<!DOCTYPE html><html lang="en"><head><title>Help for package CHNOSZ</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CHNOSZ}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CHNOSZ-package'><p>Thermodynamic Calculations and Diagrams for Geochemistry</p></a></li>
<li><a href='#add.OBIGT'><p>Functions to Work with the Thermodynamic Database</p></a></li>
<li><a href='#add.protein'><p>Amino Acid Compositions of Proteins</p></a></li>
<li><a href='#affinity'><p>Chemical Affinities of Formation Reactions</p></a></li>
<li><a href='#basis'><p>Define Basis Species</p></a></li>
<li><a href='#Berman'><p>Thermodynamic Properties of Minerals</p></a></li>
<li><a href='#buffer'><p>Calculating Buffered Chemical Activities</p></a></li>
<li><a href='#DEW'><p>Deep Earth Water (DEW) Model</p></a></li>
<li><a href='#diagram'><p>Chemical Activity Diagrams</p></a></li>
<li><a href='#EOSregress'><p>Regress Equations-of-State Parameters for Aqueous Species</p></a></li>
<li><a href='#equilibrate'><p>Equilibrium Chemical Activities of Species</p></a></li>
<li><a href='#examples'><p>Run Examples from the Documentation</p></a></li>
<li><a href='#extdata'><p>Extra Data</p></a></li>
<li><a href='#IAPWS95'><p>Properties of Water from IAPWS-95</p></a></li>
<li><a href='#info'><p>Search the Thermodynamic Database</p></a></li>
<li><a href='#ionize.aa'><p>Properties of Ionization of Proteins</p></a></li>
<li><a href='#logB.to.OBIGT'><p>Fit Thermodynamic Parameters to Formation Constants (log&thinsp;&beta;)</p></a></li>
<li><a href='#makeup'><p>Parse Chemical Formulas</p></a></li>
<li><a href='#mix'><p>Combine Diagrams for Multi-Metal Systems</p></a></li>
<li><a href='#mosaic'><p>Chemical Affinities with Changing Basis Species</p></a></li>
<li><a href='#NaCl'><p>Simple NaCl-Water Solution</p></a></li>
<li><a href='#nonideal'><p>Activity Coefficients of Aqueous Species</p></a></li>
<li><a href='#palply'><p>Conditional Parallel Processing</p></a></li>
<li><a href='#protein.info'><p>Summaries of Thermodynamic Properties of Proteins</p></a></li>
<li><a href='#rank.affinity'><p>Average Ranks of Chemical Affinities</p></a></li>
<li><a href='#retrieve'><p>Retrieve Species by Element</p></a></li>
<li><a href='#solubility'><p>Equilibrium Chemical Activities of Species</p></a></li>
<li><a href='#species'><p>Species of Interest</p></a></li>
<li><a href='#stack_mosaic'><p>Stacked mosaic diagram</p></a></li>
<li><a href='#subcrt'><p>Properties of Species and Reactions</p></a></li>
<li><a href='#swap.basis'><p>Swap Basis Species</p></a></li>
<li><a href='#taxonomy'><p>Extract Data from NCBI Taxonomy Files</p></a></li>
<li><a href='#thermo'><p>Thermodynamic Database and System Settings</p></a></li>
<li><a href='#util.array'><p>Functions to Work with Multidimensional Arrays</p></a></li>
<li><a href='#util.data'><p>Functions for Checking Thermodynamic Data</p></a></li>
<li><a href='#util.expression'><p>Functions to Express Chemical Formulas and Properties</p></a></li>
<li><a href='#util.fasta'><p>Functions for Reading FASTA Files and Downloading from UniProt</p></a></li>
<li><a href='#util.formula'><p>Functions to Work with Chemical Formulas</p></a></li>
<li><a href='#util.legend'><p>Functions to Make Legend Text</p></a></li>
<li><a href='#util.list'><p>Functions to Work with Lists</p></a></li>
<li><a href='#util.misc'><p>Functions for Miscellaneous Tasks</p></a></li>
<li><a href='#util.plot'><p>Functions to Create and Modify Plots</p></a></li>
<li><a href='#util.protein'><p>Functions for Proteins (Other Calculations)</p></a></li>
<li><a href='#util.seq'><p>Functions to Work with Sequence Data</p></a></li>
<li><a href='#util.units'><p>Functions to Convert Units</p></a></li>
<li><a href='#util.water'><p>Functions for Properties of Water and Steam</p></a></li>
<li><a href='#water'><p>Properties of Water</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Date:</td>
<td>2024-02-11</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Thermodynamic Calculations and Diagrams for Geochemistry</td>
</tr>
<tr>
<td>Author:</td>
<td>Jeffrey Dick <a href="https://orcid.org/0000-0002-0687-5890"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jeffrey Dick &lt;j3ffdick@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tinytest, knitr, rmarkdown, tufte</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, graphics, stats, utils</td>
</tr>
<tr>
<td>Description:</td>
<td>An integrated set of tools for thermodynamic calculations in
  aqueous geochemistry and geobiochemistry. Functions are provided for writing
  balanced reactions to form species from user-selected basis species and for
  calculating the standard molal properties of species and reactions, including
  the standard Gibbs energy and equilibrium constant. Calculations of the
  non-equilibrium chemical affinity and equilibrium chemical activity of species
  can be portrayed on diagrams as a function of temperature, pressure, or
  activity of basis species; in two dimensions, this gives a maximum affinity or
  predominance diagram. The diagrams have formatted chemical formulas and axis
  labels, and water stability limits can be added to Eh-pH, oxygen fugacity-
  temperature, and other diagrams with a redox variable. The package has been
  developed to handle common calculations in aqueous geochemistry, such as
  solubility due to complexation of metal ions, mineral buffers of redox or pH,
  and changing the basis species across a diagram ("mosaic diagrams"). CHNOSZ
  also implements a group additivity algorithm for the standard thermodynamic
  properties of proteins.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.chnosz.net/">https://www.chnosz.net/</a>,
<a href="https://r-forge.r-project.org/projects/chnosz/">https://r-forge.r-project.org/projects/chnosz/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-11 13:02:38 UTC; jedick</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-11 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CHNOSZ-package'>Thermodynamic Calculations and Diagrams for Geochemistry</h2><span id='topic+CHNOSZ-package'></span>

<h3>Description</h3>

<p>CHNOSZ is a package for thermodynamic calculations, primarily with applications in geochemistry and compositional biology.
It can be used to calculate the standard molal thermodynamic properties and chemical affinities of reactions relevant to geobiochemical processes, and to visualize the equilibrium activities of species on chemical speciation and predominance diagrams.
</p>


<h3>Warm Tips</h3>


<ul>
<li><p> To view the manual, run <code>help.start()</code> then select &lsquo;Packages&rsquo; and &lsquo;CHNOSZ&rsquo;. Examples in the function help pages can be run by pasting the code block into the <span class="rlang"><b>R</b></span> console.
</p>
</li>
<li><p> Also check out the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> (<em>An Introduction to CHNOSZ</em>).
</p>
</li>
<li><p> Run the command <code><a href="#topic+examples">examples</a>()</code> to run all of the examples provided in CHNOSZ. This should take about a minute.
</p>
</li></ul>



<h3>Getting Help</h3>

<p>Each help page (other than this one) has been given one of the following &ldquo;concept index entries&rdquo;:
</p>

<ul>
<li><p> Main workflow: <code><a href="#topic+info">info</a></code>, <code><a href="#topic+subcrt">subcrt</a></code>, <code><a href="#topic+basis">basis</a></code>, <code><a href="#topic+species">species</a></code>, <code><a href="#topic+affinity">affinity</a></code>, <code><a href="#topic+equilibrate">equilibrate</a></code>, <code><a href="#topic+diagram">diagram</a></code>
</p>
</li>
<li><p> Extended workflow: <code><a href="#topic+swap.basis">swap.basis</a></code>, <code><a href="#topic+buffer">buffer</a></code>, <code><a href="#topic+mosaic">mosaic</a></code>, <code><a href="#topic+EOSregress">EOSregress</a></code>
</p>
</li>
<li><p> Thermodynamic data: <code><a href="utils.html#topic+data">data</a></code>, <code><a href="#topic+extdata">extdata</a></code>, <code><a href="#topic+add.OBIGT">add.OBIGT</a></code>, <code><a href="#topic+util.data">util.data</a></code>
</p>
</li>
<li><p> Thermodynamic calculations: <code><a href="#topic+util.formula">util.formula</a></code>, <code><a href="#topic+makeup">makeup</a></code>, <code><a href="#topic+util.units">util.units</a></code>, <code><a href="#topic+Berman">Berman</a></code>, <code><a href="#topic+nonideal">nonideal</a></code>, <code><a href="#topic+util.misc">util.misc</a></code>
</p>
</li>
<li><p> Water properties: <code><a href="#topic+water">water</a></code>, <code><a href="#topic+util.water">util.water</a></code>, <code><a href="#topic+DEW">DEW</a></code>, <code><a href="#topic+IAPWS95">IAPWS95</a></code>
</p>
</li>
<li><p> Protein properties: <code><a href="#topic+protein.info">protein.info</a></code>, <code><a href="#topic+add.protein">add.protein</a></code>, <code><a href="#topic+util.fasta">util.fasta</a></code>, <code><a href="#topic+util.protein">util.protein</a></code>, <code><a href="#topic+util.seq">util.seq</a></code>, <code><a href="#topic+ionize.aa">ionize.aa</a></code>
</p>
</li>
<li><p> Other tools: <code><a href="#topic+examples">examples</a></code>, <code><a href="#topic+taxonomy">taxonomy</a></code>
</p>
</li>
<li><p> Utility functions: <code><a href="#topic+util.expression">util.expression</a></code>, <code><a href="#topic+util.plot">util.plot</a></code>, <code><a href="#topic+util.array">util.array</a></code>, <code><a href="#topic+util.list">util.list</a></code>, <code><a href="#topic+palply">palply</a></code>
</p>
</li></ul>

<p>These concept entries are visible to <code><a href="utils.html#topic+help.search">help.search</a></code> (aka <code>??</code>).
For example, help pages related to thermodynamic data can be listed using <code>??"thermodynamic data"</code>.
</p>


<h3>Warning</h3>

<p>All thermodynamic data and examples are provided on an as-is basis.
It is up to you to check not only the accuracy of the data, but also the <em>suitability of the data AND computational techniques</em> for your problem.
By combining data taken from different sources, it is possible to build an inconsistent and/or nonsensical calculation.
An attempt has been made to provide a default database (OBIGT) that is internally consistent, but no guarantee can be made.
If there is any doubt about the accuracy or suitability of data for a particular problem, please consult the primary sources (see <code><a href="#topic+thermo.refs">thermo.refs</a></code>).
</p>


<h3>Acknowledgements</h3>

<p>This package would not exist without the scientific influence and friendship of the late Professor Harold C. Helgeson.
The &lsquo;<span class="file">src/H2O92D.f</span>&rsquo; file with Fortran code for calculating the thermodynamic and electrostatic properties of H<sub>2</sub>O is modified from the <abbr><span class="acronym">SUPCRT92</span></abbr> package (Johnson et al., 1992).
</p>
<p>Work on CHNOSZ at U.C. Berkeley from ca. 2003 to 2008 was supported in part by research grants to HCH from the U.S. National Science Foundation and Department of Energy.
In 2009&ndash;2011, development of this package was partially supported by NSF grant EAR-0847616 to JMD.
</p>


<h3>References</h3>

<p>Johnson, J. W., Oelkers, E. H. and Helgeson, H. C. (1992) SUPCRT92: A software package for calculating the standard molal thermodynamic properties of minerals, gases, aqueous species, and reactions from 1 to 5000 bar and 0 to 1000&deg;C. <em>Comp. Geosci.</em> <b>18</b>, 899&ndash;947. <a href="https://doi.org/10.1016/0098-3004%2892%2990029-Q">doi:10.1016/0098-3004(92)90029-Q</a>
</p>

<hr>
<h2 id='add.OBIGT'>Functions to Work with the Thermodynamic Database</h2><span id='topic+add.OBIGT'></span><span id='topic+mod.OBIGT'></span>

<h3>Description</h3>

<p>Add or modify species in the thermodynamic database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  add.OBIGT(file, species = NULL, force = TRUE)
  mod.OBIGT(..., zap = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add.OBIGT_+3A_file">file</code></td>
<td>
<p>character, path to a file</p>
</td></tr>
<tr><td><code id="add.OBIGT_+3A_species">species</code></td>
<td>
<p>character, names of species to load from file</p>
</td></tr>
<tr><td><code id="add.OBIGT_+3A_force">force</code></td>
<td>
<p>logical, force replacement of already existing species?</p>
</td></tr>
<tr><td><code id="add.OBIGT_+3A_...">...</code></td>
<td>
<p>character or numeric, properties of species to modify in the thermodynamic database</p>
</td></tr>
<tr><td><code id="add.OBIGT_+3A_zap">zap</code></td>
<td>
<p>logical, clear preexisting parameters?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: change made to OBIGT are lost if you reload the database by calling <code><a href="#topic+reset">reset</a></code> or <code><a href="#topic+OBIGT">OBIGT</a></code> or if you quit the <span class="rlang"><b>R</b></span> session without saving it.
</p>
<p><code><a href="#topic+add.OBIGT">add.OBIGT</a></code> is used to update the thermodynamic database (<code><a href="#topic+thermo">thermo</a>$OBIGT</code>) in the running session.
The format (column names) of the specified file must be the same as the <code>extdata/OBIGT/*.csv</code> files provided with CHNOSZ.
Note that this includes both the <code>E_units</code> and <code>model</code> columns, which were added in versions 1.3.3 and 2.0.0.
</p>
<p><code>file</code> is first matched against the names of files in the <code>extdata/OBIGT</code> directory packaged with CHNOSZ.
In this case, the file suffixes are removed, so &lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;organic_aq&#8288;</span>&rsquo;, and &lsquo;<span class="samp">&#8288;organic_cr&#8288;</span>&rsquo; are valid names.
If there are no matches to a system file, then <code>file</code> is interpreted as the path a user-supplied file.
</p>
<p>If <code>species</code> is NULL (default), all species listed in the file are used.
If <code>species</code> is given and matches the name(s) of species in the file, only those species are added to the database.
</p>
<p>By default, species in the file replace any existing species having the same combination of name and state.
Set <code>force</code> to FALSE to avoid replacing species that are present in (<code>thermo()$OBIGT</code>).
</p>
<p>When adding (not replacing) species, there is no attempt made to keep the order of physical states in the database (aq-cr-liq-gas); the function simply adds new rows to the end of <code>thermo</code>$OBIGT.
As a result, retrieving the properties of an added aqueous species using <code><a href="#topic+info">info</a></code> requires an explicit <code>state="aq"</code> argument to that function if a species with the same name is present in one of the cr, liq or gas states.
</p>
<p><code>mod.OBIGT</code> changes one or more of the properties of species or adds species to the thermodynamic database.
The name of the species to add or change must be supplied as the first argument of <code>...</code> or as a named argument (named &lsquo;<span class="samp">&#8288;name&#8288;</span>&rsquo;).
Additional arguments to <code>mod.OBIGT</code> refer to the name of the property(s) to be updated and are matched to any part of compound column names in <code><a href="#topic+thermo">thermo</a>()$OBIGT</code>.
For instance, either &lsquo;<span class="samp">&#8288;z&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; matches the &lsquo;<span class="samp">&#8288;z.T&#8288;</span>&rsquo; column.
The values provided should also include order-of-magnitude scaling of HKF and DEW model parameters (see <code><a href="#topic+thermo">thermo</a></code>).
</p>
<p>When adding new species, a chemical formula should be included along with the values of any of the thermodynamic properties.
The formula is taken from the &lsquo;<span class="samp">&#8288;formula&#8288;</span>&rsquo; argument, or if that is missing, is taken to be the same as the &lsquo;<span class="samp">&#8288;name&#8288;</span>&rsquo; of the species.
An error occurs if the formula is not valid (i.e. can not be parsed by <code><a href="#topic+makeup">makeup</a></code>).
For new species, properties that are not specified become NA, except for &lsquo;<span class="samp">&#8288;state&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;E_units&#8288;</span>&rsquo;, which take default values from <code>thermo()$opt</code>.
These defaults can be overridden by giving a value for &lsquo;<span class="samp">&#8288;state&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;E_units&#8288;</span>&rsquo; in the arguments.
</p>
<p>&lsquo;<span class="samp">&#8288;model&#8288;</span>&rsquo;, if missing, is set to &lsquo;<span class="samp">&#8288;HKF&#8288;</span>&rsquo; for <code>state == "aq"</code> or &lsquo;<span class="samp">&#8288;CGL&#8288;</span>&rsquo; otherwise.
When modifying some existing minerals in OBIGT, <code>model = "CGL"</code> should be explicitly given in order to override the Berman model.
</p>
<p>When modifying species, the parameters indicated by the named arguments of <code>mod.OBIGT</code> are updated.
Use <code>zap = TRUE</code> to replace all prexisting parameters (except for <code>state</code> and <code>model</code>) with NA values.
</p>


<h3>Value</h3>

<p>The values returned (<code><a href="base.html#topic+invisible">invisible</a></code>-y) are the indices of the added and/or modified species.
</p>


<h3>References</h3>

<p>Apps, J. and Spycher, N. (2004) <em>Data qualification for thermodynamic data used to support THC calculations</em>. DOC.20041118.0004 ANL-NBS-HS-000043 REV 00. Bechtel SAIC Company, LLC.
</p>
<p>Bazarkina, E. F., Zotov, A. V., and Akinfiev, N. N. (2010)  Pressure-dependent stability of cadmium chloride complexes: Potentiometric measurements at 1-1000 bar and 25°C. <em>Geology of Ore Deposits</em> <b>52</b>, 167&ndash;178. <a href="https://doi.org/10.1134/S1075701510020054">doi:10.1134/S1075701510020054</a>
</p>
<p>Kitadai, N. (2014)  Thermodynamic prediction of glycine polymerization as a function of temperature and pH consistent with experimentally obtained results. <em>J. Mol. Evol.</em> <b>78</b>, 171&ndash;187. <a href="https://doi.org/10.1007/s00239-014-9616-1">doi:10.1007/s00239-014-9616-1</a>
</p>
<p>Shock, E. L., Helgeson, H. C. and Sverjensky, D. A. (1989) Calculation of the thermodynamic and transport properties of aqueous species at high pressures and temperatures: Standard partial molal properties of inorganic neutral species. <em>Geochim. Cosmochim. Acta</em> <b>53</b>, 2157&ndash;2183. <a href="https://doi.org/10.1016/0016-7037%2889%2990341-4">doi:10.1016/0016-7037(89)90341-4</a>
</p>
<p>Stef&aacute;nsson, A. (2001) Dissolution of primary minerals of basalt in natural waters. I. Calculation of mineral solubilities from 0&deg;C to 350&deg;C. <em>Chem. Geol.</em> <b>172</b>, 225&ndash;250. <a href="https://doi.org/10.1016/S0009-2541%2800%2900263-1">doi:10.1016/S0009-2541(00)00263-1</a>
</p>
<p>Sverjensky, D. A., Shock, E. L., and Helgeson, H. C. (1997) Prediction of the thermodynamic properties of aqueous metal complexes to 1000 °C and 5 kbar. <em>Geochim. Cosmochim. Acta</em> <b>61</b>, 1359&ndash;1412. <a href="https://doi.org/10.1016/S0016-7037%2897%2900009-4">doi:10.1016/S0016-7037(97)00009-4</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+thermo">thermo</a></code> (description of OBIGT), <code><a href="#topic+mod.buffer">mod.buffer</a></code> (modify buffer definitions), <code><a href="#topic+logB.to.OBIGT">logB.to.OBIGT</a></code> (fit thermodynamic parameters to formation constants) </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Modify an existing species (not real properties)
ialanine &lt;- mod.OBIGT("alanine", state = "cr", G = 0, H = 0, S = 0)
# We have made the values of G, H, and S inconsistent
# with the elemental composition of alanine, so the following 
# now produces a message about that
info(ialanine)

## Add an aqueous species (default) with Gibbs energy given in J/mol 
## (the same as the default) and today's date
date &lt;- as.character(Sys.Date())
iCl2O &lt;- mod.OBIGT("Cl2O", date = date, E_units = "J", G = 87738)
info(iCl2O)

## Add a solid species with a name that is different from the formula
mod.OBIGT("lorem-ipsum", formula = "C123", state = "cr", G = -12345678)
# Retrieve the data for this species using either name or formula
info(info("lorem-ipsum"))
info(info("C123"))
# Reset database
OBIGT()

## Using add.OBIGT():
# Compare stepwise stability constants of cadmium chloride complexes
# using data from Sverjensky et al., 1997 and Bazarkina et al., 2010
Cdspecies &lt;- c("Cd+2", "CdCl+", "CdCl2", "CdCl3-", "CdCl4-2")
P &lt;- c(1, seq(25, 1000, 25))
SSH97 &lt;- lapply(1:4, function(i) {
  subcrt(c(Cdspecies[i], "Cl-", Cdspecies[i+1]),
    c(-1, -1, 1), T=25, P=P)$out$logK
})
file &lt;- system.file("extdata/adds/BZA10.csv", package="CHNOSZ")
add.OBIGT(file)
BZA10 &lt;- lapply(1:4, function(i) {
  subcrt(c(Cdspecies[i], "Cl-", Cdspecies[i+1]),
    c(-1, -1, 1), T=25, P=P)$out$logK
})
matplot(P, do.call(cbind, SSH97), type="l")
matplot(P, do.call(cbind, BZA10), type="l", add=TRUE, lwd=2)
legend("topleft", legend=c("", "", "Sverjensky et al., 1997",
  "Bazarkina et al., 2010"), lwd=c(0, 0, 1, 2), bty="n")
# Make reaction labels
y &lt;- c(1.8, 0.2, -0.5, -1)
invisible(lapply(1:4, function(i) {
  text(800, y[i], describe.reaction(subcrt(c(Cdspecies[i], "Cl-",
    Cdspecies[i+1]), c(-1, -1, 1), T=25, P=1)$reaction))
}))
# Restore default database
OBIGT()

# Another use of add.OBIGT()
# Compare Delta G of AABB = UPBB + H2O
# (Figure 9 of Kitadai, 2014)
# Default database has values from Kitadai, 2014
Kit14 &lt;- subcrt(c("[AABB]", "[UPBB]", "H2O"), c(-1, 1, 1), T = seq(0, 300, 10))
# Load superseded parameters for [UPBB] from Dick et al., 2006
mod.OBIGT("[UPBB]", G = -21436, H = -45220, S = 1.62)
DLH06 &lt;- subcrt(c("[AABB]", "[UPBB]", "H2O"), c(-1, 1, 1), T = seq(0, 300, 10))
xlab &lt;- axis.label("T"); ylab &lt;- axis.label("DG", prefix="k")
plot(Kit14$out$T, Kit14$out$G/1000, type = "l", ylim = c(10, 35),
     xlab = xlab, ylab = ylab)
lines(DLH06$out$T, DLH06$out$G/1000, lty = 2)
legend("topleft", c("Dick et al., 2006", "Kitadai, 2014"), lty = c(2, 1))
title(main = "AABB = UPBB + H2O; after Figure 9 of Kitadai, 2014")
# Restore default database
OBIGT()

# Another use of add.OBIGT(): calculate Delta G of
# H4SiO4 = SiO2 + 2H2O using various sources of data for SiO2.
# First, get H4SiO4 from Stefansson, 2001
add.OBIGT("AS04", "H4SiO4")
T &lt;- seq(0, 350, 10)
s1 &lt;- subcrt(c("H4SiO4", "SiO2", "H2O"), c(-1, 1, 2), T = T)
# Now, get SiO2 from Apps and Spycher, 2004
add.OBIGT("AS04", "SiO2")
s2 &lt;- subcrt(c("H4SiO4", "SiO2", "H2O"), c(-1, 1, 2), T = T)
# Plot logK from the first and second calculations
plot(T, s1$out$G, type = "l", xlab = axis.label("T"),
  ylab = axis.label("DG"), ylim = c(-500, 2500))
lines(T, s2$out$G, lty = 2)
# Add title and legend
title(main = describe.reaction(s1$reaction))
stxt &lt;- lapply(c("H4SiO4", "SiO2", "SiO2"), expr.species)
legend("top", c("Shock et al., 1989", "Apps and Spycher, 2004"),
  title = as.expression(expr.species("SiO2")), lty = c(1, 2))
legend("topright", "Stef\u00e1nsson, 2001",
  title = as.expression(expr.species("H4SiO4")))
abline(h = 0, lty = 3, col = 8)
# Take-home message: SiO2 from Ste01 is compatible with H4SiO4 from Ste01
# at low T, but SiO2 from Shock et al., 1989 (the default in OBIGT) isn't
OBIGT()
</code></pre>

<hr>
<h2 id='add.protein'>Amino Acid Compositions of Proteins</h2><span id='topic+add.protein'></span><span id='topic+seq2aa'></span><span id='topic+aasum'></span>

<h3>Description</h3>

<p>Functions to get amino acid compositions and add them to protein list for use by other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  add.protein(aa, as.residue = FALSE)
  seq2aa(sequence, protein = NA)
  aasum(aa, abundance = 1, average = FALSE, protein = NULL, organism = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add.protein_+3A_aa">aa</code></td>
<td>
<p>data frame, amino acid composition in the format of <code>thermo()$protein</code></p>
</td></tr>
<tr><td><code id="add.protein_+3A_as.residue">as.residue</code></td>
<td>
<p>logical, normalize by protein length?</p>
</td></tr>
<tr><td><code id="add.protein_+3A_sequence">sequence</code></td>
<td>
<p>character, protein sequence</p>
</td></tr>
<tr><td><code id="add.protein_+3A_protein">protein</code></td>
<td>
<p>character, name of protein; numeric, indices of proteins (rownumbers of <code><a href="#topic+thermo">thermo</a>()$protein</code>)</p>
</td></tr>
<tr><td><code id="add.protein_+3A_abundance">abundance</code></td>
<td>
<p>numeric, abundances of proteins</p>
</td></tr>
<tr><td><code id="add.protein_+3A_average">average</code></td>
<td>
<p>logical, return the weighted average of amino acid counts?</p>
</td></tr>
<tr><td><code id="add.protein_+3A_organism">organism</code></td>
<td>
<p>character, name of organism</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A &lsquo;<span class="samp">&#8288;protein&#8288;</span>&rsquo; in CHNOSZ is defined by its identifying information and the amino acid composition, stored in <code><a href="#topic+thermo">thermo</a>$protein</code>.
The names of proteins in CHNOSZ are distinguished from those of other chemical species by having an underscore character (&quot;_&quot;) that separates two identifiers, referred to as the <code>protein</code> and <code>organism</code>.
An example is &lsquo;<span class="samp">&#8288;LYSC_CHICK&#8288;</span>&rsquo;. 
The purpose of the functions described here is to identify proteins and work with their amino acid compositions.
From the amino acid compositions, the thermodynamic properties of the proteins can be estimated by group additivity.
</p>
<p><code>seq2aa</code> returns a data frame of amino acid composition for the provided <code>sequence</code>, in the format of <code>thermo()$protein</code>.
In this function, the value of the <code>protein</code> argument is put into the <code>protein</code> column of the result.
If there is an underscore (e.g. &lsquo;<span class="samp">&#8288;LYSC_CHICK&#8288;</span>&rsquo;), it is used to split the text, and the two parts are put into the <code>protein</code> and <code>organism</code> columns.
</p>
<p>Given amino acid compositions returned by <code>seq2aa</code>, <code>add.protein</code> adds them to <code>thermo()$protein</code> for use by other functions in CHNOSZ.
The amino acid compositions of proteins in <code>aa</code> with the same name as one in <code>thermo()$protein</code> are replaced.
Set <code>as.residue</code> to TRUE to normalize by protein length; each input amino acid composition is divided by the corresponding number of residues, with the result that the sum of amino acid frequencies for each protein is 1.
</p>
<p><code>aasum</code> returns a data frame representing the sum of amino acid compositions in the rows of the input <code>aa</code> data frame.
The amino acid compositions are multiplied by the indicated <code>abundance</code>; that argument is recycled to match the number of rows of <code>aa</code>.
If <code>average</code> is TRUE the final sum is divided by the number of input compositions.
The name used in the output is taken from the first row of <code>aa</code> or from <code>protein</code> and <code>organism</code> if they are specified.
</p>


<h3>Value</h3>

<p>For <code>seq2aa</code>, a data frame of amino acid composition and identifying information for proteins.
For <code>add.protein</code>, the rownumbers of <code>thermo()$protein</code> that are added and/or replaced.
For <code>aasum</code>, a one-row data frame of amino acid composition and identifying information.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code> for another way of getting amino acid compositions that can be used with <code>add.protein</code>.
</p>
<p><code><a href="#topic+pinfo">pinfo</a></code> for protein-level functions (length, chemical formulas, reaction coefficients of basis species).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Get the amino acid composition of a protein sequence
# (Human Gastric juice peptide 1)
aa &lt;- seq2aa("LAAGKVEDSD", "GAJU_HUMAN")
# Add the protein to CHNOSZ
ip &lt;- add.protein(aa)
# Calculate the protein length and chemical formula
protein.length(ip) # 10
as.chemical.formula(protein.formula(ip)) # "C41H69N11O18"

# Calculate a formula without using add.protein
aa &lt;- seq2aa("ANLSG", "pentapeptide_test")
as.chemical.formula(protein.formula(aa))

# Sum the amino acid compositions of several poliovirus protein subunits
file &lt;- system.file("extdata/protein/POLG.csv", package = "CHNOSZ")
aa &lt;- read.csv(file, as.is = TRUE)
aasum(aa, protein = "POLG_sum")
</code></pre>

<hr>
<h2 id='affinity'>Chemical Affinities of Formation Reactions</h2><span id='topic+affinity'></span>

<h3>Description</h3>

<p>Calculate the chemical affinities of formation reactions of species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  affinity(..., property = NULL, sout = NULL, exceed.Ttr = FALSE,
    exceed.rhomin = FALSE, return.buffer = FALSE, return.sout = FALSE,
    balance = "PBB", iprotein = NULL, loga.protein = 0, transect = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="affinity_+3A_...">...</code></td>
<td>
<p>numeric, zero or more named arguments, used to identify the variables of interest in the calculations. For argument recall, pass the output from a previous calculation of <code>affinity</code> as an unnamed first argument.</p>
</td></tr>
<tr><td><code id="affinity_+3A_property">property</code></td>
<td>
<p>character, the property to be calculated. Default is &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo;, for chemical affinity of formation reactions of species of interest</p>
</td></tr>
<tr><td><code id="affinity_+3A_sout">sout</code></td>
<td>
<p>list, output from <code><a href="#topic+subcrt">subcrt</a></code></p>
</td></tr>
<tr><td><code id="affinity_+3A_exceed.ttr">exceed.Ttr</code></td>
<td>
<p>logical, allow <code><a href="#topic+subcrt">subcrt</a></code> to compute properties for phases beyond their transition temperature?</p>
</td></tr>
<tr><td><code id="affinity_+3A_exceed.rhomin">exceed.rhomin</code></td>
<td>
<p>logical, allow <code><a href="#topic+subcrt">subcrt</a></code> to compute properties of species in the HKF model below 0.35 g cm<sup>-3</sup>?</p>
</td></tr>
<tr><td><code id="affinity_+3A_return.buffer">return.buffer</code></td>
<td>
<p>logical. If <code>TRUE</code>, and a <code><a href="#topic+buffer">buffer</a></code> has been associated with one or more basis species in the system, return the values of the activities of the basis species calculated using the buffer. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="affinity_+3A_return.sout">return.sout</code></td>
<td>
<p>logical, return only the values calculated with <code><a href="#topic+subcrt">subcrt</a></code>?</p>
</td></tr>
<tr><td><code id="affinity_+3A_balance">balance</code></td>
<td>
<p>character. This argument is used to identify a conserved basis species (or &lsquo;<span class="samp">&#8288;PBB&#8288;</span>&rsquo;) in a chemical activity buffer. Default is &lsquo;<span class="samp">&#8288;PBB&#8288;</span>&rsquo;.</p>
</td></tr>
<tr><td><code id="affinity_+3A_iprotein">iprotein</code></td>
<td>
<p>numeric, indices of proteins in <code><a href="#topic+thermo">thermo</a>$protein</code> for which to calculate properties</p>
</td></tr>
<tr><td><code id="affinity_+3A_loga.protein">loga.protein</code></td>
<td>
<p>numeric, logarithms of activities of proteins identified in <code>iprotein</code></p>
</td></tr>
<tr><td><code id="affinity_+3A_transect">transect</code></td>
<td>
<p>logical, force a transect calculation, even for three or fewer values of the variables?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>affinity</code> calculates the chemical affinities of reactions to form the species of interest from the basis species.
The equation used to calculate chemical affinity (<em>A</em>), written for base-10 (decimal) logarithms, is <code class="reqn">A/(2.303RT)</code>=<code class="reqn">\log (K/Q)</code>, where <code class="reqn">K</code> is the equilibrium constant of the reaction, <code class="reqn">Q</code> is the activity product of the species in the reaction, and 2.303 is the conversion factor from natural to decimal logarithms.
The calculation of chemical affinities relies on the current definitions of the <code><a href="#topic+basis">basis</a></code> species and <code><a href="#topic+species">species</a></code> of interest.
Calculations are possible at single values of temperature, pressure, ionic strength and chemical activities of the basis species, or as a function of one or more of these variables. 
</p>
<p>The argument <code>property</code> can be changed to calculate other thermodynamic properties of formation reactions. 
Valid properties are &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo; or NULL for chemical affinity, &lsquo;<span class="samp">&#8288;logK&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;logQ&#8288;</span>&rsquo; for logarithm of equilibrium constant and reaction activity product, or any of the properties available in <code><a href="#topic+subcrt">subcrt</a></code> except for &lsquo;<span class="samp">&#8288;rho&#8288;</span>&rsquo;.
The properties returned are those of the formation reactions of the species of interest from the basis species.
It is also possible to calculate the properties of the species of interest themselves (not their formation reactions) by setting the <code>property</code> to &lsquo;<span class="samp">&#8288;G.species&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Cp.species&#8288;</span>&rsquo;, etc.
Except for &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo;, the properties of proteins or their reactions calculated in this manner are restricted to nonionized proteins.
</p>
<p>Zero, one, or more leading arguments to the function identify which of the chemical activities of basis species, temperature, pressure and/or ionic strength to vary.
The names of each of these arguments may be the formula of any of the basis species of the system, or &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pH&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo; (but names may not be repeated).
The names of charged basis species such as &lsquo;<span class="samp">&#8288;K+&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;SO4-2&#8288;</span>&rsquo; should be quoted when used as arguments.
The value of each argument is of the form <code>c(min, max)</code> or <code>c(min, max, res)</code> where <code>min</code> and <code>max</code> refer to the minimimum and maximum values of variable identified by the name of the argument, and <code>res</code> is the resolution, or number of points along which to do the calculations; <code>res</code> is assigned a default value of 256 if it is missing.
For any arguments that refer to basis species, the numerical values are the logarithms of activity (or fugacity for gases) of that basis species.
</p>
<p>If &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo;, and/or &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo; are not among the <code>var</code>s, their constant values can be supplied in <code>T</code>, <code>P</code>, or <code>IS</code> (in mol kg<code class="reqn">^{-1}</code>).
The units of &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo; are those set by <code><a href="#topic+T.units">T.units</a></code> and <code><a href="#topic+P.units">P.units</a></code> (on program start-up these are &deg;C and bar, respectively). 
<code>sout</code>, if provided, replaces the call to <code><a href="#topic+subcrt">subcrt</a></code>, which can greatly speed up the calculations if this intermediate result is stored by other functions.
<code>exceed.Ttr</code> is passed to <code><a href="#topic+subcrt">subcrt</a></code> so that the properties of mineral phases beyond their transition temperatures can optionally be calculated.
</p>
<p>If one or more buffers are assigned to the definition of <code><a href="#topic+basis">basis</a></code> species, the logarithms of activities of these basis species are taken from the buffer (see <code><a href="#topic+buffer">buffer</a></code>).
</p>
<p>The <code>iprotein</code> and <code>loga.protein</code> arguments can be used to compute the chemical affinities of formation reactions of proteins that are not in the current <code><a href="#topic+species">species</a></code> definition.
<code>iprotein</code> contains the indices (rownumbers) of desired proteins in <code><a href="#topic+thermo">thermo</a>$protein</code>.
This uses some optimizations to calculate the properties of many proteins in a fraction of the time it would take to calculate them individually.
</p>
<p>When the length(s) of the variables is(are) greater than 3, the function enters the &lsquo;<span class="samp">&#8288;transect&#8288;</span>&rsquo; mode of operation.
In this mode of operation, instead of performing the calculations on an <code class="reqn">n</code>-dimensional grid, the affinities are calculated on a transect of changing T, P, and/or chemical activity of basis species.
</p>
<p>Argument recall is invoked by passing a previous result of <code>affinity</code> as the first argument.
The function then calls itself using the settings from the previous calculation, with additions or modifications indicated by the remaining arguments in the current function call.
</p>


<h3>Value</h3>

<p>A list, elements of which are <code>fun</code> the name of the function (&lsquo;<span class="samp">&#8288;affinity&#8288;</span>&rsquo;), <code>args</code> all of the arguments except for &lsquo;<span class="samp">&#8288;sout&#8288;</span>&rsquo; (these are used for argument recall), <code>sout</code> output from <code><a href="#topic+subcrt">subcrt</a></code>, <code>property</code> name of the calculated property (&lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo; for chemical affinity), <code>basis</code> and <code>species</code> definition of basis species and species of interest in effect at runtime, <code>T</code> and <code>P</code> temperature and pressure, in the system units of Kelvin and bar, set to <code>numeric()</code> (length=0) if either one is a variable, <code>vars</code> the names of the variables, <code>vals</code> the values of the variables (a list, one element for each variable), <code>values</code> the result of the calculation (a list, one element for each species, with names taken from the species index in <code><a href="#topic+thermo">thermo</a>$OBIGT</code>).
The elements of the lists in <code>vals</code> and <code>values</code> are arrays of <code class="reqn">n</code> dimensions, where <code class="reqn">n</code> is the number of variables.
The values of chemical affinity of formation reactions of the species are returned in dimensionless units (for use with decimal logarithms, i.e., <em>A</em>/<code class="reqn">2.303RT</code>).
</p>
<p>Names other than &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo; in <code>vars</code> generally refer to basis species, and the corresponding <code>vals</code> are the logarithms of activity or fugacity.
However, if one or more of <code>pe</code>, <code>Eh</code> or <code>pH</code> is among the variables of interest, <code>vals</code> holds the values of the those variables as indicated.
</p>


<h3>References</h3>

<p>Helgeson, H. C., Richard, L, McKenzie, W. F., Norton, D. L. and Schmitt, A. (2009) A chemical and thermodynamic model of oil generation in hydrocarbon source rocks. <em>Geochim. Cosmochim. Acta</em> <b>73</b>, 594&ndash;695. <a href="https://doi.org/10.1016/j.gca.2008.03.004">doi:10.1016/j.gca.2008.03.004</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ionize.aa">ionize.aa</a></code>, activated if proteins are among the species of interest, &lsquo;<span class="samp">&#8288;H+&#8288;</span>&rsquo; is in the basis and <code>thermo()$opt$ionize.aa</code> is TRUE.
<code><a href="#topic+equilibrate">equilibrate</a></code> for using the results of <code>affinity</code> to calculate equilibrium activities of species, and <code><a href="#topic+diagram">diagram</a></code> to plot the results.
<code><a href="utils.html#topic+demo">demo</a>("saturation")</code> for an example using the argument recall feature.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Set up a system and calculate
## chemical affinities of formation reactions
basis(c("SiO2", "MgO", "H2O", "O2"), c(-5, -5, 0, 999))
species(c("quartz","enstatite","forsterite"))
# Chemical affinities (A/2.303RT) at 25 deg C and 1 bar
affinity()
# At higher temperature and pressure
affinity(T = 500, P = 2000)
# At 25 temperatures and pressures,
# some are in the low-density region so we suppress warnings
suppressWarnings(affinity(T = c(500, 1000, 5), P = c(1000, 5000, 5)))
# Equilibrium constants of formation reactions
affinity(property = "logK")
# Standard molal Gibbs energies of species,
# in units set by E.units() (default: J/mol)
affinity(property = "G.species")
# Standard molal Gibbs energies of reactions
affinity(property = "G")
# A T,P-transect
# (fluid pressure from Helgeson et al., 2009 Fig. 7)
affinity(T = c(25, 110, 115, 215), P = c(11, 335, 500, 1450))
</code></pre>

<hr>
<h2 id='basis'>Define Basis Species</h2><span id='topic+basis'></span>

<h3>Description</h3>

<p>Define the basis species of a chemical system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  basis(species = NULL, state = NULL, logact = NULL,
    delete = FALSE, add = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="basis_+3A_species">species</code></td>
<td>
<p>character, names or formulas of species, or numeric, indices of species</p>
</td></tr>
<tr><td><code id="basis_+3A_state">state</code></td>
<td>
<p>character, physical states or names of buffers</p>
</td></tr>
<tr><td><code id="basis_+3A_logact">logact</code></td>
<td>
<p>numeric, logarithms of activities or fugacities</p>
</td></tr>
<tr><td><code id="basis_+3A_delete">delete</code></td>
<td>
<p>logical, delete the current basis definition?</p>
</td></tr>
<tr><td><code id="basis_+3A_add">add</code></td>
<td>
<p>logical, add species to the current basis definition?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The basis species represent the possible range of chemical compositions for all the species of interest.
As used here, a set of basis species is valid only if it satisifes two conditions: 1) the number of basis species is the same as the number of chemical elements (including charge) in those species and 2) the square matrix representing the elemental stoichiometries of the basis species has a real inverse.
</p>
<p>To create a basis definition, call <code>basis</code> with the names or formulas of the basis species in the <code>species</code> argument, or all numeric values as species indices (rownumbers in <code><a href="#topic+thermo">thermo</a>()$OBIGT</code>).
The special names &lsquo;<span class="samp">&#8288;pH&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo; can also be used; they get translated into the names of the proton (&lsquo;<span class="samp">&#8288;H+&#8288;</span>&rsquo;) and electron (&lsquo;<span class="samp">&#8288;e-&#8288;</span>&rsquo;) as appropriate.
If desired, include the <code>state</code> for the named species and the logarithms of activity (fugacity for gases) in <code>logact</code>.
The latter defaults to zero (unit activity) if not specified.
</p>
<p>To modify an existing basis definition, the physical states or logarithms of activities of species can be changed by calling <code>basis</code> with a <code>species</code> argument that has the formulas (not names) or indices of species in the existing basis.
If either of the second or third arguments to <code>basis</code> is of type character, it refers to the physical state (if present in <code>thermo()$OBIGT$state</code>) or a chemical activity <code><a href="#topic+buffer">buffer</a></code> (if present in <code>thermo()$buffers$name</code>).
If either of these arguments is numeric it specifies the logarithms of activities (or fugacities for gases) of the basis species.
In case &lsquo;<span class="samp">&#8288;pH&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo; is named, the logarithm of activity of the basis species is converted from these values.
For example, a value of 7 for pH is stored as a logarithm of activity of -7.
</p>
<p>If <code>add</code> is TRUE, then the function attempts to <em>add</em> the indicated <code>species</code> to the basis definition.
This only works if the enlarged set of species is a valid basis set as described above.
If the formed <code><a href="#topic+species">species</a></code> are currently defined, their formation reactions are modified accordingly (with zeroes for the newly added basis species).
</p>
<p>If <code>add</code> is FALSE, and if <code>basis</code> is called with NULL values of both <code>state</code> and <code>logact</code>, the new set of species, if they are a valid basis set, completely replaces any existing basis definition.
When this occurs, any existing species definition (created by the <code>species</code> function) is deleted.
Call <code>basis</code> with <code>delete</code> set to TRUE or <code>species</code> set to &lsquo;<span class="samp">&#8288;""&#8288;</span>&rsquo; to clear the basis definition and that of the <code><a href="#topic+species">species</a></code>, if present.
</p>
<p>If the value of <code>basis</code> is one of the keywords in the following table, the corresponding set of basis species is loaded, and their activities are given preset values.
The basis species identified by these keywords are aqueous except for H<sub>2</sub>O (liq), O<sub>2</sub> (gas) and Fe<sub>2</sub>O<sub>3</sub> (hematite).
</p>

<table>
<tr>
 <td style="text-align: left;">
    <code>CHNOS</code> </td><td style="text-align: left;"> CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>2</sub>S, O<sub>2</sub> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>CHNOS+</code> </td><td style="text-align: left;"> CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>2</sub>S, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>CHNOSe</code> </td><td style="text-align: left;"> CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>2</sub>S, <i>e</i><sup>-</sup>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>CHNOPS+</code> </td><td style="text-align: left;"> CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>3</sub>PO<sub>4</sub>, H<sub>2</sub>S, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>CHNOPSe</code> </td><td style="text-align: left;"> CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>3</sub>PO<sub>4</sub>, H<sub>2</sub>S, <i>e</i><sup>-</sup>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>MgCHNOPS+</code> </td><td style="text-align: left;"> Mg<sup>+2</sup>, CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>3</sub>PO<sub>4</sub>, H<sub>2</sub>S, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>MgCHNOPSe</code> </td><td style="text-align: left;"> Mg<sup>+2</sup>, CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>3</sub>PO<sub>4</sub>, H<sub>2</sub>S, <i>e</i><sup>-</sup>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>FeCHNOS</code> </td><td style="text-align: left;"> Fe<sub>2</sub>O<sub>3</sub>, CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>2</sub>S, O<sub>2</sub> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>FeCHNOS+</code> </td><td style="text-align: left;"> Fe<sub>2</sub>O<sub>3</sub>, CO<sub>2</sub>, H<sub>2</sub>O, NH<sub>3</sub>, H<sub>2</sub>S, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>QEC4</code> </td><td style="text-align: left;"> cysteine, glutamic acid, glutamine, H<sub>2</sub>O, O<sub>2</sub> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>QEC</code> </td><td style="text-align: left;"> cysteine, glutamic acid, glutamine, H<sub>2</sub>O, O<sub>2</sub> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>QEC+</code> </td><td style="text-align: left;"> cysteine, glutamic acid, glutamine, H<sub>2</sub>O, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>QCa</code> </td><td style="text-align: left;"> glutamine, cysteine, acetic acid, H<sub>2</sub>O, O<sub>2</sub> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>QCa+</code> </td><td style="text-align: left;"> glutamine, cysteine, acetic acid, H<sub>2</sub>O, O<sub>2</sub>, H<sup>+</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>The logarithms of activities of amino acids in the &lsquo;<span class="samp">&#8288;QEC4&#8288;</span>&rsquo; basis are -4 (i.e., basis II in Dick, 2016); those in &lsquo;<span class="samp">&#8288;QEC&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;QEC+&#8288;</span>&rsquo; are set to approximate concentrations in human plasma (see Dick, 2017).
</p>


<h3>Value</h3>

<p>Returns the value of <code>thermo()$basis</code> after any modifications; or, if <code>delete</code> is TRUE, its value before deletion (invisibly).
</p>


<h3>References</h3>

<p>Dick, J. M. (2016) Proteomic indicators of oxidation and hydration state in colorectal cancer. <em>PeerJ</em> <b>4</b>:e2238. <a href="https://doi.org/10.7717/peerj.2238">doi:10.7717/peerj.2238</a>
</p>
<p>Dick, J. M. (2017) Chemical composition and the potential for proteomic transformation in cancer, hypoxia, and hyperosmotic stress. <em>PeerJ</em> <b>5</b>:e3421 <a href="https://doi.org/10.7717/peerj.3421">doi:10.7717/peerj.3421</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+info">info</a></code> to query the thermodynamic database in order to find what species are available. 
<code><a href="#topic+makeup">makeup</a></code> is used by <code>basis</code> to generate the stoichiometric matrix from chemical formulas.
<code><a href="#topic+swap.basis">swap.basis</a></code> is used to change the chemical compounds (species formulas) used in the basis definition while keeping the chemical potentials of the elements unaltered.
<code><a href="#topic+species">species</a></code> for setting up the formation reactions from basis species. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Define basis species
# with one, two or three elements
basis("O2")
basis(c("H2O", "O2"))
basis(c("H2O", "O2", "H+"))
## Clear the basis species
basis("")

## Not run: 
## Marked dontrun because they produce errors
# Fewer species than elements
basis(c("H2O", "H+"))
# More species than elements
basis(c("H2O", "O2", "H2", "H+"))
# Non-independent species
basis(c("CO2", "H2O", "HCl", "Cl-", "H+"))
## End(Not run)

## Specify activities and states
basis(c("H2O", "O2", "CO2"), c(-2, -78, -3), c("liq", "aq", "aq"))
# Change logarithms of activities/fugacities	
basis(c("H2O", "O2"), c(0, -80))	
# Change state of CO2
basis("CO2", "gas")
</code></pre>

<hr>
<h2 id='Berman'>Thermodynamic Properties of Minerals</h2><span id='topic+Berman'></span>

<h3>Description</h3>

<p>Calculate thermodynamic properties of minerals using the equations of Berman (1988).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  Berman(name, T = 298.15, P = 1, check.G = FALSE,
         calc.transition = TRUE, calc.disorder = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Berman_+3A_name">name</code></td>
<td>
<p>character, name of mineral</p>
</td></tr>
<tr><td><code id="Berman_+3A_t">T</code></td>
<td>
<p>numeric, temperature(s) at which to calculate properties (K)</p>
</td></tr>
<tr><td><code id="Berman_+3A_p">P</code></td>
<td>
<p>numeric, pressure(s) at which to calculate properties (bar)</p>
</td></tr>
<tr><td><code id="Berman_+3A_check.g">check.G</code></td>
<td>
<p>logical, check consistency of G, H, and S?</p>
</td></tr>
<tr><td><code id="Berman_+3A_calc.transition">calc.transition</code></td>
<td>
<p>logical, include calculation of polymorphic transition properties?</p>
</td></tr>
<tr><td><code id="Berman_+3A_calc.disorder">calc.disorder</code></td>
<td>
<p>logical, include calculation of disordering properties?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the thermodynamic properties of minerals at high <I>P</I> and <I>T</I> using equations given by Berman (1988).
These minerals should be listed in <code>thermo()$OBIGT</code> with the state &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; and chemical formula, and optionally an abbreviation and references, but all other properties set to NA.
</p>
<p>The standard state thermodynamic properties and parameters for the calculations are stored in data files under <code>extdata/Berman</code>, or can be read from a user-created file specified by <code><a href="#topic+thermo">thermo</a>()$opt$Berman</code>.
</p>
<p>The equation used for heat capacity is <I>C<sub>P</sub></I> = k0 + k1*<I>T</I><sup>-0.5</sup> + k2*<I>T</I><sup>-2</sup> + k3*<I>T</I><sup>-3</sup> + k4*<I>T</I><sup>-1</sup> + k5*<I>T</I> + k6*<I>T</I><sup>2</sup>.
This is an extended form Eq. 4 of Berman (1988) as used in the winTWQ program (Berman, 2007).
The equation used for volume is <I>V</I>(<I>P</I>, <I>T</I>) / <I>V</I>(1 bar, 298.15 K) =  1 + v1 * (<I>T</I> - 298.15) + v2 * (<I>T</I> - 298.15)<sup>2</sup> + v3 * (<I>P</I> - 1) + v4 * (<I>P</I> - 1)<sup>2</sup>  (Berman, 1988, Eq. 5, with terms reordered to follow winTWQ format).
The equations used for lambda transitions follow Eqs. 8-14 of Berman (1988).
The equation used for the disorder contribution between Tmin and Tmax is <I>C<sub>P</sub></I>[dis] = d0 + d1*<I>T</I><sup>-0.5</sup> + d2*<I>T</I><sup>-2</sup> + d3*<I>T</I> + d4*<I>T</I><sup>2</sup> (Berman, 1988, Eq. 15).
The parameters correspond to Tables 2 (<code>GfPrTr</code>, <code>HfPrTr</code>, <code>SPrTr</code>, <code>VPrTr</code>), 3a (<code>k0</code> to <code>k3</code>), 4 (<code>v1</code> to <code>v4</code>), 3b (transition parameters: <code>Tlambda</code> to <code>dTH</code>), and 5 (disorder parameters: <code>Tmax</code>, <code>Tmin</code>, <code>d1</code> to <code>d4</code> and <code>Vad</code>) of Berman (1988).
Following the winTWQ data format, multipliers are applied to the volume parameters only (see below).
Note that <code>VPrTr</code> is tabulated in J bar<sup>-1</sup> mol<sup>-1</sup>, which is equal to 10 cm<sup>3</sup> mol<sup>-1</sup>.
</p>
<p>A value for <code>GfPrTr</code> is not required and is only used for optional checks (see below).
Numeric values (possibly 0) should be assigned for all of <code>HfPrTr</code>, <code>SPrTr</code>, <code>VPrTr</code>, <code>k0</code> to <code>k6</code> and <code>v1</code> to <code>v4</code>.
Missing (or NA) values are permitted for the transition and disorder parameters, for minerals where they are not used.
The data files have the following 30 columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>name</code> </td><td style="text-align: left;"> mineral name (must match an entry with a formula but NA properties in <code>thermo()$OBIGT</code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>GfPrTr</code> </td><td style="text-align: left;"> standard Gibbs energy at 298.15 K and 1 bar (J mol<sup>-1</sup>) (Benson-Helgeson convention)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>HfPrTr</code> </td><td style="text-align: left;"> standard enthalpy at 298.15 K and 1 bar (J mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>SPrTr</code> </td><td style="text-align: left;"> standard entropy at 298.15 K and 1 bar (J mol<sup>-1</sup> K<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>VPrTr</code> </td><td style="text-align: left;"> standard volume at 298.15 K and 1 bar (J bar<sup>-1</sup>) [1 J bar<sup>-1</sup> = 10 cm<sup>3</sup>]</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>k0</code> ... <code>k6</code> </td><td style="text-align: left;"> k0 (J mol<sup>-1</sup> K<sup>-1</sup>) to k6 </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>v1</code> </td><td style="text-align: left;"> v1 (K<sup>-1</sup>) * 10<sup>5</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>v2</code> </td><td style="text-align: left;"> v2 (K<sup>-2</sup>) * 10<sup>5</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>v3</code> </td><td style="text-align: left;"> v3 (bar<sup>-1</sup>) * 10<sup>5</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>v4</code> </td><td style="text-align: left;"> v4 (bar<sup>-2</sup>) * 10<sup>8</sup> </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Tlambda</code> </td><td style="text-align: left;"> <I>T</I><sub>&lambda;</sub> (K) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Tref</code> </td><td style="text-align: left;"> <I>T<sub>ref</sub></I> (K) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>dTdP</code> </td><td style="text-align: left;"> d<I>T</I> / d<I>P</I> (K bar<sup>-1</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>l1</code> </td><td style="text-align: left;"> l1 ((J/mol)<sup>0.5</sup> K<sup>-1</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>l2</code> </td><td style="text-align: left;"> l2 ((J/mol)<sup>0.5</sup> K<sup>-2</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>DtH</code> </td><td style="text-align: left;"> &Delta;<I>T<sub>t</sub></I><I>H</I> (J mol<sup>-1</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Tmax</code> </td><td style="text-align: left;"> temperature at which phase is fully disordered (<I>T<sub>D</sub></I> in Berman, 1988) (K) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Tmin</code> </td><td style="text-align: left;"> reference temperature for onset of disordering (<I>t</I> in Berman, 1988) (K) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>d0</code> ... <code>d4</code> </td><td style="text-align: left;"> d0 (J mol<sup>-1</sup> K<sup>-1</sup>) to d4 </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Vad</code> </td><td style="text-align: left;"> constant that scales the disordering enthalpy to volume of disorder (<I>d</I><sub>5</sub> in Berman, 1988) </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

<p>The function outputs apparent Gibbs energies according to the Benson-Helgeson convention (&Delta;<I>G</I> = &Delta;<I>H</I> - <I>T</I>&Delta;<I>S</I>) using the entropies of the elements in the chemical formula of the mineral to calculate &Delta;<I>S</I> (cf. Anderson, 2005).
If <code>check.G</code> is TRUE, the tabulated value of <code>GfTrPr</code> (Benson-Helgeson) is compared with that calculated from <code>HfPrTr</code> - 298.15*<code>DSPrTr</code> (<code>DSPrTr</code> is the difference between the entropies of the elements in the formula and <code>SPrTr</code> in the table).
A warning is produced if the absolute value of the difference between tabulated and calculated <code>GfTrPr</code> is greater than 1000 J/mol.
</p>
<p>If the function is called with missing <code>name</code>, the parameters for all available minerals are returned.
</p>


<h3>Value</h3>

<p>A data frame with <code>T</code> (K), <code>P</code> (bar), <code>G</code>, <code>H</code>, <code>S</code>, and <code>Cp</code> (energetic units in Joules), and <code>V</code> (cm<sup>3</sup> mol<sup>-1</sup>).
</p>


<h3>References</h3>

<p>Anderson, G. M. (2005) <em>Thermodynamics of Natural Systems</em>, 2nd ed., Cambridge University Press, 648 p. <a href="https://www.worldcat.org/oclc/474880901">https://www.worldcat.org/oclc/474880901</a>
</p>
<p>Berman, R. G. (1988) Internally-consistent thermodynamic data for minerals in the system Na<sub>2</sub>O-K<sub>2</sub>O-CaO-MgO-FeO-Fe<sub>2</sub>O<sub>3</sub>-Al<sub>2</sub>O<sub>3</sub>-SiO<sub>2</sub>-TiO<sub>2</sub>-H<sub>2</sub>O-CO<sub>2</sub>. <em>J. Petrol.</em> <b>29</b>, 445-522. <a href="https://doi.org/10.1093/petrology/29.2.445">doi:10.1093/petrology/29.2.445</a>
</p>
<p>Berman, R. G. and Aranovich, L. Ya. (1996) Optimized standard state and solution properties of minerals. I. Model calibration for olivine, orthopyroxene, cordierite, garnet, and ilmenite in the system FeO-MgO-CaO-Al<sub>2</sub>O<sub>3</sub>-TiO<sub>2</sub>-SiO<sub>2</sub>. <em>Contrib. Mineral. Petrol.</em> <b>126</b>, 1-24. <a href="https://doi.org/10.1007/s004100050233">doi:10.1007/s004100050233</a>
</p>
<p>Berman, R. G. (2007) winTWQ (version 2.3): A software package for performing internally-consistent thermobarometric calculations. <em>Open File</em> <b>5462</b>, Geological Survey of Canada, 41 p. <a href="https://doi.org/10.4095/223425">doi:10.4095/223425</a>
</p>
<p>Helgeson, H. C., Delany, J. M., Nesbitt, H. W. and Bird, D. K. (1978) Summary and critique of the thermodynamic properties of rock-forming minerals. <em>Am. J. Sci.</em> <b>278-A</b>, 1&ndash;229. <a href="https://www.worldcat.org/oclc/13594862">https://www.worldcat.org/oclc/13594862</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Other than the formula, the parameters aren't stored in
# thermo()$OBIGT, so this shows NAs
info(info("quartz", "cr"))
# Properties of alpha-quartz (aQz) at 298.15 K and 1 bar
Berman("quartz")
# Gibbs energies of aQz and coesite at higher T and P
T &lt;- seq(200, 1300, 100)
P &lt;- seq(22870, 31900, length.out = length(T))
G_aQz &lt;- Berman("quartz", T = T, P = P)$G
G_Cs &lt;- Berman("coesite", T = T, P = P)$G
# That is close to the univariant curve (Ber88 Fig. 4),
# so the difference in G is close to 0
DGrxn &lt;- G_Cs - G_aQz
all(abs(DGrxn) &lt; 100)  # TRUE

# Make a P-T diagram for SiO2 minerals (Ber88 Fig. 4)
basis(c("SiO2", "O2"), c("cr", "gas"))
species(c("quartz", "quartz,beta", "coesite"), "cr")
a &lt;- affinity(T = c(200, 1700, 200), P = c(0, 50000, 200))
diagram(a)

## Getting data from a user-supplied file
## Ol-Opx exchange equilibrium, after Berman and Aranovich, 1996
species &lt;- c("fayalite", "enstatite", "ferrosilite", "forsterite")
coeffs &lt;- c(-1, -2, 2, 1)
T &lt;- seq(600, 1500, 50)
Gex_Ber88 &lt;- subcrt(species, coeffs, T = T, P = 1)$out$G
# Add data from BA96
datadir &lt;- system.file("extdata/Berman/testing", package = "CHNOSZ")
add.OBIGT(file.path(datadir, "BA96_OBIGT.csv"))
thermo("opt$Berman" = file.path(datadir, "BA96_Berman.csv"))
Gex_BA96 &lt;- subcrt(species, coeffs, T = seq(600, 1500, 50), P = 1)$out$G
# Ber88 is lower than BA96 at low T
(Gex_BA96 - Gex_Ber88)[1] &gt; 0  # TRUE
# The curves cross at about 725 deg C (BA96 Fig. 8)
# (actually, in our calculation they cross closer to 800 deg C)
T[which.min(abs(Gex_BA96 - Gex_Ber88))]  # 800
# Reset the database (thermo()$OBIGT, and thermo()$opt$Berman)
reset()
</code></pre>

<hr>
<h2 id='buffer'>Calculating Buffered Chemical Activities</h2><span id='topic+buffer'></span><span id='topic+mod.buffer'></span>

<h3>Description</h3>

<p>Calculate values of activity or fugacity of basis species buffered by an assemblage of one or more species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mod.buffer(name, species = NULL, state = "cr", logact = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buffer_+3A_name">name</code></td>
<td>
<p>character, name of buffer to add to or find in <code>thermo()$buffer</code>.</p>
</td></tr>
<tr><td><code id="buffer_+3A_species">species</code></td>
<td>
<p>character, names or formulas of species in a buffer.</p>
</td></tr>
<tr><td><code id="buffer_+3A_state">state</code></td>
<td>
<p>character, physical states of species in buffer.</p>
</td></tr>
<tr><td><code id="buffer_+3A_logact">logact</code></td>
<td>
<p>numeric, logarithms of activities of species in buffer.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A buffer is treated here as assemblage of one or more species whose presence constrains values of the chemical activity (or fugacity) of one or more basis species.
To perform calculations for buffers use <code><a href="#topic+basis">basis</a></code> to associate the name of the buffer with one or more basis species.
After this, calls to <code>affinity</code> will invoke the required calculations.
The calculated values of the buffered activites can be retrieved by setting <code>return.buffer</code> to <code>TRUE</code> (in <code><a href="#topic+affinity">affinity</a></code>).
The maximum number of buffered chemical activities possible for any buffer is equal to the number of species in the buffer; however, the user may then elect to work with the values for only one or some of the basis species calculated with the buffer. 
</p>
<p>The identification of a conserved basis species (or other reaction balancing rule) is required in calculations for buffers of more than one species.
For example, in the pyrite-pyrrhotite-magnetite buffer (<code class="reqn">\mathrm{FeS_2}</code>-<code class="reqn">\mathrm{FeS}</code>-<code class="reqn">\mathrm{Fe_3O_4}</code>) a basis species common to each species is one representing <code class="reqn">Fe</code>.
Therefore, when writing reactions between the species in this buffer <code class="reqn">Fe</code> is conserved while <code class="reqn">\mathrm{H_2S}</code> and <code class="reqn">\mathrm{O2}</code> are the variables of interest.
The calculation for buffers attempts to determine which of the available basis species qualifies as a conserved quantity.
This can be overriden with <code>balance</code>.
The default value of <code>balance</code> is &lsquo;<span class="samp">&#8288;PBB&#8288;</span>&rsquo;, which instructs the function to use the protein backbone group as the conserved quantity in buffers consisting of proteins, but has no overriding effect on the computations for buffers without proteins.
</p>
<p>To view the available buffers, print the <code><a href="#topic+thermo">thermo</a>()$buffer</code> object.
Buffer definitions can be added to this dataframe with <code>mod.buffer</code>.
The defaults for <code>state</code> and <code>logact</code> are intended for mineral buffers.
If <code>name</code> identifies an already defined buffer, this function modifies the logarithms of activities or states of species in that buffer, optionally restricted to only those species given in <code>species</code>.
</p>
<p>It is possible to assign different buffers to different basis species, in which case the order of their calculation depends on their order in <code>thermo()$buffers</code>.
This function is compatible with systems of proteins, but note that for buffers <em>made</em> of proteins the buffer calculations presently use whole protein formulas (instead of residue equivalents) and consider nonionized proteins only.
</p>


<h3>References</h3>

<p>Garrels, R. M. (1960) <em>Mineral Equilibria</em>. Harper &amp; Brothers, New York, 254 p. <a href="https://www.worldcat.org/oclc/552690">https://www.worldcat.org/oclc/552690</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diagram">diagram</a></code> with <code>type</code> set to the name of a basis species solves for the activity of the basis species.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## List the buffers
thermo()$buffer
# Another way to do it, for a specific buffer
print(mod.buffer("PPM"))

## Buffer made of one species
# Calculate the activity of CO2 in equilibrium with
# (a buffer made of) acetic acid at a given activity
basis("CHNOS")
basis("CO2", "AC")
# What activity of acetic acid are we using?
print(mod.buffer("AC"))
# Return the activity of CO2
affinity(return.buffer = TRUE)$CO2  # -7.057521
# As a function of oxygen fugacity
affinity(O2 = c(-85, -70, 4), return.buffer = TRUE)
# As a function of logfO2 and temperature
affinity(O2 = c(-85, -70, 4), T = c(25, 100, 4), return.buffer = TRUE)
# Change the activity of species in the buffer
mod.buffer("AC", logact = -10)
affinity(O2 = c(-85,-70,4), T = c(25, 100, 4), return.buffer = TRUE)

## Buffer made of three species
## Pyrite-Pyrrhotite-Magnetite (PPM)
# Specify basis species and initial activities
basis(c("FeS2", "H2S", "O2", "H2O"), c(0, -10, -50, 0))
# Note that the affinity of formation of pyrite,
# which corresponds to FeS2 in the basis, is zero
species(c("pyrite", "pyrrhotite", "magnetite"))
affinity(T = c(200, 400, 11), P = 2000)$values
# Setup H2S and O2 to be buffered by PPM
basis(c("H2S", "O2"), c("PPM", "PPM"))
# Inspect values of H2S activity and O2 fugacity
affinity(T = c(200, 400, 11), P = 2000, return.buffer = TRUE, exceed.Ttr = TRUE)
# Calculate affinities of formation reactions of species in the buffer
a &lt;- affinity(T = c(200, 400, 11), P = 2000, exceed.Ttr = TRUE)$values
# The affinities for species in the buffer are all equal to zero
all.equal(as.numeric(a[[1]]), rep(0, 11))  # TRUE
all.equal(as.numeric(a[[2]]), rep(0, 11))  # TRUE
all.equal(as.numeric(a[[3]]), rep(0, 11))  # TRUE

## Buffer made of one species: show values of logfO2 on an 
## Eh-pH diagram; after Garrels, 1960, Figure 6
basis("CHNOSe")
# Here we will buffer the activity of the electron by O2
mod.buffer("O2", "O2", "gas", 999)
basis("e-", "O2")
# Start our plot, then loop over values of logfO2
thermo.plot.new(xlim = c(0, 14), ylim = c(-0.8, 1.2),
  xlab = "pH",ylab = axis.label("Eh"))
# The upper and lower lines correspond to the upper
# and lower stability limits of water
logfO2 &lt;- c(0, -20, -40, -60, -83.1)
for(i in 1:5) {
  # Update the logarithm of fugacity (logact) of O2 in the buffer
  mod.buffer("O2", "O2", "gas", logfO2[i])
  # Get the values of the logarithm of activity of the electron
  a &lt;- affinity(pH = c(0, 14, 15), return.buffer = TRUE)
  # Convert values of pe (-logact of the electron) to Eh
  Eh &lt;- convert(-as.numeric(a$`e-`), "Eh")
  lines(seq(0, 14, length.out = 15), Eh)
  # Add some labels
  text(seq(0, 14, length.out = 15)[i*2+2], Eh[i*2+2],
    paste("logfO2 =", logfO2[i]))
}
title(main = paste("Relation between logfO2(g), Eh and pH at\n",
  "25 degC and 1 bar. After Garrels, 1960"))

## Buffer made of two species
# Conditions for metastable equilibrium among 
# CO2 and acetic acid. note their starting activities:
print(mod.buffer("CO2-AC")) 
basis("CHNOS")
basis("O2", "CO2-AC")
affinity(return.buffer = TRUE)  # logfO2 = -75.94248
basis("CO2", 123)  # what the buffer reactions are balanced on
affinity(return.buffer = TRUE)  # unchanged
# Consider more oxidizing conditions
mod.buffer("CO2-AC", logact = c(0, -10))
affinity(return.buffer = TRUE)
</code></pre>

<hr>
<h2 id='DEW'>Deep Earth Water (DEW) Model</h2><span id='topic+DEW'></span><span id='topic+calculateDensity'></span><span id='topic+calculateGibbsOfWater'></span><span id='topic+calculateEpsilon'></span><span id='topic+calculateQ'></span>

<h3>Description</h3>

<p>Calculate thermodynamic properties of water using the Deep Earth Water (DEW) model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  calculateDensity(pressure, temperature, error = 0.01)
  calculateGibbsOfWater(pressure, temperature)
  calculateEpsilon(density, temperature)
  calculateQ(density, temperature)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEW_+3A_pressure">pressure</code></td>
<td>
<p>numeric, pressure (bar)</p>
</td></tr>
<tr><td><code id="DEW_+3A_temperature">temperature</code></td>
<td>
<p>numeric, temperature (&deg;C)</p>
</td></tr>
<tr><td><code id="DEW_+3A_error">error</code></td>
<td>
<p>numeric, residual error for bisection calculation</p>
</td></tr>
<tr><td><code id="DEW_+3A_density">density</code></td>
<td>
<p>numeric, density (g/cm^3)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Deep Earth Water (DEW) model, described by Sverjensky et al., 2014, extends the applicability of the revised HKF equations of state to 60 kbar.
This implementation of DEW is based on the VBA macro code in the May, 2017 version of the DEW spreadsheet downloaded from http://www.dewcommunity.org/.
The spreadsheet provides multiple options for some calculations; here the default equations for density of water (Zhang and Duan, 2005), dielectric constant (Sverjensky et al., 2014) and Gibbs energy of water (integral of volume, equation created by Brandon Harrison) are used.
</p>
<p>Comments in the original code indicate that <code>calculateGibbsOfWater</code> is valid for 100 &le; <I>T</I> &le; 1000 &deg;C and <I>P</I> &ge; 1000 bar.
Likewise, the power function fit of the dielectric constant (epsilon) is valid for 100 &le; <I>T</I> &le; 1200 &deg;C and <I>P</I> &ge; 1000 bar (Sverjensky et al., 2014).
</p>


<h3>Value</h3>

<p>The calculated values of density, Gibbs energy, and the Q Born coefficient have units of g/cm^3, cal/mol, and bar^-1 (epsilon is dimensionless).
</p>


<h3>References</h3>

<p>Sverjensky, D. A., Harrison, B. and Azzolini, D. (2014) Water in the deep Earth: The dielectric constant and the solubilities of quartz and corundum to 60 kb and 1,200 &deg;C. <em>Geochim. Cosmochim. Acta</em> <b>129</b>, 125&ndash;145. <a href="https://doi.org/10.1016/j.gca.2013.12.019">doi:10.1016/j.gca.2013.12.019</a>
</p>
<p>Zhang, Z. and Duan, Z. (2005) Prediction of the <em>PVT</em> properties of water over wide range of temperatures and pressures from molecular dynamics simulation. <em>Phys. Earth Planet. Inter.</em> <b>149</b>, 335&ndash;354. <a href="https://doi.org/10.1016/j.pepi.2004.11.003">doi:10.1016/j.pepi.2004.11.003</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+water.DEW">water.DEW</a></code>; use <code>water("DEW")</code> to activate these equations for the main functions in CHNOSZ.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pressure &lt;- c(1000, 60000)
temperature &lt;- c(100, 1000)
calculateGibbsOfWater(pressure, temperature)
(density &lt;- calculateDensity(pressure, temperature))
calculateEpsilon(density, temperature)
calculateQ(density, temperature)
</code></pre>

<hr>
<h2 id='diagram'>Chemical Activity Diagrams</h2><span id='topic+diagram'></span><span id='topic+find.tp'></span>

<h3>Description</h3>

<p>Plot equilibrium chemical activity (1-D speciation) or equal-activity (2-D predominance) diagrams as a function of chemical activities of basis species, temperature and/or pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  diagram(
    # species affinities or activities
    eout,
    # type of plot
    type = "auto", alpha = FALSE, normalize = FALSE,
    as.residue = FALSE, balance = NULL, groups = as.list(1:length(eout$values)),
    # figure size and sides for axis tick marks
    xrange = NULL, mar = NULL, yline = par("mgp")[1]+0.3, side = 1:4,
    # axis limits and labels
    ylog = TRUE, xlim = NULL, ylim = NULL, xlab = NULL, ylab = NULL,
    # character sizes
    cex = par("cex"), cex.names = 1, cex.axis = par("cex"),
    # line styles
    lty = NULL, lty.cr = NULL, lty.aq = NULL, lwd = par("lwd"), dotted = NULL,
    spline.method = NULL, contour.method = "edge", levels = NULL,
    # colors
    col = par("col"), col.names = par("col"), fill = NULL,
    fill.NA = "gray80", limit.water = NULL,
    # field and line labels
    names = NULL, format.names = TRUE, bold = FALSE, italic = FALSE, 
    font = par("font"), family = par("family"), adj = 0.5,
    dx = 0, dy = 0, srt = 0, min.area = 0,
    # title and legend
    main = NULL, legend.x = NA,
    # plotting controls
    add = FALSE, plot.it = TRUE, tplot = TRUE, ...)
  find.tp(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diagram_+3A_eout">eout</code></td>
<td>
<p>list, object returned by <code><a href="#topic+affinity">affinity</a></code>, <code><a href="#topic+equilibrate">equilibrate</a></code> or related functions</p>
</td></tr>
<tr><td><code id="diagram_+3A_type">type</code></td>
<td>
<p>character, type of plot, or name of basis species whose activity to plot</p>
</td></tr>
<tr><td><code id="diagram_+3A_alpha">alpha</code></td>
<td>
<p>logical or character (&lsquo;<span class="samp">&#8288;balance&#8288;</span>&rsquo;), for speciation diagrams, plot degree of formation instead of activities?</p>
</td></tr>
<tr><td><code id="diagram_+3A_normalize">normalize</code></td>
<td>
<p>logical, divide chemical affinities by balance coefficients and rescale activities to whole formulas?</p>
</td></tr>
<tr><td><code id="diagram_+3A_as.residue">as.residue</code></td>
<td>
<p>logical, divide chemical affinities by balance coefficients (no rescaling)?</p>
</td></tr>
<tr><td><code id="diagram_+3A_balance">balance</code></td>
<td>
<p>character, balancing constraint; see <code><a href="#topic+equilibrate">equilibrate</a></code></p>
</td></tr>
<tr><td><code id="diagram_+3A_groups">groups</code></td>
<td>
<p>list of numeric, groups of species to consider as a single effective species</p>
</td></tr>
<tr><td><code id="diagram_+3A_xrange">xrange</code></td>
<td>
<p>numeric, range of x-values between which predominance field boundaries are plotted</p>
</td></tr>
<tr><td><code id="diagram_+3A_mar">mar</code></td>
<td>
<p>numeric, margins of plot frame</p>
</td></tr>
<tr><td><code id="diagram_+3A_yline">yline</code></td>
<td>
<p>numeric, margin line on which to plot the y-axis name</p>
</td></tr>
<tr><td><code id="diagram_+3A_side">side</code></td>
<td>
<p>numeric, which sides of plot to draw axes</p>
</td></tr>
<tr><td><code id="diagram_+3A_xlim">xlim</code></td>
<td>
<p>numeric, limits of x-axis</p>
</td></tr>
<tr><td><code id="diagram_+3A_ylim">ylim</code></td>
<td>
<p>numeric, limits of y-axis</p>
</td></tr>
<tr><td><code id="diagram_+3A_xlab">xlab</code></td>
<td>
<p>character, label to use for x-axis</p>
</td></tr>
<tr><td><code id="diagram_+3A_ylab">ylab</code></td>
<td>
<p>character, label to use for y-axis</p>
</td></tr>
<tr><td><code id="diagram_+3A_ylog">ylog</code></td>
<td>
<p>logical, use a logarithmic y-axis (on 1D degree diagrams)?</p>
</td></tr>
<tr><td><code id="diagram_+3A_cex">cex</code></td>
<td>
<p>numeric, character expansion (scaling relative to current)</p>
</td></tr>
<tr><td><code id="diagram_+3A_cex.names">cex.names</code></td>
<td>
<p>numeric, character expansion factor to be used for names of species on plots</p>
</td></tr>
<tr><td><code id="diagram_+3A_cex.axis">cex.axis</code></td>
<td>
<p>numeric, character expansion factor for names of axes</p>
</td></tr>
<tr><td><code id="diagram_+3A_lty">lty</code></td>
<td>
<p>numeric, line types to be used in plots</p>
</td></tr>
<tr><td><code id="diagram_+3A_lty.cr">lty.cr</code></td>
<td>
<p>numeric, line types for cr-cr boundaries (between two minerals)</p>
</td></tr>
<tr><td><code id="diagram_+3A_lty.aq">lty.aq</code></td>
<td>
<p>numeric, line types for aq-aq boundaries (between two aqueous species)</p>
</td></tr>
<tr><td><code id="diagram_+3A_lwd">lwd</code></td>
<td>
<p>numeric, line width</p>
</td></tr>
<tr><td><code id="diagram_+3A_dotted">dotted</code></td>
<td>
<p>numeric, how often to skip plotting points on predominance field boundaries (to gain the effect of dotted or dashed boundary lines)</p>
</td></tr>
<tr><td><code id="diagram_+3A_spline.method">spline.method</code></td>
<td>
<p>character, method used in <code><a href="stats.html#topic+splinefun">splinefun</a></code></p>
</td></tr>
<tr><td><code id="diagram_+3A_contour.method">contour.method</code></td>
<td>
<p>character, labelling method used in <code><a href="graphics.html#topic+contour">contour</a></code> (use NULL for no labels).</p>
</td></tr>
<tr><td><code id="diagram_+3A_levels">levels</code></td>
<td>
<p>numeric, levels at which to draw contour lines</p>
</td></tr>
<tr><td><code id="diagram_+3A_col">col</code></td>
<td>
<p>character, color of activity lines (1D diagram) or predominance field boundaries (2D diagram)</p>
</td></tr>
<tr><td><code id="diagram_+3A_col.names">col.names</code></td>
<td>
<p>character, colors for labels of species</p>
</td></tr>
<tr><td><code id="diagram_+3A_fill">fill</code></td>
<td>
<p>character, colors used to fill predominance fields</p>
</td></tr>
<tr><td><code id="diagram_+3A_fill.na">fill.NA</code></td>
<td>
<p>character, color for grid points with NA values</p>
</td></tr>
<tr><td><code id="diagram_+3A_limit.water">limit.water</code></td>
<td>
<p>NULL or logical, set NA values beyond water stability limits?</p>
</td></tr>
<tr><td><code id="diagram_+3A_names">names</code></td>
<td>
<p>character, names of species for activity lines or predominance fields</p>
</td></tr>
<tr><td><code id="diagram_+3A_format.names">format.names</code></td>
<td>
<p>logical, apply formatting to chemical formulas?</p>
</td></tr>
<tr><td><code id="diagram_+3A_bold">bold</code></td>
<td>
<p>logical, use bold formatting for names?</p>
</td></tr>
<tr><td><code id="diagram_+3A_italic">italic</code></td>
<td>
<p>logical, use italic formatting for names?</p>
</td></tr>
<tr><td><code id="diagram_+3A_font">font</code></td>
<td>
<p>character, font type for names (has no effect if <code>format.names</code> is TRUE)</p>
</td></tr>
<tr><td><code id="diagram_+3A_family">family</code></td>
<td>
<p>character, font family for names</p>
</td></tr>
<tr><td><code id="diagram_+3A_adj">adj</code></td>
<td>
<p>numeric, adjustment for line labels</p>
</td></tr>
<tr><td><code id="diagram_+3A_dx">dx</code></td>
<td>
<p>numeric, x offset for line or field labels</p>
</td></tr>
<tr><td><code id="diagram_+3A_dy">dy</code></td>
<td>
<p>numeric, y offset for line or field labels</p>
</td></tr>
<tr><td><code id="diagram_+3A_srt">srt</code></td>
<td>
<p>numeric, rotation for line labels</p>
</td></tr>
<tr><td><code id="diagram_+3A_min.area">min.area</code></td>
<td>
<p>numeric, minimum area of fields that should be labeled, expressed as a fraction of the total plot area</p>
</td></tr>
<tr><td><code id="diagram_+3A_main">main</code></td>
<td>
<p>character, a main <code><a href="graphics.html#topic+title">title</a></code> for the plot; <code>NULL</code> means to plot no title</p>
</td></tr>
<tr><td><code id="diagram_+3A_legend.x">legend.x</code></td>
<td>
<p>character, description of legend placement passed to <code><a href="graphics.html#topic+legend">legend</a></code></p>
</td></tr>
<tr><td><code id="diagram_+3A_add">add</code></td>
<td>
<p>logical, add to current plot?</p>
</td></tr>
<tr><td><code id="diagram_+3A_plot.it">plot.it</code></td>
<td>
<p>logical, make a plot?</p>
</td></tr>
<tr><td><code id="diagram_+3A_tplot">tplot</code></td>
<td>
<p>logical, set up plot with <code><a href="#topic+thermo.plot.new">thermo.plot.new</a></code>?</p>
</td></tr>
<tr><td><code id="diagram_+3A_x">x</code></td>
<td>
<p>matrix, value of the <code>predominant</code> list element from <code>diagram</code></p>
</td></tr>
<tr><td><code id="diagram_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="base.html#topic+plot">plot</a></code> or <code><a href="graphics.html#topic+barplot">barplot</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function displays diagrams representing either chemical affinities or equilibrium chemical activities of species.
The first argument is the output from <code><a href="#topic+affinity">affinity</a></code>, <code><a href="#topic+rank.affinity">rank.affinity</a></code>, <code><a href="#topic+equilibrate">equilibrate</a></code>, or <code><a href="#topic+solubility">solubility</a></code>.
0-D diagrams, at a single point, are shown as <code><a href="graphics.html#topic+barplot">barplot</a></code>s.
1-D diagrams, for a single variable on the x-axis, are plotted as lines.
2-D diagrams, for two variables, are plotted as predominance fields.
The allowed variables are any that <code><a href="#topic+affinity">affinity</a></code> or the other functions accepts: temperature, pressure, or the chemical activities of the basis species.
</p>
<p>A new plot is started unless <code>add</code> is TRUE.
If <code>plot.it</code> is FALSE, no plot will be generated but all the intermediate computations will be performed and the results returned.
</p>
<p>Line or field labels use the names of the species as provided in <code>eout</code>; formatting is applied to chemical formulas (see <code><a href="#topic+expr.species">expr.species</a></code>) unless <code>format.names</code> is FALSE.
Set <code>names</code> to TRUE or NULL to plot the names, or FALSE, NA, or <code>""</code> to prevent plotting the names, or a character argument to replace the default species names.
Alternatively, supply a numeric value to <code>names</code> to indicate a subset of default names that should or shouldn't be plotted (positive and negative indices, respectively).
Use <code>col.names</code> and <code>cex.names</code> to change the colors and size of the labels.
Use <code>cex</code> and <code>cex.axis</code> to adjust the overall character expansion factors (see <code><a href="graphics.html#topic+par">par</a></code>) and those of the axis labels.
The x- and y-axis labels are automatically generated unless they are supplied in <code>xlab</code> and <code>ylab</code>. 
</p>
<p>If <code>groups</code> is supplied, the activities of the species identified in each numeric element of this list are multiplied by the balance coefficients of the species, then summed together.
The names of the list are used to label the lines or fields for the summed activities of the resulting groups.
</p>
<p>Normalizing protein formulas by length gives &ldquo;residue equivalents&rdquo; (Dick and Shock, 2011) that are useful for equilibrium calculations with proteins.
<code>normalize</code> and <code>as.residue</code> are only usable when <code>eout</code> is the output from <code>affinity</code>, and only one can be TRUE.
If <code>normalize</code> is TRUE, formation reactions and their affinities are first divided by protein length, so equal activities of residue equivalents are considered; then, the residue activities are rescaled to whole proteins for making the plot.
If <code>as.residue</code> is TRUE, no rescaling is performed, so the diagram represents activities of the residues, not the whole proteins.
</p>


<h3><code>type</code> argument</h3>

<p>This paragraph describes the effect of the <code>type</code> argument when the output from <code>affinity</code> is being used.
For <code>type</code> set to &lsquo;<span class="samp">&#8288;auto&#8288;</span>&rsquo;, and with 0 or 1 variables defined in <code><a href="#topic+affinity">affinity</a></code>, the property computed by <code>affinity</code> for each species is plotted.
This is usually the affinity of the formation reactions, but can be set to some other property (using the <code>property</code> argument of <code>affinity</code>), such as the equilibrium constant (&lsquo;<span class="samp">&#8288;logK&#8288;</span>&rsquo;).
For two variables, equilibrium predominance (maximum affinity) fields are plotted.
This &ldquo;maximum affinity method&rdquo; (Dick, 2019) uses balancing coefficients that are specified by the <code>balance</code> argument.
If <code>type</code> is &lsquo;<span class="samp">&#8288;saturation&#8288;</span>&rsquo;, the function plots the line for each species where the affinity of formation equals zero (see <code>demo("saturation")</code> for an example).
If for a given species no saturation line is possible or the range of the diagram does not include the saturation line, the function prints a message instead.
If <code>type</code> is the name of a basis species, then the equilibrium activity of the selected basis species in each of the formation reactions is plotted (see the CO<sub>2</sub>-acetic acid example in <code><a href="#topic+buffer">buffer</a></code>).
In the case of 2-D diagrams, both of these options use <code><a href="graphics.html#topic+contour">contour</a></code> to draw the lines, with the method specified in <code>contour.method</code>.
</p>
<p>This paragraph describes the effect of the <code>type</code> argument when the output from <code>solubility</code> is being used.
For one mineral or gas, if <code>type</code> set to &lsquo;<span class="samp">&#8288;auto&#8288;</span>&rsquo;, the equilibrium activities of each aqueous species are plotted.
If <code>type</code> is &lsquo;<span class="samp">&#8288;loga.balance&#8288;</span>&rsquo;, the activity of the balancing basis species (i.e. total solubility) is plotted; this is represented by contours on a 2-D diagram.
For two or more minerals or gases, if <code>type</code> set to &lsquo;<span class="samp">&#8288;auto&#8288;</span>&rsquo;, the values of &lsquo;<span class="samp">&#8288;loga.balance&#8288;</span>&rsquo; (overall minimum solubility) are plotted.
If <code>type</code> is &lsquo;<span class="samp">&#8288;loga.equil&#8288;</span>&rsquo;, the solubilities of the individual minerals and gases are plotted.
For examples that use these features, see <code><a href="#topic+solubility">solubility</a></code> and various <code><a href="#topic+demos">demos</a></code>: &lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;contour&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;gold&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;solubility&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;sphalerite&#8288;</span>&rsquo;.
</p>


<h3>1-D diagrams</h3>

<p>For 1-D diagrams, the default setting for the y-axis is a logarithmic scale (unless <code>alpha</code> is TRUE) with limits corresponding to the range of logarithms of activities (or 0,1 if <code>alpha</code> is TRUE); these actions can be overridden by <code>ylog</code> and <code>ylim</code>.
If <code>legend.x</code> is NA (the default), the lines are labeled with the names of the species near the maximum value.
Otherwise, a <code><a href="graphics.html#topic+legend">legend</a></code> is placed at the location identified by <code>legend.x</code>, or omitted if <code>legend.x</code> is NULL.
</p>
<p>If <code>alpha</code> is TRUE, the fractional degrees of formation (ratios of activities to total activity) are plotted.
Or, setting <code>alpha</code> to &lsquo;<span class="samp">&#8288;balance&#8288;</span>&rsquo; allows the activities to be multiplied by the number of the balancing component; this is useful for making &ldquo;percent carbon&rdquo; diagrams where the species differ in carbon number.
The line type and line width can be controlled with <code>lty</code> and <code>lwd</code>, respectively.
To connect the points with splines instead of lines, set <code>spline.method</code> to one of the methods in <code><a href="stats.html#topic+splinefun">splinefun</a></code>.
</p>


<h3>2-D diagrams</h3>

<p>On 2-D diagrams, the fields represent the species with the highest equilibrium activity.
<code>fill</code> determines the color of the predominance fields, <code>col</code> that of the boundary lines.
The default of NULL for <code>fill</code> uses a light blue, light tan, and darker tan color for fields with aqueous species, one solid, or two solids.
These correspond to the web colors &quot;aliceblue&quot;, &quot;antiquewhite&quot;, and &quot;burlywood&quot; with some transparency added; see <a href="../doc/multi-metal.html"><strong>multi-metal</strong></a> for an example with two solids produced using <code><a href="#topic+mix">mix</a></code>.
If all the species in the diagram have the same state, or if the <code>fill</code> argument is NA or a 0-length value, the predominance fields are transparent, i.e. no fill color is used.
Otherwise, <code>fill</code> can be any <code><a href="grDevices.html#topic+colors">colors</a></code>, or the word &lsquo;<span class="samp">&#8288;rainbow&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;heat&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;terrain&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;topo&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;cm&#8288;</span>&rsquo;, indicating a palette from <span class="pkg">grDevices</span>.
Starting with <span class="rlang"><b>R</b></span> version 3.6.0, <code>fill</code> can be the name of any available HCL color palette, matched in the same way as the <code>palette</code> argument of <code>hcl.colors</code>.
</p>
<p><code>fill.NA</code> gives the color for empty fields, i.e. points at which NA values are present for any species.
This may occur when there are missing thermodynamic data or the temperature or pressure are not in the range of the equations of state.
To make overlay diagrams easier to construct, the default for <code>fill.NA</code> is automatically changed to &lsquo;<span class="samp">&#8288;transparent&#8288;</span>&rsquo; when <code>add</code> is TRUE.
</p>
<p>If <code>limit.water</code> is TRUE, the diagram is clipped to the the water stability region on Eh-pH (and some other) diagrams.
That is, predominance fields are shown only where water is stable, and <code>fill.NA</code> is used for areas where H<sub>2</sub>O is not stable.
The default of NULL for <code>limit.water</code> does not clip the main diagram but instead overlays it on the water stability fields.
Change <code>limit.water</code> to FALSE to not show the water stability regions at all; this is automatically done if <code>limit.water</code> is missing and <code>add</code> is TRUE.
</p>
<p>The default line-drawing algorithm uses <code><a href="grDevices.html#topic+contourLines">contourLines</a></code> to obtain smooth-looking diagonal and curved lines, at the expense of not coinciding exactly with the rectangular grid that is used for drawing colors.
<code>lty</code>, <code>col</code>, and <code>lwd</code> can be specified, but limiting the lines via <code>xrange</code> is not currently supported.
Set <code>lty.cr</code> or <code>lty.aq</code> to 0 to suppress boundary lines between minerals or aqueous species.
</p>
<p>To go back to the old behavior for drawing lines, set <code>dotted</code> to &lsquo;<span class="samp">&#8288;0&#8288;</span>&rsquo;.
The old behavior does not respect <code>lty</code>; instead, the style of the boundary lines on 2-D diagrams can be altered by supplying one or more non-zero integers in <code>dotted</code>, which indicates the fraction of line segments to omit; a value of &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo; or NULL for <code>dotted</code> has the effect of not drawing the boundary lines.
</p>


<h3>Activity Coefficients</h3>

<p>The wording in this page and names of variables in functions refer exclusively to &lsquo;<span class="samp">&#8288;activities&#8288;</span>&rsquo; of aqueous species.
However, if activity coefficients are calculated (using the <code>IS</code> argument in <code><a href="#topic+affinity">affinity</a></code>), then these variables are effectively transformed to molalities (see <code>inst/tinytest/test-logmolality.R</code>).
So that the labels on diagrams are adjusted accordingly, <code><a href="#topic+diagram">diagram</a></code> sets the <code>molality</code> argument of <code><a href="#topic+axis.label">axis.label</a></code> to TRUE if <code>IS</code> was supplied as an argument to <code><a href="#topic+affinity">affinity</a></code>.
The labeling as molality takes effect even if <code>IS</code> is set to 0; this way, by including (or not) the <code>IS = 0</code> argument to <code>affinity</code>, the user decides whether to label aqueous species variables as molality (or activity) for calculations at zero ionic strength (where molality = activity).
</p>


<h3>Other Functions</h3>

<p><code>find.tp</code> finds the locations in a matrix of integers that are surrounded by the greatest number of different values.
The function counts the unique values in a 3x3 grid around each point and returns a matrix of indices (similar to <code><a href="base.html#topic+which">which</a>(..., arr.ind = TRUE)</code>) for the maximum count (ties result in more than one pair of indices).
It can be used with the output from <code>diagram</code> for calculations in 2 dimensions to approximately locate the triple points on the diagram.
</p>


<h3>Value</h3>

<p><code>diagram</code> returns an <code><a href="base.html#topic+invisible">invisible</a></code> list containing, first, the contents of <code>eout</code>, i.e. the output of <code><a href="#topic+affinity">affinity</a></code> or <code><a href="#topic+equilibrate">equilibrate</a></code> supplied in the function call.
To this are added the names of the plotted variable in <code>plotvar</code>, the labels used for species (which may be <code><a href="grDevices.html#topic+plotmath">plotmath</a></code> expressions if <code>format.names</code> is TRUE) in <code>names</code>, and the values used for plotting in a list named <code>plotvals</code>.
For 1-D diagrams, <code>plotvals</code> usually corresponds to the chemical activities of the species (i.e. <code>eout$loga.equil</code>), or, if <code>alpha</code> is <code>TRUE</code>, their mole fractions (degrees of formation).
For 2-D diagrams, <code>plotvals</code> corresponds to the values of affinity (from <code>eout$values</code>) divided by the respective balancing coefficients for each species.
For 2-D diagrams, the output also contains the matrices <code>predominant</code>, which identifies the predominant species in <code>eout$species</code> at each grid point, and <code>predominant.values</code>, which has the affinities of the predominant species divided by the balancing coefficients (if <code>eout</code> is the output of <code>affinity</code>) or the activities of the predominant species (if <code>eout</code> is the output of <code>equilibrate</code>).
The rows and columns of these matrices correspond to the <em>x</em> and <em>y</em> variables, respectively.
</p>


<h3>References</h3>

<p>Aksu, S. and Doyle, F. M. (2001) Electrochemistry of copper in aqueous glycine solutions. <em>J. Electrochem. Soc.</em> <b>148</b>, B51&ndash;B57.
</p>
<p>Dick, J. M. (2019) CHNOSZ: Thermodynamic calculations and diagrams for geochemistry. <em>Front. Earth Sci.</em> <b>7</b>:180. <a href="https://doi.org/10.3389/feart.2019.00180">doi:10.3389/feart.2019.00180</a>
</p>
<p>Dick, J. M. and Shock, E. L. (2011) Calculation of the relative chemical stabilities of proteins as a function of temperature and redox chemistry in a hot spring. <em>PLOS One</em> <b>6</b>, e22782. <a href="https://doi.org/10.1371/journal.pone.0022782">doi:10.1371/journal.pone.0022782</a>
</p>
<p>Helgeson, H. C. (1970) A chemical and thermodynamic model of ore deposition in hydrothermal systems. <em>Mineral. Soc. Amer. Spec. Pap.</em> <b>3</b>, 155&ndash;186. <a href="https://www.worldcat.org/oclc/583263">https://www.worldcat.org/oclc/583263</a>
</p>
<p>Helgeson, H. C., Delany, J. M., Nesbitt, H. W. and Bird, D. K. (1978) Summary and critique of the thermodynamic properties of rock-forming minerals. <em>Am. J. Sci.</em> <b>278-A</b>, 1&ndash;229. <a href="https://www.worldcat.org/oclc/13594862">https://www.worldcat.org/oclc/13594862</a>
</p>
<p>LaRowe, D. E. and Helgeson, H. C. (2007) Quantifying the energetics of metabolic reactions in diverse biogeochemical systems: electron flow and ATP synthesis. <em>Geobiology</em> <b>5</b>, 153&ndash;168. <a href="https://doi.org/10.1111/j.1472-4669.2007.00099.x">doi:10.1111/j.1472-4669.2007.00099.x</a>
</p>
<p>Majzlan, J., Navrotsky, A., McClesky, R. B. and Alpers, C. N. (2006) Thermodynamic properties and crystal structure refinement of ferricopiapite, coquimbite, rhomboclase, and Fe<sub>2</sub>(SO<sub>4</sub>)<sub>3</sub>(H<sub>2</sub>O)<sub>5</sub>. <em>Eur. J. Mineral.</em> <b>18</b>, 175&ndash;186. <a href="https://doi.org/10.1127/0935-1221/2006/0018-0175">doi:10.1127/0935-1221/2006/0018-0175</a>
</p>
<p>Tagirov, B. and Schott, J. (2001) Aluminum speciation in crustal fluids revisited. <em>Geochim. Cosmochim. Acta</em> <b>65</b>, 3965&ndash;3992. <a href="https://doi.org/10.1016/S0016-7037%2801%2900705-0">doi:10.1016/S0016-7037(01)00705-0</a>
</p>


<h3>See Also</h3>

 
<p><code><a href="#topic+Berman">Berman</a></code>, <code><a href="#topic+mix">mix</a></code>, <code><a href="#topic+mosaic">mosaic</a></code>, <code><a href="#topic+nonideal">nonideal</a></code>, <code><a href="#topic+solubility">solubility</a></code>, and <code><a href="#topic+util.plot">util.plot</a></code> are other help topics that use <code>diagram</code> in their examples.
See the <code><a href="#topic+demos">demos</a></code> for even more examples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Calculate the equilibrium logarithm of activity of a 
## basis species in different reactions
basis("CHNOS")
species(c("ethanol", "lactic acid", "deoxyribose", "ribose"))
a &lt;- affinity(T = c(0, 150))
diagram(a, type = "O2", legend.x = "topleft", col = rev(rainbow(4)), lwd = 2)
title(main = "Equilibrium logfO2 for 1e-3 mol/kg of CO2 and ... ")

### 1-D diagrams: logarithms of activities

## Degrees of formation of ionized forms of glycine
## After Fig. 1 of Aksu and Doyle, 2001 
basis("CHNOS+")
species(ispecies &lt;- info(c("glycinium", "glycine", "glycinate")))
a &lt;- affinity(pH = c(0, 14))
e &lt;- equilibrate(a)
diagram(e, alpha = TRUE, lwd = 1)
title(main = paste("Degrees of formation of aqueous glycine species\n",
  "after Aksu and Doyle, 2001"))

## Degrees of formation of ATP species as a function of 
## temperature, after LaRowe and Helgeson, 2007, Fig. 10b
# to make a similar diagram, activity of Mg+2 here is set to
# 10^-4, which is different from LH07, who used 10^-3 total molality
basis(c("CO2", "NH3", "H2O", "H3PO4", "O2", "H+", "Mg+2"),
  c(999, 999, 999, 999, 999, -5, -4))
species(c("HATP-3", "H2ATP-2", "MgATP-2", "MgHATP-"))
a &lt;- affinity(T = c(0, 120, 25))
e &lt;- equilibrate(a)
diagram(e, alpha = TRUE)  
title(main = paste("Degrees of formation of ATP species,\n",
  "pH=5, log(aMg+2)=-3. After LaRowe and Helgeson, 2007"),
  cex.main = 0.9)

### 2-D diagrams: predominance diagrams
### These use the maximum affinity method

## Fe-S-O at 200 deg C, after Helgeson, 1970
basis(c("Fe", "oxygen", "S2"))
species(c("iron", "ferrous-oxide", "magnetite",
  "hematite", "pyrite", "pyrrhotite"))
# The calculations include the polymorphic transitions of
# pyrrhotite; no additional step is needed
a &lt;- affinity(S2 = c(-50, 0), O2 = c(-90, -10), T=200)
diagram(a, fill = "heat")
title(main = paste("Fe-S-O, 200 degrees C, 1 bar",
  "After Helgeson, 1970", sep = "\n"))

## pe-pH diagram for hydrated iron sulfides,
## goethite and pyrite, after Majzlan et al., 2006
basis(c("Fe+2", "SO4-2", "H2O", "H+", "e-"), 
  c(0, log10(3), log10(0.75), 999, 999))
species(c("rhomboclase", "ferricopiapite", "hydronium jarosite",
  "goethite", "melanterite", "pyrite"))
a &lt;- affinity(pH = c(-1, 4, 256), pe = c(-5, 23, 256))
d &lt;- diagram(a, main = "Fe-S-O-H, after Majzlan et al., 2006")
water.lines(d, lwd = 2)
text(3, 22, describe.basis(2:3, digits = 2, oneline = TRUE))
text(3, 21, describe.property(c("T", "P"), c(25, 1), oneline = TRUE))

## Aqueous Al species, after Tagirov and Schott, 2001
basis(c("Al+3", "F-", "H+", "O2", "H2O"))
AlOH &lt;- c("Al(OH)4-", "Al(OH)3", "Al(OH)2+", "AlOH+2")
Al &lt;- "Al+3"
AlF &lt;- c("AlF+2", "AlF2+", "AlF3", "AlF4-")
AlOHF &lt;- c("Al(OH)2F2-", "Al(OH)2F", "AlOHF2")
species(c(AlOH, Al, AlF, AlOHF), "aq")
res &lt;- 300
a &lt;- affinity(pH = c(0.5, 6.5, res), `F-` = c(-2, -9, res), T = 200)
diagram(a, fill = "terrain")
dprop &lt;- describe.property(c("T", "P"), c(200, "Psat"))
legend("topright", legend = dprop, bty = "n")
mtitle(c("Aqueous aluminum species",
         "After Tagirov and Schott, 2001 Fig. 4d"), cex = 0.95)

## Temperature-Pressure: kayanite-sillimanite-andalusite
# cf. Fig. 49 of Helgeson et al., 1978
# this is a system of one component (Al2SiO5), however:
# - number of basis species must be the same as of elements
# - avoid using H2O or other aqueous species because of
#     T/P limits of the water() calculations;
basis(c("corundum", "quartz", "oxygen"))
species(c("kyanite", "sillimanite", "andalusite"))
# Database has transition temperatures of kyanite and andalusite
# at 1 bar only, so we permit calculation at higher temperatures
a &lt;- affinity(T = c(200, 900, 99), P = c(0, 9000, 101), exceed.Ttr = TRUE)
d &lt;- diagram(a, fill = NULL)
slab &lt;- syslab(c("Al2O3", "SiO2", "H2O"))
mtitle(c(as.expression(slab), "after Helgeson et al., 1978"))
# Find the approximate position of the triple point
tp &lt;- find.tp(d$predominant)
Ttp &lt;- a$vals[[1]][tp[1, 2]]
Ptp &lt;- rev(a$vals[[2]])[tp[1, 1]]
points(Ttp, Ptp, pch = 10, cex = 5)
</code></pre>

<hr>
<h2 id='EOSregress'>Regress Equations-of-State Parameters for Aqueous Species</h2><span id='topic+EOSregress'></span><span id='topic+EOSvar'></span><span id='topic+EOScalc'></span><span id='topic+EOSplot'></span><span id='topic+EOSlab'></span><span id='topic+EOScoeffs'></span><span id='topic+Cp_s_var'></span><span id='topic+V_s_var'></span>

<h3>Description</h3>

<p>Fit experimental volumes and heat capacities using regression equations.
Possible models include the Helgeson-Kirkham-Flowers (HKF) equations of state, or other equations defined using any combination of terms derived from the temperature, pressure and thermodynamic and electrostatic properties of water.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  EOSregress(exptdata, var = "", T.max = 9999, ...)
  EOSvar(var, T, P, ...)
  EOScalc(coefficients, T, P, ...)
  EOSplot(exptdata, var = NULL, T.max = 9999, T.plot = NULL, 
    fun.legend = "topleft", coefficients = NULL, add = FALSE,
    lty = par("lty"), col=par("col"), ...)
  EOSlab(var, coeff = "")
  EOScoeffs(species, property, P=1)
  Cp_s_var(T = 298.15, P = 1, omega.PrTr = 0, Z = 0)
  V_s_var(T = 298.15, P = 1, omega.PrTr = 0, Z = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EOSregress_+3A_exptdata">exptdata</code></td>
<td>
<p>dataframe, experimental data</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_var">var</code></td>
<td>
<p>character, name(s) of variables in the regression equations</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_t.max">T.max</code></td>
<td>
<p>numeric, maximum temperature for regression, in degrees Kelvin</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_t">T</code></td>
<td>
<p>numeric, temperature in Kelvin</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_p">P</code></td>
<td>
<p>numeric, pressure in bars</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_...">...</code></td>
<td>
<p>arguments specifying additional dependencies of the regression variables</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_t.plot">T.plot</code></td>
<td>
<p>numeric, upper limit of temperature range to plot</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_fun.legend">fun.legend</code></td>
<td>
<p>character, where to place legend on plot</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_coefficients">coefficients</code></td>
<td>
<p>dataframe, coefficients to use to make line on plot</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_add">add</code></td>
<td>
<p>logical, add lines to an existing plot?</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_lty">lty</code></td>
<td>
<p>line style</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_col">col</code></td>
<td>
<p>color of lines</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_coeff">coeff</code></td>
<td>
<p>numeric, value of equation of state parameter for plot legend</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_species">species</code></td>
<td>
<p>character, name of aqueous species</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_property">property</code></td>
<td>
<p>character, &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo;</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_omega.prtr">omega.PrTr</code></td>
<td>
<p>numeric, value of omega at reference T and P</p>
</td></tr>
<tr><td><code id="EOSregress_+3A_z">Z</code></td>
<td>
<p>numeric, charge</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>EOSregress</code> uses a linear model (<code><a href="stats.html#topic+lm">lm</a></code>) to regress the experimental heat capacity or volume data in <code>exptdata</code>, which is a data frame with columns &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; (temperature in degrees Kelvin), &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo; (pressure in bars), and &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo; (heat capacity in cal/mol.K or volume in cm3/mol).
The &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo; data must be in the third column.
Only data below the temperature of <code>T.max</code> are included in the regression.
The regression formula is specified by a vector of names in <code>var</code>.
The names of the variables can be any combination of the following (listed in the order of search): variables listed in the following table, any available property of <code><a href="#topic+water">water</a></code> (e.g. &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;alpha&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;QBorn&#8288;</span>&rsquo;), or the name of a function that can be found using <code><a href="base.html#topic+get">get</a></code> in the default environment.
Examples of the latter are <code>Cp_s_var</code>, <code>V_s_var</code>, or functions defined by the user in the global environment; the arguments of these functions must include, but are not limited to, <code>T</code> and <code>P</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
    <code>T</code>           </td><td style="text-align: left;">  <code class="reqn">T</code> (temperature) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>P</code>           </td><td style="text-align: left;">  <code class="reqn">P</code> (pressure) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>TTheta</code>      </td><td style="text-align: left;">  <code class="reqn">(T-\Theta)</code> (<code class="reqn">\Theta</code> = 228 K) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>invTTheta</code>   </td><td style="text-align: left;">  <code class="reqn">1/(T-\Theta)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>TTheta2</code>     </td><td style="text-align: left;">  <code class="reqn">(T-\Theta)^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>invTTheta2</code>  </td><td style="text-align: left;">  <code class="reqn">1/(T-\Theta)^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>invPPsi</code>     </td><td style="text-align: left;">  <code class="reqn">1/(P+\Psi)</code> (<code class="reqn">\Psi</code> = 2600 bar) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>invPPsiTTheta</code>   </td><td style="text-align: left;">  <code class="reqn">1/((P+\Psi)(T-\Theta))</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>TXBorn</code>          </td><td style="text-align: left;">  <code class="reqn">TX</code> (temperature times <code class="reqn">X</code> Born function) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>drho.dT</code>     </td><td style="text-align: left;">  <code class="reqn">d\rho/dT</code> (temperature derivative of density of water) </td>
</tr>
<tr>
 <td style="text-align: left;">
    <code>V.kT</code>        </td><td style="text-align: left;">  <code class="reqn">V\kappa_T</code> (volume times isothermal compressibility of water) 
  </td>
</tr>

</table>

<p><code>EOSvar</code> calculates the value of the variable named <code>var</code> (defined as described above) at the specified <code>T</code> (temperature in degrees Kelvin) and <code>P</code> (pressure in bars).
This function is used by <code>EOSregress</code> to get the values of the variables used in the regression.
</p>
<p><code>EOScalc</code> calculates the predicted heat capacities or volumes using coefficients provided by the result of <code>EOSregress</code>, at the temperatures and pressures specified by <code>T</code> and <code>P</code>.
</p>
<p><code>EOSplot</code> takes a table of data in <code>exptdata</code>, runs <code>EOSregress</code> and <code>EOScalc</code> and plots the results.
The experimental data are plotted as points, and the calculated values as a smooth line.
The point symbols are filled circles where the calculated value is within 10% of the experimental value; open circles otherwise.
</p>
<p><code>EOSlab</code> produces labels for the variables listed above that can be used <code><a href="base.html#topic+as.expression">as.expression</a></code>s in plots.
The value of <code>coeff</code> is prefixed to the name of the variable (using <code><a href="base.html#topic+substitute">substitute</a></code>, with a multiplication symbol).
For the properties listed in the table above, and selected properties listed in <code><a href="#topic+water">water</a></code>, the label is formatted using <code><a href="grDevices.html#topic+plotmath">plotmath</a></code> expressions (e.g., with italicized symbols and Greek letters).
If <code>var</code> is a user-defined function, the function can be given a &lsquo;<span class="samp">&#8288;label&#8288;</span>&rsquo; attribute to provide <code><a href="grDevices.html#topic+plotmath">plotmath</a></code>-style formatting; in this case the appropriate multiplication or division symbol should be specified (see example below).
</p>
<p><code>EOScoeffs</code> retrieves coefficients in the Helgeson-Kirkham-Flowers equations from the thermodynamic database (<code><a href="#topic+thermo">thermo</a>$OBIGT</code>) for the given aqueous <code>species</code>.
If the <code>property</code> is &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo;, the resulting data frame has column names of &lsquo;<span class="samp">&#8288;(Intercept)&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;invTTheta2&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;TX&#8288;</span>&rsquo;, respectively holding the coefficients <code class="reqn">c_1</code>, <code class="reqn">c_2</code> and <code class="reqn">\omega</code> in the equation <code class="reqn">Cp^\circ = c_1 + c_2/(T-\Theta)^2 + {\omega}TX</code>.
If the <code>property</code> is &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo;, the data frame has column names of &lsquo;<span class="samp">&#8288;(Intercept)&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;invTTheta&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;Q&#8288;</span>&rsquo;, respectively holding the coefficients <code class="reqn">\sigma</code>, <code class="reqn">\xi</code> and <code class="reqn">\omega</code> in <code class="reqn">V^\circ = \sigma + \xi/(T-\Theta) - {\omega}Q</code>.
Here, <code class="reqn">\sigma</code> and <code class="reqn">\xi</code> are calculated from a1, a2, a3 and a4 in <code>thermo()$OBIGT</code> at the pressure indicated by <code>P</code> (default 1 bar).
</p>
<p>The original motivation for writing these functions was to explore alternatives or possible modifications to the revised Helgeson-Kirkham-Flowers equations applied to aqueous nonelectrolytes.
As pointed out by Schulte et al., 2001, the functional forms of the equations do not permit retrieving values of the solvation parameter (<code class="reqn">\omega</code>) that closely represent the observed trends in both heat capacity and volume at high temperatures (above ca. 200 &deg;C).
</p>
<p>The examples below assume that the <code class="reqn">\omega</code> parameter in the HKF functions is a constant (does not depend on T and P), as is appropriate for nonelectrolytes.
For charged species, the variables <code>Cp_s_var</code> and <code>V_s_var</code> can be used in the regressions.
They correspond to the solvation contribution to heat capacity or volume, respectively, in the HKF EOS, divided by the value of <code class="reqn">\omega</code> at the reference temperature and pressure.
Because these variables are themselves a function of <code>omega.PrTr</code>, an iterative procedure is needed to perform the regression.
</p>
<p>Note that variables <code>QBorn</code> and <code>V_s_var</code> are both negated, so that the value of <code class="reqn">\omega</code> has its proper sign in the corresponding equations.
</p>


<h3>Value</h3>

<p>For <code>EOSregress</code>, an object of class &ldquo;lm&rdquo;.
<code>EOSvar</code> and <code>EOScalc</code> both return numeric values.
<code>EOScoeffs</code> returns a data frame.
</p>


<h3>References</h3>

<p>Hnědkovský, L. and Wood, R. H. (1997) Apparent molar heat capacities of aqueous solutions of CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S, and NH<sub>3</sub> at temperatures from 304 K to 704 K at a pressure of 28 MPa. <em>J. Chem. Thermodyn.</em> <b>29</b>, 731&ndash;747. <a href="https://doi.org/10.1006/jcht.1997.0192">doi:10.1006/jcht.1997.0192</a>
</p>
<p>Schulte, M. D., Shock, E. L. and Wood, R. H. (1995) The temperature dependence of the standard-state thermodynamic properties of aqueous nonelectrolytes. <em>Geochim. Cosmochim. Acta</em> <b>65</b>, 3919&ndash;3930. <a href="https://doi.org/10.1016/S0016-7037%2801%2900717-7">doi:10.1016/S0016-7037(01)00717-7</a>
</p>


<h3>See Also</h3>

<p>The vignette <a href="../doc/eos-regress.html"><strong>eos-regress</strong></a> has more references and examples, including an iterative method to retrieve <code>omega.PrTr</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Fit experimental heat capacities of CH4
## using revised Helgeson-Kirkham-Flowers equations
# Read the data from Hnedkovsky and Wood, 1997
f &lt;- system.file("extdata/cpetc/HW97_Cp.csv", package = "CHNOSZ")
d &lt;- read.csv(f)
# Use data for CH4
d &lt;- d[d$species == "CH4", ]
d &lt;- d[, -1]
# Convert J to cal and MPa to bar
d$Cp &lt;- convert(d$Cp, "cal")
d$P &lt;- convert(d$P, "bar")
# Specify the terms in the HKF equations
var &lt;- c("invTTheta2", "TXBorn")
# Perform regression, with a temperature limit
EOSlm &lt;- EOSregress(d, var, T.max = 600)
# Calculate the Cp at some temperature and pressure
EOScalc(EOSlm$coefficients, 298.15, 1)
# Get the database values of c1, c2 and omega for CH4(aq)
CH4coeffs &lt;- EOScoeffs("CH4", "Cp")
## Make plots comparing the regressions
## with the accepted EOS parameters of CH4
opar &lt;- par(mfrow = c(2,2))
EOSplot(d, T.max = 600)
title("Cp of CH4(aq), fit to 600 K")
legend("bottomleft", pch = 1, legend = "Hnedkovsky and Wood, 1997")
EOSplot(d, coefficients = CH4coeffs)
title("Cp from EOS parameters in database")
EOSplot(d, T.max = 600, T.plot = 600)
title("Cp fit to 600 K, plot to 600 K")
EOSplot(d, coefficients = CH4coeffs, T.plot = 600)
title("Cp from EOS parameters in database")
par(opar)

# Continuing from above, with user-defined variables
Theta &lt;- 228  # K 
invTTTheta3 &lt;- function(T, P) (2*T) / (T-T*Theta) ^ 3
invTX &lt;- function(T, P) 1 / T * water("XBorn", T = T, P = P)[,1]
# Print the calculated values of invTTTheta3
EOSvar("invTTTheta3", d$T, d$P)
# Use invTTTheta and invTX in a regression
var &lt;- c("invTTTheta3", "invTX")
EOSregress(d, var)
# Give them a "label" attribute for use in the legend
attr(invTTTheta3, "label") &lt;-
  quote(phantom()%*%2 * italic(T) / (italic(T) - italic(T) * Theta) ^ 3)
attr(invTX, "label") &lt;- quote(phantom() / italic(T * X))
# Uncomment the following to make the plot
#EOSplot(d, var)

## Model experimental volumes of CH4
## using HKF equation and an exploratory one
f &lt;- system.file("extdata/cpetc/HWM96_V.csv", package = "CHNOSZ")
d &lt;- read.csv(f)
# Use data for CH4 near 280 bar
d &lt;- d[d$species == "CH4", ]
d &lt;- d[, -1]
d &lt;- d[abs(d$P - 28) &lt; 0.1, ]
d$P &lt;- convert(d$P, "bar")
# The HKF equation
varHKF &lt;- c("invTTheta", "QBorn")
# alpha is the expansivity coefficient of water
varal &lt;- c("invTTheta", "alpha")
opar &lt;- par(mfrow = c(2, 2))
# For both HKF and the expansivity equation,
# we'll fit up to a temperature limit
EOSplot(d, varHKF, T.max = 663, T.plot = 625)
legend("bottomright", pch = 1, legend = "Hnedkovsky et al., 1996")
title("V of CH4(aq), HKF equation")
EOSplot(d, varal, T.max = 663, T.plot = 625)
title("V of CH4(aq), expansivity equation")
EOSplot(d, varHKF, T.max = 663)
title("V of CH4(aq), HKF equation")
EOSplot(d, varal, T.max = 663)
title("V of CH4(aq), expansivity equation")
par(opar)
# Note that the volume regression using the HKF gives
# a result for omega (coefficient on Q) that is
# not consistent with the high-T heat capacities
</code></pre>

<hr>
<h2 id='equilibrate'>Equilibrium Chemical Activities of Species</h2><span id='topic+equilibrate'></span><span id='topic+equil.boltzmann'></span><span id='topic+equil.reaction'></span><span id='topic+moles'></span>

<h3>Description</h3>

<p>Calculate equilibrium chemical activities of species from the affinities of formation of the species at unit activity. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  equilibrate(aout, balance = NULL, loga.balance = NULL,
    ispecies = !grepl("cr", aout$species$state), normalize = FALSE,
    as.residue = FALSE, method = c("boltzmann", "reaction"),
    tol = .Machine$double.eps^0.25)
  equil.boltzmann(Astar, n.balance, loga.balance)
  equil.reaction(Astar, n.balance, loga.balance, tol = .Machine$double.eps^0.25)
  moles(eout)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="equilibrate_+3A_aout">aout</code></td>
<td>
<p>list, output from <code><a href="#topic+affinity">affinity</a></code></p>
</td></tr></table>
<p> or <code><a href="#topic+mosaic">mosaic</a></code>
</p>
<table role = "presentation">
<tr><td><code id="equilibrate_+3A_balance">balance</code></td>
<td>
<p>character or numeric, how to balance the transformations</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_ispecies">ispecies</code></td>
<td>
<p>numeric, which species to include</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_normalize">normalize</code></td>
<td>
<p>logical, normalize the molar formulas of species by the balancing coefficients?</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_as.residue">as.residue</code></td>
<td>
<p>logical, report results for the normalized formulas?</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_astar">Astar</code></td>
<td>
<p>numeric, affinities of formation reactions excluding species contribution</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_n.balance">n.balance</code></td>
<td>
<p>numeric, number of moles of balancing component in the formation reactions of the species of interest</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_loga.balance">loga.balance</code></td>
<td>
<p>numeric (single value or vector), logarithm of total activity of balanced quantity</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_method">method</code></td>
<td>
<p>character, equilibration method to use</p>
</td></tr>
<tr><td><code id="equilibrate_+3A_tol">tol</code></td>
<td>
<p>numeric, convergence tolerance for <code><a href="stats.html#topic+uniroot">uniroot</a></code></p>
</td></tr>
<tr><td><code id="equilibrate_+3A_eout">eout</code></td>
<td>
<p>list, output from <code><a href="#topic+equilibrate">equilibrate</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>equilibrate</code> calculates the chemical activities of species in metastable equilibrium, for constant temperature, pressure and chemical activities of basis species, using specified balancing constraints on reactions between species.
</p>
<p>It takes as input <code>aout</code>, the output from <code><a href="#topic+affinity">affinity</a></code>, giving the chemical affinities of formation reaction of each species, which may be calculated on a multidimensional grid of conditions.
Alternatively, <code>aout</code> can be the output from <code><a href="#topic+mosaic">mosaic</a></code>, in which case the equilibrium activities of the formed species are calculated and combined with those of the changing basis species to make an object that can be plotted with <code><a href="#topic+diagram">diagram</a></code>.
</p>
<p>The equilibrium chemical activities of species are calculated using either the <code>equil.reaction</code> or <code>equil.boltzmann</code> functions, the latter only if the balance is on one mole of species.
</p>
<p><code>equilibrate</code> needs to be provided constraints on how to balance the reactions representing transformations between the species.
<code>balance</code> indicates the balancing component, according to the following scheme:
</p>

<ul>
<li> <p>&lsquo;<span class="samp">&#8288;NULL&#8288;</span>&rsquo;: autoselect
</p>
</li>
<li> <p><em>name of basis species</em>: balance on this basis species
</p>
</li>
<li> <p>&lsquo;<span class="samp">&#8288;length&#8288;</span>&rsquo;: balance on length of proteins
</p>
</li>
<li> <p>&lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;: balance on one mole of species
</p>
</li>
<li> <p><em>numeric vector</em>: user-defined constraints
</p>
</li></ul>

<p>The default value of NULL for <code>balance</code> indicates to use the coefficients on the basis species that is present (i.e. with non-zero coefficients) in all formation reactions, or if that fails, to set the balance to &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;.
However, if all the species (as listed in code <code>aout$species</code>) are proteins (have an underscore character in their names), the default value of NULL for <code>balance</code> indicates to use &lsquo;<span class="samp">&#8288;length&#8288;</span>&rsquo; as the balance.
</p>
<p>NOTE: The summation of activities assumes an ideal system, so &lsquo;molality&rsquo; is equivalent to &lsquo;activity&rsquo; here.
<code>loga.balance</code> gives the logarithm of the total activity of <code>balance</code> (which is total activity of species for &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo; or total activity of amino acid residue-equivalents for &lsquo;<span class="samp">&#8288;length&#8288;</span>&rsquo;).
If <code>loga.balance</code> is missing, its value is taken from the activities of species listed in <code>aout</code>; this default is usually the desired operation.
The supplied value of <code>loga.balance</code> may also be a vector of values, with length corresponding to the number of conditions in the calculations of <code><a href="#topic+affinity">affinity</a></code>.
</p>
<p><code>normalize</code> if TRUE indicates to normalize the molar formulas of species by the balance coefficients.
This operation is intended for systems of proteins, whose conventional formulas are much larger than the basis speices.
The normalization also applies to the balancing coefficients, which as a result consist of &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;s.
After normalization and equilibration, the equilibrium activities are then re-scaled (for the original formulas of the species), unless <code>as.residue</code> is TRUE.
</p>
<p><code>equil.boltzmann</code> is used to calculate the equilibrium activities if <code>balance</code> is &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo; (or when <code>normalize</code> or <code>as.residue</code> is TRUE), otherwise <code>equil.reaction</code> is called.
The default behavior can be overriden by specifying either &lsquo;<span class="samp">&#8288;boltzmann&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;reaction&#8288;</span>&rsquo; in <code>method</code>.
Using <code>equil.reaction</code> may be needed for systems with huge (negative or positive) affinities, where <code>equil.boltzmann</code> produces a NaN result.
</p>
<p><code>ispecies</code> can be supplied to identify a subset of the species to include in the equilibrium calculation.
By default, this is all species except solids (species with &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; state).
However, the stability regions of solids are still calculated (by a call to <code><a href="#topic+diagram">diagram</a></code> without plotting).
At all points outside of their stability region, the logarithms of activities of solids are set to -999.
Likewise, where any solid species is calculated to be stable, the logarithms of activities of all aqueous species are set to -999.
</p>
<p><code>moles</code> simply calculates the total number of moles of elements corresponding to the activities of formed species in the output from <code>equilibrate</code>.
</p>


<h3>Value</h3>

<p><code>equil.reaction</code> and <code>equil.boltzmann</code> each return a list with dimensions and length equal to those of <code>Astar</code>, giving the <code><a href="base.html#topic+log10">log10</a></code> of the equilibrium activities of the species of interest.
<code>equilibrate</code> returns a list, containing first the values in <code>aout</code>, to which are appended <code>m.balance</code> (the balancing coefficients if <code>normalize</code> is TRUE, a vector of &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;s otherwise), <code>n.balance</code> (the balancing coefficients if <code>normalize</code> is FALSE, a vector of &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;s otherwise), <code>loga.balance</code>, <code>Astar</code>, and <code>loga.equil</code> (the calculated equilibrium activities of the species).
</p>


<h3>Algorithms</h3>

<p>The input values to <code>equil.reaction</code> and <code>equil.boltzmann</code> are in a list, <code>Astar</code>, all elements of the list having the same dimensions; they can be vectors, matrices, or higher-dimensionsal arrays.
Each list element contains the chemical affinities of the formation reactions of one of the species of interest (in dimensionless base-10 units, i.e. A/2.303RT), calculated at unit activity of the species of interest.
The equilibrium base-10 logarithm activities of the species of interest returned by either function satisfy the constraints that 1) the final chemical affinities of the formation reactions of the species are all equal and 2) the total activity of the balancing component is equal to (<code>loga.balance</code>). 
The first constraint does <em>not</em> impose a complete equilibrium, where the affinities of the formation reactions are all equal to zero, but allows for a metastable equilibrium, where the affinities of the formation reactions are equal to each other.
</p>
<p>In <code>equil.reaction</code> (the algorithm described in Dick, 2008 and the only one available prior to CHNOSZ_0.8), the calculations of relative abundances of species are based on a solving a system of equations representing the two constraints stated above.
The solution is found using <code><a href="stats.html#topic+uniroot">uniroot</a></code> with a flexible method for generating initial guesses.
</p>
<p>In <code>equil.boltzmann</code>, the chemical activities of species are calculated using the Boltzmann distribution.
This calculation is faster than the algorithm of <code>equil.reaction</code>, but is limited to systems where the transformations are all balanced on one mole of species.
If <code>equil.boltzmann</code> is called with <code>balance</code> other than &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;, it stops with an error.
</p>


<h3>Warning</h3>

<p>Despite its name, this function does not generally produce a complete equilibrium.
It returns activities of species such that the affinities of formation reactions are equal to each other (and transformations between species have zero affinity); this is a type of metastable equilibrium.
Although they are equal to each other, the affinities are not necessarily equal to zero.
Use <code>solubility</code> to find complete equilibrium, where the affinities of the formation reactions become zero.
</p>


<h3>References</h3>

<p>Dick, J. M. (2008) Calculation of the relative metastabilities of proteins using the CHNOSZ software package. <em>Geochem. Trans.</em> <b>9</b>:10. <a href="https://doi.org/10.1186/1467-4866-9-10">doi:10.1186/1467-4866-9-10</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diagram">diagram</a></code> has examples of using <code>equilibrate</code> to make equilibrium activity diagrams. 
<code><a href="#topic+palply">palply</a></code> is used by both <code>equil.reaction</code> and <code>equil.boltzmann</code> to parallelize intensive parts of the calculations.
</p>
<p>See the vignette <a href="../doc/multi-metal.html"><strong>multi-metal</strong></a> for an example of balancing on two elements (N in the basis species, C in the formed species).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Equilibrium in a simple system:
## ionization of carbonic acid
basis("CHNOS+")
species(c("CO2", "HCO3-", "CO3-2"))
# Set unit activity of the species (0 = log10(1))
species(1:3, 0)
# Calculate Astar (for unit activity)
res &lt;- 100
Astar &lt;- affinity(pH = c(0, 14, res))$values
# The logarithms of activity for a total activity
# of the balancing component (CO2) equal to 0.001
loga.boltz &lt;- equil.boltzmann(Astar, c(1, 1, 1), 0.001)
# Calculated another way
loga.react &lt;- equil.reaction(Astar, c(1, 1, 1), rep(0.001, 100))
# They should be pretty close
stopifnot(all.equal(loga.boltz, loga.react))
# The first ionization constant (pKa)
ipKa &lt;- which.min(abs(loga.boltz[[1]] - loga.boltz[[2]]))
pKa.equil &lt;- seq(0, 14, length.out = res)[ipKa]
# Calculate logK directly
logK &lt;- subcrt(c("CO2","H2O","HCO3-","H+"), c(-1, -1, 1, 1), T = 25)$out$logK
# We could decrease tolerance here by increasing res
stopifnot(all.equal(pKa.equil, -logK, tolerance = 1e-2))
</code></pre>

<hr>
<h2 id='examples'>Run Examples from the Documentation</h2><span id='topic+examples'></span><span id='topic+demos'></span>

<h3>Description</h3>

<p>Run the examples contained in each of the documentation topics. </p>


<h3>Usage</h3>

<pre><code class='language-R'>  examples(save.png = FALSE)
  demos(which = c("sources", "protein.equil", "affinity", "NaCl",
    "density", "ORP", "ionize", "buffer", "protbuff",
    "glycinate", "mosaic", "copper", "arsenic", "solubility", "gold",
    "contour", "sphalerite", "minsol", "Shh", "saturation",
    "adenine", "DEW", "lambda", "potassium", "TCA", "aluminum", "AD",
    "comproportionation", "Pourbaix", "E_coli", "yttrium", "rank.affinity"),
    save.png = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="examples_+3A_save.png">save.png</code></td>
<td>
<p>logical, generate PNG image files for the plots?</p>
</td></tr>
<tr><td><code id="examples_+3A_which">which</code></td>
<td>
<p>character, which example to run</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>examples</code> runs all the examples in the help pages for the package. 
<code><a href="utils.html#topic+example">example</a></code> is called for each topic with <code>ask</code> set to <code>FALSE</code> (so all of the figures are shown without prompting the user).
</p>
<p><code>demos</code> runs all the <code><a href="utils.html#topic+demo">demo</a>s</code> in the package.
The demo(s) to run is/are specified by <code>which</code>; the default is to run them in the order of the list below.
</p>

<dl>
<dt>sources</dt><dd><p>Cross-check the reference list with the thermodynamic database</p>
</dd>
<dt>protein.equil</dt><dd><p>Chemical activities of two proteins in metastable equilibrium (Dick and Shock, 2011)</p>
</dd>
<dt>affinity</dt><dd><p>Affinities of metabolic reactions and amino acid synthesis (Amend and Shock, 1998, 2001)</p>
</dd>
<dt>NaCl</dt><dd><p>Equilibrium constant for aqueous NaCl dissociation (Shock et al., 1992)</p>
</dd>
<dt>density</dt><dd><p>Density of H<sub>2</sub>O, inverted from IAPWS-95 equations (<code><a href="#topic+rho.IAPWS95">rho.IAPWS95</a></code>)</p>
</dd>
<dt>ORP</dt><dd><p>Temperature dependence of oxidation-reduction potential for redox standards</p>
</dd>
<dt>ionize</dt><dd><p>ionize.aa(): contour plots of net charge and ionization properties of LYSC_CHICK</p>
</dd>
<dt>buffer</dt><dd><p>Minerals and aqueous species as buffers of hydrogen fugacity (Schulte and Shock, 1995)</p>
</dd>
<dt>protbuff</dt><dd><p>Chemical activities buffered by thiol peroxidases or sigma factors</p>
</dd>
<dt>glycinate</dt><dd><p>Metal-glycinate complexes (Shock and Koretsky, 1995; Azadi et al., 2019)</p>
</dd>
<dt>mosaic</dt><dd><p>Eh-pH diagram with two sets of changing basis species (Garrels and Christ, 1965)</p>
</dd>
<dt>copper</dt><dd><p>Another example of <code><a href="#topic+mosaic">mosaic</a></code>: complexation of Cu with glycine (Aksu and Doyle, 2001)</p>
</dd>
<dt>arsenic</dt><dd><p>Another example of <code><a href="#topic+mosaic">mosaic</a></code>: Eh-pH diagram for the system As-O-H-S (Lu and Zhu, 2011)</p>
</dd>
<dt>solubility</dt><dd><p>Solubility of calcite (cf. Manning et al., 2013) and CO<sub>2</sub> (cf. Stumm and Morgan, 1996)</p>
</dd>
<dt>gold</dt><dd><p>Solubility of gold (Akinfiev and Zotov; 2001; Stef&aacute;nsson and Seward, 2004; Williams-Jones et al., 2009)</p>
</dd>
<dt>contour</dt><dd><p>Gold solubility contours on a log fO2 - pH diagram (Williams-Jones et al., 2009)</p>
</dd>
<dt>sphalerite</dt><dd><p>Solubility of sphalerite (Akinfiev and Tagirov, 2014)</p>
</dd>
<dt>minsol</dt><dd><p>Solubilities of multiple minerals</p>
</dd>
<dt>dehydration</dt><dd>log&thinsp;<i>K</i><p> of dehydration reactions; SVG file contains tooltips and links</p>
</dd>
<dt>Shh</dt><dd><p>Affinities of transcription factors relative to Sonic hedgehog (Dick, 2015)</p>
</dd>
<dt>saturation</dt><dd><p>Equilibrium activity diagram showing activity ratios and mineral saturation limits (Bowers et al., 1984)</p>
</dd>
<dt>adenine</dt><dd><p>HKF regression of heat capacity and volume of aqueous adenine (Lowe et al., 2017)</p>
</dd>
<dt>DEW</dt><dd><p>Deep Earth Water (DEW) model for high pressures (Sverjensky et al., 2014a and 2014b)</p>
</dd>
<dt>lambda</dt><dd><p>Effects of lambda transition on thermodynamic properties of quartz (Berman, 1988)</p>
</dd>
<dt>potassium</dt><dd><p>Comparison of thermodynamic datasets for predicting mineral stabilities (Sverjensky et al., 1991)</p>
</dd>
<dt>TCA</dt><dd><p>Standard Gibbs energies of the tricarboxylic (citric) acid cycle (Canovas and Shock, 2016)</p>
</dd>
<dt>aluminum</dt><dd><p>Reactions involving Al-bearing minerals (Zimmer et al., 2016; Tutolo et al., 2014)</p>
</dd>
<dt>carboxylase</dt><dd><p>Rank abundance distribution for RuBisCO and acetyl-CoA carboxylase</p>
</dd>
<dt>AD</dt><dd><p>Dissolved gases: Henry's constant, volume, and heat capacity (Akinfiev and Diamond, 2003)</p>
</dd>
<dt>comproportionation</dt><dd><p>Gibbs energy of sulfur comproportionation (Amend et al., 2020)</p>
</dd>
<dt>Pourbaix</dt><dd><p>Eh-pH diagram for Fe-O-H with equisolubility lines (Pourbaix, 1974)</p>
</dd>
<dt>E_coli</dt><dd><p>Gibbs energy of biomass synthesis in <em>E. coli</em> (LaRowe and Amend, 2016)</p>
</dd>
<dt>rank.affinity</dt><dd><p>Affinity ranking for proteins in yeast nutrient limitation (data from Tai et al., 2005)</p>
</dd>
<dt>yttrium</dt><dd><p><code><a href="#topic+logB.to.OBIGT">logB.to.OBIGT</a></code> fits at 800 and 1000 bar and Y speciation in <code><a href="#topic+NaCl">NaCl</a></code> solution at varying pH (Guan et al., 2020)</p>
</dd>
</dl>

<p>For either function, if <code>save.png</code> is TRUE, the plots are saved in <code><a href="grDevices.html#topic+png">png</a></code> files whose names begin with the names of the help topics or demos.
</p>
<p>Two of the demos have external dependencies and are not automatically run by <code>demos</code>.
&lsquo;<span class="samp">&#8288;dehydration&#8288;</span>&rsquo; creates an interactive SVG file; this demo depends on <a href="https://CRAN.R-project.org/package=RSVGTipsDevice"><span class="pkg">RSVGTipsDevice</span></a>, which is not available for Windows.
&lsquo;<span class="samp">&#8288;carboxylase&#8288;</span>&rsquo; creates an animated GIF; this demo requires that the ImageMagick <code>convert</code> commmand be available on the system (tested on Linux and Windows).
</p>
<p>&lsquo;<span class="samp">&#8288;carboxylase&#8288;</span>&rsquo; animates diagrams showing rankings of calculated chemical activities along a combined <I>T</I> and log<i>a</i><sub>H<sub>2</sub></sub> gradient, or makes a single plot on the default device (without conversion to animated GIF) if a single temperature (<code>T</code>) is specified in the code.
To run this demo, an empty directory named &lsquo;<span class="samp">&#8288;png&#8288;</span>&rsquo; must be present (as a subdirectory of the R working directory).
The proteins in the calculation are 24 carboxylases from a variety of organisms.
There are 12 ribulose phosphate carboxylase and 12 acetyl-coenzyme A carboxylase; 6 of each type are from nominally mesophilic organisms and 6 from nominally thermophilic organisms, shown as blue and red symbols on the diagrams.
The activities of hydrogen at each temperature are calculated using <code class="reqn">\log a_{\mathrm{H_{2}}_{\left(aq\right)}}=-11+3/\left(40\times T\left(^{\circ}C\right)\right)</code>; this equation comes from a model of relative stabilities of proteins in a hot-spring environment (Dick and Shock, 2011).
</p>


<h3>Warning</h3>

<p>The discontinuities apparent in the plot made by the <code>NaCl</code> demo illustrate limitations of the &quot;<em>g</em> function&quot; for charged species in the revised HKF model (the 355 &deg;C boundary of region II in Figure 6 of Shock et al., 1992).
Note that <abbr><span class="acronym">SUPCRT92</span></abbr> (Johnson et al., 1992) gives similar output at 500 bar.
However, <abbr><span class="acronym">SUPCRT</span></abbr> does not output thermodynamic properties above 350 &deg;C at <i>P</i><sub>SAT</sub>; see Warning in <code><a href="#topic+subcrt">subcrt</a></code>.
</p>


<h3>References</h3>

<p>Akinfiev, N. N. and Diamond, L. W. (2003) Thermodynamic description of aqueous nonelectrolytes at infinite dilution over a wide range of state parameters. <em>Geochim. Cosmochim. Acta</em> <b>67</b>, 613&ndash;629. <a href="https://doi.org/10.1016/S0016-7037%2802%2901141-9">doi:10.1016/S0016-7037(02)01141-9</a>
</p>
<p>Akinfiev, N. N. and Tagirov, B. R. (2014) Zn in hydrothermal systems: Thermodynamic description of hydroxide, chloride, and hydrosulfide complexes. <em>Geochem. Int.</em> <b>52</b>, 197&ndash;214. <a href="https://doi.org/10.1134/S0016702914030021">doi:10.1134/S0016702914030021</a>
</p>
<p>Akinfiev, N. N. and Zotov, A. V. (2001) Thermodynamic description of chloride, hydrosulfide, and hydroxo complexes of Ag(I), Cu(I), and Au(I) at temperatures of 25-500&deg;C and pressures of 1-2000 bar. <em>Geochem. Int.</em> <b>39</b>, 990&ndash;1006.
</p>
<p>Aksu, S. and Doyle, F. M. (2001) Electrochemistry of copper in aqueous glycine solutions. <em>J. Electrochem. Soc.</em> <b>148</b>, B51&ndash;B57.
</p>
<p>Amend, J. P. and Shock, E. L. (1998) Energetics of amino acid synthesis in hydrothermal ecosystems. <em>Science</em> <b>281</b>, 1659&ndash;1662. <a href="https://doi.org/10.1126/science.281.5383.1659">doi:10.1126/science.281.5383.1659</a>
</p>
<p>Amend, J. P. and Shock, E. L. (2001) Energetics of overall metabolic reactions of thermophilic and hyperthermophilic Archaea and Bacteria. <em>FEMS Microbiol. Rev.</em> <b>25</b>, 175&ndash;243. <a href="https://doi.org/10.1016/S0168-6445%2800%2900062-0">doi:10.1016/S0168-6445(00)00062-0</a>
</p>
<p>Amend, J. P., Aronson, H. S., Macalady, J. and LaRowe, D. E. (2020) Another chemolithotrophic metabolism missing in nature: sulfur comproportionation. <em>Environ. Microbiol.</em> <b>22</b>, 1971&ndash;1976. <a href="https://doi.org/10.1111/1462-2920.14982">doi:10.1111/1462-2920.14982</a>
</p>
<p>Azadi, M. R., Karrech, A., Attar, M. and Elchalakani, M. (2019) Data analysis and estimation of thermodynamic properties of aqueous monovalent metal-glycinate complexes. <em>Fluid Phase Equilib.</em> <b>480</b>, 25-40. <a href="https://doi.org/10.1016/j.fluid.2018.10.002">doi:10.1016/j.fluid.2018.10.002</a>
</p>
<p>Berman, R. G. (1988) Internally-consistent thermodynamic data for minerals in the system Na<sub>2</sub>O-K<sub>2</sub>O-CaO-MgO-FeO-Fe<sub>2</sub>O<sub>3</sub>-Al<sub>2</sub>O<sub>3</sub>-SiO<sub>2</sub>-TiO<sub>2</sub>-H<sub>2</sub>O-CO<sub>2</sub>. <em>J. Petrol.</em> <b>29</b>, 445-522. <a href="https://doi.org/10.1093/petrology/29.2.445">doi:10.1093/petrology/29.2.445</a>
</p>
<p>Bowers, T. S., Jackson, K. J. and Helgeson, H. C. (1984) <em>Equilibrium Activity Diagrams for Coexisting Minerals and Aqueous Solutions at Pressures and Temperatures to 5 kb and 600&deg;C</em>, Springer-Verlag, Berlin, 397 p. <a href="https://www.worldcat.org/oclc/11133620">https://www.worldcat.org/oclc/11133620</a>
</p>
<p>Canovas, P. A., III and Shock, E. L. (2016) Geobiochemistry of metabolism: Standard state thermodynamic properties of the citric acid cycle. <em>Geochim. Cosmochim. Acta</em> <b>195</b>, 293&ndash;322. <a href="https://doi.org/10.1016/j.gca.2016.08.028">doi:10.1016/j.gca.2016.08.028</a>
</p>
<p>Dick, J. M. and Shock, E. L. (2011) Calculation of the relative chemical stabilities of proteins as a function of temperature and redox chemistry in a hot spring. <em>PLOS One</em> <b>6</b>, e22782. <a href="https://doi.org/10.1371/journal.pone.0022782">doi:10.1371/journal.pone.0022782</a>
</p>
<p>Dick, J. M. (2015) Chemical integration of proteins in signaling and development. <em>bioRxiv</em>. <a href="https://doi.org/10.1101/015826">doi:10.1101/015826</a>
</p>
<p>Garrels, R. M. and Christ, C. L. (1965) <em>Solutions, Minerals, and Equilibria</em>, Harper &amp; Row, New York, 450 p. <a href="https://www.worldcat.org/oclc/517586">https://www.worldcat.org/oclc/517586</a>
</p>
<p>Guan, Q., Mei, Y., Etschmann, B., Testemale, D., Louvel, M. and Brugger, J. (2020) Yttrium complexation and hydration in chloride-rich hydrothermal fluids: A combined <em>ab initio</em> molecular dynamics and <em>in situ</em> X-ray absorption spectroscopy study. <em>Geochim. Cosmochim. Acta</em> <b>281</b>, 168&ndash;189. <a href="https://doi.org/10.1016/j.gca.2020.04.015">doi:10.1016/j.gca.2020.04.015</a>
</p>
<p>Johnson, J. W., Oelkers, E. H. and Helgeson, H. C. (1992) SUPCRT92: A software package for calculating the standard molal thermodynamic properties of minerals, gases, aqueous species, and reactions from 1 to 5000 bar and 0 to 1000&deg;C. <em>Comp. Geosci.</em> <b>18</b>, 899&ndash;947. <a href="https://doi.org/10.1016/0098-3004%2892%2990029-Q">doi:10.1016/0098-3004(92)90029-Q</a>
</p>
<p>LaRowe, D. E. and Amend, J. P. (2016) The energetics of anabolism in natural settings. <em>ISME J.</em> <b>10</b>, 1285&ndash;1295. <a href="https://doi.org/10.1038/ismej.2015.227">doi:10.1038/ismej.2015.227</a>
</p>
<p>Lowe, A. R., Cox, J. S. and Tremaine, P. R. (2017) Thermodynamics of aqueous adenine: Standard partial molar volumes and heat capacities of adenine, adeninium chloride, and sodium adeninate from <em>T</em> = 278.15 K to 393.15 K. <em>J. Chem. Thermodyn.</em> <b>112</b>, 129&ndash;145. <a href="https://doi.org/10.1016/j.jct.2017.04.005">doi:10.1016/j.jct.2017.04.005</a>
</p>
<p>Lu, P. and Zhu, C. (2011) Arsenic Eh&ndash;pH diagrams at 25&deg;C and 1 bar. <em>Environ. Earth Sci.</em> <b>62</b>, 1673&ndash;1683. <a href="https://doi.org/10.1007/s12665-010-0652-x">doi:10.1007/s12665-010-0652-x</a>
</p>
<p>Manning, C. E., Shock, E. L. and Sverjensky, D. A. (2013) The chemistry of carbon in aqueous fluids at crustal and upper-mantle conditions: Experimental and theoretical constraints. <em>Rev. Mineral. Geochem.</em> <b>75</b>, 109&ndash;148. <a href="https://doi.org/10.2138/rmg.2013.75.5">doi:10.2138/rmg.2013.75.5</a>
</p>
<p>Pourbaix, M. (1974) <em>Atlas of Electrochemical Equilibria in Aqueous Solutions</em>, NACE, Houston, TX and CEBELCOR, Brussels. <a href="https://www.worldcat.org/oclc/563921897">https://www.worldcat.org/oclc/563921897</a>
</p>
<p>Schulte, M. D. and Shock, E. L. (1995) Thermodynamics of Strecker synthesis in hydrothermal systems. <em>Orig. Life Evol. Biosph.</em> <b>25</b>, 161&ndash;173. <a href="https://doi.org/10.1007/BF01581580">doi:10.1007/BF01581580</a>
</p>
<p>Shock, E. L. and Koretsky, C. M. (1995) Metal-organic complexes in geochemical processes: Estimation of standard partial molal thermodynamic properties of aqueous complexes between metal cations and monovalent organic acid ligands at high pressures and temperatures. <em>Geochim. Cosmochim. Acta</em> <b>59</b>, 1497&ndash;1532. <a href="https://doi.org/10.1016/0016-7037%2895%2900058-8">doi:10.1016/0016-7037(95)00058-8</a>
</p>
<p>Shock, E. L., Oelkers, E. H., Johnson, J. W., Sverjensky, D. A. and Helgeson, H. C. (1992) Calculation of the thermodynamic properties of aqueous species at high pressures and temperatures: Effective electrostatic radii, dissociation constants and standard partial molal properties to 1000 &deg;C and 5 kbar. <em>J. Chem. Soc. Faraday Trans.</em> <b>88</b>, 803&ndash;826. <a href="https://doi.org/10.1039/FT9928800803">doi:10.1039/FT9928800803</a>
</p>
<p>Stef&aacute;nsson, A. and Seward, T. M. (2004) Gold(I) complexing in aqueous sulphide solutions to 500&deg;C at 500 bar. <em>Geochim. Cosmochim. Acta</em> <b>68</b>, 4121&ndash;4143. <a href="https://doi.org/10.1016/j.gca.2004.04.006">doi:10.1016/j.gca.2004.04.006</a>
</p>
<p>Stumm, W. and Morgan, J. J. (1996) <em>Aquatic Chemistry: Chemical Equilibria and Rates in Natural Waters</em>, John Wiley &amp; Sons, New York, 1040 p. <a href="https://www.worldcat.org/oclc/31754493">https://www.worldcat.org/oclc/31754493</a>
</p>
<p>Sverjensky, D. A., Harrison, B. and Azzolini, D. (2014a) Water in the deep Earth: The dielectric constant and the solubilities of quartz and corundum to 60 kb and 1,200 &deg;C. <em>Geochim. Cosmochim. Acta</em> <b>129</b>, 125&ndash;145. <a href="https://doi.org/10.1016/j.gca.2013.12.019">doi:10.1016/j.gca.2013.12.019</a>
</p>
<p>Sverjensky, D. A., Hemley, J. J. and D'Angelo, W. M. (1991) Thermodynamic assessment of hydrothermal alkali feldspar-mica-aluminosilicate equilibria. <em>Geochim. Cosmochim. Acta</em> <b>55</b>, 989-1004. <a href="https://doi.org/10.1016/0016-7037%2891%2990157-Z">doi:10.1016/0016-7037(91)90157-Z</a>
</p>
<p>Sverjensky, D. A., Stagno, V. and Huang, F. (2014b) Important role for organic carbon in subduction-zone fluids in the deep carbon cycle. <em>Nat. Geosci.</em> <b>7</b>, 909&ndash;913. <a href="https://doi.org/10.1038/ngeo2291">doi:10.1038/ngeo2291</a>
</p>
<p>Tai, S. L., Boer, V. M., Daran-Lapujade, P., Walsh, M. C., de Winde, J. H., Daran, J.-M. and Pronk, J. T. (2005) Two-dimensional transcriptome analysis in chemostat cultures: Combinatorial effects of oxygen availability and macronutrient limitation in <em>Saccharomyces cerevisiae</em>. <em>J. Biol. Chem.</em> <b>280</b>, 437&ndash;447. <a href="https://doi.org/10.1074/jbc.M410573200">doi:10.1074/jbc.M410573200</a>
</p>
<p>Tutolo, B. M., Kong, X.-Z., Seyfried, W. E., Jr. and Saar, M. O. (2014) Internal consistency in aqueous geochemical data revisited: Applications to the aluminum system. <em>Geochim. Cosmochim. Acta</em> <b>133</b>, 216&ndash;234. <a href="https://doi.org/10.1016/j.gca.2014.02.036">doi:10.1016/j.gca.2014.02.036</a>
</p>
<p>Williams-Jones, A. E., Bowell, R. J. and Migdisov, A. A. (2009) Gold in solution. <em>Elements</em> <b>5</b>, 281&ndash;287. <a href="https://doi.org/10.2113/gselements.5.5.281">doi:10.2113/gselements.5.5.281</a>
</p>
<p>Zimmer, K., Zhang, Y., Lu, P., Chen, Y., Zhang, G., Dalkilic, M. and Zhu, C. (2016) SUPCRTBL: A revised and extended thermodynamic dataset and software package of SUPCRT92. <em>Comp. Geosci.</em> <b>90</b>, 97&ndash;111. <a href="https://doi.org/10.1016/j.cageo.2016.02.013">doi:10.1016/j.cageo.2016.02.013</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

demos(c("ORP", "NaCl"))

</code></pre>

<hr>
<h2 id='extdata'>Extra Data</h2><span id='topic+extdata'></span>

<h3>Description</h3>

<p>The files in the subdirectories of <code>extdata</code> provide additional thermodynamic data and other data to support the examples in the package documentation and vignettes. 
See <code><a href="#topic+thermo">thermo</a></code> for a description of the files in <code>extdata/OBIGT</code>, which are used to generate the thermodynamic database.
</p>


<h3>Details</h3>

<p>Files in <code>Berman</code> contain thermodynamic data for minerals using the Berman formulation:
</p>

<ul>
<li> <p><code>Ber88_1988.csv</code> contains thermodynamic data for minerals taken from Berman (1988).
</p>
</li>
<li><p> Other files with names like <code>xxx_yyyy.csv</code> contain thermodynamic data from other sources; xxx in the filename corresponds to the reference in <code><a href="#topic+thermo">thermo</a>$OBIGT</code> and yyyy gives the year of publication.
<code><a href="#topic+Berman">Berman</a></code> uses these data for the calculation of thermodynamic properties at specified <I>P</I> and <I>T</I>, which are then available for use in <code><a href="#topic+subcrt">subcrt</a></code>.
If there are any duplicated mineral names in the files, only the most recent data are used, as determined by the year in the file name.
Following conventions used SUPCRT92 (see Helgeson et al., 1978), the names of sanidine and microcline were changed to K-feldspar,high and K-feldspar,low (by using the same names in all data files, loading the optional SUPCRT92 data file updates these minerals rather than makes new ones).
</p>
</li>
<li> <p><code>sympy.R</code> is an R script that uses <a href="https://CRAN.R-project.org/package=rSymPy"><span class="pkg">rSymPy</span></a> to symbolically integrate Bermans's equations for heat capacity and volume to write experessions for enthalpy, entropy and Gibbs energy.
</p>
</li>
<li><p> The <code>testing</code> directory contains data files based on Berman and Aranovich (1996). These are used to demonstrate the addition of data from a user-supplied file (see <code><a href="#topic+Berman">Berman</a></code>).
</p>
</li></ul>

<p>Files in <code>cpetc</code> contain experimental and calculated thermodynamic and environmental data:
</p>

<ul>
<li> <p><code>PM90.csv</code> Heat capacities of four unfolded aqueous proteins taken from Privalov and Makhatadze, 1990. Temperature in &deg;C is in the first column, and heat capacities of the proteins in J mol<code class="reqn">^{-1}</code> K<code class="reqn">^{-1}</code> in the remaining columns. See <code><a href="#topic+ionize.aa">ionize.aa</a></code> and the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> for examples that use this file.
</p>
</li>
<li> <p><code>RH95.csv</code> Heat capacity data for iron taken from Robie and Hemingway, 1995. Temperature in Kelvin is in the first column, heat capacity in J K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> in the second. See <code><a href="#topic+subcrt">subcrt</a></code> for an example that uses this file.
</p>
</li>
<li> <p><code>SOJSH.csv</code> Experimental equilibrium constants for the reaction NaCl(aq) = Na+ + Cl- as a function of temperature and pressure taken from Fig. 1 of Shock et al., 1992. See <code>demo("NaCl")</code> for an example that uses this file.
</p>
</li>
<li> <p><code>HWM96_V.csv</code>, <code>HW97_Cp.csv</code> Apparent molar volumes and heat capacities of CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S, and NH<sub>3</sub> in dilute aqueous solutions reported by Hnědkovský et al., 1996 and Hnědkovský and Wood, 1997. Units are Kelvin, MPa, J/K/mol, and cm3/mol. See <code>demo("AD")</code>, <code><a href="#topic+EOSregress">EOSregress</a></code> and the vignette <a href="../doc/eos-regress.html"><strong>eos-regress</strong></a> for examples that use these files.
</p>
</li>
<li> <p><code>SC10_Rainbow.csv</code> Values of temperature (&deg;C), pH and logarithms of activity of CO<sub>2</sub>, H<sub>2</sub>, NH<sub>4</sub><sup>+</sup>, H<sub>2</sub>S and CH<sub>4</sub> for mixing of seawater and hydrothermal fluid at Rainbow field (Mid-Atlantic Ridge), taken from Shock and Canovas, 2010. See the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> for an example that uses this file.
</p>
</li>
<li> <p><code>SS98_Fig5a.csv</code>, <code>SS98_Fig5b.csv</code> Values of logarithm of fugacity of O<sub>2</sub> and pH as a function of temperature for mixing of seawater and hydrothermal fluid, digitized from Figs. 5a and b of Shock and Schulte, 1998. See the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> for an example that uses this file.
</p>
</li>
<li> <p><code>rubisco.csv</code> UniProt IDs for Rubisco, ranges of optimal growth temperature of organisms, domain and name of organisms, and URL of reference for growth temperature, from Dick, 2014. See <code><a href="#topic+rank.affinity">rank.affinity</a></code> and the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> for examples that use this file.
</p>
</li>
<li> <p><code>bluered.txt</code> Blue - light grey - red color palette, computed using <a href="https://CRAN.R-project.org/package=colorspace"><span class="pkg">colorspace</span></a><code>::diverge_hcl(1000,</code> <code>c = 100, l = c(50, 90), power = 1)</code>. This is used by <code><a href="#topic+ZC.col">ZC.col</a></code>.
</p>
</li>
<li> <p><code>AD03_Fig1?.csv</code> Experimental data points digitized from Figure 1 of Akinfiev and Diamond, 2003, used in <code>demo("AD")</code>.
</p>
</li>
<li> <p><code>TKSS14_Fig2.csv</code> Experimental data points digitized from Figure 2 of Tutolo et al., 2014, used in <code>demo("aluminum")</code>.
</p>
</li>
<li> <p><code>Mer75_Table4.csv</code> Values of log(aK+/aH+) and log(aNa+/aH+) from Table 4 of Merino, 1975, used in <code>demo("aluminum")</code>.
</p>
</li></ul>

<p>Files in <code>protein</code> contain protein sequences and amino acid compositions for proteins.
</p>

<ul>
<li> <p><code>EF-Tu.aln</code> consists of aligned sequences (394 amino acids) of elongation factor Tu (EF-Tu). The sequences correspond to those taken from UniProtKB for ECOLI (<em>Escherichia coli</em>), THETH (<em>Thermus thermophilus</em>) and THEMA (<em>Thermotoga maritima</em>), and reconstructed ancestral sequences taken from Gaucher et al., 2003 (maximum likelihood bacterial stem and mesophilic bacterial stem, and alternative bacterial stem). See <code><a href="#topic+read.fasta">read.fasta</a></code> for an example that uses this file.
</p>
</li>
<li> <p><code>rubisco.fasta</code> Sequences of Rubisco obtained from UniProt (see Dick, 2014). See the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a> for an example that uses this file.
</p>
</li>
<li> <p><code>POLG.csv</code>
Amino acid compositions of a few proteins used for some tests and examples.
These are various subunits of the Poliovirus type 1 polyprotein (POLG_POL1M in UniProt).
</p>
</li>
<li> <p><code>TBD+05.csv</code> lists genes with transcriptomic expression changes in carbon limitation stress response experiments in yeast (Tai et al., 2005).
</p>
</li>
<li> <p><code>TBD+05_aa.csv</code> has the amino acid compositions of proteins coded by those genes.
The last two files are used in <code>demo{"rank.affinity"}</code>.
</p>
</li></ul>

<p>Files in <code>taxonomy</code> contain taxonomic data files:
</p>

<ul>
<li> <p><code>names.dmp</code> and <code>nodes.dmp</code> are excerpts of NCBI taxonomy files (<a href="https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz">https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz</a>, accessed 2010-02-15). These files contain only the entries for <em>Escherichia coli</em> K-12, <em>Saccharomyces cerevisiae</em>, <em>Homo sapiens</em>, <em>Pyrococcus furisosus</em> and <em>Methanocaldococcus jannaschii</em> (taxids 83333, 4932, 9606, 186497, 243232) and the higher-ranking nodes (genus, family, etc.) in the respective lineages. See <code><a href="#topic+taxonomy">taxonomy</a></code> for examples that use these files.
</p>
</li></ul>

<p>Files in <code>adds</code> contain additional thermodynamic data and group additivity definitions:
</p>

<ul>
<li> <p><code>BZA10.csv</code> contains supplementary thermodynamic data taken from Bazarkina et al. (2010). The data can be added to the database in the current session using <code><a href="#topic+add.OBIGT">add.OBIGT</a></code>. See <code><a href="#topic+add.OBIGT">add.OBIGT</a></code> for an example that uses this file.
</p>
</li>
<li> <p><code>OBIGT_check.csv</code> contains the results of running <code><a href="#topic+check.OBIGT">check.OBIGT</a></code> to check the internal consistency of entries in the default and optional datafiles.
</p>
</li>
<li> <p><code>RH98_Table15.csv</code> Group stoichiometries for high molecular weight crystalline and liquid organic compounds taken from Table 15 of Richard and Helgeson, 1998. The first three columns have the <code>compound</code> name, <code>formula</code> and physical <code>state</code> (&lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;liq&#8288;</span>&rsquo;). The remaining columns have the numbers of each group in the compound; the names of the groups (columns) correspond to species in <code><a href="#topic+thermo">thermo</a>$OBIGT</code>. The compound named &lsquo;<span class="samp">&#8288;5a(H),14a(H)-cholestane&#8288;</span>&rsquo; in the paper has been changed to &lsquo;<span class="samp">&#8288;5a(H),14b(H)-cholestane&#8288;</span>&rsquo; here to match the group stoichiometry given in the table. See <code><a href="#topic+RH2OBIGT">RH2OBIGT</a></code> for a function that uses this file.
</p>
</li>
<li> <p><code>SK95.csv</code> contains thermodynamic data for alanate, glycinate, and their complexes with metals, taken from Amend and Helgeson (1997) and Shock and Koretsky (1995) as corrected in slop98.dat. These data are used in <code>demo("copper")</code> and <code>demo("glycinate")</code>.
</p>
</li>
<li> <p><code>LA19_test.csv</code> contains thermodynamic data for dimethylamine and trimethylamine from LaRowe and Amend (2019) in energy units of both J and cal. This file is used in <code>test-util.data.R</code>) to check the messages produced by <code><a href="#topic+check.GHS">check.GHS</a></code> and <code><a href="#topic+check.EOS">check.EOS</a></code>.
</p>
</li></ul>



<h3>References</h3>

<p>Akinfiev, N. N. and Diamond, L. W. (2003) Thermodynamic description of aqueous nonelectrolytes at infinite dilution over a wide range of state parameters. <em>Geochim. Cosmochim. Acta</em> <b>67</b>, 613&ndash;629. <a href="https://doi.org/10.1016/S0016-7037%2802%2901141-9">doi:10.1016/S0016-7037(02)01141-9</a>
</p>
<p>Amend, J. P. and Helgeson, H. C. (1997) Calculation of the standard molal thermodynamic properties of aqueous biomolecules at elevated temperatures and pressures. Part 1. L-&alpha;-amino acids. <em>J. Chem. Soc., Faraday Trans.</em> <b>93</b>, 1927&ndash;1941. <a href="https://doi.org/10.1039/A608126F">doi:10.1039/A608126F</a>
</p>
<p>Bazarkina, E. F., Zotov, A. V. and Akinfiev, N. N. (2010) Pressure-dependent stability of cadmium chloride complexes: Potentiometric measurements at 1–1000 bar and 25°C. <em>Geol. Ore Deposits</em> <b>52</b>, 167&ndash;178. <a href="https://doi.org/10.1134/S1075701510020054">doi:10.1134/S1075701510020054</a>
</p>
<p>Berman, R. G. (1988) Internally-consistent thermodynamic data for minerals in the system Na<sub>2</sub>O-K<sub>2</sub>O-CaO-MgO-FeO-Fe<sub>2</sub>O<sub>3</sub>-Al<sub>2</sub>O<sub>3</sub>-SiO<sub>2</sub>-TiO<sub>2</sub>-H<sub>2</sub>O-CO<sub>2</sub>. <em>J. Petrol.</em> <b>29</b>, 445-522. <a href="https://doi.org/10.1093/petrology/29.2.445">doi:10.1093/petrology/29.2.445</a>
</p>
<p>Berman, R. G. and Aranovich, L. Ya. (1996) Optimized standard state and solution properties of minerals. I. Model calibration for olivine, orthopyroxene, cordierite, garnet, and ilmenite in the system FeO-MgO-CaO-Al<sub>2</sub>O<sub>3</sub>-TiO<sub>2</sub>-SiO<sub>2</sub>. <em>Contrib. Mineral. Petrol.</em> <b>126</b>, 1-24. <a href="https://doi.org/10.1007/s004100050233">doi:10.1007/s004100050233</a>
</p>
<p>Dick, J. M. (2014) Average oxidation state of carbon in proteins. <em>J. R. Soc. Interface</em> <b>11</b>, 20131095. <a href="https://doi.org/10.1098/rsif.2013.1095">doi:10.1098/rsif.2013.1095</a>
</p>
<p>Gattiker, A., Michoud, K., Rivoire, C., Auchincloss, A. H., Coudert, E., Lima, T., Kersey, P., Pagni, M., Sigrist, C. J. A., Lachaize, C., Veuthey, A.-L., Gasteiger, E. and Bairoch, A. (2003) Automatic annotation of microbial proteomes in Swiss-Prot. <em>Comput. Biol. Chem.</em> <b>27</b>, 49&ndash;58. <a href="https://doi.org/10.1016/S1476-9271%2802%2900094-4">doi:10.1016/S1476-9271(02)00094-4</a>
</p>
<p>Gaucher, E. A., Thomson, J. M., Burgan, M. F. and Benner, S. A (2003) Inferring the palaeoenvironment of ancient bacteria on the basis of resurrected proteins. <em>Nature</em> <b>425</b>(6955), 285&ndash;288. <a href="https://doi.org/10.1038/nature01977">doi:10.1038/nature01977</a>
</p>
<p>Helgeson, H. C., Delany, J. M., Nesbitt, H. W. and Bird, D. K. (1978) Summary and critique of the thermodynamic properties of rock-forming minerals. <em>Am. J. Sci.</em> <b>278-A</b>, 1&ndash;229. <a href="https://www.worldcat.org/oclc/13594862">https://www.worldcat.org/oclc/13594862</a>
</p>
<p>Hnědkovský, L., Wood, R. H. and Majer, V. (1996) Volumes of aqueous solutions of CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S, and NH<sub>3</sub> at temperatures from 298.15 K to 705 K and pressures to 35 MPa. <em>J. Chem. Thermodyn.</em> <b>28</b>, 125&ndash;142. <a href="https://doi.org/10.1006/jcht.1996.0011">doi:10.1006/jcht.1996.0011</a>
</p>
<p>Hnědkovský, L. and Wood, R. H. (1997) Apparent molar heat capacities of aqueous solutions of CH<sub>4</sub>, CO<sub>2</sub>, H<sub>2</sub>S, and NH<sub>3</sub> at temperatures from 304 K to 704 K at a pressure of 28 MPa. <em>J. Chem. Thermodyn.</em> <b>29</b>, 731&ndash;747. <a href="https://doi.org/10.1006/jcht.1997.0192">doi:10.1006/jcht.1997.0192</a>
</p>
<p>Joint Genome Institute (2007) Bison Pool Environmental Genome. Protein sequence files downloaded from IMG/M (<a href="https://img.jgi.doe.gov/">https://img.jgi.doe.gov/</a>)
</p>
<p>LaRowe, D. E. and Amend, J. P. (2019) The energetics of fermentation in natural settings. <em>Geomicrobiol. J.</em> <b>36</b>, 492&ndash;505. <a href="https://doi.org/10.1080/01490451.2019.1573278">doi:10.1080/01490451.2019.1573278</a>
</p>
<p>Merino, E. (1975) Diagenesis in teriary sandstones from Kettleman North Dome, California. II. Interstitial solutions: distribution of aqueous species at 100&amp;deg;C and chemical relation to diagenetic mineralogy. <em>Geochim. Cosmochim. Acta</em> <b>39</b>, 1629&ndash;1645. <a href="https://doi.org/10.1016/0016-7037%2875%2990085-X">doi:10.1016/0016-7037(75)90085-X</a>
</p>
<p>Privalov, P. L. and Makhatadze, G. I. (1990) Heat capacity of proteins. II. Partial molar heat capacity of the unfolded polypeptide chain of proteins: Protein unfolding effects. <em>J. Mol. Biol.</em> <b>213</b>, 385&ndash;391. <a href="https://doi.org/10.1016/S0022-2836%2805%2980198-6">doi:10.1016/S0022-2836(05)80198-6</a>
</p>
<p>Richard, L. and Helgeson, H. C. (1998) Calculation of the thermodynamic properties at elevated temperatures and pressures of saturated and aromatic high molecular weight solid and liquid hydrocarbons in kerogen, bitumen, petroleum, and other organic matter of biogeochemical interest. <em>Geochim. Cosmochim. Acta</em> <b>62</b>, 3591&ndash;3636. <a href="https://doi.org/10.1016/S0016-7037%2897%2900345-1">doi:10.1016/S0016-7037(97)00345-1</a>
</p>
<p>Robie, R. A. and Hemingway, B. S. (1995) <em>Thermodynamic Properties of Minerals and Related Substances at 298.15 K and 1 Bar (<code class="reqn">10^5</code> Pascals) Pressure and at Higher Temperatures</em>. U. S. Geol. Surv., Bull. 2131, 461 p. <a href="https://www.worldcat.org/oclc/32590140">https://www.worldcat.org/oclc/32590140</a>
</p>
<p>Shock, E. and Canovas, P. (2010) The potential for abiotic organic synthesis and biosynthesis at seafloor hydrothermal systems. <em>Geofluids</em> <b>10</b>, 161&ndash;192. <a href="https://doi.org/10.1111/j.1468-8123.2010.00277.x">doi:10.1111/j.1468-8123.2010.00277.x</a>
</p>
<p>Shock, E. L. and Koretsky, C. M. (1995) Metal-organic complexes in geochemical processes: Estimation of standard partial molal thermodynamic properties of aqueous complexes between metal cations and monovalent organic acid ligands at high pressures and temperatures. <em>Geochim. Cosmochim. Acta</em> <b>59</b>, 1497&ndash;1532. <a href="https://doi.org/10.1016/0016-7037%2895%2900058-8">doi:10.1016/0016-7037(95)00058-8</a>
</p>
<p>Shock, E. L., Oelkers, E. H., Johnson, J. W., Sverjensky, D. A. and Helgeson, H. C. (1992) Calculation of the thermodynamic properties of aqueous species at high pressures and temperatures: Effective electrostatic radii, dissociation constants and standard partial molal properties to 1000 &deg;C and 5 kbar. <em>J. Chem. Soc. Faraday Trans.</em> <b>88</b>, 803&ndash;826. <a href="https://doi.org/10.1039/FT9928800803">doi:10.1039/FT9928800803</a>
</p>
<p>Shock, E. L. and Schulte, M. D. (1998) Organic synthesis during fluid mixing in hydrothermal systems. <em>J. Geophys. Res.</em> <b>103</b>, 28513&ndash;28527. <a href="https://doi.org/10.1029/98JE02142">doi:10.1029/98JE02142</a>
</p>
<p>Tai, S. L., Boer, V. M., Daran-Lapujade, P., Walsh, M. C., de Winde, J. H., Daran, J.-M. and Pronk, J. T. (2005) Two-dimensional transcriptome analysis in chemostat cultures: Combinatorial effects of oxygen availability and macronutrient limitation in <em>Saccharomyces cerevisiae</em>. <em>J. Biol. Chem.</em> <b>280</b>, 437&ndash;447. <a href="https://doi.org/10.1074/jbc.M410573200">doi:10.1074/jbc.M410573200</a>
</p>
<p>Tutolo, B. M., Kong, X.-Z., Seyfried, W. E., Jr. and Saar, M. O. (2014) Internal consistency in aqueous geochemical data revisited: Applications to the aluminum system. <em>Geochim. Cosmochim. Acta</em> <b>133</b>, 216&ndash;234. <a href="https://doi.org/10.1016/j.gca.2014.02.036">doi:10.1016/j.gca.2014.02.036</a>
</p>

<hr>
<h2 id='IAPWS95'>Properties of Water from IAPWS-95</h2><span id='topic+IAPWS95'></span>

<h3>Description</h3>

<p>Calculate thermodynamic properties of water following the IAPWS-95 formulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  IAPWS95(property, T = 298.15, rho = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IAPWS95_+3A_property">property</code></td>
<td>
<p>character, name(s) of property(s) to calculate</p>
</td></tr>
<tr><td><code id="IAPWS95_+3A_t">T</code></td>
<td>
<p>numeric, temperature (K)</p>
</td></tr>
<tr><td><code id="IAPWS95_+3A_rho">rho</code></td>
<td>
<p>numeric, density (kg m<code class="reqn">^{-3}</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>IAPWS95</code> provides an implementation of the IAPWS-95 formulation for properties (including pressure) calculated as a function of temperature and density.
</p>
<p>The <code>IAPWS95</code> function returns values of thermodynamic properties in specific units (per gram).
The IAPWS-95 formulation follows the triple point convention used in engineering (values of internal energy and entropy are taken to be zero at the triple point).
</p>
<p>For <code>IAPWS95</code> the upper temperature limit of validity is 1000 &deg;C, but extrapolation to much higher temperatures is possible (Wagner and Pruss, 2002).
Valid pressures are from the greater of zero bar or the melting pressure at temperature to 10000 bar (with the provision for extrapolation to more extreme conditions).
The function does not check these limits and will attempt calculations for any range of input parameters, but may return <code>NA</code> for properties that fail to be calculated at given temperatures and pressures and/or produce warnings or even errors when problems are encountered.
</p>


<h3>Value</h3>

<p>A data frame the number of rows of which corresponds to the number of input temperature, pressure and/or density values.
</p>


<h3>References</h3>

<p>Wagner, W. and Pruss, A. (2002) The IAPWS formulation 1995 for the thermodynamic properties of ordinary water substance for general and scientific use. <em>J. Phys. Chem. Ref. Data</em> <b>31</b>, 387&ndash;535. <a href="https://doi.org/10.1063/1.1461829">doi:10.1063/1.1461829</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+util.water">util.water</a></code> for properties along the saturation curve (<code>WP02.auxiliary</code>) and calculation of density from pressure and temperature (<code>rho.IAPWS92</code>).
<code><a href="#topic+water.IAPWS95">water.IAPWS95</a></code> is a wrapper around <code>IAPWS95</code> and the utility functions, which converts the specific units to molar quantities, and is used in higher-level functions (<code><a href="#topic+water">water</a></code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate pressure for given temperature and density
IAPWS95("P", T = 500, rho = 838.0235)
</code></pre>

<hr>
<h2 id='info'>Search the Thermodynamic Database</h2><span id='topic+info'></span>

<h3>Description</h3>

<p>Search for species by name or formula, retrieve their thermodynamic properties and parameters, and add proteins to the thermodynamic database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  info(species = NULL, state = NULL, check.it=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="info_+3A_species">species</code></td>
<td>
<p>character, names or formulas of species, or (for <code>info</code> only) numeric with same meaning as <code>ispecies</code></p>
</td></tr>
<tr><td><code id="info_+3A_state">state</code></td>
<td>
<p>character, physical states of the species</p>
</td></tr>
<tr><td><code id="info_+3A_check.it">check.it</code></td>
<td>
<p>logical, check GHS and EOS parameters for self-consistency?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>info</code> is the primary function used for querying the thermodynamic database (<code><a href="#topic+thermo">thermo</a>()$OBIGT</code>).
It is often called recursively; first with a character value (or values) for <code>species</code> indicating the name(s) or formula(s) of the species of interest.
The result of this call is a numeric value, which can be provided as an argument in a second call to <code>info</code> in order to retrieve a data frame of the thermodynamic properties of the species.
</p>
<p>The text of <code>species</code> is searched in the names, chemical formulas, and abbreviations (in the &lsquo;<span class="samp">&#8288;abbrv&#8288;</span>&rsquo; column) in the thermodynamic database.
If the text of the <code>species</code> is matched, the index of that species is returned.
If there are multiple matches for the <code>species</code>, and <code>state</code> is NULL, the index of first match is returned.
The order of entries in the database is grouped by states in the order &lsquo;<span class="samp">&#8288;aq&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;gas&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;liq&#8288;</span>&rsquo;.
Therefore, for substances represented by both aqueous and gaseous species, the index of the aqueous species is returned, unless <code>state</code> is set to &lsquo;<span class="samp">&#8288;gas&#8288;</span>&rsquo;.
Note that names (not formulas) of inorganic species, such as &lsquo;<span class="samp">&#8288;oxygen&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;methane&#8288;</span>&rsquo;, are used only for the gas.
</p>
<p>Names of species including an underscore character are indicative of proteins, e.g. &lsquo;<span class="samp">&#8288;LYSC_CHICK&#8288;</span>&rsquo;.
If the name of a protein is provided to <code>info</code> and the composition of the protein can be found using <code><a href="#topic+pinfo">pinfo</a></code>, the thermodyamic properties and parameters of the nonionized protein (calculated using amino acid group additivity) are added to the thermodynamic database.
Included in the return value, as for other species, is the index of the protein in the thermodynamic database or <code>NA</code> if the protein is not found. Names of proteins and other species can be mixed.
</p>
<p>If no exact matches are found, <code>info</code> searches the database for similar names or formulas using <code><a href="base.html#topic+agrep">agrep</a></code>.
If any of these are found, the results are summarized on the screen, but the function always returns NA in this case.
</p>
<p>With a numeric argument, the rows of <code>thermo()$OBIGT</code> indicated by <code>ispecies</code> are returned, after removing any order-of-magnitude scaling factors (see <code><a href="#topic+thermo">thermo</a></code>).
If these species are all aqueous or are all not aqueous, the compounded column names used in <code>thermo()$OBIGT</code> are replaced with names appropriate for the corresponding equations of state.
A missing value of one of the standard molal Gibbs energy (<code>G</code>) or enthalpy (<code>H</code>) of formation from the elements or entropy (<code>S</code>) is calculated from the other two, if available.
If <code>check.it</code> is TRUE, several checks of self consistency among the thermodynamic properties and parameters are performed using <code><a href="#topic+check.GHS">check.GHS</a></code> and <code><a href="#topic+check.EOS">check.EOS</a></code>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+retrieve">retrieve</a></code> for searching species by element; <code><a href="#topic+check.OBIGT">check.OBIGT</a></code> for checking self-consistency of each species. </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Summary of available data
info()

## Species information
# Search for something named (or whose formula is) "Fe"
si &lt;- info("Fe")
# Use the number to get the full entry
info(si)
# Show data for the higher-temperature phases
info(si:(si+3))

## Dealing with states
# Order of precedence for names:
# aq &gt; cr &gt; gas &gt; liq
info(c("ethanol", "adenosine"))  # aq, aq
# State argument overrides the default
info(c("ethanol", "adenosine"), state = c("gas", "cr"))
# Exceptions: gases have precedence for names of methane and inorganic gases
info(c("methane", "oxygen")) # gas, gas

# Formulas default to aqueous species, if available
i1 &lt;- info(c("CH4", "CO2", "CS2", "MgO"))
info(i1)$state  # aq, aq, gas, cr
# State argument overrides the default
i2 &lt;- info(c("CH4", "CO2", "MgO"), "gas") 
info(i2)$state  # gas, gas, NA

## Partial name or formula searches
info("ATP")
info("thiol")
info("MgC")
# Add an extra character to refine a search
# or to search using terms that have exact matches
info("MgC ")
info("acetate ")
info(" H2O")
</code></pre>

<hr>
<h2 id='ionize.aa'>Properties of Ionization of Proteins</h2><span id='topic+ionize.aa'></span>

<h3>Description</h3>

<p>Calculate the charges of proteins and contributions of ionization to the thermodynamic properties of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ionize.aa(aa, property = "Z", T = 25, P = "Psat", pH = 7,
    ret.val = NULL, suppress.Cys = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ionize.aa_+3A_aa">aa</code></td>
<td>
<p>data frame, amino acid composition in the format of <code>thermo()$protein</code></p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_property">property</code></td>
<td>
<p>character, property to calculate</p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_t">T</code></td>
<td>
<p>numeric, temperature in &deg;C</p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_p">P</code></td>
<td>
<p>numeric, pressure in bar, or &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; for vapor pressure of H<sub>2</sub>O above 100 &deg;C</p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_ph">pH</code></td>
<td>
<p>numeric, pH</p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_ret.val">ret.val</code></td>
<td>
<p>character, return the indicated value from intermediate calculations</p>
</td></tr>
<tr><td><code id="ionize.aa_+3A_suppress.cys">suppress.Cys</code></td>
<td>
<p>logical, suppress (ignore) the ionization of the cysteine groups?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The properties of ionization of proteins calculated by this function take account of the standard molal thermodynamic properties of ionizable amino acid sidechain groups and the terminal groups in proteins ([AABB]) and their equations of state parameters taken from Dick et al., 2006. The values of the ionization constants (pK) are calculated as a function of temperature, and the charges and the ionization contributions of other thermodynamic properties to the proteins are calculated additively, without consideration of electrostatic interactions, so they are best applied to the unfolded protein reference state.
</p>
<p>For each amino acid composition in <code>aa</code>, the additive value of the <code>property</code> is calculated as a function of <code>T</code>, <code>P</code> and <code>pH</code>. <code>property</code> can be NULL to denote net charge, or if not NULL is one of the properties available in <code><a href="#topic+subcrt">subcrt</a></code>, or is &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo; to calculate the dimensionless chemical affinity (A/2.303RT) of the ionization reaction for the protein. If <code>ret.val</code> is one of &lsquo;<span class="samp">&#8288;pK&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;alpha&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;aavals&#8288;</span>&rsquo; it indicates to return the value of the ionization constant, degree of formation, or the values of the <code>property</code> for each ionizable group rather than taking their sums for the amino acid compositions in <code>aa</code>.
</p>


<h3>Value</h3>

<p>The function returns a matrix (possibly with only one row or column) with number of rows corresponding to the longest of <code>T</code>, <code>P</code> or <code>pH</code> (values of any of these with shorter length are recycled) and a column for each of the amino acid compositions in <code>aa</code>.
</p>


<h3>References</h3>

<p>Dick, J. M., LaRowe, D. E. and Helgeson, H. C. (2006) Temperature, pressure, and electrochemical constraints on protein speciation: Group additivity calculation of the standard molal thermodynamic properties of ionized unfolded proteins. <em>Biogeosciences</em> <b>3</b>, 311&ndash;336. <a href="https://doi.org/10.5194/bg-3-311-2006">doi:10.5194/bg-3-311-2006</a>
</p>
<p>Makhatadze, G. I. and Privalov, P. L. (1990) Heat capacity of proteins. 1. Partial molar heat capacity of individual amino acid residues in aqueous solution: Hydration effect. <em>J. Mol. Biol.</em> <b>213</b>, 375&ndash;384. <a href="https://doi.org/10.1016/S0022-2836%2805%2980197-4">doi:10.1016/S0022-2836(05)80197-4</a>
</p>
<p>Privalov, P. L. and Makhatadze, G. I. (1990) Heat capacity of proteins. II. Partial molar heat capacity of the unfolded polypeptide chain of proteins: Protein unfolding effects. <em>J. Mol. Biol.</em> <b>213</b>, 385&ndash;391. <a href="https://doi.org/10.1016/S0022-2836%2805%2980198-6">doi:10.1016/S0022-2836(05)80198-6</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+pinfo">pinfo</a></code>, <code><a href="#topic+affinity">affinity</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Heat capacity of LYSC_CHICK as a function of T
pH &lt;- c(5, 9, 3)
T &lt;- seq(0, 100)
# Cp of non-ionized protein
Cp.nonion &lt;- subcrt("LYSC_CHICK", T = T)$out[[1]]$Cp
plot(T, Cp.nonion, xlab = axis.label("T"), type = "l",
  ylab = axis.label("Cp"), ylim = c(20000, 35000))
# Cp of ionization and ionized protein
aa &lt;- pinfo(pinfo("LYSC_CHICK"))
for(pH in c(5, 9, 3)) {
  Cp.ionized &lt;- Cp.nonion + ionize.aa(aa, "Cp", T = T, pH = pH)[, 1]
  lines(T, Cp.ionized, lty = 2)
  text(80, Cp.ionized[70], paste("pH =", pH) )
}
# Makhatadze and Privalov's group contributions
T &lt;- c(5, 25, 50, 75, 100, 125)
points(T, MP90.cp("LYSC_CHICK", T))
# Privalov and Makhatadze's experimental values
e &lt;- read.csv(system.file("extdata/cpetc/PM90.csv", package = "CHNOSZ"))
points(e$T, e$LYSC_CHICK, pch = 16)
legend("bottomright", pch = c(16, 1, NA, NA), lty = c(NA, NA, 1, 2),
  legend = c("PM90 experiment", "MP90 groups", 
  "DLH06 groups no ion", "DLH06 groups ionized"))
title("Heat capacity of unfolded LYSC_CHICK")
</code></pre>

<hr>
<h2 id='logB.to.OBIGT'>Fit Thermodynamic Parameters to Formation Constants (log&thinsp;&beta;)</h2><span id='topic+logB.to.OBIGT'></span>

<h3>Description</h3>

<p>Fit thermodynamic parameters to experimental formation constants for an aqueous species and add the parameters to OBIGT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  logB.to.OBIGT(logB, species, coeffs, T, P, npar = 3,
    optimize.omega = FALSE, tolerance = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logB.to.OBIGT_+3A_logb">logB</code></td>
<td>
<p>Values of log&thinsp;&beta;</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_species">species</code></td>
<td>
<p>Species in the formation reaction (the formed species must be last)</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_coeffs">coeffs</code></td>
<td>
<p>Coefficients of the formation reaction</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_t">T</code></td>
<td>
<p>Temperature (units of <code><a href="#topic+T.units">T.units</a></code>)</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_p">P</code></td>
<td>
<p>Temperature (&lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; or numerical values with units of <code><a href="#topic+P.units">P.units</a></code>)</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_npar">npar</code></td>
<td>
<p>numeric, number of parameters to use in fitting</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_optimize.omega">optimize.omega</code></td>
<td>
<p>logical, optimize the omega parameter (relevant for charged species)?</p>
</td></tr>
<tr><td><code id="logB.to.OBIGT_+3A_tolerance">tolerance</code></td>
<td>
<p>Tolerance for checking equivalence of input and calculated log&thinsp;&beta; values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function updates the <code><a href="#topic+OBIGT">OBIGT</a></code> thermodynamic database with parameters fit to formation constants of aqueous species as a function of temperature.
The <code>logB</code> argument should have decimal logarithm of formation constants for an aqueous complex (log&thinsp;&beta;).
The formation reaction is defined by <code>species</code> and <code>coeffs</code>.
All species in the formation reaction must be present in OBIGT except for the <em>last</em> species, which is the newly formed species.
</p>
<p>The function works as follows.
First, the properties of the known species are combined with log&thinsp;&beta; to calculate the standard Gibbs energy (G[T]) of the formed species at each value of <code>T</code> and <code>P</code>.
Then, <code><a href="stats.html#topic+lm">lm</a></code> is used to fit one or more of the thermodynamic parameters <code>G</code>, <code>S</code>, <code>c1</code>, <code>c2</code>, and <code>omega</code> to the values of G[T].
The first two of these parameters are standard-state values at 25 &deg;C and 1 bar, and the last three are parameters in the revised HKF equations (e.g. Eq. B25 of Shock et al., 1992).
The fitted parameters for the formed species are then added to OBIGT.
Finally, <code><a href="base.html#topic+all.equal">all.equal</a></code> is used to test for approximate equivalence of the input values of log&thinsp;&beta; and calculated equilibrium constants; if the mean absolute difference exceeds <code>tolerance</code>, an error occurs.
</p>
<p>To avoid overfitting, only the first three parameters (<code>G</code>, <code>S</code>, and <code>c1</code>) are used by default.
More parameters (up to 5) or fewer (down to 1) can be selected by changing <code>npar</code>.
Volumetric parameters (<I>a</I><sub>1</sub> to <I>a</I><sub>4</sub>) in the HKF equations currently aren't included, so the resulting fits are valid only at the input pressure values.
</p>
<p>Independent of <code>npar</code>, the number of parameters used in the fit is not more than the number of data values (<em>n</em>).
If <em>n</em> is less than 5, then the values of the unused parameters are set to 0 for addition to OBIGT.
For instance, a single value of log&thinsp;&beta; would be fit only with <code>G</code>, implying that computed values of G[T] have no temperature dependence.
</p>
<p>The value of &omega; is a constant in the revised HKF equations for uncharged species, but for charged species, it is a function of <I>T</I> and <I>P</I> as described by the &ldquo;g function&rdquo; (Shock et al., 1992).
An optimization step is available to refine the value of <code>omega</code> at 25 &deg;C and 1 bar for charged species taking its temperature dependence into account for the fitting.
However, representative datasets are not well represented by variable <code>omega</code> (see first example), so this step is skipped by default.
Furthermore, <code>logB.to.OBIGT</code> by default also sets the <code>z</code> parameter in OBIGT to 0; this enforces a constant &omega; for charged species in calculations with <code><a href="#topic+subcrt">subcrt</a></code> (note that this is a parameter for the HKF equations and does not affect reaction balancing).
Set <code>optimize.omega</code> to TRUE to override the defaults and activate variable &omega; for charged species; this takes effect only if <code>npar == 5</code> and <em>n</em> &gt; 5.
</p>


<h3>Value</h3>

<p>The species index of the new species in OBIGT.
</p>


<h3>References</h3>

<p>Migdisov, Art. A., Zezin, D. and Williams-Jones, A. E. (2011) An experimental study of Cobalt (II) complexation in Cl<sup>-</sup> and H<sub>2</sub>S-bearing hydrothermal solutions. <em>Geochim. Cosmochim. Acta</em> <b>75</b>, 4065&ndash;4079. <a href="https://doi.org/10.1016/j.gca.2011.05.003">doi:10.1016/j.gca.2011.05.003</a>
</p>
<p>Mei, Y., Sherman, D. M., Liu, W., Etschmann, B., Testemale, D. and Brugger, J. (2015) Zinc complexation in chloride-rich hydrothermal fluids (25&ndash;600 &deg;C): A thermodynamic model derived from <em>ab initio</em> molecular dynamics. <em>Geochim. Cosmochim. Acta</em> <b>150</b>, 264&ndash;284. <a href="https://doi.org/10.1016/j.gca.2014.09.023">doi:10.1016/j.gca.2014.09.023</a>
</p>
<p>Shock, E. L., Oelkers, E. H., Johnson, J. W., Sverjensky, D. A. and Helgeson, H. C. (1992) Calculation of the thermodynamic properties of aqueous species at high pressures and temperatures: Effective electrostatic radii, dissociation constants and standard partial molal properties to 1000 &deg;C and 5 kbar. <em>J. Chem. Soc. Faraday Trans.</em> <b>88</b>, 803&ndash;826. <a href="https://doi.org/10.1039/FT9928800803">doi:10.1039/FT9928800803</a>
</p>


<h3>See Also</h3>

<p><code>logB.to.OBIGT</code> calls <code><a href="#topic+mod.OBIGT">mod.OBIGT</a></code> with <code>zap = TRUE</code> to clear the parameters of a formed species if it already exists in the OBIGT database.
If preexisting parameters (e.g. volumetric HKF coefficients) weren't cleared, they would interfere with the fitting done here, which uses only selected parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## CoHS+ from Migdisov et al. (2011)
logB &lt;- c(6.24, 6.02, 5.84, 5.97, 6.52)
T &lt;- c(120, 150, 200, 250, 300)
P &lt;- "Psat"
species &lt;- c("Co+2", "HS-", "CoHS+")
coeffs &lt;- c(-1, -1, 1)
opar &lt;- par(mfrow = c(2, 1))
for(o.o in c(TRUE, FALSE)) { 
  # Fit the parameters with or without variable omega
  inew &lt;- logB.to.OBIGT(logB, species, coeffs, T, P, npar = 5, optimize.omega = o.o)
  # Print the new database entry
  info(inew)
  # Plot experimental logB
  plot(T, logB, "n", c(100, 320), c(5.8, 6.8),
    xlab = axis.label("T"), ylab = quote(log~beta))
  points(T, logB, pch = 19, cex = 2)
  # Plot calculated values
  Tfit &lt;- seq(100, 320, 10)
  sres &lt;- subcrt(species, coeffs, T = Tfit)
  lines(sres$out$T, sres$out$logK, col = 4)
  title(describe.reaction(sres$reaction))
  legend &lt;- c("Migdisov et al. (2011)",paste0("logB.to.OBIGT(optimize.omega = ",o.o,")"))
  legend("top", legend, pch = c(19, NA), lty = c(0, 1), col = c(1, 4),
    pt.cex = 2, bg = "#FFFFFFB0")
}
par(opar)
# NB. Optimizing omega leads to unphysical oscillations in the logK (first plot)

## ZnCl+ from Mei et al. (2015)
# Values for 5000 bar calculated with the modified Ryzhenko-Bryzgalin (RB) model
logB &lt;- c(-1.93,-1.16,-0.38,0.45,1.15,1.76,2.30,2.80,3.26,3.70,4.12,4.53,4.92)
species &lt;- c("Zn+2", "Cl-", "ZnCl+")
coeffs &lt;- c(-1, -1, 1)
T &lt;- c(25, 60, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600)
P &lt;- 5000
# Note: ZnCl+ is in the default OBIGT database;
# logB.to.OBIGT() "zaps" the previous parameters before adding the fitted ones
inew &lt;- logB.to.OBIGT(logB, species, coeffs, T, P, npar = 5)
# Plot RB and logB.to.OBIGT values
plot(T, logB, xlab = axis.label("T"), ylab = axis.label("logB"), pch = 19, cex = 2)
Tfit &lt;- seq(25, 600, 25)
sres &lt;- subcrt(species, coeffs, T = Tfit, P = P)
lines(sres$out$T, sres$out$logK, col = 4)
title(describe.reaction(sres$reaction), line = 3)
title("5000 bar", font.main = 1, line = 1)
legend &lt;- c("Mei et al. (2015)", "logB.to.OBIGT()")
legend("topleft", legend, pch = c(19, NA), lty = c(0, 1), col = c(1, 4), pt.cex = 2)
</code></pre>

<hr>
<h2 id='makeup'>Parse Chemical Formulas</h2><span id='topic+makeup'></span><span id='topic+count.elements'></span>

<h3>Description</h3>

<p>Count the elements and charges in a chemical formula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  makeup(formula, multiplier = 1, sum = FALSE, count.zero = FALSE)
  count.elements(formula)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeup_+3A_formula">formula</code></td>
<td>
<p>character, a chemical formula</p>
</td></tr>
<tr><td><code id="makeup_+3A_multiplier">multiplier</code></td>
<td>
<p>numeric, multiplier for the elemental counts in each formula</p>
</td></tr>
<tr><td><code id="makeup_+3A_sum">sum</code></td>
<td>
<p>logical, add together the elemental counts in all formulas?</p>
</td></tr>
<tr><td><code id="makeup_+3A_count.zero">count.zero</code></td>
<td>
<p>logical, include zero counts for elements?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>makeup</code> parses a chemical formula expressed in string notation, returning the numbers of each element in the formula.
The formula may carry a charge, indicated by a + or - sign, possibly followed by a magnitude, after the uncharged part of the formula.
The formula may have multiple subformulas enclosed in parentheses (but the parentheses may not be nested), each one optionally followed by a numeric coefficient.
The formula may have one suffixed subformula, separated by &lsquo;<span class="samp">&#8288;*&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;:&#8288;</span>&rsquo;, optionally preceded by a numeric coefficient. All numbers may contain a decimal point.
</p>
<p>Each subformula (or the entire formula without subformulas) should be a simple formula.
A simple formula, processed by <code>count.elements</code>, must adhere to the following pattern: it starts with an elemental symbol; all elemental symbols start with an uppercase letter, and are followed by another elemental symbol, a number (possibly fractional, possibly signed), or nothing (the end of the formula).
Any sequence of one uppercase letter followed by zero or more lowercase letters is recognized as an elemental symbol.
<code>makeup</code> will issue a warning for elemental symbols that are not present in <code><a href="#topic+thermo">thermo</a>$element</code>.
</p>
<p><code>makeup</code> can handle numeric and length &gt; 1 values for the <code>formula</code> argument.
If the argument is numeric, it identifies row number(s) in <code>thermo()$OBIGT</code> from which to take the formulas of species.
If <code>formula</code> has length &gt; 1, the function returns a list containing the elemental counts in each of the formulas.
If <code>count.zero</code> is TRUE, the elemental counts for each formula include zeros to indicate elements that are only present in any of the other formulas.
</p>
<p>The <code>multiplier</code> argument must have either length = 1 or length equal to the number of formulas.
The elemental count in each formula is multiplied by the respective value.
If <code>sum</code> is true, the elemental counts in all formulas (after any multiplying) are summed together to yield a single bulk formula. 
</p>


<h3>Value</h3>

<p>A numeric vector with names refering to each of the elemental symbols in the formula.
If more than one formula is provided, a list of numeric vectors is returned, unless <code>sum</code> is TRUE.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+mass">mass</a></code>, <code><a href="#topic+entropy">entropy</a></code>, <code><a href="#topic+basis">basis</a></code>, <code><a href="#topic+i2A">i2A</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Elemental composition of a simple compound
makeup("CO2")     # 1 carbon, 2 oxygen
# Formula of lawsonite, with a parenthetical part and a suffix
makeup("CaAl2Si2O7(OH)2*H2O")
# Fractional coefficients are OK
reddiv10 &lt;- makeup("C10.6N1.6P0.1")
10*reddiv10 # 106, 16, 1 (Redfield ratio)

# The coefficient for charge is a number with a *preceding* sign
# e.g., ferric iron, with a charge of +3 is expressed as
makeup("Fe+3")
# Transcribing the formula the way it appears in many 
# publications produces a likely unintended result:
# 3 iron atoms and a charge of +1
makeup("Fe3+")

# These all represent a single negative charge, i.e., electron
makeup("-1")
makeup("Z-1+0")
makeup("Z0-1")   # the "old" formula for the electron in thermo()$OBIGT
makeup("(Z-1)")  # the current formula in thermo()$OBIGT

# Hypothetical compounds with negative numbers of elements
makeup("C-4(O-2)")   # -4 carbon, -2 oxygen
makeup("C-4O-2")     # -4 carbon,  1 oxygen, -2 charge
makeup("C-4O-2-2")   # -4 carbon, -2 oxygen, -2 charge

# The 'sum' argument can be used to check mass and charge
# balance in a chemical reaction
formula &lt;- c("H2O", "H+", "(Z-1)", "O2")
mf &lt;- makeup(formula, c(-1, 2, 2, 0.5), sum = TRUE)
all(mf == 0)  # TRUE
</code></pre>

<hr>
<h2 id='mix'>Combine Diagrams for Multi-Metal Systems</h2><span id='topic+mash'></span><span id='topic+rebalance'></span><span id='topic+mix'></span>

<h3>Description</h3>

<p>Combine diagrams for different systems by mashing or rebalancing two diagrams or mixing two diagrams with a third.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mash(d1, d2)
  rebalance(d1, d2, balance = NULL)
  mix(d1, d2, d3, parts = c(1, 1), .balance = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mix_+3A_d1">d1</code></td>
<td>
<p>list, output of <code><a href="#topic+diagram">diagram</a></code> for the first mono-metallic system</p>
</td></tr>
<tr><td><code id="mix_+3A_d2">d2</code></td>
<td>
<p>list, output of <code>diagram</code> for the second mono-metallic system</p>
</td></tr>
<tr><td><code id="mix_+3A_balance">balance</code></td>
<td>
<p>character or numeric, specification of secondary balancing coefficients</p>
</td></tr>
<tr><td><code id="mix_+3A_d3">d3</code></td>
<td>
<p>list, output of <code>diagram</code> for the bimetallic system</p>
</td></tr>
<tr><td><code id="mix_+3A_parts">parts</code></td>
<td>
<p>numeric, amount of each metal (i.e. fixed composition) for the mixed system</p>
</td></tr>
<tr><td><code id="mix_+3A_.balance">.balance</code></td>
<td>
<p><em>argument for internal use only</em></p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions make a new <code><a href="#topic+affinity">affinity</a></code> object from the output of <code><a href="#topic+diagram">diagram</a></code>.
The result can be used to make a new diagram that shows the combined system.
</p>
<p><code>mash</code> creates a set of intersecting predominance fields for all possible combinations of species in <code>d1</code> and <code>d2</code>.
The new names are formed from the <code>names</code> used in the source diagrams; for example if &quot;Cp&quot; and &quot;Py&quot; are predominant minerals at the same position in diagrams 1 and 2, the field for the mashed diagram will be labeled &quot;Cp+Py&quot;.
The affinities are calculated by summing the formation reactions from the two diagrams to give equal parts of the balancing coefficients in <code>d1</code> and <code>d2</code> (that is, equal parts of two different metals).
Note that the actual values of the affinities (and therefore the ratio between the metals) doesn't affect the resulting diagram because the affinities are assigned values of -Inf wherever one of the species is not predominant in the respective single-metal diagram.
</p>
<p><code>mix</code> is an expanded form of <code>mash</code> that allows combinations not only between two single-metal diagrams (<code>d1</code> and <code>d2</code>) but also between each of those diagrams and third diagram for bimetallic species (<code>d3</code>).
All combinations of species in all crosses between the diagrams (<code>d1-d2</code>, <code>d1-d3</code>, <code>d2-d3</code>, <code>d3-d3</code>) are identified.
The mole fractions of species in each combination are computed to satisfy the ratio of metals defined in <code>parts</code>.
For example, if <code>d1</code> and <code>d2</code> are balanced on Fe<sup>+2</sup> and VO<sub>4</sub><sup>-3</sup>, the species are combined by default to give equal parts of Fe and V.
Note that pairs of bimetallic species in <code>d3</code> are included as well as single bimetallic species that satisfy the composition in <code>parts</code> (e.g. FeV for <code>c(1, 1)</code> or Fe<sub>3</sub>V for <code>c(3, 1)</code>).
</p>
<p>From the possible combinations of species, combinations are removed that have a negative mole fraction of any species or that involve any mono-metallic species that has no predominance field in the corresponding single-metal diagram.
The output consists of each unique combination of species, including the combined formation reactions and affinities (in the <code>species</code> and <code>values</code> elements of the output list), 
The affinities are assigned values of -Inf wherever one of the species is not predominant in the respective single-metal diagram.
Therefore, either the single-metal diagrams (<code>d1</code> or <code>d2</code>) can be recovered by setting <code>parts</code> to <code>c(1, 0)</code> or <code>c(0, 1)</code>, respectively.
</p>
<p>NOTE: Unlike the <code>diagram</code> calls used to make <code>d1</code> and <code>d2</code>, which by themselves should produce reasonable diagrams for a single-metal system, the <code>d3</code> diagram by itself probably has no useful interpretation.
It is only used in <code>mix</code> as a way to transmit the results of <code><a href="#topic+affinity">affinity</a></code> for the bimetmal system and the formatted names that are made by <code>diagram</code>.
</p>
<p><code>rebalance</code> creates a new set of affinities of reactions <em>between</em> species in both systems.
Diagrams for different systems likely use different primary balancing coefficients, such as balancing on different metals.
<code>rebalance</code> uses <em>secondary</em> balancing coefficients, specified acording to <code>balance</code> (see <code><a href="#topic+equilibrate">equilibrate</a></code> for a description of this argument), to determine the reactions between the species in the two systems.
The affinities of these reactions are then used <em>only</em> to identify the predominant species at each grid point.
The <em>returned</em> value of affinity are carried forward from those used to make the source diagrams (&lsquo;<span class="samp">&#8288;plotvals&#8288;</span>&rsquo; in <code>d1</code> and <code>d2</code>), and therefore reflect the primary balancing coefficients.
The returned values are assigned -Inf wherever that species is determined to not predominate according to the secondary balancing.
</p>
<p>Because <code>mash</code> yields finite values of affinity for only a single species at any grid point, the final diagram can be made with any setting of <code>balance</code>.
<code>mix</code> gives combinations of species that each have the amount of metals defined in <code>parts</code>, so it makes no difference whether the final diagram is balanced on either of the metals, or on formula units (<code>balance = 1</code>).
However, for <code>rebalance</code>, <code>balance</code> in the final diagram should be set to &lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo; to balance on formula units in order to preserve the primary balancing coefficients.
</p>


<h3>Value</h3>

<p>A list object with the same structure as the output from <code><a href="#topic+affinity">affinity</a></code>, so it can be used as input to <code>diagram</code>.
</p>


<h3>See Also</h3>

<p>More examples are in the vignette <a href="../doc/multi-metal.html"><strong>multi-metal</strong></a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
par(mfrow = c(2, 2))
# Define basis species with Fe and Cu
basis(c("Fe+2", "Cu+", "hydrogen sulfide", "oxygen", "H2O", "H+"))
xlab &lt;- ratlab("Fe+2", "Cu+")
# Calculate diagram for only Fe-bearing minerals
species(c("pyrite", "pyrrhotite", "magnetite", "hematite"))
aFe &lt;- affinity("Fe+2" = c(0, 12), O2 = c(-40, -16), T = 400, P = 2000)
dFe &lt;- diagram(aFe, xlab = xlab, main = "Fe-S-O-H")
# Calculate diagram for only Cu-bearing minerals
species(c("covellite", "chalcocite", "tenorite", "cuprite"))
aCu &lt;- affinity(aFe)  # argument recall
dCu &lt;- diagram(aCu, xlab = xlab, main = "Cu-S-O-H")
### mash() diagram
ac &lt;- mash(dFe, dCu)
diagram(ac, xlab = xlab, main = "Cu-Fe-S-O-H with mash()")
### rebalance() diagram
ad &lt;- rebalance(dFe, dCu)
diagram(ad, xlab = xlab, balance = 1, main = "Cu-Fe-S-O-H with rebalance()")
db &lt;- describe.basis(3)
leg &lt;- lex(lTP(400, 2000), db)
legend("bottomleft", legend = leg, bty = "n")
</code></pre>

<hr>
<h2 id='mosaic'>Chemical Affinities with Changing Basis Species</h2><span id='topic+mosaic'></span>

<h3>Description</h3>

<p>Calculate chemical affinities of formation reactions of species using basis species that change with the conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mosaic(bases, blend = TRUE, stable = list(), loga_aq = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mosaic_+3A_bases">bases</code></td>
<td>
<p>character, basis species to be changed in the calculation, or list, each element of which defines an independent group of changing basis species</p>
</td></tr>
<tr><td><code id="mosaic_+3A_blend">blend</code></td>
<td>
<p>logical, use relative abundances of basis species?</p>
</td></tr>
<tr><td><code id="mosaic_+3A_stable">stable</code></td>
<td>
<p>list, previously determined stable species</p>
</td></tr>
<tr><td><code id="mosaic_+3A_loga_aq">loga_aq</code></td>
<td>
<p>numeric, activities of aqueous species (overrides current values in <code><a href="#topic+basis">basis</a></code>)</p>
</td></tr>
<tr><td><code id="mosaic_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to <code><a href="#topic+affinity">affinity</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mosaic</code> calculates the affinities of formation of species when the relative abundances of the basis species listed in <code>bases</code> change over the range of conditions, due to e.g. ionization, complexation or redox reactions.
This is a way to &ldquo;speciate the basis species&rdquo;.
For example, the speciation of sulfur (&lsquo;<span class="samp">&#8288;SO4-2&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;HSO4-&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;HS-&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;H2S&#8288;</span>&rsquo;) as a function of Eh and pH affects the formation affinities, and therefore relative stabilities of iron oxide and sulfide minerals.
Chemical activity diagrams constructed by assembling sub-diagrams corresponding to the predominant (i.e. most stable) basis species can described as &ldquo;mosaic diagrams&rdquo;.
</p>
<p>The function calculates the affinities using all combination of basis species given as a vector argument to <code>bases</code>.
Or, a list can be provided contain any number of vectors that specify different groups of basis species.
All combinations of basis species in these groups are used for the calculations.
</p>
<p>The first species listed in each group should be in the current basis definition, and all the basis species in each group should be related to the first basis species there (i.e. all share the same element).
The arguments in <code>...</code> are passed to <code>affinity</code> to specify the variable conditions, such as temperature, pressure, and activities of other basis species.
</p>
<p><code>blend</code> can be a single logical value or multiple values, in order to control the calculations for individual groups of basis species.
If <code>blend</code> is TRUE (the default), the relative abundances of the basis species in each group are calculated using <code><a href="#topic+equilibrate">equilibrate</a></code>, with the total activity taken from the corresponding basis species in the incoming <code><a href="#topic+basis">basis</a></code> definition.
Then, the function calculates overall affinities of the formation reactions of each species by combining reactions written using individual basis species in proportion to the relative abundances of the basis species.
</p>
<p>If <code>blend</code> is FALSE, the function returns the affinities calculated using the single predominant basis species in <code>bases</code> at each condition.
This is appropriate when minerals, rather than aqueous species, are used as the changing basis species.
Note, however, that <code>mosaic</code> is not internally recursive: the stabilities of one group of basis species (e.g. minerals) are not affected by changes in another group (e.g. aqueous species).
</p>
<p>(&ldquo;Stacked mosaic diagrams&rdquo;) are useful for making diagrams for multi-metal systems.
By using the stable minerals in one calculation as the new basis species in a subsequent calculation, a series of stacked <code>mosaic</code> diagrams with increasing complexity can be made.
Specifically, this is done by supplying previously calculated stabilities (from the <code>predominant</code> element of the output of <code><a href="#topic+diagram">diagram</a></code>) as an element of the list in the <code>stable</code> argument whose position corresponds to the appropriate group of basis species.
Note that a value in any position of the <code>stable</code> list forces <code>blend = FALSE</code> for the corresponding group of basis species, so there is no need to explicity change the <code>blend</code> argument.
</p>
<p>The activities of mosaiced basis species in each group are taken from the current <code><a href="#topic+basis">basis</a></code> definition.
Generally it makes sense to set the activity of minerals to 1 (logact = 0) and the activity of aqueous species to some smaller value.
For mosaic stacking calculations where the mosaiced basis species include both minerals and aqueous species, the <code>loga_aq</code> argument specifies the activity of aqueous species to be used <em>in each group</em>.
That is, there should be one value of <code>loga_aq</code> for each group of basis species; use NA to indicate that the activity comes from the current <code><a href="#topic+basis">basis</a></code> definition.
See the Mosaic Stacking 2 section of the vignette <a href="../doc/multi-metal.html"><strong>multi-metal</strong></a> for an example.
</p>


<h3>Value</h3>

<p>A list containing <code>A.species</code> (affinities of formation of the species with changing basis species) and <code>A.bases</code> (affinities of formation of the basis species in terms of the first basis species), each having same structure as the list returned by <code><a href="#topic+affinity">affinity</a></code>.
If <code>bases</code> is a list, <code>A.bases</code> is also a list, each element of which corresponds to a group of basis species in <code>bases</code>.
If <code>blend</code> is TRUE, the output also contains <code>E.bases</code> (the output of <code><a href="#topic+equilibrate">equilibrate</a></code> for each group of basis species)
</p>


<h3>References</h3>

<p>Garrels, R. M. and Christ, C. L. (1965) <em>Solutions, Minerals, and Equilibria</em>, Harper &amp; Row, New York, 450 p. <a href="https://www.worldcat.org/oclc/517586">https://www.worldcat.org/oclc/517586</a>
</p>


<h3>See Also</h3>

<p><code>demo("mosaic")</code>, which extends the example below with carbonate species in order to plot a siderite field.
To calculate mineral solubilities with mosaic calculations that account for ligand speciation, use <code>bases</code> as the first argument to <code><a href="#topic+solubility">solubility</a></code>.
<code><a href="#topic+stack_mosaic">stack_mosaic</a></code> implements calculations for bimetallic systems.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Fe-minerals and aqueous species in Fe-S-O-H system
# Speciate SO4-2, HSO4-, HS-, and H2S as a function of Eh and pH
# After Garrels and Christ, 1965 Figure 7.20
pH &lt;- c(0, 14)
Eh &lt;- c(-1, 1)
T &lt;- 25
basis(c("FeO", "SO4-2", "H2O", "H+", "e-"))
basis("SO4-2", -6)
species(c("Fe+2", "Fe+3"), -6)
species(c("pyrrhotite", "pyrite", "hematite", "magnetite"), add = TRUE)
# The basis species we'll swap through
bases &lt;- c("SO4-2", "HSO4-", "HS-", "H2S")
# Calculate affinities using the relative abundances of the basis species
# NOTE: set blend = FALSE for sharp transitions between the basis species
# (looks more like the diagram in GC65)
m1 &lt;- mosaic(bases, pH = pH, Eh = Eh, T = T)
# Make a diagram and add water stability lines
d &lt;- diagram(m1$A.species, lwd = 2)
water.lines(d, col = "seagreen", lwd = 1.5)
# Show lines for Fe(aq) = 10^-4 M
species(c("Fe+2", "Fe+3"), -4)
m2 &lt;- mosaic(bases, pH = pH, Eh = Eh, T = T)
diagram(m2$A.species, add = TRUE, names = FALSE)
title(main=paste("Iron oxides and sulfides in water, log(total S) = -6",
  "After Garrels and Christ, 1965", sep="\n"))
legend("bottomleft", c("log(act_Fe) = -4", "log(act_Fe) = -6"), lwd = c(2, 1), bty = "n")
# We could overlay the predominance fields for the basis species
#diagram(m1$A.bases, add = TRUE, col = "blue", col.names = "blue", lty = 3)
</code></pre>

<hr>
<h2 id='NaCl'>Simple NaCl-Water Solution</h2><span id='topic+NaCl'></span>

<h3>Description</h3>

<p>Calculate speciation and ionic strength of aqueous solutions with a given molality of NaCl.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  NaCl(m_tot = 1, T = 25, P = "Psat", pH = NA, attenuate = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NaCl_+3A_m_tot">m_tot</code></td>
<td>
<p>numeric, total molality of NaCl (single value)</p>
</td></tr>
<tr><td><code id="NaCl_+3A_t">T</code></td>
<td>
<p>numeric, temperature in &deg;C</p>
</td></tr>
<tr><td><code id="NaCl_+3A_p">P</code></td>
<td>
<p>numeric, pressure in bar</p>
</td></tr>
<tr><td><code id="NaCl_+3A_ph">pH</code></td>
<td>
<p>numeric, pH</p>
</td></tr>
<tr><td><code id="NaCl_+3A_attenuate">attenuate</code></td>
<td>
<p>logical, halve changes of variables in each step?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Thermodynamic models for metal solubility and speciation involving chloride complexes are commonly specified in terms of amount of NaCl rather than activity (or molality) of Cl<sup>-</sup> as an independent variable.
This function calculates distribution of species and ionic strength in a simple aqueous solution given a total amount (<code>m_tot</code>, in mol/kg) of NaCl.
The aqueous Cl-bearing species considered in the system are Cl<sup>-</sup>, NaCl, and optionally HCl.
Na<sup>+</sup> is present as a basis species, but the formation of Na-bearing species such as NaOH is not considered.
The activity coefficients of charged species are calculated using the extended Debye-Hückel equation (see <code><a href="#topic+nonideal">nonideal</a></code>) via the <code>IS</code> argument of <code><a href="#topic+affinity">affinity</a></code>.
The function first sets the molality of Na<sup>+</sup> and ionic strength equal to <code>m_tot</code>, then calculates the distribution of Cl-bearing species.
Based on mass balance of Na atoms, the molality of NaCl is then used to recalculate the molality of Na<sup>+</sup>, followed by ionic strength.
To find a solution, the function iterates until the change of molality of Na<sup>+</sup> and ionic strength are both less than <code>m_tot</code> / 100.
</p>
<p>At very high NaCl concentrations, which are beyond the applicability limits of the extended Debye-Hückel model and therefore not recommended for normal use, the iterations tend to oscillate without converging.
Setting <code>attenuate</code> to TRUE, which halves the amount of change in each step, may help with convergence.
If a solution is not found after 100 iterations, the function stops with an error.
</p>
<p>If <code>pH</code> is NA (the default), then HCl is not included in the calculation and its molality in the output is also assigned NA.
Note that only a single value is accepted for <code>m_tot</code>, but the other numeric arguments can have length &gt; 1, allowing multiple combinations <code>T</code>, <code>P</code>, and <code>pH</code> in a single function call.
However, due to limitations in <code><a href="#topic+affinity">affinity</a></code>, only one of <code>T</code> and <code>P</code> can have length &gt; 1.
</p>


<h3>Value</h3>

<p>A list with components &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo; (ionic strength calculated from molalities of Na<sup>+</sup> and Cl<sup>-</sup>), &lsquo;<span class="samp">&#8288;m_Cl&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;m_Cl&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;m_NaCl&#8288;</span>&rsquo;, and &lsquo;<span class="samp">&#8288;m_HCl&#8288;</span>&rsquo; (molalities of Na<sup>+</sup>, Cl<sup>-</sup>, NaCl, and HCl).
</p>


<h3>Warning</h3>

<p>It is important to keep in mind the ionic strength limits of the Debye-Hückel equation, but this function doesn't enforce them.
Furthermore, metal-ligand complexing is not calculated by this function, so metal solubility and speciation calculations will be accurate only for relatively insoluble metals in NaCl-dominated solutions.
</p>


<h3>References</h3>

<p>Shvarov, Y. and Bastrakov, E. (1999) HCh: A software package for geochemical equilibrium modelling. User's Guide. <em>Australian Geological Survey Organisation</em> <b>1999/25</b>. <a href="https://pid.geoscience.gov.au/dataset/ga/25473">https://pid.geoscience.gov.au/dataset/ga/25473</a>
</p>


<h3>See Also</h3>

<p>This function is used in a few demos (<code>demo("contour")</code>, <code>demo("gold")</code>, <code>demo("minsol")</code>, <code>demo("sphalerite")</code>).
<code>demo("yttrium")</code> uses the <code>pH</code> argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Ionic strength calculated with HCh version 3.7 (Shvarov and Bastrakov, 1999)
# at 1000 bar, 100, 200, and 300 degress C, and 1 to 6 molal NaCl
m.HCh &lt;- 1:6
IS.HCh &lt;- list(`100` = c(0.992, 1.969, 2.926, 3.858, 4.758, 5.619),
               `300` = c(0.807, 1.499, 2.136, 2.739, 3.317, 3.875),
               `500` = c(0.311, 0.590, 0.861, 1.125, 1.385, 1.642))
# Total molality in the calculation with NaCl()
m_tot &lt;- seq(1, 6, 0.5)
N &lt;- length(m_tot)
# Where we'll put the calculated values
IS.calc &lt;- data.frame(`100` = numeric(N), `300` = numeric(N), `500` = numeric(N))
# NaCl() is *not* vectorized over m_tot, so we use a loop here
for(i in 1:length(m_tot)) {
  NaCl.out &lt;- NaCl(m_tot[i], c(100, 300, 500), P = 1000)
  IS.calc[i, ] &lt;- NaCl.out$IS
}
# Plot ionic strength from HCh and NaCl() as points and lines
col &lt;- c("black", "red", "orange")
plot(c(1,6), c(0,6), xlab = "NaCl (mol/kg)", ylab = axis.label("IS"), type = "n")
for(i in 1:3) {
  # NOTE: the differences are probably mostly due to different models
  # for the properties of NaCl(aq) (HCh: B.Ryhzenko model;
  # CHONSZ: revised HKF with parameters from Shock et al., 1997)
  points(m.HCh, IS.HCh[[i]], col = col[i])
  lines(m_tot, IS.calc[, i], col = col[i])
}
# Add legend and title
dprop &lt;- describe.property(rep("T", 3), c(100, 300, 500))
legend("topleft", dprop, lty = 1, pch = 1, col = col)
title(main="H2O + NaCl; HCh (points) and 'NaCl()' (lines)")
</code></pre>

<hr>
<h2 id='nonideal'>Activity Coefficients of Aqueous Species</h2><span id='topic+nonideal'></span><span id='topic+bgamma'></span>

<h3>Description</h3>

<p>Calculate activity coefficients and adjusted molal properties of aqueous species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  nonideal(species, speciesprops, IS, T, P, A_DH, B_DH,
           m_star = NULL, method = thermo()$opt$nonideal)
  bgamma(TC, P, showsplines = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nonideal_+3A_species">species</code></td>
<td>
<p>name of method to use, or names or indices of species for which to calculate nonideal properties</p>
</td></tr>
<tr><td><code id="nonideal_+3A_speciesprops">speciesprops</code></td>
<td>
<p>list of dataframes of species properties</p>
</td></tr>
<tr><td><code id="nonideal_+3A_is">IS</code></td>
<td>
<p>numeric, ionic strength(s) used in nonideal calculations, mol kg<code class="reqn">^{-1}</code></p>
</td></tr>
<tr><td><code id="nonideal_+3A_t">T</code></td>
<td>
<p>numeric, temperature (K)</p>
</td></tr>
<tr><td><code id="nonideal_+3A_p">P</code></td>
<td>
<p>numeric, pressure (bar); required for B-dot or b_gamma equation</p>
</td></tr>
<tr><td><code id="nonideal_+3A_a_dh">A_DH</code></td>
<td>
<p>numeric, A Debye-Huckel coefficient; required for B-dot or b_gamma equation</p>
</td></tr>
<tr><td><code id="nonideal_+3A_b_dh">B_DH</code></td>
<td>
<p>numeric, B Debye-Huckel coefficient; required for B-dot or b_gamma equation</p>
</td></tr>
<tr><td><code id="nonideal_+3A_m_star">m_star</code></td>
<td>
<p>numeric, total molality of all dissolved species</p>
</td></tr>
<tr><td><code id="nonideal_+3A_method">method</code></td>
<td>
<p>character, &lsquo;<span class="samp">&#8288;Bdot&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Bdot0&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;bgamma&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;bgamma0&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;Alberty&#8288;</span>&rsquo;</p>
</td></tr>
<tr><td><code id="nonideal_+3A_tc">TC</code></td>
<td>
<p>numeric, temperature (&deg;C)</p>
</td></tr>
<tr><td><code id="nonideal_+3A_showsplines">showsplines</code></td>
<td>
<p>character, show isobaric (&lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo;) or isothermal (&lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo;) splines</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>nonideal</code> calculates activity coefficients and adjusted thermodynamic properties for charged and neutral aqueous species.
At the user level, the main use of this function is to set the method for activity coefficient calculations that gets used by other functions in CHNOSZ.
See the &ldquo;Charged Species&rdquo; section for a description of the available methods.
Activity coefficient calculations are activated by setting the <code>IS</code> argument of <code><a href="#topic+subcrt">subcrt</a></code> or <code><a href="#topic+affinity">affinity</a></code>.
Those functions then call <code>nonideal</code> with all the arguments needed to perform the calculations.
</p>


<h3>Charged Species</h3>

<p>The default is to not apply calculations for the proton (H<sup>+</sup>) and electron (<i>e</i><sup>-</sup>); this makes sense if you are setting the pH, i.e. activity of H<sup>+</sup>, to some value.
To apply the calculations to H+ and/or e-, change <code><a href="#topic+thermo">thermo</a>()$opt$ideal.H</code> or <code>ideal.e</code> to FALSE (see examples).
</p>
<p>For the &lsquo;<span class="samp">&#8288;Alberty&#8288;</span>&rsquo; method, the values of <code>IS</code> are combined with Alberty's (2003) equation 3.6-1 (extended Debye-Hückel equation with an empirical term valid up to 0.25 M ionic strength) and its derivatives (Alberty, 2001), to calculate adjusted molal properties at the specified ionic strength(s) and temperature(s).
The calculations use the equation for the Debye-Hückel constant given by Clarke and Glew, 1980, which is valid between 0 and 150 °C at saturated water vapor pressure (<i>P</i><sub>SAT</sub>).
</p>
<p>For the &lsquo;<span class="samp">&#8288;Bdot&#8288;</span>&rsquo; method (the default), the &ldquo;B-dot&rdquo; form of the extended Debye-Hückel equation is used.
This equation is valid at ionic strengths up to approximately 3 mol / kg (Hörbrand et al., 2018).
The distance of closest approach for different ions (the &ldquo;ion size parameter&rdquo;) is taken from <code><a href="#topic+thermo">thermo</a>()$Bdot_acirc</code>; any species not listed in this file is assigned a value of 4.5 &Aring;.
The extended term parameter for NaCl-dominated solutions, known as &ldquo;B-dot&rdquo;, is calculated as a function only of temperature (Helgeson, 1969).
To set the extended term parameter to zero, use the &lsquo;<span class="samp">&#8288;Bdot0&#8288;</span>&rsquo; method.
</p>
<p>For the &lsquo;<span class="samp">&#8288;bgamma&#8288;</span>&rsquo; method, the &ldquo;b_gamma&rdquo; equation is used.
The distance of closest approach is set to a constant (3.72e-8 cm) (e.g., Manning et al., 2013).
The extended term parameter is calculated by calling the <code>bgamma</code> function.
Alternatively, set the extended term parameter to zero with &lsquo;<span class="samp">&#8288;bgamma0&#8288;</span>&rsquo;.
</p>


<h3>Neutral Species</h3>

<p>For neutral species, the Setch&eacute;now equation is used, as described in Shvarov and Bastrakov, 1999.
If <code><a href="#topic+thermo">thermo</a>()$opt$Setchenow</code> is &lsquo;<span class="samp">&#8288;bgamma0&#8288;</span>&rsquo; (the default), the extended term parameter is set to zero and the only non-zero term is the mole fraction to molality conversion factor (using the value of <code>m_star</code>).
If <code>thermo()$opt$Setchenow</code> is &lsquo;<span class="samp">&#8288;bgamma&#8288;</span>&rsquo;, the extended term paramter is taken from the setting for the charged species (which can be either &lsquo;<span class="samp">&#8288;Bdot&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;bgamma&#8288;</span>&rsquo;).
Set <code>thermo()$opt$Setchenow</code> to any other value to disable the calculations for neutral species.
</p>


<h3>Additional Details</h3>

<p>This information, about the arguments and return values used to perform the calculations, is not normally needed by the user (but the usage is shown in the last example).
</p>
<p>For <code>nonideal</code>, the species can be identified by name or species index in <code>species</code>.
<code>speciesprops</code> is a list of dataframes containing the input standard molal properties; normally, at least one column is &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo; for Gibbs energy.
The function calculates the *adjusted* properties for given ionic strength (<code>IS</code>); they are equal to the *standard* values only at IS=0.
The adjusted molal properties that can be calculated include &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo;, and (currently only with the Alberty method) &lsquo;<span class="samp">&#8288;H&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;S&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo;; values of any columns with other names are left untouched.
The lengths of <code>IS</code> and <code>T</code> supplied in the arguments should be equal to the number of rows of each dataframe in <code>speciesprops</code>, or length one to use single values throughout.
</p>
<p>In addition to <code>IS</code> and <code>T</code>, the &lsquo;<span class="samp">&#8288;Bdot&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;bgamma&#8288;</span>&rsquo; methods depend on values of <code>P</code>, <code>A_DH</code>, <code>B_DH</code>, and <code>m_star</code> given in the arguments.
<code>m_star</code>, the total molality of all dissolved species, is used to compute the mole fraction to molality conversion factor.
If <code>m_star</code> is NULL, it is taken to be equal to <code>IS</code>, which is an underestimate.
For these methods, &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo; is currently the only adjusted molal property that is calculated (but this can be used by <code><a href="#topic+subcrt">subcrt</a></code> to calculate adjusted equilibrium constants).
</p>
<p>The return value is the same as the input in <code>speciesprops</code>, except the input standard thermodynamic properties (at IS=0) are replaced by adjusted properties (at higher IS).
For all affected species, a column named <code>loggam</code> (common (base-10) logarithm of gamma, the activity coefficient) is appended to the output dataframe of species properties.
</p>
<p><code>bgamma</code> calculates the extended term parameter (written as b_gamma; Helgeson et al., 1981) for activity coefficients in NaCl-dominated solutions at high temperature and pressure.
Data at <i>P</i><sub>SAT</sub> and 0.5 to 5 kb are taken from Helgeson (1969, Table 2 and Figure 3) and Helgeson et al. (1981, Table 27) and extrapolated values at 10 to 30 kb from Manning et al. (2013, Figure 11).
Furthermore, the 10 to 30 kb data were used to generate super-extrapolated values at 40, 50, and 60 kb, which may be encountered using the <code><a href="#topic+water.DEW">water.DEW</a></code> calculations.
If all <code>P</code> correspond to one of the isobaric conditions, the values of <code>Bdot</code> at <code>T</code> are calculated by spline fits to the isobaric data.
Otherwise, particular (dependent on the <code>T</code>) isobaric spline fits are themselves used to construct isothermal splines for the given values of <code>T</code>; the isothermal splines are then used to generate the values of <code>Bdot</code> for the given <code>P</code>.
To see the splines, set <code>showsplines</code> to &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; to make the first set (isobaric splines) along with the data points, or &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo; for examples of isothermal splines at even temperature intervals (here, the symbols are not data, but values generated from the isobaric splines).
This is a basic method of interpolating the data without adding any external dependencies.
</p>


<h3>References</h3>

<p>Alberty, R. A. (2001) Effect of temperature on standard transformed Gibbs energies of formation of reactants at specified pH and ionic strength and apparent equilibrium constants of biochemical reactions. <em>J. Phys. Chem. B</em> <b>105</b>, 7865&ndash;7870. <a href="https://doi.org/10.1021/jp011308v">doi:10.1021/jp011308v</a>
</p>
<p>Alberty, R. A. (2003) <em>Thermodynamics of Biochemical Reactions</em>, John Wiley &amp; Sons, Hoboken, New Jersey, 397 p. <a href="https://www.worldcat.org/oclc/51242181">https://www.worldcat.org/oclc/51242181</a>
</p>
<p>Clarke, E. C. W. and Glew, D. N. (1980) Evaluation of Debye-Hückel limiting slopes for water between 0 and 150 °C. <em>J. Chem. Soc. Faraday Trans.</em> <b>76</b>, 1911&ndash;1916. <a href="https://doi.org/10.1039/f19807601911">doi:10.1039/f19807601911</a>
</p>
<p>Helgeson, H. C. (1969) Thermodynamics of hydrothermal systems at elevated temperatures and pressures. <em>Am. J. Sci.</em> <b>267</b>, 729&ndash;804. <a href="https://doi.org/10.2475/ajs.267.7.729">doi:10.2475/ajs.267.7.729</a>
</p>
<p>Helgeson, H. C., Kirkham, D. H. and Flowers, G. C. (1981) Theoretical prediction of the thermodynamic behavior of aqueous electrolytes at high pressures and temperatures. IV. Calculation of activity coefficients, osmotic coefficients, and apparent molal and standard and relative partial molal properties to 600&deg;C and 5 Kb. <em>Am. J. Sci.</em> <b>281</b>, 1249&ndash;1516. <a href="https://doi.org/10.2475/ajs.281.10.1249">doi:10.2475/ajs.281.10.1249</a>
</p>
<p>Hörbrand, T., Baumann, T. and Moog, H. C. (2018) Validation of hydrogeochemical databases for problems in deep geothermal energy. <em>Geotherm. Energy</em> <b>6</b>, 20. <a href="https://doi.org/10.1186/s40517-018-0106-3">doi:10.1186/s40517-018-0106-3</a>
</p>
<p>Manning, C. E. (2013) Thermodynamic modeling of fluid-rock interaction at mid-crustal to upper-mantle conditions. <em>Rev. Mineral. Geochem.</em> <b>76</b>, 135&ndash;164. <a href="https://doi.org/10.2138/rmg.2013.76.5">doi:10.2138/rmg.2013.76.5</a>
</p>
<p>Manning, C. E., Shock, E. L. and Sverjensky, D. A. (2013) The chemistry of carbon in aqueous fluids at crustal and upper-mantle conditions: Experimental and theoretical constraints. <em>Rev. Mineral. Geochem.</em> <b>75</b>, 109&ndash;148. <a href="https://doi.org/10.2138/rmg.2013.75.5">doi:10.2138/rmg.2013.75.5</a>
</p>
<p>Shvarov, Y. and Bastrakov, E. (1999) HCh: A software package for geochemical equilibrium modelling. User's Guide. <em>Australian Geological Survey Organisation</em> <b>1999/25</b>. <a href="https://pid.geoscience.gov.au/dataset/ga/25473">https://pid.geoscience.gov.au/dataset/ga/25473</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Each of the available methods
nonideal("Alberty")
nonideal("bgamma0")
nonideal("bgamma")
nonideal("Bdot0")
nonideal("Bdot") # the default

## What's the activity coefficient of Na+ at
## 25 degC and 1 bar and an ionic strength of 0.7?
sres &lt;- subcrt("Na+", T = 25, IS = 0.7)
# Exponentiate to convert log10(gamma) to gamma
10^sres$out[[1]]$loggam
# Now use a different method
nonideal("bgamma")
sres &lt;- subcrt("Na+", T = 25, IS = 0.7)
10^sres$out[[1]]$loggam

## What are activity coefficients of -3, -2, -1, 0, +1, +2, +3 charged species
## as a function of ionic strength and temperature?
# First choose the method
nonideal("Bdot")
# Define the ionic strength and temperature increments
IS &lt;- c(0.001, 0.01, 0.1, 0.7)
T &lt;- seq(0, 100, 25)
# Use species with charges -3, -2, -1, 0, +1, +2, +3
species &lt;- c("PO4-3", "HPO4-2", "H2PO4-", "H3PO4", "Na+", "Ca+2", "Al+3")
# Initialize empty output table for T (rows) and charge (columns)
gamtab &lt;- matrix(nrow = length(T), ncol = length(species))
rownames(gamtab) &lt;- T
colnames(gamtab) &lt;- -3:3
# Make a list of tables to hold the activity coefficients, one for each IS
gamma &lt;- rep(list(gamtab), length(IS))
names(gamma) &lt;- IS
# Loop over the values of ionic strength
for(i in seq_along(IS)) {
  # Calculate properties of species, including logarithm of activity coefficient
  sres &lt;- subcrt(species, T = T, IS = IS[i])
  # Exponentiate to convert log10(gamma) to gamma, and put the values into the tables
  for(j in seq_along(species)) gamma[[i]][, j] &lt;- 10^sres$out[[j]]$loggam
}
# Print the output and make a plot
print(gamma)
matplot(T, gamma$`0.001`, type = "l")
title(main = "activity coefficients of -3, -2, -1, 0, +1, +2, +3 charged species")

## Alberty, 2003 p. 16 Table 1.3: adjusted pKa of acetic acid
## We use the 'IS' argument in subcrt() to calculate adjusted thermodynamic properties
# Set ideal.H to FALSE to calculate activity coefficients for the proton
# (needed for replication of the values in Alberty's book)
nonideal("Alberty")
thermo("opt$ideal.H" = FALSE)
sres &lt;- subcrt(c("acetate", "H+", "acetic acid"), c(-1, -1, 1),
  IS=c(0, 0.1, 0.25), T=25, property="logK")
# We're within 0.01 of Alberty's pK values
Alberty_logK &lt;- c(4.75, 4.54, 4.47)
# The maximum (absolute) pairwise difference between x and y
max(abs(Alberty_logK - sres$out$logK)) # 0.0072
# Reset option to default
thermo("opt$ideal.H" = TRUE)

## An example using IS with affinity():
## Speciation of phosphate as a function of ionic strength
opar &lt;- par(mfrow = c(2, 1))
basis("CHNOPS+")
Ts &lt;- c(25, 100)
species(c("PO4-3", "HPO4-2", "H2PO4-"))
for(T in Ts) {
  a &lt;- affinity(IS = c(0, 0.14), T = T)
  e &lt;- equilibrate(a)
  if(T==25) diagram(e, ylim = c(-3.0, -2.6), legend.x = NULL)
  else diagram(e, add = TRUE, names = FALSE, col = "red")
}
title(main="Non-ideality model for phosphate species")
dp &lt;- describe.property(c("pH", "T", "T"), c(7, Ts))
legend("topright", lty = c(NA, 1, 1), col = c(NA, "black", "red"), legend = dp)
text(0.07, -2.76, expr.species("HPO4-2"))
text(0.07, -2.90, expr.species("H2PO4-"))
## Phosphate predominance f(IS,pH)
a &lt;- affinity(IS = c(0, 0.14), pH = c(6, 13), T = Ts[1])
d &lt;- diagram(a, fill = NULL)
a &lt;- affinity(IS = c(0, 0.14), pH = c(6, 13), T = Ts[2])
d &lt;- diagram(a, add = TRUE, names = FALSE, col = "red")
par(opar)

## Activity coefficients for monovalent ions at 700 degC, 10 kbar
# After Manning, 2013, Fig. 7
# Here we use the b_gamma equation
nonideal("bgamma")
IS &lt;- c(0.001, 0.01, 0.1, 1, 2, 2.79)
# We're above 5000 bar, so need to use IAPWS-95 or DEW
oldwat &lt;- water("DEW")
sres &lt;- subcrt("Na+", T = 700, P = 10000, IS = IS)
water(oldwat)
# Compare the calculated activity coefficient to values from Manning's figure
gamma &lt;- 10^sres$out[[1]]$loggam
Manning_gamma &lt;- c(0.93, 0.82, 0.65, 0.76, 1.28, 2)
gamma - Manning_gamma

## Plot the data and splines used for calculating b_gamma
## (extended term parameter)
bgamma(showsplines = "T")
bgamma(showsplines = "P")

## A longer example, using nonideal() directly
# Alberty, 2003 p. 273-276: activity coefficient (gamma)
# as a function of ionic strength and temperature
nonideal("Alberty")
IS &lt;- seq(0, 0.25, 0.005)
T &lt;- c(0, 25, 40)
lty &lt;- 1:3
species &lt;- c("H2PO4-", "HADP-2", "HATP-3", "ATP-4")
col &lt;- rainbow(4)
thermo.plot.new(xlim = range(IS), ylim = c(0, 1),
  xlab = axis.label("IS"), ylab = "gamma")
for(j in 1:3) {
  # Use subcrt to generate speciesprops
  speciesprops &lt;- subcrt(species, T = rep(T[j], length(IS)))$out
  # Use nonideal to calculate loggamma; this also adjusts G, H, S, Cp,
  # but we don't use them here
  nonidealprops &lt;- nonideal(species, speciesprops, IS = IS, T = convert(T[j], "K"))
  for(i in 1:4) lines(IS, 10^(nonidealprops[[i]]$loggam), lty=lty[j], col=col[i])
}
t1 &lt;- "Activity coefficient (gamma) of -1,-2,-3,-4 charged species"
t2 &lt;- quote("at 0, 25, and 40 "*degree*"C, after Alberty, 2003")
mtitle(as.expression(c(t1, t2)))
legend("topright", lty=c(NA, 1:3), bty="n",
  legend=c(as.expression(axis.label("T")), 0, 25, 40))
legend("top", lty=1, col=col, bty="n",
  legend = as.expression(lapply(species, expr.species)))

## Reset method to default
nonideal("Bdot")  # or reset()
</code></pre>

<hr>
<h2 id='palply'>Conditional Parallel Processing</h2><span id='topic+palply'></span>

<h3>Description</h3>

<p>Use multiple processors for large calculations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  palply(varlist, X, FUN, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="palply_+3A_...">...</code></td>
<td>
<p>equivalent to the same argument in <code>parallel::parLapply</code></p>
</td></tr>
<tr><td><code id="palply_+3A_varlist">varlist</code></td>
<td>
<p>character, names of variables to export using <code>parallel::clusterExport</code></p>
</td></tr>
<tr><td><code id="palply_+3A_x">X</code></td>
<td>
<p>vector, argument for <code><a href="base.html#topic+lapply">lapply</a></code> or <code>parLapply</code></p>
</td></tr>
<tr><td><code id="palply_+3A_fun">FUN</code></td>
<td>
<p>function, argument for <code>lapply</code> or <code>parLapply</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>palply</code> is a wrapper function to run <code><a href="parallel.html#topic+clusterApply">parallel::parLapply</a></code> if length of <code>X</code> &gt; <code><a href="#topic+thermo">thermo</a>()$opt$paramin</code> and multiple cores are available, otherwise it runs <code><a href="base.html#topic+lapply">lapply</a></code>.
Note that <code>parLapply</code> is called with <code>methods</code> set to FALSE.
If lots of package startup messages are created when running <code>parallel::makeCluster</code> (which is called by <code>palply</code>), it can probably be stopped by adding a test for <code><a href="base.html#topic+interactive">interactive</a></code> sessions around any <code><a href="base.html#topic+library">library</a></code> commands in the <code><a href="base.html#topic+Rprofile">Rprofile</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>, <code><a href="#topic+count.aa">count.aa</a></code>, <code><a href="#topic+affinity">affinity</a></code>, <code><a href="#topic+equil.boltzmann">equil.boltzmann</a></code> and <code><a href="#topic+equil.reaction">equil.reaction</a></code> for functions that use <code>palply</code>.
Tests are in &lsquo;<span class="file">tests/test-util.program.R</span>&rsquo;, and a &ldquo;real world&rdquo; example is in &lsquo;<span class="file">demo/density.R</span>&rsquo;.
</p>

<hr>
<h2 id='protein.info'>Summaries of Thermodynamic Properties of Proteins</h2><span id='topic+protein.info'></span><span id='topic+pinfo'></span><span id='topic+protein.length'></span><span id='topic+protein.formula'></span><span id='topic+protein.OBIGT'></span><span id='topic+protein.basis'></span><span id='topic+protein.equil'></span>

<h3>Description</h3>

<p>Protein information, length, chemical formula, thermodynamic properties by group additivity, reaction coefficients of basis species, and metastable equilibrium example calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  pinfo(protein, organism=NULL, residue=FALSE, regexp=FALSE)
  protein.length(protein, organism = NULL)
  protein.formula(protein, organism = NULL, residue = FALSE)
  protein.OBIGT(protein, organism = NULL, state=thermo()$opt$state)
  protein.basis(protein, T = 25, normalize = FALSE)
  protein.equil(protein, T=25, loga.protein = 0, digits = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="protein.info_+3A_protein">protein</code></td>
<td>
<p>character, names of proteins; numeric, species index of proteins; data frame; amino acid composition of proteins</p>
</td></tr>
<tr><td><code id="protein.info_+3A_organism">organism</code></td>
<td>
<p>character, names of organisms</p>
</td></tr>
<tr><td><code id="protein.info_+3A_residue">residue</code></td>
<td>
<p>logical, return per-residue values (those of the proteins divided by their lengths)?</p>
</td></tr>
<tr><td><code id="protein.info_+3A_regexp">regexp</code></td>
<td>
<p>logical, find matches using regular expressions?</p>
</td></tr>
<tr><td><code id="protein.info_+3A_normalize">normalize</code></td>
<td>
<p>logical, return per-residue values (those of the proteins divided by their lengths)?</p>
</td></tr>
<tr><td><code id="protein.info_+3A_state">state</code></td>
<td>
<p>character, physical state</p>
</td></tr>
<tr><td><code id="protein.info_+3A_t">T</code></td>
<td>
<p>numeric, temperature in &deg;C</p>
</td></tr>
<tr><td><code id="protein.info_+3A_loga.protein">loga.protein</code></td>
<td>
<p>numeric, decimal logarithms of reference activities of proteins</p>
</td></tr>
<tr><td><code id="protein.info_+3A_digits">digits</code></td>
<td>
<p>integer, number of significant digits (see <code><a href="base.html#topic+signif">signif</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For character <code>protein</code>, <code>pinfo</code> returns the rownumber(s) of <code>thermo()$protein</code> that match the protein names.
The names can be supplied in the single <code>protein</code> argument (with an underscore, denoting protein_organism) or as pairs of <code>protein</code>s and <code>organism</code>s.
NA is returned for any unmatched proteins, including those for which no <code>organism</code> is given or that do not have an underscore in <code>protein</code>.
</p>
<p>Alternatively, if <code>regexp</code> is TRUE, the <code>protein</code> argument is used as a pattern (regular expression); rownumbers of all matches of <code>thermo()$protein$protein</code> to this pattern are returned.
When using <code>regexp</code>, the <code>organism</code> can optionally be provided to return only those entries that also match <code>thermo()$protein$organism</code>.
</p>
<p>For numeric <code>protein</code>, <code>pinfo</code> returns the corresponding row(s) of <code>thermo()$protein</code>.
Set <code>residue</code> to TRUE to return the per-residue composition (i.e. amino acid composition of the protein divided by total number of residues).
</p>
<p>For dataframe <code>protein</code>, <code>pinfo</code> returns it unchanged, except for possibly the per-residue calculation.
</p>
<p>The following functions accept any specification of protein(s) described above for <code>pinfo</code>:
</p>
<p><code>protein.length</code> returns the lengths (number of amino acids) of the proteins.
</p>
<p><code>protein.formula</code> returns a stoichiometrix matrix representing the chemical formulas of the proteins that can be pased to e.g. <code><a href="#topic+mass">mass</a></code> or <code><a href="#topic+ZC">ZC</a></code>.
The amino acid compositions are multiplied by the output of <code><a href="#topic+group.formulas">group.formulas</a></code> to generate the result. 
</p>
<p><code>protein.OBIGT</code> calculates the thermodynamic properties and equations-of-state parameters for the completely nonionized proteins using group additivity with parameters taken from Dick et al., 2006 (aqueous proteins) and LaRowe and Dick, 2012 (crystalline proteins and revised aqueous methionine sidechain group).
The return value is a data frame in the same format as <code>thermo()$OBIGT</code>.
<code>state</code> indicates the physical state for the parameters used in the calculation (&lsquo;<span class="samp">&#8288;aq&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo;).
</p>
<p>The following functions also depend on an existing definition of the basis species:
</p>
<p><code>protein.basis</code> calculates the numbers of the basis species (i.e. opposite of the coefficients in the formation reactions) that can be combined to form the composition of each of the proteins.
The basis species must be present in <code>thermo()$basis</code>, and if &lsquo;<span class="samp">&#8288;H+&#8288;</span>&rsquo; is among the basis species, the ionization states of the proteins are included.
The ionization state of the protein is calculated at the pH defined in <code>thermo()$basis</code> and at the temperature specified by the <code>T</code> argument.
If <code>normalize</code> is TRUE, the coefficients on the basis species are divided by the lengths of the proteins. 
</p>
<p><code>protein.equil</code> produces a series of messages showing step-by-step a calculation of the chemical activities of proteins in metastable equilibrium. For the first protein, it shows the standard Gibbs energies of the reaction to form the nonionized protein from the basis species and of the ionization reaction of the protein (if &lsquo;<span class="samp">&#8288;H+&#8288;</span>&rsquo; is in the basis), then the standard Gibbs energy/RT of the reaction to form the (possibly ionized) protein per residue. The per-residue values of &lsquo;<span class="samp">&#8288;logQstar&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;Astar/RT&#8288;</span>&rsquo; are also shown for the first protein. Equilibrium calculations are then performed, only if more than one protein is specified. This calculation applies the Boltzmann distribution to the calculation of the equilibrium degrees of formation of the residue equivalents of the proteins, then converts them to activities of proteins taking account of <code>loga.protein</code> and protein length. If the <code>protein</code> argument is numeric (indicating rownumbers in <code>thermo()$protein</code>), the values of &lsquo;<span class="samp">&#8288;Astar/RT&#8288;</span>&rsquo; are compared with the output of <code><a href="#topic+affinity">affinity</a></code>, and those of the equilibrium degrees of formation of the residues and the chemical activities of the proteins with the output of <code><a href="#topic+diagram">diagram</a></code>. If the values in any of these tests are are not <code><a href="base.html#topic+all.equal">all.equal</a></code> an error is produced indicating a bug. 
</p>


<h3>References</h3>

<p>Dick, J. M., LaRowe, D. E. and Helgeson, H. C. (2006) Temperature, pressure, and electrochemical constraints on protein speciation: Group additivity calculation of the standard molal thermodynamic properties of ionized unfolded proteins. <em>Biogeosciences</em> <b>3</b>, 311&ndash;336. <a href="https://doi.org/10.5194/bg-3-311-2006">doi:10.5194/bg-3-311-2006</a>
</p>
<p>LaRowe, D. E. and Dick, J. M. (2012) Calculation of the standard molal thermodynamic properties of crystalline peptides. <em>Geochim. Cosmochim. Acta</em> <b>80</b>, 70&ndash;91. <a href="https://doi.org/10.1016/j.gca.2011.11.041">doi:10.1016/j.gca.2011.11.041</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Search by name in thermo()$protein
# These are the same: ip1 == ip2
ip1 &lt;- pinfo("LYSC_CHICK")
ip2 &lt;- pinfo("LYSC", "CHICK")
# Two organisms with the same protein name
ip3 &lt;- pinfo("MYG", c("HORSE", "PHYCA"))
# Their amino acid compositions
pinfo(ip3)
# Their thermodynamic properties by group additivity
protein.OBIGT(ip3)

# An unknown protein name gives NA
ip4 &lt;- pinfo("MYGPHYCA")

## Example for chicken lysozyme C
# Index in thermo()$protein
ip &lt;- pinfo("LYSC_CHICK")
# Amino acid composition
pinfo(ip)
# Protein length and chemical formula
protein.length(ip)
protein.formula(ip)
# Group additivity for thermodynamic properties and HKF equation-of-state
# parameters of non-ionized protein
protein.OBIGT(ip)
# Calculation of standard thermodynamic properties
# (subcrt uses the species name, not ip)
subcrt("LYSC_CHICK")
# NOTE: subcrt() only shows the properties of the non-ionized
# protein, but affinity() uses the properties of the ionized
# protein if the basis species have H+

## These are all the same
protein.formula("P53_PIG")
protein.formula(pinfo("P53_PIG"))
protein.formula(pinfo(pinfo("P53_PIG")))
</code></pre>

<hr>
<h2 id='rank.affinity'>Average Ranks of Chemical Affinities</h2><span id='topic+rank.affinity'></span>

<h3>Description</h3>

<p>Affinity rankings for groups of species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  rank.affinity(aout, groups, percent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rank.affinity_+3A_aout">aout</code></td>
<td>
<p>list, output of <code><a href="#topic+affinity">affinity</a></code></p>
</td></tr>
<tr><td><code id="rank.affinity_+3A_groups">groups</code></td>
<td>
<p>named list of indices (integer or numeric) for species in each group</p>
</td></tr>
<tr><td><code id="rank.affinity_+3A_percent">percent</code></td>
<td>
<p>return average rank percentage for each group</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The affinities for all species are <code><a href="base.html#topic+rank">rank</a></code>ed, then the mean ranking for the species in each group is calculated.
The mean rankings of groups are converted to a percentage, or returned as-is if <code>percent</code> is FALSE.
Note that the calculations are applied to each set of conditions individually (i.e., each grid point in the affinity <code><a href="#topic+affinity">affinity</a></code> calculation).
</p>


<h3>Value</h3>

<p>The average rankings are inserted into the <code>values</code> element of <code>aout</code>, and the names of the groups are inserted into the <code>species</code> element.
The result can be used by <code><a href="#topic+diagram">diagram</a></code> to make line plots or predominance diagrams (the predominance fields correspond to the groups with highest average ranking of affinity).
</p>


<h3>Note</h3>

<p>The reaction coefficients in the <code>species</code> element of the returned value of <code>aout</code> are not valid.
Because balancing on a basis species (i.e., dividing by its reaction coefficient) would be incorrect, <code>diagram</code> enforces <code>balance = 1</code> so that that average rankings are used without further modification.
</p>


<h3>See Also</h3>

<p><code>demo("rank.affinity")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Compare Rubisco proteins from three domains
datfile &lt;- system.file("extdata/cpetc/rubisco.csv", package = "CHNOSZ")
fastafile &lt;- system.file("extdata/protein/rubisco.fasta", package = "CHNOSZ")
dat &lt;- read.csv(datfile)
aa &lt;- read.fasta(fastafile)
groups &lt;- sapply(c("A", "B", "E"), "==", dat$domain, simplify = FALSE)
names(groups) &lt;- c("Archaea", "Bacteria", "Eukaryota")
ip &lt;- add.protein(aa, as.residue = TRUE)
basis("QEC")
aout &lt;- affinity(O2 = c(-74, -66, 100), H2O = c(-4, 4, 100), iprotein = ip)
arank &lt;- rank.affinity(aout, groups = groups)
nspecies &lt;- sapply(groups, sum)
names &lt;- paste0(names(groups), " (", nspecies, ")")
diagram(arank, fill = "terrain", font = 2, names = names, format.names = FALSE)
title("Average affinity ranking of Rubisco proteins")
</code></pre>

<hr>
<h2 id='retrieve'>Retrieve Species by Element</h2><span id='topic+retrieve'></span>

<h3>Description</h3>

<p>Retrieve species in the database containing one or more chemical elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  retrieve(elements = NULL, ligands = NULL, state = NULL,
    T = NULL, P = "Psat", add.charge = TRUE, hide.groups = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="retrieve_+3A_elements">elements</code></td>
<td>
<p>character, combination of elements, or list, elements in a chemical system</p>
</td></tr>
<tr><td><code id="retrieve_+3A_ligands">ligands</code></td>
<td>
<p>character, elements present in any ligands</p>
</td></tr>
<tr><td><code id="retrieve_+3A_state">state</code></td>
<td>
<p>character, filter the result on these state(s).</p>
</td></tr>
<tr><td><code id="retrieve_+3A_t">T</code></td>
<td>
<p>numeric, temperature where &Delta;<I>G</I>&deg; of species must be not NA</p>
</td></tr>
<tr><td><code id="retrieve_+3A_p">P</code></td>
<td>
<p>numeric, pressure where &Delta;<I>G</I>&deg; of species must be not NA</p>
</td></tr>
<tr><td><code id="retrieve_+3A_add.charge">add.charge</code></td>
<td>
<p>logical, add charge to the system?</p>
</td></tr>
<tr><td><code id="retrieve_+3A_hide.groups">hide.groups</code></td>
<td>
<p>logical, exclude groups from the result?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function retrieves the species in the thermodynamic database (see <code><a href="#topic+thermo">thermo</a></code>) that have the indicated <code>elements</code>.
A character value of <code>elements</code> is interpreted as a combination of one or more elements that must be present in each species.
A list value of <code>elements</code> is used to specify a chemical system &ndash; the species must contain one or more of the indicated elements, but no other elements.
<code>ligands</code>, if present, gives the elements that may be present in any ligands; this can be used to retrieve all species in a system bearing the <code>elements</code> (usually a single metal).
</p>
<p>The result includes charged species if <code>add.charge</code> is TRUE (the default) or the user supplies the &ldquo;element&rdquo; of charge (&lsquo;<span class="samp">&#8288;Z&#8288;</span>&rsquo;).
Results can be filtered on physical state by setting the <code>state</code> argument.
Groups used in group-additivity calculations, which have names with square brackets (e.g. [-CH2-]), are excluded unless <code>hide.groups</code> is FALSE.
A special argument value &lsquo;<span class="samp">&#8288;all&#8288;</span>&rsquo; can be used to retrieve all species in the thermodynamic database, including filtering on state and hiding of the groups.
</p>
<p>The return value is a named integer vector giving the species index (i.e. rownumber(s) of <code>thermo()$OBIGT</code>) with names corresponding to the chemical formulas of the species.
If the electron is in the result, its name (&lsquo;<span class="samp">&#8288;e-&#8288;</span>&rsquo;) is used instead of its chemical formula (&lsquo;<span class="samp">&#8288;(Z-1)&#8288;</span>&rsquo;).
An empty (length 0) integer value is returned if no <code>elements</code> are specified or no species are retrieved.
</p>
<p>Set <code>T</code> (and optionally <code>P</code>) to require that species have non-NA values of &Delta;<I>G</I>&deg; at this temperature and pressure.
These values are passed to <code><a href="#topic+subcrt">subcrt</a></code> and therefore have the units set in <code><a href="#topic+T.units">T.units</a></code> and <code><a href="#topic+P.units">P.units</a></code>.
</p>
<p>To speed up operation, the function uses a precalculated stoichiometric matrix for the default database, which is loaded with the package (see <code><a href="#topic+thermo">thermo</a></code>).
If the function detects a change to any chemical formulas in the database, it updates the stoichiometric matrix using <code><a href="#topic+i2A">i2A</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+info">info</a></code> for basic database searches; <a href="../doc/anintro.html"><strong>anintro</strong></a> for a diagram made with <code>retrieve</code>d species in the Mn-O-H system.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Species index of Ti-bearing minerals
retrieve("Ti")
# Thermodynamic data for those minerals
info(retrieve("Ti"))

# All species that have Au
retrieve("Au")
# All species that have both Au and Cl
retrieve(c("Au", "Cl"))
# Au-Cl system: species that have Au and/or Cl,
# including charged species, but no other elements
retrieve(list("Au", "Cl"))
# All Au-bearing species in the Au-Cl system
retrieve("Au", "Cl")
# All uncharged Au-bearing species in the Au-Cl system
retrieve("Au", "Cl", add.charge = FALSE)

# Minerals in the system SiO2-MgO-CaO-CO2
retrieve(list("Si", "Mg", "Ca", "C", "O"), state = "cr")

</code></pre>

<hr>
<h2 id='solubility'>Equilibrium Chemical Activities of Species</h2><span id='topic+solubility'></span>

<h3>Description</h3>

<p>Calculate chemical activities of aqueous species in equilibrium with a mineral or gas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  solubility(iaq, ..., in.terms.of = NULL, dissociate = FALSE, find.IS = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solubility_+3A_iaq">iaq</code></td>
<td>
<p>character (names) or numeric (species indices) of aqueous species</p>
</td></tr>
<tr><td><code id="solubility_+3A_...">...</code></td>
<td>
<p>arguments for <code><a href="#topic+affinity">affinity</a></code> or <code><a href="#topic+mosaic">mosaic</a></code> (i.e. plotting variables)</p>
</td></tr>
<tr><td><code id="solubility_+3A_in.terms.of">in.terms.of</code></td>
<td>
<p>character, express the total solubility in terms of moles of this species</p>
</td></tr>
<tr><td><code id="solubility_+3A_dissociate">dissociate</code></td>
<td>
<p>logical, does the mineral undergo a dissociation reaction?</p>
</td></tr>
<tr><td><code id="solubility_+3A_find.is">find.IS</code></td>
<td>
<p>logical, find the equilibrium ionic strength by iteration?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>solubility</code> calculates the activities of aqueous species in equilibrium with one or more minerals or gases.
The minerals or gases should be loaded as the formed <code><a href="#topic+species">species</a></code>, and the aqueous species (including ions and/or neutral complexes) that can be produced by dissolution should be listed in the <code>iaq</code> argument.
The definitions of plotting variables should be provided in <code>...</code>, which are passed as arguments to <code><a href="#topic+affinity">affinity</a></code>, or to <code><a href="#topic+mosaic">mosaic</a></code> if the first one is named <code>bases</code>.
</p>
<p>It must be possible to obtain a valid set of basis species by substituting each of the minerals or gases in the first position of the current <code><a href="#topic+basis">basis</a></code> defintion, and all of the aqueous species must include that basis species in their formation reactions.
(This essentially means that all minerals, gases and aqueous species must share a common element, which is what the reactions are balanced on.)
</p>
<p>For a single mineral or gas, the output of <code>solubility</code> can be used by <code><a href="#topic+diagram">diagram</a></code> with <code>type = NULL</code> (the default) to plot the activities of the aqueous species or with <code>type = "loga.balance"</code> to plot the sum of activities of aqueous species, which corresponds to the solubility of the mineral or gas.
This value corresponds to the total extent of dissolution of the mineral or gas; <code>in.terms.of</code> can be used to express this value in terms of another species or element.
For example, for dissolution of gaseous S<sub>2</sub>, <code>in.terms.of = "S"</code> gives the total amount of S in solution, which is twice the amount of S<sub>2</sub> dissolved.
Likewise, the solubility of corundum (Al<sub>2</sub>O<sub>3</sub>) can be expressed in terms of the moles of Al<sup>+3</sup> in solution (see the vignette <a href="../doc/anintro.html"><strong>anintro</strong></a>).
</p>
<p>For multiple minerals, the function calculates the solubilities for each of the minerals separately; these are stored in the <code>loga.equil</code> element of the output.
The overall <em>minimum</em> solubility among all the minerals at each point is stored in <code>loga.balance</code>.
This corresponds to the total activity of dissolved species in equilibrium with the most stable mineral.
In contrast to the situation for a single mineral or gas, <code><a href="#topic+diagram">diagram</a></code> by default plots <code>loga.balance</code>; <code>type = "loga.equil"</code> can be used to plot the solubilities for the individual minerals or gases.
</p>


<h3>Backward Compatibility</h3>

<p>For compatibility with previous versions of the function, the <code>iaq</code> argument can be the output of <code><a href="#topic+affinity">affinity</a></code> or <code><a href="#topic+mosaic">mosaic</a></code> for aqueous species.
The examples for ionic strength and dissociation reactions were designed for this calling style.
</p>
<p>In this case the (single) mineral or gas being dissolved is taken from the current <code><a href="#topic+basis">basis</a></code> species.
Usually, the basis species should be set up so that the first basis species represents the substance being dissolved (a mineral such as CaCO<sub>3</sub> or gas such as CO<sub>2</sub>).
This is treated as the conserved basis species, so it must be present in all of the formation reactions of the aqueous species.
</p>
<p>The <code><a href="#topic+species">species</a></code> should be defined to represent one set of aqueous species (including ions and/or neutral complexes) formed in solution, all involving the conserved basis species.
For a dissociation reaction, the second basis species should be used to represent the counterion (cation or anion).
Other variables (pH, ionic strength, activities of other basis species) should be defined in the call to <code><a href="#topic+affinity">affinity</a></code> to make <code>iaq</code>.
</p>


<h3>Dissociation Reactions</h3>

<p>The function perfoms some additional steps to calculate the solubility of something that dissociates (not just dissolves).
For example, the dissolution of calcite (CaCO<sub>3</sub>), involves the release of both calcium ions and different forms of carbonate in solution, depending on the pH.
The equilibrium calculation must take account of the <em>total</em> activity of the shared ion (Ca<sup>+2</sup>), which is unknown at the start of the calculation.
The solution is found by recalculating the affinities, essentially working backward from the assumption that the dissociation didn't occur.
The resulting activities correspond to equilibrium considering the system-wide activity of Ca<sup>+2</sup>.
</p>
<p>A <em>not recommended</em> alternative is to set <code>dissociate</code> to a numeric value corresponding to the number of dissociated species (i.e. 2 for a 1:1 electrolyte).
This setting indicates to calculate activities on a per-reaction basis, where each reaction has its own (independent) activity of Ca<sup>+2</sup>.
That does not give a complete equilibrium in the system, but may be required to reproduce some published diagrams (see comment in the calcite example of <code>demo("solubility")</code>).
</p>


<h3>Ionic Strength</h3>

<p>Set <code>find.IS</code> to TRUE to determine the final ionic strength due to dissolution of a substance in pure water.
This works by calculating the ionic strength from the amounts of aqueous species formed, then re-running <code>affinity</code> with the calculated <code>IS</code> value.
Note that for dissociation reactions, the ionic strength is calculated from both the ions present in the species definition and the counter ion, which should be the second basis species.
The calculation is iterated until the ionic strength deviation at every point is lower than a preset tolerance (1e-4).
Alternatively, speciation of counterions (e.g. ionized forms of carbonate or sulfate) can also be performed by using the <code><a href="#topic+mosaic">mosaic</a></code> function instead of <code>affinity</code>; this is used in the second example below.
</p>


<h3>Warning</h3>

<p>This function has not been tested for systems that may form dimers or higher-order complexes (such as Au<sub>2</sub>S<sub>2</sub><sup>2-</sup>).
Except for relatively simple systems, even after careful refinement, the results from CHNOSZ, which considers chemical activities as the independent variables, will not match the results from speciation-solubility (or Gibbs energy minimization) codes, where the system is defined by its bulk composition.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+retrieve">retrieve</a></code> provides a way to list all of the aqueous species in the database that have the specified elements.
</p>
<p><code>demo("solubility")</code> shows solubilities of CO<sub>2</sub> and calcite calculated as a function of pH and T; note that for calcite, the <code>dissociate</code> argument is set to TRUE.
<code>demo("gold")</code> shows solubility calculations for Au in aqueous solutions with hydroxide, chloride, and hydrosulfide complexes.
</p>
<p>Solubility calculations for multiple minerals are used for generating isosolubility (aka. equisolubility) lines in <code>demo("Pourbaix")</code> and <code>demo("minsol")</code>.
The latter demo combines the calculation of solubilities with a <code><a href="#topic+mosaic">mosaic</a></code> calculation to account for the speciation of aqueous sulfur.
</p>
<p>Whereas <code>solubility</code> yields a stable equilibrium condition (the affinities of formation reactions of aqueous species are zero), <code><a href="#topic+equilibrate">equilibrate</a></code> generates metastable equilibrium (the affinities of formation reactions are equal to each other, but not necessarily zero).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

## EXAMPLE 1

# Calculate solubility of a single substance:
# Gaseous SO2 with a given fugacity
# Define basis species (any S-bearing basis species should be first)
basis(c("sulfur", "oxygen", "H2O", "H+"))
basis("pH", 6)
# Load the substances (minerals or gases) to be dissolved
species("sulfur dioxide", -20)
# List the formed aqueous species
# We can use retrieve() to identify all the possible aqueous species
iaq &lt;- retrieve("S", c("O", "H"), "aq")
# Place arguments for affinity() after the first argument of solubility()
s1 &lt;- solubility(iaq, O2 = c(-56, -46), T = 125, in.terms.of = "S")

# Calculate overall solubility for multiple substances:
# Gaseous S2 and SO2 with a given fugacity
basis(c("sulfur", "oxygen", "H2O", "H+"))
basis("pH", 6)
species(c("S2", "sulfur dioxide"), -20)
s2 &lt;- solubility(iaq, O2 = c(-56, -46), T = 125, in.terms.of = "S")

# Make expressions for legends
S_ &lt;- expr.species("SO2", "gas", -20, TRUE)
pH_ &lt;- quote(pH == 6)
T_ &lt;- lT(125)
lexpr &lt;- lex(S_, pH_, T_)

# Make diagrams from the results of solubility calculations
layout(matrix(c(1, 3, 2, 3), nrow = 2))
# Logarithm of activity of aqueous species in equilibrium with SO2(gas)
diagram(s1, ylim = c(-15, 0))
diagram(s1, type = "loga.balance", col = 3, lwd = 3, add = TRUE)
legend("topright", legend = lexpr, bty = "n")
# Logarithm of concentration (parts per million) of aqueous species
sppm &lt;- convert(s1, "logppm")
diagram(sppm, ylim = c(-10, 5))
diagram(sppm, type = "loga.balance", col = 3, lwd = 3, add = TRUE)
legend("topright", legend = lexpr, bty = "n")
par(xpd = NA)
text(-58, 6.5, paste("Solubility of gaseous SO2 (green line) is",
  "sum of concentrations of aqueous species"), cex = 1.2, font = 2)
par(xpd = FALSE)

# Show overall (minimum) solubility of multiple gases
diagram(s2, col = 4, lwd = 3)
# Show solubilities of individual gases
names &lt;- info(species()$ispecies)$formula
diagram(s2, type = "loga.equil", names = names, add = TRUE)
title("Minimum solubility (blue line) corresponds to the most stable gas")
layout(matrix(1))

## EXAMPLE 2

## Two ways to calculate pH-dependent solubility of calcite
## with ionic strength determination
## Method 1: CO2 and carbonate species as formed species
basis(c("CO2", "Ca+2", "H2O", "O2", "H+"))
species("calcite")
iaq &lt;- info(c("CO2", "HCO3-", "CO3-2"))
# Ionic strength calculations don't converge below around pH = 3
sa0 &lt;- solubility(iaq, pH = c(4, 14), dissociate = TRUE)
saI &lt;- solubility(iaq, pH = c(4, 14), dissociate = TRUE, find.IS = TRUE)
## Method 2: CO2 and carbonate species as basis species
basis(c("Ca+2", "CO2", "H2O", "O2", "H+"))
species("calcite")
iaq &lt;- info("Ca+2")
bases &lt;- c("CO2", "HCO3-", "CO3-2")
sm0 &lt;- solubility(iaq, bases = bases, pH = c(4, 14), dissociate = TRUE)
smI &lt;- solubility(iaq, bases = bases, pH = c(4, 14), dissociate = TRUE, find.IS = TRUE)
## Plot the results
plot(0, 0, xlab="pH", ylab="solubility, log mol", xlim = c(4, 14), ylim = c(-5, 2))
# Method 1 with/without ionic strength
lines(saI$vals[[1]], saI$loga.balance, lwd = 5, col = "lightblue")
lines(sa0$vals[[1]], sa0$loga.balance, lwd = 5, col = "pink")
# Method 2 with/without ionic strength
lines(smI$vals[[1]], smI$loga.balance, lty = 2)
lines(sm0$vals[[1]], sm0$loga.balance, lty = 2)
legend("topright", c("I = 0", "I = calculated", "mosaic method"),
       col = c("pink", "lightblue", "black"), lwd = c(5, 5, 1), lty = c(1, 1, 2))
title(main = "Solubility of calcite: Ionic strength and mosaic method")
# The two methods give nearly equivalent results
stopifnot(all.equal(sa0$loga.balance, sm0$loga.balance))
stopifnot(all.equal(saI$loga.balance, smI$loga.balance, tolerance = 0.003))
## NOTE: the second method (using mosaic) is slower, but is
## more flexible; e.g. complexes with Ca+2 could be included
</code></pre>

<hr>
<h2 id='species'>Species of Interest</h2><span id='topic+species'></span>

<h3>Description</h3>

<p>Define the species of interest in a system; modify their physical states and logarithms of activities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  species(species = NULL, state = NULL, delete = FALSE, add = FALSE,
    index.return = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="species_+3A_species">species</code></td>
<td>
<p>character, names or formulas of species to add to the species definition; numeric, rownumbers of species to modify or delete</p>
</td></tr>
<tr><td><code id="species_+3A_state">state</code></td>
<td>
<p>character, physical states; numeric, logarithms of activities or fugacities</p>
</td></tr>
<tr><td><code id="species_+3A_delete">delete</code></td>
<td>
<p>logical, delete the species identified by numeric values of <code>species</code> (or all species if that argument is missing)?</p>
</td></tr>  
<tr><td><code id="species_+3A_add">add</code></td>
<td>
<p>logical, delete a previous species definition instead of adding to it?</p>
</td></tr>  
<tr><td><code id="species_+3A_index.return">index.return</code></td>
<td>
<p>logical, return the affected rownumbers of <code>thermo()$species</code> instead of its contents?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After defining the <code><a href="#topic+basis">basis</a></code> species of your system you can use <code>species</code> to identify the species of interest.
A species is uniquely identified by a combination of a <code>name</code> and <code>state</code>, which are columns of the thermodynamic database in <code><a href="#topic+thermo">thermo</a>()$OBIGT</code>.
For each match of <code>species</code> to the name, formula, or abbreviation of a species, and of <code>state</code> to the state (&lsquo;<span class="samp">&#8288;aq&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;gas&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;liq&#8288;</span>&rsquo;), the species is added to the current species definition in <code>thermo()$species</code>.  
</p>
<p>The <code>state</code> argument can be omitted, in which case the first matching species in any state is added (in many cases, this means the aqueous species).
If there are multiple matches for a species, the one that is in the state given by <code>thermo()$opt$state</code> is chosen, otherwise the matching (or <code class="reqn">n</code>'th matching duplicate) species is used.
Note that the <code>state</code>s of species representing phases of minerals that undergo polymorphic transitions are coded as &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; (lowest-T phase), &lsquo;<span class="samp">&#8288;cr2&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;cr3&#8288;</span>&rsquo;, <code>...</code> (phases with increasing temperature).
If <code>state</code> is &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; when one of these minerals is matched, all the polymorphs are added.
</p>
<p>The data frame in <code>thermo()$species</code> holds the species names and indices as well as the stoichiometric reaction coefficients for the formation reaction from the basis species and the logarithms of activities or fugacities that are used by <code><a href="#topic+affinity">affinity</a></code>.
The default values for logarithms of activities are -3 for aqueous (&lsquo;<span class="samp">&#8288;aq&#8288;</span>&rsquo;) species and 0 for others. 
</p>
<p>To modify the logarithms of activities of species (logarithms of fugacities for gases) provide one or more numeric values of <code>species</code> referring to the rownumbers of the species dataframe, or <code>species</code> NULL, to modify all currently defined species.
The values in <code>state</code>, if numeric, are interpreted as the logarithms of activities, or if character are interpreted as new states for the species.
If <code>species</code> is numeric and <code>delete</code> is <code>TRUE</code>, these species are deleted from the dataframe; if the only argument is <code>delete</code> and it is <code>TRUE</code>, all the species are removed.
</p>
<p>By default, when identifying new species, any previous species definition is removed.
Set <code>add</code> to TRUE to add species to an existing definition.
</p>


<h3>Value</h3>

<p>With no arguments or when adding species, <code>species</code> returns the value of <code>thermo()$species</code>, unless <code>index.return</code> is TRUE, when the function returns the rownumbers of <code>thermo()$species</code> having the new species.
With &lsquo;<span class="samp">&#8288;delete=TRUE&#8288;</span>&rsquo;, the value is the definition that existed prior the deletion; with &lsquo;<span class="samp">&#8288;delete=TRUE&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;species&#8288;</span>&rsquo; not NULL, the number of species remaining after the selected ones have been deleted, or <code>NULL</code> if no species remain.
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+info">info</a></code> to search the thermodynamic database without adding species to the system. <code><a href="#topic+basis">basis</a></code> is a prerequisite for <code><a href="#topic+species">species</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Set up the basis species
basis("CHNOS")
# Define some aqueous species
species(c("CO2", "NH3"))
# Add some gases
species(c("CO2", "NH3"), "gas", add = TRUE)
# Delete the aqueous species
species(1:2, delete = TRUE)
# Modify the "logact" value
# (log10 of activity for aqueous species;
#  log10 of fugacity for gases)
species(1:2, c(-2, -5))
# Change the state to aqueous
species(1:2, "aq")
# Load a new species definition (deletes the old one first)
species(c("glycine", "alanine"))
# Delete all the species
species(delete = TRUE)

# Changing the elements in the basis definition
# causes species to be deleted
basis(c("CaO", "CO2", "H2O", "SiO2", "MgO", "O2"))
species(c("dolomite", "quartz", "calcite", "forsterite"))
basis(c("CO2", "H2O", "O2"))
species()  # NULL
</code></pre>

<hr>
<h2 id='stack_mosaic'>Stacked mosaic diagram</h2><span id='topic+stack_mosaic'></span>

<h3>Description</h3>

<p>Create a stacked mosaic diagram, where the species formed in the first layer become the basis species for the species formed in the second layer.
The species in each layer are usually minerals with different metals; any bimetallic species are added to the second layer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  stack_mosaic(bases, species1, species2, species12, names = NULL,
    col = list(4, 3, 6), col.names = list(4, 3, 6), fill = NULL,
    dx = list(0, 0, 0), dy = list(0, 0, 0), srt = list(0, 0, 0),
    lwd = list(1, 1, 1), lty = list(1, 1, 1), loga_aq = NULL,
    plot.it = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stack_mosaic_+3A_bases">bases</code></td>
<td>
<p>basis species to be changed for each layer (commonly S-bearing aqueous species)</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_species1">species1</code></td>
<td>
<p>species (minerals and/or aqueous species) with metal 1</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_species2">species2</code></td>
<td>
<p>species with metal 2</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_species12">species12</code></td>
<td>
<p>bimetallic species</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_names">names</code></td>
<td>
<p>character, species names (or chemical formulas) for labeling fields</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_col">col</code></td>
<td>
<p>line color</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_col.names">col.names</code></td>
<td>
<p>text color</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_fill">fill</code></td>
<td>
<p>field color</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_dx">dx</code></td>
<td>
<p>label x-offset</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_dy">dy</code></td>
<td>
<p>label y-offset</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_srt">srt</code></td>
<td>
<p>label rotation</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_lwd">lwd</code></td>
<td>
<p>line width</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_lty">lty</code></td>
<td>
<p>line type</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_loga_aq">loga_aq</code></td>
<td>
<p>numeric, activity of aqueous species</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_plot.it">plot.it</code></td>
<td>
<p>make plots?</p>
</td></tr>
<tr><td><code id="stack_mosaic_+3A_...">...</code></td>
<td>
<p>arguments for <code><a href="#topic+mosaic">mosaic</a></code> and <code><a href="#topic+affinity">affinity</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>stack_mosaic</code> creates a stacked mosaic diagram following steps that are described in detail in the vignette <a href="../doc/multi-metal.html"><strong>multi-metal</strong></a>.
Briefly, the first layer of the diagram is made by speciating the species in <code>bases</code> across the diagram to form the first set of species in <code>species1</code>.
Then, both <code>bases</code> <em>and</em> <code>species1</code> (the stable species at each point on the diagram) are used to form the second set of species, including those in both <code>species2</code> <em>and</em> <code>species12</code>.
</p>
<p>Note that <code>basis</code> has aqueous S species in the examples provided, and <code>species1</code> consists of minerals and/or aqueous species with a single metal (e.g. Fe).
<code>species2</code> has minerals and/or aqueous species with a second metal (e.g. Cu), and <code>species12</code> has bimetallic minerals.
For &ldquo;mixed&rdquo; diagrams (where <code>species1</code> or <code>species2</code> has both minerals and aqueous species), use <code>loga_aq</code> to set the logarithms of activities of aqueous species.
Here, only a single value of <code>loga_aq</code> is needed, unlike in <code><a href="#topic+mosaic">mosaic</a></code>, where a value for each set of basis species is required.
</p>
<p>The plot parameters <code>col</code>, <code>col.names</code>, <code>fill</code>, <code>dx</code>, <code>dy</code>, <code>srt</code>, <code>lwd</code>, and <code>lty</code> should be length-3 lists (not vectors).
The values of elements 1&ndash;3 of the list are recycled to the number of species in <code>species1</code>, <code>species2</code>, and <code>species12</code>, respectively.
</p>
<p>For <code>fill</code>, the default is to use no fill except for <code>species12</code>, where the fill color is taken from <code>col.names</code> with added transparency.
The default definition of <code>fill</code> is <code>list(NA, NA, add.alpha(col.names[3], "50"))</code>.
</p>


<h3>Value</h3>

<p>A list of length two containing the output of each of the <code><a href="#topic+diagram">diagram</a></code> calls use to make the diagram.
</p>


<h3>Warning</h3>

<p>The bimetallic species in <code>species12</code> are shown as part of the second layer, although their formation is sensitive to the presence of stable species in the first layer.
It follows that changing the order of layers (i.e., swapping <code>species1</code> and <code>species2</code>) can affect the depiction of mineral assemblages that have <code>species12</code>.
It is likely that only one of the alternatives is thermodynamically correct, but currently there is no check to determine which one it is.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define temperature (degrees C), pressure (bar), pH and logfO2 ranges
T &lt;- 200
P &lt;- "Psat"
res &lt;- 200
pH &lt;- c(0, 14, res)
O2 &lt;- c(-48, -33, res)

# Define system: Fe-Cu-O-S-Cl
# NOTE: the basis species must include the first species listed
# in each of bases, species1, and species2 below
basis(c("pyrite", "Cu", "Cl-", "H2S", "H2O", "oxygen", "H+"))
basis("H2S", -2)
# Calculate solution composition for 1 mol/kg NaCl
NaCl &lt;- NaCl(T = T, P = P, m_tot = 1)
basis("Cl-", log10(NaCl$m_Cl))

# Define arguments for stack_mosaic: Speciate aqueous sulfur
bases &lt;- c("H2S", "HS-", "HSO4-", "SO4-2")
# Calculate stabilities of Fe-bearing minerals first
species1 &lt;- c("pyrite", "pyrrhotite", "magnetite", "hematite")
# Calculate stabilities of Cu-bearing and FeCu-bearing minerals second
species2 &lt;- c("copper", "cuprite", "tenorite", "chalcocite", "covellite")
species12 &lt;- c("chalcopyrite", "bornite")
# Use abbreviations for Fe-bearing minerals and formulas for Cu-bearing minerals
names1 &lt;- c("Py", "Po", "Mag", "Hem")
names2 &lt;- info(info(species2))$formula
names12 &lt;- info(info(species12))$formula
names &lt;- list(names1, names2, names12)
# Adjust x-position for one species (chalcocite, Cu2S)
dx &lt;- list(c(0, 0, 0, 0), c(0, 0, 0, 1, 0), c(0, 0))
# Use thick dashed lines for the bimetallic species
lwd &lt;- list(1, 1, 2)
lty &lt;- list(1, 1, 2)

# Make the diagram
stack_mosaic(bases, species1, species2, species12, names = names,
  dx = dx, lwd = lwd, lty = lty,
  pH = pH, O2 = O2, T = T, P = P, IS = NaCl$IS)
# Add legend and title
lTP &lt;- lex(lTP(T, P))
db &lt;- describe.basis(c(3:4))
legend("topright", c(lTP, db), bg = "white")
title("Fe-Cu-S-O-H-Cl", font.main = 1)
</code></pre>

<hr>
<h2 id='subcrt'>Properties of Species and Reactions</h2><span id='topic+subcrt'></span>

<h3>Description</h3>

<p>Calculate the standard molal thermodynamic properties of one or more species or a reaction between species as a function of temperature and pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  subcrt(species, coeff = 1, state = NULL,
    property = c("logK","G","H","S","V","Cp"),
    T = seq(273.15,623.15,25), P = "Psat", grid = NULL, 
    convert = TRUE, exceed.Ttr = FALSE, exceed.rhomin = FALSE,
    logact = NULL, autobalance = TRUE, use.polymorphs = TRUE, IS = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subcrt_+3A_species">species</code></td>
<td>
<p>character, name or formula of species, or numeric, rownumber of species in <code>thermo()$OBIGT</code></p>
</td></tr>
<tr><td><code id="subcrt_+3A_coeff">coeff</code></td>
<td>
<p>numeric, reaction coefficients on species</p>
</td></tr>
<tr><td><code id="subcrt_+3A_state">state</code></td>
<td>
<p>character, state(s) of species</p>
</td></tr>
<tr><td><code id="subcrt_+3A_property">property</code></td>
<td>
<p>character, property(s) to calculate</p>
</td></tr>
<tr><td><code id="subcrt_+3A_t">T</code></td>
<td>
<p>numeric, temperature(s) of the calculation</p>
</td></tr>
<tr><td><code id="subcrt_+3A_p">P</code></td>
<td>
<p>numeric, pressure(s) of the calculation, or character, &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo;</p>
</td></tr>
<tr><td><code id="subcrt_+3A_grid">grid</code></td>
<td>
<p>character, type of <code>P</code><code class="reqn">\times</code><code>T</code> grid to produce (NULL, the default, means no gridding)</p>
</td></tr>
<tr><td><code id="subcrt_+3A_exceed.ttr">exceed.Ttr</code></td>
<td>
<p>logical, calculate Gibbs energies of mineral phases and other species beyond their transition temperatures?</p>
</td></tr>
<tr><td><code id="subcrt_+3A_exceed.rhomin">exceed.rhomin</code></td>
<td>
<p>logical, return properties of species in the HKF model below 0.35 g cm<sup>-3</sup>?</p>
</td></tr>
<tr><td><code id="subcrt_+3A_logact">logact</code></td>
<td>
<p>numeric, logarithms of activities of species in reaction</p>
</td></tr>
<tr><td><code id="subcrt_+3A_convert">convert</code></td>
<td>
<p>logical, are units of T, P, and energy settable by the user (default) (see <code><a href="#topic+T.units">T.units</a></code>)?</p>
</td></tr>
<tr><td><code id="subcrt_+3A_autobalance">autobalance</code></td>
<td>
<p>logical, attempt to automatically balance reaction with basis species?</p>
</td></tr>
<tr><td><code id="subcrt_+3A_use.polymorphs">use.polymorphs</code></td>
<td>
<p>logical, automatically identify available polymorphs in OBIGT and use the stable one at each value of <code>T</code>?</p>
</td></tr>
<tr><td><code id="subcrt_+3A_is">IS</code></td>
<td>
<p>numeric, ionic strength(s) at which to calculate adjusted molal properties, mol kg<code class="reqn">^{-1}</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>subcrt</code> calculates the standard molal thermodynamic properties of species and reactions as a function of temperature and pressure.
For each of the <code>species</code> (a formula or name), optionally identified in a given <code>state</code>, the standard molal thermodynamic properties and equations-of-state parameters are retrieved via <code><a href="#topic+info">info</a></code> (except for H<sub>2</sub>O liquid).
The standard molal properties of the species are computed using the thermodynamic <code>model</code> given for each species (see <code><a href="#topic+thermo">thermo</a></code>).
This function also calculates the thermodynamic properties of <em>reactions</em> by summing those of the respective species.
This functionality is inspired the <abbr><span class="acronym">SUPCRT92</span></abbr> package (Johnson et al., 1992).
</p>
<p><code>T</code> and <code>P</code> denote the temperature and pressure for the calculations.
The only valid non-numeric value is &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; for <code>P</code>, which is the default (see <code><a href="#topic+water">water</a></code>).
For calculations below 273.16 K, <code>P</code> set to 1, as <i>P</i><sub>SAT</sub> is not defined at subzero (&deg;C) temperatures.
At temperatures above the critical point of water, <code>P</code> must be set to a numeric value; unless <code>exceed.rhomin</code> is TRUE, this should correspond to a fluid density &ge; 0.35 g cm<sup>-3</sup>.
</p>
<p>Argument <code><a href="graphics.html#topic+grid">grid</a></code> if present can be one of <code>T</code> or <code>P</code> to perform the computation of a <code>T</code><code class="reqn">\times</code><code>P</code> or <code>P</code><code class="reqn">\times</code><code>T</code> grid.
The <code>property</code>s to be calculated can be one or more of those shown below:
</p>

<table>
<tr>
 <td style="text-align: left;">
         <code>rho</code>     </td><td style="text-align: left;"> Density of water                      </td><td style="text-align: left;"> g cm<code class="reqn">^{-3}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>logK</code>    </td><td style="text-align: left;"> Logarithm of equilibrium constant     </td><td style="text-align: left;"> dimensionless </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>G</code>       </td><td style="text-align: left;"> Gibbs energy                          </td><td style="text-align: left;"> (cal | J) mol<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>H</code>       </td><td style="text-align: left;"> Enthalpy                              </td><td style="text-align: left;"> (cal | J) mol<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>S</code>       </td><td style="text-align: left;"> Entropy                               </td><td style="text-align: left;"> (cal | J) K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>V</code>       </td><td style="text-align: left;"> Volume                                </td><td style="text-align: left;"> cm<code class="reqn">^3</code> mol<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>Cp</code>      </td><td style="text-align: left;"> Heat capacity                         </td><td style="text-align: left;"> (cal | J) K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>E</code>       </td><td style="text-align: left;"> Exapansibility                        </td><td style="text-align: left;"> cm<code class="reqn">^3</code> K<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
         <code>kT</code>      </td><td style="text-align: left;"> Isothermal compressibility            </td><td style="text-align: left;"> cm<code class="reqn">^3</code> bar<code class="reqn">^{-1}</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td>
</tr>

</table>

<p>If <code>convert</code> is <code>TRUE</code> (the default), the input values of <code>T</code> and <code>P</code> are interpreted to have the units given by <code><a href="#topic+T.units">T.units</a></code> and <code><a href="#topic+P.units">P.units</a></code> (default: &deg;C and bar), and the output values of <code>G</code>, <code>H</code>, <code>S</code> and <code>Cp</code> are based on the units given in <code><a href="#topic+E.units">E.units</a></code> (default: Joules).
If <code>convert</code> is <code>FALSE</code>, the user units (<code><a href="#topic+T.units">T.units</a></code>, <code><a href="#topic+P.units">P.units</a></code>, and <code><a href="#topic+E.units">E.units</a></code>) are ignored, and <code>T</code> and <code>P</code> are taken to be in Kelvin and bar, and the returned values of <code>G</code>, <code>H</code>, <code>S</code> and <code>Cp</code> are in Joules.
</p>
<p>A chemical reaction is defined if <code>coeff</code> is given.
In this mode the standard molal properties of species are summed according to the stoichiometric <code>coeff</code>icients, where negative values denote reactants.
An unbalanced reaction is signalled if the amount of any element on the reactant and product sides differs by more than 1e-7; in this case, <code>subcrt</code> prints the missing composition needed to balance the reaction and produces a warning but computes a result anyway.
Alternatively, if <code>autobalance</code> is <code>TRUE</code>, the <code><a href="#topic+basis">basis</a></code> species of a system were previously defined, and all elements in the reaction are represented by the basis species, an unbalanced reaction given in the arguments to <code>subcrt</code> will be balanced automatically.
The auto balancing doesn't change the reaction coefficients of any species in the reaction that are not among the basis species.
</p>
<p>If <code>logact</code> is provided, the chemical affinities of reactions are calculated.
<code>logact</code> indicates the logarithms of activities (fugacities for gases) of species in the reaction; if there are fewer values of <code>logact</code> than number of species those values are repeated as necessary.
If the reaction was unbalanced to start, the logarithms of activities of any basis species added to the reaction are taken from the current definition of the <code><a href="#topic+basis">basis</a></code> species.
Columns appended to the output are <code>logQ</code> for the log10 of the activity product of the reaction, and <code>A</code> for the chemical affinity, in the units set by <code><a href="#topic+E.units">E.units</a></code>.
Note that <code><a href="#topic+affinity">affinity</a></code> provides related functionality but is geared toward the properties of formation reactions of species from the basis species and can be performed in more dimensions.
Calculations of chemical affinity in <code>subcrt</code> can be performed for any reaction of interest; however, they are currently limited to constant values of the logarithms of activities of species in the reactions, and hence of <code>logQ</code>, across the computational range.
</p>
<p>If <code>IS</code> is set to a single value other than zero, <code><a href="#topic+nonideal">nonideal</a></code> is used to calculate the adjusted properties (<code>G</code>, <code>H</code>, <code>S</code> and <code>Cp</code>) of charged aqueous species at the given ionic strength.
To perform calculations at a single <code>P</code> and <code>T</code> and for multiple values of ionic strength, supply these values in <code>IS</code>.
Calculations can also be performed on a <code>P</code>-<code>IS</code>, <code>T</code>-<code>IS</code> or <code>P,T</code>-<code>IS</code> grid.
Values of <code>logK</code> of reactions calculated for <code>IS</code> not equal to zero are consistent with the adjusted Gibbs energies of the charged aqueous species.
</p>
<p>If <code>thermo()$opt$varP</code> is <code>TRUE</code>, standard Gibbs energies of gases will be converted from a standard state at 1 bar (as used in SUPCRT) to a variable pressure standard state (see chapter 12 in Anderson and Crerar, 1993).
This is useful for constructing e.g. boiling curves for organic compounds.
</p>


<h3>Value</h3>

<p>A list of length two or three.
If the properties of a reaction were calculated, the first element of the list (named &lsquo;<span class="samp">&#8288;reaction&#8288;</span>&rsquo;) contains a dataframe with the reaction parameters; the second element, named &lsquo;<span class="samp">&#8288;out&#8288;</span>&rsquo;, is a dataframe containing the calculated properties.
Otherwise, the properties of species (not reactions) are returned: the first element, named &lsquo;<span class="samp">&#8288;species&#8288;</span>&rsquo;, contains a dataframe with the species identification; the second element, named &lsquo;<span class="samp">&#8288;out&#8288;</span>&rsquo;, is itself a list, each element of which is a dataframe of properties for a given species.
If minerals with polymorphic transitions are present, a third element (a dataframe) in the list indicates for all such minerals the stable phase at each grid point.
</p>


<h3>Warning</h3>

<p>Although <abbr><span class="acronym">SUPCRT92</span></abbr> prohibits calculations above 350 &deg;C at <i>P</i><sub>SAT</sub> (&ldquo;beyond range of applicability of aqueous species equations&rdquo;), CHNOSZ does not impose this limitation, and allows calculations up to the critical temperature (373.917 &deg;C) at <i>P</i><sub>SAT</sub>.
Interpret calculations between 350 &deg;C and the critical temperature at <i>P</i><sub>SAT</sub> at your own risk.
The discontinuity in the value of log&thinsp;<i>K</i> at <i>P</i><sub>SAT</sub> that is apparent in <code>demo("NaCl")</code> demonstrates one unexpected result.
</p>
<p><code>NA</code>s are produced for calculations at &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; when the temperature exceeds the critical temperature of H<sub>2</sub>O.
In addition, properties of species using the revised HKF equations are set to <code>NA</code> wherever the density of H<sub>2</sub>O &lt; 0.35 g/cm<sup>3</sup> (threshold just above the critical isochore; Johnson et al., 1992).
Both of these situations produce warnings, which are stored in the &lsquo;<span class="samp">&#8288;warnings&#8288;</span>&rsquo; element of the return value.
</p>
<p><code>NA</code>s are also output if the T, P conditions are otherwise beyond the capabilities of the water equations of state derived from SUPCRT92 (H2O92D.f), but the messages about this are produced by <code><a href="#topic+water.SUPCRT92">water.SUPCRT92</a></code> rather than <code>subcrt</code>.
</p>


<h3>Limitations</h3>

<p>Note that <code>E</code> and <code>kT</code> can only be calculated for aqueous species and only if the option (<code><a href="#topic+thermo">thermo</a>()$opt$water</code>) for calculations of properties using <code>water</code> is set to <code>IAPWS</code>.
On the other hand, if the <code>water</code> option is &lsquo;<span class="samp">&#8288;SUPCRT&#8288;</span>&rsquo; (the default), <code>E</code> and <code>kT</code> can be calculated for water but not for aqueous species.
(This is not an inherent limitation in either formulation, but it is just a matter of implementation.)
</p>


<h3>Note on polymorphic transitions</h3>

<p>Minerals with polymorphic transitions (denoted in OBIGT by having states &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; (lowest-<I>T</I> phase), &lsquo;<span class="samp">&#8288;cr2&#8288;</span>&rsquo;, etc.) can be specified by name with &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo; for the <code>state</code> or by using a numeric species index for the lowest-<I>T</I> polymorph.
If <code>use.polymorphs</code> is TRUE, <code>subcrt</code> uses the transition temperatures calculated from those at P = 1 bar given in OBIGT together with functions of the entropies and volumes of transitions (see <code><a href="#topic+dPdTtr">dPdTtr</a></code>) to determine the stable polymorph at each grid point and uses the properties of that polymorph in the output.
A <code>polymorph</code> column is added to the output to indicate the stable polymorph at each <I>T</I>-<I>P</I> condition.
If <code>exceed.Ttr</code> is <code>FALSE</code> (the default), output values of Gibbs energy are assigned NA beyond the transition temperature of the highest-temperature polymorph.
Set <code>exceed.Ttr</code> to <code>TRUE</code> to identify the stable polymorphs by comparing their extrapolated Gibbs energies instead of the tabulated transition temperatures.
This is generally not advised, as extrapolated Gibbs energies may not reliably determine the stable polymorph at extreme temperatures.
</p>


<h3>References</h3>

<p>Anderson, G. M. and Crerar, D. A. (1993) <em>Thermodynamics in Geochemistry: The Equilibrium Model</em>, Oxford University Press. <a href="https://www.worldcat.org/oclc/803272549">https://www.worldcat.org/oclc/803272549</a>
</p>
<p>Johnson, J. W., Oelkers, E. H. and Helgeson, H. C. (1992) SUPCRT92: A software package for calculating the standard molal thermodynamic properties of minerals, gases, aqueous species, and reactions from 1 to 5000 bar and 0 to 1000&deg;C. <em>Comp. Geosci.</em> <b>18</b>, 899&ndash;947. <a href="https://doi.org/10.1016/0098-3004%2892%2990029-Q">doi:10.1016/0098-3004(92)90029-Q</a>
</p>
<p>Helgeson, H. C., Owens, C. E., Knox, A. M. and Richard, L. (1998) Calculation of the standard molal thermodynamic properties of crystalline, liquid, and gas organic molecules at high temperatures and pressures. <em>Geochim. Cosmochim. Acta</em> <b>62</b>, 985&ndash;1081. <a href="https://doi.org/10.1016/S0016-7037%2897%2900219-6">doi:10.1016/S0016-7037(97)00219-6</a>
</p>
<p>LaRowe, D. E. and Helgeson, H. C. (2007) Quantifying the energetics of metabolic reactions in diverse biogeochemical systems: electron flow and ATP synthesis. <em>Geobiology</em> <b>5</b>, 153&ndash;168. <a href="https://doi.org/10.1111/j.1472-4669.2007.00099.x">doi:10.1111/j.1472-4669.2007.00099.x</a>
</p>
<p>Shock, E. L., Oelkers, E. H., Johnson, J. W., Sverjensky, D. A. and Helgeson, H. C. (1992) Calculation of the thermodynamic properties of aqueous species at high pressures and temperatures: Effective electrostatic radii, dissociation constants and standard partial molal properties to 1000 &deg;C and 5 kbar. <em>J. Chem. Soc. Faraday Trans.</em> <b>88</b>, 803&ndash;826. <a href="https://doi.org/10.1039/FT9928800803">doi:10.1039/FT9928800803</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+info">info</a></code> can be used to find species in the thermodynamic database.
<code><a href="#topic+makeup">makeup</a></code> is used by <code>subcrt</code> for parsing formulas to check mass balance of reactions.
<code>demo("ORP")</code> and <code><a href="#topic+nonideal">nonideal</a></code> for examples using the <code>IS</code> argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Properties of species
subcrt("water")
# Change temperature
subcrt("water", T = seq(0, 100, 20))
# Change temperature and pressure 	
T &lt;- seq(500, 1000, 100)
P &lt;- seq(5000, 10000, 1000)
subcrt("water", T = T, P = P)
# Temperature-pressure grid
subcrt("water", T = c(500, 1000), P = c(5000, 10000), grid = "P")

## Properties of reactions
subcrt(c("glucose", "ethanol", "CO2"), c(-1, 2, 2), T = 25)
# Use CO2(gas) (or just change "CO2" to "carbon dioxide")
subcrt(c("glucose", "ethanol", "CO2"), c(-1, 2, 2), c("aq", "aq", "gas"), T = 25)

## Automatically balance reactions
# First define the basis species
basis(c("CO2", "H2O", "NH3", "H2S", "O2"))
# Auto-balance adds the required amount of H2O and O2
subcrt(c("ethanol", "glucose"), c(-3, 1), T = 37)
# An example with H+
basis(c("H2O", "H2S", "O2", "H+"))
subcrt(c("HS-", "SO4-2"), c(-1, 1), T = 100)

## Mineral polymorphs
# Properties of the stable polymorph at each temperature
subcrt("pyrrhotite")
# Reactions automatically use stable polymorph
subcrt(c("pyrite", "pyrrhotite", "H2O", "H2S", "O2"), c(-1, 1, -1, 1, 0.5))
# Extrapolated properties of the lowest-T polymorph (metastable at higher temperatures)
subcrt(c("pyrrhotite"), use.polymorphs = FALSE, exceed.Ttr = TRUE)

## Messages about problems with the calculation
# Above the T, P limits for the H2O equations of state
subcrt("alanine", T = c(2250, 2251), P = c(30000, 30001), grid = "T")
# Psat is not defined above the critical point
# (suppressWarnings is used so that checks of examples don't raise warnings)
suppressWarnings(subcrt("alanine", T = seq(0, 5000, by = 1000)))

## Minerals with polymorphic transitions
# Compare calculated values of heat capacity of iron with
# values from Robie and Hemingway, 1995
T.units("K")
# We set pressure here otherwise subcrt uses Psat (saturation 
# vapor pressure of H2O above 100 degrees C) which can't be 
# calculated above the critical point of H2O (~647 K)
s &lt;- subcrt("Fe", T=seq(300, 1800, 20), P=1)
plot(s$out[[1]]$T, s$out[[1]]$Cp, type="l",
  xlab=axis.label("T"), ylab=axis.label("Cp"))
# Add points from RH95
RH95 &lt;- read.csv(system.file("extdata/cpetc/RH95.csv", package="CHNOSZ"))
points(RH95[,1], RH95[,2])
title(main=paste("Heat capacity of Fe(cr)\n",
  "(points - Robie and Hemingway, 1995)"))
# Reset the units to default values
T.units("C")

## Subzero (degrees C) calculations
# Uncomment the following to try IAPWS95 instead of SUPCRT92
#water("IAPWS95")
# The limit for H2O92D.f (from SUPCRT92) is currently -20 deg C
# but we go to -30 knowing properties will become NA
sb &lt;- subcrt(c("H2O", "Na+"), T = seq(-30, 10), P = 1)$out
# Start plot with extra room on right
opar &lt;- par(mar=c(5, 4, 4, 4))
# Plot Delta G
plot(sb$water$T, sb$water$G, ylim = c(-264000, -234000),
  xlab = axis.label("T"), ylab = axis.label("DG0"))
points(sb$`Na+`$T, sb$`Na+`$G, pch = 2)
# Add Cp
# change y-axis
par("usr" = c(par("usr")[1:2], -400, 100))
axis(4)
mtext(axis.label("Cp0"), side = 4, line = 3)
points(sb$water$T, sb$water$Cp, pch = 16)
points(sb$`Na+`$T, sb$`Na+`$Cp, pch = 17)
legend("topleft", c("H2O Cp", "H2O G", "Na+ Cp", "Na+ G"), pch = c(16, 1, 17, 2))
H2O &lt;- expr.species("H2O")
Na &lt;- expr.species("Na+")
degC &lt;- expr.units("T")
title(main = substitute(H2O~and~Na~to~-20~degC, list(H2O = H2O, Na = Na, degC = degC)))
par(opar)

## Calculations using a variable-pressure standard state
thermo("opt$varP" = TRUE)
# Calculate the boiling point of octane at 2 and 20 bar
# We need exceed.Ttr = TRUE because the liquid is metastable
# at high temperatures (also, the gas is metastable at low
# temperatures, but that doesn't produce NA in the output)
sout2 &lt;- subcrt(rep("octane", 2), c("liq", "gas"),
  c(-1, 1), T = seq(-50, 300, 0.1), P = 2, exceed.Ttr = TRUE)$out
sout20 &lt;- subcrt(rep("octane", 2), c("liq", "gas"),
  c(-1, 1), T = seq(-50, 300, 0.1), P = 20, exceed.Ttr = TRUE)$out
# Find T with the Gibbs energy of reaction that is closest to zero
Tvap2 &lt;- sout2$T[which.min(abs(sout2$G))]
Tvap20 &lt;- sout20$T[which.min(abs(sout20$G))]
# Compare these with experimental values (Fig. 1 of Helgeson et al., 1998)
Tvap2.exp &lt;- 156
Tvap20.exp &lt;- 276
# Reset varP to FALSE (the default)
thermo("opt$varP" = FALSE)
</code></pre>

<hr>
<h2 id='swap.basis'>Swap Basis Species</h2><span id='topic+basis.elements'></span><span id='topic+element.mu'></span><span id='topic+basis.logact'></span><span id='topic+swap.basis'></span><span id='topic+ibasis'></span>

<h3>Description</h3>

<p>Swap the basis species defining a chemical system. One basis species is replaced by a new one with a different chemical formula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  swap.basis(species, species2, T = 25)
  basis.elements(basis = thermo()$basis)
  element.mu(basis = thermo()$basis, T = 25)
  basis.logact(emu, basis = thermo()$basis, T = 25)
  ibasis(species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="swap.basis_+3A_basis">basis</code></td>
<td>
<p>dataframe, a basis definition</p>
</td></tr>
<tr><td><code id="swap.basis_+3A_t">T</code></td>
<td>
<p>numeric, temperature in Kelvin</p>
</td></tr>
<tr><td><code id="swap.basis_+3A_emu">emu</code></td>
<td>
<p>numeric, chemical potentials of elements</p>
</td></tr>
<tr><td><code id="swap.basis_+3A_species">species</code></td>
<td>
<p>character, names or formulas of species, or numeric, indices of species</p>
</td></tr>
<tr><td><code id="swap.basis_+3A_species2">species2</code></td>
<td>
<p>character or numeric, a species to swap in to the basis definition</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>swap.basis</code> allows to change the basis definition by swapping out a basis species for a new one.
Specify the names or formulas of the old and replacement basis species in the first argument.
When the basis definition is changed, any species of interest that were present are deleted, unless the new basis definition has exactly the same elements as before.
In that case, the species are kept; also, the activities of the new basis species are set in order to maintain the chemical potentials of the elements at <code>T</code> &deg;C and 1 bar.
</p>
<p>The other functions are supporting functions: <code>basis.elements</code> returns the stoichiometric matrix for the current basis definition, <code>element.mu</code> calculates the chemical potentials of elements corresponding to the activities of the basis species, <code>basis.logact</code> does the inverse operation, and <code>ibasis</code> returns the index in the basis set for a given species index (in <code><a href="#topic+thermo">thermo</a>$OBIGT</code>), name or formula.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+basis">basis</a></code>, and <code><a href="#topic+mosaic">mosaic</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Swapping basis species
# Start with a preset basis definition
b1 &lt;- basis("CHNOS+")
# Swap H2(aq) for O2(gas)
b2 &lt;- swap.basis("O2", "H2")
# Put oxygen back in
b3 &lt;- swap.basis("H2", "oxygen")

# Interconversion of chemical potentials of elements and
# logarithms of activities of basis species at high temperature
basis("CHNOS+")
bl1 &lt;- basis()$logact
emu &lt;- element.mu(T = 100)
bl2 &lt;- basis.logact(emu, T = 100)
# There's no difference
round(bl2 - bl1, 10)
</code></pre>

<hr>
<h2 id='taxonomy'>Extract Data from NCBI Taxonomy Files</h2><span id='topic+taxonomy'></span><span id='topic+getnodes'></span><span id='topic+getrank'></span><span id='topic+parent'></span><span id='topic+allparents'></span><span id='topic+getnames'></span><span id='topic+sciname'></span>

<h3>Description</h3>

<p>Read data from NCBI taxonomy files, traverse taxonomic ranks, get scientific names of taxonomic nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  getnodes(taxdir)
  getrank(id, taxdir, nodes=NULL)
  parent(id, taxdir, rank=NULL, nodes=NULL)
  allparents(id, taxdir, nodes=NULL)
  getnames(taxdir)
  sciname(id, taxdir, names=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxonomy_+3A_taxdir">taxdir</code></td>
<td>
<p>character, directory where the taxonomy files are kept.</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_id">id</code></td>
<td>
<p>numeric, taxonomic ID(s) of the nodes of interest.</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_nodes">nodes</code></td>
<td>
<p>dataframe, output from <code>getnodes</code> (optional).</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_rank">rank</code></td>
<td>
<p>character, name of the taxonomic rank of interest.</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_names">names</code></td>
<td>
<p>dataframe, output from <code>getnames</code> (optional).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions provide a convenient way to read data from NCBI taxonomy files (i.e., the contents of <code>taxdump.tar.gz</code>, which is available from <a href="https://ftp.ncbi.nih.gov/pub/taxonomy/">https://ftp.ncbi.nih.gov/pub/taxonomy/</a>). 
</p>
<p>The <code>taxdir</code> argument is used to specify the directory where the <code>nodes.dmp</code> and <code>names.dmp</code> files are located. <code>getnodes</code> and <code>getnames</code> read these files into data frames. <code>getrank</code> returns the rank (species, genus, etc) of the node with the given taxonomic <code>id</code>. <code>parent</code> returns the taxonomic ID of the next-lowest node below that specified by the <code>id</code> in the argument, unless <code>rank</code> is supplied, in which case the function descends the tree until a node with that rank is found. <code>allparents</code> returns all the taxonomic IDs of all nodes between that specified by <code>id</code> and the root of the tree, inclusive. <code>sciname</code> returns the scientific name of the node with the given <code>id</code>.
</p>
<p>The <code>id</code> argument can be of length greater than 1 except for <code>allparents</code>. If <code>getrank</code>, <code>parent</code>, <code>allparents</code> or <code>sciname</code> need to be called repeatedly, the operation can be hastened by supplying the output of <code>getnodes</code> in the <code>nodes</code> argument and/or the output of <code>getnames</code> in the <code>names</code> argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Get information about Homo sapiens from the
## packaged taxonomy files
taxdir &lt;- system.file("extdata/taxonomy", package = "CHNOSZ")
# H. sapiens' taxonomic id
id1 &lt;- 9606
# That is a species
getrank(id1, taxdir)
# The next step up the taxonomy
id2 &lt;- parent(id1, taxdir)
print(id2)
# That is a genus
getrank(id2, taxdir)
# That genus is "Homo"
sciname(id2, taxdir)
# We can ask what phylum is it part of?
id3 &lt;- parent(id1, taxdir, "phylum")
# Answer: "Chordata"
sciname(id3, taxdir)
# H. sapiens' complete taxonomy
id4 &lt;- allparents(id1, taxdir)
sciname(id4, taxdir)

## The names of the organisms in the supplied taxonomy files
taxdir &lt;- system.file("extdata/taxonomy", package = "CHNOSZ")
id5 &lt;- c(83333, 4932, 9606, 186497, 243232)
sciname(id5, taxdir)
# These are not all species, though
# (those with "no rank" are something like strains, 
# e.g. Escherichia coli K-12)
getrank(id5, taxdir)
# Find the species for each of these
id6 &lt;- sapply(id5, function(x) parent(x, taxdir = taxdir, rank = "species"))
unique(getrank(id6, taxdir))  # "species"
# Note that the K-12 is dropped
sciname(id6, taxdir)

## The complete nodes.dmp and names.dmp files are quite large,
## so it helps to store them in memory when performing multiple queries
## (this doesn't have a noticeable speed-up for the small files in this example)
taxdir &lt;- system.file("extdata/taxonomy", package = "CHNOSZ")
nodes &lt;- getnodes(taxdir = taxdir)
# All of the node ids in this file
id7 &lt;- nodes$id
# All of the non-leaf nodes
id8 &lt;- unique(parent(id7, nodes = nodes))
names &lt;- getnames(taxdir = taxdir)
sciname(id8, names = names)
</code></pre>

<hr>
<h2 id='thermo'>Thermodynamic Database and System Settings</h2><span id='topic+reset'></span><span id='topic+OBIGT'></span><span id='topic+thermo'></span><span id='topic+CHNOSZ'></span>

<h3>Description</h3>

<p>Run <code>reset()</code> to reset all of the data used in CHNOSZ to default values.
This includes the computational settings, thermodynamic database, and system settings (chemical species).
</p>
<p>The system settings are changed using <code><a href="#topic+basis">basis</a></code> and <code><a href="#topic+species">species</a></code>.
To clear the system settings (the default, i.e. no species loaded), run <code>basis("")</code>; to clear only the formed species, run <code>species(delete = TRUE)</code>
</p>
<p>The thermodynamic database is changed using <code><a href="#topic+add.OBIGT">add.OBIGT</a></code> and <code><a href="#topic+mod.OBIGT">mod.OBIGT</a></code>.
To restore the default database without altering the species settings, run <code>OBIGT()</code>.
</p>
<p>The computational settings are changed using <code><a href="#topic+water">water</a></code>, <code><a href="#topic+P.units">P.units</a></code>, <code><a href="#topic+T.units">T.units</a></code>, <code><a href="#topic+E.units">E.units</a></code>, and some other commands (e.g. <code><a href="#topic+mod.buffer">mod.buffer</a></code>).
</p>
<p>All the data are stored in the <code>thermo</code> data object in an environment named <code>CHNOSZ</code>.
<code>thermo()</code> is a convenience function to access or modify parts of this object, in particular some computational settings, for example, <code>thermo("opt$ideal.H" = FALSE)</code> (see <code><a href="#topic+nonideal">nonideal</a></code>).
</p>
<p>The source data are provided with CHNOSZ as <code>*.csv</code> files in the <code>extdata/thermo</code> and <code>extdata/OBIGT</code> directories of the package.
These files are used to build the <code>thermo</code> object, which is described below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  reset()
  OBIGT(no.organics = FALSE)
  thermo(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="thermo_+3A_no.organics">no.organics</code></td>
<td>
<p>logical, load the database without data files for organic species (NOTE: CH<sub>4</sub> is listed as an &ldquo;inorganic&rdquo; species)?</p>
</td></tr>
<tr><td><code id="thermo_+3A_...">...</code></td>
<td>
<p>list, one or more arguments whose names correspond to the setting to modify</p>
</td></tr>
</table>


<h3>Format</h3>


<ul>
<li> <p><code>thermo()$opt</code> 
List of computational settings.
Square brackets indicate default values.
Note that the units of <code>G.tol</code> and <code>Cp.tol</code> depend on the <code>E_units</code> for each species in <code>thermo()$OBIGT</code>.
Therefore, species with <code>E_units</code> of &lsquo;<span class="samp">&#8288;J&#8288;</span>&rsquo; have a lower absolute tolerance for producing messages (because 4.184 J = 1 cal).
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>E.units</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> The user's units of energy ([&lsquo;<span class="samp">&#8288;J&#8288;</span>&rsquo;] or &lsquo;<span class="samp">&#8288;cal&#8288;</span>&rsquo;) (see <code><a href="#topic+subcrt">subcrt</a></code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>T.units</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> The user's units of temperature ([&lsquo;<span class="samp">&#8288;C&#8288;</span>&rsquo;] or &lsquo;<span class="samp">&#8288;K&#8288;</span>&rsquo;)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>P.units</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> The user's units of pressure ([&lsquo;<span class="samp">&#8288;bar&#8288;</span>&rsquo;] or &lsquo;<span class="samp">&#8288;MPa&#8288;</span>&rsquo;)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>state</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> The default physical state for searching species [&lsquo;<span class="samp">&#8288;aq&#8288;</span>&rsquo;] (see <code><a href="#topic+info">info</a></code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>water</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Computational option for properties of water ([&lsquo;<span class="samp">&#8288;SUPCRT&#8288;</span>&rsquo;] or &lsquo;<span class="samp">&#8288;IAPWS&#8288;</span>&rsquo;; see <code><a href="#topic+water">water</a></code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>G.tol</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Difference in G above which <code><a href="#topic+check.GHS">check.GHS</a></code> produces a message (cal mol<sup>-1</sup>) [100]</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Cp.tol</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Difference in Cp above which <code><a href="#topic+check.EOS">check.EOS</a></code> produces a message (cal K<sup>-1</sup> mol<sup>-1</sup>) [1]</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>V.tol</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Difference in V above which <code><a href="#topic+check.EOS">check.EOS</a></code> produces a message (cm<sup>3</sup> mol<sup>-1</sup>) [1]</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>varP</code> </td><td style="text-align: left;"> logical </td><td style="text-align: left;"> Use variable-pressure standard state for gases? [<code>FALSE</code>] (see <code><a href="#topic+subcrt">subcrt</a></code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>IAPWS.sat</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> State of water for saturation properties [&lsquo;<span class="samp">&#8288;liquid&#8288;</span>&rsquo;] (see <code><a href="#topic+util.water">util.water</a></code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>paramin</code> </td><td style="text-align: left;"> integer </td><td style="text-align: left;"> Minimum number of calculations to launch parallel processes [1000] (see <code><a href="#topic+palply">palply</a></code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ideal.H</code> </td><td style="text-align: left;"> logical </td><td style="text-align: left;"> Should <code><a href="#topic+nonideal">nonideal</a></code> ignore the proton? [<code>TRUE</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ideal.e</code> </td><td style="text-align: left;"> logical </td><td style="text-align: left;"> Should <code><a href="#topic+nonideal">nonideal</a></code> ignore the electron? [<code>TRUE</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>nonideal</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Option for charged species in <code><a href="#topic+nonideal">nonideal</a></code> [<code>Bdot</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Setchenow</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Option for neutral species in <code><a href="#topic+nonideal">nonideal</a></code> [<code>bgamma0</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Berman</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> User data file for mineral parameters in the Berman equations [<code>NA</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>maxcores</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Maximum number of cores for parallel calculations with <code><a href="#topic+palply">palply</a></code> [<code>2</code>] </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ionize.aa</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Calculate properties of ionized proteins when H<sup>+</sup> is in basis species (see <code><a href="#topic+affinity">affinity</a></code>) [<code>TRUE</code>]
</td>
</tr>

</table>

</li>
<li> <p><code>thermo()$element</code>
Dataframe containing the thermodynamic properties of elements taken from Cox et al., 1989, Wagman et al., 1982, and (for Am, Pu, Np, Cm) Thoenen et al., 2014.
The standard molal entropy (<code class="reqn">S</code>(<code>Z</code>)) at 25 &deg;C and 1 bar for the &ldquo;element&rdquo; of charge (<code>Z</code>) was calculated from <code class="reqn">S</code>(H2,g) + 2<code class="reqn">S</code>(<code>Z</code>) =  2<code class="reqn">S</code>(H+), where the standard molal entropies of H2,g and H+ were taken from Cox et al., 1989.
The mass of <code>Z</code> is taken to be zero.
Accessing this data frame using <code><a href="#topic+mass">mass</a></code> or <code><a href="#topic+entropy">entropy</a></code> will select the first entry found for a given element; i.e., values from Wagman et al., 1982 will only be retrieved if the properties of the element are not found from Cox et al., 1989.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>element</code>  </td><td style="text-align: left;"> character  </td><td style="text-align: left;"> Symbol of element</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>state</code>  </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Stable state of element at 25 &deg;C and 1 bar</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>source</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Source of data</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>mass</code>  </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Mass of element (in natural isotopic distribution;</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td><td style="text-align: left;"> </td><td style="text-align: left;"> referenced to a mass of 12 for <sup>12</sup>C)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>s</code>   </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Entropy of the compound of the element in its stable</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td><td style="text-align: left;"> </td><td style="text-align: left;"> state at 25 &deg;C and 1 bar (cal K<sup>-1</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>n</code>  </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of atoms of the element in its stable</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td><td style="text-align: left;"> </td><td style="text-align: left;"> compound at 25 &deg;C and 1 bar
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$OBIGT</code>
</p>
<p>This dataframe is a thermodynamic database of standard molal thermodynamic properties and equations of state parameters of species.
&ldquo;OrganoBioGeoTherm&rdquo; is the name of a Windows GUI interface to SUPCRT92 that was produced in Harold C. Helgeson's Laboratory of Theoretical Geochemistry and Biogeochemistry at the University of California, Berkeley.
The <abbr><span class="acronym">OBIGT</span></abbr> database was originally distributed with that program, and was the starting point for the database in CHNOSZ.
</p>
<p>Note the following database conventions:
</p>

<ul>
<li><p> The combination of <code>name</code> and <code>state</code> defines a species in <code>thermo()$OBIGT</code>. A species cannot be duplicated (this is checked when running <code>reset()</code>).
</p>
</li>
<li><p> English names of inorganic gases are used only for the gas state. The dissolved species is named with the chemical formula. Therefore, <code>info("oxygen")</code> refers to the gas, and <code>info("O2")</code> refers to the aqueous species.
</p>
</li></ul>
 
<p>Each entry is referenced to one or two literature sources listed in <code>thermo()$refs</code>.
Use <code><a href="#topic+thermo.refs">thermo.refs</a></code> to look up the citation information for the references.
See the vignette <a href="../doc/OBIGT.html"><strong>OBIGT</strong></a> for a complete description of the sources of data.
</p>
<p>The identifying characteristics of species and their standard molal thermodynamic properties at 25 &deg;C and 1 bar are located in the first 13 columns of <code>thermo()$OBIGT</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>name</code>     </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Species name</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>abbrv</code>    </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Species abbreviation</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>formula</code>  </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Species formula</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>state</code>    </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Physical state</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ref1</code>  </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Primary source</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ref2</code>  </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Secondary source</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>date</code>     </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Date of data entry (<a href="https://en.wikipedia.org/wiki/ISO_8601">ISO 8601</a> extended format)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>model</code>    </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Model for thermodynamic properties of the species</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>E_units</code>     </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Units of energy: &lsquo;<span class="samp">&#8288;J&#8288;</span>&rsquo; for Joules or &lsquo;<span class="samp">&#8288;cal&#8288;</span>&rsquo; for calories</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>G</code>        </td><td style="text-align: left;"> numeric   </td><td style="text-align: left;"> Standard molal Gibbs energy of formation</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td><td style="text-align: left;"> </td><td style="text-align: left;"> from the elements (J|cal mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>H</code>        </td><td style="text-align: left;"> numeric   </td><td style="text-align: left;"> Standard molal enthalpy of formation</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td><td style="text-align: left;"> </td><td style="text-align: left;"> from the elements (J|cal mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>S</code>        </td><td style="text-align: left;"> numeric   </td><td style="text-align: left;"> Standard molal entropy (J|cal mol<sup>-1</sup> K<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Cp</code>       </td><td style="text-align: left;"> numeric   </td><td style="text-align: left;"> Standard molal isobaric heat capacity (J|cal mol<sup>-1</sup> K<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>V</code>	      </td><td style="text-align: left;"> numeric   </td><td style="text-align: left;"> Standard molal volume (cm<sup>3</sup> mol<sup>-1</sup>)
    </td>
</tr>

</table>

<p><code>model</code> must be one of &lsquo;<span class="samp">&#8288;H2O&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;HKF&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;CGL&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Berman&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;AD&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo;.
&lsquo;<span class="samp">&#8288;H2O&#8288;</span>&rsquo; is reserved for liquid water, the properties of which are calculated using one of several available models (see <code><a href="#topic+water">water</a></code>).
Most aqueous species use &lsquo;<span class="samp">&#8288;HKF&#8288;</span>&rsquo; (the revised Helgeson-Kirkham-Flowers model).
Properties of aqueous species with <code>model</code> set to &lsquo;<span class="samp">&#8288;AD&#8288;</span>&rsquo; are calculated using the Akinfiev-Diamond model, and those with &lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo; are calculated using the <code><a href="#topic+DEW">DEW</a></code> model.
Many minerals in the default database use the &lsquo;<span class="samp">&#8288;Berman&#8288;</span>&rsquo; model (see <code><a href="#topic+Berman">Berman</a></code>).
All other species use &lsquo;<span class="samp">&#8288;CGL&#8288;</span>&rsquo; (general crystalline, gas, liquid model).
Properties of these species are calculated using a heat capacity function with up to six terms; the exponent on the final term can be defined in the database (see below).
</p>
<p>The meanings of the remaining columns depend on the model for each species.
The names of these columns are compounded from those of the parameters in the HKF equations of state and general heat capacity equation; for example, column 13 is named <code>a1.a</code>.
Scaling of the values by integral powers of ten (i.e., orders of magnitude; OOM) for the HKF parameters (this also includes the DEW model) is based on the usual (but by no means universal) convention in the literature.
</p>
<p>Columns 15-22 for aqueous species (parameters in the revised HKF equations of state).
NOTE: Most older papers use units of calories for these parameters, so &lsquo;<span class="samp">&#8288;cal&#8288;</span>&rsquo; is listed here; the actual units for each species are set in the <code>E_units</code> column.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>a1</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>a</I><sub>1</sub> * 10 (cal mol<sup>-1</sup> bar<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>a2</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>a</I><sub>2</sub> * 10<sup>-2</sup> (cal mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>a3</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>a</I><sub>3</sub> (cal K mol<sup>-1</sup> bar<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>a4</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>a</I><sub>4</sub> * 10<sup>-4</sup> (cal mol<sup>-1</sup> K)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>c1</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>c</I><sub>1</sub> (cal mol<sup>-1</sup> K<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>c2</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <I>c</I><sub>2</sub> * 10<sup>-4</sup> (cal mol<sup>-1</sup> K)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>omega</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> &omega; * 10<sup>-5</sup> (cal mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Z</code>  </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Charge
    </td>
</tr>

</table>

<p>Columns 15-22 for crystalline, gas and liquid species (<I>C<sub>P</sub></I> = <code class="reqn">a</code> + <code class="reqn">bT</code> + <code class="reqn">cT</code><sup>-2</sup> + <code class="reqn">dT</code><sup>-0.5</sup> + <code class="reqn">eT</code><sup>2</sup> + <code class="reqn">fT</code><sup>lambda</sup>).
NOTE: As of CHNOSZ 2.0.0, OOM scaling for heat capacity coefficients has been removed, and new entries use units of Joules unless indicated by setting <code>E_units</code> to &lsquo;<span class="samp">&#8288;cal&#8288;</span>&rsquo;.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>a</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">a</code> (J K<sup>-1</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>b</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">b</code> (J K<sup>-2</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>c</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">c</code> (J K mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>d</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">d</code> (J K<sup>-0.5</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>e</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">e</code> (J K<sup>-3</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>f</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">f</code> (J K<sup>-lambda-1</sup> mol<sup>-1</sup>)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>lambda</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> &lambda; (exponent on the <code class="reqn">f</code> term)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>T</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Positive value: Temperature (K) of polymorphic transition or phase stability limit</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>T</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Negative value: Opposite of temperature (K) limit of <I>C<sub>P</sub></I> equation (see <a href="../doc/FAQ.html"><strong>FAQ</strong></a> for details)</td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

<p>Columns 15-17 for aqueous species using the Akinfiev-Diamond model. Note that the <code>c</code> column is used to store the <code class="reqn">\xi</code> parameter. Columns 18-22 are not used.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>a</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">a</code> (cm<sup>3</sup> g<sup>-1</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>b</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">b</code> (cm<sup>3</sup> K<sup>0.5</sup> g<sup>-1</sup>) </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>c</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">\xi</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>d</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">XX1</code> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>e</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">XX2</code> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>f</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">XX3</code> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>lambda</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">XX4</code> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Z</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> <code class="reqn">Z</code> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$refs</code>
References for thermodynamic data.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>key</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Source key</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>author</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Author(s)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>year</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Year</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>citation</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Citation (journal title, volume, and article number or pages; or book or report title)</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>note</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Short description of the compounds or species in this data source</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>URL</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> URL</td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$buffers</code>
</p>
<p>Dataframe which contains definitions of buffers of chemical activity. Each named buffer can be composed of one or more species, which may include any species in the thermodynamic database and/or any protein. The calculations provided by <code><a href="#topic+buffer">buffer</a></code> do not take into account polymorphic transitions of minerals, so individual polymorphs of such minerals must be specified in the buffers.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>name</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Name of buffer</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>species</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Name of species</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>state</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Physical state of species</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>logact</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Logarithm of activity (fugacity for gases)
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$protein</code>
Data frame of amino acid compositions of selected proteins. Most of the compositions were taken from the SWISS-PROT/UniProt online database (Boeckmann et al., 2003) and the protein and organism names usually follow the conventions adopted there. In some cases different isoforms of proteins are identified using modifications of the protein names; for example, &lsquo;<span class="samp">&#8288;MOD5.M&#8288;</span>&rsquo; and <code>MOD5.N</code> proteins of &lsquo;<span class="samp">&#8288;YEAST&#8288;</span>&rsquo; denote the mitochondrial and nuclear isoforms of this protein. See <code><a href="#topic+pinfo">pinfo</a></code> to search this data frame by protein name, and other functions to work with the amino acid compositions.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <code>protein</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of protein</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>organism</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Identification of organism</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>ref</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Reference key for source of compositional data</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>abbrv</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Abbreviation or other ID for protein</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>chains</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of polypeptide chains in the protein</td>
</tr>
<tr>
 <td style="text-align: left;">
      <code>Ala</code>...<code>Tyr</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Number of each amino acid in the protein
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$groups</code>
This is a dataframe with 22 columns for the amino acid sidechain, backbone and protein backbone groups ([Ala]..[Tyr],[AABB],[UPBB]) whose rows correspond to the elements C, H, N, O, S. It is used to quickly calculate the chemical formulas of proteins that are selected using the <code>iprotein</code> argument in <code><a href="#topic+affinity">affinity</a></code>.
</p>
</li>
<li> <p><code>thermo()$basis</code>
Initially <code>NULL</code>, reserved for a dataframe written by <code><a href="#topic+basis">basis</a></code> upon definition of the basis species. The number of rows of this dataframe is equal to the number of columns in &ldquo;...&rdquo; (one for each element).
</p>

<table>
<tr>
 <td style="text-align: left;">
        <code>...</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> One or more columns of stoichiometric</td>
</tr>
<tr>
 <td style="text-align: left;">
        </td><td style="text-align: left;"> </td><td style="text-align: left;"> coefficients of elements in the basis species</td>
</tr>
<tr>
 <td style="text-align: left;">
        <code>ispecies</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Rownumber of basis species in <code>thermo()$OBIGT</code></td>
</tr>
<tr>
 <td style="text-align: left;">
        <code>logact</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Logarithm of activity or fugacity of basis species</td>
</tr>
<tr>
 <td style="text-align: left;">
        <code>state</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Physical state of basis species</td>
</tr>
<tr>
 <td style="text-align: left;">
     </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$species</code>
Initially <code>NULL</code>, reserved for a dataframe generated by <code><a href="#topic+species">species</a></code> to define the species of interest. The number of columns in &ldquo;...&rdquo; is equal to the number of basis species (i.e., rows of <code>thermo()$basis</code>).
</p>

<table>
<tr>
 <td style="text-align: left;">
       <code>...</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> One or more columns of stoichiometric</td>
</tr>
<tr>
 <td style="text-align: left;">
       </td><td style="text-align: left;"> </td><td style="text-align: left;"> coefficients of basis species in the species of interest</td>
</tr>
<tr>
 <td style="text-align: left;">
       <code>ispecies</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Rownumber of species in <code>thermo()$OBIGT</code></td>
</tr>
<tr>
 <td style="text-align: left;">
       <code>logact</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Logarithm of activity or fugacity of species</td>
</tr>
<tr>
 <td style="text-align: left;">
       <code>state</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Physical state of species</td>
</tr>
<tr>
 <td style="text-align: left;">
       <code>name</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Name of species</td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$stoich</code>
A precalculated stoichiometric matrix for the default database. This is a matrix, not a data frame, and as such can accept duplicated row names, corresponding to chemical formulas of the species. See <code><a href="#topic+retrieve">retrieve</a></code>, and the first test in <code>inst/tinytest/test-retrieve.R</code> for how to update this.
</p>

<table>
<tr>
 <td style="text-align: left;">
       <code>rownames</code> </td><td style="text-align: left;"> character </td><td style="text-align: left;"> Chemical formulas from <code>thermo()$OBIGT</code></td>
</tr>
<tr>
 <td style="text-align: left;">
       <code>...</code> </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> Stoichiometry, one column for each element present in any species
    </td>
</tr>

</table>

</li>
<li> <p><code>thermo()$Bdot_acirc</code>
Values of ion size parameter (&Aring;) for species, taken from the UT_SIZES.REF file of the HCh package (Shvarov and Bastrakov, 1999), which is based on Table 2.7 of Garrels and Christ, 1965.
This is used in <code><a href="#topic+nonideal">nonideal</a></code> with the default &lsquo;<span class="samp">&#8288;Bdot&#8288;</span>&rsquo; method.
Custom ion size parameters can be added to this vector; to override a default value for a species, either replace the numeric value for that species or prepend a named numeric value (for duplicated species, the first value is used).
See <code>demo("yttrium")</code> for an example of adding and overriding species.
</p>
</li>
<li> <p><code>thermo()$Berman</code>
A data frame with thermodynamic parameters for minerals in the Berman equations, assembled from files in &lsquo;<span class="samp">&#8288;extdata/Berman&#8288;</span>&rsquo; and used in <code><a href="#topic+Berman">Berman</a></code>.
</p>
</li></ul>
  


<h3>Note on polymorphic transitions</h3>

<p>To enable the calculation of thermodynamic properties of polymorphic transitions, higher-temperature polymorphs of minerals are listed in OBIGT with states &lsquo;<span class="samp">&#8288;cr2&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;cr3&#8288;</span>&rsquo;, etc.
The standard thermodynamic properties of high-temperature polymorphs at 25 &deg;C and 1 bar are apparent values that are consistent with given values of enthalpy of transition (where available) at the transition temperature (<code class="reqn">T_{tr}</code>).
See the <a href="../doc/FAQ.html"><strong>FAQ</strong></a> question &ldquo;How can minerals with polymorphic transitions be added to the database?&rdquo; for details of the retrieval of standard thermodynamic properties of polymorphs used in OBIGT.
</p>


<h3>References</h3>

<p>Cox, J. D., Wagman, D. D. and Medvedev, V. A., eds. (1989) <em>CODATA Key Values for Thermodynamics</em>. Hemisphere Publishing Corporation, New York, 271 p. <a href="https://www.worldcat.org/oclc/18559968">https://www.worldcat.org/oclc/18559968</a>
</p>
<p>Garrels, R. M. and Christ, C. L. (1965) <em>Solutions, Minerals, and Equilibria</em>, Harper &amp; Row, New York, 450 p. <a href="https://www.worldcat.org/oclc/517586">https://www.worldcat.org/oclc/517586</a>
</p>
<p>Thoenen, T., Hummel, W., Berner, U. and Curti, E. (2014) <em>The PSI/Nagra Chemical Thermodynamic Database 12/07</em>. Paul Scherrer Institut. <a href="https://www.psi.ch/en/les/database">https://www.psi.ch/en/les/database</a>
</p>
<p>Wagman, D. D., Evans, W. H., Parker, V. B., Schumm, R. H., Halow, I., Bailey, S. M., Churney, K. L. and Nuttall, R. L. (1982) The NBS tables of chemical thermodynamic properties. Selected values for inorganic and C<sub>1</sub> and C<sub>2</sub> organic substances in SI units. <em>J. Phys. Chem. Ref. Data</em> <b>11</b> (supp. 2), 1&ndash;392. <a href="https://srd.nist.gov/JPCRD/jpcrdS2Vol11.pdf">https://srd.nist.gov/JPCRD/jpcrdS2Vol11.pdf</a>
</p>


<h3>See Also</h3>

<p>Other data files, including those supporting the examples and vignettes, are documented separately at <code><a href="#topic+extdata">extdata</a></code>. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Where are the data files in CHNOSZ?
system.file("extdata", package = "CHNOSZ")
# What files make up OBIGT?
# Note: file names with _aq, _cr, _gas, or _liq
# are used in the default database
dir(system.file("extdata/OBIGT", package = "CHNOSZ"))

## Exploring thermo()$OBIGT
# What physical states are present
unique(thermo()$OBIGT$state)
# Formulas of ten random species
n &lt;- nrow(thermo()$OBIGT)
thermo()$OBIGT$formula[runif(10)*n]

## Adding an element
old &lt;- thermo()$element
# Element symbol, state, source (can be anything),
# mass, entropy, and number in compound
Xprops &lt;- data.frame(element = "X", state = "cr",
  source = "user", mass = 100, s = 100, n = 1)
new &lt;- rbind(old, Xprops)
thermo(element = new)
# Now "X" is recognized as an element in other functions
mass("X10")
# Restore default settings to remove X
reset()
</code></pre>

<hr>
<h2 id='util.array'>Functions to Work with Multidimensional Arrays</h2><span id='topic+util.array'></span><span id='topic+list2array'></span><span id='topic+slice'></span><span id='topic+dimSums'></span><span id='topic+slice.affinity'></span>

<h3>Description</h3>

<p>These functions can be used to turn a list into an array and extract or replace values or take the sum along a certain dimension of an array.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  list2array(l)
  slice(arr, d = NULL, i = 1, value = NULL)
  dimSums(arr, d = 1, i = NULL)
  slice.affinity(affinity, d = 1, i = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.array_+3A_l">l</code></td>
<td>
<p>a list.</p>
</td></tr>
<tr><td><code id="util.array_+3A_arr">arr</code></td>
<td>
<p>an array.</p>
</td></tr>
<tr><td><code id="util.array_+3A_d">d</code></td>
<td>
<p>numeric, what dimension to use.</p>
</td></tr>
<tr><td><code id="util.array_+3A_i">i</code></td>
<td>
<p>numeric, what slice to use.</p>
</td></tr>
<tr><td><code id="util.array_+3A_value">value</code></td>
<td>
<p>values to assign to the portion of an array specified by <code>d</code> and <code>i</code>.</p>
</td></tr>
<tr><td><code id="util.array_+3A_affinity">affinity</code></td>
<td>
<p>list, output from <code><a href="#topic+affinity">affinity</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>list2array</code> turns a list of <code><a href="base.html#topic+array">array</a></code>s, each with the same dimensions, into a new array having one more dimension whose size is equal to the number of initial arrays.
</p>
<p><code>slice</code> extracts or assigns values from/to the <code>i</code>th slice(s) in the <code>d</code>th dimension of an array. Values are assigned to an array if <code>value</code> is not NULL. This function works by building an expression containing the extraction operator (<code><a href="base.html#topic++5B">[</a></code>).
</p>
<p><code>slice.affinity</code> performs a slice operation on the &lsquo;<span class="samp">&#8288;values&#8288;</span>&rsquo; element of the &lsquo;<span class="samp">&#8288;affinity&#8288;</span>&rsquo; variable (which should be the output of <code><a href="#topic+affinity">affinity</a></code>).
</p>
<p><code>dimSums</code> sums an array along the <code>d</code>th dimension using only the <code>i</code>th slices in that dimension. If <code>i</code> is NULL, all slices in that dimension are summed together. For matrices, <code>dimSums(x,1)</code> has the same result as <code><a href="base.html#topic+colSums">colSums</a>(x)</code> and <code>dimSums(x,2)</code> has the same result as <code><a href="base.html#topic+rowSums">rowSums</a>(x)</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Start with a matrix
mat &lt;- matrix(1:12, ncol = 3)
# Pay attention to the following when
# writing examples that test for identity!
identical(1 * mat, mat)   # FALSE
# Create two matrices that are multiples of the first
a &lt;- 1 * mat
b &lt;- 2 * mat
# These both have two dimensions of lengths 4 and 3
dim(a)  # 4 3
# Combine them to make an array with three dimensions
x &lt;- list2array(list(a, b))
# The third dimension has length 2
dim(x)  # 4 3 2
# The first slice of the third dimension
slice(x, 3)    # a
# The second slice of the third dimension
slice(x, 3, 2) # b
# 'slice' works just like the bracket operator
slice(x, 1)        # x[1, , ]
slice(x, 1, 2)     # x[2, , ]
slice(x, 2, 1)     # x[, 1, ]
slice(x, 2, 1:2)   # x[, 1:2, ]

# Replace part of the array
y &lt;- slice(x, 3, 2, value = a)
# Now the second slice of the third dimension == a
slice(y, 3, 2)  # a
# and the sum across the third dimension == b
dimSums(y, 3)   # b
# Taking the sum removes that dimension
dim(y)              # 4 3 2
dim(dimSums(y, 1))  # 3 2
dim(dimSums(y, 2))  # 4 2
dim(dimSums(y, 3))  # 4 3

# Working with an 'affinity' object

basis("CHNOS+")
species("alanine")
a1 &lt;- affinity(O2 = c(-80, -60)) # i.e. pH = 7
a2 &lt;- affinity(O2 = c(-80, -60), pH = c(0, 14, 7))
# In the 2nd dimension (pH) get the 4th slice (pH = 7)
a3 &lt;- slice.affinity(a2, 2, 4)
all.equal(a1$values, a3$values)  # TRUE
</code></pre>

<hr>
<h2 id='util.data'>Functions for Checking Thermodynamic Data</h2><span id='topic+util.data'></span><span id='topic+thermo.refs'></span><span id='topic+check.EOS'></span><span id='topic+check.GHS'></span><span id='topic+check.OBIGT'></span><span id='topic+dumpdata'></span><span id='topic+RH2OBIGT'></span>

<h3>Description</h3>

<p>Show table of references in a web browser or get individual references for species.
Check self consistency of individual entries in database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  thermo.refs(key = NULL, keep.duplicates = FALSE)
  check.EOS(eos, model, prop, return.difference = TRUE)
  check.GHS(ghs, return.difference = TRUE)
  check.OBIGT()
  dumpdata(file)
  RH2OBIGT(compound = NULL, state = "cr", 
    file = system.file("extdata/adds/RH98_Table15.csv", package = "CHNOSZ"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.data_+3A_key">key</code></td>
<td>
<p>character, numeric, or list; bibliographic reference key(s)</p>
</td></tr>
<tr><td><code id="util.data_+3A_keep.duplicates">keep.duplicates</code></td>
<td>
<p>logical, keep duplicated references?</p>
</td></tr>
<tr><td><code id="util.data_+3A_eos">eos</code></td>
<td>
<p>dataframe, equations-of-state parameters in the format of <code>thermo()$OBIGT</code></p>
</td></tr>
<tr><td><code id="util.data_+3A_model">model</code></td>
<td>
<p>character, thermodynamic model (see <code><a href="#topic+thermo">thermo</a></code>)</p>
</td></tr>
<tr><td><code id="util.data_+3A_prop">prop</code></td>
<td>
<p>character, property of interest (&lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="util.data_+3A_return.difference">return.difference</code></td>
<td>
<p>logical, return the difference between database and calculated values?</p>
</td></tr>
<tr><td><code id="util.data_+3A_ghs">ghs</code></td>
<td>
<p>dataframe, containing G, H and S, in the format of <code>thermo()$OBIGT</code></p>
</td></tr>
<tr><td><code id="util.data_+3A_file">file</code></td>
<td>
<p>character, path to a file</p>
</td></tr>
<tr><td><code id="util.data_+3A_compound">compound</code></td>
<td>
<p>character, name of compound(s) in group additivity calculation</p>
</td></tr>
<tr><td><code id="util.data_+3A_state">state</code></td>
<td>
<p>character, physical state of species</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>thermo.refs</code> with default arguments uses <code><a href="utils.html#topic+browseURL">browseURL</a></code> to display the sources of thermodynamic data in <code>thermo()$refs</code>, with the URLs in that table showing as hyperlinks in the browser.
Otherwise, if <code>key</code> is character, the citation information for those reference <code>key</code>s (including URLs) are returned.
If <code>key</code> is numeric, the values refer to the species in those rows of <code>thermo()$OBIGT</code>, and the citation information for each listed reference (<code>thermo()$OBIGT$ref1</code>, <code>thermo()$OBIGT$ref2</code>) is returned.
If <code>key</code> is a list, it is interpreted as the result of a call to <code>subcrt</code>, and the citation information for each species involved in the calculation is returned.
Only unique references are returned, unless <code>keep.duplicates</code> is TRUE.
In that case, a single reference for each species is returned, ignoring anything in <code>thermo()$OBIGT$ref2</code>.
</p>
<p><code>check.EOS</code> calculates heat capacity (<code>prop = "Cp"</code>) or volume (<code>prop = "V"</code>) from equation-of-state parameters at 25 &deg;C and 1 bar.
<code>check.GHS</code> calculates G (standard molal Gibbs energy of formation from the elements) from H (standard molal enthalpy of formation) and S (standard molal entropy) at 25 &deg;C and 1 bar.
The calculated values of Cp, V, or G are then compared with the given values (i.e., database values).
If <code>return.difference</code> is TRUE (the default), the difference between the database and calculated values is returned.
If <code>return.difference</code> is FALSE, the difference is compared with a tolerance setting (see below).
If the absolute value of the difference exceeds the tolerance, the function prints a message and returns the calculated value (not the difference) of the property.
If the absolute value of the difference is less than the tolerance, the function returns NA with no message.
</p>
<p>For <code>check.EOS</code>, the thermodynamic parameters should be provided in <code>eos</code>, which is a data frame with column names in the same format as <code><a href="#topic+thermo">thermo</a>$OBIGT</code>.
For <code>check.GHS</code>, the data frame should include G, H, S, and the chemical formula of the species.
The default tolerances are 1 J/K.mol or 1 cal/K.mol for Cp (depending on the <code>E_units</code> for the species), 1 cm3/mol for V, and 100 cal/mol for G.
These can be changed by setting <code>thermo()$opt$Cp.tol</code>, <code>thermo()$opt$V.tol</code>, and <code>thermo()$opt$G.tol</code>.
</p>
<p><code>check.OBIGT</code> is a function to check self-consistency of each entry in the thermodynamic database, using <code>check.EOS</code> and <code>check.GHS</code>.
The output is a table listing only species that exceed at least one of the tolerance limits, giving the species index (rownumber in 'thermo()$OBIGT'), species name and state, and DCp, DV and DG, for the calculated differences (only those above the tolerances are given).
Values of DCp and DG are given in the units present in the data files.
This function is used to generate the file found at <code>extdata/thermo/OBIGT_check.csv</code>.
</p>
<p><code>dumpdata</code> returns all of the available data, from both the default and optional data files, or writes it to a file if <code>file</code> is not NULL.
The format is the same as <code><a href="#topic+thermo">thermo</a>$OBIGT</code>, except for a single prepended column named &lsquo;<span class="samp">&#8288;source&#8288;</span>&rsquo;, giving the source of the data (&lsquo;<span class="samp">&#8288;OBIGT&#8288;</span>&rsquo; refers to the default database, and &lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;SLOP98&#8288;</span>&rsquo;, and &lsquo;<span class="samp">&#8288;SUPCRT92&#8288;</span>&rsquo; are the optional data files).
</p>
<p><code>RH2OBIGT</code> implements a group additivity algorithm for standard molal thermodynamic properties and equations of state parameters of crystalline and liquid organic molecules from Richard and Helgeson, 1998. The names of the <code>compound</code>s and their physical <code>state</code> are searched for in the indicated <code>file</code>, that also contains chemical formulas and group stoichiometries; the names of the groups are stored in the column names of this file, and must be present in <code><a href="#topic+thermo">thermo</a>$OBIGT</code>. The default <code>file</code> (<code><a href="#topic+extdata">extdata</a>/thermo/RH98_Table15.csv</code>) includes data taken from Table 15 of Richard and Helgeson, 1998 for high molecular weight compounds in &lsquo;<span class="samp">&#8288;cr&#8288;</span>&rsquo;ystalline and &lsquo;<span class="samp">&#8288;liq&#8288;</span>&rsquo;uid states. An error is produced if any of the <code>compound</code>-<code>state</code> combinations is not found in the <code>file</code>, if any of the group names for a given <code>compound</code>-<code>state</code> combination is not found in <code>thermo()$OBIGT</code>, or if the chemical formula calculated from group additivity (with the aid of <code><a href="#topic+i2A">i2A</a></code> and <code><a href="#topic+as.chemical.formula">as.chemical.formula</a></code>) is not identical to that listed in the <code>file</code>.
</p>


<h3>Value</h3>

<p>The values returned (<code><a href="base.html#topic+invisible">invisible</a></code>-y) by <code>mod.OBIGT</code> are the rownumbers of the affected species.
</p>


<h3>References</h3>

<p>Richard, L. and Helgeson, H. C. (1998) Calculation of the thermodynamic properties at elevated temperatures and pressures of saturated and aromatic high molecular weight solid and liquid hydrocarbons in kerogen, bitumen, petroleum, and other organic matter of biogeochemical interest. <em>Geochim. Cosmochim. Acta</em> <b>62</b>, 3591&ndash;3636. <a href="https://doi.org/10.1016/S0016-7037%2897%2900345-1">doi:10.1016/S0016-7037(97)00345-1</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+thermo">thermo</a></code>, <code><a href="#topic+add.OBIGT">add.OBIGT</a></code>, <code><a href="#topic+mod.buffer">mod.buffer</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Citation information for Helgeson et al., 1998
thermo.refs("HOK+98")
# Two references for alanine
thermo.refs(info("alanine"))
# Three references for species in the reaction
s &lt;- subcrt(c("O2","O2"),c("gas","aq"),c(-1,1))
thermo.refs(s) 
## Not run: 
## Marked dontrun because it opens a browser
# Show the contents of thermo()$refs
thermo.refs()

## End(Not run) 

## Calculate thermodynamic properties of organic compounds
## using group additivity, after Richard and Helgeson, 1998
RH2OBIGT()
</code></pre>

<hr>
<h2 id='util.expression'>Functions to Express Chemical Formulas and Properties</h2><span id='topic+util.expression'></span><span id='topic+expr.species'></span><span id='topic+expr.property'></span><span id='topic+expr.units'></span><span id='topic+axis.label'></span><span id='topic+describe.basis'></span><span id='topic+describe.property'></span><span id='topic+describe.reaction'></span><span id='topic+syslab'></span><span id='topic+ratlab'></span>

<h3>Description</h3>

<p>Generate expressions suitable for axis labels and plot legends describing chemical species, properties and reactions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  expr.species(species, state = "aq", value=NULL, log=FALSE, molality=FALSE,
    use.state=FALSE, use.makeup=FALSE)
  expr.property(property, molality = FALSE)
  expr.units(property, prefix = "", per = "mol")
  axis.label(label, units = NULL, basis = thermo()$basis, prefix = "",
    molality = FALSE)
  describe.basis(ibasis = 1:nrow(basis), basis = thermo()$basis,
    digits = 1, oneline = FALSE, molality = FALSE, use.pH = TRUE)
  describe.property(property, value, digits = 0, oneline = FALSE,
    ret.val = FALSE)
  describe.reaction(reaction, iname = numeric(), states = NULL)
  syslab(system = c("K2O", "Al2O3", "SiO2", "H2O"), dash="-")
  ratlab(top = "K+", bottom = "H+", molality = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.expression_+3A_species">species</code></td>
<td>
<p>character, formula of a chemical species</p>
</td></tr>
<tr><td><code id="util.expression_+3A_state">state</code></td>
<td>
<p>character, designation of physical state</p>
</td></tr>
<tr><td><code id="util.expression_+3A_value">value</code></td>
<td>
<p>numeric, logarithm of activity or fugacity of species, or value of other property</p>
</td></tr>
<tr><td><code id="util.expression_+3A_log">log</code></td>
<td>
<p>logical, write logarithm of activity/fugacity/molality?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_molality">molality</code></td>
<td>
<p>logical, use molality (m) instead of activity (a) for aqueous species?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_use.state">use.state</code></td>
<td>
<p>logical, include state in expression?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_use.makeup">use.makeup</code></td>
<td>
<p>logical, use <code><a href="#topic+makeup">makeup</a></code> to count the elements?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_use.ph">use.pH</code></td>
<td>
<p>logical, use pH instead of log activity of H+?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_property">property</code></td>
<td>
<p>character, description of chemical property</p>
</td></tr>
<tr><td><code id="util.expression_+3A_prefix">prefix</code></td>
<td>
<p>character, prefix for units</p>
</td></tr>
<tr><td><code id="util.expression_+3A_per">per</code></td>
<td>
<p>character, denominator in units</p>
</td></tr>
<tr><td><code id="util.expression_+3A_label">label</code></td>
<td>
<p>character, description of species, condition or property</p>
</td></tr>
<tr><td><code id="util.expression_+3A_units">units</code></td>
<td>
<p>character, description of units</p>
</td></tr>
<tr><td><code id="util.expression_+3A_ibasis">ibasis</code></td>
<td>
<p>numeric, which basis species to include</p>
</td></tr>
<tr><td><code id="util.expression_+3A_basis">basis</code></td>
<td>
<p>data frame, definition of basis species</p>
</td></tr>
<tr><td><code id="util.expression_+3A_digits">digits</code></td>
<td>
<p>numeric, number of digits to show after decimal point</p>
</td></tr>
<tr><td><code id="util.expression_+3A_oneline">oneline</code></td>
<td>
<p>logical, make descriptions occupy a single line?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_ret.val">ret.val</code></td>
<td>
<p>logical, return only the value with the units?</p>
</td></tr>
<tr><td><code id="util.expression_+3A_reaction">reaction</code></td>
<td>
<p>data frame, definition of reaction</p>
</td></tr>
<tr><td><code id="util.expression_+3A_iname">iname</code></td>
<td>
<p>numeric, show names instead of formulas for these species</p>
</td></tr>
<tr><td><code id="util.expression_+3A_states">states</code></td>
<td>
<p>character, if &lsquo;<span class="samp">&#8288;all&#8288;</span>&rsquo;, show states for all species; numeric, which species to show states for</p>
</td></tr>
<tr><td><code id="util.expression_+3A_system">system</code></td>
<td>
<p>character, thermodynamic components</p>
</td></tr>
<tr><td><code id="util.expression_+3A_dash">dash</code></td>
<td>
<p>character to use for dash between components</p>
</td></tr>
<tr><td><code id="util.expression_+3A_top">top</code></td>
<td>
<p>character, the ion in the numerator of the ratio</p>
</td></tr>
<tr><td><code id="util.expression_+3A_bottom">bottom</code></td>
<td>
<p>character, the ion in the denominator of the ratio</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>expr.*</code> functions create <code><a href="base.html#topic+expression">expression</a></code>s using the <code><a href="grDevices.html#topic+plotmath">plotmath</a></code> syntax to describe the names and states and logarithms of activity or fugacity of chemical species, conditions including temperature and pressure and chemical properties such as Gibbs energy and volume.
</p>
<p><code>expr.species</code> constructs a formatted expression using the formula or name of a single chemical <code>species</code>.
With no other arguments, the formula is just formatted with the appropriate subscripts and superscripts.
Providing the physical <code>state</code> adds a variable to the expression (<em>a</em> for aqueous species and pure phases, except <em>f</em> for gases).
Set <code>molality</code> to TRUE to write <em>m</em> instead of <em>a</em> for aqueous species.
The state itself is written in the expression if <code>use.state</code> is TRUE.
If <code>log</code> is TRUE, the expression includes a &lsquo;<span class="samp">&#8288;log&#8288;</span>&rsquo; prefix.
Finally, provide a value in <code>value</code> to write an equation (something like logfO2 = -70), or set it to NA to only write the variable itself (e.g. logfO2).
Set <code>use.makeup</code> to TRUE to use <code><a href="#topic+makeup">makeup</a></code> to parse the chemical formula.
This was an older default action that had the undesirable effect of reordering and grouping all the elements, and has been replaced with a different splitting algorithm so that coefficients and charges are sub/superscripted without affecting the intervening text.
</p>
<p><code>expr.property</code> accepts a description in <code>property</code> that indicates the chemical property of interest.
Uppercase letters are italicized, and lowercase letters are italicized and subscripted.
Other specific characters are parsed as follows (case-sensitive):
</p>

<table>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;D&#8288;</span>&rsquo; </td><td style="text-align: left;"> Delta </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo; </td><td style="text-align: left;"> bold A (chemical affinity) </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;p&#8288;</span>&rsquo; </td><td style="text-align: left;"> subscript italic P (for isobaric heat capacity) </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;0&#8288;</span>&rsquo; </td><td style="text-align: left;"> degree sign (for a standard-state property) </td>
</tr>
<tr>
 <td style="text-align: left;">  
     &lsquo;<span class="samp">&#8288;l&#8288;</span>&rsquo; </td><td style="text-align: left;"> subscript lambda </td>
</tr>
<tr>
 <td style="text-align: left;">  
     &lsquo;<span class="samp">&#8288;'&#8288;</span>&rsquo; </td><td style="text-align: left;"> prime symbol </td>
</tr>
<tr>
 <td style="text-align: left;">  
  </td>
</tr>

</table>

<p>A &lsquo;<span class="samp">&#8288;0&#8288;</span>&rsquo; gets interpreted as a degree sign only if it does not immediately follow a number (so that e.g. &lsquo;<span class="samp">&#8288;2.303&#8288;</span>&rsquo; can be included in an expression).
</p>
<p>Every other character that is one of the <code><a href="base.html#topic+letters">letters</a></code> or <code><a href="base.html#topic+LETTERS">LETTERS</a></code> in the description of the property is italicized in the expression; other characters such as numerals or mathematical operators are shown without any special formatting.
Special cases for the <code>property</code> argument (&lsquo;<span class="samp">&#8288;logK&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pH&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;ZC&#8288;</span>&rsquo;) are interpreted as simple expressions, and are not parsed according to the above rules.
</p>
<p><code>expr.units</code> returns an expression for the units, based on one or more characters appearing in the <code>property</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;A&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;H&#8288;</span>&rsquo; </td><td style="text-align: left;"> energy </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;Cp&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;S&#8288;</span>&rsquo; </td><td style="text-align: left;"> energy per Kelvin </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;V&#8288;</span>&rsquo; </td><td style="text-align: left;"> volume </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;E&#8288;</span>&rsquo; </td><td style="text-align: left;"> volume per Kelvin </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo; </td><td style="text-align: left;"> pressure </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo; </td><td style="text-align: left;"> temperature </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo; </td><td style="text-align: left;"> electrical potential </td>
</tr>
<tr>
 <td style="text-align: left;">
     &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo; </td><td style="text-align: left;"> ionic strength </td>
</tr>
<tr>
 <td style="text-align: left;">  </td>
</tr>

</table>

<p>If none of those characters appears in the property, the expression is an empty character (no units). If a <code>prefix</code> is given, it is added to the expression. The denominator of the units (default &lsquo;<span class="samp">&#8288;mol&#8288;</span>&rsquo;) is taken from the <code>per</code> argument; it is applied to all units except for &lsquo;<span class="samp">&#8288;P&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;T&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo;, and &lsquo;<span class="samp">&#8288;IS&#8288;</span>&rsquo;.
</p>
<p><code>axis.label</code> accepts a generic description of a <code>label</code>.
If this matches the chemical formula of one of the basis species in the <code>basis</code> argument, the expression for the label is generated using <code>expr.species</code> with <code>log</code> set to the physical state of the basis species.
Otherwise, the expression is built by combining the output of <code>expr.property</code> with <code>expr.units</code> (or the value in <code>units</code>, if it is supplied), placing a comma between the two. 
This function is used extensively in <code><a href="#topic+diagram">diagram</a></code> and also appears in many of the examples.
Note that <code><a href="#topic+diagram">diagram</a></code> sets <code>molality</code> to TRUE if <code>IS</code> was supplied as an argument to <code><a href="#topic+affinity">affinity</a></code>.
</p>
<p><code>describe.basis</code> makes an expression summarizing the basis species definition (logarithms of activity or fugacity of the basis species) provided in <code>basis</code>; only the basis species identified by <code>ibasis</code> are included. 
</p>
<p><code>describe.property</code> makes an expression summarizing the properties supplied in <code>property</code>, along with their <code>value</code>s. The expressions returned by both functions consist of a property, an equals sign, and a value (with units where appropriate); the expressions have a length equal to the number of property/value pairs. If <code>oneline</code> is TRUE, the property/value pairs are combined into a single line, separated by commas. The number of digits shown after the decimal point in the values is controlled by <code>digits</code>. If <code>ret.val</code> is TRUE, only the values and their units are returned; this is useful for labeling plots with values of temperature.
</p>
<p><code>describe.reaction</code> makes an expression summarizing a chemical reaction.
The <code>reaction</code> data frame can be generated using <code><a href="#topic+subcrt">subcrt</a></code>.
Based on the sign of their reaction coefficients, species are placed on the reactant (left) or product (right) side of the reaction, where the species with their coefficients are separated by plus signs; the two sides of the reaction are separated by a reaction double arrow (Unicode U+21CC).
Coefficients equal to 1 are not shown.
Chemical formulas of species include the physical state if <code>states</code> is &lsquo;<span class="samp">&#8288;all&#8288;</span>&rsquo;, or a numeric value indicating which species to label with the state.
Names of species (as provided in <code>reaction</code>) are shown instead of chemical formulas for the species identified by <code>iname</code>.
</p>
<p><code>syslab</code> formats the given thermodynamic components (using <code>expr.species</code>) and adds intervening en dashes.
</p>
<p><code>ratlab</code> produces a expression for the activity ratio between the ions in the <code>top</code> and <code>bottom</code> arguments.
The default is a ratio with H<sup>+</sup>, i.e. (activity of the ion) / [(activity of H+) ^ (charge of the ion)].
</p>


<h3>See Also</h3>

<p><code><a href="#topic+util.legend">util.legend</a></code> for other functions to make legends.
<code>demo("saturation")</code> for examples of <code>syslab</code> and <code>ratlab</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Show descriptions of species and properties on a plot
plot(0, 0, xlim = c(1,5), ylim = c(1,5), xlab = "function", ylab = "example")
text0 &lt;- function(...) text(..., adj = 0)
# Species
text0(1, 1, expr.species("CO2"))
text0(1, 2, expr.species("CO2", use.state = TRUE))
text0(1, 3, expr.species("CO2", log = TRUE, use.state = TRUE))
text0(1, 4, expr.species("CO2", log = TRUE))
text0(1, 5, expr.species("CO2", log = TRUE, value = -3))
# Properties
text0(2, 1, expr.property("A"))
text0(2, 2, expr.property("DV"))
text0(2, 3, expr.property("DG0f"))
text0(2, 4, expr.property("DCp0,r"))
text0(2, 5, expr.property("T"))
# Units
text0(3, 1, expr.units("A", prefix = "k"))
text0(3, 2, expr.units("DV"))
text0(3, 3, expr.units("DG0f", prefix = "k"))
text0(3, 4, expr.units("DCp0,r"))
text0(3, 5, expr.units("T"))
# axis.label
text0(4, 1, axis.label("DG0f"))
text0(4, 2, axis.label("T"))
text0(4, 3, axis.label("pH"))
text0(4, 4, axis.label("Eh"))
text0(4, 5, axis.label("IS"))
# describe.basis
basis("CHNOS+")
dbasis &lt;- describe.basis(oneline = TRUE, digits = 0)
property &lt;- c("P", "T", "Eh", "pH", "IS")
value &lt;- c(1, 42.42, -1, 7, 0.1)
dprop &lt;- describe.property(property, value, oneline = TRUE)
text(3, 1.5, dbasis)
text(3, 2.5, dprop)
dbasis &lt;- describe.basis(c(1, 5))
dprop &lt;- describe.property(property[1:2], value[1:2])
legend(2.4, 3.9, legend=c(dbasis, dprop), bty = "n")
# describe.reaction
# Reaction is automatically balanced because basis species are defined
reaction &lt;- subcrt("glucose", -1)$reaction
text(3, 4.25, describe.reaction(reaction))
text(3, 4.5, describe.reaction(reaction, states = "all"))
text(3, 4.75, describe.reaction(reaction, iname = 1:4))
title(main = "Plot labels for chemical species and thermodynamic properties")
</code></pre>

<hr>
<h2 id='util.fasta'>Functions for Reading FASTA Files and Downloading from UniProt</h2><span id='topic+util.fasta'></span><span id='topic+read.fasta'></span><span id='topic+count.aa'></span>

<h3>Description</h3>

<p>Search the header lines of a FASTA file, read protein sequences from a file, count numbers of amino acids in each sequence, and download sequences from UniProt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read.fasta(file, iseq = NULL, ret = "count", lines = NULL, 
    ihead = NULL, start=NULL, stop=NULL, type="protein", id = NULL)
  count.aa(seq, start=NULL, stop=NULL, type="protein")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.fasta_+3A_file">file</code></td>
<td>
<p>character, path to FASTA file</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_iseq">iseq</code></td>
<td>
<p>numeric, which sequences to read from the file</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_ret">ret</code></td>
<td>
<p>character, specification for type of return (count, sequence, or FASTA format)</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_lines">lines</code></td>
<td>
<p>list of character, supply the lines here instead of reading them from file</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_ihead">ihead</code></td>
<td>
<p>numeric, which lines are headers</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_start">start</code></td>
<td>
<p>numeric, position in sequence to start counting</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_stop">stop</code></td>
<td>
<p>numeric, position in sequence to stop counting</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_type">type</code></td>
<td>
<p>character, sequence type (protein or DNA)</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_id">id</code></td>
<td>
<p>character, value to be used for <code>protein</code> in output table</p>
</td></tr>
<tr><td><code id="util.fasta_+3A_seq">seq</code></td>
<td>
<p>character, amino acid sequence of a protein</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>read.fasta</code> is used to retrieve entries from a FASTA file.
Use <code>iseq</code> to select the sequences to read (the default is all sequences).
The function returns various formats depending on the value of <code>ret</code>.
The default &lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo; returns a data frame of amino acid counts (the data frame can be given to <code><a href="#topic+add.protein">add.protein</a></code> in order to add the proteins to <code><a href="#topic+thermo">thermo</a>$protein</code>), &lsquo;<span class="samp">&#8288;seq&#8288;</span>&rsquo; returns a list of sequences, and &lsquo;<span class="samp">&#8288;fas&#8288;</span>&rsquo; returns a list of lines extracted from the FASTA file, including the headers (this can be used e.g. to generate a new FASTA file with only the selected sequences).
If the line numbers of the header lines were previously determined, they can be supplied in <code>ihead</code>. 
Optionally, the lines of a previously read file may be supplied in <code>lines</code> (in this case no file is needed so <code>file</code> should be set to &quot;&quot;).
When <code>ret</code> is &lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;, the names of the proteins in the resulting data frame are parsed from the header lines of the file, unless <code>id</code> is provided.
If <code>id</code> is not given, and a UniProt FASTA header is detected (regular expression <code>"\|......\|.*_"</code>), information there (accession, name, organism) is split into the <code>protein</code>, <code>abbrv</code>, and <code>organism</code> columns of the resulting data frame.
</p>
<p><code>count.aa</code> counts the occurrences of each amino acid or nucleic-acid base in a sequence (<code>seq</code>).
For amino acids, the columns in the returned data frame are in the same order as <code>thermo()$protein</code>.
The matching of letters is case-insensitive.
A warning is generated if any character in <code>seq</code>, excluding spaces, is not one of the single-letter amino acid or nucleobase abbreviations.
<code>start</code> and/or <code>stop</code> can be provided to count a fragment of the sequence (extracted using <code><a href="base.html#topic+substr">substr</a></code>).
If only one of <code>start</code> or <code>stop</code> is present, the other defaults to 1 (<code>start</code>) or the length of the sequence (<code>stop</code>).
</p>


<h3>Value</h3>

<p><code>read.fasta</code> returns a list of sequences or lines (for <code>ret</code> equal to &lsquo;<span class="samp">&#8288;seq&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;fas&#8288;</span>&rsquo;, respectively), or a data frame with amino acid compositions of proteins (for <code>ret</code> equal to &lsquo;<span class="samp">&#8288;count&#8288;</span>&rsquo;) with columns corresponding to those in <code><a href="#topic+thermo">thermo</a>$protein</code>. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+seq2aa">seq2aa</a></code>, like <code>count.aa</code>, counts amino acids in a user-input sequence, but returns a data frame in the format of <code>thermo()$protein</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Reading a protein FASTA file
# The path to the file
file &lt;- system.file("extdata/protein/EF-Tu.aln", package = "CHNOSZ")
# Read the sequences, and print the first one
read.fasta(file, ret = "seq")[[1]]
# Count the amino acids in the sequences
aa &lt;- read.fasta(file)
# Compute lengths (number of amino acids)
protein.length(aa)

## Not run: 
## Count amino acids in a sequence
count.aa("GGSGG")
# Warnings are issued for unrecognized characters
atest &lt;- count.aa("WhatAmIMadeOf?")
# There are 3 "A" (alanine)
atest[, "A"]

## End(Not run)
</code></pre>

<hr>
<h2 id='util.formula'>Functions to Work with Chemical Formulas</h2><span id='topic+util.formula'></span><span id='topic+as.chemical.formula'></span><span id='topic+mass'></span><span id='topic+entropy'></span><span id='topic+GHS'></span><span id='topic+ZC'></span><span id='topic+i2A'></span>

<h3>Description</h3>

<p>Calculate the standard molal entropy of elements in a compound; calculate the standard molal Gibbs energy or enthalpy of formation, or standard molal entropy, from the other two; list coefficients of selected elements in a chemical formula; calculate the average oxidation state of carbon. Create a stoichiometric matrix for selected species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  as.chemical.formula(makeup, drop.zero = TRUE)
  mass(formula)
  entropy(formula)
  GHS(formula, G = NA, H = NA, S = NA, T = 298.15, E_units = "J")
  ZC(formula)
  i2A(formula)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.formula_+3A_makeup">makeup</code></td>
<td>
<p>numeric, object returned by <code>makeup</code></p>
</td></tr>
<tr><td><code id="util.formula_+3A_drop.zero">drop.zero</code></td>
<td>
<p>logical, drop elements with a coefficient of zero?</p>
</td></tr>
<tr><td><code id="util.formula_+3A_formula">formula</code></td>
<td>
<p>character, chemical formulas, or numeric, rownumbers in <code>thermo()$OBIGT</code></p>
</td></tr>
<tr><td><code id="util.formula_+3A_g">G</code></td>
<td>
<p>numeric, standard molal Gibbs energy of formation from the elements</p>
</td></tr>
<tr><td><code id="util.formula_+3A_h">H</code></td>
<td>
<p>numeric, standard molal enthalpy of formation from the elements</p>
</td></tr>
<tr><td><code id="util.formula_+3A_s">S</code></td>
<td>
<p>numeric, standard molal molal entropy</p>
</td></tr>
<tr><td><code id="util.formula_+3A_t">T</code></td>
<td>
<p>numeric, temperature in Kelvin</p>
</td></tr>
<tr><td><code id="util.formula_+3A_e_units">E_units</code></td>
<td>
<p>character, energy units (J or cal)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>i2A</code> returns a stoichiometric matrix representing the elemental composition of the <code>formula</code>s.
Each column corresponds to an element that is present in at least one of the formulas; some element counts will be zero if not all formula have the same elements.
If a matrix is passed to <code>i2A</code> it is returned unchanged.
</p>
<p><code>as.chemical.formula</code> makes a character string representing a chemical formula from a vector of coefficients with names corresponding to the elements (e.g., the output of <code><a href="#topic+makeup">makeup</a></code>) or from a stoichiometric matrix (output of <code>i2A</code>).
Each elemental symbol is written followed by its coefficient; negative coefficients are signed.
Any coefficients equal to 1 are not explicitly written, and any charge (indicated by <code>makeup</code> as &lsquo;<span class="samp">&#8288;Z&#8288;</span>&rsquo;) is shown as a signed number at the end of the formula.
If the formula is uncharged, and the last element has a negative coefficient, +0 is shown at the end of the formula to indicate a charge of zero.
</p>
<p>The remaining functions documented here accept vectors of chemical formulas, species indices, or a mixture of both, or stoichiometric matrices with elements on the columns.
</p>
<p><code>mass</code> and <code>entropy</code> return the sums of masses or entropies of elements in each of the <code>formula</code>s.
The masses are calculated using the masses of the elements in their natural isotopic distribution, and the entropies, in J K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code>, are calculated using the entropies of the compounds of the pure elements in their stable states at 25 &deg;C and 1 bar.
The properties of the elements used by this function are taken from <code><a href="#topic+thermo">thermo</a>$element</code>.
</p>
<p><code>GHS</code> computes one of the standard molal Gibbs energy or enthalpy of formation from the elements, or standard molal entropy, from values of the other two.
The <code>formula</code>, <code>G</code>, <code>H</code> and <code>S</code> arguments must all have the same length.
The entropies of the elements (<code>Se</code>) in each <code>formula</code> are calculated using <code><a href="#topic+entropy">entropy</a></code>, which gives values in Joules.
If <code>E_units</code> is &lsquo;<span class="samp">&#8288;cal&#8288;</span>&rsquo;, the values are converted to calories.
The equation in effect can be written as <code class="reqn">{\Delta}G^{\circ}={\Delta}H^{\circ}-T{\Delta}S^{\circ}</code>, where <code class="reqn">{\Delta}S^{\circ}=S-S_e</code> and <code class="reqn">T</code> is the temperature given in <code>T</code> (defaults to 298.15 K) (note that <code>G</code> and <code>H</code> in the arguments correspond respectively to <code class="reqn">{\Delta}G^{\circ}</code> and <code class="reqn">{\Delta}H^{\circ}</code> in the equation).
For each formula, if one of <code>G</code>, <code>H</code>, or <code>S</code> is NA, its value is calculated from the other two.
Otherwise, the values are returned unchanged.
</p>
<p><code>ZC</code> returns the average oxidation state of carbon (<code class="reqn">Z_{\mathrm{C}}</code>) calculated from ratios of the elements in the chemical <code>formula</code>s.
The equation used is
<code class="reqn">Z_{\mathrm{C}}=\frac{Z-n_{\mathrm{H}}+2\left(n_{\mathrm{O}}+n_{\mathrm{S}}\right)+3n_{\mathrm{N}}}{n_{\mathrm{C}}}</code>, where the <code class="reqn">n</code> refer to the number of the indicated element in the formula and <code class="reqn">Z</code> is the charge (Dick and Shock, 2011).
The result is <code>NaN</code> for any formula that does not contain carbon. Elements other than those shown in the equation are not included in the calculation, and produce a warning.
</p>


<h3>Value</h3>

<p><code>mass</code>, <code>entropy</code>, and <code>ZC</code> return numeric values.
<code>as.chemical.formula</code> returns a character object.
<code>GHS</code> returns a matrix with column names &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;H&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;S&#8288;</span>&rsquo;, and <code>i2A</code> returns a matrix with column names corresponding to the elements in the formulas.
</p>


<h3>References</h3>

<p>Dick, J. M. and Shock, E. L. (2011) Calculation of the relative chemical stabilities of proteins as a function of temperature and redox chemistry in a hot spring. <em>PLOS One</em> <b>6</b>, e22782. <a href="https://doi.org/10.1371/journal.pone.0022782">doi:10.1371/journal.pone.0022782</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+makeup">makeup</a></code>, used by <code>mass</code> and <code>entropy</code>, and <code>ZC</code> and <code>i2A</code> for counting the elements in a formula (the latter two make use of the <code>count.zero</code> argument).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Mass and entropy from chemical formulas
mass("H2O")
entropy("H2O")
mass("-1")  # electron
entropy("-1")

## Different ways to get the formula of alanine
iA &lt;- info("alanine")
info(iA)$formula
as.chemical.formula(makeup(iA))

## Converting among Gibbs energy, enthalpy, entropy
# Calculate the value of G from H and S
GHS("H2O", H = water("H")$H, S = water("S")$S)
# The result is not exactly equal to the value from water("G"),
# probably because of different entropies of the elements

## Average oxidation states of carbon
ZC(c("CO2", "CH4", "CHNOSZ"))  # 4, -4, 7
si &lt;- info("LYSC_CHICK")
# Can use species index or formula
ZC(si)
ZC(info(si)$formula)

## Calculate the chemical formulas, then
## ZC of all of the proteins in CHNOSZ's database
pf &lt;- protein.formula(thermo()$protein)
range(mass(pf))
# Use na.rm = TRUE because we have a "protein" with a formula of H2O
range(ZC(pf), na.rm = TRUE)
</code></pre>

<hr>
<h2 id='util.legend'>Functions to Make Legend Text</h2><span id='topic+util.legend'></span><span id='topic+lNaCl'></span><span id='topic+lS'></span><span id='topic+lT'></span><span id='topic+lP'></span><span id='topic+lTP'></span><span id='topic+lex'></span>

<h3>Description</h3>

<p>Generate expressions describing system conditions that can be added to legends of plots.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  lNaCl(x, digits = 2)
  lS(x, digits = 3)
  lT(x, digits = 0)
  lP(x, digits = 0)
  lTP(x, y, digits = 0)
  lex(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.legend_+3A_x">x</code></td>
<td>
<p>numeric, value of the property</p>
</td></tr>
<tr><td><code id="util.legend_+3A_digits">digits</code></td>
<td>
<p>numeric, digits for rounding</p>
</td></tr>
<tr><td><code id="util.legend_+3A_y">y</code></td>
<td>
<p>numeric, value of pressure</p>
</td></tr>
<tr><td><code id="util.legend_+3A_...">...</code></td>
<td>
<p>language, objects to combine in an expression</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are used to make expressions for common chemical system variables that can be used in plot legends.
<code>lNaCl</code> describe the molality of NaCl, and <code>lS</code> the total molality of sulfur.
<code>lT</code> and <code>lP</code> describe the temperature and pressure.
<code>lTP</code> describe the temperature and pressure together, separated by a comma.
</p>
<p>The above functions return language objects, which can be combined with <code>lex</code> to make an expression that when used in <code><a href="graphics.html#topic+legend">legend</a></code> appears on multiple lines.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+util.expression">util.expression</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot.new()
l &lt;- lex(lTP(100, "Psat"), lNaCl(1), lS(1e-3))
legend("center", l)
</code></pre>

<hr>
<h2 id='util.list'>Functions to Work with Lists</h2><span id='topic+util.list'></span><span id='topic+which.pmax'></span>

<h3>Description</h3>

<p>Identify list elements that have the maximum (or minimum) values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  which.pmax(x, maximum = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.list_+3A_x">x</code></td>
<td>
<p>list of numeric vectors</p>
</td></tr>
<tr><td><code id="util.list_+3A_maximum">maximum</code></td>
<td>
<p>logical, find maximum values or minimum values?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>which.pmax</code> takes a list of equal-length numeric vectors or equal-dimension arrays in <code>x</code> and returns the index of the list element that has the maximum value at each point.
Change <code>maximum</code> to <code>FALSE</code> to find the minimum values instead.
</p>

<hr>
<h2 id='util.misc'>Functions for Miscellaneous Tasks</h2><span id='topic+util.misc'></span><span id='topic+dPdTtr'></span><span id='topic+Ttr'></span><span id='topic+GHS_Tr'></span><span id='topic+unitize'></span>

<h3>Description</h3>

<p>Calculate <code class="reqn">dP/dT</code> and temperature of polymorphic transitions; scale logarithms of activity to a desired total activity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  dPdTtr(ispecies, ispecies2 = NULL)
  Ttr(ispecies, ispecies2 = NULL, P = 1, dPdT = NULL)
  GHS_Tr(ispecies, Htr)
  unitize(logact = NULL, length = NULL, logact.tot = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.misc_+3A_ispecies">ispecies</code></td>
<td>
<p>numeric, species index of a mineral phase</p>
</td></tr>
<tr><td><code id="util.misc_+3A_ispecies2">ispecies2</code></td>
<td>
<p>numeric, species index of next mineral phase (the default is ispecies + 1)</p>
</td></tr>
<tr><td><code id="util.misc_+3A_p">P</code></td>
<td>
<p>numeric, pressure (bar)</p>
</td></tr>
<tr><td><code id="util.misc_+3A_dpdt">dPdT</code></td>
<td>
<p>numeric, values of (<code class="reqn">dP/dT</code>) of polymorphic transitions (<code>Ttr</code>)</p>
</td></tr>
<tr><td><code id="util.misc_+3A_htr">Htr</code></td>
<td>
<p>numeric, enthalpy(ies) of transition (cal/mol)</p>
</td></tr>
<tr><td><code id="util.misc_+3A_logact">logact</code></td>
<td>
<p>numeric, logarithms of activity</p>
</td></tr>
<tr><td><code id="util.misc_+3A_length">length</code></td>
<td>
<p>numeric, numbers of residues</p>
</td></tr>
<tr><td><code id="util.misc_+3A_logact.tot">logact.tot</code></td>
<td>
<p>numeric, logarithm of total activity</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>dPdTtr</code> returns values of <code class="reqn">(dP/dT)_{Ttr}</code>, where <code class="reqn">Ttr</code> represents the transition temperature, of the polymorphic transition at the high-<code class="reqn">T</code> stability limit of the <code>ispecies</code> in <code>thermo()$OBIGT</code> (other than checking that the names match, the function does not check that the species in fact represent different phases of the same mineral).
<code>dPdTtr</code> takes account of the Clapeyron equation, <code class="reqn">(dP/dT)_{Ttr}</code>=<code class="reqn">{\Delta}S/{\Delta}V</code>, where <code class="reqn">{\Delta}S</code> and <code class="reqn">{\Delta}V</code> represent the changes in entropy and volume of polymorphic transition, and are calculated using <code>subcrt</code> at Ttr from the standard molal entropies and volumes of the two phases involved.
Using values of <code>dPdT</code> calculated using <code>dPdTtr</code> or supplied in the arguments, <code>Ttr</code> returns as a function of <code>P</code> values of the upper transition temperature of the mineral phase represented by <code>ispecies</code>.
</p>
<p><code>GHS_Tr</code> can be used to calculate values of G, H, and S at Tr for the cr2, cr3, and cr4 phases in the database.
It combines the given <code>Htr</code> (enthalpies of transition) with the database values of GHS @ Tr only for the phase that is stable at 298.15 K (cr) and the transition temperatures and Cp coefficients for higher-temperature phases, to calculate the GHS @ Tr (i.e. low-temperature metastable conditions) of the phases that are stable at higher temperatures.
</p>
<p><code>unitize</code> scales the logarithms of activities given in <code>logact</code> so that the logarithm of total activity of residues is equal to zero (i.e. total activity of residues is one), or to some other value set in <code>logact.tot</code>.
<code>length</code> indicates the number of residues in each species.
If <code>logact</code> is NULL, the function takes the logarithms of activities from the current species definition.
If any of those species are proteins, the function gets their lengths using <code>protein.length</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# We need the Helgeson et al., 1978 minerals for this example
add.OBIGT("SUPCRT92")
# That replaces the existing enstatite with the first phase;
# the other phases are appended to the end of thermo()$OBIGT
i1 &lt;- info("enstatite")
i2 &lt;- info("enstatite", "cr2")
i3 &lt;- info("enstatite", "cr3")
# (dP/dT) of transitions
dPdTtr(i1, i2)  # first transition
dPdTtr(i2, i3)  # second transition
# Temperature of transitions (Ttr) as a function of P
Ttr(i1, i2, P = c(1,10,100,1000))
Ttr(i2, i3, P = c(1,10,100,1000))
# Restore default database
OBIGT()

# Calculate the GHS at Tr for the high-temperature phases of iron
# using transition enthalpies from the SUPCRT92 database (sprons92.dat)
Htr &lt;- c(326.0, 215.0, 165.0)
iiron &lt;- info("iron")
GHS_Tr(iiron, Htr)
# The results calculated above are stored in the database ...
info(1:3 + iiron)[, c("G", "H", "S")]
# ... meaning that we can recalculate the transition enthalpies using subcrt()
sapply(info(0:2 + iiron)$T, function(T) {
  # A very small T increment around the transition temperature
  T &lt;- convert(c(T-0.01, T), "C")
  # Use suppressMessages to make the output less crowded
  sres &lt;- suppressMessages(subcrt("iron", T = T, P = 1))
  diff(sres$out$iron$H)
})

## Scale logarithms of activity
# Suppose we have two proteins whose lengths are 100 and 
# 200; what are the logarithms of activity of the proteins 
# that are equal to each other and that give a total 
# activity of residues equal to unity?
logact &lt;- c(-3, -3)  # could be any two equal numbers
length &lt;- c(100, 200)
logact.tot &lt;- 0
loga &lt;- unitize(logact, length, logact.tot)
# The proteins have equal activity
loga[1] == loga[2]
# The sum of activity of the residues is unity
all.equal(sum(10^loga * length), 1)
## What if the activity of protein 2 is ten times that of protein 1?
logact &lt;- c(-3, -2)
loga &lt;- unitize(logact, length, logact.tot)
# The proteins have unequal activity,
# but the activities of residues still add up to one
all.equal(loga[2] - loga[1], 1)
all.equal(sum(10^loga * length), 1)
</code></pre>

<hr>
<h2 id='util.plot'>Functions to Create and Modify Plots</h2><span id='topic+util.plot'></span><span id='topic+thermo.plot.new'></span><span id='topic+thermo.axis'></span><span id='topic+label.plot'></span><span id='topic+usrfig'></span><span id='topic+label.figure'></span><span id='topic+water.lines'></span><span id='topic+mtitle'></span><span id='topic+ZC.col'></span><span id='topic+add.alpha'></span>

<h3>Description</h3>

<p>Initialize a new plot window using preset parameters, add an axis or title to a plot, generate labels for axes and subplots, add stability lines for water, get colors for a set of numeric values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  thermo.plot.new(xlim, ylim, xlab, ylab, cex = par("cex"),
    mar = NULL, lwd = par("lwd"), side = c(1,2,3,4), 
    mgp = c(1.7, 0.3, 0), cex.axis = par("cex"), col = par("col"),
    yline = NULL, axs = "i", plot.box = TRUE, las = 1,
    xline = NULL, grid = "", col.grid = "gray", ...)
  thermo.axis(lab = NULL, side = 1:4, line = 1.5, cex = par("cex"),
    lwd = par("lwd"), col = par("col"), grid = "", col.grid = "gray",
    plot.line = FALSE)
  label.plot(x, xfrac = 0.07, yfrac = 0.93, paren = FALSE,
    italic = FALSE, ...)
  usrfig()
  label.figure(x, xfrac = 0.05, yfrac = 0.95, paren = FALSE,
    italic = FALSE, ...)
  water.lines(eout, which = c("oxidation","reduction"),
    lty = 2, lwd = 1, col = par("fg"), plot.it = TRUE)
  mtitle(main, line = 0, spacing = 1, ...)
  ZC.col(z)
  add.alpha(col, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.plot_+3A_xlim">xlim</code></td>
<td>
<p>numeric, limits of the <code class="reqn">x</code>-axis</p>
</td></tr>
<tr><td><code id="util.plot_+3A_ylim">ylim</code></td>
<td>
<p>numeric, limits of the <code class="reqn">y</code>-axis</p>
</td></tr>
<tr><td><code id="util.plot_+3A_xlab">xlab</code></td>
<td>
<p>character, <code class="reqn">x</code>-axis label</p>
</td></tr>
<tr><td><code id="util.plot_+3A_ylab">ylab</code></td>
<td>
<p>character, <code class="reqn">y</code>-axis label</p>
</td></tr>
<tr><td><code id="util.plot_+3A_cex">cex</code></td>
<td>
<p>numeric, character expansion factor for labels</p>
</td></tr>
<tr><td><code id="util.plot_+3A_mar">mar</code></td>
<td>
<p>numeric, width (number of lines) of margins on each side of plot</p>
</td></tr>
<tr><td><code id="util.plot_+3A_lwd">lwd</code></td>
<td>
<p>numeric, line width</p>
</td></tr>
<tr><td><code id="util.plot_+3A_side">side</code></td>
<td>
<p>numeric, which sides of plot to draw axes</p>
</td></tr>
<tr><td><code id="util.plot_+3A_mgp">mgp</code></td>
<td>
<p>numeric, sizes of margins of plot</p>
</td></tr>
<tr><td><code id="util.plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>numeric, character expansion factor for names of axes</p>
</td></tr>
<tr><td><code id="util.plot_+3A_col">col</code></td>
<td>
<p>character, color</p>
</td></tr>
<tr><td><code id="util.plot_+3A_yline">yline</code></td>
<td>
<p>numeric, margin line on which to plot <code class="reqn">y</code>-axis name</p>
</td></tr>
<tr><td><code id="util.plot_+3A_axs">axs</code></td>
<td>
<p>character, setting for axis limit calculation</p>
</td></tr>
<tr><td><code id="util.plot_+3A_plot.box">plot.box</code></td>
<td>
<p>logical, draw a box around the plot?</p>
</td></tr>
<tr><td><code id="util.plot_+3A_las">las</code></td>
<td>
<p>numeric, style for axis labels</p>
</td></tr>
<tr><td><code id="util.plot_+3A_xline">xline</code></td>
<td>
<p>numeric, margin line on which to plot <code class="reqn">x</code>-axis name</p>
</td></tr>
<tr><td><code id="util.plot_+3A_grid">grid</code></td>
<td>
<p>character, type of grid (&lsquo;<span class="samp">&#8288;major&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;minor&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;both&#8288;</span>&rsquo;)</p>
</td></tr>
<tr><td><code id="util.plot_+3A_col.grid">col.grid</code></td>
<td>
<p>character, color of the grid lines</p>
</td></tr>
<tr><td><code id="util.plot_+3A_plot.line">plot.line</code></td>
<td>
<p>logical, draw axis lines?</p>
</td></tr>
<tr><td><code id="util.plot_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code>par</code> or <code>mtext</code></p>
</td></tr>
<tr><td><code id="util.plot_+3A_lab">lab</code></td>
<td>
<p>character, axis label</p>
</td></tr>
<tr><td><code id="util.plot_+3A_line">line</code></td>
<td>
<p>numeric, margin line on which to place axis label or plot title</p>
</td></tr>
<tr><td><code id="util.plot_+3A_x">x</code></td>
<td>
<p>character, label to place on plot</p>
</td></tr>
<tr><td><code id="util.plot_+3A_xfrac">xfrac</code></td>
<td>
<p>numeric, fractional location on <code class="reqn">x</code>-axis for placement of label</p>
</td></tr>
<tr><td><code id="util.plot_+3A_yfrac">yfrac</code></td>
<td>
<p>numeric, fractional location on <code class="reqn">y</code>-axis for placement of label</p>
</td></tr>
<tr><td><code id="util.plot_+3A_paren">paren</code></td>
<td>
<p>logical, add parentheses around label text?</p>
</td></tr>
<tr><td><code id="util.plot_+3A_italic">italic</code></td>
<td>
<p>logical, italicize label text?</p>
</td></tr>
<tr><td><code id="util.plot_+3A_eout">eout</code></td>
<td>
<p>data frame, output of <code><a href="#topic+affinity">affinity</a></code>, <code><a href="#topic+equilibrate">equilibrate</a></code>, or <code><a href="#topic+diagram">diagram</a></code></p>
</td></tr>
<tr><td><code id="util.plot_+3A_which">which</code></td>
<td>
<p>character, which of oxidation/reduction lines to plot</p>
</td></tr>
<tr><td><code id="util.plot_+3A_lty">lty</code></td>
<td>
<p>numeric, line type</p>
</td></tr>
<tr><td><code id="util.plot_+3A_plot.it">plot.it</code></td>
<td>
<p>logical, plot the lines?</p>
</td></tr>
<tr><td><code id="util.plot_+3A_main">main</code></td>
<td>
<p>character, text for plot title</p>
</td></tr>
<tr><td><code id="util.plot_+3A_spacing">spacing</code></td>
<td>
<p>numeric, spacing between multiple lines</p>
</td></tr>
<tr><td><code id="util.plot_+3A_z">z</code></td>
<td>
<p>numeric, set of values</p>
</td></tr>
<tr><td><code id="util.plot_+3A_alpha">alpha</code></td>
<td>
<p>character, hexadecimal value of color transparency (alpha)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>thermo.plot.new</code> sets parameters for a new plot, creates a new plot using <code><a href="graphics.html#topic+plot.new">plot.new</a></code>, and adds the axes tick marks to the plot.
Plot parameters (see <code><a href="graphics.html#topic+par">par</a></code>) including <code>cex</code>, <code>mar</code>, <code>lwd</code>, <code>mgp</code> and <code>axs</code> can be given, as well as a numeric vector in <code>side</code> identifying which sides of the plot receive tick marks.
<code>yline</code>, if present, denotes the margin line (default <code><a href="graphics.html#topic+par">par</a>('mgp')[1]</code>) where the y-axis name is plotted.
<code>thermo.axis</code> is the function that actually adds the axes, including inward-pointing major and minor tick marks (often used for thermodynamic property diagrams).
</p>
<p>Use <code>grid</code> to add a grid to the plot, corresponding to either the major ticks (solid lines), minor ticks (dashed lines), or both.
The grid can be made by adding <code>grid</code> argument to <code><a href="#topic+diagram">diagram</a></code>, or by calling <code>thermo.axis</code> after <code>diagram</code> (see example).
</p>
<p><code>water.lines</code> plots lines representing the oxidation and reduction stability limits of water on Eh/pe/log<i>f</i><sub>O<sub>2</sub></sub>/log<i>f</i><sub>H<sub>2</sub></sub> vs pH/<I>T</I>/<I>P</I> diagrams.
The x- and y-variables and their ranges are taken from <code>eout</code>.
Values of <I>T</I>, <I>P</I>, pH, and log<i>a</i><sub>H<sub>2</sub>O</sub>, not corresponding to either axis, are also taken from <code>eout</code>.
<code>which</code> controls which lines are drawn (&lsquo;<span class="samp">&#8288;oxidation&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;reduction&#8288;</span>&rsquo;, or both (the default)).
The value of <code>swapped</code> in the output reflects whether pH, <I>T</I>, or <I>P</I> is on the x-axis (TRUE) or y-axis (FALSE).
<code>NA</code> is returned for any diagram for variables that can not be processed (including diagrams with more than 2 variables).
</p>
<p><code>label.plot</code> and <code>label.figure</code> add identifying text within the plot region and figure region.
The value given for <code>x</code> is made into a label, optionally italicized and with parentheses (like (<i>a</i>)).
The location of the label is controlled by <code>xfrac</code> and <code>yfrac</code> (the fractional coordinates of either the plot or figure region), and <code>...</code> can include other parameters such as <code>cex</code> and <code>adj</code> that are passed to <code><a href="graphics.html#topic+text">text</a></code>.
</p>
<p><code>usrfig</code> returns the limits of the figure region in &ldquo;user&rdquo; coordinates (i.e. the limits of the plot region, from <code><a href="graphics.html#topic+par">par</a>("usr")</code>).
It is a supporting function for <code>label.figure</code> but is also useful for other circumstances where information must be added at a particular location in a figure.
</p>
<p><code>mtitle</code> can be used to add a multi-line title to a plot.
It loops over each element of <code>main</code> and places it on a separate margin line using <code><a href="graphics.html#topic+mtext">mtext</a></code>.
The spacing of the last (bottom) line from the edge of the plot is specified by <code>line</code>.
This function exists to facilitate using <code><a href="base.html#topic+expression">expression</a></code>s in multiline titles.
</p>
<p><code>ZC.col</code> uses <a href="https://CRAN.R-project.org/package=colorspace"><span class="pkg">colorspace</span></a>) to generate colors from a diverging palette (red - light grey - blue) corresponding to the values in <code>z</code>.
Red is associated with lower values of <code>z</code>.
This function is intended to generate colors for distinguishing average oxidation state of carbon <code><a href="#topic+ZC">ZC</a></code>, but any numeric values can be supplied.
</p>
<p><code>add.alpha</code> adds transparency to a color by appending the value of <code>alpha</code> to the hexadecimal representation of the color given in <code>col</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diagram">diagram</a></code> uses <code>thermo.plot.new</code> to set up a new plot, unless the argument <code>tplot</code> is set to FALSE in <code>diagram</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>basis(c("H2S", "H2O", "H+", "e-"))
species(c("HS-", "H2S", "HSO4-", "SO4-2"))
a &lt;- affinity(pH = c(0, 12), Eh = c(-1, 1), T = 200)
opar &lt;- par(mfrow = c(2, 2))
diagram(a, grid = "both")
title(main = 'diagram(a, grid = "both")')
diagram(a, grid = "major")
title(main = 'diagram(a, grid = "major")')
diagram(a, grid = "minor")
title(main = 'diagram(a, grid = "minor")')
diagram(a, fill = "terrain")
thermo.axis(grid = "major", col.grid = "slategray")
title(main = 'thermo.axis(grid = "major")')
par(thermo()$opar)
par(opar)
</code></pre>

<hr>
<h2 id='util.protein'>Functions for Proteins (Other Calculations)</h2><span id='topic+util.protein'></span><span id='topic+group.formulas'></span><span id='topic+MP90.cp'></span>

<h3>Description</h3>

<p>Return chemical formulas of groups in proteins, and calculate heat capacity using an additivity model from the literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  MP90.cp(protein, T)
  group.formulas()
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.protein_+3A_protein">protein</code></td>
<td>
<p>proteins specified in any format usable by <code><a href="#topic+pinfo">pinfo</a></code></p>
</td></tr>
<tr><td><code id="util.protein_+3A_t">T</code></td>
<td>
<p>numeric, temperature in &deg;C</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>group.formulas</code> returns the chemical formulas of each of the 20 common amino acid residues in proteins, as well as the terminal -H and -H (treated as the [H2O] group).
</p>
<p><code>MP90.cp</code> takes <code>protein</code> (name of protein) and <code>T</code> (one or more temperatures in &deg;C and returns the additive heat capacity (J mol <code class="reqn">^{-1}</code>) of the unfolded protein using values of heat capacities of the residues taken from Makhatadze and Privalov, 1990.
Those authors provided values of heat capacity at six points between 5 and 125 &deg;C; this function interpolates (using <code><a href="stats.html#topic+splinefun">splinefun</a></code>) values at other temperatures.
</p>


<h3>References</h3>

<p>Makhatadze, G. I. and Privalov, P. L. (1990) Heat capacity of proteins. 1. Partial molar heat capacity of individual amino acid residues in aqueous solution: Hydration effect <em>J. Mol. Biol.</em> <b>213</b>, 375&ndash;384. <a href="https://doi.org/10.1016/S0022-2836%2805%2980197-4">doi:10.1016/S0022-2836(05)80197-4</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ionize.aa">ionize.aa</a></code> for an example that compares <code>MP90.cp</code> with heat capacities calculated in CHNOSZ at different temperatures and pHs.
</p>

<hr>
<h2 id='util.seq'>Functions to Work with Sequence Data</h2><span id='topic+util.seq'></span><span id='topic+aminoacids'></span>

<h3>Description</h3>

<p>Return names or one- or three-letter abbreviations of amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  aminoacids(nchar = 1, which = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.seq_+3A_nchar">nchar</code></td>
<td>
<p>numeric, <code class="reqn">1</code> to return one-letter, <code class="reqn">3</code> to return three-letter abbreviations for amino acids</p>
</td></tr>
<tr><td><code id="util.seq_+3A_which">which</code></td>
<td>
<p>character, which amino acids to name</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>aminoacids</code> returns the one-letter abbreviations (<code>nchar</code>=&lsquo;<span class="samp">&#8288;1&#8288;</span>&rsquo;) or the three-letter abbreviations (<code>nchar</code>=&lsquo;<span class="samp">&#8288;3&#8288;</span>&rsquo;) or the names of the neutral amino acids (<code>nchar</code>=&lsquo;<span class="samp">&#8288;""&#8288;</span>&rsquo;) or the names of the amino acids with ionized side chains (<code>nchar</code>=&lsquo;<span class="samp">&#8288;"Z"&#8288;</span>&rsquo;).
The output includes 20 amino acids in alphabetic order by 1-letter abbreviation (the order used in <code>thermo()$protein</code>), unless <code>which</code> is provided, indicating the desired amino acids (either as 1- or 3-letter abbreviations or names of the neutral amino acids).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+count.aa">count.aa</a></code> for counting amino acids or nucleic-acid bases in a sequence; <code><a href="#topic+protein.formula">protein.formula</a></code> for calculating the chemical formulas of proteins.</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Count nucleobases in a sequence
bases &lt;- count.aa("ACCGGGTTT", type = "DNA")
</code></pre>

<hr>
<h2 id='util.units'>Functions to Convert Units</h2><span id='topic+util.units'></span><span id='topic+P.units'></span><span id='topic+T.units'></span><span id='topic+E.units'></span><span id='topic+convert'></span>

<h3>Description</h3>

<p>These functions convert values between units and set the user's preferred units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  P.units(units = NULL)
  T.units(units = NULL)
  E.units(units = NULL)
  convert(value, units, T = 298.15, P = 1, pH = 7, logaH2O = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.units_+3A_units">units</code></td>
<td>
<p>character, name of units to set or convert to/from</p>
</td></tr>
<tr><td><code id="util.units_+3A_value">value</code></td>
<td>
<p>numeric, value(s) to be converted</p>
</td></tr>
<tr><td><code id="util.units_+3A_t">T</code></td>
<td>
<p>numeric, temperature (Kelvin), used in &lsquo;<span class="samp">&#8288;G&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;logK&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;logfO2&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;E0&#8288;</span>&rsquo; conversions</p>
</td></tr>
<tr><td><code id="util.units_+3A_p">P</code></td>
<td>
<p>numeric, pressure (bar), used in &lsquo;<span class="samp">&#8288;logfO2&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;E0&#8288;</span>&rsquo; conversions</p>
</td></tr>
<tr><td><code id="util.units_+3A_ph">pH</code></td>
<td>
<p>numeric, pH, used in &lsquo;<span class="samp">&#8288;logfO2&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;E0&#8288;</span>&rsquo; conversions</p>
</td></tr>
<tr><td><code id="util.units_+3A_logah2o">logaH2O</code></td>
<td>
<p>numeric, logarithm of activity of water, used in &lsquo;<span class="samp">&#8288;logfO2&#8288;</span>&rsquo;-&lsquo;<span class="samp">&#8288;E0&#8288;</span>&rsquo; conversions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The units settings are used by <code><a href="#topic+subcrt">subcrt</a></code>, <code><a href="#topic+affinity">affinity</a></code>, and <code><a href="#topic+diagram">diagram</a></code> to accept input in or convert output to the units desired by the user.
The settings, which can be queried or changed with <code>T.units</code>, <code>E.units</code> and <code>P.units</code>, refer to the units of temperature (<code>C</code> or <code>K</code>), energy (<code>J</code> or <code>cal</code>), and pressure (<code>bar</code> or <code>MPa</code>).
(The first value in each of those pairs refers to the default units).
</p>
<p>The actual units conversions are handled by <code>convert</code>, through which <code>values</code> are transformed into destination <code>units</code> (names not case sensitive).
The possible conversions and settings for the <code>units</code> argument are shown in the following table.
Note that &lsquo;<span class="samp">&#8288;Eh&#8288;</span>&rsquo; and &lsquo;<span class="samp">&#8288;E0&#8288;</span>&rsquo; both stand for the value of Eh (oxidation-reduction potential in volts); they have different names so that one can choose to convert between Eh and either &lsquo;<span class="samp">&#8288;pe&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;logfO2&#8288;</span>&rsquo;. 
</p>

<table>
<tr>
 <td style="text-align: left;">
     property </td><td style="text-align: left;"> units </td><td style="text-align: left;"> setting of <code>units</code> argument </td>
</tr>
<tr>
 <td style="text-align: left;">
     temperature </td><td style="text-align: left;"> &deg;C, K </td><td style="text-align: left;"> <code>C</code>, <code>K</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     pressure </td><td style="text-align: left;"> bar, MPa </td><td style="text-align: left;"> <code>bar</code>, <code>MPa</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     energy </td><td style="text-align: left;"> cal, J </td><td style="text-align: left;"> <code>cal</code>, <code>J</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     energy </td><td style="text-align: left;"> J, cm<code class="reqn">^3</code> bar </td><td style="text-align: left;"> <code>joules</code>, <code>cm3bar</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     energy </td><td style="text-align: left;"> J, [none] </td><td style="text-align: left;"> <code>G</code>, <code>logK</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     oxidation potential </td><td style="text-align: left;"> volt, [none] </td><td style="text-align: left;"> <code>Eh</code>, <code>pe</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
     oxidation potential </td><td style="text-align: left;"> volt, [none] </td><td style="text-align: left;"> <code>E0</code>, <code>logfO2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>Another use of the function is to convert the results from <code><a href="#topic+solubility">solubility</a></code> into parts per billion, million, or thousand.
These destination units are specified by &lsquo;<span class="samp">&#8288;ppb&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;ppm&#8288;</span>&rsquo;, or &lsquo;<span class="samp">&#8288;ppt&#8288;</span>&rsquo;.
Additionally, the logarithms can be chosen with &lsquo;<span class="samp">&#8288;logppb&#8288;</span>&rsquo;, &lsquo;<span class="samp">&#8288;logppm&#8288;</span>&rsquo;, and &lsquo;<span class="samp">&#8288;logppt&#8288;</span>&rsquo;.
See <code><a href="utils.html#topic+demo">demo</a>("contour")</code> and <code><a href="utils.html#topic+demo">demo</a>("sphalerite")</code> for examples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Direct usage of convert
# Temperature (Kelvin) to degrees C
convert(273.15, "C")
# Temperature (degrees C) to Kelvin
convert(100, "K")
# Gibbs energy (J mol-1) to/from logK
convert(1000, "logK")		
convert(1000, "logK", T = 373.15)
convert(1, "G") 
# Eh (volt) to pe 
convert(-1, "pe")		
convert(-1, "pe", T = 373.15)
# logfO2 to E0 (volt)
convert(-80, "E0")             
convert(-80, "E0", pH = 5)
convert(-80, "E0", pH = 5, logaH2O = -5)
# Convert from calories to Joules
convert(1, "J")    # 1 cal = 4.184 J
# Convert from Joules to calories
convert(1, "cal")  # 1 J = 0.239 cal
# Convert cm3bar to Joules
convert(10, "joules") # 10 cm3.bar = 1 J

## Setting the units
# Make K the units for temperature arguments to subcrt() and affinity()
T.units("K") 
# Return to default - degrees C
T.units("C")
</code></pre>

<hr>
<h2 id='util.water'>Functions for Properties of Water and Steam</h2><span id='topic+util.water'></span><span id='topic+WP02.auxiliary'></span><span id='topic+rho.IAPWS95'></span><span id='topic+water.AW90'></span>

<h3>Description</h3>

<p>Utility functions for properties of water and steam.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  WP02.auxiliary(property, T = 298.15)
  rho.IAPWS95(T = 298.15, P = 1, state="", trace=0)
  water.AW90(T = 298.15, rho = 1000, P = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="util.water_+3A_property">property</code></td>
<td>
<p>character, property to calculate</p>
</td></tr>
<tr><td><code id="util.water_+3A_t">T</code></td>
<td>
<p>numeric, temperature (K)</p>
</td></tr>
<tr><td><code id="util.water_+3A_p">P</code></td>
<td>
<p>numeric, pressure (units of bar, except MPa for <code>water.AW90</code>)</p>
</td></tr>
<tr><td><code id="util.water_+3A_state">state</code></td>
<td>
<p>character, state or phase of H<sub>2</sub>O</p>
</td></tr>
<tr><td><code id="util.water_+3A_trace">trace</code></td>
<td>
<p>integer number</p>
</td></tr>
<tr><td><code id="util.water_+3A_rho">rho</code></td>
<td>
<p>numeric, density (kg m<code class="reqn">^{-3}</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Auxiliary equations to the IAPWS-95 formulation (Wagner and Pruß, 2002) are provided in <code>WP02.auxiliary</code>.
The <code>property</code> for this function can be one of &lsquo;<span class="samp">&#8288;P.sigma&#8288;</span>&rsquo; (saturation vapor pressure in MPa), &lsquo;<span class="samp">&#8288;dP.sigma.dT&#8288;</span>&rsquo; (derivative of saturation vapor pressure with respect to temperature), or &lsquo;<span class="samp">&#8288;rho.liquid&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;rho.vapor&#8288;</span>&rsquo; (density of liquid or vapor in kg m<code class="reqn">^{-3}</code>).
</p>
<p><code>rho.IAPWS95</code> implements a root-finding technique (using <code><a href="stats.html#topic+uniroot">uniroot</a></code>) to determine the values of density for the stable phase of H<sub>2</sub>O at the given temperature and pressure.
The <code>state</code> option is used internally in order to determine the stable phase at conditions close to saturation (0.9999*<i>P</i><sub>SAT</sub> &lt;= P &lt;= 1.00005*<i>P</i><sub>SAT</sub>, where <i>P</i><sub>SAT</sub> is the saturation pressure calculated by <code><a href="#topic+WP02.auxiliary">WP02.auxiliary</a></code>).
Alternatively, the user can specify a <code>state</code> of &lsquo;<span class="samp">&#8288;liquid&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;vapor&#8288;</span>&rsquo; to force the calculation of density for the corresponding phase, even if it is metastable (e.g. superheated water, supercooled steam; this option has no effect in the supercritical region).
The <code>state</code> is set in calls by <code><a href="#topic+water.IAPWS95">water.IAPWS95</a></code> to the value in <code>thermo()$opt$IAPWS.sat</code> (default &lsquo;<span class="samp">&#8288;liquid&#8288;</span>&rsquo;) so that higher-level functions (<code><a href="#topic+water">water</a></code>, <code><a href="#topic+subcrt">subcrt</a></code>) take properties for that state along the saturation curve.
Diagnostic messages are printed if <code>trace</code> is positive (it is also included in the call to <code>uniroot</code>).
</p>
<p><code>water.AW90</code> provides values of the static dielectric constant (<code>epsilon</code>) calculated using equations given by Archer and Wang, 1990.
</p>


<h3>References</h3>

<p>Archer, D. G. and Wang, P. M. (1990) The dielectric constant of water and Debye-Hückel limiting law slopes. <em>J. Phys. Chem. Ref. Data</em> <b>19</b>, 371&ndash;411. <a href="https://srd.nist.gov/JPCRD/jpcrd383.pdf">https://srd.nist.gov/JPCRD/jpcrd383.pdf</a>
</p>
<p>Wagner, W. and Pruß, A. (2002) The IAPWS formulation 1995 for the thermodynamic properties of ordinary water substance for general and scientific use. <em>J. Phys. Chem. Ref. Data</em> <b>31</b>, 387&ndash;535. <a href="https://doi.org/10.1063/1.1461829">doi:10.1063/1.1461829</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Calculate density of stable phase at 500 K, 500 bar
rho &lt;- rho.IAPWS95(T = 500, P = 500)
# Calculate pressure (= 50 MPa) at this density
IAPWS95("P", T = 500, rho = rho)
# Calculate dielectric constant 	
water.AW90(T = 500, rho = rho, P = 50)

# Density along saturation curve
T &lt;- seq(273.15, 623.15, 25)
WP02.auxiliary(T = T)  # liquid from WP02
WP02.auxiliary("rho.vapor", T)  # vapor from WP02	

# WP02.auxiliary gives a close estimate of saturation pressure...
T &lt;- 445:455
P.sigma &lt;- WP02.auxiliary("P.sigma", T)
# ... but alternates between being just on the liquid or vapor side
# (low rho: steam; high rho: water)
rho.IAPWS95(T, convert(P.sigma, "bar"))
# thermo()$opt$IAPWS.sat allows for choosing liquid or vapor or ""
thermo("opt$IAPWS.sat" = "")
# Shows artifactual vapor-liquid transition
water.IAPWS95("V", T, "Psat")
# The calculated Psat, while not exact, should be close enough for most
# geochemical calculations to select liquid or vapor
oldwat &lt;- water("IAPWS95")
thermo("opt$IAPWS.sat" = "vapor")
V.vapor &lt;- subcrt("water", T=convert(445:455, "C"))$out[[1]]$V
thermo("opt$IAPWS.sat" = "liquid") # the default
V.liquid &lt;- subcrt("water", T=convert(445:455, "C"))$out[[1]]$V
all(V.vapor &gt; V.liquid)  # TRUE
water(oldwat)
</code></pre>

<hr>
<h2 id='water'>Properties of Water</h2><span id='topic+water'></span><span id='topic+water.SUPCRT92'></span><span id='topic+water.IAPWS95'></span><span id='topic+water.DEW'></span>

<h3>Description</h3>

<p>Calculate thermodynamic and electrostatic properties of water.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  water(property = NULL, T = 298.15, P = "Psat", P1 = TRUE)
  water.SUPCRT92(property=NULL, T = 298.15, P = 1, P1 = TRUE)
  water.IAPWS95(property=NULL, T = 298.15, P = 1)
  water.DEW(property=NULL, T = 373.15, P = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="water_+3A_property">property</code></td>
<td>
<p>character, computational setting or property(s) to calculate</p>
</td></tr>
<tr><td><code id="water_+3A_t">T</code></td>
<td>
<p>numeric, temperature (K)</p>
</td></tr>
<tr><td><code id="water_+3A_p">P</code></td>
<td>
<p>numeric, pressure (bar), or &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; for vapor-liquid saturation</p>
</td></tr>
<tr><td><code id="water_+3A_p1">P1</code></td>
<td>
<p>logical, output pressure of 1 bar below 100 &deg;C instead of calculated values of &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo;?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions compute the thermodynamic (Gibbs energy and it derivatives) and electrostatic (dielectric constant and its derivatives) properties of liquid or supercritical H<sub>2</sub>O as a function of temperature and pressure using equations of state taken from the literature.
The high-level function <code>water</code> performs different computations, depending on the setting of <code><a href="#topic+thermo">thermo</a>()$opt$water</code>:
</p>

<dl>
<dt>&lsquo;<span class="samp">&#8288;SUPCRT92&#8288;</span>&rsquo; (default) or &lsquo;<span class="samp">&#8288;SUPCRT&#8288;</span>&rsquo;</dt><dd><p>Thermodynamic and electrostatic properties are calculated using a FORTRAN subroutine taken from the <abbr><span class="acronym">SUPCRT92</span></abbr> software package (Johnson et al., 1992). See more information below.</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;IAPWS95&#8288;</span>&rsquo; or &lsquo;<span class="samp">&#8288;IAPWS&#8288;</span>&rsquo;</dt><dd><p>Thermodynamic properties are calculated using an implementation in <span class="rlang"><b>R</b></span> code of the <abbr><span class="acronym">IAPWS-95</span></abbr> formulation (Wagner and Pruss, 2002), and electrostatic properties are calculated using the equations of Archer and Wang, 1990. See <code><a href="#topic+IAPWS95">IAPWS95</a></code> and more information below.</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;DEW&#8288;</span>&rsquo;</dt><dd><p>Thermodynamic and electrostatic properties are calculated using the Deep Earth Water (<abbr><span class="acronym">DEW</span></abbr>) model (Sverjensky et al., 2014). The defaults for <code>T</code> and <code>P</code> reflect the minimum values for applicability of the model; calculations at lower <code>T</code> and/or <code>P</code> points fall back to using &lsquo;<span class="samp">&#8288;SUPCRT92&#8288;</span>&rsquo;. See <code><a href="#topic+DEW">DEW</a></code>.</p>
</dd>
</dl>

<p>Calling the function with no arguments returns the current computational setting.
Use e.g. <code>water("DEW")</code> to make the setting; the previous setting (at the time of the function call) is returned invisibly.
Subsequent calculations with <code>water</code>, or other functions such as <code>subcrt</code> and <code>affinity</code>, will use that setting.
</p>
<p>The allowed <code>property</code>s for <code>water</code> are one or more of those given below, depending on the computational setting; availability is shown by an asterisk.
Note that some of the properties that can actually be calculated using the different formulations are not implemented here.
Except for <code>rho</code>, the units are those used by Johnson and Norton, 1991.
</p>

<table>
<tr>
 <td style="text-align: left;">
     Property </td><td style="text-align: left;"> Description </td><td style="text-align: left;"> Units </td><td style="text-align: left;"> IAPWS95 </td><td style="text-align: left;"> SUPCRT92 </td><td style="text-align: left;"> DEW </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>A</code> </td><td style="text-align: left;"> Helmholtz energy </td><td style="text-align: left;"> J mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>G</code> </td><td style="text-align: left;"> Gibbs energy </td><td style="text-align: left;"> J mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>S</code> </td><td style="text-align: left;"> Entropy </td><td style="text-align: left;"> J K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>U</code> </td><td style="text-align: left;"> Internal energy </td><td style="text-align: left;"> J mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>H</code> </td><td style="text-align: left;"> Enthalpy </td><td style="text-align: left;"> J mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>Cv</code> </td><td style="text-align: left;"> Isochoric heat capacity </td><td style="text-align: left;"> J K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>Cp</code> </td><td style="text-align: left;"> Isobaric heat capacity </td><td style="text-align: left;"> J K<code class="reqn">^{-1}</code> mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>Speed</code> </td><td style="text-align: left;"> Speed of sound </td><td style="text-align: left;"> cm s<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>alpha</code> </td><td style="text-align: left;"> Coefficient of isobaric expansivity </td><td style="text-align: left;"> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>beta</code> </td><td style="text-align: left;"> Coefficient of isothermal compressibility </td><td style="text-align: left;"> bar<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>epsilon</code> </td><td style="text-align: left;"> Dielectric constant </td><td style="text-align: left;"> dimensionless </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>visc</code> </td><td style="text-align: left;"> Dynamic viscosity </td><td style="text-align: left;"> g cm<code class="reqn">^{-1}</code> s<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>tcond</code> </td><td style="text-align: left;"> Thermal conductivity </td><td style="text-align: left;"> J cm<code class="reqn">^{-1}</code> s<code class="reqn">^{-1}</code> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>tdiff</code> </td><td style="text-align: left;"> Thermal diffusivity </td><td style="text-align: left;"> cm<code class="reqn">^2</code> s<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>Prndtl</code> </td><td style="text-align: left;"> Prandtl number </td><td style="text-align: left;"> dimensionless </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>visck</code> </td><td style="text-align: left;"> Kinematic viscosity </td><td style="text-align: left;"> cm<code class="reqn">^2</code> s<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>albe</code> </td><td style="text-align: left;"> Isochoric expansivity </td><td style="text-align: left;"> bar K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     </td><td style="text-align: left;"> -compressibility </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>ZBorn</code> </td><td style="text-align: left;"> Z Born function </td><td style="text-align: left;"> dimensionless </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>YBorn</code> </td><td style="text-align: left;"> Y Born function </td><td style="text-align: left;"> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>QBorn</code> </td><td style="text-align: left;"> Q Born function </td><td style="text-align: left;"> bar<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>daldT</code> </td><td style="text-align: left;"> Isobaric temperature derivative </td><td style="text-align: left;"> K<code class="reqn">^{-2}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     </td><td style="text-align: left;"> of expansibility </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>XBorn</code> </td><td style="text-align: left;"> X Born function </td><td style="text-align: left;"> K<code class="reqn">^{-2}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>NBorn</code> </td><td style="text-align: left;"> N Born function </td><td style="text-align: left;"> bar<code class="reqn">^{-2}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>UBorn</code> </td><td style="text-align: left;"> U Born function </td><td style="text-align: left;"> bar<code class="reqn">^{-1}</code> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>V</code> </td><td style="text-align: left;"> Volume </td><td style="text-align: left;"> cm<code class="reqn">^3</code> mol<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>rho</code> </td><td style="text-align: left;"> Density </td><td style="text-align: left;"> kg m<code class="reqn">^3</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">     
     <code>Psat</code> </td><td style="text-align: left;"> Saturation vapor pressure </td><td style="text-align: left;"> bar </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>E</code> </td><td style="text-align: left;"> Isobaric expansivity </td><td style="text-align: left;"> cm<code class="reqn">^3</code> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>kT</code> </td><td style="text-align: left;"> Isothermal compressibility </td><td style="text-align: left;"> cm<code class="reqn">^3</code> bar<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>de.dT</code> </td><td style="text-align: left;"> Temperature derivative </td><td style="text-align: left;"> K<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     </td><td style="text-align: left;"> of dielectric constant </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>de.dP</code> </td><td style="text-align: left;"> Pressure derivative </td><td style="text-align: left;"> bar<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     </td><td style="text-align: left;"> of dielectric constant </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>P</code> </td><td style="text-align: left;"> Pressure </td><td style="text-align: left;"> bar </td><td style="text-align: left;"> * </td><td style="text-align: left;"> NA </td><td style="text-align: left;"> NA </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>A_DH</code> </td><td style="text-align: left;"> A Debye-Huckel parameter </td><td style="text-align: left;"> kg<code class="reqn">^{0.5}</code> mol<code class="reqn">^{-0.5}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
     <code>B_DH</code> </td><td style="text-align: left;"> B Debye-Huckel parameter </td><td style="text-align: left;"> kg<code class="reqn">^{0.5}</code> mol<code class="reqn">^{-0.5}</code> cm<code class="reqn">^{-1}</code> </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td><td style="text-align: left;"> * </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>Call <code>water.SUPCRT92</code>, <code>water.IAPWS95</code>, or <code>water.DEW</code> with no arguments to list the available properties.
</p>
<p><code>water.SUPCRT92</code> interfaces to the FORTRAN subroutine taken from the <abbr><span class="acronym">SUPCRT92</span></abbr> package (H2O92D.F) for calculating properties of water.
These calculations are based on data and equations of Levelt-Sengers et al., 1983, Haar et al., 1984, and Johnson and Norton, 1991, among others (see Johnson et al., 1992).
A value of <code>P</code> set to &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; refers to one bar below 100 &deg;C, otherwise to the vapor-liquid saturation pressure at temperatures below the critical point (&lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo; is not available at temperatures above the critical point).
<code>water.SUPCRT92</code> provides a limited interface to the FORTRAN subroutine; some functions provided there are not made available here (e.g., using variable density instead of pressure, or calculating the properties of steam).
</p>
<p>The stated temperature limits of validity of calculations in <code>water.SUPCRT92</code> are from the greater of 0 &deg;C or the melting temperature at pressure, to 2250 &deg;C (Johnson et al., 1992).
Valid pressures are from the greater of zero bar or the melting pressure at temperature to 30000 bar.
The present functions do not check these limits and will attempt calculations for any range of input parameters, but may return <code>NA</code> for properties that fail to be calculated at given temperatures and pressures and/or produce warnings or even errors when problems are encountered.
</p>
<p>Starting with version 0.9-9.4, a check for minimum pressure (in valTP function in H2O92D.f) has been bypassed so that properties of H2O can be calculated using <code>water.SUPCRT92</code> at temperatures below the 0.01 &deg;C triple point.
A primary check is still enforced (Tbtm), giving a minimum valid temperature of 253.15 K.
</p>
<p><code>water.IAPWS95</code> is a wrapper around <code><a href="#topic+IAPWS95">IAPWS95</a></code>, <code><a href="#topic+rho.IAPWS95">rho.IAPWS95</a></code> and <code><a href="#topic+water.AW90">water.AW90</a></code>.
<code>water.IAPWS95</code> provides for calculations at specific temperature and pressure; density, needed for <code>IAPWS95</code>, is inverted from pressure using <code>rho.IAPWS95</code>.
The function also contains routines for calculating the Born functions as numerical derivatives of the static dielectric constant (from <code>water.AW90</code>).
For compatibility with geochemical modeling conventions, the values of Gibbs energy, enthalpy and entropy output by <code>IAPWS95</code> are converted by <code>water.IAPWS95</code> to the triple point reference state adopted in <code>SUPCRT92</code> (Johnson and Norton, 1991; Helgeson and Kirkham, 1974).
<code>water.IAPWS95</code> also accepts setting <code>P</code> to &lsquo;<span class="samp">&#8288;Psat&#8288;</span>&rsquo;, with the saturation pressure calculated from <code><a href="#topic+WP02.auxiliary">WP02.auxiliary</a></code>; by default the returned properties are for the liquid, but this can be changed to the vapor in <code><a href="#topic+thermo">thermo</a>()$opt$IAPWS.sat</code>.
</p>
<p><code>A_DH</code> and <code>B_DH</code> are solvent parameters in the &ldquo;B-dot&rdquo; (extended Debye-Huckel) equation (Helgeson, 1969; Manning, 2013).
</p>


<h3>Value</h3>

<p>A data frame, the number of rows of which corresponds to the number of input temperature-pressure pairs.
</p>


<h3>References</h3>

<p>Archer, D. G. and Wang, P. M. (1990) The dielectric constant of water and Debye-Hückel limiting law slopes. <em>J. Phys. Chem. Ref. Data</em> <b>19</b>, 371&ndash;411. <a href="https://doi.org/10.1063/1.555853">doi:10.1063/1.555853</a>
</p>
<p>Haar, L., Gallagher, J. S. and Kell, G. S. (1984) <em>NBS/NRC Steam Tables</em>. Hemisphere, Washington, D. C., 320 p. <a href="https://www.worldcat.org/oclc/301304139">https://www.worldcat.org/oclc/301304139</a>
</p>
<p>Helgeson, H. C. and Kirkham, D. H. (1974) Theoretical prediction of the thermodynamic behavior of aqueous electrolytes at high pressures and temperatures. I. Summary of the thermodynamic/electrostatic properties of the solvent. <em>Am. J. Sci.</em> <b>274</b>, 1089&ndash;1098. <a href="https://doi.org/10.2475/ajs.274.10.1089">doi:10.2475/ajs.274.10.1089</a>
</p>
<p>Helgeson, H. C. (1969) Thermodynamics of hydrothermal systems at elevated temperatures and pressures. <em>Am. J. Sci.</em> <b>267</b>, 729&ndash;804. <a href="https://doi.org/10.2475/ajs.267.7.729">doi:10.2475/ajs.267.7.729</a>
</p>
<p>Johnson, J. W. and Norton, D. (1991) Critical phenomena in hydrothermal systems: state, thermodynamic, electrostatic, and transport properties of H<code class="reqn">_2</code>O in the critical region. <em>Am. J. Sci.</em> <b>291</b>, 541&ndash;648. <a href="https://doi.org/10.2475/ajs.291.6.541">doi:10.2475/ajs.291.6.541</a>
</p>
<p>Johnson, J. W., Oelkers, E. H. and Helgeson, H. C. (1992) SUPCRT92: A software package for calculating the standard molal thermodynamic properties of minerals, gases, aqueous species, and reactions from 1 to 5000 bar and 0 to 1000&deg;C. <em>Comp. Geosci.</em> <b>18</b>, 899&ndash;947. <a href="https://doi.org/10.1016/0098-3004%2892%2990029-Q">doi:10.1016/0098-3004(92)90029-Q</a>
</p>
<p>Levelt-Sengers, J. M. H., Kamgarparsi, B., Balfour, F. W. and Sengers, J. V. (1983) Thermodynamic properties of steam in the critical region. <em>J. Phys. Chem. Ref. Data</em> <b>12</b>, 1&ndash;28. <a href="https://doi.org/10.1063/1.555676">doi:10.1063/1.555676</a>
</p>
<p>Manning, C. E. (2013) Thermodynamic modeling of fluid-rock interaction at mid-crustal to upper-mantle conditions. <em>Rev. Mineral. Geochem.</em> <b>76</b>, 135&ndash;164. <a href="https://doi.org/10.2138/rmg.2013.76.5">doi:10.2138/rmg.2013.76.5</a>
</p>
<p>Sverjensky, D. A., Harrison, B. and Azzolini, D. (2014) Water in the deep Earth: The dielectric constant and the solubilities of quartz and corundum to 60 kb and 1,200 &deg;C. <em>Geochim. Cosmochim. Acta</em> <b>129</b>, 125&ndash;145. <a href="https://doi.org/10.1016/j.gca.2013.12.019">doi:10.1016/j.gca.2013.12.019</a>
</p>
<p>Wagner, W. and Pruss, A. (2002) The IAPWS formulation 1995 for the thermodynamic properties of ordinary water substance for general and scientific use. <em>J. Phys. Chem. Ref. Data</em> <b>31</b>, 387&ndash;535. <a href="https://doi.org/10.1063/1.1461829">doi:10.1063/1.1461829</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Calculations along saturation curve
T &lt;- seq(273.15, 623.15, 25)
# Liquid density, from SUPCRT92
water("rho", T = T, P = "Psat")
# Values of the saturation pressure, Gibbs energy	
water(c("Psat", "G"), T = T, P = "Psat") 
# Derivatives of the dielectric constant (Born functions)
water(c("QBorn", "YBorn", "XBorn"), T = T, P = "Psat")
# Now at constant pressure
water(c("QBorn", "YBorn", "XBorn"), T = T, P = 2000)

## Comparing the formulations
T &lt;- convert(c(25, 100, 200, 300), "K")
# IAPWS-95
oldwat &lt;- water("IAPWS95")
water(water.IAPWS95(), T = T)
# Deep Earth Water (DEW)
water("DEW")
water(water.DEW(), T = T, P = 1000)
# SUPCRT92 (the default)
water(oldwat)
water(water.SUPCRT92(), T = T)

## Calculating Q Born function
# After Table 22 of Johnson and Norton, 1991
T &lt;- rep(c(375, 400, 425, 450, 475), each = 5)
P &lt;- rep(c(250, 300, 350, 400, 450), 5)
w &lt;- water("QBorn", T = convert(T, "K"), P = P)
# The rest is to make a neat table
w &lt;- as.data.frame(matrix(w[[1]], nrow = 5))
colnames(w) &lt;- T[1:5*5]
rownames(w) &lt;- P[1:5]
print(w)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
