<!DOCTYPE html><html lang="en"><head><title>Help for package chromoMap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {chromoMap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chromoMap'><p>render interactive chromosome plots of any living organism and annotate elements</p></a></li>
<li><a href='#chromoMap-shiny'><p>Shiny bindings for chromoMap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Genomic Visualization of Biological Data</td>
</tr>
<tr>
<td>Version:</td>
<td>4.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lakshay Anand &lt;lakshayanand15@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides interactive, configurable and elegant graphics visualization of the chromosomes or chromosome regions
    of any living organism allowing users to map chromosome elements (like genes, SNPs etc.) on the chromosome plot. It introduces
    a special plot viz. the "chromosome heatmap" that, in addition to mapping elements, can visualize the data
    associated with chromosome elements (like gene expression) in the form of heat colors which can be highly
    advantageous in the scientific interpretations and research work. Because of the large size of the chromosomes,
    it is impractical to visualize each element on the same plot. However, the plot provides a magnified view for each
    of chromosome locus to render additional information and visualization specific for that location. You can map
    thousands of genes and can view all mappings easily. Users can investigate the detailed information about the mappings
    (like gene names or total genes mapped on a location) or can view the magnified single or double stranded view of the
    chromosome at a location showing each mapped element in sequential order. The package provide multiple features
    like visualizing multiple sets, chromosome heat-maps, group annotations, adding hyperlinks, and labelling.
    The plots can be saved as HTML documents that can be customized and shared easily. In addition, you can include them in R Markdown or in R 'Shiny' applications.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>htmltools (&ge; 0.3.6), htmlwidgets (&ge; 1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Lakshay Anand [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-16 01:29:34 UTC; lakshay-anand</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-16 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='chromoMap'>render interactive chromosome plots of any living organism and annotate elements</h2><span id='topic+chromoMap'></span>

<h3>Description</h3>

<p>render an interactive graphics visualization of entire chromosomes
or chromosomal regions of any living organism. Chromosomal elements such as genes 
can be annotated easily using this tool.
</p>
<p>required for creating widgets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromoMap(
  ch.files,
  data.files,
  title = c(),
  ch_gap = 5,
  ploidy = 1,
  top_margin = 25,
  left_margin = 50,
  chr_width = 15,
  chr_length = 4,
  chr_color = c("black"),
  data_based_color_map = FALSE,
  segment_annotation = FALSE,
  lg_x = 0,
  lg_y = 0,
  data_type = c("numeric", "categorical"),
  labels = FALSE,
  canvas_width = NULL,
  canvas_height = NULL,
  data_colors = list(),
  anno_col = c("#10B85F"),
  chr_text = c(TRUE),
  discrete.domain = NULL,
  legend = c(FALSE),
  hlinks = FALSE,
  aggregate_func = c("avg"),
  plots = c("none"),
  tag_filter = list(c("none", 0)),
  plot_height = c(30),
  plot_ticks = c(4),
  plot_color = c("blue"),
  plot_y_domain = list(c(0, 0)),
  ch2D.colors = NULL,
  ch2D.cat.order = NULL,
  ch2D.lg_x = 0,
  ch2D.lg_y = 0,
  ref_line = c(FALSE),
  refl_pos = c(0),
  refl_color = c("grey"),
  refl_stroke_w = c(2),
  tagColor = c("red"),
  heat_map = c(TRUE),
  text_font_size = c(10),
  chr_curve = 5,
  title_font_size = 12,
  label_font = 9,
  label_angle = -90,
  vertical_grid = FALSE,
  grid_array = c(0, 5000, 10000),
  grid_color = "grey",
  grid_text = NULL,
  grid_text_size = 12,
  grid_text_y = 20,
  plot_filter = list(c("none", 0)),
  id = c("chromap"),
  region = NULL,
  show.links = FALSE,
  loci_links = "none",
  directed.edges = F,
  y_chr_scale = 0,
  links.colors = NULL,
  links.lg_x = 0,
  links.lg_y = 0,
  n_win.factor = 1,
  chr.scale.ticks = 5,
  export.options = F,
  fixed.window = F,
  window.size = NULL,
  win.summary.display = F,
  remove.last.window = T,
  guides = F,
  guides_color = "lightgrey",
  ann.h = 1,
  chr.2D.plot = F,
  display.chr = T,
  plot.shift = c(1),
  plot.legend.labels = c(""),
  cat.legend.label = "",
  plot.y.labels = c(""),
  plot.y.lab.x = 10,
  plot.y.lab.y = 0,
  plot.y.lab.size = 15,
  scale.suffix = "bp",
  numeric.domain = NULL,
  interactivity = T
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chromoMap_+3A_ch.files">ch.files</code></td>
<td>
<p>filename(s) as character vector OR list of data.frames containing co-ordinates of the chromosomes to render</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_data.files">data.files</code></td>
<td>
<p>filename(s) as character vector OR list of data.frames containing data to annotate on the chromosomes.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_title">title</code></td>
<td>
<p>a character string to be used as a title in plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ch_gap">ch_gap</code></td>
<td>
<p>provide spacing between chromosomes.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ploidy">ploidy</code></td>
<td>
<p>specify the number of sets of chromsomes being passed.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_top_margin">top_margin</code></td>
<td>
<p>specify the margin from top of the plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_left_margin">left_margin</code></td>
<td>
<p>specify the margin from the left of the plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr_width">chr_width</code></td>
<td>
<p>specify the width of each chromsome</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr_length">chr_length</code></td>
<td>
<p>specify the length of each chromsome.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr_color">chr_color</code></td>
<td>
<p>a vector specifying the color of each chromsome in a set. A color
can be assigned to each set by passing a different color values as vector</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_data_based_color_map">data_based_color_map</code></td>
<td>
<p>a boolean to tell the plot to use the data provided in file for
visualizing annotation</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_segment_annotation">segment_annotation</code></td>
<td>
<p>a boolean to use segment-annotation algorithm</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_lg_x">lg_x</code></td>
<td>
<p>specify the x or horizontal distance of the legend from origin(bottom right corner)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_lg_y">lg_y</code></td>
<td>
<p>specify the y or vertical distnce of the legend from the origin</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_data_type">data_type</code></td>
<td>
<p>specifying the data type of the data used. takes value either 'categorical' or
'numeric'</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_labels">labels</code></td>
<td>
<p>a boolean to include labels in plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_canvas_width">canvas_width</code></td>
<td>
<p>width of the plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_canvas_height">canvas_height</code></td>
<td>
<p>height of the plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_data_colors">data_colors</code></td>
<td>
<p>specify annotation colors for the data</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_anno_col">anno_col</code></td>
<td>
<p>a vector to specify annotation color for each set.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr_text">chr_text</code></td>
<td>
<p>a boolean vector to enable or disable chromsome texts for each ploidy.set</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_discrete.domain">discrete.domain</code></td>
<td>
<p>manually specify the order of categories.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_legend">legend</code></td>
<td>
<p>a boolean vector to enable or disable legend for each set/ploidy</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_hlinks">hlinks</code></td>
<td>
<p>a boolean to use hyperlinks supplied in data</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_aggregate_func">aggregate_func</code></td>
<td>
<p>takes either 'sum' or 'avg' to specift aggregate function for each loci</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plots">plots</code></td>
<td>
<p>specify the type of plot to visualize. takes either 'scatter' , 'bar' or 'tags'.(default: 'none')</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_tag_filter">tag_filter</code></td>
<td>
<p>a list to specify the filter operation and operands for each ploidy.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot_height">plot_height</code></td>
<td>
<p>specify plot height for each ploidy. default: c(30)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot_ticks">plot_ticks</code></td>
<td>
<p>specify number of ticks for plot axis. default: c(4)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot_color">plot_color</code></td>
<td>
<p>specify the plot color for each ploidy. default: c(&quot;blue&quot;)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot_y_domain">plot_y_domain</code></td>
<td>
<p>specify plot y-axis domain. default: list(c(0,0))</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ch2d.colors">ch2D.colors</code></td>
<td>
<p>specify the group colors for visualizing categories on 2D chromosome plots</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ch2d.cat.order">ch2D.cat.order</code></td>
<td>
<p>manually setting the order of categories for 2D-Chromsome plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ch2d.lg_x">ch2D.lg_x</code></td>
<td>
<p>specify the x or horizontal distance of 2D plot legend from the origin(bottom right corner)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ch2d.lg_y">ch2D.lg_y</code></td>
<td>
<p>specify the y or vertical distance of 2D plot legend</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ref_line">ref_line</code></td>
<td>
<p>a boolean to use horizontal reference line in plot. default: c(FALSE)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_refl_pos">refl_pos</code></td>
<td>
<p>specify the position of reference line. default: c(0)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_refl_color">refl_color</code></td>
<td>
<p>specify the color of the reference line. default: c(&quot;grey&quot;)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_refl_stroke_w">refl_stroke_w</code></td>
<td>
<p>specify the stroke width of the reference line. default: c(2)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_tagcolor">tagColor</code></td>
<td>
<p>specify the color of tags. default: c(&quot;red&quot;)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_heat_map">heat_map</code></td>
<td>
<p>a boolean to use if chromosome heatmaps are shown. default: c(TRUE),</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_text_font_size">text_font_size</code></td>
<td>
<p>specify chromosome text font-size. default: c(10)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr_curve">chr_curve</code></td>
<td>
<p>specify the chromosome curves at the telomeres or centromere loci. default:5</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_title_font_size">title_font_size</code></td>
<td>
<p>specify the font-size of the title. default:12</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_label_font">label_font</code></td>
<td>
<p>specify the font-size of the labels. default:9</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_label_angle">label_angle</code></td>
<td>
<p>specify the angle of rotation of labels. default: -90</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_vertical_grid">vertical_grid</code></td>
<td>
<p>a boolean to use vertical grid lines. default: FALSE</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_grid_array">grid_array</code></td>
<td>
<p>specify the position(s) of grid line(s) in bp to highlight locations across genome. default: c(0,5000,10000)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_grid_color">grid_color</code></td>
<td>
<p>specify the color of the grid lines. default: &quot;grey&quot;</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_grid_text">grid_text</code></td>
<td>
<p>specify the text to be attached at the top end of gridlines</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_grid_text_size">grid_text_size</code></td>
<td>
<p>specify the font-size of the text</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_grid_text_y">grid_text_y</code></td>
<td>
<p>specify the y-distance (from top) for the text</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot_filter">plot_filter</code></td>
<td>
<p>a list specify the plot filter operation, operands, and filter-color for each ploidy.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_id">id</code></td>
<td>
<p>specify a unique id doe chromoMap plot. default: c(&quot;chromap&quot;)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_region">region</code></td>
<td>
<p>specify the region of interest for chromosome(s) for zoom-in. Format: &quot;chrName:Ploidy:Start:Stop&quot;</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_show.links">show.links</code></td>
<td>
<p>a boolean to specify whether links are visualized. default: FALSE</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_loci_links">loci_links</code></td>
<td>
<p>a character vector specifying file name or a data.frame for links input data</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_directed.edges">directed.edges</code></td>
<td>
<p>a boolean to visualize directed edges</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_y_chr_scale">y_chr_scale</code></td>
<td>
<p>adjust the chromosome scale along y-axis</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_links.colors">links.colors</code></td>
<td>
<p>specify the links colors</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_links.lg_x">links.lg_x</code></td>
<td>
<p>specify x or horizontal distance of links legend from the origin</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_links.lg_y">links.lg_y</code></td>
<td>
<p>specify y or vertical distance of links</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_n_win.factor">n_win.factor</code></td>
<td>
<p>specify the factor by which the chr will be scaled;increases number of windows (default:1)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr.scale.ticks">chr.scale.ticks</code></td>
<td>
<p>specify the number of ticks for chr scale (default:5)</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_export.options">export.options</code></td>
<td>
<p>boolean to include export buttons in the plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_fixed.window">fixed.window</code></td>
<td>
<p>Boolean to specify wether to use fixed window visualization</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_window.size">window.size</code></td>
<td>
<p>specify the window size, if fixed.window is TRUE</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_win.summary.display">win.summary.display</code></td>
<td>
<p>boolean to display window summary to console</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_remove.last.window">remove.last.window</code></td>
<td>
<p>For fixed window analysis, boolean to specify whether to include last window of chromosomes</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_guides">guides</code></td>
<td>
<p>boolean to display guides</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_guides_color">guides_color</code></td>
<td>
<p>set guides color.</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_ann.h">ann.h</code></td>
<td>
<p>set annotation bar height in 2D-Chromosome plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_chr.2d.plot">chr.2D.plot</code></td>
<td>
<p>boolean to specify visualize 2d Chromosome plot</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_display.chr">display.chr</code></td>
<td>
<p>boolean to show.hide chromosome</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.shift">plot.shift</code></td>
<td>
<p>shifting the plots in y direction in case hiding chromosomes</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.legend.labels">plot.legend.labels</code></td>
<td>
<p>specify plot legend labels</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_cat.legend.label">cat.legend.label</code></td>
<td>
<p>specify categorical-data legends label</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.y.labels">plot.y.labels</code></td>
<td>
<p>specify plots y-axis labels</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.y.lab.x">plot.y.lab.x</code></td>
<td>
<p>adjust plot y labels in x-direction</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.y.lab.y">plot.y.lab.y</code></td>
<td>
<p>adjust plot y labels in y-direction</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_plot.y.lab.size">plot.y.lab.size</code></td>
<td>
<p>set size of plot y labels</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_scale.suffix">scale.suffix</code></td>
<td>
<p>set the suffix for chromosome scale(default:'bp')</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_numeric.domain">numeric.domain</code></td>
<td>
<p>manually set data domain(min,max) for heat colors for numeric data</p>
</td></tr>
<tr><td><code id="chromoMap_+3A_interactivity">interactivity</code></td>
<td>
<p>boolean to enable/disable interactivity on chromosomes</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

library(chromoMap)

#simple annotations
chromoMap("chromosome_file.txt","annotation_file.txt")

#polyploidy example
chromoMap(c("chromosome_set1.txt","chromosome_set2.txt"),
          c("annotation_set1.txt","annotation_set2.txt"), ploidy=2)

#plotting group annotation
chromoMap("chromosome_file.txt","annotation_file.txt",
           data_base_color_map=T, data_type="categorical")

#plotting chromsome heatmaps
chromoMap("chromosome_file.txt","annotation_file.txt",
           data_based_color_map=T, data_type="numeric")

#enabling hyperlinks
chromoMap("chromosome_file.txt","annotation_file.txt", hlinks=T)

#enabling labels
chromoMap("chromosome_file.txt","annotation_file.txt", labels=T)

#change chromosome color
chromoMap("chromosome_file.txt","annotation_file.txt", chr_color="red")


## End(Not run)





</code></pre>

<hr>
<h2 id='chromoMap-shiny'>Shiny bindings for chromoMap</h2><span id='topic+chromoMap-shiny'></span><span id='topic+chromoMapOutput'></span><span id='topic+renderChromoMap'></span>

<h3>Description</h3>

<p>Output and render functions for using chromoMap within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromoMapOutput(outputId, width = "100%", height = "400px")

renderChromoMap(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chromoMap-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="chromoMap-shiny_+3A_width">width</code>, <code id="chromoMap-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="chromoMap-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a chromoMap</p>
</td></tr>
<tr><td><code id="chromoMap-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="chromoMap-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
