<!DOCTYPE html><html lang="en"><head><title>Help for package chromseq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {chromseq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#id'><p>Sampled Fasta file of chromosome sequence from hg19 blacklist</p></a></li>
<li><a href='#readToList'><p>Make a list file from large chromosome Fasta file</p></a></li>
<li><a href='#replaceText'><p>Replace tedious chromosome identifier into simple format</p></a></li>
<li><a href='#sortList'><p>Sort the chromosome list  according to the chromosome number</p></a></li>
<li><a href='#splitChr'><p>Split all chromosomes from the sorted chromosome list</p></a></li>
<li><a href='#subFasID'><p>Subtract chromosome ids from Fasta file</p></a></li>
<li><a href='#tex'><p>Fasta file of chromosome sequence produced from sequence character</p></a></li>
<li><a href='#text'><p>Fasta file of chromosome sequence</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Split Chromosome 'Fasta' File</td>
</tr>
<tr>
<td>Description:</td>
<td>Chromosome files in the 'Fasta' format usually contain large sequences like human genome.
  Sometimes users have to split these chromosomes into different files according to their 
  chromosome number. The 'chromseq' can help to handle this. So the selected chromosome sequence can be
  used for downstream analysis like motif finding. Howard Y. Chang(2019)
  &lt;<a href="https://doi.org/10.1038%2Fs41587-019-0206-z">doi:10.1038/s41587-019-0206-z</a>&gt;.</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MSQ-123/chromseq">https://github.com/MSQ-123/chromseq</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MSQ-123/chromseq/issues">https://github.com/MSQ-123/chromseq/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, base</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-05-08 13:06:01 UTC; lenovo</td>
</tr>
<tr>
<td>Author:</td>
<td>Shaoqian Ma [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shaoqian Ma &lt;897341109@qq.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-05-11 15:20:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='id'>Sampled Fasta file of chromosome sequence from hg19 blacklist</h2><span id='topic+id'></span>

<h3>Description</h3>

<p>This dataset is sampled from The hg19 blacklist.
For splitting a chromosome Fasta file, sometimes the
Fasta identifier is too complicated to manipulate. This data
can be used to show how to simplify the Fasta identifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(id)
</code></pre>


<h3>Format</h3>

<p>A character sequence with 20 elements</p>


<h3>References</h3>

<p>Satpathy A T, Granja J M, Yost K E, et al. (2019) Nature biotechnology 37,925â€“936.
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/31375813">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(id)
</code></pre>

<hr>
<h2 id='readToList'>Make a list file from large chromosome Fasta file</h2><span id='topic+readToList'></span>

<h3>Description</h3>

<p>Make a list file from large chromosome Fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readToList(id = id, text = text, con = con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readToList_+3A_id">id</code></td>
<td>
<p>The id list made from subFasID function</p>
</td></tr>
<tr><td><code id="readToList_+3A_text">text</code></td>
<td>
<p>Large character read in by readLines function from Fasta file</p>
</td></tr>
<tr><td><code id="readToList_+3A_con">con</code></td>
<td>
<p>A connection object or a character string, the connection must refer to the same Fasta file as text</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Chromosome Fasta file in list format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("text")
id &lt;- subFasID(text = text)
fil &lt;- tempfile(fileext = ".data")
write(text,file = fil)
con0 &lt;- file(fil, "r")
tex &lt;- readToList(id,text = text,con = con0)
</code></pre>

<hr>
<h2 id='replaceText'>Replace tedious chromosome identifier into simple format</h2><span id='topic+replaceText'></span>

<h3>Description</h3>

<p>Make the chromosome id starting with &quot;&gt;&quot; into simple format like
&quot;&gt;chr:1091194-1093520...&quot;,this is helpful for sorting the chromosome according to their number
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replaceText(type = "text", input = input)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replaceText_+3A_type">type</code></td>
<td>
<p>This can be either &quot;text&quot; or &quot;list&quot;, The previous is a large character containing each line of the Fasta file, the latter is a list in which each element contains a unit of Fasta file</p>
</td></tr>
<tr><td><code id="replaceText_+3A_input">input</code></td>
<td>
<p>The large character or list containing ids that need to be simplified</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The large character or list of Chromosome Fasta file with simplified id.
</p>


<h3>Author(s)</h3>

<p>Shaoqian Ma
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("id")
simpleID&lt;- replaceText(type = "text",input = id)
</code></pre>

<hr>
<h2 id='sortList'>Sort the chromosome list  according to the chromosome number</h2><span id='topic+sortList'></span>

<h3>Description</h3>

<p>Sort the chromosome list  according to the chromosome number
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortList(id = id, tex = tex, chrsig = "single")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sortList_+3A_id">id</code></td>
<td>
<p>The identifier list of the Fasta file made by subFasID</p>
</td></tr>
<tr><td><code id="sortList_+3A_tex">tex</code></td>
<td>
<p>A chromosome Fasta file in list format made by readToList function</p>
</td></tr>
<tr><td><code id="sortList_+3A_chrsig">chrsig</code></td>
<td>
<p>The number of characters of the chromosome,
either &quot;single&quot; or &quot;double&quot;, the previous means a single character
following &quot;chr&quot; in the Fasta identifier, the latter means two characters
following &quot;chr&quot; in the Fasta identifier. eg.&quot;chr1,chrX,chrY,chrM&quot; is
&quot;single&quot;;&quot;chr10,chr11&quot; is &quot;double&quot;. If you want to obtain
both &quot;single&quot; and &quot;double&quot; sorted list of chromosome, try
&quot;single&quot; and &quot;double&quot; respectively</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The sorted chromosome Fasta file in list format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("tex")
data("text")
text&lt;- replaceText(type = "text",input = text)
id &lt;- subFasID(text = text)
tex2&lt;- sortList(id=id,tex = tex,chrsig = "single")
tex3 &lt;- sortList(id=id,tex = tex,chrsig = "double")
</code></pre>

<hr>
<h2 id='splitChr'>Split all chromosomes from the sorted chromosome list</h2><span id='topic+splitChr'></span>

<h3>Description</h3>

<p>Split all chromosomes from the sorted chromosome list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitChr(tex = tex, chr = chr, sex = FALSE, outdir = ".")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="splitChr_+3A_tex">tex</code></td>
<td>
<p>The sorted chromosome list made by sortList function.</p>
</td></tr>
<tr><td><code id="splitChr_+3A_chr">chr</code></td>
<td>
<p>The chromosome number sequence, if the chromosome list is &quot;single&quot; which means a single character following &quot;chr&quot; in the Fasta identifier, be sure starting with 1 and ending with 9;
if the chromosome list is &quot;double&quot; which means two characters following &quot;chr&quot; in the Fasta identifier, be sure that starting with 10 but the ending can be changed.</p>
</td></tr>
<tr><td><code id="splitChr_+3A_sex">sex</code></td>
<td>
<p>Whether to output the sex chromosomes like X chromosome and Y chromosome.</p>
</td></tr>
<tr><td><code id="splitChr_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Write the splitted chromosome Fasta file to separated txt files according to the chromosome number.
</p>


<h3>Author(s)</h3>

<p>Shaoqian Ma
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tex)
data(text)
#Simplify the Fasta id
text&lt;- replaceText(type = "text",input = text)
#Subtract id
id &lt;- subFasID(text = text)
#Sort the fasta according to the chromosome number in id
tex2&lt;- sortList(id=id,tex = tex,chrsig = "single")
tex3 &lt;- sortList(id=id,tex = tex,chrsig = "double")
outdir &lt;- tempdir()
#Output the results
splitChr(tex = tex2,chr=seq(1,9),sex = TRUE,outdir = outdir)
splitChr(tex = tex3,chr=seq(10,22),sex = FALSE,outdir = outdir)
</code></pre>

<hr>
<h2 id='subFasID'>Subtract chromosome ids from Fasta file</h2><span id='topic+subFasID'></span>

<h3>Description</h3>

<p>Subtract chromosome ids from Fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subFasID(text = text)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subFasID_+3A_text">text</code></td>
<td>
<p>Large character read by readLines from chromosome Fasta file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The id list of the Fasta file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("text")
text&lt;- replaceText(type = "text",input = text)
id &lt;- subFasID(text = text)
</code></pre>

<hr>
<h2 id='tex'>Fasta file of chromosome sequence produced from sequence character</h2><span id='topic+tex'></span>

<h3>Description</h3>

<p>Data from &quot;Three representative inter and intra-subspecific crosses
reveal the genetic architecture of reproductive isolation in rice.&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tex)
</code></pre>


<h3>Format</h3>

<p>A large list containing 62 elements.</p>


<h3>References</h3>

<p>Li, G. et al. (2017) The Plant Journal 92, 349â€“362.
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/28805257">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tex)
</code></pre>

<hr>
<h2 id='text'>Fasta file of chromosome sequence</h2><span id='topic+text'></span>

<h3>Description</h3>

<p>A downsampled dataset containing the hg19 chromosome sequence from
the hg19 blacklist. The hg19 blacklist is obtained from the supplementary
dataset from &quot;Massively parallel single-cell chromatin landscapes of human 
immune cell development and intratumoral T cell exhaustion.&quot;
The dataset is sent to the UCSC Table Browser for obtaining the corresponding 
sequence file. The sequence file is processed with replaceText function to 
simplify the fasta id. To best illustate the usage, the sequence file is downsampled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(text)
</code></pre>


<h3>Format</h3>

<p>A character sequence with 2099 elements.</p>


<h3>References</h3>

<p>Satpathy A T, Granja J M, Yost K E, et al. (2019) Nature biotechnology 37, 925â€“936.
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/31375813">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(text)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
