<!DOCTYPE html><html lang="en"><head><title>Help for package CIDER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CIDER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculateDistMatOneModel'><p>Calculate Distance Matrix Using a Single Model</p></a></li>
<li><a href='#downsampling'><p>Downsampling Cells</p></a></li>
<li><a href='#estimateProb'><p>Estimate the Empirical Probability of Whether Two Set of Cells</p>
from Distinct Batches Belong to the Same Population</a></li>
<li><a href='#finalClustering'><p>Final Clustering Step for Meta-Clustering</p></a></li>
<li><a href='#gatherInitialClusters'><p>Gather Initial Cluster Names</p></a></li>
<li><a href='#getDistMat'><p>Calculate the Similarity Matrix</p></a></li>
<li><a href='#getGroupFit'><p>Calculate IDER-Based Similarity Between Two Groups</p></a></li>
<li><a href='#getIDEr'><p>Compute IDER-Based Similarity</p></a></li>
<li><a href='#hdbscan.seurat'><p>Initial Clustering for Evaluating Integration</p></a></li>
<li><a href='#initialClustering'><p>Initial Clustering</p></a></li>
<li><a href='#mergeInitialClusters'><p>Merge Initial Clusters</p></a></li>
<li><a href='#pancreas_meta'><p>Pancreas Metadata</p></a></li>
<li><a href='#plotDistMat'><p>Plot Similarity Matrix with pheatmap</p></a></li>
<li><a href='#plotHeatmap'><p>Plot Heatmap for the IDER-Based Similarity Matrix</p></a></li>
<li><a href='#plotNetwork'><p>Plot Network Graph</p></a></li>
<li><a href='#scatterPlot'><p>Scatterplot by a selected feature</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Meta-Clustering for scRNA-Seq Integration and Evaluation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.4</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhiyuan Hu &lt;zhiyuan.cheryl.hu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A workflow of (a) meta-clustering based on 
    inter-group similarity measures and (b) a ground-truth-free test metric to 
    assess the biological correctness of integration in real datasets. See Hu Z,
    Ahmed A, Yau C (2021) &lt;<a href="https://doi.org/10.1101%2F2021.03.29.437525">doi:10.1101/2021.03.29.437525</a>&gt; for
    more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/zhiyuan-hu-lab/CIDER">https://github.com/zhiyuan-hu-lab/CIDER</a>,
<a href="https://zhiyuan-hu-lab.github.io/CIDER/">https://zhiyuan-hu-lab.github.io/CIDER/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/zhiyuan-hu-lab/CIDER/issues">https://github.com/zhiyuan-hu-lab/CIDER/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>limma (&ge; 3.42.0), edgeR (&ge; 3.28.0), stats (&ge; 3.6.2),
foreach (&ge; 1.4.7), Seurat (&ge; 3.1.0), utils (&ge; 3.6.2),
pheatmap (&ge; 1.0.0), dbscan (&ge; 1.1-5), kernlab (&ge; 0.9-29),
doParallel, igraph, parallel, graphics, ggplot2, viridis</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0), statmod (&ge; 1.2.2),
cowplot</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-06 16:58:52 UTC; zhiyuanhu</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhiyuan Hu <a href="https://orcid.org/0000-0002-1688-6032"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Christopher Yau <a href="https://orcid.org/0000-0001-7615-8523"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ahmed Ahmed [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-07 09:50:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculateDistMatOneModel'>Calculate Distance Matrix Using a Single Model</h2><span id='topic+calculateDistMatOneModel'></span>

<h3>Description</h3>

<p>This function computes a similarity matrix by utilising 
a single linear model for differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateDistMatOneModel(
  matrix,
  metadata,
  verbose = TRUE,
  method = "voom",
  additional.variate = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculateDistMatOneModel_+3A_matrix">matrix</code></td>
<td>
<p>A count matrix with rows representing genes or features and columns representing samples or cells.</p>
</td></tr>
<tr><td><code id="calculateDistMatOneModel_+3A_metadata">metadata</code></td>
<td>
<p>A data frame containing metadata corresponding to the samples or cells. 
Each row should match a column in <code>matrix</code>.</p>
</td></tr>
<tr><td><code id="calculateDistMatOneModel_+3A_verbose">verbose</code></td>
<td>
<p>Logical. If <code>TRUE</code>, the function displays progress messages 
and a progress bar. The default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="calculateDistMatOneModel_+3A_method">method</code></td>
<td>
<p>A character string specifying the method for differential 
expression analysis. Options are &quot;voom&quot; or &quot;trend&quot;, with &quot;trend&quot; as the default.</p>
</td></tr>
<tr><td><code id="calculateDistMatOneModel_+3A_additional.variate">additional.variate</code></td>
<td>
<p>A character vector of additional variates to 
include in the linear model for regression.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A similarity matrix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getDistMat">getDistMat</a></code>
</p>

<hr>
<h2 id='downsampling'>Downsampling Cells</h2><span id='topic+downsampling'></span>

<h3>Description</h3>

<p>Downsamples cells from each group for IDER-based similarity calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downsampling(
  metadata,
  n.size = 35,
  seed = NULL,
  include = FALSE,
  replace = FALSE,
  lower.cutoff = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downsampling_+3A_metadata">metadata</code></td>
<td>
<p>A data frame containing at least two columns: one for group labels
and one for batch information. Each row corresponds to a single cell. Required.</p>
</td></tr>
<tr><td><code id="downsampling_+3A_n.size">n.size</code></td>
<td>
<p>Numeric value specifying the number of cells to use in each group.
Default is <code>35</code>.</p>
</td></tr>
<tr><td><code id="downsampling_+3A_seed">seed</code></td>
<td>
<p>Numeric value to set the random seed for sampling. Default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="downsampling_+3A_include">include</code></td>
<td>
<p>Logical value indicating whether to include groups that have fewer cells than <code>n.size</code>.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="downsampling_+3A_replace">replace</code></td>
<td>
<p>Logical value specifying whether to sample with replacement if a group
is smaller than <code>n.size</code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="downsampling_+3A_lower.cutoff">lower.cutoff</code></td>
<td>
<p>Numeric value indicating the minimum group size required for inclusion.
Default is <code>3</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of numeric indices (or cell names) for cells to be kept for downstream computation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # 'meta' is a data frame with columns 'label' and 'batch'
  meta &lt;- data.frame(
    label = c(rep("A", 40), rep("A", 35), rep("B", 20)),
    batch = c(rep("X", 40), rep("Y", 35), rep("X", 20))
  )
  keep_cells &lt;- downsampling(meta, n.size = 35, seed = 12345)
  
  # Display the selected indices
  print(keep_cells)

</code></pre>

<hr>
<h2 id='estimateProb'>Estimate the Empirical Probability of Whether Two Set of Cells
from Distinct Batches Belong to the Same Population</h2><span id='topic+estimateProb'></span>

<h3>Description</h3>

<p>This function computes the empirical probability that two sets of cells from 
distinct batches belong to the same population, based on the output of <code>getIDEr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateProb(
  seu,
  ider,
  batch.var = "Batch",
  n_size = 40,
  n.perm = 5,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimateProb_+3A_seu">seu</code></td>
<td>
<p>A Seurat object.</p>
</td></tr>
<tr><td><code id="estimateProb_+3A_ider">ider</code></td>
<td>
<p>A list returned by the <code>getIDEr</code> function.</p>
</td></tr>
<tr><td><code id="estimateProb_+3A_batch.var">batch.var</code></td>
<td>
<p>Character string specifying the metadata column that contains 
batch information. Default is &quot;Batch&quot;.</p>
</td></tr>
<tr><td><code id="estimateProb_+3A_n_size">n_size</code></td>
<td>
<p>Numeric value indicating the number of cells per group used to 
compute the similarity. Default is 40.</p>
</td></tr>
<tr><td><code id="estimateProb_+3A_n.perm">n.perm</code></td>
<td>
<p>Numeric value specifying the number of permutations to perform.</p>
</td></tr>
<tr><td><code id="estimateProb_+3A_verbose">verbose</code></td>
<td>
<p>Logical. If <code>TRUE</code>, progress messages are printed. 
Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with additional columns for the IDER-based similarity 
and the empirical probability of rejection.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hdbscan.seurat">hdbscan.seurat</a></code>, <code><a href="#topic+getIDEr">getIDEr</a></code>
</p>

<hr>
<h2 id='finalClustering'>Final Clustering Step for Meta-Clustering</h2><span id='topic+finalClustering'></span>

<h3>Description</h3>

<p>This function merges initial clusters into final clusters based on the IDEr similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>finalClustering(
  seu,
  dist,
  cutree.by = "h",
  cutree.h = 0.45,
  cutree.k = 3,
  hc.method = "complete"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="finalClustering_+3A_seu">seu</code></td>
<td>
<p>A Seurat object that has undergone the <code>getIDEr</code> step. Required.</p>
</td></tr>
<tr><td><code id="finalClustering_+3A_dist">dist</code></td>
<td>
<p>A list output from the <code>getIDEr</code> function. Required.</p>
</td></tr>
<tr><td><code id="finalClustering_+3A_cutree.by">cutree.by</code></td>
<td>
<p>Character string specifying whether to cut the dendrogram by 
height (&quot;h&quot;) or by a fixed number of clusters (&quot;k&quot;). Default is &quot;h&quot;.</p>
</td></tr>
<tr><td><code id="finalClustering_+3A_cutree.h">cutree.h</code></td>
<td>
<p>Numeric value between 0 and 1 indicating the height at which 
to cut the dendrogram. This parameter is ignored if <code>cutree.by = "k"</code>. Default is 0.45.</p>
</td></tr>
<tr><td><code id="finalClustering_+3A_cutree.k">cutree.k</code></td>
<td>
<p>Numeric value specifying the number of clusters to generate 
if <code>cutree.by = "k"</code>. This parameter is ignored if <code>cutree.by = "h"</code>. Default is 3.</p>
</td></tr>
<tr><td><code id="finalClustering_+3A_hc.method">hc.method</code></td>
<td>
<p>Character string specifying the method to be used in 
hierarchical clustering (passed to <code>hclust</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with the final clustering results stored in 
the <code>CIDER_clusters</code> column of its <code>meta.data</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getIDEr">getIDEr</a></code>
</p>

<hr>
<h2 id='gatherInitialClusters'>Gather Initial Cluster Names</h2><span id='topic+gatherInitialClusters'></span>

<h3>Description</h3>

<p>Merge initial clustering results from a list of Seurat objects into a single Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gatherInitialClusters(seu_list, seu)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gatherInitialClusters_+3A_seu_list">seu_list</code></td>
<td>
<p>A list containing Seurat objects with initial clustering results. Required.</p>
</td></tr>
<tr><td><code id="gatherInitialClusters_+3A_seu">seu</code></td>
<td>
<p>A Seurat object to which the merged initial cluster information will be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object containing the initial clustering results in the <code>initial_cluster</code> column of its <code>meta.data</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mergeInitialClusters">mergeInitialClusters</a></code>
</p>

<hr>
<h2 id='getDistMat'>Calculate the Similarity Matrix</h2><span id='topic+getDistMat'></span>

<h3>Description</h3>

<p>Compute the IDER-based similarity matrix for a list of Seurat objects. This 
function does not regress out batch effects and is designed for use during the initial clustering step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDistMat(
  seu_list,
  verbose = TRUE,
  tmp.initial.clusters = "seurat_clusters",
  method = "trend",
  batch.var = "Batch",
  additional.variate = NULL,
  downsampling.size = 35,
  downsampling.include = TRUE,
  downsampling.replace = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDistMat_+3A_seu_list">seu_list</code></td>
<td>
<p>A list containing Seurat objects. Required.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_verbose">verbose</code></td>
<td>
<p>Logical. If <code>TRUE</code>, progress messages and a progress bar are displayed. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_tmp.initial.clusters">tmp.initial.clusters</code></td>
<td>
<p>Character string specifying one of the column names from <code>Seurat@meta.data</code> 
that denotes groups, e.g., initial clusters. 
Default is &quot;seurat_clusters&quot;.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_method">method</code></td>
<td>
<p>Character string specifying the method for differential expression analysis.
Options are &quot;voom&quot; or &quot;trend&quot; (default is &quot;trend&quot;).</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_batch.var">batch.var</code></td>
<td>
<p>Character string specifying the metadata column containing batch information. Default is &quot;Batch&quot;.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_additional.variate">additional.variate</code></td>
<td>
<p>Character vector of additional variates to include in the linear model for regression.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_downsampling.size">downsampling.size</code></td>
<td>
<p>Numeric value indicating the number of cells to use per group. Default is 35.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_downsampling.include">downsampling.include</code></td>
<td>
<p>Logical. Whether to include groups with fewer cells than 
<code>downsampling.size</code>. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getDistMat_+3A_downsampling.replace">downsampling.replace</code></td>
<td>
<p>Logical. Whether to sample with replacement for groups 
smaller than <code>downsampling.size</code>. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of similarity matrices.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateDistMatOneModel">calculateDistMatOneModel</a></code>
</p>

<hr>
<h2 id='getGroupFit'>Calculate IDER-Based Similarity Between Two Groups</h2><span id='topic+getGroupFit'></span>

<h3>Description</h3>

<p>This function calculates the IDER-based similarity between two groups using a linear model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGroupFit(logCPM, design, contrast_m)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getGroupFit_+3A_logcpm">logCPM</code></td>
<td>
<p>A numeric matrix of log-transformed counts per million.</p>
</td></tr>
<tr><td><code id="getGroupFit_+3A_design">design</code></td>
<td>
<p>A design matrix for the differential expression analysis.</p>
</td></tr>
<tr><td><code id="getGroupFit_+3A_contrast_m">contrast_m</code></td>
<td>
<p>A contrast matrix specifying the comparison between the two groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value representing the IDER-based similarity between the two groups.
</p>

<hr>
<h2 id='getIDEr'>Compute IDER-Based Similarity</h2><span id='topic+getIDEr'></span>

<h3>Description</h3>

<p>Calculate the similarity matrix based on Inter-group Differential Expression 
(IDER) metrics with the selected batch effects regressed out.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getIDEr(
  seu,
  group.by.var = "initial_cluster",
  batch.by.var = "Batch",
  verbose = TRUE,
  use.parallel = FALSE,
  n.cores = 1,
  downsampling.size = 40,
  downsampling.include = TRUE,
  downsampling.replace = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getIDEr_+3A_seu">seu</code></td>
<td>
<p>A Seurat S4 object that includes an <code>initial_cluster</code> column in its <code>meta.data</code>. Required.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_group.by.var">group.by.var</code></td>
<td>
<p>Character string specifying the column in <code>seu@meta.data</code> 
that defines initial clusters (batch-specific groups). Default is &quot;initial_cluster&quot;.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_batch.by.var">batch.by.var</code></td>
<td>
<p>Character string specifying the metadata column that indicates batch information. Default is &quot;Batch&quot;.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_verbose">verbose</code></td>
<td>
<p>Logical. If <code>TRUE</code>, progress messages and a progress bar are displayed. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_use.parallel">use.parallel</code></td>
<td>
<p>Logical. If <code>TRUE</code>, parallel computation is used
(requires <code>doParallel</code>); in this case, no progress bar will be shown. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_n.cores">n.cores</code></td>
<td>
<p>Numeric. The number of cores to use for parallel computing. Default is 1.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_downsampling.size">downsampling.size</code></td>
<td>
<p>Numeric. The number of cells representing each group. Default is 40.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_downsampling.include">downsampling.include</code></td>
<td>
<p>Logical. Whether to include groups with fewer 
cells than <code>downsampling.size</code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getIDEr_+3A_downsampling.replace">downsampling.replace</code></td>
<td>
<p>Logical. Whether to sample with replacement if a 
group is smaller than <code>downsampling.size</code>. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of objects: a similarity matrix, a numeric vector recording 
the cells used, and a data frame of the group combinations included.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotNetwork">plotNetwork</a></code>, <code><a href="#topic+finalClustering">finalClustering</a></code>
</p>

<hr>
<h2 id='hdbscan.seurat'>Initial Clustering for Evaluating Integration</h2><span id='topic+hdbscan.seurat'></span>

<h3>Description</h3>

<p>This function applies HDBSCAN, a density-based clustering algorithm, to the corrected dimension 
reduction of a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hdbscan.seurat(
  seu,
  batch.var = "Batch",
  reduction = "pca",
  dims = seq_len(15),
  minPts = 25
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hdbscan.seurat_+3A_seu">seu</code></td>
<td>
<p>A Seurat object containing integrated or batch-corrected data (e.g. PCA results).</p>
</td></tr>
<tr><td><code id="hdbscan.seurat_+3A_batch.var">batch.var</code></td>
<td>
<p>Character string specifying the metadata column that contains batch information. 
Default is &quot;Batch&quot;.</p>
</td></tr>
<tr><td><code id="hdbscan.seurat_+3A_reduction">reduction</code></td>
<td>
<p>Character string specifying the name of the dimension reduction to use (e.g. &quot;PCA&quot;). 
Default is &quot;PCA&quot;.</p>
</td></tr>
<tr><td><code id="hdbscan.seurat_+3A_dims">dims</code></td>
<td>
<p>Numeric vector indicating the dimensions to be used for initial clustering. Default is 1:15.</p>
</td></tr>
<tr><td><code id="hdbscan.seurat_+3A_minpts">minPts</code></td>
<td>
<p>Integer specifying the minimum number of points required to form a cluster. 
This value is passed to the <code>hdbscan</code> function. Default is 25.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat object with two additional columns in its <code>meta.data</code>: 
<code>dbscan_cluster</code> and <code>initial_cluster</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getIDEr">getIDEr</a></code>, <code><a href="#topic+estimateProb">estimateProb</a></code>
</p>

<hr>
<h2 id='initialClustering'>Initial Clustering</h2><span id='topic+initialClustering'></span>

<h3>Description</h3>

<p>Perform batch-specific initial clustering on a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialClustering(
  seu,
  batch.var = "Batch",
  cut.height = 0.4,
  nfeatures = 2000,
  additional.vars.to.regress = NULL,
  dims = seq_len(14),
  resolution = 0.6,
  downsampling.size = 50,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="initialClustering_+3A_seu">seu</code></td>
<td>
<p>A Seurat object. Required.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_batch.var">batch.var</code></td>
<td>
<p>Character string specifying one of the column names in <code>seu@meta.data</code> 
used to partition the object into subsets. Default is &quot;Batch&quot;.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_cut.height">cut.height</code></td>
<td>
<p>Numeric value specifying the height at which to cut hierarchical trees. Default is 0.4.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_nfeatures">nfeatures</code></td>
<td>
<p>Numeric value indicating the number of high-variance genes to use. Default is 2000.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_additional.vars.to.regress">additional.vars.to.regress</code></td>
<td>
<p>Character vector of additional variable names from <code>seu@meta.data</code> to regress out. 
Optional. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_dims">dims</code></td>
<td>
<p>Numeric vector specifying the dimensions to be used for clustering (passed to Seurat). Default is 1:14.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_resolution">resolution</code></td>
<td>
<p>Numeric value for clustering resolution (passed to Seurat). Default is 0.6.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_downsampling.size">downsampling.size</code></td>
<td>
<p>Numeric value indicating the number of cells representing each group. Default is 40.</p>
</td></tr>
<tr><td><code id="initialClustering_+3A_verbose">verbose</code></td>
<td>
<p>Logical. If <code>TRUE</code>, a progress bar is displayed. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Seurat S4 object with initial cluster assignments stored in the <code>initial_cluster</code> column of its <code>meta.data</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getIDEr">getIDEr</a></code>, <code><a href="#topic+finalClustering">finalClustering</a></code>
</p>

<hr>
<h2 id='mergeInitialClusters'>Merge Initial Clusters</h2><span id='topic+mergeInitialClusters'></span>

<h3>Description</h3>

<p>Merge initial clusters based on a provided similarity matrix and hierarchical clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeInitialClusters(
  seu_list,
  dist_list,
  use = "coef",
  method = "hc",
  hc.method = "average",
  cutree.by = "h",
  cutree.h = 0.6,
  cutree.k = 3,
  batch.var = "Batch"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeInitialClusters_+3A_seu_list">seu_list</code></td>
<td>
<p>A list of Seurat objects containing the single-cell data. This parameter is required.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_dist_list">dist_list</code></td>
<td>
<p>A list of similarity matrices as returned by <code>getDistMat()</code>. The order of matrices
should correspond to that of the Seurat objects in <code>seu_list</code>.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_use">use</code></td>
<td>
<p>A string specifying the similarity measure to use. Currently, only &quot;coef&quot; is supported. Default is &quot;coef&quot;.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_method">method</code></td>
<td>
<p>A string specifying the clustering method to employ. The default is &quot;hc&quot; for hierarchical clustering.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_hc.method">hc.method</code></td>
<td>
<p>A string passed to the <code>method</code> parameter of <code>hclust()</code>. Default is &quot;average&quot;.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_cutree.by">cutree.by</code></td>
<td>
<p>A character indicating whether to cut the dendrogram by height (&quot;h&quot;, default) or by
a set number of clusters (&quot;k&quot;).</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_cutree.h">cutree.h</code></td>
<td>
<p>A numeric value defining the height at which to cut the tree if <code>cutree.by = "h"</code>. Default is 0.6.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_cutree.k">cutree.k</code></td>
<td>
<p>A numeric value specifying the number of clusters to generate if <code>cutree.by = "k"</code>. Default is 3.</p>
</td></tr>
<tr><td><code id="mergeInitialClusters_+3A_batch.var">batch.var</code></td>
<td>
<p>A character string representing the metadata column name that contains batch information.
Default is &quot;Batch&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function accepts a list of Seurat objects and a corresponding list of similarity matrices,
and then merges the initial clusters using a hierarchical clustering approach. The updated cluster
assignments are stored within each Seurat object.
</p>


<h3>Value</h3>

<p>A list of Seurat objects in which the initial clustering has been updated. The new cluster
assignments are stored in the <code>inicluster</code> field of each Seurat object, whilst the original
assignments are preserved in the <code>inicluster_tmp</code> field.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gatherInitialClusters">gatherInitialClusters</a></code>, <code><a href="#topic+initialClustering">initialClustering</a></code>
</p>

<hr>
<h2 id='pancreas_meta'>Pancreas Metadata</h2><span id='topic+pancreas_meta'></span>

<h3>Description</h3>

<p>This dataset provides cell-level metadata for the human and mouse 
pancreatic data used in the study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pancreas_meta)
</code></pre>


<h3>Format</h3>

<p>A data frame with 10127 rows and 3 columns:
</p>

<dl>
<dt>Batch</dt><dd><p>Species information (human or mouse).</p>
</dd>
<dt>Group</dt><dd><p>Cell type annotation.</p>
</dd>
<dt>Sample</dt><dd><p>Donor information.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Cell-level metadata for cross-species pancreatic data.
</p>


<h3>Source</h3>

<p>The metadata were downloaded alongside the count matrix from NCBI GEO accession GSE84133.  
Reference: Baron M, Veres A, Wolock SL, Faust AL, Gaujoux R, Vetere A, et al. 
A single-cell transcriptomic map of the human and mouse pancreas reveals inter- and intra-cell 
population structure. Cell Syst. 2016;3:346â€“360.e4.
</p>

<hr>
<h2 id='plotDistMat'>Plot Similarity Matrix with pheatmap</h2><span id='topic+plotDistMat'></span>

<h3>Description</h3>

<p>This function creates a heatmap of the similarity matrix computed by <code>getDistMat()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDistMat(dist.list, use = "coef")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDistMat_+3A_dist.list">dist.list</code></td>
<td>
<p>A list representing the similarity matrix output by <code>getDistMat()</code>. Required.</p>
</td></tr>
<tr><td><code id="plotDistMat_+3A_use">use</code></td>
<td>
<p>Character string specifying the similarity measure to use. Default is &quot;coef&quot;. 
No other option is currently available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>pheatmap</code> object displaying the similarity matrix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getDistMat">getDistMat</a></code>
</p>

<hr>
<h2 id='plotHeatmap'>Plot Heatmap for the IDER-Based Similarity Matrix</h2><span id='topic+plotHeatmap'></span>

<h3>Description</h3>

<p>This function generates a heatmap that visualises the similarity between shared 
groups across batches, as computed by <code>getIDEr</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHeatmap(seu, ider, batch.var = "Batch")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotHeatmap_+3A_seu">seu</code></td>
<td>
<p>A Seurat object.</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_ider">ider</code></td>
<td>
<p>The output list from the <code>getIDEr</code> function.</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_batch.var">batch.var</code></td>
<td>
<p>Character string specifying the metadata column that contains batch information. Default is &quot;Batch&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap displaying the similarity between shared groups across batches.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getIDEr">getIDEr</a></code>
</p>

<hr>
<h2 id='plotNetwork'>Plot Network Graph</h2><span id='topic+plotNetwork'></span>

<h3>Description</h3>

<p>Visualise the network based on an IDER-based similarity matrix. 
The vertexes are initial clusters, and
the edge width denotes the similarity between two initial clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNetwork(
  seu,
  ider,
  batch.var = "Batch",
  colour.by = NULL,
  weight.factor = 6.5,
  col.vector = NULL,
  vertex.size = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotNetwork_+3A_seu">seu</code></td>
<td>
<p>Seurat S4 object after the step of <code>getIDER</code>, containing 
<code>initial_cluster</code> and <code>Batch</code> in its meta.data. Required.</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_ider">ider</code></td>
<td>
<p>A list. Output of 'getIDER'. Required.</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_batch.var">batch.var</code></td>
<td>
<p>Character. Metadata colname containing batch information.
(Default: <code>Batch</code>)</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_colour.by">colour.by</code></td>
<td>
<p>Character. It should be one of the colnames of Seurat 
object meta.data.It is used to colour the vertex of the network graph. 
(Default: <code>NULL</code>)</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_weight.factor">weight.factor</code></td>
<td>
<p>Numerical. Adjust the thickness of the edges. 
(Default: 6.5)</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_col.vector">col.vector</code></td>
<td>
<p>A vector of Hex colour codes. If no value is given 
(default), a vector of 74 colours will be used.</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_vertex.size">vertex.size</code></td>
<td>
<p>Numerical. Adjsut the size of vertexes. (Default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An igraph object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getIDEr">getIDEr</a></code>
</p>

<hr>
<h2 id='scatterPlot'>Scatterplot by a selected feature</h2><span id='topic+scatterPlot'></span>

<h3>Description</h3>

<p>Scatterplot of a Seurat object based on dimension reduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatterPlot(
  seu,
  reduction,
  colour.by,
  colvec = NULL,
  title = NULL,
  sort.by.numbers = TRUE,
  viridis_option = "B"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scatterPlot_+3A_seu">seu</code></td>
<td>
<p>Seurat S4 object after the step of <code>getIDER</code>. Required.</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_reduction">reduction</code></td>
<td>
<p>Character. The dimension reduction used to plot. Common
options: <code>"pca"</code>, <code>"tsne"</code>, <code>"umap"</code>. The availability of dimension reduction
can be checked by <code>Reductions(seu)</code>.</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_colour.by">colour.by</code></td>
<td>
<p>Character. One of the column names of <code>seu@meta.data</code>.
Can be either discreet or continuous variables.</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_colvec">colvec</code></td>
<td>
<p>A vector of Hex colour codes. If no value is given (default),
a vector of 74 colours will be used.</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_title">title</code></td>
<td>
<p>Character. Title of the figure.</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_sort.by.numbers">sort.by.numbers</code></td>
<td>
<p>Boolean. Whether to sort the groups by the number
of cells.(Default: <code>True</code>)</p>
</td></tr>
<tr><td><code id="scatterPlot_+3A_viridis_option">viridis_option</code></td>
<td>
<p>viridis_option. (Default: <code>B</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 scatter plot
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
