<!DOCTYPE html><html><head><title>Help for package cinaR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cinaR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annotatePeaks'><p>annotatePeaks</p></a></li>
<li><a href='#bed'><p>Example peaks from bone marrow of B6 mice</p></a></li>
<li><a href='#cinaR'><p>cinaR</p></a></li>
<li><a href='#color_values'><p>color values</p></a></li>
<li><a href='#differentialAnalyses'><p>Differential Analyses</p></a></li>
<li><a href='#dot_plot'><p>dot_plot</p></a></li>
<li><a href='#filterConsensus'><p>filterConsensus</p></a></li>
<li><a href='#grch37'><p>Grch37</p></a></li>
<li><a href='#grch38'><p>Grch38</p></a></li>
<li><a href='#grcm38'><p>Grcm38</p></a></li>
<li><a href='#GSEA'><p>GSEA</p></a></li>
<li><a href='#heatmap_differential'><p>heatmap_differential</p></a></li>
<li><a href='#heatmap_var_peaks'><p>heatmap_var_peaks</p></a></li>
<li><a href='#HPEA'><p>HPEA</p></a></li>
<li><a href='#normalizeConsensus'><p>normalizeConsensus</p></a></li>
<li><a href='#pca_plot'><p>pca_plot</p></a></li>
<li><a href='#run_enrichment'><p>run_enrichment</p></a></li>
<li><a href='#scale_rows'><p>scale_rows</p></a></li>
<li><a href='#show_comparisons'><p>show_comparisons</p></a></li>
<li><a href='#verboseFn'><p>verboseFn</p></a></li>
<li><a href='#vp2008'><p>Immune modules</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Computational Pipeline for Bulk 'ATAC-Seq' Profiles</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Differential analyses and Enrichment pipeline for bulk 'ATAC-seq' data
  analyses. This package combines different packages to have an ultimate package
  for both data analyses and visualization of 'ATAC-seq' data. Methods are described in 
  'Karakaslar et al.' (2021) &lt;<a href="https://doi.org/10.1101%2F2021.03.05.434143">doi:10.1101/2021.03.05.434143</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/eonurk/cinaR/">https://github.com/eonurk/cinaR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/eonurk/cinaR/issues/">https://github.com/eonurk/cinaR/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ChIPseeker, DESeq2, dplyr, edgeR, fgsea, GenomicRanges,
TxDb.Hsapiens.UCSC.hg38.knownGene,
TxDb.Hsapiens.UCSC.hg19.knownGene,
TxDb.Mmusculus.UCSC.mm10.knownGene, ggplot2, ggrepel,
grDevices, limma, utils, pheatmap, preprocessCore,
RColorBrewer, sva, writexl</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-17 08:45:32 UTC; onur-lumc</td>
</tr>
<tr>
<td>Author:</td>
<td>Onur Karakaslar [aut, cre],
  Duygu Ucar <a href="https://orcid.org/0000-0002-9772-3066"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Onur Karakaslar &lt;eonurkara@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-18 14:00:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='annotatePeaks'>annotatePeaks</h2><span id='topic+annotatePeaks'></span>

<h3>Description</h3>

<p>Runs DA pipeline and makes it ready for enrichment analyses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotatePeaks(cp, reference.genome, show.annotation.pie = FALSE, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotatePeaks_+3A_cp">cp</code></td>
<td>
<p>bed formatted consensus peak matrix: CHR, START, STOP and raw peak counts (peaks by 3+samples)</p>
</td></tr>
<tr><td><code id="annotatePeaks_+3A_reference.genome">reference.genome</code></td>
<td>
<p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p>
</td></tr>
<tr><td><code id="annotatePeaks_+3A_show.annotation.pie">show.annotation.pie</code></td>
<td>
<p>shows the annotation pie chart produced with ChipSeeker</p>
</td></tr>
<tr><td><code id="annotatePeaks_+3A_verbose">verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DApeaks returns DA peaks
</p>

<hr>
<h2 id='bed'>Example peaks from bone marrow of B6 mice</h2><span id='topic+bed'></span>

<h3>Description</h3>

<p>Example peaks from bone marrow of B6 mice
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(atac_seq_consensus_bm)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 1000 rows and 25 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(atac_seq_consensus_bm)
</code></pre>

<hr>
<h2 id='cinaR'>cinaR</h2><span id='topic+cinaR'></span>

<h3>Description</h3>

<p>Runs differential analyses and enrichment pipelines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cinaR(
  matrix,
  contrasts,
  experiment.type = "ATAC-Seq",
  DA.choice = 1,
  DA.fdr.threshold = 0.05,
  DA.lfc.threshold = 0,
  comparison.scheme = "OVO",
  save.DA.peaks = FALSE,
  DA.peaks.path = NULL,
  norm.method = "cpm",
  filter.method = "custom",
  library.threshold = 2,
  cpm.threshold = 1,
  TSS.threshold = 50000,
  show.annotation.pie = FALSE,
  reference.genome = NULL,
  batch.correction = FALSE,
  batch.information = NULL,
  additional.covariates = NULL,
  sv.number = NULL,
  run.enrichment = TRUE,
  enrichment.method = NULL,
  enrichment.FDR.cutoff = 1,
  background.genes.size = 20000,
  geneset = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cinaR_+3A_matrix">matrix</code></td>
<td>
<p>either bed formatted consensus peak matrix (peaks by 3+samples)
CHR, START, STOP and raw peak counts OR count matrix (genes by 1+samples).</p>
</td></tr>
<tr><td><code id="cinaR_+3A_contrasts">contrasts</code></td>
<td>
<p>user-defined contrasts for comparing samples</p>
</td></tr>
<tr><td><code id="cinaR_+3A_experiment.type">experiment.type</code></td>
<td>
<p>The type of experiment either set to &quot;ATAC-Seq&quot; or &quot;RNA-Seq&quot;</p>
</td></tr>
<tr><td><code id="cinaR_+3A_da.choice">DA.choice</code></td>
<td>
<p>determines which pipeline to run:
(1) edgeR, (2) limma-voom, (3) limma-trend, (4) DEseq2.
Note: Use limma-trend if consensus peaks are already normalized, otherwise use other methods.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_da.fdr.threshold">DA.fdr.threshold</code></td>
<td>
<p>fdr cut-off for differential analyses</p>
</td></tr>
<tr><td><code id="cinaR_+3A_da.lfc.threshold">DA.lfc.threshold</code></td>
<td>
<p>log-fold change cutoff for differential analyses</p>
</td></tr>
<tr><td><code id="cinaR_+3A_comparison.scheme">comparison.scheme</code></td>
<td>
<p>either one-vs-one (OVO) or one-vs-all (OVA) comparisons.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_save.da.peaks">save.DA.peaks</code></td>
<td>
<p>saves differentially accessible peaks to an excel file</p>
</td></tr>
<tr><td><code id="cinaR_+3A_da.peaks.path">DA.peaks.path</code></td>
<td>
<p>the path which the excel file of the DA peaks will be saved,
if not set it will be saved to current directory.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_norm.method">norm.method</code></td>
<td>
<p>normalization method for consensus peaks</p>
</td></tr>
<tr><td><code id="cinaR_+3A_filter.method">filter.method</code></td>
<td>
<p>filtering method for low expressed peaks</p>
</td></tr>
<tr><td><code id="cinaR_+3A_library.threshold">library.threshold</code></td>
<td>
<p>number of libraries a peak occurs so that it is not filtered default set to 2</p>
</td></tr>
<tr><td><code id="cinaR_+3A_cpm.threshold">cpm.threshold</code></td>
<td>
<p>count per million threshold for not to filter a peak</p>
</td></tr>
<tr><td><code id="cinaR_+3A_tss.threshold">TSS.threshold</code></td>
<td>
<p>Distance to transcription start site in base-pairs. Default set to 50,000.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_show.annotation.pie">show.annotation.pie</code></td>
<td>
<p>shows the annotation pie chart produced with ChipSeeker</p>
</td></tr>
<tr><td><code id="cinaR_+3A_reference.genome">reference.genome</code></td>
<td>
<p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_batch.correction">batch.correction</code></td>
<td>
<p>logical, if set will run unsupervised batch correction
via sva (default) or if the batch information is known 'batch.information'
argument should be provided by user.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_batch.information">batch.information</code></td>
<td>
<p>character vector, given by user.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_additional.covariates">additional.covariates</code></td>
<td>
<p>vector or data.frame, this parameter will be directly added to design
matrix before running the differential analyses, therefore won't affect the batch corrections but
adjust the results in down-stream analyses.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_sv.number">sv.number</code></td>
<td>
<p>number of surrogate variables to be calculated using SVA, best left untouched.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_run.enrichment">run.enrichment</code></td>
<td>
<p>logical, turns off enrichment pipeline</p>
</td></tr>
<tr><td><code id="cinaR_+3A_enrichment.method">enrichment.method</code></td>
<td>
<p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p>
</td></tr>
<tr><td><code id="cinaR_+3A_enrichment.fdr.cutoff">enrichment.FDR.cutoff</code></td>
<td>
<p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p>
</td></tr>
<tr><td><code id="cinaR_+3A_background.genes.size">background.genes.size</code></td>
<td>
<p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_geneset">geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td></tr>
<tr><td><code id="cinaR_+3A_verbose">verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns differentially accessible peaks
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10")



</code></pre>

<hr>
<h2 id='color_values'>color values</h2><span id='topic+color_values'></span>

<h3>Description</h3>

<p>color values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_values
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 8.
</p>

<hr>
<h2 id='differentialAnalyses'>Differential Analyses</h2><span id='topic+differentialAnalyses'></span>

<h3>Description</h3>

<p>Runs differential analyses pipeline of choice on consensus peaks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>differentialAnalyses(
  final.matrix,
  contrasts,
  experiment.type,
  DA.choice,
  DA.fdr.threshold,
  DA.lfc.threshold,
  comparison.scheme,
  save.DA.peaks,
  DA.peaks.path,
  batch.correction,
  batch.information,
  additional.covariates,
  sv.number,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="differentialAnalyses_+3A_final.matrix">final.matrix</code></td>
<td>
<p>Annotated Consensus peaks</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_contrasts">contrasts</code></td>
<td>
<p>user-defined contrasts for comparing samples</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_experiment.type">experiment.type</code></td>
<td>
<p>The type of experiment either set to &quot;ATAC-Seq&quot; or &quot;RNA-Seq&quot;</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_da.choice">DA.choice</code></td>
<td>
<p>determines which pipeline to run:
(1) edgeR, (2) limma-voom, (3) limma-trend, (4) DEseq2</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_da.fdr.threshold">DA.fdr.threshold</code></td>
<td>
<p>fdr cut-off for differential analyses</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_da.lfc.threshold">DA.lfc.threshold</code></td>
<td>
<p>log-fold change cutoff for differential analyses</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_comparison.scheme">comparison.scheme</code></td>
<td>
<p>either one-vs-one (OVO) or one-vs-all (OVA) comparisons.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_save.da.peaks">save.DA.peaks</code></td>
<td>
<p>logical, saves differentially accessible peaks to an excel file</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_da.peaks.path">DA.peaks.path</code></td>
<td>
<p>the path which the excel file of the DA peaks will be saved,
if not set it will be saved to current directory.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_batch.correction">batch.correction</code></td>
<td>
<p>logical, if set will run unsupervised batch correction
via sva (default) or if the batch information is known 'batch.information'
argument should be provided by user.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_batch.information">batch.information</code></td>
<td>
<p>character vector, given by user.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_additional.covariates">additional.covariates</code></td>
<td>
<p>vector or data.frame, this parameter will be directly added to design
matrix before running the differential analyses, therefore won't affect the batch corrections but
adjust the results in down-stream analyses.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_sv.number">sv.number</code></td>
<td>
<p>number of surrogate variables to be calculated using SVA, best left untouched.</p>
</td></tr>
<tr><td><code id="differentialAnalyses_+3A_verbose">verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns consensus peaks (batch corrected version if enabled) and DA peaks
</p>

<hr>
<h2 id='dot_plot'>dot_plot</h2><span id='topic+dot_plot'></span>

<h3>Description</h3>

<p>Given the results from 'cinaR' it produces dot plots for enrichment analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dot_plot(results, fdr.cutoff = 0.1, filter.pathways = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dot_plot_+3A_results">results</code></td>
<td>
<p>cinaR result object</p>
</td></tr>
<tr><td><code id="dot_plot_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>
<p>Pathways with smaller fdr values than the cut-off
will be shown as dots.</p>
</td></tr>
<tr><td><code id="dot_plot_+3A_filter.pathways">filter.pathways</code></td>
<td>
<p>logical, it will filter the pathways from dot plot
with fdr values less than 'fdr.cutoff'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10")

dot_plot(results)

</code></pre>

<hr>
<h2 id='filterConsensus'>filterConsensus</h2><span id='topic+filterConsensus'></span>

<h3>Description</h3>

<p>Filters lowly expressed peaks from down-stream analyses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterConsensus(
  cp,
  filter.method = "custom",
  library.threshold = 2,
  cpm.threshold = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterConsensus_+3A_cp">cp</code></td>
<td>
<p>consensus peak matrix, with unique ids at rownames.</p>
</td></tr>
<tr><td><code id="filterConsensus_+3A_filter.method">filter.method</code></td>
<td>
<p>filtering method for low expressed peaks</p>
</td></tr>
<tr><td><code id="filterConsensus_+3A_library.threshold">library.threshold</code></td>
<td>
<p>number of libraries a peak occurs so that it is not filtered default set to 2</p>
</td></tr>
<tr><td><code id="filterConsensus_+3A_cpm.threshold">cpm.threshold</code></td>
<td>
<p>count per million threshold for not to filter a peak</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns differentially accessible peaks
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
cp &lt;- matrix(rexp(200, rate=.1), ncol=20)

## using cpm function from `edgeR` package
cp.filtered &lt;- filterConsensus(cp)

</code></pre>

<hr>
<h2 id='grch37'>Grch37</h2><span id='topic+grch37'></span>

<h3>Description</h3>

<p>Grch37
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grch37)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 66978 rows and 3 columns.
</p>

<hr>
<h2 id='grch38'>Grch38</h2><span id='topic+grch38'></span>

<h3>Description</h3>

<p>Grch38
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grch38)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 67495 rows and 3 columns.
</p>

<hr>
<h2 id='grcm38'>Grcm38</h2><span id='topic+grcm38'></span>

<h3>Description</h3>

<p>Grcm38
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grcm38)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 25350 rows and 4 columns.
</p>

<hr>
<h2 id='GSEA'>GSEA</h2><span id='topic+GSEA'></span>

<h3>Description</h3>

<p>Gene set enrichment analyses, runs 'fgsea' package implementation with preset values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSEA(genes, geneset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GSEA_+3A_genes">genes</code></td>
<td>
<p>DA gene names to be checked if they are over-represented or not.</p>
</td></tr>
<tr><td><code id="GSEA_+3A_geneset">geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame, list of pathways and their enrichment (adjusted) p-values.
</p>


<h3>References</h3>

<p>G. Korotkevich, V. Sukhov, A. Sergushichev. Fast gene set enrichment analysis. bioRxiv (2019),
doi:10.1101/060012
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)
library(fgsea)
data(examplePathways)
data(exampleRanks)
GSEA(exampleRanks, examplePathways)

</code></pre>

<hr>
<h2 id='heatmap_differential'>heatmap_differential</h2><span id='topic+heatmap_differential'></span>

<h3>Description</h3>

<p>plot differentially accessible peaks for a given comparison
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_differential(results, comparison = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_differential_+3A_results">results</code></td>
<td>
<p>cinaR result object</p>
</td></tr>
<tr><td><code id="heatmap_differential_+3A_comparison">comparison</code></td>
<td>
<p>these are created by cinaR from 'contrasts' user provided. If not selected the first comparison will be shown!</p>
</td></tr>
<tr><td><code id="heatmap_differential_+3A_...">...</code></td>
<td>
<p>additional arguments for heatmap function, for more info '?pheatmap'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10")

heatmap_differential(results)

</code></pre>

<hr>
<h2 id='heatmap_var_peaks'>heatmap_var_peaks</h2><span id='topic+heatmap_var_peaks'></span>

<h3>Description</h3>

<p>plot most variable k peaks (default k = 100) among all samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmap_var_peaks(results, heatmap.peak.count = 100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmap_var_peaks_+3A_results">results</code></td>
<td>
<p>cinaR result object</p>
</td></tr>
<tr><td><code id="heatmap_var_peaks_+3A_heatmap.peak.count">heatmap.peak.count</code></td>
<td>
<p>number of peaks to be plotted.
If number of peaks are less than k then all peaks will be used.</p>
</td></tr>
<tr><td><code id="heatmap_var_peaks_+3A_...">...</code></td>
<td>
<p>additional arguments for heatmap function, for more info '?pheatmap'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# creating dummy results
results &lt;- NULL
results[["cp"]] &lt;- bed[,c(4:25)]

heatmap_var_peaks(results)

</code></pre>

<hr>
<h2 id='HPEA'>HPEA</h2><span id='topic+HPEA'></span>

<h3>Description</h3>

<p>Hyper-geometric p-value enrichment analyses, looking for over-representation of a set of genes on given pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HPEA(genes, geneset, background.genes.size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HPEA_+3A_genes">genes</code></td>
<td>
<p>DA gene names to be checked if they are over-represented or not.</p>
</td></tr>
<tr><td><code id="HPEA_+3A_geneset">geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td></tr>
<tr><td><code id="HPEA_+3A_background.genes.size">background.genes.size</code></td>
<td>
<p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame, list of pathways and their enrichment (adjusted) p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)

data("VP2008")
genes.to.test &lt;- vp2008[[1]][1:10]
HPEA(genes.to.test,vp2008, background.genes.size = 20e3)


</code></pre>

<hr>
<h2 id='normalizeConsensus'>normalizeConsensus</h2><span id='topic+normalizeConsensus'></span>

<h3>Description</h3>

<p>Normalizes consensus peak using different methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalizeConsensus(cp, norm.method = "cpm", log.option = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeConsensus_+3A_cp">cp</code></td>
<td>
<p>bed formatted consensus peak matrix: CHR, START, STOP and raw peak counts (peaks by 3+samples)</p>
</td></tr>
<tr><td><code id="normalizeConsensus_+3A_norm.method">norm.method</code></td>
<td>
<p>normalization method for consensus peaks</p>
</td></tr>
<tr><td><code id="normalizeConsensus_+3A_log.option">log.option</code></td>
<td>
<p>logical, log option for cpm function in edgeR</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Normalized consensus peaks
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)
cp &lt;- matrix(rexp(200, rate=.1), ncol=20)

## using cpm function from `edgeR` package
cp.normalized &lt;- normalizeConsensus(cp)

## quantile normalization option
cp.normalized &lt;- normalizeConsensus(cp, norm.method = "quantile")
</code></pre>

<hr>
<h2 id='pca_plot'>pca_plot</h2><span id='topic+pca_plot'></span>

<h3>Description</h3>

<p>pca_plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_plot(results, overlaid.info, sample.names = NULL, show.names = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_plot_+3A_results">results</code></td>
<td>
<p>cinaR result object</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_overlaid.info">overlaid.info</code></td>
<td>
<p>overlaid information onto the samples</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_sample.names">sample.names</code></td>
<td>
<p>names of the samples shown on pca plot</p>
</td></tr>
<tr><td><code id="pca_plot_+3A_show.names">show.names</code></td>
<td>
<p>logical, if set FALSE sample names will be hidden</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# creating dummy results
results &lt;- NULL
results[["cp"]] &lt;- bed[,c(4:25)]

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

## overlays the contrasts info onto PCA plots
pca_plot(results, contrasts)

## you can overlay other information as well,
## as long as it is the same length with the
## number of samples.

sample.info &lt;- c(rep("Group A", 11), rep("Group B", 11))
pca_plot(results, sample.info, show.names = FALSE)

</code></pre>

<hr>
<h2 id='run_enrichment'>run_enrichment</h2><span id='topic+run_enrichment'></span>

<h3>Description</h3>

<p>This function is run, if the enrichment pipeline wants to be called afterwards.
Setting reference genome to the same genome which cinaR was run should be given to this function!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_enrichment(
  results,
  geneset = NULL,
  experiment.type = "ATAC-Seq",
  reference.genome = NULL,
  enrichment.method = NULL,
  enrichment.FDR.cutoff = 1,
  background.genes.size = 20000,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_enrichment_+3A_results">results</code></td>
<td>
<p>list, DA peaks list for different contrasts</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_geneset">geneset</code></td>
<td>
<p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using 'read.gmt' function from 'qusage'
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_experiment.type">experiment.type</code></td>
<td>
<p>The type of experiment either set to &quot;ATAC-Seq&quot; or &quot;RNA-Seq&quot;</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_reference.genome">reference.genome</code></td>
<td>
<p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_enrichment.method">enrichment.method</code></td>
<td>
<p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_enrichment.fdr.cutoff">enrichment.FDR.cutoff</code></td>
<td>
<p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_background.genes.size">background.genes.size</code></td>
<td>
<p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p>
</td></tr>
<tr><td><code id="run_enrichment_+3A_verbose">verbose</code></td>
<td>
<p>prints messages through running the pipeline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, enrichment analyses results along with corresponding differential analyses outcomes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'

# a vector for comparing the examples
contrasts &lt;- sapply(strsplit(colnames(bed), split = "-", fixed = TRUE),
                    function(x){x[1]})[4:25]

results &lt;- cinaR(bed, contrasts, reference.genome = "mm10", run.enrichment = FALSE)

results_with_enrichment &lt;- run_enrichment(results, reference.genome = "mm10")

</code></pre>

<hr>
<h2 id='scale_rows'>scale_rows</h2><span id='topic+scale_rows'></span>

<h3>Description</h3>

<p>Normalize (z-score) rows of a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_rows(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_rows_+3A_x">x</code></td>
<td>
<p>a matrix, possibly containing gene by samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Row-normalized matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(cinaR)
data(atac_seq_consensus_bm) # calls 'bed'
bed.row.normalized &lt;- scale_rows(bed[,c(4:25)])
head(bed.row.normalized)

</code></pre>

<hr>
<h2 id='show_comparisons'>show_comparisons</h2><span id='topic+show_comparisons'></span>

<h3>Description</h3>

<p>returns the names of the created comparisons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_comparisons(results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_comparisons_+3A_results">results</code></td>
<td>
<p>output of the cinaR</p>
</td></tr>
</table>


<h3>Value</h3>

<p>comparisons created
</p>

<hr>
<h2 id='verboseFn'>verboseFn</h2><span id='topic+verboseFn'></span>

<h3>Description</h3>

<p>returns a printing function to be used with in the script
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verboseFn(verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verboseFn_+3A_verbose">verbose</code></td>
<td>
<p>boolean, determines whether the output going be printed or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>print function
</p>

<hr>
<h2 id='vp2008'>Immune modules</h2><span id='topic+vp2008'></span>

<h3>Description</h3>

<p>Immune modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(VP2008)
</code></pre>


<h3>Format</h3>

<p>An object of class GMT; see <code>read.gmt</code> from qusage package.
</p>


<h3>References</h3>

<p>Chaussabel et al. (2008) Immunity 29:150-164
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2727981/">PubMed</a>)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
