<!DOCTYPE html><html><head><title>Help for package CINNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CINNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_centralities'><p>Centrality measure calculation</p></a></li>
<li><a href='#cortex'><p>Macaque Visual Cortex Network</p></a></li>
<li><a href='#dangalchev_closeness_centrality'><p>Dangalchev Closeness Centrality</p></a></li>
<li><a href='#drugTarget'><p>Drug Target Network</p></a></li>
<li><a href='#giant_component_extract'><p>Giant component extraction of a graph</p></a></li>
<li><a href='#graph_extract_components'><p>Component extraction of a graph</p></a></li>
<li><a href='#group_centrality'><p>Group Centrality</p></a></li>
<li><a href='#harmonic_centrality'><p>Harmonic Centrality</p></a></li>
<li><a href='#kangaroo'><p>Kangaroo Network</p></a></li>
<li><a href='#local_bridging_centrality'><p>Local Bridging Centrality</p></a></li>
<li><a href='#misc_extract_components'><p>Component extraction of miscellaneous graph formats</p></a></li>
<li><a href='#pca_centralities'><p>PCA Centrality Measures</p></a></li>
<li><a href='#proper_centralities'><p>Proper centrality measure representation</p></a></li>
<li><a href='#rhesus'><p>Moreno Rhesus Network</p></a></li>
<li><a href='#summary_calculate_centralities'><p>Summarize centrality measure calculation results</p></a></li>
<li><a href='#summary_graph_extract_components'><p>Summarize component extraction of a graph</p></a></li>
<li><a href='#summary_pca_centralities'><p>Summarize PCA result related to centrality measures</p></a></li>
<li><a href='#summary_tsne_centralities'><p>Summarize t-Distributed Stochastic Neighbor Embedding (t-SNE) on centrality measures</p></a></li>
<li><a href='#tsne_centralities'><p>t-Distributed Stochastic Neighbor Embedding (t-SNE) on centrality measures</p></a></li>
<li><a href='#visualize_association'><p>Pairwise association plot between centrality measures</p></a></li>
<li><a href='#visualize_correlations'><p>Correlation plot between centrality measures</p></a></li>
<li><a href='#visualize_dendrogram'><p>Dendrogram plot among centrality measures</p></a></li>
<li><a href='#visualize_graph'><p>Graph visualization based on a specific centrality measure</p></a></li>
<li><a href='#visualize_heatmap'><p>Heatmap plot between centrality measures</p></a></li>
<li><a href='#visualize_pair_correlation'><p>Pairwise correlation plot between two centrality measures</p></a></li>
<li><a href='#wiener_index_centrality'><p>Wiener Index Centrality</p></a></li>
<li><a href='#zachary'><p>Zachary Karate Club Network</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Deciphering Central Informative Nodes in Network Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-07</td>
</tr>
<tr>
<td>Description:</td>
<td>Computing, comparing, and demonstrating top informative centrality measures within a network. "CINNA: an R/CRAN package to decipher Central Informative Nodes in Network Analysis" provides a comprehensive overview of the package functionality Ashtiani et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty819">doi:10.1093/bioinformatics/bty819</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, network, sna, centiserve, FactoMineR, factoextra,
GGally, pheatmap, corrplot, dendextend, viridis, ggplot2,
Rtsne, plyr, circlize, intergraph, utils, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-08 09:06:13 UTC; minooashtiani</td>
</tr>
<tr>
<td>Author:</td>
<td>Minoo Ashtiani [aut, cre],
  Mehdi Mirzaie [ctb],
  Mohieddin Jafari [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Minoo Ashtiani &lt;ashtiani.minoo@gmail.com&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-08 16:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_centralities'>Centrality measure calculation</h2><span id='topic+calculate_centralities'></span>

<h3>Description</h3>

<p>This function computes multitude centrality measures of an igraph object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_centralities(x, except = NULL, include = NULL, weights = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_centralities_+3A_x">x</code></td>
<td>
<p>the component of a network as an igraph object</p>
</td></tr>
<tr><td><code id="calculate_centralities_+3A_except">except</code></td>
<td>
<p>A vector containing names of centrality measures which could be omitted from the calculations.</p>
</td></tr>
<tr><td><code id="calculate_centralities_+3A_include">include</code></td>
<td>
<p>A vector including names of centrality measures which should be computed.</p>
</td></tr>
<tr><td><code id="calculate_centralities_+3A_weights">weights</code></td>
<td>
<p>A character scalar specifying the edge attribute to use. (default=NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates various types of centrality measures which are applicable to the network topology
and returns the results as a list. In the &quot;except&quot; argument, you can specify centrality measures that are
not necessary to calculate.
</p>


<h3>Value</h3>

<p>A list containing centrality measure values. Each column indicates a centrality measure, and each row corresponds to a vertex.
The structure of the output is as follows:
- The list has named elements, where each element represents a centrality measure.
- The value of each element is a numeric vector, where each element of the vector corresponds to a vertex in the network.
- The order of vertices in the numeric vectors matches the order of vertices in the input igraph object.
- The class of the output is &quot;centrality&quot;.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Bonacich, P., &amp; Lloyd, P. (2001). Eigenvector like measures of centrality for asymmetric relations. Social Networks, 23(3), 191–201.
</p>
<p>Bonacich, P. (1972). Factoring and weighting approaches to status scores and clique identification. The Journal of Mathematical Sociology, 2(1), 113–120.
</p>
<p>Bonacich, P. (1987). Power and Centrality: A Family of Measures. American Journal of Sociology, 92(5), 1170–1182.
</p>
<p>Burt, R. S. (2004). Structural Holes and Good Ideas. American Journal of Sociology, 110(2), 349–399.
</p>
<p>Batagelj, V., &amp; Zaversnik, M. (2003). An O(m) Algorithm for Cores Decomposition of Networks, 1–9. Retrieved from
</p>
<p>Seidman, S. B. (1983). Network structure and minimum degree. Social Networks, 5(3), 269–287.
</p>
<p>Kleinberg, J. M. (1999). Authoritative sources in a hyperlinked environment. Journal of the ACM, 46(5), 604–632.
</p>
<p>Wasserman, S., &amp; Faust, K. (1994). Social network analysis : methods and applications. American Ethnologist (Vol. 24).
Barrat, A., Barthélemy, M., Pastor Satorras, R., &amp; Vespignani, A. (2004). The architecture of complex weighted networks. Proceedings of the National Academy of Sciences of the United States of America , 101(11), 3747–3752.
</p>
<p>Brin, S., &amp; Page, L. (2010). The Anatomy of a Large Scale Hypertextual Web Search Engine The Anatomy of a Search Engine. Search, 30(June 2000), 1–7.
</p>
<p>Freeman, L. C. (1978). Centrality in social networks conceptual clarification. Social Networks, 1(3), 215–239.
</p>
<p>Brandes, U. (2001). A faster algorithm for betweenness centrality*. The Journal of Mathematical Sociology, 25(2), 163–177.
</p>
<p>Estrada E., Rodriguez-Velazquez J. A.: Subgraph centrality in Complex Networks. Physical Review E 71, 056103.
</p>
<p>Freeman, L. C., Borgatti, S. P., &amp; White, D. R. (1991). Centrality in valued graphs: A measure of betweenness based on network flow. Social Networks, 13(2), 141–154.
</p>
<p>Brandes, U., &amp; Erlebach, T. (Eds.). (2005). Network Analysis (Vol. 3418). Berlin, Heidelberg: Springer Berlin Heidelberg.
</p>
<p>Stephenson, K., &amp; Zelen, M. (1989). Rethinking centrality: Methods and examples. Social Networks, 11(1), 1–37.
</p>
<p>Wasserman, S., &amp; Faust, K. (1994). Social network analysis : methods and applications. American Ethnologist (Vol. 24).
</p>
<p>Brandes, U. (2008). On variants of shortest path betweenness centrality and their generic computation. Social Networks, 30(2), 136–145.
</p>
<p>Goh, K.-I., Kahng, B., &amp; Kim, D. (2001). Universal Behavior of Load Distribution in Scale Free Networks. Physical Review Letters, 87(27), 278701.
</p>
<p>Shimbel, A. (1953). Structural parameters of communication networks. The Bulletin of Mathematical Biophysics, 15(4), 501–507.
</p>
<p>Assenov, Y., Ramrez, F., Schelhorn, S.-E., Lengauer, T., &amp; Albrecht, M. (2008). Computing topological parameters of biological networks. Bioinformatics, 24(2), 282–284.
</p>
<p>Diekert, V., &amp; Durand, B. (Eds.). (2005). STACS 2005 (Vol. 3404). Berlin, Heidelberg: Springer Berlin Heidelberg.
</p>
<p>Gräßler, J., Koschützki, D., &amp; Schreiber, F. (2012). CentiLib: comprehensive analysis and exploration of network centralities. Bioinformatics (Oxford, England), 28(8), 1178–9.
</p>
<p>Latora, V., &amp; Marchiori, M. (2001). Efficient Behavior of Small World Networks. Physical Review Letters, 87(19), 198701.
</p>
<p>Opsahl, T., Agneessens, F., &amp; Skvoretz, J. (2010). Node centrality in weighted networks: Generalizing degree and shortest paths. Social Networks, 32(3), 245–251.
</p>
<p>Estrada, E., Higham, D. J., &amp; Hatano, N. (2009). Communicability betweenness in complex networks. Physica A: Statistical Mechanics and Its Applications, 388(5), 764–774.
</p>
<p>Hagberg, Aric, Pieter Swart, and Daniel S Chult. Exploring network structure, dynamics, and function using NetworkX. No. LA-UR-08-05495; LA-UR-08-5495. Los Alamos National Laboratory (LANL), 2008.
</p>
<p>Kalinka, A. T., &amp; Tomancak, P. (2011). linkcomm: an R package for the generation, visualization, and analysis of link communities in networks of arbitrary size and type. Bioinformatics, 27(14), 2011–2012.
</p>
<p>Faghani, M. R., &amp; Nguyen, U. T. (2013). A Study of XSS Worm Propagation and Detection Mechanisms in Online Social Networks. IEEE Transactions on Information Forensics and Security, 8(11), 1815–1826.
</p>
<p>Brandes, U., &amp; Erlebach, T. (Eds.). (2005). Network Analysis (Vol. 3418). Berlin, Heidelberg: Springer Berlin Heidelberg.
</p>
<p>Lin, C.-Y., Chin, C.-H., Wu, H.-H., Chen, S.-H., Ho, C.-W., &amp; Ko, M.-T. (2008). Hubba: hub objects analyzer&ndash;a framework of interactome hubs identification for network biology. Nucleic Acids Research, 36(Web Server), W438–W443.
</p>
<p>Chin, C., Chen, S., &amp; Wu, H. (2009). cyto Hubba: A Cytoscape Plug in for Hub Object Analysis in Network Biology. Genome Informatics …, 5(Java 5), 2–3.
</p>
<p>Qi, X., Fuller, E., Wu, Q., Wu, Y., &amp; Zhang, C.-Q. (2012). Laplacian centrality: A new centrality measure for weighted networks. Information Sciences, 194, 240–253.
</p>
<p>Joyce, K. E., Laurienti, P. J., Burdette, J. H., &amp; Hayasaka, S. (2010). A New Measure of Centrality for Brain Networks. PLoS ONE, 5(8), e12200.
</p>
<p>Lin, C.-Y., Chin, C.-H., Wu, H.-H., Chen, S.-H., Ho, C.-W., &amp; Ko, M.-T. (2008). Hubba: hub objects analyzer&ndash;a framework of interactome hubs identification for network biology. Nucleic Acids Research, 36(Web Server), W438–W443.
</p>
<p>Hubbell, C. H. (1965). An Input Output Approach to Clique Identification. Sociometry, 28(4), 377.
</p>
<p>Dangalchev, C. (2006). Residual closeness in networks. Physica A: Statistical Mechanics and Its Applications, 365(2), 556–564.
</p>
<p>Brandes, U. &amp; Erlebach, T. (2005). Network Analysis: Methodological Foundations, U.S. Government Printing Office.
</p>
<p>Korn, A., Schubert, A., &amp; Telcs, A. (2009). Lobby index in networks. Physica A: Statistical Mechanics and Its Applications, 388(11), 2221–2226.
</p>
<p>White, S., &amp; Smyth, P. (2003). Algorithms for estimating relative importance in networks. In Proceedings of the ninth ACM SIGKDD international conference on Knowledge discovery and data mining   KDD ’03 (p. 266). New York, New York, USA: ACM Press.
</p>
<p>Cornish, A. J., &amp; Markowetz, F. (2014). SANTA: Quantifying the Functional Content of Molecular Networks. PLoS Computational Biology, 10(9), e1003808.
</p>
<p>Scardoni, G., Petterlini, M., &amp; Laudanna, C. (2009). Analyzing biological network parameters with CentiScaPe. Bioinformatics, 25(21), 2857–2859.
</p>
<p>Lin, N. (1976). Foundations of Social Research. Mcgraw Hill.
</p>
<p>Borgatti, S. P., &amp; Everett, M. G. (2006). A Graph theoretic perspective on centrality. Social Networks, 28(4), 466–484.
</p>
<p>Newman, M. (2010). Networks. Oxford University Press.
</p>
<p>Junker, Bjorn H., Dirk Koschutzki, and Falk Schreiber(2006). &quot;Exploration of biological network centralities with CentiBiN.&quot; BMC bioinformatics 7.1 : 219.
</p>
<p>Pal, S. K., Kundu, S., &amp; Murthy, C. A. (2014). Centrality measures, upper bound, and influence maximization in large scale directed social networks. Fundamenta Informaticae, 130(3), 317–342.
</p>
<p>Lin, C.-Y., Chin, C.-H., Wu, H.-H., Chen, S.-H., Ho, C.-W., &amp; Ko, M.-T. (2008). Hubba: hub objects analyzer&ndash;a framework of interactome hubs identification for network biology. Nucleic Acids Research, 36(Web Server), W438–W443.
</p>
<p>Scardoni, G., Petterlini, M., &amp; Laudanna, C. (2009). Analyzing biological network parameters with CentiScaPe. Bioinformatics, 25(21), 2857–2859.
</p>
<p>Freeman, L. C. (1978). Centrality in social networks conceptual clarification. Social Networks, 1(3), 215–239.
</p>
<p>Chen, D.-B., Gao, H., L?, L., &amp; Zhou, T. (2013). Identifying Influential Nodes in Large Scale Directed Networks: The Role of Clustering. PLoS ONE, 8(10), e77455.
</p>
<p>Jana Hurajova, S. G. and T. M. (2014). Decay Centrality. In 15th Conference of Kosice Mathematicians. Herlany.
</p>
<p>Viswanath, M. (2009). ONTOLOGY BASED AUTOMATIC TEXT SUMMARIZATION. Vishweshwaraiah Institute of Technology.
</p>
<p>Przulj, N., Wigle, D. A., &amp; Jurisica, I. (2004). Functional topology in a network of protein interactions. Bioinformatics, 20(3), 340–348.
</p>
<p>del Rio, G., Koschtzki, D., &amp; Coello, G. (2009). How to identify essential genes from molecular networks BMC Systems Biology, 3(1), 102.
</p>
<p>Scardoni, G. and Carlo Laudanna, C.B.M.C., 2011. Network centralities for Cytoscape. University of Verona.
</p>
<p>BOLDI, P. &amp; VIGNA, S. 2014. Axioms for centrality. Internet Mathematics, 00-00.
</p>
<p>MARCHIORI, M. &amp; LATORA, V. 2000. Harmony in the small-world. Physica A: Statistical Mechanics and its Applications, 285, 539-546.
</p>
<p>OPSAHL, T., AGNEESSENS, F. &amp; SKVORETZ, J. 2010. Node centrality in weighted networks: Generalizing degree and shortest paths. Social Networks, 32, 245-251.
</p>
<p>OPSAHL, T. 2010. Closeness centrality in networks with disconnected components (http://toreopsahl.com/2010/03/20/closeness-centrality-in-networks-with-disconnected-components/)
</p>
<p>Michalak, T.P., Aadithya, K.V., Szczepanski, P.L., Ravindran, B. and Jennings, N.R., 2013. Efficient computation of the Shapley value for game-theoretic network centrality. Journal of Artificial Intelligence Research, 46, pp.607-650.
</p>
<p>Macker, J.P., 2016, November. An improved local bridging centrality model for distributed network analytics. In Military Communications Conference, MILCOM 2016-2016 IEEE (pp. 600-605). IEEE. DOI: 10.1109/MILCOM.2016.7795393
</p>
<p>DANGALCHEV, C. 2006. Residual closeness in networks. Physica A: Statistical Mechanics and its Applications, 365, 556-564. DOI: 10.1016/j.physa.2005.12.020
</p>
<p>Alain Barrat, Marc Barthelemy, Romualdo Pastor-Satorras, Alessandro Vespignani: The architecture of complex weighted networks, Proc. Natl. Acad. Sci. USA 101, 3747 (2004)
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+alpha.centrality">alpha.centrality</a></code>, <code><a href="igraph.html#topic+bonpow">bonpow</a></code>, <code><a href="igraph.html#topic+constraint">constraint</a></code>,
<code><a href="igraph.html#topic+centr_degree">centr_degree</a></code>, <code><a href="igraph.html#topic+eccentricity">eccentricity</a></code>, <code><a href="igraph.html#topic+eigen_centrality">eigen_centrality</a></code>,
<code><a href="igraph.html#topic+coreness">coreness</a></code>, <code><a href="igraph.html#topic+authority_score">authority_score</a></code>, <code><a href="igraph.html#topic+hub_score">hub_score</a></code>,
<code><a href="igraph.html#topic+transitivity">transitivity</a></code>, <code><a href="igraph.html#topic+page_rank">page_rank</a></code>, <code><a href="igraph.html#topic+betweenness">betweenness</a></code>,
<code><a href="igraph.html#topic+subgraph.centrality">subgraph.centrality</a></code>, <code><a href="sna.html#topic+flowbet">flowbet</a></code>, <code><a href="sna.html#topic+infocent">infocent</a></code>,
<code><a href="sna.html#topic+loadcent">loadcent</a></code>, <code><a href="sna.html#topic+stresscent">stresscent</a></code>, <code><a href="sna.html#topic+graphcent">graphcent</a></code>, <code><a href="centiserve.html#topic+topocoefficient">topocoefficient</a></code>,
<code><a href="centiserve.html#topic+closeness.currentflow">closeness.currentflow</a></code>, <code><a href="centiserve.html#topic+closeness.latora">closeness.latora</a></code>,
<code><a href="centiserve.html#topic+communibet">communibet</a></code>, <code><a href="centiserve.html#topic+communitycent">communitycent</a></code>,
<code><a href="centiserve.html#topic+crossclique">crossclique</a></code>, <code><a href="centiserve.html#topic+entropy">entropy</a></code>,
<code><a href="centiserve.html#topic+epc">epc</a></code>, <code><a href="centiserve.html#topic+laplacian">laplacian</a></code>, <code><a href="centiserve.html#topic+leverage">leverage</a></code>,
<code><a href="centiserve.html#topic+mnc">mnc</a></code>, <code><a href="centiserve.html#topic+hubbell">hubbell</a></code>, <code><a href="centiserve.html#topic+semilocal">semilocal</a></code>,
<code><a href="centiserve.html#topic+closeness.vitality">closeness.vitality</a></code>,
<code><a href="centiserve.html#topic+closeness.residual">closeness.residual</a></code>, <code><a href="centiserve.html#topic+lobby">lobby</a></code>,
<code><a href="centiserve.html#topic+markovcent">markovcent</a></code>, <code><a href="centiserve.html#topic+radiality">radiality</a></code>, <code><a href="centiserve.html#topic+lincent">lincent</a></code>,
<code><a href="centiserve.html#topic+geokpath">geokpath</a></code>, <code><a href="centiserve.html#topic+katzcent">katzcent</a></code>, <code><a href="centiserve.html#topic+diffusion.degree">diffusion.degree</a></code>,
<code><a href="centiserve.html#topic+dmnc">dmnc</a></code>, <code><a href="centiserve.html#topic+centroid">centroid</a></code>, <code><a href="centiserve.html#topic+closeness.freeman">closeness.freeman</a></code>,
<code><a href="centiserve.html#topic+clusterrank">clusterrank</a></code>, <code><a href="centiserve.html#topic+decay">decay</a></code>,
<code><a href="centiserve.html#topic+barycenter">barycenter</a></code>, <code><a href="centiserve.html#topic+bottleneck">bottleneck</a></code>, <code><a href="centiserve.html#topic+averagedis">averagedis</a></code>,
<code><a href="#topic+local_bridging_centrality">local_bridging_centrality</a></code>, <code><a href="#topic+wiener_index_centrality">wiener_index_centrality</a></code>, <code><a href="#topic+group_centrality">group_centrality</a></code>,
<code><a href="#topic+dangalchev_closeness_centrality">dangalchev_closeness_centrality</a></code>, <code><a href="#topic+harmonic_centrality">harmonic_centrality</a></code>, <code><a href="igraph.html#topic+strength">strength</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("zachary")
p &lt;- proper_centralities(zachary)
calculate_centralities(zachary, include = "Degree Centrality")

</code></pre>

<hr>
<h2 id='cortex'>Macaque Visual Cortex Network</h2><span id='topic+cortex'></span>

<h3>Description</h3>

<p>A graph describing the macaque visual cortex network.
Nodes are neocortical areas, 25 of them are participated in visual function in the macaque,
and 7 of which are associated with them
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("cortex")
</code></pre>


<h3>Format</h3>

<p>an igraph object with &quot;gml&quot; format
</p>


<h3>References</h3>

<p>D.J. Felleman and D.C. van Essen, &quot;Distributed hierarchical processing in the primate cerebral cortex.&quot; Cerebral Cortex 1(1), 1-47 (1991).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("cortex")
print(cortex)
</code></pre>

<hr>
<h2 id='dangalchev_closeness_centrality'>Dangalchev Closeness Centrality</h2><span id='topic+dangalchev_closeness_centrality'></span>

<h3>Description</h3>

<p>This function computes the Dangalchev Closeness Centrality for nodes in a network.
The Dangalchev Closeness Centrality measures closeness by removing nodes and edges, allowing for easier evaluation and handling of unconnected graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dangalchev_closeness_centrality(
  x,
  vids = V(x),
  mode = c("all", "out", "in"),
  weights = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dangalchev_closeness_centrality_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="dangalchev_closeness_centrality_+3A_vids">vids</code></td>
<td>
<p>Nodes to be considered in the calculation.</p>
</td></tr>
<tr><td><code id="dangalchev_closeness_centrality_+3A_mode">mode</code></td>
<td>
<p>A character value indicating whether the shortest paths &quot;in&quot; or &quot;out&quot; of the nodes in directed graphs should be considered. For undirected graphs, use &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="dangalchev_closeness_centrality_+3A_weights">weights</code></td>
<td>
<p>A numeric vector indicating the weights of the edges.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector including the centrality values for each node.
The centrality values represent the Dangalchev Closeness Centrality measure for each node in the network.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Dangalchev, C. (2006). Residual closeness in networks. Physica A: Statistical Mechanics and its Applications, 365, 556-564. DOI: 10.1016/j.physa.2005.12.020
</p>


<h3>See Also</h3>

<p><code><a href="centiserve.html#topic+closeness.residual">closeness.residual</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

dangalchev_closeness_centrality(zachary)

</code></pre>

<hr>
<h2 id='drugTarget'>Drug Target Network</h2><span id='topic+drugTarget'></span>

<h3>Description</h3>

<p>A bipartite graph extracted from DrugBank 1.0 database. The network includes two set of nodes
including Food and Drug Administration (FDA)-approved drugs and their corresponding protein targets designated by their Uniprot ID.
The 1080 drugs and their 519 target proteins nodes are connected via 3766 interactions.
Please note that it is a shrunken network in which metabolizing enzymes,
carriers and transporters associated with drug metabolism are filtered and solely
targets directly related to their pharmacological effects are included.
It is also an example of unconnected graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugTarget
</code></pre>


<h3>Format</h3>

<p>an igraph object with &quot;gml&quot; format
</p>


<h3>References</h3>

<p>Barneh, F., Jafari, M., &amp; Mirzaie, M. (2015). Updates on drug–target network; facilitating polypharmacology and data integration by growth of DrugBank database. Briefings in Bioinformatics, bbv094. https://doi.org/10.1093/bib/bbv094
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("drugTarget")
print(drugTarget)
</code></pre>

<hr>
<h2 id='giant_component_extract'>Giant component extraction of a graph</h2><span id='topic+giant_component_extract'></span>

<h3>Description</h3>

<p>This function extracts the largest connected component or the giant component
of the input graph, which can be an &quot;igraph&quot; object or a &quot;network&quot; object, and converts
it to an &quot;igraph&quot; object. For bipartite graphs, this function can perform projection
before extracting the components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>giant_component_extract(
  x,
  directed = TRUE,
  bipartite_proj = FALSE,
  num_proj = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="giant_component_extract_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="giant_component_extract_+3A_directed">directed</code></td>
<td>
<p>Whether to create a directed graph. (default = TRUE)</p>
</td></tr>
<tr><td><code id="giant_component_extract_+3A_bipartite_proj">bipartite_proj</code></td>
<td>
<p>Whether the bipartite network must be projected or not. (default = FALSE)</p>
</td></tr>
<tr><td><code id="giant_component_extract_+3A_num_proj">num_proj</code></td>
<td>
<p>A number indicating the specific projection to work with, especially for
bipartite graphs. (default = 1)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function identifies the largest component of an &quot;igraph&quot; or a &quot;network&quot; object
and returns it as a list containing the giant component as an igraph object and its
corresponding edgelist. If the input graph is bipartite and the &quot;bipartite_proj&quot; parameter
is set to TRUE, the function can perform projection before extracting the components. You
can specify which projection to work with using the &quot;num_proj&quot; parameter.
</p>


<h3>Value</h3>

<p>A list containing the giant component of the input graph. The first element is an igraph object
representing the giant component, and the second element is the corresponding edgelist.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Newman, M. (2010). Networks. Oxford University Press.
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+induced.subgraph">induced.subgraph</a></code>, <code><a href="igraph.html#topic+clusters">clusters</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A graph with 4 vertices

data(zachary)
giant_component_extract(zachary)

</code></pre>

<hr>
<h2 id='graph_extract_components'>Component extraction of a graph</h2><span id='topic+graph_extract_components'></span>

<h3>Description</h3>

<p>This function extracts all connected components of the
input, which can be an &quot;igraph&quot; object or a &quot;network&quot; object,
and converts them to &quot;igraph&quot; objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph_extract_components(
  x,
  directed = TRUE,
  bipartite_proj = FALSE,
  num_proj = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph_extract_components_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="graph_extract_components_+3A_directed">directed</code></td>
<td>
<p>Whether to create a directed graph (default = TRUE).</p>
</td></tr>
<tr><td><code id="graph_extract_components_+3A_bipartite_proj">bipartite_proj</code></td>
<td>
<p>Whether the bipartite network must be projected or not (default = FALSE).</p>
</td></tr>
<tr><td><code id="graph_extract_components_+3A_num_proj">num_proj</code></td>
<td>
<p>Numbers 1 or 2 that show the number of projects for bipartite graphs (default = 1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function separates different components of an &quot;igraph&quot; or a &quot;network&quot; object and
illustrates them as a list of independent graphs. If the input graph was bipartite and the
&quot;bipartite_proj&quot; was TRUE, it will project it, and you can decide in which project you want
to continue working with.
</p>


<h3>Value</h3>

<p>A list including the components of the input as igraph objects.
The output is a list where each element represents a component and is an igraph object.
The components are disconnected subgraphs of the input graph.
</p>
<p>A list including the components of the input as igraph objects.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+induced.subgraph">induced.subgraph</a></code>, <code><a href="igraph.html#topic+components">components</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

graph_extract_components(zachary)

</code></pre>

<hr>
<h2 id='group_centrality'>Group Centrality</h2><span id='topic+group_centrality'></span>

<h3>Description</h3>

<p>This function computes the Group Centrality for nodes in a network. Group centrality considers a consistent ranking of each node to be calculated, taking into account the diverse possible synergies among possible groups of vertices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_centrality(x, vids = V(x))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_centrality_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="group_centrality_+3A_vids">vids</code></td>
<td>
<p>Nodes to be considered in the calculation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of centrality values for each node. The length of the vector is equal to the number of nodes in the network.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Michalak, T.P., Aadithya, K.V., Szczepanski, P.L., Ravindran, B. and Jennings, N.R., 2013. Efficient computation of the Shapley value for game-theoretic network centrality. Journal of Artificial Intelligence Research, 46, pp.607-650.
</p>
<p>https://www.civilica.com/Paper-IBIS07-IBIS07_127.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

group_centrality(zachary)

</code></pre>

<hr>
<h2 id='harmonic_centrality'>Harmonic Centrality</h2><span id='topic+harmonic_centrality'></span>

<h3>Description</h3>

<p>This function computes the Harmonic Centrality for nodes in a network. The harmonic centrality metric is defined as the denormalized reciprocal of the harmonic mean of all distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonic_centrality(
  x,
  vids = V(x),
  mode = c("all", "out", "in"),
  weights = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="harmonic_centrality_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="harmonic_centrality_+3A_vids">vids</code></td>
<td>
<p>Nodes to be considered in the calculation.</p>
</td></tr>
<tr><td><code id="harmonic_centrality_+3A_mode">mode</code></td>
<td>
<p>A character value, indicating the type of degree to consider (&quot;out&quot; for out-degree, &quot;in&quot; for in-degree, &quot;total&quot; for the sum of the two). For undirected graphs, this argument is ignored. The default value is &quot;total&quot;.</p>
</td></tr>
<tr><td><code id="harmonic_centrality_+3A_weights">weights</code></td>
<td>
<p>Numeric vector indicating weights of the edges.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of centrality values for each node. The length of the vector is equal to the number of nodes in the network.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>BOLDI, P. &amp; VIGNA, S. 2014. Axioms for centrality. Internet Mathematics, 00-00.
</p>
<p>MARCHIORI, M. &amp; LATORA, V. 2000. Harmony in the small-world. Physica A: Statistical Mechanics and its Applications, 285, 539-546.
</p>
<p>OPSAHL, T., AGNEESSENS, F. &amp; SKVORETZ, J. 2010. Node centrality in weighted networks: Generalizing degree and shortest paths. Social Networks, 32, 245-251.
</p>
<p>OPSAHL, T. 2010. Closeness centrality in networks with disconnected components (http://toreopsahl.com/2010/03/20/closeness-centrality-in-networks-with-disconnected-components/)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

harmonic_centrality(zachary)

</code></pre>

<hr>
<h2 id='kangaroo'>Kangaroo Network</h2><span id='topic+kangaroo'></span>

<h3>Description</h3>

<p>An undirected graph based on interactions between free-ranging grey kangaroos.
A node displays a kangaroo and an edge between two kangaroos demonstrates an interaction.
The weights indicate the total count of interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kangaroo
</code></pre>


<h3>Format</h3>

<p>an igraph object with &quot;gml&quot; format
</p>


<h3>References</h3>

<p>Kangaroo network dataset &ndash; KONECT, October 2016.
</p>
<p>TR Grant. Dominance and association among members of a captive and a free-ranging group of grey kangaroos (Macropus giganteus). Animal Behaviour, 21(3):449&ndash;456, 1973.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("kangaroo")
print(kangaroo)
</code></pre>

<hr>
<h2 id='local_bridging_centrality'>Local Bridging Centrality</h2><span id='topic+local_bridging_centrality'></span>

<h3>Description</h3>

<p>This function computes the Local Bridging Centrality for nodes in a network. The Local Bridging Centrality classifies nodes based on their structural links among the dense components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>local_bridging_centrality(x, vids = V(x))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="local_bridging_centrality_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="local_bridging_centrality_+3A_vids">vids</code></td>
<td>
<p>Nodes to be considered in the calculation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector including the centrality values for each node.
The centrality values represent the Local Bridging Centrality measure for each node in the network.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Macker, J.P. (2016). An improved local bridging centrality model for distributed network analytics. In Military Communications Conference, MILCOM 2016-2016 IEEE (pp. 600-605). IEEE. DOI: 10.1109/MILCOM.2016.7795393
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+betweenness">betweenness</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

local_bridging_centrality(zachary)

</code></pre>

<hr>
<h2 id='misc_extract_components'>Component extraction of miscellaneous graph formats</h2><span id='topic+misc_extract_components'></span>

<h3>Description</h3>

<p>This function extracts all components of the input with various formats
and converts them to &quot;igraph&quot; objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>misc_extract_components(
  x,
  directed = TRUE,
  mode = "directed",
  weighted = NULL,
  unibipartite = FALSE,
  diag = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="misc_extract_components_+3A_x">x</code></td>
<td>
<p>The input can be an edgelist, an adjacency matrix, or a graphNEL object.</p>
</td></tr>
<tr><td><code id="misc_extract_components_+3A_directed">directed</code></td>
<td>
<p>Whether to create a directed graph. (default = TRUE)</p>
</td></tr>
<tr><td><code id="misc_extract_components_+3A_mode">mode</code></td>
<td>
<p>Character scalar, explains how to interpret the supplied matrix.
Possible values are: &quot;directed&quot;, &quot;undirected&quot;, &quot;upper&quot;, &quot;lower&quot;, &quot;max&quot;, &quot;min&quot;, &quot;plus&quot;. (default = &quot;directed&quot;)</p>
</td></tr>
<tr><td><code id="misc_extract_components_+3A_weighted">weighted</code></td>
<td>
<p>An argument for specifying whether the graph should be weighted or not.
If it is NULL, then an unweighted graph is created. (default = NULL)</p>
</td></tr>
<tr><td><code id="misc_extract_components_+3A_unibipartite">unibipartite</code></td>
<td>
<p>A boolean parameter describing whether the input edge list corresponds
to a bipartite graph. A TRUE value specifies a bipartite graph, and vice versa. (default = FALSE)</p>
</td></tr>
<tr><td><code id="misc_extract_components_+3A_diag">diag</code></td>
<td>
<p>Logical scalar, whether to consider the diagonal of the matrix or not.
If it is FALSE, then the diagonal is treated as zeros. (default = TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts components from the input object, which can be an edgelist,
an adjacency matrix, or a graphNEL object. The result is a list including the components
represented as separate graphs.
</p>


<h3>Value</h3>

<p>A list including the components of the input graph as igraph objects.
Each element of the list represents a component and is an igraph object.
The components are disconnected subgraphs of the input graph.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+induced.subgraph">induced.subgraph</a></code>, <code><a href="igraph.html#topic+components">components</a></code>,
<code><a href="igraph.html#topic+graph_from_adjacency_matrix">graph_from_adjacency_matrix</a></code>
</p>

<hr>
<h2 id='pca_centralities'>PCA Centrality Measures</h2><span id='topic+pca_centralities'></span>

<h3>Description</h3>

<p>This function performs Principal Component Analysis (PCA) on centrality measures.
It computes the contributions of variables to the principal components and visualizes them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_centralities(
  x,
  scale.unit = TRUE,
  cut.off = 80,
  ncp = 2,
  graph = FALSE,
  axes = c(1, 2)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_centralities_+3A_x">x</code></td>
<td>
<p>a list containing the computed centrality values</p>
</td></tr>
<tr><td><code id="pca_centralities_+3A_scale.unit">scale.unit</code></td>
<td>
<p>a boolean constant indicating whether data should be scaled to unit variance (default = TRUE)</p>
</td></tr>
<tr><td><code id="pca_centralities_+3A_cut.off">cut.off</code></td>
<td>
<p>The intensity that must be exceeded in cumulative percentage of variance of eigen values (default = 80)</p>
</td></tr>
<tr><td><code id="pca_centralities_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions in the final results (default = 5)</p>
</td></tr>
<tr><td><code id="pca_centralities_+3A_graph">graph</code></td>
<td>
<p>a boolean constant indicating whether the graph should be displayed (default = FALSE)</p>
</td></tr>
<tr><td><code id="pca_centralities_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector describing the number of components to plot (default = c(1, 2))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot illustrating the contributions of variables to the principal components.
The x-axis represents the centrality measures, and the y-axis represents the contribution.
The higher the contribution value, the more important the centrality measure is in the ranking.
The plot helps in identifying the most influential centrality measures.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a data frame with multiple observations
centralities &lt;- data.frame(
  Betweenness = c(0.2, 0.3, 0.5),
 Closeness = c(0.4, 0.2, 0.6),
  Degree = c(0.3, 0.1, 0.4),
  Eigenvector = c(0.1, 0.5, 0.2)
)
pca_centralities(centralities)

</code></pre>

<hr>
<h2 id='proper_centralities'>Proper centrality measure representation</h2><span id='topic+proper_centralities'></span>

<h3>Description</h3>

<p>This function indicates the proper centrality measures of an igraph object
based on the network topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proper_centralities(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="proper_centralities_+3A_x">x</code></td>
<td>
<p>An igraph object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a list including the names of centrality measures that are applicable
for the input graph based on its topology.
</p>


<h3>Value</h3>

<p>A list including the names of centrality measures that are suitable for the input graph.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculate_centralities">calculate_centralities</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("zachary")
proper_centralities(zachary)

</code></pre>

<hr>
<h2 id='rhesus'>Moreno Rhesus Network</h2><span id='topic+rhesus'></span>

<h3>Description</h3>

<p>A directed graph including observed grooming episodes between
free ranging rhesus macaques (Macaca mulatta) in Cayo Santiago
during a two month period in 1963. Cayo Santiago is an island off the coast of Puerto Rico,
which also is named as Isla de los monos (Island of the monkeys).
A node indicates a monkey and a directed edge in which
a rhesus macaque groomed another rhesus macaque.
The weights of edges demonstrates how often this behaviour
was seen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhesus
</code></pre>


<h3>Format</h3>

<p>an igraph object with &quot;gml&quot; format
</p>


<h3>References</h3>

<p>Rhesus network dataset &ndash; KONECT, October 2016.
</p>
<p>DS Sade. Sociometrics of macaca mulatta I. linkages and cliques in grooming matrices. Folia Primatologica, 18(3-4):196&ndash;223, 1972.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("rhesus")
print(rhesus)
</code></pre>

<hr>
<h2 id='summary_calculate_centralities'>Summarize centrality measure calculation results</h2><span id='topic+summary_calculate_centralities'></span>

<h3>Description</h3>

<p>This function computes the minimum, first quartile, median,
mean, third quartile, and maximum values of the computed centrality measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_calculate_centralities(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_calculate_centralities_+3A_x">x</code></td>
<td>
<p>Centrality measure calculation results, typically a numeric vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list summarizing the computed centrality measures. The list includes the following elements:
- &quot;minimum&quot;: The minimum value of the centrality measures.
- &quot;first_quartile&quot;: The first quartile (25th percentile) value of the centrality measures.
- &quot;median&quot;: The median (50th percentile) value of the centrality measures.
- &quot;mean&quot;: The mean value of the centrality measures.
- &quot;third_quartile&quot;: The third quartile (75th percentile) value of the centrality measures.
- &quot;maximum&quot;: The maximum value of the centrality measures.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>

<hr>
<h2 id='summary_graph_extract_components'>Summarize component extraction of a graph</h2><span id='topic+summary_graph_extract_components'></span>

<h3>Description</h3>

<p>This function summarizes all components of the input which can be an &quot;igraph&quot; object or a &quot;network&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_graph_extract_components(
  x,
  directed = TRUE,
  bipartite_proj = FALSE,
  num_proj = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_graph_extract_components_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="summary_graph_extract_components_+3A_directed">directed</code></td>
<td>
<p>A boolean value indicating whether to create a directed graph (default = TRUE).</p>
</td></tr>
<tr><td><code id="summary_graph_extract_components_+3A_bipartite_proj">bipartite_proj</code></td>
<td>
<p>A boolean value indicating whether the bipartite network should be projected (default = FALSE).</p>
</td></tr>
<tr><td><code id="summary_graph_extract_components_+3A_num_proj">num_proj</code></td>
<td>
<p>A number indicating the number of projects specifically for bipartite graphs (default = 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of igraph objects representing the extracted components of the graph.
Each element in the list corresponds to a component, and each component is represented by an igraph object.
The components are extracted based on the connectivity of the graph and can include single nodes, disconnected subgraphs, or connected subgraphs.
If the graph is directed and the <code>directed</code> parameter is set to FALSE, the function returns the weakly connected components.
If the graph is bipartite and the <code>bipartite_proj</code> parameter is set to TRUE, the function returns the projected graph components.
If the graph is neither directed nor bipartite, the function returns the connected components.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>

<hr>
<h2 id='summary_pca_centralities'>Summarize PCA result related to centrality measures</h2><span id='topic+summary_pca_centralities'></span>

<h3>Description</h3>

<p>This function summarizes the PCA result related to centrality measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_pca_centralities(x, scale.unit = TRUE, ncp = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_pca_centralities_+3A_x">x</code></td>
<td>
<p>A list containing the computed centrality values.</p>
</td></tr>
<tr><td><code id="summary_pca_centralities_+3A_scale.unit">scale.unit</code></td>
<td>
<p>A boolean value indicating whether the data should be scaled to unit variance (default = TRUE).</p>
</td></tr>
<tr><td><code id="summary_pca_centralities_+3A_ncp">ncp</code></td>
<td>
<p>The number of dimensions in the final results (default = 5).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of the <code>pca_centralities</code> function, which includes the PCA analysis results such as eigenvalues, variance explained, and scores.
The returned value is an object of class &quot;PCA&quot; from the FactoMineR package. It contains the following components:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>A numeric vector of eigenvalues, indicating the amount of variance explained by each principal component.</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>A numeric vector of proportions of variance explained by each principal component.</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>A data frame of individual scores, where each row represents an individual and each column represents a principal component.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>The function call used to create the PCA object.</p>
</td></tr>
<tr><td><code>call2</code></td>
<td>
<p>The call used to compute the PCA analysis.</p>
</td></tr>
<tr><td><code>call3</code></td>
<td>
<p>The call used to project the individuals.</p>
</td></tr>
<tr><td><code>svd</code></td>
<td>
<p>The singular value decomposition of the data matrix.</p>
</td></tr>
<tr><td><code>cor</code></td>
<td>
<p>The correlation matrix of the variables.</p>
</td></tr>
<tr><td><code>cos2</code></td>
<td>
<p>The squared cosines of the variables.</p>
</td></tr>
<tr><td><code>contrib</code></td>
<td>
<p>The contributions of the variables to the principal components.</p>
</td></tr>
<tr><td><code>proj</code></td>
<td>
<p>The projected coordinates of the individuals on the principal components.</p>
</td></tr>
<tr><td><code>quali.sup</code></td>
<td>
<p>The results of the supplementary qualitative variables analysis.</p>
</td></tr>
<tr><td><code>quali.sup.ind</code></td>
<td>
<p>The results of the supplementary individuals analysis.</p>
</td></tr>
<tr><td><code>quanti.sup</code></td>
<td>
<p>The results of the supplementary quantitative variables analysis.</p>
</td></tr>
<tr><td><code>quanti.sup.var</code></td>
<td>
<p>The results of the supplementary variables analysis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>

<hr>
<h2 id='summary_tsne_centralities'>Summarize t-Distributed Stochastic Neighbor Embedding (t-SNE) on centrality measures</h2><span id='topic+summary_tsne_centralities'></span>

<h3>Description</h3>

<p>This function summarizes the t-SNE analysis results on centrality measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_tsne_centralities(x, dims = 2, perplexity = 5, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_tsne_centralities_+3A_x">x</code></td>
<td>
<p>A list containing the computed centrality values.</p>
</td></tr>
<tr><td><code id="summary_tsne_centralities_+3A_dims">dims</code></td>
<td>
<p>An integer specifying the number of output dimensions (default = 2).</p>
</td></tr>
<tr><td><code id="summary_tsne_centralities_+3A_perplexity">perplexity</code></td>
<td>
<p>A numeric value representing a flexible measure of the efficient number of neighbors.
The performance of t-SNE is fairly robust to changes in perplexity, and typical values are between 5 and 50 (default = 5).</p>
</td></tr>
<tr><td><code id="summary_tsne_centralities_+3A_scale">scale</code></td>
<td>
<p>A logical value indicating whether the centrality values should be scaled or not (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following elements:
- &quot;Y&quot;: A matrix containing the new representations for the objects after t-SNE.
- &quot;costs&quot;: The cost for every object after the final iteration of t-SNE.
The &quot;costs&quot; provide information about the optimization process.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>See Also</h3>

<p><code><a href="Rtsne.html#topic+Rtsne">Rtsne</a></code>
</p>

<hr>
<h2 id='tsne_centralities'>t-Distributed Stochastic Neighbor Embedding (t-SNE) on centrality measures</h2><span id='topic+tsne_centralities'></span>

<h3>Description</h3>

<p>This function applies t-SNE, a dimensionality reduction algorithm, to centrality measures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsne_centralities(x, dims = 2, perplexity = 5, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsne_centralities_+3A_x">x</code></td>
<td>
<p>A list containing the computed centrality values.</p>
</td></tr>
<tr><td><code id="tsne_centralities_+3A_dims">dims</code></td>
<td>
<p>An integer specifying the number of output dimensions (default = 2).</p>
</td></tr>
<tr><td><code id="tsne_centralities_+3A_perplexity">perplexity</code></td>
<td>
<p>A numeric value representing a flexible measure of the efficient number of neighbors.
The performance of t-SNE is fairly robust to changes in perplexity, and typical values are between 5 and 50 (default = 5).</p>
</td></tr>
<tr><td><code id="tsne_centralities_+3A_scale">scale</code></td>
<td>
<p>A logical value indicating whether the centrality values should be scaled or not (default = TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>t-SNE is a non-linear dimensionality reduction algorithm used for exploring high-dimensional data. It maps multi-dimensional centrality measure data to a lower-dimensional space suitable for analysis and visualization.
</p>


<h3>Value</h3>

<p>A cost plot of t-SNE results, which displays centralities in order of their corresponding costs.
The cost plot provides information about the optimization process and the quality of the embedding.
</p>
<p>A cost plot of t-SNE results, which displays centralities in order of their corresponding costs.
The cost plot is a ggplot object that represents the optimization process and the quality of the embedding.
The x-axis represents the iterations of the t-SNE algorithm, and the y-axis represents the cost associated with each iteration.
The cost measures the discrepancy between the original high-dimensional space and the low-dimensional embedding.
By examining the cost plot, you can assess the convergence and stability of the t-SNE algorithm and evaluate the quality of the embedding.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>van der Maaten, L. (2014). Accelerating t-SNE using Tree-Based Algorithms. Journal of Machine Learning Research, 15, 3221–3245.
Van Der Maaten, L. J. P., &amp; Hinton, G. E. (2008). Visualizing High-Dimensional Data Using t-SNE. Journal of Machine Learning Research, 9, 2579–2605.
</p>


<h3>See Also</h3>

<p><code><a href="Rtsne.html#topic+Rtsne">Rtsne</a></code>
</p>

<hr>
<h2 id='visualize_association'>Pairwise association plot between centrality measures</h2><span id='topic+visualize_association'></span>

<h3>Description</h3>

<p>This function computes regression between pair of centrality measures
to show more details of association among them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_association(x, y, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_association_+3A_x">x</code></td>
<td>
<p>a vector containing a centrality measure as independent variable</p>
</td></tr>
<tr><td><code id="visualize_association_+3A_y">y</code></td>
<td>
<p>a vector containing a centrality measure as dependent variable</p>
</td></tr>
<tr><td><code id="visualize_association_+3A_scale">scale</code></td>
<td>
<p>Whether the centrality values should be scaled or not</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function applies regression analysis on two different centrality values in order to find out the corresponding association
between them. Regression analysis is a kind of statitiscal method for approximation the association
between variables.It asserts that the value of dependent variable changes when the
value of independent variable varies.
</p>


<h3>Value</h3>

<p>The function returns a regression plot and the results of the regression analysis between two centrality measures.
The function takes two vectors, 'x' and 'y', which represent the independent and dependent centrality measures, respectively.
If the 'scale' parameter is set to 'TRUE', the centrality values will be scaled before performing the regression analysis.
The function applies regression analysis to estimate the association between the two centrality measures.
It creates a scatter plot of the centrality values, with a regression line representing the relationship between the variables.
Additionally, the plot includes confidence intervals around the regression line to show the uncertainty of the estimates.
The function returns the regression plot and the computed regression results.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Chambers, J. M. (1992). Statistical Models in S. Wadsworth. Pacific Grove, California. Retrieved from
[Link to the reference]
</p>
<p>Wilkinson, G. N., &amp; Rogers, C. E. (1973). Symbolic Description of Factorial Models for Analysis of Variance. Applied Statistics, 22(3), 392.
</p>

<hr>
<h2 id='visualize_correlations'>Correlation plot between centrality measures</h2><span id='topic+visualize_correlations'></span>

<h3>Description</h3>

<p>This function draw correlation plot between pair of centrality measures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_correlations(x, scale = TRUE, method = "pearson")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_correlations_+3A_x">x</code></td>
<td>
<p>a list indicating calculated centrality measures</p>
</td></tr>
<tr><td><code id="visualize_correlations_+3A_scale">scale</code></td>
<td>
<p>Whether the centrality values should be scaled or not(default=TRUE)</p>
</td></tr>
<tr><td><code id="visualize_correlations_+3A_method">method</code></td>
<td>
<p>a character string describing the type of correlation coefficient (or covariance)
to be computed. The proper values are &quot;pearson&quot;, &quot;kendall&quot;, or &quot;spearman&quot;. (default=&quot;pearson&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function illustrates pairwise correlation plot of computed centrality measures.
The names of centralities shown in the result plot is abbreviated and compelete names
can be seen in &quot;proper_centralities&quot; function.
Colors from red to blue indicate the intensity of correlation value. If two centrality
measures have an inverse relationship then their correspnding color in plot have to be red
and vice versa.
</p>


<h3>Value</h3>

<p>The function generates a correlation plot between pairs of centrality measures.
The function takes a list x as input, which contains the computed centrality measures.
If the scale parameter is set to TRUE, the centrality values will be scaled before computing the correlations.
The method parameter specifies the type of correlation coefficient to be computed: &quot;pearson&quot; (default), &quot;kendall&quot;, or &quot;spearman&quot;.
The function creates a pairwise correlation plot, where each cell represents the correlation between a pair of centrality measures.
The color of each cell indicates the strength and direction of the correlation, ranging from red (negative correlation) to blue (positive correlation).
The function returns the pairwise correlation plot.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>See Also</h3>

<p><code><a href="GGally.html#topic+ggpairs">ggpairs</a></code>
</p>

<hr>
<h2 id='visualize_dendrogram'>Dendrogram plot among centrality measures</h2><span id='topic+visualize_dendrogram'></span>

<h3>Description</h3>

<p>This function demonstrates the vertice dendrogram of a graph based
on a centrality type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_dendrogram(
  x,
  centrality.type = "Degree Centrality",
  computed_centrality_value = NULL,
  k = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_dendrogram_+3A_x">x</code></td>
<td>
<p>an igraph object</p>
</td></tr>
<tr><td><code id="visualize_dendrogram_+3A_centrality.type">centrality.type</code></td>
<td>
<p>The type of centrality which should be considered.(default=&quot;Degree Centrality&quot;)</p>
</td></tr>
<tr><td><code id="visualize_dendrogram_+3A_computed_centrality_value">computed_centrality_value</code></td>
<td>
<p>A vector containing the values of calculated centrality measure for each node.(default=NULL)</p>
</td></tr>
<tr><td><code id="visualize_dendrogram_+3A_k">k</code></td>
<td>
<p>number of clusters(default=4)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function generates a dendrogram plot of a graph's vertices based on a centrality measure.
The function takes an igraph object 'x' as input, representing the graph.
The 'centrality.type' parameter specifies the type of centrality measure to be considered (default is &quot;Degree Centrality&quot;).
If the 'computed_centrality_value' parameter is provided, it should be a vector containing the computed centrality values for each node.
Otherwise, the function will compute the specified centrality measure for the graph.
The 'k' parameter specifies the number of clusters to be formed from the dendrogram (default is 4).
The function creates a dendrogram plot of the vertices, where the branching structure represents the hierarchical clustering based on the centrality measure.
The function returns the dendrogram plot.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Galili, T. (2015). dendextend: an R package for visualizing, adjusting and comparing trees of hierarchical clustering. Bioinformatics, 31(22), 3718–3720.
</p>

<hr>
<h2 id='visualize_graph'>Graph visualization based on a specific centrality measure</h2><span id='topic+visualize_graph'></span>

<h3>Description</h3>

<p>This function demonstrates the input graph in which the size of nodes
indicates calculated centrality value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_graph(
  x,
  computed_centrality_value = NULL,
  centrality.type = "Degree Centrality"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_graph_+3A_x">x</code></td>
<td>
<p>an igraph object</p>
</td></tr>
<tr><td><code id="visualize_graph_+3A_computed_centrality_value">computed_centrality_value</code></td>
<td>
<p>A vector containing the values of calculated centrality measure for each node.</p>
</td></tr>
<tr><td><code id="visualize_graph_+3A_centrality.type">centrality.type</code></td>
<td>
<p>The type of centrality which should be calculated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function represents the graph in which size of nodes are based on computed centrality value. If the values of wanted centrality
measure were computed then by placing them in computed_centrality_value argument to use it for drawing the plot. Otherwise, by only giving the
name of favorite centrality measure in centrality.type argument, this function will calculate it and  then demonstrates the corresponding graph.
</p>


<h3>Value</h3>

<p>A plot illustrating the input graph, where the size of nodes represents the calculated centrality values.
The function takes an igraph object as input and either a vector of precomputed centrality values or the name of the desired centrality measure.
If precomputed centrality values are provided in the computed_centrality_value argument, the graph will be plotted based on those values.
If only the name of the desired centrality measure is provided in the centrality.type argument, the function will calculate the centrality values
and then plot the graph.
The plot shows the nodes of the graph, with their sizes indicating the centrality values. The larger the node, the higher the centrality value.
This plot helps visualize the distribution of centrality values across the nodes of the graph.
The function returns the plot and does not modify the input graph.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>

<hr>
<h2 id='visualize_heatmap'>Heatmap plot between centrality measures</h2><span id='topic+visualize_heatmap'></span>

<h3>Description</h3>

<p>This function draws heatmap between pair of centrality measures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_heatmap(x, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_heatmap_+3A_x">x</code></td>
<td>
<p>a list indicating calculated centrality measures</p>
</td></tr>
<tr><td><code id="visualize_heatmap_+3A_scale">scale</code></td>
<td>
<p>Whether the centrality values should be scaled or not</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function illustrates the heatmap plot of computed centrality measures.
</p>


<h3>Value</h3>

<p>The function generates a heatmap plot between pairs of centrality measures.
The function takes a list x as input, which contains the computed centrality measures.
If the scale parameter is set to TRUE, the centrality values will be scaled before plotting the heatmap.
The function creates a heatmap plot, where each cell represents the correlation between a pair of centrality measures.
The color of each cell indicates the strength and direction of the correlation.
The function returns the heatmap plot.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>

<hr>
<h2 id='visualize_pair_correlation'>Pairwise correlation plot between two centrality measures</h2><span id='topic+visualize_pair_correlation'></span>

<h3>Description</h3>

<p>This function computes and plots correlation between pair of centrality
measures and histogram plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_pair_correlation(x, y, scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_pair_correlation_+3A_x">x</code></td>
<td>
<p>a vector containing a centrality measure</p>
</td></tr>
<tr><td><code id="visualize_pair_correlation_+3A_y">y</code></td>
<td>
<p>a vector containing another centrality measure</p>
</td></tr>
<tr><td><code id="visualize_pair_correlation_+3A_scale">scale</code></td>
<td>
<p>Whether the centrality values should be scaled or not</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function illustrates the correlation value between two centrality measures and their
corresponding scatterplot and histograms.
</p>


<h3>Value</h3>

<p>The function computes and plots the correlation between two centrality measures.
The function takes two vectors, 'x' and 'y', representing the centrality measures to be correlated.
If the 'scale' parameter is set to 'TRUE', the centrality values will be scaled before computing the correlation.
The function creates a correlation plot, which includes a scatterplot of the centrality values and histograms of each measure.
The correlation coefficient between the two measures is displayed on the plot.
The function returns the correlation plot.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Emerson, J. W., Green, W. A., Schloerke, B., Crowley, J., Cook, D., Hofmann, H., &amp; Wickham, H. (2013). The Generalized Pairs Plot. Journal of Computational and Graphical Statistics, 22(1), 79–91.
</p>


<h3>See Also</h3>

<p><code><a href="GGally.html#topic+ggpairs">ggpairs</a></code>
</p>

<hr>
<h2 id='wiener_index_centrality'>Wiener Index Centrality</h2><span id='topic+wiener_index_centrality'></span>

<h3>Description</h3>

<p>This function computes the Wiener Index Centrality for nodes in a network. The Wiener index is a measure of centrality that computes the sum of all shortest paths between a node and all other related nodes in the graph. In essence, it is similar to closeness centrality, but without taking the reciprocal, resulting in a measure with the opposite meaning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wiener_index_centrality(
  x,
  vids = V(x),
  mode = c("all", "out", "in"),
  weights = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wiener_index_centrality_+3A_x">x</code></td>
<td>
<p>An igraph or a network object.</p>
</td></tr>
<tr><td><code id="wiener_index_centrality_+3A_vids">vids</code></td>
<td>
<p>Nodes to be considered in the calculation.</p>
</td></tr>
<tr><td><code id="wiener_index_centrality_+3A_mode">mode</code></td>
<td>
<p>A character value indicating whether the shortest paths &quot;in&quot; or &quot;out&quot; of the nodes in directed graphs should be considered. For undirected graphs, &quot;all&quot; is used.</p>
</td></tr>
<tr><td><code id="wiener_index_centrality_+3A_weights">weights</code></td>
<td>
<p>Numeric vector indicating weights of the edges.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector including the centrality values for each node.
The centrality values represent the Wiener Index Centrality measure for each node in the network.
</p>


<h3>Author(s)</h3>

<p>Minoo Ashtiani, Mehdi Mirzaie, Mohieddin Jafari
</p>


<h3>References</h3>

<p>Scardoni, G. and Carlo Laudanna, C.B.M.C. (2011). Network centralities for Cytoscape. University of Verona.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(zachary)

wiener_index_centrality(zachary)

</code></pre>

<hr>
<h2 id='zachary'>Zachary Karate Club Network</h2><span id='topic+zachary'></span>

<h3>Description</h3>

<p>A graph describing friendships among members of a university
karate club. Includes metadata for faction membership after a social
partition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zachary
</code></pre>


<h3>Format</h3>

<p>an igraph object with &quot;gml&quot; format
</p>


<h3>References</h3>

<p>W. W. Zachary, &quot;An information flow model for conflict and fission in small groups.&quot; Journal of Anthropological Research 33, 452-473 (1977).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("zachary")
print(zachary)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
