<!DOCTYPE html><html lang="en"><head><title>Help for package circlize</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {circlize}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#circlize-package'>
<p>Circular visualization in R</p></a></li>
<li><a href='#+24.CELL_META'>
<p>Easy to way to get meta data in the current cell</p></a></li>
<li><a href='#add_transparency'>
<p>Add transparency to colors</p></a></li>
<li><a href='#adjacencyList2Matrix'>
<p>Convert adjacency list to an adjacency matrix</p></a></li>
<li><a href='#adjacencyMatrix2List'>
<p>Convert adjacency matrix to an adjacency list</p></a></li>
<li><a href='#arrange_links_evenly'>
<p>Arrange links evenly on each sector</p></a></li>
<li><a href='#calc_gap'>
<p>Calculate gaps to make two Chord diagrams in the same scale</p></a></li>
<li><a href='#CELL_META'>
<p>Easy way to get meta data in the current cell</p></a></li>
<li><a href='#chordDiagram'>
<p>Plot Chord Diagram</p></a></li>
<li><a href='#chordDiagramFromDataFrame'>
<p>Plot Chord Diagram from a data frame</p></a></li>
<li><a href='#chordDiagramFromMatrix'>
<p>Plot Chord Diagram from an adjacency matrix</p></a></li>
<li><a href='#circlize'>
<p>Convert to polar coordinate system</p></a></li>
<li><a href='#circos.arrow'>
<p>Draw arrow which is paralle to the circle</p></a></li>
<li><a href='#circos.axis'>
<p>Draw x-axis</p></a></li>
<li><a href='#circos.barplot'>
<p>Draw barplots</p></a></li>
<li><a href='#circos.boxplot'>
<p>Draw boxplots</p></a></li>
<li><a href='#circos.clear'>
<p>Reset the circular layout parameters</p></a></li>
<li><a href='#circos.connect'>
<p>Draw connecting lines/ribons between two sets of points</p></a></li>
<li><a href='#circos.dendrogram'>
<p>Add circular dendrograms</p></a></li>
<li><a href='#circos.genomicAxis'>
<p>Add genomic axes</p></a></li>
<li><a href='#circos.genomicDensity'>
<p>Calculate and add genomic density track</p></a></li>
<li><a href='#circos.genomicHeatmap'>
<p>Add heatmaps for selected regions</p></a></li>
<li><a href='#circos.genomicIdeogram'>
<p>Add an ideogram track</p></a></li>
<li><a href='#circos.genomicInitialize'>
<p>Initialize circular plot with any genomic data</p></a></li>
<li><a href='#circos.genomicLabels'>
<p>Add labels to specified genomic regions</p></a></li>
<li><a href='#circos.genomicLines'>
<p>Add lines to a plotting region, specifically for genomic graphics</p></a></li>
<li><a href='#circos.genomicLink'>
<p>Add links from two sets of genomic positions</p></a></li>
<li><a href='#circos.genomicPoints'>
<p>Add points to a plotting region, specifically for genomic graphics</p></a></li>
<li><a href='#circos.genomicPosTransformLines'>
<p>Add genomic position transformation lines between tracks</p></a></li>
<li><a href='#circos.genomicRainfall'>
<p>Genomic rainfall plot</p></a></li>
<li><a href='#circos.genomicRect'>
<p>Draw rectangle-like grid, specifically for genomic graphics</p></a></li>
<li><a href='#circos.genomicText'>
<p>Draw text in a cell, specifically for genomic graphics</p></a></li>
<li><a href='#circos.genomicTrack'>
<p>Create a track for genomic graphics</p></a></li>
<li><a href='#circos.genomicTrackPlotRegion'>
<p>Create a track for genomic graphics</p></a></li>
<li><a href='#circos.heatmap'>
<p>Make circular heatmaps</p></a></li>
<li><a href='#circos.heatmap.initialize'>
<p>Initialize circular heatmaps</p></a></li>
<li><a href='#circos.heatmap.link'>
<p>Draw a link between two matrix rows in the circular heatmap</p></a></li>
<li><a href='#circos.info'>
<p>Get information of the circular plot</p></a></li>
<li><a href='#circos.initialize'>
<p>Initialize the circular layout</p></a></li>
<li><a href='#circos.initializeCircularGenome'>
<p>Initialize a layout for circular genome</p></a></li>
<li><a href='#circos.initializeWithIdeogram'>
<p>Initialize the circular layout with an ideogram</p></a></li>
<li><a href='#circos.labels'>
<p>Add a label track</p></a></li>
<li><a href='#circos.lines'>
<p>Add lines to the plotting region</p></a></li>
<li><a href='#circos.link'>
<p>Draw links between points or/and intervals</p></a></li>
<li><a href='#circos.nested'>
<p>Nested zooming with two circular plots</p></a></li>
<li><a href='#circos.par'>
<p>Parameters for the circular layout</p></a></li>
<li><a href='#circos.points'>
<p>Add points to a plotting region</p></a></li>
<li><a href='#circos.polygon'>
<p>Draw polygon</p></a></li>
<li><a href='#circos.raster'>
<p>Add raster images</p></a></li>
<li><a href='#circos.rect'>
<p>Draw rectangle-like grid</p></a></li>
<li><a href='#circos.segments'>
<p>Draw segments through pairwise of points</p></a></li>
<li><a href='#circos.text'>
<p>Draw text in a cell</p></a></li>
<li><a href='#circos.track'>
<p>Create plotting regions for a whole track</p></a></li>
<li><a href='#circos.trackHist'>
<p>Draw histogram in cells among a whole track</p></a></li>
<li><a href='#circos.trackLines'>
<p>Add lines to the plotting regions in a same track</p></a></li>
<li><a href='#circos.trackPlotRegion'>
<p>Create plotting regions for a whole track</p></a></li>
<li><a href='#circos.trackPoints'>
<p>Add points to the plotting regions in a same track</p></a></li>
<li><a href='#circos.trackText'>
<p>Draw text in cells among the whole track</p></a></li>
<li><a href='#circos.triangle'>
<p>Draw triangles</p></a></li>
<li><a href='#circos.update'>
<p>Create plotting regions for a whole track</p></a></li>
<li><a href='#circos.updatePlotRegion'>
<p>Update the plotting region in an existed cell</p></a></li>
<li><a href='#circos.violin'>
<p>Draw violin plots</p></a></li>
<li><a href='#circos.xaxis'>
<p>Draw x-axis</p></a></li>
<li><a href='#circos.yaxis'>
<p>Draw y-axis</p></a></li>
<li><a href='#cm_h'>
<p>Convert units</p></a></li>
<li><a href='#cm_x'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#cm_y'>
<p>Convert unit on y direction in data coordinate</p></a></li>
<li><a href='#col2value'>
<p>Transform back from colors to values</p></a></li>
<li><a href='#colorRamp2'>
<p>Color interpolation</p></a></li>
<li><a href='#convert_height'>
<p>Convert units</p></a></li>
<li><a href='#convert_length'>
<p>Convert units</p></a></li>
<li><a href='#convert_x'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#convert_y'>
<p>Convert unit on y direction in data coordinate</p></a></li>
<li><a href='#cytoband.col'>
<p>Assign colors to cytogenetic band (hg19) according to the Giemsa stain results</p></a></li>
<li><a href='#degree'>
<p>Mark the value as a degree value</p></a></li>
<li><a href='#draw.sector'>
<p>Draw sectors or rings in a circle</p></a></li>
<li><a href='#fontsize'>
<p>Convert fontsize to cex</p></a></li>
<li><a href='#generateRandomBed'>
<p>Generate random genomic data</p></a></li>
<li><a href='#genomicDensity'>
<p>Calculate genomic region density</p></a></li>
<li><a href='#get_most_inside_radius'>
<p>Get the inside radius of the most inner track</p></a></li>
<li><a href='#get.all.sector.index'>
<p>Get index for all sectors</p></a></li>
<li><a href='#get.all.track.index'>
<p>Get index for all tracks</p></a></li>
<li><a href='#get.cell.meta.data'>
<p>Get the meta data of a cell</p></a></li>
<li><a href='#get.current.chromosome'>
<p>Get current chromosome name</p></a></li>
<li><a href='#get.current.sector.index'>
<p>Get current sector index</p></a></li>
<li><a href='#get.current.track.index'>
<p>Get current track index</p></a></li>
<li><a href='#getI'>
<p>Which data that <code>panel.fun</code> is using</p></a></li>
<li><a href='#highlight.chromosome'>
<p>Highlight chromosomes</p></a></li>
<li><a href='#highlight.sector'>
<p>Highlight sectors and tracks</p></a></li>
<li><a href='#inch_h'>
<p>Convert units</p></a></li>
<li><a href='#inch_x'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#inch_y'>
<p>Convert unit on y direction in data coordinate</p></a></li>
<li><a href='#inches_h'>
<p>Convert units</p></a></li>
<li><a href='#inches_x'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#inches_y'>
<p>Convert unit on y direction in data coordinate</p></a></li>
<li><a href='#mm_h'>
<p>Convert units</p></a></li>
<li><a href='#mm_x'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#mm_y'>
<p>Convert unit on y direction in data coordinate</p></a></li>
<li><a href='#names.CELL_META'>
<p>Names of all meta data in the current cell</p></a></li>
<li><a href='#posTransform.default'>
<p>Genomic position transformation function</p></a></li>
<li><a href='#posTransform.text'>
<p>Genomic position transformation function specifically for text</p></a></li>
<li><a href='#print.CELL_META'>
<p>Print CELL_META</p></a></li>
<li><a href='#rainfallTransform'>
<p>Calculate inter-distance of genomic regions</p></a></li>
<li><a href='#rand_color'>
<p>Generate random colors</p></a></li>
<li><a href='#read.chromInfo'>
<p>Read/parse chromInfo data from a data frame/file/UCSC database</p></a></li>
<li><a href='#read.cytoband'>
<p>Read/parse cytoband data from a data frame/file/UCSC database</p></a></li>
<li><a href='#reverse.circlize'>
<p>Convert to data coordinate system</p></a></li>
<li><a href='#set_track_gap'>
<p>Set gaps between tracks</p></a></li>
<li><a href='#set.current.cell'>
<p>Set flag to current cell</p></a></li>
<li><a href='#show.index'>
<p>Label the sector index and the track index on each cell</p></a></li>
<li><a href='#smartAlign'>
<p>Adjust positions of text</p></a></li>
<li><a href='#uh'>
<p>Convert units</p></a></li>
<li><a href='#ux'>
<p>Convert unit on x direction in data coordinate</p></a></li>
<li><a href='#uy'>
<p>Convert unit on y direction in data coordinate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Circular Visualization</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.16</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-20</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), graphics</td>
</tr>
<tr>
<td>Imports:</td>
<td>GlobalOptions (&ge; 0.1.2), shape, grDevices, utils, stats,
colorspace, methods, grid</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, dendextend (&ge; 1.0.1), ComplexHeatmap (&ge; 2.0.0),
gridBase, png, markdown, bezier, covr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>Circular layout is an efficient way for the visualization of huge 
    amounts of information. Here this package provides an implementation 
    of circular layout generation in R as well as an enhancement of available 
    software. The flexibility of the package is based on the usage of low-level 
    graphics functions such that self-defined high-level graphics can be easily 
    implemented by users for specific purposes. Together with the seamless 
    connection between the powerful computational and visual environment in R, 
    it gives users more convenience and freedom to design figures for 
    better understanding complex patterns behind multiple dimensional data. 
    The package is described in Gu et al. 2014 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtu393">doi:10.1093/bioinformatics/btu393</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jokergoo/circlize">https://github.com/jokergoo/circlize</a>,
<a href="https://jokergoo.github.io/circlize_book/book/">https://jokergoo.github.io/circlize_book/book/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-20 08:13:23 UTC; guz</td>
</tr>
<tr>
<td>Author:</td>
<td>Zuguang Gu <a href="https://orcid.org/0000-0002-7395-8709"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zuguang Gu &lt;z.gu@dkfz.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-20 10:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='circlize-package'>
Circular visualization in R
</h2><span id='topic+circlize-package'></span>

<h3>Description</h3>

<p>Circular visualization in R
</p>


<h3>Details</h3>

<p>This package aims to implement circular layout in R.
</p>
<p>Since most of the figures are composed of points, lines and polygons,
we just need to implement low-level functions for drawing points, lines and polygons.
</p>
<p>Current there are following low-level graphic functions:
</p>

<ul>
<li> <p><code><a href="#topic+circos.points">circos.points</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.lines">circos.lines</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.rect">circos.rect</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.polygon">circos.polygon</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.segments">circos.segments</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.text">circos.text</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.axis">circos.axis</a></code>, <code><a href="#topic+circos.xaxis">circos.xaxis</a></code>, <code><a href="#topic+circos.yaxis">circos.yaxis</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.barplot">circos.barplot</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.boxplot">circos.boxplot</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.violin">circos.violin</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.link">circos.link</a></code>
</p>
</li></ul>

<p>For drawing points, lines and text through the whole track (among several sectors), the following
functions are available:
</p>

<ul>
<li> <p><code><a href="#topic+circos.trackPoints">circos.trackPoints</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.trackLines">circos.trackLines</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.trackText">circos.trackText</a></code>
</p>
</li></ul>

<p>Draw circular heatmaps
</p>

<ul>
<li> <p><code><a href="#topic+circos.heatmap">circos.heatmap</a></code>
</p>
</li></ul>

<p>Functions to arrange circular layout:
</p>

<ul>
<li> <p><code><a href="#topic+circos.initialize">circos.initialize</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.track">circos.track</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.nested">circos.nested</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.update">circos.update</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.par">circos.par</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.info">circos.info</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.clear">circos.clear</a></code>
</p>
</li></ul>

<p>Theoretically, you are able to draw most kinds of circular plots by the above functions.
</p>
<p>For specific use in genomics, we also implement functions which add graphics in genome scale.
</p>
<p>Functions to initialize circos plot with genomic coordinates:
</p>

<ul>
<li> <p><code><a href="#topic+circos.initializeWithIdeogram">circos.initializeWithIdeogram</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code>
</p>
</li></ul>

<p>Functions to arrange genomic circular layout:
</p>

<ul>
<li> <p><code><a href="#topic+circos.genomicTrack">circos.genomicTrack</a></code>
</p>
</li></ul>

<p>Functions to add basic graphics in genomic scale:
</p>

<ul>
<li> <p><code><a href="#topic+circos.genomicPoints">circos.genomicPoints</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicLines">circos.genomicLines</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicText">circos.genomicText</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicRect">circos.genomicRect</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicLink">circos.genomicLink</a></code>
</p>
</li></ul>

<p>Functions with specific purpose:
</p>

<ul>
<li> <p><code><a href="#topic+circos.genomicDensity">circos.genomicDensity</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicRainfall">circos.genomicRainfall</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicIdeogram">circos.genomicIdeogram</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicHeatmap">circos.genomicHeatmap</a></code>
</p>
</li>
<li> <p><code><a href="#topic+circos.genomicLabels">circos.genomicLabels</a></code>
</p>
</li></ul>

<p>Finally, function that draws Chord diagram:
</p>

<ul>
<li> <p><code><a href="#topic+chordDiagram">chordDiagram</a></code>
</p>
</li></ul>

<p>Please refer to the vignettes (<a href="https://jokergoo.github.io/circlize_book/book/">https://jokergoo.github.io/circlize_book/book/</a> ) to find out how to draw basic and advanced circular plots by this package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='+24.CELL_META'>
Easy to way to get meta data in the current cell
</h2><span id='topic+Subset.CELL_META'></span><span id='topic++24.CELL_META'></span>

<h3>Description</h3>

<p>Easy to way to get meta data in the current cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CELL_META'
x$name
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B24.CELL_META_+3A_x">x</code></td>
<td>
<p>name of the variable should be &quot;CELL_META&quot;
</p>
</td></tr>
<tr><td><code id="+2B24.CELL_META_+3A_name">name</code></td>
<td>
<p>name of the cell meta name
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The variable <code><a href="#topic+CELL_META">CELL_META</a></code> can only be used to get meta data of the &quot;current&quot; cell.
Basically you can simply replace  e.g. <code>get.cell.meta.data("sector.index")</code> to <code>CELL_META$sector.index</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.cell.meta.data">get.cell.meta.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='add_transparency'>
Add transparency to colors
</h2><span id='topic+add_transparency'></span>

<h3>Description</h3>

<p>Add transparency to colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_transparency(col, transparency = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_transparency_+3A_col">col</code></td>
<td>
<p>A vector of colors.</p>
</td></tr>
<tr><td><code id="add_transparency_+3A_transparency">transparency</code></td>
<td>
<p>Transparency, numeric value between 0 and 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of colors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>add_transparency("red", 0.5)
add_transparency(1, 0.5)
add_transparency("#FF000080", 0.2)
</code></pre>

<hr>
<h2 id='adjacencyList2Matrix'>
Convert adjacency list to an adjacency matrix
</h2><span id='topic+adjacencyList2Matrix'></span>

<h3>Description</h3>

<p>Convert adjacency list to an adjacency matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjacencyList2Matrix(lt, square = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjacencyList2Matrix_+3A_lt">lt</code></td>
<td>
<p>A data frame which contains adjacency list.</p>
</td></tr>
<tr><td><code id="adjacencyList2Matrix_+3A_square">square</code></td>
<td>
<p>Should the returned matrix be a square matrix?</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
df = data.frame(from = sample(letters, 10, replace = TRUE), 
                to = sample(letters, 10, replace = TRUE), 
                value = 1:10)
adjacencyList2Matrix(df)
adjacencyList2Matrix(df, square = TRUE)
</code></pre>

<hr>
<h2 id='adjacencyMatrix2List'>
Convert adjacency matrix to an adjacency list
</h2><span id='topic+adjacencyMatrix2List'></span>

<h3>Description</h3>

<p>Convert adjacency matrix to an adjacency list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjacencyMatrix2List(mat, keep.zero = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjacencyMatrix2List_+3A_mat">mat</code></td>
<td>
<p>A numeric matrix.</p>
</td></tr>
<tr><td><code id="adjacencyMatrix2List_+3A_keep.zero">keep.zero</code></td>
<td>
<p>Whether to keep the interactions with value zero.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(999)
mat = matrix(sample(18, 18), 3, 6) 
rownames(mat) = paste0("S", 1:3)
colnames(mat) = paste0("E", 1:6)
adjacencyMatrix2List(mat)
</code></pre>

<hr>
<h2 id='arrange_links_evenly'>
Arrange links evenly on each sector
</h2><span id='topic+arrange_links_evenly'></span>

<h3>Description</h3>

<p>Arrange links evenly on each sector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arrange_links_evenly(df, directional = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arrange_links_evenly_+3A_df">df</code></td>
<td>
<p>A data frame with two columns. The values should only contain sector names.
</p>
</td></tr>
<tr><td><code id="arrange_links_evenly_+3A_directional">directional</code></td>
<td>
<p>Whether the links are directional.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function only deals with single-line links.
</p>


<h3>Value</h3>

<p>A data frame with four columns of the sectors and the positions of the links.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:20]
df = data.frame(from = sample(sectors, 40, replace = TRUE),
                to   = sample(sectors, 40, replace = TRUE),
                stringsAsFactors = FALSE)
df = unique(df)
df = df[df$from != df$to, ]

circos.initialize(sectors, xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ycenter, CELL_META$sector.index)
})

df2 = arrange_links_evenly(df, directional = 1)

for(i in seq_len(nrow(df2))) {
    s1 = df$from[i]
    s2 = df$to[i]
    circos.link(df2[i, "sector1"], df2[i, "pos1"], 
                df2[i, "sector2"], df2[i, "pos2"],
                directional = 1)
}
</code></pre>

<hr>
<h2 id='calc_gap'>
Calculate gaps to make two Chord diagrams in the same scale
</h2><span id='topic+calc_gap'></span>

<h3>Description</h3>

<p>Calculate gaps to make two Chord diagrams in the same scale
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_gap(x1, x2, big.gap = 10, small.gap = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_gap_+3A_x1">x1</code></td>
<td>
<p>The matrix or the data frame for the first Chord diagram.
</p>
</td></tr>
<tr><td><code id="calc_gap_+3A_x2">x2</code></td>
<td>
<p>The matrix or the data frame for the second Chord diagram.
</p>
</td></tr>
<tr><td><code id="calc_gap_+3A_big.gap">big.gap</code></td>
<td>
<p><code>big.gap</code> for the first Chord diagram.
</p>
</td></tr>
<tr><td><code id="calc_gap_+3A_small.gap">small.gap</code></td>
<td>
<p><code>small.gap</code> for both Chord diagrams.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both Chord diagrams should be both two-group Chord diagram.
</p>


<h3>Value</h3>

<p>A numeric value which can be directly set to <code>big.gap</code> in the second Chord diagram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
mat1 = matrix(sample(20, 25, replace = TRUE), 5)
chordDiagram(mat1, directional = 1, grid.col = rep(1:5, 2), transparency = 0.5,
    big.gap = 10, small.gap = 1)
mat2 = mat1 / 2
gap = calc_gap(mat1, mat2, big.gap = 10, small.gap = 1)
chordDiagram(mat2, directional = 1, grid.col = rep(1:5, 2), transparency = 0.5,
    big.gap = gap, small.gap = 1)
</code></pre>

<hr>
<h2 id='CELL_META'>
Easy way to get meta data in the current cell
</h2><span id='topic+CELL_META'></span>

<h3>Description</h3>

<p>Easy way to get meta data in the current cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CELL_META
</code></pre>


<h3>Details</h3>

<p>The variable <code><a href="#topic+CELL_META">CELL_META</a></code> can only be used to get meta data of the &quot;current&quot; cell.
Basically you can simply replace  e.g. <code>get.cell.meta.data("sector.index")</code> to <code>CELL_META$sector.index</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.cell.meta.data">get.cell.meta.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdf(NULL)
circos.initialize("a", xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    print(CELL_META$sector.index)
    print(CELL_META$xlim)
})
print(names(CELL_META))
dev.off()
</code></pre>

<hr>
<h2 id='chordDiagram'>
Plot Chord Diagram
</h2><span id='topic+chordDiagram'></span>

<h3>Description</h3>

<p>Plot Chord Diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chordDiagram(
    x,
    grid.col = NULL,
    grid.border = NA,
    transparency = 0.5,
    col = NULL,
    row.col = NULL,
    column.col = NULL,
    order = NULL,
    directional = 0,
    xmax = NULL,
    symmetric = FALSE,
    keep.diagonal = FALSE,
    direction.type = "diffHeight",
    diffHeight = mm_h(2),
    link.target.prop = TRUE,
    target.prop.height = mm_h(1),
    reduce = 1e-5,
    self.link = 2,
    preAllocateTracks = NULL,
    annotationTrack = c("name", "grid", "axis"),
    annotationTrackHeight = mm_h(c(3, 2)),
    link.border = NA,
    link.lwd = par("lwd"),
    link.lty = par("lty"),
    link.auto = TRUE,
    link.sort = "default",
    link.decreasing = TRUE,
    link.arr.length = ifelse(link.arr.type == "big.arrow", 0.02, 0.4),
    link.arr.width = link.arr.length/2,
    link.arr.type = "triangle",
    link.arr.lty = par("lty"),
    link.arr.lwd = par("lwd"),
    link.arr.col = par("col"),
    link.largest.ontop = FALSE,
    link.visible = TRUE,
    link.rank = NULL,
    link.zindex = NULL,
    link.overlap = FALSE,
    scale = FALSE,
    group = NULL,
    big.gap = 10,
    small.gap = 1,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chordDiagram_+3A_x">x</code></td>
<td>
<p>a matrix or a data frame. The function will pass all argument to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code> depending on the type of <code>x</code>,
also format of other arguments depends of the type of <code>x</code>. If it is in the form of a matrix,
it should be an adjacency matrix. If it is in the form of a data frame, it should be an adjacency list.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_grid.col">grid.col</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_grid.border">grid.border</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_transparency">transparency</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_col">col</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_row.col">row.col</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_column.col">column.col</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_order">order</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_directional">directional</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_xmax">xmax</code></td>
<td>
<p>maximum value on x-axes, the value should be a named vector.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_symmetric">symmetric</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_keep.diagonal">keep.diagonal</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_direction.type">direction.type</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_diffheight">diffHeight</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.target.prop">link.target.prop</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_target.prop.height">target.prop.height</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_reduce">reduce</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_self.link">self.link</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_preallocatetracks">preAllocateTracks</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_annotationtrack">annotationTrack</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_annotationtrackheight">annotationTrackHeight</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.border">link.border</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.lwd">link.lwd</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.lty">link.lty</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.auto">link.auto</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.sort">link.sort</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.decreasing">link.decreasing</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.length">link.arr.length</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.width">link.arr.width</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.type">link.arr.type</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.lty">link.arr.lty</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.lwd">link.arr.lwd</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.arr.col">link.arr.col</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.largest.ontop">link.largest.ontop</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.visible">link.visible</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.rank">link.rank</code></td>
<td>
<p>This is argument is removed.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.zindex">link.zindex</code></td>
<td>
<p>order to add links to the circle, a large value means to add it later.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_link.overlap">link.overlap</code></td>
<td>
<p>pass to <code><a href="#topic+chordDiagramFromMatrix">chordDiagramFromMatrix</a></code> or <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_scale">scale</code></td>
<td>
<p>scale each sector to same width
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_group">group</code></td>
<td>
<p>It contains the group labels and the sector names are used as the names in the vector.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_big.gap">big.gap</code></td>
<td>
<p>Gap between the two sets of sectors. If the input is a matrix, the two sets
are row sectors and column sectors. If the input is a data frame, the two sets
correspond to the first column and the second column. It only works when there
is no intersection between the two sets.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_small.gap">small.gap</code></td>
<td>
<p>Small gap between sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagram_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Chord diagram is a way to visualize numeric tables ( <a href="http://circos.ca/intro/tabular_visualization/">http://circos.ca/intro/tabular_visualization/</a> ), especially useful
when the table represents information of directional relations. This function
visualize tables in a circular way.
</p>
<p>This function is flexible and contains some settings that may be a little difficult to understand.
Please refer to vignette for better explanation.
</p>


<h3>Value</h3>

<p>A data frame which contains positions of links, columns are:
</p>

<dl>
<dt><code>rn</code></dt><dd><p>sector name corresponding to rows in the adjacency matrix or the first column in the adjacency list
</p>
</dd>
<dt><code>cn</code></dt><dd><p>sector name corresponding to columns in the adjacency matrix or the second column in the adjacency list
</p>
</dd>
<dt><code>value</code></dt><dd><p>value for the interaction or relation
</p>
</dd>
<dt><code>o1</code></dt><dd><p>order of the link on the &quot;from&quot; sector
</p>
</dd>
<dt><code>o2</code></dt><dd><p>order of the link on the &quot;to&quot; sector
</p>
</dd>
<dt><code>x1</code></dt><dd><p>and position of the link on the &quot;from&quot; sector, the interval for the link on the &quot;from&quot; sector is <code>c(x1-abs(value), x1)</code>
</p>
</dd>
<dt><code>x2</code></dt><dd><p>and position of the link on the &quot;to&quot; sector, the interval for the link on the &quot;from&quot; sector is <code>c(x2-abs(value), x2)</code>
</p>
</dd>
</dl>



<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html">https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(999)
mat = matrix(sample(18, 18), 3, 6)
rownames(mat) = paste0("S", 1:3)
colnames(mat) = paste0("E", 1:6)

df = data.frame(from = rep(rownames(mat), times = ncol(mat)),
    to = rep(colnames(mat), each = nrow(mat)),
    value = as.vector(mat),
    stringsAsFactors = FALSE)

chordDiagram(mat)
chordDiagram(df)
circos.clear()
</code></pre>

<hr>
<h2 id='chordDiagramFromDataFrame'>
Plot Chord Diagram from a data frame
</h2><span id='topic+chordDiagramFromDataFrame'></span>

<h3>Description</h3>

<p>Plot Chord Diagram from a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chordDiagramFromDataFrame(
    df,
    grid.col = NULL,
    grid.border = NA,
    transparency = 0.5,
    col = NULL,
    order = NULL,
    directional = 0,
    xmax = NULL,
    direction.type = "diffHeight",
    diffHeight = convert_height(2, "mm"),
    link.target.prop = TRUE,
    target.prop.height = mm_h(1),
    reduce = 1e-5,
    self.link = 2,
    preAllocateTracks = NULL,
    annotationTrack = c("name", "grid", "axis"),
    annotationTrackHeight = convert_height(c(3, 2), "mm"),
    link.border = NA,
    link.lwd = par("lwd"),
    link.lty = par("lty"),
    link.auto = TRUE,
    link.sort = "default",
    link.decreasing = TRUE,
    link.arr.length = ifelse(link.arr.type == "big.arrow", 0.02, 0.4),
    link.arr.width = link.arr.length/2,
    link.arr.type = "triangle",
    link.arr.lty = par("lty"),
    link.arr.lwd = par("lwd"),
    link.arr.col = par("col"),
    link.largest.ontop = FALSE,
    link.visible = TRUE,
    link.rank = NULL,
    link.zindex = seq_len(nrow(df)),
    link.overlap = FALSE,
    scale = FALSE,
    group = NULL,
    big.gap = 10,
    small.gap = 1,
    plot = TRUE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chordDiagramFromDataFrame_+3A_df">df</code></td>
<td>
<p>A data frame with at least two columns. The first two columns specify the connections and the third column (optional)
contains numeric values which are mapped to the width of links as well as the colors if <code>col</code> is specified as a color mapping function.
The sectors in the plot will be <code>union(df[[1]], df[[2]])</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_grid.col">grid.col</code></td>
<td>
<p>Grid colors which correspond to sectors. The length of the vector should be either 1 or the number of sectors.
It's preferred that <code>grid.col</code> is a named vector of which names correspond to sectors.
If it is not a named vector, the order of <code>grid.col</code> corresponds to order of sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_grid.border">grid.border</code></td>
<td>
<p>border for grids. If it is <code>NULL</code>, the border color is same as grid color
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_transparency">transparency</code></td>
<td>
<p>Transparency of link colors, 0 means no transparency and 1 means full transparency.
If transparency is already set in <code>col</code> or <code>row.col</code> or <code>column.col</code>, this argument will be ignored.
<code>NA</code>also ignores this argument.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_col">col</code></td>
<td>
<p>Colors for links. It can be a vector which corresponds to connections in <code>df</code>, or a function which generate colors
according to values (the third column) in <code>df</code>, or a single value which means colors for all links are the same. You
may use <code><a href="#topic+colorRamp2">colorRamp2</a></code> to generate a function which maps values to colors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_order">order</code></td>
<td>
<p>Order of sectors. Default order is <code>union(df[[1]], df[[2]])</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_directional">directional</code></td>
<td>
<p>Whether links have directions. 1 means the direction is from the first column in <code>df</code> to the second column, -1
is the reverse, 0 is no direction, and 2 for two directional. The value can be a vector which has same length as number of rows in <code>df</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_xmax">xmax</code></td>
<td>
<p>maximum value on x-axes, the value should be a named vector.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_direction.type">direction.type</code></td>
<td>
<p>type for representing directions. Can be one or two values in &quot;diffHeight&quot; and &quot;arrows&quot;. If the value contains &quot;diffHeight&quot;,
different heights of the links are used to represent the directions for which starting root has long height to give people feeling
that something is comming out. If the value contains &quot;arrows&quot;, users can customize arrows with following arguments.
The value can be a vector which has same length as number of rows in <code>df</code>. Note if you want to set both <code>diffHeight</code>
and <code>arrows</code> for certain links, you need to embed these two options into one string such as <code>"diffHeight+arrows"</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_diffheight">diffHeight</code></td>
<td>
<p>The difference of height between two 'roots' if <code>directional</code> is set to <code>TRUE</code>. If the value is set to
a positive value, start root is shorter than end root and if it is set to a negative value, start root is longer
than the end root. The value can be a vector which has same length as number of rows in <code>df</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.target.prop">link.target.prop</code></td>
<td>
<p>If the Chord diagram is directional, for each source sector, whether to draw bars that shows the proportion of 
target sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_target.prop.height">target.prop.height</code></td>
<td>
<p>The height of the bars when <code>link.target.prop</code> is turned on.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_reduce">reduce</code></td>
<td>
<p>if the ratio of the width of certain grid compared to the whole circle is less than this value, the grid is removed on the plot.
Set it to value less than zero if you want to keep all tiny grid.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_self.link">self.link</code></td>
<td>
<p>if there is a self link in one sector, 1 means the link will be degenerated as a 'mountain' and the width corresponds to the value for this connection.
2 means the width of the starting root and the ending root all have the same width that corresponds to the value for the connection.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_preallocatetracks">preAllocateTracks</code></td>
<td>
<p>Pre-allocate empty tracks before drawing Chord diagram. It can be a single number indicating
how many empty tracks needed to be created or a list containing settings for empty
tracks. Please refer to vignette for details.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_annotationtrack">annotationTrack</code></td>
<td>
<p>Which annotation track should be plotted? By default, a track containing sector names and a track
containing grid will be created.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_annotationtrackheight">annotationTrackHeight</code></td>
<td>
<p>Track height corresponding to values in <code>annotationTrack</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.border">link.border</code></td>
<td>
<p>border for links, single scalar or a vector which has the same length as nrows of <code>df</code> or a data frame
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.lwd">link.lwd</code></td>
<td>
<p>width for link borders, single scalar or a vector which has the same length as nrows of <code>df</code> or a data frame
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.lty">link.lty</code></td>
<td>
<p>style for link borders, single scalar or a vector which has the same length as nrows of <code>df</code> or a data frame
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.auto">link.auto</code></td>
<td>
<p>Ignored.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.sort">link.sort</code></td>
<td>
<p>whether sort links on every sector based on the width of the links on it. The value can be logical. The value can also be string
&quot;default&quot; which automatically adjusts link orders so that links have minimal overall intersections. The value can also be a string 
&quot;asis&quot; and it is only workable for input as a data frame so that the links have the same orders as in the original data frame.# -link.decreasing for <code>link.sort</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.decreasing">link.decreasing</code></td>
<td>
<p>for <code>link.sort</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.length">link.arr.length</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.width">link.arr.width</code></td>
<td>
<p>pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.type">link.arr.type</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>, same settings as <code>link.lwd</code>. Default value is <code>triangle</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.col">link.arr.col</code></td>
<td>
<p>color or the single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.lwd">link.arr.lwd</code></td>
<td>
<p>line width ofthe single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.arr.lty">link.arr.lty</code></td>
<td>
<p>line type of the single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.largest.ontop">link.largest.ontop</code></td>
<td>
<p>controls the order of adding links, whether based on the absolute value?
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.rank">link.rank</code></td>
<td>
<p>This is argument is removed.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.visible">link.visible</code></td>
<td>
<p>whether plot the link. The value is logical, if it is set to <code>FALSE</code>, the corresponding link will not
plotted, but the space is still ocuppied. The format of this argument is same as <code>link.lwd</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.zindex">link.zindex</code></td>
<td>
<p>order to add links to the circle, a large value means to add it later.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_link.overlap">link.overlap</code></td>
<td>
<p>if it is a directional Chord Diagram, whether the links that come or end in a same sector overlap?
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_scale">scale</code></td>
<td>
<p>scale each sector to same width
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_group">group</code></td>
<td>
<p>It contains the group labels and the sector names are used as the names in the vector.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_big.gap">big.gap</code></td>
<td>
<p>Gaps between the sectors in the first column of <code>df</code> and sectors in the second column in <code>df</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_small.gap">small.gap</code></td>
<td>
<p>Small gap between sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_plot">plot</code></td>
<td>
<p>Internally used.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromDataFrame_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data frame can have a column named &quot;rank&quot; which is used to control the order of adding links to the diagram.
</p>


<h3>Value</h3>

<p>A data frame which contains positions of links, see explanation in <code><a href="#topic+chordDiagram">chordDiagram</a></code>.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html">https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='chordDiagramFromMatrix'>
Plot Chord Diagram from an adjacency matrix
</h2><span id='topic+chordDiagramFromMatrix'></span>

<h3>Description</h3>

<p>Plot Chord Diagram from an adjacency matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chordDiagramFromMatrix(
    mat,
    grid.col = NULL,
    grid.border = NA,
    transparency = 0.5,
    col = NULL,
    row.col = NULL,
    column.col = NULL,
    order = NULL,
    directional = 0,
    direction.type = "diffHeight",
    diffHeight = mm_h(2),
    link.target.prop = TRUE,
    target.prop.height = mm_h(1),
    reduce = 1e-5,
    xmax = NULL,
    self.link = 2,
    symmetric = FALSE,
    keep.diagonal = FALSE,
    preAllocateTracks = NULL,
    annotationTrack = c("name", "grid", "axis"),
    annotationTrackHeight = mm_h(c(3, 2)),
    link.border = NA,
    link.lwd = par("lwd"),
    link.lty = par("lty"),
    link.auto = TRUE,
    link.sort = "default",
    link.decreasing = TRUE,
    link.arr.length = ifelse(link.arr.type == "big.arrow", 0.02, 0.4),
    link.arr.width = link.arr.length/2,
    link.arr.type = "triangle",
    link.arr.lty = par("lty"),
    link.arr.lwd = par("lwd"),
    link.arr.col = par("col"),
    link.largest.ontop = FALSE,
    link.visible = TRUE,
    link.rank = NULL,
    link.zindex = NULL,
    link.overlap = FALSE,
    scale = FALSE,
    group = NULL,
    big.gap = 10,
    small.gap = 1,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chordDiagramFromMatrix_+3A_mat">mat</code></td>
<td>
<p>A table which represents as a numeric matrix.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_grid.col">grid.col</code></td>
<td>
<p>Grid colors which correspond to matrix rows/columns (or sectors). The length of the vector should be either 1 or <code>length(union(rownames(mat), colnames(mat)))</code>.
It's preferred that <code>grid.col</code> is a named vector of which names correspond to sectors.
If it is not a named vector, the order of <code>grid.col</code> corresponds to order of sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_grid.border">grid.border</code></td>
<td>
<p>border for grids. If it is <code>NULL</code>, the border color is same as grid color
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_transparency">transparency</code></td>
<td>
<p>Transparency of link colors, 0 means no transparency and 1 means full transparency.
If transparency is already set in <code>col</code> or <code>row.col</code> or <code>column.col</code>, this argument will be ignored.
<code>NA</code>also ignores this argument.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_col">col</code></td>
<td>
<p>Colors for links. It can be a matrix which corresponds to <code>mat</code>, or a function which generate colors
according to values in <code>mat</code>, or a single value which means colors for all links are the same, or a three-column
data frame in which the first two columns correspond to row names and columns and the third column is colors. You
may use <code><a href="#topic+colorRamp2">colorRamp2</a></code> to generate a function which maps values to colors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_row.col">row.col</code></td>
<td>
<p>Colors for links. Links from the same row in <code>mat</code> will have the same color.
Length should be same as number of rows in <code>mat</code>. This argument only works when <code>col</code> is set to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_column.col">column.col</code></td>
<td>
<p>Colors for links. Links from the same column in <code>mat</code> will have the same color.
Length should be same as number of columns in <code>mat</code>. This argument only works when <code>col</code> and <code>row.col</code> is set to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_order">order</code></td>
<td>
<p>Order of sectors. Default order is <code>union(df[[1]], df[[2]])</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_directional">directional</code></td>
<td>
<p>Whether links have directions. 1 means the direction is from the first column in <code>df</code> to the second column, -1
is the reverse, 0 is no direction, and 2 for two directional. Same setting as <code>link.border</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_xmax">xmax</code></td>
<td>
<p>maximum value on x-axes, the value should be a named vector.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_direction.type">direction.type</code></td>
<td>
<p>type for representing directions. Can be one or two values in &quot;diffHeight&quot; and &quot;arrows&quot;. If the value contains &quot;diffHeight&quot;,
different heights of the links are used to represent the directions for which starting root has long height to give people feeling
that something is comming out. If the value contains &quot;arrows&quot;, users can customize arrows with following arguments.
Same setting as <code>link.border</code>. Note if you want to set both <code>diffHeight</code>
and <code>arrows</code> for certain links, you need to embed these two options into one string such as <code>"diffHeight+arrows"</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_diffheight">diffHeight</code></td>
<td>
<p>The difference of height between two 'roots' if <code>directional</code> is set to <code>TRUE</code>. If the value is set to
a positive value, start root is shorter than end root and if it is set to a negative value, start root is longer
than the end root.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.target.prop">link.target.prop</code></td>
<td>
<p>If the Chord diagram is directional, for each source sector, whether to draw bars that shows the proportion of 
target sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_target.prop.height">target.prop.height</code></td>
<td>
<p>The height of the bars when <code>link.target.prop</code> is turned on.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_reduce">reduce</code></td>
<td>
<p>if the ratio of the width of certain grid compared to the whole circle is less than this value, the grid is removed on the plot.
Set it to value less than zero if you want to keep all tiny grid.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_self.link">self.link</code></td>
<td>
<p>if there is a self link in one sector, 1 means the link will be degenerated as a 'mountain' and the width corresponds to the value for this connection.
2 means the width of the starting root and the ending root all have the width that corresponds to the value for the connection.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_symmetric">symmetric</code></td>
<td>
<p>Whether the matrix is symmetric. If the value is set to <code>TRUE</code>, only
lower triangular matrix without the diagonal will be used.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_keep.diagonal">keep.diagonal</code></td>
<td>
<p>If the matrix is specified as symmetric, whether keep diagonal for visualization.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_preallocatetracks">preAllocateTracks</code></td>
<td>
<p>Pre-allocate empty tracks before drawing Chord diagram. It can be a single number indicating
how many empty tracks needed to be created or a list containing settings for empty
tracks. Please refer to vignette for details.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_annotationtrack">annotationTrack</code></td>
<td>
<p>Which annotation track should be plotted? By default, a track containing sector names and a track
containing grid will be created.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_annotationtrackheight">annotationTrackHeight</code></td>
<td>
<p>Track height corresponding to values in <code>annotationTrack</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.border">link.border</code></td>
<td>
<p>border for links, single scalar or a matrix with names or a data frame with three columns
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.lwd">link.lwd</code></td>
<td>
<p>width for link borders, single scalar or a matrix with names or a data frame with three columns
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.lty">link.lty</code></td>
<td>
<p>style for link borders, single scalar or a matrix with names or a data frame with three columns
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.auto">link.auto</code></td>
<td>
<p>Ignored.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.sort">link.sort</code></td>
<td>
<p>whether sort links on every sector based on the width of the links on it. The value can be logical. The value can also be string
&quot;default&quot; which automatically adjusts link orders so that links have minimal overall intersections. The value can also be a string 
&quot;asis&quot; and it is only workable for input as a data frame so that the links have the same orders as in the original data frame.# -link.decreasing for <code>link.sort</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.decreasing">link.decreasing</code></td>
<td>
<p>for <code>link.sort</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.length">link.arr.length</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.width">link.arr.width</code></td>
<td>
<p>pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.type">link.arr.type</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>, same format as <code>link.lwd</code>. Default value is <code>triangle</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.col">link.arr.col</code></td>
<td>
<p>color or the single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.lwd">link.arr.lwd</code></td>
<td>
<p>line width ofthe single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.arr.lty">link.arr.lty</code></td>
<td>
<p>line type of the single line link which is put in the center of the belt. The format of this argument is same as <code>link.lwd</code>.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.largest.ontop">link.largest.ontop</code></td>
<td>
<p>controls the order of adding links, whether based on the absolute value?
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.visible">link.visible</code></td>
<td>
<p>whether plot the link. The value is logical, if it is set to <code>FALSE</code>, the corresponding link will not
plotted, but the space is still ocuppied. The format of this argument is same as <code>link.lwd</code>
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.rank">link.rank</code></td>
<td>
<p>This is argument is removed.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.zindex">link.zindex</code></td>
<td>
<p>order to add links to the circle, a large value means to add it later.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_link.overlap">link.overlap</code></td>
<td>
<p>if it is a directional Chord Diagram, whether the links that come or end in a same sector overlap?
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_scale">scale</code></td>
<td>
<p>scale each sector to same width
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_group">group</code></td>
<td>
<p>It contains the group labels and the sector names are used as the names in the vector.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_big.gap">big.gap</code></td>
<td>
<p>Gap between row sectors and column sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_small.gap">small.gap</code></td>
<td>
<p>Small gap between sectors.
</p>
</td></tr>
<tr><td><code id="chordDiagramFromMatrix_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+circos.link">circos.link</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally, the matrix is transformed to a data frame and sent to <code><a href="#topic+chordDiagramFromDataFrame">chordDiagramFromDataFrame</a></code>.
</p>


<h3>Value</h3>

<p>A data frame which contains positions of links, see explanation in <code><a href="#topic+chordDiagram">chordDiagram</a></code>.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html">https://jokergoo.github.io/circlize_book/book/the-chorddiagram-function.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circlize'>
Convert to polar coordinate system
</h2><span id='topic+circlize'></span>

<h3>Description</h3>

<p>Convert to polar coordinate system
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circlize(
    x, y,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circlize_+3A_x">x</code></td>
<td>
<p>Data points on x-axis. The value can also be a two-column matrix/data frame if you put x and y data points into one variable.</p>
</td></tr>
<tr><td><code id="circlize_+3A_y">y</code></td>
<td>
<p>Data points on y-axis.</p>
</td></tr>
<tr><td><code id="circlize_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector to convert the coordinates.</p>
</td></tr>
<tr><td><code id="circlize_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track to convert the coordinates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the core function in the package. It transform data points from data coordinate system (in a specific cell) to the polar coordinate system.
</p>


<h3>Value</h3>

<p>A matrix with two columns (<code>theta</code> and <code>rou</code>). <code>rou</code> is measured in degree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdf(NULL)
sectors = c("a", "b")
circos.initialize(sectors, xlim = c(0, 1))
circos.track(ylim = c(0, 1))
# x = 0.5, y = 0.5 in sector a and track 1
circlize(0.5, 0.5, sector.index = "a", track.index = 1)
circos.clear()
dev.off()
</code></pre>

<hr>
<h2 id='circos.arrow'>
Draw arrow which is paralle to the circle
</h2><span id='topic+circos.arrow'></span>

<h3>Description</h3>

<p>Draw arrow which is paralle to the circle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.arrow(
    x1,
    x2,
    y = get.cell.meta.data("ycenter"),
    width = get.cell.meta.data("yrange")/2,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    arrow.head.length = mm_x(5),
    arrow.head.width = width*2,
    arrow.position = c("end", "start"),
    tail = c("normal", "point"),
    border = "black",
    col = "#FFCCCC",
    lty = par("lty"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.arrow_+3A_x1">x1</code></td>
<td>
<p>Start position of the arrow on the x-axis.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_x2">x2</code></td>
<td>
<p>End position of the arrow on the x-axis. Note <code>x2</code> should be larger than <code>x1</code>. The direction of arrows can be controlled by <code>arrow.position</code> argument.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_y">y</code></td>
<td>
<p>Position of the arrow on the y-axis. Note this is the center of the arrow on y-axis.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_width">width</code></td>
<td>
<p>Width of the arrow body.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of the sector.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_track.index">track.index</code></td>
<td>
<p>Index of the track.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_arrow.head.length">arrow.head.length</code></td>
<td>
<p>Length of the arrow head. Note the value should be smaller than the length of the arrow itself (which is <code>x2 - x1</code>).</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_arrow.head.width">arrow.head.width</code></td>
<td>
<p>Width of the arrow head.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_arrow.position">arrow.position</code></td>
<td>
<p>Where is the arrow head on the arrow. If you want to the arrow in the reversed direction, set this value to <code>"start"</code>.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_tail">tail</code></td>
<td>
<p>The shape of the arrow tail (the opposite side of arrow head).</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_border">border</code></td>
<td>
<p>Border color of the arrow.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_col">col</code></td>
<td>
<p>Filled color of the arrow.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_lty">lty</code></td>
<td>
<p>Line style of the arrow.</p>
</td></tr>
<tr><td><code id="circos.arrow_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note all position values are measured in the data coordinate (the coordinate in each cell). For the values of
<code>width</code>, <code>arrow.head.Length</code>, <code>arrow.head.width</code>, they can be set with <code><a href="#topic+mm_y">mm_y</a></code>/<code><a href="#topic+cm_y">cm_y</a></code>/<code><a href="#topic+inches_y">inches_y</a></code> in absolute units.
</p>
<p>If you see points overflow warnings, you can set <code>circos.par(points.overflow.warning = FALSE)</code> to turn it off.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#circular-arrows">https://jokergoo.github.io/circlize_book/book/graphics.html#circular-arrows</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>op = par(no.readonly = TRUE)
par(mfrow = c(1, 2))
circos.initialize(letters[1:4], xlim = c(0, 1))
col = rand_color(4)
tail = c("point", "normal", "point", "normal")
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.arrow(x1 = 0, x2 = 1, y = 0.5, width = 0.4, 
        arrow.head.width = 0.6, arrow.head.length = cm_x(1), 
        col = col[CELL_META$sector.numeric.index], 
        tail = tail[CELL_META$sector.numeric.index])
}, bg.border = NA, track.height = 0.4)
circos.clear()

circos.initialize(letters[1:4], xlim = c(0, 1))
tail = c("point", "normal", "point", "normal")
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.arrow(x1 = 0, x2 = 1, y = 0.5, width = 0.4, 
        arrow.head.width = 0.6, arrow.head.length = cm_x(1), 
        col = col[CELL_META$sector.numeric.index], 
        tail = tail[CELL_META$sector.numeric.index],
        arrow.position = "start")
}, bg.border = NA, track.height = 0.4)
par(op)

########## cell cycle ###########
cell_cycle = data.frame(phase = factor(c("G1", "S", "G2", "M"), 
                                    levels = c("G1", "S", "G2", "M")),
                        hour = c(11, 8, 4, 1))
color = c("#66C2A5", "#FC8D62", "#8DA0CB", "#E78AC3")
circos.par(start.degree = 90)
circos.initialize(cell_cycle$phase, xlim = cbind(rep(0, 4), cell_cycle$hour))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  circos.arrow(CELL_META$xlim[1], CELL_META$xlim[2], 
      arrow.head.width = CELL_META$yrange*0.8, arrow.head.length = cm_x(1),
      col = color[CELL_META$sector.numeric.index])
  circos.text(CELL_META$xcenter, CELL_META$ycenter, CELL_META$sector.index, 
      facing = "downward")
}, bg.border = NA, track.height = 0.3)
circos.clear()
</code></pre>

<hr>
<h2 id='circos.axis'>
Draw x-axis
</h2><span id='topic+circos.axis'></span>

<h3>Description</h3>

<p>Draw x-axis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.axis(
    h = "top",
    major.at = NULL,
    labels = TRUE,
    major.tick = TRUE,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    labels.font = par("font"),
    labels.cex = par("cex"),
    labels.facing = "inside",
    labels.direction = NULL,
    labels.niceFacing = TRUE,
    direction = c("outside", "inside"),
    minor.ticks = 4,
    major.tick.length = mm_y(1),
    major.tick.percentage = 0.5,
    lwd = par("lwd"),
    col = par("col"),
    labels.col = par("col"),
    labels.pos.adjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.axis_+3A_h">h</code></td>
<td>
<p>Position of the x-axis, can be &quot;top&quot;, &quot;bottom&quot; or a numeric value</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_major.at">major.at</code></td>
<td>
<p>If it is numeric vector, it identifies the positions of the major ticks. It can exceed <code>xlim</code> value and the exceeding part would be trimmed automatically. If it is <code>NULL</code>, about every 10 degrees there is a major tick.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels">labels</code></td>
<td>
<p>labels of the major ticks. Also, the exceeding part would be trimmed automatically. The value can also be logical (either an atomic value or a vector) which represents which labels to show.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_major.tick">major.tick</code></td>
<td>
<p>Whether to draw major tick. If it is set to <code>FALSE</code>, there will be no minor ticks neither.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.font">labels.font</code></td>
<td>
<p>Font style for the axis labels.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.cex">labels.cex</code></td>
<td>
<p>Font size for the axis labels.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.direction">labels.direction</code></td>
<td>
<p>Deprecated, use <code>facing</code> instead.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.facing">labels.facing</code></td>
<td>
<p>Facing of labels on axis, passing to <code><a href="#topic+circos.text">circos.text</a></code></p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.nicefacing">labels.niceFacing</code></td>
<td>
<p>Should facing of axis labels be human-easy.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_direction">direction</code></td>
<td>
<p>Whether the axis ticks point to the outside or inside of the circle.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_minor.ticks">minor.ticks</code></td>
<td>
<p>Number of minor ticks between two close major ticks.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_major.tick.length">major.tick.length</code></td>
<td>
<p>Length of the major ticks, measured in &quot;current&quot; data coordinate. <code><a href="#topic+convert_y">convert_y</a></code> can be used to convert an absolute unit to the data coordinate.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_major.tick.percentage">major.tick.percentage</code></td>
<td>
<p>Not used any more, please directly use <code>major.tick.length</code>.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_lwd">lwd</code></td>
<td>
<p>Line width for ticks.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_col">col</code></td>
<td>
<p>Color for the axes.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.col">labels.col</code></td>
<td>
<p>Color for the labels.</p>
</td></tr>
<tr><td><code id="circos.axis_+3A_labels.pos.adjust">labels.pos.adjust</code></td>
<td>
<p>Whether to adjust the positions of the first label and the last label so that the first label  align to its left and the last label align to its right if they exceed the range on axes. The value can be a vector of length two which correspond to the first label and the last label.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It only draws axes on x-direction.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+circos.yaxis">circos.yaxis</a></code> draws axes on y-direction.
</p>
<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#axes">https://jokergoo.github.io/circlize_book/book/graphics.html#axes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:8]
circos.par(points.overflow.warning = FALSE)
circos.initialize(sectors, xlim = c(0, 10))
circos.trackPlotRegion(sectors, ylim = c(0, 10), track.height = 0.1,
    bg.border = NA, panel.fun = function(x, y) {
        circos.text(5, 10, get.cell.meta.data("sector.index"))
})

circos.trackPlotRegion(sectors, ylim = c(0, 10))
circos.axis(sector.index = "a")
circos.axis(sector.index = "b", direction = "inside", labels.facing = "outside")
circos.axis(sector.index = "c", h = "bottom")
circos.axis(sector.index = "d", h = "bottom", direction = "inside",
    labels.facing = "reverse.clockwise")
circos.axis(sector.index = "e", h = 5, major.at = c(1, 3, 5, 7, 9))
circos.axis(sector.index = "f", h = 5, major.at = c(1, 3, 5, 7, 9),
    labels = c("a", "c", "e", "g", "f"), minor.ticks = 0)
circos.axis(sector.index = "g", h = 5, major.at = c(1, 3, 5, 7, 9),
    labels = c("a1", "c1", "e1", "g1", "f1"), major.tick = FALSE,
    labels.facing = "reverse.clockwise")
circos.axis(sector.index = "h", h = 2, major.at = c(1, 3, 5, 7, 9),
    labels = c("a1", "c1", "e1", "g1", "f1"), minor.ticks = 2, 
    major.tick.length = mm_y(5), labels.facing = "clockwise")
circos.clear()

if(FALSE) {

############### real-time clock #################
factors = letters[1]

circos.par("gap.degree" = 0, "cell.padding" = c(0, 0, 0, 0), "start.degree" = 90)
circos.initialize(sectors, xlim = c(0, 12))
circos.trackPlotRegion(sectors, ylim = c(0, 1), bg.border = NA)
circos.axis(sector.index = "a", major.at = 0:12, labels = "",
    direction = "inside", major.tick.length = mm_y(3))
circos.text(1:12, rep(0.5, 12), 1:12, facing = "downward")

while(1) {
    current.time = as.POSIXlt(Sys.time())
    sec = ceiling(current.time$sec)
    min = current.time$min
    hour = current.time$hour

    # erase the clock hands
    draw.sector(rou1 = 0.8, border = "white", col = "white")

    sec.degree = 90 - sec/60 * 360
    arrows(0, 0, cos(sec.degree/180*pi)*0.8, sin(sec.degree/180*pi)*0.8)

    min.degree = 90 - min/60 * 360
    arrows(0, 0, cos(min.degree/180*pi)*0.7, sin(min.degree/180*pi)*0.7, lwd = 2)

    hour.degree = 90 - hour/12 * 360 - min/60 * 360/12
    arrows(0, 0, cos(hour.degree/180*pi)*0.4, sin(hour.degree/180*pi)*0.4, lwd = 2)

    Sys.sleep(1)
}
circos.clear()
}
</code></pre>

<hr>
<h2 id='circos.barplot'>
Draw barplots
</h2><span id='topic+circos.barplot'></span>

<h3>Description</h3>

<p>Draw barplots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.barplot(value, pos, bar_width = 0.6,
    col = NA, border = "black", lwd = par("lwd"), lty = par("lty"),
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.barplot_+3A_value">value</code></td>
<td>
<p>A numeric vector or a matrix. If it is a matrix, columns correspond to the height of bars.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_pos">pos</code></td>
<td>
<p>Positions of the bars.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_bar_width">bar_width</code></td>
<td>
<p>Width of bars. It assumes the bars locating at <code>x = 1, 2, ...</code>.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_col">col</code></td>
<td>
<p>Filled color of bars.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_border">border</code></td>
<td>
<p>Color for the border.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_lty">lty</code></td>
<td>
<p>Line style.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="circos.barplot_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the input variable is a matrix, it draws a stacked barplot.
</p>
<p>Please note, the x-values of barplots are normally integer indices. Just be careful
when initializing the circular layout.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    value = runif(10)
    circos.barplot(value, 1:10 - 0.5, col = 1:10)
})
circos.track(ylim = c(-1, 1), panel.fun = function(x, y) {
    value = runif(10, min = -1, max = 1)
    circos.barplot(value, 1:10 - 0.5, col = ifelse(value &gt; 0, 2, 3))
})
circos.clear()

circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 4), panel.fun = function(x, y) {
    value = matrix(runif(10*4), ncol = 4)
    circos.barplot(value, 1:10 - 0.5, col = 2:5)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.boxplot'>
Draw boxplots
</h2><span id='topic+circos.boxplot'></span>

<h3>Description</h3>

<p>Draw boxplots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.boxplot(value, pos, outline = TRUE, box_width = 0.6,
    col = NA, border = "black", lwd = par("lwd"), lty = par("lty"),
    cex = par("cex"), pch = 1, pt.col = par("col"),
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.boxplot_+3A_value">value</code></td>
<td>
<p>A numeric vector, a matrix or a list. If it is a matrix, boxplots are made by columns (each column is a box).</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_pos">pos</code></td>
<td>
<p>Positions of the boxes.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_outline">outline</code></td>
<td>
<p>Whether to draw outliers.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_box_width">box_width</code></td>
<td>
<p>Width of boxes.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_col">col</code></td>
<td>
<p>Filled color of boxes.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_border">border</code></td>
<td>
<p>Color for the border as well as the quantile lines.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_lty">lty</code></td>
<td>
<p>Line style</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_cex">cex</code></td>
<td>
<p>Point size.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_pch">pch</code></td>
<td>
<p>Point type.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_pt.col">pt.col</code></td>
<td>
<p>Point color.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="circos.boxplot_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note, the x-values of boxplots are normally integer indices. Just be careful
when initializing the circular layout.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    for(pos in seq(0.5, 9.5, by = 1)) {
        value = runif(10)
        circos.boxplot(value, pos)
    }
})
circos.clear()

circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    value = replicate(runif(10), n = 10, simplify = FALSE)
    circos.boxplot(value, 1:10 - 0.5, col = 1:10)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.clear'>
Reset the circular layout parameters
</h2><span id='topic+circos.clear'></span>

<h3>Description</h3>

<p>Reset the circular layout parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.clear()
</code></pre>


<h3>Details</h3>

<p>Because there are several
parameters for the circular plot which can only be set before <code><a href="#topic+circos.initialize">circos.initialize</a></code>. So before you draw the next
circular plot, you need to reset all these parameters.
</p>
<p>If you meet some errors when re-drawing the circular plot, try running this function and it will solve most of the problems.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.connect'>
Draw connecting lines/ribons between two sets of points
</h2><span id='topic+circos.connect'></span>

<h3>Description</h3>

<p>Draw connecting lines/ribons between two sets of points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.connect(x0, y0, x1, y1,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    type = c("normal", "segments", "bezier"),
    segments.ratio = c(1, 1, 1),
    col = par("col"),
    border = "black",
    lwd = par("lwd"),
    lty = par("lty"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.connect_+3A_x0">x0</code></td>
<td>
<p>x coordinates for point set 1. The value can also be a two-column matrix.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_y0">y0</code></td>
<td>
<p>y coordinates for point set 1.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_x1">x1</code></td>
<td>
<p>x coordinates for point set 2. The value can also be a two-column matrix.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_y1">y1</code></td>
<td>
<p>y coordinates for point set 2.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_type">type</code></td>
<td>
<p>Which type of connections. Values can be &quot;normal&quot;, &quot;segments&quot; and &quot;bezier&quot;.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_segments.ratio">segments.ratio</code></td>
<td>
<p>When <code>type</code> is set to <code>segments</code>, each connecting line is segmented into three parts. This argument controls the length of the three parts of sub-segments.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_col">col</code></td>
<td>
<p>Color of the segments.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_border">border</code></td>
<td>
<p>Border color of the links.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_lwd">lwd</code></td>
<td>
<p>Line width of the segments.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_lty">lty</code></td>
<td>
<p>Line type of the segments.</p>
</td></tr>
<tr><td><code id="circos.connect_+3A_...">...</code></td>
<td>
<p>Other arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initialize(c("a"), xlim = c(0, 1))
circos.track(ylim = c(0, 1), track.height = 0.7, bg.border = NA, 
    panel.fun = function(x, y) {
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[1], 2), col = "#CCCCCC")
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[2], 2), col = "#CCCCCC")
    x0 = runif(100)
    x1 = runif(100)

    circos.connect(x0, 0, x1, 1, 
        type = "normal", border = NA,
        col = rand_color(100, luminosity = "bright", transparency = 0.75))
})

circos.initialize(c("a"), xlim = c(0, 1))
circos.track(ylim = c(0, 1), track.height = 0.7, bg.border = NA, 
    panel.fun = function(x, y) {
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[1], 2), col = "#CCCCCC")
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[2], 2), col = "#CCCCCC")
    x0 = runif(100)
    x1 = runif(100)

    circos.connect(x0, 0, x1, 1, 
        type = "bezier", border = NA,
        col = rand_color(100, luminosity = "bright", transparency = 0.75))
})

circos.initialize(c("a"), xlim = c(0, 1))
circos.track(ylim = c(0, 1), track.height = 0.7, bg.border = NA, 
    panel.fun = function(x, y) {
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[1], 2), col = "#CCCCCC")
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[2], 2), col = "#CCCCCC")
    x0 = sort(runif(200))
    x0 = matrix(x0, ncol = 2, byrow = TRUE)
    x1 = sort(runif(200))
    x1 = matrix(x1, ncol = 2, byrow = TRUE)

    circos.connect(x0, 0, x1, 1, 
        type = "normal", border = NA,
        col = rand_color(100, luminosity = "bright", transparency = 0.5))
})

circos.initialize(c("a"), xlim = c(0, 1))
circos.track(ylim = c(0, 1), track.height = 0.7, bg.border = NA, 
    panel.fun = function(x, y) {
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[1], 2), col = "#CCCCCC")
    circos.lines(CELL_META$cell.xlim, rep(CELL_META$cell.ylim[2], 2), col = "#CCCCCC")
    x0 = sort(runif(500))
    x0 = matrix(x0, ncol = 2, byrow = TRUE)
    x0 = x0[sample(nrow(x0), nrow(x0)), ]
    x1 = sort(runif(500))
    x1 = matrix(x1, ncol = 2, byrow = TRUE)
    x1 = x1[sample(nrow(x1), nrow(x1)), ]

    l = abs(x0[, 1] - x1[, 1]) &lt; 0.5

    circos.connect(x0[l ,], 0, x1[l, ], 1, 
        type = "bezier", border = NA,
        col = rand_color(sum(l), luminosity = "bright", transparency = 0.5))
})

</code></pre>

<hr>
<h2 id='circos.dendrogram'>
Add circular dendrograms
</h2><span id='topic+circos.dendrogram'></span>

<h3>Description</h3>

<p>Add circular dendrograms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.dendrogram(
    dend,
    facing = c("outside", "inside"),
    max_height = NULL,
    use_x_attr = FALSE,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.dendrogram_+3A_dend">dend</code></td>
<td>
<p>A <code><a href="stats.html#topic+dendrogram">dendrogram</a></code> object.</p>
</td></tr>
<tr><td><code id="circos.dendrogram_+3A_facing">facing</code></td>
<td>
<p>Is the dendromgrams facing inside to the circle or outside?</p>
</td></tr>
<tr><td><code id="circos.dendrogram_+3A_max_height">max_height</code></td>
<td>
<p>Maximum height of the dendrogram. This is important if more than one dendrograms are drawn in one track and making them comparable. The height of a dendrogram can be obtained by <code>attr(dend, "height")</code>.</p>
</td></tr>
<tr><td><code id="circos.dendrogram_+3A_use_x_attr">use_x_attr</code></td>
<td>
<p>Whether use the <code>x</code> attribute to determine node positions in the dendrogram, used internally.</p>
</td></tr>
<tr><td><code id="circos.dendrogram_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="circos.dendrogram_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assuming there are <code>n</code> nodes in the dendrogram, the positions for leaves on x-axis are always <code>0.5, 1.5, ..., n - 0.5</code>.
So you must be careful with <code>xlim</code> when you initialize the cirular layout.
</p>
<p>You can use the <code>dendextend</code> package to render the dendrograms.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-plots.html#phylogenetic-trees">https://jokergoo.github.io/circlize_book/book/high-level-plots.html#phylogenetic-trees</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file(package = "circlize", "extdata", "bird.orders.RData"))

labels = hc$labels  # name of birds
ct = cutree(hc, 6)  # cut tree into 6 pieces
n = length(labels)  # number of bird species
dend = as.dendrogram(hc)

circos.par(cell.padding = c(0, 0, 0, 0))
circos.initialize(sectors = "a", xlim = c(0, n)) # only one sector
max_height = attr(dend, "height")  # maximum height of the trees
circos.trackPlotRegion(ylim = c(0, 1), bg.border = NA, track.height = 0.3,
    panel.fun = function(x, y) {
        for(i in seq_len(n)) {
            circos.text(i-0.5, 0, labels[i], adj = c(0, 0.5),
                facing = "clockwise", niceFacing = TRUE,
                col = ct[labels[i]], cex = 0.7)
        }
})

suppressPackageStartupMessages(require(dendextend))
dend = color_branches(dend, k = 6, col = 1:6)

circos.trackPlotRegion(ylim = c(0, max_height), bg.border = NA,
    track.height = 0.4, panel.fun = function(x, y) {
        circos.dendrogram(dend, max_height = max_height)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.genomicAxis'>
Add genomic axes
</h2><span id='topic+circos.genomicAxis'></span>

<h3>Description</h3>

<p>Add genomic axes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicAxis(
    h = "top",
    major.at = NULL,
    labels = NULL,
    major.by = NULL,
    tickLabelsStartFromZero = TRUE,
    labels.cex = 0.4*par("cex"),
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicAxis_+3A_h">h</code></td>
<td>
<p>Position of the axes. &quot;top&quot; or &quot;bottom&quot;.
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_major.at">major.at</code></td>
<td>
<p>Major breaks. If <code>major.at</code> is set, <code>major.by</code> is ignored.
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_labels">labels</code></td>
<td>
<p>labels corresponding to <code>major.at</code>. If <code>labels</code> is set, <code>major.at</code> must be set.
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_major.by">major.by</code></td>
<td>
<p>Increment of major ticks. It is calculated automatically if the value is not set (about every 10 degrees there is a major tick).
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_ticklabelsstartfromzero">tickLabelsStartFromZero</code></td>
<td>
<p>Whether axis tick labels start from 0? This will only affect the axis labels while not affect x-values in cells.
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_labels.cex">labels.cex</code></td>
<td>
<p>The font size for the axis tick labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track
</p>
</td></tr>
<tr><td><code id="circos.genomicAxis_+3A_...">...</code></td>
<td>
<p>Other arguments pass to <code><a href="#topic+circos.axis">circos.axis</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It assigns proper tick labels under genomic coordinate.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-axes">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-axes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initializeWithIdeogram(chromosome.index = paste0("chr", 1:4), plotType = NULL)
circos.track(ylim = c(0, 1), panel.fun = function(x, y) circos.genomicAxis())
circos.track(ylim = c(0, 1), track.height = 0.1)
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    circos.genomicAxis(h = "bottom", direction = "inside")
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.genomicDensity'>
Calculate and add genomic density track
</h2><span id='topic+circos.genomicDensity'></span>

<h3>Description</h3>

<p>Calculate and add genomic density track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicDensity(
    data,
    ylim.force = FALSE,
    window.size = NULL,
    overlap = TRUE,
    count_by = c("percent", "number"),
    col = ifelse(area, "grey", "black"),
    lwd = par("lwd"),
    lty = par("lty"),
    type = "l",
    area = TRUE,
    area.baseline = NULL,
    baseline = 0,
    border = NA,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicDensity_+3A_data">data</code></td>
<td>
<p>A bed-file-like data frame or a list of data frames. If the input is a list of data frames.
there will be multiple density plot in one same track.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_ylim.force">ylim.force</code></td>
<td>
<p>Whether to force upper bound of <code>ylim</code> to be 1. Ignored if <code>count_by</code> is set to <code>number</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_window.size">window.size</code></td>
<td>
<p>Pass to <code><a href="#topic+genomicDensity">genomicDensity</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_overlap">overlap</code></td>
<td>
<p>Pass to <code><a href="#topic+genomicDensity">genomicDensity</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_count_by">count_by</code></td>
<td>
<p>Pass to <code><a href="#topic+genomicDensity">genomicDensity</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_col">col</code></td>
<td>
<p>Colors. It should be length of one. If <code>data</code> is a list of data frames, the length of <code>col</code>
can also be the length of the list. If multiple sets of genomic regions are visualized in one
single track, you should set the colors with transparency to distinguish them.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_lwd">lwd</code></td>
<td>
<p>Width of lines, the same setting as <code>col</code> argument.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_lty">lty</code></td>
<td>
<p>Style of lines, the same setting as <code>col</code> argument.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_type">type</code></td>
<td>
<p>Type of lines, see <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_area">area</code></td>
<td>
<p>See <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_area.baseline">area.baseline</code></td>
<td>
<p>Deprecated, use <code>baseline</code> instead.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_baseline">baseline</code></td>
<td>
<p>See <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_border">border</code></td>
<td>
<p>See <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicDensity_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a high-level graphical function, and it will create a new track.
</p>
<p>If you have multiple sets of genomic regions, you should make sure the density ranges 
for all sets are similar, or I suggest you should put them into different tracks. One example
can be found in the &quot;Examples&quot; Section where the density range for <code>bed_list[[2]]</code> is too high
compared to the range for <code>bed_list[[1]]</code>, thus, it is better to put the two sets of
regions into two separate tracks.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-density-and-rainfall-plot">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-density-and-rainfall-plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file(package = "circlize", "extdata", "DMR.RData"))

# rainfall

circos.initializeWithIdeogram(plotType = c("axis", "labels"))

bed_list = list(DMR_hyper, DMR_hypo)
circos.genomicRainfall(bed_list, pch = 16, cex = 0.4, col = c("#FF000080", "#0000FF80"))

circos.genomicDensity(bed_list[[1]], col = c("#FF000080"), track.height = 0.1)
circos.genomicDensity(bed_list[[2]], col = c("#0000FF80"), track.height = 0.1)
circos.clear()

############ draw the two densities in one track  #############
circos.initializeWithIdeogram(plotType = c("axis", "labels"))
circos.genomicDensity(bed_list, col = c("#FF000080", "#0000FF80"), track.height = 0.2)
circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicHeatmap'>
Add heatmaps for selected regions
</h2><span id='topic+circos.genomicHeatmap'></span>

<h3>Description</h3>

<p>Add heatmaps for selected regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicHeatmap(
    bed,
    col,
    na_col = "grey",
    numeric.column = NULL,
    border = NA,
    border_lwd = par("lwd"),
    border_lty = par("lty"),
    connection_height = mm_h(5),
    line_col = par("col"),
    line_lwd = par("lwd"),
    line_lty = par("lty"),
    heatmap_height = 0.15,
    side = c("inside", "outside"),
    track.margin = circos.par("track.margin"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicHeatmap_+3A_bed">bed</code></td>
<td>
<p>A data frame in bed format, the matrix should be stored from the fourth column.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_col">col</code></td>
<td>
<p>Colors for the heatmaps. The value can be a matrix or a color mapping function generated by <code><a href="#topic+colorRamp2">colorRamp2</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_na_col">na_col</code></td>
<td>
<p>Color for NA values.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_numeric.column">numeric.column</code></td>
<td>
<p>Column index for the numeric columns. The values can be integer index or character index.
By default it takes all numeric columns from the fourth column.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_border">border</code></td>
<td>
<p>Border of the heatmap grids.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_border_lwd">border_lwd</code></td>
<td>
<p>Line width for borders of heatmap grids.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_border_lty">border_lty</code></td>
<td>
<p>Line style for borders of heatmap grids.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_connection_height">connection_height</code></td>
<td>
<p>Height of the connection lines. If it is set to <code>NULL</code>, no connection will be drawn.
Use <code><a href="#topic+mm_h">mm_h</a></code>/<code><a href="#topic+cm_h">cm_h</a></code>/<code><a href="#topic+inches_h">inches_h</a></code> to set a height in absolute unit.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_line_col">line_col</code></td>
<td>
<p>Color of the connection lines. The value can be a vector.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_line_lwd">line_lwd</code></td>
<td>
<p>Line width of the connection lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_line_lty">line_lty</code></td>
<td>
<p>Line style of the connection lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_heatmap_height">heatmap_height</code></td>
<td>
<p>Height of the heatmap track
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_side">side</code></td>
<td>
<p>Side of the heatmaps. Is the heatmap facing inside or outside?
</p>
</td></tr>
<tr><td><code id="circos.genomicHeatmap_+3A_track.margin">track.margin</code></td>
<td>
<p>Bottom and top margins.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function visualizes heatmaps which correspond to a subset of regions in the genome.
The correspondance between heatmaps and regions are identified by connection lines.
</p>
<p>The function actually creates two tracks, one track for the connection lines and one track
for the heamtaps. The heatmaps always fill the whole track.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-heatmap">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-heatmap</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initializeWithIdeogram(plotType = c("labels", "axis"))
bed = generateRandomBed(nr = 100, nc = 4)
col_fun = colorRamp2(c(-1, 0, 1), c("green", "black", "red"))
circos.genomicHeatmap(bed, col_fun, side = "inside", border = "white")
circos.genomicHeatmap(bed, col_fun, side = "outside", 
    line_col = as.numeric(factor(bed[[1]])))

</code></pre>

<hr>
<h2 id='circos.genomicIdeogram'>
Add an ideogram track
</h2><span id='topic+circos.genomicIdeogram'></span>

<h3>Description</h3>

<p>Add an ideogram track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicIdeogram(
    cytoband = system.file(package = "circlize", "extdata", "cytoBand.txt"),
    species = NULL,
    track.height = mm_h(2),
    track.margin = circos.par("track.margin"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicIdeogram_+3A_cytoband">cytoband</code></td>
<td>
<p>A data frame or a file path, pass to <code><a href="#topic+read.cytoband">read.cytoband</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicIdeogram_+3A_species">species</code></td>
<td>
<p>Abbreviations of the genome, pass to <code><a href="#topic+read.cytoband">read.cytoband</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicIdeogram_+3A_track.height">track.height</code></td>
<td>
<p>Height of the ideogram track.
</p>
</td></tr>
<tr><td><code id="circos.genomicIdeogram_+3A_track.margin">track.margin</code></td>
<td>
<p>Margins for the track.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#ideograms">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#ideograms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initializeWithIdeogram(plotType = c("labels", "axis"))
circos.track(ylim = c(0, 1))
circos.genomicIdeogram() # put ideogram as the third track

</code></pre>

<hr>
<h2 id='circos.genomicInitialize'>
Initialize circular plot with any genomic data
</h2><span id='topic+circos.genomicInitialize'></span>

<h3>Description</h3>

<p>Initialize circular plot with any genomic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicInitialize(
    data,
    sector.names = NULL,
    major.by = NULL,
    plotType = c("axis", "labels"),
    tickLabelsStartFromZero = TRUE,
    axis.labels.cex = 0.4*par("cex"),
    labels.cex = 0.8*par("cex"),
    track.height = NULL,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicInitialize_+3A_data">data</code></td>
<td>
<p>A data frame in bed format.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_sector.names">sector.names</code></td>
<td>
<p>Labels for each sectors which will be drawn along each sector. It will not modify values of sector index.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_major.by">major.by</code></td>
<td>
<p>Increment of major ticks. It is calculated automatically if the value is not set (about every 10 degrees there is a major tick).
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_plottype">plotType</code></td>
<td>
<p>If it is not <code>NULL</code>, there will create a new track containing axis and names for sectors.
This argument controls which part should be drawn, <code>axis</code> for genomic axis and <code>labels</code> for chromosome names
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_ticklabelsstartfromzero">tickLabelsStartFromZero</code></td>
<td>
<p>Whether axis tick labels start from 0? This will only affect the axis labels while not affect x-values in cells.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_axis.labels.cex">axis.labels.cex</code></td>
<td>
<p>The font size for the axis tick labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_labels.cex">labels.cex</code></td>
<td>
<p>The font size for the labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_track.height">track.height</code></td>
<td>
<p>If <code>PlotType</code> is not <code>NULL</code>, height of the annotation track.
</p>
</td></tr>
<tr><td><code id="circos.genomicInitialize_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.initialize">circos.initialize</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will initialize circular plot from genomic data. If <code>plotType</code> is set with value in <code>axis</code> or <code>labels</code>, there will
create a new track.
</p>
<p>The order of sectors related to data structure of <code>data</code>. If the first column in <code>data</code> is a factor, the order of sectors
is <code>levels(data[[1]])</code>; If the first column is just a simple vector, the order of sectors is <code>unique(data[[1]]</code>.
</p>
<p>For more details on initializing genomic plot, please refer to the vignettes.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/initialize-genomic-plot.html#initialize-with-general-genomic-category">https://jokergoo.github.io/circlize_book/book/initialize-genomic-plot.html#initialize-with-general-genomic-category</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = read.cytoband()$df
circos.genomicInitialize(df)

df = data.frame(name = c("TP53", "TP63", "TP73"),
                start = c(7565097, 189349205, 3569084),
                end = c(7590856, 189615068, 3652765),
                stringsAsFactors = FALSE)
circos.genomicInitialize(df)
circos.clear()

circos.genomicInitialize(df, tickLabelsStartFromZero = FALSE)
circos.clear()

circos.genomicInitialize(df, major.by = 5000)
circos.clear()

circos.genomicInitialize(df, plotType = "labels")
circos.clear()

circos.genomicInitialize(df, sector.names = c("tp53", "tp63", "tp73"))
circos.clear()

circos.genomicInitialize(df, sector.names = c("tp53x", "tp63x", "tp73"))
circos.clear()

df[[1]] = factor(df[[1]], levels = c("TP73", "TP63", "TP53"))
circos.genomicInitialize(df)
circos.clear()
</code></pre>

<hr>
<h2 id='circos.genomicLabels'>
Add labels to specified genomic regions
</h2><span id='topic+circos.genomicLabels'></span>

<h3>Description</h3>

<p>Add labels to specified genomic regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicLabels(
    bed,
    labels = NULL,
    labels.column = NULL,
    facing = "clockwise",
    niceFacing = TRUE,
    col = par("col"),
    cex = 0.8,
    font = par("font"),
    padding = 0.4,
    connection_height = mm_h(5),
    line_col = par("col"),
    line_lwd = par("lwd"),
    line_lty = par("lty"),
    labels_height = min(c(cm_h(1.5), max(strwidth(labels, cex = cex, font = font)))),
    side = c("inside", "outside"),
    labels.side = side,
    track.margin = circos.par("track.margin"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicLabels_+3A_bed">bed</code></td>
<td>
<p>A data frame in bed format.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_labels">labels</code></td>
<td>
<p>A vector of labels corresponding to rows in <code>bed</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_labels.column">labels.column</code></td>
<td>
<p>If the label column is already in <code>bed</code>, the index for this column in <code>bed</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_facing">facing</code></td>
<td>
<p>fFacing of the labels. The value can only be <code>"clockwise"</code> or <code>"reverse.clockwise"</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Whether automatically adjust the facing of the labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_col">col</code></td>
<td>
<p>Color for the labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_cex">cex</code></td>
<td>
<p>Size of the labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_font">font</code></td>
<td>
<p>Font of the labels.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_padding">padding</code></td>
<td>
<p>Padding of the labels, the value is the ratio to the height of the label.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_connection_height">connection_height</code></td>
<td>
<p>Height of the connection track.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_line_col">line_col</code></td>
<td>
<p>Color for the connection lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_line_lwd">line_lwd</code></td>
<td>
<p>Line width for the connection lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_line_lty">line_lty</code></td>
<td>
<p>Line type for the connectioin lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_labels_height">labels_height</code></td>
<td>
<p>Height of the labels track.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_side">side</code></td>
<td>
<p>Side of the labels track, is it in the inside of the track where the regions are marked?
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_labels.side">labels.side</code></td>
<td>
<p>Same as <code>side</code>. It will replace <code>side</code> in the future versions.
</p>
</td></tr>
<tr><td><code id="circos.genomicLabels_+3A_track.margin">track.margin</code></td>
<td>
<p>Bottom and top margins.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function adds labels for the specified regions. The positions of labels are arranged
so that they are not overlapping to each other.
</p>
<p>This function creates two tracks, one for the connection lines and one for the labels.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#labels">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#labels</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initializeWithIdeogram()
bed = generateRandomBed(nr = 50, fun = function(k) sample(letters, k, replace = TRUE))
bed[1, 4] = "aaaaa"
circos.genomicLabels(bed, labels.column = 4, side = "inside")
circos.clear()

circos.initializeWithIdeogram(plotType = NULL)
circos.genomicLabels(bed, labels.column = 4, side = "outside",
    col = as.numeric(factor(bed[[1]])), line_col = as.numeric(factor(bed[[1]])))
circos.genomicIdeogram()
circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicLines'>
Add lines to a plotting region, specifically for genomic graphics
</h2><span id='topic+circos.genomicLines'></span>

<h3>Description</h3>

<p>Add lines to a plotting region, specifically for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicLines(
    region,
    value,
    numeric.column = NULL,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    posTransform = NULL,
    col = ifelse(area, "grey", "black"),
    lwd = par("lwd"),
    lty = par("lty"),
    type = "l",
    area = FALSE,
    area.baseline = NULL,
    border = "black",
    baseline = "bottom",
    pt.col = par("col"),
    cex = par("cex"),
    pch = par("pch"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicLines_+3A_region">region</code></td>
<td>
<p>A data frame contains 2 column which correspond to start positions and end positions.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_value">value</code></td>
<td>
<p>A data frame contains values and other information.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_numeric.column">numeric.column</code></td>
<td>
<p>Which column in <code>value</code> data frame should be taken as y-value.
If it is not defined, the whole numeric columns in <code>value</code> will be taken.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_postransform">posTransform</code></td>
<td>
<p>Self-defined function to transform genomic positions, see <code><a href="#topic+posTransform.default">posTransform.default</a></code> for explaination.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_col">col</code></td>
<td>
<p>col of lines/areas. If there are more than one numeric column, the length of <code>col</code> can be either one or number of numeric columns.
If there is only one numeric column and type is either <code>segment</code> or <code>h</code>, 
the length of <code>col</code> can be either one or number of rows of <code>region</code>.
pass to <code><a href="#topic+circos.lines">circos.lines</a></code>
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_lwd">lwd</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_lty">lty</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_type">type</code></td>
<td>
<p>There is an additional option <code>segment</code> which plot segment lines from start position to end position. Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>. 
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_area">area</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_area.baseline">area.baseline</code></td>
<td>
<p>Deprecated, use <code>baseline</code> instead.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_baseline">baseline</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_border">border</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_pt.col">pt.col</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_cex">cex</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_pch">pch</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLines_+3A_...">...</code></td>
<td>
<p>Mysterious parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a low-level graphical function and usually is put in <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrack">circos.genomicTrack</a></code>.
</p>
<p>The function behaviours differently from different formats of input, see the examples in 
the &quot;Examples&quot; Section or go to <a href="https://jokergoo.github.io/circlize_book/book/modes-of-input.html">https://jokergoo.github.io/circlize_book/book/modes-of-input.html</a> for more details.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### test bed
circos.par("track.height" = 0.1)
circos.initializeWithIdeogram(plotType = NULL)

bed = generateRandomBed(nr = 100)
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicLines(region, value, type = "l", ...)
})

bed1 = generateRandomBed(nr = 100)
bed2 = generateRandomBed(nr = 100)
bed_list = list(bed1, bed2)

circos.genomicTrack(bed_list, panel.fun = function(region, value, ...) {
    i = getI(...)
    circos.genomicLines(region, value, col = i, ...)
})

circos.genomicTrack(bed_list, stack = TRUE, 
    panel.fun = function(region, value, ...) {
    i = getI(...)
    circos.genomicLines(region, value, col = i, ...)
})

bed = generateRandomBed(nr = 100, nc = 4)
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicLines(region, value, col = 1:4, ...)
})

circos.genomicTrack(bed, stack = TRUE, panel.fun = function(region, value, ...) {
    i = getI(...)
    circos.genomicLines(region, value, col = i, ...)
})

bed = generateRandomBed(nr = 100)
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicLines(region, value, type = "segment", lwd = 2, ...)
})

circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicLink'>
Add links from two sets of genomic positions
</h2><span id='topic+circos.genomicLink'></span>

<h3>Description</h3>

<p>Add links from two sets of genomic positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicLink(
    region1,
    region2,
    rou = get_most_inside_radius(),
    rou1 = rou,
    rou2 = rou,
    col = "black",
    lwd = par("lwd"),
    lty = par("lty"),
    border = col,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicLink_+3A_region1">region1</code></td>
<td>
<p>A data frame in bed format.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_region2">region2</code></td>
<td>
<p>A data frame in bed format.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_rou">rou</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_rou1">rou1</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_rou2">rou2</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_col">col</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>, length can be either one or nrow of <code>region1</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_lwd">lwd</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>, length can be either one or nrow of <code>region1</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_lty">lty</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>, length can be either one or nrow of <code>region1</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_border">border</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>, length can be either one or nrow of <code>region1</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicLink_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Of course, number of rows should be same in <code>region1</code> and <code>region2</code>.
</p>
<p>If you want to have more controls on links, please use <code><a href="#topic+circos.link">circos.link</a></code> directly.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/genomic-plotting-region.html#genomic-links">https://jokergoo.github.io/circlize_book/book/genomic-plotting-region.html#genomic-links</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)

bed1 = generateRandomBed(nr = 100)
bed1 = bed1[sample(nrow(bed1), 20), ]
bed2 = generateRandomBed(nr = 100)
bed2 = bed2[sample(nrow(bed2), 20), ]
circos.par("track.height" = 0.1, cell.padding = c(0, 0, 0, 0))
circos.initializeWithIdeogram()

circos.genomicLink(bed1, bed2, col = sample(1:5, 20, replace = TRUE), border = NA)
circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicPoints'>
Add points to a plotting region, specifically for genomic graphics
</h2><span id='topic+circos.genomicPoints'></span>

<h3>Description</h3>

<p>Add points to a plotting region, specifically for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicPoints(
    region,
    value,
    numeric.column = NULL,
    sector.index = get.cell.meta.data("sector.index"),
    track.index = get.cell.meta.data("track.index"),
    posTransform = NULL,
    pch = par("pch"),
    col = par("col"),
    cex = par("cex"),
    bg = par("bg"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicPoints_+3A_region">region</code></td>
<td>
<p>A data frame contains 2 columns which correspond to start positions and end positions.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_value">value</code></td>
<td>
<p>A data frame contains values and other information.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_numeric.column">numeric.column</code></td>
<td>
<p>Which column in <code>value</code> data frame should be taken as y-value.
If it is not defined, the whole numeric columns in <code>value</code> will be taken.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_postransform">posTransform</code></td>
<td>
<p>Self-defined function to transform genomic positions, see <code><a href="#topic+posTransform.default">posTransform.default</a></code> for explanation
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_col">col</code></td>
<td>
<p>Color of points. If there is only one numeric column, the length of <code>col</code> can be either one or number of rows of <code>region</code>.
If there are more than one numeric column, the length of <code>col</code> can be either one or number of numeric columns.
Pass to <code><a href="#topic+circos.points">circos.points</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_pch">pch</code></td>
<td>
<p>Type of points. Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.points">circos.points</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_cex">cex</code></td>
<td>
<p>Size of points. Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.points">circos.points</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_bg">bg</code></td>
<td>
<p>Background colors for points.
</p>
</td></tr>
<tr><td><code id="circos.genomicPoints_+3A_...">...</code></td>
<td>
<p>Mysterious parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a low-level graphical function and usually is put in <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrack">circos.genomicTrack</a></code>.
</p>
<p>The function behaviours differently from different formats of input, see the examples in 
the &quot;Examples&quot; Section or go to <a href="https://jokergoo.github.io/circlize_book/book/modes-of-input.html">https://jokergoo.github.io/circlize_book/book/modes-of-input.html</a> for more details.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.par("track.height" = 0.1)
circos.initializeWithIdeogram(plotType = NULL)

bed = generateRandomBed(nr = 100)
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicPoints(region, value, pch = 16, cex = 0.5, ...)
})

circos.genomicTrack(bed, stack = TRUE, panel.fun = function(region, value, ...) {
    circos.genomicPoints(region, value, pch = 16, cex = 0.5, ...)
    i = getI(...)
    cell.xlim = get.cell.meta.data("cell.xlim")
    circos.lines(cell.xlim, c(i, i), lty = 2, col = "#00000040")
})

bed1 = generateRandomBed(nr = 100)
bed2 = generateRandomBed(nr = 100)
bed_list = list(bed1, bed2)

# data frame list
circos.genomicTrack(bed_list, panel.fun = function(region, value, ...) {
    cex = (value[[1]] - min(value[[1]]))/(max(value[[1]]) - min(value[[1]]))
    i = getI(...)
    circos.genomicPoints(region, value, cex = cex, pch = 16, col = i, ...)
})

circos.genomicTrack(bed_list, stack = TRUE,
    panel.fun = function(region, value, ...) {
    cex = (value[[1]] - min(value[[1]]))/(max(value[[1]]) - min(value[[1]]))
    i = getI(...)
    circos.genomicPoints(region, value, cex = cex, pch = 16, col = i, ...)
    cell.xlim = get.cell.meta.data("cell.xlim")
    circos.lines(cell.xlim, c(i, i), lty = 2, col = "#00000040")
})

bed = generateRandomBed(nr = 100, nc = 4)
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    cex = (value[[1]] - min(value[[1]]))/(max(value[[1]]) - min(value[[1]]))
    circos.genomicPoints(region, value, cex = 0.5, pch = 16, col = 1:4, ...)
})

circos.genomicTrack(bed, stack = TRUE, panel.fun = function(region, value, ...) {
    cex = (value[[1]] - min(value[[1]]))/(max(value[[1]]) - min(value[[1]]))
    i = getI(...)
    circos.genomicPoints(region, value, cex = cex, pch = 16, col = i, ...)
    cell.xlim = get.cell.meta.data("cell.xlim")
    circos.lines(cell.xlim, c(i, i), lty = 2, col = "#00000040")
})

circos.clear()
</code></pre>

<hr>
<h2 id='circos.genomicPosTransformLines'>
Add genomic position transformation lines between tracks
</h2><span id='topic+circos.genomicPosTransformLines'></span>

<h3>Description</h3>

<p>Add genomic position transformation lines between tracks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicPosTransformLines(
    data,
    track.height = 0.1,
    posTransform = NULL,
    horizontalLine = c("none", "top", "bottom", "both"),
    track.margin = c(0, 0),
    direction = c("inside", "outside"),
    col = "black",
    lwd = par("lwd"),
    lty = par("lty"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicPosTransformLines_+3A_data">data</code></td>
<td>
<p>A data frame containing genomic data.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_track.height">track.height</code></td>
<td>
<p>Height of the track.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_postransform">posTransform</code></td>
<td>
<p>Genomic position transformation function, see <code><a href="#topic+posTransform.default">posTransform.default</a></code> for an example.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_horizontalline">horizontalLine</code></td>
<td>
<p>Whether to draw horizontal lines which indicate region width .
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_track.margin">track.margin</code></td>
<td>
<p>Margin of tracks.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_direction">direction</code></td>
<td>
<p>Type of the transformation. <code>inside</code> means position transformed track are located inside 
and <code>outside</code> means position transformed track are located outside.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_col">col</code></td>
<td>
<p>Color of lines, can be length of one or nrow of <code>data</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_lwd">lwd</code></td>
<td>
<p>Width of lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_lty">lty</code></td>
<td>
<p>Style of lines.
</p>
</td></tr>
<tr><td><code id="circos.genomicPosTransformLines_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There is one representative situation when such position transformation needs to be applied. 
For example, there are two sets of regions in a chromosome in which regions in one set regions are
quite densely to each other and regions in other set are far from others. Heatmap or text is going
to be drawn on the next track. If there is no position transformation, heatmap or text for those
dense regions would be overlapped and hard to identify, also ugly to visualize. Thus, a way
to transform original positions to new positions would help for the visualization.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.genomicRainfall'>
Genomic rainfall plot
</h2><span id='topic+circos.genomicRainfall'></span>

<h3>Description</h3>

<p>Genomic rainfall plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicRainfall(
    data,
    mode = "min",
    ylim = NULL,
    col = "black",
    pch = par("pch"),
    cex = par("cex"),
    normalize_to_width = FALSE,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicRainfall_+3A_data">data</code></td>
<td>
<p>A bed-file-like data frame or a list of data frames.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_mode">mode</code></td>
<td>
<p>How to calculate the distance of two neighbouring regions, pass to <code><a href="#topic+rainfallTransform">rainfallTransform</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_ylim">ylim</code></td>
<td>
<p>ylim for rainfall plot track. If <code>normalize_to_width</code> is <code>FALSE</code>, the value should correspond to <code>log10(dist+1)</code>,
and if <code>normalize_to_width</code> is <code>TRUE</code>, the value should correspond to <code>log2(rel_dist)</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_col">col</code></td>
<td>
<p>Color of points. It should be length of one. If <code>data</code> is a list, the length of <code>col</code>
can also be the length of the list.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_pch">pch</code></td>
<td>
<p>Style of points.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_cex">cex</code></td>
<td>
<p>Size of points.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_normalize_to_width">normalize_to_width</code></td>
<td>
<p>If it is <code>TRUE</code>, the value is the relative distance divided by the width of the region.
</p>
</td></tr>
<tr><td><code id="circos.genomicRainfall_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is high-level graphical function, which mean, it will create a new track.
</p>
<p>Rainfall plot can be used to visualize distribution of regions. On the plot, y-axis
corresponds to the distance to neighbour regions (log-based). So if there is a drop-down on
the plot, it means there is a cluster of regions at that area.
</p>
<p>On the plot, y-axis are log10-transformed.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-density-and-rainfall-plot">https://jokergoo.github.io/circlize_book/book/high-level-genomic-functions.html#genomic-density-and-rainfall-plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
load(system.file(package = "circlize", "extdata", "DMR.RData"))

# rainfall
circos.initializeWithIdeogram(plotType = c("axis", "labels"))

bed_list = list(DMR_hyper, DMR_hypo)
circos.genomicRainfall(bed_list, pch = 16, cex = 0.4, col = c("#FF000080", "#0000FF80"))

circos.genomicDensity(bed_list[[1]], col = c("#FF000080"), track.height = 0.1)
circos.genomicDensity(bed_list[[2]], col = c("#0000FF80"), track.height = 0.1)

circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicRect'>
Draw rectangle-like grid, specifically for genomic graphics
</h2><span id='topic+circos.genomicRect'></span>

<h3>Description</h3>

<p>Draw rectangle-like grid, specifically for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicRect(
    region,
    value = NULL,
    ytop = NULL,
    ybottom = NULL,
    ytop.column = NULL,
    ybottom.column = NULL,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    posTransform = NULL,
    col = NA,
    border = "black",
    lty = par("lty"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicRect_+3A_region">region</code></td>
<td>
<p>A data frame contains 2 column which correspond to start positions and end positions.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_value">value</code></td>
<td>
<p>A data frame contains values and other information.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_ytop">ytop</code></td>
<td>
<p>A vector or a single value indicating top position of rectangles.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_ybottom">ybottom</code></td>
<td>
<p>A vector or a single value indicating bottom position of rectangles.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_ytop.column">ytop.column</code></td>
<td>
<p>If <code>ytop</code> is in <code>value</code>, the index of the column.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_ybottom.column">ybottom.column</code></td>
<td>
<p>If <code>ybottom</code> is in <code>value</code>, the index of the column.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_postransform">posTransform</code></td>
<td>
<p>Self-defined function to transform genomic positions, see <code><a href="#topic+posTransform.default">posTransform.default</a></code> for explaination.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_col">col</code></td>
<td>
<p>The length of <code>col</code> can be either one or number of rows of <code>region</code>. Pass to <code><a href="#topic+circos.rect">circos.rect</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_border">border</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.rect">circos.rect</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_lty">lty</code></td>
<td>
<p>Settings are similar as <code>col</code>. Pass to <code><a href="#topic+circos.rect">circos.rect</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicRect_+3A_...">...</code></td>
<td>
<p>Mysterious parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a low-level graphical function and usually is put in <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrack">circos.genomicTrack</a></code>.
</p>
<p>The function behaviours differently from different formats of input, see the examples in 
the &quot;Examples&quot; Section or go to <a href="https://jokergoo.github.io/circlize_book/book/modes-of-input.html">https://jokergoo.github.io/circlize_book/book/modes-of-input.html</a> for more details.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.par("track.height" = 0.1, cell.padding = c(0, 0, 0, 0))
circos.initializeWithIdeogram(plotType = NULL)

bed1 = generateRandomBed(nr = 100)
bed2 = generateRandomBed(nr = 100)
bed_list = list(bed1, bed2)
f = colorRamp2(breaks = c(-1, 0, 1), colors = c("green", "black", "red"))
circos.genomicTrack(bed_list, stack = TRUE,
    panel.fun = function(region, value, ...) {

    circos.genomicRect(region, value, col = f(value[[1]]), 
        border = NA, ...)
    i = getI(...)
    cell.xlim = get.cell.meta.data("cell.xlim")
    circos.lines(cell.xlim, c(i, i), lty = 2, col = "#000000")
})

circos.genomicTrack(bed_list, ylim = c(0, 3),
    panel.fun = function(region, value, ...) {
    i = getI(...)
    circos.genomicRect(region, value, ytop = i+0.4, ybottom = i-0.4, col = f(value[[1]]), 
        border = NA, ...)

    cell.xlim = get.cell.meta.data("cell.xlim")
    circos.lines(cell.xlim, c(i, i), lty = 2, col = "#000000")
})

circos.genomicTrack(bed1, panel.fun = function(region, value, ...) {
    circos.genomicRect(region, value, col = "red", border = NA, ...)

})

circos.genomicTrack(bed_list, panel.fun = function(region, value, ...) {
    i = getI(...)
    circos.genomicRect(region, value, col = i, border = NA, ...)

})

circos.clear()

</code></pre>

<hr>
<h2 id='circos.genomicText'>
Draw text in a cell, specifically for genomic graphics
</h2><span id='topic+circos.genomicText'></span>

<h3>Description</h3>

<p>Draw text in a cell, specifically for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicText(
    region,
    value = NULL,
    y = NULL,
    labels = NULL,
    labels.column = NULL,
    numeric.column = NULL,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    posTransform = NULL,
    direction = NULL,
    facing = "inside",
    niceFacing = FALSE,
    adj = par("adj"),
    cex = 1,
    col = "black",
    font = par("font"),
    padding = 0,
    extend = 0,
    align_to = "region",
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicText_+3A_region">region</code></td>
<td>
<p>A data frame contains 2 column which correspond to start positions and end positions.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_value">value</code></td>
<td>
<p>A data frame contains values and other information.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_y">y</code></td>
<td>
<p>A vector or a single value indicating position of text.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_labels">labels</code></td>
<td>
<p>Labels of text corresponding to each genomic positions.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_labels.column">labels.column</code></td>
<td>
<p>If labels are in <code>value</code>, index of column in <code>value</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_numeric.column">numeric.column</code></td>
<td>
<p>Which column in <code>value</code> data frame should be taken as y-value.
If it is not defined, only the first numeric columns in <code>value</code> will be taken.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_postransform">posTransform</code></td>
<td>
<p>Self-defined function to transform genomic positions, see <code><a href="#topic+posTransform.default">posTransform.default</a></code> for explanation.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_facing">facing</code></td>
<td>
<p>Passing to <code><a href="#topic+circos.text">circos.text</a></code>. Settings are similar as <code>col</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Should the facing of text be adjusted to fit human eyes?
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_direction">direction</code></td>
<td>
<p>Deprecated, use <code>facing</code> instead. 
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_adj">adj</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.text">circos.text</a></code>. Settings are similar as <code>col</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_cex">cex</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.text">circos.text</a></code>. Settings are similar as <code>col</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_col">col</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.text">circos.text</a></code>. The length of <code>col</code> can be either one or number of rows of <code>region</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_font">font</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.text">circos.text</a></code>. Settings are similar as <code>col</code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_padding">padding</code></td>
<td>
<p>pass to <code>posTransform</code> if it is set as <code><a href="#topic+posTransform.text">posTransform.text</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_extend">extend</code></td>
<td>
<p>pass to <code>posTransform</code> if it is set as <code><a href="#topic+posTransform.text">posTransform.text</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_align_to">align_to</code></td>
<td>
<p>pass to <code>posTransform</code> if it is set as <code><a href="#topic+posTransform.text">posTransform.text</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.genomicText_+3A_...">...</code></td>
<td>
<p>Mysterious parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a low-level graphical function and usually is put in <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrack">circos.genomicTrack</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.par("track.height" = 0.1, cell.padding = c(0, 0, 0, 0))
circos.initializeWithIdeogram(plotType = NULL)

bed = generateRandomBed(nr = 20)

circos.genomicTrack(bed, ylim = c(0, 1), panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, y = 0.5, labels = "text", ...)
})

bed = cbind(bed, sample(letters, nrow(bed), replace = TRUE))
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicText(region, value, labels.column = 2, ...)
})

circos.clear()
</code></pre>

<hr>
<h2 id='circos.genomicTrack'>
Create a track for genomic graphics
</h2><span id='topic+circos.genomicTrack'></span>

<h3>Description</h3>

<p>Create a track for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicTrack(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicTrack_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.genomicTrackPlotRegion">circos.genomicTrackPlotRegion</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>shortcut function of <code><a href="#topic+circos.genomicTrackPlotRegion">circos.genomicTrackPlotRegion</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.genomicTrackPlotRegion'>
Create a track for genomic graphics
</h2><span id='topic+circos.genomicTrackPlotRegion'></span>

<h3>Description</h3>

<p>Create a track for genomic graphics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.genomicTrackPlotRegion(
    data = NULL,
    ylim = NULL,
    stack = FALSE,
    numeric.column = NULL,
    jitter = 0,
    panel.fun = function(region, value, ...) {NULL},
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_data">data</code></td>
<td>
<p>A bed-file-like data frame or a list of data frames
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_ylim">ylim</code></td>
<td>
<p>If it is <code>NULL</code>, the value will be calculated from data. If <code>stack</code> is set to <code>TRUE</code>, this value is ignored.
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_stack">stack</code></td>
<td>
<p>whether to plot in a &quot;stack&quot; mode.
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_numeric.column">numeric.column</code></td>
<td>
<p>Columns of numeric values in <code>data</code> that will be used for plotting. 
If <code>data</code> is a data frame list, <code>numeric.column</code> should be either length of one or length of <code>data</code>.
If value of <code>numeric.column</code> is not set, its value will depend on the structure of <code>data</code>.
If <code>data</code> is a data frame, the default value for <code>numeric.column</code> is all the numeric column starting from the fourth column.
If <code>data</code> is a list of data frame, the default value for <code>numeric.column</code> is a vector which have the same length as <code>data</code>
and the value in default <code>numeric.column</code> is the index of the first numeric column in corresponding data frame.
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_jitter">jitter</code></td>
<td>
<p>Numeric. Only works for adding points in <code>circos.genomicTrackPlotRegion</code> under <code>stack</code> mode
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_panel.fun">panel.fun</code></td>
<td>
<p>Self-defined function which will be applied on each sector. Please not it is different
from that in <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>. In this function, there are two arguments (<code>region</code> and <code>value</code>) plus <code>...</code>.
In them, <code>region</code> is a two-column data frame with start positions and end positions in current genomic category (e.g. chromosome). 
<code>value</code> is a data frame which is derived from <code>data</code> but excluding the first three columns. Rows in <code>value</code> correspond to 
rows in <code>region</code>. <code>...</code> is mandatory and is used to pass internal parameters to other functions. The definition of
<code>value</code> will be different according to different input data (data frame or list of data frame) and different settings (stacked or not), 
please refer to 'details' section and vignettes to detailed explanation.
</p>
</td></tr>
<tr><td><code id="circos.genomicTrackPlotRegion_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Similar as <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>, users can add customized graphics by <code>panel.fun</code>, but the behaviour of <code>panel.fun</code>
will change depending on users' input data and <code>stack</code> setting.
</p>
<p>When <code>data</code> is a single data frame, <code>region</code> in <code>panel.fun</code> is a data frame containing the second and third column in <code>data</code> in 'current' genomic category (e.g. current chromosome).
<code>value</code> is also a data frame containing columns in <code>data</code> excluding the first three columns.
</p>
<p>When <code>data</code> is a list containing data frames, <code>panel.fun</code> will be applied iteratively on each data frame, thus, 
<code>region</code> is extracted from the data frame which is in the current iteration. For example, if <code>data</code> contains two data frames, <code>panel.fun</code>
will be applied with the first data frame in current chromosome and then applied with the second data frame in the same chromosome.
</p>
<p>If <code>stack</code> is set to <code>TRUE</code>, <code>ylim</code> will be re-defined. in <code>stack</code> mode, the y-axis will be splitted into several part
with equal height and graphics will be drawn on each 'horizontal' lines (y = 1, 2, ...). In this case:
</p>
<p>When <code>data</code> is a single data frame containing one or more numeric columns, each numeric column defined in <code>numeric.column</code> will be treated as a single unit. 
<code>ylim</code> is re-defined to <code>c(0.5, n+0.5)</code> in which <code>n</code> is number of numeric columns. <code>panel.fun</code> will be applied iteratively on each numeric column. In each
iteration, in <code>panel.fun</code>, <code>region</code> is still the genomic regions in current genomic category, but <code>value</code> contains current numeric column plus all non-numeric columns.
Under <code>stack</code> mode, in <code>panel.fun</code>, all low-level genomic graphical functions will draw on the 'horizontal line' <code>y = i</code> in which <code>i</code> is the index of current numeric column 
and the value of <code>i</code> can be obtained by <code><a href="#topic+getI">getI</a></code>.
</p>
<p>When <code>data</code> is a list containing data frames, each data frame will be treated as a single unit. The situation is quite similar as described in previous paragraph.
<code>ylim</code> is re-defined to <code>c(0.5, n+0.5)</code> in which <code>n</code> is number of data frames. <code>panel.fun</code> will be applied iteratively on each data frame. In each
iteration, in <code>panel.fun</code>, <code>region</code> is still the genomic regions in current genomic category, and <code>value</code> contains columns in current data frame excluding the first three columns.
Under <code>stack</code> mode, in <code>panel.fun</code>, all low-level genomic graphical functions will draw on the 'horizontal line' <code>y = i</code> in which <code>i</code> is the index of current data frame.
</p>
<p>Being different from <code>panel.fun</code> in <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>, there should be an additional argument <code>...</code> in <code>panel.fun</code>. This additional
argument is used to pass hidden values to low-level graphical functions. So if you are using functions like <code>circos.genomicPoints</code>, you should also
add <code>...</code> as an additional argument into <code>circos.genomicPoints</code>.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/genomic-plotting-region.html">https://jokergoo.github.io/circlize_book/book/genomic-plotting-region.html</a> and <a href="https://jokergoo.github.io/circlize_book/book/modes-of-input.html">https://jokergoo.github.io/circlize_book/book/modes-of-input.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.heatmap'>
Make circular heatmaps
</h2><span id='topic+circos.heatmap'></span>

<h3>Description</h3>

<p>Make circular heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.heatmap(mat, split = NULL, col, na.col = "grey",
    cell.border = NA, cell.lty = 1, cell.lwd = 1,
    bg.border = NA, bg.lty = par("lty"), bg.lwd = par("lwd"),
    ignore.white = is.na(cell.border),
    cluster = TRUE, clustering.method = "complete", distance.method = "euclidean",
    dend.callback = function(dend, m, si) reorder(dend, rowMeans(m)),
    dend.side = c("none", "outside", "inside"), dend.track.height = 0.1,
    rownames.side = c("none", "outside", "inside"), rownames.cex = 0.5,
    rownames.font = par("font"), rownames.col = "black",
    show.sector.labels = FALSE, cell_width = rep(1, nrow(mat)), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.heatmap_+3A_mat">mat</code></td>
<td>
<p>A matrix or a vector. The vector is transformed as a one-column matrix.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_split">split</code></td>
<td>
<p>A categorical variable. It splits the matrix into a list of matrices.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_col">col</code></td>
<td>
<p>If the values in the matrices are continuous, the color should be a color mapping generated by  <code><a href="#topic+colorRamp2">colorRamp2</a></code>. If the values are characters, the color should be a named color vector.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_na.col">na.col</code></td>
<td>
<p>Color for <code>NA</code> values.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_cell.border">cell.border</code></td>
<td>
<p>Border color of cells. A single scalar.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_cell.lty">cell.lty</code></td>
<td>
<p>Line type of cell borders. A single scalar.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_cell.lwd">cell.lwd</code></td>
<td>
<p>Line width of cell borders. A single scalar.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_bg.border">bg.border</code></td>
<td>
<p>Color for background border.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_bg.lty">bg.lty</code></td>
<td>
<p>Line type of the background border.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_bg.lwd">bg.lwd</code></td>
<td>
<p>Line width of the background border.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_ignore.white">ignore.white</code></td>
<td>
<p>Whether to draw the white color?</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_cluster">cluster</code></td>
<td>
<p>whether to apply clustering on rows. The value can also be a <code>dendrogram</code>/<code>hclust</code> object or other objects that can be converted to with <code><a href="stats.html#topic+dendrogram">as.dendrogram</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_clustering.method">clustering.method</code></td>
<td>
<p>Clustering method, pass to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_distance.method">distance.method</code></td>
<td>
<p>Distance method, pass to <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_dend.callback">dend.callback</code></td>
<td>
<p>A callback function that is applied to the dendrogram in every sector.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_dend.side">dend.side</code></td>
<td>
<p>Side of the dendrograms relative to the heatmap track.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_dend.track.height">dend.track.height</code></td>
<td>
<p>Track height of the dendrograms.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_rownames.side">rownames.side</code></td>
<td>
<p>Side of the row names relative to the heatmap track.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_rownames.cex">rownames.cex</code></td>
<td>
<p>Cex of row names.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_rownames.font">rownames.font</code></td>
<td>
<p>Font of row names.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_rownames.col">rownames.col</code></td>
<td>
<p>Color of row names.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_show.sector.labels">show.sector.labels</code></td>
<td>
<p>Whether to show sector labels.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_cell_width">cell_width</code></td>
<td>
<p>Relative widths of heatmap cells.</p>
</td></tr>
<tr><td><code id="circos.heatmap_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.track">circos.track</a></code> which draws the heatmap track.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/2020/05/21/make-circular-heatmaps/">https://jokergoo.github.io/2020/05/21/make-circular-heatmaps/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)
mat1 = rbind(cbind(matrix(rnorm(50*5, mean = 1), nr = 50), 
                   matrix(rnorm(50*5, mean = -1), nr = 50)),
             cbind(matrix(rnorm(50*5, mean = -1), nr = 50), 
                   matrix(rnorm(50*5, mean = 1), nr = 50))
            )
rownames(mat1) = paste0("R", 1:100)
colnames(mat1) = paste0("C", 1:10)
mat1 = mat1[sample(100, 100), ] # randomly permute rows
split = sample(letters[1:5], 100, replace = TRUE)
spilt = factor(split, levels = letters[1:5])
col_fun1 = colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))
circos.heatmap(mat1, split = split, col = col_fun1)
circos.clear()

</code></pre>

<hr>
<h2 id='circos.heatmap.initialize'>
Initialize circular heatmaps
</h2><span id='topic+circos.heatmap.initialize'></span>

<h3>Description</h3>

<p>Initialize circular heatmaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.heatmap.initialize(mat, split = NULL, cluster = TRUE,
    clustering.method = "complete", distance.method = "euclidean",
    dend.callback = function(dend, m, si) reorder(dend, rowMeans(m)),
    cell_width = rep(1, nrow(mat)))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.heatmap.initialize_+3A_mat">mat</code></td>
<td>
<p>A matrix or a vector. The vector is transformed as a one-column matrix.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_split">split</code></td>
<td>
<p>A categorical variable. It splits the matrix into a list of matrices.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_cluster">cluster</code></td>
<td>
<p>whether to apply clustering on rows. The value can also be a <code>dendrogram</code>/<code>hclust</code> object or other objects that can be converted to with <code><a href="stats.html#topic+dendrogram">as.dendrogram</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_clustering.method">clustering.method</code></td>
<td>
<p>Clustering method, pass to <code><a href="stats.html#topic+hclust">hclust</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_distance.method">distance.method</code></td>
<td>
<p>Distance method, pass to <code><a href="stats.html#topic+dist">dist</a></code>.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_dend.callback">dend.callback</code></td>
<td>
<p>A callback function that is applied to the dendrogram in every sector.</p>
</td></tr>
<tr><td><code id="circos.heatmap.initialize_+3A_cell_width">cell_width</code></td>
<td>
<p>Relative widths of heatmap cells.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/2020/05/21/make-circular-heatmaps/">https://jokergoo.github.io/2020/05/21/make-circular-heatmaps/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.heatmap.link'>
Draw a link between two matrix rows in the circular heatmap
</h2><span id='topic+circos.heatmap.link'></span>

<h3>Description</h3>

<p>Draw a link between two matrix rows in the circular heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.heatmap.link(row_from, row_to, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.heatmap.link_+3A_row_from">row_from</code></td>
<td>
<p>The row index where the link starts. The value should be length 1. If you want to draw multiple links, put the function in a <code>for</code> loop.</p>
</td></tr>
<tr><td><code id="circos.heatmap.link_+3A_row_to">row_to</code></td>
<td>
<p>The row index where the link ends.</p>
</td></tr>
<tr><td><code id="circos.heatmap.link_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.link">circos.link</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)
mat = matrix(rnorm(100*10), nrow = 100)
rownames(mat) = paste0("R", 1:100)
col_fun = colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))
circos.heatmap(mat, col = col_fun, rownames.side = "outside")
circos.heatmap.link(10, 60)
circos.clear()

split = sample(letters[1:5], 100, replace = TRUE)
circos.heatmap(mat, col = col_fun, split = split, 
	rownames.side = "outside")
circos.heatmap.link(10, 60)
circos.clear()

</code></pre>

<hr>
<h2 id='circos.info'>
Get information of the circular plot
</h2><span id='topic+circos.info'></span>

<h3>Description</h3>

<p>Get information of the circular plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.info(sector.index = NULL, track.index = NULL, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.info_+3A_sector.index">sector.index</code></td>
<td>
<p>Which sectors you want to look at? It can be a vector.
</p>
</td></tr>
<tr><td><code id="circos.info_+3A_track.index">track.index</code></td>
<td>
<p>Which tracks you want to look at? It can be a vector.
</p>
</td></tr>
<tr><td><code id="circos.info_+3A_plot">plot</code></td>
<td>
<p>Whether to add information on the plot.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It tells you the basic parameters for sectors/tracks/cells. If both <code>sector.index</code>
and <code>track.index</code> are set to <code>NULL</code>, the function would print index for
all sectors and all tracks. If <code>sector.index</code> and/or <code>track.index</code> are set,
the function would print <code>xlim</code>, <code>ylim</code>, <code>cell.xlim</code>, <code>cell.ylim</code>,
<code>xplot</code>, <code>yplot</code>, <code>cell.width</code>, <code>cell.height</code>, <code>track.margin</code> and <code>cell.padding</code> for every cell in specified sectors and tracks.
Also, the function will print index of your current sector and current track.
</p>
<p>If <code>plot</code> is set to <code>TRUE</code>, the function will plot the index of the sector and the track
for each cell on the figure.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html#circos-info-and-circos-clear">https://jokergoo.github.io/circlize_book/book/circular-layout.html#circos-info-and-circos-clear</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.initialize'>
Initialize the circular layout
</h2><span id='topic+circos.initialize'></span>

<h3>Description</h3>

<p>Initialize the circular layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.initialize(
    sectors = NULL,
    x = NULL,
    xlim = NULL,
    sector.width = NULL,
    factors = sectors,
    ring = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.initialize_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> variable or a character vector which represent data categories
</p>
</td></tr>
<tr><td><code id="circos.initialize_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. 
</p>
</td></tr>
<tr><td><code id="circos.initialize_+3A_x">x</code></td>
<td>
<p>Data on x-axes, a vector
</p>
</td></tr>
<tr><td><code id="circos.initialize_+3A_xlim">xlim</code></td>
<td>
<p>Ranges for values on x-axes, see &quot;details&quot; section for explanation of the format
</p>
</td></tr>
<tr><td><code id="circos.initialize_+3A_sector.width">sector.width</code></td>
<td>
<p>Width for each sector. The length of the vector should be either 1 which means
all sectors have same width or as same as the number of sectors. Values for
the vector are relative, and they will be scaled by dividing their summation.
By default, it is <code>NULL</code> which means the width of sectors correspond to the data
range in sectors.
</p>
</td></tr>
<tr><td><code id="circos.initialize_+3A_ring">ring</code></td>
<td>
<p>Whether the sector represented as a ring. If yes, there should only be one sector in the circle.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function allocates the sectors according to the values on x-axis.
The number of sectors are determined by the <code>factors</code> and the order
of sectors are determined by the levels of factors. In this function,
the start and end position for each sector on the circle (measured by degree)
are calculated according to the values on x-axis or by <code>xlim</code>.
</p>
<p>If <code>x</code> is set, the length of <code>x</code> must be equal to the length of <code>factors</code>.
Then the data range for each sector are calculated from <code>x</code> by splitting <code>factors</code>.
</p>
<p>If <code>xlim</code> is set, it should be a vector containing two numbers or a matrix with 2 columns.
If <code>xlim</code> is a 2-element vector, it means all sector share the same <code>xlim</code>.
If <code>xlim</code> is a 2-column matrix, the number of rows should be equal to the number of categories
identified by <code>factors</code>, then each row of <code>xlim</code> corresponds to the data range for each sector
and the order of rows is corresponding to the order of levels of <code>factors</code>. If <code>xlim</code> is a matrix
for which row names cover all sector names, <code>xlim</code> is automatically adjusted.
</p>
<p>Normally, width of sectors will be calculated internally according to the data range in sectors. But you can
still set the width manually. However, it is not always a good idea to change the default sector width since
the width can reflect the range of data in sectors. However, in some cases, it is useful to manually set
the width such as you want to zoom some part of the sectors.
</p>
<p>The function finally calls <code><a href="graphics.html#topic+plot.default">plot</a></code> with enforing aspect ratio to be 1 and be ready for adding graphics.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html">https://jokergoo.github.io/circlize_book/book/circular-layout.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.initializeCircularGenome'>
Initialize a layout for circular genome
</h2><span id='topic+circos.initializeCircularGenome'></span>

<h3>Description</h3>

<p>Initialize a layout for circular genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.initializeCircularGenome(name, genome_size, plotType = "axis", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.initializeCircularGenome_+3A_name">name</code></td>
<td>
<p>Name of the genome (or the &quot;chromosome name&quot;).
</p>
</td></tr>
<tr><td><code id="circos.initializeCircularGenome_+3A_genome_size">genome_size</code></td>
<td>
<p>Size of the genome
</p>
</td></tr>
<tr><td><code id="circos.initializeCircularGenome_+3A_plottype">plotType</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.initializeCircularGenome_+3A_...">...</code></td>
<td>
<p>All goes to <code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code>.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.initializeWithIdeogram'>
Initialize the circular layout with an ideogram
</h2><span id='topic+circos.initializeWithIdeogram'></span>

<h3>Description</h3>

<p>Initialize the circular layout with an ideogram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.initializeWithIdeogram(
    cytoband = system.file(package = "circlize", "extdata", "cytoBand.txt"),
    species = NULL,
    sort.chr = TRUE,
    chromosome.index = usable_chromosomes(species),
    major.by = NULL,
    plotType = c("ideogram", "axis", "labels"),
    track.height = NULL,
    ideogram.height = convert_height(2, "mm"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.initializeWithIdeogram_+3A_cytoband">cytoband</code></td>
<td>
<p>A path of the cytoband file or a data frame that already contains cytoband data. By default it is cytoband for hg19.
Pass to <code><a href="#topic+read.cytoband">read.cytoband</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_species">species</code></td>
<td>
<p>Abbreviations of species. e.g. hg19 for human, mm10 for mouse. If this
value is specified, the function will download cytoBand.txt.gz from
UCSC website automatically. If there is no cytoband for user's species,
it will keep on trying to download chromInfo file. Pass to <code><a href="#topic+read.cytoband">read.cytoband</a></code> or <code><a href="#topic+read.chromInfo">read.chromInfo</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_chromosome.index">chromosome.index</code></td>
<td>
<p>subset of chromosomes, also used to reorder chromosomes.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_sort.chr">sort.chr</code></td>
<td>
<p>Whether chromosome names should be sorted (first sort by numbers then by letters).
If <code>chromosome.index</code> is set, this argumetn is enforced to <code>FALSE</code>
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_major.by">major.by</code></td>
<td>
<p>Increment of major ticks. Pass to <code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_plottype">plotType</code></td>
<td>
<p>Which tracks should be drawn. <code>ideogram</code> for ideogram rectangle, <code>axis</code> for genomic axis and <code>labels</code> for chromosome names.
If there is no ideogram for specified species, <code>ideogram</code> will be enforced to be excluded.
If it is set to <code>NULL</code>, the function just initialize the plot but draw nothing.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_track.height">track.height</code></td>
<td>
<p>Height of the track which contains &quot;axis&quot; and &quot;labels&quot;.
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_ideogram.height">ideogram.height</code></td>
<td>
<p>Height of the ideogram track
</p>
</td></tr>
<tr><td><code id="circos.initializeWithIdeogram_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will initialize the circular plot in which each sector corresponds to a chromosome. You can control the order of 
chromosomes by <code>chromosome.index</code> or by <code>sort.chr</code>, or by setting a special format of <code>cytoband</code> (please refer to <code><a href="#topic+read.cytoband">read.cytoband</a></code> 
to find out how to control a proper <code>cytoband</code>).
</p>
<p>The function finally pass data to <code><a href="#topic+circos.genomicInitialize">circos.genomicInitialize</a></code> to initialize the circular plot.
</p>
<p>The style of ideogram is almost fixed, but you can customize it with your self-sefined code. Refer to vignette for demonstration.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/initialize-genomic-plot.html#initialize-cytoband">https://jokergoo.github.io/circlize_book/book/initialize-genomic-plot.html#initialize-cytoband</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initializeWithIdeogram()

cytoband.file = system.file(package = "circlize",
    "extdata", "cytoBand.txt")
circos.initializeWithIdeogram(cytoband.file)

cytoband.df = read.table(cytoband.file, colClasses = c("character", "numeric",
    "numeric", "character", "character"), sep = "\t")
circos.initializeWithIdeogram(cytoband.df)

circos.initializeWithIdeogram(species = "hg18")

circos.initializeWithIdeogram(species = "mm10")

circos.initializeWithIdeogram(chromosome.index = c("chr1", "chr2"))

cytoband = read.table(cytoband.file, colClasses = c("character", "numeric",
    "numeric", "character", "character"), sep = "\t")
circos.initializeWithIdeogram(cytoband, sort.chr = FALSE)

cytoband[[1]] = factor(cytoband[[1]], levels = paste0("chr", c(22:1, "X", "Y")))
circos.initializeWithIdeogram(cytoband, sort.chr = FALSE)

cytoband = read.table(cytoband.file, colClasses = c("character", "numeric",
    "numeric", "character", "character"), sep = "\t")
circos.initializeWithIdeogram(cytoband, sort.chr = TRUE)

circos.initializeWithIdeogram(plotType = c("axis", "labels"))

circos.initializeWithIdeogram(plotType = NULL)

circos.par("start.degree" = 90)
circos.initializeWithIdeogram()
circos.clear()

circos.par("gap.degree" = rep(c(2, 4), 12))
circos.initializeWithIdeogram()
circos.clear()

</code></pre>

<hr>
<h2 id='circos.labels'>
Add a label track
</h2><span id='topic+circos.labels'></span>

<h3>Description</h3>

<p>Add a label track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.labels(
    sectors, x, labels,
    facing = "clockwise",
    niceFacing = TRUE,
    col = par("col"),
    cex = 0.8,
    font = par("font"),
    padding = 0.4,
    connection_height = mm_h(5),
    line_col = par("col"),
    line_lwd = par("lwd"),
    line_lty = par("lty"),
    labels_height = min(c(cm_h(1.5), max(strwidth(labels, cex = cex, font = font)))),
    side = c("inside", "outside"),
    labels.side = side,
    track.margin = circos.par("track.margin"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.labels_+3A_sectors">sectors</code></td>
<td>
<p>A vector of sector names.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_x">x</code></td>
<td>
<p>Positions of the labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_labels">labels</code></td>
<td>
<p>A vector of labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_facing">facing</code></td>
<td>
<p>Facing of the labels. The value can only be <code>"clockwise"</code> or <code>"reverse.clockwise"</code>.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Whether automatically adjust the facing of the labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_col">col</code></td>
<td>
<p>Color for the labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_cex">cex</code></td>
<td>
<p>Size of the labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_font">font</code></td>
<td>
<p>Font of the labels.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_padding">padding</code></td>
<td>
<p>Padding of the labels, the value is the ratio to the height of the label.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_connection_height">connection_height</code></td>
<td>
<p>Height of the connection track.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_line_col">line_col</code></td>
<td>
<p>Color for the connection lines.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_line_lwd">line_lwd</code></td>
<td>
<p>Line width for the connection lines.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_line_lty">line_lty</code></td>
<td>
<p>Line type for the connectioin lines.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_labels_height">labels_height</code></td>
<td>
<p>Height of the labels track.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_side">side</code></td>
<td>
<p>Side of the labels track, is it in the inside of the track where the regions are marked?</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_labels.side">labels.side</code></td>
<td>
<p>Same as <code>side</code>. It will replace <code>side</code> in the future versions.</p>
</td></tr>
<tr><td><code id="circos.labels_+3A_track.margin">track.margin</code></td>
<td>
<p>Bottom and top margins.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates two tracks, one for the connection lines and one for the labels.
</p>
<p>If two labels are too close and overlap, this function automatically adjusts the positions of neighouring labels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(sectors = letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1))
circos.labels(c("a", "a", "b", "b"), x = c(0.1, 0.12, 0.4, 0.6), labels = c(0.1, 0.12, 0.4, 0.6))
</code></pre>

<hr>
<h2 id='circos.lines'>
Add lines to the plotting region
</h2><span id='topic+circos.lines'></span>

<h3>Description</h3>

<p>Add lines to the plotting region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.lines(
    x, y,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    col = ifelse(area, "grey", par("col")),
    lwd = par("lwd"),
    lty = par("lty"),
    type = "l",
    straight = FALSE,
    area = FALSE,
    area.baseline = NULL,
    border = "black",
    baseline = "bottom",
    pt.col = par("col"),
    cex = par("cex"),
    pch = par("pch"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.lines_+3A_x">x</code></td>
<td>
<p>Data points on x-axis, measured in &quot;current&quot; data coordinate.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_y">y</code></td>
<td>
<p>Data points on y-axis, measured in &quot;current&quot; data coordinate.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_col">col</code></td>
<td>
<p>Line color.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_lty">lty</code></td>
<td>
<p>Line style.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_type">type</code></td>
<td>
<p>Line type, similar as <code>type</code> argument in <code><a href="graphics.html#topic+lines">lines</a></code>, but only in <code>c("l", "o", "h", "s")</code></p>
</td></tr>
<tr><td><code id="circos.lines_+3A_straight">straight</code></td>
<td>
<p>Whether draw straight lines between points.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_area">area</code></td>
<td>
<p>Whether to fill the area below the lines. If it is set to <code>TRUE</code>, <code>col</code> controls the filled color in the area and <code>border</code> controls color of the line.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_area.baseline">area.baseline</code></td>
<td>
<p>deprecated, use <code>baseline</code> instead.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_baseline">baseline</code></td>
<td>
<p>The base line to draw areas. By default it is the minimal of y-range (bottom). It can be a string or a number. If a string, it should be one of <code>bottom</code> and <code>top</code>. This argument also works if <code>type</code> is set to <code>h</code>.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_border">border</code></td>
<td>
<p>color for border of the area.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_pt.col">pt.col</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, point color.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_cex">cex</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, point size.</p>
</td></tr>
<tr><td><code id="circos.lines_+3A_pch">pch</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, point type.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Normally, straight lines in the Cartesian coordinate have to be transformed into curves in the circular layout.
But if you do not want to do such transformation you can use this function just drawing straight
lines between points by setting <code>straight</code> to <code>TRUE</code>.
</p>
<p>Drawing areas below lines can help to identify the direction of y-axis in cells (since it is a circle). This can be done by specifying
<code>area</code> to <code>TURE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:9]
circos.par(points.overflow.warning = FALSE)
circos.initialize(sectors, xlim = c(0, 10))
circos.trackPlotRegion(sectors, ylim = c(0, 10), track.height = 0.5)

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "a")
circos.text(5, 9, "type = 'l'", sector.index = "a", facing = "outside")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "b", type = "o")
circos.text(5, 9, "type = 'o'", sector.index = "b", facing = "outside")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "c", type = "h")
circos.text(5, 9, "type = 'h'", sector.index = "c", facing = "outside")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "d", type = "h", baseline = 5)
circos.text(5, 9, "type = 'h', baseline = 5", sector.index = "d", facing = "outside")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "e", type = "s")
circos.text(5, 9, "type = 's'", sector.index = "e", facing = "outside")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "f", area = TRUE)
circos.text(5, 9, "type = 'l', area = TRUE", sector.index = "f")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "g", type = "o", area = TRUE)
circos.text(5, 9, "type = 'o', area = TRUE", sector.index = "g")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "h", type = "s", area = TRUE)
circos.text(5, 9, "type = 's', area = TRUE", sector.index = "h")

circos.lines(sort(runif(10)*10), runif(10)*8, sector.index = "i", area = TRUE, baseline = "top")
circos.text(5, 9, "type = 'l', area = TRUE, baseline = 'top'", sector.index = "i")

circos.clear()
</code></pre>

<hr>
<h2 id='circos.link'>
Draw links between points or/and intervals
</h2><span id='topic+circos.link'></span>

<h3>Description</h3>

<p>Draw links between points or/and intervals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.link(
    sector.index1,
    point1,
    sector.index2,
    point2,
    rou = get_most_inside_radius(),
    rou1 = rou,
    rou2 = rou,
    h = NULL,
    h.ratio = 0.5,
    w = 1,
    h2 = h,
    w2 = w,
    inverse = FALSE,
    col = "black",
    lwd = par("lwd"),
    lty = par("lty"),
    border = col,
    directional = 0,
    arr.length = ifelse(arr.type == "big.arrow", 0.02, 0.4),
    arr.width = arr.length/2,
    arr.type = "triangle",
    arr.lty = lty,
    arr.lwd = lwd,
    arr.col = col,
    reduce_to_mid_line = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.link_+3A_sector.index1">sector.index1</code></td>
<td>
<p>Index for the first sector where one link end locates
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_point1">point1</code></td>
<td>
<p>A single value or a numeric vector of length 2. If it is a 2-elements vector, then
the link would be a belt/ribbon.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_sector.index2">sector.index2</code></td>
<td>
<p>Index for the other sector where the other link end locates
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_point2">point2</code></td>
<td>
<p>A single value or a numeric vector of length 2. If it is a 2-elements vector, then
the link would be a belt/ribbon.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_rou">rou</code></td>
<td>
<p>The position of the the link ends (if <code>rou1</code> and <code>rou2</code> are not set). It is the percentage of the radius of the unit circle.
By default its value is the position of bottom margin of the most inner track.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_rou1">rou1</code></td>
<td>
<p>The position of end 1 of the link.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_rou2">rou2</code></td>
<td>
<p>The position of end 2 of the link.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_h">h</code></td>
<td>
<p>Height of the link, measured as percent to the radius to the unit circle. By default it is automatically infered.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_h.ratio">h.ratio</code></td>
<td>
<p>systematically change the link height. The value is between 0 and 1.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_w">w</code></td>
<td>
<p>Since the link is a Bezier curve, it controls the shape of Bezier curve.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_h2">h2</code></td>
<td>
<p>Height of the bottom edge of the link if it is a ribbon.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_w2">w2</code></td>
<td>
<p>Shape of the bottom edge of the link if it is a ribbon.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_inverse">inverse</code></td>
<td>
<p>Whether the link is inversed.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_col">col</code></td>
<td>
<p>Color of the link. If the link is a ribbon, then it is the filled color for the ribbon.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_lwd">lwd</code></td>
<td>
<p>Line (or border) width
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_lty">lty</code></td>
<td>
<p>Line (or border) style
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_border">border</code></td>
<td>
<p>If the link is a ribbon, then it is the color for the ribbon border.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_directional">directional</code></td>
<td>
<p>0 for no direction, 1 for direction from <code>point1</code> to <code>point2</code>, -1 for direction from <code>point2</code> to <code>point1</code>.
2 for two directional. The direction is important when arrow heads are added.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.width">arr.width</code></td>
<td>
<p>Width of the arrows, pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.type">arr.type</code></td>
<td>
<p>Type of the arrows, pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>. Default value is <code>triangle</code>. There is an additional option
<code>big.arrow</code>.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.length">arr.length</code></td>
<td>
<p>Length of the arrows, measured in 'cm', pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>. If <code>arr.type</code> is set to <code>big.arrow</code>,
the value is percent to the radius of the unit circle.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.col">arr.col</code></td>
<td>
<p>Color of the arrows, pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.lwd">arr.lwd</code></td>
<td>
<p>Line width of arrows, pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_arr.lty">arr.lty</code></td>
<td>
<p>Line type of arrows, pass to <code><a href="shape.html#topic+Arrowhead">Arrowhead</a></code>.
</p>
</td></tr>
<tr><td><code id="circos.link_+3A_reduce_to_mid_line">reduce_to_mid_line</code></td>
<td>
<p>Only use the middle points of <code>point1</code> and <code>point2</code> to draw the link.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Links are implemented as quadratic Bezier curves (<a href="https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Rational_B.C3.A9zier_curves">https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Rational_B.C3.A9zier_curves</a> ).
</p>
<p>Drawing links does not create any track. So you can think it is independent of the tracks.
</p>
<p>By default you only need to set <code>sector.index1</code>, <code>point1</code>, <code>sector.index2</code> and <code>point2</code>. The
links would look nice.
</p>
<p>Please refer to the vignette for detailed explanation.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#links">https://jokergoo.github.io/circlize_book/book/graphics.html#links</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.nested'>
Nested zooming with two circular plots
</h2><span id='topic+circos.nested'></span>

<h3>Description</h3>

<p>Nested zooming with two circular plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.nested(
    f1,
    f2,
    correspondance,
    connection_height = mm_h(5),
    connection_col = NA,
    connection_border = "black",
    connection_lty = par("lty"),
    connection_lwd = par("lwd"),
    adjust_start_degree = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.nested_+3A_f1">f1</code></td>
<td>
<p>A self-defined function for making the first circular plot. The function should have no argument.</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_f2">f2</code></td>
<td>
<p>A self-defined function for making the second circular plot. The function should have no argument.</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_correspondance">correspondance</code></td>
<td>
<p>A six-column data frame which contains correspondance between the coordinates in two circular plots</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_connection_height">connection_height</code></td>
<td>
<p>The height of the connection track, measured as the percent to the radius of the unit circle. The value can be specified by <code><a href="#topic+uh">uh</a></code> or <code><a href="#topic+convert_height">convert_height</a></code> with absolute units.</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_connection_col">connection_col</code></td>
<td>
<p>Filled color of the connection track. The value can be a vector with same length as number of rows of <code>correspondance</code></p>
</td></tr>
<tr><td><code id="circos.nested_+3A_connection_border">connection_border</code></td>
<td>
<p>Border color of the connection track.</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_connection_lty">connection_lty</code></td>
<td>
<p>Line style of the connection track borders</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_connection_lwd">connection_lwd</code></td>
<td>
<p>Line width of the connection track borders</p>
</td></tr>
<tr><td><code id="circos.nested_+3A_adjust_start_degree">adjust_start_degree</code></td>
<td>
<p>If <code>circos.par(start.degree = ...)</code> is not set in <code>f2()</code>, the start degree for the second circular plot will be adjusted to make the distance of sectors between the two plots to the minimal.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function visualizes zoomings by combining two circular plots into one page where
one is the normal circular plot and the other one only contains regions that need to be zoomed.
This function automatically arranges the two plots to make it easy to correspond between
the original and the zoomed sectors.
</p>
<p>Since the function needs to know the information of the two circular plots, please do not call
<code><a href="#topic+circos.clear">circos.clear</a></code> in either <code>f1()</code> or <code>f2()</code>. It will be called internally in <code><a href="#topic+circos.nested">circos.nested</a></code>.
</p>
<p>If <code>adjust_start_degree</code> is set to <code>TRUE</code>, <code>start.degree</code> should not be set in <code>f2()</code>.
Also <code>canvas.xlim</code> and <code>canvas.ylim</code> are reset in <code>f2()</code>, they should not be set in <code>f2()</code>
either.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/nested-zooming.html">https://jokergoo.github.io/circlize_book/book/nested-zooming.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.par'>
Parameters for the circular layout
</h2><span id='topic+circos.par'></span>

<h3>Description</h3>

<p>Parameters for the circular layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.par(..., RESET = FALSE, READ.ONLY = NULL, LOCAL = FALSE, ADD = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.par_+3A_...">...</code></td>
<td>
<p>Arguments for the parameters, see &quot;details&quot; section
</p>
</td></tr>
<tr><td><code id="circos.par_+3A_reset">RESET</code></td>
<td>
<p>reset to default values
</p>
</td></tr>
<tr><td><code id="circos.par_+3A_read.only">READ.ONLY</code></td>
<td>
<p>please ignore
</p>
</td></tr>
<tr><td><code id="circos.par_+3A_local">LOCAL</code></td>
<td>
<p>please ignore
</p>
</td></tr>
<tr><td><code id="circos.par_+3A_add">ADD</code></td>
<td>
<p>please ignore
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Global parameters for the circular layout. Currently supported parameters are:
</p>

<dl>
<dt><code>start.degree</code></dt><dd><p>The starting degree from which the circle begins to draw. Note this degree is measured
in the standard polar coordinate which means it is always reverse-clockwise.
</p>
</dd>
<dt><code>gap.degree</code></dt><dd><p>Gap between two neighbour sectors. It can be a single value or a vector. If it is a vector,
the first value corresponds to the gap after the first sector.
</p>
</dd>
<dt><code>gap.after</code></dt><dd><p>identical to <code>gap.degree</code> option, but a more understandable name. Modifying this option will also affect <code>gap.degree</code>.
</p>
</dd>
<dt><code>track.margin</code></dt><dd><p>Like <code>margin</code> in Cascading Style Sheets (CSS), it is the blank area
out of the plotting region, also outside of the borders. Since left and right margin are controlled
by <code>gap.degree</code>, only bottom and top margin need to be set. And all cells in a same track share the same margins, and
that's why this parameter is called <code>track.margin</code>. The value for the <code>track.margin</code>
is the percentage according to the radius of the unit circle. <code><a href="#topic+convert_height">convert_height</a></code> can be used to set to an absolute unit (e.g cm/inche).
</p>
</dd>
<dt><code>unit.circle.segments</code></dt><dd><p>Since curves are simulated by a series of straight lines,
this parameter controls the amount of segments to represent a curve. The minimal length
of the line segmentation is the length of the unit circle (<code>2pi</code>) divided by <code>unit.circoe.segments</code>.
More segments means better approximation for the curves while larger size if you generate figures as PDF format.
</p>
</dd>
<dt><code>cell.padding</code></dt><dd><p>Padding of the cell. Like <code>padding</code> in Cascading Style Sheets
(CSS), it is the blank area around the plotting regions, but within the borders.
The parameter has four values, which controls the bottom, left, top and right paddings
respectively. The first and the third padding
values are the percentages according to the radius of the unit circle and the second and
fourth values are degrees. Similar as <code>track.margin</code> option, the first and the third value
can be set by <code><a href="#topic+convert_height">convert_height</a></code> to an absolute unit.
</p>
</dd>
<dt><code>track.height</code></dt><dd><p>The default height of tracks. It is the percentage according to the radius
of the unit circle. The height includes the top and bottom cell paddings but not the margins.
<code><a href="#topic+convert_height">convert_height</a></code> can be used to set the height to an absolute unit.
</p>
</dd>
<dt><code>points.overflow.warning</code></dt><dd><p>Since each cell is in fact not a real plotting region but only
an ordinary rectangle, it does not eliminate points that are plotted out of
the region. So if some points are out of the plotting region, <code>circlize</code> would continue drawing the points and printing warnings. In some
cases, draw something out of the plotting region is useful, such as draw
some legend or text. Set this value to <code>FALSE</code> to turn off the warnings.
</p>
</dd>
<dt><code>circle.margin</code></dt><dd><p>Margin in the horizontal and vertical direction. The value should be a positive numeric vector
and the length of it should be either 1, 2, or 4. When it has length of 1, it controls the margin on the four sides of the circle.
When it has length of 2, the first value controls the margin on the left and right, and the second value controls
the margin on the bottom and top side. When it has length of 4, the four values controls the margins on the left, right, bottom and top sides
of the circle. So A value of <code>c(x1, x2, y1, y2)</code> means <code>circos.par(canvas.xlim = c(-(1+x1), 1+x2), canvas.ylim = c(-(1+y1), 1+y2))</code>.
</p>
</dd>
<dt><code>canvas.xlim</code></dt><dd><p>The coordinate for the canvas. Because <code>circlize</code> draws everything (or almost everything) inside the unit circle,
the default <code>canvas.xlim</code> and <code>canvas.ylim</code> for the canvas would be all <code>c(-1, 1)</code>. However, you can set it to a more broad
interval if you want to draw other things out of the circle. By choosing proper
<code>canvas.xlim</code> and <code>canvas.ylim</code>, you can draw part of the circle. E.g. setting
<code>canvas.xlim</code> to <code>c(0, 1)</code> and <code>canvas.ylim</code> to <code>c(0, 1)</code> would only draw
circle in the region of (0, pi/2).
</p>
</dd>
<dt><code>canvas.ylim</code></dt><dd><p>The coordinate for the canvas. By default it is <code>c(-1, 1)</code>
</p>
</dd>
<dt><code>clock.wise</code></dt><dd><p>The direction for adding sectors. Default is <code>TRUE</code>.
</p>
</dd>
<dt><code>xaxis.clock.wise</code></dt><dd><p>The direction in the x-axes for all sectors. Default is <code>TRUE</code>.
</p>
</dd>
</dl>

<p>Similar as <code><a href="graphics.html#topic+par">par</a></code>, you can get the parameter values by specifying the
names of parameters and you can set the parameter values by specifying a
named list which contains the new values.
</p>
<p><code>gap.degree</code>, <code>start.degree</code>, <code>canvas.xlim</code>, <code>canvas.ylim</code> and <code>clock.wise</code>
only be set before the initialization of the circular layout
(i.e. before calling <code><a href="#topic+circos.initialize">circos.initialize</a></code>) because these values will not be changed after
adding sectors on the circle. The left and right padding for <code>cell.padding</code> will also be
ignored after the initialization because all cells in a sector would share the same
left and right paddings.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html#graphic-parameters">https://jokergoo.github.io/circlize_book/book/circular-layout.html#graphic-parameters</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.par
</code></pre>

<hr>
<h2 id='circos.points'>
Add points to a plotting region
</h2><span id='topic+circos.points'></span>

<h3>Description</h3>

<p>Add points to a plotting region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.points(
    x, y,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    pch = par("pch"),
    col = par("col"),
    cex = par("cex"),
    bg = par("bg"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.points_+3A_x">x</code></td>
<td>
<p>Data points on x-axis, measured in &quot;current&quot; data coordinate</p>
</td></tr>
<tr><td><code id="circos.points_+3A_y">y</code></td>
<td>
<p>Data points on y-axis, measured in &quot;current&quot; data coordinate</p>
</td></tr>
<tr><td><code id="circos.points_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.points_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.points_+3A_pch">pch</code></td>
<td>
<p>Point type</p>
</td></tr>
<tr><td><code id="circos.points_+3A_col">col</code></td>
<td>
<p>Point color</p>
</td></tr>
<tr><td><code id="circos.points_+3A_cex">cex</code></td>
<td>
<p>Point size</p>
</td></tr>
<tr><td><code id="circos.points_+3A_bg">bg</code></td>
<td>
<p>backgrond of points</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can only add points in one specified cell. Pretending a low-level plotting
function, it can only be applied in plotting region which has been created.
</p>
<p>You can think the function similar as the normal <code><a href="graphics.html#topic+points">points</a></code>
function, just adding points in the circular plotting region. The position of
cell is identified by <code>sector.index</code> and <code>track.index</code>, if they are not
specified, they are in 'current' sector and 'current' track.
</p>
<p>Data points out of the plotting region will also be added, but with warning messages.
</p>
<p>Other graphics parameters which are available in the function are <code>pch</code>, <code>col</code>
and <code>cex</code> which have same meaning as those in the <code><a href="graphics.html#topic+par">par</a></code>.
</p>
<p>It is recommended to use <code><a href="#topic+circos.points">circos.points</a></code> inside <code>panel.fun</code> in <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code> so that
it draws points directly on &quot;curent&quot; cell.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#points">https://jokergoo.github.io/circlize_book/book/graphics.html#points</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.points(runif(10), runif(10))
})
circos.points(runif(10), runif(10), sector.index = "c", pch = 16, col = "red")
circos.clear()
</code></pre>

<hr>
<h2 id='circos.polygon'>
Draw polygon
</h2><span id='topic+circos.polygon'></span>

<h3>Description</h3>

<p>Draw polygon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.polygon(
    x, y,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.polygon_+3A_x">x</code></td>
<td>
<p>Data points on x-axis</p>
</td></tr>
<tr><td><code id="circos.polygon_+3A_y">y</code></td>
<td>
<p>Data points on y-axis</p>
</td></tr>
<tr><td><code id="circos.polygon_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.polygon_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.polygon_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="graphics.html#topic+polygon">polygon</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>similar as <code><a href="graphics.html#topic+polygon">polygon</a></code>.
</p>
<p>Note: start point should overlap with the end point.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
sectors = letters[1:4]
circos.initialize(sectors, xlim = c(0, 1))
circos.trackPlotRegion(ylim = c(-3, 3), track.height = 0.4, panel.fun = function(x, y) {
    x1 = runif(20)
    y1 = x1 + rnorm(20)
    or = order(x1)
    x1 = x1[or]
    y1 = y1[or]
    loess.fit = loess(y1 ~ x1)
    loess.predict = predict(loess.fit, x1, se = TRUE)
    d1 = c(x1, rev(x1))
    d2 = c(loess.predict$fit + loess.predict$se.fit,
        rev(loess.predict$fit - loess.predict$se.fit))
    circos.polygon(d1, d2, col = "#CCCCCC", border = NA)
    circos.points(x1, y1, cex = 0.5)
    circos.lines(x1, loess.predict$fit)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.raster'>
Add raster images
</h2><span id='topic+circos.raster'></span>

<h3>Description</h3>

<p>Add raster images
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.raster(
    image, x, y,
    width, height,
    facing = c("inside", "outside", "reverse.clockwise", "clockwise",
    "downward", "bending.inside", "bending.outside"),
    niceFacing = FALSE,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    scaling = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.raster_+3A_image">image</code></td>
<td>
<p>A <code>raster</code> object, or an object that can be converted by <code><a href="grDevices.html#topic+as.raster">as.raster</a></code>.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_x">x</code></td>
<td>
<p>Position of the center of the raster image, measued in the data coordinate in the cell.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_y">y</code></td>
<td>
<p>Position of the center of the raster image, measued in the data coordinate in the cell.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_width">width</code></td>
<td>
<p>Width of the raster image. When <code>facing</code> is one of &quot;inside&quot;, &quot;outside&quot;, &quot;clockwise&quot; and &quot;reverse.clockwise&quot;, the image should have absolute size where the value of <code>width</code> should be specified like <code>20mm</code>, <code>1cm</code> or <code>0.5inche</code>. When <code>facing</code> is one of <code>bending.inside</code> and <code>bending.outside</code>, the value of <code>width</code> is measured in the data coordinate in the cell.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_height">height</code></td>
<td>
<p>Height of the raster image. Same format as <code>width</code>. If the value of <code>height</code> is omit,  default height is calculated by taking the aspect ratio of the original image. But when <code>facing</code> is one of <code>bending.inside</code> and <code>bending.outside</code>, <code>height</code> is mandatory to set.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_facing">facing</code></td>
<td>
<p>Facing of the raster image.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Facing of text. Please refer to vignette for different settings.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.raster_+3A_scaling">scaling</code></td>
<td>
<p>Scaling factor to resize the raster image.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#raster-image">https://jokergoo.github.io/circlize_book/book/graphics.html#raster-image</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(png)
image = system.file("extdata", "Rlogo.png", package = "circlize")
image = as.raster(readPNG(image))
circos.initialize(letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.raster(image, CELL_META$xcenter, CELL_META$ycenter, width = "2cm", 
        facing = "inside", niceFacing = TRUE)
})
circos.clear()

if(FALSE) {
# NOTE: following takes quite a long time to run
load(system.file("extdata", "doodle.RData", package = "circlize"))
circos.par("cell.padding" = c(0, 0, 0, 0))
circos.initialize(letters[1:16], xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    img = img_list[[CELL_META$sector.numeric.index]]
    circos.raster(img, CELL_META$xcenter, CELL_META$ycenter, width = 1, 
        height = 1, facing = "bending.inside")
}, track.height = 0.25, bg.border = NA)
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    img = img_list[[CELL_META$sector.numeric.index + 16]]
    circos.raster(img, CELL_META$xcenter, CELL_META$ycenter, width = 1, 
        height = 1, facing = "bending.inside")
}, track.height = 0.25, bg.border = NA)
circos.clear()
}
</code></pre>

<hr>
<h2 id='circos.rect'>
Draw rectangle-like grid
</h2><span id='topic+circos.rect'></span>

<h3>Description</h3>

<p>Draw rectangle-like grid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.rect(
    xleft, ybottom, xright, ytop,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    rot = 0,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.rect_+3A_xleft">xleft</code></td>
<td>
<p>x for the left bottom points</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_ybottom">ybottom</code></td>
<td>
<p>y for the left bottom points</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_xright">xright</code></td>
<td>
<p>x for the right top points</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_ytop">ytop</code></td>
<td>
<p>y for the right top points</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_rot">rot</code></td>
<td>
<p>Rotation of the rectangles. The value is measured clockwise in degree. Rotation is relative to the center of the rectangles.</p>
</td></tr>
<tr><td><code id="circos.rect_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="graphics.html#topic+polygon">polygon</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The name for this function is <code><a href="#topic+circos.rect">circos.rect</a></code>
because if you imagine the plotting region as Cartesian coordinate, then it is rectangle.
in the polar coordinate, the up and bottom edge become two arcs.
</p>
<p>This function can be vectorized.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(c("a", "b", "c", "d"), xlim = c(0, 10))
circos.track(ylim = c(0, 10), panel.fun = function(x, y) {
    for(rot in seq(0, 360, by = 30)) {
        circos.rect(2, 2, 6, 6, rot = rot)
    }
}, track.height = 0.5)
</code></pre>

<hr>
<h2 id='circos.segments'>
Draw segments through pairwise of points
</h2><span id='topic+circos.segments'></span>

<h3>Description</h3>

<p>Draw segments through pairwise of points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.segments(
    x0, y0, x1, y1,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    straight = FALSE,
    col = par("col"),
    lwd = par("lwd"),
    lty = par("lty"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.segments_+3A_x0">x0</code></td>
<td>
<p>x coordinates for starting points.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_y0">y0</code></td>
<td>
<p>y coordinates for ending points.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_x1">x1</code></td>
<td>
<p>x coordinates for starting points.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_y1">y1</code></td>
<td>
<p>y coordinates for ending points.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_straight">straight</code></td>
<td>
<p>Whether the segment is a straight line.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_col">col</code></td>
<td>
<p>Color of the segments.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_lwd">lwd</code></td>
<td>
<p>Line width of the segments.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_lty">lty</code></td>
<td>
<p>Line type of the segments.</p>
</td></tr>
<tr><td><code id="circos.segments_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="graphics.html#topic+lines">lines</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1), track.height = 0.3, panel.fun = function(x, y) {
    x = seq(0.2, 0.8, by = 0.2)
    y = seq(0.2, 0.8, by = 0.2)

    circos.segments(x, 0.1, x, 0.9)
    circos.segments(0.1, y, 0.9, y)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.text'>
Draw text in a cell
</h2><span id='topic+circos.text'></span>

<h3>Description</h3>

<p>Draw text in a cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.text(
    x, y,
    labels,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    direction = NULL,
    facing = c("inside", "outside", "reverse.clockwise", "clockwise",
    "downward", "bending", "bending.inside", "bending.outside"),
    niceFacing = FALSE,
    adj = par("adj"),
    cex = 1,
    col = par("col"),
    font = par("font"),
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.text_+3A_x">x</code></td>
<td>
<p>Data points on x-axis</p>
</td></tr>
<tr><td><code id="circos.text_+3A_y">y</code></td>
<td>
<p>Data points on y-axis</p>
</td></tr>
<tr><td><code id="circos.text_+3A_labels">labels</code></td>
<td>
<p>Labels for each points</p>
</td></tr>
<tr><td><code id="circos.text_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.text_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.text_+3A_direction">direction</code></td>
<td>
<p>deprecated, use <code>facing</code> instead.</p>
</td></tr>
<tr><td><code id="circos.text_+3A_facing">facing</code></td>
<td>
<p>Facing of text. Please refer to vignette for different settings</p>
</td></tr>
<tr><td><code id="circos.text_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Should the facing of text be adjusted to fit human eyes?</p>
</td></tr>
<tr><td><code id="circos.text_+3A_adj">adj</code></td>
<td>
<p>offset for text. By default the text position adjustment is either horizontal or vertical in the canvas coordinate system. The &quot;circular horizontal&quot; offset can be set as a value in degree unit and the value should be wrapped by <code><a href="#topic+degree">degree</a></code>.</p>
</td></tr>
<tr><td><code id="circos.text_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="graphics.html#topic+text">text</a></code></p>
</td></tr>
<tr><td><code id="circos.text_+3A_cex">cex</code></td>
<td>
<p>Font size</p>
</td></tr>
<tr><td><code id="circos.text_+3A_col">col</code></td>
<td>
<p>Font color</p>
</td></tr>
<tr><td><code id="circos.text_+3A_font">font</code></td>
<td>
<p>Font style</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is similar to <code><a href="graphics.html#topic+text">text</a></code>. All you need to note is the <code>facing</code> settings.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#text">https://jokergoo.github.io/circlize_book/book/graphics.html#text</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:4]
circos.par(points.overflow.warning = FALSE)
circos.initialize(sectors, xlim = c(0, 10))
circos.trackPlotRegion(sectors, ylim = c(0, 10),
  track.height = 0.5, panel.fun = function(x, y) {
    circos.text(3, 1, "inside", facing = "inside", cex = 0.8)
    circos.text(7, 1, "outside", facing = "outside", cex = 0.8)
    circos.text(0, 5, "reverse.clockwise", facing = "reverse.clockwise",
        adj = c(0.5, 0), cex = 0.8)
    circos.text(10, 5, "clockwise", facing = "clockwise", adj = c(0.5, 0),
        cex = 0.8)
    circos.text(5, 5, "downward", facing = "downward", cex = 0.8)
    circos.text(3, 9, "====bending.inside====", facing = "bending.inside",
        cex = 0.8)
    circos.text(7, 9, "====bending.outside====", facing = "bending.outside",
        cex = 0.8)
})
circos.clear()
</code></pre>

<hr>
<h2 id='circos.track'>
Create plotting regions for a whole track
</h2><span id='topic+circos.track'></span>

<h3>Description</h3>

<p>Create plotting regions for a whole track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.track(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.track_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Shortcut function of <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.trackHist'>
Draw histogram in cells among a whole track
</h2><span id='topic+circos.trackHist'></span>

<h3>Description</h3>

<p>Draw histogram in cells among a whole track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.trackHist(
    sectors,
    x,
    track.height = circos.par("track.height"),
    track.index = NULL,
    ylim = NULL,
    force.ylim = TRUE,
    col = ifelse(draw.density, "black", NA),
    border = "black",
    lty = par("lty"),
    lwd = par("lwd"),
    bg.col = NA,
    bg.border = "black",
    bg.lty = par("lty"),
    bg.lwd = par("lwd"),
    breaks = "Sturges",
    include.lowest = TRUE,
    right = TRUE,
    draw.density = FALSE,
    bin.size = NULL,
    area = FALSE,
    factors = sectors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.trackHist_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> or a character vector which represents the categories of data</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. </p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_x">x</code></td>
<td>
<p>Data on the x-axis</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track which is going to be updated. Setting it to <code>NULL</code> means creating the plotting regions in the next newest track.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_track.height">track.height</code></td>
<td>
<p>Height of the track. It is the percentage to the radius of the unit circle. If to update a track, this argument is disabled.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_ylim">ylim</code></td>
<td>
<p>Ranges on y-direction. By default, <code>ylim</code> is calculated automatically.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_force.ylim">force.ylim</code></td>
<td>
<p>Whether to force all cells in the track to share the same <code>ylim</code>.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_col">col</code></td>
<td>
<p>Filled color for histogram</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_border">border</code></td>
<td>
<p>Border color for histogram</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_lty">lty</code></td>
<td>
<p>Line style for histogram</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_lwd">lwd</code></td>
<td>
<p>Line width for histogram</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_bg.col">bg.col</code></td>
<td>
<p>Background color for the plotting regions</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_bg.border">bg.border</code></td>
<td>
<p>Color for the border of the plotting regions</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_bg.lty">bg.lty</code></td>
<td>
<p>Line style for the border of the plotting regions</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_bg.lwd">bg.lwd</code></td>
<td>
<p>Line width for the border of the plotting regions</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_breaks">breaks</code></td>
<td>
<p>see <code><a href="graphics.html#topic+hist">hist</a></code></p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_include.lowest">include.lowest</code></td>
<td>
<p>see <code><a href="graphics.html#topic+hist">hist</a></code></p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_right">right</code></td>
<td>
<p>see <code><a href="graphics.html#topic+hist">hist</a></code></p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_draw.density">draw.density</code></td>
<td>
<p>whether draw density lines instead of histogram bars.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_area">area</code></td>
<td>
<p>whether to fill the area below the density lines. If it is set to <code>TRUE</code>, <code>col</code> controls the filled color in the area and <code>border</code> controls color of the line.</p>
</td></tr>
<tr><td><code id="circos.trackHist_+3A_bin.size">bin.size</code></td>
<td>
<p>size of the bins of the histogram</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It draw histogram in cells among a whole track. It is also an example to show how to add self-defined
high-level graphics by this package.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/high-level-plots.html#histograms">https://jokergoo.github.io/circlize_book/book/high-level-plots.html#histograms</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x = rnorm(1600)
sectors = sample(letters[1:16], 1600, replace = TRUE)
circos.initialize(sectors, x = x)
circos.trackHist(sectors, x = x, col = "#999999",
  border = "#999999")
circos.trackHist(sectors, x = x, bin.size = 0.1,
  col = "#999999", border = "#999999")
circos.trackHist(sectors, x = x, draw.density = TRUE,
  col = "#999999", border = "#999999")
circos.clear()

</code></pre>

<hr>
<h2 id='circos.trackLines'>
Add lines to the plotting regions in a same track
</h2><span id='topic+circos.trackLines'></span>

<h3>Description</h3>

<p>Add lines to the plotting regions in a same track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.trackLines(
    sectors,
    x, y,
    track.index = get.current.track.index(),
    col = par("col"),
    lwd = par("lwd"),
    lty = par("lty"),
    type = "l",
    straight = FALSE,
    area = FALSE,
    area.baseline = NULL,
    border = "black",
    baseline = "bottom",
    pt.col = par("col"),
    cex = par("cex"),
    pch = par("pch"),
    factors = sectors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.trackLines_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> or a character vector which represents the categories of data.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. </p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_x">x</code></td>
<td>
<p>Data points on x-axis.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_y">y</code></td>
<td>
<p>Data points on y-axis.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_col">col</code></td>
<td>
<p>Line color.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_lty">lty</code></td>
<td>
<p>Line style.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_type">type</code></td>
<td>
<p>Line type, similar as <code>type</code> argument in <code><a href="graphics.html#topic+lines">lines</a></code>, but only in <code>c("l", "o", "h", "s")</code>.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_straight">straight</code></td>
<td>
<p>Whether draw straight lines between points.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_area">area</code></td>
<td>
<p>Whether to fill the area below the lines. If it is set to <code>TRUE</code>, <code>col</code> controls the filled color in the area and <code>border</code> controls the color of the line.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_area.baseline">area.baseline</code></td>
<td>
<p>Deprecated, use <code>baseline</code> instead.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_baseline">baseline</code></td>
<td>
<p>The base line to draw area, pass to <code><a href="#topic+circos.lines">circos.lines</a></code>.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_border">border</code></td>
<td>
<p>Color for border of the area.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_pt.col">pt.col</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, points color.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_cex">cex</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, points size.</p>
</td></tr>
<tr><td><code id="circos.trackLines_+3A_pch">pch</code></td>
<td>
<p>If <code>type</code> is &quot;o&quot;, points type.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function adds lines in multiple cells by first splitting data into several parts in which
each part corresponds to one factor (sector index) and then add lines in cells by calling <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>
<p>This function can be replaced by a <code>for</code> loop containing <code><a href="#topic+circos.lines">circos.lines</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.trackPlotRegion'>
Create plotting regions for a whole track
</h2><span id='topic+circos.trackPlotRegion'></span>

<h3>Description</h3>

<p>Create plotting regions for a whole track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.trackPlotRegion(
    sectors = NULL,
    x = NULL, y = NULL,
    ylim = NULL,
    force.ylim = TRUE,
    track.index = NULL,
    track.height = circos.par("track.height"),
    track.margin = circos.par("track.margin"),
    cell.padding = circos.par("cell.padding"),
    bg.col = NA,
    bg.border = "black",
    bg.lty = par("lty"),
    bg.lwd = par("lwd"),
    panel.fun = function(x, y) {NULL},
    factors = sectors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.trackPlotRegion_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> or a character vector which represents categories of data, if it is <code>NULL</code>, then it uses all sector index.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. </p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_x">x</code></td>
<td>
<p>Data on x-axis. It is only used if <code>panel.fun</code> is set.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_y">y</code></td>
<td>
<p>Data on y-axis</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_ylim">ylim</code></td>
<td>
<p>Range of data on y-axis</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_force.ylim">force.ylim</code></td>
<td>
<p>Whether to force all cells in the track to share the same <code>ylim</code>. Normally, all cells on a same track should have same <code>ylim</code>.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track which is going to be created/updated. If the specified track has already been created, this function just updated corresponding track with new plot. If the specified track is <code>NULL</code> or has not been created, this function just creates it. Note the value for this argument should not exceed maximum track index plus 1.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_track.height">track.height</code></td>
<td>
<p>Height of the track. It is the percentage to the radius of the unit circles. The value can be set by <code><a href="#topic+uh">uh</a></code> to an absolute unit. If updating a track (with proper <code>track.index</code> value), this argument is ignored.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_track.margin">track.margin</code></td>
<td>
<p>only affect current track</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_cell.padding">cell.padding</code></td>
<td>
<p>only affect current track</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_bg.col">bg.col</code></td>
<td>
<p>Background color for the plotting regions. It can be vector which has the same length of sectors.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_bg.border">bg.border</code></td>
<td>
<p>Color for the border of the plotting regions. It can be vector which has the same length of sectors.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_bg.lty">bg.lty</code></td>
<td>
<p>Line style for the border of the plotting regions. It can be vector which has the same length of sectors.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_bg.lwd">bg.lwd</code></td>
<td>
<p>Line width for the border of the plotting regions. It can be vector which has the same length of sectors.</p>
</td></tr>
<tr><td><code id="circos.trackPlotRegion_+3A_panel.fun">panel.fun</code></td>
<td>
<p>Panel function to add graphics in each cell, see &quot;details&quot; section and vignette for explanation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function tends to be a high-level plotting function, which means,
you must first call this function to create plotting regions, then those
low-level graphic function such as <code><a href="#topic+circos.points">circos.points</a></code>, <code><a href="#topic+circos.lines">circos.lines</a></code> can be
applied.
</p>
<p>Currently, all the cells that are created in a same track sharing same height, which means,
there is no cell has larger height than others.
</p>
<p>Since ranges for values on x-axis has already been defined by <code><a href="#topic+circos.initialize">circos.initialize</a></code>, only
ranges for values on y-axis should be specified in this function.
There are two ways to identify the ranges for values on y-axes either by <code>y</code>
or <code>ylim</code>. If <code>y</code> is set, it must has the same length as <code>factors</code> and the <code>ylim</code> for each cell is calculated
from y values. Also, the ylim can be specified from <code>ylim</code> which can be a two-element vector or a matrix which
has two columns and the number of rows is the same as the length of the levels of the factors.
</p>
<p>If there is no enough space for the new track or the new track overlaps with other tracks,
there will be an error.
</p>
<p>If <code>factors</code> does not cover all sectors, the cells in remaining unselected
sectors would also be created but without drawing anything. The <code>ylim</code> for these cells
are the same as that in the last created cell.
</p>
<p>The function can also update a already-created track if the index for the track
is specified. If updating an existed track, those parameters related to the position (such as track height and track margin)
of the plotting region can not be changed.
</p>


<h3>Panel</h3>

<p><code>panel.fun</code> provides a convenient way to add graphics in each cell when initializing the
tracks. The self-defined function needs two arguments: <code>x</code> and <code>y</code> which correspond to the data points
in the current cell. When <code>factors</code>, <code>x</code>, and <code>y</code> are set in <code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code>, a subset of <code>x</code>
and <code>y</code> are split by <code>factors</code> and are sent to <code>panel.fun</code> in the &quot;current&quot; cell.
<code><a href="#topic+circos.trackPlotRegion">circos.trackPlotRegion</a></code> creates plotting regions one by one on the track and
<code>panel.fun</code> adds graphics in the 'current' cell after the plotting region for a certain cell has been
created.
</p>
<p>See vignette for examples of how to use this feature.</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html">https://jokergoo.github.io/circlize_book/book/circular-layout.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:8], xlim = c(0, 1))
set.seed(123)
df = data.frame(fa = sample(letters[1:8], 100, replace = TRUE),
                x = runif(100), y = rnorm(100))
circos.track(ylim = c(0, 1), bg.col = rand_color(8))
circos.track(df$fa, x = df$x, y = df$y, panel.fun = function(x, y) {
    circos.points(x, y)
}, track.height = 0.2, bg.border = rand_color(8))
circos.clear()
</code></pre>

<hr>
<h2 id='circos.trackPoints'>
Add points to the plotting regions in a same track
</h2><span id='topic+circos.trackPoints'></span>

<h3>Description</h3>

<p>Add points to the plotting regions in a same track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.trackPoints(
    sectors,
    x, y,
    track.index = get.current.track.index(),
    pch = par("pch"),
    col = par("col"),
    cex = par("cex"),
    bg = par("bg"),
    factors = sectors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.trackPoints_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> or a character vector which represents the categories of data</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. </p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_x">x</code></td>
<td>
<p>Data points on x-axis</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_y">y</code></td>
<td>
<p>Data points on y-axis</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_pch">pch</code></td>
<td>
<p>Point type</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_col">col</code></td>
<td>
<p>Point color</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_cex">cex</code></td>
<td>
<p>Point size</p>
</td></tr>
<tr><td><code id="circos.trackPoints_+3A_bg">bg</code></td>
<td>
<p>backgrond color</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function adds points in multiple cells by first splitting data into several parts in which
each part corresponds to one factor (sector index) and then adding points in each cell by calling <code><a href="#topic+circos.points">circos.points</a></code>.
</p>
<p>Length of <code>pch</code>, <code>col</code> and <code>cex</code> can be one, length of levels of the factors or length of
factors.
</p>
<p>This function can be replaced by a <code>for</code> loop containing <code><a href="#topic+circos.points">circos.points</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:8], xlim = c(0, 1))
df = data.frame(sectors = sample(letters[1:8], 100, replace = TRUE),
                x = runif(100), y = runif(100))
circos.track(ylim = c(0, 1))
circos.trackPoints(df$sectors, x = df$x, y = df$y, pch = 16, col = as.numeric(factor(df$fa)))
circos.clear()
</code></pre>

<hr>
<h2 id='circos.trackText'>
Draw text in cells among the whole track
</h2><span id='topic+circos.trackText'></span>

<h3>Description</h3>

<p>Draw text in cells among the whole track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.trackText(
    sectors,
    x, y,
    labels,
    track.index = get.current.track.index(),
    direction = NULL,
    facing = c("inside", "outside", "reverse.clockwise", "clockwise",
    "downward", "bending", "bending.inside", "bending.outside"),
    niceFacing = FALSE,
    adj = par("adj"),
    cex = 1,
    col = par("col"),
    font = par("font"),
    factors = sectors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.trackText_+3A_sectors">sectors</code></td>
<td>
<p>A <code><a href="base.html#topic+factor">factor</a></code> or a character vector which represents the categories of data</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_factors">factors</code></td>
<td>
<p>The same as <code>sectors</code>. It will be removed in future versions. </p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_x">x</code></td>
<td>
<p>Data points on x-axis</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_y">y</code></td>
<td>
<p>Data points on y-axis</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_labels">labels</code></td>
<td>
<p>Labels</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_direction">direction</code></td>
<td>
<p>deprecated, use <code>facing</code> instead.</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_facing">facing</code></td>
<td>
<p>Facing of text</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_nicefacing">niceFacing</code></td>
<td>
<p>Should the facing of text be adjusted to fit human eyes?</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_adj">adj</code></td>
<td>
<p>Adjustment for text</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_cex">cex</code></td>
<td>
<p>Font size</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_col">col</code></td>
<td>
<p>Font color</p>
</td></tr>
<tr><td><code id="circos.trackText_+3A_font">font</code></td>
<td>
<p>Font style</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function adds texts in multiple cells by first splitting data into several parts in which
each part corresponds to one factor (sector index) and then add texts in cells by calling <code><a href="#topic+circos.text">circos.text</a></code>.
</p>
<p>This function can be replaced by a <code>for</code> loop containing <code><a href="#topic+circos.text">circos.text</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.triangle'>
Draw triangles
</h2><span id='topic+circos.triangle'></span>

<h3>Description</h3>

<p>Draw triangles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.triangle(x1, y1, x2, y2, x3, y3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.triangle_+3A_x1">x1</code></td>
<td>
<p>x-coordinates for the first point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_y1">y1</code></td>
<td>
<p>y-coordinates for the first point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_x2">x2</code></td>
<td>
<p>x-coordinates for the second point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_y2">y2</code></td>
<td>
<p>y-coordinates for the second point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_x3">x3</code></td>
<td>
<p>x-coordinates for the third point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_y3">y3</code></td>
<td>
<p>y-coordinates for the third point.</p>
</td></tr>
<tr><td><code id="circos.triangle_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+circos.polygon">circos.polygon</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(c("a", "b", "c", "d"), xlim = c(0, 10))
circos.track(ylim = c(0, 10), panel.fun = function(x, y) {
    circos.triangle(c(2, 2), c(2, 8),
                    c(8, 8), c(2, 8),
                    c(5, 5), c(8, 2))
}, track.height = 0.5)
</code></pre>

<hr>
<h2 id='circos.update'>
Create plotting regions for a whole track
</h2><span id='topic+circos.update'></span>

<h3>Description</h3>

<p>Create plotting regions for a whole track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.update(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.update_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+circos.updatePlotRegion">circos.updatePlotRegion</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>shortcut function of <code><a href="#topic+circos.updatePlotRegion">circos.updatePlotRegion</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.updatePlotRegion'>
Update the plotting region in an existed cell
</h2><span id='topic+circos.updatePlotRegion'></span>

<h3>Description</h3>

<p>Update the plotting region in an existed cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.updatePlotRegion(
    sector.index = get.cell.meta.data("sector.index"),
    track.index = get.cell.meta.data("track.index"),
    bg.col = NA,
    bg.border = "black",
    bg.lty = par("lty"),
    bg.lwd = par("lwd"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.updatePlotRegion_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.updatePlotRegion_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.updatePlotRegion_+3A_bg.col">bg.col</code></td>
<td>
<p>Background color for the plotting region</p>
</td></tr>
<tr><td><code id="circos.updatePlotRegion_+3A_bg.border">bg.border</code></td>
<td>
<p>Color for the border of the plotting region</p>
</td></tr>
<tr><td><code id="circos.updatePlotRegion_+3A_bg.lty">bg.lty</code></td>
<td>
<p>Line style for the border of the plotting region</p>
</td></tr>
<tr><td><code id="circos.updatePlotRegion_+3A_bg.lwd">bg.lwd</code></td>
<td>
<p>Line width for the border of the plotting region</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can update an existed cell by this function by erasing all the graphics.
But the <code>xlim</code> and <code>ylim</code> inside the cell still remain unchanged.
</p>
<p>Note if you use <code><a href="#topic+circos.track">circos.track</a></code> to update an already created track,
you can re-define <code>ylim</code> in these cells.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ycenter, CELL_META$sector.index)
})
circos.update(sector.index = "b", track.index = 1)
circos.rect(CELL_META$cell.xlim[1], CELL_META$cell.ylim[1],
            CELL_META$cell.xlim[2], CELL_META$cell.ylim[2],
            col = "#FF000080")
circos.clear()
</code></pre>

<hr>
<h2 id='circos.violin'>
Draw violin plots
</h2><span id='topic+circos.violin'></span>

<h3>Description</h3>

<p>Draw violin plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.violin(value, pos, violin_width = 0.8,
    col = NA, border = "black", lwd = par("lwd"), lty = par("lty"),
    show_quantile = TRUE, pt.col = par("col"), cex = par("cex"), pch = 16,
    max_density = NULL, sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.violin_+3A_value">value</code></td>
<td>
<p>A numeric vector, a matrix or a list. If it is a matrix, boxplots are made by columns.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_pos">pos</code></td>
<td>
<p>Positions of the boxes.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_violin_width">violin_width</code></td>
<td>
<p>Width of violins.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_col">col</code></td>
<td>
<p>Filled color of boxes.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_border">border</code></td>
<td>
<p>Color for the border as well as the quantile lines.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_lty">lty</code></td>
<td>
<p>Line style</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_show_quantile">show_quantile</code></td>
<td>
<p>Whether to show the quantile lines.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_cex">cex</code></td>
<td>
<p>Point size.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_pch">pch</code></td>
<td>
<p>Point type.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_pt.col">pt.col</code></td>
<td>
<p>Point color</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_max_density">max_density</code></td>
<td>
<p>The maximal density value across several violins. It is used to compare between violins.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="circos.violin_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    for(pos in seq(0.5, 9.5, by = 1)) {
        value = runif(10)
        circos.violin(value, pos)
    }
})
circos.clear()

circos.initialize(letters[1:4], xlim = c(0, 10))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    value = replicate(runif(10), n = 10, simplify = FALSE)
    circos.violin(value, 1:10 - 0.5, col = 1:10)
})
circos.clear()

</code></pre>

<hr>
<h2 id='circos.xaxis'>
Draw x-axis
</h2><span id='topic+circos.xaxis'></span>

<h3>Description</h3>

<p>Draw x-axis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.xaxis(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.xaxis_+3A_...">...</code></td>
<td>
<p>All pass to <code><a href="#topic+circos.axis">circos.axis</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is identical to <code><a href="#topic+circos.axis">circos.axis</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='circos.yaxis'>
Draw y-axis
</h2><span id='topic+circos.yaxis'></span>

<h3>Description</h3>

<p>Draw y-axis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circos.yaxis(
    side = c("left", "right"),
    at = NULL,
    labels = TRUE,
    tick = TRUE,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index(),
    labels.font = par("font"),
    labels.cex = par("cex"),
    labels.niceFacing = TRUE,
    tick.length = convert_x(1, "mm", sector.index, track.index),
    lwd = par("lwd"),
    col = par("col"),
    labels.col = par("col"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circos.yaxis_+3A_side">side</code></td>
<td>
<p>add the y-axis on the left or right of the cell</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_at">at</code></td>
<td>
<p>If it is numeric vector, it identifies the positions of the ticks. It can exceed <code>ylim</code> value and the exceeding part would be trimmed automatically.</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_labels">labels</code></td>
<td>
<p>labels of the ticks. The exceeding part would be trimmed automatically. The value can also be logical (either an atomic value or a vector) which represents which labels to show.</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_tick">tick</code></td>
<td>
<p>Whether to draw ticks.</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_labels.font">labels.font</code></td>
<td>
<p>font style for the axis labels</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_labels.cex">labels.cex</code></td>
<td>
<p>font size for the axis labels</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_labels.nicefacing">labels.niceFacing</code></td>
<td>
<p>Should facing of axis labels be human-easy</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_tick.length">tick.length</code></td>
<td>
<p>length of the tick</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_lwd">lwd</code></td>
<td>
<p>line width for ticks</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_col">col</code></td>
<td>
<p>color for the axes</p>
</td></tr>
<tr><td><code id="circos.yaxis_+3A_labels.col">labels.col</code></td>
<td>
<p>color for the labels</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, you need to set the gap between sectors manually by <code><a href="#topic+circos.par">circos.par</a></code> to make sure there is enough space
for y-axis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>op = par(no.readonly = TRUE)

sectors = letters[1:8]
circos.par(points.overflow.warning = FALSE)
circos.par(gap.degree = 8)
circos.initialize(sectors, xlim = c(0, 10))
circos.trackPlotRegion(sectors, ylim = c(0, 10), track.height = 0.5)
par(cex = 0.8)
for(a in letters[2:4]) {
  circos.yaxis(side = "left", sector.index = a)
}
for(a in letters[5:7]) {
  circos.yaxis(side = "right", sector.index = a)
}
circos.clear()

par(op)
</code></pre>

<hr>
<h2 id='cm_h'>
Convert units
</h2><span id='topic+cm_h'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cm_h(h)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cm_h_+3A_h">h</code></td>
<td>
<p>The height in numeric.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_length">convert_length</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_length` page
NULL
</code></pre>

<hr>
<h2 id='cm_x'>
Convert unit on x direction in data coordinate
</h2><span id='topic+cm_x'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cm_x(x, sector.index = get.current.sector.index(),
    track.index = get.current.track.index(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cm_x_+3A_x">x</code></td>
<td>
<p>The x-value in numeric.</p>
</td></tr>
<tr><td><code id="cm_x_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="cm_x_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
<tr><td><code id="cm_x_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+convert_x">convert_x</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_x">convert_x</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_x` page
NULL
</code></pre>

<hr>
<h2 id='cm_y'>
Convert unit on y direction in data coordinate
</h2><span id='topic+cm_y'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cm_y(y, sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cm_y_+3A_y">y</code></td>
<td>
<p>The y-value in numeric.</p>
</td></tr>
<tr><td><code id="cm_y_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="cm_y_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_y">convert_y</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_y` page
NULL
</code></pre>

<hr>
<h2 id='col2value'>
Transform back from colors to values
</h2><span id='topic+col2value'></span>

<h3>Description</h3>

<p>Transform back from colors to values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col2value(r, g, b, col_fun)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="col2value_+3A_r">r</code></td>
<td>
<p>red channel in <code><a href="colorspace.html#topic+sRGB">sRGB</a></code> color space, value should be between 0 and 1.  The <code>r</code>, <code>g</code> and <code>b</code> argumentc can be wrapped into one variable which is either a three-column matrix or a vector of colors.</p>
</td></tr>
<tr><td><code id="col2value_+3A_g">g</code></td>
<td>
<p>green channel in <code><a href="colorspace.html#topic+sRGB">sRGB</a></code> color space, value should be between 0 and 1.</p>
</td></tr>
<tr><td><code id="col2value_+3A_b">b</code></td>
<td>
<p>blue channel in <code><a href="colorspace.html#topic+sRGB">sRGB</a></code> color space, value should be between 0 and 1.</p>
</td></tr>
<tr><td><code id="col2value_+3A_col_fun">col_fun</code></td>
<td>
<p>the color mapping function generated by <code><a href="#topic+colorRamp2">colorRamp2</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+colorRamp2">colorRamp2</a></code> transforms values to colors and this function does the reversed job.
Note for some color spaces, it cannot transform back to the original value perfectly.
</p>


<h3>Value</h3>

<p>A vector of original numeric values.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = seq(0, 1, length.out = 11)
col_fun = colorRamp2(c(0, 0.5, 1), c("blue", "white", "red"))
col = col_fun(x)
col2value(col, col_fun = col_fun)
col2value("red", col_fun = col_fun)

col_fun = colorRamp2(c(0, 0.5, 1), c("blue", "white", "red"), space = "sRGB")
col = col_fun(x)
col2value(col, col_fun = col_fun)
</code></pre>

<hr>
<h2 id='colorRamp2'>
Color interpolation
</h2><span id='topic+colorRamp2'></span>

<h3>Description</h3>

<p>Color interpolation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colorRamp2(breaks, colors, transparency = 0, space = "LAB",
    hcl_palette = NULL, reverse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colorRamp2_+3A_breaks">breaks</code></td>
<td>
<p>A vector indicating numeric breaks</p>
</td></tr>
<tr><td><code id="colorRamp2_+3A_colors">colors</code></td>
<td>
<p>A vector of colors which correspond to values in <code>breaks</code></p>
</td></tr>
<tr><td><code id="colorRamp2_+3A_transparency">transparency</code></td>
<td>
<p>A single value in <code>[0, 1]</code>. 0 refers to no transparency and 1 refers to full transparency</p>
</td></tr>
<tr><td><code id="colorRamp2_+3A_space">space</code></td>
<td>
<p>color space in which colors are interpolated. Value should be one of &quot;RGB&quot;, &quot;LAB&quot;, &quot;XYZ&quot;, &quot;sRGB&quot;, &quot;LUV&quot;, see <code><a href="colorspace.html#topic+color-class">color-class</a></code> for detail.</p>
</td></tr>
<tr><td><code id="colorRamp2_+3A_hcl_palette">hcl_palette</code></td>
<td>
<p>Name of the HCL palette. Value should be supported in <code><a href="grDevices.html#topic+palettes">hcl.pals</a></code>.</p>
</td></tr>
<tr><td><code id="colorRamp2_+3A_reverse">reverse</code></td>
<td>
<p>Whether should the colors in <code>hcl_palette</code> be reversed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Colors are linearly interpolated according to break values and corresponding colors through CIE Lab color space (<code><a href="colorspace.html#topic+LAB">LAB</a></code>) by default. 
Values exceeding breaks will be assigned with corresponding maximum or minimum colors.
</p>


<h3>Value</h3>

<p>It returns a function which accepts a vector of numeric values and returns interpolated colors.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+col2value">col2value</a></code> converts back to the original values by providing the color mapping function generated by <code><a href="#topic+colorRamp2">colorRamp2</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>col_fun = colorRamp2(c(-1, 0, 1), c("green", "white", "red"))
col_fun(c(-2, -1, -0.5, 0, 0.5, 1, 2))
</code></pre>

<hr>
<h2 id='convert_height'>
Convert units
</h2><span id='topic+convert_height'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_height(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_height_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+convert_length">convert_length</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is same as <code><a href="#topic+convert_length">convert_length</a></code>. The reason for naming this function
is <code><a href="#topic+convert_length">convert_length</a></code> is mostely used for defining the height of tracks and track margins.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p>For pre-defined units, users can use <code><a href="#topic+cm_h">cm_h</a></code>, <code><a href="#topic+mm_h">mm_h</a></code> and <code><a href="#topic+inches_h">inches_h</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see example in `convert_length` page
NULL
</code></pre>

<hr>
<h2 id='convert_length'>
Convert units
</h2><span id='topic+convert_length'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_length(x, unit = c("mm", "cm", "inches"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_length_+3A_x">x</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="convert_length_+3A_unit">unit</code></td>
<td>
<p>supported units, only &quot;mm&quot;, &quot;cm&quot;, &quot;inches&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function coverts mm/cm/inches units to units measured in the canvas coordinate,
e.g. how much is it in the canvas coordinate for 1 mm/cm/inches.
</p>
<p>Since in the circular plot, the aspect ratio is always 1, it does not matter this conversion
is applied on x direction or y direction.
</p>
<p>This function is mainly used in the radical direction.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convert_x">convert_x</a></code> and <code><a href="#topic+convert_y">convert_y</a></code> convert absolute units into a data coordinate in a specified cell.
</p>
<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions">https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:10]
circos.par(cell.padding = c(0, 0, 0, 0), track.margin = c(0, 0))
circos.initialize(sectors, xlim = cbind(rep(0, 10), runif(10, 0.5, 1.5)))
circos.track(ylim = c(0, 1), track.height = mm_h(5))
circos.par(track.margin = c(0, mm_h(2)))
circos.track(ylim = c(0, 1), track.height = cm_h(1))
circos.par(track.margin = c(0, mm_h(5)))
circos.track(ylim = c(0, 1), track.height = inch_h(1))
circos.clear()
</code></pre>

<hr>
<h2 id='convert_x'>
Convert unit on x direction in data coordinate
</h2><span id='topic+convert_x'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_x(
    x,
    unit = c("mm", "cm", "inches"),
    sector.index = get.cell.meta.data("sector.index"),
    track.index = get.cell.meta.data("track.index"),
    h = get.cell.meta.data("ycenter", sector.index = sector.index,
    track.index = track.index))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_x_+3A_x">x</code></td>
<td>
<p>a numeric vector.</p>
</td></tr>
<tr><td><code id="convert_x_+3A_unit">unit</code></td>
<td>
<p>supported units, only &quot;mm&quot;, &quot;cm&quot;, &quot;inches&quot;.</p>
</td></tr>
<tr><td><code id="convert_x_+3A_sector.index">sector.index</code></td>
<td>
<p>index for the sector where the conversion is applied.</p>
</td></tr>
<tr><td><code id="convert_x_+3A_track.index">track.index</code></td>
<td>
<p>index for the track where the conversion is applied.</p>
</td></tr>
<tr><td><code id="convert_x_+3A_h">h</code></td>
<td>
<p>since the width of the cell is not identical from the top to the bottom in the cell, the position on y direction needs to be specified. By default it is at the middle point on y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numeric values which are measured in the specified data coordinate.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p>For pre-defined units, users can use <code><a href="#topic+cm_x">cm_x</a></code>, <code><a href="#topic+mm_x">mm_x</a></code> and <code><a href="#topic+inches_x">inches_x</a></code>.
</p>
<p><code><a href="#topic+convert_y">convert_y</a></code> converts on y direction.
</p>
<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions">https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:10]
circos.par(cell.padding = c(0, 0, 0, 0), track.margin = c(0, 0))
circos.initialize(sectors, xlim = cbind(rep(0, 10), runif(10, 0.5, 1.5)))
circos.track(ylim = c(0, 1), track.height = mm_h(5),
    panel.fun = function(x, y) {
        circos.lines(c(0, 0 + mm_x(5)), c(0.5, 0.5), col = "blue")
    })
circos.par(track.margin = c(0, mm_h(2)))
circos.track(ylim = c(0, 1), track.height = cm_h(1),
    panel.fun = function(x, y) {
        xcenter = get.cell.meta.data("xcenter")
        circos.lines(c(xcenter, xcenter), c(0, cm_y(1)), col = "red")
    })
circos.par(track.margin = c(0, mm_h(5)))
circos.track(ylim = c(0, 1), track.height = inch_h(1),
    panel.fun = function(x, y) {
        line_length_on_x = cm_x(1*sqrt(2)/2)
        line_length_on_y = cm_y(1*sqrt(2)/2)
        circos.lines(c(0, line_length_on_x), c(0, line_length_on_y), col = "orange")
    })
circos.clear()
</code></pre>

<hr>
<h2 id='convert_y'>
Convert unit on y direction in data coordinate
</h2><span id='topic+convert_y'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_y(
    x,
    unit = c("mm", "cm", "inches"),
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_y_+3A_x">x</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="convert_y_+3A_unit">unit</code></td>
<td>
<p>supported units, only &quot;mm&quot;, &quot;cm&quot;, &quot;inches&quot;</p>
</td></tr>
<tr><td><code id="convert_y_+3A_sector.index">sector.index</code></td>
<td>
<p>index for the sector where the conversion is applied</p>
</td></tr>
<tr><td><code id="convert_y_+3A_track.index">track.index</code></td>
<td>
<p>index for the track where the conversion is applied</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numeric values which are measured in the specified data coordinate
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>See Also</h3>

<p>For pre-defined units, users can use <code><a href="#topic+cm_y">cm_y</a></code>, <code><a href="#topic+mm_y">mm_y</a></code> and <code><a href="#topic+inches_y">inches_y</a></code>.
</p>
<p><code><a href="#topic+convert_x">convert_x</a></code> converts on x direction.
</p>
<p><a href="https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions">https://jokergoo.github.io/circlize_book/book/circular-layout.html#convert-functions</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see example on `convert_x` page 
NULL
</code></pre>

<hr>
<h2 id='cytoband.col'>
Assign colors to cytogenetic band (hg19) according to the Giemsa stain results
</h2><span id='topic+cytoband.col'></span>

<h3>Description</h3>

<p>Assign colors to cytogenetic band (hg19) according to the Giemsa stain results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytoband.col(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cytoband.col_+3A_x">x</code></td>
<td>
<p>A vector containing the Giemsa stain results
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='degree'>
Mark the value as a degree value
</h2><span id='topic+degree'></span>

<h3>Description</h3>

<p>Mark the value as a degree value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>degree(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="degree_+3A_x">x</code></td>
<td>
<p>degree value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>degree</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='draw.sector'>
Draw sectors or rings in a circle
</h2><span id='topic+draw.sector'></span>

<h3>Description</h3>

<p>Draw sectors or rings in a circle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw.sector(
    start.degree = 0,
    end.degree = 360,
    rou1 = 1,
    rou2 = NULL,
    center = c(0, 0),
    clock.wise = TRUE,
    col = NA,
    border = "black",
    lwd = par("lwd"),
    lty = par("lty"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw.sector_+3A_start.degree">start.degree</code></td>
<td>
<p>start degree for the sector</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_end.degree">end.degree</code></td>
<td>
<p>end degree for the sector</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_rou1">rou1</code></td>
<td>
<p>Radius for one of the arc in the sector</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_rou2">rou2</code></td>
<td>
<p>Radius for the other arc in the sector</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_center">center</code></td>
<td>
<p>Center of the circle</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_clock.wise">clock.wise</code></td>
<td>
<p>The direction from <code>start.degree</code> to <code>end.degree</code></p>
</td></tr>
<tr><td><code id="draw.sector_+3A_col">col</code></td>
<td>
<p>Filled color</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_border">border</code></td>
<td>
<p>Border color</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_lwd">lwd</code></td>
<td>
<p>Line width</p>
</td></tr>
<tr><td><code id="draw.sector_+3A_lty">lty</code></td>
<td>
<p>Line style</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the interval between <code>start</code> and <code>end</code> (larger or equal to 360 or smaller or equal to -360)
it would draw a full circle or ring. If <code>rou2</code> is set, it would draw part of a ring.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(c(-1, 1), c(-1, 1), type = "n", axes = FALSE, ann = FALSE, asp = 1)
draw.sector(20, 0)
draw.sector(30, 60, rou1 = 0.8, rou2 = 0.5, clock.wise = FALSE, col = "#FF000080")
draw.sector(350, 1000, col = "#00FF0080", border = NA)
draw.sector(0, 180, rou1 = 0.25, center = c(-0.5, 0.5), border = 2, lwd = 2, lty = 2)
draw.sector(0, 360, rou1 = 0.7, rou2 = 0.6, col = "#0000FF80")

sectors = letters[1:8]
circos.initialize(sectors, xlim = c(0, 1))
for(i in 1:3) {
    circos.trackPlotRegion(ylim = c(0, 1))
}
circos.info(plot = TRUE)

draw.sector(get.cell.meta.data("cell.start.degree", sector.index = "a"),
            get.cell.meta.data("cell.end.degree", sector.index = "a"),
            rou1 = 1, col = "#FF000040")

draw.sector(0, 360,
    rou1 = get.cell.meta.data("cell.top.radius", track.index = 1),
    rou2 = get.cell.meta.data("cell.bottom.radius", track.index = 1),
    col = "#00FF0040")

draw.sector(get.cell.meta.data("cell.start.degree", sector.index = "e"),
            get.cell.meta.data("cell.end.degree", sector.index = "f"),
            get.cell.meta.data("cell.top.radius", track.index = 2),
            get.cell.meta.data("cell.bottom.radius", track.index = 3),
            col = "#0000FF40")

pos = circlize(c(0.2, 0.8), c(0.2, 0.8), sector.index = "h", track.index = 2)
draw.sector(pos[1, "theta"], pos[2, "theta"], pos[1, "rou"], pos[2, "rou"],
    clock.wise = TRUE, col = "#00FFFF40")
circos.clear()
</code></pre>

<hr>
<h2 id='fontsize'>
Convert fontsize to cex
</h2><span id='topic+fontsize'></span>

<h3>Description</h3>

<p>Convert fontsize to cex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fontsize(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fontsize_+3A_x">x</code></td>
<td>
<p>value for fontsize</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='generateRandomBed'>
Generate random genomic data
</h2><span id='topic+generateRandomBed'></span>

<h3>Description</h3>

<p>Generate random genomic data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateRandomBed(
    nr = 10000,
    nc = 1,
    fun = function(k) rnorm(k, 0, 0.5),
    species = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateRandomBed_+3A_nr">nr</code></td>
<td>
<p>Number of rows
</p>
</td></tr>
<tr><td><code id="generateRandomBed_+3A_nc">nc</code></td>
<td>
<p>Number of numeric columns / value columns
</p>
</td></tr>
<tr><td><code id="generateRandomBed_+3A_fun">fun</code></td>
<td>
<p>Function for generating random values
</p>
</td></tr>
<tr><td><code id="generateRandomBed_+3A_species">species</code></td>
<td>
<p>species, pass to <code><a href="#topic+read.cytoband">read.cytoband</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will uniformly sample positions from the genome. Chromosome names start with &quot;chr&quot;
and positions are sorted. The final number of rows may not be exactly as same as <code>nr</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='genomicDensity'>
Calculate genomic region density
</h2><span id='topic+genomicDensity'></span>

<h3>Description</h3>

<p>Calculate genomic region density
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomicDensity(
    region,
    window.size = 1e7,
    n.window = NULL,
    overlap = TRUE,
    count_by = c("percent", "number"),
    chr.len = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomicDensity_+3A_region">region</code></td>
<td>
<p>Genomic positions. It can be a data frame with two
columns which are start positions and end positions on a single chromosome.
It can also be a bed-format data frame which contains the chromosome column.
</p>
</td></tr>
<tr><td><code id="genomicDensity_+3A_window.size">window.size</code></td>
<td>
<p>Window size to calculate genomic density
</p>
</td></tr>
<tr><td><code id="genomicDensity_+3A_n.window">n.window</code></td>
<td>
<p>number of windows, if it is specified, <code>window.size</code> is ignored
</p>
</td></tr>
<tr><td><code id="genomicDensity_+3A_overlap">overlap</code></td>
<td>
<p>Whether two neighbouring windows have half overlap
</p>
</td></tr>
<tr><td><code id="genomicDensity_+3A_count_by">count_by</code></td>
<td>
<p>How to count the value for each window, <code>percent</code>: percent of the window covered by the input regions; <code>number</code>: number of regions that overlap to the window.
</p>
</td></tr>
<tr><td><code id="genomicDensity_+3A_chr.len">chr.len</code></td>
<td>
<p>the chromosome length. The value should be named vector
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It calculate the percent of each genomic windows that is covered by the input regions.
</p>


<h3>Value</h3>

<p>If the input is a two-column data frame, the function returns a data frame with three columns: 
start position, end position and the overlapping (value depends on the <code>count_by</code> argument). And if the input is a bed-format
data frame, there will be an additionally chromosome name column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bed = generateRandomBed()
bed = subset(bed, chr == "chr1")
head(genomicDensity(bed))
head(genomicDensity(bed, count_by = "number"))
</code></pre>

<hr>
<h2 id='get_most_inside_radius'>
Get the inside radius of the most inner track
</h2><span id='topic+get_most_inside_radius'></span>

<h3>Description</h3>

<p>Get the inside radius of the most inner track
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_most_inside_radius()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='get.all.sector.index'>
Get index for all sectors
</h2><span id='topic+get.all.sector.index'></span>

<h3>Description</h3>

<p>Get index for all sectors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.all.sector.index()
</code></pre>


<h3>Details</h3>

<p>It simply returns a vector of all sector index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='get.all.track.index'>
Get index for all tracks
</h2><span id='topic+get.all.track.index'></span>

<h3>Description</h3>

<p>Get index for all tracks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.all.track.index()
</code></pre>


<h3>Details</h3>

<p>It simply returns a vector of all track index.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='get.cell.meta.data'>
Get the meta data of a cell
</h2><span id='topic+get.cell.meta.data'></span>

<h3>Description</h3>

<p>Get the meta data of a cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.cell.meta.data(name, sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.cell.meta.data_+3A_name">name</code></td>
<td>
<p>Only support one name at a time, see &quot;details&quot; section
</p>
</td></tr>
<tr><td><code id="get.cell.meta.data_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of the sector
</p>
</td></tr>
<tr><td><code id="get.cell.meta.data_+3A_track.index">track.index</code></td>
<td>
<p>Index of the track
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following meta information for a cell can be obtained:
</p>

<dl>
<dt><code>sector.index</code></dt><dd><p>The name (index) for the sector
</p>
</dd>
<dt><code>sector.numeric.index</code></dt><dd><p>Numeric index for the sector
</p>
</dd>
<dt><code>track.index</code></dt><dd><p>Numeric index for the track
</p>
</dd>
<dt><code>xlim</code></dt><dd><p>Minimal and maximal values on the x-axis
</p>
</dd>
<dt><code>ylim</code></dt><dd><p>Minimal and maximal values on the y-axis
</p>
</dd>
<dt><code>xrange</code></dt><dd><p>Range of <code>xlim</code>. It equals to <code>xlim[2] - xlim[1]</code>
</p>
</dd>
<dt><code>yrange</code></dt><dd><p>Range of <code>ylim</code>
</p>
</dd>
<dt><code>xcenter</code></dt><dd><p>Center of x-axis. It equals to <code>(xlim[2] + xlim[1])/2</code>
</p>
</dd>
<dt><code>ycenter</code></dt><dd><p>Center of y-axis
</p>
</dd>
<dt><code>cell.xlim</code></dt><dd><p>Minimal and maximal values on the x-axis extended by cell paddings
</p>
</dd>
<dt><code>cell.ylim</code></dt><dd><p>Minimal and maximal values on the y-axis extended by cell paddings
</p>
</dd>
<dt><code>xplot</code></dt><dd><p>Degrees for right and left borders of the cell. The values ignore the direction of the circular layout (i.e. whether it is clock wise or not).
</p>
</dd>
<dt><code>yplot</code></dt><dd><p>Radius for top and bottom borders of the cell.
</p>
</dd>
<dt><code>cell.width</code></dt><dd><p>Width of the cell, in degrees.
</p>
</dd>
<dt><code>cell.height</code></dt><dd><p>Height of the cell, simply <code>yplot[2] - yplot[1]</code>
</p>
</dd>
<dt><code>cell.start.degree</code></dt><dd><p>Same as <code>xplot[1]</code>
</p>
</dd>
<dt><code>cell.end.degree</code></dt><dd><p>Same as <code>xplot[2]</code>
</p>
</dd>
<dt><code>cell.bottom.radius</code></dt><dd><p>Same as <code>yplot[1]</code>
</p>
</dd>
<dt><code>cell.top.radius</code></dt><dd><p>Same as <code>yplot[2]</code>
</p>
</dd>
<dt><code>track.margin</code></dt><dd><p>Margin for the cell
</p>
</dd>
<dt><code>cell.padding</code></dt><dd><p>Padding for the cell
</p>
</dd>
</dl>

<p>The function is useful when using <code>panel.fun</code> in <code><a href="#topic+circos.track">circos.track</a></code> to
get detailed information of the current cell.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CELL_META">CELL_META</a></code> is a short version of <code><a href="#topic+get.cell.meta.data">get.cell.meta.data</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:4]
circos.initialize(sectors, xlim = c(0, 1))
circos.trackPlotRegion(ylim = c(0, 1), panel.fun = function(x, y) {
    print(get.cell.meta.data("xlim"))
})
print(get.cell.meta.data("xlim", sector.index = "a", track.index = 1))
circos.clear()
</code></pre>

<hr>
<h2 id='get.current.chromosome'>
Get current chromosome name
</h2><span id='topic+get.current.chromosome'></span>

<h3>Description</h3>

<p>Get current chromosome name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.current.chromosome()
</code></pre>


<h3>Details</h3>

<p>The function is same as <code><a href="#topic+get.current.sector.index">get.current.sector.index</a></code> and
should only be put inside <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrackPlotRegion">circos.genomicTrackPlotRegion</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='get.current.sector.index'>
Get current sector index
</h2><span id='topic+get.current.sector.index'></span>

<h3>Description</h3>

<p>Get current sector index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.current.sector.index()
</code></pre>


<h3>Value</h3>

<p>Simply returns the name of current sector
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='get.current.track.index'>
Get current track index
</h2><span id='topic+get.current.track.index'></span>

<h3>Description</h3>

<p>Get current track index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.current.track.index()
</code></pre>


<h3>Value</h3>

<p>Simply returns the numeric index for the current track.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='getI'>
Which data that <code>panel.fun</code> is using
</h2><span id='topic+getI'></span>

<h3>Description</h3>

<p>Which data that <code>panel.fun</code> is using
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getI(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getI_+3A_...">...</code></td>
<td>
<p>Invisible arguments that users do not need to care
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function should only be put inside <code>panel.fun</code> when using <code><a href="#topic+circos.genomicTrackPlotRegion">circos.genomicTrackPlotRegion</a></code>.
</p>
<p>If <code>stack</code> is set to <code>TRUE</code> in <code><a href="#topic+circos.genomicTrackPlotRegion">circos.genomicTrackPlotRegion</a></code>, the returned value
indicates which stack the function will be applied to.
</p>
<p>If <code>data</code> is a list of data frames, the value
indicates which data frame is being used. Please see the vignette to get a more clear explanation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='highlight.chromosome'>
Highlight chromosomes
</h2><span id='topic+highlight.chromosome'></span>

<h3>Description</h3>

<p>Highlight chromosomes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlight.chromosome(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="highlight.chromosome_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+highlight.sector">highlight.sector</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is only a shortcut function of <code><a href="#topic+highlight.sector">highlight.sector</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='highlight.sector'>
Highlight sectors and tracks
</h2><span id='topic+highlight.sector'></span>

<h3>Description</h3>

<p>Highlight sectors and tracks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlight.sector(
    sector.index,
    track.index = get.all.track.index(),
    col = "#FF000040",
    border = NA,
    lwd = par("lwd"),
    lty = par("lty"),
    padding = c(0, 0, 0, 0),
    text = NULL,
    text.col = par("col"),
    text.vjust = 0.5,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="highlight.sector_+3A_sector.index">sector.index</code></td>
<td>
<p>A vector of sector index</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_track.index">track.index</code></td>
<td>
<p>A vector of track index that you want to highlight</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_col">col</code></td>
<td>
<p>Color for highlighting. Note the color should be semi-transparent.</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_border">border</code></td>
<td>
<p>Border of the highlighted region</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_lwd">lwd</code></td>
<td>
<p>Width of borders</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_lty">lty</code></td>
<td>
<p>Style of borders</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_padding">padding</code></td>
<td>
<p>Padding for the highlighted region. It should contain four values representing ratios of the width or height of the highlighted region</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_text">text</code></td>
<td>
<p>text added in the highlight region, only support plotting one string at a time</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_text.vjust">text.vjust</code></td>
<td>
<p>adjustment on 'vertical' (radical) direction. Besides to set it as numeric values, the value can also be a string contain absoute unit, e.g. &quot;2.1mm&quot;, &quot;-1 inche&quot;, but only &quot;mm&quot;, &quot;cm&quot;, &quot;inches&quot;/&quot;inche&quot; are allowed.</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_text.col">text.col</code></td>
<td>
<p>color for the text</p>
</td></tr>
<tr><td><code id="highlight.sector_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+circos.text">circos.text</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can use <code><a href="#topic+circos.info">circos.info</a></code> to find out index for all sectors and all tracks.
</p>
<p>The function calls <code><a href="#topic+draw.sector">draw.sector</a></code>.
</p>


<h3>See Also</h3>

<p><a href="https://jokergoo.github.io/circlize_book/book/graphics.html#highlight-sectors-and-tracks">https://jokergoo.github.io/circlize_book/book/graphics.html#highlight-sectors-and-tracks</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sectors = letters[1:8]
circos.initialize(sectors, xlim = c(0, 1))
for(i in 1:4) {
    circos.trackPlotRegion(ylim = c(0, 1))
}
circos.info(plot = TRUE)

highlight.sector(c("a", "h"), track.index = 1)
highlight.sector("c", col = "#00FF0040")
highlight.sector("d", col = NA, border = "red", lwd = 2)
highlight.sector("e", col = "#0000FF40", track.index = c(2, 3))
highlight.sector(c("f", "g"), col = NA, border = "green",
    lwd = 2, track.index = c(2, 3))
highlight.sector(sectors, col = "#FFFF0040", track.index = 4)
circos.clear()
</code></pre>

<hr>
<h2 id='inch_h'>
Convert units
</h2><span id='topic+inch_h'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inch_h(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inch_h_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+inches_h">inches_h</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the same as <code><a href="#topic+inches_h">inches_h</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='inch_x'>
Convert unit on x direction in data coordinate
</h2><span id='topic+inch_x'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inch_x(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inch_x_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+inches_x">inches_x</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the same as <code><a href="#topic+inches_x">inches_x</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='inch_y'>
Convert unit on y direction in data coordinate
</h2><span id='topic+inch_y'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inch_y(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inch_y_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+inches_y">inches_y</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the same as <code><a href="#topic+inches_y">inches_y</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='inches_h'>
Convert units
</h2><span id='topic+inches_h'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inches_h(h)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inches_h_+3A_h">h</code></td>
<td>
<p>The height in numeric.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_length">convert_length</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_length` page
NULL
</code></pre>

<hr>
<h2 id='inches_x'>
Convert unit on x direction in data coordinate
</h2><span id='topic+inches_x'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inches_x(x, sector.index = get.current.sector.index(),
    track.index = get.current.track.index(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inches_x_+3A_x">x</code></td>
<td>
<p>The x-value in numeric.</p>
</td></tr>
<tr><td><code id="inches_x_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="inches_x_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
<tr><td><code id="inches_x_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+convert_x">convert_x</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_x">convert_x</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_x` page
NULL
</code></pre>

<hr>
<h2 id='inches_y'>
Convert unit on y direction in data coordinate
</h2><span id='topic+inches_y'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inches_y(y, sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inches_y_+3A_y">y</code></td>
<td>
<p>The y-value in numeric.</p>
</td></tr>
<tr><td><code id="inches_y_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="inches_y_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_y">convert_y</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_y` page
NULL
</code></pre>

<hr>
<h2 id='mm_h'>
Convert units
</h2><span id='topic+mm_h'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm_h(h)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mm_h_+3A_h">h</code></td>
<td>
<p>The height in numeric.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_length">convert_length</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_length` page
NULL
</code></pre>

<hr>
<h2 id='mm_x'>
Convert unit on x direction in data coordinate
</h2><span id='topic+mm_x'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm_x(x, sector.index = get.current.sector.index(),
    track.index = get.current.track.index(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mm_x_+3A_x">x</code></td>
<td>
<p>The x-value in numeric.</p>
</td></tr>
<tr><td><code id="mm_x_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="mm_x_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
<tr><td><code id="mm_x_+3A_...">...</code></td>
<td>
<p>Pass to <code><a href="#topic+convert_x">convert_x</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_x">convert_x</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_x` page
NULL
</code></pre>

<hr>
<h2 id='mm_y'>
Convert unit on y direction in data coordinate
</h2><span id='topic+mm_y'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm_y(y, sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mm_y_+3A_y">y</code></td>
<td>
<p>The y-value in numeric.</p>
</td></tr>
<tr><td><code id="mm_y_+3A_sector.index">sector.index</code></td>
<td>
<p>Index of sector.</p>
</td></tr>
<tr><td><code id="mm_y_+3A_track.index">track.index</code></td>
<td>
<p>Index of track.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See explanations in <code><a href="#topic+convert_y">convert_y</a></code> page.
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see examples in `convert_y` page
NULL
</code></pre>

<hr>
<h2 id='names.CELL_META'>
Names of all meta data in the current cell
</h2><span id='topic+names.CELL_META'></span>

<h3>Description</h3>

<p>Names of all meta data in the current cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CELL_META'
names(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names.CELL_META_+3A_x">x</code></td>
<td>
<p>use <code><a href="#topic+CELL_META">CELL_META</a></code>.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>names(CELL_META)
</code></pre>

<hr>
<h2 id='posTransform.default'>
Genomic position transformation function
</h2><span id='topic+posTransform.default'></span>

<h3>Description</h3>

<p>Genomic position transformation function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posTransform.default(region, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posTransform.default_+3A_region">region</code></td>
<td>
<p>Genomic positions at a single chromosome. It is a data frame with two
columns which are start position and end position.
</p>
</td></tr>
<tr><td><code id="posTransform.default_+3A_...">...</code></td>
<td>
<p>other arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default position transformation functions transforms position to be equally distributed
along the chromosome. If users want to define their own transformation function, the requirement
is that the returned value should be a data frame with two columns: transformed start position
and transformed end position. The returned value should have same number of rows as the input one.
</p>
<p>For details why need to use position transformation, please refer to <code><a href="#topic+circos.genomicPosTransformLines">circos.genomicPosTransformLines</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='posTransform.text'>
Genomic position transformation function specifically for text
</h2><span id='topic+posTransform.text'></span>

<h3>Description</h3>

<p>Genomic position transformation function specifically for text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posTransform.text(
    region,
    y,
    labels,
    cex = 1,
    font = par("font"),
    sector.index = get.cell.meta.data("sector.index"),
    track.index = get.cell.meta.data("track.index"),
    padding = 0,
    extend = 0,
    ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="posTransform.text_+3A_region">region</code></td>
<td>
<p>Genomic positions at a single chromosome. It is a data frame with two
columns which are start position and end position.
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_y">y</code></td>
<td>
<p>positions of texts
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_labels">labels</code></td>
<td>
<p>text labels
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_cex">cex</code></td>
<td>
<p>text size
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_font">font</code></td>
<td>
<p>text font style
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_sector.index">sector.index</code></td>
<td>
<p>sector index
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_track.index">track.index</code></td>
<td>
<p>track index
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_padding">padding</code></td>
<td>
<p>padding of text
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_extend">extend</code></td>
<td>
<p>extend to allow labels to be put in an region which is wider than the current chromosome.
The value should be a proportion value and the length is either one or two.
</p>
</td></tr>
<tr><td><code id="posTransform.text_+3A_...">...</code></td>
<td>
<p>other arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This position transformation function is designed specifically for text.
Under the transformation, texts will be as close as possible to the original positions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='print.CELL_META'>
Print CELL_META
</h2><span id='topic+print.CELL_META'></span>

<h3>Description</h3>

<p>Print CELL_META
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CELL_META'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.CELL_META_+3A_x">x</code></td>
<td>
<p>input
</p>
</td></tr>
<tr><td><code id="print.CELL_META_+3A_...">...</code></td>
<td>
<p>additional parameters
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='rainfallTransform'>
Calculate inter-distance of genomic regions
</h2><span id='topic+rainfallTransform'></span>

<h3>Description</h3>

<p>Calculate inter-distance of genomic regions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rainfallTransform(
    region,
    mode = c("min", "max", "mean", "left", "right"),
    normalize_to_width = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rainfallTransform_+3A_region">region</code></td>
<td>
<p>Genomic positions. It can be a data frame with two
columns which are start positions and end positions on a single chromosome.
It can also be a bed-format data frame which contains the chromosome column.
</p>
</td></tr>
<tr><td><code id="rainfallTransform_+3A_mode">mode</code></td>
<td>
<p>How to calculate inter-distance. For a region, there is a distance to the 
prevous region and also there is a distance to the next region. <code>mode</code>
controls how to merge these two distances into one value.
</p>
</td></tr>
<tr><td><code id="rainfallTransform_+3A_normalize_to_width">normalize_to_width</code></td>
<td>
<p>If it is <code>TRUE</code>, the value is the relative distance divided by the width of the region.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the input is a two-column data frame, the function returnes a data frame with three columns: start position, end position and distance.
And if the input is a bed-format data frame, there will be the chromosome column added.
</p>
<p>The row order of the returned data frame is as same as the input one.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bed = generateRandomBed()
bed = subset(bed, chr == "chr1")
head(rainfallTransform(bed))
</code></pre>

<hr>
<h2 id='rand_color'>
Generate random colors
</h2><span id='topic+rand_color'></span>

<h3>Description</h3>

<p>Generate random colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rand_color(n, hue = NULL, luminosity = "random", transparency = 0, friendly = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rand_color_+3A_n">n</code></td>
<td>
<p>number of colors</p>
</td></tr>
<tr><td><code id="rand_color_+3A_hue">hue</code></td>
<td>
<p>the hue of the generated color. You can use following default color name: <code>red</code>, <code>orange</code>,  <code>yellow</code>, <code>green</code>, <code>blue</code>, <code>purple</code>, <code>pink</code> and <code>monochrome</code>. If the value is a hexidecimal color string such as <code>#00FFFF</code>,  the function will extract its hue value and use that to generate colors.</p>
</td></tr>
<tr><td><code id="rand_color_+3A_luminosity">luminosity</code></td>
<td>
<p>controls the luminosity of the generated color. The value should be a string containing <code>bright</code>, <code>light</code>, <code>dark</code> and <code>random</code>.</p>
</td></tr>
<tr><td><code id="rand_color_+3A_transparency">transparency</code></td>
<td>
<p>transparency, numeric value between 0 and 1.</p>
</td></tr>
<tr><td><code id="rand_color_+3A_friendly">friendly</code></td>
<td>
<p>If it is true, light random colors will not be generated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code is adapted from randomColor.js (<a href="https://github.com/davidmerfield/randomColor">https://github.com/davidmerfield/randomColor</a> ).
</p>


<h3>Author(s)</h3>

<p>Zuguang Gu &lt;z.gu@dkfz.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(NULL, xlim = c(1, 10), ylim = c(1, 8), axes = FALSE, ann = FALSE)
points(1:10, rep(1, 10), pch = 16, cex = 5, 
    col = rand_color(10))
points(1:10, rep(2, 10), pch = 16, cex = 5, 
    col = rand_color(10, luminosity = "bright"))
points(1:10, rep(3, 10), pch = 16, cex = 5, 
    col = rand_color(10, luminosity = "light"))
points(1:10, rep(4, 10), pch = 16, cex = 5, 
    col = rand_color(10, luminosity = "dark"))
points(1:10, rep(5, 10), pch = 16, cex = 5, 
    col = rand_color(10, hue = "red", luminosity = "bright"))
points(1:10, rep(6, 10), pch = 16, cex = 5, 
    col = rand_color(10, hue = "green", luminosity = "bright"))
points(1:10, rep(7, 10), pch = 16, cex = 5, 
    col = rand_color(10, hue = "blue", luminosity = "bright"))
points(1:10, rep(8, 10), pch = 16, cex = 5, 
    col = rand_color(10, hue = "monochrome", luminosity = "bright"))
</code></pre>

<hr>
<h2 id='read.chromInfo'>
Read/parse chromInfo data from a data frame/file/UCSC database
</h2><span id='topic+read.chromInfo'></span>

<h3>Description</h3>

<p>Read/parse chromInfo data from a data frame/file/UCSC database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.chromInfo(
    chromInfo = system.file(package = "circlize", "extdata", "chromInfo.txt"),
    species = NULL,
    chromosome.index = usable_chromosomes(species),
    sort.chr = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.chromInfo_+3A_chrominfo">chromInfo</code></td>
<td>
<p>Path of the chromInfo file or a data frame that already contains chromInfo data
</p>
</td></tr>
<tr><td><code id="read.chromInfo_+3A_species">species</code></td>
<td>
<p>Abbreviations of species. e.g. hg19 for human, mm10 for mouse. If this
value is specified, the function will download <code>chromInfo.txt.gz</code> from
UCSC website automatically.
</p>
</td></tr>
<tr><td><code id="read.chromInfo_+3A_chromosome.index">chromosome.index</code></td>
<td>
<p>subset of chromosomes, also used to reorder chromosomes.
</p>
</td></tr>
<tr><td><code id="read.chromInfo_+3A_sort.chr">sort.chr</code></td>
<td>
<p>Whether chromosome names should be sorted (first sort by numbers then by letters).
If <code>chromosome.index</code> is set, this argument is enforced to <code>FALSE</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function read the chromInfo data, sort the chromosome names and calculate the length of each chromosome.
By default, it is human hg19 chromInfo data.
</p>
<p>You can find the data structure for the chromInfo data from <a href="https://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/chromInfo.txt.gz">https://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/chromInfo.txt.gz</a>
</p>


<h3>Value</h3>


<dl>
<dt><code>df</code></dt><dd><p>Data frame for chromInfo data (rows are sorted if <code>sort.chr</code> is set to <code>TRUE</code>)
</p>
</dd>
<dt><code>chromosome</code></dt><dd><p>Sorted chromosome names
</p>
</dd>
<dt><code>chr.len</code></dt><dd><p>Length of chromosomes. Order are same as <code>chromosome</code>
</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data = read.chromInfo(species = "hg19")
data = read.chromInfo(chromInfo = system.file(package = "circlize", "extdata", "chromInfo.txt"))
chromInfo = read.table(system.file(package = "circlize", "extdata", "chromInfo.txt"),
    colClasses = c("character", "numeric"), sep = "\t")
data = read.chromInfo(chromInfo = chromInfo)
</code></pre>

<hr>
<h2 id='read.cytoband'>
Read/parse cytoband data from a data frame/file/UCSC database
</h2><span id='topic+read.cytoband'></span>

<h3>Description</h3>

<p>Read/parse cytoband data from a data frame/file/UCSC database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.cytoband(
    cytoband = system.file(package = "circlize", "extdata", "cytoBand.txt"),
    species = NULL,
    chromosome.index = usable_chromosomes(species),
    sort.chr = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.cytoband_+3A_cytoband">cytoband</code></td>
<td>
<p>Path of the cytoband file or a data frame that already contains cytoband data
</p>
</td></tr>
<tr><td><code id="read.cytoband_+3A_species">species</code></td>
<td>
<p>Abbreviations of species. e.g. hg19 for human, mm10 for mouse. If this
value is specified, the function will download <code>cytoBand.txt.gz</code> from
UCSC website automatically.
</p>
</td></tr>
<tr><td><code id="read.cytoband_+3A_chromosome.index">chromosome.index</code></td>
<td>
<p>subset of chromosomes, also used to reorder chromosomes.
</p>
</td></tr>
<tr><td><code id="read.cytoband_+3A_sort.chr">sort.chr</code></td>
<td>
<p>Whether chromosome names should be sorted (first sort by numbers then by letters).
If <code>chromosome.index</code> is set, this argument is enforced to <code>FALSE</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function read the cytoband data, sort the chromosome names and calculate the length of each chromosome.
By default, it is human hg19 cytoband data.
</p>
<p>You can find the data structure of the cytoband data from <a href="https://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/cytoBand.txt.gz">https://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/cytoBand.txt.gz</a>
</p>


<h3>Value</h3>


<dl>
<dt><code>df</code></dt><dd><p>Data frame for cytoband data (rows are sorted if <code>sort.chr</code> is set to <code>TRUE</code>)
</p>
</dd>
<dt><code>chromosome</code></dt><dd><p>Sorted chromosome names
</p>
</dd>
<dt><code>chr.len</code></dt><dd><p>Length of chromosomes. Orders are same as <code>chromosome</code>
</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data = read.cytoband(species = "hg19")
data = read.cytoband(cytoband = system.file(package = "circlize", "extdata", "cytoBand.txt"))
cytoband = read.table(system.file(package = "circlize", "extdata", "cytoBand.txt"),
    colClasses = c("character", "numeric", "numeric", "character", "character"), sep = "\t")
data = read.cytoband(cytoband = cytoband)
</code></pre>

<hr>
<h2 id='reverse.circlize'>
Convert to data coordinate system
</h2><span id='topic+reverse.circlize'></span>

<h3>Description</h3>

<p>Convert to data coordinate system
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reverse.circlize(
    x, y,
    sector.index = get.current.sector.index(),
    track.index = get.current.track.index())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reverse.circlize_+3A_x">x</code></td>
<td>
<p>degree values. The value can also be a two-column matrix/data frame if you put x and y data points into one variable.</p>
</td></tr>
<tr><td><code id="reverse.circlize_+3A_y">y</code></td>
<td>
<p>distance to the circle center (the radius)</p>
</td></tr>
<tr><td><code id="reverse.circlize_+3A_sector.index">sector.index</code></td>
<td>
<p>Index for the sector where the data coordinate is used</p>
</td></tr>
<tr><td><code id="reverse.circlize_+3A_track.index">track.index</code></td>
<td>
<p>Index for the track where the data coordinate is used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the reverse function of <code><a href="#topic+circlize">circlize</a></code>. It transform data points from polar coordinate system to a specified data coordinate system.
</p>


<h3>Value</h3>

<p>A matrix with two columns (<code>x</code> and <code>y</code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdf(NULL)
sectors = letters[1:4]
circos.initialize(sectors, xlim = c(0, 1))
circos.trackPlotRegion(ylim = c(0, 1))
reverse.circlize(c(30, 60), c(0.9, 0.8))
reverse.circlize(c(30, 60), c(0.9, 0.8), sector.index = "d", track.index = 1)
reverse.circlize(c(30, 60), c(0.9, 0.8), sector.index = "a", track.index = 1)
circos.clear()
dev.off()
</code></pre>

<hr>
<h2 id='set_track_gap'>
Set gaps between tracks
</h2><span id='topic+set_track_gap'></span>

<h3>Description</h3>

<p>Set gaps between tracks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_track_gap(gap = 0.02)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_track_gap_+3A_gap">gap</code></td>
<td>
<p>Gap between two tracks. Use <code><a href="#topic+mm_h">mm_h</a></code>/<code><a href="#topic+cm_h">cm_h</a></code>/<code><a href="#topic+inches_h">inches_h</a></code> to set in absolute units.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>circos.initialize(letters[1:10], xlim = c(0, 1))
circos.track(ylim = c(0, 1))
set_track_gap(mm_h(2))
circos.track(ylim = c(0, 1))
circos.clear()
</code></pre>

<hr>
<h2 id='set.current.cell'>
Set flag to current cell
</h2><span id='topic+set.current.cell'></span>

<h3>Description</h3>

<p>Set flag to current cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set.current.cell(sector.index, track.index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set.current.cell_+3A_sector.index">sector.index</code></td>
<td>
<p>sector index
</p>
</td></tr>
<tr><td><code id="set.current.cell_+3A_track.index">track.index</code></td>
<td>
<p>track index
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After setting the current cell, all functions which need <code>sector.index</code> and <code>track.index</code>
arguments and are applied to the current cell do not need to specify the two arguments explicitly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdf(NULL)
circos.initialize(letters[1:8], xlim = c(0, 1))
circos.track(ylim = c(0, 1))
circos.info()
set.current.cell("b", 1)
circos.info()
circos.clear()
dev.off()
</code></pre>

<hr>
<h2 id='show.index'>
Label the sector index and the track index on each cell
</h2><span id='topic+show.index'></span>

<h3>Description</h3>

<p>Label the sector index and the track index on each cell
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show.index()
</code></pre>


<h3>Details</h3>

<p>This function is deprecated, please use <code><a href="#topic+circos.info">circos.info</a></code> instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='smartAlign'>
Adjust positions of text
</h2><span id='topic+smartAlign'></span>

<h3>Description</h3>

<p>Adjust positions of text
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smartAlign(x1, x2, xlim)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smartAlign_+3A_x1">x1</code></td>
<td>
<p>Position which corresponds to the top of the text.
</p>
</td></tr>
<tr><td><code id="smartAlign_+3A_x2">x2</code></td>
<td>
<p>Position which corresponds to the bottom of the text.
</p>
</td></tr>
<tr><td><code id="smartAlign_+3A_xlim">xlim</code></td>
<td>
<p>Ranges on x-axis.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>used internally
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='uh'>
Convert units
</h2><span id='topic+uh'></span>

<h3>Description</h3>

<p>Convert units
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uh(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uh_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+convert_length">convert_length</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please do not use this function. Use <code><a href="#topic+mm_h">mm_h</a></code>/<code><a href="#topic+cm_h">cm_h</a></code>/inches_h' instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='ux'>
Convert unit on x direction in data coordinate
</h2><span id='topic+ux'></span>

<h3>Description</h3>

<p>Convert unit on x direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ux(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ux_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+convert_x">convert_x</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please do not use this function. Use <code><a href="#topic+mm_x">mm_x</a></code>/<code><a href="#topic+cm_x">cm_x</a></code>/inches_x' instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

<hr>
<h2 id='uy'>
Convert unit on y direction in data coordinate
</h2><span id='topic+uy'></span>

<h3>Description</h3>

<p>Convert unit on y direction in data coordinate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uy(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uy_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="#topic+convert_y">convert_y</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please do not use this function. Use <code><a href="#topic+mm_y">mm_y</a></code>/<code><a href="#topic+cm_y">cm_y</a></code>/inches_y' instead.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># There is no example
NULL

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
