<!DOCTYPE html><html><head><title>Help for package cit</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cit}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cit-package'>
<p>Causal Inference Test</p></a></li>
<li><a href='#cit.bp'>
<p>Causal Inference Test for a Binary Outcome</p></a></li>
<li><a href='#cit.cp'>
<p>Causal Inference Test for a Continuous Outcome</p></a></li>
<li><a href='#fdr.cit'>
<p>Omnibus FDR Values for CIT</p></a></li>
<li><a href='#fdr.od'>
<p>Permutation-Based FDR and Confidence Interval</p></a></li>
<li><a href='#fdr.q.para'>
<p>Parametric tail-area FDR Values, q-values</p></a></li>
<li><a href='#fdr.q.perm'>
<p>Nonparametric permutation-based tail-area FDR Values, q-values</p></a></li>
<li><a href='#iuq'>
<p>Intersection/Union Q-Value</p></a></li>
<li><a href='#linreg'>
<p>F Test for Linear Model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Causal Inference Test</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-27</td>
</tr>
<tr>
<td>Description:</td>
<td>A likelihood-based hypothesis testing approach is implemented for
  assessing causal mediation. Described in Millstein, Chen, and Breton (2016),
  &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw135">doi:10.1093/bioinformatics/btw135</a>&gt;, it could be used to test for mediation
  of a known causal association between a DNA variant, the 'instrumental variable',
  and a clinical outcome or phenotype by gene expression or DNA methylation, the
  potential mediator. Another example would be testing mediation of the effect
  of a drug on a clinical outcome by the molecular target. The hypothesis test
  generates a p-value or permutation-based FDR value with confidence intervals
  to quantify uncertainty in the causal inference. The outcome can be represented
  by either a continuous or binary variable, the potential mediator is continuous,
  and the instrumental variable can be continuous or binary and is not limited to
  a single variable but may be a design matrix representing multiple variables.</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>gsl (with development libraries libgsl-dev)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tinytest</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/USCbiostats/cit/issues">https://github.com/USCbiostats/cit/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/USCbiostats/cit">https://github.com/USCbiostats/cit</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-28 19:06:04 UTC; mingzhiye</td>
</tr>
<tr>
<td>Author:</td>
<td>Joshua Millstein <a href="https://orcid.org/0000-0001-7961-8943"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Joshua Millstein &lt;joshua.millstein@usc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-28 19:20:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
</table>
<hr>
<h2 id='cit-package'>
Causal Inference Test
</h2><span id='topic+cit-package'></span>

<h3>Description</h3>

<p>This package implements a formal statistical hypothesis test for causal mediation. For example, it could be used to test for mediation of a known causal association between a DNA variant, the 'instrumental variable', and a clinical outcome or phenotype by gene expression or DNA methylation, the potential mediator. Another example would be testing mediation of the effect of a drug on a clinical outcome by the molecular target. The hypothesis test generates a p-value or permutation-based false discovery rate (FDR) value with confidence intervals to quantify uncertainty in the causal inference. The outcome can be represented by either a continuous or binary variable, the potential mediator is continuous, and the instrumental variable can be continuous or binary and is not limited to a single variable but may be a design matrix representing multiple variables.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> cit</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.3.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-06-27</td>
</tr>
<tr>
 <td style="text-align: left;">

License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package implements a novel statistical framework in which existing notions of causal mediation are formalized into a hypothesis test, thus providing a standard quantitative measure of uncertainty in the form of a p-value and optionally, permutation-based FDR. We treat the causal inference as a 'chain' of mathematical conditions that must be satisfied to conclude that the potential mediator is causal for the trait, where the inference is only as good as the weakest link in the chain. P-values are computed for the component conditions. The Intersection-Union Test, in which a series of statistical tests are combined to form an omnibus test, is then employed to generate the overall test result as the maximum of the component p-values. If we let L denote a locus of interest or other instrumental variable, T denote an outcome trait, and G denote a potential mediator, then the four component conditions are, 1) L and G are associated, 2) L and T are associated, 3) L is associated with G|T, and 4) L is independent of T|G. Test 4 requires an equivalence test and is implemented here using a permutation based approach (as described in Millstein et al. 2009).
</p>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>
<p>Maintainer: Joshua Millstein &lt;joshua.millstein@usc.edu&gt;
Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
Millstein J, Zhang B, Zhu J, Schadt EE. 2009. Disentangling molecular relationships with a causal inference test. BMC Genetics, 10:23.
</p>

<hr>
<h2 id='cit.bp'>
Causal Inference Test for a Binary Outcome
</h2><span id='topic+cit.bp'></span>

<h3>Description</h3>

<p>This function implements a formal statistical hypothesis test, resulting in a p-value, to quantify uncertainty in a causal inference pertaining to a measured factor, e.g. a molecular species, which potentially mediates a known causal association between a locus or other instrumental variable and a trait or clinical outcome. If the number of permutations is greater than zero,  then the results can be used with fdr.cit to generate permutation-based FDR values (q-values) that are returned with confidence intervals to quantify uncertainty in the estimate. The outcome is binary, the potential mediator is continuous, and the instrumental variable can be continuous, discrete (such as coding a SNP 0, 1, 2), or binary and is not limited to a single variable but may be a design matrix representing multiple variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cit.bp( L, G, T, C=NULL, maxit=10000, n.perm=0, perm.index=NULL, rseed=NULL )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cit.bp_+3A_l">L</code></td>
<td>
<p>vector or nxp design matrix representing the instrumental variable(s). 
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_g">G</code></td>
<td>
<p>continuous vector representing the potential causal mediator.</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_t">T</code></td>
<td>

<p>Continuous vector representing the clinical trait or outcome of interest.
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_c">C</code></td>
<td>

<p>Vector or nxp design matrix representing adjustment covariates. 
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_maxit">maxit</code></td>
<td>

<p>Maximum number of iterations to be conducted for the conditional independence test, test 4, which is permutation-based. The minimum number of permutations conducted is 1000, regardless of maxit. Increasing maxit will increase the precision of the p-value for test 4 if the p-value is small.
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_n.perm">n.perm</code></td>
<td>

<p>If n.perm is set to an integer greater than 0, then n.perm permutations for each component test will be conducted (randomly permuting the data to generate results under the null). 
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_perm.index">perm.index</code></td>
<td>

<p>This item is only important when the CIT is conducted multiple times, leading to a multiple testing issue, that is addressed by computing FDR using the fdr.cit() function. To accurately account for dependencies among tests when computing FDR confidence intervals, the permutations must be the same for all tests. Thus, if 100 permutations are conducted for 500 CIT test scenarios, each of these 100 permutations are applied to all 500 tests. This is achieved by passing an argument, perm.index, which is an n row by n.perm column dataframe or matrix of permutation indices, where n is the number of observations and n.perm the number of permutations. Each column of perm.index includes a random permutation of 1:n, and this same perm.index object would passed to all 500 CIT tests. fdr.cit() would then be used to compute q-values  (FDR) and q-value confidence intervals for each test.
</p>
</td></tr>
<tr><td><code id="cit.bp_+3A_rseed">rseed</code></td>
<td>

<p>If n.perm &gt; 0, and multiple tests (CITs) are being conducted, setting rseed to the same integer for all tests insures that the permutations will be the same across CITs. This is important for maintaining the observed dependencies among tests for permuted data in order to compute accurate confidence intervals for FDR estimates. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The omnibus p-value, p_cit, is the maximum of the component p-values, an intersection-union test, representing the probability of the data if at least one of the component null hypotheses is true. For component test 4, rather than using the semiparametric approach proposed by Millstein et al. (2009), here it is estimated completely by permutation, resulting in an exact test. If permutations are conducted by setting n.perm to a value greater than zero, then the results are provided in matrix (dataframe) form where each row represents an analysis using a unique permutation, except the first row (perm = 0), which has results from the observed or non-permuted analysis. These results can then be aggregated across multiple cit.bp tests and input to the function fdr.cit to generate component test FDR values (q-values) as well as omnibus q-values with confidence intervals that correspond to the p_cit omnibus p-values.
</p>


<h3>Value</h3>

<p>A dataframe which includes the following columns:
</p>
<table>
<tr><td><code>perm</code></td>
<td>
<p>Indicator for permutation results. Zero indicates that the data were not permuted and subsequent rows include an integer greater than zero for each permutation conducted.</p>
</td></tr>
<tr><td><code>p_cit</code></td>
<td>
<p>CIT (omnibus) p-value</p>
</td></tr>
<tr><td><code>p_TassocL</code></td>
<td>
<p>component p-value for the test of association between T and L.</p>
</td></tr>
<tr><td><code>p_TassocGgvnL</code></td>
<td>
<p>component p-value for the test of association between T and G|L.</p>
</td></tr>
<tr><td><code>p_GassocLgvnT</code></td>
<td>
<p>component p-value for the test of association between G and L|T.</p>
</td></tr>
<tr><td><code>p_LindTgvnG</code></td>
<td>
<p>component p-value for the equivalence test of L ind T|G</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
Millstein J, Zhang B, Zhu J, Schadt EE. 2009. Disentangling molecular relationships with a causal inference test. BMC Genetics, 10:23.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Size
ss = 100

# Errors
e1 = matrix(rnorm(ss),ncol=1)
e2 = matrix(rnorm(ss),ncol=1)

# Simulate genotypes, gene expression, covariates and a clinical trait
L = matrix(rbinom(ss*3,2,.5),ncol=3)
G =  matrix( apply(.4*L, 1, sum) + e1,ncol=1)
T =  matrix(.3*G + e2,ncol=1)
T = ifelse( T &gt; median(T), 1, 0 )
C =  matrix(matrix(rnorm(ss*2),ncol=1),ncol=2)

n.perm = 5
perm.index = matrix(NA, nrow=ss, ncol=n.perm )
for( j in 1:ncol(perm.index) ) perm.index[, j] = sample( 1:ss )

results = cit.bp(L, G, T, perm.index=perm.index, n.perm=n.perm)
results

results = cit.bp(L, G, T)
results

results = cit.bp(L, G, T, C, n.perm=5)
results

results = cit.bp(L, G, T, C)
results
</code></pre>

<hr>
<h2 id='cit.cp'>
Causal Inference Test for a Continuous Outcome
</h2><span id='topic+cit.cp'></span>

<h3>Description</h3>

<p>This function implements a formal statistical hypothesis test, resulting in a p-value, to quantify uncertainty in a causal inference pertaining to a measured factor, e.g. a molecular species, which potentially mediates a known causal association between a locus or other instrumental variable and a quantitative trait. If the number of permutations is greater than zero,  then the results can be used with fdr.cit to generate permutation-based FDR values (q-values) that are returned with confidence intervals to quantify uncertainty in the estimate. The outcome is continuous, the potential mediator is continuous, and the instrumental variable can be continuous, discrete (such as coding a SNP 0, 1, 2), or binary and is not limited to a single variable but may be a design matrix representing multiple variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   cit.cp( L, G, T, C=NULL, n.resampl=50, n.perm=0, perm.index=NULL, rseed=NULL )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cit.cp_+3A_l">L</code></td>
<td>

<p>Vector or nxp design matrix representing the instrumental variable(s). 
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_g">G</code></td>
<td>

<p>Continuous vector representing the potential causal mediator.
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_t">T</code></td>
<td>

<p>Continuous vector representing the clinical trait or outcome of interest.
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_c">C</code></td>
<td>

<p>Vector or nxp design matrix representing adjustment covariates. 
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_n.resampl">n.resampl</code></td>
<td>

<p>The number of instances of the test statistic for conditional independence (test 4) generated by permutation (Millstein et al. 2009) under the null hypothesis of no mediation (independent effects of L on G and T). These data are used to estimate the parameters of the null distribution. The default is set to 50, which we have found to provide reasonable precision.
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_n.perm">n.perm</code></td>
<td>

<p>If n.perm is set to an integer greater than 0, then n.perm permutations for each component test will be conducted (randomly permuting the data to generate results under the null). 
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_perm.index">perm.index</code></td>
<td>

<p>This item is only important when the CIT is conducted multiple times, leading to a multiple testing issue, that is addressed by computing FDR using the fdr.cit() function. To accurately account for dependencies among tests when computing FDR confidence intervals, the permutations must be the same for all tests. Thus, if 100 permutations are conducted for 500 CIT test scenarios, each of these 100 permutations are applied to all 500 tests. This is achieved by passing an argument, perm.index, which is an n row by n.perm column dataframe or matrix of permutation indices, where n is the number of observations and n.perm the number of permutations. Each column of perm.index includes a random permutation of 1:n, and this same perm.index object would passed to all 500 CIT tests. fdr.cit() would then be used to compute q-values  (FDR) and q-value confidence intervals for each test.
</p>
</td></tr>
<tr><td><code id="cit.cp_+3A_rseed">rseed</code></td>
<td>

<p>If n.perm &gt; 0, and multiple tests (CITs) are being conducted, setting rseed to the same integer for all tests insures that the permutations will be the same across CITs. This is important for maintaining the observed dependencies among tests for permuted data in order to compute accurate confidence intervals for FDR estimates. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Increasing n.resampl will increase the precision of the component test 4, the conditional independence test. This may be useful if a very small p-value is observed and high precision is desired, however, it will increase run time. The omnibus p-value, p_cit, is the maximum of the component p-values, an intersection-union test, representing the probability of the data if at least one of the component null hypotheses is true.  If permutations are conducted by setting n.perm to a value greater than zero, then the results are provided in matrix (dataframe) form, where each row represents an analysis using a unique permutation, except the first row (perm = 0), which has results from the observed or non-permuted analysis. These results can then be aggregated across multiple cit.cp tests and input to the function fdr.cit to generate component test FDR values (q-values) as well as omnibus q-values with confidence intervals that correspond to the p_cit omnibus p-values.
</p>


<h3>Value</h3>

<p>A dataframe which includes the following columns:
</p>
<table>
<tr><td><code>perm</code></td>
<td>
<p>Indicator for permutation results. Zero indicates that the data were not permuted and subsequent rows include an integer greater than zero for each permutation conducted.</p>
</td></tr>
<tr><td><code>p_cit</code></td>
<td>
<p>CIT (omnibus) p-value</p>
</td></tr>
<tr><td><code>p_TassocL</code></td>
<td>
<p>component p-value for the test of association between T and L.</p>
</td></tr>
<tr><td><code>p_TassocGgvnL</code></td>
<td>
<p>component p-value for the test of association between T and G|L.</p>
</td></tr>
<tr><td><code>p_GassocLgvnT</code></td>
<td>
<p>component p-value for the test of association between G and L|T.</p>
</td></tr>
<tr><td><code>p_LindTgvnG</code></td>
<td>
<p>component p-value for the equivalence test of L ind T|G</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
Millstein J, Zhang B, Zhu J, Schadt EE. 2009. Disentangling molecular relationships with a causal inference test. BMC Genetics, 10:23.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Size
ss = 100

# Errors
e1 = matrix(rnorm(ss),ncol=1)
e2 = matrix(rnorm(ss),ncol=1)

# Simulate genotypes, gene expression, covariates, and clinical trait matrices
L = matrix(rbinom(ss*3,2,.5),ncol=3)
G =  matrix( apply(.3*L, 1, sum) + e1,ncol=1)
T =  matrix(.3*G + e2,ncol=1)
C =  matrix(matrix(rnorm(ss*2),ncol=1),ncol=2)

n.perm = 5
perm.index = matrix(NA, nrow=ss, ncol=n.perm )
for( j in 1:ncol(perm.index) ) perm.index[, j] = sample( 1:ss )

results = cit.cp(L, G, T)
results

results = cit.cp(L, G, T, perm.index=perm.index, n.perm=5)
results

results = cit.cp(L, G, T, C)
results

results = cit.cp(L, G, T, C, n.perm=5)
results
</code></pre>

<hr>
<h2 id='fdr.cit'>
Omnibus FDR Values for CIT
</h2><span id='topic+fdr.cit'></span>

<h3>Description</h3>

<p>Results including permutation results from cit.cp or cit.bp from multiple tests are used in a permutation-based approach to compute false discovery rates (q-values) for each test. q-values are computed for the omnibus test as well as all component tests. q-values are returned with confidence intervals to quantify uncertainty in the estimate. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   fdr.cit( cit.perm.list, cl=.95, c1=NA )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.cit_+3A_cit.perm.list">cit.perm.list</code></td>
<td>

<p>List, where each item is a dataframe output from cit.cp or cit.bp, which must be run with n.perm &gt; 0. 
</p>
</td></tr>
<tr><td><code id="fdr.cit_+3A_cl">cl</code></td>
<td>

<p>Confidence level for the q-value confidence interval (default is .95).
</p>
</td></tr>
<tr><td><code id="fdr.cit_+3A_c1">c1</code></td>
<td>

<p>Over-dispersion parameter. Setting the over-dispersion parameter to one corresponds to the assumption that all tests (referring to multiple omnibus tests conducted using different sets of variables) are independent. The default (c1=NA) is to use an empirically estimated over-dispersion parameter based on the permutation results.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the initial cit.cp and cit.bp test computations, we suggest setting n.perm to 100 or greater for moderate numbers of tests, say 10s or 100s. Larger numbers of permutations will generate more precise FDR estimates but require substantial computation time. If a large number of tests, say 1000s, have been conducted, then n.perm can be as small as 10 and still generate good FDR estimates and confidence intervals. FDR confidence intervals are especially useful to quantify uncertainty when FDR value &gt; 0.05. The over-dispersion parameter should be set to one if it is known that all tests are independent, however, this is rarely the case in typical multiple testing settings. The fdr.cit function can also be used with simulated data in order to generate power calculations. In this case the over-dispersion parameter may be estimated from pilot data or other datasets and fixed to the prior estimate using the c1 argument. The omnibus CIT FDR value is generated as the maximum FDR across the four component tests, in an intesection-union type of approach analogous to the method used to generate the omnibus CIT p-value in cit.cp and cit.bp.
</p>
<p>It is important to be aware that there are certain conditions under which this permutation-based FDR is not stable. This includes the case where all observed tests achieve the significance level and the case where there are zero positive tests among the permuted results. In the latter case, we take the conservative approach of setting the number of positive permutation tests to one. Both these cases produce an NA for the upper confidence limit. The latter problem may be alleviated by increasing the number of permutations conducted.
</p>


<h3>Value</h3>

<p>A dataframe which includes the following columns:
</p>
<table>
<tr><td><code>p.raw</code></td>
<td>
<p>CIT (omnibus intersection/union test, IUT) p-value, max of component p-values</p>
</td></tr>
<tr><td><code>q.cit</code></td>
<td>
<p>CIT (omnibus FDR intersection/union test, IUT) q-value, permutation-based FDR, function of component test FDR values</p>
</td></tr>
<tr><td><code>q.ll.cit</code></td>
<td>
<p>Lower 95 percent confidence limit (if cl=.95) for q.cit estimate</p>
</td></tr>
<tr><td><code>q.ul.cit</code></td>
<td>
<p>Upper 95 percent confidence limit (if cl=.95) for q.cit estimate</p>
</td></tr>
<tr><td><code>q.TaL</code></td>
<td>
<p>component q-value (FDR) for the test of association between T and L</p>
</td></tr>
<tr><td><code>q.ll.TaL</code></td>
<td>
<p>Lower 95 percent confidence limit (if cl=.95) for q.TaL estimate</p>
</td></tr>
<tr><td><code>q.ul.TaL</code></td>
<td>
<p>Upper 95 percent confidence limit (if cl=.95) for q.TaL estimate</p>
</td></tr>
<tr><td><code>q.TaGgvL</code></td>
<td>
<p>component q-value (FDR) for the test of association between T and G|L</p>
</td></tr>
<tr><td><code>q.ll.TaGgvL</code></td>
<td>
<p>Lower 95 percent confidence limit (if cl=.95) for q.TaGgvL estimate</p>
</td></tr>
<tr><td><code>q.ul.TaGgvL</code></td>
<td>
<p>Upper 95 percent confidence limit (if cl=.95) for q.TaGgvL estimate</p>
</td></tr>
<tr><td><code>q.GaLgvT</code></td>
<td>
<p>component q-value (FDR) for the test of association between G and L|T</p>
</td></tr>
<tr><td><code>q.ll.GaLgvT</code></td>
<td>
<p>Lower 95 percent confidence limit (if cl=.95) for q.GaLgvT estimate</p>
</td></tr>
<tr><td><code>q.ul.GaLgvT</code></td>
<td>
<p>Upper 95 percent confidence limit (if cl=.95) for q.GaLgvT estimate</p>
</td></tr>
<tr><td><code>q.LiTgvG</code></td>
<td>
<p>component q-value (FDR) for the equivalence test of L ind T|G</p>
</td></tr>
<tr><td><code>q.ll.LiTgvG</code></td>
<td>
<p>Lower 95 percent confidence limit (if cl=.95) for the equivalence test of L ind T|G</p>
</td></tr>
<tr><td><code>q.ul.LiTgvG</code></td>
<td>
<p>Upper 95 percent confidence limit (if cl=.95) for the equivalence test of L ind T|G</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
Millstein J, Volfson D. 2013. Computationally efficient permutation-based confidence interval estimation for tail-area FDR. Frontiers in Genetics | Statistical Genetics and Methodology 4(179):1-11.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Size
ss = 100
n.perm = 20
perm.index = matrix(NA, nrow=ss, ncol=n.perm )
for( j in 1:ncol(perm.index) ) perm.index[, j] = sample( 1:ss )

n.tests = 20
myresults = vector('list', n.tests)

for( tst in 1:n.tests ){

	# Errors
	e1 = matrix(rnorm(ss),ncol=1)
	e2 = matrix(rnorm(ss),ncol=1)

	# Simulate genotypes, gene expression, and clinical traits
	L = matrix(rbinom(ss,2,.5),ncol=1)
	G =  matrix(.5*L + e1,ncol=1)
	T =  matrix(.3*G + e2,ncol=1)
	T = ifelse( T &gt; median(T), 1, 0 )

	myresults[[ tst ]] = cit.bp(L, G, T, perm.index=perm.index, n.perm=n.perm)
}
fdr.cit( myresults )

</code></pre>

<hr>
<h2 id='fdr.od'>
Permutation-Based FDR and Confidence Interval
</h2><span id='topic+fdr.od'></span>

<h3>Description</h3>

<p>This function can be used to estimate FDR, corresponding confidence interval, and pi0, the proportion of true null hypotheses, given a selected significance threshold, and results from permuted data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr.od(obsp, permp, pnm, ntests, thres, cl=.95, od=NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.od_+3A_obsp">obsp</code></td>
<td>

<p>observed vector of p-values.
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_permp">permp</code></td>
<td>

<p>list of dataframes that include a column of permutation p-values (or statistics) in each. The length of the list permp = number of permutations.
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_pnm">pnm</code></td>
<td>

<p>name of column in each list component dataframe that includes p-values (or statistics).
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_ntests">ntests</code></td>
<td>

<p>total number of observed tests, which is usually the same as the length of obsp and the number of rows in each permp dataframe. However, this may not be the case if results were filtered by a p-value threshold or statistic threshold. If filtering was conducted then thres must be smaller (more extreme) than the filtering criterion.
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_thres">thres</code></td>
<td>

<p>significance threshold.
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_cl">cl</code></td>
<td>

<p>confidence level for FDR confidence interval (default is .95).
</p>
</td></tr>
<tr><td><code id="fdr.od_+3A_od">od</code></td>
<td>

<p>Over-dispersion parameter. Setting the over-dispersion parameter to one corresponds to the assumption that all tests (referring to multiple omnibus tests conducted using different sets of variables) are independent. The default (c1=NA) is to use an empirically estimated over-dispersion parameter based on the permutation results.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a very large number of tests are conducted, it may be useful to filter results, that is, save only results of those tests that meet some relaxed nominal significance threshold. This alleviates the need to record results for tests that are clearly non-significant. Results from fdr_od() are valid as long as thres &lt; the relaxed nomimal significance threshold for both observed and permuted results. It is not necessary for the input to fdr_od() to be p-values, however, fdr_od() is designed for statistics in which smaller values are more extreme than larger values as is the case for p-values. Therefore, if raw statistics are used, then a transformation may be necessary to insure that smaller values are more likely associated with false null hypotheses than larger values.
</p>
<p>If there are zero positive tests among any of the permuted results (s.perm = 0) at the specified significance threshold (thres), then a conservative approximation is conducted by setting s.perm = 1 for the FDR calculation. This approximation is necessary for the permutation-based FDR to be estimable. This situation can be identified by noting that s.perm = 0 in the output. Increasing the number of permutations conducted may result in positive permutation tests, and thereby overcome the problem. In certain situations, for instance when a large proportion of tests meet the significance threshold, pi0 is estimated to be very small, and thus has a large influence on the FDR estimate. To limit this influence, pi0 is constrained to be .5 or greater, resulting in a more conservative estimate under these conditions. 
</p>


<h3>Value</h3>

<p>A vector which includes:
</p>
<table>
<tr><td><code>FDR</code></td>
<td>
<p>FDR point estimate</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>lower confidence limit</p>
</td></tr>
<tr><td><code>ul</code></td>
<td>
<p>upper confidence limit</p>
</td></tr>
<tr><td><code>pi0</code></td>
<td>
<p>proportion of true null hypotheses</p>
</td></tr>
<tr><td><code>od</code></td>
<td>
<p>overdispersion parameter</p>
</td></tr>
<tr><td><code>s.obs</code></td>
<td>
<p>observed number of positive tests</p>
</td></tr>
<tr><td><code>s.perm</code></td>
<td>
<p>total number of positive tests summed across all permuted result sets</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
Millstein J, Volfson D. 2013. Computationally efficient permutation-based confidence interval estimation for tail-area FDR. Frontiers in Genetics | Statistical Genetics and Methodology 4(179):1-11.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nrow_=100
ncol_=100
X = as.data.frame(matrix(rnorm(nrow_*ncol_),nrow=nrow_,ncol=ncol_))
Y = as.data.frame(matrix(rnorm(nrow_*ncol_),nrow=nrow_,ncol=ncol_))
nperm = 10

myanalysis = function(X,Y){
	ntests = ncol(X)
	rslts = as.data.frame(matrix(NA,nrow=ntests,ncol=2))
	names(rslts) = c("ID","pvalue")
	rslts[,"ID"] = 1:ntests
	for(i in 1:ntests){
		fit = cor.test(X[,i],Y[,i],na.action="na.exclude",
			alternative="two.sided",method="pearson")
		rslts[i,"pvalue"] = fit$p.value
	}
	return(rslts)
} # End myanalysis

# Generate observed results
obs = myanalysis(X,Y)

## Generate permuted results
perml = vector('list',nperm)
for(p_ in 1:nperm){
	X1 = X[order(runif(ncol_)),]
	perml[[p_]] = myanalysis(X1,Y)
}

## FDR results
fdr.od(obs$pvalue,perml,"pvalue",ncol_,.05)

</code></pre>

<hr>
<h2 id='fdr.q.para'>
Parametric tail-area FDR Values, q-values
</h2><span id='topic+fdr.q.para'></span>

<h3>Description</h3>

<p>Given a vector of p-values, this function will use a parametric expression for tail-area FDR to assign q-values, minimum tail-area FDR, for each p-value. This function is used by the function fdr.cit to generate q-values for the forth test, LindTgvnG.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   fdr.q.para( pvals )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.q.para_+3A_pvals">pvals</code></td>
<td>

<p>Vector of p-values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parametric expression is constructed by setting the average number of positive tests among permuted results to p*m, where p is the p-value and m is the number of tests, the length of vector pvals. Following Storey and Tibshirani (2003), the q-value is set to the minimum FDR for p-values greater than or equal to the observed p-value.
</p>


<h3>Value</h3>

<p>A vector of q-values.
</p>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135.
Millstein J, Volfson D. 2013. Computationally efficient permutation-based confidence interval estimation for tail-area FDR. Frontiers in Genetics | Statistical Genetics and Methodology 4(179):1-11.
Storey DJ, Tibshirani R. 2003. Statistical significance for genomewide studies. PNAS 100(16):9440-9445.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Size
ss = 100

n.tests = 20
mypvals = rep(NA, n.tests)

for( tst in 1:n.tests ){

	# Errors
	e1 = matrix(rnorm(ss),ncol=1)
	e2 = matrix(rnorm(ss),ncol=1)

	# Simulate genotypes, gene expression, and clinical traits
	L = matrix(rbinom(ss,2,.5),ncol=1)
	G =  matrix(.5*L + e1,ncol=1)
	T =  matrix(.3*G + e2,ncol=1)
	T = ifelse( T &gt; median(T), 1, 0 )

	mypvals[ tst ] = cit.bp(L, G, T)$p_cit
}
fdr.q.para( mypvals )

</code></pre>

<hr>
<h2 id='fdr.q.perm'>
Nonparametric permutation-based tail-area FDR Values, q-values
</h2><span id='topic+fdr.q.perm'></span>

<h3>Description</h3>

<p>Given a vector of p-values, this function uses fdr.od() to compute tail-area FDR to assign q-values, minimum tail-area FDR, for each p-value. This function is used by the function fdr.cit to generate q-values for the three component tests, TassocL, TassocGgvnL, and GassocLgvnT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   fdr.q.perm( obs.p, perml, pname, ntests, cl=.95, od=NA )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fdr.q.perm_+3A_obs.p">obs.p</code></td>
<td>

<p>observed vector of p-values.
</p>
</td></tr>
<tr><td><code id="fdr.q.perm_+3A_perml">perml</code></td>
<td>

<p>list of dataframes that include a column of permutation p-values (or statistics) in each. The length of the list permp = number of permutations.
</p>
</td></tr>
<tr><td><code id="fdr.q.perm_+3A_pname">pname</code></td>
<td>

<p>name of column in each list component dataframe that includes p-values (or statistics).
</p>
</td></tr>
<tr><td><code id="fdr.q.perm_+3A_ntests">ntests</code></td>
<td>

<p>total number of observed tests, which is usually the same as the length of obsp and the number of rows in each permp dataframe. However, this may not be the case if results were filtered by a p-value threshold or statistic threshold. If filtering was conducted then thres must be smaller (more extreme) than the filtering criterion.
</p>
</td></tr>
<tr><td><code id="fdr.q.perm_+3A_cl">cl</code></td>
<td>

<p>confidence level for FDR confidence interval (default is .95).
</p>
</td></tr>
<tr><td><code id="fdr.q.perm_+3A_od">od</code></td>
<td>

<p>Over-dispersion parameter. Setting the over-dispersion parameter to one corresponds to the assumption that all tests (referring to multiple omnibus tests conducted using different sets of variables) are independent. The default (c1=NA) is to use an empirically estimated over-dispersion parameter based on the permutation results.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Following Storey and Tibshirani (2003), the q-value is set to the minimum FDR for p-values greater than or equal to the observed p-value.
</p>


<h3>Value</h3>

<p>A vector of q-values.
</p>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135.
Millstein J, Volfson D. 2013. Computationally efficient permutation-based confidence interval estimation for tail-area FDR. Frontiers in Genetics | Statistical Genetics and Methodology 4(179):1-11.
Storey DJ, Tibshirani R. 2003. Statistical significance for genomewide studies. PNAS 100(16):9440-9445.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rowno=100
colno=100
X = as.data.frame(matrix(rnorm(rowno*colno),nrow=rowno,ncol=colno))
Y = as.data.frame(matrix(rnorm(rowno*colno),nrow=rowno,ncol=colno))
nperm = 10

myanalysis = function(X,Y){
	ntests = ncol(X)
	rslts = as.data.frame(matrix(NA,nrow=ntests,ncol=2))
	names(rslts) = c("ID","pvalue")
	rslts[,"ID"] = 1:ntests
	for(i in 1:ntests){
		fit = cor.test(X[,i],Y[,i],na.action="na.exclude",
			alternative="two.sided",method="pearson")
		rslts[i,"pvalue"] = fit$p.value
	}
	return(rslts)
} # End myanalysis

# Generate observed results
obs = myanalysis(X,Y)

## Generate permuted results
perml = vector('list',nperm)
for(perm in 1:nperm){
	X1 = X[order(runif(colno)),]
	perml[[perm]] = myanalysis(X1,Y)
}

## FDR results
fdr.q.perm(obs$pvalue,perml,"pvalue",colno)

</code></pre>

<hr>
<h2 id='iuq'>
Intersection/Union Q-Value
</h2><span id='topic+iuq'></span>

<h3>Description</h3>

<p>Tail-area false discovery rates (FDRs) or 'q-values' are combined across multiple component tests to estimate a single q-value that represents the intersection of alternative hypotheses or the union of null hypotheses. In other words, this is an estimate of the rate at which tests called 'significant' at the specified omnibus q-value include at least one component condition that is called 'significant' due to chance alone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iuq( qvec )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iuq_+3A_qvec">qvec</code></td>
<td>
<p>vector of q-values, each corresponding to a component test. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The omnibus q-value is greater than or equal to the maximum of the component q-values.
</p>


<h3>Value</h3>

<p>A single value that represents the rate at which tests called 'significant' at the specified omnibus q-value include at least one component condition that is called 'significant' due to chance alone.
</p>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Chen GK, Breton CV. 2016. cit: hypothesis testing software for mediation analysis in genomic applications. Bioinformatics. btw135. PMID: 27153715.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># component q-values
qvec = c( .1, .04, .02, .11 )

# omnibus q-value
iuq( qvec )
</code></pre>

<hr>
<h2 id='linreg'>
F Test for Linear Model
</h2><span id='topic+linreg'></span>

<h3>Description</h3>

<p>This function is used by cit.cp to compute F test given a continuous outcome and full vs reduced sets of covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linreg( nms.full, nms.redu=NULL, nm.y, mydat )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linreg_+3A_nms.full">nms.full</code></td>
<td>

<p>vector of variable names for all covariates included in the full model.
</p>
</td></tr>
<tr><td><code id="linreg_+3A_nms.redu">nms.redu</code></td>
<td>

<p>vector of variable names for all covariates included in the reduced model. If nms.redu is NULL then the reduced model is fitted with the intercept only.
</p>
</td></tr>
<tr><td><code id="linreg_+3A_nm.y">nm.y</code></td>
<td>

<p>character string, which is the name of the outcome variable.
</p>
</td></tr>
<tr><td><code id="linreg_+3A_mydat">mydat</code></td>
<td>

<p>the dataframe that includes all variables with each variable in a column.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An F test is conducted using the glm function by comparing the full and reduced models. This function is called by cit.cp.
</p>


<h3>Value</h3>

<p>A single p-value is returned.
</p>


<h3>Author(s)</h3>

<p>Joshua Millstein
</p>


<h3>References</h3>

<p>Millstein J, Zhang B, Zhu J, Schadt EE. 2009. Disentangling molecular relationships with a causal inference test. BMC Genetics, 10:23.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ss = 500
cols = 6
nm.y = "y"
nms.full = paste( "x", 1:(cols-1), sep="" )
nms.redu = paste( "x", 1:2, sep="" )

mydat = as.data.frame( matrix( rnorm( ss*cols ), ncol=cols ) )
names( mydat ) = c( nm.y, nms.full )

linreg(nms.full, nms.redu, nm.y, mydat)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
