<!DOCTYPE html><html lang="en"><head><title>Help for package CITMIC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CITMIC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CITMIC'><p>CITMIC</p></a></li>
<li><a href='#CITMIC_Data'><p>An environment variable that includes some example data</p></a></li>
<li><a href='#GetData_CITMIC'><p>GetData_CITMIC</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Cell Infiltration Based on Cell Crosstalk</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic biology tool was developed to identify cell infiltration via an Individualized Cell crosstalk network. 'CITMIC' first constructed a weighted cell crosstalk network by integrating Cell-target interaction information, biological process data from the Gene Ontology (GO) database, and gene transcriptomic data in a specific sample, and then, it used a network propagation algorithm on the network to identify cell infiltration for the sample. Ultimately, cell infiltration in the patient dataset was obtained by normalizing the centrality scores of the cells.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>fastmatch, igraph, parallel, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-08 11:12:22 UTC; 89800</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Xilong Zhao [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-08 12:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CITMIC'>CITMIC</h2><span id='topic+CITMIC'></span>

<h3>Description</h3>

<p>The function &quot;CITMIC&quot; is used to identify cell infiltration in tumor microenvironment by calculating intercellular crosstalk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CITMIC(GEP,weighted = TRUE,base = 10,damping=0.90,cl.cores=1,cell.type=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CITMIC_+3A_gep">GEP</code></td>
<td>
<p>An example gene expression profile.</p>
</td></tr>
<tr><td><code id="CITMIC_+3A_weighted">weighted</code></td>
<td>
<p>This parameter specifies whether to create a weighted graph for the cell crosstalk network. If null, an unweighted graph is created, and the elements of the adjacency matrix indicate the number of edges between vertices. If true, a weighted graph is created(default: TRUE).</p>
</td></tr>
<tr><td><code id="CITMIC_+3A_base">base</code></td>
<td>
<p>Standardized log base of data for improving data distribution(default: 10).</p>
</td></tr>
<tr><td><code id="CITMIC_+3A_damping">damping</code></td>
<td>
<p>Restart the probability of the random-walk algorithm (default: 0.9).</p>
</td></tr>
<tr><td><code id="CITMIC_+3A_cl.cores">cl.cores</code></td>
<td>
<p>The number of CPU cores applied to this task(default:1).</p>
</td></tr>
<tr><td><code id="CITMIC_+3A_cell.type">cell.type</code></td>
<td>
<p>Preset the relevant cell type (e.g. if the solid tumour tissue does not contain 'HSC', it is better to remove it when we preset it.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the cell infiltration score for each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_CITMIC("GEP")
# Run the function
lnScore&lt;-CITMIC(GEP,weighted = TRUE,base = 10,damping=0.90,cl.cores=1,cell.type=NULL)
</code></pre>

<hr>
<h2 id='CITMIC_Data'>An environment variable that includes some example data</h2><span id='topic+CITMIC_Data'></span>

<h3>Description</h3>

<p>matirx_cell_go_inter:A matrix of Jaccard score between cells and GOBP.
matirx_cell_go_jaccard:A matrix consisting of genes shared by cells targets and GOBP.
GEP:An example gene expression profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CITMIC_Data
</code></pre>


<h3>Format</h3>

<p>An environment variable
</p>

<hr>
<h2 id='GetData_CITMIC'>GetData_CITMIC</h2><span id='topic+GetData_CITMIC'></span>

<h3>Description</h3>

<p>Get the example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetData_CITMIC(Data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetData_CITMIC_+3A_data">Data</code></td>
<td>
<p>A character should be one of &quot;GEP&quot;, &quot;matrix_cell_go_inter&quot;, &quot;matrix_cell_go_jaccard&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
