<!DOCTYPE html><html><head><title>Help for package CKAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CKAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CKAT'><p>Composite kernel association test for SNP-set analysis in pharmacogenetics (PGx) studies.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Composite Kernel Association Test for Pharmacogenetics Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Hong Zhang and Judong Shen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hong Zhang &lt;hzhang@wpi.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Composite Kernel Association Test (CKAT) is a flexible and robust kernel machine based approach to jointly test the genetic main effect and gene-treatment interaction effect for a set of single-nucleotide polymorphisms (SNPs) in pharmacogenetics (PGx) assessments embedded within randomized clinical trials.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, CompQuadForm</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-07 01:39:20 UTC; consi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-09 09:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CKAT'>Composite kernel association test for SNP-set analysis in pharmacogenetics (PGx) studies.</h2><span id='topic+CKAT'></span>

<h3>Description</h3>

<p>Composite kernel association test for SNP-set analysis in pharmacogenetics (PGx) studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CKAT(G, Tr, X, y, trait = "continuous", ker = "linear", grids = c(0,
  0.5, 1), n_a = 1000, method = "liu", subdiv = 10^6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CKAT_+3A_g">G</code></td>
<td>
<p>- genotype matrix.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_tr">Tr</code></td>
<td>
<p>- treatment vector, 0 indicates placebo, 1 indicates treatment.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_x">X</code></td>
<td>
<p>- non-genetic covariates data matrix.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_y">y</code></td>
<td>
<p>- response vector. Currently continuous and binary responses are supported. Survival response will be added soon.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_trait">trait</code></td>
<td>
<p>- response indicator. trait = &quot;continuous&quot; or &quot;binary&quot;.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_ker">ker</code></td>
<td>
<p>- kernel. ker = &quot;linear&quot;, &quot;IBS&quot;, &quot;Inter&quot; (interaction kernel) and &quot;RBF&quot; (radial basis function kernel).</p>
</td></tr>
<tr><td><code id="CKAT_+3A_grids">grids</code></td>
<td>
<p>- grids of the candidate weights.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_n_a">n_a</code></td>
<td>
<p>- the number of intervals for manual integration (when integrate function fails). Default n_a = 1000.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_method">method</code></td>
<td>
<p>- method for getting density of A (see details in the reference). Default method is Liu's method.</p>
</td></tr>
<tr><td><code id="CKAT_+3A_subdiv">subdiv</code></td>
<td>
<p>- parameter of Davies' method. Default value is 1E6.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvals - p-values of each individual association test.
</p>
<p>finalp - final p-value of the CKAT test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nsamples = 500; nsnps = 10
X = rnorm(nsamples,0,1)
Tr = sample(0:1,nsamples,replace=TRUE)
G = matrix(rbinom(nsamples*nsnps, 1, 0.05), nrow = nsamples, ncol = nsnps)
GxT = G*Tr
Y0 = 0.5*X + Tr + rnorm(nsamples)
CKAT(G, Tr, X, Y0, grids=c(0,0.5,1))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
