<!DOCTYPE html><html><head><title>Help for package CKLRT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CKLRT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ColSum_C'><p>ColSum_C</p></a></li>
<li><a href='#ColSumtwomatrix_C'><p>ColSumtwomatrix_C</p></a></li>
<li><a href='#doubleloop'><p>doubleloop</p></a></li>
<li><a href='#doubleloop_LRT'><p>doubleloop_LRT</p></a></li>
<li><a href='#Eigen_C'><p>Eigen_C</p></a></li>
<li><a href='#Eigen_C_value'><p>Eigen_C_value</p></a></li>
<li><a href='#Elementwisesquare_C'><p>Elementwisesquare_C</p></a></li>
<li><a href='#getp_au1'><p>Title</p></a></li>
<li><a href='#getp_aud_estimate_pi_first'><p>Title</p></a></li>
<li><a href='#getp_Qreg_dir'><p>Title</p></a></li>
<li><a href='#getp_Qreg_logP'><p>Title</p></a></li>
<li><a href='#getp_Qreg_P'><p>Title</p></a></li>
<li><a href='#ifelsetest_C'><p>ifelsetest_C</p></a></li>
<li><a href='#LR0_fixRho_C'><p>LR0_fixRho_C</p></a></li>
<li><a href='#LR0_fixRho_LRT_C'><p>LR0_fixRho_LRT_C</p></a></li>
<li><a href='#MatMult_C'><p>MatMult_C</p></a></li>
<li><a href='#MatrixPlus_C'><p>MatrixPlus_C</p></a></li>
<li><a href='#MatrixRowMax_C'><p>MatrixRowMax_C</p></a></li>
<li><a href='#NumxMatrix_C'><p>NumxMatrix_C</p></a></li>
<li><a href='#omniLRT_fast'><p>Composite kernel machine regression based likelihood ratio test.</p></a></li>
<li><a href='#omniRLRT_fast'><p>Composite kernel machine regression based restricted likelihood ratio test</p></a></li>
<li><a href='#Sum_C'><p>Sum_C</p></a></li>
<li><a href='#VecMultMat_C'><p>VecMultMat_C</p></a></li>
<li><a href='#Vecplus_C'><p>Vecplus_C</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Composite Kernel Machine Regression Based on Likelihood Ratio
Test</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Ni Zhao [aut], Haoyu Zhang [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Haoyu Zhang &lt;andrew.haoyu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Composite Kernel Machine Regression based on Likelihood Ratio Test (CKLRT): in this package, we develop a kernel machine regression framework to model the overall genetic effect of a SNP-set, considering the possible GE interaction. Specifically, we use a composite kernel to specify the overall genetic effect via a nonparametric function and we model additional covariates parametrically within the regression framework. The composite kernel is constructed as a weighted average of two kernels, one corresponding to the genetic main effect and one corresponding to the GE interaction effect. We propose a likelihood ratio test (LRT) and a restricted likelihood ratio test (RLRT) for statistical significance. We derive a Monte Carlo approach for the finite sample distributions of LRT and RLRT statistics. (N. Zhao, H. Zhang, J. Clark, A. Maity, M. Wu. Composite Kernel Machine Regression based on Likelihood Ratio Test with Application for Combined Genetic and Gene-environment Interaction Effect (Submitted).) </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.17), mgcv, MASS, nlme, compiler, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>RcppEigen, Rcpp</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-13 21:00:37 UTC; haoyuzhang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-09-13 21:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ColSum_C'>ColSum_C</h2><span id='topic+ColSum_C'></span>

<h3>Description</h3>

<p>ColSum_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColSum_C(AA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColSum_C_+3A_aa">AA</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='ColSumtwomatrix_C'>ColSumtwomatrix_C</h2><span id='topic+ColSumtwomatrix_C'></span>

<h3>Description</h3>

<p>ColSumtwomatrix_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColSumtwomatrix_C(AA, BB)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColSumtwomatrix_C_+3A_aa">AA</code></td>
<td>
<p>Matrix</p>
</td></tr>
<tr><td><code id="ColSumtwomatrix_C_+3A_bb">BB</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='doubleloop'>doubleloop</h2><span id='topic+doubleloop'></span>

<h3>Description</h3>

<p>doubleloop
</p>


<h3>Usage</h3>

<pre><code class='language-R'>doubleloop(K1R, K2R, P0R, AR, U1R, wR, LamdasR, nminuspx, all_rho, LR0_allRhoR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doubleloop_+3A_k1r">K1R</code></td>
<td>
<p>K1 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_k2r">K2R</code></td>
<td>
<p>K2 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_p0r">P0R</code></td>
<td>
<p>P0 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_ar">AR</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_u1r">U1R</code></td>
<td>
<p>U1 vector</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_wr">wR</code></td>
<td>
<p>w matrix</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_lamdasr">LamdasR</code></td>
<td>
<p>Lamdas vector</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_nminuspx">nminuspx</code></td>
<td>
<p>n-px</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_all_rho">all_rho</code></td>
<td>
<p>the rho vector</p>
</td></tr>
<tr><td><code id="doubleloop_+3A_lr0_allrhor">LR0_allRhoR</code></td>
<td>
<p>the matrix of likelihood ratio</p>
</td></tr>
</table>

<hr>
<h2 id='doubleloop_LRT'>doubleloop_LRT</h2><span id='topic+doubleloop_LRT'></span>

<h3>Description</h3>

<p>doubleloop_LRT
</p>


<h3>Usage</h3>

<pre><code class='language-R'>doubleloop_LRT(K1R, K2R, P0R, AR, U1R, wR, LamdasR, nminuspx, all_rho,
  LR0_allRhoR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doubleloop_LRT_+3A_k1r">K1R</code></td>
<td>
<p>K1 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_k2r">K2R</code></td>
<td>
<p>K2 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_p0r">P0R</code></td>
<td>
<p>P0 matrix</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_ar">AR</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_u1r">U1R</code></td>
<td>
<p>U1 vector</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_wr">wR</code></td>
<td>
<p>w matrix</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_lamdasr">LamdasR</code></td>
<td>
<p>Lamdas vector</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_nminuspx">nminuspx</code></td>
<td>
<p>n-px</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_all_rho">all_rho</code></td>
<td>
<p>rho vector</p>
</td></tr>
<tr><td><code id="doubleloop_LRT_+3A_lr0_allrhor">LR0_allRhoR</code></td>
<td>
<p>LR0_allRhomatrix</p>
</td></tr>
</table>

<hr>
<h2 id='Eigen_C'>Eigen_C</h2><span id='topic+Eigen_C'></span>

<h3>Description</h3>

<p>Eigen_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Eigen_C(As)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Eigen_C_+3A_as">As</code></td>
<td>
<p>A sysmetric matrix</p>
</td></tr>
</table>

<hr>
<h2 id='Eigen_C_value'>Eigen_C_value</h2><span id='topic+Eigen_C_value'></span>

<h3>Description</h3>

<p>Eigen_C_value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Eigen_C_value(As)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Eigen_C_value_+3A_as">As</code></td>
<td>
<p>A sysmetric matrix</p>
</td></tr>
</table>

<hr>
<h2 id='Elementwisesquare_C'>Elementwisesquare_C</h2><span id='topic+Elementwisesquare_C'></span>

<h3>Description</h3>

<p>Elementwisesquare_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Elementwisesquare_C(AA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Elementwisesquare_C_+3A_aa">AA</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='getp_au1'>Title</h2><span id='topic+getp_au1'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getp_au1(null, LR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getp_au1_+3A_null">null</code></td>
<td>
<p>a vector of likelihood ratio under the null hypothesis. Generated from simulation</p>
</td></tr>
<tr><td><code id="getp_au1_+3A_lr">LR</code></td>
<td>
<p>the likelihood of the data</p>
</td></tr>
</table>

<hr>
<h2 id='getp_aud_estimate_pi_first'>Title</h2><span id='topic+getp_aud_estimate_pi_first'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getp_aud_estimate_pi_first(null, LR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getp_aud_estimate_pi_first_+3A_null">null</code></td>
<td>
<p>a vector of likelihood ratio under the null hypothesis. Generated from simulation</p>
</td></tr>
<tr><td><code id="getp_aud_estimate_pi_first_+3A_lr">LR</code></td>
<td>
<p>the likelihood of the data</p>
</td></tr>
</table>

<hr>
<h2 id='getp_Qreg_dir'>Title</h2><span id='topic+getp_Qreg_dir'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getp_Qreg_dir(null, LR, qmax = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getp_Qreg_dir_+3A_null">null</code></td>
<td>
<p>a vector of likelihood ratio under the null hypothesis. Generated from simulation</p>
</td></tr>
<tr><td><code id="getp_Qreg_dir_+3A_lr">LR</code></td>
<td>
<p>the likelihood of the data</p>
</td></tr>
<tr><td><code id="getp_Qreg_dir_+3A_qmax">qmax</code></td>
<td>
<p>quantile of likelihood ratio vector. Most time put as 1, all the likelihood ratio vector under the null hypothesis will be used</p>
</td></tr>
</table>

<hr>
<h2 id='getp_Qreg_logP'>Title</h2><span id='topic+getp_Qreg_logP'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getp_Qreg_logP(null, LR, qmax = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getp_Qreg_logP_+3A_null">null</code></td>
<td>
<p>a vector of likelihood ratio under the null hypothesis. Generated from simulation</p>
</td></tr>
<tr><td><code id="getp_Qreg_logP_+3A_lr">LR</code></td>
<td>
<p>the likelihood of the data</p>
</td></tr>
<tr><td><code id="getp_Qreg_logP_+3A_qmax">qmax</code></td>
<td>
<p>quantile of likelihood ratio vector. Most time put as 1, all the likelihood ratio vector under the null hypothesis will be used</p>
</td></tr>
</table>

<hr>
<h2 id='getp_Qreg_P'>Title</h2><span id='topic+getp_Qreg_P'></span>

<h3>Description</h3>

<p>Title
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getp_Qreg_P(null, LR, qmax = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getp_Qreg_P_+3A_null">null</code></td>
<td>
<p>a vector of likelihood ratio under the null hypothesis. Generated from simulation</p>
</td></tr>
<tr><td><code id="getp_Qreg_P_+3A_lr">LR</code></td>
<td>
<p>the likelihood of the data</p>
</td></tr>
<tr><td><code id="getp_Qreg_P_+3A_qmax">qmax</code></td>
<td>
<p>quantile of likelihood ratio vector. Most time put as 1, all the likelihood ratio vector under the null hypothesis will be used</p>
</td></tr>
</table>

<hr>
<h2 id='ifelsetest_C'>ifelsetest_C</h2><span id='topic+ifelsetest_C'></span>

<h3>Description</h3>

<p>ifelsetest_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ifelsetest_C(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ifelsetest_C_+3A_x">x</code></td>
<td>
<p>Vector</p>
</td></tr>
</table>

<hr>
<h2 id='LR0_fixRho_C'>LR0_fixRho_C</h2><span id='topic+LR0_fixRho_C'></span>

<h3>Description</h3>

<p>LR0_fixRho_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LR0_fixRho_C(LamdasR, muR, w1R, w2R, nminuspx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LR0_fixRho_C_+3A_lamdasr">LamdasR</code></td>
<td>
<p>Lamda Number</p>
</td></tr>
<tr><td><code id="LR0_fixRho_C_+3A_mur">muR</code></td>
<td>
<p>mu vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_C_+3A_w1r">w1R</code></td>
<td>
<p>w1 vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_C_+3A_w2r">w2R</code></td>
<td>
<p>w2 vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_C_+3A_nminuspx">nminuspx</code></td>
<td>
<p>n-px</p>
</td></tr>
</table>

<hr>
<h2 id='LR0_fixRho_LRT_C'>LR0_fixRho_LRT_C</h2><span id='topic+LR0_fixRho_LRT_C'></span>

<h3>Description</h3>

<p>LR0_fixRho_LRT_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LR0_fixRho_LRT_C(LamdasR, muR, w1R, w2R, nminuspx, xiR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_lamdasr">LamdasR</code></td>
<td>
<p>Lamda Number</p>
</td></tr>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_mur">muR</code></td>
<td>
<p>mu vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_w1r">w1R</code></td>
<td>
<p>w1 vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_w2r">w2R</code></td>
<td>
<p>w2 vector</p>
</td></tr>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_nminuspx">nminuspx</code></td>
<td>
<p>n-px</p>
</td></tr>
<tr><td><code id="LR0_fixRho_LRT_C_+3A_xir">xiR</code></td>
<td>
<p>Vector</p>
</td></tr>
</table>

<hr>
<h2 id='MatMult_C'>MatMult_C</h2><span id='topic+MatMult_C'></span>

<h3>Description</h3>

<p>MatMult_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatMult_C(A, B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatMult_C_+3A_a">A</code></td>
<td>
<p>first matrix</p>
</td></tr>
<tr><td><code id="MatMult_C_+3A_b">B</code></td>
<td>
<p>second matrix</p>
</td></tr>
</table>

<hr>
<h2 id='MatrixPlus_C'>MatrixPlus_C</h2><span id='topic+MatrixPlus_C'></span>

<h3>Description</h3>

<p>MatrixPlus_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatrixPlus_C(A, B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatrixPlus_C_+3A_a">A</code></td>
<td>
<p>First Matrix</p>
</td></tr>
<tr><td><code id="MatrixPlus_C_+3A_b">B</code></td>
<td>
<p>Second Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='MatrixRowMax_C'>MatrixRowMax_C</h2><span id='topic+MatrixRowMax_C'></span>

<h3>Description</h3>

<p>MatrixRowMax_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatrixRowMax_C(AA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatrixRowMax_C_+3A_aa">AA</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='NumxMatrix_C'>NumxMatrix_C</h2><span id='topic+NumxMatrix_C'></span>

<h3>Description</h3>

<p>NumxMatrix_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NumxMatrix_C(A, B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NumxMatrix_C_+3A_a">A</code></td>
<td>
<p>Number</p>
</td></tr>
<tr><td><code id="NumxMatrix_C_+3A_b">B</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='omniLRT_fast'>Composite kernel machine regression based likelihood ratio test.</h2><span id='topic+omniLRT_fast'></span>

<h3>Description</h3>

<p>Composite kernel machine regression based likelihood ratio test. The approximate method for likelihood ratio test tend to be too conservative for small alpha values. We recommend not using it in GWAS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omniLRT_fast(y, X, K1, K2, N = 10000, length.lambda = 200,
  length.rho = 21)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omniLRT_fast_+3A_y">y</code></td>
<td>
<p>: y is the vecgtor of the continous outcomes.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_x">X</code></td>
<td>
<p>: X denotes the additional covariates.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_k1">K1</code></td>
<td>
<p>: K1 is the first kernel corresponding to the genetic main effect.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_k2">K2</code></td>
<td>
<p>: K2 is the second kernel corresponding to the genetic and environment interaction effect.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_n">N</code></td>
<td>
<p>: N is th total number of randomly generated normal variables used to generate the emprical null distribution of LRT. Default value is 10,000.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_length.lambda">length.lambda</code></td>
<td>
<p>: the length of lambda. Dafult value is 200. The values of lambda are all more than 0.</p>
</td></tr>
<tr><td><code id="omniLRT_fast_+3A_length.rho">length.rho</code></td>
<td>
<p>: the length of rho. Default value is 21. The values of rho are between 0 and 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the result is a list containing three elements. 1. p.dir is the p-value of likelihood ratio test based on emprical distrition. 2. p.aud is the p-value by approximating the null distribution as a mixture of a point mass at zero with probability b and weighted chi square distribution with d degrees of freedom with probality of 1-b. 3. LR is the likelihood ratio test statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(6)
n = 50 # the number of observations
X = rnorm(n) # the other covariates
p = 2 # two snp in a gene will be simulated
G = runif(n*p)&lt; 0.5
G = G + runif(n*p) &lt; 0.5
G = matrix(G, n,p) #genetic matrix
E = (runif(n) &lt; 0.5)^2 #enviroment effect
y = rnorm(n) + G[,1] * 0.3 #observations
omniLRT_fast(y, X =  cbind(X, E),K1 = G %*% t(G),K2 = (G*E) %*% t(G * E))
</code></pre>

<hr>
<h2 id='omniRLRT_fast'>Composite kernel machine regression based restricted likelihood ratio test</h2><span id='topic+omniRLRT_fast'></span>

<h3>Description</h3>

<p>Composite kernel machine regression based restricted likelihood ratio test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omniRLRT_fast(y, X, K1, K2, N = 10000, length.rho = 200,
  length.lambda = 21)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omniRLRT_fast_+3A_y">y</code></td>
<td>
<p>vector of the continous outcomes.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_x">X</code></td>
<td>
<p>the additional covariates.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_k1">K1</code></td>
<td>
<p>the first kernel corresponding to the genetic main effect.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_k2">K2</code></td>
<td>
<p>the second kernel corresponding to the genetic and environment interaction effect.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_n">N</code></td>
<td>
<p>total number of randomly generated normal variables used to generate the emprical null distribution of LRT. Default value is 10,000.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_length.rho">length.rho</code></td>
<td>
<p>the length of rho. Default value is 21. The values of rho are between 0 and 1.</p>
</td></tr>
<tr><td><code id="omniRLRT_fast_+3A_length.lambda">length.lambda</code></td>
<td>
<p>the length of lambda. Dafult value is 200. The values of lambda are all more than 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the result is a list containing three elements. 1. p.dir is the p-value of restricted likelihood ratio test based on emprical distrition. 2. p.aud is the p-value by approximating the null distribution as a mixture of a point mass at zero with probability b and weighted chi square distribution with d degrees of freedom with probality of 1-b. 3. LR is the likelihood ratio test statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(6)
n = 50 # the number of observations
X = rnorm(n) # the other covariates
p = 2 # two snp in a gene will be simulated
G = runif(n*p)&lt; 0.5
G = G + runif(n*p) &lt; 0.5
G = matrix(G, n,p) #genetic matrix
E = (runif(n) &lt; 0.5)^2 #enviroment effect
y = rnorm(n) + G[,1] * 0.3 #observations
omniRLRT_fast(y, X =  cbind(X, E),K1 = G %*% t(G),K2 = (G*E) %*% t(G * E))
</code></pre>

<hr>
<h2 id='Sum_C'>Sum_C</h2><span id='topic+Sum_C'></span>

<h3>Description</h3>

<p>Sum_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sum_C(AA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sum_C_+3A_aa">AA</code></td>
<td>
<p>Vector</p>
</td></tr>
</table>

<hr>
<h2 id='VecMultMat_C'>VecMultMat_C</h2><span id='topic+VecMultMat_C'></span>

<h3>Description</h3>

<p>VecMultMat_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VecMultMat_C(A, B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VecMultMat_C_+3A_a">A</code></td>
<td>
<p>Vector</p>
</td></tr>
<tr><td><code id="VecMultMat_C_+3A_b">B</code></td>
<td>
<p>Matrix</p>
</td></tr>
</table>

<hr>
<h2 id='Vecplus_C'>Vecplus_C</h2><span id='topic+Vecplus_C'></span>

<h3>Description</h3>

<p>Vecplus_C
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Vecplus_C(A, B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Vecplus_C_+3A_a">A</code></td>
<td>
<p>Vector</p>
</td></tr>
<tr><td><code id="Vecplus_C_+3A_b">B</code></td>
<td>
<p>Vector</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
