<!DOCTYPE html><html><head><title>Help for package CleanBSequences</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CleanBSequences}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DNAStringSetOPR'><p>Curing of biological sequences</p></a></li>
<li><a href='#DNAStringSetTPR'><p>Curing of biological sequences</p></a></li>
<li><a href='#OnePrimerRemove'><p>Clean biological secuences</p></a></li>
<li><a href='#TwoPrimerRemove'><p>Clean biological secuences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Curing of Biological Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Florencia I. Pozzi, Silvina A. Felitti</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Florencia I. Pozzi &lt;florenciapoz@hotmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Curates biological sequences massively, quickly, without errors and 
 without internet connection.  Biological sequences curing is performed by 
 aligning the forward and / or revers primers or ends of cloning vectors with the sequences 
 to be cleaned. After the alignment, new subsequences are generated without biological fragment 
 not desired by the user.
 Pozzi et al (2020) &lt;<a href="https://doi.org/10.1007%2Fs00438-020-01671-z">doi:10.1007/s00438-020-01671-z</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>Biostrings</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-27 17:08:53 UTC; Florencia</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-27 17:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='DNAStringSetOPR'>Curing of biological sequences</h2><span id='topic+DNAStringSetOPR'></span>

<h3>Description</h3>

<p>Curates biological sequences of primer reverse.This cleaning is required for techniques such as cDNA-AFLP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAStringSetOPR(SEQs, PrimerR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DNAStringSetOPR_+3A_seqs">SEQs</code></td>
<td>
<p>file with fasta format containing biological sequences that are to be cleaned.</p>
</td></tr>
<tr><td><code id="DNAStringSetOPR_+3A_primerr">PrimerR</code></td>
<td>
<p>dnastring containing the reverse primer/vector sequences to be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>clean biological sequences and visualization of the alignments
</p>


<h3>Author(s)</h3>

<p>Florencia I Pozzi, Silvina A. Felitti
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SEQs = readDNAStringSet(system.file("sequences","SeqInputOPR.fasta", package = "CleanBSequences"))
PrimerR= DNAString ("GACTGCGTACCATGC")
DNAStringSetOPR (SEQs,PrimerR)
</code></pre>

<hr>
<h2 id='DNAStringSetTPR'>Curing of biological sequences</h2><span id='topic+DNAStringSetTPR'></span>

<h3>Description</h3>

<p>Curates biological sequences of two restriction enzyme primers or cloning vectors.This cleaning is required for techniques such as cDNA-AFLP.This cleaning is required for techniques such as cDNA-AFLP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAStringSetTPR(SEQs, PrimerF, PrimerR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DNAStringSetTPR_+3A_seqs">SEQs</code></td>
<td>
<p>file with fasta format containing biological sequences that are to be cleaned.</p>
</td></tr>
<tr><td><code id="DNAStringSetTPR_+3A_primerf">PrimerF</code></td>
<td>
<p>dnastring containing the foward primer/vector sequences to be removed.</p>
</td></tr>
<tr><td><code id="DNAStringSetTPR_+3A_primerr">PrimerR</code></td>
<td>
<p>dnastring containing the reverse primer/vector sequences to be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>clean biological sequences and visualization of the alignments
</p>


<h3>Author(s)</h3>

<p>Florencia I Pozzi, Silvina A. Felitti
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SEQs = readDNAStringSet(system.file("sequences","SeqInputTPR.fasta", package = "CleanBSequences"))
PrimerR= DNAString ("GACTGCGTACCATGC")
PrimerF = DNAString("GATGAGTCCTGACCGAA")
DNAStringSetTPR (SEQs,PrimerF,PrimerR)
</code></pre>

<hr>
<h2 id='OnePrimerRemove'>Clean biological secuences</h2><span id='topic+OnePrimerRemove'></span>

<h3>Description</h3>

<p>Curates biological sequences of primer reverse.This cleaning is required for techniques such as cDNA-AFLP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OnePrimerRemove(SEQs, PrimerR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OnePrimerRemove_+3A_seqs">SEQs</code></td>
<td>
<p>dnastring or file with fasta format containing biological sequences that are to be cleaned.</p>
</td></tr>
<tr><td><code id="OnePrimerRemove_+3A_primerr">PrimerR</code></td>
<td>
<p>dnastring or file with fasta format containing the reverse primer/vector sequences to be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>clean biological sequences and visualization of the alignments
</p>


<h3>Author(s)</h3>

<p>Florencia I Pozzi and Silvina A. Felitti
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SEQs = DNAString(paste("GCCTCGCCTCCCTCTTTGATCAGCTTCGCATATCAGGCAACAGCTCAATTT",
"GGTACTTGTTCAAATAAGCATTTAGACCATCTGTTCCAAGAACCTTTGCAATCTT",
"CACAAGGTGGTCATGGTACGCAGTC", sep=""))
PrimerR= DNAString("GACTGCGTACCATGC")
OnePrimerRemove (SEQs,PrimerR)
</code></pre>

<hr>
<h2 id='TwoPrimerRemove'>Clean biological secuences</h2><span id='topic+TwoPrimerRemove'></span>

<h3>Description</h3>

<p>Curates biological sequences of two restriction enzyme primers or cloning vectors.This cleaning is required for techniques such as cDNA-AFLP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TwoPrimerRemove(SEQs, PrimerF, PrimerR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TwoPrimerRemove_+3A_seqs">SEQs</code></td>
<td>
<p>DNAString containing biological sequences that are to be cleaned.</p>
</td></tr>
<tr><td><code id="TwoPrimerRemove_+3A_primerf">PrimerF</code></td>
<td>
<p>dnastring containing the foward primer/vector sequences to be removed.</p>
</td></tr>
<tr><td><code id="TwoPrimerRemove_+3A_primerr">PrimerR</code></td>
<td>
<p>dnastring containing the reverse primer/vector sequences to be removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>clean biological sequences and visualization of the alignments
</p>


<h3>Author(s)</h3>

<p>Florencia I Pozzi, Silvina A. Felitti
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SEQs = DNAString(paste("ACTTTCTGCTGCTTGTGGTCGCAATCAGAGTCCTGATGATGAGTCCTGA",
"CCGAACCCTTTTTCTCCGTCATCCGTTGGTCCATGGTACGCAATCAGAG", sep = ""))
PrimerF = DNAString("GATGAGTCCTGACCGAA")
PrimerR = DNAString("GACTGCGTACCATGC")
TwoPrimerRemove (SEQs,PrimerF,PrimerR)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
