<!DOCTYPE html><html><head><title>Help for package clinspacy</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {clinspacy}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#bind_clinspacy'><p>This function binds columns containing either the lemma of the entity or the</p>
UMLS concept unique identifier (CUI) with frequencies to a data frame. The
resulting data frame can be used to train a machine learning model or for
additional feature selection.</a></li>
<li><a href='#bind_clinspacy_embeddings'><p>This function binds columns containing entity or concept embeddings to a data</p>
frame. The entity embeddings are derived from the scispacy package, and the
concept embeddings are derived from the
<code>dataset_cui2vec_embeddings</code> dataset included with this package.</a></li>
<li><a href='#clinspacy'><p>This is the primary function for processing both data frames and character</p>
vectors in the <code>clinspacy</code> package.</a></li>
<li><a href='#clinspacy_init'><p>Initializes clinspacy. This function is optional to run but gives you more</p>
control over the parameters used by scispacy at initiation. If you do not run
this function, it will be run with default parameters the first time that any
of the package functions are run.</a></li>
<li><a href='#dataset_cui2vec_definitions'><p>Cui2vec concept definitions</p></a></li>
<li><a href='#dataset_cui2vec_embeddings'><p>Cui2vec concept embeddings</p></a></li>
<li><a href='#dataset_mtsamples'><p>Medical transcription samples.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Clinical Natural Language Processing using 'spaCy', 'scispaCy',
and 'medspaCy'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs biomedical named entity recognition,
    Unified Medical Language System (UMLS) concept mapping, and negation
    detection using the Python 'spaCy', 'scispaCy', and 'medspaCy' packages, and 
    transforms extracted data into a wide format for inclusion in machine
    learning models. The development of the 'scispaCy' package is described by
    Neumann (2019) &lt;<a href="https://doi.org/10.18653%2Fv1%2FW19-5034">doi:10.18653/v1/W19-5034</a>&gt;. The 'medspacy' package uses
    'ConText', an algorithm for determining the context of clinical statements
    described by Harkema (2009) &lt;<a href="https://doi.org/10.1016%2Fj.jbi.2009.05.002">doi:10.1016/j.jbi.2009.05.002</a>&gt;. Clinspacy
    also supports entity embeddings from 'scispaCy' and UMLS 'cui2vec' concept
    embeddings developed by Beam (2018) &lt;<a href="https://arxiv.org/abs/1804.01486">arXiv:1804.01486</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>reticulate (&ge; 1.16), data.table, assertthat, rappdirs, utils,
magrittr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ML4LHS/clinspacy">https://github.com/ML4LHS/clinspacy</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ML4LHS/clinspacy/issues">https://github.com/ML4LHS/clinspacy/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-18 16:42:43 UTC; kdpsingh</td>
</tr>
<tr>
<td>Author:</td>
<td>Karandeep Singh [aut, cre],
  Benjamin Kompa [aut],
  Andrew Beam [aut],
  Allen Schmaltz [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karandeep Singh &lt;kdpsingh@umich.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-20 10:50:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns rhs(lhs).
</p>

<hr>
<h2 id='bind_clinspacy'>This function binds columns containing either the lemma of the entity or the
UMLS concept unique identifier (CUI) with frequencies to a data frame. The
resulting data frame can be used to train a machine learning model or for
additional feature selection.</h2><span id='topic+bind_clinspacy'></span>

<h3>Description</h3>

<p>This function binds columns containing either the lemma of the entity or the
UMLS concept unique identifier (CUI) with frequencies to a data frame. The
resulting data frame can be used to train a machine learning model or for
additional feature selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind_clinspacy(
  clinspacy_output,
  df,
  cs_col = NULL,
  df_id = NULL,
  subset = "is_negated == FALSE"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind_clinspacy_+3A_clinspacy_output">clinspacy_output</code></td>
<td>
<p>A data.frame or file name containing the output from
<code><a href="#topic+clinspacy">clinspacy</a></code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_+3A_df">df</code></td>
<td>
<p>The data.frame to which you would like to bind the output of
<code><a href="#topic+clinspacy">clinspacy</a></code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_+3A_cs_col">cs_col</code></td>
<td>
<p>Name of the column in the <code>clinspacy_output</code> that you
would like to pivot. For example: <code>"entity"</code>, <code>"lemma"</code>,
<code>"cui"</code>, or <code>"definition"</code>. Defaults to <code>"lemma"</code> if
<code>use_linker</code> is set to <code>FALSE</code> and <code>"cui"</code> if
<code>use_linker</code> is set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_+3A_df_id">df_id</code></td>
<td>
<p>The name of the <code>id</code> column in the data frame with which
the <code>clinspacy_id</code> column in <code>clinspacy_output</code> will be joined.
If you supplied a <code>df_id</code> in <code><a href="#topic+clinspacy">clinspacy</a></code>, then you must
also supply it here. If you did not supply it in <code><a href="#topic+clinspacy">clinspacy</a></code>,
then it will default to the row number (similar behavior to in
<code><a href="#topic+clinspacy">clinspacy</a></code>).</p>
</td></tr>
<tr><td><code id="bind_clinspacy_+3A_subset">subset</code></td>
<td>
<p>Logical criteria represented as a string by which the
<code>clinspacy_output</code> will be subsetted prior to building the output data
frame. Defaults to <code>"is_negated == FALSE"</code>, which removes negated
concepts prior to generating the output. Any column in
<code>clinspacy_output</code> may be referenced here. To avoid any subsetting,
set this to <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the original data frame as well as additional
column names for each lemma or UMLS concept unique identifer found with
values containing frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mtsamples &lt;- dataset_mtsamples()
mtsamples[1:5,] %&gt;%
  clinspacy(df_col = 'description') %&gt;%
  bind_clinspacy(mtsamples[1:5,])

## End(Not run)
</code></pre>

<hr>
<h2 id='bind_clinspacy_embeddings'>This function binds columns containing entity or concept embeddings to a data
frame. The entity embeddings are derived from the scispacy package, and the
concept embeddings are derived from the
<code><a href="#topic+dataset_cui2vec_embeddings">dataset_cui2vec_embeddings</a></code> dataset included with this package.</h2><span id='topic+bind_clinspacy_embeddings'></span>

<h3>Description</h3>

<p>The embeddings are derived from Andrew Beam's
<a href="https://github.com/beamandrew/cui2vec">cui2vec R package</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind_clinspacy_embeddings(
  clinspacy_output,
  df,
  type = "scispacy",
  df_id = NULL,
  subset = "is_negated == FALSE"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bind_clinspacy_embeddings_+3A_clinspacy_output">clinspacy_output</code></td>
<td>
<p>A data.frame or file name containing the output from
<code><a href="#topic+clinspacy">clinspacy</a></code>. In order for scispacy embeddings to be available
to <code><a href="#topic+bind_clinspacy_embeddings">bind_clinspacy_embeddings</a></code>, you must set
<code>return_scispacy_embeddings</code> to <code>TRUE</code> when running
<code><a href="#topic+clinspacy">clinspacy</a></code> so that the embeddings are included within
<code>clinspacy_output</code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_embeddings_+3A_df">df</code></td>
<td>
<p>The data.frame to which you would like to bind the output of
<code><a href="#topic+clinspacy">clinspacy</a></code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_embeddings_+3A_type">type</code></td>
<td>
<p>The type of embeddings to return. One of <code>scispacy</code> and
<code>cui2vec</code>. Whereas <code>cui2vec</code> embeddings require the UMLS linker
to be enabled, the <code>scispacy</code> embeddings do not. Defaults to
<code>scispacy</code>.</p>
</td></tr>
<tr><td><code id="bind_clinspacy_embeddings_+3A_df_id">df_id</code></td>
<td>
<p>The name of the <code>id</code> column in the data frame with which
the <code>id</code> column in <code>clinspacy_output</code> will be joined. If you
supplied a <code>df_id</code> in <code><a href="#topic+clinspacy">clinspacy</a></code>, then you must also
supply it here. If you did not supply it in <code><a href="#topic+clinspacy">clinspacy</a></code>, then
it will default to the row number (similar behavior to in
<code><a href="#topic+clinspacy">clinspacy</a></code>).</p>
</td></tr>
<tr><td><code id="bind_clinspacy_embeddings_+3A_subset">subset</code></td>
<td>
<p>Logical criteria represented as a string by which the
<code>clinspacy_output</code> will be subsetted prior to building the output data
frame. Defaults to <code>"is_negated == FALSE"</code>, which removes negated
concepts prior to generating the output. Any column in
<code>clinspacy_output</code> may be referenced here. To avoid any subsetting,
set this to <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Citation
</p>
<p>Beam, A.L., Kompa, B., Schmaltz, A., Fried, I., Griffin, W, Palmer, N.P.,
Shi, X., Cai, T., and Kohane, I.S.,, 2019. Clinical Concept Embeddings
Learned from Massive Sources of Multimodal Medical Data. arXiv preprint
arXiv:1804.01486.
</p>
<p>License
</p>
<p>The cui2vec data is made available under a
<a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0 license</a>. The
only change made to the original dataset is the renaming of columns.
</p>


<h3>Value</h3>

<p>A data frame containing the original data frame as well as the
concept embeddings. For scispacy embeddings, this returns 200 columns of
embeddings. For cui2vec embeddings, this returns 500 columns of embedings.
The resulting data frame can be used to train a machine learning model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mtsamples &lt;- dataset_mtsamples()
mtsamples[1:5,] %&gt;%
  clinspacy(df_col = 'description', return_scispacy_embeddings = TRUE) %&gt;%
  bind_clinspacy_embeddings(mtsamples[1:5,])

## End(Not run)

</code></pre>

<hr>
<h2 id='clinspacy'>This is the primary function for processing both data frames and character
vectors in the <code>clinspacy</code> package.</h2><span id='topic+clinspacy'></span>

<h3>Description</h3>

<p>This is the primary function for processing both data frames and character
vectors in the <code>clinspacy</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clinspacy(
  x,
  df_col = NULL,
  df_id = NULL,
  threshold = 0.99,
  semantic_types = c(NA, "Acquired Abnormality", "Activity", "Age Group",
    "Amino Acid Sequence", "Amino Acid, Peptide, or Protein", "Amphibian",
    "Anatomical Abnormality", "Anatomical Structure", "Animal", "Antibiotic", "Archaeon",
    "Bacterium", "Behavior", "Biologic Function", "Biologically Active Substance",
    "Biomedical Occupation or Discipline", "Biomedical or Dental Material", "Bird",
    "Body Location or Region", "Body Part, Organ, or Organ Component",
    "Body Space or Junction", "Body Substance", "Body System", "Carbohydrate Sequence",  
       "Cell", "Cell Component", "Cell Function", "Cell or Molecular Dysfunction",
    "Chemical", "Chemical Viewed Functionally", "Chemical Viewed Structurally",
    "Classification", "Clinical Attribute", "Clinical Drug", "Conceptual Entity",
    "Congenital Abnormality", "Daily or Recreational Activity", "Diagnostic Procedure",
    "Disease or Syndrome", "Drug Delivery Device", "Educational Activity",
    "Element, Ion, or Isotope", "Embryonic Structure", "Entity",
    "Environmental Effect of Humans", "Enzyme", "Eukaryote",      "Event",
    "Experimental Model of Disease", "Family Group", "Finding", "Fish", "Food",
    "Fully Formed Anatomical Structure", "Functional Concept", "Fungus",
    "Gene or Genome", "Genetic Function", "Geographic Area",
    "Governmental or Regulatory Activity", "Group", "Group Attribute",
    "Hazardous or Poisonous Substance", "Health Care Activity",
    "Health Care Related Organization", "Hormone", "Human",
    "Human-caused Phenomenon or Process", "Idea or Concept", "Immunologic Factor",
    "Indicator, Reagent, or Diagnostic Aid",      "Individual Behavior",
    "Injury or Poisoning", "Inorganic Chemical", "Intellectual Product",
    "Laboratory or Test Result", "Laboratory Procedure", "Language", "Machine Activity",
    "Mammal", "Manufactured Object", "Medical Device",
    "Mental or Behavioral Dysfunction", "Mental Process",
    "Molecular Biology Research Technique", "Molecular Function", "Molecular Sequence",
    "Natural Phenomenon or Process", "Neoplastic Process",
    "Nucleic Acid, Nucleoside, or Nucleotide", "Nucleotide Sequence",
    "Occupation or Discipline",      "Occupational Activity", "Organ or Tissue Function",
    "Organic Chemical", "Organism", "Organism Attribute", "Organism Function",
    "Organization", "Pathologic Function", "Patient or Disabled Group",
    "Pharmacologic Substance", "Phenomenon or Process", "Physical Object",
    "Physiologic Function", "Plant", "Population Group",
    "Professional or Occupational Group", "Professional Society", "Qualitative Concept",
    "Quantitative Concept", "Receptor", "Regulation or Law", "Reptile",
    "Research Activity", "Research Device",      "Self-help or Relief Organization",
    "Sign or Symptom", "Social Behavior", "Spatial Concept", "Substance",
    "Temporal Concept", "Therapeutic or Preventive Procedure", "Tissue", "Vertebrate",
    "Virus", "Vitamin"),
  return_scispacy_embeddings = FALSE,
  verbose = TRUE,
  output_file = NULL,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clinspacy_+3A_x">x</code></td>
<td>
<p>Either a data.frame or a character vector</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_df_col">df_col</code></td>
<td>
<p>If <code>x</code> is a data.frame then you must specify the name of
the column containing text as a string.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_df_id">df_id</code></td>
<td>
<p>If <code>x</code> is a data.frame then you may *optionally* specify an
<code>id</code> column to help match up each row of text in the original data
frame with the resulting output. If you do not specify an id, the resulting
will contain the row number from the original data.frame.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_threshold">threshold</code></td>
<td>
<p>Defaults to 0.99. The confidence threshold value used by
clinspacy (can be higher than the <code>linker_threshold</code> from
<code><a href="#topic+clinspacy_init">clinspacy_init</a></code>). Note that whereas the linker_threshold can
only be set once per session, this threshold can be updated during the R
session.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_semantic_types">semantic_types</code></td>
<td>
<p>Character vector containing any combination of the
following: c(NA, &quot;Acquired Abnormality&quot;, &quot;Activity&quot;, &quot;Age Group&quot;, &quot;Amino
Acid Sequence&quot;, &quot;Amino Acid, Peptide, or Protein&quot;, &quot;Amphibian&quot;, &quot;Anatomical
Abnormality&quot;, &quot;Anatomical Structure&quot;, &quot;Animal&quot;, &quot;Antibiotic&quot;, &quot;Archaeon&quot;,
&quot;Bacterium&quot;, &quot;Behavior&quot;, &quot;Biologic Function&quot;, &quot;Biologically Active
Substance&quot;, &quot;Biomedical Occupation or Discipline&quot;, &quot;Biomedical or Dental
Material&quot;, &quot;Bird&quot;, &quot;Body Location or Region&quot;, &quot;Body Part, Organ, or Organ
Component&quot;, &quot;Body Space or Junction&quot;, &quot;Body Substance&quot;, &quot;Body System&quot;,
&quot;Carbohydrate Sequence&quot;, &quot;Cell&quot;, &quot;Cell Component&quot;, &quot;Cell Function&quot;, &quot;Cell
or Molecular Dysfunction&quot;, &quot;Chemical&quot;, &quot;Chemical Viewed Functionally&quot;,
&quot;Chemical Viewed Structurally&quot;, &quot;Classification&quot;, &quot;Clinical Attribute&quot;,
&quot;Clinical Drug&quot;, &quot;Conceptual Entity&quot;, &quot;Congenital Abnormality&quot;, &quot;Daily or
Recreational Activity&quot;, &quot;Diagnostic Procedure&quot;, &quot;Disease or Syndrome&quot;,
&quot;Drug Delivery Device&quot;, &quot;Educational Activity&quot;, &quot;Element, Ion, or Isotope&quot;,
&quot;Embryonic Structure&quot;, &quot;Entity&quot;, &quot;Environmental Effect of Humans&quot;,
&quot;Enzyme&quot;, &quot;Eukaryote&quot;, &quot;Event&quot;, &quot;Experimental Model of Disease&quot;, &quot;Family
Group&quot;, &quot;Finding&quot;, &quot;Fish&quot;, &quot;Food&quot;, &quot;Fully Formed Anatomical Structure&quot;,
&quot;Functional Concept&quot;, &quot;Fungus&quot;, &quot;Gene or Genome&quot;, &quot;Genetic Function&quot;,
&quot;Geographic Area&quot;, &quot;Governmental or Regulatory Activity&quot;, &quot;Group&quot;, &quot;Group
Attribute&quot;, &quot;Hazardous or Poisonous Substance&quot;, &quot;Health Care Activity&quot;,
&quot;Health Care Related Organization&quot;, &quot;Hormone&quot;, &quot;Human&quot;, &quot;Human-caused
Phenomenon or Process&quot;, &quot;Idea or Concept&quot;, &quot;Immunologic Factor&quot;,
&quot;Indicator, Reagent, or Diagnostic Aid&quot;, &quot;Individual Behavior&quot;, &quot;Injury or
Poisoning&quot;, &quot;Inorganic Chemical&quot;, &quot;Intellectual Product&quot;, &quot;Laboratory or
Test Result&quot;, &quot;Laboratory Procedure&quot;, &quot;Language&quot;, &quot;Machine Activity&quot;,
&quot;Mammal&quot;, &quot;Manufactured Object&quot;, &quot;Medical Device&quot;, &quot;Mental or Behavioral
Dysfunction&quot;, &quot;Mental Process&quot;, &quot;Molecular Biology Research Technique&quot;,
&quot;Molecular Function&quot;, &quot;Molecular Sequence&quot;, &quot;Natural Phenomenon or
Process&quot;, &quot;Neoplastic Process&quot;, &quot;Nucleic Acid, Nucleoside, or Nucleotide&quot;,
&quot;Nucleotide Sequence&quot;, &quot;Occupation or Discipline&quot;, &quot;Occupational Activity&quot;,
&quot;Organ or Tissue Function&quot;, &quot;Organic Chemical&quot;, &quot;Organism&quot;, &quot;Organism
Attribute&quot;, &quot;Organism Function&quot;, &quot;Organization&quot;, &quot;Pathologic Function&quot;,
&quot;Patient or Disabled Group&quot;, &quot;Pharmacologic Substance&quot;, &quot;Phenomenon or
Process&quot;, &quot;Physical Object&quot;, &quot;Physiologic Function&quot;, &quot;Plant&quot;, &quot;Population
Group&quot;, &quot;Professional or Occupational Group&quot;, &quot;Professional Society&quot;,
&quot;Qualitative Concept&quot;, &quot;Quantitative Concept&quot;, &quot;Receptor&quot;, &quot;Regulation or
Law&quot;, &quot;Reptile&quot;, &quot;Research Activity&quot;, &quot;Research Device&quot;, &quot;Self-help or
Relief Organization&quot;, &quot;Sign or Symptom&quot;, &quot;Social Behavior&quot;, &quot;Spatial
Concept&quot;, &quot;Substance&quot;, &quot;Temporal Concept&quot;, &quot;Therapeutic or Preventive
Procedure&quot;, &quot;Tissue&quot;, &quot;Vertebrate&quot;, &quot;Virus&quot;, &quot;Vitamin&quot;)</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_return_scispacy_embeddings">return_scispacy_embeddings</code></td>
<td>
<p>Defaults to <code>FALSE</code>. This is primarily
intended for use by the <code><a href="#topic+bind_clinspacy_embeddings">bind_clinspacy_embeddings</a></code> function to
obtain scispacy embeddings. In order for scispacy embeddings to be
available to <code><a href="#topic+bind_clinspacy_embeddings">bind_clinspacy_embeddings</a></code>, you must set this to
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_verbose">verbose</code></td>
<td>
<p>Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_output_file">output_file</code></td>
<td>
<p>Defaults to <code>NULL</code>. This is an optional argument that
writes the output to a comma-separated value (CSV) file.</p>
</td></tr>
<tr><td><code id="clinspacy_+3A_overwrite">overwrite</code></td>
<td>
<p>Defaults to <code>FALSE</code>. If <code>output_file</code> already
exists and <code>overwrite</code> is set to <code>FALSE</code>, then you will be
prompted to confirm whether you would like to overwrite the file. If set to
<code>TRUE</code>, then <code>output_file</code> will automatically be overwritten.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>output_file</code> is <code>NULL</code> (the default), then this
function returns a data frame containing the UMLS concept unique
identifiers (cui), entities, lemmatized entities, CyContext negation status
(<code>TRUE</code> means negated, <code>FALSE</code> means *not* negated), other
CyContext contexts, and section title from the clinical sectionizer. If
<code>output_file</code> points to a file name, then the name of the created file
will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
clinspacy('This patient has diabetes and CKD stage 3 but no HTN.')

clinspacy(c('This pt has CKD and HTN', 'Pt only has CKD but no HTN'))

data.frame(text = c('This pt has CKD and HTN', 'Diabetes is present'),
           stringsAsFactors = FALSE) %&gt;%
  clinspacy(df_col = 'text')

if (!dir.exists(rappdirs::user_data_dir('clinspacy'))) {
  dir.create(rappdirs::user_data_dir('clinspacy'), recursive = TRUE)
  }

clinspacy(c('This pt has CKD and HTN', 'Has CKD but no HTN'),
  output_file = file.path(rappdirs::user_data_dir('clinspacy'),
                          'output.csv'),
  overwrite = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='clinspacy_init'>Initializes clinspacy. This function is optional to run but gives you more
control over the parameters used by scispacy at initiation. If you do not run
this function, it will be run with default parameters the first time that any
of the package functions are run.</h2><span id='topic+clinspacy_init'></span>

<h3>Description</h3>

<p>Initializes clinspacy. This function is optional to run but gives you more
control over the parameters used by scispacy at initiation. If you do not run
this function, it will be run with default parameters the first time that any
of the package functions are run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clinspacy_init(
  miniconda = TRUE,
  use_linker = FALSE,
  linker_threshold = 0.99,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clinspacy_init_+3A_miniconda">miniconda</code></td>
<td>
<p>Defaults to TRUE, which results in miniconda being installed
(~400 MB) and configured with the &quot;clinspacy&quot; conda environment. If you
want to override this behavior, set <code>miniconda</code> to <code>FALSE</code> and
specify an alternative environment using use_python() or use_conda().</p>
</td></tr>
<tr><td><code id="clinspacy_init_+3A_use_linker">use_linker</code></td>
<td>
<p>Defaults to <code>FALSE</code>. To turn on the UMLS linker, set
this to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="clinspacy_init_+3A_linker_threshold">linker_threshold</code></td>
<td>
<p>Defaults to 0.99. This arguemtn is only relevant if
<code>use_linker</code> is set to <code>TRUE</code>. It refers to the confidence
threshold value used by the scispacy UMLS entity linker. Note: This can be
lower than the <code>threshold</code> from <code><a href="#topic+clinspacy_init">clinspacy_init</a></code>). The
linker_threshold can only be set once per session.</p>
</td></tr>
<tr><td><code id="clinspacy_init_+3A_...">...</code></td>
<td>
<p>Additional settings available from:
<a href="https://github.com/allenai/scispacy">https://github.com/allenai/scispacy</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>

<hr>
<h2 id='dataset_cui2vec_definitions'>Cui2vec concept definitions</h2><span id='topic+dataset_cui2vec_definitions'></span>

<h3>Description</h3>

<p>This dataset contains definitions for the Unified Medical Language System
(UMLS) Concept Unique Identifiers (CUIs). These come from Andrew Beam's
<a href="https://github.com/beamandrew/cui2vec">cui2vec R package</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataset_cui2vec_definitions()
</code></pre>


<h3>Format</h3>

<p>A data frame with 3053795 rows and 3 variables: </p>

<dl>
<dt>cui</dt><dd><p>A Unified Medical Language System (UMLS) Concept Unique
Identifier (CUI)</p>
</dd> <dt>semantic_type</dt><dd><p>Semantic type of the CUI</p>
</dd>
<dt>definition</dt><dd><p>Definition of the CUI</p>
</dd> </dl>



<h3>Details</h3>

<p>License
</p>
<p>This data is made available under a
<a href="https://github.com/beamandrew/cui2vec/blob/master/LICENSE.md">MIT
license</a>. The data is copyrighted in 2019 by Benjamin Kompa, Andrew Beam, and
Allen Schmaltz. The only change made to the original dataset is the renaming
of columns.
</p>


<h3>Value</h3>

<p>Returns the cui2vec UMLS definitions as a data frame.
</p>


<h3>Source</h3>

<p><a href="https://github.com/beamandrew/cui2vec">https://github.com/beamandrew/cui2vec</a>
</p>

<hr>
<h2 id='dataset_cui2vec_embeddings'>Cui2vec concept embeddings</h2><span id='topic+dataset_cui2vec_embeddings'></span>

<h3>Description</h3>

<p>This dataset contains Unified Medical Langauge System (UMLS) concept
embeddings from Andrew Beam's
<a href="https://github.com/beamandrew/cui2vec">cui2vec R package</a>. There are
500 embeddings included for each concept.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataset_cui2vec_embeddings()
</code></pre>


<h3>Format</h3>

<p>A data frame with 109053 rows and 501 variables: </p>

<dl>
<dt>cui</dt><dd><p>A Unified Medical Language System (UMLS) Concept Unique
Identifier (CUI)</p>
</dd> <dt>emb_001</dt><dd><p>Concept embedding vector #1</p>
</dd>
<dt>emb_002</dt><dd><p>Concept embedding vector #2</p>
</dd> <dt>...</dt><dd><p>and so on...</p>
</dd>
<dt>emb_500</dt><dd><p>Concept embedding vector #500</p>
</dd> </dl>



<h3>Details</h3>

<p>This dataset is not viewable until it has been downloaded, which will occur
the very first time you run <code>clinspacy_init()</code> after installing this
package.
</p>
<p>Citation
</p>
<p>Beam, A.L., Kompa, B., Schmaltz, A., Fried, I., Griffin, W, Palmer, N.P.,
Shi, X., Cai, T., and Kohane, I.S.,, 2019. Clinical Concept Embeddings
Learned from Massive Sources of Multimodal Medical Data. arXiv preprint
arXiv:1804.01486.
</p>
<p>License
</p>
<p>This data is made available under a
<a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0 license</a>. The
only change made to the original dataset is the renaming of columns.
</p>


<h3>Value</h3>

<p>Returns the cui2vec UMLS embeddings as a data frame.
</p>


<h3>Source</h3>

<p><a href="https://figshare.com/s/00d69861786cd0156d81">https://figshare.com/s/00d69861786cd0156d81</a>
</p>

<hr>
<h2 id='dataset_mtsamples'>Medical transcription samples.</h2><span id='topic+dataset_mtsamples'></span>

<h3>Description</h3>

<p>This dataset contains sample medical transcriptions for various medical
specialties.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataset_mtsamples()
</code></pre>


<h3>Format</h3>

<p>A data frame with 4999 rows and 6 variables: </p>

<dl>
<dt>note_id</dt><dd><p>A unique identifier for each note</p>
</dd> <dt>description</dt><dd><p>A
description or chief concern</p>
</dd> <dt>medical_specialty</dt><dd><p>Medical specialty of
the note</p>
</dd> <dt>sample_name</dt><dd><p>mtsamples.com note name</p>
</dd>
<dt>transcription</dt><dd><p>Transcription of note text</p>
</dd> <dt>keywords</dt><dd><p>Keywords</p>
</dd>
</dl>



<h3>Details</h3>

<p>Acknowledgements
</p>
<p>This data was scraped from
<a href="https://mtsamples.com">https://mtsamples.com</a> by Tara Boyle.
</p>
<p>License This data is made available under a
<a href="https://creativecommons.org/share-your-work/public-domain/cc0/">CC0:
Public Domain license</a>.
</p>


<h3>Value</h3>

<p>Returns the mtsamples dataset as a data frame.
</p>


<h3>Source</h3>

<p><a href="https://www.kaggle.com/tboyle10/medicaltranscriptions/data">https://www.kaggle.com/tboyle10/medicaltranscriptions/data</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
