<!DOCTYPE html><html><head><title>Help for package CLONETv2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CLONETv2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adm_table_toy'><p>Toy example of admixture table.</p></a></li>
<li><a href='#allele_specific_cna_table_toy'><p>Toy example of allele specific table of somatic copy number.</p></a></li>
<li><a href='#bt_toy'><p>Toy example of beta table.</p></a></li>
<li><a href='#check_ploidy_and_admixture'><p>Function to compute ploidy from a beta table.</p></a></li>
<li><a href='#CLONETv2-package'>
<p>CLONETv2</p></a></li>
<li><a href='#compute_allele_specific_scna_table'><p>Function to compute allele specific somatic copy number</p></a></li>
<li><a href='#compute_beta_table'><p>Function to compute beta table</p></a></li>
<li><a href='#compute_dna_admixture'><p>Function to compute DNA admixture of a tumor sample from the associatd beta</p>
table and ploidy table</a></li>
<li><a href='#compute_ploidy'><p>Function to compute ploidy from a beta table.</p></a></li>
<li><a href='#compute_scna_clonality_table'><p>Function to compute clonality of somatic copy number data</p></a></li>
<li><a href='#compute_snv_clonality'><p>Function to compute clonality of SNVs</p></a></li>
<li><a href='#error_table'><p>Beta estimation error.</p></a></li>
<li><a href='#pileup_normal_toy'><p>Toy example of normal pileup data.</p></a></li>
<li><a href='#pileup_tumor_toy'><p>Toy example of tumor pileup data.</p></a></li>
<li><a href='#pl_table_toy'><p>Toy example of ploidy table.</p></a></li>
<li><a href='#scna_clonality_table_toy'><p>Toy example of clonality table of somatic copy number.</p></a></li>
<li><a href='#seg_tb_toy'><p>Toy example of segmetd data.</p></a></li>
<li><a href='#snv_clonality_table_toy'><p>Toy example of snv clonality table.</p></a></li>
<li><a href='#snv_reads_toy'><p>Toy example of snv data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Clonality Estimates in Tumor</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Davide Prandi [aut],
  Alessandro Romanel [ctb],
  Tarcisio Fedrizzi [ctb],
  Yari Ciani [cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yari Ciani &lt;yari.ciani@unitn.it&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analyze data from next-generation sequencing experiments on genomic samples. 'CLONETv2' offers a set of functions to compute allele specific copy number and clonality from segmented data and SNPs position pileup. The package has also calculated the clonality of single nucleotide variants given read counts at mutated positions. The package has been developed at the laboratory of Computational and Functional Oncology, Department of CIBIO, University of Trento (Italy), under the supervision of prof Francesca Demichelis. References: Prandi et al. (2014) &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0439-6">doi:10.1186/s13059-014-0439-6</a>&gt;; Carreira et al. (2014) &lt;<a href="https://doi.org/10.1126%2Fscitranslmed.3009448">doi:10.1126/scitranslmed.3009448</a>&gt;; Romanel et al. (2015) &lt;<a href="https://doi.org/10.1126%2Fscitranslmed.aac9511">doi:10.1126/scitranslmed.aac9511</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, sets, ggplot2, ggrepel, arules, dbscan</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-13 15:04:06 UTC; t.fedrizzi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-13 20:40:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='adm_table_toy'>Toy example of admixture table.</h2><span id='topic+adm_table_toy'></span>

<h3>Description</h3>

<p>Toy example of admixture table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adm_table_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 1 rows and 4 columns.</p>

<hr>
<h2 id='allele_specific_cna_table_toy'>Toy example of allele specific table of somatic copy number.</h2><span id='topic+allele_specific_cna_table_toy'></span>

<h3>Description</h3>

<p>Toy example of allele specific table of somatic copy number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allele_specific_cna_table_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4 rows and 15 columns.</p>

<hr>
<h2 id='bt_toy'>Toy example of beta table.</h2><span id='topic+bt_toy'></span>

<h3>Description</h3>

<p>Toy example of beta table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bt_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4 rows and 10 columns.</p>

<hr>
<h2 id='check_ploidy_and_admixture'>Function to compute ploidy from a beta table.</h2><span id='topic+check_ploidy_and_admixture'></span>

<h3>Description</h3>

<p>This function takes the beta table of a tumor sample and returns its ploidy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_ploidy_and_admixture(beta_table, ploidy_table, admixture_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_ploidy_and_admixture_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="check_ploidy_and_admixture_+3A_ploidy_table">ploidy_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_ploidy">compute_ploidy</a></code></p>
</td></tr>
<tr><td><code id="check_ploidy_and_admixture_+3A_admixture_table">admixture_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_dna_admixture">compute_dna_admixture</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 plot reporting log2 on the x axis and beta and the y axis.
Each dot represents a segment of the input beta_table. Red transparent
circles corresponds to expected log2 vs beta position for different allele
specific copy number combinations given ploidy and admixture reported in
tables ploidy_table and admixture_table, respectively. Labels in the form
(cnA, cnB) indicate repsectively the major and minor allele copy number
value. Labels above the plot comprises sample name and ploddy/admixture estimates.
</p>


<h3>Author(s)</h3>

<p>Davide Prandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## check ploidy and admixture estimates
check_plot_toy &lt;- check_ploidy_and_admixture(beta_table = bt_toy, ploidy_table = pl_table_toy,
  admixture_table = adm_table_toy)

</code></pre>

<hr>
<h2 id='CLONETv2-package'>
CLONETv2
</h2><span id='topic+CLONETv2-package'></span>

<h3>Description</h3>

<p>This package is designed to analyze data from next-generation sequencing
experiments on genomic samples. It offers a
set of functions to compute allele specific copy number and
clonality from segmented data and SNPs position pileup. The library
also calculated the clonality of single nucleotide variants given
read counts at mutated positions.
</p>
<p>The package has been developed at the laboratory of Computational and Functional Oncology,
Department of CIBIO, University of Trento (Italy), under the supervision of prof.
Francesca Demichelis.
</p>


<h3>Author(s)</h3>

<p>Maintainer:
Davide Prandi
</p>
<p>Contributors:
Tarcisio Fedrizzi, Alessandro Romanel
</p>


<h3>References</h3>

<p>Prandi, D., Baca, S. C., Romanel, A., Barbieri, C. E., Mosquera, J. M., Fontugne, J., Beltran,
H., Sboner,
A., Garraway, L. A., Rubin, M. A., and Demichelis, F. (2014). Unraveling the clonal hierarchy of
somatic genomic aberrations. Genome biology 15, 439.
</p>
<p>Carreira, S., Romanel, A., Goodall, J., Grist, E., Ferraldeschi, R., Miranda, S., Prandi, D.,
Lorente, D.,
Frenel, J. S., Pezaro, C., et al. (2014). Tumor clone dynamics in lethal prostate cancer. Science
translational medicine 6, 254ra125.
</p>
<p>Beltran, H., Eng, K., Mosquera, J. M., Sigaras, A., Romanel, A., Rennert, H., Kossai, M.,
Pauli, C.,
Faltas, B., Fontugne, J., et al. (2015). Whole-Exome Sequencing of Metastatic Cancer and
Biomarkers
of Treatment Response. JAMA Oncol 1, 466-474.
</p>
<p>Faltas, B. M., Prandi, D., Tagawa, S. T., Molina, A. M., Nanus, D. M., Sternberg, C.,
Rosenberg, J.,
Mosquera, J. M., Robinson, B., Elemento, O., et al. (2016). Clonal evolution of
chemotherapyresistant
urothelial carcinoma. Nature genetics 48, 1490-1499.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
###############
###############
## Diploid tumor sample

## Load example data
seg_tb &lt;- read.table(system.file("sample1.seg", package = "CLONETv2"),header = TRUE, as.is=TRUE)
pileup_tumor &lt;- read.table(
  gzfile(system.file("sample1_tumor_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
pileup_normal &lt;- read.table(
  gzfile(system.file("sample1_normal_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
snv_reads &lt;- read.table(system.file("sample1_snv_read_count.tsv", package = "CLONETv2"),
  header = TRUE, as.is=TRUE, comment.char = "", check.names = FALSE, na.strings = "-")

## Compute beta table with default parameters
bt &lt;- compute_beta_table(seg_tb, pileup_tumor, pileup_normal)

## Compute ploidy table with default parameters
pl_table &lt;- compute_ploidy(bt)

## Compute admixture table with default parameters (admixture= 1-tumor_purity)
adm_table &lt;- compute_dna_admixture(beta_table = bt, ploidy_table = pl_table)

## Check ploidy and admixture estimates
check_plot &lt;-  check_ploidy_and_admixture(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)
print(check_plot)

## Compute clonality table with default parameters
scna_clonality_table &lt;- compute_scna_clonality_table(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)

## Compute allele specific scna
allele_specific_cna_table &lt;- compute_allele_specific_scna_table(beta_table = bt,
  ploidy_table = pl_table, admixture_table = adm_table)

## Compute snvs colonality
sample_id &lt;- "sample1"
snv_clonality_table &lt;- compute_snv_clonality(sample_id = sample_id, snv_read_count = snv_reads,
  beta_table = bt, ploidy_table = pl_table, admixture_table = adm_table)


###############
###############
## Aneuploid tumor sample

## Load example data
seg_tb &lt;- read.table(system.file("sample2.seg", package = "CLONETv2"),header = TRUE, as.is=TRUE)
pileup_tumor &lt;- read.table(
  gzfile(system.file("sample2_tumor_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
pileup_normal &lt;- read.table(
  gzfile(system.file("sample2_normal_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
snv_reads &lt;- read.table(system.file("sample2_snv_read_count.tsv", package = "CLONETv2"),
  header = TRUE, as.is=TRUE, comment.char = "", check.names = FALSE, na.strings = "-")

## Compute beta table with default parameters
bt &lt;- compute_beta_table(seg_tb, pileup_tumor, pileup_normal)

## Compute ploidy table with default parameters
pl_table &lt;- compute_ploidy(bt)

## Compute admixture table with default parameters (admixture= 1-tumor_purity)
adm_table &lt;- compute_dna_admixture(beta_table = bt, ploidy_table = pl_table)

## Check ploidy and admixture estimates
check_plot &lt;-  check_ploidy_and_admixture(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)
print(check_plot)

## Compute clonality table with default parameters
scna_clonality_table &lt;- compute_scna_clonality_table(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)

## Compute allele specific scna
allele_specific_cna_table &lt;- compute_allele_specific_scna_table(beta_table = bt,
  ploidy_table = pl_table, admixture_table = adm_table)

## Compute snvs colonality
sample_id &lt;- "sample2"
snv_clonality_table &lt;- compute_snv_clonality(sample_id = sample_id, snv_read_count = snv_reads,
  beta_table = bt, ploidy_table = pl_table, admixture_table = adm_table)


###############
###############
## Aneuploidy tumor sample with problematic ploidy estimate

## Load example data
seg_tb &lt;- read.table(system.file("sample3.seg", package = "CLONETv2"),header = TRUE, as.is=TRUE)
pileup_tumor &lt;- read.table(
  gzfile(system.file("sample3_tumor_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
pileup_normal &lt;- read.table(
  gzfile(system.file("sample3_normal_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)

## Compute beta table with default parameters
bt &lt;- compute_beta_table(seg_tb, pileup_tumor, pileup_normal)

## Compute ploidy table with default parameters
pl_table &lt;- compute_ploidy(bt)

## Compute admixture table with default parameters (admixture= 1-tumor_purity)
adm_table &lt;- compute_dna_admixture(beta_table = bt, ploidy_table = pl_table)

## Check ploidy and admixture estimates
check_plot &lt;-  check_ploidy_and_admixture(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)
print(check_plot)
## Observed data (gray points) does not fit with expcted positions (Red circles)


###############
###############
## Tumor sample with problem in the segmented input data

## Load example data
seg_tb &lt;- read.table(system.file("sample4.seg", package = "CLONETv2"),header = TRUE, as.is=TRUE)
pileup_tumor &lt;- read.table(
  gzfile(system.file("sample4_tumor_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
pileup_normal &lt;- read.table(
  gzfile(system.file("sample4_normal_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)

## Compute beta table with default parameters
bt &lt;- compute_beta_table(seg_tb, pileup_tumor, pileup_normal)

## Compute ploidy table with default parameters
pl_table &lt;- compute_ploidy(bt)

## Compute admixture table with default parameters (admixture= 1-tumor_purity)
adm_table &lt;- compute_dna_admixture(beta_table = bt, ploidy_table = pl_table)

## Check ploidy and admixture estimates
check_plot &lt;-  check_ploidy_and_admixture(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)
print(check_plot)
## CLONETv2 does not provide an estimate of the DNA admixture because
## (LogR, beta) data does not fit any CLONETv2 model


###############
###############
## Diploid tumor sample with subclonal hemizygous and homozygous deletions

## Load example data
seg_tb &lt;- read.table(system.file("sample5.seg", package = "CLONETv2"),header = TRUE, as.is=TRUE)
pileup_tumor &lt;- read.table(
  gzfile(system.file("sample5_tumor_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
pileup_normal &lt;- read.table(
  gzfile(system.file("sample5_normal_pileup.tsv.gz", package = "CLONETv2")),
  header = TRUE, as.is=TRUE)
snv_reads &lt;- read.table(system.file("sample5_snv_read_count.tsv", package = "CLONETv2"),
  header = TRUE, as.is=TRUE, comment.char = "", check.names = FALSE, na.strings = "-")

## Compute beta table with default parameters
bt &lt;- compute_beta_table(seg_tb, pileup_tumor, pileup_normal)

## Compute ploidy table with default parameters
pl_table &lt;- compute_ploidy(bt)

## Compute admixture table with default parameters (admixture= 1-tumor_purity)
adm_table &lt;- compute_dna_admixture(beta_table = bt, ploidy_table = pl_table)

## Check ploidy and admixture estimates
check_plot &lt;-  check_ploidy_and_admixture(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)
print(check_plot)

## Compute clonality table with default parameters
scna_clonality_table &lt;- compute_scna_clonality_table(beta_table = bt, ploidy_table = pl_table,
  admixture_table = adm_table)

## Compute allele specific scna
allele_specific_cna_table &lt;- compute_allele_specific_scna_table(beta_table = bt,
  ploidy_table = pl_table, admixture_table = adm_table)

## Compute snvs colonality
sample_id &lt;- "sample5"
snv_clonality_table &lt;- compute_snv_clonality(sample_id = sample_id, snv_read_count = snv_reads,
  beta_table = bt, ploidy_table = pl_table, admixture_table = adm_table)


</code></pre>

<hr>
<h2 id='compute_allele_specific_scna_table'>Function to compute allele specific somatic copy number</h2><span id='topic+compute_allele_specific_scna_table'></span>

<h3>Description</h3>

<p>This function takes the beta table of a tumor sample together with the
associated ploidy and admixtures tables and computes the allele specific copy
number of each segment in the beta table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_allele_specific_scna_table(beta_table, ploidy_table,
  admixture_table, error_tb = error_table, allelic_imbalance_th = 0.5,
  n_digits = 3, n_cores = 1, debug = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_allele_specific_scna_table_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_ploidy_table">ploidy_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_ploidy">compute_ploidy</a></code></p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_admixture_table">admixture_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_dna_admixture">compute_dna_admixture</a></code></p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_error_tb">error_tb</code></td>
<td>
<p>data.frame that reports for each combination of coverage and
number informative SNPs the expected estimation error around beta. The
data.frame error_tb must contains 3 columns: </p>
 <dl>
<dt>mean.cov</dt><dd><p>mean
coverage</p>
</dd> <dt>n.info.snps</dt><dd><p>number of informative SNPs</p>
</dd>
<dt>adm.estimation.error</dt><dd><p>estimated error on computed beta on a segment
with coverage mean.cov and n.info.snps informative SNPs</p>
</dd> </dl>
<p> Package CLONETv2
have built in error_tb named error_table (default=error_table)</p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_allelic_imbalance_th">allelic_imbalance_th</code></td>
<td>
<p>maximum distance from allele spefici copy number
of a segment to define integer alelle specific copy number value. Value 0.5
corresponds to round cnA and cnB (default=0.5)</p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table  (default=3)</p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_n_cores">n_cores</code></td>
<td>
<p>number of cores (default=1)</p>
</td></tr>
<tr><td><code id="compute_allele_specific_scna_table_+3A_debug">debug</code></td>
<td>
<p>return extra columns for debugging (default=F)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that extends input beta_table with columns </p>

<dl>
<dt>log2.corr</dt><dd><p>log2 ratio adjusted by ploidy and admixture</p>
</dd>
<dt>cnA</dt><dd><p>copy number of the major allele</p>
</dd> <dt>cnB</dt><dd><p>copy number of the
minor allele</p>
</dd> <dt>cnA.int</dt><dd><p>integet copy number of the major allele</p>
</dd>
<dt>cnB.int</dt><dd><p>integet copy number of the minor allele</p>
</dd> </dl>



<h3>Author(s)</h3>

<p>Davide Prandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute clonality table with default parameters
allele_specific_cna_table_toy &lt;- compute_allele_specific_scna_table(
  beta_table = bt_toy, ploidy_table = pl_table_toy,
  admixture_table = adm_table_toy)

</code></pre>

<hr>
<h2 id='compute_beta_table'>Function to compute beta table</h2><span id='topic+compute_beta_table'></span>

<h3>Description</h3>

<p>This function takes segmented data and per base pileup of tumor and matched
normal of a sample as input and associates a beta value to each genomic segment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_beta_table(seg_tb, pileup_tumor, pileup_normal,
  min_coverage = 20, min_required_snps = 10, min_af_het_snps = 0.2,
  max_af_het_snps = 0.8, n_digits = 3, n_cores = 1, plot_stats = F,
  debug = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_beta_table_+3A_seg_tb">seg_tb</code></td>
<td>
<p>data.frame in <a href="http://software.broadinstitute.org/software/igv/SEG">SEG format</a>. Rows
report per segment log2 ratio
numeric value. CLONETv2 inteprets first column as sample name, columns two
to four as genomic coordinates (chromosome, start location, and end
location), column five is not used, and column six is the log2 ratio
returned by segmentation algorithm.</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_pileup_tumor">pileup_tumor</code>, <code id="compute_beta_table_+3A_pileup_normal">pileup_normal</code></td>
<td>
<p>data.frame reporting pileup of SNPs in
tumor and normal samples respectively. First row contains column names and
subsequent rows report the pileup of a specific genomic positions. Required
information for each genomic position includes
chromosome,    position,    allelic fraction, and    coverage. Required column names
are chr, pos, af, and cov</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_min_coverage">min_coverage</code></td>
<td>
<p>minimum number of reads for considering a pileup position
valid (default=20)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_min_required_snps">min_required_snps</code></td>
<td>
<p>minimum number of snps to call beta for a segment
(default=10)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_min_af_het_snps">min_af_het_snps</code></td>
<td>
<p>minimum allowed allelic fraction of a SNP genomic
position (default=0.2)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_max_af_het_snps">max_af_het_snps</code></td>
<td>
<p>maximum allowed allelic fraction of a SNP genomic
position (default=0.8)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table  (default=3)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_n_cores">n_cores</code></td>
<td>
<p>number of available cores for computation (default=1)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_plot_stats">plot_stats</code></td>
<td>
<p>plot summary statistics of the computed beta table (default=F)</p>
</td></tr>
<tr><td><code id="compute_beta_table_+3A_debug">debug</code></td>
<td>
<p>return extra columns for debugging (default=F)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that extends input seg_tb with columns beta, nsnp, cov,
n_beta. Moreover, CLONETv2 renames colums of seg_tb as sample, chr,
start, end, XYZ, log2, with XYZ being the original name of column five
As for seg_tb, each raw of the output table represents a genomic
segments. For each raw,  the value of beta is the proportion of neutral
reads in the segment, while nsnp and cov represents respectively the number
of informative SNPs and the mean coverage of the given segment. The value
n_beta is the proportion of neutral reads in the normal sample. The value
of n_beta should be 1 as in normal samples parental chromosomes are equally
represented. Values lower than 1 of n_beta could indicate the presence of germline CNVs
or sequencing errors.
</p>


<h3>Author(s)</h3>

<p>Davide Prandi, Alessandro Romanel
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compue beta table with default parameters
bt_toy &lt;- compute_beta_table(seg_tb_toy, pileup_tumor_toy, pileup_normal_toy)
</code></pre>

<hr>
<h2 id='compute_dna_admixture'>Function to compute DNA admixture of a tumor sample from the associatd beta
table and ploidy table</h2><span id='topic+compute_dna_admixture'></span>

<h3>Description</h3>

<p>This function takes a beta table and the associated ploidy table and computes
DNA admixture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_dna_admixture(beta_table, ploidy_table, min_required_snps = 10,
  min_coverage = 20, error_tb = error_table, library_type = "WES",
  n_digits = 3, n_cores = 1, debug = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_dna_admixture_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_ploidy_table">ploidy_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_ploidy">compute_ploidy</a></code></p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_min_required_snps">min_required_snps</code></td>
<td>
<p>minimum number of informative snps in  a segment
valid for computing ploidy (default=10)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_min_coverage">min_coverage</code></td>
<td>
<p>minimum coverage of a segment valid for computing ploidy
(default=20)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_error_tb">error_tb</code></td>
<td>
<p>data.frame that reports for each combination of coverage and
number informative SNPs the expected estimation error around beta. The
data.frame error_tb must contains 3 columns: </p>

<dl>
<dt>mean.cov</dt><dd><p>mean coverage</p>
</dd> <dt>n.info.snps</dt><dd><p>number of informative
SNPs</p>
</dd> <dt>adm.estimation.error</dt><dd><p>estimated error on computed beta on a
segment with coverage mean.cov and n.info.snps informative SNPs</p>
</dd> </dl>
<p> Package
CLONETv2 have built in error_tb named error_table (default=error_table)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_library_type">library_type</code></td>
<td>
<p>WES, WGS (default=WES)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table (default=3)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_n_cores">n_cores</code></td>
<td>
<p>number of available cores for computation (default=1)</p>
</td></tr>
<tr><td><code id="compute_dna_admixture_+3A_debug">debug</code></td>
<td>
<p>return extra columns for debugging (default=F)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with two columns: sample that corresponds to column
sample of the input beta_table, and amd that represent the fraction of
estimated DNA admixture
</p>


<h3>Author(s)</h3>

<p>Davide Prandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute admixture table with default parameters
adm_table_toy &lt;- compute_dna_admixture(beta_table = bt_toy, ploidy_table = pl_table_toy)

</code></pre>

<hr>
<h2 id='compute_ploidy'>Function to compute ploidy from a beta table.</h2><span id='topic+compute_ploidy'></span>

<h3>Description</h3>

<p>This function takes the beta table of a tumor sample and returns its ploidy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_ploidy(beta_table, max_homo_dels_fraction = 0.01,
  beta_limit_for_neutral_reads = 0.9, min_coverage = 20,
  min_required_snps = 10, library_type = "WES", n_digits = 3,
  n_cores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_ploidy_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_max_homo_dels_fraction">max_homo_dels_fraction</code></td>
<td>
<p>estimated maximum proportion of genomic
segments corresponding to an homozygous deletion (default=0.01)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_beta_limit_for_neutral_reads">beta_limit_for_neutral_reads</code></td>
<td>
<p>minimum beta value of a segment valid for
computing ploidy (default=0.90)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_min_coverage">min_coverage</code></td>
<td>
<p>minimum coverage of a segment valid for computing ploidy
(default=20)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_min_required_snps">min_required_snps</code></td>
<td>
<p>minimum number of informative snps in  a segment
valid for computing ploidy (default=10)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_library_type">library_type</code></td>
<td>
<p>WES, WGS (default=WES)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table (default=3)</p>
</td></tr>
<tr><td><code id="compute_ploidy_+3A_n_cores">n_cores</code></td>
<td>
<p>number of available cores for computation (default=1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with two columns: sample that corresponds to column
sample of the input beta_table, and ploidy computed
</p>


<h3>Author(s)</h3>

<p>Davide Prandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute ploidy table with default parameters
pl_table_toy &lt;- compute_ploidy(bt_toy)

</code></pre>

<hr>
<h2 id='compute_scna_clonality_table'>Function to compute clonality of somatic copy number data</h2><span id='topic+compute_scna_clonality_table'></span>

<h3>Description</h3>

<p>This function takes the beta table of a tumor sample together with the
associated ploidy and admixtures tables and computes the clonality of each
segment in the beta table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_scna_clonality_table(beta_table, ploidy_table, admixture_table,
  error_tb = error_table, clonality_threshold = 0.85,
  beta_threshold = 0.9, n_digits = 3, n_cores = 1, debug = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_scna_clonality_table_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_ploidy_table">ploidy_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_ploidy">compute_ploidy</a></code></p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_admixture_table">admixture_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_dna_admixture">compute_dna_admixture</a></code></p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_error_tb">error_tb</code></td>
<td>
<p>data.frame that reports for each combination of coverage and
number informative SNPs the expected estimation error around beta. The
data.frame error_tb must contains 3 columns: </p>
 <dl>
<dt>mean.cov</dt><dd><p>mean
coverage</p>
</dd> <dt>n.info.snps</dt><dd><p>number of informative SNPs</p>
</dd>
<dt>adm.estimation.error</dt><dd><p>estimated error on computed beta on a segment
with coverage mean.cov and n.info.snps informative SNPs</p>
</dd> </dl>
<p> Package CLONETv2
have built in error_tb named error_table (default=error_table)</p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_clonality_threshold">clonality_threshold</code></td>
<td>
<p>threshold to discretize continuous clonality value
(default=0.85)</p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_beta_threshold">beta_threshold</code></td>
<td>
<p>threshold on beta value to determine clonality direction
(default=0.90)</p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table  (default=3)</p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_n_cores">n_cores</code></td>
<td>
<p>number of cores (default=1)</p>
</td></tr>
<tr><td><code id="compute_scna_clonality_table_+3A_debug">debug</code></td>
<td>
<p>return extra columns for debugging (default=F)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that extends input beta_table with columns </p>

<dl>
<dt>clonality</dt><dd><p>estimated fraction of tumor cell with log2 copy number</p>
</dd>
<dt>clonality.min</dt><dd><p>minum estimated fraction of tumor cell with log2 copy
number</p>
</dd> <dt>clonality.max</dt><dd><p>minum estimated fraction of tumor cell with
log2 copy number</p>
</dd> <dt>clonality.status</dt><dd><p>discretized clonality status into
five values: <em>clonal</em>, large majority of the tumor cells has the same
copy number; <em>subclonal</em>, not all the tumor cells has the same copy
number; <em>not.analysed</em>, is is not possible to determine clonality;
<em>uncertain.clonal</em> and <em>uncertain.subclonal</em> correspond
respectively to <em>clonal</em> and <em>subclonal</em> populations but with less
reliable clonality estimate </p>
</dd> </dl>



<h3>Author(s)</h3>

<p>Davide Prandi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

## Compute clonality table with default parameters
scna_clonality_table_toy &lt;- compute_scna_clonality_table(beta_table = bt_toy,
  ploidy_table = pl_table_toy, admixture_table = adm_table_toy)

</code></pre>

<hr>
<h2 id='compute_snv_clonality'>Function to compute clonality of SNVs</h2><span id='topic+compute_snv_clonality'></span>

<h3>Description</h3>

<p>This function takes as input the genomic position of a SNVs and computes the
percentage of genomic homogeneus cells harboring the mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_snv_clonality(sample_id, snv_read_count, beta_table, ploidy_table,
  admixture_table, error_tb = error_table, error_rate = 0.05,
  n_digits = 3, n_cores = 1, annotation_style = "VEP", debug = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_snv_clonality_+3A_sample_id">sample_id</code></td>
<td>
<p>the id of the analyzed sample. It must be the same value
reported in column sample of tables beta_table, ploidy_table, and
admixture_table</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_snv_read_count">snv_read_count</code></td>
<td>
<p>data.frame reporting in each row the genomic coordinates
of an SNV together with number of reference and alternative reads covering
the position in columns rc_ref_tumor and rc_alt_tumor, respectively. See
parameter annotation_style for details about column names</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_beta_table">beta_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_beta_table</a></code></p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_ploidy_table">ploidy_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_ploidy</a></code></p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_admixture_table">admixture_table</code></td>
<td>
<p>data.frame formatted as the output of function
<code><a href="#topic+compute_beta_table">compute_dna_admixture</a></code></p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_error_tb">error_tb</code></td>
<td>
<p>data.frame that reports for each combination of coverage and
number informative SNPs the expected estimation error around beta. The
data.frame error_tb must contains 3 columns: </p>
 <dl>
<dt>mean.cov</dt><dd><p>mean
coverage</p>
</dd> <dt>n.info.snps</dt><dd><p>number of informative SNPs</p>
</dd>
<dt>adm.estimation.error</dt><dd><p>estimated error on computed beta on a segment
with coverage mean.cov and n.info.snps informative SNPs</p>
</dd> </dl>
<p> Package CLONETv2
have built in error_tb named error_table (default=error_table)</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_error_rate">error_rate</code></td>
<td>
<p>expected fraction of SNV positions with outlier variant
allelic fraction (default=0.05)</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_n_digits">n_digits</code></td>
<td>
<p>number of digits in the output table  (default=3)</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_n_cores">n_cores</code></td>
<td>
<p>number of cores (default=1)</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_annotation_style">annotation_style</code></td>
<td>
<p>a string that corresponds to the format of the columns
that describe the genomic coordinates of a SNV. Accepted values are VEP and
MAF. <a href="https://www.ensembl.org/info/docs/tools/vep/index.html">VEP annotation</a>
describes genomic coordinates with a single column named Location. <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">MAF format</a> has
columns Chromosome, Start_position, and End_position for each aberrant
position</p>
</td></tr>
<tr><td><code id="compute_snv_clonality_+3A_debug">debug</code></td>
<td>
<p>return extra columns for debugging (default=F)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that extends input table snv_read_count with columns
sample, cnA, cnB, t_af, t_af_corr, SNV.clonality, and SNV.clonality.status.
Columns cnA and cnB report the allele specific copy number of the genomic
segment containing the SNV position. Columns t_af and t_af_corr are
respectively raw and ploidy/purity adjusted tumor varian allelic fractions.
SNV.clonality reports the percentage of tumor cells harboring the SNV and
with allele specific copy number cnA and cnB. SNV.clonality.status column
lists dicretized SNV.clonality values. Discrete states are clonal,
uncertain.clonal, uncertain.subclonal, and subclonal based in threshold
automatically computed on the SNV.clonality values. Empty
SNV.clonality.status of an SNV indicates that clonality cannot be assessed.
</p>


<h3>Author(s)</h3>

<p>Davide Prandi, Tarcisio Fedrizzi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Compute SNVs clonality
snv_clonality_table_toy &lt;- compute_snv_clonality("toy_sample",
  snv_reads_toy, bt_toy, pl_table_toy, adm_table_toy)

</code></pre>

<hr>
<h2 id='error_table'>Beta estimation error.</h2><span id='topic+error_table'></span>

<h3>Description</h3>

<p>A precomputed table reporting for different combinations of coverage and
number of informative SNPs  the expected error of the  beta value computed by
function <code><a href="#topic+compute_beta_table">compute_beta_table</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>error_table
</code></pre>


<h3>Format</h3>

<p>A data frame column names mean.cov, n.info.snps, and
adm.estimation.error </p>
 <dl>
<dt>mean.cov</dt><dd><p>genomic segment coverage</p>
</dd>
<dt>n.info.snps</dt><dd><p>number of informative SNPs</p>
</dd>
<dt>adm.estimation.error</dt><dd><p>expected error on beta estimate </p>
</dd> </dl>

<hr>
<h2 id='pileup_normal_toy'>Toy example of normal pileup data.</h2><span id='topic+pileup_normal_toy'></span>

<h3>Description</h3>

<p>Toy example of normal pileup data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pileup_normal_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 816 rows and 11 columns.</p>

<hr>
<h2 id='pileup_tumor_toy'>Toy example of tumor pileup data.</h2><span id='topic+pileup_tumor_toy'></span>

<h3>Description</h3>

<p>Toy example of tumor pileup data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pileup_tumor_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 816 rows and 11 columns.</p>

<hr>
<h2 id='pl_table_toy'>Toy example of ploidy table.</h2><span id='topic+pl_table_toy'></span>

<h3>Description</h3>

<p>Toy example of ploidy table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pl_table_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 1 rows and 2 columns.</p>

<hr>
<h2 id='scna_clonality_table_toy'>Toy example of clonality table of somatic copy number.</h2><span id='topic+scna_clonality_table_toy'></span>

<h3>Description</h3>

<p>Toy example of clonality table of somatic copy number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scna_clonality_table_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4 rows and 25 columns.</p>

<hr>
<h2 id='seg_tb_toy'>Toy example of segmetd data.</h2><span id='topic+seg_tb_toy'></span>

<h3>Description</h3>

<p>Toy example of segmetd data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seg_tb_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4 rows and 6 columns.</p>

<hr>
<h2 id='snv_clonality_table_toy'>Toy example of snv clonality table.</h2><span id='topic+snv_clonality_table_toy'></span>

<h3>Description</h3>

<p>Toy example of snv clonality table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snv_clonality_table_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 2 rows and 78 columns.</p>

<hr>
<h2 id='snv_reads_toy'>Toy example of snv data.</h2><span id='topic+snv_reads_toy'></span>

<h3>Description</h3>

<p>Toy example of snv data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snv_reads_toy
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 2 rows and 71 columns.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
