<!DOCTYPE html><html lang="en"><head><title>Help for package cloneRate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cloneRate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cloneRate-package'><p>The 'cloneRate' package.</p></a></li>
<li><a href='#birthDeathMCMC'><p>Growth rate estimate using MCMC</p></a></li>
<li><a href='#coal_to_tree'><p>Generate tree from coalescence times</p></a></li>
<li><a href='#exampleMutTrees'><p>Example mutation tree data</p></a></li>
<li><a href='#exampleUltraTrees'><p>Example ultrametric tree data</p></a></li>
<li><a href='#inputCheck'><p>Check the inputs to growth rate functions</p></a></li>
<li><a href='#internalLengths'><p>Growth rate estimate using the sum of internal lengths</p></a></li>
<li><a href='#longitudinalData'><p>Longitudinal validation data</p></a></li>
<li><a href='#maxLikelihood'><p>Growth rate estimate using Maximum Likelihood</p></a></li>
<li><a href='#realCloneData'><p>Real clone data from human blood</p></a></li>
<li><a href='#sharedMuts'><p>Growth rate estimate using the sum of shared mutations assuming a mutation tree</p></a></li>
<li><a href='#simMut'><p>Simulate mutation-based birth and death branching trees</p></a></li>
<li><a href='#simUltra'><p>Simulate ultrametric birth and death branching trees</p></a></li>
<li><a href='#siteFrequency'><p>Get site frequency spectrum of a tree</p></a></li>
<li><a href='#ultra2mut'><p>Add poissonian mutations to an ultrametric tree(s)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate Growth Rates from Phylogenetic Trees</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Quickly estimate the net growth rate of a
    population or clone whose growth can be approximated by a birth-death
    branching process.  Input should be phylogenetic tree(s) of clone(s) with
    edge lengths corresponding to either time or mutations. Based on
    coalescent results in Johnson et al. (2023) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtad561">doi:10.1093/bioinformatics/btad561</a>&gt;. Simulation
    techniques as well as growth rate methods build on prior work from
    Lambert A. (2018) &lt;<a href="https://doi.org/10.1016%2Fj.tpb.2018.04.005">doi:10.1016/j.tpb.2018.04.005</a>&gt; and Stadler T. (2009)
    &lt;<a href="https://doi.org/10.1016%2Fj.jtbi.2009.07.018">doi:10.1016/j.jtbi.2009.07.018</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/bdj34/cloneRate">https://github.com/bdj34/cloneRate</a>,
<a href="https://bdj34.github.io/cloneRate/">https://bdj34.github.io/cloneRate/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bdj34/cloneRate/issues">https://github.com/bdj34/cloneRate/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 4.0), methods, Rcpp (&ge; 0.12.0), Rmpfr (&ge; 0.8), rstan
(&ge; 2.18.1), rstantools (&ge; 2.3.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>car, covr, ggplot2, ggsurvfit, knitr, parallel, rmarkdown,
survival, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH (&ge; 1.66.0), Rcpp (&ge; 0.12.0), RcppEigen (&ge; 0.3.3.3.0),
RcppParallel (&ge; 5.0.1), rstan (&ge; 2.18.1), StanHeaders (&ge;
2.18.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-21 17:39:04 UTC; apple</td>
</tr>
<tr>
<td>Author:</td>
<td>Brian Johnson <a href="https://orcid.org/0000-0002-9477-4677"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut, cph],
  Yubo Shuai [aut, cph],
  Jason Schweinsberg
    <a href="https://orcid.org/0000-0002-4152-0059"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cph],
  Kit Curtius <a href="https://orcid.org/0000-0002-2678-0960"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Brian Johnson &lt;brian.d.johnson97@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-22 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cloneRate-package'>The 'cloneRate' package.</h2><span id='topic+cloneRate-package'></span><span id='topic+cloneRate'></span>

<h3>Description</h3>

<p>Quickly estimate the net growth rate of a population or clone whose growth can be approximated by a birth-death branching process.
</p>


<h3>References</h3>

<p>Johnson et al. 2023 Bioinformatics. <a href="doi:10.1093/bioinformatics/btad561">doi:10.1093/bioinformatics/btad561</a>
</p>

<hr>
<h2 id='birthDeathMCMC'>Growth rate estimate using MCMC</h2><span id='topic+birthDeathMCMC'></span>

<h3>Description</h3>

<p>Uses Rstan and the No U-turn sampler to approximate the
growth rate using the likelihood from Stadler 2009 &quot;On incomplete sampling
under birthâ€“death models and connections to the sampling-based coalescent&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>birthDeathMCMC(
  tree,
  maxGrowthRate = 4,
  alpha = 0.05,
  verbose = TRUE,
  nChains = 4,
  nCores = 1,
  chainLength = 2000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="birthDeathMCMC_+3A_tree">tree</code></td>
<td>
<p>An ultrametric tree subset to include only the clone of
interest. Alternatively, a list with several such trees.</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_maxgrowthrate">maxGrowthRate</code></td>
<td>
<p>Sets upper bound on birth rate. Default is 4 but this
will depend on the nature of the data</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_alpha">alpha</code></td>
<td>
<p>Used for calculation of confidence intervals. 1-alpha confidence intervals used with default of alpha = 0.05 (95 percent confidence intervals)</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_verbose">verbose</code></td>
<td>
<p>TRUE or FALSE, should the Rstan MCMC intermediate output and progress be printed?</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_nchains">nChains</code></td>
<td>
<p>Number of chains to run in MCMC. Default is 4</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_ncores">nCores</code></td>
<td>
<p>Number of cores to perform MCMC. Default is 1, but chains can
be run in parallel</p>
</td></tr>
<tr><td><code id="birthDeathMCMC_+3A_chainlength">chainLength</code></td>
<td>
<p>Number of iterations for each chain in MCMC. Default is
2000 (1000 warm-up + 1000 sampling), increase if stan tells you to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe including the net growth rate estimate, confidence
intervals, and other important details (clone age estimate, runtime, n,
etc.)
</p>


<h3>See Also</h3>

<p><a href="#topic+internalLengths">internalLengths</a> <a href="#topic+maxLikelihood">maxLikelihood</a> which use
alternative methods for growth rate estimation from an ultrametric tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
df &lt;- birthDeathMCMC(cloneRate::exampleUltraTrees[[1]])


</code></pre>

<hr>
<h2 id='coal_to_tree'>Generate tree from coalescence times</h2><span id='topic+coal_to_tree'></span>

<h3>Description</h3>

<p>generates a tree from a vector of coalescence times by randomly
merging lineages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coal_to_tree(coal_times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coal_to_tree_+3A_coal_times">coal_times</code></td>
<td>
<p>A numeric vector of coalescence times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ape object of class &quot;phylo&quot; representing the ultrametric
phylogenetic tree with edge lengths in units of time.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate an ape phylo tree with n tips from a vector of n-1 coalescence times
randomCoalTimes &lt;- c(9.3, 7.8, 10.15, 11.23, 9.4, 8.8, 10.01, 13)
tree &lt;- coal_to_tree(randomCoalTimes)

</code></pre>

<hr>
<h2 id='exampleMutTrees'>Example mutation tree data</h2><span id='topic+exampleMutTrees'></span>

<h3>Description</h3>

<p>Set of 100 mutation based trees reconstructed from the distribution
of a sample of n=100 tips.
All trees have a net growth rate of 1 with birth rates between 1 and 2
(sampled from a uniform distribution). Death rates are equal to the chosen
birth rate minus 1. Tree reconstruction uses the exact distribution of
coalescence times described in &quot;The coalescent of a sample from a binary
branching process&quot;, Lambert A., Theor. Pop. Bio. 2018. Tree construction and
formatting uses <code>ape</code> R package <code><a href="ape.html#topic+rtree">ape::rcoal()</a></code>. We then change the edge
lengths from time-based to mutation-based by drawing from a poisson
distribution with mean equal to edge length (in units of time) multiplied
by the mutation rate, nu, which is drawn from a uniform distribution between
10 and 20 mutations per year.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleMutTrees)
</code></pre>


<h3>Format</h3>

<p>A <code>list</code> of objects of class <code>phylo</code>
</p>

<dl>
<dt>edge</dt><dd><p>A matrix of edge connections which reconstruct the tree.</p>
</dd>
<dt>edge.length</dt><dd><p>A numeric vector of the branch lengths of the connections
in <code>edge</code> matrix. Units are mutations.</p>
</dd>
<dt>tip.label</dt><dd><p>A character vector containing the  (arbitrary in this case)
labels for the 100 tips/samples of the tree.</p>
</dd>
<dt>Nnode</dt><dd><p>Integer number of internal nodes of the tree</p>
</dd>
<dt>params</dt><dd><p><code>data.frame</code> containing info on the params used to generate
the tree</p>
</dd>
</dl>
<p>See ape package for details on class <code>phylo</code> objects.

</p>


<h3>References</h3>

<p>This data set was created for the cloneRate package using
coalescent theory approaches described in &quot;The coalescent of a sample from a
binary branching process&quot;, Lambert A., Theor. Pop. Bio. 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot first of 100 trees
ape::plot.phylo(cloneRate::exampleMutTrees[[1]],
  direction = "downwards", show.tip.label = FALSE
)

</code></pre>

<hr>
<h2 id='exampleUltraTrees'>Example ultrametric tree data</h2><span id='topic+exampleUltraTrees'></span>

<h3>Description</h3>

<p>Set of 100 time-based ultrametric trees reconstructed from the distribution
of a sample of n=100 tips.
All trees have a net growth rate of 1 with birth rates between 1 and 2
(sampled from a uniform distribution). Death rates are equal to the chosen
birth rate minus 1. Tree reconstruction uses the exact distribution of
coalescence times described in &quot;The coalescent of a sample from a binary
branching process&quot;, Lambert A., Theor. Pop. Bio. 2018. Tree construction and
formatting uses <code>ape</code> R package <code><a href="ape.html#topic+rtree">ape::rcoal()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleUltraTrees)
</code></pre>


<h3>Format</h3>

<p>A <code>list</code> of objects of class <code>phylo</code>
</p>

<dl>
<dt>edge</dt><dd><p>A matrix of edge connections which reconstruct the tree.</p>
</dd>
<dt>edge.length</dt><dd><p>A numeric vector of the branch lengths of the connections
in <code>edge</code> matrix. Units are years.</p>
</dd>
<dt>tip.label</dt><dd><p>A character vector containing the  (arbitrary in this case)
labels for the 100 tips/samples of the tree.</p>
</dd>
<dt>Nnode</dt><dd><p>Integer number of internal nodes of the tree</p>
</dd>
<dt>params</dt><dd><p><code>data.frame</code> containing info on the params used to generate
the tree</p>
</dd>
</dl>
<p>See ape package for details on class <code>phylo</code> objects.

</p>


<h3>References</h3>

<p>This data set was created for the cloneRate package using
coalescent theory approaches described in &quot;The coalescent of a sample from a
binary branching process&quot;, Lambert A., Theor. Pop. Bio. 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot first of 100 trees
ape::plot.phylo(cloneRate::exampleUltraTrees[[1]],
  direction = "downwards", show.tip.label = FALSE
)
</code></pre>

<hr>
<h2 id='inputCheck'>Check the inputs to growth rate functions</h2><span id='topic+inputCheck'></span>

<h3>Description</h3>

<p>Check the validity of inputs to growth rate fns, making sure the tree is an
ultrametric ape phylo object with reasonable alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inputCheck(tree, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inputCheck_+3A_tree">tree</code></td>
<td>
<p>An ape tree subset to include only the clone of interest</p>
</td></tr>
<tr><td><code id="inputCheck_+3A_alpha">alpha</code></td>
<td>
<p>Used for calculation of confidence intervals. 1-alpha confidence
intervals used with default of alpha = 0.05 (95 percent confidence intervals)</p>
</td></tr>
</table>

<hr>
<h2 id='internalLengths'>Growth rate estimate using the sum of internal lengths</h2><span id='topic+internalLengths'></span>

<h3>Description</h3>

<p><code>internalLengths()</code> provides an estimate for the net growth rate of the clone with confidence bounds, using the internal lengths method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>internalLengths(tree, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="internalLengths_+3A_tree">tree</code></td>
<td>
<p>An ultrametric tree subset to include only the clone of
interest. Alternatively, a list with several such trees.</p>
</td></tr>
<tr><td><code id="internalLengths_+3A_alpha">alpha</code></td>
<td>
<p>Used for calculation of confidence intervals. 1-alpha confidence intervals used with default of alpha = 0.05 (95 percent confidence intervals)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe including the net growth rate estimate, the sum of internal lengths and other important details (clone age estimate, runtime, n, etc.)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maxLikelihood">maxLikelihood()</a></code>, <code><a href="#topic+sharedMuts">sharedMuts()</a></code> for other
growth rate methods.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>internalLengths(cloneRate::exampleUltraTrees[[1]])

</code></pre>

<hr>
<h2 id='longitudinalData'>Longitudinal validation data</h2><span id='topic+longitudinalData'></span>

<h3>Description</h3>

<p>For three individuals with clonal expansions that can be estimated using our methods,
we have longitudinal data to orthogonally validate these estimates, which is included here.
Additionally, for 13 clones with a driver gene matching a driver gene in the single cell
data, but without a match to a specific clone, we include this longitudinal data as well.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>longitudinalData
</code></pre>


<h3>Format</h3>

<p>A <code>data.frame</code> containing all the information needed
</p>

<dl>
<dt>Sample.ID</dt><dd><p>The individual's ID</p>
</dd>
<dt>Age</dt><dd><p>Individual's age at the various sampling times</p>
</dd>
<dt>VAF</dt><dd><p>The variant allele frequency at the various sampling times for the clone of interest</p>
</dd>
<dt>Gene</dt><dd><p>Gene or genes with mutation that identifies the clone</p>
</dd>
<dt>Protein</dt><dd><p>Protein affected by the mutation</p>
</dd>
<dt>cellType</dt><dd><p>The type of cells used for sequencing</p>
</dd>
<dt>cloneName</dt><dd><p>The name we use for the clone to match to single cell data, if applicable.</p>
</dd>
</dl>



<h3>References</h3>

<p>These datasets were generated and annotated in:
<a href="https://pubmed.ncbi.nlm.nih.gov/35058638/">Williams et al. 2022</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/35650444/">Fabre et al. 2022</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot longitudinal data from PD9478
library(ggplot2)
ggplot(longitudinalData[longitudinalData$Sample.ID == "PD9478", ]) +
  geom_point(aes(x = Age, y = VAF))
</code></pre>

<hr>
<h2 id='maxLikelihood'>Growth rate estimate using Maximum Likelihood</h2><span id='topic+maxLikelihood'></span>

<h3>Description</h3>

<p>Uses the approximation that coalescence times H_i are equal to a+b*U_i to
find a and b. b is equal to 1/r, where r is the net growth rate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxLikelihood(tree, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maxLikelihood_+3A_tree">tree</code></td>
<td>
<p>An ultrametric tree subset to include only the clone of
interest. Alternatively, a list with several such trees.</p>
</td></tr>
<tr><td><code id="maxLikelihood_+3A_alpha">alpha</code></td>
<td>
<p>Used for calculation of confidence intervals. 1-alpha confidence intervals used with default of alpha = 0.05 (95 percent confidence intervals)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe including the net growth rate estimate, confidence
intervals, and other important details (clone age estimate, runtime, n, etc.)
</p>


<h3>See Also</h3>

<p><a href="#topic+internalLengths">internalLengths</a> which uses an alternatvie method for
growth rate estimation from an ultrametric tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- maxLikelihood(cloneRate::exampleUltraTrees[[1]])

</code></pre>

<hr>
<h2 id='realCloneData'>Real clone data from human blood</h2><span id='topic+realCloneData'></span>

<h3>Description</h3>

<p>42 clones (39 distinct) from 32 individual donors, 13 of whom have a diagnosis
of Myeloproliferative Neoplasm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(realCloneData)
</code></pre>


<h3>Format</h3>

<p>A <code>list</code> of containing one <code>list</code> with the full ultrametric
trees from 30 of the 32 individual donors (the two from Van Egeren are not included),
and one <code>list</code> containing the 42 clone trees.In three cases, there are two
timepoints from the same clone, and these are separate phylo objects.
Each <code>list</code> contains a tree as a class <code>phylo</code> object.
See ape package documentation for details on class <code>phylo</code> objects.
Names of each <code>phylo</code> object (tree) in the list matches the naming used
in the sources and also includes driver, age, and clone number.
</p>


<h3>References</h3>

<p>These datasets were generated and annotated in:
<a href="https://pubmed.ncbi.nlm.nih.gov/35058638/">Williams et al. 2022</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/35650442/">Mitchell et al. 2022</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/35650444/">Fabre et al. 2022</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/33621486/">Van Egeren et al. 2021</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot full reconstructed tree from donor PD34493
ape::plot.phylo(cloneRate::realCloneData[["fullTrees"]][["PD34493"]],
  direction = "downwards", show.tip.label = FALSE
)

</code></pre>

<hr>
<h2 id='sharedMuts'>Growth rate estimate using the sum of shared mutations assuming a mutation tree</h2><span id='topic+sharedMuts'></span>

<h3>Description</h3>

<p><code>sharedMuts()</code> provides an estimate for the net growth rate of the clone with confidence bounds, using the shared mutations method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sharedMuts(tree, nu = NULL, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sharedMuts_+3A_tree">tree</code></td>
<td>
<p>A non-ultrametric ape tree subset to include only the clone of interest</p>
</td></tr>
<tr><td><code id="sharedMuts_+3A_nu">nu</code></td>
<td>
<p>The mutation rate. If none given, sharedMuts() will first look for a <code>nu</code> column in a <code>metadata</code> data.frame of the tree, and then look for a <code>nu</code> in the tree itself. Will throw error if no <code>nu</code> given or found.</p>
</td></tr>
<tr><td><code id="sharedMuts_+3A_alpha">alpha</code></td>
<td>
<p>Used for calculation of confidence intervals. 1-alpha confidence intervals used with default of alpha = 0.05 (95 percent confidence intervals)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe including the net growth rate estimate, the sum of internal lengths and other important details (clone age estimate, runtime, n, etc.)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+internalLengths">internalLengths()</a></code> which is the ultrametric/time-based analogue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sharedMuts(cloneRate::exampleMutTrees[[1]])

</code></pre>

<hr>
<h2 id='simMut'>Simulate mutation-based birth and death branching trees</h2><span id='topic+simMut'></span>

<h3>Description</h3>

<p>Generates a sampled tree (or a list of many sampled trees) from
a supercritial (birth rate &gt; death rate) birth and death branching process
according to the coalescent point process described in &quot;Lambert, A. The
coalescent of a sample from a binary branching process. (2018).&quot; Edge
lengths will be in units of mutations, assuming poissonian mutation
accumulation. Essentially a wrapper combining simUltra() and ultra2mut()
functions into one step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simMut(
  a,
  b,
  cloneAge,
  n,
  nu,
  nTrees = 1,
  precBits = 1000,
  addStem = FALSE,
  nCores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simMut_+3A_a">a</code></td>
<td>
<p>Birth rate</p>
</td></tr>
<tr><td><code id="simMut_+3A_b">b</code></td>
<td>
<p>Death rate</p>
</td></tr>
<tr><td><code id="simMut_+3A_cloneage">cloneAge</code></td>
<td>
<p>Clone age. Make sure it's same time units as birth and death
rates</p>
</td></tr>
<tr><td><code id="simMut_+3A_n">n</code></td>
<td>
<p>Number of samples/tips of the tree to be returned</p>
</td></tr>
<tr><td><code id="simMut_+3A_nu">nu</code></td>
<td>
<p>Mutation rate in units of mutations per unit time. Make sure time
units are consistent with birth and death rates and cloneAge</p>
</td></tr>
<tr><td><code id="simMut_+3A_ntrees">nTrees</code></td>
<td>
<p>Integer indicating the number of trees to generate. Default is
1.</p>
</td></tr>
<tr><td><code id="simMut_+3A_precbits">precBits</code></td>
<td>
<p>Rmpfr param for handling high precision numbers. Needed to
draw coalescence times.</p>
</td></tr>
<tr><td><code id="simMut_+3A_addstem">addStem</code></td>
<td>
<p>Boolean indicating whether to add stem to tree preceding first
split/coalescence</p>
</td></tr>
<tr><td><code id="simMut_+3A_ncores">nCores</code></td>
<td>
<p>Integer indicating the number of cores to use if parallel pkg
is installed. Default is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ape object of class &quot;phylo&quot; representing the ultrametric
phylogenetic tree with edge lengths in units of time. Tree metadata is
located in the 'metadata' data.frame included in each &quot;phylo&quot; object. If
'nTrees' param is greater than 1, simUltra returns a list of objects of
such objects of class &quot;phylo&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a single mutation-based tree with a specified mutation rate
tree &lt;- simMut(a = 1, b = 0.5, cloneAge = 40, n = 50, nu = 10)

# Generate a list of mutation-based trees with a range of mutation rates
tree_list &lt;- simMut(
  a = 1, b = 0.5, cloneAge = 40, n = 50,
  nu = stats::runif(n = 3, min = 10, max = 20), nTrees = 3
)

</code></pre>

<hr>
<h2 id='simUltra'>Simulate ultrametric birth and death branching trees</h2><span id='topic+simUltra'></span>

<h3>Description</h3>

<p>Generates a sampled tree (or a list of many sampled trees) from
a supercritial (birth rate &gt; death rate) birth and death branching process
according to the coalescent point process described in &quot;Lambert, A. The
coalescent of a sample from a binary branching process. (2018).&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simUltra(
  a,
  b,
  cloneAge,
  n,
  nTrees = 1,
  precBits = 1000,
  addStem = FALSE,
  nCores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simUltra_+3A_a">a</code></td>
<td>
<p>Birth rate or vector of birth rates of length 'nTrees'</p>
</td></tr>
<tr><td><code id="simUltra_+3A_b">b</code></td>
<td>
<p>Death rate or vector of death rates of length 'nTrees'</p>
</td></tr>
<tr><td><code id="simUltra_+3A_cloneage">cloneAge</code></td>
<td>
<p>Clone age or vector of clone ages of length 'nTrees'. Make
sure it's same time units as birth and death rates</p>
</td></tr>
<tr><td><code id="simUltra_+3A_n">n</code></td>
<td>
<p>Number of samples/tips of the tree to be returned. Can be a vector
of length 'nTrees' as well.</p>
</td></tr>
<tr><td><code id="simUltra_+3A_ntrees">nTrees</code></td>
<td>
<p>Integer indicating the number of trees to generate. Default is
1</p>
</td></tr>
<tr><td><code id="simUltra_+3A_precbits">precBits</code></td>
<td>
<p>Rmpfr param for handling high precision numbers. Needed for
drawing the coalescence times. Can be a vector of length 'nTrees', though
it is not recommended</p>
</td></tr>
<tr><td><code id="simUltra_+3A_addstem">addStem</code></td>
<td>
<p>Boolean indicating whether to add stem to tree preceding first
split/coalescence. Can also be a vector of length 'nTrees'</p>
</td></tr>
<tr><td><code id="simUltra_+3A_ncores">nCores</code></td>
<td>
<p>Integer indicating the number of cores to use if parallel pkg
is installed. Default is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ape object of class &quot;phylo&quot; representing the ultrametric
phylogenetic tree with edge lengths in units of time. Tree metadata is
located in the 'metadata' data.frame included in each &quot;phylo&quot; object. If
'nTrees' param is greater than 1, simUltra returns a list of objects of
such objects of class &quot;phylo&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a single tree
tree &lt;- simUltra(a = 1, b = 0.5, cloneAge = 20, n = 50)

# Generate a list of trees
tree_list &lt;- simUltra(a = 1, b = 0.5, cloneAge = 20, n = 50, nTrees = 3)

</code></pre>

<hr>
<h2 id='siteFrequency'>Get site frequency spectrum of a tree</h2><span id='topic+siteFrequency'></span>

<h3>Description</h3>

<p><code>siteFrequency()</code> calculates the site frequency in
units of time or mutations, as well as a normalized frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>siteFrequency(tree, includeStem = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="siteFrequency_+3A_tree">tree</code></td>
<td>
<p>An ultrametric or mutation-based tree subset to include only the
clone of interest. Alternatively, a list with several such trees.</p>
</td></tr>
<tr><td><code id="siteFrequency_+3A_includestem">includeStem</code></td>
<td>
<p>Boolean indicating whether we should count the stem of the
tree as contributing to the site frequency distribution. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with three columns: the number of descendant cells, site
frequency in units of time or mutations, and normalized site frequency. If a
list of trees is input, output will be a list of such data.frames.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+internalLengths">internalLengths()</a></code> and <code><a href="#topic+sharedMuts">sharedMuts()</a></code> which
use the sum of edge lengths ancestral to between 2 and n-1 tips to calculate
a growth rate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get site frequency of a single tree
example.df &lt;- siteFrequency(exampleUltraTrees[[1]])

# Get site frequency of a list of trees
example.list &lt;- siteFrequency(exampleMutTrees)

</code></pre>

<hr>
<h2 id='ultra2mut'>Add poissonian mutations to an ultrametric tree(s)</h2><span id='topic+ultra2mut'></span>

<h3>Description</h3>

<p>Takes an ultrametric tree of class &quot;phylo&quot; (or a list of such
trees) and draws new edge lengths in units of mutations, with the mean of
each new edge length equal to the old edge length multiplied by the mutation
rate. Mutation rate can be set or drawn from a uniform distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ultra2mut(tree, nu)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ultra2mut_+3A_tree">tree</code></td>
<td>
<p>A single tree or list of trees of class &quot;phylo&quot;, with edge
lengths in units of time</p>
</td></tr>
<tr><td><code id="ultra2mut_+3A_nu">nu</code></td>
<td>
<p>Mutation rate in units of mutations per unit time. Can also be a
vector of mutation rates with length equal to the number of input trees.
Make sure time units are consistent in nu and tree$edge.length</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ape object of class &quot;phylo&quot; representing the phylogenetic tree
with edge lengths in units of mutations. Value of mutation rate will be
added to 'metadata' data.frame of output tree if such a data.frame exists
in the input tree. Otherwise, mutation rate value will be added to &quot;phylo&quot;
object directly. If input is a list of trees, ultra2mut() will return a
list of such &quot;phylo&quot; objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Convert the time-based, ultrametric example trees into mutation-based trees
mutTrees &lt;- ultra2mut(exampleUltraTrees,
  nu = stats::runif(n = length(exampleUltraTrees), min = 10, max = 20)
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
