<!DOCTYPE html><html lang="en"><head><title>Help for package ClusteredMutations</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ClusteredMutations}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dissmutmatrix'>
<p>Symmetric dissimilarity mutation matrix.</p></a></li>
<li><a href='#features'>
<p>Describe the features of mutation in the hyper-mutated zones.</p></a></li>
<li><a href='#imd'>
<p>Calculate of the inter-mutational distance (IMD).</p></a></li>
<li><a href='#PD4107a'>
<p>Somatic mutations data set from a primary breast cancer genome.</p></a></li>
<li><a href='#showers'>
<p>Location of clustered mutations in the cancer genome.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-04-28</td>
</tr>
<tr>
<td>Title:</td>
<td>Location and Visualization of Clustered Somatic Mutations</td>
</tr>
<tr>
<td>Depends:</td>
<td>seriation</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification and visualization of groups of closely spaced mutations in the DNA sequence of cancer genome. The extremely mutated zones are searched in the symmetric dissimilarity matrix using the anti-Robinson matrix properties. Different data sets are obtained to describe and plot the clustered mutations information. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-28 15:54:35 UTC; david</td>
</tr>
<tr>
<td>Author:</td>
<td>David Lora [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Lora &lt;david@h12o.es&gt;</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-29 07:44:36</td>
</tr>
</table>
<hr>
<h2 id='dissmutmatrix'>
Symmetric dissimilarity mutation matrix.
</h2><span id='topic+dissmutmatrix'></span>

<h3>Description</h3>

<p>This function computes and returns the Euclidean distance matrix, where each cell represents the distance in base pairs between the chromosomal position of somatic mutations. The matrix can be used to graph the anti-Robinson matrix using the seriation technique (Hahsler and Hornik 2011). Plotting the distance matrix helps to visualize and identify mutation clusters in addition to locating the micro-clustered mutated regions within the macro-clustered mutated zones that occur during the oncogenic process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dissmutmatrix(data = NULL, chr = NULL, position, subset = NULL, upper = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dissmutmatrix_+3A_data">data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td></tr>
<tr><td><code id="dissmutmatrix_+3A_chr">chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td></tr>
<tr><td><code id="dissmutmatrix_+3A_position">position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td></tr>
<tr><td><code id="dissmutmatrix_+3A_subset">subset</code></td>
<td>
<p>: chromosome where the distance between all somatic mutations will be calculated.</p>
</td></tr>
<tr><td><code id="dissmutmatrix_+3A_upper">upper</code></td>
<td>
<p>: logical value indicating whether the upper triangle of the distance matrix should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Color, in the posterior dissimilarity plot, is selected to visually identify hyper-mutated zones (min = 6; max=5000). 
</p>


<h3>Value</h3>

<p><code>dissmutmatrix()</code> returns a symmetric dissimilarity matrix in base 10.
</p>


<h3>Author(s)</h3>

<p>David Lora
</p>


<h3>References</h3>

<p>Hahsler M, Hornik K and Buchta C (2008), Getting things in order: An introduction to the R package seriation. Journal of Statistical Software 25/3. Software available at <a href="http://www.jstatsoft.org/v25/i03/">http://www.jstatsoft.org/v25/i03/</a>.  
</p>
<p>Hahsler M and Hornik K. Dissimilarity plots: A visual exploration tool for partitional clustering. Journal of Computational and Graphical Statistics, 10(2):335&ndash;354, June 2011.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>, 
<code>dissplot</code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(seriation)

###Example 1:
example1&lt;-c(1,101,201,299,301,306,307,317,318,320,418,518,528,628)
10**(dissmutmatrix(position=example1,upper=TRUE))
mut.matrix &lt;- dissmutmatrix(position=example1)
dissplot(mut.matrix,method=NA, 
 options=list(col = c("white","white","orange","orange","red","red","red")))


###Example 2:
###One hypermutated zone with Two hypermutated areas sharing somatic mutations.
example2&lt;-c(1,101,201,299,301,306,307,317,318,320,402,404,406,628)
10**(dissmutmatrix(position=example2,upper=TRUE))
mut.matrix &lt;- dissmutmatrix(position=example2)
dissplot(mut.matrix,method=NA, 
 options=list(col = c("white","white","orange","orange","red","red","red")))


###Example 3:
#data(PD4107a)

###Visualizes a dissimilarity mutation matrix using seriation and matrix shading
###using the method developed by Hahsler and Hornik (2011).
###Chromosome 1;
#mut.matrix &lt;- dissmutmatrix(data=PD4107a,chr=Chr,position=Position,subset=1)
#dissplot(mut.matrix, method=NA, options=list( col = 
# c("black","navy","blue","cyan","green","yellow","orange","red",
#	"darkred","darkred","white")))

###Chromosome 6;
#mut.matrix &lt;- dissmutmatrix(data=PD4107a,chr=Chr,position=Position,subset=6)
#dissplot(mut.matrix, method=NA, options=list( col = 
# c("black","navy","blue","cyan","green","yellow","orange","red","darkred",
#	"darkred","white")))

###Chromosome 12;
#mut.matrix &lt;- dissmutmatrix(data=PD4107a,chr=Chr,position=Position,subset=12)
#dissplot(mut.matrix, method=NA, options=list( col = 
# c("black","navy","blue","cyan","green","yellow","orange","red","darkred",
#   "darkred","white")))
</code></pre>

<hr>
<h2 id='features'>
Describe the features of mutation in the hyper-mutated zones.
</h2><span id='topic+features'></span>

<h3>Description</h3>

<p>Several features were observed in the hyper-mutated zones, for example, kataegis is the proposed name for the hyper-mutated zones with a cluster of C&gt;T and/or C&gt;G mutations that are substantially enriched at TpCpN trinucleotides, on the same DNA strand and that co-localize with large-scale genomic structural variation (Alexandrov et al. 2013; Nik-Zainal et al. 2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>features(data = NULL, chr = NULL, position, refbase, mutantbase,
						 min = 6, max = 5000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="features_+3A_data">data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td></tr>
<tr><td><code id="features_+3A_chr">chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td></tr>
<tr><td><code id="features_+3A_position">position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td></tr>
<tr><td><code id="features_+3A_refbase">refbase</code></td>
<td>
<p>: reference base in the chromosome.</p>
</td></tr>
<tr><td><code id="features_+3A_mutantbase">mutantbase</code></td>
<td>
<p>: the mutant base in the chromosome.</p>
</td></tr>
<tr><td><code id="features_+3A_min">min</code></td>
<td>
<p>: a number <code>min</code> or more consecutive mutations.</p>
</td></tr>
<tr><td><code id="features_+3A_max">max</code></td>
<td>
<p>: a distance less than or equal to a number <code>max</code> of bp.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, <code>features()</code> identifies the mutations in the hyper-mutated zones (min = 6; max=5000). Complex mutations (Roberts et al. 2012; Roberts et al. 2013) are those segments containing &gt;= 2 consecutive mutations with a distance =&lt; 100 bp.
</p>


<h3>Value</h3>

<p><code>features()</code> returns a data set with all mutations in the hyper-mutated zones. The data set contains five variables: 
</p>
<table role = "presentation">
<tr><td><code>clustered</code></td>
<td>
<p>: number of cluster.</p>
</td></tr>
<tr><td><code>chr</code></td>
<td>
<p>: chromosome.</p>
</td></tr>
<tr><td><code>position</code></td>
<td>
<p>: the position of mutation in the chromosome.</p>
</td></tr>
<tr><td><code>ref_base</code></td>
<td>
<p>: reference base in the chromosome.</p>
</td></tr>
<tr><td><code>mutant_base</code></td>
<td>
<p>: the mutant base in the chromosome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Lora
</p>


<h3>References</h3>

<p>Alexandrov LB, Nik-Zainal S, Wedge DC, et al. Signatures of mutational processes in human cancer. Nature. 2013 Aug 22;500(7463):415-21.
</p>
<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Roberts SA, Sterling J, Thompson C, et al. Clustered mutations in  yeast and in human cancers can arise from damaged long single-strand DNA regions. Mol Cell. 2012 May 25;46(4):424-35.
</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC cytidine deaminase mutagenesis pattern is widespread in human cancers. Nat Genet. 2013 Sep;45(9):970-6. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PD4107a)
###Show the features of mutations in the hypermutated zones;
features(data=PD4107a,chr=Chr,position=Position,refbase=Ref_base,
		mutantbase=Mutant_base)

###Locate complex mutations in the hypermutated zones;
kataegis&lt;-features(data=PD4107a,chr=Chr,position=Position,refbase=Ref_base,
		mutantbase=Mutant_base)
complex &lt;-features(data=PD4107a,chr=Chr,position=Position,
			refbase=Ref_base,mutantbase=Mutant_base,min=2,max=10)
sub.complex &lt;-subset(complex,select=c(clustered,chr,position))
data.hyperm  &lt;-merge(kataegis,sub.complex,by=c("chr","position"),all.x=TRUE) 

###Summary of the number of complex mutations in the hypermutated zones;
table(data.hyperm$clustered.x)
table(data.hyperm$clustered.y)
table(data.hyperm$clustered.y,data.hyperm$clustered.x)
data.hyperm&lt;-transform(data.hyperm,clustered = 
				ifelse(is.na(clustered.y)==FALSE,1,0))
table(data.hyperm$clustered.x)
table(data.hyperm$clustered,data.hyperm$clustered.x)
###All hypermutated zones had more than 6 mutations (without complex mutations).
</code></pre>

<hr>
<h2 id='imd'>
Calculate of the inter-mutational distance (IMD).
</h2><span id='topic+imd'></span>

<h3>Description</h3>

<p>The inter-mutational distance (IMD) is the distance between each somatic substitution and the substitution immediately prior (Nik-Zainal et al. 2012). The <code>imd()</code> is used to graph the rainfall plot (Nik-Zainal et al. 2012). This graph localizes the regional clustering of mutations and represents the IMD on the y-axis on a log base 10 scale, with mutations ranked and ordered on the x-axis from the first variant on the short arm of chromosome 1 to the last variant on the long arm of chromosome X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imd(data = NULL, chr = NULL, position, extra = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="imd_+3A_data">data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td></tr>
<tr><td><code id="imd_+3A_chr">chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td></tr>
<tr><td><code id="imd_+3A_position">position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td></tr>
<tr><td><code id="imd_+3A_extra">extra</code></td>
<td>
<p>: additional information to posterior graph.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>imd()</code> returns a data set with six variables: 
</p>
<table role = "presentation">
<tr><td><code>chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td></tr>
<tr><td><code>position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td></tr>
<tr><td><code>extra</code></td>
<td>
<p>: additional information.</p>
</td></tr>
<tr><td><code>number</code></td>
<td>
<p>: ordered mutations.</p>
</td></tr>
<tr><td><code>distance</code></td>
<td>
<p>: Euclidean distance between each somatic substitution and the substitution immediately prior (IMD) (Nik-Zainal et al. 2012).</p>
</td></tr>
<tr><td><code>log10distance</code></td>
<td>
<p>: Euclidean distance between each somatic substitution and the substitution immediately prior in base 10.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Lora
</p>


<h3>References</h3>

<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Wickham H. ggplot2: elegant graphics for data analysis. Springer New York, 2009.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+plot">plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PD4107a)

###New variable
extra &lt;- factor(c(),levels=c("T&gt;C","T&gt;G","T&gt;A","C&gt;T","C&gt;G","C&gt;A"))
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="G"]&lt;-"T&gt;C"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="C"]&lt;-"T&gt;C"
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="C"]&lt;-"T&gt;G"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="G"]&lt;-"T&gt;G"
extra[PD4107a$Ref_base=="A" &amp; PD4107a$Mutant_base=="T"]&lt;-"T&gt;A"
extra[PD4107a$Ref_base=="T" &amp; PD4107a$Mutant_base=="A"]&lt;-"T&gt;A"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="A"]&lt;-"C&gt;T"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="T"]&lt;-"C&gt;T"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="C"]&lt;-"C&gt;G"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="G"]&lt;-"C&gt;G"
extra[PD4107a$Ref_base=="G" &amp; PD4107a$Mutant_base=="T"]&lt;-"C&gt;A"
extra[PD4107a$Ref_base=="C" &amp; PD4107a$Mutant_base=="A"]&lt;-"C&gt;A"
PD4107a$extra&lt;-extra

###Generate new data set with intermutational distance;
rainfall&lt;-imd(data=PD4107a,chr=Chr,position=Position,extra=extra)

###Rainfall plot for PD4107a cancer sample;
plot(rainfall$number, rainfall$log10distance,pch=20,
	ylab="Intermutation distance (bp)",xlab="PD4107a",yaxt="n",
	col=c(rep(c("black","red"),14)[rainfall$chr]))
axis(2, at=c(0,1,2,3,4,6), labels=c("1","10","100","1000","10000","1000000"),
	las=2, cex.axis=0.6)

###Rainfall plot for PD4107a cancer sample (Nik-Zainal et al. 2012);
#require(ggplot2)
#graph &lt;- qplot(data=rainfall,number,log10distance,colour=extra, ylim=c(0.2,8), 
#	ylab="log10", xlab="PD4107a")
#graph &lt;- graph + 
#	scale_colour_manual(values =c("T&gt;C"="yellow","T&gt;G"="green","T&gt;A"="pink",
#		"C&gt;T"="red","C&gt;G"="black","C&gt;A"="blue"))
#graph &lt;- graph + theme(legend.title=element_blank())
#graph &lt;- graph + scale_y_continuous(breaks = c(0, 1, 2, 3, 4, 6),
#		labels=c("1","10","100","1000","10000","1000000"))
#graph
</code></pre>

<hr>
<h2 id='PD4107a'>
Somatic mutations data set from a primary breast cancer genome.
</h2><span id='topic+PD4107a'></span>

<h3>Description</h3>

<p>PD4107a is a data set of somatic substitution mutations from a primary breast cancer whole genome with a germline mutation in BRCA1 (Nik-Zainal et al. 2012). The data set contains five variables: sample name, chromosome where the somatic mutation is located, location of the somatic mutation, the reference base and the mutated base. 
</p>
<p>The complete set of somatic mutations from a patient with breast cancer (PD4107a) was provided by the Cancer Genome Project group at the Wellcome Trust Sanger Institute (Alexandrov et al. 2013). Mutations with Indel labels were deleted (only subs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PD4107a)</code></pre>


<h3>Format</h3>

<p>A data frame with 9879 observations on the following 5 variables.
</p>

<dl>
<dt><code>Sample_id</code></dt><dd><p>: PD4107a.</p>
</dd>
<dt><code>Chr</code></dt><dd><p>: From chromosome 1 to chromosome X.</p>
</dd>
<dt><code>Position</code></dt><dd><p>: Mutation locations on the chromosome.</p>
</dd>
<dt><code>Ref_base</code></dt><dd><p>: The reference base in the mutation locations.</p>
</dd>
<dt><code>Mutant_base</code></dt><dd><p>: The mutated base in the mutation locations.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Patient PD4107a has been described throughout the scientific literature (Alexandrov et al 2013; Fischer et al 2013; Muino et al 2014; Nik-Zainal et al 2012; Roberts et al 2013). 
</p>


<h3>Source</h3>

<p><a href="ftp://ftp.sanger.ac.uk/pub/cancer/AlexandrovEtAl/somatic_mutation_data/Breast/Breast_clean_somatic_mutations_for_signature_analysis.txt">ftp://ftp.sanger.ac.uk/pub/cancer/AlexandrovEtAl/somatic_mutation_data/Breast/Breast_clean_somatic_mutations_for_signature_analysis.txt</a>
</p>


<h3>References</h3>

<p>Alexandrov LB, Nik-Zainal S, Wedge DC, et al. Signatures of mutational processes in human cancer. Nature. 2013 Aug 22;500(7463):415-21.
</p>
<p>Hahsler M and Hornik K. Dissimilarity plots: A visual exploration tool for partitional clustering. Journal of Computational and Graphical Statistics, 10(2):335&ndash;354, June 2011.
</p>
<p>Fischer A, Illingworth CJ, Campbell PJ, et al. EMu: probabilistic inference of mutational processes and their localization in the cancer genome. Genome Biol. 2013 Apr 29;14(4):R39.
</p>
<p>Muino JM, Kuruoglu EE, Arndt PF. Evidence of a cancer type-specific distribution for consecutive somatic mutation distances. Comput Biol Chem. 2014 Aug 23. pii: S1476-9271(14)00091-7.
</p>
<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC cytidine deaminase mutagenesis pattern is widespread in human cancers. Nat Genet. 2013 Sep;45(9):970-6. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PD4107a)

###PD4107a data set;
head(PD4107a,12)

###generate new data set with intermutational distance;
#rainfall&lt;-imd(data=PD4107a,chr=Chr,position=Position)
###Rainfall plot for PD4107a cancer sample;
#plot(rainfall$number, rainfall$log10distance,pch=20,
#	ylab="Intermutation distance (bp)",xlab="PD4107a",yaxt="n", 
#	col=c(rep(c("black","red"),14)[rainfall$chr]))
#axis(2, at=c(0,1,2,3,4,6), labels=c("1","10","100","1000","10000","1000000"),
#	las=2, cex.axis=0.6)

###Locate the clustered mutations;
#showers(data=PD4107a,chr=Chr,position=Position)

###Visualizes a dissimilarity mutation matrix using seriation and matrix shading 
### using the method developed by Hahsler and Hornik (2011).
###Chromosome 6;
#mut.matrix &lt;- dissmutmatrix(data=PD4107a,chr=Chr,position=Position,subset=6)
#dissplot(mut.matrix, method=NA, options=list( col = 
#  c("black","navy","blue","cyan","green","yellow","orange","red",
#	"darkred","darkred","white")))
</code></pre>

<hr>
<h2 id='showers'>
Location of clustered mutations in the cancer genome.
</h2><span id='topic+showers'></span>

<h3>Description</h3>

<p><code>showers()</code> identifies all groups of closely spaced mutations using the anti-Robinson matrix. Hyper-mutated regions are defined as those segments containing a number (min = 6) or more mutations with a distance that is less than or equal to a number (max=1000) of bp, and referred to as mutation showers (Drake 2007a; Wang et al. 2007), clustered mutations (Drake 2007a; Drake 2007b; Roberts et al. 2012), or kataegis (from the Greek word for shower or thunderstorm) (Alexandrov et al. 2013; Nik-Zainal et al. 2012). <code>showers()</code> can be used to locate complex mutations (Roberts et al. 2012; Roberts et al. 2013) (min = 2; max=10). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showers(data = NULL, chr = NULL, position, min = 6, max = 5000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="showers_+3A_data">data</code></td>
<td>
<p>: somatic substitution mutations of the cancer genome data set.</p>
</td></tr>
<tr><td><code id="showers_+3A_chr">chr</code></td>
<td>
<p>: chromosome where the somatic mutation is located.</p>
</td></tr>
<tr><td><code id="showers_+3A_position">position</code></td>
<td>
<p>: position of somatic mutations in the DNA sequence of the cancer genome.</p>
</td></tr>
<tr><td><code id="showers_+3A_min">min</code></td>
<td>
<p>: a number <code>min</code> of consecutive mutations.</p>
</td></tr>
<tr><td><code id="showers_+3A_max">max</code></td>
<td>
<p>: a distance less than or equal to a number <code>max</code> of bp.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, <code>showers()</code> identifies the hyper-mutated zones (min = 6; max=5000). Complex mutations are those segments containing &gt;= 2 consecutive mutations with a distance =&lt; 100 bp.
</p>


<h3>Value</h3>

<p><code>showers()</code> returns a data set with all hyper-mutated zones in the DNA sequence of tumor cells.
</p>
<table role = "presentation">
<tr><td><code>chr</code></td>
<td>
<p>: the shower mutations data set contains seven variables: chromosome.</p>
</td></tr>
<tr><td><code>pend</code></td>
<td>
<p>: the last position in the chromosome of the mutation shower.</p>
</td></tr>
<tr><td><code>pstart</code></td>
<td>
<p>: the first position in the chromosome of the mutation shower.</p>
</td></tr>
<tr><td><code>nend</code></td>
<td>
<p>: the last number of a consecutive mutation shower.</p>
</td></tr>
<tr><td><code>nstart</code></td>
<td>
<p>: the first number of a consecutive mutation shower.</p>
</td></tr>
<tr><td><code>distance</code></td>
<td>
<p>: the length of a hyper-mutated zone and the number of mutations in the clustered mutation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Lora.
</p>


<h3>References</h3>

<p>Alexandrov LB, Nik-Zainal S, Wedge DC, et al. Signatures of mutational processes in human cancer. Nature. 2013 Aug 22;500(7463):415-21.
</p>
<p>Drake JW. Mutations in clusters and showers. Proc Natl Acad Sci U S A. 2007 May 15;104(20):8203-4. 
</p>
<p>Drake JW. Too many mutants with multiple mutations. Crit Rev Biochem Mol Biol. 2007 Jul-Aug;42(4):247-58.
</p>
<p>Nik-Zainal S, Alexandrov LB, Wedge DC, et al; Breast Cancer Working Group of the International Cancer Genome Consortium. Mutational processes molding the genomes of 21 breast cancers. Cell. 2012 May 25;149(5):979-93.
</p>
<p>Roberts SA, Sterling J, Thompson C, et al. Clustered mutations in  yeast and in human cancers can arise from damaged long single-strand DNA regions. Mol Cell. 2012 May 25;46(4):424-35.
</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC cytidine deaminase mutagenesis pattern is widespread in human cancers. Nat Genet. 2013 Sep;45(9):970-6. 
</p>
<p>Wang J, Gonzalez KD, Scaringe WA, et al. Evidence for mutation showers. Proc Natl Acad Sci U S A. 2007 May 15;104(20):8403-8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###Example 1:
example1&lt;-c(1,101,201,299,301,306,307,317,318,320,418,518,528,628)
showers(position=example1, min=5, max=100)

###Example 2:
example2&lt;-c(1,101,201,299,301,306,307,317,318,320,402,404,406,628)
showers(position=example2, min=5, max=100)

###Example 3:
#data(PD4107a)

###Locate the clustered mutations;
#showers(data=PD4107a,chr=Chr,position=Position)

###Locate the complex mutations;
#complex.showers&lt;-showers(data=PD4107a,chr=Chr,position=Position,min=2,max=10)
#nrow(complex.showers)
#table(complex.showers$chr)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
