<!DOCTYPE html><html lang="en"><head><title>Help for package clusterRepro</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {clusterRepro}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clusterRepro'><p>Gene expression clusters reproducibility and validation</p></a></li>
<li><a href='#clusterRepro-internal'><p>clusterRepro Internal Functions</p></a></li>
<li><a href='#IGP.clusterRepro'><p>In-group proportion calculation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-10-10</td>
</tr>
<tr>
<td>Title:</td>
<td>Reproducibility of Gene Expression Clusters</td>
</tr>
<tr>
<td>Author:</td>
<td>Amy Kapp &lt;Amy_Kapp@hotmail.com&gt; and Rob Tibshirani &lt;tibs@stanford.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rob Tibshirani &lt;tibs@stanford.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.2.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a function for validating microarray clusters via reproducibility,
 based on the paper referenced below.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.ncbi.nlm.nih.gov/pubmed/16613834.">https://www.ncbi.nlm.nih.gov/pubmed/16613834.</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-10-15 17:17:55 UTC; tibs</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-15 18:20:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='clusterRepro'>Gene expression clusters reproducibility and validation</h2><span id='topic+clusterRepro'></span>

<h3>Description</h3>

<p>Validate gene expression clusters by determining whether or not they
are reproducible
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterRepro(Centroids, New.data, Number.of.permutations)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clusterRepro_+3A_centroids">Centroids</code></td>
<td>
<p>The matrix of centroids with annotated rows.  The
labeled rows are either genes (for gene clusters) or samples (for
sample clusters) and the columns are the centroids.</p>
</td></tr>
<tr><td><code id="clusterRepro_+3A_new.data">New.data</code></td>
<td>
<p>The matrix of gene expression data (with annotated
rows) independent of the dataset used to form the centroids.  For
gene clusters, the rows are samples and the columns are genes.  For
sample clusters, the rows are genes and the columns are samples.</p>
</td></tr>
<tr><td><code id="clusterRepro_+3A_number.of.permutations">Number.of.permutations</code></td>
<td>
<p>The number of times the centroids will
be permuted to generate the null distribution.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function looks for gene expression clusters found in one
dataset in another independent dataset.  The centroids from the first
dataset are used to classify the independent data and the
corresponding in-group proportions (IGPs) are computed.  These in-group
proportions are compared to null distributions of in-group proportions
to produce p-values.  The IGP null distributions are generated by
repeatedly permuting the centroids within the box aligned with the principal
components, classifying the independent data, and calculating the
corresponding IGPs.</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Actual.Size</code></td>
<td>
<p>The number of columns of New.data assigned to each centroid.</p>
</td></tr>
<tr><td><code>Actual.IGP</code></td>
<td>
<p>The in-group proportions of the groups formed when
New.data is classified using Centroids.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values for each of the groups represented by the centroids.</p>
</td></tr>
<tr><td><code>Number</code></td>
<td>
<p>The number of permutations used to compute the
corresponding p-value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Amy Kapp and Robert Tibshirani</p>


<h3>References</h3>

<p>Amy Kapp and Robert Tibshirani.  Are clusters in one dataset present
in another dataset?  To be published.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IGP.clusterRepro">IGP.clusterRepro</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Generate centroids with annotated rows
Centroids &lt;- matrix(rnorm(30, sd = 10), 10)
rownames(Centroids) &lt;- letters[1:nrow(Centroids)]

### Generate data with annotated rows
Data &lt;- cbind(matrix(rep(Centroids[,1], 10), 10),
matrix(rep(Centroids[,2], 15), 10), matrix(rep(Centroids[,3], 20), 10))
Data &lt;- Data + matrix(rnorm(length(Data), sd = 10), nrow(Data))
rownames(Data) &lt;- letters[1:nrow(Data)]

### Classify the data and calculate the corresponding in-group
### proportions and group size
Result &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1)
Result$Actual.IGP
Result$Actual.Size

### Generate null distributions and compare to actual in-group proportions to obtain p-values
Result2 &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1000)

### If the number of rows in the centroid matrix does not match the
### number of rows in the data matrix and the row labels are unique, this
### function will only use the rows that the two matrices have in common. 
Data &lt;- matrix(rnorm(200), 20)
rownames(Data) &lt;- letters[(nrow(Data)+6):7]
Result &lt;- IGP.clusterRepro(Data, Centroids)
Result2 &lt;- clusterRepro(Centroids, Data, Number.of.permutations = 1000)
</code></pre>

<hr>
<h2 id='clusterRepro-internal'>clusterRepro Internal Functions</h2><span id='topic+permuteCol'></span>

<h3>Description</h3>

<p>This function randomly permutes the entries of each of the matrix's
columns independently of the other columns. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permuteCol(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clusterRepro-internal_+3A_x">x</code></td>
<td>
<p>matrix of at least one row and one column</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is not to be be called by the user.
</p>


<h3>Value</h3>

<p>The matrix of permuted columns is returned.
</p>


<h3>Author(s)</h3>

<p>Trevor Hastie
</p>

<hr>
<h2 id='IGP.clusterRepro'>In-group proportion calculation</h2><span id='topic+IGP.clusterRepro'></span>

<h3>Description</h3>

<p>This function classifies gene expression (microarray) data using centroids and
calculates the in-group proportions of the resulting groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IGP.clusterRepro(Data, Centroids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IGP.clusterRepro_+3A_data">Data</code></td>
<td>
<p>The matrix of gene expression data with annotated rows.  For gene clusters, the
labeled rows are samples and the columns are genes.  For sample clusters,
the labeled rows are genes and the columns are samples.</p>
</td></tr> 
<tr><td><code id="IGP.clusterRepro_+3A_centroids">Centroids</code></td>
<td>
<p>The matrix of centroids with annotated rows.  The
labeled rows are either genes (for gene clusters) or samples (for
sample clusters) and the columns are the centroids.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Pearson's centered correlation coefficient between each datum and
each centroid is calculated.  The datum is then classified to the
group whose centroid had the highest correlation with the datum.  The
in-group proportion is defined to be the proportion of data in a group
whose nearest neighbors (Pearson's centered correlation) are also
classified to the same group.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Class</code></td>
<td>
<p>The data classification made using the centroids</p>
</td></tr>
<tr><td><code>IGP</code></td>
<td>
<p>The in-group proportions for the groups found in the data</p>
</td></tr>
<tr><td><code>Size</code></td>
<td>
<p>The number of data classified to each of the groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Amy Kapp and Robert Tibshirani
</p>


<h3>References</h3>

<p>Amy Kapp and Robert Tibshirani.  Are clusters in one dataset present
in another dataset?  To be published.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Make centroid matrix with annotated rows
C &lt;- matrix(rnorm(30), 10)
rownames(C) &lt;- letters[1:nrow(C)]

### Make data matrix with annotated rows
D &lt;- matrix(rnorm(100), 10)
rownames(D) &lt;- letters[1:nrow(C)]

### Classify data and calculate in-group proportions
Result &lt;- IGP.clusterRepro(D, C)

### If the number of rows in the centroid matrix does not match the
### number of rows in the data matrix and the row labels are unique, this
### function will only use the rows that the two matrices have in common.
D &lt;- matrix(rnorm(200), 20)
rownames(D) &lt;- letters[(nrow(D)+6):7]
Result &lt;- IGP.clusterRepro(D, C)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
