<!DOCTYPE html><html><head><title>Help for package ClusTCR2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ClusTCR2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ClusTCR'><p>Creates ClusTCR matrix</p>
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene</a></li>
<li><a href='#ClusTCR_Large'><p>Creates ClusTCR matrix</p>
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene</a></li>
<li><a href='#ggnet2'><p>Copied code from ggnet's ggnet2 function</p></a></li>
<li><a href='#mcl_cluster'><p>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function</p></a></li>
<li><a href='#mcl_cluster_large'><p>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function</p></a></li>
<li><a href='#Motif_from_cluster_file'><p>Code for plotting the Motif based on a specific CDR3 length and V gene (see netplot_ClusTCR2 for details).</p></a></li>
<li><a href='#motif_plot'><p>Code for plotting the Motif based on a specific CDR3 length and V gene (see netplot_ClusTCR2 for ).</p></a></li>
<li><a href='#motif_plot_large'><p>Code for plotting the Motif based on a specific CDR3 length and V gene (see netplot_ClusTCR2 for details).</p></a></li>
<li><a href='#netplot_ClusTCR2'><p>Code for displaying the network.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Identifying Similar T Cell Receptor Hyper-Variable Sequences
with 'ClusTCR2'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.3.01</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-05-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Kerry A. Mullan [aut, cre],
  Sebastiaan Valkiers [aut, ctb],
  Kris Laukens [aut, ctb],
  Pieter Meysman [aut, ctb]</td>
</tr>
<tr>
<td>Description:</td>
<td>Enhancing T cell receptor (TCR) sequence analysis, 'ClusTCR2', based on 'ClusTCR' python program, leverages Hamming distance to compare the complement-determining region three (CDR3) sequences for sequence similarity, variable gene (V gene) and length. The second step employs the Markov Cluster Algorithm to identify clusters within an undirected graph, providing a summary of amino acid motifs and matrix for generating network plots. Tailored for single-cell RNA-seq data with integrated TCR-seq information, 'ClusTCR2' is integrated into the Single Cell TCR and Expression Grouped Ontologies (STEGO) R application or 'STEGO.R'. See the two publications for more details. Sebastiaan Valkiers, Max Van Houcke, Kris Laukens, Pieter Meysman (2021) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab446">doi:10.1093/bioinformatics/btab446</a>&gt;, Kerry A. Mullan, My Ha, Sebastiaan Valkiers, Nicky de Vrij, Benson Ogunjimi, Kris Laukens, Pieter Meysman (2023) &lt;<a href="https://doi.org/10.1101%2F2023.09.27.559702">doi:10.1101/2023.09.27.559702</a>&gt;.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kerry A. Mullan &lt;Kerry.Mullan@uantwerpen.be&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>DescTools, ggplot2, ggseqlogo, network, plyr, RColorBrewer,
stringr, scales, sna, VLF</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GeneTarget, SingleCell</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-15 16:22:16 UTC; kerrymullan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-16 15:00:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='ClusTCR'>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene</h2><span id='topic+ClusTCR'></span>

<h3>Description</h3>

<p>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusTCR(my_file, allele = NULL, v_gene = "v_call")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClusTCR_+3A_my_file">my_file</code></td>
<td>
<p>uploaded file with junction_aa (CD3 sequences), variable gene.</p>
</td></tr>
<tr><td><code id="ClusTCR_+3A_allele">allele</code></td>
<td>
<p>The allele, if present as *00 will be removed if the user requires it.</p>
</td></tr>
<tr><td><code id="ClusTCR_+3A_v_gene">v_gene</code></td>
<td>
<p>Variable gene column name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>X by Y matrix of structurally related CDR3 sequences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage of ClusTCR function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv", package = "ClusTCR2"))
# Perform clustering using ClusTCR function
step1 &lt;- ClusTCR(example_file, allele = FALSE)
# Print the result
print(step1)
</code></pre>

<hr>
<h2 id='ClusTCR_Large'>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene</h2><span id='topic+ClusTCR_Large'></span>

<h3>Description</h3>

<p>Creates ClusTCR matrix
This function identifies similar CDR3 amino acid sequences based on the same length and V_gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusTCR_Large(my_file, allele = NULL, v_gene = "v_call")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClusTCR_Large_+3A_my_file">my_file</code></td>
<td>
<p>uploaded file with junction_aa (CD3 sequences), variable gene.</p>
</td></tr>
<tr><td><code id="ClusTCR_Large_+3A_allele">allele</code></td>
<td>
<p>The allele, if present as *00 will be removed if the user requires it.</p>
</td></tr>
<tr><td><code id="ClusTCR_Large_+3A_v_gene">v_gene</code></td>
<td>
<p>Variable gene column name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>X by Y matrix of structurally related CDR3 sequences.
</p>

<hr>
<h2 id='ggnet2'>Copied code from ggnet's ggnet2 function</h2><span id='topic+ggnet2'></span>

<h3>Description</h3>

<p>Copied code from ggnet's ggnet2 function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggnet2(
  net,
  mode = "fruchtermanreingold",
  layout.par = NULL,
  layout.exp = 0,
  alpha = 1,
  color = "grey75",
  shape = 19,
  size = 9,
  max_size = 9,
  na.rm = NA,
  palette = NULL,
  alpha.palette = NULL,
  alpha.legend = NA,
  color.palette = palette,
  color.legend = NA,
  shape.palette = NULL,
  shape.legend = NA,
  size.palette = NULL,
  size.legend = NA,
  size.zero = FALSE,
  size.cut = FALSE,
  size.min = NA,
  size.max = NA,
  label = FALSE,
  label.alpha = 1,
  label.color = "black",
  label.size = max_size/2,
  label.trim = FALSE,
  node.alpha = alpha,
  node.color = color,
  node.label = label,
  node.shape = shape,
  node.size = size,
  edge.alpha = 1,
  edge.color = "grey50",
  edge.lty = "solid",
  edge.size = 0.25,
  edge.label = NULL,
  edge.label.alpha = 1,
  edge.label.color = label.color,
  edge.label.fill = "white",
  edge.label.size = max_size/2,
  arrow.size = 0,
  arrow.gap = 0,
  arrow.type = "closed",
  legend.size = 9,
  legend.position = "right",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggnet2_+3A_net">net</code></td>
<td>
<p>net plot from step 2.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_mode">mode</code></td>
<td>
<p>= &quot;fruchtermanreingold&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_layout.par">layout.par</code></td>
<td>
<p>= NULL,</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_layout.exp">layout.exp</code></td>
<td>
<p>= 0</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha">alpha</code></td>
<td>
<p>= 1</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color">color</code></td>
<td>
<p>= &quot;grey75&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape">shape</code></td>
<td>
<p>= 19</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size">size</code></td>
<td>
<p>= 9</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_max_size">max_size</code></td>
<td>
<p>= 9</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_na.rm">na.rm</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_palette">palette</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha.palette">alpha.palette</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_alpha.legend">alpha.legend</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color.palette">color.palette</code></td>
<td>
<p>= palette</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_color.legend">color.legend</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape.palette">shape.palette</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_shape.legend">shape.legend</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.palette">size.palette</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.legend">size.legend</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.zero">size.zero</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.cut">size.cut</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.min">size.min</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_size.max">size.max</code></td>
<td>
<p>= NA</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label">label</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.alpha">label.alpha</code></td>
<td>
<p>= 1</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.color">label.color</code></td>
<td>
<p>= &quot;black&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.size">label.size</code></td>
<td>
<p>= max_size/2</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_label.trim">label.trim</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.alpha">node.alpha</code></td>
<td>
<p>see <code>alpha</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.color">node.color</code></td>
<td>
<p>see <code>color</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.label">node.label</code></td>
<td>
<p>see <code>label</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.shape">node.shape</code></td>
<td>
<p>see <code>shape</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_node.size">node.size</code></td>
<td>
<p>see <code>size</code></p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.alpha">edge.alpha</code></td>
<td>
<p>= 1</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.color">edge.color</code></td>
<td>
<p>the color of the edges, as a color value, a vector of color
values, or as an edge attribute containing color values.
Defaults to <code>"grey50"</code>.</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.lty">edge.lty</code></td>
<td>
<p>= &quot;solid&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.size">edge.size</code></td>
<td>
<p>= 0.25</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label">edge.label</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.alpha">edge.label.alpha</code></td>
<td>
<p>= 1</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.color">edge.label.color</code></td>
<td>
<p>= label.color</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.fill">edge.label.fill</code></td>
<td>
<p>= &quot;white&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_edge.label.size">edge.label.size</code></td>
<td>
<p>= max_size/2</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.size">arrow.size</code></td>
<td>
<p>= 0</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.gap">arrow.gap</code></td>
<td>
<p>= 0</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_arrow.type">arrow.type</code></td>
<td>
<p>= &quot;closed&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_legend.size">legend.size</code></td>
<td>
<p>= 9</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_legend.position">legend.position</code></td>
<td>
<p>= &quot;right&quot;</p>
</td></tr>
<tr><td><code id="ggnet2_+3A_...">...</code></td>
<td>
<p>Other functions in ggplot2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object displaying the network plot.
</p>

<hr>
<h2 id='mcl_cluster'>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function</h2><span id='topic+mcl_cluster'></span>

<h3>Description</h3>

<p>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcl_cluster(my_file, max.iter = 10, inflation = 1, expansion = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcl_cluster_+3A_my_file">my_file</code></td>
<td>
<p>Matrix file produce from <a href="#topic+ClusTCR">ClusTCR</a></p>
</td></tr>
<tr><td><code id="mcl_cluster_+3A_max.iter">max.iter</code></td>
<td>
<p>Number of iterations to find the steady state of MCL.</p>
</td></tr>
<tr><td><code id="mcl_cluster_+3A_inflation">inflation</code></td>
<td>
<p>numeric value</p>
</td></tr>
<tr><td><code id="mcl_cluster_+3A_expansion">expansion</code></td>
<td>
<p>numeric value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li><p> 'Cluster_lab': Data frame containing information about the clusters
</p>
</li>
<li><p> 'Normalised_tabel': Normalized table used in the clustering process
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Example usage of mcl_cluster function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv",package = "ClusTCR2"))
# Perform clustering using mcl_cluster function
step1 &lt;- ClusTCR(example_file,allele = FALSE)
# perform mcl
step2 &lt;- mcl_cluster(step1)
</code></pre>

<hr>
<h2 id='mcl_cluster_large'>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function</h2><span id='topic+mcl_cluster_large'></span>

<h3>Description</h3>

<p>Create the files for labeling the linked clusters from ClusTCR_list_to_matrix function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcl_cluster_large(my_file, max.iter = 10, inflation = 1, expansion = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcl_cluster_large_+3A_my_file">my_file</code></td>
<td>
<p>Matrix file produce from <a href="#topic+ClusTCR">ClusTCR</a></p>
</td></tr>
<tr><td><code id="mcl_cluster_large_+3A_max.iter">max.iter</code></td>
<td>
<p>Number of iterations to find the steady state of MCL.</p>
</td></tr>
<tr><td><code id="mcl_cluster_large_+3A_inflation">inflation</code></td>
<td>
<p>numeric value</p>
</td></tr>
<tr><td><code id="mcl_cluster_large_+3A_expansion">expansion</code></td>
<td>
<p>numeric value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li><p> 'Cluster_lab': Data frame containing information about the clusters
</p>
</li>
<li><p> 'Normalised_tabel': Normalized table used in the clustering process
</p>
</li></ul>


<hr>
<h2 id='Motif_from_cluster_file'>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for details).</h2><span id='topic+Motif_from_cluster_file'></span>

<h3>Description</h3>

<p>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Motif_from_cluster_file(
  ClusTCR,
  Clust_selected = NULL,
  selected_cluster_column = "Clust_size_order"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Motif_from_cluster_file_+3A_clustcr">ClusTCR</code></td>
<td>
<p>Cluster file produced from <a href="#topic+mcl_cluster">mcl_cluster</a>.</p>
</td></tr>
<tr><td><code id="Motif_from_cluster_file_+3A_clust_selected">Clust_selected</code></td>
<td>
<p>Select which cluster to review.</p>
</td></tr>
<tr><td><code id="Motif_from_cluster_file_+3A_selected_cluster_column">selected_cluster_column</code></td>
<td>
<p>Select the column &quot;Clust_size_order&quot; of the cluster ordered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the motif.
</p>

<hr>
<h2 id='motif_plot'>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for ).</h2><span id='topic+motif_plot'></span>

<h3>Description</h3>

<p>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for ).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>motif_plot(
  ClusTCR,
  Clust_column_name = "Clust_size_order",
  Clust_selected = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="motif_plot_+3A_clustcr">ClusTCR</code></td>
<td>
<p>Matrix file produce from <a href="#topic+mcl_cluster">mcl_cluster</a></p>
</td></tr>
<tr><td><code id="motif_plot_+3A_clust_column_name">Clust_column_name</code></td>
<td>
<p>Name of clustering column from mcl_cluster file e.g. cluster</p>
</td></tr>
<tr><td><code id="motif_plot_+3A_clust_selected">Clust_selected</code></td>
<td>
<p>Select which cluster to display. Only one at a time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the motif.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage of mcl_cluster function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv",package = "ClusTCR2"))
# Perform clustering using mcl_cluster function
step1 &lt;- ClusTCR(example_file,allele = FALSE)
# perform mcl
step2 &lt;- mcl_cluster(step1)
# print the motif plot for the simple clustering
print(motif_plot(step2,Clust_selected = 1))
</code></pre>

<hr>
<h2 id='motif_plot_large'>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for details).</h2><span id='topic+motif_plot_large'></span>

<h3>Description</h3>

<p>Code for plotting the Motif based on a specific CDR3 length and V gene (see <a href="#topic+netplot_ClusTCR2">netplot_ClusTCR2</a> for details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>motif_plot_large(
  ClusTCRFile_large,
  Clust_column_name = "Clust_size_order",
  Clust_selected = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="motif_plot_large_+3A_clustcrfile_large">ClusTCRFile_large</code></td>
<td>
<p>Matrix file produced from <a href="#topic+mcl_cluster_large">mcl_cluster_large</a>.</p>
</td></tr>
<tr><td><code id="motif_plot_large_+3A_clust_column_name">Clust_column_name</code></td>
<td>
<p>Name of clustering column from mcl_cluster file e.g. cluster.</p>
</td></tr>
<tr><td><code id="motif_plot_large_+3A_clust_selected">Clust_selected</code></td>
<td>
<p>Select which cluster to display. Only one at a time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the motif.
</p>

<hr>
<h2 id='netplot_ClusTCR2'>Code for displaying the network.</h2><span id='topic+netplot_ClusTCR2'></span>

<h3>Description</h3>

<p>Code for displaying the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netplot_ClusTCR2(
  ClusTCR,
  filter_plot = 0,
  Clust_selected = 1,
  selected_col = "purple",
  selected_text_col = "black",
  selected_text_size = 3,
  non_selected_text_size = 2,
  Clust_column_name = "cluster",
  label = c("Name", "cluster", "CDR3", "V_gene", "Len"),
  non_selected_col = "grey80",
  non_selected_text_col = "grey40",
  alpha_selected = 1,
  alpha_non_selected = 0.5,
  colour = "color_test",
  all.colour = "default"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netplot_ClusTCR2_+3A_clustcr">ClusTCR</code></td>
<td>
<p>File produced from mcl_cluster</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_filter_plot">filter_plot</code></td>
<td>
<p>Filter's plot to remove connects grater than # e.g. 2 = 3 or more connections.</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_clust_selected">Clust_selected</code></td>
<td>
<p>Select which cluster to label.</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_selected_col">selected_col</code></td>
<td>
<p>Color of selected cluster (Default = purple)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_selected_text_col">selected_text_col</code></td>
<td>
<p>Color of selected cluster text (Default = black)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_selected_text_size">selected_text_size</code></td>
<td>
<p>Text size of selected cluster (Default = 3)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_non_selected_text_size">non_selected_text_size</code></td>
<td>
<p>Text size of non-selected clusters (Default = 2)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_clust_column_name">Clust_column_name</code></td>
<td>
<p>Name of clustering column from mcl_cluster file e.g. cluster (Re-numbering the original_cluster), Original_cluster, Clust_size_order (Based on cluster size e.g. number of nodes)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_label">label</code></td>
<td>
<p>Name to display on cluster: Name (CDR3_V_gene_Cluster), cluster, CDR3, V_gene, Len (length of CDR3 sequence), CDR3_selected, V_gene_selected, Name_selected,cluster_selected, (_selected only prints names of the chosen cluster), None</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_non_selected_col">non_selected_col</code></td>
<td>
<p>Color of selected cluster (Default = grey80)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_non_selected_text_col">non_selected_text_col</code></td>
<td>
<p>Color of selected clusters text (Default = grey40)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_alpha_selected">alpha_selected</code></td>
<td>
<p>Transparency of selected cluster (default = 1)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_alpha_non_selected">alpha_non_selected</code></td>
<td>
<p>Transparency of non-selected clusters (default = 0.5)</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_colour">colour</code></td>
<td>
<p>Colour selected = &quot;color_test&quot; or all = &quot;color_all&quot;</p>
</td></tr>
<tr><td><code id="netplot_ClusTCR2_+3A_all.colour">all.colour</code></td>
<td>
<p>Colours all points by: rainbow, random, heat.colors, terrain.colors, topo.colors, hcl.colors and default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object displaying the network plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage of mcl_cluster function with a stored file
example_file &lt;- read.csv(system.file("extdata", "my_data.csv",package = "ClusTCR2"))
# Perform clustering using mcl_cluster function
step1 &lt;- ClusTCR(example_file,allele = FALSE)
# perform mcl
step2 &lt;- mcl_cluster(step1)
# print the clustering plot after performing step 1 and step 2
print(netplot_ClusTCR2(step2, label = "Name_selected"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
