<!DOCTYPE html><html><head><title>Help for package clusterCons</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {clusterCons}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#auc'>
<p>Calculate area under the curve statistics</p></a></li>
<li><a href='#auc-class'><p>Class &quot;auc&quot;</p></a></li>
<li><a href='#aucplot'>
<p>Generate an area under the curve plot using lattice graphics</p></a></li>
<li><a href='#checks'>
<p>Functions to check the integrity of various objects</p></a></li>
<li><a href='#clrob'><p>Calculate the cluster robustness from consensus clustering results</p></a></li>
<li><a href='#cluscomp'><p>Perform consensus clustering with the option of using multiple algorithms and parameters and merging</p></a></li>
<li><a href='#clusterCons-package'><p>Calculate consensus clustering results from re-sampled clustering experiments with the option of using multiple algorithms and parameters</p></a></li>
<li><a href='#consmatrix-class'><p>Class &quot;consmatrix&quot;</p></a></li>
<li><a href='#data'>
<p>Data sets for the clusterCons package</p></a></li>
<li><a href='#deltak'>
<p>Function to calculate the change in the area under the curve (AUC) across a range of cluster number values</p></a></li>
<li><a href='#dk-class'><p>Class &quot;dk&quot;</p></a></li>
<li><a href='#dkplot'>
<p>Generate a delta-K plot from area under the curve (AUC) values across multiple cluster numbers.</p></a></li>
<li><a href='#expressionPlot'>
<p>Generate a profile plot for the data partitioned by cluster membership.</p></a></li>
<li><a href='#expSetProcess'>
<p>Internal function to extract the data from an expressionSet class object from the affy package for use with cluscomp</p></a></li>
<li><a href='#membBoxPlot'>
<p>Generate a box and whisker plot of membership robustness for all clusters</p></a></li>
<li><a href='#memrob'><p>Calculate the membership robustness from consensus clustering results</p></a></li>
<li><a href='#memroblist-class'><p>Class &quot;memroblist&quot;</p></a></li>
<li><a href='#memrobmatrix-class'><p>Class &quot;memrobmatrix&quot;</p></a></li>
<li><a href='#mergematrix-class'><p>Class &quot;mergematrix&quot;</p></a></li>
<li><a href='#wrappers'>
<p>Functions to wrap command calls to clustering functions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Title:</td>
<td>Consensus Clustering using Multiple Algorithms and Parameters</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-02-22</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dr. T. Ian Simpson &lt;ian.simpson@ed.ac.uk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>methods,cluster,lattice,RColorBrewer,grid,apcluster</td>
</tr>
<tr>
<td>Suggests:</td>
<td>latticeExtra</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for calculation of robustness measures for clusters and cluster membership based on generating consensus matrices from bootstrapped clustering experiments in which a random proportion of rows of the data set are used in each individual clustering. This allows the user to prioritise clusters and the members of clusters based on their consistency in this regime. The functions allow the user to select several algorithms to use in the re-sampling scheme and with any of the parameters that the algorithm would normally take. See Simpson, T. I., Armstrong, J. D. &amp; Jarman, A. P. (2010) &lt;<a href="https://doi.org/10.1186%2F1471-2105-11-590">doi:10.1186/1471-2105-11-590</a>&gt; and Monti, S., Tamayo, P., Mesirov, J. &amp; Golub, T. (2003) &lt;<a href="https://doi.org/10.1023%2Fa%3A1023949509487">doi:10.1023/a:1023949509487</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&gt; 2)</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/biomedicalinformaticsgroup/clusterCons">https://github.com/biomedicalinformaticsgroup/clusterCons</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/biomedicalinformaticsgroup/clusterCons/issues">https://github.com/biomedicalinformaticsgroup/clusterCons/issues</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-22 10:57:38 UTC; iansimpson</td>
</tr>
<tr>
<td>Author:</td>
<td>Dr. T. Ian Simpson
    <a href="https://orcid.org/0000-0003-0495-7187"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-22 20:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='auc'>
Calculate area under the curve statistics
</h2><span id='topic+auc'></span><span id='topic+aucs'></span>

<h3>Description</h3>

<p>These functions calculate the area under the curve (AUC) for cumulative density functions of a consensus matrix. The function <code>auc</code> operates on an indvidual consensus matrix whereas <code>aucs</code> operates on an entire <code>cluscomp</code> analysis result as described below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auc(x)
aucs(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auc_+3A_x">x</code></td>
<td>

<p>For <code>auc(x)</code>, provide a numeric square data matrix such as an individual consensus matrix. For <code>aucs(x)</code> provide a list of <code>"consmatrix"</code> class objects (see <code><a href="#topic+consmatrix-class">consmatrix-class</a> for details</code>) such as those produced directly by the <code><a href="#topic+cluscomp">cluscomp</a></code> function.
</p>
<p>The functions will not allow any missing values (NAs).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>auc(x)</code> returns an individual AUC value.
</p>
<p><code>aucs(x)</code> returns a data.frame with the following variables.
</p>
<table>
<tr><td><code>k</code></td>
<td>

<p>cluster number as a factor
</p>
</td></tr>
<tr><td><code>a</code></td>
<td>

<p>algorithm identifier as a factor
</p>
</td></tr>
<tr><td><code>aucs</code></td>
<td>

<p>the AUC value
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+consmatrix-class">consmatrix-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load up a test cluscomp result
data('testcmr');

#look at the result structure
summary(testcmr);

#calculate an individual AUC value for a consensus matrix
ac &lt;- auc(testcmr$e1_kmeans_k2@cm);

#calculate all of the AUC values from the \code{cluscomp} result for algorithm 'kmeans'
kmeanscmr &lt;- testcmr[grep('kmeans',names(testcmr))];
acs &lt;- aucs(kmeanscmr);
</code></pre>

<hr>
<h2 id='auc-class'>Class &quot;auc&quot;</h2><span id='topic+auc-class'></span>

<h3>Description</h3>

<p>Objects of class <code>'auc'</code> contain a data.frame which have three variables <code>k</code>, <code>a</code> and <code>auc</code>
as described in the <code><a href="#topic+aucs">aucs</a></code> function description. This class simply holds the result from a call to <code><a href="#topic+aucs">aucs</a></code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("auc", ...)</code>, although they are normally generated internally by the <code><a href="#topic+aucs">aucs</a></code> function. 
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+aucs">aucs</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("auc")
</code></pre>

<hr>
<h2 id='aucplot'>
Generate an area under the curve plot using lattice graphics
</h2><span id='topic+aucplot'></span>

<h3>Description</h3>

<p>This function uses the <code><a href="lattice.html#topic+lattice">lattice</a></code> function <code><a href="lattice.html#topic+xyplot">xyplot</a></code> to generate an AUC plot from a valid <code>"auc"</code> class object (see <code><a href="#topic+auc-class">auc-class</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aucplot(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aucplot_+3A_x">x</code></td>
<td>

<p>a valid &quot;auc&quot; class object (see <code><a href="#topic+auc-class">auc-class</a></code>), normally generated by the <code><a href="#topic+aucs">aucs</a></code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+consmatrix-class">consmatrix-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load up a test cluscomp result
data('testcmr');

#look at the result structure
summary(testcmr);

#calculate all of the AUC values from the \code{cluscomp} result for algorithm 'kmeans'
kmeanscmr &lt;- testcmr[grep('kmeans',names(testcmr))];
acs &lt;- aucs(kmeanscmr);

#plot the AUC curve
aucplot(acs);
</code></pre>

<hr>
<h2 id='checks'>
Functions to check the integrity of various objects
</h2><span id='topic+checks'></span><span id='topic+data_check'></span><span id='topic+validConsMatrixObject'></span><span id='topic+validMemRobListObject'></span><span id='topic+validMemRobMatrixObject'></span><span id='topic+validMergeMatrixObject'></span><span id='topic+validAUCObject'></span><span id='topic+validDkObject'></span>

<h3>Description</h3>

<p>These methods are mainly internal although the user may like to check their original data using 
<code>data_check</code> before they perform consensus clustering experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_check(x)
validConsMatrixObject(object)
validMemRobListObject(object)
validMemRobMatrixObject(object)
validMergeMatrixObject(object)
validAUCObject(object)
validDkObject(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checks_+3A_x">x</code></td>
<td>

<p>The data.frame object to be checked prior to using with the <code><a href="#topic+cluscomp">cluscomp</a></code> function.
</p>
</td></tr>
<tr><td><code id="checks_+3A_object">object</code></td>
<td>

<p>The object to be checked with the suitable function by type. These are used internally by several of the functions in the package.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns <code>TRUE</code> if check is passed or an error message if it is not
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load data
data(sim_profile);

#check if this can be used by cluscomp
data_check(sim_profile);

#perform a clusomp run
cmr &lt;- cluscomp(sim_profile,clmin=2,clmax=2,rep=10);

#check one of the consensus matrices
validConsMatrixObject(cmr$e1_kmeans_k2)

</code></pre>

<hr>
<h2 id='clrob'>Calculate the cluster robustness from consensus clustering results</h2><span id='topic+clrob'></span>

<h3>Description</h3>

<p>This function calculates the cluster robustness from a <code><a href="#topic+consmatrix">consmatrix</a></code> or <code><a href="#topic+mergematrix">mergematrix</a></code>
class object. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clrob(x,rm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clrob_+3A_x">x</code></td>
<td>

<p>either a <code><a href="#topic+consmatrix">consmatrix</a></code> or <code><a href="#topic+mergematrix">mergematrix</a></code> object. 
</p>
</td></tr>
<tr><td><code id="clrob_+3A_rm">rm</code></td>
<td>

<p>(optional) if a <code><a href="#topic+mergematrix">mergematrix</a></code> object is passed then you must provide a reference clustering
structure to calculate cluster robustness against. These structures are stored with every <code><a href="#topic+consmatrix">consmatrix</a></code>
object in the 'rm' slot. You would normally select a reference matrix for a cluster number matching that of the
<code><a href="#topic+mergematrix">mergematrix</a></code> (see example below). 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame of the cluster robustness values indexed by cluster number.
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see <code><a href="#topic+cluscomp">cluscomp</a></code>, <code><a href="#topic+consmatrix">consmatrix</a></code> and <code><a href="#topic+mergematrix">mergematrix</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load cmr (consensus clustering result produced by cluscomp)
data(testcmr);

#calculate the cluster robustness of the consensus matrix for pam where k=4
clrob(testcmr$e1_kmeans_k4);

#calculate the cluster robustness of the merge matrix in reference
#to the clustering structrure of pam where k=4
clrob(testcmr$merge_k4,testcmr$e1_kmeans_k4@rm);
</code></pre>

<hr>
<h2 id='cluscomp'>Perform consensus clustering with the option of using multiple algorithms and parameters and merging</h2><span id='topic+cluscomp'></span>

<h3>Description</h3>

<p>Calculates an NxN consensus matrix for each clustering experiment performed where each entry has a value between 0 (never observed) and 1 (always observed)<br />
When running with more than one algorithm or with the same algorithm and multiple conditions a consensus matrix will be generated for each.
These can optionally be merged into a <code><a href="#topic+mergematrix">mergematrix</a></code> by cluster number by setting merge=1.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluscomp(
  x,
  diss=FALSE,
  algorithms = list("kmeans"),
  alparams = list(),
  alweights = list(),
  clmin = 2,
  clmax = 10,
  prop = 0.8,
  reps = 50,
  merge = 0
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluscomp_+3A_x">x</code></td>
<td>

<p>data.frame of numerical data with conditions as the column names and unique ids as the row names. All variables must be numeric. Missing values(NAs) are not allowed.
Optionally you can pass a distance matrix directly, in which case you must ensure that the distance matrix is a data.frame and that the row and column names match each other (as the
distance matrix is a pair-wise distance calculation).
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_diss">diss</code></td>
<td>

<p>set to TRUE if you are providing a distance matrix, default is FALSE
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_algorithms">algorithms</code></td>
<td>

<p>list of algorithm names which can be drawn from 'agnes','diana','pam','kmeans' or 'hclust'. The user can also write a simple wrapper for any other clustering method (see details)
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_alparams">alparams</code></td>
<td>

<p>list of algorithm paramter lists using the same specification as for the individual algorithm called (see details)
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_alweights">alweights</code></td>
<td>

<p>list of integer weights for each algorithm (only used when merging consensus results between algorithms)
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_clmin">clmin</code></td>
<td>

<p>integer for the smallest cluster number to consider
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_clmax">clmax</code></td>
<td>

<p>integer for the largest cluster number to consider
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_prop">prop</code></td>
<td>

<p>numeric for the proportion of rows to sample during the process. Must be between 0 and 1
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_reps">reps</code></td>
<td>

<p>integer for the number of iterations to perform per clustering
</p>
</td></tr>
<tr><td><code id="cluscomp_+3A_merge">merge</code></td>
<td>

<p>an integer indicating whether you also want the merged matrices (1) or just the consensus ones (0), accepts only 1 or 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+cluscomp">cluscomp</a></code> is an implementation of a consensus clustering methodology first proposed by Monti et al. (2003) in which the connectivity between any two members of a data matrix is tested by resampling statistics. The principle is that by only sampling a random proportion of rows in the data matrix and performing many clustering experiments we can capture information about the robustness of the clusters identified by the full unsampled clustering result.
</p>
<p>For each re-sampling experiment run a zero square matrix is created with identical rows and columns matching the unique ids of the rows of the data matrix, this matrix is called the connectivity matrix. A second identically sized matrix is created to count the number of times that any pair of row ids are called in any one re-sampled clustering. This matrix is called the identity matrix. For each iteration within the experiment the rows sampled are recorded in the identity matrix and then the co-occurrence of all pairs are recorded in the connectivity matrix. These values are incremented for each iteration until finally a conensensus matrix is generated by dividing the connectivity matrix by the identity matrix.
</p>
<p>The consensus matrix is the raw output from <code><a href="#topic+cluscomp">cluscomp</a></code> implemented as a class 
<code><a href="#topic+consmatrix">consmatrix</a></code>. If the user has specified to return a merged matrix in addition to the consensus 
matrices then for each clustering with the same k (cluster number value) an object of class <code><a href="#topic+mergematrix">mergematrix</a></code> is also 
returned in the list which is identical to a <code><a href="#topic+consmatrix">consmatrix</a></code> with the exception that the
'cm' slot is occupied by the merged matrix (a weighted average of all the consensus matrices for
the cluster number matched consensus matrices) and there is no reference matrix slot (as there is no 
reference clustering for the merge). The user should instead call the <code><a href="#topic+memrob">memrob</a></code> 
function using the merge matrix and providing a reference matrix from one of the cluster number 
matched <code><a href="#topic+consmatrix">consmatrix</a></code> objects from which the merge was generated. This provides a way 
to quantify the difference between single and multi-algorithm resampling schemes.
</p>


<h3>Value</h3>

<p>a list of objects of class <code><a href="#topic+consmatrix">consmatrix</a></code> and (if merge specified) <code><a href="#topic+mergematrix">mergematrix</a></code>. See <code><a href="#topic+consmatrix">consmatrix</a></code> and <code><a href="#topic+mergematrix">mergematrix</a></code> for details.
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.<br />
</p>
<p>Consensus clustering: A resampling-based method for class discovery and visualization of gene expression microarray data. Monti, S., Tamayo, P., Mesirov, J. and Golub, T. Machine Learning, 52, July 2003.<br />
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+cluster">cluster</a></code>,<code><a href="#topic+clrob">clrob</a></code>,<code><a href="#topic+memrob">memrob</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load test data
data(sim_profile);

#perform a group of re-sampling clustering experiments accepting default parameters 
#for the clustering algorithms
cmr &lt;- cluscomp(
 sim_profile,
 algorithms=list('kmeans','pam'),
 merge=1,
 clmin=2,
 clmax=5,
 reps=5
)

#display resulting matrices contained in the consensus result list
summary(cmr);

#display the cluster robusteness for the kmeans k=4 consensus matrix
clrob(cmr$e2_pam_k4);

#plot a heatmap of the consensus matrix, note you access the cluster matrix object 
#through the cm slot
#heatmap(cmr$e2_pam_k4@cm);

#display the membership robustness for kmeans k=4 cluster 1
memrob(cmr$e2_pam_k4)$cluster1;

#merged consensus example
#data(testcmr);

#calculate the membership robustness for the merge matrix when cluster number k=4,
#in reference to the pam scaffold. (see memrob for more details). 
#mr &lt;- memrob(testcmr$merge_k4,testcmr$e1_kmeans_k4@rm);

#show the membership robustness for cluster 1
#mr$cluster1;
</code></pre>

<hr>
<h2 id='clusterCons-package'>Calculate consensus clustering results from re-sampled clustering experiments with the option of using multiple algorithms and parameters</h2><span id='topic+clusterCons-package'></span><span id='topic+clusterCons'></span>

<h3>Description</h3>

<p>clusterCons is a package containing functions that generate robustness measures for clusters and cluster membership based on generating consensus 
matrices from bootstrapped clustering experiments in which a random proportion of rows of the data set are used in each individual clustering.
This allows the user to prioritise clusters and the members of clusters based on their consistency in this regime. The functions allow the user to
select several algorithms to use in the re-sampling scheme and with any of the parameters that the algorithm would normally take.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> clusterCons</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2010-10-12</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> methods,cluster,lattice,RColorBrewer,grid,apcluster</td>
</tr>
<tr>
 <td style="text-align: left;">
Extends: </td><td style="text-align: left;"> cluster</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> latticeExtra</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The user should first prepare an entirely numeric <code>data.frame</code> in which the conditions to be clustered are the column names and the unique ids
of the entities are the row names. Compatibility of the resulting data.fram can be checked by using the <code>data_check</code> function. 
</p>


<h3>Functions to run the consensus clustering and retrieve robustness information</h3>

<p><code>cluscomp</code> - generate consensus matrices from re-sampled clustering experiments with the option of multiple algorithms and parameters<br />
<code>clrob</code> - calculate the robustness of the clusters from the consensus matrix<br />
<code>memrob</code> - calculate the cluster membership robustness from the consensus matrix<br />
</p>


<h3>Internal functions to call the individual clustering algorithms</h3>

<p><code>agnes_clmem</code> - wrapper for the <code>agnes</code> function of package cluster<br />
<code>diana_clmem</code> - wrapper for the <code>diana</code> function of package cluster<br />
<code>hclust_clmem</code> - wrapper for the <code>hclust</code> function of package cluster<br />
<code>kmeans_clmem</code> - wrapper for the <code>kmeans</code> function of package cluster<br />
<code>pam_clmem</code> - wrapper for the <code>pam</code> function of package cluster<br />
<code>apcluster_clmem</code> - wrapper for the <code>apclusterK</code> function of package apcluster<br />
</p>


<h3>Functions to calculate AUC related metrics</h3>

<p><code>auc</code> - calculates the area under the curve for a series of clustering experiments with the same cluster number<br />
<code>aucs</code> - calculates the areas under the curves of a series of clustering experiments over a range of cluster numbers<br />
<code>deltak</code> - calculates the change in the area under the curve<br />
</p>


<h3>Functions to check data and object validity</h3>

<p><code>data_check</code> - check that the provided <code>data.frame</code> is formatted correctly<br />
<code>expSetProcess</code> - extracts the data set from an object of class expressionSet<br />  
<code>validConsMatrixObject</code> - check the validity of a <code>consmatrix</code> object<br />
<code>validMergeMatrixObject</code> - check the validity of a <code>mergematrix</code> object<br />
<code>validMemRobListObject</code> - check the validity of a membership robustness list object<br />
<code>validMemRobMatrixObject</code> - check the validity of a membership robustness matrix object<br />
<code>validAUCObject</code> - check the validity of an <code>"auc"</code> class object<br />
<code>validDkObject</code> - check the validity of an <code>"dk"</code> class object<br />
</p>


<h3>Functions to plot out performance curves</h3>

<p><code>aucplot</code> - plot area under the curve (AUC) plots from consensus clustering results<br />
<code>dkplot</code> - plot change in AUC by cluster number (delta-K plot)<br />
<code>expressionPlot</code> - plot the original data partitioned by cluster membership<br />
<code>membBoxPlot</code> - plot a box and whisker plot of the membership robustness for each cluster<br />
</p>


<h3>Keywords</h3>

<p>cluster
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+cluster">cluster</a></code>,<code><a href="lattice.html#topic+lattice">lattice</a></code>,<code>apcluster</code>
</p>


<h3>Examples</h3>

<p>#load data
data(sim_profile);
</p>
<p>#perform consensus clustering
cmr &lt;- cluscomp(sim_profile,algo=list('agnes','pam','kmeans'),clmin=2,clmax=7,rep=10,merge=1);
</p>
<p>#see the consensus and merge matrices
summary(cmr);
</p>
<p>#fetch the cluster robustness for agnes consensus clustering with k=3
clrob(cmr$e1_agnes_k3);
</p>
<p>#show the membership robustness for cluster 1
memrob(cmr$e1_agnes_k3)$cluster1
</p>
<p>#show the same, but for the merge against the k=3 agnes clustering structure
#note we provide the reference matrix (which is the original cluster membership for agnes where k=3)
clrob(cmr$merge_k3,cmr$e1_agnes_k3@rm);
memrob(cmr$merge_k3,cmr$e1_agnes_k3@rm)$cluster1;
</p>
<p>#calculate the AUCs
acs &lt;- aucs(cmr);
</p>
<p>#plot the AUC curves
aucplot(acs);
</p>
<p>#calculate the delta-Ks
dks &lt;- deltak(acs);
</p>
<p>#plot the delta-K curves
dkplot(dks);
</p>
<p>#plot the expression profiles
expressionPlot(sim_profile,cmr$e1_agnes_k3);
</p>
<p>#plot the bwplot of membership robustness for the same
membBoxPlot(memrob(cmr$e1_agnes_k3));
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.<br />
</p>
<p>Consensus clustering: A resampling-based method for class discovery and visualization of gene expression microarray data. Monti, S., Tamayo, P., Mesirov, J. and Golub, T. Machine Learning, 52, July 2003.<br />
</p>

<hr>
<h2 id='consmatrix-class'>Class &quot;consmatrix&quot;</h2><span id='topic+consmatrix-class'></span><span id='topic+consmatrix'></span>

<h3>Description</h3>

<p>Objects of class <code>'consmatrix'</code> are created to hold the results of a consensus clustering experiment along with the necessary ancillary
data to allow the subsequent downstream calculations such as cluster and membership robustness. In addition the object holds the original call 
made when running <code><a href="#topic+cluscomp">cluscomp</a></code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("consmatrix", ...)</code>, but are normally created internally by the <code><a href="#topic+cluscomp">cluscomp</a></code> function to store
consensus matrices and their associated meta-data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>cm</code>:</dt><dd><p>Object of class <code>"matrix"</code> - the consensus matrix itself</p>
</dd>
<dt><code>rm</code>:</dt><dd><p>Object of class <code>"data.frame"</code> - the cluster membership of the full (i.e. not consensus) clustering result when 
the current algorith is called with the same algorithm parameters as the consensus clustering run. This is needed to be able to work with
merge matrices that need a clustering structure on which to operate to produce cluster and membership robustness values.</p>
</dd>
<dt><code>a</code>:</dt><dd><p>Object of class <code>"character"</code> - the clustering algorithm name </p>
</dd>
<dt><code>k</code>:</dt><dd><p>Object of class <code>"numeric"</code> - the cluster number (k) used </p>
</dd>
<dt><code>call</code>:</dt><dd><p>Object of class <code>"call"</code> - the original parameters passed to <code>cluscomp</code> for provenance and reproducibility</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+cluscomp">cluscomp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("consmatrix");

#you can access the slots in useful ways

#load a cmr
data(testcmr);

#get a consensus clustering matrix via the 'cm' slot
cm &lt;- testcmr$e1_kmeans_k4@cm;

#this can be used as a distance matrix, e.g. for a heatmap
heatmap(cm);

#or as a new distance matrix
dm &lt;- data.frame(cm) #first convert to a data.frame
#make sure names are the same for rows and columns
names(dm) &lt;- row.names(dm);

#you need to explicitly tell cluscomp that you are passing a distance matrix
cmr2 &lt;- cluscomp(dm,diss=TRUE,clmin=2,clmax=4,rep=2);

#for merge consensus clustering you take advantage of the reference matrix (rm) slot
#cluster robustness for agnes with cluster number (k) = 3
clrob(testcmr$merge_k3,testcmr$e1_kmeans_k3@rm);
#membership robustness for cluster 1
memrob(testcmr$merge_k3,testcmr$e1_kmeans_k3@rm)$cluster1;
</code></pre>

<hr>
<h2 id='data'>
Data sets for the clusterCons package
</h2><span id='topic+golub'></span><span id='topic+sim_class'></span><span id='topic+sim_profile'></span><span id='topic+testcmr'></span>

<h3>Description</h3>

<p>These data sets are used by the examples in the package function descriptions and allow the user to explore the functionality of the package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(golub);
data(sim_class);
data(sim_profile);
data(testcmr);
</code></pre>


<h3>Format</h3>

<p>golub : data.frame of gene expression values for 999 genes for 38 leukemia patients (1-27) ALL and (28-38) AML.
sim_class : data.frame of 200 simulated gene expression values for 30 conditions where there are 4 discrete classes of expression profile, for testing clustering with the transposed data (clustering by column).
sim_profile : data.frame of 120 simulated gene expression values for 4 conditions where there are 4 discrete classes of expression profile, for testing general clustering  (clustering by row). 
testcmr : list of consensus and merge matrix results from a <code><a href="#topic+cluscomp">cluscomp</a></code> run (see <code><a href="#topic+consmatrix-class">consmatrix-class</a></code> and <code><a href="#topic+mergematrix-class">mergematrix-class</a></code>). 
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>
<p>Molecular classification of cancer: class discovery and class prediction by gene expression monitoring. Golub, TR and Slonim, DK and Tamayo, P and Huard, C and Gaasenbeek, M and Mesirov, JP and Coller, H and Loh, ML and Downing, JR and Caligiuri, MA and Bloomfield, CD and Lander, ES. Science 1999, 286:531-537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#cluster by class
data(sim_class);
cutree(agnes(t(sim_class)),4);

#cluster by profile
data(sim_profile);
cutree(agnes(sim_profile),4);
</code></pre>

<hr>
<h2 id='deltak'>
Function to calculate the change in the area under the curve (AUC) across a range of cluster number values
</h2><span id='topic+deltak'></span>

<h3>Description</h3>

<p>This function takes an <code>"auc"</code> class object and calculates the difference in AUC value by cluster number (called delta-K). Peaks in delta-K
coincide with the cluster numbers that are most robust and provide estimates for the optimal cluster number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deltak(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deltak_+3A_x">x</code></td>
<td>

<p>a valid <code>"auc"</code> class object, normally provided as a result from the <code><a href="#topic+aucs">aucs</a></code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

 
<p><code>deltak(x)</code> returns a data.frame with the following variables.
</p>
<table>
<tr><td><code>k</code></td>
<td>

<p>cluster number as a factor
</p>
</td></tr>
<tr><td><code>a</code></td>
<td>

<p>algorithm identifier as a factor
</p>
</td></tr>
<tr><td><code>dk</code></td>
<td>

<p>the delta-K value
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+aucs">aucs</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load a test cluscomp result set
data(testcmr)

#calculate all of the AUC values from the \code{cluscomp} result for algorithm 'kmeans'
kmeanscmr &lt;- testcmr[grep('kmeans',names(testcmr))];
acs &lt;- aucs(kmeanscmr);

#calculate the delta-K values
dks &lt;- deltak(acs);
</code></pre>

<hr>
<h2 id='dk-class'>Class &quot;dk&quot;</h2><span id='topic+dk-class'></span>

<h3>Description</h3>

<p>Objects of class <code>'dk'</code> contain a data.frame which have three variables <code>k</code>, <code>a</code> and <code>deltak</code>
as described in the <code><a href="#topic+deltak">deltak</a></code> function description. This class simply holds the result from a call to <code><a href="#topic+deltak">deltak</a></code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("dk", ...)</code>, although they are normally generated internally by the <code><a href="#topic+deltak">deltak</a></code> function. 
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+aucs">aucs</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("dk")
</code></pre>

<hr>
<h2 id='dkplot'>
Generate a delta-K plot from area under the curve (AUC) values across multiple cluster numbers.
</h2><span id='topic+dkplot'></span>

<h3>Description</h3>

<p>This function uses the <code><a href="lattice.html#topic+lattice">lattice</a></code> function <code><a href="lattice.html#topic+xyplot">xyplot</a></code> to generate an delta-K plot from a valid <code>"dk"</code> class object (see <code><a href="#topic+dk-class">dk-class</a></code>).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dkplot(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dkplot_+3A_x">x</code></td>
<td>

<p>a valid &quot;dk&quot; class object (see <code><a href="#topic+dk-class">dk-class</a></code>), normally generated by the <code><a href="#topic+deltak">deltak</a></code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+consmatrix-class">consmatrix-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load up a test cluscomp result
data('testcmr');

#look at the result structure
summary(testcmr);

#calculate all of the AUC values from the \code{cluscomp} result for algorithm 'kmeans'
kmeanscmr &lt;- testcmr[grep('kmeans',names(testcmr))];
acs &lt;- aucs(kmeanscmr);

#calculate all of the delta-K values
dks &lt;- deltak(acs);

#plot the delta-K curve
dkplot(dks);
</code></pre>

<hr>
<h2 id='expressionPlot'>
Generate a profile plot for the data partitioned by cluster membership.
</h2><span id='topic+expressionPlot'></span>

<h3>Description</h3>

<p>This function uses the <code><a href="lattice.html#topic+lattice">lattice</a></code> function <code><a href="lattice.html#topic+xyplot">xyplot</a></code> to generate a profile plot of the data values grouped by cluster in a multi-panel plot. The function
takes as input the original data.frame() and a valid <code>"consmatrix"</code> class object (see <code><a href="#topic+consmatrix-class">consmatrix-class</a></code>) by which to segregate the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expressionPlot(x,cm);
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expressionPlot_+3A_x">x</code></td>
<td>

<p>the original data.frame() object used in the clustering.
</p>
</td></tr>
<tr><td><code id="expressionPlot_+3A_cm">cm</code></td>
<td>

<p>a valid <code>"consmatrix"</code> class object generated by the <code><a href="#topic+cluscomp">cluscomp</a></code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+consmatrix-class">consmatrix-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load up the data set
data(sim_profile);

#load up an example cluscomp result with this data
data('testcmr');

#plot the expression profiles
expressionPlot(sim_profile,testcmr$e1_kmeans_k4);

</code></pre>

<hr>
<h2 id='expSetProcess'>
Internal function to extract the data from an expressionSet class object from the affy package for use with cluscomp
</h2><span id='topic+expSetProcess'></span>

<h3>Description</h3>

<p>This is a convenience function that is used internally to allow the user to pass an expressionSet object from the microarray processing package 'affy'
directly to the <code><a href="#topic+cluscomp">cluscomp</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expSetProcess(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expSetProcess_+3A_x">x</code></td>
<td>

<p>An object of class expressionSet from the Bioconductor package 'affy'.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>when called directly, returns a suitably labeled data.frame() object of the expressionSet expression values.
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>

<hr>
<h2 id='membBoxPlot'>
Generate a box and whisker plot of membership robustness for all clusters
</h2><span id='topic+membBoxPlot'></span>

<h3>Description</h3>

<p>This function uses the <code><a href="lattice.html#topic+lattice">lattice</a></code> function <code><a href="lattice.html#topic+bwplot">bwplot</a></code> to generate a box and whisker plot of membership robustness from the result of a call to the <code>memrob</code> function.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>membBoxPlot(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="membBoxPlot_+3A_x">x</code></td>
<td>

<p>the result of a call to the <code>memrob</code> function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+memroblist-class">memroblist-class</a></code>,<code><a href="#topic+memrob">memrob</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load up a test cluscomp result
data('testcmr');

#calculate the membershpi robustness for one of the clustering results
mr &lt;- memrob(testcmr$e1_kmeans_k5);

#plot the bwplot
membBoxPlot(mr);
</code></pre>

<hr>
<h2 id='memrob'>Calculate the membership robustness from consensus clustering results</h2><span id='topic+memrob'></span>

<h3>Description</h3>

<p>This function calculates the membership robustness from a <code><a href="#topic+consmatrix">consmatrix</a></code> or <code><a href="#topic+mergematrix">mergematrix</a></code>
class object. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>memrob(x,rm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="memrob_+3A_x">x</code></td>
<td>

<p>either a <code><a href="#topic+consmatrix">consmatrix</a></code> or <code><a href="#topic+mergematrix">mergematrix</a></code> object. 
</p>
</td></tr>
<tr><td><code id="memrob_+3A_rm">rm</code></td>
<td>

<p>(optional) if a <code><a href="#topic+mergematrix">mergematrix</a></code> object is passed then you must provide a reference clustering
structure to calculate cluster robustness against. These structures are stored with every <code><a href="#topic+consmatrix">consmatrix</a></code>
object in the 'rm' slot. You would normally select a reference matrix for a cluster number matching that of the
<code><a href="#topic+mergematrix">mergematrix</a></code> (see example below). 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+memroblist">memroblist</a></code> class objects, one for each cluster, and the full membership robustness matrix as a <code><a href="#topic+memrobmatrix">memrobmatrix</a></code> class object.
</p>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see <code><a href="#topic+cluscomp">cluscomp</a></code>, <code><a href="#topic+consmatrix">consmatrix</a></code> and <code><a href="#topic+mergematrix">mergematrix</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load cmr (consensus clustering result produced by cluscomp)
data(testcmr);

#calculate the cluster robustness of the consensus matrix for pam where k=4
mr1 &lt;- memrob(testcmr$e1_kmeans_k4);

#show the membership robustness of cluster 1
mr1$cluster1;

#calculate the cluster robustness of the merge matrix in reference
#to the clustering structure of pam where k=4
mr2 &lt;- memrob(testcmr$merge_k4,testcmr$e1_kmeans_k4@rm);

#plot a heatmap of the full membership robustness matrix
heatmap(mr2$resultmatrix@mrm)
</code></pre>

<hr>
<h2 id='memroblist-class'>Class &quot;memroblist&quot;</h2><span id='topic+memroblist-class'></span><span id='topic+memroblist'></span>

<h3>Description</h3>

<p>Objects of class <code>'memroblist'</code> are created to hold the membership robustness scores for the features (e.g. genes) of a cluster.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("memroblist", ...)</code>, although these objects are normally created internally by the <code><a href="#topic+memrob">memrob</a></code> function. 
</p>


<h3>Slots</h3>


<dl>
<dt><code>mrl</code>:</dt><dd><p>Object of class <code>"data.frame"</code> - the membership robustness list itself</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+memrob">memrob</a></code> function/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("memroblist")

#load a cmr
data(testcmr);

#calculate the membership robustness for agnes, k=4
mr &lt;- memrob(testcmr$e2_agnes_k4);

#get a membership robustness list
mrl &lt;- mr$cluster1;

</code></pre>

<hr>
<h2 id='memrobmatrix-class'>Class &quot;memrobmatrix&quot;</h2><span id='topic+memrobmatrix-class'></span><span id='topic+memrobmatrix'></span>

<h3>Description</h3>

<p>Objects of class <code>'memrobmatrix'</code> hold the full membership robustness matrix generated from analysis of a consensus matrix. This
includes the calculations of membership robustness for all features (e.g. genes) for each cluster. This can be useful as it allows you to
see what conritbution a particular feature (e.g. gene) is making to other clusters. This could resonably be thought of as a measure similar
to 'fuzziness' i.e. partial cluster membership. If the value of the membership robustness for a feature is similar in many clusters then that
is additional evidence that the feature is not easily placed in any cluster.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("memrobmatrix", ...)</code>, although they are usually generated internally by the <code><a href="#topic+memrob">memrob</a></code> function. 
</p>


<h3>Slots</h3>


<dl>
<dt><code>mrm</code>:</dt><dd><p>Object of class <code>"matrix"</code> - this is the full membership robustness matrix itself and therefore has the same dimensions as the
original data object used in the clustering</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+memrob">memrob</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("memrobmatrix")

#load cmr
data(testcmr);

#calculate membership robustness
mr &lt;- memrob(testcmr$e1_kmeans_k3)

#get the full membership robustness matrix (matrix itself held in slot 'mrm')
mrm &lt;- mr$resultmatrix@mrm;

</code></pre>

<hr>
<h2 id='mergematrix-class'>Class &quot;mergematrix&quot;</h2><span id='topic+mergematrix-class'></span><span id='topic+mergematrix'></span>

<h3>Description</h3>

<p>Objects of class <code>'mergematrix'</code> hold the merge matrix  in the same way that a consmatrix object holds a consensus matrix. As merge matrices only
make sense in the context of the consensus clustering results that were used to generate them we do not store the meta-data for any one consensus clustering parameter set
as we do for a 'consmatrix' object. All we need to identify the 'mergematrix' is the cluster number.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("mergematrix", ...)</code>, although they are normally generated by the <code><a href="#topic+cluscomp">cluscomp</a></code> function when merge is specfied. 
</p>


<h3>Slots</h3>


<dl>
<dt><code>cm</code>:</dt><dd><p>Object of class <code>"matrix"</code> - the merge matrix itself</p>
</dd>
<dt><code>k</code>:</dt><dd><p>Object of class <code>"numeric"</code>  - the cluster number (k) value for which the merge was calculated</p>
</dd>
<dt><code>a</code>:</dt><dd><p>Object of class <code>"character"</code> - always takes the value of 'merge' to identify it as a merge matrix</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p>Also see the <code><a href="#topic+cluscomp">cluscomp</a></code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("mergematrix")

#load the cmr
data(testcmr);

#get a merge matrix object
mm &lt;- testcmr$merge_k4;

#plot a heatmap of the merge matrix
heatmap(mm@cm);
</code></pre>

<hr>
<h2 id='wrappers'>
Functions to wrap command calls to clustering functions
</h2><span id='topic+wrappers'></span><span id='topic+agnes_clmem'></span><span id='topic+pam_clmem'></span><span id='topic+hclust_clmem'></span><span id='topic+diana_clmem'></span><span id='topic+kmeans_clmem'></span><span id='topic+apcluster_clmem'></span>

<h3>Description</h3>

<p>These are primarily internal functions called by <code><a href="#topic+cluscomp">cluscomp</a></code> to execute 
clustering runs and are unlikely to be used directly. The wrappers are detailed in 
the <code>algorithm.R</code> file of the <code>clusterCons</code> package and the user can add
their own wrappers to this to extend the number of algorithms supported. These five
wrappers allow the user to specify the conditions under which the corresponding 
clustering algorithms are run and follow exactly the same specifications as the
corresponding <code><a href="survival.html#topic+cluster">cluster</a></code> functions (see <code><a href="cluster.html#topic+agnes">agnes</a></code>, <code><a href="cluster.html#topic+pam">pam</a></code>,
<code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="cluster.html#topic+diana">diana</a></code> and <code><a href="stats.html#topic+kmeans">kmeans</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agnes_clmem(x, clnum, params = list())
pam_clmem(x, clnum, params = list())
hclust_clmem(x, clnum, params = list())
diana_clmem(x, clnum, params = list())
kmeans_clmem(x, clnum, params = list())
apcluster_clmem(x,clnum,params = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wrappers_+3A_x">x</code></td>
<td>

<p>A data.frame of numerical values to be clustered which must pass the <code><a href="#topic+data_check">data_check</a></code>
function. This function simply checks that there are no missing values, that all of the data
is numeric and that row.names and column.names are unique. This is essential to ensure that
individual rows (e.g. genes) and columns (e.g. experimental conditions) can be identified
consistently. 
</p>
</td></tr>
<tr><td><code id="wrappers_+3A_clnum">clnum</code></td>
<td>

<p>The number of specified clusters. When using the <code><a href="#topic+cluscomp">cluscomp</a></code> function, this will
be over-ridden by the cluster range specified using the parameters <code>clmin</code>
and <code>clmax</code> (see <code><a href="#topic+cluscomp">cluscomp</a></code> for details).
</p>
</td></tr>
<tr><td><code id="wrappers_+3A_params">params</code></td>
<td>

<p>A list of key, value pairs specifying the parameters to pass to the clustering algorithm. These
follow the exact specification of the original functions in the <code><a href="survival.html#topic+cluster">cluster</a></code> package 
(see <code><a href="cluster.html#topic+agnes">agnes</a></code>, <code><a href="cluster.html#topic+pam">pam</a></code>, <code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="cluster.html#topic+diana">diana</a></code> and <code><a href="stats.html#topic+kmeans">kmeans</a></code>).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with row.names matching that of the data.
</p>
<table>
<tr><td><code>cm</code></td>
<td>

<p>cluster membership identifier specifying the cluster into which the row has been classified
</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Dr. T. Ian Simpson
<a href="mailto:ian.simpson@ed.ac.uk">ian.simpson@ed.ac.uk</a>
</p>


<h3>References</h3>

<p>Merged consensus clustering to assess and improve class discovery with microarray data. Simpson TI, Armstrong JD and Jarman AP. BMC Bioinformatics 2010, 11:590.
</p>


<h3>See Also</h3>

<p><code><a href="survival.html#topic+cluster">cluster</a></code>, <code><a href="cluster.html#topic+agnes">agnes</a></code>, <code><a href="cluster.html#topic+pam">pam</a></code>, <code><a href="stats.html#topic+hclust">hclust</a></code>, <code><a href="cluster.html#topic+diana">diana</a></code>, <code><a href="stats.html#topic+kmeans">kmeans</a></code> and <code>apclusterK</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load some data
data(sim_profile);

#run a basic agnes clustering with 3 clusters
cm &lt;- agnes_clmem(sim_profile,3);

#pass some more complex parameters
agnes_params = list(metric='manhattan',method='single');
cm &lt;- agnes_clmem(sim_profile, 3,params=agnes_params);

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
