<!DOCTYPE html><html lang="en"><head><title>Help for package ClusROC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ClusROC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ClusROC-package'>
<p>ROC Analysis in Three-Class Classification Problems for Clustered Data</p></a></li>
<li><a href='#ci_clus_vus'><p>Confidence Intervals for Covariate-specific VUS</p></a></li>
<li><a href='#clus_lme'><p>Linear Mixed-Effects Models for a continuous diagnostic test or a biomarker (or a classifier).</p></a></li>
<li><a href='#clus_opt_thres3'><p>Estimation of the covariate-specific optimal pair of thresholds for clustered data.</p></a></li>
<li><a href='#clus_roc_surface'><p>Plot an estimated covariate-specific ROC surface for clustered data.</p></a></li>
<li><a href='#clus_tcfs'><p>Estimation of the covariate-specific TCFs for clustered data.</p></a></li>
<li><a href='#clus_vus'><p>Estimation of the covariate-specific VUS for clustered data.</p></a></li>
<li><a href='#data_3class'><p>A simulated data</p></a></li>
<li><a href='#data_3class_bcx'><p>A simulated data</p></a></li>
<li><a href='#EnergyEthiopia'><p>A subset of energy choice data in 4 cities of Ethiopia</p></a></li>
<li><a href='#MouseNeurons'><p>A subset of mouse brain cells data</p></a></li>
<li><a href='#plot.clus_lme'><p>Plot an clus_lme object.</p></a></li>
<li><a href='#plot.clus_opt_thres3'><p>Plot of confidence regions for covariate-specific optimal pair of thresholds.</p></a></li>
<li><a href='#print.ci_clus_vus'><p>Print summary results from ci_clus_vus</p></a></li>
<li><a href='#print.clus_lme'><p>Print summary results of an clus_lme object</p></a></li>
<li><a href='#print.clus_opt_thres3'><p>Print summary results from <code>clus_opt_thres3</code></p></a></li>
<li><a href='#print.clus_tcfs'><p>Print summary results from clus_tcfs</p></a></li>
<li><a href='#print.clus_vus'><p>Print summary results from clus_vus</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>ROC Analysis in Three-Class Classification Problems for
Clustered Data</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-17</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Duc-Khanh To [aut, cre] (&lt;https://orcid.org/0000-0002-4641-0764&gt;), with contributions from Gianfranco Adimari and Monica Chiogna</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Duc-Khanh To &lt;toduc@stat.unipd.it&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical methods for ROC surface analysis in three-class classification problems for clustered data and in presence of covariates. In particular, the package allows to obtain covariate-specific point and interval estimation for:
  (i) true class fractions (TCFs) at fixed pairs of thresholds;
  (ii) the ROC surface;
  (iii) the volume under ROC surface (VUS);
  (iv) the optimal pairs of thresholds.
  Methods considered in points (i), (ii) and (iv) are proposed and discussed in To et al. (2022) &lt;<a href="https://doi.org/10.1177%2F09622802221089029">doi:10.1177/09622802221089029</a>&gt;. Referring to point (iv), three  different selection criteria are implemented: Generalized Youden Index (GYI), Closest to Perfection (CtP) and Maximum Volume (MV). Methods considered in point (iii) are proposed and discussed in Xiong et al. (2018) &lt;<a href="https://doi.org/10.1177%2F0962280217742539">doi:10.1177/0962280217742539</a>&gt;. Visualization tools are also provided. We refer readers to the articles cited above for all details. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), stats, utils, graphics, nlme, Rcpp (&ge; 0.12.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>rgl, ellipse, numDeriv, ggplot2, ggpubr, foreach, iterators,
parallel, doParallel</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/toduckhanh/ClusROC">https://github.com/toduckhanh/ClusROC</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/toduckhanh/ClusROC/issues">https://github.com/toduckhanh/ClusROC/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-17 14:07:13 UTC; toduc</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-17 15:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ClusROC-package'>
ROC Analysis in Three-Class Classification Problems for Clustered Data
</h2><span id='topic+ClusROC-package'></span><span id='topic+ClusROC'></span>

<h3>Description</h3>

<p>This package implements the techniques for ROC surface analysis, in cases of clustered data and in presence of covariates. In particular, the package allows to obtain covariate-specific point and interval estimation for:
(i) true class fractions (TCFs) at fixed pairs of thresholds;
(ii) the ROC surface;
(iii) the volume under ROC surface (VUS);
(iv) the optimal pairs of thresholds.
Methods considered in points (i), (ii) and (iv) are proposed and discussed in To et al. (2022). Referring to point (iv), three  different selection criteria are implemented: Generalized Youden Index (GYI), Closest to Perfection (CtP) and Maximum Volume (MV). Methods considered in point (iii) are proposed and discussed in Xiong et al. (2018). Visualization tools are also provided. We refer readers to the articles cited above for all details.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ClusROC</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0-2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-10-10</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 2 | GPL 3</td>
</tr>
<tr>
 <td style="text-align: left;">
Lazy load: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Major functions are <code><a href="#topic+clus_lme">clus_lme</a></code>, <code><a href="#topic+clus_roc_surface">clus_roc_surface</a></code>, <code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code>, <code><a href="#topic+clus_vus">clus_vus</a></code> and <code><a href="#topic+clus_tcfs">clus_tcfs</a></code>.
</p>


<h3>Author(s)</h3>

<p>Duc-Khanh To, with contributions from Gianfranco Adimari and Monica Chiogna
</p>
<p>Maintainer: Duc-Khanh To &lt;toduc@stat.unipd.it&gt;
</p>


<h3>References</h3>

<p>Bantis, L. E., Nakas, C. T., Reiser, B., Myall, D., and Dalrymple-Alford, J. C. (2017). Construction of joint confidence regions for the optimal true class fractions of Receiver Operating Characteristic (ROC) surfaces and manifolds. <em>Statistical methods in medical research</em>, <b>26</b>, 3, 1429-1442.
</p>
<p>Gurka, M. J., Edwards, L. J. , Muller, K. E., and Kupper, L. L. (2006). Extending the Box-Cox transformation to the linear mixed model. <em>Journal of the Royal Statistical Society: Series A (Statistics in Society)</em>, <b>169</b>, 2, 273-288.
</p>
<p>Gurka, M. J. and Edwards, L. J. (2011). Estimating variance components and random effects using the box-cox transformation in the linear mixed model. <em>Communications in Statistics - Theory and Methods</em>, <b>40</b>, 3, 515-531.
</p>
<p>Kauermann, G. and Carroll, R. J. (2001). A note on the efficiency of sandwich covariance matrix estimation. <em>Journal of the American Statistical Association</em>, <b>96</b>, 456, 1387-1396.
</p>
<p>Liang, K. Y. and Zeger, S. L. (1986). Longitudinal data analysis using generalized linear models. <em>Biometrika</em>, <b>73</b>, 1, 13-22.
</p>
<p>Mancl, L. A. and DeRouen, T. A. (2001). A covariance estimator for GEE with improved small-sample properties. <em>Biometrics</em>, <b>57</b>, 1, 126-134.
</p>
<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022). Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>
<p>Xiong, C., Luo, J., Chen L., Gao, F., Liu, J., Wang, G., Bateman, R. and Morris, J. C. (2018). Estimating diagnostic accuracy for clustered ordinal diagnostic groups in the three-class case &ndash; Application to the early diagnosis of Alzheimer disease. <em>Statistical Methods in Medical Research</em>, <b>27</b>, 3, 701-714.
</p>

<hr>
<h2 id='ci_clus_vus'>Confidence Intervals for Covariate-specific VUS</h2><span id='topic+ci_clus_vus'></span>

<h3>Description</h3>

<p>Computes confidence intervals for covariate-specific VUS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci_clus_vus(x, ci_level = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ci_clus_vus_+3A_x">x</code></td>
<td>
<p>an object of class &quot;VUS&quot;, a result of <code><a href="#topic+clus_vus">clus_vus</a></code> call.</p>
</td></tr>
<tr><td><code id="ci_clus_vus_+3A_ci_level">ci_level</code></td>
<td>
<p>a confidence level to be used for constructing the confidence interval; default is 0.95.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A confidence interval for covariate-specific VUS is given based on normal approximation. If the lower bound (or the upper bound) of the confidence interval is smaller than 0 (or greater than 1), it will be set as 0 (or 1). Also, logit and probit transformations are available if one wants guarantees that confidence limits are inside (0, 1).
</p>


<h3>Value</h3>

<p><code>ci_clus_vus</code> returns an object of class inheriting from &quot;ci_VUS&quot; class. An object of class &quot;ci_VUS&quot; is a list, containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>ci_vus_norm</code></td>
<td>
<p>the normal approximation-based confidence interval for covariate-specific VUS.</p>
</td></tr>
<tr><td><code>ci_vus_log</code></td>
<td>
<p>the confidence interval for covariate-specific VUS, after using logit-transformation.</p>
</td></tr>
<tr><td><code>ci_vus_prob</code></td>
<td>
<p>the confidence interval for covariate-specific VUS, after using probit-transformation.</p>
</td></tr>
<tr><td><code>ci_level</code></td>
<td>
<p>fixed confidence level.</p>
</td></tr>
<tr><td><code>newdata</code></td>
<td>
<p>value(s) of covariate(s).</p>
</td></tr>
<tr><td><code>n_p</code></td>
<td>
<p>total numbers of the regressors in the model.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+clus_vus">clus_vus</a></code>
</p>

<hr>
<h2 id='clus_lme'>Linear Mixed-Effects Models for a continuous diagnostic test or a biomarker (or a classifier).</h2><span id='topic+clus_lme'></span>

<h3>Description</h3>

<p><code>clus_lme</code> fits the cluster-effect model for a continuous diagnostic test in a three-class setting as described in Xiong et al. (2018) and To et al. (2022).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clus_lme(
  fixed_formula,
  name_class,
  name_clust,
  data = sys.frame(sys.parent()),
  subset,
  na_action = na.fail,
  levl_class = NULL,
  ap_var = TRUE,
  boxcox = FALSE,
  interval_lambda = c(-2, 2),
  trace = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clus_lme_+3A_fixed_formula">fixed_formula</code></td>
<td>
<p>a two-sided linear formula object, describing the fixed-effects part of the model for three classes, with the response on the left of ~ operator and the terms, separated by + operators, on the right. For example, <code>Y ~ X1 + X2</code>, <code>Y ~ X1 + X2 + X1:X2</code> or <code>log(Y) ~ X1 + X2 + I(X1^2)</code>.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_name_class">name_class</code></td>
<td>
<p>name of variable indicating three classes (or three groups) in the data.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_name_clust">name_clust</code></td>
<td>
<p>name of variable indicating clusters in the data.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_data">data</code></td>
<td>
<p>a data frame containing the variables in the model.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_subset">subset</code></td>
<td>
<p>an optional expression indicating the subset of the rows of data that should be used in the fit. This can be a logical vector, or a numeric vector indicating which observation numbers are to be included, or a character vector of the row names to be included. All observations are included by default.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_na_action">na_action</code></td>
<td>
<p>a function that indicates what should happen when the data contain NAs. The default action (<code><a href="stats.html#topic+na.fail">na.fail</a></code>) causes <code>clus_lme</code> to print an error message and terminate if there are any incomplete observations.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_levl_class">levl_class</code></td>
<td>
<p>a vector (of strings) containing the ordered name chosen for the disease classes. The ordering is intended to be &ldquo;increasing&rdquo; with respect to the disease severity. If <code>levl_class = NULL</code> (default), the elements of the vector will be automatically determined from data, by considering the order of the means of the test values for each disease class (diagnostic group).</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_ap_var">ap_var</code></td>
<td>
<p>a logical value. Default = <code>TRUE</code>. If set to <code>TRUE</code>, the estimated covariance matrix for all estimated parameters in the model will be obtained (by using the sandwich formula).</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_boxcox">boxcox</code></td>
<td>
<p>a logical value. Default = <code>FALSE</code>. If set to <code>TRUE</code>, a Box-Cox transformation will be applied to the model.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_interval_lambda">interval_lambda</code></td>
<td>
<p>a vector containing the end-points of the interval for searching the Box-Cox parameter, <code>lambda</code>. Default = (-2, 2).</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_trace">trace</code></td>
<td>
<p>a logical value. Default = <code>TRUE</code>. If set to <code>TRUE</code>, the information about the check for the monotonic ordering of test values will be provided.</p>
</td></tr>
<tr><td><code id="clus_lme_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="nlme.html#topic+lme">lme</a></code>, such as <code>control</code>, <code>contrasts</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function fits a linear mixed-effect model for a continuous diagnostic test in a three-class setting in order to account for the cluster and covariates effects on the test result. See Xiong et al. (2018) and To et al. (2022) for more details.
</p>

<ul>
<li><p> Estimation is done by using <code><a href="nlme.html#topic+lme">lme</a></code> with the restricted maximum log-likelihood (REML) method.
</p>
</li>
<li><p> Box-Cox transformation for the model can be used when the distributions of test results are skewed (Gurka et al. 2006). The estimation procedure is described in To et al. (2022). The Box-Cox parameter <code class="reqn">\lambda</code> is estimated by a grid search on the interval (-2, 2), as discussed in Gurka and Edwards (2011).
</p>
</li>
<li><p> The estimated variance-covariance matrix for the estimated parameters are obtained by sandwich formula (see, Liang and Zeger, 1986; Kauermann and Carroll, 2001; Mancl and DeRouen, 2001) as discussed in To et al. (2022).
</p>
</li></ul>



<h3>Value</h3>

<p><code>clus_lme</code> returns an object of class &quot;clus_lme&quot; class, i.e., a list containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>est_para</code></td>
<td>
<p>a vector containing the estimated parameters.</p>
</td></tr>
<tr><td><code>se_para</code></td>
<td>
<p>a vector containing the standard errors.</p>
</td></tr>
<tr><td><code>vcov_sand</code></td>
<td>
<p>the estimated covariance matrix for all estimated parameters.</p>
</td></tr>
<tr><td><code>residual</code></td>
<td>
<p>a list of residuals.</p>
</td></tr>
<tr><td><code>fitted</code></td>
<td>
<p>a list of fitted values.</p>
</td></tr>
<tr><td><code>randf</code></td>
<td>
<p>a vector of estimated random effects for each cluster level.</p>
</td></tr>
<tr><td><code>n_coef</code></td>
<td>
<p>total number of coefficients included in the model.</p>
</td></tr>
<tr><td><code>n_p</code></td>
<td>
<p>total numbers of regressors in the model.</p>
</td></tr>
<tr><td><code>icc</code></td>
<td>
<p>an estimate of intra-class correlation - ICC</p>
</td></tr>
<tr><td><code>terms</code></td>
<td>
<p>the <code><a href="stats.html#topic+terms">terms</a></code> object used.</p>
</td></tr>
<tr><td><code>boxcox</code></td>
<td>
<p>logical value indicating whether the Box-Cox transformation was applied or not.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>data frame is used to fitting model.</p>
</td></tr>
</table>
<p>Generic functions such as <code>print</code> and <code>plot</code> are also used to show results of the fit.
</p>


<h3>References</h3>

<p>Gurka, M. J., Edwards, L. J. , Muller, K. E., and Kupper, L. L. (2006) &ldquo;Extending the Box-Cox transformation to the linear mixed model&rdquo;. <em>Journal of the Royal Statistical Society: Series A (Statistics in Society)</em>, <b>169</b>, 2, 273-288.
</p>
<p>Gurka, M. J. and Edwards, L. J. (2011) &ldquo;Estimating variance components and random effects using the box-cox transformation in the linear mixed model&rdquo;. <em>Communications in Statistics - Theory and Methods</em>, <b>40</b>, 3, 515-531.
</p>
<p>Kauermann, G. and Carroll, R. J. (2001)
&ldquo;A note on the efficiency of sandwich covariance matrix estimation&rdquo;.
<em>Journal of the American Statistical Association</em>, <b>96</b>, 456, 1387-1396.
</p>
<p>Liang, K. Y. and Zeger, S. L. (1986)
&ldquo;Longitudinal data analysis using generalized linear models&rdquo;. <em>Biometrika</em>, <b>73</b>, 1, 13-22.
</p>
<p>Mancl, L. A. and DeRouen, T. A. (2001) &ldquo;A covariance estimator for GEE with improved small-sample properties&rdquo;.
<em>Biometrics</em>, <b>57</b>, 1, 126-134.
</p>
<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022)
&ldquo;Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data&rdquo;. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>
<p>Xiong, C., Luo, J., Chen L., Gao, F., Liu, J., Wang, G., Bateman, R. and Morris, J. C. (2018)
&ldquo;Estimating diagnostic accuracy for clustered ordinal diagnostic groups in the three-class case &ndash; Application to the early diagnosis of Alzheimer disease&rdquo;.
<em>Statistical Methods in Medical Research</em>, <b>27</b>, 3, 701-714.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1:
data(data_3class)
head(data_3class)
## A model with two covariate
out1 &lt;- clus_lme(fixed_formula = Y ~ X1 + X2, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)
print(out1)
plot(out1)


## Example 2: Box-Cox transformation
data(data_3class_bcx)
out2 &lt;- clus_lme(fixed_formula = Y ~ X, name_class = "D",
                 name_clust = "id_Clus", data = data_3class_bcx,
                 boxcox = TRUE)
print(out2)
plot(out2)


</code></pre>

<hr>
<h2 id='clus_opt_thres3'>Estimation of the covariate-specific optimal pair of thresholds for clustered data.</h2><span id='topic+clus_opt_thres3'></span>

<h3>Description</h3>

<p><code>clus_opt_thres3</code> estimates covariate-specific optimal pair of thresholds of a continuous diagnostic test in a clustered design, with three classes of diseases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clus_opt_thres3(
  method = c("GYI", "CtP", "MV"),
  out_clus_lme,
  newdata,
  ap_var = TRUE,
  control = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clus_opt_thres3_+3A_method">method</code></td>
<td>
<p>the method to be used. See 'Details'.</p>
</td></tr>
<tr><td><code id="clus_opt_thres3_+3A_out_clus_lme">out_clus_lme</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, i.e., a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="clus_opt_thres3_+3A_newdata">newdata</code></td>
<td>
<p>a data frame (containing specific value(s) of covariate(s)) in which to look for variables with which to estimate covariate-specific optimal pair of thresholds. In absence of covariate, no values have to be specified.</p>
</td></tr>
<tr><td><code id="clus_opt_thres3_+3A_ap_var">ap_var</code></td>
<td>
<p>logical value. If set to <code>TRUE</code>, the variance-covariance matrix of (estimated) covariate-specific optimal thresholds is estimated.</p>
</td></tr>
<tr><td><code id="clus_opt_thres3_+3A_control">control</code></td>
<td>
<p>a list of control parameters. See 'Details'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements estimation methods discussed in To et al. (2022) for covariate-specific optimal pair of thresholds in a clustered design with three ordinal groups. The estimators are based on the results from <code><a href="#topic+clus_lme">clus_lme</a></code> function, which fits the linear mixed-effect model by using REML approach.
</p>
<p>Before performing estimation, a check for the monotone ordering assumption is performed. This means that, for the fixed values of covariates, three predicted mean values for test results in three diagnostic groups are compared. If the assumption is not meet, the covariate-specific optimal pair of thresholds at the values of covariates are not estimated.
</p>
<p>The estimation procedure uses three criteria. Method <code>"GYI"</code> is Generalized Youden Index, which maximizes the sum of three covariate-specific True Class Fractions - TCFs. Method <code>"CtP"</code> is based on Closest to Pefection approach. By using this method, the optimal pair of thresholds is obtained by minimizing the distance, in the unit cube, between a generic point on the covariate-specific ROC surface and the top corner (1, 1, 1). Method <code>"MV"</code> is based on Maximum Volume approach, which searches for thresholds that maximize the volume of a box under the covariate-specific ROC surface. The user can select more than one method. This function allows to estimate covariate-specific optimal pair of thresholds at multiple points for covariates.
</p>
<p>The asymptotic variance-covariance matrix of the (estimated) covariate-specific optimal thresholds is estimated by using the Delta method under the normal assumption. If the Box-Cox transformation is applied to the linear mixed-effect model, a nonparametric bootstrap procedure for clustered data will be used to obtain the estimated asymptotic covariance matrix (see To et al. 2022, for more details).
</p>
<p>The <code>control</code> argument is a list that can supply any of the following components:
</p>

<dl>
<dt><code>method_optim</code></dt><dd><p>Optimization method to be used. There are three options: <code>"L-BFGS-B"</code>, <code>"BFGS"</code> and <code>"Nelder-Mead"</code>. Default is <code>"L-BFGS-B"</code>.</p>
</dd>
<dt><code>start</code></dt><dd><p>Starting values in the optimization procedure. If it is <code>NULL</code>, a starting point will be automatically obtained.</p>
</dd>
<dt><code>maxit</code></dt><dd><p>The maximum number of iterations. Default is 200.</p>
</dd>
<dt><code>lower, upper</code></dt><dd><p>Possible bounds on the threshold range, for the optimization based on &quot;L-BFGS-B&quot; method. Defaults are <code>-Inf</code> and <code>Inf</code>.</p>
</dd>
<dt><code>n_boot</code></dt><dd><p>Number of bootstrap replicates for estimating the covariance matrix (when Box-Cox transformation is applied). Default is 250.</p>
</dd>
<dt><code>parallel</code></dt><dd><p>A logical value. If set to <code>TRUE</code>, a parallel computing is employed in the bootstrap resampling process.</p>
</dd>
<dt><code>ncpus</code></dt><dd><p>Number of processes to be used in parallel computing. Default is 2.</p>
</dd>
</dl>



<h3>Value</h3>

<p><code>clus_opt_thres3</code> returns an object of &quot;clus_opt_thres3&quot; class, which is a list containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the methods used to obtain the estimated optimal pair of threholds.</p>
</td></tr>
<tr><td><code>thres3</code></td>
<td>
<p>a vector or matrix containing the estimated optimal thresholds.</p>
</td></tr>
<tr><td><code>thres3_se</code></td>
<td>
<p>a vector or matrix containing the estimated standard errors.</p>
</td></tr>
<tr><td><code>vcov_thres3</code></td>
<td>
<p>a matrix or list of matrices containing the estimated variance-covariance matrices.</p>
</td></tr>
<tr><td><code>tcfs</code></td>
<td>
<p>a vector or matrix containing the estimated TCFs at the optimal thresholds.</p>
</td></tr>
<tr><td><code>mess_order</code></td>
<td>
<p>a diagnostic message from checking the monontone ordering.</p>
</td></tr>
<tr><td><code>newdata</code></td>
<td>
<p>value(s) of covariate(s).</p>
</td></tr>
<tr><td><code>n_p</code></td>
<td>
<p>total number of regressors in the model.</p>
</td></tr>
</table>
<p>Generic functions such as <code>print</code> and <code>plot</code> are also used to show the results.
</p>


<h3>References</h3>

<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022)
&ldquo;Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data&rdquo;. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_3class)
## One covariate
out1 &lt;- clus_lme(fixed_formula = Y ~ X1, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate covariate-specific optimal thresholds at multiple values of one covariate,
### with 3 methods
out_thres_1 &lt;- clus_opt_thres3(method = c("GYI", "MV", "CtP"),
                               out_clus_lme = out1,
                               newdata = data.frame(X1 = 1), ap_var = TRUE)
print(out_thres_1)
plot(out_thres_1)

## Two covariates
out2 &lt;- clus_lme(fixed_formula = Y ~ X1 + X2, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate covariate-specific optimal thresholds at one point, with 3 methods
out_thres_2 &lt;- clus_opt_thres3(method = c("GYI", "MV", "CtP"),
                               out_clus_lme = out2,
                               newdata = data.frame(X1 = 1, X2 = 0),
                               ap_var = TRUE)
print(out_thres_2)
plot(out_thres_2)

### Estimate covariate-specific optimal thresholds at three points, with 3 methods
out_thres_3 &lt;- clus_opt_thres3(method = c("GYI", "MV", "CtP"),
                               out_clus_lme = out2,
                               newdata = data.frame(X1 = c(-0.5, 0.5, 0.5),
                                                    X2 = c(0, 0, 1)),
                               ap_var = TRUE)
print(out_thres_3)
plot(out_thres_3, colors = c("forestgreen", "blue"))

</code></pre>

<hr>
<h2 id='clus_roc_surface'>Plot an estimated covariate-specific ROC surface for clustered data.</h2><span id='topic+clus_roc_surface'></span>

<h3>Description</h3>

<p><code>clus_roc_surface</code> estimates and makes a 3D plot of a covariate-specific ROC surface for a continuous diagnostic test, in a clustered design, with three ordinal groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clus_roc_surface(
  out_clus_lme,
  newdata,
  step_tcf = 0.01,
  main = NULL,
  file_name = NULL,
  ellips = FALSE,
  thresholds = NULL,
  ci_level = ifelse(ellips, 0.95, NULL)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clus_roc_surface_+3A_out_clus_lme">out_clus_lme</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_newdata">newdata</code></td>
<td>
<p>a data frame with 1 row (containing specific value(s) of covariate(s)) in which to look for variables with which to estimate covariate-specific ROC. In absence of covariate, no values have to be specified.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_step_tcf">step_tcf</code></td>
<td>
<p>number: increment to be used in the grid for <code class="reqn">p1 = tcf1</code> and <code class="reqn">p3 = tcf3</code>.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_main">main</code></td>
<td>
<p>the main title for plot.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_file_name">file_name</code></td>
<td>
<p>File name to create on disk.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_ellips">ellips</code></td>
<td>
<p>a logical value. If set to <code>TRUE</code>, the function adds an ellipsoidal confidence region for TCFs (True Class Fractions), at a specified pair of values for the thresholds, to the plot of estimated covariate-specific ROC surface.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_thresholds">thresholds</code></td>
<td>
<p>a specified pair of thresholds, used to construct the ellipsoidal confidence region for TCFs.</p>
</td></tr>
<tr><td><code id="clus_roc_surface_+3A_ci_level">ci_level</code></td>
<td>
<p>a confidence level to be used for constructing the ellipsoidal confidence region; default is 0.95.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a method in To et al. (2022) for estimating covariate-specific ROC surface of a continuous diagnostic test in a clustered design, with three ordinal groups. The estimator is based on the results from <code><a href="#topic+clus_lme">clus_lme</a></code> with REML approach.
</p>
<p>Before performing estimation, a check for the monotone ordering assumption is performed. This means that, for the fixed values of covariates, three predicted mean values for test results in three diagnostic groups are compared. If the assumption is not meet, the covariate-specific ROC surface at the values of covariates is not estimated.
</p>
<p>The ellipsoidal confidence region for TCFs at a given pair of thresholds, if required, is constructed by using normal approximation and is plotted in the ROC surface space. The confidence level (default) is 0.95. Note that, if the Box-Cox transformation is applied for the linear mixed-effect model, the pair of thresholds must be input in the original scale. If the constructed confidence region for TCFs is outside the unit cube, a probit transformation will be automatically applied to obtain an appropriate confidence region, which is inside the unit cube (see Bantis et. al., 2017).
</p>


<h3>Value</h3>

<p><code>clus_roc_surface</code> returns a 3D <code>rgl</code> plot of the estimated covariate-specific ROC surface.
</p>


<h3>References</h3>

<p>Bantis, L. E., Nakas, C. T., Reiser, B., Myall, D., and Dalrymple-Alford, J. C. (2017).
&ldquo;Construction of joint confidence regions for the optimal true class fractions of Receiver Operating Characteristic (ROC) surfaces and manifolds&rdquo;. <em>Statistical methods in medical research</em>, <b>26</b>, 3, 1429-1442.
</p>
<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022)
&ldquo;Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data&rdquo;. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(data_3class)
## One covariate
out1 &lt;- clus_lme(fixed_formula = Y ~ X1, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### plot only covariate-specific ROC surface
clus_roc_surface(out_clus_lme = out1, newdata = data.frame(X1 = 1))

### plot covariate-specific ROC surface and a 95% ellipsoidal confidence region for TCFs
clus_roc_surface(out_clus_lme = out1, newdata = data.frame(X1 = 1),
                 ellips = TRUE, thresholds = c(0.9, 3.95))

## Two covariates
out2 &lt;- clus_lme(fixed_formula = Y ~ X1 + X2, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### plot only covariate-specific ROC surface
clus_roc_surface(out_clus_lme = out2, newdata = data.frame(X1 = 1, X2 = 1))

### plot covariate-specific ROC surface and a 95% ellipsoidal confidence region for TCFs
clus_roc_surface(out_clus_lme = out2, newdata = data.frame(X1 = 1, X2 = 1),
                 ellips = TRUE, thresholds = c(0.9, 3.95))


</code></pre>

<hr>
<h2 id='clus_tcfs'>Estimation of the covariate-specific TCFs for clustered data.</h2><span id='topic+clus_tcfs'></span>

<h3>Description</h3>

<p><code>clus_tcfs</code> estimates covariate-specific True Class Fractions (TCFs), at a specified pair of thresholds, of a continuous diagnostic test in a clustered design with three ordinal groups. This function allows to estimate covariate-specific TCFs at multiple points for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clus_tcfs(out_clus_lme, newdata, thresholds, ap_var = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clus_tcfs_+3A_out_clus_lme">out_clus_lme</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="clus_tcfs_+3A_newdata">newdata</code></td>
<td>
<p>a data frame (containing specific value(s) of covariate(s)) in which to look for variables with which to estimate covariate-specific TCFs. In absence of covariate, no values have to be specified.</p>
</td></tr>
<tr><td><code id="clus_tcfs_+3A_thresholds">thresholds</code></td>
<td>
<p>a specified pair of thresholds.</p>
</td></tr>
<tr><td><code id="clus_tcfs_+3A_ap_var">ap_var</code></td>
<td>
<p>logical value. If set to <code>TRUE</code>, the variance-covariance matrix of estimated covariate-specific TCFs is estimated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a method in To et al. (2022) for estimating covariate-specific TCFs at a specified pair of thresholds of a continuous diagnostic test in a clustered design with three ordinal groups. The estimator is based on results from  <code><a href="#topic+clus_lme">clus_lme</a></code>, which uses the REML approach. The asymptotic variance-covariance matrix of the estimated covariate-specific TCFs is estimated through the Delta method. Note that, if the Box-Cox transformation is applied for the linear mixed-effect model, the pair of thresholds must be input in the original scale.
</p>
<p>Before performing estimation, a check for the monotone ordering assumption is performed. This means that, for the fixed values of covariates, three predicted mean values for test results in three diagnostic groups are compared. If the assumption is not meet, the covariate-specific TCFs at the values of covariates are not estimated.
</p>


<h3>Value</h3>

<p><code>TCFs</code> returns an object of class &quot;TCFs&quot;, which is a list containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>tcfs_est</code></td>
<td>
<p>a vector or matrix containing the estimated TCFs.</p>
</td></tr>
<tr><td><code>tcf_vcov</code></td>
<td>
<p>a matrix or list of matrices containing the estimated variance-covariance matrices.</p>
</td></tr>
<tr><td><code>thresholds</code></td>
<td>
<p>specified pair of thresholds.</p>
</td></tr>
<tr><td><code>mess_order</code></td>
<td>
<p>a diagnostic message from checking the monontone ordering.</p>
</td></tr>
<tr><td><code>newdata</code></td>
<td>
<p>value(s) of covariate(s).</p>
</td></tr>
<tr><td><code>n_p</code></td>
<td>
<p>total number of regressors in the model.</p>
</td></tr>
</table>
<p>Generic functions such as <code>print</code> is also used to show the results.
</p>


<h3>References</h3>

<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022)
&ldquo;Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data&rdquo;. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_3class)
## One covariate
out1 &lt;- clus_lme(fixed_formula = Y ~ X1, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate TCFs at one single value of X1, (t1, t2) = (1, 4)
out_tcfs_1 &lt;- clus_tcfs(out_clus_lme = out1, newdata = data.frame(X1 = 1),
                        thresholds = c(1, 4), ap_var = TRUE)
print(out_tcfs_1)

## Two covariates
out2 &lt;- clus_lme(fixed_formula = Y ~ X1 + X2, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate covariate-specific TCFs at point (X1, X2) = (1, 0), and (t1, t2) = (1, 4)
out_tcfs_2 &lt;- clus_tcfs(out_clus_lme = out2,
                        newdata = data.frame(X1 = 1, X2 = 0),
                        thresholds = c(1, 4), ap_var = TRUE)
print(out_tcfs_2)

### Estimate covariate-specific TCFs at three points and (t1, t2) = (1, 4)
out_tcfs_3 &lt;- clus_tcfs(out_clus_lme = out2,
                        newdata = data.frame(X1 = c(-0.5, 0.5, 0.5),
                                             X2 = c(0, 0, 1)),
                        thresholds = c(1, 4), ap_var = TRUE)
print(out_tcfs_3)

</code></pre>

<hr>
<h2 id='clus_vus'>Estimation of the covariate-specific VUS for clustered data.</h2><span id='topic+clus_vus'></span>

<h3>Description</h3>

<p>This function estimates the covariate-specific VUS of a continuous diagnostic test in the setting of clustered data as described in Xiong et al. (2018). This function allows to estimate covariate-specific VUS at multiple points for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clus_vus(out_clus_lme, newdata, ap_var = TRUE, subdivisions = 1000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clus_vus_+3A_out_clus_lme">out_clus_lme</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="clus_vus_+3A_newdata">newdata</code></td>
<td>
<p>a data frame (containing specific value(s) of covariate(s)) in which to look for variables with which to estimate covariate-specific VUS. In absence of covariate, no values have to be specified.</p>
</td></tr>
<tr><td><code id="clus_vus_+3A_ap_var">ap_var</code></td>
<td>
<p>logical value. If set to <code>TRUE</code> (default), the standard error for (estimated) covariate-specific VUS are estimated.</p>
</td></tr>
<tr><td><code id="clus_vus_+3A_subdivisions">subdivisions</code></td>
<td>
<p>the maximum number of subintervals used to approximate integral. Default is 1000.</p>
</td></tr>
<tr><td><code id="clus_vus_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to <code><a href="stats.html#topic+integrate">integrate</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements a method in Xiong et al. (2018) for estimating covariate-specific VUS of a continuous diagnostic test in a clustered design with three ordinal groups. The estimator is based on results from <code><a href="#topic+clus_lme">clus_lme</a></code>, which uses the REML approach. The standard error of the estimated covariate-specific VUS is approximated through the Delta method.
</p>
<p>Before performing estimation, a check for the monotone ordering assumption is performed. This means that, for the fixed values of covariates, three predicted mean values for test results in three diagnostic groups are compared. If the assumption is not meet, the covariate-specific VUS at the values of covariates are not estimated. In addition, this function also performs the statistical test, <code class="reqn">H_0: VUS = 1/6</code> versus an alternative of interest.
</p>


<h3>Value</h3>

<p><code>clus_vus</code> returns an object of class &quot;VUS&quot; which is a list containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>the matched call.</p>
</td></tr>
<tr><td><code>vus_est</code></td>
<td>
<p>a vector containing the estimated covariate-specific VUS.</p>
</td></tr>
<tr><td><code>vus_se</code></td>
<td>
<p>a vector containing the standard errors.</p>
</td></tr>
<tr><td><code>mess_order</code></td>
<td>
<p>a diagnostic message from checking the monontone ordering.</p>
</td></tr>
<tr><td><code>newdata</code></td>
<td>
<p>value(s) of covariate(s).</p>
</td></tr>
<tr><td><code>n_p</code></td>
<td>
<p>total number of regressors in the model.</p>
</td></tr>
</table>
<p>Generic functions such as <code>print</code> is also used to show the results.
</p>


<h3>References</h3>

<p>Xiong, C., Luo, J., Chen L., Gao, F., Liu, J., Wang, G., Bateman, R. and Morris, J. C. (2018)
&ldquo;Estimating diagnostic accuracy for clustered ordinal diagnostic groups in the three-class case &ndash; Application to the early diagnosis of Alzheimer disease&rdquo;.
<em>Statistical Methods in Medical Research</em>, <b>27</b>, 3, 701-714.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_3class)
## One covariate
out1 &lt;- clus_lme(fixed_formula = Y ~ X1, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate covariate-specific VUS at one value of one covariate
out_vus1 &lt;- clus_vus(out1, newdata = data.frame(X1 = 0.5))
ci_clus_vus(out_vus1, ci_level = 0.95)

### Estimate covariate-specific VUS at multiple values of one covariate
out_vus2 &lt;- clus_vus(out1, newdata = data.frame(X1 = c(-0.5, 0, 0.5)))
ci_clus_vus(out_vus2, ci_level = 0.95)

## Two covariates
out2 &lt;- clus_lme(fixed_formula = Y ~ X1 + X2, name_class = "D",
                 name_clust = "id_Clus", data = data_3class)

### Estimate covariate-specific VUS at one point
out_vus3 &lt;- clus_vus(out2, newdata = data.frame(X1 = 1.5, X2 = 1))
ci_clus_vus(out_vus3, ci_level = 0.95)

### Estimate covariate-specific VUS at three points
out_vus4 &lt;- clus_vus(out2, newdata = data.frame(X1 = c(-0.5, 0.5, 0.5),
                                                X2 = c(0, 0, 1)))
ci_clus_vus(out_vus4, ci_level = 0.95)

</code></pre>

<hr>
<h2 id='data_3class'>A simulated data</h2><span id='topic+data_3class'></span>

<h3>Description</h3>

<p>A simulated data example with 30 clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_3class)
</code></pre>


<h3>Format</h3>

<p>A data frame with 225 observations (from 30 clusters).
</p>

<dl>
<dt><code>id_Clus</code></dt><dd><p>the id number of cluster.</p>
</dd>
<dt><code>Y</code></dt><dd><p>a vector containing test results.</p>
</dd>
<dt><code>D</code></dt><dd><p>a factor with 3 levels for the disease status, 1, 2, 3. The levels correspond to benign disease, early stage and late stage.</p>
</dd>
<dt><code>X1</code></dt><dd><p>a continuous covariate.</p>
</dd>
<dt><code>X2</code></dt><dd><p>a binary covariate.</p>
</dd>
</dl>


<hr>
<h2 id='data_3class_bcx'>A simulated data</h2><span id='topic+data_3class_bcx'></span>

<h3>Description</h3>

<p>A simulated data example with 60 clusters. This dataset is used in a example of analysis with Box-Cox transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_3class_bcx)
</code></pre>


<h3>Format</h3>

<p>A data frame with 582 observations (from 60 clusters).
</p>

<dl>
<dt><code>id_Clus</code></dt><dd><p>the id number of cluster.</p>
</dd>
<dt><code>Y</code></dt><dd><p>a vector containing test results.</p>
</dd>
<dt><code>D</code></dt><dd><p>a factor with 3 levels for disease status, 1, 2, 3. The levels correspond to benign disease, early stage and late stage.</p>
</dd>
<dt><code>X</code></dt><dd><p>a continuous covariate.</p>
</dd>
</dl>


<hr>
<h2 id='EnergyEthiopia'>A subset of energy choice data in 4 cities of Ethiopia</h2><span id='topic+EnergyEthiopia'></span>

<h3>Description</h3>

<p>A subset of energy choice data used in Alem et al. (2016). The authors are used the full dataset to investigate the determinants of household cooking fuel choice and energy transition in urban Ethiopia. A full data is publicly available at <a href="https://doi.org/10.1016/j.eneco.2016.06.025">doi:10.1016/j.eneco.2016.06.025</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(EnergyEthiopia)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2088 observations from 1123 households (or clusters) in the capital Addis Ababa and 9 variables:
</p>

<dl>
<dt><code>uqid</code></dt><dd><p>the id of household (which yield 1123 clusters).</p>
</dd>
<dt><code>energy2</code></dt><dd><p>a factor with 3 levels (types) of cooking energy state at each time (2000, 2004, 2009), i.e., 1 (clean fuel only - electricity, gas and kerosene), 2 (a mix of clean and biomass), 3 (biomass fuel only - firewood, charcoal, dung and crop residues).</p>
</dd>
<dt><code>hhs</code></dt><dd><p>household size.</p>
</dd>
<dt><code>hhs_ft</code></dt><dd><p>a factor with 4 levels of household size: small (1 <code class="reqn">\le</code> hhs <code class="reqn">\le</code> 4); medium (5 <code class="reqn">\le</code> hhs <code class="reqn">\le</code> 8); large ((9 <code class="reqn">\le</code> hhs <code class="reqn">\le</code> 12)); very large (hhs <code class="reqn">\ge</code> 13).</p>
</dd>
<dt><code>lrconsaeu</code></dt><dd><p>log of real consumption per adult equivalent units.</p>
</dd>
<dt><code>lfirewood_pr</code></dt><dd><p>Firewood log price.</p>
</dd>
<dt><code>lcharcol_pr</code></dt><dd><p>Charcoal log price.</p>
</dd>
<dt><code>lkerosene_pr</code></dt><dd><p>Kernosene log price.</p>
</dd>
<dt><code>lelectric_pr</code></dt><dd><p>Electricity log price.</p>
</dd>
</dl>



<h3>References</h3>

<p>Alem, Y., Beyene, A. D., Köhlin, G., &amp; Mekonnen, A. (2016). &quot;Modeling household cooking fuel choice: A panel multinomial logit approach&quot;. <em>Energy Economics</em>, <b>59</b>, 129-137.
</p>

<hr>
<h2 id='MouseNeurons'>A subset of mouse brain cells data</h2><span id='topic+MouseNeurons'></span>

<h3>Description</h3>

<p>A subset of mouse brain cells data used in To el al. (2022). This is used to evaluate the ability of Lamp5 gene to discriminate three types of glutamatergic neurons. A full data is publicly available at <a href="https://portal.brain-map.org/atlases-and-data/rnaseq/mouse-v1-and-alm-smart-seq">https://portal.brain-map.org/atlases-and-data/rnaseq/mouse-v1-and-alm-smart-seq</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MouseNeurons)
</code></pre>


<h3>Format</h3>

<p>A data frame with 860 observations from 23 clusters and 7 variables:
</p>

<dl>
<dt><code>sample_name</code></dt><dd><p>name of each observation.</p>
</dd>
<dt><code>subclass_label</code></dt><dd><p>a factor with 3 levels (types) of glutamatergic neurons, i.e., L2/3 IT (Layer 2/3 Intratelencephalic), L4 (Layer 4) and L5 PT (Layer 5 Pyramidal Tract) neurons.</p>
</dd>
<dt><code>genotype_id</code></dt><dd><p>the mouse genotype (which yield 23 clusters).</p>
</dd>
<dt><code>sex</code></dt><dd><p>the gender of mouse.</p>
</dd>
<dt><code>age_days</code></dt><dd><p>the age of mouse, in days.</p>
</dd>
<dt><code>Slc17a7_cpm</code></dt><dd><p>count per million of Slc17a7 (Solute Carrier Family 17 Member 7) gene expression.</p>
</dd>
<dt><code>Lamp5_cpm</code></dt><dd><p>count per million of Lamp5 (Lysosomal Associated Membrane Protein Family Member 5) gene expression.</p>
</dd>
</dl>



<h3>References</h3>

<p>To, D-K., Adimari, G., Chiogna, M. and Risso, D. (2022)
&ldquo;Receiver operating characteristic estimation and threshold selection criteria in three-class classification problems for clustered data&rdquo;. <em>Statistical Methods in Medical Research</em>, <b>7</b>, 31, 1325-1341.
</p>

<hr>
<h2 id='plot.clus_lme'>Plot an clus_lme object.</h2><span id='topic+plot.clus_lme'></span>

<h3>Description</h3>

<p>Diagnostic plots for the linear mixed-effect model, fitted by clus_lme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_lme'
plot(x, file_name = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.clus_lme_+3A_x">x</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, i.e., a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="plot.clus_lme_+3A_file_name">file_name</code></td>
<td>
<p>File name to create on disk.</p>
</td></tr>
<tr><td><code id="plot.clus_lme_+3A_...">...</code></td>
<td>
<p>further arguments used with <code><a href="ggpubr.html#topic+ggexport">ggexport</a></code> function, for example, <code>width</code>, <code>height</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot.clus_lme</code> provides three diagnostic plots: Q-Q plots for residuals, Fitted vs. Residuals values, and Q-Q plot for cluster effects, based on <code>ggplot()</code>.
</p>


<h3>Value</h3>

<p><code>plot.clus_lme</code> returns the diagnostic plots for the linear mixed-effect model, fitted by clus_lme.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_lme">clus_lme</a></code>
</p>

<hr>
<h2 id='plot.clus_opt_thres3'>Plot of confidence regions for covariate-specific optimal pair of thresholds.</h2><span id='topic+plot.clus_opt_thres3'></span>

<h3>Description</h3>

<p>This function plots confidence regions for covariate-specific optimal pair of thresholds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_opt_thres3'
plot(
  x,
  ci_level = 0.95,
  colors = NULL,
  xlims,
  ylims,
  size_point = 0.5,
  size_path = 0.5,
  names_labels,
  nrow_legend = 1,
  file_name = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.clus_opt_thres3_+3A_x">x</code></td>
<td>
<p>an object of class &quot;clus_opt_thres3&quot;, i.e., a result of <code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code>.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_ci_level">ci_level</code></td>
<td>
<p>confidence level to be used for constructing the confidence regions; default is 0.95.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_colors">colors</code></td>
<td>
<p>a string vector for the name(s) specifying color(s) to be used for drawing confidence regions. If specified, the dimension of the vector needs to be equal the number of considered points (each point corresponds to a set of values for the covariates).</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_xlims">xlims</code>, <code id="plot.clus_opt_thres3_+3A_ylims">ylims</code></td>
<td>
<p>numeric vectors of dimension 2, giving the limits for x and y axes in the plot.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_size_point">size_point</code>, <code id="plot.clus_opt_thres3_+3A_size_path">size_path</code></td>
<td>
<p>numeric values, indicating sizes for point(s) and line(s) in the plot.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_names_labels">names_labels</code></td>
<td>
<p>an optional character vector giving the label name for covariates.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_nrow_legend">nrow_legend</code></td>
<td>
<p>an optional number of rows in the legend.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_file_name">file_name</code></td>
<td>
<p>File name to create on disk.</p>
</td></tr>
<tr><td><code id="plot.clus_opt_thres3_+3A_...">...</code></td>
<td>
<p>further arguments used with <code><a href="ggpubr.html#topic+ggexport">ggexport</a></code> function, for example, <code>width</code>, <code>height</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>plot.clus_opt_thres3</code> provides plots of confidence regions (and point estimates) of covariate-specific optimal pair of thresholds. The plots are based on <code>ggplot()</code>.
</p>


<h3>Value</h3>

<p><code>plot.clus_opt_thres3</code> returns plots of confidence regions of covariate-specific optimal pair of thresholds.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code>
</p>

<hr>
<h2 id='print.ci_clus_vus'>Print summary results from ci_clus_vus</h2><span id='topic+print.ci_clus_vus'></span>

<h3>Description</h3>

<p><code>print.ci_vus</code> displays the results of the output from <code><a href="#topic+ci_clus_vus">ci_clus_vus</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ci_clus_vus'
print(x, digits = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.ci_clus_vus_+3A_x">x</code></td>
<td>
<p>an object of class &quot;ci_clus_vus&quot;, a result of <code><a href="#topic+ci_clus_vus">ci_clus_vus</a></code> call.</p>
</td></tr>
<tr><td><code id="print.ci_clus_vus_+3A_digits">digits</code></td>
<td>
<p>minimal number of significant digits, see <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
<tr><td><code id="print.ci_clus_vus_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print">print</a></code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.ci_clus_vus</code> shows a summary table for confidence interval limits for covariate-specific VUS.
</p>


<h3>Value</h3>

<p><code>print.ci_clus_vus</code> shows a summary table for confidence intervals for covariate-specific VUS.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_vus">clus_vus</a></code>
</p>

<hr>
<h2 id='print.clus_lme'>Print summary results of an clus_lme object</h2><span id='topic+print.clus_lme'></span>

<h3>Description</h3>

<p><code>print.clus_lme</code> displays results of the output from <code><a href="#topic+clus_lme">clus_lme</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_lme'
print(x, digits = max(3L, getOption("digits") - 3L), call = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.clus_lme_+3A_x">x</code></td>
<td>
<p>an object of class &quot;clus_lme&quot;, a result of <code><a href="#topic+clus_lme">clus_lme</a></code> call.</p>
</td></tr>
<tr><td><code id="print.clus_lme_+3A_digits">digits</code></td>
<td>
<p>minimal number of significant digits, see <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
<tr><td><code id="print.clus_lme_+3A_call">call</code></td>
<td>
<p>logical. If set to <code>TRUE</code>, the matched call will be printed.</p>
</td></tr>
<tr><td><code id="print.clus_lme_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print">print</a></code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.clus_lme</code> shows a summary table for the estimated parameters in the cluster-effect model (continuous diagnostic test in three-class setting).
</p>


<h3>Value</h3>

<p><code>print.clus_lme</code> returns a summary table for the estimated parameters in the cluster-effect model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_lme">clus_lme</a></code>
</p>

<hr>
<h2 id='print.clus_opt_thres3'>Print summary results from <code>clus_opt_thres3</code></h2><span id='topic+print.clus_opt_thres3'></span>

<h3>Description</h3>

<p><code>print.clus_opt_thres3</code> displays the results of the output from <code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_opt_thres3'
print(x, digits = 3, call = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.clus_opt_thres3_+3A_x">x</code></td>
<td>
<p>an object of class &quot;clus_opt_thres3&quot;, a result of <code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code> call.</p>
</td></tr>
<tr><td><code id="print.clus_opt_thres3_+3A_digits">digits</code></td>
<td>
<p>minimal number of significant digits, see <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
<tr><td><code id="print.clus_opt_thres3_+3A_call">call</code></td>
<td>
<p>logical. If set to <code>TRUE</code>, the matched call will be printed.</p>
</td></tr>
<tr><td><code id="print.clus_opt_thres3_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print">print</a></code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.clus_opt_thres3</code> shows a summary table for covariate-specific optimal pair of thresholds estimates.
</p>


<h3>Value</h3>

<p><code>print.clus_opt_thres3</code> returns a summary table for results of covariate-specific optimal pair of thresholds estimation.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_opt_thres3">clus_opt_thres3</a></code>
</p>

<hr>
<h2 id='print.clus_tcfs'>Print summary results from clus_tcfs</h2><span id='topic+print.clus_tcfs'></span>

<h3>Description</h3>

<p><code>print.clus_tcfs</code> displays the results of the output from <code><a href="#topic+clus_tcfs">clus_tcfs</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_tcfs'
print(x, digits = 3, call = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.clus_tcfs_+3A_x">x</code></td>
<td>
<p>an object of class &quot;clus_tcfs&quot;, a result of <code><a href="#topic+clus_tcfs">clus_tcfs</a></code> call.</p>
</td></tr>
<tr><td><code id="print.clus_tcfs_+3A_digits">digits</code></td>
<td>
<p>minimal number of significant digits, see <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
<tr><td><code id="print.clus_tcfs_+3A_call">call</code></td>
<td>
<p>logical. If set to <code>TRUE</code>, the matched call will be printed.</p>
</td></tr>
<tr><td><code id="print.clus_tcfs_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print">print</a></code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.clus_tcfs</code> shows a summary table for covariate-specific TCFs estimates.
</p>


<h3>Value</h3>

<p><code>print.clus_tcfs</code> returns a summary table for covariate-specific TCFs estimates.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_tcfs">clus_tcfs</a></code>
</p>

<hr>
<h2 id='print.clus_vus'>Print summary results from clus_vus</h2><span id='topic+print.clus_vus'></span>

<h3>Description</h3>

<p><code>print.clus_vus</code> displays the results of the output from <code><a href="#topic+clus_vus">clus_vus</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'clus_vus'
print(x, digits = 3, call = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.clus_vus_+3A_x">x</code></td>
<td>
<p>an object of class &quot;VUS&quot;, a result of <code><a href="#topic+clus_vus">clus_vus</a></code> call.</p>
</td></tr>
<tr><td><code id="print.clus_vus_+3A_digits">digits</code></td>
<td>
<p>minimal number of significant digits, see <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
<tr><td><code id="print.clus_vus_+3A_call">call</code></td>
<td>
<p>logical. If set to <code>TRUE</code>, the matched call will be printed.</p>
</td></tr>
<tr><td><code id="print.clus_vus_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+print">print</a></code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>print.clus_vus</code> shows a summary table for covariate-specific VUS estimates, containing estimates, standard errors, z-values and p-values for the hypothesis testing <code class="reqn">H_0: VUS = 1/6</code> versus an alternative <code class="reqn">H_A: VUS &gt; 1/6</code>.
</p>


<h3>Value</h3>

<p><code>print.clus_vus</code> returns a summary table for covariate-specific VUS estimates.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clus_vus">clus_vus</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
