<!DOCTYPE html><html lang="en"><head><title>Help for package clustermole</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {clustermole}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#clustermole_enrichment'><p>Cell types based on the expression of all genes</p></a></li>
<li><a href='#clustermole_markers'><p>Available cell type markers</p></a></li>
<li><a href='#clustermole_overlaps'><p>Cell types based on overlap of marker genes</p></a></li>
<li><a href='#read_gmt'><p>Read a GMT file into a data frame</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Unbiased Single-Cell Transcriptomic Data Cell Type
Identification</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Assignment of cell type labels to single-cell RNA sequencing (scRNA-seq) clusters is often a time-consuming process that involves manual inspection of the cluster marker genes complemented with a detailed literature search. This is especially challenging when unexpected or poorly described populations are present. The clustermole R package provides methods to query thousands of human and mouse cell identity markers sourced from a variety of databases.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://igordot.github.io/clustermole/">https://igordot.github.io/clustermole/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/igordot/clustermole/issues">https://github.com/igordot/clustermole/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, GSEABase, GSVA (&ge; 1.50.0), magrittr, methods, rlang,
singscore, tibble, tidyr, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, roxygen2, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-08 15:22:18 UTC; id460</td>
</tr>
<tr>
<td>Author:</td>
<td>Igor Dolgalev <a href="https://orcid.org/0000-0003-4451-126X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Igor Dolgalev &lt;igor.dolgalev@nyumc.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-08 22:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='clustermole_enrichment'>Cell types based on the expression of all genes</h2><span id='topic+clustermole_enrichment'></span>

<h3>Description</h3>

<p>Perform enrichment of cell type signatures based on the full gene expression matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustermole_enrichment(expr_mat, species, method = "gsva")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clustermole_enrichment_+3A_expr_mat">expr_mat</code></td>
<td>
<p>Expression matrix (logCPMs, logFPKMs, or logTPMs) with genes as rows and clusters/populations/samples as columns.</p>
</td></tr>
<tr><td><code id="clustermole_enrichment_+3A_species">species</code></td>
<td>
<p>Species: <code>hs</code> for human or <code>mm</code> for mouse.</p>
</td></tr>
<tr><td><code id="clustermole_enrichment_+3A_method">method</code></td>
<td>
<p>Enrichment method: <code>ssgsea</code>, <code>gsva</code>, <code>singscore</code>, or <code>all</code>. The method to use for the estimation of gene set enrichment scores. The options are ssGSEA (Barbie et al, 2009), GSVA (Hänzelmann et al, 2013), singscore (Foroutan et al, 2018), or a combination of all three methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of enrichment results.
</p>


<h3>References</h3>

<p>Barbie, D., Tamayo, P., Boehm, J. et al. Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. <em>Nature</em> 462, 108–112 (2009). <a href="https://doi.org/10.1038/nature08460">doi:10.1038/nature08460</a>
</p>
<p>Hänzelmann, S., Castelo, R. &amp; Guinney, J. GSVA: Gene set variation analysis for microarray and RNA-Seq data. <em>BMC Bioinformatics</em> 14, 7 (2013). <a href="https://doi.org/10.1186/1471-2105-14-7">doi:10.1186/1471-2105-14-7</a>
</p>
<p>Foroutan, M., Bhuva, D.D., Lyu, R. et al. Single sample scoring of molecular phenotypes. <em>BMC Bioinformatics</em> 19, 404 (2018). <a href="https://doi.org/10.1186/s12859-018-2435-4">doi:10.1186/s12859-018-2435-4</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># my_enrichment &lt;- clustermole_enrichment(expr_mat = my_expr_mat, species = "hs")
</code></pre>

<hr>
<h2 id='clustermole_markers'>Available cell type markers</h2><span id='topic+clustermole_markers'></span>

<h3>Description</h3>

<p>Retrieve the full list of cell type markers in the <code>clustermole</code> database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustermole_markers(species = c("hs", "mm"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clustermole_markers_+3A_species">species</code></td>
<td>
<p>Species: <code>hs</code> for human or <code>mm</code> for mouse.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of cell type markers (one gene per row).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>markers &lt;- clustermole_markers()
head(markers)
</code></pre>

<hr>
<h2 id='clustermole_overlaps'>Cell types based on overlap of marker genes</h2><span id='topic+clustermole_overlaps'></span>

<h3>Description</h3>

<p>Perform overrepresentation analysis for a set of genes compared to all cell type signatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustermole_overlaps(genes, species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clustermole_overlaps_+3A_genes">genes</code></td>
<td>
<p>A vector of genes.</p>
</td></tr>
<tr><td><code id="clustermole_overlaps_+3A_species">species</code></td>
<td>
<p>Species: <code>hs</code> for human or <code>mm</code> for mouse.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of enrichment results with hypergeometric test p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>my_genes &lt;- c("CD2", "CD3D", "CD3E", "CD3G", "TRAC", "TRBC2", "LTB")
my_overlaps &lt;- clustermole_overlaps(genes = my_genes, species = "hs")
head(my_overlaps)
</code></pre>

<hr>
<h2 id='read_gmt'>Read a GMT file into a data frame</h2><span id='topic+read_gmt'></span>

<h3>Description</h3>

<p>Read a GMT file into a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gmt(file, geneset_label = "celltype", gene_label = "gene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_gmt_+3A_file">file</code></td>
<td>
<p>A connection object or a character string (can be a URL).</p>
</td></tr>
<tr><td><code id="read_gmt_+3A_geneset_label">geneset_label</code></td>
<td>
<p>Column name for gene sets (first column of the GMT file) in the output data frame.</p>
</td></tr>
<tr><td><code id="read_gmt_+3A_gene_label">gene_label</code></td>
<td>
<p>Column name for genes (variable columns of the GMT file) in the output data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with gene sets as the first column and genes as the second column (one gene per row).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gmt &lt;- "http://software.broadinstitute.org/gsea/msigdb/supplemental/scsig.all.v1.0.symbols.gmt"
gmt_tbl &lt;- read_gmt(gmt)
head(gmt_tbl)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
