<!DOCTYPE html><html><head><title>Help for package ClussCluster</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ClussCluster}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ClussCluster'><p>Performs simultaneous detection of cell types and cell-type-specific signature genes</p></a></li>
<li><a href='#filter_gene'><p>Gene Filter</p></a></li>
<li><a href='#Hou_sim'><p>A truncated subset of the scRNA-seq expression data set from Hou et.al (2016)</p></a></li>
<li><a href='#plot_ClussCluster'><p>Plots the results of <code>ClussCluster</code></p></a></li>
<li><a href='#plot_ClussCluster_Gap'><p>Plots the results of <code>ClussCluster_Gap</code></p></a></li>
<li><a href='#print_ClussCluster'><p>Prints out the results of <code>ClussCluster</code></p></a></li>
<li><a href='#print_ClussCluster_Gap'><p>Prints out the results of <code>ClussCluster_Gap</code></p>
Prints the gap statistics and number of genes selected for each candidate tuning parameter.</a></li>
<li><a href='#sim_dat'><p>A simulated expression data set.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simultaneous Detection of Clusters and Cluster-Specific Genes in
High-Throughput Transcriptome Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a new method 'ClussCluster' descried in Ge Jiang and Jun Li, "Simultaneous Detection of Clusters and Cluster-Specific Genes in High-throughput Transcriptome Data" (Unpublished).
	Simultaneously perform clustering analysis and signature 	gene selection on high-dimensional transcriptome data sets. 	To do so, 'ClussCluster' incorporates a Lasso-type 	regularization penalty term to the objective function of K-	means so that cell-type-specific signature genes can be 	identified while clustering the cells.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown (&ge; 1.13)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats (&ge; 3.5.0), utils (&ge; 3.5.0), VennDiagram, scales (&ge;
1.0.0), reshape2 (&ge; 1.4.3), ggplot2 (&ge; 3.1.0), rlang (&ge;
0.3.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-07-02 14:40:39 UTC; Chuanqi</td>
</tr>
<tr>
<td>Author:</td>
<td>Li Jun [cre],
  Jiang Ge [aut],
  Wang Chuanqi [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Li Jun &lt;jun.li@nd.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-07-02 16:30:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='ClussCluster'>Performs simultaneous detection of cell types and cell-type-specific signature genes</h2><span id='topic+ClussCluster'></span><span id='topic+ClussCluster_Gap'></span>

<h3>Description</h3>

<p><code>ClussCluster</code> takes the single-cell transcriptome data and returns an object containing cell types and type-specific signature gene sets
</p>
<p>Selects the tuning parameter in a permutation approach. The tuning parameter controls the L1 bound on w, the feature weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClussCluster(x, nclust = NULL, centers = NULL, ws = NULL,
  nepoch.max = 10, theta = NULL, seed = 1, nstart = 20,
  iter.max = 50, verbose = FALSE)

ClussCluster_Gap(x, nclust = NULL, B = 20, centers = NULL,
  ws = NULL, nepoch.max = 10, theta = NULL, seed = 1,
  nstart = 20, iter.max = 50, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClussCluster_+3A_x">x</code></td>
<td>
<p>An nxp data matrix. There are n cells and p genes.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_nclust">nclust</code></td>
<td>
<p>Number of clusters desired if the cluster centers are not provided. If both are provided, nclust must equal the number of cluster <code>centers</code>.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_centers">centers</code></td>
<td>
<p>A set of initial (distinct) cluster centres if the number of clusters (<code>nclust</code>) is null. If both are provided, the number of cluster centres must equal <code>nclust</code>.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_ws">ws</code></td>
<td>
<p>One or multiple candidate tuning parameters to be evaluated and compared. Determines the sparsity of the selected genes. Should be greater than 1.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_nepoch.max">nepoch.max</code></td>
<td>
<p>The maximum number of epochs. In one epoch, each cell will be evaluated to determine if its label needs to be updated.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_theta">theta</code></td>
<td>
<p>Optional argument. If provided, <code>theta</code> are used as the initial cluster labels of the ClussCluster algorithm; if not, K-means is performed to produce starting cluster labels.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_seed">seed</code></td>
<td>
<p>This seed is used wherever K-means is used.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_nstart">nstart</code></td>
<td>
<p>Argument passed to <code>kmeans</code>. It is the number of random sets used in <code>kmeans</code>.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_iter.max">iter.max</code></td>
<td>
<p>Argument passed to <code>kmeans</code>. The maximum number of iterations allowed.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_verbose">verbose</code></td>
<td>
<p>Print the updates inside every epoch? If TRUE, the updates of cluster label and the value of objective function will be printed out.</p>
</td></tr>
<tr><td><code id="ClussCluster_+3A_b">B</code></td>
<td>
<p>Number of permutation samples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes the normalized and log transformed number of reads mapped to genes (e.g., log(RPKM+1) or log(TPM+1) where RPKM stands for Reads Per Kilobase of transcript per Million mapped reads and TPM stands for transcripts per million) but NOT centered.
</p>


<h3>Value</h3>

<p>a list containing the optimal tuning parameter, <code>s</code>, group labels of clustering, <code>theta</code>, and type-specific weights of genes, <code>w</code>.
</p>
<p>a list containig a vector of candidate tuning parameters, <code>ws</code>, the corresponding values of objective function, <code>O</code>, a matrix of values of objective function for each permuted data and tuning parameter, <code>O_b</code>, gap statistics and their one standard deviations, <code>Gap</code> and <code>sd.Gap</code>, the result given by <code>ClussCluster</code>, <code>run</code>, the tuning parameters with the largest Gap statistic and within one standard deviation of the largest Gap statistic, <code>bestw</code> and <code>onesd.bestw</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hou_sim)
hou.dat &lt;-Hou_sim$x
run.ft &lt;- filter_gene(hou.dat)
hou.test &lt;- ClussCluster(run.ft$dat.ft, nclust=3, ws=4, verbose = FALSE)
</code></pre>

<hr>
<h2 id='filter_gene'>Gene Filter</h2><span id='topic+filter_gene'></span>

<h3>Description</h3>

<p>Filters out genes that are not suitable for differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_gene(dfname, minmean = 2, n0prop = 0.2, minsd = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_gene_+3A_dfname">dfname</code></td>
<td>
<p>name of the expression data frame</p>
</td></tr>
<tr><td><code id="filter_gene_+3A_minmean">minmean</code></td>
<td>
<p>minimum mean expression for each gene</p>
</td></tr>
<tr><td><code id="filter_gene_+3A_n0prop">n0prop</code></td>
<td>
<p>minimum proportion of zero expression (count) for each gene</p>
</td></tr>
<tr><td><code id="filter_gene_+3A_minsd">minsd</code></td>
<td>
<p>minimum standard deviation of expression for each gene</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes an expression data frame that has been properly normalized but NOT centered. It returns a list with the slot <code>dat.ft</code> being the data set that satisfies the pre-set thresholds on minumum mean, standard deviation (sd), and proportion of zeros (n0prop) for each gene.
</p>
<p>If the data has already been centered, one can still apply the filters of <code>mean</code> and <code>sd</code> but not <code>n0prop</code>.
</p>


<h3>Value</h3>

<p>a list containing the data set with genes satisfying the thresholds, <code>dat.ft</code>, the name of <code>dat.ft</code>, and the indices of those kept genes, <code>index</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- matrix(rnbinom(300*60, mu = 2, size = 1), 300, 60)
dat_filtered &lt;- filter_gene(dat, minmean=2, n0prop=0.2, minsd=1)
</code></pre>

<hr>
<h2 id='Hou_sim'>A truncated subset of the scRNA-seq expression data set from Hou et.al (2016)</h2><span id='topic+Hou_sim'></span>

<h3>Description</h3>

<p>This data contains expression levels (normalized and log-transformed) for 33 cells and 100 genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Hou_sim)
</code></pre>


<h3>Format</h3>

<p>An object containing the following variables:
</p>

<dl>
<dt><code>x</code></dt><dd><p>An expression data frame of 33 HCC cells on 100 genes.</p>
</dd>
<dt><code>y</code></dt><dd><p>Numerical group indicator of all cells.</p>
</dd>
<dt><code>gnames</code></dt><dd><p>Gene names of all genes.</p>
</dd>
<dt><code>snames</code></dt><dd><p>Cell names of all cells.</p>
</dd>
<dt><code>groups</code></dt><dd><p>Cell group names.</p>
</dd>
<dt><code>note</code></dt><dd><p>A simple note of the data set.</p>
</dd>
</dl>


<h3>Details</h3>

<p>This data contains raw expression levels (log-transformed but not centered) for 33 HCC cells and 100 genes. The 33 cells belongs to three different subpopulations and exhibited different biological characteristics. For descriptions of how we generated this data, please refer to the paper.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE65364">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE65364</a>
</p>


<h3>References</h3>

<p>Hou, Yu, et al. &quot;Single-cell triple omics sequencing reveals genetic, epigenetic, and transcriptomic heterogeneity in hepatocellular carcinomas.&quot; Cell research 26.3 (2016): 304-319.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hou_sim)
data &lt;- Hou_sim$x
</code></pre>

<hr>
<h2 id='plot_ClussCluster'>Plots the results of <code>ClussCluster</code></h2><span id='topic+plot_ClussCluster'></span><span id='topic+top.m.hm'></span>

<h3>Description</h3>

<p>Plots the number of signature genes against the tuning parameters if multiple tuning parameters are evaluated in the object. If only one is included, then <code>plot_ClussCluster</code> returns a venn diagram and a heatmap at this particular tuning parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ClussCluster(object, m = 10, snames = NULL, gnames = NULL, ...)

top.m.hm(object, m, snames = NULL, gnames = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_ClussCluster_+3A_object">object</code></td>
<td>
<p>An object that is obtained by applying the ClussCluster function to the data set.</p>
</td></tr>
<tr><td><code id="plot_ClussCluster_+3A_m">m</code></td>
<td>
<p>The number of top signature genes selected to produce the heatmap.</p>
</td></tr>
<tr><td><code id="plot_ClussCluster_+3A_snames">snames</code></td>
<td>
<p>The names of the cells.</p>
</td></tr>
<tr><td><code id="plot_ClussCluster_+3A_gnames">gnames</code></td>
<td>
<p>The names of the genes</p>
</td></tr>
<tr><td><code id="plot_ClussCluster_+3A_...">...</code></td>
<td>
<p>Addtional parameters, sent to the method</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes the normalized and log transformed number of reads mapped to genes (e.g., log(RPKM+1) or log(TPM+1) where RPKM stands for Reads Per Kilobase of transcript per Million mapped reads and TPM stands for transcripts per million) but NOT centered.
</p>
<p>If multiple tuning parameters are evaluated in the object, the number of signature genes is computed for each cluster and is plotted against the tuning parameters. Each color and line type corresponds to a cell type.
</p>
<p>If only one tuning parameter is evaluated, two plots will be produced. One is the venn diagram of the cell-type-specific genes, the other is the heatmap of the data with the cells and top m signature genes. See more details in the paper.
</p>


<h3>Value</h3>

<p>a ggplot2 object of the heatmap with top signature genes selected by <code>ClussCluster</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hou_sim)
run.cc &lt;- ClussCluster(Hou_sim$x, nclust = 3, ws = c(2.4, 5, 8.8))
plot_ClussCluster(run.cc, m = 5, snames=Hou$snames, gnames=Hou$gnames)

</code></pre>

<hr>
<h2 id='plot_ClussCluster_Gap'>Plots the results of <code>ClussCluster_Gap</code></h2><span id='topic+plot_ClussCluster_Gap'></span>

<h3>Description</h3>

<p>Plots the gap statistics and number of genes selected as the tuning parameter varies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ClussCluster_Gap(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_ClussCluster_Gap_+3A_object">object</code></td>
<td>
<p>object obtained from <code>ClussCluster_Gap</code>()</p>
</td></tr>
</table>

<hr>
<h2 id='print_ClussCluster'>Prints out the results of <code>ClussCluster</code></h2><span id='topic+print_ClussCluster'></span>

<h3>Description</h3>

<p>Prints out the results of <code>ClussCluster</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_ClussCluster(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_ClussCluster_+3A_object">object</code></td>
<td>
<p>An object that is obtained by applying the ClussCluster function to the data set.</p>
</td></tr>
</table>

<hr>
<h2 id='print_ClussCluster_Gap'>Prints out the results of <code>ClussCluster_Gap</code>
Prints the gap statistics and number of genes selected for each candidate tuning parameter.</h2><span id='topic+print_ClussCluster_Gap'></span>

<h3>Description</h3>

<p>Prints out the results of <code>ClussCluster_Gap</code>
Prints the gap statistics and number of genes selected for each candidate tuning parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_ClussCluster_Gap(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_ClussCluster_Gap_+3A_object">object</code></td>
<td>
<p>An object that is obtained by applying the ClussCluster_Gap function to the data set.</p>
</td></tr>
</table>

<hr>
<h2 id='sim_dat'>A simulated expression data set.</h2><span id='topic+sim_dat'></span>

<h3>Description</h3>

<p>An example data set containing expressing levels for 60 cells and 200 genes. The 60 cells belong to 4 cell types with 15 cells each. Each cell type is uniquely associated with 30 signature genes, i.e., the first cell type is associated with the first 30 genes, the second cell type is associated with the next 30 genes, so on and so forth. The remaining 80 genes show indistinct expression patterns among the four cell types and are considered as noise genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sim_dat)
</code></pre>


<h3>Format</h3>

<p>A data frame with 60 cells on 200 genes.</p>


<h3>Value</h3>

<p>A simulated dataset used to demonstrate the application of <code>ClussCluster</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim_dat)
head(sim_dat)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
