<!DOCTYPE html><html><head><title>Help for package CMplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CMplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cattle50K'><p>Genotyped by Bovine50K chip</p></a></li>
<li><a href='#CMplot-package'>
<p>Circle Manhattan Plot</p></a></li>
<li><a href='#pig60K'><p>Genotyped by pig 60k chip</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Circle Manhattan Plot</td>
</tr>
<tr>
<td>Version:</td>
<td>4.5.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-15</td>
</tr>
<tr>
<td>Author:</td>
<td>LiLin-Yin</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>LiLin-Yin &lt;ylilin@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Manhattan plot, a type of scatter plot, was widely used to display the association results. However, it is usually time-consuming and laborious for a non-specialist user to write scripts and adjust parameters of an elaborate plot. Moreover, the ever-growing traits measured have necessitated the integration of results from different Genome-wide association study researches. Circle Manhattan Plot is the first open R package that can lay out. Genome-wide association study P-value results in both traditional rectangular patterns, QQ-plot and novel circular ones. United in only one bull's eye style plot, association results from multiple traits can be compared interactively, thereby to reveal both similarities and differences between signals. Additional functions include: highlight signals, a group of SNPs, chromosome visualization and candidate genes around SNPs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YinLiLin/CMplot">https://github.com/YinLiLin/CMplot</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YinLiLin/CMplot/issues/new">https://github.com/YinLiLin/CMplot/issues/new</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-19 07:29:01 UTC; yinll</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-19 08:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='cattle50K'>Genotyped by Bovine50K chip</h2><span id='topic+cattle50K'></span>

<h3>Description</h3>

<p>This dataset gives the effects of all SNPs using rrBLUP algorithm for 3 traits, individuals were genotyped by Bovine50K chip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cattle50K)</code></pre>


<h3>Format</h3>

<p>A dataframe containing 3 traits' SNPs effects</p>

<hr>
<h2 id='CMplot-package'>
Circle Manhattan Plot 
</h2><span id='topic+CMplot-package'></span><span id='topic+CMplot'></span>

<h3>Description</h3>

<p>Manhattan plot, a type of scatter plot, was widely used to display the association results. However, it is usually time-consuming and laborious for a non-specialist user to write scripts and adjust parameters of an elaborate plot. Moreover, the ever-growing traits measured have necessitated the 
integration of results from different Genome-wide association study researches. Circle Manhattan Plot is the first open R package that can lay out Genome-wide association study P-value results in QQ-plot, both traditional rectangular patterns and novel circular ones. United in only one bull's eye style 
plot, association results from multiple traits can be compared interactively, thereby to reveal both similarities and differences between signals.
Additional functions include: highlight signals, a group of SNPs, chromosome visualization and candidate genes around SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CMplot(Pmap,col=c("#4197d8","#f8c120","#413496","#495226",
    "#d60b6f","#e66519","#d581b7","#83d3ad","#7c162c","#26755d"),
    bin.size=1e6,bin.breaks=NULL,LOG10=TRUE,pch=19,type="p",band=1,
    H=1.5,ylim=NULL,axis.cex=1,axis.lwd=1.5,lab.cex=1.5,lab.font=2,
    plot.type=c("m","c","q","d"),multracks=FALSE,
    multracks.xaxis=FALSE,multraits=FALSE,points.alpha=100L,r=0.3,
    cex=c(0.5,1,1),outward=FALSE,ylab=expression(-log[10](italic(p))), 
    ylab.pos=3,xticks.pos=1,mar=c(3,6,3,3),mar.between=0,threshold=NULL,
    threshold.col="red",threshold.lwd=1,threshold.lty=2,amplify=TRUE,
    signal.cex=1.5,signal.pch=19,signal.col=NULL,signal.line=2,
    highlight=NULL,highlight.cex=1,highlight.pch=19,highlight.type="p",
    highlight.col="red",highlight.text=NULL,highlight.text.col="black",
    highlight.text.cex=1,highlight.text.font=3,chr.labels=NULL,
    chr.border=FALSE,chr.labels.angle=0,chr.den.col="black",
    chr.pos.max=FALSE,cir.band=1,cir.chr=TRUE,cir.chr.h=1.5,
    cir.axis=TRUE,cir.axis.col="black",cir.axis.grid=TRUE,conf.int=TRUE,
    conf.int.col=NULL,file.output=TRUE,file.name="",file=c("jpg","pdf",
    "tiff","png"),dpi=300,height=NULL,width=NULL,main="",main.cex=1.5,
    main.font=2,legend.ncol=NULL,legend.cex=1,
    legend.pos=c("left","middle","right"),box=FALSE,verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CMplot-package_+3A_pmap">Pmap</code></td>
<td>

<p>a dataframe, the first column is the name of genomic markers, the second column is the chromosome, the third column is the physical position, and the remaining columns are the P-values of each trait.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_col">col</code></td>
<td>

<p>a vector or a matrix, the color for chromosomes. If it is a vector, all traits use the same colors, it means that the same chromosome is drew in the same color across traits, and the length of colors are not fixed (1, 2, 3 or more colors can be used), if the length of the &quot;col&quot; is shorter than the number of chromosomes, then colors will be applied circularly. If &quot;col&quot; is a matrix, the number of row of which must be equal to the number of traits, the columns are the colors that users want to use for different traits, thus the chromosomes of each trait can be plotted in different number of colors, the missing value can be replaced by NA. For example: col=matrix(c(&quot;grey30&quot;,&quot;grey60&quot;,NA,&quot;red&quot;,&quot;blue&quot;,&quot;green&quot;,&quot;orange&quot;,NA,NA),3,3,byrow=T).
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_bin.size">bin.size</code></td>
<td>

<p>a integer, the size of bin in bp for marker density plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_bin.breaks">bin.breaks</code></td>
<td>

<p>a vector, set the breaks for the legend of density plot, e.g., seq(min, max, step), the windows in which the number of markers is out of the this range will be plotted in the same colors with the min or max value.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_log10">LOG10</code></td>
<td>

<p>logical, whether to change the p-value into log10(p-value) scale.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_pch">pch</code></td>
<td>

<p>a integer, the shape for the points, is the same with &quot;pch&quot; in &lt;plot&gt;.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_type">type</code></td>
<td>

<p>a character, could be &quot;p&quot; (point), &quot;l&quot; (cross line), &quot;h&quot; (vertical lines) and so on, is the same with &quot;type&quot; in &lt;plot&gt;.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_band">band</code></td>
<td>

<p>a number, the size of space between chromosomes, the default is 1 (if the band equals to 0, then there would be no space between chromosomes).
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_h">H</code></td>
<td>

<p>a number, the height for each circle, each circle represents a trait, the default is 1.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_ylim">ylim</code></td>
<td>

<p>vector (c(min, max)) or list, CMplot will only plot the points among this interval, it can be vector or list, if it is a list, different traits can be assigned with different range at y-axis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_axis.cex">axis.cex</code></td>
<td>

<p>a number, controls the size of ticks labels of X/Y-axis and the ticks labels of axis for circle plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_axis.lwd">axis.lwd</code></td>
<td>

<p>a number, controls the thickness of X/Y-axis lines and the thickness of axis for circle plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_lab.cex">lab.cex</code></td>
<td>

<p>a number, controls the size of labels of X/Y-axis and the labels of chromosomes for circle plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_lab.font">lab.font</code></td>
<td>

<p>a number, controls the font of labels of all axis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_plot.type">plot.type</code></td>
<td>

<p>a character or vector, only &quot;d&quot;, &quot;c&quot;, &quot;m&quot;, &quot;q&quot; can be used. if plot.type=&quot;d&quot;, SNP density will be plotted; if plot.type=&quot;c&quot;, only circle-Manhattan plot will be plotted; if plot.type=&quot;m&quot;,only Manhattan plot will be plotted; if plot.type=&quot;q&quot;,only Q-Q plot will be plotted; if plot.type=c(&quot;m&quot;,&quot;q&quot;), Both Manhattan and Q-Q plots will be plotted.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_multracks">multracks</code></td>
<td>

<p>a logical, if TRUE, all Manhattan plots will be drew in one file.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_multracks.xaxis">multracks.xaxis</code></td>
<td>

<p>a logical, if TRUE, all the traits will be added x-axis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_multraits">multraits</code></td>
<td>

<p>a logical, if TRUE, all traits will be plotted in one axis, and output one file.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_points.alpha">points.alpha</code></td>
<td>

<p>a number, the transparency of point for multiple traits plot, ranges from '0' (fully transparent) to '255' (opaque).
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_r">r</code></td>
<td>

<p>a number, the radius for the circle (the inside radius), the default is 1.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cex">cex</code></td>
<td>

<p>a number or a vector, the size for the points, is the same with &quot;size&quot; in &lt;plot&gt;, and if it is a vector, the first number controls the size of points in circle plot(the default is 0.5), the second number controls the size of points in Manhattan plot (the default is 1), the third number controls the size of points in Q-Q plot (the default is 1)
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_outward">outward</code></td>
<td>

<p>logical, if TRUE, all points will be plotted from inside to outside for circular Manhattan plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_ylab">ylab</code></td>
<td>

<p>a character, the labels for y axis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_ylab.pos">ylab.pos</code></td>
<td>

<p>the distance between ylab and yaxis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_xticks.pos">xticks.pos</code></td>
<td>

<p>the distance between labels of x ticks and x axis.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_mar">mar</code></td>
<td>

<p>the size of white gaps around the plot, 4 values should be provided, indicating the direction of bottom, left, up, and right.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_mar.between">mar.between</code></td>
<td>

<p>the distance between multiple plots for multiple tracks Manhattan plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_threshold">threshold</code></td>
<td>

<p>a number or vector or list, the significant threshold. For example, the Bonfferoni adjustment method: threshold=0.01/nrow(Pmap), more than one significant line can be added on the plots; if it is a list, each trait can be assigned with different thresholds, if threshold=0 or NULL, then the threshold line will not be added.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_threshold.col">threshold.col</code></td>
<td>

<p>a character or vector, the color for the line of threshold levels, it can also control the color of the diagonal line of QQplot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_threshold.lwd">threshold.lwd</code></td>
<td>

<p>a number or vector, the width for the line of threshold levels, it can also control the thickness of the diagonal line of QQplot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_threshold.lty">threshold.lty</code></td>
<td>

<p>a number or vector, the type for the line of threshold levels, it can also control the type of the diagonal line of QQplot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_amplify">amplify</code></td>
<td>

<p>logical, CMplot can amplify the significant points, if TRUE, then the points bigger than the minimal significant level will be amplified, the default: amplify=TRUE.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_signal.cex">signal.cex</code></td>
<td>

<p>a number, if amplify=TRUE, users can set the size of significant points.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_signal.pch">signal.pch</code></td>
<td>

<p>a number, if amplify=TRUE, users can set the shape of significant points.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_signal.col">signal.col</code></td>
<td>

<p>a character, if amplify=TRUE, users can set the colour of significant points, if signal.col=NULL, then the colors of significant points will not be changed.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_signal.line">signal.line</code></td>
<td>

<p>a number, the thickness of the lines of significant SNPs cross the circle.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight">highlight</code></td>
<td>

<p>a vector or list, names of SNPs which need to be highlighted. vector for single trait and list for multiple traits.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.cex">highlight.cex</code></td>
<td>

<p>a vector or list, the size of points for SNPs which need to be highlighted.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.pch">highlight.pch</code></td>
<td>

<p>a vector or list, the pch of points for SNPs which need to be highlighted.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.type">highlight.type</code></td>
<td>

<p>a vector or list, the type of points for SNPs which need to be highlighted.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.col">highlight.col</code></td>
<td>

<p>a vector or list, the col of points for SNPs which need to be highlighted.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.text">highlight.text</code></td>
<td>

<p>a vector or list, the text which would be added around the highlighted SNPs.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.text.col">highlight.text.col</code></td>
<td>

<p>a vector or list, the color for added text.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.text.cex">highlight.text.cex</code></td>
<td>

<p>a value, the size for added text.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_highlight.text.font">highlight.text.font</code></td>
<td>

<p>text font for the highlighted SNPs.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_chr.labels">chr.labels</code></td>
<td>

<p>a vector, the labels for the chromosomes of density plot and Manhattan plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_chr.border">chr.border</code></td>
<td>

<p>a logical, whether to plot the dot line between chromosomes.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_chr.labels.angle">chr.labels.angle</code></td>
<td>

<p>a value, rotate tick labels of x-axis for Manhattan plot (-90 &lt; chr.labels.angle &lt; 90).
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_chr.pos.max">chr.pos.max</code></td>
<td>

<p>logical, whether the physical positions of each chromosome contain the maximum length of the chromosome.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_chr.den.col">chr.den.col</code></td>
<td>

<p>a character or vector or NULL, the colour for the SNP density. If the length of parameter 'chr.den.col' is bigger than 1, SNP density that counts the number of SNP within given size ('bin.size') will be plotted around the circle. If chr.den.col=NULL, the density bar will not be attached on the bottom of manhattan plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.band">cir.band</code></td>
<td>

<p>a number, the space between circles, the default is 1.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.chr">cir.chr</code></td>
<td>

<p>logical, a boundary that represents chromosomes will be plotted on the periphery of a circle, the default is TRUE.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.chr.h">cir.chr.h</code></td>
<td>

<p>a number, the width for the boundary, if cir.chr=FALSE, then this parameter will be useless.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.axis">cir.axis</code></td>
<td>

<p>a logical, whether to add the axis of each circle.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.axis.col">cir.axis.col</code></td>
<td>

<p>a character, the color of the axis for circle.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_cir.axis.grid">cir.axis.grid</code></td>
<td>

<p>logical, whether to add axis grid line in circles.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_conf.int">conf.int</code></td>
<td>

<p>logical, whether to plot confidence interval on QQ-plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_conf.int.col">conf.int.col</code></td>
<td>

<p>character or vector, the color of confidence interval of QQplot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_file.output">file.output</code></td>
<td>

<p>a logical, users can choose whether to output the plot results.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_file.name">file.name</code></td>
<td>

<p>a character or vector, the names of output files.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_file">file</code></td>
<td>

<p>a character, users can choose the different output formats of plot, so for, &quot;jpg&quot;, &quot;pdf&quot;, &quot;tiff&quot;, and &quot;png&quot; can be selected by users. The &quot;png&quot; format has a transparent background.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_dpi">dpi</code></td>
<td>

<p>a number, the picture resolution for '.jpg', '.npg', and '.tiff' files. The default is 300.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_height">height</code></td>
<td>

<p>the height of output files.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_width">width</code></td>
<td>

<p>the width of output files.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_main">main</code></td>
<td>

<p>character of vector, the title of the plot for manhattan plot and qqplot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_main.cex">main.cex</code></td>
<td>

<p>size of title.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_main.font">main.font</code></td>
<td>

<p>font of title.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_legend.ncol">legend.ncol</code></td>
<td>

<p>number of column for the legend of multiple traits manhattan plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_legend.cex">legend.cex</code></td>
<td>

<p>set the size of the legend text of trait name.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_legend.pos">legend.pos</code></td>
<td>

<p>control the legend position of trait name on the plot, three options &quot;left&quot;, &quot;middle&quot;, &quot;right&quot; are available.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_box">box</code></td>
<td>

<p>logical, this function draws a box around the current plot.
</p>
</td></tr>
<tr><td><code id="CMplot-package_+3A_verbose">verbose</code></td>
<td>

<p>whether to print the log information.
</p>
</td></tr>
</table>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> CMplot</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 4.5.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-01-15</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL(&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p>1: In the data &quot;Pmap&quot;, autosome and heterosome can both exist simultaneously, and heterosome can also be replaced by a number. <br />
2: only when the amplify=TRUE, then the parameter signal.cex, signal.pch, signal.col will be efficacious. <br />
3: In the circle Manhattan plot, traits will be plotted from inside to outside, it means that the first trait occupies the innermost circle. <br />
</p>


<h3>Author(s)</h3>

<p>LiLin-Yin <br />
Maintainer: LiLin-Yin &lt;<b>ylilin@163.com</b>&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#plot rectangular Manhattan only one trait with 10000 markers:
data(pig60K)                        
CMplot(pig60K[sample(1:nrow(pig60K), 10000),c(1:4)],plot.type="m",
	  threshold=c(0.01,0.05)/nrow(pig60K),threshold.col=c('red','orange'),
	  multracks=FALSE, multraits=FALSE, chr.den.col=NULL, file.output=FALSE)

################################################################################
#plot circular Manhattan for all traits with all markers:                      #
################################################################################
#data(pig60K)                                                                  #         
#CMplot(pig60K,col=c("gray30","gray60"),r=0.5,plot.type="c",                   # 
#   threshold=0.01/nrow(pig60K),signal.col=NULL,outward=TRUE)                  #
################################################################################

</code></pre>

<hr>
<h2 id='pig60K'>Genotyped by pig 60k chip</h2><span id='topic+pig60K'></span>

<h3>Description</h3>

<p>This dataset gives the results of Genome-wide association study of 3 traits, individuals were genotyped by pig 60K chip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pig60K)</code></pre>


<h3>Format</h3>

<p>A dataframe containing 3 traits' Pvalue</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
