<!DOCTYPE html><html lang="en"><head><title>Help for package CNVreg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CNVreg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CNVCOVY'><p>Simulated data with copy number variants (CNV), Covariates (Cov), and outcomes traits (Y_QT for a continuous outcome</p>
and Y_BT for a binary outcome)for the illustration of CNV association analysis with penalized regression in CNVreg.</a></li>
<li><a href='#cvfit_WTSMTH'><p>Penalized Regression with Lasso and Weighted Fusion Penalties with Cross-Validation</p></a></li>
<li><a href='#fit_WTSMTH'><p>Penalized Regression with Lasso and Weighted Fusion Penalties with Given Parameters</p></a></li>
<li><a href='#prep'><p>Prepare Data for Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>CNV-Profile Regression for Copy Number Variants Association
Analysis with Penalized Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs copy number variants association analysis with Lasso and Weighted Fusion penalized regression. 
  Creates a "CNV profile curve" to represent an individualâ€™s CNV events across a genomic region so to capture variations 
  in CNV length and dosage. When evaluating association, the CNV profile curve is directly used as a predictor in the 
  regression model, avoiding the need to predefine CNV loci. CNV profile regression estimates CNV effects at each genome
  position, making the results comparable across different studies. The penalization encourages sparsity 
  in variable selection with a Lasso penalty and encourages effect smoothness between consecutive CNV events with a weighted
  fusion penalty, where the weight controls the level of smoothing between adjacent CNVs. 
  For more details, see Si (2024) &lt;<a href="https://doi.org/10.1101%2F2024.11.23.624994">doi:10.1101/2024.11.23.624994</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, Matrix, doParallel, dplyr, foreach, glmnet, tidyr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, rlang, tidyverse, markdown, kableExtra,
patchwork, ggplot2, withr</td>
</tr>
<tr>
<td>Author:</td>
<td>Yaqin Si [aut],
  Shannon T. Holloway [ctb, cre],
  Jung-Ying Tzeng [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shannon T. Holloway &lt;shannon.t.holloway@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Collate:</td>
<td>'utils.R' 'probPred.R' 'linearPred.R' 'logLH.R' 'loss.R'
'helpful_tests.R' 'ctnsSolution.R' 'rwlsSolution.R'
'Wtsmth_Fit.R' 'nfoldSplit.R' 'WTsmth_nFold_CV.R' 'breakCNV.R'
'datadescrip.R' 'wideDataRaw.R' 'wideFrequency.R'
'weightMatrix.R' 'prep.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-09 16:07:45 UTC; sth45</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-10 16:50:21 UTC</td>
</tr>
</table>
<hr>
<h2 id='CNVCOVY'>Simulated data with copy number variants (CNV), Covariates (Cov), and outcomes traits (Y_QT for a continuous outcome
and Y_BT for a binary outcome)for the illustration of CNV association analysis with penalized regression in CNVreg.</h2><span id='topic+CNVCOVY'></span><span id='topic+CNV'></span><span id='topic+Cov'></span><span id='topic+Y_QT'></span><span id='topic+Y_BT'></span>

<h3>Description</h3>

<p>Simulated data with copy number variants (CNV), Covariates (Cov), and outcomes traits (Y_QT for a continuous outcome
and Y_BT for a binary outcome)for the illustration of CNV association analysis with penalized regression in CNVreg.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("CNVCOVY")
</code></pre>


<h3>Format</h3>

<p>CNVCOVY.RData provides 4 datasets: CNV, Cov, Y_QT, and Y_BT.
</p>

<ul>
<li><p> CNV. A data frame of 2680 CNV records, it has 5 variables:
</p>

<ul>
<li><p> ID. The sample ID of the CNV records in each row. There are 797 unique IDs.
</p>
</li>
<li><p> CHR. An integer variable, the chromosome number of CNV records.
</p>
</li>
<li><p> BP1. A numeric variable, the starting breakpoint of the CNV records. 
</p>
</li>
<li><p> BP2. A numeric variable, the ending breakpoint of the CNV records. 
</p>
</li>
<li><p> TYPE. An integer variable, how many copies of the CNV present. 
</p>
</li></ul>

</li>
<li><p> Cov. A data frame contains covariats of 900 samples 
(including 797 samples in the CNV data set).
Cov has 3 variables.
</p>

<ul>
<li><p> ID. The sample ID of 900 individuals (900 unique IDs).
</p>
</li>
<li><p> Sex. An integer covariate, sex of each sample: 0 male, 1 female.
</p>
</li>
<li><p> Age. A numeric covariate, age of each sample. 
</p>
</li></ul>

</li>
<li><p> Y_QT and Y_BT are two data frames for outcomes traits. Y_QT contains a continuous trait. 
Y_BT contains a binary trait. Both have 2 variables
</p>

<ul>
<li><p> ID. The sample ID of 900 individuals (900 unique IDs).
</p>
</li>
<li><p> Y. Y_QT has a numeric variable range (-4.89 &ndash; 16.70)
Y_BT has an integer variable with controls 0 and cases 1.
</p>
</li></ul>

</li></ul>


<hr>
<h2 id='cvfit_WTSMTH'>Penalized Regression with Lasso and Weighted Fusion Penalties with Cross-Validation</h2><span id='topic+cvfit_WTSMTH'></span>

<h3>Description</h3>

<p>Uses n-fold cross-validation (CV) to fit a penalized regression model with Lasso 
penalty and weighted fusion penalty. Return the loss of all pair of tuning parameters, 
find the best pair of tuning parameters with the lowest loss, and estimate the regression coefficient.  
The CV process fine-tunes the tuning parameters required for the penalty terms
and find the pair of lambda_1 and lambda_2 that minimizes the average validation loss.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cvfit_WTSMTH(
  data,
  lambda1 = seq(-8, 0, 1),
  lambda2 = seq(-8, 8, 1),
  weight = NULL,
  family = c("gaussian", "binomial"),
  cv.control = list(n.fold = 5L, n.core = 1L, stratified = FALSE),
  iter.control = list(max.iter = 8L, tol.beta = 10^(-3), tol.loss = 10^(-6)),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cvfit_WTSMTH_+3A_data">data</code></td>
<td>
<p>An object of class &quot;WTsmth.data&quot; as generated by prep()</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_lambda1">lambda1</code></td>
<td>
<p>A numeric vector. Lambda_1 values to be considered that controls the Lasso penalty. 
Provided values will be transformed to 2^(lambda1). The default value is c(-8:0).
The user can customize the range and step_size of the candidate tuning parameters
In most cases, the user will need to run the function more than one time to 
adjust the range and step_size of tuning parameters
to locate to a reasonable range according to the 'Loss' and 'selected.lambda' 
from the previous round of model fitting</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_lambda2">lambda2</code></td>
<td>
<p>A numeric vector. Lambda_2 values to be considered that controls
the weighted fusion penalty.
Provided values will be transformed to 2^(lambda2). 
The default value is c(-8:8).
The user can customize the range and step_size of the candidate tuning parameters
In most cases, the user will need to run the function more than one time to 
adjust the range and step_size of tuning parameters
to locate to a reasonable range according to the 'Loss' and 'selected.lambda'
from the previous round of model fitting.</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_weight">weight</code></td>
<td>
<p>A character. The type of weighting. Must be one of 
('eql', 'keql', 'wcs', 'kwcs', 'wif', 'kwif'), which indicates the
equal weight, K x equal weight, Cosine similarity, K x cosine similarity, 
inverse frequency, and K x inverse frequency respectively, where K is the number of
individuals in each CNV active region. 
'eql' and 'keql' gives equal weight to adjacent CNVs.
'wcs' and 'kwcs' allow similar CNV fragments to have more similar effect size. 
'wif' and 'kwif' will encourage CNV with lower frequency to borrow 
information from nearby more frequent CNV fragments.
Considering that CNVs usually present in some CNV-active regions and there are
large regions in between with no CNV at all. K will describe the number of individuals 
having any CNV activities in a CNV-active region, and varying the weight according 
to the sample size across regions.</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_family">family</code></td>
<td>
<p>A character. The family of the outcome. Must be one of
&quot;gaussian&quot; (Y is continuous) or &quot;binomial&quot; (Y is binary).</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_cv.control">cv.control</code></td>
<td>
<p>A list object. Allows user to control cross-validation
procedure. Allowed elements are 'n.fold', the number of cross-validation
folds with a default value of 5, depends on the sample size, 
it can be chosen to have other folds (such as 3, 10); 
'n.core' is the number of cores to use in procedure, 
check available computation resource before choosing;
and 'stratified', if TRUE and 'family' = &quot;binomial&quot;, the folds will be
stratified within each category of Y (this option is recommended if either
category of the outcome is &quot;rare&quot;.)</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_iter.control">iter.control</code></td>
<td>
<p>A list object. Allows the user to control iteratively
update procedure. Allowed elements are 'max.iter', the maximum number
of iterations, it guarantees the function returns results within reasonable time;
'tol.beta' is the threshold below which the procedure is deemed converged, 
which controls the absolute difference between consecutive beta updates.
'tol.loss' is the threshold  below which the procedure
is deemed converged, which controls the difference in consecutive loss updates.</p>
</td></tr>
<tr><td><code id="cvfit_WTSMTH_+3A_verbose">verbose</code></td>
<td>
<p>A logical object. If 'TRUE', print progression updates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing 
1. 'Loss': The average loss of the validation set for all pairs of candidate tuning parameters, 
the smaller the loss, the better performance of the corresponding pair of parameters. 
2. 'selected.lambda' :The selected tuning parameter values that minimized the loss. 
3. 'coef'  the model coefficient estimate (coef) at the selected tuning parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Note we use here a very small example data set and few candidate lambda1 
# and lambda2 to expedite examples. 

# load toy dataset
data("CNVCOVY")

# prepare data format for regression analysis

## Continuous outcome Y_QT
frag_data &lt;- prep(CNV = CNV, Y = Y_QT, Z = Cov, rare.out = 0.05)
QT_tune &lt;- cvfit_WTSMTH(frag_data, 
                        lambda1 = seq(-4.75, -5.25, -0.25), 
                        lambda2 = seq(18, 22, 1), 
                        weight = "eql", 
                        family = "gaussian")
                        
## Binary outcome Y_BT

# We can directly replace frag_data$Y with Y_BT in the correct format,
# ensuring that the ordering matches that of the prepared object.

rownames(Y_BT) &lt;- Y_BT$ID
frag_data$Y &lt;- Y_BT[names(frag_data$Y), "Y"] |&gt; drop()
names(frag_data$Y) &lt;- rownames(frag_data$Z) 

# Or, we can also repeat the prep() call
# frag_data &lt;- prep(CNV = CNV, Y = Y_BT, Z = Cov, rare.out = 0.05)

BT_tune &lt;- cvfit_WTSMTH(frag_data, 
                        lambda1 = c(-5.25, -5, -4.75), 
                        lambda2 = c(5,  6, 7), 
                        weight = "eql",
                        family = "binomial")

</code></pre>

<hr>
<h2 id='fit_WTSMTH'>Penalized Regression with Lasso and Weighted Fusion Penalties with Given Parameters</h2><span id='topic+fit_WTSMTH'></span>

<h3>Description</h3>

<p>Performs penalized regression with Lasso penalty and weighted fusion penalty
for a given pair of tuning parameters (lambda1 and lambda2), which is 
determined by the user based on prior knowledge or use any number just for 
testing purpose.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_WTSMTH(
  data,
  lambda1,
  lambda2,
  weight = NULL,
  family = c("gaussian", "binomial"),
  iter.control = list(max.iter = 8L, tol.beta = 10^(-3), tol.loss = 10^(-6)),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_WTSMTH_+3A_data">data</code></td>
<td>
<p>An object of class &quot;WTsmth.data&quot; as generated by prep()</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_lambda1">lambda1</code></td>
<td>
<p>A scalar numeric. Lambda_1 value to be considered. Provided
value will be transformed to 2^(lambda1).</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_lambda2">lambda2</code></td>
<td>
<p>A scalar numeric Lambda_2 value to be considered. Provided
value will be transformed to 2^(lambda2).</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_weight">weight</code></td>
<td>
<p>A character. The type of weighting. Must be one of 
eql, keql, wcs, kwcs, wif, kwif indicating
equal weight, K x equal weight, Cosine similarity, K x cosine similarity, 
inverse frequency, and K x inverse frequency, where K is the number of
individuals in each CNV-active region. 
'eql' and 'keql' gives equal weight to adjacent CNVs.
'wcs' and 'kwcs' allow similar CNV fragments to have more similar effect size. 
'wif' and 'kwif' will encourage CNV with lower frequency to borrow 
information from nearby more frequent CNV fragments.
Considering that CNVs usually present in some CNV-active regions and there are
large regions in between with no CNV at all. K will describe the number of individuals 
having any CNV activities in a CNV-active region, and varying the weight according 
to the sample size across regions.</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_family">family</code></td>
<td>
<p>A character. The family of the outcome. Must be one of
&quot;gaussian&quot; (Y is continuous) or &quot;binomial&quot; (Y is binary).</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_iter.control">iter.control</code></td>
<td>
<p>A list object. Allows user to control iterative
update procedure. Allowed elements are &quot;max.iter&quot;, the maximum number
of iterations; &quot;tol.beta&quot;, the difference between consecutive beta
updates below which the procedure is deemed converged; and &quot;tol.loss&quot;,
the difference in consecutive loss updates below which the procedure
is deemed converged.</p>
</td></tr>
<tr><td><code id="fit_WTSMTH_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector. The estimated model parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Note we use here a very small example data set to expedite examples. 

# load toy dataset
data("CNVCOVY")

# prepare data format for regression analysis

## Continuous outcome Y_QT
frag_data &lt;- prep(CNV = CNV, Y = Y_QT, Z = Cov, rare.out = 0.05)
QT_fit &lt;- fit_WTSMTH(frag_data, 
                     lambda1 = -5, 
                     lambda2 = 21, 
                     weight = "eql", 
                     family = "gaussian")
                        
## Binary outcome Y_BT

# We can directly replace frag_data$Y with Y_BT in the correct format,
# ensuring that the ordering matches that of the prepared object.

rownames(Y_BT) &lt;- Y_BT$ID
frag_data$Y &lt;- Y_BT[names(frag_data$Y), "Y"] |&gt; drop()
names(frag_data$Y) &lt;- rownames(frag_data$Z) 

# Or, we can also repeat the prep() call
# frag_data &lt;- prep(CNV = CNV, Y = Y_BT, Z = Cov, rare.out = 0.05)

BT_fit &lt;- fit_WTSMTH(frag_data, 
                        lambda1 = -5, 
                        lambda2 = 6, 
                        weight = "eql",
                        family = "binomial")
</code></pre>

<hr>
<h2 id='prep'>Prepare Data for Analysis</h2><span id='topic+prep'></span>

<h3>Description</h3>

<p>Required preprocessing of analysis data. Function converts an individual's 
CNV events within a genomic region (from one chromosome) to a CNV profile 
curve, further processes it as CNV fragments, and filter out rare fragments. 
In addition, the adjacency relationship between CNV fragments is analyzed
and weight matrices are generated. The resulting 'WTsmth.data' object, 
is provided as input to the regression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep(CNV, Y, Z = NULL, rare.out = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_+3A_cnv">CNV</code></td>
<td>
<p>A data.frame in PLINK format. Specifically, must contain
columns: 
</p>

<ul>
<li><p> &quot;ID&quot;: character, unique identity for each sample
</p>
</li>
<li><p> &quot;CHR&quot;: integer, allowed range 1-22 NOTE: only 1 CHR can be present,
which means this function processes one chromosome at a time.
</p>
</li>
<li><p> &quot;BP1&quot;: integer, CNV event starting position, 
</p>
</li>
<li><p> &quot;BP2&quot;: integer, CNV event ending position, each record must have 
BP1 &lt;= BP2, CNV at least 1bp (or other unit length)
</p>
</li>
<li><p> &quot;TYPE&quot;: integer, range 0, 1, 3, 4, and larger allowed, i.e.,
2 is not allowed.
</p>
</li></ul>
</td></tr>
<tr><td><code id="prep_+3A_y">Y</code></td>
<td>
<p>A data.frame. Must include column &quot;ID&quot;. Must have 2 columns. For binary,
values must be 0 (control) or 1 (case). For continuous, values must be 
real number. Y$ID must contain all unique CNV$ID. Y and Z have the same IDs.</p>
</td></tr>
<tr><td><code id="prep_+3A_z">Z</code></td>
<td>
<p>A data.frame. Must include column &quot;ID&quot;. All other columns are covariates,
which can be continuous, binary, or categorical variables. At a minimum, 
Z must contain all unique CNV$ID values.</p>
</td></tr>
<tr><td><code id="prep_+3A_rare.out">rare.out</code></td>
<td>
<p>A scalar numeric in the range [0, 0.5); event rates below this value are
filtered out of the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An S3 object of class &quot;WTsmth.data&quot; extending a list object containing
</p>

<ul>
<li> <p><code>design</code> CNV data converted to design matrix.
</p>
</li>
<li> <p><code>Z</code> The processed covariate matrix.
</p>
</li>
<li> <p><code>Y</code> The processed response vector.
</p>
</li>
<li> <p><code>weight.structure</code> A Matrix object. The structure of the weight matrix.
</p>
</li>
<li> <p><code>weight.options</code> A matrix object. Each row is the multiplicative
vector to obtain each available weight. Specifically, the A matrix is
obtained as weight_option[i, ] * weight.structure where i = 1-6 with
1=&quot;eql&quot;, 2=&quot;keql&quot;, 3=&quot;wcs&quot;, 4=&quot;kwcs&quot;, 5=&quot;wif&quot;, and 6=&quot;kwif&quot;.
</p>
</li>
<li> <p><code>CNVR.info</code> A data.frame containing details about the fragment
structure.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Note we use here a very small example data set to expedite examples. 

# load toy dataset
data("CNVCOVY")

## Continuous outcome Y_QT
frag_data &lt;- prep(CNV = CNV, Y = Y_QT, Z = Cov, rare.out = 0.05)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
