<!DOCTYPE html><html><head><title>Help for package CodataGS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CodataGS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CodataGS-package'>
<p>Genomic Prediction Using SNP Codata</p></a></li>
<li><a href='#compute_GL'>
<p>Computes genomic relationship matrix</p></a></li>
<li><a href='#compute_phitau'>
<p>Computes models for the variance components</p></a></li>
<li><a href='#genomicEBV.w.codata'>
<p>Performs genomic prediction based on SNP codata.</p></a></li>
<li><a href='#hat.transf'>
<p>Transforms hat values</p></a></li>
<li><a href='#MME'>
<p>Mixed model equations</p></a></li>
<li><a href='#scaleZ'>
<p>Scales the genotype matrix.</p></a></li>
<li><a href='#summary.CodataGS'><p> Summary method for CodataGS objects</p></a></li>
<li><a href='#Transform'>
<p>Transforms hat values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genomic Prediction Using SNP Codata</td>
</tr>
<tr>
<td>Version:</td>
<td>1.43</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Lars Ronnegard</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lars Ronnegard &lt;lrn@du.se&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes genomic breeding values using external information on the markers. The package fits a linear mixed model with heteroscedastic random effects, where the random effect variance is fitted using a linear predictor and a log link. The method is described in Mouresan, Selle and Ronnegard (2019) &lt;<a href="https://doi.org/10.1101%2F636746">doi:10.1101/636746</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>Matrix</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-17 12:09:05 UTC; lrn</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-17 15:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='CodataGS-package'>
Genomic Prediction Using SNP Codata
</h2><span id='topic+CodataGS-package'></span><span id='topic+CodataGS'></span>

<h3>Description</h3>

<p>Computes genomic breeding values using external information on the markers. The package fits a linear mixed model with heteroscedastic random effects, where the random effect variance is fitted using a linear predictor and a log link. The method is described in Mouresan, Selle and Ronnegard (2019) &lt;doi:10.1101/636746&gt;.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> CodataGS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Genomic Prediction Using SNP Codata</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.43</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-05-17</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Lars Ronnegard</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Lars Ronnegard &lt;lrn@du.se&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Computes genomic breeding values using external information on the markers. The package fits a linear mixed model with heteroscedastic random effects, where the random effect variance is fitted using a linear predictor and a log link. The method is described in Mouresan, Selle and Ronnegard (2019) &lt;doi:10.1101/636746&gt;.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> Matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
CodataGS-package        Genomic Prediction Using SNP Codata
MME                     Mixed model equations
Transform               Transforms hat values
compute_GL              Computes genomic relationship matrix
compute_phitau          Computes models for the variance components
genomicEBV.w.codata     Performs genomic prediction based on SNP
                        codata.
hat.transf              Transforms hat values
scaleZ                  Scales the genotype matrix.
summary.CodataGS        Summary method for CodataGS objects
</pre>
<p>This package performs genomic prediction based on SNP codata. The main function is genomicEBV.w.codata.
</p>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>
<p>Maintainer: Lars Ronnegard &lt;lrn@du.se&gt;
</p>

<hr>
<h2 id='compute_GL'>
Computes genomic relationship matrix
</h2><span id='topic+compute_GL'></span>

<h3>Description</h3>

<p>This function computes the genomic relationship matrix, G, together with its matrix square root, L.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_GL(Z, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_GL_+3A_z">Z</code></td>
<td>

<p>Scaled matrix with genotype information
</p>
</td></tr>
<tr><td><code id="compute_GL_+3A_w">w</code></td>
<td>

<p>weights
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>L</code></td>
<td>
<p>Square root matrix of G </p>
</td></tr>
<tr><td><code>svdVec</code></td>
<td>
<p>Vectors in the Single Value Decomposition of G</p>
</td></tr>
<tr><td><code>svdD</code></td>
<td>
<p>Diagonal elements in the Single Value Decomposition of G</p>
</td></tr>
<tr><td><code>wZt</code></td>
<td>
<p>weights times the transpose of Z</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
N &lt;- 20 #Number of individuals
k &lt;- 30 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
sim.res &lt;- compute_GL(Z, w = rep(1,k))
</code></pre>

<hr>
<h2 id='compute_phitau'>
Computes models for the variance components
</h2><span id='topic+compute_phitau'></span>

<h3>Description</h3>

<p>This function computes the residual variance, the SNP variances and the linear predictor for the SNP variance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_phitau(dev, hv, devu, hvu, X.rand.disp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_phitau_+3A_dev">dev</code></td>
<td>

<p>Deviance values
</p>
</td></tr>
<tr><td><code id="compute_phitau_+3A_hv">hv</code></td>
<td>

<p>Hat values for the observed response values
</p>
</td></tr>
<tr><td><code id="compute_phitau_+3A_devu">devu</code></td>
<td>

<p>Deviance values computed for the random effects
</p>
</td></tr>
<tr><td><code id="compute_phitau_+3A_hvu">hvu</code></td>
<td>

<p>Hat values for the random effects
</p>
</td></tr>
<tr><td><code id="compute_phitau_+3A_x.rand.disp">X.rand.disp</code></td>
<td>

<p>Design matrix used in the linear predictor for the SNP variance model.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>var.e</code></td>
<td>
<p>Residual variance</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>Vector of SNP variances</p>
</td></tr>
<tr><td><code>coef</code></td>
<td>
<p>Fitted coefficients for the linear predictor in the SNP variance model</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
N &lt;- 20 #Number of individuals
k &lt;- 30 #Number of SNPs with all marker positions including a QTL
#Simulated deviances and hat values
dev &lt;- rnorm(N)^2
hv &lt;- runif(N, 0.1, 0.5)
devu &lt;- rnorm(k)^2
hvu &lt;- runif(k, 0.1, 0.85)
X.rand.disp &lt;- matrix(1, k, 1)
sim.res &lt;- compute_phitau(dev, hv, devu, hvu, X.rand.disp)
</code></pre>

<hr>
<h2 id='genomicEBV.w.codata'>
Performs genomic prediction based on SNP codata. 
</h2><span id='topic+genomicEBV.w.codata'></span>

<h3>Description</h3>

<p>The main function of the package. The input includes response values, a design matrix for the fixed effects, a matrix with SNP genotype data and a design matrix for the SNP codata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomicEBV.w.codata(y, X, Z, X.SNPcodata, Z.test = NULL, max.iter = 100, conv.crit = 1e-5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genomicEBV.w.codata_+3A_y">y</code></td>
<td>

<p>Response values
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_x">X</code></td>
<td>

<p>Design matrix for the fixed effects
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_z">Z</code></td>
<td>

<p>Genotype matrix with element values of 0, 1 or 2
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_x.snpcodata">X.SNPcodata</code></td>
<td>

<p>Design matrix for the linear predictor of the SNP variances.
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_z.test">Z.test</code></td>
<td>

<p>An optional genotype matrix for a test data set.
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_max.iter">max.iter</code></td>
<td>

<p>The maximum number of iterations
</p>
</td></tr>
<tr><td><code id="genomicEBV.w.codata_+3A_conv.crit">conv.crit</code></td>
<td>

<p>The value of the convergence criterion.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By specifying the matrix <code>Z.test</code> in the input, the function computes predicted genomic breeding values for an out-of-sample data set.
</p>


<h3>Value</h3>

<table>
<tr><td><code>gEBV</code></td>
<td>
<p>Genomic breeding values</p>
</td></tr>
<tr><td><code>predicted.gEBV</code></td>
<td>
<p>Genomic breeding values based on the genotypes in <code>Z.test</code> </p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>Computed SNP weights</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>Fitted SNP effects</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Fitted fixed effects</p>
</td></tr>
<tr><td><code>disp.beta</code></td>
<td>
<p>Fitted coefficients in the linear predictor for the SNP variance model</p>
</td></tr>
<tr><td><code>Converge</code></td>
<td>
<p>Shows whether the algorithm has converged or not</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>The number of iterations used</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#######
#Simulation part
set.seed(1234)
N &lt;- 200 #Number of individuals
k &lt;- 300 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
x1 &lt;- c(rep(1,k/2), rep(0,k/2)) #An indicator for the SNPs. 
#The first k/2 SNPs and the last k/2 have different variances
#Simulate linear predictor for the random effect variance
lin.pred &lt;- 0 + 2*x1 
X.snp &lt;- model.matrix( ~ x1 ) #Corresponding design matrix
u &lt;- rnorm(k, 0 , sqrt( exp(lin.pred) )) 
#Took the square root here because it is the SD that is specified.
#and exp() because we are modelling a log link.
u.scaled &lt;- u/as.numeric( sqrt( var( crossprod(t(Z), u) )) ) 
#Scaled by the variance of the breeding values
e &lt;- rnorm(N) #A residual variance
mu &lt;- 0
y &lt;- mu + crossprod(t(Z),u.scaled) + e
######
#Estimation part
mod1 &lt;- genomicEBV.w.codata(y = as.numeric(y), 
          X = matrix(1, N, 1), Z = Z, X.SNPcodata = X.snp)
#To fit gBLUP just specify X.SNPcodata = matrix(1, k, 1)
cat("Correlation between true and estimated BV for the codata model:")
cat(cor(crossprod(t(Z),u.scaled), mod1$gEBV), "\n")
</code></pre>

<hr>
<h2 id='hat.transf'>
Transforms hat values
</h2><span id='topic+hat.transf'></span>

<h3>Description</h3>

<p>Transforms hat values between the SNP-BLUP model and the gBLUP model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hat.transf(C22, transf, vc, k, N, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hat.transf_+3A_c22">C22</code></td>
<td>

<p>Submatrix of the inverse of the LHS in the MME
</p>
</td></tr>
<tr><td><code id="hat.transf_+3A_transf">transf</code></td>
<td>

<p>A transformation matrix.
</p>
</td></tr>
<tr><td><code id="hat.transf_+3A_vc">vc</code></td>
<td>

<p>Genetic variance
</p>
</td></tr>
<tr><td><code id="hat.transf_+3A_k">k</code></td>
<td>

<p>Number of SNPs
</p>
</td></tr>
<tr><td><code id="hat.transf_+3A_n">N</code></td>
<td>

<p>Number of individuals
</p>
</td></tr>
<tr><td><code id="hat.transf_+3A_w">w</code></td>
<td>

<p>SNP weights
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Transformed hat values
</p>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>

<hr>
<h2 id='MME'>
Mixed model equations
</h2><span id='topic+MME'></span>

<h3>Description</h3>

<p>A fast version of the Henderson's mixed model equations (MME)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MME(y, X, Z, var.e, var.u)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MME_+3A_y">y</code></td>
<td>

<p>Response
</p>
</td></tr>
<tr><td><code id="MME_+3A_x">X</code></td>
<td>

<p>Design matrix for fixed effects
</p>
</td></tr>
<tr><td><code id="MME_+3A_z">Z</code></td>
<td>

<p>Design matrix for the random effects
</p>
</td></tr>
<tr><td><code id="MME_+3A_var.e">var.e</code></td>
<td>

<p>Residual variance
</p>
</td></tr>
<tr><td><code id="MME_+3A_var.u">var.u</code></td>
<td>

<p>Genetic variance
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimates of fixed effects</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>Fitted random effects</p>
</td></tr>
<tr><td><code>hv</code></td>
<td>
<p>Hat values</p>
</td></tr>
<tr><td><code>dev</code></td>
<td>
<p>Deviances</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>

<hr>
<h2 id='scaleZ'>
Scales the genotype matrix.
</h2><span id='topic+scaleZ'></span>

<h3>Description</h3>

<p>Scales the genotype matrix so that ZZ' gives the genomic relationship matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scaleZ(Z, freq1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scaleZ_+3A_z">Z</code></td>
<td>

<p>Genotype matrix with element values 0, 1 and 2
</p>
</td></tr>
<tr><td><code id="scaleZ_+3A_freq1">freq1</code></td>
<td>

<p>Optional input parameter with allele frequencies. A vector of length equal to the number of columns in Z.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Z</code></td>
<td>
<p>Scaled genotype matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#######
#Simulation part
set.seed(1234)
N &lt;- 200 #Number of individuals
k &lt;- 300 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
sim.res &lt;- scaleZ(Z)
</code></pre>

<hr>
<h2 id='summary.CodataGS'> Summary method for CodataGS objects</h2><span id='topic+summary'></span><span id='topic+summary.CodataGS'></span>

<h3>Description</h3>

<p>A summary method for the object class <code>CodataGS</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CodataGS'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.CodataGS_+3A_object">object</code></td>
<td>
<p>A CodataGS object </p>
</td></tr>
<tr><td><code id="summary.CodataGS_+3A_...">...</code></td>
<td>
<p> arguments not used </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Provides a concise summary of CodataGS objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#######
#Simulation part
set.seed(1234)
N &lt;- 200 #Number of individuals
k &lt;- 300 #Number of SNPs with all marker positions including a QTL
Z1 &lt;- matrix(0, N, k )
Z2 &lt;- matrix(0, N, k )
Z1[1:N, 1] &lt;- rbinom(N, 1, 0.5) #Simulated phased SNP matrices
Z2[1:N, 1] &lt;- rbinom(N, 1, 0.5) 
LD.par &lt;- 0.2 #A parameter to simulate LD. 0 gives full LD, and 0.5 no LD
for (j in 2:k) {
  Z1[1:N, j] &lt;- abs( Z1[1:N, j-1] - rbinom(N, 1, LD.par) )
  Z2[1:N, j] &lt;- abs( Z2[1:N, j-1] - rbinom(N, 1, LD.par) )
} 
Z &lt;- Z1 + Z2 #Genotypic SNP matrix
x1 &lt;- c(rep(1,k/2), rep(0,k/2)) #An indicator for the SNPs. 
#The first k/2 SNPs and the last k/2 have different variances
#Simulate linear predictor for the random effect variance
lin.pred &lt;- 0 + 2*x1 
X.snp &lt;- model.matrix( ~ x1 ) #Corresponding design matrix
u &lt;- rnorm(k, 0 , sqrt( exp(lin.pred) )) 
#Took the square root here because it is the SD that is specified.
#and exp() because we are modelling a log link.
u.scaled &lt;- u/as.numeric( sqrt( var( crossprod(t(Z), u) )) ) 
#Scaled by the variance of the breeding values
e &lt;- rnorm(N) #A residual variance
mu &lt;- 0
y &lt;- mu + crossprod(t(Z),u.scaled) + e
######
#Estimation part
mod1 &lt;- genomicEBV.w.codata(y = as.numeric(y), 
          X = matrix(1, N, 1), Z = Z, X.SNPcodata = X.snp)
summary(mod1)
</code></pre>

<hr>
<h2 id='Transform'>
Transforms hat values
</h2><span id='topic+Transform'></span>

<h3>Description</h3>

<p>The function calls the hat.transf function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Transform(X, L, var.e, var.u, v, svdVec, svdD, wZt, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Transform_+3A_x">X</code></td>
<td>

<p>Design matrix for the fixed effects
</p>
</td></tr>
<tr><td><code id="Transform_+3A_l">L</code></td>
<td>

<p>Square root matrix of the genomic relationship matrix, G
</p>
</td></tr>
<tr><td><code id="Transform_+3A_var.e">var.e</code></td>
<td>

<p>Residual variance
</p>
</td></tr>
<tr><td><code id="Transform_+3A_var.u">var.u</code></td>
<td>

<p>Genetic variance
</p>
</td></tr>
<tr><td><code id="Transform_+3A_v">v</code></td>
<td>

<p>Random effects
</p>
</td></tr>
<tr><td><code id="Transform_+3A_svdvec">svdVec</code></td>
<td>

<p>Vector from the Single Value Decomposition of G
</p>
</td></tr>
<tr><td><code id="Transform_+3A_svdd">svdD</code></td>
<td>

<p>Diagonal elements of the Single Value Decomposition of G
</p>
</td></tr>
<tr><td><code id="Transform_+3A_wzt">wZt</code></td>
<td>

<p>Weights times the transpose of the scaled genotype matrix
</p>
</td></tr>
<tr><td><code id="Transform_+3A_w">w</code></td>
<td>

<p>Fitted SNP weights
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>u</code></td>
<td>
<p>SNP effects</p>
</td></tr>
<tr><td><code>qu</code></td>
<td>
<p>Hat values for the SNP effects</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Lars Ronnegard
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
