<!DOCTYPE html><html><head><title>Help for package coil</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {coil}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aa_coi_PHMM'><p>Amino acid profile hidden Markov model for coi5p.</p></a></li>
<li><a href='#censored_translation'><p>Censored Translation of a DNA string.</p></a></li>
<li><a href='#coi5p'><p>Build a coi5p object from a DNA sequence string.</p></a></li>
<li><a href='#coi5p_pipe'><p>Run the entire coi5p pipeline for an input sequence.</p></a></li>
<li><a href='#coil'><p><span class="pkg">coil</span>: evaluation of COI-5P barcode data</p></a></li>
<li><a href='#example_barcode_data'><p>Example barcode data.</p></a></li>
<li><a href='#example_nt_string'><p>Example coi5p DNA sequence string</p></a></li>
<li><a href='#flatten_coi5p'><p>Flatten a list of coi5p output objects into a dataframe.</p></a></li>
<li><a href='#frame'><p>Take a coi5p sequence and place it in reading frame.</p></a></li>
<li><a href='#indel_check'><p>Check if a coi5p sequence likely contains an error.</p></a></li>
<li><a href='#individual_AAbin'><p>build an AAbin with ape.</p></a></li>
<li><a href='#individual_DNAbin'><p>build an DNAbin with ape.</p></a></li>
<li><a href='#ins_front_trim'><p>Check sequence for an early large string of deletions.</p>
If it exists then return the starting index by which to slice the path and the string.</a></li>
<li><a href='#leading_ins'><p>Check for a large number of leading inserted bases.</p></a></li>
<li><a href='#new_coi5p'><p>Build a new coi5p class instance.</p></a></li>
<li><a href='#nt_coi_PHMM'><p>Nucleotide profile hidden Markov model for coi5p.</p></a></li>
<li><a href='#print.coi5p'><p>Print a summary of a coi5p object.</p></a></li>
<li><a href='#set_frame'><p>Take an input sequence and get it into the reading frame.</p></a></li>
<li><a href='#subsetPHMM'><p>Subset an existing PHMM.</p></a></li>
<li><a href='#trans_df'><p>Data frame containing the translation table recommendation.</p></a></li>
<li><a href='#translate'><p>Translate a coi5p sequence.</p></a></li>
<li><a href='#translate_codon'><p>Censored Translation of a codon.</p></a></li>
<li><a href='#validate_coi5p'><p>Validate the new coi5p class instance.</p></a></li>
<li><a href='#which_trans_table'><p>Determine the translation table to use for a given taxonomic group.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Contextualization and Evaluation of COI-5P Barcode Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>Author:</td>
<td>Cameron M. Nugent</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cameron M. Nugent &lt;camnugent@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Designed for the cleaning, contextualization and assessment of cytochrome c 
    oxidase I DNA barcode data (COI-5P, or the five prime portion of COI). It contains 
    functions for placing COI-5P barcode sequences into a common reading frame, 
    translating DNA sequences to amino acids and for assessing the likelihood that a 
    given barcode sequence includes an insertion or deletion error. The error assessment 
    relies on the comparison of input sequences against nucleotide and amino acid profile
    hidden Markov models (PHMMs) (for details see Durbin et al. 1998, ISBN: 9780521629713) 
    trained on a taxonomically diverse set of reference sequences. The functions are 
    provided as a complete pipeline and are also available individually for efficient and
    targeted analysis of barcode data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, aphid, seqinr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-11 11:13:16 UTC; cnuge</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-11 11:30:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='aa_coi_PHMM'>Amino acid profile hidden Markov model for coi5p.</h2><span id='topic+aa_coi_PHMM'></span>

<h3>Description</h3>

<p>This model is stored in the coi5p package and was trained on a representative
sample of the barcode of life database (http://www.boldsystems.org/index.php).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa_coi_PHMM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>PHMM</code> of length 14.
</p>

<hr>
<h2 id='censored_translation'>Censored Translation of a DNA string.</h2><span id='topic+censored_translation'></span>

<h3>Description</h3>

<p>Translate a DNA sequence using the censored translation table.
This translates codons for which the amino acids are unambiguous across
mitochondrial genetic codes across the animal kingdom and does not
translate those for which the amino acid varies,
but rather outputs a ? in the string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>censored_translation(dna_str, reading_frame = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="censored_translation_+3A_dna_str">dna_str</code></td>
<td>
<p>The DNA string to be translated.</p>
</td></tr>
<tr><td><code id="censored_translation_+3A_reading_frame">reading_frame</code></td>
<td>
<p>Set the reading frame of the sequence. reading_frame = 1 (default)
means the first bp in the string is the start of the first codon, can pass: 1, 2, or 3.
i.e. reading_frame = 2 indicates that the second bp in the string is the start of the first codon.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#translate a string of DNA:
censored_translation(example_nt_string)
#manually override the reading frame:
censored_translation(example_nt_string, reading_frame = 2)
</code></pre>

<hr>
<h2 id='coi5p'>Build a coi5p object from a DNA sequence string.</h2><span id='topic+coi5p'></span>

<h3>Description</h3>

<p>Build a coi5p object from a DNA sequence string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coi5p(x = character(), name = character())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coi5p_+3A_x">x</code></td>
<td>
<p>A nucleotide string.
Valid characters within the nucleotide string are: &quot;a&quot;, &quot;t&quot;, &quot;g&quot;, &quot;c&quot;, &quot;-&quot; and &quot;n&quot;.
coil treats both '-' and 'n' characters as placeholder nucleotides when comparing to the PHMM.
The nucleotide string can be input as upper case, but will be automatically converted to lower case.</p>
</td></tr>
<tr><td><code id="coi5p_+3A_name">name</code></td>
<td>
<p>An optional character string that serves as the identifier for the sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"coi5p"</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#build an unnamed coi5p object
dat = coi5p(example_nt_string)
#build a named coi5p sequence
dat = coi5p(example_nt_string, name = "example_seq1")
#available components in the coi5p object:
dat$raw
dat$name
</code></pre>

<hr>
<h2 id='coi5p_pipe'>Run the entire coi5p pipeline for an input sequence.</h2><span id='topic+coi5p_pipe'></span>

<h3>Description</h3>

<p>This function will take a raw DNA sequence string and run each of the coi5p methods in turn
(coi5p, frame, translate, indel_check). Note that if you are not interested in all components
of the output (i.e. only want sequences set in frame reading  or translated), then the
coi5p analysis functions can be called individually to avoid unnecessary computation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coi5p_pipe(
  x,
  ...,
  name = character(),
  trans_table = 0,
  frame_offset = 0,
  triple_translate = FALSE,
  nt_PHMM = coil::nt_coi_PHMM,
  aa_PHMM = coil::aa_coi_PHMM,
  indel_threshold = -358.88
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coi5p_pipe_+3A_x">x</code></td>
<td>
<p>A nucleotide string.
Valid characters within the nucleotide string are: 'a', 't', 'g', 'c', '-', and 'n'.
The nucleotide string can be input as upper case, but will be automatically converted to lower case.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed between methods.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_name">name</code></td>
<td>
<p>An optional character string. Identifier for the sequence.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_trans_table">trans_table</code></td>
<td>
<p>The translation table to use for translating from nucleotides to amino acids.
Default is 0, which indicates that censored translation should be performed. If the taxonomy
of the sample is known, use the function which_trans_table() to determine the translation table to use.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_frame_offset">frame_offset</code></td>
<td>
<p>The offset to the reading frame to be applied for translation. By default the offset
is zero, so the first character in the framed sequence is considered the first nucleotide of the first codon.
Passing frame_offset = 1 would make the second character in the framed sequence the first nucleotide of
the first codon.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_triple_translate">triple_translate</code></td>
<td>
<p>Optional argument indicating if the translation of sequences should be tested in all three forward
reading frames. The reading frame with the most likely amino acid PHMM score is returned.
This will decrease the rate of sequencing framing errors, at the cost of increased processing time.
Note this argument will overrule any passed frame_offset value (all options tried). Default is False.</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_nt_phmm">nt_PHMM</code></td>
<td>
<p>The profile hidden Markov model against which the raw sequence should be compared in the framing step.
Default is the full COI-5P nucleotide PHMM (nt_coi_PHMM).</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_aa_phmm">aa_PHMM</code></td>
<td>
<p>The profile hidden Markov model against which the translated amino acid sequence should be compared
in the indel_check step. Default is the full COI-5P amino acid PHMM (aa_coi_PHMM).</p>
</td></tr>
<tr><td><code id="coi5p_pipe_+3A_indel_threshold">indel_threshold</code></td>
<td>
<p>The log likelihood threshold used to assess whether or not sequences
are likely to contain an indel. Default is -358.88. Values lower than this will be classified
as likely to contain an indel and values higher will be classified as not likely to contain an indel.
For recommendations on selecting a indel_threshold value, consult: Nugent et al. 2019 (doi: https://doi.org/10.1101/2019.12.12.865014).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"coi5p"</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coi5p">coi5p</a></code>
</p>
<p><code><a href="#topic+frame">frame</a></code>
</p>
<p><code><a href="#topic+translate">translate</a></code>
</p>
<p><code><a href="#topic+indel_check">indel_check</a></code>
</p>
<p><code><a href="#topic+which_trans_table">which_trans_table</a></code>
</p>
<p><code><a href="#topic+subsetPHMM">subsetPHMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat = coi5p_pipe(example_nt_string)
#full coi5p object can then be printed
dat
#components of output coi5p object can be called individually:
dat$raw    #raw input sequence
dat$name   #name that was passed
dat$framed #sequence in common reading frame
dat$aaSeq  #sequence translated to amino acids (censored)
dat$indel_likely #whether an insertion or deletion likely exists in the sequence
dat$stop_codons #whether or not there are stop codons in the amino acid sequence
dat = coi5p_pipe(example_nt_string , trans_table = 5)
dat$aaSeq #sequence translated to amino acids using designated translation table
</code></pre>

<hr>
<h2 id='coil'><span class="pkg">coil</span>: evaluation of COI-5P barcode data</h2><span id='topic+coil'></span>

<h3>Description</h3>

<p><span class="pkg">coil</span> is an R package designed for the cleaning, contextualization and assessment of cytochrome c oxidase I DNA
barcode data (<a href="https://en.wikipedia.org/wiki/Cytochrome_c_oxidase_subunit_I">COI-5P</a>, or the five prime portion of COI).
It contains functions for placing COI-5P barcode sequences into a common reading frame, translating DNA sequences
to amino acids and for assessing the likelihood that a given barcode sequence includes an insertion or deletion error.
These functions are provided as a single function analysis pipeline and are also available individually for efficient
and targeted analysis of barcode data.
</p>


<h3>Details</h3>

<p><span class="pkg">coil</span> is built around the custom 'coi5p' object, which takes a COI-5P DNA barcode sequence as input.
The package contains functions for: setting a sequence in reading frame, translating the sequence to amino acids
and checking the sequence for evidence of insertion or deletion errors
</p>


<h3>Functions</h3>


<ul>
<li> <p><code><a href="#topic+coi5p_pipe">coi5p_pipe</a></code> Run the entire coi5p analysis pipeline for an input sequence.
</p>
</li>
<li> <p><code><a href="#topic+coi5p">coi5p</a></code> Builds a coi5p class object.
</p>
</li>
<li> <p><code><a href="#topic+frame">frame</a></code> Sets the sequence into a common reading frame
</p>
</li>
<li> <p><code><a href="#topic+which_trans_table">which_trans_table</a></code> Suggests a translation table for a taxonomic designation.
</p>
</li>
<li> <p><code><a href="#topic+censored_translation">censored_translation</a></code> Conducts translation, but ambigious codons are translated to placeholders.
</p>
</li>
<li> <p><code><a href="#topic+translate">translate</a></code> Translate a DNA sequence to amino acids.
</p>
</li>
<li> <p><code><a href="#topic+indel_check">indel_check</a></code> Check to see if an insertion or deletion error is likely.
</p>
</li></ul>



<h3>Data</h3>


<ul>
<li> <p><code><a href="#topic+example_nt_string">example_nt_string</a></code> String of DNA barcode data used in the package documentation's examples.
</p>
</li>
<li> <p><code><a href="#topic+example_barcode_data">example_barcode_data</a></code> A dataframe of coi5p barcode data, demonstrating different example cases.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Cameron M. Nugent
</p>

<hr>
<h2 id='example_barcode_data'>Example barcode data.</h2><span id='topic+example_barcode_data'></span>

<h3>Description</h3>

<p>A nine line dataframe of coi5p barcode data with the following columns:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_barcode_data
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 9 rows and 5 columns.
</p>


<h3>Details</h3>

<p>id - the unique identifier for the sample
</p>
<p>genetic_code - the genetic code for translation of the sample (features NA for unknowns)
</p>
<p>taxa -  a taxonomic designation associated with the sample
</p>
<p>sequence - the DNA sequence associated with the sample
</p>
<p>notes - notes on the sequence structure
</p>

<hr>
<h2 id='example_nt_string'>Example coi5p DNA sequence string</h2><span id='topic+example_nt_string'></span>

<h3>Description</h3>

<p>This string of barcode data is used in the package documentation's examples
and within the vignette demonstrating how to use the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_nt_string
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 1.
</p>

<hr>
<h2 id='flatten_coi5p'>Flatten a list of coi5p output objects into a dataframe.</h2><span id='topic+flatten_coi5p'></span>

<h3>Description</h3>

<p>This helper function is designed to act upon a list of coi5p objects and extract the object components
that the user requires.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flatten_coi5p(x, keep_cols = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flatten_coi5p_+3A_x">x</code></td>
<td>
<p>A list of coi5p objects.</p>
</td></tr>
<tr><td><code id="flatten_coi5p_+3A_keep_cols">keep_cols</code></td>
<td>
<p>The name of a coi5p object component, or a vector of components that should be turned into
dataframe columns. Available components are: name, raw, framed, aaSeq, aaScore, indel_likely, stop_codons.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with the coi5p object information flattened into columns.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coi5p_pipe">coi5p_pipe</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#create a list of coi5p objects
coi_output = lapply(example_barcode_data$sequence, function(x){
    coi5p_pipe(x)
  })
#flatten the list into a dataframe
coi_df = flatten_coi5p(coi_output)
#extract only a single column
coi_framed = flatten_coi5p(coi_output, keep_cols = "framed")
#or subset multiple columns
coi_framed = flatten_coi5p(coi_output, keep_cols = c("framed", "aaSeq"))
</code></pre>

<hr>
<h2 id='frame'>Take a coi5p sequence and place it in reading frame.</h2><span id='topic+frame'></span><span id='topic+frame.coi5p'></span>

<h3>Description</h3>

<p>Take a coi5p sequence and place it in reading frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frame(x, ...)

## S3 method for class 'coi5p'
frame(x, ..., nt_PHMM = nt_coi_PHMM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="frame_+3A_x">x</code></td>
<td>
<p>A coi5p class object.</p>
</td></tr>
<tr><td><code id="frame_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed between methods.</p>
</td></tr>
<tr><td><code id="frame_+3A_nt_phmm">nt_PHMM</code></td>
<td>
<p>The profile hidden Markov model against which the raw sequence should be compared. Default is the
full COI-5P nucleotide PHMM (nt_coi_PHMM).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function compares the raw sequence against the nucleotide PHMM using the Viterbi algorithm (for details see Durbin et al.
1998, ISBN: 9780521629713). The path of hidden states produced by the comparison is used to establish the reading frame of the
sequence. If leading insert states are present, the front of the sequence is trimmed to the first continuous set of match states
and the sequence is re-compared to the nucleotide PHMM. This is done because spurious or outlier matches early in the sequence
can lead to incorrect establishment of the reading frame. Realigning only the truncated version of the sequence to the PHMM
improves correct reading frame establishment, although this can also result in the loss of a few bp of true barcode sequence
on the peripherals of the sequence.
</p>


<h3>Value</h3>

<p>An object of class <code>"coi5p"</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coi5p">coi5p</a></code>
</p>
<p><code><a href="#topic+subsetPHMM">subsetPHMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#previously run function:
dat = coi5p(example_nt_string)

dat = frame(dat)

#additional components in output coi5p object:
dat$framed
</code></pre>

<hr>
<h2 id='indel_check'>Check if a coi5p sequence likely contains an error.</h2><span id='topic+indel_check'></span><span id='topic+indel_check.coi5p'></span>

<h3>Description</h3>

<p>Check if a coi5p sequence likely contains an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indel_check(x, ...)

## S3 method for class 'coi5p'
indel_check(x, ..., indel_threshold = -358.88, aa_PHMM = aa_coi_PHMM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="indel_check_+3A_x">x</code></td>
<td>
<p>A coi5p class object for which frame() and translate() have been run.</p>
</td></tr>
<tr><td><code id="indel_check_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed between methods.</p>
</td></tr>
<tr><td><code id="indel_check_+3A_indel_threshold">indel_threshold</code></td>
<td>
<p>The log likelihood threshold used to assess whether or not sequences
are likely to contain an indel. Default is -358.88. Values lower than this will be classified
as likely to contain an indel and values higher will be classified as not likely to contain an indel.</p>
</td></tr>
<tr><td><code id="indel_check_+3A_aa_phmm">aa_PHMM</code></td>
<td>
<p>The profile hidden Markov model against which the translated amino acid sequence should be compared.
Default is the full COI-5P amino acid PHMM (aa_coi_PHMM).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The indel check function analyzes the framed and translated DNA sequences in two ways in order to
allow users to make an informed decision about whether or not a DNA sequence contains a frameshift error.
This test is designed to detect insertion or deletion errors resulting from technical errors in DNA sequencing,
but can in some instances identify biological contaminants (i.e. if the contaminant sequence uses a different
genetic code than the target, or if the contaminants are things such as pseudogenes that possess sequences that
are highly divergent from animal COI-5P sequences).
</p>
<p>The two tests performed are: (1) a query for stop codons in the amino acid sequence and (2) an evaluation of the
log likelihood value resulting from the comparison of the framed coi5p amino acid sequence against the COI-5P
amino acid PHMM. The default likelihood value for identifying a sequence is likely erroneous is -358.88. Sequences with
likelihood values lower than this will receive an indel_likely value of TRUE. The threshold of -358.88 was experimentally
determined to be the optimal likelihood threshold for separating of full-length sequences with and without errors when
the censored translation option is used. Sequences will have higher likelihood values when a specific genetic code is used.
Sequences will have lower likelihood values when they are not complete barcode sequences (i.e. &lt;500bp in length). For these
reasons the likelihood threshold is not a specific value but a parameter that can be altered based on the type of translation
and length of the sequences. Below are experimentally determined suggested values for different size and translation table
combinations.
</p>
<p>Short barcode sequences, known genetic code: indel_threshold = -354.44
</p>
<p>Short barcode sequences, unknown genetic code: indel_threshold = -440.24
</p>
<p>Full length barcode sequences, known genetic code: indel_threshold = -246.20
</p>
<p>Full length barcode sequences, unknown genetic code: indel_threshold = -358.88
</p>
<p>Source: Nugent et al. 2019 (doi: https://doi.org/10.1101/2019.12.12.865014).
</p>


<h3>Value</h3>

<p>An object of class <code>"coi5p"</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coi5p">coi5p</a></code>
</p>
<p><code><a href="#topic+frame">frame</a></code>
</p>
<p><code><a href="#topic+translate">translate</a></code>
</p>
<p><code><a href="#topic+subsetPHMM">subsetPHMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#previously run functions:
dat = coi5p(example_nt_string)
dat = frame(dat)
dat = translate(dat)
#current function
dat = indel_check(dat)
#with custom indel threshold
dat = indel_check(dat, indel_threshold = -400)
#additional components in output coi5p object:
dat$stop_codons #Boolean - Indicates if there are stop codons in the amino acid sequence.
dat$indel_likely #Boolean - Indicates if the likelihood score below the specified indel_threshold.
dat$aaScore #view the amino acid log likelihood score
</code></pre>

<hr>
<h2 id='individual_AAbin'>build an AAbin with ape.</h2><span id='topic+individual_AAbin'></span>

<h3>Description</h3>

<p>build an AAbin with ape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>individual_AAbin(aa_string)
</code></pre>

<hr>
<h2 id='individual_DNAbin'>build an DNAbin with ape.</h2><span id='topic+individual_DNAbin'></span>

<h3>Description</h3>

<p>Switches the dashes in the seq - to n
</p>


<h3>Usage</h3>

<pre><code class='language-R'>individual_DNAbin(dna_string)
</code></pre>

<hr>
<h2 id='ins_front_trim'>Check sequence for an early large string of deletions.
If it exists then return the starting index by which to slice the path and the string.</h2><span id='topic+ins_front_trim'></span>

<h3>Description</h3>

<p>Check sequence for an early large string of deletions.
If it exists then return the starting index by which to slice the path and the string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ins_front_trim(path_out, search_scope = 15)
</code></pre>

<hr>
<h2 id='leading_ins'>Check for a large number of leading inserted bases.</h2><span id='topic+leading_ins'></span>

<h3>Description</h3>

<p>If this is the case, TRUE is returned and the PHMM
should be run a second time on the truncated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leading_ins(seq_path)
</code></pre>

<hr>
<h2 id='new_coi5p'>Build a new coi5p class instance.</h2><span id='topic+new_coi5p'></span>

<h3>Description</h3>

<p>Build a new coi5p class instance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>new_coi5p(x = character(), name = character())
</code></pre>

<hr>
<h2 id='nt_coi_PHMM'>Nucleotide profile hidden Markov model for coi5p.</h2><span id='topic+nt_coi_PHMM'></span>

<h3>Description</h3>

<p>This model is stored in the coi5p package and was trained on a representative
sample of the barcode of life database (http://www.boldsystems.org/index.php).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nt_coi_PHMM
</code></pre>


<h3>Format</h3>

<p>An object of class <code>PHMM</code> of length 14.
</p>

<hr>
<h2 id='print.coi5p'>Print a summary of a coi5p object.</h2><span id='topic+print.coi5p'></span>

<h3>Description</h3>

<p>Print a summary of a coi5p object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'coi5p'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.coi5p_+3A_x">x</code></td>
<td>
<p>A coi5p class object.</p>
</td></tr>
<tr><td><code id="print.coi5p_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed between methods.</p>
</td></tr>
</table>

<hr>
<h2 id='set_frame'>Take an input sequence and get it into the reading frame.</h2><span id='topic+set_frame'></span>

<h3>Description</h3>

<p>Uses the path of the ntPHMM to locate the first contiguous
set of 5 matching base pairs (5 sequential 1s) for both the
front and the back of the sequence. Sequence information outside of this
first set of matches is trimmed (low probability of being true barcode sequence).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_frame(org_seq, path_out)
</code></pre>

<hr>
<h2 id='subsetPHMM'>Subset an existing PHMM.</h2><span id='topic+subsetPHMM'></span>

<h3>Description</h3>

<p>The <code>subsetPHMM</code> function allows an existing PHMM to be subset
by profile position. This eliminates the need for the training of an additional, smaller
model if query sequences should be compared to only a subsection of an existing PHMM.
The nt_coi_PHMM and aa_coi_PHMM can therefore be subset using this function to constrain
coil's framing and error evaluation to a subset of the COI-5P region
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetPHMM(x, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetPHMM_+3A_x">x</code></td>
<td>
<p>an object of class <code>"PHMM"</code> to be subset.</p>
</td></tr>
<tr><td><code id="subsetPHMM_+3A_start">start</code></td>
<td>
<p>The first PHMM position to be inclused in the output PHMM.</p>
</td></tr>
<tr><td><code id="subsetPHMM_+3A_end">end</code></td>
<td>
<p>The last PHMM position to be included in the output PHMM.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>"PHMM"</code>
</p>


<h3>See Also</h3>

<p><code><a href="aphid.html#topic+derivePHMM">derivePHMM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## subset positions 2-100 of the COI-5P PHMM
short_nt_PHMM &lt;- subsetPHMM(nt_coi_PHMM, 2, 100)
</code></pre>

<hr>
<h2 id='trans_df'>Data frame containing the translation table recommendation.</h2><span id='topic+trans_df'></span>

<h3>Description</h3>

<p>Data frame containing the translation table recommendation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trans_df
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4609 rows and 2 columns.
</p>

<hr>
<h2 id='translate'>Translate a coi5p sequence.</h2><span id='topic+translate'></span><span id='topic+translate.coi5p'></span>

<h3>Description</h3>

<p>Translate a coi5p sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translate(x, ...)

## S3 method for class 'coi5p'
translate(x, ..., trans_table = 0, frame_offset = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translate_+3A_x">x</code></td>
<td>
<p>A coi5p class object for which frame() has been run.</p>
</td></tr>
<tr><td><code id="translate_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed between methods.</p>
</td></tr>
<tr><td><code id="translate_+3A_trans_table">trans_table</code></td>
<td>
<p>The translation table to use for translating from nucleotides to amino acids.
Default is 0, which indicates that censored translation should be performed. If the taxonomy
of the sample is known, use the function which_trans_table() to determine the translation table to use.</p>
</td></tr>
<tr><td><code id="translate_+3A_frame_offset">frame_offset</code></td>
<td>
<p>The offset to the reading frame to be applied for translation. By default the offset
is zero, so the first character in the framed sequence is considered the first nucleotide of the first codon.
Passing frame_offset = 1 would offset the sequence by one and therefore make the second character in the
framed sequence the first nucleotide of the first codon.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The translate function allows for the translation of framed sequences from nucleotides to amino acids, both
in instances when the correct genetic code corresponding to a sequence is known, and in instances when taxonomic
information is unavailable or unreliable.
</p>


<h3>Value</h3>

<p>An object of class <code>"coi5p"</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coi5p">coi5p</a></code>
</p>
<p><code><a href="#topic+frame">frame</a></code>
</p>
<p><code><a href="#topic+which_trans_table">which_trans_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#previously run functions:
dat = coi5p(example_nt_string )
dat = frame(dat)
#translate when the translation table is not known:
dat = translate(dat)
#translate when the translation table is known:
dat = translate(dat, trans_table = 5)
#additional components in output coi5p object:
dat$aaSeq
</code></pre>

<hr>
<h2 id='translate_codon'>Censored Translation of a codon.</h2><span id='topic+translate_codon'></span>

<h3>Description</h3>

<p>Translate a codon of DNA sequence using the censored translation table.
this translates codons for which the amino acids is unambiguous across
mitochondrial genetic codes across the animal kingdom and does not
translate those for which the amino acid varies,
but rather outputs a ? in the string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translate_codon(codon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translate_codon_+3A_codon">codon</code></td>
<td>
<p>a three letter DNA string.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Censored translation table:
FFLLSSSSYY?*CCWWLLLLPPPPHHQQRRRRII?MTTTTNN?KSS??VVVVAAAADDEEGGGG
Base1  = TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG
Base2  = TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG
Base3  = TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG
</p>

<hr>
<h2 id='validate_coi5p'>Validate the new coi5p class instance.</h2><span id='topic+validate_coi5p'></span>

<h3>Description</h3>

<p>Validate the new coi5p class instance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_coi5p(new_instance)
</code></pre>

<hr>
<h2 id='which_trans_table'>Determine the translation table to use for a given taxonomic group.</h2><span id='topic+which_trans_table'></span>

<h3>Description</h3>

<p>Recommends which translation table to use if taxonomic data are available.
The recommendations are based on the translation tables reported for different
taxonomic classifications on the Barcode of Life Data Systems
(BOLD - http://www.boldsystems.org/index.php).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>which_trans_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="which_trans_table_+3A_x">x</code></td>
<td>
<p>A taxonomic designation (allowed ranks: family, order, class, phylum).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If which_trans_table is unable to identify a translation table,
more information on translation tables can be found here: https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi
</p>


<h3>Value</h3>

<p>An integer indicating the correct translation table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>which_trans_table("Chordata") #phylum
which_trans_table("Actinopterygii") #class
which_trans_table("Akentrogonida")  #order
which_trans_table("Hydrobiidae") #family
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
