<!DOCTYPE html><html><head><title>Help for package colocPropTest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {colocPropTest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#colocPropTest-package'><p>The colocPropTest package</p></a></li>
<li><a href='#adjust_LD'><p>adjust LD for variable sample size</p></a></li>
<li><a href='#estprop'><p>Proportional colocalisation testing supplying only a pair of regression</p>
coefficients.</a></li>
<li><a href='#estprop_slow'><p>Proportional colocalisation testing</p></a></li>
<li><a href='#keep_from_S'><p>keep snp subset of coloc dataset</p></a></li>
<li><a href='#lp'><p>logp</p></a></li>
<li><a href='#marg_with_V'><p>create variance-covariance matrix for pair of marginal beta + vbeta, given</p>
estimate of r between snps</a></li>
<li><a href='#nform'><p>Helper function to adjust LD parameter r for differential sample size between snps</p></a></li>
<li><a href='#plot_ellipses'><p>draw two ellipses</p></a></li>
<li><a href='#run_proptests'><p>run proportional tests on extreme subset of snp pairs from two</p>
coloc style datasets. Of all functions in this package, this is
the main one that should be used.</a></li>
<li><a href='#tag'><p>Derive tag SNPs using heirarchical clustering</p></a></li>
<li><a href='#tester_marg'><p>run proportional test directly on marginal test stats from coloc datasets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Proportional Testing for Colocalisation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Colocalisation analysis tests whether two traits share a causal
  genetic variant in a specified genomic region. Proportional testing for
  colocalisation has been previously proposed
  [Wallace (2013) &lt;<a href="https://doi.org/10.1002%2Fgepi.21765">doi:10.1002/gepi.21765</a>&gt;], but is reimplemented here to
  overcome barriers to its adoption. Its use is complementary to the fine-
  mapping based colocalisation method in the 'coloc' package, and may be used in
  particular to identify false "H3" conclusions in 'coloc'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>magrittr, data.table</td>
</tr>
<tr>
<td>Imports:</td>
<td>car, coloc, graphics, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, plotrix, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-10 06:58:47 UTC; cw029707</td>
</tr>
<tr>
<td>Author:</td>
<td>Chris Wallace <a href="https://orcid.org/0000-0001-9755-1703"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chris Wallace &lt;chris.x.wallace@gsk.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-11 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='colocPropTest-package'>The colocPropTest package</h2><span id='topic+colocPropTest-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>Conducts proportional colocalisation tests. See the main function
run_proptests().
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Chris Wallace <a href="mailto:chris.x.wallace@gsk.com">chris.x.wallace@gsk.com</a> (<a href="https://orcid.org/0000-0001-9755-1703">ORCID</a>)
</p>

<hr>
<h2 id='adjust_LD'>adjust LD for variable sample size</h2><span id='topic+adjust_LD'></span>

<h3>Description</h3>

<p>adjust LD for variable sample size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_LD(S, LD)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjust_LD_+3A_s">S</code></td>
<td>
<p>coloc style dataset, with additional entries n0  and n1  which are *vectors* giving the number of cases and controls genotyped at each SNP</p>
</td></tr>
<tr><td><code id="adjust_LD_+3A_ld">LD</code></td>
<td>
<p>matrix of LD, with dimnames given by snps in S$snp</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjusted LD matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(coloc)

data(coloc_test_data)
attach(coloc_test_data)
LD=D1$LD
dimnames(LD)=list(D1$snp,D1$snp)
D1$type="cc"
D1$s=.5
D1$n1=D1$N * sample(c(0.25,.5),length(D1$snp), replace=TRUE)
D1$n0=rep(0.5*D1$N,length(D1$snp))
aLD=colocPropTest::adjust_LD(D1,LD)
LD[1:6,1:6]
aLD[1:6,1:6]
detach(coloc_test_data)
</code></pre>

<hr>
<h2 id='estprop'>Proportional colocalisation testing supplying only a pair of regression
coefficients.</h2><span id='topic+estprop'></span>

<h3>Description</h3>

<p>Proportional colocalisation testing supplying only a pair of regression
coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estprop(b1, b2, V1, V2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estprop_+3A_b1">b1</code></td>
<td>
<p>regression coefficients for trait 1, expect length(b1)=2</p>
</td></tr>
<tr><td><code id="estprop_+3A_b2">b2</code></td>
<td>
<p>regression coefficients for trait 2, expect length(b2)=2</p>
</td></tr>
<tr><td><code id="estprop_+3A_v1">V1</code></td>
<td>
<p>2x2 variance-covariance matrix for trait 1</p>
</td></tr>
<tr><td><code id="estprop_+3A_v2">V2</code></td>
<td>
<p>2x2 variance-covariance matrix for trait 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list, containing
* result: the test statistic
* plot.data: dataset for plotting the input data
* plot.eta: dataset for plotting chisq as a function of theta or eta
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>

<hr>
<h2 id='estprop_slow'>Proportional colocalisation testing</h2><span id='topic+estprop_slow'></span>

<h3>Description</h3>

<p>This should return the same as estprop for a pair of snps, but is
slower. Left here for checking. Also accomodates more than two
snps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estprop_slow(b1, b2, V1, V2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estprop_slow_+3A_b1">b1</code></td>
<td>
<p>regression coefficients for trait 1</p>
</td></tr>
<tr><td><code id="estprop_slow_+3A_b2">b2</code></td>
<td>
<p>regression coefficients for trait 2</p>
</td></tr>
<tr><td><code id="estprop_slow_+3A_v1">V1</code></td>
<td>
<p>variance-covariance matrix for trait 1</p>
</td></tr>
<tr><td><code id="estprop_slow_+3A_v2">V2</code></td>
<td>
<p>variance-covariance matrix for trait 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list, containing the test statistic and two datasets for
plotting the input data or eta
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>

<hr>
<h2 id='keep_from_S'>keep snp subset of coloc dataset</h2><span id='topic+keep_from_S'></span>

<h3>Description</h3>

<p>keep snp subset of coloc dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keep_from_S(S, keep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keep_from_S_+3A_s">S</code></td>
<td>
<p>coloc dataset</p>
</td></tr>
<tr><td><code id="keep_from_S_+3A_keep">keep</code></td>
<td>
<p>snps to keep</p>
</td></tr>
</table>


<h3>Value</h3>

<p>subset of coloc dataset
</p>

<hr>
<h2 id='lp'>logp</h2><span id='topic+lp'></span>

<h3>Description</h3>

<p>uses logs in calculation to avoid numerical issues with very small std errors / p values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lp(beta, se)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lp_+3A_beta">beta</code></td>
<td>
<p>coefficient</p>
</td></tr>
<tr><td><code id="lp_+3A_se">se</code></td>
<td>
<p>std error of coefficient</p>
</td></tr>
</table>


<h3>Value</h3>

<p>-log10 p
</p>

<hr>
<h2 id='marg_with_V'>create variance-covariance matrix for pair of marginal beta + vbeta, given
estimate of r between snps</h2><span id='topic+marg_with_V'></span>

<h3>Description</h3>

<p>create variance-covariance matrix for pair of marginal beta + vbeta, given
estimate of r between snps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marg_with_V(beta, vbeta, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marg_with_V_+3A_beta">beta</code></td>
<td>
<p>vector of two coefficients at two snps</p>
</td></tr>
<tr><td><code id="marg_with_V_+3A_vbeta">vbeta</code></td>
<td>
<p>vector of two variance of coefficients at the same two snps</p>
</td></tr>
<tr><td><code id="marg_with_V_+3A_rho">rho</code></td>
<td>
<p>LD (r) between the two snps</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of coefficient &amp; variance-covariance matrix
</p>

<hr>
<h2 id='nform'>Helper function to adjust LD parameter r for differential sample size between snps</h2><span id='topic+nform'></span>

<h3>Description</h3>

<p>Estimate the r between effect estimates at snps which were
genotyped on different sets of cases and controls.  The adjusted r
will be nform(...) * r (where r is the population correlation
between snps).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nform(n0a, n1a, n0b, n1b, n0ab = pmin(n0a, n0b), n1ab = pmin(n1a, n1b))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nform_+3A_n0a">n0a</code></td>
<td>
<p>number of controls with data at snp a</p>
</td></tr>
<tr><td><code id="nform_+3A_n1a">n1a</code></td>
<td>
<p>number of cases with data at snp a</p>
</td></tr>
<tr><td><code id="nform_+3A_n0b">n0b</code></td>
<td>
<p>number of controls with data at snp b</p>
</td></tr>
<tr><td><code id="nform_+3A_n1b">n1b</code></td>
<td>
<p>number of cases with data at snp b</p>
</td></tr>
<tr><td><code id="nform_+3A_n0ab">n0ab</code></td>
<td>
<p>number of controls with data at both snps a and b</p>
</td></tr>
<tr><td><code id="nform_+3A_n1ab">n1ab</code></td>
<td>
<p>number of cases with data at both snps a and b</p>
</td></tr>
</table>


<h3>Value</h3>

<p>proportionality constant that depends on sample size.
</p>

<hr>
<h2 id='plot_ellipses'>draw two ellipses</h2><span id='topic+plot_ellipses'></span>

<h3>Description</h3>

<p>draw two ellipses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ellipses(
  b1,
  vb1,
  b2,
  vb2,
  legend = c("inferred", "observed"),
  include_origin = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_ellipses_+3A_b1">b1</code></td>
<td>
<p>ellipse 1 centre (2d)</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_vb1">vb1</code></td>
<td>
<p>ellipse 1 vcov matrix</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_b2">b2</code></td>
<td>
<p>ellipse 2 centre (2d)</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_vb2">vb2</code></td>
<td>
<p>ellipse 2 vcov matrix</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_legend">legend</code></td>
<td>
<p>character vector length 2 naming ellipse 1 and 2</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_include_origin">include_origin</code></td>
<td>
<p>if TRUE, ensure plot includes (0,0)</p>
</td></tr>
<tr><td><code id="plot_ellipses_+3A_...">...</code></td>
<td>
<p>arguments passed to plot()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>draw ellipses on current graphics device
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_ellipses(b1=c(5,5), vb1=diag(2),
              b2=c(2,2), vb2=matrix( c(1,0.5,0.5,1), 2, 2 ),
              legend=c("circle", "ellipse"),
              include.origin=TRUE) 
</code></pre>

<hr>
<h2 id='run_proptests'>run proportional tests on extreme subset of snp pairs from two
coloc style datasets. Of all functions in this package, this is
the main one that should be used.</h2><span id='topic+run_proptests'></span>

<h3>Description</h3>

<p>run proportional tests on extreme subset of snp pairs from two
coloc style datasets. Of all functions in this package, this is
the main one that should be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_proptests(
  S1,
  S2,
  LD,
  topsnps = "auto",
  r2.thr = 0.95,
  maxtests = 10000,
  nauto = 200,
  adjust_n = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_proptests_+3A_s1">S1</code></td>
<td>
<p>coloc dataset 1</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_s2">S2</code></td>
<td>
<p>coloc dataset 2</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_ld">LD</code></td>
<td>
<p>LD matrix - rownames, colnames capture the snps and S1$snp[j] must
be represented</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_topsnps">topsnps</code></td>
<td>
<p>list of topsnps to be considered for testing or, if &quot;auto&quot;,
will be automatically selected</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_r2.thr">r2.thr</code></td>
<td>
<p>r2 threshold for initial tagging step - includes only one of
any set of snps in mutually high LD with r2 &gt; r2.thr</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_maxtests">maxtests</code></td>
<td>
<p>maximum number of test pairs to consider. if more than
maxtests pairs available, will select a random sample.</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_nauto">nauto</code></td>
<td>
<p>number of snps to use when automatically defining topsnps. only
has an effect if topsnps==&quot;auto&quot;</p>
</td></tr>
<tr><td><code id="run_proptests_+3A_adjust_n">adjust_n</code></td>
<td>
<p>TRUE if you want to adjust for variable sample size between
snps. This is only set up for case control data at the moment (ask if you
need quantitative) and requires that you supply separately the number of
cases and controls at each snp in each dataset, as vector elements of the lists
called n1 (cases) and n0 (controls)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table containing the tests run
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(colocPropTest)
library(coloc)
data(coloc_test_data)
attach(coloc_test_data)
LD=D1$LD
dimnames(LD)=list(D1$snp,D1$snp)
results=run_proptests(D1,D2,LD=LD,topsnps=D1$snp,maxtests=100)
min(results$fdr)
</code></pre>

<hr>
<h2 id='tag'>Derive tag SNPs using heirarchical clustering</h2><span id='topic+tag'></span>

<h3>Description</h3>

<p>Uses complete linkage and the <code><a href="stats.html#topic+hclust">hclust</a></code> function to define clusters,
then cuts the tree at 1-tag.threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tag(r2, r2_threshold = 0.95, quiet = FALSE, method = "complete")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tag_+3A_r2">r2</code></td>
<td>
<p>matrix of rsquared values</p>
</td></tr>
<tr><td><code id="tag_+3A_r2_threshold">r2_threshold</code></td>
<td>
<p>only 1 of a set of snps with r2 &gt; r2_threshold will be kept</p>
</td></tr>
<tr><td><code id="tag_+3A_quiet">quiet</code></td>
<td>
<p>if FALSE (default), show progress messages</p>
</td></tr>
<tr><td><code id="tag_+3A_method">method</code></td>
<td>
<p>method used for heirarchical clustering.  See hclust for options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector, names are <code>snps</code>, values are the tag for each SNP
</p>


<h3>Author(s)</h3>

<p>Chris Wallace
</p>

<hr>
<h2 id='tester_marg'>run proportional test directly on marginal test stats from coloc datasets</h2><span id='topic+tester_marg'></span>

<h3>Description</h3>

<p>run proportional test directly on marginal test stats from coloc datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tester_marg(j, S1, S2, LD1, LD2 = LD1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tester_marg_+3A_j">j</code></td>
<td>
<p>indices of thw two snps</p>
</td></tr>
<tr><td><code id="tester_marg_+3A_s1">S1</code></td>
<td>
<p>coloc dataset 1</p>
</td></tr>
<tr><td><code id="tester_marg_+3A_s2">S2</code></td>
<td>
<p>coloc dataset 2</p>
</td></tr>
<tr><td><code id="tester_marg_+3A_ld1">LD1</code></td>
<td>
<p>LD matrix for dataset 1 - rownames, colnames capture the snps and S1$snp[j] must
be represented</p>
</td></tr>
<tr><td><code id="tester_marg_+3A_ld2">LD2</code></td>
<td>
<p>LD matrix for dataset 2 - rownames, colnames capture the snps and S2$snp[j] must
be represented. if not supplied, defaults to LD1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>result from estprop
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
