<!DOCTYPE html><html><head><title>Help for package combiroc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {combiroc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#combi'><p>Compute combinations.</p></a></li>
<li><a href='#combi_score'><p>Compute the combi score using glm(link='binomial') models and optionally classifies the samples.</p></a></li>
<li><a href='#combiroc_long'><p>Reshape combiroc data in long format.</p></a></li>
<li><a href='#combs_with'><p>Show combinations with given markers.</p></a></li>
<li><a href='#demo_data'><p>A demo dataset with proteomics markers</p></a></li>
<li><a href='#demo_unclassified_data'><p>A demo dataset with unlabeled proteomics markers</p></a></li>
<li><a href='#load_data'><p>Load CombiROC data.</p></a></li>
<li><a href='#markers_distribution'><p>Show distribution of intensity values for all the markers both singularly and all together.</p></a></li>
<li><a href='#ranked_combs'><p>Rank combinations.</p></a></li>
<li><a href='#roc_reports'><p>Train logistic regression and compute ROC.</p></a></li>
<li><a href='#seurat_to_combiroc'><p>Load CombiROC data.</p></a></li>
<li><a href='#show_markers'><p>Show the composition of combinations.</p></a></li>
<li><a href='#single_markers_statistics'><p>Provide statistics for each marker.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Selection and Ranking of Omics Biomarkers Combinations Made Easy</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-07-04</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions and a workflow to easily and powerfully calculating specificity, sensitivity and ROC curves of biomarkers combinations. Allows to rank and select multi-markers signatures as well as to find the best performing sub-signatures, now also from single-cell RNA-seq datasets. The method used was first published as a Shiny app and described in Mazzara et al. (2017) &lt;<a href="https://doi.org/10.1038%2Fsrep45477">doi:10.1038/srep45477</a>&gt; and further described in Bombaci &amp; Rossi (2019) &lt;<a href="https://doi.org/10.1007%2F978-1-4939-9164-8_16">doi:10.1007/978-1-4939-9164-8_16</a>&gt;, and widely expanded as a package as presented in the bioRxiv pre print Ferrari et al. &lt;<a href="https://doi.org/10.1101%2F2022.01.17.476603">doi:10.1101/2022.01.17.476603</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>tidyr, dplyr, ggplot2, gtools, pROC, stringr, stats, utils,
moments</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, markdown, rmarkdown, httr, Seurat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://doi.org/10.1101/2022.01.17.476603">https://doi.org/10.1101/2022.01.17.476603</a>,
<a href="https://github.com/ingmbioinfo/combiroc">https://github.com/ingmbioinfo/combiroc</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ingmbioinfo/combiroc/issues">https://github.com/ingmbioinfo/combiroc/issues</a></td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-04 10:40:35 UTC; rossiriccardo</td>
</tr>
<tr>
<td>Author:</td>
<td>Ivan Ferrari <a href="https://orcid.org/0000-0003-3746-4143"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Riccardo L. Rossi <a href="https://orcid.org/0000-0002-4964-3264"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Saveria Mazzara <a href="https://orcid.org/0000-0003-1799-2360"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Mauro Bombaci <a href="https://orcid.org/0000-0002-9887-4165"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ccp, ctb, dtc]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Riccardo L. Rossi &lt;ric.rossi@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-06 12:53:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='combi'>Compute combinations.</h2><span id='topic+combi'></span>

<h3>Description</h3>

<p>A function that computes the marker combinations and counts their corresponding positive samples for each class (once thresholds are selected).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combi(data, signalthr = 0, combithr = 1, max_length = NULL, case_class)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combi_+3A_data">data</code></td>
<td>
<p>a data.frame returned by load_data().</p>
</td></tr>
<tr><td><code id="combi_+3A_signalthr">signalthr</code></td>
<td>
<p>a numeric that specifies the value above which a marker expression is considered positive in a given sample. Since the target of the analysis is the identification of marker combinations capable to correctly classify samples, the user should choose a signalthr that:
</p>

<ul>
<li><p> Positively selects most samples belonging to the case class, which must be above signalthr.
</p>
</li>
<li><p> Negatively selects most control samples, which must be below signalthr.
</p>
</li></ul>
</td></tr>
<tr><td><code id="combi_+3A_combithr">combithr</code></td>
<td>
<p>a numeric that specifies the necessary number of positively expressed markers (&gt;= signalthr), in a given combination, to cosinder that combination positivelly expressed in a sample.</p>
</td></tr>
<tr><td><code id="combi_+3A_max_length">max_length</code></td>
<td>
<p>an integer that specifies the max combination length that is allowed</p>
</td></tr>
<tr><td><code id="combi_+3A_case_class">case_class</code></td>
<td>
<p>a character that specifies which of the two classes of the dataset is the case class</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function counts how many samples are 'positive' for each combination. A sample, to be considered positive for a given combination, must have a value higher than a given signal threshold (signalthr) for at least a given number of markers composing that combination (combithr).
</p>


<h3>Value</h3>

<p>a data.frame containing how many samples of each class are &quot;positive&quot; for each combination, sensitivity and specificity.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)


# To compute the marker combinations and count their corresponding positive samples for each class.

 combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')
 # count as positive the samples with value &gt;= 450 for at least 1 marker in the combination

## End(Not run)

</code></pre>

<hr>
<h2 id='combi_score'>Compute the combi score using glm(link='binomial') models and optionally classifies the samples.</h2><span id='topic+combi_score'></span>

<h3>Description</h3>

<p>A function that applies the previously calculated models to a dataset to compute combi scores and optionally classify the samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combi_score(
  data,
  Models,
  Metrics,
  Positive_class = 1,
  Negative_class = 0,
  deal_NA = "impute",
  classify = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combi_score_+3A_data">data</code></td>
<td>
<p>a data.frame returned by load_data().</p>
</td></tr>
<tr><td><code id="combi_score_+3A_models">Models</code></td>
<td>
<p>a list of glm() objects returned by roc_reports().</p>
</td></tr>
<tr><td><code id="combi_score_+3A_metrics">Metrics</code></td>
<td>
<p>a list of data.frame objects containing ROC metrics, returned by roc_reports().</p>
</td></tr>
<tr><td><code id="combi_score_+3A_positive_class">Positive_class</code></td>
<td>
<p>a numeric or a character that specifies the label of the samples that will be classified as positives</p>
</td></tr>
<tr><td><code id="combi_score_+3A_negative_class">Negative_class</code></td>
<td>
<p>a numeric or a character that specifies the label of the samples that will be classified as negatives</p>
</td></tr>
<tr><td><code id="combi_score_+3A_deal_na">deal_NA</code></td>
<td>
<p>a character that specifies how to treat missing values. With 'impute' NAs of each marker are substituted with the median of that given marker values. With 'remove' the whole observations containing a NA are removed'.</p>
</td></tr>
<tr><td><code id="combi_score_+3A_classify">classify</code></td>
<td>
<p>a boolean that specifies if the samples will be classified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function  can take as input datasets loaded with load_data(). They MUST contain all the markers of the combinations used to train the models.
</p>


<h3>Value</h3>

<p>a data.frame containing the combi scores (classify=F) or predicted class of each sample (classify=T), for each marker/combination in Models
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)
demo_unclassified_data # combiroc built-in unclassified demo data

combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')  # compute combinations

reports &lt;- roc_reports(data= demo_data, markers_table= combs,
                       selected_combinations= c(1,11,15),
                       single_markers=c('Marker1', 'Marker2'), case_class='A') # train logistic
                                                                               # regression models


# To fit the models an retrieve the combi scores (predicted probabilities).

score_data &lt;- combi_score(data= demo_unclassified_data, Models= reports$Models,
                             Metrics= reports$Metrics)

# To classify new samples with logistic regression models.

classified_data &lt;- combi_score(data= demo_unclassified_data,
                               Models= reports$Models,Metrics= reports$Metrics,
                               Positive_class=1, Negative_class=0, classify=TRUE)

classified_data  # show samples classified using Logistic regression models

## End(Not run)
</code></pre>

<hr>
<h2 id='combiroc_long'>Reshape combiroc data in long format.</h2><span id='topic+combiroc_long'></span>

<h3>Description</h3>

<p>A function that simply wraps dyplr::pivot_longer() to reshape data in long format without taking into account NAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combiroc_long(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combiroc_long_+3A_data">data</code></td>
<td>
<p>a data.frame returned by load_data().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns the data in long format (with 'Markers' and 'Values' columns)
</p>


<h3>Value</h3>

<p>a data.frame in long format
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

# To reshape demo_data in long format

demo_data_long &lt;- combiroc_long(data = demo_data)

## End(Not run)
</code></pre>

<hr>
<h2 id='combs_with'>Show combinations with given markers.</h2><span id='topic+combs_with'></span>

<h3>Description</h3>

<p>A function to find all the combinations containing all the markers of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combs_with(markers, markers_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combs_with_+3A_markers">markers</code></td>
<td>
<p>a character vector containing one or more markers of interest.</p>
</td></tr>
<tr><td><code id="combs_with_+3A_markers_table">markers_table</code></td>
<td>
<p>a data.frame with ranked combination, reporting: SE, SP, number of markers composing the combination and the score (returned by ranked_combs()).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector containing the numbers corresponding to the combinations containing all the selected markers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')  # compute combinations

# To show all the combinations with given markers.

combs_with(markers = c('Marker1', 'Marker2') , markers_table = combs)

## End(Not run)
</code></pre>

<hr>
<h2 id='demo_data'>A demo dataset with proteomics markers</h2><span id='topic+demo_data'></span>

<h3>Description</h3>

<p>Combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104). A dataset containing signal intensity values of a 5-marker signatures for Autoimmune Hepatitis (AIH). Samples have been clinically diagnosed as “abnormal” (class A) or &quot;normal&quot; (class B).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 170 rows and 7 variables:
</p>

<dl>
<dt>Patient.ID</dt><dd><p>the ID of samples</p>
</dd>
<dt>Class</dt><dd><p>the class of the samples: A-Healthy, B-AIH</p>
</dd>
<dt>Marker1</dt><dd><p>the signal intensity value of Marker1</p>
</dd>
<dt>Marker2</dt><dd><p>the signal intensity value of Marker2</p>
</dd>
<dt>Marker3</dt><dd><p>the signal intensity value of Marker3</p>
</dd>
<dt>Marker4</dt><dd><p>the signal intensity value of Marker4</p>
</dd>
<dt>Marker5</dt><dd><p>the signal intensity value of Marker5</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

## End(Not run)
</code></pre>

<hr>
<h2 id='demo_unclassified_data'>A demo dataset with unlabeled proteomics markers</h2><span id='topic+demo_unclassified_data'></span>

<h3>Description</h3>

<p>Combiroc built-in unclassified demo dataA dataset containing signal intensity values of a 5-marker signatures for Autoimmune Hepatitis (AIH). This dataset has been obtained by randomly picking 20 samples from demo_data without their classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_unclassified_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 7 variables:
</p>

<dl>
<dt>Patient.ID</dt><dd><p>the ID of samples</p>
</dd>
<dt>Marker1</dt><dd><p>the signal intensity value of Marker1</p>
</dd>
<dt>Marker2</dt><dd><p>the signal intensity value of Marker2</p>
</dd>
<dt>Marker3</dt><dd><p>the signal intensity value of Marker3</p>
</dd>
<dt>Marker4</dt><dd><p>the signal intensity value of Marker4</p>
</dd>
<dt>Marker5</dt><dd><p>the signal intensity value of Marker5</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_unclassified_data # combiroc built-in unclassified demo data

## End(Not run)
</code></pre>

<hr>
<h2 id='load_data'>Load CombiROC data.</h2><span id='topic+load_data'></span>

<h3>Description</h3>

<p>A customized read.table() function that checks the conformity of the dataset format, and only if all checks are passed, loads it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_data(data, sep = ";", na.strings = "", labelled_data = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_data_+3A_data">data</code></td>
<td>
<p>the name of the file which the data are to be read from.</p>
</td></tr>
<tr><td><code id="load_data_+3A_sep">sep</code></td>
<td>
<p>the field separator character.</p>
</td></tr>
<tr><td><code id="load_data_+3A_na.strings">na.strings</code></td>
<td>
<p>a character vector of strings which are to be interpreted as NA values.</p>
</td></tr>
<tr><td><code id="load_data_+3A_labelled_data">labelled_data</code></td>
<td>
<p>a boolean that specifies whether the combiroc data to be loaded is labelled (with 'Class' column) or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dataset to be analysed should be in text format, which can be comma, tab or semicolon separated:
</p>

<ul>
<li><p> The 1st column must contain patient/sample IDs as characters.
</p>
</li>
<li><p> If dataset is labelled, the 2nd column must contain the class to which each sample belongs.
</p>
</li>
<li><p> The classes must be exactly 2 and they must be written in character format.
</p>
</li>
<li><p> From the 3rd column on (2nd if dataset is unlabelled), the dataset must contain numerical values that represent the signal corresponding to the markers abundance in each sample (marker-related columns).
</p>
</li>
<li><p> Marker-related columns can be called 'Marker1, Marker2, Marker3, ...' or can be called directly with the gene/protein name, but &quot;-&quot; is not allowed in the column name.
Only if all the checks are passed, it reorders alphabetically the marker-related columns depending on marker names (necessary for a proper computation of combinations), and it forces &quot;Class&quot; as 2nd column name.
</p>
</li></ul>



<h3>Value</h3>

<p>a data frame (data.frame) containing a representation of the data in the file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

# save a data.frame as a csv to be load by combiroc package
file&lt;- tempfile()
write.csv2(demo_data, file = file, row.names = FALSE)


#To load a csv file if correctly formatted

demo_data &lt;- load_data(data = file, sep = ';', na.strings = "")


demo_unclassified_data # combiroc built-in unclassified demo data

# save a data.frame as a csv to be load by combiroc package
file&lt;- tempfile()
write.csv2(demo_unclassified_data, file = file, row.names = FALSE)

# To load an unclassified dataset.

demo_unclassified_data &lt;- load_data(data = file, sep = ';', na.strings = "", labelled_data = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='markers_distribution'>Show distribution of intensity values for all the markers both singularly and all together.</h2><span id='topic+markers_distribution'></span>

<h3>Description</h3>

<p>A function that takes as input data in long format, and shows how the signal intensity value of markers are distributed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markers_distribution(
  data_long,
  min_SE = 0,
  min_SP = 0,
  x_lim = NULL,
  y_lim = NULL,
  boxplot_lim = NULL,
  signalthr_prediction = FALSE,
  case_class
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="markers_distribution_+3A_data_long">data_long</code></td>
<td>
<p>a data.frame in long format returned by combiroc_long()</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_min_se">min_SE</code></td>
<td>
<p>a numeric that specifies the min value of SE that a threshold must have to be included in $Coord.</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_min_sp">min_SP</code></td>
<td>
<p>a numeric that specifies the min value of SP that a threshold must have to be included in $Coord.</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_x_lim">x_lim</code></td>
<td>
<p>a numeric setting the max values of x that will be visualized in the density plot (zoom only, no data loss).</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_y_lim">y_lim</code></td>
<td>
<p>a numeric setting the max values of y that will be visualized in the density plot (zoom only, no data loss).</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_boxplot_lim">boxplot_lim</code></td>
<td>
<p>a numeric setting the max values of y that will be visualized in the boxplot (zoom only, no data loss).</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_signalthr_prediction">signalthr_prediction</code></td>
<td>
<p>a boolean that specifies if the density plot will also show the &quot;suggested signal threshold&quot;.</p>
</td></tr>
<tr><td><code id="markers_distribution_+3A_case_class">case_class</code></td>
<td>
<p>a character that specifies which of the two classes of the dataset is the case class.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a named list containing the following objects:
</p>

<ul>
<li><p> “Density_plot”: a density plot showing the distribution of the signal intensity values for both the classes.
</p>
</li>
<li><p> &quot;Density_summary&quot;: a data.frame showing a summary statistics of the distributions.
</p>
</li>
<li><p> “ROC”: a ROC curve showing how many real positive samples would be found positive (SE) and how many real negative samples would be found negative (SP) in function of signal threshold. NB: these SE and SP are refereed to the signal intensity threshold considering all the markers together; it is NOT equal to the SE/SP of a single marker/combination found with se_sp().
</p>
</li>
<li><p> “Coord”: a data.frame that contains the coordinates of the above described “ROC” (threshold, SP and SE) that have at least a min SE (40 by default) and a min SP (80 by default).
</p>
</li>
<li><p> &quot;Boxplot&quot;: a boxplot showing the distribution of the signal intensity values of each marker singularly, for both the classes.
</p>
</li></ul>

<p>In case of lack of a priori known threshold the user can set set signalthr_prediction= TRUE.
In this way the function provides a &quot;suggested signal threshold&quot; that corresponds to the median of the singnal threshold values (in &quot;Coord&quot;) at which SE/SP are grater or equal to their set minimal values (min_SE and min_SP),
and it adds this threshold on the &quot;Density_plot&quot; object as a dashed black line.
The use of the median allows to pick a threshold whose SE/SP are not too close to the limits (min_SE and min_SP), but it is recommended to always inspect &quot;Coord&quot; and choose the most appropriate signal threshold by considering SP, SE and Youden index.
</p>


<h3>Value</h3>

<p>a named list containing 'Coord' and 'Density_summary' data.frames, and 'ROC', 'Boxplot' and 'Density_plot' plot objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

demo_data_long &lt;- combiroc_long(data = demo_data) # long format data




# To visualize the distribution of the expression of each marker.

distributions &lt;- markers_distribution(data_long = demo_data_long,
                                       boxplot_lim = 1500, y_lim = 0.001,
                                       x_lim = 3000 , signalthr_prediction = FALSE,
                                       case_class = 'A', min_SE = 40, min_SP = 80)

distributions$Density_plot # density plot
distributions$Density_summary # summary statistics of density plot
distributions$ROC # ROC showing signal threshold range ensuring min SE and/or SP
distributions$Coord # ROC values
distributions$Boxplot # Boxplot

## End(Not run)
</code></pre>

<hr>
<h2 id='ranked_combs'>Rank combinations.</h2><span id='topic+ranked_combs'></span>

<h3>Description</h3>

<p>A function to rank combinations by a Youden index and select them if they have a min SE and/or SP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ranked_combs(combo_table, min_SE = 0, min_SP = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ranked_combs_+3A_combo_table">combo_table</code></td>
<td>
<p>a data.frame with SE, SP and number of composing markers for each combination (returned by se_sp()).</p>
</td></tr>
<tr><td><code id="ranked_combs_+3A_min_se">min_SE</code></td>
<td>
<p>a numeric that specifies the min value of SE that a combination must have to be filtered-in.</p>
</td></tr>
<tr><td><code id="ranked_combs_+3A_min_sp">min_SP</code></td>
<td>
<p>a numeric that specifies the min value of SP that a combination must have to be filtered-in.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is meant to help the user in finding the best combinations (in the first rows) and allows also (not mandatory) the SE/SP-dependent filtering of combinations.
</p>


<h3>Value</h3>

<p>a named list containing:
</p>

<ul>
<li><p> $table, a data.frame with ranked combination, reporting: SE, SP, number of markers composing the combination and the score.
</p>
</li>
<li><p> $bubble_chart, a dot plot showing the selected 'gold' combinations
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')  # compute combinations
                                                                  # compute SE and SP
                                                               # of each combination



# To rank combinations by Youden index and filter-out the ones with SE &lt; min_SE and SP &lt; min_SP

rc &lt;- ranked_combs(combo_table= combs, min_SE=40, min_SP=80)
rc$table # to visualize the selected gold combinations through a data.frame
rc$bubble_chart # to visualize the selected gold combinations through a data.frame

## End(Not run)
</code></pre>

<hr>
<h2 id='roc_reports'>Train logistic regression and compute ROC.</h2><span id='topic+roc_reports'></span>

<h3>Description</h3>

<p>A function to compute General Linear Model (binomial) and the corresponding ROC curves for each selected combination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roc_reports(
  data,
  markers_table,
  selected_combinations = NULL,
  single_markers = NULL,
  case_class,
  deal_NA = "impute"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roc_reports_+3A_data">data</code></td>
<td>
<p>a data.frame returned by load_data().</p>
</td></tr>
<tr><td><code id="roc_reports_+3A_markers_table">markers_table</code></td>
<td>
<p>a data.frame with combinations and corresponding positive samples counts, obtained with combi().</p>
</td></tr>
<tr><td><code id="roc_reports_+3A_selected_combinations">selected_combinations</code></td>
<td>
<p>a numeric vector that specifies the combinations of interest.</p>
</td></tr>
<tr><td><code id="roc_reports_+3A_single_markers">single_markers</code></td>
<td>
<p>a character vector that specifies the single markers of interest.</p>
</td></tr>
<tr><td><code id="roc_reports_+3A_case_class">case_class</code></td>
<td>
<p>a character that specifies which of the two classes of the dataset is the case class.</p>
</td></tr>
<tr><td><code id="roc_reports_+3A_deal_na">deal_NA</code></td>
<td>
<p>a character that specifies how to treat missing values. With 'impute' NAs of each marker are substituted with the median of that given marker values in the class that observation belongs to. With 'remove' the whole observations containing a NA are removed'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function trains a logistic regression model for each combination and returns a named list containing 3 objects:
</p>

<ul>
<li><p> &quot;Plot&quot;: a ggplot object with the ROC curves of the selected combinations.
</p>
</li>
<li><p> &quot;Metrics&quot;: a data.frame with the metrics of the roc curves (AUC, opt. cutoff, etc ...).
</p>
</li>
<li><p> &quot;Models&quot;: the list of models (glm() objects) that have been computed and then used to classify the samples (in which you can find the model equation for each selected combination).
</p>
</li></ul>



<h3>Value</h3>

<p>a named list containing 3 objects: &quot;Plot&quot;, &quot;Metrics&quot; and &quot;Models&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')  # compute combinations


# To train logistic regression models on each selected combinations and
# each selected marker, and compute corresponding ROCs.

reports &lt;- roc_reports(data= demo_data, markers_table= combs,
                        selected_combinations= c(1,11,15),
                        single_markers=c('Marker1', 'Marker2'), case_class='A')

reports$Plot  # Shows the ROC curves
reports$Metrics # Shows the ROC metrics
reports$Models # show models
reports$reports$Models$`Combination 11` # show model trained with Combination 11

## End(Not run)
</code></pre>

<hr>
<h2 id='seurat_to_combiroc'>Load CombiROC data.</h2><span id='topic+seurat_to_combiroc'></span>

<h3>Description</h3>

<p>A function to extract a combiroc data (both labelled and unlabelled) from a SeuratObject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seurat_to_combiroc(
  SeuratObject,
  gene_list,
  assay = "RNA",
  labelled_data = F,
  case_class = NA,
  case_label = "case",
  control_label = "control"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seurat_to_combiroc_+3A_seuratobject">SeuratObject</code></td>
<td>
<p>Defines S4 classes for single-cell genomic data and associated information.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_gene_list">gene_list</code></td>
<td>
<p>a list of gene names.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_assay">assay</code></td>
<td>
<p>a character that specifies the assay of interest.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_labelled_data">labelled_data</code></td>
<td>
<p>a boolean that specifies whether the combiroc data to be extracted must be labelled (with 'Class' column) or not.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_case_class">case_class</code></td>
<td>
<p>a character or a character vector specifying the category/ies to be considered as 'case class'. Required if labelled_data is TRUE.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_case_label">case_label</code></td>
<td>
<p>a character that will be assigned to the cells belonging to 'case class' category.</p>
</td></tr>
<tr><td><code id="seurat_to_combiroc_+3A_control_label">control_label</code></td>
<td>
<p>a character that will be assigned to the cells belonging to 'control class' category.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By specifying a gene list (if the genes are in rownames of SeuratObject assay matrix, it subsets the gene expression matrix (@data) and it retreives their expression values. If a combiroc training dataset (labelled_data=T) is required, one or more categories of Idents(SeuratObject) must be selected as case_class, the others will be merged into control class.
</p>


<h3>Value</h3>

<p>a combiroc data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_seurat# A subset of PBMC3K dataset from Satijia et al. 2015

# list of markers of interest
gene_list&lt;-c('RBP7','CYP1B1','CD14','FCN1','NKG7', 'GNLY')

# to extract the combircoc data from data from pbmc3k.final (case class: Monocytes)
data &lt;- seurat_to_combiroc(demo_seurat, gene_list = gene_list, labelled_data = TRUE,
                           case_class = c('CD14+ Mono','FCGR3A+ Mono'),
                           case_label = 'Monocyte', control_label='Other')
head(data)

## End(Not run)
</code></pre>

<hr>
<h2 id='show_markers'>Show the composition of combinations.</h2><span id='topic+show_markers'></span>

<h3>Description</h3>

<p>A function to show the composition of combinations of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_markers(markers_table, selected_combinations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_markers_+3A_markers_table">markers_table</code></td>
<td>
<p>a data.frame with combinations returned by combi().</p>
</td></tr>
<tr><td><code id="show_markers_+3A_selected_combinations">selected_combinations</code></td>
<td>
<p>a numeric vector that specifies the combinations of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame containing the selected combinations and their composing markers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

combs &lt;- combi(data= demo_data, signalthr=450, combithr=1, case_class='A')  # compute combinations

#  To show the composition of combinations of interest.

show_markers(markers_table = combs, selected_combinations = c(1,11))

## End(Not run)
</code></pre>

<hr>
<h2 id='single_markers_statistics'>Provide statistics for each marker.</h2><span id='topic+single_markers_statistics'></span>

<h3>Description</h3>

<p>A function that computes the statistics and a scatter-plot for each marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_markers_statistics(data_long)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single_markers_statistics_+3A_data_long">data_long</code></td>
<td>
<p>a data.frame in long format returned by combiroc_long().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the main statistics of the signal values distribution of each marker in both classes. In addition it also shows the values through scatter plots.
</p>


<h3>Value</h3>

<p>a list object containing:
</p>

<ul>
<li><p> 'Statistics': a dataframe containing the main statistics for each marker in each class.
</p>
</li>
<li><p> 'Plots': a named list of scatter plots showing signal intensity values.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
demo_data # combiroc built-in demo data (proteomics data from Zingaretti et al. 2012 - PMC3518104)

data_long &lt;- combiroc_long(demo_data) # reshape data in long format

sms &lt;- single_markers_statistics(data_long)

sms$Statistics # to visualize the statistics of each single marker
sms$Plots[[1]] # to visualize the scatterplot of the first marker

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
