<!DOCTYPE html><html><head><title>Help for package COMBAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {COMBAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#COMBAT'>
<p>A Combined Gene-based Association Test</p></a></li>
<li><a href='#ext_simes'>
<p>Extended Simes Procedure</p></a></li>
<li><a href='#gates'>
<p>Calling Gene-based Association Tests</p></a></li>
<li><a href='#ld.Rsquare'>
<p>Linkage Disequilibrium Estimate as Correlation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Combined Association Test for Genes using Summary Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-01-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Minghui Wang, Yiyuan Liu, Shizhong Han</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Minghui Wang &lt;m.h.wang@live.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genome-wide association studies (GWAS) have been widely used for identifying common variants associated with complex diseases. Due to the small effect sizes of common variants, the power to detect individual risk variants is generally low. Complementary to SNP-level analysis, a variety of gene-based association tests have been proposed. However, the power of existing gene-based tests is often dependent on the underlying genetic models, and it is not known a priori which test is optimal.  Here we proposed COMBined Association Test (COMBAT) to incorporate strengths from multiple existing gene-based tests, including VEGAS, GATES and simpleM. Compared to individual tests, COMBAT shows higher overall performance and robustness across a wide range of genetic models. The algorithm behind this method is described in Wang et al (2017) &lt;<a href="https://doi.org/10.1534%2Fgenetics.117.300257">doi:10.1534/genetics.117.300257</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>mvtnorm, corpcor, R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>parallel</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-14 21:31:14 UTC; Minghui</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-14 21:54:00 UTC</td>
</tr>
</table>
<hr>
<h2 id='COMBAT'>
A Combined Gene-based Association Test
</h2><span id='topic+COMBAT'></span>

<h3>Description</h3>

<p>This function implements a combined gene-based association test using SNP-level P values and reference genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COMBAT(x, snp.ref, vegas.pct = c(0.1,0.2,0.3,0.4,1), pca_cut_perc = 0.995,
   nperm = 100, seed=12345, ncores=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="COMBAT_+3A_x">x</code></td>
<td>
<p>a vector of SNP-level P values.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_snp.ref">snp.ref</code></td>
<td>
<p>a matrix of SNP genotypes (coded as allele counts) from reference samples, with samples in rows and SNPs in columns.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_vegas.pct">vegas.pct</code></td>
<td>
<p>a numeric vector, fraction of the top SNPs to be used in the VEGAS method.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_pca_cut_perc">pca_cut_perc</code></td>
<td>
<p>numeric, cutoff for percentage of sum of eigen values in the simpleM approach.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_nperm">nperm</code></td>
<td>
<p>number of permutations for computing the correlation between P values of different tests.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_seed">seed</code></td>
<td>
<p>random seed to derive consistent outcome.</p>
</td></tr>
<tr><td><code id="COMBAT_+3A_ncores">ncores</code></td>
<td>
<p>number of CPU cores for parallel computing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>COMBAT uses simulation and the extended Simes procedure (<code><a href="#topic+ext_simes">ext_simes</a></code>) to combine multiple gene-based association test statistics (currently including <code><a href="#topic+gates">gates</a></code>, <code><a href="#topic+vegas">vegas</a></code>, and <code><a href="#topic+simpleM">simpleM</a></code>) to perform a more powerful association analysis. This method does not require raw genotype or phenotype data, but needs only SNP-level P-values and correlations between SNPs from ancestry-matched samples. The technical details about the method is described in Wang et al (2017) &lt;doi:10.1534/genetics.117.300257&gt;.
</p>


<h3>Value</h3>

<p>A vector of p-values from COMAT and each individual gene-based test.
</p>


<h3>References</h3>

<p>Minghui Wang, Jianfei Huang, Yiyuan Liu, Li Ma, James B. Potash, Shizhong Han. COMBAT: A Combined Association Test for Genes using Summary Statistics. Genetics 2017, 207(3): 883-891. https://doi.org/10.1534/genetics.117.300257.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ext_simes">ext_simes</a></code>, <code><a href="#topic+gates">gates</a></code>, <code><a href="#topic+vegas">vegas</a></code>, <code><a href="#topic+simpleM">simpleM</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read SNP P values
file1 &lt;- paste(path.package("COMBAT"),"extdata","SNP_info.txt.gz",sep="/")
snp.info  &lt;- read.table(file1, header = TRUE, as.is=TRUE)
snp.pvals &lt;- as.matrix(snp.info[,2])

# read reference genotype
file2 &lt;- paste(path.package("COMBAT"),"extdata","SNP_ref.txt.gz",sep="/")
snp.ref   &lt;- read.table(file2, header = TRUE)
snp.ref   &lt;- as.matrix(snp.ref)
#call COMBAT

COMBAT(snp.pvals, snp.ref, nperm=100, ncores=2)


</code></pre>

<hr>
<h2 id='ext_simes'>
Extended Simes Procedure
</h2><span id='topic+ext_simes'></span>

<h3>Description</h3>

<p>Combine a vector of test P values by correction for number of independent tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ext_simes(x, cor_r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ext_simes_+3A_x">x</code></td>
<td>
<p>a vector of SNP-level P values.</p>
</td></tr>
<tr><td><code id="ext_simes_+3A_cor_r">cor_r</code></td>
<td>
<p>correlation among P values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+COMBAT">COMBAT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see ?COMBAT
</code></pre>

<hr>
<h2 id='gates'>
Calling Gene-based Association Tests
</h2><span id='topic+gates'></span><span id='topic+vegas'></span><span id='topic+simpleM'></span>

<h3>Description</h3>

<p>Several gene-based association tests methods are implemented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gates(x, cor_G)
vegas(x, cor_G, vegas.pct=c(0.1,0.2,0.3,0.4,1), max.simulation=1e6)
simpleM(x, cor_G, pca_cut_perc=0.995)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gates_+3A_x">x</code></td>
<td>
<p>a vector of SNP-level P values.</p>
</td></tr>
<tr><td><code id="gates_+3A_cor_g">cor_G</code></td>
<td>
<p>SNP-SNP correlation matrix.</p>
</td></tr>
<tr><td><code id="gates_+3A_vegas.pct">vegas.pct</code></td>
<td>
<p>a numeric vector, specifying the fraction of the top SNPs to be used in the VEGAS method.</p>
</td></tr>
<tr><td><code id="gates_+3A_max.simulation">max.simulation</code></td>
<td>
<p>maximum number of simulations to be performed. Must be at least 1e6.</p>
</td></tr>
<tr><td><code id="gates_+3A_pca_cut_perc">pca_cut_perc</code></td>
<td>
<p>cutoff for percentage of sum of eigen values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>gates</code> implements the GATES method (Li et al 2011, American Journal of Human Genetics 88:283-293), <code>vegas</code> implements VEGAS with different proportion tests (Liu et al 2010, American Journal of Human Genetics 87:139-145), and <code>simpleM</code> is the simpleM method (Gao et al 2008, Genetic Epidemiology 32:361-369).
</p>


<h3>Value</h3>

<p>P value(s).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+COMBAT">COMBAT</a></code>, <code><a href="#topic+ld.Rsquare">ld.Rsquare</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read SNP P values
file1 &lt;- paste(path.package("COMBAT"),"extdata","SNP_info.txt.gz",sep="/")
snp.info  &lt;- read.table(file1, header = TRUE, as.is=TRUE)
snp.pvals &lt;- as.matrix(snp.info[,2])

# read reference genotype
file2 &lt;- paste(path.package("COMBAT"),"extdata","SNP_ref.txt.gz",sep="/")
snp.ref   &lt;- read.table(file2, header = TRUE)
snp.ref   &lt;- as.matrix(snp.ref)

#compute correlation among SNPs
cor_G &lt;- ld.Rsquare(snp.ref)

#call gates
(pval_gates &lt;- gates(x=snp.pvals, cor_G=cor_G))

#call vegas
(pval_vegas &lt;- vegas(x=snp.pvals, cor_G=cor_G))

#call simpleM
(pval_simpleM &lt;- simpleM(x=snp.pvals, cor_G=cor_G))
</code></pre>

<hr>
<h2 id='ld.Rsquare'>
Linkage Disequilibrium Estimate as Correlation
</h2><span id='topic+ld.Rsquare'></span>

<h3>Description</h3>

<p>Compute linkage disequilibrium among SNPs using correlation coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld.Rsquare(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld.Rsquare_+3A_x">x</code></td>
<td>
<p>a matrix of SNP genotypes with samples in the rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A positive definite correlation matrix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gates">gates</a></code>, <code><a href="#topic+vegas">vegas</a></code>, <code><a href="#topic+simpleM">simpleM</a></code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
