<!DOCTYPE html><html lang="en"><head><title>Help for package ComICS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ComICS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ComICS'><p>Computational methods for Immune Cell-type Subsets</p></a></li>
<li><a href='#ComICS-internal'><p>ComICS-internal</p></a></li>
<li><a href='#commons'>
<p>Shared Immunological datasets</p></a></li>
<li><a href='#dcq'><p>DCQ - Digital Cell Quantifier</p></a></li>
<li><a href='#dcqEx'>
<p>Example datasets for runnning dcq</p></a></li>
<li><a href='#vocal'><p>Variation in Cell Abundance Loci</p></a></li>
<li><a href='#vocalEx'>
<p>Example datasets for runnning vocal</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Computational Methods for Immune Cell-Type Subsets</td>
</tr>
<tr>
<td>Description:</td>
<td>Provided are Computational methods for Immune Cell-type Subsets, including:(1) DCQ (Digital Cell Quantifier) to infer global dynamic changes in immune cell quantities within a complex tissue; and (2) VoCAL (Variation of Cell-type Abundance Loci) a deconvolution-based method that utilizes transcriptome data to infer the quantities of immune-cell types, and then uses these quantitative traits to uncover the underlying DNA loci.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, stats</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://dcq.tau.ac.il/">http://dcq.tau.ac.il/</a> , <a href="http://csgi.tau.ac.il/VoCAL/">http://csgi.tau.ac.il/VoCAL/</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-13 07:59:37 UTC; IritGNB9</td>
</tr>
<tr>
<td>Author:</td>
<td>Yael Steuerman [aut, cre],
  Irit Gat-Viks [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yael Steuerman &lt;yaelsteu@mail.tau.ac.il&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-13 08:19:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='ComICS'>Computational methods for Immune Cell-type Subsets</h2><span id='topic+ComICS'></span><span id='topic+ComICS-package'></span>

<h3>Description</h3>

<p>Computational methods for Immune Cell-type Subsets.
</p>


<h3>Author(s)</h3>

<p>Yael Steuerman and Irit Gat-Viks
</p>

<hr>
<h2 id='ComICS-internal'>ComICS-internal</h2><span id='topic+ComICS-internal'></span><span id='topic+.fishersMethod'></span><span id='topic+.normalizeData'></span><span id='topic+.qtlAnalysis'></span><span id='topic+.create_iQTL_association_scores'></span><span id='topic+.combine_iQTL_association_scores'></span><span id='topic+.iterative_filteration'></span>

<h3>Description</h3>

<p>Internal ComICS functions
</p>


<h3>Details</h3>

<p>These are not to be called by the user.
</p>

<hr>
<h2 id='commons'>
Shared Immunological datasets
</h2><span id='topic+immgen_dat'></span><span id='topic+DCQ_mar'></span>

<h3>Description</h3>

<p>Example datasets (Reference data and marker set):
</p>
<p><code>immgen_dat</code>: An immune cell compendium, consisting of transcriptional profiles of isolated immune cell subsets, taken from various tissues, stimulations and time points (adapted from Heng et al., 2008). 
The full immgen dataset is available for download at <a href="http://dcq.tau.ac.il/">http://dcq.tau.ac.il/</a> or <a href="http://csgi.tau.ac.il/VoCAL/">http://csgi.tau.ac.il/VoCAL/</a> .
</p>
<p><code>DCQ_mar</code>: Preselected group of genes that likely discriminate well between the immune-cell types given in the reference data (adapted from Altboum et al., 2014).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(commons)
</code></pre>

<hr>
<h2 id='dcq'>DCQ - Digital Cell Quantifier</h2><span id='topic+dcq'></span>

<h3>Description</h3>

<p>DCQ combines genome-wide gene expression data with an immune cell-type
reference data to infer changes in the quantities immune cell subpopulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dcq(reference_data, mix_data, marker_set, alpha_used=0.05,
 lambda_min=0.2, number_of_repeats=3, precent_of_data=1.0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dcq_+3A_reference_data">reference_data</code></td>
<td>
<p>a data frame representing immune cell expression profiles.
Each row represents an expression of a gene, and each column represents a
different immune cell type. <code>colnames</code> contains the name of each immune cell
type and the <code>rownames</code> includes the genes' symbol. The names of each immune
cell type and the symbol of each gene should be unique. Any gene with
missing expression values must be excluded.</p>
</td></tr>
<tr><td><code id="dcq_+3A_mix_data">mix_data</code></td>
<td>
<p>a data frame representing RNA-seq or microarray
gene-expression profiles of a given complex tissue. Each row represents an
expression of a gene, and each column represents a different experimental sample.
<code>colnames</code> contain the name of each sample and <code>rownames</code> includes the genes' symbol.
The name of each individual sample and the symbol of each gene should be unique.
Any gene with missing expression values should be excluded.</p>
</td></tr>
<tr><td><code id="dcq_+3A_marker_set">marker_set</code></td>
<td>
<p>data frames of one column, that includes a
preselected list of genes that likely discriminate well
between the immune-cell types given in the reference data.</p>
</td></tr>
<tr><td><code id="dcq_+3A_alpha_used">alpha_used</code>, <code id="dcq_+3A_lambda_min">lambda_min</code></td>
<td>
<p>parameters of the L1 and L2 regularization. It is generally
recommended to leave the default value. For more information about this
parameter, see the glmnet package.</p>
</td></tr>
<tr><td><code id="dcq_+3A_number_of_repeats">number_of_repeats</code></td>
<td>
<p>using one repeat will generate only one output model.
Using many repeats, DCQ calculates a collection of models, and outputs the
average and standard deviation for each predicted relative cell quantity.</p>
</td></tr>
<tr><td><code id="dcq_+3A_precent_of_data">precent_of_data</code></td>
<td>
<p>in order to run the analysis over all the cell types use 1.0.
For bootstrap purposes, you can use part of the data (e.g, 0.5).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list that contains two matrices
</p>
<table role = "presentation">
<tr><td><code>average</code></td>
<td>
<p>a matrix that contains the average relative quantities for each cell
type in everytest sample.</p>
</td></tr>
<tr><td><code>stdev</code></td>
<td>
<p>a matrix that contains the standard deviations over all repeats for
each cell types in each test sample.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Altboum Z, Steuerman Y, David E, Barnett-Itzhaki Z, Valadarsky L, Keren-Shaul H, et al.
Digital cell quantification identifies global immune cell dynamics during influenza
infection. Mol Syst Biol. 2014;10: 720. doi:10.1002/msb.134947
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(commons)
data(dcqEx)
results &lt;- dcq(reference_data=immgen_dat, mix_data=lung_time_series_dat, marker_set=DCQ_mar)

</code></pre>

<hr>
<h2 id='dcqEx'>
Example datasets for runnning dcq
</h2><span id='topic+lung_time_series_dat'></span>

<h3>Description</h3>

<p>Example datasets for runnning dcq (mix data):
</p>
<p><code>lung_time_series_dat</code>: RNA-seq or microarray differential gene expression profiles of a test sample compared to a reference sample (adapted from Altboum et al., 2014).
The full dataset is available for download at <a href="http://dcq.tau.ac.il/">http://dcq.tau.ac.il/</a> or <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49934">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49934</a> .
</p>
<p>uses <code>DCQ_mar</code> and <code>immgen_dat</code> from commons.RData .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dcqEx)
</code></pre>

<hr>
<h2 id='vocal'>Variation in Cell Abundance Loci</h2><span id='topic+vocal'></span>

<h3>Description</h3>

<p>Probing immune system genetics via gene expression.
VoCAL is a deconvolution-based method that utilizes transcriptome data
to infer the quantities of immune-cell types, and then uses these quantitative
traits to uncover the underlying DNA loci (iQTLs) assuming homozygosity
(such as in the case of recombinent inbred strains).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vocal(...,reference_data,expression_data,genotyping_data,normalize_data,
 T.i=5,T.e=10,eqtl_association_scores=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vocal_+3A_...">...</code></td>
<td>
<p>one or more data frames of one column, each one represents a
preselected marker set that likely discriminate well between the immune-cell
types given in the reference data. The number of data frames defines the
number of association scores that would be combined to generate the final
iQTL association score.</p>
</td></tr>
<tr><td><code id="vocal_+3A_reference_data">reference_data</code></td>
<td>
<p>a data frame representing immune cell expression profiles.
Each row represents an expression of a gene, and each column represents a
different immune cell type. <code>colnames</code> contains the name of each immune cell
type and the <code>rownames</code> includes the genes' symbol. The names of each immune
cell type and the symbol of each gene should be unique. Any gene with
missing expression values must be excluded.</p>
</td></tr>
<tr><td><code id="vocal_+3A_expression_data">expression_data</code></td>
<td>
<p>a data frame representing RNA-seq or microarray
gene-expression profiles of a given complex tissue across a population of
genetically distinct (genotyped) individuals. Each row represents an
expression of a gene, and each column represents a genetically distinct
individual. <code>colnames</code> contain the name of each individual, as written in the
<code>genotyping_data</code>, and <code>rownames</code> includes the genes' symbol.
The name of each individual sample and the symbol of each gene should be unique.
Any gene with missing expression values should be excluded.</p>
</td></tr>
<tr><td><code id="vocal_+3A_genotyping_data">genotyping_data</code></td>
<td>
<p>a data frame where each row represents a different
locus, and each column represents a genetically distinct individual.
The genotype should be taken from homozygous individuals only.
Where the genotype is unknown <code>NA</code> should be used.
The first six columns contain the following information: (1) The sequential
identifier of the locus; (2) The name of each locus Chr; (3) Chromosome
position; (4) Start genome position; (5) End genome position;
(6) position in cM.</p>
</td></tr>
<tr><td><code id="vocal_+3A_normalize_data">normalize_data</code></td>
<td>
<p>normalization type. The data will be normalized by either:
(1) &quot;All&quot; - subtraction of the mean expression of all strains;
(2) &quot;None&quot; - data is already normalized, do nothing;
(3) name of individual included in <code>colnames</code> of <code>expression_data</code>;</p>
</td></tr>
<tr><td><code id="vocal_+3A_t.i">T.i</code></td>
<td>
<p>numerical. significant iQTL association score <code>(-log10(Pvalue))</code>
cutoff for the refinement step of the VoCAL algorithm.</p>
</td></tr>
<tr><td><code id="vocal_+3A_t.e">T.e</code></td>
<td>
<p>numerical. significant eQTL association score <code>(-log10(Pvalue))</code>
cutoff for the refinement step of the VoCAL algorithm.</p>
</td></tr>
<tr><td><code id="vocal_+3A_eqtl_association_scores">eqtl_association_scores</code></td>
<td>
<p>(optional) a data frame where each entry
represents an association score for a gene given the genotype of all the
individuals that appear in the expression_data data frame, in a specific locus.
This eQTL analysis should be peformed over the normalized expression_data.
<code>colnames</code> contain the UID (as written in the genotyping_data) and
<code>rownames</code> includes the genes' symbol (as written in the expression_data).
The symbol of each gene should be unique. These scores should be in -log10(P value).
Default is NULL, meaning that eQTL analysis will be performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two martices
</p>
<table role = "presentation">
<tr><td><code>final_association_score</code></td>
<td>
<p>a matrix that contains the output iQTL association
score after applying the iterative filteration procedure. Each row represents the genome
wide-association result for a specific immune trait over a range of DNA loci.
<code>rownames</code> provides the identifier of the locus and <code>colnames</code> contains the
immune-cell type names. Each entry provides the <code>-log10(P value)</code> of an iQTL
association score.</p>
</td></tr>
<tr><td><code>marker_info</code></td>
<td>
<p>the names of all the markers removed from the different
marker sets provided</p>
</td></tr>
</table>


<h3>References</h3>

<p>Steuerman Y and Gat-Viks I.
Exploiting Gene-Expression Deconvolution to Probe the Genetics of the Immune System (2015), Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(commons)
data(vocalEx)
## Not run: 
 results &lt;- vocal(DCQ_mar, reference_data=immgen_dat, expression_data=lung_dat,
 genotyping_data=gBXD, normalize_data="B6", eqtl_association_scores=eQTL_res)

## End(Not run)

</code></pre>

<hr>
<h2 id='vocalEx'>
Example datasets for runnning vocal
</h2><span id='topic+lung_dat'></span><span id='topic+gBXD'></span><span id='topic+eQTL_res'></span>

<h3>Description</h3>

<p>Example datasets for runnning vocal (Expression data, genotype data and eQTL results data): 
</p>
<p><code>lung_dat</code>: RNA-seq or microarray gene-expression profiles of a given complex tissue across a population of genetically distinct (genotyped) individuals (adapted from E-MTAB-848).
</p>
<p><code>gBXD</code>: Genotyping of the different individuals under study (adapted from GeneNetworks).
</p>
<p><code>eQTL_res</code>: eQTL analysis results of the different genes in the expression data (specifically the genes that appear in the marker set(s) selected).
</p>
<p>uses <code>DCQ_mar</code> and <code>immgen_dat</code> from commons.RData .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(vocalEx)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
