<!DOCTYPE html><html lang="en"><head><title>Help for package compareGroups</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {compareGroups}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compareGroups-package'>
<p>Descriptive analysis by groups</p></a></li>
<li><a href='#cGroupsGUI'>
<p>Graphical user interface based on tcltk tools</p></a></li>
<li><a href='#cGroupsWUI'>
<p>Web User Interface based on Shiny tools.</p></a></li>
<li><a href='#compareGroups'>
<p>Descriptives by groups</p></a></li>
<li><a href='#compareGroups-internal'><p>Internal compareGroups functions</p></a></li>
<li><a href='#compareSNPs'>
<p>Summarise genetic data by groups.</p></a></li>
<li><a href='#createTable'>
<p>Table of descriptives by groups: bivariate table</p></a></li>
<li><a href='#descrTable'>
<p>Perform descriptives and build the bivariate table.</p></a></li>
<li><a href='#export2csv'>
<p>Exporting descriptives table to plain text (CSV) format</p></a></li>
<li><a href='#export2html'>
<p>Exporting descriptives table to HTML format</p></a></li>
<li><a href='#export2latex'>
<p>Exporting descriptives table to LaTeX format</p></a></li>
<li><a href='#export2md'>
<p>Exporting descriptives table to Markdown format</p></a></li>
<li><a href='#export2pdf'>
<p>Exports tables to PDF files.</p></a></li>
<li><a href='#export2word'>
<p>Exports tables to Word files.</p></a></li>
<li><a href='#export2xls'>
<p>Exporting descriptives table to Exel format (.xlsx or .xls)</p></a></li>
<li><a href='#getResults'>
<p>Easily retrieve summary data as R-objects (matrices and vectors).</p></a></li>
<li><a href='#missingTable'>
<p>Table of missingness counts by groups.</p></a></li>
<li><a href='#padjustCompareGroups'><p>Update p values according multiple comparisons</p></a></li>
<li><a href='#printTable'>
<p>'Nice' table format.</p></a></li>
<li><a href='#radiograph'>
<p>Lists the values in the data set.</p></a></li>
<li><a href='#regicor'>
<p>REGICOR cross-sectional data</p></a></li>
<li><a href='#report'>
<p>Report of descriptive tables and plots.</p></a></li>
<li><a href='#SNPs'><p>SNPs in a case-control study</p></a></li>
<li><a href='#strataTable'>
<p>Stratify descriptive table in stratas.</p></a></li>
<li><a href='#varinfo'>
<p>Variable names and labels extraction</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Descriptive Analysis by Groups</td>
</tr>
<tr>
<td>Version:</td>
<td>4.9.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-29</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, tools, HardyWeinberg, rmarkdown, knitr, kableExtra,
methods, chron, stats, writexl, flextable, officer, PMCMRplus</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tcltk2, shiny, shinyBS, shinyjs, shinyjqui, shinythemes,
shinyWidgets, shinydashboardPlus, DT, readxl, haven</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Isaac Subirana &lt;isubirana@imim.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Create data summaries for quality control, extensive reports for exploring data, as well as publication-ready univariate or bivariate tables in several formats (plain text, HTML,LaTeX, PDF, Word or Excel. Create figures to quickly visualise the distribution of your data (boxplots, barplots, normality-plots, etc.). Display statistics (mean, median, frequencies, incidences, etc.). Perform the appropriate tests (t-test, Analysis of variance, Kruskal-Wallis, Fisher, log-rank, ...) depending on the nature of the described variable (normal, non-normal or qualitative). Summarize genetic data (Single Nucleotide Polymorphisms) data displaying Allele Frequencies and performing Hardy-Weinberg Equilibrium tests among other typical statistics and tests for these kind of data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://isubirana.github.io/compareGroups/index.html">https://isubirana.github.io/compareGroups/index.html</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-30 11:15:49 UTC; isubirana</td>
</tr>
<tr>
<td>Author:</td>
<td>Isaac Subirana <a href="https://orcid.org/0000-0003-1676-0197"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Joan Salvador [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-30 12:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='compareGroups-package'>
Descriptive analysis by groups
</h2><span id='topic+compareGroups-package'></span>

<h3>Description</h3>

<p>Create data summaries for quality control, extensive reports for exploring data, as well as publication-ready univariate or bivariate tables in several formats (plain text, HTML,LaTeX, PDF, Word or Excel). 
Display statistics (mean, median, frequencies, incidences, etc.).
Create figures to quickly visualise the distribution of your data (boxplots, barplots, normality-plots, etc.).
Perform the appropriate tests (t-test, Analysis of variance, Kruskal-Wallis, Fisher, log-rank, ...) depending on the nature of the described variable (normal, non-normal or qualitative). 
Summarize genetic data (Single Nucleotide Polymorphisms) data displaying Allele Frequencies and performing Hardy-Weinberg Equilibrium tests among other typical statistics and tests for these kind of data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> compareGroups</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 4.9.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-10-29</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Main functions: 
<code><a href="#topic+compareGroups">compareGroups</a></code>, 
<code><a href="#topic+compareSNPs">compareSNPs</a></code>, 
<code><a href="#topic+createTable">createTable</a></code>,
<code><a href="#topic+descrTable">descrTable</a></code>,
<code><a href="#topic+strataTable">strataTable</a></code>,
<code><a href="#topic+missingTable">missingTable</a></code>, 
<code><a href="#topic+export2latex">export2latex</a></code>,
<code><a href="#topic+export2html">export2html</a></code>, 
<code><a href="#topic+export2csv">export2csv</a></code>, 
<code><a href="#topic+export2pdf">export2pdf</a></code>, 
<code><a href="#topic+export2md">export2md</a></code>,
<code><a href="#topic+export2word">export2word</a></code>,
<code><a href="#topic+export2xls">export2xls</a></code>,
<code><a href="#topic+report">report</a></code>, 
<code><a href="#topic+radiograph">radiograph</a></code>, 
<code><a href="#topic+cGroupsGUI">cGroupsGUI</a></code>, 
<code><a href="#topic+cGroupsWUI">cGroupsWUI</a></code>
</p>


<h3>Author(s)</h3>

<p>Main functions: Isaac Subirana &lt;isubirana&lt;at&gt;imim.es&gt;, Joan Vila &lt;jvila&lt;at&gt;imim.es&gt;, Héctor Sanz &lt;hsrodenas&lt;at&gt;gmail.com&gt;, Gavin Lucas &lt;gavin.lucas&lt;at&gt;cleargenetics.com&gt; and David Giménez &lt;dgimenez1&lt;at&gt;imim.es&gt;<br />
</p>
<p>Web User Interface: Isaac Subirana &lt;isubirana&lt;at&gt;imim.es&gt;, Judith Peñafiel &lt;jpenafiel&lt;at&gt;imim.es&gt;, Gavin Lucas &lt;gavin.lucas&lt;at&gt;cleargenetics.com&gt; and David Giménez &lt;dgimenez1&lt;at&gt;imim.es&gt; <br />
</p>
<p>Maintainer: Isaac Subirana &lt;isubirana&lt;at&gt;imim.es&gt;
</p>


<h3>References</h3>

<p>Isaac Subirana, Hector Sanz, Joan Vila (2014). Building Bivariate Tables: The compareGroups Package for R. Journal of Statistical Software, 57(12), 1-16. URL <a href="https://www.jstatsoft.org/v57/i12/">https://www.jstatsoft.org/v57/i12/</a>.
</p>

<hr>
<h2 id='cGroupsGUI'>
Graphical user interface based on tcltk tools
</h2><span id='topic+cGroupsGUI'></span>

<h3>Description</h3>

<p>This function allows the user to build tables in an easy and intuitive way and to modify several options, using a graphical interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cGroupsGUI(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cGroupsGUI_+3A_x">X</code></td>
<td>
<p>a matrix or a data.frame. 'X' must exist in <code>.GlobalEnv</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the vignette for more detailed examples illustrating the use of this function.
</p>


<h3>Note</h3>

<p>If a data.frame or a matrix is passed through 'X' argument or is loaded by the 'Load data' GUI menu, this object is placed in the <code>.GlobalEnv</code>. Manipulating this data.frame or matrix while GUI is opened may produce an error in executing the GUI operations. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cGroupsWUI">cGroupsWUI</a></code>, <code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(regicor)
cGroupsGUI(regicor)

## End(Not run)
</code></pre>

<hr>
<h2 id='cGroupsWUI'>
Web User Interface based on Shiny tools.
</h2><span id='topic+cGroupsWUI'></span>

<h3>Description</h3>

<p>This function opens a web browser with a graphical interface based on <span class="pkg">shiny</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cGroupsWUI(port = 8102L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cGroupsWUI_+3A_port">port</code></td>
<td>
<p>integer. Same as 'port' argument of <code><a href="shiny.html#topic+runApp">runApp</a></code>. Default value is 8102L.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If an error occurs when launching the web browser, it may be solved by changing the port number.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cGroupsGUI">cGroupsGUI</a></code>, <code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

require(compareGroups)

cGroupsWUI()


## End(Not run)

</code></pre>

<hr>
<h2 id='compareGroups'>
Descriptives by groups
</h2><span id='topic+compareGroups'></span><span id='topic+print.compareGroups'></span><span id='topic+plot.compareGroups'></span><span id='topic+update.compareGroups'></span><span id='topic+summary.compareGroups'></span><span id='topic+print.summary.compareGroups'></span>

<h3>Description</h3>

<p>This function performs descriptives by groups for several variables. Depending on the nature of these variables, different descriptive statistics are calculated (mean, median, frequencies or K-M probabilities) and different tests are computed as appropriate (t-test, ANOVA, Kruskall-Wallis, Fisher, log-rank, ...).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareGroups(formula, data, subset, na.action = NULL, y = NULL, Xext = NULL, 
  selec = NA, method = 1, timemax = NA, alpha = 0.05, min.dis = 5, max.ylev = 5, 
  max.xlev = 10, include.label = TRUE, Q1 = 0.25, Q3 = 0.75, simplify = TRUE, 
  ref = 1, ref.no = NA, fact.ratio = 1, ref.y = 1, p.corrected = TRUE, 
  compute.ratio = TRUE, include.miss = FALSE, oddsratio.method = "midp", 
  chisq.test.perm = FALSE, byrow = FALSE, chisq.test.B = 2000, chisq.test.seed = NULL, 
  Date.format = "d-mon-Y", var.equal = TRUE, conf.level = 0.95, surv=FALSE,
  riskratio = FALSE, riskratio.method = "wald", compute.prop = FALSE, 
  lab.missing = "'Missing'", p.trend.method = "spearman")
## S3 method for class 'compareGroups'
plot(x, file, type = "pdf", bivar = FALSE, z=1.5, 
  n.breaks = "Sturges", perc = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareGroups_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;formula&quot; (or one that can be coerced to that class). Right side of ~ must have the terms in an additive way, and left side of ~ must contain the name of the grouping variable or can be left in blank (in this latter case descriptives for whole sample are calculated and no test is performed).</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible by 'as.data.frame' to a data frame) containing the variables in the model. If they are not found in 'data', the variables are taken from 'environment(formula)'.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of individuals to be used in the computation process. It is applied to all row-variables. 'subset' and 'selec' are added in the sense of '&amp;' to be applied in every row-variable.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data contain NAs. The default is NULL, and that is equivalent to <code><a href="stats.html#topic+na.fail">na.pass</a></code>, which means no action. Value <code><a href="stats.html#topic+na.fail">na.exclude</a></code> can be useful if it is desired to removed all individuals with some NA in any variable.</p>
</td></tr>    
<tr><td><code id="compareGroups_+3A_y">y</code></td>
<td>
<p>a vector variable that distinguishes the groups. It must be either a numeric, character, factor or NULL. Default value is NULL which means that descriptives for whole sample are calculated and no test is performed.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_xext">Xext</code></td>
<td>
<p>a data.frame or a matrix with the same rows / individuals contained in <code>X</code>, and maybe with different variables / columns than <code>X</code>. This argument is used by <code>compareGroups.default</code> in the sense that the variables specified in the argument <code>selec</code> are searched in <code>Xext</code> and/or in the <code><a href="base.html#topic+environment">.GlobalEnv</a></code>. If <code>Xext</code> is <code>NULL</code>, then Xext is created from variables of <code>X</code> plus <code>y</code>. Default value is <code>NULL</code>.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_selec">selec</code></td>
<td>
<p>a list with as many components as row-variables. If list length is 1 it is recycled for all row-variables. Every component of 'selec' is an expression that will be evaluated to select the individuals to be analyzed for every row-variable. Otherwise, a named list specifying 'selec' row-variables is applied. '.else' is a reserved name that defines the selection for the rest of the variables; if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is NA; all individuals are analyzed (no subsetting).</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_method">method</code></td>
<td>
<p>integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for continuous row-variables (for factor row-variables it is ignored). Possible values are: 1 - forces analysis as &quot;normal-distributed&quot;; 2 - forces analysis as &quot;continuous  non-normal&quot;; 3 - forces analysis as &quot;categorical&quot;;  and 4 - NA, which performs a Shapiro-Wilks  test to decide between normal or non-normal. Otherwise, a named vector specifying 'method' row-variables is applied. 
'.else' is a reserved name that defines the method for the rest of the variables; if no '.else' variable is defined, default value is applied. Default value is 1.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_timemax">timemax</code></td>
<td>
<p>double vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for 'Surv' class row-variables (for all other row-variables it is ignored).  This value indicates at which time the K-M probability is to be computed. Otherwise, a named vector specifying 'timemax' row-variables is applied. '.else' is a reserved name that defines the 'timemax' for the rest of the variables; if no '.else' variable is defined, default value is applied. Default value is NA; K-M probability is then computed at the median of observed times.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_alpha">alpha</code></td>
<td>
<p>double between 0 and 1. Significance threshold for the <code><a href="stats.html#topic+shapiro.test">shapiro.test</a></code> normality test for continuous row-variables. Default value is 0.05.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_min.dis">min.dis</code></td>
<td>
<p>an integer. If a non-factor row-variable contains less than 'min.dis' different values and 'method' argument is set to NA, then it will be converted to a factor. Default value is 5.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_max.ylev">max.ylev</code></td>
<td>
<p>an integer indicating the maximum number of levels of grouping variable ('y'). If 'y' contains more than 'max.ylev' levels, then the function 'compareGroups' produces an error. Default value is 5.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_max.xlev">max.xlev</code></td>
<td>
<p>an integer indicating the maximum number of levels when the row-variable is a factor. If the row-variable is a factor (or converted to a factor if it is a character, for example) and contains more than 'max.xlev' levels, then it is removed from the analysis and a warning is printed. Default value is 10.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_include.label">include.label</code></td>
<td>
<p>logical, indicating whether or not variable labels have to be shown in the results. Default value is TRUE</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_q1">Q1</code></td>
<td>
<p>double between 0 and 1, indicating the quantile to be displayed as the first number inside the square brackets in the bivariate table. To compute the minimum just type 0. Default value is 0.25 which means the first quartile.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_q3">Q3</code></td>
<td>
<p>double between 0 and 1, indicating the quantile to be displayed as the second number inside the square brackets in the bivariate table. To compute the maximum just type 1. Default value is 0.75 which means the third quartile.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_simplify">simplify</code></td>
<td>
<p>logical, indicating whether levels with no values must be removed for grouping variable and for row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_ref">ref</code></td>
<td>
<p>an integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for categorical row-variables. Or a named vector specifying which row-variables 'ref' is applied (a reserved name is '.else' which defines the reference category for the rest of the variables); if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is 1.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_ref.no">ref.no</code></td>
<td>
<p>character specifying the name of the level to be the reference for Odds Ratio or Hazard Ratio. It is not case-sensitive. This is especially useful for yes/no variables. Default value is NA which means that category specified in 'ref' is the one selected to be the reference.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_fact.ratio">fact.ratio</code></td>
<td>
<p>a double vector with as many components as row-variables  indicating  the units for the HR / OR (note that it does not affect the descriptives).  If its length is 1 it is recycled for all row-variables. Otherwise, a named vector specifying 'fact.ratio' row-variables is applied. '.else' is a reserved name that defines the reference category for the rest of the variables;  if no '.else' variable is defined, default value is applied. Default value is 1.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_ref.y">ref.y</code></td>
<td>
<p>an integer indicating the reference category of y variable for computing the OR, when y is a binary factor. Default value is 1.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_p.corrected">p.corrected</code></td>
<td>
<p>logical, indicating whether p-values for pairwise comparisons must be corrected. It only applies when there is a grouping variable with more than 2 categories. Default value is TRUE.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_compute.ratio">compute.ratio</code></td>
<td>
<p>logical, indicating whether Odds Ratio (for a binary response) or Hazard Ratio (for a time-to-event response) must be computed. Default value is TRUE.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_include.miss">include.miss</code></td>
<td>
<p>logical, indicating whether to treat missing values as a new category for categorical variables. Default value is FALSE.</p>
</td></tr>   
<tr><td><code id="compareGroups_+3A_oddsratio.method">oddsratio.method</code></td>
<td>
<p>Which method to compute the Odds Ratio. See 'method' argument from <code>oddsratio</code> (<code>epitools</code> package). Default value is &quot;midp&quot;.</p>
</td></tr>    
<tr><td><code id="compareGroups_+3A_byrow">byrow</code></td>
<td>
<p>logical or NA. Percentage of categorical variables must be reported by rows (TRUE), by columns (FALSE) or by columns and rows to sum up 1 (NA). Default value is FALSE, which means that percentages are reported by columns (withing groups).</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_chisq.test.perm">chisq.test.perm</code></td>
<td>
<p>logical. It applies a permutation chi squared test (<code><a href="stats.html#topic+chisq.test">chisq.test</a></code>) instead of an exact Fisher test (<code><a href="stats.html#topic+fisher.test">fisher.test</a></code>). It only applies when expected count in some cells are lower than 5.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_chisq.test.b">chisq.test.B</code></td>
<td>
<p>integer. Number of permutation when computing permuted chi squared test for categorical variables. Default value is 2000.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_chisq.test.seed">chisq.test.seed</code></td>
<td>
<p>integer or NULL. Seed when performing permuted chi squared test for categorical variables. Default value is NULL which sets no seed. It is important to introduce some number different from NULL in order to reproduce the results when permuted chi-squared test is performed.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_date.format">Date.format</code></td>
<td>
<p>character indicating how the dates are shown. Default is &quot;d-mon-Y&quot;. See <code><a href="chron.html#topic+chron">chron</a></code> for more details.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_var.equal">var.equal</code></td>
<td>
<p>logical, indicating whether to consider equal variances when comparing means on normal distributed variables on more than two groups. If TRUE <code><a href="stats.html#topic+anova">anova</a></code> function is applied and <code><a href="stats.html#topic+oneway.test">oneway.test</a></code> otherwise. Default value is TRUE.</p>
</td></tr>   
<tr><td><code id="compareGroups_+3A_conf.level">conf.level</code></td>
<td>
<p>double. Conficende level of confidence interval for means, medians, proportions or incidence, and hazard, odds and risk ratios. Default value is 0.95.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_surv">surv</code></td>
<td>
<p>logical. Compute survival (TRUE) or incidence (FALSE) for time-to-event row-variables. Default value is FALSE.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_riskratio">riskratio</code></td>
<td>
<p>logical. Whether to compute Odds Ratio (FALSE) or Risk Ratio (TRUE). Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_riskratio.method">riskratio.method</code></td>
<td>
<p>Which method to compute the Odds Ratio. See 'method' argument from <code>riskratio</code> (<code>epitools</code> package). Default value is &quot;wald&quot;.</p>
</td></tr>     
<tr><td><code id="compareGroups_+3A_compute.prop">compute.prop</code></td>
<td>
<p>logical. Compute proportions (TRUE) or percentages (FALSE) for cathegorical row-variables. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_lab.missing">lab.missing</code></td>
<td>
<p>character. Label for missing cathegory. Only applied when <code>include.missing = TRUE</code>. Default value is 'Missing'.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_p.trend.method">p.trend.method</code></td>
<td>
<p>Character indicating the name of test to use for p-value for trend. It only applies for numerical non-normal variables. Possible values are &quot;spearman&quot;, &quot;kendall&quot; or &quot;cuzick&quot;. Default value is &quot;spearman&quot;. See section details for more info.</p>
</td></tr>
</table>
<p>Arguments passed to <code>plot</code> method.  
</p>
<table role = "presentation">
<tr><td><code id="compareGroups_+3A_x">x</code></td>
<td>
<p>an object of class 'compareGroups'.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_file">file</code></td>
<td>
<p>a character string giving the name of the file. A bmp, jpg, png or tif file is saved with an appendix added to 'file' corresponding to the row-variable name. If 'onefile' argument is set to TRUE throught '...' argument of plot method function, a unique PDF file is saved named as [file].pdf. If it is missing, multiple devices are opened, one for each row-variable of 'x' object.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_type">type</code></td>
<td>
<p>a character string indicating the file format where the plots are stored. Possibles foramts are 'bmp', 'jpg', 'png', 'tif' and 'pdf'.Default value is 'pdf'.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_bivar">bivar</code></td>
<td>
<p>logical. If bivar=TRUE, it plots a boxplot or a barplot (for a continuous or categorical row-variable, respectively) stratified by groups. If bivar=FALSE, it plots a normality plot (for continuous row-variables) or a barplot (for categorical row-variables). Default value is FALSE.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_z">z</code></td>
<td>
<p>double. Indicates threshold limits to be placed in the deviation from normality plot. It is considered that too many points beyond this threshold indicates that current variable is far to be normal-distributed. Default value is 1.5.</p>
</td></tr>
<tr><td><code id="compareGroups_+3A_n.breaks">n.breaks</code></td>
<td>
<p>same as argument 'breaks' of <code><a href="graphics.html#topic+hist">hist</a></code>.</p>
</td></tr> 
<tr><td><code id="compareGroups_+3A_perc">perc</code></td>
<td>
<p>logical. Relative frequencies (in percentatges) instead of absolute frequencies are displayed in barplots for categorical variable.</p>
</td></tr>  
<tr><td><code id="compareGroups_+3A_...">...</code></td>
<td>
<p>For 'plot' method, '...' arguments are passed to <code><a href="grDevices.html#topic+pdf">pdf</a></code>, <code><a href="grDevices.html#topic+png">bmp</a></code>, <code><a href="grDevices.html#topic+png">jpeg</a></code>, <code><a href="grDevices.html#topic+png">png</a></code> or <code><a href="grDevices.html#topic+png">tiff</a></code> if 'type' argument equals to 'pdf', 'bmp', 'jpg', 'png' or 'tif', respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Depending whether the row-variable is considered as continuous normal-distributed (1), continuous non-normal distributed (2) or categorical (3), the following descriptives and tests are performed: <br />
1- mean, standard deviation and t-test or ANOVA <br />
2- median, 1st and 3rd quartiles (by default), and Kruskall-Wallis test <br />
3- or absolute and relative frequencies and chi-squared or exact Fisher test when the expected frequencies is less than 5 in some cell<br />
Also, a row-variable can be of class 'Surv'. Then the probability of 'event' at a fixed time (set up with 'timemax' argument) is computed and a logrank test is performed.<br /> 
</p>
<p>When there are more than 2 groups, it also performs pairwise comparisons adjusting for multiple testing (Tukey when row-variable is normal-distributed and Benjamini &amp; Hochberg method otherwise), and computes p-value for trend. 
The p-value for trend is computed from the Pearson test when row-variable is normal and from the Spearman test when it is continuous non normal. Also, for continuous non normal distributed variables, it is possible to compute the p-value for trend using the Kendall's test (<code>method='kendall'</code> from <code><a href="stats.html#topic+cor.test">cor.test</a></code>) or Cuzick's test (<code><a href="PMCMRplus.html#topic+cuzickTest">cuzickTest</a></code>). 
If row-variable is of class 'Surv', the score test is computed from a Cox model where the grouping variable is introduced as an integer variable predictor. 
If the row-variable is categorical, the p-value for trend is computed from Mantel-Haenszel test of trend.
</p>
<p>If there are two groups, the Odds Ratio or Risk Ratio is computed for each row-variable. While, if the response is of class 'Surv' (i.e. time to event) Hazard Ratios are computed. 
When x-variable is a factor, the Odds Ratio and Risk Ratio are computed using <code>oddsratio</code> and <code>riskratio</code>, respectively, from <code>epitools</code> package. While when x-variable is a continuous variable, the Odds Ratio and Risk Ratio are computed under a logistic regression with a canonical link and the log link, respectively. <br />
</p>
<p>The p-values for Hazard Ratios are computed using the logrank or Wald test under a Cox proportional hazard regression when row-variable is categorical or continuous, respectively. <br />
</p>
<p>See the vignette for more detailed examples illustrating the use of this function and the methods used.
</p>


<h3>Value</h3>

<p>An object of class 'compareGroups'.  <br />
</p>
<p>'print' returns a table sample size, overall p-values, type of variable ('categorical', 'normal', 'non-normal' or 'Surv') and the subset of individuals selected.  <br />
</p>
<p>'summary' returns a much more detailed list. Every component of the list is the result for each row-variable, showing frequencies, mean, standard deviations, quartiles or K-M probabilities as appropriate. Also, it shows overall p-values as well as p-trends and pairwise p-values among the groups. <br />
</p>
<p>'plot' displays, for all the analyzed variables, normality plots (with the Shapiro-Wilks test), barplots or Kaplan-Meier plots depending on whether the row-variable is continuous, categorical or time-to-response, respectevily. Also, bivariate plots can be displayed with stratified by groups boxplots or barplots, setting 'bivar' argument to TRUE.  <br />
</p>
<p>An update method for 'compareGroups' objects has been implemented and works as usual to change all the arguments of previous analysis. <br />
</p>
<p>A subset, '[', method has been implemented for 'compareGroups' objects. The subsetting indexes can be either integers (as usual), row-variables names or row-variable labels.       <br />
</p>
<p>Combine by rows,'rbind', method has been implemented for 'compareGroups' objects. It is useful to distinguish row-variable groups. <br /> 
</p>
<p>See examples for further illustration about all previous issues. 
</p>


<h3>Note</h3>

<p>By default, the labels of the variables (row-variables and grouping variable) are displayed in the resulting tables. These labels are taken from the &quot;label&quot; attribute of each variable. And if this attribute is NULL, then the name of the variable is displayed, instead. 
To label non-labeled variables, or to change their labels, specify its &quot;label&quot; atribute directly.<br />
</p>
<p>There may be no equivalence between the intervals of the OR / HR and p-values. For example, when the response variable is binary and the row-variable is continuous, p-value is based on Mann-Whitney U test or t-test depending on whether row-variable is normal distributed or not, respectively, while the confidence interval is build using the Wald method (log(OR) -/+ 1.96*se). Or when the answer is of class 'Surv', p-value is computed with the logrank test, while confidence intervals are based on the Wald method (log(HR) -/+ 1.96*se). 
Finally, when the response is binary and the row variable is categorical, the p-value is based on the chi-squared or Fisher test when appropriate, while confidence intervals are constructed from the median-unbiased estimation method (see <code>oddsratio</code> function from <code>epitools</code> package). <br />
</p>
<p>Subjects selection criteria specified in 'selec' and 'subset' arguments are combined using '&amp;' to be applied to every row-variable.<br />
</p>
<p>Through '...' argument of 'plot' method, some parameters such as figure size, multiple figures in a unique file (only for 'pdf' files), resolution, etc. are controlled. For more information about which arguments can be passed depending on the format type, see <code><a href="grDevices.html#topic+pdf">pdf</a></code>, <code><a href="grDevices.html#topic+png">bmp</a></code>, <code><a href="grDevices.html#topic+png">jpeg</a></code>, <code><a href="grDevices.html#topic+png">png</a></code> or <code><a href="grDevices.html#topic+png">tiff</a></code>.<br />
</p>
<p>Since version 4.0, date variables are supported. For this kind of variables only method==2 is applied, i.e. non-parametric tests for continuous variables are applied. However, the descriptive statistics (medians and quantiles) are displayed in date format instead of numeric format.
</p>


<h3>References</h3>

<p>Isaac Subirana, Hector Sanz, Joan Vila (2014). Building Bivariate Tables: The compareGroups Package for R. Journal of Statistical Software, 57(12), 1-16. URL <a href="https://www.jstatsoft.org/v57/i12/">https://www.jstatsoft.org/v57/i12/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups)
require(survival)

# load REGICOR data
data(regicor)

# compute a time-to-cardiovascular event variable
regicor$tcv &lt;- with(regicor, Surv(tocv, as.integer(cv=='Yes')))
attr(regicor$tcv,"label")&lt;-"Cardiovascular"

# compute a time-to-overall death variable
regicor$tdeath &lt;- with(regicor, Surv(todeath, as.integer(death=='Yes')))
attr(regicor$tdeath,"label") &lt;- "Mortality"

# descriptives by sex
res &lt;- compareGroups(sex ~ .-id-tocv-cv-todeath-death, data = regicor)
res

# summary of each variable
summary(res)

# univariate plots of all row-variables
## Not run: 
plot(res)

## End(Not run)

# plot of all row-variables by sex
## Not run: 
plot(res, bivar = TRUE)

## End(Not run)

# update changing the response: time-to-cardiovascular event.
# note that time-to-death must be removed since it is not possible 
# not compute descriptives of a 'Surv' class object by another 'Surv' class object.

## Not run: 
update(res, tcv ~ . + sex - tdeath - tcv)

## End(Not run)

</code></pre>

<hr>
<h2 id='compareGroups-internal'>Internal compareGroups functions</h2><span id='topic+compareGroups-internal'></span><span id='topic+chisq.test2'></span><span id='topic+combn2'></span><span id='topic+compare.i'></span><span id='topic+descrip'></span><span id='topic+descripSurv'></span><span id='topic+flip'></span><span id='topic+format2'></span><span id='topic+logrank.pval'></span><span id='topic+signifdec.i'></span><span id='topic+signifdec'></span><span id='topic+summ.i'></span><span id='topic+table.i'></span><span id='topic+udpate.formula2'></span><span id='topic+KMg.plot'></span><span id='topic+Cox.plot'></span><span id='topic+bar2.plot'></span><span id='topic+box.plot'></span><span id='topic+KM.plot'></span><span id='topic+bar.plot'></span><span id='topic+norm.plot'></span><span id='topic+prepare'></span><span id='topic+trim'></span><span id='topic+compareGroups.fit'></span><span id='topic+confinterval'></span><span id='topic+formula.compareGroups'></span><span id='topic+update.formula2'></span><span id='topic+snpQC'></span><span id='topic+export2mdcbind'></span><span id='topic+export2mdword'></span><span id='topic+export2mdwordcbind'></span><span id='topic+oddsratio'></span><span id='topic+riskratio'></span><span id='topic+epitable'></span><span id='topic+table.margins'></span><span id='topic+or.midp'></span><span id='topic+tab2by2.test'></span><span id='topic+ormidp.test'></span><span id='topic+oddsratio.midp'></span><span id='topic+oddsratio.fisher'></span><span id='topic+oddsratio.wald'></span><span id='topic+oddsratio.small'></span><span id='topic+riskratio.small'></span><span id='topic+riskratio.wald'></span><span id='topic+riskratio.boot'></span><span id='topic+setupSNP2'></span><span id='topic+snp2'></span><span id='topic+reorder.snp2'></span><span id='topic+summary.snp2'></span><span id='topic+SNPHWE2'></span>

<h3>Description</h3>

<p>Internal compareGroups functions</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisq.test2(obj, chisq.test.perm, chisq.test.B, chisq.test.seed)
combn2(x)
compareGroups.fit(X, y, Xext, selec, method, timemax, alpha, min.dis, max.ylev, max.xlev, 
         include.label, Q1, Q3, simplify, ref, ref.no, fact.ratio, ref.y, p.corrected, 
         compute.ratio, include.miss, oddsratio.method, chisq.test.perm, byrow, 
         chisq.test.B, chisq.test.seed, Date.format, var.equal, conf.level, surv, 
         riskratio, riskratio.method, compute.prop, lab.missing, p.trend.method)
compare.i(x, y, selec.i, method.i, timemax.i, alpha, min.dis, max.xlev, varname, Q1, Q3, 
          groups, simplify, Xext, ref, fact.ratio, ref.y, p.corrected, compute.ratio, 
          include.miss, oddsratio.method, chisq.test.perm, byrow,chisq.test.B, 
          chisq.test.seed, Date.format, var.equal, conf.level, surv, riskratio, 
          riskratio.method, compute.prop, lab.missing, p.trend.method)
descripSurv(x, y, timemax, surv)
descrip(x, y, method, Q1, Q3, conf.level)
confinterval(x, method, conf.level)
flip(x)
format2(x, digits = NULL, stars = FALSE, ...)
## S3 method for class 'compareGroups'
formula(x, ...) 
logrank.pval(x,y)
signifdec.i(x, digits)
signifdec(x, digits)
summ.i(x)
table.i(x, hide.i, digits, digits.ratio, type, varname, hide.i.no, digits.p, sd.type, 
        q.type, spchar, show.ci, lab.ref, stars)
## S3 method for class 'formula2'
update(object, new, ...) 
KMg.plot(x, y, file, var.label.x, var.label.y, ...)
Cox.plot(x, y, file, var.label.x, var.label.y, ...)
bar2.plot(x, y, file, var.label.x, var.label.y, perc, byrow, ...)
box.plot(x, y, file, var.label.x, var.label.y, ...)
KM.plot(x, file, var.label.x, ...)
bar.plot(x, file, var.label.x, perc, ...)
norm.plot(x, file, var.label.x, z, n.breaks, ...)
prepare(x, nmax, nmax.method, header.labels)
snpQC(X, sep, verbose)
export2mdcbind(x, which.table, nmax, nmax.method, header.labels, caption, 
      strip, first.strip, background, width, size, landscape, format, 
      header.background, header.color, position, ...)
export2mdword(x, which.table, nmax, nmax.method, header.labels, caption, strip, 
      first.strip, background, size, header.background, header.color)
export2mdwordcbind(x, which.table, nmax, nmax.method, header.labels, caption, 
      strip, first.strip, background, size, header.background, header.color)         
        
trim(x)

oddsratio(x, method = c("midp", "fisher", "wald", "small"), conf.level = 0.95)
riskratio(x, method = c("wald", "small", "boot"), conf.level = 0.95)
epitable(..., ncol = 2, byrow = TRUE, rev = c("neither", "rows", "columns", "both"))
table.margins(x)
or.midp(x, conf.level = 0.95, byrow = TRUE, interval = c(0,1000))
tab2by2.test(x)
ormidp.test(a1, a0, b1, b0, or = 1)
oddsratio.midp(x, conf.level = 0.95)
oddsratio.fisher(x, conf.level = 0.95)
oddsratio.wald(x, conf.level = 0.95)
oddsratio.small(x, conf.level = 0.95)
riskratio.small(x, conf.level = 0.95)
riskratio.wald(x, conf.level = 0.95)
riskratio.boot(x, conf.level = 0.95)

setupSNP2(data, colSNPs, sep)
snp2(x, sep = "/", name.genotypes, reorder = "common", 
          remove.spaces = TRUE, allow.partial.missing = FALSE) 

reorder.snp2(x, ref = "common", ...)

## S3 method for class 'snp2'
summary(object, ...)

SNPHWE2(x)

</code></pre>


<h3>Details</h3>

<p>These are not to be called by the user</p>

<hr>
<h2 id='compareSNPs'>
Summarise genetic data by groups.
</h2><span id='topic+compareSNPs'></span><span id='topic+print.compareSNPs'></span>

<h3>Description</h3>

<p>This function provides an extensive summary range of your SNP data, allowing you to perform in-depth quality control of your genotyping results, and to explore your data before analysis. Summary measures include allele and genotype frequencies and counts, missingness rate, Hardy Weinberg equilibrium and more in the whole data set or stratified by other variables, such as case-control status. It can also test for differences in missingness between groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareSNPs(formula, data, subset, na.action = NULL, sep = "", verbose = FALSE, ...) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compareSNPs_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;formula&quot; (or one that can be coerced to that class). The right side of ~ must have the terms in an additive way, and these terms must refer to variables in 'data' must be of character or factor classes whose levels are the genotypes with the alleles written in their levels (e.g. A/A, A/T and T/T). The left side of ~ must contain the name of the grouping variable or can be left blank (in this case, summary data are provided for the whole sample, and no missingness test is performed).</p>
</td></tr>
<tr><td><code id="compareSNPs_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible by 'as.data.frame' to a data frame) containing the variables in the model. If they are not found in 'data', the variables are taken from 'environment(formula)'.</p>
</td></tr>
<tr><td><code id="compareSNPs_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of individuals to be used in the computation process (applied to all genetic variables).</p>
</td></tr>
<tr><td><code id="compareSNPs_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data contain NAs. The default is NULL, and that is equivalent to <code><a href="stats.html#topic+na.fail">na.pass</a></code>, which means no action. Value <code><a href="stats.html#topic+na.fail">na.exclude</a></code> can be useful if it is desired to removed all individuals with some NA in any variable.</p>
</td></tr>    
<tr><td><code id="compareSNPs_+3A_sep">sep</code></td>
<td>
<p>character string indicating the separator between alleles (e.g. when using A/A, A/T and T/T genotype codification, 'sep' should be set to '/'. Default value is &rdquo; indicating that genotypes are coded as AA, AT and TT.</p>
</td></tr> 
<tr><td><code id="compareSNPs_+3A_verbose">verbose</code></td>
<td>
<p>logical, print results from <code><a href="HardyWeinberg.html#topic+HWChisq">HWChisq</a></code> function. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="compareSNPs_+3A_...">...</code></td>
<td>
<p>currently ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'compareSNPs' which is a data.frame (when no groups are specified on the left of the '~' in the 'formula' argument) or a list of data.frames, otherwise. Each data.frame contains the following fields: <br />
- Ntotal: Total number of samples for which genotyping was attempted <br />
- Ntyped: Number of genotypes called <br />
- Typed.p: Percentage genotyped <br />
- Miss.t: Number of missing genotypes  <br />
- Miss.p: Proportion of missing genotypes    <br />
- Minor: Minor Allele        <br />
- MAF: Minor allele frequency  <br />
- A1: Allele 1     <br />
- A2: Allele 2     <br />
- A1.ct: Count Allele 1  <br />
- A2.ct: Count Allele 2  <br />
- A1.p: Frequency of Allele 1 <br />
- A2.p: Frequency of Allele 2   <br />
- Hom1: Allele 1 Homozygote   <br />
- Het: Heterozygote    <br />
- Hom2: Allele 2 Homozygote  <br />
- Hom1.ct: Allele 1 Homozygote count  <br />
- Het.ct: Heterozygote Count     <br />
- Hom2.ct: Allele 2 Homozygote count <br />
- Hom1.p: Frequency of Allele 1 Homozygote   <br />
- Het.p: Heterozygote frequency      <br />
- Hom2.p: Frequency of Allele 2 Homozygote  <br />
- HWE.p: Hardy-Weinberg equilibrium p-value <br />
Additionaly, when analysis is stratified by groups, the last component consists of a data.frame containing the p-values of missingness comparison among groups.   <br />
</p>
<p>'print' returns a 'nice' format table for each group with the main results for each SNP (Ntotal, Ntyped, Minor, MAF, A1, A2, HWE.p), and the missingness test when group is considered.  <br />
</p>


<h3>Note</h3>

<p>It uses some functions taken from <span class="pkg">SNPassoc</span> created by Juan Ram?n Gonz?lez et al. <br />
</p>
<p>Hardy-Weinberg equilibrium test is performed using the  <code><a href="HardyWeinberg.html#topic+HWChisqMat">HWChisqMat</a></code>
</p>


<h3>Author(s)</h3>

<p>Gavin Lucas (gavin.lucas&lt;at&gt;cleargenetics.com) <br />
</p>
<p>Isaac Subirana (isubirana&lt;at&gt;imim.es)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups) 

# load example data
data(SNPs)

# visualize first rows
head(SNPs)

# select casco and all SNPs
myDat &lt;- SNPs[,c(2,6:40)]

# QC of three SNPs by groups of cases and controls
res&lt;-compareSNPs(casco ~ .-casco, myDat)
res

# QC of three SNPs of the whole data set
res&lt;-compareSNPs( ~ .-casco, myDat)
res

</code></pre>

<hr>
<h2 id='createTable'>
Table of descriptives by groups: bivariate table 
</h2><span id='topic+createTable'></span><span id='topic+print.createTable'></span><span id='topic+summary.createTable'></span><span id='topic+print.summary.createTable'></span><span id='topic+plot.createTable'></span>

<h3>Description</h3>

<p>This functions builds a &quot;compact&quot; and &quot;nice&quot; table with the descriptives by groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createTable(x, hide = NA, digits = NA, type = NA, show.p.overall = TRUE,
           show.all, show.p.trend, show.p.mul = FALSE, show.n, show.ratio =
           FALSE, show.descr = TRUE, show.ci = FALSE, hide.no = NA, digits.ratio = NA,
           show.p.ratio = show.ratio, digits.p = 3, sd.type = 1, q.type = c(1, 1),
           extra.labels = NA, all.last = FALSE, lab.ref = "Ref.", stars = FALSE)
## S3 method for class 'createTable'
print(x, which.table = "descr", nmax = TRUE, nmax.method = 1,
           header.labels = c(), ...)
## S3 method for class 'createTable'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createTable_+3A_x">x</code></td>
<td>
<p>an object of class 'compareGroups'</p>
</td></tr>
<tr><td><code id="createTable_+3A_hide">hide</code></td>
<td>
<p>a vector (or a list) with integers or characters with as many components as row-variables. If its length is 1 it is recycled for all row-variables. Each component specifies which category (the literal name of the category if it is a character, or the position if it is an integer) must be hidden and not shown. This argument only applies to categorical row-variables, and for continuous row-variables it is ignored. If NA, all categories are displayed. Or a named vector (or a named list) specifying which row-variables 'hide' is applied, and for the rest of row-variables default value is applied. Default value is NA.</p>
</td></tr>
<tr><td><code id="createTable_+3A_digits">digits</code></td>
<td>
<p>an integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. Each component specifies the number of significant decimals to be displayed. Or a named vector specifying which row-variables 'digits' is applied (a reserved name is '.else' which defines 'digits' for the rest of the variables); if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is NA which puts the 'appropriate' number of decimals (see vignette for further details).</p>
</td></tr>
<tr><td><code id="createTable_+3A_type">type</code></td>
<td>
<p>an integer that indicates whether absolute and/or relative frequencies are displayed: 1 - only relative frequencies; 2 or NA - absolute and relative frequencies in brackets; 3 - only absolute frequencies.</p>
</td></tr> 
<tr><td><code id="createTable_+3A_show.p.overall">show.p.overall</code></td>
<td>
<p>logical indicating whether p-value of overall groups significance ('p.overall' column) is displayed or not. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.all">show.all</code></td>
<td>
<p>logical indicating whether the '[ALL]' column (all data without stratifying by groups) is displayed or not. Default value is FALSE if grouping variable is defined, and FALSE if there are no groups.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.p.trend">show.p.trend</code></td>
<td>
<p>logical indicating whether p-trend is displayed or not. It is always FALSE when there are less than 3 groups. If this argument is missing, there are more than 2 groups and the grouping variable is an ordered factor, then p-trend is displayed. By default, p-trend is not displayed, and it is displayed when there are more than 2 groups and the grouping variable is of class ordered-factor.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.p.mul">show.p.mul</code></td>
<td>
<p>logical indicating whether the pairwise (between groups) comparisons p-values are displayed or not. It is always FALSE when there are less than 3 groups. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.n">show.n</code></td>
<td>
<p>logical indicating whether number of individuals analyzed for each row-variable is displayed or not in the 'descr' table. Default value is FALSE and it is TRUE when there are no groups.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.ratio">show.ratio</code></td>
<td>
<p>logical indicating whether OR / HR is displayed or not. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="createTable_+3A_show.descr">show.descr</code></td>
<td>
<p>logical indicating whether descriptives (i.e. mean, proportions, ...) are displayed. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.ci">show.ci</code></td>
<td>
<p>logical indicating whether to show confidence intervals of means, medians, proporcions or incidences are displayed. If so, they are displayed between squared brackets. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="createTable_+3A_hide.no">hide.no</code></td>
<td>
<p>character specifying the name of the level to be hidden for all categorical variables with 2 categories. It is not case-sensitive. The result is one row for the variable with only the name displayed and not the category. This is especially useful for yes/no variables. It is ignored for the categorical row-variables with 'hide' argument different from NA. Default value is NA which means that no category is hidden.</p>
</td></tr>
<tr><td><code id="createTable_+3A_digits.ratio">digits.ratio</code></td>
<td>
<p>The same as 'digits' argument but applied for the Hazard Ratio or Odds Ratio.</p>
</td></tr>
<tr><td><code id="createTable_+3A_show.p.ratio">show.p.ratio</code></td>
<td>
<p>logical indicating whether p-values corresponding to each Hazard Ratio / Odds Ratio are shown.</p>
</td></tr>
<tr><td><code id="createTable_+3A_digits.p">digits.p</code></td>
<td>
<p>integer indicating the number of decimals displayed for all p-values. Default value is 3.</p>
</td></tr>
<tr><td><code id="createTable_+3A_sd.type">sd.type</code></td>
<td>
<p>an integer that indicates how standard deviation is shown: 1 - mean (SD), 2 -  mean ? SD.</p>
</td></tr>
<tr><td><code id="createTable_+3A_q.type">q.type</code></td>
<td>
<p>a vector with two integer components. The first component refers to the type of brackets to be displayed for non-normal row-variables (1 - squared and 2 - rounded), while the second refers to the percentile separator (1 - ';', 2 - ',' and 3 - '-'. Default value is c(1, 1).</p>
</td></tr>
<tr><td><code id="createTable_+3A_extra.labels">extra.labels</code></td>
<td>
<p>character vector of 4 components corresponding to key legend to be appended to normal, non-normal, categorical or survival row-variables labels. Default value is NA which appends no extra key. If it is set to <code>c("","","","")</code>, &quot;Mean (SD)&quot;, &quot;Median [25th; 75th]&quot;, &quot;N (%)&quot; and &quot;Incidence at time=timemax&quot; are appended (see argument <code>timemax</code> from <code>compareGroups</code> function.</p>
</td></tr>
<tr><td><code id="createTable_+3A_all.last">all.last</code></td>
<td>
<p>logical. Descriptives of the whole sample is placed after the descriptives by groups. Default value is FALSE which places the descriptives of whole cohort at first.</p>
</td></tr>
<tr><td><code id="createTable_+3A_lab.ref">lab.ref</code></td>
<td>
<p>character. String shown for reference category. &quot;Ref.&quot; as default value.</p>
</td></tr>
<tr><td><code id="createTable_+3A_stars">stars</code></td>
<td>
<p>logical, indicating whether to append stars beside p-values; '**': p-value &lt; 0.05, '*' 0.05 &lt;= p-value &lt; 0.1; &rdquo; p-value &gt;=0.1. Default value is FALSE</p>
</td></tr>
<tr><td><code id="createTable_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), printing descriptives by groups table, availability data table or both tables, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="createTable_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="createTable_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="createTable_+3A_header.labels">header.labels</code></td>
<td>
<p>a character named vector with 'all', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' components indicating the label for '[ALL]', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' (available data), respectively. Default is a zero length vector which makes no changes, i.e. '[ALL]', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' labels appear for descriptives of entire cohort, global p-value, p-value for trend, HR/OR and p-value of each HR/OR and available data, respectively.</p>
</td></tr> 
<tr><td><code id="createTable_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="base.html#topic+print.default">print.default</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'createTable', which contains a list of 2 matrix:
</p>
<table role = "presentation">
<tr><td><code>descr</code></td>
<td>
<p>a character matrix of descriptives for all row-variables by groups and p-values in a 'compact' format</p>
</td></tr>
<tr><td><code>avail</code></td>
<td>
<p>a character matrix indicating the number of available data for each group, the type of variable (categorical, continuous-normal or continuous-non-normal) and the individuals selection made (if non selection 'ALL' is displayed).</p>
</td></tr>
</table>
<p>'print' prints these two tables in a 'nice' format.
</p>
<p>'summary' prints the 'available' info table (it is a short form of <code>print(x, which.table = 'avail')</code>).
</p>
<p>'update' modifies previous results from 'createTable'.
</p>
<p>'plot' see the method in <code><a href="#topic+compareGroups">compareGroups</a></code> function.
</p>
<p>subsetting, '[', can also be applied to 'createTable' objects in the same way as 'compareGroups' objects.
</p>
<p>combine by rows, 'rbind', method can be applied to 'createTable' objects, but only if all 'createTable' objects have the same columns. It is useful to distinguish row-variable groups. The resulting object is of class 'rbind.createTable' and 'createTable'.
</p>
<p>combine by columns, 'cbind', method can be applied to 'createTable' objects, but only if all 'createTable' objects have the same rows. It may be used when combining different tables referring to different subsets of people (for example, men and women). The resulting object is of class 'cbind.createTable' and 'createTable' and has its own 'print' method.
</p>
<p>See the vignette for more details.
</p>


<h3>Note</h3>

<p>The way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail').
</p>
<p>The p-values corresponding to the OR of a two level row-variable may not me equal to its p.overall p-value. This is because statistical tests are different: the option 'midp.exact' (see <code>oddsratio</code> from <code>epitools</code> package for more details) is taken in the first case and Chi-square or Fisher exact test in the second. The same happens when OR for a continuous value is performed: the p-value corresponding to this OR is computed form a logistic regression and therefore may differ from the one computed using a Student-T test or Kruskall Wallis test.
This discordance may also be present when computing the p-value corresponding to a Hazard Ratio for a categorical two level row-variable: a Wald test or a long-rank test are peformed.
</p>


<h3>References</h3>

<p>Isaac Subirana, Hector Sanz, Joan Vila (2014). Building Bivariate Tables: The compareGroups Package for R. Journal of Statistical Software, 57(12), 1-16. URL <a href="https://www.jstatsoft.org/v57/i12/">https://www.jstatsoft.org/v57/i12/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups)
require(survival)

# load REGICOR data
data(regicor)

# compute a time-to-cardiovascular event variable
regicor$tcv &lt;- with(regicor,Surv(tocv, as.integer(cv=='Yes')))
attr(regicor$tcv, "label")&lt;-"Cardiovascular incidence"

# descriptives by time-to-cardiovascular event, taking 'no' category as 
# the reference in computing HRs.
res &lt;- compareGroups(tcv ~ age + sex + smoker + sbp + histhtn + 
         chol + txchol + bmi + phyact + pcs + tcv, regicor, ref.no='no')

# build table showing HR and hiding the 'no' category
restab &lt;- createTable(res, show.ratio = TRUE, hide.no = 'no')
restab

# prints available info table
summary(restab)


# more...

## Not run: 

# Adds the 'available data' column
update(restab, show.n=TRUE)

# Descriptive of the entire cohort
update(restab, x = update(res, ~ . ))

# .. changing the response variable to sex
# Odds Ratios (OR) are displayed instead of Hazard Ratios (HR).
# note that now it is possible to compute descriptives by time-to-death 
# or time-to-cv but not the ORs . 
# We set timemax to 5 years, to report the probability of death and CV at 5 years:
update(restab, x = update(res, sex ~ . - sex + tdeath + tcv, timemax = 5*365.25))


## Combining tables:

# a) By rows: takes the first four variables as a group and the rest as another group:
rbind("First group of variables"=restab[1:4],"Second group of variables"=
  restab[5:length(res)])

# b) By columns: puts stratified tables by sex one beside the other:
res1&lt;-compareGroups(year ~ . - id - sex, regicor)
restab1&lt;-createTable(res1, hide.no = 'no')
restab2&lt;-update(restab1, x = update(res1, subset = sex == 'Male'))
restab3&lt;-update(restab1, x = update(res1, subset = sex == 'Female'))
cbind("ALL" = restab1, "MALES" = restab2, "FEMALES" = restab3)


## End(Not run)

</code></pre>

<hr>
<h2 id='descrTable'>
Perform descriptives and build the bivariate table.
</h2><span id='topic+descrTable'></span>

<h3>Description</h3>

<p>This functions builds a bivariate table calling compareGroups and createTable function in one step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descrTable(formula, data, subset, na.action = NULL, y = NULL, Xext = NULL, 
  selec = NA, method = 1, timemax = NA, alpha = 0.05, min.dis = 5, max.ylev = 5, 
  max.xlev = 10, include.label = TRUE, Q1 = 0.25, Q3 = 0.75, simplify = TRUE, 
  ref = 1, ref.no = NA, fact.ratio = 1, ref.y = 1, p.corrected = TRUE, 
  compute.ratio = TRUE, include.miss = FALSE, oddsratio.method = "midp", 
  chisq.test.perm = FALSE, byrow = FALSE, chisq.test.B = 2000, chisq.test.seed = NULL,
  Date.format = "d-mon-Y", var.equal = TRUE, conf.level = 0.95, surv = FALSE,
  riskratio = FALSE, riskratio.method = "wald", compute.prop = FALSE, 
  lab.missing = "'Missing'", p.trend.method = "spearman",
  hide = NA, digits = NA, type = NA, show.p.overall = TRUE,
  show.all, show.p.trend, show.p.mul = FALSE, show.n, show.ratio =
  FALSE, show.descr = TRUE, show.ci = FALSE, hide.no = NA, digits.ratio = NA,
  show.p.ratio = show.ratio, digits.p = 3, sd.type = 1, q.type = c(1, 1),
  extra.labels = NA, all.last = FALSE, lab.ref="Ref.", stars = FALSE)
</code></pre>


<h3>Arguments</h3>

<p>Arguments from <code><a href="#topic+compareGroups">compareGroups</a></code> function:
</p>
<table role = "presentation">
<tr><td><code id="descrTable_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;formula&quot; (or one that can be coerced to that class). Right side of ~ must have the terms in an additive way, and left side of ~ must contain the name of the grouping variable or can be left in blank (in this latter case descriptives for whole sample are calculated and no test is performed).</p>
</td></tr>
<tr><td><code id="descrTable_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible by 'as.data.frame' to a data frame) containing the variables in the model. If they are not found in 'data', the variables are taken from 'environment(formula)'.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of individuals to be used in the computation process. It is applied to all row-variables. 'subset' and 'selec' are added in the sense of '&amp;' to be applied in every row-variable.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_na.action">na.action</code></td>
<td>
<p>a function which indicates what should happen when the data contain NAs. The default is NULL, and that is equivalent to <code><a href="stats.html#topic+na.fail">na.pass</a></code>, which means no action. Value <code><a href="stats.html#topic+na.fail">na.exclude</a></code> can be useful if it is desired to removed all individuals with some NA in any variable.</p>
</td></tr>    
<tr><td><code id="descrTable_+3A_y">y</code></td>
<td>
<p>a vector variable that distinguishes the groups. It must be either a numeric, character, factor or NULL. Default value is NULL which means that descriptives for whole sample are calculated and no test is performed.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_xext">Xext</code></td>
<td>
<p>a data.frame or a matrix with the same rows / individuals contained in <code>X</code>, and maybe with different variables / columns than <code>X</code>. This argument is used by <code>compareGroups.default</code> in the sense that the variables specified in the argument <code>selec</code> are searched in <code>Xext</code> and/or in the <code><a href="base.html#topic+environment">.GlobalEnv</a></code>. If <code>Xext</code> is <code>NULL</code>, then Xext is created from variables of <code>X</code> plus <code>y</code>. Default value is <code>NULL</code>.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_selec">selec</code></td>
<td>
<p>a list with as many components as row-variables. If list length is 1 it is recycled for all row-variables. Every component of 'selec' is an expression that will be evaluated to select the individuals to be analyzed for every row-variable. Otherwise, a named list specifying 'selec' row-variables is applied. '.else' is a reserved name that defines the selection for the rest of the variables; if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is NA; all individuals are analyzed (no subsetting).</p>
</td></tr>
<tr><td><code id="descrTable_+3A_method">method</code></td>
<td>
<p>integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for continuous row-variables (for factor row-variables it is ignored). Possible values are: 1 - forces analysis as &quot;normal-distributed&quot;; 2 - forces analysis as &quot;continuous  non-normal&quot;; 3 - forces analysis as &quot;categorical&quot;;  and 4 - NA, which performs a Shapiro-Wilks  test to decide between normal or non-normal. Otherwise, a named vector specifying 'method' row-variables is applied. 
'.else' is a reserved name that defines the method for the rest of the variables; if no '.else' variable is defined, default value is applied. Default value is 1.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_timemax">timemax</code></td>
<td>
<p>double vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for 'Surv' class row-variables (for all other row-variables it is ignored).  This value indicates at which time the K-M probability is to be computed. Otherwise, a named vector specifying 'timemax' row-variables is applied. '.else' is a reserved name that defines the 'timemax' for the rest of the variables; if no '.else' variable is defined, default value is applied. Default value is NA; K-M probability is then computed at the median of observed times.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_alpha">alpha</code></td>
<td>
<p>double between 0 and 1. Significance threshold for the <code><a href="stats.html#topic+shapiro.test">shapiro.test</a></code> normality test for continuous row-variables. Default value is 0.05.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_min.dis">min.dis</code></td>
<td>
<p>an integer. If a non-factor row-variable contains less than 'min.dis' different values and 'method' argument is set to NA, then it will be converted to a factor. Default value is 5.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_max.ylev">max.ylev</code></td>
<td>
<p>an integer indicating the maximum number of levels of grouping variable ('y'). If 'y' contains more than 'max.ylev' levels, then the function 'compareGroups' produces an error. Default value is 5.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_max.xlev">max.xlev</code></td>
<td>
<p>an integer indicating the maximum number of levels when the row-variable is a factor. If the row-variable is a factor (or converted to a factor if it is a character, for example) and contains more than 'max.xlev' levels, then it is removed from the analysis and a warning is printed. Default value is 10.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_include.label">include.label</code></td>
<td>
<p>logical, indicating whether or not variable labels have to be shown in the results. Default value is TRUE</p>
</td></tr>
<tr><td><code id="descrTable_+3A_q1">Q1</code></td>
<td>
<p>double between 0 and 1, indicating the quantile to be displayed as the first number inside the square brackets in the bivariate table. To compute the minimum just type 0. Default value is 0.25 which means the first quartile.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_q3">Q3</code></td>
<td>
<p>double between 0 and 1, indicating the quantile to be displayed as the second number inside the square brackets in the bivariate table. To compute the maximum just type 1. Default value is 0.75 which means the third quartile.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_simplify">simplify</code></td>
<td>
<p>logical, indicating whether levels with no values must be removed for grouping variable and for row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_ref">ref</code></td>
<td>
<p>an integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. It only applies for categorical row-variables. Or a named vector specifying which row-variables 'ref' is applied (a reserved name is '.else' which defines the reference category for the rest of the variables); if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is 1.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_ref.no">ref.no</code></td>
<td>
<p>character specifying the name of the level to be the reference for Odds Ratio or Hazard Ratio. It is not case-sensitive. This is especially useful for yes/no variables. Default value is NA which means that category specified in 'ref' is the one selected to be the reference.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_fact.ratio">fact.ratio</code></td>
<td>
<p>a double vector with as many components as row-variables  indicating  the units for the HR / OR (note that it does not affect the descriptives).  If its length is 1 it is recycled for all row-variables. Otherwise, a named vector specifying 'fact.ratio' row-variables is applied. '.else' is a reserved name that defines the reference category for the rest of the variables;  if no '.else' variable is defined, default value is applied. Default value is 1.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_ref.y">ref.y</code></td>
<td>
<p>an integer indicating the reference category of y variable for computing the OR, when y is a binary factor. Default value is 1.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_p.corrected">p.corrected</code></td>
<td>
<p>logical, indicating whether p-values for pairwise comparisons must be corrected. It only applies when there is a grouping variable with more than 2 categories. Default value is TRUE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_compute.ratio">compute.ratio</code></td>
<td>
<p>logical, indicating whether Odds Ratio (for a binary response) or Hazard Ratio (for a time-to-event response) must be computed. Default value is TRUE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_include.miss">include.miss</code></td>
<td>
<p>logical, indicating whether to treat missing values as a new category for categorical variables. Default value is FALSE.</p>
</td></tr>   
<tr><td><code id="descrTable_+3A_oddsratio.method">oddsratio.method</code></td>
<td>
<p>Which method to compute the Odds Ratio. See 'method' argument from <code>oddsratio</code> (<code>epitools</code> package). Default value is &quot;midp&quot;.</p>
</td></tr>    
<tr><td><code id="descrTable_+3A_byrow">byrow</code></td>
<td>
<p>logical or NA. Percentage of categorical variables must be reported by rows (TRUE), by columns (FALSE) or by columns and rows to sum up 1 (NA). Default value is FALSE, which means that percentages are reported by columns (withing groups).</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_chisq.test.perm">chisq.test.perm</code></td>
<td>
<p>logical. It applies a permutation chi squared test (<code><a href="stats.html#topic+chisq.test">chisq.test</a></code>) instead of an exact Fisher test (<code><a href="stats.html#topic+fisher.test">fisher.test</a></code>). It only applies when expected count in some cells are lower than 5.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_chisq.test.b">chisq.test.B</code></td>
<td>
<p>integer. Number of permutation when computing permuted chi squared test for categorical variables. Default value is 2000.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_chisq.test.seed">chisq.test.seed</code></td>
<td>
<p>integer or NULL. Seed when performing permuted chi squared test for categorical variables. Default value is NULL which sets no seed. It is important to introduce some number different from NULL in order to reproduce the results when permuted chi-squared test is performed.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_date.format">Date.format</code></td>
<td>
<p>character indicating how the dates are shown. Default is &quot;d-mon-Y&quot;. See <code><a href="chron.html#topic+chron">chron</a></code> for more details.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_var.equal">var.equal</code></td>
<td>
<p>logical, indicating whether to consider equal variances when comparing means on normal distributed variables on more than two groups. If TRUE <code><a href="stats.html#topic+anova">anova</a></code> function is applied and <code><a href="stats.html#topic+oneway.test">oneway.test</a></code> otherwise. Default value is TRUE.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_conf.level">conf.level</code></td>
<td>
<p>double. Conficende level of confidence interval for means, medians, proportions or incidence, and hazard, odds and risk ratios. Default value is 0.95.</p>
</td></tr>   
<tr><td><code id="descrTable_+3A_surv">surv</code></td>
<td>
<p>logical. Compute survival (TRUE) or incidence (FALSE) for time-to-event row-variables. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_riskratio">riskratio</code></td>
<td>
<p>logical. Whether to compute Odds Ratio (FALSE) or Risk Ratio (TRUE). Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_riskratio.method">riskratio.method</code></td>
<td>
<p>Which method to compute the Odds Ratio. See 'method' argument from <code>riskratio</code> (<code>epitools</code> package). Default value is &quot;wald&quot;.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_compute.prop">compute.prop</code></td>
<td>
<p>logical. Compute proportions (TRUE) or percentages (FALSE) for cathegorical row-variables. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_lab.missing">lab.missing</code></td>
<td>
<p>character. Label for missing cathegory. Only applied when <code>include.missing = TRUE</code>. Default value is 'Missing'.</p>
</td></tr>   
<tr><td><code id="descrTable_+3A_p.trend.method">p.trend.method</code></td>
<td>
<p>Character indicating the name of test to use for p-value for trend. It only applies for numerical non-normal variables. Possible values are &quot;spearman&quot;, &quot;kendall&quot; or &quot;cuzick&quot;. Default value is &quot;spearman&quot;.</p>
</td></tr>
</table>
<p>Arguments from <code><a href="#topic+createTable">createTable</a></code> function:
</p>
<table role = "presentation">
<tr><td><code id="descrTable_+3A_hide">hide</code></td>
<td>
<p>a vector (or a list) with integers or characters with as many components as row-variables. If its length is 1 it is recycled for all row-variables. Each component specifies which category (the literal name of the category if it is a character, or the position if it is an integer) must be hidden and not shown. This argument only applies to categorical row-variables, and for continuous row-variables it is ignored. If NA, all categories are displayed. Or a named vector (or a named list) specifying which row-variables 'hide' is applied, and for the rest of row-variables default value is applied. Default value is NA.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_digits">digits</code></td>
<td>
<p>an integer vector with as many components as row-variables. If its length is 1 it is recycled for all row-variables. Each component specifies the number of significant decimals to be displayed. Or a named vector specifying which row-variables 'digits' is applied (a reserved name is '.else' which defines 'digits' for the rest of the variables); if no '.else' variable is defined, default value is applied for the rest of the variables. Default value is NA which puts the 'appropriate' number of decimals (see vignette for further details).</p>
</td></tr>
<tr><td><code id="descrTable_+3A_type">type</code></td>
<td>
<p>an integer that indicates whether absolute and/or relative frequencies are displayed: 1 - only relative frequencies; 2 or NA - absolute and relative frequencies in brackets; 3 - only absolute frequencies.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_show.p.overall">show.p.overall</code></td>
<td>
<p>logical indicating whether p-value of overall groups significance ('p.overall' column) is displayed or not. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.all">show.all</code></td>
<td>
<p>logical indicating whether the '[ALL]' column (all data without stratifying by groups) is displayed or not. Default value is FALSE if grouping variable is defined, and FALSE if there are no groups.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.p.trend">show.p.trend</code></td>
<td>
<p>logical indicating whether p-trend is displayed or not. It is always FALSE when there are less than 3 groups. If this argument is missing, there are more than 2 groups and the grouping variable is an ordered factor, then p-trend is displayed. By default, p-trend is not displayed, and it is displayed when there are more than 2 groups and the grouping variable is of class ordered-factor.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.p.mul">show.p.mul</code></td>
<td>
<p>logical indicating whether the pairwise (between groups) comparisons p-values are displayed or not. It is always FALSE when there are less than 3 groups. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.n">show.n</code></td>
<td>
<p>logical indicating whether number of individuals analyzed for each row-variable is displayed or not in the 'descr' table. Default value is FALSE and it is TRUE when there are no groups.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.ratio">show.ratio</code></td>
<td>
<p>logical indicating whether OR / HR is displayed or not. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_show.descr">show.descr</code></td>
<td>
<p>logical indicating whether descriptives (i.e. mean, proportions, ...) are displayed. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.ci">show.ci</code></td>
<td>
<p>logical indicating whether to show confidence intervals of means, medians, proporcions or incidences are displayed. If so, they are displayed between squared brackets. Default value is FALSE.</p>
</td></tr>  
<tr><td><code id="descrTable_+3A_hide.no">hide.no</code></td>
<td>
<p>character specifying the name of the level to be hidden for all categorical variables with 2 categories. It is not case-sensitive. The result is one row for the variable with only the name displayed and not the category. This is especially useful for yes/no variables. It is ignored for the categorical row-variables with 'hide' argument different from NA. Default value is NA which means that no category is hidden.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_digits.ratio">digits.ratio</code></td>
<td>
<p>The same as 'digits' argument but applied for the Hazard Ratio or Odds Ratio.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_show.p.ratio">show.p.ratio</code></td>
<td>
<p>logical indicating whether p-values corresponding to each Hazard Ratio / Odds Ratio are shown.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_digits.p">digits.p</code></td>
<td>
<p>integer indicating the number of decimals displayed for all p-values. Default value is 3.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_sd.type">sd.type</code></td>
<td>
<p>an integer that indicates how standard deviation is shown: 1 - mean (SD), 2 -  mean ? SD.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_q.type">q.type</code></td>
<td>
<p>a vector with two integer components. The first component refers to the type of brackets to be displayed for non-normal row-variables (1 - squared and 2 - rounded), while the second refers to the percentile separator (1 - ';', 2 - ',' and 3 - '-'. Default value is c(1, 1).</p>
</td></tr>
<tr><td><code id="descrTable_+3A_extra.labels">extra.labels</code></td>
<td>
<p>character vector of 4 components corresponding to key legend to be appended to normal, non-normal, categorical or survival row-variables labels. Default value is NA which appends no extra key. If it is set to <code>c("","","","")</code>, &quot;Mean (SD)&quot;, &quot;Median [25th; 75th]&quot;, &quot;N (%)&quot; and &quot;Incidence at time=timemax&quot; are appended (see argument <code>timemax</code> from <code>compareGroups</code> function.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_all.last">all.last</code></td>
<td>
<p>logical. Descriptives of the whole sample is placed after the descriptives by groups. Default value is FALSE which places the descriptives of whole cohort at first.</p>
</td></tr> 
<tr><td><code id="descrTable_+3A_lab.ref">lab.ref</code></td>
<td>
<p>character. String shown for reference category. &quot;Ref.&quot; as default value.</p>
</td></tr>
<tr><td><code id="descrTable_+3A_stars">stars</code></td>
<td>
<p>logical, indicating whether to append stars beside p-values; '**': p-value &lt; 0.05, '*' 0.05 &lt;= p-value &lt; 0.1; &rdquo; p-value &gt;=0.1. Default value is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'createTable' (see <code><a href="#topic+createTable">createTable</a></code>).
</p>
<p>So, all methods implemented for createTable class objects can be applied (such as plot, '[', etc.).
</p>


<h3>Note</h3>

<p>The use of descrTable function makes easier to build the table (it only needs one line), it may be preferable to build the descriptive table in two steps when computing descriptives and p-values takes some time: first use <code>compareGroups</code> function to store the descriptives and p-values in an object, and then apply <code>createTable</code> to the this object. The two steps strategy saves time since descriptives and p-values are not recomputed every time it is desired to costumize the descriptive table (number of digits, etc.).
</p>


<h3>References</h3>

<p>Isaac Subirana, Hector Sanz, Joan Vila (2014). Building Bivariate Tables: The compareGroups Package for R. Journal of Statistical Software, 57(12), 1-16. URL <a href="https://www.jstatsoft.org/v57/i12/">https://www.jstatsoft.org/v57/i12/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups)

# load REGICOR data
data(regicor)

# perform descriptives by year and build the table.
# note the use of arguments from compareGroups (formula and data set) and
# arguments from createTable (hide.no and show.p.mul)
descrTable(year ~ ., regicor, hide.no="no", show.p.mul=TRUE)

</code></pre>

<hr>
<h2 id='export2csv'>
Exporting descriptives table to plain text (CSV) format
</h2><span id='topic+export2csv'></span>

<h3>Description</h3>

<p>This function takes the result of <code>createTable</code> and exports the tables to plain text (CSV) format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2csv(x, file, which.table="descr", sep=",", nmax = TRUE, nmax.method = 1, 
           header.labels = c(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2csv_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_file">file</code></td>
<td>
<p>file where table in CSV format will be written. Also, another file with the extension '_appendix' is written with the available data table.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_which.table">which.table</code></td>
<td>
<p>character indicating  which table is printed. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), exporting descriptives by groups table, available data table or both tables, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_sep">sep</code></td>
<td>
<p>character. The variable separator, same as 'sep' argument from <code><a href="utils.html#topic+write.table">write.table</a></code>. Default value is ','.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="export2csv_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>  
<tr><td><code id="export2csv_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code><a href="utils.html#topic+write.table">write.table</a></code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The default way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail').
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+export2md">export2md</a></code>, <code><a href="#topic+export2word">export2word</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(compareGroups)
data(regicor)
res &lt;- compareGroups(sex ~. -id-todeath-death-tocv-cv, regicor)
export2csv(createTable(res, hide.no = 'n'), file=tempfile(fileext=".csv"))

## End(Not run)

</code></pre>

<hr>
<h2 id='export2html'>
Exporting descriptives table to HTML format
</h2><span id='topic+export2html'></span>

<h3>Description</h3>

<p>This function takes the result of <code>createTable</code> and exports the tables to HTML format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2html(x, file, which.table="descr", nmax = TRUE, nmax.method = 1, 
            header.labels = c(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2html_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="export2html_+3A_file">file</code></td>
<td>
<p>file where table in HTML format will be written. Also, another file with the extension '_appendix' is written with the available data table. If missing, the HTML code is returned.</p>
</td></tr>
<tr><td><code id="export2html_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), exporting descriptives by groups table, availability data table or both tables, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2html_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2html_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="export2html_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>   
<tr><td><code id="export2html_+3A_...">...</code></td>
<td>
<p>currently ignored.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The default way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail').
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2md">export2md</a></code>, <code><a href="#topic+export2word">export2word</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(compareGroups)
data(regicor)
res &lt;- compareGroups(sex ~. -id-todeath-death-tocv-cv, regicor)
export2html(createTable(res, hide.no = 'n'), file=tempfile(fileext=".html"))

## End(Not run)

</code></pre>

<hr>
<h2 id='export2latex'>
Exporting descriptives table to LaTeX format
</h2><span id='topic+export2latex'></span><span id='topic+export2latex.createTable'></span><span id='topic+export2latex.cbind.createTable'></span>

<h3>Description</h3>

<p>This function takes the result of <code>createTable</code> and exports the tables to LaTeX format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2latex(x, ...)
## S3 method for class 'createTable'
export2latex(x, file, which.table = 'descr', size = 'same', 
    nmax = TRUE, nmax.method = 1, header.labels = c(), caption = NULL, 
    loc.caption = 'top', label = NULL, landscape = NA, colmax = 10, ...)
## S3 method for class 'cbind.createTable'
export2latex(x, file, which.table = 'descr', size = 'same', 
    nmax = TRUE, nmax.method = 1, header.labels = c(), caption = NULL, 
    loc.caption = 'top', label = NULL, landscape = NA, colmax = 10, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2latex_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_file">file</code></td>
<td>
<p>Name of file where the resulting code should be saved. If file is missing, output is displayed on screen. Also, another file with the extension '_appendix' is written with the available data table.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is exported. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), exporting descriptives by groups table, availability data table or both tables, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_size">size</code></td>
<td>
<p>character indicating the size of the table elements. Possible values are: 'tiny', 'scriptsize', 'footnotesize', 'small', 'normalsize', 'large', 'Large', 'LARGE','huge', 'Huge' or 'same' (partial matching allowed). Default value is 'same' which means that font size of the table is the same as specified in the main LaTeX document.</p>
</td></tr>  
<tr><td><code id="export2latex_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>   
<tr><td><code id="export2latex_+3A_caption">caption</code></td>
<td>
<p>character specifying the table caption for descriptives and available data table. If which.table='both' the first element of 'caption' will be assigned to descriptives table and the second to available data table. If it is set to &quot;&quot;, no caption is inserted. Default value is NULL, which writes 'Summary descriptives table by groups of 'y&rdquo; for descriptives table and 'Available data by groups of 'y&rdquo; for the available data table. </p>
</td></tr>
<tr><td><code id="export2latex_+3A_label">label</code></td>
<td>
<p>character specifying the table label for descriptives and available data table. This may be useful to cite the tables elsewhere in the LaTeX document. If which.table='both' the first element of 'label' will be assigned to descriptives table and the second to available data table. Default value is NULL, which assigns no label to the table/s. </p>
</td></tr>
<tr><td><code id="export2latex_+3A_loc.caption">loc.caption</code></td>
<td>
<p>character specifying the table caption location. Possible values are 'top' or 'bottom' (partial matching allowed). Default value is 'top'.</p>
</td></tr>
<tr><td><code id="export2latex_+3A_landscape">landscape</code></td>
<td>
<p>logical indicating whether the table must be placed in landscape, or NA that places the table in landscape when there are more than 'colmax' columns. Default value is NA.</p>
</td></tr>  
<tr><td><code id="export2latex_+3A_colmax">colmax</code></td>
<td>
<p>integer indicating the maximum number of columns to make the table not to be placed in landscape. This argument is only applied when 'landscape' argument is NA. Default value is 10.</p>
</td></tr>    
<tr><td><code id="export2latex_+3A_...">...</code></td>
<td>
<p>currently ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of two possible components corresponding to the code of 'descr' table and 'avail' table. Each component of the list is a character corresponding to the LaTeX code of these tables which can be helpful for post-processing.
</p>


<h3>Note</h3>

<p>The table is created in LaTeX language using the longtable environment. Therefore, it is necessary to type <code>\includepackage{longtable}</code> in the preamble of the LaTeX main document where the table code is inserted. Also, it it necessary to include the 'multirow' LaTeX package.  \
</p>
<p>The way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail'). \
</p>
<p>When 'landscape' argument is TRUE or there are more than 'colmax' columns and 'landscape' is set to NA, LaTeX package 'lscape' must be loaded in the tex document.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2md">export2md</a></code>, <code><a href="#topic+export2word">export2word</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(compareGroups)
data(regicor)
res &lt;- compareGroups(sex ~. -id-todeath-death-tocv-cv, regicor)
export2latex(createTable(res, hide.no = 'n'), file=tempfile(fileext=".tex"))

## End(Not run)

</code></pre>

<hr>
<h2 id='export2md'>
Exporting descriptives table to Markdown format
</h2><span id='topic+export2md'></span>

<h3>Description</h3>

<p>This function takes the result of <code>createTable</code> and exports the tables to markdown format. It may be useful when inserting R code chunks in a Markdown file (.Rmd).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2md(x, which.table = "descr", nmax = TRUE, nmax.method = 1, header.labels = c(),
          caption = NULL, format = "html", width = Inf, strip = FALSE, 
          first.strip = FALSE, background = "#D2D2D2", size = NULL, landscape=FALSE, 
          header.background=NULL, header.color=NULL, position="center", ...)

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2md_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr' or 'avail'(partial matching allowed), exporting descriptives by groups table or availability data table, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2md_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="export2md_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>   
<tr><td><code id="export2md_+3A_caption">caption</code></td>
<td>
<p>character specifying the table caption for descriptives and available data table. If which.table='both' the first element of 'caption' will be assigned to descriptives table and the second to available data table. If it is set to &quot;&quot;, no caption is inserted. Default value is NULL, which writes 'Summary descriptives table by groups of 'y&rdquo; for descriptives table and 'Available data by groups of 'y&rdquo; for the available data table. </p>
</td></tr>
<tr><td><code id="export2md_+3A_format">format</code></td>
<td>
<p>character with three options: 'html', 'latex' or 'markdown'. If missing, it tries to guess the default options of Rmarkdown file in which the table in inserted, or html if it is not in a Rmarkdown file or format not specified.</p>
</td></tr>
<tr><td><code id="export2md_+3A_width">width</code></td>
<td>
<p>character string to specify the width of first column of descriptive table. It is ignored when exporting to Word. Default value is <code>Inf</code> which makes the first column to autoadjust to variable names. Other examples are '10cm', '3in' or '30em'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_strip">strip</code></td>
<td>
<p>logical. It shadows table lines corresponding to each variable.</p>
</td></tr>
<tr><td><code id="export2md_+3A_first.strip">first.strip</code></td>
<td>
<p>logical. It determines whether to shadow the first variable (TRUE) or the second (FALSE). It only applies when <code>strip</code> argument is true.</p>
</td></tr>
<tr><td><code id="export2md_+3A_background">background</code></td>
<td>
<p>color code in HEX format for shadowed lines. You can use <code>rgb</code> function to convert red, green and blue to HEX code. Default color is '#D2D2D2'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_size">size</code></td>
<td>
<p>numeric. Size of descriptive table. Default value is NULL which creates the table in default size.</p>
</td></tr>
<tr><td><code id="export2md_+3A_landscape">landscape</code></td>
<td>
<p>logical. It determines whether to place the table in landscape (horizontal) format. It only applies when format is 'latex'. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="export2md_+3A_header.background">header.background</code></td>
<td>
<p>color character for table header or 'NULL'. Default value is 'NULL'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_header.color">header.color</code></td>
<td>
<p>color character for table header text. Default color is 'NULL'.</p>
</td></tr>
<tr><td><code id="export2md_+3A_position">position</code></td>
<td>
<p>character specifying the table location. Possible values are 'left', 'center', 'right', 'float_left' and 'float_right'. It only applies when compiling to HTML or PDF. Default value is 'center'. See <code><a href="kableExtra.html#topic+kable_styling">kable_styling</a></code> position argument for more info.</p>
</td></tr>
<tr><td><code id="export2md_+3A_...">...</code></td>
<td>
<p>arguments passed to <code><a href="knitr.html#topic+kable">kable</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It does not return anything, but the Markdown code to generate the descriptive or available table is printed.
</p>


<h3>Note</h3>

<p>The way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail').
</p>
<p>Stratified tables, i.e. <code>cbind.createTable</code> class, are not supported when creating a Word document.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+export2word">export2word</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

---
title: "Report"
output:
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
```

```{r}
library(compareGroups)
data(regicor)
res &lt;- compareGroups(year~., regicor)
restab &lt;- createTable(res)
```

## Report section

The following table contains descriptives of **REGICOR** data

```{r}
export2md(restab, strip = TRUE, first.strip = TRUE)
```

## End(Not run)

</code></pre>

<hr>
<h2 id='export2pdf'>
Exports tables to PDF files.
</h2><span id='topic+export2pdf'></span>

<h3>Description</h3>

<p>This function creates automatically a PDF with the table. Also, the LaTeX code is stored in the specified file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2pdf(x, file, which.table="descr", nmax=TRUE, header.labels=c(), caption=NULL,   
  width=Inf, strip=FALSE, first.strip=FALSE, background="#D2D2D2", size=NULL,  
  landscape=FALSE, numcompiled=2, header.background=NULL, header.color=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2pdf_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable' or that inherits it.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_file">file</code></td>
<td>
<p>character specifying the PDF file resulting after compiling the LaTeX code corresponding to the table specified in the 'x' argument. LaTeX code is also stored in the same folder with the same name but .tex extension. When 'compile' argument is FALSE, only .tex file is saved.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), printing descriptives by groups table, availability data table or both tables, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_header.labels">header.labels</code></td>
<td>
<p>a character named vector with 'all', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' components indicating the label for '[ALL]', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' (available data), respectively. Default is a zero length vector which makes no changes, i.e. '[ALL]', 'p.overall', 'p.trend', 'ratio', 'p.ratio' and 'N' labels appear for descriptives of entire cohort, global p-value, p-value for trend, HR/OR and p-value of each HR/OR and available data, respectively.</p>
</td></tr>  
<tr><td><code id="export2pdf_+3A_caption">caption</code></td>
<td>
<p>character specifying the table caption for descriptives and available data table. If which.table='both' the first element of 'caption' will be assigned to descriptives table and the second to available data table. If it is set to &quot;&quot;, no caption is inserted. Default value is NULL, which writes 'Summary descriptives table by groups of 'y&rdquo; for descriptives table and 'Available data by groups of 'y&rdquo; for the available data table. </p>
</td></tr>
<tr><td><code id="export2pdf_+3A_width">width</code></td>
<td>
<p>character string to specify the width of first column of descriptive table. Default value is <code>Inf</code> which makes the first column to autoadjust to variable names. Other examples are '10cm', '3in' or '30em'.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_strip">strip</code></td>
<td>
<p>logical. It shadows table lines corresponding to each variable.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_first.strip">first.strip</code></td>
<td>
<p>logical. It determines whether to shadow the first variable (TRUE) or the second (FALSE). It only applies when <code>strip</code> argument is true.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_background">background</code></td>
<td>
<p>color code in HEX format for shadowed lines. You can use <code>rgb</code> function to convert red, green and blue to HEX code. Default color is '#D2D2D2'.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_size">size</code></td>
<td>
<p>numeric. Size of descriptive table. Default value is NULL which creates the table in default size.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_landscape">landscape</code></td>
<td>
<p>logical. It determines whether to place the table in landscape (horizontal) format. It only applies when format is 'latex'. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_numcompiled">numcompiled</code></td>
<td>
<p>integer. Number of times LaTeX code is compiled. When creating the table it may be necessary to execute the code several times in order to fit the columns widths. By default it is compiled twice.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_header.background">header.background</code></td>
<td>
<p>color character for table header or 'NULL'. Default value is 'NULL'.</p>
</td></tr>
<tr><td><code id="export2pdf_+3A_header.color">header.color</code></td>
<td>
<p>color character for table header text. Default color is 'NULL'.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>To make the .tex file be compiled, some LaTeX compiler such as Miktex must be installed. Also, the tex file must include the following LaTeX packages:
</p>

<ul>
<li><p> longtable<br />
</p>
</li>
<li><p> multirow<br />
</p>
</li>
<li><p> multicol<br />
</p>
</li>
<li><p> booktabs<br />
</p>
</li>
<li><p> xcolor<br />
</p>
</li>
<li><p> colortbl<br />
</p>
</li>
<li><p> lscape<br />
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+export2md">export2md</a></code>, <code><a href="#topic+export2word">export2word</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

require(compareGroups)
data(regicor)

 # example on an ordinary table
res &lt;- createTable(compareGroups(year ~ . -id, regicor), hide = c(sex=1), hide.no = 'no')
export2pdf(res, file=tempfile(fileext=".pdf"), size="small")


## End(Not run)

</code></pre>

<hr>
<h2 id='export2word'>
Exports tables to Word files.
</h2><span id='topic+export2word'></span>

<h3>Description</h3>

<p>This function creates automatically a Word file with the table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2word(x, file, which.table="descr", nmax=TRUE, header.labels=c(), 
            caption=NULL, strip=FALSE, first.strip=FALSE, background="#D2D2D2", 
            size=NULL, header.background=NULL, header.color=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2word_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable' or that inherits it.</p>
</td></tr>
<tr><td><code id="export2word_+3A_file">file</code></td>
<td>
<p>character specifying the word file (.doc or .docx) resulting after compiling the Markdown code corresponding to the table specified in the 'x' argument.</p>
</td></tr>
<tr><td><code id="export2word_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr' or 'avail'(partial matching allowed), exporting descriptives by groups table or availability data table, respectively. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2word_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2word_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>   
<tr><td><code id="export2word_+3A_caption">caption</code></td>
<td>
<p>character specifying the table caption for descriptives and available data table. If which.table='both' the first element of 'caption' will be assigned to descriptives table and the second to available data table. If it is set to &quot;&quot;, no caption is inserted. Default value is NULL, which writes 'Summary descriptives table by groups of 'y&rdquo; for descriptives table and 'Available data by groups of 'y&rdquo; for the available data table. </p>
</td></tr>
<tr><td><code id="export2word_+3A_strip">strip</code></td>
<td>
<p>logical. It shadows table lines corresponding to each variable.</p>
</td></tr>
<tr><td><code id="export2word_+3A_first.strip">first.strip</code></td>
<td>
<p>logical. It determines whether to shadow the first variable (TRUE) or the second (FALSE). It only applies when <code>strip</code> argument is true.</p>
</td></tr>
<tr><td><code id="export2word_+3A_background">background</code></td>
<td>
<p>color code in HEX format for shadowed lines. You can use <code>rgb</code> function to convert red, green and blue to HEX code. Default color is '#D2D2D2'.</p>
</td></tr>
<tr><td><code id="export2word_+3A_size">size</code></td>
<td>
<p>numeric. Size of descriptive table. Default value is NULL which creates the table in default size.</p>
</td></tr>
<tr><td><code id="export2word_+3A_header.background">header.background</code></td>
<td>
<p>color character for table header or 'NULL'. Default value is 'NULL'.</p>
</td></tr>
<tr><td><code id="export2word_+3A_header.color">header.color</code></td>
<td>
<p>color character for table header text. Default color is 'NULL'.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Word file is created after compiling Markdown code created by <code><a href="#topic+export2md">export2md</a></code>. To compile it it calls <code><a href="rmarkdown.html#topic+render">render</a></code> function which requires pandoc to be installed.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+export2md">export2md</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

require(compareGroups)
data(regicor)

 # example on an ordinary table
res &lt;- createTable(compareGroups(year ~ . -id, regicor), hide = c(sex=1), hide.no = 'no')
export2word(res, file = tempfile(fileext=".docx"))


## End(Not run)

</code></pre>

<hr>
<h2 id='export2xls'>
Exporting descriptives table to Exel format (.xlsx or .xls)
</h2><span id='topic+export2xls'></span>

<h3>Description</h3>

<p>This function takes the result of <code>createTable</code> and exports the tables to Excel format (.xlsx or .xls).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export2xls(x, file, which.table="descr", nmax=TRUE, nmax.method=1, header.labels=c())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export2xls_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="export2xls_+3A_file">file</code></td>
<td>
<p>file where table in Excel format will be written.</p>
</td></tr>
<tr><td><code id="export2xls_+3A_which.table">which.table</code></td>
<td>
<p>character indicating which table is printed. Possible values are 'descr', 'avail' or 'both' (partial matching allowed), exporting descriptives by groups table, availability data table or both tables, respectively. In the latter case ('both'), two sheets are built, one for each table. Default value is 'descr'.</p>
</td></tr>
<tr><td><code id="export2xls_+3A_nmax">nmax</code></td>
<td>
<p>logical, indicating whether to show the number of subjects with at least one valid value across all row-variables. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="export2xls_+3A_nmax.method">nmax.method</code></td>
<td>
<p>integer with two possible values: 1-number of observation with valid values in at least one row-variable; 2-total number of observations or rows in the data set or in the group. Default value is 1.</p>
</td></tr>
<tr><td><code id="export2xls_+3A_header.labels">header.labels</code></td>
<td>
<p>see the 'header.labels' argument from <code><a href="#topic+createTable">createTable</a></code>.</p>
</td></tr>   
</table>


<h3>Note</h3>

<p>The way to compute the 'N' shown in the bivariate table header, controlled by 'nmax' argument, has been changed from previous versions (&lt;1.3). In the older versions 'N' was computed as the maximum across the cells withing each column (group) from the 'available data' table ('avail').
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2pdf">export2pdf</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2md">export2md</a></code>, <code><a href="#topic+export2word">export2word</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(compareGroups)
data(regicor)
res &lt;- compareGroups(sex ~. -id-todeath-death-tocv-cv, regicor)
export2xls(createTable(res, hide.no = 'n'), file=tempfile(fileext=".xlsx"))

## End(Not run)

</code></pre>

<hr>
<h2 id='getResults'>
Easily retrieve summary data as R-objects (matrices and vectors).
</h2><span id='topic+getResults'></span>

<h3>Description</h3>

<p>This functions excratcs specific results (descriptives, p-values, Odds-Ratios / Hazard-Ratios, ...) from a compareGroups object as matrix or vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResults(obj, what = "descr")

</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getResults_+3A_obj">obj</code></td>
<td>
<p>an object of class 'compareGroups' or 'createTable'</p>
</td></tr>
<tr><td><code id="getResults_+3A_what">what</code></td>
<td>
<p>character indicating which results are to be retrieved: decriptives, p-value, p-trend, pairwise p-values, or Odds-Ratios / Hazard-Ratios. Possible values are: &quot;descr&quot;, &quot;p.overall&quot;, &quot;p.trend&quot;, &quot;p.mul&quot; and &quot;ratio&quot;. Default value is &quot;descr&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>what = "descr"</code></td>
<td>
<p>An array or matrix with as many columns as variables/categories and seven columns indicating all possible descriptive statistics (mean, sd, median, Q1, Q3, absolute and relative frequencies). When different groups are analysed, the 3rd dimension of the array corresponds to the groups. Otherwise, the result will be a matrix with no 3rd dimension.</p>
</td></tr>
<tr><td><code>what = "p.overall"</code></td>
<td>
<p>A vector whose elevements are the p-value for each analysed variable.</p>
</td></tr>
<tr><td><code>what = "p.trend"</code></td>
<td>
<p>A vector whose elevements are the p-trend for each analysed variable.</p>
</td></tr> 
<tr><td><code>what = "p.mul"</code></td>
<td>
<p>A matrix with pairwise p-values where rows correspond to the analysed variables and columns to each pair of groups.</p>
</td></tr>
<tr><td><code>what = "ratio"</code></td>
<td>
<p>A matrix with as many rows as variables/categorieswith and 4 columns corresponding to the OR/HR, confidence interval and p-value.</p>
</td></tr> 
</table>


<h3>Note</h3>

<p>For descriptives, NA is placed for descriptives not appropiate for the variable. For example columns corresponding to frequencies for continuous variables will be NA.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(compareGroups)
data(regicor)
res&lt;-compareGroups(sex ~ . ,regicor,method=c(triglyc=2))
# retrieve descriptives
getResults(res)
# retrieve OR and their corresponding p-values
getResults(res,what="ratio")
</code></pre>

<hr>
<h2 id='missingTable'>
Table of missingness counts by groups.
</h2><span id='topic+missingTable'></span>

<h3>Description</h3>

<p>This functions returns a table with the non-available frequencies from a already build bivariate table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>missingTable(obj,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="missingTable_+3A_obj">obj</code></td>
<td>
<p>either a 'compareGroups' or 'createTable' object.</p>
</td></tr>
<tr><td><code id="missingTable_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code>createTable</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'createTable'. For further details, see 'value' section of <code>createTable</code> help file.
</p>


<h3>Note</h3>

<p>This function returns an object of class 'createTable', and therefore all methods implemented for 'createTable' objects can be applied, except the 'update' method.  
</p>
<p>All arguments of <code>createTable</code> can be passed throught '...' argument, except 'hide.no' argument which is fixed inside the code and cannot be changed.
</p>
<p>This function cannot be applied to stratified tables, i.e. 'rbind.createTable' and 'cbind.createTable'. If stratified missingness table is desired, apply this function first to each table and then use <code>cbind.createTable</code> or/and <code>rbind.createTable</code> functions to combine exactly in the same way as 'createTable' objects. See 'example' section below.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

require(compareGroups)

# load regicor data
data(regicor)

# table of descriptives by recruitment year
res &lt;- compareGroups(year ~ age + sex + smoker + sbp + histhtn + 
         chol + txchol + bmi + phyact + pcs + death, regicor)
restab &lt;- createTable(res, hide.no = "no")

# missingness table
missingTable(restab,type=1)


## Not run: 

# also create the missing table from a compareGroups object
miss &lt;- missingTable(res)
miss

# some methods that works for createTable objects also works for objects 
#   computed by missTable function.
miss[1:4]
varinfo(miss)
plot(miss)

#... but update methods cannot be applied (this returns an error).
update(miss,type=2) 


## End(Not run)


</code></pre>

<hr>
<h2 id='padjustCompareGroups'>Update p values according multiple comparisons</h2><span id='topic+padjustCompareGroups'></span>

<h3>Description</h3>

<p>Given a compareGroups object, returns their p-values adjusted using one of several methods (stats::p.adjust)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>padjustCompareGroups(object_compare, p = "p.overall", method = "BH")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="padjustCompareGroups_+3A_object_compare">object_compare</code></td>
<td>
<p>object of class <code>compareGroups</code></p>
</td></tr>
<tr><td><code id="padjustCompareGroups_+3A_p">p</code></td>
<td>
<p>character string. Specify which p-value must be corrected. Possible values are 'p.overall' and 'p.trend' (default: 'p.overall')</p>
</td></tr>
<tr><td><code id="padjustCompareGroups_+3A_method">method</code></td>
<td>
<p>Correction method, a character string. Can be abbreviated (see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>compareGroups class with corrected p-values
</p>


<h3>Author(s)</h3>

<p>Jordi Real &lt;jordireal&lt;at&gt;gmail.com&gt;</p>


<h3>Examples</h3>

<pre><code class='language-R'># Define simulated data 
set.seed(123)
N_obs&lt;-100
N_vars&lt;-50 
data&lt;-matrix(rnorm(N_obs*N_vars), N_obs, N_vars) 

sim_data&lt;-data.frame(data,Y=rbinom(N_obs,1,0.5))

# Execute compareGroups
res&lt;-compareGroups(Y~.,data=sim_data)
res

# update p values
res_adjusted&lt;-padjustCompareGroups(res)
res_adjusted

# update p values using FDR method
res_adjusted&lt;-padjustCompareGroups(res, method ="fdr")
res_adjusted

</code></pre>

<hr>
<h2 id='printTable'>
'Nice' table format.
</h2><span id='topic+printTable'></span>

<h3>Description</h3>

<p>This functions prints a table on the console in a 'nice' format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printTable(obj, row.names = TRUE, justify = 'right')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="printTable_+3A_obj">obj</code></td>
<td>
<p>an object of class 'data.frame' or 'matrix'. It must be at least two columns, the first columns is considered as the 'row.names' and is left justified (if the 'row.names' argument is set to TRUE), while the rest of the columns are right justified.</p>
</td></tr>
<tr><td><code id="printTable_+3A_row.names">row.names</code></td>
<td>
<p>logical indicating whether the first column or variable is treated as a 'row.names' column and must be left-justified. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="printTable_+3A_justify">justify</code></td>
<td>
<p>character as 'justify' argument from <code><a href="base.html#topic+format">format</a></code> function. It applies to all columns of the data.frame or matrix when 'row.names' argument is FALSE or all columns excluding the first one otherwise. Default value is 'right'.</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>No object is returned.
</p>


<h3>Note</h3>

<p>This function may be usefull when printing a table with some results with variables as the first column and a header. It adds 'nice' lines to highlight the header and also the bottom of the table. 
</p>
<p>It has been used to print 'compareSNPs' objects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareSNPs">compareSNPs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups)

data(regicor)

# example of the coefficients table from a linear regression
model &lt;- lm(chol ~ age + sex + bmi, regicor)
results &lt;- coef(summary(model))
results &lt;- cbind(Var = rownames(results), round(results, 4))
printTable(results)

# or visualize the first rows of the iris data frame. 
# In this example, the first column is not treated as a row.names column and it is right justified.
printTable(head(iris), FALSE)

# the same example with columns centered
printTable(head(iris), FALSE, 'centre')


</code></pre>

<hr>
<h2 id='radiograph'>
Lists the values in the data set.
</h2><span id='topic+radiograph'></span>

<h3>Description</h3>

<p>This function creates a report of raw data in your data set. For each variable an ordered list of the unique entries (read as strings), useful for checking for input errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>radiograph(file, header = TRUE, save=FALSE, out.file="", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="radiograph_+3A_file">file</code></td>
<td>
<p>character specifying the file where the data set is located.</p>
</td></tr>
<tr><td><code id="radiograph_+3A_header">header</code></td>
<td>
<p>see <code><a href="utils.html#topic+read.table">read.table</a></code>.</p>
</td></tr>
<tr><td><code id="radiograph_+3A_save">save</code></td>
<td>
<p>logical indicating whether output should be stored in a file (TRUE) or printed on the console (FALSE). Default is FALSE.</p>
</td></tr>
<tr><td><code id="radiograph_+3A_out.file">out.file</code></td>
<td>
<p>character specifying the file where the results are to be output. It only applies when 'save' argument is set to TRUE.</p>
</td></tr>
<tr><td><code id="radiograph_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="utils.html#topic+read.table">read.table</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gavin Lucas (gavin.lucas&lt;at&gt;cleargenetics.com) <br />
</p>
<p>Isaac Subirana (isubirana&lt;at&gt;imim.es)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+report">report</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

require(compareGroups)

# read example data of regicor in plain text format with variables separated by '\t'.
datafile &lt;- system.file("exdata/regicor.txt", package="compareGroups")
radiograph(datafile)


## End(Not run)

</code></pre>

<hr>
<h2 id='regicor'>
REGICOR cross-sectional data
</h2><span id='topic+regicor'></span>

<h3>Description</h3>

<p>These data come from 3 different cross-sectional surveys of individuals representative of the population from a north-west Spanish province (Girona), REGICOR study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(regicor)</code></pre>


<h3>Format</h3>

<p>A data frame with 2294 observations on the following 21 variables:
</p>

<dl>
<dt><code>id</code></dt><dd><p>Individual id</p>
</dd>
<dt><code>year</code></dt><dd><p>a factor with levels <code>1995</code> <code>2000</code> <code>2005</code>. Recruitment year</p>
</dd>
<dt><code>age</code></dt><dd><p>Patient age at recruitment date</p>
</dd>
<dt><code>sex</code></dt><dd><p>a factor with levels <code>male</code> <code>female</code>. Sex</p>
</dd>
<dt><code>smoker</code></dt><dd><p>a factor with levels <code>Never smoker</code> <code>Current or former &lt; 1y</code> <code>Never or former &gt;= 1y</code>. Smoking status</p>
</dd>
<dt><code>sbp</code></dt><dd><p>Systolic blood pressure</p>
</dd>
<dt><code>dbp</code></dt><dd><p>Diastolic blood pressure</p>
</dd>
<dt><code>histhtn</code></dt><dd><p>a factor with levels <code>Yes</code> <code>No</code>. History of hypertension</p>
</dd>
<dt><code>txhtn</code></dt><dd><p>a factor with levels <code>No</code> <code>Yes</code>. Hypertension (HTN) treatment</p>
</dd>
<dt><code>chol</code></dt><dd><p>Total cholesterol (mg/dl)</p>
</dd>
<dt><code>hdl</code></dt><dd><p>HDL cholesterol (mg/dl)</p>
</dd>
<dt><code>triglyc</code></dt><dd><p>Triglycerides (mg/dl)</p>
</dd>
<dt><code>ldl</code></dt><dd><p>LDL cholesterol (mg/dl)</p>
</dd>
<dt><code>histchol</code></dt><dd><p>a factor with levels <code>Yes</code> <code>No</code>. History of hypercholesterolemia</p>
</dd>
<dt><code>txchol</code></dt><dd><p>a factor with levels <code>No</code> <code>Yes</code>. Cholesterol treatment</p>
</dd>
<dt><code>height</code></dt><dd><p>Height (cm)</p>
</dd>
<dt><code>weight</code></dt><dd><p>Weight (Kg)</p>
</dd>
<dt><code>bmi</code></dt><dd><p>Body mass index</p>
</dd>
<dt><code>phyact</code></dt><dd><p>Physical activity (Kcal/week)</p>
</dd>
<dt><code>pcs</code></dt><dd><p>Physical component summary</p>
</dd>
<dt><code>mcs</code></dt><dd><p>Mental component summary</p>
</dd>
<dt><code>death</code></dt><dd><p>a factor with levels <code>No</code> <code>Yes</code>. Overall death</p>
</dd>
<dt><code>todeath</code></dt><dd><p>Days to overall death or end of follow-up</p>
</dd>
<dt><code>cv</code></dt><dd><p>a factor with levels <code>No</code> <code>Yes</code>. Cardiovascular event</p>
</dd>
<dt><code>tocv</code></dt><dd><p>Days to cardiovascular event or end of follow-up</p>
</dd>    
</dl>



<h3>Details</h3>

<p>The variables collected in the REGICOR study were mainly cardiovascular risk factors (hundreds of variables were collected in the different questionnaires and blood measurements), but the variables present in this data set are just a few of them. Also, for reasons of confidentiality, the individuals in this data set are a 30% approx. random subsample of the original one. <br /> <br />
Each variable of this data.frame contains label describing them in the attribute &quot;label&quot;. <br /> <br />
For more information, see the vignette.
</p>


<h3>Note</h3>

<p>Variables <code>death</code>, <code>todeath</code>, <code>cv</code>, <code>tocv</code> are not real but they have been simulated at random to complete the data example with some time-to-event variables.
</p>


<h3>Source</h3>

<p>For reasons of confidentiality, the whole data set is not publicly available. For more information about the study these data come from, visit <code>www.regicor.org</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(compareGroups)
data(regicor)
summary(regicor)
</code></pre>

<hr>
<h2 id='report'>
Report of descriptive tables and plots.
</h2><span id='topic+report'></span>

<h3>Description</h3>

<p>This function creates automatically a PDF with the descriptive table as well as availability data and all plots. This file is structured and indexed in the way that the user can navigate through all tables and figures along the document.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report(x, file, fig.folder, compile = TRUE, openfile = FALSE, title = "Report", 
       author, date, perc=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="report_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'.</p>
</td></tr>
<tr><td><code id="report_+3A_file">file</code></td>
<td>
<p>character specifying the PDF file resulting after compiling the LaTeX code of report. LaTeX code is also stored in the same folder with the same name but .tex extension. When 'compile' argument is FALSE, only .tex file is saved.</p>
</td></tr>
<tr><td><code id="report_+3A_fig.folder">fig.folder</code></td>
<td>
<p>character specifying the folder where the plots corresponding to all row-variables of the table are placed. If it is left missing, a folder with the name file_figures is created in the same folder of 'file'.</p>
</td></tr>
<tr><td><code id="report_+3A_compile">compile</code></td>
<td>
<p>logical indicating whether tex file is compiled using <code><a href="tools.html#topic+texi2dvi">texi2pdf</a></code> function. Default value is TRUE.</p>
</td></tr>
<tr><td><code id="report_+3A_openfile">openfile</code></td>
<td>
<p>logical indicating whether to open the compiled pdf file or not. Currently deprectated. Deafult value is FALSE.</p>
</td></tr>
<tr><td><code id="report_+3A_title">title</code></td>
<td>
<p>character specifying the title of the report on the cover page. Default value is 'Report'.</p>
</td></tr>
<tr><td><code id="report_+3A_author">author</code></td>
<td>
<p>character specifying the author/s name/s of the report on the cover page. When missing, no authors appear.</p>
</td></tr>
<tr><td><code id="report_+3A_date">date</code></td>
<td>
<p>character specifying the date of the report on the cover page. When missing, the present date appears.</p>
</td></tr>
<tr><td><code id="report_+3A_perc">perc</code></td>
<td>
<p>logical. Plot relative frequencies (in percentatges) instead of absolute frequencies are displayed in barplots for categorical variable.</p>
</td></tr>
<tr><td><code id="report_+3A_...">...</code></td>
<td>
<p>Arguments passed to <code><a href="#topic+export2latex">export2latex</a></code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This functions does not work with stratified tables ('cbind.createTable' class objects). To report this class of tables you can report each of its component (see second example from 'examples' section).
</p>
<p>In order to compile the tex file the following packages must be available:<br />
- babel <br />
- longtable  <br />
- hyperref   <br />
- multirow    <br />
- lscape     <br />
- geometry <br />
- float      <br />
- inputenc <br />
- epsfig <br />
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+export2latex">export2latex</a></code>, <code><a href="#topic+export2csv">export2csv</a></code>, <code><a href="#topic+export2html">export2html</a></code>, <code><a href="#topic+radiograph">radiograph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

require(compareGroups)
data(regicor)

 # example on an ordinary table
res &lt;- createTable(compareGroups(year ~ . -id, regicor), hide = c(sex=1), hide.no = 'no')
report(res, "report.pdf" ,size="small", title="\Huge \textbf{REGICOR study}", 
       author="Isaac Subirana \\ IMIM-Parc de Salut Mar")

 # example on an stratified table by sex
res.men &lt;- createTable(compareGroups(year ~ . -id-sex, regicor, subset=sex=='Male'), 
                       hide.no = 'no')
res.wom &lt;- createTable(compareGroups(year ~ . -id-sex, regicor, subset=sex=='Female'), 
                       hide.no = 'no')
res &lt;- cbind("Men"=res.men, "Wom"=res.wom)
report(res[[1]], "reportmen.pdf", size="small", 
        title="\Huge \textbf{REGICOR study \\ Men}", date="") # report for men / no date
report(res[[2]], "reportwom.pdf", size="small", 
        title="\Huge \textbf{REGICOR study \\ Women}", date="") # report for wom / no date


## End(Not run)

</code></pre>

<hr>
<h2 id='SNPs'>SNPs in a case-control study</h2><span id='topic+SNPs'></span>

<h3>Description</h3>

<p>SNPs data.frame contains selected SNPs and other clinical covariates for cases and controls 
in a case-control study
</p>
<p>SNPs.info.pos data.frame contains the names of the SNPs included in the data set 'SNPs' including
their chromosome and their genomic position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPs)</code></pre>


<h3>Format</h3>

<p>'SNPs' data.frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
	id </td><td style="text-align: left;"> identifier of each subject </td>
</tr>
<tr>
 <td style="text-align: left;">
	casco </td><td style="text-align: left;"> case or control status: 0-control, 1-case </td>
</tr>
<tr>
 <td style="text-align: left;">
	sex </td><td style="text-align: left;"> gender: Male and Female </td>
</tr>
<tr>
 <td style="text-align: left;">
	blood.pre </td><td style="text-align: left;"> arterial blood presure </td>
</tr>
<tr>
 <td style="text-align: left;">
	protein </td><td style="text-align: left;"> protein levels </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp10001 </td><td style="text-align: left;"> SNP 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp10002 </td><td style="text-align: left;"> SNP 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
	  ...        </td><td style="text-align: left;">  ...      </td>
</tr>
<tr>
 <td style="text-align: left;">
	snp100036 </td><td style="text-align: left;"> SNP 36 </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>

<p>'SNPs.info.pos' data.frame contains the following columns:
A data frame with 35 observations on the following 3 variables.
</p>

<dl>
<dt><code>snp</code></dt><dd><p>name of SNP</p>
</dd>
<dt><code>chr</code></dt><dd><p>name of chromosome</p>
</dd>
<dt><code>pos</code></dt><dd><p>genomic position</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data obtained from the &lt;code&gt;SNPassoc&lt;/code&gt; package.
</p>

<hr>
<h2 id='strataTable'>
Stratify descriptive table in stratas.
</h2><span id='topic+strataTable'></span>

<h3>Description</h3>

<p>This functions re-build a descriptive table in stratas defined by a variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strataTable(x, strata, strata.names = NULL, max.nlevels = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="strataTable_+3A_x">x</code></td>
<td>
<p>an object of class 'createTable'</p>
</td></tr>
<tr><td><code id="strataTable_+3A_strata">strata</code></td>
<td>
<p>character specifying the name of the variable whose values or levels defines strata.</p>
</td></tr>
<tr><td><code id="strataTable_+3A_strata.names">strata.names</code></td>
<td>
<p>character vector with as many components as stratas, or NULL (default value). If NULL, it takes the names of levels of strata variable.</p>
</td></tr>
<tr><td><code id="strataTable_+3A_max.nlevels">max.nlevels</code></td>
<td>
<p>an integer indicating the maximum number of unique values or levels of strata variable. Default value is 5.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'cbind.createTable'.
</p>


<h3>References</h3>

<p>Isaac Subirana, Hector Sanz, Joan Vila (2014). Building Bivariate Tables: The compareGroups Package for R. Journal of Statistical Software, 57(12), 1-16. URL <a href="https://www.jstatsoft.org/v57/i12/">https://www.jstatsoft.org/v57/i12/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+createTable">createTable</a></code>, <code><a href="#topic+descrTable">descrTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(compareGroups)

# load REGICOR data
data(regicor)

# compute the descriptive tables (by year)
restab &lt;- descrTable(year ~ . - id - sex, regicor, hide.no="no")

# re-build the table stratifying by gender
strataTable(restab, "sex")

</code></pre>

<hr>
<h2 id='varinfo'>
Variable names and labels extraction
</h2><span id='topic+varinfo'></span><span id='topic+varinfo.compareGroups'></span><span id='topic+varinfo.createTable'></span>

<h3>Description</h3>

<p>This functions builds and prints a table with the variable names and their labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varinfo(x, ...)
## S3 method for class 'compareGroups'
varinfo(x, ...)
## S3 method for class 'createTable'
varinfo(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="varinfo_+3A_x">x</code></td>
<td>
<p>an object of class 'compareGroups' or 'createTable'</p>
</td></tr>
<tr><td><code id="varinfo_+3A_...">...</code></td>
<td>
<p>other arguments currently ignored</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, a compareGroup descriptives table lists variables by label (if one exists) rather than by name.  If researchers have assigned detailed labels to their variables, this function is very useful to quickly locate the original variable name if some modification is required. This function simply lists all &quot;Analyzed variable names&quot; by &quot;Orig varname&quot; (i.e. variable name in the data.frame) and &quot;Shown varname&quot; (i.e., label).
</p>


<h3>Value</h3>

<p>A 'matrix' with two columns
</p>
<table role = "presentation">
<tr><td><code>Orig varname</code></td>
<td>
<p>actual variable name in the 'data.frame' or in the 'parent environment'.</p>
</td></tr>
<tr><td><code>Shown varname</code></td>
<td>
<p>names of the variable shown in the resulting tables.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>If a variable has no &quot;label&quot; attribute, then the 'original varname' is the same as the 'shown varname'. The first variable in the table corresponds to the grouping variable. To label non-labeled variables or to change the label, specify its &quot;label&quot; attribute..
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compareGroups">compareGroups</a></code>, <code><a href="#topic+createTable">createTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(compareGroups)
data(regicor)
res&lt;-compareGroups(sex ~ . ,regicor)
#createTable(res, hide.no = 'no')  
varinfo(res)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
