<!DOCTYPE html><html lang="en"><head><title>Help for package compound.Cox</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {compound.Cox}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compound.Cox-package'>
<p>Univariate Feature Selection and Compound Covariate for Predicting Survival, Including Copula-Based Analyses for Dependent Censoring</p></a></li>
<li><a href='#CG.Clayton'>
<p>Copula-graphic estimator under the Clayton copula.</p></a></li>
<li><a href='#CG.Frank'>
<p>Copula-graphic estimator under the Frank copula.</p></a></li>
<li><a href='#CG.Gumbel'>
<p>Copula-graphic estimator under the Gumbel copula.</p></a></li>
<li><a href='#CG.test'>
<p>Testing survival difference of two groups via the CG estimators</p></a></li>
<li><a href='#cindex.CV'>
<p>Cross-validated c-index for measuring the</p>
predictive accuracy of a prognostic index under a copula-based dependent censoring model.</a></li>
<li><a href='#compound.reg'>
<p>Compound shrinkage estimation under the Cox model</p></a></li>
<li><a href='#dependCox.reg'>
<p>Univariate Cox regression under dependent censoring.</p></a></li>
<li><a href='#dependCox.reg.CV'>
<p>Cox regression under dependent censoring.</p></a></li>
<li><a href='#Lung'>
<p>Survival data for patients with non-small-cell lung cancer.</p></a></li>
<li><a href='#PBC'>
<p>Primary biliary cirrhosis (PBC) of the liver data</p></a></li>
<li><a href='#surv.factorial'>
<p>Factorial survival analysis under dependent censoring</p></a></li>
<li><a href='#uni.score'><p>Univariate Cox score test</p></a></li>
<li><a href='#uni.selection'><p>Univariate feature selection based on univariate significance tests</p></a></li>
<li><a href='#uni.Wald'><p>Univariate Cox Wald test</p></a></li>
<li><a href='#X.pathway'><p>Generate a matrix of gene expressions in the presence of gene pathways</p></a></li>
<li><a href='#X.tag'><p>Generate a matrix of gene expressions in the presence of tag genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Univariate Feature Selection and Compound Covariate for
Predicting Survival, Including Copula-Based Analyses for
Dependent Censoring</td>
</tr>
<tr>
<td>Version:</td>
<td>3.32</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-1-11</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Takeshi Emura &lt;takeshiemura@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Univariate feature selection and compound covariate methods under the Cox model with high-dimensional features (e.g., gene expressions).
 Available are survival data for non-small-cell lung cancer patients with gene expressions (Chen et al 2007 New Engl J Med) &lt;<a href="https://doi.org/10.1056%2FNEJMoa060096">doi:10.1056/NEJMoa060096</a>&gt;,
 statistical methods in Emura et al (2012 PLoS ONE) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0047627">doi:10.1371/journal.pone.0047627</a>&gt;,
 Emura &amp; Chen (2016 Stat Methods Med Res) &lt;<a href="https://doi.org/10.1177%2F0962280214533378">doi:10.1177/0962280214533378</a>&gt;, and Emura et al (2019)&lt;<a href="https://doi.org/10.1016%2Fj.cmpb.2018.10.020">doi:10.1016/j.cmpb.2018.10.020</a>&gt;.
 Algorithms for generating correlated gene expressions are also available.
 Estimation of survival functions via copula-graphic (CG) estimators is also implemented, which is useful for
 sensitivity analyses under dependent censoring (Yeh et al 2023 Biomedicines) &lt;<a href="https://doi.org/10.3390%2Fbiomedicines11030797">doi:10.3390/biomedicines11030797</a>&gt; and
 factorial survival analyses (Emura et al 2024 Stat Methods Med Res) &lt;<a href="https://doi.org/10.1177%2F09622802231215805">doi:10.1177/09622802231215805</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>numDeriv, survival, MASS</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-11 04:25:45 UTC; takes</td>
</tr>
<tr>
<td>Author:</td>
<td>Takeshi Emura [aut, cre],
  Hsuan-Yu Chen [aut],
  Shigeyuki Matsui [aut],
  Yi-Hau Chen [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-11 06:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='compound.Cox-package'>
Univariate Feature Selection and Compound Covariate for Predicting Survival, Including Copula-Based Analyses for Dependent Censoring
</h2><span id='topic+compound.Cox-package'></span><span id='topic+compound.Cox'></span>

<h3>Description</h3>

<p>Univariate feature selection and compound covariate methods under the Cox model with high-dimensional features (e.g., gene expressions).
Available are survival data for non-small-cell lung cancer patients with gene expressions (Chen et al 2007 New Engl J Med),
statistical methods in Emura et al (2012 PLoS ONE),
Emura &amp; Chen (2016 Stat Methods Med Res), and Emura et al. (2019 Comput Methods Programs Biomed).
Algorithms for generating correlated gene expressions are also available.
Estimation of survival functions via copula-graphic (CG) estimators is also implemented, which is useful for
sensitivity analyses under dependent censoring (Yeh et al 2023 Biomedicines)
and factorial survival analyses (Emura et al 2024 Stat Methods Med Res).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> compound.Cox</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 3.32</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2025-1-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Takeshi Emura, Hsuan-Yu Chen, Shigeyuki Matsui, Yi-Hau Chen;
Maintainer: Takeshi Emura &lt;takeshiemura@gmail.com&gt;
</p>


<h3>References</h3>

<p>Chen HY, Yu SL, Chen CH, et al (2007). A Five-gene Signature and Clinical Outcome in Non-small-cell Lung Cancer, N Engl J Med 356: 11-20.
</p>
<p>Emura T, Chen YH, Chen HY (2012). Survival Prediction Based on Compound Covariate under Cox Proportional Hazard Models.
PLoS ONE 7(10): e47627. doi:10.1371/journal.pone.0047627
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring:
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57
</p>
<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival,
Computer Methods and Programs in Biomedicine 168: 21-37
</p>
<p>Matsui S (2006). Predicting Survival Outcomes Using Subsets of Significant Genes in Prognostic Marker Studies with Microarrays. BMC Bioinformatics: 7:156.
</p>
<p>Yeh CT, Liao GY, Emura T (2023). Sensitivity analysis for survival prognostic prediction
with gene selection: a copula method for dependent censoring, Biomedicines 11(3):797.
</p>
<p>Emura T, Ditzhaus M, Dobler D, Murotani K (2024). Factorial survival analysis for treatment effects under dependent censoring, Stat Methods Med Res 33(1):61-79.
</p>

<hr>
<h2 id='CG.Clayton'>
Copula-graphic estimator under the Clayton copula.
</h2><span id='topic+CG.Clayton'></span>

<h3>Description</h3>

<p>This function computes the copula-graphic (CG) estimator (Rivest &amp; Wells 2001) of a survival function under the Clayton copula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CG.Clayton(t.vec, d.vec, alpha, S.plot = TRUE, S.col = "black")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CG.Clayton_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="CG.Clayton_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="CG.Clayton_+3A_alpha">alpha</code></td>
<td>
<p>Association parameter that is related to Kendall's tau through &quot;tau= alpha/(alpha+2)&quot;</p>
</td></tr>
<tr><td><code id="CG.Clayton_+3A_s.plot">S.plot</code></td>
<td>
<p>If TRUE, the survival function is displayed</p>
</td></tr>
<tr><td><code id="CG.Clayton_+3A_s.col">S.col</code></td>
<td>
<p>Color of the survival function in the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CG estimator is a variant of the Kaplan-Meier estimator for a survival function.
The CG estimator relaxes the independent censoring assumption of the KM estimator through a copula-based dependent censoring model.
The computational formula of the CG estimator is given in Appendix D of Emura et al. (2019) or Section 3.2 of Yeh et al.(2023).
The output shows the survival probabilities at given time points of &quot;t.vec&quot;.
The input requires to specify an association parameter &quot;alpha&quot; of the Clayton copula (alpha&gt;0), where alpha=0 corresponds to the independence copula.
Emura and Chen (2016, 2018) and Yeh et al.(2023) applied the CG estimator to assess survival prognosis for lung cancer patients.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau (=alpha/(alpha+2))</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>sort(t.vec)</p>
</td></tr>
<tr><td><code>n.risk</code></td>
<td>
<p>the number of patients at-risk</p>
</td></tr>
<tr><td><code>surv</code></td>
<td>
<p>survival probability at &quot;time&quot;</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival,
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring:
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>
<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring,
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Rivest LP, Wells MT (2001). A Martingale Approach to the Copula-graphic Estimator for the
Survival Function under Dependent Censoring, J Multivar Anal; 79: 138-55.
</p>
<p>Yeh CT, Liao GY, Emura T (2023). Sensitivity analysis for survival prognostic prediction
with gene selection: a copula method for dependent censoring, Biomedicines 11(3):797.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 (a toy example of n=8) ##
t.vec=c(1,3,5,4,7,8,10,13)
d.vec=c(1,0,0,1,1,0,1,0)
CG.Clayton(t.vec,d.vec,alpha=18,S.col="blue")
### CG.Clayton gives identical results with the Kaplan-Meier estimator with alpha=0 ###
CG.Clayton(t.vec,d.vec,alpha=0,S.plot=FALSE)$surv
survfit(Surv(t.vec,d.vec)~1)$surv

## Example 2 (Analysis of the lung cancer data) ##
data(Lung) # read the data
t.vec=Lung[,"t.vec"]
d.vec=Lung[,"d.vec"]
x.vec=Lung[,"MMP16"] # the gene associated with survival (Emura and Chen 2016, 2018) #
Poor=x.vec&gt;median(x.vec) ## Indicator of poor survival
Good=x.vec&lt;=median(x.vec) ## Indicator of good survival

par(mfrow=c(1,2))
###### Predicted survival curves via the CG estimator #####
t.good=t.vec[Good]
d.good=d.vec[Good]
CG.Clayton(t.good,d.good,alpha=18,S.plot=TRUE,S.col="blue")

t.poor=t.vec[Poor]
d.poor=d.vec[Poor]
CG.Clayton(t.poor,d.poor,alpha=18,S.plot=TRUE,S.col="red")
</code></pre>

<hr>
<h2 id='CG.Frank'>
Copula-graphic estimator under the Frank copula.
</h2><span id='topic+CG.Frank'></span>

<h3>Description</h3>

<p>This function computes the copula-graphic (CG) estimator (Rivest &amp; Wells 2001) under the Frank copula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CG.Frank(t.vec, d.vec, alpha, S.plot = TRUE, S.col = "black")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CG.Frank_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="CG.Frank_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="CG.Frank_+3A_alpha">alpha</code></td>
<td>
<p>Association parameter that is related to Kendall's tau (see P.32, Table 3.1 of Emura and Chen (2018))</p>
</td></tr>
<tr><td><code id="CG.Frank_+3A_s.plot">S.plot</code></td>
<td>
<p>If TRUE, the survival function is displayed</p>
</td></tr>
<tr><td><code id="CG.Frank_+3A_s.col">S.col</code></td>
<td>
<p>Color of the survival function in the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CG estimator is a variant of the Kaplan-Meier estimator for a survival function.
The CG estimator relaxes the independent censoring assumption of the KM estimator through a copula-based dependent censoring model.
The output shows the survival probabilities at given time points of &quot;t.vec&quot;.
The input requires to specify an association parameter &quot;alpha&quot; of the Frank copula, where alpha=0 corresponds to the independence copula.
Emura and Chen (2016, 2018) and Yeh et al.(2023) applied the CG estimator to assess survival prognosis for lung cancer patients.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau (=alpha/(alpha+1))</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>sort(t.vec)</p>
</td></tr>
<tr><td><code>n.risk</code></td>
<td>
<p>the number of patients at-risk</p>
</td></tr>
<tr><td><code>surv</code></td>
<td>
<p>survival probability at &quot;time&quot;</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival,
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring:
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>
<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring,
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Rivest LP, Wells MT (2001). A Martingale Approach to the Copula-graphic Estimator for the
Survival Function under Dependent Censoring, J Multivar Anal; 79: 138-55.
</p>
<p>Yeh CT, Liao GY, Emura T (2023). Sensitivity analysis for survival prognostic prediction
with gene selection: a copula method for dependent censoring, Biomedicines 11(3):797.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 (a toy example of n=8) ##
t.vec=c(1,3,5,4,7,8,10,13)
d.vec=c(1,0,0,1,1,0,1,0)
CG.Frank(t.vec,d.vec,alpha=9,S.col="blue")
### CG.Frank gives identical results with the Kaplan-Meier estimator with alpha=0 ###
CG.Frank(t.vec,d.vec,alpha=0,S.plot=FALSE)$surv
survfit(Surv(t.vec,d.vec)~1)$surv

## Example 2 (Analysis of the lung cancer data) ##
data(Lung) # read the data
t.vec=Lung[,"t.vec"]
d.vec=Lung[,"d.vec"]
x.vec=Lung[,"MMP16"] # the gene associated with survival (Emura and Chen 2016, 2018) #
Poor=x.vec&gt;median(x.vec) ## Indicator of poor survival
Good=x.vec&lt;=median(x.vec) ## Indicator of good survival

par(mfrow=c(1,2))
###### Predicted survival curves via the CG estimator #####
t.good=t.vec[Good]
d.good=d.vec[Good]
CG.Frank(t.good,d.good,alpha=6,S.plot=TRUE,S.col="blue")

t.poor=t.vec[Poor]
d.poor=d.vec[Poor]
CG.Frank(t.poor,d.poor,alpha=6,S.plot=TRUE,S.col="red")
</code></pre>

<hr>
<h2 id='CG.Gumbel'>
Copula-graphic estimator under the Gumbel copula.
</h2><span id='topic+CG.Gumbel'></span>

<h3>Description</h3>

<p>This function computes the copula-graphic (CG) estimator (Rivest &amp; Wells 2001) under the Gumbel copula. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CG.Gumbel(t.vec, d.vec, alpha, S.plot = TRUE, S.col = "black")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CG.Gumbel_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="CG.Gumbel_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="CG.Gumbel_+3A_alpha">alpha</code></td>
<td>
<p>Association parameter that is related to Kendall's tau through &quot;tau= alpha/(alpha+1)&quot;</p>
</td></tr>
<tr><td><code id="CG.Gumbel_+3A_s.plot">S.plot</code></td>
<td>
<p>If TRUE, the survival function is displayed</p>
</td></tr>
<tr><td><code id="CG.Gumbel_+3A_s.col">S.col</code></td>
<td>
<p>Color of the survival function in the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CG estimator is a variant of the Kaplan-Meier estimator for a survival function. 
The CG estimator relaxes the independent censoring assumption of the KM estimator through a copula-based dependent censoring model. 
The computational formula of the CG estimator is given in Appendix D of Emura et al. (2019) or Section 3.2 of Yeh et al.(2023). 
The output shows the survival probabilities at given time points of &quot;t.vec&quot;. 
The input requires to specify an association parameter &quot;alpha&quot; of the Gumbel copula (alpha&gt;=0), where alpha=0 corresponds to the independence copula. 
Emura and Chen (2016, 2018) and Yeh et al.(2023) applied the CG estimator to assess survival prognosis for lung cancer patients.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau (=alpha/(alpha+1))</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>sort(t.vec)</p>
</td></tr>
<tr><td><code>n.risk</code></td>
<td>
<p>the number of patients at-risk</p>
</td></tr>
<tr><td><code>surv</code></td>
<td>
<p>survival probability at &quot;time&quot;</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>
<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring, 
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Rivest LP, Wells MT (2001). A Martingale Approach to the Copula-graphic Estimator for the 
Survival Function under Dependent Censoring, J Multivar Anal; 79: 138-55.
</p>
<p>Yeh CT, Liao GY, Emura T (2023). Sensitivity analysis for survival prognostic prediction 
with gene selection: a copula method for dependent censoring, Biomedicines 11(3):797.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 (a toy example of n=8) ##
t.vec=c(1,3,5,4,7,8,10,13)
d.vec=c(1,0,0,1,1,0,1,0)
CG.Gumbel(t.vec,d.vec,alpha=9,S.col="blue")
### CG.Gumbel gives identical results with the Kaplan-Meier estimator with alpha=0 ### 
CG.Gumbel(t.vec,d.vec,alpha=0,S.plot=FALSE)$surv
survfit(Surv(t.vec,d.vec)~1)$surv

## Example 2 (Analysis of the lung cancer data) ##
data(Lung) # read the data
t.vec=Lung[,"t.vec"]
d.vec=Lung[,"d.vec"]
x.vec=Lung[,"MMP16"] # the gene associated with survival (Emura and Chen 2016, 2018) #
Poor=x.vec&gt;median(x.vec) ## Indicator of poor survival
Good=x.vec&lt;=median(x.vec) ## Indicator of good survival

par(mfrow=c(1,2))
###### Predicted survival curves via the CG estimator #####
t.good=t.vec[Good]
d.good=d.vec[Good]
CG.Gumbel(t.good,d.good,alpha=9,S.plot=TRUE,S.col="blue")

t.poor=t.vec[Poor]
d.poor=d.vec[Poor]
CG.Gumbel(t.poor,d.poor,alpha=9,S.plot=TRUE,S.col="red")
</code></pre>

<hr>
<h2 id='CG.test'>
Testing survival difference of two groups via the CG estimators
</h2><span id='topic+CG.test'></span>

<h3>Description</h3>

<p>Testing survival difference of two prognostic groups separated by a prognostic index (PI).
Survival probabilities are computed by the CG estimators (Yeh, et al. 2023).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CG.test(t.vec,d.vec,PI,cutoff=median(PI),alpha=2,
copula=CG.Clayton,S.plot=TRUE,N=10000,mark.time=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CG.test_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="CG.test_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="CG.test_+3A_pi">PI</code></td>
<td>
<p>Vector of real numbers (the values of a prognostic index)</p>
</td></tr>
<tr><td><code id="CG.test_+3A_cutoff">cutoff</code></td>
<td>
<p>A number determining the cut-off value of a prognostic index</p>
</td></tr>
<tr><td><code id="CG.test_+3A_alpha">alpha</code></td>
<td>
<p>Copula parameter</p>
</td></tr>
<tr><td><code id="CG.test_+3A_copula">copula</code></td>
<td>
<p>Copula function: &quot;CG.Clayton&quot;,&quot;CG.Gumbel&quot; or &quot;CG.Frank&quot;</p>
</td></tr>
<tr><td><code id="CG.test_+3A_s.plot">S.plot</code></td>
<td>
<p>If TRUE, the survival curve is displayed</p>
</td></tr>
<tr><td><code id="CG.test_+3A_n">N</code></td>
<td>
<p>The number of permutations</p>
</td></tr>
<tr><td><code id="CG.test_+3A_mark.time">mark.time</code></td>
<td>
<p>If TRUE, then curves are marked at each censoring time</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two-sample comparison based on estimated survival functions
via copula-graphic estimators under dependent censoring.
The D statistic (the mean vertical difference betewen two
estimated survival functions) is used for testing the null
hypothesis of no difference in survival.
See Yeh et al.(2023) for details.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>test</code></td>
<td>
<p>Testing the difference of two survival functions</p>
</td></tr>
<tr><td><code>Good</code></td>
<td>
<p>Good prognostic group defined by PI&lt;=c</p>
</td></tr>
<tr><td><code>Poor</code></td>
<td>
<p>Poor prognostic group defined by PI&gt;c</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura, Pauline Baur</p>


<h3>References</h3>

<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring,
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Rivest LP, Wells MT (2001). A Martingale Approach to the Copula-graphic Estimator for the
Survival Function under Dependent Censoring, J Multivar Anal; 79: 138-55.
</p>
<p>Yeh CT, Liao GY, Emura T (2023). Sensitivity analysis for survival prognostic prediction
with gene selection: a copula method for dependent censoring, Biomedicines 11(3):797.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t.vec=c(1,3,5,4,7,8,10,13)
d.vec=c(1,0,0,1,1,0,1,0)
PI=c(8,7,6,5,4,3,2,1)

CG.test(t.vec,d.vec,PI,copula=CG.Clayton,alpha=18,N=100)
CG.test(t.vec,d.vec,PI,copula=CG.Gumbel,alpha=2,N=100)
</code></pre>

<hr>
<h2 id='cindex.CV'>
Cross-validated c-index for measuring the 
predictive accuracy of a prognostic index under a copula-based dependent censoring model.
</h2><span id='topic+cindex.CV'></span>

<h3>Description</h3>

<p>This function calculates the cross-validated c-index (concordance index) for measuring the 
predictive accuracy of a prognostic index under a copula-based dependent censoring model. 
Here the prognostic index is calculated as a compound covariate predictor based on the 
univariate Cox regression estimates. The expression and details are given in Section 3.2 of Emura and Chen (2016). 
The association between survival time and censoring time is modeled via the Clayton copula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cindex.CV(t.vec, d.vec, X.mat, alpha, K = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cindex.CV_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to death or time to censoring, whichever comes first)</p>
</td></tr>
<tr><td><code id="cindex.CV_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="cindex.CV_+3A_x.mat">X.mat</code></td>
<td>
<p>n by p matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
<tr><td><code id="cindex.CV_+3A_alpha">alpha</code></td>
<td>
<p>Association parameter of the Clayton copula; Kendall's tau = alpha/(alpha+2)</p>
</td></tr>
<tr><td><code id="cindex.CV_+3A_k">K</code></td>
<td>
<p>The number of cross-validation folds (K=5 is the defailt)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently, only the Clayton copula is implemented for modeling association between survival time 
and censoring time. The Clayton model yields positive association between survival time and censoring 
time with the Kendall's tau being equal to alpha/(alpha+2), 
where alpha &gt; 0. The independent copula corresponds to alpha = 0.
</p>
<p>If the number of covariates p is large (e.g., p&gt;=100), the computational time becomes very long. 
Pre-filtering for covariates is recommended to reduce p. 
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>concordant</code></td>
<td>
<p>Cross-validated c-index</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
n=25 ### sample size ###
p=3  ### the number of covariates ###
set.seed(1)
T=rexp(n) ### survival time
U=rexp(n) ### censoring time
t.vec=pmin(T,U) ### minimum of survival time and censoring time
d.vec=as.numeric( c(T&lt;=U) ) ### censoring indicator
X.mat=matrix(runif(n*p),n,p) ### covariates matrix

cindex.CV(t.vec,d.vec,X.mat,alpha=2) ### alpha=2 corresponds to Kendall's tau=0.5

</code></pre>

<hr>
<h2 id='compound.reg'>
Compound shrinkage estimation under the Cox model 
</h2><span id='topic+compound.reg'></span>

<h3>Description</h3>

<p>This function implements the &quot;compound shrinkage estimator&quot; to calculate the regression coefficients
of the Cox model, which was proposed by Emura, Chen &amp; Chen (2012).
The method is a variant of the Cox partial likelihood estimator such that
the regression coefficients are mixed with the univariate Cox regression estimators.
The resultant estimator is applicable even when the number of covariates is greater than the number of
samples (the p&gt;n setting).
The standard errors (SEs) are calculated based on the asymptotic theory (see Emura et al., 2012).    
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compound.reg(t.vec, d.vec, X.mat, K = 5, delta_a = 0.025, a_0 = 0, var = FALSE,
plot=TRUE, randomize = FALSE, var.detail = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compound.reg_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_x.mat">X.mat</code></td>
<td>
<p>n by p matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_k">K</code></td>
<td>
<p>The number of cross validation folds, K=n corresponds to a leave-one-out cross validation (default=5)</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_delta_a">delta_a</code></td>
<td>
<p>The step size for a grid search for the maximum of the cross-validated likelihood (default=0.025)</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_a_0">a_0</code></td>
<td>
<p>The starting value of a grid search for the maximum of the cross-validated likelihood (default=0)</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_var">var</code></td>
<td>
<p>If TRUE, the standard deviations and confidence intervals are given (default=FALSE, to reduce the computational cost)</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_plot">plot</code></td>
<td>
<p>If TRUE, the cross validated likelihood curve and its maximized point are drawn</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_randomize">randomize</code></td>
<td>
<p>If TRUE, randomize the subject ID's so that the subjects in the cross validation folds
are randomly chosen. Otherwise, the cross validation folds are constructed in the ascending sequence</p>
</td></tr>
<tr><td><code id="compound.reg_+3A_var.detail">var.detail</code></td>
<td>
<p>Detailed information about the covariance matrix, which is mainly used for theoretical purposes. 
Please consult Takeshi Emura for more details (default=FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>K=5 cross validation is recommended for computational efficiency, though the results appear to be robust
against the choice of the number K. If the number of covariates is greater than 200, the computational time becomes very long.
In such a case, the univariate pre-selection is recommended to reduce the number of covariates.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>a</code></td>
<td>
<p>An optimized value of the shrinkage parameter (0&lt;=a&lt;=1)</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated regression coefficients</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>Standard errors for estimated regression coefficients</p>
</td></tr>
<tr><td><code>Lower95CI</code></td>
<td>
<p>Lower ends of 95 percent confidence intervals (beta_hat-1.96*SE)</p>
</td></tr>
<tr><td><code>Upper95CI</code></td>
<td>
<p>Upper ends of 95 percent confidence intervals (beta_hat+1.96*SE)</p>
</td></tr>
<tr><td><code>Sigma</code></td>
<td>
<p>Covariance matrix for estimated regression coefficients</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>Estimates of the information matrix (-[Hessian of the loglikelihood]/n)</p>
</td></tr>
<tr><td><code>Hessian_CV</code></td>
<td>
<p>Second derivative of the cross-validated likelihood. Normally negative
since the cross-validated curve is concave</p>
</td></tr>
<tr><td><code>h_dot</code></td>
<td>
<p>Derivative of Equation (8) of Emura et al. (2012) with respect to a shrinkage parameter &quot;a&quot;</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura &amp; Yi-Hau Chen</p>


<h3>References</h3>

<p>Emura T, Chen Y-H, Chen H-Y (2012) Survival Prediction Based on Compound Covariate under Cox Proportional Hazard Models.
PLoS ONE 7(10): e47627. doi:10.1371/journal.pone.0047627
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### A simulation study ###
n=50 ### sample size
beta_true=c(1,1,0,0,0)
p=length(beta_true) 
t.vec=d.vec=numeric(n)
X.mat=matrix(0,n,p)

set.seed(1)
for(i in 1:n){
  X.mat[i,]=rnorm(p,mean=0,sd=1)
  eta=sum( as.vector(X.mat[i,])*beta_true )
  T=rexp(1,rate=exp(eta))
  C=runif(1,min=0,max=5)
  t.vec[i]=min(T,C)
  d.vec[i]=(T&lt;=C)
}
compound.reg(t.vec,d.vec,X.mat,delta_a=0.1) 
### compare the estimates (beta) with the true value ###
beta_true

### Lung cancer data analysis (Emura et al. 2012 PLoS ONE) ###
data(Lung)
temp=Lung[,"train"]==TRUE
t.vec=Lung[temp,"t.vec"]
d.vec=Lung[temp,"d.vec"]
X.mat=as.matrix( Lung[temp,-c(1,2,3)] )
#compound.reg(t.vec=t.vec,d.vec=d.vec,X.mat=X.mat,delta_a=0.025) # time-consuming process
</code></pre>

<hr>
<h2 id='dependCox.reg'>
Univariate Cox regression under dependent censoring.
</h2><span id='topic+dependCox.reg'></span>

<h3>Description</h3>

<p>This function performs univariate Cox regression under dependent censoring,
where dependence between survival time and censoring time is modeled via the Clayton copula (Emura and Chen 2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dependCox.reg(t.vec, d.vec, X.vec, alpha, var = TRUE, censor.reg=FALSE, baseline=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dependCox.reg_+3A_t.vec">t.vec</code></td>
<td>
<p>A vector of survival times (time-to-death or censoring)</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_d.vec">d.vec</code></td>
<td>
<p>A vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_x.vec">X.vec</code></td>
<td>
<p>A vector of covariates (multiple covariates are not allowed)</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_alpha">alpha</code></td>
<td>
<p>An copula parameter (Kendall's tau = alpha/(alpha+2)</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_var">var</code></td>
<td>
<p>If TRUE, the standard deviations are given (use FALSE to reduce the computational cost)</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_censor.reg">censor.reg</code></td>
<td>
<p>If TRUE, show the fitted results for both survival and censoring models</p>
</td></tr>
<tr><td><code id="dependCox.reg_+3A_baseline">baseline</code></td>
<td>
<p>If TRUE, show the cumulative baseline hazards at the values of &quot;t.vec&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Clayton model yields positive association between survival time and censoring time with Kendall's tau being equal to 
alpha/(alpha+2), where alpha &gt; 0 is a copula parameter. The independence copula corresponds to alpha = 0. 
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>The estimated regression coefficient</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>The standard error for the estimated regression coefficient</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>The Z-value for testing the null hypothesis of &quot;beta=0&quot; (the Wald test)</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>The P-value for testing the null hypothesis of &quot;beta=0&quot; (the Wald test)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Joint Cox regression of survival and censoring ### 
data(Lung)
t.vec=Lung[,"t.vec"]# death or censoring times #
d.vec=Lung[,"d.vec"]# censoring indicators #
# 16-gene prognostic index (Emura and Chen 2016; 2018) #
X.vec=0.51*Lung[,"ZNF264"]+0.50*Lung[,"MMP16"]+
  0.50*Lung[,"HGF"]-0.49*Lung[,"HCK"]+0.47*Lung[,"NF1"]+
  0.46*Lung[,"ERBB3"]+0.57*Lung[,"NR2F6"]+0.77*Lung[,"AXL"]+
  0.51*Lung[,"CDC23"]+0.92*Lung[,"DLG2"]-0.34*Lung[,"IGF2"]+
  0.54*Lung[,"RBBP6"]+0.51*Lung[,"COX11"]+
  0.40*Lung[,"DUSP6"]-0.37*Lung[,"ENG"]-0.41*Lung[,"IHPK1"]
dependCox.reg(t.vec,d.vec,X.vec,alpha=18,censor.reg=TRUE)

temp=c(Lung[,"train"]==TRUE)
t.vec=Lung[temp,"t.vec"]
d.vec=Lung[temp,"d.vec"]
dependCox.reg(t.vec,d.vec,Lung[temp,"ZNF264"],alpha=18) 
# this reproduces Table 3 of Emura and Chen (2016) #

#### A simulation study under dependent censoring ####
beta_true=1.5 # true regression coefficient
alpha_true=2 # true copula parameter corresponding to Kendall's tau=0.5
n=150
t.vec=d.vec=X.vec=numeric(n)
set.seed(1)
for(i in 1:n){
  X.vec[i]=runif(1)
  eta=X.vec[i]*beta_true
  U=runif(1)
  V=runif(1)
  T=-1/exp(eta)*log(1-U) # Exp(eta) distribution
  W=(1-U)^(-alpha_true) # dependence produced by the Clayton copula
  C=1/alpha_true/exp(eta)*log(  1-W+W*(1-V)^(-alpha_true/(alpha_true+1))  ) # Exp(eta) distribution
  t.vec[i]=min(T,C)
  d.vec[i]=(T&lt;=C) 
}

dependCox.reg(t.vec,d.vec,X.vec,alpha=alpha_true,var=FALSE) # faster computation by "var=FALSE"
beta_true# the above estimate is close to the true value
coxph(Surv(t.vec,d.vec)~X.vec)$coef
# this estimate is biased for the true value due to dependent censoring 
</code></pre>

<hr>
<h2 id='dependCox.reg.CV'>
Cox regression under dependent censoring.
</h2><span id='topic+dependCox.reg.CV'></span>

<h3>Description</h3>

<p>This function performs estimation and significance testing for survival data under a copula-based dependent censoring model proposed in
Emura and Chen (2016). The dependency between the failure and censoring times is modeled via the Clayton copula.
The method is based on the semiparametric maximum likelihood estimation, where the association parameter is estimated by maximizing
the cross-validated c-index (see Emura and Chen 2016 for details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dependCox.reg.CV(t.vec, d.vec, X.mat, K = 5, G = 20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dependCox.reg.CV_+3A_t.vec">t.vec</code></td>
<td>
<p>A vector of survival times (time-to-death or censoring)</p>
</td></tr>
<tr><td><code id="dependCox.reg.CV_+3A_d.vec">d.vec</code></td>
<td>
<p>A vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="dependCox.reg.CV_+3A_x.mat">X.mat</code></td>
<td>
<p>An (n*p) matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
<tr><td><code id="dependCox.reg.CV_+3A_k">K</code></td>
<td>
<p>The number of cross-validation folds</p>
</td></tr>
<tr><td><code id="dependCox.reg.CV_+3A_g">G</code></td>
<td>
<p>The number of grids to optimize c-index (c-index is computed for G different values of copula parameters)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Clayton model yields positive association between survival time and censoring time with Kendall's tau being equal to 
alpha/(alpha+2), where alpha &gt; 0 is a copula parameter. The independence copula corresponds to alpha = 0. 
</p>
<p>If the number of covariates p is large (p&gt;=100), the computational time becomes very long. 
We suggest using &quot;uni.selection&quot; to reduce the number such that p&lt;100. 
</p>
<p>If the number of grids G is large, the computational time becomes very long. Please take 5&lt;=G&lt;=20.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>The estimated regression coefficients</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>The standard errors for the estimated regression coefficients</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>The Z-values for testing the null hypothesis of &quot;beta=0&quot; (the Wald test)</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>The P-values for testing the null hypothesis of &quot;beta=0&quot; (the Wald test)</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The estimated copula parameter by optimizing c-index</p>
</td></tr>
<tr><td><code>c_index</code></td>
<td>
<p>The optimized value of c_index</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Reproduce Section 5 of Emura and Chen (2016) ###
data(Lung)
temp=Lung[,"train"]==TRUE
t.vec=Lung[temp,"t.vec"]
d.vec=Lung[temp,"d.vec"]
X.mat=as.matrix(Lung[temp,-c(1,2,3)])
#dependCox.reg.CV(t.vec,d.vec,X.mat,G=20) # time-consuming process #
</code></pre>

<hr>
<h2 id='Lung'>
Survival data for patients with non-small-cell lung cancer.
</h2><span id='topic+Lung'></span>

<h3>Description</h3>

<p>A subset of the lung cancer data (Chen et al. 2007) made available by Emura et al. (2019). 
The subset consists of 97 gene expressions from 125 patients with non-small-cell lung cancer.
The 97 genes were selected with P-value&lt;0.20 under univariate 
Cox regression analyses as previously done in Emura et al. (2012) and Emura and Chen (2016).
The intensity of gene expression was transformed to an ordinal level using the quantile,
i.e. if the intensity of gene expression was &lt;=25th, &gt;25th, &gt;50th, or &gt;75th percentile, 
it was coded as 1, 2, 3, or 4, respectively (Chen et al. 2007).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Lung")</code></pre>


<h3>Format</h3>

<p>A data frame with 125 observations on the following 100 variables.
</p>

<dl>
<dt><code>t.vec</code></dt><dd><p>survival times (time to either death or censoring) in months</p>
</dd>
<dt><code>d.vec</code></dt><dd><p>censoring indicators, 1=death, 0=censoring</p>
</dd>
<dt><code>train</code></dt><dd><p>TRUE=training set, FALSE=testing set, as defined in Chen et al. (2007)</p>
</dd>
<dt><code>VHL</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>IHPK1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>HMMR</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CMKOR1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PLAU</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>IGF2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>FGB</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MYBL2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ODC1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MTHFD2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>GLIPR1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>EZH2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>HCK</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CCNC</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>XRCC1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CYP1B1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CDC25A</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CD44</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>LCK</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MTHFS</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PON3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PTPN6</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>KIDINS220</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>KLHL22</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RBBP6</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>GABARAPL2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SEH1L</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CITED2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>BARD1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>TLX1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CRMP1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CTNNA1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ANXA5</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PTGS2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SMC4L1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>LOC285086</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ATP11B</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CDK10</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>IRF4</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MYH11</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ME3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CCT6A</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SNCG</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MAK3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>VCPIP1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>JMJD1A</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>STAT2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>DDX6</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ERBB3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PAX2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PCTK2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>NF1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>DLG2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>JMJD1A.1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SUCLA2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MMP16</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>AP3B2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>HGF</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MAP2K3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CPEB4</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ZNF264</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>AXL</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CDC23</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MAST3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>COX11</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PRKAG2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MAN1B1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>F8</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RSU1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MMD</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>AK5</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>IDS</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>BNIP1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>ENG</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PCDHGC3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RALY</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>WDR33</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RNF4</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PRDX1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>FXN</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>PTPRU</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>FRAP1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MMP7</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CST3</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>TIMP2</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>TAL1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>STAT1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>CCND1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>DUSP6</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SNRPF</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>MMP13</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>NR2F6</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>HOXA1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RIPK1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>IL7R</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>SEC13L1</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
<dt><code>RPL5</code></dt><dd><p>gene expression, coded as 1, 2, 3, or 4</p>
</dd>
</dl>



<h3>Details</h3>

<p>Survival data consisting of 125 patients. 
</p>


<h3>Source</h3>

<p>Chen HY, Yu SL, Chen CH, et al (2007). A Five-gene Signature and Clinical Outcome in Non-small-cell Lung Cancer, N Engl J Med 356: 11-20.
</p>
<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>


<h3>References</h3>

<p>Chen HY, Yu SL, Chen CH, et al (2007). A Five-gene Signature and Clinical Outcome in Non-small-cell Lung Cancer, N Engl J Med 356: 11-20.
</p>
<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lung)
Lung[1:3,] ## show the first 3 samples ## 

## The five-gene signature in Chen et al. (2007) ##
temp=Lung[,"train"]==TRUE
t.vec=Lung[temp,"t.vec"]
d.vec=Lung[temp,"d.vec"]
coxph(Surv(t.vec,d.vec)~Lung[temp,"ERBB3"])
coxph(Surv(t.vec,d.vec)~Lung[temp,"LCK"])
coxph(Surv(t.vec,d.vec)~Lung[temp,"DUSP6"])
coxph(Surv(t.vec,d.vec)~Lung[temp,"STAT1"])
coxph(Surv(t.vec,d.vec)~Lung[temp,"MMD"])
</code></pre>

<hr>
<h2 id='PBC'>
Primary biliary cirrhosis (PBC) of the liver data
</h2><span id='topic+PBC'></span>

<h3>Description</h3>

<p>A subset of primary biliary cirrhosis (PBC) of the liver data in the book &quot;Counting Process &amp; Survival Analysis&quot; by Fleming &amp; Harrington (1991).
This subset is used in Tibshirani (1997).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PBC)</code></pre>


<h3>Format</h3>

<p>A data frame with 276 observations on the following 19 variables.
</p>

<dl>
<dt><code>T</code></dt><dd><p>Survival times (either time to death or censoring) in days</p>
</dd>
<dt><code>d</code></dt><dd><p>Censoring indicator, 1=death, 0=censoring</p>
</dd>
<dt><code>trt</code></dt><dd><p>Treatment indicator, 1=treatment by D-penicillamine, 0=placebo</p>
</dd>
<dt><code>age</code></dt><dd><p>Age in years (days divided by 365.25)</p>
</dd>
<dt><code>sex</code></dt><dd><p>Sex, 0=male, 1=female</p>
</dd>
<dt><code>asc</code></dt><dd><p>Presence of ascites, 0=no, 1=yes</p>
</dd>
<dt><code>hep</code></dt><dd><p>Presence of hepatomegaly, 0=no, 1=yes</p>
</dd>
<dt><code>spi</code></dt><dd><p>Presence of spiders, 0=no, 1=yes</p>
</dd>
<dt><code>ede</code></dt><dd><p>Presence of edema, 0=no edema, 0.5=edema resolved by therapy, 1=edema not resolved by therapy</p>
</dd>
<dt><code>bil</code></dt><dd><p>log(bililubin, mg/dl)</p>
</dd>
<dt><code>cho</code></dt><dd><p>log(cholesterol, mg/dl)</p>
</dd>
<dt><code>alb</code></dt><dd><p>log(albumin, gm/dl)</p>
</dd>
<dt><code>cop</code></dt><dd><p>log(urine copper, mg/day)</p>
</dd>
<dt><code>alk</code></dt><dd><p>log(alkarine, U/liter)</p>
</dd>
<dt><code>SGO</code></dt><dd><p>log(SGOT, in U/ml)</p>
</dd>
<dt><code>tri</code></dt><dd><p>log(triglycerides, in mg/dl)</p>
</dd>
<dt><code>pla</code></dt><dd><p>log(platelet count, [the number of platelets per-cubic-milliliter of blood]/1000)</p>
</dd>
<dt><code>pro</code></dt><dd><p>log(prothrombin time, in seconds)</p>
</dd>
<dt><code>gra</code></dt><dd><p>Histologic stage of disease, graded 1, 2, 3, or 4</p>
</dd>
</dl>



<h3>Details</h3>

<p>Survival data consisting of 276 patients with 17 covariates.
Among them, 111 patients died (d=1) while others were censored (d=0).
The covariates consist of a treatment indicator (trt), age, sex, 5 categorical variables
(ascites, hepatomegaly, spider, edema, and stage of disease) and
9 log-transformed continuous variables (bilirubin, cholesterol, albumin, urine copper,
alkarine, SGOT, triglycerides, platelet count, and prothrombine). 
</p>


<h3>Source</h3>

<p>Fleming &amp; Harrignton (1991); Tibshirani (1997)
</p>


<h3>References</h3>

<p>Tibshirani R (1997), The Lasso method for variable selection in the Cox model, Statistics in Medicine, 385-395.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PBC)
PBC[1:5,] ### profiles for the first 5 patients ###
# See also Appendix D.1 of Fleming &amp; Harrington, Counting Process &amp; Survival Analysis (1991) # 
</code></pre>

<hr>
<h2 id='surv.factorial'>
Factorial survival analysis under dependent censoring
</h2><span id='topic+surv.factorial'></span>

<h3>Description</h3>

<p>Perform factorial survival analysis under dependent censoring under an assumed copula (Emura et al. 2024).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>surv.factorial(t.vec,d.vec,group,copula,alpha,R=1000,t.upper=min(tapply(t.vec,group,max)),
C=NULL,S.plot=TRUE,mark.time=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="surv.factorial_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_group">group</code></td>
<td>
<p>Vector of group indicators, 1, 2, ..., d</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_copula">copula</code></td>
<td>
<p>Copula function: &quot;CG.Clayton&quot;,&quot;CG.Gumbel&quot; or &quot;CG.Frank&quot;</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_alpha">alpha</code></td>
<td>
<p>Copula parameter</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_r">R</code></td>
<td>
<p>The number of Monte Carlo simulations to find the critical value of the F-test</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_t.upper">t.upper</code></td>
<td>
<p>Follow-up end (default is max(t.vec))</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_c">C</code></td>
<td>
<p>Contrast matrix</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_s.plot">S.plot</code></td>
<td>
<p>If TRUE, the survival curve is displayed</p>
</td></tr>
<tr><td><code id="surv.factorial_+3A_mark.time">mark.time</code></td>
<td>
<p>If TRUE, then curves are marked at each censoring time</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimates of treatment effects and the test results are shown.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>copula.parameter</code></td>
<td>
<p>Copula parameter</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Estimates of treatment effects</p>
</td></tr>
<tr><td><code>Var</code></td>
<td>
<p>Variance estimates</p>
</td></tr>
<tr><td><code>F</code></td>
<td>
<p>F-statistic</p>
</td></tr>
<tr><td><code>c.simu</code></td>
<td>
<p>Critical value via the simulation method</p>
</td></tr>
<tr><td><code>c.anal</code></td>
<td>
<p>Critical value via the analytical method</p>
</td></tr>
<tr><td><code>P.value</code></td>
<td>
<p>P-value of the F-test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Ditzhaus M, Dobler D, Murotani K (2024), Factorial survival analysis for treatment effects under dependent censoring, Stat Methods Med Res 33(1):61-79.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(survival)
data(cancer)
dat=subset(colon,etype==1)
## Treatment effects ##
t.vec=dat$time
d.vec=dat$status
trt=dat$rx
C12=matrix(c(1,-1,0),1,3,byrow=TRUE)
C13=matrix(c(1,0,-1),1,3,byrow=TRUE)
C23=matrix(c(0,1,-1),1,3,byrow=TRUE)
group=as.numeric(trt) # 1=Obs; 2=Lev; 3=Lev+5FU
# surv.factorial(t.vec,d.vec,group,alpha=2,copula=CG.Clayton)
# surv.factorial(t.vec,d.vec,group,alpha=2,copula=CG.Clayton,C=C12)
# surv.factorial(t.vec,d.vec,group,alpha=2,copula=CG.Clayton,C=C13)
# surv.factorial(t.vec,d.vec,group,alpha=2,copula=CG.Clayton,C=C23)
# alpha is a copula parameter

</code></pre>

<hr>
<h2 id='uni.score'>Univariate Cox score test</h2><span id='topic+uni.score'></span>

<h3>Description</h3>

<p>Univariate significance analyses via the score tests (Witten &amp; Tibshirani 2010; Emura et al. 2019) based on 
association between individual features and survival.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uni.score(t.vec, d.vec, X.mat, d0=0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uni.score_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="uni.score_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="uni.score_+3A_x.mat">X.mat</code></td>
<td>
<p>n by p matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
<tr><td><code id="uni.score_+3A_d0">d0</code></td>
<td>
<p>A positive constant to stabilize the variance (Witten &amp; Tibshirani 2010)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>score test</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>Estimated regression coefficients (one-step estimator)</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z-value for testing H_0: beta=0 (score test)</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>P-value for testing H_0: beta=0 (score test)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura and Shigeyuki Matsui</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2018-). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine, to appear.
</p>
<p>Witten DM, Tibshirani R (2010) Survival analysis with high-dimensional covariates. Stat Method Med Res 19:29-51
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lung)
t.vec=Lung$t.vec[Lung$train==TRUE]
d.vec=Lung$d.vec[Lung$train==TRUE]
X.mat=Lung[Lung$train==TRUE,-c(1,2,3)]
uni.score(t.vec, d.vec, X.mat)
</code></pre>

<hr>
<h2 id='uni.selection'>Univariate feature selection based on univariate significance tests</h2><span id='topic+uni.selection'></span>

<h3>Description</h3>

<p>This function performs univariate feature selection using significance tests (Wald tests or score tests) based on 
association between individual features and survival. 
Features are selected if their P-values are less than a given threshold (P.value). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uni.selection(t.vec, d.vec, X.mat, P.value=0.001,K=10,score=TRUE,d0=0,
                       randomize=FALSE,CC.plot=FALSE,permutation=FALSE,M=200)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uni.selection_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators (1=death, 0=censoring)</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_x.mat">X.mat</code></td>
<td>
<p>n by p matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_p.value">P.value</code></td>
<td>
<p>A threshold for selecting features</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_k">K</code></td>
<td>
<p>The number of cross-validation folds</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_score">score</code></td>
<td>
<p>If TRUE, the score tests are used; if not, the Wald tests are used</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_d0">d0</code></td>
<td>
<p>A positive constant to stabilize the variance of score statistics (Witten &amp; Tibshirani 2010)</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_randomize">randomize</code></td>
<td>
<p>If TRUE, randomize patient ID's before cross-validation</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_cc.plot">CC.plot</code></td>
<td>
<p>If TRUE, the compound covariate (CC) predictors are plotted</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_permutation">permutation</code></td>
<td>
<p>If TRUE, the FDR is computed by a permutation method (Witten &amp; Tibshirani 2010; Emura et al. 2019).</p>
</td></tr>
<tr><td><code id="uni.selection_+3A_m">M</code></td>
<td>
<p>The number of permutations to calculate the FDR</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The cross-validated likelihood (CVL) value is computed for selected features (Matsui 2006; Emura et al. 2019). 
A high CVL value corresponds to a better predictive ability of selected features. 
Hence, the CVL value can be used to find the optimal set of features. 
The CVL value is computed by a K-fold cross-validation, where the number K can be chosen by user. 
The false discovery rate (FDR) is also computed by a formula and a permutation test (if &quot;permutation=TRUE&quot;). 
The RCVL1 and RCVL2 are &quot;re-substitution&quot; CVL values and provide upper control limits for the CVL value.
If the CVL value is less than RCVL1 and RCVL2 values, the CVL value would be in-control. 
On the other hand, if the CVL value exceeds either RCVL1 or RCVL2 value, then 
the CVL may be computed again after changing the sample allocation.  
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>gene</code></td>
<td>
<p>Gene symbols</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated regression coefficients</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z-values for significance tests</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>P-values for significance tests</p>
</td></tr>
<tr><td><code>CVL</code></td>
<td>
<p>The value of CVL, RCVL1, and RCVL2 (Emura et al. 2019)</p>
</td></tr>
<tr><td><code>Genes</code></td>
<td>
<p>The number of genes, the number of selected genes, and the number of falsely selected genes</p>
</td></tr>
<tr><td><code>FDR</code></td>
<td>
<p>False discovery rate (by a formula or a permutation method)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Matsui S (2006). Predicting Survival Outcomes Using Subsets of Significant Genes in Prognostic Marker Studies with Microarrays. BMC Bioinformatics: 7:156.
</p>
<p>Witten DM, Tibshirani R (2010) Survival analysis with high-dimensional covariates. Stat Method Med Res 19:29-51
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lung)
t.vec=Lung$t.vec[Lung$train==TRUE]
d.vec=Lung$d.vec[Lung$train==TRUE]
X.mat=Lung[Lung$train==TRUE,-c(1,2,3)]
uni.selection(t.vec, d.vec, X.mat, P.value=0.05,K=5,score=FALSE)
## the outputs reproduce Table 3 of Emura and Chen (2016) ## 
</code></pre>

<hr>
<h2 id='uni.Wald'>Univariate Cox Wald test</h2><span id='topic+uni.Wald'></span>

<h3>Description</h3>

<p>Univariate significance analyses via the Wald tests (Witten &amp; Tibshirani 2010; Emura et al. 2019) based on 
association between individual features and survival. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uni.Wald(t.vec, d.vec, X.mat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uni.Wald_+3A_t.vec">t.vec</code></td>
<td>
<p>Vector of survival times (time to either death or censoring)</p>
</td></tr>
<tr><td><code id="uni.Wald_+3A_d.vec">d.vec</code></td>
<td>
<p>Vector of censoring indicators, 1=death, 0=censoring</p>
</td></tr>
<tr><td><code id="uni.Wald_+3A_x.mat">X.mat</code></td>
<td>
<p>n by p matrix of covariates, where n is the sample size and p is the number of covariates</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Wald test</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>Estimated regression coefficients</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>Z-value for testing H_0: beta=0 (Wald test)</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>P-value for testing H_0: beta=0 (Wald test)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Matsui S, Chen HY (2019). compound.Cox: Univariate Feature Selection and Compound Covariate for Predicting Survival, 
Computer Methods and Programs in Biomedicine 168: 21-37.
</p>
<p>Witten DM, Tibshirani R (2010) Survival analysis with high-dimensional covariates. Stat Method Med Res 19:29-51
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lung)
t.vec=Lung$t.vec[Lung$train==TRUE]
d.vec=Lung$d.vec[Lung$train==TRUE]
X.mat=Lung[Lung$train==TRUE,-c(1,2,3)]
uni.Wald(t.vec, d.vec, X.mat)
</code></pre>

<hr>
<h2 id='X.pathway'>Generate a matrix of gene expressions in the presence of gene pathways</h2><span id='topic+X.pathway'></span>

<h3>Description</h3>

<p>Generate a matrix of gene expressions in the presence of gene pathways (Scenario 2 of Emura et al. (2012)).</p>


<h3>Usage</h3>

<pre><code class='language-R'>X.pathway(n,p,q1,q2,rho1=0.5,rho2=0.5)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="X.pathway_+3A_n">n</code></td>
<td>
<p>the number of individuals (sample size)</p>
</td></tr>
<tr><td><code id="X.pathway_+3A_p">p</code></td>
<td>
<p>the number of genes</p>
</td></tr>
<tr><td><code id="X.pathway_+3A_q1">q1</code></td>
<td>
<p>the number of genes in the first pathway</p>
</td></tr>
<tr><td><code id="X.pathway_+3A_q2">q2</code></td>
<td>
<p>the number of genes in the second pathway</p>
</td></tr>
<tr><td><code id="X.pathway_+3A_rho1">rho1</code></td>
<td>
<p>the correlation coefficient for the first pathway</p>
</td></tr>
<tr><td><code id="X.pathway_+3A_rho2">rho2</code></td>
<td>
<p>the correlation coefficient for the second pathway</p>
</td></tr>
</table>


<h3>Details</h3>

<p>n by p matrix of gene expressions are generated. Correlation between columns (genes) is introduced to reflect the presence of gene pathways. The distribution of each column (gene) is standardized to have mean=0 and SD=1.
Two blocks of correlated genes (i.e., two gene pathways) are generated, where the first pathway include &quot;q1&quot; genes and the second pathway includes &quot;q2&quot; genes. The last p-q1-q2 genes are independent genes.
If two genes are correlated, the correlation is 0.5 (can be changed by option &quot;rho1&quot; and &quot;rho2&quot;).
Details are referred to p.4 of Emura et al. (2012).
This deta generation scheme was used in the simulations of Emura et al. (2012), Emura and Chen (2016) and
Emura et al. (2017).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>n by p matrix of gene expressions</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura &amp; Yi-Hau Chen</p>


<h3>References</h3>

<p>Emura T, Chen YH, Chen HY (2012). Survival Prediction Based on Compound Covariate under Cox Proportional Hazard Models.
PLoS ONE 7(10): e47627. doi:10.1371/journal.pone.0047627
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring:
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57
</p>
<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2017) Personalized dynamic prediction of death according to
tumour progression and high-dimensional genetic factors: meta-analysis with a joint model,
Stat Methods Med Res, doi:10.1177/0962280216688032
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## generate 6 gene expressions from 10 individuals
X.pathway(n=10,p=6,q1=2,q2=2)

## generate 200 gene expressions and check the mean and SD
X.mat=X.pathway(n=200,p=100,q1=10,q2=10)
round( colMeans(X.mat),3 ) ## mean ~ 0 ##
round( apply(X.mat, MARGIN=2, FUN=sd),3) ## SD ~ 1 ##

## Change correlation coefficients by option "rho1" and "rho2" ##
X.mat=X.pathway(n=10000,p=6,q1=2,q2=2,rho1=0.2,rho2=0.8)
round( colMeans(X.mat),2 ) ## mean ~ 0 ##
round( apply(X.mat, MARGIN=2, FUN=sd),2) ## SD ~ 1 ##
round( cor(X.mat),1) ## Correlation matrix ##
</code></pre>

<hr>
<h2 id='X.tag'>Generate a matrix of gene expressions in the presence of tag genes</h2><span id='topic+X.tag'></span>

<h3>Description</h3>

<p>Generate a matrix of gene expressions in the presence of tag genes (Scenario 1 of Emura et al. (2012)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X.tag(n, p, q, s = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="X.tag_+3A_n">n</code></td>
<td>
<p>the number of individuals (sample size)</p>
</td></tr>
<tr><td><code id="X.tag_+3A_p">p</code></td>
<td>
<p>the number of genes</p>
</td></tr>
<tr><td><code id="X.tag_+3A_q">q</code></td>
<td>
<p>the number of non-null genes</p>
</td></tr>
<tr><td><code id="X.tag_+3A_s">s</code></td>
<td>
<p>the number of null genes correlated with a non-null gene (tag)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>n by p matrix of gene expressions are generated. 
Correlation between columns is introduced to reflect the presence of tag genes. 
The distribution of each column is standardized to have mean=0 and SD=1.
If two genes are correlated, the correlation is 0.5. Otherwise, the correlation is 0. 
Details are referred to p.4 of Emura et al. (2012).
This deta generation scheme was used in the simulations of Emura et al. (2012) and Emura and Chen (2016).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>n by p matrix of gene expressions</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura &amp; Yi-Hau Chen</p>


<h3>References</h3>

<p>Emura T, Chen YH, Chen HY (2012). Survival Prediction Based on Compound Covariate under Cox Proportional Hazard Models.
PLoS ONE 7(10): e47627. doi:10.1371/journal.pone.0047627
</p>
<p>Emura T, Chen YH (2016). Gene Selection for Survival Data Under Dependent Censoring: 
a Copula-based Approach, Stat Methods Med Res 25(No.6): 2840-57.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X.mat=X.tag(n=200,p=100,q=10,s=4)
round( colMeans(X.mat),3 ) ## mean ~ 0 ##
round( apply(X.mat, MARGIN=2, FUN=sd),3) ## SD ~ 1 ##
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
