<!DOCTYPE html><html><head><title>Help for package comradesOO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {comradesOO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clusterComrades'><p>clusterComrades</p></a></li>
<li><a href='#clusterGrangesList'><p>clusterGrangesList</p></a></li>
<li><a href='#clusterGrangesList&lt;-'><p>clusterGrangesList&lt;-</p></a></li>
<li><a href='#clusterNumbers'><p>clusterNumbers</p></a></li>
<li><a href='#clusterTableFolded'><p>clusterTableFolded</p></a></li>
<li><a href='#clusterTableList'><p>clusterTableList</p></a></li>
<li><a href='#clusterTableList&lt;-'><p>clusterTableList&lt;-</p></a></li>
<li><a href='#compareKnown'><p>compareKnown</p></a></li>
<li><a href='#compareKnownStructures'><p>compareKnownStructures</p></a></li>
<li><a href='#comradesDataSet-class'><p>comradesDataSet</p></a></li>
<li><a href='#comradesOO'><p>comradesOO: A package for analysing a COMRADES dataset</p></a></li>
<li><a href='#featureInfo'><p>featureInfo</p></a></li>
<li><a href='#findBasePairsRNAcoFold2'><p>findBasePairsRNAcoFold2</p></a></li>
<li><a href='#findBasePairsRNAfold'><p>findBasePairsRNAfold</p></a></li>
<li><a href='#findBasePairsRNAfold2'><p>findBasePairsRNAfold2</p></a></li>
<li><a href='#foldComrades'><p>foldComrades</p></a></li>
<li><a href='#getAdjacancyMat'><p>getAdjacancyMat</p></a></li>
<li><a href='#getClusterClusterShortRangeWhole'><p>getClusterClusterShortRangeWhole</p></a></li>
<li><a href='#getData'><p>getData</p></a></li>
<li><a href='#getInteractions'><p>getInteractions</p></a></li>
<li><a href='#getMatrices'><p>getMatrices</p></a></li>
<li><a href='#getReverseInteractions'><p>getReverseInteractions</p></a></li>
<li><a href='#group'><p>group</p></a></li>
<li><a href='#hybFiles'><p>hybFiles</p></a></li>
<li><a href='#hybToGRanges'><p>hybToGRanges</p></a></li>
<li><a href='#makeExampleComradesDataSet'><p>makeExampleComradesDataSet</p></a></li>
<li><a href='#matrixList'><p>matrixList</p></a></li>
<li><a href='#matrixList&lt;-'><p>matrixList</p></a></li>
<li><a href='#plotComparisonArc'><p>plotComparisonArc</p></a></li>
<li><a href='#plotEnsemblePCA'><p>plotEnsemblePCA</p></a></li>
<li><a href='#plotMatrices'><p>Plots a number of contact maps to file of each sample for a stage in the analysis</p></a></li>
<li><a href='#plotMatricesAverage'><p>plotMatricesAverage</p></a></li>
<li><a href='#plotStructure'><p>plotStructure</p></a></li>
<li><a href='#printClustersFast'><p>printClustersFast</p></a></li>
<li><a href='#readNumbers'><p>readNumbers</p></a></li>
<li><a href='#rnas'><p>rnas</p></a></li>
<li><a href='#rnaSize'><p>rnaSize</p></a></li>
<li><a href='#sampleChimeras'><p>sampleChimeras</p></a></li>
<li><a href='#sampleNames'><p>sampleNames</p></a></li>
<li><a href='#sampleTable'><p>sampleTable</p></a></li>
<li><a href='#subsetHybList2'><p>subsetHybList2</p></a></li>
<li><a href='#swapHybs'><p>swapHybs</p></a></li>
<li><a href='#swapHybs2'><p>swapHybs2</p></a></li>
<li><a href='#swapHybs3'><p>swapHybs3</p></a></li>
<li><a href='#topInteracters'><p>topInteracters</p></a></li>
<li><a href='#topInteractions'><p>topInteractions</p></a></li>
<li><a href='#topTranscripts'><p>topTranscripts</p></a></li>
<li><a href='#trimClusters'><p>trimClusters</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of COMRADES (Cross-Linking Matched RNA and Deep
Sequencing) Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jonathan Price &lt;jlp76@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of RNA crosslinking data for RNA structure prediction. The package is suitable for the analysis of RNA structure cross-linking data and chemical probing data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/JLP-BioInf/comradesOO/issues">https://github.com/JLP-BioInf/comradesOO/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>seqinr, GenomicRanges, stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, reshape2, MASS, mixtools, utils, S4Vectors,
doParallel, igraph, R4RNA, RColorBrewer, IRanges,
foreach,grDevices, heatmap3, TopDom, tidyverse, RRNA, ggrepel,
methods, parallel</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Collate:</td>
<td>'comradesOO.R' 'comradesDataSet.R' 'clusterComrades.R'
'clusterComradesMethodsAndHelpers.R'
'commonHelpersAndMethods.R' 'commonStatsAndPlots.R'
'comradesDataSetMethodsAndHelpers.R' 'comradesOO-package.R'
'foldComrades.R' 'foldComradesMethodsAndHelpers.R'
'genericMethods.R'</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-10 12:17:30 UTC; jp</td>
</tr>
<tr>
<td>Author:</td>
<td>Jonathan Price [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-10 13:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='clusterComrades'>clusterComrades</h2><span id='topic+clusterComrades'></span><span id='topic+clusterComrades+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method clusters the duplexes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterComrades(cds, cores = 3, stepCount = 2, clusterCutoff = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterComrades_+3A_cds">cds</code></td>
<td>
<p>comradesDataSet object created with comradesDataSet</p>
</td></tr>
<tr><td><code id="clusterComrades_+3A_cores">cores</code></td>
<td>
<p>numeric - The number of cores to use</p>
</td></tr>
<tr><td><code id="clusterComrades_+3A_stepcount">stepCount</code></td>
<td>
<p>Stringency for clustering</p>
</td></tr>
<tr><td><code id="clusterComrades_+3A_clustercutoff">clusterCutoff</code></td>
<td>
<p>The minimum number of reads a cluster requires</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>comradesDataSet</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 0)
</code></pre>

<hr>
<h2 id='clusterGrangesList'>clusterGrangesList</h2><span id='topic+clusterGrangesList'></span><span id='topic+clusterGrangesList+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the cluster coordinates in granges format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterGrangesList(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterGrangesList_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of Granges objects showing the positions of each cluster, one entry for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusterGrangesList(cds)
</code></pre>

<hr>
<h2 id='clusterGrangesList+26lt+3B-'>clusterGrangesList&lt;-</h2><span id='topic+clusterGrangesList+3C-'></span><span id='topic+clusterGrangesList+3C-+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Set new clusterGrangesList slot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterGrangesList(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterGrangesList+2B26lt+2B3B-_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="clusterGrangesList+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>A replacement</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return - Sets a new clusterGrangesList slot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

newclusterGrangesList &lt;- clusterGrangesList(cds)
clusterGrangesList(cds) &lt;- newclusterGrangesList
</code></pre>

<hr>
<h2 id='clusterNumbers'>clusterNumbers</h2><span id='topic+clusterNumbers'></span><span id='topic+clusterNumbers+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints a table showing the number of clusters in each step 
of the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterNumbers(knowClusteredCds, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterNumbers_+3A_knowclusteredcds">knowClusteredCds</code></td>
<td>
<p>A comradesDataSet object after clustering has been performed</p>
</td></tr>
<tr><td><code id="clusterNumbers_+3A_rna">rna</code></td>
<td>
<p>The RNA ID of interest - use rna(cdsObject).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame shoing the number of clusters for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusteredCds = clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 1)
clusterNumbers(clusteredCds)
</code></pre>

<hr>
<h2 id='clusterTableFolded'>clusterTableFolded</h2><span id='topic+clusterTableFolded'></span><span id='topic+clusterTableFolded+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the cluster coordinates with fold prediciton in data frame format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterTableFolded(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterTableFolded_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table showing the vienna structures of each cluster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusterTableFolded(cds)
</code></pre>

<hr>
<h2 id='clusterTableList'>clusterTableList</h2><span id='topic+clusterTableList'></span><span id='topic+clusterTableList+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the cluster coordinates in data frame format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterTableList(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterTableList_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables showing the vienna structures of each cluster
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusterTableList(cds)
</code></pre>

<hr>
<h2 id='clusterTableList+26lt+3B-'>clusterTableList&lt;-</h2><span id='topic+clusterTableList+3C-'></span><span id='topic+clusterTableList+3C-+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Set new clusterTableList slot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterTableList(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterTableList+2B26lt+2B3B-_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="clusterTableList+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>A replacement</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return - Sets a new clusterTableList slot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

newclusterGrangesList &lt;- clusterTableList(cds)
clusterTableList(cds) &lt;- newclusterGrangesList
</code></pre>

<hr>
<h2 id='compareKnown'>compareKnown</h2><span id='topic+compareKnown'></span><span id='topic+compareKnown+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method compares the current object to a know structure.run 
<code>trimClusters()</code> on the  <code>comradesDataSet</code> first
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareKnown(trimmedClusters, knownMat, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareKnown_+3A_trimmedclusters">trimmedClusters</code></td>
<td>
<p>a <code>comradesDataSet</code> object, 
run <code>trimClusters()</code> on the  <code>comradesDataSet</code> first</p>
</td></tr>
<tr><td><code id="compareKnown_+3A_knownmat">knownMat</code></td>
<td>
<p>Matrix - A marix(ncol = lengthRNA,nrow = lengthRNA) where a
value in matrix[x,y] would indicate a known interation between nucleotide 
x and nucleotide y</p>
</td></tr>
<tr><td><code id="compareKnown_+3A_type">type</code></td>
<td>
<p>string - the Analysis stage of clusters you would like to compare you can find 
available types by just running the objects name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>comradesClusteredDataSet</code> object
</p>
<p>The 3 attributes matrixList, clusterTableList and clusterGrangesList 
will gain the <code>types</code> &quot;known&quot; and &quot;novel&quot; and &quot;knownAndNovel&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusteredCds = clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 0)
knownMat = matrix(0, ncol = rnaSize(cds), nrow = rnaSize(cds))
knownMat[7,27] = 1
# use compare known to gett he known and not know clusters
knowClusteredCds = compareKnown(clusteredCds,
                                knownMat,
                                "original")
clusterNumbers(knowClusteredCds)                 
                

</code></pre>

<hr>
<h2 id='compareKnownStructures'>compareKnownStructures</h2><span id='topic+compareKnownStructures'></span><span id='topic+compareKnownStructures+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method compares the predicted structures to a set of known interactions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareKnownStructures(foldedCds, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareKnownStructures_+3A_foldedcds">foldedCds</code></td>
<td>
<p><code>comradesDataSet</code> after running foldComrades</p>
</td></tr>
<tr><td><code id="compareKnownStructures_+3A_file">file</code></td>
<td>
<p>a two column file with column header i and j with numeric values showing
nucleoide i binds to nucleotide j</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe
</p>
<p>a tables showing the number of predicted interactions and their agreement
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cds = makeExampleComradesDataSet()
clusteredCds = clusterComrades(cds = cds,
                               cores = 3,
                               stepCount = 2,
                               clusterCutoff = 1)


trimmedClusters = trimClusters(clusteredCds = clusteredCds,trimFactor = 1, clusterCutoff = 1)



fasta = paste(c(rep('A',25),
                rep('T',25),
                rep('A',10),
                rep('T',23)),collapse = "")

header = '&gt;transcript1'


fastaFile = tempfile()
writeLines(paste(header,fasta,sep = "\n"),con = fastaFile)


rnaRefs = list()
rnaRefs[[rnas(cds)]] = read.fasta(fastaFile)
rnaRefs



foldedCds = foldComrades(trimmedClusters,
                         rnaRefs = rnaRefs,
                         start = 1,
                         end = 83,
                         shape = 0,
                         ensembl = 5,
                         constraintNumber  = 1,
                         evCutoff = 1)


# make an example table of "know" interactions
file = data.frame(V1 = c(6), 
                  V2 = c(80))
compareKnownStructures(foldedCds,file)

## End(Not run)
</code></pre>

<hr>
<h2 id='comradesDataSet-class'>comradesDataSet</h2><span id='topic+comradesDataSet-class'></span><span id='topic+comradesDataSet'></span>

<h3>Description</h3>

<p><code>comradesDataSet</code> objects are used to store the input meta-data, data and
create a framework for the storage of results. Whilst creating the object,
the original hyb files are also filtered for the RNA of interest. Check the 
package vignette for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comradesDataSet(rnas, rnaSize = 0, sampleTable)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comradesDataSet-class_+3A_rnas">rnas</code></td>
<td>
<p>vector - The names of the RNA interest, these must be displayed
the same way as in the input Hyb Files.</p>
</td></tr>
<tr><td><code id="comradesDataSet-class_+3A_rnasize">rnaSize</code></td>
<td>
<p>named list - The sizes (nt) of the RNAs of interest, the list
elements must have same names as the <code>rnas</code> vector and each each contain
one numeric value.</p>
</td></tr>
<tr><td><code id="comradesDataSet-class_+3A_sampletable">sampleTable</code></td>
<td>
<p>string - The address of the sample table, the sample table
must have 4 columns, fileName (the full path and file name of the input
hyb file for each sample ), group (&quot;s&quot; - sample or &quot;c&quot; - control),
sample (1,2,3, etc), sampleName (must be unique).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A comradesDataSet object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>clusterTableFolded</code></dt><dd><p>table - a table similar to the <code>clusterTableList</code>
it contains coordinates of the clusters along with vienna format fold and
RNA sequences for each cluster</p>
</dd>
<dt><code>clusterTableList</code></dt><dd><p>List - Follows the pattern for list slots of comradesDataSet
objects, <code>matrixList(cds)[[rna]][[type]][[sample]]</code>. contains a table
with coordinates and information about the clusters identified</p>
</dd>
<dt><code>clusterGrangesList</code></dt><dd><p>List - Follows the pattern for list slots of comradesDataSet
objects, <code>matrixList(cds)[[rna]][[type]][[sample]]</code>. contains GRanges
objects of the original duplexes with their cluster membership</p>
</dd>
<dt><code>sampleTable</code></dt><dd><p>table - Column names; fileName, group (s or c),
sample (1,2,3, etc), sampleName (must be unique)</p>
</dd>
<dt><code>rnas</code></dt><dd><p>string - a single RNA to analyse - must be present in <code>rnas(cdsObject)</code></p>
</dd>
<dt><code>rnaSize</code></dt><dd><p>if set to 0 this will be calculated</p>
</dd>
<dt><code>matrixList</code></dt><dd><p>List - Follows the pattern for list slots of comradesDataSet
objects, <code>matrixList(cds)[[rna]][[type]][[sample]]</code>. Contains a set
of contact matrices, each cell contains the number of duplexes identified
for position x,y.</p>
</dd>
<dt><code>hybFiles</code></dt><dd><p>List - Follows the pattern for list slots of comradesDataSet
objects, <code>hybFiles(cds)[[rna]][[type]][[sample]]</code>. Contains a set of
tables, these are the original Hyb files that were read in.</p>
</dd>
<dt><code>interactionTable</code></dt><dd><p>Table of interactions discovered in step1 of the folding</p>
</dd>
<dt><code>viennaStructures</code></dt><dd><p>List of vienna format structures from final prediction</p>
</dd>
<dt><code>dgs</code></dt><dd><p>List of free energies</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'># make example input
c4 = c(rep("transcript1",100),rep("transcript2",100) )
c10 = c(rep("transcript1",200) )
c1 = 1:200
c2 = rep(paste(rep("A", 40), collapse = ""),200)
c3 = rep(".",200)
c9 = rep(".",200)
c15 = rep(".",200)
c5 = rep(1,200)
c11 = rep(21,200)
c6 = rep(20,200)
c12= rep(40,200)
# short distance 50
c7 = sample(1:5, 50, replace = TRUE)
c8 = sample(20:25, 50, replace = TRUE)
c13 = sample(20:25, 50, replace = TRUE)
c14 = sample(40:45, 50, replace = TRUE)
# long distance 50
c7 = c(c7,sample(1:5, 50, replace = TRUE))
c8 = c(c8,sample(20:25, 50, replace = TRUE))
c13 = c(c13,sample(60:70, 50, replace = TRUE))
c14 = c(c14,sample(80:83, 50, replace = TRUE))
# inter RNA 100
c7 = c(c7,sample(1:5, 100, replace = TRUE))
c8 = c(c8,sample(20:25, 100, replace = TRUE))
c13 = c(c13,sample(1:5, 100, replace = TRUE))
c14 = c(c14,sample(20:25, 100, replace = TRUE))
exampleInput = data.frame(V1 = c1,
                         V2 = c2,
                         V3 = c3,
                         V4 = c4,
                         V5 = as.numeric(c5),
                         V6 = as.numeric(c6),
                         V7 = as.numeric(c7),
                         V8 = as.numeric(c8),
                         V9 = c9,
                         V10 = c10,
                         V11 = as.numeric(c11),
                         V12 = as.numeric(c12),
                         V13 = as.numeric(c13),
                         V14 = as.numeric(c14),
                         V15 = c15)
file = tempfile()
write.table(exampleInput,file = file, 
quote = FALSE, 
row.names = FALSE, sep = "\t", col.names = FALSE)


# Set up the sample table 
sampleTabler1 = c(file, "s", "1", "s1")
sampleTabler2 = c(file, "c", "1", "c1")
# make the sample table 
sampleTable2 = rbind.data.frame(sampleTabler1, sampleTabler2)
# add the column names 
colnames(sampleTable2) = c("file", "group", "sample", "sampleName")

# Choose RNA and set up the object ----
rna = c("transcript1")


# load the object
cds = comradesDataSet(rnas = rna,
                     rnaSize = 0,
                     sampleTable = sampleTable2)

cds

</code></pre>

<hr>
<h2 id='comradesOO'>comradesOO: A package for analysing a COMRADES dataset</h2><span id='topic+comradesOO'></span><span id='topic+comradesOO-package'></span>

<h3>Description</h3>

<p>The package consists of 1 Object:
comradesDataSet
</p>
<p>The package consists of 1 Object:
comradesDataSet
</p>


<h3><code><a href="#topic+comradesDataSet">comradesDataSet</a></code></h3>

<p>NA
</p>
<p>NA
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jonathan Price <a href="mailto:jlp76@cam.ac.uk">jlp76@cam.ac.uk</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li><p> Report bugs at <a href="https://github.com/JLP-BioInf/comradesOO/issues">https://github.com/JLP-BioInf/comradesOO/issues</a>
</p>
</li></ul>

<p>Useful links:
</p>

<ul>
<li><p> Report bugs at <a href="https://github.com/JLP-BioInf/comradesOO/issues">https://github.com/JLP-BioInf/comradesOO/issues</a>
</p>
</li></ul>


<hr>
<h2 id='featureInfo'>featureInfo</h2><span id='topic+featureInfo'></span><span id='topic+featureInfo+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Produces a list list of 2 elemnts 'transcript' and 'family'
Each element contains a table with the counts for each RNA in each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>featureInfo(cds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="featureInfo_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list - Feature level and transcript level counts for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
featureInfo(cds)

</code></pre>

<hr>
<h2 id='findBasePairsRNAcoFold2'>findBasePairsRNAcoFold2</h2><span id='topic+findBasePairsRNAcoFold2'></span>

<h3>Description</h3>

<p>Folds the clusters using Vienna RNAfold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBasePairsRNAcoFold2(
  startPos1,
  endPos1,
  seq1,
  startPos2,
  endPos2,
  seq2,
  fasta,
  shape
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_startpos1">startPos1</code></td>
<td>
<p>Start of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_endpos1">endPos1</code></td>
<td>
<p>End of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_seq1">seq1</code></td>
<td>
<p>Sequence of x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_startpos2">startPos2</code></td>
<td>
<p>Start of the cluster side y</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_endpos2">endPos2</code></td>
<td>
<p>End of the cluster side y</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_seq2">seq2</code></td>
<td>
<p>Sequence of y</p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_fasta">fasta</code></td>
<td>
<p><code>rnaRefs</code></p>
</td></tr>
<tr><td><code id="findBasePairsRNAcoFold2_+3A_shape">shape</code></td>
<td>
<p>shape reactivities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table of clusters and coordinates with folds
</p>

<hr>
<h2 id='findBasePairsRNAfold'>findBasePairsRNAfold</h2><span id='topic+findBasePairsRNAfold'></span>

<h3>Description</h3>

<p>Folds the clusters using Vienna RNA duplex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBasePairsRNAfold(startPos, endPos, seqs, fasta, shape)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findBasePairsRNAfold_+3A_startpos">startPos</code></td>
<td>
<p>Start of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold_+3A_endpos">endPos</code></td>
<td>
<p>End of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold_+3A_seqs">seqs</code></td>
<td>
<p>Sequence of x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold_+3A_fasta">fasta</code></td>
<td>
<p><code>rnaRefs</code></p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold_+3A_shape">shape</code></td>
<td>
<p>shape reactivities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table of clusters and coordinates with folds
</p>

<hr>
<h2 id='findBasePairsRNAfold2'>findBasePairsRNAfold2</h2><span id='topic+findBasePairsRNAfold2'></span>

<h3>Description</h3>

<p>Folds the clusters using Vienna RNA duplex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBasePairsRNAfold2(startPos, endPos, seqs, fasta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findBasePairsRNAfold2_+3A_startpos">startPos</code></td>
<td>
<p>Start of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold2_+3A_endpos">endPos</code></td>
<td>
<p>End of the cluster side x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold2_+3A_seqs">seqs</code></td>
<td>
<p>Sequence of x</p>
</td></tr>
<tr><td><code id="findBasePairsRNAfold2_+3A_fasta">fasta</code></td>
<td>
<p><code>rnaRefs</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table of clusters and coordinates with folds
</p>

<hr>
<h2 id='foldComrades'>foldComrades</h2><span id='topic+foldComrades'></span><span id='topic+foldComrades+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This methods folds an ensebl of structures for the whole RNA or chosen region
of the RNA. See <code>comradesDataSet</code> for slot information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>foldComrades(
  cdsObject,
  rnaRefs,
  start,
  end,
  evCutoff = 1,
  ensembl = 50,
  constraintNumber = 20,
  shape = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="foldComrades_+3A_cdsobject">cdsObject</code></td>
<td>
<p>comradesDataSet object created with comradesDataSet</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_rnarefs">rnaRefs</code></td>
<td>
<p>named List - a list with named elements that correspond to the
.RNA of interest. The element of the list must be a fasta file that has
been read with <code>seqinr::read.fasta()</code></p>
</td></tr>
<tr><td><code id="foldComrades_+3A_start">start</code></td>
<td>
<p>Start of segmnent to fold</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_end">end</code></td>
<td>
<p>End of segmnent to fold</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_evcutoff">evCutoff</code></td>
<td>
<p>Mininum number of read support for contraint to be included in folding</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_ensembl">ensembl</code></td>
<td>
<p>Number of structures to Nake</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_constraintnumber">constraintNumber</code></td>
<td>
<p>Number of constraints to add to each final fold</p>
</td></tr>
<tr><td><code id="foldComrades_+3A_shape">shape</code></td>
<td>
<p>shape reactivities (0 for no constraints)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a comradesDataSet object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cds = makeExampleComradesDataSet()

clusteredCds = clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 0)
                
trimmedClusters = trimClusters(clusteredCds = clusteredCds,
             trimFactor = 1, 
             clusterCutoff = 0)
             
              

fasta = paste(c(rep('A',25),
                rep('T',25),
                rep('A',10),
                rep('T',23)),collapse = "")

header = '&gt;transcript1'


fastaFile = tempfile()
writeLines(paste(header,fasta,sep = "\n"),con = fastaFile)


rnaRefs = list()
rnaRefs[[rnas(cds)]] = read.fasta(fastaFile)
rnaRefs



foldedCds = foldComrades(trimmedClusters,
                         rnaRefs = rnaRefs,
                         start = 1,
                         end = 83,
                         shape = 0,
                         ensembl = 5,
                         constraintNumber  = 1,
                         evCutoff = 1)
foldedCds

## End(Not run)
</code></pre>

<hr>
<h2 id='getAdjacancyMat'>getAdjacancyMat</h2><span id='topic+getAdjacancyMat'></span>

<h3>Description</h3>

<p>Makes and adjacency matrix list (for clustering)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAdjacancyMat(hybGranges, nucletideOrPerc, cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAdjacancyMat_+3A_hybgranges">hybGranges</code></td>
<td>
<p>list created with hybToGRanges (but just the gap section of the list)</p>
</td></tr>
<tr><td><code id="getAdjacancyMat_+3A_nucletideorperc">nucletideOrPerc</code></td>
<td>
<p>measure difference by percentage or nucleotides</p>
</td></tr>
<tr><td><code id="getAdjacancyMat_+3A_cutoff">cutoff</code></td>
<td>
<p>The maximum difference before giving these two gaps 0</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Makes and adjacency matrix list (for clustering)
</p>


<h3>Value</h3>

<p>A list of Adjacancy matrices
</p>

<hr>
<h2 id='getClusterClusterShortRangeWhole'>getClusterClusterShortRangeWhole</h2><span id='topic+getClusterClusterShortRangeWhole'></span>

<h3>Description</h3>

<p>Decides if a cluster is long or short range
then either grabs the whole sequence or the sequence of the two sides of the
interaction separately.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getClusterClusterShortRangeWhole(cluster, seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getClusterClusterShortRangeWhole_+3A_cluster">cluster</code></td>
<td>
<p>cluster positions</p>
</td></tr>
<tr><td><code id="getClusterClusterShortRangeWhole_+3A_seq">seq</code></td>
<td>
<p>sequence of transcript</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same table with an extra column
</p>

<hr>
<h2 id='getData'>getData</h2><span id='topic+getData'></span><span id='topic+getData+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Get data is more generic method for retrieving data from the object
and returns a list, the number of entries in the list is number of
samples in the dataset and the list contain entries of the data type
and analysis stage you select.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getData(x, data, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getData_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="getData_+3A_data">data</code></td>
<td>
<p>The data type to return &lt;hybFiles | matrixList | clusterGrangesList | clusterTableList&gt;</p>
</td></tr>
<tr><td><code id="getData_+3A_type">type</code></td>
<td>
<p>The analysis stage &lt;original | noHost | originalClusters | trimmedClusters&gt;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the chosen data type - one entry for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

getData(cds, 'matrixList','original')

</code></pre>

<hr>
<h2 id='getInteractions'>getInteractions</h2><span id='topic+getInteractions'></span><span id='topic+getInteractions+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method returns a table interactions of an RNA (interactor) on the RNA of
interest use topInteracters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getInteractions(cds, interactor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInteractions_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="getInteractions_+3A_interactor">interactor</code></td>
<td>
<p>The rna to show interactions with</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table showign the read coverage of the interacting RNA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
getInteractions(cds, 'transcript2')

</code></pre>

<hr>
<h2 id='getMatrices'>getMatrices</h2><span id='topic+getMatrices'></span>

<h3>Description</h3>

<p>Make a matrix of contact interactions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMatrices(hybList, rna, size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMatrices_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles or subsetHybList</p>
</td></tr>
<tr><td><code id="getMatrices_+3A_rna">rna</code></td>
<td>
<p>the RNA of interest that you want to subset</p>
</td></tr>
<tr><td><code id="getMatrices_+3A_size">size</code></td>
<td>
<p>The size of the RNA</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function used to create a list of matrices for plotting with
plotMatrixList or plotMatrixListFull, the output list will be same as the
input except for an extra list layer for the specific RNA
</p>


<h3>Value</h3>

<p>A list of matrices
</p>

<hr>
<h2 id='getReverseInteractions'>getReverseInteractions</h2><span id='topic+getReverseInteractions'></span><span id='topic+getReverseInteractions+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints interactions of the RNA of interest on another RNA
transcript.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReverseInteractions(cds, interactor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReverseInteractions_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="getReverseInteractions_+3A_interactor">interactor</code></td>
<td>
<p>The rna to show interactions with</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A long format table shoing the read coverage of chosen RNA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
getReverseInteractions(cds, 'transcript2')

</code></pre>

<hr>
<h2 id='group'>group</h2><span id='topic+group'></span><span id='topic+group+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the indeces for each group for the instance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list - The indices of the sample in the control and sample groups
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

group(cds)
</code></pre>

<hr>
<h2 id='hybFiles'>hybFiles</h2><span id='topic+hybFiles'></span><span id='topic+hybFiles+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the data in original format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hybFiles(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hybFiles_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables in the original input format, one entry for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

hybFiles(cds)
</code></pre>

<hr>
<h2 id='hybToGRanges'>hybToGRanges</h2><span id='topic+hybToGRanges'></span>

<h3>Description</h3>

<p>This function is useful to turn a list of hyb data into lists of GRanges
It creates a list for each sample one for the left side one for the right
side and one for the gap in the middle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hybToGRanges(hybList, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hybToGRanges_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles or subsetHybList</p>
</td></tr>
<tr><td><code id="hybToGRanges_+3A_rna">rna</code></td>
<td>
<p>The rna of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of GRanges data in hyb format
</p>

<hr>
<h2 id='makeExampleComradesDataSet'>makeExampleComradesDataSet</h2><span id='topic+makeExampleComradesDataSet'></span>

<h3>Description</h3>

<p>Creat a minimal example comradesdataSetObject
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeExampleComradesDataSet()
</code></pre>


<h3>Value</h3>

<p>An example comradesDataSet objext
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
</code></pre>

<hr>
<h2 id='matrixList'>matrixList</h2><span id='topic+matrixList'></span><span id='topic+matrixList+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the contact matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixList(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixList_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of contract matrices, one entry for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

matrixList(cds)
</code></pre>

<hr>
<h2 id='matrixList+26lt+3B-'>matrixList</h2><span id='topic+matrixList+3C-'></span><span id='topic+matrixList+3C-+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Set new matrixList slot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixList(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixList+2B26lt+2B3B-_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="matrixList+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>A replacement</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return - Sets a new matrixList slot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

newMatrixList &lt;- matrixList(cds)
matrixList(cds) &lt;- newMatrixList
</code></pre>

<hr>
<h2 id='plotComparisonArc'>plotComparisonArc</h2><span id='topic+plotComparisonArc'></span><span id='topic+plotComparisonArc+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method plots two structures chosen from the
plotEnsemblePCA method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotComparisonArc(foldedCds, s1 = "s1", s2 = "s2", n1 = 1, n2 = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotComparisonArc_+3A_foldedcds">foldedCds</code></td>
<td>
<p><code>comradesDataSet</code> after running foldComrades</p>
</td></tr>
<tr><td><code id="plotComparisonArc_+3A_s1">s1</code></td>
<td>
<p>sample of structure 1</p>
</td></tr>
<tr><td><code id="plotComparisonArc_+3A_s2">s2</code></td>
<td>
<p>sample of structure 2</p>
</td></tr>
<tr><td><code id="plotComparisonArc_+3A_n1">n1</code></td>
<td>
<p>number of structure from first sample</p>
</td></tr>
<tr><td><code id="plotComparisonArc_+3A_n2">n2</code></td>
<td>
<p>number of structure from first sample</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an ark diagram of the two strcutures
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cds = makeExampleComradesDataSet()
clusteredCds = clusterComrades(cds = cds,
                               cores = 3,
                               stepCount = 2,
                               clusterCutoff = 1)


trimmedClusters = trimClusters(clusteredCds = clusteredCds,trimFactor = 1, clusterCutoff = 1)



fasta = paste(c(rep('A',25),
                rep('T',25),
                rep('A',10),
                rep('T',23)),collapse = "")

header = '&gt;transcript1'


fastaFile = tempfile()
writeLines(paste(header,fasta,sep = "\n"),con = fastaFile)


rnaRefs = list()
rnaRefs[[rnas(cds)]] = read.fasta(fastaFile)
rnaRefs



foldedCds = foldComrades(trimmedClusters,
                         rnaRefs = rnaRefs,
                         start = 1,
                         end = 83,
                         shape = 0,
                         ensembl = 5,
                         constraintNumber  = 1,
                         evCutoff = 1)


plotComparisonArc(foldedCds,"s1","s1",1,3)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotEnsemblePCA'>plotEnsemblePCA</h2><span id='topic+plotEnsemblePCA'></span><span id='topic+plotEnsemblePCA+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method plots a PCA of the ensembl
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEnsemblePCA(foldedCds, labels = TRUE, split = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEnsemblePCA_+3A_foldedcds">foldedCds</code></td>
<td>
<p><code>comradesDataSet</code> after running foldComrades</p>
</td></tr>
<tr><td><code id="plotEnsemblePCA_+3A_labels">labels</code></td>
<td>
<p>plot with labels or not (TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="plotEnsemblePCA_+3A_split">split</code></td>
<td>
<p>split the plot using facets based on the samples  (TRUE/FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a PCA plot of the ensembl
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cds = makeExampleComradesDataSet()
clusteredCds = clusterComrades(cds = cds,
                               cores = 3,
                               stepCount = 2,
                               clusterCutoff = 1)


trimmedClusters = trimClusters(clusteredCds = clusteredCds,trimFactor = 1, clusterCutoff = 1)



fasta = paste(c(rep('A',25),
                rep('T',25),
                rep('A',10),
                rep('T',23)),collapse = "")

header = '&gt;transcript1'


fastaFile = tempfile()
writeLines(paste(header,fasta,sep = "\n"),con = fastaFile)


rnaRefs = list()
rnaRefs[[rnas(cds)]] = read.fasta(fastaFile)
rnaRefs



foldedCds = foldComrades(trimmedClusters,
                         rnaRefs = rnaRefs,
                         start = 1,
                         end = 83,
                         shape = 0,
                         ensembl = 5,
                         constraintNumber  = 1,
                         evCutoff = 1)


plotEnsemblePCA(foldedCds)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotMatrices'>Plots a number of contact maps to file of each sample for a stage in the analysis</h2><span id='topic+plotMatrices'></span><span id='topic+plotMatrices+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Plots a number of contact maps to file of each sample for a stage in the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMatrices(
  cds,
  type = "original",
  directory = 0,
  a = 1,
  b = 50,
  c = 1,
  d = 50,
  h = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMatrices_+3A_cds">cds</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_type">type</code></td>
<td>
<p>The analysis stage to plot</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_directory">directory</code></td>
<td>
<p>An output directory for the heatmap (use 0 for no output)</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_a">a</code></td>
<td>
<p>To make a subsetted plot (left value on x)</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_b">b</code></td>
<td>
<p>To make a subsetted plot (right value on x)</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_c">c</code></td>
<td>
<p>To make a subsetted plot (left value on y)</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_d">d</code></td>
<td>
<p>To make a subsetted plot (right value on y)</p>
</td></tr>
<tr><td><code id="plotMatrices_+3A_h">h</code></td>
<td>
<p>Height of image (inches) ( only useful if plotting)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap of the reads in the analysis stage chosen
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

plotMatrices(cds,
            b = rnaSize(cds),
            d = rnaSize(cds))
</code></pre>

<hr>
<h2 id='plotMatricesAverage'>plotMatricesAverage</h2><span id='topic+plotMatricesAverage'></span><span id='topic+plotMatricesAverage+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Plots a contact map to file of all samples for a stage in the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMatricesAverage(
  cds,
  type = "original",
  directory = 0,
  a = 1,
  b = 50,
  c = 1,
  d = 50,
  h = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMatricesAverage_+3A_cds">cds</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_type">type</code></td>
<td>
<p>The analysis stage to plot</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_directory">directory</code></td>
<td>
<p>An output directory for the heatmap (use 0 for no output)</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_a">a</code></td>
<td>
<p>To make a subsetted plot (left value on x)</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_b">b</code></td>
<td>
<p>To make a subsetted plot (right value on x)</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_c">c</code></td>
<td>
<p>To make a subsetted plot (left value on y)</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_d">d</code></td>
<td>
<p>To make a subsetted plot (right value on y)</p>
</td></tr>
<tr><td><code id="plotMatricesAverage_+3A_h">h</code></td>
<td>
<p>Height of image (inches) ( only useful if plotting)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap of the reads in the analysis stage chosen
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

plotMatricesAverage(cds,
            b = rnaSize(cds),
            d = rnaSize(cds))

</code></pre>

<hr>
<h2 id='plotStructure'>plotStructure</h2><span id='topic+plotStructure'></span><span id='topic+plotStructure+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method plots a structures chosen from the
plotEnsemblePCA method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotStructure(foldedCds, rnaRefs, s = "s1", n = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotStructure_+3A_foldedcds">foldedCds</code></td>
<td>
<p><code>comradesDataSet</code> after running foldComrades</p>
</td></tr>
<tr><td><code id="plotStructure_+3A_rnarefs">rnaRefs</code></td>
<td>
<p>A fasta of the transcript (made with seqinr::read.fasta)</p>
</td></tr>
<tr><td><code id="plotStructure_+3A_s">s</code></td>
<td>
<p>sample of structure</p>
</td></tr>
<tr><td><code id="plotStructure_+3A_n">n</code></td>
<td>
<p>number of structure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a diagram of the predicted structure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cds = makeExampleComradesDataSet()
clusteredCds = clusterComrades(cds = cds,
                               cores = 3,
                               stepCount = 2,
                               clusterCutoff = 1)


trimmedClusters = trimClusters(clusteredCds = clusteredCds,trimFactor = 1, clusterCutoff = 1)



fasta = paste(c(rep('A',25),
                rep('T',25),
                rep('A',10),
                rep('T',23)),collapse = "")

header = '&gt;transcript1'


fastaFile = tempfile()
writeLines(paste(header,fasta,sep = "\n"),con = fastaFile)


rnaRefs = list()
rnaRefs[[rnas(cds)]] = read.fasta(fastaFile)
rnaRefs



foldedCds = foldComrades(trimmedClusters,
                         rnaRefs = rnaRefs,
                         start = 1,
                         end = 83,
                         shape = 0,
                         ensembl = 5,
                         constraintNumber  = 1,
                         evCutoff = 1)


plotStructure(foldedCds,rnaRefs,"s1",3)

## End(Not run)
</code></pre>

<hr>
<h2 id='printClustersFast'>printClustersFast</h2><span id='topic+printClustersFast'></span>

<h3>Description</h3>

<p>Makes a table with the coordinates of the clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printClustersFast(dir, clustering, highest_clusters, left, right)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printClustersFast_+3A_dir">dir</code></td>
<td>
<p>the directory that contains the *hybrids.hyb files</p>
</td></tr>
<tr><td><code id="printClustersFast_+3A_clustering">clustering</code></td>
<td>
<p>The output from the iGraph function cluster_walktrap for the (made with adjacency matrix input)</p>
</td></tr>
<tr><td><code id="printClustersFast_+3A_highest_clusters">highest_clusters</code></td>
<td>
<p>The cluster you are interested in keeping</p>
</td></tr>
<tr><td><code id="printClustersFast_+3A_left">left</code></td>
<td>
<p>list created with hybToGRanges (but just the left section of the list)</p>
</td></tr>
<tr><td><code id="printClustersFast_+3A_right">right</code></td>
<td>
<p>list created with hybToGRanges (but just the right section of the list)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Does the same as printClusters but is a lot faster and does not create plots
of each cluster
</p>


<h3>Value</h3>

<p>A table of clusters and coordinates
</p>

<hr>
<h2 id='readNumbers'>readNumbers</h2><span id='topic+readNumbers'></span><span id='topic+readNumbers+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints a table showing the number of duplexes in 
the clusters in each step of the analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readNumbers(knowClusteredCds, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readNumbers_+3A_knowclusteredcds">knowClusteredCds</code></td>
<td>
<p>A comradesDataSet object after clustering has been performed</p>
</td></tr>
<tr><td><code id="readNumbers_+3A_rna">rna</code></td>
<td>
<p>The RNA ID of interest - use rna(cdsObject).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame shoing the number of reads in clusters for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusteredCds = clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 1)
readNumbers(clusteredCds)
</code></pre>

<hr>
<h2 id='rnas'>rnas</h2><span id='topic+rnas'></span><span id='topic+rnas+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the rna ID for the instance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnas(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rnas_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character - the ID of the RNA
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

rnas(cds)
</code></pre>

<hr>
<h2 id='rnaSize'>rnaSize</h2><span id='topic+rnaSize'></span><span id='topic+rnaSize+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the size of the RNA for the instance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnaSize(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rnaSize_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric - the size of the RNA (nucleotides)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

rnaSize(cds)
</code></pre>

<hr>
<h2 id='sampleChimeras'>sampleChimeras</h2><span id='topic+sampleChimeras'></span>

<h3>Description</h3>

<p>This function samples chimeras into smaller chunks so that clustering is 
quicker
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleChimeras(chimeraList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleChimeras_+3A_chimeralist">chimeraList</code></td>
<td>
<p>list of chimeras</p>
</td></tr>
</table>

<hr>
<h2 id='sampleNames'>sampleNames</h2><span id='topic+sampleNames'></span><span id='topic+sampleNames+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the sample names for the instance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleNames(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleNames_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector - the sample names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

sampleNames(cds)
</code></pre>

<hr>
<h2 id='sampleTable'>sampleTable</h2><span id='topic+sampleTable'></span><span id='topic+sampleTable+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Extract the sample table for the instance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleTable(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleTable_+3A_x">x</code></td>
<td>
<p>A comradesDataSet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame - The orginal meta-data table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

sampleTable(cds)
</code></pre>

<hr>
<h2 id='subsetHybList2'>subsetHybList2</h2><span id='topic+subsetHybList2'></span>

<h3>Description</h3>

<p>Subset a list of hyb files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetHybList2(hybList, min, max, length)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetHybList2_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles</p>
</td></tr>
<tr><td><code id="subsetHybList2_+3A_min">min</code></td>
<td>
<p>the rna of interest that you want to subset</p>
</td></tr>
<tr><td><code id="subsetHybList2_+3A_max">max</code></td>
<td>
<p>The number of randomly subsetted chimeric reads you need</p>
</td></tr>
<tr><td><code id="subsetHybList2_+3A_length">length</code></td>
<td>
<p>The number of randomly subsetted chimeric reads you need</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function used to subset a list of hyb data created by readHybFiles
This function produces the same size list as before but 
it returns ONLY the rna of interest and also
Choose duplexes where the nt difference in position between the
one side and other side of an interaction is between min and max
</p>


<h3>Value</h3>

<p>A list of subsetted hyb files
</p>

<hr>
<h2 id='swapHybs'>swapHybs</h2><span id='topic+swapHybs'></span>

<h3>Description</h3>

<p>Swap the table to ensure that 3 prime most duplex side is on he left of the
table used to make one sides heatmaps and other reasons where having the
left of the table coming after the right side is a problem. Different from
swapHybs as it ensure that BOTH duplex sides originate from the RNA of
interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swapHybs(hybList, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swapHybs_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles or subsetHybList</p>
</td></tr>
<tr><td><code id="swapHybs_+3A_rna">rna</code></td>
<td>
<p>The rna of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of &quot;swapped&quot; hyb datas
</p>

<hr>
<h2 id='swapHybs2'>swapHybs2</h2><span id='topic+swapHybs2'></span>

<h3>Description</h3>

<p>Swap the table to ensure that 3 prime most duplex side is on the left of the table
used to make one sides heatmaps and other reasons where having the left of the table
coming after the right side is a problem. Different from swapHybs as it
ensure that BOTH duplex sides originate from the RNA of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swapHybs2(hybList, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swapHybs2_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles or subsetHybList</p>
</td></tr>
<tr><td><code id="swapHybs2_+3A_rna">rna</code></td>
<td>
<p>The rna of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of &quot;swapped&quot; hyb data
</p>

<hr>
<h2 id='swapHybs3'>swapHybs3</h2><span id='topic+swapHybs3'></span>

<h3>Description</h3>

<p>Swap the table to ensure that 3 prime most duplex side is ont he left of the table
used to make one sides heatmaps and other reasons where having the left of the table
coming after the right side is a problem. Different from swapHybs as it
ensure that BOTH duplex sides originate from the RNA of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swapHybs3(hybList, rna)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swapHybs3_+3A_hyblist">hybList</code></td>
<td>
<p>the original hybList created with readHybFiles or subsetHybList</p>
</td></tr>
<tr><td><code id="swapHybs3_+3A_rna">rna</code></td>
<td>
<p>The rna of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of &quot;swapped&quot; hyb datas
</p>

<hr>
<h2 id='topInteracters'>topInteracters</h2><span id='topic+topInteracters'></span><span id='topic+topInteracters+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints the top transcripts that have the most duplexes
assigned that interact with the transcript of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topInteracters(cds, ntop = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topInteracters_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="topInteracters_+3A_ntop">ntop</code></td>
<td>
<p>the number of entries to display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table, the number of counts per sample per interacting transcript
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
topInteracters(cds)

</code></pre>

<hr>
<h2 id='topInteractions'>topInteractions</h2><span id='topic+topInteractions'></span><span id='topic+topInteractions+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints the top transcript interactions that have the most duplexes
assigned
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topInteractions(cds, ntop = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topInteractions_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="topInteractions_+3A_ntop">ntop</code></td>
<td>
<p>the number of entries to display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table, the number of counts per sample per interaction
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
topInteractions(cds)

</code></pre>

<hr>
<h2 id='topTranscripts'>topTranscripts</h2><span id='topic+topTranscripts'></span><span id='topic+topTranscripts+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>This method prints the top transcripts that have the most duplexes
assigned
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topTranscripts(cds, ntop = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topTranscripts_+3A_cds">cds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="topTranscripts_+3A_ntop">ntop</code></td>
<td>
<p>the number of entries to display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table, the number of counts per sample per transcript
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()
topTranscripts(cds)
</code></pre>

<hr>
<h2 id='trimClusters'>trimClusters</h2><span id='topic+trimClusters'></span><span id='topic+trimClusters+2CcomradesDataSet-method'></span>

<h3>Description</h3>

<p>Trimming of the clusters removes redundant information derived from random
fragmentation of the reads during library preparation. This method takes
a <code>comradesDataSet</code> object where clustering has been performed with 
the clusterCOMRADES method and trims the clusters according to the 
trimFactor argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trimClusters(clusteredCds, trimFactor = 2.5, clusterCutoff = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimClusters_+3A_clusteredcds">clusteredCds</code></td>
<td>
<p>a <code>comradesDataSet</code> object</p>
</td></tr>
<tr><td><code id="trimClusters_+3A_trimfactor">trimFactor</code></td>
<td>
<p>a positive value that defines how much the clusters will</p>
</td></tr>
<tr><td><code id="trimClusters_+3A_clustercutoff">clusterCutoff</code></td>
<td>
<p>Minimum number of reads before discarding cluster
be trimmed = mean + ( sd * trimFactor )</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 3 attributes; matrixList, clusterTableList and clusterGrangesList 
will gain the <code>types</code> &quot;superClusters&quot; and &quot;trimmedClusters&quot;
</p>


<h3>Value</h3>

<p>Returns a <code>comradesDataSet</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cds = makeExampleComradesDataSet()

clusteredCds = clusterComrades(cds,
                cores = 1,
                stepCount = 1,
                clusterCutoff = 0)
                
trimClusters(clusteredCds = clusteredCds,
             trimFactor = 1, 
             clusterCutoff = 0)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
