<!DOCTYPE html><html><head><title>Help for package concatipede</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {concatipede}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#auto_match_seqs'><p>Build a template table with automatically matched sequence names</p></a></li>
<li><a href='#concatipede'><p>Concatenate alignments</p></a></li>
<li><a href='#concatipede_example'><p>Get the path to one of concatipede example files</p></a></li>
<li><a href='#concatipede_prepare'><p>Load alignments and prepare template correspondence table for concatenate () function</p></a></li>
<li><a href='#find_fasta'><p>Find fasta files present in a folder</p></a></li>
<li><a href='#get_genbank_table'><p>Extract GenBank accession number from correspondence table</p></a></li>
<li><a href='#read_xl'><p>Read an Excel file</p></a></li>
<li><a href='#rename_sequences'><p>Rename sequences</p></a></li>
<li><a href='#write_aln'><p>Writing alignments</p></a></li>
<li><a href='#write_xl'><p>Write an Excel file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Easy Concatenation of Fasta Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Concatenation of multiple sequence alignments based on a
    correspondence table that can be edited in Excel &lt;<a href="https://doi.org/10.5281%2Fzenodo.5130603">doi:10.5281/zenodo.5130603</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tardipede/concatipede">https://github.com/tardipede/concatipede</a>,
<a href="https://tardipede.github.io/concatipede/">https://tardipede.github.io/concatipede/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tardipede/concatipede/issues">https://github.com/tardipede/concatipede/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, dplyr, igraph, qualV, magrittr, tibble, readxl,
stringdist, stringr, writexl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DT, knitr, rmarkdown, tidyverse, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-06 16:01:32 UTC; matteovecchi</td>
</tr>
<tr>
<td>Author:</td>
<td>Matteo Vecchi <a href="https://orcid.org/0000-0002-7995-6827"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Mattieu Bruneaux <a href="https://orcid.org/0000-0001-6997-192X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matteo Vecchi &lt;matteo.vecchi15@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-06 18:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='auto_match_seqs'>Build a template table with automatically matched sequence names</h2><span id='topic+auto_match_seqs'></span>

<h3>Description</h3>

<p>The algorithm used to match sequences across fasta files based on their
names is outlined below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auto_match_seqs(x, method = "lv", xlsx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auto_match_seqs_+3A_x">x</code></td>
<td>
<p>A table (data frame or tibble) typically produced by
<code><a href="#topic+concatipede_prepare">concatipede_prepare</a></code>. It must be of the same format as a
table returned by this function: a first column called &quot;name&quot; followed
by one column per fasta file. Those columns have the name of their
corresponding fasta file, and they contain the names of the sequences in
this file, with one sequence name per cell. The number of rows in the
number of sequences of the fasta file with the most sequences, and the
columns for the other fasta files are filled with <code>NA</code> for padding.</p>
</td></tr>
<tr><td><code id="auto_match_seqs_+3A_method">method</code></td>
<td>
<p>Method for string distance calculation. See
<code>?stringdist::stringdist-metrics</code> for details. Default is
<code>"lv"</code>.</p>
</td></tr>
<tr><td><code id="auto_match_seqs_+3A_xlsx">xlsx</code></td>
<td>
<p>Optional, a path to use to save the output table as an Excel
file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let's assume a situation with N fasta files, with each fasta file i having
n_i sequence names. The problem of matching the names in the best possible
way across the fasta files is similar to that of identifying homologous
proteins across species, using e.g. reciprocal blast.
</p>
<p>The algorithm steps are:
</p>

<ul>
<li><p> For each pair of fasta files, identify matching names using a
reciprocal match approach: two names match if and only if they are their
reciprocal best match.
</p>
</li>
<li><p> Those matches across fasta files define a graph.
</p>
</li>
<li><p> We identify sub-graphs such that (i) they contain at most one
sequence name per fasta file and (ii) all nodes in a given sub-graph are
fully connected (i.e., they are all their best reciprocal matches across
any pair of fasta files).
</p>
</li></ul>



<h3>Value</h3>

<p>A table (tibble) with the same columns as <code>x</code> and with sequence
names automatically matched across fasta files. Sequence names which did
not have a best reciprocal match in other fasta files are appended to
the end of the table, so that the output table columns contain all the
unique sequence names present in the corresponding column of the input
table. The first column, &quot;name&quot;, contains a suggested name for the row
(not guaranteed to be unique). If a path was provided to the <code>xlsx</code>
argument, an Excel file is saved and the table is returned invisibly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xlsx_file &lt;- concatipede_example("sequences-test-matching.xlsx")
xlsx_template &lt;- readxl::read_xlsx(xlsx_file)
auto_match_seqs(xlsx_template)
## Not run: 
  auto_match_seqs(xlsx_template, xlsx = "my-automatic-output.xlsx")

## End(Not run)

</code></pre>

<hr>
<h2 id='concatipede'>Concatenate alignments</h2><span id='topic+concatipede'></span>

<h3>Description</h3>

<p>This function concatenate sequences from alignments present in the working directory based on a correspondence table and saves the output in a new directory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concatipede(
  df = NULL,
  filename = NULL,
  format = c("fasta", "nexus", "phylip"),
  dir,
  plotimg = FALSE,
  out = NULL,
  remove.gaps = TRUE,
  write.outputs = TRUE,
  save.partitions = TRUE,
  excel.sheet = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatipede_+3A_df">df</code></td>
<td>
<p>The user-defined correspondence table, as a data frame or equivalent. This is used only if no <code>filename</code> argument is provided.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_filename">filename</code></td>
<td>
<p>Filename of input correspondence table. Alternatively, if no filename is provided, the user can provide their own correspondence table as the <code>df</code> argument.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_format">format</code></td>
<td>
<p>a string specifying in what formats you want the alignment</p>
</td></tr>
<tr><td><code id="concatipede_+3A_dir">dir</code></td>
<td>
<p>Optional, path to the directory containing the fasta files. This argument has an effect only if fasta files names are taken from the columns of the <code>df</code> argument, and that <code>df</code> does not have an attribute <code>dir_name</code> itself. If no <code>dir</code> is provided and <code>df</code> does not have a <code>dir_name</code> attribute, the current working directory is ued with a warning.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_plotimg">plotimg</code></td>
<td>
<p>Logical, save a graphical representation of the alignment in pdf format. Default: FALSE.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_out">out</code></td>
<td>
<p>specify outputs filenames</p>
</td></tr>
<tr><td><code id="concatipede_+3A_remove.gaps">remove.gaps</code></td>
<td>
<p>Logical, remove gap only columns. Useful if not using all sequences in the alignments. Default: TRUE.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_write.outputs">write.outputs</code></td>
<td>
<p>Logical, save concatenated alignment, partitions position table and graphical representation. If FALSE it overrides plotimg. Default: TRUE.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_save.partitions">save.partitions</code></td>
<td>
<p>Logical, save in the concatenated alignmeent directory a text file with partitions limits for the concatenated alignment. Default: TRUE.</p>
</td></tr>
<tr><td><code id="concatipede_+3A_excel.sheet">excel.sheet</code></td>
<td>
<p>specify what sheet from the excel spreadsheet has to be read. Either a string (the name of a sheet), or an integer (the position of the sheet).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The concatenated alignment (invisibly if <code>out</code> is not NULL).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dir &lt;- system.file("extdata", package = "concatipede")
z &lt;- concatipede(filename = paste0(dir,"/Macrobiotidae_seqnames.xlsx"), dir = dir,
                 write.outputs = FALSE)
z

</code></pre>

<hr>
<h2 id='concatipede_example'>Get the path to one of concatipede example files</h2><span id='topic+concatipede_example'></span>

<h3>Description</h3>

<p>Several example files are shipped with the concatipede package. This function facilitates the access to those files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concatipede_example(example_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatipede_example_+3A_example_file">example_file</code></td>
<td>
<p>Basename of the target example file. If <code>NULL</code> (the default), the basenames of the available files are listed.</p>
</td></tr>
</table>


<h3>Details</h3>


<dl>
<dt>COI_Macrobiotidae.fas</dt><dd><p>Example fasta file.</p>
</dd>
<dt>ITS2_Macrobiotidae.fas</dt><dd><p>Example fasta file.</p>
</dd>
<dt>LSU_Macrobiotidae.fas</dt><dd><p>Example fasta file.</p>
</dd>
<dt>SSU_Macrobiotidae.fas</dt><dd><p>Example fasta file.</p>
</dd>
<dt>sequences-test-matching.xlsx</dt><dd><p>This is an Excel file (extension .xlsx) typically used to test or demonstrate the automatic matching capabilities of the concatipede package. This file represents the Excel template that could be produced by <code><a href="#topic+concatipede_prepare">concatipede_prepare</a></code> after detecting the fasta files present in a working directory.</p>
</dd>
<dt>Macrobiotidae_seqnames.xlsx</dt><dd><p>This is an Excel file (extension .xlsx) that contains the correspondence table that can be used to concatenate the sequences contained in the example fasta files COI_Macrobiotidae.fas, ITS2_Macrobiotidae.fas, LSU_Macrobiotidae.fas, and SSU_Macrobiotidae.fas.</p>
</dd>
</dl>



<h3>Value</h3>

<p>The full path to access the example file, or a list of available example files if no <code>example_file</code> argument was provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>concatipede_example()
example &lt;- concatipede_example("sequences-test-matching.xlsx")
if (requireNamespace("readxl")) {
  seqs &lt;- readxl::read_xlsx(example)
  seqs
}

</code></pre>

<hr>
<h2 id='concatipede_prepare'>Load alignments and prepare template correspondence table for concatenate () function</h2><span id='topic+concatipede_prepare'></span>

<h3>Description</h3>

<p>This function creates a template correspondence table that can also be saved in the working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concatipede_prepare(fasta_files, out = "seqnames", excel = TRUE, exclude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatipede_prepare_+3A_fasta_files">fasta_files</code></td>
<td>
<p>Optional, a vector of paths to the fasta files that should be merged. If this argument is missing, the function automatically detects and uses all the fasta files present in the working directory.</p>
</td></tr>
<tr><td><code id="concatipede_prepare_+3A_out">out</code></td>
<td>
<p>Optional, a filename for the correspondence table template to save (without extension). No file is saved if <code>out</code> is not provided. In all cases, the function also returns a tibble with the correspondence table template (invisibly if <code>out</code> is provided).</p>
</td></tr>
<tr><td><code id="concatipede_prepare_+3A_excel">excel</code></td>
<td>
<p>Boolean, should the correspondence table template be saved in excel format? If <code>FALSE</code>, it is saved as a tab-separated text file instead. Default is <code>TRUE</code>. The correct file extension is automatically appended to the <code>out</code> argument. If <code>out</code> is missing, this argument has no effect.</p>
</td></tr>
<tr><td><code id="concatipede_prepare_+3A_exclude">exclude</code></td>
<td>
<p>If no <code>fasta_files</code> argument has been passed, fasta files matching the <code>exclude</code> pattern will be ignored by the function when it automatically detects fasta files in the working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with the correspondence table template (invisibly if an <code>out</code> argument was provided to save the table to a file).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dir &lt;- system.file("extdata", package = "concatipede")
fasta_files &lt;- find_fasta(dir)
z &lt;- concatipede_prepare(fasta_files)
z

</code></pre>

<hr>
<h2 id='find_fasta'>Find fasta files present in a folder</h2><span id='topic+find_fasta'></span>

<h3>Description</h3>

<p>Find fasta files present in a folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_fasta(dir, pattern = "\\.fa$|\\.fas$|\\.fasta$", exclude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_fasta_+3A_dir">dir</code></td>
<td>
<p>Path to the directory which should be examined. If not provided,
the current working directory (as returned by <code><a href="base.html#topic+getwd">getwd</a></code>) is
used.</p>
</td></tr>
<tr><td><code id="find_fasta_+3A_pattern">pattern</code></td>
<td>
<p>Regular expression used by <code><a href="base.html#topic+list.files">list.files</a></code> to detect
the fasta files. The default is to list all files ending in &quot;.fa&quot;,
&quot;.fas&quot;, and &quot;.fasta&quot;.</p>
</td></tr>
<tr><td><code id="find_fasta_+3A_exclude">exclude</code></td>
<td>
<p>Optional regular expression used to exclude some filenames
from the list of detected files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the full paths to the found files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get the directory containing the package example files
dir &lt;- system.file("extdata", package = "concatipede")
# List the fasta files containing in that directory
find_fasta(dir)
# Exclude some files
find_fasta(dir, exclude = "COI")

</code></pre>

<hr>
<h2 id='get_genbank_table'>Extract GenBank accession number from correspondence table</h2><span id='topic+get_genbank_table'></span>

<h3>Description</h3>

<p>Extract GenBank accession number from correspondence table formatted with the same requirements for concatipede()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genbank_table(
  df = NULL,
  filename = NULL,
  writetable = FALSE,
  out = "",
  excel.sheet = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_genbank_table_+3A_df">df</code></td>
<td>
<p>The user-defined correspondence table, as a data frame or equivalent. This is used only if no <code>filename</code> argument is provided.</p>
</td></tr>
<tr><td><code id="get_genbank_table_+3A_filename">filename</code></td>
<td>
<p>Filename of input correspondence table. Alternatively, if no filename is provided, the user can provide their own correspondence table as the <code>df</code> argument.</p>
</td></tr>
<tr><td><code id="get_genbank_table_+3A_writetable">writetable</code></td>
<td>
<p>if TRUE save the Genbank table as excel file in the working directory</p>
</td></tr>
<tr><td><code id="get_genbank_table_+3A_out">out</code></td>
<td>
<p>if writetable == T, the name to be attached to the excel filename</p>
</td></tr>
<tr><td><code id="get_genbank_table_+3A_excel.sheet">excel.sheet</code></td>
<td>
<p>specify what sheet from the excel spreadsheet you wanna read. Either a string (the name of a sheet), or an integer (the position of the sheet).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table with GenBank accession numbers
</p>

<hr>
<h2 id='read_xl'>Read an Excel file</h2><span id='topic+read_xl'></span>

<h3>Description</h3>

<p>This function loads a table from an Excel file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_xl(path, sheet = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_xl_+3A_path">path</code></td>
<td>
<p>The path to the Excel file to read.</p>
</td></tr>
<tr><td><code id="read_xl_+3A_sheet">sheet</code></td>
<td>
<p>Optional, the sheet to read (either a string with the name of the sheet or an integer with its position). Default: 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='rename_sequences'>Rename sequences</h2><span id='topic+rename_sequences'></span>

<h3>Description</h3>

<p>This function renames sequences in fasta files based on a correspondence table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_sequences(
  fasta_files,
  df = NULL,
  filename = NULL,
  marker_names = NULL,
  out = NULL,
  format = "fasta",
  excel.sheet = 1,
  unalign = FALSE,
  exclude
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_sequences_+3A_fasta_files">fasta_files</code></td>
<td>
<p>Optional, a vector of paths to the fasta files that should be renamed. If this argument is missing, the function automatically detects and uses all the fasta files present in the working directory.</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_df">df</code></td>
<td>
<p>The user-defined correspondence table, as a data frame or equivalent. This is used only if no <code>filename</code> argument is provided.</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_filename">filename</code></td>
<td>
<p>Filename of correspondence table. Alternatively, if no filename is provided, the user can provide their own correspondence table as the <code>df</code> argument.</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_marker_names">marker_names</code></td>
<td>
<p>the name of the marker for each alignment to be appended at the end of the sequences names, in the same order as in the correspondence table</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_out">out</code></td>
<td>
<p>specify outputs filename</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_format">format</code></td>
<td>
<p>a string specifying in what formats you want the alignment. Can be &quot;fasta&quot;, &quot;phylip&quot; and &quot;nexus&quot;</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_excel.sheet">excel.sheet</code></td>
<td>
<p>specify what sheet from the excel spreadsheet you wanna read. Either a string (the name of a sheet), or an integer (the position of the sheet).</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_unalign">unalign</code></td>
<td>
<p>return unaligned fasta files as output</p>
</td></tr>
<tr><td><code id="rename_sequences_+3A_exclude">exclude</code></td>
<td>
<p>Optional regular expression used to exclude some filenames from the list of detected files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effect of saving a correspondence table.
</p>

<hr>
<h2 id='write_aln'>Writing alignments</h2><span id='topic+write_aln'></span><span id='topic+write_fasta'></span><span id='topic+write_nexus'></span><span id='topic+write_phylip'></span>

<h3>Description</h3>

<p>Alignments can be saved in <code>fasta</code>, <code>nexus</code>, and <code>phylip</code> formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_fasta(x, path)

write_nexus(x, path)

write_phylip(x, path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_aln_+3A_x">x</code></td>
<td>
<p>Alignment to save (an object of class <code>DNAbin</code>).</p>
</td></tr>
<tr><td><code id="write_aln_+3A_path">path</code></td>
<td>
<p>Path of the file to be written, without file extension (the
appropriate extension is added automatically, i.e. the path will be
extended with &quot;.fasta&quot;, &quot;.nexus&quot;, or &quot;.phy&quot; depending on the file format
used).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input <code>x</code> (invisibly).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  # Path to an example alignment file
  pkg_aln &lt;- concatipede_example("COI_Macrobiotidae.fas")
  # Load the alignment into the R session
  aln &lt;- ape::read.FASTA(pkg_aln)
  # Write the alignment in various formats
  # Note that the appropriate file extension is added by the writing functions.
  write_fasta(aln, "my-alignment")
  write_nexus(aln, "my-alignment")
  write_phylip(aln, "my-alignment")

## End(Not run)
</code></pre>

<hr>
<h2 id='write_xl'>Write an Excel file</h2><span id='topic+write_xl'></span>

<h3>Description</h3>

<p>This function writes an input table to an Excel file and returns its input (invisibly).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_xl(x, path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_xl_+3A_x">x</code></td>
<td>
<p>A data frame or tibble to write to an Excel file.</p>
</td></tr>
<tr><td><code id="write_xl_+3A_path">path</code></td>
<td>
<p>The path to the Excel file to be written.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input table <code>x</code>, invisibly (so that the function can be part of a pipeline with the pipe operator).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
