<!DOCTYPE html><html><head><title>Help for package CONCUR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CONCUR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cnvData'><p>Pseudo Copy Number Variants Data</p></a></li>
<li><a href='#concur'><p>Copy Number Profile Curve-Based Association Test</p></a></li>
<li><a href='#covData'><p>Pseudo Covariate Data</p></a></li>
<li><a href='#phenoData'><p>Pseudo Phenotype Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Copy Number Profile Curve-Based Association Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Amanda Brucker, Shannon T. Holloway, Jung-Ying Tzeng</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shannon T. Holloway &lt;shannon.t.holloway@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a kernel-based association test for copy number 
  variation (CNV) aggregate analysis
  in a certain genomic region (e.g., gene set, chromosome, or genome) that is
  robust to the within-locus and across-locus etiological heterogeneity, and 
  bypass the need to define a "locus" unit for CNVs. 
  Brucker, A., et al. (2020) &lt;<a href="https://doi.org/10.1101%2F666875">doi:10.1101/666875</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>CompQuadForm, dplyr, mgcv</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Collate:</td>
<td>'commonAUC.R' 'segLength.R' 'popFunc.R' 'uniqueCombinations.R'
'createDF.R' 'deldup.R' 'CAUCkernel.R' 'cnvData.R' 'compFunc.R'
'dataChecks.R' 'test.qt.R' 'test.bt.R' 'pv_func.R' 'concur.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-05 18:27:51 UTC; 19194</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-07 12:00:28 UTC</td>
</tr>
</table>
<hr>
<h2 id='cnvData'>Pseudo Copy Number Variants Data</h2><span id='topic+cnvData'></span>

<h3>Description</h3>

<p>This data set includes simulated CNV data in PLINK CNV data format.
The data are also available from the authors through the url 
provided below. These data were generated following the simulation 
study used to illustrate the method in the original manuscript also 
referenced below; it has been reduced to include only 600 individuals. 
These data are not meaningful and are intended for demonstration purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cnvData)
</code></pre>


<h3>Format</h3>

<p>cnvData is a data.frame containing 522 observations with 5 columns: 
</p>

<dl>
<dt>ID</dt><dd><p>character patient identifier.</p>
</dd>
<dt>CHR</dt><dd><p>CNV chromosome.</p>
</dd>
<dt>BP1</dt><dd><p>starting location in base pairs.</p>
</dd> 
<dt>BP2</dt><dd><p>ending location in base pairs.</p>
</dd> 
<dt>TYPE</dt><dd><p>copy number (0,1,3,or 4).</p>
</dd>
</dl>



<h3>References</h3>

<p>Brucker, A., Lu, W., Marceau West, R., Yu, Q-Y., Hsiao, C. K.,
Hsiao, T-H., Lin, C-H., Magnusson, P. K. E., Holloway, S. T., 
Sullivan, P. F., Szatkiewicz, J. P., Lu, T-P., and
Tzeng, J-Y. Association testing using Copy Number Profile Curves (CONCUR)
enhances power in copy number variant analysis. &lt;doi:10.1101/666875&gt;.
</p>
<p><a href="https://www4.stat.ncsu.edu/~jytzeng/Software/CONCUR/">https://www4.stat.ncsu.edu/~jytzeng/Software/CONCUR/</a>
</p>

<hr>
<h2 id='concur'>Copy Number Profile Curve-Based Association Test</h2><span id='topic+concur'></span>

<h3>Description</h3>

<p>Implements a kernel-based association test for CNV aggregate analysis
in a certain genomic region (e.g., gene set, chromosome, or genome) that is
robust to the within-locus and across-locus etiologoical heterogeneity, and 
bypass the need to define a &quot;locus&quot; unit for CNVs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concur(
  cnv,
  X,
  pheno,
  phenoY,
  phenoType,
  ...,
  nCore = 1L,
  outFileKernel = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concur_+3A_cnv">cnv</code></td>
<td>
<p>A character or data.frame object. If character, the 
name of the data file containing the CNV data (with a header). If 
data.frame, the CNV data. The data must contain the following columns: 
&quot;ID&quot;, &quot;CHR&quot;, &quot;BP1&quot;, &quot;BP2&quot;, &quot;TYPE&quot;, where &quot;ID&quot; is a unique patient id,
&quot;CHR&quot; is the CNV chromosome, &quot;BP1&quot; is the start location in base pairs
or kilo-base pairs,
&quot;BP2&quot; is the end location in base pairs or kilo-base pairs, and 
&quot;TYPE&quot; is the CNV copy number.</p>
</td></tr>
<tr><td><code id="concur_+3A_x">X</code></td>
<td>
<p>A character or data.frame object. If character, the 
name of the data file containing the covariate data (with a header). If 
data.frame, the covariate data. The data must contain a column titled 
&quot;ID&quot; containing a unique patient id. This column must contain the
patient identifiers of the CNV data specified in 
input cnv; however, it can contain patient identifiers not contained in cnv.
Further, inputs X and pheno must contain the same patient identifiers. Categorical
variables must be translated into design matrix format.</p>
</td></tr>
<tr><td><code id="concur_+3A_pheno">pheno</code></td>
<td>
<p>A character or data.frame object. If character, the 
name of the data file containing the phenotype data (with a header). If 
data.frame, the phenotype data. The data must contain a column titled 
&quot;ID&quot; containing a unique patient id. This column must contain the
patient identifiers of the CNV data specified in 
input cnv; however, it can contain patient identifiers not contained in cnv.
Further, inputs X and pheno must contain the same patient identifiers.</p>
</td></tr>
<tr><td><code id="concur_+3A_phenoy">phenoY</code></td>
<td>
<p>A character object. The column name in input pheno containing
the phenotype of interest.</p>
</td></tr>
<tr><td><code id="concur_+3A_phenotype">phenoType</code></td>
<td>
<p>A character object. Must be one of of {&quot;bin&quot;, &quot;cont&quot;} indicating
if input phenoY (i.e., the phenotype of interest) is binary or continuous.</p>
</td></tr>
<tr><td><code id="concur_+3A_...">...</code></td>
<td>
<p>Ignored. Included to require named inputs.</p>
</td></tr>
<tr><td><code id="concur_+3A_ncore">nCore</code></td>
<td>
<p>An integer object. If nCore &gt; 1, package parallel is used to 
calculate the kernel. Though the methods of package CompQuadForm dominate
the time profile, setting nCore &gt; 1L can improve computation times.</p>
</td></tr>
<tr><td><code id="concur_+3A_outfilekernel">outFileKernel</code></td>
<td>
<p>A character object or NULL. If a character, the
file in which the kernel is to be saved. If NULL, the kernel is returned
by the function.</p>
</td></tr>
<tr><td><code id="concur_+3A_verbose">verbose</code></td>
<td>
<p>A logical object. If TRUE, progress information is printed
to the screen.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The CNV data must adhere to the following conditions:
</p>

<ul>
<li><p> CNVs must be at least 1 unit long.
</p>
</li>
<li><p> CNVs cannot end at the exact location another begins
</p>
</li></ul>

<p>Violations of these conditions typically occur when data are rounded to
a desired resolution.
For example
</p>
<pre>
 ID CHR      BP1      BP2 TYPE
  1  13 10112087 10112414    3
</pre>
<p>becomes upon rounding to kilo
</p>
<pre>
 ID CHR   BP1   BP2 TYPE
  1  13 10112 10112    3     .
</pre>
<p>These cases should either be discarded or modified to be of length 1, e.g.,
</p>
<pre>
 ID CHR   BP1   BP2 TYPE
  1  13 10112 10113    3     .
</pre>
<p>As an example of condition 2 
</p>
<pre>
 ID CHR    BP1   BP2 TYPE
  1  13 100768 101100    3
  1  13 101100 101299    1
</pre>
<p>should be modified to one of
</p>
<pre>
 ID CHR    BP1   BP2 TYPE
  1  13 100768 101100    3
  1  13 101101 101299    1
</pre>
<p>or
</p>
<pre>
 ID CHR    BP1   BP2 TYPE
  1  13 100768 101099    3
  1  13 101100 101299    1     .
</pre>
<p>Additionally,
</p>
<pre>
 ID CHR    BP1   BP2 TYPE
  1  13 100768 101100    3
  1  13 101100 101299    3
</pre>
<p>should be combined as
</p>
<pre>
 ID CHR    BP1   BP2 TYPE
  1  13 100768 101299    3     .
</pre>


<h3>Value</h3>

<p>A list containing the kernel (or its file name) and the p-value.
</p>


<h3>References</h3>

<p>Brucker, A., Lu, W., Marceau West, R., Yu, Q-Y., Hsiao, C. K.,
Hsiao, T-H., Lin, C-H., Magnusson, P. K. E., Holloway, S. T., 
Sullivan, P. F., Szatkiewicz, J. P., Lu, T-P., and
Tzeng, J-Y. Association testing using Copy Number Profile Curves (CONCUR)
enhances power in copy number variant analysis. &lt;doi:10.1101/666875&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cnvData)

# limit data for examples
exCNV &lt;- cnvData$ID %in% paste0("P", 1:150)
exCOV &lt;- covData$ID %in% paste0("P", 1:150)
exPHE &lt;- phenoData$ID %in% paste0("P", 1:150)

# binary phenoType
results &lt;- concur(cnv = cnvData[exCNV,],
                  X = covData[exCOV,],
                  pheno = phenoData[exPHE,],
                  phenoY = 'PHEB',
                  phenoType = 'bin',
                  nCore = 1L,
                  outFileKernel = NULL,
                  verbose = TRUE)

# continuous phenoType
results &lt;- concur(cnv = cnvData[exCNV,],
                  X = covData[exCOV,],
                  pheno = phenoData[exPHE,],
                  phenoY = 'PHEC',
                  phenoType = 'cont',
                  nCore = 1L,
                  outFileKernel = NULL,
                  verbose = TRUE)

</code></pre>

<hr>
<h2 id='covData'>Pseudo Covariate Data</h2><span id='topic+covData'></span>

<h3>Description</h3>

<p>This data set includes simulated covariate data.
These data were generated as draws from a Binom(1,0.5) distribution for the
800 individuals in the example data provided with the package.
These data are not meaningful and are intended for demonstration purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cnvData)
</code></pre>


<h3>Format</h3>

<p>covData is a data.frame containing 400 observations with 2 columns
</p>

<dl>
<dt>ID</dt><dd><p>character patient identifier.</p>
</dd>
<dt>SEX</dt><dd><p>binary indicator of M/F.</p>
</dd>
</dl>



<h3>References</h3>

<p>Brucker, A., Lu, W., Marceau West, R., Yu, Q-Y., Hsiao, C. K.,
Hsiao, T-H., Lin, C-H., Magnusson, P. K. E., Holloway, S. T., 
Sullivan, P. F., Szatkiewicz, J. P., Lu, T-P., and
Tzeng, J-Y. Association testing using Copy Number Profile Curves (CONCUR)
enhances power in copy number variant analysis. &lt;doi:10.1101/666875&gt;.
</p>

<hr>
<h2 id='phenoData'>Pseudo Phenotype Data</h2><span id='topic+phenoData'></span>

<h3>Description</h3>

<p>This data set includes simulated phenotype data.
These data include a binary phenotype and a normally distributed continuous
phenotype that are randomly generated independent of the CNV data.
These data are not meaningful and are intended for demonstration purposes only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cnvData)
</code></pre>


<h3>Format</h3>

<p>phenoData is a data.frame containing 400 observations with 3 columns
</p>

<dl>
<dt>ID</dt><dd><p>character patient identifier.</p>
</dd>
<dt>PHEB</dt><dd><p>binary phenotype.</p>
</dd>
<dt>PHEC</dt><dd><p>continuous phenotype.</p>
</dd>
</dl>



<h3>References</h3>

<p>Brucker, A., Lu, W., Marceau West, R., Yu, Q-Y., Hsiao, C. K.,
Hsiao, T-H., Lin, C-H., Magnusson, P. K. E., Holloway, S. T., 
Sullivan, P. F., Szatkiewicz, J. P., Lu, T-P., and
Tzeng, J-Y. Association testing using Copy Number Profile Curves (CONCUR)
enhances power in copy number variant analysis. &lt;doi:10.1101/666875&gt;.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
