<!DOCTYPE html><html><head><title>Help for package CoNI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CoNI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assign_colorsAnnotation'><p>Assing Colors to Class</p></a></li>
<li><a href='#barplot_VertexsPerEdgeFeature'><p>Number Vertex features per class for every shared edge feature</p></a></li>
<li><a href='#check_outputDir'><p>Output directory</p></a></li>
<li><a href='#check_previous'><p>Check previous files</p></a></li>
<li><a href='#checkInputParameters'><p>Check input parameters</p></a></li>
<li><a href='#Chow_GeneExpData'><p>Chow gene expression data</p></a></li>
<li><a href='#Chow_MetaboliteData'><p>Chow metabolite data</p></a></li>
<li><a href='#chunk2'><p>Split function</p></a></li>
<li><a href='#compare_sampleNames'><p>Compare sample names</p></a></li>
<li><a href='#Compare_Triplets'><p>Compare triplets</p></a></li>
<li><a href='#Compare_VertexClasses_sharedEdgeFeatures'><p>Table VertexClass pairs of shared Edge Features</p></a></li>
<li><a href='#CoNI'><p>Correlation guided Network Integration</p></a></li>
<li><a href='#CoNIResults_Chow'><p>CoNI Results Chow</p></a></li>
<li><a href='#CoNIResults_HFD'><p>CoNI Results HFD</p></a></li>
<li><a href='#CoNIResultsHFDToy'><p>Toy data HFD results</p></a></li>
<li><a href='#countClassPerEdgeFeature'><p>Number lipid features per class</p></a></li>
<li><a href='#create_edgeFBarplot'><p>Vertex-class pairs profile of one shared edge feature</p></a></li>
<li><a href='#create_GlobalBarplot'><p>Vertex-class pairs profile of shared features</p></a></li>
<li><a href='#create_stackedGlobalBarplot_perTreatment'><p>Stacked Global Barplot (One treatment)</p></a></li>
<li><a href='#createBipartiteGraph'><p>Bipartite Network</p></a></li>
<li><a href='#createBipartiteTable'><p>Bipartite Table</p></a></li>
<li><a href='#delIntFiles'><p>Delete intermediary files</p></a></li>
<li><a href='#do_objectsExist'><p>Check if files exist</p></a></li>
<li><a href='#find_localControllingFeatures'><p>Find local controlling features</p></a></li>
<li><a href='#flattenCorrMatrix'><p>Flatten</p></a></li>
<li><a href='#GeneExpToy'><p>Toy data gene expression</p></a></li>
<li><a href='#generate_network'><p>Create network</p></a></li>
<li><a href='#get_lowvarFeatures'><p>Low variance features</p></a></li>
<li><a href='#getcolor'><p>Get class rgb color</p></a></li>
<li><a href='#getstackedGlobalBarplot_and_Grid'><p>Stacked Global Barplot Side-by-side (two treatments)</p></a></li>
<li><a href='#getvertexes_edgeFeature'><p>Get vertexes for edge feature</p></a></li>
<li><a href='#getVertexsPerEdgeFeature'><p>Vertex Class profile per edge feature (one treatment)</p></a></li>
<li><a href='#getVertexsPerEdgeFeature_and_Grid'><p>Vertex-Class profile per edge feature Side-by-Side (two treatments)</p></a></li>
<li><a href='#HFD_GeneExpData'><p>HFD gene expression data</p></a></li>
<li><a href='#HFD_MetaboliteData'><p>HFD metabolite data</p></a></li>
<li><a href='#labels2colors_2'><p>Labels to colors</p></a></li>
<li><a href='#merge_outpuSplitFiles'><p>Merge Files.</p></a></li>
<li><a href='#MetaboExpToy'><p>Toy data metabolite expression</p></a></li>
<li><a href='#MetaboliteAnnotation'><p>Metabolite Annotation</p></a></li>
<li><a href='#MetColorTable'><p>Toy data annotation</p></a></li>
<li><a href='#NetStats'><p>Network Statistics</p></a></li>
<li><a href='#obtain_groupcolors'><p>Get colors</p></a></li>
<li><a href='#plotPcorvsCor'><p>Correlation vs Partial correlation</p></a></li>
<li><a href='#sig_correlation2'><p>Pairwise correlations</p></a></li>
<li><a href='#sig_correlation2Dfs'><p>Significant correlations 2 Df</p></a></li>
<li><a href='#split_df'><p>Split dataset</p></a></li>
<li><a href='#tableLCFs_VFs'><p>Table local controlling edge features and vertex pairs</p></a></li>
<li><a href='#top_n_LF_byMagnitude'><p>Linker Features by magnitude of effect</p></a></li>
<li><a href='#VertexClassesSharedGenes_HFDvsChow'><p>Toy data comparison treatments</p></a></li>
<li><a href='#writeTable'><p>Write table</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Correlation Guided Network Integration (CoNI)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-09-28</td>
</tr>
<tr>
<td>Description:</td>
<td>Integrates two numerical omics data sets from the same samples using partial correlations. The output can be represented as a network, bipartite graph or a hypergraph structure. The method used in the package refers to Klaus et al (2021) &lt;<a href="https://doi.org/10.1016%2Fj.molmet.2021.101295">doi:10.1016/j.molmet.2021.101295</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph (&ge; 1.2.6), doParallel (&ge; 1.0.16), cocor (&ge; 1.1.3),
ggplot2 (&ge; 3.3.3), forcats (&ge; 0.5.1), dplyr (&ge; 1.0.5),
data.table (&ge; 1.13.7), tibble (&ge; 3.1.0), foreach (&ge; 1.5.1),
genefilter (&ge; 1.72.1), ggrepel (&ge; 0.9.1), gplots (&ge; 3.1.1),
gridExtra (&ge; 2.3), plyr (&ge; 1.8.6), ppcor (&ge; 1.1), tidyr (&ge;
1.1.3), Hmisc (&ge; 4.4.2), methods (&ge; 4.0.3), rlang (&ge;
0.4.10), tidyselect (&ge; 1.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>kableExtra (&ge; 1.3.2), knitr (&ge; 1.31), rmarkdown (&ge; 2.6)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>python3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-29 16:18:07 UTC; manuel.kuhn</td>
</tr>
<tr>
<td>Author:</td>
<td>José Manuel Monroy Kuhn [aut, cre],
  Dominik Lutter [ths],
  Valentina Klaus [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>José Manuel Monroy Kuhn &lt;nolozz@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-30 09:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='assign_colorsAnnotation'>Assing Colors to Class</h2><span id='topic+assign_colorsAnnotation'></span>

<h3>Description</h3>

<p>This function assigns two color columns (color name and rgb) to an annotation data frame according to a column named 'Class' or 'class'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign_colorsAnnotation(AnnotationDf, col = "Class")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign_colorsAnnotation_+3A_annotationdf">AnnotationDf</code></td>
<td>
<p>Annotation data frame that contains a factor variable to use to assign colors</p>
</td></tr>
<tr><td><code id="assign_colorsAnnotation_+3A_col">col</code></td>
<td>
<p>Column with factor variable that will be used to assign colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input data.frame with two extra columns specifying the colors
for all vertexes according to their respective vertex-class
</p>

<hr>
<h2 id='barplot_VertexsPerEdgeFeature'>Number Vertex features per class for every shared edge feature</h2><span id='topic+barplot_VertexsPerEdgeFeature'></span>

<h3>Description</h3>

<p>Internal use. This function creates a barplot depicting the number of vertex features per class for every edge feature. To use this function one has to split first the file (or not if it is small) with the funciton chunk2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barplot_VertexsPerEdgeFeature(
  SplitFile,
  title = "Vertex Features per class",
  AnnotationWithColors,
  ggrepelL = TRUE,
  xlb = "Gene",
  szggrepel = 2.5,
  szTitle = 12,
  szaxisTxt = 12,
  szaxisTitle = 12,
  szLegendText = 10,
  szLegendKey = 1
)
</code></pre>


<h3>Value</h3>

<p>ggplot object for a barplot depicting the number of vertex features per class for every edge feature
</p>

<hr>
<h2 id='check_outputDir'>Output directory</h2><span id='topic+check_outputDir'></span>

<h3>Description</h3>

<p>Internal use. This function checks if the output directory exists and if it does not, it will create it
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_outputDir(outputDir, verb)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='check_previous'>Check previous files</h2><span id='topic+check_previous'></span>

<h3>Description</h3>

<p>Internal use. This function checks previous files and deletes according to the User input option
Requires some modifications...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_previous(del, iteration, outDir, verb)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='checkInputParameters'>Check input parameters</h2><span id='topic+checkInputParameters'></span>

<h3>Description</h3>

<p>Internal use. Function  to check if input parameters are of the right class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkInputParameters(ParaList)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='Chow_GeneExpData'>Chow gene expression data</h2><span id='topic+Chow_GeneExpData'></span>

<h3>Description</h3>

<p>Chow gene expression data
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='Chow_MetaboliteData'>Chow metabolite data</h2><span id='topic+Chow_MetaboliteData'></span>

<h3>Description</h3>

<p>Chow metabolite data
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='chunk2'>Split function</h2><span id='topic+chunk2'></span>

<h3>Description</h3>

<p>Internal use. Function to split the EdgeFeatures in smaller groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chunk2(x, n)
</code></pre>


<h3>Value</h3>

<p>A list of character vectors, each vector contains n edge features
</p>

<hr>
<h2 id='compare_sampleNames'>Compare sample names</h2><span id='topic+compare_sampleNames'></span>

<h3>Description</h3>

<p>Internal use. This function compares the sample names between the two datasets provided. If names do not match CoNI stops and outputs an error message.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_sampleNames(df1, df2)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='Compare_Triplets'>Compare triplets</h2><span id='topic+Compare_Triplets'></span>

<h3>Description</h3>

<p>Compare vertexFeature-vertexFeature-edgeFeature between two treatments, that is, find the shared triplets between two different CoNI runs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compare_Triplets(
  Treat1_path,
  Treat2_path,
  OutputName = "Shared_Genes_and_Edges_Treat1vsTreat2.csv"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compare_Triplets_+3A_treat1_path">Treat1_path</code></td>
<td>
<p>TableForNetwork_file1 (file generated by CoNI) with path for Treatment 1</p>
</td></tr>
<tr><td><code id="Compare_Triplets_+3A_treat2_path">Treat2_path</code></td>
<td>
<p>TableForNetwork_file2 (file generated by CoNI) with path for Treatment 2</p>
</td></tr>
<tr><td><code id="Compare_Triplets_+3A_outputname">OutputName</code></td>
<td>
<p>Output file name with path</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with the shared triplets (vertex1 vertex2 edge_feature) between two CoNI runs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#For an example see the vignette
</code></pre>

<hr>
<h2 id='Compare_VertexClasses_sharedEdgeFeatures'>Table VertexClass pairs of shared Edge Features</h2><span id='topic+Compare_VertexClasses_sharedEdgeFeatures'></span>

<h3>Description</h3>

<p>Compare VertexClass pairs of the shared Edge Features of two treatments (e.g., lipid-class-pairs per shared gene)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Compare_VertexClasses_sharedEdgeFeatures(
  Treat1_path,
  Treat2_path,
  OutputName = "Shared_Genes_and_Edges_Treat1vsTreat2.csv",
  Treat1Name = "Treat1",
  Treat2Name = "Treat2"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Compare_VertexClasses_sharedEdgeFeatures_+3A_treat1_path">Treat1_path</code></td>
<td>
<p>TableForNetwork_file (file generated by CoNI) with path of Treatment 1</p>
</td></tr>
<tr><td><code id="Compare_VertexClasses_sharedEdgeFeatures_+3A_treat2_path">Treat2_path</code></td>
<td>
<p>TableForNetwork_file (file generated by CoNI) with path of Treatment 2</p>
</td></tr>
<tr><td><code id="Compare_VertexClasses_sharedEdgeFeatures_+3A_outputname">OutputName</code></td>
<td>
<p>Output file name with path</p>
</td></tr>
<tr><td><code id="Compare_VertexClasses_sharedEdgeFeatures_+3A_treat1name">Treat1Name</code></td>
<td>
<p>Name of treatment one, default Treat1</p>
</td></tr>
<tr><td><code id="Compare_VertexClasses_sharedEdgeFeatures_+3A_treat2name">Treat2Name</code></td>
<td>
<p>Name of treatment one, default Treat2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with all possible vertex-class pairs and their numbers per edge-feature and treatment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#For an example see the vignette
</code></pre>

<hr>
<h2 id='CoNI'>Correlation guided Network Integration</h2><span id='topic+CoNI'></span>

<h3>Description</h3>

<p>CoNI is the main function of Correlation guided Network Integration (CoNI). Input data should come from two sources (e.g., gene expression and metabolite expression), and it should come from the same samples. It calculates all pairwise correlations of the second data input elements and the partial correlations of these pairwise elements with respect to the elements of the first data input. Both data inputs can be prefiltered to include only those elements that significantly correlate. The first data input can be prefiltered to keep just low variance elements (var&lt;0.5). A Steiger test is used to identify significant changes between the correlation and partial correlation values. Results can be visually represented in a Network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoNI(
  edgeD,
  vertexD,
  outputDir = "./CoNIOutput/",
  saveRaw = TRUE,
  outputNameRaw = "CoNIOutput",
  onlySgRes = FALSE,
  multipleTAdj = TRUE,
  padjustvertexD = TRUE,
  correlateDFs = TRUE,
  filter_highVarianceEdge = TRUE,
  splitedgeD = TRUE,
  split_number = 2,
  delPrevious = FALSE,
  delIntermediaryFiles = TRUE,
  iteration_start = 1,
  numCores = NULL,
  verbose = TRUE,
  more_coef = FALSE,
  edgeDname = "edgeD",
  vertexDname = "vertexD",
  saveFiles = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoNI_+3A_edged">edgeD</code></td>
<td>
<p>Object to use as first data input (e.g., protein expression)</p>
</td></tr>
<tr><td><code id="CoNI_+3A_vertexd">vertexD</code></td>
<td>
<p>Object to use as second data input (e.g., metabolite expression)</p>
</td></tr>
<tr><td><code id="CoNI_+3A_outputdir">outputDir</code></td>
<td>
<p>Output Directory where results are stored</p>
</td></tr>
<tr><td><code id="CoNI_+3A_saveraw">saveRaw</code></td>
<td>
<p>logical. If TRUE the raw output of CoNI is saved in the output directory (outputDir)</p>
</td></tr>
<tr><td><code id="CoNI_+3A_outputnameraw">outputNameRaw</code></td>
<td>
<p>Name for the raw output file if saved</p>
</td></tr>
<tr><td><code id="CoNI_+3A_onlysgres">onlySgRes</code></td>
<td>
<p>logical. If TRUE CoNI output is filtered and only significant results are kept</p>
</td></tr>
<tr><td><code id="CoNI_+3A_multipletadj">multipleTAdj</code></td>
<td>
<p>logical. If TRUE it will filter results after adjustment of multiple testing</p>
</td></tr>
<tr><td><code id="CoNI_+3A_padjustvertexd">padjustvertexD</code></td>
<td>
<p>logical. If TRUE vertexD is filtered according to the significant adjusted p-value of its pairwise correlations</p>
</td></tr>
<tr><td><code id="CoNI_+3A_correlatedfs">correlateDFs</code></td>
<td>
<p>logical. If TRUE the elements that significantly correlate of vertexD are correlated with the elements of edgeD. Only the elements that significantly correlate are kept</p>
</td></tr>
<tr><td><code id="CoNI_+3A_filter_highvarianceedge">filter_highVarianceEdge</code></td>
<td>
<p>logical. If TRUE features of edgeD with high variance are filtered out</p>
</td></tr>
<tr><td><code id="CoNI_+3A_splitedged">splitedgeD</code></td>
<td>
<p>logical. If TRUE edgeD will be split in n subsets for the computation (some instances n+1). Keep as TRUE unless the data input is small</p>
</td></tr>
<tr><td><code id="CoNI_+3A_split_number">split_number</code></td>
<td>
<p>Number of parts to split the elements of edgeD</p>
</td></tr>
<tr><td><code id="CoNI_+3A_delprevious">delPrevious</code></td>
<td>
<p>logical. If TRUE previous files of a previous run are deleted</p>
</td></tr>
<tr><td><code id="CoNI_+3A_delintermediaryfiles">delIntermediaryFiles</code></td>
<td>
<p>logical. If TRUE the output file of every iteration is deleted and only a single file with all results is kept</p>
</td></tr>
<tr><td><code id="CoNI_+3A_iteration_start">iteration_start</code></td>
<td>
<p>Iteration start for CoNI. Useful if run is interrupted as one can restart from the last iteration</p>
</td></tr>
<tr><td><code id="CoNI_+3A_numcores">numCores</code></td>
<td>
<p>Cores assigned for parallelization</p>
</td></tr>
<tr><td><code id="CoNI_+3A_verbose">verbose</code></td>
<td>
<p>logical. If TRUE output in the console is more verbose</p>
</td></tr>
<tr><td><code id="CoNI_+3A_more_coef">more_coef</code></td>
<td>
<p>logical. If TRUE it will include the partial correlation of edge and vertex Features</p>
</td></tr>
<tr><td><code id="CoNI_+3A_edgedname">edgeDname</code></td>
<td>
<p>File name extension for the edge features that significantly correlate with at least one vertex feature. This file will be read if the function is called again with the same input and with delPrevious=FALSE</p>
</td></tr>
<tr><td><code id="CoNI_+3A_vertexdname">vertexDname</code></td>
<td>
<p>File name extension for the vertex features that are involved in at least one significant correlation. This file will be read if the function is called again with the same input and with delPrevious=FALSE</p>
</td></tr>
<tr><td><code id="CoNI_+3A_savefiles">saveFiles</code></td>
<td>
<p>logical. If FALSE CoNI function will not save any file to disk</p>
</td></tr>
</table>


<h3>Value</h3>

<p>CoNI returns a data.frame with the correlation coefficients of the vertex-pairs, the partial correlation coefficients for every triplet, and the pvalue of the Steiger tests
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Run CoNI

#Load gene expression - Toy dataset of two treatments
data(GeneExpToy)
#Samples in rows and genes in columns
GeneExp &lt;- as.data.frame(t(GeneExpToy))
hfd_gene &lt;- GeneExp[1:8,] #high fat diet
chow_gene&lt;- GeneExp[9:nrow(GeneExp),] #chow diet
#Load metabolite expression - Toy dataset of two treatments
data(MetaboExpToy)
MetaboExp &lt;- MetaboExpToy
hfd_metabo &lt;- MetaboExp[11:18,] #high fat diet
chow_metabo &lt;- MetaboExp[1:10,] #chow diet
#Match row names both data sets
rownames(hfd_metabo)&lt;-rownames(hfd_gene)
rownames(chow_metabo)&lt;-rownames(chow_gene)

#Run CoNI with tiny example and no significance testing
#High fat diet
#For big datasets it is recommended to set splitedgeD to TRUE
#and split_number should be adjusted accordingly
#See vignette for an example
#Running CoNI with only a tiny dataset

 CoNIResultsHFD &lt;- CoNI(hfd_gene,hfd_metabo,
                        numCores = 2,
                        onlySgRes = FALSE,
                        filter_highVarianceEdge=FALSE,
                        padjustvertexD = FALSE,
                        correlateDFs = FALSE,
                        edgeDname="HFD_genes",
                        vertexDname = "HFD_metabolites",
                        saveFiles = FALSE,
                        splitedgeD = FALSE,
                        outputDir = "./")


</code></pre>

<hr>
<h2 id='CoNIResults_Chow'>CoNI Results Chow</h2><span id='topic+CoNIResults_Chow'></span>

<h3>Description</h3>

<p>CoNI Results Chow
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='CoNIResults_HFD'>CoNI Results HFD</h2><span id='topic+CoNIResults_HFD'></span>

<h3>Description</h3>

<p>CoNI Results HFD
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='CoNIResultsHFDToy'>Toy data HFD results</h2><span id='topic+CoNIResultsHFDToy'></span>

<h3>Description</h3>

<p>Toy data HFD results
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='countClassPerEdgeFeature'>Number lipid features per class</h2><span id='topic+countClassPerEdgeFeature'></span>

<h3>Description</h3>

<p>Internal use. This function counts for every edge feature the number of vertex features (e.g. lipids) per class. It disregards the number of vertex pairs...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countClassPerEdgeFeature(ResTable, treatment = "Chow")
</code></pre>


<h3>Value</h3>

<p>A data.frame with the number of vertex features per class and edge feature
</p>

<hr>
<h2 id='create_edgeFBarplot'>Vertex-class pairs profile of one shared edge feature</h2><span id='topic+create_edgeFBarplot'></span>

<h3>Description</h3>

<p>This function will create a barplot from the output of Compare_VertexClasses_sharedEdgeFeatures for a specific shared Edge Feature (e.g., a shared gene).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_edgeFBarplot(
  CompTreatTable,
  edgeF,
  treat1 = "Treatment1",
  treat2 = "Treatment2",
  factorOrder = NULL,
  col1 = "red",
  col2 = "blue",
  EdgeFeatureType = "Edge Feature",
  xlb = "Vertex-Class Pairs",
  ylb = "Number of pairs",
  szaxisTxt = 12,
  szaxisTitle = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_edgeFBarplot_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_edgef">edgeF</code></td>
<td>
<p>Edge feature present in output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_treat1">treat1</code></td>
<td>
<p>Name of treatment one, default Treatment1. It should match the column names of the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_treat2">treat2</code></td>
<td>
<p>Name of treatment one, default Treatment2. It should match the column names of the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_factororder">factorOrder</code></td>
<td>
<p>A list specifying the order of the treatments.</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_col1">col1</code></td>
<td>
<p>Color for Treatment 1</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_col2">col2</code></td>
<td>
<p>Color for Treatment 2</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_edgefeaturetype">EdgeFeatureType</code></td>
<td>
<p>Type of Edge Feature (e.g., Gene)</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_xlb">xlb</code></td>
<td>
<p>Name for x-axis</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_ylb">ylb</code></td>
<td>
<p>Name for the y-axis</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_szaxistxt">szaxisTxt</code></td>
<td>
<p>Size axis text</p>
</td></tr>
<tr><td><code id="create_edgeFBarplot_+3A_szaxistitle">szaxisTitle</code></td>
<td>
<p>Size axis titles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object for a barplot. The barplot shows the vertex-class pairs profile of a single shared edge feature between two treatments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
create_edgeFBarplot(CompTreatTable = VertexClassesSharedGenes_HFDvsChow,
                    edgeF = "Lilr4b",
                    treat1 = "HFD",
                    treat2 = "Chow",
                    factorOrder = c("HFD","Chow"),
                    EdgeFeatureType = "Gene")
</code></pre>

<hr>
<h2 id='create_GlobalBarplot'>Vertex-class pairs profile of shared features</h2><span id='topic+create_GlobalBarplot'></span>

<h3>Description</h3>

<p>This function will create a barplot from the output of Compare_VertexClasses_sharedEdgeFeatures using all shared Edge Features (e.g., genes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_GlobalBarplot(
  CompTreatTable,
  treat1 = "Treatment1",
  treat2 = "Treatment2",
  factorOrder = NULL,
  col1 = "red",
  col2 = "blue",
  maxpairs = 1,
  xlb = "Vertex-Class Pairs",
  ylb = "Number of pairs",
  szggrepel = 3.5,
  nudgey = 0.5,
  nudgex = 0.5,
  szaxisTxt = 12,
  szaxisTitle = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_GlobalBarplot_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_treat1">treat1</code></td>
<td>
<p>Name of treatment one, default Treatment1. It should match the column names of the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_treat2">treat2</code></td>
<td>
<p>Name of treatment one, default Treatment2. It should match the column names of the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_factororder">factorOrder</code></td>
<td>
<p>A list specifying the order of the treatments.</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_col1">col1</code></td>
<td>
<p>Color for Treatment 1</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_col2">col2</code></td>
<td>
<p>Color for Treatment 2</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_maxpairs">maxpairs</code></td>
<td>
<p>If number of class-vertex-pairs &gt; maxpairs, display number pairs on top of bar</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_xlb">xlb</code></td>
<td>
<p>Name for x-axis</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_ylb">ylb</code></td>
<td>
<p>Name for the y-axis</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_szggrepel">szggrepel</code></td>
<td>
<p>Size ggrepel labels</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_nudgey">nudgey</code></td>
<td>
<p>Nudge y ggrepel</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_nudgex">nudgex</code></td>
<td>
<p>Nudge x ggrepel</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_szaxistxt">szaxisTxt</code></td>
<td>
<p>Size axis text</p>
</td></tr>
<tr><td><code id="create_GlobalBarplot_+3A_szaxistitle">szaxisTitle</code></td>
<td>
<p>Size axis title</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object for a barplot. The barplot shows the vertex-class pairs profile of all shared edge features between treatments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
create_GlobalBarplot(CompTreatTable = VertexClassesSharedGenes_HFDvsChow,
                     treat1 = "HFD",
                     treat2 = "Chow",
                     factorOrder = c("HFD","Chow"),
                     col1="red",
                     col2 ="blue",
                     maxpairs = 1,
                     szggrepel = 6,
                     szaxisTxt = 15,
                     szaxisTitle = 15,
                     xlb = "Metabolite-pair classes")
</code></pre>

<hr>
<h2 id='create_stackedGlobalBarplot_perTreatment'>Stacked Global Barplot (One treatment)</h2><span id='topic+create_stackedGlobalBarplot_perTreatment'></span>

<h3>Description</h3>

<p>This function will create a stacked barplot from the output of Compare_VertexClasses_sharedEdgeFeatures using all shared Edge Features (e.g., genes) between two treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_stackedGlobalBarplot_perTreatment(
  CompTreatTable,
  treat = NULL,
  xlb = "Vertex-Class Pairs",
  ylb = "Number of pairs",
  max_pairsLegend = 2,
  mx.overlaps = Inf,
  szggrepel = 6,
  force = 0.1,
  szTitle = 12,
  szaxisTxt = 12,
  szaxisTitle = 12,
  ylim = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_treat">treat</code></td>
<td>
<p>Name of treatment to display. It should match the column name in the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_xlb">xlb</code></td>
<td>
<p>Name for x-axis</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_ylb">ylb</code></td>
<td>
<p>Name for y-axis</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_max_pairslegend">max_pairsLegend</code></td>
<td>
<p>If number of Edge Features &gt;= max_pairsLegend, display number of Edge Features as label with ggrepel</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_mx.overlaps">mx.overlaps</code></td>
<td>
<p>Max number of overlaps ggrepel</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_szggrepel">szggrepel</code></td>
<td>
<p>Size ggrepel labels</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_force">force</code></td>
<td>
<p>Repelling force for ggrepel labels</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_sztitle">szTitle</code></td>
<td>
<p>Size Title</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_szaxistxt">szaxisTxt</code></td>
<td>
<p>Size axis text</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_szaxistitle">szaxisTitle</code></td>
<td>
<p>Size axis titles</p>
</td></tr>
<tr><td><code id="create_stackedGlobalBarplot_perTreatment_+3A_ylim">ylim</code></td>
<td>
<p>Optional y-limits of the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object to create a stacked barplot. The stacked barplot shows the vertex-class pairs profile of all shared edge features but restricted to a single treatment. Every bar consists of multiple edge features (stacked) that are represented with different colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
create_stackedGlobalBarplot_perTreatment(CompTreatTable = VertexClassesSharedGenes_HFDvsChow,
                                         treat = "HFD",
                                         max_pairsLegend = 9,
                                         xlb = "Metabolite-class-pairs")
</code></pre>

<hr>
<h2 id='createBipartiteGraph'>Bipartite Network</h2><span id='topic+createBipartiteGraph'></span>

<h3>Description</h3>

<p>This function creates a simple bipartite graph, it shows the driver and linker features as nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createBipartiteGraph(TableNetwork, colorVertexTable, incidenceMatrix = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createBipartiteGraph_+3A_tablenetwork">TableNetwork</code></td>
<td>
<p>TableForNetwork_file (file generated by CoNI) with path</p>
</td></tr>
<tr><td><code id="createBipartiteGraph_+3A_colorvertextable">colorVertexTable</code></td>
<td>
<p>Table specifying the colors for the vertex features. The first column should contain the names matching the features of the vertex Data and another column should specify the colors (column name: Colors).</p>
</td></tr>
<tr><td><code id="createBipartiteGraph_+3A_incidencematrix">incidenceMatrix</code></td>
<td>
<p>logical. If TRUE it returns a hypergraph incidence matrix instead of a bipartite graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An igraph object for a bipartite graph or a hypergraph incidence matrix to represent ResultsCoNI. Basic network statistics are included in the bipartite graph. See generate_network function for details or consult the igraph package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#See vignette for an example
</code></pre>

<hr>
<h2 id='createBipartiteTable'>Bipartite Table</h2><span id='topic+createBipartiteTable'></span>

<h3>Description</h3>

<p>Internal use. This function creates a table that is used to create a simple bipartite network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createBipartiteTable(CoNINetworkTable)
</code></pre>


<h3>Value</h3>

<p>A matrix containing two columns, one for vertexes and one for edge features. The matching happens if they are adjacent to one another
</p>

<hr>
<h2 id='delIntFiles'>Delete intermediary files</h2><span id='topic+delIntFiles'></span>

<h3>Description</h3>

<p>Internal use. This function deletes intermediary files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delIntFiles(del, outDir)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='do_objectsExist'>Check if files exist</h2><span id='topic+do_objectsExist'></span>

<h3>Description</h3>

<p>Internal use. This function tests if the input files exist, if they do not it will output an error and
end CoNI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_objectsExist(gene_exp, norm_metabo_dat, verb)
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='find_localControllingFeatures'>Find local controlling features</h2><span id='topic+find_localControllingFeatures'></span>

<h3>Description</h3>

<p>This function applies for a selected subnetwork a binomial test using the frequency of appearance of an edge feature and the total number of edge features. The probability corresponds to 1/n_df, where n_df corresponds to the total number of edge features in the network.
The selected subnetwork corresponds to the second level neighborhood of a specific node. The test is applied to all possible second level neighborhoods in the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_localControllingFeatures(ResultsCoNI, network, padjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_localControllingFeatures_+3A_resultsconi">ResultsCoNI</code></td>
<td>
<p>The output of CoNI (after p-adjustment)</p>
</td></tr>
<tr><td><code id="find_localControllingFeatures_+3A_network">network</code></td>
<td>
<p>Network created with the function generate_network</p>
</td></tr>
<tr><td><code id="find_localControllingFeatures_+3A_padjust">padjust</code></td>
<td>
<p>logical. Filter output based on adjusted p values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with the results of the binomial tests. Significant results correspond to local controlling features
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load color nodes table
data(MetColorTable)

#Assign colors according to "Class" column
MetColorTable&lt;-assign_colorsAnnotation(MetColorTable)

#Load CoNI results
data(CoNIResultsHFDToy)

#Generate Network
#Note: Colors not visible when ploting in Igraph
HFDNetwork&lt;-generate_network(ResultsCoNI = CoNIResultsHFDToy,
                             colorVertexNetwork = TRUE,
                             colorVertexTable = MetColorTable,
                             Class = MetColorTable,
                             outputDir = "./",
                             outputFileName = "HFD",
                             saveFiles = FALSE)

#Note: For this tiny example nothing is significant
LCG_BinomialTestTableHFD&lt;- find_localControllingFeatures(ResultsCoNI = CoNIResultsHFDToy,
                                                         network = HFDNetwork )
LCGenes_HFD&lt;-as.character(unique(LCG_BinomialTestTableHFD$edgeFeatures))

</code></pre>

<hr>
<h2 id='flattenCorrMatrix'>Flatten</h2><span id='topic+flattenCorrMatrix'></span>

<h3>Description</h3>

<p>Internal use. This function gets the upper part of the matrix after calculating the correlation coefficients between all pairs of elements
cormat : matrix of the correlation coefficients
pmat : matrix of the correlation p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flattenCorrMatrix(cormat, pmat)
</code></pre>


<h3>Value</h3>

<p>A data.frame with all pairwise correlations of the input elements and their respective p-values
</p>

<hr>
<h2 id='GeneExpToy'>Toy data gene expression</h2><span id='topic+GeneExpToy'></span>

<h3>Description</h3>

<p>Toy data gene expression
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='generate_network'>Create network</h2><span id='topic+generate_network'></span>

<h3>Description</h3>

<p>This function creates a network using as input the output of CoNI and a table specifying the colors for the nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_network(
  ResultsCoNI,
  colorVertexNetwork = TRUE,
  colorVertexTable,
  outputDir = "./",
  outputFileName = "ResultsCoNI",
  Class = NULL,
  saveFiles = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_network_+3A_resultsconi">ResultsCoNI</code></td>
<td>
<p>The input of the function are the results of CoNI.</p>
</td></tr>
<tr><td><code id="generate_network_+3A_colorvertexnetwork">colorVertexNetwork</code></td>
<td>
<p>logical. If TRUE, the table colorVertexTable has to be provided to specify vertex colors</p>
</td></tr>
<tr><td><code id="generate_network_+3A_colorvertextable">colorVertexTable</code></td>
<td>
<p>Table specifying the colors for the nodes (vertex features). The first column should contain the names matching the features of the vertex Data and the colors or other data can be specified in the rest of the columns</p>
</td></tr>
<tr><td><code id="generate_network_+3A_outputdir">outputDir</code></td>
<td>
<p>Output directory where the network is saved as well as the file that was used to generate the network.</p>
</td></tr>
<tr><td><code id="generate_network_+3A_outputfilename">outputFileName</code></td>
<td>
<p>The name of the file used to create the network.</p>
</td></tr>
<tr><td><code id="generate_network_+3A_class">Class</code></td>
<td>
<p>Optional data frame with at least two columns, first column contains all vertex features and another column the vertex feature class (column named &quot;Class&quot;). Necessary for treatment comparisons based on class</p>
</td></tr>
<tr><td><code id="generate_network_+3A_savefiles">saveFiles</code></td>
<td>
<p>logical. If FALSE TableForNetwork_'outputFileName'.csv and Network_'outputFileName'.graphml are not saved to disk</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an igraph object (network) constructed from ResultsCoNI. The network includes the following network statistics
</p>

<ul>
<li><p>&quot;degree&quot;The number of the vertex adjacent edges
</p>
</li>
<li><p>&quot;hub_score&quot;The principal eigenvector of A*t(A), where A is the adjacency matrix of the graph
</p>
</li>
<li><p>&quot;transitivity&quot;Probability that the adjacent vertices of a vertex are connected
</p>
</li>
<li><p>&quot;closeness&quot;Steps required to access every other vertex from a given vertex
</p>
</li>
<li><p>&quot;betweenness&quot;(roughly) The number of geodesics (shortest paths) going through a vertex or an edge
</p>
</li>
<li><p>&quot;eigen_centrality&quot;Takes a graph (graph) and returns the eigenvector centralities of positions v within it
</p>
</li>
<li><p>&quot;centralized_betweenness&quot;The vertice-level centrality score according to the betweenness of vertices
</p>
</li>
<li><p>&quot;centralized_closeness&quot;The vertice-level centrality score according to the closeness of vertices
</p>
</li>
<li><p>&quot;centralized_degree&quot;The vertice-level centrality score according to the degrees of vertices
</p>
</li></ul>

<p>For more details see igraph package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate Network

#Load color nodes table
data(MetColorTable)
#Assign colors according to "Class" column
MetColorTable&lt;-assign_colorsAnnotation(MetColorTable)
#Load CoNI results
data(CoNIResultsHFDToy)

#Generate Network
HFDNetwork&lt;-generate_network(ResultsCoNI = CoNIResultsHFDToy,
                             colorVertexNetwork = TRUE,
                             colorVertexTable = MetColorTable,
                             outputDir = "./",
                             outputFileName = "HFD",
                             saveFiles = FALSE)
</code></pre>

<hr>
<h2 id='get_lowvarFeatures'>Low variance features</h2><span id='topic+get_lowvarFeatures'></span>

<h3>Description</h3>

<p>Internal use. Get low variance features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lowvarFeatures(df)
</code></pre>


<h3>Value</h3>

<p>A data.frame where low variance features were removed
</p>

<hr>
<h2 id='getcolor'>Get class rgb color</h2><span id='topic+getcolor'></span>

<h3>Description</h3>

<p>Internal use. This function gets the class rgb color of a specific (lipid) class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getcolor(ClassM, tableColor)
</code></pre>


<h3>Value</h3>

<p>A character object, that corresponds to a color in hexadecimal format
</p>

<hr>
<h2 id='getstackedGlobalBarplot_and_Grid'>Stacked Global Barplot Side-by-side (two treatments)</h2><span id='topic+getstackedGlobalBarplot_and_Grid'></span>

<h3>Description</h3>

<p>This function will create a stacked barplot from the output of Compare_VertexClasses_sharedEdgeFeatures using all shared Edge Features (e.g., genes) between two treatments. Results of both Treatments are side by side for better comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getstackedGlobalBarplot_and_Grid(
  CompTreatTable,
  Treat1,
  Treat2,
  ggrep = TRUE,
  max_pairsLegend = 1,
  force = 0.1,
  mx.overlaps = Inf,
  szggrepel = 6,
  xlb = "Vertex-Class Pairs",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_treat1">Treat1</code></td>
<td>
<p>Name treatment 1 as in table CompTreatTable</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_treat2">Treat2</code></td>
<td>
<p>Name treatment 2 as in table CompTreatTable</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_ggrep">ggrep</code></td>
<td>
<p>logical. If TRUE includes ggrepel labels for every bar</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_max_pairslegend">max_pairsLegend</code></td>
<td>
<p>If number of Edge Features &gt;= max_pairsLegend, display number of Edge Features as ggrepel label</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_force">force</code></td>
<td>
<p>Repelling force for ggrepel labels</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_mx.overlaps">mx.overlaps</code></td>
<td>
<p>Max number of overlaps ggrepel</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_szggrepel">szggrepel</code></td>
<td>
<p>Size ggrepel labels</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_xlb">xlb</code></td>
<td>
<p>Name for x-axis</p>
</td></tr>
<tr><td><code id="getstackedGlobalBarplot_and_Grid_+3A_...">...</code></td>
<td>
<p>Other parameters for inner functions, mainly ggplot2 visual parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gtable containing stacked barplots. The barplots show the vertex-class pairs profile of all shared edge features between two treatments (one bar plot per treatment). Every bar consists of multiple edge features that are depicted with different colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
VCSGs&lt;-VertexClassesSharedGenes_HFDvsChow
HFD_vs_Chow_stackedBarplot&lt;-getstackedGlobalBarplot_and_Grid(VCSGs,
                                                             Treat1 = "HFD",
                                                             Treat2 = "Chow",
                                                             xlb = "Metabolite-class-pairs",
                                                             max_pairsLegend=9)
plot(HFD_vs_Chow_stackedBarplot)
</code></pre>

<hr>
<h2 id='getvertexes_edgeFeature'>Get vertexes for edge feature</h2><span id='topic+getvertexes_edgeFeature'></span>

<h3>Description</h3>

<p>Get vertexes for edge feature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getvertexes_edgeFeature(edgeFeature, CoNIResults)
</code></pre>


<h3>Value</h3>

<p>A character vector with the vertexes connected to a given edge feature
</p>

<hr>
<h2 id='getVertexsPerEdgeFeature'>Vertex Class profile per edge feature (one treatment)</h2><span id='topic+getVertexsPerEdgeFeature'></span>

<h3>Description</h3>

<p>This function creates a barplot or barplots showing the number of vertex features per class for every shared edge feature between two treatments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVertexsPerEdgeFeature(
  CompTreatTable,
  Annotation,
  chunks = 5,
  treat = NULL,
  small = FALSE,
  ggrep = TRUE,
  xlb = "Gene",
  onlyTable = FALSE,
  szTitle = 12,
  szaxisTxt = 12,
  szaxisTitle = 12,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_annotation">Annotation</code></td>
<td>
<p>Data frame that includes the rgb colors for every class. The column 'class' (or 'Class') has to be present and also the column 'ColorRgb'</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_chunks">chunks</code></td>
<td>
<p>To avoid a non readable dense plot the results can be spitted in multiple plots</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_treat">treat</code></td>
<td>
<p>Specify the treatment for which the plot will be created. It should be one of the two treatments in the output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_small">small</code></td>
<td>
<p>logical. If only a few edge features are in the input set as TRUE. A single plot will be created</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_ggrep">ggrep</code></td>
<td>
<p>logical. If TRUE includes ggrepel labels for every bar</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_xlb">xlb</code></td>
<td>
<p>x-axis label</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_onlytable">onlyTable</code></td>
<td>
<p>logical. If TRUE a table is returned instead of a plot</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_sztitle">szTitle</code></td>
<td>
<p>Size title</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_szaxistxt">szaxisTxt</code></td>
<td>
<p>Size axis text</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_szaxistitle">szaxisTitle</code></td>
<td>
<p>Size axis title</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_+3A_...">...</code></td>
<td>
<p>Other parameters for inner functions, mainly ggplot2 visual parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot objects to create different barplots. The barplots show the number of vertex features per class for every shared
edge feature between two treatments. The barplots restrict to one of the compared treatments. An alternative output
is a data.frame with the number of vertex features per class and edge feature (onlyTable=TRUE)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
data(MetColorTable)
#Note: No differences in example as all the Output of CoNI was kept
getVertexsPerEdgeFeature(CompTreatTable = VertexClassesSharedGenes_HFDvsChow,
                         Annotation = MetColorTable,
                         chunks = 2,
                         treat = "HFD")
</code></pre>

<hr>
<h2 id='getVertexsPerEdgeFeature_and_Grid'>Vertex-Class profile per edge feature Side-by-Side (two treatments)</h2><span id='topic+getVertexsPerEdgeFeature_and_Grid'></span>

<h3>Description</h3>

<p>This function creates a grid of barplots. The barplot of one side depicts the number of class vertex features per edge feature for treatment 1 and the other side the same barplot for treatment 2. Results of both Treatments are side by side for better comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVertexsPerEdgeFeature_and_Grid(
  CompTreatTable,
  Treat1,
  Treat2,
  Annotation,
  chunks = 3,
  ggrep = TRUE,
  xlb = "Edge Feature",
  onlyT = FALSE,
  small = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_comptreattable">CompTreatTable</code></td>
<td>
<p>Output of Compare_VertexClasses_sharedEdgeFeatures</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_treat1">Treat1</code></td>
<td>
<p>Name treatment 1 as in table CompTreatTable</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_treat2">Treat2</code></td>
<td>
<p>Name treatment 2 as in table CompTreatTable</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_annotation">Annotation</code></td>
<td>
<p>Data frame that includes the rgb colors for every class. The column 'class' (or 'Class') has to be present and also the column 'ColorRgb'</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_chunks">chunks</code></td>
<td>
<p>To avoid a non readable dense plot the results can be spitted in multiple plots</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_ggrep">ggrep</code></td>
<td>
<p>logical. If TRUE includes ggrepel labels for every bar</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_xlb">xlb</code></td>
<td>
<p>Change the x-axis label</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_onlyt">onlyT</code></td>
<td>
<p>logical. If TRUE a table is returned instead of a grid of plots</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_small">small</code></td>
<td>
<p>logical. If only a few edge features are in the input set as TRUE. A single plot will be created</p>
</td></tr>
<tr><td><code id="getVertexsPerEdgeFeature_and_Grid_+3A_...">...</code></td>
<td>
<p>Other parameters for inner functions, mainly ggplot2 visual parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gtable containing side-by-side barplots, one for each treatment, showing the number of vertex features per class for every shared edge feature
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(VertexClassesSharedGenes_HFDvsChow)
VCSGs&lt;-VertexClassesSharedGenes_HFDvsChow
data(MetColorTable)
HFD_vs_Chow_LCP_Gene&lt;-getVertexsPerEdgeFeature_and_Grid(VCSGs,
                                                        "HFD","Chow",
                                                        Annotation=MetColorTable,
                                                        ggrep=FALSE,
                                                        small = FALSE,
                                                        chunks = 3,
                                                        szLegendKey=0.2)
plot(HFD_vs_Chow_LCP_Gene)
</code></pre>

<hr>
<h2 id='HFD_GeneExpData'>HFD gene expression data</h2><span id='topic+HFD_GeneExpData'></span>

<h3>Description</h3>

<p>HFD gene expression data
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='HFD_MetaboliteData'>HFD metabolite data</h2><span id='topic+HFD_MetaboliteData'></span>

<h3>Description</h3>

<p>HFD metabolite data
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='labels2colors_2'>Labels to colors</h2><span id='topic+labels2colors_2'></span>

<h3>Description</h3>

<p>Internal use. This function is modified version of labels2colors of WGCNA and the internet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labels2colors_2(
  labels,
  zeroIsGrey = TRUE,
  colorSeq = NULL,
  naColor = "grey",
  commonColorCode = TRUE
)
</code></pre>


<h3>Value</h3>

<p>A character vector with colors
</p>

<hr>
<h2 id='merge_outpuSplitFiles'>Merge Files.</h2><span id='topic+merge_outpuSplitFiles'></span>

<h3>Description</h3>

<p>Internal use. This function reads the output split files generated by CoNI and generates a single result object. It is slow. Probably there are other faster alternatives.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_outpuSplitFiles(outputDir)
</code></pre>


<h3>Value</h3>

<p>A single data.table with the results of CoNI
</p>

<hr>
<h2 id='MetaboExpToy'>Toy data metabolite expression</h2><span id='topic+MetaboExpToy'></span>

<h3>Description</h3>

<p>Toy data metabolite expression
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='MetaboliteAnnotation'>Metabolite Annotation</h2><span id='topic+MetaboliteAnnotation'></span>

<h3>Description</h3>

<p>Metabolite Annotation
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='MetColorTable'>Toy data annotation</h2><span id='topic+MetColorTable'></span>

<h3>Description</h3>

<p>Toy data annotation
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='NetStats'>Network Statistics</h2><span id='topic+NetStats'></span>

<h3>Description</h3>

<p>This function calculates simple network statistics and returns them as a dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NetStats(Network)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NetStats_+3A_network">Network</code></td>
<td>
<p>An Igraph network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with nine rows with the following network statistics:
</p>

<ul>
<li><p>&quot;net_avg_pathL&quot;Shortest paths between vertices
</p>
</li>
<li><p>&quot;net_edge_density&quot;Graph density, ratio of the number of edges and the number of possible edges
</p>
</li>
<li><p>&quot;net_transitivity&quot;Probability that the adjacent vertices of a vertex are connected
</p>
</li>
<li><p>&quot;net_diameter&quot;Length of the longest geodesic
</p>
</li>
<li><p>&quot;net_nodes_first_path_diameter&quot;The nodes along the first found path with the length of diameter
</p>
</li>
<li><p>&quot;net_eigenvalue&quot;The eigenvalue corresponding to the centrality scores.
</p>
</li>
<li><p>&quot;net_centralized_betweenessIdx&quot;The graph level centrality index after centralizing the graph according to the betweenness of vertices
</p>
</li>
<li><p>&quot;net_centralized_closenessIdx&quot;The graph level centrality index after centralizing the graph according to the closeness of vertices
</p>
</li>
<li><p>&quot;net_centralized_degreeIdx&quot;The graph level centrality index after centralizing the graph according to the degrees of vertices
</p>
</li></ul>

<p>For more information on the statistics consult the igraph package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load color nodes table
data(MetColorTable)
#Assign colors according to "Class" column
MetColorTable&lt;-assign_colorsAnnotation(MetColorTable)
#Load CoNI results
data(CoNIResultsHFDToy)

#Generate Network
HFDNetwork&lt;-generate_network(ResultsCoNI = CoNIResultsHFDToy,
                             colorVertexNetwork = TRUE,
                             colorVertexTable = MetColorTable,
                             outputDir = "./",
                             outputFileName = "HFD",
                             saveFiles = FALSE)
NetStats(HFDNetwork)
</code></pre>

<hr>
<h2 id='obtain_groupcolors'>Get colors</h2><span id='topic+obtain_groupcolors'></span>

<h3>Description</h3>

<p>Internal use. This function gets the rgb colors of every (lipid) class and names them according to the class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obtain_groupcolors(Annotation)
</code></pre>


<h3>Value</h3>

<p>A character vector with the rgb colors named after the vertex-class (e.g. lipid class)
</p>

<hr>
<h2 id='plotPcorvsCor'>Correlation vs Partial correlation</h2><span id='topic+plotPcorvsCor'></span>

<h3>Description</h3>

<p>This function fits two linear models on standardize data and plots the results. It generates a scatter plot with two regression lines, where the slopes correspond to the correlation and partial correlation coefficients (blue for cor and red for pcor)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPcorvsCor(
  ResultsCoNI,
  edgeFeature,
  vertexD,
  edgeD,
  vertexFeatures = NULL,
  outputDir = "./",
  fname,
  label_edgeFeature = "Edge Feature",
  plot_to_screen = TRUE,
  height = 10,
  width = 8,
  saveFiles = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPcorvsCor_+3A_resultsconi">ResultsCoNI</code></td>
<td>
<p>The significant results generated by CoNI</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_edgefeature">edgeFeature</code></td>
<td>
<p>The edge feature to explore e.g. Fabp2 (for a gene)</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_vertexd">vertexD</code></td>
<td>
<p>Vertex data that was given as input to CoNI</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_edged">edgeD</code></td>
<td>
<p>Edge data that was given as input to CoNI</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_vertexfeatures">vertexFeatures</code></td>
<td>
<p>The vertex features to include as a list. If not specified all metabolites available in combination with the edgeFeature will be used</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_outputdir">outputDir</code></td>
<td>
<p>Output directory with path</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_fname">fname</code></td>
<td>
<p>File name to save the plots</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_label_edgefeature">label_edgeFeature</code></td>
<td>
<p>Name for plot title e.g. Gene or Protein</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_plot_to_screen">plot_to_screen</code></td>
<td>
<p>logical. If TRUE plots will be outputted to the plotting screen</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_height">height</code></td>
<td>
<p>height of the plotting area for the saved file</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_width">width</code></td>
<td>
<p>width of the plotting are for the saved file</p>
</td></tr>
<tr><td><code id="plotPcorvsCor_+3A_savefiles">saveFiles</code></td>
<td>
<p>logical. If FALSE plot is not saved to disk</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a ggplot object for a scatter plot with two regression lines.
The blue line is the regression of the vertex features, and the red line is the regression
of the resulting residuals after regressing each vertex feature with the edge feature.
The slope of the blue line corresponds to the pearson correlation coefficient and the slope of the red line
to the partial correlation coefficient
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load gene expression - Toy dataset of two treatments
data(GeneExpToy)
#Samples in rows and genes in columns
GeneExp &lt;- as.data.frame(t(GeneExpToy))
hfd_gene &lt;- GeneExp[1:8,] #high fat diet
chow_gene&lt;- GeneExp[9:nrow(GeneExp),] #chow diet

#Load metabolite expression - Toy dataset of two treatments
data(MetaboExpToy)
MetaboExp &lt;- MetaboExpToy
hfd_metabo &lt;- MetaboExp[11:18,] #high fat diet
chow_metabo &lt;- MetaboExp[1:10,] #chow diet

#Match row names both data sets
rownames(hfd_metabo)&lt;-rownames(hfd_gene)
rownames(chow_metabo)&lt;-rownames(chow_gene)

#Load CoNI results
data(CoNIResultsHFDToy)

plotPcorvsCor(ResultsCoNI = CoNIResultsHFDToy,
              edgeFeature = "Arfrp1",
              vertexFeatures = c("PC.ae.C40.2", "SM..OH..C22.1"),
              vertexD = hfd_metabo,
              edgeD = hfd_gene,
              label_edgeFeature = "Gene",
              plot_to_screen = TRUE,
              height = 10,
              saveFiles = FALSE)
</code></pre>

<hr>
<h2 id='sig_correlation2'>Pairwise correlations</h2><span id='topic+sig_correlation2'></span>

<h3>Description</h3>

<p>Internal use. This function calculates the pairwise correlations of a matrix (it uses Hmisc::rcorr function) and gets the significant correlations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig_correlation2(input_edgeD, padj = TRUE, method = "BH", verb)
</code></pre>


<h3>Value</h3>

<p>A data.frame with the significant correlations of a correlation matrix
</p>

<hr>
<h2 id='sig_correlation2Dfs'>Significant correlations 2 Df</h2><span id='topic+sig_correlation2Dfs'></span>

<h3>Description</h3>

<p>Internal use. This function input are two data frames (e.g. metabolites and genes). It calculates the correlation matrix and creates a table with only significant pairs. No correction for multiple testing is done
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig_correlation2Dfs(metabolite_data, gene_expression)
</code></pre>


<h3>Value</h3>

<p>A data.frame with vertex-edge significant correlation coefficients and their p-values
</p>

<hr>
<h2 id='split_df'>Split dataset</h2><span id='topic+split_df'></span>

<h3>Description</h3>

<p>Internal use. This function wills split the 'big' omics data into smaller data frames to improve computations, to avoid running out of memory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_df(AbundantDF, numberSplitDF_2 = 2)
</code></pre>


<h3>Value</h3>

<p>A list of data.frames
</p>

<hr>
<h2 id='tableLCFs_VFs'>Table local controlling edge features and vertex pairs</h2><span id='topic+tableLCFs_VFs'></span>

<h3>Description</h3>

<p>This function creates a table of the local controlling edge features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableLCFs_VFs(CoNIResults, LCFs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tableLCFs_VFs_+3A_coniresults">CoNIResults</code></td>
<td>
<p>The output of CoNI (after p-adjustment)</p>
</td></tr>
<tr><td><code id="tableLCFs_VFs_+3A_lcfs">LCFs</code></td>
<td>
<p>Local controlling edge features as a vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frane of local controlling edge features and their respective vertex pairs, and unique vertexes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load CoNI results
data(CoNIResultsHFDToy)
#Note: arbitrary list of genes, not Local controlling features
tableLCFs_VFs(CoNIResultsHFDToy, c("Lilr4b","Rps12"))
</code></pre>

<hr>
<h2 id='top_n_LF_byMagnitude'>Linker Features by magnitude of effect</h2><span id='topic+top_n_LF_byMagnitude'></span>

<h3>Description</h3>

<p>This function outputs the linker features with the strongest effect on the correlation of the vertex features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_n_LF_byMagnitude(ResultsCoNI, topn = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="top_n_LF_byMagnitude_+3A_resultsconi">ResultsCoNI</code></td>
<td>
<p>The output of CoNI</p>
</td></tr>
<tr><td><code id="top_n_LF_byMagnitude_+3A_topn">topn</code></td>
<td>
<p>Top n number of features to output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame, a filtered version of ResultsCoNI, showing the top n features
with the strongest effect, that is, the highest difference between the partial correlation and correlation coefficient.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CoNIResultsHFDToy)
Top10HFD&lt;-top_n_LF_byMagnitude(CoNIResults_HFD,topn = 10)
</code></pre>

<hr>
<h2 id='VertexClassesSharedGenes_HFDvsChow'>Toy data comparison treatments</h2><span id='topic+VertexClassesSharedGenes_HFDvsChow'></span>

<h3>Description</h3>

<p>Toy data comparison Vertex classes shared edge featuresHFD vs Chow
</p>


<h3>Author(s)</h3>

<p>Jose Manuel Monroy <a href="mailto:nolozz@gmail.com">nolozz@gmail.com</a>
</p>


<h3>References</h3>

<p>doi: <a href="https://doi.org/10.1016/j.molmet.2021.101295">10.1016/j.molmet.2021.101295</a>
</p>

<hr>
<h2 id='writeTable'>Write table</h2><span id='topic+writeTable'></span>

<h3>Description</h3>

<p>Internal use. This function tries to write a table with fread, if it fails it returns &quot;NA&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeTable(results_write, num_cores, outputDir, iteration)
</code></pre>


<h3>Value</h3>

<p>Returns NA if it fails to write file to memory, otherwise no return value
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
