<!DOCTYPE html><html><head><title>Help for package contrast</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {contrast}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#contrast.lm'><p>General Contrasts of Regression Coefficients</p></a></li>
<li><a href='#print.contrast'><p>Print a Contrast Object</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#two_factor_crossed'><p>Complete Two-Factor Experiment</p></a></li>
<li><a href='#two_factor_incompl'><p>Incomplete Two-Factor Experiment with Repeated Measurments</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Collection of Contrast Methods</td>
</tr>
<tr>
<td>Version:</td>
<td>0.24.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-05</td>
</tr>
<tr>
<td>Description:</td>
<td>One degree of freedom contrasts for 'lm', 'glm', 'gls', and 'geese' objects.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Alanocallaghan/contrast">https://github.com/Alanocallaghan/contrast</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Alanocallaghan/contrast/issues">https://github.com/Alanocallaghan/contrast/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>nlme, sandwich, rms</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, kableExtra, dplyr, ggplot2, tidyr, rmarkdown,
testthat, covr, geepack, splines</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-05 16:51:48 UTC; alan</td>
</tr>
<tr>
<td>Author:</td>
<td>Alan O'Callaghan [cre],
  Max Kuhn [aut],
  Steve Weston [aut],
  Jed Wing [aut],
  James Forester [aut],
  Thorn Thaler [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alan O'Callaghan &lt;alan.ocallaghan@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-05 17:20:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='contrast.lm'>General Contrasts of Regression Coefficients</h2><span id='topic+contrast.lm'></span><span id='topic+contrast.gls'></span><span id='topic+contrast.lme'></span><span id='topic+contrast.geese'></span><span id='topic+contrast_calc'></span>

<h3>Description</h3>

<p>This function computes one or more contrasts of the estimated regression
coefficients in a fit from one of the functions in Design, along with
standard errors, confidence limits, t or Z statistics, P-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lm'
contrast(fit, ...)

## S3 method for class 'gls'
contrast(fit, ...)

## S3 method for class 'lme'
contrast(fit, ...)

## S3 method for class 'geese'
contrast(fit, ...)

contrast_calc(
  fit,
  a,
  b,
  cnames = NULL,
  type = c("individual", "average"),
  weights = "equal",
  conf.int = 0.95,
  fcType = "simple",
  fcFunc = I,
  covType = NULL,
  ...,
  env = parent.frame(2)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contrast.lm_+3A_fit">fit</code></td>
<td>
<p>A fit of class <code>lm</code>, <code>glm</code>, etc.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_...">...</code></td>
<td>
<p>For <code>contrast()</code>, these pass arguments to <code>contrast_calc()</code>. For
<code>contrast_calc()</code>, they are not used.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_a">a</code>, <code id="contrast.lm_+3A_b">b</code></td>
<td>
<p>Lists containing conditions for all predictors in the model that
will be contrasted to form the hypothesis <code>H0: a = b</code>. The <code>gendata</code>
function will generate the necessary combinations and default values for
unspecified predictors. See examples below.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_cnames">cnames</code></td>
<td>
<p>A vector of character strings naming the contrasts when
<code>type = "individual"</code>.  Usually <code>cnames</code> is not necessary as
the function tries to name the contrasts by examining which
predictors are varying consistently in the two lists.  <code>cnames</code> will be
needed when you contrast &quot;non-comparable&quot; settings, e.g., you compare
<code>list(treat = "drug", age = c(20,30))</code> with
<code style="white-space: pre;">&#8288;list(treat = "placebo", age = c(40,50)&#8288;</code>.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_type">type</code></td>
<td>
<p>A character string. Set <code>type="average"</code> to average the
individual contrasts (e.g., to obtain a &quot;Type II&quot; or &quot;Type III&quot; contrast).</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_weights">weights</code></td>
<td>
<p>A numeric vector, used when <code>type = "average"</code>, to
obtain weighted contrasts.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_conf.int">conf.int</code></td>
<td>
<p>The confidence level for confidence intervals for the
contrasts.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_fctype">fcType</code></td>
<td>
<p>A character string: &quot;simple&quot;, &quot;log&quot; or &quot;signed&quot;.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_fcfunc">fcFunc</code></td>
<td>
<p>A function to transform the numerator and denominator
of fold changes.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_covtype">covType</code></td>
<td>
<p>A string matching the method for estimating the covariance
matrix. The default value produces the typical estimate. See
<code><a href="sandwich.html#topic+vcovHC">sandwich::vcovHC()</a></code> for options.</p>
</td></tr>
<tr><td><code id="contrast.lm_+3A_env">env</code></td>
<td>
<p>An environment in which evaluate fit.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions mirror <code><a href="rms.html#topic+contrast">rms::contrast.rms()</a></code> but have fewer options.
</p>
<p>There are some between-package inconsistencies regarding degrees of freedom
in some models. See the package vignette for more details.
</p>
<p>Fold changes are calculated for each hypothesis. When <code>fcType ="simple"</code>, the
ratio of the <code>a</code> group predictions over the <code>b</code>
group predictions are used. When <code>fcType = "signed"</code>, the ratio is used
if it is greater than 1; otherwise the negative inverse (e.g.,
<code>-1/ratio</code>) is returned.
</p>


<h3>Value</h3>

<p>a list of class <code>contrast.Design</code> containing the elements
<code>Contrast</code>, <code>SE</code>, <code>Z</code>, <code>var</code>, <code>df.residual</code> <code>Lower</code>, <code>Upper</code>, <code>Pvalue</code>, <code>X</code>,
<code>cnames</code>, and <code>foldChange</code>, which denote the contrast estimates, standard
errors, Z or t-statistics, variance matrix, residual degrees of freedom
(this is <code>NULL</code> if the model was not <code>ols</code>), lower and upper confidence
limits, 2-sided P-value, design matrix, and contrast names (or <code>NULL</code>).
</p>


<h3>See Also</h3>

<p><code><a href="rms.html#topic+contrast">rms::contrast.rms()</a></code>, <code><a href="sandwich.html#topic+vcovHC">sandwich::vcovHC()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(nlme)
Orthodont2 &lt;- Orthodont
Orthodont2$newAge &lt;- Orthodont$age - 11
fm1Orth.lme2 &lt;- lme(distance ~ Sex * newAge,
    data = Orthodont2,
    random = ~ newAge | Subject
)
summary(fm1Orth.lme2)

contrast(fm1Orth.lme2,
    a = list(Sex = levels(Orthodont2$Sex), newAge = 8 - 11),
    b = list(Sex = levels(Orthodont2$Sex), newAge = 10 - 11)
)

# ---------------------------------------------------------------------------

anova_model &lt;- lm(expression ~ diet * group, data = two_factor_crossed)
anova(anova_model)

library(ggplot2)
theme_set(theme_bw() + theme(legend.position = "top"))
ggplot(two_factor_crossed) +
    aes(x = diet, y = expression, col = group, shape = group) +
    geom_point() +
    geom_smooth(aes(group = group), method = lm, se = FALSE)

int_model &lt;- lm(expression ~ diet * group, data = two_factor_crossed)
main_effects &lt;- lm(expression ~ diet + group, data = two_factor_crossed)

# Interaction effect is probably real:
anova(main_effects, int_model)

# Test treatment in low fat diet:
veh_group &lt;- list(diet = "low fat", group = "vehicle")
trt_group &lt;- list(diet = "low fat", group = "treatment")
contrast(int_model, veh_group, trt_group)

# ---------------------------------------------------------------------------

car_mod &lt;- lm(mpg ~ am + wt, data = mtcars)
print(summary(car_mod), digits = 5)

mean_wt &lt;- mean(mtcars$wt)

manual_trans &lt;- list(am = 0, wt = mean_wt)
auto_trans &lt;- list(am = 1, wt = mean_wt)
print(contrast(car_mod, manual_trans, auto_trans), digits = 5)
</code></pre>

<hr>
<h2 id='print.contrast'>Print a Contrast Object</h2><span id='topic+print.contrast'></span>

<h3>Description</h3>

<p>Print a Contrast Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'contrast'
print(x, X = FALSE, fun = function(u) u, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.contrast_+3A_x">x</code></td>
<td>
<p>Result of <code>contrast()</code>.</p>
</td></tr>
<tr><td><code id="print.contrast_+3A_x">X</code></td>
<td>
<p>A logical: set <code>TRUE</code> to print design matrix used in computing the
contrasts (or the average contrast).</p>
</td></tr>
<tr><td><code id="print.contrast_+3A_fun">fun</code></td>
<td>
<p>A function to transform the contrast, SE, and lower and upper
confidence limits before printing.  For example, specify <code>fun = exp</code> to
anti-log them for logistic models.</p>
</td></tr>
<tr><td><code id="print.contrast_+3A_...">...</code></td>
<td>
<p>Not used.</p>
</td></tr>
</table>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+contrast'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>rms</dt><dd><p><code><a href="rms.html#topic+contrast">contrast</a></code></p>
</dd>
</dl>

<hr>
<h2 id='two_factor_crossed'>Complete Two-Factor Experiment</h2><span id='topic+two_factor_crossed'></span>

<h3>Description</h3>

<p>Complete Two-Factor Experiment
</p>


<h3>Details</h3>

<p>A gene expression experiment was run to assess the effect of a compound
under two different diets: high fat and low fat. The main comparisons of
interest are the difference between the treated and untreated groups within
a diet. The interaction effect was a secondary hypothesis. For illustration,
we only include the expression value of one of the genes. The study design
was a full two-way factorial with <code>n = 24</code> samples.
</p>


<h3>Value</h3>

<table>
<tr><td><code>two_factor_crossed</code></td>
<td>
<p>A data frame</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>two_factor_crossed
</code></pre>

<hr>
<h2 id='two_factor_incompl'>Incomplete Two-Factor Experiment with Repeated Measurments</h2><span id='topic+two_factor_incompl'></span>

<h3>Description</h3>

<p>Incomplete Two-Factor Experiment with Repeated Measurments
</p>


<h3>Details</h3>

<p>In a gene expression experiment, stem cells were differentiated using a set
of factors (such as media types, cell spreads etc.). These factors were
collapsed into a single cell environment configurations variable. The cell
lines were assays over three days. Two of the configurations were only run
on the first day and the other two were assays at baseline.
</p>
<p>To get the materials, three donors provided materials. These donors
provided (almost) equal replication across the two experimental factors (day
and configuration).
</p>
<p>One of the goals of this experiment was to assess pre-specified
differences in the configuration at each time point. For example, the
differences between configurations <code>A</code> and <code>B</code> at day one is of interest.
Also, the differences between configurations <code>C</code> and <code>D</code> at each time points
were important.
</p>
<p>Since there are missing cells in the design, it is not a complete two-way
factorial. One way to analyze this experiment is to further collapse the
time and configuration data into a single variable and then specify each
comparison using this factor.
</p>


<h3>Value</h3>

<table>
<tr><td><code>two_factor_incompl</code></td>
<td>
<p>A data frame</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>two_factor_incompl
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
