<!DOCTYPE html><html><head><title>Help for package convertid</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {convertid}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.cache.writable'><p>Unexported functions</p>
Test if a path exists and is writable</a></li>
<li><a href='#.create.cache'><p>Unexported functions</p>
Create a file cache directory at a given location.</a></li>
<li><a href='#.get.Ensembl_config'><p>Unexported functions</p>
Test and retrieve Ensembl-specific CURL SSL configuration.</a></li>
<li><a href='#.get.httr_config'><p>Unexported functions</p>
Get httr configuration, i.e., current CURL options for data fetching functions.</a></li>
<li><a href='#.setCacheLocation'><p>Unexported functions</p>
Set the location for the <em><code>biomaRt</code></em> cache</a></li>
<li><a href='#convert.alias'><p>Convert Symbols to Aliases and Vice Versa.</p></a></li>
<li><a href='#convert.bm'><p>Retrieve Additional Annotations from Biomart</p></a></li>
<li><a href='#convertid'><p>Convert Gene IDs Between Each Other and Fetch Annotations from Biomart</p></a></li>
<li><a href='#convertId2'><p>Convert Gene Symbols to Ensembl Gene IDs or vice versa</p></a></li>
<li><a href='#get.bm'><p>Make a Query to Biomart.</p></a></li>
<li><a href='#likely_symbol'><p>Retrieve Symbol Aliases and Previous symbols to determine a likely current symbol</p></a></li>
<li><a href='#todisp2'><p>Convenience Function to Convert Ensembl Gene IDs to Gene Symbols</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Convert Gene IDs Between Each Other and Fetch Annotations from
Biomart</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Vidal Fey [aut, cre],
  Henrik Edgren [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vidal Fey &lt;vidal.fey@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene Symbols or Ensembl Gene IDs are converted using the Bimap interface in 'AnnotationDbi' in convertId2() but
    that function is only provided as fallback mechanism for the most common use cases in data analysis. The main function
    in the package is convert.bm() which queries BioMart using the full capacity of the API provided through the
    'biomaRt' package. Presets and defaults are provided for convenience but all "marts", "filters" and "attributes"
    can be set by the user. Function convert.alias() converts Gene Symbols to Aliases and vice versa and function likely_symbol()
    attempts to determine the most likely current Gene Symbol.</td>
</tr>
<tr>
<td>Depends:</td>
<td>AnnotationDbi</td>
</tr>
<tr>
<td>Imports:</td>
<td>org.Hs.eg.db, org.Mm.eg.db, plyr, stringr, biomaRt, stats,
xml2, utils, rappdirs, assertthat, methods, httr, BiocFileCache</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-29 12:13:00 UTC; fsvife</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-29 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.cache.writable'>Unexported functions
Test if a path exists and is writable</h2><span id='topic+.cache.writable'></span>

<h3>Description</h3>

<p><code>.cache.writable()</code> uses <code>file.access()</code> to test if a
given location exists and is writable by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.cache.writable(path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".cache.writable_+3A_path">path</code></td>
<td>
<p>(<code>character</code>). The path to be tested.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if both conditions are met, FALSE if not.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+file.access">file.access</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: .cache.writable(rappdirs::user_cache_dir())
</code></pre>

<hr>
<h2 id='.create.cache'>Unexported functions
Create a file cache directory at a given location.</h2><span id='topic+.create.cache'></span>

<h3>Description</h3>

<p><code>.create.cache()</code> attempts to create a cache directory based on a given path name. Typically, such path
is specific to the package from within the function is called. The default settings refer to the file cache framework in the <em><code>biomaRt</code></em> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.create.cache(cache.path = rappdirs::user_cache_dir("biomaRt"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".create.cache_+3A_cache.path">cache.path</code></td>
<td>
<p>(<code>character</code>). The path to use for the cached files.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the location was successfully set up, FALSE if not.
</p>


<h3>See Also</h3>

<p><code><a href="rappdirs.html#topic+user_cache_dir">user_cache_dir</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: .create.cache(rappdirs::user_cache_dir("biomaRt"))
</code></pre>

<hr>
<h2 id='.get.Ensembl_config'>Unexported functions
Test and retrieve Ensembl-specific CURL SSL configuration.</h2><span id='topic+.get.Ensembl_config'></span>

<h3>Description</h3>

<p><code>.get.Ensembl_config()</code> tests and gets CURL options used with <em>&quot;^https://.*ensembl.org&quot;</em> URLs.
The function is a modified version of <code>.getEnsemblSSL</code> from the <em><code>biomaRt</code></em> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get.Ensembl_config(use.cache = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get.Ensembl_config_+3A_use.cache">use.cache</code></td>
<td>
<p>(<code>logical</code>) Should <em><code>biomaRt</code></em> functions use the cache? Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R object of class <code>request</code> listing current CURL options.
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: .get.Ensembl_config()
</code></pre>

<hr>
<h2 id='.get.httr_config'>Unexported functions
Get httr configuration, i.e., current CURL options for data fetching functions.</h2><span id='topic+.get.httr_config'></span>

<h3>Description</h3>

<p><code>.get.httr_config()</code> retrieves the current CURL options and in particular tests and gets
the options used with <em>&quot;^https://.*ensembl.org&quot;</em> URLs. The code was partly copied from <code><a href="biomaRt.html#topic+listMarts">listMarts</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.get.httr_config(
  httr_config,
  host = "https://www.ensembl.org",
  use.cache = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".get.httr_config_+3A_httr_config">httr_config</code></td>
<td>
<p>(<code>list</code>). A R object of class <code>request</code> listing current CURL options. Missing (and meant to be created) by default.</p>
</td></tr>
<tr><td><code id=".get.httr_config_+3A_host">host</code></td>
<td>
<p>(<code>character</code>) Host URL.</p>
</td></tr>
<tr><td><code id=".get.httr_config_+3A_use.cache">use.cache</code></td>
<td>
<p>(<code>logical</code>) Should <em><code>biomaRt</code></em> functions use the cache? Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R object of class <code>request</code> listing current CURL options.
</p>


<h3>See Also</h3>

<p><code><a href="httr.html#topic+config">config</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: .get.httr_config()
</code></pre>

<hr>
<h2 id='.setCacheLocation'>Unexported functions
Set the location for the <em><code>biomaRt</code></em> cache</h2><span id='topic+.setCacheLocation'></span>

<h3>Description</h3>

<p><code>.setBiomaRtCacheLocation()</code> attempts to set the cache location
used by the functions in the <em><code>biomaRt</code></em> package and defined in the BIOMART_CACHE
environment variable.
If that variable is set and the defined location exists and is writable nothing is done.
If the system default cache location exists and is writable a sub-folder <code>app</code> is used (and created if necessary).
If the above don't work a new path is constructed from <code>cache.dir</code> and the <code>app</code> folder and an attempt is made to create that.
If all of the above fail the function attempts to create <code>file.path(tempdir(), app)</code>. If tat fails, too,
an exception is thrown.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.setCacheLocation(cache.dir = rappdirs::user_cache_dir(), app = "biomaRt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".setCacheLocation_+3A_cache.dir">cache.dir</code></td>
<td>
<p>(<code>character</code>). Optional user-defined path used as cache parent directory. Defaults to <code>rappdirs::user_cache_dir()</code>. If a custom path is given that has to exist.</p>
</td></tr>
<tr><td><code id=".setCacheLocation_+3A_app">app</code></td>
<td>
<p>(<code>character</code>). Optional application-specific cache sub-directory, i.e., the actually used cache location. Defaults to &quot;biomaRt&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The value of the BIOMART_CACHE environment variable, i.e., the cache location.
</p>


<h3>See Also</h3>

<p><code><a href="rappdirs.html#topic+user_cache_dir">user_cache_dir</a></code>, <code><a href="BiocFileCache.html#topic+BiocFileCache">BiocFileCache</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: .setCacheLocation()
</code></pre>

<hr>
<h2 id='convert.alias'>Convert Symbols to Aliases and Vice Versa.</h2><span id='topic+convert.alias'></span>

<h3>Description</h3>

<p><code>convert.alias()</code> attempts to find all possible symbol-alias combinations for a given gene symbol, i.e.,
it assumes the input ID to be either an Alias or a Symbol and performs multiple queries to find all possible
counterparts. The input IDs are converted to title and upper case before querying and all possibilities are tested.
There are species presets for Human and Mouse annotations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.alias(id, species = c("Human", "Mouse"), db = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.alias_+3A_id">id</code></td>
<td>
<p>(<code>character</code>). Vector of gene symbols.</p>
</td></tr>
<tr><td><code id="convert.alias_+3A_species">species</code></td>
<td>
<p>(<code>character</code>). One of &quot;Human&quot; and &quot;Mouse&quot;. Defaults to &quot;Human&quot;.</p>
</td></tr>
<tr><td><code id="convert.alias_+3A_db">db</code></td>
<td>
<p>(<code>AnnotationDb object</code>). Annotation package object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with two columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'SYMBOL': The official gene symbol.</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'ALIAS': All possible aliases.</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>See Also</h3>

<p><code><a href="AnnotationDbi.html#topic+select">select</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>convert.alias("TRPV4")
</code></pre>

<hr>
<h2 id='convert.bm'>Retrieve Additional Annotations from Biomart</h2><span id='topic+convert.bm'></span>

<h3>Description</h3>

<p><code>convert.bm()</code> is a wrapper for <code>get.bm()</code> which in turn makes use of <code>getBM()</code> from the <em>biomaRt</em> package.
It takes a matrix or data frame with the IDs to be converted in one column or as row names as input and returns a data frame with additional
annotations after cleaning the fetched annotations and merging them with the input data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.bm(
  dat,
  id = "ID",
  biom.data.set = c("human", "mouse"),
  biom.mart = c("ensembl", "mouse", "snp", "funcgen", "plants"),
  host = "https://www.ensembl.org",
  biom.filter = "ensembl_gene_id",
  biom.attributes = c("ensembl_gene_id", "hgnc_symbol", "description"),
  biom.cache = rappdirs::user_cache_dir("biomaRt"),
  use.cache = TRUE,
  sym.col = "hgnc_symbol",
  rm.dups = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert.bm_+3A_dat">dat</code></td>
<td>
<p><code>matrix</code> or <code>data.frame</code>. Matrix or data frame with the ids to be converted in a column or as row names.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_id">id</code></td>
<td>
<p><code>character</code>. Name of the column with the ids to be converted, special name &quot;rownames&quot; will use the row names.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_biom.data.set">biom.data.set</code></td>
<td>
<p><code>character</code> of length one. Biomart data set to use.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_biom.mart">biom.mart</code></td>
<td>
<p><code>character</code> vector. Biomart to use (uses the first element of the vector), defaults to &quot;ensembl&quot;.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_host">host</code></td>
<td>
<p><code>character</code> of length one. Host URL.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_biom.filter">biom.filter</code></td>
<td>
<p><code>character</code> of length one. Name of biomart filter, i.e., type of query ids, defaults to &quot;ensembl_gene_id&quot;.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_biom.attributes">biom.attributes</code></td>
<td>
<p><code>character</code> vector. Biomart attributes, i.e., type of desired result(s); make sure query id type is included!</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_biom.cache">biom.cache</code></td>
<td>
<p><code>character</code>. Path name giving the location of the cache <code>getBM()</code> uses if <code>use.cache=TRUE</code>. Defaults to the value in the <em>BIOMART_CACHE</em> environment variable.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_use.cache">use.cache</code></td>
<td>
<p>(<code>logical</code>). Should <code>getBM()</code> use the cache? Defaults to <code>TRUE</code> as in the <code>getBM()</code> function and is passed on to that.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_sym.col">sym.col</code></td>
<td>
<p><code>character</code>. Name of the column in the query result with gene symbols.</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_rm.dups">rm.dups</code></td>
<td>
<p><code>logical</code>. Should duplicated input IDs (<span class="option">biom.filter</span>) be removed from the result?</p>
</td></tr>
<tr><td><code id="convert.bm_+3A_verbose">verbose</code></td>
<td>
<p>(<code>logical</code>). Should verbose output be written to the console? Defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Wrapped around 'get.bm'.
</p>


<h3>Value</h3>

<p>A data frame with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Vidal Fey
</p>


<h3>See Also</h3>

<p><code><a href="biomaRt.html#topic+getBM">getBM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dat &lt;- data.frame(ID=c("ENSG00000111199", "ENSG00000134121", "ENSG00000176102", "ENSG00000171611"))
bm &lt;- convert.bm(dat)
bm

## End(Not run)
</code></pre>

<hr>
<h2 id='convertid'>Convert Gene IDs Between Each Other and Fetch Annotations from Biomart</h2><span id='topic+convertid'></span>

<h3>Description</h3>

<p>Gene Symbols or Ensembl Gene IDs are converted using the Bimap interface in 'AnnotationDbi' in convertId2() but
that function is only provided as fallback mechanism for the most common use cases in data analysis. The main function
in the package is convert.bm() which queries Biomart using the full capacity of the API provided through the
'biomaRt' package. Presets and defaults are provided for convenience but all &quot;marts&quot;, &quot;filters&quot; and &quot;attributes&quot;
can be set by the user. Function convert.alias() converts Gene Symbols to Aliases and vice versa and function
likely_symbol() attempts to determine the most likely current Gene Symbol.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> convertid</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Initial version: </td><td style="text-align: left;"> 0.1-0</td>
</tr>
<tr>
 <td style="text-align: left;">
Created: </td><td style="text-align: left;"> 2021-08-18</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Vidal Fey &lt;vidal.fey@gmail.com&gt;
Maintainer: Vidal Fey &lt;vidal.fey@gmail.com&gt;
</p>

<hr>
<h2 id='convertId2'>Convert Gene Symbols to Ensembl Gene IDs or vice versa</h2><span id='topic+convertId2'></span>

<h3>Description</h3>

<p><code>convertId2()</code> uses the Bimap interface in AnnotationDbi to extract information from
annotation packages. The function is limited to Human and Mouse annotations and is provided only as
fallback mechanism for the most common use cases in data analysis. Please use the Biomart interface
function <code>convert.bm()</code> for more flexibility.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertId2(id, species = c("Human", "Mouse"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertId2_+3A_id">id</code></td>
<td>
<p>(<code>character</code>). Vector of gene symbols.</p>
</td></tr>
<tr><td><code id="convertId2_+3A_species">species</code></td>
<td>
<p>(<code>character</code>). One of &quot;Human&quot; and &quot;Mouse&quot;. Defaults to &quot;Human&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector where the input IDs are the names and the query results the values.
</p>


<h3>See Also</h3>

<p><code><a href="AnnotationDbi.html#topic+Bimap-envirAPI">Bimap-envirAPI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>convertId2("ENSG00000111199")
convertId2("TRPV4")
</code></pre>

<hr>
<h2 id='get.bm'>Make a Query to Biomart.</h2><span id='topic+get.bm'></span>

<h3>Description</h3>

<p><code>get.bm()</code> is a user-friendly wrapper for <code>getBM()</code> from the <em>biomaRt</em> package with default
settings for Human and Mouse.
It sets all needed variables and performs the query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.bm(
  values,
  biom.data.set = c("human", "mouse"),
  biom.mart = c("ensembl", "mouse", "snp", "funcgen", "plants"),
  host = "https://www.ensembl.org",
  biom.filter = "ensembl_gene_id",
  biom.attributes = c("ensembl_gene_id", "hgnc_symbol", "description"),
  biom.cache = rappdirs::user_cache_dir("biomaRt"),
  use.cache = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.bm_+3A_values">values</code></td>
<td>
<p><code>character</code> vector of ids to be converted.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_biom.data.set">biom.data.set</code></td>
<td>
<p><code>character</code> of length one. Biomart data set to use. Defaults to 'human' (internally translated to &quot;hsapiens_gene_ensembl&quot; if <code>biom.mart="ensembl"</code>).</p>
</td></tr>
<tr><td><code id="get.bm_+3A_biom.mart">biom.mart</code></td>
<td>
<p><code>character</code> vector. Biomart to use (uses the first element of the vector), defaults to &quot;ensembl&quot;.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_host">host</code></td>
<td>
<p><code>character</code> of length one. Host URL.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_biom.filter">biom.filter</code></td>
<td>
<p><code>character</code> of length one. Name of biomart filter, i.e., type of query ids, defaults to &quot;ensembl_gene_id&quot;.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_biom.attributes">biom.attributes</code></td>
<td>
<p><code>character</code> vector. Biomart attributes, i.e., type of desired result(s); make sure query id type is included!</p>
</td></tr>
<tr><td><code id="get.bm_+3A_biom.cache">biom.cache</code></td>
<td>
<p><code>character</code>. Path name giving the location of the cache <code>getBM()</code> uses if <code>use.cache=TRUE</code>. Defaults to the value in the <em>BIOMART_CACHE</em> environment variable.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_use.cache">use.cache</code></td>
<td>
<p>(<code>logical</code>). Should <code>getBM()</code> use the cache? Defaults to <code>TRUE</code> as in the <code>getBM()</code> function and is passed on to that.</p>
</td></tr>
<tr><td><code id="get.bm_+3A_verbose">verbose</code></td>
<td>
<p>(<code>logical</code>). Should verbose output be written to the console? Defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Vidal Fey
</p>


<h3>See Also</h3>

<p><code><a href="biomaRt.html#topic+getBM">getBM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
val &lt;- c("ENSG00000111199", "ENSG00000134121", "ENSG00000176102", "ENSG00000171611")
bm &lt;- get.bm(val)
bm

## End(Not run)
</code></pre>

<hr>
<h2 id='likely_symbol'>Retrieve Symbol Aliases and Previous symbols to determine a likely current symbol</h2><span id='topic+likely_symbol'></span>

<h3>Description</h3>

<p><code>likely_symbol()</code> downloads the latest version of the HGNC gene symbol database as a text
file and query it to obtain symbol aliases, previous symbols and all symbols currently in use. (Optionally)
assuming the input ID to be either an Alias or a Symbol or a Previous Symbol it performs multiple queries and
compares the results of all possible combinations to determine a likely current Symbol.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>likely_symbol(
  syms,
  alias_sym = TRUE,
  prev_sym = TRUE,
  orgnsm = "human",
  hgnc = NULL,
  hgnc_url = NULL,
  output = c("likely", "symbols", "all"),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="likely_symbol_+3A_syms">syms</code></td>
<td>
<p>(<code>character</code>). Vector of Gene Symbols to be tested.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_alias_sym">alias_sym</code></td>
<td>
<p>(<code>logical</code>). Should the input be assumed to be an Alias? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_prev_sym">prev_sym</code></td>
<td>
<p>(<code>logical</code>). Should the input be assumed to be a Previous Symbol? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_orgnsm">orgnsm</code></td>
<td>
<p>(<code>character</code>). The organism for which the Symbols are tested.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_hgnc">hgnc</code></td>
<td>
<p>(<code>data.frame</code>). An optional data frame with the needed HGNC annotations. (Needs to match the format
available at <code>hgnc_utl</code>!)</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_hgnc_url">hgnc_url</code></td>
<td>
<p>(<code>character</code>). URL where to download the HGNC annotation dataset. Defaults to
<code>"ftp://ftp.ebi.ac.uk/pub/databases/genenames/new/tsv/hgnc_complete_set.txt"</code>.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_output">output</code></td>
<td>
<p>(<code>character</code>). One of &quot;likely&quot;, &quot;symbols&quot; and &quot;all&quot;. Determines the scope of the output data frame.
Defaults to <code>"likely"</code> which will return the inout Symbol and the determined likely Symbol.</p>
</td></tr>
<tr><td><code id="likely_symbol_+3A_verbose">verbose</code></td>
<td>
<p>(<code>logical</code>). Should messages be written to the console? Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note that the algorithm is very slow for large input vectors.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with the following columns depending on the <code>output</code> setting.
<strong><code>output="likely"</code></strong>:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'likely_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'input_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong><code>output="symbols"</code></strong>:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'current_symbols'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'likely_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'input_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'all_symbols'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong><code>output="all"</code></strong>:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'orig_input'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'organism'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'current_symbols'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'likely_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'input_symbol'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> 'all_symbols'</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p>Only fully implemented for Human for now.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
likely_symbol(c("ABCC4", "ACPP", "KIAA1524"))

## End(Not run)
</code></pre>

<hr>
<h2 id='todisp2'>Convenience Function to Convert Ensembl Gene IDs to Gene Symbols</h2><span id='topic+todisp2'></span>

<h3>Description</h3>

<p><code>todisp2()</code> uses Biomart by employing <code>get.bm()</code> to retrieve Gene Symbols for a set of Ensembl
Gene IDs. It is mainly meant as a fast way to convert IDs in standard gene expression analysis output to Symbols,
e.g., for visualisation, which is why the input ID type is hard coded to ENSG IDs. If Biomart is not available
the function can fall back to use <code>convertId2()</code> or a user-provided data frame with corresponding ENSG IDs and
Symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>todisp2(ensg, lab = NULL, biomart = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="todisp2_+3A_ensg">ensg</code></td>
<td>
<p>(<code>character</code>). Vector of Ensemble Gene IDs. Other ID types are not yet supported.</p>
</td></tr>
<tr><td><code id="todisp2_+3A_lab">lab</code></td>
<td>
<p>(<code>data.frame</code>). A data frame with Ensembl Gene IDs as row names and Gene Symbols in the only column.</p>
</td></tr>
<tr><td><code id="todisp2_+3A_biomart">biomart</code></td>
<td>
<p>(<code>logical</code>). Should Biomart be used? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="todisp2_+3A_verbose">verbose</code></td>
<td>
<p>(<code>logical</code>). Should verbose output be written to the console? Defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of Gene Symbols.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get.bm">get.bm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
val &lt;- c("ENSG00000111199", "ENSG00000134121", "ENSG00000176102", "ENSG00000171611")
sym &lt;- todisp2(val)
sym

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
