<!DOCTYPE html><html><head><title>Help for package convevol</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {convevol}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allmaxdist'><p>Calculates maxdist for all pairs of taxa in a phylogeny.</p></a></li>
<li><a href='#ancestrallineages'><p>Extracts lineages leading to two tips, t1 and t2, from their most recent common ancestor.</p></a></li>
<li><a href='#calcchanges'><p>Calculates all phenotypic changes that occur on all branches of a phylogeny.</p></a></li>
<li><a href='#calcConv'><p>Quantifies convergent evolution by the ratio of the current to maximum past phenotypic distance between two or</p>
more lineages, as described in Stayton (2015). Code written by Jonathan S. Mitchell for Zelditch et al. (2017).</a></li>
<li><a href='#calcConvCt'><p>Computes Ct-metric scores for putatively convergent tips (or groups of tips) given a set of user</p>
provided phenotypic characters and a time calibrated phylogeny.</a></li>
<li><a href='#calcCs'><p>Calculates the C1-C4 measures of convergent evolution between two lineages as described in</p>
Stayton (2015). All measures quantify convergence by the ratio of current to maximum past phenotypic
distance between lineages. Can be used as-is but more often will be used within the calcConv script.
Code written by Jonathan S. Mitchell for Zelditch et al. (2017)</a></li>
<li><a href='#calcCsCt'><p>Computes Ct values for a pair of tips. Internal, called in calcConv.</p></a></li>
<li><a href='#convnum'><p>Quantify convergence by the number of convergent events</p></a></li>
<li><a href='#convnumsig'><p>Assess the significance of convergent evolution using simulations and the convnum metric</p></a></li>
<li><a href='#convrat'><p>Quantifies convergent evolution using the C1, C2, C3, and C4 measures as described by Stayton (2015).</p></a></li>
<li><a href='#convratsig'><p>Tests the signifiance of convergent evolution by the ratio of the current to maximum past phenotypic distance</p></a></li>
<li><a href='#convSig'><p>Uses simulations to assess the significance of C1-C4 measures of convergent</p>
evolution as described in Stayton (2015). Code written by Jonathan S.
Mitchell for Zelditch et al. (2017)</a></li>
<li><a href='#convSigCt'><p>Computes and conducts significance tests on Ct-metric scores for putatively convergent tips (or groups of tips) given a set of user provided phenotypic characters and a time calibrated phylogeny.</p></a></li>
<li><a href='#findanc'><p>Find the ancestor of a given node in a phylogeny</p></a></li>
<li><a href='#labelstonumbers'><p>Converts taxon names to tip/edge numbers</p></a></li>
<li><a href='#maxdist'><p>Calculates the maximum phenotypic distance between the lineages leading to a pair of taxa.</p></a></li>
<li><a href='#multianc'><p>Reconstructs ancestral states for multiple characters</p></a></li>
<li><a href='#phy'><p>Sample phylogeny for use in convevol examples and tutorial</p></a></li>
<li><a href='#plot2D'><p>Plots calcConv or convSig output as a two-dimensional time series.</p></a></li>
<li><a href='#plotCt'><p>Plots calcConv or convSig output.</p></a></li>
<li><a href='#plotellipse'><p>Plots an ellipse</p></a></li>
<li><a href='#pullNodeSeq'><p>Extracts a vector of ancestors for a given taxon. Code written by Jonathan S. Mitchell for</p>
Zelditch et al. (2017)</a></li>
<li><a href='#pwCheck'><p>Calculates the number of measurements that can be made between two lineages for each pairwise comparison within a set of putatively convergent tips (group identity may also be taken into account). Useful for determining which comparisons are not informative, and constructing a group object before running calcConvCt or convSigCt.</p></a></li>
<li><a href='#trait'><p>Sample data for use in convevol examples and tutorial</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysis of Convergent Evolution</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Quantifies and assesses the significance of convergent evolution using multiple methods and measures as described in Stayton (2015) &lt;<a href="https://doi.org/10.1111%2Fevo.12729">doi:10.1111/evo.12729</a>&gt; and Grossnickle et al. 2023.  Also displays results in various ways.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), ape, cluster, geiger, magick, phytools</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-08 13:02:12 UTC; tstayton</td>
</tr>
<tr>
<td>Author:</td>
<td>William Brightly [aut],
  C. Tristan Stayton [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>C. Tristan Stayton &lt;tstayton@bucknell.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-08 13:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='allmaxdist'>Calculates maxdist for all pairs of taxa in a phylogeny.</h2><span id='topic+allmaxdist'></span>

<h3>Description</h3>

<p>allmaxdist Uses maxdist to calcualte the maximum phenotypic.  distance between the ancestors of all pairs of taxa in a phylogeny.  By default outputs these as a matrix, but can also output a list.  Can take some time to run for large trees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allmaxdist(phyl, phendata, mat = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allmaxdist_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="allmaxdist_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="allmaxdist_+3A_mat">mat</code></td>
<td>
<p>Whether or not to export the values in a matrix (default) or a list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Regarding the output:  the matrix is better organized, but the list avoids all the zeroes and is probably better for making histograms.
</p>


<h3>Value</h3>

<p>A matrix or list of all maxdist values for all pairs of taxa in the phylogeny.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=2)
answer&lt;-allmaxdist(phyl,phendata,mat=TRUE)
</code></pre>

<hr>
<h2 id='ancestrallineages'>Extracts lineages leading to two tips, t1 and t2, from their most recent common ancestor.</h2><span id='topic+ancestrallineages'></span>

<h3>Description</h3>

<p>Extracts lineages leading to two tips, t1 and t2, from their most recent common ancestor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancestrallineages(phyl, phendata, t1, t2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancestrallineages_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="ancestrallineages_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="ancestrallineages_+3A_t1">t1</code></td>
<td>
<p>The first tip of interest</p>
</td></tr>
<tr><td><code id="ancestrallineages_+3A_t2">t2</code></td>
<td>
<p>The second tip of interest</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>A list containing two matrices.  Each matrix corresponds to a tip.  The matrix
consists of reconstructed ancestral values for all nodes leading from the mrca of both
tips to the tip.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>
<p>Stayton, C.T.  (2015).  The definition, recognition, and interpretation of
convergent evolution, and two new measure for quantifying and assessing the
significance of convergence.  Evolution 69:2140-2153.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=2)
answer&lt;-ancestrallineages(phyl,phendata,"t1","t2")
</code></pre>

<hr>
<h2 id='calcchanges'>Calculates all phenotypic changes that occur on all branches of a phylogeny.</h2><span id='topic+calcchanges'></span>

<h3>Description</h3>

<p>Calculates the Euclidean distance between all ancestors and descendants on a phylogeny to reconstruct the phenotypic changes that occur along all edges of a phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcchanges(phyl, phendata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcchanges_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="calcchanges_+3A_phendata">phendata</code></td>
<td>
<p>A matrix of phenotypic data for all tips, with taxa in rows and characters in columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the Euclidean distance between all ancestors and descendants on a phylogeny to reconstruct the phenotypic changes that occur along all edges of a phylogeny.
</p>


<h3>Value</h3>

<p>A vector in which each element represents an edge of the phylogeny, and the values are the magnitudes of evolutionary change that occur along those edges.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=5)
changes&lt;-calcchanges(phyl,phendata)
</code></pre>

<hr>
<h2 id='calcConv'>Quantifies convergent evolution by the ratio of the current to maximum past phenotypic distance between two or
more lineages, as described in Stayton (2015). Code written by Jonathan S. Mitchell for Zelditch et al. (2017).</h2><span id='topic+calcConv'></span>

<h3>Description</h3>

<p>calcConv prepares arguements for the CalcCs function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcConv(phy, traits, focaltaxa, anc = NULL, VERBOSE = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcConv_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="calcConv_+3A_traits">traits</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="calcConv_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>A list consisting of the names of all putatively convergent taxa</p>
</td></tr>
<tr><td><code id="calcConv_+3A_anc">anc</code></td>
<td>
<p>A matrix of user supplied ancestral trait values at internal nodes (formatted as &quot;traits&quot; but with node number as rownames)</p>
</td></tr>
<tr><td><code id="calcConv_+3A_verbose">VERBOSE</code></td>
<td>
<p>Whether or not to print progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calcConv is a wrapper function which formats data, performs ancestral state reconstructions, obtains
distance matrices, and determines pairwise combinations of focal taxa, which are then used as arguements for
the CalcCs function, which calculates values for C1-C4 of Stayton (2015) for each pair of putatively
convergent taxa.
</p>


<h3>Value</h3>

<p>C1-C4 convergence measures for all pairs of putatively convergent taxa.
</p>


<h3>References</h3>

<p>Stayton, C.T. 2015. The definition, recognition, and interpretation of convergent evolution, and
two new measures for quantifying and assessing the significance of convergence. Evolution 69:2140-2453.
</p>
<p>Zelditch, M.L., J. Ye, J.S. Mitchell, and D.L. Swiderski. 2017. Rare ecomorphological
convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in
squirrels (Sciuridae). Evolution 71:633-649.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phy&lt;-rtree(100)
traits&lt;-fastBM(phy,nsim=3)
focaltaxa&lt;-c("t1","t50","t100")
answer&lt;-calcConv(phy,traits,focaltaxa,anc=NULL,VERBOSE=FALSE)
</code></pre>

<hr>
<h2 id='calcConvCt'>Computes Ct-metric scores for putatively convergent tips (or groups of tips) given a set of user
provided phenotypic characters and a time calibrated phylogeny.</h2><span id='topic+calcConvCt'></span>

<h3>Description</h3>

<p>calcConvCt Computes Ct-metric scores for putatively convergent tips (or groups of tips)
given a set of user provided phenotypic characters and a time calibrated phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcConvCt(
  phy,
  traits,
  focaltaxa,
  groups = NULL,
  ace = NULL,
  conservative = FALSE,
  VERBOSE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcConvCt_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_traits">traits</code></td>
<td>
<p>A matrix of numeric phenotypic traits with rownames matching tip labels of phy</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>A vector of tip labels for the putatively convergent taxa to be compared</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_groups">groups</code></td>
<td>
<p>An optional vector of groups with names matching focaltaxa. Indicates the group identity of all putatively convergent taxa and limits Ct measures to intergroup comparisons only</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_ace">ace</code></td>
<td>
<p>A matrix of user supplied ancestral trait values at internal nodes (formatted as &quot;traits&quot; but with node number as rownames)</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_conservative">conservative</code></td>
<td>
<p>Logical value indicating whether Dmax.t should be restricted to occurr before the oldest stem lineage of the two groups involved in each pairwise comparison. Stem lineage age for each group is defined as the height of the parent node of the branch subtending the most recent common ancestor of tips within a group. Where groups include a single tip, the parent node of the tip's subtending branch is used. Requires group object to be provided by user.</p>
</td></tr>
<tr><td><code id="calcConvCt_+3A_verbose">VERBOSE</code></td>
<td>
<p>Logical value indicating whether model information should be printed during computation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function incorporates the optimizations introduced by Zelditch et al. (2017), which significantly improve runtimes
</p>
<p>Reconstructions part way along branches are obtained using equation 2 of Felsenstein (1985), following code modified from the phytools (Revell, 2012) function contMap
</p>


<h3>Value</h3>

<p>A list of the following components:
</p>
<p>mean a named vector of Ct-metrics averaged from all pairwise comparisons of focaltaxa. If user provided groups, this is based only on comparisons between taxa belonging to different groups.
</p>
<p>Cmat a matrix of Ct-metrics for each pairwise comparison.
</p>
<p>path_df a list of dataframes, one per pairwise comparison of the focal taxa, each containing information from all timepoint measurements of the two putatively convergent lineages. These provide the nodes at which comparisons were drawn, the evolutionary path along which that node fell (i.e., leading to one of two tips), the node height, reconstructed ancestral states at that node for each phenotypic trait, reconstructed ancestral values for each trait along the opposite path, and the phenotypic distance between the two lineages at that point.
</p>
<p>grp.mean a matrix of Ct-metrics summarized for inter-group comparisons, returned only if user defined groups were specified. Provides overall results matching those reported in &quot;mean&quot;, results for each unique inter-group comparison, and results averaged with equal weight given to each unique inter-group comparison (i.e., ignoring differences in the number of tips in each group).
</p>
<p>limits a list of tree heights used to constrain Dmax.t calculations for each pairwise comparison in conservative analyses. Only returned if conservative == TRUE.
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(phytools)
library(geiger)

# create time calibrated tree
mytree&lt;-rtree(100)
mycalibration &lt;- makeChronosCalib(mytree, node="root", age.max=50)
phy &lt;- chronos(mytree, lambda = 1, model = "correlated", calibration = mycalibration, 
control = chronos.control() )
class(phy)&lt;-"phylo"

# create three normally distributed phenotypic traits
traits &lt;- cbind(rnorm(Ntip(phy)),rnorm(Ntip(phy)),rnorm(Ntip(phy)))
rownames(traits) &lt;- phy$tip.label

#	select two random tips, excluding sister taxa
pairs &lt;- apply(combn(phy$tip.label,2),2,function(x) nodepath(phy,which(phy$tip.label == x[1]),
which(phy$tip.label == x[2])))
nosis &lt;- combn(phy$tip.label,2)[,unlist(lapply(pairs, function(x) length(x) &gt; 3))]
focaltaxa &lt;- nosis[,sample(1:ncol(nosis),1)]

system.time(run &lt;- calcConvCt(phy, traits, focaltaxa))
system.time(run2 &lt;- convSigCt(phy, traits, focaltaxa, nsim=10))

plotCt(output = run2,phy = phy,focaltaxa = focaltaxa)

## End(Not run)
</code></pre>

<hr>
<h2 id='calcCs'>Calculates the C1-C4 measures of convergent evolution between two lineages as described in
Stayton (2015). All measures quantify convergence by the ratio of current to maximum past phenotypic
distance between lineages. Can be used as-is but more often will be used within the calcConv script.
Code written by Jonathan S. Mitchell for Zelditch et al. (2017)</h2><span id='topic+calcCs'></span>

<h3>Description</h3>

<p>calcCs calculates the C1-C4 measures of convergent evolution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcCs(tips, ancList, allDists, phy, VERBOSE = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcCs_+3A_tips">tips</code></td>
<td>
<p>Two putatively convergent tips</p>
</td></tr>
<tr><td><code id="calcCs_+3A_anclist">ancList</code></td>
<td>
<p>A list of ancestors of all tips. Most often obtained from calcConv</p>
</td></tr>
<tr><td><code id="calcCs_+3A_alldists">allDists</code></td>
<td>
<p>A matrix of phenotypic distances between all nodes (tips and ancestors). Most often
obtained from calcConv</p>
</td></tr>
<tr><td><code id="calcCs_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest</p>
</td></tr>
<tr><td><code id="calcCs_+3A_verbose">VERBOSE</code></td>
<td>
<p>Whether or not to print progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calcCs calculates values of C1-C4, all of which are fundamentally based on comparing the
current phenotypic distance between two tips to the maximum past distances between the ancestors of
those tips. Higher values indicate a greater amount of past phenotypic distance which has been &quot;closed&quot;
by subsequent evolution, and thus greater convergence. C1 is the ratio of tip to maximum ancestral
distance. C2 is the difference of those two values. C3 scales C2 by the total amount of evolution that
has occured in the two lineages. C4 scales C2 by the total amount of evolution that has occurred in the
entire phylogeny. The arguments for this function will usually be obtained from the calcConv script in
convevol - this allows certain computationally-intensive steps (e.g., calculating ancestral states) to
only be performed once, thus saving a great deal of time. This script also corrects an error in the
calculation of C4 of previous versions of convevol.
</p>


<h3>Value</h3>

<p>C1-C4 convergence measures for all pairs of putatively convergent taxa.
</p>


<h3>References</h3>

<p>Stayton, C.T. 2015. The definition, recognition, and interpretation of convergent evolution, and
two new measures for quantifying and assessing the significance of convergence. Evolution 69:2140-2453.
</p>
<p>Zelditch, M.L., J. Ye, J.S. Mitchell, and D.L. Swiderski. 2017. Rare ecomorphological
convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in
squirrels (Sciuridae). Evolution 71:633-649.
</p>

<hr>
<h2 id='calcCsCt'>Computes Ct values for a pair of tips. Internal, called in calcConv.</h2><span id='topic+calcCsCt'></span>

<h3>Description</h3>

<p>calcCsCt Computes Ct values for a pair of tips. Internal, called in calcConv.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcCsCt(
  tips,
  ancList,
  allDists,
  phy,
  VERBOSE = FALSE,
  allVals,
  edge,
  lim.height = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcCsCt_+3A_tips">tips</code></td>
<td>
<p>vector of two tips</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_anclist">ancList</code></td>
<td>
<p>list of node paths for all tips in the user provided phylogeny</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_alldists">allDists</code></td>
<td>
<p>matrix of phenotypic distances between all exterior and interior node pairs</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_verbose">VERBOSE</code></td>
<td>
<p>logical value indicating whether model information should be printed during computation</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_allvals">allVals</code></td>
<td>
<p>a matrix of observed and reconstructed phenotypes for all user supplied traits at interior and exterior nodes</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_edge">edge</code></td>
<td>
<p>a list of data frames, each including the edge matrix of user supplied phylogeny, along with node heights and reconstructed phenotype for each of the user supplied traits</p>
</td></tr>
<tr><td><code id="calcCsCt_+3A_lim.height">lim.height</code></td>
<td>
<p>an optional tree height used to limit Dmax.t, passed only if groups are defined and a conservative test is run (see calcConv)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function incorporates the optimizations introduced by Zelditch et al. (2017), which significantly improve runtimes
</p>
<p>Reconstructions part way along branches are obtained using equation 2 of Felsenstein (1985), following code modified from the phytools (Revell, 2012) function contMap
</p>


<h3>Value</h3>

<p>A list of the Ct values
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>

<hr>
<h2 id='convnum'>Quantify convergence by the number of convergent events</h2><span id='topic+convnum'></span>

<h3>Description</h3>

<p>This program takes in a set of taxa that are already suspected to be convergent in a particular area of  morphospace.  It then counts the number of times that a lineage has invaded that region of morphospace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convnum(
  phy,
  phendata,
  convtips,
  plot = TRUE,
  ellipse = NULL,
  plotellipse = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convnum_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convnum_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="convnum_+3A_convtips">convtips</code></td>
<td>
<p>A list consisting of the names of all convergent taxa</p>
</td></tr>
<tr><td><code id="convnum_+3A_plot">plot</code></td>
<td>
<p>Whether or not to plot a phylomorphospace with lineages that cross into the region of interest highlighted as red arrows.  Default=TRUE</p>
</td></tr>
<tr><td><code id="convnum_+3A_ellipse">ellipse</code></td>
<td>
<p>Optional.  An ellipse defining the region of interest, into which groups may or may not converge.</p>
</td></tr>
<tr><td><code id="convnum_+3A_plotellipse">plotellipse</code></td>
<td>
<p>Optional.  The ellipse defining the region of interest in the first two dimensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will construct an ellipse around all convergent taxa.  Then it will reconstruct ancestral states throughout the phylogeny, and use those to determine how many lineages have crossed into this ellipse from the outside.
</p>


<h3>Value</h3>

<p>The number of lineages that have crossed into the region of trait space occupied by the convergent taxa.
</p>


<h3>References</h3>

<p>Maechler, M., Rousseeuw, P., Struyf, A., Hubert, M., Hornik, K.(2013).
cluster: Cluster Analysis Basics and Extensions. R package version 1.14.4.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phy&lt;-rtree(10)
phendata&lt;-fastBM(phy,nsim=2)
convtips&lt;-c("t1","t2","t3")
answer&lt;-convnum(phy,phendata,convtips,plot=TRUE,ellipse=NULL,plotellipse=NULL)
</code></pre>

<hr>
<h2 id='convnumsig'>Assess the significance of convergent evolution using simulations and the convnum metric</h2><span id='topic+convnumsig'></span>

<h3>Description</h3>

<p>Simulates evolution along a given phylogeny, using parameters derived from observed data, and calculates the convnum metric for each simulation for a set of user-defined taxa.  Then compares the observed convnum value to the simulated values to assess the significance of the observed levels of convergent evolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convnumsig(
  phy,
  phendata,
  convtips,
  nsim,
  ellipse = NULL,
  plot = FALSE,
  plotellipse = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convnumsig_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_convtips">convtips</code></td>
<td>
<p>A list consisting of the names of all convergent taxa</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulatons to conduct</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_ellipse">ellipse</code></td>
<td>
<p>Optional.  An ellipse defining the region of interest, into which groups may or may not converge.</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_plot">plot</code></td>
<td>
<p>Optional.  Describes whether or not to show phylomorphospaces for all of the simulations.</p>
</td></tr>
<tr><td><code id="convnumsig_+3A_plotellipse">plotellipse</code></td>
<td>
<p>Optional.  The ellipse defining the region of interest in the first two dimensions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>A list, consisting first of the p-value for the observed convnum, and second of a vector containing all of the simulated convnum values.  Also displays a histogram of all of the simulated convnum values.
</p>


<h3>References</h3>

<p>Maechler, M., Rousseeuw, P., Struyf, A., Hubert, M., Hornik, K.(2013).
cluster: Cluster Analysis Basics and Extensions. R package version 1.14.4.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phy&lt;-rtree(10)
phendata&lt;-fastBM(phy,nsim=2)
convtips&lt;-c("t1","t2","t3")
answer&lt;-convnumsig(phy,phendata,convtips,10,plot=FALSE,ellipse=NULL,plotellipse=NULL)
</code></pre>

<hr>
<h2 id='convrat'>Quantifies convergent evolution using the C1, C2, C3, and C4 measures as described by Stayton (2015).</h2><span id='topic+convrat'></span>

<h3>Description</h3>

<p>Calculates the current phenotypic distance (Euclidean) between two taxa.  Then uses ancestral state reconstruction under a BM model to calculate the maximum phenotypic distance at any time between lineages leading from the most recent common ancestor of those two taxa to the tips.  Also calculate the total amount of phenotypic evolution in the clade defined by the most recent common ancestor of those lineages, and the total amount of phenotypic evolution in the input tree.  These quantities are used to calcualte C1-C4:  C1 = 1-(current distance / maximum ancestral distance); C2 = maximum ancestral distance - current distance; C3=C2/(total phenotypic evolution in the clade defined by the two taxa); C4 = C2/(total amount of phenotypic evolution in the entire tree).  If more than two convergent taxa are input, then C1-C4 are calculated for all possible pairs of taxa, and averaged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convrat(phyl, phendata, convtips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convrat_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convrat_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="convrat_+3A_convtips">convtips</code></td>
<td>
<p>A list consisting of the names of all convergent taxa</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>Four numbers - C1, C2, C3, C4.
</p>


<h3>References</h3>

<p>Maechler, M., Rousseeuw, P., Struyf, A., Hubert, M., Hornik, K.(2013).
cluster: Cluster Analysis Basics and Extensions. R package version 1.14.4.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>
<p>Stayton, C.T.  (2015).  The definition, recognition, and interpretation of
convergent evolution, and two new measure for quantifying and assessing the
significance of convergence.  Evolution 69:2140-2153.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=2)
convtips&lt;-c("t1","t2","t3")
answer&lt;-convrat(phyl,phendata,convtips)
</code></pre>

<hr>
<h2 id='convratsig'>Tests the signifiance of convergent evolution by the ratio of the current to maximum past phenotypic distance</h2><span id='topic+convratsig'></span>

<h3>Description</h3>

<p>convratsig tests the significance of convergence (as quantified by convrat) using evolutionary simulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convratsig(phyl, phendata, convtips, nsim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convratsig_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convratsig_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="convratsig_+3A_convtips">convtips</code></td>
<td>
<p>A list consisting of the names of all convergent taxa</p>
</td></tr>
<tr><td><code id="convratsig_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulations to conduct</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function simulates evolution via Brownian motion using the input tree and parameters derived from the observed data.  It calculates a convergence metric for each simulation and calculates statistics from the number of times the simulated value exceeds the observed value.
</p>


<h3>Value</h3>

<p>The convergence metric of interest (C1, C2, etc...), a cutoff value (the value that the observed measure would have to exceed in order to be considered significant), a P-value for the statistic, and all simulated values.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=2)
convtips&lt;-c("t1","t2","t3")
answer&lt;-convratsig(phyl,phendata,convtips,10)
</code></pre>

<hr>
<h2 id='convSig'>Uses simulations to assess the significance of C1-C4 measures of convergent
evolution as described in Stayton (2015). Code written by Jonathan S.
Mitchell for Zelditch et al. (2017)</h2><span id='topic+convSig'></span>

<h3>Description</h3>

<p>convSig  calculates the significance of measures of convergent evolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convSig(phy, traits, focaltaxa, user.ace = NULL, nsim = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convSig_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convSig_+3A_traits">traits</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="convSig_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>A list consisting of the names of all putatively convergent taxa</p>
</td></tr>
<tr><td><code id="convSig_+3A_user.ace">user.ace</code></td>
<td>
<p>A matrix of user supplied ancestral trait values at internal nodes (formatted as &quot;traits&quot; but with node number as rownames)</p>
</td></tr>
<tr><td><code id="convSig_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulations to use to assess significance</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script simulates data according to a Brownian motion model of
evolution, and then assesses convergene on that simulated data. The number
of times that the simulated data produces greater convergence than that
observed in the empirical data is used to calculate a P-value.
</p>


<h3>Value</h3>

<p>C1-C4 convergence measures for all pairs of putatively convergent
taxa and their associated p-values.
</p>


<h3>References</h3>

<p>Stayton, C.T. 2015. The definition, recognition, and interpretation of convergent evolution, and
two new measures for quantifying and assessing the significance of convergence. Evolution 69:2140-2453.
</p>
<p>Zelditch, M.L., J. Ye, J.S. Mitchell, and D.L. Swiderski. 2017. Rare ecomorphological
convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in
squirrels (Sciuridae). Evolution 71:633-649.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phy&lt;-rtree(100)
traits&lt;-fastBM(phy,nsim=3)
focaltaxa&lt;-c("t1","t50","t100")
answer&lt;-convSig(phy,traits,focaltaxa,user.ace=NULL,nsim=10)
</code></pre>

<hr>
<h2 id='convSigCt'>Computes and conducts significance tests on Ct-metric scores for putatively convergent tips (or groups of tips) given a set of user provided phenotypic characters and a time calibrated phylogeny.</h2><span id='topic+convSigCt'></span>

<h3>Description</h3>

<p>calcConvCt Computes and conducts significance tests on Ct-metric scores for putatively convergent tips (or groups of tips) given a set of user provided phenotypic characters and a time calibrated phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convSigCt(
  phy,
  traits,
  focaltaxa,
  groups = NULL,
  user.ace = NULL,
  nsim = 1000,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convSigCt_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_traits">traits</code></td>
<td>
<p>A matrix of numeric phenotypic traits with rownames matching tip labels of phy</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>A vector of tip labels for the putatively convergent taxa to be compared</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_groups">groups</code></td>
<td>
<p>An optional vector of groups with names matching focaltaxa. Indicates the group identity of all putatively convergent taxa and limits Ct measures to intergroup comparisons only</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_user.ace">user.ace</code></td>
<td>
<p>A matrix of user supplied ancestral trait values at internal nodes (formatted as &quot;traits&quot; but with node number as rownames)</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_nsim">nsim</code></td>
<td>
<p>Number of simulated (Brownian motion) datasets used to build the null distribution</p>
</td></tr>
<tr><td><code id="convSigCt_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to calcConvCt. If conservative == TRUE, Dmax.t will be restricted to occurr before the oldest stem lineage of the two groups involved in each pairwise comparison. Stem lineage age for each group is defined as the height of the parent node of the branch subtending the most recent common ancestor of tips within a group. Where groups include a single tip, the parent node of the tip's subtending branch is used. Requires group object to be provided by user.. If VERBOSE is TRUE, model information will be printed during computation, including time limits imposed on Dmax.t if the conservative option is chosen.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function incorporates the optimizations introduced by Zelditch et al. (2017), which significantly improve runtimes
</p>
<p>Reconstructions part way along branches are obtained using equation 2 of Felsenstein (1985), following code modified from the phytools (Revell, 2012) function contMap
</p>


<h3>Value</h3>

<p>A list of the following components:
</p>
<p>pvals a matrix containing Ct1 - Ct4 and p-values from significance tests for each
</p>
<p>meas.Cmat a matrix of Ct values for each pairwise comparison of focaltaxa
</p>
<p>meas.path a list of dataframes, one per pairwise comparison of focaltaxa, each containing information from all timepoint measurements of the two putatively convergent lineages. These provide the nodes at which comparisons were drawn, the evolutionary path along which that node fell (i.e., leading to one of two tips), the node height, reconstructed ancestral states at that node for each phenotypic trait, reconstructed ancestral values for each trait along the opposite path, and the phenotypic distance between the two lineages at that point.
</p>
<p>sim.avg average Ct values from all pairwise comparisons between focaltaxa using simulated Brownian motion traits, number of columns corresponds to the user provided number of simulations
</p>
<p>sim.path a list of dataframes as in meas.path, but obtained using simulated data. Length of object determined by number of pairwise comparisons multiplied by the number of simulated datasets.
</p>
<p>grp.mean a matrix of Ct-metrics summarized for inter-group comparisons, returned only if user defined groups were specified. Provides overall results matching those reported in &quot;mean&quot;, results for each unique inter-group comparison, and results averaged with equal weight given to each unique inter-group comparison (i.e., ignoring differences in the number of tips in each group).
</p>
<p>grp.pvals a matrix of p-values associated with Ct values in grp.mean object. Returned only if user defined groups were specified.
</p>
<p>limits a list of tree heights used to constrain Dmax.t calculations for each pairwise comparison in conservative analyses. Only returned if conservative == TRUE.
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(phytools)
library(geiger)

# create time calibrated tree
mytree&lt;-rtree(100)
mycalibration &lt;- makeChronosCalib(mytree, node="root", age.max=50)
phy &lt;- chronos(mytree, lambda = 1, model = "correlated", calibration = mycalibration, 
control = chronos.control() )
class(phy)&lt;-"phylo"

# create three normally distributed phenotypic traits
traits &lt;- cbind(rnorm(Ntip(phy)),rnorm(Ntip(phy)),rnorm(Ntip(phy)))
rownames(traits) &lt;- phy$tip.label

#	select two random tips, excluding sister taxa
pairs &lt;- apply(combn(phy$tip.label,2),2,function(x) nodepath(phy,which(phy$tip.label == x[1]),
which(phy$tip.label == x[2])))
nosis &lt;- combn(phy$tip.label,2)[,unlist(lapply(pairs, function(x) length(x) &gt; 3))]
focaltaxa &lt;- nosis[,sample(1:ncol(nosis),1)]

system.time(run &lt;- calcConvCt(phy, traits, focaltaxa))
system.time(run2 &lt;- convSigCt(phy, traits, focaltaxa, nsim=100))

plotCt(output = run2,phy = phy,focaltaxa = focaltaxa)

## End(Not run)
</code></pre>

<hr>
<h2 id='findanc'>Find the ancestor of a given node in a phylogeny</h2><span id='topic+findanc'></span>

<h3>Description</h3>

<p>This function will find the ancestor of a given node in a phylogeny.  It will return a two-element vector, which will contain both the node of the ancestor and the number of the edge that connects the node and ancestor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findanc(phyl, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findanc_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="findanc_+3A_node">node</code></td>
<td>
<p>The number of the node that you want the ancestor for</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns a two-element vector.  The node of the ancestor is first;  the edge that connects that node with its ancestor is second.
</p>


<h3>Value</h3>

<p>A two-element vector, where the first element is the node of the ancestor and the second element is the number of the edge that connects the node and ancestor (i.e., the row number in phyl$edge).
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
Paradis, E. (2012) Analysis of Phylogenetics and Evolution with R (Second Edition). New York: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
ancestor&lt;-findanc(phyl,1)
</code></pre>

<hr>
<h2 id='labelstonumbers'>Converts taxon names to tip/edge numbers</h2><span id='topic+labelstonumbers'></span>

<h3>Description</h3>

<p>Converts taxon names to corresponding tip/edge numbers in the phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labelstonumbers(phyl, tips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labelstonumbers_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="labelstonumbers_+3A_tips">tips</code></td>
<td>
<p>The names of the tips in question</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simply reads in taxon names, determines which tip or edge number they correspond to, and returns those values
</p>


<h3>Value</h3>

<p>The numbers of all of the tips of interest.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
Paradis, E. (2012) Analysis of Phylogenetics and Evolution with R (Second Edition). New York: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
nums&lt;-labelstonumbers(phyl,c("t1","t2","t3"))
</code></pre>

<hr>
<h2 id='maxdist'>Calculates the maximum phenotypic distance between the lineages leading to a pair of taxa.</h2><span id='topic+maxdist'></span>

<h3>Description</h3>

<p>maxdist uses ancestral state reconstruction to determine the maximum distance between any ancestors of those two taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxdist(phyl, phendata, t1, t2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maxdist_+3A_phyl">phyl</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="maxdist_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
<tr><td><code id="maxdist_+3A_t1">t1</code></td>
<td>
<p>The first taxon of interest</p>
</td></tr>
<tr><td><code id="maxdist_+3A_t2">t2</code></td>
<td>
<p>The second taxon of interest</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns the maximum Euclidean distance between any pair of ancestors of the two taxa, whether or not those two ancestors are contemporaries.
</p>


<h3>Value</h3>

<p>The maximum phenotypic distance between the two taxa
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phyl&lt;-rtree(10)
phendata&lt;-fastBM(phyl,nsim=2)
answer&lt;-maxdist(phyl,phendata,1,10)
</code></pre>

<hr>
<h2 id='multianc'>Reconstructs ancestral states for multiple characters</h2><span id='topic+multianc'></span>

<h3>Description</h3>

<p>Uses fastAnc to reconstruct ancestral states for multiple phenotypic characters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multianc(phy, phendata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multianc_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="multianc_+3A_phendata">phendata</code></td>
<td>
<p>Phenotypic data for all tips</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>A matrix with the tips data in the first n rows and the ancestral data in the remaining n-1 rows.
</p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics
and evolution in R langauge. Bioinformatics, 20, 289-290.
</p>
<p>Revell, L. J. (2012) phytools: An R package for phylogenetic comparative
biology (and other things). Methods Ecol. Evol. 3 217-223.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phy&lt;-rtree(10)

phendata&lt;-fastBM(phy,nsim=2)

ancs&lt;-multianc(phy,phendata)
</code></pre>

<hr>
<h2 id='phy'>Sample phylogeny for use in convevol examples and tutorial</h2><span id='topic+phy'></span>

<h3>Description</h3>

<p>It's a sample phylogeny with 100 tips.
</p>


<h3>Format</h3>

<p>A tree file
</p>

<hr>
<h2 id='plot2D'>Plots calcConv or convSig output as a two-dimensional time series.</h2><span id='topic+plot2D'></span>

<h3>Description</h3>

<p>plot2D Plots calcConv or convSig output as a two-dimensional time series.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot2D(
  Ct,
  phy,
  tip,
  foc.trt,
  trait,
  pca = FALSE,
  save = FALSE,
  filename = "frame",
  dir = NULL,
  leg = FALSE,
  leg.pos = "topleft",
  width = 480,
  height = 480,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot2D_+3A_ct">Ct</code></td>
<td>
<p>object containing calcConvCt or convSigCt output</p>
</td></tr>
<tr><td><code id="plot2D_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="plot2D_+3A_tip">tip</code></td>
<td>
<p>vector of two tip labels indicating the putatively convergent taxa to be plotted in morphospace</p>
</td></tr>
<tr><td><code id="plot2D_+3A_foc.trt">foc.trt</code></td>
<td>
<p>vector of two traits to be used in plotting. If pca == TRUE this should indicate which PC axes should be plotted in format &quot;PC1&quot;</p>
</td></tr>
<tr><td><code id="plot2D_+3A_trait">trait</code></td>
<td>
<p>the matrix of trait values used to compute Ct</p>
</td></tr>
<tr><td><code id="plot2D_+3A_pca">pca</code></td>
<td>
<p>logical value indicating whether to conduct principal component analysis and plot PC scores instead of raw trait values</p>
</td></tr>
<tr><td><code id="plot2D_+3A_save">save</code></td>
<td>
<p>logical value indicating whether an animated time series (along with individual time slices) should be saved to a user specified folder</p>
</td></tr>
<tr><td><code id="plot2D_+3A_filename">filename</code></td>
<td>
<p>character indicating the desired prefix for filenames to be saved</p>
</td></tr>
<tr><td><code id="plot2D_+3A_dir">dir</code></td>
<td>
<p>optional character indicating the folderpath of the desired save location</p>
</td></tr>
<tr><td><code id="plot2D_+3A_leg">leg</code></td>
<td>
<p>logical value indicating whether a legend should be added to plots</p>
</td></tr>
<tr><td><code id="plot2D_+3A_leg.pos">leg.pos</code></td>
<td>
<p>character indicating the position of the legend</p>
</td></tr>
<tr><td><code id="plot2D_+3A_width">width</code></td>
<td>
<p>pixel width of saved png files</p>
</td></tr>
<tr><td><code id="plot2D_+3A_height">height</code></td>
<td>
<p>pixel height of saved png files</p>
</td></tr>
<tr><td><code id="plot2D_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None
</p>


<h3>Value</h3>

<p>Plots tracking putatively convergent taxa in two-dimensional morphospace through time
</p>
<p>A table with trait values, morphospace distance, and nodeheights conincident with Dmax.t. See meas.path output from calcConvCt and convSigCt
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(phytools)
library(geiger)

# create time calibrated tree
mytree&lt;-rtree(100)
mycalibration &lt;- makeChronosCalib(mytree, node="root", age.max=50)
phy &lt;- chronos(mytree, lambda = 1, model = "correlated", 
calibration = mycalibration, control = chronos.control() )
class(phy)&lt;-"phylo"

# create three normally distributed phenotypic traits
traits &lt;- cbind(rnorm(Ntip(phy)),rnorm(Ntip(phy)),rnorm(Ntip(phy)))
colnames(traits) &lt;- c("V1","V2","V3")
rownames(traits) &lt;- phy$tip.label

#	select two random tips, excluding sister taxa
pairs &lt;- apply(combn(phy$tip.label,2),2,function(x) nodepath(phy,
which(phy$tip.label == x[1]),which(phy$tip.label == x[2])))
nosis &lt;- combn(phy$tip.label,2)[,unlist(lapply(pairs, function(x) length(x) &gt; 3))]
focaltaxa &lt;- nosis[,sample(1:ncol(nosis),1)]

system.time(run &lt;- calcConvCt(phy, traits, focaltaxa))
system.time(run2 &lt;- convSigCt(phy, traits, focaltaxa, nsim=100))

plot2D(run, phy, focaltaxa[1:2], colnames(traits)[1:2], traits)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotCt'>Plots calcConv or convSig output.</h2><span id='topic+plotCt'></span>

<h3>Description</h3>

<p>plotCt Plots calcConv or convSig output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotCt(
  output,
  phy,
  focaltaxa,
  nsim = 25,
  col = c("black", "forest green", "dodgerblue2", "firebrick1", "purple", "orange",
    "salmon", "goldenrod", "springgreen2", "plum1"),
  groups = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotCt_+3A_output">output</code></td>
<td>
<p>object containing calcConv or convSig output</p>
</td></tr>
<tr><td><code id="plotCt_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="plotCt_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>a vector of tip labels for the putatively convergent taxa to be compared</p>
</td></tr>
<tr><td><code id="plotCt_+3A_nsim">nsim</code></td>
<td>
<p>number of null simulations to plot</p>
</td></tr>
<tr><td><code id="plotCt_+3A_col">col</code></td>
<td>
<p>vector of colors to use for all unique intergroup comparisons a default option is given usable with up to five groups. If number of groups is 1 or less than length of col, not all colors will be used</p>
</td></tr>
<tr><td><code id="plotCt_+3A_groups">groups</code></td>
<td>
<p>an optional vector of groups with names matching focaltaxa, indicating the group identity of all focaltaxa</p>
</td></tr>
<tr><td><code id="plotCt_+3A_...">...</code></td>
<td>
<p>optional arguments to be passed to tiplabels</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a plot that shows the phenotypic distances between pairs of putatively convergent lineages over time. When these distances decrease, convergence has occurred. When more than two putatively
convergent taxa are analyzed, all pairs are plotted.
</p>


<h3>Value</h3>

<p>A plot identifying putatively convergent taxa in the provided phylogeny and tracking the change in phenotypic distance between taxa since their most recent common ancestor
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#' create time calibrated tree
phy&lt;-rcoal(100)

# create three normally distributed phenotypic traits
traits &lt;- cbind(rnorm(Ntip(phy)),rnorm(Ntip(phy)),rnorm(Ntip(phy)))
rownames(traits) &lt;- phy$tip.label
focaltaxa &lt;- sample(phy$tip.label, 5)

system.time(run2 &lt;- convSigCt(phy, traits, focaltaxa, nsim=10))

plotCt(output = run2,phy = phy,focaltaxa = focaltaxa)
</code></pre>

<hr>
<h2 id='plotellipse'>Plots an ellipse</h2><span id='topic+plotellipse'></span>

<h3>Description</h3>

<p>Plots a minimum ellipse around a set of data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotellipse(ellipse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotellipse_+3A_ellipse">ellipse</code></td>
<td>
<p>Gives the parameters of the ellipse - output from the ellipsoidhull functon in cluster.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Routine adapted from a suggestion made on CrossValidated:  http://stats.stackexchange.com/questions/9898/how-to-plot-an-ellipse-from-eigenvalues-and-eigenvectors-in-r
</p>


<h3>Value</h3>

<p>Nothing - just plots the ellipse.
</p>


<h3>References</h3>

<p>Maechler, M., Rousseeuw, P., Struyf, A., Hubert, M., Hornik, K.(2013).
cluster: Cluster Analysis Basics and Extensions. R package version 1.14.4.
</p>

<hr>
<h2 id='pullNodeSeq'>Extracts a vector of ancestors for a given taxon. Code written by Jonathan S. Mitchell for
Zelditch et al. (2017)</h2><span id='topic+pullNodeSeq'></span>

<h3>Description</h3>

<p>pullNodeSeq Extracts a vector of all ancestors of a given taxon in a phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pullNodeSeq(phy, tip)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pullNodeSeq_+3A_phy">phy</code></td>
<td>
<p>The phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="pullNodeSeq_+3A_tip">tip</code></td>
<td>
<p>The tip of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of ancestors
</p>


<h3>References</h3>

<p>Zelditch, M.L., J. Ye, J.S. Mitchell, and D.L. Swiderski. 2017. Rare ecomorphological
convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in
squirrels (Sciuridae). Evolution 71:633-649.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
phylogeny&lt;-rtree(100)
answer&lt;-pullNodeSeq(phy=phylogeny,tip="t1")
</code></pre>

<hr>
<h2 id='pwCheck'>Calculates the number of measurements that can be made between two lineages for each pairwise comparison within a set of putatively convergent tips (group identity may also be taken into account). Useful for determining which comparisons are not informative, and constructing a group object before running calcConvCt or convSigCt.</h2><span id='topic+pwCheck'></span>

<h3>Description</h3>

<p>pwCheck Calculates the number of measurements that can be made between two lineages for each pairwise comparison within a set of putatively convergent tips (group identity may also be taken into account). Useful for determining which comparisons are not informative, and constructing a group object before running calcConvCt or convSigCt.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pwCheck(phy, focaltaxa, groups = NULL, conservative = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pwCheck_+3A_phy">phy</code></td>
<td>
<p>The time calibrated phylogeny of interest in phylo format</p>
</td></tr>
<tr><td><code id="pwCheck_+3A_focaltaxa">focaltaxa</code></td>
<td>
<p>a vector of tip labels for the putatively convergent taxa to be compared</p>
</td></tr>
<tr><td><code id="pwCheck_+3A_groups">groups</code></td>
<td>
<p>an optional vector of groups with names matching focaltaxa. Indicates the group identity of all putatively convergent taxa and limits Ct measures to intergroup comparisons only</p>
</td></tr>
<tr><td><code id="pwCheck_+3A_conservative">conservative</code></td>
<td>
<p>logical value indicating whether candidate nodes for measurement of Dmax.t should be restricted to occurr before the oldest stem lineage of the two groups involved in each pairwise comparison. Stem lineage age for each group is defined as the height of the parent node of the branch subtending the most recent common ancestor of tips within a group. Where groups include a single tip, the parent node of the tip's subtending branch is used. Requires group object to be provided by user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the following components:
</p>
<p>taxa a matrix of uninformative tip comparisons
</p>
<p>path a vector with the number of measurements for all pairwise comparisons - a summary of this is also printed when running the function
</p>


<h3>References</h3>

<p>Grossnickle DM, Brightly WH, Weaver LN, Stanchak KE, Roston RA, Pevsner SK, Stayton CT, Polly PD, Law CJ. 2022. A cautionary note on quantitative measures of phenotypic convergence. in revision
Zelditch ML, Ye J, Mitchell JS, Swiderski DL. 2017. Rare ecomorphological convergence on a complex adaptive landscape: Body size and diet mediate evolution of jaw shape in squirrels (Sciuridae). Evolution 71: 633-649
Stayton CT. 2015. The definition, recognition, and interpretation of convergent evolution and two new measures for quantifying and assessing the significance of convergence. Evolution 69(8): 2140-2153.
Revell, L. J. (2012) phytools: An R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol., 3, 217-223.
Felsenstein, J. 1985. Phylogenies and the comparative method. American Naturalist, 125, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(phytools)
library(geiger)

# create time calibrated tree
mytree&lt;-rtree(100)
mycalibration &lt;- makeChronosCalib(mytree, node="root", age.max=50)
phy &lt;- chronos(mytree, lambda = 1, model = "correlated", calibration = mycalibration, 
control = chronos.control() )
class(phy)&lt;-"phylo"

# create three normally distributed phenotypic traits
traits &lt;- cbind(rnorm(Ntip(phy)),rnorm(Ntip(phy)),rnorm(Ntip(phy)))
rownames(traits) &lt;- phy$tip.label

#	select two random tips, excluding sister taxa
pairs &lt;- apply(combn(phy$tip.label,2),2,function(x) nodepath(phy,which(phy$tip.label == x[1]),
which(phy$tip.label == x[2])))
nosis &lt;- combn(phy$tip.label,2)[,unlist(lapply(pairs, function(x) length(x) &gt; 3))]
focaltaxa &lt;- nosis[,sample(1:ncol(nosis),1)]

pwCheck(phy,focaltaxa)

## End(Not run)
</code></pre>

<hr>
<h2 id='trait'>Sample data for use in convevol examples and tutorial</h2><span id='topic+trait'></span>

<h3>Description</h3>

<p>It's a data matrix with 6 variables measured across 100 tips of a phylogeny (&quot;phy&quot; - also included)
</p>


<h3>Format</h3>

<p>I'll need to look into this
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
