<!DOCTYPE html><html><head><title>Help for package cophescan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cophescan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust_priors'><p>adjust_priors</p></a></li>
<li><a href='#average_piks'><p>Average of priors: pnk, pak and pck</p></a></li>
<li><a href='#average_piks_list'><p>Average of priors: pnk, pak and pck from list (memory intensive)</p></a></li>
<li><a href='#average_posterior_prob'><p>Average of posterior probabilities: Hn, Ha and Hc</p></a></li>
<li><a href='#average_posterior_prob_list'><p>Average of posterior probabilities: Hn, Ha and Hc from list (memory intensive)</p></a></li>
<li><a href='#combine.bf'><p>combine.bf</p></a></li>
<li><a href='#cophe_heatmap'><p>Heatmap of multi-trait cophescan results</p></a></li>
<li><a href='#cophe_multi_trait_data'><p>Simulated multi-trait data</p></a></li>
<li><a href='#cophe_plot'><p>cophe_plots showing the Ha and Hc of all traits and labelled above the specified threshold</p></a></li>
<li><a href='#cophe.bf_bf'><p>extract data through Bayes factors</p></a></li>
<li><a href='#cophe.hyp.predict'><p>Predict cophescan hypothesis for tested associations</p></a></li>
<li><a href='#cophe.multitrait'><p>Run cophescan on multiple traits at once</p></a></li>
<li><a href='#cophe.prepare.dat.single'><p>Prepare data for cophe.single</p></a></li>
<li><a href='#cophe.prepare.dat.susie'><p>Prepare data for <code>cophe.susie</code></p></a></li>
<li><a href='#cophe.single'><p>Bayesian cophescan analysis using Approximate Bayes Factors</p></a></li>
<li><a href='#cophe.single.lbf'><p>cophe.single.lbf</p></a></li>
<li><a href='#cophe.susie'><p>run <code>cophe.susie</code> using susie to detect separate signals</p></a></li>
<li><a href='#cophe.susie.lbf'><p>cophe.susie.lbf</p></a></li>
<li><a href='#cophescan-package'><p>The 'cophescan' package.</p></a></li>
<li><a href='#get_beta'><p>Extract beta and p-values of queried variant</p></a></li>
<li><a href='#get_posterior_prob'><p>Calculation of the posterior prob of Hn, Ha and Hc</p></a></li>
<li><a href='#Hc.cutoff.fdr'><p>Estimate the Hc.cutoff for the required FDR</p></a></li>
<li><a href='#hypothesis.priors'><p>hypothesis.priors</p></a></li>
<li><a href='#logd_alpha'><p>dnorm for alpha</p></a></li>
<li><a href='#logd_beta'><p>dgamma for beta</p></a></li>
<li><a href='#logd_gamma'><p>dgamma for gamma</p></a></li>
<li><a href='#loglik'><p>Log likelihood calculation</p></a></li>
<li><a href='#logpost'><p>Log posterior calculation</p></a></li>
<li><a href='#logpriors'><p>Calculate log priors</p></a></li>
<li><a href='#logsum'><p>logsum</p></a></li>
<li><a href='#logsumexp'><p>Log sum</p></a></li>
<li><a href='#metrop_run'><p>Run the hierarchical mcmc model to infer priors</p></a></li>
<li><a href='#multitrait.simplify'><p>Simplifying the output obtained from <code>cophe.multitrait</code>, <code>cophe.single</code> or <code>cophe.susie</code></p></a></li>
<li><a href='#pars_init'><p>Initiate parameters alpha, beta and gamma</p></a></li>
<li><a href='#pars2pik'><p>Conversion of parameters alpha, beta and gamma to pnk, pak and pck</p></a></li>
<li><a href='#per.snp.priors'><p>per.snp.priors</p></a></li>
<li><a href='#piks'><p>List of priors: pn, pa and pc over all iterations</p></a></li>
<li><a href='#plot_trait_manhat'><p>Plot region Manhattan for a trait highlighting the queried variant</p></a></li>
<li><a href='#posterior_prob'><p>List of posterior probabilities: Hn, Ha and Hc over all iterations</p></a></li>
<li><a href='#prepare_plot_data'><p>Prepare data for plotting</p></a></li>
<li><a href='#propose'><p>Proposal distribution</p></a></li>
<li><a href='#run_metrop_priors'><p>Run the hierarchical Metropolis Hastings model to infer priors</p></a></li>
<li><a href='#sample_alpha'><p>sample alpha</p></a></li>
<li><a href='#sample_beta'><p>sample beta</p></a></li>
<li><a href='#sample_gamma'><p>sample gamma</p></a></li>
<li><a href='#summary.cophe'><p>print the summary of results from cophescan single or susie</p></a></li>
<li><a href='#target'><p>Target distribution</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Adaptation of the Coloc Method for PheWAS</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ichcha Manipur &lt;im504@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
     A Bayesian method for Phenome-wide association studies (PheWAS) that identifies causal associations between genetic variants and traits, while simultaneously addressing confounding due to linkage disequilibrium. For details see   
     Manipur et al (2023) &lt;<a href="https://doi.org/10.1101%2F2023.06.29.546856">doi:10.1101/2023.06.29.546856</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ichcha-m/cophescan">https://github.com/ichcha-m/cophescan</a>,
<a href="https://ichcha-m.github.io/cophescan/">https://ichcha-m.github.io/cophescan/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ichcha-m/cophescan/issues">https://github.com/ichcha-m/cophescan/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.7), coloc, data.table, ggplot2, ggrepel,
pheatmap, methods, viridis, stats, grDevices, magrittr, utils,
matrixStats, dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat (&ge; 3.0.0), rmarkdown, RColorBrewer, ggpubr</td>
</tr>
<tr>
<td>Collate:</td>
<td>'cophescan-package.R' 'singlevar.R' 'multivarsusie.R'
'multitrait.R' 'cophe_hyp_predict.R' 'copheplots.R'
'testdata.R' 'RcppExports.R' 'metrop_hier_priors.R' 'zzz.R'</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-11 19:48:22 UTC; ichcha</td>
</tr>
<tr>
<td>Author:</td>
<td>Ichcha Manipur [aut, cre],
  Chris Wallace [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-11 20:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjust_priors'>adjust_priors</h2><span id='topic+adjust_priors'></span>

<h3>Description</h3>

<p>adjust fixed priors when nsnps in region is high
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_priors(
  nsnps,
  pa = 3.82e-05,
  pc = 0.00182,
  p1 = NULL,
  p2 = NULL,
  p12 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjust_priors_+3A_nsnps">nsnps</code></td>
<td>
<p>number of SNPs</p>
</td></tr>
<tr><td><code id="adjust_priors_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait (cophescan prior), default 3.82e-5</p>
</td></tr>
<tr><td><code id="adjust_priors_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait (cophescan prior), default 1.82e-3 (cophescan prior)</p>
</td></tr>
<tr><td><code id="adjust_priors_+3A_p1">p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code>; use p1, p2, p12 only when pa and pc are unavailable (See vignettes)</p>
</td></tr>
<tr><td><code id="adjust_priors_+3A_p2">p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2,  (coloc prior), pa derived by using <code class="reqn">pa = p2</code></p>
</td></tr>
<tr><td><code id="adjust_priors_+3A_p12">p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits,  (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of pn, pa and pc adjusted prior probabilities
</p>

<hr>
<h2 id='average_piks'>Average of priors: pnk, pak and pck</h2><span id='topic+average_piks'></span>

<h3>Description</h3>

<p>Average of priors: pnk, pak and pck
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average_piks(params, nsnps, covar_vec, nits, thin, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average_piks_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="average_piks_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="average_piks_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="average_piks_+3A_nits">nits</code></td>
<td>
<p>Number of iterations run in mcmc</p>
</td></tr>
<tr><td><code id="average_piks_+3A_thin">thin</code></td>
<td>
<p>thinning</p>
</td></tr>
<tr><td><code id="average_piks_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>average pik matrix of priors: pnk, pak and pck
</p>

<hr>
<h2 id='average_piks_list'>Average of priors: pnk, pak and pck from list (memory intensive)</h2><span id='topic+average_piks_list'></span>

<h3>Description</h3>

<p>Average of priors: pnk, pak and pck from list (memory intensive)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average_piks_list(params, nsnps, covar_vec, nits, thin, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average_piks_list_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="average_piks_list_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="average_piks_list_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="average_piks_list_+3A_nits">nits</code></td>
<td>
<p>Number of iterations run in mcmc</p>
</td></tr>
<tr><td><code id="average_piks_list_+3A_thin">thin</code></td>
<td>
<p>thinning</p>
</td></tr>
<tr><td><code id="average_piks_list_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>average pik matrix of priors: pnk, pak and pck
</p>

<hr>
<h2 id='average_posterior_prob'>Average of posterior probabilities: Hn, Ha and Hc</h2><span id='topic+average_posterior_prob'></span>

<h3>Description</h3>

<p>Average of posterior probabilities: Hn, Ha and Hc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average_posterior_prob(
  params,
  lbf_mat,
  nsnps,
  covar_vec,
  nits,
  thin,
  covar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average_posterior_prob_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_nits">nits</code></td>
<td>
<p>Number of iterations run in mcmc</p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_thin">thin</code></td>
<td>
<p>thinning</p>
</td></tr>
<tr><td><code id="average_posterior_prob_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with average of all the posterior probabilities: Hn, Ha and Hc
</p>

<hr>
<h2 id='average_posterior_prob_list'>Average of posterior probabilities: Hn, Ha and Hc from list (memory intensive)</h2><span id='topic+average_posterior_prob_list'></span>

<h3>Description</h3>

<p>Average of posterior probabilities: Hn, Ha and Hc from list (memory intensive)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>average_posterior_prob_list(
  params,
  lbf_mat,
  nsnps,
  covar_vec,
  nits,
  thin,
  covar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="average_posterior_prob_list_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_nits">nits</code></td>
<td>
<p>Number of iterations run in mcmc</p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_thin">thin</code></td>
<td>
<p>thinning</p>
</td></tr>
<tr><td><code id="average_posterior_prob_list_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with average of all the posterior probabilities: Hn, Ha and Hc
</p>

<hr>
<h2 id='combine.bf'>combine.bf</h2><span id='topic+combine.bf'></span>

<h3>Description</h3>

<p>Calculate posterior probabilities for all the configurations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine.bf(lBF_df, pn, pa, pc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine.bf_+3A_lbf_df">lBF_df</code></td>
<td>
<p>dataframe with log bayes factors of hypothesis Ha  and Hn: column names should be lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="combine.bf_+3A_pn">pn</code></td>
<td>
<p>prior probability that none of the SNPs/variants in the region are associated with the query trait</p>
</td></tr>
<tr><td><code id="combine.bf_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait</p>
</td></tr>
<tr><td><code id="combine.bf_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named numeric vector of posterior probabilities and bayes factors
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>

<hr>
<h2 id='cophe_heatmap'>Heatmap of multi-trait cophescan results</h2><span id='topic+cophe_heatmap'></span>

<h3>Description</h3>

<p>Heatmap of multi-trait cophescan results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe_heatmap(
  multi.dat,
  querysnpid,
  query_trait_names,
  thresh_Hc = 0.5,
  thresh_Ha = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe_heatmap_+3A_multi.dat">multi.dat</code></td>
<td>
<p>multi trait cophescan results returned from <code>cophe.multitrait</code> or formatted in the same way with multitrait.simplify</p>
</td></tr>
<tr><td><code id="cophe_heatmap_+3A_querysnpid">querysnpid</code></td>
<td>
<p>query variant</p>
</td></tr>
<tr><td><code id="cophe_heatmap_+3A_query_trait_names">query_trait_names</code></td>
<td>
<p>names of phenotypes corresponding to the multi.dat results</p>
</td></tr>
<tr><td><code id="cophe_heatmap_+3A_thresh_hc">thresh_Hc</code></td>
<td>
<p>Hc threshold to be displayed</p>
</td></tr>
<tr><td><code id="cophe_heatmap_+3A_thresh_ha">thresh_Ha</code></td>
<td>
<p>Ha threshold to be displayed</p>
</td></tr>
<tr><td><code id="cophe_heatmap_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to
<a href="pheatmap.html#topic+pheatmap">pheatmap</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>heatmap of posterior probabilities of the phentypes above the set threshold
</p>

<hr>
<h2 id='cophe_multi_trait_data'>Simulated multi-trait data</h2><span id='topic+cophe_multi_trait_data'></span>

<h3>Description</h3>

<p>Simulated multi-trait data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cophe_multi_trait_data)
</code></pre>


<h3>Format</h3>

<p>list of coloc structred datasets for 24 traits (cophe_multi_trait_data$summ_stat), LD matrix (cophe_multi_trait_data$LD) and the id of the query snp  (cophe_multi_trait_data$querysnpid). #' The trait dataset  are simulated summary statistics (1000 SNPs) for 10 Hn, 10 Ha and 10 Hc.
</p>

<hr>
<h2 id='cophe_plot'>cophe_plots showing the Ha and Hc of all traits and labelled above the specified threshold</h2><span id='topic+cophe_plot'></span>

<h3>Description</h3>

<p>cophe_plots showing the Ha and Hc of all traits and labelled above the specified threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe_plot(
  multi.dat,
  querysnpid,
  query_trait_names,
  thresh_Hc = 0.5,
  thresh_Ha = 0.5,
  beta_p = NULL,
  traits.dat = NULL,
  group_pheno = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe_plot_+3A_multi.dat">multi.dat</code></td>
<td>
<p>multi trait cophescan results returned from cophe.multitrait or multitrait.simplify</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_querysnpid">querysnpid</code></td>
<td>
<p>query variant (only a single variant for PheWAS plots)</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_query_trait_names">query_trait_names</code></td>
<td>
<p>list of phenotype names</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_thresh_hc">thresh_Hc</code></td>
<td>
<p>Hc threshold to be displayed</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_thresh_ha">thresh_Ha</code></td>
<td>
<p>Ha threshold to be displayed</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_beta_p">beta_p</code></td>
<td>
<p>data.frame (from the <code>get.beta</code> function) with four columns : 1. &quot;beta_plot&quot;: indicating beta direction (p or n) 2. &quot;beta_plot&quot;: -log10(pval) of the queried variant 3. &quot;querysnp&quot; 4. &quot;querytrait&quot;.</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_traits.dat">traits.dat</code></td>
<td>
<p>list of multi-trait coloc structured datasets</p>
</td></tr>
<tr><td><code id="cophe_plot_+3A_group_pheno">group_pheno</code></td>
<td>
<p>Vector with additional grouping of phenotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>cophescan plots of Ha and Hc
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.single">cophe.single</a></code>, <code><a href="#topic+cophe.susie">cophe.susie</a></code>, <code><a href="#topic+cophe.multitrait">cophe.multitrait</a></code>, , <code><a href="#topic+multitrait.simplify">multitrait.simplify</a></code>
</p>

<hr>
<h2 id='cophe.bf_bf'>extract data through Bayes factors</h2><span id='topic+cophe.bf_bf'></span>

<h3>Description</h3>

<p>a dataset represented by Bayes factors from SuSIE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.bf_bf(
  sus_dat,
  cred_set,
  querysnpid,
  querytrait,
  pn = NULL,
  pa = NULL,
  pc = NULL,
  ret_pp = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.bf_bf_+3A_sus_dat">sus_dat</code></td>
<td>
<p>a list with the output of running susie</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_cred_set">cred_set</code></td>
<td>
<p>credible set extracted from susie</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_querytrait">querytrait</code></td>
<td>
<p>Query trait name</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_pn">pn</code></td>
<td>
<p>prior probability that none of the SNPs/variants in the region are associated with the query trait</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait</p>
</td></tr>
<tr><td><code id="cophe.bf_bf_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait</p>
</td></tr>
</table>


<h3>Value</h3>

<p>bayes factors of signals
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.susie">cophe.susie</a></code>
</p>

<hr>
<h2 id='cophe.hyp.predict'>Predict cophescan hypothesis for tested associations</h2><span id='topic+cophe.hyp.predict'></span>

<h3>Description</h3>

<p>Predict cophescan hypothesis for tested associations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.hyp.predict(
  cophe.res,
  grouping.vars = c("querysnp", "querytrait"),
  Hc.cutoff = 0.6,
  Hn.cutoff = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.hyp.predict_+3A_cophe.res">cophe.res</code></td>
<td>
<p>results obtained from <code>cophe.single</code>, <code>cophe.susie</code> or <code>cophe.multitrait</code> or data.frame with the following columns: PP.Hn, PP.Hc, PP.Ha, querysnp, querytrait</p>
</td></tr>
<tr><td><code id="cophe.hyp.predict_+3A_grouping.vars">grouping.vars</code></td>
<td>
<p>This is important for results from <code>cophe.susie</code> where there are multiple signals. These will be collapsed into one call. If you want to return all signals set this to a single variable eg: grouping.vars = c('querysnp')</p>
</td></tr>
<tr><td><code id="cophe.hyp.predict_+3A_hc.cutoff">Hc.cutoff</code></td>
<td>
<p>threshold for PP.Hc above which the associations are called Hc</p>
</td></tr>
<tr><td><code id="cophe.hyp.predict_+3A_hn.cutoff">Hn.cutoff</code></td>
<td>
<p>threshold for PP.Hn above which the associations are called Hn</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns dataframe with posterior probabilties of Hn, Hc and Ha with the predicted hypothesis based on the provided cut.offs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.single">cophe.single</a></code>, <code><a href="#topic+cophe.susie">cophe.susie</a></code>, <code><a href="#topic+cophe.multitrait">cophe.multitrait</a></code>, , <code><a href="#topic+multitrait.simplify">multitrait.simplify</a></code>
</p>

<hr>
<h2 id='cophe.multitrait'>Run cophescan on multiple traits at once</h2><span id='topic+cophe.multitrait'></span>

<h3>Description</h3>

<p>Run cophescan on multiple traits at once
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.multitrait(
  trait.dat,
  querysnpid,
  querytrait.names,
  LDmat = NULL,
  method = "single",
  simplify = FALSE,
  predict.hyp = TRUE,
  Hn.cutoff = 0.2,
  Hc.cutoff = 0.6,
  est.fdr.based.cutoff = FALSE,
  fdr = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.multitrait_+3A_trait.dat">trait.dat</code></td>
<td>
<p>Named(traits) list of coloc structured data for k traits (Total number of traits)</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_querysnpid">querysnpid</code></td>
<td>
<p>vector of query variant ids = length(trait.dat), if the same variant</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_querytrait.names">querytrait.names</code></td>
<td>
<p>vector of names for the query traits, if the names of
the multi.dat list contain the trait names please pass querytrait.names=names(multi.dat)</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_ldmat">LDmat</code></td>
<td>
<p>LD matrix</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_method">method</code></td>
<td>
<p>either 'single' for <code>cophe.single</code> or 'susie' for <code>cophe.susie</code></p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_simplify">simplify</code></td>
<td>
<p>if TRUE removes intermediate results from output using 'multitrait.simplify'</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_predict.hyp">predict.hyp</code></td>
<td>
<p>if TRUE predicts the hypothesis based on the provided thresholds for pp.Hc and pp.Hn (overrides simplify) using <code>cophe.hyp.predict</code></p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_hn.cutoff">Hn.cutoff</code></td>
<td>
<p>threshold for PP.Hc above which the associations are called Hc</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_hc.cutoff">Hc.cutoff</code></td>
<td>
<p>threshold for PP.Hc above which the associations are called Hn</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_est.fdr.based.cutoff">est.fdr.based.cutoff</code></td>
<td>
<p>if True calculates the Hc.cutoff using 1-mean(PP.Hc)|PP.Hc &gt; cutoff</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_fdr">fdr</code></td>
<td>
<p>fdr threshold to estimate Hc.cutoff</p>
</td></tr>
<tr><td><code id="cophe.multitrait_+3A_...">...</code></td>
<td>
<p>additional arguments of priors for <code>cophe.susie</code> or <code>cophe.single</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>if simplify is False returns multi-trait list of lists, each with:
</p>

<ul>
<li><p> a summary data.frame of the cophescan results
</p>
</li>
<li><p> priors used
</p>
</li>
<li><p> querysnp
</p>
</li>
<li><p> querytrait
</p>
</li></ul>

<p>if simplify is TRUE only returns dataframe with posterior probabilties of Hn, Hc and Ha with no intermediate results
if predict.hyp is TRUE returns a dataframe with output of simplify and the predicted hypotheses for all associations
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>

<hr>
<h2 id='cophe.prepare.dat.single'>Prepare data for cophe.single</h2><span id='topic+cophe.prepare.dat.single'></span>

<h3>Description</h3>

<p>Prepare data for cophe.single
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.prepare.dat.single(dataset, querysnpid, MAF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.prepare.dat.single_+3A_dataset">dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td></tr>
<tr><td><code id="cophe.prepare.dat.single_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list containing the per snp BFs (df) and position of the query variant (querypos)
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>

<hr>
<h2 id='cophe.prepare.dat.susie'>Prepare data for <code>cophe.susie</code></h2><span id='topic+cophe.prepare.dat.susie'></span>

<h3>Description</h3>

<p>Prepare data for <code>cophe.susie</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.prepare.dat.susie(dataset, querysnpid, susie.args)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.prepare.dat.susie_+3A_dataset">dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td></tr>
<tr><td><code id="cophe.prepare.dat.susie_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td></tr>
<tr><td><code id="cophe.prepare.dat.susie_+3A_susie.args">susie.args</code></td>
<td>
<p>a named list of additional arguments to be passed to
<a href="coloc.html#topic+runsusie">runsusie</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the output of running susie
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.susie">cophe.susie</a></code>
</p>

<hr>
<h2 id='cophe.single'>Bayesian cophescan analysis using Approximate Bayes Factors</h2><span id='topic+cophe.single'></span>

<h3>Description</h3>

<p>Bayesian cophescan analysis under single causal variant assumption
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.single(
  dataset,
  querysnpid,
  querytrait,
  MAF = NULL,
  pa = 3.82e-05,
  pc = 0.00182,
  p1 = NULL,
  p2 = NULL,
  p12 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.single_+3A_dataset">dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_querytrait">querytrait</code></td>
<td>
<p>Query trait name</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_maf">MAF</code></td>
<td>
<p>Minor allele frequency vector</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait (cophescan prior), default 3.82e-5</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait (cophescan prior), default 1.82e-3 (cophescan prior)</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_p1">p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code>; use p1, p2, p12 only when pa and pc are unavailable (See vignettes)</p>
</td></tr>
<tr><td><code id="cophe.single_+3A_p2">p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2,  (coloc prior), pa derived by using <code class="reqn">pa = p2</code></p>
</td></tr>
<tr><td><code id="cophe.single_+3A_p12">p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits,  (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates posterior probabilities of different
causal variant configurations under the assumption of a single
causal variant for each trait.
</p>
<p>If regression coefficients and variances are available, it
calculates Bayes factors for association at each SNP.  If only p
values are available, it uses an approximation that depends on the
SNP's MAF and ignores any uncertainty in imputation.  Regression
coefficients should be used if available. Find more input data structure details
in the coloc package
</p>


<h3>Value</h3>

<p>a list of two <code>data.frame</code>s:
</p>

<ul>
<li><p> summary is a vector giving the number of SNPs analysed, and the posterior probabilities of Hn (no shared causal variant), Ha (two distinct causal variants) and Hc (one common causal variant)
</p>
</li>
<li><p> results is an annotated version of the input data containing log Approximate Bayes Factors and intermediate calculations, and the posterior probability SNP.PP.Hc of the SNP being causal for the shared signal <em>if</em> Hc is true. This is only relevant if the posterior support for Hc in summary is convincing.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(cophescan)
data(cophe_multi_trait_data)
query_trait_1 &lt;- cophe_multi_trait_data$summ_stat[['Trait_1']]
querysnpid &lt;- cophe_multi_trait_data$querysnpid
res.single &lt;- cophe.single(query_trait_1, querysnpid = querysnpid, querytrait='Trait_1')
summary(res.single)
</code></pre>

<hr>
<h2 id='cophe.single.lbf'>cophe.single.lbf</h2><span id='topic+cophe.single.lbf'></span>

<h3>Description</h3>

<p>Calculate log bayes factors for each hypothesis (Single causal variant assumption)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.single.lbf(dataset, querysnpid, querytrait, MAF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.single.lbf_+3A_dataset">dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td></tr>
<tr><td><code id="cophe.single.lbf_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td></tr>
<tr><td><code id="cophe.single.lbf_+3A_querytrait">querytrait</code></td>
<td>
<p>Query trait name</p>
</td></tr>
<tr><td><code id="cophe.single.lbf_+3A_maf">MAF</code></td>
<td>
<p>Minor allele frequency vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with log bayes factors for Hn and Ha hypotheses
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.single">cophe.single</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(cophescan)
data(cophe_multi_trait_data)
query_trait_1 &lt;- cophe_multi_trait_data$summ_stat[['Trait_1']]
querysnpid &lt;- cophe_multi_trait_data$querysnpid
res.single.lbf &lt;- cophe.single.lbf(query_trait_1, querysnpid = querysnpid, querytrait='Trait_1')
res.single.lbf
</code></pre>

<hr>
<h2 id='cophe.susie'>run <code>cophe.susie</code> using susie to detect separate signals</h2><span id='topic+cophe.susie'></span>

<h3>Description</h3>

<p>Check if a variant causally associated in one trait might be causal in another trait
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.susie(
  dataset,
  querysnpid,
  querytrait,
  pa = 3.82e-05,
  pc = 0.00182,
  p1 = NULL,
  p2 = NULL,
  p12 = NULL,
  susie.args = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.susie_+3A_dataset">dataset</code></td>
<td>
<p><em>either</em> a list with specifically named elements defining the dataset
to be analysed. (see
<a href="coloc.html#topic+check_dataset">check_dataset</a>), or the result of running <a href="coloc.html#topic+runsusie">runsusie</a> on such a
dataset</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_querytrait">querytrait</code></td>
<td>
<p>Query trait name</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait (cophescan prior), default 3.82e-5</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait (cophescan prior), default 1.82e-3 (cophescan prior)</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_p1">p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code>; use p1, p2, p12 only when pa and pc are unavailable (See vignettes)</p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_p2">p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2,  (coloc prior), pa derived by using <code class="reqn">pa = p2</code></p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_p12">p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits,  (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code></p>
</td></tr>
<tr><td><code id="cophe.susie_+3A_susie.args">susie.args</code></td>
<td>
<p>a named list of additional arguments to be passed to
<a href="coloc.html#topic+runsusie">runsusie</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list, containing elements
</p>

<ul>
<li><p> summary a data.table of posterior
probabilities of each global hypothesis, one row per pairwise comparison
of signals from the two traits
</p>
</li>
<li><p> results a data.table of detailed results giving the posterior probability
for each snp to be jointly causal for both traits <em>assuming Hc is true</em>.
Please ignore this column if the corresponding posterior support for H4
is not high.
</p>
</li>
<li><p> priors a vector of the priors used for the analysis
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(cophescan)
data(cophe_multi_trait_data)
query_trait_1 &lt;- cophe_multi_trait_data$summ_stat[['Trait_1']]
querysnpid &lt;- cophe_multi_trait_data$querysnpid
query_trait_1$LD &lt;- cophe_multi_trait_data$LD
res.susie &lt;- cophe.susie(query_trait_1, querysnpid = querysnpid, querytrait='Trait_1')
summary(res.susie)
</code></pre>

<hr>
<h2 id='cophe.susie.lbf'>cophe.susie.lbf</h2><span id='topic+cophe.susie.lbf'></span>

<h3>Description</h3>

<p>Calculate log bayes factors for each hypothesis (SuSIE - multiple causal variant assumption)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cophe.susie.lbf(
  dataset,
  querysnpid,
  querytrait,
  switch = TRUE,
  susie.args = list(),
  MAF = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cophe.susie.lbf_+3A_dataset">dataset</code></td>
<td>
<p>a list with specifically named elements defining the query trait dataset
to be analysed.</p>
</td></tr>
<tr><td><code id="cophe.susie.lbf_+3A_querysnpid">querysnpid</code></td>
<td>
<p>Id of the query variant, (id in dataset$snp)</p>
</td></tr>
<tr><td><code id="cophe.susie.lbf_+3A_querytrait">querytrait</code></td>
<td>
<p>Query trait name</p>
</td></tr>
<tr><td><code id="cophe.susie.lbf_+3A_switch">switch</code></td>
<td>
<p>Set switch=TRUE to obtain single BF when credible sets not found with SuSIE</p>
</td></tr>
<tr><td><code id="cophe.susie.lbf_+3A_susie.args">susie.args</code></td>
<td>
<p>a named list of additional arguments to be passed to
<a href="coloc.html#topic+runsusie">runsusie</a></p>
</td></tr>
<tr><td><code id="cophe.susie.lbf_+3A_maf">MAF</code></td>
<td>
<p>Minor allele frequency vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with log bayes factors for Hn and Ha hypotheses
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.susie">cophe.susie</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(cophescan)
data(cophe_multi_trait_data)
query_trait_1 &lt;- cophe_multi_trait_data$summ_stat[['Trait_1']]
query_trait_1$LD &lt;- cophe_multi_trait_data$LD
querysnpid &lt;- cophe_multi_trait_data$querysnpid
res.susie.lbf &lt;- cophe.susie.lbf(query_trait_1, querysnpid = querysnpid,
                                  querytrait='Trait_1', switch=T)
res.susie.lbf
</code></pre>

<hr>
<h2 id='cophescan-package'>The 'cophescan' package.</h2><span id='topic+cophescan-package'></span><span id='topic+cophescan'></span>

<h3>Description</h3>

<p>Coloc adapted Phenome-wide Scans
</p>

<hr>
<h2 id='get_beta'>Extract beta and p-values of queried variant</h2><span id='topic+get_beta'></span>

<h3>Description</h3>

<p>Extract beta and p-values of queried variant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_beta(traits.dat, querysnpid, querytrait)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_beta_+3A_traits.dat">traits.dat</code></td>
<td>
<p>list of coloc structured dataset</p>
</td></tr>
<tr><td><code id="get_beta_+3A_querysnpid">querysnpid</code></td>
<td>
<p>vector of querysnpid</p>
</td></tr>
<tr><td><code id="get_beta_+3A_querytrait">querytrait</code></td>
<td>
<p>vector of querytrait names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with one column named beta_plot: indicating beta direction (n/p) and another column named pval_plot with -log10(pval) of the queried variant
</p>

<hr>
<h2 id='get_posterior_prob'>Calculation of the posterior prob of Hn, Ha and Hc</h2><span id='topic+get_posterior_prob'></span>

<h3>Description</h3>

<p>Calculation of the posterior prob of Hn, Ha and Hc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_posterior_prob(params, lbf_mat, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_posterior_prob_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="get_posterior_prob_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="get_posterior_prob_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="get_posterior_prob_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="get_posterior_prob_+3A_covar">covar</code></td>
<td>
<p>logical: should the covariate inflormation be used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>posterior prob of Hn, Ha and Hc
</p>

<hr>
<h2 id='Hc.cutoff.fdr'>Estimate the Hc.cutoff for the required FDR</h2><span id='topic+Hc.cutoff.fdr'></span>

<h3>Description</h3>

<p>Estimate the Hc.cutoff for the required FDR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hc.cutoff.fdr(ppHc, fdr = 0.05, return_plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hc.cutoff.fdr_+3A_pphc">ppHc</code></td>
<td>
<p>a vector containing the PP.Hc (the posterior probability of causal association) of all tests</p>
</td></tr>
<tr><td><code id="Hc.cutoff.fdr_+3A_fdr">fdr</code></td>
<td>
<p>FDR default: 0.05</p>
</td></tr>
<tr><td><code id="Hc.cutoff.fdr_+3A_return_plot">return_plot</code></td>
<td>
<p>default: TRUE, plot the fdr estimated at the different Hc.cutoff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Hc.cutoff value for the specified FDR, if return_plot is True returns a plot showing the FDR calculated at different Hc thresholds
</p>

<hr>
<h2 id='hypothesis.priors'>hypothesis.priors</h2><span id='topic+hypothesis.priors'></span>

<h3>Description</h3>

<p>Estimate priors for each hypothesis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hypothesis.priors(nsnps, pn, pa, pc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hypothesis.priors_+3A_nsnps">nsnps</code></td>
<td>
<p>number of SNPs</p>
</td></tr>
<tr><td><code id="hypothesis.priors_+3A_pn">pn</code></td>
<td>
<p>prior probability that none of the SNPs/variants in the region are associated with the query trait</p>
</td></tr>
<tr><td><code id="hypothesis.priors_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait</p>
</td></tr>
<tr><td><code id="hypothesis.priors_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait</p>
</td></tr>
</table>


<h3>Value</h3>

<p>hypotheses priors
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>

<hr>
<h2 id='logd_alpha'>dnorm for alpha</h2><span id='topic+logd_alpha'></span>

<h3>Description</h3>

<p>dnorm for alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logd_alpha(a, alpha_mean = -10, alpha_sd = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logd_alpha_+3A_a">a</code></td>
<td>
<p>current alpha</p>
</td></tr>
<tr><td><code id="logd_alpha_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="logd_alpha_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log dnorm
</p>

<hr>
<h2 id='logd_beta'>dgamma for beta</h2><span id='topic+logd_beta'></span>

<h3>Description</h3>

<p>dgamma for beta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logd_beta(b, beta_shape = 2, beta_scale = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logd_beta_+3A_b">b</code></td>
<td>
<p>current beta</p>
</td></tr>
<tr><td><code id="logd_beta_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="logd_beta_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log dgamma
</p>

<hr>
<h2 id='logd_gamma'>dgamma for gamma</h2><span id='topic+logd_gamma'></span>

<h3>Description</h3>

<p>dgamma for gamma
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logd_gamma(g, gamma_shape = 2, gamma_scale = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logd_gamma_+3A_g">g</code></td>
<td>
<p>current gamma</p>
</td></tr>
<tr><td><code id="logd_gamma_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="logd_gamma_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log dgamma
</p>

<hr>
<h2 id='loglik'>Log likelihood calculation</h2><span id='topic+loglik'></span>

<h3>Description</h3>

<p>Log likelihood calculation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loglik(params, lbf_mat, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loglik_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="loglik_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="loglik_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="loglik_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="loglik_+3A_covar">covar</code></td>
<td>
<p>logical: should the covariate inflormation be used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logpost flog of the posteriors
</p>

<hr>
<h2 id='logpost'>Log posterior calculation</h2><span id='topic+logpost'></span>

<h3>Description</h3>

<p>Log posterior calculation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logpost(params, lbf_mat, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logpost_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="logpost_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="logpost_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="logpost_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="logpost_+3A_covar">covar</code></td>
<td>
<p>logical: should the covariate inflormation be used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logpost flog of the posteriors
</p>

<hr>
<h2 id='logpriors'>Calculate log priors</h2><span id='topic+logpriors'></span>

<h3>Description</h3>

<p>Calculate log priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logpriors(
  params,
  covar = FALSE,
  alpha_mean = -10,
  alpha_sd = 0.5,
  beta_shape = 2,
  beta_scale = 2,
  gamma_shape = 2,
  gamma_scale = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logpriors_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="logpriors_+3A_covar">covar</code></td>
<td>
<p>logical: Should the covariate inflormation be used? default: False</p>
</td></tr>
<tr><td><code id="logpriors_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="logpriors_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
<tr><td><code id="logpriors_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="logpriors_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
<tr><td><code id="logpriors_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="logpriors_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log priors
</p>

<hr>
<h2 id='logsum'>logsum</h2><span id='topic+logsum'></span>

<h3>Description</h3>

<p>Internal function, logsum
Function directly taken from coloc
This function calculates the log of the sum of the exponentiated
logs taking out the max, i.e. insuring that the sum is not Inf
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logsum(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logsum_+3A_x">x</code></td>
<td>
<p>numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>max(x) + log(sum(exp(x - max(x))))
</p>

<hr>
<h2 id='logsumexp'>Log sum</h2><span id='topic+logsumexp'></span>

<h3>Description</h3>

<p>Log sum
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logsumexp(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logsumexp_+3A_x">x</code></td>
<td>
<p>vector of log scale values to be added</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log sum of input
</p>

<hr>
<h2 id='metrop_run'>Run the hierarchical mcmc model to infer priors</h2><span id='topic+metrop_run'></span>

<h3>Description</h3>

<p>Run the hierarchical mcmc model to infer priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metrop_run(
  lbf_mat,
  nsnps,
  covar_vec,
  covar = FALSE,
  nits = 10000L,
  thin = 1L,
  alpha_mean = -10,
  alpha_sd = 0.5,
  beta_shape = 2,
  beta_scale = 2,
  gamma_shape = 2,
  gamma_scale = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metrop_run_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_covar">covar</code></td>
<td>
<p>logical: Should the covariate inflormation be used? default: False</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_nits">nits</code></td>
<td>
<p>Number of iterations run in mcmc</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_thin">thin</code></td>
<td>
<p>thinning</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="metrop_run_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list of log likelihood (ll) and parameters: alpha, beta and gamma
</p>

<hr>
<h2 id='multitrait.simplify'>Simplifying the output obtained from <code>cophe.multitrait</code>, <code>cophe.single</code> or <code>cophe.susie</code></h2><span id='topic+multitrait.simplify'></span>

<h3>Description</h3>

<p>Simplifying the output obtained from <code>cophe.multitrait</code>, <code>cophe.single</code> or <code>cophe.susie</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multitrait.simplify(multi.dat, only_BF = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multitrait.simplify_+3A_multi.dat">multi.dat</code></td>
<td>
<p>output obtained from <code>cophe.multitrait</code>, <code>cophe.single</code> or <code>cophe.susie</code></p>
</td></tr>
<tr><td><code id="multitrait.simplify_+3A_only_bf">only_BF</code></td>
<td>
<p>return only bayes factors and not posterior probabilities (default=FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with posterior probabilties of Hn, Hc and Ha
</p>

<hr>
<h2 id='pars_init'>Initiate parameters alpha, beta and gamma</h2><span id='topic+pars_init'></span>

<h3>Description</h3>

<p>Initiate parameters alpha, beta and gamma
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pars_init(
  covar = FALSE,
  alpha_mean = -10,
  alpha_sd = 0.5,
  beta_shape = 2,
  beta_scale = 2,
  gamma_shape = 2,
  gamma_scale = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pars_init_+3A_covar">covar</code></td>
<td>
<p>logical: Should the covariate inflormation be used? default: False</p>
</td></tr>
<tr><td><code id="pars_init_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="pars_init_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
<tr><td><code id="pars_init_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="pars_init_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
<tr><td><code id="pars_init_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="pars_init_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>params <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code>
</p>

<hr>
<h2 id='pars2pik'>Conversion of parameters alpha, beta and gamma to pnk, pak and pck</h2><span id='topic+pars2pik'></span>

<h3>Description</h3>

<p>Conversion of parameters alpha, beta and gamma to pnk, pak and pck
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pars2pik(params, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pars2pik_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="pars2pik_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="pars2pik_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="pars2pik_+3A_covar">covar</code></td>
<td>
<p>logical: should the covariate information be used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pik matrix of priors: pnk, pak and pck
</p>

<hr>
<h2 id='per.snp.priors'>per.snp.priors</h2><span id='topic+per.snp.priors'></span>

<h3>Description</h3>

<p>Estimate per snp priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>per.snp.priors(
  nsnps,
  pa = 3.82e-05,
  pc = 0.00182,
  p1 = NULL,
  p2 = NULL,
  p12 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="per.snp.priors_+3A_nsnps">nsnps</code></td>
<td>
<p>number of SNPs</p>
</td></tr>
<tr><td><code id="per.snp.priors_+3A_pa">pa</code></td>
<td>
<p>prior probability that a non-query variant is causally associated with the query trait (cophescan prior), default 3.82e-5</p>
</td></tr>
<tr><td><code id="per.snp.priors_+3A_pc">pc</code></td>
<td>
<p>prior probability that the query variant is causally associated with the query trait (cophescan prior), default 1.82e-3 (cophescan prior)</p>
</td></tr>
<tr><td><code id="per.snp.priors_+3A_p1">p1</code></td>
<td>
<p>prior probability a SNP is associated with trait 1, (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code>; use p1, p2, p12 only when pa and pc are unavailable (See vignettes)</p>
</td></tr>
<tr><td><code id="per.snp.priors_+3A_p2">p2</code></td>
<td>
<p>prior probability a SNP is associated with trait 2,  (coloc prior), pa derived by using <code class="reqn">pa = p2</code></p>
</td></tr>
<tr><td><code id="per.snp.priors_+3A_p12">p12</code></td>
<td>
<p>prior probability a SNP is associated with both traits,  (coloc prior), pc derived by using <code class="reqn">pc =  p12/p1+p12</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>priors at the query variant
</p>


<h3>Author(s)</h3>

<p>Ichcha Manipur
</p>

<hr>
<h2 id='piks'>List of priors: pn, pa and pc over all iterations</h2><span id='topic+piks'></span>

<h3>Description</h3>

<p>List of priors: pn, pa and pc over all iterations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>piks(params, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="piks_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="piks_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="piks_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="piks_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of priors (len: iterations): pnk, pak and pck
</p>

<hr>
<h2 id='plot_trait_manhat'>Plot region Manhattan for a trait highlighting the queried variant</h2><span id='topic+plot_trait_manhat'></span>

<h3>Description</h3>

<p>Plot region Manhattan for a trait highlighting the queried variant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_trait_manhat(trait.dat, querysnpid, alt.snpid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_trait_manhat_+3A_trait.dat">trait.dat</code></td>
<td>
<p>dataset used as input for running cophescan</p>
</td></tr>
<tr><td><code id="plot_trait_manhat_+3A_querysnpid">querysnpid</code></td>
<td>
<p>the id of the causal variant as present in trait.dat$snp, plotted in red</p>
</td></tr>
<tr><td><code id="plot_trait_manhat_+3A_alt.snpid">alt.snpid</code></td>
<td>
<p>the id of the other variants as a vector to be plotted, plotted in blue</p>
</td></tr>
</table>


<h3>Value</h3>

<p>regional manhattan plot
</p>

<hr>
<h2 id='posterior_prob'>List of posterior probabilities: Hn, Ha and Hc over all iterations</h2><span id='topic+posterior_prob'></span>

<h3>Description</h3>

<p>List of posterior probabilities: Hn, Ha and Hc over all iterations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posterior_prob(params, lbf_mat, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posterior_prob_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="posterior_prob_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="posterior_prob_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="posterior_prob_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="posterior_prob_+3A_covar">covar</code></td>
<td>
<p>logical: was the covariate inflormation  used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of posterior probabilties (len: iterations): Hn, Ha and Hc
</p>

<hr>
<h2 id='prepare_plot_data'>Prepare data for plotting</h2><span id='topic+prepare_plot_data'></span>

<h3>Description</h3>

<p>Prepare data for plotting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_plot_data(
  multi.dat,
  querysnpid,
  query_trait_names,
  thresh_Ha = 0.5,
  thresh_Hc = 0.5,
  hmp = FALSE,
  cophe.plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_plot_data_+3A_multi.dat">multi.dat</code></td>
<td>
<p>multi trait cophescan results returned from cophe.multitrait or multitrait.simplify</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_querysnpid">querysnpid</code></td>
<td>
<p>query variant</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_query_trait_names">query_trait_names</code></td>
<td>
<p>vector of names of the query traits</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_thresh_ha">thresh_Ha</code></td>
<td>
<p>Ha threshold to be displayed</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_thresh_hc">thresh_Hc</code></td>
<td>
<p>Hc threshold to be displayed</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_hmp">hmp</code></td>
<td>
<p>return for heatmap</p>
</td></tr>
<tr><td><code id="prepare_plot_data_+3A_cophe.plot">cophe.plot</code></td>
<td>
<p>default: TRUE, return for <code>cophe_plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot list
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe_plot">cophe_plot</a></code>, <code><a href="#topic+cophe.susie">cophe.susie</a></code>, <code><a href="#topic+cophe.multitrait">cophe.multitrait</a></code>,  <code><a href="#topic+multitrait.simplify">multitrait.simplify</a></code>
default NULL
</p>

<hr>
<h2 id='propose'>Proposal distribution</h2><span id='topic+propose'></span>

<h3>Description</h3>

<p>Proposal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>propose(params, propsd = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propose_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="propose_+3A_propsd">propsd</code></td>
<td>
<p>Standard deviation for the proposal</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector : proposal
</p>

<hr>
<h2 id='run_metrop_priors'>Run the hierarchical Metropolis Hastings model to infer priors</h2><span id='topic+run_metrop_priors'></span>

<h3>Description</h3>

<p>Run the hierarchical Metropolis Hastings model to infer priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_metrop_priors(
  multi.dat,
  covar = FALSE,
  covar_vec = NULL,
  is_covar_categorical = FALSE,
  nits = 10000,
  thin = 1,
  posterior = FALSE,
  avg_pik = TRUE,
  avg_posterior = TRUE,
  pik = FALSE,
  alpha_mean = -10,
  alpha_sd = 0.5,
  beta_shape = 2,
  beta_scale = 2,
  gamma_shape = 2,
  gamma_scale = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_metrop_priors_+3A_multi.dat">multi.dat</code></td>
<td>
<p>matrix of bf values, rows=traits, named columns=(&quot;lBF.Ha&quot;,&quot;lBF.Hc&quot;,&quot;nsnps&quot;)</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_covar">covar</code></td>
<td>
<p>whether to include covariates</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_covar_vec">covar_vec</code></td>
<td>
<p>vector of covariates</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_is_covar_categorical">is_covar_categorical</code></td>
<td>
<p>only two categories supported (default=FALSE) - Experimental</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_nits">nits</code></td>
<td>
<p>number of iterations</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_thin">thin</code></td>
<td>
<p>burnin</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_posterior">posterior</code></td>
<td>
<p>default: FALSE, estimate posterior probabilities of the hypotheses</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_avg_pik">avg_pik</code></td>
<td>
<p>default: FALSE, estimate the average of the pik</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_avg_posterior">avg_posterior</code></td>
<td>
<p>default: FALSE, estimate the average of the posterior probabilities of the hypotheses</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_pik">pik</code></td>
<td>
<p>default: FALSE, inferred prior probabilities</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="run_metrop_priors_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the posterior distribution of the parameters alpha, beta, gamma (if covariate included) and the loglikelihood
</p>
<p>if avg_posterior=TRUE matrix with average of all the posterior probabilities of Hn, Ha and Hc
</p>
<p>if avg_pik=TRUE matrix with average of all the priors: pn, pa and pc
</p>
<p>data, nits and thin contain the input data, number of iterations and burnin respectively specified for the hierarchical model
</p>

<hr>
<h2 id='sample_alpha'>sample alpha</h2><span id='topic+sample_alpha'></span>

<h3>Description</h3>

<p>sample alpha
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_alpha(alpha_mean = -10, alpha_sd = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_alpha_+3A_alpha_mean">alpha_mean</code></td>
<td>
<p>prior for the mean of  alpha</p>
</td></tr>
<tr><td><code id="sample_alpha_+3A_alpha_sd">alpha_sd</code></td>
<td>
<p>prior for the standard deviation of  alpha</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample from rnorm for <code class="reqn">\alpha</code>
</p>

<hr>
<h2 id='sample_beta'>sample beta</h2><span id='topic+sample_beta'></span>

<h3>Description</h3>

<p>sample beta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_beta(beta_shape = 2, beta_scale = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_beta_+3A_beta_shape">beta_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of beta</p>
</td></tr>
<tr><td><code id="sample_beta_+3A_beta_scale">beta_scale</code></td>
<td>
<p>prior for the scale of beta</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample from rgamma for <code class="reqn">\beta</code>
</p>

<hr>
<h2 id='sample_gamma'>sample gamma</h2><span id='topic+sample_gamma'></span>

<h3>Description</h3>

<p>sample gamma
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_gamma(gamma_shape = 2, gamma_scale = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_gamma_+3A_gamma_shape">gamma_shape</code></td>
<td>
<p>prior for the shape (gamma distibution) of gamma</p>
</td></tr>
<tr><td><code id="sample_gamma_+3A_gamma_scale">gamma_scale</code></td>
<td>
<p>prior for the scale of gamma</p>
</td></tr>
</table>


<h3>Value</h3>

<p>sample from rgamma for <code class="reqn">\gamma</code>
</p>

<hr>
<h2 id='summary.cophe'>print the summary of results from cophescan single or susie</h2><span id='topic+summary.cophe'></span>

<h3>Description</h3>

<p>print the summary of results from cophescan single or susie
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cophe'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.cophe_+3A_object">object</code></td>
<td>
<p>Result from either <code>cophe.susie</code> or <code>cophe.single</code></p>
</td></tr>
<tr><td><code id="summary.cophe_+3A_...">...</code></td>
<td>
<p>additional arguments affecting the summary produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log bayes and posterior probabilities
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cophe.single">cophe.single</a></code>, <code><a href="#topic+cophe.susie">cophe.susie</a></code>
</p>

<hr>
<h2 id='target'>Target distribution</h2><span id='topic+target'></span>

<h3>Description</h3>

<p>Target distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>target(params, lbf_mat, nsnps, covar_vec, covar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="target_+3A_params">params</code></td>
<td>
<p>Vector of parameters: <code class="reqn">\alpha</code>, <code class="reqn">\beta</code> and <code class="reqn">\gamma</code></p>
</td></tr>
<tr><td><code id="target_+3A_lbf_mat">lbf_mat</code></td>
<td>
<p>matrix of log bayes factors: lBF.Ha and lBF.Hc</p>
</td></tr>
<tr><td><code id="target_+3A_nsnps">nsnps</code></td>
<td>
<p>number of snps</p>
</td></tr>
<tr><td><code id="target_+3A_covar_vec">covar_vec</code></td>
<td>
<p>Vector of the covariate</p>
</td></tr>
<tr><td><code id="target_+3A_covar">covar</code></td>
<td>
<p>logical: Should the covariate inflormation be used? default: False</p>
</td></tr>
</table>


<h3>Value</h3>

<p>target
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
