<!DOCTYPE html><html lang="en"><head><title>Help for package CopSens</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CopSens}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bcalibrate'><p>Calibration for Binary Outcomes</p></a></li>
<li><a href='#cal_rv'><p>Calculate Robustness Value When Executing Worstcase Calibration</p></a></li>
<li><a href='#cali_mean_ybinary_algm'><p>Calibrate Estimate of Intervention Mean for Binary Outcome</p></a></li>
<li><a href='#GaussianT_BinaryY'><p>Dataset with Gaussian Treatments and Binary Outcomes</p></a></li>
<li><a href='#GaussianT_GaussianY'><p>Dataset with Gaussian Treatments and Outcomes</p></a></li>
<li><a href='#gcalibrate'><p>Calibration for Gaussian Outcomes</p></a></li>
<li><a href='#get_opt_gamma'><p>Obtain Optimized Sensitivity Parameters Using Multivariate Calibration Criterion</p></a></li>
<li><a href='#mice_est_nulltr'><p>Estimates of genes' effects on mice body weight using null treatments approach from Miao et al. (2020)</p></a></li>
<li><a href='#micedata'><p>Body weight and gene expressions of 287 mice</p></a></li>
<li><a href='#plot_estimates'><p>Visualize Estimates of Treatment Effects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Copula-Based Sensitivity Analysis for Observational Causal
Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the copula-based sensitivity analysis method, 
  as discussed in Copula-based Sensitivity Analysis for Multi-Treatment Causal Inference with Unobserved Confounding
  &lt;<a href="https://doi.org/10.48550/arXiv.2102.09412">doi:10.48550/arXiv.2102.09412</a>&gt;, with Gaussian copula adopted in particular.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/JiajingZ/CopSens">https://github.com/JiajingZ/CopSens</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>pcaMethods (&ge; 1.78.0), CVXR (&ge; 1.0), MASS (&ge; 7.3-53),
ggplot2 (&ge; 3.3.3), stats (&ge; 4.0.3), dplyr (&ge; 1.0.3), tidyr
(&ge; 1.1.2), tidyverse (&ge; 1.3.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-11 06:23:54 UTC; jiajing</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiajing Zheng [aut, cre],
  Alexander Franks [aut],
  Alex D'Amour [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiajing Zheng &lt;carolinez6850@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-12 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bcalibrate'>Calibration for Binary Outcomes</h2><span id='topic+bcalibrate'></span>

<h3>Description</h3>

<p>Calibrates the naive estimates to account for unobserved confounding when outcome
variables are binary. The calibration can be done with user-specific sensitivity parameter or
with our pre-provided calibration methods, the worst-case calibration for a single contrast
or multivariate calibration for multiple contrasts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcalibrate(
  y,
  tr,
  t,
  gamma,
  R2 = NULL,
  mu_y_t = NULL,
  mu_u_tr = NULL,
  mu_u_t = NULL,
  cov_u_t = NULL,
  nU = NULL,
  nsim = 4000,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bcalibrate_+3A_y">y</code></td>
<td>
<p><code>data.frame</code>, <code>matrix</code> or <code>vector</code>. Binary outcome variable.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_tr">tr</code></td>
<td>
<p><code>data.frame</code>. Treatment variables with rows corresponding to observations and columns
to variables.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_t">t</code></td>
<td>
<p><code>data.frame</code>. Treatment arms of interest.
May contain a single or multiple treatments in rows.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_gamma">gamma</code></td>
<td>
<p>a vector specifying the direction of sensitivity parameters.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_r2">R2</code></td>
<td>
<p>an optional scalar or vector specifying the proportion of residual variance in outcome given the
treatment that can be explained by confounders, which determines the magnitude of sensitivity parameters.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_mu_y_t">mu_y_t</code></td>
<td>
<p>an optional scalar or vector that contains naive estimates of treatment effects
ignoring confounding.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_mu_u_tr">mu_u_tr</code></td>
<td>
<p>an optional matrix of conditional confounder means for all observed treatments
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_mu_u_t">mu_u_t</code></td>
<td>
<p>an optional matrix of conditional confounder means for treatments of interest
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_cov_u_t">cov_u_t</code></td>
<td>
<p>an optional covariance matrix of confounders conditional on treatments.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_nu">nU</code></td>
<td>
<p>Number of latent confounders to consider.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_nsim">nsim</code></td>
<td>
<p>an optional scalar specifying the number of sample draws.</p>
</td></tr>
<tr><td><code id="bcalibrate_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code>kEstimate</code>or <code>pca</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with naive and calibrated estimates of population average outcome receiving
treatment <code>t</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load the example data #
y &lt;- GaussianT_BinaryY$y
tr &lt;- subset(GaussianT_BinaryY, select = -c(y))
t1 &lt;- tr[1:5,]
t2 &lt;- rep(0, times = ncol(tr))
# calibration #
est_b &lt;- bcalibrate(y = y, tr = tr, t = rbind(t1, t2),
                    nU = 3, gamma = c(1.27, -0.28, 0),
                    R2 = c(0.2, 0.7))
est_b_rr &lt;- list(est_df = est_b$est_df[1:5,] / as.numeric(est_b$est_df[6,]),
                 R2 = c(0.2, 0.7))
plot_estimates(est_b_rr)

</code></pre>

<hr>
<h2 id='cal_rv'>Calculate Robustness Value When Executing Worstcase Calibration</h2><span id='topic+cal_rv'></span>

<h3>Description</h3>

<p>Calculate Robustness Value When Executing Worstcase Calibration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cal_rv(
  y,
  tr,
  t1,
  t2,
  mu_y_dt = NULL,
  sigma_y_t = NULL,
  mu_u_dt = NULL,
  cov_u_t = NULL,
  nU = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cal_rv_+3A_y">y</code></td>
<td>
<p><code>data.frame</code>, <code>matrix</code> or <code>vector</code>. Gaussian outcome variable.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_tr">tr</code></td>
<td>
<p><code>data.frame</code>. Treatment variables with rows corresponding to observations and columns
to variables.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_t1">t1</code></td>
<td>
<p><code>data.frame</code>. First treatment arms of interest.
May contain a single or multiple treatments in rows.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_t2">t2</code></td>
<td>
<p><code>data.frame</code>. Second treatment arms of interest,
which has same number of row as <code>t1</code>.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_mu_y_dt">mu_y_dt</code></td>
<td>
<p>an optional scalar or vector that contains naive estimates of treatment effects
ignoring confounding.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_sigma_y_t">sigma_y_t</code></td>
<td>
<p>an optional scalar of the standard deviation of outcome conditional on treatments.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_mu_u_dt">mu_u_dt</code></td>
<td>
<p>an optional matrix of difference in conditional confounder means, <code class="reqn">E(U \mid t1) - E(U \mid t2)</code>,
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_cov_u_t">cov_u_t</code></td>
<td>
<p>an optional covariance matrix of confounders conditional on treatments.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_nu">nU</code></td>
<td>
<p>Number of latent confounders to consider.</p>
</td></tr>
<tr><td><code id="cal_rv_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code>kEstimate</code>, <code>pca</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>numeric vector</code> with elements being the robustness value or <code>NA</code> if the ignorance region doesn't
contains 0 for each contrast of interest.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load the example data #
y &lt;- GaussianT_GaussianY$y
tr &lt;- subset(GaussianT_GaussianY, select = -c(y))
# calculate robustness value #
cal_rv(y = y, tr = tr, t1 = tr[1:2,], t2 = tr[3:4,])

</code></pre>

<hr>
<h2 id='cali_mean_ybinary_algm'>Calibrate Estimate of Intervention Mean for Binary Outcome</h2><span id='topic+cali_mean_ybinary_algm'></span>

<h3>Description</h3>

<p>Calibrate Estimate of Intervention Mean for Binary Outcome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cali_mean_ybinary_algm(i, gamma, mu_u_tr, mu_u_t, mu_y_t, nsim = 4000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cali_mean_ybinary_algm_+3A_i">i</code></td>
<td>
<p>Observation index.</p>
</td></tr>
<tr><td><code id="cali_mean_ybinary_algm_+3A_gamma">gamma</code></td>
<td>
<p>Scalar or vector specifying the sensitivity parameters.</p>
</td></tr>
<tr><td><code id="cali_mean_ybinary_algm_+3A_mu_u_tr">mu_u_tr</code></td>
<td>
<p>Matrix of conditional confounder means for all observed treatments
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="cali_mean_ybinary_algm_+3A_mu_u_t">mu_u_t</code></td>
<td>
<p>Matrix of conditional confounder means for treatments of interest
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="cali_mean_ybinary_algm_+3A_mu_y_t">mu_y_t</code></td>
<td>
<p>Scalar or vector that contains naive estimates of treatment effects
ignoring confounding.</p>
</td></tr>
<tr><td><code id="cali_mean_ybinary_algm_+3A_nsim">nsim</code></td>
<td>
<p>Number of simulation sample draws.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scalar of calibrated intervention mean.
</p>

<hr>
<h2 id='GaussianT_BinaryY'>Dataset with Gaussian Treatments and Binary Outcomes</h2><span id='topic+GaussianT_BinaryY'></span>

<h3>Description</h3>

<p>A dataset containing Gaussian treatments and binary outcomes of 10,000 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GaussianT_BinaryY
</code></pre>


<h3>Format</h3>

<p>A data frame with eleven variables: one binary outcome, <code>y</code>, and ten Gaussian
treatments, <code>t1</code>, <code>t2</code>, ..., <code>t10</code>.
</p>


<h3>Source</h3>

<p>For data generating process, see <code>data-raw/Data_Generation.R</code>.
</p>

<hr>
<h2 id='GaussianT_GaussianY'>Dataset with Gaussian Treatments and Outcomes</h2><span id='topic+GaussianT_GaussianY'></span>

<h3>Description</h3>

<p>A dataset containing Gaussian treatments and outcomes of 10,000 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GaussianT_GaussianY
</code></pre>


<h3>Format</h3>

<p>A data frame with eleven variables: one Gaussian outcome, <code>y</code>, and ten Gaussian
treatments, <code>t1</code>, <code>t2</code>, ..., <code>t10</code>.
</p>


<h3>Source</h3>

<p>For data generating process, see <code>data-raw/Data_Generation.R</code>.
</p>

<hr>
<h2 id='gcalibrate'>Calibration for Gaussian Outcomes</h2><span id='topic+gcalibrate'></span>

<h3>Description</h3>

<p>Calibrates the naive estimates to account for unobserved confounding when outcome
variables are Gaussian. The calibration can be done with user-specific sensitivity parameters or
with our pre-provided calibration methods, the worst-case calibration for a single contrast
or multivariate calibration for multiple contrasts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gcalibrate(
  y,
  tr,
  t1,
  t2,
  calitype = c("worstcase", "multicali", "null"),
  mu_y_dt = NULL,
  sigma_y_t = NULL,
  mu_u_dt = NULL,
  cov_u_t = NULL,
  nU = NULL,
  R2 = 1,
  gamma = NULL,
  R2_constr = 1,
  nc_index = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gcalibrate_+3A_y">y</code></td>
<td>
<p><code>data.frame</code>, <code>matrix</code> or <code>vector</code>. Gaussian outcome variable.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_tr">tr</code></td>
<td>
<p><code>data.frame</code>. Treatment variables with rows corresponding to observations and columns
to variables.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_t1">t1</code></td>
<td>
<p><code>data.frame</code>. First treatment arms of interest.
May contain a single or multiple treatments in rows.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_t2">t2</code></td>
<td>
<p><code>data.frame</code>. Second treatment arms of interest,
which has same number of row as <code>t1</code>.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_calitype">calitype</code></td>
<td>
<p>character. The calibration method to be applied. Can be one of: <br />
&quot;worstcase&quot; - apply worst-case calibration when considering a single contrast. <br />
&quot;multicali&quot; - apply mutlivariate calibration when considering multiple contrasts.<br />
&quot;null&quot; - apply calibration with user-specified sensitivity parameter, <code class="reqn">\gamma</code>. <br /></p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_mu_y_dt">mu_y_dt</code></td>
<td>
<p>an optional scalar or vector that contains naive estimates of treatment effects
ignoring confounding.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_sigma_y_t">sigma_y_t</code></td>
<td>
<p>an optional scalar of the standard deviation of outcome conditional on treatments.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_mu_u_dt">mu_u_dt</code></td>
<td>
<p>an optional matrix of difference in conditional confounder means, <code class="reqn">E(U \mid t1) - E(U \mid t2)</code>,
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_cov_u_t">cov_u_t</code></td>
<td>
<p>an optional covariance matrix of confounders conditional on treatments.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_nu">nU</code></td>
<td>
<p>Number of latent confounders to consider.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_r2">R2</code></td>
<td>
<p>an optional scalar or vector specifying the proportion of residual variance in outcome given the
treatment that can be explained by confounders.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_gamma">gamma</code></td>
<td>
<p>sensitivity parameter vector. Must be given when <code>calitype = "null"</code>.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_r2_constr">R2_constr</code></td>
<td>
<p>an optional scalar or vector specifying the upper limit constraint on <code class="reqn">R^2</code> .
By default, <code>R2_constr = 1</code>.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_nc_index">nc_index</code></td>
<td>
<p>an optional vector containing indexes of negative control treatments. If not <code>NULL</code>,
worstcase calibration will be executed with constraints imposed by negative control treatments.</p>
</td></tr>
<tr><td><code id="gcalibrate_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code>kEstimate</code>, <code>pca</code> or
<code><a href="#topic+get_opt_gamma">get_opt_gamma</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>gcalibrate</code> returns a list containing the following components:
</p>

<dl>
<dt><code>est_df</code></dt><dd><p>a <code>data.frame</code> with naive and calibrated estimates of average treatment effects.</p>
</dd>
<dt><code>R2</code></dt><dd><p>a vector of <code class="reqn">R^2</code> with elements corresponding to columns of <code>est_df</code>.</p>
</dd>
<dt><code>gamma</code></dt><dd><p>a matrix returned when <code>calitype = "multicali"</code> or <code>"worstcase"</code>.
If <code>calitype = "multicali"</code>, optimized gamma are in columns,
respectively resulting in estimates in columns of <code>est_df</code>.
If <code>calitype = "worstcase"</code>, gamma are in rows,
which respectively lead to the worstcase ignorance region with <code class="reqn">R^2=1</code> for each contrast of interest.</p>
</dd>
<dt><code>rv</code></dt><dd><p>a <code>numeric vector</code> returned when <code>calitype = "worstcase"</code>,
with elements being the robustness value or <code>NA</code> if the ignorance region doesn't
contains 0 for each contrast of interest.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># load the example data #
y &lt;- GaussianT_GaussianY$y
tr &lt;- subset(GaussianT_GaussianY, select = -c(y))

# worst-case calibration #
t1 &lt;- data.frame(diag(ncol(tr)))
t2 &lt;- data.frame(matrix(0, nrow = ncol(tr), ncol = ncol(tr)))
colnames(t1) = colnames(t2) &lt;- colnames(tr)
est_g1 &lt;- gcalibrate(y = y, tr = tr, t1 = t1, t2 = t2, nU = 3,
                     calitype = "worstcase", R2 = c(0.3, 1))
plot_estimates(est_g1)
# with negative conotrls #
est_g1_nc &lt;- gcalibrate(y = y, tr = tr, t1 = t1, t2 = t2, nU = 3,
                        calitype = "worstcase", R2 = c(0.3, 1), nc_index = c(3, 6))
plot_estimates(est_g1_nc)


# multivariate calibration #
est_g2 &lt;- gcalibrate(y = y, tr = tr, t1 = tr[1:10,], t2 = tr[11:20,], nU = 3,
                     calitype = "multicali", R2_constr = c(1, 0.15))
plot_estimates(est_g2)

# user-specified calibration #
est_g3 &lt;- gcalibrate(y = y, tr = tr, t1 = tr[1:2,], t2 = tr[3:4,],
                     nU = 3, calitype = "null",
                     gamma = c(0.96, -0.29, 0), R2 = c(0.2, 0.6, 1))
plot_estimates(est_g3)
# apply gamma that maximizes the bias for the first contrast considered in est_g1 #
est_g4 &lt;- gcalibrate(y = y, tr = tr, t1 = tr[1:2,], t2 = tr[3:4,],
                     nU = 3, calitype = "null",
                     gamma = est_g1$gamma[1,], R2 = c(0.2, 0.6, 1))
plot_estimates(est_g4)

</code></pre>

<hr>
<h2 id='get_opt_gamma'>Obtain Optimized Sensitivity Parameters Using Multivariate Calibration Criterion</h2><span id='topic+get_opt_gamma'></span>

<h3>Description</h3>

<p>Obtain Optimized Sensitivity Parameters Using Multivariate Calibration Criterion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_opt_gamma(
  mu_y_dt,
  mu_u_dt,
  cov_u_t,
  sigma_y_t,
  R2_constr = 1,
  normtype = "L2",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_opt_gamma_+3A_mu_y_dt">mu_y_dt</code></td>
<td>
<p>Scalar or vector that contains naive estimates of treatment effects
ignoring confounding.</p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_mu_u_dt">mu_u_dt</code></td>
<td>
<p>Matrix of difference in conditional confounder means, <code class="reqn">E(U \mid t1) - E(U \mid t2)</code>,
with latent variables in columns.</p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_cov_u_t">cov_u_t</code></td>
<td>
<p>Covariance matrix of confounders conditional on treatments.</p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_sigma_y_t">sigma_y_t</code></td>
<td>
<p>Scalar of the standard deviation of outcome conditional on treatments.</p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_r2_constr">R2_constr</code></td>
<td>
<p>an optional scalar or vector specifying the upper limit constraint on <code class="reqn">R^2</code> .
By default, <code>R2_constr = 1</code>.</p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_normtype">normtype</code></td>
<td>
<p>character. Optional function <code>m</code> for the multivariate calibration criterion.
By default, the L2 norm will be applied.<br />
&quot;L1&quot; - apply the L1 norm, <code>sum(abs(x))</code>. <br />
&quot;L2&quot; - apply the L2 norm, <code>sqrt(sum(x^2))</code>.<br />
&quot;Inf&quot; - apply the infinity norm, <code>max(abs(x))</code>. <br /></p>
</td></tr>
<tr><td><code id="get_opt_gamma_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="base.html#topic+solve">solve</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Optimized sensitivity parameters.
</p>

<hr>
<h2 id='mice_est_nulltr'>Estimates of genes' effects on mice body weight using null treatments approach from Miao et al. (2020)</h2><span id='topic+mice_est_nulltr'></span>

<h3>Description</h3>

<p>The dataset consists of estimates of treatment effects of 17 genes, which are likely to affect mouse weight,
by using the null treatments approach from Miao et al. (2020),
assuming that at least half of the confounded treatments have no causal effect on the outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mice_est_nulltr
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 rows and 6 variables:
</p>

<dl>
<dt>esti</dt><dd><p>mean estimates of genes' treatment effects on mouse body weight</p>
</dd>
<dt>X2.5.</dt><dd><p>2.5% percentile of the estimates of genes' treatment effects on mouse body weight</p>
</dd>
<dt>X97.5.</dt><dd><p>97.5% percentile of the estimates of genes' treatment effects on mouse body weight</p>
</dd>
<dt>X5.</dt><dd><p>5% percentile of the estimates of genes' treatment effects on mouse body weight</p>
</dd>
<dt>X95.</dt><dd><p>95% percentile of the estimates of genes' treatment effects on mouse body weight</p>
</dd>
<dt>signif</dt><dd><p>significance</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://arxiv.org/abs/2011.04504">https://arxiv.org/abs/2011.04504</a>
</p>

<hr>
<h2 id='micedata'>Body weight and gene expressions of 287 mice</h2><span id='topic+micedata'></span>

<h3>Description</h3>

<p>A dataset are collected from 287 mice, including the body weight, 37 gene expressions,
and 5 single nucleotide polymorphisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>micedata
</code></pre>


<h3>Format</h3>

<p>A data frame with forty-three variables: the mice body weight, <code>y</code>,
5 single nucleotide polymorphisms, <code>rs3663003</code>, <code>rs4136518</code>, <code>rs3694833</code>,
<code>rs4231406</code>, <code>rs3661189</code>, and the rest are thirty-seven genes.
</p>


<h3>Source</h3>

<p><a href="https://arxiv.org/abs/2011.04504">https://arxiv.org/abs/2011.04504</a>
</p>

<hr>
<h2 id='plot_estimates'>Visualize Estimates of Treatment Effects</h2><span id='topic+plot_estimates'></span>

<h3>Description</h3>

<p>Visualize Estimates of Treatment Effects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_estimates(est, show_rv = TRUE, order = "naive", labels = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_estimates_+3A_est">est</code></td>
<td>
<p>an return object from <code><a href="#topic+gcalibrate">gcalibrate</a></code> or <code><a href="#topic+bcalibrate">bcalibrate</a></code>, or
<code>data.frame</code> containing estimates of treatment effects with estimates' type in columns and
contrasts of interest in rows.</p>
</td></tr>
<tr><td><code id="plot_estimates_+3A_show_rv">show_rv</code></td>
<td>
<p>logical. Whether robustness values should be printed in the plot or not?
Available only for the &quot;worstcase&quot; calibration.</p>
</td></tr>
<tr><td><code id="plot_estimates_+3A_order">order</code></td>
<td>
<p>character. The type of order used to plot treatment effects from left to right.
Can be one of the following:<br />
&quot;naive&quot; - order by the naive estimate from smallest to largest.
&quot;worstcase&quot; - place all treatments with negative robust effects on the left, with positive robust
effects on the right, and all sensitive ones in the middle. Within the negative robust group, order
treatments by the upper bound of the worst-case ignorance region from smallest to largest;
within the positive robust group, order treatments by the lower bound of the worst-case ignorance
region from smallest to largest; and within the sensitive group, order by  the naive estimate from
smallest to largest.</p>
</td></tr>
<tr><td><code id="plot_estimates_+3A_labels">labels</code></td>
<td>
<p>character. Labels of treatments.</p>
</td></tr>
<tr><td><code id="plot_estimates_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code>theme</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graph plotting ignorance regions of the causal estimands of interest.
</p>


<h3>Note</h3>

<p>For examples, please refer to <code><a href="#topic+bcalibrate">bcalibrate</a></code> or <code><a href="#topic+gcalibrate">gcalibrate</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
