<!DOCTYPE html><html><head><title>Help for package copyseparator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {copyseparator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#copy_assemble'><p>copy_assemble</p></a></li>
<li><a href='#copy_detect'><p>copy_detect</p></a></li>
<li><a href='#copy_separate'><p>copy_separate</p></a></li>
<li><a href='#copy_validate'><p>copy_validate</p></a></li>
<li><a href='#sep_assem'><p>sep_assem</p></a></li>
<li><a href='#subset_downsize'><p>subset_downsize</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Assembling Long Gene Copies from Short Read Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Lei Yang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lei Yang &lt;leiyangslu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Assembles two or more gene copies from short-read Next-Generation Sequencing data. Works best when there are only two gene copies and read length &gt;=250 base pairs. High and relatively even coverage are important.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/LeiYang-Fish/copyseparator">https://github.com/LeiYang-Fish/copyseparator</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/LeiYang-Fish/copyseparator/issues">https://github.com/LeiYang-Fish/copyseparator/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, seqinr, stringr, kmer, DECIPHER, beepr, Biostrings,
grDevices, doParallel, foreach, parallel</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-25 00:14:04 UTC; lyang1</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-25 00:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='copy_assemble'>copy_assemble</h2><span id='topic+copy_assemble'></span>

<h3>Description</h3>

<p>Assembles a small number of overlapping DNA sequences into their respective gene copies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copy_assemble(filename, copy_number, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="copy_assemble_+3A_filename">filename</code></td>
<td>
<p>A fasta alignment of a small number of overlapping DNA sequences (results from &quot;copy_separate&quot;) covering the entire length of the target gene. Check the alignment carefully before proceeding.</p>
</td></tr>
<tr><td><code id="copy_assemble_+3A_copy_number">copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the anticipated number of gene copies. Must be the same value as used for &quot;copy_separate&quot;.</p>
</td></tr>
<tr><td><code id="copy_assemble_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fasta alignment of the anticipated number of full-length gene copies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
copy_assemble("inst/extdata/combined_con.fasta",2,1)

## End(Not run)

</code></pre>

<hr>
<h2 id='copy_detect'>copy_detect</h2><span id='topic+copy_detect'></span>

<h3>Description</h3>

<p>Separates two or more gene copies from a single subset of short reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copy_detect(filename, copy_number, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="copy_detect_+3A_filename">filename</code></td>
<td>
<p>A fasta file contains short reads from a single subset generated by &quot;subset_downsize&quot;.</p>
</td></tr>
<tr><td><code id="copy_detect_+3A_copy_number">copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the anticipated number of gene copies in the input file.</p>
</td></tr>
<tr><td><code id="copy_detect_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fasta alignment of the anticipated number of gene copies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
copy_detect("inst/extdata/toysubset.fasta",2,1)

## End(Not run)

</code></pre>

<hr>
<h2 id='copy_separate'>copy_separate</h2><span id='topic+copy_separate'></span>

<h3>Description</h3>

<p>Separates two or more gene copies from short-read Next-Generation Sequencing data into a small number of overlapping DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copy_separate(
  filename,
  copy_number,
  read_length,
  overlap = 225,
  rare_read = 10,
  verbose = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="copy_separate_+3A_filename">filename</code></td>
<td>
<p>A fasta file contains thousands of short reads that have been mapped to a reference. The reference and reads that are not directly mapped to the reference need to be removed after mapping.</p>
</td></tr>
<tr><td><code id="copy_separate_+3A_copy_number">copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the anticipated number of gene copies in the input file.</p>
</td></tr>
<tr><td><code id="copy_separate_+3A_read_length">read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data. This method is designed for read length &gt;=250bp.</p>
</td></tr>
<tr><td><code id="copy_separate_+3A_overlap">overlap</code></td>
<td>
<p>An integer describing number of base pairs of overlap between adjacent subsets. More overlap means more subsets. Default 225.</p>
</td></tr>
<tr><td><code id="copy_separate_+3A_rare_read">rare_read</code></td>
<td>
<p>A positive integer. During clustering analyses, clusters with less than this number of reads will be ignored. Default 10.</p>
</td></tr>
<tr><td><code id="copy_separate_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fasta alignment of a small number of overlapping DNA sequences covering the entire length of the target gene. Gene copies can be assembled by reordering the alignment manually or use the function &quot;copy_assemble&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
copy_separate("inst/extdata/toydata.fasta",2,300,225,10,1)

## End(Not run)

</code></pre>

<hr>
<h2 id='copy_validate'>copy_validate</h2><span id='topic+copy_validate'></span>

<h3>Description</h3>

<p>A tool to help identify incorrectly assembled chimeric sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>copy_validate(filename, copy_number, read_length, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="copy_validate_+3A_filename">filename</code></td>
<td>
<p>A DNA alignment in fasta format that contains sequences of two or more gene copies (e.g. results from &quot;copy_assemble&quot;).</p>
</td></tr>
<tr><td><code id="copy_validate_+3A_copy_number">copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the number of gene copies in the input file.</p>
</td></tr>
<tr><td><code id="copy_validate_+3A_read_length">read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data.</p>
</td></tr>
<tr><td><code id="copy_validate_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A histogram in pdf format showing the relationships between the physical distance between neighboring variable sites and read length.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
copy_validate("inst/extdata/Final_two_copies.fasta",2,300,1)

## End(Not run)

</code></pre>

<hr>
<h2 id='sep_assem'>sep_assem</h2><span id='topic+sep_assem'></span>

<h3>Description</h3>

<p>Separates two or more gene copies from short-read Next-Generation Sequencing data into a small number of overlapping DNA sequences and assemble them into their respective gene copies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sep_assem(
  copy_number,
  read_length,
  overlap = 225,
  rare_read = 10,
  core_number = 1,
  verbose = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sep_assem_+3A_copy_number">copy_number</code></td>
<td>
<p>An integer (e.g. 2,3, or 4) giving the anticipated number of gene copies in the input file.</p>
</td></tr>
<tr><td><code id="sep_assem_+3A_read_length">read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data. This method is designed for read length &gt;=250bp.</p>
</td></tr>
<tr><td><code id="sep_assem_+3A_overlap">overlap</code></td>
<td>
<p>An integer describing number of base pairs of overlap between adjacent subsets. More overlap means more subsets. Default 225.</p>
</td></tr>
<tr><td><code id="sep_assem_+3A_rare_read">rare_read</code></td>
<td>
<p>A positive integer. During clustering analyses, clusters with less than this number of reads will be ignored. Default 10.</p>
</td></tr>
<tr><td><code id="sep_assem_+3A_core_number">core_number</code></td>
<td>
<p>An integer describing number of cores to use.</p>
</td></tr>
<tr><td><code id="sep_assem_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fasta alignment of the anticipated number of full-length gene copies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sep_assem(2,300,225,10,1,1) # all input fasta files in the working directory will be processed

## End(Not run)

</code></pre>

<hr>
<h2 id='subset_downsize'>subset_downsize</h2><span id='topic+subset_downsize'></span>

<h3>Description</h3>

<p>Subdivides the imported read alignment into subsets and then downsizes each subset by deleting those sequences that have too many gaps or missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_downsize(filename, read_length, overlap, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_downsize_+3A_filename">filename</code></td>
<td>
<p>A fasta file contains thousands of short reads that have been mapped to a reference. The reference and reads that are not directly mapped to the reference need to be removed after mapping.</p>
</td></tr>
<tr><td><code id="subset_downsize_+3A_read_length">read_length</code></td>
<td>
<p>An integer (e.g. 250, or 300) giving the read length of your Next-generation Sequencing data. This method is designed for read length &gt;=250bp.</p>
</td></tr>
<tr><td><code id="subset_downsize_+3A_overlap">overlap</code></td>
<td>
<p>An integer describing number of base pairs of overlap between adjacent subsets. More overlap means more subsets.</p>
</td></tr>
<tr><td><code id="subset_downsize_+3A_verbose">verbose</code></td>
<td>
<p>Turn on (verbose=1; default) or turn off (verbose=0) the output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number of overlapping subsets (before and after downsizing) of the input alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
subset_downsize("inst/extdata/toydata.fasta",300,225,1)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
