<!DOCTYPE html><html><head><title>Help for package CoreMicrobiomeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CoreMicrobiomeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CoreMicrobiome'><p>Identification of Core Microbiome</p></a></li>
<li><a href='#demo_md'><p>Arabidopsis thaliana - Metadata dataset</p></a></li>
<li><a href='#demo_otu'><p>Arabidopsis thaliana - OTU dataset</p></a></li>
<li><a href='#demo_tax'><p>Arabidopsis thaliana - Taxonomy dataset</p></a></li>
<li><a href='#group_bar_plots'><p>Grouped Bar Plots Based on Sample Size</p></a></li>
<li><a href='#significance'><p>Testing the Significance of the Identified Core Microbiome</p></a></li>
<li><a href='#stacked_bar_plots'><p>Stacked Bar Plots Based on Relative Abundance Data</p></a></li>
<li><a href='#visualize'><p>Visualizing the effect of minimum count on the core size</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Core Microbiome</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>fastmatch, vegan, SRS, edgeR, ggplot2, ggrepel, stats, plotly,
reshape2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohammad Samir Farooqi &lt;samirfarooqi8@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The Core Microbiome refers to the group of microorganisms that are consistently present in a particular environment, habitat, or host species. These microorganisms play a crucial role in the functioning and stability of that ecosystem. Identifying these microorganisms can contribute to the emerging field of personalized medicine. The 'CoreMicrobiomeR' is designed to facilitate the identification, statistical testing, and visualization of this group of microorganisms.This package offers three key functions to analyze and visualize microbial community data. This package has been developed based on the research papers published by Pereira et al.(2018) &lt;<a href="https://doi.org/10.1186%2Fs12864-018-4637-6">doi:10.1186/s12864-018-4637-6</a>&gt; and Beule L, Karlovsky P. (2020) &lt;<a href="https://doi.org/10.7717%2Fpeerj.9593">doi:10.7717/peerj.9593</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Author:</td>
<td>Sorna A M [aut],
  Mohammad Samir Farooqi [aut, cre],
  Dwijesh Chandra Mishra [aut],
  Krishna Kumar Chaturvedi [aut],
  Anu Sharma [aut],
  Prawin Arya [aut],
  Sudhir Srivastava [aut],
  Sharanbasappa [aut],
  Girish Kumar Jha [aut],
  Kabilan S [ctb]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-03 05:57:33 UTC; kabil</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-03 20:03:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CoreMicrobiome'>Identification of Core Microbiome</h2><span id='topic+CoreMicrobiome'></span>

<h3>Description</h3>

<p>This function provides a comprehensive pipeline for processing
OTU (Operational Taxonomic Unit) tables, taxonomic tables, and metadata tables.
It applies various filtering methods based on user-defined parameters to select
core OTUs and non-core OTUs calculates alpha and beta diversity measures.
The pipeline can be customized with different normalization methods and filtering
criteria. Taxa are ranked in descending according to the cumulative sum obtained.
This method assigns taxa to the core if they are in the top X% of reads. Any taxa which appears before some cutoff percentage is included in the core.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CoreMicrobiome(otu_table, tax_table, metadata_table, filter_type, ...,
method, beta_diversity_method, top_percentage)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CoreMicrobiome_+3A_otu_table">otu_table</code></td>
<td>
<p>A dataframe of OTUs where the first row is the OTU ID and column names refer to sites/sample names.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_tax_table">tax_table</code></td>
<td>
<p>A dataframe of taxonomies where the first row is the OTU ID and column names refer to taxonomic classification.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_metadata_table">metadata_table</code></td>
<td>
<p>A dataframe of sites/samples where the first row is the sites/sample names and column names refer to groups of samples.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_filter_type">filter_type</code></td>
<td>
<p>Filtering method type, includes &quot;abundance_fun_filter&quot;, &quot;occupancy_fun_filter&quot;, or &quot;combined_filter&quot;.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_...">...</code></td>
<td>
<p>Other parameters. These are ignored, except in filter_type = &quot;abundance_fun_filter&quot; which accepts min_count, prop, min_total_count parameter, and in filter_type = &quot;occupancy_fun_filter&quot; which accepts percent parameter, and also filter_type = &quot;combined_filter&quot; which accepts percent, min_count, prop, min_total_count parameters.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_method">method</code></td>
<td>
<p>Different normalization methods, includes &quot;rrarefy&quot;, &quot;srs&quot;, &quot;css&quot;, &quot;tmm&quot;, &quot;tmmwsp&quot;, &quot;rle&quot;, &quot;upperquartile&quot; or &quot;none&quot;. The default method is tmm.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_beta_diversity_method">beta_diversity_method</code></td>
<td>
<p>Different beta diversity methods, includes &quot;bray&quot;, &quot;jaccard&quot;, &quot;mountford&quot;. The default method is bray.</p>
</td></tr>
<tr><td><code id="CoreMicrobiome_+3A_top_percentage">top_percentage</code></td>
<td>
<p>Percentage used for Core OTUs identification and the default is 10 percent.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function gives the list  which consist of following results.
</p>
<p>'final_otu_table_bef_filter'  otu_table obtained after sorting according to the provided tax_table and metadata_table
</p>
<p>'filtered_md_table' metadata_table obtained after sorting according to the provided otu_table
</p>
<p>'final_otu_aft_filter' otu_table obtained after filtering according to the user defined filtering method
</p>
<p>'normalized_table' normalized_otu_table obtained after normalizing according to the user defined normalization method
</p>
<p>'alpha_diversity' Alpha diversity measures of the samples
</p>
<p>'beta_diversity' Beta diversity measures between the samples
</p>
<p>'core_otus' Core OTUs obtained
</p>
<p>'non_core_otus' Non Core OTUs obtained
</p>
<p>'core_otus_tax' Taxonomy of the obtained Core OTUs
</p>
<p>'core_otus_count_data' Original count data of the obtained core OTUs
</p>
<p>'core_otus_relative_abundance' Relative abundance data of the obtained core OTUs
</p>


<h3>References</h3>

<p>Pereira, M., Wallroth, M., Jonsson, V. et al. (2018). Comparison of normalization methods for the analysis of metagenomic gene abundance data. BMC Genomics 19, 274. &lt;doi:https://doi.org/10.1186/s12864-018-4637-6&gt;
</p>
<p>Beule L, Karlovsky P. (2020). Improved normalization of species count data in ecology by scaling with ranked subsampling (SRS): application to microbial communities. PeerJ 8:e9593.&lt;doi:https://doi.org/10.7717/peerj.9593&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To run input data
core_1 &lt;- CoreMicrobiome(
 otu_table = demo_otu,
 tax_table = demo_tax,
 metadata_table = demo_md,
 filter_type = "occupancy_fun_filter", #Or "abundance_fun_filter", Or "combined_filter"
 percent = 0.5,
 method = "css",  # Or "srs", "rrarefy", "tmm", "tmmwsp", "rle", "upperquartile", "none"
 beta_diversity_method = "jaccard",
 top_percentage = 10  # Adjust the percentage as needed for core/non-core OTUs
)

#To view the core otus obtained
core_1[["core_otus"]]
#To view the taxonomy of the obtained core otus
core_1[["core_otus_tax"]]
</code></pre>

<hr>
<h2 id='demo_md'>Arabidopsis thaliana - Metadata dataset</h2><span id='topic+demo_md'></span>

<h3>Description</h3>

<p>This dataset was given by Lundberg et al., (2012). The metadata table contains
additional information about each sample (root) included in the study. It
typically includes details about the experimental conditions, environmental
factors, sample genotype, location, and other relevant contextual information.
Metadata is crucial for linking the microbial community data to specific
experimental variables and understanding how the root microbiome might vary in
response to different factors.
The original dataset contains 1049 samples(rows) and factors like soil type,
genotype, treatment, developmental stage and replication information in columns
for each particular sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_md
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 103 rows and 6 columns.
</p>


<h3>Details</h3>

<p>Here only the portion of the dataset is taken for running the functions.
The dataset contains 103 rows and 6 columns.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1038/nature11237">doi:10.1038/nature11237</a>
</p>

<hr>
<h2 id='demo_otu'>Arabidopsis thaliana - OTU dataset</h2><span id='topic+demo_otu'></span>

<h3>Description</h3>

<p>This dataset was given by Lundberg et al., (2012). The OTU table is a central
part of the data set. It represents the abundance or presence/absence of
different microbial taxa (operational taxonomic units) in the root samples of
Arabidopsis thaliana. Each column in the OTU table corresponds to a specific
sample (root) from the study, and each row represents a different OTU, which
could be a species or a group of closely related organisms. The table contains
numerical values representing the count of each OTU in the corresponding samples.
The original dataset contains 18783 rows of OTUs and 1439 samples(columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_otu
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 188 rows and 1440 columns.
</p>


<h3>Details</h3>

<p>Here only the portion of the dataset is taken for running the functions.
</p>
<p>The dataset contains 188 rows and 1440 columns.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1038/nature11237">doi:10.1038/nature11237</a>
</p>

<hr>
<h2 id='demo_tax'>Arabidopsis thaliana - Taxonomy dataset</h2><span id='topic+demo_tax'></span>

<h3>Description</h3>

<p>This dataset was given by Lundberg et al., (2012). The taxonomy table provides
information about the taxonomic identity of the OTUs listed in the OTU table.
Each row in the taxonomy table corresponds to an OTU from the OTU table, and
the columns provide details about the taxonomic classification of that OTU,
such as kingdom, phylum, class, order, family, genus, and species. This
information allows researchers to identify the microbial species or groups
that are present in the root samples.
The original dataset contains 777 rows of OTUs and Phylum, Class, Order,
Family in columns corresponding to particular OTU.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_tax
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 188 rows and 5 columns.
</p>


<h3>Details</h3>

<p>Here only the portion of the dataset is taken for running the functions.
The dataset contains 188 rows and 5 columns.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1038/nature11237">doi:10.1038/nature11237</a>
</p>

<hr>
<h2 id='group_bar_plots'>Grouped Bar Plots Based on Sample Size</h2><span id='topic+group_bar_plots'></span>

<h3>Description</h3>

<p>The grouped_bar_plots function is designed for generating grouped
bar plots to visualize data. It takes a OTU table before filtering and OTU table
after filtering as input containing data for multiple samples and creates a
series of grouped bar plots, each representing a specific group of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_bar_plots(otu_table_bef_filtering, otu_table_aft_filtering,
num_samples_per_plot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_bar_plots_+3A_otu_table_bef_filtering">otu_table_bef_filtering</code></td>
<td>
<p>A data frame of OTUs before filtering where the first row is the OTU ID and column names refer to sites/sample names</p>
</td></tr>
<tr><td><code id="group_bar_plots_+3A_otu_table_aft_filtering">otu_table_aft_filtering</code></td>
<td>
<p>A data frame of OTUs after filtering where the first row is the OTU ID and column names refer to sites/sample names</p>
</td></tr>
<tr><td><code id="group_bar_plots_+3A_num_samples_per_plot">num_samples_per_plot</code></td>
<td>
<p>The number of samples to be displayed in each grouped bar plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of interactive grouped bar plots, showing the change in sample size before and after filtering OTU table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To run input data
core_1 &lt;- CoreMicrobiome(
 otu_table = demo_otu,
 tax_table = demo_tax,
 metadata_table = demo_md,
 filter_type = "occupancy_fun_filter", #Or "abundance_fun_filter", Or "combined_filter"
 percent = 0.5,
 method = "css",  # Or "srs", "rrarefy", "tmm", "tmmwsp", "rle", "upperquartile", "none"
 beta_diversity_method = "jaccard",
 top_percentage = 10  # Adjust the percentage as needed for core/non-core OTUs
)
#To run grouped bar plot function
plot_group_bar &lt;- group_bar_plots(core_1$final_otu_table_bef_filter,
core_1$final_otu_aft_filter, 10)
#To view the grouped bar plot
plot_group_bar[[1]]
</code></pre>

<hr>
<h2 id='significance'>Testing the Significance of the Identified Core Microbiome</h2><span id='topic+significance'></span>

<h3>Description</h3>

<p>This function performs a two-sample variance test to assess the
statistical significance of differences in abundance between core OTUs and
non-core OTUs. It takes two data frames as input, representing the abundance
of core OTUs and non-core OTUs, and returns the results of the variance test.
It tells whether the identified core represents the particular environment or
habitat.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>significance(core_ids, non_core_ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="significance_+3A_core_ids">core_ids</code></td>
<td>
<p>A Dataframe of core OTUs where the first row is the OTU ID and column names refer to sites/sample names</p>
</td></tr>
<tr><td><code id="significance_+3A_non_core_ids">non_core_ids</code></td>
<td>
<p>A Dataframe of non_core OTUs where the first row is the OTU ID and column names refer to sites/sample names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function gives the list  which consist of following results.
</p>
<p>'statistic'  Calculated F test statistic
</p>
<p>'parameter' The numerator degrees of freedom (num df), and the denominator degrees of freedom (denom df)
</p>
<p>'p-value' Probability value
</p>
<p>'alternative' The alternative hypothesis for this test is that the true ratio of variances is not equal to 1. This suggests that the variances of the two data sets are different
</p>
<p>'conf.int' 95 percent confidence interval limit for the ratio of variances
</p>
<p>'estimate' Ratio of variances between core_data and non_core_data calculated
</p>
<p>'method' The test performed is an F test, which compares the variances of the two data sets
</p>
<p>'data.name' The data used for the test are core_ids and non_core_ids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To run input data
core_1 &lt;- CoreMicrobiome(
 otu_table = demo_otu,
 tax_table = demo_tax,
 metadata_table = demo_md,
 filter_type = "occupancy_fun_filter", #Or "abundance_fun_filter", Or "combined_filter"
 percent = 0.5,
 method = "css",  # Or "srs", "rrarefy", "tmm", "tmmwsp", "rle", "upperquartile", "none"
 beta_diversity_method = "jaccard",
 top_percentage = 10  # Adjust the percentage as needed for core/non-core OTUs
)
#To run significance test
f_test &lt;- significance(core_1[["core_otus"]] , core_1[["non_core_otus"]] )

#To view the significance test result
f_test
</code></pre>

<hr>
<h2 id='stacked_bar_plots'>Stacked Bar Plots Based on Relative Abundance Data</h2><span id='topic+stacked_bar_plots'></span>

<h3>Description</h3>

<p>This function generates stacked bar plots for visualizing
the relative abundance data of different operational taxonomic units (OTUs)
in various samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stacked_bar_plots(data, num_samples_per_plot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stacked_bar_plots_+3A_data">data</code></td>
<td>
<p>A data frame containing the relative abundance data for the OTUs. The first column should contain the OTU IDs, and the subsequent columns should represent samples.</p>
</td></tr>
<tr><td><code id="stacked_bar_plots_+3A_num_samples_per_plot">num_samples_per_plot</code></td>
<td>
<p>The number of samples to be displayed in each stacked bar plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of interactive stacked bar plots, one for each group of samples, showing the relative abundance of OTUs in the samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To run input data
core_1 &lt;- CoreMicrobiome(
 otu_table = demo_otu,
 tax_table = demo_tax,
 metadata_table = demo_md,
 filter_type = "occupancy_fun_filter", #Or "abundance_fun_filter", Or "combined_filter"
 percent = 0.5,
 method = "css",  # Or "srs", "rrarefy", "tmm", "tmmwsp", "rle", "upperquartile", "none"
 beta_diversity_method = "jaccard",
 top_percentage = 10  # Adjust the percentage as needed for core/non-core OTUs
)
#To run the stacked bar plots function
stacked_plots &lt;- stacked_bar_plots(core_1$core_otus_relative_abundance, 10)
#To view the stacked bar plot
stacked_plots[[1]]
</code></pre>

<hr>
<h2 id='visualize'>Visualizing the effect of minimum count on the core size</h2><span id='topic+visualize'></span>

<h3>Description</h3>

<p>The visualize function generates interactive line plots that
allow users to explore the impact of different min_count values on the number of
core OTUs. Users can interact with the plots to examine the relationship between
filtering criteria and core OTU identification visually.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize(filtered_otu, min_count_val, max_count_val, count_val_interval,
prop, min_total_count, method, top_percentage)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_+3A_filtered_otu">filtered_otu</code></td>
<td>
<p>A dataframe of OTUs obtained before filtering which is retrieved from CoreMicrobiome function where the first row is the OTU ID and column names refer to sites/sample names</p>
</td></tr>
<tr><td><code id="visualize_+3A_min_count_val">min_count_val</code></td>
<td>
<p>A numeric value of Minimum count for each OTU to be present in each to be included after the filtering</p>
</td></tr>
<tr><td><code id="visualize_+3A_max_count_val">max_count_val</code></td>
<td>
<p>A numeric value of Maximum count for each OTU to be present in each to be included after the filtering</p>
</td></tr>
<tr><td><code id="visualize_+3A_count_val_interval">count_val_interval</code></td>
<td>
<p>Count value interval for each OTU to be present in each to be included after the filtering</p>
</td></tr>
<tr><td><code id="visualize_+3A_prop">prop</code></td>
<td>
<p>Minimum proportion of samples in which an OTU must be present</p>
</td></tr>
<tr><td><code id="visualize_+3A_min_total_count">min_total_count</code></td>
<td>
<p>Minimum total count for each OTU to be included after the filtering</p>
</td></tr>
<tr><td><code id="visualize_+3A_method">method</code></td>
<td>
<p>Different normalization methods, includes &quot;rrarefy&quot;, &quot;srs&quot;, &quot;css&quot;, &quot;tmm&quot;, or &quot;none&quot;</p>
</td></tr>
<tr><td><code id="visualize_+3A_top_percentage">top_percentage</code></td>
<td>
<p>Percentage used for obtaining the Core OTUs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function gives a line plot which shows change in number of core OTUs with minimum count
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To run input data
core_1 &lt;- CoreMicrobiome(
 otu_table = demo_otu,
 tax_table = demo_tax,
 metadata_table = demo_md,
 filter_type = "occupancy_fun_filter", #Or "abundance_fun_filter", Or "combined_filter"
 percent = 0.5,
 method = "css",  # Or "srs", "rrarefy", "tmm", "tmmwsp", "rle", "upperquartile", "none"
 beta_diversity_method = "jaccard",
 top_percentage = 10  # Adjust the percentage as needed for core/non-core OTUs
)
#To view the line plot
visualize(filtered_otu = core_1[["final_otu_table_bef_filter"]],
         min_count_val = 5,
         max_count_val = 25,
         count_val_interval = 5,
         prop = 0.1,
         min_total_count = 10,
         method = "srs",
         top_percentage =10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
