<!DOCTYPE html><html><head><title>Help for package CORE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CORE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CORE'><p>Cores of Recurrent Events</p></a></li>
<li><a href='#testInputBoundaries'>
<p>A table of chromosome boundary positions for DNA copy number analysis</p>
</a></li>
<li><a href='#testInputCORE'>
<p>A table of DNA copy number gain events observed in 100 individual tumor cells</p>
</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Cores of Recurrent Events</td>
</tr>
<tr>
<td>Version:</td>
<td>3.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Alex Krasnitz, Guoli Sun</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guoli Sun &lt;guolisun87@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Given a collection of intervals with integer start and end 
    positions, find recurrently targeted regions and estimate the significance 
    of finding. Randomization is implemented by parallel methods, either 
    using local host machines, or submitting grid engine jobs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, stats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-23</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-24 03:00:01 UTC</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-24 00:29:12 UTC; Astrid</td>
</tr>
</table>
<hr>
<h2 id='CORE'>Cores of Recurrent Events</h2><span id='topic+CORE'></span>

<h3>Description</h3>

<p>Given a collection of intervals s_1,...,s_N, find K intervals c_1,...,c_K which 
approximately minimize Sum_i Prod_k (1-E(s_i,c_k)), where E(s_i,c_k) is a 
geometric measure of association between s_i and c_k. Perform permutation 
tests to estimate the significance of finding.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CORE(dataIn, keep = NULL, startcol = "start", endcol = "end", 
chromcol = "chrom", weightcol = "weight", maxmark = 1, minscore = 0, 
pow = 1, assoc = c("I", "J", "P"), nshuffle = 0, boundaries = NULL, 
seedme = sample(1e+08, 1), shufflemethod = c("SIMPLE", "RESCALE"), 
tiny = -1, distrib = c("vanilla", "Rparallel","Grid"), njobs = 1,qmem=NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CORE_+3A_datain">dataIn</code></td>
<td>

<p>A matrix, a data frame or an object of class &quot;CORE&quot;. If <code>dataIn</code> 
is a matrix or a data frame, it should have columns with names specified by 
the <code>startcol</code> and <code>endcol</code> arguments, otherwise the function exits 
with an error.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_keep">keep</code></td>
<td>

<p>A character vector. If <code>dataIn</code> is of class &quot;CORE&quot;, <code>keep</code> specifies 
the names of items of <code>dataIn</code> to be kept at their input values. These 
values take precedence over the corresponding argument values as specified in 
the function call. <code>keep</code> is ignored if <code>dataIn</code> is not of class 
&quot;CORE&quot;.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_startcol">startcol</code></td>
<td>

<p>A character string. If <code>dataIn</code> is a matrix or a data frame, 
<code>startcol</code> specifies the name of the column containing start coordinates 
of the input intervals. Otherwise <code>startcol</code> is ignored.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_endcol">endcol</code></td>
<td>

<p>A character string. If <code>dataIn</code> is a matrix or a data frame, 
<code>endcol</code> specifies the name of the column containing end coordinates 
of the input intervals. Otherwise <code>endcol</code> is ignored.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_chromcol">chromcol</code></td>
<td>

<p>A character string. If <code>dataIn</code> is a matrix or a data frame, 
<code>chromcol</code> specifies the name of the column containing chromosome numbers 
of the input intervals. Otherwise <code>chromcol</code> is ignored.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_weightcol">weightcol</code></td>
<td>

<p>A character string. If <code>dataIn</code> is a matrix or a data frame, 
<code>weightcol</code> specifies the name of the column containing initial weights 
of the input intervals. Otherwise <code>weightcol</code> is ignored.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_maxmark">maxmark</code></td>
<td>

<p>An integer for the maximal number of cores to be computed. The actual number 
of cores to be computed is the smaller of <code>maxmark</code> and the number of 
cores with scores exceeding <code>minscore</code>.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_minscore">minscore</code></td>
<td>

<p>A single numeric value for the minimal allowed score of the cores to be 
reported.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_pow">pow</code></td>
<td>

<p>A single numeric value of at least 1 for the power parameter used in 
computing the association measure beween the cores and the input intervals
(see Details).
</p>
</td></tr>
<tr><td><code id="CORE_+3A_assoc">assoc</code></td>
<td>

<p>A character specifying the type of association measure to be used (see Details).
</p>
</td></tr>
<tr><td><code id="CORE_+3A_nshuffle">nshuffle</code></td>
<td>

<p>An integer specifying the number of randomizations to be performed for 
estimating significance.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_boundaries">boundaries</code></td>
<td>

<p>A matrix or a data frame that must have three columns whose names are given by 
<code>chromcol</code>, <code>startcol</code> and <code>endcol</code>. These specify the 
chromosome numbers and their start and end positions (see Details).
</p>
</td></tr>
<tr><td><code id="CORE_+3A_seedme">seedme</code></td>
<td>

<p>An integer specifying the random number generator seed (see Details).
</p>
</td></tr>
<tr><td><code id="CORE_+3A_shufflemethod">shufflemethod</code></td>
<td>

<p>A character string specifying the event randomization method used for 
estimation of significance. If &quot;SIMPLE&quot; (default), each event is placed at 
random with equal probability for any position where it can fit within 
chromosome boundaries. If &quot;RESCALE&quot;, each event is placed at random in a 
randomly chosen chromosome, and the event length is multiplied by the length 
ratio of the new to the original chromosome.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_tiny">tiny</code></td>
<td>

<p>A single numeric value specifying the weight below which events are removed 
from the input event set.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_distrib">distrib</code></td>
<td>

<p>A character string specifying the method of distributed computing used for 
estimation of significance. If &quot;vanilla&quot; (default), no distributed computing 
is performed. If &quot;Rparallel&quot;, parallel computation with the local machine 
is performed using functions from CRAN core package parallel, with 
the number of worker processes being the smaller number of <code>njobs</code>,and 
<code>nshuffle</code>. If &quot;Grid&quot;, parallel computation with grid engine is performed. 
The number of submitted array jobs, or cores that are distributed, is the 
smaller number of <code>njobs</code>,and <code>nshuffle</code>. When using &quot;Grid&quot;, make 
sure you have write premission to the current work space.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_njobs">njobs</code></td>
<td>

<p>If distributed computing is used for estimation of significance, a single 
integer specifying the desired number of worker processes.
</p>
</td></tr>
<tr><td><code id="CORE_+3A_qmem">qmem</code></td>
<td>

<p>A character string that can customize grid engine <code>qsub</code> command. The 
command decides memory size per core(each job). The default substring is 
&quot;-l virtual_free=2G&quot;.  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The three measures of association specified by <code>assoc</code> are defined as 
follows (|| denotes the length of an interval). For &quot;I&quot; (inclusion) 
E(s_i,c_k) = (|c_k|/|s_i|)^pow if c_k is contained in s_i and 0 otherwise. 
For &quot;J&quot; (Jaccard) E(s_i,c_k) = J(s_i,c_k)^pow, where J is the Jaccard index. 
For &quot;P&quot; (piercing) E(s_i,c_k) = 1 if c_k is contained and 0 otherwise. 
In all cases the left (right) boundary of an optimal c_k is one of the left 
(right) boundaries in the set of input interval events. In addition, there 
are no event interval boundaries in the interior of an optimal c_k in case &quot;P&quot;.
</p>
<p>The <code>boundaries</code> argument is used for assessing statistical significance 
of the solution. If <code>boundaries</code> is not specified, the chromosome 
boundaries for each chromosome are taken to be the leftmost left and the 
rightmost right boundaries of all events in the chromosome.
</p>
<p>If significance of finding is estimated, the random number generator stream, 
and hence the resultant estimate, only depends on <code>seedme</code> and is 
independent of the parallelization option chosen.
</p>


<h3>Value</h3>

<p>An object of class &quot;CORE&quot; with the following items.
</p>
<table>
<tr><td><code>input</code></td>
<td>

<p>A matrix with four columns called &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot; and &quot;weight&quot;, 
specifying the input interval events.
</p>
</td></tr>
<tr><td><code>call</code></td>
<td>

<p>A character string specifying the function call.
</p>
</td></tr>
<tr><td><code>coreTable</code></td>
<td>

<p>A matrix with columns named &quot;start&quot;, &quot;end&quot; and &quot;score&quot;, for start and end 
positions and CORE scores of the cores found by the algorithm.
</p>
</td></tr>
<tr><td><code>seedme</code></td>
<td>

<p>If significance estimate was performed, the random number generator seed.
</p>
</td></tr>
<tr><td><code>assoc</code></td>
<td>

<p>One of &quot;I&quot;, &quot;J&quot; or &quot;P&quot;, indicating the geometric measure of association used.
</p>
</td></tr>
<tr><td><code>shufflemethod</code></td>
<td>

<p>One of &quot;SIMPLE&quot; or &quot;RESCALE&quot;, indicating the randomization method used.
</p>
</td></tr>
<tr><td><code>p</code></td>
<td>

<p>A numeric vector of the length equal to the row dimension of <code>coreTable</code> 
containing estimated p-values for the cores.
</p>
</td></tr>
<tr><td><code>simscores</code></td>
<td>

<p>A matrix with the row dimension equal to that of <code>coreTable</code> and 
<code>nshuffle</code> columns, containing core scores computed for <code>nshuffle</code> 
sets of randomized events.
</p>
</td></tr>
<tr><td><code>minscore</code></td>
<td>

<p>A single numeric value for the minimal score of the reported cores.
</p>
</td></tr>
<tr><td><code>maxmark</code></td>
<td>

<p>A single numeric value for the requested maximal number of cores to be 
computed.
</p>
</td></tr>
<tr><td><code>tiny</code></td>
<td>

<p>A single numeric value for the weight below which events were removed from 
the input set.
</p>
</td></tr>
<tr><td><code>pow</code></td>
<td>

<p>A single numeric value for the power used in computing the association 
measures.
</p>
</td></tr>
<tr><td><code>boundaries</code></td>
<td>

<p>A matrix with three columns named &quot;chrom&quot;, &quot;start&quot; and &quot;end&quot;, indicating 
chromosome numbers and boundary positions used for estimation of significance.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Alex Krasnitz,Guoli Sun</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Compute 3 cores and perform no randomization 
#(meaningless for estimate of significance).
data(testInputCORE)
data(testInputBoundaries)
myCOREobj&lt;-CORE(dataIn=testInputCORE,maxmark=3,nshuffle=0,
boundaries=testInputBoundaries,seedme=123)
## Not run: 
#Extend this computation to a much larger number of randomizations,
#using 2 cores of a host computer.
newCOREobj&lt;-CORE(dataIn=myCOREobj,keep=c("maxmark","seedme","boundaries"),
nshuffle=20,distrib="Rparallel",njobs=2)
#When using "Grid", make sure you have write premission to the current 
#work space.
newCOREobj&lt;-CORE(dataIn=myCOREobj,keep=c("maxmark","seedme","boundaries"),
nshuffle=20,distrib="Grid",njobs=2)

## End(Not run)
</code></pre>

<hr>
<h2 id='testInputBoundaries'>
A table of chromosome boundary positions for DNA copy number analysis

</h2><span id='topic+testInputBoundaries'></span>

<h3>Description</h3>

<p>The entire length of the genome was divided into 50009 bins, with chromosomes
laid out in the usual genomic order: 1,...,22,X,Y. Each observation in the
table provides the <code>start</code> and <code>end</code> bin numbers of each chromosome
(<code>chrom</code>). 

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(testInputBoundaries)</code></pre>


<h3>Format</h3>

<p>A data frame with 24 observations on the following 3 variables.
</p>

<dl>
<dt><code>chrom</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>end</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Navin N, Kendall J, Troge J, Andrews P, Rodgers L, McIndoo J, Cook K,
Stepansky A, Levy D, Esposito D et al. 2011.
Tumour evolution inferred by single-cell sequencing. Nature 472(7341): 90-U119.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(testInputBoundaries)
## maybe str(testInputBoundaries) ; plot(testInputBoundaries) ...
</code></pre>

<hr>
<h2 id='testInputCORE'>
A table of DNA copy number gain events observed in 100 individual tumor cells

</h2><span id='topic+testInputCORE'></span>

<h3>Description</h3>

<p>Each observation in the table corresponds to a DNA copy number gain event in
one of 100 individual breast cancer cells. The entire length of the genome was
divided into 50009 bins. An event is an interval in chromosome <code>chrom</code>
whose start and end bin numbers are given by <code>start</code> and <code>end</code>.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(testInputCORE)</code></pre>


<h3>Format</h3>

<p>A data frame with 2490 observations on the following 3 variables.
</p>

<dl>
<dt><code>chrom</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>end</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Navin N, Kendall J, Troge J, Andrews P, Rodgers L, McIndoo J, Cook K, 
Stepansky A, Levy D, Esposito D et al. 2011. 
Tumour evolution inferred by single-cell sequencing. Nature 472(7341): 90-U119.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(testInputCORE)
## maybe str(testInputCORE) ; plot(testInputCORE) ...
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
