<!DOCTYPE html><html><head><title>Help for package corncob</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {corncob}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#bbdml'><p>Maximum Likelihood for the Beta-binomial Distribution</p></a></li>
<li><a href='#checkNested'><p>Check for nested models</p></a></li>
<li><a href='#clean_taxa_names'><p>Rename taxa</p></a></li>
<li><a href='#contrastsTest'><p>Identify differentially-abundant and differentially-variable taxa using contrasts</p></a></li>
<li><a href='#convert_phylo'><p>Function to subset and convert phyloseq data</p></a></li>
<li><a href='#corncob-package'><p>Corncob package documentation.</p></a></li>
<li><a href='#coth'><p>Hyperbolic cotangent transformation</p></a></li>
<li><a href='#dbetabin'><p>Betabinomial density</p></a></li>
<li><a href='#dbetabin_neg'><p>Negative betabinomial density</p></a></li>
<li><a href='#dbetabinom_cts'><p>Densities of beta binomial distributions, permitting non integer x and size</p></a></li>
<li><a href='#differentialTest'><p>Identify differentially-abundant and differentially-variable taxa</p></a></li>
<li><a href='#doBoot'><p>Function to run a bootstrap iteration</p></a></li>
<li><a href='#fishZ'><p>Fisher's z transformation</p></a></li>
<li><a href='#genInits'><p>Generate initialization for optimization</p></a></li>
<li><a href='#getRestrictionTerms'><p>Get index of restricted terms for Wald test</p></a></li>
<li><a href='#gr_full'><p>Parameter Gradient Vector</p></a></li>
<li><a href='#HDIbetabinom'><p>Get highest density interval of beta-binomial</p></a></li>
<li><a href='#hessian'><p>Compute Hessian matrix at the MLE</p></a></li>
<li><a href='#ibd_phylo_otu'><p>IBD data, OTU count data frame.</p></a></li>
<li><a href='#ibd_phylo_sample'><p>IBD data, sample data frame.</p></a></li>
<li><a href='#ibd_phylo_taxa'><p>IBD data, taxonomy data frame.</p></a></li>
<li><a href='#invfishZ'><p>Inverse Fisher's z transformation</p></a></li>
<li><a href='#invlogit'><p>Inverse logit transformation</p></a></li>
<li><a href='#logit'><p>Logit transformation</p></a></li>
<li><a href='#lrtest'><p>Likelihood ratio test</p></a></li>
<li><a href='#objfun'><p>Objective function</p></a></li>
<li><a href='#otu_to_taxonomy'><p>Transform OTUs to their taxonomic label</p></a></li>
<li><a href='#pbLRT'><p>Parametric bootstrap likelihood ratio test</p></a></li>
<li><a href='#pbRao'><p>Parametric bootstrap Rao test</p></a></li>
<li><a href='#pbWald'><p>Parametric bootstrap Wald test</p></a></li>
<li><a href='#plot.bbdml'><p>Plotting function</p></a></li>
<li><a href='#plot.differentialTest'><p>differentialTest plot function</p></a></li>
<li><a href='#print.bbdml'><p>Print function</p></a></li>
<li><a href='#print.differentialTest'><p>differentialTest print function</p></a></li>
<li><a href='#print.summary.bbdml'><p>Print summary function</p></a></li>
<li><a href='#qbetabinom'><p>Get quantiles of beta binom</p></a></li>
<li><a href='#raotest'><p>Rao-type chi-squared test (model-based or robust)</p></a></li>
<li><a href='#sand_vcov'><p>Compute sandwich estimate of variance-covariance matrix</p></a></li>
<li><a href='#sandSE'><p>Compute sandwich standard errors. Legacy function. Use sand_vcov instead.</p></a></li>
<li><a href='#score'><p>Compute score at the MLE</p></a></li>
<li><a href='#simulate.bbdml'><p>Simulate from beta-binomial model</p></a></li>
<li><a href='#soil_phylo_otu'><p>Soil data, otu table as data frame.</p></a></li>
<li><a href='#soil_phylo_sample'><p>Soil data, sample data.</p></a></li>
<li><a href='#soil_phylo_taxa'><p>Soil data, taxa table as data frame.</p></a></li>
<li><a href='#soil_phylum_contrasts_otu'><p>Small soil phylum data for contrasts examples, otu table as data frame</p></a></li>
<li><a href='#soil_phylum_contrasts_sample'><p>Small soil phylum data for contrasts examples, sample data as data frame</p></a></li>
<li><a href='#soil_phylum_small_otu'><p>Small soil phylum data for examples, otu table as data frame</p></a></li>
<li><a href='#soil_phylum_small_otu1'><p>Small soil phylum data for examples, sample data as data frame combined with counts for OTU 1 and sequencing depth.</p></a></li>
<li><a href='#soil_phylum_small_sample'><p>Small soil phylum data for examples, sample data as data frame</p></a></li>
<li><a href='#summary.bbdml'><p>Summary function</p></a></li>
<li><a href='#waldchisq'><p>Wald-type chi-squared test</p></a></li>
<li><a href='#waldchisq_test'><p>Wald-type chi-squared test statistic (model-based or robust)</p></a></li>
<li><a href='#waldt'><p>Wald-type t test (model-based or robust)</p></a></li>
<li><a href='#warn_phyloseq'><p>Function to throw error if the 'phyloseq' package is called but it is not installed</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Count Regression for Correlated Observations with the
Beta-Binomial</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical modeling for correlated count data using the beta-binomial distribution, described in Martin et al. (2020) &lt;<a href="https://doi.org/10.1214%2F19-AOAS1283">doi:10.1214/19-AOAS1283</a>&gt;. It allows for both mean and overdispersion covariates.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/statdivlab/corncob">https://github.com/statdivlab/corncob</a>,
<a href="https://statdivlab.github.io/corncob/">https://statdivlab.github.io/corncob/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/statdivlab/corncob/issues">https://github.com/statdivlab/corncob/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, VGAM, numDeriv, ggplot2, trust, dplyr, magrittr,
detectseparation, scales, rlang</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, covr, limma, slam, R.rsp, optimx,
phyloseq</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-10 20:23:01 UTC; sarahteichman</td>
</tr>
<tr>
<td>Author:</td>
<td>Bryan D Martin [aut],
  Daniela Witten [aut],
  Sarah Teichman [ctb],
  Amy D Willis [aut, cre],
  Thomas W Yee [ctb] (VGAM library),
  Xiangjie Xue [ctb] (VGAM library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Amy D Willis &lt;adwillis@uw.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-10 21:03:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Value</h3>

<p>No return value. Created to import the pipe operator.
</p>

<hr>
<h2 id='bbdml'>Maximum Likelihood for the Beta-binomial Distribution</h2><span id='topic+bbdml'></span>

<h3>Description</h3>

<p>Maximum Likelihood for the Beta-binomial Distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbdml(
  formula,
  phi.formula,
  data,
  link = "logit",
  phi.link = "logit",
  method = "trust",
  control = list(maxit = 1000, reltol = 1e-14),
  numerical = FALSE,
  nstart = 1,
  inits = NULL,
  allow_noninteger = FALSE,
  robust = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbdml_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code>: a symbolic description of the model to be fitted to the abundance</p>
</td></tr>
<tr><td><code id="bbdml_+3A_phi.formula">phi.formula</code></td>
<td>
<p>an object of class <code>formula</code> without the response: a symbolic description of the model to be fitted to the dispersion</p>
</td></tr>
<tr><td><code id="bbdml_+3A_data">data</code></td>
<td>
<p>a data frame or <code>phyloseq</code> object containing the variables in the models</p>
</td></tr>
<tr><td><code id="bbdml_+3A_link">link</code></td>
<td>
<p>link function for abundance covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="bbdml_+3A_phi.link">phi.link</code></td>
<td>
<p>link function for dispersion covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="bbdml_+3A_method">method</code></td>
<td>
<p>optimization method, defaults to <code>"trust"</code>, or see <code><a href="optimx.html#topic+optimr">optimr</a></code> for other options</p>
</td></tr>
<tr><td><code id="bbdml_+3A_control">control</code></td>
<td>
<p>optimization control parameters (see <code><a href="optimx.html#topic+optimr">optimr</a></code>)</p>
</td></tr>
<tr><td><code id="bbdml_+3A_numerical">numerical</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether to use the numeric Hessian (not recommended).</p>
</td></tr>
<tr><td><code id="bbdml_+3A_nstart">nstart</code></td>
<td>
<p>Integer. Defaults to <code>1</code>. Number of starts for optimization.</p>
</td></tr>
<tr><td><code id="bbdml_+3A_inits">inits</code></td>
<td>
<p>Optional initializations as rows of a matrix. Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="bbdml_+3A_allow_noninteger">allow_noninteger</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Should noninteger W's and M's be allowed? This behavior was not permitted prior to v4.1, needs to be explicitly allowed.</p>
</td></tr>
<tr><td><code id="bbdml_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be returned? If not, model-based standard arras are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="bbdml_+3A_...">...</code></td>
<td>
<p>Optional additional arguments for <code><a href="optimx.html#topic+optimr">optimr</a></code> or <code><a href="trust.html#topic+trust">trust</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>bbdml</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data frame example
data(soil_phylum_small_otu1)
bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

# phyloseq example (only run this if you have phyloseq installed)
## Not run: 
data(soil_phylum_small_sample)
data(soil_phylum_small_otu)
data_phylo &lt;- phyloseq::phyloseq(phyloseq::sample_data(soil_phylum_small_sample),
phyloseq::otu_table(soil_phylum_small_otu, taxa_are_rows = TRUE))
bbdml(formula = Proteobacteria ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = data_phylo)

## End(Not run)

</code></pre>

<hr>
<h2 id='checkNested'>Check for nested models</h2><span id='topic+checkNested'></span>

<h3>Description</h3>

<p>Check for nested models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkNested(mod, mod_null)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkNested_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="checkNested_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>TRUE</code> if <code>mod_null</code> is nested within <code>mod</code>, otherwise it throws an error.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)

checkNested(mod1, mod2)

</code></pre>

<hr>
<h2 id='clean_taxa_names'>Rename taxa</h2><span id='topic+clean_taxa_names'></span>

<h3>Description</h3>

<p>Renames taxa to have short human-readable names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_taxa_names(x, name = "OTU")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_taxa_names_+3A_x">x</code></td>
<td>
<p>Object of class <code>phyloseq</code></p>
</td></tr>
<tr><td><code id="clean_taxa_names_+3A_name">name</code></td>
<td>
<p>Character, defaults to <code>"OTU"</code>. Optional. String to use in every taxa name.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The original taxa names are saved as the <code>original_names</code> attribute. See the example for an example of how to access the original names.
</p>


<h3>Value</h3>

<p>Object of class <code>phyloseq</code>, with taxa renamed (defaults to OTU1, OTU2, ...), with the original taxa names saved as an attribute.
</p>

<hr>
<h2 id='contrastsTest'>Identify differentially-abundant and differentially-variable taxa using contrasts</h2><span id='topic+contrastsTest'></span>

<h3>Description</h3>

<p>Identify differentially-abundant and differentially-variable taxa using contrasts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrastsTest(
  formula,
  phi.formula,
  contrasts_DA = NULL,
  contrasts_DV = NULL,
  data,
  link = "logit",
  phi.link = "logit",
  sample_data = NULL,
  taxa_are_rows = TRUE,
  filter_discriminant = TRUE,
  fdr_cutoff = 0.05,
  fdr = "fdr",
  inits = NULL,
  try_only = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contrastsTest_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code> without the response: a symbolic description of the model to be fitted to the abundance</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_phi.formula">phi.formula</code></td>
<td>
<p>an object of class <code>formula</code> without the response: a symbolic description of the model to be fitted to the dispersion</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_contrasts_da">contrasts_DA</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>. Must include at least one of <code>contrasts_DA</code> or <code>contrasts_DV</code>.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_contrasts_dv">contrasts_DV</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>phi.formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>. Must include at least one of <code>contrasts_DA</code> or <code>contrasts_DV</code>.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_data">data</code></td>
<td>
<p>a data frame containing the OTU table, or <code>phyloseq</code> object containing the variables in the models</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_link">link</code></td>
<td>
<p>link function for abundance covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_phi.link">phi.link</code></td>
<td>
<p>link function for dispersion covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_sample_data">sample_data</code></td>
<td>
<p>Data frame or matrix. Defaults to <code>NULL</code>. If <code>data</code> is a data frame or matrix, this must be included as covariates/sample data.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_taxa_are_rows">taxa_are_rows</code></td>
<td>
<p>Boolean. Optional. If <code>data</code> is a data frame or matrix, this indicates whether taxa are rows. Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_filter_discriminant">filter_discriminant</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. If <code>FALSE</code>, discriminant taxa will not be filtered out.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_fdr_cutoff">fdr_cutoff</code></td>
<td>
<p>Integer. Defaults to <code>0.05</code>. Desired type 1 error rate</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_fdr">fdr</code></td>
<td>
<p>Character. Defaults to <code>"fdr"</code>. False discovery rate control method, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for more options.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_inits">inits</code></td>
<td>
<p>Optional initializations for model fit using <code>formula</code> and <code>phi.formula</code> as rows of a matrix. Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_try_only">try_only</code></td>
<td>
<p>Optional numeric. Will try only the <code>try_only</code> taxa, specified either via numeric input or character taxa names. Useful for speed when troubleshooting. Defaults to <code>NULL</code>, testing all taxa.</p>
</td></tr>
<tr><td><code id="contrastsTest_+3A_...">...</code></td>
<td>
<p>Optional additional arguments for <code><a href="#topic+bbdml">bbdml</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses contrast matrices to test for differential abundance and differential variability using a Wald-type chi-squared test. To use a formula implementation, see <code><a href="#topic+differentialTest">differentialTest</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>contrastsTest</code>. List with elements <code>p</code> containing the p-values for each contrast, <code>p_fdr</code> containing the p-values after false discovery rate control,  <code>significant_taxa</code> containing the taxa names of the statistically significant taxa,  <code>contrasts_DA</code> containing the contrast matrix for parameters associated with the abundance, <code>contrasts_DV</code> containing the contrast matrix for parameters associated with the dispersion, <code>discriminant_taxa_DA</code> containing the taxa for which at least one covariate associated with the abundance was perfectly discriminant, <code>discriminant_taxa_DV</code> containing the taxa for which at least one covariate associated with the dispersion was perfectly discriminant, and <code>data</code> containing the data used to fit the models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data frame example
data(soil_phylum_contrasts_sample)
data(soil_phylum_contrasts_otu)
da_analysis &lt;- contrastsTest(formula = ~ DayAmdmt,
                             phi.formula = ~ DayAmdmt,
                             contrasts_DA = list("DayAmdmt21 - DayAmdmt11",
                                                 "DayAmdmt22 - DayAmdmt21"),
                             data = soil_phylum_contrasts_otu,
                             sample_data = soil_phylum_contrasts_sample,
                             fdr_cutoff = 0.05,
                             try_only = 1:5)

# phyloseq example (only run if you have phyloseq installed)
## Not run: 
contrasts_phylo &lt;- phyloseq::phyloseq(phyloseq::sample_data(soil_phylum_contrasts_sample),
phyloseq::otu_table(soil_phylum_contrasts_otu, taxa_are_rows = TRUE))
da_analysis &lt;- contrastsTest(formula = ~ DayAmdmt,
                             phi.formula = ~ DayAmdmt,
                             contrasts_DA = list("DayAmdmt21 - DayAmdmt11",
                                                 "DayAmdmt22 - DayAmdmt21"),
                             data = contrasts_phylo,
                             fdr_cutoff = 0.05,
                             try_only = 1:5)

## End(Not run)

</code></pre>

<hr>
<h2 id='convert_phylo'>Function to subset and convert phyloseq data</h2><span id='topic+convert_phylo'></span>

<h3>Description</h3>

<p>Function to subset and convert phyloseq data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_phylo(data, select)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_phylo_+3A_data">data</code></td>
<td>
<p>a <code>phyloseq</code> object</p>
</td></tr>
<tr><td><code id="convert_phylo_+3A_select">select</code></td>
<td>
<p>Name of OTU or taxa to select, must match taxa name in <code>data</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> object, with elements <code>W</code> as the observed counts, <code>M</code> as the sequencing depth, and the sample data with their original names.
</p>

<hr>
<h2 id='corncob-package'>Corncob package documentation.</h2><span id='topic+corncob-package'></span><span id='topic+corncob'></span>

<h3>Description</h3>

<p>Corncob provides methods for estimating and plotting count data. Specifically, corncob is designed to account for the challenges of modeling sequencing data from microbial abundance studies.
</p>


<h3>Details</h3>

<p>For details on the model implemented in this package, see Martin et al. (2020) &lt;doi:10.1214/19-AOAS1283&gt;.
</p>
<p>The development version of the package will be maintained on <a href="https://github.com/statdivlab/corncob">https://github.com/statdivlab/corncob</a>.
</p>


<h3>Value</h3>

<p>No return value. Created for documentation.
</p>

<hr>
<h2 id='coth'>Hyperbolic cotangent transformation</h2><span id='topic+coth'></span>

<h3>Description</h3>

<p>Hyperbolic cotangent transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coth(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coth_+3A_x">x</code></td>
<td>
<p>data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Hyperbolic cotangent transformation of x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- .5
coth(x)

</code></pre>

<hr>
<h2 id='dbetabin'>Betabinomial density</h2><span id='topic+dbetabin'></span>

<h3>Description</h3>

<p>Betabinomial density
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbetabin(theta, W, M, X, X_star, np, npstar, link, phi.link, logpar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dbetabin_+3A_theta">theta</code></td>
<td>
<p>Numeric vector. Parameters associated with <code>X</code> and <code>X_star</code></p>
</td></tr>
<tr><td><code id="dbetabin_+3A_w">W</code></td>
<td>
<p>Numeric vector of counts</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_x">X</code></td>
<td>
<p>Matrix of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_x_star">X_star</code></td>
<td>
<p>Matrix of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_np">np</code></td>
<td>
<p>Number of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_npstar">npstar</code></td>
<td>
<p>Number of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_link">link</code></td>
<td>
<p>ink function for abundance covariates</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_phi.link">phi.link</code></td>
<td>
<p>ink function for dispersion covariates</p>
</td></tr>
<tr><td><code id="dbetabin_+3A_logpar">logpar</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. Indicator of whether to return log-likelihood.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Negative beta-binomial (log-)likelihood
</p>

<hr>
<h2 id='dbetabin_neg'>Negative betabinomial density</h2><span id='topic+dbetabin_neg'></span>

<h3>Description</h3>

<p>Created as a convenient helper function for optimization. Not intended for users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbetabin_neg(theta, W, M, X, X_star, np, npstar, link, phi.link, logpar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dbetabin_neg_+3A_theta">theta</code></td>
<td>
<p>Numeric vector. Parameters associated with <code>X</code> and <code>X_star</code></p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_w">W</code></td>
<td>
<p>Numeric vector of counts</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_x">X</code></td>
<td>
<p>Matrix of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_x_star">X_star</code></td>
<td>
<p>Matrix of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_np">np</code></td>
<td>
<p>Number of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_npstar">npstar</code></td>
<td>
<p>Number of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_link">link</code></td>
<td>
<p>ink function for abundance covariates</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_phi.link">phi.link</code></td>
<td>
<p>ink function for dispersion covariates</p>
</td></tr>
<tr><td><code id="dbetabin_neg_+3A_logpar">logpar</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. Indicator of whether to return log-likelihood.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Negative beta-binomial (log-)likelihood
</p>

<hr>
<h2 id='dbetabinom_cts'>Densities of beta binomial distributions, permitting non integer x and size</h2><span id='topic+dbetabinom_cts'></span>

<h3>Description</h3>

<p>In some cases we may not have integer W and M's. In these cases,
we can still use corncob to estimate parameters, but we need to think of them as
no longer coming from the specific beta binomial parametric model,
and instead from an estimating equations framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbetabinom_cts(x, size, prob, rho = 0, log = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dbetabinom_cts_+3A_x">x</code></td>
<td>
<p>the value at which defined the density</p>
</td></tr>
<tr><td><code id="dbetabinom_cts_+3A_size">size</code></td>
<td>
<p>number of trials</p>
</td></tr>
<tr><td><code id="dbetabinom_cts_+3A_prob">prob</code></td>
<td>
<p>the probability of success</p>
</td></tr>
<tr><td><code id="dbetabinom_cts_+3A_rho">rho</code></td>
<td>
<p>the correlation parameter</p>
</td></tr>
<tr><td><code id="dbetabinom_cts_+3A_log">log</code></td>
<td>
<p>if TRUE, log-densities p are given</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas W Yee
</p>
<p>Xiangjie Xue
</p>
<p>Amy D Willis
</p>

<hr>
<h2 id='differentialTest'>Identify differentially-abundant and differentially-variable taxa</h2><span id='topic+differentialTest'></span>

<h3>Description</h3>

<p>Identify differentially-abundant and differentially-variable taxa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>differentialTest(
  formula,
  phi.formula,
  formula_null,
  phi.formula_null,
  data,
  link = "logit",
  phi.link = "logit",
  test,
  boot = FALSE,
  B = 1000,
  sample_data = NULL,
  taxa_are_rows = TRUE,
  filter_discriminant = TRUE,
  fdr_cutoff = 0.05,
  fdr = "fdr",
  full_output = FALSE,
  inits = NULL,
  inits_null = NULL,
  try_only = NULL,
  verbose = FALSE,
  robust = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="differentialTest_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code> without the response: a symbolic description of the model to be fitted to the abundance</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_phi.formula">phi.formula</code></td>
<td>
<p>an object of class <code>formula</code> without the response: a symbolic description of the model to be fitted to the dispersion</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_formula_null">formula_null</code></td>
<td>
<p>Formula for mean under null, without response</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_phi.formula_null">phi.formula_null</code></td>
<td>
<p>Formula for overdispersion under null, without response</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_data">data</code></td>
<td>
<p>a data frame containing the OTU table, or <code>phyloseq</code> object containing the variables in the models</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_link">link</code></td>
<td>
<p>link function for abundance covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="differentialTest_+3A_phi.link">phi.link</code></td>
<td>
<p>link function for dispersion covariates, defaults to <code>"logit"</code></p>
</td></tr>
<tr><td><code id="differentialTest_+3A_test">test</code></td>
<td>
<p>Character. Hypothesis testing procedure to use. One of <code>"Wald"</code>, <code>"LRT"</code> (likelihood ratio test), or <code>"Rao"</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_boot">boot</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether or not to use parametric bootstrap algorithm. (See <code><a href="#topic+pbWald">pbWald</a></code> and <code><a href="#topic+pbLRT">pbLRT</a></code>).</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_b">B</code></td>
<td>
<p>Optional integer. Number of bootstrap iterations. Ignored if <code>boot</code> is <code>FALSE</code>. Otherwise, defaults to <code>1000</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_sample_data">sample_data</code></td>
<td>
<p>Data frame or matrix. Defaults to <code>NULL</code>. If <code>data</code> is a data frame or matrix, this must be included as covariates/sample data.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_taxa_are_rows">taxa_are_rows</code></td>
<td>
<p>Boolean. Optional. If <code>data</code> is a data frame or matrix, this indicates whether taxa are rows. Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_filter_discriminant">filter_discriminant</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. If <code>FALSE</code>, discriminant taxa will not be filtered out.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_fdr_cutoff">fdr_cutoff</code></td>
<td>
<p>Integer. Defaults to <code>0.05</code>. Desired false discovery rate.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_fdr">fdr</code></td>
<td>
<p>Character. Defaults to <code>"fdr"</code>. False discovery rate control method, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for more options.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_full_output">full_output</code></td>
<td>
<p>Boolean. Opetional. Defaults to <code>FALSE</code>. Indicator of whether to include full <code>bbdml</code> model output for all taxa.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_inits">inits</code></td>
<td>
<p>Optional initializations for model fit using <code>formula</code> and <code>phi.formula</code> as rows of a matrix. Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_inits_null">inits_null</code></td>
<td>
<p>Optional initializations for model fit using <code>formula_null</code> and <code>phi.formula_null</code> as rows of a matrix. Defaults to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_try_only">try_only</code></td>
<td>
<p>Optional numeric. Will try only the <code>try_only</code> taxa, specified either via numeric input or character taxa names. Useful for speed when troubleshooting. Defaults to <code>NULL</code>, testing all taxa.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_verbose">verbose</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>; print status updates for long-running analyses</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be used? If not, model-based standard errors are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="differentialTest_+3A_...">...</code></td>
<td>
<p>Optional additional arguments for <code><a href="#topic+bbdml">bbdml</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>See package vignette for details and example usage. Make sure the number of columns in all of the initializations are correct! <code>inits</code> probably shouldn't match <code>inits_null</code>. To use a contrast matrix, see <code><a href="#topic+contrastsTest">contrastsTest</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code>differentialTest</code>. List with elements <code>p</code> containing the p-values, <code>p_fdr</code> containing the p-values after false discovery rate control,  <code>significant_taxa</code> containing the taxa names of the statistically significant taxa, <code>significant_models</code> containing a list of the model fits for the significant taxa, <code>all_models</code> containing a list of the model fits for all taxa, <code>restrictions_DA</code> containing a list of covariates that were tested for differential abundance, <code>restrictions_DV</code> containing a list of covariates that were tested for differential variability, <code>discriminant_taxa_DA</code> containing the taxa for which at least one covariate associated with the abundance was perfectly discriminant, <code>discriminant_taxa_DV</code> containing the taxa for which at least one covariate associated with the dispersion was perfectly discriminant, <code>data</code> containing the data used to fit the models. If <code>full_output = TRUE</code>, it will also include <code>full_output</code>, a list of all model output from <code>bbdml</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# data frame example
data(soil_phylum_small_sample)
data(soil_phylum_small_otu)
da_analysis &lt;- differentialTest(formula = ~ DayAmdmt,
                                phi.formula = ~ DayAmdmt,
                                formula_null = ~ 1,
                                phi.formula_null = ~ DayAmdmt,
                                test = "Wald", boot = FALSE,
                                data = soil_phylum_small_otu,
                                sample_data = soil_phylum_small_sample,
                                fdr_cutoff = 0.05,
                                try_only = 1:5)

# phyloseq example (only run if you have phyloseq installed)
## Not run: 
data_phylo &lt;- phyloseq::phyloseq(phyloseq::sample_data(soil_phylum_small_sample),
phyloseq::otu_table(soil_phylum_small_otu, taxa_are_rows = TRUE))
da_analysis &lt;- differentialTest(formula = ~ DayAmdmt,
                               phi.formula = ~ DayAmdmt,
                               formula_null = ~ 1,
                               phi.formula_null = ~ DayAmdmt,
                               test = "Wald", boot = FALSE,
                               data = data_phylo,
                               fdr_cutoff = 0.05,
                               try_only = 1:5)

## End(Not run)

</code></pre>

<hr>
<h2 id='doBoot'>Function to run a bootstrap iteration</h2><span id='topic+doBoot'></span>

<h3>Description</h3>

<p>Internal function. Not intended for users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>doBoot(mod, mod_null, test, robust = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doBoot_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="doBoot_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="doBoot_+3A_test">test</code></td>
<td>
<p>Character. Hypothesis testing procedure to use. One of <code>"Wald"</code> or <code>"LRT"</code> (likelihood ratio test).</p>
</td></tr>
<tr><td><code id="doBoot_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be used? If not, model-based standard arras are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>test statistic from one bootstrap iteration
</p>

<hr>
<h2 id='fishZ'>Fisher's z transformation</h2><span id='topic+fishZ'></span>

<h3>Description</h3>

<p>Fisher's z transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishZ(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fishZ_+3A_x">x</code></td>
<td>
<p>data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fisher's z transformation of x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- .5
fishZ(x)

</code></pre>

<hr>
<h2 id='genInits'>Generate initialization for optimization</h2><span id='topic+genInits'></span>

<h3>Description</h3>

<p>Generate initialization for optimization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genInits(W, M, X, X_star, np, npstar, link, phi.link, nstart = 1, use = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genInits_+3A_w">W</code></td>
<td>
<p>Numeric vector of counts</p>
</td></tr>
<tr><td><code id="genInits_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="genInits_+3A_x">X</code></td>
<td>
<p>Matrix of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="genInits_+3A_x_star">X_star</code></td>
<td>
<p>Matrix of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="genInits_+3A_np">np</code></td>
<td>
<p>Number of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="genInits_+3A_npstar">npstar</code></td>
<td>
<p>Number of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="genInits_+3A_link">link</code></td>
<td>
<p>ink function for abundance covariates</p>
</td></tr>
<tr><td><code id="genInits_+3A_phi.link">phi.link</code></td>
<td>
<p>ink function for dispersion covariates</p>
</td></tr>
<tr><td><code id="genInits_+3A_nstart">nstart</code></td>
<td>
<p>Integer. Defaults to <code>1</code>. Number of starts for optimization.</p>
</td></tr>
<tr><td><code id="genInits_+3A_use">use</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. Indicator of whether to use deterministic intialization.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of initializations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
seq_depth &lt;- rpois(20, lambda = 10000)
my_counts &lt;- rbinom(20, size = seq_depth, prob = 0.001) * 10
my_covariate &lt;- cbind(rep(c(0,1), each = 10))
colnames(my_covariate) &lt;- c("X1")

genInits(W = my_counts, M = seq_depth,
       X = cbind(1, my_covariate), X_star = cbind(1, my_covariate),
       np = 2, npstar = 2,
       link = "logit",
       phi.link = "logit", nstart = 2, use = TRUE)
</code></pre>

<hr>
<h2 id='getRestrictionTerms'>Get index of restricted terms for Wald test</h2><span id='topic+getRestrictionTerms'></span>

<h3>Description</h3>

<p>Created as a convenient helper function. Not intended for users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRestrictionTerms(
  mod,
  mod_null = NULL,
  restrictions = NULL,
  restrictions.phi = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRestrictionTerms_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="getRestrictionTerms_+3A_mod_null">mod_null</code></td>
<td>
<p>Optional. An object of class <code>bbdml</code>. Defaults to <code>NULL</code></p>
</td></tr>
<tr><td><code id="getRestrictionTerms_+3A_restrictions">restrictions</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the abundance to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the abundance.</p>
</td></tr>
<tr><td><code id="getRestrictionTerms_+3A_restrictions.phi">restrictions.phi</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the dispersion to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the dispersion.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with <code>mu</code> representing the index of the restricted covariates associated with abundance and <code>phi</code> representing the index of the restricted covarates associated with the dispersion
</p>

<hr>
<h2 id='gr_full'>Parameter Gradient Vector</h2><span id='topic+gr_full'></span>

<h3>Description</h3>

<p>Used for internal optimization. Not intended for users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gr_full(theta, W, M, X, X_star, np, npstar, link, phi.link, logpar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gr_full_+3A_theta">theta</code></td>
<td>
<p>Numeric vector. Parameters associated with <code>X</code> and <code>X_star</code></p>
</td></tr>
<tr><td><code id="gr_full_+3A_w">W</code></td>
<td>
<p>Numeric vector of counts</p>
</td></tr>
<tr><td><code id="gr_full_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="gr_full_+3A_x">X</code></td>
<td>
<p>Matrix of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="gr_full_+3A_x_star">X_star</code></td>
<td>
<p>Matrix of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="gr_full_+3A_np">np</code></td>
<td>
<p>Number of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="gr_full_+3A_npstar">npstar</code></td>
<td>
<p>Number of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="gr_full_+3A_link">link</code></td>
<td>
<p>ink function for abundance covariates</p>
</td></tr>
<tr><td><code id="gr_full_+3A_phi.link">phi.link</code></td>
<td>
<p>ink function for dispersion covariates</p>
</td></tr>
<tr><td><code id="gr_full_+3A_logpar">logpar</code></td>
<td>
<p>Boolean. Defaults to <code>TRUE</code>. Indicator of whether to return log-likelihood.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Gradient of likelihood with respect to parameters
</p>

<hr>
<h2 id='HDIbetabinom'>Get highest density interval of beta-binomial</h2><span id='topic+HDIbetabinom'></span>

<h3>Description</h3>

<p>Get highest density interval of beta-binomial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HDIbetabinom(percent, M, mu, phi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HDIbetabinom_+3A_percent">percent</code></td>
<td>
<p>Numeric. Percent interval desired.</p>
</td></tr>
<tr><td><code id="HDIbetabinom_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="HDIbetabinom_+3A_mu">mu</code></td>
<td>
<p>Numeric vector of abundance parameter</p>
</td></tr>
<tr><td><code id="HDIbetabinom_+3A_phi">phi</code></td>
<td>
<p>Numeric vector of dispersion parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List where <code>lower</code> represents the lower bound and <code>upper</code> represents the upper bound
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
HDIbetabinom(.95, M = mod$M[1], mu = mod$mu.resp[1], phi = mod$phi.resp[1])

</code></pre>

<hr>
<h2 id='hessian'>Compute Hessian matrix at the MLE</h2><span id='topic+hessian'></span>

<h3>Description</h3>

<p>Compute Hessian matrix at the MLE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hessian(mod, numerical = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hessian_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="hessian_+3A_numerical">numerical</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether to use the numeric Hessian (not recommended).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Hessian matrix at the MLE. In this setting, it's hard to compute expectations to calculate the information matrix,
so we return the consistent estimate using sample moments:
<code class="reqn">\hat{A}(\hat{\theta}) = \sum_i \frac{\partial^2}{\partial \theta \partial \theta^T} l(\theta, W_i)</code> evaluated at <code class="reqn">\theta = \hat{\theta}</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
hessian(mod)

</code></pre>

<hr>
<h2 id='ibd_phylo_otu'>IBD data, OTU count data frame.</h2><span id='topic+ibd_phylo_otu'></span>

<h3>Description</h3>

<p>OTU data frame from a phyloseq object from an IBD microbiome study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ibd_phylo_otu
</code></pre>


<h3>Format</h3>

<p>A data frame of OTU counts.
</p>


<h3>References</h3>

<p>Papa, E., Docktor, M., Smillie, C., Weber, S., Preheim, S. P., Gevers, D., Giannoukos, G., Ciulla, D., Tabbaa, D., Ingram, J., Schauer, D. B., Ward, D. V., Korzenik, J. R., Xavier, R. J., Bousvaros, A., Alm, E. J. &amp; Schauer, D. B. (2012). <em>Non-invasive mapping of the gastrointestinal microbiota identifies children with inflammatory bowel disease</em>. PloS One, 7(6), e39242. &lt;doi.org/10.1371/journal.pone.0039242&gt;.
</p>
<p>Duvallet, C., Gibbons, S., Gurry, T., Irizarry, R., &amp; Alm, E. (2017). <em>MicrobiomeHD: the human gut microbiome in health and disease [Data set]</em>. Zenodo. &lt;doi.org/10.5281/zenodo.1146764&gt;.
</p>

<hr>
<h2 id='ibd_phylo_sample'>IBD data, sample data frame.</h2><span id='topic+ibd_phylo_sample'></span>

<h3>Description</h3>

<p>Sample data from a phyloseq object from an IBD microbiome study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ibd_phylo_sample
</code></pre>


<h3>Format</h3>

<p>A data frame of sample data.
</p>


<h3>References</h3>

<p>Papa, E., Docktor, M., Smillie, C., Weber, S., Preheim, S. P., Gevers, D., Giannoukos, G., Ciulla, D., Tabbaa, D., Ingram, J., Schauer, D. B., Ward, D. V., Korzenik, J. R., Xavier, R. J., Bousvaros, A., Alm, E. J. &amp; Schauer, D. B. (2012). <em>Non-invasive mapping of the gastrointestinal microbiota identifies children with inflammatory bowel disease</em>. PloS One, 7(6), e39242. &lt;doi.org/10.1371/journal.pone.0039242&gt;.
</p>
<p>Duvallet, C., Gibbons, S., Gurry, T., Irizarry, R., &amp; Alm, E. (2017). <em>MicrobiomeHD: the human gut microbiome in health and disease [Data set]</em>. Zenodo. &lt;doi.org/10.5281/zenodo.1146764&gt;.
</p>

<hr>
<h2 id='ibd_phylo_taxa'>IBD data, taxonomy data frame.</h2><span id='topic+ibd_phylo_taxa'></span>

<h3>Description</h3>

<p>Taxonomy data from a phyloseq object from an IBD microbiome study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ibd_phylo_taxa
</code></pre>


<h3>Format</h3>

<p>A data frame of taxonomy data.
</p>


<h3>References</h3>

<p>Papa, E., Docktor, M., Smillie, C., Weber, S., Preheim, S. P., Gevers, D., Giannoukos, G., Ciulla, D., Tabbaa, D., Ingram, J., Schauer, D. B., Ward, D. V., Korzenik, J. R., Xavier, R. J., Bousvaros, A., Alm, E. J. &amp; Schauer, D. B. (2012). <em>Non-invasive mapping of the gastrointestinal microbiota identifies children with inflammatory bowel disease</em>. PloS One, 7(6), e39242. &lt;doi.org/10.1371/journal.pone.0039242&gt;.
</p>
<p>Duvallet, C., Gibbons, S., Gurry, T., Irizarry, R., &amp; Alm, E. (2017). <em>MicrobiomeHD: the human gut microbiome in health and disease [Data set]</em>. Zenodo. &lt;doi.org/10.5281/zenodo.1146764&gt;.
</p>

<hr>
<h2 id='invfishZ'>Inverse Fisher's z transformation</h2><span id='topic+invfishZ'></span>

<h3>Description</h3>

<p>Inverse Fisher's z transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invfishZ(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="invfishZ_+3A_x">x</code></td>
<td>
<p>data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Inverse Fisher's z transformation of x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- .5
invfishZ(x)

</code></pre>

<hr>
<h2 id='invlogit'>Inverse logit transformation</h2><span id='topic+invlogit'></span>

<h3>Description</h3>

<p>Inverse logit transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invlogit(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="invlogit_+3A_x">x</code></td>
<td>
<p>data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Inverse logit transformation of x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- .5
invlogit(x)

</code></pre>

<hr>
<h2 id='logit'>Logit transformation</h2><span id='topic+logit'></span>

<h3>Description</h3>

<p>Logit transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logit(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logit_+3A_x">x</code></td>
<td>
<p>data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logit of x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- .5
logit(x)

</code></pre>

<hr>
<h2 id='lrtest'>Likelihood ratio test</h2><span id='topic+lrtest'></span>

<h3>Description</h3>

<p>Likelihood ratio test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrtest(mod, mod_null)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lrtest_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="lrtest_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code>, should be nested within <code>mod</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-value from likelihood ratio test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)
lrtest(mod1, mod2)
</code></pre>

<hr>
<h2 id='objfun'>Objective function</h2><span id='topic+objfun'></span>

<h3>Description</h3>

<p>Used for internal optimization. Not intended for users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>objfun(theta, W, M, X, X_star, np, npstar, link, phi.link)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="objfun_+3A_theta">theta</code></td>
<td>
<p>Numeric vector. Parameters associated with <code>X</code> and <code>X_star</code></p>
</td></tr>
<tr><td><code id="objfun_+3A_w">W</code></td>
<td>
<p>Numeric vector of counts</p>
</td></tr>
<tr><td><code id="objfun_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="objfun_+3A_x">X</code></td>
<td>
<p>Matrix of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="objfun_+3A_x_star">X_star</code></td>
<td>
<p>Matrix of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="objfun_+3A_np">np</code></td>
<td>
<p>Number of covariates associated with abundance (including intercept)</p>
</td></tr>
<tr><td><code id="objfun_+3A_npstar">npstar</code></td>
<td>
<p>Number of covariates associated with dispersion (including intercept)</p>
</td></tr>
<tr><td><code id="objfun_+3A_link">link</code></td>
<td>
<p>ink function for abundance covariates</p>
</td></tr>
<tr><td><code id="objfun_+3A_phi.link">phi.link</code></td>
<td>
<p>ink function for dispersion covariates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of negative log-likelihood, gradient, and hessian
</p>

<hr>
<h2 id='otu_to_taxonomy'>Transform OTUs to their taxonomic label</h2><span id='topic+otu_to_taxonomy'></span>

<h3>Description</h3>

<p>Transform OTUs to their taxonomic label
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otu_to_taxonomy(OTU, data, level = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otu_to_taxonomy_+3A_otu">OTU</code></td>
<td>
<p>String vector. Names of OTU labels in <code>data</code></p>
</td></tr>
<tr><td><code id="otu_to_taxonomy_+3A_data">data</code></td>
<td>
<p><code>phyloseq</code> object with a taxonomy table</p>
</td></tr>
<tr><td><code id="otu_to_taxonomy_+3A_level">level</code></td>
<td>
<p>(Optional). Character vector. Desired taxonomic levels for output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>String vector. Names of taxonomic labels matching labels of <code>OTU</code>.
</p>

<hr>
<h2 id='pbLRT'>Parametric bootstrap likelihood ratio test</h2><span id='topic+pbLRT'></span>

<h3>Description</h3>

<p>Parametric bootstrap likelihood ratio test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbLRT(mod, mod_null, B = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbLRT_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="pbLRT_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code>, should be nested within <code>mod</code></p>
</td></tr>
<tr><td><code id="pbLRT_+3A_b">B</code></td>
<td>
<p>Integer. Defaults to <code>1000</code>. Number of bootstrap iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-value from parametric bootstrap likelihood ratio test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)
pbLRT(mod1, mod2, B = 50)
</code></pre>

<hr>
<h2 id='pbRao'>Parametric bootstrap Rao test</h2><span id='topic+pbRao'></span>

<h3>Description</h3>

<p>Parametric bootstrap Rao test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbRao(mod, mod_null, B = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbRao_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="pbRao_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code>, should be nested within <code>mod</code></p>
</td></tr>
<tr><td><code id="pbRao_+3A_b">B</code></td>
<td>
<p>Integer. Defaults to <code>1000</code>. Number of bootstrap iterations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-value from parametric bootstrap Rao test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)
pbRao(mod1, mod2, B = 10)
</code></pre>

<hr>
<h2 id='pbWald'>Parametric bootstrap Wald test</h2><span id='topic+pbWald'></span>

<h3>Description</h3>

<p>Parametric bootstrap Wald test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbWald(mod, mod_null, B = 1000, robust = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbWald_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="pbWald_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code>, should be nested within <code>mod</code></p>
</td></tr>
<tr><td><code id="pbWald_+3A_b">B</code></td>
<td>
<p>Integer. Defaults to <code>1000</code>. Number of bootstrap iterations.</p>
</td></tr>
<tr><td><code id="pbWald_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be used? If not, model-based standard arras are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-value from parametric bootstrap Wald test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)
pbWald(mod1, mod2, B = 50)
</code></pre>

<hr>
<h2 id='plot.bbdml'>Plotting function</h2><span id='topic+plot.bbdml'></span>

<h3>Description</h3>

<p>Plotting function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bbdml'
plot(
  x,
  total = FALSE,
  color = NULL,
  shape = NULL,
  facet = NULL,
  title = NULL,
  B = 1000,
  sample_names = TRUE,
  data_only = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.bbdml_+3A_x">x</code></td>
<td>
<p>Object of class <code>bbdml</code>.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_total">total</code></td>
<td>
<p>(Optional). Default <code>FALSE</code>. Boolean indicator for whether to plot on total counts scale</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_color">color</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. The sample variable to map to different colors. Can be a single character string of the variable name in <code>sample_data</code> or a custom supplied vector with length equal to the number of samples. Use a character vector to have <code>ggplot2</code> default.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_shape">shape</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. The sample variable to map to different shapes. Can be a single character string of the variable name in <code>sample_data</code> or a custom supplied vector with length equal to the number of samples.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_facet">facet</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. The sample variable to map to different panels in a facet grid. Must be a single character string of a variable name in <code>sample_data</code>.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_title">title</code></td>
<td>
<p>(Optional). Default <code>NULL</code>. Character string. The main title for the graphic.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_b">B</code></td>
<td>
<p>(Optional). Default <code>1000</code>. Integer. Number of bootstrap simulations for prediction intervals. Use <code>B = 0</code> for no prediction intervals.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_sample_names">sample_names</code></td>
<td>
<p>(Optional). Default <code>TRUE</code>. Boolean. If <code>FALSE</code>, remove sample names from the plot.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_data_only">data_only</code></td>
<td>
<p>(Optional). Default <code>FALSE</code>. Boolean. If <code>TRUE</code>, only returns data frame.</p>
</td></tr>
<tr><td><code id="plot.bbdml_+3A_...">...</code></td>
<td>
<p>There are no optional parameters at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>ggplot</code>. Plot of <code>bbdml</code> model fit with 95
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
# Here we use B = 50 for quick demonstration purposes.
# In practice, we recommend a higher value for B for more accurate intervals
plot(mod, color = "DayAmdmt", B = 50)
</code></pre>

<hr>
<h2 id='plot.differentialTest'>differentialTest plot function</h2><span id='topic+plot.differentialTest'></span>

<h3>Description</h3>

<p>differentialTest plot function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'differentialTest'
plot(x, level = NULL, data_only = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.differentialTest_+3A_x">x</code></td>
<td>
<p>Object of class <code>differentialTest</code></p>
</td></tr>
<tr><td><code id="plot.differentialTest_+3A_level">level</code></td>
<td>
<p>(Optional). Character vector. Desired taxonomic levels for taxa labels.</p>
</td></tr>
<tr><td><code id="plot.differentialTest_+3A_data_only">data_only</code></td>
<td>
<p>(Optional). Default <code>FALSE</code>. Boolean. If <code>TRUE</code>, only returns data frame.</p>
</td></tr>
<tr><td><code id="plot.differentialTest_+3A_...">...</code></td>
<td>
<p>No optional arguments are accepted at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>ggplot</code>. Plot of coefficients from models for significant taxa from <code>differentialTest</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># phyloseq example
data(soil_phylum_small_sample)
data(soil_phylum_small_otu)
da_analysis &lt;- differentialTest(formula = ~ DayAmdmt,
                                phi.formula = ~ DayAmdmt,
                                formula_null = ~ 1,
                                phi.formula_null = ~ DayAmdmt,
                                test = "Wald", boot = FALSE,
                                data = soil_phylum_small_otu,
                                sample_data = soil_phylum_small_sample,
                                fdr_cutoff = 0.05,
                                try_only = 1:5)
plot(da_analysis, level = "Phylum")

</code></pre>

<hr>
<h2 id='print.bbdml'>Print function</h2><span id='topic+print.bbdml'></span>

<h3>Description</h3>

<p>Print function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bbdml'
print(
  x,
  digits = max(3L, getOption("digits") - 3L),
  signif.stars = getOption("show.signif.stars"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.bbdml_+3A_x">x</code></td>
<td>
<p>Object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="print.bbdml_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.bbdml_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical. If <code>TRUE</code>, &lsquo;significance stars&rsquo; are printed for each coefficient.</p>
</td></tr>
<tr><td><code id="print.bbdml_+3A_...">...</code></td>
<td>
<p>No optional arguments are accepted at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>. Displays printed model summary.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
print(mod)
</code></pre>

<hr>
<h2 id='print.differentialTest'>differentialTest print function</h2><span id='topic+print.differentialTest'></span>

<h3>Description</h3>

<p>differentialTest print function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'differentialTest'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.differentialTest_+3A_x">x</code></td>
<td>
<p>Object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="print.differentialTest_+3A_...">...</code></td>
<td>
<p>No optional arguments are accepted at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>. Displays printed <code>differentialTest</code> summary.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># phyloseq example
data(soil_phylum_small_sample)
data(soil_phylum_small_otu)
da_analysis &lt;- differentialTest(formula = ~ DayAmdmt,
                                phi.formula = ~ DayAmdmt,
                                formula_null = ~ 1,
                                phi.formula_null = ~ DayAmdmt,
                                test = "Wald", boot = FALSE,
                                data = soil_phylum_small_otu,
                                sample_data = soil_phylum_small_sample,
                                fdr_cutoff = 0.05,
                                try_only = 1:5)
print(da_analysis)

</code></pre>

<hr>
<h2 id='print.summary.bbdml'>Print summary function</h2><span id='topic+print.summary.bbdml'></span>

<h3>Description</h3>

<p>Print summary function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.bbdml'
print(
  x,
  digits = max(3L, getOption("digits") - 3L),
  signif.stars = getOption("show.signif.stars"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.summary.bbdml_+3A_x">x</code></td>
<td>
<p>Object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="print.summary.bbdml_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.summary.bbdml_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical. If <code>TRUE</code>, &lsquo;significance stars&rsquo; are printed for each coefficient.</p>
</td></tr>
<tr><td><code id="print.summary.bbdml_+3A_...">...</code></td>
<td>
<p>No optional arguments are accepted at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>. Displays printed model summary.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
print(summary(mod))
</code></pre>

<hr>
<h2 id='qbetabinom'>Get quantiles of beta binom</h2><span id='topic+qbetabinom'></span>

<h3>Description</h3>

<p>Get quantiles of beta binom
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qbetabinom(p, M, mu, phi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qbetabinom_+3A_p">p</code></td>
<td>
<p>Numeric. Probability for quantile</p>
</td></tr>
<tr><td><code id="qbetabinom_+3A_m">M</code></td>
<td>
<p>Numeric vector of sequencing depth</p>
</td></tr>
<tr><td><code id="qbetabinom_+3A_mu">mu</code></td>
<td>
<p>Numeric vector of abundance parameter</p>
</td></tr>
<tr><td><code id="qbetabinom_+3A_phi">phi</code></td>
<td>
<p>Numeric vector of dispersion parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>quantile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
qbetabinom(.5, M = mod$M[1], mu = mod$mu.resp[1], phi = mod$phi.resp[1])
</code></pre>

<hr>
<h2 id='raotest'>Rao-type chi-squared test (model-based or robust)</h2><span id='topic+raotest'></span>

<h3>Description</h3>

<p>Rao-type chi-squared test (model-based or robust)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raotest(mod, mod_null)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raotest_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="raotest_+3A_mod_null">mod_null</code></td>
<td>
<p>an object of class <code>bbdml</code>, should be nested within <code>mod</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>P-value from likelihood ratio test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)
raotest(mod1, mod2)
</code></pre>

<hr>
<h2 id='sand_vcov'>Compute sandwich estimate of variance-covariance matrix</h2><span id='topic+sand_vcov'></span>

<h3>Description</h3>

<p>Compute sandwich estimate of variance-covariance matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sand_vcov(mod, numerical = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sand_vcov_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="sand_vcov_+3A_numerical">numerical</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether to use the numeric Hessian and score (not recommended).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sandwich variance-covariance matrix. <code class="reqn">\hat{A}^{-1} \hat{B} \hat{A}^{-1}</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
sand_vcov(mod)

</code></pre>

<hr>
<h2 id='sandSE'>Compute sandwich standard errors. Legacy function. Use sand_vcov instead.</h2><span id='topic+sandSE'></span>

<h3>Description</h3>

<p>Compute sandwich standard errors. Legacy function. Use sand_vcov instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sandSE(mod, numerical = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sandSE_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="sandSE_+3A_numerical">numerical</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether to use the numeric Hessian and score (not recommended).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sandwich variance-covariance matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
sandSE(mod)

</code></pre>

<hr>
<h2 id='score'>Compute score at the MLE</h2><span id='topic+score'></span>

<h3>Description</h3>

<p>Compute score at the MLE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score(mod, numerical = FALSE, get_score_covariance = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="score_+3A_numerical">numerical</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Indicator of whether to use the numeric Hessian and score (not recommended).</p>
</td></tr>
<tr><td><code id="score_+3A_get_score_covariance">get_score_covariance</code></td>
<td>
<p>Boolean. Defaults to <code>FALSE</code>. Should we return a robust estimate of variance of score: <code class="reqn">\hat{B}(\hat{\theta}) = \sum_i G(\hat{\theta}; W_i) G(\hat{\theta}; W_i)^T</code>. This parameter is not intended for users.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Score at the MLE. For <code class="reqn">G(\theta, w)</code> score function, returns <code class="reqn">\sum_i G(\hat{\theta}, W_i)</code> if get_score_covariance = FALSE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
score(mod)

</code></pre>

<hr>
<h2 id='simulate.bbdml'>Simulate from beta-binomial model</h2><span id='topic+simulate.bbdml'></span>

<h3>Description</h3>

<p>Simulate from beta-binomial model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bbdml'
simulate(object, nsim, seed = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate.bbdml_+3A_object">object</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="simulate.bbdml_+3A_nsim">nsim</code></td>
<td>
<p>Integer. Number of simulations</p>
</td></tr>
<tr><td><code id="simulate.bbdml_+3A_seed">seed</code></td>
<td>
<p>Optional integer to set a random seed</p>
</td></tr>
<tr><td><code id="simulate.bbdml_+3A_...">...</code></td>
<td>
<p>There are no additional parameters at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>nsim</code> simulations from <code>object</code>
</p>

<hr>
<h2 id='soil_phylo_otu'>Soil data, otu table as data frame.</h2><span id='topic+soil_phylo_otu'></span>

<h3>Description</h3>

<p>A data frame made from a soil 'phyloseq' object with only otu count data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylo_otu
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with an OTU table.
</p>

<dl>
<dt>otu_table</dt><dd><p>OTU table with 7,770 taxa and 119 samples</p>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylo_sample'>Soil data, sample data.</h2><span id='topic+soil_phylo_sample'></span>

<h3>Description</h3>

<p>A data frame made from a soil 'phyloseq' object with only sample data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylo_sample
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with sample data.
</p>

<dl>
<dt>sam_data</dt><dd><p>sample data with the following covariates:
</p>

<ul>
<li> <p><code>Plants</code>, values <code>0</code> and <code>1</code>. Index for different plants
</p>
</li>
<li> <p><code>Day</code>, values <code>0</code> (initial sampling point), <code>1</code> (12 days after treatment additions), and <code>2</code> (82 days after treatment additions). Index for different days of measurement
</p>
</li>
<li> <p><code>Amdmt</code>, values <code>0</code> (no additions), <code>1</code> (biochar additions), and <code>2</code> (fresh biomass additions). Index for different soil additives.
</p>
</li>
<li> <p><code>DayAmdmt</code>, values <code>00</code>, <code>01</code>, <code>02</code>, <code>10</code>, <code>11</code>, <code>12</code>, <code>20</code>, <code>21</code>, and <code>22</code>. A single index for the combination of <code>Day</code> and <code>Amdmt</code> with <code>Day</code> as the first digit and <code>Amdmt</code> as the second digit.
</p>
</li>
<li> <p><code>ID</code>, values <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>,  and <code>F</code>. Index for different soil plots.
</p>
</li></ul>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylo_taxa'>Soil data, taxa table as data frame.</h2><span id='topic+soil_phylo_taxa'></span>

<h3>Description</h3>

<p>A data frame made from a soil 'phyloseq' object with only taxonomy data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylo_taxa
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with an OTU table.
</p>

<dl>
<dt>tax_table</dt><dd><p>taxonomy table</p>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylum_contrasts_otu'>Small soil phylum data for contrasts examples, otu table as data frame</h2><span id='topic+soil_phylum_contrasts_otu'></span>

<h3>Description</h3>

<p>A small subset of <code><a href="#topic+soil_phylo_otu">soil_phylo_otu</a></code> used for examples of testing contrasts. A data frame made from the 'phyloseq' object with only otu counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylum_contrasts_otu
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with an OTU table.
</p>

<dl>
<dt>otu_table</dt><dd><p>OTU table with 39 taxa and 56 samples</p>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylum_contrasts_sample'>Small soil phylum data for contrasts examples, sample data as data frame</h2><span id='topic+soil_phylum_contrasts_sample'></span>

<h3>Description</h3>

<p>A small subset of <code><a href="#topic+soil_phylo_sample">soil_phylo_sample</a></code> used for examples of testing contrasts. A data frame made from the 'phyloseq' object with only sample data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylum_contrasts_sample
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with sample data.
</p>

<dl>
<dt>sam_data</dt><dd><p>sample data with the following covariates:
</p>

<ul>
<li> <p><code>Plants</code>, values <code>0</code> and <code>1</code>. Index for different plants
</p>
</li>
<li> <p><code>Day</code>, values <code>0</code> (initial sampling point), <code>1</code> (12 days after treatment additions), and <code>2</code> (82 days after treatment additions). Index for different days of measurement
</p>
</li>
<li> <p><code>Amdmt</code>, values <code>0</code> (no additions), <code>1</code> (biochar additions), and <code>2</code> (fresh biomass additions). Index for different soil additives.
</p>
</li>
<li> <p><code>DayAmdmt</code>, values <code>00</code>, <code>01</code>, <code>02</code>, <code>10</code>, <code>11</code>, <code>12</code>, <code>20</code>, <code>21</code>, and <code>22</code>. A single index for the combination of <code>Day</code> and <code>Amdmt</code> with <code>Day</code> as the first digit and <code>Amdmt</code> as the second digit.
</p>
</li>
<li> <p><code>ID</code>, values <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>,  and <code>F</code>. Index for different soil plots.
</p>
</li></ul>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylum_small_otu'>Small soil phylum data for examples, otu table as data frame</h2><span id='topic+soil_phylum_small_otu'></span>

<h3>Description</h3>

<p>A small subset of <code><a href="#topic+soil_phylo_otu">soil_phylo_otu</a></code> used for examples. A data frame made from the 'phyloseq' object with only otu counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylum_small_otu
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with an OTU table.
</p>

<dl>
<dt>otu_table</dt><dd><p>OTU table with 39 taxa and 32 samples</p>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylum_small_otu1'>Small soil phylum data for examples, sample data as data frame combined with counts for OTU 1 and sequencing depth.</h2><span id='topic+soil_phylum_small_otu1'></span>

<h3>Description</h3>

<p>A small subset of <code><a href="#topic+soil_phylo_sample">soil_phylo_sample</a></code> used for examples. A data frame made from the 'phyloseq' object with only sample data and counts for OTU 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylum_small_otu1
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with sample data and OTU 1 counts.
</p>

<dl>
<dt>sam_data</dt><dd><p>sample data with the following covariates:
</p>

<ul>
<li> <p><code>Plants</code>, values <code>0</code> and <code>1</code>. Index for different plants
</p>
</li>
<li> <p><code>Day</code>, values <code>0</code> (initial sampling point), <code>1</code> (12 days after treatment additions), and <code>2</code> (82 days after treatment additions). Index for different days of measurement
</p>
</li>
<li> <p><code>Amdmt</code>, values <code>0</code> (no additions), <code>1</code> (biochar additions), and <code>2</code> (fresh biomass additions). Index for different soil additives.
</p>
</li>
<li> <p><code>DayAmdmt</code>, values <code>00</code>, <code>01</code>, <code>02</code>, <code>10</code>, <code>11</code>, <code>12</code>, <code>20</code>, <code>21</code>, and <code>22</code>. A single index for the combination of <code>Day</code> and <code>Amdmt</code> with <code>Day</code> as the first digit and <code>Amdmt</code> as the second digit.
</p>
</li>
<li> <p><code>ID</code>, values <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>,  and <code>F</code>. Index for different soil plots.
</p>
</li>
<li> <p><code>W</code>, counts for OTU1 in each sample. This OTU corresponds with the phylum <em>Proteobacteria</em>.
</p>
</li>
<li> <p><code>M</code>, the sequencing depth for each sample.
</p>
</li></ul>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='soil_phylum_small_sample'>Small soil phylum data for examples, sample data as data frame</h2><span id='topic+soil_phylum_small_sample'></span>

<h3>Description</h3>

<p>A small subset of <code><a href="#topic+soil_phylo_sample">soil_phylo_sample</a></code> used for examples. A data frame made from the 'phyloseq' object with only sample data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>soil_phylum_small_sample
</code></pre>


<h3>Format</h3>

<p>A phyloseq-class experiment-level object with sample data.
</p>

<dl>
<dt>sam_data</dt><dd><p>sample data with the following covariates:
</p>

<ul>
<li> <p><code>Plants</code>, values <code>0</code> and <code>1</code>. Index for different plants
</p>
</li>
<li> <p><code>Day</code>, values <code>0</code> (initial sampling point), <code>1</code> (12 days after treatment additions), and <code>2</code> (82 days after treatment additions). Index for different days of measurement
</p>
</li>
<li> <p><code>Amdmt</code>, values <code>0</code> (no additions), <code>1</code> (biochar additions), and <code>2</code> (fresh biomass additions). Index for different soil additives.
</p>
</li>
<li> <p><code>DayAmdmt</code>, values <code>00</code>, <code>01</code>, <code>02</code>, <code>10</code>, <code>11</code>, <code>12</code>, <code>20</code>, <code>21</code>, and <code>22</code>. A single index for the combination of <code>Day</code> and <code>Amdmt</code> with <code>Day</code> as the first digit and <code>Amdmt</code> as the second digit.
</p>
</li>
<li> <p><code>ID</code>, values <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>,  and <code>F</code>. Index for different soil plots.
</p>
</li></ul>
</dd>
</dl>



<h3>References</h3>

<p>Whitman, T., Pepe-Ranney, C., Enders, A., Koechli, C., Campbell, A.,  Buckley, D. H., Lehmann, J. (2016). <em>Dynamics of microbial community composi-tion and soil organic carbon mineralization in soil following addition of pyrogenic andfresh organic matter</em>. The ISME journal, 10(12):2918. &lt;doi: 10.1038/ismej.2016.68&gt;.
</p>

<hr>
<h2 id='summary.bbdml'>Summary function</h2><span id='topic+summary.bbdml'></span>

<h3>Description</h3>

<p>Summary function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'bbdml'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.bbdml_+3A_object">object</code></td>
<td>
<p>Object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="summary.bbdml_+3A_...">...</code></td>
<td>
<p>No optional arguments are accepted at this time.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>summary.bbdml</code>. Displays printed model summary.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
summary(mod)
</code></pre>

<hr>
<h2 id='waldchisq'>Wald-type chi-squared test</h2><span id='topic+waldchisq'></span>

<h3>Description</h3>

<p>Wald-type chi-squared test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waldchisq(
  mod,
  mod_null = NULL,
  restrictions = NULL,
  restrictions.phi = NULL,
  contrasts_DA = NULL,
  contrasts_DV = NULL,
  robust = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="waldchisq_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="waldchisq_+3A_mod_null">mod_null</code></td>
<td>
<p>Optional. An object of class <code>bbdml</code>, should be nested within <code>mod</code>. If not included, need to include <code>restrictions</code> or <code>restrictions.phi</code>.</p>
</td></tr>
<tr><td><code id="waldchisq_+3A_restrictions">restrictions</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the abundance to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the abundance.</p>
</td></tr>
<tr><td><code id="waldchisq_+3A_restrictions.phi">restrictions.phi</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the dispersion to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the dispersion.</p>
</td></tr>
<tr><td><code id="waldchisq_+3A_contrasts_da">contrasts_DA</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>.</p>
</td></tr>
<tr><td><code id="waldchisq_+3A_contrasts_dv">contrasts_DV</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>phi.formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>.</p>
</td></tr>
<tr><td><code id="waldchisq_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be used? If not, model-based standard arras are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with wald test statistics and p-values. Only performs univariate tests.
</p>
<p>P-value from Wald test.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod1 &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)

mod2 &lt;- bbdml(formula = cbind(W, M - W) ~ 1,
phi.formula = ~ 1,
data = soil_phylum_small_otu1)

# Example using mod_null
waldchisq(mod = mod1, mod_null = mod2)
waldchisq(mod = mod1, mod_null = mod2, robust = TRUE)

# Example using restrictions and restrictions.phi
waldchisq(mod = mod1, restrictions = 2, restrictions.phi = 2)
waldchisq(mod = mod1, restrictions = "DayAmdmt", restrictions.phi = "DayAmdmt")
waldchisq(mod = mod1, restrictions = 2, restrictions.phi = "DayAmdmt")
waldchisq(mod = mod1, restrictions = 2, restrictions.phi = 2, robust = TRUE)
</code></pre>

<hr>
<h2 id='waldchisq_test'>Wald-type chi-squared test statistic (model-based or robust)</h2><span id='topic+waldchisq_test'></span>

<h3>Description</h3>

<p>This is a helper function and not intended for users
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waldchisq_test(
  mod,
  restrictions = NULL,
  restrictions.phi = NULL,
  contrasts_DA = NULL,
  contrasts_DV = NULL,
  robust = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="waldchisq_test_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
<tr><td><code id="waldchisq_test_+3A_restrictions">restrictions</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the abundance to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the abundance.</p>
</td></tr>
<tr><td><code id="waldchisq_test_+3A_restrictions.phi">restrictions.phi</code></td>
<td>
<p>Optional. Defaults to <code>NULL</code>. Numeric vector indicating the parameters associated with the dispersion to test, or character vector with name of variable to test. Note that <code>1</code> is the intercept associated with the dispersion.</p>
</td></tr>
<tr><td><code id="waldchisq_test_+3A_contrasts_da">contrasts_DA</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>.</p>
</td></tr>
<tr><td><code id="waldchisq_test_+3A_contrasts_dv">contrasts_DV</code></td>
<td>
<p>List. Optional. Constructs a contrast matrix. List elements should be characters specifying contrasts in the parameters within <code>phi.formula</code>. Note that this is only available with <code>"Wald"</code> value for <code>test</code>.</p>
</td></tr>
<tr><td><code id="waldchisq_test_+3A_robust">robust</code></td>
<td>
<p>Should robust standard errors be used? If not, model-based standard arras are used. Logical, defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Test statistic for Wald test.
</p>

<hr>
<h2 id='waldt'>Wald-type t test (model-based or robust)</h2><span id='topic+waldt'></span>

<h3>Description</h3>

<p>Wald-type t test (model-based or robust)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>waldt(mod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="waldt_+3A_mod">mod</code></td>
<td>
<p>an object of class <code>bbdml</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with wald test statistics and p-values. Only performs univariate tests.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soil_phylum_small_otu1)
mod &lt;- bbdml(formula = cbind(W, M - W) ~ DayAmdmt,
phi.formula = ~ DayAmdmt,
data = soil_phylum_small_otu1)
waldt(mod)

</code></pre>

<hr>
<h2 id='warn_phyloseq'>Function to throw error if the 'phyloseq' package is called but it is not installed</h2><span id='topic+warn_phyloseq'></span>

<h3>Description</h3>

<p>Function to throw error if the 'phyloseq' package is called but it is not installed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>warn_phyloseq()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
