<!DOCTYPE html><html lang="en"><head><title>Help for package corrDNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {corrDNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assoc_comb'>
<p>Complete association matrix.</p></a></li>
<li><a href='#assoc_Zi.Zj'>
<p>Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{j}</code>.</p></a></li>
<li><a href='#assoc_Zi.ZjR'>
<p>Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{jR}</code>.</p></a></li>
<li><a href='#assoc_Zi.ZjY'>
<p>Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{jY}</code>.</p></a></li>
<li><a href='#assoc_ZiR.ZjR'>
<p>Association between variable <code class="reqn">Z_{iR}</code> and <code class="reqn">Z_{jR}</code>.</p></a></li>
<li><a href='#assoc_ZiR.ZjY'>
<p>Association between variable <code class="reqn">Z_{iR}</code> and <code class="reqn">Z_{jY}</code>.</p></a></li>
<li><a href='#assoc_ZiY.ZjY'>
<p>Association between variable <code class="reqn">Z_{iY}</code> and <code class="reqn">Z_{jY}</code>.</p></a></li>
<li><a href='#don_dat'>
<p>A sample dataset of human donor splice sites.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Finding Associations in Position-Wise Aligned DNA Sequence
Dataset</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-03-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Prabina Kumar Meher &amp; A. R. Rao</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prabina Kumar Meher &lt;meherprabin@yahoo.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Description:</td>
<td>Can be useful for finding associations among different positions in a position-wise aligned sequence dataset. The approach adopted for finding associations among positions is based on the latent multivariate normal distribution.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-03-31 10:25:23 UTC; USER</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-04-05 15:31:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='assoc_comb'>
Complete association matrix.
</h2><span id='topic+assoc_comb'></span>

<h3>Description</h3>

<p>All the six possible association matrices can be merged in to a single matrix to visualize the overall association among positions as well as among the occurences of nucleotides of different positions, in a position-wise aligned sequence dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_comb(x, rZiZj, rZiZjR, rZiZjY, rZiRZjR, rZiRZjY, rZiYZjY)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_comb_+3A_x">x</code></td>
<td>

<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rzizj">rZiZj</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rzizjr">rZiZjR</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjR">assoc_Zi.ZjR</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rzizjy">rZiZjY</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjY">assoc_Zi.ZjY</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rzirzjr">rZiRZjR</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_ZiR.ZjR">assoc_ZiR.ZjR</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rzirzjy">rZiRZjY</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_ZiR.ZjY">assoc_ZiR.ZjY</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_comb_+3A_rziyzjy">rZiYZjY</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_ZiY.ZjY">assoc_ZiY.ZjY</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the six association matrices are required to be generated prior to merging them into a single matrix.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>3L</em> by <em>3L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjr &lt;- assoc_Zi.ZjR(x=kk, rZiZj=zizj)
zizjy &lt;- assoc_Zi.ZjY(x=kk, rZiZj=zizj)
zirzjr &lt;- assoc_ZiR.ZjR(x=kk,rZiZj=zizj,rZiZjR=zizjr)
zirzjy &lt;- assoc_ZiR.ZjY(x=kk,rZiZj=zizj,rZiZjR=zizjr,rZiZjY=zizjy)
ziyzjy &lt;- assoc_ZiY.ZjY(x=kk,rZiZj=zizj,rZiZjY=zizjy)
fin_corr &lt;- assoc_comb(x=kk, rZiZj=zizj,rZiZjR=zizjr,rZiZjY=zizjy,
rZiRZjR=zirzjr,rZiRZjY=zirzjy,rZiYZjY=ziyzjy)
fin_corr

</code></pre>

<hr>
<h2 id='assoc_Zi.Zj'>
Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{j}</code>.
</h2><span id='topic+assoc_Zi.Zj'></span>

<h3>Description</h3>

<p>Finding association between variables of <code class="reqn">i^{th}</code> position and <code class="reqn">j^{th}</code> position.
In any position wise aligned sequence dataset, occurences of R=(A,G) and Y=(C, T) at each position can be explained by a standard normal variate Z based on certain threshold value.
So, an association between any two position in the datast can be obtained which will be the association beween the two standard normal variate at this two positions.
However, the two nomal variates reprsenting the occurences of R and Y are independent of each other at a given position. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_Zi.Zj(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_Zi.Zj_+3A_x">x</code></td>
<td>
<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the sequence dataset in tab delimited format and not in FASTA format.
Each sequence (row) should contain only standard nucleotides (A, T, G and C).
Each sequence should be same length.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizj
</code></pre>

<hr>
<h2 id='assoc_Zi.ZjR'>
Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{jR}</code>.
</h2><span id='topic+assoc_Zi.ZjR'></span>

<h3>Description</h3>

<p>Finding association between variable Z at <code class="reqn">i^{th}</code> position and <code class="reqn">Z_{R}</code> at <code class="reqn">j^{th}</code> position.
Here, the standard normal variable Z represents the occurence of R=(A,G) and Y=(C, T) at each position in the position wise aligned dataset,
whereas the the standard normal variable <code class="reqn">Z_{R}</code> reprsents the occurences of nucleotides A and G at any position based on some threshold value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_Zi.ZjR(x, rZiZj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_Zi.ZjR_+3A_x">x</code></td>
<td>
<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_Zi.ZjR_+3A_rzizj">rZiZj</code></td>
<td>
<p> An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the input dataset as well as the output generated from the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Note</h3>

<p>It may happen that the convergence will not reach after a ceratin number of iterations and will not produce any output. 
In such situation, the user is advised to exclude or include some positions , or otherwise include or exclude certain sequences.
The user should exploit both options till convergence is reached.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjr &lt;- assoc_Zi.ZjR(x=kk, rZiZj=zizj)
zizjr
</code></pre>

<hr>
<h2 id='assoc_Zi.ZjY'>
Association between variable <code class="reqn">Z_{i}</code> and <code class="reqn">Z_{jY}</code>.
</h2><span id='topic+assoc_Zi.ZjY'></span>

<h3>Description</h3>

<p>Finding association between variable Z at <code class="reqn">i^{th}</code> position and <code class="reqn">Z_{Y}</code> at <code class="reqn">j^{th}</code> position.
Here, the standard normal variable Z represents the occurence of R=(A,G) and Y=(C, T) at each position in the position wise aligned dataset,
whereas the the standard normal variable <code class="reqn">Z_{R}</code> reprsents the occurences of nucleotides A and G at any position based on some threshold values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_Zi.ZjY(x, rZiZj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_Zi.ZjY_+3A_x">x</code></td>
<td>

<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_Zi.ZjY_+3A_rzizj">rZiZj</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the input dataset as well as the output generated from the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Note</h3>

<p>It may happen that the convergence will not reach after a ceratin number of iterations and will not produce any output. 
In such situation, the user is advised to exclude or include some positions , or otherwise include or exclude certain sequences.
The user should exploit both options till convergence is reached.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjy &lt;- assoc_Zi.ZjY(x=kk, rZiZj=zizj)
zizjy
</code></pre>

<hr>
<h2 id='assoc_ZiR.ZjR'>
Association between variable <code class="reqn">Z_{iR}</code> and <code class="reqn">Z_{jR}</code>.
</h2><span id='topic+assoc_ZiR.ZjR'></span>

<h3>Description</h3>

<p>Finding association between variable <code class="reqn">Z_{R}</code> at <code class="reqn">i^{th}</code> position and <code class="reqn">Z_{R}</code> at <code class="reqn">j^{th}</code> position.
Here, the standard normal variable <code class="reqn">Z_{R}</code> reprsents the occurences of nucleotides A and G at any position based on some threshold value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_ZiR.ZjR(x, rZiZj, rZiZjR)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_ZiR.ZjR_+3A_x">x</code></td>
<td>

<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_ZiR.ZjR_+3A_rzizj">rZiZj</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_ZiR.ZjR_+3A_rzizjr">rZiZjR</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjR">assoc_Zi.ZjR</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the input dataset as well as the outputs generated from the functions <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code> and <code><a href="#topic+assoc_Zi.ZjR">assoc_Zi.ZjR</a></code>.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Note</h3>

<p>It may happen that the convergence will not reach after a ceratin number of iterations and will not produce any output. 
In such situation, the user is advised to exclude or include some positions , or otherwise include or exclude certain sequences.
The user should exploit both options till convergence is reached.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjr &lt;- assoc_Zi.ZjR(x=kk, rZiZj=zizj)
zirzjr &lt;- assoc_ZiR.ZjR(x=kk, rZiZj=zizj, rZiZjR=zizjr)
zirzjr

</code></pre>

<hr>
<h2 id='assoc_ZiR.ZjY'>
Association between variable <code class="reqn">Z_{iR}</code> and <code class="reqn">Z_{jY}</code>.
</h2><span id='topic+assoc_ZiR.ZjY'></span>

<h3>Description</h3>

<p>Finding association between variable <code class="reqn">Z_{R}</code> at <code class="reqn">i^{th}</code> position and <code class="reqn">Z_{Y}</code> at <code class="reqn">j^{th}</code> position.
Here, the standard normal variable <code class="reqn">Z_{Y}</code> represents the occurences C and T at each position in the position wise aligned dataset, 
and the standard normal variable <code class="reqn">Z_{R}</code> reprsents the occurences of nucleotides A and G at any position based on some threshold values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_ZiR.ZjY(x, rZiZj, rZiZjR, rZiZjY)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_ZiR.ZjY_+3A_x">x</code></td>
<td>

<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_ZiR.ZjY_+3A_rzizj">rZiZj</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_ZiR.ZjY_+3A_rzizjr">rZiZjR</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjR">assoc_Zi.ZjR</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_ZiR.ZjY_+3A_rzizjy">rZiZjY</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjY">assoc_Zi.ZjY</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the input dataset as well as the outputs generated from the functions <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>, <code><a href="#topic+assoc_Zi.ZjR">assoc_Zi.ZjR</a></code> and <code><a href="#topic+assoc_Zi.ZjY">assoc_Zi.ZjY</a></code>.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Note</h3>

<p>It may happen that the convergence will not reach after a ceratin number of iterations and will not produce any output. 
In such situation, the user is advised to exclude or include some positions , or otherwise include or exclude certain sequences.
The user should exploit both options till convergence is reached.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjr &lt;- assoc_Zi.ZjR(x=kk, rZiZj=zizj)
zizjy &lt;- assoc_Zi.ZjY(x=kk, rZiZj=zizj)
zirzjy &lt;- assoc_ZiR.ZjY(x=kk, rZiZj=zizj, rZiZjR=zizjr, rZiZjY=zizjy)
zirzjy

</code></pre>

<hr>
<h2 id='assoc_ZiY.ZjY'>
Association between variable <code class="reqn">Z_{iY}</code> and <code class="reqn">Z_{jY}</code>.
</h2><span id='topic+assoc_ZiY.ZjY'></span>

<h3>Description</h3>

<p>Finding association between variable <code class="reqn">Z_{Y}</code> at <code class="reqn">i^{th}</code> position and <code class="reqn">Z_{Y}</code> at <code class="reqn">j^{th}</code> position.
Here, the standard normal variable <code class="reqn">Z_{Y}</code> reprsents the occurences of nucleotides C and T at any position based on some threshold values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assoc_ZiY.ZjY(x, rZiZj, rZiZjY)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assoc_ZiY.ZjY_+3A_x">x</code></td>
<td>

<p>A dataframe of position wise aligned sequence dataset having A, T, G and C only.
</p>
</td></tr>
<tr><td><code id="assoc_ZiY.ZjY_+3A_rzizj">rZiZj</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code>.
</p>
</td></tr>
<tr><td><code id="assoc_ZiY.ZjY_+3A_rzizjy">rZiZjY</code></td>
<td>

<p>An object generated by using the function <code><a href="#topic+assoc_Zi.ZjY">assoc_Zi.ZjY</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user has to supply the input dataset as well as the outputs generated from the functions <code><a href="#topic+assoc_Zi.Zj">assoc_Zi.Zj</a></code> and <code><a href="#topic+assoc_Zi.ZjY">assoc_Zi.ZjY</a></code>.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <em>L</em> by <em>L</em> for the dataset of <em>L</em> nucleotides long sequences.
</p>


<h3>Note</h3>

<p>It may happen that the convergence will not reach after a ceratin number of iterations and will not produce any output. 
In such situation, the user is advised to exclude or include some positions , or otherwise include or exclude certain sequences.
The user should exploit both options till convergence is reached.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher &amp; A. R. Rao
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(don_dat)
kk &lt;- don_dat[1:300,]
zizj &lt;- assoc_Zi.Zj(x=kk)
zizjy &lt;- assoc_Zi.ZjY(x=kk, rZiZj=zizj)
ziyzjy &lt;- assoc_ZiY.ZjY(x=kk, rZiZj=zizj, rZiZjY=zizjy)
ziyzjy

</code></pre>

<hr>
<h2 id='don_dat'>
A sample dataset of human donor splice sites.
</h2><span id='topic+don_dat'></span>

<h3>Description</h3>

<p>This dataset comprises 1000 donor splice site sequences, where each sequence is of length 20 with 10 at the exon end and 10 at the intron start excluding the conserved di-nucleotide GT at the begining of intron.
This dataset was randomly taken from true donor splice sites of HS3D dataset. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(don_dat)</code></pre>


<h3>References</h3>

<p>Pollastro P, Rampone S: HS3D: Homosapiens Splice Site Data Set. <em>Nucleic Acids Res</em>. 2003, Molecular Biology Database Collection entry number 36; Annual Database Issue.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(don_dat)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
