<!DOCTYPE html><html><head><title>Help for package CorrectedFDR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CorrectedFDR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CorrectedFDR-package'>
<p>Correcting False Discovery Rates</p></a></li>
<li><a href='#BlendedLFDR'><p>Blended Estimator of Local False Discovery Rate (LFDR)</p></a></li>
<li><a href='#EstimatorsFDR'><p>Estimators of Local False Discovery Rate (LFDR)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Correcting False Discovery Rates</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-10-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Abbas Rahal, Anna Akpawu, Justin Chitpin and David R. Bickel</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Abbas Rahal &lt;Abbas.Rahal13@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>There are many estimators of false discovery rate. In this package we compute the Nonlocal False
   Discovery Rate (NFDR) and the estimators of local false discovery rate: Corrected False discovery
   Rate (CFDR), Re-ranked False Discovery rate (RFDR) and the blended estimator. 
   Bickel, D.R., Rahal, A. (2019) <a href="https://tinyurl.com/kkdc9rk8">https://tinyurl.com/kkdc9rk8</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.14.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>LFDR.MLE, LFDREmpiricalBayes, ProData</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bayesian Statistics, MathematicalBiology</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-06 17:58:35 UTC; Rahal</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-07 13:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='CorrectedFDR-package'>
Correcting False Discovery Rates
</h2><span id='topic+CorrectedFDR-package'></span>

<h3>Description</h3>

<p>There are many estimators of false discovery rate. In this package we compute the Nonlocal False
Discovery Rate (NFDR) and the estimators of local false discovery rate: Corrected False discovery
Rate (CFDR), Re-ranked False Discovery rate (RFDR) and the blended estimator. 
Bickel, D.R., Rahal, A. (2019) &lt;https://tinyurl.com/kkdc9rk8&gt;.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> CorrectedFDR</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2021-10-06</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R(&gt;= 2.14.2)</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> LFDR.MLE, LFDREmpiricalBayes, ProData</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Two functions in <code>CorrectedFDR</code> package to compute the LFDR estimators. The function <code>EstimatorsFDR</code> computes the nonlocal false discovery rate (NFDR), the CFDR and the RFDR. The function <code>BlendedLFDR</code> uses a Benchmark of FDR, and other estimators of LFDR in order to get an estimate of LFDR.
</p>


<h3>Author(s)</h3>

<p>Abbas Rahal, Anna Akpawu, Justin Chitpin and David R. Bickel
</p>
<p>Maintainer: Abbas Rahal &lt;Abbas.Rahal13@gmail.com&gt;
</p>


<h3>References</h3>

<p>Bickel, D.R., Rahal, A. (2019). Correcting false discovery rates for their bias toward
false positives. Communications in Statistics - Simulation and Computation, <a href="https://tinyurl.com/kkdc9rk8">https://tinyurl.com/kkdc9rk8</a>.
</p>
<p>Bickel, D. R. (2015). Corrigendum to: Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2015, 14, 225.
</p>
<p>Bickel, D. R. (2015). Blending Bayesian and frequentist methods according to the precision of prior information with applications to hypothesis testing. Statistical Methods and Applications, 24(4), pp. 523-546.
</p>
<p>Bickel, D. R. (2013). Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2013, 12, 529-543.
</p>

<hr>
<h2 id='BlendedLFDR'>Blended Estimator of Local False Discovery Rate (LFDR)</h2><span id='topic+BlendedLFDR'></span>

<h3>Description</h3>

<p>BlendedLFDR is a function used to compute the blended estimator based on a benchmark estimator, usually the nonlocal false discovery rate (NFDR), and a set of estimators of local false discovery rates (LFDR).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BlendedLFDR(Benchmark, EstLFDR)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BlendedLFDR_+3A_benchmark">Benchmark</code></td>
<td>
<p>Input numeric vector for benchmark estimator (often NFDR).</p>
</td></tr>
<tr><td><code id="BlendedLFDR_+3A_estlfdr">EstLFDR</code></td>
<td>
<p>Input a matrix containing two or more sets of LFDR estimators.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Benchmark</code> is an estimator of the FDR. This is usually the nonlocal false discovery rate (NFDR). <br /> <code>EstLFDR</code> is a matrix of several LFDR estimators such as corrected FDR (CFDR), re-ranked FDR (RFDR), MLE (Maximum Likelihood Estimator), BBE1(Binomial Based Estimator), etc. <br /> The output returns a single numeric vector containing the blended estimator of the LFDR.
</p>


<h3>Value</h3>

<p>The value of the blended estimator is an estimator of the LFDR.
</p>


<h3>Note</h3>

<p>The number of rows for the <code>Benchmark</code> and <code>EstLFDR</code> must have equal lengths.
</p>


<h3>Author(s)</h3>

<p>Code: Abbas Rahal. <br /> Documentation: Anna Akpawu, Justin Chitpin and Abbas Rahal. <br /> Maintainer: Abbas Rahal &lt;Abbas.Rahal13@gmail.com&gt;
</p>


<h3>References</h3>

<p>Bickel, D. R. (2015). Blending Bayesian and frequentist methods according to the precision of prior information with applications to hypothesis testing. Statistical Methods and Applications, 24(4), pp. 523-546.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The data used to compute the LFDR estimators (CFDR, RFDR, MLE, and BBE1)
#comes from the ER/PR breast cancer data from the "ProData" package.
#To read more about the data, visit the website: https://www.bioconductor.org/
#Test statistics were first obtain, then the estimators for the FDR and LFDR were estimated.
#Benchmark vector
NFDR&lt;-c(0.5661106448, 0.6897735492, 0.0000288516, 0.1549745113, 0.1305508970, 0.2421032979,
0.1482335568, 1, 1, 1, 0.6602562820, 0.7034682859, 0.7036332234, 0.0071192090,
0.8204536037, 0.9757716498, 0.7379329991, 1, 0.6333245479, 0.9904389701)
#Estimators of LFDR
CFDR&lt;- c(1, 1, 0.0000288516, 0.2841199373, 0.2980912149, 0.5931530799, 0.3088199101,
1, 1, 1, 1, 1, 1, 0.0106788135, 1, 1, 1, 1, 1, 1)

RFDR&lt;- c(0.689773549, 1, 0.007119209, 0.130550897, 0.703633223, 0.660256282, 0.242103298,
1, 1, 1, 0.820453604, 1, 0.703468286, 0.154974511, 1, 1, 1, 1, 0.975771650,1)

MLE&lt;- c(0.9865479126, 0.9969935995, 0.0002372158, 0.6531633437, 0.7611453549, 0.9187425383,
0.7359259207, 0.9996548155, 0.9997310453, 0.9997437131, 0.9944712582, 0.9981685029,
0.9937604664, 0.0215892618, 0.9990504315, 0.9997493086, 0.9967673540, 0.9997016985,
0.9970142319, 0.9997625673)

BBE1&lt;- c(1,1, 0.0003169812, 0.1138333734, 1, 1, 1, 1, 1, 1, 0.3279109564, 1, 0.0504755806,
0.0091823115, 0.0182614994, 0.0165386682, 1, 0.6964403713, 0.1001337298, 0.8415641198 )

#Matrix of LFDR Estimators
Est.LFDR&lt;- matrix(c(CFDR,RFDR,MLE,BBE1), ncol=4)
output&lt;-BlendedLFDR(Benchmark = NFDR, EstLFDR = Est.LFDR)
output$Blended
</code></pre>

<hr>
<h2 id='EstimatorsFDR'>Estimators of Local False Discovery Rate (LFDR)</h2><span id='topic+EstimatorsFDR'></span>

<h3>Description</h3>

<p>EstimatorsFDR is an R function that computes the Nonlocal False Discovery Rate (NFDR) and the estimators of local false discovery rate: Corrected False discovery Rate (CFDR) and Re-ranked False Discovery rate (RFDR).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EstimatorsFDR(pvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstimatorsFDR_+3A_pvalue">pvalue</code></td>
<td>
<p>Input numeric vector of pvalues.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input is a list of pvalues. The pvalues can be obtained for example by performing Student's t-test between two datasets. The two groups can be data from healthy and disease states. Let <code class="reqn">i=1, 2, ..., N</code>, where <code class="reqn">i</code> represents the <code class="reqn">ith</code> feature (SNP or gene, for example). Then, for each <code class="reqn">i</code>, the hypothesis indicator <code class="reqn">A_i</code> can have two possible values.<br /> <code class="reqn">A_i=0</code>, if the <code class="reqn">ith</code> null hypothesis is true, or <br /> <code class="reqn">A_i=1</code>,  if the <code class="reqn">ith</code> null hypothesis is not true, <br />
where the null hypothesis is defined by: the <code class="reqn">ith</code> feature is unaffected by a treatment, unassociated with a disease, etc.
The values for each estimator (NFDR, CFDR, RFDR) indicate the probability that the null hypothesis of the <code class="reqn">ith</code> feature is true (<code class="reqn">A_i=0</code>) given the statistics <code class="reqn">T_i</code>. The alternative hypothesis is true if <code class="reqn">A_i=1</code>.
For example, in gene expression data analysis, if the null hypothesis is true, this would mean that the genes are not differentially expressed.
</p>


<h3>Value</h3>

<p>The output returns three lists. It returns the NFDR, CFDR, and RFDR estimators: </p>

<table>
<tr>
 <td style="text-align: left;"> NFDR</td><td style="text-align: left;"> nonlocal FDR</td>
</tr>
<tr>
 <td style="text-align: left;"> CFDR</td><td style="text-align: left;"> corrected FDR</td>
</tr>
<tr>
 <td style="text-align: left;"> RFDR</td><td style="text-align: left;"> re-ranked FDR</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Code: Abbas Rahal. <br /> Documentation: Anna Akpawu, Justin Chitpin and Abbas Rahal. <br /> Maintainer: Abbas Rahal &lt;Abbas.Rahal13@gmail.com&gt;
</p>


<h3>References</h3>

<p>Bickel, D.R., Rahal, A. (2019). Correcting false discovery rates for their bias toward
false positives. Communications in Statistics - Simulation and Computation, <a href="https://tinyurl.com/kkdc9rk8">https://tinyurl.com/kkdc9rk8</a>.
</p>
<p>Bickel, D. R. (2015). Corrigendum to: Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2015, 14, 225.
</p>
<p>Bickel, D. R. (2013). Simple estimators of false discovery rates given as few as one or two p-values without strong parametric assumptions. Statistical Applications in Genetics and Molecular Biology, 2013, 12, 529-543.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The examples below are from the "ProData" package.
#In order to use the "Prodata" input you would first need to install the ProData package.
#You will also need the function exprs in this package.
#First, make sure that the ProData package is properly installed:
#source("https://bioconductor.org/biocLite.R")
#biocLite("ProData")
#library(ProData)
#data("f45cbmk")
#q1&lt;- quantile(as(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "B"]), "numeric"), probs = 0.25)
#logish&lt;- function(x){log(x + q1)}
#Vectors of proteins for 20 patients ER/PR-positive and Healthy
#Y&lt;- logish(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "B"])) # Control (Healthy)
#X.ER&lt;- logish(exprs(f45cbmk[, pData(f45cbmk)$GROUP == "C"])) # Case ER/PR-positive
#pvalue&lt;- NULL
#for (i in 1:nrow(X.ER))
#{
# t&lt;-t.test(x=X.ER[i,], y=Y[i,], alternative = "two.sided")
# pvalue[i]&lt;- t$p.value
#}

#The pvalues obtained from the t-test:
pvalue&lt;- c(0.1981, 0.3794, 0.000001443, 0.02325, 0.03264, 0.07263, 0.02965, 0.8016, 0.8888,
          0.9133, 0.2971, 0.4573, 0.2815, 0.0007119, 0.5743, 0.927, 0.369, 0.8478, 0.38, 0.9904)
output&lt;- EstimatorsFDR(pvalue)
#Three lists
output$NFDR
output$CFDR
output$RFDR
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
