<!DOCTYPE html><html lang="en"><head><title>Help for package corto</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {corto}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#barplot2'><p>barplot2 - Bar plot with upper error bars</p></a></li>
<li><a href='#corto'><p>Calculate a regulon from a data matrix</p></a></li>
<li><a href='#fcor'><p>A fast correlation function</p></a></li>
<li><a href='#fisherp'><p>Fisher integration of p-values</p></a></li>
<li><a href='#gsea'><p>GSEA</p></a></li>
<li><a href='#gsea2'><p>2-way GSEA</p>
GSEA Gene set enrichment analysis of two complementary gene sets using gsea</a></li>
<li><a href='#kmgformat'><p>kmgformat - Nice Formatting of Numbers</p></a></li>
<li><a href='#mra'><p>Perform Master Regulator Analysis (mra).</p></a></li>
<li><a href='#mraplot'><p>Plot a master regulator analysis</p></a></li>
<li><a href='#p2r'><p>p2r Convert a P-value to the corresponding Correlation Coefficient</p></a></li>
<li><a href='#p2z'><p>p2z</p></a></li>
<li><a href='#plot_gsea'><p>Plot GSEA results</p></a></li>
<li><a href='#plot_gsea2'><p>Plot 2-way GSEA results</p></a></li>
<li><a href='#r2p'><p>r2p Convert Correlation Coefficient to P-value</p></a></li>
<li><a href='#scatter'><p>scatter - XY scatter plot with extra information</p></a></li>
<li><a href='#scinot'><p>scinot - Convert a number to a scientific notation expression</p></a></li>
<li><a href='#slice'><p>Slice</p></a></li>
<li><a href='#ssgsea'><p>ssGSEA</p></a></li>
<li><a href='#stouffer'><p>Stouffer integration of Z scores</p></a></li>
<li><a href='#textrepel'><p>textrepel - Plot text with non-overlapping labels</p></a></li>
<li><a href='#val2col'><p>val2col - Convert a numeric vector into colors</p></a></li>
<li><a href='#wstouffer'><p>Weighted Stouffer integration of Z scores</p></a></li>
<li><a href='#z2p'><p>z2p</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inference of Gene Regulatory Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Federico M. Giorgi &lt;federico.giorgi@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We present 'corto' (Correlation Tool), a simple package to infer
 gene regulatory networks and visualize master regulators from gene expression
 data using DPI (Data Processing Inequality) and bootstrapping to recover edges.
 An initial step is performed to calculate all significant
 edges between a list of source nodes (centroids) and target genes.
 Then all triplets containing two centroids and one target are tested
 in a DPI step which removes edges. A bootstrapping process then calculates
 the robustness of the network, eventually re-adding edges previously removed by DPI.
 The algorithm has been optimized to run outside a computing cluster, using a fast correlation
 implementation. The package finally provides functions to calculate network enrichment
 analysis from RNA-Seq and ATAC-Seq signatures as described in the article by
 Giorgi lab (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtaa223">doi:10.1093/bioinformatics/btaa223</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, gplots, knitr, methods, rmarkdown, parallel, pbapply,
plotrix, stats, utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-06 18:47:20 UTC; feder</td>
</tr>
<tr>
<td>Author:</td>
<td>Federico M. Giorgi [aut, cre],
  Daniele Mercatelli [ctb],
  Gonzalo Lopez-Garcia [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-06 23:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='barplot2'>barplot2 - Bar plot with upper error bars</h2><span id='topic+barplot2'></span>

<h3>Description</h3>

<p>barplot2 - Bar plot with upper error bars
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barplot2(values, errors, lower = FALSE, flat = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barplot2_+3A_values">values</code></td>
<td>
<p>A matrix of values</p>
</td></tr>
<tr><td><code id="barplot2_+3A_errors">errors</code></td>
<td>
<p>A matrix of values for upper error bar</p>
</td></tr>
<tr><td><code id="barplot2_+3A_lower">lower</code></td>
<td>
<p>Boolean, whether the lower error bar should be plotted, default FALSE</p>
</td></tr>
<tr><td><code id="barplot2_+3A_flat">flat</code></td>
<td>
<p>Boolean, whether the head of bars should be flat, default TRUE</p>
</td></tr>
<tr><td><code id="barplot2_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to the core _barplot_ function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>values&lt;-matrix(rnorm(10*4,mean=10),nrow=4,ncol=10)
errors&lt;-matrix(runif(10*4),nrow=4,ncol=10)
colnames(values)&lt;-colnames(errors)&lt;-LETTERS[1:10]
barplot2(values,errors,main="Bar plot with error bars")
</code></pre>

<hr>
<h2 id='corto'>Calculate a regulon from a data matrix</h2><span id='topic+corto'></span>

<h3>Description</h3>

<p>This function applies Correlation and DPI to generate a robust
regulon object based on the input data matrix and the selected centroids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corto(
  inmat,
  centroids,
  nbootstraps = 100,
  p = 1e-30,
  nthreads = 1,
  verbose = FALSE,
  cnvmat = NULL,
  boot_threshold = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="corto_+3A_inmat">inmat</code></td>
<td>
<p>Input matrix, with features (e.g. genes) as rows and samples
as columns</p>
</td></tr>
<tr><td><code id="corto_+3A_centroids">centroids</code></td>
<td>
<p>A character vector indicating which features (e.g. genes)
to consider as centroids (a.k.a. Master Regulators) for DPI</p>
</td></tr>
<tr><td><code id="corto_+3A_nbootstraps">nbootstraps</code></td>
<td>
<p>Number of bootstraps to be performed. Default is 100</p>
</td></tr>
<tr><td><code id="corto_+3A_p">p</code></td>
<td>
<p>The p-value threshold for correlation significance (by default 1E-30)</p>
</td></tr>
<tr><td><code id="corto_+3A_nthreads">nthreads</code></td>
<td>
<p>The number of threads to use for bootstrapping. Default is 1</p>
</td></tr>
<tr><td><code id="corto_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Whether to print progress messages. Default is FALSE</p>
</td></tr>
<tr><td><code id="corto_+3A_cnvmat">cnvmat</code></td>
<td>
<p>An optional matrix with copy-number variation data. If specified, the program
will calculate linear regression between the gene expression data in the input matrix (exp)
and the cnv data, and target profiles will be transformed to the residuals
of each linear model exp~cnv. Default is NULL</p>
</td></tr>
<tr><td><code id="corto_+3A_boot_threshold">boot_threshold</code></td>
<td>
<p>The fraction of bootstraps in which the edge should appear
to be included in the final network. It can be any number between 0.0 and 1.0.
Default is 0.0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list (object of class regulon), where each element is a centroid
</p>

<ul>
<li><p> tfmode: a named vector containing correlation coefficients between
features and the centroid
</p>
</li>
<li><p> likelihood: a numeric vector indicating the likelihood of interaction
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Load data matrix inmat (from TCGA mesothelioma project)
load(system.file("extdata","inmat.rda",package="corto",mustWork=TRUE))
# Load centroids
load(system.file("extdata","centroids.rda",package="corto",mustWork=TRUE))
# Run corto
regulon &lt;- corto(inmat,centroids=centroids,nthreads=2,nbootstraps=10,verbose=TRUE)

# In a second example, a CNV matrix is provided. The analysis will be run only
# for the features (rows) and samples (columns) present in both matrices
load(system.file("extdata","cnvmat.rda",package="corto",mustWork=TRUE))
regulon &lt;- corto(inmat,centroids=centroids,nthreads=2,nbootstraps=6,verbose=TRUE,cnvmat=cnvmat,
p=1e-8)
</code></pre>

<hr>
<h2 id='fcor'>A fast correlation function</h2><span id='topic+fcor'></span>

<h3>Description</h3>

<p>A fast correlation function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcor(inmat, centroids, r)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fcor_+3A_inmat">inmat</code></td>
<td>
<p>An input matrix with features as rows and samples as columns</p>
</td></tr>
<tr><td><code id="fcor_+3A_centroids">centroids</code></td>
<td>
<p>A character vector indicating the centroids</p>
</td></tr>
<tr><td><code id="fcor_+3A_r">r</code></td>
<td>
<p>A numeric correlation threshold</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix describing which edges were significant in the input matrix
matrix according to the r correlation threshold provided
</p>

<hr>
<h2 id='fisherp'>Fisher integration of p-values</h2><span id='topic+fisherp'></span>

<h3>Description</h3>

<p>This function applies the Fisher integration of pvalues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisherp(ps)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisherp_+3A_ps">ps</code></td>
<td>
<p>a vector of p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p.val an integrated p-value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ps&lt;-c(0.01,0.05,0.03,0.2)
fisherp(ps)
</code></pre>

<hr>
<h2 id='gsea'>GSEA</h2><span id='topic+gsea'></span>

<h3>Description</h3>

<p>This function performs Gene Set Enrichment Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea(
  reflist,
  set,
  method = c("permutation", "pareto"),
  np = 1000,
  w = 1,
  gsea_null = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsea_+3A_reflist">reflist</code></td>
<td>
<p>named vector of reference scores</p>
</td></tr>
<tr><td><code id="gsea_+3A_set">set</code></td>
<td>
<p>element set</p>
</td></tr>
<tr><td><code id="gsea_+3A_method">method</code></td>
<td>
<p>one of 'permutation' or 'pareto'</p>
</td></tr>
<tr><td><code id="gsea_+3A_np">np</code></td>
<td>
<p>Number of permutations (Default: 1000)</p>
</td></tr>
<tr><td><code id="gsea_+3A_w">w</code></td>
<td>
<p>exponent used to raise the supplied scores. Default is 1 (original
scores unchanged)</p>
</td></tr>
<tr><td><code id="gsea_+3A_gsea_null">gsea_null</code></td>
<td>
<p>a GSEA null distribution (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A GSEA object. Basically a list of s components:
</p>

<dl>
<dt>ES</dt><dd><p>The enrichment score</p>
</dd>
<dt>NES</dt><dd><p>The normalized enrichment socre</p>
</dd>
<dt>ledge</dt><dd><p>The items in the leading edge</p>
</dd>
<dt>p.value</dt><dd><p>The permutation-based p-value</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>reflist&lt;-setNames(-sort(rnorm(1000)),paste0('gene',1:1000))
set&lt;-paste0('gene',sample(1:200,50))
obj&lt;-gsea(reflist,set,method='pareto',np=1000)
obj$p.value
</code></pre>

<hr>
<h2 id='gsea2'>2-way GSEA
GSEA Gene set enrichment analysis of two complementary gene sets using gsea</h2><span id='topic+gsea2'></span>

<h3>Description</h3>

<p>2-way GSEA
GSEA Gene set enrichment analysis of two complementary gene sets using gsea
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsea2(
  reflist,
  set1,
  set2,
  method = c("permutation", "pareto"),
  np = 1000,
  w = 1,
  gsea_null = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsea2_+3A_reflist">reflist</code></td>
<td>
<p>named vector of reference scores</p>
</td></tr>
<tr><td><code id="gsea2_+3A_set1">set1</code></td>
<td>
<p>element set 1</p>
</td></tr>
<tr><td><code id="gsea2_+3A_set2">set2</code></td>
<td>
<p>element set 1</p>
</td></tr>
<tr><td><code id="gsea2_+3A_method">method</code></td>
<td>
<p>one of 'permutation' or 'pareto'</p>
</td></tr>
<tr><td><code id="gsea2_+3A_np">np</code></td>
<td>
<p>Number of permutations (Default: 1000)</p>
</td></tr>
<tr><td><code id="gsea2_+3A_w">w</code></td>
<td>
<p>exponent used to raise the supplied scores. Default is 1 (original
scores unchanged)</p>
</td></tr>
<tr><td><code id="gsea2_+3A_gsea_null">gsea_null</code></td>
<td>
<p>a GSEA null distribution (Optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 2 GSEA objects. Each of which is a list of components:
</p>

<dl>
<dt>ES</dt><dd><p>The enrichment score</p>
</dd>
<dt>NES</dt><dd><p>The normalized enrichment socre</p>
</dd>
<dt>ledge</dt><dd><p>The items in the leading edge</p>
</dd>
<dt>p.value</dt><dd><p>The permutation-based p-value</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>reflist&lt;-setNames(-sort(rnorm(1000)),paste0('gene',1:1000))
set1&lt;-paste0('gene',sample(1:200,50))
set2&lt;-paste0('gene',sample(801:1000,50))
obj&lt;-gsea2(reflist,set1,set2,method='pareto',np=1000)
obj$p.value
</code></pre>

<hr>
<h2 id='kmgformat'>kmgformat - Nice Formatting of Numbers</h2><span id='topic+kmgformat'></span>

<h3>Description</h3>

<p>This function will convert thousand numbers to K, millions to M, billions
to G, trillions to T, quadrillions to P
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmgformat(input, roundParam = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kmgformat_+3A_input">input</code></td>
<td>
<p>A vector of values</p>
</td></tr>
<tr><td><code id="kmgformat_+3A_roundparam">roundParam</code></td>
<td>
<p>How many decimal digits you want</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of formatted numebr names
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Thousands
set.seed(1)
a&lt;-runif(1000,0,1e4)
plot(a,yaxt='n')
kmg&lt;-kmgformat(pretty(a))
axis(2,at=pretty(a),labels=kmg)

# Millions to Billions
set.seed(1)
a&lt;-runif(1000,0,1e9)
plot(a,yaxt='n',pch=20,col="black")
kmg&lt;-kmgformat(pretty(a))
axis(2,at=pretty(a),labels=kmg)
</code></pre>

<hr>
<h2 id='mra'>Perform Master Regulator Analysis (mra).</h2><span id='topic+mra'></span>

<h3>Description</h3>

<p>The analysis is performed between two groups of samples in the form of expression matrices,
with genes/features as rows and samples as columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mra(
  expmat1,
  expmat2 = NULL,
  regulon,
  minsize = 10,
  nperm = NULL,
  nthreads = 2,
  verbose = FALSE,
  atacseq = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mra_+3A_expmat1">expmat1</code></td>
<td>
<p>A numeric expression matrix, with genes/features as rows and samples as columns.
If only expmat1 is provided (without expmat2), the function will perform a sample-by-sample
master regulator analysis, with the mean of the dataset as a reference. If expmat2 is provided,
expmat1 will be considered the &quot;treatment&quot; sample set. If a named vector is provided, with names
as genes/features and values as signature values (e.g. T-test statistics), signature
master regulator analysis is performed.</p>
</td></tr>
<tr><td><code id="mra_+3A_expmat2">expmat2</code></td>
<td>
<p>A numeric expression matrix, with genes/features as rows and samples as columns.
If provided, it will be considered as the &quot;control&quot; or &quot;reference&quot; sample set for expmat1.</p>
</td></tr>
<tr><td><code id="mra_+3A_regulon">regulon</code></td>
<td>
<p>A _regulon_ object, output of the _corto_ function.</p>
</td></tr>
<tr><td><code id="mra_+3A_minsize">minsize</code></td>
<td>
<p>A minimum network size for each centroid/TF to be analyzed. Default is 10.</p>
</td></tr>
<tr><td><code id="mra_+3A_nperm">nperm</code></td>
<td>
<p>The number of times the input data will be permuted to generate null signatures.
Default is 1000 if expmat2 is provided, and 10 if expmat2 is not provided (single sample mra).</p>
</td></tr>
<tr><td><code id="mra_+3A_nthreads">nthreads</code></td>
<td>
<p>The number of threads to use for generating null signatures. Default is 1</p>
</td></tr>
<tr><td><code id="mra_+3A_verbose">verbose</code></td>
<td>
<p>Boolean, whether to print full messages on progress analysis. Default is FALSE</p>
</td></tr>
<tr><td><code id="mra_+3A_atacseq">atacseq</code></td>
<td>
<p>An optional 3 column matrix derived from an ATAC-Seq analysis, indicating
1) gene symbol, 2) -log10(FDR)*sing(log2FC) of an ATAC-Seq design, 3) distance from TSS.
If provided, the output will contain an _atacseq_ field.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list summarizing the master regulator analysis
</p>

<ul>
<li><p> nes: the normalized enrichment score: positive if the centroid/TF network is upregulated
in expmat1 vs expmat2 (or in expmat1 vs the mean of the dataset), negative if downregulated. A
vector in multisample mode, a matrix in sample-by-sample mode.
</p>
</li>
<li><p> pvalue: the pvalue of the enrichment.
</p>
</li>
<li><p> sig: the calculated signature (useful for plotting).
</p>
</li>
<li><p> regulon: the original regulon used in the analysis (but filtered for _minsize_)
</p>
</li>
<li><p> atac: Optionally present if atacseq data is provided. For each centroid/TF a number
ranging from 0 to 1 will indicate the fraction of changes in activity due to promoter effects
rather than distal effects.
</p>
</li></ul>


<hr>
<h2 id='mraplot'>Plot a master regulator analysis</h2><span id='topic+mraplot'></span>

<h3>Description</h3>

<p>Plotting function for master regulator analysis performed by the _mra_ function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mraplot(
  mraobj,
  mrs = 5,
  title = "corto - Master Regulator Analysis",
  pthr = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mraplot_+3A_mraobj">mraobj</code></td>
<td>
<p>The input object, output of the function mra</p>
</td></tr>
<tr><td><code id="mraplot_+3A_mrs">mrs</code></td>
<td>
<p>Either a numeric value indicating how many MRs to show, sorted by
significance, or a character vector specifying which TFs to show. Default is 5</p>
</td></tr>
<tr><td><code id="mraplot_+3A_title">title</code></td>
<td>
<p>Title of the plot (optional, default is &quot;corto - Master Regulator Analysis&quot;)</p>
</td></tr>
<tr><td><code id="mraplot_+3A_pthr">pthr</code></td>
<td>
<p>The p-value at which the MR is considered significant. Default is 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot is generated
</p>

<hr>
<h2 id='p2r'>p2r Convert a P-value to the corresponding Correlation Coefficient</h2><span id='topic+p2r'></span>

<h3>Description</h3>

<p>p2r Convert a P-value to the corresponding Correlation Coefficient
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p2r(p, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p2r_+3A_p">p</code></td>
<td>
<p>the p-value</p>
</td></tr>
<tr><td><code id="p2r_+3A_n">n</code></td>
<td>
<p>the number of samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a correlation coefficient
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p2r(p=0.08,n=20)
</code></pre>

<hr>
<h2 id='p2z'>p2z</h2><span id='topic+p2z'></span>

<h3>Description</h3>

<p>This function gives a gaussian Z-score corresponding to the provided p-value
Careful: sign is not provided
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p2z(p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p2z_+3A_p">p</code></td>
<td>
<p>a p-value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>z a Z score
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p&lt;-0.05
p2z(p)
</code></pre>

<hr>
<h2 id='plot_gsea'>Plot GSEA results</h2><span id='topic+plot_gsea'></span>

<h3>Description</h3>

<p>This function generates a GSEA plot from a gsea object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gsea(
  gsea.obj,
  twoColors = c("red", "blue"),
  plotNames = FALSE,
  colBarcode = "black",
  title = "Running Enrichment Score",
  bottomTitle = "List Values",
  bottomYlabel = "Signature values",
  ext_nes = NULL,
  ext_pvalue = NULL,
  ext_es = NULL,
  omit_middle = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gsea_+3A_gsea.obj">gsea.obj</code></td>
<td>
<p>GSEA object produced by the <code>gsea</code> function</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_twocolors">twoColors</code></td>
<td>
<p>the two colors to use for positive[1] and negative[2]
enrichment scores</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_plotnames">plotNames</code></td>
<td>
<p>Logical. Should the set names be plotted?</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_colbarcode">colBarcode</code></td>
<td>
<p>The color of the barcode</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_title">title</code></td>
<td>
<p>String to be plotted above the Running Enrichment Score</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_bottomtitle">bottomTitle</code></td>
<td>
<p>String for the title of the bottom part of the plot</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_bottomylabel">bottomYlabel</code></td>
<td>
<p>String for the Y label of the bottom plot</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_ext_nes">ext_nes</code></td>
<td>
<p>Provide a NES from an external calculation</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_ext_pvalue">ext_pvalue</code></td>
<td>
<p>Provide a pvalue from an external calculation</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_ext_es">ext_es</code></td>
<td>
<p>Provide an ES from an external calculation</p>
</td></tr>
<tr><td><code id="plot_gsea_+3A_omit_middle">omit_middle</code></td>
<td>
<p>If TRUE, will not plot the running score
(FALSE by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing, a plot is generated in the default output device
</p>


<h3>Examples</h3>

<pre><code class='language-R'>reflist&lt;-setNames(-sort(rnorm(1000)),paste0('gene',1:1000))
set&lt;-paste0('gene',sample(1:200,50))
obj&lt;-gsea(reflist,set,method='pareto',np=1000)
plot_gsea(obj)
</code></pre>

<hr>
<h2 id='plot_gsea2'>Plot 2-way GSEA results</h2><span id='topic+plot_gsea2'></span>

<h3>Description</h3>

<p>This function generates a GSEA plot from a gsea object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gsea2(
  gsea.obj,
  twoColors = c("red", "blue"),
  plotNames = FALSE,
  title = "Running Enrichment Score",
  bottomTitle = "List Values",
  bottomYlabel = "Signature values",
  legside1 = NULL,
  legside2 = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_gsea2_+3A_gsea.obj">gsea.obj</code></td>
<td>
<p>GSEA object produced by the <code>gsea</code> function</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_twocolors">twoColors</code></td>
<td>
<p>the two colors to use for positive[1] and negative[2]
enrichment scores, and of the barcodes</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_plotnames">plotNames</code></td>
<td>
<p>Logical. Should the set names be plotted?</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_title">title</code></td>
<td>
<p>String to be plotted above the Running Enrichment Score</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_bottomtitle">bottomTitle</code></td>
<td>
<p>String for the title of the bottom part of the plot</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_bottomylabel">bottomYlabel</code></td>
<td>
<p>String for the Y label of the bottom plot
(FALSE by default)</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_legside1">legside1</code></td>
<td>
<p>String specifying the position of the first NES legend,
for example &quot;topright&quot;, &quot;bottomleft&quot;. Default is NULL, letting the function
automatically place it</p>
</td></tr>
<tr><td><code id="plot_gsea2_+3A_legside2">legside2</code></td>
<td>
<p>String specifying the position of the second NES legend,
for example &quot;topright&quot;, &quot;bottomleft&quot;. Default is NULL, letting the function
automatically place it</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing, a plot is generated in the default output device
</p>


<h3>Examples</h3>

<pre><code class='language-R'>reflist&lt;-setNames(-sort(rnorm(1000)),paste0('gene',1:1000))
set1&lt;-paste0('gene',sample(1:200,50))
set2&lt;-paste0('gene',sample(801:1000,50))
obj&lt;-gsea2(reflist,set1,set2,method='pareto',np=1000)
plot_gsea2(obj)
</code></pre>

<hr>
<h2 id='r2p'>r2p Convert Correlation Coefficient to P-value</h2><span id='topic+r2p'></span>

<h3>Description</h3>

<p>r2p Convert Correlation Coefficient to P-value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r2p(r, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="r2p_+3A_r">r</code></td>
<td>
<p>the correlation coefficient</p>
</td></tr>
<tr><td><code id="r2p_+3A_n">n</code></td>
<td>
<p>the number of samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric p-value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>r2p(r=0.4,n=20) # 0.08
</code></pre>

<hr>
<h2 id='scatter'>scatter - XY scatter plot with extra information</h2><span id='topic+scatter'></span>

<h3>Description</h3>

<p>This function will plot two variables (based on their common names), calculate their
Coefficient of Correlation (CC), plot a linear regression line and color the background
if the correlation is positive (red), negative
(blue) or non-significant (white)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatter(
  x,
  y,
  method = "pearson",
  threshold = 0.01,
  showLine = TRUE,
  grid = TRUE,
  bgcol = FALSE,
  pch = 20,
  subtitle = NULL,
  extendXlim = FALSE,
  ci = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scatter_+3A_x">x</code></td>
<td>
<p>The first named vector</p>
</td></tr>
<tr><td><code id="scatter_+3A_y">y</code></td>
<td>
<p>The second named vector</p>
</td></tr>
<tr><td><code id="scatter_+3A_method">method</code></td>
<td>
<p>a character string indicating which correlation coefficient is to
be computed. One of &quot;pearson&quot; (default), &quot;kendall&quot;, or &quot;spearman&quot;: can be abbreviated.</p>
</td></tr>
<tr><td><code id="scatter_+3A_threshold">threshold</code></td>
<td>
<p>a numeric value indicating the significance threshold (p-value) of the correlation,
in order to show a colored background. Default is 0.01.</p>
</td></tr>
<tr><td><code id="scatter_+3A_showline">showLine</code></td>
<td>
<p>a boolean indicating if a linear regression line should be plotted. Default is
TRUE</p>
</td></tr>
<tr><td><code id="scatter_+3A_grid">grid</code></td>
<td>
<p>a boolean indicating whether to show a plot grid. Default is TRUE</p>
</td></tr>
<tr><td><code id="scatter_+3A_bgcol">bgcol</code></td>
<td>
<p>Boolean. Should a background coloring associated to significance and sign of
correlation be used? Default is TRUE, and it will color the background in red if the correlation
coefficient is positive, in blue if negative, in white if not significant (accordin to the
_threshold_ parameter)</p>
</td></tr>
<tr><td><code id="scatter_+3A_pch">pch</code></td>
<td>
<p>the _pch_ parameter indicating the points shape. Default is 20</p>
</td></tr>
<tr><td><code id="scatter_+3A_subtitle">subtitle</code></td>
<td>
<p>NULL by default, in which case the function will print as a subtitle the correlation
coefficient (CC) and its pvalue. Otherwise, a user-provided string, bypassing the predefined subtitle</p>
</td></tr>
<tr><td><code id="scatter_+3A_extendxlim">extendXlim</code></td>
<td>
<p>logical. If TRUE, the x-axis limits are extended by a fraction (useful for
labeling points on the margins of the plot area). Default is FALSE</p>
</td></tr>
<tr><td><code id="scatter_+3A_ci">ci</code></td>
<td>
<p>logical. If TRUE, confidence intervals of linear regression are
shown at 95 percent confidence.</p>
</td></tr>
<tr><td><code id="scatter_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to the core _plot_ function (if a new plot is created)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x&lt;-setNames(rnorm(200),paste0("var",1:200))
y&lt;-setNames(rnorm(210),paste0("var",11:220))
scatter(x,y,xlab="Variable x",ylab="Variable y",main="Scatter plot by corto package",ci=TRUE)
</code></pre>

<hr>
<h2 id='scinot'>scinot - Convert a number to a scientific notation expression</h2><span id='topic+scinot'></span>

<h3>Description</h3>

<p>This function will convert any numeric vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scinot(v, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scinot_+3A_v">v</code></td>
<td>
<p>The input numeric object. It can be a single value or a vector</p>
</td></tr>
<tr><td><code id="scinot_+3A_digits">digits</code></td>
<td>
<p>An integer indicating how many significant digits to show. Default is 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class _expression_.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Usage on single value
scinot(0.00000543)
# Demonstration on a vector
numbers&lt;-c(3.456e-12,0.00901,5670000,-3.16e18,0.000004522,rnorm(5,sd=0.0000001))
plot(0,xlim=c(0,10),ylim=c(0,10),type="n")
text(c(2,6),c(10,10),labels=c("Before","After"),font=2)
for(i in 10:1){
    text(c(2,6),c(i-1,i-1),labels=c(numbers[i],scinot(numbers)[i]))
}
</code></pre>

<hr>
<h2 id='slice'>Slice</h2><span id='topic+slice'></span>

<h3>Description</h3>

<p>This function prints a slice of a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slice(matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slice_+3A_matrix">matrix</code></td>
<td>
<p>A matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A visualization of the first 5 rows and columns of the input matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
example&lt;-matrix(rnorm(1000),nrow=100,ncol=10)
slice(example)
</code></pre>

<hr>
<h2 id='ssgsea'>ssGSEA</h2><span id='topic+ssgsea'></span>

<h3>Description</h3>

<p>This function performs single sample GSEA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssgsea(inmat, groups, scale = TRUE, minsize = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ssgsea_+3A_inmat">inmat</code></td>
<td>
<p>A numeric matrix, with rownames/rows as genes or features, and
colnames/columns as sample names</p>
</td></tr>
<tr><td><code id="ssgsea_+3A_groups">groups</code></td>
<td>
<p>a named list. Names are names of the groups (e.g. pathways) and
elements are character vectors indicating gene or feature names (that should
match, at least partially, with the rownames of inmat)</p>
</td></tr>
<tr><td><code id="ssgsea_+3A_scale">scale</code></td>
<td>
<p>Boolean. Wheter the matrix should be row-scaled.</p>
</td></tr>
<tr><td><code id="ssgsea_+3A_minsize">minsize</code></td>
<td>
<p>Numeric. Include only groups with at least this many elements
Default is 10</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of Normalized Enrichment Scores (NES), which can be converted to
p-values using the function _corto::z2p_
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A random matrix
set.seed(1)
inmat&lt;-matrix(rnorm(200*50),nrow=200,ncol=50)
rownames(inmat)&lt;-paste0("gene",1:nrow(inmat))
# A random list of groups
groups&lt;-list()
for(i in 1:10){
    somegenes&lt;-sample(rownames(inmat),30)
    groups[[paste0("pathway_",i)]]&lt;-somegenes
}
# Run ssGSEA
nesmat&lt;-ssgsea(inmat,groups)

</code></pre>

<hr>
<h2 id='stouffer'>Stouffer integration of Z scores</h2><span id='topic+stouffer'></span>

<h3>Description</h3>

<p>This function gives a gaussian Z-score corresponding to the provided p-value
Careful: sign is not provided
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stouffer(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stouffer_+3A_x">x</code></td>
<td>
<p>a vector of Z scores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Z an integrated Z score
</p>


<h3>Examples</h3>

<pre><code class='language-R'>zs&lt;-c(1,3,5,2,3)
stouffer(zs)
</code></pre>

<hr>
<h2 id='textrepel'>textrepel - Plot text with non-overlapping labels</h2><span id='topic+textrepel'></span>

<h3>Description</h3>

<p>This function plots text with x and y coordinates, forcing overlapping labels to not overlap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>textrepel(
  x,
  y,
  labels = NULL,
  padding = " ",
  rstep = 0.1,
  tstep = 0.1,
  vertical = FALSE,
  textSize = 1,
  showLines = TRUE,
  lineColor = "#00000066",
  lineWidth = 2,
  showPoints = TRUE,
  pointColor = "#00000033",
  pointSize = 2,
  pointPch = 16,
  add = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="textrepel_+3A_x">x</code></td>
<td>
<p>A numeric vector of x coordinates</p>
</td></tr>
<tr><td><code id="textrepel_+3A_y">y</code></td>
<td>
<p>A numeric vector of y coordinates (must have the same length of x)</p>
</td></tr>
<tr><td><code id="textrepel_+3A_labels">labels</code></td>
<td>
<p>A vector of labels associated with x and y (must have the same length of x)</p>
</td></tr>
<tr><td><code id="textrepel_+3A_padding">padding</code></td>
<td>
<p>A character object specifying left and right padding for words. Default is a
single whitespace &quot; &quot;</p>
</td></tr>
<tr><td><code id="textrepel_+3A_rstep">rstep</code></td>
<td>
<p>Decimal numeric specifying the lateral step length for label distancing.
Default is 0.1</p>
</td></tr>
<tr><td><code id="textrepel_+3A_tstep">tstep</code></td>
<td>
<p>Decimal numeric specifying the theta step length for label distancing.
Default is 0.1</p>
</td></tr>
<tr><td><code id="textrepel_+3A_vertical">vertical</code></td>
<td>
<p>Boolean. If FALSE (default), the labels are plotted horizontally. If TRUE,
vertically</p>
</td></tr>
<tr><td><code id="textrepel_+3A_textsize">textSize</code></td>
<td>
<p>Numeric. Size of text. Default is 1</p>
</td></tr>
<tr><td><code id="textrepel_+3A_showlines">showLines</code></td>
<td>
<p>Boolean. Whether to show lines connecting displaced labels to their original
plot. Default is TRUE</p>
</td></tr>
<tr><td><code id="textrepel_+3A_linecolor">lineColor</code></td>
<td>
<p>String indicating the color of the connecting line</p>
</td></tr>
<tr><td><code id="textrepel_+3A_linewidth">lineWidth</code></td>
<td>
<p>Numeric indicating the width of the connecting line</p>
</td></tr>
<tr><td><code id="textrepel_+3A_showpoints">showPoints</code></td>
<td>
<p>Boolean. Whether to show points over original x-y coordinates</p>
</td></tr>
<tr><td><code id="textrepel_+3A_pointcolor">pointColor</code></td>
<td>
<p>String indicating the color of the point</p>
</td></tr>
<tr><td><code id="textrepel_+3A_pointsize">pointSize</code></td>
<td>
<p>Numeric indicating the size of the point</p>
</td></tr>
<tr><td><code id="textrepel_+3A_pointpch">pointPch</code></td>
<td>
<p>Integer applying to shape of points. Default is 16 (filled circle)</p>
</td></tr>
<tr><td><code id="textrepel_+3A_add">add</code></td>
<td>
<p>Boolean. If FALSE (default), a new plot is generated. If TRUE, the textrepel labels
are plotted over the existing plot</p>
</td></tr>
<tr><td><code id="textrepel_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to the core _plot_ function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simple example, generating a new plot, taking care of some overlapping labels
set.seed(1)
x&lt;-rnorm(100)
y&lt;-abs(x)+rnorm(100)
names(x)&lt;-names(y)&lt;-paste0("OBJ",1:length(x))
labels&lt;-names(x)
textrepel(x,y,labels)
# More advanced example, adding textrepel over an existing plot
set.seed(1)
x&lt;-rnorm(1000)
y&lt;-abs(x)+rnorm(1000)
names(x)&lt;-names(y)&lt;-paste0("GENE",1:length(x))
labels&lt;-names(x)
plot(x,y,pch=16,col="#00000066",xlim=1.3*c(min(x),max(x)))
subset1&lt;-which(x&lt;(-2.2))
textrepel(x[subset1],y[subset1],labels[subset1],add=TRUE,pointCol="cornflowerblue")
subset2&lt;-which(x&gt;(+2.2))
textrepel(x[subset2],y[subset2],labels[subset2],add=TRUE,pointCol="salmon")
</code></pre>

<hr>
<h2 id='val2col'>val2col - Convert a numeric vector into colors</h2><span id='topic+val2col'></span>

<h3>Description</h3>

<p>val2col - Convert a numeric vector into colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>val2col(
  z,
  col1 = "navy",
  col2 = "white",
  col3 = "red3",
  nbreaks = 1000,
  center = TRUE,
  rank = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="val2col_+3A_z">z</code></td>
<td>
<p>a vector of numbers</p>
</td></tr>
<tr><td><code id="val2col_+3A_col1">col1</code></td>
<td>
<p>a color name for the min value, default 'navy'</p>
</td></tr>
<tr><td><code id="val2col_+3A_col2">col2</code></td>
<td>
<p>a color name for the middle value, default 'white'</p>
</td></tr>
<tr><td><code id="val2col_+3A_col3">col3</code></td>
<td>
<p>a color name for the max value, default 'red3'</p>
</td></tr>
<tr><td><code id="val2col_+3A_nbreaks">nbreaks</code></td>
<td>
<p>Number of colors to be generated. Default is 30.</p>
</td></tr>
<tr><td><code id="val2col_+3A_center">center</code></td>
<td>
<p>boolean, should the data be centered? Default is TRUE</p>
</td></tr>
<tr><td><code id="val2col_+3A_rank">rank</code></td>
<td>
<p>boolean, should the data be ranked? Default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a&lt;-rnorm(1000)
cols&lt;-val2col(a)
plot(a,col=cols,pch=16)
</code></pre>

<hr>
<h2 id='wstouffer'>Weighted Stouffer integration of Z scores</h2><span id='topic+wstouffer'></span>

<h3>Description</h3>

<p>This function gives a gaussian Z-score corresponding to the provided p-value
Careful: sign is not provided
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wstouffer(x, w)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wstouffer_+3A_x">x</code></td>
<td>
<p>a vector of Z scores</p>
</td></tr>
<tr><td><code id="wstouffer_+3A_w">w</code></td>
<td>
<p>weight for each Z score</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Z an integrated Z score
</p>


<h3>Examples</h3>

<pre><code class='language-R'>zs&lt;-c(1,-3,5,2,3)
ws&lt;-c(1,10,1,2,1)
wstouffer(zs,ws)
</code></pre>

<hr>
<h2 id='z2p'>z2p</h2><span id='topic+z2p'></span>

<h3>Description</h3>

<p>This function gives a gaussian p-value corresponding to the provided Z-score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z2p(z)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="z2p_+3A_z">z</code></td>
<td>
<p>a Z score</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a p-value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>z&lt;-1.96
z2p(z)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
