<!DOCTYPE html><html lang="en-US"><head><title>Help for package cosmicsig</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cosmicsig}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cosmicsig'><p>cosmicsig: A package of mutational signatures from COSMIC (Catalogue</p>
Of Somatic Mutations In Cancer)
<a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.</a></li>
<li><a href='#COSMIC_v3.0'><p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In</p>
Cancer (v3.0 - May 2019)</a></li>
<li><a href='#COSMIC_v3.1'><p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In</p>
Cancer (v3.1 - June 2020)</a></li>
<li><a href='#COSMIC_v3.2'><p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In</p>
Cancer (v3.2 - March 2021)</a></li>
<li><a href='#COSMIC_v3.3'><p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In</p>
Cancer (v3.3 - June 2022)</a></li>
<li><a href='#etiology'><p>List of mutational signatures's proposed etiology summarized from COSMIC,</p>
Catalogue Of Somatic Mutations In Cancer (v3.3 - June 2022)</a></li>
<li><a href='#get_etiology'><p>Get the proposed etiology of mutational signatures.</p></a></li>
<li><a href='#possible_artifacts'><p>Return a character vector of the names of possible SBS96 signature artifacts</p></a></li>
<li><a href='#rare_signatures'><p>Return a character vector of the names of rare SBS96 signatures</p></a></li>
<li><a href='#SBS96_ID_to_SBS192_ID'><p>Translate SBS96 signature IDs to SBS192 signature IDs by adding &quot;-E&quot; if necessary</p></a></li>
<li><a href='#signature'><p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In</p>
Cancer (v3.3 - June 2022)</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Mutational Signatures from COSMIC (Catalogue of Somatic
Mutations in Cancer)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A data package with 2 main package variables: 'signature' and 'etiology'. 
     The 'signature' variable contains the latest mutational signature profiles 
     released on COSMIC <a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a> for 3 mutation
     types:
     * Single base substitutions in the context of preceding and following bases,
     * Doublet base substitutions, and
     * Small insertions and deletions.
    The 'etiology' variable provides the known or hypothesized causes of signatures. 
    'cosmicsig' stands for COSMIC signatures. Please run ?'cosmicsig' for more
    information.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Rozen-Lab/cosmicsig">https://github.com/Rozen-Lab/cosmicsig</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Rozen-Lab/cosmicsig/issues">https://github.com/Rozen-Lab/cosmicsig/issues</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), ICAMS, usethis</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-30 06:49:49 UTC; User</td>
</tr>
<tr>
<td>Author:</td>
<td>Steven Rozen <a href="https://orcid.org/0000-0002-4288-0056"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Nanhai Jiang <a href="https://orcid.org/0000-0003-4974-2753"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Steven Rozen &lt;steverozen@pm.me&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-30 08:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cosmicsig'>cosmicsig: A package of mutational signatures from COSMIC (Catalogue
Of Somatic Mutations In Cancer)
<a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.</h2><span id='topic+cosmicsig'></span><span id='topic+cosmicsig-package'></span>

<h3>Description</h3>

<p>For a general introduction to mutational signatures
and the techniques used to discover them, see
Alexandrov et al., 2020 <a href="https://doi.org/10.1038/s41586-020-1943-3">doi:10.1038/s41586-020-1943-3</a>.
</p>


<h3>Details</h3>

<p>This is a data package with 2 main package variables:
<code><a href="#topic+signature">signature</a></code> and <code><a href="#topic+etiology">etiology</a></code>.
</p>
<p>The <code><a href="#topic+signature">signature</a></code> variable contains the
latest mutational signature profiles released on
<a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a> for
3 mutation types:
</p>

<ul>
<li><p> SBS (single base substitutions in the context
of preceding and following bases, called SBS96
in this package)
</p>
</li>
<li><p> DBS (doublet base substitutions, called DBS78 in
this package)
</p>
</li>
<li><p> ID (small insertions and deletions)
</p>
</li></ul>

<p>The package variable <code><a href="#topic+etiology">etiology</a></code> contains
information on known or hypothesized causes of
mutational signatures. In general, it is better to use
<code><a href="#topic+get_etiology">get_etiology</a></code>.
</p>
<p>Earlier releases are available in the
variables COSMIC_<em>version</em>, e.g.
<code><a href="#topic+COSMIC_v3.2">COSMIC_v3.2</a></code>.
</p>
<p>The profiles of SBSs signatures depend on the
frequencies of trinucleotides in a genome
and profiles of DBS signatures depend on the
frequencies of dinucleotides in a genome.
Therefore COSMIC and this package provide
slightly different signatures for different
reference genomes. COSMIC and this package offer
versions of SBS and DBS signatures for
human GRCh37 (also known as hg19) and GRCh38,
and for mouse and rat.
ID signatures do not take into consideration
differing nucleotide composition between
reference genomes because relating this
to the ID mutational categories would be
extremely complicated.
</p>
<p>Some signatures are due to experimental
or laboratory artifacts. Function
<code><a href="#topic+possible_artifacts">possible_artifacts</a></code>
returns these.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>

<hr>
<h2 id='COSMIC_v3.0'>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.0 - May 2019)</h2><span id='topic+COSMIC_v3.0'></span>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.0 - May 2019)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COSMIC_v3.0
</code></pre>


<h3>Format</h3>

<p>A list with one element <code>signature</code>, with the
same structure as <code><a href="#topic+signature">signature</a></code>, except that
sub-element GRCh37 does contain <code>SBS192</code>.
</p>


<h3>Remark</h3>

<p>The signatures are all genome signatures. <br /> See <a href="ICAMS.html#topic+CatalogRowOrder">CatalogRowOrder</a>
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_sig_v3.0 &lt;- COSMIC_v3.0$signature$GRCh37$SBS96

</code></pre>

<hr>
<h2 id='COSMIC_v3.1'>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.1 - June 2020)</h2><span id='topic+COSMIC_v3.1'></span>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.1 - June 2020)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COSMIC_v3.1
</code></pre>


<h3>Format</h3>

<p>A list with one element <code>signature</code>, with the
same structure as <code><a href="#topic+signature">signature</a></code>.
</p>


<h3>Remark</h3>

<p>The signatures are all genome signatures. <br /> See <a href="ICAMS.html#topic+CatalogRowOrder">CatalogRowOrder</a>
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_sig_v3.1 &lt;- COSMIC_v3.1$signature$GRCh37$SBS96

</code></pre>

<hr>
<h2 id='COSMIC_v3.2'>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.2 - March 2021)</h2><span id='topic+COSMIC_v3.2'></span>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.2 - March 2021)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COSMIC_v3.2
</code></pre>


<h3>Format</h3>

<p>A list with two elements, <code>signature</code> and <code>etiology</code>.
</p>

<ul>
<li> <p><code>signature</code> is a list with the same structure as
<code><a href="#topic+signature">signature</a></code>.
</p>
</li>
<li> <p><code>etiology</code> is a list with the same structure as
<code><a href="#topic+etiology">etiology</a></code>.
</p>
</li></ul>



<h3>Note</h3>

<p>SBS10c, SBS10d, SBS91, SBS92, SBS93, SBS94 (total 6) new SBS signatures were
added in COSMIC v3.2. See the news from COSMIC release for more details
<a href="https://cosmic-blog.sanger.ac.uk/cosmic-mutational-signatures-release-v3-2/">https://cosmic-blog.sanger.ac.uk/cosmic-mutational-signatures-release-v3-2/</a>
</p>


<h3>Remark</h3>

<p>The signatures are all genome signatures. <br /> See <a href="ICAMS.html#topic+CatalogRowOrder">CatalogRowOrder</a>
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_sig_v3.2 &lt;- COSMIC_v3.2$signature$GRCh37$SBS96

</code></pre>

<hr>
<h2 id='COSMIC_v3.3'>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)</h2><span id='topic+COSMIC_v3.3'></span>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>COSMIC_v3.3
</code></pre>


<h3>Format</h3>

<p>A list with two elements, <code>signature</code> and <code>etiology</code>.
</p>

<ul>
<li> <p><code>signature</code> is a list with the same structure as
<code><a href="#topic+signature">signature</a></code>.
</p>
</li>
<li> <p><code>etiology</code> is a list with the same structure as
<code><a href="#topic+etiology">etiology</a></code>.
</p>
</li></ul>



<h3>Remark</h3>

<p>The signatures are all genome signatures. <br /> See <a href="ICAMS.html#topic+CatalogRowOrder">CatalogRowOrder</a>
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_sig_v3.3 &lt;- COSMIC_v3.3$signature$GRCh37$SBS96

</code></pre>

<hr>
<h2 id='etiology'>List of mutational signatures's proposed etiology summarized from COSMIC,
Catalogue Of Somatic Mutations In Cancer (v3.3 - June 2022)</h2><span id='topic+etiology'></span>

<h3>Description</h3>

<p>List of mutational signatures's proposed etiology summarized from COSMIC,
Catalogue Of Somatic Mutations In Cancer (v3.3 - June 2022)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>etiology
</code></pre>


<h3>Format</h3>

<p>A list with the elements:
</p>

<ul>
<li><p> SBS96
</p>
</li>
<li><p> DBS78
</p>
</li>
<li><p> ID
</p>
</li></ul>

<p>Each element is a single-column matrix with rownames being the signature
IDs and values being a short string describing the proposed
etiology.
</p>
<p>In general use <code><a href="#topic+get_etiology">get_etiology</a></code>, which handles new
signatures do not have an element in <code>etiology</code>.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_etiology &lt;- etiology$SBS96
</code></pre>

<hr>
<h2 id='get_etiology'>Get the proposed etiology of mutational signatures.</h2><span id='topic+get_etiology'></span>

<h3>Description</h3>

<p>Return the known or hypothesized causes of mutational signatures.
The level of evidence supporting the proposed etiologies varies.
In addition, some proposed etiologies are more akin to associations than
specific, mechanistic causes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_etiology(mutation_type, sig_id)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_etiology_+3A_mutation_type">mutation_type</code></td>
<td>
<p>Character string, one of &quot;SBS96&quot;, &quot;SBS192&quot;, &quot;DBS78&quot;,
&quot;ID&quot;.</p>
</td></tr>
<tr><td><code id="get_etiology_+3A_sig_id">sig_id</code></td>
<td>
<p>Character vector with signature ids, e.g. <code>c("SBS3",
"SBS5")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of the same length as <code>sig_id</code>,
each element of which is the etiology of the
corresponding signature, if available, or else the empty string.
</p>


<h3>Note</h3>

<p>The etiology information is not versioned at the COSMIC website.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_etiology">get_etiology</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_etiology(mutation_type = "ID", sig_id = c("ID1", "foo", "ID3"))
</code></pre>

<hr>
<h2 id='possible_artifacts'>Return a character vector of the names of possible SBS96 signature artifacts</h2><span id='topic+possible_artifacts'></span>

<h3>Description</h3>

<p>Return a character vector of the names of possible SBS96 signature artifacts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possible_artifacts()
</code></pre>


<h3>Value</h3>

<p>A character vector of the names of possible SBS96 signature
artifacts.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>artifact_sigs &lt;- possible_artifacts()
</code></pre>

<hr>
<h2 id='rare_signatures'>Return a character vector of the names of rare SBS96 signatures</h2><span id='topic+rare_signatures'></span>

<h3>Description</h3>

<p>Return a character vector of the names of rare SBS96 signatures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_signatures()
</code></pre>


<h3>Value</h3>

<p>A character vector of the names of rare SBS96 signatures.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rare_sigs &lt;- rare_signatures()
</code></pre>

<hr>
<h2 id='SBS96_ID_to_SBS192_ID'>Translate SBS96 signature IDs to SBS192 signature IDs by adding &quot;-E&quot; if necessary</h2><span id='topic+SBS96_ID_to_SBS192_ID'></span>

<h3>Description</h3>

<p>&quot;-E&quot; added to the name of a transcriptional strand bias
signature indicates that it was extracted only from exome sequencing data,
and thus reflects transcriptional strand bias in the exome rather than in
the entire transcript, including introns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SBS96_ID_to_SBS192_ID(sig_ids)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SBS96_ID_to_SBS192_ID_+3A_sig_ids">sig_ids</code></td>
<td>
<p>Character vector of SBS96 signature IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of corresponding SBS192 signature IDs; some
have &quot;-E&quot; (for exome) post-pended.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_ids &lt;- c("SBS1", "SBS23", "SBS25")
sbs192_ids &lt;- SBS96_ID_to_SBS192_ID(sbs96_ids)
</code></pre>

<hr>
<h2 id='signature'>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)</h2><span id='topic+signature'></span>

<h3>Description</h3>

<p>Mutational signatures data from COSMIC, Catalogue Of Somatic Mutations In
Cancer (v3.3 - June 2022)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>signature
</code></pre>


<h3>Format</h3>

<p>A list with the following elements:
</p>

<ul>
<li><p> GRCh37: Homo sapiens (human) genome assembly GRCh37.
</p>
</li>
<li><p> GRCh38: Homo sapiens (human) genome assembly GRCh38.
</p>
</li>
<li><p> mm9: Mus musculus (house mouse) genome assembly mm9.
</p>
</li>
<li><p> mm10: Mus musculus (house mouse) genome assembly mm10.
</p>
</li>
<li><p> rn6: Rattus norvegicus (Norway rat) genome assembly rn6.
</p>
</li></ul>

<p>Each element contains the sub elements:
</p>

<ul>
<li><p> SBS96: Strand-agnostic single-base substitutions in trinucleotide context.
</p>
</li>
<li><p> DBS78: Strand-agnostic doublet-base substitutions.
</p>
</li></ul>

<p>Element GRCh37 contains the additional sub elements:
</p>

<ul>
<li><p> ID: Strand-agnostic indels (short insertions and deletions).
</p>
</li></ul>



<h3>Remark</h3>

<p>The signatures are all genome signatures. <br /> See <a href="ICAMS.html#topic+CatalogRowOrder">CatalogRowOrder</a>
in package ICAMS for the classification of mutation types.
</p>


<h3>Source</h3>

<p><a href="https://cancer.sanger.ac.uk/signatures/">https://cancer.sanger.ac.uk/signatures/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sbs96_sig &lt;- signature$GRCh37$SBS96

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
