<!DOCTYPE html><html><head><title>Help for package countTransformers</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {countTransformers}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#es'>
<p>A Simulated Data Set</p></a></li>
<li><a href='#getJaccard'>
<p>Calculate Jaccard Index for Two Binary Vectors</p></a></li>
<li><a href='#l2Transformer'>
<p>Log Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference</p></a></li>
<li><a href='#lTransformer'>
<p>Log-based transformation</p></a></li>
<li><a href='#lv2Transformer'>
<p>Log and VOOM Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference</p></a></li>
<li><a href='#lvTransformer'>
<p>Log and VOOM Transformation</p></a></li>
<li><a href='#r2Transformer'>
<p>Root Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference</p></a></li>
<li><a href='#rTransformer'>
<p>Root Based Transformation</p></a></li>
<li><a href='#rv2Transformer'>
<p>Root and VOOM Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference</p></a></li>
<li><a href='#rvTransformer'>
<p>Root and VOOM Transformation</p></a></li>
<li><a href='#wilcoxWrapper'>
<p>Wrapper Function for Wilcoxon Rank Sum Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transform Counts in RNA-Seq Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-03-20</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zeyu Zhang &lt;zhzyvv@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), Biobase, limma</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, graphics, stats</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bioinformatics, DifferentialExpression</td>
</tr>
<tr>
<td>Description:</td>
<td>Provide data transformation functions to transform counts in RNA-seq data analysis. Please see the reference: Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. (2019) &lt;doi.org/10.1038/s41598-019-41315-w&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-03-20 11:24:28 UTC; ICSA</td>
</tr>
<tr>
<td>Author:</td>
<td>Zeyu Zhang [aut, cre],
  Danyang Yu [aut, ctb],
  Minseok Seo [aut, ctb],
  Craig P. Hersh [aut, ctb],
  Scott T. Weiss [aut, ctb],
  Weiliang Qiu [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-03-20 12:56:43 UTC</td>
</tr>
</table>
<hr>
<h2 id='es'>
A Simulated Data Set
</h2><span id='topic+es'></span>

<h3>Description</h3>

<p>A simulated data set based on the R code provided by Law et al.'s (2014) paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("es")</code></pre>


<h3>Format</h3>

<p>The format is:
Formal class 'ExpressionSet' [package &quot;Biobase&quot;]
</p>


<h3>Details</h3>

<p>The simulated data set contains RNA-seq counts of 1000 genes for 6 samples (3 cases and 3 controls). 
The library sizes of the 6 samples are not equal.
</p>


<h3>Source</h3>

<p>The dataset was generated based on the R code Simulation_Full.R from the website <a href="http://bioinf.wehi.edu.au/voom/">http://bioinf.wehi.edu.au/voom/</a>.
</p>


<h3>References</h3>

<p>Law CW, Chen Y, Shi W, Smyth GK.  
voom: precision weights unlock linear model analysis tools for RNA-seq read counts.
Genome Biology. 2014; 15:R29
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# phenotype data
pDat = pData(es)
print(dim(pDat))
print(pDat[1:2,])

# feature data
fDat = fData(es)
print(dim(fDat))
print(fDat[1:2,])

</code></pre>

<hr>
<h2 id='getJaccard'>
Calculate Jaccard Index for Two Binary Vectors
</h2><span id='topic+getJaccard'></span>

<h3>Description</h3>

<p>Calculate Jaccard index for two binary vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getJaccard(cl1, cl2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getJaccard_+3A_cl1">cl1</code></td>
<td>

<p>n by 1 binary vector of classification 1 for the n subjects
</p>
</td></tr>
<tr><td><code id="getJaccard_+3A_cl2">cl2</code></td>
<td>

<p>n by 1 binary vector of classification 2 for the n subjects
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Jaccard Index is defined as the ratio </p>
<p style="text-align: center;"><code class="reqn">d/(b+c+d</code>
</p>
<p>, where
<code class="reqn">d</code> is the number of subjects who were classified to group 1 by both classification rules,
<code class="reqn">b</code> is the number of subjects who were classified to group 1 by classification rule 1 and were classified to group 0 by classification rule 2,
<code class="reqn">c</code> is the number of subjects who were classified to group 0 by classification rule 1 and were classified to group 1 by classification rule 2.
</p>


<h3>Value</h3>

<p>The Jaccard Index
</p>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  n = 10
  set.seed(1234567)

  # generate two random binary vector of size n
  cl1 = sample(c(1,0), size = n, prob = c(0.5, 0.5), replace = TRUE)
  cl2 = sample(c(1,0), size = n, prob = c(0.5, 0.5), replace = TRUE)
  cat("\n2x2 contingency table &gt;&gt;\n")
  print(table(cl1, cl2))

  JI = getJaccard(cl1, cl2)
  cat("Jaccard index = ", JI, "\n")


</code></pre>

<hr>
<h2 id='l2Transformer'>
Log Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference
</h2><span id='topic+l2Transformer'></span>

<h3>Description</h3>

<p>Log based count transformation minimizing sum of sample-specific squared difference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l2Transformer(mat, low = 1e-04, upp = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="l2Transformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="l2Transformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="l2Transformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the log transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\log_2\left(x_{gi} + \frac{1}{\delta}\right)</code>
</p>
<p>. The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the sum of the squared 
difference between the sample mean and the sample median across <code class="reqn">n</code> subjects </p>
<p style="text-align: center;"><code class="reqn">\sum_{i=1}^{n}\left(\bar{y}_i - \tilde{y}_i\right)^2</code>
</p>
<p>,
<code class="reqn">\bar{y}_i=\sum_{g=1}^{G}y_{gi}/G</code> and 
<code class="reqn">\tilde{y}_i</code> is the median of <code class="reqn">y_{1i}, \ldots, y_{Gi}</code>, and
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = l2Transformer(mat = ex)

# estimated model parameter
print(res$delta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")


</code></pre>

<hr>
<h2 id='lTransformer'>
Log-based transformation
</h2><span id='topic+lTransformer'></span>

<h3>Description</h3>

<p>Log-based transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lTransformer(mat, low = 1e-04, upp = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lTransformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="lTransformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="lTransformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the log transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\log_2\left(x_{gi} + \frac{1}{\delta}\right)</code>
</p>
<p>. The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the squared 
difference between the sample mean and the sample median of the pooled data <code class="reqn">y_{gi}</code>, <code class="reqn">g=1, \ldots, G</code>, <code class="reqn">i=1, \ldots, n</code>,
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = lTransformer(mat = ex)

# estimated model parameter
print(res$delta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='lv2Transformer'>
Log and VOOM Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference
</h2><span id='topic+lv2Transformer'></span>

<h3>Description</h3>

<p>Log and VOOM based count transformation minimizing sum of sample-specific squared difference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lv2Transformer(mat, lib.size = NULL, low = 0.001, upp = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lv2Transformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="lv2Transformer_+3A_lib.size">lib.size</code></td>
<td>

<p>By default, <code>lib.size</code> is a vector of column sums of <code>mat</code>
</p>
</td></tr>
<tr><td><code id="lv2Transformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="lv2Transformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the log transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\log_2\left(t_{gi} + \frac{1}{\delta}\right)</code>
</p>
<p>, 
where
</p>
<p style="text-align: center;"><code class="reqn">t_{gi}=\frac{\left(x_{gi}+0.5\right)}{X_i+1}\times 10^6</code>
</p>
<p> and
<code class="reqn">X_i=\sum_{g=1}^{G} x_{gi}</code> is the column sum for the <code class="reqn">i</code>-th column
of the matrix <code>mat</code>. 
The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the sum of the squared 
difference between the sample mean and the sample median across <code class="reqn">n</code> subjects </p>
<p style="text-align: center;"><code class="reqn">\sum_{i=1}^{n}\left(\bar{y}_i - \tilde{y}_i\right)^2</code>
</p>
<p>,
<code class="reqn">\bar{y}_i=\sum_{g=1}^{G}y_{gi}/G</code> and 
<code class="reqn">\tilde{y}_i</code> is the median of <code class="reqn">y_{1i}, \ldots, y_{Gi}</code>, and
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = lv2Transformer(mat = ex)

# estimated model parameter
print(res$delta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='lvTransformer'>
Log and VOOM Transformation
</h2><span id='topic+lvTransformer'></span>

<h3>Description</h3>

<p>Log and VOOM Transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lvTransformer(mat, lib.size=NULL, low=0.001, upp=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lvTransformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="lvTransformer_+3A_lib.size">lib.size</code></td>
<td>

<p>By default, <code>lib.size</code> is a vector of column sums of <code>mat</code>
</p>
</td></tr>
<tr><td><code id="lvTransformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="lvTransformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the log transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\log_2\left(t_{gi} + \frac{1}{\delta}\right)</code>
</p>
<p>, where
</p>
<p style="text-align: center;"><code class="reqn">t_{gi}=\frac{\left(x_{gi}+0.5\right)}{X_i+1}\times 10^6</code>
</p>
<p> and
<code class="reqn">X_i=\sum_{g=1}^{G} x_{gi}</code> is the column sum for the <code class="reqn">i</code>-th column
of the matrix <code>mat</code>. The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the squared 
difference between the sample mean and the sample median of the pooled data <code class="reqn">y_{gi}</code>, <code class="reqn">g=1, \ldots, G</code>, <code class="reqn">i=1, \ldots, n</code>,
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = lvTransformer(mat = ex)

# estimated model parameter
print(res$delta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='r2Transformer'>
Root Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference
</h2><span id='topic+r2Transformer'></span>

<h3>Description</h3>

<p>Root based count transformation minimizing sum of sample-specific squared difference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r2Transformer(mat, low = 1e-04, upp = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r2Transformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="r2Transformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="r2Transformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the root and voom transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\frac{x_{gi}^{(1/\eta)}}{(1/\eta)}</code>
</p>
<p>,
The optimal value for the parameter <code class="reqn">\eta</code> is to minimize the sum of the squared 
difference between the sample mean and the sample median across <code class="reqn">n</code> subjects </p>
<p style="text-align: center;"><code class="reqn">\sum_{i=1}^{n}\left(\bar{y}_i - \tilde{y}_i\right)^2</code>
</p>
<p>,
<code class="reqn">\bar{y}_i=\sum_{g=1}^{G}y_{gi}/G</code> and 
<code class="reqn">\tilde{y}_i</code> is the median of <code class="reqn">y_{1i}, \ldots, y_{Gi}</code>, and
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = r2Transformer(mat = ex)

# estimated model parameter
print(res$eta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='rTransformer'>
Root Based Transformation
</h2><span id='topic+rTransformer'></span>

<h3>Description</h3>

<p>Root based transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rTransformer(mat, low = 1e-04, upp = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rTransformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="rTransformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="rTransformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the root transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\frac{x_{gi}^{(1/\eta)}}{(1/\eta)}</code>
</p>
<p>. The optimal value for the parameter <code class="reqn">\eta</code> is to minimize the squared 
difference between the sample mean and the sample median of the pooled data <code class="reqn">y_{gi}</code>, <code class="reqn">g=1, \ldots, G</code>, <code class="reqn">i=1, \ldots, n</code>,
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<table>
<tr><td><code>res.eta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = rTransformer(mat = ex)

# estimated model parameter
print(res$eta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='rv2Transformer'>
Root and VOOM Based Count Transformation Minimizing Sum of Sample-Specific Squared Difference
</h2><span id='topic+rv2Transformer'></span>

<h3>Description</h3>

<p>Root and VOOM based count transformation minimizing sum of sample-specific squared difference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rv2Transformer(mat, low = 1e-04, upp = 1000, lib.size = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rv2Transformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="rv2Transformer_+3A_lib.size">lib.size</code></td>
<td>

<p>By default, <code>lib.size</code> is a vector of column sums of <code>mat</code>
</p>
</td></tr>
<tr><td><code id="rv2Transformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="rv2Transformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the root and voom transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\frac{t_{gi}^{(1/\eta)}}{(1/\eta)}</code>
</p>
<p>,
where
</p>
<p style="text-align: center;"><code class="reqn">t_{gi}=\frac{\left(x_{gi}+0.5\right)}{X_i+1}\times 10^6</code>
</p>
<p> and
<code class="reqn">X_i=\sum_{g=1}^{G} x_{gi}</code> is the column sum for the <code class="reqn">i</code>-th column
of the matrix <code>mat</code>. 
The optimal value for the parameter <code class="reqn">\eta</code> is to minimize the sum of the squared 
difference between the sample mean and the sample median across <code class="reqn">n</code> subjects </p>
<p style="text-align: center;"><code class="reqn">\sum_{i=1}^{n}\left(\bar{y}_i - \tilde{y}_i\right)^2</code>
</p>
<p>,
<code class="reqn">\bar{y}_i=\sum_{g=1}^{G}y_{gi}/G</code> and 
<code class="reqn">\tilde{y}_i</code> is the median of <code class="reqn">y_{1i}, \ldots, y_{Gi}</code>, and
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.delta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = rv2Transformer(mat = ex)

# estimated model parameter
print(res$eta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='rvTransformer'>
Root and VOOM Transformation
</h2><span id='topic+rvTransformer'></span>

<h3>Description</h3>

<p>Root and vOOM transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rvTransformer(mat, lib.size = NULL, low = 0.001, upp = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rvTransformer_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="rvTransformer_+3A_lib.size">lib.size</code></td>
<td>

<p>By default, <code>lib.size</code> is a vector of column sums of <code>mat</code>
</p>
</td></tr>
<tr><td><code id="rvTransformer_+3A_low">low</code></td>
<td>

<p>lower bound for the model parameter
</p>
</td></tr>
<tr><td><code id="rvTransformer_+3A_upp">upp</code></td>
<td>

<p>upper bound for the model parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Denote <code class="reqn">x_{gi}</code> as the expression level of the <code class="reqn">g</code>-th gene for
the <code class="reqn">i</code>-th subject.
We perform the root transformation
</p>
<p style="text-align: center;"><code class="reqn">y_{gi}=\frac{t_{gi}^{(1/\eta)}}{(1/\eta)}</code>
</p>
<p>,
where
</p>
<p style="text-align: center;"><code class="reqn">t_{gi}=\frac{\left(x_{gi}+0.5\right)}{X_i+1}\times 10^6</code>
</p>
<p> and
<code class="reqn">X_i=\sum_{g=1}^{G} x_{gi}</code> is the column sum for the <code class="reqn">i</code>-th column
of the matrix <code>mat</code>. The optimal value for the parameter <code class="reqn">\delta</code> is to minimize the squared 
difference between the sample mean and the sample median of the pooled data <code class="reqn">y_{gi}</code>, <code class="reqn">g=1, \ldots, G</code>, <code class="reqn">i=1, \ldots, n</code>,
where <code class="reqn">G</code> is the number of genes and <code class="reqn">n</code> is the number of subjects.
</p>


<h3>Value</h3>

<p>A list with 3 elements:	   
</p>
<table>
<tr><td><code>res.eta</code></td>
<td>
<p>An object returned by <code>optimize</code> function</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>model parameter</p>
</td></tr>
<tr><td><code>mat2</code></td>
<td>
<p>transformed data matrix having the same dimension as <code>mat</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Biobase)

data(es)
print(es)

# expression set
ex = exprs(es)
print(dim(ex))
print(ex[1:3,1:2])

# mean-median before transformation
vec = c(ex)
m = mean(vec)
md = median(vec)
diff = m - md
cat("m=", m, ", md=", md, ", diff=", diff, "\n")

res = rvTransformer(mat = ex)

# estimated model parameter
print(res$eta)

# mean-median after transformation
vec2 = c(res$mat2)
m2 = mean(vec2)
md2 = median(vec2)
diff2 = m2 - md2
cat("m2=", m2, ", md2=", md2, ", diff2=", diff2, "\n")
</code></pre>

<hr>
<h2 id='wilcoxWrapper'>
Wrapper Function for Wilcoxon Rank Sum Test
</h2><span id='topic+wilcoxWrapper'></span>

<h3>Description</h3>

<p>Wrapper function for wilcoxon rank sum test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcoxWrapper(mat, grp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcoxWrapper_+3A_mat">mat</code></td>
<td>

<p>G x n data matrix, where G is the number of genes and n is the number of subjects
</p>
</td></tr>
<tr><td><code id="wilcoxWrapper_+3A_grp">grp</code></td>
<td>

<p>n x 1 vector of subject group info
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each row of <code>mat</code>, we perform Wilcoxon rank sum test.
</p>


<h3>Value</h3>

<p>A G x 1 vector of p-values.
</p>


<h3>Author(s)</h3>

<p>Zeyu Zhang, Danyang Yu, Minseok Seo, Craig P. Hersh, Scott T. Weiss, Weiliang Qiu
</p>


<h3>References</h3>

<p>Zhang Z, Yu D, Seo M, Hersh CP, Weiss ST, Qiu W. 
Novel Data Transformations for RNA-seq Differential Expression Analysis. 
(2019) 9:4820 https://rdcu.be/brDe5
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
