<!DOCTYPE html><html lang="en"><head><title>Help for package countsplit</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {countsplit}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#countsplit'><p>Count splitting</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Splitting a Count Matrix into Independent Folds</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the count splitting methodology from Neufeld et al. (2022) &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2Fkxac047">doi:10.1093/biostatistics/kxac047</a>&gt; and Neufeld et al. (2023) &lt;<a href="https://doi.org/10.48550/arXiv.2307.12985">doi:10.48550/arXiv.2307.12985</a>&gt;. Intended for turning a matrix of single-cell RNA sequencing counts, or similar count datasets, into independent folds that can be used for training/testing or cross validation. Assumes that the entries in the matrix are from a Poisson or a negative binomial distribution.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, methods, Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/anna-neufeld/countsplit">https://github.com/anna-neufeld/countsplit</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/anna-neufeld/countsplit/issues">https://github.com/anna-neufeld/countsplit/issues</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-23 23:56:19 UTC; aneufeld</td>
</tr>
<tr>
<td>Author:</td>
<td>Anna Neufeld [aut, cre, cph],
  Mischko Heming [ctb],
  Joshua Popp [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anna Neufeld &lt;aneufeld@fredhutch.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-24 10:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='countsplit'>Count splitting</h2><span id='topic+countsplit'></span>

<h3>Description</h3>

<p>Takes one matrix of counts and splits it into a specified number of folds. Each fold is a matrix of counts with the same dimension
as the original matrix. Summing element-wise across the folds yields the original data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countsplit(X, folds = 2, epsilon = rep(1/folds, folds), overdisps = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="countsplit_+3A_x">X</code></td>
<td>
<p>A cell-by-gene matrix of integer counts</p>
</td></tr>
<tr><td><code id="countsplit_+3A_folds">folds</code></td>
<td>
<p>An integer specifying how many folds you would like to split your data into.</p>
</td></tr>
<tr><td><code id="countsplit_+3A_epsilon">epsilon</code></td>
<td>
<p>A vector, which has length <code>folds</code>, that stores non-zero elements that sum to one. Determines the proportion of information from X that is allocated to each fold.
When <code>folds</code> is not equal to 2, the recommended (and default) setting is to allocate equal amounts of information to each fold, such that each element is <code>1/folds</code>.
When <code>folds=2</code>, the default is still <code style="white-space: pre;">&#8288;(1/2, 1/2)&#8288;</code>, but other values may be beneficial.</p>
</td></tr>
<tr><td><code id="countsplit_+3A_overdisps">overdisps</code></td>
<td>
<p>If NULL, then Poisson count splitting will be performed. Otherwise, this parameter should be a vector of non-negative numbers whose length is equal to the number of columns of <code>X</code>.
These numbers are the overdispersion parameters for each column in <code>X</code>. If these are unknown, they can be estimated with a function such as
<code>vst</code> in the package <code>sctransform</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When the argument <code>overdisps</code> is set to NULL, this function performs the Poisson count splitting methodology outlined in
Neufeld et al. (2022). With this setting, the folds of data are independent only if the original data were drawn from a Poisson distribution.
</p>
<p>If the data are thought to be overdispersed relative to the Poisson, then we may instead model them as coming from a negative binomial distribution,
If we assume that <code class="reqn">X_{ij} \sim NB(\mu_{ij}, b_j)</code>, where this parameterization means that <code class="reqn"> E[X_{ij}] = \mu_{ij}</code> and <code class="reqn"> Var[X_{ij}] = \mu_{ij} + \mu_{ij}^2/b_j</code>, then
we should pass in <code>overdisps</code> = <code class="reqn">c(b_1, \ldots, b_j)</code>. If this is the correct assumption, then the resulting folds of data will be independent.
This is the negative binomial count splitting method of Neufeld et al. (2023).
</p>
<p>Please see our tutorials and vignettes for more details.
</p>


<h3>Value</h3>

<p>A list of length <code>folds</code>. Each element in the list stores a sparse matrix with the same dimensions as the data <code>X</code>. Each list element is a fold of data.
</p>


<h3>References</h3>

<p>reference
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(countsplit)
library(Matrix)
library(Rcpp)
# A Poisson count splitting example.
n=400
p=2
X &lt;- matrix(rpois(n*p, 7), nrow=n, ncol=p)
split &lt;- countsplit(X, folds=2)
Xtrain &lt;- split[[1]]
Xtest &lt;- split[[2]]
cor(Xtrain[,1], Xtest[,1])
cor(Xtrain[,2], Xtest[,2])

# A negative binomial count splitting example.
X &lt;- matrix(rnbinom(n*p, mu=7, size=7), nrow=n, ncol=p)
split &lt;- countsplit(X, folds=2, overdisps=c(7,7))
Xtrain &lt;- split[[1]]
Xtest &lt;- split[[2]]
cor(Xtrain[,1], Xtest[,1])
cor(Xtrain[,2], Xtest[,2])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
