<!DOCTYPE html><html><head><title>Help for package countToFPKM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {countToFPKM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fpkm'>
<p>Convert Counts to Fragments per Kilobase of Transcript per Million (FPKM)</p>
Convert counts to Fragments Per Kilobase of transcript per Million mapped reads (FPKM)</a></li>
<li><a href='#fpkmheatmap'>
<p>Convert Counts to Fragments per Kilobase of Transcript per Million (FPKM)</p>
A FPKM heatmap plot of the highly variable features in RNA-Seq dataset</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Convert Counts to Fragments per Kilobase of Transcript per
Million (FPKM)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-03-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Ahmed Alhendi [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ahmed Alhendi &lt;asna4@le.ac.uk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), ComplexHeatmap, circlize, stats</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the algorithm described in Trapnell,C. et al. (2010) &lt;<a href="https://doi.org/10.1038%2Fnbt.1621">doi:10.1038/nbt.1621</a>&gt;. This function takes read counts matrix of RNA-Seq data, feature lengths which can be retrieved using 'biomaRt' package, and the mean fragment lengths which can be calculated using the 'CollectInsertSizeMetrics(Picard)' tool. It then returns a matrix of FPKM normalised data by library size and feature effective length. It also provides the user with a quick and reliable function to generate FPKM heatmap plot of the highly variable features in RNA-Seq dataset.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/AAlhendi1707/countToFPKM">https://github.com/AAlhendi1707/countToFPKM</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/AAlhendi1707/countToFPKM/issues">https://github.com/AAlhendi1707/countToFPKM/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-04-07 16:59:02 UTC; asna4</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-04-07 17:42:43 UTC</td>
</tr>
</table>
<hr>
<h2 id='fpkm'>
Convert Counts to Fragments per Kilobase of Transcript per Million (FPKM)
Convert counts to Fragments Per Kilobase of transcript per Million mapped reads (FPKM)
</h2><span id='topic+fpkm'></span>

<h3>Description</h3>

<p>fpkm() function returns a numeric matrix normalized by library size and feature length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fpkm (counts, featureLength, meanFragmentLength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fpkm_+3A_counts">counts</code></td>
<td>
<p>A numeric matrix of raw feature counts</p>
</td></tr>
<tr><td><code id="fpkm_+3A_featurelength">featureLength</code></td>
<td>
<p>A numeric vector with feature lengths which can be obtained using 'biomaRt' package.
The length of items should be as the same of rows in read count matrix.</p>
</td></tr>
<tr><td><code id="fpkm_+3A_meanfragmentlength">meanFragmentLength</code></td>
<td>
<p>A numeric vector with mean fragment lengths, 
which can be calculated using 'CollectInsertSizeMetrics(Picard)' tool.
The length of items should be as the same of columns in read count matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implements the algorithm described in Trapnell,C. et al. (2010). &quot;Transcript assembly and quantification by RNA-seq reveals unannotated transcripts and isoform switching during cell differentiation&quot;. Nat. Biotechnol., 28, 511-515. doi: 10.1038/nbt.1621. This function takes a matrix of read feature counts of RNA-seq, a numeric vector with feature lengths which can be retrieved using the 'biomaRt' package, and a numeric vector with mean fragment length which can be calculated using the 'CollectInsertSizeMetrics(Picard)' tool. It then returns a matrix of FPKM normalised data by library size and feature effective length. Please see the original manuscript for further details.
</p>


<h3>Value</h3>

<p>A data matrix normalized by library size and feature length.
</p>


<h3>References</h3>

<p>Trapnell,C. et al. (2010) Transcript assembly and quantification by RNA-seq reveals unannotated transcripts and isoform switching during cell differentiation. 
Nat. Biotechnol., 28, 511-515. doi: 10.1038/nbt.1621.
</p>
<p>Lior Pachter. Models for transcript quantification from RNA-Seq.
arXiv:1104.3889v2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(countToFPKM)

file.readcounts &lt;- system.file("extdata", "RNA-seq.read.counts.csv", package="countToFPKM")
file.annotations &lt;- system.file("extdata", "Biomart.annotations.hg38.txt", package="countToFPKM")
file.sample.metrics &lt;- system.file("extdata", "RNA-seq.samples.metrics.txt", package="countToFPKM")


# Import the read count matrix data into R.
counts &lt;- as.matrix(read.csv(file.readcounts))

# Import feature annotations.
# Assign feature length into a numeric vector.
gene.annotations &lt;- read.table(file.annotations, sep="\t", header=TRUE)
featureLength &lt;- gene.annotations$length

# Import sample metrics.
# Assign mean fragment length into a numeric vector.
samples.metrics &lt;- read.table(file.sample.metrics, sep="\t", header=TRUE)
meanFragmentLength &lt;- samples.metrics$meanFragmentLength

# Return FPKM into a numeric matrix.
fpkm_matrix &lt;- fpkm (counts, featureLength, meanFragmentLength)
</code></pre>

<hr>
<h2 id='fpkmheatmap'>
Convert Counts to Fragments per Kilobase of Transcript per Million (FPKM)
A FPKM heatmap plot of the highly variable features in RNA-Seq dataset
</h2><span id='topic+fpkmheatmap'></span>

<h3>Description</h3>

<p>fpkmheatmap() function returns a heatmap plot of the highly variable features in RNA-Seq dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fpkmheatmap(fpkm_matrix, topvar=30, showfeaturenames=TRUE, return_log = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fpkmheatmap_+3A_fpkm_matrix">fpkm_matrix</code></td>
<td>
<p>A data matrix normalized by library size and feature length.</p>
</td></tr>
<tr><td><code id="fpkmheatmap_+3A_topvar">topvar</code></td>
<td>
<p>Number of highly variable features to show in heatmap plot.</p>
</td></tr>
<tr><td><code id="fpkmheatmap_+3A_showfeaturenames">showfeaturenames</code></td>
<td>
<p>whether to show the name of features in heatmap plot.
The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="fpkmheatmap_+3A_return_log">return_log</code></td>
<td>
<p>whether to use log10 transformation of (fpkm+1).
The default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fpkmheatmap() function provides the user with a quick and reliable way to generate FPKM heatmap plot of the highly variable features in RNA-Seq dataset. It takes an FPKM numeric matrix which can be obtained using the fpkm() function as input. By default using Pearson correlation - 1 to measure the distance between features, and Spearman correlation -1 for clustering of samples. By default log10 transformation of (FPKM+1) is applied to make variation similar across orders of magnitude. It then uses the var() function to identify the highly variable features to create the heatmap plot using the Heatmap() function from the 'ComplexHeatmap' package.
</p>


<h3>Value</h3>

<p>A FPKM heatmap plot of the highly variable features in RNA-Seq dataset.
</p>


<h3>References</h3>

<p>Trapnell,C. et al. (2010) Transcript assembly and quantification by RNA-seq reveals unannotated transcripts and isoform switching during cell differentiation. 
Nat. Biotechnol., 28, 511-515. doi: 10.1038/nbt.1621.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(countToFPKM)

file.readcounts &lt;- system.file("extdata", "RNA-seq.read.counts.csv", package="countToFPKM")
file.annotations &lt;- system.file("extdata", "Biomart.annotations.hg38.txt", package="countToFPKM")
file.sample.metrics &lt;- system.file("extdata", "RNA-seq.samples.metrics.txt", package="countToFPKM")


# Import the read count matrix data into R.
counts &lt;- as.matrix(read.csv(file.readcounts))

# Import feature annotations. 
# Assign feature length into a numeric vector.
gene.annotations &lt;- read.table(file.annotations, sep="\t", header=TRUE)
featureLength &lt;- gene.annotations$length

# Import sample metrics. 
# Assign mean fragment length into a numeric vector.
samples.metrics &lt;- read.table(file.sample.metrics, sep="\t", header=TRUE)
meanFragmentLength &lt;- samples.metrics$meanFragmentLength

# Return FPKM into a numeric matrix.
fpkm_matrix &lt;- fpkm (counts, featureLength, meanFragmentLength)

# Plot log10(FPKM+1) heatmap of top 30 highly variable features
fpkmheatmap(fpkm_matrix, topvar=30, showfeaturenames=TRUE, return_log = TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
