<!DOCTYPE html><html><head><title>Help for package covid19.analytics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {covid19.analytics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#avecRentrez'><p>function to query NCBI database servers using the &quot;rentrex&quot; library</p></a></li>
<li><a href='#badOption'><p>bad argument error handling function</p></a></li>
<li><a href='#c19.fasta.data'><p>function to obtain FASTA sequence of the SARS-CoV-2 virus</p></a></li>
<li><a href='#c19.genomic.data'><p>function to obtain genomic data from SARS-CoV-2019</p></a></li>
<li><a href='#c19.NP_fasta.data'><p>function to obtain FASTA seqs for nucleotides or proteins from SARS-CoV-2</p></a></li>
<li><a href='#c19.NPs.data'><p>function to obtain data for nucleotides or proteins from SARS-CoV-2</p></a></li>
<li><a href='#c19.OWID.data'><p>function to read data from OWID repos</p></a></li>
<li><a href='#c19.ptree.data'><p>function to obtain &quot;Tree of complete SARS-CoV-2 Sequences as obtained from NCBI&quot;</p></a></li>
<li><a href='#c19.refGenome.data'><p>function to obtain sequencing data grom NCBI</p>
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2</a></li>
<li><a href='#checkGeoLoc'><p>function to check the geographical location</p></a></li>
<li><a href='#chk.TS.data'><p>auxiliary function to check whether a dataset is composed of time series data</p></a></li>
<li><a href='#confBand'><p>function to draw confidence bands, using generalized moving averages/sds</p></a></li>
<li><a href='#consistency.check'><p>function that determines whether there are consistency issues within the data, such as, anomalies in the cumulative quantities of the data as reported by JHU/CCSEGIS</p></a></li>
<li><a href='#covid19.Canada.data'><p>function to import data for Canada</p>
as reported by Health Canada
https://health-infobase.canada.ca/src/data/covidLive/covid19.csv</a></li>
<li><a href='#covid19.data'><p>function to read &quot;live&quot; data from reported covid19 cases</p></a></li>
<li><a href='#covid19.genomic.data'><p>main master (wrapper) function to obtain different types of genomic data for the SARS-CoV-2 virus</p></a></li>
<li><a href='#covid19.JHU.data'><p>function to read &quot;live&quot; data as reported by JHU's CCSE repository</p></a></li>
<li><a href='#covid19.testing.data'><p>function to read data related to covid19 Testing from OWID repo</p></a></li>
<li><a href='#covid19.Toronto_city.data'><p>function to import data from the city of Toronto, ON - Canada</p>
as reported by the City of Toronto
https://www.toronto.ca/home/covid-19/covid-19-pandemic-data/</a></li>
<li><a href='#covid19.Toronto_OD.data'><p>function to import data from the city of Toronto, ON - Canada</p>
as reported by Open Data Toronto
https://open.toronto.ca/dataset/covid-19-cases-in-toronto/
This dataset is updated WEEKLY.</a></li>
<li><a href='#covid19.Toronto.data'><p>function to import data from the city of Toronto, ON - Canada</p>
as reported by the City of Toronto OR Open Data Toronto</a></li>
<li><a href='#covid19.URL_csv.data'><p>function to read CSV from URLs or local replicas</p></a></li>
<li><a href='#covid19.US.data'><p>function to read the TimeSeries US detailed data</p></a></li>
<li><a href='#covid19.vaccination'><p>function to read data related to covid19 vaccinations</p></a></li>
<li><a href='#covid19dashboard'><p>covid19.analytics explorer dashboard</p></a></li>
<li><a href='#covid19Explorer'><p>covid19.analytics explorer dashboard</p></a></li>
<li><a href='#data.checks'><p>function to check for data integrity and data consistency</p></a></li>
<li><a href='#estimateRRs'><p>estimate rolling rates for a given geographical location for an specific TS data</p></a></li>
<li><a href='#gen.glm.model'><p>function to generate models using GLM</p></a></li>
<li><a href='#generate.SIR.model'><p>function to generate a simple SIR (Susceptible-Infected-Recovered) model based on the actual data of the coivd19 cases</p></a></li>
<li><a href='#genModel'><p>function to generate models using Linear Regression &quot;LM&quot;</p></a></li>
<li><a href='#geographicalRegions'><p>function to define continents and its constituent countries</p></a></li>
<li><a href='#getFile'><p>auxiliary function to download files in a protected fashion, i.e. against</p>
errors in the downloading procedure</a></li>
<li><a href='#growth.rate'><p>function to compute daily changes and &quot;Growth Rates&quot; per location; &quot;Growth Rates&quot; defined as the ratio between changes in consecutive days</p></a></li>
<li><a href='#header'><p>auxiliary fn to print &quot;headers&quot; adn 'titles'</p></a></li>
<li><a href='#header0'><p>auxiliary fn to print &quot;headers&quot; adn 'titles'</p></a></li>
<li><a href='#integrity.check'><p>function that determines whether there are integrity issues within the datasets</p>
or changes to the structure of the data as reported by JHU/CCSEGIS</a></li>
<li><a href='#itrends'><p>function to visualize trends in daily changes in time series data interactively</p></a></li>
<li><a href='#live.map'><p>function to map cases in an interactive map</p></a></li>
<li><a href='#loadLibrary'><p>auxiliary function to check and load an specific set of libraries</p></a></li>
<li><a href='#log.sc.setup'><p>aux fn to create a log-scale pulldown option in plotly plots</p></a></li>
<li><a href='#movingFn'><p>generic fn that computes the &quot;fn&quot; on a moving window</p></a></li>
<li><a href='#mrollingRates'><p>function to compute a rolling fn (rate) of multiple quantities from TS data, eg. fatality and recovery rate</p></a></li>
<li><a href='#mtrends'><p>function to visualize different indicators for trends in daily changes of cases reported as time series data, for mutliple (or single) locations</p></a></li>
<li><a href='#nullify.data'><p>remove inconsistencies from data by removing 'suspicious' entries</p></a></li>
<li><a href='#OWID.repos'><p>function to define the OWID repos URLs</p></a></li>
<li><a href='#pandemics.data'><p>function to retrieve historical pandemics data</p></a></li>
<li><a href='#pandemics.loaddata'><p>internal function to retrieve historical data on pandemics</p></a></li>
<li><a href='#plt.SIR.model'><p>function to plot the results from the SIR model fn</p></a></li>
<li><a href='#preProcessingData'><p>auxiliary function to pre-process data per geographical location</p></a></li>
<li><a href='#print'><p>method associated with 'report' objects</p></a></li>
<li><a href='#print.report'><p>function associated to the 'report' object method</p></a></li>
<li><a href='#red.devel.ver'><p>function to redirect users to install devel version from github</p></a></li>
<li><a href='#report.summary'><p>function to summarize the current situation, will download the latest data and summarize the top provinces/cities per case</p></a></li>
<li><a href='#report.Tor'><p>function to obtain main indicators from Toronto data</p></a></li>
<li><a href='#rollingRate'><p>function to compute a rolling fn of a TS data</p></a></li>
<li><a href='#select.per.loc'><p>auxiliary function to select data based on geographical location</p></a></li>
<li><a href='#set.plt.canvas'><p>auxiliary function to set the graphical layout</p></a></li>
<li><a href='#simple.SIR.ODE'><p>Define an ODE system for a simple SIR model</p></a></li>
<li><a href='#single.trend'><p>function to visualize different indicators for trends in daily changes of cases reported as time series data</p></a></li>
<li><a href='#sweep.SIR.models'><p>function to perform a sweep of models and generate values of R0</p></a></li>
<li><a href='#totals.plt'><p>function to plot total number of cases per day for different groups</p></a></li>
<li><a href='#tots.per.location'><p>function to compute totals per location</p></a></li>
<li><a href='#X.covid19.genomic.data'><p>function to obtain sequencing data grom NCBI</p>
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Load and Analyze Live Data from the COVID-19 Pandemic</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.3.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Marcelo Ponce [aut, cre], Amit Sandhel [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marcelo Ponce &lt;m.ponce@utoronto.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Load and analyze updated time series worldwide data of reported cases for the Novel Coronavirus Disease (COVID-19) from different sources, including the Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE) data repository <a href="https://github.com/CSSEGISandData/COVID-19">https://github.com/CSSEGISandData/COVID-19</a>, "Our World in Data" <a href="https://github.com/owid/">https://github.com/owid/</a> among several others. The datasets reporting the COVID-19 cases are available in two main modalities, as a time series sequences and aggregated data for the last day with greater spatial resolution. Several analysis, visualization and modelling functions are available in the package that will allow the user to compute and visualize total number of cases, total number of changes and growth rate globally or for an specific geographical location, while at the same time generating models using these trends; generate interactive visualizations and generate Susceptible-Infected-Recovered (SIR) model for the disease spread.</td>
</tr>
<tr>
<td>Imports:</td>
<td>readxl, ape, rentrez, curl, plotly, htmlwidgets, deSolve,
gplots, pheatmap, shiny, shinydashboard, shinycssloaders, DT,
dplyr, collapsibleTree</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, devtools, roxygen2, markdown, rmarkdown, testthat</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mponce0.github.io/covid19.analytics/">https://mponce0.github.io/covid19.analytics/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mponce0/covid19.analytics/issues">https://github.com/mponce0/covid19.analytics/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-15 05:36:02 UTC; marcelo</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-15 21:50:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='avecRentrez'>function to query NCBI database servers using the &quot;rentrex&quot; library</h2><span id='topic+avecRentrez'></span>

<h3>Description</h3>

<p>function to query NCBI database servers using the &quot;rentrex&quot; library
</p>


<h3>Usage</h3>

<pre><code class='language-R'>avecRentrez(
  DB = "nucleotide",
  max.nr.recs = 20000,
  chunkSize = 100,
  accOnly = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="avecRentrez_+3A_db">DB</code></td>
<td>
<p>database</p>
</td></tr>
<tr><td><code id="avecRentrez_+3A_max.nr.recs">max.nr.recs</code></td>
<td>
<p>maximun number of records to retrieve, there are limitations in the fns and server sides</p>
</td></tr>
<tr><td><code id="avecRentrez_+3A_chunksize">chunkSize</code></td>
<td>
<p>number of records to retrieve at once</p>
</td></tr>
<tr><td><code id="avecRentrez_+3A_acconly">accOnly</code></td>
<td>
<p>boolean indicator for getting only accession codes or whole records</p>
</td></tr>
</table>

<hr>
<h2 id='badOption'>bad argument error handling function</h2><span id='topic+badOption'></span>

<h3>Description</h3>

<p>bad argument error handling function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>badOption(arg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="badOption_+3A_arg">arg</code></td>
<td>
<p>argument choosen</p>
</td></tr>
</table>

<hr>
<h2 id='c19.fasta.data'>function to obtain FASTA sequence of the SARS-CoV-2 virus</h2><span id='topic+c19.fasta.data'></span>

<h3>Description</h3>

<p>function to obtain FASTA sequence of the SARS-CoV-2 virus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.fasta.data(src = "livedata")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.fasta.data_+3A_src">src</code></td>
<td>
<p>argument to indicate where the data is being retrieved from</p>
</td></tr>
</table>

<hr>
<h2 id='c19.genomic.data'>function to obtain genomic data from SARS-CoV-2019</h2><span id='topic+c19.genomic.data'></span>

<h3>Description</h3>

<p>function to obtain genomic data from SARS-CoV-2019
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.genomic.data(src = "livedata", accOnly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.genomic.data_+3A_src">src</code></td>
<td>
<p>argument to indicate what sources are going to be used for retrieving the data: &quot;livedata&quot;, &quot;repo&quot; or &quot;local&quot;
'livedata' will access NCBI servers to acquire the latest possible data, this may incur in significant longer times
'repo' will access an updated replica of the data from a github repository (faster but not necessarily upto the latest udpates)
'local' will access previously archived records within teh package (fastest but not updated)</p>
</td></tr>
<tr><td><code id="c19.genomic.data_+3A_acconly">accOnly</code></td>
<td>
<p>boolean indicator for getting only accession codes or whole records</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing reference genome, annotation data, nucleotides, proteins and list of SRA runs
</p>

<hr>
<h2 id='c19.NP_fasta.data'>function to obtain FASTA seqs for nucleotides or proteins from SARS-CoV-2</h2><span id='topic+c19.NP_fasta.data'></span>

<h3>Description</h3>

<p>function to obtain FASTA seqs for nucleotides or proteins from SARS-CoV-2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.NP_fasta.data(src = "repo", target = "nucleotide")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.NP_fasta.data_+3A_src">src</code></td>
<td>
<p>origin for the data source: &quot;livedata&quot; OR &quot;repo&quot;</p>
</td></tr>
<tr><td><code id="c19.NP_fasta.data_+3A_target">target</code></td>
<td>
<p>&quot;nucleotide&quot;, &quot;protein&quot; or &quot;codingRegion&quot;</p>
</td></tr>
</table>

<hr>
<h2 id='c19.NPs.data'>function to obtain data for nucleotides or proteins from SARS-CoV-2</h2><span id='topic+c19.NPs.data'></span>

<h3>Description</h3>

<p>function to obtain data for nucleotides or proteins from SARS-CoV-2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.NPs.data(
  src = "livedata",
  DB = "nucleotide",
  max.nr.recs = NULL,
  accOnly = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.NPs.data_+3A_src">src</code></td>
<td>
<p>origin for the data source: &quot;livedata&quot;, &quot;repo&quot;, &quot;local&quot;</p>
</td></tr>
<tr><td><code id="c19.NPs.data_+3A_db">DB</code></td>
<td>
<p>database</p>
</td></tr>
<tr><td><code id="c19.NPs.data_+3A_max.nr.recs">max.nr.recs</code></td>
<td>
<p>maximun number of records to retrieve, there are limitations in the fns and server sides</p>
</td></tr>
<tr><td><code id="c19.NPs.data_+3A_acconly">accOnly</code></td>
<td>
<p>boolean indicator for getting only accession codes or whole records</p>
</td></tr>
</table>

<hr>
<h2 id='c19.OWID.data'>function to read data from OWID repos</h2><span id='topic+c19.OWID.data'></span>

<h3>Description</h3>

<p>function to read data from OWID repos
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.OWID.data(repo, disclaimer = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.OWID.data_+3A_repo">repo</code></td>
<td>
<p>URL for reading the data</p>
</td></tr>
<tr><td><code id="c19.OWID.data_+3A_disclaimer">disclaimer</code></td>
<td>
<p>indicate whether the information about the source of the data is disclosed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>datafram object
</p>

<hr>
<h2 id='c19.ptree.data'>function to obtain &quot;Tree of complete SARS-CoV-2 Sequences as obtained from NCBI&quot;</h2><span id='topic+c19.ptree.data'></span>

<h3>Description</h3>

<p>function to obtain &quot;Tree of complete SARS-CoV-2 Sequences as obtained from NCBI&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.ptree.data(src = "livedata")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.ptree.data_+3A_src">src</code></td>
<td>
<p>argument to indicate where the data is being retrieved from</p>
</td></tr>
</table>

<hr>
<h2 id='c19.refGenome.data'>function to obtain sequencing data grom NCBI
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2</h2><span id='topic+c19.refGenome.data'></span>

<h3>Description</h3>

<p>function to obtain sequencing data grom NCBI
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c19.refGenome.data(src = "livedata", graphics.ON = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c19.refGenome.data_+3A_src">src</code></td>
<td>
<p>data origin source: 'livedata', 'repo', 'local'</p>
</td></tr>
<tr><td><code id="c19.refGenome.data_+3A_graphics.on">graphics.ON</code></td>
<td>
<p>flag to activate/deactivate graphical output</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# obtain covid19's genomic data
covid19.gen.seq &lt;- c19.refGenome.data()
# display the actual RNA seq
covid19.gen.seq$NC_045512.2

## End(Not run)
</code></pre>

<hr>
<h2 id='checkGeoLoc'>function to check the geographical location</h2><span id='topic+checkGeoLoc'></span>

<h3>Description</h3>

<p>function to check the geographical location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkGeoLoc(data, geo.loc = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkGeoLoc_+3A_data">data</code></td>
<td>
<p>data.frame with data from covid19</p>
</td></tr>
<tr><td><code id="checkGeoLoc_+3A_geo.loc">geo.loc</code></td>
<td>
<p>list of locations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of capitalized locations within the data
</p>

<hr>
<h2 id='chk.TS.data'>auxiliary function to check whether a dataset is composed of time series data</h2><span id='topic+chk.TS.data'></span>

<h3>Description</h3>

<p>auxiliary function to check whether a dataset is composed of time series data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chk.TS.data(data, xtp = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chk.TS.data_+3A_data">data</code></td>
<td>
<p>data set to consider</p>
</td></tr>
<tr><td><code id="chk.TS.data_+3A_xtp">xtp</code></td>
<td>
<p>indicator whether to stop the program if data is not time series</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a boolean indicator whether the data contains time series values or not
</p>

<hr>
<h2 id='confBand'>function to draw confidence bands, using generalized moving averages/sds</h2><span id='topic+confBand'></span>

<h3>Description</h3>

<p>importFrom  grDevices  rgb
importFrom  graphics  lines polygon
importFrom  stats  sd
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confBand(
  x,
  y,
  x0,
  x1,
  y0,
  y1,
  windowsNbr = 10,
  period = ceiling(length(y)/windowsNbr),
  lcolour = "gray",
  ltype = 4,
  lwidth = 2,
  filling = TRUE
)
</code></pre>

<hr>
<h2 id='consistency.check'>function that determines whether there are consistency issues within the data, such as, anomalies in the cumulative quantities of the data as reported by JHU/CCSEGIS</h2><span id='topic+consistency.check'></span>

<h3>Description</h3>

<p>function that determines whether there are consistency issues within the data, such as, anomalies in the cumulative quantities of the data as reported by JHU/CCSEGIS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consistency.check(
  data,
  n0 = 5,
  nf = ncol(data),
  datasetName = "",
  disclose = FALSE,
  details = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="consistency.check_+3A_data">data</code></td>
<td>
<p>dataset to analyze</p>
</td></tr>
<tr><td><code id="consistency.check_+3A_n0">n0</code></td>
<td>
<p>column where the cumulative data begins</p>
</td></tr>
<tr><td><code id="consistency.check_+3A_nf">nf</code></td>
<td>
<p>column where the cumulative data ends</p>
</td></tr>
<tr><td><code id="consistency.check_+3A_datasetname">datasetName</code></td>
<td>
<p>optional argument to display the name of the dataset</p>
</td></tr>
<tr><td><code id="consistency.check_+3A_disclose">disclose</code></td>
<td>
<p>boolean flag to indicate whether index of problematic entries are returned</p>
</td></tr>
<tr><td><code id="consistency.check_+3A_details">details</code></td>
<td>
<p>optional argument to specify whether to show details about the records where inconsistencies were detected</p>
</td></tr>
</table>

<hr>
<h2 id='covid19.Canada.data'>function to import data for Canada
as reported by Health Canada
https://health-infobase.canada.ca/src/data/covidLive/covid19.csv</h2><span id='topic+covid19.Canada.data'></span>

<h3>Description</h3>

<p>function to import data for Canada
as reported by Health Canada
https://health-infobase.canada.ca/src/data/covidLive/covid19.csv
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.Canada.data(
  data.fmt = "TS",
  local.data = FALSE,
  debrief = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.Canada.data_+3A_data.fmt">data.fmt</code></td>
<td>
<p>&quot;TS&quot; for TimeSeries of cumulative cases or &quot;original&quot; for the data as original reported</p>
</td></tr>
<tr><td><code id="covid19.Canada.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.Canada.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.Canada.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with the latest data reported by &quot;OpenData Toronto&quot; for the city of Toronto, ON - Canada
</p>

<hr>
<h2 id='covid19.data'>function to read &quot;live&quot; data from reported covid19 cases</h2><span id='topic+covid19.data'></span>

<h3>Description</h3>

<p>function to read &quot;live&quot; data from reported covid19 cases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.data(
  case = "aggregated",
  local.data = FALSE,
  debrief = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.data_+3A_case">case</code></td>
<td>
<p>a string indicating the category of the data, possible values are:
&quot;aggregated&quot; :  latest number of cases *aggregated* by country,
&quot;ts-confirmed&quot; :  time data of confirmed cases,
&quot;ts-deaths&quot;    :  time series data of fatal cases,
&quot;ts-recovered&quot; :  time series data of recovered cases,
&quot;ts-ALL&quot;       :  all time series data combined,
&quot;ts-confirmed-US&quot;  : time series data of confirmed cases for the United States,
&quot;ts-deaths-US&quot;     : time series data of fatal cases for the United States,
&quot;ts-dep-confirmed&quot; : time series data of confirmed cases as originally reported (depricated),
&quot;ts-dep-deaths&quot;    : time series data of deaths as originally reported (depricated),
&quot;ts-dep-recovered&quot; : time series data of recovered cases as originally reported (depricated),
&quot;ALL&quot;: all of the above
&quot;ts-Toronto&quot;       :  data for the City of Toronto, ON - Canada</p>
</td></tr>
<tr><td><code id="covid19.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe (or a list in the case of &quot;ALL&quot;) with the daily worlwide indicated type of data per country/region/city
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# reads all possible datastest, returnin a list
covid19.all.datasets &lt;- covid19.data("ALL")

# reads the latest aggregated data
covid19.ALL.agg.cases &lt;- covid19.data("aggregated")
# reads time series data for casualities
covid19.TS.deaths &lt;- covid19.data("ts-deaths")

</code></pre>

<hr>
<h2 id='covid19.genomic.data'>main master (wrapper) function to obtain different types of genomic data for the SARS-CoV-2 virus</h2><span id='topic+covid19.genomic.data'></span>

<h3>Description</h3>

<p>main master (wrapper) function to obtain different types of genomic data for the SARS-CoV-2 virus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.genomic.data(
  type = "genome",
  src = "livedata",
  graphics.ON = TRUE,
  accOnly = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.genomic.data_+3A_type">type</code></td>
<td>
<p>type of data to retrieve, options are: 'genome', 'genomic', 'fasta', 'nucleotide', 'protein', 'ptree'</p>
</td></tr>
<tr><td><code id="covid19.genomic.data_+3A_src">src</code></td>
<td>
<p>source of the data: &quot;livedata&quot;, &quot;repo&quot; or &quot;local&quot;</p>
</td></tr>
<tr><td><code id="covid19.genomic.data_+3A_graphics.on">graphics.ON</code></td>
<td>
<p>boolean option for display associated graphics</p>
</td></tr>
<tr><td><code id="covid19.genomic.data_+3A_acconly">accOnly</code></td>
<td>
<p>boolean indicator for getting only accession codes or whole records</p>
</td></tr>
</table>

<hr>
<h2 id='covid19.JHU.data'>function to read &quot;live&quot; data as reported by JHU's CCSE repository</h2><span id='topic+covid19.JHU.data'></span>

<h3>Description</h3>

<p>function to read &quot;live&quot; data as reported by JHU's CCSE repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.JHU.data(
  case = "aggregated",
  local.data = FALSE,
  debrief = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.JHU.data_+3A_case">case</code></td>
<td>
<p>a string indicating the category of the data, possible values are:
&quot;aggregated&quot; :  latest number of cases *aggregated* by country,
&quot;ts-confirmed&quot; :  time data of confirmed cases,
&quot;ts-deaths&quot;    :  time series data of fatal cases,
&quot;ts-recovered&quot; :  time series data of recovered cases,
&quot;ts-ALL&quot;       :  all time series data combined,
&quot;ts-confirmed-US&quot;  : time series data of confirmed cases for the United States,
&quot;ts-deaths-US&quot;     : time series data of fatal cases for the United States,
&quot;ts-dep-confirmed&quot; : time series data of confirmed cases as originally reported (depricated),
&quot;ts-dep-deaths&quot;    : time series data of deaths as originally reported (depricated),
&quot;ts-dep-recovered&quot; : time series data of recovered cases as originally reported (depricated),
&quot;ALL&quot;: all of the above
&quot;Toronto&quot;    :  data for the City of Toronto, ON - Canada</p>
</td></tr>
<tr><td><code id="covid19.JHU.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.JHU.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.JHU.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe (or a list in the case of &quot;ALL&quot;) with the daily worlwide indicated type of data per country/region/city
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# reads all possible datastest, returnin a list
covid19.all.datasets &lt;- covid19.data("ALL")

# reads the latest aggregated data
covid19.ALL.agg.cases &lt;- covid19.data("aggregated")
# reads time series data for casualities
covid19.TS.deaths &lt;- covid19.data("ts-deaths")

</code></pre>

<hr>
<h2 id='covid19.testing.data'>function to read data related to covid19 Testing from OWID repo</h2><span id='topic+covid19.testing.data'></span>

<h3>Description</h3>

<p>function to read data related to covid19 Testing from OWID repo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.testing.data(tgt = "testing", disclaimer = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.testing.data_+3A_tgt">tgt</code></td>
<td>
<p>selects between time series data ('testing') and details and overall view of the data (&quot;testing.details&quot;)</p>
</td></tr>
<tr><td><code id="covid19.testing.data_+3A_disclaimer">disclaimer</code></td>
<td>
<p>indicates whether the information about the source of the data is disclosed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe containing list of countries (Entity) and testing data
</p>

<hr>
<h2 id='covid19.Toronto_city.data'>function to import data from the city of Toronto, ON - Canada
as reported by the City of Toronto
https://www.toronto.ca/home/covid-19/covid-19-pandemic-data/</h2><span id='topic+covid19.Toronto_city.data'></span>

<h3>Description</h3>

<p>function to import data from the city of Toronto, ON - Canada
as reported by the City of Toronto
https://www.toronto.ca/home/covid-19/covid-19-pandemic-data/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.Toronto_city.data(
  data.fmt = "TS",
  local.data = FALSE,
  debrief = FALSE,
  OLD.fmt = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.Toronto_city.data_+3A_data.fmt">data.fmt</code></td>
<td>
<p>&quot;TS&quot; for TimeSeries of cumulative cases or &quot;original&quot; for the data as reported in the google-document with multiple sheets</p>
</td></tr>
<tr><td><code id="covid19.Toronto_city.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.Toronto_city.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.Toronto_city.data_+3A_old.fmt">OLD.fmt</code></td>
<td>
<p>boolean flag to specify if the data is being read in an old format</p>
</td></tr>
<tr><td><code id="covid19.Toronto_city.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe (or a list in the case of &quot;original&quot;) with the latest data reported for the city of Toronto, ON - Canada
</p>

<hr>
<h2 id='covid19.Toronto_OD.data'>function to import data from the city of Toronto, ON - Canada
as reported by Open Data Toronto
https://open.toronto.ca/dataset/covid-19-cases-in-toronto/
This dataset is updated WEEKLY.</h2><span id='topic+covid19.Toronto_OD.data'></span>

<h3>Description</h3>

<p>function to import data from the city of Toronto, ON - Canada
as reported by Open Data Toronto
https://open.toronto.ca/dataset/covid-19-cases-in-toronto/
This dataset is updated WEEKLY.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.Toronto_OD.data(
  data.fmt = "TS",
  local.data = FALSE,
  debrief = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.Toronto_OD.data_+3A_data.fmt">data.fmt</code></td>
<td>
<p>&quot;TS&quot; for TimeSeries of cumulative cases or &quot;original&quot; for the data as original reported</p>
</td></tr>
<tr><td><code id="covid19.Toronto_OD.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.Toronto_OD.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.Toronto_OD.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with the latest data reported by &quot;OpenData Toronto&quot; for the city of Toronto, ON - Canada
</p>

<hr>
<h2 id='covid19.Toronto.data'>function to import data from the city of Toronto, ON - Canada
as reported by the City of Toronto OR Open Data Toronto</h2><span id='topic+covid19.Toronto.data'></span>

<h3>Description</h3>

<p>function to import data from the city of Toronto, ON - Canada
as reported by the City of Toronto OR Open Data Toronto
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.Toronto.data(
  origin = "OD",
  data.fmt = "TS",
  local.data = FALSE,
  debrief = FALSE,
  OLD.fmt = FALSE,
  acknowledge = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.Toronto.data_+3A_origin">origin</code></td>
<td>
<p>select between the &quot;City of Toronto&quot; ('city') OR &quot;Open Data Toronto&quot; ('OD')</p>
</td></tr>
<tr><td><code id="covid19.Toronto.data_+3A_data.fmt">data.fmt</code></td>
<td>
<p>&quot;TS&quot; for TimeSeries of cumulative cases or &quot;original&quot; for the data as reported in the google-document with multiple sheets</p>
</td></tr>
<tr><td><code id="covid19.Toronto.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.Toronto.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.Toronto.data_+3A_old.fmt">OLD.fmt</code></td>
<td>
<p>boolean flag to specify if the data is being read in an old format</p>
</td></tr>
<tr><td><code id="covid19.Toronto.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe (or a list in the case of &quot;original&quot;) with the latest data reported for the city of Toronto, ON - Canada
</p>

<hr>
<h2 id='covid19.URL_csv.data'>function to read CSV from URLs or local replicas</h2><span id='topic+covid19.URL_csv.data'></span>

<h3>Description</h3>

<p>function to read CSV from URLs or local replicas
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.URL_csv.data(
  local.data = FALSE,
  acknowledge = FALSE,
  srcURL = "",
  srcName = "",
  locFileName = NA,
  locVarName = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.URL_csv.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.URL_csv.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
<tr><td><code id="covid19.URL_csv.data_+3A_srcurl">srcURL</code></td>
<td>
<p>URL from where to obtain the data</p>
</td></tr>
<tr><td><code id="covid19.URL_csv.data_+3A_srcname">srcName</code></td>
<td>
<p>name of the source</p>
</td></tr>
<tr><td><code id="covid19.URL_csv.data_+3A_locfilename">locFileName</code></td>
<td>
<p>name of the file to read from local repo</p>
</td></tr>
<tr><td><code id="covid19.URL_csv.data_+3A_locvarname">locVarName</code></td>
<td>
<p>name of the variable loaded from local file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data as oriignally obtained from the URL src
</p>

<hr>
<h2 id='covid19.US.data'>function to read the TimeSeries US detailed data</h2><span id='topic+covid19.US.data'></span>

<h3>Description</h3>

<p>function to read the TimeSeries US detailed data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.US.data(local.data = FALSE, debrief = FALSE, acknowledge = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.US.data_+3A_local.data">local.data</code></td>
<td>
<p>boolean flag to indicate whether the data will be read from the local repo, in case of connectivity issues or data integrity</p>
</td></tr>
<tr><td><code id="covid19.US.data_+3A_debrief">debrief</code></td>
<td>
<p>boolean specifying whether information about the read data is going to be displayed in screen</p>
</td></tr>
<tr><td><code id="covid19.US.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>boolean flag to indicate that the user acknowledges where the data is coming from.  If FALSE, display data acquisition messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TimeSeries dataframe with data for the US
</p>

<hr>
<h2 id='covid19.vaccination'>function to read data related to covid19 vaccinations</h2><span id='topic+covid19.vaccination'></span>

<h3>Description</h3>

<p>function to read data related to covid19 vaccinations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19.vaccination(tgt = "global", data.fmt = "orig", disclaimer = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19.vaccination_+3A_tgt">tgt</code></td>
<td>
<p>selects data type: 'global','us','country','locations'</p>
</td></tr>
<tr><td><code id="covid19.vaccination_+3A_data.fmt">data.fmt</code></td>
<td>
<p>selects the format of the data, options are: 'orig' (original as reported by the OWID repo)</p>
</td></tr>
<tr><td><code id="covid19.vaccination_+3A_disclaimer">disclaimer</code></td>
<td>
<p>indicates whether the information about the source of the data is disclosed</p>
</td></tr>
</table>

<hr>
<h2 id='covid19dashboard'>covid19.analytics explorer dashboard</h2><span id='topic+covid19dashboard'></span>

<h3>Description</h3>

<p>covid19.analytics explorer dashboard
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19dashboard(locn = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19dashboard_+3A_locn">locn</code></td>
<td>
<p>geographical location to use as default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with shinyApp UI and server
</p>

<hr>
<h2 id='covid19Explorer'>covid19.analytics explorer dashboard</h2><span id='topic+covid19Explorer'></span>

<h3>Description</h3>

<p>covid19.analytics explorer dashboard
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid19Explorer(locn = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid19Explorer_+3A_locn">locn</code></td>
<td>
<p>geographical location to use as default</p>
</td></tr>
</table>

<hr>
<h2 id='data.checks'>function to check for data integrity and data consistency</h2><span id='topic+data.checks'></span>

<h3>Description</h3>

<p>function to check for data integrity and data consistency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data.checks(
  data,
  n0 = 5,
  nf = ncol(data),
  datasetName = "",
  details = TRUE,
  disclose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data.checks_+3A_data">data</code></td>
<td>
<p>dataset to analyze</p>
</td></tr>
<tr><td><code id="data.checks_+3A_n0">n0</code></td>
<td>
<p>column where the cumulative data begins</p>
</td></tr>
<tr><td><code id="data.checks_+3A_nf">nf</code></td>
<td>
<p>column where the cumulative data ends</p>
</td></tr>
<tr><td><code id="data.checks_+3A_datasetname">datasetName</code></td>
<td>
<p>optional argument to display the name of the dataset</p>
</td></tr>
<tr><td><code id="data.checks_+3A_details">details</code></td>
<td>
<p>optional argument to specify whether to show details about the records where inconsistencies were detected</p>
</td></tr>
<tr><td><code id="data.checks_+3A_disclose">disclose</code></td>
<td>
<p>boolean flag to indicate whether index of problematic entries are returned</p>
</td></tr>
</table>

<hr>
<h2 id='estimateRRs'>estimate rolling rates for a given geographical location for an specific TS data</h2><span id='topic+estimateRRs'></span>

<h3>Description</h3>

<p>estimate rolling rates for a given geographical location for an specific TS data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateRRs(
  data = NULL,
  geo.loc = NULL,
  period = NULL,
  graphics.ON = TRUE,
  splitG = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimateRRs_+3A_data">data</code></td>
<td>
<p>time series dataset to consider</p>
</td></tr>
<tr><td><code id="estimateRRs_+3A_geo.loc">geo.loc</code></td>
<td>
<p>country/region to analyze</p>
</td></tr>
<tr><td><code id="estimateRRs_+3A_period">period</code></td>
<td>
<p>length of window</p>
</td></tr>
<tr><td><code id="estimateRRs_+3A_graphics.on">graphics.ON</code></td>
<td>
<p>boolean flag to activate/deactivate graphical output</p>
</td></tr>
<tr><td><code id="estimateRRs_+3A_splitg">splitG</code></td>
<td>
<p>boolean flag for having the graphical output separated or not</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># the following examples take longer than 10 sec, and triggers CRAN checks
## Not run: 
estimateRRs(covid19.data("TS-all"), geo.loc='Peru', period=7)
estimateRRs(covid19.data("TS-all"),
	geo.loc=c('Peru','Argentina','Uruguay','US','Spain','Japan'), period=7)

## End(Not run)

</code></pre>

<hr>
<h2 id='gen.glm.model'>function to generate models using GLM</h2><span id='topic+gen.glm.model'></span>

<h3>Description</h3>

<p>function to generate models using GLM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen.glm.model(y, family = Gamma(link = "log"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen.glm.model_+3A_y">y</code></td>
<td>
<p>vector containing the data to fit</p>
</td></tr>
<tr><td><code id="gen.glm.model_+3A_family">family</code></td>
<td>
<p>family to use in the GLM method</p>
</td></tr>
</table>

<hr>
<h2 id='generate.SIR.model'>function to generate a simple SIR (Susceptible-Infected-Recovered) model based on the actual data of the coivd19 cases</h2><span id='topic+generate.SIR.model'></span>

<h3>Description</h3>

<p>function to generate a simple SIR (Susceptible-Infected-Recovered) model based on the actual data of the coivd19 cases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate.SIR.model(
  data = NULL,
  geo.loc = "Hubei",
  t0 = NULL,
  t1 = NULL,
  deltaT = NULL,
  tfinal = 90,
  fatality.rate = 0.02,
  tot.population = 1.4e+09,
  staticPlt = TRUE,
  interactiveFig = FALSE,
  add.extras = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate.SIR.model_+3A_data">data</code></td>
<td>
<p>time series dataset to consider</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_geo.loc">geo.loc</code></td>
<td>
<p>country/region to analyze</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_t0">t0</code></td>
<td>
<p>initial period of time for data consideration</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_t1">t1</code></td>
<td>
<p>final period of time for data consideration</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_deltat">deltaT</code></td>
<td>
<p>interval period of time from t0, ie. number of days to consider since t0</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_tfinal">tfinal</code></td>
<td>
<p>total number of days</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_fatality.rate">fatality.rate</code></td>
<td>
<p>rate of causality, deafault value of 2 percent</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_tot.population">tot.population</code></td>
<td>
<p>total population of the country/region</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_staticplt">staticPlt</code></td>
<td>
<p>optional flag to activate/deactive plotting of the data and the SIR model generated</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_interactivefig">interactiveFig</code></td>
<td>
<p>optional flag to activate/deactive the generation of an interactive plot of the data and the SIR model generated</p>
</td></tr>
<tr><td><code id="generate.SIR.model_+3A_add.extras">add.extras</code></td>
<td>
<p>boolean flag to add extra indicators, such as, the &quot;force of infection&quot; and time derivatives</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- covid19.data("ts-confirmed")
generate.SIR.model(data,"Hubei", t0=1,t1=15)
generate.SIR.model(data,"Germany",tot.population=83149300)
generate.SIR.model(data,"Uruguay", tot.population=3500000)
generate.SIR.model(data,"Canada", tot.population=37590000, add.extras=TRUE)

</code></pre>

<hr>
<h2 id='genModel'>function to generate models using Linear Regression &quot;LM&quot;</h2><span id='topic+genModel'></span>

<h3>Description</h3>

<p>function to generate models using Linear Regression &quot;LM&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genModel(y, deg = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genModel_+3A_y">y</code></td>
<td>
<p>vector containing the data to fit</p>
</td></tr>
<tr><td><code id="genModel_+3A_deg">deg</code></td>
<td>
<p>degree of the polynomial fit</p>
</td></tr>
</table>

<hr>
<h2 id='geographicalRegions'>function to define continents and its constituent countries</h2><span id='topic+geographicalRegions'></span>

<h3>Description</h3>

<p>function to define continents and its constituent countries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geographicalRegions(cont = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geographicalRegions_+3A_cont">cont</code></td>
<td>
<p>optional argumetn, to specify a particular continent; if no argument is given then it returns all the continents and countries for each</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with the composition of continents
</p>

<hr>
<h2 id='getFile'>auxiliary function to download files in a protected fashion, i.e. against
errors in the downloading procedure</h2><span id='topic+getFile'></span>

<h3>Description</h3>

<p>auxiliary function to download files in a protected fashion, i.e. against
errors in the downloading procedure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFile(url = NULL, fileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFile_+3A_url">url</code></td>
<td>
<p>resource's URL on the web</p>
</td></tr>
<tr><td><code id="getFile_+3A_filename">fileName</code></td>
<td>
<p>name of the resource (file) to download
</p>
<p>keywords internal</p>
</td></tr>
</table>

<hr>
<h2 id='growth.rate'>function to compute daily changes and &quot;Growth Rates&quot; per location; &quot;Growth Rates&quot; defined as the ratio between changes in consecutive days</h2><span id='topic+growth.rate'></span>

<h3>Description</h3>

<p>function to compute daily changes and &quot;Growth Rates&quot; per location; &quot;Growth Rates&quot; defined as the ratio between changes in consecutive days
</p>


<h3>Usage</h3>

<pre><code class='language-R'>growth.rate(
  data0,
  geo.loc = NULL,
  stride = 1,
  info = "",
  staticPlt = TRUE,
  interactiveFig = FALSE,
  interactive.display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="growth.rate_+3A_data0">data0</code></td>
<td>
<p>data.frame with *time series* data from covid19</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_geo.loc">geo.loc</code></td>
<td>
<p>list of locations</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_stride">stride</code></td>
<td>
<p>how frequently to compute the growth rate in units of days</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_info">info</code></td>
<td>
<p>additional information to include in plots' title</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_staticplt">staticPlt</code></td>
<td>
<p>boolean flag to indicate whether static plots would be generated or not</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_interactivefig">interactiveFig</code></td>
<td>
<p>boolean flag to indicate whether interactice figures would be generated or not</p>
</td></tr>
<tr><td><code id="growth.rate_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean flag to indicate whether the interactive plot will be displayed (pushed) to your browser</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing two dataframes: one reporting changes on daily baisis and a second one reporting growth rates, for the indicated regions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###\donttest{
# read data for confirmed cases
data &lt;- covid19.data("ts-confirmed")
# compute changes and growth rates per location for all the countries
# growth.rate(data)
# compute changes and growth rates per location for 'Italy'
growth.rate(data,geo.loc="Italy")
# compute changes and growth rates per location for 'Italy' and 'Germany'
growth.rate(data,geo.loc=c("Italy","Germany"))
###}
</code></pre>

<hr>
<h2 id='header'>auxiliary fn to print &quot;headers&quot; adn 'titles'</h2><span id='topic+header'></span>

<h3>Description</h3>

<p>auxiliary fn to print &quot;headers&quot; adn 'titles'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>header(x = "-", title = "", total.len = 80, eol = "\n")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="header_+3A_x">x</code></td>
<td>
<p>character to use as lines</p>
</td></tr>
<tr><td><code id="header_+3A_title">title</code></td>
<td>
<p>title to dispo</p>
</td></tr>
<tr><td><code id="header_+3A_total.len">total.len</code></td>
<td>
<p>length of the line</p>
</td></tr>
<tr><td><code id="header_+3A_eol">eol</code></td>
<td>
<p>end of line character</p>
</td></tr>
</table>

<hr>
<h2 id='header0'>auxiliary fn to print &quot;headers&quot; adn 'titles'</h2><span id='topic+header0'></span>

<h3>Description</h3>

<p>auxiliary fn to print &quot;headers&quot; adn 'titles'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>header0(x = "-", title = "", total.len = 80, eol = "\n")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="header0_+3A_x">x</code></td>
<td>
<p>character to use as lines</p>
</td></tr>
<tr><td><code id="header0_+3A_title">title</code></td>
<td>
<p>title to dispo</p>
</td></tr>
<tr><td><code id="header0_+3A_total.len">total.len</code></td>
<td>
<p>length of the line</p>
</td></tr>
<tr><td><code id="header0_+3A_eol">eol</code></td>
<td>
<p>end of line character</p>
</td></tr>
</table>

<hr>
<h2 id='integrity.check'>function that determines whether there are integrity issues within the datasets
or changes to the structure of the data as reported by JHU/CCSEGIS</h2><span id='topic+integrity.check'></span>

<h3>Description</h3>

<p>function that determines whether there are integrity issues within the datasets
or changes to the structure of the data as reported by JHU/CCSEGIS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrity.check(data, datasetName = "", disclose = FALSE, recommend = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrity.check_+3A_data">data</code></td>
<td>
<p>dataset to analyze</p>
</td></tr>
<tr><td><code id="integrity.check_+3A_datasetname">datasetName</code></td>
<td>
<p>optional argument to display the name of the dataset</p>
</td></tr>
<tr><td><code id="integrity.check_+3A_disclose">disclose</code></td>
<td>
<p>boolean flag to indicate whether index of problematic entries are returned</p>
</td></tr>
<tr><td><code id="integrity.check_+3A_recommend">recommend</code></td>
<td>
<p>optional flag to recommend further actions</p>
</td></tr>
</table>

<hr>
<h2 id='itrends'>function to visualize trends in daily changes in time series data interactively</h2><span id='topic+itrends'></span>

<h3>Description</h3>

<p>function to visualize trends in daily changes in time series data interactively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>itrends(
  ts.data = NULL,
  geo.loc = NULL,
  with.totals = FALSE,
  fileName = NULL,
  interactive.display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="itrends_+3A_ts.data">ts.data</code></td>
<td>
<p>time series dataset to process</p>
</td></tr>
<tr><td><code id="itrends_+3A_geo.loc">geo.loc</code></td>
<td>
<p>geographical location, country/region or province/state to restrict the analysis to</p>
</td></tr>
<tr><td><code id="itrends_+3A_with.totals">with.totals</code></td>
<td>
<p>a boolean flag to indicate whether the global totals should be displayed with the records for the specific location</p>
</td></tr>
<tr><td><code id="itrends_+3A_filename">fileName</code></td>
<td>
<p>file where to save the HTML version of the interactive figure</p>
</td></tr>
<tr><td><code id="itrends_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean flag to indicate whether the interactive plot will be displayed (pushed) to your browser</p>
</td></tr>
</table>

<hr>
<h2 id='live.map'>function to map cases in an interactive map</h2><span id='topic+live.map'></span>

<h3>Description</h3>

<p>function to map cases in an interactive map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>live.map(
  data = covid19.data(),
  select.projctn = TRUE,
  projctn = "orthographic",
  title = "",
  no.legend = FALSE,
  szRef = 0.2,
  fileName = NULL,
  interactive.display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="live.map_+3A_data">data</code></td>
<td>
<p>data to be used</p>
</td></tr>
<tr><td><code id="live.map_+3A_select.projctn">select.projctn</code></td>
<td>
<p>argument to activate or deactivate the pulldown menu for selecting the type of projection</p>
</td></tr>
<tr><td><code id="live.map_+3A_projctn">projctn</code></td>
<td>
<p>initial type of map-projection to use, possible values are:
&quot;equirectangular&quot; | &quot;mercator&quot; | &quot;orthographic&quot; | &quot;natural earth&quot; | &quot;kavrayskiy7&quot; | &quot;miller&quot; | &quot;robinson&quot; | &quot;eckert4&quot; | &quot;azimuthal equal area&quot; | &quot;azimuthal equidistant&quot; | &quot;conic equal area&quot; | &quot;conic conformal&quot; | &quot;conic equidistant&quot; | &quot;gnomonic&quot; | &quot;stereographic&quot; | &quot;mollweide&quot; | &quot;hammer&quot; | &quot;transverse mercator&quot; | &quot;albers usa&quot; | &quot;winkel tripel&quot; | &quot;aitoff&quot; | &quot;sinusoidal&quot;</p>
</td></tr>
<tr><td><code id="live.map_+3A_title">title</code></td>
<td>
<p>a string with a title to add to the plot</p>
</td></tr>
<tr><td><code id="live.map_+3A_no.legend">no.legend</code></td>
<td>
<p>parameter to turn off or on the legend on the right with the list of countries</p>
</td></tr>
<tr><td><code id="live.map_+3A_szref">szRef</code></td>
<td>
<p>numerical value to use as reference, to scale up the size of the bubbles in the map, from 0 to 1 (smmaller value &ndash;&gt; larger bubbles)</p>
</td></tr>
<tr><td><code id="live.map_+3A_filename">fileName</code></td>
<td>
<p>file where to save the HTML version of the interactive figure</p>
</td></tr>
<tr><td><code id="live.map_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean argument for enabling or not displaying the figure</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieve aggregated data
data &lt;- covid19.data("aggregated")
# interactive map of aggregated cases -- with more spatial resolution
live.map(data)

# interactive map of the time series data of the confirmed cases
# with less spatial resolution, ie. aggregated by country
live.map(covid19.data("ts-confirmed"))

## End(Not run)

</code></pre>

<hr>
<h2 id='loadLibrary'>auxiliary function to check and load an specific set of libraries</h2><span id='topic+loadLibrary'></span>

<h3>Description</h3>

<p>auxiliary function to check and load an specific set of libraries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadLibrary(lib)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadLibrary_+3A_lib">lib</code></td>
<td>
<p>is a list of packages to be loaded</p>
</td></tr>
</table>

<hr>
<h2 id='log.sc.setup'>aux fn to create a log-scale pulldown option in plotly plots</h2><span id='topic+log.sc.setup'></span>

<h3>Description</h3>

<p>aux fn to create a log-scale pulldown option in plotly plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sc.setup'
log(nbr.traces, tot.traces)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log.sc.setup_+3A_nbr.traces">nbr.traces</code></td>
<td>
<p>number of traces in order to set Ts/Fs</p>
</td></tr>
</table>

<hr>
<h2 id='movingFn'>generic fn that computes the &quot;fn&quot; on a moving window</h2><span id='topic+movingFn'></span>

<h3>Description</h3>

<p>generic fn that computes the &quot;fn&quot; on a moving window
</p>


<h3>Usage</h3>

<pre><code class='language-R'>movingFn(x, fn = mean, period = length(x), direction = "forward")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="movingFn_+3A_x">x</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="movingFn_+3A_fn">fn</code></td>
<td>
<p>a function to be applied/computed, default is set to mean()</p>
</td></tr>
<tr><td><code id="movingFn_+3A_period">period</code></td>
<td>
<p>size of the &quot;moving window&quot;, default set to the lenght of the vector</p>
</td></tr>
<tr><td><code id="movingFn_+3A_direction">direction</code></td>
<td>
<p>type of moving avergage to consider: &quot;forward&quot;, &quot;centered&quot;, &quot;backward&quot;; ie. whether the window computation is ( &quot;centered&quot; / &quot;forward&quot; / &quot;backward&quot; ) wrt the data series</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with the 'moving operation' applied to the x vector
</p>

<hr>
<h2 id='mrollingRates'>function to compute a rolling fn (rate) of multiple quantities from TS data, eg. fatality and recovery rate</h2><span id='topic+mrollingRates'></span>

<h3>Description</h3>

<p>function to compute a rolling fn (rate) of multiple quantities from TS data, eg. fatality and recovery rate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrollingRates(data = NULL, geo.loc = NULL, fn = mean, period)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrollingRates_+3A_data">data</code></td>
<td>
<p>time series dataset to consider</p>
</td></tr>
<tr><td><code id="mrollingRates_+3A_geo.loc">geo.loc</code></td>
<td>
<p>country/region to analyze</p>
</td></tr>
<tr><td><code id="mrollingRates_+3A_fn">fn</code></td>
<td>
<p>function to compute rolling</p>
</td></tr>
<tr><td><code id="mrollingRates_+3A_period">period</code></td>
<td>
<p>length of window</p>
</td></tr>
</table>

<hr>
<h2 id='mtrends'>function to visualize different indicators for trends in daily changes of cases reported as time series data, for mutliple (or single) locations</h2><span id='topic+mtrends'></span>

<h3>Description</h3>

<p>function to visualize different indicators for trends in daily changes of cases reported as time series data, for mutliple (or single) locations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtrends(data, geo.loc = NULL, confBnd = TRUE, info = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mtrends_+3A_data">data</code></td>
<td>
<p>data.frame with *time series* data from covid19</p>
</td></tr>
<tr><td><code id="mtrends_+3A_geo.loc">geo.loc</code></td>
<td>
<p>list of locations</p>
</td></tr>
<tr><td><code id="mtrends_+3A_confbnd">confBnd</code></td>
<td>
<p>flag to activate/deactivate drawing of confidence bands base on a moving average window</p>
</td></tr>
<tr><td><code id="mtrends_+3A_info">info</code></td>
<td>
<p>additional info to display in the plot</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># triggers CRAN checks for timing
## Not run: 
ts.data &lt;- covid19.data("ts-confirmed")
mtrends(ts.data, geo.loc=c("Canada","Ontario","Uruguay","Italy"))

## End(Not run)
</code></pre>

<hr>
<h2 id='nullify.data'>remove inconsistencies from data by removing 'suspicious' entries</h2><span id='topic+nullify.data'></span>

<h3>Description</h3>

<p>remove inconsistencies from data by removing 'suspicious' entries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nullify.data(data, stringent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nullify.data_+3A_data">data</code></td>
<td>
<p>dataset to process</p>
</td></tr>
<tr><td><code id="nullify.data_+3A_stringent">stringent</code></td>
<td>
<p>only return records with &quot;complete cases&quot;</p>
</td></tr>
</table>

<hr>
<h2 id='OWID.repos'>function to define the OWID repos URLs</h2><span id='topic+OWID.repos'></span>

<h3>Description</h3>

<p>function to define the OWID repos URLs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OWID.repos(tgt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OWID.repos_+3A_tgt">tgt</code></td>
<td>
<p>target case:  'VAC.global','VAC.us','VAC.country','VAC.locations', 'testing', 'testing.details'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>URL
</p>

<hr>
<h2 id='pandemics.data'>function to retrieve historical pandemics data</h2><span id='topic+pandemics.data'></span>

<h3>Description</h3>

<p>function to retrieve historical pandemics data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pandemics.data(acknowledge = TRUE, show = FALSE, tgt = "pandemics")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pandemics.data_+3A_acknowledge">acknowledge</code></td>
<td>
<p>displays details on the data sources</p>
</td></tr>
<tr><td><code id="pandemics.data_+3A_show">show</code></td>
<td>
<p>displays data</p>
</td></tr>
<tr><td><code id="pandemics.data_+3A_tgt">tgt</code></td>
<td>
<p>which data set to read &ndash; options are  'pandemics' OR 'pandemics_vaccines'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='pandemics.loaddata'>internal function to retrieve historical data on pandemics</h2><span id='topic+pandemics.loaddata'></span>

<h3>Description</h3>

<p>internal function to retrieve historical data on pandemics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pandemics.loaddata(
  tgt.file = "pandemics.RDS",
  acknowledge = TRUE,
  show = FALSE,
  src.descr = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pandemics.loaddata_+3A_tgt.file">tgt.file</code></td>
<td>
<p>which data set to read</p>
</td></tr>
<tr><td><code id="pandemics.loaddata_+3A_acknowledge">acknowledge</code></td>
<td>
<p>displays details on the data sources</p>
</td></tr>
<tr><td><code id="pandemics.loaddata_+3A_show">show</code></td>
<td>
<p>displays data</p>
</td></tr>
<tr><td><code id="pandemics.loaddata_+3A_src.descr">src.descr</code></td>
<td>
<p>description of the source of the data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='plt.SIR.model'>function to plot the results from the SIR model fn</h2><span id='topic+plt.SIR.model'></span>

<h3>Description</h3>

<p>function to plot the results from the SIR model fn
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plt.SIR.model(
  SIR.model,
  geo.loc = "",
  interactiveFig = FALSE,
  fileName = NULL,
  interactive.display = TRUE,
  add.extras = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plt.SIR.model_+3A_sir.model">SIR.model</code></td>
<td>
<p>model resulting from the generate.SIR.model() fn</p>
</td></tr>
<tr><td><code id="plt.SIR.model_+3A_geo.loc">geo.loc</code></td>
<td>
<p>optional string to specify geographical location</p>
</td></tr>
<tr><td><code id="plt.SIR.model_+3A_interactivefig">interactiveFig</code></td>
<td>
<p>optional flag to activate interactive plot</p>
</td></tr>
<tr><td><code id="plt.SIR.model_+3A_filename">fileName</code></td>
<td>
<p>file where to save the HTML version of the interactive figure</p>
</td></tr>
<tr><td><code id="plt.SIR.model_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean flag to indicate whether the interactive plot will be displayed (pushed) to your browser</p>
</td></tr>
<tr><td><code id="plt.SIR.model_+3A_add.extras">add.extras</code></td>
<td>
<p>boolean flag to add extra indicators, such as, the &quot;force of infection&quot; and time derivatives</p>
</td></tr>
</table>

<hr>
<h2 id='preProcessingData'>auxiliary function to pre-process data per geographical location</h2><span id='topic+preProcessingData'></span>

<h3>Description</h3>

<p>auxiliary function to pre-process data per geographical location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preProcessingData(data0, geo.loc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preProcessingData_+3A_data0">data0</code></td>
<td>
<p>data set</p>
</td></tr>
<tr><td><code id="preProcessingData_+3A_geo.loc">geo.loc</code></td>
<td>
<p>geopgraphical location, can be a country, region, province or city
</p>
<p>#@keywords internal</p>
</td></tr>
</table>

<hr>
<h2 id='print'>method associated with 'report' objects</h2><span id='topic+print'></span>

<h3>Description</h3>

<p>method associated with 'report' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print(report.obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_+3A_report.obj">report.obj</code></td>
<td>
<p>'report' object</p>
</td></tr>
</table>

<hr>
<h2 id='print.report'>function associated to the 'report' object method</h2><span id='topic+print.report'></span>

<h3>Description</h3>

<p>function associated to the 'report' object method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'report'
print(report.obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.report_+3A_report.obj">report.obj</code></td>
<td>
<p>'report' object</p>
</td></tr>
</table>

<hr>
<h2 id='red.devel.ver'>function to redirect users to install devel version from github</h2><span id='topic+red.devel.ver'></span>

<h3>Description</h3>

<p>function to redirect users to install devel version from github
</p>


<h3>Usage</h3>

<pre><code class='language-R'>red.devel.ver(fileRDS, force = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="red.devel.ver_+3A_filerds">fileRDS</code></td>
<td>
<p>missing data file</p>
</td></tr>
<tr><td><code id="red.devel.ver_+3A_force">force</code></td>
<td>
<p>boolean flag to force stoping the code</p>
</td></tr>
</table>

<hr>
<h2 id='report.summary'>function to summarize the current situation, will download the latest data and summarize the top provinces/cities per case</h2><span id='topic+report.summary'></span>

<h3>Description</h3>

<p>function to summarize the current situation, will download the latest data and summarize the top provinces/cities per case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report.summary(
  cases.to.process = "ALL",
  Nentries = 10,
  geo.loc = NULL,
  graphical.output = TRUE,
  saveReport = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.summary_+3A_cases.to.process">cases.to.process</code></td>
<td>
<p>which data to process: &quot;TS&quot; &ndash;time series&ndash;, &quot;AGG&quot; &ndash;aggregated&ndash; or &quot;ALL&quot; &ndash;time series and aggregated&ndash;</p>
</td></tr>
<tr><td><code id="report.summary_+3A_nentries">Nentries</code></td>
<td>
<p>number of top cases to display</p>
</td></tr>
<tr><td><code id="report.summary_+3A_geo.loc">geo.loc</code></td>
<td>
<p>geographical location to process</p>
</td></tr>
<tr><td><code id="report.summary_+3A_graphical.output">graphical.output</code></td>
<td>
<p>flag to deactivate graphical output</p>
</td></tr>
<tr><td><code id="report.summary_+3A_savereport">saveReport</code></td>
<td>
<p>flag to indicate whether the report should be saved in a file</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># triggers CRAN checks for timing
## Not run: 
# displaying top 10s
report.summary()

# get the top 20
report.summary(Nentries=20,graphical.output=FALSE)

# specify a location
report.summary(geo.loc="NorthAmerica")

## End(Not run)

</code></pre>

<hr>
<h2 id='report.Tor'>function to obtain main indicators from Toronto data</h2><span id='topic+report.Tor'></span>

<h3>Description</h3>

<p>function to obtain main indicators from Toronto data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report.Tor(
  colTgts = c("Source.of.Infection", "Age.Group", "Client.Gender", "Outcome",
    "Neighbourhood.Name"),
  report = TRUE,
  staticPlt = TRUE,
  horiz.plts = 4,
  vert.plts = 3,
  same.Yaxis = TRUE,
  interactiveFig = FALSE,
  interactive.display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report.Tor_+3A_coltgts">colTgts</code></td>
<td>
<p>optional argument to indicate which columns from the Toronto data to process</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_report">report</code></td>
<td>
<p>optional argument indicating whether a report will be printed to the screen</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_staticplt">staticPlt</code></td>
<td>
<p>optional argument to indicate whether the 'static' graphical output is wanted or not</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_horiz.plts">horiz.plts</code></td>
<td>
<p>number of plots in the horizontal direction</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_vert.plts">vert.plts</code></td>
<td>
<p>number of plots in the vertical direction</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_same.yaxis">same.Yaxis</code></td>
<td>
<p>graphical argument to indicate if plots will use same y-axis</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_interactivefig">interactiveFig</code></td>
<td>
<p>boolean flag to indicate whether interactice figures would be generated or not</p>
</td></tr>
<tr><td><code id="report.Tor_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean flag to indicate whether the interactive plot will be displayed (pushed) to your browser</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with statistics by selected as indicated in colTgts
</p>
<p>#@export
</p>

<hr>
<h2 id='rollingRate'>function to compute a rolling fn of a TS data</h2><span id='topic+rollingRate'></span>

<h3>Description</h3>

<p>function to compute a rolling fn of a TS data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rollingRate(data, fn = mean, period = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rollingRate_+3A_data">data</code></td>
<td>
<p>TS data</p>
</td></tr>
<tr><td><code id="rollingRate_+3A_fn">fn</code></td>
<td>
<p>function to compute rolling</p>
</td></tr>
<tr><td><code id="rollingRate_+3A_period">period</code></td>
<td>
<p>length of window</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with rolling values
</p>

<hr>
<h2 id='select.per.loc'>auxiliary function to select data based on geographical location</h2><span id='topic+select.per.loc'></span>

<h3>Description</h3>

<p>auxiliary function to select data based on geographical location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select.per.loc(data, geo.loc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select.per.loc_+3A_data">data</code></td>
<td>
<p>data set to process</p>
</td></tr>
<tr><td><code id="select.per.loc_+3A_geo.loc">geo.loc</code></td>
<td>
<p>geographical location, can be a country, region, province or city</p>
</td></tr>
</table>

<hr>
<h2 id='set.plt.canvas'>auxiliary function to set the graphical layout</h2><span id='topic+set.plt.canvas'></span>

<h3>Description</h3>

<p>auxiliary function to set the graphical layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set.plt.canvas(geo.loc, ylayers = 1, minBreaks = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set.plt.canvas_+3A_geo.loc">geo.loc</code></td>
<td>
<p>list of locations so that the fn can determine how many plots would be</p>
</td></tr>
<tr><td><code id="set.plt.canvas_+3A_ylayers">ylayers</code></td>
<td>
<p>parameter to set a multiplier times the nbr of initial plots</p>
</td></tr>
<tr><td><code id="set.plt.canvas_+3A_minbreaks">minBreaks</code></td>
<td>
<p>cut-off to set the minimum nbr of plots</p>
</td></tr>
</table>

<hr>
<h2 id='simple.SIR.ODE'>Define an ODE system for a simple SIR model</h2><span id='topic+simple.SIR.ODE'></span>

<h3>Description</h3>

<p>Define an ODE system for a simple SIR model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simple.SIR.ODE(time, state, parameters)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simple.SIR.ODE_+3A_time">time</code></td>
<td>
<p>time variable</p>
</td></tr>
<tr><td><code id="simple.SIR.ODE_+3A_state">state</code></td>
<td>
<p>state variable</p>
</td></tr>
<tr><td><code id="simple.SIR.ODE_+3A_parameters">parameters</code></td>
<td>
<p>parameters of the ODE</p>
</td></tr>
</table>

<hr>
<h2 id='single.trend'>function to visualize different indicators for trends in daily changes of cases reported as time series data</h2><span id='topic+single.trend'></span>

<h3>Description</h3>

<p>function to visualize different indicators for trends in daily changes of cases reported as time series data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single.trend(ts.data, confBnd = TRUE, info = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single.trend_+3A_ts.data">ts.data</code></td>
<td>
<p>time series data</p>
</td></tr>
<tr><td><code id="single.trend_+3A_confbnd">confBnd</code></td>
<td>
<p>optional argument to remove the drawing of a confidence band</p>
</td></tr>
<tr><td><code id="single.trend_+3A_info">info</code></td>
<td>
<p>addtional information to display in plots</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
tor.data &lt;- covid19.Toronto.data()
single.trend(tor.data[tor.data$status=="Active Cases",]) 


ts.data &lt;- covid19.data("ts-confirmed")
ont.data &lt;- ts.data[ ts.data$Province.State == "Ontario",]
single.trend(ont.data)

single.trend(ts.data[ ts.data$Country.Region=="Italy",])
</code></pre>

<hr>
<h2 id='sweep.SIR.models'>function to perform a sweep of models and generate values of R0</h2><span id='topic+sweep.SIR.models'></span>

<h3>Description</h3>

<p>function to perform a sweep of models and generate values of R0
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sweep.SIR.models(
  data = NULL,
  geo.loc = "Hubei",
  t0_range = 15:20,
  t1 = NULL,
  deltaT = NULL,
  tfinal = 90,
  fatality.rate = 0.02,
  tot.population = 1.4e+09
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sweep.SIR.models_+3A_data">data</code></td>
<td>
<p>time series dataset to consider</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_geo.loc">geo.loc</code></td>
<td>
<p>country/region to analyze</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_t0_range">t0_range</code></td>
<td>
<p>range of initial date for data consideration</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_t1">t1</code></td>
<td>
<p>final period of time for data consideration</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_deltat">deltaT</code></td>
<td>
<p>interval period of time from t0, ie. number of days to consider since t0</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_tfinal">tfinal</code></td>
<td>
<p>total number of days</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_fatality.rate">fatality.rate</code></td>
<td>
<p>rate of causality, deafault value of 2 percent</p>
</td></tr>
<tr><td><code id="sweep.SIR.models_+3A_tot.population">tot.population</code></td>
<td>
<p>total population of the country/region</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># read TimeSeries data
TS.data &lt;- covid19.data("TS-confirmed")
# select a location of interest, eg. France
# France has many entries, just pick "la France"
France.data &lt;- TS.data[ (TS.data$Country.Region == "France") &amp; (TS.data$Province.State == ""),]
# sweep values of R0 based on range of dates to consider for the model
ranges &lt;- 15:20
deltaT &lt;- 20
params_sweep &lt;- sweep.SIR.models(data=France.data,geo.loc="France", t0_range=ranges, deltaT=deltaT)
# obtain the R0 values from the parameters
R0s &lt;- unlist(params_sweep["R0",])
# nbr of infected cases
FR.infs&lt;- preProcessingData(France.data,"France")
# average per range
# define ranges
lst.ranges &lt;- lapply(ranges, function(x) x:(x+deltaT))
# compute averages
avg.FR.infs &lt;- lapply(lst.ranges, function(x) mean(FR.infs[x]))
# plots
plot(R0s, type='b')
# plot vs average number of infected cases
plot(avg.FR.infs, R0s, type='b')

</code></pre>

<hr>
<h2 id='totals.plt'>function to plot total number of cases per day for different groups</h2><span id='topic+totals.plt'></span>

<h3>Description</h3>

<p>function to plot total number of cases per day for different groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>totals.plt(
  data0 = NULL,
  geo.loc0 = NULL,
  one.plt.per.page = FALSE,
  log.plt = TRUE,
  with.totals = FALSE,
  interactive.fig = TRUE,
  fileName = NULL,
  interactive.display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="totals.plt_+3A_data0">data0</code></td>
<td>
<p>time series dataset to process, default all the possible cases: 'confirmed' and 'deaths' for all countries/regions</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_geo.loc0">geo.loc0</code></td>
<td>
<p>geographical location, country/region or province/state to restrict the analysis to</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_one.plt.per.page">one.plt.per.page</code></td>
<td>
<p>boolean flag to have one plot per figure</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_log.plt">log.plt</code></td>
<td>
<p>include a log scale plot in the static plot</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_with.totals">with.totals</code></td>
<td>
<p>a boolean flag to indicate whether the totals should be displayed with the records for the specific location</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_interactive.fig">interactive.fig</code></td>
<td>
<p>switch to turn off/on an interactive plot</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_filename">fileName</code></td>
<td>
<p>file where to save the HTML version of the interactive figure</p>
</td></tr>
<tr><td><code id="totals.plt_+3A_interactive.display">interactive.display</code></td>
<td>
<p>boolean argument for enabling or not displaying the interactive figure</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># retrieve time series data
TS.data &lt;- covid19.data("ts-ALL")

# static and interactive plot 
totals.plt(TS.data)


</code></pre>

<hr>
<h2 id='tots.per.location'>function to compute totals per location</h2><span id='topic+tots.per.location'></span>

<h3>Description</h3>

<p>function to compute totals per location
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tots.per.location(
  data,
  geo.loc = NULL,
  confBnd = FALSE,
  nbr.plts = 1,
  info = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tots.per.location_+3A_data">data</code></td>
<td>
<p>data.frame with *time series* data from covid19</p>
</td></tr>
<tr><td><code id="tots.per.location_+3A_geo.loc">geo.loc</code></td>
<td>
<p>list of locations</p>
</td></tr>
<tr><td><code id="tots.per.location_+3A_confbnd">confBnd</code></td>
<td>
<p>flag to activate/deactivate drawing of confidence bands base on a moving average window</p>
</td></tr>
<tr><td><code id="tots.per.location_+3A_nbr.plts">nbr.plts</code></td>
<td>
<p>parameter to control the number of plots to display per figure</p>
</td></tr>
<tr><td><code id="tots.per.location_+3A_info">info</code></td>
<td>
<p>additional info to display in plots' titles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list or dataframe with totals per specified locations and type of case
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read data for confirmed cases
data &lt;- covid19.data("ts-confirmed")
# compute totals per location for all the countries

tots.per.location(data)

# compute totals per location for 'Italy'
tots.per.location(data,geo.loc="Italy")
# compute totals per location for 'Italy' and 'Germany'
tots.per.location(data,geo.loc=c("Italy","Germany"))

</code></pre>

<hr>
<h2 id='X.covid19.genomic.data'>function to obtain sequencing data grom NCBI
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2</h2><span id='topic+X.covid19.genomic.data'></span>

<h3>Description</h3>

<p>function to obtain sequencing data grom NCBI
Reference:  https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>X.covid19.genomic.data(graphics.ON = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="X.covid19.genomic.data_+3A_graphics.on">graphics.ON</code></td>
<td>
<p>flag to activate/deactivate graphical output</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># obtain covid19's genomic data
covid19.gen.seq &lt;- covid19.genomic.data()
# display the actual RNA seq
covid19.gen.seq$NC_045512.2

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
