<!DOCTYPE html><html lang="en"><head><title>Help for package CovidMutations</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CovidMutations}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AssayMutRatio'><p>Calculate the mutation detection rate using different assays</p></a></li>
<li><a href='#assays'><p>Assays for mutation detection using different primers and probes</p></a></li>
<li><a href='#chinalist'><p>A list of places in China</p></a></li>
<li><a href='#covid_annot'><p>Mutation annotation results produced by &quot;indelSNP&quot; function</p></a></li>
<li><a href='#doubleAssay'><p>Detection of co-occurring mutations using double-assay information</p></a></li>
<li><a href='#gene_position'><p>&quot;GFF3&quot; format gene position data for SARS-Cov-2</p></a></li>
<li><a href='#gff3'><p>&quot;GFF3&quot; format annotation data for SARS-Cov-2</p></a></li>
<li><a href='#globalProteinMut'><p>Global mutational events profiling of proteins</p></a></li>
<li><a href='#globalSNPprofile'><p>Global single nucleotide polymorphism (SNP) profiling in virus genome</p></a></li>
<li><a href='#indelSNP'><p>Provide effects of each single nucleotide polymorphism (SNP), insertion and deletion in virus genome</p></a></li>
<li><a href='#LastfiveNrMutation'><p>Bacth assay analysis for last five Nr of primers</p></a></li>
<li><a href='#mergeEvents'><p>Merge neighboring events of single nucleotide polymorphism (SNP), insertion and deletion.</p></a></li>
<li><a href='#MutByGene'><p>Plot mutation counts for certain genes</p></a></li>
<li><a href='#mutStat'><p>Plot mutation statistics for nucleiotide</p></a></li>
<li><a href='#nucmer'><p>Mutation information derived from &quot;nucmer&quot; SNP analysis</p></a></li>
<li><a href='#nucmerr'><p>Preprocessed &quot;nucmer.snps&quot; file using &quot;nucmerRMD&quot; function</p></a></li>
<li><a href='#nucmerRMD'><p>Preprocess &quot;nucmer&quot; object to add group information</p></a></li>
<li><a href='#plotMutAnno'><p>Plot the mutation statistics after annotating the &quot;nucmer&quot; object by &quot;indelSNP&quot; function</p></a></li>
<li><a href='#plotMutProteins'><p>Plot the most frequent mutational events for proteins in the SARS-CoV-2 genome</p></a></li>
<li><a href='#refseq'><p>SARS-Cov-2 genomic reference sequence from NCBI</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Mutation Analysis Toolkit for COVID-19 (Coronavirus Disease
2019)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-08-29</td>
</tr>
<tr>
<td>Description:</td>
<td>A feasible framework for mutation analysis and reverse transcription 
  polymerase chain reaction (RT-PCR) assay evaluation of COVID-19, including 
  mutation profile visualization, statistics and mutation ratio of each assay. 
  The mutation ratio is conducive to evaluating the coverage of RT-PCR assays in 
  large-sized samples. Mercatelli, D. and Giorgi, F. M. (2020) 
  &lt;<a href="https://doi.org/10.20944%2Fpreprints202004.0529.v1">doi:10.20944/preprints202004.0529.v1</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, cowplot, seqinr, stringr, grDevices, graphics, utils,
ggpubr, dplyr, VennDiagram</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MSQ-123/CovidMutations">https://github.com/MSQ-123/CovidMutations</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MSQ-123/CovidMutations/issues">https://github.com/MSQ-123/CovidMutations/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-11 08:10:05 UTC; lenovo</td>
</tr>
<tr>
<td>Author:</td>
<td>Shaoqian Ma <a href="https://orcid.org/0000-0001-8950-0711"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Yongyou Zhang <a href="https://orcid.org/0000-0003-2413-9106"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shaoqian Ma &lt;shaoqianma@qq.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-18 12:00:39 UTC</td>
</tr>
</table>
<hr>
<h2 id='AssayMutRatio'>Calculate the mutation detection rate using different assays</h2><span id='topic+AssayMutRatio'></span>

<h3>Description</h3>

<p>This function is to use the well established assays information to detect mutations in different
SARS-CoV-2 genomic sites. The output will be series of figures presenting the mutation profile using a
specific assay and a figure for comparison between the mutation detection rate in each primers binding region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AssayMutRatio(
  nucmerr = nucmerr,
  assays = assays,
  totalsample = totalsample,
  plotType = "barplot",
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AssayMutRatio_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="AssayMutRatio_+3A_assays">assays</code></td>
<td>
<p>Assays dataframe including the detection ranges of mutations.</p>
</td></tr>
<tr><td><code id="AssayMutRatio_+3A_totalsample">totalsample</code></td>
<td>
<p>Total sample number, total cleared GISAID fasta data.</p>
</td></tr>
<tr><td><code id="AssayMutRatio_+3A_plottype">plotType</code></td>
<td>
<p>Figure type for either &quot;barplot&quot; or &quot;logtrans&quot;.</p>
</td></tr>
<tr><td><code id="AssayMutRatio_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
data("assays")
Total &lt;- 52 ## Total Cleared GISAID fasta data, sekitseq
#outdir &lt;- tempdir()
#Output the results
AssayMutRatio(nucmerr = nucmerr,
              assays = assays,
              totalsample = Total,
              plotType = "logtrans",
              outdir = NULL)
</code></pre>

<hr>
<h2 id='assays'>Assays for mutation detection using different primers and probes</h2><span id='topic+assays'></span>

<h3>Description</h3>

<p>These assays include the primer detection ranges in which
mutations may occur.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(assays)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 10 rows and 7 columns.
</p>


<h3>References</h3>

<p>Kilic T, Weissleder R, Lee H (2019) iScience 23, 101406.
(<a href="https://pubmed.ncbi.nlm.nih.gov/32771976/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(assays)
</code></pre>

<hr>
<h2 id='chinalist'>A list of places in China</h2><span id='topic+chinalist'></span>

<h3>Description</h3>

<p>The list is used for displacing some original cities' names with &quot;China&quot; in order
to make the downstream analysis easier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chinalist)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 31 rows and 1 column.
</p>


<h3>Source</h3>

<p>This data is created by Zhanglab in Xiamen University.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chinalist)
</code></pre>

<hr>
<h2 id='covid_annot'>Mutation annotation results produced by &quot;indelSNP&quot; function</h2><span id='topic+covid_annot'></span>

<h3>Description</h3>

<p>A dataframe which could be used for downstream analysis like mutation
statistics description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covid_annot)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 394 rows and 10 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.gisaid.org/">https://www.gisaid.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(covid_annot)
</code></pre>

<hr>
<h2 id='doubleAssay'>Detection of co-occurring mutations using double-assay information</h2><span id='topic+doubleAssay'></span>

<h3>Description</h3>

<p>The detection of SARS-CoV-2 is important for the
prevention of the outbreak and management of patients.
Real-time reverse-transcription polymerase chain reaction (RT-PCR)
assay is one of the most effective molecular diagnosis strategies
to detect virus in clinical laboratory. It will be more accurate
and practical to use double assays to detect some samples with
co-occurring mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>doubleAssay(nucmerr = nucmerr, assay1 = assay1, assay2 = assay2, outdir = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="doubleAssay_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="doubleAssay_+3A_assay1">assay1</code></td>
<td>
<p>Information of the first assay(containing primers
locations and probe location, see the format of assays provided as
example data. e.g. data(assays); assay1&lt;- assays[1,])</p>
</td></tr>
<tr><td><code id="doubleAssay_+3A_assay2">assay2</code></td>
<td>
<p>Information of the second assay, the format is the
same as the first assay.</p>
</td></tr>
<tr><td><code id="doubleAssay_+3A_outdir">outdir</code></td>
<td>
<p>The output directory. If NULL print the plot in Rstudio.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot three figures in a single panel, including two results of assays and a
&quot;venn&quot; plot for co-occurring mutated samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
data("assays")
assay1 &lt;- assays[1,]
assay2 &lt;- assays[2,]
#outdir &lt;- tempdir()
doubleAssay(nucmerr = nucmerr,
            assay1 = assay1,
            assay2 = assay2,
            outdir = NULL)
</code></pre>

<hr>
<h2 id='gene_position'>&quot;GFF3&quot; format gene position data for SARS-Cov-2</h2><span id='topic+gene_position'></span>

<h3>Description</h3>

<p>This &quot;GFF3&quot; data is used for counting the mutations in each gene in virus sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_position)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 26 rows and 10 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/">https://www.ncbi.nlm.nih.gov/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gene_position)
</code></pre>

<hr>
<h2 id='gff3'>&quot;GFF3&quot; format annotation data for SARS-Cov-2</h2><span id='topic+gff3'></span>

<h3>Description</h3>

<p>This &quot;GFF3&quot; data is used for annotating the effects of mutations in virus sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gff3)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 26 rows and 10 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=2697049">https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=2697049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gff3)
</code></pre>

<hr>
<h2 id='globalProteinMut'>Global mutational events profiling of proteins</h2><span id='topic+globalProteinMut'></span>

<h3>Description</h3>

<p>This function is to visualize the global protein mutational pattern in the
SARS-CoV-2 genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>globalProteinMut(
  covid_annot = covid_annot,
  outdir = NULL,
  figure_Type = "heatmap",
  top = 10,
  country = "global"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="globalProteinMut_+3A_covid_annot">covid_annot</code></td>
<td>
<p>The mutation effects provided by &quot;indelSNP&quot; function.</p>
</td></tr>
<tr><td><code id="globalProteinMut_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
<tr><td><code id="globalProteinMut_+3A_figure_type">figure_Type</code></td>
<td>
<p>Figure type for either &quot;heatmap&quot; or &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="globalProteinMut_+3A_top">top</code></td>
<td>
<p>The number of variants to plot.</p>
</td></tr>
<tr><td><code id="globalProteinMut_+3A_country">country</code></td>
<td>
<p>Choose a country to plot the mutational pattern or
choose &quot;global&quot; to profile mutations across all countries. The
default is &quot;global&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("covid_annot")
outdir &lt;- tempdir()
# make sure the covid_annot is a dataframe
covid_annot &lt;- as.data.frame(covid_annot)
globalProteinMut(covid_annot = covid_annot,
                 outdir = outdir,
                 figure_Type = "heatmap",
                 top = 10,
                 country = "USA")
</code></pre>

<hr>
<h2 id='globalSNPprofile'>Global single nucleotide polymorphism (SNP) profiling in virus genome</h2><span id='topic+globalSNPprofile'></span>

<h3>Description</h3>

<p>This function is to visualize the global SNP pattern in the
SARS-CoV-2 genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>globalSNPprofile(
  nucmerr = nucmerr,
  outdir = NULL,
  figure_Type = "heatmap",
  country = "global",
  top = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="globalSNPprofile_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="globalSNPprofile_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
<tr><td><code id="globalSNPprofile_+3A_figure_type">figure_Type</code></td>
<td>
<p>Figure type for either &quot;heatmap&quot; or &quot;count&quot;.</p>
</td></tr>
<tr><td><code id="globalSNPprofile_+3A_country">country</code></td>
<td>
<p>Choose a country to plot the mutational pattern or
choose &quot;global&quot; to profile mutations across all countries. The
default is &quot;global&quot;.</p>
</td></tr>
<tr><td><code id="globalSNPprofile_+3A_top">top</code></td>
<td>
<p>The number of mutational classes to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
outdir &lt;- tempdir()
globalSNPprofile(nucmerr = nucmerr,
                 outdir = outdir,
                 figure_Type = "heatmap",
                 country = "global",
                 top = 5)
</code></pre>

<hr>
<h2 id='indelSNP'>Provide effects of each single nucleotide polymorphism (SNP), insertion and deletion in virus genome</h2><span id='topic+indelSNP'></span>

<h3>Description</h3>

<p>This function is to annotate the mutational events and indicate their potential effects on the proteins.
Mutational events include SNP, insertion and deletion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indelSNP(
  nucmer = nucmer,
  saveRda = FALSE,
  refseq = refseq,
  gff3 = gff3,
  annot = annot,
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indelSNP_+3A_nucmer">nucmer</code></td>
<td>
<p>An object called &quot;nucmer&quot;, mutation information derived
from &quot;nucmer.snp&quot; variant file by &quot;seqkit&quot; software and &quot;nucmer
SNP-calling&quot; scripts. To be processed by &quot;indelSNP&quot; function, The nucmer
object should be first transformed by &quot;mergeEvents&quot; function.</p>
</td></tr>
<tr><td><code id="indelSNP_+3A_saverda">saveRda</code></td>
<td>
<p>Whether to save the results as &quot;.rda&quot; file.</p>
</td></tr>
<tr><td><code id="indelSNP_+3A_refseq">refseq</code></td>
<td>
<p>SARS-Cov-2 genomic reference sequence.</p>
</td></tr>
<tr><td><code id="indelSNP_+3A_gff3">gff3</code></td>
<td>
<p>&quot;GFF3&quot; format annotation data for SARS-Cov-2.</p>
</td></tr>
<tr><td><code id="indelSNP_+3A_annot">annot</code></td>
<td>
<p>Annotation of genes(corresponding proteins) list from &quot;GFF3&quot; file by &quot;setNames(gff3[,10],gff3[,9])&quot;.</p>
</td></tr>
<tr><td><code id="indelSNP_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Write the result as &quot;.csv&quot; file to the specified directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmer")
# Fix IUPAC codes
nucmer&lt;-nucmer[!nucmer$qvar%in%c("B","D","H","K","M","N","R","S","V","W","Y"),]
nucmer&lt;- mergeEvents(nucmer = nucmer)## This will update the nucmer object
data("refseq")
data("gff3")
annot &lt;- setNames(gff3[,10],gff3[,9])
#outdir &lt;- tempdir()
nucmer&lt;- indelSNP(nucmer = nucmer,
                  saveRda = FALSE,
                  refseq = refseq,
                  gff3 = gff3,
                  annot = annot,
                  outdir = NULL)
</code></pre>

<hr>
<h2 id='LastfiveNrMutation'>Bacth assay analysis for last five Nr of primers</h2><span id='topic+LastfiveNrMutation'></span>

<h3>Description</h3>

<p>Last five nucleotides of primer mutation count/type for any reverse transcription
polymerase chain reaction (RT-PCR) primer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LastfiveNrMutation(
  nucmerr = nucmerr,
  assays = assays,
  totalsample = totalsample,
  figurelist = FALSE,
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LastfiveNrMutation_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="LastfiveNrMutation_+3A_assays">assays</code></td>
<td>
<p>Assays dataframe including the detection ranges of mutations.</p>
</td></tr>
<tr><td><code id="LastfiveNrMutation_+3A_totalsample">totalsample</code></td>
<td>
<p>Total sample number, total cleared GISAID fasta data.</p>
</td></tr>
<tr><td><code id="LastfiveNrMutation_+3A_figurelist">figurelist</code></td>
<td>
<p>Whether to output the integrated plot list for each assay.</p>
</td></tr>
<tr><td><code id="LastfiveNrMutation_+3A_outdir">outdir</code></td>
<td>
<p>The output directory. if the figurelist = TRUE, output the figure in the R session.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the mutation counts(last five nucleotides for each primer) for each assay as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
data("assays")
totalsample &lt;- 434
#outdir &lt;- tempdir()
LastfiveNrMutation(nucmerr = nucmerr,
                   assays = assays,
                   totalsample = totalsample,
                   figurelist = FALSE,
                   outdir = NULL)
</code></pre>

<hr>
<h2 id='mergeEvents'>Merge neighboring events of single nucleotide polymorphism (SNP), insertion and deletion.</h2><span id='topic+mergeEvents'></span>

<h3>Description</h3>

<p>The first step for handling the nucmer object, then effects of mutations
can be analysed using &quot;indelSNP&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeEvents(nucmer = nucmer)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeEvents_+3A_nucmer">nucmer</code></td>
<td>
<p>An object called &quot;nucmer&quot;, mutation information derived from
&quot;nucmer.snp&quot; variant file by &quot;seqkit&quot; software and &quot;nucmer SNP-calling&quot;
scripts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated &quot;nucmer&quot; object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The example data:
data("nucmer")
#options(stringsAsFactors = FALSE)

#The input nucmer object can be made by the comment below:
#nucmer&lt;-read.delim("nucmer.snps",as.is=TRUE,skip=4,header=FALSE)
#colnames(nucmer)&lt;-c("rpos","rvar","qvar","qpos","","","","",
#"rlength","qlength","","","rname","qname")
#rownames(nucmer)&lt;-paste0("var",1:nrow(nucmer))

# Fix IUPAC codes
nucmer&lt;-nucmer[!nucmer$qvar%in%c("B","D","H","K","M","N","R","S","V","W","Y"),]
nucmer&lt;- mergeEvents(nucmer = nucmer)## This will update the nucmer object
</code></pre>

<hr>
<h2 id='MutByGene'>Plot mutation counts for certain genes</h2><span id='topic+MutByGene'></span>

<h3>Description</h3>

<p>After annotating the mutations, this function is to plot the counts of mutational events for
each gene in the SARS-CoV-2 genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MutByGene(nucmerr = nucmerr, gff3 = gff3, figurelist = FALSE, outdir = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MutByGene_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="MutByGene_+3A_gff3">gff3</code></td>
<td>
<p>&quot;GFF3&quot; format gene position data for SARS-Cov-2(the &quot;GFF3&quot; file should include columns named: &quot;Gene&quot;, &quot;Start&quot;, &quot;Stop&quot;).</p>
</td></tr>
<tr><td><code id="MutByGene_+3A_figurelist">figurelist</code></td>
<td>
<p>Whether to output the integrated plot list for each gene.</p>
</td></tr>
<tr><td><code id="MutByGene_+3A_outdir">outdir</code></td>
<td>
<p>The output directory, if the figurelist = TRUE, output the figure in the R session.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the mutation counts figure for each gene as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
data("gene_position")
#outdir &lt;- tempdir()
MutByGene(nucmerr = nucmerr, gff3 = gene_position, figurelist = FALSE, outdir = NULL)
#if figurelist = TRUE, the recommendation for figure display(in pixel)is: width=1650, height=1300
</code></pre>

<hr>
<h2 id='mutStat'>Plot mutation statistics for nucleiotide</h2><span id='topic+mutStat'></span>

<h3>Description</h3>

<p>Visualization for the top mutated samples, average mutational counts,
top mutated position in the genome, mutational density across the genome
and distribution of mutations across countries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutStat(
  nucmerr = nucmerr,
  outdir = NULL,
  figure_Type = "TopMuSample",
  type_top = 10,
  country = FALSE,
  mutpos = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mutStat_+3A_nucmerr">nucmerr</code></td>
<td>
<p>Mutation information containing group list(derived from &quot;nucmer&quot; object using &quot;nucmerRMD&quot; function).</p>
</td></tr>
<tr><td><code id="mutStat_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
<tr><td><code id="mutStat_+3A_figure_type">figure_Type</code></td>
<td>
<p>Figure type for: &quot;TopMuSample&quot;, &quot;AverageMu&quot;, &quot;TopMuPos&quot;, &quot;MutDens&quot;, &quot;CountryMutCount&quot;, &quot;TopCountryMut&quot;.</p>
</td></tr>
<tr><td><code id="mutStat_+3A_type_top">type_top</code></td>
<td>
<p>To plot the figure involving &quot;top n&quot;(&quot;TopMuSample&quot;, &quot;TopMuPos&quot;, &quot;TopCountryMut&quot;), the &quot;type_top&quot; should specify the number of objects to display.</p>
</td></tr>
<tr><td><code id="mutStat_+3A_country">country</code></td>
<td>
<p>To plot the figure using country as groups(&quot;CountryMutCount&quot; and &quot;TopCountryMut&quot;), the &quot;country&quot; should be TRUE.</p>
</td></tr>
<tr><td><code id="mutStat_+3A_mutpos">mutpos</code></td>
<td>
<p>If the figure type is &quot;TopCountryMut&quot;, &quot;mutpos&quot; can specify
A range of genomic position(eg. 28831:28931) for plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmerr")
outdir &lt;- tempdir()
mutStat(nucmerr = nucmerr,
        outdir = outdir,
        figure_Type = "TopCountryMut",
        type_top = 10,
        country = FALSE,
        mutpos = NULL)
</code></pre>

<hr>
<h2 id='nucmer'>Mutation information derived from &quot;nucmer&quot; SNP analysis</h2><span id='topic+nucmer'></span>

<h3>Description</h3>

<p>The &quot;nucmer.snps&quot; variant file is obtained by processing the SARS-Cov-2 sequence
from Gisaid website (complete, high coverage only, low coverage exclusion, Host=human,
Virus name = hCoV-19) with &quot;seqkit&quot; software and &quot;nucmer&quot; scripts. The example data
is downsampled from complete data in 2020-07-28 (0.001 proportion, 52 samples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nucmer)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 437 rows (mutation sites) and 14 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.gisaid.org/">https://www.gisaid.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nucmer)
</code></pre>

<hr>
<h2 id='nucmerr'>Preprocessed &quot;nucmer.snps&quot; file using &quot;nucmerRMD&quot; function</h2><span id='topic+nucmerr'></span>

<h3>Description</h3>

<p>A dataset contains some group information subtracted from the &quot;nucmer&quot;
object by &quot;nucmerRMD&quot; function in order to best describe the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nucmerr)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 437 rows (downsampled mutation sites) and 10
columns.
</p>


<h3>Source</h3>

<p><a href="https://www.gisaid.org/">https://www.gisaid.org/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nucmerr)
</code></pre>

<hr>
<h2 id='nucmerRMD'>Preprocess &quot;nucmer&quot; object to add group information</h2><span id='topic+nucmerRMD'></span>

<h3>Description</h3>

<p>Manipulate the &quot;nucmer&quot; object to make the analysis easier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nucmerRMD(nucmer = nucmer, outdir = NULL, chinalist = chinalist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nucmerRMD_+3A_nucmer">nucmer</code></td>
<td>
<p>An object called &quot;nucmer&quot;, mutation information derived from
&quot;nucmer.snp&quot; variant file by &quot;seqkit&quot; software and &quot;nucmer SNP-calling&quot;
scripts.</p>
</td></tr>
<tr><td><code id="nucmerRMD_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
<tr><td><code id="nucmerRMD_+3A_chinalist">chinalist</code></td>
<td>
<p>A list of places in China, for displacing some original cities with &quot;China&quot; in order
to make the downstream analysis easier.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saving the updated &quot;nucmer&quot; object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("nucmer")
data("chinalist")
#outdir &lt;- tempdir()
nucmerr&lt;- nucmerRMD(nucmer = nucmer, outdir = NULL, chinalist = chinalist)
</code></pre>

<hr>
<h2 id='plotMutAnno'>Plot the mutation statistics after annotating the &quot;nucmer&quot; object by &quot;indelSNP&quot; function</h2><span id='topic+plotMutAnno'></span>

<h3>Description</h3>

<p>Basic descriptions for the mutational events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMutAnno(covid_annot = covid_annot, figureType = "MostMut", outdir = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMutAnno_+3A_covid_annot">covid_annot</code></td>
<td>
<p>The mutation effects provided by &quot;indelSNP&quot; function.</p>
</td></tr>
<tr><td><code id="plotMutAnno_+3A_figuretype">figureType</code></td>
<td>
<p>Figure type for: &quot;MostMut&quot;, &quot;MutPerSample&quot;, &quot;VarClasses&quot;, &quot;VarType&quot;, &quot;NucleoEvents&quot;, &quot;ProEvents&quot;.</p>
</td></tr>
<tr><td><code id="plotMutAnno_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the selected figure type as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("covid_annot")
# make sure the covid_annot is a dataframe
covid_annot &lt;- as.data.frame(covid_annot)
#outdir &lt;- tempdir() specify your output directory
plotMutAnno(covid_annot = covid_annot, figureType = "MostMut", outdir = NULL)
</code></pre>

<hr>
<h2 id='plotMutProteins'>Plot the most frequent mutational events for proteins in the SARS-CoV-2 genome</h2><span id='topic+plotMutProteins'></span>

<h3>Description</h3>

<p>Plot the most frequent mutational events for proteins selected.
The protein name should be specified correctly (only for SARS-CoV-2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMutProteins(
  covid_annot = covid_annot,
  proteinName = "NSP2",
  top = 20,
  outdir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMutProteins_+3A_covid_annot">covid_annot</code></td>
<td>
<p>The mutation effects provided by &quot;indelSNP&quot; function.</p>
</td></tr>
<tr><td><code id="plotMutProteins_+3A_proteinname">proteinName</code></td>
<td>
<p>Proteins in the SARS-CoV-2 genome, available choices:
5'UTR, NSP1~NSP10, NSP12a, NSP12b, NSP13, NSP14, NSP15, NSP16, S, ORF3a,
E, M, ORF6, ORF7a, ORF7b, ORF8, N, ORF10.</p>
</td></tr>
<tr><td><code id="plotMutProteins_+3A_top">top</code></td>
<td>
<p>The number of objects to display.</p>
</td></tr>
<tr><td><code id="plotMutProteins_+3A_outdir">outdir</code></td>
<td>
<p>The output directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot the mutational events for selected proteins as output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("covid_annot")
# make sure the covid_annot is a dataframe
covid_annot &lt;- as.data.frame(covid_annot)
#outdir &lt;- tempdir() specify your output directory
plotMutProteins(covid_annot = covid_annot,proteinName = "NSP2", top = 20, outdir = NULL)
</code></pre>

<hr>
<h2 id='refseq'>SARS-Cov-2 genomic reference sequence from NCBI</h2><span id='topic+refseq'></span>

<h3>Description</h3>

<p>This reference sequence is derived from &quot;fasta&quot; file, preprocessed by
&quot;read.fasta&quot; function(refseq&lt;-read.fasta(&quot;NC_045512.2.fa&quot;,forceDNAtolower=FALSE)[[1]]). It is used for annotating mutations in virus samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(refseq)
</code></pre>


<h3>Format</h3>

<p>&quot;SeqFastadna&quot; characters.
</p>


<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/32015508/">https://pubmed.ncbi.nlm.nih.gov/32015508/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(refseq)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
