<!DOCTYPE html><html><head><title>Help for package cp4p</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cp4p}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cp4p-package'><p>Introduction to the CP4P package</p></a></li>
<li><a href='#adjust.p'><p>Adjusted p-values for adaptive FDR control</p></a></li>
<li><a href='#calibration.plot'>
<p>Displaying the &quot;Calibration Plot&quot; of a vector of p-values.</p></a></li>
<li><a href='#estim.pi0'>
<p>Estimation of the proportion of true null hypotheses</p></a></li>
<li><a href='#LFQRatio2'><p>Dataset &quot;LFQRatio2&quot;</p></a></li>
<li><a href='#LFQRatio25'><p>Dataset &quot;LFQRatio25&quot;</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calibration Plot for Proteomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-02-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Quentin Giai Gianetto, Florence Combes, Claire Ramus, Christophe Bruley, Yohann Couté, Thomas Burger</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to check whether a vector of p-values respects the assumptions of FDR (false discovery rate) control procedures and to compute adjusted p-values.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0), MESS, graphics, stats, multtest, qvalue, limma</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-22 17:17:44 UTC; Quentin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-24 15:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='cp4p-package'>Introduction to the CP4P package</h2><span id='topic+cp4p-package'></span><span id='topic+cp4p'></span>

<h3>Description</h3>

<p>This package provides tools to check whether a vector of p-values respects the assumptions of classical FDR (false discovery rate) control procedures. 
</p>
<p>It is built to be easily used by non-statisticians in the context of quantitative proteomics (yet, it can be applied in other contexts). 
</p>
<p>Concretely, it allows estimating the proportion of true null hypotheses (i.e. proportion of non-differentially abundant proteins or peptides in a relative quantification experiment), 
as well as checking whether the p-values are adequately distributed for further FDR  control. 
</p>
<p>In addition, the package allows performing an adequately chosen adaptive FDR control procedure to get adjusted p-values.  
</p>
<p>A tutorial giving a practical introduction to this package is available in the supplementary material of Giai Gianetto et al. (2016).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> cp4p</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends:</td><td style="text-align: left;"> multtest, qvalue, limma, MESS, graphics, stats</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package is composed of three functions that take as input a vector of p-values resulting from multiple two-sided hypothesis testing (such as multiple t-tests for equal means for instance). 
</p>
<p>First, the function <code><a href="#topic+estim.pi0">estim.pi0</a></code> allows determining the proportion of true null hypotheses among the set of tests using eight different estimation methods proposed in literature. 
</p>
<p>Second, the function <code><a href="#topic+calibration.plot">calibration.plot</a></code> proposes an intuitive plot of the p-values, so as to visually assess their behavior and well-calibration. 
</p>
<p>Third, the function <code><a href="#topic+adjust.p">adjust.p</a></code> allows obtaining adjusted p-values in view to perform an adaptive FDR control from a chosen level.
</p>
<p>Two proteomic datasets named <code><a href="#topic+LFQRatio2">LFQRatio2</a></code> and <code><a href="#topic+LFQRatio25">LFQRatio25</a></code> allow to use these functions in a concrete framework where the proportion of non-differentially abundant proteins is known.
</p>


<h3>Author(s)</h3>

<p>Quentin Giai Gianetto, Florence Combes, Claire Ramus, Christophe Bruley, Yohann Couté, Thomas Burger
</p>
<p>Maintainer: Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;
</p>


<h3>References</h3>

<p>Giai Gianetto, Q., Combes, F., Ramus, C., Bruley, C., Couté, Y., Burger, T. (2016). Calibration plot for proteomics: A graphical tool to visually check the assumptions underlying FDR control in quantitative experiments. Proteomics, 16(1), 29-32.</p>

<hr>
<h2 id='adjust.p'>Adjusted p-values for adaptive FDR control</h2><span id='topic+adjust.p'></span>

<h3>Description</h3>

<p>This function computes adjusted p-values for adaptive FDR control from a vector of raw (unadjusted) p-values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust.p(p, pi0.method = 1, alpha = 0.05, nbins = 20, pz = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjust.p_+3A_p">p</code></td>
<td>
<p>Numeric vector of raw p-values. Raw p-values are assumed without missing values, and between 0 and 1.</p>
</td></tr>
<tr><td><code id="adjust.p_+3A_pi0.method">pi0.method</code></td>
<td>
<p>Numeric value between 0 and 1 corresponding to the proportion of true null hypotheses (non-differentially abundant proteins or peptides), or the name of an estimation method for this proportion among <code>"st.boot"</code>, <code>"st.spline"</code>, <code>"langaas"</code>, <code>"jiang"</code>, <code>"histo"</code>, <code>"pounds"</code>, <code>"abh"</code> or <code>"slim"</code> (see function <code><a href="#topic+estim.pi0">estim.pi0</a></code> for details). The two-stage Benjamini and Hochberg procedure (Benjamini et al. (2006)) is also available according to an expected FDR given by the <code>alpha</code> parameter (write <code>pi0.method="bky"</code>). Default is 1 (classical Benjamini and Hochberg procedure (1995) is performed in this case).</p>
</td></tr>
<tr><td><code id="adjust.p_+3A_alpha">alpha</code></td>
<td>
<p>A nominal type I error rate used for estimating the proportion of true null hypotheses (non-differentially abundant proteins or peptides) in the two-stage Benjamini and Hochberg procedure (used only if <code>pi0.method="bky"</code>). Default is 0.05.</p>
</td></tr>
<tr><td><code id="adjust.p_+3A_nbins">nbins</code></td>
<td>
<p>Number of bins. Parameter used for the <code>"jiang"</code> and  <code>"histo"</code> methods in <code><a href="#topic+estim.pi0">estim.pi0</a></code>. Default is 20.</p>
</td></tr>
<tr><td><code id="adjust.p_+3A_pz">pz</code></td>
<td>
<p>P-value threshold such as p-values below are associated to false null hypotheses. Used for the <code>"slim"</code> method in <code><a href="#topic+estim.pi0">estim.pi0</a></code>. Default is 0.05.</p>
</td></tr>

</table>


<h3>Details</h3>

<p>The procedure uses an estimation of the proportion of true null hypotheses (non-differentially abundant proteins or peptides), the value or the name of which is precised in input. Next, this estimation is multiplied by the adjusted p-values of the Benjamini and Hochberg procedure (1995) to obtain the final adjusted p-values (see section 3 in Craiu and Sun (2008) for details).
</p>
<p>The adjusted p-values of the Benjamini and Hochberg procedure (1995) and of the two-stage Benjamini and Hochberg procedure (Benjamini et al. (2006)) are computed using the R package <code>multtest</code> (Pollard et al. (2005)).
</p>


<h3>Value</h3>

<p>A list composed of :
</p>
<table>
<tr><td><code>pi0</code></td>
<td>
<p>The proportion of true null hypotheses (non-differentially abundant proteins or peptides) used to adjust p-values.</p>
</td></tr>
<tr><td><code>adjp</code></td>
<td>
<p>A matrix of raw and adjusted p-values with rows corresponding to each test. First column corresponds to raw p-values and second column to adjusted p-values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;
</p>


<h3>References</h3>

<p>Y. Benjamini and Y. Hochberg. Controlling the false discovery rate: a practical and powerful approach to multiple testing. Journal of the Royal Statistical Society. Series B (Methodological), 289-300, 1995.
</p>
<p>Y. Benjamini, A.M. Krieger, and D.Yekutieli. Adaptive linear step-up procedures that control the false discovery rate. Biometrika, 93(3):491-507, 2006.
</p>
<p>R.V. Craiu and L. Sun. Choosing the lesser evil: trade-off between false discovery rate and non-discovery rate. Statistica Sinica, 18:861-879, 2008.
</p>
<p>K.S. Pollard, S. Dudoit and M.J. van der Laan. Multiple Testing Procedures: R multtest Package and Applications to Genomics, in Bioinformatics and Computational Biology Solutions Using R and Bioconductor. Springer. 2005.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estim.pi0">estim.pi0</a></code>, <code><a href="#topic+calibration.plot">calibration.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get p-values
data(LFQRatio2)
p=LFQRatio2[,7]

#adjust p-values by estimating the proportion of true null hypotheses 
#with the "pounds" method.
res_pounds=adjust.p(p, pi0.method = "pounds")

#proportion of true null hypotheses with the "pounds" method.
res_pounds$pi0

#plot ajusted p-values in function of raw p-values
plot(res_pounds$adjp)

#adjust p-values by estimating the proportion of true null hypotheses
#using the two-stage Benjamini and Hochberg procedure with a FDR of 0.1.
res_bky=adjust.p(p, pi0.method = "bky", alpha = 0.1)

#proportion of true null hypotheses with the two-stage BH procedure.
res_bky$pi0

#plot adjusted p-values in function of raw p-values
plot(res_bky$adjp)

#compare the two-stage Benjamini and Hochberg procedure 
#with the "pounds" method
plot(res_pounds$adjp[,2],res_bky$adjp[,2])

</code></pre>

<hr>
<h2 id='calibration.plot'>
Displaying the &quot;Calibration Plot&quot; of a vector of p-values.
</h2><span id='topic+calibration.plot'></span>

<h3>Description</h3>

<p>From a proteomics viewpoint, this function displays a graph (the &quot;Calibration Plot&quot;) which allows to visually assess the compliance of a differential abundance analysis with FDR control procedure assumptions.
</p>
<p>From a statistical viewpoint, this function performs a plot of the cumulative distribution function of 1-p-values. It allows checking whether p-values respect several assumptions of FDR control procedures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibration.plot(p, pi0.method = "pounds", nbins = 20, pz = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calibration.plot_+3A_p">p</code></td>
<td>
<p>Numeric vector of raw p-values. Raw p-values are assumed without missing values, and between 0 and 1.</p>
</td></tr>
<tr><td><code id="calibration.plot_+3A_pi0.method">pi0.method</code></td>
<td>

<p>Numeric value between 0 and 1 corresponding to the proportion of true null hypotheses if known by the user, or the name of an estimation method among <code>"st.boot"</code>, <code>"st.spline"</code>, <code>"langaas"</code>, <code>"jiang"</code>, <code>"histo"</code>, <code>"pounds"</code>, <code>"abh"</code> or <code>"slim"</code> (see function <code><a href="#topic+estim.pi0">estim.pi0</a></code> for details). Default is <code>"pounds"</code>. If <code>pi0.method="ALL"</code>, a plot allowing the comparison of the eight estimation methods is displayed.
</p>
</td></tr>
<tr><td><code id="calibration.plot_+3A_nbins">nbins</code></td>
<td>
<p>Number of bins. Parameter used for the <code>"jiang"</code> and  <code>"histo"</code> methods in <code><a href="#topic+estim.pi0">estim.pi0</a></code>. Default is 20.</p>
</td></tr>
<tr><td><code id="calibration.plot_+3A_pz">pz</code></td>
<td>
<p>P-value threshold such as p-values below are associated to false null hypotheses. Used for the <code>"slim"</code> method in <code><a href="#topic+estim.pi0">estim.pi0</a></code>. Default is 0.05.</p>
</td></tr>

</table>


<h3>Details</h3>

<p>This function provides a graph which displays the cumulative distribution function of 1-p-values as a function of 1-p-values (black curve) as advocated by Schweder and Spjotvoll (1982). 
</p>
<p>The blue straight line has a slope equals to the proportion of true null hypotheses (estimated by <code><a href="#topic+estim.pi0">estim.pi0</a></code>) that is recalled in the caption of the plot. It is close to the black curve for small 1-pvalues if the p-values are independently and uniformly distributed under the null hypothesis.
</p>
<p>In addition, two other measures are given in the caption of the graphic. Each has a color that matches that of various areas of the plot and should be carefully consider to assess the well-calibration of p-values (see Giai Gianetto et al. (2016) for details). 
</p>
<p>The first measure corresponds to one minus the ratio between the green area and the grey area (referred to as &quot;differentially abundant protein concentration&quot;). The closer to 100% this measure is, the smaller the false nondiscovery rate is expected. 
</p>
<p>The second measure corresponds to the total red area observed on the graph (referred to as &quot;uniformity underestimation&quot;). The smaller this measure is, the more the proportion of true null hypotheses is expected to be not under-estimated (so as to get a conservative p-value adjustment).
</p>
<p>Supplementary theoretical justifications on these measures can be found in the tutorial available in the supplementary material of Giai Gianetto et al. (2016).
</p>


<h3>Value</h3>

<p>A list composed of :
</p>
<table>
<tr><td><code>pi0</code></td>
<td>
<p>Numeric value corresponding to the proportion of true null hypotheses (non-differentially abundant proteins or peptides) used for the plot. Numeric vector if <code>pi0.method="ALL"</code>.</p>
</td></tr>
<tr><td><code>h1.concentration</code></td>
<td>
<p>Numeric value corresponding to one minus the ratio between the green area and the grey area. NULL if <code>pi0.method="ALL"</code>.</p>
</td></tr>
<tr><td><code>unif.under</code></td>
<td>
<p>Numeric value corresponding to the total red area observed on the graph (multiplied by 100). NULL if <code>pi0.method="ALL"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;
</p>


<h3>References</h3>

<p>Giai Gianetto, Q., Combes, F., Ramus, C., Bruley, C., Couté, Y., Burger, T. (2016). Calibration plot for proteomics: A graphical tool to visually check the assumptions underlying FDR control in quantitative experiments. Proteomics, 16(1), 29-32.
</p>
<p>Schweder, T., Spjotvoll, E. (1982). Plots of p-values to evaluate many tests simultaneously. Biometrika, 69(3), 493-502.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estim.pi0">estim.pi0</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get p-values
data(LFQRatio25)
p=LFQRatio25[,7]

#Plot straight lines whose slopes correspond to different estimates of 
#the proportion of true null hypotheses
r=calibration.plot(p, pi0.method="ALL")
r$pi0
 
#Plot of the graph with the "pounds" method (default)
r=calibration.plot(p)
#Estimate of the proportion of true null hypotheses
r$pi0
#Estimate of the differentially abundant protein concentration 
#(the closer to one, the better)
r$h1.concentration
#Estimate of the "uniformity underestimation" quantity 
#(If null, pi0 is not underestimated.)
r$unif.under

#Plot of the graph using the "slim" method
r=calibration.plot(p, pi0.method="slim")

</code></pre>

<hr>
<h2 id='estim.pi0'>
Estimation of the proportion of true null hypotheses
</h2><span id='topic+estim.pi0'></span>

<h3>Description</h3>

<p>From a proteomics viewpoint, this function estimates the global proportion of proteins (resp. of peptides) that are non differentially abundant from the tested protein list (resp. from the tested peptide list).
This proportion is later used as a correcting factor to compute the adjusted p-values, that are in turn used to tune a threshold according to a desired false discovery rate.
</p>
<p>From a statistical viewpoint, this function allows estimating the proportion of true null hypotheses (pi0) from a vector of raw p-values following eight different estimation methods from the literature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estim.pi0(p, pi0.method = "ALL", nbins = 20, pz = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estim.pi0_+3A_p">p</code></td>
<td>
<p>Numeric vector of raw p-values. Raw p-values are assumed without missing values, and between 0 and 1.</p>
</td></tr>
<tr><td><code id="estim.pi0_+3A_pi0.method">pi0.method</code></td>
<td>
<p>Name of an estimation method for the proportion of true null hypotheses among <code>"st.boot"</code>, <code>"st.spline"</code>, <code>"langaas"</code>, <code>"jiang"</code>, <code>"histo"</code>, <code>"pounds"</code>, <code>"abh"</code> or <code>"slim"</code>. Default is <code>"ALL"</code>: all the eight estimation methods are performed simultaneously.</p>
</td></tr>
<tr><td><code id="estim.pi0_+3A_nbins">nbins</code></td>
<td>
<p>Number of bins. Parameter used for the <code>"jiang"</code> and  <code>"histo"</code> methods. Default is 20.</p>
</td></tr>
<tr><td><code id="estim.pi0_+3A_pz">pz</code></td>
<td>
<p>P-value threshold such as p-values below are associated to false null hypotheses. Used for the <code>"slim"</code> method. Wang, Tuominen and Tsai (2011) suggest to take a value between 0.01 and 0.1. Default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows to estimate the proportion of true null hypotheses following different estimation methods :
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>"abh"</code></td><td style="text-align: left;"> the least slope method proposed in Benjamini and Hochberg (2000).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"st.spline"</code></td><td style="text-align: left;"> the smoother method described in Storey and Tibshirani (2003). </td>
</tr>
<tr>
 <td style="text-align: left;"> 
 </td><td style="text-align: left;"> The <code>qvalue</code> function of R package <code>qvalue</code> with default tuning is used (Storey (2015)).</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"st.boot"</code></td><td style="text-align: left;"> the bootstrap method described in Storey et al. (2004).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> The <code>qvalue</code> function of R package <code>qvalue</code> with default tuning is used (Storey (2015)).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"langaas"</code></td><td style="text-align: left;"> the method described in Langaas, Ferkingstad and Lindqvist (2005) using a convex</td>
</tr>
<tr>
 <td style="text-align: left;">
  </td><td style="text-align: left;"> decreasing density estimate for p-values. The <code>convest</code> function of R package <code>limma</code></td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;">  with default tuning is used (Ritchie et al. (2015)).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"histo"</code></td><td style="text-align: left;"> the histogram method described in Nettleton, Hwang, Caldo and Wise (2006).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"pounds"</code></td><td style="text-align: left;"> the conservative estimate described in Pounds and Cheng (2006).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"jiang"</code></td><td style="text-align: left;"> the average estimate method described in Jiang and Doerge (2008).</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"slim"</code></td><td style="text-align: left;"> the method of Wang, Tuominen and Tsai (2011) using a sliding linear model.</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> The default tuning suggested by Wang, Tuominen and Tsai (2011) is used.</td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> Using their notations, lambda1 is fixed to 0.1, n to 10 and B to 100.
</td>
</tr>

</table>

<p>To take into account of right censorship on the vector of p-values, each p-value is divided by the maximum p-value present in <code>p</code>. Accordingly, the p-values of the true null hypotheses are assumed uniformly distributed between 0 and this maximum. This kind of censorship happens in proteomics when a first thresholding is performed on the fold-changes. 
</p>
<p>If you want to assume that the p-values are uniformly distributed between 0 and 1, replace <code>p</code> by <code>c(p,1)</code> when using <code>estim.pi0</code>. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>pi0</code></td>
<td>
<p>Numeric value of the estimated proportion of true null hypotheses from the selected method; Numeric vector if <code>pi0.method="ALL"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Quentin Giai Gianetto &lt;quentin2g@yahoo.fr&gt;
</p>


<h3>References</h3>

<p>Y. Benjamini and Y. Hochberg. On the adaptive control of the false discovery rate in multiple testing with independent statistics. Journal of Educational and Behavioral Statistics, 25(1):60-83, 2000.
</p>
<p>H. Jiang and R.W. Doerge. Estimating the proportion of true null hypotheses for multiple comparisons. Cancer informatics, 6:25, 2008.
</p>
<p>M. Langaas, B.H. Lindqvist, and E. Ferkingstad. Estimating the proportion of true null hypotheses, with application to dna microarray data. Journal of the Royal Statistical Society: Series B (Statistical Methodology), 67(4):555-572, 2005.
</p>
<p>D. Nettleton, J.T.G. Hwang, R.A. Caldo, and R.P. Wise. Estimating the number of true null hypotheses from a histogram of p values. Journal of Agricultural, Biological, and Environmental Statistics, 11(3):337-356, 2006.
</p>
<p>S. Pounds and C. Cheng. Robust estimation of the false discovery rate. Bioinformatics, 22(16):1979-1987, 2006.
</p>
<p>M.E. Ritchie, B. Phipson, D. Wu, Y. Hu, C.W. Law, W. Shi and G.K. Smyth. “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), pp.e47. 2015.
</p>
<p>J.D. Storey, J.E. Taylor, and D. Siegmund. Strong control, conservative point estimation and simultaneous conservative consistency of false discovery rates: a unified approach. Journal of the Royal Statistical Society: Series B (Statistical Methodology), 66(1):187-205, 2004.
</p>
<p>J.D. Storey and R. Tibshirani. Statistical significance for genomewide studies. Proceedings of the National Academy of Sciences, 100(16):9440-9445, 2003.
</p>
<p>J.D. Storey. qvalue: Q-value estimation for false discovery rate control. R package version 2.0.0, http://qvalue.princeton.edu/, http://github.com/jdstorey/qvalue. 2015.
</p>
<p>H.-Q. Wang, L.K. Tuominen, and C.-J. Tsai. SLIM: a sliding linear model for estimating the proportion of true null hypotheses in datasets with dependence structures. Bioinformatics, 27(2):225-231, 2011.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calibration.plot">calibration.plot</a></code>, <code><a href="#topic+adjust.p">adjust.p</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get p-values
data(LFQRatio2)
p=LFQRatio2[,7]

#estimate the proportion of true null hypotheses with different methods
r=estim.pi0(p)
r$pi0

#estimate the proportion of true null hypotheses with the "abh" method
r=estim.pi0(p, pi0.method="abh")
r$pi0

#compare with one minus the proportion of human proteins 
prop_human=sum(LFQRatio2$Organism=="human")/length(LFQRatio2$Organism)
pi0_true=1-prop_human
pi0_true

</code></pre>

<hr>
<h2 id='LFQRatio2'>Dataset &quot;LFQRatio2&quot;</h2><span id='topic+LFQRatio2'></span>

<h3>Description</h3>

<p>This dataset is the final outcome of a quantitative mass spectrometry-based proteomic analysis of two samples containing different concentrations of 48 human proteins (UPS1 standard from Sigma-Aldrich) within a constant yeast background (see Giai Gianetto et al. (2016) for details). It contains the abundance values of the different human and yeast proteins identified and quantified in these two conditions. The conditions A and B represent the measured abundances of proteins when respectively 10fmol and 5fmol of UPS1 human proteins were mixed with the yeast extract before mass spectrometry analyses. Three technical replicates were acquired for each condition.
</p>
<p>To identify and quantify proteins, spectra were searched using MaxQuant (version 1.5.1.2) against the Uniprot database, the UPS database and the frequently observed contaminants database. Maximum false discovery rates were set to 0.01 at peptide and protein levels by employing a reverse database strategy.
</p>
<p>The abundance values of the dataset were obtained from LFQ values calculated using MaxQuant from MS intensity of unique peptides (see Cox et al. (2014)). The following pre-processing steps were performed to obtain these values using the Perseus toolbox (version 1.5.1.6) available in the MaxQuant environment: log2-transformation of LFQ values, filtering of proteins with at least 3 measured values in one condition and data imputation by replacing missing values with values generated from a normal distribution (see Deeb et al. (2012)).
</p>
<p>From a statistical viewpoint, the goal is to find which proteins are differentially abundant between the two conditions among the 1481 quantified proteins. Ideally, the 38 quantified human proteins (out of the original 48 ones) should be concluded as differentially abundant (in such a case the proportion of non-differentially abundant proteins will be pi0=1-38/1481).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LFQRatio2)</code></pre>


<h3>Format</h3>

<p>A data frame with the 1481 identified proteins in row. 
</p>
<p>Columns <code>A.R1</code>, <code>A.R2</code> and <code>A.R3</code> correspond to the (numeric) abundance values of proteins in the three replicates of condition A. 
</p>
<p>Columns <code>B.R1</code>, <code>B.R2</code> and <code>B.R3</code> correspond to the (numeric) abundance values of proteins in the three replicates of condition B. 
</p>
<p>Column <code>Welch.test.pval</code> contains the p-values of the Welch t-test between condition A and condition B computed with the Perseus software.
</p>
<p>Column <code>Organism</code> contains categorical values: <code>human</code> if the protein is identified as human and <code>yeast</code> otherwise. 
</p>
<p>Column <code>Majority.protein.IDs</code> contains the IDs of proteins.
</p>


<h3>References</h3>

<p>Cox J., Hein M.Y., Luber C.A., Paron I., Nagaraj N., Mann M. Accurate proteome-wide label-free quantification by delayed normalization and maximal peptide ratio extraction, termed MaxLFQ. Mol Cell Proteomics. 2014 Sep, 13(9):2513-26.
</p>
<p>Deeb S.J., D'Souza R.C., Cox J., Schmidt-Supprian M., Mann M. Super-SILAC allows classification of diffuse large B-cell lymphoma subtypes by their protein expression profiles. Mol Cell Proteomics. 2012 May, 11(5):77-89.
</p>
<p>Giai Gianetto, Q., Combes, F., Ramus, C., Bruley, C., Couté, Y., Burger, T. (2016). Calibration plot for proteomics: A graphical tool to visually check the assumptions underlying FDR control in quantitative experiments. Proteomics, 16(1), 29-32.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LFQRatio2)

#p-values of the Welch t-test between condition A and condition B
p=LFQRatio2[,7]

</code></pre>

<hr>
<h2 id='LFQRatio25'>Dataset &quot;LFQRatio25&quot;</h2><span id='topic+LFQRatio25'></span>

<h3>Description</h3>

<p>This dataset is the final outcome of a quantitative mass spectrometry-based proteomic analysis of two samples containing different concentrations of 48 human proteins (UPS1 standard from Sigma-Aldrich) within a constant yeast background (see Giai Gianetto et al. (2016) for details). It contains the abundance values of the different human and yeast proteins identified and quantified in these two conditions. The conditions A and B represent the measured abundances of proteins when respectively 25fmol and 10fmol of UPS1 human proteins were mixed with the yeast extract before mass spectrometry analyses. Three technical replicates were acquired for each condition.
</p>
<p>To identify and quantify proteins, spectra were searched using MaxQuant (version 1.5.1.2) against the Uniprot database, the UPS database and the frequently observed contaminants database. Maximum false discovery rates were set to 0.01 at peptide and protein levels by employing a reverse database strategy.
</p>
<p>The abundance values of the dataset were obtained from LFQ values calculated using MaxQuant from MS intensity of unique peptides (see Cox et al. (2014)). The following pre-processing steps were performed to obtain these values using the Perseus toolbox (version 1.5.1.6) available in the MaxQuant environment: log2-transformation of LFQ values, filtering of proteins with at least 3 measured values in one condition and data imputation by replacing missing values with values generated from a normal distribution (see Deeb et al. (2012)).
</p>
<p>From a statistical viewpoint, the goal is to find which proteins are differentially abundant between the two conditions among the 1472 quantified proteins. Ideally, the 46 quantified human proteins (out of the original 48 ones) should be concluded as differentially abundant (in such a case the proportion of non-differentially abundant proteins will be pi0=1-46/1472).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LFQRatio25)</code></pre>


<h3>Format</h3>

<p>A data frame with the 1472 identified proteins in row. 
</p>
<p>Columns <code>A.R1</code>, <code>A.R2</code> and <code>A.R3</code> correspond to the (numeric) abundance values of proteins in the three replicates of condition A. 
</p>
<p>Columns <code>B.R1</code>, <code>B.R2</code> and <code>B.R3</code> correspond to the (numeric) abundance values of proteins in the three replicates of condition B. 
</p>
<p>Column <code>Welch.test.pval</code> contains the p-values of the Welch t-test between condition A and condition B computed with the Perseus software.
</p>
<p>Column <code>Organism</code> contains categorical values: <code>human</code> if the protein is identified as human and <code>yeast</code> otherwise. 
</p>
<p>Column <code>Majority.protein.IDs</code> contains the IDs of proteins.
</p>


<h3>References</h3>

<p>Cox J., Hein M.Y., Luber C.A., Paron I., Nagaraj N., Mann M. Accurate proteome-wide label-free quantification by delayed normalization and maximal peptide ratio extraction, termed MaxLFQ. Mol Cell Proteomics. 2014 Sep, 13(9):2513-26.
</p>
<p>Deeb S.J., D'Souza R.C., Cox J., Schmidt-Supprian M., Mann M. Super-SILAC allows classification of diffuse large B-cell lymphoma subtypes by their protein expression profiles. Mol Cell Proteomics. 2012 May, 11(5):77-89.
</p>
<p>Giai Gianetto, Q., Combes, F., Ramus, C., Bruley, C., Couté, Y., Burger, T. (2016). Calibration plot for proteomics: A graphical tool to visually check the assumptions underlying FDR control in quantitative experiments. Proteomics, 16(1), 29-32.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LFQRatio25)

#p-values of the Welch t-test between condition A and condition B
p=LFQRatio25[,7]

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
