<!DOCTYPE html><html><head><title>Help for package CPBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CPBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#analytic_locFDR_BF_cor'><p>Analytic calculation of the local FDR &amp; Bayes factor for correlated summary statistics.</p></a></li>
<li><a href='#analytic_locFDR_BF_uncor'><p>Analytic calculation of the local FDR &amp; Bayes factor for uncorrelated summary statistics.</p></a></li>
<li><a href='#CPBayes'><p>CPBayes: An R-package implemeting a Bayesian meta analysis method for studying cross-phenotype</p>
genetic associations.</a></li>
<li><a href='#cpbayes_cor'><p>Run correlated version of CPBayes.</p></a></li>
<li><a href='#cpbayes_uncor'><p>Run uncorrelated version of CPBayes.</p></a></li>
<li><a href='#estimate_corln'><p>Estimate correlation structure of beta-hat vector for multiple overlapping case-control studies</p>
using sample-overlap matrices.</a></li>
<li><a href='#ExampleDataCor'><p>An example data for correlated summary statistics.</p></a></li>
<li><a href='#ExampleDataUncor'><p>An example data for uncorrelated summary statistics.</p></a></li>
<li><a href='#forest_cpbayes'><p>Forest plot presenting pleiotropy result obtained by CPBayes.</p></a></li>
<li><a href='#post_summaries'><p>Post summary of the MCMC data generated by the uncorrelated or correlated version of CPBayes.</p></a></li>
<li><a href='#SampleOverlapMatrix'><p>An example data of sample-overlap matrices.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Bayesian Meta Analysis for Studying Cross-Phenotype Genetic
Associations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-12-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Arunabha Majumdar &lt;statgen.arunabha@gmail.com&gt; [aut, cre],
    Tanushree Haldar &lt;tanushree.haldar@gmail.com&gt; [aut],
    John Witte [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arunabha Majumdar &lt;statgen.arunabha@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A Bayesian meta-analysis method for studying cross-phenotype
    genetic associations. It uses summary-level data across multiple phenotypes to
    simultaneously measure the evidence of aggregate-level pleiotropic association
    and estimate an optimal subset of traits associated with the risk locus. CPBayes
    is based on a spike and slab prior. The methodology is available from: A Majumdar, T Haldar, S Bhattacharya, JS Witte (2018) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1007139">doi:10.1371/journal.pgen.1007139</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ArunabhaCodes/CPBayes">https://github.com/ArunabhaCodes/CPBayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ArunabhaCodes/CPBayes/issues">https://github.com/ArunabhaCodes/CPBayes/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, stats, forestplot, grDevices, purrr, mvtnorm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-12-01 21:33:03 UTC; arunmacpro13</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-12-02 07:40:23 UTC</td>
</tr>
</table>
<hr>
<h2 id='analytic_locFDR_BF_cor'>Analytic calculation of the local FDR &amp; Bayes factor for correlated summary statistics.</h2><span id='topic+analytic_locFDR_BF_cor'></span>

<h3>Description</h3>

<p>Run the <code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code> function to analytically compute the local FDR &amp; Bayes factor (BF)
that quantifies the evidence of aggregate-level pleiotropic association for correlated summary statistics.
Here a fixed value of slab variance is considred instead of a range of it in <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analytic_locFDR_BF_cor(BetaHat, SE, Corln, SpikeVar = 1e-04, SlabVar = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analytic_locFDR_BF_cor_+3A_betahat">BetaHat</code></td>
<td>
<p>A numeric vector of length K where K is the number of phenotypes. It
contains the beta-hat values across studies/traits. No default.</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_cor_+3A_se">SE</code></td>
<td>
<p>A numeric vector with the same dimension as BetaHat providing the standard errors
corresponding to BetaHat. Every element of SE must be positive. No default.</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_cor_+3A_corln">Corln</code></td>
<td>
<p>A numeric square matrix of order K by K providing the correlation matrix of BetaHat.
The number of rows &amp; columns of Corln must be the same as the length of BetaHat. No default
is specified. See <code><a href="#topic+estimate_corln">estimate_corln</a></code>.</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_cor_+3A_spikevar">SpikeVar</code></td>
<td>
<p>Variance of spike (normal distribution with small variance) representing the null effect distribution.
Default is 10^(-4).</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_cor_+3A_slabvar">SlabVar</code></td>
<td>
<p>Variance of slab normal distribution representing the non-null effect distribution.
Default is 0.8.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by the function is a list which consists of the local FDR and log10(Bayes factor).
</p>
<table>
<tr><td><code>locFDR</code></td>
<td>
<p>It provides the analytically computed local false discovery rate (posterior probability of null association) under CPBayes model
(a Bayesian analog of the p-value) which is a measure of the evidence of the
aggregate-level pleiotropic association. Bayes factor is adjusted for prior odds, but
locFDR is solely a function of the posterior odds.</p>
</td></tr>
<tr><td><code>log10_BF</code></td>
<td>
<p>It provides the analytically computed log10(Bayes factor) produced by CPBayes that measures the
evidence of the overall pleiotropic association.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>, <code><a href="#topic+estimate_corln">estimate_corln</a></code>, <code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code>, <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>, <code><a href="#topic+post_summaries">post_summaries</a></code>, <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataCor)
BetaHat &lt;- ExampleDataCor$BetaHat
BetaHat
SE &lt;- ExampleDataCor$SE
SE
cor &lt;- ExampleDataCor$cor
cor
result &lt;- cpbayes_cor(BetaHat, SE, cor)
str(result)

</code></pre>

<hr>
<h2 id='analytic_locFDR_BF_uncor'>Analytic calculation of the local FDR &amp; Bayes factor for uncorrelated summary statistics.</h2><span id='topic+analytic_locFDR_BF_uncor'></span>

<h3>Description</h3>

<p>Run the <code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code> function to analytically compute the local FDR &amp; Bayes factor (BF)
that quantifies the evidence of aggregate-level pleiotropic association for uncorrelated summary statistics.
Here a fixed value of slab variance is considred instead of a range of it in <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analytic_locFDR_BF_uncor(BetaHat, SE, SpikeVar = 1e-04, SlabVar = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analytic_locFDR_BF_uncor_+3A_betahat">BetaHat</code></td>
<td>
<p>A numeric vector of length K where K is the number of phenotypes. It
contains the beta-hat values across studies/traits. No default.</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_uncor_+3A_se">SE</code></td>
<td>
<p>A numeric vector with the same dimension as BetaHat providing the standard errors
corresponding to BetaHat. Every element of SE must be positive. No default.</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_uncor_+3A_spikevar">SpikeVar</code></td>
<td>
<p>Variance of spike (normal distribution with small variance) representing the null effect distribution.
Default is 10^(-4).</p>
</td></tr>
<tr><td><code id="analytic_locFDR_BF_uncor_+3A_slabvar">SlabVar</code></td>
<td>
<p>Variance of slab normal distribution representing the non-null effect distribution.
Default is 0.8.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by the function is a list which consists of the local FDR and log10(Bayes factor).
</p>
<table>
<tr><td><code>locFDR</code></td>
<td>
<p>It provides the analytically computed local false discovery rate (posterior probability of null association) under CPBayes model
(a Bayesian analog of the p-value) which is a measure of the evidence of the
aggregate-level pleiotropic association. Bayes factor is adjusted for prior odds, but
locFDR is solely a function of the posterior odds.</p>
</td></tr>
<tr><td><code>log10_BF</code></td>
<td>
<p>It provides the analytically computed log10(Bayes factor) produced by CPBayes that measures the
evidence of the overall pleiotropic association.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>, <code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code>, <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>, <code><a href="#topic+estimate_corln">estimate_corln</a></code>, <code><a href="#topic+post_summaries">post_summaries</a></code>, <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
BetaHat &lt;- ExampleDataUncor$BetaHat
BetaHat
SE &lt;- ExampleDataUncor$SE
SE
result &lt;- analytic_locFDR_BF_uncor(BetaHat, SE)
str(result)

</code></pre>

<hr>
<h2 id='CPBayes'>CPBayes: An R-package implemeting a Bayesian meta analysis method for studying cross-phenotype
genetic associations.</h2><span id='topic+CPBayes'></span>

<h3>Description</h3>

<p>Simultaneous analysis of genetic associations with multiple phenotypes may reveal shared
genetic susceptibility across traits (pleiotropy). CPBayes is a Bayesian meta analysis
method for studying cross-phenotype genetic associations. It uses summary-level data
across multiple phenotypes to simultaneously measure the evidence of aggregate-level
pleiotropic association and estimate an optimal subset of traits associated with the
risk locus. CPBayes is based on a spike and slab prior.
</p>


<h3>Details</h3>

<p>The package consists of following functions:
<code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code>, <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>; <code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code>, <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>; <code><a href="#topic+post_summaries">post_summaries</a></code>, <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>, <code><a href="#topic+estimate_corln">estimate_corln</a></code>.
</p>


<h3>Functions</h3>


<dl>
<dt><code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code></dt><dd><p>It analytically computes the local FDR (locFDR)
and Bayes factor (BF) quantifying the evidence of aggregate-level pleiotropic association
for uncorrelated summary statistics.</p>
</dd>
<dt><code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code></dt><dd><p>It implements CPBayes (based on MCMC) for uncorrelated summary statistics to
figure out the optimal subset of non-null traits underlying a pleiotropic signal and other insights.
The summary statistics across traits/studies are uncorrelated when the studies
have no overlapping/genetically related subjects.</p>
</dd>
<dt><code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code></dt><dd><p>It analytically computes the local FDR (locFDR)
and Bayes factor (BF) for correlated summary statistics.</p>
</dd>
<dt><code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code></dt><dd><p>It implements CPBayes (based on MCMC) for correlated summary statistics to figure out
the optimal subset of non-null traits underlying a pleiotropic signal and other insights. The summary statistics across
traits/studies are correlated when the studies have overlapping/genetically related subjects
or the phenotypes were measured in a cohort study.</p>
</dd>
<dt><code><a href="#topic+post_summaries">post_summaries</a></code></dt><dd><p>It summarizes the MCMC data produced by
<code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>.
It computes additional summaries to provide a better insight into a pleiotropic signal.
It works in the same way for both <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> and <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>.</p>
</dd>
<dt><code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code></dt><dd><p>It creates a forest plot presenting the pleiotropy result obtained by
<code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. It works in the same way for
both <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> and <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>.</p>
</dd>
<dt><code><a href="#topic+estimate_corln">estimate_corln</a></code></dt><dd><p>It computes an approximate correlation matrix of
the beta-hat vector for multiple overlapping case-control studies using the
sample-overlap count matrices.</p>
</dd>
</dl>



<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>

<hr>
<h2 id='cpbayes_cor'>Run correlated version of CPBayes.</h2><span id='topic+cpbayes_cor'></span>

<h3>Description</h3>

<p>Run correlated version of CPBayes when the main genetic effect (beta/log(odds ratio)) estimates across
studies/traits are correlated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpbayes_cor(
  BetaHat,
  SE,
  Corln,
  Phenotypes,
  Variant,
  UpdateSlabVar = TRUE,
  MinSlabVar = 0.6,
  MaxSlabVar = 1,
  MCMCiter = 7500,
  Burnin = 500
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpbayes_cor_+3A_betahat">BetaHat</code></td>
<td>
<p>A numeric vector of length K where K is the number of phenotypes. It
contains the beta-hat values across studies/traits. No default is specified.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_se">SE</code></td>
<td>
<p>A numeric vector with the same dimension as BetaHat providing the standard errors
corresponding to BetaHat. Every element of SE must be positive. No default is specified.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_corln">Corln</code></td>
<td>
<p>A numeric square matrix of order K by K providing the correlation matrix of BetaHat.
The number of rows &amp; columns of Corln must be the same as the length of BetaHat. No default
is specified. See <code><a href="#topic+estimate_corln">estimate_corln</a></code>.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_phenotypes">Phenotypes</code></td>
<td>
<p>A character vector of the same length as BetaHat providing the name
of the phenotypes. Default is specified as trait1, trait2, . . . , traitK. Note that
BetaHat, SE, Corln, and Phenotypes must be in the same order.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_variant">Variant</code></td>
<td>
<p>A character vector of length one providing the name of the genetic variant.
Default is &lsquo;Variant&rsquo;.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_updateslabvar">UpdateSlabVar</code></td>
<td>
<p>A logical vector of length one. If TRUE, the variance of the slab distribution
that presents the prior distribution of non-null effects is
updated at each MCMC iteration in a range (MinSlabVar &ndash; MaxSlabVar) (see next). If FALSE,
it is fixed at (MinSlabVar + MaxSlabVar)/2. Default is TRUE.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_minslabvar">MinSlabVar</code></td>
<td>
<p>A numeric value greater than 0.01 providing the minimum value of the
variance of the slab distribution. Default is 0.6.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_maxslabvar">MaxSlabVar</code></td>
<td>
<p>A numeric value smaller than 10.0 providing the maximum value of the
variance of the slab distribution. Default is 1.0. **Note that,
a smaller value of the slab variance will increase the sensitivity of CPBayes while selecting the optimal
subset of associated traits but at the expense of lower specificity. Hence the slab variance
parameter in CPBayes is inversely related to the level of false discovery rate (FDR) in a frequentist
FDR controlling procedure. For a specific dataset, an user
can experiment different choices of these three arguments: UpdateSlabVar, MinSlabVar, and MaxSlabVar.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_mcmciter">MCMCiter</code></td>
<td>
<p>A positive integer greater than or equal to 2200 providing the total number of
iterations in the MCMC. Default is 7500.</p>
</td></tr>
<tr><td><code id="cpbayes_cor_+3A_burnin">Burnin</code></td>
<td>
<p>A positive integer greater than or equal to 200 providing the burn in period
in the MCMC. Default is 200. Note that the MCMC sample size (MCMCiter - Burnin) must be at least 2000, which is 7000 by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code> is a list which consists of various components.
</p>
<table>
<tr><td><code>variantName</code></td>
<td>
<p>It is the name of the genetic variant provided by the user. If
not specified by the user, default name is &lsquo;Variant&rsquo;.</p>
</td></tr>
<tr><td><code>log10_BF</code></td>
<td>
<p>It provides the log10(Bayes factor) produced by CPBayes that measures the
evidence of the overall pleiotropic association.</p>
</td></tr>
<tr><td><code>locFDR</code></td>
<td>
<p>It provides the local false discovery rate (posterior probability of null association) produced by
CPBayes which is a measure of the evidence
of aggregate-level pleiotropic association. Bayes factor is adjusted for prior odds, but
locFDR is solely a function of the posterior odds. locFDR can sometimes be small
indicating an association, but log10_BF may not indicate an association. Hence, always check both log10_BF and locFDR.</p>
</td></tr>
<tr><td><code>subset</code></td>
<td>
<p>It provides the optimal subset of associated/non-null traits selected by
CPBayes. It is NULL if no phenotype is selected.</p>
</td></tr>
<tr><td><code>important_traits</code></td>
<td>
<p>It provides the traits which yield a trait-specific posterior
probability of association (PPAj) &gt; 20%. Even if a phenotype is not
selected in the optimal subset of non-null traits, it can produce a non-negligible
value of PPAj. Note that, &lsquo;important_traits&rsquo; is expected to include the traits already
contained in &lsquo;subset&rsquo;. It provides both the name of the important traits and their
corresponding value of PPAj. Always check 'important_traits' even if 'subset' contains
a single trait. It helps to better explain an observed pleiotropic signal.</p>
</td></tr>
<tr><td><code>auxi_data</code></td>
<td>
<p>It contains supplementary data including the MCMC data which is used later
by <code><a href="#topic+post_summaries">post_summaries</a></code> and <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>:
</p>

<ol>
<li><p> traitNames: Name of all the phenotypes.
</p>
</li>
<li><p> K: Total number of phenotypes.
</p>
</li>
<li><p> mcmc.samplesize: MCMC sample size.
</p>
</li>
<li><p> PPAj: Trait-specific posterior probability of association for all the traits.
</p>
</li>
<li><p> Z.data: MCMC data on the latent association status of all the traits (Z).
</p>
</li>
<li><p> sim.beta: MCMC data on the unknown true genetic effect (beta) on each trait.
</p>
</li>
<li><p> betahat: The beta-hat vector provided by the user which will be used by <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>.
</p>
</li>
<li><p> se: The standard error vector provided by the user which will be used by <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>.
</p>
</li></ol>

</td></tr>
<tr><td><code>uncor_use</code></td>
<td>
<p>'Yes' or 'No'. Whether the combined strategy of CPBayes (implemented for correlated
summary statistics) used the uncorrelated version or not.</p>
</td></tr>
<tr><td><code>runtime</code></td>
<td>
<p>It provides the runtime (in seconds) taken by <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. It will help the user to
plan the whole analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code>, <code><a href="#topic+estimate_corln">estimate_corln</a></code>, <code><a href="#topic+post_summaries">post_summaries</a></code>, <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>, <code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code>, <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataCor)
BetaHat &lt;- ExampleDataCor$BetaHat
BetaHat
SE &lt;- ExampleDataCor$SE
SE
cor &lt;- ExampleDataCor$cor
cor
traitNames &lt;- paste("Disease", 1:10, sep = "")
SNP1 &lt;- "rs1234"
result &lt;- cpbayes_cor(BetaHat, SE, cor, Phenotypes = traitNames, Variant = SNP1)
str(result)

</code></pre>

<hr>
<h2 id='cpbayes_uncor'>Run uncorrelated version of CPBayes.</h2><span id='topic+cpbayes_uncor'></span>

<h3>Description</h3>

<p>Run uncorrelated version of CPBayes when the main genetic effect (beta/log(odds ratio)) estimates across
studies/traits are uncorrelated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpbayes_uncor(
  BetaHat,
  SE,
  Phenotypes,
  Variant,
  UpdateSlabVar = TRUE,
  MinSlabVar = 0.6,
  MaxSlabVar = 1,
  MCMCiter = 7500,
  Burnin = 500
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpbayes_uncor_+3A_betahat">BetaHat</code></td>
<td>
<p>A numeric vector of length K where K is the number of phenotypes. It
contains the beta-hat values across studies/traits. No default is specified.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_se">SE</code></td>
<td>
<p>A numeric vector with the same dimension as BetaHat providing the standard errors
corresponding to BetaHat. Every element of SE must be positive. No default is specified.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_phenotypes">Phenotypes</code></td>
<td>
<p>A character vector of the same length as BetaHat providing the name of
the phenotypes. Default is specified as trait1, trait2, . . . , traitK. Note that BetaHat,
SE, and Phenotypes must be in the same order.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_variant">Variant</code></td>
<td>
<p>A character vector of length one specifying the name of the genetic variant.
Default is &lsquo;Variant&rsquo;.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_updateslabvar">UpdateSlabVar</code></td>
<td>
<p>A logical vector of length one. If TRUE, the variance of the slab distribution
that presents the prior distribution of non-null effects
is updated at each MCMC iteration in a range (MinSlabVar &ndash; MaxSlabVar) (see next). If FALSE,
it is fixed at (MinSlabVar + MaxSlabVar)/2. Default is TRUE.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_minslabvar">MinSlabVar</code></td>
<td>
<p>A numeric value greater than 0.01 providing the minimum value of
the variance of the slab distribution. Default is 0.6.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_maxslabvar">MaxSlabVar</code></td>
<td>
<p>A numeric value smaller than 10.0 providing the maximum value of
the variance of the slab distribution. Default is 1.0. **Note that,
a smaller value of the slab variance will increase the sensitivity of CPBayes while selecting the optimal
subset of associated traits but at the expense of lower specificity. Hence the slab variance
parameter in CPBayes is inversely related to the level of false discovery rate (FDR) in a frequentist
FDR controlling procedure. For a specific dataset, an user
can experiment different choices of these three arguments: UpdateSlabVar, MinSlabVar, and MaxSlabVar.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_mcmciter">MCMCiter</code></td>
<td>
<p>A positive integer greater than or equal to 2200 providing the total number of
iterations in the MCMC. Default is 7500.</p>
</td></tr>
<tr><td><code id="cpbayes_uncor_+3A_burnin">Burnin</code></td>
<td>
<p>A positive integer greater than or equal to 200 providing the burn in period
in the MCMC. Default is 500. Note that the MCMC sample size (MCMCiter - Burnin) must be at least 2000, which is 7000 by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by the function is a list which consists of various components.
</p>
<table>
<tr><td><code>variantName</code></td>
<td>
<p>It is the name of the genetic variant provided by the user. If not
specified by the user, default name is &lsquo;Variant&rsquo;.</p>
</td></tr>
<tr><td><code>log10_BF</code></td>
<td>
<p>It provides the log10(Bayes factor) produced by CPBayes that measures the
evidence of the overall pleiotropic association.</p>
</td></tr>
<tr><td><code>locFDR</code></td>
<td>
<p>It provides the local false discovery rate (posterior probability of null association) produced by CPBayes
which is a measure of the evidence of the
aggregate-level pleiotropic association. Bayes factor is adjusted for prior odds, but
locFDR is solely a function of the posterior odds. locFDR can sometimes be small
indicating an association, but log10_BF may not indicate an association. Hence, always check both log10_BF and locFDR.</p>
</td></tr>
<tr><td><code>subset</code></td>
<td>
<p>It provides the optimal subset of associated/non-null traits selected
by CPBayes. It is NULL if no phenotype is selected.</p>
</td></tr>
<tr><td><code>important_traits</code></td>
<td>
<p>It provides the traits which yield a trait-specific posterior probability of
association (PPAj) &gt; 20%. Even if a phenotype is not selected in the optimal subset of non-null
traits, it can produce a non-negligible value of trait-specific posterior probability of
association (PPAj). Note that, &lsquo;important_traits&rsquo; is expected to include the traits
already contained in &lsquo;subset&rsquo;. It provides both the name of the important traits and
their corresponding values of PPAj. Always check 'important_traits' even if 'subset' contains
a single trait. It helps to better explain an observed pleiotropic signal.</p>
</td></tr>
<tr><td><code>auxi_data</code></td>
<td>
<p>It contains supplementary data including the MCMC data which is used later
by <code><a href="#topic+post_summaries">post_summaries</a></code> and <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>:
</p>

<ol>
<li><p> traitNames: Name of all the phenotypes.
</p>
</li>
<li><p> K: Total number of phenotypes.
</p>
</li>
<li><p> mcmc.samplesize: MCMC sample size.
</p>
</li>
<li><p> PPAj: Trait-specific posterior probability of association for all the traits.
</p>
</li>
<li><p> Z.data: MCMC data on the latent association status of all the traits (Z).
</p>
</li>
<li><p> sim.beta: MCMC data on the unknown true genetic effect (beta) on all the traits.
</p>
</li>
<li><p> betahat: The beta-hat vector provided by the user which will be used by <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>.
</p>
</li>
<li><p> se: The standard error vector provided by the user which will be used by <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>.
</p>
</li></ol>

</td></tr>
<tr><td><code>runtime</code></td>
<td>
<p>It provides the runtime (in seconds) taken by <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>. It will help the user
to plan the whole analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+analytic_locFDR_BF_uncor">analytic_locFDR_BF_uncor</a></code>, <code><a href="#topic+post_summaries">post_summaries</a></code>, <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code>, <code><a href="#topic+analytic_locFDR_BF_cor">analytic_locFDR_BF_cor</a></code>, <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>, <code><a href="#topic+estimate_corln">estimate_corln</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
BetaHat &lt;- ExampleDataUncor$BetaHat
BetaHat
SE &lt;- ExampleDataUncor$SE
SE
traitNames &lt;- paste("Disease", 1:10, sep = "")
SNP1 &lt;- "rs1234"
result &lt;- cpbayes_uncor(BetaHat, SE, Phenotypes = traitNames, Variant = SNP1)
str(result)

</code></pre>

<hr>
<h2 id='estimate_corln'>Estimate correlation structure of beta-hat vector for multiple overlapping case-control studies
using sample-overlap matrices.</h2><span id='topic+estimate_corln'></span>

<h3>Description</h3>

<p>It computes an approximate correlation matrix of the estimated  beta (log odds ratio) vector for multiple overlapping
case-control studies using the sample-overlap matrices which describe
the number of cases or controls shared between studies/traits, and the number of subjects
who are case for one study/trait but control for another study/trait. For a cohort study,
the phenotypic correlation matrix should be a reasonable substitute of this correlation matrix.
These approximations are accurate when none of the diseases/traits is associated with
the environmental covariates and genetic variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_corln(n11, n00, n10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_corln_+3A_n11">n11</code></td>
<td>
<p>An integer square matrix (number of rows must be the same as the number of
studies/traits) providing the
number of cases shared between all possible pairs of studies/traits. So (k,l)-th element of n11
is the number of subjects who are case for both k-th and l-th study/trait. Note that the diagonal elements of
n11 are the number of cases in the studies/traits. If no case is shared between studies/traits,
the off-diagonal elements of n11 will be zero. No default is specified.</p>
</td></tr>
<tr><td><code id="estimate_corln_+3A_n00">n00</code></td>
<td>
<p>An integer square matrix (number of rows must be the same as the
number of studies/traits) providing the
number of controls shared between all possible pairs of studies/traits. So (k,l)-th element of n00
is the number subjects who are control for both k-th and l-th study/trait. Note that the diagonal
elements of n00 are the number of controls in the studies/traits. If no control is
shared between studies/traits,
the off-diagonal elements will be zero. No default is specified.</p>
</td></tr>
<tr><td><code id="estimate_corln_+3A_n10">n10</code></td>
<td>
<p>An integer square matrix (number of rows must be the same as the
number of studies/traits) providing the
number of subjects who are case for one study/trait and control for another study/trait.
Clearly, the diagonal elements
will be zero. An off diagonal element, e.g., (k,l)-th element of n10 is the number of subjects who
are case for k-th study/trait and control for l-th study/trait. If there is no such overlap,
all the elements
of n10 will be zero. No default is specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>***Important note on the estimation of correlation structure of correlated beta-hat vector:***
In general, environmental covariates are expected to be present in a study and associated
with the phenotypes of interest. Also, a small proportion of genome-wide genetic variants
are expected to be associated. Hence the above approximation of the correlation matrix
may not be accurate. So in general, we recommend
an alternative strategy to estimate the correlation matrix using the genome-wide summary
statistics data across traits as follows. First, extract all the SNPs for each of which the
trait-specific univariate association p-value across all the traits are &gt; 0.1. The
trait-specific univariate association p-values are obtained using the beta-hat
and standard error for each trait. Each of the SNPs selected in this way is either weakly
or not associated with any of the phenotypes (null SNP). Next, select a set of independent
null SNPs from the initial set of null SNPs by using a threshold of r^2 &lt; 0.01 (r: the
correlation between the genotypes at a pair of SNPs). In the absence of in-sample
linkage disequilibrium (LD) information, one can use the reference panel LD information
for this screening. Finally, compute the correlation
matrix of the effect estimates (beta-hat vector) as the sample correlation matrix of the
beta-hat vector across all the selected independent null SNPs. This strategy is more
general and applicable to a cohort study or multiple overlapping studies for binary or
quantitative traits with arbitrary distributions. It is also useful when the beta-hat vector
for multiple non-overlapping studies become correlated due to genetically related
individuals across studies. Misspecification of the correlation
structure can affect the results produced by CPBayes to some extent. Hence, if
genome-wide summary statistics data across traits is available, we highly recommend to use
this alternative strategy to estimate the correlation matrix of the beta-hat vector.
See our paper for more details.
</p>


<h3>Value</h3>

<p>This function returns an approximate correlation matrix of the beta-hat vector for
multiple overlapping case-control studies. See the example below.
</p>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SampleOverlapMatrix)
n11 &lt;- SampleOverlapMatrix$n11
n11
n00 &lt;- SampleOverlapMatrix$n00
n00
n10 &lt;- SampleOverlapMatrix$n10
n10
cor &lt;- estimate_corln(n11, n00, n10)
cor

</code></pre>

<hr>
<h2 id='ExampleDataCor'>An example data for correlated summary statistics.</h2><span id='topic+ExampleDataCor'></span>

<h3>Description</h3>

<p>ExampleDataCor is a list consisting of three components: BetaHat, SE, cor. ExampleDataCor$BetaHat is a
numeric vector that contains the main genetic effect (beta/log(odds ratio)) estimates
for a SNP across 10 overlapping case-control studies for 10 different diseases. Each of the 10 studies has
a distinct set of 7000 cases and a common set of 10000 controls shared across all the studies.
In each case-control study, we fit a logistic regression of the case-control status on the genotype
coded as the minor allele count for all the individuals in the sample. One can also include various
covariates, such as, age, gender, principal components (PCs) of ancestries in the logistic regression.
From each logistic regression for a disease, we obtain the estimate of the main genetic association
parameter (beta/log(odds ratio)) along with the corresponding standard error. Since the studies have overlapping
subjects, the beta-hat across traits are correlated. ExampleDataCor$SE contains the standard error vector
corresponding to the correlated beta-hat vector. ExampleDataCor$cor is a numeric square matrix providing
the correlation matrix of the correlated beta-hat vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleDataCor)
</code></pre>


<h3>Format</h3>

<p>A list consisting of two numeric vectors (each of length 10) and a numeric square matrix of
dimension 10 by 10: 
</p>

<dl>
<dt>BetaHat</dt><dd><p>beta hat vector of length 10.</p>
</dd>
<dt>SE</dt><dd><p>standard error vector corresponding to the beta-hat vector.</p>
</dd>
<dt>cor</dt><dd><p>correlation matrix of the beta-hat vector.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataCor)
BetaHat &lt;- ExampleDataCor$BetaHat
BetaHat
SE &lt;- ExampleDataCor$SE
SE
cor &lt;- ExampleDataCor$cor
cor
cpbayes_cor(BetaHat, SE, cor)
</code></pre>

<hr>
<h2 id='ExampleDataUncor'>An example data for uncorrelated summary statistics.</h2><span id='topic+ExampleDataUncor'></span>

<h3>Description</h3>

<p>ExampleDataUncor is a list which has two components: BetaHat, SE. The numeric vector 
ExampleDataUncor$BetaHat contains the main genetic effect (beta/log(odds ratio)) estimates
for a single nucleotide polymorphism (SNP) obtained from 10 separate case-control studies
for 10 different diseases. In each case-control study comprising a
distinct set of 7000 cases and 10000 controls, we fit a logistic regression
of the case-control status on the genotype coded as the minor allele count
for all the individuals in the sample. One can also include various covariates,
such as, age, gender, principal components (PCs) of ancestries in the logistic regression.
From each logistic regression for a
disease, we obtain the estimate of the main genetic association parameter
(beta/log(odds ratio)) along with the corresponding standard error.
Since the studies do not have any overlapping subject, the beta-hat across the traits are uncorrelated.
ExampleDataUncor$SE is the second numeric vector that contains the standard errors corresponding to
the uncorrelated beta-hat vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
</code></pre>


<h3>Format</h3>

<p>A list of two numeric vectors each of length 10 (for 10 studies):
</p>

<dl>
<dt>BetaHat</dt><dd><p>beta hat vector of length 10.</p>
</dd>
<dt>SE</dt><dd><p>standard error vector corresponding to beta-hat vector.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
BetaHat &lt;- ExampleDataUncor$BetaHat
BetaHat
SE &lt;- ExampleDataUncor$SE
SE
cpbayes_uncor(BetaHat, SE)
</code></pre>

<hr>
<h2 id='forest_cpbayes'>Forest plot presenting pleiotropy result obtained by CPBayes.</h2><span id='topic+forest_cpbayes'></span>

<h3>Description</h3>

<p>Run the <code><a href="#topic+forest_cpbayes">forest_cpbayes</a></code> function to create a forest plot that presents the pleiotropy result obtained
by <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forest_cpbayes(mcmc_output, level = 0.05, PPAj_cutoff = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forest_cpbayes_+3A_mcmc_output">mcmc_output</code></td>
<td>
<p>A list returned by either
<code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. This list
contains all the primary results and MCMC data produced by <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>
or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. No default is specified. See the example below.</p>
</td></tr>
<tr><td><code id="forest_cpbayes_+3A_level">level</code></td>
<td>
<p>A numeric value. (1-level)% confidence interval of the unknown true genetic effect (beta/log(odds ratio))
on each trait is plotted in the forest plot. Default choice is 0.05.</p>
</td></tr>
<tr><td><code id="forest_cpbayes_+3A_ppaj_cutoff">PPAj_cutoff</code></td>
<td>
<p>A numeric value. It's a user-specified threshold of PPAj (trait-specific posterior probability
of association). Only those traits having PPAj values above this cut-off are included in the forest plot. So, the choice of
this variable as '0.0' includes all traits in the forest plot. Default is 0.01.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by this function is a diagram file in .pdf format. The details of the diagram are as follows:
</p>
<table>
<tr><td><code>file_name</code></td>
<td>
<p>The pdf file is named after the genetic variant. So, if the argument &lsquo;Variant&rsquo;
in <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code> is specified as 'rs1234', the figure file is named as rs1234.pdf.</p>
</td></tr>
<tr><td><code>Column1</code></td>
<td>
<p>First column in the figure specifies the name of the phenotypes.</p>
</td></tr>
<tr><td><code>Column2</code></td>
<td>
<p>Second column provides the trait-specific univariate association p-value for a trait.</p>
</td></tr>
<tr><td><code>Column3</code></td>
<td>
<p>Third column provides the trait-specific posterior probability of association (PPAj) produced by CPBayes.</p>
</td></tr>
<tr><td><code>Column4</code></td>
<td>
<p>Fourth column states whether a phenotype was selected in the optimal subset of associated/non-null traits
detected by CPBayes. If a phenotype was not selected, selected and positively associated, selected and negatively associated,
its association status is stated as null, positive and negative, respectively.</p>
</td></tr>
<tr><td><code>Column5</code></td>
<td>
<p>In the right section of the figure, the primary eatimate and confidence interval of the beta/log odds ratio parameter for
a trait is plotted.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>, <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
BetaHat &lt;- ExampleDataUncor$BetaHat
SE &lt;- ExampleDataUncor$SE
traitNames &lt;- paste("Disease", 1:10, sep = "")
SNP1 &lt;- "rs1234"
result &lt;- cpbayes_uncor(BetaHat, SE, Phenotypes = traitNames, Variant = SNP1)
## Not run: forest_cpbayes(result, level = 0.05)

</code></pre>

<hr>
<h2 id='post_summaries'>Post summary of the MCMC data generated by the uncorrelated or correlated version of CPBayes.</h2><span id='topic+post_summaries'></span>

<h3>Description</h3>

<p>Run the <code><a href="#topic+post_summaries">post_summaries</a></code> function to summarize the MCMC data produced by
<code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code> and obtain meaningful insights
into an observed pleiotropic signal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>post_summaries(mcmc_output, level = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="post_summaries_+3A_mcmc_output">mcmc_output</code></td>
<td>
<p>A list returned by either
<code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code> or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. This list
contains the primary results and MCMC data produced by <code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>
or <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>. No default is specified. See the example below.</p>
</td></tr>
<tr><td><code id="post_summaries_+3A_level">level</code></td>
<td>
<p>A numeric value. (1-level)% credible interval (Bayesian analog of the
confidence interval) of the unknown true genetic effect (beta/odds ratio)
on each trait is computed. Default choice is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by this function is a list that consists of various components.
</p>
<table>
<tr><td><code>variantName</code></td>
<td>
<p>It is the name of the genetic variant provided by the user. If not
specified by the user, default name is &lsquo;Variant&rsquo;.</p>
</td></tr>
<tr><td><code>log10_BF</code></td>
<td>
<p>It provides the log10(Bayes factor) produced by CPBayes that measures
the evidence of the overall pleiotropic association.</p>
</td></tr>
<tr><td><code>locFDR</code></td>
<td>
<p>It provides the local false discovery rate (posterior probability of null association) produced by
CPBayes (a Bayesian analog of the p-value) which is a measure of the evidence
of aggregate-level pleiotropic association. Bayes factor is adjusted for prior odds, but
locFDR is solely a function of posterior odds. locFDR can sometimes be significantly small
indicating an association, but log10_BF may not. Hence, always check both log10_BF and locFDR.</p>
</td></tr>
<tr><td><code>subset</code></td>
<td>
<p>A data frame providing the optimal subset of associated/non-null traits
along with their trait-specific posterior probability of association (PPAj) and direction
of associations. It is NULL if no phenotype is selected by CPBayes.</p>
</td></tr>
<tr><td><code>important_traits</code></td>
<td>
<p>It provides the traits which yield a trait-specific posterior
probability of association (PPAj) &gt; 20%. Even if a phenotype is not selected in the
optimal subset of non-null traits, it can produce a non-negligible value of
trait-specific posterior probability of association. We note that &lsquo;important_traits&rsquo;
is expected to include the traits already contained in &lsquo;subset&rsquo;. It provides the
name of the important traits and their trait-specific posterior probability of
association (PPAj) and the direction of associations. Always check
'important_traits' even if 'subset' contains a single trait.
It helps to better explain an observed pleiotropic signal.</p>
</td></tr>
<tr><td><code>traitNames</code></td>
<td>
<p>It returns the name of all the phenotypes specified by the user.
Default is trait1, trait2, ... , traitK.</p>
</td></tr>
<tr><td><code>PPAj</code></td>
<td>
<p>Data frame providing the trait-specific posterior probability of
association for all the phenotypes.</p>
</td></tr>
<tr><td><code>poste_summary_beta</code></td>
<td>
<p>Data frame providing the posterior summary of the
unknown true genetic effect (beta) on each trait. It gives posterior mean,
median, standard error, credible interval (lower and upper limits) of the
true beta corresponding to each trait.</p>
</td></tr>
<tr><td><code>poste_summary_OR</code></td>
<td>
<p>Data frame providing the posterior summary of the unknown
true genetic effect (odds ratio) on each trait. It gives posterior mean, median,
standard error, credible interval (lower and upper limits) of the true odds
ratio corresponding to each trait.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Majumdar A, Haldar T, Bhattacharya S, Witte JS (2018) An efficient Bayesian meta analysis approach for studying cross-phenotype genetic associations. PLoS Genet 14(2): e1007139.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpbayes_uncor">cpbayes_uncor</a></code>, <code><a href="#topic+cpbayes_cor">cpbayes_cor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleDataUncor)
BetaHat &lt;- ExampleDataUncor$BetaHat
BetaHat
SE &lt;- ExampleDataUncor$SE
SE
traitNames &lt;- paste("Disease", 1:10, sep = "")
SNP1 &lt;- "rs1234"
result &lt;- cpbayes_uncor(BetaHat, SE, Phenotypes = traitNames, Variant = SNP1)
PleioSumm &lt;- post_summaries(result, level = 0.05)
str(PleioSumm)

</code></pre>

<hr>
<h2 id='SampleOverlapMatrix'>An example data of sample-overlap matrices.</h2><span id='topic+SampleOverlapMatrix'></span>

<h3>Description</h3>

<p>An example data of sample-overlap matrices for five different diseases in the Kaiser 
GERA cohort (a real data).
SampleOverlapMatrix is a list that contains an example of the sample overlap matrices
for five different diseases in the Kaiser GERA cohort. SampleOverlapMatrix$n11
provides the number of cases shared between all possible pairs of diseases.
SampleOverlapMatrix$n00 provides the number of controls shared between all possible
pairs of diseases. SampleOverlapMatrix$n10 provides the number of subjects who
are case for one disease and control for another disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SampleOverlapMatrix)
</code></pre>


<h3>Format</h3>

<p>A list consisting of three integer square matrices (each of dimension 5 by 5): 
</p>

<dl>
<dt>n11</dt><dd><p>number of cases shared between all possible pairs of diseases.</p>
</dd>
<dt>n00</dt><dd><p>number of controls shared between all possible pairs of diseases.</p>
</dd>
<dt>n10</dt><dd><p>number of subjects who are case for one disease and control for another disease.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(SampleOverlapMatrix)
n11 &lt;- SampleOverlapMatrix$n11
n11
n00 &lt;- SampleOverlapMatrix$n00
n00
n10 &lt;- SampleOverlapMatrix$n10
n10
estimate_corln(n11,n00,n10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
