<!DOCTYPE html><html><head><title>Help for package CpGassoc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CpGassoc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CpGassoc-package'>
<p>Association Between Methylation and a Phenotype of Interest</p></a></li>
<li><a href='#Class cpg.perm'>
<p>Methods for object of class <code>"cpg.perm"</code>.</p></a></li>
<li><a href='#cpg'><p>Methods for object of class <code>"cpg"</code></p></a></li>
<li><a href='#cpg.assoc'>
<p>Association Analysis Between Methylation Beta Values and Phenotype of Interest</p></a></li>
<li><a href='#cpg.combine'>
<p>Combine various objects of class <code>"cpg"</code></p></a></li>
<li><a href='#cpg.everything'>
<p>Multi-Task function</p></a></li>
<li><a href='#cpg.GC'>
<p>cpg.GC and methods for output of function</p></a></li>
<li><a href='#cpg.perm'>
<p>Perform a Permutation Test of the Association Between Methylation and a Phenotype of Interest</p></a></li>
<li><a href='#cpg.qc'>
<p>Performs quality control on Illumina data.</p></a></li>
<li><a href='#cpg.work'>
<p>Does the analysis between the CpG sites and phenotype of interest</p></a></li>
<li><a href='#design'>
<p>Create full and reduced design matrices for the cpg.assoc function.</p></a></li>
<li><a href='#manhattan'>
<p>Create a manhattan plot</p></a></li>
<li><a href='#manhattan.reflect'>
<p>Create a Reflective Manhattan plot</p></a></li>
<li><a href='#Other CpGassoc Functions'>
<p>Information on miscellaneous other functions</p></a></li>
<li><a href='#Sample Data CpGassoc'>
<p>Sample data from <span class="pkg">CpGassoc</span></p></a></li>
<li><a href='#scatterplot'>
<p>Plot beta values of individual CpG sites against the independent variable.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Association Between Methylation and a Phenotype of Interest</td>
</tr>
<tr>
<td>Version:</td>
<td>2.60</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-05-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Barfield, R., Conneely, K., Kilaru,V</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>R Barfield &lt;barfieldrichard8@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Is designed to test for association between methylation at CpG sites across the genome and a phenotype of interest, adjusting for any relevant covariates. The package can perform standard analyses of large datasets very quickly with no need to impute the data. It can also handle mixed effects models with chip or batch entering the model as a random intercept. Also includes tools to apply quality control filters, perform permutation tests, and create QQ plots, manhattan plots, and scatterplots for individual CpG sites.	</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), nlme, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>qvalue</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-05-30 16:47:03 UTC; richardbarfield</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-05-30 22:23:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='CpGassoc-package'>
Association Between Methylation and a Phenotype of Interest 
</h2><span id='topic+CpGassoc-package'></span><span id='topic+CpGassoc'></span>

<h3>Description</h3>

<p>Is designed to test for association between methylation at CpG sites across the genome and a phenotype of interest, 
adjusting for any relevant covariates. The package can perform standard analyses of large datasets very quickly with no 
need to impute the data. It can also handle mixed effects models with chip or batch entering the model as a random intercept. 
Also includes tools to apply quality control filters, perform permutation tests, and create QQ plots, manhattan plots, 
and scatterplots for individual CpG sites.
</p>
<p>There is a tutorial for CpGassoc available at the website &lt;genetics.emory.edu/conneely/&gt;
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: CpGassoc
Type: Package
Title: Association between Methylation and a phenotype of interest
Version: 2.60
Date: 2017-05-30
Author: Barfield, R., Conneely, K., Kilaru,V
Maintainer: R Barfield &lt;barfieldrichard8@gmail.com&gt;
Description: CpGassoc is designed to test for association between methylation at CpG sites across the genome and a phenotype of interest, adjusting for any relevant covariates. The package can perform standard analyses of large datasets very quickly with no need to impute the data. It can also handle mixed effects models with chip or batch entering the model as a random intercept. CpGassoc also includes tools to apply quality control filters, perform permutation tests, and create QQ plots, manhattan plots, and scatterplots for individual CpG sites.
Depends:nlme,methods
Suggests:qvalue
License: GPL (&gt;= 2)
</td>
</tr>

</table>

<p>CpGassoc is a suite of R functions designed to perform flexible analyses of methylation array data. 
The two main functions are  <code>cpg.assoc</code> and  <code>cpg.perm</code>.  <code>cpg.assoc</code> will perform an association test 
between the CpG sites and the phenotype of interest. Covariates can be added to the model, and can be 
continuous or categorical in nature.  <code>cpg.assoc</code> allows users to set their own false discovery rate threshold,
to transform the beta values to log(beta/(1-beta)), and to subset if required.  cpg.assoc can also fit a 
linear mixed effects model with a single random effect to control for possible technical difference due to 
batch or chip.  <code>cpg.assoc</code> uses the Holm method to determine significance. The user can also specify 
an FDR method to determine significance based on the function  <code>p.adjust</code> or  <code>qvalue</code>.  <code>cpg.perm</code> 
performs the same tasks as <code>cpg.assoc</code> followed by a permutation test on the data, repeating the analysis 
multiple times after randomly permuting the main phenotype of interest. The user can 
specify the seed and the number of permutations. If over one hundred permutations are performed 
QQ plots can be created with empirical confidence intervals based on the permuted t-statistics. 
For more information see <code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>. For more information on how to perform  <code>cpg.assoc</code> or 
<code>cpg.perm</code> see their corresponding help pages. CpGassoc can also perform quality control (see <code><a href="#topic+cpg.qc">cpg.qc</a></code>).
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+cpg.qc">cpg.qc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###NOTE: If you are dealing with large data, do not specify large.data=FALSE. 
###The default option is true
##This will involve partitioning up the data and performing more gc() to clear up space
##Using cpg.assoc:
data(samplecpg,samplepheno,package="CpGassoc")
results&lt;-cpg.assoc(samplecpg,samplepheno$weight,large.data=FALSE)
results

##Using cpg.perm:
Testperm&lt;-cpg.perm(samplecpg[1:200,],samplepheno$weight,data.frame(samplepheno$Dose),
                seed=2314,nperm=10,large.data=FALSE)
Testperm


#For more examples go to those two pages main help pages.
</code></pre>

<hr>
<h2 id='Class+20cpg.perm'>
Methods for object of class <code>"cpg.perm"</code>.
</h2><span id='topic+plot.cpg.perm'></span><span id='topic+summary.cpg.perm'></span><span id='topic+print.cpg.perm'></span><span id='topic+sort.cpg.perm'></span>

<h3>Description</h3>

<p>Methods and extra functions for class <code>"cpg.perm"</code>.
<code>plot.cpg.perm</code> creates a QQ plot based on the association p-values or t-statistics from the function <code>cpg.perm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'cpg.perm'
plot(x, save.plot = NULL, file.type = "pdf", popup.pdf = FALSE,
  main.title = NULL, eps.size = c(5, 5), tplot = FALSE, perm.ci = TRUE, classic = TRUE,
  gc.p.val = FALSE, gcdisplay = FALSE, ...)

  ## S3 method for class 'cpg.perm'
summary(object,...)

  ## S3 method for class 'cpg.perm'
print(x,...)

  ## S3 method for class 'cpg.perm'
sort(x,decreasing,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Class+2B20cpg.perm_+3A_x">x</code></td>
<td>

<p>Output from <code>cpg.perm</code>. Of class <code>"cpg.perm"</code>.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_save.plot">save.plot</code></td>
<td>

<p>Name of the file for the plot to be saved to. If not specified, plot will not be saved.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_file.type">file.type</code></td>
<td>

<p>Type of file to be saved. Can either be <code>"pdf"</code> or <code>"eps"</code>. Selecting <code>file.type="eps"</code> will
result in publication quality editable postscript files that can be opened by Adobe Illustrator or Photoshop.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_popup.pdf">popup.pdf</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>. If creating a pdf file, this indicates if the plot should appear in a popup window as well. If running in a 
cluster-like environment, best to leave <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_main.title">main.title</code></td>
<td>

<p>Main title to be put on the graph. If <code>NULL</code> one based on the analysis will be used
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_eps.size">eps.size</code></td>
<td>

<p>Vector indicating the size of .eps file (if creating one). Correponds to the options horizontal and height in the
<code>postscript</code> function.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_tplot">tplot</code></td>
<td>

<p>Logical. If <code>TRUE</code>, ordered t-statistics will be plotted against their expected quanties. If <code>FALSE</code> (default), -log(p) will be plotted. If indep is a class variable this option will be
ignored.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_perm.ci">perm.ci</code></td>
<td>

<p>Logical. If <code>TRUE</code>, the confidence intervals computed will be from the permutated values, otherwise will be based on the theoretical values.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_classic">classic</code></td>
<td>

<p>Logical. If <code>TRUE</code>, a classic qq-plot will be generated, with all p-values plotted against predicted values (including significant).
If <code>FALSE</code> Holm-significant CpG sites will not be used to compute expected quantiles and will be plotted separately.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_gc.p.val">gc.p.val</code></td>
<td>

<p>Logical. If true, plot will use the genomic control adjusted p-values.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_gcdisplay">gcdisplay</code></td>
<td>

<p>Logical.If true, plot will display the genomic control value in the legend.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_object">object</code></td>
<td>

<p>Output of class <code>"cpg.perm"</code> from <code>"cpg.perm"</code>.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_decreasing">decreasing</code></td>
<td>

<p>logical. Should the sort be increasing or decreasing? Not available for partial sorting.
</p>
</td></tr>
<tr><td><code id="Class+2B20cpg.perm_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Empirical confidence intervals will be computed only if there are a hundred or more permutations. Otherwise the theoretical confidence intervals will be plotted.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplecpg,samplepheno,package="CpGassoc")
##We will do the analysis on a subset to save time
###NOTE: If you are dealing with large data, do not specify large.data=FALSE.
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
#The qq plot:
Testperm&lt;-cpg.perm(samplecpg[1:300,],samplepheno$weight,seed=2314,nperm=10,large.data=FALSE)
plot(Testperm)
#The t-statistic plot from cpg.perm has confidence intervals since we were allowed 
#to perform permutations on the T-values.
plot(Testperm,tplot=TRUE)
#If there was 100 or more permutations, there would be emperical confidence intervals.

#Getting an example of the non classic QQ plot
plot(Testperm,classic=FALSE)


###Now for Sort
head(sort(Testperm)$results)
head(Testperm$results)



</code></pre>

<hr>
<h2 id='cpg'>Methods for object of class <code>"cpg"</code></h2><span id='topic+plot.cpg'></span><span id='topic+summary.cpg'></span><span id='topic+print.cpg'></span><span id='topic+sort.cpg'></span>

<h3>Description</h3>

<p>Methods and extra functions for class <code>"cpg"</code>.
<code>plot.cpg</code> creates a QQ plot based on the association p-values or t-statistics from the function <code>cpg.assoc</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
  ## S3 method for class 'cpg'
plot(x, save.plot = NULL, file.type = "pdf", popup.pdf = FALSE, 
                     tplot = FALSE, classic = TRUE,main.title = NULL, eps.size = c(5, 5),
                     gc.p.val = FALSE, gcdisplay = FALSE, ...)

  ## S3 method for class 'cpg'
summary(object,...)

  ## S3 method for class 'cpg'
print(x,...)

  ## S3 method for class 'cpg'
sort(x,decreasing,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg_+3A_x">x</code></td>
<td>

<p>Output of class <code>"cpg"</code> from cpg.assoc or cpg.work.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_save.plot">save.plot</code></td>
<td>

<p>Name of the file for the plot to be saved to. If not specified, plot will not be saved.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_file.type">file.type</code></td>
<td>

<p>Type of file to be saved. Can either be <code>"pdf"</code> or <code>"eps"</code>. Selecting <code>file.type="eps"</code> will
result in publication quality editable postscript files that can be opened by Adobe Illustrator or Photoshop.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_popup.pdf">popup.pdf</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>. If creating a pdf file, this indicates if the plot should appear in a popup window as well. If running in a 
cluster-like environment, best to leave <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_tplot">tplot</code></td>
<td>

<p>Logical. If <code>TRUE</code>, t-statistics will be plotted vs. their expected quantiles. If <code>FALSE</code> (default), -log(p) will be
plotted. (Note: if <code>class(x$indep)=='factor'</code> this option will be ignored.)
</p>
</td></tr>
<tr><td><code id="cpg_+3A_classic">classic</code></td>
<td>

<p>Logical. If <code>TRUE</code>, a classic qq-plot will be generated, with all p-values plotted against predicted values (including significant).
If <code>FALSE</code> Holm-significant CpG sites will not be used to compute expected quantiles and will be plotted separately.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_main.title">main.title</code></td>
<td>

<p>Main title to be put on the graph. If <code>NULL</code> one based on the analysis will be used.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_eps.size">eps.size</code></td>
<td>

<p>Vector indicating the size of .eps file (if creating one). Correponds to the options horizontal and height in the
<code>postscript</code> function.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_gc.p.val">gc.p.val</code></td>
<td>

<p>Logical. If true, plot will use the genomic control adjusted p-values.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_gcdisplay">gcdisplay</code></td>
<td>

<p>Logical.If true, plot will display the genomic control value in the legend.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_object">object</code></td>
<td>

<p>Output of class <code>"cpg"</code> from <code>cpg.assoc</code> or <code>cpg.work</code>.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_decreasing">decreasing</code></td>
<td>

<p>logical. Should the sort be increasing or decreasing? Not available for partial sorting.
</p>
</td></tr>
<tr><td><code id="cpg_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>sort.cpg</code> returns an item of class <code>"cpg"</code> that is sorted by p-value.
<code>summary.cpg</code> creates a qq-plot based on the data, and scatterplots or boxplots for the top sites.
</p>


<h3>Note</h3>

<p>Plots with empirical confidence intervals based on permutation tests can be obtained from <code>cpg.perm</code>.
See <code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code> for more info.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Using the results from the example given in cpg.assoc.
###NOTE: If you are dealing with large data, do not specify large.data=FALSE. 
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
##QQ Plot:
data(samplecpg,samplepheno,package="CpGassoc")
test&lt;-cpg.assoc(samplecpg,samplepheno$weight,large.data=FALSE)
plot(test)
##t-statistic plot:
plot(test,tplot=TRUE)


#Getting our plot:
plot(test,classic=FALSE)


##Now an example of sort
head(sort(test)$results)

##Summary
summary(test)
</code></pre>

<hr>
<h2 id='cpg.assoc'>
Association Analysis Between Methylation Beta Values and Phenotype of Interest
</h2><span id='topic+cpg.assoc'></span>

<h3>Description</h3>

<p>Association Analysis Between Methylation Beta Values and Phenotype of Interest. For more detail see Tutorial.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.assoc(beta.val, indep, covariates = NULL, data = NULL, logit.transform = FALSE, 
chip.id = NULL, subset = NULL, random = FALSE, fdr.cutoff = 0.05, large.data = TRUE,
fdr.method = "BH", logitperm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.assoc_+3A_beta.val">beta.val</code></td>
<td>

<p>A vector, matrix, or data frame containing the beta values of interest (1 row per CpG site, 1 column per individual).
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_indep">indep</code></td>
<td>

<p>A vector containing the variable to be tested for association. <code>cpg.assoc</code> will evaluate the association between the beta values (dependent
variable) and indep (independent variable).
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_covariates">covariates</code></td>
<td>

<p>A data frame consisting of additional covariates to be included in the model. covariates can also be specified as a matrix 
if it takes the form of a model matrix with no intercept column, or can be specified as a vector if there is
only one covariate of interest. Can also be a formula(e.g. <code>~cov1+cov2</code>).
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_data">data</code></td>
<td>

<p>an optional data frame, list or environment (or object coercible by as.data.frame to a data frame) containing the
variables in the model. If not found in data, the variables are taken from the environment from which cpg.assoc is
called.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_logit.transform">logit.transform</code></td>
<td>

<p>logical. If <code>TRUE</code>, the logit transform of the beta values log(beta.val/(1-beta.val)) will be used. Any values equal to zero or one will be set to the 
next smallest or  next largest value respectively; values &lt;0 or &gt;1 will be set to NA.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_chip.id">chip.id</code></td>
<td>

<p>An optional vector containing chip, batch identities, or other categorical factor of interest to the researcher. If specified, chip id will be included as a factor in the model.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_subset">subset</code></td>
<td>

<p>An optional logical vector specifying a subset of observations to be used in the fitting process.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_random">random</code></td>
<td>

<p>Logical. If <code>TRUE</code>, <span class="option">chip.id</span> will be included in the model as a random effect, and a random intercept model will be
fitted. If <code>FALSE</code>, <span class="option">chip.id</span> will be included in the model as an ordinary categorical covariate, for a much faster analysis.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>

<p>The desired FDR threshold. The default setting is .05. The set of CpG sites with FDR &lt; <span class="option">fdr.cutoff</span> will be labeled as significant.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_large.data">large.data</code></td>
<td>

<p>Logical.  Enables analyses of large datasets.  When <code>large.data=TRUE</code>, <code>cpg.assoc</code> avoids memory problems by performing the analysis in chunks. 
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_fdr.method">fdr.method</code></td>
<td>

<p>Character. Method used to calculate False Discovery Rate. Choices include any of the methods available in <code>p.adjust()</code> or
&quot;qvalue&quot; for John Storey's qvalue method (requires that <span class="pkg">qvalue</span> package is installed). The default method is &quot;BH&quot; for the Benjamini &amp; Hochberg method.
</p>
</td></tr>
<tr><td><code id="cpg.assoc_+3A_logitperm">logitperm</code></td>
<td>

<p>Logical. For internal use only.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cpg.assoc</code> is designed to test for association between an independent variable and methylation at a number of CpG sites, with the option to include additional covariates and factors. <br />
<code>cpg.assoc</code> assesses significance with the Holm (step-down Bonferroni) and FDR methods. <br />
</p>
<p>If <code>class(indep)='factor'</code>, <code>cpg.assoc</code> will perform an ANOVA test of the variable conditional on the covariates specified. Covariates, if entered, should be in the form of a data 
frame, matrix, or vector. For example, <code>covariates=data.frame(weight,age,factor(city))</code>. The data frame can also be specified prior to calling <code>cpg.assoc</code>.  The covariates 
should either be vectors or columns of a matrix or data.frame.<br />
<code>cpg.assoc</code> is also designed to deal with large data sets. Setting <code>large.data=TRUE</code> will make <code>cpg.assoc</code> split up the data to enable efficient analysis of large datasets.
</p>


<h3>Value</h3>

<p><code>cpg.assoc</code> will return an object of class <code>"cpg"</code>. The functions <code>summary</code> and <code>plot</code> can be called to get a summary of results and to create QQ plots.
</p>
<table>
<tr><td><code>results</code></td>
<td>

<p>A data frame consisting of the t or F statistics and P-values for each CpG site, as well as indicators of Holm and FDR significance. CpG sites will be in the same order as the original
input, but the <code>sort()</code> function can be used directly on the <code>cpg.assoc</code> object to sort CpG sites by p-value.
</p>
</td></tr>
<tr><td><code>Holm.sig</code></td>
<td>

<p>A list of sites that met criteria for Holm significance.
</p>
</td></tr>
<tr><td><code>FDR.sig</code></td>
<td>

<p>A data.frame of the CpG sites that were significant by the FDR method specified.
</p>
</td></tr>
<tr><td><code>info</code></td>
<td>

<p>A data frame consisting of the minimum P-value observed, the FDR method that was used, the phenotype of interest, the number of covariates in the model, the name of the matrix or data frame the methylation beta values were
taken from, the FDR cutoff value and whether a mixed effects analysis was performed.
</p>
</td></tr>
<tr><td><code>indep</code></td>
<td>

<p>The independent variable that was tested for association.
</p>
</td></tr>
<tr><td><code>covariates</code></td>
<td>

<p>Data.frame or matrix of covariates, if specified (otherwise <code>NULL</code>).
</p>
</td></tr>
<tr><td><code>chip</code></td>
<td>

<p>chip.id vector, if specified (otherwise <code>NULL</code>).
</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>

<p>A data frame consisting of the degrees of freedom, and if object is continous the intercept effect adjusted for possible covariates in the model, the estimated effect size, and the standard error. 
The degrees of freedom is used in <code><a href="#topic+plot.cpg">plot.cpg</a></code> to compute the genomic inflation factors.<br />
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barfield, R.; Conneely, K.; Kilaru,V.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+sort.cpg.perm">sort.cpg.perm</a></code>
<code><a href="#topic+sort.cpg">sort.cpg</a></code>
<code><a href="#topic+cpg.qc">cpg.qc</a></code>
<code><a href="#topic+cpg.GC">cpg.GC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Sample output from CpGassoc
###NOTE: If you are dealing with large data, do not specify large.data=FALSE. 
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
data(samplecpg,samplepheno,package="CpGassoc")
results&lt;-cpg.assoc(samplecpg,samplepheno$weight,large.data=FALSE)
results
#Analysis with covariates. There are multiple ways to do this. One can define the
#dataframe prior or do it in the function call.
test&lt;-cpg.assoc(samplecpg,samplepheno$weight,data.frame(samplepheno$Distance,
samplepheno$Dose),large.data=FALSE)
# or
covar&lt;-data.frame(samplepheno$Distance,samplepheno$Dose)
test2&lt;-cpg.assoc(samplecpg,samplepheno$weight,covar,large.data=FALSE)


#Doing a mixed effects model. This does take more time, so we will do a subset of
#the samplecpg
randtest&lt;-cpg.assoc(samplecpg[1:10,],samplepheno$weight,chip.id=samplepheno$chip,
random=TRUE,large.data=FALSE)
          </code></pre>

<hr>
<h2 id='cpg.combine'>
Combine various objects of class <code>"cpg"</code>
</h2><span id='topic+cpg.combine'></span>

<h3>Description</h3>

<p>Takes a list containing objects of class <code>"cpg"</code> and combines them into one cpg item. Assumes that there are no repeated CpG sites bewtween the various objects (i.e. analysis wasn't performed on the same sites twice).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.combine(allvalues, fdr.method="BH",fdr.cutoff=.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.combine_+3A_allvalues">allvalues</code></td>
<td>

<p>A list containing the <code>"cpg"</code> objects that are desired to be consolidated.
</p>
</td></tr>
<tr><td><code id="cpg.combine_+3A_fdr.method">fdr.method</code></td>
<td>

<p>FDR method that user wants to use. For options see the <code>cpg.assoc</code> help page.
</p>
</td></tr>
<tr><td><code id="cpg.combine_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>

<p>The desired FDR threshold. The default setting is .05. The set of CpG sites with FDR &lt; fdr.cutoff will be labeled as significant.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>info.data</code></td>
<td>
<p>An object of class <code>"cpg"</code> that is the consolidated version of the objects of class cpg that were passed in.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is designed to be used by <code>cpg.assoc</code> when it does analysis on large data sets or by the user if they split up the analysis by chromosome or some other such partition.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplecpg,samplepheno,package="CpGassoc")
###NOTE: If you are dealing with large data, do not specify large.data=FALSE. 
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
test1&lt;-cpg.assoc(samplecpg[1:100,],samplepheno$weight,large.data=FALSE)
test2&lt;-cpg.assoc(samplecpg[101:200,],samplepheno$weight,large.data=FALSE)
bigtest&lt;-list(test1,test2)
overall&lt;-cpg.combine(bigtest)
overall
      
</code></pre>

<hr>
<h2 id='cpg.everything'>
Multi-Task function   
</h2><span id='topic+cpg.everything'></span><span id='topic+cpg.everything.character'></span><span id='topic+cpg.everything.complex'></span><span id='topic+cpg.everything.logical'></span><span id='topic+cpg.everything.matrix'></span><span id='topic+cpg.everything.numeric'></span>

<h3>Description</h3>

<p>A function designed to do a group of smaller functions required for the <code>cpg.assoc</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.everything(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.everything_+3A_x">x</code></td>
<td>

<p>Just a generic object
</p>
</td></tr>
<tr><td><code id="cpg.everything_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function created to do a bunch of much smaller tasks within <span class="pkg">CpGassoc</span> based on the class of x.
</p>


<h3>Note</h3>

<p><code>cpg.everything</code> is designed to perform a multitude of smaller tasks for <code>cpg.assoc</code> that do not warrant a full
help page. These include:warnings, getting the name of the independent variable, designing the random function
to be used, and getting the names for the values returned.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Has four methods:character,complex, numeric/matrix, and logical
#They correspon to getting the indep variable name, warnings, getting the random function,
#and getting the names for the values returned. For the design of these functions see
#the R code. 
</code></pre>

<hr>
<h2 id='cpg.GC'>
cpg.GC and methods for output of function
</h2><span id='topic+cpg.GC'></span><span id='topic+print.cpg.gc'></span><span id='topic+print.cpg.perm.gc'></span>

<h3>Description</h3>

<p>cpg.GC accepts an object of class <code>"cpg.perm"</code> or <code>"cpg"</code> and returns information
regarding Holm and FDR-significance of the GC (genomic control) adjusted test statistics. For <code>"cpg.perm"</code>
will return permutation p-values based on the GC-adjusted values from each permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   cpg.GC(x)

  ## S3 method for class 'cpg.gc'
print(x,...)
  
  ## S3 method for class 'cpg.perm.gc'
print(x,...)
 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.GC_+3A_x">x</code></td>
<td>

<p>Object of class <code>"cpg.perm"</code> or <code>"cpg"</code>. 
</p>
</td></tr>
<tr><td><code id="cpg.GC_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cpg.GC</code> will display the number of Holm and FDR-significant sites using the genomic control adjusted p-values
test statistics. It will also display the estimated genomic control inflation factor.
</p>


<h3>Value</h3>

<p><code>cpg.GC</code> returns an object of class <code>"cpg.gc"</code> or <code>"cpg.perm.gc"</code>
</p>
<table>
<tr><td><code>gc.results</code></td>
<td>

<p>Matrix consisting of GC-adjusted test statistics for each CpG site. Similar to the results output of <code>cpg.assoc</code>. 
</p>
</td></tr>
<tr><td><code>gc.info</code></td>
<td>

<p>Data frame with information on the number of Holm and FDR significant sites. Will also have the genomic control
inflation estimate. Objects from <code>"cpg.perm"</code> will also have information concerning the permutation p-values.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+sort.cpg.perm">sort.cpg.perm</a></code>
<code><a href="#topic+sort.cpg">sort.cpg</a></code>
<code><a href="#topic+cpg.qc">cpg.qc</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplecpg,samplepheno,package="CpGassoc")
results&lt;-cpg.assoc(samplecpg,samplepheno$weight,large.data=FALSE)

cpg.GC(results)
##If the genomic inflation factor is less than one there is no need for adjustment

</code></pre>

<hr>
<h2 id='cpg.perm'>
Perform a Permutation Test of the Association Between Methylation and a Phenotype of Interest
</h2><span id='topic+cpg.perm'></span>

<h3>Description</h3>

<p>Calls <code>cpg.assoc</code> to get the observed P-values from the study and then performs a user-specified number of permutations to calculate an emperical p-value.
In addition to the same test statistics computed by <code>cpg.assoc</code>, <code>cpg.perm</code> will compute the permutation p-values for the observed p-value, the number of Holm significant sites, and the number of FDR significant sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.perm(beta.values, indep, covariates = NULL, nperm, data = NULL, seed = NULL,
logit.transform = FALSE, chip.id = NULL, subset = NULL, random = FALSE,
fdr.cutoff = 0.05, fdr.method = "BH",large.data=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.perm_+3A_beta.values">beta.values</code></td>
<td>

<p>A vector, matrix, or data frame containing the beta values of interest (1 row per CpG site, 1 column per individual).
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_indep">indep</code></td>
<td>

<p>A vector containing the main variable of interest.  <code>cpg.assoc</code> will evaluate the association between indep and the beta values. 
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_covariates">covariates</code></td>
<td>

<p>A data frame consisting of the covariates of interest. covariates can also be a matrix if it is a model matrix minus the intercept column. 
It can also be a vector if there is only one covariate of interest.  Can also be a formula(e.g. <code>~cov1+cov2</code>).
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_nperm">nperm</code></td>
<td>

<p>The number of permutations to be performed.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_data">data</code></td>
<td>

<p>an optional data frame, list or environment (or object coercible by <code>as.data.frame</code> to a data frame) containing the variables in the model. If not found in data, the variables are taken from the environment from which <code>cpg.perm</code> is called.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_seed">seed</code></td>
<td>

<p>The required seed for random number generation. If not input, will use R's internal seed.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_logit.transform">logit.transform</code></td>
<td>

<p>logical. If <code>TRUE</code>, the logit transform of the beta values log(beta.val/(1-beta.val)) will be used. Any values equal to zero or one will be set to the 
next smallest or  next largest value respectively; values &lt;0 or &gt;1 will be set to NA.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_chip.id">chip.id</code></td>
<td>

<p>An optional vector containing chip, batch identities, or other categorical factor of interest to the researcher. If specified, chip id will be included as a factor in the model.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_subset">subset</code></td>
<td>

<p>An optional logical vector specifying a subset of observations to be used in the fitting process.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_random">random</code></td>
<td>

<p>logical. If <code>TRUE</code>, the <span class="option">chip.id</span> will be processed as a random effect, and a random intercept model will be fitted.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>

<p>The threshold at which to compare the FDR values. The default setting is .05. Any FDR values less than .05 will be considered significant.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_fdr.method">fdr.method</code></td>
<td>

<p>Character. Method used to calculate False Discovery Rate. Can be any of the methods listed in <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> or &quot;qvalue&quot; 
for John Storey's qvalue method (required to have <span class="pkg">qvalue</span> package installed). The default method is &quot;BH&quot; for the Benjamini &amp; Hochberg method.
</p>
</td></tr>
<tr><td><code id="cpg.perm_+3A_large.data">large.data</code></td>
<td>

<p>Logical.  Enables analyses of large datasets.  When <code>large.data=TRUE</code>, <code>cpg.assoc</code> avoids memory problems by performing the analysis in chunks. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The item returned will be of class <code>"cpg.perm"</code>. It will contain all of the values of class cpg (<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>) and a few more:
</p>
<table>
<tr><td><code>permutation.matrix</code></td>
<td>

<p>A matrix consisting of the minimum observed P-value, the number of Holm significant CpG sites, and the number of FDR significant sites for each permutation.
</p>
</td></tr>
<tr><td><code>gc.permutation.matrix</code></td>
<td>

<p>Similar to the permutation.matrix only in relation to the genomic control adjusted p-values.
</p>
</td></tr>
<tr><td><code>perm.p.values</code></td>
<td>

<p>A data frame consisting of the permutation P-values, and the number of permutations performed.
</p>
</td></tr>
<tr><td><code>perm.tstat</code></td>
<td>

<p>If one hundred or more permutations were performed and indep is a continuous variable, consists of the quantile .025 and .975 of observed t-statistcs for each permutation, ordered from smallest to largest.
perm.tstat is used by <code>plot.cpg.perm</code> to compute the confidence intervals for the QQ plot of t-statistics. Otherwise <code>NULL</code>.
</p>
</td></tr>
<tr><td><code>perm.pval</code></td>
<td>

<p>If one hundred or more permutations were performed, consists of the observed p-values for each permutation, ordered from smallest to largest. perm.pval is usd by <code>plot.cpg.perm</code> to compute
the confidence intervals for the QQ plot of the p-values. Otherwise NULL.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barfield, R.; Conneely, K.; Kilaru,V.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+sort.cpg.perm">sort.cpg.perm</a></code>
<code><a href="#topic+sort.cpg">sort.cpg</a></code>
<code><a href="#topic+cpg.qc">cpg.qc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##Loading the data
data(samplecpg,samplepheno,package="CpGassoc")
###NOTE: If you are dealing with large data, do not specify large.data=FALSE.
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
#Performing a permutation 10 times
Testperm&lt;-cpg.perm(samplecpg[1:200,],samplepheno$weight,seed=2314,nperm=10,large.data=FALSE)
Testperm
#All the contents of CpGassoc are included in the output from Testperm

#summary function works on objects of class cpg.perm
summary(Testperm)

</code></pre>

<hr>
<h2 id='cpg.qc'>
Performs quality control on Illumina data.
</h2><span id='topic+cpg.qc'></span>

<h3>Description</h3>

<p><code>cpg.qc</code> is designed to perform quality control on Illumina data prior to analysis. 
In addition to the matrix of beta values, this function requires as input matrices of Signal A, Signal B, and detection p-values.
It will remove samples that have low intensity (mean signal intensity less than half of the overall median or 2000). 
It can also set to NA datapoints with detection p-values exceeding a user-specified cutoff, and can remove samples or 
sites that have a missing rate above a user-specified value. 
Finally, users can opt to compute beta values as M/(U+M) or M/(U+M+100).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.qc(beta.orig,siga,sigb,pval,p.cutoff=.001,cpg.miss=NULL,sample.miss=NULL,
constant100=FALSE,sig.return=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.qc_+3A_beta.orig">beta.orig</code></td>
<td>

<p>The original beta values matrix obtained from GenomeStudio.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_siga">siga</code></td>
<td>

<p>The unmethylated signals matrix obtained from GenomeStudio.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_sigb">sigb</code></td>
<td>

<p>The methylated signals matrix obtained from GenomeStudio.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_pval">pval</code></td>
<td>

<p>A matrix of detection p-values obtained from GenomeStudio. 
pval should have the same dimension as the beta values and signals: one row for each site and one column for each individual.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_p.cutoff">p.cutoff</code></td>
<td>

<p>The user-specified cutoff for detection p-values (default=.001).
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_cpg.miss">cpg.miss</code></td>
<td>

<p>Optional cutoff value. If specified, cpg.qc will remove cpg sites where the proportion of missing values exceeds this cutoff.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_sample.miss">sample.miss</code></td>
<td>

<p>Optional cutoff value. If specified, cpg.qc will remove samples where the proportion of missing values exceeds this cutoff.
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_constant100">constant100</code></td>
<td>

<p>Logical. If true, the new beta values will be calculated as M/(U+M+100); if false (default) they will be calculated as M/(U+M).
</p>
</td></tr>
<tr><td><code id="cpg.qc_+3A_sig.return">sig.return</code></td>
<td>

<p>Logical. If true, cpg.qc returns the qced signal data as well.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important that all the matrices or data frames listed above (<span class="option">pval</span>, <span class="option">siga</span>, <span class="option">sigb</span>, <span class="option">beta.orig</span>) 
are ordered similarly with respect to samples and CpG sites.
</p>


<h3>Value</h3>

<p><code>cpg.qc</code> returns a new matrix of beta values that has been subjected to the specified quality control filters.  This matrix can be input directly into <code>cpg.assoc</code>.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Conneely, K.; Kilaru,V.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
</p>

<hr>
<h2 id='cpg.work'>
Does the analysis between the CpG sites and phenotype of interest
</h2><span id='topic+cpg.work'></span>

<h3>Description</h3>

<p>Association Analysis Between Methylation Beta Values and Phenotype of Interest. This function contains the code that does the brunt of the work for <code>cpg.assoc</code> and <code>cpg.perm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.work(beta.values, indep, covariates = NULL, data = NULL, logit.transform = FALSE,
chip.id = NULL, subset = NULL, random = FALSE, fdr.cutoff = 0.05, callarge = FALSE,
fdr.method = "BH", logitperm = FALSE,big.split=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpg.work_+3A_beta.values">beta.values</code></td>
<td>

<p>A vector, matrix, or data frame containing the beta values of interest (1 row per CpG site, 1 column per individual).
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_indep">indep</code></td>
<td>

<p>A vector containing the main variable of interest.  <code>cpg.work</code> will evaluate the association between indep and the beta values. 
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_covariates">covariates</code></td>
<td>

<p>A data frame consisting of the covariates of interest. covariates can also be a matrix if it is a model matrix minus the intercept column. 
It can also be a vector if there is only one covariate of interest. Can also be a formula (e.g. <code>~cov1+cov2</code>).
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_data">data</code></td>
<td>

<p>an optional data frame, list or environment (or object coercible by <code>as.data.frame</code> to a data frame) containing the variables in the model. If not found in data, the variables are taken from the environment from which <code>cpg.work</code> is called.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_logit.transform">logit.transform</code></td>
<td>

<p>logical. If <code>TRUE</code>, the logit transform of the beta values log(beta.val/(1-beta.val)) will be used. Any values equal to zero or one will be set to the 
next smallest or  next largest value respectively; values &lt;0 or &gt;1 will be set to NA.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_chip.id">chip.id</code></td>
<td>

<p>An optional vector containing chip, batch identities, or other categorical factor of interest to the researcher. If specified, chip id will be included as a factor in the model.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_subset">subset</code></td>
<td>

<p>an optional logical vector specifying a subset of observations to be used in the fitting process.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_random">random</code></td>
<td>

<p>logical. If <code>TRUE</code>, the <span class="option">chip.id</span> will be included in the model as a random effect, and a random intercept model will be fitted. If <code>FALSE</code>, <span class="option">chip.id</span> will be included
in the model as an ordinary categorical covariate, for a much faster analysis.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>

<p>The threshold at which to compare the FDR values. The default setting is .05. Any FDR values less than .05 will be considered significant.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_callarge">callarge</code></td>
<td>

<p>logical. Used by <code>cpg.assoc</code> when it calls <code>cpg.work</code>. If <code>TRUE</code> it means that beta.values is actually split up from a larger data set and that <code>memory.limit</code> may be a problem.
This tells <code>cpg.work</code> to perform more <code>rm()</code> and <code>gc()</code> to clear up space.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_fdr.method">fdr.method</code></td>
<td>

<p>Character.Method used to calculate False Discovery Rate. Can be any of the methods listed in <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> or &quot;qvalue&quot; 
for John Storey's qvalue method (required to have <span class="pkg">qvalue</span> package installed). The default method is &quot;BH&quot; for the Benjamini &amp; Hochberg method.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_logitperm">logitperm</code></td>
<td>

<p>Passes from <code>cpg.perm</code> when permutation test is performed. Stops from future checks involving the logistic transformation.
</p>
</td></tr>
<tr><td><code id="cpg.work_+3A_big.split">big.split</code></td>
<td>

<p>Passes from <code>cpg.assoc</code>. Internal flag to inform <code>cpg.work</code> that the large data did not need to be split up.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cpg.work</code> does the analysis between the methylation and the phenotype of interest. It is called by <code>cpg.assoc</code> to do the brunt of the work. It can be called itself with the same input as <code>cpg.assoc</code>, it just cannot handle large data sets.
</p>


<h3>Value</h3>

<p><code>cpg.work</code> will return an object of class <code>"cpg"</code>.
The functions summary and plot can be called to get a summary of results and to create QQ plots. The output is in the same order as the original input. To sort it by p-value, use the <code>sort</code> function.
</p>
<table>
<tr><td><code>results</code></td>
<td>

<p>A data frame consisting of the statistics and P-values for each CpG site. Also has the adjusted p-value based on the fdr.method and whether the site was Holm significant.
</p>
</td></tr>
<tr><td><code>Holm.sig</code></td>
<td>

<p>A list of sites that met criteria for Holm significance.
</p>
</td></tr>
<tr><td><code>FDR.sig</code></td>
<td>

<p>A data.frame of the sites that were FDR significant by the fdr method. 
</p>
</td></tr>
<tr><td><code>info</code></td>
<td>

<p>A data frame consisting of the minimum P-value observed, the fdr method used, what the phenotype of interest was, and the number of covariates in the model.
</p>
</td></tr>
<tr><td><code>indep</code></td>
<td>

<p>The main phenotype of interest.
</p>
</td></tr>
<tr><td><code>covariates</code></td>
<td>

<p>If covariates was non <code>NULL</code>, the covariates will be included.Otherwise will be <code>NULL</code>.
</p>
</td></tr>
<tr><td><code>chip</code></td>
<td>

<p>If chip.id was non <code>NULL</code>, the chip will be included. Otherwise will be <code>NULL</code>.
</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>

<p>A data frame consisting of the degrees of freedom, and if object is continous the intercept effect adjusted for possible covariates in the model, the estimated effect size, and the standard error. 
The degrees of freedom is used in <code><a href="#topic+plot.cpg">plot.cpg</a></code> to compute the genomic inflation factors.<br />
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+sort.cpg.perm">sort.cpg.perm</a></code>
<code><a href="#topic+sort.cpg">sort.cpg</a></code>
<code><a href="#topic+cpg.qc">cpg.qc</a></code>
<code><a href="#topic+cpg.GC">cpg.GC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##See the examples listed in cpg.assoc for ways in which to use cpg.work.
##Just change the cpg.assoc to cpg.work.
</code></pre>

<hr>
<h2 id='design'>
Create full and reduced design matrices for the cpg.assoc function.
</h2><span id='topic+design'></span><span id='topic+design.data.frame'></span><span id='topic+design.NULL'></span>

<h3>Description</h3>

<p>Designed to be used by <code>cpg.assoc</code> and <code>cpg.perm</code>. Creates a full and reduced design matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design(covariates, indep, chip.id, random)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="design_+3A_covariates">covariates</code></td>
<td>

<p>A data frame consisting of the covariates of interest. covariates can also be a matrix if it is a model matrix minus the intercept column. 
It can also be a vector if there is only one covariate of interest.
If no covariates must be specified as <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="design_+3A_indep">indep</code></td>
<td>

<p>A vector containing the main variable of interest.  <code>cpg.assoc</code> will evaluate the association between indep and the beta values.
</p>
</td></tr>
<tr><td><code id="design_+3A_chip.id">chip.id</code></td>
<td>

<p>An optional vector containing chip or batch identities. If specified, <code>chip.id</code> will be included as a factor in the model.
</p>
</td></tr>
<tr><td><code id="design_+3A_random">random</code></td>
<td>

<p>Is the model going to be a mixed effects. If so, <code>chip.id</code> will not be included in the design matrices.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing the full and reduced design matrices.
</p>
<table>
<tr><td><code>full</code></td>
<td>
<p>The full design matrix.</p>
</td></tr>
<tr><td><code>reduced</code></td>
<td>
<p>The reduced design matrix.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The <code>design</code> function  is designed to be used exclusively by the <code>cpg.assoc</code> and <code>cpg.perm</code> functions.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.work">cpg.work</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.combine">cpg.combine</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(samplecpg,samplepheno,package="CpGassoc")
#Example where there are covariates:
covar&lt;-data.frame(samplepheno$weight,samplepheno$Distance)
test&lt;-design(covar,samplepheno$SBP,samplepheno$chip,FALSE)
dim(test$full)
dim(test$reduced)
test$reduced[1:5,1:5]
test$full[1:5,1:5]
#When no covariates or chip.id:
test2&lt;-design(NULL,samplepheno$SBP,NULL,FALSE)
dim(test2$full)
dim(test2$reduced)
</code></pre>

<hr>
<h2 id='manhattan'>
Create a manhattan plot
</h2><span id='topic+manhattan'></span>

<h3>Description</h3>

<p>This function will produce a manhattan plot for the observed P-values from a object of class <code>"cpg"</code> or <code>"cpg.perm"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan(x, cpgname, chr, pos, save.plot = NULL, file.type="pdf", 
popup.pdf = FALSE, eps.size = c(15, 5), main.title = NULL, cpg.labels = NULL,
chr.list = NULL, color.list = NULL, point.size = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan_+3A_x">x</code></td>
<td>

<p>Object of class <code>"cpg"</code> or <code>"cpg.perm"</code>.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_cpgname">cpgname</code></td>
<td>

<p>A vector consisting of the labels for each CpG site.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chr">chr</code></td>
<td>

<p>A vector consisting of the chromosome number for each CpG site.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_pos">pos</code></td>
<td>

<p>The map position of each CpG site within its chromosome.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_save.plot">save.plot</code></td>
<td>

<p>Name of the file for the plot to be saved to. If not specified, plot will not be saved. 
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_file.type">file.type</code></td>
<td>

<p>Type of file to be saved. Can either be <code>"pdf"</code> or <code>"eps"</code>. Selecting <code>file.type="eps"</code> will
result in publication quality editable postscript files that can be opened by Adobe Illustrator or Photoshop.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_popup.pdf">popup.pdf</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>. If creating a pdf file, this indicates if the plot should appear in a popup window as well. If running in a 
cluster-like environment, best to leave <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_eps.size">eps.size</code></td>
<td>

<p>Vector indicating the size of .eps file (if creating one). Corresponds to horrizontal and height.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_main.title">main.title</code></td>
<td>

<p>Main title to be put on the graph. If <code>NULL</code> one based on the analysis will be used.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_cpg.labels">cpg.labels</code></td>
<td>

<p>A character scalar of either <code>"FDR"</code> or <code>"HOLM"</code> which will label the significant sites on the manhattan plot.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_chr.list">chr.list</code></td>
<td>

<p>A vector listing the chromosomes to be plotted (all available chromosomes are plotted by default). The X and Y chromosomes
can be denoted by 23 and 24
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_color.list">color.list</code></td>
<td>

<p>A vector of custom colors to be used for each chromosomes in the manhattan plot.
</p>
</td></tr>
<tr><td><code id="manhattan_+3A_point.size">point.size</code></td>
<td>

<p>The Size of the points in the manhattan plot, if NULL, default to our default, where significant probes have different sizes.  
</p>
</td></tr>  
<tr><td><code id="manhattan_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p><span class="option">cpgname</span>, <span class="option">chr</span>, and <span class="option">pos</span> must be sorted in the same order, so that the first cpgname[1] corresponds to chr[1] and pos[1], and so on. 
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+manhattan.reflect">manhattan.reflect</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Doing a Manhattan plot. First load the data:
###NOTE: If you are dealing with large data, do not specify large.data=FALSE. 
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
data(samplecpg,samplepheno,annotation,package="CpGassoc")

examplemanhat&lt;-cpg.assoc(samplecpg,samplepheno$Disease,large.data=FALSE)

manhattan(examplemanhat,annotation$TargetID,annotation$CHR,annotation$MAPINFO)


</code></pre>

<hr>
<h2 id='manhattan.reflect'>
Create a Reflective Manhattan plot
</h2><span id='topic+manhattan.reflect'></span>

<h3>Description</h3>

<p>This function will produce a reflective manhattan plot for the observed P-values from an object of class <code>"cpg"</code> or <code>"cpg.perm"</code>.
The original analysis needs to be performed on a continuous variable (need T-statistics).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan.reflect(x, cpgname, chr, pos, save.plot = NULL, file.type="pdf",
popup.pdf = FALSE, eps.size = c(15, 5), main.title = NULL, cpg.labels = NULL,
chr.list = NULL, color.list = NULL, fdr.cutoff=NULL, point.size=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan.reflect_+3A_x">x</code></td>
<td>

<p>Object of class <code>"cpg"</code> or <code>"cpg.perm"</code>.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_cpgname">cpgname</code></td>
<td>

<p>A vector consisting of the labels for each CpG site.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_chr">chr</code></td>
<td>

<p>A vector consisting of the chromosome number for each CpG site.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_pos">pos</code></td>
<td>

<p>The map position of each CpG site within its chromosome.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_save.plot">save.plot</code></td>
<td>

<p>Name of the file for the plot to be saved to. If not specified, plot will not be saved.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_file.type">file.type</code></td>
<td>

<p>Type of file to be saved. Can either be <code>"pdf"</code> or <code>"eps"</code>. Selecting <code>file.type="eps"</code> will
result in publication quality editable postscript files that can be opened by Adobe Illustrator or Photoshop.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_popup.pdf">popup.pdf</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>. If creating a pdf file, this indicates if the plot should appear in a popup window as well. If running in a
cluster-like environment, best to leave <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_eps.size">eps.size</code></td>
<td>

<p>Vector indicating the size of .eps file (if creating one). Corresponds to horrizontal and height.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_main.title">main.title</code></td>
<td>

<p>Main title to be put on the graph. If <code>NULL</code> one based on the analysis will be used.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_cpg.labels">cpg.labels</code></td>
<td>

<p>A character scalar of either <code>"FDR"</code> or <code>"HOLM"</code> which will label the significant sites on the manhattan plot.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_chr.list">chr.list</code></td>
<td>

<p>A vector listing the chromosomes to be plotted (all available chromosomes are plotted by default). The X and Y chromosomes
can be denoted by 23 and 24
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_color.list">color.list</code></td>
<td>

<p>A vector of custom colors to be used for each chromosomes in the manhattan plot.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>

<p>A numeric scalar between 0 and 1 to indicate what to consider FDR significant. Defaults to <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="manhattan.reflect_+3A_point.size">point.size</code></td>
<td>

<p>The Size of the points in the manhattan plot, if NULL, default to our default, where significant probes have different sizes.  
</p>
</td></tr>  
<tr><td><code id="manhattan.reflect_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p><span class="option">cpgname</span>, <span class="option">chr</span>, and <span class="option">pos</span> must be sorted in the same order, so that the first cpgname[1] corresponds to chr[1] and pos[1], and so on.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Doing a Manhattan plot. First load the data:
###NOTE: If you are dealing with large data, do not specify large.data=FALSE.
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
data(samplecpg,samplepheno,annotation,package="CpGassoc")

examplemanhat&lt;-cpg.assoc(samplecpg,samplepheno$Disease,large.data=FALSE)

manhattan.reflect(examplemanhat,annotation$TargetID,annotation$CHR,annotation$MAPINFO)


</code></pre>

<hr>
<h2 id='Other+20CpGassoc+20Functions'>
Information on miscellaneous other functions
</h2><span id='topic+cpg.length'></span><span id='topic+pointsizefunction'></span><span id='topic+cpgassocsummary'></span>

<h3>Description</h3>

<p><code>cpg.length</code> compares the dimensions of the covariates, the independent phenotype, chip, and the matrix of beta values from
<code>cpg.assoc</code> or <code>cpg.perm</code>. If the number of individuals does not match up. Stops the code.
</p>
<p><code>pointsizefunction</code> simply scales the size of the points for the qq-plot for <code>plot.cpg</code> and <code>plot.cpg.perm</code>
</p>
<p><code>cpgassocsummary</code> is used by <code>cpg.work</code> when the phenotypes contains such small amount of variance such that matrix methods
can not be used. Takes an object of class <code>"aov"</code> or <code>"mlm"</code> and gets the test statistics from these objects while avoiding doing a loop
or the list structures which is typically done if one does summary on one of these objects. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpg.length(indep, numpatients, covariates, chip.id)

pointsizefunction(x)

cpgassocsummary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_indep">indep</code></td>
<td>

<p>object <code>indep</code> from <code>cpg.assoc</code>, <code>cpg.work</code> or <code>cpg.perm</code>.
</p>
</td></tr>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_numpatients">numpatients</code></td>
<td>

<p>The number of column of the object beta.values in <code>cpg.work</code> or <code>cpg.perm</code> or number of columns of beta.val in <code>cpg.assoc</code>
</p>
</td></tr>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_covariates">covariates</code></td>
<td>

<p>object covariate from <code>cpg.assoc</code>, <code>cpg.work</code> or <code>cpg.perm</code>.
</p>
</td></tr>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_chip.id">chip.id</code></td>
<td>

<p>object <code>chip.id</code> from <code>cpg.assoc</code>, <code>cpg.work</code> or <code>cpg.perm</code>.
</p>
</td></tr>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_x">x</code></td>
<td>

<p>the p-values from the object <code>"cpg"</code> or <code>"cpg.perm"</code> 
</p>
</td></tr>
<tr><td><code id="Other+2B20CpGassoc+2B20Functions_+3A_object">object</code></td>
<td>

<p>An object of <code>"aov"</code> or <code>"mlm"</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cpg.length</code> stops the functions if the dimensions do not add up. Used Internally by <span class="pkg">CpGassoc</span>
</p>
<p><code>pointsizefunction</code> simply returns a vector of equal length of x to be used in plot for the cex option.
</p>
<p><code>cpgassocsummary</code> returns a matrix of columns two and three of the <code>results</code> value of an <code>"cpg"</code> or <code>"cpg.perm"</code> object. This is for the cpg sites with non-missing data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>

<hr>
<h2 id='Sample+20Data+20CpGassoc'>
Sample data from <span class="pkg">CpGassoc</span>
</h2><span id='topic+samplecpg'></span><span id='topic+samplepheno'></span><span id='topic+annotation'></span>

<h3>Description</h3>


<ul>
<li><p>samplecpgMatrix containing sample (fake) methylation data for 258 individuals over 1228 CpG sites
</p>
</li>
<li><p>samplephenoMatrix with phenotype info for the 258 individuals
</p>
</li>
<li><p>annotationMatrix with annotation information for the 1228 CpG sites
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(samplecpg)</code></pre>


<h3>Format</h3>


<ul>
<li><p>samplecpgMatrix, dimensions 1228 x 258
</p>
</li>
<li><p>samplephenoMatrix, dimensions 258 x 6. Phenotype info on <code>"Dose"</code>, <code>"SBP"</code>, <code>"Distance"</code>, <code>"weight"</code>, <code>"Disease"</code>, and <code>"chip"</code>
</p>
</li>
<li><p>annotationMatrix, dimensions 1228 x 3. Header names are <code>"TargetID"</code>, <code>"CHR"</code>, and <code>"MAPINFO"</code>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+scatterplot">scatterplot</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##See help pages for other functions for usage of these datasets
</code></pre>

<hr>
<h2 id='scatterplot'>
Plot beta values of individual CpG sites against the independent variable.
</h2><span id='topic+scatterplot'></span>

<h3>Description</h3>

<p>Plot beta values of individual CpG sites against the independent variable. Can create scatterplots and boxplots. If scatterplots the intercept will be adjusted
for any covariates that were included in the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatterplot(x, cpg.rank = NULL, cpg.name = NULL, save.plot = NULL, file.type="pdf", 
eps.size = c(5, 5), popup.pdf = FALSE, beta.values = NULL,main.title=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatterplot_+3A_x">x</code></td>
<td>

<p>Object of class <code>"cpg"</code> or <code>"cpg.perm"</code>.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_cpg.rank">cpg.rank</code></td>
<td>

<p>A vector listing the rank of sites to be plotted. The rank is based on the ordered p-values.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_cpg.name">cpg.name</code></td>
<td>

<p>A character vector containing the names of CpG sites to be plotted against the phenotype of interest. This option is ignored if <span class="option">cpg.rank</span> is specified.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_save.plot">save.plot</code></td>
<td>

<p>Prefix of the filename for the plot(s) to be saved to. If specified, plot filenames will be created by appending this prefix to  
either cpg.rank or cpg.name.  If not specified, plot will not be saved.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_file.type">file.type</code></td>
<td>

<p>Type of file to be saved. Can either be <code>"pdf"</code> or <code>"eps"</code>. Selecting <code>file.type="eps"</code> will
result in publication quality editable postscript files that can be opened by Adobe Illustrator or Photoshop.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_eps.size">eps.size</code></td>
<td>

<p>Vector indicating the size of .eps file (if creating one). Correponds to horrizontal and height.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_popup.pdf">popup.pdf</code></td>
<td>

<p><code>TRUE</code> or <code>FALSE</code>. If creating a pdf file, this indicates if the plot should appear in a popup window as well. If running in a 
cluster-like environment, best to leave <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_beta.values">beta.values</code></td>
<td>

<p>If the object has been renamed (i.e. <code>x$info$betainfo</code> is no longer in <code>ls(.GlobalEnv)</code>) then specify the new object here.
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_main.title">main.title</code></td>
<td>

<p>Main title to be put on the graph. If <code>NULL</code> one based on the analysis will be used
</p>
</td></tr>
<tr><td><code id="scatterplot_+3A_...">...</code></td>
<td>

<p>Arguments to be passed to methods, such as graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An unlimited number of CpG sites can be selected for plotting by specifying either <span class="option">cpg.rank</span> or <span class="option">cpg.name</span>, as shown in the Examples below.  Note that only one of these options is needed; if  both are entered, <span class="option">cpg.rank</span> will be used.
</p>


<h3>Author(s)</h3>

<p>Barfield, R.; Kilaru,V.; Conneely, K.<br />
Maintainer: R. Barfield: &lt;rbarfield01@fas.harvard.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cpg.assoc">cpg.assoc</a></code>
<code><a href="#topic+plot.cpg">plot.cpg</a></code>
<code><a href="#topic+manhattan">manhattan</a></code>
<code><a href="#topic+cpg.perm">cpg.perm</a></code>
<code><a href="#topic+plot.cpg.perm">plot.cpg.perm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load the data:
data(samplecpg,samplepheno,package="CpGassoc")
###NOTE: If you are dealing with large data, do not specify large.data=FALSE.
###The default option is true.
##This will involve partitioning up the data and performing more gc() to clear up space
test&lt;-cpg.assoc(samplecpg,samplepheno$weight,large.data=FALSE)
##Using rank, will plot the top three sites in order of significance:
scatterplot(test,c(1:3))
##Using name, specify three sites:
scatterplot(test,cpg.name=c("CpG1182","CpG1000","CpG42"))

##Plotting something that is categorical in nature:
test2&lt;-cpg.assoc(samplecpg[1:200,],factor(samplepheno$Disease),large.data=FALSE)
scatterplot(test2,c(2),beta.values=samplecpg[1:200,])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
