<!DOCTYPE html><html><head><title>Help for package CpmERCCutoff</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CpmERCCutoff}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#exp_input'><p>A data frame of expected ERCC1 and ERCC2 ratios</p></a></li>
<li><a href='#getLowLcpmCutoff'><p>Function to empirically determine a log2 CPM cutoff based on ERCC RNA spike-in</p></a></li>
<li><a href='#mta_dta'><p>A data frame containing sample-level ERCC meta data</p></a></li>
<li><a href='#obs_input'><p>A data frame of observed spike in ERCC normalized LCPM data</p></a></li>
<li><a href='#plot.empLCPM'><p>Plot the empirically derived LCPM cutoff</p></a></li>
<li><a href='#print.empLCPM'><p>Print the empirically derived LCPM cutoff results</p></a></li>
<li><a href='#summary.empLCPM'><p>Summarizing the empirically derived LCPM cutoff</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculation of Log2 Counts per Million Cutoff from ERCC Controls</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the empirical method to derive log2 counts per million (CPM) cutoff to filter out lowly expressed genes using ERCC spike-ins as described in Goll and Bosinger et.al (2022)&lt;<a href="https://doi.org/10.1101%2F2022.06.23.497396">doi:10.1101/2022.06.23.497396</a>&gt;. This package utilizes the synthetic mRNA control pairs developed by the External RNA Controls Consortium (ERCC) (ERCC 1 / ERCC 2) that are spiked into sample pairs at known ratios at various absolute abundances.  The relationship between the observed and expected fold changes is then used to empirically determine an optimal log2 CPM cutoff for filtering out lowly expressed genes.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-12 18:36:42 UTC; ksteenbergen</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Author:</td>
<td>Tyler Grimes <a href="https://orcid.org/0000-0002-5653-3418"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Travis L. Jensen <a href="https://orcid.org/0000-0002-0322-0469"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Kristen Steenbergen [ctb, cre],
  The Emmes Company LLC [cph] (Copyright holder of CpmERCCutoff package
    (Copyright (C) 2022)),
  Johannes B. Goll <a href="https://orcid.org/0000-0002-9968-4080"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kristen Steenbergen &lt;ksteenbergen@emmes.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-13 11:20:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='exp_input'>A data frame of expected ERCC1 and ERCC2 ratios</h2><span id='topic+exp_input'></span>

<h3>Description</h3>

<p>A data frame that contains the expected spike in ERCC data.  This data can be obtained
from 'ERCC Controls Analysis' manual located on Thermo Fisher's ERCC RNA
Spike-In Mix product
[page](https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt).
The 'exp_input' data frame mirrors the fields shown in the ERCC manual.
For the LCPM cutoff calculation, the last column of the log2 expected fold
change ratios are used.  Ensure that this column is titled
&quot;expected_lfc_ratio&quot;. See the example code below for formatting the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_input
</code></pre>


<h3>Format</h3>

<p>A data frame with 92 rows and 4 columns: Each row represents an ERCC
transcript.  Columns are described below:
</p>

<dl>
<dt>ercc_id</dt><dd><p>ERCC spike-in mRNA Ids (ERCC-00002 &ndash; ERCC-00171)</p>
</dd>
<dt>subgroup</dt><dd><p>ERCC subgroups (A &ndash; D)</p>
</dd>
<dt>ercc1_conc</dt><dd><p>ERCC1 concentration (0.014 &ndash; 30,000)</p>
</dd>
<dt>ercc2_conc</dt><dd><p>ERCC2 concentration (0.007 &ndash; 30,000)</p>
</dd>
<dt>expected_fc_ratio</dt><dd><p>Expected fold change ratio (.5 &ndash; 4)</p>
</dd>
<dt>expected_lfc_ratio</dt><dd><p>Expected log2 fold change ratio (-1 &ndash; 2)</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'># Order rows by ERCC ID and assign to rownames.
exp_input = exp_input[order(exp_input$ercc_id), ]
rownames(exp_input) = exp_input$ercc_id

</code></pre>

<hr>
<h2 id='getLowLcpmCutoff'>Function to empirically determine a log2 CPM cutoff based on ERCC RNA spike-in</h2><span id='topic+getLowLcpmCutoff'></span>

<h3>Description</h3>

<p>This function uses spike-in ERCC data, known control RNA probes,
and paired samples to fit a 3rd order polynomial to determine an expression
cutoff that meets the specified correlation between expected and observed
fold changes.  The <code>obs</code> data frame used as input for the observed
expression of the 92 ERCC RNA spike-ins and stores the coverage-normalized
read log2 counts per million (LCPM) that mapped to the respective ERCC
sequences.  Typically, prior to LCPM calculation, the read count data is
normalized for any systematic differences in read coverage between samples,
for example, by using the TMM normalization method as implemented in
the <code>edgeR</code> package.
</p>
<p>For each bootstrap replicate, the paired samples are subsampled with
replacement.  The mean LCPM of each ERCC transcript is determined by
first calculating the average LCPM value for each paired sample, and
then taking the mean of those averages. The ERCC transcripts are sorted
based on these means, and are then grouped into <code>n.bins</code> ERCC bins.
Next, the Spearman correlation metric is used to calculate the association
between the empirical and expected log fold change (LFC) of the ERCCs in
each bin for each sample.
Additionally, the average LCPM for the ERCCs in each bin are calculated
for each sample. This leads to a pair of values - the average LCPM and the
association value - for each sample and each ERCC bin.  Outliers within
each ERCC bin are identified and removed based on &gt;1.5 IQR.
A 3rd order polynomial is fit with the explanatory variable being the
average LCPM and the response variable being the Spearman correlation
value between expected and observed log2 fold changes.
The fitted curve is used to identify the average LCPM value with a Spearman
correlation of <code>cor.value</code>. The results are output as an &quot;empLCM&quot;
object as described below.  The <code><a href="#topic+summary.empLCPM">summary.empLCPM</a></code> function can
be used to extract a summary of the results, and the
<code><a href="#topic+plot.empLCPM">plot.empLCPM</a></code> function to plot the results for visualization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLowLcpmCutoff(
  obs,
  exp,
  pairs,
  n.bins = 7,
  rep = 1000,
  ci = 0.95,
  cor.value = 0.9,
  remove.outliers = TRUE,
  seed = 20220719
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLowLcpmCutoff_+3A_obs">obs</code></td>
<td>
<p>A data frame  of observed spike-in ERCC data.  Each row is an ERCC
transcript, and each column is a sample.  Data are read
coverage-normalized log2 counts per million (LCPM).</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_exp">exp</code></td>
<td>
<p>A data frame of expected ERCC Mix 1 and Mix 2 ratios with a column
titled 'expected_lfc_ratio' containing the expected log2 fold-change
ratios. This data can be obtained from 'ERCC Controls Analysis' manual
located on Thermo Fisher's ERCC RNA Spike-In Mix product
[page](https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt).
The 'exp_input' data frame mirrors the fields shown in the ERCC manual.
For the LCPM cutoff calculation, the last column containing the log2
expected fold change ratios are used.  Ensure that this column is titled
&quot;expected_lfc_ratio&quot;. See the example code below for formatting the data.     #</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_pairs">pairs</code></td>
<td>
<p>A 2-column data frame where each row indicates a sample pair
with the first column indicating the sample that received ERCC spike-ins
from Mix 1 and the second column indicating the sample receiving Mix 2.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_n.bins">n.bins</code></td>
<td>
<p>Integer.  The number of abundance bins to create.  Default is 7.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_rep">rep</code></td>
<td>
<p>Integer.  The number of bootstrap replicates.  Default is 1000.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_ci">ci</code></td>
<td>
<p>Numeric.  The confidence interval.  Default is 0.95.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_cor.value">cor.value</code></td>
<td>
<p>Numeric.  The desired Spearman correlation between the
empirical log2 fold change across the ERCC transcripts.  Default is 0.9.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_remove.outliers">remove.outliers</code></td>
<td>
<p>If TRUE (default) outliers are identified as exceeding
1.5 IQR, and are removed prior to fitting the polynomial. Set to FALSE
to keep all points.</p>
</td></tr>
<tr><td><code id="getLowLcpmCutoff_+3A_seed">seed</code></td>
<td>
<p>Integer.  The reproducibility seed.  Default is 20220719.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An &quot;empLCPM&quot; object is returned, which contains the following named
elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
   <code>cutoff</code> </td><td style="text-align: left;"> a vector containing 3 values: the threshold value,
   upper confidence interval, </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> and the lower confidence interval value. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>args</code> </td><td style="text-align: left;"> a key: value list of arguments that were provided. </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>res</code> </td><td style="text-align: left;"> a list containing the main results and other
   information from the input. </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> The <code><a href="#topic+summary.empLCPM">summary.empLCPM</a></code>
   function should be used to extract a summary table. </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td>
</tr>

</table>



<h3>See Also</h3>

<p><code><a href="#topic+summary.empLCPM">summary.empLCPM</a></code>, <code><a href="#topic+plot.empLCPM">plot.empLCPM</a></code>,
<code><a href="#topic+print.empLCPM">print.empLCPM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(CpmERCCutoff)
##############################
# Load and wrangle input data:
##############################
# Load observed read counts
data("obs_input")

# Set ERCC Ids to rownames
rownames(obs_input) = obs_input$X

# Load expected ERCC data:
data("exp_input")

# Order rows by ERCC ID.
exp_input = exp_input[order(exp_input$ercc_id), ]
rownames(exp_input) = exp_input$ercc_id

# Load metadata:
data("mta_dta")

# Pair samples that received ERCC Mix 1 with samples that received ERCC Mix 2.
# The resulting 2-column data frame is used for the 'pairs' argument.
# Note: the code here will depend on the details of the given experiment. In
#       this example, the post-vaccination samples (which received Mix 2)
#       for each subject are paired to their pre-vaccination samples (which
#       received Mix 1).
pairs_input = cbind(
  mta_dta[mta_dta$spike == 2, 'samid'],
  mta_dta[match(mta_dta[mta_dta$spike == 2, 'subid'],
                mta_dta[mta_dta$spike == 1,'subid']), 'samid'])
# Put Mix 1 in the first column and Mix 2 in the second.
pairs_input = pairs_input[, c(2, 1)]

###############################
# Run getLowLcpmCutoff Function:
###############################'
# Note: Here we use `rep = 10` for only 10 bootstrap replicates
#       to decrease the run time for this example; a lager number
#       should be used in practice (default = 1000).
res = getLowLcpmCutoff(obs = obs_input,
                       exp = exp_input,
                       pairs = pairs_input,
                       n.bins = 7,
                       rep = 10,
                       cor.value = 0.9,
		                  remove.outliers = TRUE,
                       seed = 20220719)

# Print a short summary of the results:
res

# Extract a summary table of the results:
summary(res)

# Create a plot of the results:
plot(x = res,
     main = "Determination of Empirical Minimum Expression Cutoffs using ERCCs",
     col.trend = "blue",
     col.outlier = c("black", "red"))

</code></pre>

<hr>
<h2 id='mta_dta'>A data frame containing sample-level ERCC meta data</h2><span id='topic+mta_dta'></span>

<h3>Description</h3>

<p>A data frame containing sample-level ERCC meta data.  In this experiment,
subjects had repeated measures and the baseline samples were spiked with
ERCC1 and four post-vaccination samples were each spiked with ERCC2.   This
meta data is used to create the 2-column data frame of sample pairs where the
first column will contain sample IDs that received ERCC1, and the second
column will contain sample IDs that received ERCC2.  The pairs data frame is
used as input to the 'pairs' argument in the 'getLowCpmCutOff' function.  Use
'?getLowCpmcutOff' to review example code regarding the formatting of the
'pairs' data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mta_dta
</code></pre>


<h3>Format</h3>

<p>A data frame with 49 rows and 4 columns: Each row is a sample, and each
column contains metadata such as subject IDs, spike in control type, and
treatment groups.  For this study, data was collected at various time points,
however, under different experiment conditions, the 'day' column can be
represented as treatment group.
</p>

<dl>
<dt>samid</dt><dd><p>Sample IDs (SAM01 &ndash; SAM49)</p>
</dd>
<dt>subid</dt><dd><p>Subject/Participant ID (SUB01 &ndash; SUB10)</p>
</dd>
<dt>day</dt><dd><p>collection day (0 &ndash; 14)</p>
</dd>
<dt>spike</dt><dd><p>ERCC spike in control Mix (1 or 2)</p>
</dd>
</dl>


<hr>
<h2 id='obs_input'>A data frame of observed spike in ERCC normalized LCPM data</h2><span id='topic+obs_input'></span>

<h3>Description</h3>

<p>A data frame of observed gene expression results for ERCC RNA that was spiked
into samples.  Data are read coverage-normalized log2 counts per million (LCPM).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obs_input
</code></pre>


<h3>Format</h3>

<p>A data frame with 92 rows and 50 Columns: Each row is an ERCC
transcript, and each column is a sample.  Data are read
coverage-normalized LCPM.
</p>

<dl>
<dt>X</dt><dd><p>This first column is ERCC spike in mRNA Ids (ERCC-00002 &ndash; ERCC-00171)</p>
</dd>
<dt>SAM15</dt><dd><p>Sample 15 containing normalized log2 counts per million</p>
</dd>
<dt>SAM36</dt><dd><p>Sample 36 containing normalized log2 counts per million</p>
</dd>
<dt>SAM19</dt><dd><p>Sample 19 containing normalized log2 counts per million</p>
</dd>
<dt>SAM53</dt><dd><p>Sample 53 containing normalized log2 counts per million</p>
</dd>
<dt>SAM42</dt><dd><p>Sample 42 containing normalized log2 counts per million</p>
</dd>
<dt>SAM32</dt><dd><p>Sample 32 containing normalized log2 counts per million</p>
</dd>
<dt>SAM18</dt><dd><p>Sample 18 containing normalized log2 counts per million</p>
</dd>
<dt>SAM48</dt><dd><p>Sample 48 containing normalized log2 counts per million</p>
</dd>
<dt>SAM26</dt><dd><p>Sample 26 containing normalized log2 counts per million</p>
</dd>
<dt>SAM37</dt><dd><p>Sample 37 containing normalized log2 counts per million</p>
</dd>
<dt>SAM38</dt><dd><p>Sample 38 containing normalized log2 counts per million</p>
</dd>
<dt>SAM29</dt><dd><p>Sample 29 containing normalized log2 counts per million</p>
</dd>
<dt>SAM17</dt><dd><p>Sample 17 containing normalized log2 counts per million</p>
</dd>
<dt>SAM41</dt><dd><p>Sample 41 containing normalized log2 counts per million</p>
</dd>
<dt>SAM09</dt><dd><p>Sample 09 containing normalized log2 counts per million</p>
</dd>
<dt>SAM07</dt><dd><p>Sample 07 containing normalized log2 counts per million</p>
</dd>
<dt>SAM14</dt><dd><p>Sample 14 containing normalized log2 counts per million</p>
</dd>
<dt>SAM02</dt><dd><p>Sample 02 containing normalized log2 counts per million</p>
</dd>
<dt>SAM05</dt><dd><p>Sample 05 containing normalized log2 counts per million</p>
</dd>
<dt>SAM25</dt><dd><p>Sample 25 containing normalized log2 counts per million</p>
</dd>
<dt>SAM08</dt><dd><p>Sample 08 containing normalized log2 counts per million</p>
</dd>
<dt>SAM28</dt><dd><p>Sample 28 containing normalized log2 counts per million</p>
</dd>
<dt>SAM44</dt><dd><p>Sample 44 containing normalized log2 counts per million</p>
</dd>
<dt>SAM04</dt><dd><p>Sample 04 containing normalized log2 counts per million</p>
</dd>
<dt>SAM10</dt><dd><p>Sample 10 containing normalized log2 counts per million</p>
</dd>
<dt>SAM31</dt><dd><p>Sample 31 containing normalized log2 counts per million</p>
</dd>
<dt>SAM21</dt><dd><p>Sample 21 containing normalized log2 counts per million</p>
</dd>
<dt>SAM20</dt><dd><p>Sample 20 containing normalized log2 counts per million</p>
</dd>
<dt>SAM52</dt><dd><p>Sample 52 containing normalized log2 counts per million</p>
</dd>
<dt>SAM46</dt><dd><p>Sample 46 containing normalized log2 counts per million</p>
</dd>
<dt>SAM01</dt><dd><p>Sample 01 containing normalized log2 counts per million</p>
</dd>
<dt>SAM13</dt><dd><p>Sample 13 containing normalized log2 counts per million</p>
</dd>
<dt>SAM39</dt><dd><p>Sample 39 containing normalized log2 counts per million</p>
</dd>
<dt>SAM49</dt><dd><p>Sample 49 containing normalized log2 counts per million</p>
</dd>
<dt>SAM30</dt><dd><p>Sample 30 containing normalized log2 counts per million</p>
</dd>
<dt>SAM50</dt><dd><p>Sample 50 containing normalized log2 counts per million</p>
</dd>
<dt>SAM11</dt><dd><p>Sample 11 containing normalized log2 counts per million</p>
</dd>
<dt>SAM35</dt><dd><p>Sample 35 containing normalized log2 counts per million</p>
</dd>
<dt>SAM06</dt><dd><p>Sample 06 containing normalized log2 counts per million</p>
</dd>
<dt>SAM27</dt><dd><p>Sample 27 containing normalized log2 counts per million</p>
</dd>
<dt>SAM33</dt><dd><p>Sample 33 containing normalized log2 counts per million</p>
</dd>
<dt>SAM22</dt><dd><p>Sample 22 containing normalized log2 counts per million</p>
</dd>
<dt>SAM24</dt><dd><p>Sample 24 containing normalized log2 counts per million</p>
</dd>
<dt>SAM16</dt><dd><p>Sample 16 containing normalized log2 counts per million</p>
</dd>
<dt>SAM34</dt><dd><p>Sample 34 containing normalized log2 counts per million</p>
</dd>
<dt>SAM03</dt><dd><p>Sample 03 containing normalized log2 counts per million</p>
</dd>
<dt>SAM47</dt><dd><p>Sample 47 containing normalized log2 counts per million</p>
</dd>
<dt>SAM40</dt><dd><p>Sample 40 containing normalized log2 counts per million</p>
</dd>
<dt>SAM23</dt><dd><p>Sample 23 containing normalized log2 counts per million</p>
</dd>
</dl>


<hr>
<h2 id='plot.empLCPM'>Plot the empirically derived LCPM cutoff</h2><span id='topic+plot.empLCPM'></span>

<h3>Description</h3>

<p>Plot method for class &quot;empLCPM&quot; that plots the 3rd order polynomial fit for
the Spearman correlation between the expected and observed ERCC fold changes
across log2 CPM abundance bins and highlights the empirically derived LCPM
cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'empLCPM'
plot(x, main = "", col.trend = "purple", col.outlier = c("black", "red"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.empLCPM_+3A_x">x</code></td>
<td>
<p>A 'empLCPM' object from <code><a href="#topic+getLowLcpmCutoff">getLowLcpmCutoff</a></code>.</p>
</td></tr>
<tr><td><code id="plot.empLCPM_+3A_main">main</code></td>
<td>
<p>An (optional) title for the plot.</p>
</td></tr>
<tr><td><code id="plot.empLCPM_+3A_col.trend">col.trend</code></td>
<td>
<p>Color used for the 3rd order polynomial fit.</p>
</td></tr>
<tr><td><code id="plot.empLCPM_+3A_col.outlier">col.outlier</code></td>
<td>
<p>A vector specifying the default color for the points
in the scatterplot, and the color for the outlier points. The default is
to color all non-outlier points black and the outliers red.</p>
</td></tr>
<tr><td><code id="plot.empLCPM_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function <code>plot.empLCPM</code> creates a plot that summarizes 3rd
order polynomial fit for the Spearman correlation between the expected and
observed ERCC fold changes across log2 CPM abundance bins.  Vertical lines
are used to indicate the optimal cutoff value, along with the lower and upper
95% bootstrap confidence interval highlighted by dashed vertical lines.
</p>

<hr>
<h2 id='print.empLCPM'>Print the empirically derived LCPM cutoff results</h2><span id='topic+print.empLCPM'></span>

<h3>Description</h3>

<p>Print method for class &quot;empLCPM&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'empLCPM'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.empLCPM_+3A_x">x</code></td>
<td>
<p>A 'empLCPM' object from <code><a href="#topic+getLowLcpmCutoff">getLowLcpmCutoff</a></code>.</p>
</td></tr>
<tr><td><code id="print.empLCPM_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function <code>print.empLCPM</code> extracts the key results and
returns a string of the summary.
</p>

<hr>
<h2 id='summary.empLCPM'>Summarizing the empirically derived LCPM cutoff</h2><span id='topic+summary.empLCPM'></span>

<h3>Description</h3>

<p>Summary method for class &quot;empLCPM&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'empLCPM'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.empLCPM_+3A_object">object</code></td>
<td>
<p>A 'empLCPM' object from <code><a href="#topic+getLowLcpmCutoff">getLowLcpmCutoff</a></code>.</p>
</td></tr>
<tr><td><code id="summary.empLCPM_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function <code>summary.empLCPM</code> extracts the key results and
returns a string of the summary.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
