<!DOCTYPE html><html><head><title>Help for package crispRdesignR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {crispRdesignR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#crispRdesignRUI'><p>UI caller for crispRdesignR</p></a></li>
<li><a href='#Doench_2016_processing'><p>Donech 2016 Processsing</p></a></li>
<li><a href='#FindsgRNAfunction'>
<p>sgRNA target design for Shiny App</p></a></li>
<li><a href='#getofftargetdata'><p>Off Target Data Frame Creation</p></a></li>
<li><a href='#getsgRNAdata'><p>sgRNA Data Frame Creation</p></a></li>
<li><a href='#sgRNA_design'><p>sgRNA Target Design</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Guide Sequence Design for CRISPR/Cas9</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.7</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Description:</td>
<td>Designs guide sequences for CRISPR/Cas9 genome editing and 
    provides information on sequence features pertinent to guide 
    efficiency. Sequence features include annotated off-target 
    predictions in a user-selected genome and a predicted efficiency 
    score based on the model described in Doench et al. (2016) 
    &lt;<a href="https://doi.org/10.1038%2Fnbt.3437">doi:10.1038/nbt.3437</a>&gt;. Users are able to import additional genomes 
    and genome annotation files to use when searching and annotating 
    off-target hits. All guide sequences and off-target data can be 
    generated through the 'R' console with sgRNA_Design() or through 
    'crispRdesignR's' user interface with crispRdesignRUI(). CRISPR 
    (Clustered Regularly Interspaced Short Palindromic Repeats) and the 
    associated protein Cas9 refer to a technique used in genome editing.</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/dylanbeeber/crispRdesignR&amp;gt;">https://github.com/dylanbeeber/crispRdesignR&gt;</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, gbm, GenomicRanges, BiocGenerics, IRanges,
GenomeInfoDb, S4Vectors, rtracklayer, stringr, vtreat, shiny,
DT</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BSgenome.Scerevisiae.UCSC.sacCer3</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-11 20:33:17 UTC; admin</td>
</tr>
<tr>
<td>Author:</td>
<td>Dylan Beeber [aut, cre],
  Frederic Chain [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dylan Beeber &lt;dylan.beeber@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-12 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='crispRdesignRUI'>UI caller for crispRdesignR</h2><span id='topic+crispRdesignRUI'></span>

<h3>Description</h3>

<p>Activates the shiny UI for the crispRdesignR package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crispRdesignRUI(max_gtf_size = 150)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crispRdesignRUI_+3A_max_gtf_size">max_gtf_size</code></td>
<td>

<p>The maximum size (in MB) of the geneome annotation file (.gtf) that can be used with the shiny App. By default this is set to 150.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called to initiate user interface.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>


<h3>Examples</h3>

<pre><code class='language-R'>requireNamespace("gbm", quietly = TRUE)
requireNamespace("Biostrings", quietly = TRUE)
if (interactive()) {
  crispRdesignRUI()
  }
</code></pre>

<hr>
<h2 id='Doench_2016_processing'>Donech 2016 Processsing</h2><span id='topic+Doench_2016_processing'></span>

<h3>Description</h3>

<p>Warning: This function is not designed to be directly called by the user. This function is used internally in sgRNA_design() and sgRNA_design_function().
</p>
<p>Internal function that encodes all sgRNA sequence information into a data frame. This data frame is then used in conjunction with the Rule_Set_2_Model to predict effciency scores for the generated sgRNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Doench_2016_processing(seqlist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Doench_2016_processing_+3A_seqlist">seqlist</code></td>
<td>

<p>A list of 30-mer sgRNA (as a character string) with the sgRNA sequence spanning from positions 5 to 24.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing processed data on the presence of relevant sequence features to the Rule_Set_2_Model for effciency scoring. Includes information on single nucleotide positions, dinucleotide positions, single nucleotide count, dinucleotide count, GC count, PAM neighboring nucleotides, and melting temperatures. Single nulceotide positions, dinucleotide positions, and PAM neighboring nucleotides are all one-hot encoded.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>

<hr>
<h2 id='FindsgRNAfunction'>
sgRNA target design for Shiny App</h2><span id='topic+sgRNA_design_function'></span>

<h3>Description</h3>

<p>Warning: This function should not be directly called by the user - it must be called though RunShiny.R
</p>
<p>Designs sgRNA based on inputs provided in the Shiny App.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sgRNA_design_function(userseq, genomename, gtf,
designprogress, userPAM, calloffs, annotateoffs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindsgRNAfunction_+3A_userseq">userseq</code></td>
<td>

<p>The target sequence to generate sgRNA guides for. Can either be a character sequence containing DNA bases or the name of a fasta file in the working directory.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_genomename">genomename</code></td>
<td>

<p>The name of a geneome (from the BSgenome package) to check off-targets for.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_gtf">gtf</code></td>
<td>
<p>The name of a genome annotation file (.gtf) in the working directory to check off-target sequences against.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_designprogress">designprogress</code></td>
<td>
<p>Assists in communicating the progress of the sgRNA design to the Shiny App.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_userpam">userPAM</code></td>
<td>
<p>An optional argument used to set a custom PAM for the sgRNA. If not set, the function will default to the &quot;NGG&quot; PAM. Warning: Doench efficieny scores are only accurate for the &quot;NGG&quot; PAM.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_calloffs">calloffs</code></td>
<td>
<p>If TRUE, the function will search for off-targets in the genome chosen specified by the genomename argument. If FALSE, off-target calling will be skipped.
</p>
</td></tr>
<tr><td><code id="FindsgRNAfunction_+3A_annotateoffs">annotateoffs</code></td>
<td>
<p>If TRUE, the function will provide annotations for the off-targets called using the genome annotation file specified by the gtfname argument. If FALSE, off-target annotation will be skipped.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all data on the generated sgRNA and all off-target information. List items 1 through 15 include information on each individual sgRNA, including the sgRNA sequence itself, PAM, location, direction relative to the target sequence, GC content, homopolymer presence, presence of self-complementarity, off-target matches, predicted efficiency score, and a notes column that summarizes unfavorable sequence features. List items 16 through 27 include all information on off-target matches, including the original sgRNA sequence, off-target sequence, chromosome, location, direction relative to the target sequence, number of mismatches, gene ID, gene name, type of DNA, and exon number.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>

<hr>
<h2 id='getofftargetdata'>Off Target Data Frame Creation</h2><span id='topic+getofftargetdata'></span>

<h3>Description</h3>

<p>Will provide a data frame with all information about the generated sgRNA returned by the sgRNA_design function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getofftargetdata(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getofftargetdata_+3A_x">x</code></td>
<td>
<p>the data list generated by the sgRNA_design function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing all information on potential off-target sequences generated by the sgRNA_design function. Information includes the original sgRNA sequence, off-target sequence, chromosome, location, direction relative to the target sequence, number of mismatches, gene ID, gene name, type of DNA, and exon number.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Quick example without off-target searching or annotation
## First generate data with the sgRNA_Design Function
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- "placeholder"
gtfname &lt;- "placeholder"
alldata &lt;- sgRNA_design(testseq, usergenome, gtfname, calloffs = FALSE)
## Then separate and format the off-target data with getofftargetdata()
final_data &lt;- getofftargetdata(alldata)


## Longer example with off-target searching and annotation
## First generate data with the sgRNA_Design Function
requireNamespace("BSgenome.Scerevisiae.UCSC.sacCer3", quietly = TRUE)
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- BSgenome.Scerevisiae.UCSC.sacCer3::BSgenome.Scerevisiae.UCSC.sacCer3
gtfname &lt;- "Saccharomyces_cerevisiae.R64-1-1.92.gtf.gz"
annotation_file &lt;- system.file("example_data", gtfname, package = "crispRdesignR")
alldata &lt;- sgRNA_design(testseq, usergenome, annotation_file)
## Then separate and format the sgRNA data with getofftargetdata()
final_data &lt;- getofftargetdata(alldata)

</code></pre>

<hr>
<h2 id='getsgRNAdata'>sgRNA Data Frame Creation</h2><span id='topic+getsgRNAdata'></span>

<h3>Description</h3>

<p>Will provide a data frame with all information about the generated sgRNA returned by the sgRNA_design function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getsgRNAdata(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getsgRNAdata_+3A_x">x</code></td>
<td>
<p>the data list generated by the sgRNA_design function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing all information specific to sgRNA sequences generated by the sgRNA_design function. Information includes the sgRNA sequence itself, PAM, location, direction relative to the target sequence, GC content, homopolymer presence, presence of self-complementarity, off-target matches, predicted efficiency score, and a notes column that summarizes unfavorable sequence features.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Quick example without off-target searching or annotation
## First generate data with the sgRNA_Design Function
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- "placeholder"
gtfname &lt;- "placeholder"
alldata &lt;- sgRNA_design(testseq, usergenome, gtfname, calloffs = FALSE)
## Then separate and format the sgRNA data with getsgRNAdata()
final_data &lt;- getsgRNAdata(alldata)


## Longer example with off-target searching and annotation
## First generate data with the sgRNA_Design Function
requireNamespace("BSgenome.Scerevisiae.UCSC.sacCer3", quietly = TRUE)
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- BSgenome.Scerevisiae.UCSC.sacCer3::BSgenome.Scerevisiae.UCSC.sacCer3
gtfname &lt;- "Saccharomyces_cerevisiae.R64-1-1.92.gtf.gz"
annotation_file &lt;- system.file("example_data", gtfname, package = "crispRdesignR")
alldata &lt;- sgRNA_design(testseq, usergenome, annotation_file)
## Then separate and format the sgRNA data with getsgRNAdata()
final_data &lt;- getsgRNAdata(alldata)

</code></pre>

<hr>
<h2 id='sgRNA_design'>sgRNA Target Design</h2><span id='topic+sgRNA_design'></span>

<h3>Description</h3>

<p>sgRNA_design returns information to design sgRNA sequences based on a given target sequence, a genome to annotate off-target information, and a genome annoation file (.gtf), to annotate the off-target findings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sgRNA_design(userseq, genomename, gtfname, userPAM, calloffs = TRUE, annotateoffs = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sgRNA_design_+3A_userseq">userseq</code></td>
<td>

<p>The target sequence to generate sgRNA guides for. Can either be a character sequence containing DNA bases or the name of a fasta file in the working directory.
</p>
</td></tr>
<tr><td><code id="sgRNA_design_+3A_genomename">genomename</code></td>
<td>

<p>The name of a geneome (from the BSgenome package) to check off-targets for.
</p>
</td></tr>
<tr><td><code id="sgRNA_design_+3A_gtfname">gtfname</code></td>
<td>
<p>The name of a genome annotation file (.gtf) in the working directory to check off-target sequences against.
</p>
</td></tr>
<tr><td><code id="sgRNA_design_+3A_userpam">userPAM</code></td>
<td>
<p>An optional argument used to set a custom PAM for the sgRNA. If not set, the function will default to the &quot;NGG&quot; PAM. Warning: Doench efficieny scores are only accurate for the &quot;NGG&quot; PAM.
</p>
</td></tr>
<tr><td><code id="sgRNA_design_+3A_calloffs">calloffs</code></td>
<td>
<p>If TRUE, the function will search for off-targets in the genome chosen specified by the genomename argument. If FALSE, off-target calling will be skipped.
</p>
</td></tr>
<tr><td><code id="sgRNA_design_+3A_annotateoffs">annotateoffs</code></td>
<td>
<p>If TRUE, the function will provide annotations for the off-targets called using the genome annotation file specified by the gtfname argument. If FALSE, off-target annotation will be skipped.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Important Note: When designing sgRNA for large genomes (billions of base pairs), use short query DNA sequences (under 500 bp). Depending on your hardware checking for off-targets can be quite computationally intensive and may take several hours if not limited to smaller query sequences.
</p>


<h3>Value</h3>

<p>A list containing all data on the generated sgRNA and all off-target information. List items 1 through 15 include information on each individual sgRNA, including the sgRNA sequence itself, PAM, location, direction relative to the target sequence, GC content, homopolymer presence, presence of self-complementarity, off-target matches, predicted efficiency score, and a notes column that summarizes unfavorable sequence features. List items 16 through 27 include all information on off-target matches, including the original sgRNA sequence, off-target sequence, chromosome, location, direction relative to the target sequence, number of mismatches, gene ID, gene name, type of DNA, and exon number.
</p>


<h3>Author(s)</h3>

<p>Dylan Beeber
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Quick example without off-target searching or annotation
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- "placeholder"
gtfname &lt;- "placeholder"
alldata &lt;- sgRNA_design(testseq, usergenome, gtfname, calloffs = FALSE)


## Designing guide RNA for a target region as a test string, using
## the Saccharomyces Cerevisiae genome and genome annotation file:
requireNamespace("BSgenome.Scerevisiae.UCSC.sacCer3", quietly = TRUE)
testseq &lt;- "GGCAGAGCTTCGTATGTCGGCGATTCATCTCAAGTAGAAGATCCTGGTGCAGTAGG"
usergenome &lt;- BSgenome.Scerevisiae.UCSC.sacCer3::BSgenome.Scerevisiae.UCSC.sacCer3
gtfname &lt;- "Saccharomyces_cerevisiae.R64-1-1.92.gtf.gz"
annotation_file &lt;- system.file("example_data", gtfname, package = "crispRdesignR")
alldata &lt;- sgRNA_design(testseq, usergenome, annotation_file)

## Designing guide RNA for a target region as a text file, using
## the Saccharomyces Cerevisiae genome and genome annotation file,
## while switching genome annotation off:
testseq &lt;- system.file("example_data", "ExampleDAK1seq.txt", package = "crispRdesignR")
alldata2 &lt;- sgRNA_design(testseq, usergenome, annotation_file, annotateoffs = FALSE)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
