<!DOCTYPE html><html><head><title>Help for package CRMetrics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CRMetrics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addDetailedMetricsInner'><p>Add detailed metrics</p></a></li>
<li><a href='#addPlotStats'><p>Add statistics to plot</p></a></li>
<li><a href='#addPlotStatsSamples'><p>Add statistics to plot</p></a></li>
<li><a href='#addSummaryMetrics'><p>Add summary metrics</p></a></li>
<li><a href='#checkCompGroup'><p>Set correct 'comp.group' parameter</p></a></li>
<li><a href='#checkCompMeta'><p>Check whether 'comp.group' is in metadata</p></a></li>
<li><a href='#checkDataPath'><p>Check data path</p></a></li>
<li><a href='#createUniqueCellNames'><p>Create unique cell names</p></a></li>
<li><a href='#CRMetrics'><p>CRMetrics class object</p></a></li>
<li><a href='#filterVector'><p>Create filtering vector</p></a></li>
<li><a href='#getH5Paths'><p>Get H5 file paths</p></a></li>
<li><a href='#labelsFilter'><p>Get labels for percentage of filtered cells</p></a></li>
<li><a href='#percFilter'><p>Calculate percentage of filtered cells</p></a></li>
<li><a href='#plotGeom'><p>Plot the data as points, as bars as a histogram, or as a violin</p></a></li>
<li><a href='#read10x'><p>Load 10x count matrices</p></a></li>
<li><a href='#read10xH5'><p>Read 10x HDF5 files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Cell Ranger Output Filtering and Metrics Visualization</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Sample and cell filtering as well as visualisation of output metrics from 'Cell Ranger' by Grace X.Y. Zheng et al. (2017) &lt;<a href="https://doi.org/10.1038%2Fncomms14049">doi:10.1038/ncomms14049</a>&gt;. 'CRMetrics' allows for easy plotting of output metrics across multiple samples as well as comparative plots including statistical assessments of these. 'CRMetrics' allows for easy removal of ambient RNA using 'SoupX' by Matthew D Young and Sam Behjati (2020) &lt;<a href="https://doi.org/10.1093%2Fgigascience%2Fgiaa151">doi:10.1093/gigascience/giaa151</a>&gt; or 'CellBender' by Stephen J Fleming et al. (2022) &lt;<a href="https://doi.org/10.1101%2F791699">doi:10.1101/791699</a>&gt;. Furthermore, it is possible to preprocess data using 'Pagoda2' by Nikolas Barkas et al. (2021) <a href="https://github.com/kharchenkolab/pagoda2">https://github.com/kharchenkolab/pagoda2</a> or 'Seurat' by Yuhan Hao et al. (2021) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2021.04.048">doi:10.1016/j.cell.2021.04.048</a>&gt; followed by embedding of cells using 'Conos' by Nikolas Barkas et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41592-019-0466-z">doi:10.1038/s41592-019-0466-z</a>&gt;. Finally, doublets can be detected using 'scrublet' by Samuel L. Wolock et al. (2019) &lt;<a href="https://doi.org/10.1016%2Fj.cels.2018.11.005">doi:10.1016/j.cels.2018.11.005</a>&gt; or 'DoubletDetection' by Gayoso et al. (2020) &lt;<a href="https://doi.org/10.5281%2Fzenodo.2678041">doi:10.5281/zenodo.2678041</a>&gt;. In the end, cells are filtered based on user input for use in downstream applications.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cowplot, dplyr, ggbeeswarm, ggplot2, ggpmisc, ggpubr, ggrepel,
magrittr, Matrix, methods, R6, scales, sccore,
sparseMatrixStats, stats, tibble, tidyr, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>conos, data.table, markdown, pagoda2, reticulate, rhdf5,
Seurat, SoupX, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/khodosevichlab/CRMetrics">https://github.com/khodosevichlab/CRMetrics</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/khodosevichlab/CRMetrics/issues">https://github.com/khodosevichlab/CRMetrics/issues</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rasmus Rydbirk &lt;rrydbirk@bmb.sdu.dk&gt;</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-01 08:32:46 UTC; ucloud</td>
</tr>
<tr>
<td>Author:</td>
<td>Rasmus Rydbirk [aut, cre],
  Fabienne Kick [aut],
  Henrietta Holze [aut],
  Xian Xin [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-01 09:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='addDetailedMetricsInner'>Add detailed metrics</h2><span id='topic+addDetailedMetricsInner'></span>

<h3>Description</h3>

<p>Add detailed metrics, requires to load raw count matrices using pagoda2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addDetailedMetricsInner(cms, verbose = TRUE, n.cores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addDetailedMetricsInner_+3A_cms">cms</code></td>
<td>
<p>List containing the count matrices.</p>
</td></tr>
<tr><td><code id="addDetailedMetricsInner_+3A_verbose">verbose</code></td>
<td>
<p>Print messages (default = TRUE).</p>
</td></tr>
<tr><td><code id="addDetailedMetricsInner_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores for the calculations (default = 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='addPlotStats'>Add statistics to plot</h2><span id='topic+addPlotStats'></span>

<h3>Description</h3>

<p>Use ggpubr to add statistics to plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addPlotStats(
  p,
  comp.group,
  metadata,
  h.adj = 0.05,
  primary.test,
  secondary.test,
  exact = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addPlotStats_+3A_p">p</code></td>
<td>
<p>Plot to add statistics to.</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_comp.group">comp.group</code></td>
<td>
<p>Comparison metric.</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_metadata">metadata</code></td>
<td>
<p>Metadata for samples.</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_h.adj">h.adj</code></td>
<td>
<p>Position of statistics test p value as % of max(y) (default = 0.05).</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_primary.test">primary.test</code></td>
<td>
<p>Primary statistical test, e.g. &quot;anova&quot;, &quot;kruskal.test&quot;.</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_secondary.test">secondary.test</code></td>
<td>
<p>Secondary statistical test, e.g. &quot;t-test&quot;, &quot;wilcox.test&quot;</p>
</td></tr>
<tr><td><code id="addPlotStats_+3A_exact">exact</code></td>
<td>
<p>Whether to calculate exact p values (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>

<hr>
<h2 id='addPlotStatsSamples'>Add statistics to plot</h2><span id='topic+addPlotStatsSamples'></span>

<h3>Description</h3>

<p>Use ggpubr to add statistics to samples or plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addPlotStatsSamples(
  p,
  comp.group,
  metadata,
  h.adj = 0.05,
  exact = FALSE,
  second.comp.group
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addPlotStatsSamples_+3A_p">p</code></td>
<td>
<p>Plot to add statistics to.</p>
</td></tr>
<tr><td><code id="addPlotStatsSamples_+3A_comp.group">comp.group</code></td>
<td>
<p>Comparison metric.</p>
</td></tr>
<tr><td><code id="addPlotStatsSamples_+3A_metadata">metadata</code></td>
<td>
<p>Metadata for samples.</p>
</td></tr>
<tr><td><code id="addPlotStatsSamples_+3A_h.adj">h.adj</code></td>
<td>
<p>Position of statistics test p value as % of max(y) (default = 0.05).</p>
</td></tr>
<tr><td><code id="addPlotStatsSamples_+3A_exact">exact</code></td>
<td>
<p>Whether to calculate exact p values (default = FALSE).</p>
</td></tr>
<tr><td><code id="addPlotStatsSamples_+3A_second.comp.group">second.comp.group</code></td>
<td>
<p>Second comparison metric.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>

<hr>
<h2 id='addSummaryMetrics'>Add summary metrics</h2><span id='topic+addSummaryMetrics'></span>

<h3>Description</h3>

<p>Add summary metrics by reading Cell Ranger metrics summary files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addSummaryMetrics(data.path, metadata, n.cores = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addSummaryMetrics_+3A_data.path">data.path</code></td>
<td>
<p>Path to cellranger count data.</p>
</td></tr>
<tr><td><code id="addSummaryMetrics_+3A_metadata">metadata</code></td>
<td>
<p>Metadata for samples.</p>
</td></tr>
<tr><td><code id="addSummaryMetrics_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores for the calculations (default = 1).</p>
</td></tr>
<tr><td><code id="addSummaryMetrics_+3A_verbose">verbose</code></td>
<td>
<p>Print messages (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='checkCompGroup'>Set correct 'comp.group' parameter</h2><span id='topic+checkCompGroup'></span>

<h3>Description</h3>

<p>Set comp.group to 'category' if null.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkCompGroup(comp.group, category, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkCompGroup_+3A_comp.group">comp.group</code></td>
<td>
<p>Comparison metric.</p>
</td></tr>
<tr><td><code id="checkCompGroup_+3A_category">category</code></td>
<td>
<p>Comparison metric to use if comp.group is not provided.</p>
</td></tr>
<tr><td><code id="checkCompGroup_+3A_verbose">verbose</code></td>
<td>
<p>Print messages (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector
</p>

<hr>
<h2 id='checkCompMeta'>Check whether 'comp.group' is in metadata</h2><span id='topic+checkCompMeta'></span>

<h3>Description</h3>

<p>Checks whether 'comp.group' is any of the column names in metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkCompMeta(comp.group, metadata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkCompMeta_+3A_comp.group">comp.group</code></td>
<td>
<p>Comparison metric.</p>
</td></tr>
<tr><td><code id="checkCompMeta_+3A_metadata">metadata</code></td>
<td>
<p>Metadata for samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing or stop
</p>

<hr>
<h2 id='checkDataPath'>Check data path</h2><span id='topic+checkDataPath'></span>

<h3>Description</h3>

<p>Helper function to check that data.path is not NULL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkDataPath(data.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkDataPath_+3A_data.path">data.path</code></td>
<td>
<p>character Path to be checked</p>
</td></tr>
</table>

<hr>
<h2 id='createUniqueCellNames'>Create unique cell names</h2><span id='topic+createUniqueCellNames'></span>

<h3>Description</h3>

<p>Create unique cell names from sample IDs and cell IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createUniqueCellNames(cms, samples, sep = "!!")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createUniqueCellNames_+3A_cms">cms</code></td>
<td>
<p>list List of count matrices, should be named (optional)</p>
</td></tr>
<tr><td><code id="createUniqueCellNames_+3A_samples">samples</code></td>
<td>
<p>character Optional, list of sample names</p>
</td></tr>
<tr><td><code id="createUniqueCellNames_+3A_sep">sep</code></td>
<td>
<p>character Separator between sample IDs and cell IDs (default = &quot;!!&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='CRMetrics'>CRMetrics class object</h2><span id='topic+CRMetrics'></span>

<h3>Description</h3>

<p>Functions to analyze Cell Ranger count data. To initialize a new object, 'data.path' or 'cms' is needed. 'metadata' is also recommended, but not required.
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>metadata</code></dt><dd><p>data.frame or character Path to metadata file or name of metadata data.frame object. Metadata must contain a column named 'sample' containing sample names that must match folder names in 'data.path' (default = NULL)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path(s) to Cell Ranger count data, one directory per sample. If multiple paths, do c(&quot;path1&quot;,&quot;path2&quot;) (default = NULL)</p>
</dd>
<dt><code>cms</code></dt><dd><p>list List with count matrices (default = NULL)</p>
</dd>
<dt><code>cms.preprocessed</code></dt><dd><p>list List with preprocessed count matrices after $doPreprocessing() (default = NULL)</p>
</dd>
<dt><code>cms.raw</code></dt><dd><p>list List with raw, unfiltered count matrices, i.e., including all CBs detected also empty droplets (default = NULL)</p>
</dd>
<dt><code>summary.metrics</code></dt><dd><p>data.frame Summary metrics from Cell Ranger (default = NULL)</p>
</dd>
<dt><code>detailed.metrics</code></dt><dd><p>data.frame Detailed metrics, i.e., no. genes and UMIs per cell (default = NULL)</p>
</dd>
<dt><code>comp.group</code></dt><dd><p>character A group present in the metadata to compare the metrics by, can be added with addComparison (default = NULL)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = TRUE)</p>
</dd>
<dt><code>theme</code></dt><dd><p>ggplot2 theme (default: theme_bw())</p>
</dd>
<dt><code>pal</code></dt><dd><p>Plotting palette (default = NULL)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>numeric Number of cores for calculations (default = 1)
Initialize a CRMetrics object</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-CRMetrics-new"><code>CRMetrics$new()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-addDetailedMetrics"><code>CRMetrics$addDetailedMetrics()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-addComparison"><code>CRMetrics$addComparison()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotSamples"><code>CRMetrics$plotSamples()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotSummaryMetrics"><code>CRMetrics$plotSummaryMetrics()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotDetailedMetrics"><code>CRMetrics$plotDetailedMetrics()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotEmbedding"><code>CRMetrics$plotEmbedding()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotDepth"><code>CRMetrics$plotDepth()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotMitoFraction"><code>CRMetrics$plotMitoFraction()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-detectDoublets"><code>CRMetrics$detectDoublets()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-doPreprocessing"><code>CRMetrics$doPreprocessing()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-createEmbedding"><code>CRMetrics$createEmbedding()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-filterCms"><code>CRMetrics$filterCms()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-selectMetrics"><code>CRMetrics$selectMetrics()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotFilteredCells"><code>CRMetrics$plotFilteredCells()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-getDepth"><code>CRMetrics$getDepth()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-getMitoFraction"><code>CRMetrics$getMitoFraction()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-prepareCellbender"><code>CRMetrics$prepareCellbender()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-saveCellbenderScript"><code>CRMetrics$saveCellbenderScript()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-getExpectedCells"><code>CRMetrics$getExpectedCells()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-getTotalDroplets"><code>CRMetrics$getTotalDroplets()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-addCms"><code>CRMetrics$addCms()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotCbTraining"><code>CRMetrics$plotCbTraining()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotCbCellProbs"><code>CRMetrics$plotCbCellProbs()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotCbAmbExp"><code>CRMetrics$plotCbAmbExp()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotCbAmbGenes"><code>CRMetrics$plotCbAmbGenes()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-addSummaryFromCms"><code>CRMetrics$addSummaryFromCms()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-runSoupX"><code>CRMetrics$runSoupX()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotSoupX"><code>CRMetrics$plotSoupX()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-plotCbCells"><code>CRMetrics$plotCbCells()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-addDoublets"><code>CRMetrics$addDoublets()</code></a>
</p>
</li>
<li> <p><a href="#method-CRMetrics-clone"><code>CRMetrics$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-CRMetrics-new"></a>



<h4>Method <code>new()</code></h4>

<p>To initialize new object, 'data.path' or 'cms' is needed. 'metadata' is also recommended, but not required.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$new(
  data.path = NULL,
  metadata = NULL,
  cms = NULL,
  samples = NULL,
  unique.names = TRUE,
  sep.cells = "!!",
  comp.group = NULL,
  verbose = TRUE,
  theme = theme_bw(),
  n.cores = 1,
  sep.meta = ",",
  raw.meta = FALSE,
  pal = NULL
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data.path</code></dt><dd><p>character Path to directory with Cell Ranger count data, one directory per sample (default = NULL).</p>
</dd>
<dt><code>metadata</code></dt><dd><p>data.frame or character Path to metadata file (comma-separated) or name of metadata dataframe object. Metadata must contain a column named 'sample' containing sample names that must match folder names in 'data.path' (default = NULL)</p>
</dd>
<dt><code>cms</code></dt><dd><p>list List with count matrices (default = NULL)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names. Only relevant is cms is provided (default = NULL)</p>
</dd>
<dt><code>unique.names</code></dt><dd><p>logical Create unique cell names. Only relevant if cms is provided (default = TRUE)</p>
</dd>
<dt><code>sep.cells</code></dt><dd><p>character Sample-cell separator. Only relevant if cms is provided and <code>unique.names=TRUE</code> (default = &quot;!!&quot;)</p>
</dd>
<dt><code>comp.group</code></dt><dd><p>character A group present in the metadata to compare the metrics by, can be added with addComparison (default = NULL)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = TRUE)</p>
</dd>
<dt><code>theme</code></dt><dd><p>ggplot2 theme (default: theme_bw())</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores for the calculations (default = self$n.cores)</p>
</dd>
<dt><code>sep.meta</code></dt><dd><p>character Separator for metadata file (default = &quot;,&quot;)</p>
</dd>
<dt><code>raw.meta</code></dt><dd><p>logical Keep metadata in its raw format. If FALSE, classes will be converted using &quot;type.convert&quot; (default = FALSE)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = NULL)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>CRMetrics object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-addDetailedMetrics"></a>



<h4>Method <code>addDetailedMetrics()</code></h4>

<p>Function to read in detailed metrics. This is not done upon initialization for speed.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$addDetailedMetrics(
  cms = self$cms,
  min.transcripts.per.cell = 100,
  n.cores = self$n.cores,
  verbose = self$verbose
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list List of (sparse) count matrices (default = self$cms)</p>
</dd>
<dt><code>min.transcripts.per.cell</code></dt><dd><p>numeric Minimal number of transcripts per cell (default = 100)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores for the calculations (default = self$n.cores).</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = self$verbose).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Count matrices
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Run function
crm$addDetailedMetrics()
</pre>
</div>


<hr>
<a id="method-CRMetrics-addComparison"></a>



<h4>Method <code>addComparison()</code></h4>

<p>Add comparison group for statistical testing.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$addComparison(comp.group, metadata = self$metadata)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>comp.group</code></dt><dd><p>character Comparison metric (default = self$comp.group).</p>
</dd>
<dt><code>metadata</code></dt><dd><p>data.frame Metadata for samples (default = self$metadata).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Vector
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add metadata
crm$metadata &lt;- data.frame(sex = c("male","female"))

# Add comparison group
crm$addComparison(comp.group = "sex")
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotSamples"></a>



<h4>Method <code>plotSamples()</code></h4>

<p>Plot the number of samples.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotSamples(
  comp.group = self$comp.group,
  h.adj = 0.05,
  exact = FALSE,
  metadata = self$metadata,
  second.comp.group = NULL,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>comp.group</code></dt><dd><p>character Comparison metric, must match a column name of metadata (default = self$comp.group).</p>
</dd>
<dt><code>h.adj</code></dt><dd><p>numeric Position of statistics test p value as % of max(y) (default = 0.05).</p>
</dd>
<dt><code>exact</code></dt><dd><p>logical Whether to calculate exact p values (default = FALSE).</p>
</dd>
<dt><code>metadata</code></dt><dd><p>data.frame Metadata for samples (default = self$metadata).</p>
</dd>
<dt><code>second.comp.group</code></dt><dd><p>character Second comparison metric, must match a column name of metadata (default = NULL).</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>samples &lt;- c("sample1", "sample2")

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})
names(testdata.cms) &lt;- samples

# Create metadata
metadata &lt;- data.frame(sample = samples,
sex = c("male","female"),
condition = c("a","b"))

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, metadata = metadata, n.cores = 1)

# Plot
crm$plotSamples(comp.group = "sex", second.comp.group = "condition")
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotSummaryMetrics"></a>



<h4>Method <code>plotSummaryMetrics()</code></h4>

<p>Plot all summary stats or a selected list.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotSummaryMetrics(
  comp.group = self$comp.group,
  second.comp.group = NULL,
  metrics = NULL,
  h.adj = 0.05,
  plot.stat = TRUE,
  stat.test = c("non-parametric", "parametric"),
  exact = FALSE,
  metadata = self$metadata,
  summary.metrics = self$summary.metrics,
  plot.geom = "bar",
  se = FALSE,
  group.reg.lines = FALSE,
  secondary.testing = TRUE,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>comp.group</code></dt><dd><p>character Comparison metric (default = self$comp.group).</p>
</dd>
<dt><code>second.comp.group</code></dt><dd><p>character Second comparison metric, used for the metric &quot;samples per group&quot; or when &quot;comp.group&quot; is a numeric or an integer (default = NULL).</p>
</dd>
<dt><code>metrics</code></dt><dd><p>character Metrics to plot (default = NULL).</p>
</dd>
<dt><code>h.adj</code></dt><dd><p>numeric Position of statistics test p value as % of max(y) (default = 0.05)</p>
</dd>
<dt><code>plot.stat</code></dt><dd><p>logical Show statistics in plot. Will be FALSE if &quot;comp.group&quot; = &quot;sample&quot; or if &quot;comp.group&quot; is a numeric or an integer (default = TRUE)</p>
</dd>
<dt><code>stat.test</code></dt><dd><p>character Statistical test to perform to compare means. Can either be &quot;non-parametric&quot; or &quot;parametric&quot; (default = &quot;non-parametric&quot;).</p>
</dd>
<dt><code>exact</code></dt><dd><p>logical Whether to calculate exact p values (default = FALSE).</p>
</dd>
<dt><code>metadata</code></dt><dd><p>data.frame Metadata for samples (default = self$metadata).</p>
</dd>
<dt><code>summary.metrics</code></dt><dd><p>data.frame Summary metrics (default = self$summary.metrics).</p>
</dd>
<dt><code>plot.geom</code></dt><dd><p>character Which geometric is used to plot the data (default = &quot;point&quot;).</p>
</dd>
<dt><code>se</code></dt><dd><p>logical For regression lines, show SE (default = FALSE)</p>
</dd>
<dt><code>group.reg.lines</code></dt><dd><p>logical For regression lines, if FALSE show one line, if TRUE show line per group defined by second.comp.group (default = FALSE)</p>
</dd>
<dt><code>secondary.testing</code></dt><dd><p>logical Whether to show post hoc testing (default = TRUE)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary metrics
crm$addSummaryFromCms()

crm$plotSummaryMetrics(plot.geom = "point")
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotDetailedMetrics"></a>



<h4>Method <code>plotDetailedMetrics()</code></h4>

<p>Plot detailed metrics from the detailed.metrics object
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotDetailedMetrics(
  comp.group = self$comp.group,
  detailed.metrics = self$detailed.metrics,
  metadata = self$metadata,
  metrics = NULL,
  plot.geom = "violin",
  hline = TRUE,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>comp.group</code></dt><dd><p>character Comparison metric (default = self$comp.group).</p>
</dd>
<dt><code>detailed.metrics</code></dt><dd><p>data.frame Object containing the count matrices (default = self$detailed.metrics).</p>
</dd>
<dt><code>metadata</code></dt><dd><p>data.frame Metadata for samples (default = self$metadata).</p>
</dd>
<dt><code>metrics</code></dt><dd><p>character Metrics to plot. NULL plots both plots (default = NULL).</p>
</dd>
<dt><code>plot.geom</code></dt><dd><p>character How to plot the data (default = &quot;violin&quot;).</p>
</dd>
<dt><code>hline</code></dt><dd><p>logical Whether to show median as horizontal line (default = TRUE)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger count data (default = self$data.path).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add detailed metrics
crm$addDetailedMetrics()

# Plot
crm$plotDetailedMetrics()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotEmbedding"></a>



<h4>Method <code>plotEmbedding()</code></h4>

<p>Plot cells in embedding using Conos and color by depth and doublets.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotEmbedding(
  depth = FALSE,
  doublet.method = NULL,
  doublet.scores = FALSE,
  depth.cutoff = 1000,
  mito.frac = FALSE,
  mito.cutoff = 0.05,
  species = c("human", "mouse"),
  size = 0.3,
  sep = "!!",
  pal = NULL,
  ...
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>depth</code></dt><dd><p>logical Plot depth or not (default = FALSE).</p>
</dd>
<dt><code>doublet.method</code></dt><dd><p>character Doublet detection method (default = NULL).</p>
</dd>
<dt><code>doublet.scores</code></dt><dd><p>logical Plot doublet scores or not (default = FALSE).</p>
</dd>
<dt><code>depth.cutoff</code></dt><dd><p>numeric Depth cutoff (default = 1e3).</p>
</dd>
<dt><code>mito.frac</code></dt><dd><p>logical Plot mitochondrial fraction or not (default = FALSE).</p>
</dd>
<dt><code>mito.cutoff</code></dt><dd><p>numeric Mitochondrial fraction cutoff (default = 0.05).</p>
</dd>
<dt><code>species</code></dt><dd><p>character Species to calculate the mitochondrial fraction for (default = c(&quot;human&quot;,&quot;mouse&quot;)).</p>
</dd>
<dt><code>size</code></dt><dd><p>numeric Dot size (default = 0.3)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = NULL)</p>
</dd>
<dt><code>...</code></dt><dd><p>Plotting parameters passed to <code>sccore::embeddingPlot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding() 

crm$plotEmbedding()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotDepth"></a>



<h4>Method <code>plotDepth()</code></h4>

<p>Plot the sequencing depth in histogram.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotDepth(
  cutoff = 1000,
  samples = self$metadata$sample,
  sep = "!!",
  keep.col = "#E7CDC2",
  filter.col = "#A65141"
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cutoff</code></dt><dd><p>numeric The depth cutoff to color the cells in the embedding (default = 1e3).</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include for plotting (default = $metadata$sample).</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>keep.col</code></dt><dd><p>character Color for density of cells that are kept (default = &quot;#E7CDC2&quot;)</p>
</dd>
<dt><code>filter.col</code></dt><dd><p>Character Color for density of cells to be filtered (default = &quot;#A65141&quot;)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot
crm$plotDepth()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotMitoFraction"></a>



<h4>Method <code>plotMitoFraction()</code></h4>

<p>Plot the mitochondrial fraction in histogram.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotMitoFraction(
  cutoff = 0.05,
  species = c("human", "mouse"),
  samples = self$metadata$sample,
  sep = "!!",
  keep.col = "#E7CDC2",
  filter.col = "#A65141"
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cutoff</code></dt><dd><p>numeric The mito. fraction cutoff to color the embedding (default = 0.05)</p>
</dd>
<dt><code>species</code></dt><dd><p>character Species to calculate the mitochondrial fraction for (default = &quot;human&quot;)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include for plotting (default = $metadata$sample)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>keep.col</code></dt><dd><p>character Color for density of cells that are kept (default = &quot;#E7CDC2&quot;)</p>
</dd>
<dt><code>filter.col</code></dt><dd><p>Character Color for density of cells to be filtered (default = &quot;#A65141&quot;)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot
crm$plotMitoFraction()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-detectDoublets"></a>



<h4>Method <code>detectDoublets()</code></h4>

<p>Detect doublet cells.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$detectDoublets(
  method = c("scrublet", "doubletdetection"),
  cms = self$cms,
  samples = self$metadata$sample,
  env = "r-reticulate",
  conda.path = system("whereis conda"),
  n.cores = self$n.cores,
  verbose = self$verbose,
  args = list(),
  export = FALSE,
  data.path = self$data.path
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>method</code></dt><dd><p>character Which method to use, either <code>scrublet</code> or <code>doubletdetection</code> (default=&quot;scrublet&quot;).</p>
</dd>
<dt><code>cms</code></dt><dd><p>list List containing the count matrices (default=self$cms).</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Vector of sample names. If NULL, samples are extracted from cms (default = self$metadata$sample)</p>
</dd>
<dt><code>env</code></dt><dd><p>character Environment to run python in (default=&quot;r-reticulate&quot;).</p>
</dd>
<dt><code>conda.path</code></dt><dd><p>character Path to conda environment (default=system(&quot;whereis conda&quot;)).</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores to use (default = self$n.cores)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = self$verbose)</p>
</dd>
<dt><code>args</code></dt><dd><p>list A list with additional arguments for either <code>DoubletDetection</code> or <code>scrublet</code>. Please check the respective manuals.</p>
</dd>
<dt><code>export</code></dt><dd><p>boolean Export CMs in order to detect doublets outside R (default = FALSE)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to write data, only relevant if <code>export = TRUE</code>. Last character must be <code>/</code> (default = self$data.path)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>data.frame
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)


# Detect doublets
crm$detectDoublets(method = "scrublet", 
conda.path = "/opt/software/miniconda/4.12.0/condabin/conda")
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-doPreprocessing"></a>



<h4>Method <code>doPreprocessing()</code></h4>

<p>Perform conos preprocessing.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$doPreprocessing(
  cms = self$cms,
  preprocess = c("pagoda2", "seurat"),
  min.transcripts.per.cell = 100,
  verbose = self$verbose,
  n.cores = self$n.cores,
  get.largevis = FALSE,
  tsne = FALSE,
  make.geneknn = FALSE,
  cluster = FALSE,
  ...
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list List containing the count matrices (default = self$cms).</p>
</dd>
<dt><code>preprocess</code></dt><dd><p>character Method to use for preprocessing (default = c(&quot;pagoda2&quot;,&quot;seurat&quot;)).</p>
</dd>
<dt><code>min.transcripts.per.cell</code></dt><dd><p>numeric Minimal transcripts per cell (default = 100)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = self$verbose).</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores for the calculations (default = self$n.cores).</p>
</dd>
<dt><code>get.largevis</code></dt><dd><p>logical For Pagoda2, create largeVis embedding (default = FALSE)</p>
</dd>
<dt><code>tsne</code></dt><dd><p>logical Create tSNE embedding (default = FALSE)</p>
</dd>
<dt><code>make.geneknn</code></dt><dd><p>logical For Pagoda2, estimate gene kNN (default = FALSE)</p>
</dd>
<dt><code>cluster</code></dt><dd><p>logical For Seurat, estimate clusters (default = FALSE)</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional arguments for <code>Pagaoda2::basicP2Proc</code> or <code>conos:::basicSeuratProc</code></p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Conos object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Perform preprocessing
crm$doPreprocessing(preprocess = "pagoda2")
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-createEmbedding"></a>



<h4>Method <code>createEmbedding()</code></h4>

<p>Create Conos embedding.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$createEmbedding(
  cms = self$cms.preprocessed,
  verbose = self$verbose,
  n.cores = self$n.cores,
  arg.buildGraph = list(),
  arg.findCommunities = list(),
  arg.embedGraph = list(method = "UMAP")
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list List containing the preprocessed count matrices (default = self$cms.preprocessed).</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Print messages or not (default = self$verbose).</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores for the calculations (default = self$n.cores).</p>
</dd>
<dt><code>arg.buildGraph</code></dt><dd><p>list A list with additional arguments for the <code>buildGraph</code> function in Conos (default = list())</p>
</dd>
<dt><code>arg.findCommunities</code></dt><dd><p>list A list with additional arguments for the <code>findCommunities</code> function in Conos (default = list())</p>
</dd>
<dt><code>arg.embedGraph</code></dt><dd><p>list A list with additional arguments for the <code>embedGraph</code> function in Conos (default = list(method = &quot;UMAP))</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Conos object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-filterCms"></a>



<h4>Method <code>filterCms()</code></h4>

<p>Filter cells based on depth, mitochondrial fraction and doublets from the count matrix.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$filterCms(
  depth.cutoff = NULL,
  mito.cutoff = NULL,
  doublets = NULL,
  species = c("human", "mouse"),
  samples.to.exclude = NULL,
  verbose = self$verbose,
  sep = "!!",
  raw = FALSE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>depth.cutoff</code></dt><dd><p>numeric Depth (transcripts per cell) cutoff (default = NULL).</p>
</dd>
<dt><code>mito.cutoff</code></dt><dd><p>numeric Mitochondrial fraction cutoff (default = NULL).</p>
</dd>
<dt><code>doublets</code></dt><dd><p>character Doublet detection method to use (default = NULL).</p>
</dd>
<dt><code>species</code></dt><dd><p>character Species to calculate the mitochondrial fraction for (default = &quot;human&quot;).</p>
</dd>
<dt><code>samples.to.exclude</code></dt><dd><p>character Sample names to exclude (default = NULL)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Show progress (default = self$verbose)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>raw</code></dt><dd><p>boolean Filter on raw, unfiltered count matrices. Usually not intended (default = FALSE)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>list of filtered count matrices
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()


# Filter CMs
crm$filterCms(depth.cutoff = 1e3, mito.cutoff = 0.05)
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-selectMetrics"></a>



<h4>Method <code>selectMetrics()</code></h4>

<p>Select metrics from summary.metrics
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$selectMetrics(ids = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>ids</code></dt><dd><p>character Metric id to select (default = NULL).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>vector
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Select metrics
crm$selectMetrics()
selection.metrics &lt;- crm$selectMetrics(c(1:4))
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotFilteredCells"></a>



<h4>Method <code>plotFilteredCells()</code></h4>

<p>Plot filtered cells in an embedding, in a bar plot, on a tile or export the data frame
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotFilteredCells(
  type = c("embedding", "bar", "tile", "export"),
  depth = TRUE,
  depth.cutoff = 1000,
  doublet.method = NULL,
  mito.frac = TRUE,
  mito.cutoff = 0.05,
  species = c("human", "mouse"),
  size = 0.3,
  sep = "!!",
  cols = c("grey80", "red", "blue", "green", "yellow", "black", "pink", "purple"),
  ...
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>type</code></dt><dd><p>character The type of plot to use: embedding, bar, tile or export (default = c(&quot;embedding&quot;,&quot;bar&quot;,&quot;tile&quot;,&quot;export&quot;)).</p>
</dd>
<dt><code>depth</code></dt><dd><p>logical Plot the depth or not (default = TRUE).</p>
</dd>
<dt><code>depth.cutoff</code></dt><dd><p>numeric Depth cutoff, either a single number or a vector with cutoff per sample and with sampleIDs as names (default = 1e3).</p>
</dd>
<dt><code>doublet.method</code></dt><dd><p>character Method to detect doublets (default = NULL).</p>
</dd>
<dt><code>mito.frac</code></dt><dd><p>logical Plot the mitochondrial fraction or not (default = TRUE).</p>
</dd>
<dt><code>mito.cutoff</code></dt><dd><p>numeric Mitochondrial fraction cutoff, either a single number or a vector with cutoff per sample and with sampleIDs as names (default = 0.05).</p>
</dd>
<dt><code>species</code></dt><dd><p>character Species to calculate the mitochondrial fraction for (default = c(&quot;human&quot;,&quot;mouse&quot;)).</p>
</dd>
<dt><code>size</code></dt><dd><p>numeric Dot size (default = 0.3)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>cols</code></dt><dd><p>character Colors used for plotting (default = c(&quot;grey80&quot;,&quot;red&quot;,&quot;blue&quot;,&quot;green&quot;,&quot;yellow&quot;,&quot;black&quot;,&quot;pink&quot;,&quot;purple&quot;))</p>
</dd>
<dt><code>...</code></dt><dd><p>Plotting parameters passed to <code>sccore::embeddingPlot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object or data frame
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot and extract result
crm$plotFilteredCells(type = "embedding")
filtered.cells &lt;- crm$plotFilteredCells(type = "export")
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-getDepth"></a>



<h4>Method <code>getDepth()</code></h4>

<p>Extract sequencing depth from Conos object.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$getDepth(cms = self$cms)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list List of (sparse) count matrices (default = self$cms)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>data frame
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Get depth
crm$getDepth()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-getMitoFraction"></a>



<h4>Method <code>getMitoFraction()</code></h4>

<p>Calculate the fraction of mitochondrial genes.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$getMitoFraction(species = c("human", "mouse"), cms = self$cms)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>species</code></dt><dd><p>character Species to calculate the mitochondrial fraction for (default = &quot;human&quot;).</p>
</dd>
<dt><code>cms</code></dt><dd><p>list List of (sparse) count matrices (default = self$cms)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>data frame
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\donttest{
if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Get mito. fraction
crm$getMitoFraction(species = c("human", "mouse"))
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-prepareCellbender"></a>



<h4>Method <code>prepareCellbender()</code></h4>

<p>Create plots and script call for CellBender
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$prepareCellbender(
  shrinkage = 100,
  show.expected.cells = TRUE,
  show.total.droplets = TRUE,
  expected.cells = NULL,
  total.droplets = NULL,
  cms.raw = self$cms.raw,
  umi.counts = self$cellbender$umi.counts,
  data.path = self$data.path,
  samples = self$metadata$sample,
  verbose = self$verbose,
  n.cores = self$n.cores,
  unique.names = FALSE,
  sep = "!!"
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>shrinkage</code></dt><dd><p>integer Select every nth UMI count per cell for plotting. Improves plotting speed drastically. To plot all cells, set to 1 (default = 100)</p>
</dd>
<dt><code>show.expected.cells</code></dt><dd><p>logical Plot line depicting expected number of cells (default = TRUE)</p>
</dd>
<dt><code>show.total.droplets</code></dt><dd><p>logical Plot line depicting total droplets included for CellBender run (default = TRUE)</p>
</dd>
<dt><code>expected.cells</code></dt><dd><p>named numeric If NULL, expected cells will be deduced from the number of cells per sample identified by Cell Ranger. Otherwise, a named vector of expected cells with sample IDs as names. Sample IDs must match those in summary.metrics (default: stored named vector)</p>
</dd>
<dt><code>total.droplets</code></dt><dd><p>named numeric If NULL, total droplets included will be deduced from expected cells multiplied by 3. Otherwise, a named vector of total droplets included with sample IDs as names. Sample IDs must match those in summary.metrics (default: stored named vector)</p>
</dd>
<dt><code>cms.raw</code></dt><dd><p>list Raw count matrices from HDF5 Cell Ranger outputs (default = self$cms.raw)</p>
</dd>
<dt><code>umi.counts</code></dt><dd><p>list UMI counts calculated as column sums of raw count matrices from HDF5 Cell Ranger outputs (default: stored list)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Show progress (default: stored vector)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores (default: stored vector)</p>
</dd>
<dt><code>unique.names</code></dt><dd><p>logical Create unique cell names (default = FALSE)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator for creating unique cell names (default = &quot;!!&quot;)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>ggplot2 object and bash script
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data")
crm$prepareCellbender()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-saveCellbenderScript"></a>



<h4>Method <code>saveCellbenderScript()</code></h4>



<h5>Usage</h5>

<div class="r"><pre>CRMetrics$saveCellbenderScript(
  file = "cellbender_script.sh",
  fpr = 0.01,
  epochs = 150,
  use.gpu = TRUE,
  expected.cells = NULL,
  total.droplets = NULL,
  data.path = self$data.path,
  samples = self$metadata$sample,
  args = NULL
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>file</code></dt><dd><p>character File name for CellBender script. Will be stored in <code>data.path</code> (default: &quot;cellbender_script.sh&quot;)</p>
</dd>
<dt><code>fpr</code></dt><dd><p>numeric False positive rate for CellBender (default = 0.01)</p>
</dd>
<dt><code>epochs</code></dt><dd><p>integer Number of epochs for CellBender (default = 150)</p>
</dd>
<dt><code>use.gpu</code></dt><dd><p>logical Use CUDA capable GPU (default = TRUE)</p>
</dd>
<dt><code>expected.cells</code></dt><dd><p>named numeric If NULL, expected cells will be deduced from the number of cells per sample identified by Cell Ranger. Otherwise, a named vector of expected cells with sample IDs as names. Sample IDs must match those in summary.metrics (default: stored named vector)</p>
</dd>
<dt><code>total.droplets</code></dt><dd><p>named numeric If NULL, total droplets included will be deduced from expected cells multiplied by 3. Otherwise, a named vector of total droplets included with sample IDs as names. Sample IDs must match those in summary.metrics (default: stored named vector)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>args</code></dt><dd><p>character (optional) Additional parameters for CellBender</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>bash script
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-getExpectedCells"></a>



<h4>Method <code>getExpectedCells()</code></h4>

<p>Extract the expected number of cells per sample based on the Cell Ranger summary metrics
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$getExpectedCells(samples = self$metadata$sample)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A numeric vector
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Get summary
crm$addSummaryFromCms()

# Get no. cells
crm$getExpectedCells()
</pre>
</div>


<hr>
<a id="method-CRMetrics-getTotalDroplets"></a>



<h4>Method <code>getTotalDroplets()</code></h4>

<p>Get the total number of droplets included in the CellBender estimations. Based on the Cell Ranger summary metrics and multiplied by a preset multiplier.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$getTotalDroplets(samples = self$metadata$sample, multiplier = 3)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>samples</code></dt><dd><p>character Samples names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>multiplier</code></dt><dd><p>numeric Number to multiply expected number of cells with (default = 3)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A numeric vector
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary
crm$addSummaryFromCms()

# Get no. droplets
crm$getTotalDroplets()
</pre>
</div>


<hr>
<a id="method-CRMetrics-addCms"></a>



<h4>Method <code>addCms()</code></h4>

<p>Add a list of count matrices to the CRMetrics object.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$addCms(
  cms = NULL,
  data.path = self$data.path,
  samples = self$metadata$sample,
  cellbender = FALSE,
  raw = FALSE,
  symbol = TRUE,
  unique.names = TRUE,
  sep = "!!",
  add.metadata = TRUE,
  n.cores = self$n.cores,
  verbose = self$verbose
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list List of (sparse) count matrices (default = NULL)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to cellranger count data (default = self$data.path).</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Vector of sample names. If NULL, samples are extracted from cms (default = self$metadata$sample)</p>
</dd>
<dt><code>cellbender</code></dt><dd><p>logical Add CellBender filtered count matrices in HDF5 format. Requires that &quot;cellbender&quot; is in the names of the files (default = FALSE)</p>
</dd>
<dt><code>raw</code></dt><dd><p>logical Add raw count matrices from Cell Ranger output. Cannot be combined with <code>cellbender=TRUE</code> (default = FALSE)</p>
</dd>
<dt><code>symbol</code></dt><dd><p>character The type of gene IDs to use, SYMBOL (TRUE) or ENSEMBLE (default = TRUE)</p>
</dd>
<dt><code>unique.names</code></dt><dd><p>logical Make cell names unique based on <code>sep</code> parameter (default = TRUE)</p>
</dd>
<dt><code>sep</code></dt><dd><p>character Separator used to create unique cell names (default = &quot;!!&quot;)</p>
</dd>
<dt><code>add.metadata</code></dt><dd><p>boolean Add metadata from cms or not (default = TRUE)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores to use (default = self$n.cores)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>boolean Print progress (default = self$verbose)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Add list of (sparse) count matrices to R6 class object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

crm$addCms(cms = testdata.cms)
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotCbTraining"></a>



<h4>Method <code>plotCbTraining()</code></h4>

<p>Plot the results from the CellBender estimations
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotCbTraining(
  data.path = self$data.path,
  samples = self$metadata$sample,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbTraining()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotCbCellProbs"></a>



<h4>Method <code>plotCbCellProbs()</code></h4>

<p>Plot the CellBender assigned cell probabilities
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotCbCellProbs(
  data.path = self$data.path,
  samples = self$metadata$sample,
  low.col = "gray",
  high.col = "red"
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>low.col</code></dt><dd><p>character Color for low probabilities (default = &quot;gray&quot;)</p>
</dd>
<dt><code>high.col</code></dt><dd><p>character Color for high probabilities (default = &quot;red&quot;)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run the CellBender script
crm$plotCbCellProbs()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotCbAmbExp"></a>



<h4>Method <code>plotCbAmbExp()</code></h4>

<p>Plot the estimated ambient gene expression per sample from CellBender calculations
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotCbAmbExp(
  cutoff = 0.005,
  data.path = self$data.path,
  samples = self$metadata$sample
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cutoff</code></dt><dd><p>numeric Horizontal line included in the plot to indicate highly expressed ambient genes (default = 0.005)</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbAmbExp()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotCbAmbGenes"></a>



<h4>Method <code>plotCbAmbGenes()</code></h4>

<p>Plot the most abundant estimated ambient genes from the CellBender calculations
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotCbAmbGenes(
  cutoff = 0.005,
  data.path = self$data.path,
  samples = self$metadata$sample,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cutoff</code></dt><dd><p>numeric Cutoff of ambient gene expression to use to extract ambient genes per sample</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbAmbGenes()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-addSummaryFromCms"></a>



<h4>Method <code>addSummaryFromCms()</code></h4>

<p>Add summary metrics from a list of count matrices
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$addSummaryFromCms(
  cms = self$cms,
  n.cores = self$n.cores,
  verbose = self$verbose
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>cms</code></dt><dd><p>list A list of filtered count matrices (default = self$cms)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>integer Number of cores to use (default = self$n.cores)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Show progress (default = self$verbose)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>data.frame
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre># Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary
crm$addSummaryFromCms()
</pre>
</div>


<hr>
<a id="method-CRMetrics-runSoupX"></a>



<h4>Method <code>runSoupX()</code></h4>

<p>Run SoupX ambient RNA estimation and correction
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$runSoupX(
  data.path = self$data.path,
  samples = self$metadata$sample,
  n.cores = self$n.cores,
  verbose = self$verbose,
  arg.load10X = list(),
  arg.autoEstCont = list(),
  arg.adjustCounts = list()
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>n.cores</code></dt><dd><p>numeric Number of cores (default = self$n.cores)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Show progress (default = self$verbose)</p>
</dd>
<dt><code>arg.load10X</code></dt><dd><p>list A list with additional parameters for <code>SoupX::load10X</code> (default = list())</p>
</dd>
<dt><code>arg.autoEstCont</code></dt><dd><p>list A list with additional parameters for <code>SoupX::autoEstCont</code> (default = list())</p>
</dd>
<dt><code>arg.adjustCounts</code></dt><dd><p>list A list with additional parameters for <code>SoupX::adjustCounts</code> (default = list())</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>List containing a list with corrected counts, and a data.frame containing plotting estimations
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$runSoupX()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotSoupX"></a>



<h4>Method <code>plotSoupX()</code></h4>

<p>Plot the results from the SoupX estimations
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotSoupX(plot.df = self$soupx$plot.df)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>plot.df</code></dt><dd><p>data.frame SoupX estimations (default = self$soupx$plot.df)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$runSoupX()
crm$plotSoupX()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-plotCbCells"></a>



<h4>Method <code>plotCbCells()</code></h4>

<p>Plot CellBender cell estimations against the estimated cell numbers from Cell Ranger
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$plotCbCells(
  data.path = self$data.path,
  samples = self$metadata$sample,
  pal = self$pal
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>pal</code></dt><dd><p>character Plotting palette (default = self$pal)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 object
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbCells()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-addDoublets"></a>



<h4>Method <code>addDoublets()</code></h4>

<p>Add doublet results created from exported Python script
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$addDoublets(
  method = c("scrublet", "doubletdetection"),
  data.path = self$data.path,
  samples = self$metadata$sample,
  cms = self$cms,
  verbose = self$verbose
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>method</code></dt><dd><p>character Which method to use, either <code>scrublet</code> or <code>doubletdetection</code> (default is both).</p>
</dd>
<dt><code>data.path</code></dt><dd><p>character Path to Cell Ranger outputs (default = self$data.path)</p>
</dd>
<dt><code>samples</code></dt><dd><p>character Sample names to include (default = self$metadata$sample)</p>
</dd>
<dt><code>cms</code></dt><dd><p>list List containing the count matrices (default = self$cms).</p>
</dd>
<dt><code>verbose</code></dt><dd><p>boolean Print progress (default = self$verbose)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>List of doublet results
</p>



<h5>Examples</h5>

<div class="r example copy">
<pre>\dontrun{
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$detectDoublets(export = TRUE)
## Run Python script
crm$addDoublets()
}
</pre>
</div>


<hr>
<a id="method-CRMetrics-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>CRMetrics$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>




<h3>Examples</h3>

<pre><code class='language-R'>
## ------------------------------------------------
## Method `CRMetrics$new`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$addDetailedMetrics`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Run function
crm$addDetailedMetrics()

## ------------------------------------------------
## Method `CRMetrics$addComparison`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add metadata
crm$metadata &lt;- data.frame(sex = c("male","female"))

# Add comparison group
crm$addComparison(comp.group = "sex")

## ------------------------------------------------
## Method `CRMetrics$plotSamples`
## ------------------------------------------------

samples &lt;- c("sample1", "sample2")

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})
names(testdata.cms) &lt;- samples

# Create metadata
metadata &lt;- data.frame(sample = samples,
sex = c("male","female"),
condition = c("a","b"))

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, metadata = metadata, n.cores = 1)

# Plot
crm$plotSamples(comp.group = "sex", second.comp.group = "condition")

## ------------------------------------------------
## Method `CRMetrics$plotSummaryMetrics`
## ------------------------------------------------


# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary metrics
crm$addSummaryFromCms()

crm$plotSummaryMetrics(plot.geom = "point")


## ------------------------------------------------
## Method `CRMetrics$plotDetailedMetrics`
## ------------------------------------------------


# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add detailed metrics
crm$addDetailedMetrics()

# Plot
crm$plotDetailedMetrics()


## ------------------------------------------------
## Method `CRMetrics$plotEmbedding`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding() 

crm$plotEmbedding()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$plotDepth`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot
crm$plotDepth()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$plotMitoFraction`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot
crm$plotMitoFraction()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$detectDoublets`
## ------------------------------------------------

## Not run: 
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)


# Detect doublets
crm$detectDoublets(method = "scrublet", 
conda.path = "/opt/software/miniconda/4.12.0/condabin/conda")

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$doPreprocessing`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Perform preprocessing
crm$doPreprocessing(preprocess = "pagoda2")
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$createEmbedding`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$filterCms`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()


# Filter CMs
crm$filterCms(depth.cutoff = 1e3, mito.cutoff = 0.05)
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$selectMetrics`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Select metrics
crm$selectMetrics()
selection.metrics &lt;- crm$selectMetrics(c(1:4))

## ------------------------------------------------
## Method `CRMetrics$plotFilteredCells`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Plot and extract result
crm$plotFilteredCells(type = "embedding")
filtered.cells &lt;- crm$plotFilteredCells(type = "export")
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$getDepth`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Get depth
crm$getDepth()
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$getMitoFraction`
## ------------------------------------------------


if (requireNamespace("pagoda2", quietly = TRUE)) {
if (requireNamespace("conos", quietly = TRUE)) {
# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Create embedding
crm$doPreprocessing()
crm$createEmbedding()

# Get mito. fraction
crm$getMitoFraction(species = c("human", "mouse"))
} else {
message("Package 'conos' not available.")
}
} else {
message("Package 'pagoda2' not available.")
}


## ------------------------------------------------
## Method `CRMetrics$prepareCellbender`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data")
crm$prepareCellbender()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$saveCellbenderScript`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$getExpectedCells`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Get summary
crm$addSummaryFromCms()

# Get no. cells
crm$getExpectedCells()

## ------------------------------------------------
## Method `CRMetrics$getTotalDroplets`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary
crm$addSummaryFromCms()

# Get no. droplets
crm$getTotalDroplets()

## ------------------------------------------------
## Method `CRMetrics$addCms`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

crm$addCms(cms = testdata.cms)

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotCbTraining`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbTraining()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotCbCellProbs`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run the CellBender script
crm$plotCbCellProbs()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotCbAmbExp`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbAmbExp()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotCbAmbGenes`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbAmbGenes()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$addSummaryFromCms`
## ------------------------------------------------

# Simulate data
testdata.cms &lt;- lapply(seq_len(2), \(x) {
out &lt;- Matrix::rsparsematrix(2e3, 1e3, 0.1)
out[out &lt; 0] &lt;- 1
dimnames(out) &lt;- list(sapply(seq_len(2e3), \(x) paste0("gene",x)),
sapply(seq_len(1e3), \(x) paste0("cell",x)))
return(out)
})

# Initialize
crm &lt;- CRMetrics$new(cms = testdata.cms, samples = c("sample1", "sample2"), n.cores = 1)

# Add summary
crm$addSummaryFromCms()

## ------------------------------------------------
## Method `CRMetrics$runSoupX`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$runSoupX()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotSoupX`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$runSoupX()
crm$plotSoupX()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$plotCbCells`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$prepareCellbender()
crm$saveCellbenderScript()
## Run CellBender script
crm$plotCbCells()

## End(Not run)

## ------------------------------------------------
## Method `CRMetrics$addDoublets`
## ------------------------------------------------

## Not run: 
crm &lt;- CRMetrics$new(data.path = "/path/to/count/data/")
crm$detectDoublets(export = TRUE)
## Run Python script
crm$addDoublets()

## End(Not run)
</code></pre>

<hr>
<h2 id='filterVector'>Create filtering vector</h2><span id='topic+filterVector'></span>

<h3>Description</h3>

<p>Create logical filtering vector based on a numeric vector and a (sample-wise) cutoff
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterVector(num.vec, name, filter, samples, sep = "!!")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterVector_+3A_num.vec">num.vec</code></td>
<td>
<p>numeric Numeric vector to create filter on</p>
</td></tr>
<tr><td><code id="filterVector_+3A_name">name</code></td>
<td>
<p>character Name of filter</p>
</td></tr>
<tr><td><code id="filterVector_+3A_filter">filter</code></td>
<td>
<p>numeric Either a single numeric value or a numeric value with length of samples</p>
</td></tr>
<tr><td><code id="filterVector_+3A_samples">samples</code></td>
<td>
<p>character Sample IDs</p>
</td></tr>
<tr><td><code id="filterVector_+3A_sep">sep</code></td>
<td>
<p>character Separator to split cells by into sample-wise lists (default = &quot;!!&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='getH5Paths'>Get H5 file paths</h2><span id='topic+getH5Paths'></span>

<h3>Description</h3>

<p>Get file paths for H5 files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getH5Paths(data.path, samples = NULL, type = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getH5Paths_+3A_data.path">data.path</code></td>
<td>
<p>character Path for directory containing sample-wise directories with Cell Ranger count outputs</p>
</td></tr>
<tr><td><code id="getH5Paths_+3A_samples">samples</code></td>
<td>
<p>character Sample names to include (default = NULL)</p>
</td></tr>
<tr><td><code id="getH5Paths_+3A_type">type</code></td>
<td>
<p>character Type of H5 files to get paths for, one of &quot;raw&quot;, &quot;filtered&quot; (Cell Ranger count outputs), &quot;cellbender&quot; (raw CellBender outputs), &quot;cellbender_filtered&quot; (CellBender filtered outputs) (default = &quot;type&quot;)</p>
</td></tr>
</table>

<hr>
<h2 id='labelsFilter'>Get labels for percentage of filtered cells</h2><span id='topic+labelsFilter'></span>

<h3>Description</h3>

<p>Labels the percentage of filtered cells based on mitochondrial fraction, sequencing depth and doublets as low, medium or high
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labelsFilter(filter.data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labelsFilter_+3A_filter.data">filter.data</code></td>
<td>
<p>Data frame containing the mitochondrial fraction, depth and doublets per sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='percFilter'>Calculate percentage of filtered cells</h2><span id='topic+percFilter'></span>

<h3>Description</h3>

<p>Calculate percentage of filtered cells based on the filter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>percFilter(filter.data, filter = "mito", no.vars = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="percFilter_+3A_filter.data">filter.data</code></td>
<td>
<p>Data frame containing the mitochondrial fraction, depth and doublets per sample.</p>
</td></tr>
<tr><td><code id="percFilter_+3A_filter">filter</code></td>
<td>
<p>The variable to filter (default = &quot;mito&quot;)</p>
</td></tr>
<tr><td><code id="percFilter_+3A_no.vars">no.vars</code></td>
<td>
<p>numeric Number of variables (default = 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector
</p>

<hr>
<h2 id='plotGeom'>Plot the data as points, as bars as a histogram, or as a violin</h2><span id='topic+plotGeom'></span>

<h3>Description</h3>

<p>Plot the data as points, barplot, histogram or violin
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGeom(g, plot.geom, col, pal = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGeom_+3A_g">g</code></td>
<td>
<p>ggplot2 object</p>
</td></tr>
<tr><td><code id="plotGeom_+3A_plot.geom">plot.geom</code></td>
<td>
<p>The plot.geom to use, &quot;point&quot;, &quot;bar&quot;, &quot;histogram&quot;, or &quot;violin&quot;.</p>
</td></tr>
<tr><td><code id="plotGeom_+3A_pal">pal</code></td>
<td>
<p>character Palette (default = NULL)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>geom
</p>

<hr>
<h2 id='read10x'>Load 10x count matrices</h2><span id='topic+read10x'></span>

<h3>Description</h3>

<p>Load gene expression count data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read10x(
  data.path,
  samples = NULL,
  raw = FALSE,
  symbol = TRUE,
  sep = "!!",
  unique.names = TRUE,
  n.cores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read10x_+3A_data.path">data.path</code></td>
<td>
<p>Path to cellranger count data.</p>
</td></tr>
<tr><td><code id="read10x_+3A_samples">samples</code></td>
<td>
<p>Vector of sample names (default = NULL)</p>
</td></tr>
<tr><td><code id="read10x_+3A_raw">raw</code></td>
<td>
<p>logical Add raw count matrices (default = FALSE)</p>
</td></tr>
<tr><td><code id="read10x_+3A_symbol">symbol</code></td>
<td>
<p>The type of gene IDs to use, SYMBOL (TRUE) or ENSEMBLE (default = TRUE).</p>
</td></tr>
<tr><td><code id="read10x_+3A_sep">sep</code></td>
<td>
<p>Separator for cell names (default = &quot;!!&quot;).</p>
</td></tr>
<tr><td><code id="read10x_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores for the calculations (default = 1).</p>
</td></tr>
<tr><td><code id="read10x_+3A_verbose">verbose</code></td>
<td>
<p>Print messages (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cms &lt;- read10x(data.path = "/path/to/count/data", 
samples = crm$metadata$samples, 
raw = FALSE, 
symbol = TRUE, 
n.cores = crm$n.cores)

## End(Not run)
</code></pre>

<hr>
<h2 id='read10xH5'>Read 10x HDF5 files</h2><span id='topic+read10xH5'></span>

<h3>Description</h3>

<p>Read 10x HDF5 files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read10xH5(
  data.path,
  samples = NULL,
  type = c("raw", "filtered", "cellbender", "cellbender_filtered"),
  symbol = TRUE,
  sep = "!!",
  n.cores = 1,
  verbose = TRUE,
  unique.names = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read10xH5_+3A_data.path">data.path</code></td>
<td>
<p>character</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_samples">samples</code></td>
<td>
<p>character vector, select specific samples for processing (default = NULL)</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_type">type</code></td>
<td>
<p>name of H5 file to search for, &quot;raw&quot; and &quot;filtered&quot; are Cell Ranger count outputs, &quot;cellbender&quot; is output from CellBender after running script from saveCellbenderScript</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_symbol">symbol</code></td>
<td>
<p>logical Use gene SYMBOLs (TRUE) or ENSEMBL IDs (FALSE) (default = TRUE)</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_sep">sep</code></td>
<td>
<p>character Separator for creating unique cell names from sample IDs and cell IDs (default = &quot;!!&quot;)</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_n.cores">n.cores</code></td>
<td>
<p>integer Number of cores (default = 1)</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_verbose">verbose</code></td>
<td>
<p>logical Print progress (default = TRUE)</p>
</td></tr>
<tr><td><code id="read10xH5_+3A_unique.names">unique.names</code></td>
<td>
<p>logical Create unique cell IDs (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with sparse count matrices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cms.h5 &lt;- read10xH5(data.path = "/path/to/count/data")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
