<!DOCTYPE html><html><head><title>Help for package crosshap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {crosshap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#arith_mode'><p>Mode utility function</p></a></li>
<li><a href='#build_bot_halfeyeplot'><p>Bot hap-pheno raincloud plot</p></a></li>
<li><a href='#build_left_alleleplot'><p>Left SNP-allele plot</p></a></li>
<li><a href='#build_left_posplot'><p>Left SNP-position plot</p></a></li>
<li><a href='#build_mid_dotplot'><p>Middle MG/hap dot plot</p></a></li>
<li><a href='#build_right_clusterplot'><p>Right intra-cluster linkage plot</p></a></li>
<li><a href='#build_right_phenoplot'><p>Right SNP-pheno phenoplot</p></a></li>
<li><a href='#build_summary_tables'><p>Hap/MG summary tables</p></a></li>
<li><a href='#build_top_metaplot'><p>Top metadata-hap bar plot</p></a></li>
<li><a href='#clustree_viz'><p>Clustering tree</p></a></li>
<li><a href='#crosshap_viz'><p>Visualize haplotypes</p></a></li>
<li><a href='#crosshap-package'><p>crosshap: Local Haplotype Clustering and Visualization</p></a></li>
<li><a href='#HapObject'><p>Example Haplotype object</p></a></li>
<li><a href='#LD'><p>Example LD matrix</p></a></li>
<li><a href='#mean_na.rm'><p>Mean utility function</p></a></li>
<li><a href='#metadata'><p>Example Domestication metadata</p></a></li>
<li><a href='#pheno'><p>Example phenotype data</p></a></li>
<li><a href='#prepare_hap_umap'><p>UMAP haplotype visualization helper</p></a></li>
<li><a href='#pseudo_haps'><p>Identify haplotypes from clustered SNPs</p></a></li>
<li><a href='#read_LD'><p>Read LD correlation matrix to tibble</p></a></li>
<li><a href='#read_metadata'><p>Read metadata to tibble</p></a></li>
<li><a href='#read_pheno'><p>Read phenotype data to tibble</p></a></li>
<li><a href='#read_vcf'><p>Read VCF to tibble</p></a></li>
<li><a href='#run_haplotyping'><p>Cluster SNPs and identify haplotypes</p></a></li>
<li><a href='#run_hdbscan_haplotyping'><p>Cluster SNPs with HDBSCAN and identify haplotypes</p></a></li>
<li><a href='#tagphenos'><p>Calculate SNP phenotypic associations</p></a></li>
<li><a href='#vcf'><p>Example VCF</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Local Haplotype Clustering and Visualization</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jacob Marsh &lt;jake.marsh@live.com.au&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A local haplotyping visualization toolbox to capture major patterns 
    of co-inheritance between clusters of linked variants, whilst connecting findings 
    to phenotypic and demographic traits across individuals. 'crosshap' enables users 
    to explore and understand genomic variation across a trait-associated region. 
    For an example of successful local haplotype analysis, see Marsh et al. (2022) 
    &lt;<a href="https://doi.org/10.1007%2Fs00122-022-04045-8">doi:10.1007/s00122-022-04045-8</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, clustree, data.table, dbscan, dplyr, ggdist, ggplot2,
ggpp, gridExtra, gtable, magrittr, patchwork, rlang, scales,
tibble, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, testthat (&ge; 3.0.0), umap, vdiffr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.00)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://jacobimarsh.github.io/crosshap/">https://jacobimarsh.github.io/crosshap/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jacobimarsh/crosshap/issues">https://github.com/jacobimarsh/crosshap/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-31 15:17:05 UTC; jmarsh96</td>
</tr>
<tr>
<td>Author:</td>
<td>Jacob Marsh <a href="https://orcid.org/0000-0003-3734-2023"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Brady Johnston <a href="https://orcid.org/0000-0001-6301-2269"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Jakob Petereit <a href="https://orcid.org/0000-0003-2159-0380"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-31 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='arith_mode'>Mode utility function</h2><span id='topic+arith_mode'></span>

<h3>Description</h3>

<p>Mode utility function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arith_mode(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="arith_mode_+3A_x">x</code></td>
<td>
<p>Input vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mode numerical values
</p>

<hr>
<h2 id='build_bot_halfeyeplot'>Bot hap-pheno raincloud plot</h2><span id='topic+build_bot_halfeyeplot'></span>

<h3>Description</h3>

<p>build_bot_halfeyeplot() builds a vertical plot displaying the
phenotypic scores for each individual, grouped by haplotype, coloured by
metadata variable. Metadata groups can be isolated using the isolate_groups
argument. Makes use of the $Indfile information from haplotype
object. It is an internal function called by crosshap_viz(), though can be
called separately to build a stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_bot_halfeyeplot(
  HapObject,
  epsilon,
  hide_labels = TRUE,
  isolate_group = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_bot_halfeyeplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_bot_halfeyeplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="build_bot_halfeyeplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
<tr><td><code id="build_bot_halfeyeplot_+3A_isolate_group">isolate_group</code></td>
<td>
<p>If a Metadata group is provided, all other Metadata
groups will be masked from the plot. NOTE: it does change the summary tables
or marker group phenotype scores.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
build_bot_halfeyeplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_left_alleleplot'>Left SNP-allele plot</h2><span id='topic+build_left_alleleplot'></span>

<h3>Description</h3>

<p>build_left_alleleplot() builds a horizontal plot displaying mean allelic
frequencies (reference/alternate/missing/heterozygous) of all SNP loci,
grouped by marker group. Makes use of $Varfile information from a HapObject
created by run_haplotyping(). This is an internal function called by
crosshap_viz(), though can be called separately to build a stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_left_alleleplot(HapObject, epsilon, hide_labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_left_alleleplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_left_alleleplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon matching the haplotype object used for umap_in.</p>
</td></tr>
<tr><td><code id="build_left_alleleplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>build_left_alleleplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_left_posplot'>Left SNP-position plot</h2><span id='topic+build_left_posplot'></span>

<h3>Description</h3>

<p>build_left_alleleplot() builds a horizontal plot displaying the chromosomal
position of each SNP locus, grouped by marker group. Makes use of the $Varfile
file from haplotype object. It is an internal function called by
crosshap_viz(), though can be called separately to build a stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_left_posplot(HapObject, epsilon, hide_labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_left_posplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_left_posplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon matching the haplotype object used for umap_in.</p>
</td></tr>
<tr><td><code id="build_left_posplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>build_left_posplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_mid_dotplot'>Middle MG/hap dot plot</h2><span id='topic+build_mid_dotplot'></span>

<h3>Description</h3>

<p>build_mid_dotplot() builds a central dot plot displaying the relationship
between haplotype combinations and the characteristic marker group alleles
that define them. Makes use of the $Hapfile information from a haplotype
object. This is an internal function called by crosshap_viz(), though can
be called separately to build a stand-alone plot (can be useful when patched
to a peripheral plot).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_mid_dotplot(HapObject, epsilon, hide_labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_mid_dotplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping</p>
</td></tr>
<tr><td><code id="build_mid_dotplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="build_mid_dotplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>build_mid_dotplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_right_clusterplot'>Right intra-cluster linkage plot</h2><span id='topic+build_right_clusterplot'></span>

<h3>Description</h3>

<p>build_right_jitterplot() builds a horizontal plot displaying the mean
pairwise R^2 linkage between each SNP and all other SNPs in its marker group,
grouped by marker group, coloured by alternate allele frequency. Makes use of
the $Varfile information from haplotyping object. It is an internal function
called by crosshap_viz(), though can be called separately to build a
stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_right_clusterplot(HapObject, epsilon, hide_labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_right_clusterplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_right_clusterplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="build_right_clusterplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>build_right_clusterplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_right_phenoplot'>Right SNP-pheno phenoplot</h2><span id='topic+build_right_phenoplot'></span>

<h3>Description</h3>

<p>build_right_phenoplot() builds a horizontal plot displaying the mean
difference in phenotype score between individuals with the alternate vs
reference alleles for each SNP locus, grouped by marker group, coloured by
the alternate allele frequency of each SNP. Makes use of the $Varfile
phenotypic information from haplotyping object. It is an internal function
called by crosshap_viz(), though can be called separately to build a
stand-alone plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_right_phenoplot(HapObject, epsilon, hide_labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_right_phenoplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_right_phenoplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="build_right_phenoplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    build_right_phenoplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='build_summary_tables'>Hap/MG summary tables</h2><span id='topic+build_summary_tables'></span>

<h3>Description</h3>

<p>build_summary_tables() builds summary tables for each haplotype and Marker
Group with some of the information shown in the peripheral crosshap plots.
It is an internal function called by crosshap_viz(), though can be called
separately to build stand-along grob tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_summary_tables(HapObject, epsilon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_summary_tables_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="build_summary_tables_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two TableGrob objects.
</p>

<hr>
<h2 id='build_top_metaplot'>Top metadata-hap bar plot</h2><span id='topic+build_top_metaplot'></span>

<h3>Description</h3>

<p>build_top_metaplot() builds a vertical stacked bar plot displaying the
frequency of each haplotype combination, broken down by each categorical
metadata variable provided. Makes use of the $Indfile information from a
haplotype object. This is an in internal function called by crosshap_viz(),
though can be called separately to build a stand-alone plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_top_metaplot(HapObject, epsilon, hide_labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_top_metaplot_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping()</p>
</td></tr>
<tr><td><code id="build_top_metaplot_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="build_top_metaplot_+3A_hide_labels">hide_labels</code></td>
<td>
<p>If TRUE, legend is hidden.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>build_top_metaplot(HapObject, epsilon = 0.6, hide_labels = FALSE)

</code></pre>

<hr>
<h2 id='clustree_viz'>Clustering tree</h2><span id='topic+clustree_viz'></span>

<h3>Description</h3>

<p>clustree_viz() builds a clustering tree displaying changes in haplotype
assignment between individuals or changes in Marker Group assignment for
SNPs, across different epsilon values. This function is a 'clustree' wrapper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustree_viz(HapObject, type = "MG")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clustree_viz_+3A_hapobject">HapObject</code></td>
<td>
<p>A haplotyping object with a range of results from different
epsilons created by run_haplotyping()</p>
</td></tr>
<tr><td><code id="clustree_viz_+3A_type">type</code></td>
<td>
<p>When type = &quot;hap&quot;, nodes represent haplotype populations, when
type = &quot;MG&quot;, nodes represent marker groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object.
</p>

<hr>
<h2 id='crosshap_viz'>Visualize haplotypes</h2><span id='topic+crosshap_viz'></span>

<h3>Description</h3>

<p>crosshap_viz() builds five individual plots using various elements of a
HapObject created by run_haplotyping(). The central dotplot displays
relationship between clusters of linked SNPs (Marker Groups), and distinct
haplotypes present within the population. Vertical plots (top/bottom)
visualize individuals and populations, grouped by haplotype. Horizontal plots
(left/right) visualize SNP information, grouped by Marker Group cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crosshap_viz(
  HapObject,
  epsilon,
  plot_left = "allele",
  plot_right = "pheno",
  hide_labels = FALSE,
  isolate_group = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crosshap_viz_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="crosshap_viz_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon to visualize haplotyping results for.</p>
</td></tr>
<tr><td><code id="crosshap_viz_+3A_plot_left">plot_left</code></td>
<td>
<p>When plot_left = &quot;allele&quot;, SNP allele frequency information
is displayed, when plot_left = &quot;pos&quot;, SNP position information is displayed.</p>
</td></tr>
<tr><td><code id="crosshap_viz_+3A_plot_right">plot_right</code></td>
<td>
<p>When plot_right = &quot;pheno&quot;, phenotype associations for SNPs
are displayed, when plot_right = &quot;cluster&quot;, internal marker group linkage is
displayed.</p>
</td></tr>
<tr><td><code id="crosshap_viz_+3A_hide_labels">hide_labels</code></td>
<td>
<p>When TRUE, legends from plots are hidden.</p>
</td></tr>
<tr><td><code id="crosshap_viz_+3A_isolate_group">isolate_group</code></td>
<td>
<p>If one or more Metadata groups are provided, all other
Metadata groups will be masked from the plot. NOTE: it does change the
summary tables or marker group phenotype scores.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A patchwork object.
</p>

<hr>
<h2 id='crosshap-package'>crosshap: Local Haplotype Clustering and Visualization</h2><span id='topic+crosshap'></span><span id='topic+crosshap-package'></span>

<h3>Description</h3>

<p>A local haplotyping visualization toolbox to capture major patterns of co-inheritance between clusters of linked variants, whilst connecting findings to phenotypic and demographic traits across individuals. 'crosshap' enables users to explore and understand genomic variation across a trait-associated region. For an example of successful local haplotype analysis, see Marsh et al. (2022) <a href="https://doi.org/10.1007/s00122-022-04045-8">doi:10.1007/s00122-022-04045-8</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Jacob Marsh <a href="mailto:jake.marsh@live.com.au">jake.marsh@live.com.au</a> (<a href="https://orcid.org/0000-0003-3734-2023">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Brady Johnston <a href="mailto:brady.johnston@me.com">brady.johnston@me.com</a> (<a href="https://orcid.org/0000-0001-6301-2269">ORCID</a>)
</p>
</li>
<li><p> Jakob Petereit <a href="mailto:petinho86@gmail.com">petinho86@gmail.com</a> (<a href="https://orcid.org/0000-0003-2159-0380">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://jacobimarsh.github.io/crosshap/">https://jacobimarsh.github.io/crosshap/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/jacobimarsh/crosshap/issues">https://github.com/jacobimarsh/crosshap/issues</a>
</p>
</li></ul>


<hr>
<h2 id='HapObject'>Example Haplotype object</h2><span id='topic+HapObject'></span>

<h3>Description</h3>

<p>A haplotyping object created by run_haplotyping() for example cqProt-003 soy data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HapObject
</code></pre>


<h3>Format</h3>

<p>A haplotype (S3) object containing results needed for haplotype visualization across five epsilon values (0.2,0.4,0.6,0.8,1)
</p>

<dl>
<dt>epsilon</dt><dd><p>Epsilon value chosen for haplotyping with DBSCAN</p>
</dd>
<dt>MGmin</dt><dd><p>MGmin value (minPts) chosen for haplotyping with DBSCAN</p>
</dd>
<dt>Hapfile</dt><dd><p>Summary of Marker Groups defining haplotype combinations</p>
</dd>
<dt>Indfile</dt><dd><p>Haplotype assignments for individuals</p>
</dd>
<dt>Varfile</dt><dd><p>Marker Group assignments for SNPs, with additional calculated information</p>
</dd>
</dl>


<hr>
<h2 id='LD'>Example LD matrix</h2><span id='topic+LD'></span>

<h3>Description</h3>

<p>A pairwise R^2 linkage matrix generated by PLINK for example cqProt-003 soy data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LD
</code></pre>


<h3>Format</h3>

<p>A square matrix read in by read_LD()
</p>

<hr>
<h2 id='mean_na.rm'>Mean utility function</h2><span id='topic+mean_na.rm'></span>

<h3>Description</h3>

<p>Mean utility function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean_na.rm(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mean_na.rm_+3A_x">x</code></td>
<td>
<p>Input vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mean numerical values
</p>

<hr>
<h2 id='metadata'>Example Domestication metadata</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>Metadata file with level of domestication for each individual in example cqProt-003 soy data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata
</code></pre>


<h3>Format</h3>

<p>A two-column tibble read in by read_metadata()
</p>


<h3>Source</h3>

<p>https://doi.org/10.1007/s00122-022-04045-8
</p>

<hr>
<h2 id='pheno'>Example phenotype data</h2><span id='topic+pheno'></span>

<h3>Description</h3>

<p>Seed protein scores for each individual in example cqProt-003 soy data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pheno
</code></pre>


<h3>Format</h3>

<p>A two-column tibble read in by read_pheno()
</p>


<h3>Source</h3>

<p>https://doi.org/10.1007/s00122-022-04045-8
</p>

<hr>
<h2 id='prepare_hap_umap'>UMAP haplotype visualization helper</h2><span id='topic+prepare_hap_umap'></span>

<h3>Description</h3>

<p>prepare_hap_umap() builds a large composite ggplot2 object ready for faceting
and animation (see vignette) for visualizing SNP alleles (coloured by Marker
Group) possessed by individuals with each haplotype. UMAP
coordinates for each SNP can be generated using umap::umap(), with the LD
matrix generated for run_haplotyping() as input. When fully rendered and
faceted, the resultant GIF intuitively visualizes the shared loci within each
Marker Group that are constant within each haplotype combination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_hap_umap(
  umap_in,
  hetmiss_as = "allele",
  HapObject,
  epsilon,
  vcf,
  nsamples = 25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_hap_umap_+3A_umap_in">umap_in</code></td>
<td>
<p>UMAP results produced for a haplotype object at a given epsilon.</p>
</td></tr>
<tr><td><code id="prepare_hap_umap_+3A_hetmiss_as">hetmiss_as</code></td>
<td>
<p>If hetmiss_as = &quot;allele&quot;, heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = &quot;miss&quot;, the site is recoded as missing.</p>
</td></tr>
<tr><td><code id="prepare_hap_umap_+3A_hapobject">HapObject</code></td>
<td>
<p>Haplotype object created by run_haplotyping().</p>
</td></tr>
<tr><td><code id="prepare_hap_umap_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon matching the haplotype object used for umap_in.</p>
</td></tr>
<tr><td><code id="prepare_hap_umap_+3A_vcf">vcf</code></td>
<td>
<p>Input vcf.</p>
</td></tr>
<tr><td><code id="prepare_hap_umap_+3A_nsamples">nsamples</code></td>
<td>
<p>Number of times to sample each haplotype group, will directly
translate to the number of frames in animation. Should be the same as the
nframes passed to gganimate::animate().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A large ggplot2 object.
</p>

<hr>
<h2 id='pseudo_haps'>Identify haplotypes from clustered SNPs</h2><span id='topic+pseudo_haps'></span>

<h3>Description</h3>

<p>pseudo_haps() calls the most common allelic states for each SNP marker group
across individuals, before building dummy SNPs for each marker group that
mimic the binary vcf format. This is the step which determines the haplotype
combinations, and therefore enables several summaries to be returned - as
contained in the $Hapfile and preliminary $Indfile and finalised $MGfile,
following marker group smoothing. This is an internal function not intended
for external use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pseudo_haps(preMGfile, bin_vcf, minHap, LD, keep_outliers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pseudo_haps_+3A_premgfile">preMGfile</code></td>
<td>
<p>SNP clusters from DBscan.</p>
</td></tr>
<tr><td><code id="pseudo_haps_+3A_bin_vcf">bin_vcf</code></td>
<td>
<p>Binary VCF for region of interest reformatted by
run_haplotyping().</p>
</td></tr>
<tr><td><code id="pseudo_haps_+3A_minhap">minHap</code></td>
<td>
<p>Minimum size (nIndividuals) to keep haplotype combinations</p>
</td></tr>
<tr><td><code id="pseudo_haps_+3A_ld">LD</code></td>
<td>
<p>LD matrix input.</p>
</td></tr>
<tr><td><code id="pseudo_haps_+3A_keep_outliers">keep_outliers</code></td>
<td>
<p>When FALSE, marker group smoothing is performed to
remove outliers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns intermediate of haplotype object
</p>

<hr>
<h2 id='read_LD'>Read LD correlation matrix to tibble</h2><span id='topic+read_LD'></span>

<h3>Description</h3>

<p>If your correlation matrix does not have rownames and column names, a VCF
will need to be provided so it can be added with read_LD().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_LD(LDin, vcf = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_LD_+3A_ldin">LDin</code></td>
<td>
<p>Square correlation matrix</p>
</td></tr>
<tr><td><code id="read_LD_+3A_vcf">vcf</code></td>
<td>
<p>VCF object created by read_vcf() that can be used to assign column names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='read_metadata'>Read metadata to tibble</h2><span id='topic+read_metadata'></span>

<h3>Description</h3>

<p>Requires two column text file without a header (Ind | Metadata)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_metadata(Metain)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_metadata_+3A_metain">Metain</code></td>
<td>
<p>Input phenotype file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='read_pheno'>Read phenotype data to tibble</h2><span id='topic+read_pheno'></span>

<h3>Description</h3>

<p>Requires two column text file without a header (Ind | Pheno)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_pheno(Phenoin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_pheno_+3A_phenoin">Phenoin</code></td>
<td>
<p>Input phenotype file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='read_vcf'>Read VCF to tibble</h2><span id='topic+read_vcf'></span>

<h3>Description</h3>

<p>Dashes,'-', in individual names are recoded to '.' for downstream
compatability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_vcf(VCFin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_vcf_+3A_vcfin">VCFin</code></td>
<td>
<p>Input VCF</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble.
</p>

<hr>
<h2 id='run_haplotyping'>Cluster SNPs and identify haplotypes</h2><span id='topic+run_haplotyping'></span>

<h3>Description</h3>

<p>run_haplotyping() performs density-based clustering of SNPs in region of
interest to identify Marker Groups. Individuals are classified by haplotype
combination based on shared combinations of Marker Group alleles. Returns a
haplotyping object (HapObject), which can be used as input to build clustering
tree for epsilon optimization using clustree_viz(), and can be visualized with
reference to phenotype and metadata using crosshap_viz().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_haplotyping(
  vcf,
  LD,
  pheno,
  metadata = NULL,
  epsilon = c(0.2, 0.4, 0.6, 0.8, 1),
  MGmin = 30,
  minHap = 9,
  hetmiss_as = "allele",
  het_phenos = FALSE,
  keep_outliers = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_haplotyping_+3A_vcf">vcf</code></td>
<td>
<p>Input VCF for region of interest.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_ld">LD</code></td>
<td>
<p>Pairwise correlation matrix of SNPs in region (e.g. from PLINK).</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_pheno">pheno</code></td>
<td>
<p>Input numeric phenotype data for each individual.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_metadata">metadata</code></td>
<td>
<p>Metadata input (optional).</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_epsilon">epsilon</code></td>
<td>
<p>Epsilon values for clustering SNPs with DBscan.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_mgmin">MGmin</code></td>
<td>
<p>Minimum SNPs in marker groups, MinPts parameter for DBscan.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_minhap">minHap</code></td>
<td>
<p>Minimum nIndividuals in a haplotype combination.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_hetmiss_as">hetmiss_as</code></td>
<td>
<p>If hetmiss_as = &quot;allele&quot;, heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = &quot;miss&quot;, the site is recoded as missing.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_het_phenos">het_phenos</code></td>
<td>
<p>When FALSE, phenotype associations for SNPs are calculated
from reference and alternate allele individuals only, when TRUE, heterozygous
individuals are included assuming additive allele effects.</p>
</td></tr>
<tr><td><code id="run_haplotyping_+3A_keep_outliers">keep_outliers</code></td>
<td>
<p>When FALSE, marker group smoothing is performed to
remove outliers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A comprehensive haplotyping S3 object (HapObject) for each provided
epsilon value, needed for clustree_viz() and crosshap_viz().
</p>

<hr>
<h2 id='run_hdbscan_haplotyping'>Cluster SNPs with HDBSCAN and identify haplotypes</h2><span id='topic+run_hdbscan_haplotyping'></span>

<h3>Description</h3>

<p>run_hdbscan_haplotyping() performs HDBSCAN clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be visualized with reference
to phenotype and metadata using crosshap_viz() (set epsilon to 1 as a dummy value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_hdbscan_haplotyping(
  vcf,
  LD,
  pheno,
  MGmin,
  minHap = 5,
  hetmiss_as = "allele",
  metadata = NULL,
  keep_outliers = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_hdbscan_haplotyping_+3A_vcf">vcf</code></td>
<td>
<p>Input VCF for region of interest.</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_ld">LD</code></td>
<td>
<p>Pairwise correlation matrix of SNPs in region (e.g. from PLINK).</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_pheno">pheno</code></td>
<td>
<p>Input numeric phenotype data for each individual.</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_mgmin">MGmin</code></td>
<td>
<p>Minimum SNPs in marker groups, MinPts parameter for DBscan.</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_minhap">minHap</code></td>
<td>
<p>Minimum nIndividuals in a haplotype combination.</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_hetmiss_as">hetmiss_as</code></td>
<td>
<p>If hetmiss_as = &quot;allele&quot;, heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = &quot;miss&quot;, the site is recoded as missing.</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_metadata">metadata</code></td>
<td>
<p>Metadata input (optional).</p>
</td></tr>
<tr><td><code id="run_hdbscan_haplotyping_+3A_keep_outliers">keep_outliers</code></td>
<td>
<p>When FALSE, marker group smoothing is performed to
remove outliers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A comprehensive haplotyping S3 object (HapObject) for each provided
epsilon value, needed for clustree_viz() and crosshap_viz().
</p>

<hr>
<h2 id='tagphenos'>Calculate SNP phenotypic associations</h2><span id='topic+tagphenos'></span>

<h3>Description</h3>

<p>tagphenos() reports the frequency of allele types for each SNP and calculates
phenotype associations for the different alleles, before returning this
information in a $Varfile in a HapObject. This is an internal function that
is not intended for external use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tagphenos(MGfile, bin_vcf, pheno, het_phenos = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tagphenos_+3A_mgfile">MGfile</code></td>
<td>
<p>SNP marker groups clustered using DBscan.</p>
</td></tr>
<tr><td><code id="tagphenos_+3A_bin_vcf">bin_vcf</code></td>
<td>
<p>Binary VCF for region of interest reformatted by
run_haplotyping().</p>
</td></tr>
<tr><td><code id="tagphenos_+3A_pheno">pheno</code></td>
<td>
<p>Input numeric phenotype data for each individual.</p>
</td></tr>
<tr><td><code id="tagphenos_+3A_het_phenos">het_phenos</code></td>
<td>
<p>When FALSE, phenotype associations for SNPs are calculated
from reference and alternate allele individuals only, when TRUE, heterozygous
individuals are included assuming additive allele effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns intermediate of haplotype object.
</p>

<hr>
<h2 id='vcf'>Example VCF</h2><span id='topic+vcf'></span>

<h3>Description</h3>

<p>A VCF containing SNPs for example cqProt-003 soy data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf
</code></pre>


<h3>Format</h3>

<p>A VCF read in by read_vcf()
</p>


<h3>Source</h3>

<p>https://doi.org/10.1007/s00122-022-04045-8
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
