<!DOCTYPE html><html><head><title>Help for package CrossValidate</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CrossValidate}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#balancedSplit'>
<p>Split a dataset into training and testing sets, balancing a factor</p></a></li>
<li><a href='#CrossValidate'><p>Creating CrossValidate objects</p></a></li>
<li><a href='#CrossValidate-class'><p>Class &quot;CrossValidate&quot;</p></a></li>
<li><a href='#CrossValSummary-class'><p>Class &quot;CrossValSummary&quot;</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-06</td>
</tr>
<tr>
<td>Title:</td>
<td>Classes and Methods for Cross Validation of "Class Prediction"
Algorithms</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0), Modeler</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, oompaBase (&ge; 3.0.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biobase</td>
</tr>
<tr>
<td>Description:</td>
<td>Defines classes and methods to cross-validate various
  binary classification algorithms used for "class prediction"
  problems.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microarray, Clustering</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org">http://oompa.r-forge.r-project.org</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-06 17:10:23 UTC; Kevin Coombes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-06 17:40:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='balancedSplit'>
Split a dataset into training and testing sets, balancing a factor
</h2><span id='topic+balancedSplit'></span>

<h3>Description</h3>

<p>When performing cross-validation on a dataset, it often becomes
necessary to split the data into training and test sets that are
balanced for a factor.  This function implements such a
balanced split.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>balancedSplit(fac, size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="balancedSplit_+3A_fac">fac</code></td>
<td>
<p>A factor that should be balanced between the two subsets.</p>
</td></tr>
<tr><td><code id="balancedSplit_+3A_size">size</code></td>
<td>

<p>A number between 0 and 1 indicating the fraction of the dataset
to be used for training.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function randomly samples the same fraction of items from each
level of a factor to include in a training set.  In most cases, this
will be a binary factor (and might even be the outcome that one wants to
predict).  However, the implementation works for factors with an
arbitrary number of levels. 
</p>


<h3>Value</h3>

<p>Returns a logical vector with length equal to the length of
<code>fac</code>. TRUE values designate samples selected for the training
set.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes &lt;krc@silicovore.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CrossValidate">CrossValidate</a></code>, <code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nFeatures &lt;- 40
nSamples &lt;- 2*10
dataset &lt;- matrix(rnorm(nSamples*nFeatures), ncol=nSamples)
groups &lt;- factor(rep(c("A", "B"), each=10))
balancedSplit(dataset, groups)
</code></pre>

<hr>
<h2 id='CrossValidate'>Creating CrossValidate objects</h2><span id='topic+CrossValidate'></span>

<h3>Description</h3>

<p>Given a model classifier and a data set, this function performs
cross-validation by repeatedly splitting the data into training and
testing subsets in order to estimate the performance of this kind of
classifer on new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CrossValidate(model, data, status, frac, nLoop, prune=keepAll, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CrossValidate_+3A_model">model</code></td>
<td>

<p>An element of the <code><a href="Modeler.html#topic+Modeler">Modeler</a></code> class.
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_data">data</code></td>
<td>

<p>A matrix containing the data to be used for cross-validation. As with
most gene expression data, columns are the independent samples or
observations and rows are the measured features.
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_status">status</code></td>
<td>

<p>A binary-valued factor with the classes to be predicted.
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_frac">frac</code></td>
<td>

<p>A number between 0 and 1; the fraction of the data that should be used
in each iteration to train the model. 
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_nloop">nLoop</code></td>
<td>

<p>An integer; the number of times to split the data into training and test
sets. 
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_prune">prune</code></td>
<td>

<p>A function  that takes two inoputs, a data matrix and a factor with
two levels, and rteturns a logical vector whose length equals the
number of rows in the data matrix.
</p>
</td></tr>
<tr><td><code id="CrossValidate_+3A_verbose">verbose</code></td>
<td>

<p>A logical value; should the cross-validation routine report interim
progress.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>CrossValidate</code> package provides generic tools for performing
cross-validation on classificaiton methods in the context of
high-throughput data sets such as those produced by gene expression
microarrays. In order to use a classifier with this implementaiton of
cross-validation, you must first prepare a pair of functions (one for
learning models from training data, and one for making predictions on
test data). These functions, along with any required meta-parameters,
are used to create an object of the <code><a href="Modeler.html#topic+Modeler-class">Modeler-class</a></code>. That
object is then passed to the <code><a href="#topic+CrossValidate">CrossValidate</a></code> function along
with the full training data set. The full data set is then repeatedly
split into its own training and test sets; you can specify the
fraction to be used for training and the number of iterations. The
result is a detailed look at the accuracy, sensitivity, specificity,
and positive and negative predictive value of the model, as estimated
by cross-validation.
</p>


<h3>Value</h3>

<p>An object of the <code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code>.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krcoombes@mdanderson.org">krcoombes@mdanderson.org</a>
</p>


<h3>References</h3>

<p>See the manual page for the <code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code> for a list
of related references.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code> for a description of the slots in
the object created by this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- matrix(rnorm(50*100), nrow=50)
pseudoclass &lt;- factor(rep(c("A", "B"), each=50))
model &lt;- modelerCCP # obviously, other models can be used
numTimes &lt;- 10 # and more is probably better
cv &lt;- CrossValidate(model, dataset, pseudoclass, 0.5, numTimes)
summary(cv)
</code></pre>

<hr>
<h2 id='CrossValidate-class'>Class &quot;CrossValidate&quot;</h2><span id='topic+CrossValidate-class'></span><span id='topic+summary+2CCrossValidate-method'></span>

<h3>Description</h3>

<p>A class that contains the results of internal cross-validation (by
multiple splits into training and test sets) of an algorithm that
builds a model to predict a binary outcome.
</p>


<h3>Objects from the Class</h3>

<p>Objects should be created by calls to the constructor function,
<code><a href="#topic+CrossValidate">CrossValidate</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>nIterations</code>:</dt><dd><p>An integer; the number of times the
data was split into training and test sets.</p>
</dd>
<dt><code>trainPercent</code>:</dt><dd><p>A number between 0 and 1; the fraction
of data used in each training set.</p>
</dd>
<dt><code>outcome</code>:</dt><dd><p>A binary factor containing the true outcome
for each sample.</p>
</dd>
<dt><code>trainOutcome</code>:</dt><dd><p>A data frame containing the true
outcomes for each member of the training set.  The value 'NA' is
used for samples that were reserved for testing. Each column is a
different split into training and test sets.</p>
</dd> 
<dt><code>trainPredict</code>:</dt><dd><p>A data frame containing the predicted
outcome from the model for each member of the training set.  The
value 'NA' is used for samples that were reserved for testing.
Each column is a different split into training and test sets.</p>
</dd>
<dt><code>validOutcome</code>:</dt><dd><p>A data frame containing the true
outcomes for each member of the test set.  The
value 'NA' is used for samples that were used for training.
Each column is a different split into training and test sets.</p>
</dd>
<dt><code>validPredict</code>:</dt><dd><p>A data frame containing the predicted
outcome from the model for each member of the test set.  The
value 'NA' is used for samples that were used for training.
Each column is a different split into training and test sets.</p>
</dd>
<dt><code>extras</code>:</dt><dd><p>A list, whose length equals the number of
plsits into trainin and test sets.  Each entry contains any
&quot;extra&quot; information collected during the fitting of the model; the
kinds of items stored here depend on the actual classification
algorithm used.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>summary</dt><dd><p><code>signature(object = "CrossValidate")</code>: Produces
a summary of the performance of the algorithm on both the trinaing
sets and the test sets, in terms of specificity, sensitivity, and
positive or negative predictive value.  Specifically, this method
returns an object of the <code><a href="#topic+CrossValSummary-class">CrossValSummary-class</a></code>.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes &lt;krcoombes@mdanderson.org&gt;
</p>


<h3>References</h3>

<p>Braga-Neto U, Dougherty ER.<br />
<em>Is cross-validation valid for small-sample microarray
classification?</em><br />
Bioinformatics, 2004; <b>20</b>:374&ndash;380.
</p>
<p>Jiang W, Varma S, Simon R.<br />
<em>Calculating confidence intervals for
prediction error in microarray classification using resampling.</em><br />
Stat Appl Genet Mol Biol. 2008; <b>7</b>:Article8.
</p>
<p>Fu LM, Youn ES.<br />
<em>Improving reliability of gene selection from
microarray functional genomics data.</em><br />
IEEE Trans Inf Technol Biomed. 2003; <b>7</b>:191&ndash;6.
</p>
<p>Man MZ, Dyson G, Johnson K, Liao B.<br />
<em>Evaluating methods for classifying expression data.</em><br />
J Biopharm Stat. 2004; <b>14</b>:1065&ndash;84.
</p>
<p>Fu WJ, Carroll RJ, Wang S.<br />
<em>Estimating misclassification error with small samples via
bootstrap cross-validation.</em><br />
Bioinformatics, 2005; <b>21</b>:1979&ndash;86.
</p>
<p>Ancona N, Maglietta R, Piepoli A, D'Addabbo A, Cotugno R, Savino M,
Liuni S, Carella M, Pesole G, Perri F.<br />
<em>On the statistical assessment of classifiers using DNA
microarray data.</em><br />
BMC Bioinformatics, 2006; <b>7</b>:387.
</p>
<p>Lecocke M, Hess K.<br />
<em>An empirical study of univariate and genetic
algorithm-based feature selection in binary classification with
microarray data.</em><br />
Cancer Inform, 2007; <b>2</b>:313&ndash;27.
</p>
<p>Lee S.<br />
<em>Mistakes in validating the accuracy of a prediction classifier
in high-dimensional but small-sample microarray data.</em><br />
Stat Methods Med Res, 2008; <b>17</b>:635&ndash;42.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+CrossValidate">CrossValidate</a></code> for the constructor function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("CrossValidate")
</code></pre>

<hr>
<h2 id='CrossValSummary-class'>Class &quot;CrossValSummary&quot;</h2><span id='topic+CrossValSummary-class'></span><span id='topic+show+2CCrossValSummary-method'></span>

<h3>Description</h3>

<p>Represents the effect of summarizing a <code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code> object
by computing the performance of predictions on each split into
training and test sets.
</p>


<h3>Objects from the Class</h3>

<p>Objects are almost always created automatically by applying the
<code>summary</code> method to an object of the
<code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>call</code>:</dt><dd><p>A <code>"call"</code> object recoding how the summary
method was invoked. </p>
</dd>
<dt><code>parent</code>:</dt><dd><p>A character vector containing the name of the
<code>CrossValidate</code> object being summarized.</p>
</dd>
<dt><code>trainAcc</code>:</dt><dd><p>A <code>"list"</code> containing five numeric
vectors: the sensitivity, specificity, accuracy, PPV, and NPV on
each training data set.</p>
</dd> 
<dt><code>validAcc</code>:</dt><dd><p>A <code>"list"</code> containing five numeric
vectors: the sensitivity, specificity, accuracy, PPV, and NPV on
each test data set.</p>
</dd> 
</dl>



<h3>Methods</h3>


<dl>
<dt>show</dt><dd><p><code>signature(object = "CrossValSummary")</code>: Summarizes
the algorithm performance.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes &lt;krcoombes@mdanderson.org&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CrossValidate-class">CrossValidate-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("CrossValSummary")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
