<!DOCTYPE html><html><head><title>Help for package CSclone</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CSclone}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust.mpear'><p>Adjust MPEAR</p></a></li>
<li><a href='#CSclone'><p>Clustering model which contains the CNAs mutation and SNVs mutation.</p></a></li>
<li><a href='#Hamming'><p>Hamming distance</p></a></li>
<li><a href='#l.s'><p>Linear relation for SNVs</p></a></li>
<li><a href='#simu.data'><p>Simulate data</p></a></li>
<li><a href='#tf_similar'><p>True or false of similar</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Nonparametric Modeling in R</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-11-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter Wu &lt;peter123wu0@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter Wu &lt;peter123wu0@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Germline and somatic locus data which contain the total read depth and B allele 
    read depth using Bayesian model (Dirichlet Process) to cluster. Meanwhile, the cluster 
    model can deal with the SNVs mutation and the CNAs mutation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>lpSolve, mcclust, moments, DNAcopy, stats, R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-11-11 17:13:34 UTC; Administrator</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-11-12 00:52:58</td>
</tr>
</table>
<hr>
<h2 id='adjust.mpear'>Adjust MPEAR</h2><span id='topic+adjust.mpear'></span>

<h3>Description</h3>

<p>adjust.mpear is modified the mcclust package to suit our model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust.mpear(psm, method = "ward.D2", max.k = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjust.mpear_+3A_psm">psm</code></td>
<td>
<p>is posterior similarity matrix with entries between 0 and 1 and 1's on the diagonals which is distance matrix.</p>
</td></tr>
<tr><td><code id="adjust.mpear_+3A_method">method</code></td>
<td>
<p>is clustering method. We offer many kinds of method, like as kmeans and hierarchical clustering. The default is ward.D2 of hierarchical clustering.</p>
</td></tr>
<tr><td><code id="adjust.mpear_+3A_max.k">max.k</code></td>
<td>
<p>is number with the user has the professional judgment to consider the max group and the default is NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>adjust.mpear is modified the mcclust package to suit our model. And the function use distance matrix by criteria of MPEAR to cluster the data.
</p>


<h3>Value</h3>

<p>cl is vector with the clustering result.
</p>
<p>value is number with MPEAR.
</p>
<p>method is clustering method.
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>


<h3>References</h3>

<p>Fritsch, A. and Ickstadt, K. Improved criteria for clustering based on the posterior similarity matrix. Bayesian analysis 2009;4(2):367-391.
</p>

<hr>
<h2 id='CSclone'>Clustering model which contains the CNAs mutation and SNVs mutation.</h2><span id='topic+CSclone'></span>

<h3>Description</h3>

<p>CSclone is the main function of the package and a clustering model which contains the CNAs mutation and SNVs mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSclone(somatic.id, DNAcopy.object, mcmc = list(nburn = 5000, nsave = 10000,
  nskip = 1, ndisplay = 1000), y, set = 100, alpha = 1, max.k = NULL,
  method = "ward.D2", prior = c(1, 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSclone_+3A_somatic.id">somatic.id</code></td>
<td>
<p>is a vector of s observiations and mark which loci has somatic mutation.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_dnacopy.object">DNAcopy.object</code></td>
<td>
<p>is a list and output of the DNAcopy package.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_mcmc">mcmc</code></td>
<td>
<p>is a list giving the MCMC parameters. The list must include the following integers: nburn giving the number of burn-in scans, nskip giving the thinning interval, nsave giving the total number of scans to be saved, and ndisplay giving the number of saved scans to be displayed on screen (the function reports on the screen when every ndisplay iterations have been carried out) and default is list(nburn=5000,nsave=10000,nskip=1,ndisplay=1000).</p>
</td></tr>
<tr><td><code id="CSclone_+3A_y">y</code></td>
<td>
<p>is a matrix giving the binomial data. The first column is B allele read depth and the second column is total read depth.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_set">set</code></td>
<td>
<p>is number that you can reproduce the simulation result and default is 100.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_alpha">alpha</code></td>
<td>
<p>is a number giving the concentration parameter of Dirichlet process and default is 1.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_max.k">max.k</code></td>
<td>
<p>is a number giving limit the maximum cluster number and default is NULL.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_method">method</code></td>
<td>
<p>is the agglomeration method to be used and the method is used to result of Hamming distance. The default is ward.D2.</p>
</td></tr>
<tr><td><code id="CSclone_+3A_prior">prior</code></td>
<td>
<p>is a vector has tow number which is the parameter of Beta distribution and the default is (1,1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>CSclone is a two steps model. The first step is clustering the non-CNAs somatic mutations by DPMM with Binomial distribution and the posterior distribution is Beta distribution. The second step classifies the CNAs mutation and the somatic mutation with CNAs mutation. There are an example and the dataset is small case in order to run quickly.
</p>


<h3>Value</h3>

<p>cluster is a vector giving the group result of the somatic mutation.
</p>
<p>segment is a matrix. The first column is chromosome, the second column is starting position, the third column is end position, the fourth column is copy number, the fifth column is proportion of mutation, the sixth is number of loci, the seventh column is starting loci number, and the eighth column is end loci number.
</p>
<p>mcmc is a list giving the MCMC parameters.
</p>
<p>alpha is number giving the concentration parameter of Dirichlet process.
</p>
<p>y is a matrix giving the binomial data.
</p>
<p>group.prop is number giving the predict the proportion of group.
</p>
<p>SNV is matrix giving the somatic mutation of every loci. The first column is B allele read depth, the second column is total read depth, the third column proportion of somatic mutation.
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mcmc=list(nburn=200,nsave=500,nskip=1,ndisplay=1000)
p=c(0.2,0.4,0.6,0.8)
chrs=rep(c(1:2),times=rep(500,2))
pos=sort(sample(size=1000,x=1:10^7))
pc=sample(rep(p,4*c(0,0,0.5,0.5)))
ps=sample(rep(p,50*c(0.3,0.3,0.2,0.2)))
x=simu.data(n.germline=1000,pc=pc,read=200,ps=ps,dis="Negative binomial",parameter=0.75)
snv.id=x$snv.id
y=x$y
row.names(y)=paste0(chrs,"_",pos)
logR=log(y[,2],base=2)-median(log(y[,2],base=2))
data=data.frame(chr=chrs,pos=pos,logR=logR)
rownames(data)=paste0("SNP",1:nrow(data))
CNA.object=CNA(data$logR,data$chr,data$pos,data.type="logratio",sampleid="test")
smoothed.CNA.object=smooth.CNA(CNA.object)
DNAcopy.object=segment(smoothed.CNA.object, undo.splits = "sdundo",undo.SD = 3, verbose = 1)
fit=CSclone(somatic.id=snv.id,DNAcopy.object=DNAcopy.object,mcmc=mcmc,y=y)
result=fit$cluster
tf_similar(real=ps,cluster=result)

</code></pre>

<hr>
<h2 id='Hamming'>Hamming distance</h2><span id='topic+Hamming'></span>

<h3>Description</h3>

<p>Hamming is computing the co-occurrence matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hamming(k.matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hamming_+3A_k.matrix">k.matrix</code></td>
<td>
<p>is matrix with the result-retaining of MCMC.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hamming use the result-retaining of MCMC to compute the co-occurrence matrix. And we use the co-occurrence matrix to cluster which the criteria is MPEAR.
</p>


<h3>Value</h3>

<p>Hamming.matrix is the co-occurrence matrix and the idea is Hamming distance.
</p>
<p>pro.matrix is the standardization of co-occurrence matrix.
</p>
<p>data is data.frame with matrix. The purpose is constructing the heat map.
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>

<hr>
<h2 id='l.s'>Linear relation for SNVs</h2><span id='topic+l.s'></span>

<h3>Description</h3>

<p>l.s is the linear relation between B allele frequence and the proportion of SNVs mutation(PS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l.s(cnt, cnb1, cnb2, pc = 0, ps = NULL, baf = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="l.s_+3A_cnt">cnt</code></td>
<td>
<p>is number with the total copy number after the CNAs mutation.</p>
</td></tr>
<tr><td><code id="l.s_+3A_cnb1">cnb1</code></td>
<td>
<p>is number with the B copy number of two types of periods. The first period, the proportion of CNAs mutation is more than the proportion of SNVs mutation then cnb1 is B copy number after SNVs mutation. The second period, the proportion of CNAs mutation is less than the proportion of SNVs mutation then cnb1 is B copy number between the SNVs mutation and CNAs mutation.</p>
</td></tr>
<tr><td><code id="l.s_+3A_cnb2">cnb2</code></td>
<td>
<p>is number with the B copy number of two types of periods. The first period, the proportion of CNAs mutation is less than the proportion of SNVs mutation then cnb1 is B copy number after CNAs mutation. The second period, the proportion of CNAs mutation is more than the proportion of SNVs mutation then cnb1 is B copy number between the CNAs mutation and SNVs mutation.</p>
</td></tr>
<tr><td><code id="l.s_+3A_pc">pc</code></td>
<td>
<p>is number with the proportion of CNAs mutation and the default is 0.</p>
</td></tr>
<tr><td><code id="l.s_+3A_ps">ps</code></td>
<td>
<p>is number with the proportion of SNVs mutation and the default is NULL.</p>
</td></tr>
<tr><td><code id="l.s_+3A_baf">baf</code></td>
<td>
<p>is number with the B allele frequence and the default is NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>l.s is bidirection function. The first function is given the proportion of SNVs mutation(ps) to predict the B allele frequence(baf). The second function is given the B allele frequence(baf) to predict the proportion of SNVs mutation(ps).
</p>


<h3>Value</h3>

<p>baf is number with B allele frequence(baf) if the input is given the proportion of SNVs mutation(ps).
</p>
<p>ps is number with the proportion of SNVs mutation(ps) if the input is given the B allele frequence(baf).
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>

<hr>
<h2 id='simu.data'>Simulate data</h2><span id='topic+simu.data'></span>

<h3>Description</h3>

<p>simu.data is generating the data which simulates the locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simu.data(n.germline = 10000, segment.length = 100, seed = NULL, pc,
  read = 100, ps, dis = "Normal", parameter = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simu.data_+3A_n.germline">n.germline</code></td>
<td>
<p>is number with the total number of generating data which contains non-CNAs and non-SNVs. The default is 10000.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_segment.length">segment.length</code></td>
<td>
<p>is number with the number of loci of every segment and the number is common divisor of n.germline. The default is 100.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_seed">seed</code></td>
<td>
<p>is number that you can reproduce the simulation result and default is NULL.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_pc">pc</code></td>
<td>
<p>is vector with the proportion of CNAs mutation of every segment.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_read">read</code></td>
<td>
<p>is number with the standard total read depth.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_ps">ps</code></td>
<td>
<p>is vector with the proportion of SNVs mutation.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_dis">dis</code></td>
<td>
<p>is distribution with generating the data and the default is Normal distribution. We provide Poisson and Negative binomial distribution to choose.</p>
</td></tr>
<tr><td><code id="simu.data_+3A_parameter">parameter</code></td>
<td>
<p>is number and default is NULL. If dis is Normal and parameter is not NULL, then the standard deviation is parameter x mean. If dis is Normal and parameter is NULL, then the standard deviation is mean. If dis is Negative binomial and parameter is NULL, then the parameter is second parameter. If dis is Negative binomial and parameter NULL, then the second parameter is 0.5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The simu.data can consider not only SNVs mutation but also CNAs mutation and generate the data which contains germline data and mutation data.
</p>


<h3>Value</h3>

<p>y is n x 2 matrix. The first column is the B allele read depth and the second column is the total read depth.
</p>
<p>snv.id is vector which denotes which loci with SNVs mutation.
</p>
<p>cnt.id is A x B matrix which denotes which loci with CNAs mutation.. The A is segment.length and the B is the number of segment with CNAs mutation.
</p>
<p>cnv is n x 4 matrix. The first column is the total copy number after CNAs mutation, the second column is the cnb1, the third column is the cnb2, and the fourth column is the proportion of CNAs mutation. The cnb1 and the cnb2 has detailed explanation at l.s function.
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>

<hr>
<h2 id='tf_similar'>True or false of similar</h2><span id='topic+tf_similar'></span>

<h3>Description</h3>

<p>tf_similar is comparing the similarity of different group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tf_similar(real, cluster)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tf_similar_+3A_real">real</code></td>
<td>
<p>is vector with the real group number. If we don't have the real group number, we also can input the result of some clustering method.</p>
</td></tr>
<tr><td><code id="tf_similar_+3A_cluster">cluster</code></td>
<td>
<p>is vector with the result of clustering.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>tf_similar is comparing the similarity of different group. So the main purpose is checking the quality of clustering method. We show the rand index(RI), weight rand index(WRI), and adjust rand index(ARI) to compare. In order to compute the three kinds of index, we count the TT, TF, FT, FF, and the relation weight. The details of the function of the function  are in paper of Tumor subclones detection with Dirichlet Process Mixture Model.
</p>


<h3>Value</h3>

<p>result is data.frame with 10 columns which contains the TT, TF, FT, FF, rand index(RI), weight rand index(WRI), adjust rand index(ARI), the weighted TT, the weighted TF, and the weight FT.
</p>
<p>weight is data.frame with 3 columns which contains the weight about the TT, TF, and FT.
</p>


<h3>Author(s)</h3>

<p>Peter Wu (peter123wu0@gmail.com)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
