<!DOCTYPE html><html lang="en"><head><title>Help for package CSeQTL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CSeQTL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CSeQTL_dataGen'><p>CSeQTL_dataGen</p></a></li>
<li><a href='#CSeQTL_full_analysis'><p>CSeQTL_full_analysis</p></a></li>
<li><a href='#CSeQTL_GS'><p>CSeQTL_GS</p></a></li>
<li><a href='#CSeQTL_linearTest'><p>CSeQTL_linearTest</p></a></li>
<li><a href='#CSeQTL_oneExtremeSim'><p>CSeQTL_oneExtremeSim</p></a></li>
<li><a href='#CSeQTL_run_MatrixEQTL'><p>CSeQTL_run_MatrixEQTL</p></a></li>
<li><a href='#CSeQTL_smart'><p>CSeQTL_smart</p></a></li>
<li><a href='#gen_true_RHO'><p>gen_true_RHO</p></a></li>
<li><a href='#OLS_sim'><p>OLS_sim</p></a></li>
<li><a href='#plot_RHO'><p>plot_RHO</p></a></li>
<li><a href='#prep_gene_info'><p>prep_gene_info</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Cell Type-Specific Expression Quantitative Trail Loci Mapping</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-03-04</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform bulk and cell type-specific expression quantitative trail loci mapping with our novel method (Little et al. (2023) &lt;<a href="https://doi.org/10.1038%2Fs41467-023-38795-w">doi:10.1038/s41467-023-38795-w</a>&gt;).</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, smarter, ggplot2, multcomp, emdbook, MatrixEQTL,
data.table, HelpersMG, R.utils, GenomicFeatures, methods</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-04 22:11:04 UTC; Admin</td>
</tr>
<tr>
<td>Author:</td>
<td>Paul Little [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Paul Little &lt;pllittle321@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-06 17:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='CSeQTL_dataGen'>CSeQTL_dataGen</h2><span id='topic+CSeQTL_dataGen'></span>

<h3>Description</h3>

<p>Simulates a gene/SNP pair with baseline covariates <code>XX</code>,
cell type compositions <code>true_RHO</code>, phased SNP genotypes <code>true_SNP</code>,
and total (TReC) and allele-specific read counts (ASReC) contained in <code>dat</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_dataGen(
  NN,
  MAF,
  true_BETA0 = log(1000),
  true_KAPPA,
  true_ETA,
  true_PHI = 0.1,
  true_PSI = 0.05,
  prob_phased = 0.05,
  true_ALPHA = NULL,
  batch = 1,
  RHO = NULL,
  cnfSNP = FALSE,
  show = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_dataGen_+3A_nn">NN</code></td>
<td>
<p>Positive integer for sample size.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_maf">MAF</code></td>
<td>
<p>Positive numeric value between 0 and 1 for the minor 
allele frequency to simulate phased SNP genotypes assuming Hardy-Weinberg.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_beta0">true_BETA0</code></td>
<td>
<p>A positive numeric value denoting the reference cell type
and reference base's expression multiplied by two and log transformed.
For example, if the TReC for reference base and cell type is 500, then
<code>true_BETA0 = log{2 * 500}</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_kappa">true_KAPPA</code></td>
<td>
<p>A numeric vector denoting the baseline fold change in TReC
between a cell type and reference. By definition, the first element is 1.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_eta">true_ETA</code></td>
<td>
<p>A numeric vector where each element denotes the fold change 
in TReC between the non-reference and reference base in a cell type.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_phi">true_PHI</code></td>
<td>
<p>A non-negative numeric value denoting the over-dispersion term
associated with TReC. If <code>true_PHI &gt; 0</code>, TReC is simulated with the 
negative binomial. If <code>true_PHI = 0</code>, TReC is simulated with the poisson.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_psi">true_PSI</code></td>
<td>
<p>A non-negative numeric value denoting the over-dispersion term
associated with ASReC. If <code>true_PSI &gt; 0</code>, ASReC is simulated with the
beta-binomial, otherwise it is simulated with the binomial distribution.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_prob_phased">prob_phased</code></td>
<td>
<p>A positive numeric value denoting the simulated proportion of 
simulated TReC that are ASReC.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_true_alpha">true_ALPHA</code></td>
<td>
<p>By default, it is set to <code>NULL</code> setting each cell 
type with an eQTL to be cis-eQTL. Otherwise, a positive numeric vector
of fold changes between TReC eQTL effect sizes and ASReC eQTL effect sizes.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_batch">batch</code></td>
<td>
<p>A numeric value set to 1 by default to allow underlying batch effects. 
Set to zero to eliminate batch effects.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions where each row sums to one.
If set to <code>NULL</code>, a matrix of cell type proportions will be simulated.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_cnfsnp">cnfSNP</code></td>
<td>
<p>A boolean value where <code>TRUE</code> re-arranges simulated SNPs to
correlate with baseline bulk expression. When fitting the marginal model 
(not accounting for cell type proportions) and in the presence of cell 
type-specific differentiated expression, a marginal eQTL may be incorrectly inferred.</p>
</td></tr>
<tr><td><code id="CSeQTL_dataGen_+3A_show">show</code></td>
<td>
<p>A boolean value to display verbose output and plot intermediate 
simulated results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list containing true parameters governing the simulated dataset,
simulated covariate matrix <code>XX</code>, observed outcomes in <code>dat</code>.
</p>

<hr>
<h2 id='CSeQTL_full_analysis'>CSeQTL_full_analysis</h2><span id='topic+CSeQTL_full_analysis'></span>

<h3>Description</h3>

<p>Performs marginal and cell type-specific eQTL analysis
with both CSeQTL and OLS models for simulation and comparative purposes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_full_analysis(
  TREC,
  hap2,
  ASREC,
  PHASE,
  SNP,
  RHO,
  XX,
  log_lib_size,
  vec_MARG = c(TRUE, FALSE),
  vec_TRIM = c(TRUE, FALSE),
  vec_PERM = c(TRUE, FALSE),
  thres_TRIM = 10,
  ncores = 1,
  show = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_full_analysis_+3A_trec">TREC</code></td>
<td>
<p>An integer vector of total read counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_hap2">hap2</code></td>
<td>
<p>An integer vector of second haplotype counts</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_asrec">ASREC</code></td>
<td>
<p>An integer vector of total haplotype counts</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_phase">PHASE</code></td>
<td>
<p>An integer vector of 0s and 1s denoting if a subject 
has available haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_snp">SNP</code></td>
<td>
<p>An integer vector of phased genotypes coded 0 (AA),
1 (AB), 2 (BA), 3 (BB), and 5 (NA).</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions. Rows 
correspond to subjects and columns correspond to cell types.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_xx">XX</code></td>
<td>
<p>A numeric design matrix of baseline covariates 
including the intercept in the first column and centered 
continuous covariates.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_log_lib_size">log_lib_size</code></td>
<td>
<p>A positive numeric vector of log transformed
library sizes per subject.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_vec_marg">vec_MARG</code></td>
<td>
<p>A boolean vector for marginal and/or cell type-specific
analyses to be run. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_vec_trim">vec_TRIM</code></td>
<td>
<p>A boolean vector for whether or not analyses with 
trimmed outcomes are included. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_vec_perm">vec_PERM</code></td>
<td>
<p>A boolean vector for whether or not permuted SNP analyses
are included. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_ncores">ncores</code></td>
<td>
<p>A positive integer specifying the number of threads available
to decrease computational runtime.</p>
</td></tr>
<tr><td><code id="CSeQTL_full_analysis_+3A_show">show</code></td>
<td>
<p>A boolean value to display verbose output and plot intermediate 
simulated results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list containing multiple objects. <code>res</code> is a R dataframe
containing the model fitted, marginal model indicator, TRIM indicator, 
permutation indicator, cell types, utilized allele-specific reads indicator,
A and B allele-specific expression, eta/eqtl fold change estimate, p-value.
<code>out</code> contains lists of detailed estimates per model fit.
</p>

<hr>
<h2 id='CSeQTL_GS'>CSeQTL_GS</h2><span id='topic+CSeQTL_GS'></span>

<h3>Description</h3>

<p>Main function that performs eQTL mapping on one 
gene with its associated SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_GS(
  XX,
  TREC,
  SNP,
  hap2,
  ASREC,
  PHASE,
  RHO,
  trim = TRUE,
  thres_TRIM = 20,
  numAS = 5,
  numASn = 5,
  numAS_het = 5,
  cistrans = 0.01,
  ncores = 1,
  show = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_GS_+3A_xx">XX</code></td>
<td>
<p>A numeric design matrix of baseline covariates 
including the intercept in the first column and centered 
continuous covariates. One of the non-intercept columns
should correspond to centered log-transformed library size.
The <code>rownames(XX)</code> needs to be specified.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_trec">TREC</code></td>
<td>
<p>An integer vector containing one gene's 
total read counts.
The <code>names(TREC)</code> needs to be specified.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_snp">SNP</code></td>
<td>
<p>A nonnegative integer matrix with genotypes. Values
should be coded 0, 1, 2, 3, 5 for genotypes AA, AB, BA, BB, NA, respectively.
Rows correspond to SNPs and columns correspond to subjects.
The <code>rownames(SNP)</code> needs to be specified.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_hap2">hap2</code></td>
<td>
<p>An integer vector of the second haplotype's
counts for the gene. The <code>names(hap2)</code> needs to be specified
and match <code>names(TREC)</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_asrec">ASREC</code></td>
<td>
<p>An integer vector of the total haplotype
counts (ASReC) for the gene. The <code>names(ASREC)</code> needs to be specified
and match <code>names(TREC)</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_phase">PHASE</code></td>
<td>
<p>A binary 0/1 vector indicating if haplotype counts
for the gene will be used. The <code>names(PHASE)</code> needs to be specified
and match <code>names(TREC)</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions. Rows 
correspond to subjects and columns correspond to cell types. 
The <code>rownames(RHO)</code> needs to be specified
and match <code>names(TREC)</code>. The <code>colnames(RHO)</code> 
also needs to be specified.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_trim">trim</code></td>
<td>
<p>Boolean value set to <code>FALSE</code> by default to prevent
outcome trimming. If <code>TRUE</code>, the CSeQTL model will be fitted
without SNP genotype to calculate each subject's Cooks' distance
for the gene.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_numas">numAS</code></td>
<td>
<p>A positive integer to determine if a subject has
enough total haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_numasn">numASn</code></td>
<td>
<p>A positive integer to determine how many subjects
have at least <code>numAS</code> to use the haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_numas_het">numAS_het</code></td>
<td>
<p>A positive integer to determine how many subjects
with at least <code>numAS</code> are heterozygous (AB or BA). If
<code>sum(PHASE == 1 &amp; ASREC &gt;= numAS &amp; (SNP == 1 | SNP == 2)) &gt;= numAS_het</code>,
those subjects haplotype counts will be used for TReCASE and cis/trans 
testing and estimation.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_cistrans">cistrans</code></td>
<td>
<p>A numeric value specifying the cis/trans test p-value cutoff
to determine if the eQTLs from TReC-only or TReCASE model is reported.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_ncores">ncores</code></td>
<td>
<p>A positive integer specifying the number of threads available
to decrease computational runtime when performing trimming and looping through SNPs.</p>
</td></tr>
<tr><td><code id="CSeQTL_GS_+3A_show">show</code></td>
<td>
<p>A boolean value to display verbose output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list of statistics and metrics after optimizing the model across multiple SNPs.
The list contains parameter MLEs, gene expression estimates, fold change estimates, 
convergence indicators, likelihood ratio test statistics, and associated p-values.
</p>

<hr>
<h2 id='CSeQTL_linearTest'>CSeQTL_linearTest</h2><span id='topic+CSeQTL_linearTest'></span>

<h3>Description</h3>

<p>Runs marginal and cell type-specific 
analysis using ordinary least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_linearTest(
  input,
  XX,
  RHO,
  SNP,
  YY = NULL,
  MARG = FALSE,
  impute_geno = FALSE,
  trim = FALSE,
  thres_TRIM = 10,
  show_plot = TRUE,
  main_plot = "",
  CTs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_linearTest_+3A_input">input</code></td>
<td>
<p>A data.frame containing columns <code>total</code>
(total read counts) and <code>log_lib_size</code> (log transformed
library size).</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_xx">XX</code></td>
<td>
<p>A numeric design matrix of baseline covariates 
including the intercept in the first column and centered 
continuous covariates.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions. Rows 
correspond to subjects and columns correspond to cell types.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_snp">SNP</code></td>
<td>
<p>An integer vector of phased genotypes coded 0 (AA),
1 (AB), 2 (BA), 3 (BB), and 5 (NA).</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_yy">YY</code></td>
<td>
<p>Default is <code>NULL</code>. By default, the outcome
for OLS is the inverse rank quantile normalized TReC 
after library size correction. Otherwise, the user can input
their own transformed outcome as a numeric vector.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_marg">MARG</code></td>
<td>
<p>Boolean value. Set to <code>TRUE</code> to fit the
marginal OLS model. Default is set to <code>FALSE</code> to fit
the cell type-specific interaction OLS model.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_impute_geno">impute_geno</code></td>
<td>
<p>Boolean value. Default is set to <code>FALSE</code>
to only analyze subjects with non-missing genotype.
If <code>TRUE</code>, missing genotypes are imputed with the 
mean of non-missing genotypes mimicing <code>MatrixEQTL</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_trim">trim</code></td>
<td>
<p>Boolean value set to <code>FALSE</code> by default to prevent
outcome trimming. If <code>TRUE</code>, the OLS model will be fitted
without covariates containing SNP genotype to calculate each 
subject's Cooks' distance.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_show_plot">show_plot</code></td>
<td>
<p>Boolean value set to <code>TRUE</code> to visualize
boxplot or interactions.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_main_plot">main_plot</code></td>
<td>
<p>Character string for the visual's main title.</p>
</td></tr>
<tr><td><code id="CSeQTL_linearTest_+3A_cts">CTs</code></td>
<td>
<p>Set to <code>NULL</code> by default. If <code>NULL</code> and 
<code>show_plot = TRUE</code>, all cell type interaction plots are shown.
Otherwise a subset of cell types can be displayed using an integer vector
of columns or character vector of column names of <code>RHO</code> can 
be provided.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list containing <code>lm()</code> output for <code>lm_out</code>, 
a R dataframe for <code>out_df</code> containing regression estimates, 
standard errors, p-values. <code>res_trim</code> provides a summary of trimmed 
results over a grid of cut-off values and number of samples trimmed.
<code>cooksd</code> is a numeric vector of median shifted and MAD scaled 
Cook's distances per sample. <code>prop_trim</code>, a numeric value for 
number of samples with outcome values trimmed for the user-specified
<code>thres_TRIM</code> value.
</p>

<hr>
<h2 id='CSeQTL_oneExtremeSim'>CSeQTL_oneExtremeSim</h2><span id='topic+CSeQTL_oneExtremeSim'></span>

<h3>Description</h3>

<p>Performs a simulation with multiple replicates
for a pre-specified set of arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_oneExtremeSim(
  NN,
  MAF,
  true_BETA0,
  true_KAPPA,
  true_ETA,
  true_PHI = 0.1,
  wRHO,
  noiseRHO = 0,
  RR = 50,
  vec_MARG = c(TRUE, FALSE),
  vec_TRIM = c(TRUE, FALSE),
  vec_PERM = c(TRUE, FALSE),
  thres_TRIM = 10,
  ncores = 1,
  sim_fn
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_nn">NN</code></td>
<td>
<p>Positive integer for sample size.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_maf">MAF</code></td>
<td>
<p>Positive numeric value between 0 and 1 for the minor 
allele frequency to simulate phased SNP genotypes assuming Hardy-Weinberg.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_true_beta0">true_BETA0</code></td>
<td>
<p>A positive numeric value denoting the reference cell type
and reference base's expression multiplied by two and log transformed.
For example, if the TReC for reference base and cell type is 500, then
<code>true_BETA0 = log{2 * 500}</code>.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_true_kappa">true_KAPPA</code></td>
<td>
<p>A numeric vector denoting the baseline fold change in TReC
between a cell type and reference. By definition, the first element is 1.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_true_eta">true_ETA</code></td>
<td>
<p>A numeric vector where each element denotes the fold change 
in TReC between the non-reference and reference base in a cell type.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_true_phi">true_PHI</code></td>
<td>
<p>A non-negative numeric value denoting the over-dispersion term
associated with TReC. If <code>true_PHI &gt; 0</code>, TReC is simulated with the 
negative binomial. If <code>true_PHI = 0</code>, TReC is simulated with the poisson.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_wrho">wRHO</code></td>
<td>
<p>Takes integer values 1, 2, or 3 to simulate three 
scenarios of cell type proportions.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_noiserho">noiseRHO</code></td>
<td>
<p>A positive numeric value to purposely distort simulated 
cell type compositions.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_rr">RR</code></td>
<td>
<p>A positive integer for number of replicates to generate and analyze.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_vec_marg">vec_MARG</code></td>
<td>
<p>A boolean vector for marginal and/or cell type-specific
analyses to be run. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_vec_trim">vec_TRIM</code></td>
<td>
<p>A boolean vector for whether or not analyses with 
trimmed outcomes are included. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_vec_perm">vec_PERM</code></td>
<td>
<p>A boolean vector for whether or not permuted SNP analyses
are included. By default, both sets of analyses are run.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_ncores">ncores</code></td>
<td>
<p>A positive integer specifying the number of threads available
to decrease computational runtime.</p>
</td></tr>
<tr><td><code id="CSeQTL_oneExtremeSim_+3A_sim_fn">sim_fn</code></td>
<td>
<p>Character value specifying the full path and filename
to store intermediate simulation replicates should errors arise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list containing the results of a small-scale simulation.
<code>res</code> contains replicate-specific results while <code>ures</code> contains
overall simulation results include power, frequency of replicates with 
constrained eta estimates and gene expressions inferred to be zero after optimization.
</p>

<hr>
<h2 id='CSeQTL_run_MatrixEQTL'>CSeQTL_run_MatrixEQTL</h2><span id='topic+CSeQTL_run_MatrixEQTL'></span>

<h3>Description</h3>

<p>CSeQTL_run_MatrixEQTL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_run_MatrixEQTL(TREC, RD, XX, SNP, out_cis_fn, cisDist = 1e+06)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_trec">TREC</code></td>
<td>
<p>A matrix of integer total read counts, rows are genes
with row labels with syntax &quot;gene_name:chrom:start:end&quot;, 
columns are samples with column labels.</p>
</td></tr>
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_rd">RD</code></td>
<td>
<p>A positive numeric vector of library sizes per sample.</p>
</td></tr>
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_xx">XX</code></td>
<td>
<p>A numeric design matrix of baseline covariates 
including the intercept in the first column and centered 
continuous covariates.</p>
</td></tr>
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_snp">SNP</code></td>
<td>
<p>An integer matrix, rows are genomic loci with row labels
such as &quot;chrom:pos&quot;, columns correspond to samples with column labels.</p>
</td></tr>
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_out_cis_fn">out_cis_fn</code></td>
<td>
<p>A full path filename string to store
MatrixEQTL output</p>
</td></tr>
<tr><td><code id="CSeQTL_run_MatrixEQTL_+3A_cisdist">cisDist</code></td>
<td>
<p>A positive integer for number of SNPs to
include relative to the gene body</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MatrixEQTL output file/R dataframe generated containing genes, 
SNPs, associated p-values, effect sizes, etc.
</p>

<hr>
<h2 id='CSeQTL_smart'>CSeQTL_smart</h2><span id='topic+CSeQTL_smart'></span>

<h3>Description</h3>

<p>A function to understand the novelties within CSeQTL. 
The user can experiment with trimming TReC, assess parameter estimation, 
perform hypothesis testing, actively constrain cell type-specific parameters,
when analyzing a single gene/SNP pair.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSeQTL_smart(
  TREC,
  hap2,
  ASREC,
  PHASE,
  SNP,
  RHO,
  XX,
  upPHI,
  upKAPPA,
  upETA,
  upPSI,
  upALPHA,
  iFullModel = FALSE,
  trim = FALSE,
  thres_TRIM = 10,
  hypotest = TRUE,
  swap = TRUE,
  numAS = 5,
  numASn = 5,
  numAS_het = 5,
  cistrans_thres = 0.01,
  gr_eps = 0.01,
  conv_eps = 0.001,
  ncores = 1,
  show = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CSeQTL_smart_+3A_trec">TREC</code></td>
<td>
<p>An integer vector of total read counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_hap2">hap2</code></td>
<td>
<p>An integer vector of second haplotype counts</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_asrec">ASREC</code></td>
<td>
<p>An integer vector of total haplotype counts</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_phase">PHASE</code></td>
<td>
<p>An integer vector of 0s and 1s denoting if a subject 
has available haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_snp">SNP</code></td>
<td>
<p>An integer vector of phased genotypes coded 0 (AA),
1 (AB), 2 (BA), 3 (BB), and 5 (NA).</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions. Rows 
correspond to subjects and columns correspond to cell types.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_xx">XX</code></td>
<td>
<p>A numeric design matrix of baseline covariates 
including the intercept in the first column and centered 
continuous covariates.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_upphi">upPHI</code></td>
<td>
<p>A value of 0 or 1 indicating if a Poisson or
Negative binomial distribution is fitted, respectively.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_upkappa">upKAPPA</code></td>
<td>
<p>An integer vector of zeroes and ones where 
<code>length(upKAPPA) == ncol(RHO)</code>. A requirement is <code>upKAPPA[1] = 1</code>.
Cell types with indices equal to one have the baseline fold change
between the q-th and reference cell type are estimated. Otherwise
that cell type's parameter is constrained to zero.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_upeta">upETA</code></td>
<td>
<p>An integer vector of zeroes and ones where
<code>length(upETA) == ncol(RHO)</code> indicating which
cell types eQTL parameters are estimated or constrained to their null.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_uppsi">upPSI</code></td>
<td>
<p>A value of 0 or 1 indicating if a Binomial or
Beta-binomial distribution is fitted, respectively.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_upalpha">upALPHA</code></td>
<td>
<p>An integer vector of zeroes and ones where
<code>length(upALPHA) == ncol(RHO)</code> indicating which
cell types' cis-trans parameters are estimated or 
constrained to their null.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_ifullmodel">iFullModel</code></td>
<td>
<p>A boolean that if set to <code>TRUE</code> will 
determine the submodel of the full model, based by upKAPPA, 
upETA, and upALPHA parameters, that can be estimated with stability.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_trim">trim</code></td>
<td>
<p>Boolean value. If <code>TRUE</code>, the TReC model will be fitted
without SNP genotype to calculate each subject's Cooks' distance.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_hypotest">hypotest</code></td>
<td>
<p>A boolean to perform eQTL significance 
testing and cis-trans eQTL testing.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_swap">swap</code></td>
<td>
<p>A boolean to determine if the reference cell type should
be swapped with the cell type with highest TReC across alleles.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_numas">numAS</code></td>
<td>
<p>A positive integer to determine if a subject has
enough total haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_numasn">numASn</code></td>
<td>
<p>A positive integer to determine how many subjects
have at least <code>numAS</code> to use the haplotype counts.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_numas_het">numAS_het</code></td>
<td>
<p>A positive integer to determine how many subjects
with at least <code>numAS</code> are heterozygous (AB or BA). If
<code>sum(PHASE == 1 &amp; ASREC &gt;= numAS &amp; (SNP == 1 | SNP == 2)) &gt;= numAS_het</code>,
those subjects haplotype counts will be used for TReCASE and cis/trans 
testing and estimation.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_cistrans_thres">cistrans_thres</code></td>
<td>
<p>A numeric value to determine the 
cis/trans test p-value cutoff.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_gr_eps">gr_eps</code></td>
<td>
<p>A numeric value to determine if convergence
is achieved based on the L2 norm of the gradient.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_conv_eps">conv_eps</code></td>
<td>
<p>A numeric value to determine if convergence
is achieved based on the L2 norm of the product of the 
inverse hessian and gradient.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_ncores">ncores</code></td>
<td>
<p>A positive integer specifying the number of threads available
to decrease computational runtime.</p>
</td></tr>
<tr><td><code id="CSeQTL_smart_+3A_show">show</code></td>
<td>
<p>A boolean value to display verbose output and plot intermediate 
simulated results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A R list of statistics and metrics after optimizing the model for a single SNP.
The list contains parameter MLEs, gradient vectors, covariance matrices,
gene expression estimates, fold change estimates, convergence indicators, 
likelihood ratio test statistics, and associated p-values.
</p>

<hr>
<h2 id='gen_true_RHO'>gen_true_RHO</h2><span id='topic+gen_true_RHO'></span>

<h3>Description</h3>

<p>Simulates cell type proportions under three scenarios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_true_RHO(wRHO = 1, NN, QQ, RHO = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gen_true_RHO_+3A_wrho">wRHO</code></td>
<td>
<p>An integer taking values 1, 2, or 3 for one of the three
scenarios. For <code>wRHO</code> equal to 2 or 3, <code>QQ</code> needs to be 
set to 3 cell types.</p>
</td></tr>
<tr><td><code id="gen_true_RHO_+3A_nn">NN</code></td>
<td>
<p>Positive integer for sample size.</p>
</td></tr>
<tr><td><code id="gen_true_RHO_+3A_qq">QQ</code></td>
<td>
<p>Positive integer for number of cell types.</p>
</td></tr>
<tr><td><code id="gen_true_RHO_+3A_rho">RHO</code></td>
<td>
<p>Default to <code>NULL</code> leads to simulating cell 
type proportions. If a matrix of proportions is supplied, 
this function will append row/column names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric matrix with <code>QQ</code> columns and <code>NN</code> rows
of cell type proportions per sample and cell type.
</p>

<hr>
<h2 id='OLS_sim'>OLS_sim</h2><span id='topic+OLS_sim'></span>

<h3>Description</h3>

<p>OLS_sim
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OLS_sim(
  NN,
  MAF,
  true_BETA0,
  true_KAPPA,
  true_ETA,
  wRHO,
  RR,
  trim = FALSE,
  thres_TRIM = 10,
  showRHO = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="OLS_sim_+3A_nn">NN</code></td>
<td>
<p>Positive integer for sample size.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_maf">MAF</code></td>
<td>
<p>Positive numeric value between 0 and 1 for the minor 
allele frequency to simulate phased SNP genotypes assuming Hardy-Weinberg.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_true_beta0">true_BETA0</code></td>
<td>
<p>A positive numeric value denoting the reference cell type
and reference base's expression multiplied by two and log transformed.
For example, if the TReC for reference base and cell type is 500, then
<code>true_BETA0 = log{2 * 500}</code>.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_true_kappa">true_KAPPA</code></td>
<td>
<p>A numeric vector denoting the baseline fold change in TReC
between a cell type and reference. By definition, the first element is 1.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_true_eta">true_ETA</code></td>
<td>
<p>A numeric vector where each element denotes the fold change 
in TReC between the non-reference and reference base in a cell type.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_wrho">wRHO</code></td>
<td>
<p>Takes integer values 1, 2, or 3 to simulate three 
scenarios of cell type proportions.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_rr">RR</code></td>
<td>
<p>A positive integer for number of replicates to generate and analyze.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_trim">trim</code></td>
<td>
<p>Boolean value set to <code>FALSE</code> by default to prevent
outcome trimming. If <code>TRUE</code>, the OLS model will be fitted
without covariates containing SNP genotype to calculate each 
subject's Cooks' distance.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_thres_trim">thres_TRIM</code></td>
<td>
<p>A positive numeric value to perform subject outcome trimming.
Subjects with standardized Cooks' Distances greater than the threshold are trimmed.</p>
</td></tr>
<tr><td><code id="OLS_sim_+3A_showrho">showRHO</code></td>
<td>
<p>Boolean for seeing a preview of how 
proportions are distributed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simulation results for OLS-based approach with per replicate
and per cell type metrics including p-values, OLS effect sizes,
and proportion of samples with trimmed outcomes.
</p>

<hr>
<h2 id='plot_RHO'>plot_RHO</h2><span id='topic+plot_RHO'></span>

<h3>Description</h3>

<p>plot_RHO
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RHO(RHO, main_plot = "", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RHO_+3A_rho">RHO</code></td>
<td>
<p>A numeric matrix of cell type proportions. Rows 
correspond to subjects and columns correspond to cell types.</p>
</td></tr>
<tr><td><code id="plot_RHO_+3A_main_plot">main_plot</code></td>
<td>
<p>Character string for the visual's main title.</p>
</td></tr>
<tr><td><code id="plot_RHO_+3A_...">...</code></td>
<td>
<p>Arguments for <code>plot(x,y,bty = "n",...)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Null. A plot of cell type proportions per cell type across <code>NN</code> samples.
</p>

<hr>
<h2 id='prep_gene_info'>prep_gene_info</h2><span id='topic+prep_gene_info'></span>

<h3>Description</h3>

<p>Prepare GTF files for gene and 
exon level information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_gene_info(work_dir, gtf_gz_fn = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prep_gene_info_+3A_work_dir">work_dir</code></td>
<td>
<p>A character string specifying
the working directory to store gene and exon
data.</p>
</td></tr>
<tr><td><code id="prep_gene_info_+3A_gtf_gz_fn">gtf_gz_fn</code></td>
<td>
<p>A character string full path
of the gz compressed gtf file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Null. No value returned.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
