<!DOCTYPE html><html><head><title>Help for package ctgt</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ctgt}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#1-alpha quantile'><p>The 1-alpha quantile of globaltest</p></a></li>
<li><a href='#closed testing with globaltest'><p>Approximated Closed Testing with Global Test</p></a></li>
<li><a href='#ctgt-package'>
<p>'Closed Testing with Globaltest for Pathway Analysis'</p></a></li>
<li><a href='#internalfunctions'><p>'Internal Functions (ctgt)'</p></a></li>
<li><a href='#majorizing vector'><p>The majorizing vector</p></a></li>
<li><a href='#non-standardized globaltest'><p>'Non-standardized globaltest'</p></a></li>
<li><a href='#p-value'><p>p-value of globaltest</p></a></li>
<li><a href='#true discoveries'><p>True discoveries</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Closed Testing with Globaltest for Pathway Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Ningning Xu</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ningning Xu &lt;ningningxu312@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A shortcut procedure is proposed to implement closed testing for large-scale multiple testings, especially with the global test. This shortcut is asymptotically equivalent to closed testing and post hoc. Users could detect any possible sets of features or pathways with family-wise error rate controlled. The global test is powerful to detect associations between a group of features and an outcome of interest. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.3)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, BH</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-15 14:53:18 UTC; nxu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-16 13:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='1-alpha+20quantile'>The 1-alpha quantile of globaltest</h2><span id='topic+criticalvalue'></span>

<h3>Description</h3>

<p>Robbins and Pitman Algorithm to calculate the criticalvalue given eigenvalue vector and alpha level. </p>


<h3>Usage</h3>

<pre><code class='language-R'>  criticalvalue(lam, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="1-alpha+2B20quantile_+3A_lam">lam</code></td>
<td>
<p>The numeric vector with eigenvalues as elements.</p>
</td></tr>
<tr><td><code id="1-alpha+2B20quantile_+3A_alpha">alpha</code></td>
<td>
<p>The type I error rate allowed. The default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a real number.</p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>

<hr>
<h2 id='closed+20testing+20with+20globaltest'>Approximated Closed Testing with Global Test</h2><span id='topic+actgt'></span>

<h3>Description</h3>

<p> To detect the significance of the set of features after correcting for multiple global tests, with family-wise error rate controlled.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  actgt (y, X, xs, hyps, maxit = 0, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_y">y</code></td>
<td>
<p>The response vector (numeric vector).</p>
</td></tr>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_x">X</code></td>
<td>
<p>The full design matrix, whose columns are named by the covariates.</p>
</td></tr>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_xs">xs</code></td>
<td>
<p>The name vector of all covariates (character vector). </p>
</td></tr>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_hyps">hyps</code></td>
<td>
<p>The name vector of the covariates in the pathway of interest (character vector).</p>
</td></tr>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_maxit">maxit</code></td>
<td>
<p>An optional integer to denote the maximal interations for branch and bound method. The default value 0 means the single-step shortcut without branch and bound method. Note that larger value is more time-consuming. </p>
</td></tr>
<tr><td><code id="closed+2B20testing+2B20with+2B20globaltest_+3A_alpha">alpha</code></td>
<td>
<p>The type I error rate allowed. The default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of rejection indicator and the number of iterations. </p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Generate the design matrix and response vector for logistic regression models
  n= 100
  m = 5
  X = matrix(data = 0, nrow = n, ncol = m,byrow = TRUE )
  for ( i in 1:n){
    set.seed(1234+i)
    X[i,] =  as.vector(arima.sim(model = list(order = c(1, 0, 0), ar = 0.2), n = m) )
  }

  y = rbinom(n,1,0.6)
  X[which(y==1),1:3] = X[which(y==1),1:3] + 0.8


  xs = paste("x",seq(1,m,1),sep="") 
  colnames(X) = xs

  hyps=xs[1]

  #The sinle-step ctgt procedure
  actgt(y = y, X = X, xs = xs, hyps = hyps, maxit = 0, alpha = 0.05)
  #Result Iterations 
  #"unsure"        "0" 

  # The iterative ctgt procedure with more iterations
  actgt(y = y, X = X, xs = xs, hyps = hyps, maxit = 0, alpha = 0.05)
  #Result Iterations 
  #"reject"       "2" 
  #which means that x1 is rejected by closed testing within two more iterations of the shortcut


  # For a group of feature sets
  mysets = list(xs[1:5], xs[c(1,4)], xs[c(1,4,5)])
  sapply(mysets, function(i) actgt(y = y, X = X, xs = xs, hyps = i, maxit = 0, alpha = 0.05))
  #Result     "reject" "unsure" "reject"
  #Iterations "0"      "0"      "0"         

  mysets = list(xs[1:5], xs[c(1,4)], xs[c(1,4,5)])
  sapply(mysets, function(i) actgt(y = y, X = X, xs = xs, hyps = i, maxit = 0, alpha = 0.05))
  #Result     "reject" "reject" "reject"
  #Iterations "0"      "2"          "0"         

</code></pre>

<hr>
<h2 id='ctgt-package'>
'Closed Testing with Globaltest for Pathway Analysis'
</h2><span id='topic+ctgt-package'></span><span id='topic+ctgt'></span>

<h3>Description</h3>

<p>A shortcut procedure for closed testing with the global test is presented. 
</p>


<h3>Details</h3>

<p>See examples in <a href="#topic+actgt">actgt</a> function.
</p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
Jelle J. Goeman, Sara A. van de Geer, Floor de Kort, Hans C. van Houwelingen, A global test for groups of genes: testing association with a clinical outcome, Bioinformatics, Volume 20, Issue 1, 1 January 2004, Pages 93-99, https://doi.org/10.1093/bioinformatics/btg382
</p>

<hr>
<h2 id='internalfunctions'>'Internal Functions (ctgt)'</h2><span id='topic+actgt_it'></span><span id='topic+tacmax'></span><span id='topic+tactrue'></span>

<h3>Description</h3>

<p>Internal functions of ctgt.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## iterative shortcut with branch and bound
actgt_it(y,Tmatrix, Cmatrix,fxs, sxs,Tf,Lamf,Cf,Ts,Lams,Cs,count=1,maxIt=1,a = 0.05)
  
## to check whether tmin is above cmax
tacmax(tmins,levels,tw, cf,lf,ls,alp )
  
## to check whether tmin is above ctrue
tactrue(tmins,hyxs,cfull,Wmatrix,alp )


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="internalfunctions_+3A_y">y</code></td>
<td>
<p>The response vector (numeric vector).</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_tmatrix">Tmatrix</code></td>
<td>
<p>The matrix used to calculate the test statistics.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_cmatrix">Cmatrix</code></td>
<td>
<p>The matrix used to calculate the critical values.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_fxs">fxs</code></td>
<td>
<p>The name vector of upper model (character vector). </p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_sxs">sxs</code></td>
<td>
<p>The name vector of lower model (character vector). </p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_tf">Tf</code>, <code id="internalfunctions_+3A_lamf">Lamf</code>, <code id="internalfunctions_+3A_cf">Cf</code></td>
<td>
<p>Test statistic, eigenvalues, and critical value of fxs.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_ts">Ts</code>, <code id="internalfunctions_+3A_lams">Lams</code>, <code id="internalfunctions_+3A_cs">Cs</code></td>
<td>
<p>Test statistic, eigenvalues, and critical value of sxs.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_count">count</code></td>
<td>
<p>count the branches, default is 1.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_maxit">maxIt</code></td>
<td>
<p>maximal number of branches chosen by user, default is 1.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_a">a</code>, <code id="internalfunctions_+3A_alp">alp</code></td>
<td>
<p>alpha level.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_tmins">tmins</code></td>
<td>
<p>Minimum test statistics. </p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_levels">levels</code></td>
<td>
<p>levels</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_tw">tw</code>, <code id="internalfunctions_+3A_cf">cf</code>, <code id="internalfunctions_+3A_lf">lf</code>, <code id="internalfunctions_+3A_ls">ls</code></td>
<td>
<p>sorted weights, critical values and level for fxs and sxs. </p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_hyxs">hyxs</code></td>
<td>
<p>The name vector of the covariates of interest (character vector).</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_cfull">cfull</code></td>
<td>
<p> critical value of full model.</p>
</td></tr>
<tr><td><code id="internalfunctions_+3A_wmatrix">Wmatrix</code></td>
<td>
<p>matrix to calculate majorizing vector.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>

<hr>
<h2 id='majorizing+20vector'>The majorizing vector</h2><span id='topic+getL'></span>

<h3>Description</h3>

<p>To get the majorizing vector at a specific level, given the upbound and lowbound.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  getL (ub, lb, level)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="majorizing+2B20vector_+3A_ub">ub</code></td>
<td>
<p>upper bound.</p>
</td></tr>
<tr><td><code id="majorizing+2B20vector_+3A_lb">lb</code></td>
<td>
<p>lower bound.</p>
</td></tr>
<tr><td><code id="majorizing+2B20vector_+3A_level">level</code></td>
<td>
<p>level of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector with the same length as ub and lb. </p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>

<hr>
<h2 id='non-standardized+20globaltest'>'Non-standardized globaltest'</h2><span id='topic+gt2'></span>

<h3>Description</h3>

<p>This is the sencond version of the globaltest, the non-standardized globaltest</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## a powerful variant of globaltest
  gt2 (y, X, hyps, alpha = 0.05)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="non-standardized+2B20globaltest_+3A_y">y</code></td>
<td>
<p>The response vector (numeric vector).</p>
</td></tr>
<tr><td><code id="non-standardized+2B20globaltest_+3A_x">X</code></td>
<td>
<p>The full design matrix, whose columns are named by the covariates.</p>
</td></tr>
<tr><td><code id="non-standardized+2B20globaltest_+3A_hyps">hyps</code></td>
<td>
<p>The name vector of the covariates in the pathway of interest (character vector).</p>
</td></tr>
<tr><td><code id="non-standardized+2B20globaltest_+3A_alpha">alpha</code></td>
<td>
<p>The type I error rate allowed. The default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the p-value, the observed and expected test statistics and the number of covariates. </p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Generate the design matrix and response vector for logistic regression models
  n= 100
  m = 5
  X = matrix(data = 0, nrow = n, ncol = m,byrow = TRUE )
  for ( i in 1:n){
    set.seed(1234+i)
    X[i,] =  as.vector(arima.sim(model = list(order = c(1, 0, 0), ar = 0.2), n = m) )
  }

  y = rbinom(n,1,0.6)
  X[which(y==1),1:3] = X[which(y==1),1:3] + 0.8


  xs = paste("x",seq(1,m,1),sep="") 
  colnames(X) = xs

  hyps=xs[1]

  #The raw p-values of globaltest
  gt2(y = y, X = X, hyps = hyps, alpha = 0.05)
  #p-value Statistic  Expected      #Cov 
  #7.64e-03  2.30e+02  1.24e+02  1.00e+00 
</code></pre>

<hr>
<h2 id='p-value'>p-value of globaltest</h2><span id='topic+pv'></span>

<h3>Description</h3>

<p>Robbins and Pitman Algorithm to calculate the p-value given the observed value and the eigenvalue vector.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  pv(x, lam)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p-value_+3A_x">x</code></td>
<td>
<p>The observed value that is used to calculate the coresponding the p-value.</p>
</td></tr>
<tr><td><code id="p-value_+3A_lam">lam</code></td>
<td>
<p>The numeric vector with eigenvalues as elements.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a value between 0 and 1. </p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>

<hr>
<h2 id='true+20discoveries'>True discoveries</h2><span id='topic+discoveries'></span>

<h3>Description</h3>

<p>To count the number of true discoveries within a given pathway or feature set of interest.</p>


<h3>Usage</h3>

<pre><code class='language-R'>  discoveries (y, X, xs, hyps, maxit = 0, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true+2B20discoveries_+3A_y">y</code></td>
<td>
<p>The response vector (numeric vector).</p>
</td></tr>
<tr><td><code id="true+2B20discoveries_+3A_x">X</code></td>
<td>
<p>The full design matrix, whose columns are named by the covariates.</p>
</td></tr>
<tr><td><code id="true+2B20discoveries_+3A_xs">xs</code></td>
<td>
<p>The name vector of all covariates (character vector). </p>
</td></tr>
<tr><td><code id="true+2B20discoveries_+3A_hyps">hyps</code></td>
<td>
<p>The name vector of the covariates in the pathway of interest (character vector).</p>
</td></tr>
<tr><td><code id="true+2B20discoveries_+3A_maxit">maxit</code></td>
<td>
<p>An optional integer to denote the maximal interations for branch and bound method. The default value 0 means the single-step shortcut without branch and bound method. Note that larger value is more time-consuming. </p>
</td></tr>
<tr><td><code id="true+2B20discoveries_+3A_alpha">alpha</code></td>
<td>
<p>The type I error rate allowed. The default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a non-negative interger. </p>


<h3>Author(s)</h3>

<p>Ningning Xu
</p>
<p>Maintainer: Ningning Xu &lt;n.xu@lumc.nl; xu15263142750@gmail.com&gt;
</p>


<h3>References</h3>

<p>Ningning Xu, Aldo solari, Jelle Goeman, Clsoed testing with global test, with applications on metabolomics data, arXiv:2001.01541, https://arxiv.org/abs/2001.01541
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #Generate the design matrix and response vector for logistic regression models
  n= 100
  m = 5
  X = matrix(data = 0, nrow = n, ncol = m,byrow = TRUE )
  for ( i in 1:n){
    set.seed(1234+i)
    X[i,] =  as.vector(arima.sim(model = list(order = c(1, 0, 0), ar = 0.2), n = m) )
  }

  y = rbinom(n,1,0.6)
  X[which(y==1),1:3] = X[which(y==1),1:3] + 0.8


  xs = paste("x",seq(1,m,1),sep="") 
  colnames(X) = xs

  # For standarized data
  X = scale(x = X,center = FALSE,scale = TRUE)/sqrt(n-1)
  interest = xs

  discoveries(y=y, X = X, xs = xs, hyps = interest)
  #2
  discoveries(y=y, X = X, xs = xs, hyps = interest, maxit=10)
  #2
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
