<!DOCTYPE html><html><head><title>Help for package ctl</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ctl}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ath.churchill'>
<p>Example metabolite expression data from Arabidopsis Thaliana on 9 metabolites.</p></a></li>
<li><a href='#ath.metabolites'>
<p>Example metabolite expression data from Arabidopsis Thaliana on 24 metabolites.</p></a></li>
<li><a href='#ath.result'>
<p>Output of QCLscan after 5000 permutations on the metabolite expression data from Arabidopsis Thaliana.</p></a></li>
<li><a href='#basic.qc'><p>Create quality control plots.</p></a></li>
<li><a href='#ctl-internal'><p>Internal CTL functions</p></a></li>
<li><a href='#ctl-package'>
<p>CTL - CTL mapping in experimental crosses</p></a></li>
<li><a href='#ctl.circle'><p>Circleplot CTL on multiple traits</p></a></li>
<li><a href='#ctl.lineplot'><p>Lineplot CTL on multiple traits</p></a></li>
<li><a href='#ctl.load'><p>ctl.load - Load CTLs calculated by the D2.0 version</p></a></li>
<li><a href='#CTLhelper'><p>CTLhelper - Helper functions for CTL mapping</p></a></li>
<li><a href='#CTLmapping'><p>CTLmapping - Scan for correlated trait loci (CTL)</p></a></li>
<li><a href='#CTLnetwork'><p>CTLnetwork - Interaction network from a genome-wide CTLscan of multiple traits</p></a></li>
<li><a href='#CTLprofiles'><p>CTLprofiles - Extract CTL interaction profiles</p></a></li>
<li><a href='#CTLregions'><p>CTLregions - Get all significant interactions from a genome-wide CTLscan</p></a></li>
<li><a href='#CTLscan'><p>CTLscan - Scan for Correlated Trait Locus (CTL)</p></a></li>
<li><a href='#CTLscan.cross'><p>CTLscan.cross - Scan for Correlated Trait Locus (CTL) (R/qtl cross object)</p></a></li>
<li><a href='#CTLsignificant'><p>CTLsignificant - Get all significant interactions from a genome-wide CTLscan</p></a></li>
<li><a href='#detect.peaks'><p>detect.peaks - Peak detection algorithm to 'flatten' data above a certain threshold</p></a></li>
<li><a href='#hist.CTLobject'><p>Plot histogram of CTL permutations</p></a></li>
<li><a href='#image.CTLobject'><p>Plot genome-wide CTL on multiple traits</p></a></li>
<li><a href='#plot.CTLobject'><p>Plot CTL curves or heatmaps</p></a></li>
<li><a href='#plot.CTLpermute'><p>Differential correlation versus likelihood plotted in curves</p></a></li>
<li><a href='#plot.CTLscan'><p>Plot CTL results as bar, line or GWAS plot.</p></a></li>
<li><a href='#plotTraits'><p>plotTraits - Trait vs Trait scatterplot, colored by the selected genetic locus</p></a></li>
<li><a href='#print.CTLobject'><p>Print the results of a CTL genome scan</p></a></li>
<li><a href='#print.CTLscan'><p>Print the results of a single phenotype CTL scan</p></a></li>
<li><a href='#qtlimage'><p>Plot a QTL heatmap of the phenotypes scanned by CTLscan</p></a></li>
<li><a href='#QTLmapping'><p>QTLmapping - QTL mapping method for CTL analysis</p></a></li>
<li><a href='#scanSD'><p>scanSD - Analyze the differences in Standard Deviation between genotypes between two traits</p></a></li>
<li><a href='#scanSD.cross'><p>scanSD.cross - Analyze the differences in standard deviation between two traits at a certain genetic marker (R/qtl cross object)</p></a></li>
<li><a href='#scanSlopes'><p>scanSlopes - Create a slope difference profile between two traits</p></a></li>
<li><a href='#scanSlopes.cross'><p>scanSlopes.cross - Create a slope difference profile between two traits (R/qtl cross object)</p></a></li>
<li><a href='#yeast.brem'>
<p>Example gene expression data from Saccharomyces cerevisiae on 301 RNA expressions.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.0-10</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Correlated Trait Locus Mapping</td>
</tr>
<tr>
<td>Author:</td>
<td>Danny Arends &lt;Danny.Arends@gmail.com&gt;, Yang Li, Gudrun A Brockmann, 
  Ritsert C Jansen, Robert W Williams, and Pjotr Prins</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Danny Arends &lt;Danny.Arends@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), MASS, parallel, qtl</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification and network inference of genetic loci associated
  with correlation changes in quantitative traits (called correlated trait loci, CTLs).
  Arends et al. (2016) &lt;<a href="https://doi.org/10.21105%2Fjoss.00087">doi:10.21105/joss.00087</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-15 15:27:28 UTC; rqdt9</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-16 12:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ath.churchill'> 
Example metabolite expression data from Arabidopsis Thaliana on 9 metabolites.
</h2><span id='topic+ath.churchill'></span><span id='topic+ath.gary'></span>

<h3>Description</h3>

<p>Arabidopsis recombinant inbred lines by selfing.  There are 403 lines, 9 phenotypes, 
and 69 markers on 5 chromosomes stored as a list with 3 matrices: genotypes, phenotypes, map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(ath.churchill)
</code></pre>


<h3>Format</h3>

<p>Data stored in a list holding 3 matrices genotypes, phenotypes and map
</p>


<h3>Details</h3>

<p>Arabidopsis recombinant inbred lines by selfing.  There are 403 lines, 9 metabolic 
phenotypes, and 69 markers on 5 chromosomes.
</p>


<h3>Source</h3>

<p>Arabidopsis Bay-0 x Sha metabolite data from XX, senior author: Gary Churchill 2012, 
Published in: Plos
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.churchill)           # Arabidopsis thaliana dataset
  
  ath.gary$genotypes[1:5, 1:5]  # ath.gary is the short name
  ath.gary$phenotypes[1:5, 1:5]
  ath.gary$map[1:5, ]
</code></pre>

<hr>
<h2 id='ath.metabolites'> 
Example metabolite expression data from Arabidopsis Thaliana on 24 metabolites.
</h2><span id='topic+ath.metabolites'></span><span id='topic+ath.metab'></span>

<h3>Description</h3>

<p>Arabidopsis recombinant inbred lines by selfing.  There are 162 lines, 24 phenotypes, 
and 117 markers on 5 chromosomes stored as a list with 3 matrices: genotypes, phenotypes, map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(ath.metabolites)
</code></pre>


<h3>Format</h3>

<p>Data stored in a list holding 3 matrices genotypes, phenotypes and map
</p>


<h3>Details</h3>

<p>Arabidopsis recombinant inbred lines by selfing.  There are 162 lines, 24 phenotypes, and 
117 markers on 5 chromosomes.
</p>


<h3>Source</h3>

<p>Part of the Arabidopsis RIL selfing experiment with Landsberg Erecta (Ler) and Cape Verde 
Islands (Cvi) with 162 individuals scored (with errors) at 117 markers. Dataset obtained 
from GBIC - Groningen BioInformatics Centre, University of Groningen.
</p>


<h3>References</h3>


<ul>
<li><p> Keurentjes, J. J. and Fu, J. and de Vos, C. H. and Lommen, A. and Hall, R. D. and 
Bino, R. J. and van der Plas, L. H. and Jansen, R. C. and Vreugdenhil, D. and 
Koornneef, M. (2006), The genetics of plant metabolism. <em>Nature Genetics.</em> 
<b>38</b>-7, 842&ndash;849.
</p>
</li>
<li><p> Alonso-Blanco, C. and Peeters, A. J. and Koornneef, M. and Lister, C. and Dean, C. 
and van den Bosch, N. and Pot, J. and Kuiper, M. T. (1998), Development of an AFLP 
based linkage map of Ler, Col and Cvi Arabidopsis thaliana ecotypes and construction 
of a Ler/Cvi recombinant inbred line population. <em>Plant J.</em> <b>14</b>(2), 259&ndash;271.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites)           # Arabidopsis thaliana dataset

  ath.metab$genotypes[1:5, 1:5]   # ath.metab is the short name
  ath.metab$phenotypes[1:5, 1:5]
  ath.metab$map[1:5, ]
</code></pre>

<hr>
<h2 id='ath.result'> 
Output of QCLscan after 5000 permutations on the metabolite expression data from Arabidopsis Thaliana.
</h2><span id='topic+ath.result'></span>

<h3>Description</h3>

<p>Results from a QCLscan on Arabidopsis recombinant inbred lines by selfing.  There are 162 lines, 24 phenotypes, 
and 117 markers on 5 chromosomes stored as a list with 3 matrices: genotypes, phenotypes, map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(ath.result)
</code></pre>


<h3>Format</h3>

<p>Cross object from R/QTL
</p>


<h3>Details</h3>

<p>Results from a QCLscan on Arabidopsis recombinant inbred lines by selfing.  There are 162 lines, 24 
phenotypes, and 117 markers on 5 chromosomes. the QCLscan also includes 5000 permutations
</p>


<h3>Source</h3>

<p>Part of the Arabidopsis RIL selfing experiment with Landsberg Erecta (Ler) and Cape Verde 
Islands (Cvi) with 162 individuals scored (with errors) at 117 markers. Dataset obtained 
from GBIC - Groningen BioInformatics Centre, University of Groningen.
</p>


<h3>References</h3>


<ul>
<li><p> Keurentjes, J. J. and Fu, J. and de Vos, C. H. and Lommen, A. and Hall, 
R. D. and Bino, R. J. and van der Plas, L. H. and Jansen, R. C. and 
Vreugdenhil, D. and Koornneef, M.
(2006), The genetics of plant metabolism. <em>Nature Genetics.</em>
<b>38</b>-7, 842&ndash;849. 
</p>
</li>
<li><p> Alonso-Blanco, C. and Peeters, A. J. and Koornneef, M. and Lister, C. 
and Dean, C. and van den Bosch, N. and Pot, J. and Kuiper, M. T. (1998),
Development of an AFLP based linkage map of Ler, Col and Cvi
Arabidopsis thaliana ecotypes and construction of a Ler/Cvi recombinant
inbred line population. <em>Plant J.</em> <b>14</b>(2), 259&ndash;271.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(ath.result)           # Arabidopsis thaliana dataset
ath.result[[1]]            # Print the QCLscan summary of the phenotype 1
</code></pre>

<hr>
<h2 id='basic.qc'>Create quality control plots.</h2><span id='topic+basic.qc'></span>

<h3>Description</h3>

<p>Create quality control plots, used in the examples of CTL mapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  basic.qc(genotypes, phenotypes, map_info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basic.qc_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="basic.qc_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="basic.qc_+3A_map_info">map_info</code></td>
<td>
<p>    Matrix of genetic map information </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None. 
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot a CTLscan object
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  #TODO
</code></pre>

<hr>
<h2 id='ctl-internal'>Internal CTL functions</h2><span id='topic+CTLnetwork.addmarker'></span><span id='topic+ctleffects'></span><span id='topic+ctlmarker'></span><span id='topic+correlation'></span><span id='topic+chiSQN'></span><span id='topic+test.correlation'></span><span id='topic+test.chiSQN'></span><span id='topic+getCorrelations'></span><span id='topic+getCorrelations.cross'></span><span id='topic+test.getCorrelation.cross'></span><span id='topic+correlated.phenotype'></span><span id='topic+check.marker'></span><span id='topic+check.pheno'></span><span id='topic+create.marker'></span><span id='topic+CTLpowerAnalysis'></span><span id='topic+dcor.create'></span><span id='topic+getRatio'></span><span id='topic+plot.CTLnetwork'></span><span id='topic+test.power.test'></span><span id='topic+CTLanalyseRatios'></span><span id='topic+CTLanalyseEffects'></span><span id='topic+chiSQtoP'></span><span id='topic+mapinfotomarkerlocs'></span><span id='topic+draw.element'></span><span id='topic+draw.spline'></span><span id='topic+circle.loc'></span><span id='topic+nfrom'></span><span id='topic+nto'></span><span id='topic+openmp'></span><span id='topic+extrapolateBeyondRange'></span><span id='topic+CTLscantoScanone'></span><span id='topic+CTLtoLODvector'></span><span id='topic+deltaSD'></span><span id='topic+stdSlopeEstimate'></span><span id='topic+stdSlopeIntercept'></span><span id='topic+printSignificantSlope'></span><span id='topic+stdSlopeError'></span><span id='topic+stdSlopeBeta'></span><span id='topic+stdSlopeTest'></span><span id='topic+munique'></span><span id='topic+check.genotypes'></span><span id='topic+getRVM'></span><span id='topic+lodscorestoscanone'></span><span id='topic+gcLoop'></span><span id='topic+internal.image'></span><span id='topic+addChromosomeLines'></span><span id='topic+getPermuteThresholds'></span><span id='topic+print.CTLpermute'></span><span id='topic+chr_length'></span><span id='topic+chr_total_length'></span><span id='topic+a_loc'></span><span id='topic+m_loc'></span><span id='topic+plot.CTLscan2'></span><span id='topic+plotCTLscan3'></span><span id='topic+plotExpression'></span><span id='topic+CTLasLOD'></span><span id='topic+remove.diag'></span><span id='topic+dcor'></span><span id='topic+get.chr.edges'></span><span id='topic+up'></span><span id='topic+dw'></span><span id='topic+redblue'></span><span id='topic+whiteblack'></span><span id='topic+top.correlated'></span><span id='topic+ctl.version'></span>

<h3>Description</h3>

<p>Internal CTL functions. These are generally not to be called by the user.
</p>


<h3>Author(s)</h3>

<p>Danny Arends, <a href="mailto:Danny.Arends@Gmail.com">Danny.Arends@Gmail.com</a> </p>

<hr>
<h2 id='ctl-package'>
CTL - CTL mapping in experimental crosses
</h2><span id='topic+ctl'></span><span id='topic+CTL'></span>

<h3>Description</h3>

<p>Analysis of experimental crosses to identify genetic markers associated with correlation 
changes in quantitative traits (CTL). The additional correlation information obtained 
can be combined with QTL information to perform de novo reconstruction of interaction 
networks.
</p>
<p>For more background information about the method we refer to the methodology article 
published in XX (201X).
</p>
<p>The R package is a basic iomplementation and it includes the following core functionality:
</p>

<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL.
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>.
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>.
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a CTLscan.
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait.
</p>
</li>
<li> <p><code><a href="#topic+ctl.circle">ctl.circle</a></code> - Circle plot CTLs on single and multiple traits.
</p>
</li>
<li> <p><code><a href="#topic+ctl.lineplot">ctl.lineplot</a></code> - Line plot CTLs on single and multiple traits.
</p>
</li>
<li> <p><code><a href="#topic+CTLprofiles">CTLprofiles</a></code> - Extract CTL interaction profiles.
</p>
</li></ul>

<p>For all these functions we also provide examples and demonstrations on real genetical 
genomics data. We thank all contributors for publishing their data online and will 
accept submissions of intrestion datasets, currently ctl provides:
</p>

<ul>
<li> <p><code><a href="#topic+ath.metabolites">ath.metabolites</a></code> - Metabolite expression data from Arabidopsis Thaliana
</p>
</li>
<li> <p><code><a href="#topic+ath.churchill">ath.churchill</a></code> - Metabolite expression data from Arabidopsis Thaliana
</p>
</li>
<li> <p><code><a href="#topic+yeast.brem">yeast.brem</a></code> - Gene expression data from Saccharomyces cerevisiae
</p>
</li></ul>



<h3>Details</h3>

<p>More detailed information and/or examples are given per function as needed.
Some additional functionality:
</p>

<ul>
<li> <p><code><a href="#topic+basic.qc">basic.qc</a></code> - Some basic quality checks for phenotype and genotype data
</p>
</li>
<li> <p><code><a href="#topic+CTLscan.cross">CTLscan.cross</a></code> - Use an R/qtl cross object with <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li></ul>



<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Contributions from: Bruno Tesson, Pjotr Prins and Ritsert C. Jansen
</p>


<h3>References</h3>


<ul>
<li><p> TODO
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLscan.cross">CTLscan.cross</a></code> - Use an R/qtl cross object with <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li></ul>


<hr>
<h2 id='ctl.circle'>Circleplot CTL on multiple traits</h2><span id='topic+ctl.circle'></span>

<h3>Description</h3>

<p>Plot the CTL for genome-wide CTL on multiple traits (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctl.circle(CTLobject, mapinfo, phenocol, significance = 0.05, gap = 50, cex = 1,
verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctl.circle_+3A_ctlobject">CTLobject</code></td>
<td>
<p>      An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="ctl.circle_+3A_mapinfo">mapinfo</code></td>
<td>
<p>        The mapinfo matrix with 3 columns: &quot;Chr&quot; - the chromosome number, &quot;cM&quot; - the 
location of the marker in centiMorgans and the 3rd column &quot;Mbp&quot; - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) should 
match those in the CTLobject. </p>
</td></tr>
<tr><td><code id="ctl.circle_+3A_phenocol">phenocol</code></td>
<td>
<p>       Which phenotype results to plot. Defaults to plot all phenotypes. </p>
</td></tr> 
<tr><td><code id="ctl.circle_+3A_significance">significance</code></td>
<td>
<p>   Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="ctl.circle_+3A_gap">gap</code></td>
<td>
<p>            Gap between chromosomes in cM. </p>
</td></tr>
<tr><td><code id="ctl.circle_+3A_cex">cex</code></td>
<td>
<p>            Global magnificantion factor for the image elements. </p>
</td></tr>
<tr><td><code id="ctl.circle_+3A_verbose">verbose</code></td>
<td>
<p>        Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLprofiles">CTLprofiles</a></code> - Extract CTL interaction profiles
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result)       # Arabidopsis Thaliana results
  data(ath.metabolites)  # Arabidopsis Thaliana data set

  ctl.circle(ath.result, ath.metab$map, sign=0.001)
  ctl.circle(ath.result, ath.metab$map, phenocol = 1:6, sign = 0.01)
</code></pre>

<hr>
<h2 id='ctl.lineplot'>Lineplot CTL on multiple traits</h2><span id='topic+ctl.lineplot'></span>

<h3>Description</h3>

<p>Plot the CTL for genome-wide CTL on multiple traits (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctl.lineplot(CTLobject, mapinfo, phenocol, significance = 0.05, gap = 50, 
col = "orange", bg.col = "lightgray", cex = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctl.lineplot_+3A_ctlobject">CTLobject</code></td>
<td>
<p>     An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_mapinfo">mapinfo</code></td>
<td>
<p>       The mapinfo matrix with 3 columns: &quot;Chr&quot; - the chromosome number, &quot;cM&quot; - the 
location of the marker in centiMorgans and the 3rd column &quot;Mbp&quot; - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) 
should match those in the CTLobject (only significant markers will be annotated). </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_phenocol">phenocol</code></td>
<td>
<p>      Which phenotype results to plot. Defaults to plot all phenotypes. </p>
</td></tr> 
<tr><td><code id="ctl.lineplot_+3A_significance">significance</code></td>
<td>
<p>  Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_gap">gap</code></td>
<td>
<p>           The gap between chromosomes in cM. </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_col">col</code></td>
<td>
<p>           Line color used. </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_bg.col">bg.col</code></td>
<td>
<p>        Node background color. </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_cex">cex</code></td>
<td>
<p>           Global magnificantion factor for the image elements. </p>
</td></tr>
<tr><td><code id="ctl.lineplot_+3A_verbose">verbose</code></td>
<td>
<p>       Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLprofiles">CTLprofiles</a></code> - Extract CTL interaction profiles
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  require(ctl)
  data(ath.result)       # Arabidopsis Thaliana results
  data(ath.metabolites)  # Arabidopsis Thaliana data set

  todo &lt;- c(1,3,4,5,6,8,9,10,11,12,14,17,18,19,22,23)
  op   &lt;- par(mfrow = c(4,4))
  op   &lt;- par(oma = c(0.1,0.1,0.1,0.1))
  op   &lt;- par(mai = c(0.1,0.1,0.1,0.1))
  for(x in todo){ # Overview of the 16 traits with CTLs
    ctl.lineplot(ath.result, ath.metab$map, phenocol = x, sign=0.1)
  }
</code></pre>

<hr>
<h2 id='ctl.load'>ctl.load - Load CTLs calculated by the D2.0 version </h2><span id='topic+ctl.load'></span>

<h3>Description</h3>

<p>Load CTLs calculated by the D2.0 version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctl.load(genotypes = "ngenotypes.txt", phenotypes = "nphenotypes.txt",
output = "ctlout", from=1, to, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctl.load_+3A_genotypes">genotypes</code></td>
<td>
<p> Original datafile containing the genotypes scanned. </p>
</td></tr>
<tr><td><code id="ctl.load_+3A_phenotypes">phenotypes</code></td>
<td>
<p> Original datafile containing the phenotypes scanned. </p>
</td></tr>
<tr><td><code id="ctl.load_+3A_output">output</code></td>
<td>
<p> Directory containing the output files. </p>
</td></tr>
<tr><td><code id="ctl.load_+3A_from">from</code></td>
<td>
<p> Start loading at which phenotype. </p>
</td></tr>
<tr><td><code id="ctl.load_+3A_to">to</code></td>
<td>
<p> Continue loading untill this phenotype. </p>
</td></tr>
<tr><td><code id="ctl.load_+3A_verbose">verbose</code></td>
<td>
<p> Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>CTLobject, a list with at each index a CTLscan object:
</p>

<ul>
<li><p> $ctls - Matrix of differential correlation scores for each trait at each marker
</p>
</li>
<li><p> $qtl  - Vector of QTL lodscores for each marker (if a QTL scan was perfomed -qtl)
</p>
</li>
<li><p> $p    - Vector of maximum scores per marker obtained during permutations
</p>
</li>
<li><p> $l    - Matrix of LOD scores for CTL likelihood
</p>
</li></ul>



<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)         # Load CTL library
</code></pre>

<hr>
<h2 id='CTLhelper'>CTLhelper - Helper functions for CTL mapping </h2><span id='topic+ctl.name'></span><span id='topic+ctl.names'></span><span id='topic+ctl.ctlmatrix'></span><span id='topic+ctl.qtlmatrix'></span><span id='topic+ctl.dcormatrix'></span><span id='topic+ctl.qtlprofile'></span>

<h3>Description</h3>

<p>Helper functions for Correlated Trait Locus (CTL) mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctl.names(CTLobject)
ctl.qtlmatrix(CTLobject)

ctl.name(CTLscan)
ctl.ctlmatrix(CTLscan)
ctl.dcormatrix(CTLscan)
ctl.qtlprofile(CTLscan)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLhelper_+3A_ctlobject">CTLobject</code></td>
<td>
<p>  An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="CTLhelper_+3A_ctlscan">CTLscan</code></td>
<td>
<p> An object of class <code>"CTLscan"</code>. This is a single element from an <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #TODO
</code></pre>

<hr>
<h2 id='CTLmapping'>CTLmapping - Scan for correlated trait loci (CTL) </h2><span id='topic+CTLmapping'></span>

<h3>Description</h3>

<p>Scan for correlated trait loci (CTL)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLmapping(genotypes, phenotypes, phenocol = 1, nperm = 100, nthreads = 1,
strategy = c("Exact", "Full", "Pairwise"), adjust = TRUE, qtl = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLmapping_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse a single phenotype. </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_nperm">nperm</code></td>
<td>
<p>       Number of permutations to perform. This parameter is not used when method=&quot;Exact&quot;. </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_nthreads">nthreads</code></td>
<td>
<p>    Number of CPU cores to use during the analysis. </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_strategy">strategy</code></td>
<td>
<p>    The permutation strategy to use, either 
</p>

<ul>
<li><p> Exact: Uses exact calculations to calculate the likelihood of a difference 
in correlation: Cor(AA) - Cor(BB). Using a Bonferroni correction.
</p>
</li>
<li><p> Full: Most powerful analysis method - Compensate for marker and trait 
correlation structure (Breitling et al.).
</p>
</li>
<li><p> Pairwise: Suitable when we have a lot of markers and only a few 
traits (&lt; 50) (human GWAS)- Compensates only for marker correlation structure.
</p>
</li></ul>

<p>Note: Exact is the default and fastest option it uses a normal distribution for 
estimating p-values and uses bonferoni correction. It has however the least power to 
detect CTLs, the two other methods (Full and Pairwise) perform permutations to
assign significance. </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_adjust">adjust</code></td>
<td>
<p>      Adjust p-values for multiple testing (only used when strategy = Exact). </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_qtl">qtl</code></td>
<td>
<p>         Use the internal slow QTL mapping method to map QTLs. </p>
</td></tr>
<tr><td><code id="CTLmapping_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>

<ul>
<li><p> NOTE: Main bottleneck of the algorithm is the RAM available to the system
</p>
</li></ul>



<h3>Value</h3>

<p>CTLscan, a list of:
</p>

<ul>
<li><p> $dcor  - Matrix of differential correlation scores for each trait at each marker
</p>
</li>
<li><p> $perms - Vector of maximums per marker obtained during permutations
</p>
</li>
<li><p> $ctls  - Matrix of LOD scores for CTL likelihood
</p>
</li></ul>



<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLscan.cross">CTLscan.cross</a></code> - Use an R/qtl cross object with <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites) # Arabidopsis Thaliana dataset
  singlescan &lt;- CTLmapping(ath.metab$genotypes, ath.metab$phenotypes, phenocol = 23)

  plot(singlescan)      # Plot the results of the CTL scan for the phenotype

  summary &lt;- CTLsignificant(singlescan)
  summary               # Get a list of significant CTLs
</code></pre>

<hr>
<h2 id='CTLnetwork'>CTLnetwork - Interaction network from a genome-wide CTLscan of multiple traits </h2><span id='topic+CTLnetwork'></span>

<h3>Description</h3>

<p>Create a file containing the interaction network from a genome-wide CTLscan of multiple traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLnetwork(CTLobject, mapinfo, significance = 0.05, LODdrop = 2, 
what = c("names", "ids"), short = FALSE, add.qtls = FALSE, file = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLnetwork_+3A_ctlobject">CTLobject</code></td>
<td>
<p>     An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_mapinfo">mapinfo</code></td>
<td>
<p>       The mapinfo matrix with 3 columns: &quot;Chr&quot; - the chromosome number, &quot;cM&quot; - the 
location of the marker in centiMorgans and the 3rd column &quot;Mbp&quot; - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) should 
match those in the CTLobject (only significant markers will be annotated). </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_significance">significance</code></td>
<td>
<p>  Significance threshold for a genome wide false discovery rate (FDR). </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_loddrop">LODdrop</code></td>
<td>
<p>       Drop in LOD score needed before we assign an edge type. </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_what">what</code></td>
<td>
<p>          Return trait and marker names or column numbers (for indexing). </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_short">short</code></td>
<td>
<p>         Edges are markers when TRUE, otherwise markers are nodes (default). </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_add.qtls">add.qtls</code></td>
<td>
<p>      Should marker QTL trait interactions be added to the generated sif network file, 
QTLs are included when they are above -log10(significance/n.markers). </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_file">file</code></td>
<td>
<p>          A connection, or a character string naming the file to print to. If &quot;&quot; 
(the default), CTLnetwork prints to the standard output connection, the console 
unless redirected by sink. </p>
</td></tr>
<tr><td><code id="CTLnetwork_+3A_verbose">verbose</code></td>
<td>
<p>       Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Outputs a sif network file, and a node attribute file:
</p>

<ul>
<li><p> ctlnet&lt;FILE&gt;.sif - Shows CTL connections from Trait to Marker with edge descriptions
</p>
</li>
<li><p> ctlnet&lt;FILE&gt;.nodes - Attributes of the nodes (Traits and Genetic markers) nodes to this 
file can be used to either color chromosomes, or add chromosome locations.
</p>
</li></ul>



<h3>Value</h3>

<p>A matrix with significant CTL interactions and information in 5 Columns:
</p>

<ul>
<li><p> TRAIT1 - Trait ID of the origin trait
</p>
</li>
<li><p> MARKER - Marker ID at which the CTL was found
</p>
</li>
<li><p> TRAIT2 - Trait ID of the target trait
</p>
</li>
<li><p> LOD_C  - LOD score of the CTL interaction
</p>
</li>
<li><p> CAUSAL - Type of edge determined by QTL LOD-drop:
</p>

<ul>
<li><p> NA - CTL/QTL for TRAIT1 and/or TRAIT2 not found
</p>
</li>
<li><p> -1 - TRAIT1 is DOWNSTREAM of TRAIT2
</p>
</li>
<li><p> 0  - UNDETERMINED Edge
</p>
</li>
<li><p> 1  - TRAIT1 is UPSTREAM of TRAIT2
</p>
</li></ul>

</li>
<li><p> LOD_T1 - QTL LOD-score of TRAIT1 at MARKER
</p>
</li>
<li><p> LOD_T2 - QTL LOD-score of TRAIT2 at MARKER
</p>
</li></ul>



<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result)       # Arabidopsis Thaliana results
  data(ath.metabolites)  # Arabidopsis Thaliana data set

  ctls &lt;- CTLnetwork(ath.result, significance = 0.1)
  op &lt;- par(mfrow = c(2,1))
  plot(ctls)
  ctl.lineplot(ath.result, ath.metab$map, significance=0.1)
</code></pre>

<hr>
<h2 id='CTLprofiles'>CTLprofiles - Extract CTL interaction profiles </h2><span id='topic+CTLprofiles'></span>

<h3>Description</h3>

<p>Extract the CTL interaction profiles: phenotype x marker (p2m matrix) 
and phenotype x phenotype (p2p matrix) from a <code><a href="#topic+CTLscan">CTLscan</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLprofiles(CTLobject, against = c("markers","phenotypes"), significance = 0.05, 
verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLprofiles_+3A_ctlobject">CTLobject</code></td>
<td>
<p>     An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="CTLprofiles_+3A_against">against</code></td>
<td>
<p>       Plot the CTL against either: markers or phenotypes. </p>
</td></tr>
<tr><td><code id="CTLprofiles_+3A_significance">significance</code></td>
<td>
<p>  Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="CTLprofiles_+3A_verbose">verbose</code></td>
<td>
<p>       Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>These matrices can be combined with QTL information to perform de novo 
reconstruction of interaction networks.
</p>
<p>The 'against' parameter is by default set to &quot;markers&quot; which returns a 
phenotype x markers matrix (p2m matrix), which should be comparible to 
the QTL profiles of the traits. 
</p>
<p>When the 'against' parameter is set to &quot;phenotypes&quot; a phenotype x phenotype
matrix (p2p matrix) is returned, showing the interactions between the phenotypes.
</p>


<h3>Value</h3>

<p>Matrix: phenotypes x marker or phenotypes x phenotypes
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)         # Load CTL library
  data(ath.result)     # Arabidopsis Thaliana results
  p2m_matrix &lt;- CTLprofiles(ath.result, against="markers")
  p2p_matrix &lt;- CTLprofiles(ath.result, against="phenotypes")
</code></pre>

<hr>
<h2 id='CTLregions'>CTLregions - Get all significant interactions from a genome-wide CTLscan </h2><span id='topic+CTLregions'></span>

<h3>Description</h3>

<p>Get all significant interactions from a genome-wide CTLscan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLregions(CTLobject, mapinfo, phenocol = 1, significance = 0.05, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLregions_+3A_ctlobject">CTLobject</code></td>
<td>
<p>         An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="CTLregions_+3A_mapinfo">mapinfo</code></td>
<td>
<p>           The mapinfo matrix with 3 columns: &quot;Chr&quot; - the chromosome number, &quot;cM&quot; - the 
location of the marker in centiMorgans and the 3rd column &quot;Mbp&quot; - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) 
should match those in the CTLobject. </p>
</td></tr>
<tr><td><code id="CTLregions_+3A_phenocol">phenocol</code></td>
<td>
<p>          Which phenotype column should we analyse. </p>
</td></tr>
<tr><td><code id="CTLregions_+3A_significance">significance</code></td>
<td>
<p>      Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="CTLregions_+3A_verbose">verbose</code></td>
<td>
<p>           Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>A matrix significant CTL interactions with 4 columns: trait, marker, trait, lod
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  
  data(ath.metabolites)                 # Arabidopsis Thaliana data set
  data(ath.result)                      # Arabidopsis Thaliana CTL results
  regions &lt;- CTLregions(ath.result, ath.metab$map)
</code></pre>

<hr>
<h2 id='CTLscan'>CTLscan - Scan for Correlated Trait Locus (CTL) </h2><span id='topic+CTLscan'></span>

<h3>Description</h3>

<p>Scan for Correlated Trait Locus (CTL) in populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLscan(genotypes, phenotypes, phenocol, nperm=100, nthreads = 1, 
strategy = c("Exact", "Full", "Pairwise"),
parametric = FALSE, adjust=TRUE, qtl = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLscan_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse all phenotypes. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_nperm">nperm</code></td>
<td>
<p>       Number of permutations to perform. This parameter is not used when method=&quot;Exact&quot;. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_nthreads">nthreads</code></td>
<td>
<p>    Number of CPU cores to use during the analysis. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_strategy">strategy</code></td>
<td>
<p>    The permutation strategy to use, either 
</p>

<ul>
<li><p> Exact - Uses exact calculations to calculate the likelihood of a difference 
in correlation: Cor(AA) - Cor(BB). Using a Bonferroni correction.
</p>
</li>
<li><p> Full - Most powerful analysis method - Compensate for marker and trait 
correlation structure (Breitling et al.).
</p>
</li>
<li><p> Pairwise - Suitable when we have a lot of markers and only a few 
traits (&lt; 50) (human GWAS)- Compensates only for marker correlation structure.
</p>
</li></ul>

<p>Note: Exact is the default and fastest option it uses a normal distribution for 
estimating p-values and uses bonferoni correction. It has however the least power to 
detect CTLs, the two other methods (Full and Pairwise) perform permutations to
assign significance. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_parametric">parametric</code></td>
<td>
<p>  Use non-parametric testing (Spearman) or parametric testing (Pearson). The DEFAULT is to 
use non-parametric tests which are less sensitive to outliers in the phenotype data. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_adjust">adjust</code></td>
<td>
<p>      Adjust p-values for multiple testing (only used when strategy = Exact). </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_qtl">qtl</code></td>
<td>
<p>         Use the internal slow QTL mapping method to map QTLs. </p>
</td></tr>
<tr><td><code id="CTLscan_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the algorithm will not do QTL mapping, the qtl component of the output is an vector of 0 scores 
for LOD. This is to remove some computational burden, please use the have.qtls parameter to provide QTL data. 
Some computational bottleneck of the algorithm are:
</p>

<ul>
<li><p> RAM available to the system with large number of markers (100K+) and/or  phenotypes (100K+).
</p>
</li>
<li><p> Computational time with large sample sizes (5000+) and/or huge amount of phenotype data (100K+).
</p>
</li>
<li><p> Very very huge amounts of genotype markers (1M+)
</p>
</li></ul>

<p>Some way of avoiding these problems are: CTL mapping using only a single chromosome at a time and / or
selecting a smaller subsets of phenotype data for analysis.
</p>


<h3>Value</h3>

<p>CTLobject, a list with at each index (i) an CTLscan object:
</p>

<ul>
<li><p> $dcor  - Matrix of Z scores (method=Exact), or Power/Adjacency Z scores or for each trait at each
marker (n.markers x n.phenotypes)
</p>
</li>
<li><p> $perms - Vector of maximum scores obtained during permutations  (n.perms)
</p>
</li>
<li><p> $ctl   - Matrix of LOD scores for CTL likelihood of phenotype i (n.markers x n.phenotypes)
</p>
</li>
<li><p> $qtl   - Vector of LOD scores for QTL likelihood of phenotype i (n.markers)
</p>
</li></ul>



<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan.cross">CTLscan.cross</a></code> - Use an R/qtl cross object with <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLregions">CTLregions</a></code> - Regions with significant CTLs from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites)                 # Arabidopsis Thaliana data set

  ctlscan &lt;- CTLscan(ath.metab$genotypes, ath.metab$phenotypes, phenocol=1:4)
  ctlscan

  # Genetic regions with significant CTLs found for the first phenotype
  CTLregions(ctlscan, ath.metab$map, phenocol = 1)
  
  summary &lt;- CTLsignificant(ctlscan)    # Matrix of Trait, Marker, Trait interactions 
  summary                               # Get a list of significant CTLs

  nodes &lt;- ctl.lineplot(ctlscan, ath.metab$map)  # Line plot the phenotypes
  nodes
</code></pre>

<hr>
<h2 id='CTLscan.cross'>CTLscan.cross - Scan for Correlated Trait Locus (CTL) (R/qtl cross object) </h2><span id='topic+CTLscan.cross'></span>

<h3>Description</h3>

<p>Scan for Correlated Trait Locus (CTL) in populations (using an R/qtl cross object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLscan.cross(cross, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLscan.cross_+3A_cross">cross</code></td>
<td>
<p> An object of class <code>cross</code>. See <code><a href="qtl.html#topic+read.cross">read.cross</a></code> for details. </p>
</td></tr>
<tr><td><code id="CTLscan.cross_+3A_...">...</code></td>
<td>
<p> Passed to <code><a href="#topic+CTLscan">CTLscan</a></code> function: 
</p>

<ul>
<li><p> phenocol   - Which phenotype column should we analyse.
</p>
</li>
<li><p> method     - We provide 3 ways of mapping correlation differences across the genome:
</p>

<ul>
<li><p> Exact: Uses a Correlation to Z score transformation to calculate the
likelihood of a difference in correlation: 
Cor(AA) - Cor(BB)
</p>
</li>
<li><p> Power: More powerful analysis method using the squared difference in     
correlation:
(Cor(AA) - Cor(BB))^2
</p>
</li>
<li><p> Adjacency: Adjacency method which using the squared difference in squared   
correlation, but keeping the sign of correlation:
(sign*Cor(AA)^2 - sign*Cor(BB)^2)^2
</p>
</li></ul>

<p>Note: Exact is the default and fastest option it uses a normal distribution for 
estimating p-values and uses bonferoni correction. It has however the least power 
to detect CTLs, the two other methods (Power and Adjacency) perform permutations to
assign significance.
</p>
</li>
<li><p> n.perm     - Number of permutations to perform.
</p>
</li>
<li><p> strategy   - The permutation strategy to use, either Full (Compensate for marker and trait
correlation structure) or Pairwise (Compensate for marker correlation structure). 
This parameter is not used when method=&quot;Exact&quot;.
</p>
</li>
<li><p> conditions - A vector of experimental conditions applied during the experiment. These conditions 
will be used as covariates in the QTL modeling step.
</p>
</li>
<li><p> n.cores    - Number of CPU cores to use during the analysis.
</p>
</li>
<li><p> verbose    - Be verbose.
</p>
</li></ul>

</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>

<ul>
<li><p> NOTE: Main bottleneck of the algorithm is the RAM available to the system
</p>
</li></ul>



<h3>Value</h3>

<p>CTLscan object, a list with at each index a CTL matrix (Rows: Phenotypes, 
Columns: Genetic markers) for the phenotype.
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(multitrait)      # Arabidopsis Thaliana (R/qtl cross object)

  mtrait &lt;- calc.genoprob(multitrait)          # Calculate genotype probabilities
  qtls   &lt;- scanone(mtrait, pheno.col = 1)     # Scan for QTLS using R/qtl

  ctls   &lt;- CTLscan.cross(mtrait, phenocol = 1, qtl = FALSE)
  ctls[[1]]$qtl &lt;- qtls[,3]

  ctl.lineplot(ctls, qtls[,1:2], significance = 0.05) # Line plot all the phenotypes

  summary &lt;- CTLsignificant(ctls)              # Get a list of significant CTLs
  summary
</code></pre>

<hr>
<h2 id='CTLsignificant'>CTLsignificant - Get all significant interactions from a genome-wide CTLscan </h2><span id='topic+CTLsignificant'></span>

<h3>Description</h3>

<p>Get all significant interactions from a genome-wide CTLscan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTLsignificant(CTLobject, significance = 0.05, what = c("names","ids"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CTLsignificant_+3A_ctlobject">CTLobject</code></td>
<td>
<p>    An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="CTLsignificant_+3A_significance">significance</code></td>
<td>
<p> Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="CTLsignificant_+3A_what">what</code></td>
<td>
<p>         Return trait and marker names or column numbers (for indexing). </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>A matrix significant CTL interactions with 4 columns: trait, marker, trait, lod
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result)
  all_interactions &lt;- CTLsignificant(ath.result)
  all_interactions[1:10, ]
  trait1_interactions &lt;- CTLsignificant(ath.result[[1]])
  trait1_interactions
</code></pre>

<hr>
<h2 id='detect.peaks'>detect.peaks - Peak detection algorithm to 'flatten' data above a certain threshold </h2><span id='topic+detect.peaks'></span>

<h3>Description</h3>

<p>Peak detection algorithm to 'flatten' data above a certain threshold
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detect.peaks(data, chrEdges = c(1), threshold = 4, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detect.peaks_+3A_data">data</code></td>
<td>
<p>        A vector of scores per marker/locus. </p>
</td></tr>
<tr><td><code id="detect.peaks_+3A_chredges">chrEdges</code></td>
<td>
<p>    Start positions of the chromosomes. </p>
</td></tr>
<tr><td><code id="detect.peaks_+3A_threshold">threshold</code></td>
<td>
<p>   Threshold to determine regions. </p>
</td></tr>
<tr><td><code id="detect.peaks_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  #TODO
</code></pre>

<hr>
<h2 id='hist.CTLobject'>Plot histogram of CTL permutations</h2><span id='topic+hist.CTLobject'></span>

<h3>Description</h3>

<p>Plot histogram of CTL permutations (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLobject'
hist(x, phenocol=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hist.CTLobject_+3A_x">x</code></td>
<td>
<p>           An object of class <code>"CTLscan"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="hist.CTLobject_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Defaults to analyse all phenotype columns </p>
</td></tr>
<tr><td><code id="hist.CTLobject_+3A_...">...</code></td>
<td>
<p>       Passed to the function <code><a href="graphics.html#topic+image">image</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>For a detailed description, see <code><a href="#topic+CTLprofiles">CTLprofiles</a></code>
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLprofiles">CTLprofiles</a></code> - Extract CTL interaction profiles
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)                      # Load CTL library
  data(ath.result)
  hist(ath.result, phenocol = 1:3)  # Compare the results of the first 3 scans
</code></pre>

<hr>
<h2 id='image.CTLobject'>Plot genome-wide CTL on multiple traits</h2><span id='topic+image.CTLobject'></span>

<h3>Description</h3>

<p>Plot the CTL for genome-wide CTL on multiple traits (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLobject'
image(x, marker_info, against = c("markers","phenotypes"), significance = 0.05,
col=whiteblack, do.grid=TRUE, grid.col = "white", verbose = FALSE, add=FALSE, 
breaks = c(0, 1, 2, 3, 10, 10000), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="image.CTLobject_+3A_x">x</code></td>
<td>
<p>             An object of class <code>"CTLscan"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_marker_info">marker_info</code></td>
<td>
<p>   Information used to plot chromosome lines. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_against">against</code></td>
<td>
<p>       Plot which interaction matrice, options are: markers: the phenotype*marker or 
phenotypes: the phenotype*phenotypes matrix. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_significance">significance</code></td>
<td>
<p>  Significance threshold to set a genome wide False Discovery Rate (FDR). </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_col">col</code></td>
<td>
<p>           Color-range used in plotting. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_do.grid">do.grid</code></td>
<td>
<p>       When TRUE, grid lines are added to the plot. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_grid.col">grid.col</code></td>
<td>
<p>      Color used for the grid lines, only used when do.grid = TRUE. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_verbose">verbose</code></td>
<td>
<p>       Be verbose. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_add">add</code></td>
<td>
<p>           Add this plot to a previously opened plot window. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_breaks">breaks</code></td>
<td>
<p>        See par. </p>
</td></tr>
<tr><td><code id="image.CTLobject_+3A_...">...</code></td>
<td>
<p>         Passed to the function <code><a href="graphics.html#topic+image">image</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>For a detailed description, see <code><a href="#topic+CTLprofiles">CTLprofiles</a></code>
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLprofiles">CTLprofiles</a></code> - Extract CTL interaction profiles
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result)       # Arabidopsis Thaliana results
  #Phenotype to phenotype matrix
  p2p_matrix &lt;- image(ath.result, against="phenotypes")
  #Phenotype to marker matrix
  p2m_matrix &lt;- image(ath.result, against="markers")
</code></pre>

<hr>
<h2 id='plot.CTLobject'>Plot CTL curves or heatmaps</h2><span id='topic+plot.CTLobject'></span>

<h3>Description</h3>

<p>Plot the CTL curve or heatmaps for a genome scan (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLobject'
plot(x, phenocol = 1:length(x), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CTLobject_+3A_x">x</code></td>
<td>
<p>           An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="plot.CTLobject_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we plot. Defaults to creating an image of all phenotype columns </p>
</td></tr>
<tr><td><code id="plot.CTLobject_+3A_...">...</code></td>
<td>
<p>       Passed to the function <code><a href="#topic+image.CTLobject">image.CTLobject</a></code> <code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result) # Arabidopsis Thaliana dataset
  plot(ath.result)
</code></pre>

<hr>
<h2 id='plot.CTLpermute'>Differential correlation versus likelihood plotted in curves</h2><span id='topic+plot.CTLpermute'></span>

<h3>Description</h3>

<p>Differential correlation versus likelihood plot curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLpermute'
plot(x, type="s", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CTLpermute_+3A_x">x</code></td>
<td>
<p>       An object of class <code>"CTLscan"</code>. </p>
</td></tr>
<tr><td><code id="plot.CTLpermute_+3A_type">type</code></td>
<td>
<p>    What type of plot should be drawn. for possible options see <code><a href="base.html#topic+plot">plot</a></code>. </p>
</td></tr>
<tr><td><code id="plot.CTLpermute_+3A_...">...</code></td>
<td>
<p>   Passed to the function <code><a href="base.html#topic+plot">plot</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result) # Arabidopsis Thaliana dataset
  plot(ath.result[[1]]$perms)
</code></pre>

<hr>
<h2 id='plot.CTLscan'>Plot CTL results as bar, line or GWAS plot. </h2><span id='topic+plot.CTLscan'></span>

<h3>Description</h3>

<p>Plot the CTL results for a genome scan (the output of <code><a href="#topic+CTLscan">CTLscan</a></code>) as a barplot, curved line 
or GWAS plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLscan'
plot(x, mapinfo = NULL, type = c("barplot","gwas","line"), 
onlySignificant = TRUE, significance = 0.05, gap = 25, plot.cutoff = FALSE, 
do.legend=TRUE, legend.pos = "topleft", cex.legend=1.0, ydim=NULL, 
ylab="-log10(P-value)", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CTLscan_+3A_x">x</code></td>
<td>
<p>                 An object of class <code>"CTLscan"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_mapinfo">mapinfo</code></td>
<td>
<p>           The mapinfo matrix with 3 columns: &quot;Chr&quot; - the chromosome number, &quot;cM&quot; - the 
location of the marker in centiMorgans and the 3rd column &quot;Mbp&quot; - The location 
of the marker in Mega basepairs. If supplied the marker names (rownames) 
should match those in the CTLobject. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_type">type</code></td>
<td>
<p>              Type of plot: Summed barplot, GWAS style plot or a basic line plot. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_onlysignificant">onlySignificant</code></td>
<td>
<p>   Plot only the significant contributions to the CTL profile. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_significance">significance</code></td>
<td>
<p>      Significance threshold for setting a genomewide FDR. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_gap">gap</code></td>
<td>
<p>               Gap in Cm between chromosomes. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_plot.cutoff">plot.cutoff</code></td>
<td>
<p>       Adds a line at -log10(significance) and adds a legend showing the 
significance level. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_do.legend">do.legend</code></td>
<td>
<p>         Adds a legend showing which phenotypes contribute to the CTL profile. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_legend.pos">legend.pos</code></td>
<td>
<p>        Position of the legend in the plot window. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_cex.legend">cex.legend</code></td>
<td>
<p>        Maginification of the text in the legend. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_ydim">ydim</code></td>
<td>
<p>              Dimension of the y-axis, if NULL then it will be calculated. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_ylab">ylab</code></td>
<td>
<p>              Label for the y-axis. </p>
</td></tr>
<tr><td><code id="plot.CTLscan_+3A_...">...</code></td>
<td>
<p>             Passed to the function <code><a href="base.html#topic+plot">plot</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.result)       # Arabidopsis thaliana results
  data(ath.metabolites)  # Arabidopsis thaliana data (phenotypes, genotypes and mapinfo

  plot(ath.result[[3]])
  plot(ath.result[[2]], mapinfo = ath.metab[[3]])
  plot(ath.result[[1]], mapinfo = ath.metab[[3]])
  plot(ath.result[[3]], mapinfo = ath.metab[[3]])
  plot(ath.result[[3]], mapinfo = ath.metab[[3]], type="gwas")
  plot(ath.result[[3]], mapinfo = ath.metab[[3]], type="line")
</code></pre>

<hr>
<h2 id='plotTraits'>plotTraits - Trait vs Trait scatterplot, colored by the selected genetic locus </h2><span id='topic+plotTraits'></span>

<h3>Description</h3>

<p>Trait vs Trait scatterplot, colored by the selected genetic locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTraits(genotypes, phenotypes, phenocol = c(1, 2), marker = 1, doRank = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTraits_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="plotTraits_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="plotTraits_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should be plotted against each other, Default: phenotype 1 versus 2 </p>
</td></tr>
<tr><td><code id="plotTraits_+3A_marker">marker</code></td>
<td>
<p>      Which marker (column in genotypes) should be used to add genotype as a color of the dots. </p>
</td></tr>
<tr><td><code id="plotTraits_+3A_dorank">doRank</code></td>
<td>
<p>      Transform quantitative data into ranked data before analyzing the slope. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites)                 # Arabidopsis Thaliana data set

  plotTraits(ath.metab$genotypes, ath.metab$phenotypes, marker=75, doRank = TRUE)
</code></pre>

<hr>
<h2 id='print.CTLobject'>Print the results of a CTL genome scan</h2><span id='topic+print.CTLobject'></span>

<h3>Description</h3>

<p>Print the results of a multiple phenotype CTL genome scan produced by <code><a href="#topic+CTLscan">CTLscan</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ## S3 method for class 'CTLobject'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.CTLobject_+3A_x">x</code></td>
<td>
<p>       An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="print.CTLobject_+3A_...">...</code></td>
<td>
<p>   Ignored.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None. 
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot a CTLscan object
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  #TODO
</code></pre>

<hr>
<h2 id='print.CTLscan'>Print the results of a single phenotype CTL scan</h2><span id='topic+print.CTLscan'></span>

<h3>Description</h3>

<p>Print the results of a single phenotype CTL scan produced by either <code><a href="#topic+CTLmapping">CTLmapping</a></code> (Single phenotype scan) or 
<code><a href="#topic+CTLscan">CTLscan</a></code> (Multi phenotype scan).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CTLscan'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.CTLscan_+3A_x">x</code></td>
<td>
<p> An object of class <code>"CTLscan"</code>. This is a single element from an <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="print.CTLscan_+3A_...">...</code></td>
<td>
<p> Ignored.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None. 
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot a CTLscan object
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  #TODO
</code></pre>

<hr>
<h2 id='qtlimage'>Plot a QTL heatmap of the phenotypes scanned by CTLscan</h2><span id='topic+qtlimage'></span>

<h3>Description</h3>

<p>Plots the QTL heatmap of a genome wide QTL scan (part of the output of <code><a href="#topic+CTLscan">CTLscan</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  qtlimage(x, marker_info, do.grid=TRUE, grid.col="white", verbose=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtlimage_+3A_x">x</code></td>
<td>
<p>         An object of class <code>"CTLobject"</code>, as output by <code><a href="#topic+CTLscan">CTLscan</a></code>. </p>
</td></tr>
<tr><td><code id="qtlimage_+3A_marker_info">marker_info</code></td>
<td>
<p>   Information used to plot chromosome lines. </p>
</td></tr>
<tr><td><code id="qtlimage_+3A_do.grid">do.grid</code></td>
<td>
<p>   When TRUE, grid lines are added to the plot. </p>
</td></tr>
<tr><td><code id="qtlimage_+3A_grid.col">grid.col</code></td>
<td>
<p>  Color used for the grid lines, only used when do.grid = TRUE. </p>
</td></tr>
<tr><td><code id="qtlimage_+3A_verbose">verbose</code></td>
<td>
<p>   Be verbose. </p>
</td></tr>
<tr><td><code id="qtlimage_+3A_...">...</code></td>
<td>
<p>     Passed to the function <code><a href="base.html#topic+plot">plot</a></code> when it is called. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+print.CTLscan">print.CTLscan</a></code> - Print a summary of a CTLscan
</p>
</li>
<li> <p><code><a href="graphics.html#topic+par">par</a></code> - Plot parameters
</p>
</li>
<li> <p><code><a href="grDevices.html#topic+colors">colors</a></code> - Colors used in plotting
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)          # Load CTL library
  data(ath.metabolites) # Arabidopsis Thaliana data
  data(ath.result)      # Arabidopsis Thaliana results
  qtlimage(ath.result, ath.metab$map)  # Plot only the qtls
</code></pre>

<hr>
<h2 id='QTLmapping'>QTLmapping - QTL mapping method for CTL analysis </h2><span id='topic+QTLmapping'></span>

<h3>Description</h3>

<p>Internal QTL mapping method used by the CTL analysis, associates every column in the genotypes with a single phenotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QTLmapping(genotypes, phenotypes, phenocol = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QTLmapping_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="QTLmapping_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="QTLmapping_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse a single phenotype. </p>
</td></tr>
<tr><td><code id="QTLmapping_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>

<ul>
<li><p> NOTE: Slow approach, it is adviced to use your own QTL mapping data
</p>
</li></ul>



<h3>Value</h3>

<p>vector of LOD scores for each genotype column, for phenotype column phenocol
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLscan.cross">CTLscan.cross</a></code> - Use an R/qtl cross object with <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites) # Arabidopsis Thaliana dataset
  qtldata &lt;- QTLmapping(ath.metab$genotypes, ath.metab$phenotypes, phenocol = 23)
  plot(qtldata)         # Plot the results of the QTL scan for the phenotype
</code></pre>

<hr>
<h2 id='scanSD'>scanSD - Analyze the differences in Standard Deviation between genotypes between two traits </h2><span id='topic+scanSD'></span>

<h3>Description</h3>

<p>Analyze the differences in Standard Deviation between genotypes between two traits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanSD(genotypes, phenotypes, phenocol=c(1,2), doRank = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanSD_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="scanSD_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="scanSD_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should be plotted against each other, Default: phenotype 1 versus 2 </p>
</td></tr>
<tr><td><code id="scanSD_+3A_dorank">doRank</code></td>
<td>
<p>      Transform quantitative data into ranked data before analyzing the slope. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(multitrait)      # Arabidopsis Thaliana (R/qtl cross object)

  sds &lt;- scanSD(pull.geno(multitrait),pull.pheno(multitrait))
</code></pre>

<hr>
<h2 id='scanSD.cross'>scanSD.cross - Analyze the differences in standard deviation between two traits at a certain genetic marker (R/qtl cross object) </h2><span id='topic+scanSD.cross'></span>

<h3>Description</h3>

<p>Analyze the differences in standard deviation between two traits at a certain genetic marker
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanSD.cross(cross, phenocol = c(1,2), doRank = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanSD.cross_+3A_cross">cross</code></td>
<td>
<p>       An object of class <code>cross</code>. See <code><a href="qtl.html#topic+read.cross">read.cross</a></code> for details. </p>
</td></tr>
<tr><td><code id="scanSD.cross_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should be plotted against each other, Default: phenotype 1 versus 2 </p>
</td></tr>
<tr><td><code id="scanSD.cross_+3A_dorank">doRank</code></td>
<td>
<p>      Transform quantitative data into ranked data before analyzing the slope. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(multitrait)      # Arabidopsis Thaliana (R/qtl cross object)

  sds &lt;- scanSD.cross(multitrait)
</code></pre>

<hr>
<h2 id='scanSlopes'>scanSlopes - Create a slope difference profile between two traits </h2><span id='topic+scanSlopes'></span>

<h3>Description</h3>

<p>Create a slope difference profile between two traits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanSlopes(genotypes, phenotypes, phenocol = 1, doRank = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanSlopes_+3A_genotypes">genotypes</code></td>
<td>
<p>   Matrix of genotypes. (individuals x markers) </p>
</td></tr>
<tr><td><code id="scanSlopes_+3A_phenotypes">phenotypes</code></td>
<td>
<p>  Matrix of phenotypes. (individuals x phenotypes) </p>
</td></tr>
<tr><td><code id="scanSlopes_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse phenotype 1. </p>
</td></tr>
<tr><td><code id="scanSlopes_+3A_dorank">doRank</code></td>
<td>
<p>      Transform quantitative data into ranked data before analyzing the slope. </p>
</td></tr>
<tr><td><code id="scanSlopes_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(ath.metabolites)                 # Arabidopsis Thaliana data set

  slopes &lt;- scanSlopes(ath.metab$genotypes, ath.metab$phenotypes[,1:4], phenocol = 2)
  image(1:nrow(slopes), 1:ncol(slopes), -log10(slopes))
</code></pre>

<hr>
<h2 id='scanSlopes.cross'>scanSlopes.cross - Create a slope difference profile between two traits (R/qtl cross object) </h2><span id='topic+scanSlopes.cross'></span>

<h3>Description</h3>

<p>Create a slope difference profile between two traits (using an R/qtl cross object)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scanSlopes.cross(cross, phenocol = 1, doRank = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scanSlopes.cross_+3A_cross">cross</code></td>
<td>
<p>       An object of class <code>cross</code>. See <code><a href="qtl.html#topic+read.cross">read.cross</a></code> for details. </p>
</td></tr>
<tr><td><code id="scanSlopes.cross_+3A_phenocol">phenocol</code></td>
<td>
<p>    Which phenotype column(s) should we analyse. Default: Analyse phenotype 1 </p>
</td></tr>
<tr><td><code id="scanSlopes.cross_+3A_dorank">doRank</code></td>
<td>
<p>      Transform quantitative data into ranked data before analyzing the slope. </p>
</td></tr>
<tr><td><code id="scanSlopes.cross_+3A_verbose">verbose</code></td>
<td>
<p>     Be verbose. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Note</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

 
<p>Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a><br />
Maintainer: Danny Arends <a href="mailto:Danny.Arends@gmail.com">Danny.Arends@gmail.com</a> 
</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+CTLscan">CTLscan</a></code> - Main function to scan for CTL
</p>
</li>
<li> <p><code><a href="#topic+CTLsignificant">CTLsignificant</a></code> - Significant interactions from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+CTLnetwork">CTLnetwork</a></code> - Create a CTL network from a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+image.CTLobject">image.CTLobject</a></code> - Heatmap overview of a <code><a href="#topic+CTLscan">CTLscan</a></code>
</p>
</li>
<li> <p><code><a href="#topic+plot.CTLscan">plot.CTLscan</a></code> - Plot the CTL curve for a single trait
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(multitrait)      # Arabidopsis Thaliana (R/qtl cross object)
  multitrait$pheno &lt;- multitrait$pheno[,1:4]
  slopes &lt;- scanSlopes.cross(multitrait)
  image(1:nrow(slopes), 1:ncol(slopes), -log10(slopes))
</code></pre>

<hr>
<h2 id='yeast.brem'> 
Example gene expression data from Saccharomyces cerevisiae on 301 RNA expressions.
</h2><span id='topic+yeast.brem'></span>

<h3>Description</h3>

<p>Saccharomyces recombinant inbred lines. There are 109 lines, 301 phenotypes, genotyped 
at 282 markers on 16 chromosomes stored as a list with 3 matrices: genotypes, phenotypes
and map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(yeast.brem)
</code></pre>


<h3>Format</h3>

<p>Data stored in a list holding 3 matrices genotypes, phenotypes and map
</p>


<h3>Details</h3>

<p>Saccharomyces recombinant inbred lines. There are 109 lines, 301 RNA expression phenotypes.
The individuals are genotyped at 282 markers on 16 chromosomes.
</p>


<h3>Source</h3>

<p>Saccharomyces cerevisiae RNA expression data from XX, senior author: Rachel Brem 20XX, 
Published in: Plos
</p>


<h3>References</h3>

<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(ctl)
  data(yeast.brem)              # Yeast data set
  
  yeast.brem$genotypes[1:5, 1:5]
  yeast.brem$phenotypes[1:5, 1:5]
  yeast.brem$map[1:5, ]
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
