<!DOCTYPE html><html lang="en"><head><title>Help for package cubar</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cubar}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aa2codon'><p>amino acids to codons</p></a></li>
<li><a href='#check_cds'><p>Quality control of CDS</p></a></li>
<li><a href='#codon_diff'><p>Differential codon usage analysis</p></a></li>
<li><a href='#codon_optimize'><p>Optimize codons</p></a></li>
<li><a href='#count_codons'><p>Count occurrences of different codons</p></a></li>
<li><a href='#create_codon_table'><p>create custom codon table from a data frame</p></a></li>
<li><a href='#est_csc'><p>Estimate Codon Stabilization Coefficient</p></a></li>
<li><a href='#est_optimal_codons'><p>Estimate optimal codons</p></a></li>
<li><a href='#est_rscu'><p>Estimate RSCU</p></a></li>
<li><a href='#est_trna_weight'><p>Estimate tRNA weight w</p></a></li>
<li><a href='#get_cai'><p>Calculate CAI</p></a></li>
<li><a href='#get_codon_table'><p>get codon table by NCBI gene code ID</p></a></li>
<li><a href='#get_cscg'><p>Mean Codon Stabilization Coefficients</p></a></li>
<li><a href='#get_dp'><p>Deviation from Proportionality</p></a></li>
<li><a href='#get_enc'><p>Calculate ENC</p></a></li>
<li><a href='#get_fop'><p>Fraction of optimal codons (Fop)</p></a></li>
<li><a href='#get_gc'><p>GC contents</p></a></li>
<li><a href='#get_gc3s'><p>GC contents at synonymous 3rd codon positions</p></a></li>
<li><a href='#get_gc4d'><p>GC contents at 4-fold degenerate sites</p></a></li>
<li><a href='#get_tai'><p>Calculate TAI</p></a></li>
<li><a href='#human_mt'><p>human mitochondrial CDS sequences</p></a></li>
<li><a href='#plot_ca_pairing'><p>Plot codon-anticodon pairing relationship</p></a></li>
<li><a href='#rev_comp'><p>Reverse complement</p></a></li>
<li><a href='#seq_to_codons'><p>Convert CDS to codons</p></a></li>
<li><a href='#show_codon_tables'><p>show available codon tables</p></a></li>
<li><a href='#slide'><p>slide window interval generator</p></a></li>
<li><a href='#slide_apply'><p>apply a cub index to a sliding window</p></a></li>
<li><a href='#slide_codon'><p>sliding window of codons</p></a></li>
<li><a href='#slide_plot'><p>plot sliding window codon usage</p></a></li>
<li><a href='#yeast_cds'><p>yeast CDS sequences</p></a></li>
<li><a href='#yeast_exp'><p>yeast mRNA expression levels</p></a></li>
<li><a href='#yeast_half_life'><p>Half life of yeast mRNAs</p></a></li>
<li><a href='#yeast_trna'><p>yeast tRNA sequences</p></a></li>
<li><a href='#yeast_trna_gcn'><p>yeast tRNA gene copy numbers (GCN)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Codon Usage Bias Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite of functions for rapid and flexible analysis of codon 
    usage bias. It provides in-depth analysis at the codon level, including 
    relative synonymous codon usage (RSCU), tRNA weight calculations, machine 
    learning predictions for optimal or preferred codons, and visualization of 
    codon-anticodon pairing. Additionally, it can calculate various gene-
    specific codon indices such as codon adaptation index (CAI), effective 
    number of codons (ENC), fraction of optimal codons (Fop), tRNA adaptation 
    index (tAI), mean codon stabilization coefficients (CSCg), and GC contents 
    (GC/GC3s/GC4d). It also supports both standard and non-standard genetic 
    code tables found in NCBI, as well as custom genetic code tables.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mt1022/cubar">https://github.com/mt1022/cubar</a>, <a href="https://mt1022.github.io/cubar/">https://mt1022.github.io/cubar/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mt1022/cubar/issues">https://github.com/mt1022/cubar/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>bzip2</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings (&ge; 2.60.0), IRanges (&ge; 2.34.0), data.table (&ge;
1.14.0), ggplot2 (&ge; 3.3.5), rlang (&ge; 0.4.11)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-07 07:31:04 UTC; mt1022</td>
</tr>
<tr>
<td>Author:</td>
<td>Hong Zhang <a href="https://orcid.org/0000-0002-4064-9432"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Mengyue Liu [aut],
  Bu Zi [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hong Zhang &lt;mt1022.dev@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-07 07:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aa2codon'>amino acids to codons</h2><span id='topic+aa2codon'></span>

<h3>Description</h3>

<p>A data.frame of mapping from amino acids to codons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa2codon
</code></pre>


<h3>Format</h3>

<p>a data.frame with two columns: amino_acid, and codon.
</p>

<dl>
<dt>amino_acid</dt><dd><p>amino acid corresponding to the codon</p>
</dd>
<dt>codon</dt><dd><p>codon identity</p>
</dd>
</dl>



<h3>Source</h3>

<p>It is actually the standard genetic code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aa2codon

</code></pre>

<hr>
<h2 id='check_cds'>Quality control of CDS</h2><span id='topic+check_cds'></span>

<h3>Description</h3>

<p><code>check_cds</code> performs quality control of CDS sequences by filtering some
peculiar sequences and optionally remove start or stop codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_cds(
  seqs,
  codon_table = get_codon_table(),
  min_len = 6,
  check_len = TRUE,
  check_start = TRUE,
  check_stop = TRUE,
  check_istop = TRUE,
  rm_start = TRUE,
  rm_stop = TRUE,
  start_codons = c("ATG")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_cds_+3A_seqs">seqs</code></td>
<td>
<p>input CDS sequences</p>
</td></tr>
<tr><td><code id="check_cds_+3A_codon_table">codon_table</code></td>
<td>
<p>codon table matching the genetic code of <code>seqs</code></p>
</td></tr>
<tr><td><code id="check_cds_+3A_min_len">min_len</code></td>
<td>
<p>minimum CDS length in nt</p>
</td></tr>
<tr><td><code id="check_cds_+3A_check_len">check_len</code></td>
<td>
<p>check whether CDS length is divisible by 3</p>
</td></tr>
<tr><td><code id="check_cds_+3A_check_start">check_start</code></td>
<td>
<p>check whether CDSs have start codons</p>
</td></tr>
<tr><td><code id="check_cds_+3A_check_stop">check_stop</code></td>
<td>
<p>check whether CDSs have stop codons</p>
</td></tr>
<tr><td><code id="check_cds_+3A_check_istop">check_istop</code></td>
<td>
<p>check internal stop codons</p>
</td></tr>
<tr><td><code id="check_cds_+3A_rm_start">rm_start</code></td>
<td>
<p>whether to remove start codons</p>
</td></tr>
<tr><td><code id="check_cds_+3A_rm_stop">rm_stop</code></td>
<td>
<p>whether to remove stop codons</p>
</td></tr>
<tr><td><code id="check_cds_+3A_start_codons">start_codons</code></td>
<td>
<p>vector of start codons</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DNAStringSet of filtered (and trimmed) CDS sequences
</p>


<h3>Examples</h3>

<pre><code class='language-R'># CDS sequence QC for a sample of yeast genes
s &lt;- head(yeast_cds, 10)
print(s)
check_cds(s)

</code></pre>

<hr>
<h2 id='codon_diff'>Differential codon usage analysis</h2><span id='topic+codon_diff'></span>

<h3>Description</h3>

<p><code>codon_diff</code> takes two set of coding sequences and
perform differential codon usage analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>codon_diff(seqs1, seqs2, codon_table = get_codon_table())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="codon_diff_+3A_seqs1">seqs1</code></td>
<td>
<p>DNAStringSet, or an object that can be coerced to a DNAStringSet</p>
</td></tr>
<tr><td><code id="codon_diff_+3A_seqs2">seqs2</code></td>
<td>
<p>DNAStringSet, or an object that can be coerced to a DNAStringSet</p>
</td></tr>
<tr><td><code id="codon_diff_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table of the differential codon usage analysis. Global tests examine wthether a codon
is used differently relative to all the other codons. Family tests examine whether a codon is used
differently relative to other codons that encode the same amino acid. Subfamily tests examine whether
a codon is used differently relative to other synonymous codons that share the same first two nucleotides.
Odds ratio &gt; 1 suggests a codon is used at higher frequency in <code>seqs1</code> than in <code>seqs2</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>yeast_exp_sorted &lt;- yeast_exp[order(yeast_exp$fpkm),]
seqs1 &lt;- yeast_cds[names(yeast_cds) %in% head(yeast_exp_sorted$gene_id, 1000)]
seqs2 &lt;- yeast_cds[names(yeast_cds) %in% tail(yeast_exp_sorted$gene_id, 1000)]
cudiff &lt;- codon_diff(seqs1, seqs2)

</code></pre>

<hr>
<h2 id='codon_optimize'>Optimize codons</h2><span id='topic+codon_optimize'></span>

<h3>Description</h3>

<p><code>codon_optimize</code> takes a coding sequence (without stop codon) and replace
each codon to the corresponding synonymous optimal codon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>codon_optimize(
  seq,
  optimal_codons,
  codon_table = get_codon_table(),
  level = "subfam"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="codon_optimize_+3A_seq">seq</code></td>
<td>
<p>DNAString, or an object that can be coerced to a DNAString.</p>
</td></tr>
<tr><td><code id="codon_optimize_+3A_optimal_codons">optimal_codons</code></td>
<td>
<p>table optimze codons as generated by <code>est_optimal_codons</code>.</p>
</td></tr>
<tr><td><code id="codon_optimize_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="codon_optimize_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. Optimize codon usage at which level.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a DNAString of the optimized coding sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cf_all &lt;- count_codons(yeast_cds)
optimal_codons &lt;- est_optimal_codons(cf_all)
seq &lt;- 'ATGCTACGA'
codon_optimize(seq, optimal_codons)

</code></pre>

<hr>
<h2 id='count_codons'>Count occurrences of different codons</h2><span id='topic+count_codons'></span>

<h3>Description</h3>

<p><code>count_codons</code> tabulates the occurrences of all the 64 codons in input CDSs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_codons(seqs, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="count_codons_+3A_seqs">seqs</code></td>
<td>
<p>CDS sequences, DNAStringSet.</p>
</td></tr>
<tr><td><code id="count_codons_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code>Biostrings::trinucleotideFrequency</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of codon (column) frequencies of each CDS (row).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># count codon occurrences
cf_all &lt;- count_codons(yeast_cds)
dim(cf_all)
cf_all[1:5, 1:5]
count_codons(yeast_cds[1])

</code></pre>

<hr>
<h2 id='create_codon_table'>create custom codon table from a data frame</h2><span id='topic+create_codon_table'></span>

<h3>Description</h3>

<p><code>create_codon_table</code> creates codon table from data frame of aa to codon mapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_codon_table(aa2codon)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_codon_table_+3A_aa2codon">aa2codon</code></td>
<td>
<p>a data frame with two columns: amino_acid (Ala, Arg, etc.) and codon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.table</code> with four columns: aa_code, amino_acid, codon, and subfam.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(aa2codon)
create_codon_table(aa2codon = aa2codon)
</code></pre>

<hr>
<h2 id='est_csc'>Estimate Codon Stabilization Coefficient</h2><span id='topic+est_csc'></span>

<h3>Description</h3>

<p><code>get_csc</code> calculate codon occurrence to mRNA stability correlation coefficients (Default to Pearson's).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_csc(
  seqs,
  half_life,
  codon_table = get_codon_table(),
  cor_method = "pearson"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_csc_+3A_seqs">seqs</code></td>
<td>
<p>CDS sequences of all protein-coding genes. One for each gene.</p>
</td></tr>
<tr><td><code id="est_csc_+3A_half_life">half_life</code></td>
<td>
<p>data.frame of mRNA half life (gene_id &amp; half_life are column names).</p>
</td></tr>
<tr><td><code id="est_csc_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or <code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="est_csc_+3A_cor_method">cor_method</code></td>
<td>
<p>method name passed to 'cor.test' used for calculating correlation coefficients.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table of optimal codons.
</p>


<h3>References</h3>

<p>Presnyak V, Alhusaini N, Chen YH, Martin S, Morris N, Kline N, Olson S, Weinberg D, Baker KE, Graveley BR, et al. 2015. Codon optimality is a major determinant of mRNA stability. Cell 160:1111-1124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate yeast mRNA CSC
est_csc(yeast_cds, yeast_half_life)

</code></pre>

<hr>
<h2 id='est_optimal_codons'>Estimate optimal codons</h2><span id='topic+est_optimal_codons'></span>

<h3>Description</h3>

<p><code>est_optimal_codons</code> determine optimal codon of each codon family with binomial regression.
Usage of optimal codons should correlate negatively with enc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_optimal_codons(
  cf,
  codon_table = get_codon_table(),
  level = "subfam",
  gene_score = NULL,
  fdr = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_optimal_codons_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="est_optimal_codons_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or <code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="est_optimal_codons_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine optimal codons.</p>
</td></tr>
<tr><td><code id="est_optimal_codons_+3A_gene_score">gene_score</code></td>
<td>
<p>a numeric vector of scores for genes. The order of values should match with
gene orders in the codon frequency matrix. The length of the vector should be equal to the
number of rows in the matrix. The scores could be gene expression levels (RPKM or TPM) that are
optionally log-transformed (for example, with <code>log1p</code>). The opposite of ENC will be used by
default if <code>gene_score</code> is not provided.</p>
</td></tr>
<tr><td><code id="est_optimal_codons_+3A_fdr">fdr</code></td>
<td>
<p>false discovery rate used to determine optimal codons.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table of optimal codons.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># perform binomial regression for optimal codon estimation
cf_all &lt;- count_codons(yeast_cds)
codons_opt &lt;- est_optimal_codons(cf_all)
codons_opt &lt;- codons_opt[optimal == TRUE]
codons_opt

</code></pre>

<hr>
<h2 id='est_rscu'>Estimate RSCU</h2><span id='topic+est_rscu'></span>

<h3>Description</h3>

<p><code>est_rscu</code> returns the RSCU value of codons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_rscu(
  cf,
  weight = 1,
  pseudo_cnt = 1,
  codon_table = get_codon_table(),
  level = "subfam"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_rscu_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="est_rscu_+3A_weight">weight</code></td>
<td>
<p>a vector of the same length as <code>seqs</code> that gives different weights
to CDSs when count codons. for example, it could be gene expression levels.</p>
</td></tr>
<tr><td><code id="est_rscu_+3A_pseudo_cnt">pseudo_cnt</code></td>
<td>
<p>pseudo count to avoid dividing by zero. This may occur when
only a few sequences are available for RSCU calculation.</p>
</td></tr>
<tr><td><code id="est_rscu_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="est_rscu_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine RSCU.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table of codon info. RSCU values are reported in the last column.
</p>


<h3>References</h3>

<p>Sharp PM, Tuohy TM, Mosurski KR. 1986. Codon usage in yeast: cluster analysis clearly differentiates highly and lowly expressed genes. Nucleic Acids Res 14:5125-5143.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># compute RSCU of all yeast genes
cf_all &lt;- count_codons(yeast_cds)
est_rscu(cf_all)

# compute RSCU of highly expressed (top 500) yeast genes
heg &lt;- head(yeast_exp[order(-yeast_exp$fpkm), ], n = 500)
cf_heg &lt;- count_codons(yeast_cds[heg$gene_id])
est_rscu(cf_heg)

</code></pre>

<hr>
<h2 id='est_trna_weight'>Estimate tRNA weight w</h2><span id='topic+est_trna_weight'></span>

<h3>Description</h3>

<p><code>est_trna_weight</code> compute the tRNA weight per codon for TAI calculation.
This weight reflects relative tRNA availability for each codon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_trna_weight(
  trna_level,
  codon_table = get_codon_table(),
  s = list(WC = 0, IU = 0, IC = 0.4659, IA = 0.9075, GU = 0.7861, UG = 0.6295)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_trna_weight_+3A_trna_level">trna_level</code></td>
<td>
<p>named vector of tRNA level (or gene copy numbers), one value for each anticodon.
vector names are anticodons.</p>
</td></tr>
<tr><td><code id="est_trna_weight_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or <code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="est_trna_weight_+3A_s">s</code></td>
<td>
<p>list of non-Waston-Crick pairing panelty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table of tRNA expression information.
</p>


<h3>References</h3>

<p>dos Reis M, Savva R, Wernisch L. 2004. Solving the riddle of codon usage preferences: a test for translational selection. Nucleic Acids Res 32:5036-5044.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate codon tRNA weight for yeasts
est_trna_weight(yeast_trna_gcn)

</code></pre>

<hr>
<h2 id='get_cai'>Calculate CAI</h2><span id='topic+get_cai'></span>

<h3>Description</h3>

<p><code>get_cai</code> calculates Codon Adaptation Index (CAI) of each input CDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cai(cf, rscu, level = "subfam")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_cai_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_cai_+3A_rscu">rscu</code></td>
<td>
<p>rscu table containing CAI weight for each codon. This table could be
generated with <code>est_rscu</code> or prepared manually.</p>
</td></tr>
<tr><td><code id="get_cai_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine CAI.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of CAI values
</p>


<h3>References</h3>

<p>Sharp PM, Li WH. 1987. The codon Adaptation Index&ndash;a measure of directional
synonymous codon usage bias, and its potential applications. Nucleic Acids Res 15:1281-1295.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate CAI of yeast genes based on RSCU of highly expressed genes
heg &lt;- head(yeast_exp[order(-yeast_exp$fpkm), ], n = 500)
cf_all &lt;- count_codons(yeast_cds)
cf_heg &lt;- cf_all[heg$gene_id, ]
rscu_heg &lt;- est_rscu(cf_heg)
cai &lt;- get_cai(cf_all, rscu_heg)
head(cai)
hist(cai)

</code></pre>

<hr>
<h2 id='get_codon_table'>get codon table by NCBI gene code ID</h2><span id='topic+get_codon_table'></span>

<h3>Description</h3>

<p><code>get_codon_table</code> creates a codon table based on the given id of genetic code in NCBI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_codon_table(gcid = "1")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_codon_table_+3A_gcid">gcid</code></td>
<td>
<p>a string of genetic code id. run <code>show_codon_tables()</code> to see available codon tables.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.table</code> with four columns: aa_code, amino_acid, codon, and subfam.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Standard genetic code
get_codon_table()

# Vertebrate Mitochondrial genetic code
get_codon_table(gcid = '2')
</code></pre>

<hr>
<h2 id='get_cscg'>Mean Codon Stabilization Coefficients</h2><span id='topic+get_cscg'></span>

<h3>Description</h3>

<p><code>get_cscg</code> calculates Mean Codon Stabilization Coefficients of each CDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cscg(cf, csc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_cscg_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_cscg_+3A_csc">csc</code></td>
<td>
<p>table of Codon Stabilization Coefficients as calculated by <code>est_csc()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of cscg values.
</p>


<h3>References</h3>

<p>Presnyak V, Alhusaini N, Chen YH, Martin S, Morris N, Kline N, Olson S, Weinberg D,
Baker KE, Graveley BR, et al. 2015. Codon optimality is a major determinant of mRNA stability.
Cell 160:1111-1124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate CSCg of yeast genes
yeast_csc &lt;- est_csc(yeast_cds, yeast_half_life)
cf_all &lt;- count_codons(yeast_cds)
cscg &lt;- get_cscg(cf_all, csc = yeast_csc)
head(cscg)
hist(cscg)

</code></pre>

<hr>
<h2 id='get_dp'>Deviation from Proportionality</h2><span id='topic+get_dp'></span>

<h3>Description</h3>

<p><code>get_dp</code> calculates Deviation from Proportionality of each CDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dp(
  cf,
  host_weights,
  codon_table = get_codon_table(),
  level = "subfam",
  missing_action = "ignore"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_dp_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_dp_+3A_host_weights">host_weights</code></td>
<td>
<p>a named vector of tRNA weights for each codon that reflects the relative
availability of tRNAs in the host organism.</p>
</td></tr>
<tr><td><code id="get_dp_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="get_dp_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. If &quot;subfam&quot;, the deviation is calculated at
the codon subfamily level. Otherwise, the deviation is calculated at the amino acid level.</p>
</td></tr>
<tr><td><code id="get_dp_+3A_missing_action">missing_action</code></td>
<td>
<p>Actions to take when no codon of a group were found in a CDS. Options are
&quot;ignore&quot; (default), or &quot;zero&quot; (set codon proportions to 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of dp values.
</p>


<h3>References</h3>

<p>Chen F, Wu P, Deng S, Zhang H, Hou Y, Hu Z, Zhang J, Chen X, Yang JR. 2020.
Dissimilation of synonymous codon usage bias in virus-host coevolution due to translational
selection. Nat Ecol Evol 4:589-600.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate DP of yeast genes
cf_all &lt;- count_codons(yeast_cds)
trna_weight &lt;- est_trna_weight(yeast_trna_gcn)
trna_weight &lt;- setNames(trna_weight$w, trna_weight$codon)
dp &lt;- get_dp(cf_all, host_weights = trna_weight)
head(dp)
hist(dp)

</code></pre>

<hr>
<h2 id='get_enc'>Calculate ENC</h2><span id='topic+get_enc'></span>

<h3>Description</h3>

<p><code>get_enc</code> computes ENC of each CDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_enc(cf, codon_table = get_codon_table(), level = "subfam")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_enc_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_enc_+3A_codon_table">codon_table</code></td>
<td>
<p>codon_table a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="get_enc_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine ENC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of ENC values, sequence names are used as vector names
</p>


<h3>References</h3>

<p>- Wright F. 1990. The 'effective number of codons' used in a gene. Gene 87:23-29.
- Sun X, Yang Q, Xia X. 2013. An improved implementation of effective number of codons (NC).
Mol Biol Evol 30:191-196.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate ENC of yeast genes
cf_all &lt;- count_codons(yeast_cds)
enc &lt;- get_enc(cf_all)
head(enc)
hist(enc)
</code></pre>

<hr>
<h2 id='get_fop'>Fraction of optimal codons (Fop)</h2><span id='topic+get_fop'></span>

<h3>Description</h3>

<p><code>get_fop</code> calculates the fraction of optimal codons (Fop) of each CDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_fop(cf, op = NULL, codon_table = get_codon_table(), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_fop_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_fop_+3A_op">op</code></td>
<td>
<p>a character vector of optimal codons. Can be determined automatically by running
<code>est_optimal_codons</code>.</p>
</td></tr>
<tr><td><code id="get_fop_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="get_fop_+3A_...">...</code></td>
<td>
<p>other arguments passed to <code>est_optimal_codons</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of fop values.
</p>


<h3>References</h3>

<p>Ikemura T. 1981. Correlation between the abundance of Escherichia coli transfer RNAs
and the occurrence of the respective codons in its protein genes: a proposal for a synonymous
codon choice that is optimal for the E. coli translational system. J Mol Biol 151:389-409.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate Fop of yeast genes
cf_all &lt;- count_codons(yeast_cds)
fop &lt;- get_fop(cf_all)
head(fop)
hist(fop)

</code></pre>

<hr>
<h2 id='get_gc'>GC contents</h2><span id='topic+get_gc'></span>

<h3>Description</h3>

<p>Calculate GC content of the whole sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gc(cf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gc_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by 'count_codons()'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of GC contents.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate GC content of yeast genes
cf_all &lt;- count_codons(yeast_cds)
gc &lt;- get_gc(cf_all)
head(gc)
hist(gc)

</code></pre>

<hr>
<h2 id='get_gc3s'>GC contents at synonymous 3rd codon positions</h2><span id='topic+get_gc3s'></span>

<h3>Description</h3>

<p>Calculate GC content at synonymous 3rd codon positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gc3s(cf, codon_table = get_codon_table(), level = "subfam")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gc3s_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_gc3s_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or <code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="get_gc3s_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine 
GC content at synonymous 3rd codon positions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of GC3s values.
</p>


<h3>References</h3>

<p>Peden JF. 2000. Analysis of codon usage.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate GC3s of yeast genes
cf_all &lt;- count_codons(yeast_cds)
gc3s &lt;- get_gc3s(cf_all)
head(gc3s)
hist(gc3s)

</code></pre>

<hr>
<h2 id='get_gc4d'>GC contents at 4-fold degenerate sites</h2><span id='topic+get_gc4d'></span>

<h3>Description</h3>

<p>Calculate GC content at synonymous position of codons (using four-fold degenerate sites only).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gc4d(cf, codon_table = get_codon_table(), level = "subfam")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gc4d_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_gc4d_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="get_gc4d_+3A_level">level</code></td>
<td>
<p>&quot;subfam&quot; (default) or &quot;amino_acid&quot;. For which level to determine 
GC contents at 4-fold degenerate sites.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of GC4d values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># estimate GC4d of yeast genes
cf_all &lt;- count_codons(yeast_cds)
gc4d &lt;- get_gc4d(cf_all)
head(gc4d)
hist(gc4d)

</code></pre>

<hr>
<h2 id='get_tai'>Calculate TAI</h2><span id='topic+get_tai'></span>

<h3>Description</h3>

<p><code>get_tai</code> calculates tRNA Adaptation Index (TAI) of each CDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tai(cf, trna_w)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_tai_+3A_cf">cf</code></td>
<td>
<p>matrix of codon frequencies as calculated by <code>count_codons()</code>.</p>
</td></tr>
<tr><td><code id="get_tai_+3A_trna_w">trna_w</code></td>
<td>
<p>tRNA weight for each codon, can be generated with <code>est_trna_weight()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named vector of TAI values
</p>


<h3>References</h3>

<p>dos Reis M, Savva R, Wernisch L. 2004. Solving the riddle of codon usage
preferences: a test for translational selection. Nucleic Acids Res 32:5036-5044.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculate TAI of yeast genes based on genomic tRNA copy numbers
w &lt;- est_trna_weight(yeast_trna_gcn)
cf_all &lt;- count_codons(yeast_cds)
tai &lt;- get_tai(cf_all, w)
head(tai)
hist(tai)

</code></pre>

<hr>
<h2 id='human_mt'>human mitochondrial CDS sequences</h2><span id='topic+human_mt'></span>

<h3>Description</h3>

<p>CDSs of 13 protein-coding genes in the human mitochondrial genome extracted from ENSEMBL Biomart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human_mt
</code></pre>


<h3>Format</h3>

<p>a DNAStringSet of 13 sequences
</p>


<h3>Source</h3>

<p>&lt;https://www.ensembl.org/index.html&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(human_mt)
</code></pre>

<hr>
<h2 id='plot_ca_pairing'>Plot codon-anticodon pairing relationship</h2><span id='topic+plot_ca_pairing'></span>

<h3>Description</h3>

<p><code>plot_ca_pairing</code> show possible codon-anticodons pairings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ca_pairing(codon_table = get_codon_table(), plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ca_pairing_+3A_codon_table">codon_table</code></td>
<td>
<p>a table of genetic code derived from <code>get_codon_table</code> or
<code>create_codon_table</code>.</p>
</td></tr>
<tr><td><code id="plot_ca_pairing_+3A_plot">plot</code></td>
<td>
<p>whether to plot the pairing relationship</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.table of codon info and RSCU values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ctab &lt;- get_codon_table(gcid = '2')
pairing &lt;- plot_ca_pairing(ctab)
head(pairing)

</code></pre>

<hr>
<h2 id='rev_comp'>Reverse complement</h2><span id='topic+rev_comp'></span>

<h3>Description</h3>

<p><code>rev_comp</code> creates reverse complemented version of the input sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rev_comp(seqs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rev_comp_+3A_seqs">seqs</code></td>
<td>
<p>input sequences, DNAStringSet or named vector of sequences</p>
</td></tr>
</table>


<h3>Value</h3>

<p>reverse complemented input sequences as a DNAStringSet.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># reverse complement of codons
rev_comp(Biostrings::DNAStringSet(c('TAA', 'TAG')))

</code></pre>

<hr>
<h2 id='seq_to_codons'>Convert CDS to codons</h2><span id='topic+seq_to_codons'></span>

<h3>Description</h3>

<p><code>seq_to_codons</code> converts a coding sequence to a vector of codons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seq_to_codons(seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="seq_to_codons_+3A_seq">seq</code></td>
<td>
<p>DNAString, or an object that can be coerced to a DNAString</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of codons
</p>


<h3>Examples</h3>

<pre><code class='language-R'># convert a CDS sequence to a sequence of codons
seq_to_codons('ATGTGGTAG')
seq_to_codons(yeast_cds[[1]])

</code></pre>

<hr>
<h2 id='show_codon_tables'>show available codon tables</h2><span id='topic+show_codon_tables'></span>

<h3>Description</h3>

<p><code>show_codon_tables</code> print a table of available genetic code from NCBI through
<code>Biostrings::GENETIC_CODE_TABLE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_codon_tables()
</code></pre>


<h3>Value</h3>

<p>No return value (NULL). Available codon tables will be printed out directly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># print available NCBI codon table IDs and descriptions.
show_codon_tables()
</code></pre>

<hr>
<h2 id='slide'>slide window interval generator</h2><span id='topic+slide'></span>

<h3>Description</h3>

<p><code>slide</code> generates a data.table with start, center, and end columns
for a sliding window analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slide(from, to, step = 1, before = 0, after = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slide_+3A_from">from</code></td>
<td>
<p>integer, the start of the sequence</p>
</td></tr>
<tr><td><code id="slide_+3A_to">to</code></td>
<td>
<p>integer, the end of the sequence</p>
</td></tr>
<tr><td><code id="slide_+3A_step">step</code></td>
<td>
<p>integer, the step size</p>
</td></tr>
<tr><td><code id="slide_+3A_before">before</code></td>
<td>
<p>integer, the number of values before the center of a window</p>
</td></tr>
<tr><td><code id="slide_+3A_after">after</code></td>
<td>
<p>integer, the number of values after the center of a window</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table with start, center, and end columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>slide(1, 10, step = 2, before = 1, after = 1)

</code></pre>

<hr>
<h2 id='slide_apply'>apply a cub index to a sliding window</h2><span id='topic+slide_apply'></span>

<h3>Description</h3>

<p><code>slide_apply</code> applies a function to a sliding window of codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slide_apply(seq, .f, step = 1, before = 0, after = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slide_apply_+3A_seq">seq</code></td>
<td>
<p>DNAString, the sequence</p>
</td></tr>
<tr><td><code id="slide_apply_+3A_.f">.f</code></td>
<td>
<p>function, the codon index calculation function to apply, for
example, <code>get_enc</code>.</p>
</td></tr>
<tr><td><code id="slide_apply_+3A_step">step</code></td>
<td>
<p>integer, the step size in number of codons</p>
</td></tr>
<tr><td><code id="slide_apply_+3A_before">before</code></td>
<td>
<p>integer, the number of codons before the center of a window</p>
</td></tr>
<tr><td><code id="slide_apply_+3A_after">after</code></td>
<td>
<p>integer, the number of codons after the center of a window</p>
</td></tr>
<tr><td><code id="slide_apply_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to the function <code>.f</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table with start, center, end, and codon usage index columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>slide_apply(yeast_cds[[1]], get_enc, step = 1, before = 10, after = 10)

</code></pre>

<hr>
<h2 id='slide_codon'>sliding window of codons</h2><span id='topic+slide_codon'></span>

<h3>Description</h3>

<p><code>slide_codon</code> generates a data.table with start, center, and end columns
for a sliding window analysis of codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slide_codon(seq, step = 1, before = 0, after = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slide_codon_+3A_seq">seq</code></td>
<td>
<p>DNAString, the sequence</p>
</td></tr>
<tr><td><code id="slide_codon_+3A_step">step</code></td>
<td>
<p>integer, the step size</p>
</td></tr>
<tr><td><code id="slide_codon_+3A_before">before</code></td>
<td>
<p>integer, the number of codons before the center of a window</p>
</td></tr>
<tr><td><code id="slide_codon_+3A_after">after</code></td>
<td>
<p>integer, the number of codons after the center of a window</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.table with start, center, and end columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- Biostrings::DNAString('ATCTACATAGCTACGTAGCTCGATGCTAGCATGCATCGTACGATCGTCGATCGTAG')
slide_codon(x, step = 3, before = 1, after = 1)

</code></pre>

<hr>
<h2 id='slide_plot'>plot sliding window codon usage</h2><span id='topic+slide_plot'></span>

<h3>Description</h3>

<p><code>slide_plot</code> visualizes codon usage in sliding window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slide_plot(windt, index_name = "Index")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slide_plot_+3A_windt">windt</code></td>
<td>
<p>data.table, the sliding window codon usage
generated by <code>slide_apply</code>.</p>
</td></tr>
<tr><td><code id="slide_plot_+3A_index_name">index_name</code></td>
<td>
<p>character, the name of the index to display.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sw &lt;- slide_apply(yeast_cds[[1]], get_enc, step = 1, before = 10, after = 10)
slide_plot(sw)
</code></pre>

<hr>
<h2 id='yeast_cds'>yeast CDS sequences</h2><span id='topic+yeast_cds'></span>

<h3>Description</h3>

<p>CDSs of all protein-coding genes in Saccharomyces_cerevisiae
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_cds
</code></pre>


<h3>Format</h3>

<p>a DNAStringSet of 6600 sequences
</p>


<h3>Source</h3>

<p>&lt;https://ftp.ensembl.org/pub/release-107/fasta/saccharomyces_cerevisiae/cds/Saccharomyces_cerevisiae.R64-1-1.cds.all.fa.gz&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(yeast_cds)
</code></pre>

<hr>
<h2 id='yeast_exp'>yeast mRNA expression levels</h2><span id='topic+yeast_exp'></span>

<h3>Description</h3>

<p>Yeast mRNA FPKM determined from rRNA-depleted (RiboZero) total RNA-Seq libraries. RUN1_0_WT and RUN2_0_WT (0 min after RNA Pol II repression) were averaged and used here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_exp
</code></pre>


<h3>Format</h3>

<p>a data.frame with 6717 rows and three columns:
</p>

<dl>
<dt>gene_id</dt><dd><p>gene ID</p>
</dd>
<dt>gene_name</dt><dd><p>gene name</p>
</dd>
<dt>fpkm</dt><dd><p>mRNA expression level in Fragments per kilobase per million reads</p>
</dd>
</dl>



<h3>Source</h3>

<p>&lt;https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57385&gt;
</p>


<h3>References</h3>

<p>Presnyak V, Alhusaini N, Chen YH, Martin S, Morris N, Kline N, Olson S, Weinberg D, Baker KE, Graveley BR, et al. 2015. Codon optimality is a major determinant of mRNA stability. Cell 160:1111-1124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(yeast_exp)
</code></pre>

<hr>
<h2 id='yeast_half_life'>Half life of yeast mRNAs</h2><span id='topic+yeast_half_life'></span>

<h3>Description</h3>

<p>Half life of yeast mRNAs in Saccharomyces_cerevisiae calculated from rRNA-deleted total RNAs by Presnyak et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_half_life
</code></pre>


<h3>Format</h3>

<p>a data.frame with 3888 rows and three columns:
</p>

<dl>
<dt>gene_id</dt><dd><p>gene id</p>
</dd>
<dt>gene_name</dt><dd><p>gene name</p>
</dd>
<dt>half_life</dt><dd><p>mRNA half life in minutes</p>
</dd>
</dl>



<h3>Source</h3>

<p>&lt;https://doi.org/10.1016/j.cell.2015.02.029&gt;
</p>


<h3>References</h3>

<p>Presnyak V, Alhusaini N, Chen YH, Martin S, Morris N, Kline N, Olson S, Weinberg D, Baker KE, Graveley BR, et al. 2015. Codon optimality is a major determinant of mRNA stability. Cell 160:1111-1124.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(yeast_half_life)
</code></pre>

<hr>
<h2 id='yeast_trna'>yeast tRNA sequences</h2><span id='topic+yeast_trna'></span>

<h3>Description</h3>

<p>Yeast tRNA sequences obtained from gtRNAdb.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_trna
</code></pre>


<h3>Format</h3>

<p>a RNAStringSet with a length of 275.
</p>


<h3>Source</h3>

<p>&lt;http://gtrnadb.ucsc.edu/genomes/eukaryota/Scere3/sacCer3-mature-tRNAs.fa&gt;
</p>


<h3>References</h3>

<p>Chan PP, Lowe TM. 2016. GtRNAdb 2.0: an expanded database of transfer RNA genes identified in complete and draft genomes. Nucleic Acids Res 44:D184-189.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>yeast_trna
</code></pre>

<hr>
<h2 id='yeast_trna_gcn'>yeast tRNA gene copy numbers (GCN)</h2><span id='topic+yeast_trna_gcn'></span>

<h3>Description</h3>

<p>Yeast tRNA gene copy numbers (GCN) by anticodon obtained from gtRNAdb.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_trna_gcn
</code></pre>


<h3>Format</h3>

<p>a named vector with a length of 41. Value names are anticodons.
</p>


<h3>Source</h3>

<p>&lt;http://gtrnadb.ucsc.edu/genomes/eukaryota/Scere3/sacCer3-mature-tRNAs.fa&gt;
</p>


<h3>References</h3>

<p>Chan PP, Lowe TM. 2016. GtRNAdb 2.0: an expanded database of transfer RNA genes identified in complete and draft genomes. Nucleic Acids Res 44:D184-189.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>yeast_trna_gcn
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
