<!DOCTYPE html><html lang="en"><head><title>Help for package CUMP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CUMP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CUMP-package'>
<p>Analyze Multivariate Phenotypes by Combining Univariate results</p></a></li>
<li><a href='#combGWAS'>
<p>Combining Univariate Association Test Results of Multiple Phenotypes for Detecting Pleiotropy</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyze Multivariate Phenotypes by Combining Univariate Results</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Shuo Li &lt;skyli@bu.edu&gt;, Xuan Liu &lt;liuxuan@bu.edu&gt; and Qiong Yang &lt;qyang@bu.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shuo Li &lt;skyli@bu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Combining Univariate Association Test Results of Multiple Phenotypes for Detecting Pleiotropy.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-14 21:44:38 UTC; skyli</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-15 00:27:00</td>
</tr>
</table>
<hr>
<h2 id='CUMP-package'>
Analyze Multivariate Phenotypes by Combining Univariate results 
</h2><span id='topic+CUMP-package'></span><span id='topic+CUMP'></span>

<h3>Description</h3>

<p>This package can be used to detect pleiotropy by combining univariate association test results of
multiple phenotypes in genome-wide association studies (or studies of a large number of SNPs). In this package, we have several combination approaches including the O'Brien's method that is weighted sum of the Z or beta statistic (direction sensitive) and other methods that are weighted sum of the squared Z statistics (direction insensitive).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> CUMP</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-08-25</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Shuo Li &lt;skyli@bu.edu&gt;, Xuan Liu &lt;liuxuan@bu.edu&gt; and Qiong Yang &lt;qyang@bu.edu&gt;
</p>
<p>Maintainer: Shuo Li &lt;skyli@bu.edu&gt;
</p>


<h3>References</h3>

<p>CUMP: an R package for analyzing multivariate phenotypes in genetic association studies
</p>

<hr>
<h2 id='combGWAS'>
Combining Univariate Association Test Results of Multiple Phenotypes for Detecting Pleiotropy
</h2><span id='topic+combGWAS'></span>

<h3>Description</h3>

<p>combGWAS() can be used to detect pleiotropy by combining univariate association test results of multiple phenotypes in genome-wide association studies (or studies of a large number of SNPs). In this function, we have several combination approaches including the O'Brien's method that is weighted sum of the Z or beta statistic (direction sensitive) and other methods that are weighted sum of the squared Z statistics (direction insensitive).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combGWAS(project = "mv", traitlist, traitfile, comb_method = c("z"), 
betasign = rep(1, length(traitlist)), snpid, beta = NULL, SE = NULL, 
Z = NULL, coded_all, AF_coded_all, n_total = NULL, pvalue = NULL, 
Z_sample_weighted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combGWAS_+3A_project">project</code></td>
<td>

<p>a character string for project name, for labeling output file.
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_traitlist">traitlist</code></td>
<td>

<p>a vector of character strings of the phenotype names for naming the output file. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_traitfile">traitfile</code></td>
<td>

<p>a vector of character strings containing the univariate association results file names corresponding to the order in <code>traitlist</code> for reading in for analysis. Each univariate result file should contain the header corresponding to following fields: <code>snpid</code>, <code>beta</code>, <code>SE</code>, <code>coded_all</code>, <code>AF_coded_all</code>, <code>n_total</code>, <code>pvalue</code>. These fields can be labeled in any name in the header. You are asked to give the names of these fields as arguments for this function. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_comb_method">comb_method</code></td>
<td>

<p>a vector of character strings  indicating the method to be used in combing univariate association results file. It can be any subset of c(&quot;z&quot;, &quot;beta&quot;, &quot;chisq&quot; and &quot;sumsq&quot;). The details of the combination methods are given in Details. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_betasign">betasign</code></td>
<td>

<p>a numeric vector for changing the signs (1 or -1) of the univariate beta (or Z) statistics. It should be of the same length and correspond to the order in <code>traitfile</code> (or <code>traitlist</code>).
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_snpid">snpid</code></td>
<td>

<p>the name of the genetic marker in the header of input association results files. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_beta">beta</code></td>
<td>

<p>the name of the beta estimate (if have) in the header of input association results files. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_se">SE</code></td>
<td>

<p>the name of the standard error of the beta estimate (if have) in the header of input association results files. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_z">Z</code></td>
<td>

<p>the name of the Z statistic (if have) in the header of input association results files. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_coded_all">coded_all</code></td>
<td>

<p>the name of coded allele in the header of input association results files.
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_af_coded_all">AF_coded_all</code></td>
<td>

<p>the name of the allele frequency of the coded allele in the header of input association results file.
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_n_total">n_total</code></td>
<td>

<p>the sample size with phenotype and genotype for the genetic marker in the header of input association results file. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_pvalue">pvalue</code></td>
<td>

<p>the name of the p-value of the beta estimate (if have) in the header of input association results file. 
</p>
</td></tr>
<tr><td><code id="combGWAS_+3A_z_sample_weighted">Z_sample_weighted</code></td>
<td>

<p>a logical value. True if the results of Z method are combined by sample size weighted. False if the results are combined equally weighted.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The orders of traits in <code>traitlist</code> and <code>traitfile</code> should be the same. 
<br />
<br />
Currently, 4 combination methods (&quot;z&quot;, &quot;beta&quot;, &quot;chisq&quot; and &quot;sumsq&quot;) can be implemented by the package. The default is to implement equally weighted &quot;z&quot; method only, but you can ask for the 4 methods simultaneously. 
<br />
<br />
<code>betasign</code> should be a vector of 1 or -1 with the number of traits as the length. 1 means the beta of corresponding trait remains and -1 means the beta sign will be reversed. It only affects &quot;z&quot; and &quot;beta&quot; methods. 
<br />
<br />
<code>snpid</code>, <code>coded_all</code> and <code>AF_coded_all</code> must be assigned explicitly and the corresponding columns must appear in the input datasets. 
<br />
<br />
At least one of <code>beta(SE)</code> and <code>Z</code> should be assigned. In particular, if &quot;beta&quot; method is implemented, <code>beta</code> and <code>SE</code> must be assigned. 
<br />
<br />
<code>n_total</code> and/or <code>pvalue</code> can be missing in the input datasets. In particular, if &quot;z&quot; method is set to be sample size weighted, <code>n_total</code> must be assigned.
</p>


<h3>Value</h3>

<p>No value is returned. Instead, results are written to <code>outfile</code> (named as &quot;project_traits_method.csv&quot;) in the current working directory. In the <code>outfile</code>, there are some new variables (listed below) created by the package along with the existing variables in the original datasets. 
</p>
<table role = "presentation">
<tr><td><code>zi</code></td>
<td>
<p>Z statistic for the ith phenotype in traitlist. They will appear in &quot;z&quot;, &quot;chisq&quot; and &quot;sumsq&quot; methods.</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>p-value for the ith phenotype in traitlist.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>combined statistic of &quot;beta&quot; methods.</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>standard error of the combined statistic of &quot;beta&quot; methods.</p>
</td></tr>
<tr><td><code>Z.comb</code></td>
<td>
<p>Z statistic (Z.comb=beta/SE) of &quot;beta&quot; and &quot;z&quot; methods.</p>
</td></tr>
<tr><td><code>betai</code></td>
<td>
<p>beta statistic for the ith phenotype in traitlist. They will appear in output for &quot;beta&quot; method.</p>
</td></tr>
<tr><td><code>chisq.comb</code></td>
<td>
<p>combined and test statistic of &quot;chisq&quot; and &quot;sumsq&quot; methods.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the combined statistic.</p>
</td></tr>
<tr><td><code>meanN</code></td>
<td>
<p>the mean sample size with phenotype and genotype for the genetic marker. N/A if <code>n_total</code> is not specified.</p>
</td></tr>
<tr><td><code>minN</code></td>
<td>
<p>the minimum sample size with phenotype and genotype for the genetic marker. N/A if <code>n_total</code> is not specified.</p>
</td></tr>
<tr><td><code>maxN</code></td>
<td>
<p>the maximum sample size with phenotype and genotype for the genetic marker. N/A if <code>n_total</code> is not specified.</p>
</td></tr>
<tr><td><code>remark1</code></td>
<td>
<p>The sign of beta will be flipped if coded alleles different between two datasets.</p>
</td></tr>
<tr><td><code>remark2</code></td>
<td>
<p>If the minimum eigen value of the covariance matrix is less than 0.01, we consider it as nearly singular and the analyses will stop.</p>
</td></tr>
<tr><td><code>remark3</code></td>
<td>
<p>The alleles are supposed called on positive strand. If not, the user should convert
the coded allele to that on a positive strand in the result file.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Shuo Li &lt;skyli@bu.edu&gt;, Xuan Liu &lt;liuxuan@bu.edu&gt; and Qiong Yang &lt;qyang@bu.edu&gt; 
</p>


<h3>References</h3>

<p>CUMP: an R package for analyzing multivariate phenotypes in genetic association studies 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##The following are two fake examples. Do NOT run. 
##Please refer to example.pdf for details.
##no change of beta signs before combining
##combGWAS(project="mv",traitlist=c("phen1","phne2"),
## traitfile=c("Phen1GWAS.csv", "Phen2GWAS.csv"), comb_method=c("z","chisq"), 
## betasign=c(1,1), snpid="SNPID", beta="beta", SE="SE", 
## coded_all="coded_all"", AF_coded_all=" AF_coded_all ", pvalue="pval") 

##change of  beta signs before combining: the beta sign for the 2nd phenotype reversed
##combGWAS(project="mv",traitlist=c("phen1","phne2"),
## traitfile=c("Phen1GWAS.csv", "Phen2GWAS.csv"), comb_method=c("z","chisq"),
## betasign=c(1,-1), snpid="SNPID", beta="beta", SE="SE", 
## coded_all="coded_all ", AF_coded_all=" AF_coded_all ", pvalue="pval") 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
