<!DOCTYPE html><html><head><title>Help for package cumSeg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cumSeg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cumSeg-package'>
<p>Change point detection and estimation in genomic sequences</p></a></li>
<li><a href='#fibroblast'>
<p>Fibroblast Cell Line dataset</p></a></li>
<li><a href='#fit.control'>
<p>Auxiliary function for controlling model fitting</p></a></li>
<li><a href='#jumpoints'>
<p>Change point estimation in piecewise constant models</p></a></li>
<li><a href='#plot.aCGHsegmented'>
<p>Plot method for the class 'aCGHsegmented'</p></a></li>
<li><a href='#print.aCGHsegmented'><p> Print method for the aCGHsegmented class</p></a></li>
<li><a href='#sel.control'>
<p>Auxiliary function for controlling model selection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Change Point Detection in Genomic Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-07-18</td>
</tr>
<tr>
<td>Author:</td>
<td>Vito M.R. Muggeo</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vito M.R. Muggeo &lt;vito.muggeo@unipa.it&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimation of number and location of change points in
        mean-shift (piecewise constant) models. Particularly useful (but not confined) to
        model genomic sequences of continuous measurements.</td>
</tr>
<tr>
<td>Depends:</td>
<td>lars</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-07-17 08:45:44 UTC; vito</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-07-17 09:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cumSeg-package'>
Change point detection and estimation in genomic sequences
</h2><span id='topic+cumSeg-package'></span><span id='topic+cumSeg'></span>

<h3>Description</h3>

<p>Estimation of number and location of change points in &lsquo;mean-shift&rsquo;
(&lsquo;piecewise constant&rsquo; or &lsquo;step-function&rsquo;) models. 
Particularly useful (but not confined) to model genomic sequences of continuous measurements.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> cumSeg</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-07-18</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Package <code>cumSeg</code> estimates the number and location of change points in &lsquo;mean-shift&rsquo;
(also said &lsquo;piecewise constant&rsquo; or &lsquo;step-function&rsquo;) models. These models are particularly useful in Biology, Medicine, or Genomics, where 
it is of interest to know the location of changes in some genomic sequences (e.g. in array comparative genomic hybridization analysis).
The algorithm works by first estimating an high number of change points (via the efficient &lsquo;segmented&rsquo; algorithm of Muggeo (2003)) 
and then by applying the <em>lars</em> algorithm of Efron et al. (2004) to select some of them via a generalized BIC criterion. 
The procedure appears to be (somewhat) robust to some forms of model mis-specifications and, 
from a computational standpoint, it is substantially independent of the number of change points to be estimated.
</p>


<h3>Author(s)</h3>

<p>Vito M.R. Muggeo <a href="mailto:vito.muggeo@unipa.it">vito.muggeo@unipa.it</a>
</p>


<h3>References</h3>

<p>Muggeo, V.M.R., Adelfio, G., Efficient change point detection for genomic sequences 
of continuous measurements, <em>Bioinformatics</em> <b>27</b>, 161-166.
</p>
<p>Efron, B., Hastie, T., Johnstone, I., Tibshirani, R. (2004) Least angle regression, 
<em>Annals of Statistics</em> <b>32</b>, 407-489.
</p>
<p>Muggeo, V.M.R. (2003) Estimating regression models with unknown break-points.
<em>Statistics in Medicine</em> <b>22</b>, 3055-3071.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(cumSeg)
data(fibroblast)
#select chromosomes 1.. but the same for chromosomes 3,9,11
z&lt;-na.omit(fibroblast$gm03563[fibroblast$Chromosome==1]) 
o&lt;-jumpoints(z,k=30,output="3")
plot(z)
plot(o,add=TRUE,y=FALSE,col=4)

## End(Not run)
</code></pre>

<hr>
<h2 id='fibroblast'>
Fibroblast Cell Line dataset
</h2><span id='topic+fibroblast'></span>

<h3>Description</h3>

<p>Genomic sequences of 15 fibroblast cell lines. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fibroblast)</code></pre>


<h3>Format</h3>

<p>A data frame with 2462 observations on the following 11 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>a numeric vector to identify the chromosome</p>
</dd>
<dt><code>Genome.Order</code></dt><dd><p>a numeric vector meaning the genome index</p>
</dd>
<dt><code>gm05296</code></dt><dd><p>cell line GM05296</p>
</dd>
<dt><code>gm03563</code></dt><dd><p>cell line GM03563</p>
</dd>
<dt><code>gm01535</code></dt><dd><p>cell line GM01535</p>
</dd>
<dt><code>gm07081</code></dt><dd><p>cell line GM07081</p>
</dd>
<dt><code>gm01750</code></dt><dd><p>cell line GM01750</p>
</dd>
<dt><code>gm03134</code></dt><dd><p>cell line GM03134</p>
</dd>
<dt><code>gm13330</code></dt><dd><p>cell line GM13330</p>
</dd>
<dt><code>gm13031</code></dt><dd><p>cell line GM13031</p>
</dd>
<dt><code>gm01524</code></dt><dd><p>cell line GM01524</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data come from a single experiments on 15 fibroblast cell lines with each array
containing over 2000 (mapped) BACs spotted in triplicate. The variable in the 
dataset is the normalized average of the log base 2 test over reference ratio.</p>


<h3>References</h3>

<p>Snijders, A. M., Nowak, N., Segraves, R., Blackwood, S., Brown, N., Conroy, J., Hamilton, G.,
Hindle, A. K., Huey, B., Kimura, K. et al., (2001). Assembly of microarrays for genome-wide measurement
of DNA copy number. Nature Genetics 29, 263-264.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(fibroblast)
#select chromosome 1
z&lt;-na.omit(fibroblast$gm03563[fibroblast$Chromosome==1]) 
o&lt;-jumpoints(z,k=30,output="3")
plot(z)
plot(o,add=TRUE,y=FALSE,col=4)

## End(Not run)
</code></pre>

<hr>
<h2 id='fit.control'>
Auxiliary function for controlling model fitting
</h2><span id='topic+fit.control'></span>

<h3>Description</h3>

<p>Auxiliary function as user interface for model fitting. Typically
only used when calling 'jumpoints'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit.control(toll = 0.001, it.max = 5, display = FALSE, last = TRUE, 
    maxit.glm = 25, h = 1, stop.if.error = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit.control_+3A_toll">toll</code></td>
<td>

<p>positive convergence tolerance.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_it.max">it.max</code></td>
<td>

<p>integer giving the maximal number of iterations.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_display">display</code></td>
<td>

<p>logical indicating if the value of the objective function should be printed at each iteration.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_last">last</code></td>
<td>

<p>Currently ignored.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_maxit.glm">maxit.glm</code></td>
<td>

<p>Currently ignored.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_h">h</code></td>
<td>

<p>Currently ignored.
</p>
</td></tr>
<tr><td><code id="fit.control_+3A_stop.if.error">stop.if.error</code></td>
<td>

<p>logical indicating if the algorithm should stop when one or more estimated changepoints 
do not assume admissible values. Default is <code>FALSE</code> which implies automatic 
changepoint selection.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the arguments as components to be used by 'jumpoints'.
</p>


<h3>Author(s)</h3>

<p> Vito M. R. Muggeo </p>


<h3>See Also</h3>

<p><code><a href="#topic+jumpoints">jumpoints</a></code></p>

<hr>
<h2 id='jumpoints'>
Change point estimation in piecewise constant models
</h2><span id='topic+jumpoints'></span>

<h3>Description</h3>

<p>Estimation of change points and model selection via generalized BIC and other criteria
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jumpoints(y, x, k = min(30, round(length(y)/10)), output = "2", 
    psi = NULL, round = TRUE, control = fit.control(), 
    selection = sel.control(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jumpoints_+3A_y">y</code></td>
<td>

<p>the observed (genomic) sequence supposed to have a piecewise constant mean function.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_x">x</code></td>
<td>

<p>the &lsquo;segmented&rsquo; variable, e.g. the genomic location. If missing simple indices 1,2,... n (length of <code>y</code>) are assumed.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_k">k</code></td>
<td>

<p>the starting number of changepoints. It should be quite larger than the supposed number of 
(true) changepoints. This argument is ignored if starting values of the changepoints are specified via
<code>psi</code>.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_output">output</code></td>
<td>

<p>which output should be produced? Possible values are <code>"1"</code>, <code>"2"</code>, or <code>"3"</code>; see Details
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_psi">psi</code></td>
<td>

<p>numeric vector to indicate the starting values for the changepoints. When <code>psi=NULL</code> (default), <code>k</code> 
quantiles are assumed.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_round">round</code></td>
<td>

<p>logical; should the values of the changepoints be rounded?
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_control">control</code></td>
<td>

<p>a list returned by <code>fit.control</code>.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_selection">selection</code></td>
<td>

<p>a list returned by <code>sel.control</code>.
</p>
</td></tr>
<tr><td><code id="jumpoints_+3A_...">...</code></td>
<td>

<p>additional arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algorithm works by suitably transforming the observed responses to fit a continuous piecewise linear model.
Starting from <code>k</code> changepoints, a large number of changepoints is first estimated. This number will be (typically slightly) lower than <code>k</code> since some changepoints will be discarded during the iterative steps when taking non admissible values. If <code>output="1"</code>, <code>jumpoints</code> returns them which typically will be more than the actual ones. If <code>output="2"</code> the appropriate number of changepoints is 
selected via the criterion specified in argument <code>selection</code> via <code><a href="#topic+sel.control">sel.control</a></code> (e.g. BIC, MDL, ..). 
Finally if <code>output="3"</code>, the segmented algorithm is run again to try to improve the changepoint estimates returned by the previous step. 
</p>


<h3>Value</h3>

<p>A list including several components depending on the value of <code>output</code> <br />
</p>
<p>If <code>output="1"</code> the most relevant components are 
</p>
<table>
<tr><td><code>fitted.values</code></td>
<td>
<p>the fitted values</p>
</td></tr>
<tr><td><code>n.psi</code></td>
<td>
<p>the estimated number of changepoints</p>
</td></tr>
<tr><td><code>est.means</code></td>
<td>
<p>the estimated means</p>
</td></tr>
<tr><td><code>psi</code></td>
<td>
<p>the estimated changepoints</p>
</td></tr></table>
<p><br />
</p>
<p>If <code>output="2"</code> the most relevant components are 
</p>
<table>
<tr><td><code>fitted.values</code></td>
<td>
<p>the fitted values</p>
</td></tr>
<tr><td><code>n.psi</code></td>
<td>
<p>the estimated number of changepoints</p>
</td></tr>
<tr><td><code>criterion</code></td>
<td>
<p>the values of the selection criterion</p>
</td></tr>
<tr><td><code>psi</code></td>
<td>
<p>the estimated changepoints</p>
</td></tr>
<tr><td><code>est.means</code></td>
<td>
<p>the estimated means</p>
</td></tr>
<tr><td><code>psi0</code></td>
<td>
<p>the estimated changepoints at ouput 1 (before applying the selection criterion)</p>
</td></tr>
<tr><td><code>est.means0</code></td>
<td>
<p>the estimated means at ouput 1 (before applying the selection criterion)</p>
</td></tr></table>
<p><br />
</p>
<p>If <code>output="3"</code> the most relevant components are those of output 2 but
</p>
<table>
<tr><td><code>psi0</code></td>
<td>
<p>the estimated changepoints at ouput 1</p>
</td></tr>
<tr><td><code>psi1</code></td>
<td>
<p>the estimated changepoints at ouput 2</p>
</td></tr>
<tr><td><code>psi</code></td>
<td>
<p>the estimated changepoints at ouput 3 (after applying again the segmented algorithm).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Vito Muggeo
</p>


<h3>References</h3>

<p>Muggeo, V.M.R., Adelfio, G., Efficient change point detection for genomic sequences 
of continuous measurements, <em>Bioinformatics</em> <b>27</b>, 161-166.
</p>


<h3>See Also</h3>

<p><code><a href="lars.html#topic+lars">lars</a></code>, <code><a href="#topic+sel.control">sel.control</a></code>, <code><a href="#topic+fit.control">fit.control</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
n&lt;-100
x&lt;-1:n/n
 
lp&lt;-I(x&gt;.1) -1*I(x&gt;.15)+.585*I(x&gt;.45)-.585*I(x&gt;.6) -I(x&gt;.9)
e&lt;-rnorm(n,0,.154)
y&lt;-lp+e #data

#fit the model without selecting the changepoints
o1&lt;-jumpoints(y,output="1")
plot(o1, typeL="l")
lines(lp, col=2) #true regression function
legend("topright", c("true","fit with output=1"),bty="n", col=c(2,1), lty=1)

#fit model and select the changepoints
o2&lt;-jumpoints(y,output="2")
par(mfrow=c(1,2))
plot(o2, what="c")
plot(o2, typeL="s")
lines(lp, col=3) #true regression function
legend("topright", c("true","fit with output=2"),bty="n", col=c(3,1), lty=1)

## End(Not run)
    </code></pre>

<hr>
<h2 id='plot.aCGHsegmented'>
Plot method for the class 'aCGHsegmented'
</h2><span id='topic+plot.aCGHsegmented'></span>

<h3>Description</h3>

<p>Plots fitted piecewise constant lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'aCGHsegmented'
plot(x, add = FALSE, y = TRUE, psi.lines = TRUE, typeL="l", 
    what=c("lines","criterion"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.aCGHsegmented_+3A_x">x</code></td>
<td>

<p>object of class <code>"aCGHsegmented"</code> returned by <code>jumpoints</code>.
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_add">add</code></td>
<td>

<p>logical; if <code>TRUE</code> the fitted piecewise constant lines are added 
to an existing plot.
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_y">y</code></td>
<td>

<p>logical; if <code>TRUE</code> the observations are also plotted, otherwise only the fitted lines.
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_psi.lines">psi.lines</code></td>
<td>

<p>logical; if <code>TRUE</code> vertical lines corresponding to the estimated changepoints are added.
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_typel">typeL</code></td>
<td>

<p>argument <code>type</code> for the fitted lines. Possible options are <code style="white-space: pre;">&#8288;typeL="s"&#8288;</code> to plot the horizontal and vertical lines of the step-function, and 
<code style="white-space: pre;">&#8288;typeL="l"&#8288;</code> to draw the horizontal lines only.  
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_what">what</code></td>
<td>

<p>If <code>'lines'</code> the fitted lines are plotted, otherwise the criterion values versus the number of change points, provided the fitted object <code>x</code> has been 
called with argument <code>output='2'</code> or <code>output='3'</code>.  
</p>
</td></tr>
<tr><td><code id="plot.aCGHsegmented_+3A_...">...</code></td>
<td>

<p>possible additional graphical arguments, such as <code>col</code>, <code>xlab</code>, and so on.  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This fuction takes a fitted object returned by <code>jumpoints</code> and plots 
the resulting fit, namely the estimated step-function and changepoints.
</p>


<h3>Value</h3>

<p>The function simply plots the fit returned by 'jumpoints'.
</p>


<h3>Author(s)</h3>

<p>Vito Muggeo
</p>


<h3>See Also</h3>

<p><code><a href="#topic+jumpoints">jumpoints</a></code>
</p>

<hr>
<h2 id='print.aCGHsegmented'> Print method for the aCGHsegmented class </h2><span id='topic+print.aCGHsegmented'></span>

<h3>Description</h3>

<p>Printing the most important feautures of a model returned by <code>jumpoints</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'aCGHsegmented'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.aCGHsegmented_+3A_x">x</code></td>
<td>
<p> object of class <code>aCGHsegmented</code> </p>
</td></tr>
<tr><td><code id="print.aCGHsegmented_+3A_digits">digits</code></td>
<td>
<p> number of digits to be printed </p>
</td></tr>
<tr><td><code id="print.aCGHsegmented_+3A_...">...</code></td>
<td>
<p> arguments passed to other functions </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Vito M.R. Muggeo </p>


<h3>See Also</h3>

<p><code><a href="#topic+jumpoints">jumpoints</a></code>, <code><a href="#topic+plot.aCGHsegmented">plot.aCGHsegmented</a></code></p>

<hr>
<h2 id='sel.control'>
Auxiliary function for controlling model selection
</h2><span id='topic+sel.control'></span>

<h3>Description</h3>

<p>Auxiliary function as user interface for model selection. Typically
only used when calling 'jumpoints'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel.control(display = FALSE, type = c("bic", "mdl", "rss"), S = 1, 
    Cn = "log(log(n))", alg = c("stepwise", "lasso"), edf.psi = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sel.control_+3A_display">display</code></td>
<td>

<p>logical to be passed to the argument <code>trace</code> of <code>lars</code>
</p>
</td></tr>
<tr><td><code id="sel.control_+3A_type">type</code></td>
<td>

<p>the criterion to be used to perform model selection. 
</p>
</td></tr>
<tr><td><code id="sel.control_+3A_s">S</code></td>
<td>

<p>if <code>type="rss"</code> the optimal model is selected when the residual sum of squares decreases 
by the threshold <code>S</code>.
</p>
</td></tr>
<tr><td><code id="sel.control_+3A_cn">Cn</code></td>
<td>

<p>if <code>type="bic"</code> a character string (as a function of <code>'n'</code>) to specify to generalized BIC. 
If <code>Cn=1</code> the standard BIC is used.
</p>
</td></tr>
<tr><td><code id="sel.control_+3A_alg">alg</code></td>
<td>

<p>which procedure should be used to perform model selection? The value of <code>alg</code> is 
passed to the argument <code>'type'</code> of <code>lars</code>.
</p>
</td></tr>
<tr><td><code id="sel.control_+3A_edf.psi">edf.psi</code></td>
<td>

<p>logical indicating if the number of changepoints should be computed in the model df.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function specifies how to perform model seletion, namely how many change points should be selected. 
</p>


<h3>Value</h3>

<p>A list with the arguments as components to be used by <code>jumpoints</code> and in turn by <code>lars</code>.
</p>


<h3>Author(s)</h3>

<p> Vito Muggeo </p>


<h3>See Also</h3>

<p><code><a href="#topic+jumpoints">jumpoints</a></code>, <code><a href="lars.html#topic+lars">lars</a></code></p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
