<!DOCTYPE html><html><head><title>Help for package cuperdec</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cuperdec}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adaptive_burnin_filter'><p>Calculate adaptive burn-in retain/discard list</p></a></li>
<li><a href='#calculate_curve'><p>Calculate cumulative decay percent curve</p></a></li>
<li><a href='#cuperdec_database_ex'><p>Example isolation source database input for cuperdec</p></a></li>
<li><a href='#cuperdec_metadata_ex'><p>Example metadata file input for cuperdec</p></a></li>
<li><a href='#cuperdec_taxatable_ex'><p>Example taxon table input for cuperdec</p></a></li>
<li><a href='#hard_burnin_filter'><p>Calculate hard burn-in retain/discard list</p></a></li>
<li><a href='#load_database'><p>Load database</p></a></li>
<li><a href='#load_map'><p>Load metadata table</p></a></li>
<li><a href='#load_taxa_table'><p>Load OTU table</p></a></li>
<li><a href='#plot_cuperdec'><p>Plot cumulative percent decay curves</p></a></li>
<li><a href='#simple_filter'><p>Apply simple percentage filter</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Cumulative Percent Decay Curve Generator</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculates and visualises cumulative percent 'decay' curves,
    which are typically calculated from metagenomic taxonomic profiles.
    These can be used to estimate the level of expected 'endogenous' taxa
    at different abundance levels retrieved from metagenomic samples, when
    comparing to samples of known sampling site or source. Method
    described in Fellows Yates, J. A. et. al. (2021) Proceedings of the
    National Academy of Sciences USA &lt;<a href="https://doi.org/10.1073%2Fpnas.2021655118">doi:10.1073/pnas.2021655118</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jfy133/cuperdec">https://github.com/jfy133/cuperdec</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jfy133/cuperdec/issues">https://github.com/jfy133/cuperdec/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, magrittr, readr, rlang, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, tibble</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-12 18:12:11 UTC; jfellows</td>
</tr>
<tr>
<td>Author:</td>
<td>James A. Fellows Yates
    <a href="https://orcid.org/0000-0001-5585-6277"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James A. Fellows Yates &lt;jfy133@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-12 21:40:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='adaptive_burnin_filter'>Calculate adaptive burn-in retain/discard list</h2><span id='topic+adaptive_burnin_filter'></span>

<h3>Description</h3>

<p>Automates a selection of a per-sample 'burn in' based on the
nature of the sample's curve itself (rather than supplying a hard value) by
finding the point from which the 'fluctuation' of the curve doesn't exceed
the mean +- SD of the total curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adaptive_burnin_filter(curves, percent_threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adaptive_burnin_filter_+3A_curves">curves</code></td>
<td>
<p>A cuperdec curve table calculated with
<code><a href="#topic+calculate_curve">calculate_curve</a></code>.</p>
</td></tr>
<tr><td><code id="adaptive_burnin_filter_+3A_percent_threshold">percent_threshold</code></td>
<td>
<p>A percentage of the target-source in a sample above
which a sample is considered 'retained'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with each row showing each sample and whether it
passed the specified filter.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
data(cuperdec_database_ex)

taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")

curve_results &lt;- calculate_curve(taxa_table, iso_database)
adaptive_burnin_filter(curve_results, percent_threshold = 0.1)
</code></pre>

<hr>
<h2 id='calculate_curve'>Calculate cumulative decay percent curve</h2><span id='topic+calculate_curve'></span>

<h3>Description</h3>

<p>Performs the initial decay curve based on percentage of 'target' isolation
source along a rank of most to least abundant taxa for a given sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_curve(taxa_table, database)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_curve_+3A_taxa_table">taxa_table</code></td>
<td>
<p>An OTU table loaded with <code><a href="#topic+load_taxa_table">load_taxa_table</a></code>.</p>
</td></tr>
<tr><td><code id="calculate_curve_+3A_database">database</code></td>
<td>
<p>A database file loaded with <code><a href="#topic+load_database">load_database</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object in the form of a tibble with taxa of each given sample
ordered by rank and the proportion of taxa up to that rank deriving
from your target source.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
data(cuperdec_database_ex)

taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")

calculate_curve(taxa_table, iso_database)
</code></pre>

<hr>
<h2 id='cuperdec_database_ex'>Example isolation source database input for cuperdec</h2><span id='topic+cuperdec_database_ex'></span>

<h3>Description</h3>

<p>Example isolation source database used for input to
cuperdec based. Species names are from a NCBI Nt database
and isolation sources gather from the Human Oral Microbiome
database, NCBI GenBank, and manual curation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cuperdec_database_ex)
</code></pre>


<h3>Format</h3>

<p>An TSV table loaded as a <code>tibble</code>.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.5281/zenodo.3740492">10.5281/zenodo.3740492</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cuperdec_database_ex)
load_database(cuperdec_database_ex, target = "oral")
</code></pre>

<hr>
<h2 id='cuperdec_metadata_ex'>Example metadata file input for cuperdec</h2><span id='topic+cuperdec_metadata_ex'></span>

<h3>Description</h3>

<p>Example metadata map file corresponding to samples
in example data &quot;cuperdec_taxatable_ex&quot;. Includes
a grouping column corresponding to sample species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cuperdec_metadata_ex)
</code></pre>


<h3>Format</h3>

<p>An TSV table loaded as a <code>tibble</code>.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.5281/zenodo.3740492">10.5281/zenodo.3740492</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cuperdec_metadata_ex)
load_map(cuperdec_metadata_ex, sample_col = "#SampleID", source_col = "Env")
</code></pre>

<hr>
<h2 id='cuperdec_taxatable_ex'>Example taxon table input for cuperdec</h2><span id='topic+cuperdec_taxatable_ex'></span>

<h3>Description</h3>

<p>Example taxon table used for input to cuperdec based
on data including shotgun-sequenced ancient calculus
samples aligned against the NCBI Nt database from Oct
2017 using MALT. Samples are columns, rows are taxa
and counts are assigned reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
</code></pre>


<h3>Format</h3>

<p>An TSV table loaded as a <code>tibble</code>.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.5281/zenodo.3740492">10.5281/zenodo.3740492</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cuperdec_taxatable_ex)
load_taxa_table(cuperdec_taxatable_ex)
</code></pre>

<hr>
<h2 id='hard_burnin_filter'>Calculate hard burn-in retain/discard list</h2><span id='topic+hard_burnin_filter'></span>

<h3>Description</h3>

<p>Returns a table of whether each sample passes a given threshold, after
considering a 'burn-in', in the form of a fraction of the abundance ranks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hard_burnin_filter(curves, percent_threshold, rank_burnin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hard_burnin_filter_+3A_curves">curves</code></td>
<td>
<p>A cuperdec curve table calculated with
<code><a href="#topic+calculate_curve">calculate_curve</a></code>.</p>
</td></tr>
<tr><td><code id="hard_burnin_filter_+3A_percent_threshold">percent_threshold</code></td>
<td>
<p>A percentage of the target-source in a sample above
which a sample is considered 'retained'.</p>
</td></tr>
<tr><td><code id="hard_burnin_filter_+3A_rank_burnin">rank_burnin</code></td>
<td>
<p>A number between 0 and 1 indicating the fraction of taxa
to ignore before applying the threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with each row showing each sample and whether it
passed the specified filter.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
data(cuperdec_database_ex)

taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")

curve_results &lt;- calculate_curve(taxa_table, iso_database)
hard_burnin_filter(curve_results, percent_threshold = 50, rank_burnin = 0.1)
</code></pre>

<hr>
<h2 id='load_database'>Load database</h2><span id='topic+load_database'></span>

<h3>Description</h3>

<p>Loads a taxon/isolation source database file, i.e. first column is
a list of taxa, and the second column is a list of isolation sources, and
formats for downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_database(x, target)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_database_+3A_x">x</code></td>
<td>
<p>Path to a (minimum) two column TSV file or tidy dataframe (e.g.
tibble), one column with taxon names and other indicating if from
target isolation source.</p>
</td></tr>
<tr><td><code id="load_database_+3A_target">target</code></td>
<td>
<p>the string in the 'Isolation Source' (i.e. 2nd) column which
is the expected target source of the samples</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Taxon names should match that with the taxa table.
</p>


<h3>Value</h3>

<p>A tibble, formatted for use in downstream cuperdec functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_database_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")
</code></pre>

<hr>
<h2 id='load_map'>Load metadata table</h2><span id='topic+load_map'></span>

<h3>Description</h3>

<p>Loads a metadata table and reformats it for downstream analysis. This needs
to include at minimum two columns: sample name, and sample source.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_map(x, sample_col, source_col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_map_+3A_x">x</code></td>
<td>
<p>Path to a TSV file or tidy dataframe (e.g. tibble) with a column
containing sample names and other grouping metadata columns.</p>
</td></tr>
<tr><td><code id="load_map_+3A_sample_col">sample_col</code></td>
<td>
<p>A column name specifying which column should be used to
specify sample names.</p>
</td></tr>
<tr><td><code id="load_map_+3A_source_col">source_col</code></td>
<td>
<p>A column name specifying which group or the source the
sample is from.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The two columns required need to include the following information:
</p>

<ul>
<li><p>Sample name - a unique identifier for each sample
</p>
</li>
<li><p>Sample source - a grouping ID indicating what 'source' the sample is
from This is used for plotting to separate comparative 'sources' to your
own samples.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble, formatted for use in downstream cuperdec functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_metadata_ex)
metadata_table &lt;- load_map(cuperdec_metadata_ex,
  sample_col = "#SampleID",
  source_col = "Env"
)
</code></pre>

<hr>
<h2 id='load_taxa_table'>Load OTU table</h2><span id='topic+load_taxa_table'></span>

<h3>Description</h3>

<p>Loads a typical taxa table (Samples: columns; Taxa: rows) in TSV format
and standardises some columns, storing the table in the form of a tibble.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_taxa_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_taxa_table_+3A_x">x</code></td>
<td>
<p>Path to a TSV file or tidy dataframe (e.g. tibble) consisting of an
OTU table of samples as columns, except first column with taxon names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble, formatted for use in downstream cuperdec functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
</code></pre>

<hr>
<h2 id='plot_cuperdec'>Plot cumulative percent decay curves</h2><span id='topic+plot_cuperdec'></span>

<h3>Description</h3>

<p>Generates visual representation of curves, with optional separate plotting
of different groups, and also indication of individuals passing different
on types filters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cuperdec(
  curves,
  metadata,
  burnin_result,
  restrict_x = 0,
  facet_cols = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_cuperdec_+3A_curves">curves</code></td>
<td>
<p>Output tibble from <code><a href="#topic+calculate_curve">calculate_curve</a></code>.</p>
</td></tr>
<tr><td><code id="plot_cuperdec_+3A_metadata">metadata</code></td>
<td>
<p>Output from <code><a href="#topic+load_map">load_map</a></code>.</p>
</td></tr>
<tr><td><code id="plot_cuperdec_+3A_burnin_result">burnin_result</code></td>
<td>
<p>Output from <code>apply_*_burnin</code>.
functions (optional).</p>
</td></tr>
<tr><td><code id="plot_cuperdec_+3A_restrict_x">restrict_x</code></td>
<td>
<p>Restrict viewing of abundance rank to X number
of ranks (useful for closer inspection of curves) (optional).</p>
</td></tr>
<tr><td><code id="plot_cuperdec_+3A_facet_cols">facet_cols</code></td>
<td>
<p>Custom number of columns for faceted plots (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 image object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
data(cuperdec_database_ex)
data(cuperdec_metadata_ex)

taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")
metadata_table &lt;- load_map(cuperdec_metadata_ex,
  sample_col = "#SampleID",
  source_col = "Env"
)

curves &lt;- calculate_curve(taxa_table, iso_database)
burnin_results &lt;- adaptive_burnin_filter(curves, percent_threshold = 0.1)

plot_cuperdec(curves, metadata_table, burnin_results)
</code></pre>

<hr>
<h2 id='simple_filter'>Apply simple percentage filter</h2><span id='topic+simple_filter'></span>

<h3>Description</h3>

<p>Performs the initial decay curve based on percentage of 'target' isolation
source along a rank of most to least abundant taxa for a given sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simple_filter(curves, percent_threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simple_filter_+3A_curves">curves</code></td>
<td>
<p>A cuperdec curve table calculated with
<code><a href="#topic+calculate_curve">calculate_curve</a></code>.</p>
</td></tr>
<tr><td><code id="simple_filter_+3A_percent_threshold">percent_threshold</code></td>
<td>
<p>A database file loaded with
<code><a href="#topic+load_database">load_database</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with each row showing each sample and whether it
passed the specified filter.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cuperdec_taxatable_ex)
data(cuperdec_database_ex)

taxa_table &lt;- load_taxa_table(cuperdec_taxatable_ex)
iso_database &lt;- load_database(cuperdec_database_ex, target = "oral")

curve_results &lt;- calculate_curve(taxa_table, iso_database)
simple_filter(curve_results, percent_threshold = 50)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
