<!DOCTYPE html><html><head><title>Help for package cwot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {cwot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#awot'><p>Adaptively weighted joint test for main effect and genotype-by-treatment interaction effect for continuous endpoints.</p></a></li>
<li><a href='#awot_null'><p>Prepare null model for awot.</p></a></li>
<li><a href='#cwot'><p>Cauchy weighted joint test for main effect and genotype-by-treatment interaction effect for binary endpoints.</p></a></li>
<li><a href='#cwot_null'><p>Prepare null model for cwot.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Cauchy Weighted Joint Test for Pharmacogenetics Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A flexible and robust joint test of the single nucleotide polymorphism (SNP) main effect and genotype-by-treatment interaction effect for continuous and binary endpoints. Two analytic procedures, Cauchy weighted joint test (CWOT) and adaptively weighted joint test (AWOT), are proposed to accurately calculate the joint test p-value. The proposed methods are evaluated through extensive simulations under various scenarios. The results show that the proposed AWOT and CWOT control type I error well and outperform existing methods in detecting the most interesting signal patterns in pharmacogenetics (PGx) association studies. For reference, see Hong Zhang, Devan Mehrotra and Judong Shen (2022) &lt;<a href="https://doi.org/10.13140%2FRG.2.2.28323.53280">doi:10.13140/RG.2.2.28323.53280</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, SPAtest, mvtnorm</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-22 18:54:08 UTC; consi</td>
</tr>
<tr>
<td>Author:</td>
<td>Hong Zhang <a href="https://orcid.org/0000-0002-8869-8671"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hong Zhang &lt;hzhang@wpi.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-23 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='awot'>Adaptively weighted joint test for main effect and genotype-by-treatment interaction effect for continuous endpoints.</h2><span id='topic+awot'></span>

<h3>Description</h3>

<p>Adaptively weighted joint test for main effect and genotype-by-treatment interaction effect for continuous endpoints.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>awot(nullmod, g, weights = seq(-1, 1, 0.1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="awot_+3A_nullmod">nullmod</code></td>
<td>
<p>- The null model object from the output of awot_null.</p>
</td></tr>
<tr><td><code id="awot_+3A_g">g</code></td>
<td>
<p>- The variable of interest, e.g., the genotype.</p>
</td></tr>
<tr><td><code id="awot_+3A_weights">weights</code></td>
<td>
<p>- The pre-specified weights. The default choice is a vector of -1, -0.9,..., 0.9, 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The p-value of AWOT and the individual p-values of the composite genotypes.
</p>


<h3>References</h3>

<p>Hong Zhang, Qing Li, Devan Mehrotra and Judong Shen. &quot;CauchyCP: a powerful test under non-proportional hazards using Cauchy combination of change-point Cox regressions&quot;, arXiv:2101.00059.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 100
y = rbinom(n, 1, 0.3)
x = data.frame(x1=rnorm(n))
tr = rbinom(n, 1, 0.5)
g = rbinom(n, 2, 0.1)
nullmod = awot_null(y, tr, x)
awot(nullmod, g, weights=seq(-1,1,0.1))
</code></pre>

<hr>
<h2 id='awot_null'>Prepare null model for awot.</h2><span id='topic+awot_null'></span>

<h3>Description</h3>

<p>Prepare null model for awot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>awot_null(y, tr, x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="awot_null_+3A_y">y</code></td>
<td>
<p>- Continuous response variable.</p>
</td></tr>
<tr><td><code id="awot_null_+3A_tr">tr</code></td>
<td>
<p>- Binary treatment variable, 1 for treated, 0 for placebo.</p>
</td></tr>
<tr><td><code id="awot_null_+3A_x">x</code></td>
<td>
<p>- Covariates in addition to treatment variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of objects needed for awot.
</p>


<h3>References</h3>

<p>Hong Zhang, Devan Mehrotra and Judong Shen, &quot;AWOT and CWOT for Genotype and Genotype by Treatment Interaction Joint Analysis in Pharmacogenetics GWAS&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 100
y = rnorm(n)
x = data.frame(x1=rnorm(n))
tr = rbinom(n, 1, 0.5)
nullmod = awot_null(y, tr, x)
</code></pre>

<hr>
<h2 id='cwot'>Cauchy weighted joint test for main effect and genotype-by-treatment interaction effect for binary endpoints.</h2><span id='topic+cwot'></span>

<h3>Description</h3>

<p>Cauchy weighted joint test for main effect and genotype-by-treatment interaction effect for binary endpoints.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cwot(nullmod, g, weights = seq(-1, 1, 0.1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cwot_+3A_nullmod">nullmod</code></td>
<td>
<p>- The null model object from the output of cwot_null.</p>
</td></tr>
<tr><td><code id="cwot_+3A_g">g</code></td>
<td>
<p>- The variable of interest, e.g., the genotype.</p>
</td></tr>
<tr><td><code id="cwot_+3A_weights">weights</code></td>
<td>
<p>- The pre-specified weights. The default choice is a vector of -1, -0.9,..., 0.9, 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The p-values of CWOT, CWOT_Score, CWOT_LRT and the individual p-values of the composite genotypes.
</p>


<h3>References</h3>

<p>Hong Zhang, Qing Li, Devan Mehrotra and Judong Shen. &quot;CauchyCP: a powerful test under non-proportional hazards using Cauchy combination of change-point Cox regressions&quot;, arXiv:2101.00059.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 100
y = rbinom(n, 1, 0.3)
x = data.frame(x1=rnorm(n))
tr = rbinom(n, 1, 0.5)
g = rbinom(n, 2, 0.1)
nullmod = cwot_null(y, tr, x)
cwot(nullmod, g, weights=seq(-1,1,0.1))
</code></pre>

<hr>
<h2 id='cwot_null'>Prepare null model for cwot.</h2><span id='topic+cwot_null'></span>

<h3>Description</h3>

<p>Prepare null model for cwot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cwot_null(y, tr, x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cwot_null_+3A_y">y</code></td>
<td>
<p>- Binary response variable.</p>
</td></tr>
<tr><td><code id="cwot_null_+3A_tr">tr</code></td>
<td>
<p>- Binary treatment variable, 1 for treated, 0 for placebo.</p>
</td></tr>
<tr><td><code id="cwot_null_+3A_x">x</code></td>
<td>
<p>- Covariates in addition to treatment variable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of objects needed for cwot.
</p>


<h3>References</h3>

<p>Hong Zhang, Devan Mehrotra and Judong Shen, &quot;AWOT and CWOT for Genotype and Genotype by Treatment Interaction Joint Analysis in Pharmacogenetics GWAS&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 100
y = rbinom(n, 1, 0.3)
x = data.frame(x1=rnorm(n))
tr = rbinom(n, 1, 0.5)
nullmod = cwot_null(y, tr, x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
