<!DOCTYPE html><html><head><title>Help for package CytoSimplex</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {CytoSimplex}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#colNormalize'><p>Normalize each column of the input matrix by the column sum</p></a></li>
<li><a href='#plotBinary'><p>Create binary plots</p></a></li>
<li><a href='#plotQuaternary'><p>Create quaternary simplex plots</p></a></li>
<li><a href='#plotTernary'><p>Create ternary plots</p></a></li>
<li><a href='#rnaCluster'><p>Major cell type annotation of the example mouse bone marrow data</p></a></li>
<li><a href='#rnaRaw'><p>Mouse bone marrow scRNAseq example data</p></a></li>
<li><a href='#rnaVelo'><p>Velocity graph of the example mouse bone marrow data</p></a></li>
<li><a href='#selectTopFeatures'><p>Pick top differentially presented features for similarity calculation</p></a></li>
<li><a href='#show,plist-method'><p>Show plist object produced with plot3D package</p></a></li>
<li><a href='#writeQuaternaryGIF'><p>Create GIF image for dynamic rotating view of 3D quaternary simplex plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simplex Visualization of Cell Fate Similarity in Single-Cell
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yichen Wang &lt;wayichen@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Create simplex plots to visualize the similarity between 
    single-cells and selected clusters in a 1-/2-/3-simplex space.
    Velocity information can be added as an additional layer. 
    See Liu J, Wang Y et al (2023) &lt;<a href="https://doi.org/10.1101%2F2023.12.07.570655">doi:10.1101/2023.12.07.570655</a>&gt; for more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://welch-lab.github.io/CytoSimplex/">https://welch-lab.github.io/CytoSimplex/</a>,
<a href="https://github.com/welch-lab/CytoSimplex">https://github.com/welch-lab/CytoSimplex</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>stats, methods, R (&ge; 3.6)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, Matrix, plot3D, Rcpp, rlang,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, magick, patchwork, rgl, rmarkdown, Seurat,
SeuratObject, SingleCellExperiment, SummarizedExperiment,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-14 16:17:27 UTC; wangych</td>
</tr>
<tr>
<td>Author:</td>
<td>Yichen Wang <a href="https://orcid.org/0000-0003-4347-5199"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Jialin Liu <a href="https://orcid.org/0000-0002-9984-7695"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Joshua Welch [cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-15 09:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='colNormalize'>Normalize each column of the input matrix by the column sum</h2><span id='topic+colNormalize'></span><span id='topic+colNormalize.default'></span><span id='topic+colNormalize.dgCMatrix'></span><span id='topic+colNormalize.Seurat'></span><span id='topic+colNormalize.SingleCellExperiment'></span>

<h3>Description</h3>

<p>Normalize each column of the input matrix by the column sum
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colNormalize(x, scaleFactor = NULL, log = FALSE, ...)

## Default S3 method:
colNormalize(x, scaleFactor = NULL, log = FALSE, ...)

## S3 method for class 'dgCMatrix'
colNormalize(x, scaleFactor = NULL, log = FALSE, ...)

## S3 method for class 'Seurat'
colNormalize(
  x,
  scaleFactor = NULL,
  log = FALSE,
  assay = NULL,
  layer = "counts",
  ...
)

## S3 method for class 'SingleCellExperiment'
colNormalize(x, scaleFactor = NULL, log = FALSE, assay.type = "counts", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colNormalize_+3A_x">x</code></td>
<td>
<p>Feature by observation matrix. Alternatively, <code>Seurat</code> object
or <code>SingleCellExperiment</code> object with raw counts available are also
supported.</p>
</td></tr>
<tr><td><code id="colNormalize_+3A_scalefactor">scaleFactor</code></td>
<td>
<p>Multiplier on normalized data. Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="colNormalize_+3A_log">log</code></td>
<td>
<p>Logical. Whether to take log1p transformation after scaling.
Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="colNormalize_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to methods</p>
</td></tr>
<tr><td><code id="colNormalize_+3A_assay">assay</code></td>
<td>
<p>For &quot;Seurat&quot; method, the specific assay to get data from.
Default <code>NULL</code> to the default assay.</p>
</td></tr>
<tr><td><code id="colNormalize_+3A_layer">layer</code></td>
<td>
<p>For &quot;Seurat&quot; method, which layer of the assay to be used.
Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="colNormalize_+3A_assay.type">assay.type</code></td>
<td>
<p>For &quot;SingleCellExperiment&quot; method, the assay type to get
data from. Default <code>"counts"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Normalized matrix of the same size
</p>
<p>A Seurat object with normalized data in the specified slot of the
specified assay.
</p>
<p>A SingleCellExperiment object with normalized data in the specified
assay. <code>"normcounts"</code> if <code>log = FALSE</code> and <code>"logcounts"</code> if
<code>log = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rnaNorm &lt;- colNormalize(rnaRaw)

# Seurat example
library(Seurat)
srt &lt;- CreateSeuratObject(rnaRaw)
srt &lt;- colNormalize(srt)


# SingleCellExperiment example
library(SingleCellExperiment)
sce &lt;- SingleCellExperiment(assays = list(counts = rnaRaw))
sce &lt;- colNormalize(sce)

</code></pre>

<hr>
<h2 id='plotBinary'>Create binary plots</h2><span id='topic+plotBinary'></span><span id='topic+plotBinary.default'></span><span id='topic+plotBinary.Seurat'></span><span id='topic+plotBinary.SingleCellExperiment'></span><span id='topic+plotBinary.simMat'></span>

<h3>Description</h3>

<p>Create binary plots that show similarity between single cells and two
selected terminals in a barycentric coordinate. The two vertices are placed
at the left and right of a 2D plot where x-axis measures the similarity.
Y-axis is jittered for a clear view. A density (histogram) curve is added for
indicating the distribution.
</p>
<p>See <code><a href="#topic+plotTernary">plotTernary</a></code> manual for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBinary(x, ...)

## Default S3 method:
plotBinary(
  x,
  clusterVar,
  vertices,
  features = NULL,
  byCluster = NULL,
  processed = FALSE,
  method = c("euclidean", "cosine", "pearson", "spearman"),
  force = FALSE,
  sigma = 0.08,
  scale = TRUE,
  dotColor = "grey60",
  returnData = FALSE,
  ...
)

## S3 method for class 'Seurat'
plotBinary(
  x,
  layer = "counts",
  assay = NULL,
  clusterVar = NULL,
  processed = FALSE,
  ...
)

## S3 method for class 'SingleCellExperiment'
plotBinary(x, assay.type = "counts", clusterVar = NULL, processed = FALSE, ...)

## S3 method for class 'simMat'
plotBinary(
  x,
  dotSize = 0.6,
  dotColor = "grey60",
  densLinewidth = 0.8,
  labelColors = c("#3B4992FF", "#EE0000FF"),
  title = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBinary_+3A_x">x</code></td>
<td>
<p>Input data. Can be a <code>matrix</code> or <code>dgCMatrix</code> object with
cells as columns, a <code>Seurat</code> or <code>SingleCellExperiment</code> object.
&quot;simMat&quot; method takes intermediate values.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_clustervar">clusterVar</code></td>
<td>
<p>A vector/factor assigning the cluster variable to each
column of the matrix object. For &quot;Seurat&quot; method, <code>NULL</code> (default) for
<code>Idents(x)</code>, or a variable name in <code>meta.data</code> slot. For
&quot;SingleCellExperiment&quot; method, <code>NULL</code> (default) for <code>colLabels(x)</code>,
or a variable name in <code>colData</code> slot.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_vertices">vertices</code></td>
<td>
<p>Vector of three unique cluster names that will be used for
plotting. Or a named list that groups clusters as three terminal vertices.
There must not be any overlap between groups.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_features">features</code></td>
<td>
<p>Valid matrix row subsetting index to select features for
similarity calculation. Default <code>NULL</code> uses all available features.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_bycluster">byCluster</code></td>
<td>
<p>Default <code>NULL</code> to generate one plot with all cells.
Set <code>"all"</code> to split cells in plot by cluster and returns a list of
subplots for each cluster as well as the plot including all cells. Otherwise,
a vector of cluster names to generate a list of subplots for the specified
clusters.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_processed">processed</code></td>
<td>
<p>Logical. Whether the input matrix is already processed.
<code>TRUE</code> will bypass internal preprocessing and input matrix will be
directly used for similarity calculation. Default <code>FALSE</code> and raw count
input is recommended. If missing in call, using <code>slot = "counts"</code> in
&quot;Seurat&quot; method or using <code>assay.type = "counts"</code> in
&quot;SingleCellExperiment&quot; method will force this argument to be <code>FALSE</code> and
others for <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_method">method</code></td>
<td>
<p>Similarity calculation method. Default <code>"euclidean"</code>.
Choose from <code>"euclidean"</code>, <code>"cosine"</code>, <code>"pearson"</code>,
<code>"spearman"</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_force">force</code></td>
<td>
<p>Whether to force calculate the similarity when more then 500
features are detected, which is generally not recommended. Default
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_sigma">sigma</code></td>
<td>
<p>Gaussian kernel parameter that controls the effect of variance.
Only effective when using a distance metric (i.e. <code>method</code> is
<code>"euclidian"</code> or <code>"cosine"</code>). Larger value tighten the dot
spreading on figure. Default <code>0.08</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_scale">scale</code></td>
<td>
<p>Whether to min-max scale the distance matrix by clusters.
Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_returndata">returnData</code></td>
<td>
<p>Logical. Whether to return similarity data instead of
generating plot. Default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_layer">layer</code></td>
<td>
<p>For &quot;Seurat&quot; method, which layer of the assay to be used.
Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_assay">assay</code></td>
<td>
<p>For &quot;Seurat&quot; method, the specific assay to get data from.
Default <code>NULL</code> to the default assay.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_assay.type">assay.type</code></td>
<td>
<p>For &quot;SingleCellExperiment&quot; methods. Which assay to use for
calculating the similarity. Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_dotsize">dotSize</code>, <code id="plotBinary_+3A_dotcolor">dotColor</code></td>
<td>
<p>Dot aesthetics passed to
<code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. Default <code>0.6</code> and <code>"grey60"</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_denslinewidth">densLinewidth</code></td>
<td>
<p>Density plot line aesthetic. Default <code>0.8</code>.</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_labelcolors">labelColors</code></td>
<td>
<p>Color of the axis lines and vertex labels. Default
<code>c("#3B4992FF", "#EE0000FF")</code> (blue and red).</p>
</td></tr>
<tr><td><code id="plotBinary_+3A_title">title</code></td>
<td>
<p>Title text of the plot. Default <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For 'simMat' method, a ggplot object. For other methods, a ggplot
object when <code>splitCluster = FALSE</code>, or a list of ggplot objects when
<code>splitCluster = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene &lt;- selectTopFeatures(rnaRaw, rnaCluster, c("RE", "OS"))
plotBinary(rnaRaw, rnaCluster, c("RE", "OS"), gene)

# Seurat example
library(Seurat)
srt &lt;- CreateSeuratObject(rnaRaw)
Idents(srt) &lt;- rnaCluster
gene &lt;- selectTopFeatures(srt, vertices = c("OS", "RE"))
plotBinary(srt, features = gene, vertices = c("OS", "RE"))


# SingleCellExperiment example
library(SingleCellExperiment)
sce &lt;- SingleCellExperiment(assays = list(counts = rnaRaw))
colLabels(sce) &lt;- rnaCluster
gene &lt;- selectTopFeatures(sce, vertices = c("OS", "RE"))
plotBinary(sce, features = gene, vertices = c("OS", "RE"))

</code></pre>

<hr>
<h2 id='plotQuaternary'>Create quaternary simplex plots</h2><span id='topic+plotQuaternary'></span><span id='topic+plotQuaternary.default'></span><span id='topic+plotQuaternary.Seurat'></span><span id='topic+plotQuaternary.SingleCellExperiment'></span><span id='topic+plotQuaternary.simMat'></span>

<h3>Description</h3>

<p>Create quaternary plots that show similarity between single cells and
selected four terminals in a baricentric coordinate.
</p>
<p>See <code><a href="#topic+plotTernary">plotTernary</a></code> for more details on methodologies.
</p>
<p>A dynamic rotating view in a GIF image file can be created with
<code><a href="#topic+writeQuaternaryGIF">writeQuaternaryGIF</a></code>. Package <code>magick</code> must be installed in
advance. Linux users may refer to this
<a href="https://cran.r-project.org/package=magick/vignettes/intro.html#Build_from_source">installation guide</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotQuaternary(x, ...)

## Default S3 method:
plotQuaternary(
  x,
  clusterVar,
  vertices,
  features = NULL,
  veloGraph = NULL,
  byCluster = NULL,
  processed = FALSE,
  method = c("euclidean", "cosine", "pearson", "spearman"),
  force = FALSE,
  sigma = 0.05,
  scale = TRUE,
  dotColor = "grey60",
  returnData = FALSE,
  ...
)

## S3 method for class 'Seurat'
plotQuaternary(
  x,
  layer = "counts",
  assay = NULL,
  clusterVar = NULL,
  processed = FALSE,
  ...
)

## S3 method for class 'SingleCellExperiment'
plotQuaternary(
  x,
  assay.type = "counts",
  clusterVar = NULL,
  processed = FALSE,
  ...
)

## S3 method for class 'simMat'
plotQuaternary(
  x,
  veloMat = NULL,
  nGrid = 10,
  radius = 0.2,
  dotSize = 0.6,
  dotColor = "grey60",
  labelColors = c("#3B4992FF", "#EE0000FF", "#008B45FF", "#631879FF"),
  arrowLinewidth = 0.6,
  arrowAngle = 20,
  arrowLen = 0.1,
  vertexLabelSize = 1,
  edgeLinewidth = 1,
  title = NULL,
  titleSize = 1,
  titleColor = "black",
  theta = 20,
  phi = 0,
  interactive = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotQuaternary_+3A_x">x</code></td>
<td>
<p>Input data. Can be a <code>matrix</code> or <code>dgCMatrix</code> object with
cells as columns, a <code>Seurat</code> or <code>SingleCellExperiment</code> object.
&quot;simMat&quot; method takes intermediate values.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_clustervar">clusterVar</code></td>
<td>
<p>A vector/factor assigning the cluster variable to each
column of the matrix object. For &quot;Seurat&quot; method, <code>NULL</code> (default) for
<code>Idents(x)</code>, or a variable name in <code>meta.data</code> slot. For
&quot;SingleCellExperiment&quot; method, <code>NULL</code> (default) for <code>colLabels(x)</code>,
or a variable name in <code>colData</code> slot.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_vertices">vertices</code></td>
<td>
<p>Vector of three unique cluster names that will be used for
plotting. Or a named list that groups clusters as three terminal vertices.
There must not be any overlap between groups.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_features">features</code></td>
<td>
<p>Valid matrix row subsetting index to select features for
similarity calculation. Default <code>NULL</code> uses all available features.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_velograph">veloGraph</code></td>
<td>
<p>Cell x cell <code>dgCMatrix</code> object containing velocity
information. Shows velocity grid-arrow layer when specified. Default
<code>NULL</code> does not show velocity.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_bycluster">byCluster</code></td>
<td>
<p>Default <code>NULL</code> to generate one plot with all cells.
Set <code>"all"</code> to split cells in plot by cluster and returns a list of
subplots for each cluster as well as the plot including all cells. Otherwise,
a vector of cluster names to generate a list of subplots for the specified
clusters.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_processed">processed</code></td>
<td>
<p>Logical. Whether the input matrix is already processed.
<code>TRUE</code> will bypass internal preprocessing and input matrix will be
directly used for similarity calculation. Default <code>FALSE</code> and raw count
input is recommended. If missing in call, using <code>slot = "counts"</code> in
&quot;Seurat&quot; method or using <code>assay.type = "counts"</code> in
&quot;SingleCellExperiment&quot; method will force this argument to be <code>FALSE</code> and
others for <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_method">method</code></td>
<td>
<p>Similarity calculation method. Default <code>"euclidean"</code>.
Choose from <code>"euclidean"</code>, <code>"cosine"</code>, <code>"pearson"</code>,
<code>"spearman"</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_force">force</code></td>
<td>
<p>Whether to force calculate the similarity when more then 500
features are detected, which is generally not recommended. Default
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_sigma">sigma</code></td>
<td>
<p>Gaussian kernel parameter that controls the effect of variance.
Only effective when using a distance metric (i.e. <code>method</code> is
<code>"euclidian"</code> or <code>"cosine"</code>). Larger values tighten the dot
spreading on figure. Default <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_scale">scale</code></td>
<td>
<p>Whether to min-max scale the distance matrix by clusters.
Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_returndata">returnData</code></td>
<td>
<p>Logical. Whether to return similarity and aggregated
velocity data if applicable instead of generating plot. Default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_layer">layer</code></td>
<td>
<p>For &quot;Seurat&quot; method, which layer of the assay to be used.
Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_assay">assay</code></td>
<td>
<p>For &quot;Seurat&quot; method, the specific assay to get data from.
Default <code>NULL</code> to the default assay.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_assay.type">assay.type</code></td>
<td>
<p>For &quot;SingleCellExperiment&quot; methods. Which assay to use for
calculating the similarity. Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_velomat">veloMat</code></td>
<td>
<p>Aggregated velocity matrix. Output of <code>aggrVeloGraph</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_ngrid">nGrid</code></td>
<td>
<p>Number of grids along the x-axis of the tetrahedron
triangle. Default <code>10</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_radius">radius</code></td>
<td>
<p>Arrow length of unit velocity. Lower this when arrows point
outside of the tetrahedron. Default <code>0.2</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_dotsize">dotSize</code>, <code id="plotQuaternary_+3A_dotcolor">dotColor</code></td>
<td>
<p>Dot aesthetics. Default <code>0.6</code> and
<code>"grey60"</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_labelcolors">labelColors</code></td>
<td>
<p>Colors of the vertex labels. Default
<code>c("#3B4992FF", "#EE0000FF", "#008B45FF", "#631879FF")</code> (blue, red,
green and purple).</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_arrowlinewidth">arrowLinewidth</code></td>
<td>
<p>Arrow aesthetics. Default <code>0.6</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_arrowangle">arrowAngle</code>, <code id="plotQuaternary_+3A_arrowlen">arrowLen</code></td>
<td>
<p>Arrow aesthetics passed to
<code>grid::<a href="grid.html#topic+arrow">arrow</a></code>. The length of the arrow will be internally
converted to unit onject in inches. Default <code>20</code> and <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_vertexlabelsize">vertexLabelSize</code></td>
<td>
<p>Numeric, size of vertex text label relative to default
size. Default <code>1</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_edgelinewidth">edgeLinewidth</code></td>
<td>
<p>Controls the linewidth of the edges of the tetrahedron.
Default <code>1</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_title">title</code></td>
<td>
<p>Title text of the plot. Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_titlesize">titleSize</code>, <code id="plotQuaternary_+3A_titlecolor">titleColor</code></td>
<td>
<p>Setting on the main title text. Default <code>1</code>,
and <code>"black"</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_theta">theta</code>, <code id="plotQuaternary_+3A_phi">phi</code></td>
<td>
<p>Numeric scalar. The angles defining the viewing direction.
<code>theta</code> gives the azimuthal direction and <code>phi</code> the colatitude.
Default <code>20</code> and <code>0</code>.</p>
</td></tr>
<tr><td><code id="plotQuaternary_+3A_interactive">interactive</code></td>
<td>
<p>Logical. Whether to use &quot;rgl&quot; library to create
interactive device. Default <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For &quot;simMat&quot; method, a &quot;plist&quot; (plot3D package product) object. For
other methods, a &quot;plist&quot; object when <code>splitCluster = FALSE</code>, or a list
of &quot;plist&quot; objects when <code>splitCluster = TRUE</code>. A &quot;plist&quot; object can be
viewed with <code>print()</code>, <code>show()</code> or a direct run of the object
variable name in interactive console.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene &lt;- selectTopFeatures(rnaRaw, rnaCluster, c("RE", "OS", "CH", "ORT"))
plotQuaternary(rnaRaw, rnaCluster, c("RE", "OS", "CH", "ORT"), gene)

# Seurat example
library(Seurat)
srt &lt;- CreateSeuratObject(rnaRaw)
Idents(srt) &lt;- rnaCluster
gene &lt;- selectTopFeatures(srt, vertices = c("OS", "RE", "CH", "ORT"))
plotQuaternary(srt, features = gene,
               vertices = c("OS", "RE", "CH", "ORT"))


# SingleCellExperiment example
library(SingleCellExperiment)
sce &lt;- SingleCellExperiment(assays = list(counts = rnaRaw))
colLabels(sce) &lt;- rnaCluster
gene &lt;- selectTopFeatures(sce, vertices = c("OS", "RE", "CH", "ORT"))
plotQuaternary(sce, features = gene,
               vertices = c("OS", "RE", "CH", "ORT"))

</code></pre>

<hr>
<h2 id='plotTernary'>Create ternary plots</h2><span id='topic+plotTernary'></span><span id='topic+plotTernary.default'></span><span id='topic+plotTernary.Seurat'></span><span id='topic+plotTernary.SingleCellExperiment'></span><span id='topic+plotTernary.simMat'></span>

<h3>Description</h3>

<p>Create ternary plots that show similarity between single cells and
selected three terminals in a ternary baricentric coordinate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTernary(x, ...)

## Default S3 method:
plotTernary(
  x,
  clusterVar,
  vertices,
  features = NULL,
  veloGraph = NULL,
  byCluster = NULL,
  processed = FALSE,
  method = c("euclidean", "cosine", "pearson", "spearman"),
  force = FALSE,
  sigma = 0.08,
  scale = TRUE,
  dotColor = "grey60",
  returnData = FALSE,
  ...
)

## S3 method for class 'Seurat'
plotTernary(
  x,
  layer = "counts",
  assay = NULL,
  clusterVar = NULL,
  processed = FALSE,
  ...
)

## S3 method for class 'SingleCellExperiment'
plotTernary(
  x,
  assay.type = "counts",
  clusterVar = NULL,
  processed = FALSE,
  ...
)

## S3 method for class 'simMat'
plotTernary(
  x,
  title = NULL,
  veloMat = NULL,
  nGrid = 10,
  radius = 0.1,
  dotSize = 0.6,
  dotColor = "grey60",
  labelColors = c("#3B4992FF", "#EE0000FF", "#008B45FF"),
  vertexLabelSize = 5,
  vertexLabelDrift = 0.03,
  axisBreak = 5,
  axisTextShow = TRUE,
  axisTextSize = 4,
  axisTextDrift = 0.02,
  gridLineAlpha = 0.6,
  arrowLinewidth = 0.25,
  arrowAngle = 20,
  arrowLen = 0.2,
  titleSize = 14,
  equilateral = TRUE,
  margin = 0.1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTernary_+3A_x">x</code></td>
<td>
<p>Input data. Can be a <code>matrix</code> or <code>dgCMatrix</code> object with
cells as columns, a <code>Seurat</code> or <code>SingleCellExperiment</code> object.
&quot;simMat&quot; method takes intermediate values.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_...">...</code></td>
<td>
<p>Arguments passed to other methods.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_clustervar">clusterVar</code></td>
<td>
<p>A vector/factor assigning the cluster variable to each
column of the matrix object. For &quot;Seurat&quot; method, <code>NULL</code> (default) for
<code>Idents(x)</code>, or a variable name in <code>meta.data</code> slot. For
&quot;SingleCellExperiment&quot; method, <code>NULL</code> (default) for <code>colLabels(x)</code>,
or a variable name in <code>colData</code> slot.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_vertices">vertices</code></td>
<td>
<p>Vector of three unique cluster names that will be used for
plotting. Or a named list that groups clusters as three terminal vertices.
There must not be any overlap between groups.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_features">features</code></td>
<td>
<p>Valid matrix row subsetting index to select features for
similarity calculation. Default <code>NULL</code> uses all available features.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_velograph">veloGraph</code></td>
<td>
<p>Cell x cell <code>dgCMatrix</code> object containing velocity
information. Shows velocity grid-arrow layer when specified. Default
<code>NULL</code> does not show velocity.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_bycluster">byCluster</code></td>
<td>
<p>Default <code>NULL</code> to generate one plot with all cells.
Set <code>"all"</code> to split cells in plot by cluster and returns a list of
subplots for each cluster as well as the plot including all cells. Otherwise,
a vector of cluster names to generate a list of subplots for the specified
clusters.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_processed">processed</code></td>
<td>
<p>Logical. Whether the input matrix is already processed.
<code>TRUE</code> will bypass internal preprocessing and input matrix will be
directly used for similarity calculation. Default <code>FALSE</code> and raw count
input is recommended. If missing in call, using <code>slot = "counts"</code> in
&quot;Seurat&quot; method or using <code>assay.type = "counts"</code> in
&quot;SingleCellExperiment&quot; method will force this argument to be <code>FALSE</code> and
others for <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_method">method</code></td>
<td>
<p>Similarity calculation method. Default <code>"euclidean"</code>.
Choose from <code>"euclidean"</code>, <code>"cosine"</code>, <code>"pearson"</code>,
<code>"spearman"</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_force">force</code></td>
<td>
<p>Whether to force calculate the similarity when more then 500
features are detected, which is generally not recommended. Default
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_sigma">sigma</code></td>
<td>
<p>Gaussian kernel parameter that controls the effect of variance.
Only effective when using a distance metric (i.e. <code>method</code> is
<code>"euclidian"</code> or <code>"cosine"</code>). Larger values tighten the dot
spreading on figure. Default <code>0.08</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_scale">scale</code></td>
<td>
<p>Whether to min-max scale the distance matrix by clusters.
Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_returndata">returnData</code></td>
<td>
<p>Logical. Whether to return similarity and aggregated
velocity data if applicable instead of generating plot. Default <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_layer">layer</code></td>
<td>
<p>For &quot;Seurat&quot; method, which layer of the assay to be used.
Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_assay">assay</code></td>
<td>
<p>For &quot;Seurat&quot; method, the specific assay to get data from.
Default <code>NULL</code> to the default assay.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_assay.type">assay.type</code></td>
<td>
<p>For &quot;SingleCellExperiment&quot; methods. Which assay to use for
calculating the similarity. Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_title">title</code></td>
<td>
<p>Title text of the plot. Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_velomat">veloMat</code></td>
<td>
<p>Aggregated velocity matrix. Output of <code>aggrVeloGraph</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_ngrid">nGrid</code></td>
<td>
<p>Number of grids along the bottom side of the equilateral
triangle. Default <code>10</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_radius">radius</code></td>
<td>
<p>Arrow length of unit velocity. Lower this when arrows point
outside of the coordinate. Default <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_dotsize">dotSize</code>, <code id="plotTernary_+3A_dotcolor">dotColor</code></td>
<td>
<p>Dot aesthetics passed to
<code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>. Default <code>0.6</code> and <code>"grey60"</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_labelcolors">labelColors</code></td>
<td>
<p>Colors of the axis lines and vertex labels.
Default <code>c("#3B4992FF", "#EE0000FF", "#008B45FF")</code> (blue, red and green)</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_vertexlabelsize">vertexLabelSize</code>, <code id="plotTernary_+3A_vertexlabeldrift">vertexLabelDrift</code></td>
<td>
<p>Adjustment on the three vertex text
labels. Drift means the distance that the labels should be moved against the
center of the plot. Default size <code>5</code>, drifted distance <code>0.03</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_axisbreak">axisBreak</code></td>
<td>
<p>Number of breaks to be labeled along axis. Default
<code>5</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_axistextshow">axisTextShow</code></td>
<td>
<p>Logical, whether to show axis text. Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_axistextsize">axisTextSize</code>, <code id="plotTernary_+3A_axistextdrift">axisTextDrift</code></td>
<td>
<p>Similar to the vertex adjustment applied
to the text label along the axis breaks. Default size <code>4</code>, drifted
distance <code>0.02</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_gridlinealpha">gridLineAlpha</code></td>
<td>
<p>Transparency of background grid lines. Default
<code>0.6</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_arrowlinewidth">arrowLinewidth</code>, <code id="plotTernary_+3A_arrowangle">arrowAngle</code>, <code id="plotTernary_+3A_arrowlen">arrowLen</code></td>
<td>
<p>Arrow aesthetics, see Details.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_titlesize">titleSize</code></td>
<td>
<p>Size of title text. Default <code>14</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_equilateral">equilateral</code></td>
<td>
<p>Logical, whether to always display the triangle as
equilateral. Default <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotTernary_+3A_margin">margin</code></td>
<td>
<p>Margin allowed around of the triangle plotting region when
<code>equilateral = TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><b>Argument inheritance</b> - For matrix/dgCMatrix (&quot;default&quot; method), we
first calculate the similarity matrix and obtain a &quot;simMat&quot; object. Then the
&quot;simMat&quot; method is internally called. For data container objects (e.g.
Seurat), we obtain the correct data matrix first and then call the &quot;default&quot;
method. The arguments inherits as the flow described above.
</p>
<p><b>The calculation of similarity matrix</b> - The similarity is calculated
either by converting a distance metric (&quot;euclidean&quot; or &quot;cosine&quot;) with
Gaussian kernel, or directly computed with correlation metrics (&quot;pearson&quot; or
&quot;spearman&quot;). The centroid of each terminal is obtained first, and the
specified metric from each cell to each terminal is calculated. The
similarity matrix (n cells by v terminals) is lastly normalized to sum to 1
for each cell, so it becomes a baricentric coordinate.
</p>
<p><b>Arrow aesthetics parameters</b> - The shape of arrows is controlled by 3
arguments. Considering an arrow as the combination of a line segment and a
triangle, <code>arrowLinewidth</code> controls the width of the line as well as
the edge line of the triangle; <code>arrowAngle</code> equals to angle of the
arrow-tip vertex of the triangle devided by 2 (e.g. the triangle is
equilateral when <code>arrowAngle = 20</code>); <code>arrowLen</code> controls the
absolute length from the arrow-tip vertex to its opposite edge.
</p>


<h3>Value</h3>

<p>For &quot;simMat&quot; method, a ggplot object. For other methods, a ggplot
object when <code>splitCluster = FALSE</code>, or a list of ggplot objects when
<code>splitCluster = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene &lt;- selectTopFeatures(rnaRaw, rnaCluster, c("OS", "RE", "CH"))
plotTernary(rnaRaw, rnaCluster, c("OS", "RE", "CH"), gene)

# Seurat example
library(Seurat)
srt &lt;- CreateSeuratObject(rnaRaw)
Idents(srt) &lt;- rnaCluster
gene &lt;- selectTopFeatures(srt, vertices = c("OS", "RE", "CH"))
plotTernary(srt, features = gene, vertices = c("OS", "RE", "CH"))


# SingleCellExperiment example
library(SingleCellExperiment)
sce &lt;- SingleCellExperiment(assays = list(counts = rnaRaw))
colLabels(sce) &lt;- rnaCluster
gene &lt;- selectTopFeatures(sce, vertices = c("OS", "RE", "CH"))
plotTernary(sce, features = gene, vertices = c("OS", "RE", "CH"))

</code></pre>

<hr>
<h2 id='rnaCluster'>Major cell type annotation of the example mouse bone marrow data</h2><span id='topic+rnaCluster'></span>

<h3>Description</h3>

<p>Major cell type annotation of the example mouse bone marrow data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnaCluster
</code></pre>


<h3>Format</h3>

<p><code>factor</code> object
</p>


<h3>Source</h3>

<p>https://www.nature.com/articles/s41467-023-38034-2
</p>


<h3>References</h3>

<p>Matsushita, Y., Liu, J., Chu, A.K.Y. et al. Bone marrow endosteal
stem cells dictate active osteogenesis and aggressive tumorigenesis. Nat
Commun 14, 2383 (2023).
</p>

<hr>
<h2 id='rnaRaw'>Mouse bone marrow scRNAseq example data</h2><span id='topic+rnaRaw'></span>

<h3>Description</h3>

<p>Mouse bone marrow scRNAseq example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnaRaw
</code></pre>


<h3>Format</h3>

<p><a href="Matrix.html#topic+dgCMatrix-class">dgCMatrix</a> object
</p>


<h3>Source</h3>

<p>https://www.nature.com/articles/s41467-023-38034-2
</p>


<h3>References</h3>

<p>Matsushita, Y., Liu, J., Chu, A.K.Y. et al. Bone marrow endosteal
stem cells dictate active osteogenesis and aggressive tumorigenesis. Nat
Commun 14, 2383 (2023).
</p>

<hr>
<h2 id='rnaVelo'>Velocity graph of the example mouse bone marrow data</h2><span id='topic+rnaVelo'></span>

<h3>Description</h3>

<p>Velocity graph of the example mouse bone marrow data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnaVelo
</code></pre>


<h3>Format</h3>

<p><a href="Matrix.html#topic+dgCMatrix-class">dgCMatrix</a> object
</p>


<h3>Source</h3>

<p>https://www.nature.com/articles/s41467-023-38034-2
</p>


<h3>References</h3>

<p>Matsushita, Y., Liu, J., Chu, A.K.Y. et al. Bone marrow endosteal
stem cells dictate active osteogenesis and aggressive tumorigenesis. Nat
Commun 14, 2383 (2023).
</p>

<hr>
<h2 id='selectTopFeatures'>Pick top differentially presented features for similarity calculation</h2><span id='topic+selectTopFeatures'></span><span id='topic+selectTopFeatures.default'></span><span id='topic+selectTopFeatures.Seurat'></span><span id='topic+selectTopFeatures.SingleCellExperiment'></span>

<h3>Description</h3>

<p>Performs wilcoxon rank-sum test on input matrix. While <code>clusterVar</code> and
<code>vertices</code> together defines the groups of cells to be set as terminals
of the simplex, this function will test each of these groups against the rest
of the cells. The U-Statistics (<code>statistic</code>), p-value (<code>pval</code>) and
adjusted p-value (<code>padj</code>), together with average presence in group
(<code>avgExpr</code>), log fold-change (<code>logFC</code>), AUC (<code>auc</code>),
percentage in group (<code>pct_in</code>) and percentage out of group
(<code>pct_out</code>) will be calculated. Set <code>returnStats = TRUE</code> to return
the full statistics table.
</p>
<p>Top features are selected by sorting primarily on adjusted p-value, and
secondarily on log fold-change, after filtering for up-regulated features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selectTopFeatures(x, clusterVar, vertices, ...)

## Default S3 method:
selectTopFeatures(
  x,
  clusterVar,
  vertices,
  nTop = 30,
  processed = FALSE,
  lfcThresh = 0.1,
  returnStats = FALSE,
  ...
)

## S3 method for class 'Seurat'
selectTopFeatures(
  x,
  clusterVar = NULL,
  vertices,
  assay = NULL,
  layer = "counts",
  processed = FALSE,
  ...
)

## S3 method for class 'SingleCellExperiment'
selectTopFeatures(
  x,
  clusterVar = NULL,
  vertices,
  assay.type = "counts",
  processed = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selectTopFeatures_+3A_x">x</code></td>
<td>
<p>Dense or sparse matrix, observation per column. Preferrably a raw
count matrix. Alternatively, a <code>Seurat</code> object or a
<code>SingleCellExperiment</code> object.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_clustervar">clusterVar</code></td>
<td>
<p>A vector/factor assigning the cluster variable to each
column of the matrix object. For &quot;Seurat&quot; method, <code>NULL</code> (default) for
<code>Idents(x)</code>, or a variable name in <code>meta.data</code> slot. For
&quot;SingleCellExperiment&quot; method, <code>NULL</code> (default) for <code>colLabels(x)</code>,
or a variable name in <code>colData</code> slot.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_vertices">vertices</code></td>
<td>
<p>Vector of cluster names that will be used for plotting. Or a
named list that groups clusters as a terminal vertex. There must not be any
overlap between groups.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_...">...</code></td>
<td>
<p>Arguments passed to methods.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_ntop">nTop</code></td>
<td>
<p>Number of top differentially presented features per terminal.
Default <code>30</code>.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_processed">processed</code></td>
<td>
<p>Logical. Whether the input matrix is already processed.
<code>TRUE</code> will bypass internal preprocessing and input matrix will be
directly used for rank-sum calculation. Default <code>FALSE</code> and raw count
input is recommended.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_lfcthresh">lfcThresh</code></td>
<td>
<p>Threshold on log fold-change to identify up-regulated
features. Default <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_returnstats">returnStats</code></td>
<td>
<p>Logical. Whether to return the full statistics table
rather then returning the selected genes. Default <code>FALSE</code></p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_assay">assay</code></td>
<td>
<p>Assay name of the Seurat object to be used. Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_layer">layer</code></td>
<td>
<p>For &quot;Seurat&quot; method, which layer of the assay to be used.
Default <code>"counts"</code>.</p>
</td></tr>
<tr><td><code id="selectTopFeatures_+3A_assay.type">assay.type</code></td>
<td>
<p>Assay name of the SingleCellExperiment object to be used.
Default <code>"counts"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>When <code>returnStats = FALSE</code> (default), a character vector of at
most <code>length(unique(vertices))*nTop</code> feature names. When
<code>returnStats = TRUE</code>, a data.frame of wilcoxon rank sum test statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>selectTopFeatures(rnaRaw, rnaCluster, c("OS", "RE"))

# Seurat example
library(Seurat)
srt &lt;- CreateSeuratObject(rnaRaw)
Idents(srt) &lt;- rnaCluster
gene &lt;- selectTopFeatures(srt, vertices = c("OS", "RE"))


# SingleCellExperiment example
library(SingleCellExperiment)
sce &lt;- SingleCellExperiment(assays = list(counts = rnaRaw))
colLabels(sce) &lt;- rnaCluster
gene &lt;- selectTopFeatures(sce, vertices = c("OS", "RE"))

</code></pre>

<hr>
<h2 id='show+2Cplist-method'>Show plist object produced with plot3D package</h2><span id='topic+show+2Cplist-method'></span><span id='topic+print.plist'></span>

<h3>Description</h3>

<p>Show plist object produced with plot3D package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'plist'
show(object)

## S3 method for class 'plist'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2Cplist-method_+3A_object">object</code>, <code id="show+2B2Cplist-method_+3A_x">x</code></td>
<td>
<p>plist object</p>
</td></tr>
<tr><td><code id="show+2B2Cplist-method_+3A_...">...</code></td>
<td>
<p>Graphic parameters passed to <code><a href="base.html#topic+plot">plot</a></code>. <code>mar</code> is
pre-specified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. It displays the plot described in a 'plist' object
returned by <code><a href="#topic+plotQuaternary">plotQuaternary</a></code>, internally created by package
'plot3D'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene &lt;- selectTopFeatures(rnaRaw, rnaCluster, c("RE", "OS", "CH", "ORT"))
plistObj &lt;- plotQuaternary(rnaRaw, rnaCluster, c("RE", "OS", "CH", "ORT"), gene)
print(plistObj)
# equivalent to
show(plistObj)
</code></pre>

<hr>
<h2 id='writeQuaternaryGIF'>Create GIF image for dynamic rotating view of 3D quaternary simplex plot</h2><span id='topic+writeQuaternaryGIF'></span>

<h3>Description</h3>

<p>Create GIF image for dynamic rotating view of 3D quaternary simplex plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeQuaternaryGIF(
  x,
  ...,
  cluster = NULL,
  gifPath = "quaternary.gif",
  tmpDir = tempdir(),
  fps = 10,
  degreePerFrame = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeQuaternaryGIF_+3A_x">x</code></td>
<td>
<p>Input object that <code><a href="#topic+plotQuaternary">plotQuaternary</a></code> accepts.</p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_...">...</code></td>
<td>
<p>All other arguments needed for <code><a href="#topic+plotQuaternary">plotQuaternary</a></code>. Must
be specified with exact argument names instead of a positional manner.</p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_cluster">cluster</code></td>
<td>
<p>One cluster that exists in <code>clusterVar</code>, if users
need to view the plot for specific group. Default <code>NULL</code> plot all cells.</p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_gifpath">gifPath</code></td>
<td>
<p>Output GIF image file path. Default <code>"quaternary.gif"</code></p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_tmpdir">tmpDir</code></td>
<td>
<p>A temprorary directory to store all PNG files for all
perspectives created. Default <code>tempdir()</code>.</p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_fps">fps</code></td>
<td>
<p>Number of frame per second, must be a factor of 100. Default
<code>10</code>.</p>
</td></tr>
<tr><td><code id="writeQuaternaryGIF_+3A_degreeperframe">degreePerFrame</code></td>
<td>
<p>Number of degree that the tetrahedron is rotated per
frame. Default <code>10</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No object is returned. The <code>tmpDir</code> folder will be created with
<code>360 / degreePerFrame</code> PNG image files in it. A GIF image file will be
created at <code>gifPath</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene &lt;- selectTopFeatures(rnaRaw, rnaCluster, c("RE", "OS", "CH", "ORT"))

writeQuaternaryGIF(rnaRaw, clusterVar = rnaCluster, features = gene,
                   vertices = c("RE", "OS", "CH", "ORT"),
                   gifPath = paste0(tempfile(), ".gif"))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
