<!DOCTYPE html><html lang="en"><head><title>Help for package D3GB</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {D3GB}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#D3GB'><p>Interactive genome browsers with R</p></a></li>
<li><a href='#createAssembly'><p>Creates a genome assembly for <code>genomebrowser</code> or <code>genomemap</code>.</p></a></li>
<li><a href='#gbk2genomebrowser'><p>Generates an interactive genome browser.</p></a></li>
<li><a href='#genome_addGFF'><p>Add track in a gff file to <code>genomebrowser</code>.</p></a></li>
<li><a href='#genome_addSequence'><p>Add genome sequence on Fasta format to <code>genomebrowser</code>.</p></a></li>
<li><a href='#genome_addTrack'><p>Add tracks to <code>genomebrowser</code>.</p></a></li>
<li><a href='#genome_addVCF'><p>Add VCF tracks to <code>genomebrowser</code>.</p></a></li>
<li><a href='#genome_createLocalMode'><p>Generates an interactive genome browser.</p></a></li>
<li><a href='#genome_createServerMode'><p>Generates an interactive genome browser.</p></a></li>
<li><a href='#genomebrowser'><p>Generates an interactive genome browser.</p></a></li>
<li><a href='#genomemap'><p>Create an interative genome map.</p></a></li>
<li><a href='#getAssemblyFromFasta'><p>Create an assembly from a FASTA file for its use as parameter of <code>genomebrowser</code> or <code>genomemap</code>.</p></a></li>
<li><a href='#GRCh37'><p>Length of human chromosomes based on GRCh37 assembly</p></a></li>
<li><a href='#GRCh37.bands'><p>Cytoband information based on GRCh37</p></a></li>
<li><a href='#GRCh38'><p>Length of human chromosomes based on GRCh38 assembly</p></a></li>
<li><a href='#GRCh38.bands'><p>Cytoband information based on GRCh38</p></a></li>
<li><a href='#NCBI36'><p>Length of human chromosomes based on NCBI36 assembly</p></a></li>
<li><a href='#segmentation'><p>Function for segmentate tracks in BED format.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-05-24</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Genome Browser</td>
</tr>
<tr>
<td>Description:</td>
<td>Creates interactive genome browser. It joins the data analysis power of 'R' and the visualization libraries of 'JavaScript' in one package. Barrios, D. &amp; Prieto, C. (2017) &lt;<a href="https://doi.org/10.1089%2Fcmb.2016.0213">doi:10.1089/cmb.2016.0213</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://d3gb.usal.es">http://d3gb.usal.es</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>RSQLite (&ge; 1.0.0), DBI (&ge; 0.3.1), base64enc</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carlos Prieto &lt;cprietos@usal.es&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-29 10:00:32 UTC; dbarrios</td>
</tr>
<tr>
<td>Author:</td>
<td>Carlos Prieto <a href="https://orcid.org/0000-0001-8178-9768"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, cre],
  David Barrios <a href="https://orcid.org/0000-0003-4465-0200"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-30 08:10:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='D3GB'>Interactive genome browsers with R</h2><span id='topic+D3GB'></span><span id='topic+D3GB-package'></span>

<h3>Description</h3>

<p>Create interactive genome browsers with 'R'.</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>

<hr>
<h2 id='createAssembly'>Creates a genome assembly for <code>genomebrowser</code> or <code>genomemap</code>.</h2><span id='topic+createAssembly'></span>

<h3>Description</h3>

<p><code>createAssembly</code> create assemblies for their use as parameter of <code>genomebrowser</code> or <code>genomemap</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createAssembly(name, size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createAssembly_+3A_name">name</code></td>
<td>
<p>a vector with the chomosome or scaffold names.</p>
</td></tr>
<tr><td><code id="createAssembly_+3A_size">size</code></td>
<td>
<p>a vector with the chomosome or scaffold sizes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame in BED format to be used as an assembly in <code>genomebrowser</code> or <code>genomemap</code>.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+genomemap">genomemap</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A Micromonospora scaffolds' names and sizes
scaffolds &lt;- c("NZ_HF570105.1", "NZ_HF570106.1", "NZ_HF570107.1", "NZ_HF570108.1")
sizes &lt;- c(583, 1327, 241394, 7082520)

# Create an assembly
createAssembly(scaffolds, sizes)
</code></pre>

<hr>
<h2 id='gbk2genomebrowser'>Generates an interactive genome browser.</h2><span id='topic+gbk2genomebrowser'></span>

<h3>Description</h3>

<p><code>gbk2genomebrowser</code> creates an interactive genome browser from a GenBank file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbk2genomebrowser(gbkfile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gbk2genomebrowser_+3A_gbkfile">gbkfile</code></td>
<td>
<p>a &quot;character&quot; string representing the input GenBank file to be represented in the genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>genomebrowser</code>.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Download GenBank file
gbff &lt;- tempfile()
download.file(paste0("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/",
"GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.gbff.gz"),gbff)

# Genome browser generation.
gb &lt;- gbk2genomebrowser(gbff)

plot(gb)

</code></pre>

<hr>
<h2 id='genome_addGFF'>Add track in a gff file to <code>genomebrowser</code>.</h2><span id='topic+genome_addGFF'></span>

<h3>Description</h3>

<p><code>genome_addGFF</code> add track in a gff file to <code>genomebrowser</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_addGFF(gb, gfffile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_addGFF_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_addGFF_+3A_gfffile">gfffile</code></td>
<td>
<p>a &quot;character&quot; string representing the input gff file to be represented in the genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+genome_addSequence">genome_addSequence</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>, <code><a href="#topic+genome_addVCF">genome_addVCF</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Download fasta file
fasta &lt;- tempfile()
download.file(paste0("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/",
"GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.fna.gz"),fasta)

# Genome browser generation.
gb &lt;- genomebrowser(getAssemblyFromFasta(fasta))

genome_addSequence(gb,fasta)

# Download gff file and add to the genome browser
gff &lt;- tempfile()
download.file(paste0("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/",
"GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.gff.gz"),gff)

genome_addGFF(gb,gff)

plot(gb)

</code></pre>

<hr>
<h2 id='genome_addSequence'>Add genome sequence on Fasta format to <code>genomebrowser</code>.</h2><span id='topic+genome_addSequence'></span>

<h3>Description</h3>

<p><code>genome_addSequence</code> add FASTA sequences to <code>genomeviewer</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_addSequence(gb, fastafile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_addSequence_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_addSequence_+3A_fastafile">fastafile</code></td>
<td>
<p>a &quot;character&quot; string representing the input Fasta file to be added in the genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+genome_addGFF">genome_addGFF</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>, <code><a href="#topic+genome_addVCF">genome_addVCF</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Download fasta file
fasta &lt;- tempfile()
download.file(paste0("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/",
"GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.fna.gz"),fasta)

# Genome browser generation.
gb &lt;- genomebrowser(getAssemblyFromFasta(fasta))

genome_addSequence(gb,fasta)

plot(gb)

</code></pre>

<hr>
<h2 id='genome_addTrack'>Add tracks to <code>genomebrowser</code>.</h2><span id='topic+genome_addTrack'></span>

<h3>Description</h3>

<p><code>genome_addTrack</code> add tracks to <code>genomebrowser</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_addTrack(gb, track, trackname = NULL, type = "gene", color = "#000", scale = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_addTrack_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_addTrack_+3A_track">track</code></td>
<td>
<p>a &quot;character&quot; string giving the input BED file or a data frame with track information to be represented in the genome browser.</p>
</td></tr>
<tr><td><code id="genome_addTrack_+3A_trackname">trackname</code></td>
<td>
<p>a &quot;character&quot; string giving a name for the track.</p>
</td></tr>
<tr><td><code id="genome_addTrack_+3A_type">type</code></td>
<td>
<p>a &quot;character&quot; string with the type of track should be drawn. Possible types are: &quot;gene&quot;, &quot;exons&quot;, &quot;domain&quot;, &quot;value&quot; or &quot;score&quot;.</p>
</td></tr>
<tr><td><code id="genome_addTrack_+3A_color">color</code></td>
<td>
<p>a &quot;character&quot; string giving the color of the track.</p>
</td></tr>
<tr><td><code id="genome_addTrack_+3A_scale">scale</code></td>
<td>
<p>a vector with two values which specifies the minimun and maximun limits in the representation of the &quot;score&quot; or &quot;value&quot; tracks. By default maximun and minimun scores are taken as the limits.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>

<hr>
<h2 id='genome_addVCF'>Add VCF tracks to <code>genomebrowser</code>.</h2><span id='topic+genome_addVCF'></span>

<h3>Description</h3>

<p><code>genome_addVCF</code> add VCF tracks to <code>genomebrowser</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_addVCF(gb, vcffile, trackname = NULL, show = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_addVCF_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_addVCF_+3A_vcffile">vcffile</code></td>
<td>
<p>a &quot;character&quot; string representing the input VCF file to be represented in the genome browser.</p>
</td></tr>
<tr><td><code id="genome_addVCF_+3A_trackname">trackname</code></td>
<td>
<p>a &quot;character&quot; string giving a name for the track.</p>
</td></tr>
<tr><td><code id="genome_addVCF_+3A_show">show</code></td>
<td>
<p>a &quot;character&quot; vector giving the info features to display.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>

<hr>
<h2 id='genome_createLocalMode'>Generates an interactive genome browser.</h2><span id='topic+genome_createLocalMode'></span>

<h3>Description</h3>

<p><code>genome_createLocalMode</code> creates an interactive genome browser to open directly in your browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_createLocalMode(gb, dir="GenomeBrowser")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_createLocalMode_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_createLocalMode_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a folder in the working directory of your computer with an HTML document (index.html) which contains the graph. This file can be directly opened with your browser.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+createAssembly">createAssembly</a></code>, <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>, <code><a href="#topic+GRCh37.bands">GRCh37.bands</a></code>, <code><a href="#topic+GRCh38.bands">GRCh38.bands</a></code>, <code><a href="#topic+genome_addGFF">genome_addGFF</a></code>, <code><a href="#topic+genome_addSequence">genome_addSequence</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>, <code><a href="#topic+genome_addVCF">genome_addVCF</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Genome browser generation.
gb &lt;- genomebrowser(GRCh38.bands)
# It creates a genome browser ready to be viewed in your browser.
genome_createLocalMode(gb,dir=tempfile())
</code></pre>

<hr>
<h2 id='genome_createServerMode'>Generates an interactive genome browser.</h2><span id='topic+genome_createServerMode'></span>

<h3>Description</h3>

<p><code>genome_createServerMode</code> creates an interactive genome browser to be shared as a website.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_createServerMode(gb, dir="GenomeBrowser")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_createServerMode_+3A_gb">gb</code></td>
<td>
<p>a genome browser object produced by the function <code>genomebrowser</code>.</p>
</td></tr>
<tr><td><code id="genome_createServerMode_+3A_dir">dir</code></td>
<td>
<p>a &quot;character&quot; string representing the directory where the graph will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a website. Resulting folder should be added to the Apache applications directory. In this way the genome browser will be working as a web site.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+createAssembly">createAssembly</a></code>, <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>, <code><a href="#topic+GRCh37.bands">GRCh37.bands</a></code>, <code><a href="#topic+GRCh38.bands">GRCh38.bands</a></code>, <code><a href="#topic+genome_addGFF">genome_addGFF</a></code>, <code><a href="#topic+genome_addSequence">genome_addSequence</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>, <code><a href="#topic+genome_addVCF">genome_addVCF</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Genome browser generation.
gb &lt;- genomebrowser(GRCh38.bands)
# It creates a website. You can add it to your Apache server.
genome_createServerMode(gb,dir=tempfile())
</code></pre>

<hr>
<h2 id='genomebrowser'>Generates an interactive genome browser.</h2><span id='topic+genomebrowser'></span>

<h3>Description</h3>

<p><code>genomebrowser</code> creates an interactive genome browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomebrowser(assembly, tracks = NULL, types = NA, colors = NA, mapTrack = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomebrowser_+3A_tracks">tracks</code></td>
<td>
<p>a list of data frames with track information that will be viewed in the genome browser. These data frames should have a BED format with the following columns: chromosome/scaffold, start, end, name, score, strand.</p>
</td></tr>
<tr><td><code id="genomebrowser_+3A_types">types</code></td>
<td>
<p>a &quot;character&quot; vector with type of tracks that will be represented. It should be one of: &quot;gene&quot;, &quot;exons&quot;, &quot;domain&quot;, &quot;value&quot; or &quot;score&quot;.</p>
</td></tr>
<tr><td><code id="genomebrowser_+3A_colors">colors</code></td>
<td>
<p>a &quot;character&quot; vector with color used for the track representation.</p>
</td></tr>
<tr><td><code id="genomebrowser_+3A_maptrack">mapTrack</code></td>
<td>
<p>a data frame with values to represent on the genome map in BED format: chromosome, start, end, name, value.</p>
</td></tr>
<tr><td><code id="genomebrowser_+3A_assembly">assembly</code></td>
<td>
<p>a genome assembly data frame. D3GB provides human assemblies (NCBI36, GRCh37, GRCh38), human assemblies with cytobands (GRCh37.bands or GRCh38.bands), or methods for creating your own assembly from a FASTA file <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>genomebrowser</code>.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+createAssembly">createAssembly</a></code>, <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>, <code><a href="#topic+GRCh37.bands">GRCh37.bands</a></code>, <code><a href="#topic+GRCh38.bands">GRCh38.bands</a></code>, <code><a href="#topic+genome_addGFF">genome_addGFF</a></code>, <code><a href="#topic+genome_addSequence">genome_addSequence</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>, <code><a href="#topic+genome_addVCF">genome_addVCF</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Genome browser generation.
gb &lt;- genomebrowser(GRCh38.bands)
# Now you can add sequences, tracks, gff or vcf
</code></pre>

<hr>
<h2 id='genomemap'>Create an interative genome map.</h2><span id='topic+genomemap'></span>

<h3>Description</h3>

<p><code>genomemap</code> creates an interactive genome map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomemap(assembly, mapTrack = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomemap_+3A_assembly">assembly</code></td>
<td>
<p>a genome assembly data frame. D3GB provides human assemblies (NCBI36, GRCh37, GRCh38), human assemblies with cytobands (GRCh37.bands or GRCh38.bands), or methods for creating your own assembly from a FASTA file <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>.</p>
</td></tr>
<tr><td><code id="genomemap_+3A_maptrack">mapTrack</code></td>
<td>
<p>a data frame with values to represent on the genome map in BED format: chromosome, start, end, name, value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>genomebrowser</code>.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+createAssembly">createAssembly</a></code>, <code><a href="#topic+getAssemblyFromFasta">getAssemblyFromFasta</a></code>, <code><a href="#topic+GRCh37.bands">GRCh37.bands</a></code>, <code><a href="#topic+GRCh38.bands">GRCh38.bands</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gm &lt;- genomemap(GRCh38.bands)
</code></pre>

<hr>
<h2 id='getAssemblyFromFasta'>Create an assembly from a FASTA file for its use as parameter of <code>genomebrowser</code> or <code>genomemap</code>.</h2><span id='topic+getAssemblyFromFasta'></span>

<h3>Description</h3>

<p><code>getAssemblyFromFasta</code> create assemblies from a FASTA file for their use as parameter of <code>genomebrowser</code> or <code>genomemap</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAssemblyFromFasta(fasta)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAssemblyFromFasta_+3A_fasta">fasta</code></td>
<td>
<p>a &quot;character&quot; string representing the input Fasta file to be added in the genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame in BED format ready to use as an assembly in <code>genomebrowser</code> or <code>genomemap</code>.
</p>


<h3>Note</h3>

<p>FASTA format is a text-based format for representing either nucleotide sequences or peptide sequences, in which nucleotides or amino acids are represented using single-letter codes.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+genomemap">genomemap</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Download fasta file
fasta &lt;- tempfile()
download.file(paste0("https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/297/395/",
"GCF_000297395.2_ASM29739v2/GCF_000297395.2_ASM29739v2_genomic.fna.gz"),fasta)

# Assembly generation
getAssemblyFromFasta(fasta)

</code></pre>

<hr>
<h2 id='GRCh37'>Length of human chromosomes based on GRCh37 assembly</h2><span id='topic+GRCh37'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on GRCh37 assembly.</p>


<h3>Value</h3>

<p>A Data Frame containing length of chromosomes of GRCh37 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/">https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13/</a>.</p>

<hr>
<h2 id='GRCh37.bands'>Cytoband information based on GRCh37</h2><span id='topic+GRCh37.bands'></span>

<h3>Description</h3>

<p>A Data Frame containing cytobands of GRCh37 assembly.</p>


<h3>Value</h3>

<p>A Data Frame containing cytobands of GRCh37 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://grch37.ensembl.org/Homo_sapiens/Location/Genome">https://grch37.ensembl.org/Homo_sapiens/Location/Genome</a>.</p>

<hr>
<h2 id='GRCh38'>Length of human chromosomes based on GRCh38 assembly</h2><span id='topic+GRCh38'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on GRCh38 assembly.</p>


<h3>Value</h3>

<p>A Data Frame containing length of chromosomes of GRCh38 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/883148/">https://www.ncbi.nlm.nih.gov/assembly/883148/</a>.</p>

<hr>
<h2 id='GRCh38.bands'>Cytoband information based on GRCh38</h2><span id='topic+GRCh38.bands'></span>

<h3>Description</h3>

<p>A Data Frame containing cytobands of GRCh38 assembly.</p>


<h3>Value</h3>

<p>A Data Frame containing cytobands of GRCh38 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ensembl.org/Homo_sapiens/Location/Genome">https://www.ensembl.org/Homo_sapiens/Location/Genome</a>.</p>

<hr>
<h2 id='NCBI36'>Length of human chromosomes based on NCBI36 assembly</h2><span id='topic+NCBI36'></span>

<h3>Description</h3>

<p>Length of human chromosomes based on NCBI36 assembly.</p>


<h3>Value</h3>

<p>A Data Frame containing length of chromosomes of NCBI36 assembly.</p>


<h3>Source</h3>

<p>See <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.12/">https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.12/</a>.</p>

<hr>
<h2 id='segmentation'>Function for segmentate tracks in BED format.</h2><span id='topic+segmentation'></span>

<h3>Description</h3>

<p><code>segmentation</code> performs a segmentation of track data in bed format in order to enable its effective representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segmentation(track, cell)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="segmentation_+3A_track">track</code></td>
<td>
<p>a Data Frame with genomic features (in BED format).</p>
</td></tr>
<tr><td><code id="segmentation_+3A_cell">cell</code></td>
<td>
<p>a numeric value with the size of each segment.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a segmented Data Frame in BED format.
</p>


<h3>Note</h3>

<p>Input file is provided in BED format which is an universal format for genomic features.
</p>


<h3>Author(s)</h3>

<p>David Barrios and Carlos Prieto. Bioinformatics, University of Salamanca. See <a href="http://d3gb.usal.es/">http://d3gb.usal.es/</a>
</p>


<h3>See Also</h3>

<p>The &lsquo;D3GB&rsquo; Website: <a href="http://d3gb.usal.es">http://d3gb.usal.es</a>
</p>
<p><code><a href="#topic+genomebrowser">genomebrowser</a></code>, <code><a href="#topic+genomemap">genomemap</a></code>, <code><a href="#topic+genome_addTrack">genome_addTrack</a></code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
