<!DOCTYPE html><html><head><title>Help for package DAAG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DAAG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DAAG-package'>
<p>The R DAAG Package</p></a></li>
<li><a href='#ACF1'><p>Aberrant Crypt Foci in Rat Colons</p></a></li>
<li><a href='#ais'><p>Australian athletes data set</p></a></li>
<li><a href='#align2D'>
<p>Function to align points from ordination with known locations</p></a></li>
<li><a href='#allbacks'><p>Measurements on a Selection of Books</p></a></li>
<li><a href='#anesthetic'><p>Anesthetic Effectiveness</p></a></li>
<li><a href='#antigua'><p>Averages by block of yields for the Antigua Corn data</p></a></li>
<li><a href='#appletaste'><p>Tasting experiment that compared four apple varieties</p></a></li>
<li><a href='#audists'><p>Road distances between 10 Australian cities</p></a></li>
<li><a href='#aulatlong'><p>Latitudes and longitudes for ten Australian cities</p></a></li>
<li><a href='#austpop'><p>Population figures for Australian States and Territories</p></a></li>
<li><a href='#bestsetNoise'><p>Best Subset Selection Applied to Noise</p></a></li>
<li><a href='#biomass'><p>Biomass Data</p></a></li>
<li><a href='#bomregions2021'><p>Australian and Related Historical Annual Climate Data, by Region</p></a></li>
<li><a href='#bomsoi'><p>Southern Oscillation Index Data</p></a></li>
<li><a href='#bostonc'><p>Boston Housing Data &ndash; Corrected</p></a></li>
<li><a href='#bounce'><p>Separate plotting positions for labels, to avoid overlap</p></a></li>
<li><a href='#capstring'><p>Converts initial character of a string to upper case</p></a></li>
<li><a href='#carprice'><p>US Car Price Data</p></a></li>
<li><a href='#Cars93.summary'><p>A Summary of the Cars93 Data set</p></a></li>
<li><a href='#cerealsugar'><p>Percentage of Sugar in Breakfast Cereal</p></a></li>
<li><a href='#cfseal'><p>Cape Fur Seal Data</p></a></li>
<li><a href='#cities'><p>Populations of Major Canadian Cities (1992-96)</p></a></li>
<li><a href='#codling'><p>Dose-mortality data, for fumigation of codling moth with methyl bromide</p></a></li>
<li><a href='#compareTreecalcs'><p>Error rate comparisons for tree-based classification</p></a></li>
<li><a href='#component.residual'><p>Component + Residual Plot</p></a></li>
<li><a href='#confusion'>
<p>Given actual and predicted group assignments, give the confusion matrix</p></a></li>
<li><a href='#coralPval'>
<p>P-values from biological expression array data</p></a></li>
<li><a href='#cottonworkers'><p>Occupation and wage profiles of British cotton workers</p></a></li>
<li><a href='#cricketer'><p>Lifespans of UK 1st class cricketers born 1840-1960</p></a></li>
<li><a href='#cuckoohosts'><p>Comparison of cuckoo eggs with host eggs</p></a></li>
<li><a href='#cuckoos'><p>Cuckoo Eggs Data</p></a></li>
<li><a href='#CVbinary'><p>Cross-Validation for Regression with a Binary Response</p></a></li>
<li><a href='#CVlm'><p>Cross-Validation for Linear Regression</p></a></li>
<li><a href='#DAAGtheme'>
<p>Function to generate lattice themes for graphs.</p></a></li>
<li><a href='#DAAGxdb'><p>List, each of whose elements hold rows of a file, in character format</p></a></li>
<li><a href='#datafile'><p>Write an ASCII data file to the working directory.</p></a></li>
<li><a href='#dengue'><p>Dengue prevalence, by administrative region</p></a></li>
<li><a href='#dewpoint'><p>Dewpoint Data</p></a></li>
<li><a href='#droughts'><p>Periods Between Rain Events</p></a></li>
<li><a href='#edcCO2'><p>EPICA Dome C Ice Core 800KYr Carbon Dioxide Data</p></a></li>
<li><a href='#edcT'><p>EPICA Dome C Ice Core 800KYr Temperature Estimates</p></a></li>
<li><a href='#elastic1'><p>Elastic Band Data Replicated</p></a></li>
<li><a href='#elasticband'><p>Elastic Band Data</p></a></li>
<li><a href='#errorsINseveral'>
<p>Simulation of classical errors in x model, with multiple</p>
explanatory variables.</a></li>
<li><a href='#errorsINx'><p>Simulate data for straight line regression, with &quot;errors in x&quot;.</p></a></li>
<li><a href='#excessRisk'><p>Create and analyze multiway frequency or weighted frequency</p>
table</a></li>
<li><a href='#fossilfuel'><p>Fossil Fuel Data</p></a></li>
<li><a href='#frogs'><p>Frogs Data</p></a></li>
<li><a href='#frostedflakes'><p>Frosted Flakes data</p></a></li>
<li><a href='#fruitohms'><p>Electrical Resistance of Kiwi Fruit</p></a></li>
<li><a href='#gaba'><p>Effect of pentazocine on post-operative pain (average VAS scores)</p></a></li>
<li><a href='#geophones'><p>Seismic Timing Data</p></a></li>
<li><a href='#greatLakes'>
<p>Yearly averages of Great Lake heights: 1918 - 2009</p></a></li>
<li><a href='#grog'><p>Alcohol consumption in Australia and New Zealand</p></a></li>
<li><a href='#hardcopy'><p>Graphical Output for Hardcopy</p></a></li>
<li><a href='#headInjury'><p>Minor Head Injury (Simulated) Data</p></a></li>
<li><a href='#hills'><p>Scottish Hill Races Data</p></a></li>
<li><a href='#hotspots'><p>Hawaian island chain hotspot Potassium-Argon ages</p></a></li>
<li><a href='#hotspots2006'><p>Hawaian island chain hotspot Argon-Argon ages</p></a></li>
<li><a href='#houseprices'><p>Aranda House Prices</p></a></li>
<li><a href='#humanpower'><p>Oxygen uptake versus mechanical power, for humans</p></a></li>
<li><a href='#hurricNamed'>
<p>Named US Atlantic Hurricanes</p></a></li>
<li><a href='#intersalt'><p>Blood pressure versus Salt; inter-population data</p></a></li>
<li><a href='#ironslag'><p>Iron Content Measurements</p></a></li>
<li><a href='#jobs'><p>Canadian Labour Force Summary Data (1995-96)</p></a></li>
<li><a href='#kiwishade'><p>Kiwi Shading Data</p></a></li>
<li><a href='#leafshape'><p>Full Leaf Shape Data Set</p></a></li>
<li><a href='#leaftemp'><p>Leaf and Air Temperature Data</p></a></li>
<li><a href='#leaftemp.all'><p>Full Leaf and Air Temperature Data Set</p></a></li>
<li><a href='#litters'><p>Mouse Litters</p></a></li>
<li><a href='#lmdiags'>
<p>Return data required for diagnostic plots</p></a></li>
<li><a href='#logisticsim'><p>Simple Logistic Regression Data Simulator</p></a></li>
<li><a href='#Lottario'><p>Ontario Lottery Data</p></a></li>
<li><a href='#lung'><p>Cape Fur Seal Lung Measurements</p></a></li>
<li><a href='#Manitoba.lakes'><p>The Nine Largest Lakes in Manitoba</p></a></li>
<li><a href='#mdbAVtJtoD'>
<p>Murray-Darling basin monthly temperatures</p></a></li>
<li><a href='#measles'><p>Deaths in London from measles</p></a></li>
<li><a href='#medExpenses'><p>Family Medical Expenses</p></a></li>
<li><a href='#mignonette'><p>Darwin's Wild Mignonette Data</p></a></li>
<li><a href='#milk'><p>Milk Sweetness Study</p></a></li>
<li><a href='#modelcars'><p>Model Car Data</p></a></li>
<li><a href='#monica'><p>WHO Monica Data</p></a></li>
<li><a href='#moths'><p>Moths Data</p></a></li>
<li><a href='#multilap'><p>Data Filtering Function</p></a></li>
<li><a href='#nassCDS'><p>Airbag and other influences on accident fatalities</p></a></li>
<li><a href='#nasshead'><p>Documentation of names of columns in nass9702cor</p></a></li>
<li><a href='#nihills'><p>Record times for Northern Ireland mountain running events</p></a></li>
<li><a href='#nsw74demo'><p>Labour Training Evaluation Data</p></a></li>
<li><a href='#nswdemo'><p>Labour Training Evaluation Data</p></a></li>
<li><a href='#nswpsid1'><p>Labour Training Evaluation Data</p></a></li>
<li><a href='#obounce'><p>Bounce - obsolete</p></a></li>
<li><a href='#oddbooks'><p>Measurements on 12 books</p></a></li>
<li><a href='#onesamp'><p> Paired Sample t-test</p></a></li>
<li><a href='#onet.permutation'><p>One Sample Permutation t-test</p></a></li>
<li><a href='#onetPermutation'><p>One Sample Permutation t-test</p></a></li>
<li><a href='#onewayPlot'><p>Display of One Way Analysis Results</p></a></li>
<li><a href='#orings'><p>Challenger O-rings Data</p></a></li>
<li><a href='#overlapDensity'><p>Overlapping Density Plots</p></a></li>
<li><a href='#ozone'><p>Ozone Data</p></a></li>
<li><a href='#pair65'><p>Heated Elastic Bands</p></a></li>
<li><a href='#panel.corr'><p>Scatterplot Panel</p></a></li>
<li><a href='#panelCorr'><p>Scatterplot Panel</p></a></li>
<li><a href='#panelplot'><p>Panel Plot</p></a></li>
<li><a href='#pause'>
<p>Pause before continuing execution</p></a></li>
<li><a href='#plotSampDist'><p>Plot(s) of simulated sampling distributions</p></a></li>
<li><a href='#plotSimDiags'>
<p>Diagnostic plots for simulated data</p></a></li>
<li><a href='#plotSimScat'>
<p>Simulate scatterplots, from <code>lm</code> object with a single explanatory variable.</p></a></li>
<li><a href='#poissonsim'><p>Simple Poisson Regression Data Simulator</p></a></li>
<li><a href='#possum'><p>Possum Measurements</p></a></li>
<li><a href='#possumsites'><p>Possum Sites</p></a></li>
<li><a href='#powerplot'><p>Plot of Power Functions</p></a></li>
<li><a href='#poxetc'><p>Deaths from various causes, in London from 1629 till 1881, with gaps</p></a></li>
<li><a href='#press'><p>Predictive Error Sum of Squares</p></a></li>
<li><a href='#primates'><p>Primate Body and Brain Weights</p></a></li>
<li><a href='#progression'><p>Progression of Record times for track races, 1912 - 2008</p></a></li>
<li><a href='#qreference'><p>Normal QQ Reference Plot</p></a></li>
<li><a href='#races2000'><p>Scottish Hill Races Data - 2000</p></a></li>
<li><a href='#rainforest'><p>Rainforest Data</p></a></li>
<li><a href='#rareplants'><p>Rare and Endangered Plant Species</p></a></li>
<li><a href='#rice'><p>Genetically Modified and Wild Type Rice Data</p></a></li>
<li><a href='#rockArt'><p>Pacific Rock Art features</p></a></li>
<li><a href='#roller'><p>Lawn Roller Data</p></a></li>
<li><a href='#sampdist'>
<p>Plot sampling distribution of mean or other sample statistic.</p></a></li>
<li><a href='#science'><p>School Science Survey Data</p></a></li>
<li><a href='#seedrates'><p>Barley Seeding Rate Data</p></a></li>
<li><a href='#show.colors'><p>Show R's Colors</p></a></li>
<li><a href='#simulateLinear'><p>Simulation of Linear Models for ANOVA vs. Regression Comparison</p></a></li>
<li><a href='#simulateSampDist'><p>Simulated sampling distribution of mean or other statistic</p></a></li>
<li><a href='#socsupport'><p>Social Support Data</p></a></li>
<li><a href='#softbacks'><p>Measurements on a Selection of Paperback Books</p></a></li>
<li><a href='#sorption'><p>sorption data set</p></a></li>
<li><a href='#SP500close'><p>Closing Numbers for S and P 500 Index</p></a></li>
<li><a href='#SP500W90'><p>Closing Numbers for S and P 500 Index - First 100 Days of 1990</p></a></li>
<li><a href='#spam7'><p>Spam E-mail Data</p></a></li>
<li><a href='#stVincent'><p>Averages by block of yields for the St. Vincent Corn data</p></a></li>
<li><a href='#sugar'><p>Sugar Data</p></a></li>
<li><a href='#sumry'>
<p>A more flexible alternatives to <code>summary</code>.</p></a></li>
<li><a href='#sumry.glm'><p>Summarizing Generalized Linear Model Fits</p></a></li>
<li><a href='#tinting'><p>Car Window Tinting Experiment Data</p></a></li>
<li><a href='#tomato'><p>Root weights of tomato plants exposed to 4 different treatments</p></a></li>
<li><a href='#toycars'><p>Toy Cars Data</p></a></li>
<li><a href='#two65'><p>Unpaired Heated Elastic Bands</p></a></li>
<li><a href='#twot.permutation'><p>Two Sample Permutation Test - Obsolete</p></a></li>
<li><a href='#twotPermutation'><p>Two Sample Permutation Test</p></a></li>
<li><a href='#vif'><p>Variance Inflation Factors</p></a></li>
<li><a href='#vince111b'><p>Averages by block of corn yields, for treatment 111 only</p></a></li>
<li><a href='#vlt'><p>Video Lottery Terminal Data</p></a></li>
<li><a href='#wages1833'><p>Wages of Lancashire Cotton Factory Workers in 1833</p></a></li>
<li><a href='#whoops'><p>Deaths from whooping cough, in London</p></a></li>
<li><a href='#worldRecords'><p>Record times for track and road races, at August 9th 2006</p></a></li>
<li><a href='#zzDAAGxdb'><p>List, each of whose elements hold rows of a file, in character format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.25.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-13</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Analysis and Graphics Data and Functions</td>
</tr>
<tr>
<td>Author:</td>
<td>John H Maindonald &lt;john@statsresearch.co.nz&gt; and W. John Braun &lt;john.braun@ubc.ca&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>W. John Braun &lt;john.braun@ubc.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions and data sets used in examples and exercises in the
        text Maindonald, J.H. and Braun, W.J. (2003, 2007, 2010) "Data
        Analysis and Graphics Using R", and in an upcoming Maindonald,
        Braun, Andrews, and Narayan text that builds on this earlier text.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>true</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>lattice, latticeExtra, methods, Rdpack</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Suggests:</td>
<td>leaps, oz, lme4, quantreg, knitr, boot, rpart, randomForest,
MASS, survival, mgcv, rmarkdown, bookdown</td>
</tr>
<tr>
<td>ZipData:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gitlab.com/daagur">https://gitlab.com/daagur</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-14 03:14:45 UTC; peterhall</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-14 07:20:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='DAAG-package'>
The R DAAG Package
</h2><span id='topic+DAAG-package'></span><span id='topic+DAAG'></span>

<h3>Description</h3>

<p>Various data sets and functions used or referred to in the book
Maindonald, J.H. and Braun, W.J. (3rd edn 2010)
&quot;Data Analysis and Graphics Using R&quot;, plus other
selected datasets and functions.
</p>


<h3>Details</h3>

<p>For a list of , use <code>library(help="DAAG")</code>.
</p>


<h3>Author(s)</h3>

<p>Author: John H Maindonald
</p>
<p>Maintainer: W John Braun <a href="mailto:braun@stats.uwo.ca">braun@stats.uwo.ca</a>
</p>

<hr>
<h2 id='ACF1'>Aberrant Crypt Foci in Rat Colons</h2><span id='topic+ACF1'></span>

<h3>Description</h3>

<p>Numbers of aberrant crypt foci (ACF) in the 
section 1 of the colons of 22 rats subjected to a single
dose of the carcinogen azoxymethane (AOM), sacrificed
at 3 different times. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ACF1</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>count</dt><dd><p>Observed number of ACF in section 1 of
each rat colon</p>
</dd>
<dt>endtime</dt><dd><p>Time of sacrifice, in weeks following injection
of AOM</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ranjana P. Bird, Faculty of Human Ecology, University of Manitoba,
Winnipeg, Canada.
</p>


<h3>References</h3>

<p>E.A. McLellan, A. Medline and R.P. Bird.  Dose response and
proliferative characteristics of aberrant crypt foci: putative
preneoplastic lesions in rat colon.  Carcinogenesis, 12(11): 2093-2098, 
1991.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sapply(split(ACF1$count,ACF1$endtime),var)
plot(count ~ endtime, data=ACF1, pch=16)
pause()
print("Poisson Regression - Example 8.3")
ACF.glm0 &lt;- glm(formula = count ~ endtime, family = poisson, data = ACF1)
summary(ACF.glm0)

# Is there a quadratic effect?
pause()

ACF.glm &lt;- glm(formula = count ~ endtime + I(endtime^2),
  family = poisson, data = ACF1)
summary(ACF.glm)

# But is the data really Poisson?  If not, try quasipoisson:
pause()

ACF.glm &lt;- glm(formula = count ~ endtime + I(endtime^2),
  family = quasipoisson, data = ACF1)
summary(ACF.glm)
</code></pre>

<hr>
<h2 id='ais'>Australian athletes data set</h2><span id='topic+ais'></span>

<h3>Description</h3>

<p>These data were collected in a study of how data on 
various characteristics of the blood varied with sport,
body size, and sex of the athlete.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ais)</code></pre>


<h3>Format</h3>

<p>A data frame with 202 observations on the following 13 variables.
</p>

<dl>
<dt>rcc</dt><dd><p>red blood cell count, in <code class="reqn">10^{12} l^{-1}</code></p>
</dd>
<dt>wcc</dt><dd><p>while blood cell count, in <code class="reqn">10^{12}</code> per liter</p>
</dd>
<dt>hc</dt><dd><p>hematocrit, percent</p>
</dd>
<dt>hg</dt><dd><p>hemaglobin concentration, in g per decaliter</p>
</dd>
<dt>ferr</dt><dd><p>plasma ferritins, ng <code class="reqn">dl^{-1}</code></p>
</dd>
<dt>bmi</dt><dd><p>Body mass index, kg <code class="reqn">cm^{-2} 10^2</code></p>
</dd>
<dt>ssf</dt><dd><p>sum of skin folds</p>
</dd>
<dt>pcBfat</dt><dd><p>percent Body fat</p>
</dd>
<dt>lbm</dt><dd><p>lean body mass, kg</p>
</dd>
<dt>ht</dt><dd><p>height, cm</p>
</dd>
<dt>wt</dt><dd><p>weight, kg</p>
</dd>
<dt>sex</dt><dd><p>a factor with levels <code>f</code> <code>m</code></p>
</dd>
<dt>sport</dt><dd><p>a factor with levels <code>B_Ball</code> <code>Field</code> 
<code>Gym</code> <code>Netball</code> <code>Row</code> <code>Swim</code> <code>T_400m</code> 
<code>T_Sprnt</code> <code>Tennis</code> <code>W_Polo</code></p>
</dd>
</dl>



<h3>Details</h3>

<p>Do blood hemoglobin concentrations of athletes in endurance-related
events differ from those in power-related events?
</p>


<h3>Source</h3>

<p>These data were the basis for the analyses that are reported in
Telford and Cunningham (1991).
</p>


<h3>References</h3>

<p>Telford, R.D. and Cunningham, R.B. 1991.  Sex, sport and
body-size dependency of hematology in highly trained athletes.
Medicine and Science in Sports and Exercise 23: 788-794.
</p>

<hr>
<h2 id='align2D'>
Function to align points from ordination with known locations
</h2><span id='topic+align2D'></span>

<h3>Description</h3>

<p>Find the linear transformation which, applied to one set of points
in the ($x$, $y$) plane, gives the best match in a least squares
sense to a second set of points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align2D(lat, long, x1, x2, wts=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="align2D_+3A_lat">lat</code></td>
<td>

<p>Latitude or other co-ordinate of point to align to
</p>
</td></tr>
<tr><td><code id="align2D_+3A_long">long</code></td>
<td>

<p>Longitude or other co-ordinate of point to align to
</p>
</td></tr>
<tr><td><code id="align2D_+3A_x1">x1</code></td>
<td>

<p>First coordinate of point to align
</p>
</td></tr>
<tr><td><code id="align2D_+3A_x2">x2</code></td>
<td>

<p>First coordinate of point to align
</p>
</td></tr>
<tr><td><code id="align2D_+3A_wts">wts</code></td>
<td>

<p>If non-NULL, specifies weights for the points.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Achieves the best match, in a least squares sense, between an ordination
and known locations in two-dimensionaL space.</p>


<h3>Value</h3>

<table>
<tr><td><code>fitlat</code></td>
<td>
<p>Fitted values of <code>lat</code></p>
</td></tr>
<tr><td><code>fitlong</code></td>
<td>
<p>Fitted values of <code>long</code></p>
</td></tr>
<tr><td><code>lat</code></td>
<td>
<p>Input values of <code>lat</code></p>
</td></tr>
<tr><td><code>long</code></td>
<td>
<p>Input values of <code>long</code></p>
</td></tr>
</table>


<h3>Note</h3>

<p>An ordination that is designed to reproduce distances between
points is specified only to within an arbitrary rotation about
the centroid.  What linear transformation of the points ($x1$, $x2$)
given by the ordination gives the best match to the known
co-ordinates?
</p>


<h3>Author(s)</h3>

<p>John H Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(DAAG)&amp;require(oz)){
aupts &lt;- cmdscale(audists)
xy &lt;- align2D(lat = aulatlong$latitude, long = aulatlong$longitude,
              x1 = aupts[, 1], x2 = aupts[, 2], wts = NULL)
oz()
fitcoords &lt;- align2D(lat=aulatlong$latitude,
                      long=aulatlong$longitude,
                      x1=aupts[,1], x2 = aupts[,2],
                      wts=NULL)
x &lt;-with(fitcoords,
         as.vector(rbind(lat, fitlat, rep(NA,length(lat)))))
y &lt;-with(fitcoords,
         as.vector(rbind(long, fitlong, rep(NA,length(long)))))
points(aulatlong, col="red", pch=16, cex=1.5)
lines(x, y, col="gray40", lwd=3)
}

## The function is currently defined as
function(lat, long, x1, x2, wts=NULL){
    ## Get best fit in space of (latitude, longitude)
    if(is.null(wts))wts &lt;- rep(1,length(x1))
    fitlat &lt;- predict(lm(lat ~ x1+x2, weights=wts))
    fitlong &lt;- predict(lm(long ~ x1+x2, weights=wts))
    list(fitlat = fitlat, fitlong=fitlong, lat=lat, long=long)
}
</code></pre>

<hr>
<h2 id='allbacks'>Measurements on a Selection of Books</h2><span id='topic+allbacks'></span>

<h3>Description</h3>

<p>The <code>allbacks</code> data frame gives measurements
on the volume and weight of 15 books, some of which
are softback (pb) and some of which are hardback (hb).  Area
of the hardback covers is also included.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allbacks</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>volume</dt><dd><p>book volumes
in cubic centimeters</p>
</dd>
<dt>area</dt><dd><p>hard board cover
areas in square centimeters</p>
</dd>
<dt>weight</dt><dd><p>book weights in grams</p>
</dd>
<dt>cover</dt><dd><p>a factor with levels
<code>hb</code> hardback,   <code>pb</code> paperback</p>
</dd>
</dl>



<h3>Source</h3>

<p>The bookshelf of J. H. Maindonald.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Multiple Regression - Example 6.1")
attach(allbacks)
volume.split &lt;- split(volume, cover)
weight.split &lt;- split(weight, cover)
plot(weight.split$hb ~ volume.split$hb, pch=16, xlim=range(volume), ylim=range(weight),
     ylab="Weight (g)", xlab="Volume (cc)")
points(weight.split$pb ~ volume.split$pb, pch=16, col=2)
pause()

allbacks.lm &lt;- lm(weight ~ volume+area)
summary(allbacks.lm)
detach(allbacks)
pause()

anova(allbacks.lm)
pause()

model.matrix(allbacks.lm)
pause()

print("Example 6.1.1")
allbacks.lm0 &lt;- lm(weight ~ -1+volume+area, data=allbacks)
summary(allbacks.lm0)
pause()

print("Example 6.1.2")
oldpar &lt;- par(mfrow=c(2,2))
plot(allbacks.lm0)
par(oldpar)
allbacks.lm13 &lt;- lm(weight ~ -1+volume+area, data=allbacks[-13,])
summary(allbacks.lm13)
pause()

print("Example 6.1.3")
round(coef(allbacks.lm0),2)  # Baseline for changes
round(lm.influence(allbacks.lm0)$coef,2)
</code></pre>

<hr>
<h2 id='anesthetic'>Anesthetic Effectiveness</h2><span id='topic+anesthetic'></span>

<h3>Description</h3>

<p>Thirty patients were given an anesthetic agent maintained
at a predetermined level (conc) for 15 minutes before making
an incision.  It was then noted whether the patient
moved, i.e. jerked or twisted.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anesthetic</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>move</dt><dd><p>a binary numeric vector coded for 
patient movement (0 = no movement, 1 = movement)</p>
</dd>
<dt>conc</dt><dd><p>anesthetic concentration</p>
</dd>
<dt>logconc</dt><dd><p>logarithm of concentration</p>
</dd>
<dt>nomove</dt><dd><p>the complement of move</p>
</dd>
</dl>



<h3>Details</h3>

<p>The interest is in estimating
how the probability of jerking or twisting varies with
increasing concentration of the anesthetic agent.
</p>


<h3>Source</h3>

<p>unknown
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Logistic Regression - Example 8.1.4")

z &lt;- table(anesthetic$nomove, anesthetic$conc)
tot &lt;- apply(z, 2, sum)         # totals at each concentration
prop &lt;- z[2,  ]/(tot)           # proportions at each concentration
oprop &lt;- sum(z[2,  ])/sum(tot)  # expected proportion moving if concentration had no effect
conc &lt;- as.numeric(dimnames(z)[[2]])
plot(conc, prop, xlab = "Concentration", ylab = "Proportion", xlim = c(.5,2.5),
    ylim = c(0, 1), pch = 16)
chw &lt;- par()$cxy[1]
text(conc - 0.75 * chw, prop, paste(tot), adj = 1)
abline(h = oprop, lty = 2)

pause()

anes.logit &lt;- glm(nomove ~ conc, family = binomial(link = logit),
  data = anesthetic)
anova(anes.logit)
summary(anes.logit)

</code></pre>

<hr>
<h2 id='antigua'>Averages by block of yields for the Antigua Corn data</h2><span id='topic+antigua'></span><span id='topic+ant111b'></span>

<h3>Description</h3>

<p>These data frames have yield averages by blocks (parcels). The
<code>ant111b</code>dataset is a subset that has block averages of
corn yields for treatment 111 only
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(antigua)
  data(ant111b)
  </code></pre>


<h3>Format</h3>

<p> A data frame with 324 observations on 7 variables.
</p>

<dl>
<dt>id</dt><dd><p>a numeric vector</p>
</dd>
<dt>site</dt><dd><p>a factor with 8 levels.</p>
</dd>
<dt>block</dt><dd><p>a factor with levels <code>I</code> <code>II</code> <code>III</code> <code>IV</code></p>
</dd>
<dt>plot</dt><dd><p>a numeric vector</p>
</dd>
<dt>trt</dt><dd><p>a factor consisting of 12 levels</p>
</dd>
<dt>ears</dt><dd><p>a numeric vector; note that -9999 is used as a missing value code.</p>
</dd>
<dt>harvwt</dt><dd><p>a numeric vector; the average yield</p>
</dd>
</dl>



<h3>Source</h3>

<p>Andrews DF; Herzberg AM, 1985. Data. A Collection of Problems from
Many Fields for the Student and Research Worker. Springer-Verlag.
(pp. 339-353)
</p>

<hr>
<h2 id='appletaste'>Tasting experiment that compared four apple varieties</h2><span id='topic+appletaste'></span>

<h3>Description</h3>

<p>Each of 20 tasters each assessed three out of the four varieties.
The experiment was conducted according to a balanced incomplete
block design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(appletaste)</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on the following 3 variables.
</p>

<dl>
<dt>aftertaste</dt><dd><p>a numeric vector</p>
</dd></dl>
<p> Apple samples were rated for
<code>aftertaste</code>, by making a mark on a continuous scale that
ranged from 0 (extreme dislike) to 150 (like very much).
</p>
<dl>
<dt>panelist</dt><dd><p>a factor with levels <code>a</code> <code>b</code> <code>c</code> <code>d</code> <code>e</code> <code>f</code> <code>g</code> <code>h</code> <code>i</code> <code>j</code> <code>k</code> <code>l</code> <code>m</code> <code>n</code> <code>o</code> <code>p</code> <code>q</code> <code>r</code> <code>s</code> <code>t</code></p>
</dd>
<dt>product</dt><dd><p>a factor with levels <code>298</code> <code>493</code> <code>649</code> <code>937</code></p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(appletaste)
appletaste.aov &lt;- aov(aftertaste ~ panelist + product, data=appletaste)
termplot(appletaste.aov)
</code></pre>

<hr>
<h2 id='audists'>Road distances between 10 Australian cities</h2><span id='topic+audists'></span>

<h3>Description</h3>

<p>Distances between the Australian cities of
Adelaide, Alice, Brisbane, Broome, Cairns, Canberra, 
Darwin, Melbourne, Perth and Sydney   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>audists</code></pre>


<h3>Format</h3>

<p>The format is:
Class 'dist', i.e., a distance matrix.
</p>


<h3>Source</h3>

<p>Australian road map
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(audists)
## Not run: 
audists.cmd &lt;- cmdscale(audists)
library(lattice)
xyplot(audists.cmd[,2] ~ audists.cmd[,1], 
       groups=row.names(audists.cmd),
       panel = function(x, y, subscripts, groups)  
                        ltext(x = x, y = y, label = groups[subscripts],
                        cex=1, fontfamily = "HersheySans"))

## End(Not run)
</code></pre>

<hr>
<h2 id='aulatlong'>Latitudes and longitudes for ten Australian cities</h2><span id='topic+aulatlong'></span>

<h3>Description</h3>

<p>Latitudes and longitudes for
Adelaide, Alice, Brisbane, Broome, Cairns, Canberra, 
Darwin, Melbourne, Perth and Sydney; i.e., for the cities to which the
road distances in <code>audists</code> relate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aulatlong</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 2 variables.
</p>

<dl>
<dt><code>latitude</code></dt><dd><p>Latitude, as a decimal number</p>
</dd>
<dt><code>longitude</code></dt><dd><p>Latitude, as a decimal number</p>
</dd>
</dl>



<h3>Source</h3>

<p>Map of Australia showing latitude and longitude information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(aulatlong)
## maybe str(aulatlong) ; plot(aulatlong) ...
</code></pre>

<hr>
<h2 id='austpop'>Population figures for Australian States and Territories</h2><span id='topic+austpop'></span>

<h3>Description</h3>

<p>Population figures for Australian states and territories for 1917, 1927,
..., 1997.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>austpop</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>year</dt><dd><p>a numeric vector</p>
</dd>
<dt>NSW</dt><dd><p>New South Wales population counts </p>
</dd>
<dt>Vic</dt><dd><p>Victoria population counts</p>
</dd>
<dt>Qld</dt><dd><p>Queensland population counts</p>
</dd>
<dt>SA</dt><dd><p>South Australia population counts</p>
</dd>
<dt>WA</dt><dd><p>Western Australia population counts</p>
</dd>
<dt>Tas</dt><dd><p>Tasmania population counts</p>
</dd>
<dt>NT</dt><dd><p>Northern Territory population 
counts</p>
</dd>
<dt>ACT</dt><dd><p>Australian Capital Territory 
population counts</p>
</dd>
<dt>Aust</dt><dd><p>Population counts for
the whole country</p>
</dd>
</dl>



<h3>Source</h3>

<p>Australian Bureau of Statistics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Looping - Example 1.7")

growth.rates &lt;- numeric(8)
for (j in seq(2,9)) {
    growth.rates[j-1] &lt;- (austpop[9, j]-austpop[1, j])/austpop[1, j] }
growth.rates &lt;- data.frame(growth.rates)
row.names(growth.rates) &lt;- names(austpop[c(-1,-10)])
  # Note the use of row.names() to name the rows of the data frame
growth.rates

pause()
print("Avoiding Loops - Example 1.7b")

sapply(austpop[,-c(1,10)], function(x){(x[9]-x[1])/x[1]})

pause()
print("Plot - Example 1.8a")
attach(austpop)
plot(year, ACT, type="l") # Join the points ("l" = "line")
detach(austpop)

pause()
print("Exerice 1.12.9")
attach(austpop)
oldpar &lt;- par(mfrow=c(2,4))  
for (i in 2:9){
plot(austpop[,1], log(austpop[, i]), xlab="Year",
    ylab=names(austpop)[i], pch=16, ylim=c(0,10))}
par(oldpar) 
detach(austpop)

</code></pre>

<hr>
<h2 id='bestsetNoise'>Best Subset Selection Applied to Noise</h2><span id='topic+bestsetNoise'></span><span id='topic+bestset.noise'></span><span id='topic+bsnCV'></span><span id='topic+bsnOpt'></span><span id='topic+bsnVaryNvar'></span>

<h3>Description</h3>

<p>Best subset selection applied to completely random noise.  This
function demonstrates how variable selection techniques in
regression can often err in including explanatory variables that
are indistinguishable from noise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bestsetNoise(m = 100, n = 40, method = "exhaustive", nvmax = 3,
              X = NULL, y=NULL, intercept=TRUE,
              print.summary = TRUE, really.big = FALSE, ...)

bestset.noise(m = 100, n = 40, method = "exhaustive", nvmax = 3,
              X = NULL, y=NULL, intercept=TRUE,
              print.summary = TRUE, really.big = FALSE, ...)

bsnCV(m = 100, n = 40, method = "exhaustive", nvmax = 3,
              X = NULL, y=NULL, intercept=TRUE, nfolds = 2,
              print.summary = TRUE, really.big = FALSE)

bsnOpt(X = matrix(rnorm(25 * 10), ncol = 10), y = NULL, method = "exhaustive",
              nvmax = NULL, nbest = 1, intercept = TRUE, criterion = "cp",
              tcrit = NULL, print.summary = TRUE, really.big = FALSE,
         ...)

bsnVaryNvar(m = 100, nvar = nvmax:50, nvmax = 3, method = "exhaustive",
              intercept=TRUE,
              plotit = TRUE, xlab = "# of variables from which to select",
              ylab = "p-values for t-statistics", main = paste("Select 'best'",
                                                  nvmax, "variables"),
              details = FALSE, really.big = TRUE, smooth = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bestsetNoise_+3A_m">m</code></td>
<td>
<p>the number of observations to be simulated, ignored if X is supplied.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_n">n</code></td>
<td>
<p>the number of predictor variables in the simulated
model, ignored if X is supplied.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_method">method</code></td>
<td>
<p>Use <code>exhaustive</code> search, or <code>backward</code> selection,
or <code>forward</code> selection, or <code>sequential</code> replacement.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_nvmax">nvmax</code></td>
<td>
<p>Number of explanatory variables in model.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_x">X</code></td>
<td>
<p>Use columns from this matrix.  Alternatively, X may be a
data frame, in which case a model matrix will be formed from it.
If not <code>NULL</code>, <code>m</code> and <code>n</code> are ignored.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_y">y</code></td>
<td>
<p>If not supplied, random normal noise will be generated.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_nbest">nbest</code></td>
<td>
<p>Number of models, for each choice of number of columns
of explanatory variables, to return (<code>bsnOpt</code>). If <code>tcrit</code>
is non-NULL, it may be important to set this greater than one, in
order to have a good chance of finding models with minimum absolute
<em>t</em>-statistic greater than <code>tcrit</code>.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_intercept">intercept</code></td>
<td>
<p>Should an intercept be added?</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_nvar">nvar</code></td>
<td>
<p>range of number of candidate variables (<code>bsnVaryVvar</code>).</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_nfolds">nfolds</code></td>
<td>
<p>For splitting the data into training and text sets,
the number of folds.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_criterion">criterion</code></td>
<td>
<p>Criterion to use in choosing between models with
different numbers of explanatory variables (<code>bsnOpt</code>).
Alternatives are &ldquo;bic&rdquo;, or &ldquo;cip&rdquo; or &ldquo;adjr2&rdquo;.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_tcrit">tcrit</code></td>
<td>
<p>Consider only those models for which the minimum absolute
<em>t</em>-statistic is greater than <code>tcrit</code>.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_print.summary">print.summary</code></td>
<td>
<p>Should summary information be printed.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_plotit">plotit</code></td>
<td>

<p>Plot a graph? (<code>bsnVaryVvar</code>)</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_xlab">xlab</code></td>
<td>

<p><em>x</em>-label for graph (<code>bsnVaryVvar</code>)
</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_ylab">ylab</code></td>
<td>

<p><em>y</em>-label for graph (<code>bsnVaryVvar</code>.)
</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_main">main</code></td>
<td>

<p>main title for graph (<code>bsnVaryVvar</code>.)
</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_details">details</code></td>
<td>
<p>Return detailed output list (<code>bsnVaryVvar</code>)</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_really.big">really.big</code></td>
<td>
<p>Set to <code>TRUE</code> to allow (currently) for more than
50 explanatory variables.</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_smooth">smooth</code></td>
<td>
<p>Fit smooth to graph? (<code>bsnVaryVvar</code>).</p>
</td></tr>
<tr><td><code id="bestsetNoise_+3A_...">...</code></td>
<td>
<p>Additional arguments, to be passed through to
<code>regsubsets()</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>X</code> is not supplied, and in any case for <code>bsnVaryNvar</code>, a
set of <code>n</code> predictor variables are simulated as independent
standard normal, i.e. N(0,1), variates.  Additionally a N(0,1) response
variable is simulated.  The function <code>bsnOpt</code> selects the
&lsquo;best&rsquo; model with <code>nvmax</code> or fewer explanatory variables,
where the argument <code>criterion</code> specifies the criterion that will
be used to choose between models with different numbers of explanatory
columns.  Other functions select the &lsquo;best&rsquo; model with
<code>nvmax</code> explanatory columns.  In any case, the selection is made
using the <code>regsubsets()</code> function from the leaps package.
(The leaps package must be installed for this function to work.)
</p>
<p>The function <code>bsnCV</code> splits the data (randomly) into <code>nfolds</code>
(2 or more) parts.  It puts each part aside in turn for use to fit
the model (effectively, test data), with the remaining data used
for selecting the variables that will be used for fitting. One model
fit is returned for each of the <code>nfolds</code> parts.
</p>
<p>The function <code>bsnVaryVvar</code> makes repeated calls to
<code>bestsetNoise</code>
</p>


<h3>Value</h3>

<p><code>bestsetNoise</code> returns the <code>lm</code> model object for the &quot;best&quot;
model with <code>nvmax</code> explanatory columns.
</p>
<p><code>bsnCV</code> returns as many models as there are folds.
</p>
<p><code>bsnVaryVvar</code> silently returns either (<code>details=FALSE</code>) a
matrix that has <em>p</em>-values of the coefficients for the &lsquo;best&rsquo;
choice of model for
each different number of candidate variables, or
(<code>details=TRUE</code>) a list with elements:
</p>
<table>
<tr><td><code>coef</code></td>
<td>
<p>A matrix of sets of regression coefficients</p>
</td></tr>
<tr><td><code>SE</code></td>
<td>
<p>A matrix of standard errors</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>A matrix of <em>p</em>-values</p>
</td></tr>
</table>
<p>Matrices have one row for each choice of <code>nvar</code>.  The statistics
returned are for the &lsquo;best&rsquo; model with nvmax explanatory
variables.
</p>
<p><code>bsnOpt</code> silently returns a list with elements:
</p>
<table>
<tr><td><code>u1</code></td>
<td>
<p>&lsquo;best&rsquo; model (<code>lm</code> object) with <code>nvmax</code> or
fewer columns of predictors. If <code>tcrit</code> is non-NULL, and there
is no model for which all coefficients have <em>t</em>-statistics
less than <code>tcrit</code> in absolute value, <code>u1</code> will be NULL.</p>
</td></tr>
</table>
<p><code>tcrit</code>For each model, the minimum of the absolute values of
the <em>t</em>-statistics.
<code>regsubsets_obj</code>The object returned by the call to <code>regsubsets</code>.
</p>


<h3>Note</h3>

<p>These functions are primarily designed to demonstrate the biases
that can be expected, relative to theoretical estimates of standard
errors of parameters and other fitted model statistics, when there
is prior selection of the columns that are to be included in the
model. With the exception of <code>bsnVaryNvar</code>, they can also be
used with an <code>X</code> and <code>y</code> for actual data.  In that case,
the <em>p</em>-values should be compared with those
obtained from repeated use of the function where <code>y</code> is random
noise, as a check on the extent of selection effects.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+lm">lm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>leaps.out &lt;- try(require(leaps, quietly=TRUE))
leaps.out.log &lt;- is.logical(leaps.out)
if ((leaps.out.log==TRUE)&amp;(leaps.out==TRUE)){
bestsetNoise(20,6) # `best' 3-variable regression for 20 simulated observations
                   # on 7 unrelated variables (including the response)
bsnCV(20,6) # `best' 3-variable regressions (one for each fold) for 20
                   # simulated observations on 7 unrelated variables
                   # (including the response)
bsnVaryNvar(m = 50, nvar = 3:6, nvmax = 3, method = "exhaustive",
            plotit=FALSE, details=TRUE)
bsnOpt()
}
</code></pre>

<hr>
<h2 id='biomass'>Biomass Data</h2><span id='topic+biomass'></span>

<h3>Description</h3>

<p>The <code>biomass</code> data frame has 135 rows and 8 columns.  The
<code>rainforest</code> data frame is a subset of this one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomass</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>dbh</dt><dd><p>a numeric vector</p>
</dd>
<dt>wood</dt><dd><p>a numeric vector</p>
</dd>
<dt>bark</dt><dd><p>a numeric vector</p>
</dd>
<dt>fac26</dt><dd><p>a factor with 3 levels</p>
</dd>
<dt>root</dt><dd><p>a numeric vector</p>
</dd>
<dt>rootsk</dt><dd><p>a numeric vector</p>
</dd>
<dt>branch</dt><dd><p>a numeric vector</p>
</dd>
<dt>species</dt><dd><p>a factor with levels
<code>Acacia mabellae</code>,
<code>C. fraseri</code>,
<code>Acmena smithii</code>,
<code>B. myrtifolia</code> </p>
</dd>
</dl>



<h3>Source</h3>

<p>J. Ash, Australian National University
</p>


<h3>References</h3>

<p>Ash, J. and Helman, C. (1990) Floristics and vegetation
biomass of a forest catchment, Kioloa, south coastal N.S.W.
Cunninghamia, 2: 167-182.
</p>

<hr>
<h2 id='bomregions2021'>Australian and Related Historical Annual Climate Data, by Region</h2><span id='topic+bomregions2021'></span><span id='topic+bomregions2018'></span><span id='topic+bomregions'></span>

<h3>Description</h3>

<p>Australian regional temperature data, Australian regional rainfall
data, and Annual SOI, are given for the years 1900-2021.  The regional
rainfall and temperature data are area-weighted averages for the 
respective regions.  The Southern Oscillation Index (SOI) is the 
difference in barometric pressure at sea level between Tahiti and Darwin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("bomregions2021")</code></pre>


<h3>Format</h3>

<p>These data frames contains the following columns:
</p>

<dl>
<dt>Year</dt><dd><p>Year</p>
</dd>
<dt>seAVt</dt><dd><p>Southeastern region average temperature (degrees C)</p>
</dd>
<dt>southAVt</dt><dd><p>Southern temperature</p>
</dd>
<dt>eastAVt</dt><dd><p>Eastern temperature</p>
</dd>
<dt>northAVt</dt><dd><p>Northern temperature</p>
</dd>
<dt>swAVt</dt><dd><p>Southwestern temperature</p>
</dd>
<dt>qldAVt</dt><dd><p> temperature</p>
</dd>
<dt>nswAVt</dt><dd><p> temperature</p>
</dd>
<dt>ntAVt</dt><dd><p> temperature</p>
</dd>
<dt>saAVt</dt><dd><p> temperature</p>
</dd>
<dt>tasAVt</dt><dd><p> temperature</p>
</dd>
<dt>vicAVt</dt><dd><p> temperature</p>
</dd>
<dt>waAVt</dt><dd><p> temperature</p>
</dd>
<dt>mdbAVt</dt><dd><p>Murray-Darling basin temperature</p>
</dd>
<dt>ausAVt</dt><dd><p>Australian average temperature, area-weighted mean</p>
</dd>
<dt>seRain</dt><dd><p>Southeast Australian annual rainfall (mm)</p>
</dd>
<dt>southRain</dt><dd><p>Southern rainfall</p>
</dd>
<dt>eastRain</dt><dd><p>Eastern rainfall</p>
</dd>
<dt>northRain</dt><dd><p>Northern rainfall</p>
</dd>
<dt>swRain</dt><dd><p>Southwest rainfall</p>
</dd>
<dt>qldRain</dt><dd><p>Queensland rainfall</p>
</dd>
<dt>nswRain</dt><dd><p>NSW rainfall</p>
</dd>
<dt>ntRain</dt><dd><p>Northern Territory rainfall</p>
</dd> 
<dt>saRain</dt><dd><p>South Australian rainfall</p>
</dd> 
<dt>tasRain</dt><dd><p>Tasmanian rainfall</p>
</dd>
<dt>vicRain</dt><dd><p>Victorian rainfall</p>
</dd>
<dt>waRain</dt><dd><p>West Australian rainfall</p>
</dd> 
<dt>mdbRain</dt><dd><p>Murray-Darling basin rainfall</p>
</dd>
<dt>ausRain</dt><dd><p>Australian average rainfall, area weighted</p>
</dd>
<dt>SOI</dt><dd><p>Annual average Southern Oscillation Index</p>
</dd>
<dt>sunspot</dt><dd><p>Yearly mean sunspot number</p>
</dd>
<dt>co2mlo</dt><dd><p>Moana Loa CO2 concentrations, from 1959</p>
</dd>
<dt>co2law</dt><dd><p>Moana Loa CO2 concentrations, 1900 to 1978</p>
</dd>
<dt>CO2</dt><dd><p>CO2 concentrations, composite series</p>
</dd>
<dt>avDMI</dt><dd><p>Annual average Dipole Mode Index, for the Indian Ocean Dipole,
from 1950</p>
</dd>
</dl>



<h3>Source</h3>

<p>Australian Bureau of Meteorology web pages:
</p>
<p><a href="http://www.bom.gov.au/climate/change/index.shtml">http://www.bom.gov.au/climate/change/index.shtml</a>
</p>
<p>Go to website, choose timeseries to display, then click &quot;Download data&quot;
</p>
<p>The SOI data are from <a href="http://www.bom.gov.au/climate/enso/soi_monthly.txt">http://www.bom.gov.au/climate/enso/soi_monthly.txt</a>.
</p>
<p>The CO2 series <code>co2law</code>, for Law Dome ice core data. is from
<a href="https://cdiac.ess-dive.lbl.gov/trends/co2/lawdome-data.html">https://cdiac.ess-dive.lbl.gov/trends/co2/lawdome-data.html</a>.
</p>
<p>The Moana Loa CO2 series <code>co2mlo</code> is from Dr. Pieter Tans, 
NOAA/ESRL (<a href="https://gml.noaa.gov/ccgg/trends/">https://gml.noaa.gov/ccgg/trends/</a>)
</p>
<p>The series <code>CO2</code> is a composite series, obtained by adding 0.46 to
the Law data for 1900 to 1958, then following this with the Moana Loa
data that is avaiable from 1959.  The addition of 0.46 brings the
average of the Law data into agreement with that for the Moana Loa data
for the period 1959 to 1968.
</p>
<p>The yearly mean sunspot number is a subset of one of several sunspot
series that are available from WDC-SILSO, Royal Observatory of Belgium, Brussels.
<a href="https://www.sidc.be/silso/datafiles">https://www.sidc.be/silso/datafiles</a>
</p>
<p>The dipole mode index data are from 
<a href="https://ds.data.jma.go.jp/tcc/tcc/products/elnino/index/Readme_iod.txt">https://ds.data.jma.go.jp/tcc/tcc/products/elnino/index/Readme_iod.txt</a>.
Note also <a href="https://stateoftheocean.osmc.noaa.gov/sur/ind/dmi.php">https://stateoftheocean.osmc.noaa.gov/sur/ind/dmi.php</a>, which has details of several other such series.
</p>


<h3>References</h3>

<p>D.M. Etheridge, L.P. Steele, R.L. Langenfelds, R.J. Francey,
J.-M. Barnola and V.I. Morgan, 1998, <em>Historical CO2 records from the
Law Dome DE08, DE08-2, and DSS ice cores</em>, in Trends: A Compendium of
Data on Global Change, on line at Carbon Dioxide Information Analysis
Center, Oak Ridge National Laboratory, U.S. Department of Energy, Oak
Ridge, Tenn., U.S.A.
</p>
<p>Lavery, B., Joung, G. and Nicholls, N. 1997. An extended high-quality
historical rainfall dataset for Australia. Australian Meteorological
Magazine, 46, 27-38.
</p>
<p>Nicholls, N., Lavery, B., Frederiksen, C.\ and Drosdowsky, W. 1996.
Recent apparent changes in relationships between the El Nino &ndash;
southern oscillation and Australian rainfall and temperature.
Geophysical Research Letters 23: 3357-3360.
</p>
<p>SIDC-team, World Data Center for the Sunspot Index, Royal Observatory of
Belgium, Monthly Report on the International Sunspot Number, online
catalogue of the sunspot index: <a href="https://www.sidc.be/silso/datafiles">https://www.sidc.be/silso/datafiles</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(ts(bomregions2021[, c("mdbRain","SOI")], start=1900),
     panel=function(y,...)panel.smooth(bomregions2021$Year, y,...))
avrain &lt;- bomregions2021[,"mdbRain"]
xbomsoi &lt;- with(bomregions2021, data.frame(Year=Year, SOI=SOI,
                cuberootRain=avrain^0.33))
xbomsoi$trendSOI &lt;- lowess(xbomsoi$SOI, f=0.1)$y
xbomsoi$trendRain &lt;- lowess(xbomsoi$cuberootRain, f=0.1)$y
xbomsoi$detrendRain &lt;-
  with(xbomsoi, cuberootRain - trendRain + mean(trendRain))
xbomsoi$detrendSOI &lt;-
  with(xbomsoi, SOI - trendSOI + mean(trendSOI))
## Plot time series avrain and SOI: ts object xbomsoi
plot(ts(xbomsoi[, c("cuberootRain","SOI")], start=1900),
     panel=function(y,...)panel.smooth(xbomsoi$Year, y,...),
     xlab = "Year", main="", ylim=list(c(250, 800),c(-20,25)))
par(mfrow=c(1,2))
rainpos &lt;- pretty(xbomsoi$cuberootRain^3, 6)
plot(cuberootRain ~ SOI, data = xbomsoi,
     ylab = "Rainfall (cube root scale)", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
mtext(side = 3, line = 0.8, "A", adj = -0.025)
with(xbomsoi, lines(lowess(cuberootRain ~ SOI, f=0.75)))
plot(detrendRain ~ detrendSOI, data = xbomsoi,
     xlab="Detrended SOI", ylab = "Detrended rainfall", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
with(xbomsoi, lines(lowess(detrendRain ~ detrendSOI, f=0.75)))
mtext(side = 3, line = 0.8, "B", adj = -0.025)
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='bomsoi'>Southern Oscillation Index Data</h2><span id='topic+bomsoi'></span>

<h3>Description</h3>

<p>The Southern Oscillation Index (SOI) is the difference in barometric  
pressure at sea level between Tahiti and Darwin.  Annual SOI and
Australian rainfall data, for the years 1900-2005, are given.
Australia's annual mean rainfall is an area-weighted average of the total 
annual precipitation at approximately 370 rainfall stations 
around the country.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bomsoi</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Year</dt><dd><p>a numeric vector</p>
</dd>
<dt>Jan</dt><dd><p>average January SOI values for each year</p>
</dd>
<dt>Feb</dt><dd><p>average February SOI values for each year</p>
</dd>
<dt>Mar</dt><dd><p>average March SOI values for each year</p>
</dd>
<dt>Apr</dt><dd><p>average April SOI values for each year</p>
</dd>
<dt>May</dt><dd><p>average May SOI values for each year</p>
</dd>
<dt>Jun</dt><dd><p>average June SOI values for each year</p>
</dd>
<dt>Jul</dt><dd><p>average July SOI values for each year</p>
</dd>
<dt>Aug</dt><dd><p>average August SOI values for each year</p>
</dd>
<dt>Sep</dt><dd><p>average September SOI values for each year</p>
</dd>
<dt>Oct</dt><dd><p>average October SOI values for each year</p>
</dd>
<dt>Nov</dt><dd><p>average November SOI values for each year</p>
</dd>
<dt>Dec</dt><dd><p>average December SOI values for each year</p>
</dd>
<dt>SOI</dt><dd><p>a numeric vector consisting of average annual SOI
values</p>
</dd>
<dt>avrain</dt><dd><p>a numeric vector consisting of a weighted average annual
rainfall at a large number of Australian sites</p>
</dd>
<dt>NTrain</dt><dd><p>Northern Territory rain</p>
</dd>
<dt>northRain</dt><dd><p>north rain</p>
</dd>
<dt>seRain</dt><dd><p>southeast rain</p>
</dd>
<dt>eastRain</dt><dd><p>east rain</p>
</dd>
<dt>southRain</dt><dd><p>south rain</p>
</dd>
<dt>swRain</dt><dd><p>southwest rain</p>
</dd>
</dl>
 


<h3>Source</h3>

 
<p>Australian Bureau of Meteorology web pages:
</p>
<p>http://www.bom.gov.au/climate/change/rain02.txt and
http://www.bom.gov.au/climate/current/soihtm1.shtml
</p>


<h3>References</h3>

<p>Nicholls, N., Lavery, B., Frederiksen, C.\ and Drosdowsky, W. 1996.
Recent apparent changes in relationships between the El Nino &ndash;
southern oscillation and Australian rainfall and temperature.
Geophysical Research Letters 23: 3357-3360.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
plot(ts(bomsoi[, 15:14], start=1900),
     panel=function(y,...)panel.smooth(1900:2005, y,...))
pause()

# Check for skewness by comparing the normal probability plots for 
# different a, e.g.
par(mfrow = c(2,3))
for (a in c(50, 100, 150, 200, 250, 300))
qqnorm(log(bomsoi[, "avrain"] - a))
  # a = 250 leads to a nearly linear plot

pause()

par(mfrow = c(1,1))
plot(bomsoi$SOI, log(bomsoi$avrain - 250), xlab = "SOI",
     ylab = "log(avrain = 250)")
lines(lowess(bomsoi$SOI)$y, lowess(log(bomsoi$avrain - 250))$y, lwd=2)
  # NB: separate lowess fits against time
lines(lowess(bomsoi$SOI, log(bomsoi$avrain - 250)))
pause()

xbomsoi &lt;-
  with(bomsoi, data.frame(SOI=SOI, cuberootRain=avrain^0.33))
xbomsoi$trendSOI &lt;- lowess(xbomsoi$SOI)$y
xbomsoi$trendRain &lt;- lowess(xbomsoi$cuberootRain)$y
rainpos &lt;- pretty(bomsoi$avrain, 5)
with(xbomsoi,
     {plot(cuberootRain ~ SOI, xlab = "SOI",
           ylab = "Rainfall (cube root scale)", yaxt="n")
     axis(2, at = rainpos^0.33, labels=paste(rainpos))
## Relative changes in the two trend curves
     lines(lowess(cuberootRain ~ SOI))
     lines(lowess(trendRain ~ trendSOI), lwd=2)
  })
pause()

xbomsoi$detrendRain &lt;-
  with(xbomsoi, cuberootRain - trendRain + mean(trendRain))
xbomsoi$detrendSOI &lt;-
  with(xbomsoi, SOI - trendSOI + mean(trendSOI))
oldpar &lt;- par(mfrow=c(1,2), pty="s")
plot(cuberootRain ~ SOI, data = xbomsoi,
     ylab = "Rainfall (cube root scale)", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
with(xbomsoi, lines(lowess(cuberootRain ~ SOI)))
plot(detrendRain ~ detrendSOI, data = xbomsoi,
  xlab="Detrended SOI", ylab = "Detrended rainfall", yaxt="n")
axis(2, at = rainpos^0.33, labels=paste(rainpos))
with(xbomsoi, lines(lowess(detrendRain ~ detrendSOI)))
pause()

par(oldpar)
attach(xbomsoi)
xbomsoi.ma0 &lt;- arima(detrendRain, xreg=detrendSOI, order=c(0,0,0))
# ordinary regression model

xbomsoi.ma12 &lt;- arima(detrendRain, xreg=detrendSOI,
                      order=c(0,0,12))
# regression with MA(12) errors -- all 12 MA parameters are estimated
xbomsoi.ma12
pause()

xbomsoi.ma12s &lt;- arima(detrendRain, xreg=detrendSOI,
                      seasonal=list(order=c(0,0,1), period=12))
# regression with seasonal MA(1) (lag 12) errors -- only 1 MA parameter
# is estimated
xbomsoi.ma12s
pause()

xbomsoi.maSel &lt;- arima(x = detrendRain, order = c(0, 0, 12),
                        xreg = detrendSOI, fixed = c(0, 0, 0,
                        NA, rep(0, 4), NA, 0, NA, NA, NA, NA),
                        transform.pars=FALSE)
# error term is MA(12) with fixed 0's at lags 1, 2, 3, 5, 6, 7, 8, 10
# NA's are used to designate coefficients that still need to be estimated
# transform.pars is set to FALSE, so that MA coefficients are not
# transformed (see help(arima))

detach(xbomsoi)
pause()

Box.test(resid(lm(detrendRain ~ detrendSOI, data = xbomsoi)),
          type="Ljung-Box", lag=20)

pause()

attach(xbomsoi)
 xbomsoi2.maSel &lt;- arima(x = detrendRain, order = c(0, 0, 12),
                         xreg = poly(detrendSOI,2), fixed = c(0,
                         0, 0, NA, rep(0, 4), NA, 0, rep(NA,5)),
                         transform.pars=FALSE)
 xbomsoi2.maSel
qqnorm(resid(xbomsoi.maSel, type="normalized"))
detach(xbomsoi)

</code></pre>

<hr>
<h2 id='bostonc'>Boston Housing Data &ndash; Corrected</h2><span id='topic+bostonc'></span>

<h3>Description</h3>

<p>The corrected Boston housing data (from http://lib.stat.cmu.edu/datasets/).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bostonc</code></pre>


<h3>Format</h3>

<p>A single vector containing the contents of &quot;boston_corrected.txt&quot;. 
</p>


<h3>Source</h3>

<p>Harrison, D. and Rubinfeld, D.L. 'Hedonic prices and the demand for clean air', J. Environ. Economics &amp; Management, vol.5, 81-102, 1978.
corrected by Kelley Pace (kpace@unix1.sncc.lsu.edu)
</p>

<hr>
<h2 id='bounce'>Separate plotting positions for labels, to avoid overlap</h2><span id='topic+bounce'></span>

<h3>Description</h3>

  
<p>Return univariate plotting positions in which neighboring points are
separated, if and as necessary, so that they are the specified minimum
distance apart.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bounce(y, d, log = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bounce_+3A_y">y</code></td>
<td>
<p>A numeric vector of plotting positions</p>
</td></tr>
<tr><td><code id="bounce_+3A_d">d</code></td>
<td>
<p>Minimum required distance between neighboring positions</p>
</td></tr>
<tr><td><code id="bounce_+3A_log">log</code></td>
<td>
<p><code>TRUE</code> if values are will be plotted on a logarithmic scale.</p>
</td></tr>
</table>


<h3>Details</h3>

  
<p>The centroid(s) of groups of points that are moved relative to each
other remain the same.
</p>


<h3>Value</h3>

  
<p>A vector of values such that, when plotted along a line, neighboring
points are the required minimum distance apart.  
</p>


<h3>Note</h3>

<p>If values are plotted on a logarithmic scale, <code>d</code> is the required
distance apart on that scale. If a base other than 10 is required, set
<code>log</code> equal to that base.  (Note that base 10 is the default for
<code>plot</code> with <code>log=TRUE</code>.)
</p>


<h3>Author(s)</h3>

<p>John Maindonald</p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+onewayPlot">onewayPlot</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>bounce(c(4, 1.8, 2, 6), d=.4)
bounce(c(4, 1.8, 2, 6), d=.1, log=TRUE)
</code></pre>

<hr>
<h2 id='capstring'>Converts initial character of a string to upper case</h2><span id='topic+capstring'></span>

<h3>Description</h3>

<p>This function is useful for use before
plotting, if one wants capitalized axis labels or factor
levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>capstring(names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="capstring_+3A_names">names</code></td>
<td>
<p>a character vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with upper case initial values.
</p>


<h3>Author(s)</h3>

<p>W.J. Braun</p>


<h3>Examples</h3>

<pre><code class='language-R'>capstring(names(tinting)[c(3,4)])

library(lattice)
levels(tinting$agegp) &lt;- capstring(levels(tinting$agegp))
xyplot(csoa ~ it | sex * agegp, data=tinting) 
</code></pre>

<hr>
<h2 id='carprice'>US Car Price Data</h2><span id='topic+carprice'></span>

<h3>Description</h3>

<p>U.S. data extracted from <code>Cars93</code>, a data frame in the 
MASS package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carprice</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Type</dt><dd><p>Type of car, e.g. Sporty, Van, Compact</p>
</dd>
<dt>Min.Price</dt><dd><p>Price for a basic model</p>
</dd>
<dt>Price</dt><dd><p>Price for a mid-range model</p>
</dd>
<dt>Max.Price</dt><dd><p>Price for a &lsquo;premium&rsquo; model</p>
</dd>
<dt>Range.Price</dt><dd><p>Difference between Max.Price and Min.Price</p>
</dd>
<dt>RoughRange</dt><dd><p>Rough.Range plus some N(0,.0001) noise</p>
</dd>
<dt>gpm100</dt><dd><p>The number of gallons required to travel 100 miles</p>
</dd>
<dt>MPG.city</dt><dd><p>Average number of miles per gallon for city driving</p>
</dd>
<dt>MPG.highway</dt><dd><p>Average number of miles per gallon for highway 
driving</p>
</dd>
</dl>
 


<h3>Source</h3>

<p>MASS package
</p>


<h3>References</h3>

<p>Venables, W.N.\ and Ripley, B.D., 4th edn 2002. Modern
Applied Statistics with S. Springer, New York.
</p>
<p>See also
&lsquo;R&rsquo; Complements to Modern Applied Statistics with S-Plus,
available from <a href="http://www.stats.ox.ac.uk/pub/MASS3/">http://www.stats.ox.ac.uk/pub/MASS3/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
print("Multicollinearity - Example 6.8")
pairs(carprice[,-c(1,8,9)])

carprice1.lm &lt;- lm(gpm100 ~ Type+Min.Price+Price+Max.Price+Range.Price,
    data=carprice)
round(summary(carprice1.lm)$coef,3)
pause()

alias(carprice1.lm)
pause()

carprice2.lm &lt;- lm(gpm100 ~ Type+Min.Price+Price+Max.Price+RoughRange, data=carprice)
round(summary(carprice2.lm)$coef, 2)
pause()

carprice.lm &lt;- lm(gpm100 ~ Type + Price, data = carprice)
round(summary(carprice.lm)$coef,4)  
pause()

summary(carprice1.lm)$sigma   # residual standard error when fitting all 3 price variables
pause()

summary(carprice.lm)$sigma    # residual standard error when only price is used
pause()

vif(lm(gpm100 ~ Price, data=carprice)) # Baseline Price
pause()

vif(carprice1.lm)    # includes Min.Price, Price &amp; Max.Price
pause()

vif(carprice2.lm)    # includes Min.Price, Price, Max.Price &amp; RoughRange
pause()

vif(carprice.lm)     # Price alone
</code></pre>

<hr>
<h2 id='Cars93.summary'>A Summary of the Cars93 Data set</h2><span id='topic+Cars93.summary'></span>

<h3>Description</h3>

<p>The <code>Cars93.summary</code> data frame has 6 rows and 4 columns
created from information in the <code>Cars93</code> data set in the Venables 
and Ripley MASS package. Each row corresponds to a different
class of car (e.g. Compact, Large, etc.).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cars93.summary</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Min.passengers</dt><dd><p>minimum
passenger capacity for each class of car</p>
</dd>
<dt>Max.passengers</dt><dd><p>maximum
passenger capacity for each class of car</p>
</dd>
<dt>No.of.cars</dt><dd><p>number of
cars in each class</p>
</dd>
<dt>abbrev</dt><dd><p>a factor with levels
<code>C</code> Compact, <code>L</code> Large, 
<code>M</code> Mid-Size, <code>Sm</code> Small, 
<code>Sp</code> Sporty, <code>V</code> Van</p>
</dd>
</dl>



<h3>Source</h3>

<p>Lock, R. H. (1993) 1993 New Car Data. Journal of Statistics
Education 1(1)
</p>


<h3>References</h3>

<p>MASS library
</p>


<h3>Examples</h3>

<pre><code class='language-R'>type &lt;- Cars93.summary$abbrev
type &lt;- Cars93.summary[,4]
type &lt;- Cars93.summary[,"abbrev"]
type &lt;- Cars93.summary[[4]] # Take the object that is stored
                            # in the fourth list element.
type
pause()

attach(Cars93.summary)
  # R can now access the columns of Cars93.summary directly
abbrev
detach("Cars93.summary")
pause()

#  To change the name of the \verb!abbrev! variable (the fourth column)
names(Cars93.summary)[4] &lt;- "code"
pause()

#  To change all of the names, try
names(Cars93.summary) &lt;- c("minpass","maxpass","number","code")

</code></pre>

<hr>
<h2 id='cerealsugar'>Percentage of Sugar in Breakfast Cereal</h2><span id='topic+cerealsugar'></span>

<h3>Description</h3>

<p>Measurements of sugar content in frosted flakes breakfast cereal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cerealsugar</code></pre>


<h3>Format</h3>

<p>A vector of 100 measurements.
</p>

<hr>
<h2 id='cfseal'>Cape Fur Seal Data</h2><span id='topic+cfseal'></span>

<h3>Description</h3>

<p>The <code>cfseal</code> data frame has 30 rows and 11 columns consisting
of weight measurements for various organs taken from 30 Cape Fur
Seals that died as an unintended consequence of commercial fishing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cfseal</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>age</dt><dd><p>a numeric vector</p>
</dd>
<dt>weight</dt><dd><p>a numeric vector</p>
</dd>
<dt>heart</dt><dd><p>a numeric vector</p>
</dd>
<dt>lung</dt><dd><p>a numeric vector</p>
</dd>
<dt>liver</dt><dd><p>a numeric vector</p>
</dd>
<dt>spleen</dt><dd><p>a numeric vector</p>
</dd>
<dt>stomach</dt><dd><p>a numeric vector</p>
</dd>
<dt>leftkid</dt><dd><p>a numeric vector</p>
</dd>
<dt>rightkid</dt><dd><p>a numeric vector</p>
</dd>
<dt>kidney</dt><dd><p>a numeric vector</p>
</dd>
<dt>intestines</dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Stewardson, C.L., Hemsley, S., Meyer, M.A., Canfield,
P.J. and Maindonald, J.H. 1999.  Gross and microscopic visceral
anatomy of the male Cape fur seal, Arctocephalus pusillus pusillus
(Pinnepedia: Otariidae), with reference to organ size and growth.
Journal of Anatomy (Cambridge) 195: 235-255.  (WWF project ZA-348)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Allometric Growth - Example 5.7")

cfseal.lm &lt;- lm(log(heart) ~ log(weight), data=cfseal); summary(cfseal.lm)
plot(log(heart) ~ log(weight), data = cfseal, pch=16, xlab = "Heart Weight (g, log scale)", 
ylab = "Body weight (kg, log scale)", axes=FALSE)
heartaxis &lt;- 100*(2^seq(0,3))
bodyaxis &lt;- c(20,40,60,100,180)
axis(1, at = log(bodyaxis), lab = bodyaxis)
axis(2, at = log(heartaxis), lab = heartaxis)
box()
abline(cfseal.lm)
</code></pre>

<hr>
<h2 id='cities'>Populations of Major Canadian Cities (1992-96)</h2><span id='topic+cities'></span>

<h3>Description</h3>

<p>Population estimates for several Canadian cities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cities</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>CITY</dt><dd><p>a factor, consisting of the city names</p>
</dd>
<dt>REGION</dt><dd><p>a factor with 5 levels (ATL=Atlantic, ON=Ontario,
QC=Quebec, PR=Prairies, WEST=Alberta and British Columbia) representing the location
of the cities</p>
</dd>
<dt>POP1992</dt><dd><p>a numeric vector giving population in 1000's for 1992</p>
</dd>
<dt>POP1993</dt><dd><p>a numeric vector giving population in 1000's for 1993</p>
</dd>
<dt>POP1994</dt><dd><p>a numeric vector giving population in 1000's for 1994</p>
</dd>
<dt>POP1995</dt><dd><p>a numeric vector giving population in 1000's for 1995</p>
</dd>
<dt>POP1996</dt><dd><p>a numeric vector giving population in 1000's for 1996</p>
</dd>
</dl>



<h3>Source</h3>

<p>Statistics Canada
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cities$have &lt;- factor((cities$REGION=="ON")|(cities$REGION=="WEST"))
plot(POP1996~POP1992, data=cities, col=as.integer(cities$have))
</code></pre>

<hr>
<h2 id='codling'>Dose-mortality data, for fumigation of codling moth with methyl bromide</h2><span id='topic+codling'></span>

<h3>Description</h3>

<p>Data are from trials that studied the mortality response of codling moth
to fumigation with methyl bromide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(codling)</code></pre>


<h3>Format</h3>

<p>A data frame with 99 observations on the following 10 variables.
</p>

<dl>
<dt>dose</dt><dd><p>Injected dose of methyl bromide, in gm per cubic meter</p>
</dd>
<dt>tot</dt><dd><p>Number of insects in chamber</p>
</dd>
<dt>dead</dt><dd><p>Number of insects dying</p>
</dd>
<dt>pobs</dt><dd><p>Proportion dying</p>
</dd>
<dt>cm</dt><dd><p>Control mortality, i.e., at dose 0</p>
</dd>
<dt>ct</dt><dd><p>Concentration-time sum</p>
</dd>
<dt>Cultivar</dt><dd><p>a factor with levels <code>BRAEBURN</code> <code>FUJI</code> <code>GRANNY</code> <code>Gala</code> <code>ROYAL</code> <code>Red Delicious</code> <code>Splendour</code></p>
</dd>
<dt>gp</dt><dd><p>a factor which has a different level for each different
combination of <code>Cultivar</code>, <code>year</code> and <code>rep</code> (replicate).</p>
</dd>
<dt>year</dt><dd><p>a factor with levels <code>1988</code> <code>1989</code></p>
</dd>
<dt>numcm</dt><dd><p>a numeric vector: total number of control insects</p>
</dd>
</dl>



<h3>Details</h3>

<p>The research that generated these data was in part funded by New Zealand
pipfruit growers.  The published analysis was funded by New Zealand
pipfruit growers. See also <code>sorption</code>.
</p>


<h3>Source</h3>

<p>Maindonald, J.H.; Waddell, B.C.; Petry, R.J. 2001.
Apple cultivar effects on codling moth (Lepidoptera: Tortricidae)
egg mortality following fumigation with methyl bromide.
Postharvest Biology and Technology 22: 99-110.
</p>

<hr>
<h2 id='compareTreecalcs'>Error rate comparisons for tree-based classification</h2><span id='topic+compareTreecalcs'></span>

<h3>Description</h3>

<p>Compare error rates, between different functions and different
selection rules, for an approximately equal random division of the
data into a training and test set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareTreecalcs(x = yesno ~ ., data = DAAG::spam7, cp = 0.00025, fun = c("rpart",
"randomForest"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareTreecalcs_+3A_x">x</code></td>
<td>
<p>model formula</p>
</td></tr>
<tr><td><code id="compareTreecalcs_+3A_data">data</code></td>
<td>
<p>an data frame in which to interpret the variables
named in the formula</p>
</td></tr>
<tr><td><code id="compareTreecalcs_+3A_cp">cp</code></td>
<td>
<p>setting for the cost complexity parameter <code>cp</code>,
used by rpart()</p>
</td></tr>
<tr><td><code id="compareTreecalcs_+3A_fun">fun</code></td>
<td>
<p>one or both of &quot;rpart&quot; and &quot;randomForest&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Data are randomly divided into two subsets, I and II. The function(s)
are used in the standard way for calculations on subset I, and error
rates returined that come from the calculations carried out by the
function(s).  Predictions are made for subset II, allowing the
calculation of a completely independent set of error rates.
</p>


<h3>Value</h3>

<p>If <code>rpart</code> is specified in <code>fun</code>, the following:
</p>
<table>
<tr><td><code>rpSEcvI</code></td>
<td>
<p>the estimated cross-validation error rate
when <code>rpart()</code> is run on the training data (I), and the
one-standard error rule is used</p>
</td></tr>
<tr><td><code>rpcvI</code></td>
<td>
<p>the estimated cross-validation error rate when 
<code>rpart()</code> is run on subset I, and the model used that
gives the minimum cross-validated error rate</p>
</td></tr>
<tr><td><code>rpSEtest</code></td>
<td>
<p>the error rate when the model that leads to <code>rpSEcvI</code>
is used to make predictions for subset II</p>
</td></tr>
<tr><td><code>rptest</code></td>
<td>
<p>the error rate when the model that leads to <code>rpcvI</code>
is used to make predictions for subset II</p>
</td></tr>
<tr><td><code>nSErule</code></td>
<td>
<p>number of splits required by the one standard error rule</p>
</td></tr>
<tr><td><code>nREmin</code></td>
<td>
<p>number of splits to give the minimum error</p>
</td></tr>
</table>
<p>If <code>rpart</code> is specified in <code>fun</code>, the following:
</p>
<table>
<tr><td><code>rfcvI</code></td>
<td>
<p>the out-of-bag (OOB) error rate when 
<code>randomForest()</code> is run on subset I</p>
</td></tr>
<tr><td><code>rftest</code></td>
<td>
<p>the error rate when the model that leads to <code>rfcvI</code>
is used to make predictions for subset II</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald</p>

<hr>
<h2 id='component.residual'>Component + Residual Plot</h2><span id='topic+component.residual'></span>

<h3>Description</h3>

<p>Component + Residual plot for a term in a <code>lm</code> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>component.residual(lm.obj, which = 1, xlab = "Component",
    ylab = "C+R")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="component.residual_+3A_lm.obj">lm.obj</code></td>
<td>
<p> A <code>lm</code> object </p>
</td></tr>
<tr><td><code id="component.residual_+3A_which">which</code></td>
<td>
<p>numeric code for the term in the <code>lm</code> formula to be 
plotted</p>
</td></tr>
<tr><td><code id="component.residual_+3A_xlab">xlab</code></td>
<td>
<p>label for the x-axis</p>
</td></tr>
<tr><td><code id="component.residual_+3A_ylab">ylab</code></td>
<td>
<p>label for the y-axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatterplot with a smooth curve overlaid.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>mice12.lm &lt;- lm(brainwt ~ bodywt + lsize, data=litters)
oldpar &lt;- par(mfrow = c(1,2))
component.residual(mice12.lm, 1, xlab = "Body weight", ylab= "t(Body weight) + e")
component.residual(mice12.lm, 2, xlab = "Litter size", ylab= "t(Litter size) + e")
par(oldpar)
</code></pre>

<hr>
<h2 id='confusion'>
Given actual and predicted group assignments, give the confusion matrix
</h2><span id='topic+confusion'></span>

<h3>Description</h3>

<p>Given actual and predicted group assignments, give the confusion matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confusion(actual, predicted, gpnames = NULL, rowcol=c("actual", "predicted"),
printit = c("overall","confusion"), prior = NULL, digits=3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="confusion_+3A_actual">actual</code></td>
<td>

<p>Actual (prior) group assigments
</p>
</td></tr>
<tr><td><code id="confusion_+3A_predicted">predicted</code></td>
<td>

<p>Predicted group assigments.
</p>
</td></tr>
<tr><td><code id="confusion_+3A_gpnames">gpnames</code></td>
<td>

<p>Names for groups, if different from <code>levels(actual)</code>
</p>
</td></tr>
<tr><td><code id="confusion_+3A_rowcol">rowcol</code></td>
<td>
<p>For predicted categories to appear as rows,
specify <code>rowcol="predicted"</code>
</p>
</td></tr>
<tr><td><code id="confusion_+3A_printit">printit</code></td>
<td>

<p>Character vector. Print <code>"overall"</code>, or <code>"confusion"</code> matrix, or both.
</p>
</td></tr>
<tr><td><code id="confusion_+3A_prior">prior</code></td>
<td>

<p>Prior probabilities for groups, if different from the relative group
frequencies
</p>
</td></tr>
<tr><td><code id="confusion_+3A_digits">digits</code></td>
<td>
<p>Number of decimal digits to display in printed output
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted group assignments should be estimated from cross-validation
or from bootstrap out-of-bag data.  Better still, work with assignments
for test data that are completely separate from the data used to
dervive the model.
</p>


<h3>Value</h3>

<p>A list with elements overall (overall accuracy), confusion (confusion
matrix) and prior (prior used for calculation of overall accuracy)
</p>


<h3>Author(s)</h3>

<p>John H Maindonald
</p>


<h3>References</h3>

<p>Maindonald and Braun: 'Data Analysis and Graphics Using R',
3rd edition 2010, Section 12.2.2
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(MASS)
library(DAAG)
cl &lt;- lda(species ~ length+breadth, data=cuckoos, CV=TRUE)$class
confusion(cl, cuckoos$species)

## The function is currently defined as
function (actual, predicted, gpnames = NULL,
            rowcol = c("actual", "predicted"),
            printit = c("overall","confusion"),
            prior = NULL, digits = 3) 
{
  if (is.null(gpnames)) 
    gpnames &lt;- levels(actual)
  if (is.logical(printit)){
    if(printit)printit &lt;- c("overall","confusion")
    else printit &lt;- ""
  }
  tab &lt;- table(actual, predicted)
  acctab &lt;- t(apply(tab, 1, function(x) x/sum(x)))
  dimnames(acctab) &lt;- list(Actual = gpnames, `Predicted (cv)` = gpnames)
  if (is.null(prior)) {
    relnum &lt;- table(actual)
    prior &lt;- relnum/sum(relnum)
    acc &lt;- sum(tab[row(tab) == col(tab)])/sum(tab)
  }
  else {
    acc &lt;- sum(prior * diag(acctab))
  }
  names(prior) &lt;- gpnames
  if ("overall"%in%printit) {
    cat("Overall accuracy =", round(acc, digits), "\n")
    if(is.null(prior)){
      cat("This assumes the following prior frequencies:", 
          "\n")
      print(round(prior, digits))
    }
  }
  if ("confusion"%in%printit) {
    cat("\nConfusion matrix", "\n")
    print(round(acctab, digits))
  }
  invisible(list(overall=acc, confusion=acctab, prior=prior))
}
</code></pre>

<hr>
<h2 id='coralPval'>
P-values from biological expression array data
</h2><span id='topic+coralPval'></span>

<h3>Description</h3>

<p>P-values were calculated for each of 3072 genes, for data that
compared expression values between post-settlement coral larvae
and pre-settlement coral larvae.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("coralPval")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:3072, 1] 8.60e-01 3.35e-08 3.96e-01 2.79e-01 6.36e-01 ...
</p>


<h3>Details</h3>

<p>t-statistics, and hence p-values, were derived from five replicate
two-colour micro-array slides.  Details are in a vignette that
accompanies the <span class="pkg">DAAGbio</span> package.
</p>


<h3>Source</h3>

<p>See the <code>?DAAGbio::coralRG</code>
</p>


<h3>References</h3>

<p>Grasso, L. C.; Maindonald, J.; Rudd, S.; Hayward, D. C.; Saint, R.;
Miller, D. J.; and Ball, E. E., 2008. Microarray analysis identifies
candidate genes for key roles in coral development. BMC Genomics, 9:540.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## From p-values, calculate Benjamini-Hochberg false discrimination rates
fdr &lt;- p.adjust(DAAG::coralPval, method='BH')
## Number of genes identified as differentially expressed for FDR = 0.01
sum(fdr&lt;=0.01)
</code></pre>

<hr>
<h2 id='cottonworkers'>Occupation and wage profiles of British cotton workers</h2><span id='topic+cottonworkers'></span>

<h3>Description</h3>

<p>Numbers are given in different categories of worker, in each of
two investigations.  The first source of information is the
Board of Trade Census that was conducted on 1886.  The second
is a relatively informal survey conducted by US Bureau of
Labor representatives in 1889, for use in official reports.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cottonworkers)</code></pre>


<h3>Format</h3>

<p>A data frame with 14 observations on the following 3 variables.
</p>

<dl>
<dt>census1886</dt><dd><p>Numbers of workers in each of 14 different
categories, according to the Board of Trade wage census that
was conducted in 1886</p>
</dd>
<dt>survey1889</dt><dd><p>Numbers of workers in each of 14 different
categories, according to data collected in 1889 by the US Bureau
of Labor, for use in a report to the US Congress and House of
Representatives</p>
</dd>
<dt>avwage</dt><dd><p>Average wage, in pence, as estimated in the US Bureau
of Labor survey</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data in <code>survey1889</code> were collected in a relatively informal
manner, by approaching individuals on the street.  Biases might
therefore be expected.
</p>


<h3>Source</h3>

<p>United States congress, House of Representatives, Sixth Annual Report
of the Commissioner of Labor, 1890, Part III, Cost of Living
(Washington D.C. 1891); idem.,  Seventh Annual Report
of the Commissioner of Labor, 1891, Part III, Cost of Living
(Washington D.C. 1892)
</p>
<p>Return of wages in the principal textile trades of the United Kingdom,
with report therein. (P.P. 1889, LXX). United Kingdom Official
Publication.
</p>


<h3>References</h3>

<p>Boot, H. M. and Maindonald, J. H. 2007. New estimates of age- and sex-
specific earnings and the male-female earnings gap in the British
cotton industry, 1833-1906. <em>Economic History Review</em>. Published
online 28-Aug-2007 doi: 10.1111/j.1468-0289.2007.00398.x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cottonworkers)
str(cottonworkers)
plot(survey1889 ~ census1886, data=cottonworkers)
plot(I(avwage*survey1889) ~ I(avwage*census1886), data=cottonworkers)
</code></pre>

<hr>
<h2 id='cricketer'>Lifespans of UK 1st class cricketers born 1840-1960</h2><span id='topic+cricketer'></span>

<h3>Description</h3>

<p>Year and birth, lifespan, etc, of British first class cricketers,
born 1840-1960, whose handedness could be determined from
information in the Who's who of cricketers.  The status
(alive=0, dead =1), and lifetime or lifespan, is for 1992.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cricketer)</code></pre>


<h3>Format</h3>

<p>A data frame with 5960 observations on the following 8 variables.
</p>

<dl>
<dt><code>left</code></dt><dd><p>a factor with levels <code>right</code> <code>left</code></p>
</dd>
<dt><code>year</code></dt><dd><p>numeric, year of birth</p>
</dd>
<dt><code>life</code></dt><dd><p>numeric, lifetime or lifespan to 1992</p>
</dd>
<dt><code>dead</code></dt><dd><p>numeric (0 = alive (censored), 1 = dead, in 1992)</p>
</dd>
<dt><code>acd</code></dt><dd><p>numeric (0 = not accidental or not dead, 1
= accidental death)</p>
</dd>
<dt><code>kia</code></dt><dd><p>numeric (0 = not killed in action, 1
= killed in action)</p>
</dd>
<dt><code>inbed</code></dt><dd><p>numeric (0 = did not die in bed, 1
= died in bed)</p>
</dd>
<dt><code>cause</code></dt><dd><p>a factor with levels <code>alive</code> <code>acd</code>
(accidental death) <code>inbed</code> (died in bed)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note that those 'killed in action' (mostly during World Wars I and II)
form a subset of those who died by accident.
</p>


<h3>Source</h3>

<p>John Aggleton, Martin Bland.  Data were collated as described in
Aggleton et al.
</p>


<h3>References</h3>

<p>Aggleton JP, Bland JM, Kentridge RW, Neave NJ 1994.  Handedness and
longevity: an archival study of cricketers.  British Medical Journal
309, 1681-1684.
</p>
<p>Bailey P, Thorne P, Wynne-Thomas P. 1993. Who's Who of Cricketers.
2nd ed, London, Hamlyn.
</p>
<p>Bland M and Altman D. 2005. Do the left-handed die young?
Significance 2, 166-170.
</p>


<h3>See Also</h3>

<p><code>earlycrcktr</code>.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cricketer)
numLH &lt;- xtabs(~ left+year, data=cricketer)
propLH &lt;- prop.table(numLH, margin=2)[2,]
yr &lt;- as.numeric(colnames(numLH))
plot(propLH ~ yr)
cricketer$lh &lt;- unclass(cricketer$left)-1
left2.hat &lt;- fitted(lm(lh ~ poly(year,2), data=cricketer))
ord &lt;- order(cricketer$year)
lines(left2.hat[ord] ~ cricketer$year[ord])
library(splines)
ns3.hat &lt;- fitted(lm(lh ~ ns(year,3), data=cricketer))
lines(ns3.hat[ord] ~ cricketer$year[ord], col="red")
require(survival)
summary(coxph(Surv(life, kia) ~ bs(year,3) +left, data=cricketer))
cricketer$notacdDead &lt;- with(cricketer, {dead[acd==1]&lt;-0; dead})
summary(coxph(Surv(life, notacdDead) ~ ns(year,2) +left, data=cricketer))
</code></pre>

<hr>
<h2 id='cuckoohosts'>Comparison of cuckoo eggs with host eggs</h2><span id='topic+cuckoohosts'></span>

<h3>Description</h3>

<p>These data compare mean length, mean breadth, and egg color,
between cuckoos and their hosts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cuckoohosts</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 12 variables.
</p>

<dl>
<dt>clength</dt><dd><p>mean length of cuckoo eggs in given host's nest</p>
</dd>
<dt>cl.sd</dt><dd><p>standard deviation of cuckoo egg lengths</p>
</dd>
<dt>cbreadth</dt><dd><p>mean breadth of cuckoo eggs in given host's nest</p>
</dd>
<dt>cb.sd</dt><dd><p>standard deviation of cuckoo egg breadths</p>
</dd>
<dt>cnum</dt><dd><p>number of cuckoo eggs</p>
</dd>
<dt>hlength</dt><dd><p>length of host eggs</p>
</dd>
<dt>hl.sd</dt><dd><p>standard deviation of host egg lengths</p>
</dd>
<dt>hbreadth</dt><dd><p>breadth of host eggs</p>
</dd>
<dt>hb.sd</dt><dd><p>standard deviation of host egg breadths</p>
</dd>
<dt>hnum</dt><dd><p>number of host eggs</p>
</dd>
<dt>match</dt><dd><p>number of eggs where color matched</p>
</dd>
<dt>nomatch</dt><dd><p>number where color did not match</p>
</dd>
</dl>



<h3>Details</h3>

<p>Although from the same study that generated data in the data frame
<code>cuckoos</code>, the data do not match precisely.  The cuckoo egg
lengths and breadths are from the tables on page 168, the host egg
lengths and breadths from Appendix IV on page 176, and the color
match counts from the table on page 171.
</p>


<h3>Source</h3>

<p>Latter, O.H., 1902.
The egg of <em>cuculus canorus</em>. an inquiry into the dimensions of
the cuckoo's egg and the relation of the variations to the size of the eggs
of the foster-parent, with notes on coloration, &amp;c.
<em>Biometrika</em>, 1:164&ndash;176.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cuckoohosts
str(cuckoohosts)
plot(cuckoohosts)
with(cuckoohosts,
     plot(c(clength,hlength),c(cbreadth,hbreadth),col=rep(1:2,c(6,6))))
</code></pre>

<hr>
<h2 id='cuckoos'>Cuckoo Eggs Data</h2><span id='topic+cuckoos'></span>

<h3>Description</h3>

<p>Length and breadth measurements of 120 eggs lain in the nests of six 
different species of host bird.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cuckoos</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>length</dt><dd><p>the egg lengths in millimeters</p>
</dd>
<dt>breadth</dt><dd><p>the egg breadths in millimeters</p>
</dd>
<dt>species</dt><dd><p>a factor with levels
<code>hedge.sparrow</code>, 
<code>meadow.pipit</code>, 
<code>pied.wagtail</code>, 
<code>robin</code>, 
<code>tree.pipit</code>, 
<code>wren</code> 
</p>
</dd> 
<dt>id</dt><dd><p>a numeric vector</p>
</dd>
</dl>
 


<h3>Source</h3>

<p> Latter, O.H. (1902). The eggs of Cuculus canorus. An
Inquiry into the dimensions of the cuckoo's egg and the relation of the
variations to the size of the eggs of the foster-parent, with notes on
coloration, &amp;c. Biometrika i, 164.  Tippett (1931) gives summary details
of the data.
</p>


<h3>References</h3>

<p>Tippett, L.H.C. 1931: &quot;The Methods of Statistics&quot;. Williams &amp; Norgate, 
London.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
print("Strip and Boxplots - Example 2.1.2")

attach(cuckoos)
oldpar &lt;- par(las = 2) # labels at right angle to axis.
stripchart(length ~ species) 
boxplot(split(cuckoos$length, cuckoos$species),
         xlab="Length of egg", horizontal=TRUE)
detach(cuckoos)
par(oldpar)
pause()

print("Summaries - Example 2.2.2")
sapply(split(cuckoos$length, cuckoos$species), sd)
pause()

print("Example 4.1.4")
wren &lt;- split(cuckoos$length, cuckoos$species)$wren
median(wren)
n &lt;- length(wren)
sqrt(pi/2)*sd(wren)/sqrt(n)  # this s.e. computation assumes normality
</code></pre>

<hr>
<h2 id='CVbinary'>Cross-Validation for Regression with a Binary Response</h2><span id='topic+CVbinary'></span><span id='topic+cv.binary'></span>

<h3>Description</h3>

<p>These functions give training (internal) and cross-validation measures
of predictive accuracy for regression with a binary response.  The
data are randomly divided between a number of &lsquo;folds&rsquo;.  Each fold is
removed, in turn, while the remaining data are used to re-fit the
regression model and to predict at the omitted observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVbinary(obj, rand=NULL, nfolds=10, print.details=TRUE)

cv.binary(obj, rand=NULL, nfolds=10, print.details=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CVbinary_+3A_obj">obj</code></td>
<td>
<p>a <code>glm</code> object</p>
</td></tr>
<tr><td><code id="CVbinary_+3A_rand">rand</code></td>
<td>
<p>a vector which assigns each observation to a fold</p>
</td></tr>
<tr><td><code id="CVbinary_+3A_nfolds">nfolds</code></td>
<td>
<p> the number of folds</p>
</td></tr>
<tr><td><code id="CVbinary_+3A_print.details">print.details</code></td>
<td>
<p> logical variable (TRUE = print detailed output,
the default)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>cvhat</code></td>
<td>
<p>predicted values from cross-validation</p>
</td></tr>
<tr><td><code>internal</code></td>
<td>
<p>internal or (better) training predicted values</p>
</td></tr>
<tr><td><code>training</code></td>
<td>
<p>training predicted values</p>
</td></tr>
<tr><td><code>acc.cv</code></td>
<td>
<p>cross-validation estimate of accuracy</p>
</td></tr>
<tr><td><code>acc.internal</code></td>
<td>
<p>internal or (better) training estimate of accuracy</p>
</td></tr>
<tr><td><code>acc.training</code></td>
<td>
<p>training estimate of accuracy</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The term &lsquo;training&rsquo; seems preferable to the term
&lsquo;internal&rsquo; in connection with predicted values, and
the accuracy measure, that are based on the observations used to derive
the model.</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>frogs.glm &lt;- glm(pres.abs ~ log(distance) + log(NoOfPools),
                 family=binomial,data=frogs)
CVbinary(frogs.glm)
mifem.glm &lt;- glm(outcome ~ ., family=binomial, data=mifem)
CVbinary(mifem.glm)
</code></pre>

<hr>
<h2 id='CVlm'>Cross-Validation for Linear Regression</h2><span id='topic+CVlm'></span><span id='topic+cv.lm'></span>

<h3>Description</h3>

<p>This function gives internal and cross-validation measures of predictive
accuracy for multiple linear regression. (For binary logistic
regression, use the <code>CVbinary</code> function.) The data are
randomly assigned to a number of &lsquo;folds&rsquo;.
Each fold is removed, in turn, while the remaining data is used
to re-fit the regression model and to predict at the deleted observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVlm(data = DAAG::houseprices, form.lm = formula(sale.price ~ area),
              m = 3, dots = FALSE, seed = 29, plotit = c("Observed","Residual"),
              col.folds=NULL,               
              main="Small symbols show cross-validation predicted values",
              legend.pos="topleft", 
              printit = TRUE, ...)
cv.lm(data = DAAG::houseprices, form.lm = formula(sale.price ~ area),
              m = 3, dots = FALSE, seed = 29, plotit = c("Observed","Residual"),
              col.folds=NULL,               
              main="Small symbols show cross-validation predicted values",
              legend.pos="topleft", printit = TRUE, ...)
              </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CVlm_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="CVlm_+3A_form.lm">form.lm</code></td>
<td>
<p>a formula or <code>lm</code> call or <code>lm</code> object</p>
</td></tr>
<tr><td><code id="CVlm_+3A_m">m</code></td>
<td>
<p>the number of folds</p>
</td></tr>
<tr><td><code id="CVlm_+3A_dots">dots</code></td>
<td>
<p>uses pch=16 for the plotting character</p>
</td></tr>
<tr><td><code id="CVlm_+3A_seed">seed</code></td>
<td>
<p>random number generator seed</p>
</td></tr>
<tr><td><code id="CVlm_+3A_plotit">plotit</code></td>
<td>
<p>This can be one of the text strings <code>"Observed"</code>,
<code>"Residual"</code>, or a logical value.  The logical <code>TRUE</code> is
equivalent to <code>"Observed"</code>, while <code>FALSE</code> is equivalent
to <code>""</code> (no plot)</p>
</td></tr>
<tr><td><code id="CVlm_+3A_col.folds">col.folds</code></td>
<td>
<p>Per fold color settings</p>
</td></tr>
<tr><td><code id="CVlm_+3A_main">main</code></td>
<td>
<p>main title for graph</p>
</td></tr>
<tr><td><code id="CVlm_+3A_legend.pos">legend.pos</code></td>
<td>
<p>position of legend: one of
<code>"bottomright"</code>, <code>"bottom"</code>, <code>"bottomleft"</code>,
<code>"left"</code>, <code>"topleft"</code>, <code>"top"</code>, <code>"topright"</code>,
<code>"right"</code>, <code>"center"</code>.
</p>
</td></tr>  
<tr><td><code id="CVlm_+3A_printit">printit</code></td>
<td>
<p>if TRUE, output is printed to the screen</p>
</td></tr>
<tr><td><code id="CVlm_+3A_...">...</code></td>
<td>
<p>Other arguments, to be passed through to the function <code>legend()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>plotit="Residual"</code> and there is more than one explanatory
variable, the fitted lines that are shown for the individual folds
are approximations.
</p>


<h3>Value</h3>

<p>The input data frame is returned, with additional columns
<code>Predicted</code> (Predicted values using all observations)
and <code>cvpred</code> (cross-validation predictions). The
cross-validation residual sum of squares (<code>ss</code>) and
degrees of freedom (<code>df</code>) are returned as attributes of
the data frame.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a></code>, <code><a href="#topic+CVbinary">CVbinary</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>CVlm()
## Not run: 
CVlm(data=nihills, form.lm=formula(log(time)~log(climb)+log(dist)),
          plotit="Observed")
CVlm(data=nihills, form.lm=formula(log(time)~log(climb)+log(dist)),
     plotit="Residual")
out &lt;- CVlm(data=nihills, form.lm=formula(log(time)~log(climb)+log(dist)),
               plotit="Observed")
out[c("ms","df")]

## End(Not run)
</code></pre>

<hr>
<h2 id='DAAGtheme'>
Function to generate lattice themes for graphs.
</h2><span id='topic+DAAGtheme'></span>

<h3>Description</h3>

<p>This generates themes for use in a planned 4th edition of 
&quot;Data Analysis and Graphics Using R&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DAAGtheme(fontsize = list(text = 10, points = 6), box = "gray40",
          sides = list(tck = 0.6, pad1 = 0.75, pad2 = 0.75),...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DAAGtheme_+3A_fontsize">fontsize</code></td>
<td>

<p>Fontize for text and points. Specify as, e.g., <code>list(text = 10, points = 6)</code>.
</p>
</td></tr>
<tr><td><code id="DAAGtheme_+3A_box">box</code></td>
<td>
<p>Color for the panel and strip borders.</p>
</td></tr>
<tr><td><code id="DAAGtheme_+3A_sides">sides</code></td>
<td>
<p>List, with elements <code>tck</code> (Tick length, as fraction of lattice default),
and margin paddings <code>pad1</code>, <code>pad2</code>, <code>pad3</code>, and <code>pad4</code>.
Margin paddings set the distance, in lines, from the tick marks to the tick labels.
</p>
</td></tr>
<tr><td><code id="DAAGtheme_+3A_...">...</code></td>
<td>

<p>Settings that will be passed to <code>simpleTheme()</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Setting the color of the bounding box and of the strip boxes set gray, 
which is the default, reduces the focus on them.
</p>


<h3>Value</h3>

<p>A list which can be used as the <code>par.settings</code> argument to lattice 
graphics functions, or as the <code>theme</code> argument to <code>trellis.par.set()</code>.
</p>


<h3>Note</h3>

<p>The code provides an example of the creation of a functions that 
generates themes that are tuned to specific user requirements.
In this connection, see also <code><a href="latticeExtra.html#topic+theEconomist.theme">theEconomist.theme</a></code>.
</p>


<h3>Author(s)</h3>

<p>John Maindonald.
</p>


<h3>See Also</h3>

<p><code><a href="lattice.html#topic+standard.theme">standard.theme</a></code>, <code><a href="lattice.html#topic+simpleTheme">simpleTheme</a></code>,
<code><a href="latticeExtra.html#topic+theEconomist.theme">theEconomist.theme</a></code>,
<code><a href="latticeExtra.html#topic+custom.theme">custom.theme</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bwtheme &lt;- DAAGtheme(pch=2:4)
lattice::xyplot(csoa ~ age | target, groups=sex, 
                data=DAAG::tinting, par.settings=bwtheme)
</code></pre>

<hr>
<h2 id='DAAGxdb'>List, each of whose elements hold rows of a file, in character format</h2><span id='topic+DAAGxdb'></span>

<h3>Description</h3>

<p>This is the default database for use with the function
<code>datafile</code>, which uses elements of this list to place files
in the working directory. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DAAGxdb)</code></pre>


<h3>Format</h3>

  
<p>Successive elements in this list hold character vectors from which the
corresponding files can be generated.  The names of the list elements
are fuel, fuel.csv, oneBadRow, scan-demo, molclock1, molclock2, and
travelbooks. 
</p>


<h3>Details</h3>

<p>The files fuel.txt and fuel.csv are used in Chapter 1 of DAAGUR, while
the files oneBadRow.txt and scan-demo.txt are used in Chapter 14 of
DAAGUR.</p>


<h3>References</h3>

<p>Maindonald, J.H. and Braun, W.J. 2007. Data Analysis and Graphics
Using R: An Example-Based Approach.  2nd edn, Cambridge University
Press (DAAGUR).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DAAGxdb)
names(DAAGxdb)
</code></pre>

<hr>
<h2 id='datafile'>Write an ASCII data file to the working directory.</h2><span id='topic+datafile'></span>

<h3>Description</h3>

<p>Invoking this function writes one or more nominated files to the
working directory.  In particular, it may be used to write the
files 'fuel.txt' and 'fuel.csv' that are used in Chapter 1 of
DAAGUR, and the files 'oneBadRow.txt' and 'scan-demo.txt' that
are used in Chapter 14 of DAAGUR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>datafile(file = c("fuel", "travelbooks"), datastore =
                 DAAG::DAAGxdb, altstore = DAAG::zzDAAGxdb, showNames =
                 FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="datafile_+3A_file">file</code></td>
<td>
<p>character; with the defaults for <code>datastore</code> and
<code>altstore</code> the options are
&quot;fuel&quot;, for fuel.txt;
&quot;fuel.csv&quot;, for fuel.csv;
&quot;oneBadRow&quot;, for oneBadRow.txt;
&quot;scan-demo&quot;, for scan-demo.txt;
&quot;molclock1&quot;, for molclock1.txt;
&quot;molclock2&quot;, for molclock2.txt;
&quot;travelbooks&quot;, for travelbooks.txt;
&quot;bestTimes&quot;, for bestTimes.txt;
&quot;bostonc&quot;, for bostonc.txt
</p>
</td></tr>
<tr><td><code id="datafile_+3A_datastore">datastore</code></td>
<td>
<p>Each element of this list is a character vector
that holds the rows of a file.</p>
</td></tr>
<tr><td><code id="datafile_+3A_altstore">altstore</code></td>
<td>
<p>An alternative list.  The default alternative list 
is used for the two files that are more than a few lines.</p>
</td></tr>
<tr><td><code id="datafile_+3A_shownames">showNames</code></td>
<td>
<p>if <code>TRUE</code>, returns the names of available datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

  
<p>An ASCII file is output to the current working directory. The names of
all available datasets are returned invisibly.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>datafile(file="", showNames=TRUE)
</code></pre>

<hr>
<h2 id='dengue'>Dengue prevalence, by administrative region</h2><span id='topic+dengue'></span>

<h3>Description</h3>

<p>Data record, for each of 2000 administrative regions,
whether or not dengue was recorded at any time between 1961 and 1990.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dengue)</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 observations on the following 13 variables.
</p>

<dl>
<dt>humid</dt><dd><p>Average vapour density: 1961-1990</p>
</dd>
<dt>humid90</dt><dd><p>90th percentile of <code>humid</code></p>
</dd>
<dt>temp</dt><dd><p>Average temperature: 1961-1990</p>
</dd>
<dt>temp90</dt><dd><p>90th percentile of <code>temp</code></p>
</dd>
<dt>h10pix</dt><dd><p>maximum of <code>humid</code>, within a 10 pixel radius</p>
</dd>
<dt>h10pix90</dt><dd><p>maximum of <code>humid90</code>, within a 10 pixel radius</p>
</dd>
<dt>trees</dt><dd><p>Percent tree cover, from satellite data</p>
</dd>
<dt>trees90</dt><dd><p>90th percentile of <code>trees</code></p>
</dd>
<dt>NoYes</dt><dd><p>Was dengue observed? (1=yes)</p>
</dd>
<dt>Xmin</dt><dd><p>minimum longitude</p>
</dd>
<dt>Xmax</dt><dd><p>maximum longitude</p>
</dd>
<dt>Ymin</dt><dd><p>minimum latitude</p>
</dd>
<dt>Ymax</dt><dd><p>maximum latitude</p>
</dd>
</dl>



<h3>Details</h3>

<p>This is derived from a data set in which the climate and tree cover
information were given for each half degree of latitude by half
degreee of longitude pixel.
The variable <code>NoYes</code> was given by administrative region.
The climate data and tree cover data given here are 50th or 90th
percentiles, where percetiles were calculates across pixels for an
administrative region.
</p>


<h3>Source</h3>

<p>Simon Hales, Environmental Research New Zealand Ltd.
</p>


<h3>References</h3>

<p>Hales, S., de Wet, N., Maindonald, J. and Woodward, A.
2002.  Potential effect of population and climate change global
distribution of dengue fever: an empirical model.  The Lancet 2002;
360: 830-34. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(dengue)
glm(NoYes ~ humid, data=dengue, family=binomial)
glm(NoYes ~ humid90, data=dengue, family=binomial)
</code></pre>

<hr>
<h2 id='dewpoint'>Dewpoint Data</h2><span id='topic+dewpoint'></span>

<h3>Description</h3>

<p>The <code>dewpoint</code> data frame has 72 rows and 3 columns.
Monthly data were obtained for a number of sites (in Australia)
and a number of months. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dewpoint</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>maxtemp</dt><dd><p>monthly minimum temperatures</p>
</dd>
<dt>mintemp</dt><dd><p>monthly maximum temperatures</p>
</dd>
<dt>dewpt</dt><dd><p>monthly average dewpoint for each combination of
minimum and maximum temperature readings (formerly dewpoint)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Dr Edward Linacre,
visiting fellow in the Australian National University Department
of Geography. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Additive Model - Example 7.5")
require(splines)
attach(dewpoint)   
ds.lm &lt;- lm(dewpt ~ bs(maxtemp,5) + bs(mintemp,5), data=dewpoint)
ds.fit &lt;-predict(ds.lm, type="terms", se=TRUE)
oldpar &lt;- par(mfrow=c(1,2))
plot(maxtemp, ds.fit$fit[,1], xlab="Maximum temperature",
     ylab="Change from dewpoint mean",type="n")
lines(maxtemp,ds.fit$fit[,1])
lines(maxtemp,ds.fit$fit[,1]-2*ds.fit$se[,1],lty=2)
lines(maxtemp,ds.fit$fit[,1]+2*ds.fit$se[,1],lty=2)
plot(mintemp,ds.fit$fit[,2],xlab="Minimum temperature",
     ylab="Change from dewpoint mean",type="n")
ord&lt;-order(mintemp)
lines(mintemp[ord],ds.fit$fit[ord,2])
lines(mintemp[ord],ds.fit$fit[ord,2]-2*ds.fit$se[ord,2],lty=2)
lines(mintemp[ord],ds.fit$fit[ord,2]+2*ds.fit$se[ord,2],lty=2)
detach(dewpoint)
par(oldpar)

</code></pre>

<hr>
<h2 id='droughts'>Periods Between Rain Events</h2><span id='topic+droughts'></span>

<h3>Description</h3>

<p>Data collected at Winnipeg International Airport (Canada)
on periods (in days) between rain events.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>droughts</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>length</dt><dd><p>the length of time from the completion of
the last rain event to the beginning of the next rain event.</p>
</dd>
<dt>year</dt><dd><p>the calendar year.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>  boxplot(length ~ year, data=droughts)
  boxplot(log(length) ~ year, data=droughts)
  hist(droughts$length, main="Winnipeg Droughts", xlab="length (in days)")
  hist(log(droughts$length), main="Winnipeg Droughts", xlab="length (in days, log scale)")

</code></pre>

<hr>
<h2 id='edcCO2'>EPICA Dome C Ice Core 800KYr Carbon Dioxide Data</h2><span id='topic+edcCO2'></span>

<h3>Description</h3>

<p>Carbon dioxide record from the EPICA (European Project for Ice Coring 
in Antarctica) Dome C ice core covering 0 to 800 kyr BP.    
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(edcCO2)</code></pre>


<h3>Format</h3>

<p>A data frame with 1096 observations on the following 2 variables.
</p>

<dl>
<dt><code>age</code></dt><dd><p>Age in years before present (BP)</p>
</dd>
<dt><code>co2</code></dt><dd><p>CO2 level (ppmv)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data are a composite series.
</p>


<h3>Source</h3>

<p><a href="https://www.ncei.noaa.gov/products/paleoclimatology/ice-core">https://www.ncei.noaa.gov/products/paleoclimatology/ice-core</a>  
</p>


<h3>References</h3>

<p>Luthi, D., M. et al.  2008.
High-resolution carbon dioxide concentration record 650,000-800,000 
years before present.
Nature, Vol. 453, pp. 379-382, 15 May 2008.  doi:10.1038/nature06949
</p>
<p>Indermuhle, A., E. et al, 1999, 
Atmospheric CO2 concentration from 60 to 20 kyr BP from the 
Taylor Dome ice core, Antarctica. Geophysical Research Letters, 27, 735-738.
</p>
<p>Monnin, E., A. et al. 2001. 
Atmospheric CO2 concentrations over the last glacial termination. 
Science, Vol. 291, pp. 112-114.
</p>
<p>Petit, J.R. et al. 1999. 
Climate and atmospheric history of the past 420,000 years from 
the Vostok ice core, Antarctica. 
Nature 399: 429-436.
</p>
<p>Siegenthaler, U. et al.  2005. 
Stable Carbon Cycle-Climate Relationship During the Late Pleistocene.
Science, v. 310 , pp. 1313-1317, 25 November 2005.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(edcCO2)
</code></pre>

<hr>
<h2 id='edcT'>EPICA Dome C Ice Core 800KYr Temperature Estimates</h2><span id='topic+edcT'></span>

<h3>Description</h3>

<p>Temperature record, using Deuterium as a proxy, from the EPICA
(European Project for Ice Coring in Antarctica) Dome C ice core
covering 0 to 800 kyr BP.    
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(edcT)</code></pre>


<h3>Format</h3>

<p>A data frame with 5788 observations on the following 5 variables.
</p>

<dl>
<dt><code>Bag</code></dt><dd><p>Bag number</p>
</dd>
<dt><code>ztop</code></dt><dd><p>Top depth (m)</p>
</dd>
<dt><code>Age</code></dt><dd><p>Years before 1950</p>
</dd>
<dt><code>Deuterium</code></dt><dd><p>Deuterium dD data</p>
</dd>
<dt><code>dT</code></dt><dd><p>Temperature difference from the average of the last
1000 years ~ -54.5degC</p>
</dd>
</dl>



<h3>Details</h3>

<p>Temperature was estimated from the deuterium data, after making various
corrections.
</p>


<h3>Source</h3>

<p><a href="https://www.ncei.noaa.gov/products/paleoclimatology/ice-core">https://www.ncei.noaa.gov/products/paleoclimatology/ice-core</a>
</p>


<h3>References</h3>

<p>Jouzel, J., et al.  2007.
EPICA Dome C Ice Core 800KYr Deuterium Data and Temperature Estimates. 
IGBP PAGES/World Data Center for Paleoclimatology 
Data Contribution Series No. 2007-091.
NOAA/NCDC Paleoclimatology Program, Boulder CO, USA.
</p>
<p>Jouzel, J., et al.  2007.  Orbital and Millennial Antarctic Climate
Variability over the Past 800,000 Years.  Science, Vol. 317, No. 5839,
pp.793-797, 10 August 2007.  </p>


<h3>Examples</h3>

<pre><code class='language-R'> data(edcT) </code></pre>

<hr>
<h2 id='elastic1'>Elastic Band Data Replicated</h2><span id='topic+elastic1'></span><span id='topic+elastic2'></span>

<h3>Description</h3>

<p>Both datasets
give, for each amount by which an elastic band is stretched
over the end of a ruler, the distance that the band traveled when
released.  The <code>elastic1</code> data frame has 7 rows.
The <code>elastic2</code> data frame, whose data span a wider range
of stretches and distances, has 9 rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(elastic1)
  data(elastic2)
  </code></pre>


<h3>Format</h3>

<p>These data frames contains the following columns:
</p>

<dl>
<dt>stretch</dt><dd><p>the amount by which the
elastic band was stretched</p>
</dd>
<dt>distance</dt><dd><p>the distance traveled</p>
</dd>
</dl>



<h3>Source</h3>

<p>J. H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(elastic1)
sapply(elastic1, mean)
pause()
sapply(elastic1, function(x)mean(x))
pause()
sapply(elastic1, function(x)sum(log(x)))
pause()
yrange &lt;- range(c(elastic1$distance, elastic2$distance))
xrange &lt;- range(c(elastic1$stretch, elastic2$stretch))
plot(distance ~ stretch, data = elastic1, pch = 16, ylim = yrange, xlim = 
xrange)
points(distance ~ stretch, data = elastic2, pch = 15, col = 2)
legend(xrange[1], yrange[2], legend = c("Data set 1", "Data set 2"), pch = 
c(16, 15), col = c(1, 2))
elastic1.lm &lt;- lm(distance ~ stretch, data = elastic1)
elastic2.lm &lt;- lm(distance ~ stretch, data = elastic2)
abline(elastic1.lm)
abline(elastic2.lm, col = 2)
summary(elastic1.lm)
summary(elastic2.lm)
pause()
predict(elastic1.lm, se.fit=TRUE)
predict(elastic2.lm, se.fit=TRUE)
</code></pre>

<hr>
<h2 id='elasticband'>Elastic Band Data</h2><span id='topic+elasticband'></span>

<h3>Description</h3>

<p>The <code>elasticband</code> data frame has 7 rows and 2 columns
giving, for each amount by which an elastic band is stretched
over the end of a ruler, the distance that the band traveled when
released.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elasticband</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>stretch</dt><dd><p>the amount by which the
elastic band was stretched</p>
</dd>
<dt>distance</dt><dd><p>the distance traveled</p>
</dd>
</dl>



<h3>Source</h3>

<p>J. H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Example 1.8.1")

attach(elasticband)     # R now knows where to find stretch and distance
plot(stretch, distance) # Alternative: plot(distance ~ stretch)
detach(elasticband)

print("Lists - Example 12.7")

elastic.lm &lt;- lm(distance ~ stretch, data=elasticband)
 names(elastic.lm)
 elastic.lm$coefficients
elastic.lm[["coefficients"]]
pause()

elastic.lm[[1]]
pause()

elastic.lm[1]
pause()

options(digits=3)
elastic.lm$residuals 
pause()

elastic.lm$call
pause()

 mode(elastic.lm$call)

</code></pre>

<hr>
<h2 id='errorsINseveral'>
Simulation of classical errors in x model, with multiple
explanatory variables.
</h2><span id='topic+errorsINseveral'></span>

<h3>Description</h3>

<p>Simulates $y-$ and $x-$values for a classical &ldquo;errors in $x$&rdquo;
linear regression model. One or more $x-$values are subject to random
measurement error, independently of the corresponding covariate
values that are measured without error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>errorsINseveral(n = 1000, a0 = 2.5, beta = c(1.5, 0), mu = 12.5, SDyerr = 0.5,
default.Vpar = list(SDx = 2, rho = -0.5, timesSDx = 1.5),
V = with(default.Vpar, matrix(c(1, rho, rho, 1), ncol = 2) * SDx^2),
xerrV = with(default.Vpar, matrix(c(1, 0, 0, 0), ncol = 2) * (SDx * timesSDx)^2),
parset = NULL, print.summary = TRUE, plotit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="errorsINseveral_+3A_n">n</code></td>
<td>

<p>Number of observations
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_a0">a0</code></td>
<td>

<p>Intercept in linear regression model
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_beta">beta</code></td>
<td>

<p>Regression coefficients.  If one coefficient only is given, this will
be repeated as many times as necessary
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_mu">mu</code></td>
<td>

<p>Vector of covariate means.
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_sdyerr">SDyerr</code></td>
<td>

<p>SD of $y$, conditional on the covariates measured without error
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_default.vpar">default.Vpar</code></td>
<td>

<p>Parameters for the default model with two explanatory variables,
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_v">V</code></td>
<td>

<p>Variance-covariance matrix for the z's, measured without error.
(These are generated from a multivariate normal distribution,
mainly as a matter of convenience)
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_xerrv">xerrV</code></td>
<td>

<p>Variance-covariance matrix for the added &ldquo;errors in x&rdquo;
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_parset">parset</code></td>
<td>

<p>Parameter list (theme) in a form suitable for supplying to
<code>trellis.par.set()</code>.
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_print.summary">print.summary</code></td>
<td>

<p>If <code>TRUE</code>, print summary details of the regression results from the
simulation.
</p>
</td></tr>
<tr><td><code id="errorsINseveral_+3A_plotit">plotit</code></td>
<td>

<p>If <code>TRUE</code>, plot the fitted values for the model with covariates with
error, against the fitted values for covariates without error.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With default arguments, simulates a model in which two
covariates are in contention, the first measured without error,
and the second with coefficient 0 in the model that includes
both covariates measured without error.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ERRfree</code></td>
<td>
<p>Data frame holding covariates without error, plus $y$</p>
</td></tr>
<tr><td><code>addedERR</code></td>
<td>
<p>Data frame holding covariates with error, plus $y$</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>References</h3>

<p>Data Analysis and Graphics Using R, 3rd edn, Section 6.8.1
</p>


<h3>See Also</h3>

<p><code><a href="#topic+errorsINx">errorsINx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lattice)
function(n=1000, a0=2.5, beta=c(1.5,0), mu=12.5, SDyerr=0.5, 
           default.Vpar=list(SDx=2, rho=-0.5, timesSDx=1.5),
           V=with(default.Vpar, matrix(c(1,rho,rho,1), ncol=2)*SDx^2),
           xerrV=with(default.Vpar, matrix(c(1,0,0,0), ncol=2)*(SDx*timesSDx)^2),
           parset=NULL, print.summary=TRUE, plotit=TRUE){
    m &lt;- dim(V)[1]
    if(length(mu)==1)mu &lt;- rep(mu,m)
    ow &lt;- options(warn=-1)
    xxmat &lt;- sweep(matrix(rnorm(m*n, 0, 1), ncol=m) %*% chol(V), 2, mu, "+")
    errxx &lt;- matrix(rnorm(m*n, 0, 1), ncol=m) %*% chol(xerrV, pivot=TRUE)
    options(ow)
    dimnames(xxmat)[[2]] &lt;- paste("z", 1:m, sep="")
    xxWITHerr &lt;- xxmat+errxx
    xxWITHerr &lt;- data.frame(xxWITHerr)
    names(xxWITHerr) &lt;- paste("xWITHerr", 1:m, sep="")
    xxWITHerr[, "y"] &lt;- a0 + xxmat %*% matrix(beta,ncol=1) + rnorm(n, sd=SDyerr)
    err.lm &lt;- lm(y ~ ., data=xxWITHerr)
    xx &lt;- data.frame(xxmat)
    names(xx) &lt;- paste("z", 1:m, sep="")
    xx$y &lt;- xxWITHerr$y
    xx.lm &lt;- lm(y ~ ., data=xx)
    B &lt;- coef(err.lm)
    b &lt;- coef(xx.lm)
    SE &lt;- summary(err.lm)$coef[,2]
    se &lt;- summary(xx.lm)$coef[,2]
    if(print.summary){
      beta0 &lt;- c(mean(xx$y)-sum(beta*apply(xx[,1:m],2,mean)), beta)
      tab &lt;- rbind(beta0, b, B)
      dimnames(tab) &lt;- list(c("Values for simulation",
                              "Estimates: no error in x1",
                              "LS Estimates: error in x1"),
                            c("Intercept", paste("b", 1:m, sep="")))
      tabSE &lt;- rbind(rep(NA,m+1),se,SE)
      rownames(tabSE) &lt;- rownames(tab)
      colnames(tabSE) &lt;- c("SE(Int)", paste("SE(", colnames(tab)[-1],")", sep=""))
      tab &lt;- cbind(tab,tabSE)
      print(round(tab,3))
    }
    if(m==2 &amp; print.summary){
      tau &lt;- default.Vpar$timesSDx
      s1 &lt;- sqrt(V[1,1])
      s2 &lt;- sqrt(V[2,2])
      rho &lt;- default.Vpar$rho
      s12 &lt;- s1*sqrt(1-rho^2)
      lambda &lt;- (1-rho^2)/(1-rho^2+tau^2)
      gam12 &lt;- rho*sqrt(V[1,1]/V[2,2])
      expB2 &lt;- beta[2]+beta[1]*(1-lambda)*gam12
      print(c("Theoretical attenuation of b1" = lambda, "Theoretical b2" = expB2))
    }
    if(is.null(parset))parset &lt;- simpleTheme(col=c("gray40","gray40"),
                                             col.line=c("black","black"))
    if(plotit){
      library(lattice)
      zhat &lt;- fitted(xx.lm)
      xhat &lt;- fitted(err.lm)
      plt &lt;- xyplot(xhat ~ zhat, aspect=1, scales=list(tck=0.5),
                    panel=function(x,y,...){
                      panel.xyplot(x,y,type="p",...)
                      panel.abline(lm(y ~ x), lty=2)
                      panel.abline(0,1)
                    },
                    xlab="Fitted values; regress on exact z",
                    ylab="Fitted values; regress on x = xWITHerr",
                    key=list(space="top", columns=2,
                      text=list(lab=c("Line y=x", "Regression fit to points")),
                      lines=list(lty=1:2)),
                    par.settings=parset
                    )
      print(plt)}
    invisible(list(ERRfree=xx, addedERR=xxWITHerr))
  }
</code></pre>

<hr>
<h2 id='errorsINx'>Simulate data for straight line regression, with &quot;errors in x&quot;.</h2><span id='topic+errorsINx'></span>

<h3>Description</h3>

<p>Simulates $y-$ and $x-$values for the straight line regression model,
but with $x-$values subject to random measurement error, following
the classical &ldquo;errors in x&rdquo; model. Optionally, the x-values
can be split into two groups, with one group shifted relative to the other
</p>


<h3>Usage</h3>

<pre><code class='language-R'>errorsINx(mu = 12.5, n = 200, a = 15, b = 1.5, SDx=2, SDyerr = 1.5,
           timesSDx=(1:5)/2.5, gpfactor=if(missing(gpdiff))FALSE else TRUE,
           gpdiff=if(gpfactor) 1.5 else 0, layout=NULL,
           parset = simpleTheme(alpha = 0.75, col = c("black","gray45"),
             col.line = c("black","gray45"), lwd=c(1,1.5), pch=c(1,2),
           lty=c(1,2)), print.summary=TRUE, plotit=TRUE, xrelation="same")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="errorsINx_+3A_mu">mu</code></td>
<td>
<p>Mean of $z$</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_n">n</code></td>
<td>
<p>Number of points</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_a">a</code></td>
<td>
<p>Intercept in model where $z$ is measured without error</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_b">b</code></td>
<td>
<p>Slope in model where $z$ is measured without error</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_sdx">SDx</code></td>
<td>
<p>SD of $z$-values, measured without error</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_sdyerr">SDyerr</code></td>
<td>
<p>SD of error term in <code>y</code> where $z$ is measured without error</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_timessdx">timesSDx</code></td>
<td>
<p>SD of measurement error is <code>timesSDx</code>, as a
multiple of <code>SDx</code></p>
</td></tr>
<tr><td><code id="errorsINx_+3A_gpfactor">gpfactor</code></td>
<td>
<p>Should x-values be split into two groups, with one
shifted relative to the other?</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_gpdiff">gpdiff</code></td>
<td>
<p>Amount of shift of one group of z-values relative to the other</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_layout">layout</code></td>
<td>
<p>Layout for lattice graph, if requested</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_parset">parset</code></td>
<td>
<p>Parameters to be supplied to the lattice plot, if any</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_print.summary">print.summary</code></td>
<td>
<p>Print summary information on fits?</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_plotit">plotit</code></td>
<td>
<p>logical: plot the data?</p>
</td></tr>
<tr><td><code id="errorsINx_+3A_xrelation">xrelation</code></td>
<td>
<p>character: sets the x-axis <code>relation</code> component of
<code>scales</code> to <code>"same"</code> or <code>"free"</code> or (though this does
not make make sense here) <code>"sliced"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>timesSDx</code> can be a numeric vector.
One set of $x$-values that are contaminated with measurement error
is simulated for each element of <code>timesSDx</code>.
</p>


<h3>Value</h3>

<table>
<tr><td><code>gph</code></td>
<td>
<p>the trellis graphics object</p>
</td></tr>
<tr><td><code>mat</code></td>
<td>

<p>A matrix, with <code>length(timesSDx)+2</code> columns.  Values of $z$ are
in the first column. There is one further column (x with error) for
each element of <code>timesSDx</code>, followed by a column for $y$.
If there is a grouping variable, a further column identifies the
groups.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald</p>


<h3>References</h3>

<p>Data Analysis and Graphics Using R, 3rd edn, Section 6.7</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lattice)
errorsINx()
errorsINx(gpdiff=2, timesSDx=1.25, SDyerr=2.5, n=80)
</code></pre>

<hr>
<h2 id='excessRisk'>Create and analyze multiway frequency or weighted frequency
table</h2><span id='topic+excessRisk'></span>

<h3>Description</h3>

<p>This function creates a multi-way table of counts
for the <code>response</code> given a set of classifying factors.  Output
facilitates a check on how the factor specified as <code>margin</code>
may, after accounting for other classifying factors, affect the
response.  </p>


<h3>Usage</h3>

<pre><code class='language-R'>excessRisk(form = weight ~ seatbelt + airbag, response = "dead", margin = "airbag",
data = DAAG::nassCDS, decpl = 4, printResults = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="excessRisk_+3A_form">form</code></td>
<td>
<p><code>form</code> is a formula in which classifying factors
appear on the right, with an optional weight variable on the left.</p>
</td></tr>
<tr><td><code id="excessRisk_+3A_response">response</code></td>
<td>
<p><code>response</code> is a binary variable or two-level
factor such that the response of interest is the relative number
in the two levels.
</p>
</td></tr>
<tr><td><code id="excessRisk_+3A_margin">margin</code></td>
<td>
<p><code>margin</code> is the factor whose effect on the
response,
after accounting for other classifying factors, is of interest</p>
</td></tr>
<tr><td><code id="excessRisk_+3A_data">data</code></td>
<td>
<p><code>data</code> is a data frame in which variables and
factors may be found</p>
</td></tr>
<tr><td><code id="excessRisk_+3A_decpl">decpl</code></td>
<td>
<p><code>decpl</code> is the number of decimal places in
proportions that appear in the output</p>
</td></tr>
<tr><td><code id="excessRisk_+3A_printresults">printResults</code></td>
<td>
<p>if <code>TRUE</code>, a tabular summary is printed.</p>
</td></tr>  
</table>


<h3>Details</h3>

<p>The best way to understand what this function does may be to run it
with the default parameters, and/or with examples that appear below.
</p>


<h3>Value</h3>

<p>The function returns a data frame, with one row for each combination of
levels of factors on the right of the formula, but excluding the
factor specified as <code>margin</code>.
The final three columns show the count for level 1 as a fraction
of the margin by total, the count for level 2 as a fraction of
the margin by total, and the excess count for level 2 of response in 
the row, under the assumption that, in that row, there is no association 
between <code>response</code> and <code>margin</code>.  This is the observed response
(for the default arguments, number of dead) for level 2 (airbag deployed),
less the number that would have been expected if the proportion
had been that for level 1. (Negative values favor airbags.)
</p>


<h3>Author(s)</h3>

<p>John Maindonald</p>


<h3>References</h3>

<p>See <code>help(nassCDS)</code></p>


<h3>See Also</h3>

<p><code>xtabs</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>excessRisk()
excessRisk(weight ~ airbag+seatbelt+dvcat)
UCB &lt;- as.data.frame.table(UCBAdmissions)
excessRisk(Freq~Gender, response="Admit", margin="Gender",data=UCB)
excessRisk(Freq~Gender+Dept, response="Admit", margin="Gender",data=UCB)
</code></pre>

<hr>
<h2 id='fossilfuel'>Fossil Fuel Data</h2><span id='topic+fossilfuel'></span>

<h3>Description</h3>

<p>Estimates of total worldwide carbon emissions from
fossil fuel use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fossilfuel</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>year</dt><dd><p>a numeric vector giving the year the measurement
was taken.</p>
</dd>
<dt>carbon</dt><dd><p>a numeric vector giving the total worldwide
carbon emissions from fossil fuel use, in millions of tonnes.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data for the years 1751 through to 2014 is available
from Data for the years 2014
<a href="https://cdiac.ess-dive.lbl.gov/ftp/ndp030/global.1751_2014.ems">https://cdiac.ess-dive.lbl.gov/ftp/ndp030/global.1751_2014.ems</a></p>


<h3>Source</h3>

<p>Boden TA, Marland G, Andres RJ (2017).
&ldquo;Global, Regional, and National Fossil-Fuel CO2 Emissions.&rdquo;
<a href="https://cdiac.ess-dive.lbl.gov/trends/emis/meth_reg.html">https://cdiac.ess-dive.lbl.gov/trends/emis/meth_reg.html</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(fossilfuel)
</code></pre>

<hr>
<h2 id='frogs'>Frogs Data</h2><span id='topic+frogs'></span>

<h3>Description</h3>

<p>The <code>frogs</code> data frame has 212 rows and 11 columns.
The data are on the distribution of the Southern Corroboree
frog, which occurs in the Snowy Mountains area of New South Wales,
Australia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frogs</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>pres.abs</dt><dd><p>0 = frogs were absent, 1 = frogs were present</p>
</dd>
<dt>northing</dt><dd><p>reference point</p>
</dd>
<dt>easting</dt><dd><p>reference point</p>
</dd>
<dt>altitude</dt><dd><p>altitude , in meters</p>
</dd>
<dt>distance</dt><dd><p>distance in meters to nearest extant population</p>
</dd>
<dt>NoOfPools</dt><dd><p>number of potential breeding pools</p>
</dd>
<dt>NoOfSites</dt><dd><p>(number of potential breeding sites within a 2 km 
radius</p>
</dd>
<dt>avrain</dt><dd><p>mean rainfall for Spring period</p>
</dd>
<dt>meanmin</dt><dd><p>mean minimum Spring temperature</p>
</dd>
<dt>meanmax</dt><dd><p>mean maximum Spring temperature</p>
</dd>
</dl>



<h3>Source</h3>

<p>Hunter, D. (2000) The conservation and demography of 
the southern corroboree frog (Pseudophryne corroboree). M.Sc. thesis,
University of Canberra, Canberra.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Multiple Logistic Regression - Example 8.2")

plot(northing ~ easting, data=frogs, pch=c(1,16)[frogs$pres.abs+1],
  xlab="Meters east of reference point", ylab="Meters north")
pairs(frogs[,4:10])
attach(frogs)
pairs(cbind(altitude,log(distance),log(NoOfPools),NoOfSites),
  panel=panel.smooth, labels=c("altitude","log(distance)",
  "log(NoOfPools)","NoOfSites"))
detach(frogs)

frogs.glm0 &lt;- glm(formula = pres.abs ~ altitude + log(distance) +
  log(NoOfPools) + NoOfSites + avrain + meanmin + meanmax,
  family = binomial, data = frogs)
summary(frogs.glm0)

frogs.glm &lt;- glm(formula = pres.abs ~ log(distance) + log(NoOfPools) + 
meanmin +
  meanmax, family = binomial, data = frogs)
oldpar &lt;- par(mfrow=c(2,2))
termplot(frogs.glm, data=frogs)

termplot(frogs.glm, data=frogs, partial.resid=TRUE)

cv.binary(frogs.glm0)   # All explanatory variables
pause()

cv.binary(frogs.glm)    # Reduced set of explanatory variables

for (j in 1:4){
 rand &lt;- sample(1:10, 212, replace=TRUE)
 all.acc &lt;- cv.binary(frogs.glm0, rand=rand, print.details=FALSE)$acc.cv
 reduced.acc &lt;- cv.binary(frogs.glm, rand=rand, print.details=FALSE)$acc.cv
 cat("\nAll:", round(all.acc,3), "  Reduced:", round(reduced.acc,3))
}

</code></pre>

<hr>
<h2 id='frostedflakes'>Frosted Flakes data</h2><span id='topic+frostedflakes'></span>

<h3>Description</h3>

<p>The <code>frosted flakes</code> data frame has 101 rows and 2 columns
giving the sugar concentration (in percent) for 25 g samples
of a cereal as measured by 2 methods &ndash; high performance liquid
chromatography (a slow accurate lab method) and a quick method
using the infra-analyzer 400.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elastic1</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Lab</dt><dd><p>careful laboratory analysis measurements using
high performance liquid chromatography</p>
</dd>
<dt>IA400</dt><dd><p>measurements based on the infra-analyzer 400</p>
</dd>
</dl>



<h3>Source</h3>

<p>W. J. Braun
</p>

<hr>
<h2 id='fruitohms'>Electrical Resistance of Kiwi Fruit</h2><span id='topic+fruitohms'></span>

<h3>Description</h3>

<p>Data are from a study that examined how the electrical
resistance of a slab of kiwifruit changed with the apparent
juice content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fruitohms</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>juice</dt><dd><p>apparent juice content (percent) </p>
</dd>
<dt>ohms</dt><dd><p>electrical resistance (in ohms)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Harker, F. R. and Maindonald J.H. 1994. Ripening of nectarine
fruit. <em>Plant Physiology</em> 106: 165 - 171.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(ohms ~ juice, xlab="Apparent juice content (%)",ylab="Resistance (ohms)", data=fruitohms)
lines(lowess(fruitohms$juice, fruitohms$ohms), lwd=2)
pause()

require(splines)
attach(fruitohms)
plot(ohms ~ juice, cex=0.8, xlab="Apparent juice content (%)",
     ylab="Resistance (ohms)", type="n")
fruit.lmb4 &lt;- lm(ohms ~ bs(juice,4))
ord &lt;- order(juice)
lines(juice[ord], fitted(fruit.lmb4)[ord], lwd=2)
ci &lt;- predict(fruit.lmb4, interval="confidence")
lines(juice[ord], ci[ord,"lwr"])
lines(juice[ord], ci[ord,"upr"])
</code></pre>

<hr>
<h2 id='gaba'>Effect of pentazocine on post-operative pain (average VAS scores)</h2><span id='topic+gaba'></span>

<h3>Description</h3>

<p>The table shows, separately for males and females, the effect 
of pentazocine on post-operative pain profiles (average VAS scores), 
with (mbac and fbac) and without (mpl and fpl) preoperatively administered 
baclofen.  Pain scores are recorded every 20 minutes, from 10 minutes
to 170 minutes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaba</code></pre>


<h3>Format</h3>

<p>A data frame with 9 observations on the following 7 variables.
</p>

<dl>
<dt><code>min</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>mbac</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>mpl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>fbac</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>fpl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>avbac</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>avplac</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>15 females were given baclofen, as against 3 males.
7 females received the placebo, as against 16 males.
Averages for the two treatments (baclofen/placebo),
taken over all trial participants and ignoring sex,
are misleading.
</p>


<h3>Source</h3>

<p>Gordon, N. C. et al.(1995): 'Enhancement of Morphine Analgesia 
by the GABA<code class="reqn">_B</code> against Baclofen'. <em>Neuroscience</em> 69: 345-349.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gaba)
mr &lt;- range(gaba$min)
tran &lt;- range(gaba[, c("mbac","mpl","fbac","fpl")])
## Means by treatment and sex
par(mfrow=c(1,2))
plot(mr, tran, xlab = "Time post pentazocine (min)",
     ylab = "Reduction in VAS pain rating", 
     type = "n", xlim = c(0, 170), ylim = tran)
points(gaba$min, gaba$fbac, pch = 1, col = 8, lwd = 2, lty = 2, 
       type = "b")
points(gaba$min, gaba$fpl, pch = 0, col = 8, lwd = 2, lty = 2, 
       type = "b")
points(gaba$min, gaba$mbac, pch = 16, col = 8, lty = 2, type = "b")
points(gaba$min, gaba$mpl, pch = 15, col = 8, lty = 2, type = "b")
box()
## Now plot means, by treatment, averaged over all participants
plot(mr, tran, xlab = "Time post pentazocine (min)",
     ylab = "Reduction in VAS pain rating", 
     type = "n", xlim = c(0, 170), ylim = tran)
bac &lt;- (15 * gaba$fbac + 3 * gaba$mbac)/18
plac &lt;- (7 * gaba$fpl + 9 * gaba$mpl)/16
points(gaba$min, plac, pch = 15, lty = 1, col=1, type = "b")
points(gaba$min, bac, pch = 16, lty = 1, col=1, type = "b")
box()
par(mfrow=c(1,1))
</code></pre>

<hr>
<h2 id='geophones'>Seismic Timing Data</h2><span id='topic+geophones'></span>

<h3>Description</h3>

<p>The <code>geophones</code> data frame has 56 rows and 2 columns.
Thickness of a layer of Alberta substratum as measured by
a line of geophones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geophones</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>distance</dt><dd><p>location of geophone.</p>
</dd>
<dt>thickness</dt><dd><p>time for signal to pass through substratum.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>plot(geophones)
lines(lowess(geophones, f=.25))
</code></pre>

<hr>
<h2 id='greatLakes'>
Yearly averages of Great Lake heights: 1918 - 2009
</h2><span id='topic+greatLakes'></span>

<h3>Description</h3>

<p>Heights, stored as a multivariate time series, are for the lakes Erie,
Michigan/Huron, Ontario and St Clair
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(greatLakes)</code></pre>


<h3>Format</h3>

<p>The format is:
mts [1:92, 1:4] 174 174 174 174 174 ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : NULL
..$ : chr [1:4] &quot;Erie&quot; &quot;michHuron&quot; &quot;Ontario&quot; &quot;StClair&quot;
- attr(*, &quot;tsp&quot;)= num [1:3] 1918 2009 1
- attr(*, &quot;class&quot;)= chr [1:2] &quot;mts&quot; &quot;ts&quot;
</p>


<h3>Details</h3>

<p>For more details, go to the website that is the source of the data.
</p>


<h3>Source</h3>

<p><a href="https://www.lre.usace.army.mil/Missions/Great-Lakes-Information/Great-Lakes-Information-2/Water-Level-Data/">https://www.lre.usace.army.mil/Missions/Great-Lakes-Information/Great-Lakes-Information-2/Water-Level-Data/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(greatLakes)
plot(greatLakes)
## maybe str(greatLakes)
</code></pre>

<hr>
<h2 id='grog'>Alcohol consumption in Australia and New Zealand</h2><span id='topic+grog'></span>

<h3>Description</h3>

<p>Data are annual apparent alcohol consumption in  Australia and New
Zealand, in liters of pure alcohol content per annum, separately for
beer, wine, and spirits (including spirit-based products).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(grog)</code></pre>


<h3>Format</h3>

<p>A data frame with 18 observations on the following 5 variables.
</p>

<dl>
<dt><code>Beer</code></dt><dd><p>liters per annum</p>
</dd>
<dt><code>Wine</code></dt><dd><p>liters per annum</p>
</dd>
<dt><code>Spirit</code></dt><dd><p>liters per annum</p>
</dd>
<dt><code>Country</code></dt><dd><p>a factor with levels <code>Australia</code>
<code>NewZealand</code></p>
</dd>
<dt><code>Year</code></dt><dd><p>Year ending in June of the given year</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data are total available pure alcohol content, for the three
categories, divided by numbers of persons aged 15 years or more.  The
source data for New Zealand included quarterly figures from December
1997, and annual data to December for all years.  The annual New
Zealand figure to June 1998 required an estimate for September
1997 that was obtained by extrapolating back the third quarter trend
line from later years.
</p>


<h3>Source</h3>

<p>Australian data are from <a href="https://www.abs.gov.au">https://www.abs.gov.au</a>.
For New Zealand data, go to
<a href="https://infoshare.stats.govt.nz/">https://infoshare.stats.govt.nz/</a> Click on 'Industry sectors'
and then on 'Alcohol Available for Consumption - ALC'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grog)
library(lattice)
xyplot(Beer+Wine+Spirit ~ Year | Country, data=grog)
xyplot(Beer+Wine+Spirit ~ Year, groups=Country, data=grog, outer=TRUE)
</code></pre>

<hr>
<h2 id='hardcopy'>Graphical Output for Hardcopy</h2><span id='topic+hardcopy'></span>

<h3>Description</h3>

<p>This function streamlines graphical output to the screen, pdf or ps
files. File names for hard copy devices can be generated
automatically from function names of the form <code>g3.2</code> or
<code>fig3.2</code> (the choice of alphabetic characters prior to
<code>3.2</code> is immaterial).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hardcopy(width = 3.75, height = 3.75, color = FALSE, trellis = FALSE,
                 device = c("", "pdf", "ps"), path = getwd(), file =
                 NULL, format = c("nn-nn", "name"), split = "\\.",
                 pointsize = c(8, 4), fonts=NULL, horiz = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hardcopy_+3A_width">width</code></td>
<td>
<p>width of plot in inches (sic!)</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_height">height</code></td>
<td>
<p>height of plot in inches (sic!)</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_color">color</code></td>
<td>
<p>(lattice plots only) TRUE if plot is not black on white only</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_trellis">trellis</code></td>
<td>
<p>TRUE if plot uses trellis graphics</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_device">device</code></td>
<td>
<p>screen &quot;&quot;, pdf or ps</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_path">path</code></td>
<td>
<p>external path name</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_file">file</code></td>
<td>
<p>name of file to hold output, else <code>NULL</code></p>
</td></tr>
<tr><td><code id="hardcopy_+3A_format">format</code></td>
<td>
<p>Alternatives are <code>"nn-nn"</code> and <code>"name"</code>.</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_split">split</code></td>
<td>
<p>character on which to split function name (file=NULL)</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_pointsize">pointsize</code></td>
<td>
<p>Pointsize. For trellis devices a vector of length
2 giving font sizes for text and for points respectively</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_fonts">fonts</code></td>
<td>
<p>For postscript devices, specify families that will be
used in addition to the intial device</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_horiz">horiz</code></td>
<td>
<p>FALSE for landscape mode; applies only to postscript
files</p>
</td></tr>
<tr><td><code id="hardcopy_+3A_...">...</code></td>
<td>
<p>Other arguments for passing to the <code>pdf</code> or
<code>postscript</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a file name (<code>file</code>, without extension) is not
supplied, the <code>format</code> argument determines how the name is
constructed.  With <code>format="name"</code>, the function name is used.
With <code>format="nn-nn"</code> and <code>dotsplit</code> unchanged from the
default, a function name of the form g3.1 leads to the name
<code>03-01</code>.  Here <code>g</code> can be replaced by any other non-numeric
characters; the result is the same. The relevant extension is in any
case added.
</p>


<h3>Value</h3>

<p>Graphical output to screen, pdf or ps file. 
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code><a href="grDevices.html#topic+postscript">postscript</a></code></p>

<hr>
<h2 id='headInjury'>Minor Head Injury (Simulated) Data</h2><span id='topic+headInjury'></span><span id='topic+head.injury'></span>

<h3>Description</h3>

<p>The <code>headInjury</code> data frame has 3121 rows and 11 columns.
The data were simulated according to a simple logistic
regression model to match roughly the clinical characteristics
of a sample of individuals who suffered minor head injuries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>headInjury</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>age.65</dt><dd><p>age factor (0 = under 65, 1 = over 65).</p>
</dd>
<dt>amnesia.before</dt><dd><p>amnesia before impact (less than 30 minutes = 0, 
more 
than 30 minutes =1).</p>
</dd>
<dt>basal.skull.fracture</dt><dd><p>(0 = no fracture, 1 = fracture).</p>
</dd>
<dt>GCS.decrease</dt><dd><p>Glasgow Coma Scale decrease (0 = no 
deterioration, 1 = deterioration).</p>
</dd>
<dt>GCS.13</dt><dd><p>initial Glasgow Coma Scale (0 = not &lsquo;13&rsquo;, 1 = &lsquo;13&rsquo;). </p>
</dd>
<dt>GCS.15.2hours</dt><dd><p>Glasgow Coma Scale after 2 hours
(0 = not &lsquo;15&rsquo;, 1 = '15').</p>
</dd>
<dt>high.risk</dt><dd><p>assessed by clinician as high risk for neurological
intervention (0 = not high risk, 1 = high risk).</p>
</dd>
<dt>loss.of.consciousness</dt><dd><p>(0 = conscious, 1 = loss of 
consciousness).</p>
</dd>
<dt>open.skull.fracture</dt><dd><p>(0 = no fracture, 1 = fracture) </p>
</dd>
<dt>vomiting</dt><dd><p>(0 = no vomiting, 1 = vomiting)</p>
</dd>
<dt>clinically.important.brain.injury</dt><dd><p>any acute brain finding
revealed on CT (0 = not present, 1 = present).</p>
</dd>
</dl>



<h3>References</h3>

<p>Stiell, I.G., Wells, G.A., Vandemheen, K., Clement, C., Lesiuk, H.,
Laupacis, A., McKnight, R.D., Verbee, R., Brison, R., Cass, D., 
Eisenhauer, M., Greenberg, G.H., and Worthington, J. (2001) 
The Canadian CT Head Rule for Patients with Minor Head Injury,
The Lancet. 357: 1391-1396. 
</p>

<hr>
<h2 id='hills'>Scottish Hill Races Data</h2><span id='topic+hills'></span><span id='topic+hills2000'></span>

<h3>Description</h3>

<p>The record times in 1984 (<code>hills</code>) for 35 Scottish hill races,
or in 2000 (<code>hills2000</code>) for 56 hill races.  The <code>hills2000</code>
dataset is the subset of <code><a href="#topic+races2000">races2000</a></code> for which <code>type</code>
is <code>hill</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(hills)
  data(hills2000)
  </code></pre>


<h3>Format</h3>


<dl>
<dt>dist</dt><dd><p>distance, in miles (on the map)</p>
</dd>
<dt>climb</dt><dd><p>total height gained during the route, in feet</p>
</dd>
<dt>time</dt><dd><p>record time in hours</p>
</dd>
<dt>timef</dt><dd><p>record time in hours for females, in the
<code>hills2000</code> dataset.
</p>
</dd>
</dl>



<h3>Source</h3>

<p>A.C. Atkinson (1986) Comment: Aspects of diagnostic regression
analysis. Statistical Science  1, 397-402.
</p>
<p>Also, in MASS library, with time in minutes.
</p>
<p>The Scottish Running Resource, http://www.hillrunning.co.uk
</p>


<h3>References</h3>

<p>A.C. Atkinson (1988) Transformations unmasked. Technometrics 30,
311-318. [ &quot;corrects&quot; the time for Knock Hill, in the <code>hills</code>
dataset, from 78.65 to 18.65. It   
is unclear if this based on the original records.]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Transformation - Example 6.4.3")
pairs(hills, labels=c("dist\n\n(miles)", "climb\n\n(feet)", 
"time\n\n(hours)"))
pause()

pairs(log(hills), labels=c("dist\n\n(log(miles))", "climb\n\n(log(feet))",
  "time\n\n(log(hours))"))
pause()

hills0.loglm &lt;- lm(log(time) ~ log(dist) + log(climb), data = hills)  
oldpar &lt;- par(mfrow=c(2,2))
plot(hills0.loglm)
pause()


hills.loglm &lt;- lm(log(time) ~ log(dist) + log(climb), data = hills[-18,])
summary(hills.loglm) 
plot(hills.loglm)
pause()

hills2.loglm &lt;- lm(log(time) ~ log(dist)+log(climb)+log(dist):log(climb), 
data=hills[-18,])
anova(hills.loglm, hills2.loglm)
pause()

step(hills2.loglm)
pause()

summary(hills.loglm, corr=TRUE)$coef
pause()

summary(hills2.loglm, corr=TRUE)$coef
par(oldpar)
pause()

print("Nonlinear - Example 6.9.4")
hills.nls0 &lt;- nls(time ~ (dist^alpha)*(climb^beta), start =
   c(alpha = .909, beta = .260), data = hills[-18,])
summary(hills.nls0)
plot(residuals(hills.nls0) ~ predict(hills.nls0)) # residual plot
pause()

hills$climb.mi &lt;- hills$climb/5280
hills.nls &lt;- nls(time ~ alpha + beta*dist + gamma*(climb.mi^delta),
  start=c(alpha = 1, beta = 1, gamma = 1, delta = 1), data=hills[-18,])
summary(hills.nls)
plot(residuals(hills.nls) ~ predict(hills.nls)) # residual plot


</code></pre>

<hr>
<h2 id='hotspots'>Hawaian island chain hotspot Potassium-Argon ages</h2><span id='topic+hotspots'></span>

<h3>Description</h3>

<p>K-Ar Ages (millions of years) and distances (km) from Kilauea 
along the trend of the chain of Hawaian volcanic islands and
other seamounts that are believed to have been created by a
moving &quot;hot spot&quot;. The age of Kilauea is given as 0-0.4 Ma.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hotspots)</code></pre>


<h3>Format</h3>

<p>A data frame with 36 observations on the following 6 variables.
</p>

<dl>
<dt><code>ID</code></dt><dd><p>Volcano identifier</p>
</dd>
<dt><code>name</code></dt><dd><p>Name</p>
</dd>
<dt><code>distance</code></dt><dd><p>Distance in kilometers</p>
</dd>
<dt><code>age</code></dt><dd><p>K-Ar age in millions of years</p>
</dd>
<dt><code>error</code></dt><dd><p>Standard error of estimate?</p>
</dd>
<dt><code>source</code></dt><dd><p>Data source; see information on web site below.</p>
</dd>
</dl>



<h3>Details</h3>

<p>For details of the way that errors werre calculated, refer to the
original papers. See also the comments under <code>hotspots2006</code>.
In general, errors do not account for geological uncertainty.
</p>


<h3>Source</h3>

<p><a href="http://www.soest.hawaii.edu/GG/HCV/haw_formation.html">http://www.soest.hawaii.edu/GG/HCV/haw_formation.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hotspots)
plot(age ~ distance, data=hotspots)
abline(lm(age ~ distance, data=hotspots))
</code></pre>

<hr>
<h2 id='hotspots2006'>Hawaian island chain hotspot Argon-Argon ages</h2><span id='topic+hotspots2006'></span>

<h3>Description</h3>

<p>Ar-Ar Ages (millions of years) and distances (km) from Kilauea 
along the trend of the chain of Hawaian volcanic islands and
other seamounts that are believed to have been created by a
moving &quot;hot spot&quot;. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hotspots2006)</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations on the following 6 variables.
</p>

<dl>
<dt><code>age</code></dt><dd><p>Ar-Ar age</p>
</dd>
<dt><code>CI95lim</code></dt><dd><p>Measurement error; 95% CI</p>
</dd>
<dt><code>geoErr</code></dt><dd><p>Geological Uncertainty</p>
</dd>
<dt><code>totplus</code></dt><dd><p>Total uncertainty (+)</p>
</dd>
<dt><code>totminus</code></dt><dd><p>Total uncertainty (-)</p>
</dd>
<dt><code>distance</code></dt><dd><p>Distance in kilometers</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note that measurement error is small relative to geological
uncertainty. Geological uncertainty arises because lavas are
likely to have erupted, over a period of up to 2 million years,
somewhat after passage over the hot spot's centre.  Dredging
or drilling will in general have accessed larvas from the
younger half of this interval.  Hence the asymmetry in the
geological uncertainty.
</p>


<h3>Source</h3>

<p>Warren D. Sharp and David A. Clague, 50-Ma initiation of
Hawaiian-Emperor bend records major change in Pacific Plate
motion. Science 313: 1281-1284 (2006).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hotspots2006)
</code></pre>

<hr>
<h2 id='houseprices'>Aranda House Prices</h2><span id='topic+houseprices'></span>

<h3>Description</h3>

<p>The <code>houseprices</code> data frame consists of the floor
area, price, and the number
of bedrooms for a sample of houses sold in Aranda in 1999. 
Aranda is a suburb of Canberra, Australia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>houseprices</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>area</dt><dd><p>a numeric vector giving the floor area</p>
</dd>
<dt>bedrooms</dt><dd><p>a numeric vector giving the number of bedrooms</p>
</dd>
<dt>sale.price</dt><dd><p>a numeric vector giving the sale price
in thousands of Australian dollars</p>
</dd>
</dl>



<h3>Source</h3>

<p>J.H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(sale.price~area, data=houseprices)
pause()

coplot(sale.price~area|bedrooms, data=houseprices)
pause()

print("Cross-Validation - Example 5.5.2")

houseprices.lm &lt;- lm(sale.price ~ area, data=houseprices)
summary(houseprices.lm)$sigma^2
pause()

CVlm()
pause()

print("Bootstrapping - Example 5.5.3")
houseprices.fn &lt;- function (houseprices, index){
house.resample &lt;- houseprices[index,]
house.lm &lt;- lm(sale.price ~ area, data=house.resample)
coef(house.lm)[2]    # slope estimate for resampled data
}
require(boot)       # ensure that the boot package is loaded
houseprices.boot &lt;- boot(houseprices, R=999, statistic=houseprices.fn)

houseprices1.fn &lt;- function (houseprices, index){
house.resample &lt;- houseprices[index,]
house.lm &lt;- lm(sale.price ~ area, data=house.resample)
predict(house.lm, newdata=data.frame(area=1200))
}

houseprices1.boot &lt;- boot(houseprices, R=999, statistic=houseprices1.fn)
boot.ci(houseprices1.boot, type="perc") # "basic" is an alternative to "perc"
houseprices2.fn &lt;- function (houseprices, index){
house.resample &lt;- houseprices[index,]
house.lm &lt;- lm(sale.price ~ area, data=house.resample)
houseprices$sale.price-predict(house.lm, houseprices)  # resampled prediction errors
}

n &lt;- length(houseprices$area)
R &lt;- 200   
houseprices2.boot &lt;- boot(houseprices, R=R, statistic=houseprices2.fn)
house.fac &lt;- factor(rep(1:n, rep(R, n)))
plot(house.fac, as.vector(houseprices2.boot$t), ylab="Prediction Errors", 
xlab="House")
pause()

plot(apply(houseprices2.boot$t,2, sd)/predict.lm(houseprices.lm, se.fit=TRUE)$se.fit,
     ylab="Ratio of Bootstrap SE's to Model-Based SE's", xlab="House", pch=16)
abline(1,0)

</code></pre>

<hr>
<h2 id='humanpower'>Oxygen uptake versus mechanical power, for humans</h2><span id='topic+humanpower1'></span><span id='topic+humanpower2'></span>

<h3>Description</h3>

<p>Data are from Daedalus project; see the reference below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(humanpower1)</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations on the following 3 variables.
</p>

<dl>
<dt>wattsPerKg</dt><dd><p>a numeric vector: watts per kilogram of body weight</p>
</dd>
<dt>o2</dt><dd><p>a numeric vector: ml/min/kg</p>
</dd>
<dt>id</dt><dd><p>a factor with levels 1 - 5 (<code>humanpower1</code>)
or 1 - 4 (<code>humanpower2</code>), identifying the different athletes</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data in <code>humanpower1</code> are from investigations (Bussolari 1987)
designed to assess the feasibility of a proposed 119 kilometer human
powered flight from the island of Crete &ndash; in the initial phase of the
Daedalus project.  Data are for five athletes
&ndash; a female hockey player, a male amateur tri-athlete, a female
amateur triathlete, a male wrestler and a male cyclist &ndash; who were
selected from volunteers who were recruited through the news media,
Data in <code>humanpower2) are for four out of the 25 applicants who
    were selected for further testing, in the lead-up to the eventual
    selection of a pilot for the Daedalus project (Nadel and Bussolari 1988).
</code>
</p>


<h3>Source</h3>

<p>Bussolari, S.R.(1987). Human factors of long-distance human-powered
aircraft flights.  Human Power 5: 8-12.
</p>
<p>Nadel and Bussolari, S.R.(1988). The Daedalus project: physiological
problems and solutions.  American Scientist 76: 351-360.
</p>


<h3>References</h3>

<p>Nadel and Bussolari, S.R.(1989). The physiological limits of
long-duration human-power production &ndash; lessons learned from
the Daedalus project.  Human Power 7: 7-10.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(humanpower1)
plot(humanpower1)
lm(o2 ~ id + wattsPerKg:id, data=humanpower1)
lm(o2 ~ id + wattsPerKg:id, data=humanpower2)
</code></pre>

<hr>
<h2 id='hurricNamed'>
Named US Atlantic Hurricanes
</h2><span id='topic+hurricNamed'></span>

<h3>Description</h3>

<p>Details are given of atmospheric pressure at landfall, estimated
damage in millions of dollars, and deaths, for named hurricanes
that made landfall in the US mainland from 1950 through to 2012.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("hurricNamed")</code></pre>


<h3>Format</h3>

<p>A data frame with 94 observations on the following 11 variables.
</p>

<dl>
<dt><code>Name</code></dt><dd><p>Hurricane name</p>
</dd>
<dt><code>Year</code></dt><dd><p>Numeric</p>
</dd>
<dt><code>LF.WindsMPH</code></dt><dd><p>Maximum sustained windspeed (&gt;= 1 minute) to occur along
the US coast. Prior to 1980, this is estimated from the maximum
windspeed associated with the Saffir-Simpson index at landfall.
If 2 or more landfalls, the maximum is taken</p>
</dd>
<dt><code>LF.PressureMB</code></dt><dd><p>Atmospheric pressure at landfall in
millibars. If 2 or more landfalls, the minimum is taken</p>
</dd>
<dt><code>LF.times</code></dt><dd><p>Date of first landfall</p>
</dd>
<dt><code>BaseDam2014</code></dt><dd><p>Property damage (millions of 2014 US dollars)</p>
</dd>
<dt><code>BaseDamage</code></dt><dd><p>Property damage (in millions of dollars for that year)</p>
</dd>
<dt><code>NDAM2014</code></dt><dd><p>Damage, had hurricane appeared in 2014</p>
</dd>
<dt><code>AffectedStates</code></dt><dd><p>Affected states (2-digit
abbreviations), pasted together</p>
</dd>
<dt><code>firstLF</code></dt><dd><p>Date of first landfall</p>
</dd>
<dt><code>deaths</code></dt><dd><p>Number of continental US direct and indirect deaths</p>
</dd>
<dt><code>mf</code></dt><dd><p>Gender of name; a factor with levels <code>f</code> <code>m</code></p>
</dd>
</dl>



<h3>Details</h3>

<p>An earlier version of these data was the subject of a controversial
paper that claimed to have found that hurricanes with female names,
presumably because taken less seriously, did more human damage after
adjusting for the severity of the storm than those with male names.
</p>


<h3>Source</h3>

<p><a href="https://www.icat.com/storms/catastrophe-resources">https://www.icat.com/storms/catastrophe-resources</a>
Deaths except for Audrey and Katrina, are in the Excel file
that is available from
<a href="https://www.pnas.org/doi/10.1073/pnas.1402786111">https://www.pnas.org/doi/10.1073/pnas.1402786111</a>
</p>
<p>NOAA Monthly Weather Reports (MWRs) supplied the numbers of deaths for
all except Donna, Celia, Audrey and Katrina.  The figure for Celia is
from <a href="https://www.nhc.noaa.gov/pdf/NWS-TPC-5.pdf">https://www.nhc.noaa.gov/pdf/NWS-TPC-5.pdf</a>.  For the other
three hurricanes, it is from the Atlantic hurricane list in Wikipedia
(see the references.)
</p>


<h3>References</h3>

<p><a href="https://www.aoml.noaa.gov/hrd/hurdat/mwr_pdf/">https://www.aoml.noaa.gov/hrd/hurdat/mwr_pdf/</a>
<a href="https://en.wikipedia.org/wiki/List_of_Atlantic_hurricanes">https://en.wikipedia.org/wiki/List_of_Atlantic_hurricanes</a>
<a href="https://www.nhtsa.gov/file-downloads">https://www.nhtsa.gov/file-downloads</a>
</p>
<p>Jung, Kiju, et al. &quot;Female hurricanes are deadlier than male hurricanes.&quot; Proceedings of the National Academy of Sciences 111.24 (2014): 8782-8787.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hurricNamed)
str(hurricNamed)
plot(log(deaths+0.5) ~ log(NDAM2014), data=hurricNamed)
with(hurricNamed, lines(lowess(log(deaths+0.5) ~ log(NDAM2014))))
plot(log(deaths+0.5) ~ I(NDAM2014^0.14), data=hurricNamed)
with(hurricNamed, lines(lowess(log(deaths+0.1) ~ I(NDAM2014^0.14))))
</code></pre>

<hr>
<h2 id='intersalt'>Blood pressure versus Salt; inter-population data</h2><span id='topic+intersalt'></span>

<h3>Description</h3>

<p>Median blood pressure, as a fuction of salt intake, for each of 52
human populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intersalt</code></pre>


<h3>Format</h3>

<p>A data frame with 52 observations on the following 4 variables.
</p>

<dl>
<dt><code>b</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>bp</code></dt><dd><p>mean diastolic blood pressure (mm Hg)</p>
</dd>
<dt><code>na</code></dt><dd><p>mean sodium excretion (mmol/24h)</p>
</dd>
<dt><code>country</code></dt><dd><p>a character vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>For each population took a sample of 25 males and 25 females from each
decade in the age range 20 - 50, i.e. 200 individuals in all.
</p>


<h3>Source</h3>

<p>Intersalt Cooperative Research Group. 1988.  Intersalt: an international
study of electrolyte excretion and blood pressure: results for 24 hour
urinary sodium and potassium excretion.  <em>British Medical Journal</em>
297: 319-328.  
</p>


<h3>References</h3>

<p>Maindonald, J.H. <em>The Design of Research Studies ? A Statistical
Perspective</em>, viii + 109pp. Graduate School Occasional Paper 00/2, 
Australian National University 2000.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(intersalt)
plot(bp ~ na, data=intersalt, xlab="Median sodium excretion (mmol/24h)",
     ylab="Median diatoluc blood pressure (mm Hg)")
</code></pre>

<hr>
<h2 id='ironslag'>Iron Content Measurements</h2><span id='topic+ironslag'></span>

<h3>Description</h3>

<p>The <code>ironslag</code> data frame has 53 rows and 2 columns.
Two methods for measuring the iron content in samples of slag 
were compared, a chemical and a magnetic method.  The chemical
method requires greater effort than the magnetic method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ironslag</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>chemical</dt><dd><p>a numeric vector containing the measurements
coming from the chemical method</p>
</dd>
<dt>magnetic</dt><dd><p>a numeric vector containing the measurments
coming from the magnetic method</p>
</dd>
</dl>



<h3>Source</h3>

<p>Hand, D.J., Daly, F., McConway, K., Lunn, D., and Ostrowski, E. eds (1993)
A Handbook of Small Data Sets. London: Chapman &amp; Hall. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>iron.lm &lt;- lm(chemical ~ magnetic, data = ironslag)
oldpar &lt;- par(mfrow = c(2,2))
plot(iron.lm)
par(oldpar)
</code></pre>

<hr>
<h2 id='jobs'>Canadian Labour Force Summary Data (1995-96)</h2><span id='topic+jobs'></span>

<h3>Description</h3>

<p>The number of workers in the Canadian
labour force broken down by region (BC, Alberta, Prairies,
Ontario, Quebec, Atlantic) for the 24-month period from January,
1995 to December, 1996 (a time when Canada was emerging from a
deep economic recession).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jobs</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>BC</dt><dd><p>monthly labour force counts in British Columbia</p>
</dd>
<dt>Alberta</dt><dd><p>monthly labour force counts in Alberta</p>
</dd>
<dt>Prairies</dt><dd><p>monthly labour force counts in Saskatchewan and 
Manitoba</p>
</dd>
<dt>Ontario</dt><dd><p>monthly labour force counts in Ontario</p>
</dd>
<dt>Quebec</dt><dd><p>monthly labour force counts in Quebec</p>
</dd>
<dt>Atlantic</dt><dd><p>monthly labour force counts in Newfoundland, Nova 
Scotia, Prince Edward Island and New Brunswick</p>
</dd>
<dt>Date</dt><dd><p>year (in decimal form)</p>
</dd>
</dl>



<h3>Details</h3>

<p>These data have been seasonally adjusted.
</p>


<h3>Source</h3>

<p>Statistics Canada
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Multiple Variables and Times - Example 2.1.4")
sapply(jobs, range)
pause()

matplot(jobs[,7], jobs[,-7], type="l", xlim=c(95,97.1))
 # Notice that we have been able to use a data frame as the second argument to matplot().
 # For more information on matplot(), type help(matplot)
text(rep(jobs[24,7], 6), jobs[24,1:6], names(jobs)[1:6], adj=0)
pause()

sapply(log(jobs[,-7]), range)
apply(sapply(log(jobs[,-7]), range), 2, diff)
pause()

oldpar &lt;- par(mfrow=c(2,3))
range.log &lt;- sapply(log(jobs[,-7], 2), range)
maxdiff &lt;- max(apply(range.log, 2, diff))
range.log[2,] &lt;- range.log[1,] + maxdiff
titles &lt;- c("BC Jobs","Alberta Jobs","Prairie Jobs",
   "Ontario Jobs", "Quebec Jobs", "Atlantic Jobs")
for (i in 1:6){
plot(jobs$Date, log(jobs[,i], 2), type = "l", ylim = range.log[,i],
    xlab = "Time", ylab = "Number of jobs", main = titles[i])
}
par(oldpar)
</code></pre>

<hr>
<h2 id='kiwishade'>Kiwi Shading Data</h2><span id='topic+kiwishade'></span>

<h3>Description</h3>

<p>The <code>kiwishade</code> data frame has 48 rows and 4 columns.
The data are from a designed experiment that
compared different kiwifruit shading treatments.
There are four vines in each plot, and four plots (one for each of four
treatments: none, Aug2Dec, Dec2Feb, and Feb2May) in each of three blocks
(locations: west, north, east). Each
plot has the same number of vines, each block has the same number of
plots, with each treatment occurring the same number of times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kiwishade</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>yield</dt><dd><p>Total yield (in kg)</p>
</dd>
<dt>plot</dt><dd><p>a factor with levels    <code>east.Aug2Dec</code>,
<code>east.Dec2Feb</code>,    <code>east.Feb2May</code>,
<code>east.none</code>,    <code>north.Aug2Dec</code>,
<code>north.Dec2Feb</code>,    <code>north.Feb2May</code>,
<code>north.none</code>,    <code>west.Aug2Dec</code>,
<code>west.Dec2Feb</code>,    <code>west.Feb2May</code>,
<code>west.none</code> </p>
</dd>
<dt>block</dt><dd><p>a factor indicating the location of the plot with levels
<code>east</code>, <code>north</code>, <code>west</code> </p>
</dd>
<dt>shade</dt><dd><p>a factor representing the period for which
the experimenter placed shading over the vines; with levels:
<code>none</code> no shading,    <code>Aug2Dec</code> August - December,
<code>Dec2Feb</code> December - February,    <code>Feb2May</code> February - May</p>
</dd>
</dl>



<h3>Details</h3>

<p>The northernmost plots were grouped together because they
were similarly affected by shading from the sun in the north.
For the remaining two blocks shelter effects, whether from the
west or from the east, were thought more important.
</p>


<h3>Source</h3>

<p>Snelgar, W.P., Manson. P.J., Martin, P.J. 1992. Influence of
time of shading on flowering and yield of kiwifruit vines. Journal of
Horticultural Science 67: 481-487.
</p>


<h3>References</h3>

<p>Maindonald J H 1992. Statistical design, analysis and presentation
issues. New Zealand Journal of Agricultural Research 35: 121-141.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Data Summary - Example 2.2.1")
attach(kiwishade)
kiwimeans &lt;- aggregate(yield, by=list(block, shade), mean)
names(kiwimeans) &lt;- c("block","shade","meanyield")

kiwimeans[1:4,]
pause()

print("Multilevel Design - Example 9.3")
kiwishade.aov &lt;- aov(yield ~ shade+Error(block/shade),data=kiwishade)
summary(kiwishade.aov)
pause()


sapply(split(yield, shade), mean)

pause()

kiwi.table &lt;- t(sapply(split(yield, plot), as.vector))
kiwi.means &lt;- sapply(split(yield, plot), mean)
kiwi.means.table &lt;- matrix(rep(kiwi.means,4), nrow=12, ncol=4)
kiwi.summary &lt;- data.frame(kiwi.means, kiwi.table-kiwi.means.table)
names(kiwi.summary)&lt;- c("Mean", "Vine 1", "Vine 2", "Vine 3", "Vine 4")
kiwi.summary
mean(kiwi.means) # the grand mean (only for balanced design)

if(require(lme4, quietly=TRUE)) {
kiwishade.lmer &lt;- lmer(yield ~ shade + (1|block) + (1|block:plot),
                       data=kiwishade)
## block:shade is an alternative to block:plot

kiwishade.lmer


##                  Residuals and estimated effects
library(lattice)
xyplot(residuals(kiwishade.lmer) ~ fitted(kiwishade.lmer)|block,
                data=kiwishade, groups=shade,
                layout=c(3,1), par.strip.text=list(cex=1.0),
                xlab="Fitted values (Treatment + block + plot effects)",
                ylab="Residuals", pch=1:4, grid=TRUE,
                scales=list(x=list(alternating=FALSE), tck=0.5),
                key=list(space="top", points=list(pch=1:4),
                         text=list(labels=levels(kiwishade$shade)),columns=4))
ploteff &lt;- ranef(kiwishade.lmer, drop=TRUE)[[1]]
qqmath(ploteff, xlab="Normal quantiles", ylab="Plot effect estimates",
       scales=list(tck=0.5))
}
</code></pre>

<hr>
<h2 id='leafshape'>Full Leaf Shape Data Set</h2><span id='topic+leafshape'></span><span id='topic+leafshape17'></span>

<h3>Description</h3>

<p>Leaf length, width and petiole measurements taken at various
sites worldwide.  The <code>leafshape17</code> data frame is the
subset that has data for North Queensland sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leafshape)
data(leafshape17)
</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>bladelen</dt><dd><p>leaf length (in mm)</p>
</dd>
<dt>petiole</dt><dd><p>a numeric vector</p>
</dd>
<dt>bladewid</dt><dd><p>leaf width (in mm)</p>
</dd>
<dt>latitude</dt><dd><p>latitude</p>
</dd>
<dt>logwid</dt><dd><p>natural logarithm of width</p>
</dd>
<dt>logpet</dt><dd><p>logarithm of petiole</p>
</dd>
<dt>loglen</dt><dd><p>logarithm of length</p>
</dd>
<dt>arch</dt><dd><p>leaf architecture (0 = plagiotropic, 1 = orthotropic</p>
</dd>
<dt>location</dt><dd><p>a factor with levels
<code>Sabah</code>, <code>Panama</code>, <code>Costa Rica</code>,
<code>N Queensland</code>, <code>S Queensland</code>, 
<code>Tasmania</code> </p>
</dd>
</dl>



<h3>Source</h3>

<p>King, D.A. and Maindonald, J.H. 1999. Tree architecture in relation to
leaf dimensions and tree stature in temperate and tropical rain
forests. Journal of Ecology 87: 1012-1024.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(MASS)
leaf17.lda &lt;- lda(arch ~ logwid+loglen, data=leafshape17)
leaf17.hat &lt;- predict(leaf17.lda)
leaf17.lda
 table(leafshape17$arch, leaf17.hat$class)
pause()

tab &lt;- table(leafshape17$arch, leaf17.hat$class)
 sum(tab[row(tab)==col(tab)])/sum(tab)
leaf17cv.lda &lt;- lda(arch ~ logwid+loglen, data=leafshape17, CV=TRUE)
tab &lt;- table(leafshape17$arch, leaf17cv.lda$class)
pause()

leaf17.glm &lt;- glm(arch ~ logwid + loglen, family=binomial, data=leafshape17)
 options(digits=3)
summary(leaf17.glm)$coef
pause()

leaf17.one &lt;- cv.binary(leaf17.glm)
table(leafshape17$arch, round(leaf17.one$internal))     # Resubstitution
pause()

table(leafshape17$arch, round(leaf17.one$cv))           # Cross-validation
</code></pre>

<hr>
<h2 id='leaftemp'>Leaf and Air Temperature Data</h2><span id='topic+leaftemp'></span>

<h3>Description</h3>

<p>Data are measurements of vapour pressure and of the 
difference between leaf and air temperature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leaftemp</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>CO2level</dt><dd><p>Carbon Dioxide level
<code>low</code>, <code>medium</code>, <code>high</code> </p>
</dd>
<dt>vapPress</dt><dd><p>Vapour pressure</p>
</dd>
<dt>tempDiff</dt><dd><p>Difference between leaf and air temperature</p>
</dd>
<dt>BtempDiff</dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Katharina Siebke and Susan von Cammerer, Australian National University.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Fitting Multiple Lines - Example 7.3")

leaf.lm1 &lt;- lm(tempDiff ~ 1 , data = leaftemp)
leaf.lm2 &lt;- lm(tempDiff ~ vapPress, data = leaftemp)
leaf.lm3 &lt;- lm(tempDiff ~ CO2level + vapPress, data = leaftemp)
leaf.lm4 &lt;- lm(tempDiff ~ CO2level + vapPress + vapPress:CO2level,
  data = leaftemp)

anova(leaf.lm1, leaf.lm2, leaf.lm3, leaf.lm4)

summary(leaf.lm2)
plot(leaf.lm2)

</code></pre>

<hr>
<h2 id='leaftemp.all'>Full Leaf and Air Temperature Data Set</h2><span id='topic+leaftemp.all'></span>

<h3>Description</h3>

<p>The <code>leaftemp.all</code> data frame has 62 rows and 9 columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leaftemp.all</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>glasshouse</dt><dd><p>a factor with levels
<code>A</code>,
<code>B</code>,
<code>C</code> </p>
</dd>
<dt>CO2level</dt><dd><p>a factor with Carbon Dioxide Levels:
<code>high</code>,
<code>low</code>,
<code>medium</code> </p>
</dd>
<dt>day</dt><dd><p>a factor</p>
</dd>
<dt>light</dt><dd><p>a numeric vector</p>
</dd>
<dt>CO2</dt><dd><p>a numeric vector</p>
</dd>
<dt>tempDiff</dt><dd><p>Difference between Leaf and Air Temperature</p>
</dd>
<dt>BtempDiff</dt><dd><p>a numeric vector</p>
</dd>
<dt>airTemp</dt><dd><p>Air Temperature</p>
</dd>
<dt>vapPress</dt><dd><p>Vapour Pressure</p>
</dd>
</dl>



<h3>Source</h3>

<p>J.H. Maindonald
</p>

<hr>
<h2 id='litters'>Mouse Litters</h2><span id='topic+litters'></span>

<h3>Description</h3>

<p>Data on the body and brain weights of 20 mice, together
with the size of the litter.  Two mice were taken from each
litter size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>litters</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>lsize</dt><dd><p>litter size</p>
</dd>
<dt>bodywt</dt><dd><p>body weight</p>
</dd>
<dt>brainwt</dt><dd><p>brain weight</p>
</dd>
</dl>



<h3>Source</h3>

<p>Wainright P, Pelkman C and Wahlsten D 1989. The quantitative
relationship between nutritional effects on preweaning growth      
and behavioral development in mice. Developmental Psychobiology   
22: 183-193.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Multiple Regression - Example 6.2")

pairs(litters, labels=c("lsize\n\n(litter size)", "bodywt\n\n(Body Weight)",
                        "brainwt\n\n(Brain Weight)"))
  # pairs(litters) gives a scatterplot matrix with less adequate labeling

mice1.lm &lt;- lm(brainwt ~ lsize, data = litters) # Regress on lsize
mice2.lm &lt;- lm(brainwt ~ bodywt, data = litters) #Regress on bodywt
mice12.lm &lt;- lm(brainwt ~ lsize + bodywt, data = litters) # Regress on lsize &amp; bodywt

summary(mice1.lm)$coef # Similarly for other coefficients.
# results are consistent with the biological concept of brain sparing

pause()

hat(model.matrix(mice12.lm))  # hat diagonal
pause()

plot(lm.influence(mice12.lm)$hat, residuals(mice12.lm))

print("Diagnostics - Example 6.3")

mice12.lm &lt;- lm(brainwt ~ bodywt+lsize, data=litters)
oldpar &lt;-par(mfrow = c(1,2))
bx &lt;- mice12.lm$coef[2]; bz &lt;- mice12.lm$coef[3]
res &lt;- residuals(mice12.lm)
plot(litters$bodywt, bx*litters$bodywt+res, xlab="Body weight",
  ylab="Component + Residual")
panel.smooth(litters$bodywt, bx*litters$bodywt+res) # Overlay
plot(litters$lsize, bz*litters$lsize+res, xlab="Litter size", 
  ylab="Component + Residual")
panel.smooth(litters$lsize, bz*litters$lsize+res)
par(oldpar)
</code></pre>

<hr>
<h2 id='lmdiags'>
Return data required for diagnostic plots
</h2><span id='topic+lmdiags'></span>

<h3>Description</h3>

<p>This extracts the code that provides the major part of the
statistical information used by plot.lm, leaving out the
code used to provide the graphs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmdiags(x, which = c(1L:3L, 5L), cook.levels = c(0.5, 1), hii=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lmdiags_+3A_x">x</code></td>
<td>

<p>This must be an  object of class <code>lm</code> object, or that inherits from
an object of class  <code>lm</code>.
</p>
</td></tr>
<tr><td><code id="lmdiags_+3A_which">which</code></td>
<td>
<p>a subset of the numbers '1:6', indicating the plots for
which statistical information is required
</p>
</td></tr>
<tr><td><code id="lmdiags_+3A_cook.levels">cook.levels</code></td>
<td>

<p>Levels for contours of <code>cook.levels</code>, by default <code>c(0.5,1)</code>
</p>
</td></tr>
<tr><td><code id="lmdiags_+3A_hii">hii</code></td>
<td>
<p>Diagonal elements for the hat matrix.  If not supplied
(<code>hii=NULL</code>), they will be calculated from the argument <code>x</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code><a href="stats.html#topic+plot.lm">plot.lm</a></code> for additional information.
</p>


<h3>Value</h3>

<table>
<tr><td><code>yh</code></td>
<td>
<p>fitted values</p>
</td></tr>
<tr><td><code>rs</code></td>
<td>
<p>standardized residuals (for <code>glm</code> models standardized
deviance residuals)</p>
</td></tr>
<tr><td><code>yhn0</code></td>
<td>
<p>As <code>yh</code>, but omitting observations with zero weight</p>
</td></tr>
<tr><td><code>cook</code></td>
<td>
<p>Cook's statistics</p>
</td></tr>
<tr><td><code>rsp</code></td>
<td>
<p>standardized residuals (for <code>glm</code> models standardized
Pearson residuals)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is designed, in the first place, for use in connection
with <code><a href="#topic+plotSimDiags">plotSimDiags</a></code>, used to give simulations of
diagnostic plots for <code>lm</code> objects.
</p>


<h3>Author(s)</h3>

<p>John Maindonald, using code that John Maindonald, Martin Maechler and
others had contributed to <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>
</p>


<h3>References</h3>

<p>See references for <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimDiags">plotSimDiags</a></code>, <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>women.lm &lt;- lm(weight ~ height, data=women)
veclist &lt;- lmdiags(x=women.lm)
## Returns the statistics that are required for graphs 1, 2, 3, and 5
</code></pre>

<hr>
<h2 id='logisticsim'>Simple Logistic Regression Data Simulator</h2><span id='topic+logisticsim'></span>

<h3>Description</h3>

<p>This function simulates simple regression data from a 
logistic model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logisticsim(x = seq(0, 1, length=101), a = 2, b = -4, seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logisticsim_+3A_x">x</code></td>
<td>
<p>a numeric vector representing the explanatory variable</p>
</td></tr>
<tr><td><code id="logisticsim_+3A_a">a</code></td>
<td>
<p>the regression function intercept</p>
</td></tr>
<tr><td><code id="logisticsim_+3A_b">b</code></td>
<td>
<p>the regression function slope</p>
</td></tr>
<tr><td><code id="logisticsim_+3A_seed">seed</code></td>
<td>
<p>numeric constant</p>
</td></tr>
</table>


<h3>Value</h3>

<p> a list consisting of
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>the explanatory variable vector</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>the Poisson response vector</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>logisticsim()
</code></pre>

<hr>
<h2 id='Lottario'>Ontario Lottery Data</h2><span id='topic+Lottario'></span>

<h3>Description</h3>

<p>The data frame <code>Lottario</code>
is a summary of 122 weekly draws of an Ontario lottery, beginning in
November, 1978.  Each draw consists of 7 numbered balls, drawn without
replacement from an urn consisting of balls numbered from 1 through 39.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Lottario</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Number</dt><dd><p>the integers from 1 to 39, representing the
numbered balls</p>
</dd>
<dt>Frequency</dt><dd><p>the number of occurrences of each numbered ball</p>
</dd>
</dl>



<h3>Source</h3>

 
<p>The Ontario Lottery Corporation
</p>


<h3>References</h3>

<p>Bellhouse, D.R. (1982). Fair is fair:  new rules for Canadian lotteries.
Canadian Public Policy - Analyse de Politiques 8: 311-320.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
order(Lottario$Frequency)[33:39]  # the 7 most frequently chosen numbers
</code></pre>

<hr>
<h2 id='lung'>Cape Fur Seal Lung Measurements</h2><span id='topic+lung'></span>

<h3>Description</h3>

<p>The <code>lung</code> vector consists  
of weight measurements of lungs taken from 30 Cape Fur
Seals that died as an unintended consequence of commercial fishing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lung</code></pre>

<hr>
<h2 id='Manitoba.lakes'>The Nine Largest Lakes in Manitoba</h2><span id='topic+Manitoba.lakes'></span>

<h3>Description</h3>

<p>The <code>Manitoba.lakes</code> data frame has 9 rows and 2 columns.
The areas and elevations of the nine largest lakes in
Manitoba, Canada.  The geography of Manitoba (a relatively
flat province) can be divided crudely into three main
areas: a very flat prairie in the south which is at a 
relatively high elevation, a middle region consisting
of mainly of forest and Precambrian rock, and a northern
region which drains more rapidly into Hudson
Bay.  All water in Manitoba, which does not evaporate, eventually drains 
into Hudson Bay.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Manitoba.lakes</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>elevation</dt><dd><p>a numeric vector consisting of the elevations
of the lakes (in meters)</p>
</dd>
<dt>area</dt><dd><p>a numeric vector consisting of the areas of
the lakes (in square kilometers)</p>
</dd>
</dl>



<h3>Source</h3>

<p>The CANSIM data base at Statistics Canada.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(Manitoba.lakes)
plot(Manitoba.lakes[-1,])
</code></pre>

<hr>
<h2 id='mdbAVtJtoD'>
Murray-Darling basin monthly temperatures
</h2><span id='topic+mdbAVtJtoD'></span>

<h3>Description</h3>

<p>Australian Murray-Darling basin monthly temperatures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mdbAVtJtoD")</code></pre>


<h3>Format</h3>

<p>The format is:
Time-Series [1:867] from 1950 to 2022: 27.44 26.84 24.4 22.27 8.41 ...
</p>


<h3>Source</h3>

<p>Australian Bureau of Meteorology web pages:
</p>
<p><a href="http://www.bom.gov.au/climate/change/index.shtml">http://www.bom.gov.au/climate/change/index.shtml</a>
</p>
<p>Go to website, choose timeseries to display, then click &quot;Download data&quot;
</p>
<p>The website gives anomalies from 1961-1990 averages. The monthly
means have been added, in order to obtain a series.  The monthly
means are shown along with plots for the individual months.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mdbAVtJtoD)
plot(window(mdbAVtJtoD, start=c(2000,1)), ylab="Mean monthly data")
</code></pre>

<hr>
<h2 id='measles'>Deaths in London from measles</h2><span id='topic+measles'></span>

<h3>Description</h3>

<p>Deaths in London from measles: 1629 &ndash; 1939, with gaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(measles)</code></pre>


<h3>Format</h3>

<p>The format is:
Time-Series [1:311] from 1629 to 1939: 42 2 3 80 21 33 27 12 NA NA ...
</p>


<h3>Source</h3>

<p>Guy, W. A. 1882. Two hundred and fifty years of small pox in London.
Journal of the Royal Statistical Society 399-443.
</p>
<p>Stocks, P. 1942. Measles and whooping cough during the dispersal of 
1939-1940. Journal of the Royal Statistical Society 105:259-291.
</p>


<h3>References</h3>

<p>Lancaster, H. O. 1990. Expectations of Life. Springer.
</p>

<hr>
<h2 id='medExpenses'>Family Medical Expenses</h2><span id='topic+medExpenses'></span>

<h3>Description</h3>

<p>The <code>medExpenses</code> data frame  
contains average weekly medical expenses including drugs for 33 families
randomly sampled from a community of 600 families which contained
2700 individuals.  These data were collected in the 1970's at an 
unknown location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>medExpenses</code></pre>


<h3>Format</h3>


<dl>
<dt>familysize</dt><dd><p>number of individuals in a family</p>
</dd>
<dt>expenses</dt><dd><p>average weekly cost for medical expenses per
family member</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>with(medExpenses, weighted.mean(expenses, familysize))
</code></pre>

<hr>
<h2 id='mignonette'>Darwin's Wild Mignonette Data</h2><span id='topic+mignonette'></span>

<h3>Description</h3>

<p>Data compare the heights of crossed plants with
self-fertilized plants of the wild mignonette
<em>reseda lutea</em>.  Plants were paired within the
pots in which they were grown, with one on one side and
one on the other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mignonette</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>cross</dt><dd><p>heights of 
the crossed plants</p>
</dd>
<dt>self</dt><dd><p>heights of
the self-fertilized plants</p>
</dd>
</dl>



<h3>Source</h3>

<p>Darwin, Charles. 1877.  The Effects of Cross and Self
Fertilisation in the Vegetable Kingdom.  Appleton and Company, New
York, page 118.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Is Pairing Helpful? - Example 4.3.1")

attach(mignonette)
plot(cross ~ self, pch=rep(c(4,1), c(3,12))); abline(0,1) 
abline(mean(cross-self), 1, lty=2)
detach(mignonette)
</code></pre>

<hr>
<h2 id='milk'>Milk Sweetness Study</h2><span id='topic+milk'></span>

<h3>Description</h3>

<p>The <code>milk</code> data frame has 17 rows and 2 columns.
Each of 17 panelists compared two milk samples
for sweetness. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>milk</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>four</dt><dd><p>a numeric vector consisting of the assessments for four units
of additive</p>
</dd>
<dt>one</dt><dd><p>a numeric vector  while the is the assessment for one unit
of additive</p>
</dd>
</dl>



<h3>Source</h3>

<p>J.H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Rug Plot - Example 1.8.1")
xyrange &lt;- range(milk)
plot(four ~ one, data = milk, xlim = xyrange, ylim = xyrange, pch = 16)
rug(milk$one)
rug(milk$four, side = 2)
abline(0, 1)
</code></pre>

<hr>
<h2 id='modelcars'>Model Car Data</h2><span id='topic+modelcars'></span>

<h3>Description</h3>

<p>The <code>modelcars</code> data frame has 12 rows and 2 columns.
The data are for an experiment in which a model car was released
three times at each of four different distances up a 20 degree
ramp.  The experimenter recorded distances traveled from the 
bottom of the ramp across a concrete floor.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modelcars</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>distance.traveled</dt><dd><p>a numeric vector consisting
of the lengths traveled (in cm)</p>
</dd>
<dt>starting.point</dt><dd><p>a numeric vector consisting
of the distance of the starting point from the top of
the ramp (in cm)</p>
</dd>
</dl>



<h3>Source</h3>

<p>W.J. Braun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(modelcars)
modelcars.lm &lt;- lm(distance.traveled ~ starting.point, data=modelcars)
aov(modelcars.lm)
pause()

print("Response Curves - Example 4.6")
attach(modelcars)
stripchart(distance.traveled ~ starting.point, vertical=TRUE, pch=15,
           xlab = "Distance up ramp", ylab="Distance traveled")
detach(modelcars)

</code></pre>

<hr>
<h2 id='monica'>WHO Monica Data</h2><span id='topic+monica'></span><span id='topic+mifem'></span>

<h3>Description</h3>

<p>The <code>monica</code> data frame has 6357 rows and 12 columns. The
dataset <code>mifem</code> (1295 rows) is the subset that has data
for females.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(monica)
  data(mifem)
  </code></pre>


<h3>Format</h3>

<p>Columns are:
</p>

<dl>
<dt>outcome</dt><dd><p>mortality outcome, a factor with levels <code>live</code>, 
<code>dead</code></p>
</dd> 
<dt>age</dt><dd><p>age at onset</p>
</dd>
<dt>sex</dt><dd><p>m = male, f = female</p>
</dd>
<dt>hosp</dt><dd><p>y = hospitalized, n = not hospitalized</p>
</dd>
<dt>yronset</dt><dd><p>year of onset</p>
</dd>
<dt>premi</dt><dd><p>previous myocardial infarction event, a 
factor with levels <code>y</code>, <code>n</code>, <code>nk</code> not known </p>
</dd>
<dt>smstat</dt><dd><p>smoking status, a factor with levels <code>c</code> 
current, <code>x</code> ex-smoker, <code>n</code> non-smoker, <code>nk</code> not known</p>
</dd>
<dt>diabetes</dt><dd><p>a factor with levels <code>y</code>, <code>n</code>, <code>nk</code> not 
known</p>
</dd>
<dt>highbp</dt><dd><p>high blood pressure, a factor with levels
<code>y</code>, <code>n</code>, <code>nk</code> not known </p>
</dd>
<dt>hichol</dt><dd><p>high cholesterol, a factor with levels
<code>y</code>, <code>n</code> <code>nk</code> not known</p>
</dd>
<dt>angina</dt><dd><p>a factor with levels <code>y</code>,
<code>n</code>, <code>nk</code> not known</p>
</dd>
<dt>stroke</dt><dd><p>a factor with levels
<code>y</code>, <code>n</code>, <code>nk</code> not known</p>
</dd>
</dl>



<h3>Source</h3>

<p>Newcastle (Australia) centre of the Monica
project; see the web site http://www.ktl.fi/monica
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("CART - Example 10.7")
summary(monica)
pause()

library(rpart)
monica.rpart &lt;- rpart(outcome ~ ., data = monica, cp = 0.0025)
plotcp(monica.rpart)
printcp(monica.rpart)
pause()
monicab.rpart &lt;- prune(monica.rpart, cp=0.006)
print(monicab.rpart)
summary(mifem)
pause()
mifem.rpart &lt;- rpart(outcome ~ ., data = mifem, cp = 0.0025)
plotcp(mifem.rpart)
printcp(mifem.rpart)
pause()

mifemb.rpart &lt;- prune(mifem.rpart, cp=0.006)
print(mifemb.rpart)
</code></pre>

<hr>
<h2 id='moths'>Moths Data</h2><span id='topic+moths'></span>

<h3>Description</h3>

<p>The <code>moths</code> data frame has 41 rows and 4 columns.
These data are from a study of the effect of habitat on the
densities of two species of moth (A and P).  Transects were
set across the search area.  Within transects, sections
were identified according to habitat type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moths</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>meters</dt><dd><p>length of transect</p>
</dd>
<dt>A</dt><dd><p>number of type A moths found</p>
</dd>
<dt>P</dt><dd><p>number of type P moths found</p>
</dd>
<dt>habitat</dt><dd><p>a factor with levels
<code>Bank</code>,
<code>Disturbed</code>,
<code>Lowerside</code>,
<code>NEsoak</code>,
<code>NWsoak</code>,
<code>SEsoak</code>,
<code>SWsoak</code>,
<code>Upperside</code> </p>
</dd>
</dl>



<h3>Source</h3>

<p>Sharyn Wragg, formerly of Australian National University
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Quasi Poisson Regression - Example 8.3")
rbind(table(moths[,4]), sapply(split(moths[,-4], moths$habitat), apply,2,
sum))
A.glm &lt;- glm(formula = A ~ log(meters) + factor(habitat), family =
quasipoisson, data = moths)
summary(A.glm)
  # Note the huge standard errors
moths$habitat &lt;- relevel(moths$habitat, ref="Lowerside")
A.glm &lt;- glm(A ~ habitat + log(meters), family=quasipoisson, data=moths)
summary(A.glm)$coef
## Consider as another possibility
A2.glm &lt;- glm(formula = A ~ sqrt(meters) + factor(habitat), family =
                  quasipoisson(link=sqrt), data = moths)
summary(A2.glm)
</code></pre>

<hr>
<h2 id='multilap'>Data Filtering Function</h2><span id='topic+multilap'></span>

<h3>Description</h3>

<p>A subset of data is selected for which the treatment to control
ratio of non-binary covariates is never outside a specified range.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multilap(df = DAAG::nsw74psid1, maxf = 20, colnames = c("educ",
                 "age", "re74", "re75", "re78"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multilap_+3A_df">df</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="multilap_+3A_maxf">maxf</code></td>
<td>
<p>filtering parameter</p>
</td></tr>
<tr><td><code id="multilap_+3A_colnames">colnames</code></td>
<td>
<p>columns to be compared for filtering</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>

<hr>
<h2 id='nassCDS'>Airbag and other influences on accident fatalities</h2><span id='topic+nassCDS'></span>

<h3>Description</h3>

<p>US data, for 1997-2002, from police-reported car crashes in
which there is a harmful event (people or property), and from which at
least one vehicle was towed. Data are restricted to front-seat
occupants, include only a subset of the variables recorded, and
are restricted in other ways also.</p>


<h3>Usage</h3>

<pre><code class='language-R'>nassCDS</code></pre>


<h3>Format</h3>

<p>A data frame with 26217 observations on the following 15 variables.
</p>

<dl>
<dt>dvcat</dt><dd><p>ordered factor with levels (estimated impact
speeds) <code>1-9km/h</code>, <code>10-24</code>, <code>25-39</code>, <code>40-54</code>,
<code>55+</code>
</p>
</dd>
<dt>weight</dt><dd><p>Observation weights, albeit of uncertain
accuracy, designed to account for varying sampling probabilities.</p>
</dd>
<dt>dead</dt><dd><p>factor with levels <code>alive</code> <code>dead</code></p>
</dd>
<dt>airbag</dt><dd><p>a factor with levels <code>none</code> <code>airbag</code></p>
</dd>
<dt>seatbelt</dt><dd><p>a factor with levels <code>none</code> <code>belted</code></p>
</dd>
<dt>frontal</dt><dd><p>a numeric vector; 0 = non-frontal, 1=frontal impact</p>
</dd>
<dt>sex</dt><dd><p>a factor with levels <code>f</code> <code>m</code></p>
</dd>
<dt>ageOFocc</dt><dd><p>age of occupant in years</p>
</dd>
<dt>yearacc</dt><dd><p>year of accident</p>
</dd>
<dt>yearVeh</dt><dd><p>Year of model of vehicle; a numeric vector</p>
</dd>
<dt>abcat</dt><dd><p>Did one or more (driver or passenger) airbag(s)
deploy? This factor has levels <code>deploy</code> <code>nodeploy</code>
<code>unavail</code></p>
</dd>
<dt>occRole</dt><dd><p>a factor with levels <code>driver</code> <code>pass</code></p>
</dd>
<dt>deploy</dt><dd><p>a numeric vector: 0 if an airbag was
unavailable or did not deploy; 1 if one or more bags deployed.</p>
</dd>
<dt>injSeverity</dt><dd><p>a numeric vector; 0:none, 1:possible injury,
2:no incapacity, 3:incapacity, 4:killed; 5:unknown, 6:prior death</p>
</dd>
<dt>caseid</dt><dd><p>character, created by pasting together the
populations sampling unit, the case number, and the vehicle
number.  Within each year, use this to uniquely identify the
vehicle.
</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data collection used a multi-stage probabilistic sampling scheme.
The observation weight, called national inflation factor
(<code>national</code>) in the data from NASS, is the inverse
of an estimate of the selection probability.  These data
include a subset of the variables from the NASS dataset.  Variables
that are coded here as factors are coded as numeric values in that
dataset.
</p>


<h3>Source</h3>

<p><a href="https://www.stat.colostate.edu/~meyer/airbags.htm">https://www.stat.colostate.edu/~meyer/airbags.htm</a>\
<a href="https://www.nhtsa.gov/file-downloads">https://www.nhtsa.gov/file-downloads</a>
</p>
<p>See also\
<a href="https://maths-people.anu.edu.au/~johnm/datasets/airbags/">https://maths-people.anu.edu.au/~johnm/datasets/airbags/</a>
</p>


<h3>References</h3>

<p>Meyer, M.C. and Finney, T. (2005): <em>Who wants airbags?</em>. Chance
18:3-16.
</p>
<p>Farmer, C.H. 2006. <em>Another look at Meyer and Finney's &lsquo;Who wants
airbags?&rsquo;</em>. Chance 19:15-22.
</p>
<p>Meyer, M.C. 2006.  <em>Commentary on &quot;Another look at Meyer and
Finney's &lsquo;Who wants airbags?&rsquo;</em>. Chance 19:23-24.
</p>
<p>For analyses based on the alternative FARS (Fatal Accident Recording
System) data, and associated commentary, see:
</p>
<p>Cummings, P; McKnight, B, 2010. <em>Accounting for vehicle, crash, and
occupant characteristics in traffic crash studies.</em> Injury Prevention 16:
363-366. [The relatively definitive analyses in this paper use a matched
cohort design,
</p>
<p>Olson, CM; Cummings, P, Rivara, FP, 2006. <em>Association of first- and
second-generation air bags with front occupant death in car crashes: a
matched cohort study.</em> Am J Epidemiol 164:161-169. [The relatively
definitive analyses in this paper use a matched cohort design, using
data taken from the FARS (Fatal Accident Recording System) database.]
</p>
<p>Braver, ER; Shardell, M; Teoh, ER, 2010. <em>How have changes in air
bag designs affected frontal crash mortality?</em> Ann Epidemiol 20:499-510.
</p>
<p>The web page <a href="http://www-fars.nhtsa.dot.gov/Main/index.aspx">http://www-fars.nhtsa.dot.gov/Main/index.aspx</a> has a
menu-based interface into the FARS (Fatality Analysis Recording
System) data. The FARS database aims to include every accident in
which there was at least one fatality.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nassCDS)
xtabs(weight ~ dead + airbag, data=nassCDS)
xtabs(weight ~ dead + airbag + seatbelt + dvcat, data=nassCDS)
tab &lt;- xtabs(weight ~ dead + abcat, data=nassCDS,
             subset=dvcat=="25-39"&amp;frontal==0)[, c(3,1,2)]
round(tab[2, ]/apply(tab,2,sum)*100,2)
</code></pre>

<hr>
<h2 id='nasshead'>Documentation of names of columns in nass9702cor</h2><span id='topic+nasshead'></span>

<h3>Description</h3>

<p><code>SASname</code> and <code>longname</code> are from the SAS XPT file
nass9702cor.XPT that is available from the website noted below.
The name <code>shortname</code> is the name used in the data frame
<code>nass9702cor</code>, not included in this package, but available
from my website that is noted below.  It is also used in
<code>nassCDS</code>, for columns that <code>nassCDS</code> includes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nasshead)</code></pre>


<h3>Format</h3>

<p>A data frame with 56 observations on the following 3 variables.
</p>

<dl>
<dt>shortname</dt><dd><p>a character vector</p>
</dd>
<dt>SASname</dt><dd><p>a character vector</p>
</dd>
<dt>longname</dt><dd><p>a character vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>For full details of the coding of values in columns of
<code>nass9702cor</code>, consult one of the SAS format files that
can be obtained by following the instructions on Dr Meyer's web
site that is noted below.
</p>


<h3>Source</h3>

<p><a href="https://www.stat.colostate.edu/~meyer/airbags.htm">https://www.stat.colostate.edu/~meyer/airbags.htm</a>\
<a href="https://www.nhtsa.gov/file-downloads">https://www.nhtsa.gov/file-downloads</a>\
</p>
<p>See also
<a href="https://maths-people.anu.edu.au/~johnm/datasets/airbags/">https://maths-people.anu.edu.au/~johnm/datasets/airbags/</a>
</p>


<h3>References</h3>

<p>Meyer, M.C. and Finney, T. (2005): <em>Who wants airbags?</em>. Chance
18:3-16.
</p>
<p>Farmer, C.H. 2006. <em>Another look at Meyer and Finney's 'Who wants
airbags?'</em>. Chance 19:15-22.
</p>
<p>Meyer, M.C. 2006.  <em>Commentary on &quot;Another look at Meyer and
Finney's &lsquo;Who wants airbags?&rsquo;&quot;</em>. Chance 19:23-24.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nasshead)
</code></pre>

<hr>
<h2 id='nihills'>Record times for Northern Ireland mountain running events</h2><span id='topic+nihills'></span><span id='topic+lognihills'></span>

<h3>Description</h3>

<p>Data were from the 2007 calendar for the Northern Ireland Mountain
Running Association.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nihills)
data(lognihills)
</code></pre>


<h3>Format</h3>

<p>A data frame with 23 observations on the following 4 variables.
</p>

<dl>
<dt><code>dist</code></dt><dd><p>distances in miles</p>
</dd>
<dt><code>climb</code></dt><dd><p>amount of climb in feet</p>
</dd>
<dt><code>time</code></dt><dd><p>record time in hours for males</p>
</dd>
<dt><code>timef</code></dt><dd><p>record time in hours for females</p>
</dd>
<dt><code>logdist</code></dt><dd><p>distances, log(miles)</p>
</dd>
<dt><code>logclimb</code></dt><dd><p>climb, log(feet)</p>
</dd>
<dt><code>logtime</code></dt><dd><p>record time for males, log(hours)</p>
</dd>
<dt><code>logtimef</code></dt><dd><p>record time for females, log(hours)</p>
</dd>
</dl>



<h3>Details</h3>

<p>These data make an interesting comparison with the dataset
<code>hills2000</code> in the DAAG package.</p>


<h3>Source</h3>

<p>For more recent information, see
<a href="https://www.nimra.org.uk/index.php/fixtures/">https://www.nimra.org.uk/index.php/fixtures/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nihills)
lm(formula = log(time) ~ log(dist) + log(climb), data = nihills)
lm(formula = log(time) ~ log(dist) + log(climb/dist), data = nihills)
lm(formula = logtime ~ logdist + I(logclimb-logdist), data = lognihills)
</code></pre>

<hr>
<h2 id='nsw74demo'>Labour Training Evaluation Data</h2><span id='topic+nsw74demo'></span><span id='topic+nsw74psid1'></span><span id='topic+nsw74psid3'></span><span id='topic+nsw74psidA'></span>

<h3>Description</h3>

<p>This <code>nsw74demo</code> data frame, with 445 rows and 10 columns,
is the subset of the <code><a href="#topic+nswdemo">nswdemo</a></code> dataset for which 1974
earnings are available. 
Data are for the male experimental control and treatment
groups, in an investigation of the effect of training
on changes, between 1974-1975 and 1978, in the earnings
of individuals who had experienced employment difficulties.
</p>
<p>Likewise, <code>nsw74psid1</code> (2675 rows) is the subset of the
<code>nswpsid1</code> data, and <code>nsw74psid3</code> (313 rows) is the subset of
the  <code>nswpsid3</code> data, for which 1974 income is available.
NB, also, the <code>nsw74psidA</code> data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(nsw74demo)
  data(nsw74psid1)
  data(nsw74psid3)
  data(nsw74psidA)
</code></pre>


<h3>Format</h3>

<p>Columns are:
</p>

<dl>
<dt>trt</dt><dd><p>a numeric vector 
identifying the study in which the subjects were enrolled
(0 = PSID, 1 = NSW).</p>
</dd>
<dt>age</dt><dd><p> age (in years).</p>
</dd>
<dt>educ</dt><dd><p> years of education.</p>
</dd>
<dt>black</dt><dd><p> (0 = not black, 1 = black).</p>
</dd>
<dt>hisp</dt><dd><p> (0 = not hispanic, 1 = hispanic).</p>
</dd>
<dt>marr</dt><dd><p> (0 = not married, 1 = married).</p>
</dd>
<dt>nodeg</dt><dd><p> (0 = completed high school, 1 = dropout).</p>
</dd>
<dt>re74</dt><dd><p> real earnings in 1974.</p>
</dd>
<dt>re75</dt><dd><p> real earnings in 1975.</p>
</dd>
<dt>re78</dt><dd><p> real earnings in 1978.</p>
</dd> 
</dl>



<h3>Details</h3>

<p>The <code>nsw74psidA</code> data set (252 rows) was obtained from
<code>nsw74psid1</code> using:
</p>
<p><code>here &lt;- age &lt;= 40 &amp; re74&lt;=5000 &amp; re75 &lt;= 5000 &amp; re78 &lt; 30000</code>
</p>
<p><code>nsw74psidA &lt;- nsw74psid1[here, ]</code>
</p>


<h3>Source</h3>

<p>http://www.columbia.edu/~rd247/nswdata.html
</p>


<h3>References</h3>

<p>Dehejia, R.H. and Wahba, S. 1999. Causal effects in
non-experimental studies: re-evaluating the evaluation of training
programs. Journal of the American Statistical Association 94:
1053-1062.
</p>
<p>Lalonde, R. 1986. Evaluating the economic evaluations of
training programs. American Economic Review 76: 604-620.
</p>

<hr>
<h2 id='nswdemo'>Labour Training Evaluation Data</h2><span id='topic+nswdemo'></span>

<h3>Description</h3>

<p>The <code>nswdemo</code> data frame contains 722 rows and 10 columns.
These data are pertinent to an investigation of the way that   
earnings changed, between 1974-1975 and 1978, for an experimental
treatment who were given job training as compared with a control
group who did not receive such training.
</p>
<p>The <code>psid1</code> data set is an alternative non-experimental &quot;control&quot;
group. <code>psid2</code> and <code>psid3</code> are subsets of <code>psid1</code>,
designed to be better matched to the experimental data than
<code>psid1</code>. Note also the <code>cps1</code>, <code>cps2</code> and <code>cps3</code>
datasets (<span class="pkg">DAAGxtras</span>) that have been proposed as
non-experimental controls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nswdemo)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>trt</dt><dd><p>a numeric vector 
identifying the study in which the subjects were enrolled
(0 = Control, 1 = treated).</p>
</dd>
<dt>age</dt><dd><p> age (in years).</p>
</dd>
<dt>educ</dt><dd><p> years of education.</p>
</dd>
<dt>black</dt><dd><p> (0 = not black, 1 = black).</p>
</dd>
<dt>hisp</dt><dd><p> (0 = not hispanic, 1 = hispanic).</p>
</dd>
<dt>marr</dt><dd><p> (0 = not married, 1 = married).</p>
</dd>
<dt>nodeg</dt><dd><p> (0 = completed high school, 1 = dropout).</p>
</dd>
<dt>re74</dt><dd><p> real earnings in 1974.</p>
</dd>
<dt>re75</dt><dd><p> real earnings in 1975.</p>
</dd>
<dt>re78</dt><dd><p> real earnings in 1978.</p>
</dd> 
</dl>



<h3>Source</h3>

<p><a href="https://users.nber.org/~rdehejia/nswdata.html">https://users.nber.org/~rdehejia/nswdata.html</a>
</p>


<h3>References</h3>

<p>Dehejia, R.H. and Wahba, S. 1999. Causal effects in
non-experimental studies: re-evaluating the evaluation of training
programs. Journal of the American Statistical Association 94:
1053-1062.
</p>
<p>Lalonde, R. 1986. Evaluating the economic evaluations of
training programs. American Economic Review 76: 604-620.
</p>
<p>Smith, J. A. and Todd, P.E. 2005,&quot;Does Matching overcome. LaLonde?s
critique of nonexperimental estimators&quot;, <em>Journal of
Econometrics</em> 125: 305-353.
</p>
<p>Dehejia, R.H. 2005. Practical propensity score matching: a reply to
Smith and Todd. <em>Journal of Econometrics</em> 125: 355-364.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+psid1">psid1</a></code>, <code><a href="#topic+psid2">psid2</a></code>, <code><a href="#topic+psid3">psid3</a></code>
</p>

<hr>
<h2 id='nswpsid1'>Labour Training Evaluation Data</h2><span id='topic+cps1'></span><span id='topic+cps2'></span><span id='topic+cps3'></span><span id='topic+psid1'></span><span id='topic+psid2'></span><span id='topic+psid3'></span><span id='topic+nswpsid1'></span>

<h3>Description</h3>

<p>The <code>cps1</code>  (15992 rows) and <code>psid1</code> (2490 rows)
datasets are from
non-experimental &quot;control&quot; groups, used in various studies of
the effect of a labor training program, alternative to the
experimental control group in <code><a href="#topic+nswdemo">nswdemo</a></code>.
The <code>cps2</code> (2369 rows) and <code>cps3</code> (429 rows) subsets
of <code>cps1</code> are designed to 
be better matched to the experimental data than <code>cps1</code>.
Likewise, <code>psid2</code> (253 rows) and <code>psid3</code> (128 rows)
are subsets of <code>psid1</code> that are
designed to be better matched to the experimental data than
<code>psid1</code>.
The <code>nswpsid1</code> dataset (2675 rows) combines the experimental
treatment group in <code>nswdemo</code> with the <code>psid1</code> 
control data from the Panel Study of Income Dynamics
(PSID) study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(psid1)
  data(nswpsid1)
  </code></pre>


<h3>Format</h3>

<p>Columns are:
</p>

<dl>
<dt>trt</dt><dd><p>a numeric vector 
identifying the study in which the subjects were enrolled
(0 = Control, 1 = treated).</p>
</dd>
<dt>age</dt><dd><p> age (in years).</p>
</dd>
<dt>educ</dt><dd><p> years of education.</p>
</dd>
<dt>black</dt><dd><p> (0 = not black, 1 = black).</p>
</dd>
<dt>hisp</dt><dd><p> (0 = not hispanic, 1 = hispanic).</p>
</dd>
<dt>marr</dt><dd><p> (0 = not married, 1 = married).</p>
</dd>
<dt>nodeg</dt><dd><p> (0 = completed high school, 1 = dropout).</p>
</dd>
<dt>re74</dt><dd><p> real earnings in 1974.</p>
</dd>
<dt>re75</dt><dd><p> real earnings in 1975.</p>
</dd>
<dt>re78</dt><dd><p> real earnings in 1978.</p>
</dd> 
</dl>



<h3>Details</h3>

 
<p>The <code>cps1</code> and <code>psid1</code> data sets are two non-experimental
&quot;control&quot; groups, alternative to that in <code>nswdemo</code>, used in
investigating whether use of such a non-experimental control group can
be satisfactory. <code>cps2</code> and <code>cps3</code> are subsets of <code>cps1</code>,
designed to be better matched to the experimental data than <code>cps1</code>.
Similary <code>psid2</code> and <code>psid3</code> are subsets of <code>psid1</code>,
designed to be better matched to the experimental data than
<code>psid1</code>. <code>nswpsid1</code> combines data for the experimental
treatment group in <code>nswdemo</code> with the <code>psid1</code> control data
from the Panel Study of Income Dynamics (PSID) study.
</p>


<h3>Source</h3>

<p><a href="https://users.nber.org/~rdehejia/nswdata.html">https://users.nber.org/~rdehejia/nswdata.html</a>
</p>


<h3>References</h3>

<p>Dehejia, R.H. and Wahba, S. 1999. Causal effects in
non-experimental studies: re-evaluating the evaluation of training
programs. <em>Journal of the American Statistical Association</em> 94:
1053-1062.
</p>
<p>Lalonde, R. 1986. Evaluating the economic evaluations of
training programs. <em>American Economic Review</em> 76: 604-620.
</p>
<p>Smith, J. A. and Todd, P.E. &quot;Does Matching overcome. LaLonde?s
critique of nonexperimental estimators&quot;, <em>Journal of
Econometrics</em> 125: 305-353.
</p>
<p>Dehejia, R.H. 2005. Practical propensity score matching: a reply to
Smith and Todd. <em>Journal of Econometrics</em> 125: 355-364.
</p>

<hr>
<h2 id='obounce'>Bounce - obsolete</h2><span id='topic+obounce'></span>

<h3>Description</h3>

<p>A utility function for oneway.plot
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>

<hr>
<h2 id='oddbooks'>Measurements on 12 books</h2><span id='topic+oddbooks'></span>

<h3>Description</h3>

<p>Data giving thickness (mm), height (cm), width (cm)
and weight (g), of 12 books.  Books were selected so that thickness
decreased as page area increased
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(oddbooks)</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 4 variables.
</p>

<dl>
<dt>thick</dt><dd><p>a numeric vector</p>
</dd>
<dt>height</dt><dd><p>a numeric vector</p>
</dd>
<dt>breadth</dt><dd><p>a numeric vector</p>
</dd>
<dt>weight</dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>JM took books from his library.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(oddbooks)
str(oddbooks)
plot(oddbooks) 
</code></pre>

<hr>
<h2 id='onesamp'> Paired Sample t-test </h2><span id='topic+onesamp'></span>

<h3>Description</h3>

<p>This function performs a t-test for the mean difference for paired data,
and produces a scatterplot of one column against the other column, showing 
whether there was any benefit to using the paired design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onesamp(dset, x="unsprayed", y="sprayed", xlab=NULL, ylab=NULL,
        dubious=NULL, conv=NULL, dig=2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onesamp_+3A_dset">dset</code></td>
<td>
<p>a matrix or dataframe having two columns</p>
</td></tr>
<tr><td><code id="onesamp_+3A_x">x</code></td>
<td>
<p>name of column to play the role of the &lsquo;predictor&rsquo; </p>
</td></tr>
<tr><td><code id="onesamp_+3A_y">y</code></td>
<td>
<p>name of column to play the role of the &lsquo;response&rsquo; </p>
</td></tr>
<tr><td><code id="onesamp_+3A_xlab">xlab</code></td>
<td>
<p> horizontal axis label </p>
</td></tr>
<tr><td><code id="onesamp_+3A_ylab">ylab</code></td>
<td>
<p>vertical axis label </p>
</td></tr>
<tr><td><code id="onesamp_+3A_dubious">dubious</code></td>
<td>
<p>vector of logical (<code>FALSE</code>/<code>TRUE</code>) values,
specifying points that are to be omitted</p>
</td></tr>
<tr><td><code id="onesamp_+3A_conv">conv</code></td>
<td>
<p>scaling factor that should be applied to data</p>
</td></tr>
<tr><td><code id="onesamp_+3A_dig">dig</code></td>
<td>
<p>round SE to this number of digits for dispplay on graph </p>
</td></tr>
<tr><td><code id="onesamp_+3A_...">...</code></td>
<td>
<p>Further arguments, to be passed to <code>plot()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatterplot of <code>y</code> against <code>x</code> together with estimates
of standard errors and standard errors of the difference 
(<code>y</code>-<code>x</code>).
</p>
<p>Also produced is a confidence interval and p-value for the test.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>onesamp(dset = pair65, x = "ambient", y = "heated", xlab =
        "Amount of stretch (ambient)", ylab =
        "Amount of stretch (heated)") 
</code></pre>

<hr>
<h2 id='onet.permutation'>One Sample Permutation t-test</h2><span id='topic+onet.permutation'></span>

<h3>Description</h3>

<p>This function computes the p-value for the one sample
t-test using a permutation test.  The permutation
density can  also be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onet.permutation(x = DAAG::pair65$heated - DAAG::pair65$ambient, nsim = 2000, 
plotit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onet.permutation_+3A_x">x</code></td>
<td>
<p>a numeric vector containing the sample values (centered
at the null hypothesis value) </p>
</td></tr>
<tr><td><code id="onet.permutation_+3A_nsim">nsim</code></td>
<td>
<p>the number of permutations (randomly selected)</p>
</td></tr>
<tr><td><code id="onet.permutation_+3A_plotit">plotit</code></td>
<td>
<p>if TRUE, the permutation density is plotted </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The p-value for the test of the hypothesis that the mean of <code>x</code>
differs from 0
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>References</h3>

 
<p>Good, P. 2000. Permutation Tests. Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>onet.permutation()
</code></pre>

<hr>
<h2 id='onetPermutation'>One Sample Permutation t-test</h2><span id='topic+onetPermutation'></span>

<h3>Description</h3>

<p>This function computes the p-value for the one sample
t-test using a permutation test.  The permutation
density can  also be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onetPermutation(x = DAAG::pair65$heated - DAAG::pair65$ambient, nsim = 2000, 
plotit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onetPermutation_+3A_x">x</code></td>
<td>
<p>a numeric vector containing the sample values (centered
at the null hypothesis value) </p>
</td></tr>
<tr><td><code id="onetPermutation_+3A_nsim">nsim</code></td>
<td>
<p>the number of permutations (randomly selected)</p>
</td></tr>
<tr><td><code id="onetPermutation_+3A_plotit">plotit</code></td>
<td>
<p>if TRUE, the permutation density is plotted </p>
</td></tr>
</table>


<h3>Value</h3>

<p>The p-value for the test of the hypothesis that the mean of <code>x</code>
differs from 0
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>References</h3>

 
<p>Good, P. 2000. Permutation Tests. Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>onetPermutation()
</code></pre>

<hr>
<h2 id='onewayPlot'>Display of One Way Analysis Results</h2><span id='topic+onewayPlot'></span><span id='topic+oneway.plot'></span>

<h3>Description</h3>

<p>A line plot of estimates for unstructured comparison of factor levels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onewayPlot(obj, trtnam = "trt", axisht = 4.5, xlim = NULL,
    xlab = NULL, lsdht = 1.5, hsdht = 0.5, textht = axisht -
        2.5, oma = rep(1, 4), angle = 80, alpha = 0.05)
oneway.plot(obj, trtnam = "trt", axisht = 4.5, xlim = NULL,
    xlab = NULL, lsdht = 1.5, hsdht = 0.5, textht = axisht -
        2.5, oma = rep(1, 4), angle = 80, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onewayPlot_+3A_obj">obj</code></td>
<td>
<p>One way analysis of variance object (from aov)</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_trtnam">trtnam</code></td>
<td>
<p>name of factor for which line plot is required</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_axisht">axisht</code></td>
<td>
<p>Axis height</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_xlim">xlim</code></td>
<td>
<p>Range on horizontal axis</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_xlab">xlab</code></td>
<td>
<p>Horizontal axis label</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_lsdht">lsdht</code></td>
<td>
<p>Height adjustment parameter for display of LSD</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_hsdht">hsdht</code></td>
<td>
<p>Height adjustment parameter for display of Tukey's HSD</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_textht">textht</code></td>
<td>
<p>Height of text</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_oma">oma</code></td>
<td>
<p>Outer margin area</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_angle">angle</code></td>
<td>
<p>Text angle (in degrees)</p>
</td></tr>
<tr><td><code id="onewayPlot_+3A_alpha">alpha</code></td>
<td>
<p>Test size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimates, labeled with level names, are set out along a line</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>rice.aov &lt;- aov(ShootDryMass ~ trt, data=rice)
onewayPlot(obj=rice.aov)
</code></pre>

<hr>
<h2 id='orings'>Challenger O-rings Data</h2><span id='topic+orings'></span>

<h3>Description</h3>

<p>Record of the number and type of O-ring failures prior to the
tragic Challenger mission in January, 1986.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orings</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Temperature</dt><dd><p>O-ring temperature for each test firing or
actual launch of the shuttle rocket engine</p>
</dd>
<dt>Erosion</dt><dd><p>Number of erosion incidents </p>
</dd>
<dt>Blowby</dt><dd><p>Number of blowby incidents</p>
</dd>
<dt>Total</dt><dd><p>Total number of incidents</p>
</dd>
</dl>



<h3>Source</h3>

<p>Presidential Commission on the Space Shuttle Challenger Accident, 
Vol. 1, 1986: 129-131.  
</p>


<h3>References</h3>

<p>Tufte, E. R. 1997. Visual Explanations. Graphics Press, Cheshire,
Connecticut, U.S.A.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oldpar &lt;- par(mfrow=c(1,2))
plot(Total~Temperature, data = orings[c(1,2,4,11,13,18),]) # the 
               # observations included in the pre-launch charts
plot(Total~Temperature, data = orings)
par(oldpar)
</code></pre>

<hr>
<h2 id='overlapDensity'>Overlapping Density Plots</h2><span id='topic+overlapDensity'></span><span id='topic+overlap.density'></span>

<h3>Description</h3>

<p>Densities for two distinct samples are estimated and plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlapDensity(x0, x1, ratio = c(0.05, 20), ratio.number = FALSE,
          plotvalues = c("Density", "Numbers"), gpnames = c("Control", "Treatment"),
          cutoffs = c(lower = TRUE, upper = TRUE), bw = FALSE,
          xlab = "Score", ylab = NULL,
          col = 1:2, lty = 1:2, lwd = c(1, 1), ...)

overlap.density(x0, x1, ratio = c(0.05, 20), ratio.number = FALSE,
          plotvalues = c("Density", "Numbers"), gpnames = c("Control", "Treatment"),
          cutoffs = c(lower = TRUE, upper = TRUE), bw = FALSE,
          xlab = "Score", ylab = NULL,
          col = 1:2, lty = 1:2, lwd = c(1, 1), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlapDensity_+3A_x0">x0</code></td>
<td>
<p>control group measurements</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_x1">x1</code></td>
<td>
<p>treatment group measurements</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_ratio">ratio</code></td>
<td>
<p>if not <code>NULL</code>, the range within which the relative
number per unit interval (<code>ratio.number=TRUE</code>) or
relative probability density (<code>ratio.number=FALSE</code>)
of observations from the two groups are required to lie
will be used to determine lower and upper bounds on the
values of <code>x0</code> and <code>x1</code>.
[The relative numbers at any point are estimated from
(density1*n1)/(density0*x0)]</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_ratio.number">ratio.number</code></td>
<td>
<p>If TRUE (default), then <code>ratio</code>
is taken as the ratio of number of points per unit
interval</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_plotvalues">plotvalues</code></td>
<td>
<p>If set to <code>Number</code> then the y-axis
scale is chosen so that total area undere the curve is equal
to the sample size; otherwise (<code>plotvalues="Density"</code>)
total area under each cueve is 1. Any other setting does
not give a plot.</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_gpnames">gpnames</code></td>
<td>
<p>Names of the two samples</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_cutoffs">cutoffs</code></td>
<td>
<p>logical vector, indicating whether density estimates
should be truncated below (lower=TRUE) or above (upper=TRUE)</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_bw">bw</code></td>
<td>
<p>logical, indicates whether to overwrite with a gray
scale plot</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_xlab">xlab</code></td>
<td>
<p>Label for x-axis</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_ylab">ylab</code></td>
<td>
<p>Label for y-axis</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_col">col</code></td>
<td>
<p>standard color parameter</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_lty">lty</code></td>
<td>
<p>standard line type preference</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_lwd">lwd</code></td>
<td>
<p>standard line width preference</p>
</td></tr>
<tr><td><code id="overlapDensity_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to <code>plot()</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

 <p><code>t.test</code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(two65)
overlapDensity(ambient,heated)
t.test(ambient,heated)
</code></pre>

<hr>
<h2 id='ozone'>Ozone Data</h2><span id='topic+ozone'></span>

<h3>Description</h3>

<p>Monthly provisional mean total ozone (in Dobson units) at Halley Bay 
(approximately corrected to Bass-Paur).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ozone</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Year</dt><dd><p>the year</p>
</dd>
<dt>Aug</dt><dd><p>August mean total ozone</p>
</dd>
<dt>Sep</dt><dd><p>September mean total ozone</p>
</dd>
<dt>Oct</dt><dd><p>October mean total ozone</p>
</dd>
<dt>Nov</dt><dd><p>November mean total ozone</p>
</dd>
<dt>Dec</dt><dd><p>December mean total ozone</p>
</dd>
<dt>Jan</dt><dd><p>January mean total ozone</p>
</dd>
<dt>Feb</dt><dd><p>February mean total ozone</p>
</dd>
<dt>Mar</dt><dd><p>March mean total ozone</p>
</dd>
<dt>Apr</dt><dd><p>April mean total ozone</p>
</dd>
<dt>Annual</dt><dd><p>Yearly mean total ozone</p>
</dd>
</dl>



<h3>Source</h3>

<p>Shanklin, J. (2001) Ozone at Halley, Rothera and Vernadsky/Faraday.  
</p>
<p>http://www.antarctica.ac.uk/met/jds/ozone/data/zoz5699.dat
</p>


<h3>References</h3>

<p>Christie, M. (2000) The Ozone Layer: a Philosophy of Science Perspective.
Cambridge University Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AnnualOzone &lt;- ts(ozone$Annual, start=1956)
plot(AnnualOzone)
</code></pre>

<hr>
<h2 id='pair65'>Heated Elastic Bands</h2><span id='topic+pair65'></span>

<h3>Description</h3>

<p>The <code>pair65</code> data frame has 9 rows and 2 columns.
Eighteen elastic bands were divided into nine pairs, with bands
of similar stretchiness placed in the same pair. One member of
each pair was placed in hot water (60-65 degrees C) for four
minutes, while the other was left at ambient temperature.  After
a wait of about ten minutes, the amounts of stretch, under a 1.35 kg
weight, were recorded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pair65</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>heated</dt><dd><p>a numeric vector giving the stretch lengths for
the heated bands</p>
</dd>
<dt>ambient</dt><dd><p>a numeric vector giving the stretch lengths for
the unheated bands</p>
</dd>
</dl>



<h3>Source</h3>

<p>J.H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mean(pair65$heated - pair65$ambient)
sd(pair65$heated - pair65$ambient)
</code></pre>

<hr>
<h2 id='panel.corr'>Scatterplot Panel</h2><span id='topic+panel.corr'></span>

<h3>Description</h3>

<p>This function produces a bivariate scatterplot with the Pearson 
correlation.  This is for use with the function <code>panelplot</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.corr(data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel.corr_+3A_data">data</code></td>
<td>
<p>A data frame with columns x and y</p>
</td></tr>
<tr><td><code id="panel.corr_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# correlation between body and brain weights for 20 mice:

weights &lt;- litters[,-1]
names(weights) &lt;-  c("x","y")
weights &lt;- list(weights)
weights[[1]]$xlim &lt;- range(litters[,2])
weights[[1]]$ylim &lt;- range(litters[,3])
panelplot(weights, panel.corr, totrows=1, totcols=1)
</code></pre>

<hr>
<h2 id='panelCorr'>Scatterplot Panel</h2><span id='topic+panelCorr'></span>

<h3>Description</h3>

<p>This function produces a bivariate scatterplot with the Pearson 
correlation.  This is for use with the function <code>panelplot</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panelCorr(data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panelCorr_+3A_data">data</code></td>
<td>
<p>A data frame with columns x and y</p>
</td></tr>
<tr><td><code id="panelCorr_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# correlation between body and brain weights for 20 mice:

weights &lt;- litters[,-1]
names(weights) &lt;-  c("x","y")
weights &lt;- list(weights)
weights[[1]]$xlim &lt;- range(litters[,2])
weights[[1]]$ylim &lt;- range(litters[,3])
panelplot(weights, panelCorr, totrows=1, totcols=1)
</code></pre>

<hr>
<h2 id='panelplot'>Panel Plot</h2><span id='topic+panelplot'></span>

<h3>Description</h3>

<p>Panel plots of various types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panelplot(data, panel=points, totrows=3, totcols=2, oma=rep(2.5, 4), 
par.strip.text=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panelplot_+3A_data">data</code></td>
<td>
<p> A list consisting of elements, each of which
consists of x, y, xlim and ylim vectors</p>
</td></tr>
<tr><td><code id="panelplot_+3A_panel">panel</code></td>
<td>
<p>The panel function to be plotted</p>
</td></tr>
<tr><td><code id="panelplot_+3A_totrows">totrows</code></td>
<td>
<p>The number of rows in the plot layout</p>
</td></tr>
<tr><td><code id="panelplot_+3A_totcols">totcols</code></td>
<td>
<p>The number of columns in the plot layout</p>
</td></tr>
<tr><td><code id="panelplot_+3A_oma">oma</code></td>
<td>
<p>Outer margin area</p>
</td></tr>
<tr><td><code id="panelplot_+3A_par.strip.text">par.strip.text</code></td>
<td>
<p>A data frame with column cex</p>
</td></tr>
<tr><td><code id="panelplot_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to plotting functions</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>       x1 &lt;- x2 &lt;- x3 &lt;- (11:30)/5
          y1 &lt;- x1 + rnorm(20)/2
          y2 &lt;- 2 - 0.05 * x1 + 0.1 * ((x1 - 1.75))^4 + 1.25 * rnorm(20)
          r &lt;- round(cor(x1, y2), 3)
          rho &lt;- round(cor(rank(x1), rank(y2)), 3)
          y3 &lt;- (x1 - 3.85)^2 + 0.015 + rnorm(20)/4
          theta &lt;- ((2 * pi) * (1:20))/20
          x4 &lt;- 10 + 4 * cos(theta)
          y4 &lt;- 10 + 4 * sin(theta) + (0.5 * rnorm(20))
          r1 &lt;- cor(x1, y1)
          xy &lt;- data.frame(x = c(rep(x1, 3), x4), y = c(y1, y2, y3, y4),
                           gp = rep(1:4, rep(20, 4)))
          xy &lt;- split(xy,xy$gp)
          xlimdf &lt;- lapply(list(x1,x2,x3,x4), range)
          ylimdf &lt;- lapply(list(y1,y2,y3,y4), range)
          xy &lt;- lapply(1:4, function(i,u,v,w){list(xlim=v[[i]],ylim=w[[i]],
                             x=u[[i]]$x, y=u[[i]]$y)},
                                u=xy, v=xlimdf, w=ylimdf)

          panel.corr &lt;- function (data, ...)
              {
              x &lt;- data$x
              y &lt;- data$y
              points(x, y, pch = 16)
              chh &lt;- par()$cxy[2]
              x1 &lt;- min(x)
              y1 &lt;- max(y) - chh/4
              r1 &lt;- cor(x, y)
              text(x1, y1, paste(round(r1, 3)), cex = 0.8, adj = 0)
          }


          panelplot(xy, panel=panel.corr, totrows=2, totcols=2,oma=rep(1,4))
</code></pre>

<hr>
<h2 id='pause'>
Pause before continuing execution  
</h2><span id='topic+pause'></span>

<h3>Description</h3>

<p>If a program produces several plots, isertion of <code>pause()</code> between
two plots suspends execution until the &lt;Enter&gt; key is pressed, to
allow inspection of the current plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pause()
</code></pre>


<h3>Author(s)</h3>

<p>From the &lsquo;sm&rsquo; package of Bowman and Azzalini (1997)
</p>

<hr>
<h2 id='plotSampDist'>Plot(s) of simulated sampling distributions</h2><span id='topic+plotSampDist'></span>

<h3>Description</h3>

<p>Plots are based on the output from <code>simulateSampDist()</code>.
By default, both density plots and normal probability plots
are given, for a sample from the specified population and
for samples of the relevant size(s)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSampDist(sampvalues, graph = c("density", "qq"), cex = 0.925,
             titletext = "Empirical sampling distributions of the",
             popsample=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSampDist_+3A_sampvalues">sampvalues</code></td>
<td>
<p>Object output from <code>simulateSampDist()</code></p>
</td></tr>
<tr><td><code id="plotSampDist_+3A_graph">graph</code></td>
<td>
<p>Either or both of <code>"density"</code> and <code>"qq"</code></p>
</td></tr>
<tr><td><code id="plotSampDist_+3A_cex">cex</code></td>
<td>
<p>Character size parameter, relative to default</p>
</td></tr>
<tr><td><code id="plotSampDist_+3A_titletext">titletext</code></td>
<td>
<p>Title for graph</p>
</td></tr>
<tr><td><code id="plotSampDist_+3A_popsample">popsample</code></td>
<td>
<p>If <code>TRUE</code> show distribution of random
sample from population</p>
</td></tr>
<tr><td><code id="plotSampDist_+3A_...">...</code></td>
<td>
<p>Other graphics parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots graph(s), as described above.
</p>


<h3>Author(s)</h3>

<p>John Maindonald</p>


<h3>References</h3>

<p>Maindonald, J.H. and Braun, W.J. (3rd edn, 2010)
&ldquo;Data Analysis and Graphics Using R&rdquo;, Sections 3.3 and 3.4.</p>


<h3>See Also</h3>

<p>See Also <code>help(simulateSampDist)</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## By default, sample from normal population
simAvs &lt;- simulateSampDist()
par(pty="s")
plotSampDist(simAvs)
## Sample from empirical distribution
simAvs &lt;- simulateSampDist(rpop=rivers)
plotSampDist(simAvs)

## The function is currently defined as
function(sampvalues, graph=c("density", "qq"), cex=0.925,
           titletext="Empirical sampling distributions of the",
           popsample=TRUE, ...){
    if(length(graph)==2)oldpar &lt;- par(mfrow=c(1,2), mar=c(3.1,4.1,1.6,0.6),
               mgp=c(2.5, 0.75, 0), oma=c(0,0,1.5,0), cex=cex)
    values &lt;- sampvalues$values
    numINsamp &lt;- sampvalues$numINsamp
    funtxt &lt;- sampvalues$FUN
    nDists &lt;- length(numINsamp)+1
    nfirst &lt;- 2
    legitems &lt;- paste("Size", numINsamp)
    if(popsample){nfirst &lt;- 1
                  legitems &lt;- c("Size 1", legitems)
                }
    if(match("density", graph)){
      popdens &lt;- density(values[,1], ...)
      avdens &lt;- vector("list", length=nDists)
      maxht &lt;- max(popdens$y)
      ## For each sample size specified in numINsamp, calculate mean
      ## (or other statistic specified by FUN) for numsamp samples
      for(j in nfirst:nDists){
        av &lt;- values[, j]
        avdens[[j]] &lt;- density(av, ...)
        maxht &lt;- max(maxht, avdens[[j]]$y)
      }
    }
    if(length(graph)&gt;0)
      for(graphtype in graph){
        if(graphtype=="density"){
          if(popsample)
          plot(popdens, ylim=c(0, 1.2*maxht), type="l", yaxs="i",
               main="")
          else plot(avdens[[2]], type="n", ylim=c(0, 1.2*maxht),
                    yaxs="i", main="")
          for(j in 2:nDists)lines(avdens[[j]], col=j)
          legend("topleft",
                 legend=legitems,
                 col=nfirst:nDists, lty=rep(1,nDists-nfirst+1), cex=cex)
        }
        if(graphtype=="qq"){
          if(popsample) qqnorm(values[,1], main="")
          else qqnorm(values[,2], type="n")
          for(j in 2:nDists){
            qqav &lt;- qqnorm(values[, j], plot.it=FALSE)
            points(qqav, col=j, pch=j)
           }
            legend("topleft", legend=legitems,
                   col=nfirst:nDists, pch=nfirst:nDists, cex=cex)
       }
      }
    if(par()$oma[3]&gt;0){
      outer &lt;- TRUE
      line=0
    }  else
    {
      outer &lt;- FALSE
      line &lt;- 1.25
    }
    if(!is.null(titletext))
      mtext(side=3, line=line,
            paste(titletext, funtxt),
            cex=1.1, outer=outer)
    if(length(graph)&gt;1)par(oldpar)
  }
</code></pre>

<hr>
<h2 id='plotSimDiags'>
Diagnostic plots for simulated data
</h2><span id='topic+plotSimDiags'></span>

<h3>Description</h3>

<p>This provides diagnostic plots, closely equivalent to those provided by
<code><a href="stats.html#topic+plot.lm">plot.lm</a></code>, for simulated data.  By default, simulated data
are for the fitted model.  Alternatively, simulated data can be
supplied, making it possible to check the effct of fitting, e.g.,
an AR1 model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSimDiags(obj, simvalues = NULL, seed = NULL,
types = NULL, which = c(1:3, 5), layout = c(4, 1), qqline=TRUE,
cook.levels = c(0.5, 1), caption = list("Residuals vs Fitted",
"Normal Q-Q", "Scale-Location", "Cook's distance", "Residuals vs Leverage",
expression("Cook's dist vs Leverage  " * h[ii]/(1 - h[ii]))),
...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSimDiags_+3A_obj">obj</code></td>
<td>

<p>Fitted model object - <code>lm</code> or an object inheriting from <code>lm</code>
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_simvalues">simvalues</code></td>
<td>

<p>Optional matrix of simulated data.
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_seed">seed</code></td>
<td>

<p>Random number seed - set this to make results repeatable.
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_types">types</code></td>
<td>

<p>If set, this should be a list with six elements, ordinarily with
each list element either <code>"p"</code> or <code>c("p","smooth")</code> or
(<code>which=2</code>, <code>which=6</code>) NULL or (<code>which=4</code>) &quot;h&quot;
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_which">which</code></td>
<td>

<p>Set to be a subset of the numbers 1 to 6, as for <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_layout">layout</code></td>
<td>

<p>Controls the number of simulations and the layout of the plots.
For example <code>layout=c(3,4)</code> will give 12 plots in a 3 by 4 layout.
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_qqline">qqline</code></td>
<td>
<p>logical: add line to normal Q-Q plot</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_cook.levels">cook.levels</code></td>
<td>

<p>Levels of Cook's statistics for which contours are to be plotted.
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_caption">caption</code></td>
<td>
<p>list: Captions for the six graphs
</p>
</td></tr>
<tr><td><code id="plotSimDiags_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to plotting functions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diagnotic plots from repeated simulations from the fitted model
provide a useful indication of the range of variation in the model
diagnistics that are consistent with the fitted model.
</p>


<h3>Value</h3>

<p>A list of lattice graphics objects is returned, one for each value of
<code>which</code>.  List elements for which a graphics object is not
returned are set to NULL.  Or if <code>which</code> is of length 1,
a lattice graphics object.
</p>
<table>
<tr><td><code>residVSfitted</code></td>
<td>
<p>Residuals vs fitted</p>
</td></tr>
<tr><td><code>normalQQ</code></td>
<td>
<p>Normal quantile-quantile plot</p>
</td></tr>
<tr><td><code>scaleVSloc</code></td>
<td>
<p>Scale versus location</p>
</td></tr>
<tr><td><code>CookDist</code></td>
<td>
<p>Cook's distance vs observation number</p>
</td></tr>
<tr><td><code>residVSlev</code></td>
<td>
<p>Standardized residuals (for GLMs, standardized
Pearson residuals) vs leverage</p>
</td></tr>
<tr><td><code>CookVSlev</code></td>
<td>
<p>Cook's distance vs leverage</p>
</td></tr>
</table>
<p>For the default <code>which=c(1:3,5)</code>, list items 1, 2, 3 and 5 above
contain graphics objects, with list elements 4 and 6 set to <code>NULL</code>.
</p>


<h3>Note</h3>

<p>The graphics objects contained in individual list elements can be
extracted for printing, or updating and printing, as required.
If the value is returned to the command line, list elements that
are not <code>NULL</code> will be printed in turn.
</p>


<h3>Author(s)</h3>

<p>John Maindonald, with some code chunks adapted from <code>plot.lm</code>
</p>


<h3>References</h3>

<p>See <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+plot.lm">plot.lm</a></code>, <code><a href="#topic+lmdiags">lmdiags</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>women.lm &lt;- lm(height ~ weight, data=women)
gphlist &lt;- plotSimDiags(obj=women.lm, which=c(1:3,5))
</code></pre>

<hr>
<h2 id='plotSimScat'>
Simulate scatterplots, from <code>lm</code> object with a single explanatory variable.
</h2><span id='topic+plotSimScat'></span>

<h3>Description</h3>

<p>This plots simulated y-values, or residuals from such simulations,
against x-values
.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSimScat(obj, sigma = NULL, layout = c(4, 1), type = c("p", "r"),
show = c("points", "residuals"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSimScat_+3A_obj">obj</code></td>
<td>

<p>An <code>lm</code> object with a single explanatory variable.
</p>
</td></tr>
<tr><td><code id="plotSimScat_+3A_sigma">sigma</code></td>
<td>

<p>Standard deviation, if different from that for the supplied <code>lm</code>
object.
</p>
</td></tr>
<tr><td><code id="plotSimScat_+3A_layout">layout</code></td>
<td>

<p>Columns by Rows layout for plots from the simulations.
</p>
</td></tr>
<tr><td><code id="plotSimScat_+3A_type">type</code></td>
<td>

<p>See <code>type</code> as in <code><a href="stats.html#topic+plot.lm">plot.lm</a></code>.
</p>
</td></tr>
<tr><td><code id="plotSimScat_+3A_show">show</code></td>
<td>

<p>Specify <code>points</code> or residuals.
</p>
</td></tr>
<tr><td><code id="plotSimScat_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed to plotting functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A lattice graphics object is returned.
</p>


<h3>Author(s)</h3>

<p>J H Maindonald
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSimDiags">plotSimDiags</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nihills.lm &lt;- lm(timef~time, data=nihills)
plotSimDiags(nihills.lm)

## The function is currently defined as
function (obj, sigma = NULL, layout = c(4, 1), type = c("p",
    "r"), show = c("points", "residuals"))
{
    nsim &lt;- prod(layout)
    if (is.null(sigma))
        sigma &lt;- summary(obj)[["sigma"]]
    hat &lt;- fitted(obj)
    xnam &lt;- all.vars(formula(obj))[2]
    ynam &lt;- all.vars(formula(obj))[1]
    df &lt;- data.frame(sapply(1:nsim, function(x) rnorm(length(hat),
        sd = sigma)))
    if (show[1] == "points")
        df &lt;- df + hat
    simnam &lt;- names(df) &lt;- paste("Simulation", 1:nsim, sep = "")
    df[, c(xnam, ynam)] &lt;- model.frame(obj)[, c(xnam, ynam)]
    if (show[1] != "points") {
        df[, "Residuals"] &lt;- df[, ynam] - hat
        ynam &lt;- "Residuals"
        legadd &lt;- "residuals"
    }
    else legadd &lt;- "data"
    leg &lt;- list(text = paste(c("Simulated", "Actual"), legadd),
        columns = 2)
    formula &lt;- formula(paste(paste(simnam, collapse = "+"), "~",
        xnam))
    parset &lt;- simpleTheme(pch = c(16, 16), lty = 2, col = c("black",
        "gray"))
    gph &lt;- xyplot(formula, data = df, outer = TRUE, par.settings = parset,
        auto.key = leg, lty = 2, layout = layout, type = type)
    formxy &lt;- formula(paste(ynam, "~", xnam))
    addgph &lt;- xyplot(formxy, data = df, pch = 16, col = "gray")
    gph + as.layer(addgph, under = TRUE)
  }
</code></pre>

<hr>
<h2 id='poissonsim'>Simple Poisson Regression Data Simulator</h2><span id='topic+poissonsim'></span>

<h3>Description</h3>

<p>This function simulates simple regression data from a Poisson
model.  It also has the option to create over-dispersed data
of a particular type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poissonsim(x = seq(0, 1, length=101), a = 2, b = -4, intcp.sd=NULL,
           slope.sd=NULL, seed=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="poissonsim_+3A_x">x</code></td>
<td>
<p>a numeric vector representing the explanatory variable</p>
</td></tr>
<tr><td><code id="poissonsim_+3A_a">a</code></td>
<td>
<p>the regression function intercept</p>
</td></tr>
<tr><td><code id="poissonsim_+3A_b">b</code></td>
<td>
<p>the regression function slope</p>
</td></tr>
<tr><td><code id="poissonsim_+3A_intcp.sd">intcp.sd</code></td>
<td>
<p>standard deviation of the (random) intercept</p>
</td></tr>
<tr><td><code id="poissonsim_+3A_slope.sd">slope.sd</code></td>
<td>
<p>standard deviation of the (random) slope</p>
</td></tr>
<tr><td><code id="poissonsim_+3A_seed">seed</code></td>
<td>
<p>numeric constant</p>
</td></tr>
</table>


<h3>Value</h3>

<p> a list consisting of
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>the explanatory variable vector</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>the Poisson response vector</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>poissonsim()
</code></pre>

<hr>
<h2 id='possum'>Possum Measurements</h2><span id='topic+possum'></span><span id='topic+fossum'></span>

<h3>Description</h3>

<p>The <code>possum</code> data frame consists of nine morphometric
measurements on each of 104 mountain brushtail possums, trapped
at seven Australian sites from Southern Victoria to central Queensland.
See <code><a href="#topic+possumsites">possumsites</a></code> for further details.
The <code>fossum</code> data frame is the subset of <code>possum</code> that has
measurements for the 43 females.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(possum)
  data(fossum)
  </code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>case</dt><dd><p>observation number</p>
</dd>
<dt>site</dt><dd><p>one of seven locations where possums were
trapped. The sites were, in order,Cambarville, Bellbird, Whian Whian,
Byrangery, Conondale, Allyn River and Bulburin </p>
</dd>
<dt>Pop</dt><dd><p>a factor which classifies the sites as <code>Vic</code> Victoria,
<code>other</code> New South Wales or Queensland</p>
</dd>
<dt>sex</dt><dd><p>a factor with levels
<code>f</code> female,
<code>m</code> male </p>
</dd>
<dt>age</dt><dd><p>age</p>
</dd>
<dt>hdlngth</dt><dd><p>head length</p>
</dd>
<dt>skullw</dt><dd><p>skull width</p>
</dd>
<dt>totlngth</dt><dd><p>total length</p>
</dd>
<dt>taill</dt><dd><p>tail length</p>
</dd>
<dt>footlgth</dt><dd><p>foot length</p>
</dd>
<dt>earconch</dt><dd><p>ear conch length</p>
</dd>
<dt>eye</dt><dd><p>distance from medial canthus to lateral canthus of
right eye</p>
</dd>
<dt>chest</dt><dd><p>chest girth (in cm)</p>
</dd>
<dt>belly</dt><dd><p>belly girth (in cm)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Lindenmayer, D. B., Viggers, K. L., Cunningham, R. B., and
Donnelly, C. F. 1995. Morphological variation among columns of the
mountain brushtail possum, Trichosurus caninus Ogilby
(Phalangeridae: Marsupiala). Australian Journal of Zoology 43:
449-458.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>boxplot(earconch~sex, data=possum)
pause()

sex &lt;- as.integer(possum$sex)
oldpar &lt;- par(oma=c(2,4,5,4))
pairs(possum[, c(9:11)], pch=c(0,2:7), col=c("red","blue"),
  labels=c("tail\nlength","foot\nlength","ear conch\nlength"))
chh &lt;- par()$cxy[2]; xleg &lt;- 0.05; yleg &lt;- 1.04
oldpar &lt;- par(xpd=TRUE)
legend(xleg, yleg, c("Cambarville", "Bellbird", "Whian Whian  ",
  "Byrangery", "Conondale  ","Allyn River", "Bulburin"), pch=c(0,2:7),
  x.intersp=1, y.intersp=0.75, cex=0.8, xjust=0, bty="n", ncol=4)
text(x=0.2, y=yleg - 2.25*chh, "female", col="red", cex=0.8, bty="n")
text(x=0.75, y=yleg - 2.25*chh, "male", col="blue", cex=0.8, bty="n")
par(oldpar)
pause()

sapply(possum[,6:14], function(x)max(x,na.rm=TRUE)/min(x,na.rm=TRUE))
pause()

here &lt;- na.omit(possum$footlgth)
possum.prc &lt;- princomp(possum[here, 6:14])
pause()

plot(possum.prc$scores[,1] ~ possum.prc$scores[,2],
  col=c("red","blue")[as.numeric(possum$sex[here])],
  pch=c(0,2:7)[possum$site[here]], xlab = "PC1", ylab = "PC2")
  # NB: We have abbreviated the axis titles
chh &lt;- par()$cxy[2]; xleg &lt;- -15; yleg &lt;- 20.5
oldpar &lt;- par(xpd=TRUE)
legend(xleg, yleg, c("Cambarville", "Bellbird", "Whian Whian  ",
  "Byrangery", "Conondale  ","Allyn River", "Bulburin"), pch=c(0,2:7),
  x.intersp=1, y.intersp=0.75, cex=0.8, xjust=0, bty="n", ncol=4)
text(x=-9, y=yleg - 2.25*chh, "female", col="red", cex=0.8, bty="n")
summary(possum.prc, loadings=TRUE, digits=2)
par(oldpar)
pause()

require(MASS)
here &lt;- !is.na(possum$footlgth)
possum.lda &lt;- lda(site ~ hdlngth+skullw+totlngth+ taill+footlgth+
  earconch+eye+chest+belly, data=possum, subset=here)
options(digits=4)
possum.lda$svd   # Examine the singular values
plot(possum.lda, dimen=3)
  # Scatterplot matrix - scores on 1st 3 canonical variates (Figure 11.4)
possum.lda
pause()
boxplot(fossum$totlngth)
</code></pre>

<hr>
<h2 id='possumsites'>Possum Sites</h2><span id='topic+possumsites'></span>

<h3>Description</h3>

<p>The <code>possumsites</code> data frame consists of Longitudes, Latitudes,
and altitudes for the seven sites from Southern Victoria to central Queensland
where the <code><a href="#topic+possum">possum</a></code> observations were made.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possumsites</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Longitude</dt><dd><p>a numeric vector</p>
</dd>
<dt>Latitude</dt><dd><p>a numeric vector</p>
</dd>
<dt>altitude</dt><dd><p>in meters</p>
</dd>
</dl>



<h3>Source</h3>

<p>Lindenmayer, D. B., Viggers, K. L., Cunningham, R. B., and
Donnelly, C. F. 1995. Morphological variation among columns of the
mountain brushtail possum, Trichosurus caninus Ogilby
(Phalangeridae: Marsupiala). Australian Journal of Zoology 43:
449-458.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(oz)
oz(sections=c(3:5, 11:16))
attach(possumsites)
points(Longitude, Latitude, pch=16, col=2)
chw &lt;- par()$cxy[1]
chh &lt;- par()$cxy[2]
posval &lt;- c(2,4,2,2,4,2,2)
text(Longitude+(3-posval)*chw/4, Latitude, row.names(possumsites), pos=posval)
</code></pre>

<hr>
<h2 id='powerplot'>Plot of Power Functions</h2><span id='topic+powerplot'></span>

<h3>Description</h3>

<p>This function plots powers of a variable on the interval [0,10].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>powerplot(expr="x^2", xlab="x", ylab="y", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="powerplot_+3A_expr">expr</code></td>
<td>
<p>Functional form to be plotted</p>
</td></tr>
<tr><td><code id="powerplot_+3A_xlab">xlab</code></td>
<td>
<p>x-axis label</p>
</td></tr>
<tr><td><code id="powerplot_+3A_ylab">ylab</code></td>
<td>
<p>y-axis label</p>
</td></tr>
<tr><td><code id="powerplot_+3A_...">...</code></td>
<td>
<p>Further arguments, to be passed to <code>plot()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Other expressions such as &quot;sin(x)&quot; and &quot;cos(x)&quot;, etc.
could also be plotted with this function, but results are
not guaranteed.
</p>


<h3>Value</h3>

<p>A plot of the given expression on the interval [0,10].
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>   oldpar &lt;- par(mfrow = c(2, 3), mar = par()$mar - c(
        1, 1, 1.0, 1),  mgp = c(1.5, 0.5, 0),  oma=c(0,1,0,1))
#    on.exit(par(oldpar))
    powerplot(expr="sqrt(x)", xlab="")
    powerplot(expr="x^0.25", xlab="", ylab="")
    powerplot(expr="log(x)", xlab="", ylab="")
    powerplot(expr="x^2")
    powerplot(expr="x^4", ylab="")
    powerplot(expr="exp(x)", ylab="")
par(oldpar)</code></pre>

<hr>
<h2 id='poxetc'>Deaths from various causes, in London from 1629 till 1881, with gaps</h2><span id='topic+poxetc'></span>

<h3>Description</h3>

<p>Deaths from &quot;flux&quot; or smallpox, measles, all causes, and ratios
of the the first two categories to total deaths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poxetc)</code></pre>


<h3>Format</h3>

<p>This is a multiple time series consisting of 5 series:
<code>fpox</code>, <code>measles</code>, <code>all</code>, <code>fpox2all</code>, <code>measles2all</code>.
</p>


<h3>Source</h3>

<p>Guy, W. A. 1882. Two hundred and fifty years of small pox in London.
Journal of the Royal Statistical Society 399-443.
</p>


<h3>References</h3>

<p>Lancaster, H. O. 1990. Expectations of Life. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(poxetc)
str(poxetc)
plot(poxetc) 
</code></pre>

<hr>
<h2 id='press'>Predictive Error Sum of Squares</h2><span id='topic+press'></span>

<h3>Description</h3>

<p>Allen's PRESS statistic is computed for a fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>press(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="press_+3A_obj">obj</code></td>
<td>
<p> A <code>lm</code> object </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numeric value.
</p>


<h3>Author(s)</h3>

<p>W.J. Braun</p>


<h3>See Also</h3>

<p><code>lm</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>litters.lm &lt;- lm(brainwt ~ bodywt + lsize, data = litters)
press(litters.lm)
litters.lm0 &lt;- lm(brainwt ~ bodywt + lsize -1, data=litters)
press(litters.lm0) # no intercept
litters.lm1 &lt;- lm(brainwt ~ bodywt, data=litters)
press(litters.lm1) # bodywt only
litters.lm2 &lt;- lm(brainwt ~ bodywt + lsize + lsize:bodywt, data=litters)
press(litters.lm2) # include an interaction term
</code></pre>

<hr>
<h2 id='primates'>Primate Body and Brain Weights</h2><span id='topic+primates'></span>

<h3>Description</h3>

<p>A subset of <code>Animals</code> data frame from the MASS library.
It contains the average body and brain measurements of five
primates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>primates</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>Bodywt</dt><dd><p>a numeric vector consisting of the body weights (in kg) 
of five
different primates</p>
</dd>
<dt>Brainwt</dt><dd><p>a numeric vector consisting of the corresponding brain weights (in g) </p>
</dd>
</dl>



<h3>Source</h3>

<p>P. J. Rousseeuw  and A. M. Leroy (1987) Robust Regression and
Outlier Detection. Wiley, p. 57.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(primates)
plot(x=Bodywt, y=Brainwt, pch=16,
       xlab="Body weight (kg)", ylab="Brain weight (g)",
       xlim=c(5,300), ylim=c(0,1500))
chw &lt;- par()$cxy[1]
chh &lt;- par()$cxy[2]
text(x=Bodywt+chw, y=Brainwt+c(-.1,0,0,.1,0)*chh,
       labels=row.names(primates), adj=0)
detach(primates)
</code></pre>

<hr>
<h2 id='progression'>Progression of Record times for track races, 1912 - 2008</h2><span id='topic+progression'></span>

<h3>Description</h3>

<p>Progression in world record times for track and road races.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(progression)</code></pre>


<h3>Format</h3>

<p>A data frame with 227 observations on the following 4 columns.
</p>

<dl>
<dt><code>year</code></dt><dd><p>Year that time was first recorded</p>
</dd>
<dt><code>Distance</code></dt><dd><p>distance in kilometers</p>
</dd>
<dt><code>Time</code></dt><dd><p>time in minutes</p>
</dd>
<dt><code>race</code></dt><dd><p>character; descriptor for event (100m, mile, ...)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Record times for men's track events, from 1912 onwards.  The
series starts with times that were recognized as record times in 1912,
where available.
</p>


<h3>Source</h3>

<p>Links to sources for the data are at
</p>
<p><a href="https://en.wikipedia.org/wiki/Athletics_world_record">https://en.wikipedia.org/wiki/Athletics_world_record</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(progression)
plot(log(Time) ~ log(Distance), data=progression)
res &lt;- resid(lm(log(Time) ~ log(Distance), data=progression))
plot(res ~ log(Distance), data=progression,
     ylab="Residuals from regression line on log scales")
library(lattice)
xyplot(log(Time) ~ log(Distance), data=progression, type=c("p","r"))
xyplot(log(Time) ~ log(Distance), data=progression,
       type=c("p","smooth"))

</code></pre>

<hr>
<h2 id='qreference'>Normal QQ Reference Plot</h2><span id='topic+qreference'></span>

<h3>Description</h3>

<p>This function computes the normal QQ plot for given data and
allows for comparison with normal QQ plots of simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qreference(test = NULL, m = 50, nrep = 6, distribution = function(x) qnorm(x, 
    mean = ifelse(is.null(test), 0, mean(test)), sd = ifelse(is.null(test), 
    1, sd(test))), seed = NULL, nrows = NULL, cex.strip = 0.75, 
    xlab = NULL, ylab = NULL) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qreference_+3A_test">test</code></td>
<td>
<p>a vector containing a sample to be tested; if not supplied,
all qq-plots are of the reference distribution</p>
</td></tr>
<tr><td><code id="qreference_+3A_m">m</code></td>
<td>
<p>the sample size for the reference samples; default is
test sample size if test sample is supplied</p>
</td></tr>
<tr><td><code id="qreference_+3A_nrep">nrep</code></td>
<td>
<p>the total number of samples, including reference
samples and test sample if any</p>
</td></tr>
<tr><td><code id="qreference_+3A_distribution">distribution</code></td>
<td>
<p>reference distribution; default is standard normal</p>
</td></tr>
<tr><td><code id="qreference_+3A_seed">seed</code></td>
<td>
<p>the random number generator seed</p>
</td></tr>
<tr><td><code id="qreference_+3A_nrows">nrows</code></td>
<td>
<p>number of rows in the plot layout</p>
</td></tr>
<tr><td><code id="qreference_+3A_cex.strip">cex.strip</code></td>
<td>
<p>character expansion factor for labels</p>
</td></tr>
<tr><td><code id="qreference_+3A_xlab">xlab</code></td>
<td>
<p>label for x-axis</p>
</td></tr>
<tr><td><code id="qreference_+3A_ylab">ylab</code></td>
<td>
<p>label for y-axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>QQ plots of the sample (if test is non-null) and all reference samples
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'># qreference(rt(180,1))

# qreference(rt(180,1), distribution=function(x) qt(x, df=1))

# qreference(rexp(180), nrep = 4)

# toycars.lm &lt;- lm(distance ~ angle + factor(car), data = toycars)
# qreference(residuals(toycars.lm), nrep = 9)
</code></pre>

<hr>
<h2 id='races2000'>Scottish Hill Races Data - 2000</h2><span id='topic+races2000'></span>

<h3>Description</h3>

<p>The record times in 2000 for 77 Scottish long distance
races.  We
believe the data are, for the most part, trustworthy.  However,
the <code>dist</code> variable for Caerketton (record 58) seems
to have been variously recorded as 1.5 mi and 2.5 mi.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>races2000</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>dist</dt><dd><p>distance, in miles (on the map)</p>
</dd>
<dt>climb</dt><dd><p>total height gained during the route, in feet</p>
</dd>
<dt>time</dt><dd><p>record time in hours</p>
</dd>
<dt>timef</dt><dd><p>record time in hours for females</p>
</dd>
<dt>type</dt><dd><p>a factor, with levels indicating type of race, 
i.e. hill, marathon, relay, uphill or other</p>
</dd>
</dl>



<h3>Source</h3>

<p>The Scottish Running Resource, http://www.hillrunning.co.uk
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    pairs(races2000[,-5])
</code></pre>

<hr>
<h2 id='rainforest'>Rainforest Data</h2><span id='topic+rainforest'></span>

<h3>Description</h3>

<p>The <code>rainforest</code> data frame has 65 rows and 7 columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rainforest</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>dbh</dt><dd><p>a numeric vector</p>
</dd>
<dt>wood</dt><dd><p>a numeric vector</p>
</dd>
<dt>bark</dt><dd><p>a numeric vector</p>
</dd>
<dt>root</dt><dd><p>a numeric vector</p>
</dd>
<dt>rootsk</dt><dd><p>a numeric vector</p>
</dd>
<dt>branch</dt><dd><p>a numeric vector</p>
</dd>
<dt>species</dt><dd><p>a factor with levels
<code>Acacia mabellae</code>,
<code>C. fraseri</code>,
<code>Acmena smithii</code>,
<code>B. myrtifolia</code> </p>
</dd>
</dl>



<h3>Source</h3>

<p>J. Ash, Australian National University
</p>


<h3>References</h3>

<p>Ash, J. and Helman, C. (1990) Floristics and vegetation
biomass of a forest catchment, Kioloa, south coastal N.S.W.
Cunninghamia, 2: 167-182.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(rainforest$species)
</code></pre>

<hr>
<h2 id='rareplants'>Rare and Endangered Plant Species</h2><span id='topic+rareplants'></span>

<h3>Description</h3>

<p>These data were taken from species lists for South Australia, Victoria and 
Tasmania.  Species were classified as CC, CR, RC and RR, with C denoting
common and R denoting rare.  The first code relates to South Australia 
and Victoria, and the second to Tasmania.  They were further classified
by habitat according to the Victorian register, where D = dry only, 
W = wet only, and WD = wet or dry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rareplants</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;rareplants&quot;
</p>


<h3>Source</h3>

<p>Jasmyn Lynch, Department of Botany and Zoology at Australian National 
University
</p>


<h3>Examples</h3>

<pre><code class='language-R'>chisq.test(rareplants)
</code></pre>

<hr>
<h2 id='rice'>Genetically Modified and Wild Type Rice Data</h2><span id='topic+rice'></span>

<h3>Description</h3>

<p>The <code>rice</code> data frame has 72 rows and 7 columns.
The data are from an experiment that compared wild type (wt)
and genetically modified rice plants (ANU843), each
with three different chemical treatments (F10, NH4Cl, and NH4NO3).   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rice</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>PlantNo</dt><dd><p>a numeric vector</p>
</dd>
<dt>Block</dt><dd><p>a numeric vector</p>
</dd>
<dt>RootDryMass</dt><dd><p>a numeric vector</p>
</dd>
<dt>ShootDryMass</dt><dd><p>a numeric vector</p>
</dd>
<dt>trt</dt><dd><p>a factor with levels
<code>F10</code>,
<code>NH4Cl</code>,
<code>NH4NO3</code>,
<code>F10 +ANU843</code>,
<code>NH4Cl +ANU843</code>,
<code>NH4NO3 +ANU843</code></p>
</dd>
<dt>fert</dt><dd><p>a factor with levels
<code>F10</code>
<code>NH4Cl</code>
<code>NH4NO3</code></p>
</dd>
<dt>variety</dt><dd><p>a factor with levels
<code>wt</code>
<code>ANU843</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Perrine, F.M., Prayitno, J., Weinman, J.J., Dazzo, F.B. and
Rolfe, B. 2001.  Rhizobium plasmids are involved in the inhibition or
stimulation of rice growth and development.  Australian Journal of
Plant Physiology 28: 923-927.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("One and Two-Way Comparisons - Example 4.5")
attach(rice)
oldpar &lt;- par(las = 2)
stripchart(ShootDryMass ~ trt, pch=1, cex=1, xlab="Level of factor 1")
detach(rice)
pause()

rice.aov &lt;- aov(ShootDryMass ~ trt, data=rice); anova(rice.aov)
anova(rice.aov)
pause()

summary.lm(rice.aov)$coef
pause()

rice$trt &lt;- relevel(rice$trt, ref="NH4Cl")
  # Set NH4Cl as the baseline

fac1 &lt;- factor(sapply(strsplit(as.character(rice$trt)," \\+"), function(x)x[1]))
anu843 &lt;- sapply(strsplit(as.character(rice$trt), "\\+"), 
function(x)c("wt","ANU843")[length(x)])
anu843 &lt;- factor(anu843, levels=c("wt", "ANU843"))
attach(rice)
interaction.plot(fac1, anu843, ShootDryMass)
detach(rice)
par(oldpar)

</code></pre>

<hr>
<h2 id='rockArt'>Pacific Rock Art features</h2><span id='topic+rockArt'></span>

<h3>Description</h3>

<p>Data characterise rock art at 103 sites in the Pacific.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rockArt</code></pre>


<h3>Format</h3>

<p>A data frame with 103 observations on the following 641 variables.
</p>

<dl>
<dt><code>Site.No.</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Site.Name</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Site.Code</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>District</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Island</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Country</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Technique</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Engtech</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>red</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>black</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>yellow</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>white</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>green</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>red.blk</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>red.wh</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>red.yell</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>r.w.y</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>black.white</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>blue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Geology</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Topography</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Location</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Proxhab.km.</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Proxcoast.km.</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Maxheight.m.</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Language</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>No.motif</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>Ca1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ca2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ca3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ca4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cb5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cb6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cc20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cd27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ce28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ce29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf33</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf34</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf35</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf37</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cf38</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cg39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cg40</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ch41</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ch42</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ci43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ci44</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cj45</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ck46</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ck47</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cl48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm49</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm50</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm51</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm52</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm53</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm54</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm55</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm56</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm57</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cm58</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn59</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn60</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn61</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn62</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn63</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn64</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn65</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn66</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn67</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn68</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn69</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn70</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cn71</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co72</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co73</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co74</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co75</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co76</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co77</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co78</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Co79</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cp80</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq81</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq82</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq83</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq84</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq85</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq86</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq87</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq88</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq89</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq90</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq91</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq92</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq93</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq94</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq95</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq96</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cq97</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cr98</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cr99</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cr100</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cr101</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cs102</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cs103</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cs104</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cs105</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cs106</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ct107</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C108</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C109</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C110</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C111</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSa1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSd8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSf9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSg10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSj11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSj12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSj13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSl14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSm15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSm16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn33</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn34</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSn35</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSo36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSo37</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSp38</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSq39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSq40</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSt41</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSu42</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oa1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oc2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Od3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Od4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oe5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Of6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Of7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Of8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Of9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Og15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oi16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Om17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Om18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Om19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Om20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Om21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>On22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>On23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>On24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oq25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oq26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Oq27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>.u28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ov29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ov30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>O31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>O32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>O33</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sa1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sb2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sb3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sd5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sd6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sd7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Se8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Si9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sm10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sm11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sx14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sx15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sx16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sx17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sy18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sz19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>S25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCd7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCm8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCn9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCn10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCw11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCx12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCx13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCx14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCx15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCx16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCy17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SCy18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SC26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRf5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRf6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRf7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRj8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SR9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SR10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bn2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bn3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bn4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bt5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bx6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ha1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hg2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hn3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hq4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hq5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TDd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TDf2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TDj3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TDn4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TDq5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TD6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TD7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TD8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TD9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dc1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dg2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dh3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dk4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dm5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dm6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ta1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tc2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tc3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tc4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Td5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tf6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tf7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Tg8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Th9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>To10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>T16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CNg1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CN8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ld1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lf2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lg3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lp4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LS2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LL2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LL3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LL4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LL5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EGd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EGf2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCn2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCn3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMc1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMf4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMf5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMn6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMx7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EM8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EM9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EM10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EM11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EM12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TE2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TE3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TE4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TE5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BWe1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BWn2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BWn3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TS9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Pg1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Pg2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Pg3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DUaa1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DUw2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DU3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CP12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STg4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STaa5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STaa6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STaa7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STaa8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ST9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ST10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ST11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ST12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Wd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Wd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Wd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Wd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Wn5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Waa6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Waa7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>W13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zn3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zw4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zw5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zaa6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Z12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd33</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd34</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd35</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd37</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLd38</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn40</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn41</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn42</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn44</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn45</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn46</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn47</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLn48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLw49</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL50</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL51</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL52</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL53</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL54</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL55</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL56</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL57</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL58</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CL59</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xd13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xf14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xk15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xn32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xp33</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xp34</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xp35</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xq36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xq37</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xq38</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X40</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X41</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X42</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X44</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X45</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X46</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X47</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X49</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X50</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qe2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qe3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qh9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qn10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qn11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Qt12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Q22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TZd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TZf2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TZh3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TZ4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CRd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CR2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CR3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EUd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EUd2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EUg3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EUm4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EUw5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EU6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ud1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ud2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ud3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Uaa4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>U5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Vd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LWE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LWE2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ad1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Al2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Am3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>An4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Aw5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Aaa6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EVd1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EVg2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ECL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EFe1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EFm2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EFm3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EF4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LPo1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LPq2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LP5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PT1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CSC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CSR</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCRC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Anthrop</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Turtle</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Boat</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Canoe</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hand</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Foot</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Lizard</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Crocodile</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Jellyfish</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bird</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Anthrobird</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Axe</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Marine</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Face</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zoo6</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note the vignette <b>rockArt</b>.
</p>


<h3>Source</h3>

<p>Meredith Wilson: <em>Picturing Pacific Pre-History</em> (PhD thesis), 2002,
Australian National University.
</p>


<h3>References</h3>

<p>Meredith Wilson: Rethinking regional analyses of Western Pacific
rock-art. <em>Records of the Australian Museum</em>, Supplement 29: 173-186.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rockArt)
rockart.dist &lt;- dist(x = as.matrix(rockArt[, 28:641]), method = "binary")
sum(rockart.dist==1)/length(rockart.dist)
plot(density(rockart.dist, to = 1))
rockart.cmd &lt;- cmdscale(rockart.dist)
tab &lt;- table(rockArt$District)
district &lt;- as.character(rockArt$District)
district[!(rockArt$District %in% names(tab)[tab&gt;5])] &lt;- "other"
## Not run: 
xyplot(rockart.cmd[,2] ~ rockart.cmd[,1], groups=district,
       auto.key=list(columns=5),
       par.settings=list(superpose.symbol=list(pch=16)))
library(MASS)
## For sammon, need to avoid zero distances
omit &lt;- c(47, 54, 60, 63, 92)
rockart.dist &lt;- dist(x = as.matrix(rockArt[-omit, 28:641]), method = "binary")
rockart.cmd &lt;- cmdscale(rockart.dist)
rockart.sam &lt;- sammon(rockart.dist, rockart.cmd)
xyplot(rockart.sam$points[,2] ~ rockart.sam$points[,1],
       groups=district[-omit], auto.key=list(columns=5),
       par.settings=list(superpose.symbol=list(pch=16)))
## Notice the very different appearance of the Sammon plot

## End(Not run)
</code></pre>

<hr>
<h2 id='roller'>Lawn Roller Data</h2><span id='topic+roller'></span>

<h3>Description</h3>

<p>The <code>roller</code> data frame has 10 rows and 2 columns.
Different weights of roller were rolled over different parts
of a lawn, and the depression was recorded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roller</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>weight</dt><dd><p>a numeric vector consisting of the roller weights</p>
</dd>
<dt>depression</dt><dd><p>the depth of the depression made in the grass
under the roller</p>
</dd>
</dl>



<h3>Source</h3>

<p>Stewart, K.M., Van Toor, R.F., Crosbie,
S.F. 1988. Control of grass grub (Coleoptera: Scarabaeidae) with
rollers of different design. N.Z. Journal of Experimental Agriculture
16: 141-150.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(roller)
roller.lm &lt;- lm(depression ~ weight, data = roller)
plot(roller.lm, which = 4)
</code></pre>

<hr>
<h2 id='sampdist'>
Plot sampling distribution of mean or other sample statistic.
</h2><span id='topic+sampdist'></span>

<h3>Description</h3>

<p>The function <code>sampvals</code> generates the data.  A density plot
of a normal probability plot is provided, for one or mare sample
sizes.  For a density plot, the density estimate for the population
is superimposed in gray.  For the normal probability plot, the
population plot is a dashed gray line. Default arguments give the
sampling distribution of the mean, for a distribution that is
mildly positively skewed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampdist(sampsize = c(3, 9, 30), seed = NULL, nsamp = 1000, FUN = mean,
         sampvals = function(n) exp(rnorm(n, mean = 0.5, sd = 0.3)),
         tck = NULL, plot.type = c("density", "qq"), layout = c(3, 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampdist_+3A_sampvals">sampvals</code></td>
<td>

<p>Function that generates the data.  For sampling from existing data
values, this might be function that generates bootstrap samples.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_sampsize">sampsize</code></td>
<td>

<p>One or more sample sizes.  A plot will be provided for each different
sample size.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_seed">seed</code></td>
<td>

<p>Specify a seed if it is required to make the exact set(s) of sample
values reproducible.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_nsamp">nsamp</code></td>
<td>

<p>Number of samples.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_fun">FUN</code></td>
<td>

<p>Function that calculates the sample statistic.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_plot.type">plot.type</code></td>
<td>

<p>Specify <code>density</code>, or <code>qq</code>.  Or if no plot is required,
specify <code>""</code>.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_tck">tck</code></td>
<td>

<p>Tick size on lattice plots, by default 1, but 0.5 may be suitable
for plots that are, for example, 50% of the default dimensions
in each direction.
</p>
</td></tr>
<tr><td><code id="sampdist_+3A_layout">layout</code></td>
<td>
<p>Layout on page, e.g. <code>c(3,1)</code> for a 3 columns
by one row layout.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame
</p>


<h3>Author(s)</h3>

<p>John Maindonald.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sampdist(plot.type="density")
sampdist(plot.type="qq")

## The function is currently defined as
  function (sampsize = c(3, 9, 30), seed = NULL, nsamp = 1000, FUN = mean,
            sampvals = function(n) exp(rnorm(n, mean = 0.5, sd = 0.3)),
            tck = NULL, plot.type = c("density", "qq"), layout = c(3,
                                                          1))
{
  if (!is.null(seed))
    set.seed(seed)
  ncases &lt;- length(sampsize)
  y &lt;- sampvals(nsamp)
  xlim = quantile(y, c(0.01, 0.99))
  xlim &lt;- xlim + c(-1, 1) * diff(xlim) * 0.1
  samplingDist &lt;- function(sampsize=3, nsamp=1000, FUN=mean)
    apply(matrix(sampvals(sampsize*nsamp), ncol=sampsize), 1, FUN)
  df &lt;- data.frame(sapply(sampsize, function(x)samplingDist(x, nsamp=nsamp)))
  names(df) &lt;- paste("y", sampsize, sep="")
  form &lt;- formula(paste("~", paste(names(df), collapse="+")))
  lab &lt;- lapply(sampsize, function(x) substitute(A, list(A = paste(x))))
  if (plot.type[1] == "density")
    gph &lt;- densityplot(form, data=df, layout = layout, outer=TRUE,
                       plot.points = FALSE, panel = function(x, ...) {
                         panel.densityplot(x, ..., col = "black")
                         panel.densityplot(y, col = "gray40", lty = 2,
                                           ...)
                       }, xlim = xlim, xlab = "", scales = list(tck = tck),
                       between = list(x = 0.5), strip = strip.custom(strip.names = TRUE,
                       factor.levels = as.expression(lab), var.name = "Sample size",
                                                  sep = expression(" = ")))
  else if (plot.type[1] == "qq")
    gph &lt;- qqmath(form, data = df, layout = layout, plot.points = FALSE,
                  outer=TRUE,
                  panel = function(x, ...) {
                    panel.qqmath(x, ..., col = "black", alpha=0.5)
                    panel.qqmath(y, col = "gray40", lty = 2, type = "l",
                                 ...)
                  }, xlab = "", xlim = c(-3, 3), ylab = "", scales = list(tck = tck),
                  between = list(x = 0.5), strip = strip.custom(strip.names = TRUE,
                  factor.levels = as.expression(lab), var.name = "Sample size",
                                             sep = expression(" = ")))
  if (plot.type[1] %in% c("density", "qq"))
    print(gph)
  invisible(df)
}
</code></pre>

<hr>
<h2 id='science'>School Science Survey Data</h2><span id='topic+science'></span>

<h3>Description</h3>

<p>The <code>science</code> data frame has 1385 rows and 7 columns.
</p>
<p>The data are on attitudes to science, from a survey where there were
results from 20 classes in private schools and 46 classes in public
schools.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>science</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>State</dt><dd><p>a factor with levels
<code>ACT</code> Australian Capital Territory,
<code>NSW</code> New South Wales</p>
</dd>
<dt>PrivPub</dt><dd><p>a factor with levels
<code>private</code> school,
<code>public</code> school</p>
</dd>
<dt>school</dt><dd><p>a factor, coded to identify the school</p>
</dd>
<dt>class</dt><dd><p>a factor, coded to identify the class</p>
</dd>
<dt>sex</dt><dd><p>a factor with levels
<code>f</code>, <code>m</code> </p>
</dd>
<dt>like</dt><dd><p> a summary score
based on two of the questions, on a scale from 1 (dislike)
to 12 (like)</p>
</dd>
<dt>Class</dt><dd><p>a factor with levels corresponding to each class</p>
</dd>
</dl>



<h3>Source</h3>

<p>Francine Adams,  Rosemary Martin and Murali Nayadu, Australian
National University
</p>


<h3>Examples</h3>

<pre><code class='language-R'>classmeans &lt;- with(science, aggregate(like, by=list(PrivPub, Class), mean))
names(classmeans) &lt;- c("PrivPub","Class","like")
dim(classmeans)

attach(classmeans)
boxplot(split(like, PrivPub), ylab = "Class average of attitude to science score", boxwex = 0.4)
rug(like[PrivPub == "private"], side = 2)
rug(like[PrivPub == "public"], side = 4)
detach(classmeans)
if(require(lme4, quietly=TRUE)) {
science.lmer &lt;- lmer(like ~ sex + PrivPub + (1 | school) +
                     (1 | school:class), data = science,
                     na.action=na.exclude)
summary(science.lmer)
science1.lmer &lt;- lmer(like ~ sex + PrivPub + (1 | school:class),
                      data = science, na.action=na.exclude)
summary(science1.lmer)
ranf &lt;- ranef(obj = science1.lmer, drop=TRUE)[["school:class"]]
flist &lt;- science1.lmer@flist[["school:class"]]
privpub &lt;- science[match(names(ranf), flist), "PrivPub"]
num &lt;- unclass(table(flist)); numlabs &lt;- pretty(num)
## Plot effect estimates vs numbers
plot(sqrt(num), ranf, xaxt="n", pch=c(1,3)[as.numeric(privpub)],
     xlab="# in class (square root scale)",
     ylab="Estimate of class effect")
lines(lowess(sqrt(num[privpub=="private"]),
             ranf[privpub=="private"], f=1.1), lty=2)
lines(lowess(sqrt(num[privpub=="public"]),
             ranf[privpub=="public"], f=1.1), lty=3)
axis(1, at=sqrt(numlabs), labels=paste(numlabs))
}
</code></pre>

<hr>
<h2 id='seedrates'>Barley Seeding Rate Data</h2><span id='topic+seedrates'></span>

<h3>Description</h3>

<p>The <code>seedrates</code> data frame has 5 rows and 2 columns on 
the effect of seeding rate of barley on yield.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedrates</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>rate</dt><dd><p>the seeding rate</p>
</dd>
<dt>grain</dt><dd><p>the number of grain per head of barley</p>
</dd>
</dl>



<h3>Source</h3>

<p>McLeod, C.C. 1982.
Effect of rates of seeding on barley grown for grain. New Zealand 
Journal of Agriculture 10: 133-136.</p>


<h3>References</h3>

<p>Maindonald J H 1992. Statistical design, analysis and presentation
issues. New Zealand Journal of Agricultural Research 35: 121-141.</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(grain~rate,data=seedrates,xlim=c(50,180),ylim=c(15.5,22),axes=FALSE)
new.df&lt;-data.frame(rate=(2:8)*25)
seedrates.lm1&lt;-lm(grain~rate,data=seedrates)
seedrates.lm2&lt;-lm(grain~rate+I(rate^2),data=seedrates)
hat1&lt;-predict(seedrates.lm1,newdata=new.df,interval="confidence")
hat2&lt;-predict(seedrates.lm2,newdata=new.df,interval="confidence")
axis(1,at=new.df$rate); axis(2); box()
z1&lt;-spline(new.df$rate, hat1[,"fit"]); z2&lt;-spline(new.df$rate,   
hat2[,"fit"])
rate&lt;-new.df$rate; lines(z1$x,z1$y)
lines(spline(rate,hat1[,"lwr"]),lty=1,col=3)
lines(spline(rate,hat1[,"upr"]),lty=1,col=3)
lines(z2$x,z2$y,lty=4)
lines(spline(rate,hat2[,"lwr"]),lty=4,col=3)
lines(spline(rate,hat2[,"upr"]),lty=4,col=3)</code></pre>

<hr>
<h2 id='show.colors'>Show R's Colors</h2><span id='topic+show.colors'></span>

<h3>Description</h3>

<p>This function displays the built-in colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show.colors(type=c("singles", "shades", "gray"), order.cols=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show.colors_+3A_type">type</code></td>
<td>
<p>type of display - single, multiple or gray shades</p>
</td></tr>
<tr><td><code id="show.colors_+3A_order.cols">order.cols</code></td>
<td>
<p> Arrange colors in order </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot of colors for which there is a single shade (type = &quot;single&quot;),
multiple shades (type = &quot;multiple&quot;), or gray shades (type = &quot;gray&quot;)
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(MASS)
show.colors()
</code></pre>

<hr>
<h2 id='simulateLinear'>Simulation of Linear Models for ANOVA vs. Regression Comparison</h2><span id='topic+simulateLinear'></span>

<h3>Description</h3>

<p>This function simulates a number of bivariate data sets
in which there are replicates at each level of the predictor.
The p-values for ANOVA and for the regression slope are
compared, and a lattice graphics object returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateLinear(sd=2, npoints=5, nrep=4, nsets=200, graphtype="xy",
seed=21, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateLinear_+3A_sd">sd</code></td>
<td>
<p>The error standard deviation</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_npoints">npoints</code></td>
<td>
<p>Number of distinct predictor levels</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_nrep">nrep</code></td>
<td>
<p>Number of replications at each level</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_nsets">nsets</code></td>
<td>
<p>Number of simulation runs</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_graphtype">graphtype</code></td>
<td>
<p>Type of graph; x-y plot (<code>graphtype="xy"</code>),
overlaid density plots (<code>graphtype="density"</code>), or density
plot for x-y difference (<code>graphtype="density-diff"</code>)
</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_seed">seed</code></td>
<td>
<p>Random Number generator seed</p>
</td></tr>
<tr><td><code id="simulateLinear_+3A_...">...</code></td>
<td>
<p>Additional arguments, to be passed through to the
lattice function that is called</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A lattice graphics object.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>Examples</h3>

<pre><code class='language-R'>simulateLinear()
</code></pre>

<hr>
<h2 id='simulateSampDist'>Simulated sampling distribution of mean or other statistic</h2><span id='topic+simulateSampDist'></span>

<h3>Description</h3>

<p>Simulates the sample distribution of the specified statistic,
for samples of the size(s) specified in <code>numINsamp</code>.
Additionally a with replacement) sample is drawn from the
specified population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateSampDist(rpop = rnorm, numsamp = 100, numINsamp = c(4, 16),
                 FUN = mean, seed=NULL
      )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateSampDist_+3A_rpop">rpop</code></td>
<td>
<p>Either a function that generates random samples from
the specified distribution, or a vector of values that define the
population (i.e., an empirical distribution)</p>
</td></tr>
<tr><td><code id="simulateSampDist_+3A_numsamp">numsamp</code></td>
<td>
<p>Number of samples that should be taken.  For close
approximation of the asymptotic distribution (e.g., for the mean)
this number should be large</p>
</td></tr>
<tr><td><code id="simulateSampDist_+3A_numinsamp">numINsamp</code></td>
<td>
<p>Size(s) of each of the <code>numsamp</code> sample(s)</p>
</td></tr>
<tr><td><code id="simulateSampDist_+3A_fun">FUN</code></td>
<td>
<p>Function to calculate the statistic whose sampling
distribution is to be simulated</p>
</td></tr>
<tr><td><code id="simulateSampDist_+3A_seed">seed</code></td>
<td>
<p>Optional seed for random number generation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List, with elements <code>values</code>, <code>numINsamp</code> and <code>FUN</code>
</p>
<table>
<tr><td><code>values</code></td>
<td>
<p>Matrix, with dimensions <code>numsamp</code> by
<code>numINsamp + 1</code>.  The first column has a random with
replacement sample from the population, while the remaining
<code>length(numINsamp)</code> columns hold simulated values from sampling
distributions with samples of the specified size(s)</p>
</td></tr>
<tr><td><code>numINsamp</code></td>
<td>
<p>Input value of <code>numINsamp</code></p>
</td></tr>
<tr><td><code>numsamp</code></td>
<td>
<p>Input value of <code>numsamp</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>John Maindonald</p>


<h3>References</h3>

<p>Maindonald, J.H. and Braun, W.J. (3rd edn, 2010) <em>Data
Analysis and Graphics Using R</em>, 3rd edn, Sections 3.3 and
3.4</p>


<h3>See Also</h3>

<p><code>help(plotSampDist)</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## By default, sample from normal population
simAvs &lt;- simulateSampDist()
par(pty="s")
plotSampDist(simAvs)
## Sample from empirical distribution
simAvs &lt;- simulateSampDist(rpop=rivers)
plotSampDist(simAvs)


## The function is currently defined as
function(rpop=rnorm, numsamp=100, numINsamp=c(4,16), FUN=mean,
seed=NULL){
    if(!is.null(seed))set.seed(seed)
    funtxt &lt;- deparse(substitute(FUN))
    nDists &lt;- length(numINsamp)+1
    values &lt;- matrix(0, nrow=numsamp, ncol=nDists)
    if(!is.function(rpop)) {
      x &lt;- rpop
      rpop &lt;- function(n)sample(x, n, replace=TRUE)
    }
    values[,1] &lt;- rpop(numsamp)
    for(j in 2:nDists){
      n &lt;- numINsamp[j-1]
      for(i in 1:numsamp)values[i, j] &lt;- FUN(rpop(n))
    }
    colnames(values) &lt;- paste("Size", c(1, numINsamp))
    invisible(list(values=values, numINsamp=numINsamp, FUN=funtxt))
  }
</code></pre>

<hr>
<h2 id='socsupport'>Social Support Data</h2><span id='topic+socsupport'></span>

<h3>Description</h3>

<p>Data from a survey on social and other kinds of support.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>socsupport</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>gender</dt><dd><p>a factor with levels
<code>female</code>, <code>male</code> </p>
</dd>
<dt>age</dt><dd><p>age, in years, with levels
<code>18-20</code>, <code>21-24</code>, <code>25-30</code>, 
<code>31-40</code>,<code>40+</code> </p>
</dd>
<dt>country</dt><dd><p>a factor with levels <code>australia</code>, 
<code>other</code> </p>
</dd>
<dt>marital</dt><dd><p>a factor with levels <code>married</code>,
<code>other</code>, <code>single</code> </p>
</dd>
<dt>livewith</dt><dd><p>a factor with levels <code>alone</code>, 
<code>friends</code>, <code>other</code>, <code>parents</code>, 
<code>partner</code>, <code>residences</code></p>
</dd>
<dt>employment</dt><dd><p>a factor with levels
<code>employed fulltime</code>, <code>employed part-time</code>, 
<code>govt assistance</code>, <code>other</code>, <code>parental support</code></p>
</dd> 
<dt>firstyr</dt><dd><p>a factor with levels <code>first year</code>, 
<code>other</code> </p>
</dd>
<dt>enrolment</dt><dd><p>a factor with levels
<code>full-time</code>, <code>part-time</code>, <code>&lt;NA&gt;</code></p>
</dd>
<dt>emotional</dt><dd><p>summary of 5 questions on emotional support availability</p>
</dd>
<dt>emotionalsat</dt><dd><p>summary of 5 questions on emotional support satisfaction</p>
</dd>
<dt>tangible</dt><dd><p>summary of 4 questions on availability of tangible support</p>
</dd>
<dt>tangiblesat</dt><dd><p>summary of 4 questions on satisfaction with tangible support</p>
</dd>
<dt>affect</dt><dd><p>summary of 3 questions on availability of affectionate support sources</p>
</dd>
<dt>affectsat</dt><dd><p>summary of 3 questions on satisfaction with affectionate support sources</p>
</dd>
<dt>psi</dt><dd><p>summary of 3 questions on availability of positive social interaction</p>
</dd>
<dt>psisat</dt><dd><p>summary of 3 questions on satisfaction with positive social 
interaction</p>
</dd>
<dt>esupport</dt><dd><p>summary of 4 questions on extent of emotional support sources</p>
</dd>
<dt>psupport</dt><dd><p>summary of 4 questions on extent of practical support sources</p>
</dd>
<dt>supsources</dt><dd><p>summary of 4 questions on extent of social support 
sources (formerly, socsupport)</p>
</dd>
<dt>BDI</dt><dd><p>Score on the Beck depression index (summary of 21 questions)</p>
</dd>
</dl>



<h3>Source</h3>

<p>Melissa Manning, Psychology, Australian National University
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(socsupport)

not.na &lt;- apply(socsupport[,9:19], 1, function(x)!any(is.na(x)))
ss.pr1 &lt;- princomp(as.matrix(socsupport[not.na, 9:19]), cor=TRUE)  
pairs(ss.pr1$scores[,1:3])
sort(-ss.pr1$scores[,1])        # Minus the largest value appears first
pause()

not.na[36] &lt;- FALSE
ss.pr &lt;- princomp(as.matrix(socsupport[not.na, 9:19]), cor=TRUE)  
summary(ss.pr)          # Examine the contribution of the components
pause()

# We now regress BDI on the first six principal components:
ss.lm &lt;- lm(BDI[not.na] ~ ss.pr$scores[, 1:6], data=socsupport)
summary(ss.lm)$coef
pause()

ss.pr$loadings[,1]
plot(BDI[not.na] ~  ss.pr$scores[ ,1], col=as.numeric(gender), 
pch=as.numeric(gender),  xlab ="1st principal component", ylab="BDI")
topleft &lt;- par()$usr[c(1,4)]
legend(topleft[1], topleft[2], col=1:2, pch=1:2, legend=levels(gender))
</code></pre>

<hr>
<h2 id='softbacks'>Measurements on a Selection of Paperback Books</h2><span id='topic+softbacks'></span>

<h3>Description</h3>

<p>This is a subset of the <code>allbacks</code> data frame 
which gives measurements
on the volume and weight of 8 paperback books.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>softbacks</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>volume</dt><dd><p>a numeric vector giving the book volumes
in cubic centimeters</p>
</dd>
<dt>weight</dt><dd><p>a numeric vector giving the weights in grams</p>
</dd>
</dl>



<h3>Source</h3>

<p>The bookshelf of J. H. Maindonald.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print("Outliers in Simple Regression - Example 5.2")
paperback.lm &lt;- lm(weight ~ volume, data=softbacks)
summary(paperback.lm)
plot(paperback.lm)
</code></pre>

<hr>
<h2 id='sorption'>sorption data set</h2><span id='topic+sorption'></span>

<h3>Description</h3>

<p>Concentration-time measurements on different varieties of apples
under methyl bromide injection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sorption)</code></pre>


<h3>Format</h3>

<p>A data frame with 192 observations on the following 14 variables.
</p>

<dl>
<dt>m5</dt><dd><p>a numeric vector</p>
</dd>
<dt>m10</dt><dd><p>a numeric vector</p>
</dd>
<dt>m30</dt><dd><p>a numeric vector</p>
</dd>
<dt>m60</dt><dd><p>a numeric vector</p>
</dd>
<dt>m90</dt><dd><p>a numeric vector</p>
</dd>
<dt>m120</dt><dd><p>a numeric vector</p>
</dd>
<dt>ct</dt><dd><p>concentration-time</p>
</dd>
<dt>Cultivar</dt><dd><p>a factor with levels <code>Pacific Rose</code> <code>BRAEBURN</code> <code>Fuji</code> <code>GRANNY</code> <code>Gala</code> <code>ROYAL</code> <code>Red Delicious</code> <code>Splendour</code></p>
</dd>
<dt>Dose</dt><dd><p>injected dose of methyl bromide</p>
</dd>
<dt>rep</dt><dd><p>replicate number, within Cultivar and year</p>
</dd>
<dt>year</dt><dd><p>a factor with levels <code>1988</code> <code>1989</code> <code>1998</code> <code>1999</code></p>
</dd>
<dt>year.rep</dt><dd><p>a factor with levels <code>1988:1</code> <code>1988:2</code> <code>1988:3</code> <code>1989:1</code> <code>1989:2</code> <code>1998:1</code> <code>1998:2</code> <code>1998:3</code> <code>1999:1</code> <code>1999:2</code></p>
</dd>
<dt>gp</dt><dd><p>a factor with levels <code>BRAEBURN1</code> <code>BRAEBURN2</code> <code>Fuji1</code> <code>Fuji10</code> <code>Fuji2</code> <code>Fuji6</code> <code>Fuji7</code> <code>Fuji8</code> <code>Fuji9</code> <code>GRANNY1</code> <code>GRANNY2</code> <code>Gala4</code> <code>Gala5</code> <code>Pacific Rose10</code> <code>Pacific Rose6</code> <code>Pacific Rose7</code> <code>Pacific Rose8</code> <code>Pacific Rose9</code> <code>ROYAL1</code> <code>ROYAL2</code> <code>Red Del10</code> <code>Red Del9</code> <code>Red Delicious1</code> <code>Red Delicious2</code> <code>Red Delicious3</code> <code>Red Delicious4</code> <code>Red Delicious5</code> <code>Red Delicious6</code> <code>Red Delicious7</code> <code>Red Delicious8</code> <code>Splendour4</code> <code>Splendour5</code></p>
</dd>
<dt>inyear</dt><dd><p>a factor with levels <code>1</code> <code>2</code> <code>3</code> <code>4</code> <code>5</code> <code>6</code></p>
</dd>
</dl>


<hr>
<h2 id='SP500close'>Closing Numbers for S and P 500 Index</h2><span id='topic+SP500close'></span>

<h3>Description</h3>

<p>Closing numbers for S and P 500 Index, Jan. 1, 1990 through
early 2000.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SP500close</code></pre>


<h3>Source</h3>

<p>Derived from SP500 in the MASS library.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ts.plot(SP500close)
</code></pre>

<hr>
<h2 id='SP500W90'>Closing Numbers for S and P 500 Index - First 100 Days of 1990</h2><span id='topic+SP500W90'></span>

<h3>Description</h3>

<p>Closing numbers for S and P 500 Index, Jan. 1, 1990 through
early 2000.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SP500W90</code></pre>


<h3>Source</h3>

<p>Derived from SP500 in the MASS library.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ts.plot(SP500W90)
</code></pre>

<hr>
<h2 id='spam7'>Spam E-mail Data</h2><span id='topic+spam7'></span>

<h3>Description</h3>

<p>The data consist of 4601 email items, of which 1813 items were identified
as spam.  This is a subset of the full dataset, with six only of the
57 explanatory variables in the complete dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spam7</code></pre>


<h3>Format</h3>

<p>Columns included are:
</p>

<dl>
<dt>crl.tot</dt><dd><p>total length of uninterrupted sequences of capitals</p>
</dd>
<dt>dollar</dt><dd><p>Occurrences of the dollar sign, as percent of total number of characters</p>
</dd>
<dt>bang</dt><dd><p>Occurrences of &lsquo;!&rsquo;, as percent of total number of characters</p>
</dd>
<dt>money</dt><dd><p>Occurrences of &lsquo;money&rsquo;, as percent of total number of words</p>
</dd>
<dt>n000</dt><dd><p>Occurrences of the string &lsquo;000&rsquo;, as percent of total
number of words</p>
</dd>
<dt>make</dt><dd><p>Occurrences of &lsquo;make&rsquo;, as a percent of total number of words</p>
</dd>
<dt>yesno</dt><dd><p>outcome variable, a factor with levels
<code>n</code> not spam,
<code>y</code> spam</p>
</dd>
</dl>



<h3>Source</h3>

<p>George Forman, Hewlett-Packard Laboratories
</p>
<p>The complete dataset, and documentation, are available from
<a href="https://archive.ics.uci.edu/ml/datasets/spambasehttps://archive.ics.uci.edu/ml/datasets/spambase/">Spam database</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(rpart)
spam.rpart &lt;- rpart(formula = yesno ~ crl.tot + dollar + bang +
   money + n000 + make, data=spam7)
plot(spam.rpart)
text(spam.rpart)

</code></pre>

<hr>
<h2 id='stVincent'>Averages by block of yields for the St. Vincent Corn data</h2><span id='topic+stVincent'></span>

<h3>Description</h3>

<p>These data frames have yield averages by blocks (parcels).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stVincent</code></pre>


<h3>Format</h3>

<p> A data frame with 324 observations on 8 variables.
</p>

<dl>
<dt>code</dt><dd><p>a numeric vector</p>
</dd>
<dt>island</dt><dd><p>a numeric vector</p>
</dd>
<dt>id</dt><dd><p>a numeric vector</p>
</dd>
<dt>site</dt><dd><p>a factor with 8 levels.</p>
</dd>
<dt>block</dt><dd><p>a factor with levels <code>I</code> <code>II</code> <code>III</code> <code>IV</code></p>
</dd>
<dt>plot</dt><dd><p>a numeric vector</p>
</dd>
<dt>trt</dt><dd><p>a factor consisting of 12 levels</p>
</dd>
<dt>harvwt</dt><dd><p>a numeric vector; the average yield</p>
</dd>
</dl>



<h3>Source</h3>

<p>Andrews DF; Herzberg AM, 1985. Data. A Collection of Problems from
Many Fields for the Student and Research Worker. Springer-Verlag.
(pp. 339-353)
</p>

<hr>
<h2 id='sugar'>Sugar Data</h2><span id='topic+sugar'></span>

<h3>Description</h3>

<p>The <code>sugar</code> data frame has 12 rows and 2 columns.
They are from an experiment that
compared an unmodified wild type plant with three different
genetically modified forms.  The measurements are
weights  of sugar that were obtained by breaking down the
cellulose.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sugar</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>weight</dt><dd><p>weight, in mg</p>
</dd>
<dt>trt</dt><dd><p>a factor with levels
<code>Control</code> i.e. unmodified Wild form,  
<code>A</code> Modified 1,
<code>B</code> Modified 2,
<code>C</code> Modified 3</p>
</dd>
</dl>



<h3>Source</h3>

<p>Anonymous
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sugar.aov &lt;- aov(weight ~ trt, data=sugar)
fitted.values(sugar.aov)
summary.lm(sugar.aov)
sugar.aov &lt;- aov(formula = weight ~ trt, data = sugar)
summary.lm(sugar.aov)
</code></pre>

<hr>
<h2 id='sumry'>
A more flexible alternatives to <code>summary</code>.
</h2><span id='topic+sumry'></span>

<h3>Description</h3>

<p>At present this has a method only for <code>glm</code> objects.
The function <code>print.sumry.glm</code> allows greater control over 
what is printed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sumry(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sumry_+3A_object">object</code></td>
<td>

<p>An object for with a summary is required.  At present, this must be a glm object.
</p>
</td></tr>
<tr><td><code id="sumry_+3A_...">...</code></td>
<td>

<p>additional arguments affecting the summary produced.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns summary information.
</p>


<h3>Author(s)</h3>

<p>John Maindonald
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.sumry.glm">print.sumry.glm</a></code>, <code><a href="#topic+sumry">sumry</a></code>, <code><a href="stats.html#topic+glm">glm</a></code>
</p>

<hr>
<h2 id='sumry.glm'>Summarizing Generalized Linear Model Fits</h2><span id='topic+sumry.glm'></span><span id='topic+print.sumry.glm'></span>

<h3>Description</h3>

<p>These functions are <code><a href="utils.html#topic+methods">methods</a></code> for class <code>glm</code> or
<code>sumry.glm</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'glm'
sumry(object, dispersion = NULL, correlation = FALSE,
        symbolic.cor = FALSE, ...)

## S3 method for class 'sumry.glm'
print(x, digits = max(3L, getOption("digits") - 3L), 
                          symbolic.cor = FALSE, 
                          signif.stars = getOption("show.signif.stars"), 
                          call=FALSE, deviance.residuals=FALSE, 
                          show.iter=10, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sumry.glm_+3A_object">object</code></td>
<td>
<p>an object of class <code>"glm"</code>, usually, a result of a
call to <code><a href="stats.html#topic+glm">glm</a></code>.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_x">x</code></td>
<td>
<p>an object of class <code>"summary.glm"</code>, usually, a result of a
call to <code>summary.glm</code>.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_dispersion">dispersion</code></td>
<td>
<p>the dispersion parameter for the family used.
Either a single numerical value or <code>NULL</code> (the default), when
it is inferred from <code>object</code> (see &lsquo;Details&rsquo;).</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_correlation">correlation</code></td>
<td>
<p>logical; if <code>TRUE</code>, the correlation matrix of
the estimated parameters is returned and printed.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_symbolic.cor">symbolic.cor</code></td>
<td>
<p>logical. If <code>TRUE</code>, print the correlations in
a symbolic form (see <code><a href="stats.html#topic+symnum">symnum</a></code>) rather than as numbers.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_signif.stars">signif.stars</code></td>
<td>
<p>logical. If <code>TRUE</code>, &lsquo;significance stars&rsquo;
are printed for each coefficient.</p>
</td></tr>
<tr><td><code id="sumry.glm_+3A_call">call</code></td>
<td>
<p>logical. If <code>TRUE</code>, details of the function call
are printed.</p>
</td></tr>  
<tr><td><code id="sumry.glm_+3A_deviance.residuals">deviance.residuals</code></td>
<td>
<p>logical. If <code>TRUE</code>, deviance residuals
are printed.</p>
</td></tr>      
<tr><td><code id="sumry.glm_+3A_show.iter">show.iter</code></td>
<td>
<p><code>NULL</code> or integer. If <code>NULL</code>, or if
the number of iterations is greater than the specified integer,
then the number of iterations will be printed.</p>
</td></tr>    
<tr><td><code id="sumry.glm_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>print.sumry.glm</code> allows, relative to 
<code>print.summary.glm</code>, some greater flexibility in what is
printed.  By default, details of the call to <code>glm</code> are omitted,
and details of the number of interations only in the unusual case
where this number is greater than 10.  See the help page for 
<code><a href="stats.html#topic+summary.glm">summary.glm</a></code> for further details.
</p>


<h3>Value</h3>

<p><code>sumry.glm</code> returns an object of class <code>"sumry.glm"</code>, a
list with the same components as <code>summary.glm</code>.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+glm">glm</a></code>, <code><a href="base.html#topic+summary">summary</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## For examples see example(glm)
</code></pre>

<hr>
<h2 id='tinting'>Car Window Tinting Experiment Data</h2><span id='topic+tinting'></span>

<h3>Description</h3>

<p>These data are from an experiment that aimed to model the
effects of the tinting of car windows on visual
performance. The authors
were mainly interested in effects on side window vision, and hence in
visual recognition tasks that would be performed when looking through
side windows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tinting</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>case</dt><dd><p>observation number</p>
</dd>
<dt>id</dt><dd><p>subject identifier code (1-26)</p>
</dd>
<dt>age</dt><dd><p>age (in years) </p>
</dd>
<dt>sex</dt><dd><p>a factor with levels
<code>f</code> female,
<code>m</code> male</p>
</dd>
<dt>tint</dt><dd><p>an ordered factor with levels representing degree of
tinting: <code>no</code> &lt; <code>lo</code> &lt; <code>hi</code></p>
</dd>
<dt>target</dt><dd><p>a factor with levels
<code>locon</code>: low contrast,
<code>hicon</code>: high contrast </p>
</dd>
<dt>it</dt><dd><p>the inspection time, the time required
to perform a simple discrimination task (in milliseconds)</p>
</dd>
<dt>csoa</dt><dd><p>critical stimulus onset asynchrony, the time
to recognize an alphanumeric target (in milliseconds)</p>
</dd>
<dt>agegp</dt><dd><p>a factor with levels
<code>younger</code>, 21-27,
<code>older</code>, 70-78 </p>
</dd>
</dl>



<h3>Details</h3>

<p>Visual light transmittance (VLT) levels were 100% (tint=none),
81.3% (tint=lo), and 35.1% (tint=hi).  Based on these and other
data, Burns et al. argue that road safety may be compromised if
the front side windows of cars are tinted to 35
</p>


<h3>Source</h3>

<p>Burns, N.R., Nettlebeck, T., White, M. and Willson, J., 1999.
Effects of car window tinting on visual performance: a comparison of
younger and older drivers. Ergonomics 42: 428-443.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lattice)
levels(tinting$agegp) &lt;- capstring(levels(tinting$agegp))
xyplot(csoa ~ it | sex * agegp, data=tinting) # Simple use of xyplot()
pause()

xyplot(csoa ~ it|sex*agegp, data=tinting, panel=panel.superpose, groups=target)
pause()

xyplot(csoa ~ it|sex*agegp, data=tinting, panel=panel.superpose, col=1:2,
  groups=target, key=list(x=0.14, y=0.84, points=list(pch=rep(1,2),
  col=1:2), text=list(levels(tinting$target), col=1:2), border=TRUE))
## Not run: 
xyplot(csoa ~ it|sex*agegp, data=tinting, panel=panel.superpose,
  groups=tint, type=c("p","smooth"), span=0.8, col=1:3,
  key=list(x=0.14, y=0.84, points=list(pch=rep(1,2), col=1:3),
  text=list(levels(tinting$tint), col=1:3), border=TRUE))
  
## End(Not run)
</code></pre>

<hr>
<h2 id='tomato'>Root weights of tomato plants exposed to 4 different treatments</h2><span id='topic+tomato'></span>

<h3>Description</h3>

<p>The <code>tomato</code> data frame has 24 rows and 2 columns.
They are from an experiment that exposed tomato plants
to four different 'nutrients'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tomato)</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>weight</dt><dd><p>weight, in g</p>
</dd>
<dt>trt</dt><dd><p>a factor with levels
<code>water only</code>,
<code>conc nutrient</code>,
<code>2-4-D + conc nutrient</code>,
<code>3x conc nutrient</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Dr Ron Balham, Victoria University of Wellington NZ, sometime
in 1971 - 1976.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tomato.aov &lt;- aov(log(weight) ~ trt, data=tomato)
fitted.values(tomato.aov)
summary.lm(tomato.aov)
tomato.aov &lt;- aov(formula = weight ~ trt, data = tomato)
summary.lm(tomato.aov)
</code></pre>

<hr>
<h2 id='toycars'>Toy Cars Data</h2><span id='topic+toycars'></span>

<h3>Description</h3>

<p>The <code>toycars</code> data frame has 27 rows and 3 columns.
Observations are on the
distance traveled by one of three different toy cars on
a smooth surface, starting from rest at the top of a 16 inch long ramp
tilted at varying angles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toycars</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>angle</dt><dd><p>tilt of ramp, in degrees</p>
</dd>
<dt>distance</dt><dd><p>distance traveled, in meters</p>
</dd>
<dt>car</dt><dd><p>a numeric code (1 = first car, 2 = second car, 3 = third 
car)</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>toycars.lm &lt;- lm(distance ~ angle + factor(car), data=toycars)
summary(toycars.lm)
</code></pre>

<hr>
<h2 id='two65'>Unpaired Heated Elastic Bands</h2><span id='topic+two65'></span>

<h3>Description</h3>

<p>Twenty-one elastic bands were divided into two groups.
</p>
<p>One of the sets was placed in hot water (60-65 degrees C) for four
minutes, while the other was left at ambient temperature.  After
a wait of about ten minutes, the amounts of stretch, under a 1.35 kg
weight, were recorded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pair65</code></pre>


<h3>Format</h3>

<p>This list contains the following elements:
</p>

<dl>
<dt>heated</dt><dd><p>a numeric vector giving the stretch lengths for
the heated bands</p>
</dd>
<dt>ambient</dt><dd><p>a numeric vector giving the stretch lengths for
the unheated bands</p>
</dd>
</dl>



<h3>Source</h3>

<p>J.H. Maindonald
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twot.permutation(two65$ambient,two65$heated) # two sample permutation test
</code></pre>

<hr>
<h2 id='twot.permutation'>Two Sample Permutation Test - Obsolete</h2><span id='topic+twot.permutation'></span>

<h3>Description</h3>

<p>This function computes the p-value for the two sample
t-test using a permutation test.  The permutation density
can  also be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twot.permutation(x1 = DAAG::two65$ambient, x2 = DAAG::two65$heated, nsim = 2000,
plotit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twot.permutation_+3A_x1">x1</code></td>
<td>
<p>Sample 1</p>
</td></tr>
<tr><td><code id="twot.permutation_+3A_x2">x2</code></td>
<td>
<p>Sample 2</p>
</td></tr>
<tr><td><code id="twot.permutation_+3A_nsim">nsim</code></td>
<td>
<p>Number of simulations</p>
</td></tr>
<tr><td><code id="twot.permutation_+3A_plotit">plotit</code></td>
<td>
<p>If TRUE, the permutation density will be plotted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have n1 values in one group and n2 in a
second, with n = n1 + n2. The permutation distribution
results from taking all possible samples of n2 values from
the total of n values.
</p>


<h3>Value</h3>

<p>The p-value for the test of the hypothesis that the mean of
<code>x1</code> differs from <code>x2</code>
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>References</h3>

 
<p>Good, P. 2000. Permutation Tests. Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twot.permutation()
</code></pre>

<hr>
<h2 id='twotPermutation'>Two Sample Permutation Test</h2><span id='topic+twotPermutation'></span>

<h3>Description</h3>

<p>This function computes the p-value for the two sample
t-test using a permutation test.  The permutation density
can  also be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twotPermutation(x1 = DAAG::two65$ambient, x2 = DAAG::two65$heated, nsim = 2000,
plotit = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twotPermutation_+3A_x1">x1</code></td>
<td>
<p>Sample 1</p>
</td></tr>
<tr><td><code id="twotPermutation_+3A_x2">x2</code></td>
<td>
<p>Sample 2</p>
</td></tr>
<tr><td><code id="twotPermutation_+3A_nsim">nsim</code></td>
<td>
<p>Number of simulations</p>
</td></tr>
<tr><td><code id="twotPermutation_+3A_plotit">plotit</code></td>
<td>
<p>If TRUE, the permutation density will be plotted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose we have n1 values in one group and n2 in a
second, with n = n1 + n2. The permutation distribution
results from taking all possible samples of n2 values from
the total of n values.
</p>


<h3>Value</h3>

<p>The p-value for the test of the hypothesis that the mean of
<code>x1</code> differs from <code>x2</code>
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>References</h3>

 
<p>Good, P. 2000. Permutation Tests. Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twotPermutation()
</code></pre>

<hr>
<h2 id='vif'>Variance Inflation Factors</h2><span id='topic+vif'></span>

<h3>Description</h3>

<p>Variance inflation factors are computed for the standard
errors of linear model coefficient estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vif(obj, digits=5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vif_+3A_obj">obj</code></td>
<td>
<p> A <code>lm</code> object </p>
</td></tr>
<tr><td><code id="vif_+3A_digits">digits</code></td>
<td>
<p> Number of digits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of variance inflation factors corresponding to 
the coefficient estimates given in the <code>lm</code> object.
</p>


<h3>Author(s)</h3>

<p>J.H. Maindonald</p>


<h3>See Also</h3>

<p><code>lm</code></p>


<h3>Examples</h3>

<pre><code class='language-R'>litters.lm &lt;- lm(brainwt ~ bodywt + lsize, data = litters)
vif(litters.lm)

carprice1.lm &lt;- lm(gpm100 ~ Type+Min.Price+Price+Max.Price+Range.Price,
    data=carprice)
vif(carprice1.lm)

carprice.lm &lt;- lm(gpm100 ~ Type + Price, data = carprice)
vif(carprice1.lm)

</code></pre>

<hr>
<h2 id='vince111b'>Averages by block of corn yields, for treatment 111 only</h2><span id='topic+vince111b'></span>

<h3>Description</h3>

<p>These data frames have averages by blocks (parcels) for the treatment
<code>111</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vince111b</code></pre>


<h3>Format</h3>

<p> A data frame with 36 observations on 8 variables.
</p>

<dl>
<dt>site</dt><dd><p>a factor with levels 
<code>AGSV</code> <code>CASV</code> <code>CPSV</code>
<code>LPSV</code> <code>MPSV</code> <code>OOSV</code> <code>OTSV</code> <code>SSSV</code> <code>UISV</code>
</p>
</dd>
<dt>parcel</dt><dd><p>a factor with levels <code>I</code> <code>II</code> <code>III</code> <code>IV</code></p>
</dd>
<dt>code</dt><dd><p>a numeric vector</p>
</dd>
<dt>island</dt><dd><p>a numeric vector</p>
</dd>
<dt>id</dt><dd><p>a numeric vector</p>
</dd>
<dt>plot</dt><dd><p>a numeric vector</p>
</dd>
<dt>trt</dt><dd><p>a numeric vector</p>
</dd>
<dt>harvwt</dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Andrews DF; Herzberg AM, 1985. Data. A Collection of Problems from
Many Fields for the Student and Research Worker. Springer-Verlag.
(pp. 339-353)
</p>

<hr>
<h2 id='vlt'>Video Lottery Terminal Data</h2><span id='topic+vlt'></span>

<h3>Description</h3>

<p>Data on objects appearing in three windows on a video
lottery terminal, together with the prize payout (usually 0).
Observations were taken on two successive days in late 1994
at a hotel lounge north of Winnipeg, Manitoba.  Each observation
cost 25 cents (Canadian).  The game played was &lsquo;Double Diamond&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vlt</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>window1</dt><dd><p>object appearing in the first window.</p>
</dd>
<dt>window2</dt><dd><p>object appearing in the second window.</p>
</dd>
<dt>window3</dt><dd><p>object appearing in the third window.</p>
</dd>
<dt>prize</dt><dd><p>cash prize awarded (in Canadian dollars).</p>
</dd>
<dt>night</dt><dd><p>1, if observation was taken on day 1; 2,
if observation was taken on day 2.</p>
</dd>
</dl>



<h3>Details</h3>

<p>At each play, each of three windows shows one of 7 possible objects.
Apparently, the three windows are independent of each other, and
the objects should appear with equal probability across the three windows.  
The objects are coded as follows: blank (0), single bar (1), double
bar (2), triple bar (3), double diamond (5), cherries (6), and 
the numeral &quot;7&quot; (7).  
</p>
<p>Prizes (in quarters) are awarded according to the following scheme:  
800 (5-5-5), 80 (7-7-7), 40 (3-3-3), 25 (2-2-2), 10 (1-1-1), 10 (6-6-6),
5 (2 &quot;6&quot;'s), 2 (1 &quot;6&quot;) and 5 (any combination of &quot;1&quot;, &quot;2&quot; and &quot;3&quot;).
In addition, a &quot;5&quot; doubles any winning combination, e.g. (5-3-3) pays
80 and (5-3-5) pays 160.
</p>


<h3>Source</h3>

<p>Braun, W. J. (1995) An illustration of bootstrapping using
video lottery terminal data.  Journal of Statistics Education
http://www.amstat.org/publications/jse/v3n2/datasets.braun.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>     vlt.stk &lt;- stack(vlt[,1:3])
     table(vlt.stk)
</code></pre>

<hr>
<h2 id='wages1833'>Wages of Lancashire Cotton Factory Workers in 1833</h2><span id='topic+wages1833'></span>

<h3>Description</h3>

<p>The <code>wages1833</code> data frame gives the wages 
of Lancashire cotton factory workers in 1833. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wages1833</code></pre>


<h3>Format</h3>

<p>This data frame contains the following columns:
</p>

<dl>
<dt>age</dt><dd><p>age in years</p>
</dd>
<dt>mnum</dt><dd><p>number of male workers</p>
</dd>
<dt>mwage</dt><dd><p>average wage of male workers</p>
</dd>
<dt>fnum</dt><dd><p>number of female workers</p>
</dd>
<dt>fwage</dt><dd><p>average wage of female workers</p>
</dd>
</dl>



<h3>Source</h3>

<p>Boot, H.M. 1995. How Skilled Were the Lancashire Cotton Factory
Workers in 1833? Economic History Review 48: 283-303.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(wages1833)
plot(mwage~age,ylim=range(c(mwage,fwage[fwage&gt;0])))
points(fwage[fwage&gt;0]~age[fwage&gt;0],pch=15,col="red")
lines(lowess(age,mwage))
lines(lowess(age[fwage&gt;0],fwage[fwage&gt;0]),col="red")
</code></pre>

<hr>
<h2 id='whoops'>Deaths from whooping cough, in London</h2><span id='topic+whoops'></span>

<h3>Description</h3>

<p>Deaths from whooping cough, in London from 1740 to 1881.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(whoops)</code></pre>


<h3>Format</h3>

<p>This is a multiple time series consisting of 3 series:
<code>wcough</code>, <code>ratio</code>, and <code>alldeaths</code>.
</p>


<h3>Source</h3>

<p>Guy, W. A. 1882. Two hundred and fifty years of small pox in London.
Journal of the Royal Statistical Society 399-443.
</p>


<h3>References</h3>

<p>Lancaster, H. O. 1990. Expectations of Life. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whoops)
str(whoops)
plot(whoops) 
</code></pre>

<hr>
<h2 id='worldRecords'>Record times for track and road races, at August 9th 2006</h2><span id='topic+worldRecords'></span>

<h3>Description</h3>

<p>Record times for track and road races, at August 9th 2006</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(worldRecords)</code></pre>


<h3>Format</h3>

<p>A data frame with 40 observations on the following 9 variables.
</p>

<dl>
<dt><code>Distance</code></dt><dd><p>distance in kilometers</p>
</dd>
<dt><code>roadORtrack</code></dt><dd><p>a factor with levels <code>road</code> <code>track</code></p>
</dd>
<dt><code>Place</code></dt><dd><p>place; a character vector</p>
</dd>
<dt><code>Time</code></dt><dd><p>time in minutes</p>
</dd>
<dt><code>Date</code></dt><dd><p>a Date</p>
</dd>
</dl>



<h3>Details</h3>

<p>For further details, and some additional details, see the web site
that is the source of the data.
</p>


<h3>Source</h3>

<p><a href="http://www.gbrathletics.com/wrec.htm">http://www.gbrathletics.com/wrec.htm</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(worldRecords)
library(lattice)
xyplot(log(Time) ~ log(Distance), groups=roadORtrack, data=worldRecords)
xyplot(log(Time) ~ log(Distance), groups=roadORtrack, data=worldRecords,
       type=c("p","r"))
xyplot(log(Time) ~ log(Distance), groups=roadORtrack, data=worldRecords,
       type=c("p","smooth"))
</code></pre>

<hr>
<h2 id='zzDAAGxdb'>List, each of whose elements hold rows of a file, in character format</h2><span id='topic+zzDAAGxdb'></span>

<h3>Description</h3>

<p>This is the default alternative database for use with the function
<code>datafile</code>, which uses elements of this list to place files
in the working directory. The names of the list elements are
bestTimes and bostonc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(zzDAAGxdb)</code></pre>


<h3>Format</h3>

<p>Successive elements in this list hold character vectors from which the
corresponding files can be readily generated.  
</p>


<h3>Details</h3>

<p>The web site given as the source of the data has additional
information on the bestTimes data. Records are as at August 7 2006.
</p>


<h3>Source</h3>

<p><a href="http://www.gbrathletics.com/wrec.htm">http://www.gbrathletics.com/wrec.htm</a> (bestTimes)
</p>
<p><a href="http://lib.stat.cmu.edu/datasets/">http://lib.stat.cmu.edu/datasets/</a> (bostonc)
</p>


<h3>References</h3>

<p>Harrison, D. and Rubinfeld, D.L. 'Hedonic prices and the demand
for clean air', J. Environ. Economics &amp; Management, vol.5, 81-102,
1978. corrected by Kelley Pace (kpace@unix1.sncc.lsu.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(zzDAAGxdb)
names(zzDAAGxdb)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
