<!DOCTYPE html><html><head><title>Help for package dafs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dafs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abaz.df'><p>DNA from drinking containers</p></a></li>
<li><a href='#abduct.age.df'><p>Victim abduct and the age data</p></a></li>
<li><a href='#anneal.df'><p>Annealing float glass</p></a></li>
<li><a href='#anscombe.df'><p>Anscombe quartet</p></a></li>
<li><a href='#bennett.df'><p>Bennett data</p></a></li>
<li><a href='#bottle.df'><p>Bottle data</p></a></li>
<li><a href='#casework.df'><p>Casework RI data</p></a></li>
<li><a href='#cck196.df'><p>CCK-196G/A data</p></a></li>
<li><a href='#cck45.df'><p>CCK-45C/T data</p></a></li>
<li><a href='#docexam.df'><p>Misleading signatures data</p></a></li>
<li><a href='#dpd.df'><p>Deoxypyridinoline data</p></a></li>
<li><a href='#fiber.color.df'><p>Textile fibres in human hair data</p></a></li>
<li><a href='#fitPvalue'>
<p>Return the P-value from an F-test for a linear model</p></a></li>
<li><a href='#formatScientificTeX'>
<p>Format a number in scientific notation into LaTeX.</p></a></li>
<li><a href='#gc.df'><p>FBI-Gc data</p></a></li>
<li><a href='#ghb.df'><p>Gamma-hydroxybutyric acid data</p></a></li>
<li><a href='#gustafson.df'><p>Teeth data</p></a></li>
<li><a href='#halfnorm'><p>Produce a half normal plot</p></a></li>
<li><a href='#intPlot'><p>Modified Interactions Plot for Two-way Analysis of Variance</p></a></li>
<li><a href='#jauhari1.df'><p>Pellet pattern data 1</p></a></li>
<li><a href='#jauhari2.df'><p>Pellet pattern data 2</p></a></li>
<li><a href='#kent.df'><p>Chest deflection tolerance</p></a></li>
<li><a href='#kerckring.df'><p>Kerckring data</p></a></li>
<li><a href='#last'>
<p>Return the last element of a vector.</p></a></li>
<li><a href='#liver.df'><p>Liver data</p></a></li>
<li><a href='#morphine.df'><p>Morphine concentration in heart and perpipheral blood samples</p></a></li>
<li><a href='#nasal.spline.df'><p>Nasal spine data</p></a></li>
<li><a href='#newton.df'><p>Glass strata data</p></a></li>
<li><a href='#nzglass.df'><p>New Zealand glass RI data</p></a></li>
<li><a href='#pairsDAFS'><p>Modified Pairwise Scatter Plots with Histograms and Correlations</p></a></li>
<li><a href='#palatal.df'><p>Palatal arch shape data</p></a></li>
<li><a href='#plotBoth'><p>Produce postscript and pdf images simulataneously</p></a></li>
<li><a href='#plotBoth.control'><p>Control over plotBoth function</p></a></li>
<li><a href='#plotRegDiagPlots'><p>Plot some standard regression diagnostic plots</p></a></li>
<li><a href='#printANODEVTable'><p>Print an analysis of deviance (ANODEV) table for a GLM</p></a></li>
<li><a href='#printANOVATable'><p>Print a ANOVA table for a Normal GLM</p></a></li>
<li><a href='#printCI'><p>Produce a formatted confidence interval</p></a></li>
<li><a href='#printRegTable'><p>Print a regression table</p></a></li>
<li><a href='#pvalue'><p>Formats an P-value figure for LaTeX</p></a></li>
<li><a href='#ri.calibration.df'><p>Combined calibration data</p></a></li>
<li><a href='#ri.calibration2.df'><p>GRIM2 calibration data</p></a></li>
<li><a href='#Rsq'><p>Formats an R-squared figure for LaTeX</p></a></li>
<li><a href='#salting1.df'><p>Salting out effects in forensic blood alcohol determination</p></a></li>
<li><a href='#salting2.df'><p>Salting out effects in forensic blood alcohol determination</p></a></li>
<li><a href='#sex.age.df'><p>Victims data</p></a></li>
<li><a href='#shadeDens'><p>Add a shaded region to a pdf plot</p></a></li>
<li><a href='#shotgun.df'><p>Shotgun data</p></a></li>
<li><a href='#squamous.df'><p>Occipital squamous bone data</p></a></li>
<li><a href='#StangleBook'>
<p>Stangle a set of files</p></a></li>
<li><a href='#SweaveBook'>
<p>Sweave a set of files</p></a></li>
<li><a href='#tryptase.df'><p>Tryptase data</p></a></li>
<li><a href='#velocity.df'><p>Calibration data</p></a></li>
<li><a href='#vhcdt.df'><p>Vitreous Humour Carbohydrate Deficient Transferrin</p></a></li>
<li><a href='#wong.df'><p>Glass fragments data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0-38</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-11</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Analysis for Forensic Scientists</td>
</tr>
<tr>
<td>Author:</td>
<td>James Curran, Danny Chang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Curran &lt;j.curran@auckland.ac.nz&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>s20x</td>
</tr>
<tr>
<td>Description:</td>
<td>Data and miscellanea to support the book "Introduction to Data analysis with R for Forensic Scientists." This book was written by James Curran and published by CRC Press in 2010 (ISBN: 978-1-4200-8826-7).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-11 05:57:14 UTC; james</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-11 09:12:33 UTC</td>
</tr>
</table>
<hr>
<h2 id='abaz.df'>DNA from drinking containers</h2><span id='topic+abaz.df'></span>

<h3>Description</h3>

<p>The amount of DNA left on different types of drinking containers.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(abaz.df)
</code></pre>


<h3>Format</h3>

<p>A data frame with 21 columns.
</p>

<table>
<tr>
 <td style="text-align: left;">
    [,1] </td><td style="text-align: right;"> person </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> A label (A..F) for the 6
    experimental subjects </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,2] </td><td style="text-align: right;"> sample </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> A treatement factor indicating the
    different beverage/container combinations </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,3] </td><td style="text-align: right;"> ab.sample </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> An abbreviated treatment label
    </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,4] </td><td style="text-align: right;"> time </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> time when DNA concentration was
    measured. Levels: 24hrs, 48hrs </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,5] </td><td style="text-align: right;"> amylase </td><td style="text-align: right;"> numeric </td><td style="text-align: right;"> the relative amount of
    alpha-amylase activity </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,6] </td><td style="text-align: right;"> quant </td><td style="text-align: right;"> numeric </td><td style="text-align: right;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,7] </td><td style="text-align: right;"> amp.volume </td><td style="text-align: right;"> numeric </td><td style="text-align: right;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,8] </td><td style="text-align: right;"> dna.conc </td><td style="text-align: right;"> numeric </td><td style="text-align: right;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,9] </td><td style="text-align: right;"> gel.profile </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
    [,10] </td><td style="text-align: right;"> failed.profile </td><td style="text-align: right;"> factor </td><td style="text-align: right;"> the failure or success of obtaining a usable DNA profile</td>
</tr>
<tr>
 <td style="text-align: left;">
    [,11:21] </td><td style="text-align: right;"> d3..fga </td><td style="text-align: right;"> numeric </td><td style="text-align: right;"> Total peak heights at each
    locus </td>
</tr>
<tr>
 <td style="text-align: left;">
  </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Abaz et al.</p>


<h3>References</h3>

<p>Abaz, J., Walsh, S.J., Curran, J.M., Moss, D.S., Cullen, J., Bright,
J.A., Crowe, G.A., Cockerton, S.L. and Power, T.E. 'Comparison of the
variables affecting the recovery of DNA from common drinking
containers' Forensic Sci Int. 2002 May 23;126(3):233-40.
</p>

<hr>
<h2 id='abduct.age.df'>Victim abduct and the age data</h2><span id='topic+abduct.age.df'></span>

<h3>Description</h3>

<p>This data was studied to investigate whether there was a relationship between whether the victim had been abducted and the age of the victims in certain crimes. The age of the victims had been classiffed as 0-10 and 11+.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(abduct.age.df)</code></pre>


<h3>References</h3>

<p>C. G. G. Aitken,  T. Connolly,  A. Gammerman,  G. Zhang,  D. Bailey, R. Gordon, and R. Oldfield. <em>Statistical modelling in specific case analysis</em>. Science &amp; Justice, 36(4):245-255, October 1996.
</p>

<hr>
<h2 id='anneal.df'>Annealing float glass</h2><span id='topic+anneal.df'></span>

<h3>Description</h3>

<p>Rushton was interested in the effect of annealing on the refractive
index of glass. It is well known that annealing float glass changes
the refractive index (RI). The change in RI - called delta-RI - can
tell a forensic scientist something about the glass that they are
examining.
</p>
<p>In this data set, 3 replicate measurements were made from 150 squares
of glass from a single pane. Each fragment was measured pre and post
annealing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(anneal.df)
</code></pre>


<h3>Format</h3>

<p>A data frame with 900 observations on 3 variables.
</p>

<table>
<tr>
 <td style="text-align: right;">
[,1] </td><td style="text-align: left;"> ri </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> The fragment's refractive index  </td>
</tr>
<tr>
 <td style="text-align: right;">
[,2] </td><td style="text-align: left;"> temp  </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> The fragment's match temperature -
  this will be almost perfectly correlated with ri </td>
</tr>
<tr>
 <td style="text-align: right;">
[,3] </td><td style="text-align: left;"> anneal  </td><td style="text-align: left;"> factor </td><td style="text-align: left;"> either pre or post for pre or post
  annealing </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>K.P. Rushton</p>


<h3>References</h3>

<p>Rushton, K.P., Analysis of the variation of glass refractive index
with respect to annealing, (2009) MSc Thesis, Forensic Science,
University of Auckland.
</p>

<hr>
<h2 id='anscombe.df'>Anscombe quartet</h2><span id='topic+anscombe.df'></span>

<h3>Description</h3>

<p>This data is known as the Anscombe quartet. Each of sets has two variables, x and y. In each data set, x and y have the same mean (9 and 7.5), the same standard deviation (3.32 and 2.03) and the same correlation (0.82). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(anscombe.df)</code></pre>


<h3>References</h3>

<p>F. J. Anscombe. <em>Graphs in statistical analysis</em>. The American Statistician, 27(1):17-21, 1973.
</p>

<hr>
<h2 id='bennett.df'>Bennett data</h2><span id='topic+bennett.df'></span>

<h3>Description</h3>

<p>This data has 10 rows and 49 columns corresponding 10 refractive index (RI) measurements from 49 different locations in a windowpane.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bennett.df)</code></pre>


<h3>References</h3>

<p>R. L. Bennett, N. D. Kim, J. M. Curran, S. A. Coulson, and A. W. N. Newton. <em>Spatial variation of refractive index in a pane of float glass</em>. Science &amp; Justice, 43(2):71-76, April 2003.
</p>

<hr>
<h2 id='bottle.df'>Bottle data</h2><span id='topic+bottle.df'></span>

<h3>Description</h3>

<p>This data contains the elemental concentration of five different elements (Manganese, Barium, Strontium, Zirconium, and Titanium) in samples of glass taken from six different Heineken beer bottles at four different locations (Base, Body, Shoulder, and Neck). Five repeat measurements are made on each sample at each location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bottle.df)</code></pre>


<h3>References</h3>

<p>R. L. Bennett. <em>Aspects of the analysis and interpretation of glass trace evidence</em>. Master's thesis, Department of Chemistry, University of Waikato, 2002.
</p>

<hr>
<h2 id='casework.df'>Casework RI data</h2><span id='topic+casework.df'></span>

<h3>Description</h3>

<p>This data has 10 refractive index (RI) measurement from recovered glass fragments and .....
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(casework.df)</code></pre>


<h3>Author(s)</h3>

<p>James Curran
</p>

<hr>
<h2 id='cck196.df'>CCK-196G/A data</h2><span id='topic+cck196.df'></span>

<h3>Description</h3>

<p>This data has genotype of Cholecystokinin (CCK) gene promoter regions of -196G/A from selected suicide victims (S) and from control subjects (C).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cck196.df)</code></pre>


<h3>References</h3>

<p>Shindo S, Yoshioka N. <em>Polymorphisms of the cholecystokinin gene promoter region in suicide victims in Japan</em>. Forensic Science International, 150(1):85-90, May 2005.
</p>

<hr>
<h2 id='cck45.df'>CCK-45C/T data</h2><span id='topic+cck45.df'></span>

<h3>Description</h3>

<p>This data has genotype of Cholecystokinin (CCK) gene promoter regions of -45C/T from selected suicide victims (S) and from control subjects (C).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cck45.df)</code></pre>


<h3>References</h3>

<p>S. Shindo and N. Yoshioka. <em>Polymorphisms of the cholecystokinin gene promoter region in suicide victims in Japan</em>. Forensic Science International, 150(1):85-90, May 2005.
</p>

<hr>
<h2 id='docexam.df'>Misleading signatures data</h2><span id='topic+docexam.df'></span>

<h3>Description</h3>

<p>This data has number of misleading signatures in comparisons of 16 genuine signatures and 64 simulated signatures from 15 document examiners.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(docexam.df)</code></pre>


<h3>References</h3>

<p>B. Found and D. K. Rogers. <em>Investigating forensic document examiners' skill relating to opinions on photocopied signatures</em>. Science &amp; Justice, 45(4):199-206, 2005.
</p>

<hr>
<h2 id='dpd.df'>Deoxypyridinoline data</h2><span id='topic+dpd.df'></span>

<h3>Description</h3>

<p>This data measured deoxypyridinoline (DPD) to estimate human age. DPD is a nonreducible collagen crosslink that can be measured in human dentin samples extracted from permanent individual molars. Measurements were made in dentin samples from 22 patients with ages ranging from 15 to 73.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dpd.df)</code></pre>


<h3>References</h3>

<p>S. Martin-de las Heras, A. Valenzuela, and E. Villanueva. <em>Deoxypyridinoline crosslinks in human dentin and estimation of age</em>. International Journal of Legal Medicine, 112(4):222-226, June 1999.
</p>

<hr>
<h2 id='fiber.color.df'>Textile fibres in human hair data</h2><span id='topic+fiber.color.df'></span>

<h3>Description</h3>

<p>This data comes from a population study of textile fibres in human hair in Cambridgeshire, UK, 2002. It was carried out using 26 volunteers, and 12149 fibres were recovered from a variety of hair lengths using low adhesive tape and classified according to colour, generic type and fibre length. This data has percentage distribution of fibres in head hair according to colour.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fiber.color.df)</code></pre>


<h3>References</h3>

<p>R. Palmer and S. Oliver. <em>The population of coloured fibres in human head hair</em>. Science &amp; Justice, 44(2):83-88, April 2004.
</p>

<hr>
<h2 id='fitPvalue'>
Return the P-value from an F-test for a linear model
</h2><span id='topic+fitPvalue'></span>

<h3>Description</h3>

<p>This functions the P-value from the (null) hypothesis that all of the
linear predictors are zero or not-significant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitPvalue(fit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitPvalue_+3A_fit">fit</code></td>
<td>

<p>a <code>lm</code> object.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>summary</code> returns an invisible vector called fstatistic that
contains the F-statistic and the degrees of freedom used to test the
hypothesis that all of the linear predictors are zero or
not-significant. This function takes those values and returns the
appropriate upper tail probability from the F-distribution.
</p>


<h3>Value</h3>

<p>A P-value
</p>


<h3>Author(s)</h3>

<p>James Curran
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+summary.lm">summary.lm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = runif(100,1,10)
y = 2 + 3*x + rnorm(100)
fit = lm(y~x)
fitPvalue(fit)
</code></pre>

<hr>
<h2 id='formatScientificTeX'>
Format a number in scientific notation into LaTeX.
</h2><span id='topic+formatScientificTeX'></span><span id='topic+fmtST'></span>

<h3>Description</h3>

<p>Format a number in standard scientific format XXX.XXEXXX into a string
that can be typeset by LaTeX
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatScientificTeX(x, width, digits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatScientificTeX_+3A_x">x</code></td>
<td>

<p>The number to be formatted. This number does not need to be in
scientific format.
</p>
</td></tr>
<tr><td><code id="formatScientificTeX_+3A_width">width</code></td>
<td>

<p><code>width</code> and <code>digits</code> are interpreted in the same way
that they are for
the C function sprintf. The number will be formatted so that it
takes x times 10 to the power of y, where x is formatted so that the
number has <code>width</code> characters in total (including a decimal point) and
<code>digits</code> decimal places. 
</p>
</td></tr>
<tr><td><code id="formatScientificTeX_+3A_digits">digits</code></td>
<td>

<p>See <code>width</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>x = 300.123</code>, <code>width = 4</code> and <code>digits = 1</code>, then this
function will return  &quot; 3.0\\times 10^2&quot;
</p>


<h3>Value</h3>

<p>A string that will format in a LaTeX inline math environment.
</p>


<h3>Author(s)</h3>

<p>James Curran
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+sprintf">sprintf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fmtST(300.123, 1, 4)
</code></pre>

<hr>
<h2 id='gc.df'>FBI-Gc data</h2><span id='topic+gc.df'></span>

<h3>Description</h3>

<p>This data has the human group-specific component (Gc) genotypes of African Americans, Caucasians, and South Western Hispanics. The Gc locus has alleles A, B, and C and hence, the possible genotype are AA, AB, AC, BB, BC, and CC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gc.df)</code></pre>


<h3>References</h3>

<p>FBI reference database
</p>

<hr>
<h2 id='ghb.df'>Gamma-hydroxybutyric acid data</h2><span id='topic+ghb.df'></span>

<h3>Description</h3>

<p>This data comes from a gamma-hydroxybutyric acid (GHB) experiment, which has three groups of individuals:  (1) volunteers with no alcoholic history, (2) alcoholics who were waiting to start a GHB treatment program but had not yet received the drug, (3) alcoholics currently being treated with GHB, and measured the endogenous GHB concentration (ug/mL) in each.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ghb.df)</code></pre>


<h3>References</h3>

<p>F. Mari, L. Politi, C. Trignano, M. Grazia Di Milia, M. Di Padua, and E. Bertoli. <em>What constitutes a normal ante-mortem urine GHB concentration?</em> Journal of Forensic and Legal Medicine, 16(3):148-151, April 2009.
</p>

<hr>
<h2 id='gustafson.df'>Teeth data</h2><span id='topic+gustafson.df'></span>

<h3>Description</h3>

<p>This data is collected from 41 teeth, which were scored on a number of variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gustafson.df)</code></pre>


<h3>References</h3>

<p>G. Gustafson. <em>Age determinations on teeth</em>.  Journal of the American Dental Association, 41(1):45-54, 1950.
</p>

<hr>
<h2 id='halfnorm'>Produce a half normal plot</h2><span id='topic+halfnorm'></span>

<h3>Description</h3>

<p>Produce a half normal plot for a fitted <code>lm</code> or <code>glm</code>
object. This function should work for any class that implements
<code>residuals</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>halfnorm(fit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="halfnorm_+3A_fit">fit</code></td>
<td>

<p>A <code>lm</code> or <code>glm</code> object.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The absolute value of the residuals are plotted against the positive
quantiles of the normal distribution. The largest 5 percent of the
empirical quantiles are labelled to help identify potential outliers.
</p>


<h3>Value</h3>

<p>No values are returned.
</p>


<h3>Note</h3>

<p>The labelling of the largest 5 percent is utterly arbitrary.
</p>


<h3>Author(s)</h3>

<p>J.M. Curran
</p>

<hr>
<h2 id='intPlot'>Modified Interactions Plot for Two-way Analysis of Variance</h2><span id='topic+intPlot'></span><span id='topic+intPlot.default'></span><span id='topic+intPlot.formula'></span>

<h3>Description</h3>

<p>This is a modified version of the function <code>interactionPlots</code>
from the s20x library which produces greyscale plots.
</p>
<p>Displays data with intervals for each
combination of the two factors and shows the mean differences between
levels of the first factor for each level of the second factor. Note
that there should be more than one observation for each combination of
factors.</p>


<h3>Usage</h3>

<pre><code class='language-R'>intPlot(y, ...)
## Default S3 method:
 intPlot(y, 
                           fac1 = NULL, 
                           fac2 = NULL, 
                           xlab = NULL, 
                           xlab2 = NULL, 
                           ylab = NULL, 
                           data.order = TRUE, 
                           exlim = 0.1, 
                           jitter = 0.02, 
                           conf.level = 0.95, 
                           interval.type = "tukey", 
                           pooled = TRUE, 
                           tick.length = 0.1,
                           interval.distance = 0.2,
                           col.width = 2/3, 
                           xlab.distance = 0.1, 
                           xlen = 1.5, 
                           ylen = 1,
                           ...)
## S3 method for class 'formula'
 intPlot(y, 
                           data,
                           xlab = NULL,
                           xlab2 = NULL,
                           ylab = NULL,
                           data.order = TRUE,
                           exlim=0.1,
                           jitter=0.02,
                           conf.level=0.95, 
                           interval.type = "tukey",
                           pooled = TRUE,
                           tick.length = 0.1,
                           interval.distance = 0.2,
                           col.width = 2/3,
                           xlab.distance = 0.1,
                           xlen=1.5,
                           ylen = 1,
                           ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intPlot_+3A_y">y</code></td>
<td>
<p>either a formula of the form: y~fac1+fac2 where y is the
response and fac1 and fac2 are the two explanatory variables used as
factors, or a single response vector</p>
</td></tr>
<tr><td><code id="intPlot_+3A_fac1">fac1</code></td>
<td>
<p>if 'y' is a vector, then fac1 contains the levels of factor
1 which correspond to the y value</p>
</td></tr>
<tr><td><code id="intPlot_+3A_fac2">fac2</code></td>
<td>
<p>if 'y' is a vector, then fac1 contains the levels of factor
2 which correspond to the y value</p>
</td></tr>
<tr><td><code id="intPlot_+3A_data">data</code></td>
<td>
<p>an optional data frame containing the variables in the model.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_xlab">xlab</code></td>
<td>
<p>an optional label for the x-axis. If not specified the
name of fac1 will be used.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_xlab2">xlab2</code></td>
<td>
<p>an optional label for the lines. If not specified the
name of fac2 will be used.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_ylab">ylab</code></td>
<td>
<p>An optional label for the y-axis. If not specified the
name of y will be used.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_data.order">data.order</code></td>
<td>
<p>if TRUE the levels of fac1 and fac2 will be set to
unique(fac1) and unique(fac2) respectively.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_exlim">exlim</code></td>
<td>
<p>provide extra limits.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_jitter">jitter</code></td>
<td>
<p>the amount of horizontal jitter to show in the plot.
The actual jitter is determined as the function is called,
and will likely be different each time the function is used.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_conf.level">conf.level</code></td>
<td>
<p>confidence level of the intervals.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_interval.type">interval.type</code></td>
<td>
<p>four options for intervals appearing on plot:
&quot;tukey&quot;, &quot;hsd&quot;, &quot;lsd&quot; or &quot;ci&quot;.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_pooled">pooled</code></td>
<td>
<p>two options: pooled or unpooled standard deviation used
for plotted intervals.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_tick.length">tick.length</code></td>
<td>
<p>size of tick, in inches.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_interval.distance">interval.distance</code></td>
<td>
<p>distance, as a fraction of the column width,
between the points and interval.  This is in addition to the
extra space allocated for the jitter.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_col.width">col.width</code></td>
<td>
<p>width of a factor &lsquo;column&rsquo;, as a fraction of the
space between the centres of two columns.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_xlab.distance">xlab.distance</code></td>
<td>
<p>distance of x-axis labels from bottom of
plot, as a fraction of the overall height of the plot.</p>
</td></tr>
<tr><td><code id="intPlot_+3A_xlen">xlen</code>, <code id="intPlot_+3A_ylen">ylen</code></td>
<td>
<p>xxx</p>
</td></tr>
<tr><td><code id="intPlot_+3A_...">...</code></td>
<td>
<p>optional arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(s20x)
data(mtcars)
intPlot(wt~vs+gear, mtcars)
</code></pre>

<hr>
<h2 id='jauhari1.df'>Pellet pattern data 1</h2><span id='topic+jauhari1.df'></span>

<h3>Description</h3>

<p>This data comes from a pilot experiment, which fired a sequence of 10 shots through a set of six paper targets that were equally spaced from 3 ft to 18 ft and recorded the size of the pellet pattern at 18 ft.  They then repeated the experiment with five of the targets removed and a single target at 18 ft. The third variable indicate with and without intermediate targets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(jauhari1.df)</code></pre>


<h3>References</h3>

<p>M. Jauhari, S. M. Chatterjee, and P. K. Ghosh. <em>Statistical treatment of pellet dispersion data for estimating range of firing</em>. Journal of Forensic Sciences, 17(1):141-149, 1972.
</p>

<hr>
<h2 id='jauhari2.df'>Pellet pattern data 2</h2><span id='topic+jauhari2.df'></span>

<h3>Description</h3>

<p>This data comes from a pilot experiment by Jauhari et al. They fired a sequence of 10 shots through a set of six paper targets that were equally spaced from 3 ft to 18 ft and recorded the size of the pellet pattern at 18 ft.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(jauhari2.df)</code></pre>


<h3>References</h3>

<p>M. Jauhari, S. M. Chatterjee, and P. K. Ghosh. <em>Statistical treatment of pellet dispersion data for estimating range of firing</em>. Journal of Forensic Sciences, 17(1):141-149, 1972.
</p>

<hr>
<h2 id='kent.df'>Chest deflection tolerance</h2><span id='topic+kent.df'></span>

<h3>Description</h3>

<p>In this data set are the results of 93 human cadaver crash
tests. The tests were used in the development of thoracic injury risk
functions with consideration of age and restraint condition. The data
can be used with logistic regression models by recoding the variable
fracture into &lt;6 and &gt;=6, or with a Poisson/quasi-Poisson/negative
binomial GLM.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kent.df)
</code></pre>


<h3>Value</h3>

<p>A data frame with ... variables
</p>


<h3>Author(s)</h3>

<p>Kent. R, and Petrie, J.</p>


<h3>References</h3>

<p>Kent, R. and Petrie, J., Chest deflection tolerance to blunt anterior
loading is sensitive to age but not load distribution , Forensic
Science International 149(2004):2-3 p.121-128.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kent.df)

##recode the response fracture to minor injury (&lt;6 rib fractures) and
##severe injury (&gt;=6 rib fractures)
##kent.df = within(kent.df, {
##                    injury = factor(ifelse(fractures&lt;6,'minor','severe'),
##                                    levels = c('severe','minor'))})

## fit a binomial GLM

## kent.fit = glm(injury~cmax*
</code></pre>

<hr>
<h2 id='kerckring.df'>Kerckring data</h2><span id='topic+kerckring.df'></span>

<h3>Description</h3>

<p>This data was analyzed the process of Kerckring on the occipital bone of numbers black and white perinates. The process of Kerckring is a projection of bone occasionally observed emerging from the inferior margin of the supraoccipital portion of the occipital squamous at the midline. Its status was regarded as either absent or present.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kerckring.df)</code></pre>


<h3>References</h3>

<p>S. M. Weinberg, D. A. Putz, M. P. Mooney, and M. I. Siegel. <em>Evaluation of non-metric variation in the crania of black and white perinates</em>. Forensic Science International, 151(2-3):177-185, July 2005.
</p>

<hr>
<h2 id='last'>
Return the last element of a vector.
</h2><span id='topic+last'></span>

<h3>Description</h3>

<p>Return the last element of a vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>last(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="last_+3A_x">x</code></td>
<td>

<p>a vector
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>James Curran
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = 1:10
last(x)
</code></pre>

<hr>
<h2 id='liver.df'>Liver data</h2><span id='topic+liver.df'></span>

<h3>Description</h3>

<p>The data record information on the presence or absence of extramedullary haematopoiesis (EMH) symptoms in 51 liver of sudden infant death (SIDS) and 102 non-SIDS cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(liver.df)</code></pre>


<h3>References</h3>

<p>K. Toro, M.Hubay, and E. Keller. <em>Extramedullary haematopoiesis in liver of sudden infant death cases</em>. Forensic Science International, 170(1):15-19, July 2007.
</p>

<hr>
<h2 id='morphine.df'>Morphine concentration in heart and perpipheral blood samples</h2><span id='topic+morphine.df'></span>

<h3>Description</h3>

<p>Data was compiled from 126 morphine-involved cases
investigated by the Office of the Chief Medical Examiner, State
of Maryland, USA. An investigation was conducted into whether
comparison of morphine concentrations from a central and
peripheral site could be used to determine whether a morphine
death was acute or delayed. Fifty cases were identified as 'acute'
because the urine free morphine concentration by
radioimmunoassay (RIA) was less than 25 nglmL; 76 cases were
classified as 'random' because they had a urine morphine
concentration greater than 25 ng/mL by RIA. The average heart
blood to peripheral blood morphine concentration ratio in the
acute deaths was 1.40. The average heart blood to peripheral
blood morphine concentration ratio in the random deaths was
1.18. Because there was considerable overlap between the two
groups of data, the authors conclude that it was not possible to
predict 'acute' opiate intoxication deaths versus 'delayed' deaths
when the only information available is heart and peripheral
blood free morphine concentrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(morphine.df)
</code></pre>


<h3>Author(s)</h3>

<p>Levine et al.</p>


<h3>References</h3>

<p>B. Levine, D. Green-Johnson, K.A. Moore, D. Fowler, A. Jenkins, Assessment of the acuteness of heroin deaths from the analysis of multiple blood specimens, Science &amp; Justice, Volume 42, Issue 1, January 2002, Pages 17-20.
</p>

<hr>
<h2 id='nasal.spline.df'>Nasal spine data</h2><span id='topic+nasal.spline.df'></span>

<h3>Description</h3>

<p>This data was analyzed the projection of the anterior nasal spine on the maxillae of numbers black and white perinates. The anterior nasal spine was assessed by its forward projection away from the frontal plane of the anterior maxillary surface. This was best assessed from the lateral and/or superior perspective. Projection was described as slight, moderate or pronounced.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nasal.spline.df)</code></pre>


<h3>References</h3>

<p>S. M. Weinberg, D. A. Putz, M. P. Mooney, and M. I. Siegel. <em>Evaluation of non-metric variation in the crania of black and white perinates</em>. Forensic Science International, 151(2-3):177-185, July 2005.
</p>

<hr>
<h2 id='newton.df'>Glass strata data</h2><span id='topic+newton.df'></span>

<h3>Description</h3>

<p>This data has refractive index (RI) measurements for 30 fragments on each of the five strata - float surface (FS), near float surface (NFS), bulk (B), near anti-float (NAFS), and anti-float (AFS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(newton.df)</code></pre>


<h3>References</h3>

<p>A. W. N. Newton, J. M. Curran, C. M. Triggs, and J. S. Buckleton. <em>The consequences of potentially differing distributions of the refractive indices of glass fragments from control and recovered sources</em>.  Forensic Science International, 140(2?3):185-193, March 2004.
</p>

<hr>
<h2 id='nzglass.df'>New Zealand glass RI data</h2><span id='topic+nzglass.df'></span>

<h3>Description</h3>

<p>This data has refractive index (RI) measurements made on glass fragments recovered in New Zealand case work.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nzglass.df)</code></pre>


<h3>Author(s)</h3>

<p>John Buckleton, Sally Coulson, Tony Gummer, Angus Newton, Gerhard Wevers, Kevan Walsh (and ESR)
</p>

<hr>
<h2 id='pairsDAFS'>Modified Pairwise Scatter Plots with Histograms and Correlations</h2><span id='topic+pairsDAFS'></span>

<h3>Description</h3>

<p>This is a modified verison of the <code>pairs20x</code> function
from the s20x library which produces greyscale plots.
</p>
<p>Plots pairwise scatter plots with histograms and correlations for the data frame.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairsDAFS(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairsDAFS_+3A_x">x</code></td>
<td>
<p>a data frame.</p>
</td></tr>
<tr><td><code id="pairsDAFS_+3A_...">...</code></td>
<td>
<p>optional argumments which are passed to the generic pairs
function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the plots.</p>


<h3>Examples</h3>

<pre><code class='language-R'>##peruvian indians
library(s20x)
data(peru.df)
pairsDAFS(peru.df)
</code></pre>

<hr>
<h2 id='palatal.df'>Palatal arch shape data</h2><span id='topic+palatal.df'></span>

<h3>Description</h3>

<p>This data was analyzed the shape of the palatal arch on the maxillae of numbers black and white perinates. Palatal arch shape was determined by the curve formed by the inner alveolar margin. In order to determine shape, the separate maxillary halves were joined together at the midline. If the inner alveolar margins were observed to converge gradually along a continuous arc towards the midline, the palate was considered parabolic. On the other hand, if the inner alveolar margins were parallel to each other without converging until anterior of the premolars, the palate was considered hyperbolic. If the palate fell somewhere in between, it was considered intermediate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(palatal.df)</code></pre>


<h3>References</h3>

<p>S. M. Weinberg, D. A. Putz, M. P. Mooney, and M. I. Siegel. <em>Evaluation of non-metric variation in the crania of black and white perinates</em>. Forensic Science International, 151(2-3):177-185, July 2005.
</p>

<hr>
<h2 id='plotBoth'>Produce postscript and pdf images simulataneously</h2><span id='topic+plotBoth'></span>

<h3>Description</h3>

<p>Produce simulataneous postscript and pdf images using a
user defined plot function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBoth(plotfn, filename, control = plotBoth.control(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBoth_+3A_plotfn">plotfn</code></td>
<td>
<p>A function containing the plotting commands</p>
</td></tr>
<tr><td><code id="plotBoth_+3A_filename">filename</code></td>
<td>
<p>The name of the output file - .eps and .pdf will be appended to the postscript and pdf images respectively</p>
</td></tr>
<tr><td><code id="plotBoth_+3A_control">control</code></td>
<td>
<p>The results of <code>plotBoth.control</code> controls font embedding and plotting.</p>
</td></tr>
<tr><td><code id="plotBoth_+3A_...">...</code></td>
<td>
<p>Any additional arguments that need to be fed to plotfn</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J.M. Curran</p>


<h3>See Also</h3>

<p>plotBoth.control
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: plotBoth(function(){
                    plot(rnorm(100),rnorm(100))}, 'test')
## End(Not run)
</code></pre>

<hr>
<h2 id='plotBoth.control'>Control over plotBoth function</h2><span id='topic+plotBoth.control'></span>

<h3>Description</h3>

<p>Gives user control over font embedding and plot generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBoth.control(genPlots = .genPlots, embedF = .embedF, embedFoptions = .embedFoptions)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBoth.control_+3A_genplots">genPlots</code></td>
<td>
<p>TRUE or FALSE depending on whether plotBoth should
produce a plot</p>
</td></tr>
<tr><td><code id="plotBoth.control_+3A_embedf">embedF</code></td>
<td>
<p>TRUE or FALSE depending on whether font embedding is
desired</p>
</td></tr>
<tr><td><code id="plotBoth.control_+3A_embedfoptions">embedFoptions</code></td>
<td>
<p>A string containing all the font embedding
options to be sent to Ghostscript</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the values of the variables genPlots, embedF, and embedFoptions
</p>


<h3>Author(s)</h3>

<p>J Curran</p>

<hr>
<h2 id='plotRegDiagPlots'>Plot some standard regression diagnostic plots</h2><span id='topic+plotRegDiagPlots'></span>

<h3>Description</h3>

<p>Produces 3 plots in a 2 x 2 array. The plots are a plot of
residuals versus predicted values (pred-res plot), a histogram of the
residuals with a normal distribution superimposed, and a normal
QQ-plot of the residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRegDiagPlots(fit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRegDiagPlots_+3A_fit">fit</code></td>
<td>
<p>A <code>lm</code> object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>plot.lm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = runif(100)
y = 3*x+2+dnorm(100)
fit = lm(y~x)
plotRegDiagPlots(fit)
</code></pre>

<hr>
<h2 id='printANODEVTable'>Print an analysis of deviance (ANODEV) table for a GLM</h2><span id='topic+printANODEVTable'></span>

<h3>Description</h3>

<p>Formats the column headings an  table for a GLM using the input of <code>xtable</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printANODEVTable(xtbl,  sanitize.text.function = function(x){x},
                 test = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printANODEVTable_+3A_xtbl">xtbl</code></td>
<td>
<p>The output of <code>xtable</code> applied to the <code>summary</code>
output of a normal GLM.</p>
</td></tr>
<tr><td><code id="printANODEVTable_+3A_sanitize.text.function">sanitize.text.function</code></td>
<td>
<p>Don't change this</p>
</td></tr>
<tr><td><code id="printANODEVTable_+3A_test">test</code></td>
<td>
<p>&quot;Chisq&quot; or &quot;F&quot; depending on the desired test</p>
</td></tr>
<tr><td><code id="printANODEVTable_+3A_...">...</code></td>
<td>
<p>Extra arguments to be fed to <code>print.xtable</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A LaTeX formatted ANODEV table
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>print.xtable
</p>

<hr>
<h2 id='printANOVATable'>Print a ANOVA table for a Normal GLM</h2><span id='topic+printANOVATable'></span>

<h3>Description</h3>

<p>Formats the column headings an ANOVA table for a normal GLM using the input of <code>xtable</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printANOVATable(xtbl,  sanitize.text.function = function(x){x}, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printANOVATable_+3A_xtbl">xtbl</code></td>
<td>
<p>The output of <code>xtable</code> applied to the <code>summary</code>
output of a normal GLM.</p>
</td></tr>
<tr><td><code id="printANOVATable_+3A_sanitize.text.function">sanitize.text.function</code></td>
<td>
<p>Don't change this</p>
</td></tr>
<tr><td><code id="printANOVATable_+3A_...">...</code></td>
<td>
<p>Extra arguments to be fed to <code>print.xtable</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A LaTeX formatted ANOVA table
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>print.xtable
</p>

<hr>
<h2 id='printCI'>Produce a formatted confidence interval</h2><span id='topic+printCI'></span>

<h3>Description</h3>

<p>Takes a vector of length two and prints out a confidence interval in a
user specified format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printCI(x, fmt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printCI_+3A_x">x</code></td>
<td>
<p>A vector of length two containing the lower and upper bounds of a confidence interval</p>
</td></tr>
<tr><td><code id="printCI_+3A_fmt">fmt</code></td>
<td>
<p>A format string to be used by <code>sprintf</code>. Note that
this format string is repeated twice in the code for each bound of
the confidence interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string containing the formatted CI
</p>


<h3>Author(s)</h3>

<p>James Curran</p>


<h3>See Also</h3>

<p>sprintf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 100
x = rnorm(n)
mx = mean(x)
se = sd(x)/sqrt(n)
ci = mx + qnorm(c(0.025,0.975))*se
printCI(ci, '%5.2f')
</code></pre>

<hr>
<h2 id='printRegTable'>Print a regression table</h2><span id='topic+printRegTable'></span>

<h3>Description</h3>

<p>Formats the column headings a regression table using the input of <code>xtable</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printRegTable(xtbl,  sanitize.text.function = function(x){x},
                 test = 't', ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printRegTable_+3A_xtbl">xtbl</code></td>
<td>
<p>The output of <code>xtable</code> applied to the <code>summary</code>
output of a normal GLM.</p>
</td></tr>
<tr><td><code id="printRegTable_+3A_sanitize.text.function">sanitize.text.function</code></td>
<td>
<p>Don't change this</p>
</td></tr>
<tr><td><code id="printRegTable_+3A_test">test</code></td>
<td>
<p>&quot;t&quot; or &quot;z&quot; depending on the desired test</p>
</td></tr>
<tr><td><code id="printRegTable_+3A_...">...</code></td>
<td>
<p>Extra arguments to be fed to <code>print.xtable</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A LaTeX formatted regression table
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>print.xtable
</p>

<hr>
<h2 id='pvalue'>Formats an P-value figure for LaTeX</h2><span id='topic+pvalue'></span>

<h3>Description</h3>

<p>Produces a math-mode formatted string for a P-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalue(p, fmt = "$P = %6.4f$")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalue_+3A_p">p</code></td>
<td>
<p>The P-value</p>
</td></tr>
<tr><td><code id="pvalue_+3A_fmt">fmt</code></td>
<td>
<p>A formatting string for LaTeX</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string that will be correctly formatted by LaTeX
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>Sweave
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = 0.04
pvalue(p)
</code></pre>

<hr>
<h2 id='ri.calibration.df'>Combined calibration data</h2><span id='topic+ri.calibration.df'></span>

<h3>Description</h3>

<p>This data is combined from two calibration experiments which are conduct by Dr. Grzegorz Zadora and Bennett. The factor owner has two levels, RB for Rachel Bennett, and GZ for Grzegorz Zadora. The calibration of the instrument was established using standard glasses (Locke Scientific) set B1-B12 (RI=1.52912-1.520226). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ri.calibration.df)</code></pre>


<h3>References</h3>

<p>Dr. Grzegorz Zadora from the Institute for Forensic Research in Krakow, Poland.
R. L. Bennett. <em>Aspects of the analysis and interpretation of glass trace evidence</em>. Master's thesis, Department of Chemistry, University of Waikato, 2002.
</p>

<hr>
<h2 id='ri.calibration2.df'>GRIM2 calibration data</h2><span id='topic+ri.calibration2.df'></span>

<h3>Description</h3>

<p>This data comes from a 2nd generation Glass Refractive Index Measurer (GRIM2) calibration experiment. The calibration of the instrument was established using standard glasses (Locke Scientific) set B1-B12 (RI=1.52912-1.520226). Each of the twelve reference sample glasses B1-B12 was measured five times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ri.calibration2.df)</code></pre>


<h3>References</h3>

<p>Dr. Grzegorz Zadora from the Institute for Forensic Research in Krakow, Poland.
</p>

<hr>
<h2 id='Rsq'>Formats an R-squared figure for LaTeX</h2><span id='topic+Rsq'></span>

<h3>Description</h3>

<p>Produces a math-mode formatted string for R-squared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rsq(R, fmt = "$R^2 = %6.4f$", adj = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Rsq_+3A_r">R</code></td>
<td>
<p>The R-squared value</p>
</td></tr>
<tr><td><code id="Rsq_+3A_fmt">fmt</code></td>
<td>
<p>A formatting string for LaTeX</p>
</td></tr>
<tr><td><code id="Rsq_+3A_adj">adj</code></td>
<td>
<p>If TRUE appends 'adjusted '  to the format string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string that will be correctly formatted by LaTeX
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>See Also</h3>

<p>Sweave
</p>


<h3>Examples</h3>

<pre><code class='language-R'>r = 0.99
Rsq(r)
</code></pre>

<hr>
<h2 id='salting1.df'>Salting out effects in forensic blood alcohol determination</h2><span id='topic+salting1.df'></span>

<h3>Description</h3>

<p>Blood alcohol measurements determined by headspace gas
chromatography have been challenged on the grounds that the
presence of the preservative sodium fluoride in blood samples
artificially increases headspace alcohol concentrations due to a
salting out effect. Blood samples containing varying amounts of
ethanol and sodium fluoride (NaF) were tested using semi-automated
headspace gas chromatography with n-propyl alcohol as the
internal standard to assess the validity of this challenge. Miller et al found,
in fact, that under these test conditions the measured alcohol
levels are systematically depressed as the amount of sodium
fluoride in the blood sample increases.
</p>
<p>Blood was drawn from each of six subjects near the time of estimated peak blood
alcohol concentration. Each subject had three blood samples taken to
which 0, 5, and 10mg/mL of NaF were added. The blood alcohol
concentration for each tube was determined twice for each tube
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(salting1.df)
</code></pre>


<h3>Format</h3>

<p>A data frame containing four variables
</p>

<table>
<tr>
 <td style="text-align: right;">
  [,1] </td><td style="text-align: left;"> subject </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> subject identifier 1-6 </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,2] </td><td style="text-align: left;"> tube </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> tube 1,2,3 for each subject
  </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,3] </td><td style="text-align: left;"> rep </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> levels 1,2 indicating replicate
  measurement </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,4] </td><td style="text-align: left;"> NaF </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> the level of sodium fluoride
  added in mg/mL </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,5] </td><td style="text-align: left;"> conc </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> alcohol concentration in g/100mL </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Details</h3>

<p>Note that the blocking and treatment factors in this data frame are
numeric. Therefore, to use them as such will require the use of <code>factor</code>
or <code>ordered</code>.
</p>


<h3>Author(s)</h3>

<p>B. A. Miller et al.</p>


<h3>References</h3>

<p>B.A. Miller, S.M. Day, T.E. Vasquez, F.M. Evans, Absence of salting out
effects in forensic blood alcohol determination at various
concentrations of sodium fluoride using semi-automated headspace gas
chromatography, Science &amp; Justice, Volume 44, Issue 2, April 2004, Pages
73-76.
</p>

<hr>
<h2 id='salting2.df'>Salting out effects in forensic blood alcohol determination</h2><span id='topic+salting2.df'></span>

<h3>Description</h3>

<p>Blood alcohol measurements determined by headspace gas
chromatography have been challenged on the grounds that the
presence of the preservative sodium fluoride in blood samples
artificially increases headspace alcohol concentrations due to a
salting out effect. Blood samples containing varying amounts of
ethanol and sodium fluoride (NaF) were tested using semi-automated
headspace gas chromatography with n-propyl alcohol as the
internal standard to assess the validity of this challenge. Miller et al found,
in fact, that under these test conditions the measured alcohol
levels are systematically depressed as the amount of sodium
fluoride in the blood sample increases.
</p>
<p>Blood was drawn from each of four subjects at
two time points, first near the time of estimated peak blood
alcohol concentration and then approximately 1.5 hours later.
Samples were initially analyzed with NaF
at manufacturer's levels (ca. 10 mg/mL).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(salting2.df)
</code></pre>


<h3>Format</h3>

<p>A data frame containing four variables
</p>

<table>
<tr>
 <td style="text-align: right;">
  [,1] </td><td style="text-align: left;"> subject </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> subject identifier 1-4 </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,2] </td><td style="text-align: left;"> time </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> time sample taken 0 or 1.5h </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,3] </td><td style="text-align: left;"> NaF </td><td style="text-align: left;"> numeric factor </td><td style="text-align: left;"> the level of sodium fluoride
  added in mg/mL </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,4] </td><td style="text-align: left;"> EtOH </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> alcohol concentration in g/100mL </td>
</tr>
<tr>
 <td style="text-align: right;">
  </td>
</tr>

</table>



<h3>Details</h3>

<p>Note that the blocking and treatment factors in this data frame are
numeric. Therefore, to use them as such will require the use of <code>factor</code>
or <code>ordered</code>.
</p>


<h3>Author(s)</h3>

<p>B. A. Miller et al.</p>


<h3>References</h3>

<p>B.A. Miller, S.M. Day, T.E. Vasquez, F.M. Evans, Absence of salting out
effects in forensic blood alcohol determination at various
concentrations of sodium fluoride using semi-automated headspace gas
chromatography, Science &amp; Justice, Volume 44, Issue 2, April 2004, Pages
73-76.
</p>

<hr>
<h2 id='sex.age.df'>Victims data</h2><span id='topic+sex.age.df'></span>

<h3>Description</h3>

<p>This data is collected gender and age of victims in certain crimes. The age of the victims had been classified as 0-10 and 11+.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sex.age.df)</code></pre>


<h3>References</h3>

<p>C. G. G. Aitken, T. Connolly, A. Gammerman, G. Zhang, D. Bailey, R. Gordon, and R. Oldfield. <em>Statistical modelling in specific case analysis</em>. Science &amp; Justice, 36(4):245-255, October 1996.
</p>

<hr>
<h2 id='shadeDens'>Add a shaded region to a pdf plot</h2><span id='topic+shadeDens'></span>

<h3>Description</h3>

<p>Useful for shading regions of interest (critical regions perhaps) on a
theoretical pdf to illustrate concepts such as P-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shadeDens(x0, x1, dens, col = "lightgrey", n.points  = 200,
         lty = 1,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shadeDens_+3A_x0">x0</code></td>
<td>
<p>A starting x-value for the region to be shaded</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_x1">x1</code></td>
<td>
<p>An ending x-value for the region to be shaded</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_dens">dens</code></td>
<td>
<p>A function that calculates the pdf</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_col">col</code></td>
<td>
<p>A color for the shaded region</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_n.points">n.points</code></td>
<td>
<p>The number of points to calculate the pdf at over the
interval [x0,x1]</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_lty">lty</code></td>
<td>
<p>Line type</p>
</td></tr>
<tr><td><code id="shadeDens_+3A_...">...</code></td>
<td>
<p>Additional arguments to be fed to <code>dens</code> such as
degrees of freedom, mean or standard deviation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds a filled polygon to an existing pdf plot.
</p>


<h3>Author(s)</h3>

<p>J Curran</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = seq(-4.5,4.5,by = 0.01)
plot(x, dnorm(x), type = 'l')
x0 = qnorm(0.975)
x1 = 4.5
shadeDens(x0, x1, dnorm)
</code></pre>

<hr>
<h2 id='shotgun.df'>Shotgun data</h2><span id='topic+shotgun.df'></span>

<h3>Description</h3>

<p>In order to test the validity of range-of-fire estimates obtained by
the application of regression analysis to shotgun pellet patterns, a
blind study was conducted in which questioned pellet patterns were
fired at randomly selected ranges between 3.0 and 15.2 m (10 and 50
ft) with two different 12-gauge shotguns. each firing a different type
of buckshot cartridge. Test firings at known ranges were also
conducted with the same weapons and ammunition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(shotgun.df)
</code></pre>


<h3>Format</h3>

<p>A data frame with 3 variables:
</p>

<table>
<tr>
 <td style="text-align: right;">
  [,1] </td><td style="text-align: left;"> range </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the range in feet of the firing </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,2] </td><td style="text-align: left;"> sqrt.area </td><td style="text-align: left;"> numeric </td><td style="text-align: left;"> the square root of the area of the smallest rectangle that would enclose the pellet pattern </td>
</tr>
<tr>
 <td style="text-align: right;">
  [,3] </td><td style="text-align: left;"> model </td><td style="text-align: left;"> factor </td><td style="text-align: left;"> the model of shotgun used in the experiment </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>J.M. Curran</p>


<h3>References</h3>

<p>Rowe, W.F. and Hanson, S.R. (1985) Range-of-fire estimates from regression
analysis applied to the spreads of shotgun pellet patterns: Results of
a blind study, Forensic Science International, 28(3-4): 239-250.
</p>

<hr>
<h2 id='squamous.df'>Occipital squamous bone data</h2><span id='topic+squamous.df'></span>

<h3>Description</h3>

<p>This data has been recorded the shape of the occipital squamous bone as narrow, equal, or greater for different races.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(squamous.df)</code></pre>


<h3>References</h3>

<p>S. M. Weinberg, D. A. Putz, M. P. Mooney, and M. I. Siegel. <em>Evaluation of non-metric variation in the crania of black and white perinates</em>. Forensic Science International, 151(2-3):177-185, July 2005.
</p>

<hr>
<h2 id='StangleBook'>
Stangle a set of files
</h2><span id='topic+StangleBook'></span>

<h3>Description</h3>

<p>Stangle more than one file at once. This is a helper function to Stangle all the
chapters in Data Analysis for Forensic Scientists (DAFS) at once. I have put it
into the library for completeness and in the hope that someone else
might find it useful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StangleBook(idx = 0:6,
           fileList = paste(paste("Chapter", idx, "/", sep = ""),
                            paste("ch", idx, ".rnw", sep = ""),
                            sep = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StangleBook_+3A_idx">idx</code></td>
<td>

<p>There seven chapters in Data Analysis for Forensic
Scientists. Corresponding to each chapter is a folder named ChapterI
where I is a number from 0 to 6, and an Sweave file named
chI.rnw. This parameter lets the user select which subset of files
need to be run through Stangle
</p>
</td></tr>
<tr><td><code id="StangleBook_+3A_filelist">fileList</code></td>
<td>

<p>This parameter may be altered to suit the structure of your book. By
default it is set to create a list of files that have the same name
and directory structure as DAFS. The function iterates over a set of
file names specified by this parameter.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No values are returned
</p>


<h3>Author(s)</h3>

<p>James Curran
</p>


<h3>See Also</h3>

<p>Stangle
</p>

<hr>
<h2 id='SweaveBook'>
Sweave a set of files
</h2><span id='topic+SweaveBook'></span>

<h3>Description</h3>

<p>Sweave more than one file at once. This is a helper function to Sweave all the
chapters in Data Analysis for Forensic Scientists (DAFS) at once. I have put it
into the library for completeness and in the hope that someone else
might find it useful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SweaveBook(idx = 0:6,
           fileList = paste(paste("Chapter", idx, "/", sep = ""),
                            paste("ch", idx, ".rnw", sep = ""),
                            sep = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SweaveBook_+3A_idx">idx</code></td>
<td>

<p>There seven chapters in Data Analysis for Forensic
Scientists. Corresponding to each chapter is a folder named ChapterI
where I is a number from 0 to 6, and an Sweave file named
chI.rnw. This parameter lets the user select which subset of files
need to be run through Sweave
</p>
</td></tr>
<tr><td><code id="SweaveBook_+3A_filelist">fileList</code></td>
<td>

<p>This parameter may be altered to suit the structure of your book. By
default it is set to create a list of files that have the same name
and directory structure as DAFS. The function iterates over a set of
file names specified by this parameter.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No values are returned
</p>


<h3>Author(s)</h3>

<p>James Curran
</p>


<h3>See Also</h3>

<p>Sweave
</p>

<hr>
<h2 id='tryptase.df'>Tryptase data</h2><span id='topic+tryptase.df'></span>

<h3>Description</h3>

<p>This data has tryptase concentrations, which was measured in blood from the femoral vein in 60 deaths: 39 control cases who died rapidly (within  minutes) from natural causes (sudden cardiac death and acute aortic dissection), 16 with death caused by prolonged asphyxia (traumatic compression of the chest and suffocation due to body position or smothering), and five anaphylactic deaths. In 44 of these cases, tryptase was measured in both heart (Tryp.cor) and femoral blood (Tryp.fem).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tryptase.df)</code></pre>


<h3>References</h3>

<p>E. Edston, O. Eriksson, M. V. Hage. <em>Mast cell tryptase in postmortem serum - reference values and confounders</em>. International Journal of Legal Medicine, 121(4):275-280, May 2006
</p>

<hr>
<h2 id='velocity.df'>Calibration data</h2><span id='topic+velocity.df'></span>

<h3>Description</h3>

<p>This data is about calibration between gun powder loading and speed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(velocity.df)</code></pre>


<h3>References</h3>

<p>S. C. K. Wong. <em>The effects of projectile properties on glass backscatter: A statistical analysis</em>. Master's thesis, Department of Chemistry, University of Auckland, 2007. Forensic Science.
</p>

<hr>
<h2 id='vhcdt.df'>Vitreous Humour Carbohydrate Deficient Transferrin</h2><span id='topic+vhcdt.df'></span>

<h3>Description</h3>

<p>This data has the carbohydrate-deficient transferrin concentration in vitreous humour (VH-CDT) in 21 alcoholics and 7 non-alcoholics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(vhcdt.df)
</code></pre>


<h3>Format</h3>

<p>The data consist of a data frame with 28 observations on 7 variables.
</p>

<table>
<tr>
 <td style="text-align: right;">
  age alc vhcdt1 vhcdt2 vhtf td th
[,1] </td><td style="text-align: left;"> age </td><td style="text-align: left;"> integer </td><td style="text-align: left;"> age in years </td>
</tr>
<tr>
 <td style="text-align: right;">
[,2] </td><td style="text-align: left;"> alc </td><td style="text-align: left;"> factor </td><td style="text-align: left;"> levels (Y, N) </td>
</tr>
<tr>
 <td style="text-align: right;">
[,3] </td><td style="text-align: left;"> vhcdt1 </td><td style="text-align: left;"> double </td><td style="text-align: left;"> vitreous humour carbohydrate deficient
transferrin concentration (micrograms per liter) - first assay run </td>
</tr>
<tr>
 <td style="text-align: right;">
[,4] </td><td style="text-align: left;"> vhcdt2 </td><td style="text-align: left;"> double </td><td style="text-align: left;"> vitreous humour carbohydrate deficient
transferrin concentration (micrograms per liter) second assay run</td>
</tr>
<tr>
 <td style="text-align: right;">
[,5] </td><td style="text-align: left;"> vhtf </td><td style="text-align: left;"> double </td><td style="text-align: left;"> vitreous humour transferrin (micrograms
per liter)</td>
</tr>
<tr>
 <td style="text-align: right;">
[,6] </td><td style="text-align: left;"> td </td><td style="text-align: left;"> integer </td><td style="text-align: left;"> time interval to autopsy since found dead
</td>
</tr>
<tr>
 <td style="text-align: right;">
[,7] </td><td style="text-align: left;"> td </td><td style="text-align: left;"> integer </td><td style="text-align: left;"> time interval to autopsy since found alive </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>vhcdt1 had a detection limit of 5 micrograms per liter. Observations
below this level were coded as 2.5. Similarly in the second assay a
detection limit of 2.5 micrograms per liter was used. Observations below
this threshold were coded as 1.125.
</p>


<h3>Author(s)</h3>

<p>Berkowicz, A. et al.</p>


<h3>References</h3>

<p>Berkowicz, A., Wallerstedt, S., Wall, K. and Denison, H., Analysis of
carbohydrate-deficient transferrin (CDT) in vitreous humour as a
forensic tool for detection of alcohol misuse, Forensic Science
International 137:(2003) 119-124.
</p>

<hr>
<h2 id='wong.df'>Glass fragments data</h2><span id='topic+wong.df'></span>

<h3>Description</h3>

<p>This data comes from a glass fragments experiment, count the number of glass fragments on the ground that were recovered after a window pane was shot with a handgun. The projectile velocity was controlled by altering the amount of gunpowder added to each bullet. The hardness (as measured on the Rockwell scale of hardness) of each projectile was altered by changing the amount of antimony (Sb) added to the projectile lead during casting. The profile of the projectile was changed by using a round-nose (RN) or wad-cutter (WC) mold. A full factorial design was used to allocate combinations of the factors (velocity, hardness, and profile) to the experimental units (shots). There were four velocity levels, three hardness levels, and two profile levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wong.df)</code></pre>


<h3>References</h3>

<p>S. C. K. Wong. <em>The effects of projectile properties on glass backscatter: A statistical analysis</em>. Master's thesis, Department of Chemistry, University of Auckland, 2007. Forensic Science.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
