<!DOCTYPE html><html><head><title>Help for package DArand</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DArand}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#build_example'><p>Simulation of gene expressions using independant negative binomials</p></a></li>
<li><a href='#DArand'><p>Do Differential Analysis with Random Reference Genes</p></a></li>
<li><a href='#select_prob'><p>Probabilities to select a normalization set without DE-gene</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Differential Analysis with Random Reference Genes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Differential Analysis of short RNA transcripts that can be modeled by either Poisson or Negative binomial distribution.  The statistical methodology implemented in this package is based on the random selection of references genes (Desaulle et al. (2021) &lt;<a href="https://doi.org/10.48550/arXiv.2103.09872">doi:10.48550/arXiv.2103.09872</a>&gt;). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-10 14:19:28 UTC; Dorota</td>
</tr>
<tr>
<td>Author:</td>
<td>Dorota Desaulle <a href="https://orcid.org/0000-0002-3419-9447"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Yves Rozenholc <a href="https://orcid.org/0000-0002-3907-5101"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dorota Desaulle &lt;dorota.desaulle@u-paris.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-10 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='build_example'>Simulation of gene expressions using independant negative binomials</h2><span id='topic+build_example'></span>

<h3>Description</h3>

<p>Simulation of gene expressions using independant negative binomials
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_example(
  m = 500,
  m1,
  n1 = 6,
  n2 = n1,
  fold = 100,
  mu0 = 100,
  use.scales = FALSE,
  nb.size = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_example_+3A_m">m</code></td>
<td>
<p>number of genes</p>
</td></tr>
<tr><td><code id="build_example_+3A_m1">m1</code></td>
<td>
<p>number of differentially expressed genes. In the expression matrix, m1 first columns contain differentially expressed genes.</p>
</td></tr>
<tr><td><code id="build_example_+3A_n1">n1</code></td>
<td>
<p>number of samples under the first condition. The first n1 rows in the expression matrix.</p>
</td></tr>
<tr><td><code id="build_example_+3A_n2">n2</code></td>
<td>
<p>number of samples under the second condition (default n2=n1)</p>
</td></tr>
<tr><td><code id="build_example_+3A_fold">fold</code></td>
<td>
<p>maximal fold change added to the first m1 genes. The fold decreases proportionally to <code>1/sqrt(1:m1)</code>.</p>
</td></tr>
<tr><td><code id="build_example_+3A_mu0">mu0</code></td>
<td>
<p>mean relative expression</p>
</td></tr>
<tr><td><code id="build_example_+3A_use.scales">use.scales</code></td>
<td>
<p>if TRUE random scales are used, otherwise all scales are set to 1.</p>
</td></tr>
<tr><td><code id="build_example_+3A_nb.size">nb.size</code></td>
<td>
<p>number of successful trials in the negative binomial distribution. If nb.size is set to Inf (default), the Poisson model is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function generates a list, of which the first element <code>X</code> is a matrix of <code>n1+n2</code> and m dimension with simulated expressions under Poisson or Negative Binomial distribution. Lines <code>1:n1</code> correspond to the first condition (or sub-group) and lines <code>(n1+1):(n1+n2)</code> to the second one. Columns <code>1:m1</code> contain counts imitating differential expressions.
</p>
<p>In the ideal situation there is no microscopical variability between samples and all scales (so-called scaling factors) would be the same.  To simulate examples corresponding to this perfect situation, use argument <code>use.scales=FALSE</code> which will set all scales to 1.  When <code>use.scales=TRUE</code>, scales are simulated under uniform distribution <em>Unif(0.25,4)</em>.
</p>
<p>The fold is maximal for the first expression and decreases proportionally to <code>1/sqrt(1:m1)</code>. The smallest fold  <code>fold/sqrt(m1)</code> is set to the <em>m1</em>-th expression.
</p>


<h3>Value</h3>

<p>A list with components
</p>

<dl>
<dt><code>X</code></dt><dd><p>a two-dimensional array containing the expression table of n individuals in rows and m gene expressions in columns. </p>
</dd>
<dt><code>m1</code></dt><dd><p>number of differentially expressed genes (as in arguments).</p>
</dd>
<dt><code>n1</code></dt><dd><p>number of samples under the first condition (as in arguments).</p>
</dd>
<dt><code>n2</code></dt><dd><p>number of samples under the second condition (as in arguments).</p>
</dd>
<dt><code>fold</code></dt><dd><p>maximal fold change between the differentally expressed genes and invariant genes (as in arguments). </p>
</dd>
<dt><code>scales</code></dt><dd><p>vector of simulated scales.</p>
</dd>
<dt><code>mu0</code></dt><dd><p>mean relative expression (as in arguments).</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
L = build_example(m=500,m1=25,n1=6,fold=20,mu0=100,use.scales=FALSE,nb.size=Inf)
</code></pre>

<hr>
<h2 id='DArand'>Do Differential Analysis with Random Reference Genes</h2><span id='topic+DArand'></span>

<h3>Description</h3>

<p>Implement the <code>DArand</code> procedure for transcriptomic data.
The procedure is based on random and repeated selection of subsets of reference genes as described in the paper cited below. Observed counts data are normalized with counts from the subset and a differential analysis is used to detect differentially expressed genes. Thought repetitions, the number times a gene is detected is recorded and the final selection is determined from p-values computed under Binomial distribution and adjusted with the Holm's correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DArand(
  X,
  n1,
  k = NULL,
  alpha = 0.05,
  eta = 0.05,
  beta = 0.1,
  r = 1000,
  with.info = FALSE,
  clog = 1,
  use.multi.core = TRUE,
  step = 0,
  scales = NULL,
  use.Iter = TRUE,
  set.seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DArand_+3A_x">X</code></td>
<td>
<p>a two-dimensional array (or data.frame) containing the expression table of n individuals in rows and m gene expressions in columns.</p>
</td></tr>
<tr><td><code id="DArand_+3A_n1">n1</code></td>
<td>
<p>integer, number of individuals of the first category, should be smaller than n</p>
</td></tr>
<tr><td><code id="DArand_+3A_k">k</code></td>
<td>
<p>integer, number of random genes selected (default <code>k = ceiling(log2(m))</code>) as reference genes.</p>
</td></tr>
<tr><td><code id="DArand_+3A_alpha">alpha</code></td>
<td>
<p>numeric, global test level (default 0.05)</p>
</td></tr>
<tr><td><code id="DArand_+3A_eta">eta</code></td>
<td>
<p>numeric, inner test level (default 0.05)</p>
</td></tr>
<tr><td><code id="DArand_+3A_beta">beta</code></td>
<td>
<p>numeric, inner type II error (default 0.1)</p>
</td></tr>
<tr><td><code id="DArand_+3A_r">r</code></td>
<td>
<p>integer, number of random 'reference' set selected (the default 1000)</p>
</td></tr>
<tr><td><code id="DArand_+3A_with.info">with.info</code></td>
<td>
<p>logical, if <code>TRUE</code> results are displayed (the default <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="DArand_+3A_clog">clog</code></td>
<td>
<p>numeric, constant (default 1) controlling the gaussian approximation of the test statistic (in Negative Binomial and Poisson case) .</p>
</td></tr>
<tr><td><code id="DArand_+3A_use.multi.core">use.multi.core</code></td>
<td>
<p>logical, if <code>TRUE</code> (the default) parallel computing with <code>mclapply</code> is used.</p>
</td></tr>
<tr><td><code id="DArand_+3A_step">step</code></td>
<td>
<p>integer, only used when use.Iter is TRUE to get information on the number of iterations (default 0). Not for use.</p>
</td></tr>
<tr><td><code id="DArand_+3A_scales">scales</code></td>
<td>
<p>numeric, only used for simulation of oracle purpose (default <code>NULL</code>). Not for use.</p>
</td></tr>
<tr><td><code id="DArand_+3A_use.iter">use.Iter</code></td>
<td>
<p>logical, applies iterative procedure  (default FALSE)</p>
</td></tr>
<tr><td><code id="DArand_+3A_set.seed">set.seed</code></td>
<td>
<p>numeric, set random seed (as is in <code><a href="base.html#topic+set.seed">set.seed</a></code> function for random number generation ), here default is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The expression table should be organized in such a way that individuals are represented in rows and genes in columns of the <code>X</code> array. Furthermore, in the current version, the procedure provides a differential analysis comparing exactly two experimental conditions. Hence, lines from 1 to <code>n1</code> should correspond to the first condition and the remaining lines to the second condition.
</p>
<p>In the inner part of the procedure, called further <em>randomization</em>, scaling factors are estimated using a normalization subset of <code>k</code> genes randomly selected from  all m genes.  These <code>k</code> genes are used as reference genes.  The normalized data are compared between the experimental conditions within an approximately gaussian test for Poisson  or negative-binomial counts as proposed in the methodology cited below. For this inner test the type I (<code>eta</code>) and the type II (<code>beta</code>) errors should be specified, otherwise the default values will be used.  Since true reference genes (<em>housekeeping genes</em>) are unknown, the inner part is repeated <code>r</code> times.
</p>
<p>Through all <code>r</code> randomization, for each gene, the number of detections (<em>i.e.</em> the number of randomizations when a given gene is identified as differentially expressed) is collected. For these detection counts, the corresponding p-values are computed under the Binomial distribution. The finale detection uses the p-values and,  owing to Holm's correction, controls FWER at specified level <code>alpha</code>.
</p>
<p>The maximal number of discoveries is limited to Delta -  the parameter that is a function of <code>eta</code>, <code>beta</code> and the probability of selecting a subset containing at least one differentially expressed gene leading to a wrong normalization (see <code><a href="#topic+select_prob">select_prob</a></code> ) . If <code>use.Iter</code> is TRUE (the default), the maximal number of discoveries is limited (per iteration) to Delta. The procedure is iterated as long as the number of discoveries is equal to the value of Delta computed in the iteration. Starting from step=1, at each iteration the one-type error is  halved <code>alpha=alpha/2</code> to ensure the overall test level respects the initial alpha.
</p>
<p><code>clog</code> is a constant that controls gaussian approximation of the test statistic for the count data arising from Negative Binomial or Poisson distribution. The constant should be ajusted to keep the probability <code>1-5*n^(-clog)</code> high while shift term <code>1+sqrt(clog*n)</code> low.
</p>


<h3>Value</h3>

<p>position vector of the gene expressions found as differentially expressed.
</p>


<h3>Author(s)</h3>

<p>D. Desaulle and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Differential analysis in Transcriptomic: The strengh of randomly picking 'reference' genes</em>, D. Desaulle, C. Hoffman, B. Hainque and Y. Rozenholc.
<a href="https://arxiv.org/abs/2103.09872">https://arxiv.org/abs/2103.09872</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
L = build_example(m=500,m1=25,n1=6,fold=20,mu0=100,use.scales=FALSE,nb.size=Inf)
DArand(L$X,L$n1,alpha=0.05)

</code></pre>

<hr>
<h2 id='select_prob'>Probabilities to select a normalization set without DE-gene</h2><span id='topic+select_prob'></span>

<h3>Description</h3>

<p>Probabilities to select a normalization set without DE-gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_prob(m, k, invariant = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_prob_+3A_m">m</code></td>
<td>
<p>number of genes</p>
</td></tr>
<tr><td><code id="select_prob_+3A_k">k</code></td>
<td>
<p>normalization subset size</p>
</td></tr>
<tr><td><code id="select_prob_+3A_invariant">invariant</code></td>
<td>
<p>boolean, when TRUE, probability of selection is evaluated for invariant gene</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of probabilities of having at least one differential expression used as an reference selected in the normalization subset for any number of differential expressions d in the gene collection.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
select_prob(500, 10, invariant=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
