<!DOCTYPE html><html><head><title>Help for package dartR.base</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR.base}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[,dartR,ANY,ANY,ANY-method'><p>indexing dartR objects correctly...</p></a></li>
<li><a href='#bandicoot.gl'><p>A genlight object created via the read.dart functions</p>
This a test data set to test the validity of functions within dartR and is
based on a DArT SNP data set of simulated bandicoots across Australia. It
contains 96 individuals and 1000 SNPs.</a></li>
<li><a href='#cbind.dartR'><p>adjust cbind for dartR</p></a></li>
<li><a href='#gl.alf'><p>Calculates allele frequency of the first and second allele for each locus</p>
A very simple function to report allele frequencies</a></li>
<li><a href='#gl.allele.freq'><p>Generates percentage allele frequencies by locus and population</p></a></li>
<li><a href='#gl.amova'><p>Performs AMOVA using genlight data</p></a></li>
<li><a href='#gl.check.verbosity'><p>Checks the current global verbosity</p></a></li>
<li><a href='#gl.check.wd'><p>Checks the global working directory</p></a></li>
<li><a href='#gl.colors'><p>Returns a list of colors for use in plots</p></a></li>
<li><a href='#gl.compliance.check'><p>Checks a genlight object to see if it complies with dartR</p>
expectations and amends it to comply if necessary
@family environment</a></li>
<li><a href='#gl.define.pop'><p>Defines a new population in a genlight object for specified</p>
individuals</a></li>
<li><a href='#gl.diagnostics.hwe'><p>Provides descriptive stats and plots to diagnose potential problems</p>
with Hardy-Weinberg proportions
@family matched report</a></li>
<li><a href='#gl.dist.ind'><p>Calculates a distance matrix for individuals defined in a genlight object</p></a></li>
<li><a href='#gl.dist.pop'><p>Calculates a distance matrix for populations with SNP genotypes in a</p>
genlight object</a></li>
<li><a href='#gl.drop.ind'><p>Removes specified individuals from a dartR genlight object</p></a></li>
<li><a href='#gl.drop.loc'><p>Removes specified loci from a dartR genlight object</p></a></li>
<li><a href='#gl.drop.pop'><p>Removes specified populations from a dartR genlight object</p></a></li>
<li><a href='#gl.edit.recode.ind'><p>Creates or edits individual (=specimen) names, creates a recode_ind</p>
file and applies the changes to a genlight object
@family data manipulation</a></li>
<li><a href='#gl.edit.recode.pop'><p>Creates or edits and applies a population re-assignment table</p></a></li>
<li><a href='#gl.fdsim'><p>Estimates the rate of false positives in a fixed difference analysis</p></a></li>
<li><a href='#gl.filter.allna'><p>Filters loci that are all NA across individuals and/or populations</p>
with all NA across loci</a></li>
<li><a href='#gl.filter.callrate'><p>Filters loci or specimens in a genlight {adegenet} object based on</p>
call rate</a></li>
<li><a href='#gl.filter.factorloadings'><p>Filters loci based on factor loadings for a PCA or PCoA</p></a></li>
<li><a href='#gl.filter.hamming'><p>Filters loci based on pairwise Hamming distance between sequence tags</p></a></li>
<li><a href='#gl.filter.heterozygosity'><p>Filters individuals with average heterozygosity greater than a</p>
specified upper threshold or less than a specified lower threshold
@family matched filter</a></li>
<li><a href='#gl.filter.hwe'><p>Filters loci that show significant departure from Hardy-Weinberg</p>
Equilibrium
@family matched filter</a></li>
<li><a href='#gl.filter.ld'><p>Filters loci based on linkage disequilibrium (LD)</p></a></li>
<li><a href='#gl.filter.locmetric'><p>Filters loci on the basis of numeric information stored in</p>
other$loc.metrics in a genlight {adegenet} object</a></li>
<li><a href='#gl.filter.maf'><p>Filters loci on the basis of minor allele frequency (MAF) in a</p>
genlight object</a></li>
<li><a href='#gl.filter.monomorphs'><p>Filters monomorphic loci, including those with all NAs</p></a></li>
<li><a href='#gl.filter.overshoot'><p>Filters loci for which the SNP has been trimmed from the sequence tag</p>
along with the adaptor</a></li>
<li><a href='#gl.filter.pa'><p>Filters loci that contain private (and fixed alleles) between two populations</p></a></li>
<li><a href='#gl.filter.rdepth'><p>Filters loci based on counts of sequence tags scored at a locus (read</p>
depth)
@family matched filter</a></li>
<li><a href='#gl.filter.reproducibility'><p>Filters loci in a genlight {adegenet} object based on average</p>
repeatability of alleles at a locus
@family matched filter</a></li>
<li><a href='#gl.filter.secondaries'><p>Filters loci that represent secondary SNPs in a genlight object</p></a></li>
<li><a href='#gl.filter.taglength'><p>Filters loci in a genlight {adegenet} object based on sequence tag</p>
length
@family matched filter</a></li>
<li><a href='#gl.fixed.diff'><p>Generates a matrix of fixed differences and associated statistics for</p>
populations taken pairwise</a></li>
<li><a href='#gl.fst.pop'><p>Calculates a pairwise Fst values for populations in a genlight object</p>
This script calculates pairwise Fst values based on the implementation in the
StAMPP package (?stamppFst). It allows to run bootstrap to estimate
probability of Fst values to be different from zero. For detailed information
please check the help pages (?stamppFst).</a></li>
<li><a href='#gl.He'><p>Estimates expected Heterozygosity</p></a></li>
<li><a href='#gl.Ho'><p>Estimates observed Heterozygosity</p></a></li>
<li><a href='#gl.hwe.pop'><p>Performs Hardy-Weinberg tests over loci and populations</p></a></li>
<li><a href='#gl.impute'><p>Imputes missing data</p></a></li>
<li><a href='#gl.join'><p>Combines two dartR genlight objects</p></a></li>
<li><a href='#gl.keep.ind'><p>Removes all but the specified individuals from a dartR genlight object</p></a></li>
<li><a href='#gl.keep.loc'><p>Removes all but the specified loci from a genlight object</p></a></li>
<li><a href='#gl.keep.pop'><p>Removes all but the specified populations from a dartR genlight object</p></a></li>
<li><a href='#gl.load'><p>Loads an object from compressed binary format produced by gl.save()</p></a></li>
<li><a href='#gl.mahal.assign'><p>Assigns individuals to populations with an associated probability</p></a></li>
<li><a href='#gl.make.recode.ind'><p>Creates a proforma recode_ind file for reassigning individual</p>
(=specimen) names</a></li>
<li><a href='#gl.make.recode.pop'><p>Creates a proforma recode_pop_table file for reassigning population</p>
names
@family data manipulation</a></li>
<li><a href='#gl.map.interactive'><p>Creates an interactive map (based on latlon) from a genlight object</p></a></li>
<li><a href='#gl.merge.pop'><p>Merges two or more populations in a dartR genlight object into one population</p></a></li>
<li><a href='#gl.pcoa'><p>Ordination applied to genotypes in a genlight object (PCA), in an fd</p>
object, or to a distance matrix (PCoA)</a></li>
<li><a href='#gl.pcoa.plot'><p>Bivariate or trivariate plot of the results of an ordination generated</p>
using gl.pcoa()</a></li>
<li><a href='#gl.plot.heatmap'><p>Represents a distance matrix as a heatmap</p></a></li>
<li><a href='#gl.print.history'><p>Prints history of a genlight object</p></a></li>
<li><a href='#gl.prop.shared'><p>Calculates a similarity (distance) matrix for individuals on the proportion of</p>
shared alleles
@family distance</a></li>
<li><a href='#gl.randomize.snps'><p>Randomly changes the allocation of 0's and 2's in a genlight object</p></a></li>
<li><a href='#gl.read.csv'><p>Reads SNP data from a csv file into a genlight object</p></a></li>
<li><a href='#gl.read.dart'><p>Imports DArT data into dartR and converts it into a dartR genlight object</p></a></li>
<li><a href='#gl.read.fasta'><p>Reads FASTA files and converts them to genlight object</p></a></li>
<li><a href='#gl.read.PLINK'><p>Reads PLINK data file into a genlight object</p></a></li>
<li><a href='#gl.read.silicodart'><p>Imports presence/absence data from SilicoDArT to genlight {agegenet}</p>
format (ploidy=1)</a></li>
<li><a href='#gl.read.vcf'><p>Converts a vcf file into a genlight object</p></a></li>
<li><a href='#gl.reassign.pop'><p>Assigns an individual metric as pop in a genlight {adegenet} object</p></a></li>
<li><a href='#gl.recalc.metrics'><p>Recalculates locus metrics when individuals or populations are deleted from a</p>
genlight {adegenet} object
@family environment</a></li>
<li><a href='#gl.recode.ind'><p>Recodes individual (=specimen = sample) labels in a genlight object</p></a></li>
<li><a href='#gl.recode.pop'><p>Recodes population assignments in a genlight object</p></a></li>
<li><a href='#gl.rename.pop'><p>Renames a population in a genlight object</p></a></li>
<li><a href='#gl.report.bases'><p>Reports summary of base pair frequencies</p></a></li>
<li><a href='#gl.report.basics'><p>Basic statistics for a genlight object</p></a></li>
<li><a href='#gl.report.callrate'><p>Reports summary of Call Rate for loci or individuals</p></a></li>
<li><a href='#gl.report.diversity'><p>Calculates diversity indexes for SNPs</p></a></li>
<li><a href='#gl.report.factorloadings'><p>Reports factor loadings for a PCA or PCoA</p></a></li>
<li><a href='#gl.report.fstat'><p>Reports various statistics of genetic differentiation between</p>
populations with confident intervals</a></li>
<li><a href='#gl.report.hamming'><p>Calculates the pairwise Hamming distance between DArT trimmed DNA</p>
sequences</a></li>
<li><a href='#gl.report.heterozygosity'><p>Reports observed, expected and unbiased heterozygosities and FIS</p>
(inbreeding coefficient) by population or by individual from SNP data</a></li>
<li><a href='#gl.report.hwe'><p>Reports departure from Hardy-Weinberg proportions</p></a></li>
<li><a href='#gl.report.ld'><p>Calculates pairwise population based Linkage Disequilibrium across all loci</p>
using the specified number of cores
@family matched report</a></li>
<li><a href='#gl.report.ld.map'><p>Calculates pairwise linkage disequilibrium by population</p></a></li>
<li><a href='#gl.report.locmetric'><p>Reports summary of the slot $other$loc.metrics</p></a></li>
<li><a href='#gl.report.maf'><p>Reports minor allele frequency (MAF) for each locus in a SNP dataset</p></a></li>
<li><a href='#gl.report.monomorphs'><p>Reports monomorphic loci</p></a></li>
<li><a href='#gl.report.overshoot'><p>Reports loci for which the SNP has been trimmed from the sequence tag</p>
along with the adaptor</a></li>
<li><a href='#gl.report.pa'><p>Reports private alleles (and fixed alleles) per pair of populations</p></a></li>
<li><a href='#gl.report.rdepth'><p>Reports summary of Read Depth for each locus</p></a></li>
<li><a href='#gl.report.reproducibility'><p>Reports summary of RepAvg (repeatability averaged over both alleles</p>
for
each locus) or reproducibility (repeatability of the scores for fragment
presence/absence)</a></li>
<li><a href='#gl.report.secondaries'><p>Reports loci containing secondary SNPs in sequence tags and calculates</p>
number of invariant sites</a></li>
<li><a href='#gl.report.taglength'><p>Reports summary of sequence tag length across loci</p></a></li>
<li><a href='#gl.sample'><p>Samples individuals from populations</p></a></li>
<li><a href='#gl.save'><p>Saves an object in compressed binary format for later rapid retrieval</p></a></li>
<li><a href='#gl.select.colors'><p>Selects colors from one of several palettes and outputs as a vector</p></a></li>
<li><a href='#gl.select.shapes'><p>Selects shapes from the base R shape palette and outputs as a vector</p></a></li>
<li><a href='#gl.set.verbosity'><p>Sets the default verbosity level</p></a></li>
<li><a href='#gl.set.wd'><p>Sets the default working directory</p></a></li>
<li><a href='#gl.sim.cross'><p>Generates random crosses between fathers and mothers</p></a></li>
<li><a href='#gl.sim.genotypes'><p>Generate random genotypes</p></a></li>
<li><a href='#gl.smearplot'><p>Smear plot</p></a></li>
<li><a href='#gl.sort'><p>re-sorts genlight objects</p></a></li>
<li><a href='#gl.subsample.ind'><p>Subsample individuals from a genlight object</p></a></li>
<li><a href='#gl.subsample.loc'><p>Subsample loci from a genlight object</p></a></li>
<li><a href='#gl.test.heterozygosity'><p>Tests the difference in heterozygosity between populations taken</p>
pairwise</a></li>
<li><a href='#gl.tree.nj'><p>Outputs an nj tree to summarize genetic similarity among populations</p></a></li>
<li><a href='#gl.write.csv'><p>Writes out data from a genlight object to csv file</p></a></li>
<li><a href='#gl2bayesAss'><p>Converts a genlight object into bayesAss (BA3) input format</p></a></li>
<li><a href='#gl2bayescan'><p>Converts a genlight object into a format suitable for input to Bayescan</p></a></li>
<li><a href='#gl2bpp'><p>Converts a genlight object into a format suitable for input to the BPP</p>
program</a></li>
<li><a href='#gl2demerelate'><p>Creates a dataframe suitable for input to package {Demerelate} from a</p>
genlight {adegenet} object</a></li>
<li><a href='#gl2eigenstrat'><p>Converts a genlight object into eigenstrat format</p></a></li>
<li><a href='#gl2fasta'><p>Concatenates DArT trimmed sequences and outputs a FASTA file</p></a></li>
<li><a href='#gl2faststructure'><p>Converts a genlight object into faststructure format (to run faststructure</p>
elsewhere)</a></li>
<li><a href='#gl2gds'><p>Converts a genlight object into gds format</p></a></li>
<li><a href='#gl2genalex'><p>Converts a genlight object into a format suitable for input to genalex</p></a></li>
<li><a href='#gl2genepop'><p>Converts a genlight object into genepop format (and file)</p></a></li>
<li><a href='#gl2geno'><p>Converts a genlight object to geno format from package LEA</p></a></li>
<li><a href='#gl2gi'><p>Converts a genind object into a genlight object</p></a></li>
<li><a href='#gl2hiphop'><p>Converts a genlight objects into hiphop format</p></a></li>
<li><a href='#gl2phylip'><p>Creates a Phylip input distance matrix from a genlight (SNP) {adegenet}</p>
object</a></li>
<li><a href='#gl2plink'><p>Converts a genlight object into PLINK format</p></a></li>
<li><a href='#gl2related'><p>Converts a genlight object to format suitable to be run with Coancestry</p></a></li>
<li><a href='#gl2sa'><p>Converts genlight objects to the format used in the SNPassoc package</p></a></li>
<li><a href='#gl2snapp'><p>Converts a genlight object to nexus format suitable for phylogenetic analysis</p>
by SNAPP (via BEAUti)
@family linker</a></li>
<li><a href='#gl2structure'><p>Converts a genlight object to STRUCTURE formatted files</p></a></li>
<li><a href='#gl2svdquartets'><p>Converts a genlight object to nexus format PAUP SVDquartets</p></a></li>
<li><a href='#gl2treemix'><p>Converts a genlight object to a treemix input file</p></a></li>
<li><a href='#gl2vcf'><p>Converts a genlight object into vcf format</p></a></li>
<li><a href='#possums.gl'><p>A simulated genlight object created to run a landscape genetic example</p>
This a test data set to run a landscape genetics example. It contains 10
populations of 30 individuals each and each individual has 300 loci. There
are no covariates for individuals or loci.</a></li>
<li><a href='#rbind.dartR'><p>adjust rbind for dartR</p></a></li>
<li><a href='#testset.gl'><p>A genlight object created via the gl.read.dart function</p>
This is a test data set on turtles. 250 individuals, 255 loci in &gt;30
populations.</a></li>
<li><a href='#testset.gs'><p>A genlight object created via the gl.read.silicodart function</p>
This is a test data set on turtles. 218 individuals, 255 loci in &gt;30
populations.</a></li>
<li><a href='#theme_dartR'><p>Default theme for dartR plots</p></a></li>
<li><a href='#utils.basic.stats'><p>Calculates mean observed heterozygosity, mean expected heterozygosity</p>
and FIS per locus, per population and various population differentiation
measures
@family utilities</a></li>
<li><a href='#utils.check.datatype'><p>Utility function to check the class of an object passed to a function</p></a></li>
<li><a href='#utils.dart2genlight'><p>An internal function to converts DarT to genlight.</p></a></li>
<li><a href='#utils.dist.binary'><p>Calculates a distance matrix for individuals defined in a dartR</p>
genlight object using binary P/A data (SilicoDArT)</a></li>
<li><a href='#utils.dist.ind.snp'><p>Calculates a distance matrix for individuals defined in a</p>
genlight object using SNP data (DArTseq)</a></li>
<li><a href='#utils.flag.start'><p>A utility script to flag the start of a script</p></a></li>
<li><a href='#utils.hamming'><p>Calculates the Hamming distance between two DArT trimmed DNA sequences</p></a></li>
<li><a href='#utils.het.pop'><p>An internal function that calculates expected mean heterozygosity per population</p></a></li>
<li><a href='#utils.impute'><p>An internal script [Custodian to provide a title]</p></a></li>
<li><a href='#utils.is.fixed'><p>An internal function to tests if two populations are fixed at a given locus</p></a></li>
<li><a href='#utils.jackknife'><p>An internal function to conducts jackknife resampling using a genlight object</p></a></li>
<li><a href='#utils.n.var.invariant'><p>An internal utility function to calculate the number of variant and invariant sites by</p>
locus</a></li>
<li><a href='#utils.plink.run'><p>Runs PLINK from within R</p></a></li>
<li><a href='#utils.plot.save'><p>An internal function to save a ggplot object to disk in RDS binary format</p></a></li>
<li><a href='#utils.read.dart'><p>Utility to import DarT data to R</p></a></li>
<li><a href='#utils.read.fasta'><p>An internal script to read a fastA file into a genlight object</p></a></li>
<li><a href='#utils.read.ped'><p>An internal script [Custodian to provide a title]</p></a></li>
<li><a href='#utils.recalc.avgpic'><p>A utility function to recalculate intermediate locus metrics</p></a></li>
<li><a href='#utils.recalc.callrate'><p>A utility script to recalculate the callrate by locus after some populations</p>
have been deleted</a></li>
<li><a href='#utils.recalc.freqhets'><p>A utility script to recalculate the frequency of the heterozygous SNPs by</p>
locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.freqhomref'><p>#' An internal utility function to recalculate the frequency of the homozygous reference</p>
SNP by locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.freqhomsnp'><p>A utility function to recalculate the frequency of the homozygous alternate</p>
SNP by locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.maf'><p>A utility function to recalculate the minor allele frequency by locus,</p>
typically after some populations have been deleted</a></li>
<li><a href='#utils.reset.flags'><p>#' An internal utility function to reset to FALSE (or TRUE) the locus metric flags after</p>
some individuals or populations have been deleted.</a></li>
<li><a href='#utils.transpose'><p>An internal utility function to transpose a genlight object.</p></a></li>
<li><a href='#zzz'><p>Setting up the package</p>
Setting theme, colors and verbosity</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysing 'SNP' and 'Silicodart' Data - Basic Functions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.65</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-16</td>
</tr>
<tr>
<td>Description:</td>
<td>Facilitates the import and analysis of 'SNP' (single nucleotide 'polymorphism') 
    and 'silicodart' (presence/absence) data. The main focus is on data generated by 'DarT' 
    (Diversity Arrays Technology), however, data from other sequencing platforms can be used 
    once 'SNP' or related fragment presence/absence data from any source is imported. Genetic 
    datasets are stored in a derived 'genlight' format (package 'adegenet'), that allows for 
    a very compact storage of data and metadata. Functions are available for 
    importing and exporting of 'SNP' and 'silicodart' data, for reporting on and 
    filtering on various criteria (e.g. 'callrate', 'heterozygosity', 'reproducibility', 
    maximum allele frequency). Additional functions are available for visualization 
    (e.g. Principle Coordinate Analysis) and creating a spatial representation 
    using maps. 'dartR.base' is the 'base' package of the 'dartRverse' suits of packages. 
    To install the other packages, we recommend to install the 'dartRverse' package, that 
    supports the installation of all packages in the 'dartRverse'.
    If you want to cite 'dartR', you find the information by typing citation('dartR.base')
    in the console.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), adegenet (&ge; 2.0.0), ggplot2, dplyr, dartR.data</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape,crayon,data.table,foreach,gridExtra,methods,patchwork,plyr,reshape2,SNPRelate,StAMPP,stats,stringr,tidyr,utils,MASS</td>
</tr>
<tr>
<td>Suggests:</td>
<td>boot, devtools, directlabels, dismo, doParallel, expm,
gdistance, gganimate, ggrepel, grid, gtable, ggthemes, gplots,
HardyWeinberg, hierfstat, igraph, iterpc, knitr,
label.switching, lattice, leaflet, leaflet.minicharts,
markdown, mmod, networkD3, parallel, pegas, pheatmap, plotly,
poppr, proxy, purrr, qvalue, RColorBrewer, Rcpp, rgl,
rmarkdown, rrBLUP, scales, seqinr, sf, shinyBS, shinyjs,
shinythemes, shinyWidgets, SIBER, snpStats, stringi, terra,
tibble, vcfR, zoo, viridis, fields, testthat (&ge; 3.0.0), ggtern</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 12:52:28 UTC; bernd</td>
</tr>
<tr>
<td>Author:</td>
<td>Bernd Gruber [aut, cre],
  Arthur Georges [aut],
  Jose L. Mijangos [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb],
  Lindsay V. Clark [ctb],
  Floriaan Devloo-Delva [ctb],
  Eric Archer [ctb]</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/dartr?pli=1">https://groups.google.com/g/dartr?pli=1</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bernd Gruber &lt;bernd.gruber@canberra.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 20:00:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B+2CdartR+2CANY+2CANY+2CANY-method'>indexing dartR objects correctly...</h2><span id='topic++5B+2CdartR+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>indexing dartR objects correctly...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'dartR,ANY,ANY,ANY'
x[i, j, ..., pop = NULL, treatOther = TRUE, quiet = TRUE, drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>dartR object</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>index for individuals</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_j">j</code></td>
<td>
<p>index for loci</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_...">...</code></td>
<td>
<p>other parameters</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_pop">pop</code></td>
<td>
<p>list of populations to be kept</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_treatother">treatOther</code></td>
<td>
<p>elements in other (and ind.metrics &amp; loci.metrics) as 
indexed as well. default: TRUE</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_quiet">quiet</code></td>
<td>
<p>warnings are suppressed. default: TRUE</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_drop">drop</code></td>
<td>
<p>reduced to a vector if a single individual/loci is selected. 
default: FALSE [should never set to TRUE]</p>
</td></tr>
</table>

<hr>
<h2 id='bandicoot.gl'>A genlight object created via the read.dart functions
This a test data set to test the validity of functions within dartR and is 
based on a DArT SNP data set of simulated bandicoots across Australia. It 
contains 96 individuals and 1000 SNPs.</h2><span id='topic+bandicoot.gl'></span>

<h3>Description</h3>

<p>A genlight object created via the read.dart functions
This a test data set to test the validity of functions within dartR and is 
based on a DArT SNP data set of simulated bandicoots across Australia. It 
contains 96 individuals and 1000 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bandicoot.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='cbind.dartR'>adjust cbind for dartR</h2><span id='topic+cbind.dartR'></span>

<h3>Description</h3>

<p>cbind is a bit lazy and does not take care for the metadata (so data in the
other slot is lost). You can get most of the loci metadata back using
gl.compliance.check.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dartR'
cbind(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbind.dartR_+3A_...">...</code></td>
<td>
<p>list of dartR objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t1 &lt;- platypus.gl
class(t1) &lt;- "dartR"
t2 &lt;- cbind(t1[,1:10],t1[,11:20])
</code></pre>

<hr>
<h2 id='gl.alf'>Calculates allele frequency of the first and second allele for each locus
A very simple function to report allele frequencies</h2><span id='topic+gl.alf'></span>

<h3>Description</h3>

<p>Calculates allele frequency of the first and second allele for each locus
A very simple function to report allele frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.alf(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.alf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple data.frame with ref (reference allele), alt (alternate allele).
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#for the first 10 loci only
#Deprecated:
gl.alf(possums.gl[,1:10])
barplot(t(as.matrix(gl.alf(possums.gl[,1:10]))))
#Current:
gl.allele.freq(possums.gl[,1:10],simple=TRUE)
barplot(t(as.matrix(gl.allele.freq(possums.gl[,1:10],simple=TRUE))))
</code></pre>

<hr>
<h2 id='gl.allele.freq'>Generates percentage allele frequencies by locus and population</h2><span id='topic+gl.allele.freq'></span>

<h3>Description</h3>

<p>This is a support script, to take SNP data or SilicoDArT presence/absence
data grouped into populations in a genlight object {adegenet} and generate
a table of allele frequencies for each population and locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.allele.freq(x, percent = FALSE, by = "pop", simple = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.allele.freq_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or Tag P/A
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.allele.freq_+3A_percent">percent</code></td>
<td>
<p>If TRUE, percentage allele frequencies are given, if FALSE
allele proportions are given [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.allele.freq_+3A_by">by</code></td>
<td>
<p>If by='popxloc' then breakdown is given by population and locus; if by='pop'
then breakdown is given by population with statistics averaged across loci; if by='loc'
then breakdown is given by locus with statistics averaged across individuals [default 'pop']</p>
</td></tr>
<tr><td><code id="gl.allele.freq_+3A_simple">simple</code></td>
<td>
<p>A legacy option to return a dataframe with the frequency of the 
reference allele (alf1) and the frequency of the alternate allele (alf2) by locus [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.allele.freq_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with allele (SNP data) or presence/absence frequencies
(Tag P/A data) broken down by population and locus
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other unmatched report: 
<code><a href="#topic+gl.report.basics">gl.report.basics</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.allele.freq(testset.gl,percent=FALSE,by='pop')
gl.allele.freq(testset.gl,percent=FALSE,by="loc")
gl.allele.freq(testset.gl,percent=FALSE,by="popxloc")
gl.allele.freq(testset.gl,simple=TRUE)


</code></pre>

<hr>
<h2 id='gl.amova'>Performs AMOVA using genlight data</h2><span id='topic+gl.amova'></span>

<h3>Description</h3>

<p>This script performs an AMOVA based on the genetic distance matrix from
stamppNeisD() [package StAMPP] using the amova() function from the package
PEGAS for exploring within and between population variation. For detailed
information use their help pages: ?pegas::amova, ?StAMPP::stamppAmova. Be
aware due to a conflict of the amova functions from various packages I had
to 'hack' StAMPP::stamppAmova to avoid a namespace conflict.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.amova(x, distance = NULL, permutations = 100, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.amova_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes, with
population information [required].</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_distance">distance</code></td>
<td>
<p>Distance matrix between individuals (if not provided NeisD
from StAMPP::stamppNeisD is calculated) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_permutations">permutations</code></td>
<td>
<p>Number of permutations to perform for hypothesis
testing [default 100]. Please note should be set to 1000 for analysis.</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'amova' which is a list with a table of sums of
square deviations (SSD), mean square deviations (MSD), and the number of
degrees of freedom, and a vector of variance components.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#permutations should be higher, here set to 1 because of speed
out &lt;- gl.amova(bandicoot.gl, permutations=1)

</code></pre>

<hr>
<h2 id='gl.check.verbosity'>Checks the current global verbosity</h2><span id='topic+gl.check.verbosity'></span>

<h3>Description</h3>

<p>The verbosity can be set in one of two ways &ndash; (a) explicitly by the user by
passing a value using the parameter verbose in a function, or (b) by setting
the verbosity globally as part of the r environment (gl.set.verbosity).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.check.verbosity(x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.check.verbosity_+3A_x">x</code></td>
<td>
<p>User requested level of verbosity [default NULL].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The verbosity, in variable verbose
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other environment: 
<code><a href="#topic+gl.check.wd">gl.check.wd</a>()</code>,
<code><a href="#topic+gl.print.history">gl.print.history</a>()</code>,
<code><a href="#topic+gl.set.wd">gl.set.wd</a>()</code>,
<code><a href="#topic+theme_dartR">theme_dartR</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.check.verbosity()

</code></pre>

<hr>
<h2 id='gl.check.wd'>Checks the global working directory</h2><span id='topic+gl.check.wd'></span>

<h3>Description</h3>

<p>The working directory can be set in one of two ways &ndash; (a) explicitly by the user by
passing a value using the parameter plot.dir in a function, or (b) by setting
the working directory globally as part of the r environment (gl.setwd). The default is in acccordance to CRAN set to tempdir().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.check.wd(wd = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.check.wd_+3A_wd">wd</code></td>
<td>
<p>path to the working directory [default: tempdir()].</p>
</td></tr>
<tr><td><code id="gl.check.wd_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the working directory
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other environment: 
<code><a href="#topic+gl.check.verbosity">gl.check.verbosity</a>()</code>,
<code><a href="#topic+gl.print.history">gl.print.history</a>()</code>,
<code><a href="#topic+gl.set.wd">gl.set.wd</a>()</code>,
<code><a href="#topic+theme_dartR">theme_dartR</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.check.wd()

</code></pre>

<hr>
<h2 id='gl.colors'>Returns a list of colors for use in plots</h2><span id='topic+gl.colors'></span>

<h3>Description</h3>

<p>Creates a vector of colors in hex (e.g. &quot;#3B9AB2&quot; &quot;#78B7C5&quot;) based on user selected category 
(parameter type).
</p>

<ul>
<li><p> &quot;2&quot; [two colors]
</p>
</li>
<li><p> &quot;2c&quot; [two colors contrast]
</p>
</li>
<li><p> &quot;3&quot; [three colors]
</p>
</li>
<li><p>  4&quot; [four colors]
</p>
</li>
<li><p> &quot;pal&quot; [need to be specify the palette type and the number of colors]
</p>
</li></ul>

<p>A palette of colors 
can be specified via &quot;div&quot; [divergent], &quot;dis&quot; [discrete], &quot;con&quot; [convergent], &quot;vir&quot; [viridis]. 
Be aware a palette needs the number of colors specified as well. It returns a function 
and therefore the number of colors needs to be a part of the function call. 
Check the examples to see how this works.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.colors(type = 2, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.colors_+3A_type">type</code></td>
<td>
<p>Type of color (2, 3 or 4 colors, or palette, see description) [default 2].</p>
</td></tr>
<tr><td><code id="gl.colors_+3A_verbose">verbose</code></td>
<td>
<p>&ndash; verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns colors as a vector
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other graphics: 
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.colors(2)
gl.colors("2")
gl.colors("2c")
#five discrete colors
gl.colors(type="dis")(5)
#seven divergent colors
gl.colors("div")(7)
</code></pre>

<hr>
<h2 id='gl.compliance.check'>Checks a genlight object to see if it complies with dartR
expectations and amends it to comply if necessary
@family environment</h2><span id='topic+gl.compliance.check'></span>

<h3>Description</h3>

<p>This function will check to see that the genlight object conforms to
expectation in regard to dartR requirements (see details), and if it does
not, will rectify it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.compliance.check(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.compliance.check_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.compliance.check_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A genlight object used by dartR has a number of requirements that allow
functions within the package to operate correctly. The genlight object
comprises:
</p>

<ol>
<li><p> The SNP genotypes or Tag Presence/Absence data (SilicoDArT);
</p>
</li>
<li><p> An associated dataframe (gl@other$loc.metrics) containing the locus
metrics (e.g. Call Rate, Repeatability, etc);
</p>
</li>
<li><p> An associated dataframe (gl@other$ind.metrics) containing the
individual/sample metrics (e.g. sex, latitude (=lat), longitude(=lon), etc);
</p>
</li>
<li><p> A specimen identity field (indNames(gl)) with the unique labels applied
to each individual/sample;
</p>
</li>
<li><p> A population assignment (popNames) for each individual/specimen;
</p>
</li>
<li><p> Flags that indicate whether or not calculable locus metrics have been
updated.
</p>
</li></ol>



<h3>Value</h3>

<p>A genlight object that conforms to the expectations of dartR
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- gl.compliance.check(testset.gl)
x &lt;- gl.compliance.check(testset.gs)

</code></pre>

<hr>
<h2 id='gl.define.pop'>Defines a new population in a genlight object for specified 
individuals</h2><span id='topic+gl.define.pop'></span>

<h3>Description</h3>

<p>The script reassigns existing individuals to a new population and removes
their existing population assignment.
The script returns a genlight object with the new population assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.define.pop(x, ind.list, new, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.define.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_ind.list">ind.list</code></td>
<td>
<p>A list of individuals to be assigned to the new population
[required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_new">new</code></td>
<td>
<p>Name of the new population [required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the redefined population structure.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>popNames(testset.gl)
gl &lt;- gl.define.pop(testset.gl, ind.list=c('AA019073','AA004859'), 
new='newguys')
popNames(gl)
indNames(gl)[pop(gl)=='newguys']

</code></pre>

<hr>
<h2 id='gl.diagnostics.hwe'>Provides descriptive stats and plots to diagnose potential problems
with Hardy-Weinberg proportions
@family matched report</h2><span id='topic+gl.diagnostics.hwe'></span>

<h3>Description</h3>

<p>Different causes may be responsible for lack of Hardy-Weinberg
proportions. This function helps diagnose potential problems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.diagnostics.hwe(
  x,
  alpha_val = 0.05,
  bins = 20,
  stdErr = TRUE,
  colors.hist = gl.colors(2),
  colors.barplot = gl.colors("2c"),
  plot.theme = theme_dartR(),
  n.cores = "auto",
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.diagnostics.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 20].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_stderr">stdErr</code></td>
<td>
<p>Whether standard errors for Fis and Fst should be computed 
(default: TRUE)</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_colors.hist">colors.hist</code></td>
<td>
<p>List of two color names for the borders and fill of the
histogram [default gl.colors(2)].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_colors.barplot">colors.barplot</code></td>
<td>
<p>Vector with two color names for the observed and
expected number of significant HWE tests [default gl.colors(&quot;2c&quot;)].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use. If &quot;auto&quot;, it will 
use all but one available cores [default &quot;auto&quot;].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension)
[default NULL]</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function initially runs <code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code> and reports
the ternary plots. The remaining outputs follow the recommendations from
Waples
(2015) paper and De Meeûs 2018. These include: 
</p>
 
<ol>
<li><p> A histogram
with the distribution of p-values of the HWE tests. The distribution should
be roughly uniform across equal-sized bins. 
</p>
</li>
<li><p> A bar plot with observed
and expected (null expectation) number of significant HWE tests for the same
locus in multiple populations (that is, the x-axis shows whether a locus
results significant in 1, 2, ..., n populations. The y axis is the count of
these occurrences. The zero value on x-axis shows the number of
non-significant tests). If HWE tests are significant by chance alone,
observed and expected number of HWE tests should have roughly a similar
distribution. 
</p>
</li>
<li><p> A scatter plot with a linear regression between Fst and Fis,
averaged across subpopulations. De Meeûs 2018 suggests that in the case of
Null alleles, a strong positive relationship is expected (together with the
Fis standard error much larger than the Fst standard error, see below).
<b>Note</b>, this is not the scatter plot that Waples 2015 presents in his
paper. In the lower right corner of the plot, the Pearson correlation
coefficient is reported. 
</p>
</li>
<li><p> The Fis and Fst (averaged over loci and
subpopulations) standard errors are also printed on screen and reported in
the returned list (if <code>stdErr=TRUE</code>). These are computed with the 
Jackknife method over loci (See De Meeûs 2007 for details on how this is 
computed) and it may take some time for these computations to complete. 
De Meeûs 2018 suggests that under a global significant heterozygosity 
deficit: 
- if the
correlation between Fis and Fst is strongly positive, and StdErrFis &gt;&gt;
StdErrFst, Null alleles are likely to be the cause. 
- if the correlation
between Fis and Fst is ~0 or mildly positive, and StdErrFis &gt; StdErrFst,
Wahlund may be the cause. 
- if the correlation between Fis and Fst is ~0, and
StdErrFis ~ StdErrFst, selfing or sib mating could to be the cause.
It is
important to realise that these statistics only suggest a pattern (pointers).
Their absence is not conclusive evidence of the absence of the problem, as 
their presence does not confirm the cause of the problem. 
</p>
</li>
<li><p> A table where the
number of observed and expected significant HWE tests are reported by each
population, indicating whether these are due to heterozygosity excess or
deficiency. These can be used to have a clue of potential problems (e.g.
deficiency might be due to a Wahlund effect, presence of null alleles or
non-random sampling; excess might be due to sex linkage or different
selection between sexes, demographic changes or small Ne. See Table 1 in
Wapples 2015). The last two columns of the table generated by this function
report chisquare values and their associated p-values. Chisquare is computed
following Fisher's procedure for a global test (Fisher 1970). This basically
tests whether there is at least one test that is truly significant in the
series of tests conducted (De Meeûs et al 2009).
</p>
</li></ol>



<h3>Value</h3>

<p>A list with the table with the summary of the HWE tests and (if 
stdErr=TRUE) a named vector with the StdErrFis and StdErrFst.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>

 
<ul>
<li><p> de Meeûs, T., McCoy, K.D., Prugnolle, F.,
Chevillon, C., Durand, P., Hurtrez-Boussès, S., Renaud, F., 2007. Population
genetics and molecular epidemiology or how to “débusquer la bête”. Infection,
Genetics and Evolution 7, 308-332. 
</p>
</li>
<li><p> De Meeûs, T., Guégan, J.-F., Teriokhin, A.T., 2009. MultiTest V.1.2, 
a program to binomially combine independent tests and performance comparison 
with other related methods on
proportional data. BMC Bioinformatics 10, 443-443. 
</p>
</li>
<li><p> De Meeûs, T., 2018. Revisiting FIS, FST, Wahlund Effects, and Null 
Alleles. Journal of Heredity 109, 446-456.
</p>
</li>
<li><p> Fisher, R., 1970.
Statistical methods for research workers Edinburgh: Oliver and Boyd. 
</p>
</li>
<li>
<p>Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we lost
the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
res &lt;- gl.diagnostics.hwe(x = gl.filter.allna(platypus.gl[,1:50]), 
stdErr=FALSE, n.cores=1)

</code></pre>

<hr>
<h2 id='gl.dist.ind'>Calculates a distance matrix for individuals defined in a genlight object</h2><span id='topic+gl.dist.ind'></span>

<h3>Description</h3>

<p>Calculates various distances between individuals based on allele
frequencies or presence-absence data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.dist.ind(
  x,
  method = NULL,
  scale = FALSE,
  swap = FALSE,
  type = "dist",
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.dist.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight [required].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_method">method</code></td>
<td>
<p>Specify distance measure [SNP: Euclidean; P/A: Simple].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_scale">scale</code></td>
<td>
<p>If TRUE, the distances are scaled to fall in the range [0,1] [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_swap">swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_type">type</code></td>
<td>
<p>Specify the type of output, dist or matrix [default dist]</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure for SNP genotypes can be one of:
</p>

<ul>
<li><p> Euclidean Distance [method = &quot;Euclidean&quot;]
</p>
</li>
<li><p> Scaled Euclidean Distance [method='Euclidean&quot;, scale=TRUE]
</p>
</li>
<li><p> Simple Mismatch Distance [method=&quot;Simple&quot;]
</p>
</li>
<li><p> Absolute Mismatch Distance [method=&quot;Absolute&quot;]
</p>
</li>
<li><p> Czekanowski (Manhattan) Distance [method=&quot;Manhattan&quot;]
</p>
</li></ul>

<p>The distance measure for Sequence Tag Presence/Absence data (binary) can be one of:
</p>

<ul>
<li><p> Euclidean Distance [method = &quot;Euclidean&quot;]
</p>
</li>
<li><p> Scaled Euclidean Distance [method='Euclidean&quot;, scale=TRUE]
</p>
</li>
<li><p> Simple Matching Distance [method=&quot;Simple&quot;]
</p>
</li>
<li><p> Jaccard Distance [method=&quot;Jaccard&quot;]
</p>
</li>
<li><p> Bray-Curtis Distance [method=&quot;Bray-Curtis&quot;]
</p>
</li></ul>

<p>Refer to the documentation of functions in
https://doi.org/10.1101/2023.03.22.533737 for algorithms
and definitions.
</p>


<h3>Value</h3>

<p>An object of class 'matrix' or dist' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Custodian: Arthur Georges &ndash; Post to #' <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other distance: 
<code><a href="#topic+gl.dist.pop">gl.dist.pop</a>()</code>,
<code><a href="#topic+gl.fdsim">gl.fdsim</a>()</code>,
<code><a href="#topic+utils.dist.ind.snp">utils.dist.ind.snp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
D &lt;- gl.dist.ind(testset.gl[1:20,], method='manhattan')
D &lt;- gl.dist.ind(testset.gs[1:20,], method='Jaccard',swap=TRUE)

D &lt;- gl.dist.ind(testset.gl[1:20,], method='euclidean',scale=TRUE)

</code></pre>

<hr>
<h2 id='gl.dist.pop'>Calculates a distance matrix for populations with SNP genotypes in a
genlight object</h2><span id='topic+gl.dist.pop'></span>

<h3>Description</h3>

<p>This script calculates various distances between populations based on allele
frequencies (SNP genotypes) or frequency of presences in PA (SilicoDArT) data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.dist.pop(
  x,
  as.pop = NULL,
  method = "euclidean",
  scale = FALSE,
  type = "dist",
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.dist.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing [required].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Temporarily assign another locus metric as the population for
the purposes of deletions [default NULL].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default euclidean].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_type">type</code></td>
<td>
<p>Specify the type of output, dist or matrix [default dist]</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure can be one of 'euclidean', 'fixed-diff', 'reynolds',
'nei' and 'chord'. Refer to the documentation of functions in
https://doi.org/10.1101/2023.03.22.533737 for algorithms
and definitions.
</p>


<h3>Value</h3>

<p>An object of class 'dist' giving distances between populations
</p>


<h3>Author(s)</h3>

<p>author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other distance: 
<code><a href="#topic+gl.dist.ind">gl.dist.ind</a>()</code>,
<code><a href="#topic+gl.fdsim">gl.fdsim</a>()</code>,
<code><a href="#topic+utils.dist.ind.snp">utils.dist.ind.snp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP genotypes
D &lt;- gl.dist.pop(possums.gl, method='euclidean')
D &lt;- gl.dist.pop(possums.gl, method='euclidean',scale=TRUE)
D &lt;- gl.dist.pop(possums.gl, method='nei')
D &lt;- gl.dist.pop(possums.gl, method='reynolds')
D &lt;- gl.dist.pop(possums.gl, method='chord')
D &lt;- gl.dist.pop(possums.gl, method='fixed-diff')
#Presence-Absence data [only 10 individuals due to speed]
D &lt;- gl.dist.pop(testset.gs[1:10,], method='euclidean')


</code></pre>

<hr>
<h2 id='gl.drop.ind'>Removes specified individuals from a dartR genlight object</h2><span id='topic+gl.drop.ind'></span>

<h3>Description</h3>

<p>This function deletes individuals and their associated metadata.
Monomorphic loci and loci that are scored all NA are optionally deleted (mono.rm=TRUE). 
The script also optionally recalculates locus metatdata statistics to accommodate
the deletion of individuals from the dataset (recalc=TRUE).
</p>
<p>The script returns a dartR genlight object with the retained individuals 
and the recalculated locus metadata. The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.ind(x, ind.list, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_ind.list">ind.list</code></td>
<td>
<p>List of individuals to be removed [required].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic and all NA loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reduced dartR genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.ind">gl.keep.ind</a></code> to keep rather than drop specified
individuals
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.drop.ind(testset.gl,
   ind.list=c('AA019073','AA004859'))
 # Tag P/A data
   gs2 &lt;- gl.drop.ind(testset.gs,
   ind.list=c('AA020656','AA19077','AA004859'))
   gs2 &lt;- gl.drop.ind(testset.gs, ind.list=c('AA020656'
   ,'AA19077','AA004859'),mono.rm=TRUE, recalc=TRUE)
   
</code></pre>

<hr>
<h2 id='gl.drop.loc'>Removes specified loci from a dartR genlight object</h2><span id='topic+gl.drop.loc'></span>

<h3>Description</h3>

<p>This function deletes individuals and their associated metadata. 
The script returns a dartR genlight object with the retained loci. 
The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.loc(x, loc.list = NULL, first = NULL, last = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.loc_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_loc.list">loc.list</code></td>
<td>
<p>A list of loci to be deleted
[required, if loc.range not specified].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_first">first</code></td>
<td>
<p>First of a range of loci to be deleted
[required, if loc.list not specified].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_last">last</code></td>
<td>
<p>Last of a range of loci to be deleted
[if not specified, last locus in the dataset].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reduced dartR genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.loc">gl.keep.loc</a></code> to keep rather than drop specified loci
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl2 &lt;- gl.drop.loc(testset.gl, loc.list=c('100051468|42-A/T', '100049816-51-A/G'),verbose=3)
# Tag P/A data
  gs2 &lt;- gl.drop.loc(testset.gs, loc.list=c('20134188','19249144'),verbose=3)
  
</code></pre>

<hr>
<h2 id='gl.drop.pop'>Removes specified populations from a dartR genlight object</h2><span id='topic+gl.drop.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on associated specimen metadata
stored in the dartR genlight object. 
This function deletes all individuals in the nominated populations (pop.list).
Monomorphic loci and loci that are scored all NA are optionally deleted (mono.rm=TRUE). 
The script also optionally recalculates locus metatdata statistics to accommodate
the deletion of individuals from the dataset (recalc=TRUE).
The script returns a dartR genlight object with the retained populations 
and the recalculated locus metadata. The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.pop(
  x,
  pop.list,
  as.pop = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_pop.list">pop.list</code></td>
<td>
<p>List of populations to be removed [required].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Temporarily assign another locus metric as the population for
the purposes of deletions [default NULL].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic and all NA loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reduced dartR genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.pop">gl.keep.pop</a></code> to keep rather than drop specified populations
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.drop.pop(testset.gl,
   pop.list=c('EmsubRopeMata','EmvicVictJasp'),verbose=3)
   gl2 &lt;- gl.drop.pop(testset.gl, pop.list=c('EmsubRopeMata','EmvicVictJasp'),
   mono.rm=TRUE,recalc=TRUE)
   gl2 &lt;- gl.drop.pop(testset.gl,as.pop='sex',pop.list=c('Male','Unknown'),verbose=3)
 # Tag P/A data
   gs2 &lt;- gl.drop.pop(testset.gs, pop.list=c('EmsubRopeMata','EmvicVictJasp'))
</code></pre>

<hr>
<h2 id='gl.edit.recode.ind'>Creates or edits individual (=specimen) names, creates a recode_ind
file and applies the changes to a genlight object
@family data manipulation</h2><span id='topic+gl.edit.recode.ind'></span>

<h3>Description</h3>

<p>A function to edit names of individual in a dartR genlight object, or to create a
reassignment table taking the individual labels from a genlight object, or to
edit existing individual labels in an existing recode_ind file. The amended 
recode table is then applied to the genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.edit.recode.ind(
  x,
  out.recode.file = NULL,
  outpath = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.edit.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>Name of the file to output the new individual labels
[optional].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_outpath">outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Renaming individuals may be required when there have been errors in labeling
arising in the passage of samples to sequencing. There may be occasions
where renaming individuals is required for preparation of figures. 
This function will input an existing recode table for editing and optionally
save it as a new table, or if the name of an input table is not supplied,
will generate a table using the individual labels in the parent genlight
object.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a durable record of the changes.
For SNP genotype data, the function, having deleted individuals, optionally 
identifies resultant monomorphic loci or loci with all values missing 
and deletes them. The script also optionally recalculates the
locus metadata as appropriate. The optional deletion of monomorphic loci
and the optional recalculation of locus statistics is not available for
Tag P/A data (SilicoDArT).
Use outpath=getwd() when calling this function to direct
output files to your working directory.
The function returns a dartR genlight object with the new population assignments  
and the recalculated locus metadata.
</p>


<h3>Value</h3>

<p>An object of class ('genlight') with the revised individual labels.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.recode.ind">gl.recode.ind</a></code>, <code><a href="#topic+gl.drop.ind">gl.drop.ind</a></code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#this is an interactive example
if(interactive()){
gl &lt;- gl.edit.recode.ind(testset.gl)
gl &lt;- gl.edit.recode.ind(testset.gl, out.recode.file='ind.recode.table.csv')
}

</code></pre>

<hr>
<h2 id='gl.edit.recode.pop'>Creates or edits and applies a population re-assignment table</h2><span id='topic+gl.edit.recode.pop'></span>

<h3>Description</h3>

<p>A function to edit population assignments in a dartR genlight object, or to
create a reassignment table taking the population assignments
from a genlight object, or to edit existing population assignments in
a pop.recode.table. The amended recode table is then applied to the genlight
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.edit.recode.pop(
  x,
  pop.recode = NULL,
  out.recode.file = NULL,
  outpath = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.edit.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object  [required].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_pop.recode">pop.recode</code></td>
<td>
<p>Path to recode file [default NULL].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>Name of the file to output the new individual labels
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_outpath">outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].
</p>
<p>@details
Genlight objects assign specimens to populations based on information in the
ind.metadata file provided when the genlight object is first generated.
Often one wishes to subset the data by deleting populations or to amalgamate
populations. This can be done with a pop.recode table with two columns. The
first column is the population assignment in the genlight object, the second
column provides the new assignment.
This function will input an existing reassignment table for editing and
optionally save it as a new table, or if the name of an input table is not
supplied, will generate a table using the population assignments in the
parent genlight object. It will then apply the recodings to the genlight 
object.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a durable record of the changes.
For SNP genotype data, the function, having deleted populations, optionally 
identifies resultant monomorphic loci or loci with all values missing 
and deletes them. The script also optionally recalculates the
locus metadata as appropriate. The optional deletion of monomorphic loci
and the optional recalculation of locus statistics is not available for
Tag P/A data (SilicoDArT).
Use outpath=getwd() when calling this function to direct
output files to your working directory.
The function returns a dartR genlight object with the new population assignments  
and the recalculated locus metadata.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the revised population assignments
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code>, <code><a href="#topic+gl.drop.pop">gl.drop.pop</a></code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a></code>, <code><a href="#topic+gl.merge.pop">gl.merge.pop</a></code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a></code>
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#this is an interactive example
if(interactive()){
gl &lt;- gl.edit.recode.pop(testset.gl)
gs &lt;- gl.edit.recode.pop(testset.gs)
}

# See also -------------------
</code></pre>

<hr>
<h2 id='gl.fdsim'>Estimates the rate of false positives in a fixed difference analysis</h2><span id='topic+gl.fdsim'></span>

<h3>Description</h3>

<p>This function takes two populations and generates allele frequency profiles
for them. It then samples an allele frequency for each, at random, and
estimates a sampling distribution for those two allele frequencies. Drawing
two samples from those sampling distributions, it calculates whether or not
they represent a fixed difference. This is applied to all loci, and the
number of fixed differences so generated are counted, as an expectation. The
script distinguished between true fixed differences (with a tolerance of
delta), and false positives. The simulation is repeated a given number of
times (default=1000) to provide an expectation of the number of false
positives, given the observed allele frequency profiles and the sample sizes.
The probability of the observed count of fixed differences is greater than
the expected number of false positives is calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fdsim(
  x,
  poppair,
  obs = NULL,
  sympatric = FALSE,
  reps = 1000,
  delta = 0.02,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fdsim_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_poppair">poppair</code></td>
<td>
<p>Labels of two populations for comparison in the form
c(popA,popB) [required].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_obs">obs</code></td>
<td>
<p>Observed number of fixed differences between the two populations
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_sympatric">sympatric</code></td>
<td>
<p>If TRUE, the two populations are sympatric, if FALSE then
allopatric [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_reps">reps</code></td>
<td>
<p>Number of replications to undertake in the simulation
[default 1000].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_delta">delta</code></td>
<td>
<p>The threshold value for the minor allele frequency to regard the
difference between two populations to be fixed [default 0.02].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent, fatal errors only; 1, flag function
begin and end; 2, progress log; 3, progress and results summary; 5, full
report [default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following square matrices
[[1]] observed fixed differences;
[[2]] mean expected number of false positives for each comparison;
[[3]] standard deviation of the no. of false positives for each
comparison;
[[4]] probability the observed fixed differences arose by chance for
each comparison.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other distance: 
<code><a href="#topic+gl.dist.ind">gl.dist.ind</a>()</code>,
<code><a href="#topic+gl.dist.pop">gl.dist.pop</a>()</code>,
<code><a href="#topic+utils.dist.ind.snp">utils.dist.ind.snp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fd &lt;- gl.fdsim(testset.gl[,1:100],poppair=c('EmsubRopeMata','EmmacBurnBara'),
sympatric=TRUE,verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.allna'>Filters loci that are all NA across individuals and/or populations 
with all NA across loci</h2><span id='topic+gl.filter.allna'></span>

<h3>Description</h3>

<p>This script deletes deletes loci or individuals with all calls missing (NA),
from a genlight object
A DArT dataset will not have loci for which the calls are scored all as
missing (NA) for a particular individual, but such loci can arise rarely when
populations or individuals are deleted. Similarly, a DArT dataset will not
have individuals for which the calls are scored all as missing (NA) across
all loci, but such individuals may sneak in to the dataset when loci are
deleted. Retaining individual or loci with all NAs can cause issues for
several functions.
Also, on occasion an analysis will require that there are some loci scored
in each population. Setting by.pop=TRUE will result in removal of loci when
they are all missing in any one population.
Note that loci that are missing for all individuals in a population are
not imputed with method 'frequency' or 'HW'. Consider 
using the function <code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> with by.pop=TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.allna(x, by.pop = FALSE, recalc = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.allna_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_by.pop">by.pop</code></td>
<td>
<p>If TRUE, loci that are all missing in any one population
are deleted [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object having removed individuals that are scored NA
across all loci, or loci that are scored NA across all individuals.
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  result &lt;- gl.filter.allna(testset.gl, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.allna(testset.gs, verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.callrate'>Filters loci or specimens in a genlight {adegenet} object based on
call rate</h2><span id='topic+gl.filter.callrate'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the restriction
enzyme recognition sites. The script gl.filter.callrate() will filter out the
loci with call rates below a specified threshold.
Tag Presence/Absence datasets (SilicoDArT) have missing values where it is
not possible to determine reliably if there the sequence tag can be called at
a particular locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.callrate(
  x,
  method = "loc",
  threshold = 0.95,
  mono.rm = FALSE,
  recalc = FALSE,
  recursive = FALSE,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  bins = 25,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data, or the genind
object containing the SilocoDArT data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_method">method</code></td>
<td>
<p>Use method='loc' to specify that loci are to be filtered, 'ind'
to specify that specimens are to be filtered, 'pop' to remove loci that 
fail to meet the specified threshold in any one population [default 'loc'].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value below which loci will be removed
[default 0.95].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci after analysis is complete
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_recursive">recursive</code></td>
<td>
<p>Repeatedly filter individuals on call rate, each time
removing monomorphic loci. Only applies if method='ind' and mono.rm=TRUE
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because this filter operates on call rate, this function recalculates Call
Rate, if necessary, before filtering. If individuals are removed using
method='ind', then the call rate stored in the genlight object is, 
optionally,
recalculated after filtering.
Note that when filtering individuals on call rate, the initial call rate is
calculated and compared against the threshold. After filtering, if
mono.rm=TRUE, the removal of monomorphic loci will alter the call rates.
Some individuals with a call rate initially greater than the nominated
threshold, and so retained, may come to have a call rate lower than the
threshold. If this is a problem, repeated iterations of this function will
resolve the issue. This is done by setting mono.rm=TRUE and recursive=TRUE,
or it can be done manually.
Callrate is summarized by locus or by individual to allow sensible decisions
on thresholds for filtering taking into consideration consequential loss of
data. The summary is in the form of a tabulation and plots.
Plot themes can be obtained from </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Resultant ggplot(s) and the tabulation(s) are saved to the session's
temporary directory.
</p>


<h3>Value</h3>

<p>The reduced genlight or genind object, plus a summary
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.callrate">gl.report.callrate</a></code>
</p>
<p>Other matched filter: 
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='loc', threshold=0.8,
   verbose=3)
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='ind', threshold=0.8,
   verbose=3)
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='pop', threshold=0.8,
   verbose=3)
# Tag P/A data
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='loc', 
  threshold=0.95, verbose=3)
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='ind', 
  threshold=0.8, verbose=3)
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='pop', 
  threshold=0.8, verbose=3)
  
  res &lt;- gl.filter.callrate(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.filter.factorloadings'>Filters loci based on factor loadings for a PCA or PCoA</h2><span id='topic+gl.filter.factorloadings'></span>

<h3>Description</h3>

<p>Extracts the factor loadings from a glPCA object (generated by gl.pcoa) and 
filters loci based on a user specified threshold for the ABSOLUTE value of the 
factor loadings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.factorloadings(
  x,
  pca,
  axis = 1,
  threshold,
  retain = FALSE,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  bins = 25,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.factorloadings_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data or the 
SilocoDArT data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_pca">pca</code></td>
<td>
<p>Name of the glPCA object containing factor loadings [required].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_axis">axis</code></td>
<td>
<p>Axis in the ordination used to display the factor loadings [default 1]</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_threshold">threshold</code></td>
<td>
<p>Remove [retain=FALSE] or retain [retain=TRUE] only those loci that
load high (greater than the threshold) against the specified axis [required].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_retain">retain</code></td>
<td>
<p>If true, the resultant genlight object holds only the loci that load
high on the specified axis; if FALSE, the resultant genlight object has the
loci loading high on the specified axis filtered out [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.filter.factorloadings_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="ggplot2.html#topic+ggsave">ggsave</a>, 
such as width and height, when the ggplot is to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function extracts the factor loadings for a given axis from a PCA object 
generated by gl.pcoa and then filters loci on the basis of a user specified
threshold. The threshold value is decided using gl.report.factorloadings.  The function
can be used to filter out loci that load high with a particular axis or alternatively
if retain=TRUE, to retain loci that load high on a specified axis.  
</p>
<p>A color vector can be obtained with gl.select.colors() and then passed to the function
with the plot.colors parameter.
</p>
<p>Themes can be obtained from in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir().
</p>


<h3>Value</h3>

<p>The unchanged genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- gl.pcoa(testset.gl)
gl.report.factorloadings(pca = pca)
gl2 &lt;- gl.filter.factorloadings(pca=pca,x=testset.gl,threshold=0.2)

</code></pre>

<hr>
<h2 id='gl.filter.hamming'>Filters loci based on pairwise Hamming distance between sequence tags</h2><span id='topic+gl.filter.hamming'></span>

<h3>Description</h3>

<p>Hamming distance is calculated as the number of base differences between two
sequences which can be expressed as a count or a proportion. Typically, it is
calculated between two sequences of equal length. In the context of DArT
trimmed sequences, which differ in length but which are anchored to the left
by the restriction enzyme recognition sequence, it is sensible to compare the
two trimmed sequences starting from immediately after the common recognition
sequence and terminating at the last base of the shorter sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.hamming(
  x,
  threshold = 0.2,
  rs = 5,
  tag.length = 69,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  pb = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.hamming_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_threshold">threshold</code></td>
<td>
<p>A threshold Hamming distance for filtering loci
[default threshold 0.2].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_rs">rs</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_tag.length">tag.length</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_pb">pb</code></td>
<td>
<p>If TRUE, a progress bar will be displayed [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hamming distance can be computed
by exploiting the fact that the dot product of two binary vectors x and (1-y)
counts the corresponding elements that are different between x and y.
This approach can also be used for vectors that contain more than two 
possible values at each position (e.g. A, C, T or G).
If a pair of DNA sequences are of differing length, the longer is truncated.
The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
as implemented in <code><a href="#topic+utils.hamming">utils.hamming</a></code>.
Only one of two loci are retained if their Hamming distance is less that a 
specified
percentage. 5 base differences out of 100 bases is a 20
</p>


<h3>Value</h3>

<p>A genlight object filtered on Hamming distance.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
test &lt;- gl.subsample.loc(platypus.gl,n=50)
result &lt;- gl.filter.hamming(test, threshold=0.6, verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.heterozygosity'>Filters individuals with average heterozygosity greater than a
specified upper threshold or less than a specified lower threshold
@family matched filter</h2><span id='topic+gl.filter.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the observed heterozygosity for each individual in a genlight
object and filters individuals based on specified threshold values.
Use gl.report.heterozygosity to determine the appropriate thresholds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.heterozygosity(x, t.upper = 0.7, t.lower = 0, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.heterozygosity_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_t.upper">t.upper</code></td>
<td>
<p>Filter individuals &gt; the threshold [default 0.7].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_t.lower">t.lower</code></td>
<td>
<p>Filter individuals &lt; the threshold [default 0].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The filtered genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> result &lt;- gl.filter.heterozygosity(testset.gl,t.upper=0.06,verbose=3)
 tmp &lt;- gl.report.heterozygosity(result,method='ind')
 
</code></pre>

<hr>
<h2 id='gl.filter.hwe'>Filters loci that show significant departure from Hardy-Weinberg
Equilibrium
@family matched filter</h2><span id='topic+gl.filter.hwe'></span>

<h3>Description</h3>

<p>This function filters out loci showing significant departure from H-W
proportions based on observed frequencies of reference homozygotes,
heterozygotes and alternate homozygotes.
Loci are filtered out if they show HWE departure either in any one population 
(n.pop.threshold =1) or in at least X number of populations 
(n.pop.threshold &gt; 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.hwe(
  x,
  subset = "each",
  n.pop.threshold = 1,
  test.type = "Exact",
  mult.comp.adj = FALSE,
  mult.comp.adj.method = "BY",
  alpha = 0.05,
  pvalue.type = "midp",
  cc.val = 0.5,
  n.min = 5,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_subset">subset</code></td>
<td>
<p>Way to group individuals to perform H-W tests. Either a vector
with population names, 'each', 'all' (see details) [default 'each'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_n.pop.threshold">n.pop.threshold</code></td>
<td>
<p>The minimum number of populations where the same locus 
has to be out of H-W proportions to be removed [default 1].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_test.type">test.type</code></td>
<td>
<p>Method for determining statistical significance: 
'ChiSquare'
or 'Exact' [default 'Exact'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_mult.comp.adj">mult.comp.adj</code></td>
<td>
<p>Whether to adjust p-values for multiple comparisons
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_mult.comp.adj.method">mult.comp.adj.method</code></td>
<td>
<p>Method to adjust p-values for multiple comparisons:
'holm', 'hochberg', 'hommel', 'bonferroni', 'BH', 'BY', 'fdr'
(see details) [default 'fdr'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_alpha">alpha</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_pvalue.type">pvalue.type</code></td>
<td>
<p>Type of p-value to be used in the Exact method.
Either 'dost','selome','midp' (see details) [default 'midp'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_cc.val">cc.val</code></td>
<td>
<p>The continuity correction applied to the ChiSquare test
[default 0.5].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_n.min">n.min</code></td>
<td>
<p>Minimum number of individuals per population in which
perform H-W tests [default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Several factors can cause deviations from Hardy-Weinberg
equilibrium including: mutation, finite population size, selection,
population structure, age structure, assortative mating, sex linkage,
nonrandom sampling and genotyping errors. Refer to Waples (2015).
Note that tests for departure from H-W equilibrium are only valid if there is no population
substructure (assuming random mating) and have sufficient power only when
there is sufficient sample size (n individuals &gt; 15).
Populations can be defined in three ways:
</p>

<ul>
<li><p> Merging all populations in the dataset using subset = 'all'.
</p>
</li>
<li><p> Within each population separately using: subset = 'each'.
</p>
</li>
<li><p> Within selected populations using for example: subset = 
c('pop1','pop2').
</p>
</li></ul>

<p>Two different statistical methods to test for deviations from Hardy Weinberg
proportions:
</p>

<ul>
<li><p> The classical chi-square test (test.type='ChiSquare') based on the
function <code><a href="HardyWeinberg.html#topic+HWChisq">HWChisq</a></code> of the R package HardyWeinberg.
By default a continuity correction is applied (cc.val=0.5). The
continuity correction can be turned off (by specifying cc.val=0), for example
when extreme allele frequencies occur continuity correction can
lead to excessive Type I error rates.
</p>
</li>
<li><p> The exact test (test.type='Exact') based on the exact calculations
contained in the function <code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code> of the R
package HardyWeinberg as described by Wigginton  et al. (2005). The exact
test is recommended in most cases.
Three different methods to estimate p-values (pvalue.type) in the Exact test
can be used:
</p>

<ul>
<li><p> 'dost' p-value is computed as twice the tail area of a one-sided test.
</p>
</li>
<li><p> 'selome' p-value is computed as the sum of the probabilities of all
samples less or equally likely as the current sample.
</p>
</li>
<li><p> 'midp', p-value is computed as half the probability of the current
sample + the probabilities of all samples that are more extreme.
</p>
</li></ul>

<p>The standard exact p-value is overly conservative, in particular
for small minor allele frequencies. The mid p-value ameliorates this problem
by bringing the rejection rate closer to the nominal level, at the price of
occasionally exceeding the nominal level (Graffelman &amp; Moreno, 2013).
</p>
</li></ul>

<p>Correction for multiple tests can be applied using the following methods
based on the function <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>:
</p>

<ul>
<li><p> 'holm' is also known as the sequential Bonferroni technique 
(Rice, 1989).
This method has a greater statistical power than the standard Bonferroni 
test,
however this method becomes very stringent when many tests are performed and
many real deviations from the null hypothesis can go undetected 
(Waples, 2015).
</p>
</li>
<li><p> 'hochberg' based on Hochberg, 1988.
</p>
</li>
<li><p> 'hommel' based on Hommel, 1988. This method is more powerful than
Hochberg's, but the difference is usually small.
</p>
</li>
<li><p> 'bonferroni' in which p-values are multiplied by the number of tests.
This method is very stringent and therefore has reduced power to detect
multiple departures from the null hypothesis.
</p>
</li>
<li><p> 'BH' based on Benjamini &amp; Hochberg, 1995.
</p>
</li>
<li><p> 'BY' based on Benjamini &amp; Yekutieli, 2001.
</p>
</li></ul>

<p>The first four methods are designed to give strong control of the family-wise
error rate. The last two methods control the false discovery rate (FDR),
the expected proportion of false discoveries among the rejected hypotheses.
The false discovery rate is a less stringent condition than the family-wise
error rate, so these methods are more powerful than the others, especially
when number of tests is large.
The number of tests on which the adjustment for multiple comparisons is
the number of populations times the number of loci.
<b>From v2.1</b> <code>gl.filter.hwe</code> takes the argument
<code>n.pop.threshold</code>.
if <code>n.pop.threshold &gt; 1</code> loci will be removed only if they are 
concurrently 
significant (after adjustment if applied) out of hwe in &gt;= 
<code>n.pop.threshold &gt; 1</code>.
</p>


<h3>Value</h3>

<p>A genlight object with the loci departing significantly from H-W
proportions removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Benjamini, Y., and Yekutieli, D. (2001). The control of the false
discovery rate in multiple testing under dependency. Annals of Statistics,
29, 1165–1188.
</p>
</li>
<li><p> Graffelman, J. (2015). Exploring Diallelic Genetic Markers: The Hardy
Weinberg Package. Journal of Statistical Software 64:1-23.
</p>
</li>
<li><p> Graffelman, J. &amp; Morales-Camarena, J. (2008). Graphical tests for
Hardy-Weinberg equilibrium based on the ternary plot. Human Heredity 
65:77-84.
</p>
</li>
<li><p> Graffelman, J., &amp; Moreno, V. (2013). The mid p-value in exact tests for
Hardy-Weinberg equilibrium. Statistical applications in genetics and
molecular
biology, 12(4), 433-448.
</p>
</li>
<li><p> Hochberg, Y. (1988). A sharper Bonferroni procedure for multiple tests
of significance. Biometrika, 75, 800–803.
</p>
</li>
<li><p> Hommel, G. (1988). A stagewise rejective multiple test procedure based
on a modified Bonferroni test. Biometrika, 75, 383–386.
</p>
</li>
<li><p> Rice, W. R. (1989). Analyzing tables of statistical tests. Evolution,
43(1), 223-225.
</p>
</li>
<li><p> Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we
lost the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li>
<li><p> Wigginton, J.E., Cutler, D.J., &amp; Abecasis, G.R. (2005). A Note on Exact
Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics
76:887-893.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.hwe(x = bandicoot.gl)
</code></pre>

<hr>
<h2 id='gl.filter.ld'>Filters loci based on linkage disequilibrium (LD)</h2><span id='topic+gl.filter.ld'></span>

<h3>Description</h3>

<p>This function uses the statistic set in the parameter <code>stat_keep</code> from 
function <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code> to choose the SNP to keep when two 
SNPs are in LD. When a SNP is selected to be filtered out in each pairwise 
comparison, the function stores its  name in a list. In subsequent pairwise
comparisons, if the SNP is already in the list, the other SNP will be kept.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.ld(
  x,
  ld.report,
  threshold = 0.2,
  pop.limit = ceiling(nPop(x)/2),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.ld_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_ld.report">ld.report</code></td>
<td>
<p>Output from function <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code> 
[required].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value above which loci will be removed
[default 0.2].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_pop.limit">pop.limit</code></td>
<td>
<p>Minimum number of populations in which LD should be more
than the threshold for a locus to be filtered out.
The default value is half of the populations [default ceiling(nPop(x)/2)].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code>
</p>
<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>

<hr>
<h2 id='gl.filter.locmetric'>Filters loci on the basis of numeric information stored in
other$loc.metrics in a genlight {adegenet} object</h2><span id='topic+gl.filter.locmetric'></span>

<h3>Description</h3>

<p>This script uses any field with numeric values stored in $other$loc.metrics
to filter loci. The loci to keep can be within the upper and lower thresholds
('within') or outside of the upper and lower thresholds ('outside').
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.locmetric(x, metric, upper, lower, keep = "within", verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.locmetric_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_metric">metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_upper">upper</code></td>
<td>
<p>Filter upper threshold [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_lower">lower</code></td>
<td>
<p>Filter lower threshold  [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_keep">keep</code></td>
<td>
<p>Whether keep loci within of upper and lower thresholds or keep
loci outside of upper and lower thresholds [within].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own filter by adding a
vector into $other$loc.metrics as shown in the example.
</p>

<ol>
<li><p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li><p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li><p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li><p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li><p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li><p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li><p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li><p> PICRef - polymorphism information content (PIC) for the Reference
allele.
</p>
</li>
<li><p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li><p> AvgPIC - average of the polymorphism information content (PIC) of the
Reference and SNP alleles.
</p>
</li>
<li><p> AvgCountRef - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Reference allele
row.
</p>
</li>
<li><p> AvgCountSnp - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Alternate (SNP)
allele row.
</p>
</li>
<li><p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li></ol>



<h3>Value</h3>

<p>The reduced genlight dataset.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># adding dummy data
test &lt;- testset.gl
test$other$loc.metrics$test &lt;- 1:nLoc(test)
result &lt;- gl.filter.locmetric(x=test, metric= 'test', upper=255,
lower=200, keep= 'within', verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.maf'>Filters loci on the basis of minor allele frequency (MAF) in a 
genlight object</h2><span id='topic+gl.filter.maf'></span>

<h3>Description</h3>

<p>This script calculates the minor allele frequency for each locus and updates
the locus metadata for FreqHomRef, FreqHomSnp, FreqHets and MAF (if it
exists). It then uses the updated metadata for MAF to filter loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.maf(
  x,
  threshold = 0.01,
  by.pop = FALSE,
  pop.limit = ceiling(nPop(x)/2),
  ind.limit = 10,
  recalc = FALSE,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  bins = 25,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_threshold">threshold</code></td>
<td>
<p>Threshold MAF &ndash; loci with a MAF less than the threshold
will be removed. If a value &gt; 1 is provided it will be 
interpreted as MAC (i.e. the minimum number of times an allele needs to be 
observed) [default 0.01].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_by.pop">by.pop</code></td>
<td>
<p>Whether MAF should be calculated by population [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_pop.limit">pop.limit</code></td>
<td>
<p>Minimum number of populations in which MAF should be less 
than the threshold for a locus to be filtered out. Only used if by.pop=TRUE. 
The default value is half of the populations [default ceiling(nPop(x)/2)].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should 
contain to calculate MAF. Only used if by.pop=TRUE [default 10].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms of base composition are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].
</p>
<p>@details 
Careful consideration needs to be given to the settings to be used for this 
fucntion. When the filter is applied globally (i.e. <code>by.pop=FALSE</code>) but 
the data include multiple population, there is the risk to remove markers 
because the allele frequencies is low (at global level) but the allele 
frequencies
for the same markers may be high within some of the populations (especially 
if 
the per-population sample size is small). Similarly, not always it is a 
sensible choice to run this function using <code>by.pop=TRUE</code> because allele 
that are rare in a population may be very common in other, but the (possible) 
allele frequencies will depend on the sample size within each population. 
Where the purpose of filtering for MAF is to remove possible spurious alleles 
(i.e. sequencing errors), it is perhaps better to filter based on the number 
of times an allele is observed (MAC, Minimum Allele Count), under the 
assumption that if an allele is observed &gt;MAC, it is fairly rare to be an 
error.
<b>From v2.1</b> The threshold can take values &gt; 1. In this case, these are 
interpreted as a threshold for MAC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight dataset
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>

<hr>
<h2 id='gl.filter.monomorphs'>Filters monomorphic loci, including those with all NAs</h2><span id='topic+gl.filter.monomorphs'></span>

<h3>Description</h3>

<p>This script deletes monomorphic loci from a genlight {adegenet} object
A DArT dataset will not have monomorphic loci, but they can arise, along with
loci that are scored all NA, when populations or individuals are deleted.
Retaining monomorphic loci unnecessarily increases the size of the dataset
and will affect some calculations.
Note that for SNP data, NAs likely represent null alleles; in tag
presence/absence data, NAs represent missing values (presence/absence could
not be reliably scored)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.monomorphs(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.monomorphs_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.monomorphs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with monomorphic (and all NA) loci removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  result &lt;- gl.filter.monomorphs(testset.gl, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.monomorphs(testset.gs, verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.overshoot'>Filters loci for which the SNP has been trimmed from the sequence tag
along with the adaptor</h2><span id='topic+gl.filter.overshoot'></span>

<h3>Description</h3>

<p>This function checks the position of the SNP within the trimmed sequence tag
and identifies those for which the SNP position is outside the trimmed
sequence tag. This can happen, rarely, when the sequence containing the SNP
resembles the adaptor.
The SNP genotype can still be used in most analyses, but functions like
gl2fasta() will present challenges if the SNP has been trimmed from the
sequence tag.
Not fatal, but should apply this filter before gl.filter.secondaries, for
obvious reasons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.overshoot(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.overshoot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.overshoot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new genlight object with the recalcitrant loci deleted
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges; Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.overshoot(testset.gl, verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.pa'>Filters loci that contain private (and fixed alleles) between two populations</h2><span id='topic+gl.filter.pa'></span>

<h3>Description</h3>

<p>This script is meant to be used prior to <code>gl.nhybrids</code> to maximise the
information content of the SNPs used to identify hybrids (currently
newhybrids does allow only 200 SNPs). The idea is to use first all loci that
have fixed alleles between the potential source populations and then 'fill
up' to 200 loci using loci that have private alleles between those. The
functions filters for those loci (if invers is set to TRUE, the opposite
is returned (all loci that are not fixed and have no private alleles - not
sure why yet, but maybe useful.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.pa(x, pop1, pop2, invers = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.pa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_pop1">pop1</code></td>
<td>
<p>Name of the first parental population (in quotes) [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_pop2">pop2</code></td>
<td>
<p>Name of the second parental population (in quotes) [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_invers">invers</code></td>
<td>
<p>Switch to filter for all loci that have no private alleles and
are not fixed [FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight dataset, containing now only fixed and private
alleles.
</p>


<h3>Author(s)</h3>

<p>Authors: Bernd Gruber &amp; Ella Kelly (University of Melbourne);
Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.pa(testset.gl, pop1=pop(testset.gl)[1], 
pop2=pop(testset.gl)[2],verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.rdepth'>Filters loci based on counts of sequence tags scored at a locus (read
depth)
@family matched filter</h2><span id='topic+gl.filter.rdepth'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT report AvgCountRef and AvgCountSnp as counts
of sequence tags for the reference and alternate alleles respectively. These
can be used to back calculate Read Depth. Fragment presence/absence datasets
as provided by DArT (SilicoDArT) provide Average Read Depth and Standard
Deviation of Read Depth as standard columns in their report.
Filtering on Read Depth using the companion script gl.filter.rdepth can be on
the basis of loci with exceptionally low counts,
or loci with exceptionally high counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.rdepth(
  x,
  lower = 5,
  upper = 1000,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.rdepth_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or tag
presence/absence data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_lower">lower</code></td>
<td>
<p>Lower threshold value below which loci will be removed
[default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_upper">upper</code></td>
<td>
<p>Upper threshold value above which loci will be removed
[default infinite=1000].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms of base composition are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For examples of themes, see:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a genlight object retaining loci with a Read Depth in the
range specified by the lower and upper threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.rdepth(testset.gl)
  result &lt;- gl.filter.rdepth(testset.gl, lower=8, upper=50, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.rdepth(testset.gs, lower=8, upper=50, verbose=3)
  
  res &lt;- gl.filter.rdepth(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.filter.reproducibility'>Filters loci in a genlight {adegenet} object based on average
repeatability of alleles at a locus
@family matched filter</h2><span id='topic+gl.filter.reproducibility'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have an index, RepAvg, generated by
reproducing the data independently for 30
of alleles that give a repeatable result, averaged over both alleles for each
locus.
SilicoDArT datasets generated by DArT have a similar index, Reproducibility.
For these fragment presence/absence data, repeatability is the percentage of
scores that are repeated in the technical replicate dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.reproducibility(
  x,
  threshold = 0.99,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.reproducibility_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value below which loci will be removed
[default 0.99].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms of base composition are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object retaining loci with repeatability (Repavg
or Reproducibility) greater than the specified threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.reproducibility(testset.gl)
  result &lt;- gl.filter.reproducibility(testset.gl, threshold=0.99, verbose=3)
# Tag P/A data
  gl.report.reproducibility(testset.gs)
  result &lt;- gl.filter.reproducibility(testset.gs, threshold=0.99)
  
  res &lt;- gl.filter.reproducibility(testset.gl)
  
</code></pre>

<hr>
<h2 id='gl.filter.secondaries'>Filters loci that represent secondary SNPs in a genlight object</h2><span id='topic+gl.filter.secondaries'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT include fragments with more than one SNP and
record them separately with the same CloneID (=AlleleID). These multiple SNP
loci within a fragment (secondaries) are likely to be linked, and so you may
wish to remove secondaries.
This script filters out all but the first sequence tag with the same CloneID
after ordering the genlight object on based on repeatability, avgPIC in that
order (method='best') or at random (method='random').
The filter has not been implemented for tag presence/absence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.secondaries(x, method = "random", verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.secondaries_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.secondaries_+3A_method">method</code></td>
<td>
<p>Method of selecting SNP locus to retain, 'best' or 'random'
[default 'random'].</p>
</td></tr>
<tr><td><code id="gl.filter.secondaries_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The genlight object, with the secondary SNP loci removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched filter: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.report.secondaries(testset.gl)
result &lt;- gl.filter.secondaries(testset.gl)
</code></pre>

<hr>
<h2 id='gl.filter.taglength'>Filters loci in a genlight {adegenet} object based on sequence tag
length
@family matched filter</h2><span id='topic+gl.filter.taglength'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT typically have sequence tag lengths ranging
from 20 to 69 base pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.taglength(x, lower = 20, upper = 69, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.taglength_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_lower">lower</code></td>
<td>
<p>Lower threshold value below which loci will be removed
[default 20].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_upper">upper</code></td>
<td>
<p>Upper threshold value above which loci will be removed
[default 69].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object retaining loci with a sequence tag length
in the range specified by the lower and upper threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.taglength(testset.gl)
  result &lt;- gl.filter.taglength(testset.gl,lower=60)
  gl.report.taglength(result)
# Tag P/A data
  gl.report.taglength(testset.gs)
  result &lt;- gl.filter.taglength(testset.gs,lower=60)
  gl.report.taglength(result)
  
  
</code></pre>

<hr>
<h2 id='gl.fixed.diff'>Generates a matrix of fixed differences and associated statistics for
populations taken pairwise</h2><span id='topic+gl.fixed.diff'></span>

<h3>Description</h3>

<p>This script takes SNP data or sequence tag P/A data grouped into populations
in a genlight object (DArTSeq) and generates a matrix of fixed differences
between populations taken pairwise
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fixed.diff(
  x,
  tloc = 0,
  test = FALSE,
  delta = 0.02,
  alpha = 0.05,
  reps = 1000,
  mono.rm = TRUE,
  pb = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fixed.diff_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes or tag P/A data
(SilicoDArT) or an object of class 'fd' [required].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_tloc">tloc</code></td>
<td>
<p>Threshold defining a fixed difference (e.g. 0.05 implies 95:5 vs
5:95 is fixed) [default 0].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_test">test</code></td>
<td>
<p>If TRUE, calculate p values for the observed fixed differences
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_delta">delta</code></td>
<td>
<p>Threshold value for the true population minor allele frequency
(MAF) from which resultant sample fixed differences are considered true
positives [default 0.02].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_alpha">alpha</code></td>
<td>
<p>Level of significance used to display non-significant
differences between populations as they are compared pairwise [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_reps">reps</code></td>
<td>
<p>Number of replications to undertake in the simulation to estimate
probability of false positives [default 1000].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, loci that are monomorphic across all individuals are
removed before beginning computations [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_pb">pb</code></td>
<td>
<p>If TRUE, show a progress bar on time consuming loops
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A fixed difference at a locus occurs when two populations share no alleles or
where all members of one population has a sequence tag scored, and all
members of the other population has the sequence tag absent. The challenge
with this approach is that when sample sizes are finite, fixed differences
will occur through sampling error, compounded when many loci are examined.
Simulations suggest that sample sizes of n1=5 and n2=5 are adequate to reduce
the probability of [experiment-wide] type 1 error to negligible levels
[ploidy=2]. A warning is issued if comparison between two populations
involves sample sizes less than 5, taking into account allele drop-out.
Optionally, if test=TRUE, the script will test the fixed differences between
final OTUs for statistical significance, using simulation, and then further
amalgamate populations that for which there are no significant fixed
differences at a specified level of significance (alpha). To avoid conflation
of true fixed differences with false positives in the simulations, it is
necessary to decide a threshold value (delta) for extreme true allele
frequencies that will be considered fixed for practical purposes. That is,
fixed differences in the sample set will be considered to be positives (not
false positives) if they arise from true allele frequencies of less than
1-delta in one or both populations.  The parameter delta is typically set to
be small (e.g. delta = 0.02).
NOTE: The above test will only be calculated if tloc=0, that is, for analyses
of absolute fixed differences. The test applies in comparisons of allopatric
populations only. For sympatric populations, use gl.pval.sympatry().
An absolute fixed difference is as defined above. However, one might wish to
score fixed differences at some lower level of allele frequency difference,
say where percent allele frequencies are 95,5 and 5,95 rather than 100:0 and
0:100. This adjustment can be done with the tloc parameter. For example,
tloc=0.05 means that SNP allele frequencies of 95,5 and 5,95 percent will be
regarded as fixed when comparing two populations at a locus.
</p>


<h3>Value</h3>

<p>A list of Class 'fd' containing the gl object and square matrices,
as follows:
</p>

<ol>
<li><p> $gl &ndash; the output genlight object;
</p>
</li>
<li><p> $fd &ndash; raw fixed differences;
</p>
</li>
<li><p> $pcfd &ndash; percent fixed differences;
</p>
</li>
<li><p> $nobs &ndash; mean no. of individuals used in each comparison;
</p>
</li>
<li><p> $nloc &ndash; total number of loci used in each comparison;
</p>
</li>
<li><p> $expfpos &ndash; if test=TRUE, the expected count of false positives
for each comparison [by simulation];
</p>
</li>
<li><p> $sdfpos &ndash; if test=TRUE, the standard deviation of the count of
false positives for each comparison [by simulation];
</p>
</li>
<li><p> $prob &ndash; if test=TRUE, the significance of the count of fixed
differences [by simulation])
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utils.is.fixed">utils.is.fixed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fd &lt;- gl.fixed.diff(testset.gl, tloc=0, verbose=3 )
fd &lt;- gl.fixed.diff(testset.gl, tloc=0, test=TRUE, delta=0.02, reps=100, verbose=3 )


</code></pre>

<hr>
<h2 id='gl.fst.pop'>Calculates a pairwise Fst values for populations in a genlight object
This script calculates pairwise Fst values based on the implementation in the
StAMPP package (?stamppFst). It allows to run bootstrap to estimate
probability of Fst values to be different from zero. For detailed information
please check the help pages (?stamppFst).</h2><span id='topic+gl.fst.pop'></span>

<h3>Description</h3>

<p>Calculates a pairwise Fst values for populations in a genlight object
This script calculates pairwise Fst values based on the implementation in the
StAMPP package (?stamppFst). It allows to run bootstrap to estimate
probability of Fst values to be different from zero. For detailed information
please check the help pages (?stamppFst).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fst.pop(x, nboots = 1, percent = 95, nclusters = 1, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fst.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_nboots">nboots</code></td>
<td>
<p>Number of bootstraps to perform across loci to generate
confidence intervals and p-values [default 1].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_percent">percent</code></td>
<td>
<p>Percentile to calculate the confidence interval around
[default 95].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_nclusters">nclusters</code></td>
<td>
<p>Number of processor threads or cores to use during
calculations [default 1].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of distances between populations (class dist), if nboots =1,
otherwise a list with Fsts (in a matrix), Pvalues (a matrix of pvalues),
Bootstraps results (data frame of all runs). Hint: Use
<code>as.matrix(as.dist(fsts))</code> if you want to have a squared matrix with
symmetric entries returned, instead of a dist object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
test &lt;- gl.filter.monomorphs(test)
out &lt;- gl.fst.pop(test, nboots=1)
</code></pre>

<hr>
<h2 id='gl.He'>Estimates expected Heterozygosity</h2><span id='topic+gl.He'></span>

<h3>Description</h3>

<p>Estimates expected Heterozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.He(gl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.He_+3A_gl">gl</code></td>
<td>
<p>A genlight object [required]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple vector whit Ho for each loci
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.Ho'>Estimates observed Heterozygosity</h2><span id='topic+gl.Ho'></span>

<h3>Description</h3>

<p>Estimates observed Heterozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.Ho(gl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.Ho_+3A_gl">gl</code></td>
<td>
<p>A genlight object [required]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple vector whit Ho for each loci
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.hwe.pop'>Performs Hardy-Weinberg tests over loci and populations</h2><span id='topic+gl.hwe.pop'></span>

<h3>Description</h3>

<p>Hardy-Weinberg tests are performed for each loci in each of the populations
as defined by the pop slot in a genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.hwe.pop(
  x,
  alpha_val = 0.05,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = c("gray90", "deeppink"),
  HWformat = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.hwe.pop_+3A_x">x</code></td>
<td>
<p>A genlight object with a population defined
[pop(x) does not return NULL].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, returns a plot object compatible with ggplot,
otherwise returns a dataframe [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default gl.colors(2)].
[default gl.colors(&quot;dis&quot;)].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_hwformat">HWformat</code></td>
<td>
<p>Switch if data should be returned in HWformat (counts of
Genotypes to be used in package <code>HardyWeinberg</code>)</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function employs the <code>HardyWeinberg</code> package, which needs to be
installed. The function that is used is
<code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code>, but there are several other great
functions implemented in the package regarding HWE. Therefore, this function
can return the data in the format expected by the HWE package expects, via
<code>HWformat=TRUE</code> and then use this to run other functions of the package.
This functions performs a HWE test for every population (rows) and loci
(columns) and returns a true false matrix. True is reported if the p-value of
an HWE-test for a particular loci and population was below the specified
threshold (alpha_val, default=0.05). The thinking behind this approach is
that loci that are not in HWE in several populations have most likely to be
treated (e.g. filtered if loci under selection are of interest). If plot=TRUE
a barplot on the loci and the sum of deviation over all population is
returned. Loci that deviate in the majority of populations can be identified
via colSums on the resulting matrix.
Plot themes can be obtained from </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Resultant ggplots and the tabulation are saved to the session's temporary
directory.
</p>


<h3>Value</h3>

<p>The function returns a list with up to three components:
</p>

<ul>
<li><p> 'HWE' is the matrix over loci and populations
</p>
</li>
<li><p> 'plot' is a plot (ggplot) which shows the significant results
for population and loci (can be amended further using ggplot syntax)
</p>
</li>
<li><p> 'HWEformat=TRUE' the 'HWformat' entails SNP data for each population
in 'HardyWeinberg'-format to be used with other functions of the package
(e.g <code><a href="HardyWeinberg.html#topic+HWPerm">HWPerm</a></code> or
<code><a href="HardyWeinberg.html#topic+HWExactPrevious">HWExactPrevious</a></code>).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.hwe.pop(bandicoot.gl[,1:33], alpha_val=0.05, plot.out=TRUE, HWformat=FALSE)
</code></pre>

<hr>
<h2 id='gl.impute'>Imputes missing data</h2><span id='topic+gl.impute'></span>

<h3>Description</h3>

<p>This function imputes genotypes on a population-by-population basis, where
populations can be considered panmictic, or imputes the state for
presence-absence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.impute(
  x,
  method = "neighbour",
  fill.residual = TRUE,
  parallel = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.impute_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence-absence
data [required].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_method">method</code></td>
<td>
<p>Imputation method, either &quot;frequency&quot; or &quot;HW&quot; or &quot;neighbour&quot; 
or &quot;random&quot; [default &quot;neighbour&quot;].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_fill.residual">fill.residual</code></td>
<td>
<p>Should any residual missing values remaining after 
imputation be set to 0, 1, 2 at random, taking into account global allele 
frequencies at the particular locus [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_parallel">parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We recommend that imputation be performed on sampling locations, before
any aggregation. Imputation is achieved by replacing missing values using
either of two methods:
</p>

<ul>
<li><p> If &quot;frequency&quot;, genotypes scored as missing at a locus in an individual
are imputed using the average allele frequencies at that locus in the 
population from which the individual was drawn.
</p>
</li>
<li><p> If &quot;HW&quot;, genotypes scored as missing at a locus in an individual are 
imputed by sampling at random assuming Hardy-Weinberg equilibrium. Applies 
only to genotype data.
</p>
</li>
<li><p> If &quot;neighbour&quot;, substitute the missing values for the focal individual
with the values taken from the nearest neighbour. Repeat with next nearest
and so on until all missing values are replaced.
</p>
</li>
<li><p> if &quot;random&quot;, missing data are substituted by random values (0, 1 or 2). 
</p>
</li></ul>

<p>The nearest neighbour is the one with the smallest Euclidean distance in 
all the dataset.
The advantage of this approach is that it works regardless of how many
individuals are in the population to which the focal individual belongs,
and the displacement of the individual is haphazard as opposed to:
(a) Drawing the individual toward the population centroid (HW and Frequency).
(b) Drawing the individual toward the global centroid (glPCA).
Note that loci that are missing for all individuals in a population are not 
imputed with method 'frequency' or 'HW'. Consider using the function 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> with by.pop=TRUE to remove them first.
</p>


<h3>Value</h3>

<p>A genlight object with the missing data imputed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
require("dartR.data")
# SNP genotype data
gl &lt;- gl.filter.callrate(platypus.gl,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gl &lt;- gl.impute(gl,method="neighbour")
# Sequence Tag presence-absence data
gs &lt;- gl.filter.callrate(testset.gs,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gs &lt;- gl.impute(gs, method="neighbour")

gs &lt;- gl.impute(platypus.gl,method ="random")

</code></pre>

<hr>
<h2 id='gl.join'>Combines two dartR genlight objects</h2><span id='topic+gl.join'></span>

<h3>Description</h3>

<p>This function combines two genlight objects and their associated metadata.
The history associated with the two genlight objects is cleared from the new
genlight object. The individuals/samples must be the same in each genlight
object.
The function is typically used to combine datasets from the same service
where the files have been split because of size limitations. The data is read
in from multiple csv files, then the resultant genlight objects are combined.
This function works with both SNP and Tag P/A data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.join(x1, x2, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.join_+3A_x1">x1</code></td>
<td>
<p>Name of the first genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.join_+3A_x2">x2</code></td>
<td>
<p>Name of the first genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.join_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 &lt;- testset.gl[,1:100]
x1@other$loc.metrics &lt;-  testset.gl@other$loc.metrics[1:100,]
nLoc(x1)
x2 &lt;- testset.gl[,101:150]
x2@other$loc.metrics &lt;-  testset.gl@other$loc.metrics[101:150,]
nLoc(x2)
gl &lt;- gl.join(x1, x2, verbose=2)
nLoc(gl)

</code></pre>

<hr>
<h2 id='gl.keep.ind'>Removes all but the specified individuals from a dartR genlight object</h2><span id='topic+gl.keep.ind'></span>

<h3>Description</h3>

<p>This script deletes all individuals apart from those listed (ind.list).
Monomorphic loci and loci that are scored all NA are optionally deleted (mono.rm=TRUE). 
The script also optionally recalculates locus metatdata statistics to accommodate
the deletion of individuals from the dataset (recalc=TRUE).
The script returns a dartR genlight object with the retained individuals 
and the recalculated locus metadata. The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.ind(x, ind.list, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_ind.list">ind.list</code></td>
<td>
<p>A list of individuals to be retained [required].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic and all NA loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reduced dartR genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.pop">gl.drop.pop</a></code> to drop rather than keep specified populations
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # SNP data
    gl2 &lt;- gl.keep.ind(testset.gl, ind.list=c('AA019073','AA004859'))
  # Tag P/A data
   gs2 &lt;- gl.keep.ind(testset.gs, ind.list=c('AA020656','AA19077','AA004859'))
   
</code></pre>

<hr>
<h2 id='gl.keep.loc'>Removes all but the specified loci from a genlight object</h2><span id='topic+gl.keep.loc'></span>

<h3>Description</h3>

<p>This function deletes loci that are not specified to keep, and their associated metadata. 
The script returns a dartR genlight object with the retained loci. 
The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.loc(x, loc.list = NULL, first = NULL, last = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.loc_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_loc.list">loc.list</code></td>
<td>
<p>A list of loci to be kept
[required, if loc.range not specified].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_first">first</code></td>
<td>
<p>First of a range of loci to be kept
[required, if loc.list not specified].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_last">last</code></td>
<td>
<p>Last of a range of loci to be kept
[if not specified, last locus in the dataset].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.loc">gl.drop.loc</a></code> to drop rather than keep specified loci
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl2 &lt;- gl.keep.loc(testset.gl, loc.list=c('100051468|42-A/T', '100049816-51-A/G'))
# Tag P/A data
  gs2 &lt;- gl.keep.loc(testset.gs, loc.list=c('20134188','19249144'))
  
</code></pre>

<hr>
<h2 id='gl.keep.pop'>Removes all but the specified populations from a dartR genlight object</h2><span id='topic+gl.keep.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on associated specimen metadata
stored in the dartR genlight object. 
This script deletes all individuals apart from those in listed populations (pop.list).
Monomorphic loci and loci that are scored all NA are optionally deleted (mono.rm=TRUE). 
The script also optionally recalculates locus metatdata statistics to accommodate
the deletion of individuals from the dataset (recalc=TRUE).
The script returns a dartR genlight object with the retained populations 
and the recalculated locus metadata. The script works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.pop(
  x,
  pop.list,
  as.pop = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_pop.list">pop.list</code></td>
<td>
<p>List of populations to be retained [required].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Temporarily assign another locus metric as the population for
the purposes of deletions [default NULL].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic and all NA loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A reduced dartR genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.pop">gl.drop.pop</a></code> to drop rather than keep specified populations
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('EmsubRopeMata', 'EmvicVictJasp'))
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('EmsubRopeMata', 'EmvicVictJasp'),
   mono.rm=TRUE,recalc=TRUE)
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('Female'),as.pop='sex')
 # Tag P/A data
   gs2 &lt;- gl.keep.pop(testset.gs, pop.list=c('EmsubRopeMata','EmvicVictJasp'))
   
</code></pre>

<hr>
<h2 id='gl.load'>Loads an object from compressed binary format produced by gl.save()</h2><span id='topic+gl.load'></span>

<h3>Description</h3>

<p>This is a wrapper for readRDS()
The function loads the object from the current workspace and returns the
gl object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.load(file, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.load_+3A_file">file</code></td>
<td>
<p>Name of the file to receive the binary version of the object
[required].</p>
</td></tr>
<tr><td><code id="gl.load_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The loaded object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.save">gl.save</a></code>
</p>
<p>Other io: 
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.save(testset.gl,file.path(tempdir(),'testset.rds'))
gl &lt;- gl.load(file.path(tempdir(),'testset.rds'))
</code></pre>

<hr>
<h2 id='gl.mahal.assign'>Assigns individuals to populations with an associated probability</h2><span id='topic+gl.mahal.assign'></span>

<h3>Description</h3>

<p>Uses Mahalanobis Distances between individuals and group centroids to
calculate probability of group membership
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.mahal.assign(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.mahal.assign_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.mahal.assign_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function generates 200 simulated individuals for each group (=population in the
genlight object) drawing from the observed allele frequencies for each group.
The group centroids and covariance matricies are calculated for these simulated
groups. The covariance matrix is inverted using package MASS::ginv to overcome the
singularities that would otherwise arise with typical SNP data. Mahanobilis
Distances are calculated using stats::mahanalobis for each individual in the
dataset and associated Chi Square probabilities of group membership are calculated for
each individual in the original genlight object. The resultant table can be
used for decisions on group membership. A special group (=population in the
genlight object) called 'unknowns' can be used to specifically identify 
individuals with unknown group membership.
</p>
<p>A color vector can be obtained with gl.select.colors() and then passed to the function
with the plot.colors parameter.
Themes can be obtained from in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir().
</p>


<h3>Value</h3>

<p>The unchanged genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched reports: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.factorloadings">gl.report.factorloadings</a>()</code>,
<code><a href="#topic+gl.report.fstat">gl.report.fstat</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>
</p>

<hr>
<h2 id='gl.make.recode.ind'>Creates a proforma recode_ind file for reassigning individual
(=specimen) names</h2><span id='topic+gl.make.recode.ind'></span>

<h3>Description</h3>

<p>Renaming individuals may be required when there have been errors in labeling
arising in the process from sample to sequencing files. There may be occasions
where renaming individuals is required for preparation of figures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.make.recode.ind(
  x,
  out.recode.file = "default_recode_ind.csv",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.make.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>File name of the output file (including extension)
[default default_recode_ind.csv].</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_outpath">outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function facilitates the construction of a recode table by producing a
proforma file with current individual (=specimen) names in two identical
columns. Edit the second column to reassign individual names. Use keyword
'Delete' to delete an individual.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a clear record of the changes.
Use outpath=getwd() or when calling this function to direct output files 
to your working directory.
The function works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
Apply the recoding using gl.recode.ind().
</p>


<h3>Value</h3>

<p>A vector containing the new individual names.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.make.recode.ind(testset.gl, out.recode.file ='Emmac_recode_ind.csv',outpath=tempdir())
</code></pre>

<hr>
<h2 id='gl.make.recode.pop'>Creates a proforma recode_pop_table file for reassigning population
names
@family data manipulation</h2><span id='topic+gl.make.recode.pop'></span>

<h3>Description</h3>

<p>Renaming populations may be required when there have been errors in
assignment arising in the process from sample to sequence files or when one
wishes to amalgamate populations, or delete populations. Recoding populations
can also be done with a recode table (csv).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.make.recode.pop(
  x,
  out.recode.file = "recode_pop_table.csv",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.make.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>File name of the output file (including extension)
[default recode_pop_table.csv].</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_outpath">outpath</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function facilitates the construction of a recode table by producing a
proforma file with current population names in two identical columns. Edit
the second column to reassign populations. Use keyword 'Delete' to delete a
population.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a clear record of the changes.
Use outpath=getwd() or when calling this function to direct output files 
to your working directory.
The function works with both genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
Apply the recoding using gl.recode.pop().
</p>


<h3>Value</h3>

<p>A vector containing the new population names.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.make.recode.pop(testset.gl,out.recode.file='test.csv',outpath=tempdir(),verbose=2)
</code></pre>

<hr>
<h2 id='gl.map.interactive'>Creates an interactive map (based on latlon) from a genlight object</h2><span id='topic+gl.map.interactive'></span>

<h3>Description</h3>

<p>Creates an interactive map (based on latlon) from a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.map.interactive(
  x,
  matrix = NULL,
  standard = TRUE,
  symmetric = TRUE,
  pop.labels = TRUE,
  pop.labels.cex = 12,
  ind.circles = TRUE,
  ind.circle.cols = NULL,
  ind.circle.cex = 10,
  ind.circle.transparency = 0.8,
  palette.links = NULL,
  legend.title = NULL,
  provider = "Esri.NatGeoWorldMap",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.map.interactive_+3A_x">x</code></td>
<td>
<p>A genlight object (including coordinates within the latlon slot) 
[required].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_matrix">matrix</code></td>
<td>
<p>A distance matrix between populations or individuals. The
matrix is visualised as lines between individuals/populations. If matrix is
asymmetric two lines with arrows are plotted [default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_standard">standard</code></td>
<td>
<p>If a matrix is provided line width will be standardised to be
between 1 to 10, if set to true, otherwise taken as given [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_symmetric">symmetric</code></td>
<td>
<p>If a symmetric matrix is provided only one line is drawn
based on the lower triangle of the matrix. If set to false arrows indicating
the direction are used instead [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_pop.labels">pop.labels</code></td>
<td>
<p>Population labels at the center of the individuals of
populations [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_pop.labels.cex">pop.labels.cex</code></td>
<td>
<p>Size of population labels [default 12].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circles">ind.circles</code></td>
<td>
<p>Should individuals plotted as circles [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.cols">ind.circle.cols</code></td>
<td>
<p>Colors of circles. Colors can be provided as usual by 
names (e.g. &quot;black&quot;) and are re-cycled. So a color c(&quot;blue&quot;,&quot;red&quot;) colors 
individuals alternatively between blue and red using the genlight object
order of individuals. For transparency see parameter 
ind.circle.transparency. Defaults to rainbow colors by population  if not
provided. If you want to have your own colors for each population, check
the platypus.gl example below.</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.cex">ind.circle.cex</code></td>
<td>
<p>(size or circles in pixels ) [default 10].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.transparency">ind.circle.transparency</code></td>
<td>
<p>Transparency of circles between 0=invisible 
and 1=no transparency. Defaults to 0.8.</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_palette.links">palette.links</code></td>
<td>
<p>Color palette for the links in case a matrix is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_legend.title">legend.title</code></td>
<td>
<p>Legend's title for the links in case a matrix is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_provider">provider</code></td>
<td>
<p>Passed to leaflet [default &quot;Esri.NatGeoWorldMap&quot;].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A wrapper around the <span class="pkg">leaflet</span> package. For possible background 
maps check as specified via the provider:
<a href="http://leaflet-extras.github.io/leaflet-providers/preview/index.html">http://leaflet-extras.github.io/leaflet-providers/preview/index.html</a>
The palette.links argument can be any of the following:
A character vector of RGB or named colors. Examples: palette(), 
c(&quot;#000000&quot;, &quot;#0000FF&quot;, &quot;#FFFFFF&quot;), topo.colors(10)
The name of an RColorBrewer palette, e.g. &quot;BuPu&quot; or &quot;Greens&quot;.
The full name of a viridis palette: &quot;viridis&quot;, &quot;magma&quot;, &quot;inferno&quot;, 
or &quot;plasma&quot;.
A function that receives a single value between 0 and 1 and returns a color.
Examples: colorRamp(c(&quot;#000000&quot;, &quot;#FFFFFF&quot;), interpolate = &quot;spline&quot;).
</p>


<h3>Value</h3>

<p>plots a map
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &ndash; Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
gl.map.interactive(bandicoot.gl)
cols &lt;- c("red","blue","yellow")[as.numeric(pop(platypus.gl))]
gl.map.interactive(platypus.gl, ind.circle.cols=cols, ind.circle.cex=10, 
ind.circle.transparency=0.5)

</code></pre>

<hr>
<h2 id='gl.merge.pop'>Merges two or more populations in a dartR genlight object into one population</h2><span id='topic+gl.merge.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart().
This function assigns individuals from two nominated populations into a new
single population. It can also be used to rename populations.
The function works with both SNP and Tag P/A (silicoDArT) data.
The function returns a genlight object with the new population assignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.merge.pop(x, old = NULL, new = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.merge.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_old">old</code></td>
<td>
<p>A list of populations to be merged [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_new">new</code></td>
<td>
<p>Name of the new population [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the new population assignments.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   gl &lt;- gl.merge.pop(testset.gl, old=c('EmsubRopeMata','EmvicVictJasp'), new='Outgroup')
   
</code></pre>

<hr>
<h2 id='gl.pcoa'>Ordination applied to genotypes in a genlight object (PCA), in an fd
object, or to a distance matrix (PCoA)</h2><span id='topic+gl.pcoa'></span>

<h3>Description</h3>

<p>This function takes the genotypes for individuals and undertakes a Pearson
Principal Component analysis (PCA) on SNP or Sequence tag P/A (SilicoDArT) data; it
undertakes a Gower Principal Coordinate analysis (PCoA) if supplied with a
distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.pcoa(
  x,
  nfactors = 5,
  correction = NULL,
  mono.rm = TRUE,
  parallel = FALSE,
  n.cores = 1,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = gl.colors(2),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.pcoa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object or fd object containing the SNP data, or
a distance matrix of type dist [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_nfactors">nfactors</code></td>
<td>
<p>Number of axes to retain in the output of factor scores
[default 5].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_correction">correction</code></td>
<td>
<p>Method applied to correct for negative eigenvalues, either
'lingoes' or 'cailliez' [Default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_parallel">parallel</code></td>
<td>
<p>TRUE if parallel processing is required (does fail under
Windows) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores to use if parallel processing is requested
[default 16].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, a diagnostic plot is displayed showing a scree plot
for the &quot;informative&quot; axes and a histogram of eigenvalues of the remaining 
&quot;noise&quot; axes [Default TRUE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plot [default gl.colors(2)].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_verbose">verbose</code></td>
<td>
<p>verbose= 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is essentially a wrapper for glPca {adegenet} or pcoa {ape}
with default settings apart from those specified as parameters in this
function.
<strong> Sources of stress in the visual representation </strong>
While, technically, any distance matrix can be represented in an ordinated
space, the representation will not typically be exact.There are three major
sources of stress in a reduced-representation of distances or dissimilarities
among entities using PCA or PCoA. By far the greatest source comes from the
decision to select only the top two or three axes from the ordinated set of
axes derived from the PCA or PCoA. The representation of the entities such a
heavily reduced space will not faithfully represent the distances in the
input distance matrix simply because of the loss of information in deeper
informative dimensions. For this reason, it is not sensible to be too
precious about managing the other two sources of stress in the visual
representation.
The measure of distance between entities in a PCA is the Pearson Correlation
Coefficient, essentially a standardized Euclidean distance. This is both a
metric distance and a Euclidean distance and so the distances in the final 
ordination are faithful to those between entities in
the dataset. Note that missing values are imputed in PCA, and that this can be
a source of disparity between the distances between the entities in the dataset
and the distances in the ordinated space.
</p>
<p>In PCoA, the second source of stress is the choice of distance 
measure or dissimilarity measure. While pretty much any distance or dissimilarity matrix 
can be represented in an ordinated space, the distances between entities can 
be faithfully represented in that space (that is, without stress) only if the 
distances are metric. Furthermore, for distances between entities to be 
faithfully represented in a rigid Cartesian space, the distance measure 
needs to be Euclidean. If this is not the case, the distances between the 
entities in the ordinated visualized space will not #' exactly represent 
the distances in the input matrix (stress will be non-zero).
This source of stress will be evident as negative eigenvalues in the deeper
dimensions. 
</p>
<p>A third source of stress arises from having a sparse dataset, one with
missing values. This affects both PCA and PCoA. If the original data matrix
is not fully populated, that is, if there are missing values, then even a
Euclidean distance matrix will not necessarily be 'positive definite'. It
follows that some of the eigenvalues may be negative, even though the
distance metric is Euclidean. This issue is exacerbated when the number of
loci greatly exceeds the number of individuals, as is typically the case when
working with SNP data. The impact of missing values can be minimized by
stringently filtering on Call Rate, albeit with loss of data. An alternative
is given in a paper 'Honey, I shrunk the sample covariance matrix' and more
recently by Ledoit and Wolf (2018), but their approach has not been
implemented here.
</p>
<p>Options for imputing missing values while minimizing distortion are provided in 
the function gl.impute().
The good news is that, unless the sum of the negative eigenvalues, arising
from a non-Euclidean distance measure or from missing values, approaches
those of the final PCA or PCoA axes to be displayed, the distortion is
probably of no practical consequence and certainly not comparable to the
stress arising from selecting only two or three final dimensions out of
several informative dimensions for the visual representation.
<strong> Function's output </strong>
Two diagnostic plots are produced. The first is a Scree Plot, showing the
percentage variation explained by each of the PCA or PCoA axes, for those
axes are considered informative. The scree plot informs a decision on the number of
dimensions to be retained in the visual summaries. Various approaches are available
for identifying which axes are informative (in terms of containing biologically
significant variation) and which are noise axes. The simplest method is to consider
only those axes that explain more variance than the original variables on average as
being informative (the Kaiser-Guttman criterion).
</p>
<p>Once you have the informative axes identified, a judgement call is made as to how
many dimensions to retain and present as results. This requires a decision on how 
much information on structure in the data is to be discarded. Retaining at least those axes 
that explain 10
The second graph is for diagnostic purposes only. It shows the distribution of 
eigenvalues for the remaining uninformative (noise) axes, including those with 
negative eigenvalues. If a plot.file is given, the ggplot arising from this function 
is saved as an binary file using gl.save(); can be reloaded with gl.load(). 
A file name must be  specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir(). 
Action is recommended (verbose &gt;= 2) if the negative eigenvalues are
dominant, their sum approaching in magnitude the eigenvalues for axes
selected for the final visual solution.
Output is a glPca object conforming to adegenet::glPca but with only the
following retained.
</p>

<ul>
<li><p>  $call - The call that generated the PCA/PCoA
</p>
</li>
<li><p>  $eig - Eigenvalues &ndash; All eigenvalues (positive, null, negative).
</p>
</li>
<li><p>  $scores - Scores (coefficients) for each individual
</p>
</li>
<li><p>  $loadings - Loadings of each SNP for each principal component
</p>
</li></ul>

<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>PCA was developed by Pearson (1901) and Hotelling (1933), whilst the best
modern reference is Jolliffe (2002). PCoA was developed by Gower (1966) while
the best modern reference is Legendre &amp; Legendre (1998).
</p>


<h3>Value</h3>

<p>An object of class pcoa containing the eigenvalues and factor scores
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Cailliez, F. (1983) The analytical solution of the additive constant
problem. Psychometrika, 48, 305-308.
</p>
</li>
<li><p> Gower, J. C. (1966) Some distance properties of latent root and vector
methods used in multivariate analysis. Biometrika, 53, 325-338.
</p>
</li>
<li><p> Hotelling, H., 1933. Analysis of a complex of statistical variables into
Principal Components. Journal of Educational Psychology 24:417-441, 498-520.
</p>
</li>
<li><p> Jolliffe, I. (2002) Principal Component Analysis. 2nd Edition, Springer,
New York.
</p>
</li>
<li><p> Ledoit, O. and Wolf, M. (2018). Analytical nonlinear shrinkage of
large-dimensional covariance matrices. University of Zurich, Department of
Economics, Working Paper No. 264, Revised version. Available at SSRN:
https://ssrn.com/abstract=3047302 or http://dx.doi.org/10.2139/ssrn.3047302
</p>
</li>
<li><p> Legendre, P. and Legendre, L. (1998). Numerical Ecology, Volume 24, 2nd
Edition. Elsevier Science, NY.
</p>
</li>
<li><p> Lingoes, J. C. (1971) Some boundary conditions for a monotone analysis
of symmetric matrices. Psychometrika, 36, 195-203.
</p>
</li>
<li><p> Pearson, K. (1901). On lines and planes of closest fit to systems of
points in space. Philosophical Magazine. Series 6, vol. 2, no. 11, pp.
559-572.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.pcoa.plot">gl.pcoa.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># PCA (using SNP genlight object)
gl &lt;- possums.gl
pca &lt;- gl.pcoa(possums.gl[1:50,],verbose=2)
gl.pcoa.plot(pca,gl)

gs &lt;- testset.gs
levels(pop(gs))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',6),'Em.subglobosa','Em.victoriae')
# PCA (using SilicoDArT genlight object)
pca &lt;- gl.pcoa(gs)
gl.pcoa.plot(pca,gs)
# Using a distance matrix
D &lt;- gl.dist.ind(testset.gs, method='jaccard')
pcoa &lt;- gl.pcoa(D,correction="cailliez")
gl.pcoa.plot(pcoa,gs)

</code></pre>

<hr>
<h2 id='gl.pcoa.plot'>Bivariate or trivariate plot of the results of an ordination generated
using gl.pcoa()</h2><span id='topic+gl.pcoa.plot'></span>

<h3>Description</h3>

<p>This script takes output from the ordination generated by gl.pcoa() and plots
the individuals classified by population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.pcoa.plot(
  glPca,
  x,
  scale = FALSE,
  ellipse = FALSE,
  plevel = 0.95,
  pop.labels = "pop",
  interactive = FALSE,
  as.pop = NULL,
  hadjust = 1.5,
  vadjust = 1,
  xaxis = 1,
  yaxis = 2,
  zaxis = NULL,
  pt.size = 2,
  pt.colors = NULL,
  pt.shapes = NULL,
  label.size = 1,
  axis.label.size = 1.5,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.pcoa.plot_+3A_glpca">glPca</code></td>
<td>
<p>Name of the PCA or PCoA object containing the factor scores and
eigenvalues [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object or fd object containing the SNP
genotypes or Tag P/A (SilicoDArT) genotypes or the Distance Matrix used to
generate the ordination [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_scale">scale</code></td>
<td>
<p>If TRUE, scale the x and y axes in proportion to % variation
explained [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_ellipse">ellipse</code></td>
<td>
<p>If TRUE, display ellipses to encapsulate points for each
population [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_plevel">plevel</code></td>
<td>
<p>Value of the percentile for the ellipse to encapsulate points
for each population [default 0.95].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pop.labels">pop.labels</code></td>
<td>
<p>How labels will be added to the plot
['none'|'pop'|'legend', default = 'pop'].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_interactive">interactive</code></td>
<td>
<p>If TRUE then the populations are plotted without labels,
mouse-over to identify points [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_as.pop">as.pop</code></td>
<td>
<p>Assign another metric to represent populations for the plot
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_hadjust">hadjust</code></td>
<td>
<p>Horizontal adjustment of label position in 2D plots
[default 1.5].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_vadjust">vadjust</code></td>
<td>
<p>Vertical adjustment of label position in 2D plots [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_xaxis">xaxis</code></td>
<td>
<p>Identify the x axis from those available in the ordination
(xaxis &lt;= nfactors) [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_yaxis">yaxis</code></td>
<td>
<p>Identify the y axis from those available in the ordination
(yaxis &lt;= nfactors) [default 2].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_zaxis">zaxis</code></td>
<td>
<p>Identify the z axis from those available in the ordination for a
3D plot (zaxis &lt;= nfactors) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.size">pt.size</code></td>
<td>
<p>Specify the size of the displayed points [default 2].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.colors">pt.colors</code></td>
<td>
<p>Optionally provide a vector of nPop colors
(run gl.select.colors() for color options) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.shapes">pt.shapes</code></td>
<td>
<p>Optionally provide a vector of nPop shapes
(run gl.select.shapes() for shape options) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_label.size">label.size</code></td>
<td>
<p>Specify the size of the point labels [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_axis.label.size">axis.label.size</code></td>
<td>
<p>Specify the size of the displayed axis labels
[default 1.5].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The factor scores are taken from the output of gl.pcoa() and the population
assignments are taken from from the original data file. In the bivariate
plots, the specimens are shown optionally with adjacent labels and enclosing
ellipses. Population labels on the plot are shuffled so as not to overlap
(using package {directlabels}).
This can be a bit clunky, as the labels may be some distance from the points
to which they refer, but it provides the opportunity for moving labels around
using graphics software (e.g. Adobe Illustrator).
3D plotting is activated by specifying a zaxis.
Any pair or trio of axes can be specified from the ordination, provided they
are within the range of the nfactors value provided to gl.pcoa().
In the 2D plots, axes can be scaled to represent the proportion of variation
explained. In any case, the proportion of variation explained by each axis is
provided in the axis label.
Colors and shapes of the points can be altered by passing a vector of shapes
and/or a vector of colors. These vectors can be created with
gl.select.shapes() and gl.select.colors() and passed to this script using the
pt.shapes and pt.colors parameters.
Points displayed in the ordination can be identified if the option
interactive=TRUE is chosen, in which case the resultant plot is ggplotly()
friendly. Identification of points is by moving the mouse over them. Refer
to the plotly package for further information.
The interactive option is automatically enabled for 3D plotting.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.pcoa">gl.pcoa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> test &lt;- gl.pcoa(platypus.gl)
 gl.pcoa.plot(glPca = test, x = platypus.gl)

# SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# RUN PCA
pca&lt;-gl.pcoa(gl,nfactors=5)
# VARIOUS EXAMPLES
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.95, pop.labels='pop', 
axis.label.size=1, hadjust=1.5,vadjust=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend', 
axis.label.size=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend', 
axis.label.size=1.5,scale=TRUE)
gl.pcoa.plot(pca, gl, ellipse=TRUE, axis.label.size=1.2, xaxis=1, yaxis=3, 
scale=TRUE)
gl.pcoa.plot(pca, gl, pop.labels='none',scale=TRUE)
gl.pcoa.plot(pca, gl, axis.label.size=1.2, interactive=TRUE)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, xaxis=1, yaxis=2, zaxis=3)
# color AND SHAPE ADJUSTMENTS
shp &lt;- gl.select.shapes(select=c(16,17,17,0,2))
col &lt;- gl.select.colors(library='brewer',palette='Spectral',ncolors=11,
select=c(1,9,3,11,11))
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.95, pop.labels='pop', 
pt.colors=col, pt.shapes=shp, axis.label.size=1, hadjust=1.5,vadjust=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend',
 pt.colors=col, pt.shapes=shp, axis.label.size=1)
 
</code></pre>

<hr>
<h2 id='gl.plot.heatmap'>Represents a distance matrix as a heatmap</h2><span id='topic+gl.plot.heatmap'></span>

<h3>Description</h3>

<p>The script plots a heat map to represent the distances in the distance or
dissimilarity matrix. This function is a wrapper for
<a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package gplots).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.heatmap(D, palette.divergent = gl.colors("div"), verbose = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.heatmap_+3A_d">D</code></td>
<td>
<p>Name of the distance matrix or class fd object [required].</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_palette.divergent">palette.divergent</code></td>
<td>
<p>A divergent palette for the distance values
[default gl.colors(&quot;div&quot;)].</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_...">...</code></td>
<td>
<p>Parameters passed to function
<a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package gplots)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
   gl &lt;- testset.gl[1:10,]
   D &lt;- dist(as.matrix(gl),upper=TRUE,diag=TRUE)
   gl.plot.heatmap(D)
   D2 &lt;- gl.dist.pop(possums.gl)
   gl.plot.heatmap(D2)
   D3 &lt;- gl.fixed.diff(testset.gl)
   gl.plot.heatmap(D3)
   
   if ((requireNamespace("gplots", quietly = TRUE))) {
   D2 &lt;- gl.dist.pop(possums.gl)
   gl.plot.heatmap(D2)
   }
   
</code></pre>

<hr>
<h2 id='gl.print.history'>Prints history of a genlight object</h2><span id='topic+gl.print.history'></span>

<h3>Description</h3>

<p>Prints history of a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.print.history(x = NULL, history = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.print.history_+3A_x">x</code></td>
<td>
<p>A genlight object (with history) [optional].</p>
</td></tr>
<tr><td><code id="gl.print.history_+3A_history">history</code></td>
<td>
<p>Either a link to a history slot
(gl\@other$history), or a vector indicating which part of the history of x is
used [c(1,3,4) uses the first, third and forth entry from x\@other$history].
If no history is provided the complete history of x is used (recreating the
identical object x) [optional].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a table with all history records. Currently the style cannot
be changed.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other environment: 
<code><a href="#topic+gl.check.verbosity">gl.check.verbosity</a>()</code>,
<code><a href="#topic+gl.check.wd">gl.check.wd</a>()</code>,
<code><a href="#topic+gl.set.wd">gl.set.wd</a>()</code>,
<code><a href="#topic+theme_dartR">theme_dartR</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dartfile &lt;- system.file('extdata','testset_SNPs_2Row.csv', package='dartR.data')
metadata &lt;- system.file('extdata','testset_metadata.csv', package='dartR.data')
gl &lt;- gl.read.dart(dartfile, ind.metafile = metadata, probar=FALSE)
gl2 &lt;- gl.filter.callrate(gl, method='loc', threshold=0.9)
gl3 &lt;- gl.filter.callrate(gl2, method='ind', threshold=0.95)
gl.print.history(gl3)


</code></pre>

<hr>
<h2 id='gl.prop.shared'>Calculates a similarity (distance) matrix for individuals on the proportion of
shared alleles
@family distance</h2><span id='topic+gl.prop.shared'></span><span id='topic+gl.propShared'></span>

<h3>Description</h3>

<p>This script calculates an individual based distance matrix. It uses an C++
implementation, so package Rcpp needs to be installed and it is therefore
really fast (once it has compiled the function after the first run).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.propShared(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.prop.shared_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A similarity matrix
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#takes some time at the first run of the function...

res &lt;- gl.propShared(bandicoot.gl)
res[1:5,1:7] #show only a small part of the matrix

</code></pre>

<hr>
<h2 id='gl.randomize.snps'>Randomly changes the allocation of 0's and 2's in a genlight object</h2><span id='topic+gl.randomize.snps'></span>

<h3>Description</h3>

<p>This function samples randomly half of the SNPs and re-codes, in the sampled
SNP's, 0's by 2's.
</p>
<p>This function samples randomly half of the SNPs and re-codes, in the sampled
SNP's, 0's by 2's.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.randomize.snps(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)

gl.randomize.snps(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.randomize.snps_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.randomize.snps_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default NULL,
unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DArT calls the most common allele as the reference allele. In a genlight
object, homozygous for the reference allele are coded with a '0' and
homozygous for the alternative allele are coded with a '2'. This causes some
distortions in visuals from time to time.
If plot.display = TRUE, two smear plots (pre-randomisation and
post-randomisation) are presented using a random subset of individuals (10)
and loci (100) to provide an overview of the changes.
Resultant ggplots are saved to the session's temporary directory.
</p>
<p>DArT calls the most common allele as the reference allele. In a genlight
object, homozygous for the reference allele are coded with a '0' and
homozygous for the alternative allele are coded with a '2'. This causes some
distortions in visuals from time to time.
If plot.display = TRUE, two smear plots (pre-randomisation and
post-randomisation) are presented using a random subset of individuals (10)
and loci (100) to provide an overview of the changes.
Resultant ggplots are saved to the session's temporary directory.
</p>


<h3>Value</h3>

<p>Returns a genlight object with half of the loci re-coded.
</p>
<p>Returns a genlight object with half of the loci re-coded.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- gl.randomize.snps(platypus.gl[1:5,1:5],verbose = 5)
gl &lt;- gl.filter.monomorphs(testset.gl)
res &lt;- gl.randomize.snps(gl,verbose = 5)
</code></pre>

<hr>
<h2 id='gl.read.csv'>Reads SNP data from a csv file into a genlight object</h2><span id='topic+gl.read.csv'></span>

<h3>Description</h3>

<p>This script takes SNP genotypes from a csv file, combines them with
individual and locus metrics and creates a genlight object.
The SNP data need to be in one of two forms. SNPs can be coded 0 for
homozygous reference, 2 for homozygous alternate, 1 for heterozygous, and NA 
for missing values; or the SNP data can be coded A/A, A/C, C/T, G/A etc,
and -/- for missing data. In this format, the reference allele is the most 
frequent allele, as used by DArT. Other formats will throw an error.
The SNP data need to be individuals as rows, labeled, and loci as columns,
also labeled. If the orientation is individuals as columns and loci by rows,
then set transpose=TRUE.
The individual metrics need to be in a csv file, with headings, with a
mandatory id column corresponding exactly to the individual identity labels
provided with the SNP data and in the same order.
The locus metadata needs to be in a csv file with headings, with a mandatory
column headed AlleleID corresponding exactly to the locus identity labels
provided with the SNP data and in the same order.
Note that the locus metadata will be complemented by calculable statistics
corresponding to those that would be provided by Diversity Arrays Technology
(e.g. CallRate).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.csv(
  filename,
  transpose = FALSE,
  ind.metafile = NULL,
  loc.metafile = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.csv_+3A_filename">filename</code></td>
<td>
<p>Name of the csv file containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_transpose">transpose</code></td>
<td>
<p>If TRUE, rows are loci and columns are individuals
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
individuals [optional].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_loc.metafile">loc.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
loci [optional].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the SNP data and associated metadata included.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>csv_file &lt;- system.file('extdata','platy_test.csv', package='dartR.data')
ind_metadata &lt;- system.file('extdata','platy_ind.csv', package='dartR.data')
gl  &lt;- gl.read.csv(filename = csv_file, ind.metafile = ind_metadata)

</code></pre>

<hr>
<h2 id='gl.read.dart'>Imports DArT data into dartR and converts it into a dartR genlight object</h2><span id='topic+gl.read.dart'></span>

<h3>Description</h3>

<p>This function is a wrapper function that allows you to convert your DArT file
into a genlight object of class dartR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.dart(
  filename,
  ind.metafile = NULL,
  recalc = TRUE,
  mono.rm = FALSE,
  nas = "-",
  topskip = NULL,
  lastmetric = NULL,
  covfilename = NULL,
  service.row = 1,
  plate.row = 3,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.dart_+3A_filename">filename</code></td>
<td>
<p>File containing the SNP data (csv file) [required].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>File that contains additional information on individuals
[required].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, force the recalculation of locus metrics [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, force the removal of monomorphic loci (including all NAs.
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_nas">nas</code></td>
<td>
<p>A character specifying NAs [default '-'].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_topskip">topskip</code></td>
<td>
<p>A number specifying the number of initial rows to be skipped
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Deprecated, specifies the last column of locus metadata. Can be 
specified as a column number [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_covfilename">covfilename</code></td>
<td>
<p>Deprecated, sse ind.metafile parameter [NULL].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_service.row">service.row</code></td>
<td>
<p>The row number for the DArT service
is contained [default 1].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_plate.row">plate.row</code></td>
<td>
<p>The row number the plate well [default 3].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, or as set by gl.set.verbose()].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will determine automatically if the data are in Diversity Arrays
one-row csv format or two-row csv format. 
The first 
row of data is determined from the number of rows with an * in the first 
column. This can be alternatively specified with the topskip parameter.
The DArT service code is added to the ind.metrics of the genlight object. 
The row containing the service code for each individual can be specified with 
the service.row parameter.
#'The DArT plate well is added to the ind.metrics of the genlight object. 
The row containing the plate well for each individual can be specified with 
the plate.row parameter.
If individuals have been deleted from the input file manually, then the locus
metrics supplied by DArT will no longer be correct and some loci may be
monomorphic. To accommodate this, set mono.rm and recalc to TRUE.
</p>


<h3>Value</h3>

<p>A dartR genlight object that contains individual and locus metrics
[if data were provided] and locus metrics [from a DArT report].
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utils.read.dart">utils.read.dart</a></code>
</p>
<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dartfile &lt;- system.file('extdata','testset_SNPs_2Row.csv', package='dartR.data')
metadata &lt;- system.file('extdata','testset_metadata.csv', package='dartR.data')
gl &lt;- gl.read.dart(dartfile, ind.metafile = metadata, probar=TRUE)
</code></pre>

<hr>
<h2 id='gl.read.fasta'>Reads FASTA files and converts them to genlight object</h2><span id='topic+gl.read.fasta'></span>

<h3>Description</h3>

<p>The following IUPAC Ambiguity Codes are taken as heterozygotes:
</p>

<ul>
<li><p> M is heterozygote for	AC and CA	
</p>
</li>
<li><p> R is heterozygote	for AG and GA	
</p>
</li>
<li><p> W is heterozygote	for AT and TA	
</p>
</li>
<li><p> S is heterozygote	for CG and GC
</p>
</li>
<li><p> Y is heterozygote	for CT and TC
</p>
</li>
<li><p> K is heterozygote	for GT and TG
</p>
</li></ul>

<p>The following IUPAC Ambiguity Codes are taken as missing data:
</p>

<ul>
<li><p> V
</p>
</li>
<li><p> H 
</p>
</li>
<li><p> D 
</p>
</li>
<li><p> B
</p>
</li>
<li><p> N 
</p>
</li></ul>

<p>The function can deal with missing data in individuals, e.g. when FASTA 
files have different number of individuals due to missing data.
The allele with the highest frequency is taken as the reference allele.
SNPs with more than two alleles are skipped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.fasta(fasta.files, parallel = FALSE, n.cores = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.fasta_+3A_fasta.files">fasta.files</code></td>
<td>
<p>Fasta files to read [required].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_parallel">parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_n.cores">n.cores</code></td>
<td>
<p>If parallel is TRUE, the number of cores to be used in the
computations; if NULL, then the maximum number of cores available on the
computer is used [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Ambiguity characters are often used to code heterozygotes. However, using
heterozygotes as ambiguity characters may bias many estimates. See more
information in the link below:
<a href="https://evodify.com/heterozygotes-ambiguity-characters/">https://evodify.com/heterozygotes-ambiguity-characters/</a>
</p>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>

<hr>
<h2 id='gl.read.PLINK'>Reads PLINK data file into a genlight object</h2><span id='topic+gl.read.PLINK'></span>

<h3>Description</h3>

<p>This function imports PLINK data into a genlight object and append 
available metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.PLINK(
  filename,
  ind.metafile = NULL,
  loc.metafile = NULL,
  plink.cmd = "plink",
  plink.path = "path",
  plink.flags = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.PLINK_+3A_filename">filename</code></td>
<td>
<p>Fully qualified path to PLINK input file (without including the extension)</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
individuals [optional].</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_loc.metafile">loc.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
loci [optional].</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_plink.cmd">plink.cmd</code></td>
<td>
<p>The 'name' to call plink. This will depend on the file name 
(without the extension '.exe' if on windows) or the name of the PATH variable</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_plink.path">plink.path</code></td>
<td>
<p>The path where the executable is. If plink is listed in
the PATH then there is no need for this. This is what the option &quot;path&quot;
means</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_plink.flags">plink.flags</code></td>
<td>
<p>additional possible parameters passed on to plink.</p>
</td></tr>
<tr><td><code id="gl.read.PLINK_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function handles .ped or .bed file (with the associate files - 
e.g. .fam, .bim). However, if a .ped file is provided, PLINK needs to be 
installed and it is used to convert the .ped into a .bed, which is then
converted into a genlight.
</p>
<p>Additional metadata can be included passing .csv files. These will be appended 
to the existing metadata present in the PLINK files. 
</p>
<p>The locus metadata needs to be in a csv file with headings, with a mandatory
column headed AlleleID corresponding exactly to the locus identity labels
provided with the SNP data
</p>


<h3>Value</h3>

<p>A genlight object with the SNP data and associated metadata included.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='gl.read.silicodart'>Imports presence/absence data from SilicoDArT to genlight {agegenet}
format (ploidy=1)</h2><span id='topic+gl.read.silicodart'></span>

<h3>Description</h3>

<p>DaRT provide the data as a matrix of entities (individual animals) across the
top and attributes (P/A of sequenced fragment) down the side in a format
that is unique to DArT. This program reads the data in to adegenet format
for consistency with other programming activity. The script may require
modification as DArT modify their data formats from time to time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.silicodart(
  filename,
  ind.metafile = NULL,
  nas = "-",
  topskip = NULL,
  lastmetric = "Reproducibility",
  probar = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.silicodart_+3A_filename">filename</code></td>
<td>
<p>Name of csv file containing the SilicoDArT data [required].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Name of csv file containing metadata assigned to each
entity (individual) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_nas">nas</code></td>
<td>
<p>Missing data character [default '-'].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_topskip">topskip</code></td>
<td>
<p>Number of rows to skip before the header row (containing the
specimen identities) [optional].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Specifies the last non genetic column (Default is
'Reproducibility'). Be sure to check if that is true, otherwise the number of
individuals will not match. You can also specify the last column by a number
[default &quot;Reproducibility&quot;].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, or as set by gl.set.verbose()].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>gl.read.silicodart() opens the data file (csv comma delimited) and skips the
first n=topskip lines. The script assumes that the next line contains the
entity labels (specimen ids) followed immediately by the SNP data for the
first locus.
It reads the presence/absence data into a matrix of 1s and 0s, and inputs the
locus metadata and specimen metadata. The locus metadata comprises a series
of columns of values for each locus including the essential columns of
CloneID and the desirable variables Reproducibility and PIC. Refer to
documentation provide by DArT for an explanation of these columns.
The specimen metadata provides the opportunity to reassign specimens to
populations, and to add other data relevant to the specimen. The key
variables are id (specimen identity which must be the same and in the same
order as the SilicoDArT file, each unique), pop (population assignment), lat
(latitude, optional) and lon (longitude, optional). id, pop, lat, lon are
the column headers in the csv file. Other optional columns can be added.
The data matrix, locus names (forced to be unique), locus metadata, specimen
names, specimen metadata are combined into a genind object. Refer to the
documentation for {adegenet} for further details.
</p>


<h3>Value</h3>

<p>An object of class <code>genlight</code> with ploidy set to 1, containing
the presence/absence data, and locus and individual metadata.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.read.dart">gl.read.dart</a></code>
</p>
<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>silicodartfile &lt;- system.file('extdata','testset_SilicoDArT.csv', package='dartR.data')
metadata &lt;- system.file('extdata',ind.metafile ='testset_metadata_silicodart.csv',
package='dartR.data')
testset.gs &lt;- gl.read.silicodart(filename = silicodartfile, ind.metafile = metadata)
</code></pre>

<hr>
<h2 id='gl.read.vcf'>Converts a vcf file into a genlight object</h2><span id='topic+gl.read.vcf'></span>

<h3>Description</h3>

<p>This function needs package vcfR, please install it. The converted genlight
object does not have individual metrics. You need to add them 'manually' to
the other$ind.metrics slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.vcf(vcf.file, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.vcf_+3A_vcf.file">vcf.file</code></td>
<td>
<p>A vcf file (works only for diploid data) [required].</p>
</td></tr>
<tr><td><code id="gl.read.vcf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># you need to provide the path to the vcf file
#obj &lt;- gl.read.vcf("yourvcffile.vcf", package='dartR.data'))

</code></pre>

<hr>
<h2 id='gl.reassign.pop'>Assigns an individual metric as pop in a genlight {adegenet} object</h2><span id='topic+gl.reassign.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the
individual/sample/specimen metrics file (csv) used with gl.read.dart().
One might want to define the population structure in accordance with another
classification, such as using an individual metric (e.g. sex, male or
female). This script discards the current population assignments and replaces
them with new population assignments defined by a specified individual
metric.
The function returns a genlight object with the new population assignments.
Note that the original population assignments are lost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.reassign.pop(x, as.pop, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.reassign.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.reassign.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Specify the name of the individual metric to set as the pop
variable [required].</p>
</td></tr>
<tr><td><code id="gl.reassign.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reassigned populations.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
   popNames(testset.gl)
   gl &lt;- gl.reassign.pop(testset.gl, as.pop='sex',verbose=3)
   popNames(gl)
# Tag P/A data
   popNames(testset.gs)
   gs &lt;- gl.reassign.pop(testset.gs, as.pop='sex',verbose=3)
   popNames(gs)
   
</code></pre>

<hr>
<h2 id='gl.recalc.metrics'>Recalculates locus metrics when individuals or populations are deleted from a
genlight {adegenet} object
@family environment</h2><span id='topic+gl.recalc.metrics'></span>

<h3>Description</h3>

<p>When individuals,or populations, are deleted from a genlight object, the
locus metrics no longer apply. For example, the Call Rate may be different
considering the subset of individuals, compared with the full set. This
script recalculates those affected locus metrics, namely, avgPIC, CallRate,
freqHets, freqHomRef, freqHomSnp, OneRatioRef, OneRatioSnp, PICRef and
PICSnp.
Metrics that remain unaltered are RepAvg and TrimmedSeq as they are
unaffected by the removal of individuals.
The script optionally removes resultant monomorphic loci or loci
with all values missing and deletes them (using gl.filter.monomorphs.r).
The script returns a genlight object with the recalculated locus metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recalc.metrics(x, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recalc.metrics_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.recalc.metrics_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, removes monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recalc.metrics_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the recalculated locus metadata.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  gl &lt;- gl.recalc.metrics(testset.gl, verbose=2)
  
</code></pre>

<hr>
<h2 id='gl.recode.ind'>Recodes individual (=specimen = sample) labels in a genlight object</h2><span id='topic+gl.recode.ind'></span>

<h3>Description</h3>

<p>This function recodes individual labels and/or deletes individuals from a DaRT
genlight SNP file based on a lookup table provided as a csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recode.ind(x, ind.recode, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object  [required].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_ind.recode">ind.recode</code></td>
<td>
<p>Name of the csv file containing the individual relabelling
[required].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics if any 
individuals are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Renaming individuals may be required when there have been errors in labeling
arising in the process from sample to sequence files. There may be occasions
where renaming individuals is required for preparation of figures. When
caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a durable record of the changes.
The function works with genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
For SNP genotype data, the function, having deleted individuals, optionally 
identifies resultant monomorphic loci or loci with all values missing 
and deletes them. The script also optionally recalculates the
locus metadata as appropriate. The optional deletion of monomorphic loci
and the optional recalculation of locus statistics is not available for
Tag P/A data (SilicoDArT).
The script returns a dartR genlight object with the new individual names  
and the recalculated locus metadata.
</p>


<h3>Value</h3>

<p>A genlight or genind object with the recoded and reduced data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code> for filtering monomorphs,
<code><a href="#topic+gl.recalc.metrics">gl.recalc.metrics</a></code> for recalculating locus metrics,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code> for recoding populations
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  file &lt;- system.file('extdata','testset_ind_recode.csv', package='dartR.data')
  gl &lt;- gl.recode.ind(testset.gl, ind.recode=file, verbose=3)
  
</code></pre>

<hr>
<h2 id='gl.recode.pop'>Recodes population assignments in a genlight object</h2><span id='topic+gl.recode.pop'></span>

<h3>Description</h3>

<p>This function recodes population assignments and/or deletes populations from a
DaRT genlight object based on information provided in a csv population
recode file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recode.pop(x, pop.recode, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_pop.recode">pop.recode</code></td>
<td>
<p>Name of the csv file containing the population
reassignments [required].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculates the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, removes monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart(). Recoding can be used to amalgamate
populations or to selectively delete or retain populations.
When caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a durable record of the changes.
The population recode file contains a list of populations taken from the genlight
object as the first column of the csv file, and the new population
assignments are located in the second column of the csv file. The keyword 
'Delete' used as a new population assignment will result in the associated 
specimen being dropped from the dataset.
The function works with genlight objects
containing SNP genotypes and Tag P/A data (SilicoDArT).
For SNP genotype data, the function, having deleted populations, optionally 
identifies resultant monomorphic loci or loci with all values missing 
and deletes them. The script also optionally recalculates the
locus metadata as appropriate. The optional deletion of monomorphic loci
and the optional recalculation of locus statistics is not available for
Tag P/A data (SilicoDArT).
</p>


<h3>Value</h3>

<p>A genlight object with the recoded and reduced data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>
<p><code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code>
</p>
<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  mfile &lt;- system.file('extdata', 'testset_pop_recode.csv', package='dartR.data')
  nPop(testset.gl)
  gl &lt;- gl.recode.pop(testset.gl, pop.recode=mfile, verbose=3)
 
</code></pre>

<hr>
<h2 id='gl.rename.pop'>Renames a population in a genlight object</h2><span id='topic+gl.rename.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart().
This script renames a nominated population.
The script returns a genlight object with the new population name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.rename.pop(x, old = NULL, new = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.rename.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_old">old</code></td>
<td>
<p>Name of population to be changed [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_new">new</code></td>
<td>
<p>New name for the population [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the new population name.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   gl &lt;- gl.rename.pop(testset.gl, old='EmsubRopeMata', new='Outgroup')
   
</code></pre>

<hr>
<h2 id='gl.report.bases'>Reports summary of base pair frequencies</h2><span id='topic+gl.report.bases'></span>

<h3>Description</h3>

<p>Calculates the frequencies of the four DNA nucleotide bases:
adenine (A), cytosine (C), 'guanine (G) and thymine (T), and the frequency of
transitions (Ts) and transversions (Tv) in a DArT genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.bases(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.bases_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="ggplot2.html#topic+ggsave">ggsave</a>, 
such as width and height, when the ggplot is to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function checks first if trimmed sequences are included in the
locus metadata (@other$loc.metrics$TrimmedSequence), and if so, tallies up
the numbers of A, T, G and C bases. Only the reference state at the SNP locus
is counted. Counts of transitions (Ts) and transversions (Tv) assume that
there is no directionality, that is C-&gt;T is the same as T-&gt;C, because the
reference state is arbitrary.
For presence/absence data (SilicoDArT), it is not possible to count
transversions or transitions or transversions/transitions ratio because the
SNP data are not available, only a single sequence tag per locus.
A color vector can be obtained with gl.select.colors() and then passed to the function
with the plot.colors parameter.
Themes can be obtained from in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir().
</p>


<h3>Value</h3>

<p>The unchanged genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched reports: 
<code><a href="#topic+gl.mahal.assign">gl.mahal.assign</a>()</code>,
<code><a href="#topic+gl.report.factorloadings">gl.report.factorloadings</a>()</code>,
<code><a href="#topic+gl.report.fstat">gl.report.fstat</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  out &lt;- gl.report.bases(testset.gl)
  col &lt;- gl.select.colors(select=c(6,1),palette=rainbow)
  out &lt;- gl.report.bases(testset.gl,plot.colors=col)
# Tag P/A data
  out &lt;- gl.report.bases(testset.gs)
</code></pre>

<hr>
<h2 id='gl.report.basics'>Basic statistics for a genlight object</h2><span id='topic+gl.report.basics'></span>

<h3>Description</h3>

<p>Calculates basic statistics for a genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.basics(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.basics_+3A_x">x</code></td>
<td>
<p>Name of the genlight object  [required].</p>
</td></tr>
<tr><td><code id="gl.report.basics_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other unmatched report: 
<code><a href="#topic+gl.allele.freq">gl.allele.freq</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.report.basics(platypus.gl)

</code></pre>

<hr>
<h2 id='gl.report.callrate'>Reports summary of Call Rate for loci or individuals</h2><span id='topic+gl.report.callrate'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the restriction
enzyme recognition sites. P/A datasets (SilicoDArT) have missing values
because it was not possible to call whether a sequence tag was amplified or
not. This function tabulates the number of missing values as quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.callrate(
  x,
  method = "loc",
  ind.to.list = 20,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  bins = 50,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_method">method</code></td>
<td>
<p>Specify the type of report by locus (method='loc') or
individual (method='ind') [default 'loc'].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_ind.to.list">ind.to.list</code></td>
<td>
<p>Number of individuals to list for callrate [default 20]</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be displayed in the graphics window [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="ggplot2.html#topic+ggsave">ggsave</a>, 
such as width and height, when the ggplot is to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function expects a genlight object, containing either SNP data or
SilicoDArT (=presence/absence data).
Callrate is summarized by locus or by individual to allow sensible decisions
on thresholds for filtering taking into consideration consequential loss of
data. The summary is in the form of a tabulation and plots.
</p>
<p>To avoid issues from inadvertent use of this function in an assignment statement,
the function returns the genlight object unaltered.
Plot themes can be obtained from:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Plot colours can be set with gl.select.colors().
</p>
<p>If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
</p>


<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  test.gl &lt;- testset.gl[1:20,]
  gl.report.callrate(test.gl)
  gl.report.callrate(test.gl,method='ind')
  gl.report.callrate(test.gl,method='ind',plot.file="test")
  gl.report.callrate(test.gl,method='loc',by.pop=TRUE)
  gl.report.callrate(test.gl,method='loc',by.pop=TRUE,plot.file="test")
# Tag P/A data
  test.gs &lt;- testset.gs[1:20,]
  gl.report.callrate(test.gs)
  gl.report.callrate(test.gs,method='ind')
  
  test.gl &lt;- testset.gl[1:20,]
  gl.report.callrate(test.gl)
  
</code></pre>

<hr>
<h2 id='gl.report.diversity'>Calculates diversity indexes for SNPs</h2><span id='topic+gl.report.diversity'></span>

<h3>Description</h3>

<p>This script takes a genlight object and calculates alpha and beta diversity
for q = 0:2. Formulas are taken from Sherwin et al. 2017. The paper describes
nicely the relationship between the different q levels and how they relate to
population genetic processes such as dispersal and selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.diversity(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  library = NULL,
  palette = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  pbar = TRUE,
  table = "DH",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.diversity_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be displayed in the graphics window [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_library">library</code></td>
<td>
<p>Name of the color library to be used [default scales::hue_pl].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_palette">palette</code></td>
<td>
<p>Name of the color palette to be pulled from the specified library [default is library specific].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot.file">plot.file</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_pbar">pbar</code></td>
<td>
<p>Report on progress. Silent if set to FALSE [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_table">table</code></td>
<td>
<p>Prints a tabular output to the console either 'D'=D values, or
'H'=H values or 'DH','HD'=both or 'N'=no table. [default 'DH'].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For all indexes, the entropies (H) and corresponding effective
numbers, i.e. Hill numbers (D), which reflect the number of needed entities
to get the observed values, are calculated. In a nutshell, the alpha indexes
between the different q-values should be similar if there is no deviation
from expected allele frequencies and occurrences (e.g. all loci in HWE &amp;
equilibrium). If there is a deviation of an index, this links to a process
causing it, such as dispersal, selection or strong drift. For a detailed
explanation of all the indexes, we recommend resorting to the literature
provided below. Error bars are +/- 1 standard deviation.
</p>
<p><strong> Function's output </strong>
</p>
<p>If the function's parameter &quot;table&quot; = &quot;DH&quot; (the default value) is used, the 
output of the function is 20 tables.
</p>
<p>The first two show the number of loci used. The name of each of the rest of 
the tables starts with three terms separated by underscores.
</p>
<p>The first term refers to the q value (0 to 2).
</p>
<p>The second term refers to whether it is the diversity measure (H) or its 
transformation to Hill numbers (D). 
</p>
<p>The third term refers to whether the diversity is calculated within 
populations (alpha) or between populations (beta). 
</p>
<p>In the case of alpha diversity tables, standard deviations have their own 
table, which finishes with a fourth term: &quot;sd&quot;.
</p>
<p>In the case of beta diversity tables, standard deviations are in the upper 
triangle of the matrix and diversity values are in the lower triangle of the 
matrix.
</p>
<p><strong> Plotting </strong>
</p>
<p>Plot colours can be set with gl.select.colors().
</p>
<p>If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A list of entropy indexes for each level of q and equivalent numbers
for alpha and beta diversity.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>),
Contributors: William B. Sherwin, Alexander Sentinella
</p>


<h3>References</h3>

<p>Sherwin, W.B., Chao, A., Johst, L., Smouse, P.E. (2017). Information Theory
Broadens the Spectrum of Molecular Ecology and Evolution. TREE 32(12)
948-963. doi:10.1016/j.tree.2017.09.12
</p>


<h3>See Also</h3>

<p>Other unmatched report: 
<code><a href="#topic+gl.allele.freq">gl.allele.freq</a>()</code>,
<code><a href="#topic+gl.report.basics">gl.report.basics</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>div &lt;- gl.report.diversity(bandicoot.gl[1:10,1:100],library='brewer', table=FALSE,pbar=FALSE)
div$zero_H_alpha
div$two_H_beta
names(div)
</code></pre>

<hr>
<h2 id='gl.report.factorloadings'>Reports factor loadings for a PCA or PCoA</h2><span id='topic+gl.report.factorloadings'></span>

<h3>Description</h3>

<p>Extracts the factor loadings from a glPCA object (generated by gl.pcoa) and plots
their distribuion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.factorloadings(
  pca,
  axis = 1,
  n.display = 15,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  bins = 25,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.factorloadings_+3A_pca">pca</code></td>
<td>
<p>Name of the glPCA object containing factor loadings [required].</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_axis">axis</code></td>
<td>
<p>Axis in the ordination used to display the factor loadings [default 1]</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_n.display">n.display</code></td>
<td>
<p>Number of loci for which to display factorloadings [default 15]</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, resultant plots are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;,&quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.report.factorloadings_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="ggplot2.html#topic+ggsave">ggsave</a>, 
such as width and height, when the ggplot is to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function extracts the factor loadings for a given axis from a PCA object 
generated by gl.pcoa and plots their magnitudes. Useful for identifying loci
that load high for a given axis. 
</p>
<p>A color vector can be obtained with gl.select.colors() and then passed to the function
with the plot.colors parameter.
</p>
<p>Themes can be obtained from in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir().
</p>


<h3>Value</h3>

<p>The unchanged genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other matched reports: 
<code><a href="#topic+gl.mahal.assign">gl.mahal.assign</a>()</code>,
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.fstat">gl.report.fstat</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- gl.pcoa(testset.gl)
gl.report.factorloadings(pca = pca)

</code></pre>

<hr>
<h2 id='gl.report.fstat'>Reports various statistics of genetic differentiation between
populations with confident intervals</h2><span id='topic+gl.report.fstat'></span>

<h3>Description</h3>

<p>This function calculates four genetic differentiation between populations
statistics (see the &quot;Details&quot; section for further information).
</p>

<ul>
<li> <p><strong>Fst</strong> - Measure of the degree of genetic differentiation of 
subpopulations (Nei, 1987).
</p>
</li>
<li> <p><strong>Fstp</strong> - Unbiased (i.e. corrected for sampling error, see 
explanation below) Fst (Nei, 1987).
</p>
</li>
<li> <p><strong>Dest</strong> - Jost’s D (Jost, 2008).
</p>
</li>
<li> <p><strong>Gst_H</strong> - Gst standardized by the maximum level that it can obtain for
the observed amount of genetic variation (Hedrick 2005).
</p>
</li></ul>

<p>Sampling errors arise because allele frequencies in our samples differ from 
those in the subpopulations from which they were taken (Holsinger, 2012).
</p>
<p>Confident Intervals are obtained using bootstrapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.fstat(
  x,
  nboots = 0,
  conf = 0.95,
  CI.type = "bca",
  ncpus = 1,
  plot.stat = "Fstp",
  plot.display = TRUE,
  palette.divergent = gl.colors("div"),
  font.size = 0.5,
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.fstat_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_nboots">nboots</code></td>
<td>
<p>Number of bootstrap replicates to obtain confident intervals
[default 0].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_conf">conf</code></td>
<td>
<p>The confidence level of the required interval  [default 0.95].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_ci.type">CI.type</code></td>
<td>
<p>Method to estimate confident intervals. One of
&quot;norm&quot;, &quot;basic&quot;, &quot;perc&quot; or &quot;bca&quot; [default &quot;bca&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_ncpus">ncpus</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncpus
&gt; 1 parallel operation is activated,see &quot;Details&quot; section [default 1].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_plot.stat">plot.stat</code></td>
<td>
<p>Statistic to plot. One of &quot;Fst&quot;,&quot;Fstp&quot;,&quot;Dest&quot; or &quot;Gst_H&quot;
[default &quot;Fstp&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, a heatmap of the pairwise static chosen is
displayed in the plot window [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_palette.divergent">palette.divergent</code></td>
<td>
<p>A color palette function for the heatmap plot
[default gl.colors(&quot;div&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_font.size">font.size</code></td>
<td>
<p>Size of font for the labels of horizontal and vertical axes
of the heatmap [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default working directory].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only,
exclude extension) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.report.fstat_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package
gplots).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Even though Fst and its relatives can predict evolutionary processes
(Holsinger &amp; Weir, 2009), they are not true measures of genetic
differentiation in the sense that they are dependent on the diversity
within populations (Meirmans &amp; Hedrick, 2011), the number of populations
analysed (Alcala &amp; Rosenberg, 2017) and are not monotonic
(Sherwin et al., 2017). Recent approaches have been developed to
accommodate these mathematical restrictions (G'ST; &quot;Gst_H&quot;; Hedrick, 2005,
and Jost's D; &quot;Dest&quot;; Jost, 2008). More recently, novel approaches based on
information theory (Mutual Information; Sherwin et al., 2017) and allele
frequencies (Allele Frequency Difference; Berner, 2019) have distinct
properties that make them valuable resources to interpret genetic
differentiation between populations.
</p>
<p>Note that each measure of genetic differentiation has advantages and
drawbacks, and the decision of using a particular measure is usually
based on the research question.
</p>
<p><strong>Statistics calculated</strong>
</p>
<p>The equations used to calculate the statistics are shown below.
</p>

<ul>
<li>
<p><em>Ho</em> - Unbiased estimate of observed heterozygosity across 
subpopulations (Nei, 1987, pp. 164, eq. 7.38) is calculated as:
</p>
<p><img src="../help/figures/Hoequation.jpg" alt="Hoequation.jpg" />
</p>
<p>where <em>Pkii</em> represents the proportion of homozygote <em>ii</em> for 
allele <em>i</em> in individual <em>k</em> and <em>s</em> represents the number
of subpopulations.
</p>
</li>
<li>
<p><em>Hs</em> - Unbiased estimate of the expected heterozygosity under 
Hardy-Weinberg equilibrium across subpopulations (Nei, 1987, pp. 164,
eq. 7.39) is calculated as:
</p>
<p><img src="../help/figures/Hsequation.jpg" alt="Hsequation.jpg" />
</p>
<p>where <em>ñ</em> is the harmonic mean of <em>nk</em> (the number of 
individuals in each subpopulation), <em>pki</em> is the proportion 
(sometimes misleadingly called frequency) of allele <em>i</em> in 
subpopulation <em>k</em>. 
</p>
</li>
<li>
<p><em>Ht</em> - Heterozygosity for the total population (Nei, 1987, pp. 164,
eq. 7.40) is calculated as:
</p>
<p><img src="../help/figures/Htequation.jpg" alt="Htequation.jpg" />
</p>
</li>
<li>
<p><em>Dst</em> - The average allele frequency differentiation between 
populations (Nei, 1987, pp. 163) is calculated as:
</p>
<p><img src="../help/figures/Dstequation.jpg" alt="Dstequation.jpg" />
</p>
</li>
<li>
<p><em>Htp</em> - Unbiased estimate of Heterozygosity for the total population
(Nei, 1987, pp. 165) is calculated as:
</p>
<p><img src="../help/figures/Htpequation.jpg" alt="Htpequation.jpg" />
</p>
</li>
<li>
<p><em>Dstp</em> - Unbiased estimate of the average allele frequency 
differentiation between populations (Nei, 1987, pp. 165) is calculated
as:
</p>
<p><img src="../help/figures/Dstequation.jpg" alt="Dstequation.jpg" />
</p>
</li>
<li>
<p><em>Fst</em> - Measure of the extent of genetic differentiation 
of subpopulations (Nei, 1987, pp. 162, eq. 7.34) is calculated as:
</p>
<p><img src="../help/figures/Fstequation.jpg" alt="Fstequation.jpg" />
</p>
</li>
<li>
<p><em>Fstp</em> - Unbiased measure of the extent of genetic differentiation 
of subpopulations (Nei, 1987, pp. 163, eq. 7.36) is calculated as:
</p>
<p><img src="../help/figures/Fstpequation.jpg" alt="Fstpequation.jpg" />
</p>
</li>
<li>
<p><em>Dest</em> - Jost’s D (Jost, 2008, eq. 12) is calculated as:
</p>
<p><img src="../help/figures/Dstequation.jpg" alt="Dstequation.jpg" />
</p>
</li>
<li>
<p><em>Gst-max</em> - The maximum level that Gst can obtain for the observed 
amount of genetic variation (Hedrick 2005, eq. 4a) is calculated as:
</p>
<p><img src="../help/figures/GstMaxequation.jpg" alt="GstMaxequation.jpg" />
</p>
</li>
<li>
<p><em>Gst-H</em> - Gst standardized by the maximum level that it can obtain 
for the observed amount of genetic variation (Hedrick 2005, eq. 4b) is 
calculated as:
</p>
<p><img src="../help/figures/GstH.jpg" alt="GstH.jpg" />
</p>
</li></ul>

<p><strong>Confident Intervals</strong>
</p>
<p>The uncertainty of a parameter, in this case the mean of the statistic, can
be summarised by a confidence interval (CI) which includes the true parameter
value with a specified probability (i.e. confidence level; the parameter
&quot;conf&quot; in this function).
</p>
<p>In this function, CI are obtained using Bootstrap which is an inference
method that samples with replacement the data (i.e. loci) and calculates the
statistics every time.
</p>
<p>This function uses the function <a href="boot.html#topic+boot">boot</a> (package boot) to perform
the bootstrap replicates and the function <a href="boot.html#topic+boot.ci">boot.ci</a>
(package boot) to perform the calculations for the CI.
</p>
<p>Four different types of nonparametric CI can be calculated
(parameter &quot;CI.type&quot; in this function):
</p>

<ul>
<li><p> First order normal approximation interval (&quot;norm&quot;).
</p>
</li>
<li><p> Basic bootstrap interval (&quot;basic&quot;).
</p>
</li>
<li><p> Bootstrap percentile interval (&quot;perc&quot;).
</p>
</li>
<li><p> Adjusted bootstrap percentile interval (&quot;bca&quot;).
</p>
</li></ul>

<p>The studentized bootstrap interval (&quot;stud&quot;) was not included in the CI types
because it is computationally intensive, it may produce estimates outside
the range of plausible values and it has been found to be erratic in
practice, see for example the &quot;Studentized (t) Intervals&quot; section in:
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Nice tutorials about the different types of CI can be found in:
</p>
<p><a href="https://www.datacamp.com/tutorial/bootstrap-r">https://www.datacamp.com/tutorial/bootstrap-r</a>
</p>
<p>and
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Efron and Tibshirani (1993, p. 162) and Davison and Hinkley
(1997, p. 194) suggest that the number of bootstrap replicates should
be between 1000 and 2000.
</p>
<p><strong>It is important</strong> to note that unreliable confident intervals will be
obtained if too few number of bootstrap replicates are used.
Therefore, the function <a href="boot.html#topic+boot.ci">boot.ci</a> will throw warnings and errors
if bootstrap replicates are too few. Consider increasing then number of
bootstrap replicates to at least 200.
</p>
<p>The &quot;bca&quot; interval is often cited as the best for theoretical reasons,
however it may produce unstable results if the bootstrap distribution
is skewed or has extreme values. For example, you might get the warning
&quot;extreme order statistics used as endpoints&quot; or the error &quot;estimated
adjustment 'a' is NA&quot;. In this case, you may want to use more bootstrap
replicates or a different method or check your data for outliers.
</p>
<p>The error &quot;estimated adjustment 'w' is infinite&quot; means that the estimated
adjustment ‘w’ for the &quot;bca&quot; interval is infinite, which can happen when
the empirical influence values are zero or very close to zero. This can
be caused by various reasons, such as:
</p>
<p>The number of bootstrap replicates is too small, the statistic of interest
is constant or nearly constant across the bootstrap samples, the data
contains outliers or extreme values.
</p>
<p>You can try some possible solutions, such as:
</p>
<p>Increasing the number of bootstrap replicates, using a different type of
bootstrap confidence interval or removing or transforming the outliers or
extreme values.
</p>
<p><strong>Plotting</strong>
</p>
<p>The plot can be customised by including any parameter(s) from the function
<a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package gplots).
</p>
<p>For the color palette you could try for example:
</p>
<p>&gt; <code>library(viridis)</code>
</p>
<p>&gt; <code>res &lt;- gl.report.fstat(platypus.gl, palette.divergent = viridis)</code>
</p>
<p>If a plot.file is given, the plot arising from this function is saved as an
&quot;RDS&quot; binary file using the function <a href="base.html#topic+saveRDS">saveRDS</a> (package base);
can be reloaded with function <a href="base.html#topic+readRDS">readRDS</a> (package base). A file
name must be specified for the plot to be saved.
</p>
<p>If a plot directory (plot.dir) is specified, the gplot binary is saved to
that directory; otherwise to the tempdir().
</p>
<p>Your plot might not shown in full because your 'Plots' pane is too small
(in RStudio).
Increase the size of the 'Plots' pane before running the function.
Alternatively, use the parameter 'plot.file' to save the plot to a file.
</p>
<p><strong>Parallelisation</strong>
</p>
<p>If the parameter ncpus &gt; 1, parallelisation is enabled. In Windows, parallel
computing employs a &quot;socket&quot; approach that starts new copies of R on each
core. POSIX systems, on the other hand (Mac, Linux, Unix, and BSD),
utilise a &quot;forking&quot; approach that replicates the whole current version of
R and transfers it to a new core.
</p>
<p>Opening and terminating R sessions in each core involves a significant
amount of processing time, therefore parallelisation in Windows machines
is only quicker than not usung parallelisation when nboots &gt; 1000-2000.
</p>


<h3>Value</h3>

<p>Two lists, the first list contains matrices with genetic statistics
taken pairwise by population, the second list contains tables with the
genetic statistics for each pair of populations. If nboots &gt; 0, tables with
the four statistics calculated with Low Confidence Intervals (LCI) and High
Confidence Intervals (HCI).
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li>
<p>Alcala, N., &amp; Rosenberg, N. A. (2017). Mathematical constraints on FST:
Biallelic markers in arbitrarily many populations. Genetics (206), 1581-1600.
</p>
</li>
<li>
<p>Berner, D. (2019). Allele frequency difference AFD–an intuitive alternative
to FST for quantifying genetic population differentiation. Genes, 10(4), 308.
</p>
</li>
<li>
<p>Davison AC, Hinkley DV (1997). Bootstrap Methods and their Application.
Cambridge University Press: Cambridge.
</p>
</li>
<li>
<p>Efron, B. (1979). Bootstrap methods: Another look at the jackknife. Annals of
Statistics 7, 1–26.
</p>
</li>
<li>
<p>Efron B, Tibshirani RJ (1993). An Introduction to the Bootstrap. Chapman and
Hall: London.
</p>
</li>
<li>
<p>Hedrick, P. W. (2005). A standardized genetic differentiation measure.
Evolution, 59(8), 1633-1638.
</p>
</li>
<li> 
<p>Holsinger, K. E. (2012). Lecture notes in population genetics.
</p>
</li>
<li>
<p>Holsinger, K. E., &amp; Weir, B. S. (2009). Genetics in geographically structured
populations: defining, estimating and interpreting FST. Nature Reviews
Genetics, 10(9), 639- 650.
</p>
</li>
<li>
<p>Jost, L. (2008). GST and its relatives do not measure differentiation.
Molecular Ecology, 17(18), 4015-4026.
</p>
</li>
<li>
<p>Meirmans, P. G., &amp; Hedrick, P. W. (2011). Assessing population structure:
FST and related measures. Molecular Ecology Resources, 11(1), 5-18.
</p>
</li>
<li>
<p>Nei, M. (1987). Molecular evolutionary genetics: Columbia University Press.
</p>
</li>
<li>
<p>Sherwin, W. B., Chao, A., Jost, L., &amp; Smouse, P. E. (2017). Information
theory broadens the spectrum of molecular ecology and evolution. Trends in
Ecology &amp; Evolution, 32(12), 948-963.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other matched reports: 
<code><a href="#topic+gl.mahal.assign">gl.mahal.assign</a>()</code>,
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.factorloadings">gl.report.factorloadings</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res &lt;- gl.report.fstat(platypus.gl)

</code></pre>

<hr>
<h2 id='gl.report.hamming'>Calculates the pairwise Hamming distance between DArT trimmed DNA
sequences</h2><span id='topic+gl.report.hamming'></span>

<h3>Description</h3>

<p>Hamming distance is calculated as the number of base differences
between two sequences which can be expressed as a count or a proportion.
Typically, it is calculated between two sequences of equal length. In the
context of DArT trimmed sequences, which differ in length but which are
anchored to the left by the restriction enzyme recognition sequence, it is
sensible to compare the two trimmed sequences starting from immediately after
the common recognition sequence and terminating at the last base of the
shorter sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.hamming(
  x,
  rs = 5,
  threshold = 3,
  tag.length = 69,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.hamming_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_rs">rs</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 5].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_threshold">threshold</code></td>
<td>
<p>Minimum acceptable base pair difference for display on the
boxplot and histogram [default 3].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_tag.length">tag.length</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_probar">probar</code></td>
<td>
<p>If TRUE, a progress bar is displayed during run [defalut FALSE]</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a></code> will filter out one of
two loci if their Hamming distance is less than a specified percentage
Hamming distance can be computed by exploiting the fact that the dot product
of two binary vectors x and (1-y) counts the corresponding elements that are
different between x and y. This approach can also be used for vectors that
contain more than two possible values at each position (e.g. A, C, T or G).
If a pair of DNA sequences are of differing length, the longer is truncated.
The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
as implemented in <code><a href="#topic+utils.hamming">utils.hamming</a></code>
If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
gl.report.hamming(testset.gl[,1:100])
gl.report.hamming(testset.gs[,1:100])

#' # SNP data
test &lt;- platypus.gl
test &lt;- gl.subsample.loc(platypus.gl,n=50)
result &lt;- gl.report.hamming(test, verbose=3)
result &lt;- gl.report.hamming(test, plot.file="ttest", verbose=3)
</code></pre>

<hr>
<h2 id='gl.report.heterozygosity'>Reports observed, expected and unbiased heterozygosities and FIS
(inbreeding coefficient) by population or by individual from SNP data</h2><span id='topic+gl.report.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the observed, expected and unbiased expected (i.e.
corrected for sample size) heterozygosities and FIS (inbreeding coefficient)
for each population or the observed heterozygosity for each individual in a
genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.heterozygosity(
  x,
  method = "pop",
  n.invariant = 0,
  nboots = 0,
  conf = 0.95,
  CI.type = "bca",
  ncpus = 1,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors.pop = gl.colors("dis"),
  plot.colors.ind = gl.colors(2),
  error.bar = "SD",
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.heterozygosity_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP [required].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_method">method</code></td>
<td>
<p>Calculate heterozygosity by population (method='pop') or by
individual (method='ind') [default 'pop'].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_n.invariant">n.invariant</code></td>
<td>
<p>An estimate of the number of invariant sequence tags used
to adjust the heterozygosity rate [default 0].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_nboots">nboots</code></td>
<td>
<p>Number of bootstrap replicates to obtain confidence intervals
[default 0].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_conf">conf</code></td>
<td>
<p>The confidence level of the required interval  [default 0.95].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_ci.type">CI.type</code></td>
<td>
<p>Method to estimate confidence intervals. One of
&quot;norm&quot;, &quot;basic&quot;, &quot;perc&quot; or &quot;bca&quot; [default &quot;bca&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_ncpus">ncpus</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncpus
&gt; 1 parallel operation is activated,see &quot;Details&quot; section [default 1].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot.colors.pop">plot.colors.pop</code></td>
<td>
<p>A color palette for population plots or a list with
as many colors as there are populations in the dataset
[default gl.colors(&quot;dis&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot.colors.ind">plot.colors.ind</code></td>
<td>
<p>List of two color names for the borders and fill of
the plot by individual [default gl.colors(2)].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_error.bar">error.bar</code></td>
<td>
<p>statistic to be plotted as error bar either &quot;SD&quot; (standard 
deviation) or &quot;SE&quot; (standard error) or &quot;CI&quot; (confident intervals)
[default &quot;SD&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Observed heterozygosity for a population takes the proportion of
heterozygous loci for each individual and averages it over all individuals in
that population. The calculations take into account missing values.
</p>
<p>Expected heterozygosity for a population takes the expected proportion of
heterozygotes, that is, expected under Hardy-Weinberg equilibrium, for each
locus, then averages this across the loci for an average estimate for the
population.
</p>
<p>The unbiased expected heterozygosity is calculated using the correction for 
sample size following equation 2 from Nei 1978.
</p>
<p>Accuracy of all heterozygosity estimates is affected by small sample sizes,
and so is their comparison between populations or repeated analysis. Expected
heterozygosities are less affected because their calculations are based on 
allele frequencies while observed heterozygosities are strongly susceptible 
to sampling effects when the sample size is small.  
</p>
<p>Observed heterozygosity for individuals is calculated as the proportion of
loci that are heterozygous for that individual.
</p>
<p>Finally, the loci that are invariant across all individuals in the dataset
(that is, across populations), is typically unknown. This can render
estimates of heterozygosity analysis specific, and so it is not valid to
compare such estimates across species or even across different analyses 
(see Schimdt et al 2021). This is a similar problem faced by microsatellites. 
If you have an estimate of the
number of invariant sequence tags (loci) in your data, such as provided by
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a></code>, you can specify it with the n.invariant
parameter to standardize your estimates of heterozygosity. This is called
autosomal heterozygosities by Schimddt et al (2021).
</p>
<p><strong>NOTE</strong>: It is important to realise that estimation of adjusted (autosomal)
heterozygosity requires that secondaries not to be removed.
</p>
<p>Heterozygosities and FIS (inbreeding coefficient) are calculated by locus
within each population using the following equations, and then averaged across 
all loci:
</p>

<ul>
<li><p> Observed heterozygosity (Ho) = number of heterozygotes / n_Ind,
where n_Ind is the number of individuals without missing data for that locus.
</p>
</li>
<li><p> Observed heterozygosity adjusted (Ho.adj) &lt;- Ho * n_Loc /
(n_Loc + n.invariant),
where n_Loc is the number of loci that do not have all missing data  and
n.invariant is an estimate of the number of invariant loci to adjust
heterozygosity.
</p>
</li>
<li><p> Expected heterozygosity (He) = 1 - (p^2 + q^2),
where p is the frequency of the reference allele and q is the frequency of
the alternative allele.
</p>
</li>
<li><p> Expected heterozygosity adjusted (He.adj) = He * n_Loc /
(n_Loc + n.invariant)
</p>
</li>
<li><p> Unbiased expected heterozygosity (uHe) = He * (2 * n_Ind /
(2 * n_Ind - 1))
</p>
</li>
<li><p> Inbreeding coefficient (FIS) = 1 - Ho / uHe
</p>
</li></ul>

<p><strong> Function's output </strong>
Output for method='pop' is an ordered barchart of observed heterozygosity,
unbiased expected heterozygosity and FIS (Inbreeding coefficient) across populations
together with a table of mean observed and expected heterozygosities and FIS
by population and their respective standard deviations (SD).
In the output, it is also reported by population: the number of loci used to
estimate heterozygosity(n.Loc), the number of polymorphic loci (polyLoc),
the number of monomorphic loci (monoLoc) and loci with all missing data
(all_NALoc).
Output for method='ind' is a histogram and a boxplot of heterozygosity across
individuals.
Plots and table are saved to the session temporary directory (tempdir)
Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p><strong>Error bars</strong>
</p>
<p>The best method for presenting or assessing genetic statistics depends on 
the type of data you have and the specific questions you're trying to 
answer. Here's a brief overview of when you might use each method:
</p>
<p><strong>1. Confidence Intervals (&quot;CI&quot;):</strong>
</p>
<p>- Usage: Often used to convey the precision of an estimate.
</p>
<p>- Advantage: Confidence intervals give a range in which the true parameter 
(like a population mean) is likely to fall, given the data and a specified 
probability (like 95
</p>
<p>- In Context: For genetic statistics, if you're estimating a parameter,
a 95
lies.
</p>
<p><strong>2. Standard Deviation (&quot;SD&quot;):</strong>
</p>
<p>- Usage: Describes the amount of variation from the average in a set of data.
</p>
<p>- Advantage: Allows for an understanding of the spread of individual data
points around the mean.
</p>
<p>- In Context: If you're looking at the distribution of a quantitative trait 
(like height) in a population with a particular genotype, the SD can 
describe how much individual heights vary around the average height.
</p>
<p><strong>3. Standard Error (&quot;SE&quot;):</strong>
</p>
<p>- Usage: Describes the precision of the sample mean as an estimate of the 
population mean.
</p>
<p>- Advantage: Smaller than the SD in large samples; it takes into account 
both the SD and the sample size. 
</p>
<p>- In Context: If you want to know how accurately your sample mean represents
the population mean, you'd look at the SE.
</p>
<p><strong>Recommendation:</strong>
</p>
<p>- If you're trying to convey the precision of an estimate, confidence 
intervals are very useful.
</p>
<p>- For understanding variability within a sample, standard deviation is key.
</p>
<p>- To see how well a sample mean might estimate a population mean, consider 
the standard error.
</p>
<p>In practice, geneticists often use a combination of these methods to 
analyze and present their data, depending on their research questions and 
the nature of the data.
</p>
<p><strong>Confident Intervals</strong>
</p>
<p>The uncertainty of a parameter, in this case the mean of the statistic, can
be summarised by a confidence interval (CI) which includes the true parameter
value with a specified probability (i.e. confidence level; the parameter
&quot;conf&quot; in this function).
</p>
<p>In this function, CI are obtained using Bootstrap which is an inference
method that samples with replacement the data (i.e. loci) and calculates the
statistics every time.
</p>
<p>This function uses the function <a href="boot.html#topic+boot">boot</a> (package boot) to perform
the bootstrap replicates and the function <a href="boot.html#topic+boot.ci">boot.ci</a>
(package boot) to perform the calculations for the CI.
</p>
<p>Four different types of nonparametric CI can be calculated
(parameter &quot;CI.type&quot; in this function):
</p>

<ul>
<li><p> First order normal approximation interval (&quot;norm&quot;).
</p>
</li>
<li><p> Basic bootstrap interval (&quot;basic&quot;).
</p>
</li>
<li><p> Bootstrap percentile interval (&quot;perc&quot;).
</p>
</li>
<li><p> Adjusted bootstrap percentile interval (&quot;bca&quot;).
</p>
</li></ul>

<p>The studentized bootstrap interval (&quot;stud&quot;) was not included in the CI types
because it is computationally intensive, it may produce estimates outside
the range of plausible values and it has been found to be erratic in
practice, see for example the &quot;Studentized (t) Intervals&quot; section in:
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Nice tutorials about the different types of CI can be found in:
</p>
<p><a href="https://www.datacamp.com/tutorial/bootstrap-r">https://www.datacamp.com/tutorial/bootstrap-r</a>
</p>
<p>and
</p>
<p><a href="https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/">https://www.r-bloggers.com/2019/09/understanding-bootstrap-confidence-interval-output-from-the-r-boot-package/</a>
</p>
<p>Efron and Tibshirani (1993, p. 162) and Davison and Hinkley
(1997, p. 194) suggest that the number of bootstrap replicates should
be between 1000 and 2000.
</p>
<p><strong>It is important</strong> to note that unreliable confident intervals will be
obtained if too few number of bootstrap replicates are used.
Therefore, the function <a href="boot.html#topic+boot.ci">boot.ci</a> will throw warnings and errors
if bootstrap replicates are too few. Consider increasing the number of
bootstrap replicates to at least 200.
</p>
<p>The &quot;bca&quot; interval is often cited as the best for theoretical reasons,
however it may produce unstable results if the bootstrap distribution
is skewed or has extreme values. For example, you might get the warning
&quot;extreme order statistics used as endpoints&quot; or the error &quot;estimated
adjustment 'a' is NA&quot;. In this case, you may want to use more bootstrap
replicates or a different method or check your data for outliers.
</p>
<p>The error &quot;estimated adjustment 'w' is infinite&quot; means that the estimated
adjustment ‘w’ for the &quot;bca&quot; interval is infinite, which can happen when
the empirical influence values are zero or very close to zero. This can
be caused by various reasons, such as:
</p>
<p>The number of bootstrap replicates is too small, the statistic of interest
is constant or nearly constant across the bootstrap samples, the data
contains outliers or extreme values.
</p>
<p>You can try some possible solutions, such as:
</p>
<p>Increasing the number of bootstrap replicates, using a different type of
bootstrap confidence interval or removing or transforming the outliers or
extreme values.
</p>
<p><strong>Parallelisation</strong>
</p>
<p>If the parameter ncpus &gt; 1, parallelisation is enabled. In Windows, parallel
computing employs a &quot;socket&quot; approach that starts new copies of R on each
core. POSIX systems, on the other hand (Mac, Linux, Unix, and BSD),
utilise a &quot;forking&quot; approach that replicates the whole current version of
R and transfers it to a new core.
</p>
<p>Opening and terminating R sessions in each core involves a significant
amount of processing time, therefore parallelisation in Windows machines
is only quicker than not using parallelisation when nboots &gt; 1000-2000.
</p>


<h3>Value</h3>

<p>A dataframe containing population labels, heterozygosities, FIS,
their standard deviations and sample sizes
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Nei, M. (1978). Estimation of average heterozygosity and genetic distance
from a small number of individuals. Genetics, 89(3), 583-590.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a></code>
</p>
<p>Other unmatched report: 
<code><a href="#topic+gl.allele.freq">gl.allele.freq</a>()</code>,
<code><a href="#topic+gl.report.basics">gl.report.basics</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
require("dartR.data")
df &lt;- gl.report.heterozygosity(platypus.gl)
df &lt;- gl.report.heterozygosity(platypus.gl,method='ind')
n.inv &lt;- gl.report.secondaries(platypus.gl)
gl.report.heterozygosity(platypus.gl, n.invariant = n.inv[7, 2])

df &lt;- gl.report.heterozygosity(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.report.hwe'>Reports departure from Hardy-Weinberg proportions</h2><span id='topic+gl.report.hwe'></span>

<h3>Description</h3>

<p>Calculates the probabilities of agreement with H-W proportions based on 
observed
frequencies of reference homozygotes, heterozygotes and alternate 
homozygotes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.hwe(
  x,
  subset = "each",
  method_sig = "Exact",
  multi_comp = FALSE,
  multi_comp_method = "BY",
  alpha_val = 0.05,
  pvalue_type = "midp",
  cc_val = 0.5,
  sig_only = TRUE,
  min_sample_size = 5,
  plot.out = TRUE,
  plot_colors = gl.colors("2c"),
  max_plots = 4,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_subset">subset</code></td>
<td>
<p>Way to group individuals to perform H-W tests. Either a vector
with population names, 'each', 'all' (see details) [default 'each'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_method_sig">method_sig</code></td>
<td>
<p>Method for determining statistical significance: 
'ChiSquare'
or 'Exact' [default 'Exact'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_multi_comp">multi_comp</code></td>
<td>
<p>Whether to adjust p-values for multiple comparisons
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_multi_comp_method">multi_comp_method</code></td>
<td>
<p>Method to adjust p-values for multiple comparisons:
'holm', 'hochberg', 'hommel', 'bonferroni', 'BH', 'BY', 'fdr'
(see details) [default 'fdr'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_pvalue_type">pvalue_type</code></td>
<td>
<p>Type of p-value to be used in the Exact method.
Either 'dost','selome','midp' (see details) [default 'midp'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_cc_val">cc_val</code></td>
<td>
<p>The continuity correction applied to the ChiSquare test
[default 0.5].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_sig_only">sig_only</code></td>
<td>
<p>Whether the returned table should include loci with a 
significant departure from Hardy-Weinberg proportions [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_min_sample_size">min_sample_size</code></td>
<td>
<p>Minimum number of individuals per population in which
perform H-W tests [default 5].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, will produce Ternary Plot(s) [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the significant and
not-significant loci [default gl.colors(&quot;2c&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_max_plots">max_plots</code></td>
<td>
<p>Maximum number of plots to print per page [default 4].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are several factors that can cause deviations from Hardy-Weinberg
proportions including: mutation, finite population size, selection,
population structure, age structure, assortative mating, sex linkage,
nonrandom sampling and genotyping errors. Therefore, testing for
Hardy-Weinberg proportions should be a process that involves a careful
evaluation of the results, a good place to start is Waples (2015).
Note that tests for H-W proportions are only valid if there is no population
substructure (assuming random mating) and have sufficient power only when
there is sufficient sample size (n individuals &gt; 15).
Populations can be defined in three ways:
</p>

<ul>
<li><p> Merging all populations in the dataset using subset = 'all'.
</p>
</li>
<li><p> Within each population separately using: subset = 'each'.
</p>
</li>
<li><p> Within selected populations using for example: subset =
c('pop1','pop2').
</p>
</li></ul>

<p>Two different statistical methods to test for deviations from Hardy Weinberg
proportions:
</p>

<ul>
<li><p> The classical chi-square test (method_sig='ChiSquare') based on the
function <code><a href="HardyWeinberg.html#topic+HWChisq">HWChisq</a></code> of the R package HardyWeinberg.
By default a continuity correction is applied (cc_val=0.5). The
continuity correction can be turned off (by specifying cc_val=0), for example
in cases of extreme allele frequencies in which the continuity correction can
lead to excessive type 1 error rates.
</p>
</li>
<li><p> The exact test (method_sig='Exact') based on the exact calculations
contained in the function <code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code> of the R
package HardyWeinberg, and described in Wigginton  et al. (2005). The exact
test is recommended in most cases (Wigginton  et al., 2005).
Three different methods to estimate p-values (pvalue_type) in the Exact test
can be used:
</p>

<ul>
<li><p> 'dost' p-value is computed as twice the tail area of a one-sided test.
</p>
</li>
<li><p> 'selome' p-value is computed as the sum of the probabilities of all
samples less or equally likely as the current sample.
</p>
</li>
<li><p> 'midp', p-value is computed as half the probability of the current
sample + the probabilities of all samples that are more extreme.
</p>
</li></ul>

<p>The standard exact p-value is overly conservative, in particular
for small minor allele frequencies. The mid p-value ameliorates this problem
by bringing the rejection rate closer to the nominal level, at the price of
occasionally exceeding the nominal level (Graffelman &amp; Moreno, 2013).
</p>
</li></ul>

<p>Correction for multiple tests can be applied using the following methods
based on the function <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>:
</p>

<ul>
<li><p> 'holm' is also known as the sequential Bonferroni technique (Rice,
1989). This method has a greater statistical power than the standard
Bonferroni test, however this method becomes very stringent when many tests
are performed and many real deviations from the null hypothesis can go
undetected (Waples, 2015).
</p>
</li>
<li><p> 'hochberg' based on Hochberg, 1988.
</p>
</li>
<li><p> 'hommel' based on Hommel, 1988. This method is more powerful than
Hochberg's, but the difference is usually small.
</p>
</li>
<li><p> 'bonferroni' in which p-values are multiplied by the number of tests.
This method is very stringent and therefore has reduced power to detect
multiple departures from the null hypothesis.
</p>
</li>
<li><p> 'BH' based on Benjamini &amp; Hochberg, 1995.
</p>
</li>
<li><p> 'BY' based on Benjamini &amp; Yekutieli, 2001.
</p>
</li></ul>

<p>The first four methods are designed to give strong control of the family-wise
error rate. The last two methods control the false discovery rate (FDR),
the expected proportion of false discoveries among the rejected hypotheses.
The false discovery rate is a less stringent condition than the family-wise
error rate, so these methods are more powerful than the others, especially
when number of tests is large.
The number of tests on which the adjustment for multiple comparisons is
the number of populations times the number of loci.
<strong>Ternary plots</strong>
Ternary plots can be used to visualise patterns of H-W proportions (plot.out
= TRUE). P-values and the statistical (non)significance of a large number of
bi-allelic markers can be inferred from their position in a ternary plot.
See Graffelman &amp; Morales-Camarena (2008) for further details. Ternary plots
are based on the function  <code><a href="HardyWeinberg.html#topic+HWTernaryPlot">HWTernaryPlot</a></code> from
the package HardyWeinberg. Each vertex of the Ternary plot represents one of 
the three possible genotypes for SNP data: homozygous for the reference 
allele (AA), heterozygous (AB) and homozygous for the alternative allele
(BB). Loci deviating significantly from Hardy-Weinberg proportions after 
correction for multiple tests are shown in pink. The blue parabola 
represents Hardy-Weinberg equilibrium, and the area between green lines 
represents the acceptance region.
For these plots to work it is necessary to install the package ggtern.
</p>


<h3>Value</h3>

<p>A dataframe containing loci, counts of reference SNP homozygotes,
heterozygotes and alternate SNP homozygotes; probability of departure from
H-W proportions, per locus significance with and without correction for
multiple comparisons and the number of population where the same locus is 
significantly out of HWE.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Benjamini, Y., and Yekutieli, D. (2001). The control of the false
discovery rate in multiple testing under dependency. Annals of Statistics,
29, 1165–1188.
</p>
</li>
<li><p> Graffelman, J. (2015). Exploring Diallelic Genetic Markers: The Hardy
Weinberg Package. Journal of Statistical Software 64:1-23.
</p>
</li>
<li><p> Graffelman, J. &amp; Morales-Camarena, J. (2008). Graphical tests for
Hardy-Weinberg equilibrium based on the ternary plot. Human Heredity
65:77-84.
</p>
</li>
<li><p> Graffelman, J., &amp; Moreno, V. (2013). The mid p-value in exact tests for
Hardy-Weinberg equilibrium. Statistical applications in genetics and
molecular biology, 12(4), 433-448.
</p>
</li>
<li><p> Hochberg, Y. (1988). A sharper Bonferroni procedure for multiple tests
of significance. Biometrika, 75, 800–803.
</p>
</li>
<li><p> Hommel, G. (1988). A stagewise rejective multiple test procedure based
on a modified Bonferroni test. Biometrika, 75, 383–386.
</p>
</li>
<li><p> Rice, W. R. (1989). Analyzing tables of statistical tests. Evolution,
43(1), 223-225.
</p>
</li>
<li><p> Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we
lost the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li>
<li><p> Wigginton, J.E., Cutler, D.J., &amp; Abecasis, G.R. (2005). A Note on Exact
Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics
76:887-893.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a></code>
</p>

<hr>
<h2 id='gl.report.ld'>Calculates pairwise population based Linkage Disequilibrium across all loci
using the specified number of cores
@family matched report</h2><span id='topic+gl.report.ld'></span>

<h3>Description</h3>

<p>This function is implemented in a parallel fashion to speed up the process.
There is also the ability to restart the function if crashed by specifying
the chunk file names or restarting the function exactly in the same way as in
the first run. This is implemented because sometimes, due to connectivity loss
between cores, the function may crash half way. Before running the function, 
it is advisable to use the function <code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> to remove
loci with all missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.ld(
  x,
  name = NULL,
  save = TRUE,
  outpath = tempdir(),
  nchunks = 2,
  ncores = 1,
  chunkname = NULL,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.ld_+3A_x">x</code></td>
<td>
<p>A genlight or genind object created (genlight objects are internally
converted via <code><a href="#topic+gl2gi">gl2gi</a></code> to genind) [required].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_name">name</code></td>
<td>
<p>Character string for rdata file. If not given genind object name
is used [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_save">save</code></td>
<td>
<p>Switch if results are saved in a file [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_outpath">outpath</code></td>
<td>
<p>Folder where chunks and results are saved (if save=TRUE)
[default tempdir()].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_nchunks">nchunks</code></td>
<td>
<p>How many subchunks will be used (the less the faster, but if
the routine crashes more bits are lost) [default 2].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_ncores">ncores</code></td>
<td>
<p>How many cores should be used [default 1].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_chunkname">chunkname</code></td>
<td>
<p>The name of the chunks for saving [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_probar">probar</code></td>
<td>
<p>if TRUE, a progress bar is displayed for long loops
[default = TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns calculation of pairwise LD across all loci between
subpopulations. This functions uses if specified many cores on your computer
to speed up. And if save is used can restart (if save=TRUE is used) with the
same command starting where it crashed. The final output is a data frame that
holds all statistics of pairwise LD between loci. (See ?LD in package
genetics for details).
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.report.ld.map'>Calculates pairwise linkage disequilibrium by population</h2><span id='topic+gl.report.ld.map'></span>

<h3>Description</h3>

<p>This function calculates pairwise linkage disequilibrium (LD) by population 
using the function <code><a href="snpStats.html#topic+ld">ld</a></code> (package snpStats).
If SNPs are not mapped to a reference genome, the parameter
<code>ld.max.pairwise</code>
should be set as NULL (the default). In this case, the 
function will assign the same chromosome (&quot;1&quot;) to all the SNPs in the dataset
and assign a sequence from 1 to n loci as the position of each SNP. The 
function will then calculate LD for all possible SNP pair combinations. 
If SNPs are mapped to a reference genome, the parameter 
<code>ld.max.pairwise</code>
should be filled out (i.e. not NULL). In this case, the
information for SNP's position should be stored in the genlight accessor
&quot;@position&quot; and the SNP's chromosome name in the accessor &quot;@chromosome&quot;
(see examples). The function will then calculate LD within each chromosome
and for all possible SNP pair combinations within a distance of
<code>ld.max.pairwise</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.ld.map(
  x,
  ld.max.pairwise = NULL,
  maf = 0.05,
  ld.stat = "R.squared",
  ind.limit = 10,
  stat.keep = "AvgPIC",
  ld.threshold.pops = 0.2,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.file = NULL,
  plot.dir = NULL,
  histogram.colors = NULL,
  boxplot.colors = NULL,
  bins = 50,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.ld.map_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld.max.pairwise">ld.max.pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD 
should be calculated [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out 
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld.stat">ld.stat</code></td>
<td>
<p>The LD measure to be calculated: &quot;LLR&quot;, &quot;OR&quot;, &quot;Q&quot;, &quot;Covar&quot;,
&quot;D.prime&quot;, &quot;R.squared&quot;, and &quot;R&quot;. See <code><a href="snpStats.html#topic+ld">ld</a></code>
(package snpStats) for details [default &quot;R.squared&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_stat.keep">stat.keep</code></td>
<td>
<p>Name of the column from the slot <code>loc.metrics</code> to be
used to choose SNP to be kept [default &quot;AvgPIC&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld.threshold.pops">ld.threshold.pops</code></td>
<td>
<p>LD threshold to report in the plot of &quot;Number of 
populations in which the same SNP pair are in LD&quot; [default 0.2].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms of base composition are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_histogram.colors">histogram.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_boxplot.colors">boxplot.colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 50].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reports LD between SNP pairs by population. 
The function <code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code> filters out the SNPs in LD using as
input the results of <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code>. The actual number of 
SNPs to be filtered out depends on the parameters set in the function 
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code>.
Boxplots of LD by population and
a histogram showing LD frequency are presented.
</p>


<h3>Value</h3>

<p>A dataframe with information for each SNP pair in LD.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code>
</p>
<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.filter.monomorphs(x)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
ld_res &lt;- gl.report.ld.map(x,ld.max.pairwise = 10000000)
</code></pre>

<hr>
<h2 id='gl.report.locmetric'>Reports summary of the slot $other$loc.metrics</h2><span id='topic+gl.report.locmetric'></span>

<h3>Description</h3>

<p>This function reports summary statistics (mean, minimum, average, quantiles), histograms
and boxplots for any loc.metric with numeric values (stored in 
$other$loc.metrics) to assist the decision of choosing thresholds for the filter
function <code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.locmetric(
  x,
  metric,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.locmetric_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_metric">metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code> will filter out the
loci with a locmetric value below a specified threshold.
The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own field by adding a vector
into $other$loc.metrics as shown in the example. You can check the names of
all available loc.metrics via: names(gl$other$loc.metrics).
</p>

<ul>
<li><p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li><p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li><p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li><p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li><p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li><p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li><p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li><p> PICRef - polymorphism information content (PIC) for the Reference allele.
</p>
</li>
<li><p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li><p> AvgPIC - average of the polymorphism information content (PIC) of the
reference and SNP alleles.
</p>
</li>
<li><p> AvgCountRef - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Reference allele row.
</p>
</li>
<li><p> AvgCountSnp - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Alternate (SNP) allele
row.
</p>
</li>
<li><p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li>
<li><p> rdepth - read depth.
</p>
</li></ul>

<p><strong> Function's output </strong>
The minimum, maximum, mean and a tabulation of quantiles of the locmetric
values against thresholds rate are provided. Output also includes a boxplot
and a histogram.
Quantiles are partitions of a finite set of values into q subsets of (nearly)
equal sizes. In this function q = 20. Quantiles are useful measures because
they are less susceptible to long-tailed distributions and outliers.
Plot colours can be set with gl.select.colors().
If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
out &lt;- gl.report.locmetric(testset.gl,metric='SnpPosition')
# Tag P/A data
out &lt;- gl.report.locmetric(testset.gs,metric='AvgReadDepth')
</code></pre>

<hr>
<h2 id='gl.report.maf'>Reports minor allele frequency (MAF) for each locus in a SNP dataset</h2><span id='topic+gl.report.maf'></span>

<h3>Description</h3>

<p>This script provides summary histograms of MAF for each
population and an overall histogram to assist the decision of
choosing thresholds for the filter function <code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.maf(
  x,
  as.pop = NULL,
  maf.limit = 0.5,
  ind.limit = 5,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  bins = 25,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_as.pop">as.pop</code></td>
<td>
<p>Temporarily assign another locus metric as the population for
the purposes of deletions [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_maf.limit">maf.limit</code></td>
<td>
<p>Show histograms MAF range &lt;= maf.limit [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Show histograms only for populations of size greater than
ind.limit [default 5].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be displayed in the graphics window [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code> will filter out the
loci with MAF below a specified threshold.
<strong> Function's output </strong>
The minimum, maximum, mean and a tabulation of MAF quantiles against
thresholds rate are provided. Output also includes a boxplot and a
histogram.
This function reports the
MAF for each of several quantiles. Quantiles are
partitions of a finite set of values into q subsets of (nearly) equal sizes.
In this function q = 20. Quantiles are useful measures because they are less
susceptible to long-tailed distributions and outliers.
</p>
<p>Plot colours can be set with gl.select.colors().
</p>
<p>If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>

<hr>
<h2 id='gl.report.monomorphs'>Reports monomorphic loci</h2><span id='topic+gl.report.monomorphs'></span>

<h3>Description</h3>

<p>This script reports the number of monomorphic loci and those with all NAs in
a genlight {adegenet} object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.monomorphs(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.monomorphs_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.report.monomorphs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A DArT dataset will not have monomorphic loci, but they can arise, along with
loci that are scored all NA, when populations or individuals are deleted.
Retaining monomorphic loci unnecessarily increases the size of the dataset
and will affect some calculations.
Note that for SNP data, NAs likely represent null alleles; in tag
presence/absence data, NAs represent missing values (presence/absence could
not be reliably scored)
</p>


<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>
<p>Other matched reports: 
<code><a href="#topic+gl.mahal.assign">gl.mahal.assign</a>()</code>,
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.factorloadings">gl.report.factorloadings</a>()</code>,
<code><a href="#topic+gl.report.fstat">gl.report.fstat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl.report.monomorphs(testset.gl)
# SilicoDArT data
  gl.report.monomorphs(testset.gs)
</code></pre>

<hr>
<h2 id='gl.report.overshoot'>Reports loci for which the SNP has been trimmed from the sequence tag
along with the adaptor</h2><span id='topic+gl.report.overshoot'></span>

<h3>Description</h3>

<p>This function checks the position of the SNP within the trimmed sequence tag
and identifies those for which the SNP position is outside the trimmed
sequence tag. This can happen, rarely, when the sequence containing the SNP
resembles the adaptor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.overshoot(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.overshoot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.report.overshoot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SNP genotype can still be used in most analyses, but functions like
gl2fasta() will present challenges if the SNP has been trimmed from the 
sequence tag.
Resultant ggplot(s) and the tabulation(s) are saved to the session's
temporary directory.
</p>


<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.report.overshoot(testset.gl)

</code></pre>

<hr>
<h2 id='gl.report.pa'>Reports private alleles (and fixed alleles) per pair of populations</h2><span id='topic+gl.report.pa'></span>

<h3>Description</h3>

<p>This function reports private alleles in one population compared with a
second population, for all populations taken pairwise. It also reports a
count of fixed allelic differences and the mean absolute allele frequency
differences (AFD) between pairs of populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.pa(
  x,
  x2 = NULL,
  method = "pairwise",
  loc.names = FALSE,
  test.asym = FALSE,
  test.asym.boot = 100,
  plot.display = FALSE,
  plot.font = 14,
  map.interactive = FALSE,
  provider = "Esri.NatGeoWorldMap",
  palette.discrete = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.pa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_x2">x2</code></td>
<td>
<p>If two separate genlight objects are to be compared this can be
provided here, but they must have the same number of SNPs [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_method">method</code></td>
<td>
<p>Method to calculate private alleles: 'pairwise' comparison or
compare each population against the rest 'one2rest' [default 'pairwise'].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_loc.names">loc.names</code></td>
<td>
<p>Whether names of loci with private alleles and fixed 
differences should reported. If TRUE, loci names are reported using a list</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_test.asym">test.asym</code></td>
<td>
<p>bootstrap test for significant differences of private alleles. 
This test uses a bootstrap simulation by shuffling individuals between a pair
of population and drawing with replacement. For each bootstrap the ratio of 
private alleles is compared to the actual ratio and recorded how often it is 
larger than the simulated one. If number of individuals are different between
population bootstrap is done using the smaller number of samples in both 
populations.</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_test.asym.boot">test.asym.boot</code></td>
<td>
<p>number of bootstraps [default 100]
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if Sankey plot is to be produced [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_plot.font">plot.font</code></td>
<td>
<p>Numeric font size in pixels for the node text labels
[default 14].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_map.interactive">map.interactive</code></td>
<td>
<p>Specify whether an interactive map showing private
alleles between populations is to be produced [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_provider">provider</code></td>
<td>
<p>Passed to leaflet [default &quot;Esri.NatGeoWorldMap&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_palette.discrete">palette.discrete</code></td>
<td>
<p>A discrete palette for the color of populations or a
list with as many colors as there are populations in the dataset
[default gl.select.colors(x)].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent, fatal errors only; 1, flag function
begin and end; 2, progress log; 3, progress and results summary; 5, full
report [default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the number of paired alleles between two populations is not a
symmetric dissimilarity measure.
If no x2 is provided, the function uses the pop(gl) hierarchy to determine
pairs of populations, otherwise it runs a single comparison between x and
x2.
<strong>Hint:</strong> in case you want to run comparisons between individuals
(assuming individual names are unique), you can simply redefine your
population names with your individual names, as below:
<code>pop(gl) &lt;- indNames(gl)</code>
<strong> Definition of fixed and private alleles </strong>
The table below shows the possible cases of allele frequencies between
two populations (0 = homozygote for Allele 1, x = both Alleles are present,
1 = homozygote for Allele 2).
</p>

<ul>
<li><p> p: cases where there is a private allele in pop1 compared to pop2 (but
not vice versa)
</p>
</li>
<li><p> f: cases where there is a fixed allele in pop1 (and pop2, as those cases
are symmetric)
</p>
</li></ul>


<table>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> </td><td style="text-align: center;"> </td><td style="text-align: center;"> <em>pop1</em> </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> </td><td style="text-align: center;"> <strong>0</strong> </td><td style="text-align: center;">   <strong>x</strong>  </td><td style="text-align: center;">  <strong>1</strong></td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;">     <strong>0</strong></td><td style="text-align: center;"> -  </td><td style="text-align: center;">  p </td><td style="text-align: center;">  p,f</td>
</tr>
<tr>
 <td style="text-align: center;">
<em>pop2</em> </td><td style="text-align: center;"> <strong>x</strong></td><td style="text-align: center;"> -  </td><td style="text-align: center;">- </td><td style="text-align: center;"> -</td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> <strong>1</strong> </td><td style="text-align: center;"> p,f</td><td style="text-align: center;"> p </td><td style="text-align: center;">   -</td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p>The absolute allele frequency difference (AFD) in this function is a simple
differentiation metric displaying intuitive properties which provides a
valuable alternative to FST. For details about its properties and how it is
calculated see Berner (2019).
The function also reports an estimation of the lower bound of the number of
undetected private alleles using the Good-Turing frequency formula,
originally developed for cryptography, which estimates in an ecological 
context the true frequencies of rare species in a single assemblage based on
an incomplete sample of individuals. The approach is described in Chao et 
al. (2017). For this function, the equation 2c is used. This estimate is 
reported in the output table as Chao1 and Chao2. 
In this function a Sankey Diagram is used to visualize patterns of private
alleles between populations. This diagram allows to display flows (private
alleles) between nodes (populations). Their links are represented with arcs
that have a width proportional to the importance of the flow (number of
private alleles).
if save2temp=TRUE, resultant plot(s) and the tabulation(s) are saved to the
session's temporary directory.
</p>


<h3>Value</h3>

<p>A data.frame. Each row shows, for each pair of populations the number
of individuals in each population, the number of loci with fixed differences
(same for both populations) in pop1 (compared to pop2) and vice versa. Same
for private alleles and finally the absolute mean allele frequency
difference between loci (AFD). If loc.names = TRUE, loci names with private
alleles and fixed differences are reported in a list in addition to the 
dataframe.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Berner, D. (2019). Allele frequency difference AFD – an intuitive
alternative to FST for quantifying genetic population differentiation. Genes,
10(4), 308.
</p>
</li>
<li><p> Chao, Anne, et al. &quot;Deciphering the enigma of undetected species,
phylogenetic, and functional diversity based on Good-Turing theory.&quot; 
Ecology 98.11 (2017): 2914-2929.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.pa(platypus.gl)

out &lt;- gl.report.pa(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.report.rdepth'>Reports summary of Read Depth for each locus</h2><span id='topic+gl.report.rdepth'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT report AvgCountRef and AvgCountSnp as counts
of sequence tags for the reference and alternate alleles respectively.
These can be used to back calculate Read Depth. Fragment presence/absence
datasets as provided by DArT (SilicoDArT) provide Average Read Depth and
Standard Deviation of Read Depth as standard columns in their report. This
function reports the read depth by locus for each of several quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.rdepth(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.rdepth_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.theme">plot.theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function displays a table of minimum, maximum, mean and quantiles for
read depth against possible thresholds that might subsequently be specified
in <code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>. If plot.display=TRUE, display also includes a
boxplot and a histogram to guide in the selection of a threshold for
filtering on read depth.
Plot colours can be set with gl.select.colors().
If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
For examples of themes, see  </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
df &lt;- gl.report.rdepth(testset.gl)
df &lt;- gl.report.rdepth(testset.gs)

</code></pre>

<hr>
<h2 id='gl.report.reproducibility'>Reports summary of RepAvg (repeatability averaged over both alleles 
for
each locus) or reproducibility (repeatability of the scores for fragment
presence/absence)</h2><span id='topic+gl.report.reproducibility'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have an index, RepAvg, generated by
reproducing the data independently for 30
of alleles that give a repeatable result, averaged over both alleles for each
locus.
In the case of fragment presence/absence data (SilicoDArT), repeatability is
the percentage of scores that are repeated in the technical replicate
dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.reproducibility(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.reproducibility_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options [default
theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function displays a table of minimum, maximum, mean and quantiles for
repeatbility against possible thresholds that might subsequently be
specified in <code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a></code>.
If plot.display=TRUE, display also includes a boxplot and a histogram to guide
in the selection of a threshold for filtering on repeatability.
If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
For examples of themes, see:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and 
</p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  out &lt;- gl.report.reproducibility(testset.gl)
  
# Tag P/A data
  out &lt;- gl.report.reproducibility(testset.gs)
</code></pre>

<hr>
<h2 id='gl.report.secondaries'>Reports loci containing secondary SNPs in sequence tags and calculates
number of invariant sites</h2><span id='topic+gl.report.secondaries'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT include fragments with more than
one SNP (that is, with secondaries). They are recorded separately with the
same CloneID (=AlleleID). These multiple SNP loci within a fragment are
likely to be linked, and so you may wish to remove secondaries.
This function reports statistics associated with secondaries, and the
consequences of filtering them out, and provides three plots. The first is
a boxplot, the second is a barplot of the frequency of secondaries per
sequence tag, and the third is the Poisson expectation for those
frequencies including an estimate of the zero class (no. of sequence tags
with no SNP scored).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.secondaries(
  x,
  nsim = 1000,
  taglength = 69,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.secondaries_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulations to estimate the mean of the Poisson
distribution [default 1000].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_taglength">taglength</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.display">plot.display</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options [default
theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.colors">plot.colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.file">plot.file</code></td>
<td>
<p>Filename (minus extension) for the RDS plot file [Required for plot save]</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code> will filter out the
loci with secondaries retaining only one sequence tag.
Heterozygosity as estimated by the function
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code> is in a sense relative, because it
is calculated against a background of only those loci that are polymorphic
somewhere in the dataset. To allow intercompatibility across studies and
species, any measure of heterozygosity needs to accommodate loci that are
invariant (autosomal heterozygosity. See Schmidt et al 2021). However, the
number of invariant loci are unknown given the SNPs are detected as single
point mutational variants and invariant sequences are discarded, and
because of the particular additional filtering pre-analysis. Modelling the
counts of SNPs per sequence tag as a Poisson distribution in this script
allows estimate of the zero class, that is, the number of invariant loci.
This is reported, and the veracity of the estimate can be assessed by the
correspondence of the observed frequencies against those under Poisson
expectation in the associated graphs. The number of invariant loci can then
be optionally provided to the function
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code> via the parameter n.invariants.
In case the calculations for the Poisson expectation of the number of
invariant sequence tags fail to converge, try to rerun the analysis with a
larger <code>nsim</code> values.
This function now also calculates the number of invariant sites (i.e.
nucleotides) of the sequence tags (if <code>TrimmedSequence</code> is present in
<code>x$other$loc.metrics</code>) or estimate these by assuming that the average
length of the sequence tags is 69 nucleotides. Based on the Poisson
expectation of the number of invariant sequence tags, it also estimates the
number of invariant sites for these to eventually provide an estimate of
the total number of invariant sites.
<strong>Note</strong>, previous version of
<code>dartR</code> would only return an estimate of the number of invariant
sequence tags (not sites).
If plot.file is specified, plots are saved to the directory specified by the user, or the global
default working directory set by gl.set.wd() or to the tempdir().
Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>


<ul>
<li><p> n.total.tags Number of sequence tags in total
</p>
</li>
<li><p> n.SNPs.secondaries Number of secondary SNP loci that would be removed
on filtering
</p>
</li>
<li><p> n.invariant.tags Estimated number of invariant sequence tags
</p>
</li>
<li><p> n.tags.secondaries Number of sequence tags with secondaries
</p>
</li>
<li><p> n.inv.gen Number of invariant sites in sequenced tags
</p>
</li>
<li><p> mean.len.tag Mean length of sequence tags
</p>
</li>
<li><p> n.invariant Total Number of invariant sites (including invariant
sequence tags)
</p>
</li>
<li><p> k Lambda: mean of the Poisson distribution of number of SNPs in the
sequence tags
</p>
</li></ul>



<h3>Value</h3>

<p>A data.frame with the list of parameter values
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Schmidt, T.L., Jasper, M.-E., Weeks, A.R., Hoffmann, A.A., 2021.
Unbiased population heterozygosity estimates from genome-wide sequence
data. Methods in Ecology and Evolution n/a.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code>,<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
n.inv &lt;- gl.report.secondaries(test)
gl.report.heterozygosity(test, n.invariant = n.inv[7, 2])

</code></pre>

<hr>
<h2 id='gl.report.taglength'>Reports summary of sequence tag length across loci</h2><span id='topic+gl.report.taglength'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT typically have sequence tag lengths ranging
from 20 to 69 base pairs. This function reports summary statistics of the tag
lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.taglength(
  x,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.taglength_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP [required].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, histograms of base composition are displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default c(&quot;#2171B5&quot;, &quot;#6BAED6&quot;)].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a></code> will filter out the
loci with a tag length below a specified threshold.
Quantiles are partitions of a finite set of values into q subsets of (nearly)
equal sizes. In this function q = 20. Quantiles are useful measures because
they are less susceptible to long-tailed distributions and outliers.
<strong> Function's output </strong>
The minimum, maximum, mean and a tabulation of tag length quantiles against
thresholds are provided. Output also includes a boxplot and a
histogram to guide in the selection of a threshold for filtering on tag
length.
If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir(). 
Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a></code>
</p>
<p>Other matched report: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.taglength(testset.gl)

</code></pre>

<hr>
<h2 id='gl.sample'>Samples individuals from populations</h2><span id='topic+gl.sample'></span>

<h3>Description</h3>

<p>A function to subsample individuals in a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sample(
  x,
  nsample = min(table(pop(x))),
  replace = TRUE,
  onepop = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sample_+3A_x">x</code></td>
<td>
<p>genlight object containing SNP/silicodart genotypes</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_nsample">nsample</code></td>
<td>
<p>the number of individuals that should be sampled</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_replace">replace</code></td>
<td>
<p>a switch to sample by replacement (default).</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_onepop">onepop</code></td>
<td>
<p>switch to ignore population settings of the genlight object and sample from all individuals disregarding the population definition. [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_verbose">verbose</code></td>
<td>
<p>set verbosity</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is convenience function to facilitate a bootstrap approach
</p>
<p>This function is often used to support a bootstrap 
approach in dartR. For a bootstrap approach it is often 
desirable to sample a defined number of individuals for 
each of the populations in a genlight object and then 
calculate a certain quantity for that subset (redo a 
1000 times)
</p>


<h3>Value</h3>

<p>returns a genlight object with nsample samples from each populations.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#bootstrap for 2 possums populations to check effect of sample size on fixed alleles
gl.set.verbosity(0)
pp &lt;- possums.gl[c(1:30,91:120),]
nrep &lt;- 1:10
nss &lt;- seq(1,10,2)
res &lt;- expand.grid(nrep=nrep, nss=nss)
for (i in 1:nrow(res)) {
dummy &lt;- gl.sample(pp, nsample=res$nss[i], replace=TRUE)
dummy &lt;- gl.compliance.check(dummy)
pas &lt;- gl.report.pa(dummy, plot.display= FALSE)
res$fixed[i] &lt;- pas$fixed[1]
}
boxplot(fixed ~ nss, data=res)

</code></pre>

<hr>
<h2 id='gl.save'>Saves an object in compressed binary format for later rapid retrieval</h2><span id='topic+gl.save'></span>

<h3>Description</h3>

<p>This is a wrapper for saveRDS().
The script saves the object in binary form to the current workspace and
returns the input gl object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.save(x, file, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.save_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.save_+3A_file">file</code></td>
<td>
<p>Name of the file to receive the binary version of the object
[required].</p>
</td></tr>
<tr><td><code id="gl.save_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.load">gl.load</a></code>
</p>
<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.save(testset.gl,file.path(tempdir(),'testset.rds'))
</code></pre>

<hr>
<h2 id='gl.select.colors'>Selects colors from one of several palettes and outputs as a vector</h2><span id='topic+gl.select.colors'></span>

<h3>Description</h3>

<p>This function draws upon a number of specified color libraries to extract a
vector of colors for plotting. For use where the function that follows has a color
parameter expecting a vector of colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.select.colors(
  x = NULL,
  library = NULL,
  palette = NULL,
  ncolors = NULL,
  select = NULL,
  plot.display = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.select.colors_+3A_x">x</code></td>
<td>
<p>Optionally, provide a gl object from which to determine the number
of populations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_library">library</code></td>
<td>
<p>Name of the color library to be used, one of 'brewer'
'gr.palette', 'r.hcl' or 'baseR' [default scales::hue_pl].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_palette">palette</code></td>
<td>
<p>Name of the color palette to be pulled from the specified
library, refer function help [default is library specific].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_ncolors">ncolors</code></td>
<td>
<p>number of colors to be displayed and returned [default 9 or nPop(gl)].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_select">select</code></td>
<td>
<p>select bu number the colors to retain in the output vector; 
can repeat colors. [default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_plot.display">plot.display</code></td>
<td>
<p>if TRUE, plot the colours in the plot window [default=TRUE]</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_verbose">verbose</code></td>
<td>
<p>&ndash; verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Colors are chosen by specifying a library (one of 'brewer'
'gr.palette', 'r.hcl' or 'baseR') and a palette within that library. Each
library has its own array of palettes, which can be listed as outlined below.
Alternatively, if you specify an incorrect palette, the list of available
palettes for the specified library will be listed.
</p>
<p>The available color libraries and their palettes include:
</p>

<ul>
<li><p> library 'brewer' and the palettes available can be listed by
RColorBrewer::display.brewer.all() and RColorBrewer::brewer.pal.info.
</p>
</li>
<li><p> library 'gr.palette' and the palettes available can be listed by
grDevices::palette.pals()
</p>
</li>
<li><p> library 'r.hcl' and the palettes available can be listed by
grDevices::hcl.pals()
</p>
</li>
<li><p> library 'baseR' and the palettes available are: 'rainbow','heat',
'topo.colors','terrain.colors','cm.colors'.
</p>
</li></ul>

<p>If the library is not specified, then the default library 'scales' is set and
the default palette of 'hue_pal is set.
</p>
<p>If the library is set but the palette is not specified, all palettes for that
library will be listed and a default palette will then be chosen.
The color palette will be displayed in the graphics window for the requested
number of colors (or 9 if not specified or nPop(gl) if a genlight object is 
specified),and the vector of colors returned by assignment for later use.
The select parameter can be used to select colors from the specified ncolors.
For example, select=c(1,1,3) will select color 1, 1 again and 3 to retain in
the final vector. This can be useful for fine-tuning color selection, and
matching colors and shapes.
</p>


<h3>Value</h3>

<p>A vector with the required number of colors
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.select.shapes">gl.select.shapes</a></code>
</p>
<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# EXAMPLES -- SIMPLE
colors &lt;- gl.select.colors()
colors &lt;- gl.select.colors(library='brewer',palette='Spectral',ncolors=6)
colors &lt;- gl.select.colors(library='baseR',palette='terrain.colors',ncolors=6)
colors &lt;- gl.select.colors(library='baseR',palette='rainbow',ncolors=12)
colors &lt;- gl.select.colors(library='gr.hcl',palette='RdBu',ncolors=12)
colors &lt;- gl.select.colors(library='gr.palette',palette='Pastel 1',ncolors=6)
# EXAMPLES -- SELECTING colorS
colors &lt;- gl.select.colors(library='baseR',palette='rainbow',ncolors=12,select=c(1,1,1,5,8))
# EXAMPLES -- CROSS-CHECKING WITH A GENLIGHT OBJECT
colors &lt;- gl.select.colors(x=gl,library='baseR',palette='rainbow',ncolors=12,select=c(1,1,1,5,8))

</code></pre>

<hr>
<h2 id='gl.select.shapes'>Selects shapes from the base R shape palette and outputs as a vector</h2><span id='topic+gl.select.shapes'></span>

<h3>Description</h3>

<p>This script draws upon the standard R shape palette to extract a vector of
shapes for plotting, where the script that follows has a shape parameter
expecting a vector of shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.select.shapes(x = NULL, select = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.select.shapes_+3A_x">x</code></td>
<td>
<p>Optionally, provide a gl object from which to determine the number
of populations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.shapes_+3A_select">select</code></td>
<td>
<p>Select the shapes to retain in the output vector
[default NULL, all shapes shown and returned].</p>
</td></tr>
<tr><td><code id="gl.select.shapes_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the shape palette will be displayed in full in the graphics window
from which shapes can be selected in a subsequent run, and the vector of
shapes returned for later use.
The select parameter can be used to select shapes from the specified 26
shapes available (0-25). For example, select=c(1,1,3) will select shape 1, 1
again and 3 to retain in the final vector. This can be useful for fine-tuning
shape selection, and matching colors and shapes.
</p>


<h3>Value</h3>

<p>A vector with the required number of shapes
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.select.colors">gl.select.colors</a></code>
</p>
<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# EXAMPLES
shapes &lt;- gl.select.shapes() # Select and display available shapes
# Select and display a restricted set of shapes
shapes &lt;- gl.select.shapes(select=c(1,1,1,5,8)) 
 # Select set of shapes and check with no. of pops.
shapes &lt;- gl.select.shapes(x=gl,select=c(1,1,1,5,8))
</code></pre>

<hr>
<h2 id='gl.set.verbosity'>Sets the default verbosity level</h2><span id='topic+gl.set.verbosity'></span>

<h3>Description</h3>

<p>dartR functions have a verbosity parameter that sets the level of reporting
during the execution of the function. The verbosity level, set by parameter
'verbose' can be one of verbose 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report. The
default value for verbosity is stored in the r environment. This script sets
the default value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.set.verbosity(value = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.set.verbosity_+3A_value">value</code></td>
<td>
<p>Set the default verbosity to be this value: 0, silent only fatal
errors; 1, begin and end; 2, progress log; 3, progress and results summary;
5, full report [default 2]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>verbosity value [set for all functions]
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl &lt;- gl.set.verbosity(value=2)
</code></pre>

<hr>
<h2 id='gl.set.wd'>Sets the default working directory</h2><span id='topic+gl.set.wd'></span>

<h3>Description</h3>

<p>Many dartR functions have a plot.dir parameter which is used to save output to (e.g. ggplots as rds files)
With this functions users can set the working directory globally so it is used in all functions, without setting is explicitely. 
The value for wd is stored in the r environment and if not set defaults to tempdir(). This script sets the default value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.set.wd(wd = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.set.wd_+3A_wd">wd</code></td>
<td>
<p>Set the path to the wd directory globally to be used by all functions if not set explicitely in the function.</p>
</td></tr>
<tr><td><code id="gl.set.wd_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>path the the working directory [set for all functions]
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other environment: 
<code><a href="#topic+gl.check.verbosity">gl.check.verbosity</a>()</code>,
<code><a href="#topic+gl.check.wd">gl.check.wd</a>()</code>,
<code><a href="#topic+gl.print.history">gl.print.history</a>()</code>,
<code><a href="#topic+theme_dartR">theme_dartR</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#set to current working directory
wd &lt;- gl.set.wd(wd=getwd())

</code></pre>

<hr>
<h2 id='gl.sim.cross'>Generates random crosses between fathers and mothers</h2><span id='topic+gl.sim.cross'></span><span id='topic+gl.sim.crosses'></span>

<h3>Description</h3>

<p>Generates random crosses between fathers (in one genlight object) and
mothers (in a second genlight object) then randomly selects a specified
number of offspring to retain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.crosses(
  fathers,
  mothers,
  broodsize = 10,
  sexratio = 0.5,
  n = 1000,
  error.check = TRUE,
  compliance.check = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.cross_+3A_fathers">fathers</code></td>
<td>
<p>Genlight object of potential fathers [required].</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_mothers">mothers</code></td>
<td>
<p>Genlight object of potential mothers simulated [required].</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_broodsize">broodsize</code></td>
<td>
<p>Number of offspring per mother [required].</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_sexratio">sexratio</code></td>
<td>
<p>Sex ratio of simulated offspring [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_n">n</code></td>
<td>
<p>Number of offspring to retain [default 1000 or mothers*broodsize whichever is the lesser]</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_error.check">error.check</code></td>
<td>
<p>If TRUE, will perform error checks on the provided parameters [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_compliance.check">compliance.check</code></td>
<td>
<p>If TRUE, will perform a compliance check on the resultant
genlight object before returning it [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.sim.cross_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script is to be used in conjunction with gl.subsample.ind() applied
initially to a base genlight object containing initial male and female 
genotypes. The workflow is to
</p>
<p>(a) Select the males from the base genlight object using gl.keep.pop() with 
pop.list=&quot;male&quot; and the as.pop parameter set to sex.
(b) Select the females from the base genlight object using gl.keep.pop() with 
pop.list=&quot;female&quot; and the as.pop parameter set to sex.
(c) Subsample a cohort of males for breeding and a cohort of females for breeding
using gl.subsample.ind() and the replace parameter as follows:
</p>
<p>To enforce monogamy &ndash; generate the fathers and mothers from the base genlight object
using gl.subsample.ind() with replace=FALSE.
To admit polygyny &ndash; generate the fathers from the base genlight object using 
gl.subsample.ind() with replace=FALSE and the mothers from the base genlight
object using gl.subsample.ind() with replace=TRUE.
To admit polyandry &ndash; generate the fathers from the base genlight object using 
gl.subsample.ind() with replace=TRUE and the mothers from the base genlight
object using gl.subsample.ind() with replace=FALSE.
To admit promiscuity &ndash; generate the fathers and mothers from the base genlight object using
gl.subsample.ind() with replace=TRUE.
</p>
<p>These are simple scenarios that leave the number of maternal mates per 
father (polygyny) and the number of paternal mates per mother (polyandry) 
to chance, depending on the random selection of males and females with 
replacement from the base genlight object.
</p>
<p>(d) Cross the males with the females using gl.sim.crosses() retaining a
subset of offspring at random.
</p>
<p>So the input for this function is a genlight object with a sample of male
individuals (fathers) selected from a larger set at random with or without 
replacement; a similar sample of female individuals in a second genlight 
object (mothers); specified broodsize; and desired offspring sex ratio.
</p>
<p>Set check.error to FALSE if using this script in simulations
</p>


<h3>Value</h3>

<p>A genlight object with n offspring of both sexes.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.sim.genotypes'>Generate random genotypes</h2><span id='topic+gl.sim.genotypes'></span>

<h3>Description</h3>

<p>Generate random genotypes for a single population drawing upon the allele
frequencies from that population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.genotypes(x, n.ind = 200, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.genotypes_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.sim.genotypes_+3A_n.ind">n.ind</code></td>
<td>
<p>Number of individuals to be simulated (should be less than the number of loci) [default 200]</p>
</td></tr>
<tr><td><code id="gl.sim.genotypes_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object with the simulated genotypes
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>

<hr>
<h2 id='gl.smearplot'>Smear plot</h2><span id='topic+gl.smearplot'></span>

<h3>Description</h3>

<p>Each locus is color coded for scores of 0, 1, 2 and NA for SNP data and 0, 1
and NA for presence/absence (SilicoDArT) data. Individual labels can be added.
Plot may become cluttered if ind.labels If there are too many individuals, 
it is best to use ind.labels = FALSE.
</p>
<p>Works with both SNP data and P/A data (SilicoDArT)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.smearplot(
  x,
  plot.display = TRUE,
  ind.labels = FALSE,
  label.size = 10,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  het.only = FALSE,
  legend = "bottom",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.smearplot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot.display">plot.display</code></td>
<td>
<p>If TRUE, the plot is displayed in the plot window
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_ind.labels">ind.labels</code></td>
<td>
<p>If TRUE, individual IDs are shown [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_label.size">label.size</code></td>
<td>
<p>Size of the individual labels [default 10].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of four color names for the column fill for homozygous reference,
heterozygous, homozygous alternate, and missing value (NA) [default c(&quot;#0000FF&quot;,&quot;#00FFFF&quot;,&quot;#FF0000&quot;,&quot;#e0e0e0&quot;)].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]#'</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_het.only">het.only</code></td>
<td>
<p>If TRUE, show only the heterozygous state [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_legend">legend</code></td>
<td>
<p>Position of the legend: “left”, “top”, “right”, “bottom” or
'none' [default = 'bottom'].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the ggplot object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.tree.nj">gl.tree.nj</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.smearplot(testset.gl,ind.labels=FALSE)
gl.smearplot(testset.gs,ind.labels=FALSE)
gl.smearplot(testset.gl[1:10,],ind.labels=TRUE)
gl.smearplot(testset.gs[1:10,],ind.labels=TRUE)
</code></pre>

<hr>
<h2 id='gl.sort'>re-sorts genlight objects</h2><span id='topic+gl.sort'></span>

<h3>Description</h3>

<p>Often it is desirable to have the genlight object sorted individuals by population names, indiviual name, for example to have a more informative gl.smearplot (showing banding patterns for populations). Also sorting by loci can be informative in some instances. This function provides the ability to sort individuals of a genlight object by providing the order of individuals or populations and also by loci metric providing the order of locis. See examples below for specifics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sort(x, sort.by = "pop", order.by = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sort_+3A_x">x</code></td>
<td>
<p>genlight object containing SNP/silicodart genotypes</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_sort.by">sort.by</code></td>
<td>
<p>either &quot;ind&quot;, &quot;pop&quot;. Default is pop</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_order.by">order.by</code></td>
<td>
<p>that is used to order individuals or loci. Depening on the order.by parameter, this needs to be a vector of length of nPop(genlight) for populations or  nInd(genlight) for individuals. If not specified alphabetical order of populations or individuals is used. For sort.by=&quot;ind&quot; order.by can be also a vector specifying the order for each individual (for example another ind.metrics)</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_verbose">verbose</code></td>
<td>
<p>set verbosity</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is convenience function to facilitate sorting of individuals within the genlight object. For example if you want to visualise the &quot;band&quot; of population in a gl.smearplot then the order of individuals is important. Also
</p>


<h3>Value</h3>

<p>Returns a reordered genlight object. Sorts also the ind/loc.metrics and coordinates accordingly
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sort by populations
bc &lt;- gl.sort(bandicoot.gl)
#sort from West to East
bc2 &lt;- gl.sort(bandicoot.gl, sort.by="pop" ,
order.by=c("WA", "SA", "VIC", "NSW", "QLD"))
#sort by missing values
miss &lt;- rowSums(is.na(as.matrix(bandicoot.gl)))
bc3 &lt;- gl.sort(bandicoot.gl, sort.by="ind", order.by=miss)
gl.smearplot(bc3)
</code></pre>

<hr>
<h2 id='gl.subsample.ind'>Subsample individuals from a genlight object</h2><span id='topic+gl.subsample.ind'></span>

<h3>Description</h3>

<p>A function to subsample individuals at random in a genlight object
with and without replacement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.subsample.ind(
  x,
  n = NULL,
  replace = TRUE,
  by.pop = TRUE,
  error.check = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.subsample.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.subsample.ind_+3A_n">n</code></td>
<td>
<p>Number of individuals to include in the subsample [default NULL]</p>
</td></tr>
<tr><td><code id="gl.subsample.ind_+3A_replace">replace</code></td>
<td>
<p>If TRUE, sampling is with replacement [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.subsample.ind_+3A_by.pop">by.pop</code></td>
<td>
<p>If FALSE, ignore population settings [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.subsample.ind_+3A_error.check">error.check</code></td>
<td>
<p>If TRUE, will undertake error checks on input paramaters [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.subsample.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Retain a subset of individuals at random, with or without replacement.
If subsampling globally, n must be less than or equal to nInd(x). If subsampling
by population, then n must be less than the minimum sample size for any population.
</p>
<p>Set error.check = FALSE for speedy execution in simulations
</p>


<h3>Value</h3>

<p>Returns the subsampled genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.loc">gl.subsample.loc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl &lt;- gl.subsample.ind(testset.gl, n=30, by.pop=FALSE, replace=TRUE)
gl &lt;- gl.subsample.ind(platypus.gl, n=10, by.pop=TRUE, replace=TRUE)
</code></pre>

<hr>
<h2 id='gl.subsample.loc'>Subsample loci from a genlight object</h2><span id='topic+gl.subsample.loc'></span>

<h3>Description</h3>

<p>A function to subsample loci at random in a genlight object
with and without replacement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.subsample.loc(x, n, replace = TRUE, error.check = TRUE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.subsample.loc_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.subsample.loc_+3A_n">n</code></td>
<td>
<p>Number of loci to include in the subsample [default NULL]</p>
</td></tr>
<tr><td><code id="gl.subsample.loc_+3A_replace">replace</code></td>
<td>
<p>If TRUE, sampling is with replacement [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.subsample.loc_+3A_error.check">error.check</code></td>
<td>
<p>If TRUE, will undertake error checks on input paramaters [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.subsample.loc_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Retain a subset of loci at random, with or without replacement.
Parameter n must be less than or equal to nLoc(x). 
</p>
<p>#' Set error.check = FALSE for speedy execution in simulations
</p>


<h3>Value</h3>

<p>Returns the subsampled genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other data manipulation: 
<code><a href="#topic+gl.define.pop">gl.define.pop</a>()</code>,
<code><a href="#topic+gl.drop.ind">gl.drop.ind</a>()</code>,
<code><a href="#topic+gl.drop.loc">gl.drop.loc</a>()</code>,
<code><a href="#topic+gl.drop.pop">gl.drop.pop</a>()</code>,
<code><a href="#topic+gl.edit.recode.pop">gl.edit.recode.pop</a>()</code>,
<code><a href="#topic+gl.impute">gl.impute</a>()</code>,
<code><a href="#topic+gl.join">gl.join</a>()</code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a>()</code>,
<code><a href="#topic+gl.keep.loc">gl.keep.loc</a>()</code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a>()</code>,
<code><a href="#topic+gl.make.recode.ind">gl.make.recode.ind</a>()</code>,
<code><a href="#topic+gl.merge.pop">gl.merge.pop</a>()</code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a>()</code>,
<code><a href="#topic+gl.recode.ind">gl.recode.ind</a>()</code>,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a>()</code>,
<code><a href="#topic+gl.rename.pop">gl.rename.pop</a>()</code>,
<code><a href="#topic+gl.sample">gl.sample</a>()</code>,
<code><a href="#topic+gl.sim.genotypes">gl.sim.genotypes</a>()</code>,
<code><a href="#topic+gl.sort">gl.sort</a>()</code>,
<code><a href="#topic+gl.subsample.ind">gl.subsample.ind</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2 &lt;- gl.subsample.loc(testset.gl, n=50, replace=TRUE, verbose=3)
</code></pre>

<hr>
<h2 id='gl.test.heterozygosity'>Tests the difference in heterozygosity between populations taken
pairwise</h2><span id='topic+gl.test.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the expected heterozygosities for each population in a genlight
object, and uses re-randomization to test the statistical significance of
differences in heterozygosity between populations taken pairwise.
</p>
<p>Expected heterozygosity is calculated using the correction for sample size 
following equation 2 from Nei 1978.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.test.heterozygosity(
  x,
  nreps = 100,
  alpha1 = 0.05,
  alpha2 = 0.01,
  plot.out = TRUE,
  max_plots = 6,
  plot.theme = theme_dartR(),
  plot.colors = gl.select.colors(ncolors = 2, verbose = 0),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.test.heterozygosity_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_nreps">nreps</code></td>
<td>
<p>Number of replications of the re-randomization [default 1,000].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_alpha1">alpha1</code></td>
<td>
<p>First significance level for comparison with diff=0 on plot
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_alpha2">alpha2</code></td>
<td>
<p>Second significance level for comparison with diff=0 on plot
[default 0.01].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, plots a sampling distribution of the differences for
each comparison [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_max_plots">max_plots</code></td>
<td>
<p>Maximum number of plots to print per page [default 6].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.colors">plot.colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default gl.colors(2)].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong> Function's output </strong>
If plot.out = TRUE, plots are created showing the sampling distribution for
the difference between each pair of heterozygosities, marked with the
critical limits alpha1 and alpha2, the observed heterozygosity, and the zero
value (if in range).
If a plot.file is given, the ggplot arising from this function is saved as an &quot;RDS&quot; 
binary file using saveRDS(); can be reloaded with readRDS(). A file name must be 
specified for the plot to be saved.
If a plot directory (plot.dir) is specified, the ggplot binary is saved to that
directory; otherwise to the tempdir(). 
Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A dataframe containing population labels, heterozygosities and sample
sizes
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Nei, M. (1978). Estimation of average heterozygosity and genetic distance 
from a small number of individuals. Genetics, 89(3), 583-590.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.test.heterozygosity(platypus.gl, nreps=1, verbose=3, plot.out=TRUE)
</code></pre>

<hr>
<h2 id='gl.tree.nj'>Outputs an nj tree to summarize genetic similarity among populations</h2><span id='topic+gl.tree.nj'></span>

<h3>Description</h3>

<p>This function is a wrapper for the nj function or package ape applied to Euclidean
distances calculated from the genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.tree.nj(
  x,
  dist.matrix = NULL,
  type = "phylogram",
  outgroup = NULL,
  labelsize = 0.7,
  treefile = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.tree.nj_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_dist.matrix">dist.matrix</code></td>
<td>
<p>Distance matrix [default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_type">type</code></td>
<td>
<p>Type of dendrogram &quot;phylogram&quot;|&quot;cladogram&quot;|&quot;fan&quot;|&quot;unrooted&quot;
[default &quot;phylogram&quot;].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_outgroup">outgroup</code></td>
<td>
<p>Vector containing the population names that are the outgroups
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_labelsize">labelsize</code></td>
<td>
<p>Size of the labels as a proportion of the graphics default
[default 0.7].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_treefile">treefile</code></td>
<td>
<p>Name of the file for the tree topology using Newick format 
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent, fatal errors only; 1, flag function
begin and end; 2, progress log; 3, progress and results summary; 5, full
report [default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An euclidean distance matrix is calculated by default [dist.matrix = NULL]. 
Optionally the user can use as input for the tree any other distance matrix
using this parameter, see for example the function  <code><a href="#topic+gl.dist.pop">gl.dist.pop</a></code>.
</p>


<h3>Value</h3>

<p>A tree file of class phylo.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other graphics: 
<code><a href="#topic+gl.colors">gl.colors</a>()</code>,
<code><a href="#topic+gl.map.interactive">gl.map.interactive</a>()</code>,
<code><a href="#topic+gl.plot.heatmap">gl.plot.heatmap</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.tree.nj(testset.gl,type='fan')
# Tag P/A data
  gl.tree.nj(testset.gs,type='fan')
  
  res &lt;- gl.tree.nj(platypus.gl)
  
</code></pre>

<hr>
<h2 id='gl.write.csv'>Writes out data from a genlight object to csv file</h2><span id='topic+gl.write.csv'></span>

<h3>Description</h3>

<p>This script writes to file the SNP genotypes with specimens as entities
(columns) and loci as attributes (rows). Each row has associated locus
metadata. Each column, with header of specimen id, has population in the
first row.
The data coding differs from the DArT 1row format in that 0 = reference
homozygous, 2 = alternate homozygous, 1 = heterozygous, and NA = missing SNP
assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.write.csv(x, outfile = "outfile.csv", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.write.csv_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;outfile.csv&quot;].</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saves a genlight object to csv, returns NULL.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+utils.read.dart">utils.read.dart</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl.write.csv(testset.gl, outfile='SNP_1row.csv')
# Tag P/A data
  gl.write.csv(testset.gs, outfile='PA_1row.csv')
</code></pre>

<hr>
<h2 id='gl2bayesAss'>Converts a genlight object into bayesAss (BA3) input format</h2><span id='topic+gl2bayesAss'></span>

<h3>Description</h3>

<p>This function exports a genlight object into bayesAss format and save it into a
file.
This function only caters for <code>ploidy=2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2bayesAss(
  x,
  ploidy = 2,
  outfile = "gl.BayesAss.txt",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2bayesAss_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2bayesAss_+3A_ploidy">ploidy</code></td>
<td>
<p>Set the ploidy [defaults 2].</p>
</td></tr>
<tr><td><code id="gl2bayesAss_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl.BayesAss.txt'].</p>
</td></tr>
<tr><td><code id="gl2bayesAss_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2bayesAss_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the input file as data.table
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Mussmann S. M., Douglas M. R., Chafin T. K. and Douglas M. E. (2019) BA3-SNPs: 
Contemporary migration reconfigured in BayesAss for next-generation sequence data. 
Methods in Ecology and Evolution 10, 1808-1813.
</p>
<p>Wilson G. A. and Rannala B. (2003) Bayesian Inference of Recent Migration Rates 
Using Multilocus Genotypes. Genetics 163, 1177-1191.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
#only the first 100 due to check time
gl2bayesAss(platypus.gl[,1:100], outpath=tempdir())
</code></pre>

<hr>
<h2 id='gl2bayescan'>Converts a genlight object into a format suitable for input to Bayescan</h2><span id='topic+gl2bayescan'></span>

<h3>Description</h3>

<p>The output text file contains the SNP data and relevant BAyescan command
lines to guide input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2bayescan(x, outfile = "bayescan.txt", outpath = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2bayescan_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default bayescan.txt].</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Foll M and OE Gaggiotti (2008) A genome scan method to identify selected loci
appropriate for both dominant and codominant markers: A Bayesian
perspective. Genetics 180: 977-993.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl2bayescan(testset.gl, outpath = tempdir())

</code></pre>

<hr>
<h2 id='gl2bpp'>Converts a genlight object into a format suitable for input to the BPP 
program</h2><span id='topic+gl2bpp'></span>

<h3>Description</h3>

<p>This function generates the sequence alignment file and the Imap file. The 
control file should produced by the user. 
If method = 1, heterozygous positions are replaced by standard ambiguity 
codes.
If method = 2, the heterozygous state is resolved by randomly assigning one 
or the other SNP variant to the individual.
Trimmed sequences for which the SNP has been trimmed out, rarely, by adapter
mis-identity are deleted.
This function requires 'TrimmedSequence' to be among the locus metrics
(<code>@other$loc.metrics</code>) and information of the type of alleles (slot
loc.all e.g. 'G/A') and the position of the SNP in slot position of the
&ldquo;'genlight&ldquo;' object (see testset.gl@position and testset.gl@loc.all for
how to format these slots.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2bpp(
  x,
  method = 1,
  outfile = "output_bpp.txt",
  imap = "Imap.txt",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2bpp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_method">method</code></td>
<td>
<p>One of 1 | 2, see details [default = 1].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_outfile">outfile</code></td>
<td>
<p>Name of the saved sequence alignment file [&quot;output_bpp.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_imap">imap</code></td>
<td>
<p>Name of the saved Imap file [&quot;Imap.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It's important to keep in mind that analyses based on coalescent theory, 
like those done by the programme BPP, are meant to be used with sequence
data. In this type of data, large chunks of DNA are sequenced, so when we
find polymorphic sites along the sequence, we know they are all on the same
chromosome. This kind of data, in which we know which chromosome each 
allele comes from, is called &quot;phased data.&quot; Most data from reduced 
representation genome-sequencing methods, like DArTseq, is unphased, 
which means that we don't know which chromosome each allele comes from. 
So, if we apply coalescence theory to data that is not phased, we will get
biased results. As in Ellegren et al., one way to deal with this is to 
&quot;haplodize&quot; each genotype by randomly choosing one allele from 
heterozygous genotypes (2012) by using method = 2.
</p>
<p>Be mindful that there is little information in the literature on the
validity of this method.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Ellegren, Hans, et al. &quot;The genomic landscape of species divergence in 
Ficedula flycatchers.&quot; Nature 491.7426 (2012): 756-760.
</p>
</li>
<li><p> Flouri T., Jiao X., Rannala B., Yang Z. (2018) Species Tree Inference 
with BPP using Genomic Sequences and the Multispecies Coalescent. Molecular
Biology and Evolution, 35(10):2585-2593. doi:10.1093/molbev/msy147
</p>
</li></ul>



<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(dartR.data)
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
test &lt;- gl.filter.monomorphs(test)
test &lt;- gl.subsample.loc(test,n=25)
gl2bpp(x = test, outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2demerelate'>Creates a dataframe suitable for input to package {Demerelate} from a
genlight {adegenet} object</h2><span id='topic+gl2demerelate'></span>

<h3>Description</h3>

<p>Creates a dataframe suitable for input to package {Demerelate} from a
genlight {adegenet} object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2demerelate(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2demerelate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2demerelate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe suitable as input to package {Demerelate}
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- gl2demerelate(testset.gl)

</code></pre>

<hr>
<h2 id='gl2eigenstrat'>Converts a genlight object into eigenstrat format</h2><span id='topic+gl2eigenstrat'></span>

<h3>Description</h3>

<p>The output of this function are three files:
</p>

<ul>
<li><p> genotype file: contains genotype data for each individual at each SNP
with an extension 'eigenstratgeno.'
</p>
</li>
<li><p> snp file: contains information about each SNP with an extension 'snp.'
</p>
</li>
<li><p> indiv file: contains information about each individual with an
extension 'ind.'
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>gl2eigenstrat(
  x,
  outfile = "gl_eigenstrat",
  outpath = NULL,
  snp.pos = 1,
  snp.chr = 1,
  pos.cM = 0,
  sex.code = "unknown",
  phen.value = "Case",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2eigenstrat_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_eigenstrat'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_snp.pos">snp.pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_snp.chr">snp.chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_pos.cm">pos.cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_sex.code">sex.code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default 'unknown'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_phen.value">phen.value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value ('Case', 'Control') [default 'Case'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Eigenstrat only accepts chromosomes coded as numeric values, as follows:
X chromosome is encoded as 23, Y is encoded as 24, mtDNA is encoded as
90, and XY is encoded as 91. SNPs with illegal chromosome values, such
as 0, will be removed.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Patterson, N., Price, A. L., &amp; Reich, D. (2006). Population structure
and eigenanalysis. PLoS genetics, 2(12), e190.
</p>
</li>
<li><p> Price, A. L., Patterson, N. J., Plenge, R. M., Weinblatt, M. E.,
Shadick, N. A., &amp; Reich, D. (2006). Principal components analysis corrects
for stratification in genome-wide association studies. Nature genetics,
38(8), 904-909.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
gl2eigenstrat(platypus.gl,snp.pos='ChromPos_Platypus_Chrom_NCBIv1',
snp.chr = 'Chrom_Platypus_Chrom_NCBIv1', outpath=tempdir())


</code></pre>

<hr>
<h2 id='gl2fasta'>Concatenates DArT trimmed sequences and outputs a FASTA file</h2><span id='topic+gl2fasta'></span>

<h3>Description</h3>

<p>Concatenated sequence tags are useful for phylogenetic methods where
information on base frequencies and transition and transversion ratios are
required (for example, Maximum Likelihood methods). Where relevant,
heterozygous loci are resolved before concatenation by either assigning
ambiguity codes or by random allele assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2fasta(
  x,
  method = 1,
  outfile = "output.fasta",
  outpath = tempdir(),
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2fasta_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_method">method</code></td>
<td>
<p>One of 1 | 2 | 3 | 4. Type method=0 for a list of options 
[method=1].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_outfile">outfile</code></td>
<td>
<p>Name of the output file (fasta format) [&quot;output.fasta&quot;].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file (set to tempdir by default)</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_probar">probar</code></td>
<td>
<p>If TRUE, a progress bar will be displayed for long loops 
[default = TRUE].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Four methods are employed:
</p>
<p>Method 1 &ndash; heterozygous positions are replaced by the standard ambiguity 
codes. The resultant sequence fragments are concatenated across loci to
generate a single combined sequence to be used in subsequent ML phylogenetic
analyses.
</p>
<p>Method 2 &ndash; the heterozygous state is resolved by randomly assigning one or
the other SNP variant to the individual. The resultant sequence fragments are
concatenated across loci to generate a single composite haplotype to be used
in subsequent ML phylogenetic analyses.
</p>
<p>Method 3 &ndash; heterozygous positions are replaced by the standard ambiguity
codes. The resultant SNP bases are concatenated across loci to generate a
single combined sequence to be used in subsequent MP phylogenetic analyses.
</p>
<p>Method 4 &ndash; the heterozygous state is resolved by randomly assigning one or
the other SNP variant to the individual. The resultant SNP bases are
concatenated across loci to generate a single composite haplotype to be used
in subsequent MP phylogenetic analyses.
</p>
<p>Trimmed sequences for which the SNP has been trimmed out, rarely, by adapter
mis-identity are deleted.
</p>
<p>The script writes out the composite haplotypes for each individual as a
fastA file. Requires 'TrimmedSequence' to be among the locus metrics
(<code>@other$loc.metrics</code>) and information of the type of alleles (slot
loc.all e.g. 'G/A') and the position of the SNP in slot position of the
&ldquo;'genlight&ldquo;' object (see testset.gl@position and testset.gl@loc.all for
how to format these slots.)
</p>


<h3>Value</h3>

<p>A new gl object with all loci rendered homozygous.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl2faststructure'>Converts a genlight object into faststructure format (to run faststructure
elsewhere)</h2><span id='topic+gl2faststructure'></span>

<h3>Description</h3>

<p>Recodes in the quite specific faststructure format (e.g first six columns
need to be there, but are ignored...check faststructure documentation
(if you find any :-( )))
The script writes out the a file in faststructure format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2faststructure(
  x,
  outfile = "gl.str",
  outpath = NULL,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2faststructure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;gl.str&quot;].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_probar">probar</code></td>
<td>
<p>Switch to show/hide progress bar [default FALSE].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>

<hr>
<h2 id='gl2gds'>Converts a genlight object into gds format</h2><span id='topic+gl2gds'></span>

<h3>Description</h3>

<p>Package SNPRelate relies on a bit-level representation of a SNP dataset that
competes with {adegenet} genlight objects and associated files. This
function converts a genlight object to a gds format file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2gds(
  x,
  outfile = "gl_gds.gds",
  outpath = NULL,
  snp.pos = "0",
  snp.chr = "0",
  chr.format = "character",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2gds_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'gl_gds.gds'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_snp.pos">snp.pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_snp.chr">snp.chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_chr.format">chr.format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function orders the SNPS by chromosome and by position before converting
to SNPRelate format, as required by this package.
The chromosome of each SNP can be a character or numeric, as described in the
vignette of SNPRelate:
'snp.chromosome, an integer or character mapping for each chromosome.
Integer: numeric values 1-26, mapped in order from 1-22, 23=X, 24=XY
(the pseudoautosomal region), 25=Y, 26=M (the mitochondrial probes), and 0
for probes with unknown positions; it does not allow NA. Character: “X”,
“XY”, “Y” and “M” can be used here, and a blank string indicating unknown
position.'
When using some functions from package SNPRelate with datasets other than
humans it might be necessary to use the option autosome.only=FALSE to avoid
detecting chromosome coding. So, it is important to read the documentation of
the function before using it.
The chromosome information for unmapped SNPS is coded as 0, as required by
SNPRelate.
Remember to close the GDS file before working in a different GDS object with
the function <a href="SNPRelate.html#topic+snpgdsClose">snpgdsClose</a> (package SNPRelate).
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
gl2gds(platypus.gl,snp.pos='ChromPos_Platypus_Chrom_NCBIv1',
snp.chr = 'Chrom_Platypus_Chrom_NCBIv1', outpath=tempdir())


</code></pre>

<hr>
<h2 id='gl2genalex'>Converts a genlight object into a format suitable for input to genalex</h2><span id='topic+gl2genalex'></span>

<h3>Description</h3>

<p>The output csv file contains the snp data and other relevant lines suitable
for genalex. This function is a wrapper for  <a href="poppr.html#topic+genind2genalex">genind2genalex</a>
(package poppr).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2genalex(
  x,
  outfile = "genalex.csv",
  outpath = NULL,
  overwrite = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2genalex_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_outfile">outfile</code></td>
<td>
<p>Name of the output file (including extension)
[default 'genalex.csv'].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE and filename exists, then the file will not be
overwritten [default FALSE].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos, Author: Katrin Hohwieler, wrapper Arthur
Georges (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Peakall, R. and Smouse P.E. (2012) GenAlEx 6.5: genetic analysis
in Excel. Population genetic software for teaching and research-an update.
Bioinformatics 28, 2537-2539.
http://bioinformatics.oxfordjournals.org/content/28/19/2537
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gl2genalex(testset.gl, outfile='testset.csv', outpath=tempdir())


</code></pre>

<hr>
<h2 id='gl2genepop'>Converts a genlight object into genepop format (and file)</h2><span id='topic+gl2genepop'></span>

<h3>Description</h3>

<p>The genepop format is used by several external applications (for example
Neestimator2
(<a href="http://www.molecularfisherieslaboratory.com.au/neestimator-software/">http://www.molecularfisherieslaboratory.com.au/neestimator-software/</a>).
So the main idea is to create the genepop file and then run the other
software externally. As a feature, the genepop file is also returned as an
invisible data.frame by the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2genepop(
  x,
  outfile = "genepop.gen",
  outpath = NULL,
  pop.order = "alphabetic",
  output.format = "2_digits",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2genepop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'genepop.gen'].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_pop.order">pop.order</code></td>
<td>
<p>Order of the output populations either &quot;alphabetic&quot; or a 
vector of population names in the order required by the user (see examples)
[default &quot;alphabetic&quot;].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_output.format">output.format</code></td>
<td>
<p>Whether to use a 2-digit format (&quot;2_digits&quot;) or 3-digits
format (&quot;3_digits&quot;) [default &quot;2_digits&quot;].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible data frame in genepop format
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
# SNP data
geno &lt;- gl2genepop(possums.gl[1:3,1:9], outpath = tempdir())
head(geno)
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
popNames(test)
gl2genepop(test, pop.order = c("TENTERFIELD","SEVERN_ABOVE","SEVERN_BELOW"),
           output.format="3_digits", outpath = tempdir())


</code></pre>

<hr>
<h2 id='gl2geno'>Converts a genlight object to geno format from package LEA</h2><span id='topic+gl2geno'></span>

<h3>Description</h3>

<p>The function converts a genlight object (SNP or presence/absence
i.e. SilicoDArT data) into a file in the 'geno' and the 'lfmm' formats from 
(package LEA).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2geno(x, outfile = "gl_geno", outpath = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2geno_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_geno'].</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
gl2geno(testset.gl, outpath=tempdir())
# Tag P/A data
gl2geno(testset.gs, outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2gi'>Converts a genind object into a genlight object</h2><span id='topic+gl2gi'></span><span id='topic+gi2gl'></span>

<h3>Description</h3>

<p>Converts a genind object into a genlight object
</p>
<p>Converts a genlight object to genind object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gi2gl(gi, parallel = FALSE, verbose = NULL)

gl2gi(x, probar = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2gi_+3A_gi">gi</code></td>
<td>
<p>A genind object [required].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_parallel">parallel</code></td>
<td>
<p>Switch to deactivate parallel version. It might not be worth
to run it parallel most of the times [default FALSE].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_x">x</code></td>
<td>
<p>A genlight object [required].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_probar">probar</code></td>
<td>
<p>If TRUE, a progress bar will be displayed for long loops
[default TRUE].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Be aware due to ambiguity which one is the reference allele a combination of
gi2gl(gl2gi(gl)) does not return an identical object (but in terms of
analysis this conversions are equivalent)
</p>
<p>This function uses a faster version of df2genind (from the adegenet
package)
</p>


<h3>Value</h3>

<p>A genlight object, with all slots filled.
</p>
<p>A genind object, with all slots filled.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>
<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>

<hr>
<h2 id='gl2hiphop'>Converts a genlight objects into hiphop format</h2><span id='topic+gl2hiphop'></span>

<h3>Description</h3>

<p>This function exports genlight objects to the format used by the parentage
assignment R package hiphop. Hiphop can be used for paternity and maternity
assignment and outperforms conventional methods where closely related
individuals occur in the pool of possible parents. The method compares the
genotypes of offspring with any combination of potentials parents and scores
the number of mismatches of these individuals at bi-allelic genetic markers
(e.g. Single Nucleotide Polymorphisms).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2hiphop(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2hiphop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2hiphop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe containing all the genotyped individuals (offspring and
potential parents) and their genotypes scored using bi-allelic markers.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Cockburn, A., Penalba, J.V.,Jaccoud, D.,Kilian, A., Brouwer, L.,Double, M.C.,
Margraf, N., Osmond, H.L., van de Pol, M. and Kruuk, L.E.B.(in revision).
HIPHOP: improved paternity assignment among close relatives using a simple
exclusion method for bi-allelic markers. Molecular Ecology Resources, DOI to
be added upon acceptance
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- gl2hiphop(testset.gl)


</code></pre>

<hr>
<h2 id='gl2phylip'>Creates a Phylip input distance matrix from a genlight (SNP) {adegenet}
object</h2><span id='topic+gl2phylip'></span>

<h3>Description</h3>

<p>This function calculates and returns a matrix of Euclidean distances between 
populations and produces an input file for the phylogenetic program Phylip 
(Joe Felsenstein).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2phylip(
  x,
  outfile = "phyinput.txt",
  outpath = tempdir(),
  bstrap = 1,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2phylip_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_outfile">outfile</code></td>
<td>
<p>Name of the file to become the input file for phylip
[default &quot;phyinput.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file 
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_bstrap">bstrap</code></td>
<td>
<p>Number of bootstrap replicates [default 1].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of Euclidean distances between populations.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- gl2phylip(testset.gl, outfile='test.txt', bstrap=10)


</code></pre>

<hr>
<h2 id='gl2plink'>Converts a genlight object into PLINK format</h2><span id='topic+gl2plink'></span>

<h3>Description</h3>

<p>This function exports a genlight object into PLINK format and save it into a
file.
This function produces the following PLINK files: bed, bim, fam, ped and map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2plink(
  x,
  plink.bin.path = getwd(),
  bed.files = FALSE,
  outfile = "gl_plink",
  outpath = NULL,
  chr.format = "character",
  pos.cM = "0",
  ID.dad = "0",
  ID.mum = "0",
  sex.code = "unknown",
  phen.value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2plink_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_plink.bin.path">plink.bin.path</code></td>
<td>
<p>Path of PLINK binary file [default getwd()].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_bed.files">bed.files</code></td>
<td>
<p>Whether create PLINK files .bed, .bim and .fam
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_plink'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_chr.format">chr.format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_pos.cm">pos.cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_id.dad">ID.dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_id.mum">ID.mum</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_sex.code">sex.code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown'). Sex information needs 
just to start with an &quot;F&quot; or &quot;f&quot; for females, with an &quot;M&quot; or &quot;m&quot; for males 
and with a &quot;U&quot;, &quot;u&quot; or being empty if the sex is unknown [default 'unknown'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_phen.value">phen.value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To create PLINK files .bed, .bim and .fam (bed.files = TRUE), it is necessary
to download the binary file of PLINK 1.9 and provide its path (plink.bin.path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
After downloading, unzip the file, access the unzipped folder and move the 
binary file (&quot;plink&quot;) to your working directory.
If you are using a Mac, you might need to open the binary first to grant 
access to the binary. 
The chromosome of each SNP can be a character or numeric. The chromosome
information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop.
Within-family ID (cannot be '0') is taken from indNames(x).
Variant identifier is taken from locNames(x).
SNP position is taken from the accessor x$position.
Chromosome name is taken from the accessor x$chromosome
Note that if names of populations or individuals contain spaces, they are 
replaced by an underscore &quot;_&quot;.
If you like to use chromosome information when converting to plink format and
your chromosome names are not from human, you need to change the chromosome 
names as 'contig1', 'contig2', etc. as described in the section &quot;Nonstandard
chromosome IDs&quot; in the following link:
https://www.cog-genomics.org/plink/1.9/input
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Purcell, Shaun, et al. 'PLINK: a tool set for whole-genome association and
population-based linkage analyses.' The American journal of human genetics
81.3 (2007): 559-575.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
test &lt;- platypus.gl
# assigning SNP position
test$position &lt;- test$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
# assigning a dummy name for chromosomes
test$chromosome &lt;- as.factor("1")
gl2plink(test, outpath=tempdir())


</code></pre>

<hr>
<h2 id='gl2related'>Converts a genlight object to format suitable to be run with Coancestry</h2><span id='topic+gl2related'></span>

<h3>Description</h3>

<p>The output txt file contains the SNP data and an additional column with the
names of the individual. The file then can be used and loaded into coancestry
or - if installed - run with the related package. Be aware the related
package was crashing in previous versions, but in general is using the same
code as coancestry and therefore should have identical results. Also running
coancestry with thousands of SNPs via the GUI seems to be not reliable and
therefore for comparisons between coancestry and related we suggest to use
the command line version of coancestry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2related(
  x,
  outfile = "related.txt",
  outpath = NULL,
  save = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2related_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'related.txt'].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_save">save</code></td>
<td>
<p>A switch if you want to save the file or not. This might be
useful for someone who wants to use the coancestry function to calculate
relatedness and not export to coancestry. See the example below
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that can be used to run with the related package
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Jack Pew, Jinliang Wang, Paul Muir and Tim Frasier (2014).
related: related: an R package for analyzing pairwise relatedness
data based on codominant molecular markers.
R package version 0.8/r2.
<a href="https://R-Forge.R-project.org/projects/related/">https://R-Forge.R-project.org/projects/related/</a>
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gtd &lt;- gl2related(bandicoot.gl[1:10,1:20], save=FALSE, )
## Not run: 
##running with the related package, use
#install.packages('related', repos='http://R-Forge.R-project.org')
library(related)
coan &lt;- coancestry(gtd, wang=1)
head(coan$relatedness)
##check ?coancestry for information how to use the function.

## End(Not run)

</code></pre>

<hr>
<h2 id='gl2sa'>Converts genlight objects to the format used in the SNPassoc package</h2><span id='topic+gl2sa'></span>

<h3>Description</h3>

<p>This function exports a genlight object into a SNPassoc object. See package
SNPassoc for details. This function needs package SNPassoc. At the time of
writing (August 2020) the package was no longer available from CRAN. To
install the package check their github repository.
<a href="https://github.com/isglobal-brge/SNPassoc">https://github.com/isglobal-brge/SNPassoc</a> and/or use
<code>install_github('isglobal-brge/SNPassoc')</code> to install the function and
uncomment the function code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2sa(x, installed = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2sa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2sa_+3A_installed">installed</code></td>
<td>
<p>Switch to run the function once SNPassoc package i
s installed [default FALSE].#' @param verbose Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl2sa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class 'snp' to be used with SNPassoc.
</p>


<h3>Author(s)</h3>

<p>Bernd Guber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Gonzalez, J.R., Armengol, L., Sol?, X., Guin?, E., Mercader, J.M., Estivill,
X. and Moreno, V. (2017). SNPassoc: an R package to perform whole genome
association studies. Bioinformatics 23:654-655.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>

<hr>
<h2 id='gl2snapp'>Converts a genlight object to nexus format suitable for phylogenetic analysis
by SNAPP (via BEAUti)
@family linker</h2><span id='topic+gl2snapp'></span>

<h3>Description</h3>

<p>The output nexus file contains the SNP data and relevant PAUP command lines
suitable for BEAUti.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2snapp(x, outfile = "snapp.nex", outpath = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2snapp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;snapp.nex&quot;].</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Bryant, D., Bouckaert, R., Felsenstein, J., Rosenberg, N.A. and
RoyChoudhury, A. (2012). Inferring species trees directly from biallelic
genetic markers: bypassing gene trees in a full coalescent analysis.
Molecular Biology and Evolution 29:1917-1932.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2snapp(testset.gl, outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2structure'>Converts a genlight object to STRUCTURE formatted files</h2><span id='topic+gl2structure'></span>

<h3>Description</h3>

<p>This function exports genlight objects to STRUCTURE formatted files (be aware
there is a gl2faststructure version as well). It is based on the code
provided by Lindsay Clark (see
<a href="https://github.com/lvclark/R_genetics_conv">https://github.com/lvclark/R_genetics_conv</a>) and this function is
basically a wrapper around her numeric2structure function. See also: Lindsay
Clark. (2017, August 22). lvclark/R_genetics_conv: R_genetics_conv 1.1
(Version v1.1). Zenodo: doi.org/10.5281/zenodo.846816.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2structure(
  x,
  ind.names = NULL,
  add.columns = NULL,
  ploidy = 2,
  export.marker.names = TRUE,
  outfile = "gl.str",
  outpath = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data and location
data, lat longs [required].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_ind.names">ind.names</code></td>
<td>
<p>Specify individuals names to be added 
[if NULL, defaults to ind.names(x)].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_add.columns">add.columns</code></td>
<td>
<p>Additional columns to be added before genotypes 
[default NULL].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_ploidy">ploidy</code></td>
<td>
<p>Set the ploidy [defaults 2].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_export.marker.names">export.marker.names</code></td>
<td>
<p>If TRUE, locus names locNames(x) will be included 
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension) 
[default &quot;gl.str&quot;].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (wrapper) and Lindsay V. Clark [lvclark@illinois.edu]; 
Custodian Bernd Gruber
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2structure(testset.gl[1:10,1:50], outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2svdquartets'>Converts a genlight object to nexus format PAUP SVDquartets</h2><span id='topic+gl2svdquartets'></span>

<h3>Description</h3>

<p>The output nexus file contains the SNP data in one of two forms, depending
upon what you regard as most appropriate. One form, that used by Chifman and
Kubatko, has two lines per individual, one providing the reference SNP the
second providing the alternate SNP (method=1).
A second form, recommended by Dave Swofford, has a single line per
individual, resolving heterozygous SNPs by replacing them with standard
ambiguity codes (method=2).
If the data are tag presence/absence, then method=2 is assumed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2svdquartets(
  x,
  outfile = "svd.nex",
  outpath = NULL,
  method = 2,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2svdquartets_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data or tag P/A data
[required].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'svd.nex'].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_method">method</code></td>
<td>
<p>Method = 1, nexus file with two lines per individual; method =
2, nexus file with one line per individual, ambiguity codes for SNP
genotypes, 0 or 1 for presence/absence data [default 2].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Chifman, J. and L. Kubatko. 2014. Quartet inference from SNP data
under the coalescent. Bioinformatics 30: 3317-3324
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gg &lt;- testset.gl[1:20,1:100]
gg@other$loc.metrics &lt;- gg@other$loc.metrics[1:100,]
gl2svdquartets(gg, outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2treemix'>Converts a genlight object to a treemix input file</h2><span id='topic+gl2treemix'></span>

<h3>Description</h3>

<p>The output file contains the SNP data in the format expected by treemix &ndash;
see the treemix manual. The file will be gzipped before in order to be
recognised by treemix. Plotting functions provided with treemix will need to
be sourced from the treemix download page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2treemix(x, outfile = "treemix_input.gz", outpath = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2treemix_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including gz extension)
[default 'treemix_input.gz'].</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
</p>
<p>[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Pickrell and Pritchard (2012). Inference of population splits and
mixtures from genome-wide allele frequency data. PLoS Genetics
https://doi.org/10.1371/journal.pgen.1002967
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2vcf">gl2vcf</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2treemix(testset.gl, outpath=tempdir())

</code></pre>

<hr>
<h2 id='gl2vcf'>Converts a genlight object into vcf format</h2><span id='topic+gl2vcf'></span>

<h3>Description</h3>

<p>This function exports a genlight object into VCF format and save it into a
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2vcf(
  x,
  plink.bin.path = getwd(),
  outfile = "gl_vcf",
  outpath = NULL,
  snp.pos = "0",
  snp.chr = "0",
  chr.format = "character",
  pos.cM = "0",
  ID.dad = "0",
  ID.mum = "0",
  sex.code = "unknown",
  phen.value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2vcf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_plink.bin.path">plink.bin.path</code></td>
<td>
<p>Path of PLINK binary file [default getwd())].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_vcf'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default global working 
directory or if not specified, tempdir()].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_snp.pos">snp.pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_snp.chr">snp.chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_chr.format">chr.format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_pos.cm">pos.cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_id.dad">ID.dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_id.mum">ID.mum</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_sex.code">sex.code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default  'unknown'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_phen.value">phen.value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function requires to download the binary file of PLINK 1.9 and provide
its path (plink.bin.path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
The chromosome information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop
Within-family ID (cannot be '0') is taken from indNames(x)
Variant identifier is taken from locNames(x)
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M.
A., ... &amp; 1000 Genomes Project Analysis Group. (2011). The variant call
format and VCFtools. Bioinformatics, 27(15), 2156-2158.
</p>


<h3>See Also</h3>

<p>Other linker: 
<code><a href="#topic+gl2bayesAss">gl2bayesAss</a>()</code>,
<code><a href="#topic+gl2bayescan">gl2bayescan</a>()</code>,
<code><a href="#topic+gl2bpp">gl2bpp</a>()</code>,
<code><a href="#topic+gl2demerelate">gl2demerelate</a>()</code>,
<code><a href="#topic+gl2eigenstrat">gl2eigenstrat</a>()</code>,
<code><a href="#topic+gl2faststructure">gl2faststructure</a>()</code>,
<code><a href="#topic+gl2gds">gl2gds</a>()</code>,
<code><a href="#topic+gl2genalex">gl2genalex</a>()</code>,
<code><a href="#topic+gl2genepop">gl2genepop</a>()</code>,
<code><a href="#topic+gl2geno">gl2geno</a>()</code>,
<code><a href="#topic+gl2gi">gl2gi</a>()</code>,
<code><a href="#topic+gl2hiphop">gl2hiphop</a>()</code>,
<code><a href="#topic+gl2phylip">gl2phylip</a>()</code>,
<code><a href="#topic+gl2plink">gl2plink</a>()</code>,
<code><a href="#topic+gl2related">gl2related</a>()</code>,
<code><a href="#topic+gl2sa">gl2sa</a>()</code>,
<code><a href="#topic+gl2structure">gl2structure</a>()</code>,
<code><a href="#topic+gl2treemix">gl2treemix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#this example needs plink installed to work
require("dartR.data")
gl2vcf(platypus.gl,snp.pos='ChromPos_Platypus_Chrom_NCBIv1',
 snp.chr = 'Chrom_Platypus_Chrom_NCBIv1')

## End(Not run)

</code></pre>

<hr>
<h2 id='possums.gl'>A simulated genlight object created to run a landscape genetic example
This a test data set to run a landscape genetics example. It contains 10 
populations of 30 individuals each and each individual has 300 loci. There 
are no covariates for individuals or loci.</h2><span id='topic+possums.gl'></span>

<h3>Description</h3>

<p>A simulated genlight object created to run a landscape genetic example
This a test data set to run a landscape genetics example. It contains 10 
populations of 30 individuals each and each individual has 300 loci. There 
are no covariates for individuals or loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possums.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='rbind.dartR'>adjust rbind for dartR</h2><span id='topic+rbind.dartR'></span>

<h3>Description</h3>

<p>rbind is a bit lazy and does not take care for the metadata (so data in the
other slot is lost). You can get most of the loci metadata back using
gl.compliance.check.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dartR'
rbind(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind.dartR_+3A_...">...</code></td>
<td>
<p>list of dartR objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t1 &lt;- platypus.gl
class(t1) &lt;- "dartR"
t2 &lt;- rbind(t1[1:5,],t1[6:10,])
</code></pre>

<hr>
<h2 id='testset.gl'>A genlight object created via the gl.read.dart function
This is a test data set on turtles. 250 individuals, 255 loci in &gt;30 
populations.</h2><span id='topic+testset.gl'></span>

<h3>Description</h3>

<p>A genlight object created via the gl.read.dart function
This is a test data set on turtles. 250 individuals, 255 loci in &gt;30 
populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testset.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='testset.gs'>A genlight object created via the gl.read.silicodart function
This is a test data set on turtles. 218 individuals, 255 loci in &gt;30 
populations.</h2><span id='topic+testset.gs'></span>

<h3>Description</h3>

<p>A genlight object created via the gl.read.silicodart function
This is a test data set on turtles. 218 individuals, 255 loci in &gt;30 
populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testset.gs
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='theme_dartR'>Default theme for dartR plots</h2><span id='topic+theme_dartR'></span>

<h3>Description</h3>

<p>This is the theme used as default for dartR plots.
This function controls all non-data display elements in the plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_dartR(
  base_size = 11,
  base_family = "",
  base_line_size = base_size/22,
  base_rect_size = base_size/22
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_dartR_+3A_base_size">base_size</code></td>
<td>
<p>base font size, given in pts.</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_family">base_family</code></td>
<td>
<p>base font family</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_line_size">base_line_size</code></td>
<td>
<p>base size for line elements</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_rect_size">base_rect_size</code></td>
<td>
<p>base size for rect elements</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a the standard dartR theme to be used in ggplots
</p>


<h3>See Also</h3>

<p>Other environment: 
<code><a href="#topic+gl.check.verbosity">gl.check.verbosity</a>()</code>,
<code><a href="#topic+gl.check.wd">gl.check.wd</a>()</code>,
<code><a href="#topic+gl.print.history">gl.print.history</a>()</code>,
<code><a href="#topic+gl.set.wd">gl.set.wd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggplot(data.frame(dummy=rnorm(1000)),aes(dummy)) +
geom_histogram(binwidth=0.1) + theme_dartR()

</code></pre>

<hr>
<h2 id='utils.basic.stats'>Calculates mean observed heterozygosity, mean expected heterozygosity 
and FIS per locus, per population and various population differentiation 
measures
@family utilities</h2><span id='topic+utils.basic.stats'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY
END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.basic.stats(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.basic.stats_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a re-implementation of <code>hierfstat::basics.stats</code> specifically 
for genlight objects. Formula (and hence results) match exactly the original 
version of <code>hierfstat::basics.stats</code> but it is much faster.
</p>


<h3>Value</h3>

<p>A list with with the statistics for each population
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos and Carlo Pacioni (post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
out &lt;- utils.basic.stats(platypus.gl)

</code></pre>

<hr>
<h2 id='utils.check.datatype'>Utility function to check the class of an object passed to a function</h2><span id='topic+utils.check.datatype'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.check.datatype(
  x,
  accept = c("genlight", "SNP", "SilicoDArT"),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.check.datatype_+3A_x">x</code></td>
<td>
<p>Name of the genlight object, dist matrix, data matrix, glPCA, or
fixed difference list (fd) [required].</p>
</td></tr>
<tr><td><code id="utils.check.datatype_+3A_accept">accept</code></td>
<td>
<p>Vector containing the classes of objects that are to be
accepted [default c('genlight','SNP','SilicoDArT'].</p>
</td></tr>
<tr><td><code id="utils.check.datatype_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Most functions require access to a genlight object, dist matrix, data matrix
or fixed difference list (fd), and this function checks that a genlight
object or one of the above has been passed, whether the genlight object is a
SNP dataset or a SilicoDArT object, and reports back if verbosity is &gt;=2.
</p>
<p>This function checks the class of passed object and sets the datatype to
'SNP', 'SilicoDArT', 'dist', 'mat', or class[1](x) as appropriate.
Note also that this function checks to see if there are individuals or loci
scored as all missing (NA) and if so, issues the user with a warning.
Note: One and only one of gl.check, fd.check, dist.check or mat.check can be
TRUE.
</p>


<h3>Value</h3>

<p>datatype, 'SNP' for SNP data, 'SilicoDArT' for P/A data, 'dist' for a
distance matrix, 'mat' for a data matrix, 'glPCA' for an ordination file, or
class(x)[1].
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>datatype &lt;- utils.check.datatype(testset.gl)
datatype &lt;- utils.check.datatype(as.matrix(testset.gl),accept='matrix')
fd &lt;- gl.fixed.diff(testset.gl)
datatype &lt;- utils.check.datatype(fd,accept='fd')
 
</code></pre>

<hr>
<h2 id='utils.dart2genlight'>An internal function to converts DarT to genlight.</h2><span id='topic+utils.dart2genlight'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dart2genlight(
  dart,
  ind.metafile = NULL,
  covfilename = NULL,
  probar = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dart2genlight_+3A_dart">dart</code></td>
<td>
<p>A dart object created via read.dart [required].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Optional file in csv format with metadata for each
individual (see details for explanation) [default NULL].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_covfilename">covfilename</code></td>
<td>
<p>Depreciated, use parameter ind.metafile.</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a DArT file (read via <code>read.dart</code>) into an
genlight object <code><a href="adegenet.html#topic+adegenet">adegenet</a></code>. #' Internal function called by gl.read.dart().
</p>
<p>The ind.metadata file needs to have very specific headings. First a heading
called id. Here the ids have to match the ids in the dart object
<code>colnames(dart[[4]])</code>. The following column headings are optional.
pop: specifies the population membership of each individual. lat and lon
specify spatial coordinates (in decimal degrees WGS1984 format). Additional
columns with individual metadata can be imported (e.g. age, gender).
</p>


<h3>Value</h3>

<p>A genlight object. Including all available slots are filled.
loc.names, ind.names, pop, lat, lon (if provided via the ind.metadata file)
</p>


<h3>Author(s)</h3>

<p>Maintainer: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.dist.binary'>Calculates a distance matrix for individuals defined in a dartR
genlight object using binary P/A data (SilicoDArT)</h2><span id='topic+utils.dist.binary'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dist.binary(
  x,
  method = "simple",
  scale = FALSE,
  swap = FALSE,
  type = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dist.binary_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default simple].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_swap">swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_type">type</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for N11 object of class dist, matrix for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].
</p>
<p>@details
This script calculates various distances between individuals based on sequence tag
Presence/Absence data.
</p>
<p>The distance measure can be one of:
</p>

<ul>
<li><p> Euclidean &ndash; Euclidean Distance applied to cartesian coordinates defined
by the loci, scored as 0 or 1. Presence and absence equally weighted.
</p>
</li>
<li><p> simple &ndash; simple matching, both 1 or both 0 = 0; one 1 and the other
0 = 1. Presence and absence equally weighted.
</p>
</li>
<li><p> Jaccard &ndash; ignores matching 0, both 1 = 0; one 1 and the other 0 = 1.
Absences could be for different reasons.
</p>
</li>
<li><p> Bray-Curtis &ndash; both 0 = 0; both 1 = 2; one 1 and the other 0 = 1. Absences
could be for different reasons. Sometimes called the Dice or Sorensen
distance.
</p>
</li></ul>

<p>One might choose to disregard or downweight absences in comparison with
presences because the homology of absences is less clear (mutation at one or
the other, or both restriction sites). Your call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author: Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.dist.ind.snp'>Calculates a distance matrix for individuals defined in a 
genlight object using SNP data (DArTseq)</h2><span id='topic+utils.dist.ind.snp'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dist.ind.snp(
  x,
  method = "Euclidean",
  scale = FALSE,
  type = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dist.ind.snp_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default Euclidean].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_type">type</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script calculates various distances between individuals based on 
SNP genotypes.
The distance measure can be one of:
</p>

<ul>
<li><p> Euclidean &ndash; Euclidean Distance applied to Cartesian coordinates defined
by the loci, scored as 0, 1 or 2. 
</p>
</li>
<li><p> Simple &ndash; simple mismatch, 0 where no alleles are shared, 1 where one
allele is shared, 2 where both alleles are shared. 
</p>
</li>
<li><p> Absolute &ndash; absolute mismatch, 0 where no alleles are shared, 1 where
one or both alleles are shared.
</p>
</li>
<li><p> Czekanowski (or Manhattan) calculates the city block metric distance
by summing the scores on each axis (locus).
</p>
</li></ul>



<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other distance: 
<code><a href="#topic+gl.dist.ind">gl.dist.ind</a>()</code>,
<code><a href="#topic+gl.dist.pop">gl.dist.pop</a>()</code>,
<code><a href="#topic+gl.fdsim">gl.fdsim</a>()</code>
</p>

<hr>
<h2 id='utils.flag.start'>A utility script to flag the start of a script</h2><span id='topic+utils.flag.start'></span>

<h3>Description</h3>

<p>A utility script to flag the start of a script
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.flag.start(func = NULL, build = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.flag.start_+3A_func">func</code></td>
<td>
<p>Name of the function that is starting [required].</p>
</td></tr>
<tr><td><code id="utils.flag.start_+3A_build">build</code></td>
<td>
<p>Name of the build [default NULL].</p>
</td></tr>
<tr><td><code id="utils.flag.start_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>calling function name
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.hamming'>Calculates the Hamming distance between two DArT trimmed DNA sequences</h2><span id='topic+utils.hamming'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.hamming(str1, str2, r = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.hamming_+3A_str1">str1</code></td>
<td>
<p>String containing the first sequence [required].</p>
</td></tr>
<tr><td><code id="utils.hamming_+3A_str2">str2</code></td>
<td>
<p>String containing the second sequence [required].</p>
</td></tr>
<tr><td><code id="utils.hamming_+3A_r">r</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 4].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hamming distance is calculated as the number of base differences between two
sequences which can be expressed as a count or a proportion. Typically, it is
calculated between two sequences of equal length. In the context of DArT
trimmed sequences, which differ in length but which are anchored to the left
by the restriction enzyme recognition sequence, it is sensible to compare the
two trimmed sequences starting from immediately after the common recognition
sequence and terminating at the last base of the shorter sequence.
The Hamming distance between the rows of a matrix can be computed quickly
by exploiting the fact that the dot product of two binary vectors x and (1-y)
counts the corresponding elements that are different between x and y.
This matrix multiplication can also be used for matrices with more than two
possible values, and different types of elements, such as DNA sequences.
The function calculates the Hamming distance between all columns of a
matrix X, or two matrices X and Y. Again matrix multiplication is used, this
time for counting, between two columns x and y, the number of cases in which
corresponding elements have the same value (e.g. A, C, G or T). This counting
is done for each of the possible values individually, while iteratively adding
the results. The end result of the iterative adding is the sum of all
corresponding elements that are the same, i.e. the inverse of the Hamming
distance. Therefore, the last step is to subtract this end result H from the
maximum possible distance, which is the number of rows of matrix X.
If the two DNA sequences are of differing length, the longer is truncated. The
initial common restriction enzyme recognition sequence is ignored.
</p>


<h3>Value</h3>

<p>Hamming distance between the two strings
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.het.pop'>An internal function that calculates expected mean heterozygosity per population</h2><span id='topic+utils.het.pop'></span><span id='topic+ind.count'></span>

<h3>Description</h3>

<p>An internal function that calculates expected mean heterozygosity per population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ind.count(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.het.pop_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the mean expected heterozygosity for each population
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.impute'>An internal script [Custodian to provide a title]</h2><span id='topic+utils.impute'></span><span id='topic+matrix2gen'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrix2gen(snp_matrix, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.impute_+3A_snp_matrix">snp_matrix</code></td>
<td>
<p>[Custodian to provide parameter description]</p>
</td></tr>
<tr><td><code id="utils.impute_+3A_parallel">parallel</code></td>
<td>
<p>[Custodian to provide parameter description]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>#[Custodian to provide details for future you]
</p>


<h3>Value</h3>

<p>The resultant genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.is.fixed'>An internal function to tests if two populations are fixed at a given locus</h2><span id='topic+utils.is.fixed'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.is.fixed(s1, s2, tloc = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.is.fixed_+3A_s1">s1</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the first 
population [required].</p>
</td></tr>
<tr><td><code id="utils.is.fixed_+3A_s2">s2</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the second 
population [required].</p>
</td></tr>
<tr><td><code id="utils.is.fixed_+3A_tloc">tloc</code></td>
<td>
<p>Threshold value for tolerance in when a difference is regarded as
fixed [default 0].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script compares two percent allele frequencies
and reports TRUE if they represent a fixed difference, FALSE otherwise.
</p>
<p>A fixed difference at a locus occurs when two populations share no alleles,
noting that SNPs are biallelic (ploidy=2).
Tolerance in the definition of a fixed difference is provided by the t
parameter. For example, t=0.05 means that SNP allele frequencies of 95,5 and
5,95 percent will be reported as fixed (TRUE).
</p>


<h3>Value</h3>

<p>TRUE (fixed difference) or FALSE (alleles shared) or NA (one or both s1 or s2 missing)
</p>


<h3>Author(s)</h3>

<p>Maintainer: Arthur Georges (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.fixed.diff">gl.fixed.diff</a></code>
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.jackknife'>An internal function to conducts jackknife resampling using a genlight object</h2><span id='topic+utils.jackknife'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.jackknife(
  x,
  FUN,
  unit = "loc",
  recalc = FALSE,
  mono.rm = FALSE,
  n.cores = 1,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.jackknife_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_fun">FUN</code></td>
<td>
<p>the name of the function to be used to calculate the statistic</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_unit">unit</code></td>
<td>
<p>The unit to use for resampling. One of c(&quot;loc&quot;, &quot;ind&quot;, &quot;pop&quot;): 
loci, individuals or populations</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic and all NA loci [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use. If &quot;auto&quot;, it will 
use all but one available cores.</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress but not results; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_...">...</code></td>
<td>
<p>any additional arguments to be passed to FUN</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Jackknife resampling is a statistical procedure where for a dataset of sample 
size n, subsamples of size n-1 are used to compute a statistic. The collection 
of the values obtained can be used to evaluate the variability around the point 
estimate. This function can take the loci, the individuals or the populations 
as units over which to conduct resampling.
Note: when n is very small, jackknife resampling is not recommended.
Parallel computation is implemented. The argument  <code>n.cores</code> indicates the 
number of core to use. If &quot;auto&quot;, it will use all but one available 
cores. If the number of units is small (e.g. a few populations), there is not 
real advantage in using parallel computation. On the other hand, if the number 
of units is large (e.g. thousands of loci), even with parallel computation, 
this function can be very slow.
</p>


<h3>Value</h3>

<p>A list of length n where each element is the output of FUN
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
platMod.gl &lt;- gl.filter.allna(platypus.gl) 
chk.pop &lt;- utils.jackknife(x=platMod.gl, FUN="gl.alf", unit="pop", 
recalc = FALSE, mono.rm = FALSE, n.cores = 1, verbose=0)
</code></pre>

<hr>
<h2 id='utils.n.var.invariant'>An internal utility function to calculate the number of variant and invariant sites by
locus</h2><span id='topic+utils.n.var.invariant'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.n.var.invariant(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.n.var.invariant_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.n.var.invariant_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL].
</p>
<p>@details
Calculate the number of variant and invariant sites by locus and add them as
columns in <code>loc.metrics</code>. This can be useful to conduct further
filtering, for example where only loci with secondaries are wanted for
phylogenetic analyses.
Invariant sites are the sites (nucleotide) that are not polymorphic. When the
locus metadata supplied by DArT includes the sequence of the allele
(<code>TrimmedSequence</code>), it is used by this function to estimate the number
of sites that were sequenced in each tag (read). This script then subtracts
the number of polymorphic sites. The length of the trimmed sequence
(lenTrimSeq), the number of variant (n.variant) and
invariant (n.invariant) sites are the added to the table in
<code>gl@others$loc.metrics</code>.
<strong>NOTE</strong>: It is important to realise that this function correctly
estimates the number of variant and invariant sites only when it is executed on
<code>genlight</code> objects before secondaries are removed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code>,<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code>
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.plink.run'>Runs PLINK from within R</h2><span id='topic+utils.plink.run'></span>

<h3>Description</h3>

<p>Runs PLINK from within R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.plink.run(
  dir.in,
  plink.cmd = "plink",
  plink.path = "path",
  out = "hapmap1",
  syntax,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.plink.run_+3A_dir.in">dir.in</code></td>
<td>
<p>The path where the data files are</p>
</td></tr>
<tr><td><code id="utils.plink.run_+3A_plink.cmd">plink.cmd</code></td>
<td>
<p>The 'name' to call plink. This will depend on the file name 
(without the extension '.exe' if on windows) or the name of the PATH variable</p>
</td></tr>
<tr><td><code id="utils.plink.run_+3A_plink.path">plink.path</code></td>
<td>
<p>The path where the executable is. If plink is listed in
the PATH then there is no need for this. This is what the option &quot;path&quot;
means</p>
</td></tr>
<tr><td><code id="utils.plink.run_+3A_out">out</code></td>
<td>
<p>The root of the output file name</p>
</td></tr>
<tr><td><code id="utils.plink.run_+3A_syntax">syntax</code></td>
<td>
<p>the flags to pass to plink call</p>
</td></tr>
<tr><td><code id="utils.plink.run_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PLINK needs to be installed on the 
machine and syntax used need to be appropriate for the version installed.
</p>


<h3>Value</h3>

<p>A character vector with the command used for PLINK.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni and Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Purcell, Shaun, et al. 'PLINK: a tool set for whole-genome association and
population-based linkage analyses.' The American journal of human genetics
81.3 (2007): 559-575.
</p>

<hr>
<h2 id='utils.plot.save'>An internal function to save a ggplot object to disk in RDS binary format</h2><span id='topic+utils.plot.save'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.plot.save(x, dir = NULL, file = NULL, verbose = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.plot.save_+3A_x">x</code></td>
<td>
<p>Name of the ggplot object.</p>
</td></tr>
<tr><td><code id="utils.plot.save_+3A_dir">dir</code></td>
<td>
<p>Name of the directory to save the file.</p>
</td></tr>
<tr><td><code id="utils.plot.save_+3A_file">file</code></td>
<td>
<p>Name of the file to save the plot to (omit file extension)</p>
</td></tr>
<tr><td><code id="utils.plot.save_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="utils.plot.save_+3A_...">...</code></td>
<td>
<p>Parameters passed to function <a href="ggplot2.html#topic+ggsave">ggsave</a>, 
such as width and height, when the ggplot is to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An internal function to save a ggplot object to disk in RDS binary format.
Uses saveRDS() to save the file with an .RDS extension; can be reloaded with gl.load().
</p>


<h3>Value</h3>

<p>returns NULL
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.read.dart'>Utility to import DarT data to R</h2><span id='topic+utils.read.dart'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.read.dart(
  filename,
  nas = "-",
  topskip = NULL,
  lastmetric = "RepAvg",
  service.row = 1,
  plate.row = 3,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.read.dart_+3A_filename">filename</code></td>
<td>
<p>Path to file (csv file only currently) [required].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_nas">nas</code></td>
<td>
<p>A character specifying NAs [default '-'].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_topskip">topskip</code></td>
<td>
<p>A number specifying the number of rows to be skipped. If not
provided the number of rows to be skipped are 'guessed' by the number of rows
with '*' at the beginning [default NULL].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Specifies the last non genetic column [default 'RepAvg'].
Be sure to check if that is true, otherwise the number of individuals will
not match. You can also specify the last column by a number.</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_service.row">service.row</code></td>
<td>
<p>The row number in which the information of the DArT
service is contained [default 1].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_plate.row">plate.row</code></td>
<td>
<p>The row number in which the information of the plate
location is contained [default 3].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internal function called by gl.read.dart()
</p>


<h3>Value</h3>

<p>A list of length 5. #dart format (one or two rows) #individuals,
#snps, #non genetic metrics, #genetic data (still two line format, rows=snps,
columns=individuals)
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other io: 
<code><a href="#topic+gl.load">gl.load</a>()</code>,
<code><a href="#topic+gl.read.csv">gl.read.csv</a>()</code>,
<code><a href="#topic+gl.read.dart">gl.read.dart</a>()</code>,
<code><a href="#topic+gl.read.fasta">gl.read.fasta</a>()</code>,
<code><a href="#topic+gl.read.silicodart">gl.read.silicodart</a>()</code>,
<code><a href="#topic+gl.read.vcf">gl.read.vcf</a>()</code>,
<code><a href="#topic+gl.save">gl.save</a>()</code>,
<code><a href="#topic+gl.write.csv">gl.write.csv</a>()</code>
</p>

<hr>
<h2 id='utils.read.fasta'>An internal script to read a fastA file into a genlight object</h2><span id='topic+utils.read.fasta'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.read.fasta(file, parallel = parallel, n.cores = NULL, verbose = verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.read.fasta_+3A_file">file</code></td>
<td>
<p>Name of the fastA file [required]</p>
</td></tr>
<tr><td><code id="utils.read.fasta_+3A_parallel">parallel</code></td>
<td>
<p>Switch to deactivate parallel version. It might not be worth
to run it parallel most of the times [default FALSE]</p>
</td></tr>
<tr><td><code id="utils.read.fasta_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores to use in parallel [default 4]</p>
</td></tr>
<tr><td><code id="utils.read.fasta_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent, fatal errors only; 1, flag function
begin and end; 2, progress log; 3, progress and results summary; 5, full
report [default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The resultant genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.read.ped'>An internal script [Custodian to provide a title]</h2><span id='topic+utils.read.ped'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.read.ped(
  file,
  snps,
  which,
  split = "\t| +",
  sep = ".",
  na.strings = "0",
  lex.order = FALSE,
  show_warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.read.ped_+3A_file">file</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_snps">snps</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_which">which</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_split">split</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_sep">sep</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_na.strings">na.strings</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_lex.order">lex.order</code></td>
<td>
<p>Custodian to provide</p>
</td></tr>
<tr><td><code id="utils.read.ped_+3A_show_warnings">show_warnings</code></td>
<td>
<p>Custodian to provide
</p>
<p>#[Custodian to provide other parameter descriptions]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>#[Custodian to provide details]
</p>


<h3>Value</h3>

<p>The resultant genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.avgpic'>A utility function to recalculate intermediate locus metrics</h2><span id='topic+utils.recalc.avgpic'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.avgpic(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.avgpic_+3A_x">x</code></td>
<td>
<p>Name of the genlight [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.avgpic_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Recalculates OneRatioRef, OneRatioSnp, PICRef, PICSnp,
and AvgPIC by locus after some individuals or populations have been deleted.
</p>
<p>The locus metadata supplied by DArT has OneRatioRef, OneRatioSnp, PICRef,
PICSnp, and AvgPIC included, but the allelic composition will change when
some individuals,or populations, are removed from the dataset and so the
initial statistics will no longer apply. This script recalculates these
statistics and places the recalculated values in the appropriate place in
the genlight object.
If the locus metadata OneRatioRef|Snp, PICRef|Snp and/or AvgPIC do not exist,
the script creates and populates them.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.callrate'>A utility script to recalculate the callrate by locus after some populations
have been deleted</h2><span id='topic+utils.recalc.callrate'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.callrate(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the
restriction enzyme recognition sites. The locus metadata supplied by DArT has
callrate included, but the call rate will change when some individuals are
removed from the dataset. This script recalculates the callrate and places
these recalculated values in the appropriate place in the genlight object.
It sets the Call Rate flag to TRUE.
</p>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.avgpic</code> for recalculating avgPIC,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.freqhets'>A utility script to recalculate the frequency of the heterozygous SNPs by
locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhets'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhets(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhets_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhets_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The locus metadata supplied by DArT has FreqHets included, but the frequency
of the heterozygotes will change when some individuals are removed from the
dataset.
This script recalculates the FreqHets and places these recalculated values in
the appropriate place in the genlight object.
Note that the frequency of the homozygote reference SNPS is calculated from
the individuals that could be scored.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate,
<code>utils.recalc.AvgPIC</code> for recalculating RepAvg, <code>gl.recalc.maf</code> for
recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.freqhomref'>#' An internal utility function to recalculate the frequency of the homozygous reference
SNP by locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhomref'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhomref(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhomref_+3A_x">x</code></td>
<td>
<p>Name of the genlight [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhomref_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The locus metadata supplied by DArT has FreqHomRef included, but the
frequency of the homozygous reference will change when some individuals are
removed from the dataset.
This script recalculates the FreqHomRef and places these recalculated values
in the appropriate place in the genlight object.
Note that the frequency of the homozygote reference SNPS is calculated from
the individuals that could be scored.
</p>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.avgpic</code> for recalculating AvgPIC,
<code>utils.recalc.freqhomsnp</code> for recalculating frequency of homozygous
alternate, <code>utils.recalc.freqhet</code> for recalculating frequency of
heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.freqhomsnp'>A utility function to recalculate the frequency of the homozygous alternate
SNP by locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhomsnp'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhomsnp(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhomsnp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhomsnp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The locus metadata supplied by DArT has FreqHomSnp included, but the
frequency of the homozygous alternate will change when some individuals are
removed from the dataset.
This function recalculates the FreqHomSnp and places these recalculated values
in the appropriate place in the genlight object.
Note that the frequency of the homozygote alternate SNPS is calculated from
the individuals that could be scored.
This function only applies to SNP genotype data not Tag P/A data (SilicoDArT).
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.avgpic</code> for recalculating AvgPIC,
<code>utils.recalc.freqhet</code> for recalculating frequency of heterozygotes,
<code>gl.recalc.maf</code> for recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.recalc.maf'>A utility function to recalculate the minor allele frequency by locus,
typically after some populations have been deleted</h2><span id='topic+utils.recalc.maf'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.maf(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The locus metadata supplied by DArT does not have MAF included, so it is
calculated and added to the locus.metadata by this script. The minimum allele
frequency will change when some individuals are removed from the dataset.
This script recalculates the MAF and places these recalculated values in the
appropriate place in the genlight object.
This function only applies to SNP genotype data.
</p>


<h3>Value</h3>

<p>The modified genlight dataset.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.avgpic</code> for recalculating AvgPIC,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>

<hr>
<h2 id='utils.reset.flags'>#' An internal utility function to reset to FALSE (or TRUE) the locus metric flags after
some individuals or populations have been deleted.</h2><span id='topic+utils.reset.flags'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.reset.flags(x, set = FALSE, value = 2, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.reset.flags_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_set">set</code></td>
<td>
<p>Set the flags to TRUE or FALSE [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_value">value</code></td>
<td>
<p>Set the default verbosity for all functions, where verbosity is
not specified [default 2].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The locus metadata supplied by DArT has OneRatioRef, OneRatioSnp, PICRef,
PICSnp, and AvgPIC included, but the allelic composition will change when
some individuals are removed from the dataset and so the initial statistics
will no longer apply. This applies also to some variable calculated by dartR
(e.g. maf). This script resets the locus metrics flags to FALSE to indicate
that these statistics in the genlight object are no longer current. The
verbosity default is also set, and in the case of SilcoDArT, the flags PIC
and OneRatio are also set.
If the locus metrics do not exist then they are added to the genlight object
but not populated. If the locus metrics flags do not exist, then they are
added to the genlight object and set to FALSE (or TRUE).
</p>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele 
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>
<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.transpose">utils.transpose</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- utils.reset.flags(testset.gl)

</code></pre>

<hr>
<h2 id='utils.transpose'>An internal utility function to transpose a genlight object.</h2><span id='topic+utils.transpose'></span>

<h3>Description</h3>

<p>WARNING: UTILITY SCRIPTS ARE FOR INTERNAL USE ONLY AND SHOULD NOT BE USED BY END USERS AS THEIR USE OUT OF CONTEXT COULD LEAD TO UNPREDICTABLE OUTCOMES.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.transpose(x, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.transpose_+3A_x">x</code></td>
<td>
<p>name of the genlight object</p>
</td></tr>
<tr><td><code id="utils.transpose_+3A_parallel">parallel</code></td>
<td>
<p>if TRUE, use parallel processing capability</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a function to transpose a genlight object, that is, to set loci as
entities and individuals as attributes.
</p>


<h3>Value</h3>

<p>a transposed genlight object
</p>


<h3>See Also</h3>

<p>Other utilities: 
<code><a href="#topic+gl.alf">gl.alf</a>()</code>,
<code><a href="#topic+utils.check.datatype">utils.check.datatype</a>()</code>,
<code><a href="#topic+utils.dart2genlight">utils.dart2genlight</a>()</code>,
<code><a href="#topic+utils.dist.binary">utils.dist.binary</a>()</code>,
<code><a href="#topic+utils.flag.start">utils.flag.start</a>()</code>,
<code><a href="#topic+utils.hamming">utils.hamming</a>()</code>,
<code><a href="#topic+utils.het.pop">utils.het.pop</a>()</code>,
<code><a href="#topic+utils.impute">utils.impute</a></code>,
<code><a href="#topic+utils.is.fixed">utils.is.fixed</a>()</code>,
<code><a href="#topic+utils.jackknife">utils.jackknife</a>()</code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a>()</code>,
<code><a href="#topic+utils.plot.save">utils.plot.save</a>()</code>,
<code><a href="#topic+utils.read.fasta">utils.read.fasta</a>()</code>,
<code><a href="#topic+utils.read.ped">utils.read.ped</a>()</code>,
<code><a href="#topic+utils.recalc.avgpic">utils.recalc.avgpic</a>()</code>,
<code><a href="#topic+utils.recalc.callrate">utils.recalc.callrate</a>()</code>,
<code><a href="#topic+utils.recalc.freqhets">utils.recalc.freqhets</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomref">utils.recalc.freqhomref</a>()</code>,
<code><a href="#topic+utils.recalc.freqhomsnp">utils.recalc.freqhomsnp</a>()</code>,
<code><a href="#topic+utils.recalc.maf">utils.recalc.maf</a>()</code>,
<code><a href="#topic+utils.reset.flags">utils.reset.flags</a>()</code>
</p>

<hr>
<h2 id='zzz'>Setting up the package
Setting theme, colors and verbosity</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting up the package
Setting theme, colors and verbosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
