<!DOCTYPE html><html><head><title>Help for package dartR.spatial</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR.spatial}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gl.costdistances'><p>Calculates cost distances for a given landscape (resistance matrix)</p></a></li>
<li><a href='#gl.genleastcost'><p>Performs least-cost path analysis based on a friction matrix</p></a></li>
<li><a href='#gl.grm2'><p>Calculates an identity by descent matrix</p></a></li>
<li><a href='#gl.ibd'><p>Performs isolation by distance analysis</p></a></li>
<li><a href='#gl.kosman'><p>Calculates a Kosman distance matrix for each pair of individuals</p></a></li>
<li><a href='#gl.spatial.autoCorr'><p>Spatial autocorrelation following Smouse and Peakall 1999</p></a></li>
<li><a href='#gl2shp'><p>Converts a genlight object to ESRI shapefiles or kml files</p></a></li>
<li><a href='#utils.spautocor'><p>Spatial autocorrelation coefficient calculations</p></a></li>
<li><a href='#zzz'><p>Setting up the package dartR.spatial</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Applying Landscape Genomic Methods on 'SNP' and 'Silicodart'
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.78</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-15</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides landscape genomic functions to analyse 'SNP' (single nuclear
    polymorphism) data, such as least cost path analysis and isolation by distance. 
    Therefore each sample needs to have coordinate data attached (lat/lon) to be 
    able to run most of the functions. 'dartR.spatial' is a package that belongs
    to the 'dartRverse' suit of packages and depends on 'dartR.base' and 'dartR.data'.</td>
</tr>
<tr>
<td>Revision:</td>
<td>Attractive Abalone</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), adegenet (&ge; 2.0.0), dartR.base, dartR.data</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, utils, crayon, StAMPP, raster, sp, tidyr, vegan,
MASS, ggplot2, data.table</td>
</tr>
<tr>
<td>Suggests:</td>
<td>mmod, dismo, gdistance, gplots, rrBLUP, terra (&ge; 1.7-39),
sf, PopGenReport</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-14 23:22:35 UTC; s425824</td>
</tr>
<tr>
<td>Author:</td>
<td>Bernd Gruber [aut, cre],
  Arthur Georges [aut],
  Jose L. Mijangos [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/dartr?pli=1">https://groups.google.com/g/dartr?pli=1</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bernd Gruber &lt;bernd.gruber@canberra.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-15 00:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gl.costdistances'>Calculates cost distances for a given landscape (resistance matrix)</h2><span id='topic+gl.costdistances'></span>

<h3>Description</h3>

<p>Calculates a cost distance matrix, to be used with run.popgensim.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.costdistances(landscape, locs, method, NN, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.costdistances_+3A_landscape">landscape</code></td>
<td>
<p>A raster object coding the resistance of the landscape
[required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_locs">locs</code></td>
<td>
<p>Coordinates of the subpopulations. If a genlight object is
provided coordinates are taken from @other$latlon and centers for population
(pop(gl)) are calculated. In case you want to calculate costdistances between
individuals redefine pop(gl) via: <code>pop(gl)&lt;- indNames(gl)</code> [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_method">method</code></td>
<td>
<p>Defines the type of cost distance, types are 'leastcost',
'rSPDistance' or 'commute' (Circuitscape type) [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_nn">NN</code></td>
<td>
<p>Number of next neighbours recommendation is 8 [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A costdistance matrix between all pairs of locs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(possums.gl)
library(raster)  #needed for that example
landscape.sim &lt;- readRDS(system.file('extdata','landscape.sim.rdata', 
package='dartR.data'))
#calculate mean centers of individuals per population
xy &lt;- apply(possums.gl@other$xy, 2, function(x) tapply(x, pop(possums.gl),
 mean))
cd &lt;- gl.costdistances(landscape.sim, xy, method='leastcost', NN=8)
round(cd,3)
</code></pre>

<hr>
<h2 id='gl.genleastcost'>Performs least-cost path analysis based on a friction matrix</h2><span id='topic+gl.genleastcost'></span>

<h3>Description</h3>

<p>This function calculates the pairwise distances (Euclidean, cost path
distances and genetic distances) of populations using a friction matrix and
a spatial genind object. The genind object needs to have coordinates in the
same projected coordinate system as the friction matrix. The friction
matrix can be either a single raster of a stack of several layers. If a
stack is provided the specified cost distance is calculated for each layer
in the stack. The output of this function can be used with the functions
wassermann from package PopGenReport and
lgrMMRR from package PopGenReport to test for the significance of a
layer on the genetic structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.genleastcost(
  x,
  fric.raster,
  gen.distance,
  NN = NULL,
  pathtype = "leastcost",
  plotpath = TRUE,
  theta = 1,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.genleastcost_+3A_x">x</code></td>
<td>
<p>A spatial gelight object. [required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_fric.raster">fric.raster</code></td>
<td>
<p>A friction matrix [required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_gen.distance">gen.distance</code></td>
<td>
<p>Specification which genetic distance method should be
used to calculate pairwise genetic distances between populations ( 'D',
'Gst.Nei', 'Gst.Hedrick') or individuals ('kosman', 'propShared')
[required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_nn">NN</code></td>
<td>
<p>Number of neighbours used when calculating the cost distance
(possible values 4, 8 or 16). As the default is NULL a value has to be
provided if pathtype='leastcost'. NN=8 is most commonly used. Be aware that
linear structures may cause artefacts in the least-cost paths, therefore
inspect the actual least-cost paths in the provided output [default NULL].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_pathtype">pathtype</code></td>
<td>
<p>Type of cost distance to be calculated (based on function in
the <code>gdistance</code> package. Available distances are 'leastcost', 'commute'
or 'rSPDistance'. See functions in the gdistance package for futher
explanations. If the path type is set to 'leastcost' then paths and also
pathlength are returned [default 'leastcost'].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_plotpath">plotpath</code></td>
<td>
<p>switch if least cost paths should be plotted (works only if
pathtype='leastcost'. Be aware this slows down the computation, but it is
recommended to do this to check least cost paths visually.</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_theta">theta</code></td>
<td>
<p>value needed for rSPDistance function. See
<code><a href="gdistance.html#topic+rSPDistance">rSPDistance</a></code> in package <code>gdistance</code> [default 1].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list that consists of four pairwise distance matrices
(Euclidean, Cost, length of path and genetic) and the actual paths as spatial
line objects.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Cushman, S., Wasserman, T., Landguth, E. and Shirk, A. (2013).
Re-Evaluating Causal Modeling with Mantel Tests in Landscape Genetics.
Diversity, 5(1), 51-72.
</p>
</li>
<li><p> Landguth, E. L., Cushman, S. A., Schwartz, M. K., McKelvey, K. S.,
Murphy, M. and Luikart, G. (2010). Quantifying the lag time to detect
barriers in landscape genetics. Molecular ecology, 4179-4191.
</p>
</li>
<li><p> Wasserman, T. N., Cushman, S. A., Schwartz, M. K. and Wallin, D. O.
(2010). Spatial scaling and multi-model inference in landscape genetics:
Martes americana in northern Idaho. Landscape Ecology, 25(10), 1601-1612.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>#this example takes about 20 seconds to run...

data(possums.gl)
library(raster)  #needed for that example
landscape.sim &lt;- readRDS(system.file('extdata','landscape.sim.rdata', 
package='dartR.data'))
#use only 3 population (first 90 individuals) due to speed
glc &lt;- gl.genleastcost(x=possums.gl,fric.raster=landscape.sim ,
gen.distance = 'D', NN=8, pathtype = 'leastcost',plotpath = TRUE)
#### run tests as implemented in PopGenreport (maybe need to install)
if (require("PopGenReport", quietly=TRUE)) {
PopGenReport::wassermann(eucl.mat = glc$eucl.mat, cost.mat = glc$cost.mats, 
 gen.mat = glc$gen.mat)
lgrMMRR(gen.mat = glc$gen.mat, cost.mats = glc$cost.mats,  
eucl.mat = glc$eucl.mat)
}

</code></pre>

<hr>
<h2 id='gl.grm2'>Calculates an identity by descent matrix</h2><span id='topic+gl.grm2'></span>

<h3>Description</h3>

<p>This function calculates the mean probability of identity by state (IBS)
across loci that would result from all the possible crosses of the
individuals analyzed. IBD is calculated by an additive relationship matrix
approach developed by Endelman and Jannink (2012) as implemented in the
function <a href="rrBLUP.html#topic+A.mat">A.mat</a> (package rrBLUP).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.grm2(
  x,
  plotheatmap = TRUE,
  palette_discrete = NULL,
  palette_convergent = NULL,
  legendx = 0,
  legendy = 0.5,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.grm2_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_plotheatmap">plotheatmap</code></td>
<td>
<p>A switch if a heatmap should be shown [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_palette_discrete">palette_discrete</code></td>
<td>
<p>the color of populations [gl.select.colors].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_palette_convergent">palette_convergent</code></td>
<td>
<p>A convergent palette for the IBD values
[default convergent_palette].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_legendx">legendx</code></td>
<td>
<p>x coordinates for the legend[default 0].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_legendy">legendy</code></td>
<td>
<p>y coordinates for the legend[default 1].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.grm2_+3A_...">...</code></td>
<td>
<p>Parameters passed to function A.mat from package rrBLUP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two or more alleles are identical by descent (IBD) if they are identical
copies of the same ancestral allele in a base population. The additive
relationship matrix is a theoretical framework for estimating a relationship
matrix that is consistent with an approach to estimate the probability that
the alleles at a random locus are identical in state (IBS).
</p>
<p>This function also plots a heatmap, and a dendrogram, of IBD values where
each diagonal element has a mean that equals 1+f, where f is the inbreeding
coefficient (i.e. the probability that the two alleles at a randomly chosen
locus are IBD from the base population). As this probability lies between 0
and 1, the diagonal elements range from 1 to 2. Because the inbreeding
coefficients are expressed relative to the current population, the mean of
the off-diagonal elements is -(1+f)/n, where n is the number of loci.
Individual names are shown in the margins of the heatmap and colors
represent different populations.
</p>


<h3>Value</h3>

<p>An identity by descent matrix
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Endelman, J. B. (2011). Ridge regression and other kernels for genomic
selection with r package rrblup. The Plant Genome 4, 250.
</p>
</li>
<li><p> Endelman, J. B. , Jannink, J.-L. (2012). Shrinkage estimation of the
realized relationship matrix. G3: Genes, Genomics, Genetics 2, 1405.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>require(dartR.base)  
gl.grm2(platypus.gl[1:10,1:100])

</code></pre>

<hr>
<h2 id='gl.ibd'>Performs isolation by distance analysis</h2><span id='topic+gl.ibd'></span>

<h3>Description</h3>

<p>This function performs an isolation by distance analysis based on a Mantel
test and also produces an isolation by distance plot. If a genlight object
with coordinates is provided, then an Euclidean and genetic distance matrices
are calculated.'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ibd(
  x = NULL,
  distance = "Fst",
  coordinates = "latlon",
  Dgen = NULL,
  Dgeo = NULL,
  Dgeo_trans = "Dgeo",
  Dgen_trans = "Dgen",
  permutations = 999,
  plot.out = TRUE,
  paircols = NULL,
  plot.theme = theme_dartR(),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ibd_+3A_x">x</code></td>
<td>
<p>Genlight object. If provided a standard analysis on Fst/1-Fst and
log(distance) is performed [required].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_distance">distance</code></td>
<td>
<p>Type of distance that is calculated and used for the
analysis. Can be either population based 'Fst' [<a href="StAMPP.html#topic+stamppFst">stamppFst</a>],
'D' [<a href="StAMPP.html#topic+stamppNeisD">stamppNeisD</a>] or individual based 'propShared',
[gl.propShared], 'euclidean' [gl.dist.ind, method='Euclidean'], 'kosman' [gl.kosman]
[default &quot;Fst&quot;].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_coordinates">coordinates</code></td>
<td>
<p>Can be either 'latlon', 'xy' or a two column data.frame
with column names 'lat','lon', 'x', 'y'). Coordinates are provided via
<code>gl@other$latlon</code> ['latlon'] or via <code>gl@other$xy</code> ['xy']. If latlon
data will be projected to meters using Mercator system [google maps] or if
xy then distance is directly calculated on the coordinates.</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgen">Dgen</code></td>
<td>
<p>Genetic distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgeo">Dgeo</code></td>
<td>
<p>Euclidean distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgeo_trans">Dgeo_trans</code></td>
<td>
<p>Transformation to be used on the Euclidean distances. See
Dgen_trans [default &quot;Dgeo&quot;].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgen_trans">Dgen_trans</code></td>
<td>
<p>You can provide a formula to transform the genetic
distance. The transformation can be applied as a formula using Dgen as the
variable to be transformed. For example: <code>Dgen_trans = 'Dgen/(1-Dgen)'.
  Any valid R expression can be used here
   [default 'Dgen', which is the identity function.]</code></p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_permutations">permutations</code></td>
<td>
<p>Number of permutations in the Mantel test [default 999].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot.out">plot.out</code></td>
<td>
<p>Should an isolation by distance plot be returned
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_paircols">paircols</code></td>
<td>
<p>Should pairwise dots colored by 'pop'ulation/'ind'ividual
pairs [default 'pop']. You can color pairwise individuals by pairwise
population colors.</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently pairwise Fst and D between populations and
1-propShared and Euclidean distance between individuals are
implemented. Coordinates are expected as lat long and converted to Google
Earth Mercator projection. If coordinates are already projected, provide them
at the x@other$xy slot.
</p>
<p>You can provide also your own genetic and Euclidean distance matrices. The
function is based on the code provided by the adegenet tutorial
(<a href="http://adegenet.r-forge.r-project.org/files/tutorial-basics.pdf">http://adegenet.r-forge.r-project.org/files/tutorial-basics.pdf</a>),
using the functions  <a href="vegan.html#topic+mantel">mantel</a> (package vegan),
<a href="StAMPP.html#topic+stamppFst">stamppFst</a>, <a href="StAMPP.html#topic+stamppNeisD">stamppNeisD</a> (package StAMPP) and
gl.propShared or gl.dist.ind. For transformation you need to have the dismo
package installed. As a new feature you can plot pairwise relationship using
double colored points (paircols=TRUE). Pairwise relationship can be
visualised via populations or individuals, depending which distance is
calculated. Please note: Often a problem arises, if an individual based 
distance is calculated (e.g. propShared) and some individuals have identical
coordinates as this results in distances of zero between those pairs of 
individuals.
</p>
<p>If the standard transformation [log(Dgeo)] is used, this results in an 
infinite value, because of trying to calculate'log(0)'. To avoid this, the 
easiest fix is to change the transformation from log(Dgeo) to log(Dgeo+1) or 
you could add some &quot;noise&quot; to the coordinates of the individuals (e.g. +- 1m,
but be aware if you use lat lon then you rather want to add +0.00001 degrees
or so).
</p>


<h3>Value</h3>

<p>Returns a list of the following components: Dgen (the genetic
distance matrix), Dgeo (the Euclidean distance matrix), Mantel (the
statistics of the Mantel test).
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Rousset, F. (1997). Genetic differentiation and estimation of gene flow from
F-statistics under isolation by distance. Genetics, 145(4), 1219-1228.
</p>


<h3>See Also</h3>

<p><a href="vegan.html#topic+mantel">mantel</a>, <a href="StAMPP.html#topic+stamppFst">stamppFst</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
#because of speed only the first 100 loci
ibd &lt;- gl.ibd(bandicoot.gl[,1:100], Dgeo_trans='log(Dgeo)' ,
Dgen_trans='Dgen/(1-Dgen)')
#because of speed only the first 10 individuals)
ibd &lt;- gl.ibd(bandicoot.gl[1:10,], distance='euclidean', paircols='pop', 
Dgeo_trans='Dgeo')

#only first 100 loci
ibd &lt;- gl.ibd(bandicoot.gl[,1:100], paircols='pop')
</code></pre>

<hr>
<h2 id='gl.kosman'>Calculates a Kosman distance matrix for each pair of individuals</h2><span id='topic+gl.kosman'></span>

<h3>Description</h3>

<p>This script calculates an individual based distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.kosman(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.kosman_+3A_x">x</code></td>
<td>
<p>genlight/dartR object with a unique ploidy</p>
</td></tr>
<tr><td><code id="gl.kosman_+3A_verbose">verbose</code></td>
<td>
<p>verbosity of the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a matrix of [dimensions nInd(x) x nInd(x)] of kosman distances between individuals,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#use only five individuals and seven loci
gg &lt;- gl.kosman(possums.gl[1:5,14:21])
gg$kosman
gg$nloci
</code></pre>

<hr>
<h2 id='gl.spatial.autoCorr'>Spatial autocorrelation following Smouse and Peakall 1999</h2><span id='topic+gl.spatial.autoCorr'></span>

<h3>Description</h3>

<p>Global spatial autocorrelation is a multivariate approach
combining all loci into a single analysis. The autocorrelation coefficient
&quot;r&quot; is calculated for each pair of individuals in each specified distance
class. For more information see Smouse and Peakall 1999, Peakall et al. 2003
and Smouse et al. 2008.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.spatial.autoCorr(
  x = NULL,
  Dgeo = NULL,
  Dgen = NULL,
  coordinates = "latlon",
  Dgen_method = "Euclidean",
  Dgeo_trans = "Dgeo",
  Dgen_trans = "Dgen",
  bins = 5,
  reps = 100,
  plot.pops.together = FALSE,
  permutation = TRUE,
  bootstrap = TRUE,
  plot.theme = theme_dartR(),
  plot.colors.pop = NULL,
  CI.color = "red",
  plot.out = TRUE,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.spatial.autoCorr_+3A_x">x</code></td>
<td>
<p>Genlight object [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgeo">Dgeo</code></td>
<td>
<p>Geographic distance matrix if no genlight object is provided.
This is typically an Euclidean distance but it can be any meaningful 
(geographical) distance metrics [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen">Dgen</code></td>
<td>
<p>Genetic distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_coordinates">coordinates</code></td>
<td>
<p>Can be either 'latlon', 'xy' or a two column data.frame
with column names 'lat','lon', 'x', 'y')  Coordinates are provided via
<code>gl@other$latlon</code> ['latlon'] or via <code>gl@other$xy</code> ['xy']. If latlon
data will be projected to meters using Mercator system [google maps] or if
xy then distance is directly calculated on the coordinates [default &quot;latlon&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen_method">Dgen_method</code></td>
<td>
<p>Method to calculate genetic distances. See details
[default &quot;Euclidean&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgeo_trans">Dgeo_trans</code></td>
<td>
<p>Transformation to be used on the geographic distances. See
Dgen_trans [default &quot;Dgeo&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen_trans">Dgen_trans</code></td>
<td>
<p>You can provide a formula to transform the genetic
distance. The transformation can be applied as a formula using Dgen as the
variable to be transformed. For example: <code>Dgen_trans = 'Dgen/(1-Dgen)'.
  Any valid R expression can be used here 
  [default 'Dgen', which is the identity function.]</code></p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_bins">bins</code></td>
<td>
<p>The number of bins for the distance classes
(i.e. <code>length(bins) == 1)</code> or a vectors with the break points. See 
details [default 5].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_reps">reps</code></td>
<td>
<p>The number to be used for permutation and bootstrap analyses
[default 100].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.pops.together">plot.pops.together</code></td>
<td>
<p>Plot all the populations in one plot. Confidence 
intervals from permutations are not shown [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_permutation">permutation</code></td>
<td>
<p>Whether permutation calculations for the null hypothesis 
of no spatial structure should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_bootstrap">bootstrap</code></td>
<td>
<p>Whether bootstrap calculations to compute the 95% 
confidence intervals around r should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See details [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.colors.pop">plot.colors.pop</code></td>
<td>
<p>A color palette for populations or a list with
as many colors as there are populations in the dataset [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_ci.color">CI.color</code></td>
<td>
<p>Color for the shade of the 95% confidence intervals around 
the r estimates [default &quot;red&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function executes a modified version
of <code>spautocorr</code> from the package <code>PopGenReport</code>. Differently 
from <code>PopGenReport</code>, this function also computes the 95% confidence 
intervals around the r via bootstraps, the 95
null hypothesis of no spatial structure and the one-tail test via permutation, 
and the correction factor described by Peakall et al 2003.
</p>
<p>The input can be i) a genlight object (which has to have the latlon slot 
populated), ii) a pair of <code>Dgeo</code> and <code>Dgen</code>, which have to be
either
<code>matrix</code> or <code>dist</code> objects, or iii) a <code>list</code> of the 
<code>matrix</code> or <code>dist</code> objects if the 
analysis needs to be carried out for multiple populations (in this case, 
all the elements of the <code>list</code> have to be of the same class (i.e. 
<code>matrix</code> or <code>dist</code>) and the population order in the two lists has 
to be the same. 
</p>
<p>If the input is a genlight object, the function calculates the linear 
distance
for <code>Dgeo</code> and the relevant <code>Dgen</code> matrix (see <code>Dgen_method</code>) 
for each population. 
When the method selected is a genetic similarity matrix (e.g. &quot;simple&quot; 
distance), the matrix is internally transformed with <code>1 - Dgen</code> so that 
positive values of autocorrelation coefficients indicates more related 
individuals similarly as implemented in GenAlEx. If the user provide the 
distance matrices, care must be taken in interpreting the results because
similarity matrix will generate negative values for closely related 
individuals.
</p>
<p>If <code>max(Dgeo)&gt;1000</code> (e.g. the geographic distances are in thousands of 
metres), values are divided by 1000 (in the example before these would then 
become km) to facilitate readability of the plots.
</p>
<p>If <code>bins</code> is of length = 1 it is interpreted as the number of (even)
bins to use. In this case the starting point is always the minimum value in 
the distance matrix, and the last is the maximum. If it is a numeric vector 
of length&gt;1, it is interpreted as the breaking points. In this case, the 
first has to be the lowest value, and the last has to be the highest. There 
are no internal checks for this and it is user responsibility to ensure that
distance classes are properly set up. If that is not the case, data that fall
outside the range provided will be dropped. The number of bins will be 
<code>length(bins) - 1</code>.
</p>
<p>The permutation constructs the 95% confidence intervals around the null
hypothesis of no spatial structure (this is a two-tail test). The same data
are also used to calculate the probability of the one-tail test (See 
references below for details).
</p>
<p>Bootstrap calculations are skipped and <code>NA</code> is returned when the number 
of possible combinations given the sample size of any given distance class is
&lt; <code>reps</code>.
</p>
<p>Methods available to calculate genetic distances for SNP data:
</p>

<ul>
<li><p> &quot;propShared&quot; using the function <code><a href="dartR.base.html#topic+gl.propShared">gl.propShared</a></code>.
</p>
</li>
<li><p> &quot;grm&quot; using the function <code>gl.grm2</code>.
</p>
</li>
<li><p> &quot;Euclidean&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Simple&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Absolute&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Manhattan&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li></ul>

<p>Methods available to calculate genetic distances for SilicoDArT data:
</p>

<ul>
<li><p> &quot;Euclidean&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Simple&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Jaccard&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Bray-Curtis&quot; using the function <code><a href="dartR.base.html#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li></ul>

<p>Plots and table are saved plot.file in plot.dir if specified.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a data frame with the following columns:
</p>

<ol>
<li><p> Bin  The distance classes
</p>
</li>
<li><p> N The number of pairwise comparisons within each distance class
</p>
</li>
<li><p> r.uc The uncorrected autocorrelation coefficient
</p>
</li>
<li><p> Correction the correction
</p>
</li>
<li><p> r The corrected autocorrelation coefficient
</p>
</li>
<li><p> L.r The corrected autocorrelation coefficient lower limit
(if <code>bootstap = TRUE</code>)
</p>
</li>
<li><p> U.r The corrected autocorrelation coefficient upper limit
(if <code>bootstap = TRUE</code>)
</p>
</li>
<li><p> L.r.null.uc The uncorrected lower limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> U.r.null.uc  The uncorrected upper limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> L.r.null The corrected lower limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> U.r.null The corrected upper limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> p.one.tail The p value of the one tail statistical test
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Carlo Pacioni, Bernd Gruber &amp; Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Smouse PE, Peakall R. 1999. Spatial autocorrelation analysis of
individual multiallele and multilocus genetic structure. Heredity 82:
561-573.
</p>
</li>
<li><p> Double, MC, et al. 2005. Dispersal, philopatry and infidelity: 
dissecting local genetic structure in superb fairy-wrens (Malurus cyaneus). 
Evolution 59, 625-635.
</p>
</li>
<li><p> Peakall, R, et al. 2003. Spatial autocorrelation analysis offers new
insights into gene flow in the Australian bush rat, Rattus fuscipes.
Evolution 57, 1182-1195.
</p>
</li>
<li><p> Smouse, PE, et al. 2008. A heterogeneity test for fine-scale genetic
structure. Molecular Ecology 17, 3389-3400.
</p>
</li>
<li><p> Gonzales, E, et al. 2010. The impact of landscape disturbance on 
spatial genetic structure in the Guanacaste tree, Enterolobium
cyclocarpum (Fabaceae). Journal of Heredity 101, 133-143.
</p>
</li>
<li><p> Beck, N, et al. 2008. Social constraint and an absence of sex-biased
dispersal drive fine-scale genetic structure in white-winged choughs.
Molecular Ecology 17, 4346-4358.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
res &lt;- gl.spatial.autoCorr(platypus.gl, bins=seq(0,10000,2000))
# using one population, showing sample size
test &lt;- gl.keep.pop(platypus.gl,pop.list = "TENTERFIELD")
res &lt;- gl.spatial.autoCorr(test, bins=seq(0,10000,2000),CI.color = "green")

test &lt;- gl.keep.pop(platypus.gl,pop.list = "TENTERFIELD")
res &lt;- gl.spatial.autoCorr(test, bins=seq(0,10000,2000),CI.color = "green")
</code></pre>

<hr>
<h2 id='gl2shp'>Converts a genlight object to ESRI shapefiles or kml files</h2><span id='topic+gl2shp'></span>

<h3>Description</h3>

<p>This function exports coordinates in a genlight object to a point shape file
(including also individual meta data if available).
Coordinates are provided under x@other$latlon and assumed to be in WGS84
coordinates, if not proj4 string is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2shp(
  x,
  type = "shp",
  proj4 = "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs",
  outfile = "gl",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2shp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data and location
data, lat longs [required].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_type">type</code></td>
<td>
<p>Type of output 'kml' or 'shp' [default 'shp'].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_proj4">proj4</code></td>
<td>
<p>Proj4string of data set (see spatialreference.org for
projections) [default WGS84].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_outfile">outfile</code></td>
<td>
<p>Name (path) of the output shape file [default 'gl']. shp
extension is added automatically.</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a SpatVector file
</p>


<h3>Author(s)</h3>

<p>Bernd Guber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl2shp(testset.gl, outpath=tempdir())
</code></pre>

<hr>
<h2 id='utils.spautocor'>Spatial autocorrelation coefficient calculations</h2><span id='topic+utils.spautocor'></span>

<h3>Description</h3>

<p>Carries out calculation for spatial autocorrelation coefficient
starting from a genetic and geogaphic distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.spautocor(
  GD,
  GGD,
  permutation = FALSE,
  bootstrap = FALSE,
  bins = 10,
  reps
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.spautocor_+3A_gd">GD</code></td>
<td>
<p>Genetic distance matrix.</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_ggd">GGD</code></td>
<td>
<p>Geographic distance matrix.</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_permutation">permutation</code></td>
<td>
<p>Whether permutation calculations for the null hypothesis 
of no spatial structure should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_bootstrap">bootstrap</code></td>
<td>
<p>Whether bootstrap calculations to compute the 95% 
confidence intervals around r should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_bins">bins</code></td>
<td>
<p>The number of bins for the distance classes
(i.e. <code>length(bins) == 1)</code> or a vectors with the break points. See 
details [default 5].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_reps">reps</code></td>
<td>
<p>The number to be used for permutation and bootstrap analyses
[default 100].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code of this function is based one <code>spautocorr</code> from the package
<code>PopGenReport</code>, which has been modified to fix a few bugs (as of
<code>PopGenReport v 3.0.4</code> and allow calculations of bootstraps estimates.
</p>
<p>See details from <code>gl.spatial.autoCorr</code> for a detailed explanation.
</p>


<h3>Value</h3>

<p>Returns a data frame with the following columns:
</p>

<ol>
<li><p> Bin  The distance classes
</p>
</li>
<li><p> N The number of pairwise comparisons within each distance class
</p>
</li>
<li><p> r.uc The uncorrected autocorrelation coefficient
</p>
</li></ol>

<p>if both <code>bootstap</code> and <code>permutation</code> are <code>FALSE</code> otherwise only
<code>r</code> estimates are returned
</p>


<h3>Author(s)</h3>

<p>Carlo Pacioni &amp; Bernd Gruber
</p>


<h3>References</h3>


<ul>
<li><p> Smouse PE, Peakall R. 1999. Spatial autocorrelation analysis of
individual multiallele and multilocus genetic structure. Heredity 82:
561-573.
</p>
</li>
<li><p> Double, MC, et al. 2005. Dispersal, philopatry and infidelity: dissecting
local genetic structure in superb fairy-wrens (Malurus cyaneus). Evolution
59, 625-635.
</p>
</li>
<li><p> Peakall, R, et al. 2003. Spatial autocorrelation analysis offers new
insights into gene flow in the Australian bush rat, Rattus fuscipes.
Evolution 57, 1182-1195.
</p>
</li>
<li><p> Smouse, PE, et al. 2008. A heterogeneity test for fine-scale genetic
structure. Molecular Ecology 17, 3389-3400.
</p>
</li>
<li><p> Gonzales, E, et al. 2010. The impact of landscape disturbance on spatial
genetic structure in the Guanacaste tree, Enterolobium
cyclocarpum(Fabaceae). Journal of Heredity 101, 133-143.
</p>
</li>
<li><p> Beck, N, et al. 2008. Social constraint and an absence of sex-biased
dispersal drive fine-scale genetic structure in white-winged choughs.
Molecular Ecology 17, 4346-4358.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.spatial.autoCorr">gl.spatial.autoCorr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See gl.spatial.autoCorr
</code></pre>

<hr>
<h2 id='zzz'>Setting up the package dartR.spatial</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting up dartR.spatial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
