<!DOCTYPE html><html><head><title>Help for package dartR.sim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR.sim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gl.diagnostics.sim'><p>Comparing simulations against theoretical expectations</p></a></li>
<li><a href='#gl.sim.create_dispersal'><p>Creates a dispersal file as input for the function gl.sim.WF.run</p></a></li>
<li><a href='#gl.sim.emigration'><p>Simulates emigration between populations</p></a></li>
<li><a href='#gl.sim.ind'><p>Simulates individuals based on allele frequencies</p></a></li>
<li><a href='#gl.sim.mutate'><p>Simulates mutations within a genlight object</p></a></li>
<li><a href='#gl.sim.offspring'><p>Simulates offspring based on alleles provided by parents</p></a></li>
<li><a href='#gl.sim.WF.run'><p>Runs Wright-Fisher simulations</p></a></li>
<li><a href='#gl.sim.WF.table'><p>Creates the reference table for running gl.sim.WF.run</p></a></li>
<li><a href='#interactive_reference'><p>Shiny app for the input of the reference table for the simulations</p></a></li>
<li><a href='#interactive_sim_run'><p>Shiny app for the input of the simulations variables</p></a></li>
<li><a href='#zzz'><p>Setting up the package</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computer Simulations of 'SNP' Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.70</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows to simulate SNP data using genlight objects. For example, it is straight forward to simulate a simple drift scenario with exchange of individuals between two populations or create a new genlight object based on allele frequencies of an existing genlight object.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), adegenet (&ge; 2.0.0), dartR.base, dartR.data,
ggplot2</td>
</tr>
<tr>
<td>Imports:</td>
<td>shiny, fields, utils, methods, stringi, stringr, data.table,
Rcpp, shinyBS, shinyjs, shinythemes, shinyWidgets, hierfstat,
reshape2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-20 06:56:23 UTC; mijangos</td>
</tr>
<tr>
<td>Author:</td>
<td>Jose L. Mijangos [aut, cre],
  Bernd Gruber [aut],
  Arthur Georges [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a>,
<a href="https://github.com/green-striped-gecko/dartR.sim">https://github.com/green-striped-gecko/dartR.sim</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/green-striped-gecko/dartR.sim/issues">https://github.com/green-striped-gecko/dartR.sim/issues</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jose L. Mijangos &lt;luis.mijangos@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-20 19:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gl.diagnostics.sim'>Comparing simulations against theoretical expectations</h2><span id='topic+gl.diagnostics.sim'></span>

<h3>Description</h3>

<p>Comparing simulations against theoretical expectations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.diagnostics.sim(
  x,
  Ne,
  iteration = 1,
  pop_he = 1,
  pops_fst = c(1, 2),
  plot_theme = theme_dartR(),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.diagnostics.sim_+3A_x">x</code></td>
<td>
<p>Output from function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_ne">Ne</code></td>
<td>
<p>Effective population size to use as input to compare theoretical 
expectations [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_iteration">iteration</code></td>
<td>
<p>Iteration number to analyse [default 1].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_pop_he">pop_he</code></td>
<td>
<p>Population name in which the rate of loss of heterozygosity is
going to be compared against theoretical expectations [default 1].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_pops_fst">pops_fst</code></td>
<td>
<p>Pair of populations in which FST is going to be compared 
against theoretical expectations [default c(1,2)].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two plots are presented comparing the simulations against theoretical 
expectations:
</p>

<ol>
<li><p> Expected heterozygosity under neutrality (Crow &amp; Kimura, 1970, p. 329) is 
calculated as:
</p>
<p>Het = He0(1-(1/2Ne))^t,
</p>
<p>where Ne is effective population size, He0  is heterozygosity at generation 0
and t is the number of generations.
</p>
</li>
<li><p> Expected FST under neutrality (Takahata, 1983) is calculated as:
</p>
<p>FST=1/(4Nem(n/(n-1))^2+1),
</p>
<p>where Ne is effective populations size of each individual subpopulation, m is
dispersal rate and n the number of subpopulations (always 2).
</p>
</li></ol>



<h3>Value</h3>

<p>Returns plots comparing simulations against theoretical expectations
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Crow JF, Kimura M. An introduction to population genetics theory. An 
introduction to population genetics theory. 1970.
</p>
</li>
<li><p> Takahata N. Gene identity and genetic differentiation of populations in 
the finite island model. Genetics. 1983;104(3):497-512.
</p>
</li></ul>



<h3>See Also</h3>

<p><code>gl.filter.callrate</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ref_table &lt;- gl.sim.WF.table(file_var=system.file('extdata', 
'ref_variables.csv', package = 'dartR.data'),interactive_vars = FALSE)

res_sim &lt;- gl.sim.WF.run(file_var = system.file('extdata',
 'sim_variables.csv', package ='dartR.data'),ref_table=ref_table,
 interactive_vars = FALSE,number_pops_phase2=2,population_size_phase2="50 50")
 
 res &lt;- gl.diagnostics.sim(x=res_sim,Ne=50)
 
</code></pre>

<hr>
<h2 id='gl.sim.create_dispersal'>Creates a dispersal file as input for the function gl.sim.WF.run</h2><span id='topic+gl.sim.create_dispersal'></span>

<h3>Description</h3>

<p>This function writes a csv file called &quot;dispersal_table.csv&quot; which contains
the dispersal variables for each pair of populations to be used as input for
the function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>.
</p>
<p>The values of the variables can be modified using the columns
&quot;transfer_each_gen&quot; and &quot;number_transfers&quot; of this file.
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed by typing in the R console:
browseVignettes(package=&quot;dartR”)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.create_dispersal(
  number_pops,
  dispersal_type = "all_connected",
  number_transfers = 1,
  transfer_each_gen = 1,
  outpath = tempdir(),
  outfile = "dispersal_table.csv",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.create_dispersal_+3A_number_pops">number_pops</code></td>
<td>
<p>Number of populations [required].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_dispersal_type">dispersal_type</code></td>
<td>
<p>One of: &quot;all_connected&quot;, &quot;circle&quot; or &quot;line&quot;
[default &quot;all_connected&quot;].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_number_transfers">number_transfers</code></td>
<td>
<p>Number of dispersing individuals. This value can be .
modified by hand after the file has been created [default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_transfer_each_gen">transfer_each_gen</code></td>
<td>
<p>Interval of number of generations in which dispersal
occur. This value can be modified by hand after the file has been created
[default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'dispersal_table.csv'].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A csv file containing the dispersal variables for each pair of
populations to be used as input for the function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a>()</code>,
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.sim.create_dispersal(number_pops=10)
</code></pre>

<hr>
<h2 id='gl.sim.emigration'>Simulates emigration between populations</h2><span id='topic+gl.sim.emigration'></span>

<h3>Description</h3>

<p>A function that allows to exchange individuals of populations within a
genlight object (=simulate emigration between populations).
</p>
<p>There are two ways to specify emigration. If an emi.table is provided (a
square matrix of dimension of the populations that specifies the emigration
from column x to row y), then emigration is deterministic in terms of numbers
of individuals as specified in the table. If perc.mig and emi.m are provided,
then emigration is probabilistic. The number of emigrants is determined by
the population size times the perc.mig and then the population where to
migrate to is taken from the relative probability in the columns of the emi.m
table.
</p>
<p>Be aware if the diagonal is non zero then migration can occur into the same
patch. So most often you want to set the diagonal of the emi.m matrix to
zero. Which individuals is moved is random, but the order is in the order of
populations. It is possible that an individual moves twice within an
emigration call(as there is no check, so an individual moved from population
1 to 2 can move again from population 2 to 3).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.emigration(x, perc.mig = NULL, emi.m = NULL, emi.table = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.emigration_+3A_x">x</code></td>
<td>
<p>A genlight or list of genlight objects [required].</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_perc.mig">perc.mig</code></td>
<td>
<p>Percentage of individuals that migrate
(emigrates = nInd times perc.mig) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_emi.m">emi.m</code></td>
<td>
<p>Probabilistic emigration matrix (emigrate from=column to=row)
[default NULL]</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_emi.table">emi.table</code></td>
<td>
<p>If presented emi.m matrix is ignored. Deterministic
emigration as specified in the matrix (a square matrix of dimension of the
number of populations). e.g. an entry in the 'emi.table[2,1]&lt;- 5' means that
five individuals emigrate from population 1 to population 2 (from=columns and
to=row) [default NULL].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list or a single [depends on the input] genlight object, where
emigration between population has happened
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- possums.gl
#one individual moves from every population to
#every other population
emi.tab &lt;- matrix(1, nrow=nPop(x), ncol=nPop(x))
diag(emi.tab)&lt;- 0
np &lt;- gl.sim.emigration(x, emi.table=emi.tab)
np
</code></pre>

<hr>
<h2 id='gl.sim.ind'>Simulates individuals based on allele frequencies</h2><span id='topic+gl.sim.ind'></span>

<h3>Description</h3>

<p>This function simulates individuals based on the allele frequencies of a
genlight object. The output is a genlight object with the same number of loci
as the input genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.ind(x, n = 50, popname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.sim.ind_+3A_n">n</code></td>
<td>
<p>Number of individuals that should be simulated [default 50].</p>
</td></tr>
<tr><td><code id="gl.sim.ind_+3A_popname">popname</code></td>
<td>
<p>A population name for the simulated individuals [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be used to simulate populations for sampling designs or for
power analysis. Check the example below where the effect of drift is
explored, by simply simulating several generation a genlight object and 
putting in the allele frequencies of the previous generation. The beauty of
the function is, that it is lightning fast. Be aware this is a simulation 
and to avoid lengthy error checking the function crashes if there are loci 
that have just NAs. If such a case can occur during your simulation, those
loci need to be removed, before the function is called.
</p>


<h3>Value</h3>

<p>A genlight object with n individuals.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bernd.gruber@canberra.edu.au)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>glsim &lt;- gl.sim.ind(testset.gl, n=10, popname='sims')
glsim
###Simulate drift over 10 generation
# assuming a bottleneck of only 10 individuals
# [ignoring effect of mating and mutation]
# Simulate 20 individuals with no structure and 50 SNP loci
founder &lt;- glSim(n.ind = 20, n.snp.nonstruc = 50, ploidy=2)
#number of fixed loci in the first generation

res &lt;- sum(colMeans(as.matrix(founder), na.rm=TRUE) %%2 ==0)
simgl &lt;- founder
#49 generations of only 10 individuals
for (i in 2:50)
{
   simgl &lt;- gl.sim.ind(simgl, n=10, popname='sims')
   res[i]&lt;- sum(colMeans(as.matrix(simgl), na.rm=TRUE) %%2 ==0)
}
plot(1:50, res, type='b', xlab='generation', ylab='# fixed loci')
</code></pre>

<hr>
<h2 id='gl.sim.mutate'>Simulates mutations within a genlight object</h2><span id='topic+gl.sim.mutate'></span>

<h3>Description</h3>

<p>This script is intended to be used within the simulation framework of dartR. 
It adds the ability to add a constant mutation rate across all loci. Only 
works currently for biallelic data sets (SNPs). Mutation rate is checking for 
all alleles position and mutations at loci with missing values are ignored 
and in principle 'double mutations' at the same loci can occur, but should be 
rare.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.mutate(x, mut.rate = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.mutate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.sim.mutate_+3A_mut.rate">mut.rate</code></td>
<td>
<p>Constant mutation rate over nInd*nLoc*2 possible locations
[default 1e-6]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object with the applied mutations
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b2 &lt;- gl.sim.mutate(bandicoot.gl,mut.rate=1e-4 )
#check the mutations that have occurred
table(as.matrix(bandicoot.gl), as.matrix(b2))
</code></pre>

<hr>
<h2 id='gl.sim.offspring'>Simulates offspring based on alleles provided by parents</h2><span id='topic+gl.sim.offspring'></span>

<h3>Description</h3>

<p>This takes a population (or a single individual) of fathers (provided as a
genlight object) and mother(s) and simulates offspring based on 'random'
mating. It can be used to simulate population dynamics and check the effect
of those dynamics and allele frequencies, number of alleles. Another
application is to simulate relatedness of siblings and compare it to actual
relatedness found in the population to determine kinship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.offspring(
  fathers,
  mothers,
  noffpermother,
  sexratio = 0.5,
  popname = "offspring",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.offspring_+3A_fathers">fathers</code></td>
<td>
<p>Genlight object of potential fathers [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_mothers">mothers</code></td>
<td>
<p>Genlight object of potential mothers simulated [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_noffpermother">noffpermother</code></td>
<td>
<p>Number of offspring per mother [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_sexratio">sexratio</code></td>
<td>
<p>The sex ratio of simulated offspring 
(females / females +males, 1 equals 100 percent females) [default 0.5.].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_popname">popname</code></td>
<td>
<p>population name of the returned genlight object 
[default offspring]</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with n individuals.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Simulate 10 potential fathers
gl.fathers &lt;- glSim(10, 20, ploidy=2)
#Simulate 10 potential mothers
gl.mothers &lt;- glSim(10, 20, ploidy=2)
res &lt;- gl.sim.offspring(gl.fathers, gl.mothers, 2, sexratio=0.5)
</code></pre>

<hr>
<h2 id='gl.sim.WF.run'>Runs Wright-Fisher simulations</h2><span id='topic+gl.sim.WF.run'></span>

<h3>Description</h3>

<p>This function simulates populations made up of diploid organisms that 
reproduce in non-overlapping generations. Each individual has a pair of 
homologous chromosomes that contains interspersed selected and neutral loci. 
For the initial generation, the genotype for each individual’s chromosomes is
randomly drawn from distributions at linkage equilibrium and in 
Hardy-Weinberg equilibrium. 
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed at http://georges.biomatix.org/dartR 
</p>
<p>Take into account that the simulations will take a little bit longer the
first time you use the function gl.sim.WF.run() because C++ functions must
be compiled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.WF.run(
  file_var,
  ref_table,
  x = NULL,
  file_dispersal = NULL,
  number_iterations = 1,
  every_gen = 10,
  sample_percent = 50,
  store_phase1 = FALSE,
  interactive_vars = TRUE,
  seed = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.WF.run_+3A_file_var">file_var</code></td>
<td>
<p>Path of the variables file 'sim_variables.csv' (see details) 
[required if interactive_vars = FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_ref_table">ref_table</code></td>
<td>
<p>Reference table created by the function 
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data to extract
values for some simulation variables (see details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_file_dispersal">file_dispersal</code></td>
<td>
<p>Path of the file with the dispersal table created with
the function <code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a></code> [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_number_iterations">number_iterations</code></td>
<td>
<p>Number of iterations of the simulations [default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_every_gen">every_gen</code></td>
<td>
<p>Generation interval at which simulations should be stored in
a genlight object [default 10].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_sample_percent">sample_percent</code></td>
<td>
<p>Percentage of individuals, from the total population, 
to sample and save in the genlight object every generation [default 50].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_store_phase1">store_phase1</code></td>
<td>
<p>Whether to store simulations of phase 1 in genlight
objects [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_interactive_vars">interactive_vars</code></td>
<td>
<p>Run a shiny app to input interactively the values of
simulations variables [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_seed">seed</code></td>
<td>
<p>Set the seed for the simulations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_...">...</code></td>
<td>
<p>Any variable and its value can be added separately within the 
function, will be changed over the input value supplied by the csv file. See 
tutorial.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Values for simulation variables can be submitted into the function 
interactively through a shiny app if interactive_vars = TRUE. Optionally, if 
interactive_vars = FALSE, values for variables can be submitted by using the
csv file 'sim_variables.csv' which can be found by typing in the R console:
system.file('extdata', 'sim_variables.csv', package ='dartR.data').
</p>
<p>The values of the variables can be modified using the third column (“value”) 
of this file. 
</p>
<p>The output of the simulations can be analysed seemingly with other dartR
functions.
</p>
<p>If a genlight object is used as input for some of the simulation variables, 
this function access the information stored in the slots x$position and 
x$chromosome.
</p>
<p>To show further information of the variables in interactive mode, it might be
necessary to call first: 'library(shinyBS)' for the information to be 
displayed.
</p>
<p>The main characteristics of the simulations are:
</p>
 
<ul>
<li><p> Simulations can be parameterised with real-life genetic 
characteristics such as the number, location, allele frequency and the 
distribution of fitness effects (selection coefficients and dominance) of 
loci under selection. 
</p>
</li>
<li><p> Simulations can recreate specific life histories and demographics, such
as source populations, dispersal rate, number of generations, founder 
individuals, effective population size and census population size.
</p>
</li>
<li><p> Each allele in each individual is an agent (i.e., each allele is 
explicitly simulated).
</p>
</li>
<li><p> Each locus can be customisable regarding its allele frequencies, 
selection coefficients, and dominance.
</p>
</li>
<li><p> The number of loci, individuals, and populations to be simulated is 
only limited by computing resources.
</p>
</li>
<li><p> Recombination is accurately modeled, and it is possible to use real 
recombination maps as input.
</p>
</li>
<li><p> The ratio between effective population size and census population size 
can be easily controlled.
</p>
</li>
<li><p> The output of the simulations are genlight objects for each generation 
or a subset of generations.
</p>
</li>
<li><p> Genlight objects can be used as input for some simulation variables.
</p>
</li></ul>



<h3>Value</h3>

<p>Returns genlight objects with simulated data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a>()</code>,
<code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
ref_table &lt;- gl.sim.WF.table(file_var=system.file("extdata", 
"ref_variables.csv", package = "dartR.data"),interactive_vars = FALSE)

res_sim &lt;- gl.sim.WF.run(file_var = system.file("extdata",
 "sim_variables.csv", package ="dartR.data"),ref_table=ref_table,
 interactive_vars = FALSE)
 
</code></pre>

<hr>
<h2 id='gl.sim.WF.table'>Creates the reference table for running gl.sim.WF.run</h2><span id='topic+gl.sim.WF.table'></span>

<h3>Description</h3>

<p>This function creates a reference table to be used as input for the function
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>. The created table has eight columns with the 
following information for each locus to be simulated:
</p>
 
<ul>
<li><p> q - initial frequency.
</p>
</li>
<li><p> h - dominance coefficient.
</p>
</li>
<li><p> s - selection coefficient.
</p>
</li>
<li><p> c - recombination rate.
</p>
</li>
<li><p> loc_bp - chromosome location in base pairs.
</p>
</li>
<li><p> loc_cM - chromosome location in centiMorgans.
</p>
</li>
<li><p> chr_name - chromosome name.
</p>
</li>
<li><p> type - SNP type.
</p>
</li></ul>
 
<p>The reference table can be further modified as required. 
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed at http://georges.biomatix.org/dartR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.WF.table(
  file_var,
  x = NULL,
  file_targets_sel = NULL,
  file_r_map = NULL,
  interactive_vars = TRUE,
  seed = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.WF.table_+3A_file_var">file_var</code></td>
<td>
<p>Path of the variables file 'ref_variables.csv' (see details) 
[required if interactive_vars = FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data to extract
values for some simulation variables (see details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_file_targets_sel">file_targets_sel</code></td>
<td>
<p>Path of the file with the targets for selection (see 
details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_file_r_map">file_r_map</code></td>
<td>
<p>Path of the file with the recombination map (see details)
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_interactive_vars">interactive_vars</code></td>
<td>
<p>Run a shiny app to input interactively the values of
simulation variables [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_seed">seed</code></td>
<td>
<p>Set the seed for the simulations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_...">...</code></td>
<td>
<p>Any variable and its value can be added separately within the 
function, will be changed over the input value supplied by the csv file. See 
tutorial.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Values for the variables to create the reference table can be submitted into 
the function interactively through a Shiny app if interactive_vars = TRUE. 
Optionally, if interactive_vars = FALSE, values for variables can be 
submitted by using the csv file 'ref_variables.csv' which can be found by 
typing in the R console:
system.file('extdata', 'ref_variables.csv', package ='dartR.data').
</p>
<p>The values of the variables can be modified using the third column (“value”) 
of this file. 
</p>
<p>If a genlight object is used as input for some of the simulation variables, 
this function access the information stored in the slots x$position and 
x$chromosome.
</p>
<p>Examples of the format required for the recombination map file and the 
targets for selection file can be found by typing in the R console:
</p>
 
<ul>
<li><p> system.file('extdata', 'fly_recom_map.csv', package ='dartR.data')
</p>
</li>
<li><p> system.file('extdata', 'fly_targets_of_selection.csv', package ='dartR.data')
</p>
</li></ul>

<p>To show further information of the variables in interactive mode, it might be
necessary to call first: 'library(shinyBS)' for the information to be 
displayed.
</p>


<h3>Value</h3>

<p>Returns a list with the reference table used as input for the function
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code> and a table with the values variables used to 
create the reference table.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a>()</code>,
<code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
ref_table &lt;- gl.sim.WF.table(file_var=system.file("extdata", 
"ref_variables.csv", package = "dartR.data"),interactive_vars = FALSE)

res_sim &lt;- gl.sim.WF.run(file_var = system.file("extdata",
 "sim_variables.csv", package ="dartR.data"),ref_table=ref_table,
 interactive_vars = FALSE)
 
 
</code></pre>

<hr>
<h2 id='interactive_reference'>Shiny app for the input of the reference table for the simulations</h2><span id='topic+interactive_reference'></span>

<h3>Description</h3>

<p>Shiny app for the input of the reference table for the simulations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interactive_reference()
</code></pre>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='interactive_sim_run'>Shiny app for the input of the simulations variables</h2><span id='topic+interactive_sim_run'></span>

<h3>Description</h3>

<p>Shiny app for the input of the simulations variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interactive_sim_run()
</code></pre>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='zzz'>Setting up the package</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting up dartR.sim
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
