<!DOCTYPE html><html lang="en"><head><title>Help for package dartR.sexlinked</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR.sexlinked}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gl.drop.sexlinked'><p>Removes loci that are sex linked</p></a></li>
<li><a href='#gl.infer.sex'><p>Uses sex-linked loci to infer sex of individuals</p></a></li>
<li><a href='#gl.keep.sexlinked'><p>Keeps loci that are sex linked</p></a></li>
<li><a href='#gl.report.sexlinked'><p>Filters loci that are sex linked</p></a></li>
<li><a href='#zzz'><p>Setting up the package dartR.sexlinked</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysing SNP Data to Identify Sex-Linked Markers</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Diana Robledo-Ruiz [aut, cre],
  Floriaan Devloo-Delva [aut],
  Bernd Gruber [aut],
  Arthur Georges [aut],
  Jose L. Mijangos [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb]</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifies, filters and exports sex linked markers using 'SNP' 
    (single nucleotide polymorphism) data. To install the other packages, we 
    recommend to install the 'dartRverse' package, that supports the installation of all packages in the 'dartRverse'.
    If you want understand the applied rational to identify sexlinked markers 
    and/or want to cite 'dartR.sexlinked', you find the information by typing 
    citation('dartR.sexlinked') in the console.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/dartr?pli=1">https://groups.google.com/g/dartr?pli=1</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>adegenet (&ge; 2.0.0), dartR.base, dartR.data, R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel (&ge; 1.0.17), ggplot2, methods, utils, foreach,
patchwork</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-22 06:09:12 UTC; bernd</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Diana Robledo-Ruiz &lt;diana.robledoruiz1@monash.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-24 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gl.drop.sexlinked'>Removes loci that are sex linked</h2><span id='topic+gl.drop.sexlinked'></span>

<h3>Description</h3>

<p>This function identifies sex-linked and autosomal loci present in a SNP 
dataset (genlight object) using individuals with known sex. It identifies
five types of loci: w-linked or y-linked, sex-biased, z-linked or
x-linked, gametologous and autosomal.
</p>
<p>This function produces as output a genlight object with autosomal loci only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.sexlinked(
  x,
  system = NULL,
  ncores = 1,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gl.drop.sexlinked_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data. This genlight
object needs to contain the sex of the individuals. See explanation in 
details [required].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_system">system</code></td>
<td>
<p>String that declares the sex-determination system of the 
species: 'zw' or 'xy' [required].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_ncores">ncores</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncores
&gt; 1 parallel operation is activated, see &quot;Details&quot; section [default 1].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_plot.display">plot.display</code></td>
<td>
<p>Creates four output plots. See explanation in details
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].[not yet implemented]</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_plot.colors">plot.colors</code></td>
<td>
<p>[not implemented yet]</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, 
exclude extension) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()].</p>
</td></tr>
<tr><td><code id="gl.drop.sexlinked_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genlight object must contain in <code>gl@other$ind.metrics</code> a column 
named &quot;id&quot;, and a column named &quot;sex&quot; in which individuals with known-sex are 
assigned 'M' for male, or 'F' for female. The function ignores individuals 
that are assigned anything else or nothing at all (unknown-sex).
</p>
<p>The creation of plots can be turned-off (<code>plot.display = FALSE</code>) in order
to save a little bit of running time for very large datasets (&gt;50,000 SNPs). 
However, we strongly encourage you to always inspect the output plots at 
least once to make sure everything is working properly.
</p>
<p><strong> Function's output </strong>
</p>
<p>This function returns as output a genlight object that contains only autosomal
loci (i.e. sex-linked loci have been dropped.)
</p>
<p>And four plots:</p>

<ul>
<li> <p>A BEFORE plot based on loci call rate by sex, with w/y-linked loci colored 
in yellow and sex-biased loci in blue
</p>
</li>
<li> <p>An AFTER plot based on loci call rate by sex, with sex-linked loci removed
</p>
</li>
<li> <p>A BEFORE plot based on loci heterozygosity by sex, with z/x-linked loci colored 
in orange and gametologs in green 
</p>
</li>
<li> <p>An AFTER plot based on loci heterozygosity by sex, with sex-linked loci removed
</p>
</li></ul>



<h3>Value</h3>

<p>A genlight object and 4 plots.
</p>


<h3>Author(s)</h3>

<p>Custodian: Diana Robledo-Ruiz &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Robledo‐Ruiz, D. A., Austin, L., Amos, J. N., Castrejón‐Figueroa, J.,
Harley, D. K., Magrath, M. J., Sunnucks, P., &amp; Pavlova, A. (2023). 
Easy‐to‐use R functions to separate reduced‐representation genomic datasets
into sex‐linked and autosomal loci, and conduct sex assignment. Molecular 
Ecology Resources, 00, 1-21.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>LBP_noSexLinked &lt;- gl.drop.sexlinked(x = LBP, system = "xy", plot.display = TRUE, ncores = 1)
LBP_noSexLinked

</code></pre>

<hr>
<h2 id='gl.infer.sex'>Uses sex-linked loci to infer sex of individuals</h2><span id='topic+gl.infer.sex'></span>

<h3>Description</h3>

<p>This function uses the output of function gl.keep.sexlinked (list of 5 
elements) to infer the sex of all individuals in the dataset. 
It uses 3 types of sex-linked loci (W-/Y-linked, Z-/X-linked, and 
gametologs), assigns a preliminary genetic sex for each type of sex-linked
loci available, and outputs an agreed sex.
</p>
<p>This function produces as output a dataframe with individuals in rows and 11 
columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.infer.sex(gl_sexlinked, system = NULL, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gl.infer.sex_+3A_gl_sexlinked">gl_sexlinked</code></td>
<td>
<p>The output of function gl.keep.sexlinked (complete 
list with 5 elements). See explanation in &quot;Details&quot; section [required].</p>
</td></tr>
<tr><td><code id="gl.infer.sex_+3A_system">system</code></td>
<td>
<p>String that declares the sex-determination system of the 
species: 'zw' or 'xy' [required].</p>
</td></tr>
<tr><td><code id="gl.infer.sex_+3A_seed">seed</code></td>
<td>
<p>User-defined integer for repeatability purposes. If not provided
by user, it is chosen randomly by default. See &quot;Details&quot; section.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parameter <code>gl_sexlinked</code> must be the name of the output object (a 
list of 5 elements) produced by function <code>gl.keep.sexlinked</code>. Parameter 
<code>seed</code> must be an integer that will be used on the KMeans algorithm used
by the function. We highly recommend choosing the seed to guarantee 
repeatability.
</p>
<p>Note that this function was created with the explicit intent that a human 
checks the evidence for the sex assignments that do NOT agree for all 
types of sex-linked loci (called &quot;indefinite sex assignments&quot; and denoted 
as &quot;*M&quot; or &quot;*F&quot; in the last column of dataframe output). This human can then 
use their criterion to validate these assignments.
</p>
<p><strong> Function's output </strong>
</p>
<p>This function creates a dataframe with one row per individual and 11 columns: </p>

<ul>
<li> <p>id &gt; Individuals' ID.
</p>
</li>
<li> <p>w.linked.sex or y.linked.sex &gt; Sex inferred using w-linked or y-linked
loci.
</p>
</li>
<li> <p>#called &gt; Number of W-linked or Y-linked loci for which the individual 
had a called genotype (cf. missing genotype).
</p>
</li>
<li> <p>#missing &gt; Number of W-linked or Y-linked loci for which the 
individual had a missing genotype (cf. called genotype). 
</p>
</li>
<li> <p>z.linked.sex or x.linked.sex &gt; Sex inferred using z-linked or x-linked
loci.
</p>
</li>
<li> <p>#Hom.z or #Hom.x &gt; Number of z-linked or x-linked loci for which the 
individual is homozygous.
</p>
</li>
<li> <p>#Het.z or #Het.x &gt; Number of z-linked or x-linked loci for which the 
individual is heterozygous.
</p>
</li>
<li> <p>gametolog.sex &gt; Sex inferred using gametologs. 
</p>
</li>
<li> <p>#Hom.g &gt; Number of gametologous loci for which the individual is 
homozygous.
</p>
</li>
<li> <p>#Het.g &gt; Number of gametologous loci for which the individual is 
heterozygous.
</p>
</li>
<li> <p>agreed.sex &gt; Agreed sex: 'F' or 'M' if all preliminary sex-assignments
match (i.e., definite sex assignment), and '*F' or '*M' if NOT all 
preliminary sex-assignments match (i.e., indefinite sex assignment). 
</p>
</li></ul>



<h3>Value</h3>

<p>A dataframe.
</p>


<h3>Author(s)</h3>

<p>Custodian: Diana Robledo-Ruiz  &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Robledo‐Ruiz, D. A., Austin, L., Amos, J. N., Castrejón‐Figueroa, J.,
Harley, D. K., Magrath, M. J., Sunnucks, P., &amp; Pavlova, A. (2023). 
Easy‐to‐use R functions to separate reduced‐representation genomic datasets
into sex‐linked and autosomal loci, and conduct sex assignment. Molecular 
Ecology Resources, 00, 1-21.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>LBP_sexLinked &lt;- gl.keep.sexlinked(x = LBP, system = "xy", plot.display = TRUE, ncores = 1)
inferred.sexes &lt;- gl.infer.sex(gl_sexlinked = LBP_sexLinked, system = "xy", seed = 100)
inferred.sexes

</code></pre>

<hr>
<h2 id='gl.keep.sexlinked'>Keeps loci that are sex linked</h2><span id='topic+gl.keep.sexlinked'></span>

<h3>Description</h3>

<p>This function identifies sex-linked and autosomal loci present in a SNP 
dataset (genlight object) using individuals with known sex. It identifies
five types of loci: w-linked or y-linked, sex-biased, z-linked or
x-linked, gametologous and autosomal.
</p>
<p>This function produces as output a list with 5 elements, including one 
dataframe and 4 genlight objects with sex-linked loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.sexlinked(
  x,
  system = NULL,
  ncores = 1,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gl.keep.sexlinked_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data. This genlight
object needs to contain the sex of the individuals. See explanation in 
details [required].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_system">system</code></td>
<td>
<p>String that declares the sex-determination system of the 
species: 'zw' or 'xy' [required].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_ncores">ncores</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncores
&gt; 1 parallel operation is activated, see &quot;Details&quot; section [default 1].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_plot.display">plot.display</code></td>
<td>
<p>Creates four output plots. See explanation in details
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].[not yet implemented]</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_plot.colors">plot.colors</code></td>
<td>
<p>[not implemented yet]</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, 
exclude extension) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()].</p>
</td></tr>
<tr><td><code id="gl.keep.sexlinked_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genlight object must contain in <code>gl@other$ind.metrics</code> a column 
named &quot;id&quot;, and a column named &quot;sex&quot; in which individuals with known-sex are 
assigned 'M' for male, or 'F' for female. The function ignores individuals 
that are assigned anything else or nothing at all (unknown-sex).
</p>
<p>The creation of plots can be turned-off (<code>plot.display = FALSE</code>) in order
to save a little bit of running time for very large datasets (&gt;50,000 SNPs). 
However, we strongly encourage you to always inspect the output plots at 
least once to make sure everything is working properly.
</p>
<p><strong> Function's output </strong>
</p>
<p>This function returns a list of 5 elements: </p>

<ul>
<li> <p>$results.table &gt; Table with statistics (columns) for each loci (rows)
</p>
</li>
<li> <p>$w.linked or $y.linked &gt; Genlight object with w-linked/y-linked loci
</p>
</li>
<li> <p>$sex.biased &gt; Genlight object with sex-biased scoring rate loci 
</p>
</li>
<li> <p>$z.linked or $x.linked &gt; Genlight object with z-linked/x-linked loci
</p>
</li>
<li> <p>$gametolog &gt; Genlight object with gametologs 
</p>
</li></ul>

<p>And four plots:</p>

<ul>
<li> <p>A BEFORE plot based on loci call rate by sex, with w/y-linked loci colored 
in yellow and sex-biased loci in blue
</p>
</li>
<li> <p>An AFTER plot based on loci call rate by sex, with only sex-linked loci
</p>
</li>
<li> <p>A BEFORE plot based on loci heterozygosity by sex, with z/x-linked loci colored 
in orange and gametologs in green 
</p>
</li>
<li> <p>An AFTER plot based on loci heterozygosity by sex, with only sex-linked loci
</p>
</li></ul>



<h3>Value</h3>

<p>A list of 5 elements and 4 plots.
</p>


<h3>Author(s)</h3>

<p>Custodian: Diana Robledo-Ruiz &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Robledo‐Ruiz, D. A., Austin, L., Amos, J. N., Castrejón‐Figueroa, J.,
Harley, D. K., Magrath, M. J., Sunnucks, P., &amp; Pavlova, A. (2023). 
Easy‐to‐use R functions to separate reduced‐representation genomic datasets
into sex‐linked and autosomal loci, and conduct sex assignment. Molecular 
Ecology Resources, 00, 1-21.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>LBP_sexLinked &lt;- gl.keep.sexlinked(x = LBP, system = "xy", plot.display = TRUE, ncores = 1)
LBP_sexLinked$gametolog

</code></pre>

<hr>
<h2 id='gl.report.sexlinked'>Filters loci that are sex linked</h2><span id='topic+gl.report.sexlinked'></span>

<h3>Description</h3>

<p>This function identifies sex-linked and autosomal loci present in a SNP 
dataset (genlight object) using individuals with known sex. It identifies
five types of loci: w-linked or y-linked, sex-biased, z-linked or
x-linked, gametologous and autosomal.
</p>
<p>This function produces as output a dataframe and 2 plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.sexlinked(
  x,
  system = NULL,
  ncores = 1,
  plot.display = TRUE,
  plot.theme = theme_dartR(),
  plot.colors = NULL,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gl.report.sexlinked_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data. This genlight
object needs to contain the sex of the individuals. See explanation in 
details [required].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_system">system</code></td>
<td>
<p>String that declares the sex-determination system of the 
species: 'zw' or 'xy' [required].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_ncores">ncores</code></td>
<td>
<p>Number of processes to be used in parallel operation. If ncores
&gt; 1 parallel operation is activated, see &quot;Details&quot; section [default 1].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.display">plot.display</code></td>
<td>
<p>Creates two output plots. See explanation in details
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.theme">plot.theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].[not yet implemented]</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.colors">plot.colors</code></td>
<td>
<p>[not implemented yet]</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, 
exclude extension) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The genlight object must contain in <code>gl@other$ind.metrics</code> a column 
named &quot;id&quot;, and a column named &quot;sex&quot; in which individuals with known-sex are 
assigned 'M' for male, or 'F' for female. The function ignores individuals 
that are assigned anything else or nothing at all (unknown-sex).
</p>
<p>The creation of plots can be turned-off (<code>plot.display = FALSE</code>) in order to save a 
little bit of running time for very large datasets (&gt;50,000 SNPs). However, 
we strongly encourage you to always inspect the output plots at least once to
make sure everything is working properly.
</p>
<p><strong> Function's output </strong>
</p>
<p>This function returns two plots:</p>

<ul>
<li> <p>A plot based on loci call rate by sex, with w/y-linked loci colored 
in yellow and sex-biased loci in blue
</p>
</li>
<li> <p>A plot based on loci heterozygosity by sex, with z/x-linked loci colored 
in orange and gametologs in green 
</p>
</li></ul>

<p>And a dataframe in which loci are in rows, and columns:</p>

<ul>
<li> <p>index - Index number to identify loci
</p>
</li>
<li> <p>count.F.miss - Count of females that have this locus as missing data (NA).
</p>
</li>
<li> <p>count.M.miss - Count of males that have this locus as missing data (NA)
</p>
</li>
<li> <p>count.F.scored - Count of females that have this locus scored (0, 1 or 2; i.e. non-missing)
</p>
</li>
<li> <p>count.M.scored - Count of males that have this locus scored (0, 1 or 2; i.e. non-missing)
</p>
</li>
<li> <p>ratio - Fisher's exact test estimate testing for the independence of call rate and sex for this locus
</p>
</li>
<li> <p>p.value - P-value for the Fisher's exact test estimate
</p>
</li>
<li> <p>p.adjusted - P-value adjusted for false discovery rate
</p>
</li>
<li> <p>scoringRate.F - Female call rate (proportion of females that were scored for this locus; x-axis in the 1st plot)
</p>
</li>
<li> <p>scoringRate.M - Male call rate (proportion of males that were scored for this locus; y-axis in the 1st plot)
</p>
</li>
<li> <p>w.linked/y.linked - Boolean for this locus being w-linked/y-linked
</p>
</li>
<li> <p>sex.biased - Boolean for this locus having sex-biased call rate
</p>
</li>
<li> <p>count.F.het - Count of females that are heterozygous for this locus
</p>
</li>
<li> <p>count.M.het - Count of males that are heterozygous for this locus
</p>
</li>
<li> <p>count.F.hom - Count of females that are homozygous for this locus
</p>
</li>
<li> <p>count.M.hom - Count of males that are homozygous for this locus
</p>
</li>
<li> <p>stat - Fisher's exact test estimate testing for the independence of heterozygosity and sex for this locus
</p>
</li>
<li> <p>stat.p.value - P-value for the Fisher's exact test estimate
</p>
</li>
<li> <p>stat.p.adjusted - P-value adjusted for false discovery rate
</p>
</li>
<li> <p>heterozygosity.F - Proportion of females that are heterozygotes for this locus (x-axis in the 2nd plot)
</p>
</li>
<li> <p>heterozygosity.M - Proportion of males that are heterozygotes for this locus (y-axis in the 2nd plot)
</p>
</li>
<li> <p>z.linked/x.linked - Boolean for this locus being z-linked/x.linked
</p>
</li>
<li> <p>gametolog - Boolean for this locus being a gametolog
</p>
</li></ul>



<h3>Value</h3>

<p>A dataframe and 2 plots.
</p>


<h3>Author(s)</h3>

<p>Custodian: Diana Robledo-Ruiz &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Robledo‐Ruiz, D. A., Austin, L., Amos, J. N., Castrejón‐Figueroa, J.,
Harley, D. K., Magrath, M. J., Sunnucks, P., &amp; Pavlova, A. (2023). 
Easy‐to‐use R functions to separate reduced‐representation genomic datasets
into sex‐linked and autosomal loci, and conduct sex assignment. Molecular 
Ecology Resources, 00, 1-21.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.sexlinked(x = LBP, system = "xy", plot.display = TRUE, ncores = 1)

</code></pre>

<hr>
<h2 id='zzz'>Setting up the package dartR.sexlinked</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting up dartR.sexlinked
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
