<!DOCTYPE html><html><head><title>Help for package dartR.popgen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR.popgen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gl.blast'><p>Aligns nucleotides sequences against those present in a target database</p>
using blastn</a></li>
<li><a href='#gl.collapse'><p>Collapses a distance matrix by amalgamating populations with pairwise</p>
fixed difference count less that a threshold</a></li>
<li><a href='#gl.evanno'><p>Creates an Evanno plot from a STRUCTURE run object</p></a></li>
<li><a href='#gl.ld.distance'><p>Plots linkage disequilibrium against distance by population</p>
disequilibrium patterns</a></li>
<li><a href='#gl.ld.haplotype'><p>Visualize patterns of linkage disequilibrium and identification of</p>
haplotypes</a></li>
<li><a href='#gl.LDNe'><p>Estimates effective population size using the Linkage Disequilibrium</p>
method based on NeEstimator (V2)</a></li>
<li><a href='#gl.map.structure'><p>Maps a STRUCTURE plot using a genlight object</p></a></li>
<li><a href='#gl.nhybrids'><p>Creates an input file for the program NewHybrids and runs it if</p>
NewHybrids is installed</a></li>
<li><a href='#gl.outflank'><p>Identifies loci under selection per population using the outflank</p>
method of Whitlock and Lotterhos (2015)</a></li>
<li><a href='#gl.plot.faststructure'><p>Plots fastStructure analysis results (Q-matrix)</p></a></li>
<li><a href='#gl.plot.structure'><p>Plots STRUCTURE analysis results (Q-matrix)</p></a></li>
<li><a href='#gl.run.faststructure'><p>Runs a faststructure analysis using a genlight object</p></a></li>
<li><a href='#gl.run.structure'><p>Runs a STRUCTURE analysis using a genlight object</p></a></li>
<li><a href='#gl.sfs'><p>Creates a site frequency spectrum based on a dartR or genlight object</p></a></li>
<li><a href='#utils.outflank'><p>OutFLANK:  An Fst outlier approach by Mike Whitlock and Katie Lotterhos,</p>
University of British Columbia.</a></li>
<li><a href='#utils.outflank.MakeDiploidFSTMat'><p>Creates OutFLANK input file from individual genotype info.</p></a></li>
<li><a href='#utils.outflank.plotter'><p>Plotting functions for Fst distributions after OutFLANK</p></a></li>
<li><a href='#utils.structure.evanno'><p>Util function for evanno plots</p></a></li>
<li><a href='#utils.structure.genind2gtypes'><p>structure util functions</p></a></li>
<li><a href='#utils.structure.run'><p>Utility function to run Structure</p></a></li>
<li><a href='#zzz'><p>Setting up the package dartR.popgen</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysing 'SNP' and 'Silicodart' Data Generated by Genome-Wide
Restriction Fragment Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.32</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-21</td>
</tr>
<tr>
<td>Description:</td>
<td>Facilitates the analysis of SNP (single nucleotide polymorphism) 
    and silicodart (presence/absence) data. 'dartR.popgen' provides a suit of 
    functions to analyse such data in a population genetics context. It provides
    several functions to calculate population genetic metrics and to study 
    population structure. Quite a few functions need additional software to be 
    able to run (gl.run.structure(), gl.blast(), gl.LDNe()). You find detailed description 
    in the help pages how to download and link the packages so the function can 
    run the software. 'dartR.popgen' is part of the the 'dartRverse' suit of packages.
    Gruber et al. (2018) &lt;<a href="https://doi.org/10.1111%2F1755-0998.12745">doi:10.1111/1755-0998.12745</a>&gt;.
    Mijangos et al. (2022) &lt;<a href="https://doi.org/10.1111%2F2041-210X.13918">doi:10.1111/2041-210X.13918</a>&gt;.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), adegenet (&ge; 2.0.0), dartR.base, dartR.data</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, utils, MASS, dplyr, patchwork, crayon, ggplot2,
data.table, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>SIBER, expm, fields, gplots, gridExtra, igraph, iterpc,
label.switching, leaflet, plyr, proxy, purrr, qvalue, raster,
reshape2, scales, snpStats, tidyr, viridis, zoo, gsubfn, sp</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-20 23:51:05 UTC; s425824</td>
</tr>
<tr>
<td>Author:</td>
<td>Bernd Gruber [aut, cre],
  Arthur Georges [aut],
  Jose L. Mijangos [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb],
  Lindsay V. Clark [ctb],
  Floriaan Devloo-Delva [ctb],
  Eric Archer [ctb]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/dartr?pli=1">https://groups.google.com/g/dartr?pli=1</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bernd Gruber &lt;bernd.gruber@canberra.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-21 18:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gl.blast'>Aligns nucleotides sequences against those present in a target database
using blastn</h2><span id='topic+gl.blast'></span>

<h3>Description</h3>

<p>Basic Local Alignment Search Tool (BLAST; Altschul et al., 1990 &amp;
1997) is a sequence comparison algorithm optimized for speed used to search
sequence databases for optimal local alignments to a query. This function
creates fasta files, creates databases to run BLAST, runs blastn and filters
these results to obtain the best hit per sequence.
</p>
<p>This function can be used to run BLAST alignment of short-read (DArTseq
data) and long-read sequences (Illumina, PacBio... etc). You can use
reference genomes from NCBI, genomes from your private collection, contigs,
scaffolds or any other genetic sequence that you would like to use as
reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.blast(
  x,
  ref_genome,
  task = "megablast",
  Percentage_identity = 70,
  Percentage_overlap = 0.8,
  bitscore = 50,
  number_of_threads = 2,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.blast_+3A_x">x</code></td>
<td>
<p>Either a genlight object containing a column named
'TrimmedSequence' containing the sequence of the SNPs (the sequence tag)
trimmed of adapters as provided by DArT; or a path to a fasta file with the
query sequences [required].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_ref_genome">ref_genome</code></td>
<td>
<p>Path to a reference genome in fasta of fna format
[required].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_task">task</code></td>
<td>
<p>Four different tasks are supported: 1) “megablast”, for very
similar sequences (e.g, sequencing errors), 2) “dc-megablast”, typically
used for inter-species comparisons, 3) “blastn”, the traditional program
used for inter-species comparisons, 4) “blastn-short”, optimized for
sequences less than 30 nucleotides [default 'megablast'].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_percentage_identity">Percentage_identity</code></td>
<td>
<p>Not a very sensitive or reliable measure of
sequence similarity, however it is a reasonable proxy for evolutionary
distance. The evolutionary distance associated with a 10 percent change in
Percentage_identity is much greater at longer distances. Thus, a change from
80 – 70 percent identity might reflect divergence 200 million years earlier
in time, but the change from 30 percent to 20 percent might correspond to a
billion year divergence time change [default 70].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_percentage_overlap">Percentage_overlap</code></td>
<td>
<p>Calculated as alignment length divided by the
query length or subject length (whichever is shortest of the two lengths,
i.e.  length / min(qlen,slen) ) [default 0.8].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_bitscore">bitscore</code></td>
<td>
<p>A rule-of-thumb for inferring homology, a bit score of 50
is almost always significant [default 50].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_number_of_threads">number_of_threads</code></td>
<td>
<p>Number of threads (CPUs) to use in blastn search
[default 2].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_verbose">verbose</code></td>
<td>
<p>verbose= 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Installing BLAST</strong>
</p>
<p>You can download the BLAST installs from:
<a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>It is important to install BLAST in a path that does not contain spaces for
this function to work.
</p>
<p><strong>Running BLAST</strong>
</p>
<p>Four different tasks are supported: </p>
 <ul>
<li><p> “megablast”, for very
similar sequences (e.g, sequencing errors) </p>
</li>
<li><p> “dc-megablast”, typically
used for inter-species comparisons </p>
</li>
<li><p> “blastn”, the traditional program
used for inter-species comparisons </p>
</li>
<li><p> “blastn-short”, optimized for
sequences less than 30 nucleotides </p>
</li></ul>

<p>If  you  are  running  a  BLAST alignment of  similar  sequences,  for
example  Turtle  Genome  Vs Turtle Sequences, the recommended parameters
are: task = “megablast”, Percentage_identity = 70, Percentage_overlap =  0.8
and bitscore = 50.
</p>
<p>If you are running a BLAST alignment of highly dissimilar sequences because
you are probably looking for sex linked  hits in  a distantly  related
species,  and  you  are aligning for example sequences of Chicken Genome Vs
Bassiana, the recommended parameters are: task = “dc-megablast”,
Percentage_identity = 50, Percentage_overlap =  0.01 and bitscore = 30.
</p>
<p>Be aware that running BLAST might take a long time (i.e. days) depending of
the size of your query, the size of your database and the number of threads
selected for your computer.
</p>
<p><strong>BLAST output</strong>
</p>
<p>The BLAST output is formatted as a table using output format 6, with columns
defined in the following order: </p>
 <ul>
<li><p> qseqid - Query Seq-id </p>
</li>
<li>
<p>sacc - Subject accession </p>
</li>
<li><p> stitle - Subject Title </p>
</li>
<li><p> qseq - Aligned
part of query sequence </p>
</li>
<li><p> sseq - Aligned part of subject sequence </p>
</li>
<li>
<p>nident - Number of identical matches </p>
</li>
<li><p> mismatch - Number of mismatches
</p>
</li>
<li><p> pident - Percentage of identical matches </p>
</li>
<li><p> length - Alignment
length </p>
</li>
<li><p> evalue - Expect value </p>
</li>
<li><p> bitscore - Bit score </p>
</li>
<li><p> qstart -
Start of alignment in query </p>
</li>
<li><p> qend - End of alignment in query </p>
</li>
<li>
<p>sstart - Start of alignment in subject </p>
</li>
<li><p> send - End of alignment in
subject </p>
</li>
<li><p> gapopen - Number of gap openings </p>
</li>
<li><p> gaps - Total number of
gaps </p>
</li>
<li><p> qlen - Query sequence length </p>
</li>
<li><p> slen - Subject sequence length
</p>
</li>
<li><p> PercentageOverlap - length / min(qlen,slen) </p>
</li></ul>

<p>Databases containing unfiltered aligned sequences, filtered aligned
sequences and one hit per sequence are saved to the working directory
(plot.dir tempdir if not set).
</p>
<p><strong>BLAST filtering</strong>
</p>
<p>BLAST output is filtered by ordering the hits of each sequence first by the
highest percentage identity, then the highest percentage overlap and then
the highest bitscore. Only one hit per sequence is kept based on these
selection criteria.
</p>


<h3>Value</h3>

<p>If the input is a genlight object: returns a genlight object with one
hit per sequence merged to the slot $other$loc.metrics. If the input is a
fasta file: returns a dataframe with one hit per sequence.
</p>


<h3>Author(s)</h3>

<p>Berenice Talamantes Becerra &amp; Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Altschul, S. F., Gish, W., Miller, W., Myers, E. W., &amp; Lipman, D.
J. (1990). Basic local alignment search tool. Journal of molecular biology,
215(3), 403-410.
</p>
</li>
<li><p> Altschul, S. F., Madden, T. L., Schäffer, A. A., Zhang, J., Zhang,
Z., Miller, W., &amp; Lipman, D. J. (1997). Gapped BLAST and PSI-BLAST: a new
generation of protein database search programs. Nucleic acids research,
25(17), 3389-3402.
</p>
</li>
<li><p> Pearson, W. R. (2013). An introduction to sequence similarity
(“homology”) searching. Current protocols in bioinformatics, 42(1), 3-1.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="dartR.base.html#topic+gl.print.history">gl.print.history</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- gl.blast(x = testset.gl, ref_genome = "sequence.fasta")
# display of reports saved in the temporal directory
# open the reports saved in the temporal directory

## End(Not run)

</code></pre>

<hr>
<h2 id='gl.collapse'>Collapses a distance matrix by amalgamating populations with pairwise
fixed difference count less that a threshold</h2><span id='topic+gl.collapse'></span>

<h3>Description</h3>

<p>This script takes a file generated by gl.fixed.diff and amalgamates
populations with distance less than or equal to a specified threshold. The
distance matrix is generated by gl.fixed.diff().
</p>
<p>The script then applies the new population assignments to the genlight object
and recalculates the distance and associated matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.collapse(fd, tpop = 0, tloc = 0, pb = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.collapse_+3A_fd">fd</code></td>
<td>
<p>Name of the list of matrices produced by gl.fixed.diff() 
[required].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_tpop">tpop</code></td>
<td>
<p>Threshold number of fixed differences above which populations
will not be amalgamated [default 0].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_tloc">tloc</code></td>
<td>
<p>Threshold defining a fixed difference (e.g. 0.05 implies 95:5 vs
5:95 is fixed) [default 0].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_pb">pb</code></td>
<td>
<p>If TRUE, show a progress bar on time consuming loops 
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the gl object x and the following square matrices:
</p>

<ol>
<li><p> $gl &ndash; the new genlight object with populations collapsed;
</p>
</li>
<li><p> $fd &ndash; raw fixed differences;
</p>
</li>
<li><p> $pcfd &ndash; percent fixed differences;
</p>
</li>
<li><p> $nobs &ndash; mean no. of individuals used in each comparison;
</p>
</li>
<li><p> $nloc &ndash; total number of loci used in each comparison;
</p>
</li>
<li><p> $expfpos &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li>
<li><p> $expfpos &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li>
<li><p> $prob &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
fd &lt;- gl.fixed.diff(testset.gl,tloc=0.05)
fd
fd2 &lt;- gl.collapse(fd,tpop=1)
fd2
fd3 &lt;- gl.collapse(fd2,tpop=1)
fd3

 fd &lt;- gl.fixed.diff(testset.gl,tloc=0.05)
 fd2 &lt;- gl.collapse(fd)
</code></pre>

<hr>
<h2 id='gl.evanno'>Creates an Evanno plot from a STRUCTURE run object</h2><span id='topic+gl.evanno'></span>

<h3>Description</h3>

<p>This function takes a genlight object and runs a STRUCTURE analysis based on
functions from <code>strataG</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.evanno(sr, plot.out = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.evanno_+3A_sr">sr</code></td>
<td>
<p>structure run object from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.evanno_+3A_plot.out">plot.out</code></td>
<td>
<p>TRUE: all four plots are shown. FALSE: all four plots are
returned as a ggplot but not shown [default TRUE].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is basically a convenient wrapper around the beautiful
strataG function <code>evanno</code> (Archer et al. 2016). For a detailed
description please refer to this package (see references below).
</p>


<h3>Value</h3>

<p>An Evanno plot is created and a list of all four plots is returned.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li>
<li><p> Evanno, G., Regnaut, S., and J. Goudet. 2005. Detecting the number of
clusters of individuals using the software STRUCTURE: a simulation study.
Molecular Ecology 14:2611-2620.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.run.structure">gl.run.structure</a></code>,  <code>clumpp</code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'># examples need structure to be installed on the system (see above)
## Not run: 
 bc &lt;- bandicoot.gl[,1:100]
 sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure.exe')
 ev &lt;- gl.evanno(sr)
 ev
 qmat &lt;- gl.plot.structure(sr, K=3)
 head(qmat)
 gl.map.structure(qmat, bc, K=3, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.ld.distance'>Plots linkage disequilibrium against distance by population
disequilibrium patterns</h2><span id='topic+gl.ld.distance'></span>

<h3>Description</h3>

<p>The function creates a plot showing
the pairwise LD measure against distance in number of base pairs pooled over
all the chromosomes and a red line representing the threshold (R.squared =
0.2) that is commonly used to imply that two loci are unlinked (Delourme et
al., 2013; Li et al., 2014).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ld.distance(
  ld_report,
  ld_resolution = 1e+05,
  pop_colors = NULL,
  plot_theme = NULL,
  plot.out = TRUE,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ld.distance_+3A_ld_report">ld_report</code></td>
<td>
<p>Output from function <code><a href="dartR.base.html#topic+gl.report.ld.map">gl.report.ld.map</a></code>
[required].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_ld_resolution">ld_resolution</code></td>
<td>
<p>Resolution at which LD should be reported in number of
base pairs [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_pop_colors">pop_colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with information of LD against distance by population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Delourme, R., Falentin, C., Fomeju, B. F., Boillot, M., Lassalle, G.,
André, I., . . . Marty, A. (2013). High-density SNP-based genetic map
development and linkage disequilibrium assessment in Brassica napusL. BMC
genomics, 14(1), 120.
</p>
</li>
<li><p> Li, X., Han, Y., Wei, Y., Acharya, A., Farmer, A. D., Ho, J., . . .
Brummer, E. C. (2014). Development of an alfalfa SNP array and its use to
evaluate patterns of population structure and linkage disequilibrium. PLoS
One, 9(1), e84329.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other ld functions: 
<code><a href="#topic+gl.ld.haplotype">gl.ld.haplotype</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if ((requireNamespace("snpStats", quietly = TRUE)) &amp; (requireNamespace("fields", quietly = TRUE))) {
  require("dartR.data")
  x &lt;- platypus.gl
  x &lt;- gl.filter.callrate(x, threshold = 1)
  x &lt;- gl.filter.monomorphs(x)
  x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
  x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
  ld_res &lt;- gl.report.ld.map(x, ld.max.pairwise = 10000000)
  ld_res_2 &lt;- gl.ld.distance(ld_res, ld_resolution = 1000000)
}
</code></pre>

<hr>
<h2 id='gl.ld.haplotype'>Visualize patterns of linkage disequilibrium and identification of
haplotypes</h2><span id='topic+gl.ld.haplotype'></span>

<h3>Description</h3>

<p>This function plots a Linkage disequilibrium (LD) heatmap, where the colour
shading indicates the strength of LD. Chromosome positions (Mbp) are shown on
the horizontal axis, and haplotypes appear as triangles and delimited by
dark yellow vertical lines. Numbers identifying each haplotype are shown in
the upper part of the plot.
</p>
<p>The heatmap also shows heterozygosity for each SNP.
</p>
<p>The function identifies haplotypes based on contiguous SNPs that are in
linkage disequilibrium using as threshold <code>ld_threshold_haplo</code> and
containing more than <code>min_snps</code> SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ld.haplotype(
  x,
  pop_name = NULL,
  chrom_name = NULL,
  ld_max_pairwise = 1e+07,
  maf = 0.05,
  ld_stat = "R.squared",
  ind.limit = 10,
  min_snps = 10,
  ld_threshold_haplo = 0.5,
  coordinates = NULL,
  color_haplo = "viridis",
  color_het = "deeppink",
  plot.out = TRUE,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ld.haplotype_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_pop_name">pop_name</code></td>
<td>
<p>Name of the population to analyse. If NULL all the
populations are analised [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_chrom_name">chrom_name</code></td>
<td>
<p>Nme of the chromosome to analyse. If NULL all the
chromosomes are analised [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_max_pairwise">ld_max_pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD
should be calculated [default 10000000].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_stat">ld_stat</code></td>
<td>
<p>The LD measure to be calculated: &quot;LLR&quot;, &quot;OR&quot;, &quot;Q&quot;, &quot;Covar&quot;,
&quot;D.prime&quot;, &quot;R.squared&quot;, and &quot;R&quot;. See <code><a href="snpStats.html#topic+ld">ld</a></code>
(package snpStats) for details [default &quot;R.squared&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_min_snps">min_snps</code></td>
<td>
<p>Minimum number of SNPs that should have a haplotype to call
it [default 10].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_threshold_haplo">ld_threshold_haplo</code></td>
<td>
<p>Minimum LD between adjacent SNPs to call a
haplotype [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_coordinates">coordinates</code></td>
<td>
<p>A vector of two elements with the start and end
coordinates in base pairs to which restrict the
analysis e.g. c(1,1000000) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_color_haplo">color_haplo</code></td>
<td>
<p>Color palette for haplotype plot. See details
[default &quot;viridis&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_color_het">color_het</code></td>
<td>
<p>Color for heterozygosity [default &quot;deeppink&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if heatmap plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The information for SNP's position should be stored in the genlight accessor
&quot;@position&quot; and the SNP's chromosome name in the accessor &quot;@chromosome&quot;
(see examples). The function will then calculate LD within each chromosome.
</p>
<p>The output of the function includes a table with the haplotypes
that were identified and their location.
</p>
<p>Colors of the heatmap (<code>color_haplo</code>) are based on the function
<code><a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a></code> from  package <code>viridis</code>.
Other color palettes options are &quot;magma&quot;, &quot;inferno&quot;, &quot;plasma&quot;, &quot;viridis&quot;,
&quot;cividis&quot;, &quot;rocket&quot;, &quot;mako&quot; and &quot;turbo&quot;.
</p>


<h3>Value</h3>

<p>A table with the haplotypes that were identified.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other ld functions: 
<code><a href="#topic+gl.ld.distance">gl.ld.distance</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x, threshold = 1)
# only the first 20 individuals because of speed during tests
x &lt;- gl.keep.pop(x, pop.list = "TENTERFIELD")[1:20, ]
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
ld_res &lt;- gl.ld.haplotype(x,
  chrom_name = "NC_041728.1_chromosome_1",
  ld_max_pairwise = 10000000
)

</code></pre>

<hr>
<h2 id='gl.LDNe'>Estimates effective population size using the Linkage Disequilibrium
method based on NeEstimator (V2)</h2><span id='topic+gl.LDNe'></span>

<h3>Description</h3>

<p>This function is basically a convenience function that runs the LD Ne
estimator using Neestimator2
(<a href="http://www.molecularfisherieslaboratory.com.au/neestimator-software/">http://www.molecularfisherieslaboratory.com.au/neestimator-software/</a>)
within R using the provided genlight object. To be able to do so, the
software has to be downloaded from their website and the appropriate
executable Ne2-1 has to be copied into the path as specified in the function
(see example below).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.LDNe(
  x,
  outfile = "genepopLD.txt",
  outpath = tempdir(),
  neest.path = getwd(),
  critical = 0,
  singleton.rm = TRUE,
  mating = "random",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = gl.select.colors(x, verbose = 0),
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.LDNe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file  with
all results from Neestimator 2 [default 'genepopLD.txt'].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_neest.path">neest.path</code></td>
<td>
<p>Path to the folder of the NE2-1 file.
Please note there are 3 different executables depending on your OS:
Ne2-1.exe (=Windows), Ne2-1M (=Mac), Ne2-1L (=Linux). You only need to point
to the folder (the function will recognise which OS you are running)
[default getwd()].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_critical">critical</code></td>
<td>
<p>(vector of) Critical values that are used to remove alleles
based on their minor allele frequency. This can be done before using the
gl.filter.maf function, therefore the default is set to 0 (no loci are
removed). To run for MAF 0 and MAF 0.05 at the same time specify: critical =
c(0,0.05) [default 0].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_singleton.rm">singleton.rm</code></td>
<td>
<p>Whether to remove singleton alleles [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_mating">mating</code></td>
<td>
<p>Formula for Random mating='random' or monogamy= 'monogamy'
[default 'random'].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>population colors with as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe with the results as table
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# SNP data (use two populations and only the first 100 SNPs)
pops &lt;- possums.gl[1:60, 1:100]
nes &lt;- gl.LDNe(pops,
  outfile = "popsLD.txt", outpath = tempdir(),
  neest.path = "./path_to Ne-21",
  critical = c(0, 0.05), singleton.rm = TRUE, mating = "random"
)
nes

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.map.structure'>Maps a STRUCTURE plot using a genlight object</h2><span id='topic+gl.map.structure'></span>

<h3>Description</h3>

<p>This function takes the output of plotstructure (the q matrix) and maps the
q-matrix across using the population centers from the genlight object that
was used to run the structure analysis via <code><a href="#topic+gl.run.structure">gl.run.structure</a></code>)
and plots the typical structure bar plots on a spatial map, providing a
barplot for each subpopulation. Therefore it requires coordinates from a
genlight object. This kind of plots should support the interpretation of the
spatial structure of a population, but in principle is not different from
<code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.map.structure(
  qmat,
  x,
  K,
  provider = "Esri.NatGeoWorldMap",
  scalex = 1,
  scaley = 1,
  movepops = NULL,
  pop.labels = TRUE,
  pop.labels.cex = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.map.structure_+3A_qmat">qmat</code></td>
<td>
<p>Q-matrix from a structure run followed by a clumpp run object
[from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> and <code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>]
[required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the coordinates in the
<code>\@other$latlon</code> slot to calculate the population centers [required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_k">K</code></td>
<td>
<p>The number for K to be plotted [required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_provider">provider</code></td>
<td>
<p>Provider    passed to leaflet. Check <a href="leaflet.html#topic+providers">providers</a>
for a list of possible backgrounds [default &quot;Esri.NatGeoWorldMap&quot;].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_scalex">scalex</code></td>
<td>
<p>Scaling factor to determine the size of the bars in x direction
[default 1].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_scaley">scaley</code></td>
<td>
<p>Scaling factor to determine the size of the bars in y direction
[default 1].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_movepops">movepops</code></td>
<td>
<p>A two-dimensional data frame that allows to move the center of
the barplots manually in case they overlap. Often if populations are
horizontally close to each other. This needs to be a data.frame of the
dimensions [rows=number of populations, columns = 2 (lon/lat)]. For each
population you have to specify the x and y (lon and lat) units you want to
move the center of the plot, (see example for details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_pop.labels">pop.labels</code></td>
<td>
<p>Switch for population labels below the parplots
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_pop.labels.cex">pop.labels.cex</code></td>
<td>
<p>Size of population labels [default 12].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a mapped version of structure plots. For possible background maps
check as specified via the provider:
<a href="http://leaflet-extras.github.io/leaflet-providers/preview/index.html">http://leaflet-extras.github.io/leaflet-providers/preview/index.html</a>.
You may need to adjust scalex and scaley values [default 1], as the size
depends on the scale of the map and the position of the populations.
</p>


<h3>Value</h3>

<p>An interactive map that shows the structure plots broken down by
population.
</p>
<p>returns the map and a list of the qmat split into sorted matrices per
population. This can be used to create your own map.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li>
<li><p> Evanno, G., Regnaut, S., and J. Goudet. 2005. Detecting the number of
clusters of individuals using the software STRUCTURE: a simulation study.
Molecular Ecology 14:2611-2620.
</p>
</li>
<li><p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.run.structure">gl.run.structure</a></code>,  <code>clumpp</code>,
<code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># examples need structure to be installed on the system (see above)
## Not run: 
bc &lt;- bandicoot.gl[,1:100]
sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure.exe')
ev &lt;- gl.evanno(sr)
ev
qmat &lt;- gl.plot.structure(sr, k=2:4)#' #head(qmat)
gl.map.structure(qmat, bc,K=3)
gl.map.structure(qmat, bc,K=4)
# move population 4 (out of 5) 0.5 degrees to the right and populations 1
# 0.3 degree to the north of the map.
mp &lt;- data.frame(lon=c(0,0,0,0.5,0), lat=c(-0.3,0,0,0,0))
gl.map.structure(qmat, bc,K=4, movepops=mp)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.nhybrids'>Creates an input file for the program NewHybrids and runs it if
NewHybrids is installed</h2><span id='topic+gl.nhybrids'></span>

<h3>Description</h3>

<p>This function compares two sets of parental populations to identify loci that
exhibit a fixed difference, returns an genlight object with the reduced
data, and creates an input file for the program NewHybrids using the top 200
(or hard specified loc.limit) loci. In the absence of two identified
parental populations, the script will select a random set 200 loci only
(method='random') or the first 200 loci ranked on information content
(method='AvgPIC').
</p>
<p>A fixed difference occurs when a SNP allele is present in all individuals
of one population and absent in the other. There is provision for setting
a level of tolerance, e.g. threshold = 0.05 which considers alleles present
at greater than 95
a fixed difference. Only the 200 loci are retained, because of limitations
of NewHybids.
</p>
<p>If you specify a directory for the NewHybrids executable file, then the
script will create the input file from the SNP data then run NewHybrids. If
the directory is set to NULL, the execution will stop once the input file
(default='nhyb.txt') has been written to disk. Note: the executable option
will not work on a Mac; Mac users should generate the NewHybrids input file
and run this on their local installation of NewHybrids.
</p>
<p>Refer to the New Hybrids manual for further information on the parameters to
set
&ndash; http://ib.berkeley.edu/labs/slatkin/eriq/software/new_hybs_doc1_1Beta3.pdf
</p>
<p>It is important to stringently filter the data on RepAvg and CallRate if
using the random option. One might elect to repeat the analysis
(method='random') and combine the resultant posterior probabilities should
200 loci be considered insufficient.
</p>
<p>The F1 individuals should be homozygous at all loci for which the parental
populations are fixed and different, assuming parental populations have been
specified. Sampling errors can result in this not being the case, especially
where the sample sizes for the parental populations are small. Alternatively,
the threshold for posterior probabilities used to determine assignment
(pprob) or the definition of a fixed difference (threshold) may be too lax.
To assess the error rate in the determination of assignment of F1
individuals, a plot of the frequency of homozygous reference, heterozygotes
and homozygous alternate (SNP) can be produced by setting plot=TRUE (the
default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.nhybrids(
  gl,
  outpath = tempdir(),
  p0 = NULL,
  p1 = NULL,
  threshold = 0,
  method = "random",
  plot = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = gl.select.colors(ncolors = 2, verbose = 0),
  pprob = 0.95,
  nhyb.directory = NULL,
  BurnIn = 10000,
  sweeps = 10000,
  GtypFile = "TwoGensGtypFreq.txt",
  AFPriorFile = NULL,
  PiPrior = "Jeffreys",
  ThetaPrior = "Jeffreys",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.nhybrids_+3A_gl">gl</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir()].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_p0">p0</code></td>
<td>
<p>List of populations to be regarded as parental population 0
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_p1">p1</code></td>
<td>
<p>List of populations to be regarded as parental population 1
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_threshold">threshold</code></td>
<td>
<p>Sets the level at which a gene frequency difference is
considered to be fixed [default 0].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_method">method</code></td>
<td>
<p>Specifies the method (random) to select 200 loci for
NewHybrids [default random]. Previous AvgPic does not work anymore!</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot">plot</code></td>
<td>
<p>If TRUE, a plot of the frequency of homozygous reference,
heterozygotes and homozygous alternate (SNP) is produced for the F1
individuals
[default TRUE, applies only if both parental populations are specified].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two colors].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_pprob">pprob</code></td>
<td>
<p>Threshold level for assignment to likelihood bins
[default 0.95, used only if plot=TRUE].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_nhyb.directory">nhyb.directory</code></td>
<td>
<p>Directory that holds the NewHybrids executable file
e.g. C:/NewHybsPC [default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_burnin">BurnIn</code></td>
<td>
<p>Number of sweeps to use in the burn in [default 10000].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_sweeps">sweeps</code></td>
<td>
<p>Number  of  sweeps  to  use  in  computing  the  actual Monte
Carlo averages [default 10000].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_gtypfile">GtypFile</code></td>
<td>
<p>Name of a file containing the genotype frequency classes
[default TwoGensGtypFreq.txt].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_afpriorfile">AFPriorFile</code></td>
<td>
<p>Name of the file containing prior allele frequency
information [default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_piprior">PiPrior</code></td>
<td>
<p>Jeffreys-like priors or Uniform priors for the parameter pi
[default Jeffreys].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_thetaprior">ThetaPrior</code></td>
<td>
<p>Jeffreys-like priors or Uniform priors for the parameter
theta [default Jeffreys].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight object, if parentals are provided; output of
NewHybrids is saved to the working directory.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>

<p>Anderson, E.C. and Thompson, E.A.(2002). A model-based method for identifying
species hybrids using multilocus genetic data. Genetics. 160:1217-1229.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
m &lt;- gl.nhybrids(testset.gl,
  p0 = NULL, p1 = NULL,
  nhyb.directory = "D:/workspace/R/NewHybsPC", # Specify as necessary
  outpath = "D:/workspace", # Specify as necessary, usually getwd() [= workspace]
  BurnIn = 100,
  sweeps = 100,
  verbose = 3
)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.outflank'>Identifies loci under selection per population using the outflank
method of Whitlock and Lotterhos (2015)</h2><span id='topic+gl.outflank'></span>

<h3>Description</h3>

<p>Identifies loci under selection per population using the outflank
method of Whitlock and Lotterhos (2015)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.outflank(
  gi,
  plot = TRUE,
  LeftTrimFraction = 0.05,
  RightTrimFraction = 0.05,
  Hmin = 0.1,
  qthreshold = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.outflank_+3A_gi">gi</code></td>
<td>
<p>A genlight or genind object, with a defined population structure
[required].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_plot">plot</code></td>
<td>
<p>A switch if a barplot is wanted [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_lefttrimfraction">LeftTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
lower end of the range of Fst before the likelihood function is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_righttrimfraction">RightTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
upper end of the range of Fst before the likelihood function is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including calculations
from a locus [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_qthreshold">qthreshold</code></td>
<td>
<p>The desired false discovery rate threshold for calculating
q-values [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_...">...</code></td>
<td>
<p>additional parameters (see documentation of outflank on github).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper around the outflank function provided by
Whitlock and Lotterhos. To be able to run this function the packages qvalue
(from bioconductor) and outflank (from github) needs to be installed. To do
so see example below.
</p>


<h3>Value</h3>

<p>Returns an index of outliers and the full outflank list
</p>


<h3>References</h3>

<p>Whitlock, M.C. and Lotterhos K.J. (2015) Reliable detection of loci
responsible for local adaptation: inference of a neutral model through
trimming the distribution of Fst. The American Naturalist 186: 24 - 36.
</p>
<p>Github repository: Whitlock &amp; Lotterhos:
<a href="https://github.com/whitlock/OutFLANK">https://github.com/whitlock/OutFLANK</a> (Check the readme.pdf within the
repository for an explanation. Be aware you now can run OufFLANK from a
genlight object)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utils.outflank">utils.outflank</a></code>, <code><a href="#topic+utils.outflank.plotter">utils.outflank.plotter</a></code>,
<code><a href="#topic+utils.outflank.MakeDiploidFSTMat">utils.outflank.MakeDiploidFSTMat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gl.outflank(bandicoot.gl, plot = TRUE)

</code></pre>

<hr>
<h2 id='gl.plot.faststructure'>Plots fastStructure analysis results (Q-matrix)</h2><span id='topic+gl.plot.faststructure'></span>

<h3>Description</h3>

<p>This function takes a fastStructure run object (output from
<code><a href="#topic+gl.run.faststructure">gl.run.faststructure</a></code>) and plots the typical structure bar
plot that visualize the q matrix of a fastStructure run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.faststructure(
  sr,
  k.range,
  met_clumpp = "greedyLargeK",
  iter_clumpp = 100,
  clumpak = TRUE,
  plot_theme = NULL,
  colors_clusters = NULL,
  ind_name = TRUE,
  border_ind = 0.15
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.faststructure_+3A_sr">sr</code></td>
<td>
<p>fastStructure run object from <code><a href="#topic+gl.run.faststructure">gl.run.faststructure</a></code>
[required].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_k.range">k.range</code></td>
<td>
<p>The number for K of the q matrix that should be plotted. Needs
to be within you simulated range of K's in your sr structure run object. If
NULL, all the K's are plotted [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_met_clumpp">met_clumpp</code></td>
<td>
<p>The algorithm to use to infer the correct permutations.
One of 'greedy' or 'greedyLargeK' or 'stephens' [default &quot;greedyLargeK&quot;].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_iter_clumpp">iter_clumpp</code></td>
<td>
<p>The number of iterations to use if running either 'greedy'
'greedyLargeK' [default 100].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_clumpak">clumpak</code></td>
<td>
<p>Whether use the Clumpak method (see details) [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_colors_clusters">colors_clusters</code></td>
<td>
<p>A color palette for clusters (K) or a list with
as many colors as there are clusters (K) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_ind_name">ind_name</code></td>
<td>
<p>Whether to plot individual names [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.faststructure_+3A_border_ind">border_ind</code></td>
<td>
<p>The width of the border line between individuals
[default 0.25].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function outputs a barplot which is the typical output of
fastStructure.
</p>
<p>This function is based on the methods of CLUMPP and Clumpak as implemented
in the R package starmie (https://github.com/sa-lee/starmie).
</p>
<p>The Clumpak method identifies sets of highly similar runs among
all the replicates of the same K. The method then separates the distinct
groups of runs representing distinct modes in the space of possible solutions.
</p>
<p>The CLUMPP method permutes the clusters output by independent runs of
clustering programs such as structure, so that they match up as closely as
possible.
</p>
<p>This function averages the replicates within each mode identified by the
Clumpak method.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>List of Q-matrices
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Raj, A., Stephens, M., &amp; Pritchard, J. K. (2014). fastSTRUCTURE:
variational inference of population structure in large SNP data sets.
Genetics, 197(2), 573-589.
</p>
</li>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Kopelman, Naama M., et al. &quot;Clumpak: a program for identifying
clustering modes and packaging population structure inferences across K.&quot;
Molecular ecology resources 15.5 (2015): 1179-1191.
</p>
</li>
<li><p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code>gl.run.faststructure</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
t1 &lt;- gl.filter.callrate(platypus.gl, threshold = 1)
res &lt;- gl.run.faststructure(t1,
  exec = "./fastStructure", k.range = 2:3,
  num.k.rep = 2, output = paste0(getwd(), "/res_str")
)
qmat &lt;- gl.plot.faststructure(res, k.range = 2:3)
gl.map.structure(qmat, K = 2, t1, scalex = 1, scaley = 0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.plot.structure'>Plots STRUCTURE analysis results (Q-matrix)</h2><span id='topic+gl.plot.structure'></span>

<h3>Description</h3>

<p>This function takes a structure run object (output from
<code><a href="#topic+gl.run.structure">gl.run.structure</a></code>) and plots the typical structure bar
plot that visualize the q matrix of a structure run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.structure(
  sr,
  K = NULL,
  met_clumpp = "greedyLargeK",
  iter_clumpp = 100,
  clumpak = TRUE,
  plot_theme = NULL,
  color_clusters = NULL,
  ind_name = TRUE,
  border_ind = 0.15,
  plot.out = TRUE,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.structure_+3A_sr">sr</code></td>
<td>
<p>Structure run object from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_k">K</code></td>
<td>
<p>The number for K of the q matrix that should be plotted. Needs to
be within you simulated range of K's in your sr structure run object. If
NULL, all the K's are plotted [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_met_clumpp">met_clumpp</code></td>
<td>
<p>The algorithm to use to infer the correct permutations.
One of 'greedy' or 'greedyLargeK' or 'stephens' [default &quot;greedyLargeK&quot;].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_iter_clumpp">iter_clumpp</code></td>
<td>
<p>The number of iterations to use if running either 'greedy'
'greedyLargeK' [default 100].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_clumpak">clumpak</code></td>
<td>
<p>Whether use the Clumpak method (see details) [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_color_clusters">color_clusters</code></td>
<td>
<p>A color palette for clusters (K) or a list with
as many colors as there are clusters (K) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_ind_name">ind_name</code></td>
<td>
<p>Whether to plot individual names [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_border_ind">border_ind</code></td>
<td>
<p>The width of the border line between individuals
[default 0.25].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude
extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default
NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function outputs a barplot which is the typical output of
structure. For a Evanno plot use gl.evanno.
</p>
<p>This function is based on the methods of CLUMPP and Clumpak as implemented
in the R package starmie (https://github.com/sa-lee/starmie).
</p>
<p>The Clumpak method identifies sets of highly similar runs among
all the replicates of the same K. The method then separates the distinct
groups of runs representing distinct modes in the space of possible solutions.
</p>
<p>The CLUMPP method permutes the clusters output by independent runs of
clustering programs such as structure, so that they match up as closely as
possible.
</p>
<p>This function averages the replicates within each mode identified by the
Clumpak method.
</p>
<p>Plots and table are saved to the working directory specified in plot.dir (tempdir )
if plot.file is set.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>List of Q-matrices
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Kopelman, Naama M., et al. &quot;Clumpak: a program for identifying
clustering modes and packaging population structure inferences across K.&quot;
Molecular ecology resources 15.5 (2015): 1179-1191.
</p>
</li>
<li><p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code>gl.run.structure</code>, <code>gl.plot.structure</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># examples need structure to be installed on the system (see above)
## Not run: 
bc &lt;- bandicoot.gl[,1:100]
sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure')
ev &lt;- gl.evanno(sr)
ev
qmat &lt;- gl.plot.structure(sr, K=3)
head(qmat)
gl.map.structure(qmat, K=3, bc, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.run.faststructure'>Runs a faststructure analysis using a genlight object</h2><span id='topic+gl.run.faststructure'></span>

<h3>Description</h3>

<p>This function takes a genlight object and runs a faststructure analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.run.faststructure(
  x,
  k.range,
  num.k.rep,
  exec = "./fastStructure",
  output = getwd(),
  tol = 1e-05,
  prior = "simple",
  cv = 0,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.run.faststructure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_k.range">k.range</code></td>
<td>
<p>Range of the number of populations [required].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_num.k.rep">num.k.rep</code></td>
<td>
<p>Number of replicates [required].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_exec">exec</code></td>
<td>
<p>Full path and name+extension where the fastStructure executable
is located [default working directory &quot;./fastStructure&quot;].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_output">output</code></td>
<td>
<p>Path to output file [default getwd()].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_tol">tol</code></td>
<td>
<p>Convergence criterion [default 10e-6].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_prior">prior</code></td>
<td>
<p>Choice of prior: simple or logistic [default &quot;simple&quot;].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_cv">cv</code></td>
<td>
<p>Number of test sets for cross-validation, 0 implies no CV step
[default 0].</p>
</td></tr>
<tr><td><code id="gl.run.faststructure_+3A_seed">seed</code></td>
<td>
<p>Seed for random number generator [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Download faststructure binary for your system from here (only runs on Mac or
Linux):
</p>
<p>https://github.com/StuntsPT/Structure_threader/tree/master/structure_threader/bins
</p>
<p>Move faststructure file to working directory. Make file executable using
terminal app.
</p>
<p><code>system(paste0("chmod u+x ",getwd(), "/faststructure"))</code>
</p>
<p>Download plink binary for your system from here:
</p>
<p>https://www.cog-genomics.org/plink/
</p>
<p>Move plink file to working directory. Make file executable using
terminal app.
</p>
<p><code>system(paste0("chmod u+x ",getwd(), "/plink"))</code>
</p>
<p>To install fastStructure dependencies follow these directions:
https://github.com/rajanil/fastStructure
</p>
<p>fastStructure performs inference for the simplest, independent-loci,
admixture model, with two choices of priors that can be specified using
the &ndash;prior parameter. Thus, unlike Structure, fastStructure does not require
the mainparams and extraparam files. The inference algorithm used by
fastStructure is fundamentally different from that of Structure and
requires the setting of far fewer options.
</p>
<p>To identify the number of populations that best approximates the marginal
likelihood of the data, the marginal likelihood is extracted from each run
of K, averaged across replications and plotted.
</p>


<h3>Value</h3>

<p>A list in which each list entry is a single faststructure run output
(there are k.range * num.k.rep number of runs).
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Raj, A., Stephens, M., &amp; Pritchard, J. K. (2014). fastSTRUCTURE:
variational inference of population structure in large SNP data sets.
Genetics, 197(2), 573-589.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Please note: faststructure needs to be installed
# Please note: faststructure is not available for windows
t1 &lt;- gl.filter.callrate(platypus.gl, threshold = 1)
res &lt;- gl.run.faststructure(t1,
  exec = "./fastStructure", k.range = 2:3,
  num.k.rep = 2, output = paste0(getwd(), "/res_str")
)
qmat &lt;- gl.plot.faststructure(res, k.range = 2:3)
gl.map.structure(qmat, K = 2, t1, scalex = 1, scaley = 0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.run.structure'>Runs a STRUCTURE analysis using a genlight object</h2><span id='topic+gl.run.structure'></span>

<h3>Description</h3>

<p>This function takes a genlight object and runs a STRUCTURE analysis based on
functions from <code>strataG</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.run.structure(
  x,
  ...,
  exec = ".",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot.dir = NULL,
  plot.file = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.run.structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_...">...</code></td>
<td>
<p>Parameters to specify the STRUCTURE run (check <code>structureRun</code>
within strataG.
for more details). Parameters are passed to the <code>structureRun</code> function.
For example you need to set the k.range and the type of model you would like
to run (noadmix, locprior) etc. If those parameter names do not tell you
anything, please make sure you familiarize with the STRUCTURE program
(Pritchard 2000).</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_exec">exec</code></td>
<td>
<p>Full path and name+extension where the structure executable is
located. E.g. <code>'c:/structure/structure.exe'</code> under Windows. For Mac and
Linux it might be something like <code>'./structure/structure'</code> if the
executable is in a subfolder 'structure' in your home directory
[default working directory &quot;.&quot;].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot.out">plot.out</code></td>
<td>
<p>Create an Evanno plot once finished. Be aware k.range needs
to be at least three different k steps [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory to save the plot RDS files [default as specified 
by the global working directory or tempdir()]</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_verbose">verbose</code></td>
<td>
<p>Set verbosity for this function (though structure output
cannot be switched off currently) [default NULL]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is basically a convenient wrapper around the beautiful
strataG function <code>structureRun</code> (Archer et al. 2016). For a detailed
description please refer to this package (see references below).
To make use of this function you need to download STRUCTURE for you system
(<b>non GUI version</b>) from here
<a href="https://web.stanford.edu/group/pritchardlab/structure_software/release_versions/v2.3.4/html/structure.html">STRUCTURE</a>.
</p>
<p><b>Format note</b>
</p>
<p>For this function to work, make sure that individual and population names
have no spaces. To substitute spaces by underscores you could use the R
function <code>gsub</code> as below.
</p>
<p><code>
popNames(gl) &lt;- gsub(" ","_",popNames(gl));
</code>
<code>
indNames(gl) &lt;- gsub(" ","_",indNames(gl))
</code>
</p>
<p>It's also worth noting that Structure truncates individual names at 11
characters. The function will fail if the names of individuals are not unique
after truncation. To avoid this possible problem, a number sequence, as
shown in the code below, might be used instead of individual names.
<code>
indNames(gl) &lt;- as.character(1:length(indNames(gl)))
</code>
</p>


<h3>Value</h3>

<p>An sr object (structure.result list output). Each list entry is a
single structurerun output (there are k.range * num.k.rep number of runs).
For example the summary output of the first run can be accessed via
<code>sr[[1]]$summary</code> or the q-matrix of the third run via
<code>sr[[3]]$q.mat</code>. To conveniently summarise the outputs across runs
(clumpp) you need to run gl.plot.structure on the returned sr object. For
Evanno plots run gl.evanno on your sr object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># examples need structure to be installed on the system (see above)
## Not run: 
bc &lt;- bandicoot.gl[,1:100]
sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3,
exec = './structure.exe')
ev &lt;- gl.evanno(sr)
ev
qmat &lt;- gl.plot.structure(sr, K=3)
head(qmat)
gl.map.structure(qmat, bc, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.sfs'>Creates a site frequency spectrum based on a dartR or genlight object</h2><span id='topic+gl.sfs'></span>

<h3>Description</h3>

<p>Creates a site frequency spectrum based on a dartR or genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sfs(
  x,
  minbinsize = 0,
  folded = TRUE,
  singlepop = FALSE,
  plot.out = TRUE,
  plot.file = NULL,
  plot.dir = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sfs_+3A_x">x</code></td>
<td>
<p>dartR/genlight object</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_minbinsize">minbinsize</code></td>
<td>
<p>remove bins from the left of the sfs. For example to remove
singletons (alleles only occurring once among all individuals) set
minbinsize to 2. If set to zero, also monomorphic (d0) loci are returned.</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_folded">folded</code></td>
<td>
<p>if set to TRUE (default) a folded sfs (minor allele frequency
sfs) is returned. If set to FALSE then an unfolded (derived allele frequency
sfs) is returned. It is assumed that 0 is homozygote for the reference and
2 is homozygote for the derived allele. So you need to make sure your
coding is correct.</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_singlepop">singlepop</code></td>
<td>
<p>switch to force to create a one-dimensional sfs, even
though the genlight/dartR object contains more than one population</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_plot.file">plot.file</code></td>
<td>
<p>Name for the RDS binary file to save (base name only, exclude extension) [default NULL]</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_plot.dir">plot.dir</code></td>
<td>
<p>Directory in which to save files [default = working directory]</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a site frequency spectrum, either a one dimensional vector
(only a single population in the dartR/genlight object or singlepop=TRUE) or
an n-dimensional array (n is the number of populations in the genlight/dartR
object). If the dartR/genlight object consists of several populations the
multidimensional site frequency spectrum for each population is returned
[=a multidimensional site frequency spectrum]. Be aware the
multidimensional spectrum works only for a limited number of population
and individuals [if too high the table command used internally will
through an error as the number of populations and individuals (and
therefore dimensions) are too large]. To get a single sfs for a
genlight/dartR object with multiple populations, you need to set
singlepop to TRUE. The returned sfs can be used to analyse demographics,
e.g. using fastsimcoal2.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &amp; Carlo Pacioni (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Excoffier L., Dupanloup I., Huerta-Sánchez E., Sousa V. C. and
Foll M. (2013) Robust demographic inference from genomic and SNP data. PLoS
genetics 9(10)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.sfs(bandicoot.gl, singlepop = TRUE)
gl.sfs(possums.gl[c(1:5, 31:33), ], minbinsize = 1)
</code></pre>

<hr>
<h2 id='utils.outflank'>OutFLANK:  An Fst outlier approach by Mike Whitlock and Katie Lotterhos,
University of British Columbia.</h2><span id='topic+utils.outflank'></span>

<h3>Description</h3>

<p>This function is the original implementation of Outflank by Whitlock and
Lotterhos. dartR simply provides a convenient wrapper around their functions
and an easier install being an r package (for information please refer to
their github repository)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank(
  FstDataFrame,
  LeftTrimFraction = 0.05,
  RightTrimFraction = 0.05,
  Hmin = 0.1,
  NumberOfSamples,
  qthreshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank_+3A_fstdataframe">FstDataFrame</code></td>
<td>
<p>A data frame that includes a row for each locus, with
columns as follows:
</p>

<ul>
<li><p> $LocusName: a character string that uniquely names
each locus.
</p>
</li>
<li><p> $FST: Fst calculated for this locus. (Kept here to
report the unbased Fst of the results)
</p>
</li>
<li><p> $T1: The numerator of the estimator for Fst
(necessary, with $T2, to calculate mean Fst)
</p>
</li>
<li><p> $T2: The denominator of the estimator of Fst
</p>
</li>
<li><p> $FSTNoCorr: Fst calculated for this locus without
sample size correction. (Used to find outliers)
</p>
</li>
<li><p> $T1NoCorr: The numerator of the estimator for Fst
without sample size correction (necessary, with $T2, to
calculate mean Fst)
</p>
</li>
<li><p> $T2NoCorr: The denominator of the estimator of Fst
without sample size correction
</p>
</li>
<li><p> $He: The heterozygosity of the locus (used to screen
out low heterozygosity loci that have a different distribution)
</p>
</li></ul>
</td></tr>
<tr><td><code id="utils.outflank_+3A_lefttrimfraction">LeftTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
lower end of the range of Fst before the likelihood funciton is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_righttrimfraction">RightTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
upper end of the range of Fst before the likelihood funciton is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including calculations
from a locus [default 0.1].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_numberofsamples">NumberOfSamples</code></td>
<td>
<p>The number of spatial locations included in the data
set.</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_qthreshold">qthreshold</code></td>
<td>
<p>The desired false discovery rate threshold for calculating
q-values [default 0.05].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method looks for Fst outliers from a list of Fst's for different loci.
It assumes that each locus has been genotyped in all populations with
approximately equal coverage.
</p>
<p>OutFLANK estimates the distribution of Fst based on a trimmed sample of Fst's.
It assumes that the majority of loci in the center of the distribution are
neutral and infers the shape of the distribution of neutral Fst using a
trimmed set of loci. Loci with the highest and lowest Fst's are trimmed from
the data set before this inference, and the distribution of Fst df/(mean Fst)
is assumed to'follow a chi-square distribution. Based on this inferred
distribution, each locus is given a q-value based on its quantile in the
inferred null'distribution.
</p>
<p>The main procedure is called OutFLANK &ndash; see comments in that function
immediately below for input and output formats. The other functions here are
necessary and must be uploaded, but are not necessarily needed by the user
directly.
</p>
<p>Steps:
</p>


<h3>Value</h3>

<p>The function returns a list with seven elements:
</p>

<ul>
<li><p> FSTbar: the mean FST inferred from loci not marked as outliers
</p>
</li>
<li><p> FSTNoCorrbar: the mean FST (not corrected for sample size -gives an
upwardly biased estimate of FST)
</p>
</li>
<li><p> dfInferred: the inferred number of degrees of freedom for the
chi-square distribution of neutral FST
</p>
</li>
<li><p> numberLowFstOutliers: Number of loci flagged as having a significantly
low FST (not reliable)
</p>
</li>
<li><p> numberHighFstOutliers: Number of loci identified as having
significantly high FST
</p>
</li>
<li><p> results: a data frame with a row for each locus. This data frame
includes all the original columns in the
data set, and six new ones:
</p>

<ul>
<li><p> $indexOrder (the original order of the input data set),
</p>
</li>
<li><p> $GoodH (Boolean variable which is TRUE if the expected
heterozygosity is greater than the Hemin set by input),
</p>
</li>
<li><p> $OutlierFlag (TRUE if the method identifies the locus as
an outlier, FALSE otherwise), and
</p>
</li>
<li><p> $q (the q-value for the test of neutrality for the locus)
</p>
</li>
<li><p> $pvalues (the p-value for the test of neutrality for the
locus)
</p>
</li>
<li><p> $pvaluesRightTail the one-sided (right tail) p-value for
a locus
</p>
</li></ul>

</li></ul>



<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Whitlock &amp; Lotterhos
</p>

<hr>
<h2 id='utils.outflank.MakeDiploidFSTMat'>Creates OutFLANK input file from individual genotype info.</h2><span id='topic+utils.outflank.MakeDiploidFSTMat'></span>

<h3>Description</h3>

<p>Creates OutFLANK input file from individual genotype info.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank.MakeDiploidFSTMat(SNPmat, locusNames, popNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_snpmat">SNPmat</code></td>
<td>
<p>This is an array of genotypes with a row for each individual.
There should be a column for each SNP, with the number of copies of the focal
allele (0, 1, or 2) for that individual. If that individual is missing data
for that SNP, there should be a 9, instead.</p>
</td></tr>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_locusnames">locusNames</code></td>
<td>
<p>A list of names for each SNP locus. There should be the
same number of locus names as there are columns in SNPmat.</p>
</td></tr>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_popnames">popNames</code></td>
<td>
<p>A list of population names to give location for each
individual. Typically multiple individuals will have the same popName. The
list popNames should have the same length as the number of rows in SNPmat.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame in the form needed for the main OutFLANK
function.
</p>

<hr>
<h2 id='utils.outflank.plotter'>Plotting functions for Fst distributions after OutFLANK</h2><span id='topic+utils.outflank.plotter'></span>

<h3>Description</h3>

<p>This function takes the output of OutFLANK as
input with the OFoutput parameter.  It plots a histogram of the FST (by
default, the uncorrected FSTs used by OutFLANK) of loci and overlays the
inferred null histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank.plotter(
  OFoutput,
  withOutliers = TRUE,
  NoCorr = TRUE,
  Hmin = 0.1,
  binwidth = 0.005,
  Zoom = FALSE,
  RightZoomFraction = 0.05,
  titletext = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank.plotter_+3A_ofoutput">OFoutput</code></td>
<td>
<p>The output of the function OutFLANK()</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_withoutliers">withOutliers</code></td>
<td>
<p>Determines whether the loci marked as outliers (with
$OutlierFlag) are included in the histogram.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_nocorr">NoCorr</code></td>
<td>
<p>Plots the distribution of FSTNoCorr when TRUE. Recommended,
because this is the data used by OutFLANK to infer the distribution.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including  a locus in
the plot.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_binwidth">binwidth</code></td>
<td>
<p>The width of bins in the histogram.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_zoom">Zoom</code></td>
<td>
<p>If Zoom is set to TRUE, then the graph will zoom in on the right
tail of the distirbution (based on argument RightZoomFraction)</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_rightzoomfraction">RightZoomFraction</code></td>
<td>
<p>Used when Zoom = TRUE. Defines the proportion of the
distribution to plot.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_titletext">titletext</code></td>
<td>
<p>Allows a test string to be printed as a title on the graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>produces a histogram of the FST
</p>

<hr>
<h2 id='utils.structure.evanno'>Util function for evanno plots</h2><span id='topic+utils.structure.evanno'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN
(maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.evanno(sr, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.evanno_+3A_sr">sr</code></td>
<td>
<p>structure run object</p>
</td></tr>
<tr><td><code id="utils.structure.evanno_+3A_plot">plot</code></td>
<td>
<p>should the plots be returned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of dataframes (structure results) and a list of plots
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='utils.structure.genind2gtypes'>structure util functions</h2><span id='topic+utils.structure.genind2gtypes'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN (maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.genind2gtypes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.genind2gtypes_+3A_x">x</code></td>
<td>
<p>a genind object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gtypes object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='utils.structure.run'>Utility function to run Structure</h2><span id='topic+utils.structure.run'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN
(maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.run(
  g,
  k.range = NULL,
  num.k.rep = 1,
  label = NULL,
  delete.files = TRUE,
  exec = "structure",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.run_+3A_g">g</code></td>
<td>
<p>a gtypes object [see <code>strataG</code>].</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_k.range">k.range</code></td>
<td>
<p>vector of values to for <code>maxpop</code> in multiple runs. If set
to <code>NULL</code>, a single STRUCTURE run is conducted with <code>maxpops</code>
groups. If specified, do not also specify <code>maxpops</code>.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_num.k.rep">num.k.rep</code></td>
<td>
<p>number of replicates for each value in <code>k.range</code>.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_label">label</code></td>
<td>
<p>label to use for input and output files</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_delete.files">delete.files</code></td>
<td>
<p>logical. Delete all files when STRUCTURE is finished?</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_exec">exec</code></td>
<td>
<p>name of executable for STRUCTURE. Defaults to &quot;structure&quot;.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code>structureWrite</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

 <dl>
<dt><code>structureRun</code></dt><dd><p>a list where each element is a
list with results from <code>structureRead</code> and a vector of the filenames
used</p>
</dd> <dt><code>structureWrite</code></dt><dd><p>a vector of the filenames used by
STRUCTURE</p>
</dd> <dt><code>structureRead</code></dt><dd><p>a list containing: </p>

<dl>
<dt><code>summary</code></dt><dd><p>new locus name, which is a combination of loci in
group</p>
</dd> <dt><code>q.mat</code></dt><dd><p>data.frame of assignment probabilities for each
id</p>
</dd> <dt><code>prior.anc</code></dt><dd><p>list of prior ancestry estimates for each
individual where population priors were used</p>
</dd> <dt><code>files</code></dt><dd><p>vector of
input and output files used by STRUCTURE</p>
</dd> <dt><code>label</code></dt><dd><p>label for the
run</p>
</dd> </dl>
 </dd> </dl>



<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='zzz'>Setting up the package dartR.popgen</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting up dartR.popgen
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
