<!DOCTYPE html><html><head><title>Help for package dartR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dartR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[,dartR,ANY,ANY,ANY-method'><p>indexing dartR objects correctly...</p></a></li>
<li><a href='#bandicoot.gl'><p>A genlight object created via the read.dart functions</p></a></li>
<li><a href='#cbind.dartR'><p>adjust cbind for dartR</p></a></li>
<li><a href='#gi2gl'><p>Converts a genind object into a genlight object</p></a></li>
<li><a href='#gl.alf'><p>Calculates allele frequency of the first and second allele for each loci</p>
A very simple function to report allele frequencies</a></li>
<li><a href='#gl.amova'><p>Performs AMOVA using genlight data</p></a></li>
<li><a href='#gl.assign.grm'><p>Population assignment using grm</p></a></li>
<li><a href='#gl.assign.mahalanobis'><p>Assign an individual of unknown provenance to population based on</p>
Mahalanobis Distance</a></li>
<li><a href='#gl.assign.pa'><p>Eliminates populations as possible source populations for an</p>
individual of unknown provenance, using private alleles</a></li>
<li><a href='#gl.assign.pca'><p>Assign an individual of unknown provenance to population based on PCA</p></a></li>
<li><a href='#gl.basic.stats'><p>Calculates basic statistics for each loci (Hs, Ho, Fis etc.)</p></a></li>
<li><a href='#gl.blast'><p>Aligns nucleotides sequences against those present in a target database</p>
using blastn</a></li>
<li><a href='#gl.check.verbosity'><p>Checks the current global verbosity</p></a></li>
<li><a href='#gl.collapse'><p>Collapses a distance matrix by amalgamating populations with pairwise</p>
fixed difference count less that a threshold</a></li>
<li><a href='#gl.compliance.check'><p>Checks a genlight object to see if it complies with dartR</p>
expectations and amends it to comply if necessary</a></li>
<li><a href='#gl.costdistances'><p>Calculates cost distances for a given landscape (resistance matrix)</p></a></li>
<li><a href='#gl.define.pop'><p>Defines a new population in a genlight object for specified</p>
individuals</a></li>
<li><a href='#gl.diagnostics.hwe'><p>Provides descriptive stats and plots to diagnose potential problems</p>
with Hardy-Weinberg proportions</a></li>
<li><a href='#gl.diagnostics.sim'><p>Comparing simulations against theoretical expectations</p></a></li>
<li><a href='#gl.dist.ind'><p>Calculates a distance matrix for individuals defined in a genlight object</p></a></li>
<li><a href='#gl.dist.pop'><p>Calculates a distance matrix for populations with SNP genotypes in a</p>
genlight object</a></li>
<li><a href='#gl.drop.ind'><p>Removes specified individuals from a genlight object</p></a></li>
<li><a href='#gl.drop.loc'><p>Removes specified loci from a genlight object</p></a></li>
<li><a href='#gl.drop.pop'><p>Removes specified populations from a genlight object</p></a></li>
<li><a href='#gl.edit.recode.ind'><p>Creates or edits individual (=specimen) names, creates a recode_ind</p>
file and applies the changes to a genlight object</a></li>
<li><a href='#gl.edit.recode.pop'><p>Creates or edits a population re-assignment table</p></a></li>
<li><a href='#gl.evanno'><p>Creates an Evanno plot from a STRUCTURE run object</p></a></li>
<li><a href='#gl.fdsim'><p>Estimates the rate of false positives in a fixed difference analysis</p></a></li>
<li><a href='#gl.filter.allna'><p>Filters loci that are all NA across individuals and/or populations</p>
with all NA across loci</a></li>
<li><a href='#gl.filter.callrate'><p>Filters loci or specimens in a genlight {adegenet} object based on</p>
call rate</a></li>
<li><a href='#gl.filter.hamming'><p>Filters loci based on pairwise Hamming distance between sequence tags</p></a></li>
<li><a href='#gl.filter.heterozygosity'><p>Filters individuals with average heterozygosity greater than a</p>
specified upper threshold or less than a specified lower threshold</a></li>
<li><a href='#gl.filter.hwe'><p>Filters loci that show significant departure from Hardy-Weinberg</p>
Equilibrium</a></li>
<li><a href='#gl.filter.ld'><p>Filters loci based on linkage disequilibrium (LD)</p></a></li>
<li><a href='#gl.filter.locmetric'><p>Filters loci on the basis of numeric information stored in</p>
other$loc.metrics in a genlight {adegenet} object</a></li>
<li><a href='#gl.filter.maf'><p>Filters loci on the basis of minor allele frequency (MAF) in a</p>
genlight
adegenet object</a></li>
<li><a href='#gl.filter.monomorphs'><p>Filters monomorphic loci, including those with all NAs</p></a></li>
<li><a href='#gl.filter.overshoot'><p>Filters loci for which the SNP has been trimmed from the sequence tag</p>
along with the adaptor</a></li>
<li><a href='#gl.filter.pa'><p>Filters loci that contain private (and fixed alleles) between two populations</p></a></li>
<li><a href='#gl.filter.parent.offspring'><p>Filters putative parent offspring within a population</p></a></li>
<li><a href='#gl.filter.rdepth'><p>Filters loci based on counts of sequence tags scored at a locus (read</p>
depth)</a></li>
<li><a href='#gl.filter.reproducibility'><p>Filters loci in a genlight {adegenet} object based on average</p>
repeatability of alleles at a locus</a></li>
<li><a href='#gl.filter.secondaries'><p>Filters loci that represent secondary SNPs in a genlight object</p></a></li>
<li><a href='#gl.filter.sexlinked'><p>Filters loci that are sex linked</p></a></li>
<li><a href='#gl.filter.taglength'><p>Filters loci in a genlight {adegenet} object based on sequence tag</p>
length</a></li>
<li><a href='#gl.fixed.diff'><p>Generates a matrix of fixed differences and associated statistics for</p>
populations taken pairwise</a></li>
<li><a href='#gl.fst.pop'><p>Calculates a pairwise Fst values for populations in a genlight object</p></a></li>
<li><a href='#gl.genleastcost'><p>Performs least-cost path analysis based on a friction matrix</p></a></li>
<li><a href='#gl.grm'><p>Calculates an identity by descent matrix</p></a></li>
<li><a href='#gl.grm.network'><p>Represents a genomic relationship matrix (GRM) as a network</p></a></li>
<li><a href='#gl.He'><p>Estimates expected Heterozygosity</p></a></li>
<li><a href='#gl.Ho'><p>Estimates observed Heterozygosity</p></a></li>
<li><a href='#gl.hwe.pop'><p>Performs Hardy-Weinberg tests over loci and populations</p></a></li>
<li><a href='#gl.ibd'><p>Performs isolation by distance analysis</p></a></li>
<li><a href='#gl.impute'><p>Imputates missing data</p></a></li>
<li><a href='#gl.install.vanilla.dartR'><p>Installs all required packages for using all functions</p>
available in dartR</a></li>
<li><a href='#gl.join'><p>Combines two genlight objects</p></a></li>
<li><a href='#gl.keep.ind'><p>Removes all but the specified individuals from a genlight object</p></a></li>
<li><a href='#gl.keep.loc'><p>Removes all but the specified loci from a genlight object</p></a></li>
<li><a href='#gl.keep.pop'><p>Removes all but the specified populations from a genlight object</p></a></li>
<li><a href='#gl.ld.distance'><p>Plots linkage disequilibrium against distance by population</p>
disequilibrium patterns</a></li>
<li><a href='#gl.ld.haplotype'><p>Visualize patterns of linkage disequilibrium and identification of</p>
haplotypes</a></li>
<li><a href='#gl.LDNe'><p>Estimates effective population size using the Linkage Disequilibrium</p>
method based on NeEstimator (V2)</a></li>
<li><a href='#gl.list.reports'><p>Prints dartR reports saved in tempdir</p></a></li>
<li><a href='#gl.load'><p>Loads an object from compressed binary format produced by gl.save()</p></a></li>
<li><a href='#gl.make.recode.ind'><p>Creates a proforma recode_ind file for reassigning individual</p>
(=specimen) names</a></li>
<li><a href='#gl.make.recode.pop'><p>Creates a proforma recode_pop_table file for reassigning population</p>
names</a></li>
<li><a href='#gl.map.interactive'><p>Creates an interactive map (based on latlon) from a genlight object</p></a></li>
<li><a href='#gl.map.structure'><p>Maps a STRUCTURE plot using a genlight object</p></a></li>
<li><a href='#gl.merge.pop'><p>Merges two or more populations in a genlight object into one population</p></a></li>
<li><a href='#gl.nhybrids'><p>Creates an input file for the program NewHybrids and runs it if</p>
NewHybrids is installed</a></li>
<li><a href='#gl.outflank'><p>Identifies loci under selection per population using the outflank</p>
method of Whitlock and Lotterhos (2015)</a></li>
<li><a href='#gl.pcoa'><p>Ordination applied to genotypes in a genlight object (PCA), in an fd</p>
object, or to a distance matrix (PCoA)</a></li>
<li><a href='#gl.pcoa.plot'><p>Bivariate or trivariate plot of the results of an ordination generated</p>
using gl.pcoa()</a></li>
<li><a href='#gl.percent.freq'><p>Generates percentage allele frequencies by locus and population</p></a></li>
<li><a href='#gl.play.history'><p>Replays the history and applies it to a genlight object</p></a></li>
<li><a href='#gl.plot.heatmap'><p>Represents a distance matrix as a heatmap</p></a></li>
<li><a href='#gl.plot.network'><p>Represents a distance or dissimilarity matrix as a network</p></a></li>
<li><a href='#gl.plot.structure'><p>Plots STRUCTURE analysis results (Q-matrix)</p></a></li>
<li><a href='#gl.print.history'><p>Prints history of a genlight object</p></a></li>
<li><a href='#gl.print.reports'><p>Prints dartR reports saved in tempdir</p></a></li>
<li><a href='#gl.propShared'><p>Calculates a similarity (distance) matrix for individuals on the proportion of</p>
shared alleles</a></li>
<li><a href='#gl.random.snp'><p>Randomly changes the allocation of 0's and 2's in a genlight object</p></a></li>
<li><a href='#gl.read.csv'><p>Reads SNP data from a csv file into a genlight object</p></a></li>
<li><a href='#gl.read.dart'><p>Imports DArT data into dartR and converts it into a genlight object</p></a></li>
<li><a href='#gl.read.fasta'><p>Reads FASTA files and converts them to genlight object</p></a></li>
<li><a href='#gl.read.silicodart'><p>Imports presence/absence data from SilicoDArT to genlight {agegenet}</p>
format (ploidy=1)</a></li>
<li><a href='#gl.read.vcf'><p>Converts a vcf file into a genlight object</p></a></li>
<li><a href='#gl.reassign.pop'><p>Assigns an individual metric as pop in a genlight {adegenet} object</p></a></li>
<li><a href='#gl.recalc.metrics'><p>Recalculates locus metrics when individuals or populations are deleted from a</p>
genlight {adegenet} object</a></li>
<li><a href='#gl.recode.ind'><p>Recodes individual (=specimen = sample) labels in a genlight object</p></a></li>
<li><a href='#gl.recode.pop'><p>Recodes population assignments in a genlight object</p></a></li>
<li><a href='#gl.rename.pop'><p>Renames a population in a genlight object</p></a></li>
<li><a href='#gl.report.bases'><p>Reports summary of base pair frequencies</p></a></li>
<li><a href='#gl.report.callrate'><p>Reports summary of Call Rate for loci or individuals</p></a></li>
<li><a href='#gl.report.diversity'><p>Calculates diversity indexes for SNPs</p></a></li>
<li><a href='#gl.report.hamming'><p>Calculates the pairwise Hamming distance between DArT trimmed DNA</p>
sequences</a></li>
<li><a href='#gl.report.heterozygosity'><p>Reports observed, expected and unbiased heterozygosities and FIS</p>
(inbreeding coefficient) by population or by individual from SNP data</a></li>
<li><a href='#gl.report.hwe'><p>Reports departure from Hardy-Weinberg proportions</p></a></li>
<li><a href='#gl.report.ld'><p>Calculates pairwise population based Linkage Disequilibrium across all loci</p>
using the specified number of cores</a></li>
<li><a href='#gl.report.ld.map'><p>Calculates pairwise linkage disequilibrium by population</p></a></li>
<li><a href='#gl.report.locmetric'><p>Reports summary of the slot $other$loc.metrics</p></a></li>
<li><a href='#gl.report.maf'><p>Reports minor allele frequency (MAF) for each locus in a SNP dataset</p></a></li>
<li><a href='#gl.report.monomorphs'><p>Reports monomorphic loci</p></a></li>
<li><a href='#gl.report.overshoot'><p>Reports loci for which the SNP has been trimmed from the sequence tag</p>
along with the adaptor</a></li>
<li><a href='#gl.report.pa'><p>Reports private alleles (and fixed alleles) per pair of populations</p></a></li>
<li><a href='#gl.report.parent.offspring'><p>Identifies putative parent offspring within a population</p></a></li>
<li><a href='#gl.report.rdepth'><p>Reports summary of Read Depth for each locus</p></a></li>
<li><a href='#gl.report.reproducibility'><p>Reports summary of RepAvg (repeatability averaged over both alleles</p>
for
each locus) or reproducibility (repeatability of the scores for fragment
presence/absence)</a></li>
<li><a href='#gl.report.secondaries'><p>Reports loci containing secondary SNPs in sequence tags and calculates</p>
number of invariant sites</a></li>
<li><a href='#gl.report.sexlinked'><p>Identifies loci that are sex linked</p></a></li>
<li><a href='#gl.report.taglength'><p>Reports summary of sequence tag length across loci</p></a></li>
<li><a href='#gl.run.structure'><p>Runs a STRUCTURE analysis using a genlight object</p></a></li>
<li><a href='#gl.sample'><p>Samples individuals from populations</p></a></li>
<li><a href='#gl.save'><p>Saves an object in compressed binary format for later rapid retrieval</p></a></li>
<li><a href='#gl.select.colors'><p>Selects colors from one of several palettes and output as a vector</p></a></li>
<li><a href='#gl.select.shapes'><p>Selects shapes from the base R shape palette and outputs as a vector</p></a></li>
<li><a href='#gl.set.verbosity'><p>Sets the default verbosity level</p></a></li>
<li><a href='#gl.sfs'><p>Creates a site frequency spectrum based on a dartR or genlight object</p></a></li>
<li><a href='#gl.sim.create_dispersal'><p>Creates a dispersal file as input for the function gl.sim.WF.run</p></a></li>
<li><a href='#gl.sim.emigration'><p>Simulates emigration between populations</p></a></li>
<li><a href='#gl.sim.ind'><p>Simulates individuals based on the allele frequencies provided via a genlight</p>
object.</a></li>
<li><a href='#gl.sim.mutate'><p>Simulates mutations within a genlight object</p></a></li>
<li><a href='#gl.sim.offspring'><p>Simulates a specified number of offspring based on alleles provided by</p>
potential father(s) and mother(s)</a></li>
<li><a href='#gl.sim.WF.run'><p>Runs Wright-Fisher simulations</p></a></li>
<li><a href='#gl.sim.WF.table'><p>Creates the reference table for running gl.sim.WF.run</p></a></li>
<li><a href='#gl.smearplot'><p>Smear plot of SNP or presence/absence (SilicoDArT) data</p></a></li>
<li><a href='#gl.sort'><p>re-sorts genlight objects</p></a></li>
<li><a href='#gl.spatial.autoCorr'><p>Spatial autocorrelation following Smouse and Peakall 1999</p></a></li>
<li><a href='#gl.subsample.loci'><p>Subsamples n loci from a genlight object and return it as a genlight object</p></a></li>
<li><a href='#gl.test.heterozygosity'><p>Tests the difference in heterozygosity between populations taken</p>
pairwise</a></li>
<li><a href='#gl.tree.nj'><p>Outputs an nj tree to summarize genetic similarity among populations</p></a></li>
<li><a href='#gl.write.csv'><p>Writes out data from a genlight object to csv file</p></a></li>
<li><a href='#gl2bayescan'><p>Converts a genlight object into a format suitable for input to Bayescan</p></a></li>
<li><a href='#gl2bpp'><p>Converts a genlight object into a format suitable for input to the BPP</p>
program</a></li>
<li><a href='#gl2demerelate'><p>Creates a dataframe suitable for input to package {Demerelate} from a</p>
genlight {adegenet} object</a></li>
<li><a href='#gl2eigenstrat'><p>Converts a genlight object into eigenstrat format</p></a></li>
<li><a href='#gl2fasta'><p>Concatenates DArT trimmed sequences and outputs a FASTA file</p></a></li>
<li><a href='#gl2faststructure'><p>Converts a genlight object into faststructure format (to run faststructure</p>
elsewhere)</a></li>
<li><a href='#gl2gds'><p>Converts a genlight object into gds format</p></a></li>
<li><a href='#gl2genalex'><p>Converts a genlight object into a format suitable for input to genalex</p></a></li>
<li><a href='#gl2genepop'><p>Converts a genlight object into genepop format (and file)</p></a></li>
<li><a href='#gl2geno'><p>Converts a genlight object to geno format from package LEA</p></a></li>
<li><a href='#gl2gi'><p>Converts a genlight object to genind object</p></a></li>
<li><a href='#gl2hiphop'><p>Converts a genlight objects into hiphop format</p></a></li>
<li><a href='#gl2phylip'><p>Creates a Phylip input distance matrix from a genlight (SNP) {adegenet}</p>
object</a></li>
<li><a href='#gl2plink'><p>Converts a genlight object into PLINK format</p></a></li>
<li><a href='#gl2related'><p>Converts a genlight object to format suitable to be run with Coancestry</p></a></li>
<li><a href='#gl2sa'><p>Converts genlight objects to the format used in the SNPassoc package</p></a></li>
<li><a href='#gl2sfs'><p>Converts a genlight object into a sfs input file</p></a></li>
<li><a href='#gl2shp'><p>Converts a genlight object to ESRI shapefiles or kml files</p></a></li>
<li><a href='#gl2snapp'><p>Converts a genlight object to nexus format suitable for phylogenetic analysis</p>
by SNAPP (via BEAUti)</a></li>
<li><a href='#gl2structure'><p>Converts a genlight object to STRUCTURE formatted files</p></a></li>
<li><a href='#gl2svdquartets'><p>Converts a genlight object to nexus format PAUP SVDquartets</p></a></li>
<li><a href='#gl2treemix'><p>Converts a genlight object to a treemix input file</p></a></li>
<li><a href='#gl2vcf'><p>Converts a genlight object into vcf format</p></a></li>
<li><a href='#interactive_reference'><p>Shiny app for the input of the reference table for the simulations</p></a></li>
<li><a href='#interactive_sim_run'><p>Shiny app for the input of the simulations variables</p></a></li>
<li><a href='#is.fixed'><p>Tests if two populations are fixed at a given locus</p></a></li>
<li><a href='#platy'><p>Example data set as text file to be imported into a genlight object</p></a></li>
<li><a href='#possums.gl'><p>A simulated genlight object created to run a landscape genetic example</p></a></li>
<li><a href='#rbind.dartR'><p>adjust rbind for dartR</p></a></li>
<li><a href='#testset_metadata'><p>Metadata file. Can be integrated via the dart2genlight function.</p></a></li>
<li><a href='#testset_pop_recode'><p>Recode file to be used with the function.</p></a></li>
<li><a href='#testset_SNPs_2Row'><p>Testfile in DArT format (as provided by DArT)</p></a></li>
<li><a href='#testset.gl'><p>A genlight object created via the gl.read.dart function</p></a></li>
<li><a href='#testset.gs'><p>A genlight object created via the gl.read.silicodart function</p></a></li>
<li><a href='#theme_dartR'><p>dartR theme</p></a></li>
<li><a href='#utils.assignment'><p>Population assignment probabilities</p></a></li>
<li><a href='#utils.assignment_2'><p>Population assignment probabilities</p></a></li>
<li><a href='#utils.assignment_3'><p>Population assignment probabilities</p></a></li>
<li><a href='#utils.assignment_4'><p>Population assignment probabilities</p></a></li>
<li><a href='#utils.basic.stats'><p>Calculates mean observed heterozygosity, mean expected heterozygosity and Fis</p>
per locus, per population and various population differentiation measures</a></li>
<li><a href='#utils.check.datatype'><p>Utility function to check the class of an object passed to a function</p></a></li>
<li><a href='#utils.dart2genlight'><p>Converts DarT to genlight</p></a></li>
<li><a href='#utils.dist.binary'><p>Calculates a distance matrix for individuals defined in a dartR</p>
genlight object using binary P/A data (SilicoDArT)</a></li>
<li><a href='#utils.dist.ind.snp'><p>Calculates a distance matrix for individuals defined in a dartR</p>
genlight object using SNP data (DArTseq)</a></li>
<li><a href='#utils.flag.start'><p>A utility script to flag the start of a script</p></a></li>
<li><a href='#utils.hamming'><p>Calculates the Hamming distance between two DArT trimmed DNA sequences</p></a></li>
<li><a href='#utils.het.pop'><p>Calculates expected mean expected heterozygosity per population</p></a></li>
<li><a href='#utils.jackknife'><p>Conducts jackknife resampling using a genlight object</p></a></li>
<li><a href='#utils.n.var.invariant'><p>A utility script to calculate the number of variant and invariant sites by</p>
locus</a></li>
<li><a href='#utils.outflank'><p>OutFLANK:  An Fst outlier approach by Mike Whitlock and Katie Lotterhos,</p>
University of British Columbia.</a></li>
<li><a href='#utils.outflank.MakeDiploidFSTMat'><p>Creates OutFLANK input file from individual genotype info.</p></a></li>
<li><a href='#utils.outflank.plotter'><p>Plotting functions for Fst distributions after OutFLANK</p></a></li>
<li><a href='#utils.read.dart'><p>Imports DarT data to R</p></a></li>
<li><a href='#utils.recalc.avgpic'><p>A utility script to recalculate the OneRatioRef, OneRatioSnp, PICRef, PICSnp,</p>
and AvgPIC by locus after some individuals or populations have been deleted.</a></li>
<li><a href='#utils.recalc.callrate'><p>A utility script to recalculate the callrate by locus after some populations</p>
have been deleted</a></li>
<li><a href='#utils.recalc.freqhets'><p>A utility script to recalculate the frequency of the heterozygous SNPs by</p>
locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.freqhomref'><p>A utility script to recalculate the frequency of the homozygous reference</p>
SNP by locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.freqhomsnp'><p>A utility script to recalculate the frequency of the homozygous alternate</p>
SNP by locus after some populations have been deleted</a></li>
<li><a href='#utils.recalc.maf'><p>A utility script to recalculate the minor allele frequency by locus,</p>
typically after some populations have been deleted</a></li>
<li><a href='#utils.reset.flags'><p>A utility script to reset to FALSE (or TRUE) the locus metric flags after</p>
some individuals or populations have been deleted.</a></li>
<li><a href='#utils.spautocor'><p>Spatial autocorrelation coefficient calculations</p></a></li>
<li><a href='#utils.structure.evanno'><p>Util function for evanno plots</p></a></li>
<li><a href='#utils.structure.genind2gtypes'><p>structure util functions</p></a></li>
<li><a href='#utils.structure.run'><p>Utility function to run Structure</p></a></li>
<li><a href='#zzz'><p>Setting up the package</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Importing and Analysing 'SNP' and 'Silicodart' Data Generated by
Genome-Wide Restriction Fragment Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.9.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-07</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions are provided that facilitate the import and analysis of
    'SNP' (single nucleotide polymorphism) and 'silicodart' (presence/absence) data. 
    The main focus is on data generated by 'DarT' (Diversity Arrays Technology), 
    however, data from other sequencing platforms can be used once 'SNP' or related 
    fragment presence/absence data from any source is imported. Genetic datasets 
    are stored in a derived 'genlight' format (package 'adegenet'), that allows for 
    a very compact storage of data and metadata. Functions are available for 
    importing and exporting of 'SNP' and 'silicodart' data, for reporting on and 
    filtering on various criteria (e.g. 'CallRate', heterozygosity, reproducibility, 
    maximum allele frequency). Additional functions are available for visualization 
    (e.g. Principle Coordinate Analysis) and creating a spatial representation 
    using maps. 'dartR' supports also the analysis of 3rd party software package 
    such as 'newhybrid', 'structure', 'NeEstimator' and 'blast'. Since version 
    2.0.3 we also implemented simulation functions, that allow to forward simulate
    'SNP' dynamics under different population and evolutionary dynamics. 
    Comprehensive tutorials and support can be found at our 'github' repository: 
    github.com/green-striped-gecko/dartR/. If you want to cite 'dartR', you find 
    the information by typing citation('dartR') in the console.</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), adegenet (&ge; 2.0.0), ggplot2, dplyr, dartR.data</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape,crayon,data.table,fields,foreach,gridExtra,MASS,methods,patchwork,plyr,PopGenReport,raster,reshape2,shiny,SNPRelate,sp
(&ge; 1.6.1),StAMPP,stats,stringr,tidyr,utils,vegan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>boot, devtools, directlabels, dismo, doParallel, expm,
gdistance, ggtern, gganimate, ggrepel, grid, gtable, ggthemes,
gplots, HardyWeinberg, hierfstat, igraph, iterpc, knitr,
label.switching, lattice, leaflet, leaflet.minicharts,
markdown, mmod, networkD3, parallel, pegas, pheatmap, plotly,
poppr, proxy, purrr, qvalue, RColorBrewer, Rcpp, rgl,
rmarkdown, rrBLUP, scales, seqinr, shinyBS, shinyjs,
shinythemes, shinyWidgets, SIBER, snpStats, stringi, terra,
tibble, vcfR, zoo, viridis, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-07 08:38:16 UTC; s425824</td>
</tr>
<tr>
<td>Author:</td>
<td>Bernd Gruber [aut, cre],
  Arthur Georges [aut],
  Jose L. Mijangos [aut],
  Carlo Pacioni [aut],
  Peter J. Unmack [ctb],
  Oliver Berry [ctb],
  Lindsay V. Clark [ctb],
  Floriaan Devloo-Delva [ctb],
  Eric Archer [ctb]</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://green-striped-gecko.github.io/dartR/">https://green-striped-gecko.github.io/dartR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/dartr?pli=1">https://groups.google.com/g/dartr?pli=1</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bernd Gruber &lt;bernd.gruber@canberra.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-07 10:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B+2CdartR+2CANY+2CANY+2CANY-method'>indexing dartR objects correctly...</h2><span id='topic++5B+2CdartR+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>indexing dartR objects correctly...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'dartR,ANY,ANY,ANY'
x[i, j, ..., pop = NULL, treatOther = TRUE, quiet = TRUE, drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>dartR object</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>index for individuals</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_j">j</code></td>
<td>
<p>index for loci</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_...">...</code></td>
<td>
<p>other parameters</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_pop">pop</code></td>
<td>
<p>list of populations to be kept</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_treatother">treatOther</code></td>
<td>
<p>elements in other (and ind.metrics &amp; loci.metrics) as 
indexed as well. default: TRUE</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_quiet">quiet</code></td>
<td>
<p>warnings are suppressed. default: TRUE</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CdartR+2B2CANY+2B2CANY+2B2CANY-method_+3A_drop">drop</code></td>
<td>
<p>reduced to a vector if a single individual/loci is selected. 
default: FALSE [should never set to TRUE]</p>
</td></tr>
</table>

<hr>
<h2 id='bandicoot.gl'>A genlight object created via the read.dart functions</h2><span id='topic+bandicoot.gl'></span>

<h3>Description</h3>

<p>This a test data set to test the validity of functions within dartR and is 
based on a DArT SNP data set of simulated bandicoots across Australia. It 
contains 96 individuals and 1000 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bandicoot.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='cbind.dartR'>adjust cbind for dartR</h2><span id='topic+cbind.dartR'></span>

<h3>Description</h3>

<p>cbind is a bit lazy and does not take care for the metadata (so data in the
other slot is lost). You can get most of the loci metadata back using
gl.compliance.check.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dartR'
cbind(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbind.dartR_+3A_...">...</code></td>
<td>
<p>list of dartR objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t1 &lt;- platypus.gl
class(t1) &lt;- "dartR"
t2 &lt;- cbind(t1[,1:10],t1[,11:20])
</code></pre>

<hr>
<h2 id='gi2gl'>Converts a genind object into a genlight object</h2><span id='topic+gi2gl'></span>

<h3>Description</h3>

<p>Converts a genind object into a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gi2gl(gi, parallel = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gi2gl_+3A_gi">gi</code></td>
<td>
<p>A genind object [required].</p>
</td></tr>
<tr><td><code id="gi2gl_+3A_parallel">parallel</code></td>
<td>
<p>Switch to deactivate parallel version. It might not be worth
to run it parallel most of the times [default FALSE].</p>
</td></tr>
<tr><td><code id="gi2gl_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Be aware due to ambiguity which one is the reference allele a combination of
gi2gl(gl2gi(gl)) does not return an identical object (but in terms of
analysis this conversions are equivalent)
</p>


<h3>Value</h3>

<p>A genlight object, with all slots filled.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.alf'>Calculates allele frequency of the first and second allele for each loci
A very simple function to report allele frequencies</h2><span id='topic+gl.alf'></span>

<h3>Description</h3>

<p>Calculates allele frequency of the first and second allele for each loci
A very simple function to report allele frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.alf(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.alf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple data.frame with alf1, alf2.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#for the first 10 loci only
gl.alf(possums.gl[,1:10])
barplot(t(as.matrix(gl.alf(possums.gl[,1:10]))))
</code></pre>

<hr>
<h2 id='gl.amova'>Performs AMOVA using genlight data</h2><span id='topic+gl.amova'></span>

<h3>Description</h3>

<p>This script performs an AMOVA based on the genetic distance matrix from
stamppNeisD() [package StAMPP] using the amova() function from the package
PEGAS for exploring within and between population variation. For detailed
information use their help pages: ?pegas::amova, ?StAMPP::stamppAmova. Be
aware due to a conflict of the amova functions from various packages I had
to 'hack' StAMPP::stamppAmova to avoid a namespace conflict.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.amova(x, distance = NULL, permutations = 100, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.amova_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes, with
population information [required].</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_distance">distance</code></td>
<td>
<p>Distance matrix between individuals (if not provided NeisD
from StAMPP::stamppNeisD is calculated) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_permutations">permutations</code></td>
<td>
<p>Number of permutations to perform for hypothesis
testing [default 100]. Please note should be set to 1000 for analysis.</p>
</td></tr>
<tr><td><code id="gl.amova_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'amova' which is a list with a table of sums of
square deviations (SSD), mean square deviations (MSD), and the number of
degrees of freedom, and a vector of variance components.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#permutations should be higher, here set to 1 because of speed
out &lt;- gl.amova(bandicoot.gl, permutations=1)

</code></pre>

<hr>
<h2 id='gl.assign.grm'>Population assignment using grm</h2><span id='topic+gl.assign.grm'></span>

<h3>Description</h3>

<p>This function takes one individual and estimates
their probability of coming from individual populations
from multilocus genotype frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.assign.grm(x, unknown, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.assign.grm_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.assign.grm_+3A_unknown">unknown</code></td>
<td>
<p>Name of the individual to be assigned to a population [required].</p>
</td></tr>
<tr><td><code id="gl.assign.grm_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a re-implementation of the function multilocus_assignment
from package gstudio.
Description of the method used in this function can be found at:
https://dyerlab.github.io/applied_population_genetics/population-assignment.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> consisting of assignment probabilities for each
population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
if ((requireNamespace("rrBLUP", quietly = TRUE)) &amp;(requireNamespace("gplots", quietly = TRUE)) ) {
res &lt;- gl.assign.grm(platypus.gl,unknown="T27")
}
</code></pre>

<hr>
<h2 id='gl.assign.mahalanobis'>Assign an individual of unknown provenance to population based on 
Mahalanobis Distance</h2><span id='topic+gl.assign.mahalanobis'></span>

<h3>Description</h3>

<p>This script assigns an individual of unknown provenance to one or more target
populations based on the unknown individual's proximity to population 
centroids; proximity is estimated using Mahalanobis Distance. 
</p>
<p>The following process is followed:
</p>

<ol>
<li><p> An ordination is undertaken on the populations to again yield a
series of orthogonal (independent) axes.
</p>
</li>
<li><p> A workable subset of dimensions is chosen, that specified, or
equal to the number of dimensions with substantive eigenvalues, whichever is
the smaller.
</p>
</li>
<li><p> The Mahalobalis Distance is calculated for the unknown against each
population and probability of membership of each population is calculated.
The assignment probabilities are listed in support of a decision.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>gl.assign.mahalanobis(
  x,
  dim.limit = 2,
  plevel = 0.999,
  plot.out = TRUE,
  unknown,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.assign.mahalanobis_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.assign.mahalanobis_+3A_dim.limit">dim.limit</code></td>
<td>
<p>Maximum number of dimensions to consider for the
confidence ellipses [default 2]</p>
</td></tr>
<tr><td><code id="gl.assign.mahalanobis_+3A_plevel">plevel</code></td>
<td>
<p>Probability level for bounding ellipses
[default 0.999].</p>
</td></tr>
<tr><td><code id="gl.assign.mahalanobis_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, produces a plot showing the position of the 
unknown in relation to putative source populations [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.assign.mahalanobis_+3A_unknown">unknown</code></td>
<td>
<p>Identity label of the focal individual whose provenance is
unknown [required].</p>
</td></tr>
<tr><td><code id="gl.assign.mahalanobis_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are three considerations to assignment. First, consider only those
populations for which the unknown has no private alleles. Private alleles are
an indication that the unknown does not belong to a target population
(provided that the sample size is adequate, say &gt;=10). This can be evaluated
with gl.assign.pa().
</p>
<p>A next step is to consider the PCoA plot for populations where no private
alleles have been detected. The position of the unknown in relation to the
confidence ellipses is plotted by this script as a basis for narrowing down
the list of putative source populations. This can be evaluated with 
gl.assign.pca().
</p>
<p>The third step (delivered by this script) is to consider the assignment 
probabilities based on the squared Generalised Linear Distance 
(Mahalanobis distance) of the unknown from the centroid for each population, 
then to consider the probability associated with its quantile using the 
Chisquare approximation. In effect, this index takes into account position 
of the unknown in relation to the confidence envelope in all selected 
dimensions of the ordination. The larger the assignment probability, 
the greater the confidence in the assignment. 
</p>
<p>If dim.limit is set to 2, to correspond with the dimensions used in
gl.assign.pa(), then the output provides a ranking of the final set
of putative source populations.
</p>
<p>If dim.limit is set to be &gt; 2, then this script provides a basis for
further narrowing the set of putative populations.If the unknown individual
is an extreme outlier, say at less than 0.001 probability of population 
membership (0.999 confidence envelope), then the associated population 
can be eliminated from further consideration.
</p>
<p>Warning: gl.assign.mahal() treats each specified dimension equally, without
regard to the percentage variation explained after ordination. If the 
unknown is an outlier in a lower dimension with an explanatory variance of,
say, 0.1
dimensions from the ordination.
</p>
<p>Each of these above approaches provides evidence, none are 100
They need to be interpreted cautiously.
</p>
<p>In deciding the assignment, the script considers an individual to be an
outlier with respect to a particular population at alpha = 0.001 as default
</p>


<h3>Value</h3>

<p>A data frame with the results of the assignment analysis.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash;
Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Test run with a focal individual from the Macleay River (EmmacMaclGeor) 
test &lt;- gl.assign.pa(testset.gl, unknown='UC_01044', nmin=10, threshold=1,
verbose=3) 
test_2  &lt;- gl.assign.pca(test, unknown='UC_01044', plevel=0.95, verbose=3)
df &lt;- gl.assign.mahalanobis(test_2, unknown='UC_01044', verbose=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.assign.pa'>Eliminates populations as possible source populations for an 
individual of unknown provenance, using private alleles</h2><span id='topic+gl.assign.pa'></span>

<h3>Description</h3>

<p>This script eliminates from consideration as putative source populations,
those populations for which the individual has too many private alleles. The
populations that remain are putative source populations, subject to further
consideration.
</p>
<p>The algorithm identifies those target populations for which the individual
has no private alleles or for which the number of private alleles does not
exceed a user specified threshold.
</p>
<p>An excessive count of private alleles is an indication that the unknown does
not belong to a target population (provided that the sample size is
adequate, say &gt;=10).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.assign.pa(
  x,
  unknown,
  nmin = 10,
  threshold = 0,
  n.best = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.assign.pa_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.assign.pa_+3A_unknown">unknown</code></td>
<td>
<p>SpecimenID label (indName) of the focal individual whose
provenance is unknown [required].</p>
</td></tr>
<tr><td><code id="gl.assign.pa_+3A_nmin">nmin</code></td>
<td>
<p>Minimum sample size for a target population to be included in the
analysis [default 10].</p>
</td></tr>
<tr><td><code id="gl.assign.pa_+3A_threshold">threshold</code></td>
<td>
<p>Populations to retain for consideration; those for which the
focal individual has less than or equal to threshold loci with private
alleles [default 0].</p>
</td></tr>
<tr><td><code id="gl.assign.pa_+3A_n.best">n.best</code></td>
<td>
<p>If given a value, dictates the best n=n.best populations to
retain for consideration (or more if their are ties) based on private alleles
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.assign.pa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object containing the focal individual (assigned to
population 'unknown') and populations for which the focal individual is not
distinctive (number of loci with private alleles less than or equal to the
threshold). If no such populations, the genlight object contains only data
for the unknown individual.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.assign.pca">gl.assign.pca</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Test run with a focal individual from the Macleay River (EmmacMaclGeor)
  test &lt;- gl.assign.pa(testset.gl, unknown='UC_00146', nmin=10, threshold=1,
  verbose=3)

</code></pre>

<hr>
<h2 id='gl.assign.pca'>Assign an individual of unknown provenance to population based on PCA</h2><span id='topic+gl.assign.pca'></span>

<h3>Description</h3>

<p>This script assigns an individual of unknown provenance to one or more target
populations based on its proximity to each population defined by a
confidence ellipse in ordinated space of two dimensions.
</p>
<p>The following process is followed:
</p>

<ol>
<li><p> The space defined by the loci is ordinated to yield a series of
orthogonal axes (independent), and the top two dimensions are considered.
Populations for which the unknown lies outside the specified confidence
limits are no longer removed from the dataset.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>gl.assign.pca(x, unknown, plevel = 0.999, plot.out = TRUE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.assign.pca_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.assign.pca_+3A_unknown">unknown</code></td>
<td>
<p>Identity label of the focal individual whose provenance is
unknown [required].</p>
</td></tr>
<tr><td><code id="gl.assign.pca_+3A_plevel">plevel</code></td>
<td>
<p>Probability level for bounding ellipses in the PCoA plot
[default 0.999].</p>
</td></tr>
<tr><td><code id="gl.assign.pca_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, plot the 2D PCA showing the position 
of the unknown [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.assign.pca_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are three considerations to assignment. First, consider only those
populations for which the unknown has no private alleles. Private alleles are
an indication that the unknown does not belong to a target population
(provided that the sample size is adequate, say &gt;=10). This can be evaluated
with gl.assign.pa().
</p>
<p>A next step is to consider the PCoA plot for populations where no private
alleles have been detected and the position of the unknown in relation to the
confidence ellipses as is plotted by this script. Note, this plot is
considering only the top two dimensions of the ordination, and so an unknown
lying outside the confidence ellipse can be unambiguously interpreted as it 
lying outside the confidence envelope. However, if the unknown lies inside 
the confidence ellipse in two dimensions, then it may still lie outside the 
confidence envelope in deeper dimensions. This second step is good for 
eliminating populations from consideration, but does not provide confidence 
in assignment.
</p>
<p>The third step is to consider the assignment probabilities, using the script
gl.assign.mahalanobis(). This approach calculates the squared Generalised 
Linear Distance (Mahalanobis distance) of the unknown from the centroid 
for each population, and calculates the probability associated with its 
quantile under the zero truncated normal distribution. This index takes 
into account position of the unknown in relation to the confidence envelope 
in all selected dimensions of the ordination. 
</p>
<p>Each of these approaches provides evidence, none are 100
need to be interpreted cautiously. They are best applied sequentially.
</p>
<p>In deciding the assignment, the script considers an individual to be an
outlier with respect to a particular population at alpha = 0.001 as default.
</p>


<h3>Value</h3>

<p>A genlight object containing only those populations that are
putative source populations for the unknown individual.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Test run with a focal individual from the Macleay River (EmmacMaclGeor) 
test &lt;- gl.assign.pa(testset.gl, unknown='UC_00146', nmin=10, threshold=1,
verbose=3) 
test_2 &lt;- gl.assign.pca(test, unknown='UC_00146', plevel=0.95, verbose=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.basic.stats'>Calculates basic statistics for each loci (Hs, Ho, Fis etc.)</h2><span id='topic+gl.basic.stats'></span>

<h3>Description</h3>

<p>Based on function <code><a href="hierfstat.html#topic+basic.stats">basic.stats</a></code>. Check ?basic.stats
for help.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.basic.stats(x, digits = 4, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.basic.stats_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.basic.stats_+3A_digits">digits</code></td>
<td>
<p>Number of digits that should be returned [default 4].</p>
</td></tr>
<tr><td><code id="gl.basic.stats_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Several tables and lists with all basic stats.
<code><a href="hierfstat.html#topic+basic.stats">basic.stats</a></code> for details.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (!(requireNamespace("hierfstat", quietly = TRUE))) {
out &lt;- gl.basic.stats(possums.gl[1:10,1:100])
}
</code></pre>

<hr>
<h2 id='gl.blast'>Aligns nucleotides sequences against those present in a target database 
using blastn</h2><span id='topic+gl.blast'></span>

<h3>Description</h3>

<p>Basic Local Alignment Search Tool (BLAST; Altschul et al., 1990 &amp;
1997) is a sequence comparison algorithm optimized for speed used to search
sequence databases for optimal local alignments to a query. This function
creates fasta files, creates databases to run BLAST, runs blastn and filters
these results to obtain the best hit per sequence.
</p>
<p>This function can be used to run BLAST alignment of short-read (DArTseq
data) and long-read sequences (Illumina, PacBio... etc). You can use
reference genomes from NCBI, genomes from your private collection, contigs,
scaffolds or any other genetic sequence that you would like to use as
reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.blast(
  x,
  ref_genome,
  task = "megablast",
  Percentage_identity = 70,
  Percentage_overlap = 0.8,
  bitscore = 50,
  number_of_threads = 2,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.blast_+3A_x">x</code></td>
<td>
<p>Either a genlight object containing a column named
'TrimmedSequence' containing the sequence of the SNPs (the sequence tag)
trimmed of adapters as provided by DArT; or a path to a fasta file with the
query sequences [required].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_ref_genome">ref_genome</code></td>
<td>
<p>Path to a reference genome in fasta of fna format
[required].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_task">task</code></td>
<td>
<p>Four different tasks are supported: 1) “megablast”, for very
similar sequences (e.g, sequencing errors), 2) “dc-megablast”, typically
used for inter-species comparisons, 3) “blastn”, the traditional program
used for inter-species comparisons, 4) “blastn-short”, optimized for
sequences less than 30 nucleotides [default 'megablast'].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_percentage_identity">Percentage_identity</code></td>
<td>
<p>Not a very sensitive or reliable measure of
sequence similarity, however it is a reasonable proxy for evolutionary
distance. The evolutionary distance associated with a 10 percent change in
Percentage_identity is much greater at longer distances. Thus, a change from
80 – 70 percent identity might reflect divergence 200 million years earlier
in time, but the change from 30 percent to 20 percent might correspond to a
billion year divergence time change [default 70].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_percentage_overlap">Percentage_overlap</code></td>
<td>
<p>Calculated as alignment length divided by the
query length or subject length (whichever is shortest of the two lengths,
i.e.  length / min(qlen,slen) ) [default 0.8].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_bitscore">bitscore</code></td>
<td>
<p>A rule-of-thumb for inferring homology, a bit score of 50
is almost always significant [default 50].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_number_of_threads">number_of_threads</code></td>
<td>
<p>Number of threads (CPUs) to use in blastn search
[default 2].</p>
</td></tr>
<tr><td><code id="gl.blast_+3A_verbose">verbose</code></td>
<td>
<p>verbose= 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Installing BLAST</strong>
</p>
<p>You can download the BLAST installs from:
<a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
<p>It is important to install BLAST in a path that does not contain spaces for
this function to work.
</p>
<p><strong>Running BLAST</strong>
</p>
<p>Four different tasks are supported: </p>
 <ul>
<li><p> “megablast”, for very
similar sequences (e.g, sequencing errors) </p>
</li>
<li><p> “dc-megablast”, typically
used for inter-species comparisons </p>
</li>
<li><p> “blastn”, the traditional program
used for inter-species comparisons </p>
</li>
<li><p> “blastn-short”, optimized for
sequences less than 30 nucleotides </p>
</li></ul>

<p>If  you  are  running  a  BLAST alignment of  similar  sequences,  for
example  Turtle  Genome  Vs Turtle Sequences, the recommended parameters
are: task = “megablast”, Percentage_identity = 70, Percentage_overlap =  0.8
and bitscore = 50.
</p>
<p>If you are running a BLAST alignment of highly dissimilar sequences because
you are probably looking for sex linked  hits in  a distantly  related
species,  and  you  are aligning for example sequences of Chicken Genome Vs
Bassiana, the recommended parameters are: task = “dc-megablast”,
Percentage_identity = 50, Percentage_overlap =  0.01 and bitscore = 30.
</p>
<p>Be aware that running BLAST might take a long time (i.e. days) depending of
the size of your query, the size of your database and the number of threads
selected for your computer.
</p>
<p><strong>BLAST output</strong>
</p>
<p>The BLAST output is formatted as a table using output format 6, with columns
defined in the following order: </p>
 <ul>
<li><p> qseqid - Query Seq-id </p>
</li>
<li>
<p>sacc - Subject accession </p>
</li>
<li><p> stitle - Subject Title </p>
</li>
<li><p> qseq - Aligned
part of query sequence </p>
</li>
<li><p> sseq - Aligned part of subject sequence </p>
</li>
<li>
<p>nident - Number of identical matches </p>
</li>
<li><p> mismatch - Number of mismatches
</p>
</li>
<li><p> pident - Percentage of identical matches </p>
</li>
<li><p> length - Alignment
length </p>
</li>
<li><p> evalue - Expect value </p>
</li>
<li><p> bitscore - Bit score </p>
</li>
<li><p> qstart -
Start of alignment in query </p>
</li>
<li><p> qend - End of alignment in query </p>
</li>
<li>
<p>sstart - Start of alignment in subject </p>
</li>
<li><p> send - End of alignment in
subject </p>
</li>
<li><p> gapopen - Number of gap openings </p>
</li>
<li><p> gaps - Total number of
gaps </p>
</li>
<li><p> qlen - Query sequence length </p>
</li>
<li><p> slen - Subject sequence length
</p>
</li>
<li><p> PercentageOverlap - length / min(qlen,slen) </p>
</li></ul>

<p>Databases containing unfiltered aligned sequences, filtered aligned
sequences and one hit per sequence are saved to the temporal directory
(tempdir) and can be accessed with the function
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with the function
<code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed only in the
current R session because tempdir is cleared each time that the R session is
closed.
</p>
<p><strong>BLAST filtering</strong>
</p>
<p>BLAST output is filtered by ordering the hits of each sequence first by the
highest percentage identity, then the highest percentage overlap and then
the highest bitscore. Only one hit per sequence is kept based on these
selection criteria.
</p>


<h3>Value</h3>

<p>If the input is a genlight object: returns a genlight object with one
hit per sequence merged to the slot $other$loc.metrics. If the input is a
fasta file: returns a dataframe with one hit per sequence.
</p>


<h3>Author(s)</h3>

<p>Berenice Talamantes Becerra &amp; Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Altschul, S. F., Gish, W., Miller, W., Myers, E. W., &amp; Lipman, D.
J. (1990). Basic local alignment search tool. Journal of molecular biology,
215(3), 403-410.
</p>
</li>
<li><p> Altschul, S. F., Madden, T. L., Schäffer, A. A., Zhang, J., Zhang,
Z., Miller, W., &amp; Lipman, D. J. (1997). Gapped BLAST and PSI-BLAST: a new
generation of protein database search programs. Nucleic acids research,
25(17), 3389-3402.
</p>
</li>
<li><p> Pearson, W. R. (2013). An introduction to sequence similarity
(“homology”) searching. Current protocols in bioinformatics, 42(1), 3-1.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.print.history">gl.print.history</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
res &lt;- gl.blast(x= testset.gl,ref_genome = 'sequence.fasta')
# display of reports saved in the temporal directory
gl.list.reports()
# open the reports saved in the temporal directory
blast_databases &lt;- gl.print.reports(1)

## End(Not run)

</code></pre>

<hr>
<h2 id='gl.check.verbosity'>Checks the current global verbosity</h2><span id='topic+gl.check.verbosity'></span>

<h3>Description</h3>

<p>The verbosity can be set in one of two ways &ndash; (a) explicitly by the user by
passing a value using the parameter verbose in a function, or (b) by setting
the verbosity globally as part of the r environment (gl.set.verbosity).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.check.verbosity(x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.check.verbosity_+3A_x">x</code></td>
<td>
<p>User requested level of verbosity [default NULL].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The verbosity, in variable verbose
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.check.verbosity()
</code></pre>

<hr>
<h2 id='gl.collapse'>Collapses a distance matrix by amalgamating populations with pairwise
fixed difference count less that a threshold</h2><span id='topic+gl.collapse'></span>

<h3>Description</h3>

<p>This script takes a file generated by gl.fixed.diff and amalgamates
populations with distance less than or equal to a specified threshold. The
distance matrix is generated by gl.fixed.diff().
</p>
<p>The script then applies the new population assignments to the genlight object
and recalculates the distance and associated matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.collapse(fd, tpop = 0, tloc = 0, pb = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.collapse_+3A_fd">fd</code></td>
<td>
<p>Name of the list of matrices produced by gl.fixed.diff() 
[required].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_tpop">tpop</code></td>
<td>
<p>Threshold number of fixed differences above which populations
will not be amalgamated [default 0].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_tloc">tloc</code></td>
<td>
<p>Threshold defining a fixed difference (e.g. 0.05 implies 95:5 vs
5:95 is fixed) [default 0].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_pb">pb</code></td>
<td>
<p>If TRUE, show a progress bar on time consuming loops 
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.collapse_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the gl object x and the following square matrices:
</p>

<ol>
<li><p> $gl &ndash; the new genlight object with populations collapsed;
</p>
</li>
<li><p> $fd &ndash; raw fixed differences;
</p>
</li>
<li><p> $pcfd &ndash; percent fixed differences;
</p>
</li>
<li><p> $nobs &ndash; mean no. of individuals used in each comparison;
</p>
</li>
<li><p> $nloc &ndash; total number of loci used in each comparison;
</p>
</li>
<li><p> $expfpos &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li>
<li><p> $expfpos &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li>
<li><p> $prob &ndash; NA's, populated by gl.fixed.diff [by simulation]
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
fd &lt;- gl.fixed.diff(testset.gl,tloc=0.05)
fd
fd2 &lt;- gl.collapse(fd,tpop=1)
fd2
fd3 &lt;- gl.collapse(fd2,tpop=1)
fd3

 fd &lt;- gl.fixed.diff(testset.gl,tloc=0.05)
 fd2 &lt;- gl.collapse(fd)
</code></pre>

<hr>
<h2 id='gl.compliance.check'>Checks a genlight object to see if it complies with dartR
expectations and amends it to comply if necessary</h2><span id='topic+gl.compliance.check'></span>

<h3>Description</h3>

<p>This function will check to see that the genlight object conforms to
expectation in regard to dartR requirements (see details), and if it does
not, will rectify it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.compliance.check(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.compliance.check_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.compliance.check_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A genlight object used by dartR has a number of requirements that allow
functions within the package to operate correctly. The genlight object
comprises:
</p>

<ol>
<li><p> The SNP genotypes or Tag Presence/Absence data (SilicoDArT);
</p>
</li>
<li><p> An associated dataframe (gl@other$loc.metrics) containing the locus
metrics (e.g. Call Rate, Repeatability, etc);
</p>
</li>
<li><p> An associated dataframe (gl@other$ind.metrics) containing the
individual/sample metrics (e.g. sex, latitude (=lat), longitude(=lon), etc);
</p>
</li>
<li><p> A specimen identity field (indNames(gl)) with the unique labels applied
to each individual/sample;
</p>
</li>
<li><p> A population assignment (popNames) for each individual/specimen;
</p>
</li>
<li><p> Flags that indicate whether or not calculable locus metrics have been
updated.
</p>
</li></ol>



<h3>Value</h3>

<p>A genlight object that conforms to the expectations of dartR
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- gl.compliance.check(testset.gl)
x &lt;- gl.compliance.check(testset.gs)
</code></pre>

<hr>
<h2 id='gl.costdistances'>Calculates cost distances for a given landscape (resistance matrix)</h2><span id='topic+gl.costdistances'></span>

<h3>Description</h3>

<p>Calculates a cost distance matrix, to be used with run.popgensim.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.costdistances(landscape, locs, method, NN, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.costdistances_+3A_landscape">landscape</code></td>
<td>
<p>A raster object coding the resistance of the landscape
[required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_locs">locs</code></td>
<td>
<p>Coordinates of the subpopulations. If a genlight object is
provided coordinates are taken from @other$latlon and centers for population
(pop(gl)) are calculated. In case you want to calculate costdistances between
individuals redefine pop(gl) via: <code>pop(gl)&lt;- indNames(gl)</code> [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_method">method</code></td>
<td>
<p>Defines the type of cost distance, types are 'leastcost',
'rSPDistance' or 'commute' (Circuitscape type) [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_nn">NN</code></td>
<td>
<p>Number of next neighbours recommendation is 8 [required].</p>
</td></tr>
<tr><td><code id="gl.costdistances_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A costdistance matrix between all pairs of locs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(possums.gl)
library(raster)  #needed for that example
landscape.sim &lt;- readRDS(system.file('extdata','landscape.sim.rdata', 
package='dartR'))
#calculate mean centers of individuals per population
xy &lt;- apply(possums.gl@other$xy, 2, function(x) tapply(x, pop(possums.gl),
 mean))
cd &lt;- gl.costdistances(landscape.sim, xy, method='leastcost', NN=8)
round(cd,3)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.define.pop'>Defines a new population in a genlight object for specified 
individuals</h2><span id='topic+gl.define.pop'></span>

<h3>Description</h3>

<p>The script reassigns existing individuals to a new population and removes
their existing population assignment.
</p>
<p>The script returns a genlight object with the new population assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.define.pop(x, ind.list, new, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.define.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_ind.list">ind.list</code></td>
<td>
<p>A list of individuals to be assigned to the new population
[required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_new">new</code></td>
<td>
<p>Name of the new population [required].</p>
</td></tr>
<tr><td><code id="gl.define.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the redefined population structure.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>popNames(testset.gl)
gl &lt;- gl.define.pop(testset.gl, ind.list=c('AA019073','AA004859'), 
new='newguys')
popNames(gl)
indNames(gl)[pop(gl)=='newguys']
</code></pre>

<hr>
<h2 id='gl.diagnostics.hwe'>Provides descriptive stats and plots to diagnose potential problems
with Hardy-Weinberg proportions</h2><span id='topic+gl.diagnostics.hwe'></span>

<h3>Description</h3>

<p>Different causes may be responsible for lack of Hardy-Weinberg
proportions. This function helps diagnose potential problems.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.diagnostics.hwe(
  x,
  alpha_val = 0.05,
  bins = 20,
  stdErr = TRUE,
  colors_hist = two_colors,
  colors_barplot = two_colors_contrast,
  plot_theme = theme_dartR(),
  save2tmp = FALSE,
  n.cores = "auto",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.diagnostics.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 20].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_stderr">stdErr</code></td>
<td>
<p>Whether standard errors for Fis and Fst should be computed 
(default: TRUE)</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_colors_hist">colors_hist</code></td>
<td>
<p>List of two color names for the borders and fill of the
histogram [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_colors_barplot">colors_barplot</code></td>
<td>
<p>Vector with two color names for the observed and
expected number of significant HWE tests [default two_colors_contrast].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use. If &quot;auto&quot;, it will 
use all but one available cores [default &quot;auto&quot;].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function initially runs <code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code> and reports
the ternary plots. The remaining outputs follow the recommendations from
Waples
(2015) paper and De Meeûs 2018. These include: 
</p>
 
<ol>
<li><p> A histogram
with the distribution of p-values of the HWE tests. The distribution should
be roughly uniform across equal-sized bins. 
</p>
</li>
<li><p> A bar plot with observed
and expected (null expectation) number of significant HWE tests for the same
locus in multiple populations (that is, the x-axis shows whether a locus
results significant in 1, 2, ..., n populations. The y axis is the count of
these occurrences. The zero value on x-axis shows the number of
non-significant tests). If HWE tests are significant by chance alone,
observed and expected number of HWE tests should have roughly a similar
distribution. 
</p>
</li>
<li><p> A scatter plot with a linear regression between Fst and Fis,
averaged across subpopulations. De Meeûs 2018 suggests that in the case of
Null alleles, a strong positive relationship is expected (together with the
Fis standard error much larger than the Fst standard error, see below).
<b>Note</b>, this is not the scatter plot that Waples 2015 presents in his
paper. In the lower right corner of the plot, the Pearson correlation
coefficient is reported. 
</p>
</li>
<li><p> The Fis and Fst (averaged over loci and
subpopulations) standard errors are also printed on screen and reported in
the returned list (if <code>stdErr=TRUE</code>). These are computed with the 
Jackknife method over loci (See De Meeûs 2007 for details on how this is 
computed) and it may take some time for these computations to complete. 
De Meeûs 2018 suggests that under a global significant heterozygosity 
deficit: 
</p>
<p>- if the
correlation between Fis and Fst is strongly positive, and StdErrFis &gt;&gt;
StdErrFst, Null alleles are likely to be the cause. 
</p>
<p>- if the correlation
between Fis and Fst is ~0 or mildly positive, and StdErrFis &gt; StdErrFst,
Wahlund may be the cause. 
</p>
<p>- if the correlation between Fis and Fst is ~0, and
StdErrFis ~ StdErrFst, selfing or sib mating could to be the cause.
</p>
<p>It is
important to realise that these statistics only suggest a pattern (pointers).
Their absence is not conclusive evidence of the absence of the problem, as 
their presence does not confirm the cause of the problem. 
</p>
</li>
<li><p> A table where the
number of observed and expected significant HWE tests are reported by each
population, indicating whether these are due to heterozygosity excess or
deficiency. These can be used to have a clue of potential problems (e.g.
deficiency might be due to a Wahlund effect, presence of null alleles or
non-random sampling; excess might be due to sex linkage or different
selection between sexes, demographic changes or small Ne. See Table 1 in
Wapples 2015). The last two columns of the table generated by this function
report chisquare values and their associated p-values. Chisquare is computed
following Fisher's procedure for a global test (Fisher 1970). This basically
tests whether there is at least one test that is truly significant in the
series of tests conducted (De Meeûs et al 2009).
</p>
</li></ol>



<h3>Value</h3>

<p>A list with the table with the summary of the HWE tests and (if 
stdErr=TRUE) a named vector with the StdErrFis and StdErrFst.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>

 
<ul>
<li><p> de Meeûs, T., McCoy, K.D., Prugnolle, F.,
Chevillon, C., Durand, P., Hurtrez-Boussès, S., Renaud, F., 2007. Population
genetics and molecular epidemiology or how to “débusquer la bête”. Infection,
Genetics and Evolution 7, 308-332. 
</p>
</li>
<li><p> De Meeûs, T., Guégan, J.-F., Teriokhin, A.T., 2009. MultiTest V.1.2, 
a program to binomially combine independent tests and performance comparison 
with other related methods on
proportional data. BMC Bioinformatics 10, 443-443. 
</p>
</li>
<li><p> De Meeûs, T., 2018. Revisiting FIS, FST, Wahlund Effects, and Null 
Alleles. Journal of Heredity 109, 446-456.
</p>
</li>
<li><p> Fisher, R., 1970.
Statistical methods for research workers Edinburgh: Oliver and Boyd. 
</p>
</li>
<li>
<p>Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we lost
the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("dartR.data")
res &lt;- gl.diagnostics.hwe(x = gl.filter.allna(platypus.gl[,1:50]), 
stdErr=FALSE, n.cores=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.diagnostics.sim'>Comparing simulations against theoretical expectations</h2><span id='topic+gl.diagnostics.sim'></span>

<h3>Description</h3>

<p>Comparing simulations against theoretical expectations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.diagnostics.sim(
  x,
  Ne,
  iteration = 1,
  pop_he = 1,
  pops_fst = c(1, 2),
  plot_theme = theme_dartR(),
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.diagnostics.sim_+3A_x">x</code></td>
<td>
<p>Output from function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_ne">Ne</code></td>
<td>
<p>Effective population size to use as input to compare theoretical 
expectations [required].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_iteration">iteration</code></td>
<td>
<p>Iteration number to analyse [default 1].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_pop_he">pop_he</code></td>
<td>
<p>Population name in which the rate of loss of heterozygosity is
going to be compared against theoretical expectations [default 1].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_pops_fst">pops_fst</code></td>
<td>
<p>Pair of populations in which FST is going to be compared 
against theoretical expectations [default c(1,2)].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.diagnostics.sim_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two plots are presented comparing the simulations against theoretical 
expectations:
</p>

<ol>
<li><p> Expected heterozygosity under neutrality (Crow &amp; Kimura, 1970, p. 329) is 
calculated as:
</p>
<p>Het = He0(1-(1/2Ne))^t,
</p>
<p>where Ne is effective population size, He0  is heterozygosity at generation 0
and t is the number of generations.
</p>
</li>
<li><p> Expected FST under neutrality (Takahata, 1983) is calculated as:
</p>
<p>FST=1/(4Nem(n/(n-1))^2+1),
</p>
<p>where Ne is effective populations size of each individual subpopulation, m is
dispersal rate and n the number of subpopulations (always 2).
</p>
</li></ol>



<h3>Value</h3>

<p>Returns plots comparing simulations against theoretical expectations
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Crow JF, Kimura M. An introduction to population genetics theory. An 
introduction to population genetics theory. 1970.
</p>
</li>
<li><p> Takahata N. Gene identity and genetic differentiation of populations in 
the finite island model. Genetics. 1983;104(3):497-512.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ref_table &lt;- gl.sim.WF.table(file_var=system.file('extdata', 
'ref_variables.csv', package = 'dartR'),interactive_vars = FALSE)
res_sim &lt;- gl.sim.WF.run(file_var = system.file('extdata', 
'sim_variables.csv', package ='dartR'),ref_table=ref_table,
interactive_vars = FALSE,number_pops_phase2=2,population_size_phase2="50 50")
res &lt;- gl.diagnostics.sim(x=res_sim,Ne=50)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.dist.ind'>Calculates a distance matrix for individuals defined in a genlight object</h2><span id='topic+gl.dist.ind'></span>

<h3>Description</h3>

<p>This script calculates various distances between individuals based on allele
frequencies or presence-absence data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.dist.ind(
  x,
  method = NULL,
  scale = FALSE,
  swap = FALSE,
  output = "dist",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.dist.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes or presence-absence data [required].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_method">method</code></td>
<td>
<p>Specify distance measure [SNP: Euclidean; P/A: Simple].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_scale">scale</code></td>
<td>
<p>If TRUE, the distances are scaled to fall in the range [0,1] [default TRUE]</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_swap">swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_output">output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
'dist' for a object of class dist, 'matrix' for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, display a histogram and a boxplot of the genetic distances [TRUE].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots to the session temporary directory [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure for SNP genotypes can be one of:
</p>

<ul>
<li><p> Euclidean Distance [method = &quot;Euclidean&quot;]
</p>
</li>
<li><p> Scaled Euclidean Distance [method='Euclidean&quot;, scale=TRUE]
</p>
</li>
<li><p> Simple Mismatch Distance [method=&quot;Simple&quot;]
</p>
</li>
<li><p> Absolute Mismatch Distance [method=&quot;Absolute&quot;]
</p>
</li>
<li><p> Czekanowski (Manhattan) Distance [method=&quot;Manhattan&quot;]
</p>
</li></ul>

<p>The distance measure for Sequence Tag Presence/Absence data (binary) can be one of:
</p>

<ul>
<li><p> Euclidean Distance [method = &quot;Euclidean&quot;]
</p>
</li>
<li><p> Scaled Euclidean Distance [method='Euclidean&quot;, scale=TRUE]
</p>
</li>
<li><p> Simple Matching Distance [method=&quot;Simple&quot;]
</p>
</li>
<li><p> Jaccard Distance [method=&quot;Jaccard&quot;]
</p>
</li>
<li><p> Bray-Curtis Distance [method=&quot;Bray-Curtis&quot;]
</p>
</li></ul>

<p>Refer to the dartR Technical Note on Distances in Genetics.
</p>


<h3>Value</h3>

<p>An object of class 'matrix' or dist' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to #' <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
D &lt;- gl.dist.ind(testset.gl[1:20,], method='manhattan')
D &lt;- gl.dist.ind(testset.gs[1:20,], method='Jaccard',swap=TRUE)

D &lt;- gl.dist.ind(testset.gl[1:20,], method='euclidean',scale=TRUE)
</code></pre>

<hr>
<h2 id='gl.dist.pop'>Calculates a distance matrix for populations with SNP genotypes in a
genlight object</h2><span id='topic+gl.dist.pop'></span>

<h3>Description</h3>

<p>This script calculates various distances between populations based on allele
frequencies (SNP genotypes) or frequency of presences in presence-absence data 
(Euclidean and Fixed-diff distances only).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.dist.pop(
  x,
  method = "euclidean",
  plot.out = TRUE,
  scale = FALSE,
  output = "dist",
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.dist.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default euclidean].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, display a histogram of the genetic distances, and a
whisker plot [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_output">output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two_colors].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.dist.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure can be one of 'euclidean', 'fixed-diff', 'reynolds',
'nei' and 'chord'. Refer to the documentation of functions
described in the the dartR Distance Analysis tutorial for algorithms
and definitions.
</p>


<h3>Value</h3>

<p>An object of class 'dist' giving distances between populations
</p>


<h3>Author(s)</h3>

<p>author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
# SNP genotypes
D &lt;- gl.dist.pop(possums.gl[1:90,1:100], method='euclidean')
D &lt;- gl.dist.pop(possums.gl[1:90,1:100], method='euclidean',scale=TRUE)
#D &lt;- gl.dist.pop(possums.gl, method='nei')
#D &lt;- gl.dist.pop(possums.gl, method='reynolds')
#D &lt;- gl.dist.pop(possums.gl, method='chord')
#D &lt;- gl.dist.pop(possums.gl, method='fixed-diff')
#Presence-Absence data [only 10 individuals due to speed]
D &lt;- gl.dist.pop(testset.gs[1:10,], method='euclidean')

## End(Not run)
res &lt;- gl.dist.pop(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.drop.ind'>Removes specified individuals from a genlight object</h2><span id='topic+gl.drop.ind'></span>

<h3>Description</h3>

<p>The script, having deleted individuals, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally recalculates
statistics made redundant by the deletion of individuals from the dataset.
</p>
<p>The script returns a genlight object with the individuals deleted and,
optionally, the recalculated locus metadata.
</p>
<p>#' See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.ind(x, ind.list, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_ind.list">ind.list</code></td>
<td>
<p>A list of individuals to be removed [required].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.ind">gl.keep.ind</a></code> to keep rather than drop specified
individuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.drop.ind(testset.gl,
   ind.list=c('AA019073','AA004859'))
   gl2 &lt;- gl.drop.ind(testset.gl,
   ind.list=c('AA019073','AA004859'))
 # Tag P/A data
   gs2 &lt;- gl.drop.ind(testset.gs,
   ind.list=c('AA020656','AA19077','AA004859'))
   gs2 &lt;- gl.drop.ind(testset.gs, ind.list=c('AA020656'
   ,'AA19077','AA004859'),mono.rm=TRUE, recalc=TRUE)

</code></pre>

<hr>
<h2 id='gl.drop.loc'>Removes specified loci from a genlight object</h2><span id='topic+gl.drop.loc'></span>

<h3>Description</h3>

<p>The script returns a genlight object with specified loci deleted.
</p>
<p>#' See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.loc(x, loc.list = NULL, first = NULL, last = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.loc_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes or
presence/absence data [required].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_loc.list">loc.list</code></td>
<td>
<p>A list of loci to be deleted
[required, if loc.range not specified].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_first">first</code></td>
<td>
<p>First of a range of loci to be deleted
[required, if loc.list not specified].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_last">last</code></td>
<td>
<p>Last of a range of loci to be deleted
[if not specified, last locus in the dataset].</p>
</td></tr>
<tr><td><code id="gl.drop.loc_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.loc">gl.keep.loc</a></code> to keep rather than drop specified loci
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl2 &lt;- gl.drop.loc(testset.gl, loc.list=c('100051468|42-A/T', '100049816-51-A/G'),verbose=3)
# Tag P/A data
  gs2 &lt;- gl.drop.loc(testset.gs, loc.list=c('20134188','19249144'),verbose=3)

</code></pre>

<hr>
<h2 id='gl.drop.pop'>Removes specified populations from a genlight object</h2><span id='topic+gl.drop.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata
file (csv) used with <code><a href="#topic+gl.read.dart">gl.read.dart</a></code>.
The script, having deleted populations, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally
recalculates statistics made redundant by the deletion of individuals from
the dataset.
</p>
<p>The script returns a genlight object with the new population assignments and
the recalculated locus metadata.
</p>
<p>#' See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.drop.pop(
  x,
  pop.list,
  as.pop = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.drop.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes or Tag P/A data
(SilicoDArT) [required].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_pop.list">pop.list</code></td>
<td>
<p>A list of populations to be removed [required].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Temporarily assign another metric to represent population for
the purposes of deletions [default NULL].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.drop.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.keep.pop">gl.keep.pop</a></code> to keep rather than drop specified
populations
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.drop.pop(testset.gl,
   pop.list=c('EmsubRopeMata','EmvicVictJasp'),verbose=3)
   gl2 &lt;- gl.drop.pop(testset.gl, pop.list=c('EmsubRopeMata','EmvicVictJasp'),
   mono.rm=TRUE,recalc=TRUE)
   gl2 &lt;- gl.drop.pop(testset.gl, pop.list=c('Male','Unknown'),as.pop='sex',verbose=3)
 # Tag P/A data
   gs2 &lt;- gl.drop.pop(testset.gs, pop.list=c('EmsubRopeMata','EmvicVictJasp'))

</code></pre>

<hr>
<h2 id='gl.edit.recode.ind'>Creates or edits individual (=specimen) names, creates a recode_ind
file and applies the changes to a genlight object</h2><span id='topic+gl.edit.recode.ind'></span>

<h3>Description</h3>

<p>A script to edit individual names in a genlight object, or to create a
reassignment table taking the individual labels from a genlight object, or to
edit existing individual labels in an existing recode_ind file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.edit.recode.ind(
  x,
  out.recode.file = NULL,
  outpath = tempdir(),
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.edit.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object for which individuals are to be
relabelled [required].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>Name of the file to output the new individual labels
[optional].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Renaming individuals may be required when there have been errors in labelling
arising in the process from sample to DArT files. There may be occasions
where renaming individuals is required for preparation of figures. Caution
needs to be exercised because of the potential for breaking the 'chain of
evidence' between the samples themselves and the analyses. Recoding
individuals can also be done with a recode table (csv).
</p>
<p>This script will input an existing recode table for editing and optionally
save it as a new table, or if the name of an input table is not supplied,
will generate a table using the individual labels in the parent genlight
object.
</p>
<p>The script, having deleted individuals, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them (using
gl.filter.monomorphs.r). The script also optionally recalculates statistics
made redundant by the deletion of individuals from the dataset.
</p>
<p>Use outpath=getwd() or outpath='.' when calling this function to direct
output files to your working directory.
</p>
<p>The script returns a genlight object with the new individual labels and the
recalculated locus metadata.
</p>


<h3>Value</h3>

<p>An object of class ('genlight') with the revised individual labels.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.recode.ind">gl.recode.ind</a></code>, <code><a href="#topic+gl.drop.ind">gl.drop.ind</a></code>,
<code><a href="#topic+gl.keep.ind">gl.keep.ind</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gl &lt;- gl.edit.recode.ind(testset.gl)
gl &lt;- gl.edit.recode.ind(testset.gl, out.recode.file='ind.recode.table.csv')

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.edit.recode.pop'>Creates or edits a population re-assignment table</h2><span id='topic+gl.edit.recode.pop'></span>

<h3>Description</h3>

<p>A script to edit population assignments in a genlight object, or to
create a reassignment table taking the population assignments
from a genlight object, or to edit existing population assignments in
a pop.recode.table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.edit.recode.pop(
  x,
  pop.recode = NULL,
  out.recode.file = NULL,
  outpath = tempdir(),
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.edit.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object for which populations are to be
reassigned [required].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_pop.recode">pop.recode</code></td>
<td>
<p>Path to recode file [default NULL].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>Name of the file to output the new individual labels
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals are deleted
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.edit.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genlight objects assign specimens to populations based on information in the
ind.metadata file provided when the genlight object is first generated.
Often one wishes to subset the data by deleting populations or to amalgamate
populations. This can be done with a pop.recode table with two columns. The
first column is the population assignment in the genlight object, the second
column provides the new assignment.
</p>
<p>This script will input an existing reassignment table for editing and
optionally save it as a new table, or if the name of an input table is not
supplied, will generate a table using the population assignments in the
parent genlight object.
</p>
<p>The script, having deleted populations, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them (using
gl.filter.monomorphs.r). The script also optionally recalculates statistics
made redundant by the deletion of individuals from the dataset.
</p>
<p>Use outpath=getwd() or outpath='.' when calling this function to direct
output files to your working directory.
</p>
<p>The script returns a genlight object with the new population assignments and
the recalculated locus metadata.
</p>


<h3>Value</h3>

<p>An object of class ('genlight') with the revised population assignments
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash;Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code>, <code><a href="#topic+gl.drop.pop">gl.drop.pop</a></code>,
<code><a href="#topic+gl.keep.pop">gl.keep.pop</a></code>, <code><a href="#topic+gl.merge.pop">gl.merge.pop</a></code>,
<code><a href="#topic+gl.reassign.pop">gl.reassign.pop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gl &lt;- gl.edit.recode.pop(testset.gl)
gs &lt;- gl.edit.recode.pop(testset.gs)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.evanno'>Creates an Evanno plot from a STRUCTURE run object</h2><span id='topic+gl.evanno'></span>

<h3>Description</h3>

<p>This function takes a genlight object and runs a STRUCTURE analysis based on
functions from <code>strataG</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.evanno(sr, plot.out = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.evanno_+3A_sr">sr</code></td>
<td>
<p>structure run object from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.evanno_+3A_plot.out">plot.out</code></td>
<td>
<p>TRUE: all four plots are shown. FALSE: all four plots are
returned as a ggplot but not shown [default TRUE].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is basically a convenient wrapper around the beautiful
strataG function <code>evanno</code> (Archer et al. 2016). For a detailed
description please refer to this package (see references below).
</p>


<h3>Value</h3>

<p>An Evanno plot is created and a list of all four plots is returned.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li>
<li><p> Evanno, G., Regnaut, S., and J. Goudet. 2005. Detecting the number of
clusters of individuals using the software STRUCTURE: a simulation study.
Molecular Ecology 14:2611-2620.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.run.structure">gl.run.structure</a></code>,  <code>clumpp</code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#CLUMPP and STRUCTURE need to be installed to be able to run the example
#bc &lt;- bandicoot.gl[,1:100]
#sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure.exe')
#ev &lt;- gl.evanno(sr)
#ev
#qmat &lt;- gl.plot.structure(sr, k=3, CLUMPP='d:/structure/')
#head(qmat)
#gl.map.structure(qmat, bc, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.fdsim'>Estimates the rate of false positives in a fixed difference analysis</h2><span id='topic+gl.fdsim'></span>

<h3>Description</h3>

<p>This function takes two populations and generates allele frequency profiles
for them. It then samples an allele frequency for each, at random, and
estimates a sampling distribution for those two allele frequencies. Drawing
two samples from those sampling distributions, it calculates whether or not
they represent a fixed difference. This is applied to all loci, and the
number of fixed differences so generated are counted, as an expectation. The
script distinguished between true fixed differences (with a tolerance of
delta), and false positives. The simulation is repeated a given number of
times (default=1000) to provide an expectation of the number of false
positives, given the observed allele frequency profiles and the sample sizes.
The probability of the observed count of fixed differences is greater than
the expected number of false positives is calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fdsim(
  x,
  poppair,
  obs = NULL,
  sympatric = FALSE,
  reps = 1000,
  delta = 0.02,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fdsim_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_poppair">poppair</code></td>
<td>
<p>Labels of two populations for comparison in the form
c(popA,popB) [required].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_obs">obs</code></td>
<td>
<p>Observed number of fixed differences between the two populations
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_sympatric">sympatric</code></td>
<td>
<p>If TRUE, the two populations are sympatric, if FALSE then
allopatric [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_reps">reps</code></td>
<td>
<p>Number of replications to undertake in the simulation
[default 1000].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_delta">delta</code></td>
<td>
<p>The threshold value for the minor allele frequency to regard the
difference between two populations to be fixed [default 0.02].</p>
</td></tr>
<tr><td><code id="gl.fdsim_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following square matrices
[[1]] observed fixed differences;
[[2]] mean expected number of false positives for each comparison;
[[3]] standard deviation of the no. of false positives for each
comparison;
[[4]] probability the observed fixed differences arose by chance for
each comparison.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fd &lt;- gl.fdsim(testset.gl[,1:100],poppair=c('EmsubRopeMata','EmmacBurnBara'),
sympatric=TRUE,verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.allna'>Filters loci that are all NA across individuals and/or populations 
with all NA across loci</h2><span id='topic+gl.filter.allna'></span>

<h3>Description</h3>

<p>This script deletes deletes loci or individuals with all calls missing (NA),
from a genlight object
</p>
<p>A DArT dataset will not have loci for which the calls are scored all as
missing (NA) for a particular individual, but such loci can arise rarely when
populations or individuals are deleted. Similarly, a DArT dataset will not
have individuals for which the calls are scored all as missing (NA) across
all loci, but such individuals may sneak in to the dataset when loci are
deleted. Retaining individual or loci with all NAs can cause issues for
several functions.
</p>
<p>Also, on occasion an analysis will require that there are some loci scored
in each population. Setting by.pop=TRUE will result in removal of loci when
they are all missing in any one population.
</p>
<p>Note that loci that are missing for all individuals in a population are
not imputed with method 'frequency' or 'HW'. Consider 
using the function <code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> with by.pop=TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.allna(x, by.pop = FALSE, recalc = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.allna_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_by.pop">by.pop</code></td>
<td>
<p>If TRUE, loci that are all missing in any one population
are deleted [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.allna_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object having removed individuals that are scored NA
across all loci, or loci that are scored NA across all individuals.
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  result &lt;- gl.filter.allna(testset.gl, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.allna(testset.gs, verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.callrate'>Filters loci or specimens in a genlight {adegenet} object based on
call rate</h2><span id='topic+gl.filter.callrate'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the restriction
enzyme recognition sites. The script gl.filter.callrate() will filter out the
loci with call rates below a specified threshold.
</p>
<p>Tag Presence/Absence datasets (SilicoDArT) have missing values where it is
not possible to determine reliably if there the sequence tag can be called at
a particular locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.callrate(
  x,
  method = "loc",
  threshold = 0.95,
  mono.rm = FALSE,
  recalc = FALSE,
  recursive = FALSE,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  bins = 25,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data, or the genind
object containing the SilocoDArT data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_method">method</code></td>
<td>
<p>Use method='loc' to specify that loci are to be filtered, 'ind'
to specify that specimens are to be filtered, 'pop' to remove loci that 
fail to meet the specified threshold in any one population [default 'loc'].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value below which loci will be removed
[default 0.95].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci after analysis is complete
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_recursive">recursive</code></td>
<td>
<p>Repeatedly filter individuals on call rate, each time
removing monomorphic loci. Only applies if method='ind' and mono.rm=TRUE
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if histograms of call rate, before and after, are to
be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme for the plot [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Because this filter operates on call rate, this function recalculates Call
Rate, if necessary, before filtering. If individuals are removed using
method='ind', then the call rate stored in the genlight object is, 
optionally,
recalculated after filtering.
</p>
<p>Note that when filtering individuals on call rate, the initial call rate is
calculated and compared against the threshold. After filtering, if
mono.rm=TRUE, the removal of monomorphic loci will alter the call rates.
Some individuals with a call rate initially greater than the nominated
threshold, and so retained, may come to have a call rate lower than the
threshold. If this is a problem, repeated iterations of this function will
resolve the issue. This is done by setting mono.rm=TRUE and recursive=TRUE,
or it can be done manually.
</p>
<p>Callrate is summarized by locus or by individual to allow sensible decisions
on thresholds for filtering taking into consideration consequential loss of
data. The summary is in the form of a tabulation and plots.
</p>
<p>Plot themes can be obtained from </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Resultant ggplot(s) and the tabulation(s) are saved to the session's
temporary directory.
</p>


<h3>Value</h3>

<p>The reduced genlight or genind object, plus a summary
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.callrate">gl.report.callrate</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='loc', threshold=0.8,
   verbose=3)
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='ind', threshold=0.8,
   verbose=3)
  result &lt;- gl.filter.callrate(testset.gl[1:10], method='pop', threshold=0.8,
   verbose=3)
# Tag P/A data
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='loc', 
  threshold=0.95, verbose=3)
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='ind', 
  threshold=0.8, verbose=3)
  result &lt;- gl.filter.callrate(testset.gs[1:10], method='pop', 
  threshold=0.8, verbose=3)
  
  res &lt;- gl.filter.callrate(platypus.gl)

</code></pre>

<hr>
<h2 id='gl.filter.hamming'>Filters loci based on pairwise Hamming distance between sequence tags</h2><span id='topic+gl.filter.hamming'></span>

<h3>Description</h3>

<p>Hamming distance is calculated as the number of base differences between two
sequences which can be expressed as a count or a proportion. Typically, it is
calculated between two sequences of equal length. In the context of DArT
trimmed sequences, which differ in length but which are anchored to the left
by the restriction enzyme recognition sequence, it is sensible to compare the
two trimmed sequences starting from immediately after the common recognition
sequence and terminating at the last base of the shorter sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.hamming(
  x,
  threshold = 0.2,
  rs = 5,
  taglength = 69,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  pb = FALSE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.hamming_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_threshold">threshold</code></td>
<td>
<p>A threshold Hamming distance for filtering loci
[default threshold 0.2].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_rs">rs</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_taglength">taglength</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_pb">pb</code></td>
<td>
<p>Switch to output progress bar [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.hamming_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Hamming distance can be computed
by exploiting the fact that the dot product of two binary vectors x and (1-y)
counts the corresponding elements that are different between x and y.
This approach can also be used for vectors that contain more than two 
possible values at each position (e.g. A, C, T or G).
</p>
<p>If a pair of DNA sequences are of differing length, the longer is truncated.
</p>
<p>The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
as implemented in <code><a href="#topic+utils.hamming">utils.hamming</a></code>.
</p>
<p>Only one of two loci are retained if their Hamming distance is less that a 
specified
percentage. 5 base differences out of 100 bases is a 20
</p>


<h3>Value</h3>

<p>A genlight object filtered on Hamming distance.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
test &lt;- platypus.gl
test &lt;- gl.subsample.loci(platypus.gl,n=50)
result &lt;- gl.filter.hamming(test, threshold=0.25, verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.heterozygosity'>Filters individuals with average heterozygosity greater than a
specified upper threshold or less than a specified lower threshold</h2><span id='topic+gl.filter.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the observed heterozygosity for each individual in a genlight
object and filters individuals based on specified threshold values.
Use gl.report.heterozygosity to determine the appropriate thresholds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.heterozygosity(x, t.upper = 0.7, t.lower = 0, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.heterozygosity_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_t.upper">t.upper</code></td>
<td>
<p>Filter individuals &gt; the threshold [default 0.7].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_t.lower">t.lower</code></td>
<td>
<p>Filter individuals &lt; the threshold [default 0].</p>
</td></tr>
<tr><td><code id="gl.filter.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The filtered genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> result &lt;- gl.filter.heterozygosity(testset.gl,t.upper=0.06,verbose=3)
 tmp &lt;- gl.report.heterozygosity(result,method='ind')
</code></pre>

<hr>
<h2 id='gl.filter.hwe'>Filters loci that show significant departure from Hardy-Weinberg
Equilibrium</h2><span id='topic+gl.filter.hwe'></span>

<h3>Description</h3>

<p>This function filters out loci showing significant departure from H-W
proportions based on observed frequencies of reference homozygotes,
heterozygotes and alternate homozygotes.
</p>
<p>Loci are filtered out if they show HWE departure either in any one population 
(n.pop.threshold =1) or in at least X number of populations 
(n.pop.threshold &gt; 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.hwe(
  x,
  subset = "each",
  n.pop.threshold = 1,
  method_sig = "Exact",
  multi_comp = FALSE,
  multi_comp_method = "BY",
  alpha_val = 0.05,
  pvalue_type = "midp",
  cc_val = 0.5,
  min_sample_size = 5,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_subset">subset</code></td>
<td>
<p>Way to group individuals to perform H-W tests. Either a vector
with population names, 'each', 'all' (see details) [default 'each'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_n.pop.threshold">n.pop.threshold</code></td>
<td>
<p>The minimum number of populations where the same locus 
has to be out of H-W proportions to be removed [default 1].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_method_sig">method_sig</code></td>
<td>
<p>Method for determining statistical significance: 
'ChiSquare'
or 'Exact' [default 'Exact'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_multi_comp">multi_comp</code></td>
<td>
<p>Whether to adjust p-values for multiple comparisons
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_multi_comp_method">multi_comp_method</code></td>
<td>
<p>Method to adjust p-values for multiple comparisons:
'holm', 'hochberg', 'hommel', 'bonferroni', 'BH', 'BY', 'fdr'
(see details) [default 'fdr'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_pvalue_type">pvalue_type</code></td>
<td>
<p>Type of p-value to be used in the Exact method.
Either 'dost','selome','midp' (see details) [default 'midp'].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_cc_val">cc_val</code></td>
<td>
<p>The continuity correction applied to the ChiSquare test
[default 0.5].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_min_sample_size">min_sample_size</code></td>
<td>
<p>Minimum number of individuals per population in which
perform H-W tests [default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are several factors that can cause deviations from Hardy-Weinberg
proportions including: mutation, finite population size, selection,
population structure, age structure, assortative mating, sex linkage,
nonrandom sampling and genotyping errors. Therefore, testing for
Hardy-Weinberg proportions should be a process that involves a careful
evaluation of the results, a good place to start is Waples (2015).
</p>
<p>Note that tests for H-W proportions are only valid if there is no population
substructure (assuming random mating) and have sufficient power only when
there is sufficient sample size (n individuals &gt; 15).
</p>
<p>Populations can be defined in three ways:
</p>

<ul>
<li><p> Merging all populations in the dataset using subset = 'all'.
</p>
</li>
<li><p> Within each population separately using: subset = 'each'.
</p>
</li>
<li><p> Within selected populations using for example: subset = 
c('pop1','pop2').
</p>
</li></ul>

<p>Two different statistical methods to test for deviations from Hardy Weinberg
proportions:
</p>

<ul>
<li><p> The classical chi-square test (method_sig='ChiSquare') based on the
function <code><a href="HardyWeinberg.html#topic+HWChisq">HWChisq</a></code> of the R package HardyWeinberg.
By default a continuity correction is applied (cc_val=0.5). The
continuity correction can be turned off (by specifying cc_val=0), for example
in cases of extreme allele frequencies in which the continuity correction can
lead to excessive type 1 error rates.
</p>
</li>
<li><p> The exact test (method_sig='Exact') based on the exact calculations
contained in the function <code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code> of the R
package HardyWeinberg, and described in Wigginton  et al. (2005). The exact
test is recommended in most cases (Wigginton  et al., 2005).
Three different methods to estimate p-values (pvalue_type) in the Exact test
can be used:
</p>

<ul>
<li><p> 'dost' p-value is computed as twice the tail area of a one-sided test.
</p>
</li>
<li><p> 'selome' p-value is computed as the sum of the probabilities of all
samples less or equally likely as the current sample.
</p>
</li>
<li><p> 'midp', p-value is computed as half the probability of the current
sample + the probabilities of all samples that are more extreme.
</p>
</li></ul>

<p>The standard exact p-value is overly conservative, in particular
for small minor allele frequencies. The mid p-value ameliorates this problem
by bringing the rejection rate closer to the nominal level, at the price of
occasionally exceeding the nominal level (Graffelman &amp; Moreno, 2013).
</p>
</li></ul>

<p>Correction for multiple tests can be applied using the following methods
based on the function <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>:
</p>

<ul>
<li><p> 'holm' is also known as the sequential Bonferroni technique 
(Rice, 1989).
This method has a greater statistical power than the standard Bonferroni 
test,
however this method becomes very stringent when many tests are performed and
many real deviations from the null hypothesis can go undetected 
(Waples, 2015).
</p>
</li>
<li><p> 'hochberg' based on Hochberg, 1988.
</p>
</li>
<li><p> 'hommel' based on Hommel, 1988. This method is more powerful than
Hochberg's, but the difference is usually small.
</p>
</li>
<li><p> 'bonferroni' in which p-values are multiplied by the number of tests.
This method is very stringent and therefore has reduced power to detect
multiple departures from the null hypothesis.
</p>
</li>
<li><p> 'BH' based on Benjamini &amp; Hochberg, 1995.
</p>
</li>
<li><p> 'BY' based on Benjamini &amp; Yekutieli, 2001.
</p>
</li></ul>

<p>The first four methods are designed to give strong control of the family-wise
error rate. The last two methods control the false discovery rate (FDR),
the expected proportion of false discoveries among the rejected hypotheses.
The false discovery rate is a less stringent condition than the family-wise
error rate, so these methods are more powerful than the others, especially
when number of tests is large.
The number of tests on which the adjustment for multiple comparisons is
the number of populations times the number of loci.
</p>
<p><b>From v2.1</b> <code>gl.filter.hwe</code> takes the argument
<code>n.pop.threshold</code>.
if <code>n.pop.threshold &gt; 1</code> loci will be removed only if they are 
concurrently 
significant (after adjustment if applied) out of hwe in &gt;= 
<code>n.pop.threshold &gt; 1</code>.
</p>


<h3>Value</h3>

<p>A genlight object with the loci departing significantly from H-W
proportions removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Benjamini, Y., and Yekutieli, D. (2001). The control of the false
discovery rate in multiple testing under dependency. Annals of Statistics,
29, 1165–1188.
</p>
</li>
<li><p> Graffelman, J. (2015). Exploring Diallelic Genetic Markers: The Hardy
Weinberg Package. Journal of Statistical Software 64:1-23.
</p>
</li>
<li><p> Graffelman, J. &amp; Morales-Camarena, J. (2008). Graphical tests for
Hardy-Weinberg equilibrium based on the ternary plot. Human Heredity 
65:77-84.
</p>
</li>
<li><p> Graffelman, J., &amp; Moreno, V. (2013). The mid p-value in exact tests for
Hardy-Weinberg equilibrium. Statistical applications in genetics and
molecular
biology, 12(4), 433-448.
</p>
</li>
<li><p> Hochberg, Y. (1988). A sharper Bonferroni procedure for multiple tests
of significance. Biometrika, 75, 800–803.
</p>
</li>
<li><p> Hommel, G. (1988). A stagewise rejective multiple test procedure based
on a modified Bonferroni test. Biometrika, 75, 383–386.
</p>
</li>
<li><p> Rice, W. R. (1989). Analyzing tables of statistical tests. Evolution,
43(1), 223-225.
</p>
</li>
<li><p> Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we
lost the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li>
<li><p> Wigginton, J.E., Cutler, D.J., &amp; Abecasis, G.R. (2005). A Note on Exact
Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics
76:887-893.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.report.hwe">gl.report.hwe</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.hwe(x = bandicoot.gl)
</code></pre>

<hr>
<h2 id='gl.filter.ld'>Filters loci based on linkage disequilibrium (LD)</h2><span id='topic+gl.filter.ld'></span>

<h3>Description</h3>

<p>This function uses the statistic set in the parameter <code>stat_keep</code> from 
function <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code> to choose the SNP to keep when two 
SNPs are in LD. When a SNP is selected to be filtered out in each pairwise 
comparison, the function stores its  name in a list. In subsequent pairwise
comparisons, if the SNP is already in the list, the other SNP will be kept.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.ld(
  x,
  ld_report,
  threshold = 0.2,
  pop.limit = ceiling(nPop(x)/2),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.ld_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_ld_report">ld_report</code></td>
<td>
<p>Output from function <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code> 
[required].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value above which loci will be removed
[default 0.2].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_pop.limit">pop.limit</code></td>
<td>
<p>Minimum number of populations in which LD should be more
than the threshold for a locus to be filtered out.
The default value is half of the populations [default ceiling(nPop(x)/2)].</p>
</td></tr>
<tr><td><code id="gl.filter.ld_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
test &lt;- bandicoot.gl
test &lt;- gl.filter.callrate(test,threshold = 1)
res &lt;- gl.report.ld.map(test)
res_2 &lt;- gl.filter.ld(x=test,ld_report = res)
res_3 &lt;- gl.report.ld.map(res_2)

## End(Not run)
if ((requireNamespace("snpStats", quietly = TRUE)) &amp; (requireNamespace("fields", quietly = TRUE))) {
test &lt;- gl.filter.callrate(platypus.gl, threshold = 1)
test &lt;- gl.filter.monomorphs(test)
test &lt;- test[,1:20]
report &lt;- gl.report.ld.map(test)
res &lt;- gl.filter.ld(x=test,ld_report = report)
}
</code></pre>

<hr>
<h2 id='gl.filter.locmetric'>Filters loci on the basis of numeric information stored in
other$loc.metrics in a genlight {adegenet} object</h2><span id='topic+gl.filter.locmetric'></span>

<h3>Description</h3>

<p>This script uses any field with numeric values stored in $other$loc.metrics
to filter loci. The loci to keep can be within the upper and lower thresholds
('within') or outside of the upper and lower thresholds ('outside').
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.locmetric(x, metric, upper, lower, keep = "within", verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.locmetric_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_metric">metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_upper">upper</code></td>
<td>
<p>Filter upper threshold [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_lower">lower</code></td>
<td>
<p>Filter lower threshold  [required].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_keep">keep</code></td>
<td>
<p>Whether keep loci within of upper and lower thresholds or keep
loci outside of upper and lower thresholds [within].</p>
</td></tr>
<tr><td><code id="gl.filter.locmetric_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own filter by adding a
vector into $other$loc.metrics as shown in the example.
</p>

<ol>
<li><p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li><p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li><p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li><p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li><p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li><p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li><p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li><p> PICRef - polymorphism information content (PIC) for the Reference
allele.
</p>
</li>
<li><p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li><p> AvgPIC - average of the polymorphism information content (PIC) of the
Reference and SNP alleles.
</p>
</li>
<li><p> AvgCountRef - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Reference allele
row.
</p>
</li>
<li><p> AvgCountSnp - sum of the tag read counts for all samples, divided by
the number of samples with non-zero tag read counts, for the Alternate (SNP)
allele row.
</p>
</li>
<li><p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li></ol>



<h3>Value</h3>

<p>The reduced genlight dataset.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># adding dummy data
test &lt;- testset.gl
test$other$loc.metrics$test &lt;- 1:nLoc(test)
result &lt;- gl.filter.locmetric(x=test, metric= 'test', upper=255,
lower=200, keep= 'within', verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.maf'>Filters loci on the basis of minor allele frequency (MAF) in a 
genlight
adegenet object</h2><span id='topic+gl.filter.maf'></span>

<h3>Description</h3>

<p>This script calculates the minor allele frequency for each locus and updates
the locus metadata for FreqHomRef, FreqHomSnp, FreqHets and MAF (if it
exists). It then uses the updated metadata for MAF to filter loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.maf(
  x,
  threshold = 0.01,
  by.pop = FALSE,
  pop.limit = ceiling(nPop(x)/2),
  ind.limit = 10,
  recalc = FALSE,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = discrete_palette,
  plot_colors_all = two_colors,
  bins = 25,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_threshold">threshold</code></td>
<td>
<p>Threshold MAF &ndash; loci with a MAF less than the threshold
will be removed. If a value &gt; 1 is provided it will be 
interpreted as MAC (i.e. the minimum number of times an allele needs to be 
observed) [default 0.01].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_by.pop">by.pop</code></td>
<td>
<p>Whether MAF should be calculated by population [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_pop.limit">pop.limit</code></td>
<td>
<p>Minimum number of populations in which MAF should be less 
than the threshold for a locus to be filtered out. Only used if by.pop=TRUE. 
The default value is half of the populations [default ceiling(nPop(x)/2)].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should 
contain to calculate MAF. Only used if by.pop=TRUE [default 10].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if histograms of call rate, before and after, are to
be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme for the plot [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>A color palette for population plots
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_plot_colors_all">plot_colors_all</code></td>
<td>
<p>List of two color names for the borders and fill of
the overall plot [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Careful consideration needs to be given to the settings to be used for this 
fucntion. When the filter is applied globally (i.e. <code>by.pop=FALSE</code>) but 
the data include multiple population, there is the risk to remove markers 
because the allele frequencies is low (at global level) but the allele 
frequencies
for the same markers may be high within some of the populations (especially 
if 
the per-population sample size is small). Similarly, not always it is a 
sensible choice to run this function using <code>by.pop=TRUE</code> because allele 
that are rare in a population may be very common in other, but the (possible) 
allele frequencies will depend on the sample size within each population. 
Where the purpose of filtering for MAF is to remove possible spurious alleles 
(i.e. sequencing errors), it is perhaps better to filter based on the number 
of times an allele is observed (MAC, Minimum Allele Count), under the 
assumption that if an allele is observed &gt;MAC, it is fairly rare to be an 
error.
<b>From v2.1</b> The threshold can take values &gt; 1. In this case, these are 
interpreted as a threshold for MAC.
</p>


<h3>Value</h3>

<p>The reduced genlight dataset
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.monomorphs(testset.gl)
result &lt;- gl.filter.maf(result, threshold=0.05, verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.monomorphs'>Filters monomorphic loci, including those with all NAs</h2><span id='topic+gl.filter.monomorphs'></span>

<h3>Description</h3>

<p>This script deletes monomorphic loci from a genlight {adegenet} object
</p>
<p>A DArT dataset will not have monomorphic loci, but they can arise, along with
loci that are scored all NA, when populations or individuals are deleted.
</p>
<p>Retaining monomorphic loci unnecessarily increases the size of the dataset
and will affect some calculations.
</p>
<p>Note that for SNP data, NAs likely represent null alleles; in tag
presence/absence data, NAs represent missing values (presence/absence could
not be reliably scored)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.monomorphs(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.monomorphs_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.monomorphs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with monomorphic (and all NA) loci removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  result &lt;- gl.filter.monomorphs(testset.gl, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.monomorphs(testset.gs, verbose=3)

</code></pre>

<hr>
<h2 id='gl.filter.overshoot'>Filters loci for which the SNP has been trimmed from the sequence tag
along with the adaptor</h2><span id='topic+gl.filter.overshoot'></span>

<h3>Description</h3>

<p>This function checks the position of the SNP within the trimmed sequence tag
and identifies those for which the SNP position is outside the trimmed
sequence tag. This can happen, rarely, when the sequence containing the SNP
resembles the adaptor.
</p>
<p>The SNP genotype can still be used in most analyses, but functions like
gl2fasta() will present challenges if the SNP has been trimmed from the
sequence tag.
</p>
<p>Not fatal, but should apply this filter before gl.filter.secondaries, for
obvious reasons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.overshoot(x, save2tmp = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.overshoot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.filter.overshoot_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.overshoot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new genlight object with the recalcitrant loci deleted
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.overshoot(testset.gl, verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.pa'>Filters loci that contain private (and fixed alleles) between two populations</h2><span id='topic+gl.filter.pa'></span>

<h3>Description</h3>

<p>This script is meant to be used prior to <code>gl.nhybrids</code> to maximise the
information content of the SNPs used to identify hybrids (currently
newhybrids does allow only 200 SNPs). The idea is to use first all loci that
have fixed alleles between the potential source populations and then 'fill
up' to 200 loci using loci that have private alleles between those. The
functions filters for those loci (if invers is set to TRUE, the opposite
is returned (all loci that are not fixed and have no private alleles - not
sure why yet, but maybe useful.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.pa(x, pop1, pop2, invers = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.pa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_pop1">pop1</code></td>
<td>
<p>Name of the first parental population (in quotes) [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_pop2">pop2</code></td>
<td>
<p>Name of the second parental population (in quotes) [required].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_invers">invers</code></td>
<td>
<p>Switch to filter for all loci that have no private alleles and
are not fixed [FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.pa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight dataset, containing now only fixed and private
alleles.
</p>


<h3>Author(s)</h3>

<p>Authors: Bernd Gruber &amp; Ella Kelly (University of Melbourne);
Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.filter.pa(testset.gl, pop1=pop(testset.gl)[1], 
pop2=pop(testset.gl)[2],verbose=3)
</code></pre>

<hr>
<h2 id='gl.filter.parent.offspring'>Filters putative parent offspring within a population</h2><span id='topic+gl.filter.parent.offspring'></span>

<h3>Description</h3>

<p>This script removes individuals suspected of being related as
parent-offspring,using the output of the function
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a></code>, which examines the frequency of
pedigree inconsistent loci, that is, those loci that are homozygotes in the
parent for the reference allele, and homozygous in the offspring for the
alternate allele. This condition is not consistent with any pedigree,
regardless of the (unknown) genotype of the other parent.
The pedigree inconsistent loci are counted as an indication of whether or not
it is reasonable to propose the two individuals are in a parent-offspring
relationship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.parent.offspring(
  x,
  min.rdepth = 12,
  min.reproducibility = 1,
  range = 1.5,
  method = "best",
  rm.monomorphs = FALSE,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.parent.offspring_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_min.rdepth">min.rdepth</code></td>
<td>
<p>Minimum read depth to include in analysis [default 12].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_min.reproducibility">min.reproducibility</code></td>
<td>
<p>Minimum reproducibility to include in analysis
[default 1].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_range">range</code></td>
<td>
<p>Specifies the range to extend beyond the interquartile range for
delimiting outliers [default 1.5 interquartile ranges].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_method">method</code></td>
<td>
<p>Method of selecting the individual to retain from each pair of
parent offspring relationship, 'best' (based on CallRate) or 'random'
[default 'best'].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_rm.monomorphs">rm.monomorphs</code></td>
<td>
<p>If TRUE, remove monomorphic loci after filtering
individuals [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.parent.offspring_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If two individuals are in a parent offspring relationship, the true number of
pedigree inconsistent loci should be zero, but SNP calling is not infallible.
Some loci will be miss-called. The problem thus becomes one of determining if
the two focal individuals have a count of pedigree inconsistent loci less
than would be expected of typical unrelated individuals. There are some quite
sophisticated software packages available to formally apply likelihoods to
the decision, but we use a simple outlier comparison.
</p>
<p>To reduce the frequency of miss-calls, and so emphasize the difference
between true parent-offspring pairs and unrelated pairs, the data can be
filtered on read depth. Typically minimum read depth is set to 5x, but you
can examine the distribution of read depths with the function
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a></code> and push this up with an acceptable loss of
loci. 12x might be a good minimum for this particular analysis. It is
sensible also to push the minimum reproducibility up to 1, if that does not
result in an unacceptable loss of loci. Reproducibility is stored in the slot
<code>@other$loc.metrics$RepAvg</code> and is defined as the proportion of
technical replicate assay pairs for which the marker score is consistent.
You can examine the distribution of reproducibility with the function
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>.
</p>
<p>Note that the null expectation is not well defined, and the power reduced, if
the population from which the putative parent-offspring pairs are drawn
contains many sibs. Note also that if an individual has been genotyped twice
in the dataset, the replicate pair will be assessed by this script as being
in a parent-offspring relationship.
</p>
<p>You should run <code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a></code> before filtering. Use
this report to decide min.rdepth and min.reproducibility and assess impact on
your dataset.
</p>
<p>Note that if your dataset does not contain RepAvg or rdepth among the locus
metrics, the filters for reproducibility and read depth are no used.
</p>
<p><strong> Function's output </strong>
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>the filtered genlight object without A set of individuals in
parent-offspring relationship. NULL if no parent-offspring relationships were
found.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.list.reports">gl.list.reports</a></code>, <code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a></code> ,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>,<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.filter.parent.offspring(testset.gl[1:10,1:50])

</code></pre>

<hr>
<h2 id='gl.filter.rdepth'>Filters loci based on counts of sequence tags scored at a locus (read
depth)</h2><span id='topic+gl.filter.rdepth'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT report AvgCountRef and AvgCountSnp as counts
of sequence tags for the reference and alternate alleles respectively. These
can be used to back calculate Read Depth. Fragment presence/absence datasets
as provided by DArT (SilicoDArT) provide Average Read Depth and Standard
Deviation of Read Depth as standard columns in their report.
</p>
<p>Filtering on Read Depth using the companion script gl.filter.rdepth can be on
the basis of loci with exceptionally low counts,
or loci with exceptionally high counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.rdepth(
  x,
  lower = 5,
  upper = 50,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.rdepth_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or tag
presence/absence data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_lower">lower</code></td>
<td>
<p>Lower threshold value below which loci will be removed
[default 5].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_upper">upper</code></td>
<td>
<p>Upper threshold value above which loci will be removed
[default 50].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.rdepth_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For examples of themes, see:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a genlight object retaining loci with a Read Depth in the
range specified by the lower and upper threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.rdepth(testset.gl)
  result &lt;- gl.filter.rdepth(testset.gl, lower=8, upper=50, verbose=3)
# Tag P/A data
  result &lt;- gl.filter.rdepth(testset.gs, lower=8, upper=50, verbose=3)
  
  res &lt;- gl.filter.rdepth(platypus.gl)

</code></pre>

<hr>
<h2 id='gl.filter.reproducibility'>Filters loci in a genlight {adegenet} object based on average
repeatability of alleles at a locus</h2><span id='topic+gl.filter.reproducibility'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have an index, RepAvg, generated by
reproducing the data independently for 30
of alleles that give a repeatable result, averaged over both alleles for each
locus.
</p>
<p>SilicoDArT datasets generated by DArT have a similar index, Reproducibility.
For these fragment presence/absence data, repeatability is the percentage of
scores that are repeated in the technical replicate dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.reproducibility(
  x,
  threshold = 0.99,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.reproducibility_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_threshold">threshold</code></td>
<td>
<p>Threshold value below which loci will be removed
[default 0.99].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, displays a plots of the distribution of
reproducibility values before and after filtering [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.filter.reproducibility_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object retaining loci with repeatability (Repavg
or Reproducibility) greater than the specified threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>
</p>
<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.reproducibility(testset.gl)
  result &lt;- gl.filter.reproducibility(testset.gl, threshold=0.99, verbose=3)
# Tag P/A data
  gl.report.reproducibility(testset.gs)
  result &lt;- gl.filter.reproducibility(testset.gs, threshold=0.99)
  
  
  test &lt;- gl.subsample.loci(platypus.gl,n=100)
  res &lt;- gl.filter.reproducibility(test)
  
</code></pre>

<hr>
<h2 id='gl.filter.secondaries'>Filters loci that represent secondary SNPs in a genlight object</h2><span id='topic+gl.filter.secondaries'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT include fragments with more than one SNP and
record them separately with the same CloneID (=AlleleID). These multiple SNP
loci within a fragment (secondaries) are likely to be linked, and so you may
wish to remove secondaries.
</p>
<p>This script filters out all but the first sequence tag with the same CloneID
after ordering the genlight object on based on repeatability, avgPIC in that
order (method='best') or at random (method='random').
</p>
<p>The filter has not been implemented for tag presence/absence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.secondaries(x, method = "random", verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.secondaries_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.secondaries_+3A_method">method</code></td>
<td>
<p>Method of selecting SNP locus to retain, 'best' or 'random'
[default 'random'].</p>
</td></tr>
<tr><td><code id="gl.filter.secondaries_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The genlight object, with the secondary SNP loci removed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.report.secondaries(testset.gl)
result &lt;- gl.filter.secondaries(testset.gl)
</code></pre>

<hr>
<h2 id='gl.filter.sexlinked'>Filters loci that are sex linked</h2><span id='topic+gl.filter.sexlinked'></span>

<h3>Description</h3>

<p>Alleles unique to the Y or W chromosome and monomorphic on the X chromosomes
will appear in the SNP dataset as genotypes that are heterozygotic in all
individuals of the heterogametic sex and homozygous in all individuals of the
homogametic sex. This function keeps or drops loci with alleles that behave
in this way, as putative sex specific SNP markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.sexlinked(
  x,
  sex = NULL,
  filter = NULL,
  read.depth = 0,
  t.het = 0.1,
  t.hom = 0.1,
  t.pres = 0.1,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = three_colors,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.sexlinked_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_sex">sex</code></td>
<td>
<p>Factor that defines the sex of individuals. See explanation in
details [default NULL].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_filter">filter</code></td>
<td>
<p>Either 'keep' to keep sex linked markers only or 'drop' to drop
sex linked markers [required].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_read.depth">read.depth</code></td>
<td>
<p>Additional filter option to keep only loci above a certain
read.depth. Default to 0, which means read.depth is not taken into account
[default 0].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_t.het">t.het</code></td>
<td>
<p>Tolerance in the heterogametic sex, that is t.het=0.05 means
that 5% of the heterogametic sex can be homozygous and still be regarded as
consistent with a sex specific marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_t.hom">t.hom</code></td>
<td>
<p>Tolerance in the homogametic sex, that is t.hom=0.05 means that
5% of the homogametic sex can be heterozygous and still be regarded as
consistent with a sex specific marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_t.pres">t.pres</code></td>
<td>
<p>Tolerance in presence, that is t.pres=0.05 means that a
silicodart marker can be present in either of the sexes and still be regarded
as a sex-linked marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_plot.out">plot.out</code></td>
<td>
<p>Creates a plot that shows the heterozygosity of males and
females at each loci be regarded as consistent with a sex specific marker 
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of three color names for the not sex-linked loci, for
the sex-linked loci and for the area in which sex-linked loci appear 
[default three_colors].</p>
</td></tr>
<tr><td><code id="gl.filter.sexlinked_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sex of the individuals for which sex is known with certainty can be provided
via a factor (equal to the length of the number of individuals) or to be held
in the variable <code>x@other$ind.metrics$sex</code>.
Coding is: M for male, F for female, U or NA for unknown/missing.
The script abbreviates the entries here to the first character. So, coding of
'Female' and 'Male' works as well. Character are also converted to upper 
cases.
</p>
<p>'<strong> Function's output </strong>
</p>
<p>This function creates also a plot that shows the heterozygosity of males and
females at each loci for SNP data or percentage of present/absent in the case 
of SilicoDArT data.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>The filtered genlight object (filter = 'keep': sex linked loci,
filter='drop', everything except sex linked loci).
</p>


<h3>Author(s)</h3>

<p>Arthur Georges, Bernd Gruber &amp; Floriaan Devloo-Delva (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
out &lt;- gl.filter.sexlinked(testset.gl, filter='drop')

out &lt;- gl.filter.sexlinked(testset.gs, filter='drop')

</code></pre>

<hr>
<h2 id='gl.filter.taglength'>Filters loci in a genlight {adegenet} object based on sequence tag
length</h2><span id='topic+gl.filter.taglength'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT typically have sequence tag lengths ranging
from 20 to 69 base pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.filter.taglength(x, lower = 20, upper = 69, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.filter.taglength_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_lower">lower</code></td>
<td>
<p>Lower threshold value below which loci will be removed
[default 20].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_upper">upper</code></td>
<td>
<p>Upper threshold value above which loci will be removed
[default 69].</p>
</td></tr>
<tr><td><code id="gl.filter.taglength_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object retaining loci with a sequence tag length
in the range specified by the lower and upper threshold.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other filter functions: 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a>()</code>,
<code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a>()</code>,
<code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a>()</code>,
<code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a>()</code>,
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a>()</code>,
<code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a>()</code>,
<code><a href="#topic+gl.filter.maf">gl.filter.maf</a>()</code>,
<code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a>()</code>,
<code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a>()</code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a>()</code>,
<code><a href="#topic+gl.filter.pa">gl.filter.pa</a>()</code>,
<code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a>()</code>,
<code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a>()</code>,
<code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a>()</code>,
<code><a href="#topic+gl.filter.sexlinked">gl.filter.sexlinked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.report.taglength(testset.gl)
  result &lt;- gl.filter.taglength(testset.gl,lower=60)
  gl.report.taglength(result)
# Tag P/A data
  gl.report.taglength(testset.gs)
  result &lt;- gl.filter.taglength(testset.gs,lower=60)
  gl.report.taglength(result)
  
  test &lt;- gl.subsample.loci(platypus.gl, n =100)
  res &lt;- gl.report.taglength(test)
</code></pre>

<hr>
<h2 id='gl.fixed.diff'>Generates a matrix of fixed differences and associated statistics for
populations taken pairwise</h2><span id='topic+gl.fixed.diff'></span>

<h3>Description</h3>

<p>This script takes SNP data or sequence tag P/A data grouped into populations
in a genlight object (DArTSeq) and generates a matrix of fixed differences
between populations taken pairwise
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fixed.diff(
  x,
  tloc = 0,
  test = FALSE,
  delta = 0.02,
  alpha = 0.05,
  reps = 1000,
  mono.rm = TRUE,
  pb = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fixed.diff_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes or tag P/A data
(SilicoDArT) or an object of class 'fd' [required].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_tloc">tloc</code></td>
<td>
<p>Threshold defining a fixed difference (e.g. 0.05 implies 95:5 vs
5:95 is fixed) [default 0].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_test">test</code></td>
<td>
<p>If TRUE, calculate p values for the observed fixed differences
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_delta">delta</code></td>
<td>
<p>Threshold value for the true population minor allele frequency
(MAF) from which resultant sample fixed differences are considered true
positives [default 0.02].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_alpha">alpha</code></td>
<td>
<p>Level of significance used to display non-significant
differences between populations as they are compared pairwise [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_reps">reps</code></td>
<td>
<p>Number of replications to undertake in the simulation to estimate
probability of false positives [default 1000].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, loci that are monomorphic across all individuals are
removed before beginning computations [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_pb">pb</code></td>
<td>
<p>If TRUE, show a progress bar on time consuming loops
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.fixed.diff_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A fixed difference at a locus occurs when two populations share no alleles or
where all members of one population has a sequence tag scored, and all
members of the other population has the sequence tag absent. The challenge
with this approach is that when sample sizes are finite, fixed differences
will occur through sampling error, compounded when many loci are examined.
Simulations suggest that sample sizes of n1=5 and n2=5 are adequate to reduce
the probability of [experiment-wide] type 1 error to negligible levels
[ploidy=2]. A warning is issued if comparison between two populations
involves sample sizes less than 5, taking into account allele drop-out.
</p>
<p>Optionally, if test=TRUE, the script will test the fixed differences between
final OTUs for statistical significance, using simulation, and then further
amalgamate populations that for which there are no significant fixed
differences at a specified level of significance (alpha). To avoid conflation
of true fixed differences with false positives in the simulations, it is
necessary to decide a threshold value (delta) for extreme true allele
frequencies that will be considered fixed for practical purposes. That is,
fixed differences in the sample set will be considered to be positives (not
false positives) if they arise from true allele frequencies of less than
1-delta in one or both populations.  The parameter delta is typically set to
be small (e.g. delta = 0.02).
</p>
<p>NOTE: The above test will only be calculated if tloc=0, that is, for analyses
of absolute fixed differences. The test applies in comparisons of allopatric
populations only. For sympatric populations, use gl.pval.sympatry().
</p>
<p>An absolute fixed difference is as defined above. However, one might wish to
score fixed differences at some lower level of allele frequency difference,
say where percent allele frequencies are 95,5 and 5,95 rather than 100:0 and
0:100. This adjustment can be done with the tloc parameter. For example,
tloc=0.05 means that SNP allele frequencies of 95,5 and 5,95 percent will be
regarded as fixed when comparing two populations at a locus.
</p>


<h3>Value</h3>

<p>A list of Class 'fd' containing the gl object and square matrices,
as follows:
</p>

<ol>
<li><p> $gl &ndash; the output genlight object;
</p>
</li>
<li><p> $fd &ndash; raw fixed differences;
</p>
</li>
<li><p> $pcfd &ndash; percent fixed differences;
</p>
</li>
<li><p> $nobs &ndash; mean no. of individuals used in each comparison;
</p>
</li>
<li><p> $nloc &ndash; total number of loci used in each comparison;
</p>
</li>
<li><p> $expfpos &ndash; if test=TRUE, the expected count of false positives
for each comparison [by simulation];
</p>
</li>
<li><p> $sdfpos &ndash; if test=TRUE, the standard deviation of the count of
false positives for each comparison [by simulation];
</p>
</li>
<li><p> $prob &ndash; if test=TRUE, the significance of the count of fixed
differences [by simulation])
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.fixed">is.fixed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fd &lt;- gl.fixed.diff(testset.gl, tloc=0, verbose=3 )
fd &lt;- gl.fixed.diff(testset.gl, tloc=0, test=TRUE, delta=0.02, reps=100, verbose=3 )

</code></pre>

<hr>
<h2 id='gl.fst.pop'>Calculates a pairwise Fst values for populations in a genlight object</h2><span id='topic+gl.fst.pop'></span>

<h3>Description</h3>

<p>This script calculates pairwise Fst values based on the implementation in the
StAMPP package (?stamppFst). It allows to run bootstrap to estimate
probability of Fst values to be different from zero. For detailed information
please check the help pages (?stamppFst).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.fst.pop(x, nboots = 1, percent = 95, nclusters = 1, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.fst.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_nboots">nboots</code></td>
<td>
<p>Number of bootstraps to perform across loci to generate
confidence intervals and p-values [default 1].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_percent">percent</code></td>
<td>
<p>Percentile to calculate the confidence interval around
[default 95].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_nclusters">nclusters</code></td>
<td>
<p>Number of processor threads or cores to use during
calculations [default 1].</p>
</td></tr>
<tr><td><code id="gl.fst.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of distances between populations (class dist), if nboots =1,
otherwise a list with Fsts (in a matrix), Pvalues (a matrix of pvalues),
Bootstraps results (data frame of all runs). Hint: Use
<code>as.matrix(as.dist(fsts))</code> if you want to have a squared matrix with
symmetric entries returned, instead of a dist object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
test &lt;- gl.filter.monomorphs(test)
out &lt;- gl.fst.pop(test, nboots=1)
</code></pre>

<hr>
<h2 id='gl.genleastcost'>Performs least-cost path analysis based on a friction matrix</h2><span id='topic+gl.genleastcost'></span>

<h3>Description</h3>

<p>This function calculates the pairwise distances (Euclidean, cost path
distances and genetic distances) of populations using a friction matrix and
a spatial genind object. The genind object needs to have coordinates in the
same projected coordinate system as the friction matrix. The friction
matrix can be either a single raster of a stack of several layers. If a
stack is provided the specified cost distance is calculated for each layer
in the stack. The output of this function can be used with the functions
<code><a href="PopGenReport.html#topic+wassermann">wassermann</a></code> or
<code><a href="PopGenReport.html#topic+lgrMMRR">lgrMMRR</a></code> to test for the significance of a
layer on the genetic structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.genleastcost(
  x,
  fric.raster,
  gen.distance,
  NN = NULL,
  pathtype = "leastcost",
  plotpath = TRUE,
  theta = 1,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.genleastcost_+3A_x">x</code></td>
<td>
<p>A spatial genind object. See ?popgenreport how to provide
coordinates in genind objects [required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_fric.raster">fric.raster</code></td>
<td>
<p>A friction matrix [required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_gen.distance">gen.distance</code></td>
<td>
<p>Specification which genetic distance method should be
used to calculate pairwise genetic distances between populations ( 'D',
'Gst.Nei', 'Gst.Hedrick') or individuals ('Smouse', 'Kosman', 'propShared')
[required].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_nn">NN</code></td>
<td>
<p>Number of neighbours used when calculating the cost distance
(possible values 4, 8 or 16). As the default is NULL a value has to be
provided if pathtype='leastcost'. NN=8 is most commonly used. Be aware that
linear structures may cause artefacts in the least-cost paths, therefore
inspect the actual least-cost paths in the provided output [default NULL].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_pathtype">pathtype</code></td>
<td>
<p>Type of cost distance to be calculated (based on function in
the <code>gdistance</code> package. Available distances are 'leastcost', 'commute'
or 'rSPDistance'. See functions in the gdistance package for futher
explanations. If the path type is set to 'leastcost' then paths and also
pathlength are returned [default 'leastcost'].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_plotpath">plotpath</code></td>
<td>
<p>switch if least cost paths should be plotted (works only if
pathtype='leastcost'. Be aware this slows down the computation, but it is
recommended to do this to check least cost paths visually.</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_theta">theta</code></td>
<td>
<p>value needed for rSPDistance function. See
<code><a href="gdistance.html#topic+rSPDistance">rSPDistance</a></code> in package <code>gdistance</code> [default 1].</p>
</td></tr>
<tr><td><code id="gl.genleastcost_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list that consists of four pairwise distance matrices
(Euclidean, Cost, length of path and genetic) and the actual paths as spatial
line objects.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Cushman, S., Wasserman, T., Landguth, E. and Shirk, A. (2013).
Re-Evaluating Causal Modeling with Mantel Tests in Landscape Genetics.
Diversity, 5(1), 51-72.
</p>
</li>
<li><p> Landguth, E. L., Cushman, S. A., Schwartz, M. K., McKelvey, K. S.,
Murphy, M. and Luikart, G. (2010). Quantifying the lag time to detect
barriers in landscape genetics. Molecular ecology, 4179-4191.
</p>
</li>
<li><p> Wasserman, T. N., Cushman, S. A., Schwartz, M. K. and Wallin, D. O.
(2010). Spatial scaling and multi-model inference in landscape genetics:
Martes americana in northern Idaho. Landscape Ecology, 25(10), 1601-1612.
</p>
</li></ul>



<h3>See Also</h3>

<p><code>landgenreport</code>, <code>popgenreport</code>,
<code>wassermann</code>, <code>lgrMMRR</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(possums.gl)
library(raster)  #needed for that example
landscape.sim &lt;- readRDS(system.file('extdata','landscape.sim.rdata', 
package='dartR'))
glc &lt;- gl.genleastcost(x=possums.gl,fric.raster=landscape.sim ,
gen.distance = 'D', NN=8, pathtype = 'leastcost',plotpath = TRUE)
library(PopGenReport)
PopGenReport::wassermann(eucl.mat = glc$eucl.mat, cost.mat = glc$cost.mats, 
 gen.mat = glc$gen.mat)
lgrMMRR(gen.mat = glc$gen.mat, cost.mats = glc$cost.mats,  
eucl.mat = glc$eucl.mat)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.grm'>Calculates an identity by descent matrix</h2><span id='topic+gl.grm'></span>

<h3>Description</h3>

<p>This function calculates the mean probability of identity by state (IBS)
across loci that would result from all the possible crosses of the
individuals analyzed. IBD is calculated by an additive relationship matrix
approach developed by Endelman and Jannink (2012) as implemented in the
function <a href="rrBLUP.html#topic+A.mat">A.mat</a> (package rrBLUP).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.grm(
  x,
  plotheatmap = TRUE,
  palette_discrete = discrete_palette,
  palette_convergent = convergent_palette,
  legendx = 0,
  legendy = 0.5,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.grm_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_plotheatmap">plotheatmap</code></td>
<td>
<p>A switch if a heatmap should be shown [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_palette_discrete">palette_discrete</code></td>
<td>
<p>A discrete palette for the color of populations or a
list with as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_palette_convergent">palette_convergent</code></td>
<td>
<p>A convergent palette for the IBD values
[default convergent_palette].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_legendx">legendx</code></td>
<td>
<p>x coordinates for the legend[default 0].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_legendy">legendy</code></td>
<td>
<p>y coordinates for the legend[default 1].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.grm_+3A_...">...</code></td>
<td>
<p>Parameters passed to function A.mat from package rrBLUP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two or more alleles are identical by descent (IBD) if they are identical
copies of the same ancestral allele in a base population. The additive
relationship matrix is a theoretical framework for estimating a relationship
matrix that is consistent with an approach to estimate the probability that
the alleles at a random locus are identical in state (IBS).
</p>
<p>This function also plots a heatmap, and a dendrogram, of IBD values where
each diagonal element has a mean that equals 1+f, where f is the inbreeding
coefficient (i.e. the probability that the two alleles at a randomly chosen
locus are IBD from the base population). As this probability lies between 0
and 1, the diagonal elements range from 1 to 2. Because the inbreeding
coefficients are expressed relative to the current population, the mean of
the off-diagonal elements is -(1+f)/n, where n is the number of loci.
Individual names are shown in the margins of the heatmap and colors
represent different populations.
</p>


<h3>Value</h3>

<p>An identity by descent matrix
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Endelman, J. B. (2011). Ridge regression and other kernels for genomic
selection with r package rrblup. The Plant Genome 4, 250.
</p>
</li>
<li><p> Endelman, J. B. , Jannink, J.-L. (2012). Shrinkage estimation of the
realized relationship matrix. G3: Genes, Genomics, Genetics 2, 1405.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.grm.network">gl.grm.network</a></code>
</p>
<p>Other inbreeding functions: 
<code><a href="#topic+gl.grm.network">gl.grm.network</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.grm(platypus.gl[1:10,1:100])

</code></pre>

<hr>
<h2 id='gl.grm.network'>Represents a genomic relationship matrix (GRM) as a network</h2><span id='topic+gl.grm.network'></span>

<h3>Description</h3>

<p>This script takes a G matrix generated by <code><a href="#topic+gl.grm">gl.grm</a></code> and represents
the relationship among the specimens as a network diagram. In order to use
this script, a decision is required on a threshold for relatedness to be
represented as link in the network, and on the layout used to create the
diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.grm.network(
  G,
  x,
  method = "fr",
  node.size = 8,
  node.label = TRUE,
  node.label.size = 2,
  node.label.color = "black",
  link.color = NULL,
  link.size = 2,
  relatedness_factor = 0.125,
  title = "Network based on a genomic relationship matrix",
  palette_discrete = NULL,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.grm.network_+3A_g">G</code></td>
<td>
<p>A genomic relationship matrix (GRM) generated by
<code><a href="#topic+gl.grm">gl.grm</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_x">x</code></td>
<td>
<p>A genlight object from which the G matrix was generated [required].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_method">method</code></td>
<td>
<p>One of 'fr', 'kk', 'gh' or 'mds' [default 'fr'].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_node.size">node.size</code></td>
<td>
<p>Size of the symbols for the network nodes [default 8].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_node.label">node.label</code></td>
<td>
<p>TRUE to display node labels [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_node.label.size">node.label.size</code></td>
<td>
<p>Size of the node labels [default 3].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_node.label.color">node.label.color</code></td>
<td>
<p>Color of the text of the node labels
[default 'black'].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_link.color">link.color</code></td>
<td>
<p>Color palette for links [default NULL].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_link.size">link.size</code></td>
<td>
<p>Size of the links [default 2].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_relatedness_factor">relatedness_factor</code></td>
<td>
<p>Factor of relatedness [default 0.125].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_title">title</code></td>
<td>
<p>Title for the plot
[default 'Network based on genomic relationship matrix'].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_palette_discrete">palette_discrete</code></td>
<td>
<p>A discrete palette for the color of populations or a
list with as many colors as there are populations in the dataset
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.grm.network_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gl.grm.network function takes a genomic relationship matrix (GRM) 
generated by the gl.grm function to represent the relationship among 
individuals in the dataset as a network diagram. To generate the GRM, the 
function gl.grm uses the function A.mat from package rrBLUP, which implements
the approach developed by Endelman and Jannink (2012).
</p>
<p>The GRM is an estimate of the proportion of alleles that two individuals have
in common. It is generated by estimating the covariance of the genotypes 
between two individuals, i.e. how much genotypes in the two individuals
correspond with each other. This covariance depends on the probability that
alleles at a random locus are identical by state (IBS). Two alleles are 
IBS if they represent the same allele. Two alleles are identical by 
descent (IBD) if one is a physical copy of the other or if they are both 
physical copies of the same ancestral allele. Note that IBD is complicated
to determine. IBD implies IBS, but not conversely. However, as the number
of SNPs in a dataset increases, the mean probability of IBS approaches 
the mean probability of IBD.
</p>
<p>It follows that the off-diagonal elements of the GRM are two times the 
kinship coefficient, i.e. the probability that two alleles at a random locus
drawn from two individuals are IBD. Additionally, the diagonal elements of
the GRM are 1+f, where f is the inbreeding coefficient of each individual,
i.e. the probability that the two alleles at a random locus are IBD.
</p>
<p>Choosing a meaningful threshold to represent the relationship between 
individuals is tricky because IBD is not an absolute state but is relative to
a reference population for which there is generally little information so 
that we can estimate the kinship of a pair of individuals only relative to 
some other quantity. To deal with this, we can use the average inbreeding 
coefficient of the diagonal elements as the reference value. For this, the 
function subtracts 1 from the mean of the diagonal elements of the GRM. In a
second step, the off-diagonal elements are divided by 2, and finally, the 
mean of the diagonal elements is subtracted from each off-diagonal element 
after dividing them by 2. This approach is similar to the one used by 
Goudet et al. (2018).
</p>
<p>Below is a table modified from Speed &amp; Balding (2015) showing kinship values,
and their confidence intervals (CI), for different relationships that could 
be used to guide the choosing of the relatedness threshold in the function.
</p>
<p>|Relationship                               |Kinship  |     95
</p>
<p>|Identical twins/clones/same individual     | 0.5     |        -         |
</p>
<p>|Sibling/Parent-Offspring                   | 0.25    |    (0.204, 0.296)|
</p>
<p>|Half-sibling                               | 0.125   |    (0.092, 0.158)|
</p>
<p>|First cousin                               | 0.062   |    (0.038, 0.089)|
</p>
<p>|Half-cousin                                | 0.031   |    (0.012, 0.055)|
</p>
<p>|Second cousin                              | 0.016   |    (0.004, 0.031)|
</p>
<p>|Half-second cousin                         | 0.008   |    (0.001, 0.020)|
</p>
<p>|Third cousin                               | 0.004   |    (0.000, 0.012)|
</p>
<p>|Unrelated                                  | 0       |        -         | 
</p>
<p>Four layout options are implemented in this function:
</p>

<ul>
<li><p> 'fr' Fruchterman-Reingold layout  <a href="igraph.html#topic+layout_with_fr">layout_with_fr</a>
(package igraph)
</p>
</li>
<li><p> 'kk' Kamada-Kawai layout <a href="igraph.html#topic+layout_with_kk">layout_with_kk</a> (package igraph)
</p>
</li>
<li><p> 'gh' Graphopt layout <a href="igraph.html#topic+layout_with_graphopt">layout_with_graphopt</a>
(package igraph)
</p>
</li>
<li><p> 'mds' Multidimensional scaling layout <a href="igraph.html#topic+layout_with_mds">layout_with_mds</a>
(package igraph)
</p>
</li></ul>



<h3>Value</h3>

<p>A network plot showing relatedness between individuals
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Endelman, J. B. , Jannink, J.-L. (2012). Shrinkage estimation of the 
realized relationship matrix. G3: Genes, Genomics, Genetics 2, 1405.
</p>
</li>
<li><p> Goudet, J., Kay, T., &amp; Weir, B. S. (2018). How to estimate kinship.
Molecular Ecology, 27(20), 4121-4135.
</p>
</li>
<li><p> Speed, D., &amp; Balding, D. J. (2015). Relatedness in the post-genomic era: 
is it still useful?. Nature Reviews Genetics, 16(1), 33-44.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.grm">gl.grm</a></code>
</p>
<p>Other inbreeding functions: 
<code><a href="#topic+gl.grm">gl.grm</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("igraph", quietly = TRUE) &amp; requireNamespace("rrBLUP",
quietly = TRUE) &amp; requireNamespace("fields", quietly=TRUE)) {
t1 &lt;- possums.gl
# filtering on call rate 
t1 &lt;- gl.filter.callrate(t1)
t1 &lt;- gl.subsample.loci(t1,n = 100)
# relatedness matrix
res &lt;- gl.grm(t1,plotheatmap = FALSE)
# relatedness network
res2 &lt;- gl.grm.network(res,t1,relatedness_factor = 0.125)
}
</code></pre>

<hr>
<h2 id='gl.He'>Estimates expected Heterozygosity</h2><span id='topic+gl.He'></span>

<h3>Description</h3>

<p>Estimates expected Heterozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.He(gl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.He_+3A_gl">gl</code></td>
<td>
<p>A genlight object [required]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple vector whit Ho for each loci
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.Ho'>Estimates observed Heterozygosity</h2><span id='topic+gl.Ho'></span>

<h3>Description</h3>

<p>Estimates observed Heterozygosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.Ho(gl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.Ho_+3A_gl">gl</code></td>
<td>
<p>A genlight object [required]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A simple vector whit Ho for each loci
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.hwe.pop'>Performs Hardy-Weinberg tests over loci and populations</h2><span id='topic+gl.hwe.pop'></span>

<h3>Description</h3>

<p>Hardy-Weinberg tests are performed for each loci in each of the populations
as defined by the pop slot in a genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.hwe.pop(
  x,
  alpha_val = 0.05,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = c("gray90", "deeppink"),
  HWformat = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.hwe.pop_+3A_x">x</code></td>
<td>
<p>A genlight object with a population defined
[pop(x) does not return NULL].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, returns a plot object compatible with ggplot,
otherwise returns a dataframe [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two_colors].
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_hwformat">HWformat</code></td>
<td>
<p>Switch if data should be returned in HWformat (counts of
Genotypes to be used in package <code>HardyWeinberg</code>)</p>
</td></tr>
<tr><td><code id="gl.hwe.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function employs the <code>HardyWeinberg</code> package, which needs to be
installed. The function that is used is
<code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code>, but there are several other great
functions implemented in the package regarding HWE. Therefore, this function
can return the data in the format expected by the HWE package expects, via
<code>HWformat=TRUE</code> and then use this to run other functions of the package.
</p>
<p>This functions performs a HWE test for every population (rows) and loci
(columns) and returns a true false matrix. True is reported if the p-value of
an HWE-test for a particular loci and population was below the specified
threshold (alpha_val, default=0.05). The thinking behind this approach is
that loci that are not in HWE in several populations have most likely to be
treated (e.g. filtered if loci under selection are of interest). If plot=TRUE
a barplot on the loci and the sum of deviation over all population is
returned. Loci that deviate in the majority of populations can be identified
via colSums on the resulting matrix.
</p>
<p>Plot themes can be obtained from </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Resultant ggplots and the tabulation are saved to the session's temporary
directory.
</p>


<h3>Value</h3>

<p>The function returns a list with up to three components:
</p>

<ul>
<li><p> 'HWE' is the matrix over loci and populations
</p>
</li>
<li><p> 'plot' is a plot (ggplot) which shows the significant results
for population and loci (can be amended further using ggplot syntax)
</p>
</li>
<li><p> 'HWEformat=TRUE' the 'HWformat' entails SNP data for each population
in 'HardyWeinberg'-format to be used with other functions of the package
(e.g <code><a href="HardyWeinberg.html#topic+HWPerm">HWPerm</a></code> or
<code><a href="HardyWeinberg.html#topic+HWExactPrevious">HWExactPrevious</a></code>).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.hwe.pop(bandicoot.gl[,1:33], alpha_val=0.05, plot.out=TRUE, HWformat=FALSE)
</code></pre>

<hr>
<h2 id='gl.ibd'>Performs isolation by distance analysis</h2><span id='topic+gl.ibd'></span>

<h3>Description</h3>

<p>This function performs an isolation by distance analysis based on a Mantel
test and also produces an isolation by distance plot. If a genlight object
with coordinates is provided, then an Euclidean and genetic distance matrices
are calculated.'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ibd(
  x = NULL,
  distance = "Fst",
  coordinates = "latlon",
  Dgen = NULL,
  Dgeo = NULL,
  Dgeo_trans = "Dgeo",
  Dgen_trans = "Dgen",
  permutations = 999,
  plot.out = TRUE,
  paircols = NULL,
  plot_theme = theme_dartR(),
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ibd_+3A_x">x</code></td>
<td>
<p>Genlight object. If provided a standard analysis on Fst/1-Fst and
log(distance) is performed [required].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_distance">distance</code></td>
<td>
<p>Type of distance that is calculated and used for the
analysis. Can be either population based 'Fst' [<a href="StAMPP.html#topic+stamppFst">stamppFst</a>],
'D' [<a href="StAMPP.html#topic+stamppNeisD">stamppNeisD</a>] or individual based 'propShared',
[gl.propShared], 'euclidean' [gl.dist.ind, method='Euclidean']
[default &quot;Fst&quot;].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_coordinates">coordinates</code></td>
<td>
<p>Can be either 'latlon', 'xy' or a two column data.frame
with column names 'lat','lon', 'x', 'y'). Coordinates are provided via
<code>gl@other$latlon</code> ['latlon'] or via <code>gl@other$xy</code> ['xy']. If latlon
data will be projected to meters using Mercator system [google maps] or if
xy then distance is directly calculated on the coordinates.</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgen">Dgen</code></td>
<td>
<p>Genetic distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgeo">Dgeo</code></td>
<td>
<p>Euclidean distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgeo_trans">Dgeo_trans</code></td>
<td>
<p>Transformation to be used on the Euclidean distances. See
Dgen_trans [default &quot;Dgeo&quot;].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_dgen_trans">Dgen_trans</code></td>
<td>
<p>You can provide a formula to transform the genetic
distance. The transformation can be applied as a formula using Dgen as the
variable to be transformed. For example: <code>Dgen_trans = 'Dgen/(1-Dgen)'.
  Any valid R expression can be used here
   [default 'Dgen', which is the identity function.]</code></p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_permutations">permutations</code></td>
<td>
<p>Number of permutations in the Mantel test [default 999].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot.out">plot.out</code></td>
<td>
<p>Should an isolation by distance plot be returned
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_paircols">paircols</code></td>
<td>
<p>Should pairwise dots colored by 'pop'ulation/'ind'ividual
pairs [default 'pop']. You can color pairwise individuals by pairwise
population colors.</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.ibd_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently pairwise Fst and D between populations and
1-propShared and Euclidean distance between individuals are
implemented. Coordinates are expected as lat long and converted to Google
Earth Mercator projection. If coordinates are already projected, provide them
at the x@other$xy slot.
</p>
<p>You can provide also your own genetic and Euclidean distance matrices. The
function is based on the code provided by the adegenet tutorial
(<a href="http://adegenet.r-forge.r-project.org/files/tutorial-basics.pdf">http://adegenet.r-forge.r-project.org/files/tutorial-basics.pdf</a>),
using the functions  <a href="vegan.html#topic+mantel">mantel</a> (package vegan),
<a href="StAMPP.html#topic+stamppFst">stamppFst</a>, <a href="StAMPP.html#topic+stamppNeisD">stamppNeisD</a> (package StAMPP) and
gl.propShared or gl.dist.ind. For transformation you need to have the dismo
package installed. As a new feature you can plot pairwise relationship using
double colored points (paircols=TRUE). Pairwise relationship can be
visualised via populations or individuals, depending which distance is
calculated. Please note: Often a problem arises, if an individual based 
distance is calculated (e.g. propShared) and some individuals have identical
coordinates as this results in distances of zero between those pairs of 
individuals.
</p>
<p>If the standard transformation [log(Dgeo)] is used, this results in an 
infinite value, because of trying to calculate'log(0)'. To avoid this, the 
easiest fix is to change the transformation from log(Dgeo) to log(Dgeo+1) or 
you could add some &quot;noise&quot; to the coordinates of the individuals (e.g. +- 1m,
but be aware if you use lat lon then you rather want to add +0.00001 degrees
or so).
</p>


<h3>Value</h3>

<p>Returns a list of the following components: Dgen (the genetic
distance matrix), Dgeo (the Euclidean distance matrix), Mantel (the
statistics of the Mantel test).
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Rousset, F. (1997). Genetic differentiation and estimation of gene flow from
F-statistics under isolation by distance. Genetics, 145(4), 1219-1228.
</p>


<h3>See Also</h3>

<p><a href="vegan.html#topic+mantel">mantel</a>, <a href="StAMPP.html#topic+stamppFst">stamppFst</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
#because of speed only the first 100 loci
ibd &lt;- gl.ibd(bandicoot.gl[,1:100], Dgeo_trans='log(Dgeo)' ,
Dgen_trans='Dgen/(1-Dgen)')
#because of speed only the first 10 individuals)
ibd &lt;- gl.ibd(bandicoot.gl[1:10,], distance='euclidean', paircols='pop', 
Dgeo_trans='Dgeo')

#only first 100 loci
ibd &lt;- gl.ibd(bandicoot.gl[,1:100])
</code></pre>

<hr>
<h2 id='gl.impute'>Imputates missing data</h2><span id='topic+gl.impute'></span>

<h3>Description</h3>

<p>This function imputes genotypes on a population-by-population basis, where
populations can be considered panmictic, or imputes the state for
presence-absence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.impute(
  x,
  method = "neighbour",
  fill.residual = TRUE,
  parallel = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.impute_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence-absence
data [required].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_method">method</code></td>
<td>
<p>Imputation method, either &quot;frequency&quot; or &quot;HW&quot; or &quot;neighbour&quot; 
or &quot;random&quot; [default &quot;neighbour&quot;].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_fill.residual">fill.residual</code></td>
<td>
<p>Should any residual missing values remaining after 
imputation be set to 0, 1, 2 at random, taking into account global allele 
frequencies at the particular locus [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_parallel">parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.impute_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We recommend that imputation be performed on sampling locations, before
any aggregation. Imputation is achieved by replacing missing values using
either of two methods:
</p>

<ul>
<li><p> If &quot;frequency&quot;, genotypes scored as missing at a locus in an individual
are imputed using the average allele frequencies at that locus in the 
population from which the individual was drawn.
</p>
</li>
<li><p> If &quot;HW&quot;, genotypes scored as missing at a locus in an individual are 
imputed by sampling at random assuming Hardy-Weinberg equilibrium. Applies 
only to genotype data.
</p>
</li>
<li><p> If &quot;neighbour&quot;, substitute the missing values for the focal individual
with the values taken from the nearest neighbour. Repeat with next nearest
and so on until all missing values are replaced.
</p>
</li>
<li><p> if &quot;random&quot;, missing data are substituted by random values (0, 1 or 2). 
</p>
</li></ul>

<p>The nearest neighbour is the one with the smallest Euclidean distance in 
all the dataset.
</p>
<p>The advantage of this approach is that it works regardless of how many
individuals are in the population to which the focal individual belongs,
and the displacement of the individual is haphazard as opposed to:
</p>
<p>(a) Drawing the individual toward the population centroid (HW and Frequency).
</p>
<p>(b) Drawing the individual toward the global centroid (glPCA).
</p>
<p>Note that loci that are missing for all individuals in a population are not 
imputed with method 'frequency' or 'HW'. Consider using the function 
<code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> with by.pop=TRUE to remove them first.
</p>


<h3>Value</h3>

<p>A genlight object with the missing data imputed.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
require("dartR.data")
# SNP genotype data
gl &lt;- gl.filter.callrate(platypus.gl,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gl &lt;- gl.impute(gl,method="neighbour")
# Sequence Tag presence-absence data
gs &lt;- gl.filter.callrate(testset.gs,threshold=0.95)
gl &lt;- gl.filter.allna(gl)
gs &lt;- gl.impute(gs, method="neighbour")

gs &lt;- gl.impute(platypus.gl,method ="random")
</code></pre>

<hr>
<h2 id='gl.install.vanilla.dartR'>Installs all required packages for using all functions
available in dartR</h2><span id='topic+gl.install.vanilla.dartR'></span>

<h3>Description</h3>

<p>The function compares the installed packages with the the currently available
ones on CRAN. Be aware this function only works if a version of dartR is
already installed on your system. You can choose if you also want to have a
specific version of dartR installed ('CRAN', 'master', 'beta' or 'dev' ). 'master'
, 'beta' and 'dev' are installed from Github. Be aware that the dev version from github is
not fully tested and most certainly will contain untested functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.install.vanilla.dartR(flavour = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.install.vanilla.dartR_+3A_flavour">flavour</code></td>
<td>
<p>The version of R you want to install. If NULL
then only packages needed for the current version will be installed. If
'CRAN' current CRAN version will be installed. 'master' installs the GitHub
master branch, 'beta' installs the latest stable version, and 'dev' installs 
the experimental development branch from
GitHub [default NULL].</p>
</td></tr>
<tr><td><code id="gl.install.vanilla.dartR_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a message if the installation was successful/required.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.join'>Combines two genlight objects</h2><span id='topic+gl.join'></span>

<h3>Description</h3>

<p>This function combines two genlight objects and their associated metadata.
The history associated with the two genlight objects is cleared from the new
genlight object. The individuals/samples must be the same in each genlight
object.
</p>
<p>The function is typically used to combine datasets from the same service
where the files have been split because of size limitations. The data is read
in from multiple csv files, then the resultant genlight objects are combined.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.join(x1, x2, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.join_+3A_x1">x1</code></td>
<td>
<p>Name of the first genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.join_+3A_x2">x2</code></td>
<td>
<p>Name of the first genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.join_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 &lt;- testset.gl[,1:100]
x1@other$loc.metrics &lt;-  testset.gl@other$loc.metrics[1:100,]
nLoc(x1)
x2 &lt;- testset.gl[,101:150]
x2@other$loc.metrics &lt;-  testset.gl@other$loc.metrics[101:150,]
nLoc(x2)
gl &lt;- gl.join(x1, x2, verbose=2)
nLoc(gl)
</code></pre>

<hr>
<h2 id='gl.keep.ind'>Removes all but the specified individuals from a genlight object</h2><span id='topic+gl.keep.ind'></span>

<h3>Description</h3>

<p>The script, having deleted individuals, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally recalculates
statistics made redundant by the deletion of individuals from the dataset.
</p>
<p>The script returns a genlight object with the individuals deleted and,
optionally, the recalculated locus metadata.
</p>
<p>See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.ind(x, ind.list, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_ind.list">ind.list</code></td>
<td>
<p>A list of individuals to be removed [required].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.ind">gl.drop.ind</a></code> to drop rather than keep specified
individuals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # SNP data
    gl2 &lt;- gl.keep.ind(testset.gl, ind.list=c('AA019073','AA004859'))
  # Tag P/A data
   gs2 &lt;- gl.keep.ind(testset.gs, ind.list=c('AA020656','AA19077','AA004859'))
</code></pre>

<hr>
<h2 id='gl.keep.loc'>Removes all but the specified loci from a genlight object</h2><span id='topic+gl.keep.loc'></span>

<h3>Description</h3>

<p>The script returns a genlight object with the all but the specified loci
deleted.
</p>
<p>#' See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.loc(x, loc.list = NULL, first = NULL, last = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.loc_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes or
presence/absence data [required].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_loc.list">loc.list</code></td>
<td>
<p>A list of loci to be kept
[required, if loc.range not specified].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_first">first</code></td>
<td>
<p>First of a range of loci to be kept
[required, if loc.list not specified].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_last">last</code></td>
<td>
<p>Last of a range of loci to be kept
[if not specified, last locus in the dataset].</p>
</td></tr>
<tr><td><code id="gl.keep.loc_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.loc">gl.drop.loc</a></code> to drop rather than keep specified loci
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl2 &lt;- gl.keep.loc(testset.gl, loc.list=c('100051468|42-A/T', '100049816-51-A/G'))
# Tag P/A data
  gs2 &lt;- gl.keep.loc(testset.gs, loc.list=c('20134188','19249144'))

</code></pre>

<hr>
<h2 id='gl.keep.pop'>Removes all but the specified populations from a genlight object</h2><span id='topic+gl.keep.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart().
</p>
<p>The script, having deleted the specified populations, optionally identifies
resultant monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally recalculates
statistics made redundant by the deletion of individuals from the dataset.
</p>
<p>The script returns a genlight object with the new population assignments and
the recalculated locus metadata.
</p>
<p>#' See more about data manipulation in the [tutorial](http://georges.biomatix.org/storage/app/media/uploaded-files/tutorial4dartrdatamanipulation22-dec-21-3.pdf).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.keep.pop(
  x,
  pop.list,
  as.pop = NULL,
  recalc = FALSE,
  mono.rm = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.keep.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_pop.list">pop.list</code></td>
<td>
<p>A list of populations to be kept [required].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Assign another metric to represent population [default NULL].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.keep.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reduced data
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.drop.pop">gl.drop.pop</a></code> to drop rather than keep specified populations
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # SNP data
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('EmsubRopeMata', 'EmvicVictJasp'))
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('EmsubRopeMata', 'EmvicVictJasp'),
   mono.rm=TRUE,recalc=TRUE)
   gl2 &lt;- gl.keep.pop(testset.gl, pop.list=c('Female'),as.pop='sex')
 # Tag P/A data
   gs2 &lt;- gl.keep.pop(testset.gs, pop.list=c('EmsubRopeMata','EmvicVictJasp'))

</code></pre>

<hr>
<h2 id='gl.ld.distance'>Plots linkage disequilibrium against distance by population 
disequilibrium patterns</h2><span id='topic+gl.ld.distance'></span>

<h3>Description</h3>

<p>The function creates a plot showing
the pairwise LD measure against distance in number of base pairs pooled over
all the chromosomes and a red line representing the threshold (R.squared = 
0.2) that is commonly used to imply that two loci are unlinked (Delourme et
al., 2013; Li et al., 2014).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ld.distance(
  ld_report,
  ld_resolution = 1e+05,
  pop_colors = NULL,
  plot_theme = NULL,
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ld.distance_+3A_ld_report">ld_report</code></td>
<td>
<p>Output from function <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code> 
[required].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_ld_resolution">ld_resolution</code></td>
<td>
<p>Resolution at which LD should be reported in number of 
base pairs [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_pop_colors">pop_colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.ld.distance_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with information of LD against distance by population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Delourme, R., Falentin, C., Fomeju, B. F., Boillot, M., Lassalle, G., 
André, I., . . . Marty, A. (2013). High-density SNP-based genetic map 
development and linkage disequilibrium assessment in Brassica napusL. BMC 
genomics, 14(1), 120.
</p>
</li>
<li><p> Li, X., Han, Y., Wei, Y., Acharya, A., Farmer, A. D., Ho, J., . . . 
Brummer, E. C. (2014). Development of an alfalfa SNP array and its use to 
evaluate patterns of population structure and linkage disequilibrium. PLoS 
One, 9(1), e84329.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other ld functions: 
<code><a href="#topic+gl.ld.haplotype">gl.ld.haplotype</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if ((requireNamespace("snpStats", quietly = TRUE)) &amp; (requireNamespace("fields", quietly = TRUE))) {
require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.filter.monomorphs(x)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
ld_res &lt;- gl.report.ld.map(x,ld_max_pairwise = 10000000)
ld_res_2 &lt;- gl.ld.distance(ld_res,ld_resolution= 1000000)
}
</code></pre>

<hr>
<h2 id='gl.ld.haplotype'>Visualize patterns of linkage disequilibrium and identification of 
haplotypes</h2><span id='topic+gl.ld.haplotype'></span>

<h3>Description</h3>

<p>This function plots a Linkage disequilibrium (LD) heatmap, where the colour 
shading indicates the strength of LD. Chromosome positions (Mbp) are shown on
the horizontal axis, and haplotypes appear as triangles and delimited by 
dark yellow vertical lines. Numbers identifying each haplotype are shown in 
the upper part of the plot. 
</p>
<p>The heatmap also shows heterozygosity for each SNP. 
</p>
<p>The function identifies haplotypes based on contiguous SNPs that are in 
linkage disequilibrium using as threshold <code>ld_threshold_haplo</code> and
containing more than <code>min_snps</code> SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.ld.haplotype(
  x,
  pop_name = NULL,
  chrom_name = NULL,
  ld_max_pairwise = 1e+07,
  maf = 0.05,
  ld_stat = "R.squared",
  ind.limit = 10,
  min_snps = 10,
  ld_threshold_haplo = 0.5,
  coordinates = NULL,
  color_haplo = "viridis",
  color_het = "deeppink",
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.ld.haplotype_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_pop_name">pop_name</code></td>
<td>
<p>Name of the population to analyse. If NULL all the 
populations are analised [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_chrom_name">chrom_name</code></td>
<td>
<p>Nme of the chromosome to analyse. If NULL all the 
chromosomes are analised [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_max_pairwise">ld_max_pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD
should be calculated [default 10000000].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out 
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_stat">ld_stat</code></td>
<td>
<p>The LD measure to be calculated: &quot;LLR&quot;, &quot;OR&quot;, &quot;Q&quot;, &quot;Covar&quot;,
&quot;D.prime&quot;, &quot;R.squared&quot;, and &quot;R&quot;. See <code><a href="snpStats.html#topic+ld">ld</a></code>
(package snpStats) for details [default &quot;R.squared&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_min_snps">min_snps</code></td>
<td>
<p>Minimum number of SNPs that should have a haplotype to call 
it [default 10].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_ld_threshold_haplo">ld_threshold_haplo</code></td>
<td>
<p>Minimum LD between adjacent SNPs to call a 
haplotype [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_coordinates">coordinates</code></td>
<td>
<p>A vector of two elements with the start and end 
coordinates in base pairs to which restrict the 
analysis e.g. c(1,1000000) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_color_haplo">color_haplo</code></td>
<td>
<p>Color palette for haplotype plot. See details
[default &quot;viridis&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_color_het">color_het</code></td>
<td>
<p>Color for heterozygosity [default &quot;deeppink&quot;].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if heatmap plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.ld.haplotype_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The information for SNP's position should be stored in the genlight accessor
&quot;@position&quot; and the SNP's chromosome name in the accessor &quot;@chromosome&quot;
(see examples). The function will then calculate LD within each chromosome.
</p>
<p>The output of the function includes a table with the haplotypes
that were identified and their location.
</p>
<p>Colors of the heatmap (<code>color_haplo</code>) are based on the function
<code><a href="viridis.html#topic+scale_fill_viridis">scale_fill_viridis</a></code> from  package <code>viridis</code>. 
Other color palettes options are &quot;magma&quot;, &quot;inferno&quot;, &quot;plasma&quot;, &quot;viridis&quot;,
&quot;cividis&quot;, &quot;rocket&quot;, &quot;mako&quot; and &quot;turbo&quot;.
</p>


<h3>Value</h3>

<p>A table with the haplotypes that were identified.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other ld functions: 
<code><a href="#topic+gl.ld.distance">gl.ld.distance</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.keep.pop(x, pop.list = "TENTERFIELD")
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
ld_res &lt;- gl.ld.haplotype(x,chrom_name = "NC_041728.1_chromosome_1",
                          ld_max_pairwise = 10000000 )
     
</code></pre>

<hr>
<h2 id='gl.LDNe'>Estimates effective population size using the Linkage Disequilibrium
method based on NeEstimator (V2)</h2><span id='topic+gl.LDNe'></span>

<h3>Description</h3>

<p>This function is basically a convenience function that runs the LD Ne
estimator using Neestimator2
(<a href="http://www.molecularfisherieslaboratory.com.au/neestimator-software/">http://www.molecularfisherieslaboratory.com.au/neestimator-software/</a>)
within R using the provided genlight object. To be able to do so, the
software has to be downloaded from their website and the appropriate
executable Ne2-1 has to be copied into the path as specified in the function
(see example below).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.LDNe(
  x,
  outfile = "genepopLD.txt",
  outpath = tempdir(),
  neest.path = getwd(),
  critical = 0,
  singleton.rm = TRUE,
  mating = "random",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = discrete_palette,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.LDNe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file  with
all results from Neestimator 2 [default 'genepopLD.txt'].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_neest.path">neest.path</code></td>
<td>
<p>Path to the folder of the NE2-1 file.
Please note there are 3 different executables depending on your OS:
Ne2-1.exe (=Windows), Ne2-1M (=Mac), Ne2-1L (=Linux). You only need to point
to the folder (the function will recognise which OS you are running)
[default getwd()].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_critical">critical</code></td>
<td>
<p>(vector of) Critical values that are used to remove alleles
based on their minor allele frequency. This can be done before using the
gl.filter.maf function, therefore the default is set to 0 (no loci are
removed). To run for MAF 0 and MAF 0.05 at the same time specify: critical =
c(0,0.05) [default 0].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_singleton.rm">singleton.rm</code></td>
<td>
<p>Whether to remove singleton alleles [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_mating">mating</code></td>
<td>
<p>Formula for Random mating='random' or monogamy= 'monogamy'
[default 'random'].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>A discrete palette for population colors or a list
with as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.LDNe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe with the results as table
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# SNP data (use two populations and only the first 100 SNPs)
pops &lt;- possums.gl[1:60,1:100]
nes &lt;- gl.LDNe(pops, outfile="popsLD.txt", outpath=tempdir(),
neest.path = "./path_to Ne-21",
critical=c(0,0.05), singleton.rm=TRUE, mating='random')
nes

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.list.reports'>Prints dartR reports saved in tempdir</h2><span id='topic+gl.list.reports'></span>

<h3>Description</h3>

<p>Prints dartR reports saved in tempdir
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.list.reports()
</code></pre>


<h3>Value</h3>

<p>Prints a table with all reports saved in tempdir. Currently the style
cannot be changed.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.print.reports">gl.print.reports</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gl.report.callrate(testset.gl,save2tmp=TRUE)
gl.list.reports()

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.load'>Loads an object from compressed binary format produced by gl.save()</h2><span id='topic+gl.load'></span>

<h3>Description</h3>

<p>This is a wrapper for readRDS()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.load(file, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.load_+3A_file">file</code></td>
<td>
<p>Name of the file to receive the binary version of the object
[required].</p>
</td></tr>
<tr><td><code id="gl.load_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The script loads the object from the current workspace and returns the
gl object.
</p>


<h3>Value</h3>

<p>The loaded object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.save">gl.save</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.save(testset.gl,file.path(tempdir(),'testset.rds'))
gl &lt;- gl.load(file.path(tempdir(),'testset.rds'))
</code></pre>

<hr>
<h2 id='gl.make.recode.ind'>Creates a proforma recode_ind file for reassigning individual
(=specimen) names</h2><span id='topic+gl.make.recode.ind'></span>

<h3>Description</h3>

<p>Renaming individuals may be required when there have been errors in
labelling arising in the process from sample to DArT files. There may be
occasions where renaming individuals is required for preparation of figures.
Caution needs to be exercised because of the potential for breaking the
'chain of evidence' between the samples themselves and the analyses. Recoding
individuals can be done with a recode table (csv).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.make.recode.ind(
  x,
  out.recode.file = "default_recode_ind.csv",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.make.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>File name of the output file (including extension)
[default default_recode_ind.csv].</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl.make.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script facilitates the construction of a recode table by producing a
proforma file with current individual (=specimen) names in two identical
columns. Edit the second column to reassign individual names. Use keyword
Delete to delete an individual.
</p>
<p>Apply the recoding using gl.recode.ind(). Deleting individuals can
potentially generate monomorphic loci or loci with all values missing. Clean
this up with gl.filter.monomorphic().
</p>


<h3>Value</h3>

<p>A vector containing the new individual names.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.make.recode.ind(testset.gl, out.recode.file ='Emmac_recode_ind.csv',outpath=tempdir())
</code></pre>

<hr>
<h2 id='gl.make.recode.pop'>Creates a proforma recode_pop_table file for reassigning population
names</h2><span id='topic+gl.make.recode.pop'></span>

<h3>Description</h3>

<p>Renaming populations may be required when there have been errors in
assignment arising in the process from sample to DArT files or when one
wishes to amalgamate populations, or delete populations. Recoding populations
can also be done with a recode table (csv).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.make.recode.pop(
  x,
  out.recode.file = "recode_pop_table.csv",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.make.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_out.recode.file">out.recode.file</code></td>
<td>
<p>File name of the output file (including extension)
[default recode_pop_table.csv].</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl.make.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script facilitates the construction of a recode table by producing a
proforma file with current population names in two identical columns. Edit
the second column to reassign populations. Use keyword Delete to delete a
population.
</p>
<p>Apply the recoding using gl.recode.pop().
</p>


<h3>Value</h3>

<p>A vector containing the new population names.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- gl.make.recode.pop(testset.gl,out.recode.file='test.csv',outpath=tempdir(),verbose=2)
</code></pre>

<hr>
<h2 id='gl.map.interactive'>Creates an interactive map (based on latlon) from a genlight object</h2><span id='topic+gl.map.interactive'></span>

<h3>Description</h3>

<p>Creates an interactive map (based on latlon) from a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.map.interactive(
  x,
  matrix = NULL,
  standard = TRUE,
  symmetric = TRUE,
  pop.labels = TRUE,
  pop.labels.cex = 12,
  ind.circles = TRUE,
  ind.circle.cols = NULL,
  ind.circle.cex = 10,
  ind.circle.transparency = 0.8,
  palette_links = NULL,
  leg_title = NULL,
  provider = "Esri.NatGeoWorldMap",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.map.interactive_+3A_x">x</code></td>
<td>
<p>A genlight object (including coordinates within the latlon slot) 
[required].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_matrix">matrix</code></td>
<td>
<p>A distance matrix between populations or individuals. The
matrix is visualised as lines between individuals/populations. If matrix is
asymmetric two lines with arrows are plotted [default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_standard">standard</code></td>
<td>
<p>If a matrix is provided line width will be standardised to be
between 1 to 10, if set to true, otherwise taken as given [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_symmetric">symmetric</code></td>
<td>
<p>If a symmetric matrix is provided only one line is drawn
based on the lower triangle of the matrix. If set to false arrows indicating
the direction are used instead [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_pop.labels">pop.labels</code></td>
<td>
<p>Population labels at the center of the individuals of
populations [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_pop.labels.cex">pop.labels.cex</code></td>
<td>
<p>Size of population labels [default 12].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circles">ind.circles</code></td>
<td>
<p>Should individuals plotted as circles [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.cols">ind.circle.cols</code></td>
<td>
<p>Colors of circles. Colors can be provided as usual by 
names (e.g. &quot;black&quot;) and are re-cycled. So a color c(&quot;blue&quot;,&quot;red&quot;) colors 
individuals alternatively between blue and red using the genlight object
order of individuals. For transparency see parameter 
ind.circle.transparency. Defaults to rainbow colors by population  if not
provided. If you want to have your own colors for each population, check
the platypus.gl example below.</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.cex">ind.circle.cex</code></td>
<td>
<p>(size or circles in pixels ) [default 10].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_ind.circle.transparency">ind.circle.transparency</code></td>
<td>
<p>Transparency of circles between 0=invisible 
and 1=no transparency. Defaults to 0.8.</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_palette_links">palette_links</code></td>
<td>
<p>Color palette for the links in case a matrix is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_leg_title">leg_title</code></td>
<td>
<p>Legend's title for the links in case a matrix is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_provider">provider</code></td>
<td>
<p>Passed to leaflet [default &quot;Esri.NatGeoWorldMap&quot;].</p>
</td></tr>
<tr><td><code id="gl.map.interactive_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A wrapper around the <span class="pkg">leaflet</span> package. For possible background 
maps check as specified via the provider:
<a href="http://leaflet-extras.github.io/leaflet-providers/preview/index.html">http://leaflet-extras.github.io/leaflet-providers/preview/index.html</a>
</p>
<p>The palette_links argument can be any of the following:
A character vector of RGB or named colors. Examples: palette(), 
c(&quot;#000000&quot;, &quot;#0000FF&quot;, &quot;#FFFFFF&quot;), topo.colors(10)
</p>
<p>The name of an RColorBrewer palette, e.g. &quot;BuPu&quot; or &quot;Greens&quot;.
</p>
<p>The full name of a viridis palette: &quot;viridis&quot;, &quot;magma&quot;, &quot;inferno&quot;, 
or &quot;plasma&quot;.
</p>
<p>A function that receives a single value between 0 and 1 and returns a color.
Examples: colorRamp(c(&quot;#000000&quot;, &quot;#FFFFFF&quot;), interpolate = &quot;spline&quot;).
</p>


<h3>Value</h3>

<p>plots a map
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &ndash; Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
gl.map.interactive(bandicoot.gl)
cols &lt;- c("red","blue","yellow")[as.numeric(pop(platypus.gl))]
gl.map.interactive(platypus.gl, ind.circle.cols=cols, ind.circle.cex=10, 
ind.circle.transparency=0.5)
</code></pre>

<hr>
<h2 id='gl.map.structure'>Maps a STRUCTURE plot using a genlight object</h2><span id='topic+gl.map.structure'></span>

<h3>Description</h3>

<p>This function takes the output of plotstructure (the q matrix) and maps the
q-matrix across using the population centers from the genlight object that
was used to run the structure analysis via <code><a href="#topic+gl.run.structure">gl.run.structure</a></code>)
and plots the typical structure bar plots on a spatial map, providing a
barplot for each subpopulation. Therefore it requires coordinates from a
genlight object. This kind of plots should support the interpretation of the
spatial structure of a population, but in principle is not different from
<code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.map.structure(
  qmat,
  x,
  K,
  provider = "Esri.NatGeoWorldMap",
  scalex = 1,
  scaley = 1,
  movepops = NULL,
  pop.labels = TRUE,
  pop.labels.cex = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.map.structure_+3A_qmat">qmat</code></td>
<td>
<p>Q-matrix from a structure run followed by a clumpp run object
[from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> and <code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>]
[required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the coordinates in the
<code>\@other$latlon</code> slot to calculate the population centers [required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_k">K</code></td>
<td>
<p>The number for K to be plotted [required].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_provider">provider</code></td>
<td>
<p>Provider    passed to leaflet. Check <a href="leaflet.html#topic+providers">providers</a>
for a list of possible backgrounds [default &quot;Esri.NatGeoWorldMap&quot;].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_scalex">scalex</code></td>
<td>
<p>Scaling factor to determine the size of the bars in x direction 
[default 1].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_scaley">scaley</code></td>
<td>
<p>Scaling factor to determine the size of the bars in y direction
[default 1].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_movepops">movepops</code></td>
<td>
<p>A two-dimensional data frame that allows to move the center of
the barplots manually in case they overlap. Often if populations are
horizontally close to each other. This needs to be a data.frame of the
dimensions [rows=number of populations, columns = 2 (lon/lat)]. For each
population you have to specify the x and y (lon and lat) units you want to
move the center of the plot, (see example for details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_pop.labels">pop.labels</code></td>
<td>
<p>Switch for population labels below the parplots 
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.map.structure_+3A_pop.labels.cex">pop.labels.cex</code></td>
<td>
<p>Size of population labels [default 12].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a mapped version of structure plots. For possible background maps
check as specified via the provider:
<a href="http://leaflet-extras.github.io/leaflet-providers/preview/index.html">http://leaflet-extras.github.io/leaflet-providers/preview/index.html</a>.
You may need to adjust scalex and scaley values [default 1], as the size
depends on the scale of the map and the position of the populations.
</p>


<h3>Value</h3>

<p>An interactive map that shows the structure plots broken down by 
population.
</p>
<p>returns the map and a list of the qmat split into sorted matrices per
population. This can be used to create your own map.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li>
<li><p> Evanno, G., Regnaut, S., and J. Goudet. 2005. Detecting the number of
clusters of individuals using the software STRUCTURE: a simulation study.
Molecular Ecology 14:2611-2620.
</p>
</li>
<li><p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.run.structure">gl.run.structure</a></code>,  <code>clumpp</code>,
<code><a href="#topic+gl.plot.structure">gl.plot.structure</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#bc &lt;- bandicoot.gl[,1:100]
#sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure.exe')
#ev &lt;- gl.evanno(sr)
#ev
#qmat &lt;- gl.plot.structure(sr, k=2:4)#' #head(qmat)
#gl.map.structure(qmat, bc,K=3)
#gl.map.structure(qmat, bc,K=4)
#move population 4 (out of 5) 0.5 degrees to the right and populations 1
#0.3 degree to the north of the map.
#mp &lt;- data.frame(lon=c(0,0,0,0.5,0), lat=c(-0.3,0,0,0,0))
#gl.map.structure(qmat, bc,K=4, movepops=mp)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.merge.pop'>Merges two or more populations in a genlight object into one population</h2><span id='topic+gl.merge.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart().
</p>
<p>This script assigns individuals from two nominated populations into a new
single population. It can also be used to rename populations.
</p>
<p>The script returns a genlight object with the new population assignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.merge.pop(x, old = NULL, new = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.merge.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_old">old</code></td>
<td>
<p>A list of populations to be merged [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_new">new</code></td>
<td>
<p>Name of the new population [required].</p>
</td></tr>
<tr><td><code id="gl.merge.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the new population assignments.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   gl &lt;- gl.merge.pop(testset.gl, old=c('EmsubRopeMata','EmvicVictJasp'), new='Outgroup')
</code></pre>

<hr>
<h2 id='gl.nhybrids'>Creates an input file for the program NewHybrids and runs it if 
NewHybrids is installed</h2><span id='topic+gl.nhybrids'></span>

<h3>Description</h3>

<p>This function compares two sets of parental populations to identify loci that
exhibit a fixed difference, returns an genlight object with the reduced
data, and creates an input file for the program NewHybrids using the top 200
(or hard specified loc.limit) loci. In the absence of two identified
parental populations, the script will select a random set 200 loci only
(method='random') or the first 200 loci ranked on information content
(method='AvgPIC').
</p>
<p>A fixed difference occurs when a SNP allele is present in all individuals
of one population and absent in the other. There is provision for setting
a level of tolerance, e.g. threshold = 0.05 which considers alleles present
at greater than 95
a fixed difference. Only the 200 loci are retained, because of limitations
of NewHybids.
</p>
<p>If you specify a directory for the NewHybrids executable file, then the
script will create the input file from the SNP data then run NewHybrids. If
the directory is set to NULL, the execution will stop once the input file
(default='nhyb.txt') has been written to disk. Note: the executable option
will not work on a Mac; Mac users should generate the NewHybrids input file
and run this on their local installation of NewHybrids.
</p>
<p>Refer to the New Hybrids manual for further information on the parameters to
set
&ndash; http://ib.berkeley.edu/labs/slatkin/eriq/software/new_hybs_doc1_1Beta3.pdf
</p>
<p>It is important to stringently filter the data on RepAvg and CallRate if
using the random option. One might elect to repeat the analysis
(method='random') and combine the resultant posterior probabilities should
200 loci be considered insufficient.
</p>
<p>The F1 individuals should be homozygous at all loci for which the parental
populations are fixed and different, assuming parental populations have been
specified. Sampling errors can result in this not being the case, especially
where the sample sizes for the parental populations are small. Alternatively,
the threshold for posterior probabilities used to determine assignment
(pprob) or the definition of a fixed difference (threshold) may be too lax.
To assess the error rate in the determination of assignment of F1
individuals, a plot of the frequency of homozygous reference, heterozygotes
and homozygous alternate (SNP) can be produced by setting plot=TRUE (the
default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.nhybrids(
  gl,
  outpath = tempdir(),
  p0 = NULL,
  p1 = NULL,
  threshold = 0,
  method = "random",
  plot = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  pprob = 0.95,
  nhyb.directory = NULL,
  BurnIn = 10000,
  sweeps = 10000,
  GtypFile = "TwoGensGtypFreq.txt",
  AFPriorFile = NULL,
  PiPrior = "Jeffreys",
  ThetaPrior = "Jeffreys",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.nhybrids_+3A_gl">gl</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir()].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_p0">p0</code></td>
<td>
<p>List of populations to be regarded as parental population 0
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_p1">p1</code></td>
<td>
<p>List of populations to be regarded as parental population 1
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_threshold">threshold</code></td>
<td>
<p>Sets the level at which a gene frequency difference is
considered to be fixed [default 0].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_method">method</code></td>
<td>
<p>Specifies the method (random or AvgPIC) to select 200 loci for
NewHybrids [default random].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot">plot</code></td>
<td>
<p>If TRUE, a plot of the frequency of homozygous reference,
heterozygotes and homozygous alternate (SNP) is produced for the F1
individuals
[default TRUE, applies only if both parental populations are specified].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two_colors].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_pprob">pprob</code></td>
<td>
<p>Threshold level for assignment to likelihood bins
[default 0.95, used only if plot=TRUE].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_nhyb.directory">nhyb.directory</code></td>
<td>
<p>Directory that holds the NewHybrids executable file
e.g. C:/NewHybsPC [default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_burnin">BurnIn</code></td>
<td>
<p>Number of sweeps to use in the burn in [default 10000].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_sweeps">sweeps</code></td>
<td>
<p>Number  of  sweeps  to  use  in  computing  the  actual Monte
Carlo averages [default 10000].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_gtypfile">GtypFile</code></td>
<td>
<p>Name of a file containing the genotype frequency classes
[default TwoGensGtypFreq.txt].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_afpriorfile">AFPriorFile</code></td>
<td>
<p>Name of the file containing prior allele frequency
information [default NULL].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_piprior">PiPrior</code></td>
<td>
<p>Jeffreys-like priors or Uniform priors for the parameter pi
[default Jeffreys].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_thetaprior">ThetaPrior</code></td>
<td>
<p>Jeffreys-like priors or Uniform priors for the parameter
theta [default Jeffreys].</p>
</td></tr>
<tr><td><code id="gl.nhybrids_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The reduced genlight object, if parentals are provided; output of
NewHybrids is saved to the working directory.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>

<p>Anderson, E.C. and Thompson, E.A.(2002). A model-based method for identifying 
species hybrids using multilocus genetic data. Genetics. 160:1217-1229.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
m &lt;- gl.nhybrids(testset.gl, 
p0=NULL, p1=NULL,
nhyb.directory='D:/workspace/R/NewHybsPC', # Specify as necessary
outpath="D:/workspace",  # Specify as necessary, usually getwd() [= workspace]
BurnIn=100,
sweeps=100,
verbose=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.outflank'>Identifies loci under selection per population using the outflank
method of Whitlock and Lotterhos (2015)</h2><span id='topic+gl.outflank'></span>

<h3>Description</h3>

<p>Identifies loci under selection per population using the outflank
method of Whitlock and Lotterhos (2015)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.outflank(
  gi,
  plot = TRUE,
  LeftTrimFraction = 0.05,
  RightTrimFraction = 0.05,
  Hmin = 0.1,
  qthreshold = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.outflank_+3A_gi">gi</code></td>
<td>
<p>A genlight or genind object, with a defined population structure
[required].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_plot">plot</code></td>
<td>
<p>A switch if a barplot is wanted [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_lefttrimfraction">LeftTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
lower end of the range of Fst before the likelihood function is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_righttrimfraction">RightTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
upper end of the range of Fst before the likelihood function is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including calculations
from a locus [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_qthreshold">qthreshold</code></td>
<td>
<p>The desired false discovery rate threshold for calculating
q-values [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.outflank_+3A_...">...</code></td>
<td>
<p>additional parameters (see documentation of outflank on github).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper around the outflank function provided by
Whitlock and Lotterhos. To be able to run this function the packages qvalue
(from bioconductor) and outflank (from github) needs to be installed. To do
so see example below.
</p>


<h3>Value</h3>

<p>Returns an index of outliers and the full outflank list
</p>


<h3>References</h3>

<p>Whitlock, M.C. and Lotterhos K.J. (2015) Reliable detection of loci
responsible for local adaptation: inference of a neutral model through
trimming the distribution of Fst. The American Naturalist 186: 24 - 36.
</p>
<p>Github repository: Whitlock &amp; Lotterhos:
<a href="https://github.com/whitlock/OutFLANK">https://github.com/whitlock/OutFLANK</a> (Check the readme.pdf within the
repository for an explanation. Be aware you now can run OufFLANK from a
genlight object)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utils.outflank">utils.outflank</a></code>, <code><a href="#topic+utils.outflank.plotter">utils.outflank.plotter</a></code>,
<code><a href="#topic+utils.outflank.MakeDiploidFSTMat">utils.outflank.MakeDiploidFSTMat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gl.outflank(bandicoot.gl, plot = TRUE)

</code></pre>

<hr>
<h2 id='gl.pcoa'>Ordination applied to genotypes in a genlight object (PCA), in an fd
object, or to a distance matrix (PCoA)</h2><span id='topic+gl.pcoa'></span>

<h3>Description</h3>

<p>This function takes the genotypes for individuals and undertakes a Pearson
Principal Component analysis (PCA) on SNP or Tag P/A (SilicoDArT) data; it
undertakes a Gower Principal Coordinate analysis (PCoA) if supplied with a
distance matrix. Technically, any distance matrix can be represented in an
ordinated space using PCoA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.pcoa(
  x,
  nfactors = 5,
  correction = NULL,
  mono.rm = TRUE,
  parallel = FALSE,
  n.cores = 16,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.pcoa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object or fd object containing the SNP data, or
a distance matrix of type dist [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_nfactors">nfactors</code></td>
<td>
<p>Number of axes to retain in the output of factor scores
[default 5].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_correction">correction</code></td>
<td>
<p>Method applied to correct for negative eigenvalues, either
'lingoes' or 'cailliez' [Default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_parallel">parallel</code></td>
<td>
<p>TRUE if parallel processing is required (does fail under
Windows) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_n.cores">n.cores</code></td>
<td>
<p>Number of cores to use if parallel processing is requested
[default 16].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, a diagnostic plot is displayed showing a scree plot
for the &quot;informative&quot; axes and a histogram of eigenvalues of the remaining 
&quot;noise&quot; axes [Default TRUE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plot [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa_+3A_verbose">verbose</code></td>
<td>
<p>verbose= 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is essentially a wrapper for glPca adegenet or pcoa {ape}
with default settings apart from those specified as parameters in this
function.
<strong> Sources of stress in the visual representation </strong>
</p>
<p>While, technically, any distance matrix can be represented in an ordinated
space, the representation will not typically be exact.There are three major
sources of stress in a reduced-representation of distances or dissimilarities
among entities using PCA or PCoA. By far the greatest source comes from the
decision to select only the top two or three axes from the ordinated set of
axes derived from the PCA or PCoA. The representation of the entities such a
heavily reduced space will not faithfully represent the distances in the
input distance matrix simply because of the loss of information in deeper
informative dimensions. For this reason, it is not sensible to be too
precious about managing the other two sources of stress in the visual
representation.
</p>
<p>The measure of distance between entities in a PCA is the Pearson Correlation
Coefficient, essentially a standardized Euclidean distance. This is both a
metric distance and a Euclidean distance. In PCoA, the second source of
stress is the choice of distance measure or dissimilarity measure. While any
distance or dissimilarity matrix can be represented in an ordinated space,
the distances between entities can be faithfully represented in that space
(that is, without stress) only if the distances are metric. Furthermore, for
distances between entities to be faithfully represented in a rigid Cartesian
space, the distance measure needs to be Euclidean. If this is not the case,
the distances between the entities in the ordinated visualized space will not
exactly represent the distances in the input matrix (stress will be non-zero).
This source of stress will be evident as negative eigenvalues in the deeper
dimensions.
</p>
<p>A third source of stress arises from having a sparse dataset, one with
missing values. This affects both PCA and PCoA. If the original data matrix
is not fully populated, that is, if there are missing values, then even a
Euclidean distance matrix will not necessarily be 'positive definite'. It
follows that some of the eigenvalues may be negative, even though the
distance metric is Euclidean. This issue is exacerbated when the number of
loci greatly exceeds the number of individuals, as is typically the case when
working with SNP data. The impact of missing values can be minimized by
stringently filtering on Call Rate, albeit with loss of data. An alternative
is given in a paper 'Honey, I shrunk the sample covariance matrix' and more
recently by Ledoit and Wolf (2018), but their approach has not been
implemented here.
</p>
<p>The good news is that, unless the sum of the negative eigenvalues, arising
from a non-Euclidean distance measure or from missing values, approaches
those of the final PCA or PCoA axes to be displayed, the distortion is
probably of no practical consequence and certainly not comparable to the
stress arising from selecting only two or three final dimensions out of
several informative dimensions for the visual representation.
</p>
<p><strong> Function's output </strong>
</p>
<p>Two diagnostic plots are produced. The first is a Scree Plot, showing the
percentage variation explained by each of the PCA or PCoA axes, for those
axes that explain more than the original variables (loci) on average. That
is, only informative axes are displayed. The scree plot informs the number of
dimensions to be retained in the visual summaries. As a rule of thumb, axes
with more than 10
</p>
<p>The second graph shows the distribution of eigenvalues for the remaining
uninformative (noise) axes, including those with negative eigenvalues.
</p>
<p>Action is recommended (verbose &gt;= 2) if the negative eigenvalues are
dominant, their sum approaching in magnitude the eigenvalues for axes
selected for the final visual solution.
</p>
<p>Output is a glPca object conforming to adegenet::glPca but with only the
following retained.
</p>

<ul>
<li><p>  $call - The call that generated the PCA/PCoA
</p>
</li>
<li><p>  $eig - Eigenvalues &ndash; All eigenvalues (positive, null, negative).
</p>
</li>
<li><p>  $scores - Scores (coefficients) for each individual
</p>
</li>
<li><p>  $loadings - Loadings of each SNP for each principal component
</p>
</li></ul>

<p>Plots and table were saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the R
session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>PCA was developed by Pearson (1901) and Hotelling (1933), whilst the best
modern reference is Jolliffe (2002). PCoA was developed by Gower (1966) while
the best modern reference is Legendre &amp; Legendre (1998).
</p>


<h3>Value</h3>

<p>An object of class pcoa containing the eigenvalues and factor scores
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Cailliez, F. (1983) The analytical solution of the additive constant
problem. Psychometrika, 48, 305-308.
</p>
</li>
<li><p> Gower, J. C. (1966) Some distance properties of latent root and vector
methods used in multivariate analysis. Biometrika, 53, 325-338.
</p>
</li>
<li><p> Hotelling, H., 1933. Analysis of a complex of statistical variables into
Principal Components. Journal of Educational Psychology 24:417-441, 498-520.
</p>
</li>
<li><p> Jolliffe, I. (2002) Principal Component Analysis. 2nd Edition, Springer,
New York.
</p>
</li>
<li><p> Ledoit, O. and Wolf, M. (2018). Analytical nonlinear shrinkage of
large-dimensional covariance matrices. University of Zurich, Department of
Economics, Working Paper No. 264, Revised version. Available at SSRN:
https://ssrn.com/abstract=3047302 or http://dx.doi.org/10.2139/ssrn.3047302
</p>
</li>
<li><p> Legendre, P. and Legendre, L. (1998). Numerical Ecology, Volume 24, 2nd
Edition. Elsevier Science, NY.
</p>
</li>
<li><p> Lingoes, J. C. (1971) Some boundary conditions for a monotone analysis
of symmetric matrices. Psychometrika, 36, 195-203.
</p>
</li>
<li><p> Pearson, K. (1901). On lines and planes of closest fit to systems of
points in space. Philosophical Magazine. Series 6, vol. 2, no. 11, pp.
559-572.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.pcoa.plot">gl.pcoa.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gl &lt;- possums.gl
# PCA (using SNP genlight object)
pca &lt;- gl.pcoa(possums.gl[1:50,],verbose=2)
gl.pcoa.plot(pca,gl)

gs &lt;- testset.gs
levels(pop(gs))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',6),'Em.subglobosa','Em.victoriae')

# PCA (using SilicoDArT genlight object)
pca &lt;- gl.pcoa(gs)
gl.pcoa.plot(pca,gs)

# Collapsing pops to OTUs using Fixed Difference Analysis (using fd object)
fd &lt;- gl.fixed.diff(testset.gl)
fd &lt;- gl.collapse(fd)
pca &lt;- gl.pcoa(fd)
gl.pcoa.plot(pca,fd$gl)

# Using a distance matrix
D &lt;- gl.dist.ind(testset.gs, method='jaccard')
pcoa &lt;- gl.pcoa(D,correction="cailliez")
gl.pcoa.plot(pcoa,gs)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.pcoa.plot'>Bivariate or trivariate plot of the results of an ordination generated
using gl.pcoa()</h2><span id='topic+gl.pcoa.plot'></span>

<h3>Description</h3>

<p>This script takes output from the ordination generated by gl.pcoa() and plots
the individuals classified by population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.pcoa.plot(
  glPca,
  x,
  scale = FALSE,
  ellipse = FALSE,
  plevel = 0.95,
  pop.labels = "pop",
  interactive = FALSE,
  as.pop = NULL,
  hadjust = 1.5,
  vadjust = 1,
  xaxis = 1,
  yaxis = 2,
  zaxis = NULL,
  pt.size = 2,
  pt.colors = NULL,
  pt.shapes = NULL,
  label.size = 1,
  axis.label.size = 1.5,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.pcoa.plot_+3A_glpca">glPca</code></td>
<td>
<p>Name of the PCA or PCoA object containing the factor scores and
eigenvalues [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object or fd object containing the SNP
genotypes or Tag P/A (SilicoDArT) genotypes or the Distance Matrix used to
generate the ordination [required].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_scale">scale</code></td>
<td>
<p>If TRUE, scale the x and y axes in proportion to % variation
explained [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_ellipse">ellipse</code></td>
<td>
<p>If TRUE, display ellipses to encapsulate points for each
population [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_plevel">plevel</code></td>
<td>
<p>Value of the percentile for the ellipse to encapsulate points
for each population [default 0.95].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pop.labels">pop.labels</code></td>
<td>
<p>How labels will be added to the plot
['none'|'pop'|'legend', default = 'pop'].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_interactive">interactive</code></td>
<td>
<p>If TRUE then the populations are plotted without labels,
mouse-over to identify points [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_as.pop">as.pop</code></td>
<td>
<p>Assign another metric to represent populations for the plot
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_hadjust">hadjust</code></td>
<td>
<p>Horizontal adjustment of label position in 2D plots
[default 1.5].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_vadjust">vadjust</code></td>
<td>
<p>Vertical adjustment of label position in 2D plots [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_xaxis">xaxis</code></td>
<td>
<p>Identify the x axis from those available in the ordination
(xaxis &lt;= nfactors) [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_yaxis">yaxis</code></td>
<td>
<p>Identify the y axis from those available in the ordination
(yaxis &lt;= nfactors) [default 2].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_zaxis">zaxis</code></td>
<td>
<p>Identify the z axis from those available in the ordination for a
3D plot (zaxis &lt;= nfactors) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.size">pt.size</code></td>
<td>
<p>Specify the size of the displayed points [default 2].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.colors">pt.colors</code></td>
<td>
<p>Optionally provide a vector of nPop colors
(run gl.select.colors() for color options) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_pt.shapes">pt.shapes</code></td>
<td>
<p>Optionally provide a vector of nPop shapes
(run gl.select.shapes() for shape options) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_label.size">label.size</code></td>
<td>
<p>Specify the size of the point labels [default 1].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_axis.label.size">axis.label.size</code></td>
<td>
<p>Specify the size of the displayed axis labels
[default 1.5].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.pcoa.plot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The factor scores are taken from the output of gl.pcoa() and the population
assignments are taken from from the original data file. In the bivariate
plots, the specimens are shown optionally with adjacent labels and enclosing
ellipses. Population labels on the plot are shuffled so as not to overlap
(using package {directlabels}).
This can be a bit clunky, as the labels may be some distance from the points
to which they refer, but it provides the opportunity for moving labels around
using graphics software (e.g. Adobe Illustrator).
</p>
<p>3D plotting is activated by specifying a zaxis.
</p>
<p>Any pair or trio of axes can be specified from the ordination, provided they
are within the range of the nfactors value provided to gl.pcoa().
In the 2D plots, axes can be scaled to represent the proportion of variation
explained. In any case, the proportion of variation explained by each axis is
provided in the axis label.
</p>
<p>Colors and shapes of the points can be altered by passing a vector of shapes
and/or a vector of colors. These vectors can be created with
gl.select.shapes() and gl.select.colors() and passed to this script using the
pt.shapes and pt.colors parameters.
</p>
<p>Points displayed in the ordination can be identified if the option
interactive=TRUE is chosen, in which case the resultant plot is ggplotly()
friendly. Identification of points is by moving the mouse over them. Refer
to the plotly package for further information.
The interactive option is automatically enabled for 3D plotting.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.pcoa">gl.pcoa</a></code>
</p>
<p>Other Exploration/visualisation functions: 
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# RUN PCA
pca&lt;-gl.pcoa(gl,nfactors=5)
# VARIOUS EXAMPLES
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.95, pop.labels='pop', 
axis.label.size=1, hadjust=1.5,vadjust=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend', 
axis.label.size=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend', 
axis.label.size=1.5,scale=TRUE)
gl.pcoa.plot(pca, gl, ellipse=TRUE, axis.label.size=1.2, xaxis=1, yaxis=3, 
scale=TRUE)
gl.pcoa.plot(pca, gl, pop.labels='none',scale=TRUE)
gl.pcoa.plot(pca, gl, axis.label.size=1.2, interactive=TRUE)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, xaxis=1, yaxis=2, zaxis=3)
# color AND SHAPE ADJUSTMENTS
shp &lt;- gl.select.shapes(select=c(16,17,17,0,2))
col &lt;- gl.select.colors(library='brewer',palette='Spectral',ncolors=11,
select=c(1,9,3,11,11))
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.95, pop.labels='pop', 
pt.colors=col, pt.shapes=shp, axis.label.size=1, hadjust=1.5,vadjust=1)
gl.pcoa.plot(pca, gl, ellipse=TRUE, plevel=0.99, pop.labels='legend',
 pt.colors=col, pt.shapes=shp, axis.label.size=1)
 
 test &lt;- gl.pcoa(platypus.gl)
 gl.pcoa.plot(glPca = test, x = platypus.gl)
</code></pre>

<hr>
<h2 id='gl.percent.freq'>Generates percentage allele frequencies by locus and population</h2><span id='topic+gl.percent.freq'></span>

<h3>Description</h3>

<p>This is a support script, to take SNP data or SilicoDArT presence/absence
data grouped into populations in a genlight object {adegenet} and generate
a table of allele frequencies for each population and locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.percent.freq(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.percent.freq_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or Tag P/A
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.percent.freq_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with allele (SNP data) or presence/absence frequencies
(Tag P/A data) broken down by population and locus
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;-  gl.percent.freq(testset.gl)
</code></pre>

<hr>
<h2 id='gl.play.history'>Replays the history and applies it to a genlight object</h2><span id='topic+gl.play.history'></span>

<h3>Description</h3>

<p>Replays the history and applies it to a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.play.history(x, history = NULL, verbose = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.play.history_+3A_x">x</code></td>
<td>
<p>A genlight object (with a history slot) [optional].</p>
</td></tr>
<tr><td><code id="gl.play.history_+3A_history">history</code></td>
<td>
<p>If no history is provided the complete history of
x is used (recreating the identical object x). If history is a vector it
indicates which which part of the history of x is used [<code>c(1,3,4)</code> uses
the first, third and forth entry from <code>x@other$history</code>]. Or a simple
link to a history slot of another genlight object (e.g.
codex2@other$history[c(1,4,5)]). [optional].</p>
</td></tr>
<tr><td><code id="gl.play.history_+3A_verbose">verbose</code></td>
<td>
<p>If set to one then history commands are printed,
which may facilitate reading the output [default 0].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function basically allows to create a 'template history'
(=set of filters) and apply them to any other genlight object. Histories can
also be saved and loaded (see. gl.save.history and gl.load.history).
</p>


<h3>Value</h3>

<p>Returns a genlight object that was created by replaying the provided
applied to the genlight object x. Please note you can 'mix' histories or
part of them and apply them to different genlight objects. If the history
does not contain <code>gl.read.dart</code>, histories of x and history are
concatenated.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dartfile &lt;- system.file('extdata','testset_SNPs_2Row.csv', package='dartR')
metadata &lt;- system.file('extdata','testset_metadata.csv', package='dartR')
gl &lt;- gl.read.dart(dartfile, ind.metafile = metadata, probar=FALSE)
gl2 &lt;- gl.filter.callrate(gl, method='loc', threshold=0.9)
gl3 &lt;- gl.filter.callrate(gl2, method='ind', threshold=0.95)
#Now 'replay' part of the history 'onto' another genlight object
#bc.fil &lt;- gl.play.history(gl.compliance.check(bandicoot.gl),
#history=gl3@other$history[c(2,3)], verbose=1)
#gl.print.history(bc.fil)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.plot.heatmap'>Represents a distance matrix as a heatmap</h2><span id='topic+gl.plot.heatmap'></span>

<h3>Description</h3>

<p>The script plots a heat map to represent the distances in the distance or
dissimilarity matrix. This function is a wrapper for
<a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package gplots).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.heatmap(D, palette_divergent = diverging_palette, verbose = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.heatmap_+3A_d">D</code></td>
<td>
<p>Name of the distance matrix or class fd object [required].</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_palette_divergent">palette_divergent</code></td>
<td>
<p>A divergent palette for the distance values
[default diverging_palette].</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
<tr><td><code id="gl.plot.heatmap_+3A_...">...</code></td>
<td>
<p>Parameters passed to function
<a href="gplots.html#topic+heatmap.2">heatmap.2</a> (package gplots)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   gl &lt;- testset.gl[1:10,]
   D &lt;- dist(as.matrix(gl),upper=TRUE,diag=TRUE)
   gl.plot.heatmap(D)
   D2 &lt;- gl.dist.pop(possums.gl)
   gl.plot.heatmap(D2)
   D3 &lt;- gl.fixed.diff(testset.gl)
   gl.plot.heatmap(D3)
   
## End(Not run)
   if ((requireNamespace("gplots", quietly = TRUE))) {
   D2 &lt;- gl.dist.pop(possums.gl)
   gl.plot.heatmap(D2)
   }
</code></pre>

<hr>
<h2 id='gl.plot.network'>Represents a distance or dissimilarity matrix as a network</h2><span id='topic+gl.plot.network'></span>

<h3>Description</h3>

<p>This script takes a distance matrix generated by dist() and represents the
relationship among the specimens as a network diagram. In order to use this
script, a decision is required on a threshold for relatedness to be
represented as link in the network, and on the layout used to create the
diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.network(
  D,
  x = NULL,
  method = "fr",
  node.size = 3,
  node.label = FALSE,
  node.label.size = 0.7,
  node.label.color = "black",
  alpha = 0.005,
  title = "Network based on genetic distance",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.network_+3A_d">D</code></td>
<td>
<p>A distance or dissimilarity matrix generated by dist() or gl.dist()
[required].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_x">x</code></td>
<td>
<p>A genlight object from which the D matrix was generated
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_method">method</code></td>
<td>
<p>One of &quot;fr&quot;, &quot;kk&quot; or &quot;drl&quot; [default &quot;fr&quot;].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_node.size">node.size</code></td>
<td>
<p>Size of the symbols for the network nodes [default 3].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_node.label">node.label</code></td>
<td>
<p>TRUE to display node labels [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_node.label.size">node.label.size</code></td>
<td>
<p>Size of the node labels [default 0.7].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_node.label.color">node.label.color</code></td>
<td>
<p>Color of the text of the node labels
[default 'black'].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_alpha">alpha</code></td>
<td>
<p>Upper threshold to determine which links between nodes to display
[default 0.005].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_title">title</code></td>
<td>
<p>Title for the plot
[default &quot;Network based on genetic distance&quot;].</p>
</td></tr>
<tr><td><code id="gl.plot.network_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The threshold for relatedness to be represented as a link in the network is
specified as a quantile. Those relatedness measures above the quantile are
plotted as links, those below the quantile are not. Often you are looking for
relatedness outliers in comparison with the overall relatedness among
individuals, so a very conservative quantile is used (e.g. 0.004), but
ultimately, this decision is made as a matter of trial and error. One way to
approach this trial and error is to try to achieve a sparse set of links
between unrelated 'background' individuals so that the stronger links are
preferentially shown.
</p>
<p>There are several layouts from which to choose. The most popular are given as
options in this script.
</p>

<ul>
<li><p> fr &ndash; Fruchterman, T.M.J. and Reingold, E.M. (1991). Graph Drawing by
Force-directed Placement. Software &ndash; Practice and Experience 21:1129-1164.
</p>
</li>
<li><p> kk &ndash; Kamada, T. and Kawai, S.: An Algorithm for Drawing General
Undirected Graphs. Information Processing Letters 31:7-15, 1989.
</p>
</li>
<li><p> drl &ndash; Martin, S., Brown, W.M., Klavans, R., Boyack, K.W., DrL:
Distributed Recursive (Graph) Layout. SAND Reports 2936:1-10, 2008.
</p>
</li></ul>

<p>Colors of node symbols are those of the rainbow.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if ((requireNamespace("rrBLUP", quietly = TRUE)) &amp; (requireNamespace("gplots", quietly = TRUE))) {
test &lt;- gl.subsample.loci(platypus.gl, n = 100)
test &lt;- gl.keep.ind(test,ind.list = indNames(test)[1:10])
D &lt;- gl.grm(test, legendx=0.04)
gl.plot.network(D,test)
}
</code></pre>

<hr>
<h2 id='gl.plot.structure'>Plots STRUCTURE analysis results (Q-matrix)</h2><span id='topic+gl.plot.structure'></span>

<h3>Description</h3>

<p>This function takes a structure run object (output from
<code><a href="#topic+gl.run.structure">gl.run.structure</a></code>) and plots the typical structure bar
plot that visualize the q matrix of a structure run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.plot.structure(
  sr,
  K = NULL,
  met_clumpp = "greedyLargeK",
  iter_clumpp = 100,
  clumpak = TRUE,
  plot_theme = NULL,
  colors_clusters = NULL,
  ind_name = TRUE,
  border_ind = 0.15,
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.plot.structure_+3A_sr">sr</code></td>
<td>
<p>Structure run object from <code><a href="#topic+gl.run.structure">gl.run.structure</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_k">K</code></td>
<td>
<p>The number for K of the q matrix that should be plotted. Needs to
be within you simulated range of K's in your sr structure run object. If 
NULL, all the K's are plotted [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_met_clumpp">met_clumpp</code></td>
<td>
<p>The algorithm to use to infer the correct permutations.
One of 'greedy' or 'greedyLargeK' or 'stephens' [default &quot;greedyLargeK&quot;].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_iter_clumpp">iter_clumpp</code></td>
<td>
<p>The number of iterations to use if running either 'greedy'
'greedyLargeK' [default 100].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_clumpak">clumpak</code></td>
<td>
<p>Whether use the Clumpak method (see details) [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_colors_clusters">colors_clusters</code></td>
<td>
<p>A color palette for clusters (K) or a list with
as many colors as there are clusters (K) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_ind_name">ind_name</code></td>
<td>
<p>Whether to plot individual names [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_border_ind">border_ind</code></td>
<td>
<p>The width of the border line between individuals 
[default 0.25].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.plot.structure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default
NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function outputs a barplot which is the typical output of
structure. For a Evanno plot use gl.evanno.
</p>
<p>This function is based on the methods of CLUMPP and Clumpak as implemented 
in the R package starmie (https://github.com/sa-lee/starmie). 
</p>
<p>The Clumpak method identifies sets of highly similar runs among 
all the replicates of the same K. The method then separates the distinct 
groups of runs representing distinct modes in the space of possible solutions.
</p>
<p>The CLUMPP method permutes the clusters output by independent runs of 
clustering programs such as structure, so that they match up as closely as 
possible.
</p>
<p>This function averages the replicates within each mode identified by the 
Clumpak method.
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>List of Q-matrices
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Kopelman, Naama M., et al. &quot;Clumpak: a program for identifying 
clustering modes and packaging population structure inferences across K.&quot; 
Molecular ecology resources 15.5 (2015): 1179-1191.
</p>
</li>
<li><p> Mattias Jakobsson and Noah A. Rosenberg. 2007. CLUMPP: a cluster
matching and permutation program for dealing with label switching and
multimodality in analysis of population structure. Bioinformatics
23(14):1801-1806. Available at
<a href="http://web.stanford.edu/group/rosenberglab/clumppDownload.html">clumpp</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code>gl.run.structure</code>, <code>gl.plot.structure</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#bc &lt;- bandicoot.gl[,1:100]
#sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, exec = './structure')
#ev &lt;- gl.evanno(sr)
#ev
#qmat &lt;- gl.plot.structure(sr, K=3)
#head(qmat)
#gl.map.structure(qmat, K=3, bc, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.print.history'>Prints history of a genlight object</h2><span id='topic+gl.print.history'></span>

<h3>Description</h3>

<p>Prints history of a genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.print.history(x = NULL, history = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.print.history_+3A_x">x</code></td>
<td>
<p>A genlight object (with history) [optional].</p>
</td></tr>
<tr><td><code id="gl.print.history_+3A_history">history</code></td>
<td>
<p>Either a link to a history slot
(gl\@other$history), or a vector indicating which part of the history of x is
used [c(1,3,4) uses the first, third and forth entry from x\@other$history].
If no history is provided the complete history of x is used (recreating the
identical object x) [optional].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a table with all history records. Currently the style cannot
be changed.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dartfile &lt;- system.file('extdata','testset_SNPs_2Row.csv', package='dartR')
metadata &lt;- system.file('extdata','testset_metadata.csv', package='dartR')
gl &lt;- gl.read.dart(dartfile, ind.metafile = metadata, probar=FALSE)
gl2 &lt;- gl.filter.callrate(gl, method='loc', threshold=0.9)
gl3 &lt;- gl.filter.callrate(gl2, method='ind', threshold=0.95)
#Now 'replay' part of the history 'onto' another genlight object
#bc.fil &lt;- gl.play.history(gl.compliance.check(bandicoot.gl),
#history=gl3@other$history[c(2,3)], verbose=1)
#gl.print.history(bc.fil)

</code></pre>

<hr>
<h2 id='gl.print.reports'>Prints dartR reports saved in tempdir</h2><span id='topic+gl.print.reports'></span>

<h3>Description</h3>

<p>Prints dartR reports saved in tempdir
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.print.reports(print_report)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.print.reports_+3A_print_report">print_report</code></td>
<td>
<p>Number of report from <code><a href="#topic+gl.list.reports">gl.list.reports</a></code> that 
is to be printed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints reports that were saved in tempdir.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.list.reports">gl.list.reports</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
reports &lt;- gl.print.reports(1)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.propShared'>Calculates a similarity (distance) matrix for individuals on the proportion of
shared alleles</h2><span id='topic+gl.propShared'></span>

<h3>Description</h3>

<p>This script calculates an individual based distance matrix. It uses an C++
implementation, so package Rcpp needs to be installed and it is therefore
really fast (once it has compiled the function after the first run).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.propShared(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.propShared_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the SNP genotypes [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A similarity matrix
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#takes some time at the first run of the function...
## Not run: 
res &lt;- gl.propShared(bandicoot.gl)
res[1:5,1:7] #show only a small part of the matrix

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.random.snp'>Randomly changes the allocation of 0's and 2's in a genlight object</h2><span id='topic+gl.random.snp'></span>

<h3>Description</h3>

<p>This function samples randomly half of the SNPs and re-codes, in the sampled
SNP's, 0's by 2's.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.random.snp(x, plot.out = TRUE, save2tmp = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.random.snp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.random.snp_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if a plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.random.snp_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots to the session temporary directory
(tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.random.snp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default NULL,
unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DArT calls the most common allele as the reference allele. In a genlight
object, homozygous for the reference allele are coded with a '0' and
homozygous for the alternative allele are coded with a '2'. This causes some
distortions in visuals from time to time.
</p>
<p>If plot.out = TRUE, two smear plots (pre-randomisation and
post-randomisation) are presented using a random subset of individuals (10)
and loci (100) to provide an overview of the changes.
</p>
<p>Resultant ggplots are saved to the session's temporary directory.
</p>


<h3>Value</h3>

<p>Returns a genlight object with half of the loci re-coded.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- gl.random.snp(platypus.gl[1:5,1:5],verbose = 5)

</code></pre>

<hr>
<h2 id='gl.read.csv'>Reads SNP data from a csv file into a genlight object</h2><span id='topic+gl.read.csv'></span>

<h3>Description</h3>

<p>This script takes SNP genotypes from a csv file, combines them with
individual and locus metrics and creates a genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.csv(
  filename,
  transpose = FALSE,
  ind.metafile = NULL,
  loc.metafile = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.csv_+3A_filename">filename</code></td>
<td>
<p>Name of the csv file containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_transpose">transpose</code></td>
<td>
<p>If TRUE, rows are loci and columns are individuals
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
individuals [optional].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_loc.metafile">loc.metafile</code></td>
<td>
<p>Name of the csv file containing the metrics for
loci [optional].</p>
</td></tr>
<tr><td><code id="gl.read.csv_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SNP data need to be in one of two forms. SNPs can be coded 0 for
homozygous reference, 2 for homozygous alternate, 1 for heterozygous, and NA 
for missing values; or the SNP data can be coded A/A, A/C, C/T, G/A etc,
and -/- for missing data. In this format, the reference allele is the most 
frequent allele, as used by DArT. Other formats will throw an error.
</p>
<p>The SNP data need to be individuals as rows, labeled, and loci as columns,
also labeled. If the orientation is individuals as columns and loci by rows,
then set transpose=TRUE.
</p>
<p>The individual metrics need to be in a csv file, with headings, with a
mandatory id column corresponding exactly to the individual identity labels
provided with the SNP data and in the same order.
</p>
<p>The locus metadata needs to be in a csv file with headings, with a mandatory
column headed AlleleID corresponding exactly to the locus identity labels
provided with the SNP data and in the same order.
</p>
<p>Note that the locus metadata will be complemented by calculable statistics
corresponding to those that would be provided by Diversity Arrays Technology
(e.g. CallRate).
</p>


<h3>Value</h3>

<p>A genlight object with the SNP data and associated metadata included.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>csv_file &lt;- system.file('extdata','platy_test.csv', package='dartR')
ind_metadata &lt;- system.file('extdata','platy_ind.csv', package='dartR')
gl  &lt;- gl.read.csv(filename = csv_file, ind.metafile = ind_metadata)
</code></pre>

<hr>
<h2 id='gl.read.dart'>Imports DArT data into dartR and converts it into a genlight object</h2><span id='topic+gl.read.dart'></span>

<h3>Description</h3>

<p>This function is a wrapper function that allows you to convert your DArT file
into a genlight object in one step. In previous versions you had to use
read.dart and then dart2genlight. In case you have individual metadata for
each individual/sample you can specify as before in the dart2genlight command
the file that combines the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.dart(
  filename,
  ind.metafile = NULL,
  recalc = TRUE,
  mono.rm = FALSE,
  nas = "-",
  topskip = NULL,
  lastmetric = "RepAvg",
  covfilename = NULL,
  service_row = 1,
  plate_row = 3,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.dart_+3A_filename">filename</code></td>
<td>
<p>File containing the SNP data (csv file) [required].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>File that contains additional information on individuals
[required].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_recalc">recalc</code></td>
<td>
<p>Force the recalculation of locus metrics, in case individuals
have been manually deleted from the input csv file [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Force the removal of monomorphic loci (including all NAs), in
case individuals have been manually deleted from the input csv file
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_nas">nas</code></td>
<td>
<p>A character specifying NAs [default '-'].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_topskip">topskip</code></td>
<td>
<p>A number specifying the number of rows to be skipped. If not
provided the number of rows to be skipped are 'guessed' by the number of rows
with '*' at the beginning [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Specifies the last non-genetic column (Default is 'RepAvg').
Be sure to check if that is true, otherwise the number of individuals will
not match. You can also specify the last column by a number
[default 'RepAvg'].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_covfilename">covfilename</code></td>
<td>
<p>Use ind.metafile parameter [depreciated, NULL].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_service_row">service_row</code></td>
<td>
<p>The row number in which the information of the DArT service
is contained [default 1].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_plate_row">plate_row</code></td>
<td>
<p>The row number in which the information of the plate location
is contained [default 3].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.dart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, or as set by gl.set.verbose()].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The dartR genlight object can then be fed into a number of initial screening,
export and export functions provided by the package. For some of the
functions it is necessary to have the metadata that was provided from DArT.
Please check the vignette for more information. Additional information can
also be found in the help documents for <code><a href="#topic+utils.read.dart">utils.read.dart</a></code>.
</p>


<h3>Value</h3>

<p>A genlight object that contains individual metrics
[if data were provided] and locus metrics [from a DArT report].
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utils.read.dart">utils.read.dart</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dartfile &lt;- system.file('extdata','testset_SNPs_2Row.csv', package='dartR')
metadata &lt;- system.file('extdata','testset_metadata.csv', package='dartR')
gl &lt;- gl.read.dart(dartfile, ind.metafile = metadata, probar=TRUE)

</code></pre>

<hr>
<h2 id='gl.read.fasta'>Reads FASTA files and converts them to genlight object</h2><span id='topic+gl.read.fasta'></span>

<h3>Description</h3>

<p>The following IUPAC Ambiguity Codes are taken as heterozygotes:
</p>

<ul>
<li><p> M is heterozygote for	AC and CA	
</p>
</li>
<li><p> R is heterozygote	for AG and GA	
</p>
</li>
<li><p> W is heterozygote	for AT and TA	
</p>
</li>
<li><p> S is heterozygote	for CG and GC
</p>
</li>
<li><p> Y is heterozygote	for CT and TC
</p>
</li>
<li><p> K is heterozygote	for GT and TG
</p>
</li></ul>

<p>The following IUPAC Ambiguity Codes are taken as missing data:
</p>

<ul>
<li><p> V
</p>
</li>
<li><p> H 
</p>
</li>
<li><p> D 
</p>
</li>
<li><p> B
</p>
</li>
<li><p> N 
</p>
</li></ul>

<p>The function can deal with missing data in individuals, e.g. when FASTA 
files have different number of individuals due to missing data.
</p>
<p>The allele with the highest frequency is taken as the reference allele.
</p>
<p>SNPs with more than two alleles are skipped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.fasta(fasta_files, parallel = FALSE, n_cores = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.fasta_+3A_fasta_files">fasta_files</code></td>
<td>
<p>Fasta files to read [required].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_parallel">parallel</code></td>
<td>
<p>A logical indicating whether multiple cores -if available-
should be used for the computations (TRUE), or not (FALSE); requires the
package parallel to be installed [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_n_cores">n_cores</code></td>
<td>
<p>If parallel is TRUE, the number of cores to be used in the
computations; if NULL, then the maximum number of cores available on the
computer is used [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.fasta_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Ambiguity characters are often used to code heterozygotes. However, using
heterozygotes as ambiguity characters may bias many estimates. See more
information in the link below:
<a href="https://evodify.com/heterozygotes-ambiguity-characters/">https://evodify.com/heterozygotes-ambiguity-characters/</a>
</p>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> # Folder where the fasta files are located. 
 folder_samples &lt;- system.file('extdata', package ='dartR')
 # listing the FASTA files, including their path. Files have an extension
 # that contains "fas".
 file_names &lt;- list.files(path = folder_samples, pattern = "*.fas", 
                          full.names = TRUE)
 # reading fasta files
  obj &lt;- gl.read.fasta(file_names)
</code></pre>

<hr>
<h2 id='gl.read.silicodart'>Imports presence/absence data from SilicoDArT to genlight {agegenet}
format (ploidy=1)</h2><span id='topic+gl.read.silicodart'></span>

<h3>Description</h3>

<p>DaRT provide the data as a matrix of entities (individual animals) across the
top and attributes (P/A of sequenced fragment) down the side in a format
that is unique to DArT. This program reads the data in to adegenet format
for consistency with other programming activity. The script may require
modification as DArT modify their data formats from time to time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.silicodart(
  filename,
  ind.metafile = NULL,
  nas = "-",
  topskip = NULL,
  lastmetric = "Reproducibility",
  probar = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.silicodart_+3A_filename">filename</code></td>
<td>
<p>Name of csv file containing the SilicoDArT data [required].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Name of csv file containing metadata assigned to each
entity (individual) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_nas">nas</code></td>
<td>
<p>Missing data character [default '-'].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_topskip">topskip</code></td>
<td>
<p>Number of rows to skip before the header row (containing the
specimen identities) [optional].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Specifies the last non genetic column (Default is
'Reproducibility'). Be sure to check if that is true, otherwise the number of
individuals will not match. You can also specify the last column by a number
[default &quot;Reproducibility&quot;].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.read.silicodart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, or as set by gl.set.verbose()].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>gl.read.silicodart() opens the data file (csv comma delimited) and skips the
first n=topskip lines. The script assumes that the next line contains the
entity labels (specimen ids) followed immediately by the SNP data for the
first locus.
</p>
<p>It reads the presence/absence data into a matrix of 1s and 0s, and inputs the
locus metadata and specimen metadata. The locus metadata comprises a series
of columns of values for each locus including the essential columns of
CloneID and the desirable variables Reproducibility and PIC. Refer to
documentation provide by DArT for an explanation of these columns.
</p>
<p>The specimen metadata provides the opportunity to reassign specimens to
populations, and to add other data relevant to the specimen. The key
variables are id (specimen identity which must be the same and in the same
order as the SilicoDArT file, each unique), pop (population assignment), lat
(latitude, optional) and lon (longitude, optional). id, pop, lat, lon are
the column headers in the csv file. Other optional columns can be added.
</p>
<p>The data matrix, locus names (forced to be unique), locus metadata, specimen
names, specimen metadata are combined into a genind object. Refer to the
documentation for {adegenet} for further details.
</p>


<h3>Value</h3>

<p>An object of class <code>genlight</code> with ploidy set to 1, containing
the presence/absence data, and locus and individual metadata.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.read.dart">gl.read.dart</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>silicodartfile &lt;- system.file('extdata','testset_SilicoDArT.csv', package='dartR')
metadata &lt;- system.file('extdata',ind.metafile ='testset_metadata_silicodart.csv', package='dartR')
testset.gs &lt;- gl.read.silicodart(filename = silicodartfile, ind.metafile = metadata)
</code></pre>

<hr>
<h2 id='gl.read.vcf'>Converts a vcf file into a genlight object</h2><span id='topic+gl.read.vcf'></span>

<h3>Description</h3>

<p>This function needs package vcfR, please install it. The converted genlight
object does not have individual metrics. You need to add them 'manually' to
the other$ind.metrics slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.read.vcf(vcffile, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.read.vcf_+3A_vcffile">vcffile</code></td>
<td>
<p>A vcf file (works only for diploid data) [required].</p>
</td></tr>
<tr><td><code id="gl.read.vcf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
obj &lt;- gl.read.vcf(system.file('extdata/test.vcf', package='dartR'))

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.reassign.pop'>Assigns an individual metric as pop in a genlight {adegenet} object</h2><span id='topic+gl.reassign.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the
individual/sample/specimen metrics file (csv) used with gl.read.dart().
</p>
<p>One might want to define the population structure in accordance with another
classification, such as using an individual metric (e.g. sex, male or
female). This script discards the current population assignments and replaces
them with new population assignments defined by a specified individual
metric.
</p>
<p>The script returns a genlight object with the new population assignments.
Note that the original population assignments are lost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.reassign.pop(x, as.pop, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.reassign.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.reassign.pop_+3A_as.pop">as.pop</code></td>
<td>
<p>Specify the name of the individual metric to set as the pop
variable [required].</p>
</td></tr>
<tr><td><code id="gl.reassign.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the reassigned populations.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
   popNames(testset.gl)
   gl &lt;- gl.reassign.pop(testset.gl, as.pop='sex',verbose=3)
   popNames(gl)
# Tag P/A data
   popNames(testset.gs)
   gs &lt;- gl.reassign.pop(testset.gs, as.pop='sex',verbose=3)
   popNames(gs)
</code></pre>

<hr>
<h2 id='gl.recalc.metrics'>Recalculates locus metrics when individuals or populations are deleted from a
genlight {adegenet} object</h2><span id='topic+gl.recalc.metrics'></span>

<h3>Description</h3>

<p>When individuals,or populations, are deleted from a genlight object, the
locus metrics no longer apply. For example, the Call Rate may be different
considering the subset of individuals, compared with the full set. This
script recalculates those affected locus metrics, namely, avgPIC, CallRate,
freqHets, freqHomRef, freqHomSnp, OneRatioRef, OneRatioSnp, PICRef and
PICSnp.
Metrics that remain unaltered are RepAvg and TrimmedSeq as they are
unaffected by the removal of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recalc.metrics(x, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recalc.metrics_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.recalc.metrics_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, removes monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recalc.metrics_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The script optionally removes resultant monomorphic loci or loci
with all values missing and deletes them (using gl.filter.monomorphs.r).
</p>
<p>The script returns a genlight object with the recalculated locus metadata.
</p>


<h3>Value</h3>

<p>A genlight object with the recalculated locus metadata.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  gl &lt;- gl.recalc.metrics(testset.gl, verbose=2)
</code></pre>

<hr>
<h2 id='gl.recode.ind'>Recodes individual (=specimen = sample) labels in a genlight object</h2><span id='topic+gl.recode.ind'></span>

<h3>Description</h3>

<p>This script recodes individual labels and/or deletes individuals from a DaRT
genlight SNP file based on a lookup table provided as a csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recode.ind(x, ind.recode, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recode.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_ind.recode">ind.recode</code></td>
<td>
<p>Name of the csv file containing the individual relabelling
[required].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_recalc">recalc</code></td>
<td>
<p>If TRUE, recalculate the locus metadata statistics if any 
individuals are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_mono.rm">mono.rm</code></td>
<td>
<p>If TRUE, remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.ind_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Renaming individuals may be required when there have been errors in labelling
arising in the process from sample to DArT files. There may be occasions
where renaming individuals is required for preparation of figures. When
caution needs to be exercised because of the potential for breaking the
'chain of evidence' associated with the samples, recoding individuals using
a recode table (csv) can provide a clear record of the changes.
</p>
<p>The script, having deleted individuals, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally recalculates
statistics made incorrect by the deletion of individuals from the dataset.
</p>
<p>The script returns a genlight object with the new individual labels, the
monomorphic loci optionally removed and the optionally recalculated locus
metadata.
</p>


<h3>Value</h3>

<p>A genlight or genind object with the recoded and reduced data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code> for filtering monomorphs,
<code><a href="#topic+gl.recalc.metrics">gl.recalc.metrics</a></code> for recalculating locus metrics,
<code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code> for recoding populations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  file &lt;- system.file('extdata','testset_ind_recode.csv', package='dartR')
  gl &lt;- gl.recode.ind(testset.gl, ind.recode=file, verbose=3)
</code></pre>

<hr>
<h2 id='gl.recode.pop'>Recodes population assignments in a genlight object</h2><span id='topic+gl.recode.pop'></span>

<h3>Description</h3>

<p>This script recodes population assignments and/or deletes populations from a
DaRT genlight SNP file based on information provided in a csv population
recode file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.recode.pop(x, pop.recode, recalc = FALSE, mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.recode.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_pop.recode">pop.recode</code></td>
<td>
<p>Name of the csv file containing the population
reassignments [required].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics if any individuals
are deleted in the filtering [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.recode.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart(). Recoding can be used to amalgamate
populations or to selectively delete or retain populations.
</p>
<p>The population recode file contains a list of populations in the genlight
object as the first column of the csv file, and the new population
assignments in the second column of the csv file. The keyword Delete used as
a new population assignment will result in the associated specimen being
dropped from the dataset.
</p>
<p>The script, having deleted populations, optionally identifies resultant
monomorphic loci or loci with all values missing and deletes them
(using gl.filter.monomorphs.r). The script also optionally recalculates the
locus metadata as appropriate.
</p>


<h3>Value</h3>

<p>A genlight object with the recoded and reduced data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>
<p><code><a href="#topic+gl.recode.pop">gl.recode.pop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  mfile &lt;- system.file('extdata', 'testset_pop_recode.csv', package='dartR')
  nPop(testset.gl)
  gl &lt;- gl.recode.pop(testset.gl, pop.recode=mfile, verbose=3)
 
</code></pre>

<hr>
<h2 id='gl.rename.pop'>Renames a population in a genlight object</h2><span id='topic+gl.rename.pop'></span>

<h3>Description</h3>

<p>Individuals are assigned to populations based on the specimen metadata data
file (csv) used with gl.read.dart().
</p>
<p>This script renames a nominated population.
</p>
<p>The script returns a genlight object with the new population name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.rename.pop(x, old = NULL, new = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.rename.pop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_old">old</code></td>
<td>
<p>Name of population to be changed [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_new">new</code></td>
<td>
<p>New name for the population [required].</p>
</td></tr>
<tr><td><code id="gl.rename.pop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with the new population name.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   gl &lt;- gl.rename.pop(testset.gl, old='EmsubRopeMata', new='Outgroup')
</code></pre>

<hr>
<h2 id='gl.report.bases'>Reports summary of base pair frequencies</h2><span id='topic+gl.report.bases'></span>

<h3>Description</h3>

<p>This script calculates the frequencies of the four DNA nucleotide bases:
adenine (A), cytosine (C), 'guanine (G) and thymine (T), and the frequency of
transitions (Ts) and transversions (Tv) in a DArT genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.bases(
  x,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.bases_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, histograms of base composition are produced
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE]</p>
</td></tr>
<tr><td><code id="gl.report.bases_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The script checks first if trimmed sequences are included in the
locus metadata (@other$loc.metrics$TrimmedSequence), and if so, tallies up
the numbers of A, T, G and C bases. Only the reference state at the SNP locus
is counted. Counts of transitions (Ts) and transversions (Tv) assume that
there is no directionality, that is C-&gt;T is the same as T-&gt;C, because the
reference state is arbitrary.
</p>
<p>For presence/absence data (SilicoDArT), it is not possible to count
transversions or transitions or transversions/transitions ratio because the
SNP data is not available, only a single sequence tag.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>The unchanged genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other report functions: 
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  out &lt;- gl.report.bases(testset.gl)
  #' # Tag P/A data
  out &lt;- gl.report.bases(testset.gs)

</code></pre>

<hr>
<h2 id='gl.report.callrate'>Reports summary of Call Rate for loci or individuals</h2><span id='topic+gl.report.callrate'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the restriction
enzyme recognition sites. P/A datasets (SilicoDArT) have missing values
because it was not possible to call whether a sequence tag was amplified or
not. This function tabulates the number of missing values as quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.callrate(
  x,
  method = "loc",
  by_pop = FALSE,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  bins = 50,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_method">method</code></td>
<td>
<p>Specify the type of report by locus (method='loc') or
individual (method='ind') [default 'loc'].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_by_pop">by_pop</code></td>
<td>
<p>Whether report by population [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function expects a genlight object, containing either SNP data or
SilicoDArT (=presence/absence data).
</p>
<p>Callrate is summarized by locus or by individual to allow sensible decisions
on thresholds for filtering taking into consideration consequential loss of
data. The summary is in the form of a tabulation and plots.
</p>
<p>Plot themes can be obtained from:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>

<p>Resultant ggplots and the tabulation are saved to the session's temporary
directory.
</p>


<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.callrate">gl.filter.callrate</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  test.gl &lt;- testset.gl[1:20,]
  gl.report.callrate(test.gl)
  gl.report.callrate(test.gl,method='ind')
# Tag P/A data
  test.gs &lt;- testset.gs[1:20,]
  gl.report.callrate(test.gs)
  gl.report.callrate(test.gs,method='ind')
  
  test.gl &lt;- testset.gl[1:20,]
  gl.report.callrate(test.gl)
  
</code></pre>

<hr>
<h2 id='gl.report.diversity'>Calculates diversity indexes for SNPs</h2><span id='topic+gl.report.diversity'></span>

<h3>Description</h3>

<p>This script takes a genlight object and calculates alpha and beta diversity
for q = 0:2. Formulas are taken from Sherwin et al. 2017. The paper describes
nicely the relationship between the different q levels and how they relate to
population genetic processes such as dispersal and selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.diversity(
  x,
  plot.out = TRUE,
  pbar = TRUE,
  table = "DH",
  plot_theme = theme_dartR(),
  plot_colors = discrete_palette,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.diversity_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_pbar">pbar</code></td>
<td>
<p>Report on progress. Silent if set to FALSE [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_table">table</code></td>
<td>
<p>Prints a tabular output to the console either 'D'=D values, or
'H'=H values or 'DH','HD'=both or 'N'=no table. [default 'DH'].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_plot_colors">plot_colors</code></td>
<td>
<p>A color palette or a list with as many colors as there are 
populations in the dataset [default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.diversity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For all indexes, the entropies (H) and corresponding effective
numbers, i.e. Hill numbers (D), which reflect the number of needed entities
to get the observed values, are calculated. In a nutshell, the alpha indexes
between the different q-values should be similar if there is no deviation
from expected allele frequencies and occurrences (e.g. all loci in HWE &amp;
equilibrium). If there is a deviation of an index, this links to a process
causing it, such as dispersal, selection or strong drift. For a detailed
explanation of all the indexes, we recommend resorting to the literature
provided below. Confidence intervals are +/- 1 standard deviation.
</p>
<p><strong> Function's output </strong>
</p>
<p>If the function's parameter &quot;table&quot; = &quot;DH&quot; (the default value) is used, the 
output of the function is 20 tables.
</p>
<p>The first two show the number of loci used. The name of each of the rest of 
the tables starts with three terms separated by underscores.
</p>
<p>The first term refers to the q value (0 to 2).
</p>
<p>The second term refers to whether it is the diversity measure (H) or its 
transformation to Hill numbers (D). 
</p>
<p>The third term refers to whether the diversity is calculated within 
populations (alpha) or between populations (beta). 
</p>
<p>In the case of alpha diversity tables, standard deviations have their own 
table, which finishes with a fourth term: &quot;sd&quot;.
</p>
<p>In the case of beta diversity tables, standard deviations are in the upper 
triangle of the matrix and diversity values are in the lower triangle of the 
matrix.
</p>
<p>Plots are saved to the temporal directory (tempdir) and can be accessed with
the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with the function
<code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed only in the
current R session because tempdir is cleared each time that the R session
is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A list of entropy indexes for each level of q and equivalent numbers
for alpha and beta diversity.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>),
Contributors: William B. Sherwin, Alexander Sentinella
</p>


<h3>References</h3>

<p>Sherwin, W.B., Chao, A., Johst, L., Smouse, P.E. (2017). Information Theory
Broadens the Spectrum of Molecular Ecology and Evolution. TREE 32(12)
948-963. doi:10.1016/j.tree.2017.09.12
</p>


<h3>See Also</h3>

<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>div &lt;- gl.report.diversity(bandicoot.gl[1:10,1:100], table = FALSE,
 pbar=FALSE)
div$zero_H_alpha
div$two_H_beta
names(div)

</code></pre>

<hr>
<h2 id='gl.report.hamming'>Calculates the pairwise Hamming distance between DArT trimmed DNA
sequences</h2><span id='topic+gl.report.hamming'></span>

<h3>Description</h3>

<p>Hamming distance is calculated as the number of base differences
between two sequences which can be expressed as a count or a proportion.
Typically, it is calculated between two sequences of equal length. In the
context of DArT trimmed sequences, which differ in length but which are
anchored to the left by the restriction enzyme recognition sequence, it is
sensible to compare the two trimmed sequences starting from immediately after
the common recognition sequence and terminating at the last base of the
shorter sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.hamming(
  x,
  rs = 5,
  threshold = 3,
  taglength = 69,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  probar = FALSE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.hamming_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_rs">rs</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 5].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_threshold">threshold</code></td>
<td>
<p>Minimum acceptable base pair difference for display on the
boxplot and histogram [default 3].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_taglength">taglength</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_probar">probar</code></td>
<td>
<p>If TRUE, then a progress bar is displayed on long loops
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.hamming_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a></code> will filter out one of
two loci if their Hamming distance is less than a specified percentage
</p>
<p>Hamming distance can be computed by exploiting the fact that the dot product
of two binary vectors x and (1-y) counts the corresponding elements that are
different between x and y. This approach can also be used for vectors that
contain more than two possible values at each position (e.g. A, C, T or G).
</p>
<p>If a pair of DNA sequences are of differing length, the longer is truncated.
</p>
<p>The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
as implemented in <code><a href="#topic+utils.hamming">utils.hamming</a></code>
</p>
<p>Plots and table are saved to the session's temporary directory (tempdir)
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.hamming">gl.filter.hamming</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
gl.report.hamming(testset.gl[,1:100])
gl.report.hamming(testset.gs[,1:100])


#' # SNP data
test &lt;- platypus.gl
test &lt;- gl.subsample.loci(platypus.gl,n=50)
result &lt;- gl.filter.hamming(test, threshold=0.25, verbose=3)

</code></pre>

<hr>
<h2 id='gl.report.heterozygosity'>Reports observed, expected and unbiased heterozygosities and FIS
(inbreeding coefficient) by population or by individual from SNP data</h2><span id='topic+gl.report.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the observed, expected and unbiased expected (i.e.
corrected for sample size) heterozygosities and FIS (inbreeding coefficient)
for each population or the observed heterozygosity for each individual in a
genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.heterozygosity(
  x,
  method = "pop",
  n.invariant = 0,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = discrete_palette,
  plot_colors_ind = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.heterozygosity_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP [required].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_method">method</code></td>
<td>
<p>Calculate heterozygosity by population (method='pop') or by
individual (method='ind') [default 'pop'].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_n.invariant">n.invariant</code></td>
<td>
<p>An estimate of the number of invariant sequence tags used
to adjust the heterozygosity rate [default 0].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>A color palette for population plots or a list with
as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_plot_colors_ind">plot_colors_ind</code></td>
<td>
<p>List of two color names for the borders and fill of
the plot by individual [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Observed heterozygosity for a population takes the proportion of
heterozygous loci for each individual then averages over the individuals in
that population. The calculations take into account missing values.
</p>
<p>Expected heterozygosity for a population takes the expected proportion of
heterozygotes, that is, expected under Hardy-Weinberg equilibrium, for each
locus, then averages this across the loci for an average estimate for the
population.
</p>
<p>Observed heterozygosity for individuals is calculated as the proportion of
loci that are heterozygous for that individual.
</p>
<p>Finally, the loci that are invariant across all individuals in the dataset
(that is, across populations), is typically unknown. This can render
estimates of heterozygosity analysis specific, and so it is not valid to
compare such estimates across species or even across different analyses. This
is a similar problem faced by microsatellites. If you have an estimate of the
number of invariant sequence tags (loci) in your data, such as provided by
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a></code>, you can specify it with the n.invariant
parameter to standardize your estimates of heterozygosity.
</p>
<p><strong>NOTE</strong>: It is important to realise that estimation of adjusted
heterozygosity requires that secondaries not to be removed.
</p>
<p>Heterozygosities and FIS (inbreeding coefficient) are calculated by locus
within each population using the following equations:
</p>

<ul>
<li><p> Observed heterozygosity (Ho) = number of homozygotes / n_Ind,
where n_Ind is the number of individuals without missing data.
</p>
</li>
<li><p> Observed heterozygosity adjusted (Ho.adj) &lt;- Ho * n_Loc /
(n_Loc + n.invariant),
where n_Loc is the number of loci that do not have all missing data  and
n.invariant is an estimate of the number of invariant loci to adjust
heterozygosity.
</p>
</li>
<li><p> Expected heterozygosity (He) = 1 - (p^2 + q^2),
where p is the frequency of the reference allele and q is the frequency of
the alternative allele.
</p>
</li>
<li><p> Expected heterozygosity adjusted (He.adj) = He * n_Loc /
(n_Loc + n.invariant)
</p>
</li>
<li><p> Unbiased expected heterozygosity (uHe) = He * (2 * n_Ind /
(2 * n_Ind - 1))
</p>
</li>
<li><p> Inbreeding coefficient (FIS) = 1 - (mean(Ho) / mean(uHe))
</p>
</li></ul>

<p><strong> Function's output </strong>
</p>
<p>Output for method='pop' is an ordered barchart of observed heterozygosity,
unbiased expected heterozygosity and FIS (Inbreeding coefficient) across populations
together with a table of mean observed and expected heterozygosities and FIS
by population and their respective standard deviations (SD).
</p>
<p>In the output, it is also reported by population: the number of loci used to
estimate heterozygosity(nLoc), the number of polymorphic loci (polyLoc), 
the number of monomorphic loci (monoLoc) and loci with all missing data
(all_NALoc).
</p>
<p>Output for method='ind' is a histogram and a boxplot of heterozygosity across
individuals.
</p>
<p>Plots and table are saved to the session temporary directory (tempdir)
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A dataframe containing population labels, heterozygosities, FIS,
their standard deviations and sample sizes
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.heterozygosity">gl.filter.heterozygosity</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
require("dartR.data")
df &lt;- gl.report.heterozygosity(platypus.gl)
df &lt;- gl.report.heterozygosity(platypus.gl,method='ind')
n.inv &lt;- gl.report.secondaries(platypus.gl)
gl.report.heterozygosity(platypus.gl, n.invariant = n.inv[7, 2])

df &lt;- gl.report.heterozygosity(platypus.gl)

</code></pre>

<hr>
<h2 id='gl.report.hwe'>Reports departure from Hardy-Weinberg proportions</h2><span id='topic+gl.report.hwe'></span>

<h3>Description</h3>

<p>Calculates the probabilities of agreement with H-W proportions based on 
observed
frequencies of reference homozygotes, heterozygotes and alternate 
homozygotes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.hwe(
  x,
  subset = "each",
  method_sig = "Exact",
  multi_comp = FALSE,
  multi_comp_method = "BY",
  alpha_val = 0.05,
  pvalue_type = "midp",
  cc_val = 0.5,
  sig_only = TRUE,
  min_sample_size = 5,
  plot.out = TRUE,
  plot_colors = two_colors_contrast,
  max_plots = 4,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.hwe_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_subset">subset</code></td>
<td>
<p>Way to group individuals to perform H-W tests. Either a vector
with population names, 'each', 'all' (see details) [default 'each'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_method_sig">method_sig</code></td>
<td>
<p>Method for determining statistical significance: 
'ChiSquare'
or 'Exact' [default 'Exact'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_multi_comp">multi_comp</code></td>
<td>
<p>Whether to adjust p-values for multiple comparisons
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_multi_comp_method">multi_comp_method</code></td>
<td>
<p>Method to adjust p-values for multiple comparisons:
'holm', 'hochberg', 'hommel', 'bonferroni', 'BH', 'BY', 'fdr'
(see details) [default 'fdr'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_alpha_val">alpha_val</code></td>
<td>
<p>Level of significance for testing [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_pvalue_type">pvalue_type</code></td>
<td>
<p>Type of p-value to be used in the Exact method.
Either 'dost','selome','midp' (see details) [default 'midp'].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_cc_val">cc_val</code></td>
<td>
<p>The continuity correction applied to the ChiSquare test
[default 0.5].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_sig_only">sig_only</code></td>
<td>
<p>Whether the returned table should include loci with a 
significant departure from Hardy-Weinberg proportions [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_min_sample_size">min_sample_size</code></td>
<td>
<p>Minimum number of individuals per population in which
perform H-W tests [default 5].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, will produce Ternary Plot(s) [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with two color names for the significant and
not-significant loci [default two_colors_contrast].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_max_plots">max_plots</code></td>
<td>
<p>Maximum number of plots to print per page [default 4].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.hwe_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are several factors that can cause deviations from Hardy-Weinberg
proportions including: mutation, finite population size, selection,
population structure, age structure, assortative mating, sex linkage,
nonrandom sampling and genotyping errors. Therefore, testing for
Hardy-Weinberg proportions should be a process that involves a careful
evaluation of the results, a good place to start is Waples (2015).
</p>
<p>Note that tests for H-W proportions are only valid if there is no population
substructure (assuming random mating) and have sufficient power only when
there is sufficient sample size (n individuals &gt; 15).
</p>
<p>Populations can be defined in three ways:
</p>

<ul>
<li><p> Merging all populations in the dataset using subset = 'all'.
</p>
</li>
<li><p> Within each population separately using: subset = 'each'.
</p>
</li>
<li><p> Within selected populations using for example: subset =
c('pop1','pop2').
</p>
</li></ul>

<p>Two different statistical methods to test for deviations from Hardy Weinberg
proportions:
</p>

<ul>
<li><p> The classical chi-square test (method_sig='ChiSquare') based on the
function <code><a href="HardyWeinberg.html#topic+HWChisq">HWChisq</a></code> of the R package HardyWeinberg.
By default a continuity correction is applied (cc_val=0.5). The
continuity correction can be turned off (by specifying cc_val=0), for example
in cases of extreme allele frequencies in which the continuity correction can
lead to excessive type 1 error rates.
</p>
</li>
<li><p> The exact test (method_sig='Exact') based on the exact calculations
contained in the function <code><a href="HardyWeinberg.html#topic+HWExactStats">HWExactStats</a></code> of the R
package HardyWeinberg, and described in Wigginton  et al. (2005). The exact
test is recommended in most cases (Wigginton  et al., 2005).
Three different methods to estimate p-values (pvalue_type) in the Exact test
can be used:
</p>

<ul>
<li><p> 'dost' p-value is computed as twice the tail area of a one-sided test.
</p>
</li>
<li><p> 'selome' p-value is computed as the sum of the probabilities of all
samples less or equally likely as the current sample.
</p>
</li>
<li><p> 'midp', p-value is computed as half the probability of the current
sample + the probabilities of all samples that are more extreme.
</p>
</li></ul>

<p>The standard exact p-value is overly conservative, in particular
for small minor allele frequencies. The mid p-value ameliorates this problem
by bringing the rejection rate closer to the nominal level, at the price of
occasionally exceeding the nominal level (Graffelman &amp; Moreno, 2013).
</p>
</li></ul>

<p>Correction for multiple tests can be applied using the following methods
based on the function <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>:
</p>

<ul>
<li><p> 'holm' is also known as the sequential Bonferroni technique (Rice,
1989). This method has a greater statistical power than the standard
Bonferroni test, however this method becomes very stringent when many tests
are performed and many real deviations from the null hypothesis can go
undetected (Waples, 2015).
</p>
</li>
<li><p> 'hochberg' based on Hochberg, 1988.
</p>
</li>
<li><p> 'hommel' based on Hommel, 1988. This method is more powerful than
Hochberg's, but the difference is usually small.
</p>
</li>
<li><p> 'bonferroni' in which p-values are multiplied by the number of tests.
This method is very stringent and therefore has reduced power to detect
multiple departures from the null hypothesis.
</p>
</li>
<li><p> 'BH' based on Benjamini &amp; Hochberg, 1995.
</p>
</li>
<li><p> 'BY' based on Benjamini &amp; Yekutieli, 2001.
</p>
</li></ul>

<p>The first four methods are designed to give strong control of the family-wise
error rate. The last two methods control the false discovery rate (FDR),
the expected proportion of false discoveries among the rejected hypotheses.
The false discovery rate is a less stringent condition than the family-wise
error rate, so these methods are more powerful than the others, especially
when number of tests is large.
The number of tests on which the adjustment for multiple comparisons is
the number of populations times the number of loci.
</p>
<p><strong>Ternary plots</strong>
</p>
<p>Ternary plots can be used to visualise patterns of H-W proportions (plot.out
= TRUE). P-values and the statistical (non)significance of a large number of
bi-allelic markers can be inferred from their position in a ternary plot.
See Graffelman &amp; Morales-Camarena (2008) for further details. Ternary plots
are based on the function  <code><a href="HardyWeinberg.html#topic+HWTernaryPlot">HWTernaryPlot</a></code> from
the package HardyWeinberg. Each vertex of the Ternary plot represents one of 
the three possible genotypes for SNP data: homozygous for the reference 
allele (AA), heterozygous (AB) and homozygous for the alternative allele
(BB). Loci deviating significantly from Hardy-Weinberg proportions after 
correction for multiple tests are shown in pink. The blue parabola 
represents Hardy-Weinberg equilibrium, and the area between green lines 
represents the acceptance region.
</p>
<p>For these plots to work it is necessary to install the package ggtern.
</p>


<h3>Value</h3>

<p>A dataframe containing loci, counts of reference SNP homozygotes,
heterozygotes and alternate SNP homozygotes; probability of departure from
H-W proportions, per locus significance with and without correction for
multiple comparisons and the number of population where the same locus is 
significantly out of HWE.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Benjamini, Y., and Yekutieli, D. (2001). The control of the false
discovery rate in multiple testing under dependency. Annals of Statistics,
29, 1165–1188.
</p>
</li>
<li><p> Graffelman, J. (2015). Exploring Diallelic Genetic Markers: The Hardy
Weinberg Package. Journal of Statistical Software 64:1-23.
</p>
</li>
<li><p> Graffelman, J. &amp; Morales-Camarena, J. (2008). Graphical tests for
Hardy-Weinberg equilibrium based on the ternary plot. Human Heredity
65:77-84.
</p>
</li>
<li><p> Graffelman, J., &amp; Moreno, V. (2013). The mid p-value in exact tests for
Hardy-Weinberg equilibrium. Statistical applications in genetics and
molecular biology, 12(4), 433-448.
</p>
</li>
<li><p> Hochberg, Y. (1988). A sharper Bonferroni procedure for multiple tests
of significance. Biometrika, 75, 800–803.
</p>
</li>
<li><p> Hommel, G. (1988). A stagewise rejective multiple test procedure based
on a modified Bonferroni test. Biometrika, 75, 383–386.
</p>
</li>
<li><p> Rice, W. R. (1989). Analyzing tables of statistical tests. Evolution,
43(1), 223-225.
</p>
</li>
<li><p> Waples, R. S. (2015). Testing for Hardy–Weinberg proportions: have we
lost the plot?. Journal of heredity, 106(1), 1-19.
</p>
</li>
<li><p> Wigginton, J.E., Cutler, D.J., &amp; Abecasis, G.R. (2005). A Note on Exact
Tests of Hardy-Weinberg Equilibrium. American Journal of Human Genetics
76:887-893.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.hwe">gl.filter.hwe</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>

<hr>
<h2 id='gl.report.ld'>Calculates pairwise population based Linkage Disequilibrium across all loci
using the specified number of cores</h2><span id='topic+gl.report.ld'></span>

<h3>Description</h3>

<p>This function is implemented in a parallel fashion to speed up the process.
There is also the ability to restart the function if crashed by specifying
the chunk file names or restarting the function exactly in the same way as in
the first run. This is implemented because sometimes, due to connectivity loss
between cores, the function may crash half way. Before running the function, 
it is advisable to use the function <code><a href="#topic+gl.filter.allna">gl.filter.allna</a></code> to remove
loci with all missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.ld(
  x,
  name = NULL,
  save = TRUE,
  outpath = tempdir(),
  nchunks = 2,
  ncores = 1,
  chunkname = NULL,
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.ld_+3A_x">x</code></td>
<td>
<p>A genlight or genind object created (genlight objects are internally
converted via <code><a href="#topic+gl2gi">gl2gi</a></code> to genind) [required].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_name">name</code></td>
<td>
<p>Character string for rdata file. If not given genind object name
is used [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_save">save</code></td>
<td>
<p>Switch if results are saved in a file [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_outpath">outpath</code></td>
<td>
<p>Folder where chunks and results are saved (if save=TRUE)
[default tempdir()].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_nchunks">nchunks</code></td>
<td>
<p>How many subchunks will be used (the less the faster, but if
the routine crashes more bits are lost) [default 2].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_ncores">ncores</code></td>
<td>
<p>How many cores should be used [default 1].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_chunkname">chunkname</code></td>
<td>
<p>The name of the chunks for saving [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_probar">probar</code></td>
<td>
<p>if TRUE, a progress bar is displayed for long loops
[default = TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns calculation of pairwise LD across all loci between
subpopulations. This functions uses if specified many cores on your computer
to speed up. And if save is used can restart (if save=TRUE is used) with the
same command starting where it crashed. The final output is a data frame that
holds all statistics of pairwise LD between loci. (See ?LD in package
genetics for details).
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl.report.ld.map'>Calculates pairwise linkage disequilibrium by population</h2><span id='topic+gl.report.ld.map'></span>

<h3>Description</h3>

<p>This function calculates pairwise linkage disequilibrium (LD) by population 
using the function <code><a href="snpStats.html#topic+ld">ld</a></code> (package snpStats).
</p>
<p>If SNPs are not mapped to a reference genome, the parameter
<code>ld_max_pairwise</code>
should be set as NULL (the default). In this case, the 
function will assign the same chromosome (&quot;1&quot;) to all the SNPs in the dataset
and assign a sequence from 1 to n loci as the position of each SNP. The 
function will then calculate LD for all possible SNP pair combinations. 
</p>
<p>If SNPs are mapped to a reference genome, the parameter 
<code>ld_max_pairwise</code>
should be filled out (i.e. not NULL). In this case, the
information for SNP's position should be stored in the genlight accessor
&quot;@position&quot; and the SNP's chromosome name in the accessor &quot;@chromosome&quot;
(see examples). The function will then calculate LD within each chromosome
and for all possible SNP pair combinations within a distance of
<code>ld_max_pairwise</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.ld.map(
  x,
  ld_max_pairwise = NULL,
  maf = 0.05,
  ld_stat = "R.squared",
  ind.limit = 10,
  stat_keep = "AvgPIC",
  ld_threshold_pops = 0.2,
  plot.out = TRUE,
  plot_theme = NULL,
  histogram_colors = NULL,
  boxplot_colors = NULL,
  bins = 50,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.ld.map_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld_max_pairwise">ld_max_pairwise</code></td>
<td>
<p>Maximum distance in number of base pairs at which LD 
should be calculated [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_maf">maf</code></td>
<td>
<p>Minor allele frequency (by population) threshold to filter out 
loci. If a value &gt; 1 is provided it will be interpreted as MAC (i.e. the
minimum number of times an allele needs to be observed) [default 0.05].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld_stat">ld_stat</code></td>
<td>
<p>The LD measure to be calculated: &quot;LLR&quot;, &quot;OR&quot;, &quot;Q&quot;, &quot;Covar&quot;,
&quot;D.prime&quot;, &quot;R.squared&quot;, and &quot;R&quot;. See <code><a href="snpStats.html#topic+ld">ld</a></code>
(package snpStats) for details [default &quot;R.squared&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Minimum number of individuals that a population should
contain to take it in account to report loci in LD [default 10].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_stat_keep">stat_keep</code></td>
<td>
<p>Name of the column from the slot <code>loc.metrics</code> to be
used to choose SNP to be kept [default &quot;AvgPIC&quot;].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_ld_threshold_pops">ld_threshold_pops</code></td>
<td>
<p>LD threshold to report in the plot of &quot;Number of 
populations in which the same SNP pair are in LD&quot; [default 0.2].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_plot_theme">plot_theme</code></td>
<td>
<p>User specified theme [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_histogram_colors">histogram_colors</code></td>
<td>
<p>Vector with two color names for the borders and fill
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_boxplot_colors">boxplot_colors</code></td>
<td>
<p>A color palette for box plots by population or a list
with as many colors as there are populations in the dataset
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 50].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.ld.map_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reports LD between SNP pairs by population. 
The function <code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code> filters out the SNPs in LD using as
input the results of <code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a></code>. The actual number of 
SNPs to be filtered out depends on the parameters set in the function 
<code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code>.
</p>
<p>Boxplots of LD by population and
a histogram showing LD frequency are presented.
</p>


<h3>Value</h3>

<p>A dataframe with information for each SNP pair in LD.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.ld">gl.filter.ld</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
x &lt;- platypus.gl
x &lt;- gl.filter.callrate(x,threshold = 1)
x &lt;- gl.filter.monomorphs(x)
x$position &lt;- x$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
x$chromosome &lt;- as.factor(x$other$loc.metrics$Chrom_Platypus_Chrom_NCBIv1)
ld_res &lt;- gl.report.ld.map(x,ld_max_pairwise = 10000000)
</code></pre>

<hr>
<h2 id='gl.report.locmetric'>Reports summary of the slot $other$loc.metrics</h2><span id='topic+gl.report.locmetric'></span>

<h3>Description</h3>

<p>This script uses any field with numeric values stored in $other$loc.metrics
to produce summary statistics (mean, minimum, average, quantiles), histograms
and boxplots to assist the decision of choosing thresholds for the filter
function <code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.locmetric(
  x,
  metric,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.locmetric_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_metric">metric</code></td>
<td>
<p>Name of the metric to be used for filtering [required].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.locmetric_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code> will filter out the
loci with a locmetric value below a specified threshold.
</p>
<p>The fields that are included in dartR, and a short description, are found
below. Optionally, the user can also set his/her own field by adding a vector
into $other$loc.metrics as shown in the example. You can check the names of
all available loc.metrics via: names(gl$other$loc.metrics).
</p>

<ul>
<li><p> SnpPosition - position (zero is position 1) in the sequence tag of the
defined SNP variant base.
</p>
</li>
<li><p> CallRate - proportion of samples for which the genotype call is
non-missing (that is, not '-' ).
</p>
</li>
<li><p> OneRatioRef - proportion of samples for which the genotype score is 0.
</p>
</li>
<li><p> OneRatioSnp - proportion of samples for which the genotype score is 2.
</p>
</li>
<li><p> FreqHomRef - proportion of samples homozygous for the Reference allele.
</p>
</li>
<li><p> FreqHomSnp - proportion of samples homozygous for the Alternate (SNP)
allele.
</p>
</li>
<li><p> FreqHets - proportion of samples which score as heterozygous, that is,
scored as 1.
</p>
</li>
<li><p> PICRef - polymorphism information content (PIC) for the Reference allele.
</p>
</li>
<li><p> PICSnp - polymorphism information content (PIC) for the SNP.
</p>
</li>
<li><p> AvgPIC - average of the polymorphism information content (PIC) of the
reference and SNP alleles.
</p>
</li>
<li><p> AvgCountRef - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Reference allele row.
</p>
</li>
<li><p> AvgCountSnp - sum of the tag read counts for all samples, divided by the
number of samples with non-zero tag read counts, for the Alternate (SNP) allele
row.
</p>
</li>
<li><p> RepAvg - proportion of technical replicate assay pairs for which the
marker score is consistent.
</p>
</li>
<li><p> rdepth - read depth.
</p>
</li></ul>

<p><strong> Function's output </strong>
</p>
<p>The minimum, maximum, mean and a tabulation of quantiles of the locmetric
values against thresholds rate are provided. Output also includes a boxplot
and a histogram.
</p>
<p>Quantiles are partitions of a finite set of values into q subsets of (nearly)
equal sizes. In this function q = 20. Quantiles are useful measures because
they are less susceptible to long-tailed distributions and outliers.
</p>
<p>Plots and table were saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.locmetric">gl.filter.locmetric</a></code>, <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># adding dummy data
test &lt;- testset.gl
test$other$loc.metrics$test &lt;- 1:nLoc(test)
# SNP data
out &lt;- gl.report.locmetric(test,metric='test')

# adding dummy data
test.gs &lt;- testset.gs
test.gs$other$loc.metrics$test &lt;- 1:nLoc(test.gs)
# Tag P/A data
out &lt;- gl.report.locmetric(test.gs,metric='test')

</code></pre>

<hr>
<h2 id='gl.report.maf'>Reports minor allele frequency (MAF) for each locus in a SNP dataset</h2><span id='topic+gl.report.maf'></span>

<h3>Description</h3>

<p>This script provides summary histograms of MAF for each
population in the dataset and an overall histogram to assist the decision of
choosing thresholds for the filter function <code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.maf(
  x,
  maf.limit = 0.5,
  ind.limit = 5,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors_pop = discrete_palette,
  plot_colors_all = two_colors,
  bins = 25,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_maf.limit">maf.limit</code></td>
<td>
<p>Show histograms MAF range &lt;= maf.limit [default 0.5].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_ind.limit">ind.limit</code></td>
<td>
<p>Show histograms only for populations of size greater than
ind.limit [default 5].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>A color palette for population plots
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_plot_colors_all">plot_colors_all</code></td>
<td>
<p>List of two color names for the borders and fill of
the overall plot [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_bins">bins</code></td>
<td>
<p>Number of bins to display in histograms [default 25].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code> will filter out the
loci with MAF below a specified threshold.
</p>
<p><strong> Function's output </strong>
</p>
<p>The minimum, maximum, mean and a tabulation of MAF quantiles against
thresholds rate are provided. Output also includes a boxplot and a
histogram.
</p>
<p>This function reports the
MAF for each of several quantiles. Quantiles are
partitions of a finite set of values into q subsets of (nearly) equal sizes.
In this function q = 20. Quantiles are useful measures because they are less
susceptible to long-tailed distributions and outliers.
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.maf">gl.filter.maf</a></code>, <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl &lt;- gl.report.maf(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.report.monomorphs'>Reports monomorphic loci</h2><span id='topic+gl.report.monomorphs'></span>

<h3>Description</h3>

<p>This script reports the number of monomorphic loci and those with all NAs in
a genlight {adegenet} object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.monomorphs(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.monomorphs_+3A_x">x</code></td>
<td>
<p>Name of the input genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.report.monomorphs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A DArT dataset will not have monomorphic loci, but they can arise, along with
loci that are scored all NA, when populations or individuals are deleted.
Retaining monomorphic loci unnecessarily increases the size of the dataset
and will affect some calculations.
</p>
<p>Note that for SNP data, NAs likely represent null alleles; in tag
presence/absence data, NAs represent missing values (presence/absence could
not be reliably scored)
</p>


<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.monomorphs">gl.filter.monomorphs</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl.report.monomorphs(testset.gl)
# SilicoDArT data
  gl.report.monomorphs(testset.gs)
</code></pre>

<hr>
<h2 id='gl.report.overshoot'>Reports loci for which the SNP has been trimmed from the sequence tag
along with the adaptor</h2><span id='topic+gl.report.overshoot'></span>

<h3>Description</h3>

<p>This function checks the position of the SNP within the trimmed sequence tag
and identifies those for which the SNP position is outside the trimmed
sequence tag. This can happen, rarely, when the sequence containing the SNP
resembles the adaptor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.overshoot(x, save2tmp = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.overshoot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object [required].</p>
</td></tr>
<tr><td><code id="gl.report.overshoot_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.overshoot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The SNP genotype can still be used in most analyses, but functions like
gl2fasta() will present challenges if the SNP has been trimmed from the 
sequence tag.
</p>
<p>Resultant ggplot(s) and the tabulation(s) are saved to the session's
temporary directory.
</p>


<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.overshoot">gl.filter.overshoot</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.report.overshoot(testset.gl)
</code></pre>

<hr>
<h2 id='gl.report.pa'>Reports private alleles (and fixed alleles) per pair of populations</h2><span id='topic+gl.report.pa'></span>

<h3>Description</h3>

<p>This function reports private alleles in one population compared with a
second population, for all populations taken pairwise. It also reports a
count of fixed allelic differences and the mean absolute allele frequency
differences (AFD) between pairs of populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.pa(
  x,
  x2 = NULL,
  method = "pairwise",
  loc_names = FALSE,
  plot.out = TRUE,
  font_plot = 14,
  map.interactive = FALSE,
  palette_discrete = discrete_palette,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.pa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_x2">x2</code></td>
<td>
<p>If two separate genlight objects are to be compared this can be
provided here, but they must have the same number of SNPs [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_method">method</code></td>
<td>
<p>Method to calculate private alleles: 'pairwise' comparison or
compare each population against the rest 'one2rest' [default 'pairwise'].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_loc_names">loc_names</code></td>
<td>
<p>Whether names of loci with private alleles and fixed 
differences should reported. If TRUE, loci names are reported using a list
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if Sankey plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_font_plot">font_plot</code></td>
<td>
<p>Numeric font size in pixels for the node text labels
[default 14].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_map.interactive">map.interactive</code></td>
<td>
<p>Specify whether an interactive map showing private
alleles between populations is to be produced [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_palette_discrete">palette_discrete</code></td>
<td>
<p>A discrete palette for the color of populations or a
list with as many colors as there are populations in the dataset
[default discrete_palette].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.pa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent, fatal errors only; 1, flag function
begin and end; 2, progress log; 3, progress and results summary; 5, full
report [default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the number of paired alleles between two populations is not a
symmetric dissimilarity measure.
</p>
<p>If no x2 is provided, the function uses the pop(gl) hierarchy to determine
pairs of populations, otherwise it runs a single comparison between x and
x2.
</p>
<p><strong>Hint:</strong> in case you want to run comparisons between individuals
(assuming individual names are unique), you can simply redefine your
population names with your individual names, as below:
</p>
<p><code>pop(gl) &lt;- indNames(gl)</code>
</p>
<p><strong> Definition of fixed and private alleles </strong>
</p>
<p>The table below shows the possible cases of allele frequencies between
two populations (0 = homozygote for Allele 1, x = both Alleles are present,
1 = homozygote for Allele 2).
</p>

<ul>
<li><p> p: cases where there is a private allele in pop1 compared to pop2 (but
not vice versa)
</p>
</li>
<li><p> f: cases where there is a fixed allele in pop1 (and pop2, as those cases
are symmetric)
</p>
</li></ul>


<table>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> </td><td style="text-align: center;"> </td><td style="text-align: center;"> <em>pop1</em> </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> </td><td style="text-align: center;"> <strong>0</strong> </td><td style="text-align: center;">   <strong>x</strong>  </td><td style="text-align: center;">  <strong>1</strong></td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;">     <strong>0</strong></td><td style="text-align: center;"> -  </td><td style="text-align: center;">  p </td><td style="text-align: center;">  p,f</td>
</tr>
<tr>
 <td style="text-align: center;">
<em>pop2</em> </td><td style="text-align: center;"> <strong>x</strong></td><td style="text-align: center;"> -  </td><td style="text-align: center;">- </td><td style="text-align: center;"> -</td>
</tr>
<tr>
 <td style="text-align: center;">
</td><td style="text-align: center;"> <strong>1</strong> </td><td style="text-align: center;"> p,f</td><td style="text-align: center;"> p </td><td style="text-align: center;">   -</td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p>The absolute allele frequency difference (AFD) in this function is a simple
differentiation metric displaying intuitive properties which provides a
valuable alternative to FST. For details about its properties and how it is
calculated see Berner (2019).
</p>
<p>The function also reports an estimation of the lower bound of the number of
undetected private alleles using the Good-Turing frequency formula,
originally developed for cryptography, which estimates in an ecological 
context the true frequencies of rare species in a single assemblage based on
an incomplete sample of individuals. The approach is described in Chao et 
al. (2017). For this function, the equation 2c is used. This estimate is 
reported in the output table as Chao1 and Chao2. 
</p>
<p>In this function a Sankey Diagram is used to visualize patterns of private
alleles between populations. This diagram allows to display flows (private
alleles) between nodes (populations). Their links are represented with arcs
that have a width proportional to the importance of the flow (number of
private alleles).
</p>
<p>if save2temp=TRUE, resultant plot(s) and the tabulation(s) are saved to the
session's temporary directory.
</p>


<h3>Value</h3>

<p>A data.frame. Each row shows, for each pair of populations the number
of individuals in each population, the number of loci with fixed differences
(same for both populations) in pop1 (compared to pop2) and vice versa. Same
for private alleles and finally the absolute mean allele frequency
difference between loci (AFD). If loc_names = TRUE, loci names with private
alleles and fixed differences are reported in a list in addition to the 
dataframe.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>References</h3>


<ul>
<li><p> Berner, D. (2019). Allele frequency difference AFD – an intuitive
alternative to FST for quantifying genetic population differentiation. Genes,
10(4), 308.
</p>
</li>
<li><p> Chao, Anne, et al. &quot;Deciphering the enigma of undetected species,
phylogenetic, and functional diversity based on Good-Turing theory.&quot; 
Ecology 98.11 (2017): 2914-2929.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.list.reports">gl.list.reports</a></code>,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.pa(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.report.parent.offspring'>Identifies putative parent offspring within a population</h2><span id='topic+gl.report.parent.offspring'></span>

<h3>Description</h3>

<p>This script examines the frequency of pedigree inconsistent loci, that is,
those loci that are homozygotes in the parent for the reference allele, and
homozygous in the offspring for the alternate allele. This condition is not
consistent with any pedigree, regardless of the (unknown) genotype of the
other parent. The pedigree inconsistent loci are counted as an indication of
whether or not it is reasonable to propose the two individuals are in a
parent-offspring relationship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.parent.offspring(
  x,
  min.rdepth = 12,
  min.reproducibility = 1,
  range = 1.5,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.parent.offspring_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_min.rdepth">min.rdepth</code></td>
<td>
<p>Minimum read depth to include in analysis [default 12].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_min.reproducibility">min.reproducibility</code></td>
<td>
<p>Minimum reproducibility to include in analysis
[default 1].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_range">range</code></td>
<td>
<p>Specifies the range to extend beyond the interquartile range for
delimiting outliers [default 1.5 interquartile ranges].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_plot.out">plot.out</code></td>
<td>
<p>Creates a plot that shows the sex linked markers
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.parent.offspring_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If two individuals are in a parent offspring relationship, the true number of
pedigree inconsistent loci should be zero, but SNP calling is not infallible.
Some loci will be miss-called. The problem thus becomes one of determining
if the two focal individuals have a count of pedigree inconsistent loci less
than would be expected of typical unrelated individuals. There are some quite
sophisticated software packages available to formally apply likelihoods to
the decision, but we use a simple outlier comparison.
</p>
<p>To reduce the frequency of miss-calls, and so emphasize the difference
between true parent-offspring pairs and unrelated pairs, the data can be
filtered on read depth.
</p>
<p>Typically minimum read depth is set to 5x, but you can examine the
distribution of read depths with the function <code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a></code>
and push this up with an acceptable loss of loci. 12x might be a good minimum
for this particular analysis. It is sensible also to push the minimum
reproducibility up to 1, if that does not result in an unacceptable loss of
loci. Reproducibility is stored in the slot <code>@other$loc.metrics$RepAvg</code>
and is defined as the proportion of technical replicate assay pairs for which
the marker score is consistent. You can examine the distribution of
reproducibility with the function <code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>.
</p>
<p>Note that the null expectation is not well defined, and the power reduced, if
the population from which the putative parent-offspring pairs are drawn
contains many sibs. Note also that if an individual has been genotyped twice
in the dataset, the replicate pair will be assessed by this script as being
in a parent-offspring relationship.
</p>
<p>The function <code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a></code> will filter out those
individuals in a parent offspring relationship.
</p>
<p>Note that if your dataset does not contain RepAvg or rdepth among the locus
metrics, the filters for reproducibility and read depth are no used.
</p>
<p><strong> Function's output </strong>
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A set of individuals in parent-offspring relationship. NULL if no
parent-offspring relationships were found.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.list.reports">gl.list.reports</a></code>, <code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a></code> ,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>,<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a></code>,
<code><a href="#topic+gl.filter.parent.offspring">gl.filter.parent.offspring</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.parent.offspring(testset.gl[1:10,1:100])
</code></pre>

<hr>
<h2 id='gl.report.rdepth'>Reports summary of Read Depth for each locus</h2><span id='topic+gl.report.rdepth'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT report AvgCountRef and AvgCountSnp as counts
of sequence tags for the reference and alternate alleles respectively.
These can be used to back calculate Read Depth. Fragment presence/absence
datasets as provided by DArT (SilicoDArT) provide Average Read Depth and
Standard Deviation of Read Depth as standard columns in their report. This
function reports the read depth by locus for each of several quantiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.rdepth(
  x,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.rdepth_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.rdepth_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function displays a table of minimum, maximum, mean and quantiles for
read depth against possible thresholds that might subsequently be specified
in <code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>. If plot.out=TRUE, display also includes a
boxplot and a histogram to guide in the selection of a threshold for
filtering on read depth.
</p>
<p>If save2tmp=TRUE, ggplots and relevant tabulations are saved to the
session's temp directory (tempdir).
</p>
<p>For examples of themes, see  </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.rdepth">gl.filter.rdepth</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
df &lt;- gl.report.rdepth(testset.gl)

df &lt;- gl.report.rdepth(testset.gs)
</code></pre>

<hr>
<h2 id='gl.report.reproducibility'>Reports summary of RepAvg (repeatability averaged over both alleles 
for
each locus) or reproducibility (repeatability of the scores for fragment
presence/absence)</h2><span id='topic+gl.report.reproducibility'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have an index, RepAvg, generated by
reproducing the data independently for 30
of alleles that give a repeatable result, averaged over both alleles for each
locus.
</p>
<p>In the case of fragment presence/absence data (SilicoDArT), repeatability is
the percentage of scores that are repeated in the technical replicate
dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.reproducibility(
  x,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.reproducibility_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, displays a plot to guide the decision on a filter
threshold [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.reproducibility_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function displays a table of minimum, maximum, mean and quantiles for
repeatbility against possible thresholds that might subsequently be
specified in <code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a></code>.
</p>
<p>If plot.out=TRUE, display also includes a boxplot and a histogram to guide
in the selection of a threshold for filtering on repeatability.
</p>
<p>If save2tmp=TRUE, ggplots and relevant tabulations are saved to the
session's temp directory (tempdir)
</p>
<p>For examples of themes, see:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and 
</p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>An unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.reproducibility">gl.filter.reproducibility</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  out &lt;- gl.report.reproducibility(testset.gl)
  
# Tag P/A data
  out &lt;- gl.report.reproducibility(testset.gs)

</code></pre>

<hr>
<h2 id='gl.report.secondaries'>Reports loci containing secondary SNPs in sequence tags and calculates
number of invariant sites</h2><span id='topic+gl.report.secondaries'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT include fragments with more than
one SNP (that is, with secondaries). They are recorded separately with the
same CloneID (=AlleleID). These multiple SNP loci within a fragment are
likely to be linked, and so you may wish to remove secondaries.
</p>
<p>This function reports statistics associated with secondaries, and the
consequences of filtering them out, and provides three plots. The first is
a boxplot, the second is a barplot of the frequency of secondaries per
sequence tag, and the third is the Poisson expectation for those
frequencies including an estimate of the zero class (no. of sequence tags
with no SNP scored).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.secondaries(
  x,
  nsim = 1000,
  taglength = 69,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.secondaries_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_nsim">nsim</code></td>
<td>
<p>The number of simulations to estimate the mean of the Poisson
distribution [default 1000].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_taglength">taglength</code></td>
<td>
<p>Typical length of the sequence tags [default 69].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options [default
theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.secondaries_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code> will filter out the
loci with secondaries retaining only one sequence tag.
</p>
<p>Heterozygosity as estimated by the function
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code> is in a sense relative, because it
is calculated against a background of only those loci that are polymorphic
somewhere in the dataset. To allow intercompatibility across studies and
species, any measure of heterozygosity needs to accommodate loci that are
invariant (autosomal heterozygosity. See Schmidt et al 2021). However, the
number of invariant loci are unknown given the SNPs are detected as single
point mutational variants and invariant sequences are discarded, and
because of the particular additional filtering pre-analysis. Modelling the
counts of SNPs per sequence tag as a Poisson distribution in this script
allows estimate of the zero class, that is, the number of invariant loci.
This is reported, and the veracity of the estimate can be assessed by the
correspondence of the observed frequencies against those under Poisson
expectation in the associated graphs. The number of invariant loci can then
be optionally provided to the function
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code> via the parameter n.invariants.
</p>
<p>In case the calculations for the Poisson expectation of the number of
invariant sequence tags fail to converge, try to rerun the analysis with a
larger <code>nsim</code> values.
</p>
<p>This function now also calculates the number of invariant sites (i.e.
nucleotides) of the sequence tags (if <code>TrimmedSequence</code> is present in
<code>x$other$loc.metrics</code>) or estimate these by assuming that the average
length of the sequence tags is 69 nucleotides. Based on the Poisson
expectation of the number of invariant sequence tags, it also estimates the
number of invariant sites for these to eventually provide an estimate of
the total number of invariant sites.
</p>
<p><strong>Note</strong>, previous version of
<code>dartR</code> would only return an estimate of the number of invariant
sequence tags (not sites).
</p>
<p>Plots are saved to the session temporary directory (tempdir).
</p>
<p>Examples of other themes that can be used can be consulted in:
</p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A data.frame with the list of parameter values
</p>

<ul>
<li><p> n.total.tags Number of sequence tags in total
</p>
</li>
<li><p> n.SNPs.secondaries Number of secondary SNP loci that would be removed
on filtering
</p>
</li>
<li><p> n.invariant.tags Estimated number of invariant sequence tags
</p>
</li>
<li><p> n.tags.secondaries Number of sequence tags with secondaries
</p>
</li>
<li><p> n.inv.gen Number of invariant sites in sequenced tags
</p>
</li>
<li><p> mean.len.tag Mean length of sequence tags
</p>
</li>
<li><p> n.invariant Total Number of invariant sites (including invariant
sequence tags)
</p>
</li>
<li><p> k Lambda: mean of the Poisson distribution of number of SNPs in the
sequence tags
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Schmidt, T.L., Jasper, M.-E., Weeks, A.R., Hoffmann, A.A., 2021.
Unbiased population heterozygosity estimates from genome-wide sequence
data. Methods in Ecology and Evolution n/a.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code>,<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code>,
<code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
n.inv &lt;- gl.report.secondaries(test)
gl.report.heterozygosity(test, n.invariant = n.inv[7, 2])
</code></pre>

<hr>
<h2 id='gl.report.sexlinked'>Identifies loci that are sex linked</h2><span id='topic+gl.report.sexlinked'></span>

<h3>Description</h3>

<p>Alleles unique to the Y or W chromosome and monomorphic on the X chromosomes
will appear in the SNP dataset as genotypes that are heterozygotic in all
individuals of the heterogametic sex and homozygous in all individuals of the
homogametic sex. This function identifies loci with alleles that behave in
this way, as putative sex specific SNP markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.sexlinked(
  x,
  sex = NULL,
  t.het = 0.1,
  t.hom = 0.1,
  t.pres = 0.1,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = three_colors,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.sexlinked_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_sex">sex</code></td>
<td>
<p>Factor that defines the sex of individuals. See explanation in
details [default NULL].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_t.het">t.het</code></td>
<td>
<p>Tolerance in the heterogametic sex, that is t.het=0.05 means
that 5% of the heterogametic sex can be homozygous and still be regarded as
consistent with a sex specific marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_t.hom">t.hom</code></td>
<td>
<p>Tolerance in the homogametic sex, that is t.hom=0.05 means that
5% of the homogametic sex can be heterozygous and still be regarded as
consistent with a sex specific marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_t.pres">t.pres</code></td>
<td>
<p>Tolerance in presence, that is t.pres=0.05 means that a
silicodart marker can be present in either of the sexes and still be regarded
as a sex-linked marker [default 0.1].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot.out">plot.out</code></td>
<td>
<p>Creates a plot that shows the heterozygosity of males and
females at each loci and shaded area in which loci can be regarded as 
consistent with a sex specific marker [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of three color names for the not sex-linked loci, for
the sex-linked loci and for the area in which sex-linked loci appear
[default three_colors].</p>
</td></tr>
<tr><td><code id="gl.report.sexlinked_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sex of the individuals for which sex is known with certainty can be provided
via a factor (equal to the length of the number of individuals) or to be held
in the variable <code>x@other$ind.metrics$sex</code>.
Coding is: M for male, F for female, U or NA for unknown/missing.
The script abbreviates the entries here to the first character. So, coding of
'Female' and 'Male' works as well. Character are also converted to upper
cases.
</p>
<p>'<strong> Function's output </strong>
</p>
<p>This function creates a plot that shows the heterozygosity of males and
females at each loci or SNP data or percentage of present/absent in the case
of SilicoDArT data.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Two lists of sex-linked loci, one for XX/XY and one for ZZ/ZW systems
and a plot.
</p>


<h3>Author(s)</h3>

<p>Arthur Georges, Bernd Gruber &amp; Floriaan Devloo-Delva
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.taglength">gl.report.taglength</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
out &lt;- gl.report.sexlinked(testset.gl)
out &lt;- gl.report.sexlinked(testset.gs)

test &lt;- gl.filter.callrate(platypus.gl)
test &lt;- gl.filter.monomorphs(test)
out &lt;- gl.report.sexlinked(test)
</code></pre>

<hr>
<h2 id='gl.report.taglength'>Reports summary of sequence tag length across loci</h2><span id='topic+gl.report.taglength'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT typically have sequence tag lengths ranging
from 20 to 69 base pairs. This function reports summary statistics of the tag
lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.report.taglength(
  x,
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.report.taglength_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP [required].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, displays a plot to guide the decision on a filter
threshold [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.report.taglength_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a></code> will filter out the
loci with a tag length below a specified threshold.
</p>
<p>Quantiles are partitions of a finite set of values into q subsets of (nearly)
equal sizes. In this function q = 20. Quantiles are useful measures because
they are less susceptible to long-tailed distributions and outliers.
</p>
<p><strong> Function's output </strong>
</p>
<p>The minimum, maximum, mean and a tabulation of tag length quantiles against
thresholds are provided. Output also includes a boxplot and a
histogram to guide in the selection of a threshold for filtering on tag
length.
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.taglength">gl.filter.taglength</a></code>, <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>,
<code><a href="#topic+gl.print.reports">gl.print.reports</a></code>
</p>
<p>Other report functions: 
<code><a href="#topic+gl.report.bases">gl.report.bases</a>()</code>,
<code><a href="#topic+gl.report.callrate">gl.report.callrate</a>()</code>,
<code><a href="#topic+gl.report.diversity">gl.report.diversity</a>()</code>,
<code><a href="#topic+gl.report.hamming">gl.report.hamming</a>()</code>,
<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a>()</code>,
<code><a href="#topic+gl.report.hwe">gl.report.hwe</a>()</code>,
<code><a href="#topic+gl.report.ld.map">gl.report.ld.map</a>()</code>,
<code><a href="#topic+gl.report.locmetric">gl.report.locmetric</a>()</code>,
<code><a href="#topic+gl.report.maf">gl.report.maf</a>()</code>,
<code><a href="#topic+gl.report.monomorphs">gl.report.monomorphs</a>()</code>,
<code><a href="#topic+gl.report.overshoot">gl.report.overshoot</a>()</code>,
<code><a href="#topic+gl.report.parent.offspring">gl.report.parent.offspring</a>()</code>,
<code><a href="#topic+gl.report.pa">gl.report.pa</a>()</code>,
<code><a href="#topic+gl.report.rdepth">gl.report.rdepth</a>()</code>,
<code><a href="#topic+gl.report.reproducibility">gl.report.reproducibility</a>()</code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a>()</code>,
<code><a href="#topic+gl.report.sexlinked">gl.report.sexlinked</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.report.taglength(testset.gl)
</code></pre>

<hr>
<h2 id='gl.run.structure'>Runs a STRUCTURE analysis using a genlight object</h2><span id='topic+gl.run.structure'></span>

<h3>Description</h3>

<p>This function takes a genlight object and runs a STRUCTURE analysis based on
functions from <code>strataG</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.run.structure(
  x,
  ...,
  exec = ".",
  plot.out = TRUE,
  plot_theme = theme_dartR(),
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.run.structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_...">...</code></td>
<td>
<p>Parameters to specify the STRUCTURE run (check <code>structureRun</code>
within strataG.
for more details). Parameters are passed to the <code>structureRun</code> function.
For example you need to set the k.range and the type of model you would like
to run (noadmix, locprior) etc. If those parameter names do not tell you
anything, please make sure you familiarize with the STRUCTURE program
(Pritchard 2000).</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_exec">exec</code></td>
<td>
<p>Full path and name+extension where the structure executable is
located. E.g. <code>'c:/structure/structure.exe'</code> under Windows. For Mac and
Linux it might be something like <code>'./structure/structure'</code> if the
executable is in a subfolder 'structure' in your home directory
[default working directory &quot;.&quot;].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot.out">plot.out</code></td>
<td>
<p>Create an Evanno plot once finished. Be aware k.range needs
to be at least three different k steps [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.run.structure_+3A_verbose">verbose</code></td>
<td>
<p>Set verbosity for this function (though structure output
cannot be switched off currently) [default NULL]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is basically a convenient wrapper around the beautiful
strataG function <code>structureRun</code> (Archer et al. 2016). For a detailed
description please refer to this package (see references below).
To make use of this function you need to download STRUCTURE for you system
(<b>non GUI version</b>) from here
<a href="https://web.stanford.edu/group/pritchardlab/structure_software/release_versions/v2.3.4/html/structure.html">STRUCTURE</a>.
</p>
<p><b>Format note</b>
</p>
<p>For this function to work, make sure that individual and population names 
have no spaces. To substitute spaces by underscores you could use the R 
function <code>gsub</code> as below.
</p>
<p><code>popNames(gl) &lt;- gsub(" ","_",popNames(gl))</code>
</p>
<p><code>indNames(gl) &lt;- gsub(" ","_",indNames(gl)) </code>
</p>
<p>It's also worth noting that Structure truncates individual names at 11 
characters. The function will fail if the names of individuals are not unique
after truncation. To avoid this possible problem, a number sequence, as 
shown in the code below, might be used instead of individual names.
<code>
indNames(gl) &lt;- as.character(1:length(indNames(gl)))
</code>
</p>


<h3>Value</h3>

<p>An sr object (structure.result list output). Each list entry is a
single structurerun output (there are k.range * num.k.rep number of runs).
For example the summary output of the first run can be accessed via
<code>sr[[1]]$summary</code> or the q-matrix of the third run via
<code>sr[[3]]$q.mat</code>. To conveniently summarise the outputs across runs
(clumpp) you need to run gl.plot.structure on the returned sr object. For
Evanno plots run gl.evanno on your sr object.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Pritchard, J.K., Stephens, M., Donnelly, P. (2000) Inference of
population structure using multilocus genotype data. Genetics 155, 945-959.
</p>
</li>
<li><p> Archer, F. I., Adams, P. E. and Schneiders, B. B. (2016) strataG: An R
package for manipulating, summarizing and analysing population genetic data.
Mol Ecol Resour. doi:10.1111/1755-0998.12559
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#bc &lt;- bandicoot.gl[,1:100]
#sr &lt;- gl.run.structure(bc, k.range = 2:5, num.k.rep = 3, 
# exec = './structure.exe')
#ev &lt;- gl.evanno(sr)
#ev
#qmat &lt;- gl.plot.structure(sr, K=3)
#head(qmat)
#gl.map.structure(qmat, bc, scalex=1, scaley=0.5)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.sample'>Samples individuals from populations</h2><span id='topic+gl.sample'></span>

<h3>Description</h3>

<p>This is a convenience function to prepare a bootstrap approach in dartR. For a bootstrap approach it is often desirable to sample a defined number of individuals for each of the populations in a genlight object and then calculate a certain quantity for that subset (redo a 1000 times)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sample(
  x,
  nsample = min(table(pop(x))),
  replace = TRUE,
  onepop = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sample_+3A_x">x</code></td>
<td>
<p>genlight object containing SNP/silicodart genotypes</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_nsample">nsample</code></td>
<td>
<p>the number of individuals that should be sampled</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_replace">replace</code></td>
<td>
<p>a switch to sample by replacement (default).</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_onepop">onepop</code></td>
<td>
<p>switch to ignore population settings of the genlight object and sample from all individuals disregarding the population definition. [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.sample_+3A_verbose">verbose</code></td>
<td>
<p>set verbosity</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is convenience function to facilitate a bootstrap approach
</p>


<h3>Value</h3>

<p>returns a genlight object with nsample samples from each populations.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other base dartR: 
<code><a href="#topic+gl.sort">gl.sort</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#bootstrap for 2 possums populations to check effect of sample size on fixed alleles
gl.set.verbosity(0)
pp &lt;- possums.gl[1:60,]
nrep &lt;- 1:10
nss &lt;- seq(1,10,2)
res &lt;- expand.grid(nrep=nrep, nss=nss)
for (i in 1:nrow(res)) {
dummy &lt;- gl.sample(pp, nsample=res$nss[i], replace=TRUE)
pas &lt;- gl.report.pa(dummy, plot.out = F)
res$fixed[i] &lt;- pas$fixed[1]
}
boxplot(fixed ~ nss, data=res)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.save'>Saves an object in compressed binary format for later rapid retrieval</h2><span id='topic+gl.save'></span>

<h3>Description</h3>

<p>This is a wrapper for saveRDS().
</p>
<p>The script saves the object in binary form to the current workspace and
returns the input gl object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.save(x, file, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.save_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.save_+3A_file">file</code></td>
<td>
<p>Name of the file to receive the binary version of the object
[required].</p>
</td></tr>
<tr><td><code id="gl.save_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.load">gl.load</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.save(testset.gl,file.path(tempdir(),'testset.rds'))
</code></pre>

<hr>
<h2 id='gl.select.colors'>Selects colors from one of several palettes and output as a vector</h2><span id='topic+gl.select.colors'></span>

<h3>Description</h3>

<p>This script draws upon a number of specified color libraries to extract a
vector of colors for plotting, where the script that follows has a color
parameter expecting a vector of colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.select.colors(
  x = NULL,
  library = NULL,
  palette = NULL,
  ncolors = NULL,
  select = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.select.colors_+3A_x">x</code></td>
<td>
<p>Optionally, provide a gl object from which to determine the number
of populations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_library">library</code></td>
<td>
<p>Name of the color library to be used [default scales::hue_pl].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_palette">palette</code></td>
<td>
<p>Name of the color palette to be pulled from the specified
library [default is library specific] .</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_ncolors">ncolors</code></td>
<td>
<p>number of colors to be displayed and returned [default 9].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_select">select</code></td>
<td>
<p>select the colors to retain in the output vector
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.colors_+3A_verbose">verbose</code></td>
<td>
<p>&ndash; verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The available color libraries and their palettes include:
</p>

<ul>
<li><p> library 'brewer' and the palettes available can be listed by
RColorBrewer::display.brewer.all() and RColorBrewer::brewer.pal.info.
</p>
</li>
<li><p> library 'gr.palette' and the palettes available can be listed by
grDevices::palette.pals()
</p>
</li>
<li><p> library 'r.hcl' and the palettes available can be listed by
grDevices::hcl.pals()
</p>
</li>
<li><p> library 'baseR' and the palettes available are: 'rainbow','heat',
'topo.colors','terrain.colors','cm.colors'.
</p>
</li></ul>

<p>If the nominated palette is not specified, all the palettes will be listed a
nd a default palette will then be chosen.
</p>
<p>The color palette will be displayed in the graphics window for the requested
number of colors (or 9 if not specified),and the vector of colors returned
for later use.
</p>
<p>The select parameter can be used to select colors from the specified ncolors.
For example, select=c(1,1,3) will select color 1, 1 again and 3 to retain in
the final vector. This can be useful for fine-tuning color selection, and
matching colors and shapes.
</p>


<h3>Value</h3>

<p>A vector with the required number of colors
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.select.shapes">gl.select.shapes</a></code>
</p>
<p>Other Exploration/visualisation functions: 
<code><a href="#topic+gl.pcoa.plot">gl.pcoa.plot</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# EXAMPLES -- SIMPLE
colors &lt;- gl.select.colors()
colors &lt;- gl.select.colors(library='brewer',palette='Spectral',ncolors=6)
colors &lt;- gl.select.colors(library='baseR',palette='terrain.colors',ncolors=6)
colors &lt;- gl.select.colors(library='baseR',palette='rainbow',ncolors=12)
colors &lt;- gl.select.colors(library='gr.hcl',palette='RdBu',ncolors=12)
colors &lt;- gl.select.colors(library='gr.palette',palette='Pastel 1',ncolors=6)
# EXAMPLES -- SELECTING colorS
colors &lt;- gl.select.colors(library='baseR',palette='rainbow',ncolors=12,select=c(1,1,1,5,8))
# EXAMPLES -- CROSS-CHECKING WITH A GENLIGHT OBJECT
colors &lt;- gl.select.colors(x=gl,library='baseR',palette='rainbow',ncolors=12,select=c(1,1,1,5,8))

</code></pre>

<hr>
<h2 id='gl.select.shapes'>Selects shapes from the base R shape palette and outputs as a vector</h2><span id='topic+gl.select.shapes'></span>

<h3>Description</h3>

<p>This script draws upon the standard R shape palette to extract a vector of
shapes for plotting, where the script that follows has a shape parameter
expecting a vector of shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.select.shapes(x = NULL, select = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.select.shapes_+3A_x">x</code></td>
<td>
<p>Optionally, provide a gl object from which to determine the number
of populations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.select.shapes_+3A_select">select</code></td>
<td>
<p>Select the shapes to retain in the output vector
[default NULL, all shapes shown and returned].</p>
</td></tr>
<tr><td><code id="gl.select.shapes_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the shape palette will be displayed in full in the graphics window
from which shapes can be selected in a subsequent run, and the vector of
shapes returned for later use.
</p>
<p>The select parameter can be used to select shapes from the specified 26
shapes available (0-25). For example, select=c(1,1,3) will select shape 1, 1
again and 3 to retain in the final vector. This can be useful for fine-tuning
shape selection, and matching colors and shapes.
</p>


<h3>Value</h3>

<p>A vector with the required number of shapes
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.select.colors">gl.select.colors</a></code>
</p>
<p>Other Exploration/visualisation functions: 
<code><a href="#topic+gl.pcoa.plot">gl.pcoa.plot</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.smearplot">gl.smearplot</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SET UP DATASET
gl &lt;- testset.gl
levels(pop(gl))&lt;-c(rep('Coast',5),rep('Cooper',3),rep('Coast',5),
rep('MDB',8),rep('Coast',7),'Em.subglobosa','Em.victoriae')
# EXAMPLES
shapes &lt;- gl.select.shapes() # Select and display available shapes
# Select and display a restricted set of shapes
shapes &lt;- gl.select.shapes(select=c(1,1,1,5,8)) 
 # Select set of shapes and check with no. of pops.
shapes &lt;- gl.select.shapes(x=gl,select=c(1,1,1,5,8))
</code></pre>

<hr>
<h2 id='gl.set.verbosity'>Sets the default verbosity level</h2><span id='topic+gl.set.verbosity'></span>

<h3>Description</h3>

<p>dartR functions have a verbosity parameter that sets the level of reporting
during the execution of the function. The verbosity level, set by parameter
'verbose' can be one of verbose 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report. The
default value for verbosity is stored in the r environment. This script sets
the default value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.set.verbosity(value = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.set.verbosity_+3A_value">value</code></td>
<td>
<p>Set the default verbosity to be this value: 0, silent only fatal
errors; 1, begin and end; 2, progress log; 3, progress and results summary;
5, full report [default 2]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>verbosity value [set for all functions]
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl &lt;- gl.set.verbosity(value=2)
</code></pre>

<hr>
<h2 id='gl.sfs'>Creates a site frequency spectrum based on a dartR or genlight object</h2><span id='topic+gl.sfs'></span>

<h3>Description</h3>

<p>Creates a site frequency spectrum based on a dartR or genlight object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sfs(
  x,
  minbinsize = 0,
  folded = TRUE,
  singlepop = FALSE,
  plot.out = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sfs_+3A_x">x</code></td>
<td>
<p>dartR/genlight object</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_minbinsize">minbinsize</code></td>
<td>
<p>remove bins from the left of the sfs. For example to remove
singletons (alleles only occurring once among all individuals) set
minbinsize to 2. If set to zero, also monomorphic (d0) loci are returned.</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_folded">folded</code></td>
<td>
<p>if set to TRUE (default) a folded sfs (minor allele frequency
sfs) is returned. If set to FALSE then an unfolded (derived allele frequency
sfs) is returned. It is assumed that 0 is homozygote for the reference and
2 is homozygote for the derived allele. So you need to make sure your
coding is correct.</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_singlepop">singlepop</code></td>
<td>
<p>switch to force to create a one-dimensional sfs, even
though the genlight/dartR object contains more than one population</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sfs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a site frequency spectrum, either a one dimensional vector
(only a single population in the dartR/genlight object or singlepop=TRUE) or
an n-dimensional array (n is the number of populations in the genlight/dartR
object). If the dartR/genlight object consists of several populations the
multidimensional site frequency spectrum for each population is returned
[=a multidimensional site frequency spectrum]. Be aware the
multidimensional spectrum works only for a limited number of population
and individuals [if too high the table command used internally will
through an error as the number of populations and individuals (and
therefore dimensions) are too large]. To get a single sfs for a
genlight/dartR object with multiple populations, you need to set
singlepop to TRUE. The returned sfs can be used to analyse demographics,
e.g. using fastsimcoal2.
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber &amp; Carlo Pacioni (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Excoffier L., Dupanloup I., Huerta-Sánchez E., Sousa V. C. and
Foll M. (2013) Robust demographic inference from genomic and SNP data. PLoS
genetics 9(10)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.sfs(bandicoot.gl, singlepop=TRUE)
gl.sfs(possums.gl[c(1:5,31:33),], minbinsize=1)
</code></pre>

<hr>
<h2 id='gl.sim.create_dispersal'>Creates a dispersal file as input for the function gl.sim.WF.run</h2><span id='topic+gl.sim.create_dispersal'></span>

<h3>Description</h3>

<p>This function writes a csv file called &quot;dispersal_table.csv&quot; which contains
the dispersal variables for each pair of populations to be used as input for
the function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>.
</p>
<p>The values of the variables can be modified using the columns
&quot;transfer_each_gen&quot; and &quot;number_transfers&quot; of this file.
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed by typing in the R console:
browseVignettes(package=&quot;dartR”)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.create_dispersal(
  number_pops,
  dispersal_type = "all_connected",
  number_transfers = 1,
  transfer_each_gen = 1,
  outpath = tempdir(),
  outfile = "dispersal_table.csv",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.create_dispersal_+3A_number_pops">number_pops</code></td>
<td>
<p>Number of populations [required].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_dispersal_type">dispersal_type</code></td>
<td>
<p>One of: &quot;all_connected&quot;, &quot;circle&quot; or &quot;line&quot;
[default &quot;all_connected&quot;].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_number_transfers">number_transfers</code></td>
<td>
<p>Number of dispersing individuals. This value can be .
modified by hand after the file has been created [default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_transfer_each_gen">transfer_each_gen</code></td>
<td>
<p>Interval of number of generations in which dispersal
occur. This value can be modified by hand after the file has been created
[default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'dispersal_table.csv'].</p>
</td></tr>
<tr><td><code id="gl.sim.create_dispersal_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A csv file containing the dispersal variables for each pair of
populations to be used as input for the function <code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a>()</code>,
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.sim.create_dispersal(number_pops=10)
</code></pre>

<hr>
<h2 id='gl.sim.emigration'>Simulates emigration between populations</h2><span id='topic+gl.sim.emigration'></span>

<h3>Description</h3>

<p>A function that allows to exchange individuals of populations within a
genlight object (=simulate emigration between populations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.emigration(x, perc.mig = NULL, emi.m = NULL, emi.table = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.emigration_+3A_x">x</code></td>
<td>
<p>A genlight or list of genlight objects [required].</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_perc.mig">perc.mig</code></td>
<td>
<p>Percentage of individuals that migrate
(emigrates = nInd times perc.mig) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_emi.m">emi.m</code></td>
<td>
<p>Probabilistic emigration matrix (emigrate from=column to=row)
[default NULL]</p>
</td></tr>
<tr><td><code id="gl.sim.emigration_+3A_emi.table">emi.table</code></td>
<td>
<p>If presented emi.m matrix is ignored. Deterministic
emigration as specified in the matrix (a square matrix of dimension of the
number of populations). e.g. an entry in the 'emi.table[2,1]&lt;- 5' means that
five individuals emigrate from population 1 to population 2 (from=columns and
to=row) [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two ways to specify emigration. If an emi.table is provided (a
square matrix of dimension of the populations that specifies the emigration
from column x to row y), then emigration is deterministic in terms of numbers
of individuals as specified in the table. If perc.mig and emi.m are provided,
then emigration is probabilistic. The number of emigrants is determined by
the population size times the perc.mig and then the population where to
migrate to is taken from the relative probability in the columns of the emi.m
table.
</p>
<p>Be aware if the diagonal is non zero then migration can occur into the same
patch. So most often you want to set the diagonal of the emi.m matrix to
zero. Which individuals is moved is random, but the order is in the order of
populations. It is possible that an individual moves twice within an
emigration call(as there is no check, so an individual moved from population
1 to 2 can move again from population 2 to 3).
</p>


<h3>Value</h3>

<p>A list or a single [depends on the input] genlight object, where
emigration between population has happened
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- possums.gl
#one individual moves from every population to
#every other population
emi.tab &lt;- matrix(1, nrow=nPop(x), ncol=nPop(x))
diag(emi.tab)&lt;- 0
np &lt;- gl.sim.emigration(x, emi.table=emi.tab)
np
</code></pre>

<hr>
<h2 id='gl.sim.ind'>Simulates individuals based on the allele frequencies provided via a genlight
object.</h2><span id='topic+gl.sim.ind'></span>

<h3>Description</h3>

<p>This function simulates individuals based on the allele frequencies of a
genlight object. The output is a genlight object with the same number of loci
as the input genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.ind(x, n = 50, popname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.ind_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.sim.ind_+3A_n">n</code></td>
<td>
<p>Number of individuals that should be simulated [default 50].</p>
</td></tr>
<tr><td><code id="gl.sim.ind_+3A_popname">popname</code></td>
<td>
<p>A population name for the simulated individuals [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be used to simulate populations for sampling designs or for
power analysis. Check the example below where the effect of drift is
explored, by simply simulating several generation a genlight object and 
putting in the allele frequencies of the previous generation. The beauty of
the function is, that it is lightning fast. Be aware this is a simulation 
and to avoid lengthy error checking the function crashes if there are loci 
that have just NAs. If such a case can occur during your simulation, those
loci need to be removed, before the function is called.
</p>


<h3>Value</h3>

<p>A genlight object with n individuals.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bernd.gruber@canberra.edu.au)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>glsim &lt;- gl.sim.ind(testset.gl, n=10, popname='sims')
glsim
###Simulate drift over 10 generation
# assuming a bottleneck of only 10 individuals
# [ignoring effect of mating and mutation]
# Simulate 20 individuals with no structure and 50 SNP loci
founder &lt;- glSim(n.ind = 20, n.snp.nonstruc = 50, ploidy=2)
#number of fixed loci in the first generation

res &lt;- sum(colMeans(as.matrix(founder), na.rm=TRUE) %%2 ==0)
simgl &lt;- founder
#49 generations of only 10 individuals
for (i in 2:50)
{
   simgl &lt;- gl.sim.ind(simgl, n=10, popname='sims')
   res[i]&lt;- sum(colMeans(as.matrix(simgl), na.rm=TRUE) %%2 ==0)
}
plot(1:50, res, type='b', xlab='generation', ylab='# fixed loci')
</code></pre>

<hr>
<h2 id='gl.sim.mutate'>Simulates mutations within a genlight object</h2><span id='topic+gl.sim.mutate'></span>

<h3>Description</h3>

<p>This script is intended to be used within the simulation framework of dartR. 
It adds the ability to add a constant mutation rate across all loci. Only 
works currently for biallelic data sets (SNPs). Mutation rate is checking for 
all alleles position and mutations at loci with missing values are ignored 
and in principle 'double mutations' at the same loci can occur, but should be 
rare.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.mutate(x, mut.rate = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.mutate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.sim.mutate_+3A_mut.rate">mut.rate</code></td>
<td>
<p>Constant mutation rate over nInd*nLoc*2 possible locations
[default 1e-6]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a genlight object with the applied mutations
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b2 &lt;- gl.sim.mutate(bandicoot.gl,mut.rate=1e-4 )
#check the mutations that have occurred
table(as.matrix(bandicoot.gl), as.matrix(b2))
</code></pre>

<hr>
<h2 id='gl.sim.offspring'>Simulates a specified number of offspring based on alleles provided by
potential father(s) and mother(s)</h2><span id='topic+gl.sim.offspring'></span>

<h3>Description</h3>

<p>This takes a population (or a single individual) of fathers (provided as a
genlight object) and mother(s) and simulates offspring based on 'random'
mating. It can be used to simulate population dynamics and check the effect
of those dynamics and allele frequencies, number of alleles. Another
application is to simulate relatedness of siblings and compare it to actual
relatedness found in the population to determine kinship.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.offspring(fathers, mothers, noffpermother, sexratio = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.offspring_+3A_fathers">fathers</code></td>
<td>
<p>Genlight object of potential fathers [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_mothers">mothers</code></td>
<td>
<p>Genlight object of potential mothers simulated [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_noffpermother">noffpermother</code></td>
<td>
<p>Number of offspring per mother [required].</p>
</td></tr>
<tr><td><code id="gl.sim.offspring_+3A_sexratio">sexratio</code></td>
<td>
<p>The sex ratio of simulated offspring 
(females / females +males, 1 equals 100 percent females) [default 0.5.].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with n individuals.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Simulate 10 potential fathers
gl.fathers &lt;- glSim(10, 20, ploidy=2)
#Simulate 10 potential mothers
gl.mothers &lt;- glSim(10, 20, ploidy=2)
gl.sim.offspring(gl.fathers, gl.mothers, 2, sexratio=0.5)
</code></pre>

<hr>
<h2 id='gl.sim.WF.run'>Runs Wright-Fisher simulations</h2><span id='topic+gl.sim.WF.run'></span>

<h3>Description</h3>

<p>This function simulates populations made up of diploid organisms that 
reproduce in non-overlapping generations. Each individual has a pair of 
homologous chromosomes that contains interspersed selected and neutral loci. 
For the initial generation, the genotype for each individual’s chromosomes is
randomly drawn from distributions at linkage equilibrium and in 
Hardy-Weinberg equilibrium. 
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed at http://georges.biomatix.org/dartR 
</p>
<p>Take into account that the simulations will take a little bit longer the
first time you use the function gl.sim.WF.run() because C++ functions must
be compiled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.WF.run(
  file_var,
  ref_table,
  x = NULL,
  file_dispersal = NULL,
  number_iterations = 1,
  every_gen = 10,
  sample_percent = 50,
  store_phase1 = FALSE,
  interactive_vars = TRUE,
  seed = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.WF.run_+3A_file_var">file_var</code></td>
<td>
<p>Path of the variables file 'sim_variables.csv' (see details) 
[required if interactive_vars = FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_ref_table">ref_table</code></td>
<td>
<p>Reference table created by the function 
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a></code> [required].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data to extract
values for some simulation variables (see details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_file_dispersal">file_dispersal</code></td>
<td>
<p>Path of the file with the dispersal table created with
the function <code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a></code> [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_number_iterations">number_iterations</code></td>
<td>
<p>Number of iterations of the simulations [default 1].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_every_gen">every_gen</code></td>
<td>
<p>Generation interval at which simulations should be stored in
a genlight object [default 10].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_sample_percent">sample_percent</code></td>
<td>
<p>Percentage of individuals, from the total population, 
to sample and save in the genlight object every generation [default 50].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_store_phase1">store_phase1</code></td>
<td>
<p>Whether to store simulations of phase 1 in genlight
objects [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_interactive_vars">interactive_vars</code></td>
<td>
<p>Run a shiny app to input interactively the values of
simulations variables [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_seed">seed</code></td>
<td>
<p>Set the seed for the simulations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.run_+3A_...">...</code></td>
<td>
<p>Any variable and its value can be added separately within the 
function, will be changed over the input value supplied by the csv file. See 
tutorial.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Values for simulation variables can be submitted into the function 
interactively through a shiny app if interactive_vars = TRUE. Optionally, if 
interactive_vars = FALSE, values for variables can be submitted by using the
csv file 'sim_variables.csv' which can be found by typing in the R console:
system.file('extdata', 'sim_variables.csv', package ='dartR').
</p>
<p>The values of the variables can be modified using the third column (“value”) 
of this file. 
</p>
<p>The output of the simulations can be analysed seemingly with other dartR 
functions.
</p>
<p>If a genlight object is used as input for some of the simulation variables, 
this function access the information stored in the slots x$position and 
x$chromosome.
</p>
<p>To show further information of the variables in interactive mode, it might be
necessary to call first: 'library(shinyBS)' for the information to be 
displayed.
</p>
<p>The main characteristics of the simulations are:
</p>
 
<ul>
<li><p> Simulations can be parameterised with real-life genetic 
characteristics such as the number, location, allele frequency and the 
distribution of fitness effects (selection coefficients and dominance) of 
loci under selection. 
</p>
</li>
<li><p> Simulations can recreate specific life histories and demographics, such
as source populations, dispersal rate, number of generations, founder 
individuals, effective population size and census population size.
</p>
</li>
<li><p> Each allele in each individual is an agent (i.e., each allele is 
explicitly simulated).
</p>
</li>
<li><p> Each locus can be customisable regarding its allele frequencies, 
selection coefficients, and dominance.
</p>
</li>
<li><p> The number of loci, individuals, and populations to be simulated is 
only limited by computing resources.
</p>
</li>
<li><p> Recombination is accurately modeled, and it is possible to use real 
recombination maps as input.
</p>
</li>
<li><p> The ratio between effective population size and census population size 
can be easily controlled.
</p>
</li>
<li><p> The output of the simulations are genlight objects for each generation 
or a subset of generations.
</p>
</li>
<li><p> Genlight objects can be used as input for some simulation variables.
</p>
</li></ul>



<h3>Value</h3>

<p>Returns genlight objects with simulated data.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.table">gl.sim.WF.table</a>()</code>,
<code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ref_table &lt;- gl.sim.WF.table(file_var=system.file('extdata', 
'ref_variables.csv', package = 'dartR'),interactive_vars = FALSE)
res_sim &lt;- gl.sim.WF.run(file_var = system.file('extdata', 
'sim_variables.csv', package ='dartR'),ref_table=ref_table,
interactive_vars = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.sim.WF.table'>Creates the reference table for running gl.sim.WF.run</h2><span id='topic+gl.sim.WF.table'></span>

<h3>Description</h3>

<p>This function creates a reference table to be used as input for the function
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>. The created table has eight columns with the 
following information for each locus to be simulated:
</p>
 
<ul>
<li><p> q - initial frequency.
</p>
</li>
<li><p> h - dominance coefficient.
</p>
</li>
<li><p> s - selection coefficient.
</p>
</li>
<li><p> c - recombination rate.
</p>
</li>
<li><p> loc_bp - chromosome location in base pairs.
</p>
</li>
<li><p> loc_cM - chromosome location in centiMorgans.
</p>
</li>
<li><p> chr_name - chromosome name.
</p>
</li>
<li><p> type - SNP type.
</p>
</li></ul>
 
<p>The reference table can be further modified as required. 
</p>
<p>See documentation and tutorial for a complete description of the simulations.
These documents can be accessed at http://georges.biomatix.org/dartR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sim.WF.table(
  file_var,
  x = NULL,
  file_targets_sel = NULL,
  file_r_map = NULL,
  interactive_vars = TRUE,
  seed = NULL,
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sim.WF.table_+3A_file_var">file_var</code></td>
<td>
<p>Path of the variables file 'ref_variables.csv' (see details) 
[required if interactive_vars = FALSE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data to extract
values for some simulation variables (see details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_file_targets_sel">file_targets_sel</code></td>
<td>
<p>Path of the file with the targets for selection (see 
details) [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_file_r_map">file_r_map</code></td>
<td>
<p>Path of the file with the recombination map (see details)
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_interactive_vars">interactive_vars</code></td>
<td>
<p>Run a shiny app to input interactively the values of
simulation variables [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_seed">seed</code></td>
<td>
<p>Set the seed for the simulations [default NULL].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl.sim.WF.table_+3A_...">...</code></td>
<td>
<p>Any variable and its value can be added separately within the 
function, will be changed over the input value supplied by the csv file. See 
tutorial.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Values for the variables to create the reference table can be submitted into 
the function interactively through a Shiny app if interactive_vars = TRUE. 
Optionally, if interactive_vars = FALSE, values for variables can be 
submitted by using the csv file 'ref_variables.csv' which can be found by 
typing in the R console:
system.file('extdata', 'ref_variables.csv', package ='dartR').
</p>
<p>The values of the variables can be modified using the third column (“value”) 
of this file. 
</p>
<p>If a genlight object is used as input for some of the simulation variables, 
this function access the information stored in the slots x$position and 
x$chromosome.
</p>
<p>Examples of the format required for the recombination map file and the 
targets for selection file can be found by typing in the R console:
</p>
 
<ul>
<li><p> system.file('extdata', 'fly_recom_map.csv', package ='dartR')
</p>
</li>
<li><p> system.file('extdata', 'fly_targets_of_selection.csv', package ='dartR')
</p>
</li></ul>

<p>To show further information of the variables in interactive mode, it might be
necessary to call first: 'library(shinyBS)' for the information to be 
displayed.
</p>


<h3>Value</h3>

<p>Returns a list with the reference table used as input for the function
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code> and a table with the values variables used to 
create the reference table.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a></code>
</p>
<p>Other simulation functions: 
<code><a href="#topic+gl.sim.WF.run">gl.sim.WF.run</a>()</code>,
<code><a href="#topic+gl.sim.create_dispersal">gl.sim.create_dispersal</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref_table &lt;- gl.sim.WF.table(file_var=system.file('extdata', 
'ref_variables.csv', package = 'dartR'),interactive_vars = FALSE)
## Not run: 
#uncomment to run 
res_sim &lt;- gl.sim.WF.run(file_var = system.file('extdata', 
'sim_variables.csv', package ='dartR'),ref_table=ref_table,
interactive_vars = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='gl.smearplot'>Smear plot of SNP or presence/absence (SilicoDArT) data</h2><span id='topic+gl.smearplot'></span>

<h3>Description</h3>

<p>Each locus is color coded for scores of 0, 1, 2 and NA for SNP data and 0, 1
and NA for presence/absence (SilicoDArT) data. Individual labels can be added
and individuals can be grouped by population.
</p>
<p>Plot may become cluttered if ind_labels If there are too many individuals, 
it is best to use ind_labels_size = 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.smearplot(
  x,
  ind_labels = FALSE,
  group_pop = FALSE,
  ind_labels_size = 10,
  plot_colors = NULL,
  posi = "bottom",
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.smearplot_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_ind_labels">ind_labels</code></td>
<td>
<p>If TRUE, individuals are labelled with indNames(x) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_group_pop">group_pop</code></td>
<td>
<p>If ind_labels is TRUE, group by population [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_ind_labels_size">ind_labels_size</code></td>
<td>
<p>Size of the individual labels [default 10].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_plot_colors">plot_colors</code></td>
<td>
<p>Vector with four color names for homozygotes for the
reference allele, heterozygotes, homozygotes for the alternative allele and
for missing values (NA), e.g. four_colours [default NULL].
Can be set to &quot;hetonly&quot;, which defines colors to only show heterozygotes in the genlight object</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_posi">posi</code></td>
<td>
<p>Position of the legend: “left”, “top”, “right”, “bottom” or
'none' [default = 'bottom'].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves plot to the session temporary directory
(tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.smearplot_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns unaltered genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>See Also</h3>

<p>Other Exploration/visualisation functions: 
<code><a href="#topic+gl.pcoa.plot">gl.pcoa.plot</a>()</code>,
<code><a href="#topic+gl.select.colors">gl.select.colors</a>()</code>,
<code><a href="#topic+gl.select.shapes">gl.select.shapes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl.smearplot(testset.gl,ind_labels=FALSE)
gl.smearplot(testset.gs[1:10,],ind_labels=TRUE)
</code></pre>

<hr>
<h2 id='gl.sort'>re-sorts genlight objects</h2><span id='topic+gl.sort'></span>

<h3>Description</h3>

<p>Often it is desirable to have the genlight object sorted individuals by population names, indiviual name, for example to have a more informative gl.smearplot (showing banding patterns for populations). Also sorting by loci can be informative in some instances. This function provides the ability to sort individuals of a genlight object by providing the order of individuals or populations and also by loci metric providing the order of locis. See examples below for specifics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.sort(x, sort.by = "pop", order.by = NULL, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.sort_+3A_x">x</code></td>
<td>
<p>genlight object containing SNP/silicodart genotypes</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_sort.by">sort.by</code></td>
<td>
<p>either &quot;ind&quot;, &quot;pop&quot;. Default is pop</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_order.by">order.by</code></td>
<td>
<p>that is used to order individuals or loci. Depening on the order.by parameter, this needs to be a vector of length of nPop(genlight) for populations or  nInd(genlight) for individuals. If not specified alphabetical order of populations or individuals is used. For sort.by=&quot;ind&quot; order.by can be also a vector specifying the order for each individual (for example another ind.metrics)</p>
</td></tr>
<tr><td><code id="gl.sort_+3A_verbose">verbose</code></td>
<td>
<p>set verbosity</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is convenience function to facilitate sorting of individuals within the genlight object. For example if you want to visualise the &quot;band&quot; of population in a gl.smearplot then the order of individuals is important. Also
</p>


<h3>Value</h3>

<p>Returns a reordered genlight object. Sorts also the ind/loc.metrics and coordinates accordingly
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p>Other base dartR: 
<code><a href="#topic+gl.sample">gl.sample</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#sort by populations
bc &lt;- gl.sort(bandicoot.gl)
#sort from West to East
bc2 &lt;- gl.sort(bandicoot.gl, sort.by="pop" ,
order.by=c("WA", "SA", "VIC", "NSW", "QLD"))
#sort by missing values
miss &lt;- rowSums(is.na(as.matrix(bandicoot.gl)))
bc3 &lt;- gl.sort(bandicoot.gl, sort.by="ind", order.by=miss)
gl.smearplot(bc3)
</code></pre>

<hr>
<h2 id='gl.spatial.autoCorr'>Spatial autocorrelation following Smouse and Peakall 1999</h2><span id='topic+gl.spatial.autoCorr'></span>

<h3>Description</h3>

<p>Global spatial autocorrelation is a multivariate approach
combining all loci into a single analysis. The autocorrelation coefficient
&quot;r&quot; is calculated for each pair of individuals in each specified distance
class. For more information see Smouse and Peakall 1999, Peakall et al. 2003
and Smouse et al. 2008.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.spatial.autoCorr(
  x = NULL,
  Dgeo = NULL,
  Dgen = NULL,
  coordinates = "latlon",
  Dgen_method = "Euclidean",
  Dgeo_trans = "Dgeo",
  Dgen_trans = "Dgen",
  bins = 5,
  reps = 100,
  plot.pops.together = FALSE,
  permutation = TRUE,
  bootstrap = TRUE,
  plot_theme = NULL,
  plot_colors_pop = NULL,
  CI_color = "red",
  plot.out = TRUE,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.spatial.autoCorr_+3A_x">x</code></td>
<td>
<p>Genlight object [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgeo">Dgeo</code></td>
<td>
<p>Geographic distance matrix if no genlight object is provided.
This is typically an Euclidean distance but it can be any meaningful 
(geographical) distance metrics [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen">Dgen</code></td>
<td>
<p>Genetic distance matrix if no genlight object is provided
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_coordinates">coordinates</code></td>
<td>
<p>Can be either 'latlon', 'xy' or a two column data.frame
with column names 'lat','lon', 'x', 'y')  Coordinates are provided via
<code>gl@other$latlon</code> ['latlon'] or via <code>gl@other$xy</code> ['xy']. If latlon
data will be projected to meters using Mercator system [google maps] or if
xy then distance is directly calculated on the coordinates [default &quot;latlon&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen_method">Dgen_method</code></td>
<td>
<p>Method to calculate genetic distances. See details
[default &quot;Euclidean&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgeo_trans">Dgeo_trans</code></td>
<td>
<p>Transformation to be used on the geographic distances. See
Dgen_trans [default &quot;Dgeo&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_dgen_trans">Dgen_trans</code></td>
<td>
<p>You can provide a formula to transform the genetic
distance. The transformation can be applied as a formula using Dgen as the
variable to be transformed. For example: <code>Dgen_trans = 'Dgen/(1-Dgen)'.
  Any valid R expression can be used here 
  [default 'Dgen', which is the identity function.]</code></p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_bins">bins</code></td>
<td>
<p>The number of bins for the distance classes
(i.e. <code>length(bins) == 1)</code> or a vectors with the break points. See 
details [default 5].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_reps">reps</code></td>
<td>
<p>The number to be used for permutation and bootstrap analyses
[default 100].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.pops.together">plot.pops.together</code></td>
<td>
<p>Plot all the populations in one plot. Confidence 
intervals from permutations are not shown [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_permutation">permutation</code></td>
<td>
<p>Whether permutation calculations for the null hypothesis 
of no spatial structure should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_bootstrap">bootstrap</code></td>
<td>
<p>Whether bootstrap calculations to compute the 95% 
confidence intervals around r should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See details [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot_colors_pop">plot_colors_pop</code></td>
<td>
<p>A color palette for populations or a list with
as many colors as there are populations in the dataset [default NULL].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_ci_color">CI_color</code></td>
<td>
<p>Color for the shade of the 95% confidence intervals around 
the r estimates [default &quot;red&quot;].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_plot.out">plot.out</code></td>
<td>
<p>Specify if plot is to be produced [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.spatial.autoCorr_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default
NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function executes a modified version
of <code>spautocorr</code> from the package <code>PopGenReport</code>. Differently 
from <code>PopGenReport</code>, this function also computes the 95% confidence 
intervals around the r via bootstraps, the 95
null hypothesis of no spatial structure and the one-tail test via permutation, 
and the correction factor described by Peakall et al 2003.
</p>
<p>The input can be i) a genlight object (which has to have the latlon slot 
populated), ii) a pair of <code>Dgeo</code> and <code>Dgen</code>, which have to be
either
<code>matrix</code> or <code>dist</code> objects, or iii) a <code>list</code> of the 
<code>matrix</code> or <code>dist</code> objects if the 
analysis needs to be carried out for multiple populations (in this case, 
all the elements of the <code>list</code> have to be of the same class (i.e. 
<code>matrix</code> or <code>dist</code>) and the population order in the two lists has 
to be the same. 
</p>
<p>If the input is a genlight object, the function calculates the linear 
distance
for <code>Dgeo</code> and the relevant <code>Dgen</code> matrix (see <code>Dgen_method</code>) 
for each population. 
When the method selected is a genetic similarity matrix (e.g. &quot;simple&quot; 
distance), the matrix is internally transformed with <code>1 - Dgen</code> so that 
positive values of autocorrelation coefficients indicates more related 
individuals similarly as implemented in GenAlEx. If the user provide the 
distance matrices, care must be taken in interpreting the results because
similarity matrix will generate negative values for closely related 
individuals.
</p>
<p>If <code>max(Dgeo)&gt;1000</code> (e.g. the geographic distances are in thousands of 
metres), values are divided by 1000 (in the example before these would then 
become km) to facilitate readability of the plots.
</p>
<p>If <code>bins</code> is of length = 1 it is interpreted as the number of (even)
bins to use. In this case the starting point is always the minimum value in 
the distance matrix, and the last is the maximum. If it is a numeric vector 
of length&gt;1, it is interpreted as the breaking points. In this case, the 
first has to be the lowest value, and the last has to be the highest. There 
are no internal checks for this and it is user responsibility to ensure that
distance classes are properly set up. If that is not the case, data that fall
outside the range provided will be dropped. The number of bins will be 
<code>length(bins) - 1</code>.
</p>
<p>The permutation constructs the 95% confidence intervals around the null
hypothesis of no spatial structure (this is a two-tail test). The same data
are also used to calculate the probability of the one-tail test (See 
references below for details).
</p>
<p>Bootstrap calculations are skipped and <code>NA</code> is returned when the number 
of possible combinations given the sample size of any given distance class is
&lt; <code>reps</code>.
</p>
<p>Methods available to calculate genetic distances for SNP data:
</p>

<ul>
<li><p> &quot;propShared&quot; using the function <code><a href="#topic+gl.propShared">gl.propShared</a></code>.
</p>
</li>
<li><p> &quot;grm&quot; using the function <code><a href="#topic+gl.grm">gl.grm</a></code>.
</p>
</li>
<li><p> &quot;Euclidean&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Simple&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Absolute&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Manhattan&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li></ul>

<p>Methods available to calculate genetic distances for SilicoDArT data:
</p>

<ul>
<li><p> &quot;Euclidean&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Simple&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Jaccard&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li>
<li><p> &quot;Bray-Curtis&quot; using the function <code><a href="#topic+gl.dist.ind">gl.dist.ind</a></code>.
</p>
</li></ul>

<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with
the function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed
only in the current R session because tempdir is cleared each time that the
R session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a data frame with the following columns:
</p>

<ol>
<li><p> Bin  The distance classes
</p>
</li>
<li><p> N The number of pairwise comparisons within each distance class
</p>
</li>
<li><p> r.uc The uncorrected autocorrelation coefficient
</p>
</li>
<li><p> Correction the correction
</p>
</li>
<li><p> r The corrected autocorrelation coefficient
</p>
</li>
<li><p> L.r The corrected autocorrelation coefficient lower limit
(if <code>bootstap = TRUE</code>)
</p>
</li>
<li><p> U.r The corrected autocorrelation coefficient upper limit
(if <code>bootstap = TRUE</code>)
</p>
</li>
<li><p> L.r.null.uc The uncorrected lower limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> U.r.null.uc  The uncorrected upper limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> L.r.null The corrected lower limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> U.r.null The corrected upper limit for the null hypothesis of no 
spatial autocorrelation (if <code>permutation = TRUE</code>)
</p>
</li>
<li><p> p.one.tail The p value of the one tail statistical test
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Carlo Pacioni, Bernd Gruber &amp; Luis Mijangos 
(Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Smouse PE, Peakall R. 1999. Spatial autocorrelation analysis of
individual multiallele and multilocus genetic structure. Heredity 82:
561-573.
</p>
</li>
<li><p> Double, MC, et al. 2005. Dispersal, philopatry and infidelity: 
dissecting local genetic structure in superb fairy-wrens (Malurus cyaneus). 
Evolution 59, 625-635.
</p>
</li>
<li><p> Peakall, R, et al. 2003. Spatial autocorrelation analysis offers new
insights into gene flow in the Australian bush rat, Rattus fuscipes.
Evolution 57, 1182-1195.
</p>
</li>
<li><p> Smouse, PE, et al. 2008. A heterogeneity test for fine-scale genetic
structure. Molecular Ecology 17, 3389-3400.
</p>
</li>
<li><p> Gonzales, E, et al. 2010. The impact of landscape disturbance on 
spatial genetic structure in the Guanacaste tree, Enterolobium
cyclocarpum (Fabaceae). Journal of Heredity 101, 133-143.
</p>
</li>
<li><p> Beck, N, et al. 2008. Social constraint and an absence of sex-biased
dispersal drive fine-scale genetic structure in white-winged choughs.
Molecular Ecology 17, 4346-4358.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
res &lt;- gl.spatial.autoCorr(platypus.gl, bins=seq(0,10000,2000))
# using one population, showing sample size
test &lt;- gl.keep.pop(platypus.gl,pop.list = "TENTERFIELD")
res &lt;- gl.spatial.autoCorr(test, bins=seq(0,10000,2000),CI_color = "green")

test &lt;- gl.keep.pop(platypus.gl,pop.list = "TENTERFIELD")
res &lt;- gl.spatial.autoCorr(test, bins=seq(0,10000,2000),CI_color = "green")
</code></pre>

<hr>
<h2 id='gl.subsample.loci'>Subsamples n loci from a genlight object and return it as a genlight object</h2><span id='topic+gl.subsample.loci'></span>

<h3>Description</h3>

<p>This is a support script, to subsample a genlight {adegenet} object based
on loci. Two methods are used to subsample, random and based on information
content.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.subsample.loci(x, n, method = "random", mono.rm = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.subsample.loci_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl.subsample.loci_+3A_n">n</code></td>
<td>
<p>Number of loci to include in the subsample [required].</p>
</td></tr>
<tr><td><code id="gl.subsample.loci_+3A_method">method</code></td>
<td>
<p>Method: 'random', in which case the loci are sampled at random;
or 'pic', in which case the top n loci ranked on information content are
chosen. Information content is stored in AvgPIC in the case of SNP data and in
PIC in the the case of presence/absence (SilicoDArT) data [default 'random'].</p>
</td></tr>
<tr><td><code id="gl.subsample.loci_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Delete monomorphic loci before sampling [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.subsample.loci_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object with n loci
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl2 &lt;- gl.subsample.loci(testset.gl, n=200, method='pic')
# Tag P/A data
  gl2 &lt;- gl.subsample.loci(testset.gl, n=100, method='random')
</code></pre>

<hr>
<h2 id='gl.test.heterozygosity'>Tests the difference in heterozygosity between populations taken
pairwise</h2><span id='topic+gl.test.heterozygosity'></span>

<h3>Description</h3>

<p>Calculates the expected heterozygosities for each population in a genlight
object, and uses re-randomization to test the statistical significance of
differences in heterozygosity between populations taken pairwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.test.heterozygosity(
  x,
  nreps = 100,
  alpha1 = 0.05,
  alpha2 = 0.01,
  plot.out = TRUE,
  max_plots = 6,
  plot_theme = theme_dartR(),
  plot_colors = two_colors,
  save2tmp = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.test.heterozygosity_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_nreps">nreps</code></td>
<td>
<p>Number of replications of the re-randomization [default 1,000].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_alpha1">alpha1</code></td>
<td>
<p>First significance level for comparison with diff=0 on plot
[default 0.05].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_alpha2">alpha2</code></td>
<td>
<p>Second significance level for comparison with diff=0 on plot
[default 0.01].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot.out">plot.out</code></td>
<td>
<p>If TRUE, plots a sampling distribution of the differences for
each comparison [default TRUE].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_max_plots">max_plots</code></td>
<td>
<p>Maximum number of plots to print per page [default 6].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot_theme">plot_theme</code></td>
<td>
<p>Theme for the plot. See Details for options
[default theme_dartR()].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_plot_colors">plot_colors</code></td>
<td>
<p>List of two color names for the borders and fill of the
plots [default two_colors].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_save2tmp">save2tmp</code></td>
<td>
<p>If TRUE, saves any ggplots and listings to the session
temporary directory (tempdir) [default FALSE].</p>
</td></tr>
<tr><td><code id="gl.test.heterozygosity_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong> Function's output </strong>
</p>
<p>If plot.out = TRUE, plots are created showing the sampling distribution for
the difference between each pair of heterozygosities, marked with the
critical limits alpha1 and alpha2, the observed heterozygosity, and the zero
value (if in range).
</p>
<p>Plots and table are saved to the temporal directory (tempdir) and can be
accessed with the function <code><a href="#topic+gl.print.reports">gl.print.reports</a></code> and listed with the
function <code><a href="#topic+gl.list.reports">gl.list.reports</a></code>. Note that they can be accessed only
in the current R session because tempdir is cleared each time that the R
session is closed.
</p>
<p>Examples of other themes that can be used can be consulted in </p>

<ul>
<li> <p><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">https://ggplot2.tidyverse.org/reference/ggtheme.html</a> and </p>
</li>
<li>
<p><a href="https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/">https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A dataframe containing population labels, heterozygosities and sample
sizes
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl.test.heterozygosity(platypus.gl, nreps=1, verbose=3, plot.out=TRUE)
</code></pre>

<hr>
<h2 id='gl.tree.nj'>Outputs an nj tree to summarize genetic similarity among populations</h2><span id='topic+gl.tree.nj'></span>

<h3>Description</h3>

<p>This function is a wrapper for the nj function or package ape applied to Euclidean
distances calculated from the genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.tree.nj(
  x,
  d_mat = NULL,
  type = "phylogram",
  outgroup = NULL,
  labelsize = 0.7,
  treefile = NULL,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.tree.nj_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_d_mat">d_mat</code></td>
<td>
<p>Distance matrix [default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_type">type</code></td>
<td>
<p>Type of dendrogram &quot;phylogram&quot;|&quot;cladogram&quot;|&quot;fan&quot;|&quot;unrooted&quot;
[default &quot;phylogram&quot;].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_outgroup">outgroup</code></td>
<td>
<p>Vector containing the population names that are the outgroups
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_labelsize">labelsize</code></td>
<td>
<p>Size of the labels as a proportion of the graphics default
[default 0.7].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_treefile">treefile</code></td>
<td>
<p>Name of the file for the tree topology using Newick format 
[default NULL].</p>
</td></tr>
<tr><td><code id="gl.tree.nj_+3A_verbose">verbose</code></td>
<td>
<p>Specify the level of verbosity: 0, silent, fatal errors only; 
1, flag function begin and end; 2, progress log; 3, progress and results 
summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An euclidean distance matrix is calculated by default [d_mat = NULL]. 
Optionally the user can use as input for the tree any other distance matrix
using this parameter, see for example the function  <code><a href="#topic+gl.dist.pop">gl.dist.pop</a></code>.
</p>


<h3>Value</h3>

<p>A tree file of class phylo.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# SNP data
  gl.tree.nj(testset.gl,type='fan')
# Tag P/A data
  gl.tree.nj(testset.gs,type='fan')
  
  res &lt;- gl.tree.nj(platypus.gl)
</code></pre>

<hr>
<h2 id='gl.write.csv'>Writes out data from a genlight object to csv file</h2><span id='topic+gl.write.csv'></span>

<h3>Description</h3>

<p>This script writes to file the SNP genotypes with specimens as entities
(columns) and loci as attributes (rows). Each row has associated locus
metadata. Each column, with header of specimen id, has population in the
first row.
</p>
<p>The data coding differs from the DArT 1row format in that 0 = reference
homozygous, 2 = alternate homozygous, 1 = heterozygous, and NA = missing SNP
assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl.write.csv(x, outfile = "outfile.csv", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl.write.csv_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;outfile.csv&quot;].</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl.write.csv_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Saves a genlight object to csv, returns NULL.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
  gl.write.csv(testset.gl, outfile='SNP_1row.csv')
# Tag P/A data
  gl.write.csv(testset.gs, outfile='PA_1row.csv')
</code></pre>

<hr>
<h2 id='gl2bayescan'>Converts a genlight object into a format suitable for input to Bayescan</h2><span id='topic+gl2bayescan'></span>

<h3>Description</h3>

<p>The output text file contains the SNP data and relevant BAyescan command
lines to guide input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2bayescan(x, outfile = "bayescan.txt", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2bayescan_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default bayescan.txt].</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2bayescan_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Foll M and OE Gaggiotti (2008) A genome scan method to identify selected loci
appropriate for both dominant and codominant markers: A Bayesian
perspective. Genetics 180: 977-993.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl2bayescan(testset.gl)
</code></pre>

<hr>
<h2 id='gl2bpp'>Converts a genlight object into a format suitable for input to the BPP 
program</h2><span id='topic+gl2bpp'></span>

<h3>Description</h3>

<p>This function generates the sequence alignment file and the Imap file. The 
control file should produced by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2bpp(
  x,
  method = 1,
  outfile = "output_bpp.txt",
  imap = "Imap.txt",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2bpp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_method">method</code></td>
<td>
<p>One of 1 | 2, see details [default = 1].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_outfile">outfile</code></td>
<td>
<p>Name of the sequence alignment file [&quot;output_bpp.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_imap">imap</code></td>
<td>
<p>Name of the Imap file [&quot;Imap.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file (set to tempdir by default)</p>
</td></tr>
<tr><td><code id="gl2bpp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If method = 1, heterozygous positions are replaced by standard ambiguity 
codes.
</p>
<p>If method = 2, the heterozygous state is resolved by randomly assigning one 
or the other SNP variant to the individual.
</p>
<p>Trimmed sequences for which the SNP has been trimmed out, rarely, by adapter
mis-identity are deleted.
</p>
<p>This function requires 'TrimmedSequence' to be among the locus metrics
(<code>@other$loc.metrics</code>) and information of the type of alleles (slot
loc.all e.g. 'G/A') and the position of the SNP in slot position of the
&ldquo;'genlight&ldquo;' object (see testset.gl@position and testset.gl@loc.all for
how to format these slots.)
</p>
<p>It's important to keep in mind that analyses based on coalescent theory, 
like those done by the programme BPP, are meant to be used with sequence
data. In this type of data, large chunks of DNA are sequenced, so when we
find polymorphic sites along the sequence, we know they are all on the same
chromosome. This kind of data, in which we know which chromosome each 
allele comes from, is called &quot;phased data.&quot; Most data from reduced 
representation genome-sequencing methods, like DArTseq, is unphased, 
which means that we don't know which chromosome each allele comes from. 
So, if we apply coalescence theory to data that is not phased, we will get
biased results. As in Ellegren et al., one way to deal with this is to 
&quot;haplodize&quot; each genotype by randomly choosing one allele from 
heterozygous genotypes (2012) by using method = 2.
</p>
<p>Be mindful that there is little information in the literature on the
validity of this method.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Ellegren, Hans, et al. &quot;The genomic landscape of species divergence in 
Ficedula flycatchers.&quot; Nature 491.7426 (2012): 756-760.
</p>
</li>
<li><p> Flouri T., Jiao X., Rannala B., Yang Z. (2018) Species Tree Inference 
with BPP using Genomic Sequences and the Multispecies Coalescent. Molecular
Biology and Evolution, 35(10):2585-2593. doi:10.1093/molbev/msy147
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>require(dartR.data)
test &lt;- platypus.gl
test &lt;- gl.filter.callrate(test,threshold = 1)
test &lt;- gl.filter.monomorphs(test)
test &lt;- gl.subsample.loci(test,n=25)
gl2bpp(x = test)
</code></pre>

<hr>
<h2 id='gl2demerelate'>Creates a dataframe suitable for input to package {Demerelate} from a
genlight {adegenet} object</h2><span id='topic+gl2demerelate'></span>

<h3>Description</h3>

<p>Creates a dataframe suitable for input to package {Demerelate} from a
genlight {adegenet} object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2demerelate(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2demerelate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2demerelate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe suitable as input to package {Demerelate}
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- gl2demerelate(testset.gl)
</code></pre>

<hr>
<h2 id='gl2eigenstrat'>Converts a genlight object into eigenstrat format</h2><span id='topic+gl2eigenstrat'></span>

<h3>Description</h3>

<p>The output of this function are three files:
</p>

<ul>
<li><p> genotype file: contains genotype data for each individual at each SNP
with an extension 'eigenstratgeno.'
</p>
</li>
<li><p> snp file: contains information about each SNP with an extension 'snp.'
</p>
</li>
<li><p> indiv file: contains information about each individual with an
extension 'ind.'
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>gl2eigenstrat(
  x,
  outfile = "gl_eigenstrat",
  outpath = tempdir(),
  snp_pos = 1,
  snp_chr = 1,
  pos_cM = 0,
  sex_code = "unknown",
  phen_value = "Case",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2eigenstrat_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_eigenstrat'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_snp_pos">snp_pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_snp_chr">snp_chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_pos_cm">pos_cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default 1].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_sex_code">sex_code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default 'unknown'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_phen_value">phen_value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value ('Case', 'Control') [default 'Case'].</p>
</td></tr>
<tr><td><code id="gl2eigenstrat_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Eigenstrat only accepts chromosomes coded as numeric values, as follows:
X chromosome is encoded as 23, Y is encoded as 24, mtDNA is encoded as
90, and XY is encoded as 91. SNPs with illegal chromosome values, such
as 0, will be removed.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>


<ul>
<li><p> Patterson, N., Price, A. L., &amp; Reich, D. (2006). Population structure
and eigenanalysis. PLoS genetics, 2(12), e190.
</p>
</li>
<li><p> Price, A. L., Patterson, N. J., Plenge, R. M., Weinblatt, M. E.,
Shadick, N. A., &amp; Reich, D. (2006). Principal components analysis corrects
for stratification in genome-wide association studies. Nature genetics,
38(8), 904-909.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
gl2eigenstrat(platypus.gl,snp_pos='ChromPos_Platypus_Chrom_NCBIv1',
snp_chr = 'Chrom_Platypus_Chrom_NCBIv1')

</code></pre>

<hr>
<h2 id='gl2fasta'>Concatenates DArT trimmed sequences and outputs a FASTA file</h2><span id='topic+gl2fasta'></span>

<h3>Description</h3>

<p>Concatenated sequence tags are useful for phylogenetic methods where
information on base frequencies and transition and transversion ratios are
required (for example, Maximum Likelihood methods). Where relevant,
heterozygous loci are resolved before concatenation by either assigning
ambiguity codes or by random allele assignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2fasta(
  x,
  method = 1,
  outfile = "output.fasta",
  outpath = tempdir(),
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2fasta_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_method">method</code></td>
<td>
<p>One of 1 | 2 | 3 | 4. Type method=0 for a list of options 
[method=1].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_outfile">outfile</code></td>
<td>
<p>Name of the output file (fasta format) [&quot;output.fasta&quot;].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file (set to tempdir by default)</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_probar">probar</code></td>
<td>
<p>If TRUE, a progress bar will be displayed for long loops 
[default = TRUE].</p>
</td></tr>
<tr><td><code id="gl2fasta_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Four methods are employed:
</p>
<p>Method 1 &ndash; heterozygous positions are replaced by the standard ambiguity 
codes. The resultant sequence fragments are concatenated across loci to
generate a single combined sequence to be used in subsequent ML phylogenetic
analyses.
</p>
<p>Method 2 &ndash; the heterozygous state is resolved by randomly assigning one or
the other SNP variant to the individual. The resultant sequence fragments are
concatenated across loci to generate a single composite haplotype to be used
in subsequent ML phylogenetic analyses.
</p>
<p>Method 3 &ndash; heterozygous positions are replaced by the standard ambiguity
codes. The resultant SNP bases are concatenated across loci to generate a
single combined sequence to be used in subsequent MP phylogenetic analyses.
</p>
<p>Method 4 &ndash; the heterozygous state is resolved by randomly assigning one or
the other SNP variant to the individual. The resultant SNP bases are
concatenated across loci to generate a single composite haplotype to be used
in subsequent MP phylogenetic analyses.
</p>
<p>Trimmed sequences for which the SNP has been trimmed out, rarely, by adapter
mis-identity are deleted.
</p>
<p>The script writes out the composite haplotypes for each individual as a
fastA file. Requires 'TrimmedSequence' to be among the locus metrics
(<code>@other$loc.metrics</code>) and information of the type of alleles (slot
loc.all e.g. 'G/A') and the position of the SNP in slot position of the
&ldquo;'genlight&ldquo;' object (see testset.gl@position and testset.gl@loc.all for
how to format these slots.)
</p>


<h3>Value</h3>

<p>A new gl object with all loci rendered homozygous.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
gl &lt;- gl.filter.reproducibility(testset.gl,t=1)
gl &lt;- gl.filter.overshoot(gl,verbose=3)
gl &lt;- gl.filter.callrate(testset.gl,t=.98)
gl &lt;- gl.filter.monomorphs(gl)
gl2fasta(gl, method=1, outfile='test.fasta',verbose=3)

test &lt;- gl.subsample.loci(platypus.gl,n=100)
gl2fasta(test)
</code></pre>

<hr>
<h2 id='gl2faststructure'>Converts a genlight object into faststructure format (to run faststructure
elsewhere)</h2><span id='topic+gl2faststructure'></span>

<h3>Description</h3>

<p>Recodes in the quite specific faststructure format (e.g first six columns
need to be there, but are ignored...check faststructure documentation
(if you find any :-( )))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2faststructure(
  x,
  outfile = "gl.str",
  outpath = tempdir(),
  probar = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2faststructure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;gl.str&quot;].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_probar">probar</code></td>
<td>
<p>Switch to show/hide progress bar [default FALSE].</p>
</td></tr>
<tr><td><code id="gl2faststructure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The script writes out the a file in faststructure format.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl2gds'>Converts a genlight object into gds format</h2><span id='topic+gl2gds'></span>

<h3>Description</h3>

<p>Package SNPRelate relies on a bit-level representation of a SNP dataset that
competes with {adegenet} genlight objects and associated files. This
function converts a genlight object to a gds format file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2gds(
  x,
  outfile = "gl_gds.gds",
  outpath = tempdir(),
  snp_pos = "0",
  snp_chr = "0",
  chr_format = "character",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2gds_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'gl_gds.gds'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_snp_pos">snp_pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_snp_chr">snp_chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_chr_format">chr_format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2gds_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function orders the SNPS by chromosome and by position before converting
to SNPRelate format, as required by this package.
</p>
<p>The chromosome of each SNP can be a character or numeric, as described in the
vignette of SNPRelate:
'snp.chromosome, an integer or character mapping for each chromosome.
Integer: numeric values 1-26, mapped in order from 1-22, 23=X, 24=XY
(the pseudoautosomal region), 25=Y, 26=M (the mitochondrial probes), and 0
for probes with unknown positions; it does not allow NA. Character: “X”,
“XY”, “Y” and “M” can be used here, and a blank string indicating unknown
position.'
</p>
<p>When using some functions from package SNPRelate with datasets other than
humans it might be necessary to use the option autosome.only=FALSE to avoid
detecting chromosome coding. So, it is important to read the documentation of
the function before using it.
</p>
<p>The chromosome information for unmapped SNPS is coded as 0, as required by
SNPRelate.
</p>
<p>Remember to close the GDS file before working in a different GDS object with
the function <a href="SNPRelate.html#topic+snpgdsClose">snpgdsClose</a> (package SNPRelate).
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
gl2gds(platypus.gl,snp_pos='ChromPos_Platypus_Chrom_NCBIv1',
snp_chr = 'Chrom_Platypus_Chrom_NCBIv1')

</code></pre>

<hr>
<h2 id='gl2genalex'>Converts a genlight object into a format suitable for input to genalex</h2><span id='topic+gl2genalex'></span>

<h3>Description</h3>

<p>The output csv file contains the snp data and other relevant lines suitable
for genalex. This script is a wrapper for  <a href="poppr.html#topic+genind2genalex">genind2genalex</a>
(package poppr).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2genalex(
  x,
  outfile = "genalex.csv",
  outpath = tempdir(),
  overwrite = FALSE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2genalex_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'genalex.csv'].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir()].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_overwrite">overwrite</code></td>
<td>
<p>If FALSE and filename exists, then the file will not be
overwritten. Set this option to TRUE to overwrite the file [default FALSE].</p>
</td></tr>
<tr><td><code id="gl2genalex_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end;
2, progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos, Author: Katrin Hohwieler, wrapper Arthur
Georges (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Peakall, R. and Smouse P.E. (2012) GenAlEx 6.5: genetic analysis
in Excel. Population genetic software for teaching and research-an update.
Bioinformatics 28, 2537-2539.
http://bioinformatics.oxfordjournals.org/content/28/19/2537
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
gl2genalex(testset.gl, outfile='testset.csv')

</code></pre>

<hr>
<h2 id='gl2genepop'>Converts a genlight object into genepop format (and file)</h2><span id='topic+gl2genepop'></span>

<h3>Description</h3>

<p>The genepop format is used by several external applications (for example
Neestimator2
(<a href="http://www.molecularfisherieslaboratory.com.au/neestimator-software/">http://www.molecularfisherieslaboratory.com.au/neestimator-software/</a>).
So the main idea is to create the genepop file and then run the other
software externally. As a feature, the genepop file is also returned as an
invisible data.frame by the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2genepop(
  x,
  outfile = "genepop.gen",
  outpath = tempdir(),
  pop_order = "alphabetic",
  output_format = "2_digits",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2genepop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'genepop.gen'].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file. Use outpath=getwd() or
outpath='.' when calling this function to direct output files to your working
directory [default tempdir(), mandated by CRAN].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_pop_order">pop_order</code></td>
<td>
<p>Order of the output populations either &quot;alphabetic&quot; or a 
vector of population names in the order required by the user (see examples)
[default &quot;alphabetic&quot;].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_output_format">output_format</code></td>
<td>
<p>Whether to use a 2-digit format (&quot;2_digits&quot;) or 3-digits
format (&quot;3_digits&quot;) [default &quot;2_digits&quot;].</p>
</td></tr>
<tr><td><code id="gl2genepop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible data frame in genepop format
</p>


<h3>Author(s)</h3>

<p>Custodian: Bernd Gruber (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("dartR.data")
# SNP data
geno &lt;- gl2genepop(testset.gl[1:3,1:9])
head(geno)
test &lt;- gl.filter.callrate(platypus.gl,threshold = 1)
popNames(test)
gl2genepop(test, pop_order = c("TENTERFIELD","SEVERN_ABOVE","SEVERN_BELOW"),
           output_format="3_digits")

## End(Not run)
</code></pre>

<hr>
<h2 id='gl2geno'>Converts a genlight object to geno format from package LEA</h2><span id='topic+gl2geno'></span>

<h3>Description</h3>

<p>The function converts a genlight object (SNP or presence/absence
i.e. SilicoDArT data) into a file in the 'geno' and the 'lfmm' formats from 
(package LEA).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2geno(x, outfile = "gl_geno", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2geno_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_geno'].</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2geno_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP data
gl2geno(testset.gl)
# Tag P/A data
gl2geno(testset.gs)
</code></pre>

<hr>
<h2 id='gl2gi'>Converts a genlight object to genind object</h2><span id='topic+gl2gi'></span>

<h3>Description</h3>

<p>Converts a genlight object to genind object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2gi(x, probar = FALSE, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2gi_+3A_x">x</code></td>
<td>
<p>A genlight object [required].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_probar">probar</code></td>
<td>
<p>If TRUE, a progress bar will be displayed for long loops
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl2gi_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses a faster version of df2genind (from the adegenet
package)
</p>


<h3>Value</h3>

<p>A genind object, with all slots filled.
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='gl2hiphop'>Converts a genlight objects into hiphop format</h2><span id='topic+gl2hiphop'></span>

<h3>Description</h3>

<p>This function exports genlight objects to the format used by the parentage
assignment R package hiphop. Hiphop can be used for paternity and maternity
assignment and outperforms conventional methods where closely related
individuals occur in the pool of possible parents. The method compares the
genotypes of offspring with any combination of potentials parents and scores
the number of mismatches of these individuals at bi-allelic genetic markers
(e.g. Single Nucleotide Polymorphisms).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2hiphop(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2hiphop_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2hiphop_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe containing all the genotyped individuals (offspring and
potential parents) and their genotypes scored using bi-allelic markers.
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Cockburn, A., Penalba, J.V.,Jaccoud, D.,Kilian, A., Brouwer, L.,Double, M.C.,
Margraf, N., Osmond, H.L., van de Pol, M. and Kruuk, L.E.B.(in revision).
HIPHOP: improved paternity assignment among close relatives using a simple
exclusion method for bi-allelic markers. Molecular Ecology Resources, DOI to
be added upon acceptance
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- gl2hiphop(testset.gl)

</code></pre>

<hr>
<h2 id='gl2phylip'>Creates a Phylip input distance matrix from a genlight (SNP) {adegenet}
object</h2><span id='topic+gl2phylip'></span>

<h3>Description</h3>

<p>This function calculates and returns a matrix of Euclidean distances between 
populations and produces an input file for the phylogenetic program Phylip 
(Joe Felsenstein).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2phylip(
  x,
  outfile = "phyinput.txt",
  outpath = tempdir(),
  bstrap = 1,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2phylip_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP or presence/absence
(SilicoDArT) data [required].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_outfile">outfile</code></td>
<td>
<p>Name of the file to become the input file for phylip
[default &quot;phyinput.txt&quot;].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file 
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_bstrap">bstrap</code></td>
<td>
<p>Number of bootstrap replicates [default 1].</p>
</td></tr>
<tr><td><code id="gl2phylip_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2, 
progress log; 3, progress and results summary; 5, full report 
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of Euclidean distances between populations.
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- gl2phylip(testset.gl, outfile='test.txt', bstrap=10)

</code></pre>

<hr>
<h2 id='gl2plink'>Converts a genlight object into PLINK format</h2><span id='topic+gl2plink'></span>

<h3>Description</h3>

<p>This function exports a genlight object into PLINK format and save it into a
file.
This function produces the following PLINK files: bed, bim, fam, ped and map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2plink(
  x,
  plink_path = getwd(),
  bed_file = FALSE,
  outfile = "gl_plink",
  outpath = tempdir(),
  chr_format = "character",
  pos_cM = "0",
  ID_dad = "0",
  ID_mom = "0",
  sex_code = "unknown",
  phen_value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2plink_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_plink_path">plink_path</code></td>
<td>
<p>Path of PLINK binary file [default getwd()].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_bed_file">bed_file</code></td>
<td>
<p>Whether create PLINK files .bed, .bim and .fam
[default FALSE].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_plink'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_chr_format">chr_format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_pos_cm">pos_cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_id_dad">ID_dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_id_mom">ID_mom</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_sex_code">sex_code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown'). Sex information needs 
just to start with an &quot;F&quot; or &quot;f&quot; for females, with an &quot;M&quot; or &quot;m&quot; for males 
and with a &quot;U&quot;, &quot;u&quot; or being empty if the sex is unknown [default 'unknown'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_phen_value">phen_value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td></tr>
<tr><td><code id="gl2plink_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To create PLINK files .bed, .bim and .fam (bed_file = TRUE), it is necessary
to download the binary file of PLINK 1.9 and provide its path (plink_path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
</p>
<p>After downloading, unzip the file, access the unzipped folder and move the 
binary file (&quot;plink&quot;) to your working directory.
</p>
<p>If you are using a Mac, you might need to open the binary first to grant 
access to the binary. 
</p>
<p>The chromosome of each SNP can be a character or numeric. The chromosome
information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop.
Within-family ID (cannot be '0') is taken from indNames(x).
Variant identifier is taken from locNames(x).
SNP position is taken from the accessor x$position.
Chromosome name is taken from the accessor x$chromosome
Note that if names of populations or individuals contain spaces, they are 
replaced by an underscore &quot;_&quot;.
</p>
<p>If you like to use chromosome information when converting to plink format and
your chromosome names are not from human, you need to change the chromosome 
names as 'contig1', 'contig2', etc. as described in the section &quot;Nonstandard
chromosome IDs&quot; in the following link:
https://www.cog-genomics.org/plink/1.9/input
</p>
<p>Note that the function might not work if there are spaces in the path to the
plink executable.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Purcell, Shaun, et al. 'PLINK: a tool set for whole-genome association and
population-based linkage analyses.' The American journal of human genetics
81.3 (2007): 559-575.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require("dartR.data")
test &lt;- platypus.gl
# assigning SNP position
test$position &lt;- test$other$loc.metrics$ChromPos_Platypus_Chrom_NCBIv1
# assigning a dummy name for chromosomes
test$chromosome &lt;- as.factor("1")
gl2plink(test)

</code></pre>

<hr>
<h2 id='gl2related'>Converts a genlight object to format suitable to be run with Coancestry</h2><span id='topic+gl2related'></span>

<h3>Description</h3>

<p>The output txt file contains the SNP data and an additional column with the
names of the individual. The file then can be used and loaded into coancestry
or - if installed - run with the related package. Be aware the related
package was crashing in previous versions, but in general is using the same
code as coancestry and therefore should have identical results. Also running
coancestry with thousands of SNPs via the GUI seems to be not reliable and
therefore for comparisons between coancestry and related we suggest to use
the command line version of coancestry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2related(
  x,
  outfile = "related.txt",
  outpath = tempdir(),
  save = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2related_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'related.txt'].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir()].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_save">save</code></td>
<td>
<p>A switch if you want to save the file or not. This might be
useful for someone who wants to use the coancestry function to calculate
relatedness and not export to coancestry. See the example below
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl2related_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame that can be used to run with the related package
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Jack Pew, Jinliang Wang, Paul Muir and Tim Frasier (2014).
related: related: an R package for analyzing pairwise relatedness
data based on codominant molecular markers.
R package version 0.8/r2.
<a href="https://R-Forge.R-project.org/projects/related/">https://R-Forge.R-project.org/projects/related/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gtd &lt;- gl2related(bandicoot.gl[1:10,1:20], save=FALSE)
## Not run: 
##running with the related package
#install.packages('related', repos='http://R-Forge.R-project.org')
library(related)
coan &lt;- coancestry(gtd, wang=1)
head(coan$relatedness)
##check ?coancestry for information how to use the function.

## End(Not run)
</code></pre>

<hr>
<h2 id='gl2sa'>Converts genlight objects to the format used in the SNPassoc package</h2><span id='topic+gl2sa'></span>

<h3>Description</h3>

<p>This function exports a genlight object into a SNPassoc object. See package
SNPassoc for details. This function needs package SNPassoc. At the time of
writing (August 2020) the package was no longer available from CRAN. To
install the package check their github repository.
<a href="https://github.com/isglobal-brge/SNPassoc">https://github.com/isglobal-brge/SNPassoc</a> and/or use
<code>install_github('isglobal-brge/SNPassoc')</code> to install the function and
uncomment the function code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2sa(x, verbose = NULL, installed = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2sa_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2sa_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="gl2sa_+3A_installed">installed</code></td>
<td>
<p>Switch to run the function once SNPassoc package i
s installed [default FALSE].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class 'snp' to be used with SNPassoc.
</p>


<h3>Author(s)</h3>

<p>Bernd Guber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Gonzalez, J.R., Armengol, L., Sol?, X., Guin?, E., Mercader, J.M., Estivill,
X. and Moreno, V. (2017). SNPassoc: an R package to perform whole genome
association studies. Bioinformatics 23:654-655.
</p>

<hr>
<h2 id='gl2sfs'>Converts a genlight object into a sfs input file</h2><span id='topic+gl2sfs'></span>

<h3>Description</h3>

<p>The output of this function is suitable for analysis in fastsimcoal2 or dada.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2sfs(
  x,
  n.invariant.tags = 0,
  outfile_root = "gl2sfs",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2sfs_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2sfs_+3A_n.invariant.tags">n.invariant.tags</code></td>
<td>
<p>Number of invariant sites[default 0].</p>
</td></tr>
<tr><td><code id="gl2sfs_+3A_outfile_root">outfile_root</code></td>
<td>
<p>The root of the name of the output file [default &quot;gl2sfs&quot;].</p>
</td></tr>
<tr><td><code id="gl2sfs_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file [default tempdir()].</p>
</td></tr>
<tr><td><code id="gl2sfs_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It saves a derived sfs, assuming that the reference allele is the ancestral,
and a MAF sfs.
</p>
<p>At this stage this function caters only for diploid organisms, for samples
from one population only, and for genotypes without missing data. Note that
sfs uses frequencies considered <b>independent</b>, data are assumed to be
from independent (i.e. not linked) loci. This means that only one site per tag
should be considered 9i.e. secondaries should be removed). If no monomorphic
site estimates is provided (with <code>n.invariant.tags</code>), the sfs will only
include the number of monomorphic sites in the data (but this will be a biased
estimates as it doesn't take into account the invariant tags that have not
been included. This will affect parameter estimates in the analyses). Note
that the number of invariant tags can be estimated with
<code>gl.report.secondaries</code>. In a limited number of cases, ascertainment bias
can be explicitly modelled in fastsimcoal2. See fastsimcoal2 manual for
details.
</p>
<p>It expects a dartR formatted genlight object, but it should also  work with
other genlight objects.
</p>


<h3>Value</h3>

<p>Deprecated. Please use gl.sfs instead.
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Excoffier L., Dupanloup I., Huerta-Sánchez E., Sousa V. C. and
Foll M. (2013) Robust demographic inference from genomic and SNP data. PLoS
genetics 9(10)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code>,
<code><a href="#topic+gl.report.secondaries">gl.report.secondaries</a></code>, <code><a href="#topic+utils.n.var.invariant">utils.n.var.invariant</a></code>
</p>

<hr>
<h2 id='gl2shp'>Converts a genlight object to ESRI shapefiles or kml files</h2><span id='topic+gl2shp'></span>

<h3>Description</h3>

<p>This function exports coordinates in a genlight object to a point shape file
(including also individual meta data if available).
Coordinates are provided under x@other$latlon and assumed to be in WGS84
coordinates, if not proj4 string is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2shp(
  x,
  type = "shp",
  proj4 = "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs",
  outfile = "gl",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2shp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data and location
data, lat longs [required].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_type">type</code></td>
<td>
<p>Type of output 'kml' or 'shp' [default 'shp'].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_proj4">proj4</code></td>
<td>
<p>Proj4string of data set (see spatialreference.org for
projections) [default WGS84].</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_outfile">outfile</code></td>
<td>
<p>Name (path) of the output shape file [default 'gl']. shp
extension is added automatically.</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2shp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a SpatVector file
</p>


<h3>Author(s)</h3>

<p>Bernd Guber (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- gl2shp(testset.gl)
</code></pre>

<hr>
<h2 id='gl2snapp'>Converts a genlight object to nexus format suitable for phylogenetic analysis
by SNAPP (via BEAUti)</h2><span id='topic+gl2snapp'></span>

<h3>Description</h3>

<p>The output nexus file contains the SNP data and relevant PAUP command lines
suitable for BEAUti.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2snapp(x, outfile = "snapp.nex", outpath = tempdir(), verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2snapp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default &quot;snapp.nex&quot;].</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2snapp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Bryant, D., Bouckaert, R., Felsenstein, J., Rosenberg, N.A. and
RoyChoudhury, A. (2012). Inferring species trees directly from biallelic
genetic markers: bypassing gene trees in a full coalescent analysis.
Molecular Biology and Evolution 29:1917-1932.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2snapp(testset.gl)
</code></pre>

<hr>
<h2 id='gl2structure'>Converts a genlight object to STRUCTURE formatted files</h2><span id='topic+gl2structure'></span>

<h3>Description</h3>

<p>This function exports genlight objects to STRUCTURE formatted files (be aware
there is a gl2faststructure version as well). It is based on the code
provided by Lindsay Clark (see
<a href="https://github.com/lvclark/R_genetics_conv">https://github.com/lvclark/R_genetics_conv</a>) and this function is
basically a wrapper around her numeric2structure function. See also: Lindsay
Clark. (2017, August 22). lvclark/R_genetics_conv: R_genetics_conv 1.1
(Version v1.1). Zenodo: doi.org/10.5281/zenodo.846816.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2structure(
  x,
  indNames = NULL,
  addcolumns = NULL,
  ploidy = 2,
  exportMarkerNames = TRUE,
  outfile = "gl.str",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2structure_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data and location
data, lat longs [required].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_indnames">indNames</code></td>
<td>
<p>Specify individuals names to be added 
[if NULL, defaults to indNames(x)].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_addcolumns">addcolumns</code></td>
<td>
<p>Additional columns to be added before genotypes 
[default NULL].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_ploidy">ploidy</code></td>
<td>
<p>Set the ploidy [defaults 2].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_exportmarkernames">exportMarkerNames</code></td>
<td>
<p>If TRUE, locus names locNames(x) will be included 
[default TRUE].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension) 
[default &quot;gl.str&quot;].</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.' 
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2structure_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (wrapper) and Lindsay V. Clark [lvclark@illinois.edu]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#not run here
#gl2structure(testset.gl)
</code></pre>

<hr>
<h2 id='gl2svdquartets'>Converts a genlight object to nexus format PAUP SVDquartets</h2><span id='topic+gl2svdquartets'></span>

<h3>Description</h3>

<p>The output nexus file contains the SNP data in one of two forms, depending
upon what you regard as most appropriate. One form, that used by Chifman and
Kubatko, has two lines per individual, one providing the reference SNP the
second providing the alternate SNP (method=1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2svdquartets(
  x,
  outfile = "svd.nex",
  outpath = tempdir(),
  method = 2,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2svdquartets_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data or tag P/A data
[required].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including extension)
[default 'svd.nex'].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() when calling this
function or set.tempdir &lt;- getwd() elsewhere in your script to direct output
files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_method">method</code></td>
<td>
<p>Method = 1, nexus file with two lines per individual; method =
2, nexus file with one line per individual, ambiguity codes for SNP
genotypes, 0 or 1 for presence/absence data [default 2].</p>
</td></tr>
<tr><td><code id="gl2svdquartets_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity]</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A second form, recommended by Dave Swofford, has a single line per
individual, resolving heterozygous SNPs by replacing them with standard
ambiguity codes (method=2).
</p>
<p>If the data are tag presence/absence, then method=2 is assumed.
</p>
<p>Note that the genlight object must contain at least two populations for this
function to work.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Chifman, J. and L. Kubatko. 2014. Quartet inference from SNP data
under the coalescent. Bioinformatics 30: 3317-3324
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gg &lt;- testset.gl[1:20,1:100]
gg@other$loc.metrics &lt;- gg@other$loc.metrics[1:100,]
gl2svdquartets(gg)
</code></pre>

<hr>
<h2 id='gl2treemix'>Converts a genlight object to a treemix input file</h2><span id='topic+gl2treemix'></span>

<h3>Description</h3>

<p>The output file contains the SNP data in the format expected by treemix &ndash;
see the treemix manual. The file will be gzipped before in order to be
recognised by treemix. Plotting functions provided with treemix will need to
be sourced from the treemix download page.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2treemix(
  x,
  outfile = "treemix_input.gz",
  outpath = tempdir(),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2treemix_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file (including gz extension)
[default 'treemix_input.gz'].</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() when calling this
function or set.tempdir &lt;- getwd() elsewhere in your script to direct
output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2treemix_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Pickrell and Pritchard (2012). Inference of population splits and
mixtures from genome-wide allele frequency data. PLoS Genetics
https://doi.org/10.1371/journal.pgen.1002967
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gl2treemix(testset.gl, outpath=tempdir())
</code></pre>

<hr>
<h2 id='gl2vcf'>Converts a genlight object into vcf format</h2><span id='topic+gl2vcf'></span>

<h3>Description</h3>

<p>This function exports a genlight object into VCF format and save it into a
file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gl2vcf(
  x,
  plink_path = getwd(),
  outfile = "gl_vcf",
  outpath = tempdir(),
  snp_pos = "0",
  snp_chr = "0",
  chr_format = "character",
  pos_cM = "0",
  ID_dad = "0",
  ID_mom = "0",
  sex_code = "unknown",
  phen_value = "0",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gl2vcf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_plink_path">plink_path</code></td>
<td>
<p>Path of PLINK binary file [default getwd())].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_outfile">outfile</code></td>
<td>
<p>File name of the output file [default 'gl_vcf'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_outpath">outpath</code></td>
<td>
<p>Path where to save the output file
[default tempdir(), mandated by CRAN]. Use outpath=getwd() or outpath='.'
when calling this function to direct output files to your working directory.</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_snp_pos">snp_pos</code></td>
<td>
<p>Field name from the slot loc.metrics where the SNP position is
stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_snp_chr">snp_chr</code></td>
<td>
<p>Field name from the slot loc.metrics where the chromosome of
each is stored [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_chr_format">chr_format</code></td>
<td>
<p>Whether chromosome information is stored as 'numeric' or as
'character', see details [default 'character'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_pos_cm">pos_cM</code></td>
<td>
<p>A vector, with as many elements as there are loci, containing
the SNP position in morgans or centimorgans [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_id_dad">ID_dad</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the father, '0' if father isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_id_mom">ID_mom</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the ID of the mother, '0' if mother isn't in dataset [default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_sex_code">sex_code</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the sex code ('male', 'female', 'unknown') [default  'unknown'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_phen_value">phen_value</code></td>
<td>
<p>A vector, with as many elements as there are individuals,
containing the phenotype value. '1' = control, '2' = case, '0' = unknown
[default '0'].</p>
</td></tr>
<tr><td><code id="gl2vcf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function requires to download the binary file of PLINK 1.9 and provide
its path (plink_path).
The binary file can be downloaded from:
<a href="https://www.cog-genomics.org/plink/">https://www.cog-genomics.org/plink/</a>
</p>
<p>The chromosome information for unmapped SNPS is coded as 0.
Family ID is taken from  x$pop
Within-family ID (cannot be '0') is taken from indNames(x)
Variant identifier is taken from locNames(x)
</p>
<p>#' Note that if names of populations or individuals contain spaces, they are 
replaced by an underscore &quot;_&quot;.
</p>
<p>If you like to use chromosome information when converting to plink format and
your chromosome names are not from human, you need to change the chromosome 
names as 'contig1', 'contig2', etc. as described in the section &quot;Nonstandard
chromosome IDs&quot; in the following link:
https://www.cog-genomics.org/plink/1.9/input
</p>
<p>Note that the function might not work if there are spaces in the path to the
plink executable.
</p>


<h3>Value</h3>

<p>returns no value (i.e. NULL)
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M.
A., ... &amp; 1000 Genomes Project Analysis Group. (2011). The variant call
format and VCFtools. Bioinformatics, 27(15), 2156-2158.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("dartR.data")
gl2vcf(platypus.gl,snp_pos='ChromPos_Platypus_Chrom_NCBIv1',
 snp_chr = 'Chrom_Platypus_Chrom_NCBIv1')

## End(Not run)
</code></pre>

<hr>
<h2 id='interactive_reference'>Shiny app for the input of the reference table for the simulations</h2><span id='topic+interactive_reference'></span>

<h3>Description</h3>

<p>Shiny app for the input of the reference table for the simulations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interactive_reference()
</code></pre>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='interactive_sim_run'>Shiny app for the input of the simulations variables</h2><span id='topic+interactive_sim_run'></span>

<h3>Description</h3>

<p>Shiny app for the input of the simulations variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interactive_sim_run()
</code></pre>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='is.fixed'>Tests if two populations are fixed at a given locus</h2><span id='topic+is.fixed'></span>

<h3>Description</h3>

<p>This script compares two percent allele frequencies
and reports TRUE if they represent a fixed difference, FALSE otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.fixed(s1, s2, tloc = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.fixed_+3A_s1">s1</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the first 
population [required].</p>
</td></tr>
<tr><td><code id="is.fixed_+3A_s2">s2</code></td>
<td>
<p>Percentage SNP allele or sequence tag frequency for the second 
population [required].</p>
</td></tr>
<tr><td><code id="is.fixed_+3A_tloc">tloc</code></td>
<td>
<p>Threshold value for tolerance in when a difference is regarded as
fixed [default 0].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A fixed difference at a locus occurs when two populations share no alleles,
noting that SNPs are biallelic (ploidy=2).
Tolerance in the definition of a fixed difference is provided by the t
parameter. For example, t=0.05 means that SNP allele frequencies of 95,5 and
5,95 percent will be reported as fixed (TRUE).
</p>


<h3>Value</h3>

<p>TRUE (fixed difference) or FALSE (alleles shared) or NA (one or both 
s1 or s2 missing)
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.fixed.diff">gl.fixed.diff</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is.fixed(s1=100, s2=0, tloc=0)
is.fixed(96, 4, tloc=0.05)
</code></pre>

<hr>
<h2 id='platy'>Example data set as text file to be imported into a genlight object</h2><span id='topic+platy'></span>

<h3>Description</h3>

<p>Check ?read.genetable in pacakge PopGenReport for details on the format.
</p>


<h3>Format</h3>

<p>csv
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(PopGenReport)
read.csv( paste(.libPaths()[1],'/dartR/extdata/platy.csv',sep='' ))
platy &lt;- read.genetable( paste(.libPaths()[1],'/dartR/extdata/platy.csv',
sep='' ), ind=1, pop=2, lat=3, long=4, other.min=5, other.max=6, 
oneColPerAll=FALSE, sep='/')
platy.gl &lt;- gi2gl(platy, parallel=FALSE)
df.loc &lt;- data.frame(RepAvg = runif(nLoc(platy.gl)), CallRate = 1)
platy.gl@other$loc.metrics &lt;- df.loc
gl.report.reproducibility(platy.gl)

</code></pre>

<hr>
<h2 id='possums.gl'>A simulated genlight object created to run a landscape genetic example</h2><span id='topic+possums.gl'></span>

<h3>Description</h3>

<p>This a test data set to run a landscape genetics example. It contains 10 
populations of 30 individuals each and each individual has 300 loci. There 
are no covariates for individuals or loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>possums.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='rbind.dartR'>adjust rbind for dartR</h2><span id='topic+rbind.dartR'></span>

<h3>Description</h3>

<p>rbind is a bit lazy and does not take care for the metadata (so data in the
other slot is lost). You can get most of the loci metadata back using
gl.compliance.check.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dartR'
rbind(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind.dartR_+3A_...">...</code></td>
<td>
<p>list of dartR objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A genlight object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t1 &lt;- platypus.gl
class(t1) &lt;- "dartR"
t2 &lt;- rbind(t1[1:5,],t1[6:10,])
</code></pre>

<hr>
<h2 id='testset_metadata'>Metadata file. Can be integrated via the dart2genlight function.</h2><span id='topic+testset_metadata'></span>

<h3>Description</h3>

<p>Metadata file. Can be integrated via the dart2genlight function.
</p>


<h3>Format</h3>

<p>csv
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='testset_pop_recode'>Recode file to be used with the function.</h2><span id='topic+testset_pop_recode'></span>

<h3>Description</h3>

<p>This test data set is provided to show a typical recode file format.
</p>


<h3>Format</h3>

<p>csv
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='testset_SNPs_2Row'>Testfile in DArT format (as provided by DArT)</h2><span id='topic+testset_SNPs_2Row'></span>

<h3>Description</h3>

<p>This test data set is provided to show a typical DArT file format. Can be 
used to create a genlight object using the read.dart function.
</p>


<h3>Format</h3>

<p>csv
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='testset.gl'>A genlight object created via the gl.read.dart function</h2><span id='topic+testset.gl'></span>

<h3>Description</h3>

<p>This is a test data set on turtles. 250 individuals, 255 loci in &gt;30 
populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testset.gl
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='testset.gs'>A genlight object created via the gl.read.silicodart function</h2><span id='topic+testset.gs'></span>

<h3>Description</h3>

<p>This is a test data set on turtles. 218 individuals, 255 loci in &gt;30 
populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testset.gs
</code></pre>


<h3>Format</h3>

<p>genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (bugs? Post to 
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>

<hr>
<h2 id='theme_dartR'>dartR theme</h2><span id='topic+theme_dartR'></span>

<h3>Description</h3>

<p>This is the theme used as default for dartR plots.
This function controls all non-data display elements in the plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_dartR(
  base_size = 11,
  base_family = "",
  base_line_size = base_size/22,
  base_rect_size = base_size/22
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_dartR_+3A_base_size">base_size</code></td>
<td>
<p>base font size, given in pts.</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_family">base_family</code></td>
<td>
<p>base font family</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_line_size">base_line_size</code></td>
<td>
<p>base size for line elements</p>
</td></tr>
<tr><td><code id="theme_dartR_+3A_base_rect_size">base_rect_size</code></td>
<td>
<p>base size for rect elements</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#ggplot(data.frame(dummy=rnorm(1000)),aes(dummy)) +
#geom_histogram(binwidth=0.1) + theme_dartR()

</code></pre>

<hr>
<h2 id='utils.assignment'>Population assignment probabilities</h2><span id='topic+utils.assignment'></span>

<h3>Description</h3>

<p>This function takes one individual and estimates
their probability of coming from individual populations
from multilocus genotype frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.assignment(x, unknown, verbose = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.assignment_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_+3A_unknown">unknown</code></td>
<td>
<p>Name of the individual to be assigned to a population [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a re-implementation of the function multilocus_assignment
from package gstudio.
Description of the method used in this function can be found at:
https://dyerlab.github.io/applied_population_genetics/population-assignment.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> consisting of assignment probabilities for each
population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- utils.assignment(platypus.gl,unknown="T27")
</code></pre>

<hr>
<h2 id='utils.assignment_2'>Population assignment probabilities</h2><span id='topic+utils.assignment_2'></span>

<h3>Description</h3>

<p>This function takes one individual and estimates
their probability of coming from individual populations
from multilocus genotype frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.assignment_2(x, unknown, verbose = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.assignment_2_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_2_+3A_unknown">unknown</code></td>
<td>
<p>Name of the individual to be assigned to a population [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_2_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a re-implementation of the function multilocus_assignment
from package gstudio.
Description of the method used in this function can be found at:
https://dyerlab.github.io/applied_population_genetics/population-assignment.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> consisting of assignment probabilities for each
population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- utils.assignment_2(platypus.gl,unknown="T27")
</code></pre>

<hr>
<h2 id='utils.assignment_3'>Population assignment probabilities</h2><span id='topic+utils.assignment_3'></span>

<h3>Description</h3>

<p>This function takes one individual and estimates
their probability of coming from individual populations
from multilocus genotype frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.assignment_3(x, unknown, verbose = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.assignment_3_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_3_+3A_unknown">unknown</code></td>
<td>
<p>Name of the individual to be assigned to a population [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_3_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a re-implementation of the function multilocus_assignment
from package gstudio.
Description of the method used in this function can be found at:
https://dyerlab.github.io/applied_population_genetics/population-assignment.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> consisting of assignment probabilities for each
population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- utils.assignment_2(platypus.gl,unknown="T27")
</code></pre>

<hr>
<h2 id='utils.assignment_4'>Population assignment probabilities</h2><span id='topic+utils.assignment_4'></span>

<h3>Description</h3>

<p>This function takes one individual and estimates
their probability of coming from individual populations
from multilocus genotype frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.assignment_4(x, unknown, verbose = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.assignment_4_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_4_+3A_unknown">unknown</code></td>
<td>
<p>Name of the individual to be assigned to a population [required].</p>
</td></tr>
<tr><td><code id="utils.assignment_4_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a re-implementation of the function multilocus_assignment
from package gstudio.
Description of the method used in this function can be found at:
https://dyerlab.github.io/applied_population_genetics/population-assignment.html
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> consisting of assignment probabilities for each
population.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
res &lt;- utils.assignment_2(platypus.gl,unknown="T27")
</code></pre>

<hr>
<h2 id='utils.basic.stats'>Calculates mean observed heterozygosity, mean expected heterozygosity and Fis
per locus, per population and various population differentiation measures</h2><span id='topic+utils.basic.stats'></span>

<h3>Description</h3>

<p>This is a re-implementation of <code>hierfstat::basics.stats</code> specifically 
for genlight objects. Formula (and hence results) match exactly the original 
version of <code>hierfstat::basics.stats</code> but it is much faster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.basic.stats(x, digits = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.basic.stats_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.basic.stats_+3A_digits">digits</code></td>
<td>
<p>Number of decimals to report [default 4]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with with the statistics for each population
</p>


<h3>Author(s)</h3>

<p>Luis Mijangos and Carlo Pacioni (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
out &lt;- utils.basic.stats(platypus.gl)
</code></pre>

<hr>
<h2 id='utils.check.datatype'>Utility function to check the class of an object passed to a function</h2><span id='topic+utils.check.datatype'></span>

<h3>Description</h3>

<p>Most functions require access to a genlight object, dist matrix, data matrix
or fixed difference list (fd), and this function checks that a genlight
object or one of the above has been passed, whether the genlight object is a
SNP dataset or a SilicoDArT object, and reports back if verbosity is &gt;=2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.check.datatype(
  x,
  accept = c("genlight", "SNP", "SilicoDArT"),
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.check.datatype_+3A_x">x</code></td>
<td>
<p>Name of the genlight object, dist matrix, data matrix, glPCA, or
fixed difference list (fd) [required].</p>
</td></tr>
<tr><td><code id="utils.check.datatype_+3A_accept">accept</code></td>
<td>
<p>Vector containing the classes of objects that are to be
accepted [default c('genlight','SNP','SilicoDArT'].</p>
</td></tr>
<tr><td><code id="utils.check.datatype_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL, unless specified using gl.set.verbosity].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function checks the class of passed object and sets the datatype to
'SNP', 'SilicoDArT', 'dist', 'mat', or class[1](x) as appropriate.
</p>
<p>Note also that this function checks to see if there are individuals or loci
scored as all missing (NA) and if so, issues the user with a warning.
</p>
<p>Note: One and only one of gl.check, fd.check, dist.check or mat.check can be
TRUE.
</p>


<h3>Value</h3>

<p>datatype, 'SNP' for SNP data, 'SilicoDArT' for P/A data, 'dist' for a
distance matrix, 'mat' for a data matrix, 'glPCA' for an ordination file, or
class(x)[1].
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
datatype &lt;- utils.check.datatype(testset.gl)
datatype &lt;- utils.check.datatype(as.matrix(testset.gl),accept='matrix')
fd &lt;- gl.fixed.diff(testset.gl)
datatype &lt;- utils.check.datatype(fd,accept='fd')

datatype &lt;- utils.check.datatype(testset.gl)

</code></pre>

<hr>
<h2 id='utils.dart2genlight'>Converts DarT to genlight</h2><span id='topic+utils.dart2genlight'></span>

<h3>Description</h3>

<p>Converts a DArT file (read via <code>read.dart</code>) into an
genlight object <code>adegenet</code>. Internal function called by
gl.read.dart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dart2genlight(
  dart,
  ind.metafile = NULL,
  covfilename = NULL,
  probar = TRUE,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dart2genlight_+3A_dart">dart</code></td>
<td>
<p>A dart object created via read.dart [required].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_ind.metafile">ind.metafile</code></td>
<td>
<p>Optional file in csv format with metadata for each
individual (see details for explanation) [default NULL].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_covfilename">covfilename</code></td>
<td>
<p>Depreciated, use parameter ind.metafile.</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_probar">probar</code></td>
<td>
<p>Show progress bar [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.dart2genlight_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ind.metadata file needs to have very specific headings. First a heading
called id. Here the ids have to match the ids in the dart object
<code>colnames(dart[[4]])</code>. The following column headings are optional.
pop: specifies the population membership of each individual. lat and lon
specify spatial coordinates (in decimal degrees WGS1984 format). Additional
columns with individual metadata can be imported (e.g. age, gender).
</p>


<h3>Value</h3>

<p>A genlight object. Including all available slots are filled.
loc.names, ind.names, pop, lat, lon (if provided via the ind.metadata file)
</p>

<hr>
<h2 id='utils.dist.binary'>Calculates a distance matrix for individuals defined in a dartR
genlight object using binary P/A data (SilicoDArT)</h2><span id='topic+utils.dist.binary'></span>

<h3>Description</h3>

<p>This script calculates various distances between individuals based on sequence tag
Presence/Absence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dist.binary(
  x,
  method = "simple",
  scale = FALSE,
  swap = FALSE,
  output = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dist.binary_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default simple].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_swap">swap</code></td>
<td>
<p>If TRUE and working with presence-absence data, then presence 
(no disrupting mutation) is scored as 0 and absence (presence of a disrupting 
mutation) is scored as 1 [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_output">output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="utils.dist.binary_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure can be one of:
</p>

<ul>
<li><p> Euclidean &ndash; Euclidean Distance applied to cartesian coordinates defined
by the loci, scored as 0 or 1. Presence and absence equally weighted.
</p>
</li>
<li><p> simple &ndash; simple matching, both 1 or both 0 = 0; one 1 and the other
0 = 1. Presence and absence equally weighted.
</p>
</li>
<li><p> Jaccard &ndash; ignores matching 0, both 1 = 0; one 1 and the other 0 = 1.
Absences could be for different reasons.
</p>
</li>
<li><p> Bray-Curtis &ndash; both 0 = 0; both 1 = 2; one 1 and the other 0 = 1. Absences
could be for different reasons. Sometimes called the Dice or Sorensen
distance.
</p>
</li></ul>

<p>One might choose to disregard or downweight absences in comparison with
presences because the homology of absences is less clear (mutation at one or
the other, or both restriction sites). Your call.
</p>


<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author: Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
D &lt;- utils.dist.binary(testset.gs, method='Jaccard')
D &lt;- utils.dist.binary(testset.gs, method='Euclidean',scale=TRUE)

D &lt;- utils.dist.binary(testset.gs, method='Simple')

</code></pre>

<hr>
<h2 id='utils.dist.ind.snp'>Calculates a distance matrix for individuals defined in a dartR
genlight object using SNP data (DArTseq)</h2><span id='topic+utils.dist.ind.snp'></span>

<h3>Description</h3>

<p>This script calculates various distances between individuals based on 
SNP genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.dist.ind.snp(
  x,
  method = "Euclidean",
  scale = FALSE,
  output = "dist",
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.dist.ind.snp_+3A_x">x</code></td>
<td>
<p>Name of the genlight containing the genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_method">method</code></td>
<td>
<p>Specify distance measure [default Euclidean].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_scale">scale</code></td>
<td>
<p>If TRUE and method='Euclidean', the distance will be scaled to 
fall in the range [0,1] [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_output">output</code></td>
<td>
<p>Specify the format and class of the object to be returned, 
dist for a object of class dist, matrix for an object of class matrix [default &quot;dist&quot;].</p>
</td></tr>
<tr><td><code id="utils.dist.ind.snp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance measure can be one of:
</p>

<ul>
<li><p> Euclidean &ndash; Euclidean Distance applied to Cartesian coordinates defined
by the loci, scored as 0, 1 or 2. 
</p>
</li>
<li><p> Simple &ndash; simple mismatch, 0 where no alleles are shared, 1 where one
allele is shared, 2 where both alleles are shared. 
</p>
</li>
<li><p> Absolute &ndash; absolute mismatch, 0 where no alleles are shared, 1 where
one or both alleles are shared.
</p>
</li>
<li><p> Czekanowski (or Manhattan) calculates the city block metric distance
by summing the scores on each axis (locus).
</p>
</li></ul>



<h3>Value</h3>

<p>An object of class 'dist' or 'matrix' giving distances between individuals
</p>


<h3>Author(s)</h3>

<p>Author(s): Arthur Georges. Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
D &lt;- utils.dist.ind.snp(testset.gl, method='Manhattan')
D &lt;- utils.dist.ind.snp(testset.gl, method='Euclidean',scale=TRUE)

D &lt;- utils.dist.ind.snp(testset.gl, method='Simple')

</code></pre>

<hr>
<h2 id='utils.flag.start'>A utility script to flag the start of a script</h2><span id='topic+utils.flag.start'></span>

<h3>Description</h3>

<p>A utility script to flag the start of a script
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.flag.start(func = NULL, build = NULL, verbosity = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.flag.start_+3A_func">func</code></td>
<td>
<p>Name of the function that is starting [required].</p>
</td></tr>
<tr><td><code id="utils.flag.start_+3A_build">build</code></td>
<td>
<p>Name of the build [default NULL].</p>
</td></tr>
<tr><td><code id="utils.flag.start_+3A_verbosity">verbosity</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>calling function name
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
@export
</p>

<hr>
<h2 id='utils.hamming'>Calculates the Hamming distance between two DArT trimmed DNA sequences</h2><span id='topic+utils.hamming'></span>

<h3>Description</h3>

<p>Hamming distance is calculated as the number of base differences between two
sequences which can be expressed as a count or a proportion. Typically, it is
calculated between two sequences of equal length. In the context of DArT
trimmed sequences, which differ in length but which are anchored to the left
by the restriction enzyme recognition sequence, it is sensible to compare the
two trimmed sequences starting from immediately after the common recognition
sequence and terminating at the last base of the shorter sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.hamming(str1, str2, r = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.hamming_+3A_str1">str1</code></td>
<td>
<p>String containing the first sequence [required].</p>
</td></tr>
<tr><td><code id="utils.hamming_+3A_str2">str2</code></td>
<td>
<p>String containing the second sequence [required].</p>
</td></tr>
<tr><td><code id="utils.hamming_+3A_r">r</code></td>
<td>
<p>Number of bases in the restriction enzyme recognition sequence
[default 4].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Hamming distance between the rows of a matrix can be computed quickly
by exploiting the fact that the dot product of two binary vectors x and (1-y)
counts the corresponding elements that are different between x and y.
This matrix multiplication can also be used for matrices with more than two
possible values, and different types of elements, such as DNA sequences.
</p>
<p>The function calculates the Hamming distance between all columns of a
matrix X, or two matrices X and Y. Again matrix multiplication is used, this
time for counting, between two columns x and y, the number of cases in which
corresponding elements have the same value (e.g. A, C, G or T). This counting
is done for each of the possible values individually, while iteratively adding
the results. The end result of the iterative adding is the sum of all
corresponding elements that are the same, i.e. the inverse of the Hamming
distance. Therefore, the last step is to subtract this end result H from the
maximum possible distance, which is the number of rows of matrix X.
</p>
<p>If the two DNA sequences are of differing length, the longer is truncated. The
initial common restriction enzyme recognition sequence is ignored.
</p>
<p>The algorithm is that of Johann de Jong
<a href="https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/">https://johanndejong.wordpress.com/2015/10/02/faster-hamming-distance-in-r-2/</a>
</p>


<h3>Value</h3>

<p>Hamming distance between the two strings
</p>


<h3>Author(s)</h3>

<p>Custodian: Arthur Georges (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>

<hr>
<h2 id='utils.het.pop'>Calculates expected mean expected heterozygosity per population</h2><span id='topic+utils.het.pop'></span>

<h3>Description</h3>

<p>Calculates expected mean expected heterozygosity per population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.het.pop(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.het.pop_+3A_x">x</code></td>
<td>
<p>A genlight object containing the SNP genotypes [required].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with the mean expected heterozygosity for each population
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber &amp; Luis Mijangos (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>out &lt;- utils.het.pop(testset.gl)
</code></pre>

<hr>
<h2 id='utils.jackknife'>Conducts jackknife resampling using a genlight object</h2><span id='topic+utils.jackknife'></span>

<h3>Description</h3>

<p>Jackknife resampling is a statistical procedure where for a dataset of sample 
size n, subsamples of size n-1 are used to compute a statistic. The collection 
of the values obtained can be used to evaluate the variability around the point 
estimate. This function can take the loci, the individuals or the populations 
as units over which to conduct resampling.
</p>
<p>boldNote that when n is very small, jackknife resampling is not recommended.
</p>
<p>Parallel computation is implemented. The argument coden.cores indicates the 
number of core to use. If &quot;auto&quot; [default], it will use all but one available 
cores. If the number of units is small (e.g. a few populations), there is not 
real advantage in using parallel computation. On the other hand, if the number 
of units is large (e.g. thousands of loci), even with parallel computation, 
this function can be very slow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.jackknife(
  x,
  FUN,
  unit = "loc",
  recalc = FALSE,
  mono.rm = FALSE,
  n.cores = "auto",
  verbose = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.jackknife_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing SNP genotypes [required].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_fun">FUN</code></td>
<td>
<p>the name of the function to be used to calculate the statistic</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_unit">unit</code></td>
<td>
<p>The unit to use for resampling. One of c(&quot;loc&quot;, &quot;ind&quot;, &quot;pop&quot;): 
loci, individuals or populations</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_recalc">recalc</code></td>
<td>
<p>Recalculate the locus metadata statistics [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_mono.rm">mono.rm</code></td>
<td>
<p>Remove monomorphic loci [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use. If &quot;auto&quot; [default], it will 
use all but one available cores.</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default 2 or as specified using gl.set.verbosity].</p>
</td></tr>
<tr><td><code id="utils.jackknife_+3A_...">...</code></td>
<td>
<p>any additional arguments to be passed to FUN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length n where each element is the output of FUN
</p>


<h3>Author(s)</h3>

<p>Custodian: Carlo Pacioni &ndash; Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
platMod.gl &lt;- gl.filter.allna(platypus.gl) 
chk.pop &lt;- utils.jackknife(x=platMod.gl, FUN="gl.alf", unit="pop", 
recalc = FALSE, mono.rm = FALSE, n.cores = 1, verbose=0)
</code></pre>

<hr>
<h2 id='utils.n.var.invariant'>A utility script to calculate the number of variant and invariant sites by
locus</h2><span id='topic+utils.n.var.invariant'></span>

<h3>Description</h3>

<p>Calculate the number of variant and invariant sites by locus and add them as
columns in <code>loc.metrics</code>. This can be useful to conduct further
filtering, for example where only loci with secondaries are wanted for
phylogenetic analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.n.var.invariant(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.n.var.invariant_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.n.var.invariant_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report
[default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Invariant sites are the sites (nucleotide) that are not polymorphic. When the
locus metadata supplied by DArT includes the sequence of the allele
(<code>TrimmedSequence</code>), it is used by this function to estimate the number
of sites that were sequenced in each tag (read). This script then subtracts
the number of polymorphic sites. The length of the trimmed sequence
(lenTrimSeq), the number of variant (n.variant) and
invariant (n.invariant) sites are the added to the table in
<code>gl@others$loc.metrics</code>.
</p>
<p><strong>NOTE</strong>: It is important to realise that this function correctly
estimates the number of variant and invariant sites only when it is executed on
<code>genlight</code> objects before secondaries are removed.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Carlo Pacioni (Post to <a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gl.filter.secondaries">gl.filter.secondaries</a></code>,<code><a href="#topic+gl.report.heterozygosity">gl.report.heterozygosity</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require("dartR.data")
out &lt;- utils.n.var.invariant(platypus.gl)
</code></pre>

<hr>
<h2 id='utils.outflank'>OutFLANK:  An Fst outlier approach by Mike Whitlock and Katie Lotterhos,
University of British Columbia.</h2><span id='topic+utils.outflank'></span>

<h3>Description</h3>

<p>This function is the original implementation of Outflank by Whitlock and
Lotterhos. dartR simply provides a convenient wrapper around their functions
and an easier install being an r package (for information please refer to
their github repository)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank(
  FstDataFrame,
  LeftTrimFraction = 0.05,
  RightTrimFraction = 0.05,
  Hmin = 0.1,
  NumberOfSamples,
  qthreshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank_+3A_fstdataframe">FstDataFrame</code></td>
<td>
<p>A data frame that includes a row for each locus, with
columns as follows:
</p>

<ul>
<li><p> $LocusName: a character string that uniquely names
each locus.
</p>
</li>
<li><p> $FST: Fst calculated for this locus. (Kept here to
report the unbased Fst of the results)
</p>
</li>
<li><p> $T1: The numerator of the estimator for Fst
(necessary, with $T2, to calculate mean Fst)
</p>
</li>
<li><p> $T2: The denominator of the estimator of Fst
</p>
</li>
<li><p> $FSTNoCorr: Fst calculated for this locus without
sample size correction. (Used to find outliers)
</p>
</li>
<li><p> $T1NoCorr: The numerator of the estimator for Fst
without sample size correction (necessary, with $T2, to
calculate mean Fst)
</p>
</li>
<li><p> $T2NoCorr: The denominator of the estimator of Fst
without sample size correction
</p>
</li>
<li><p> $He: The heterozygosity of the locus (used to screen
out low heterozygosity loci that have a different distribution)
</p>
</li></ul>
</td></tr>
<tr><td><code id="utils.outflank_+3A_lefttrimfraction">LeftTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
lower end of the range of Fst before the likelihood funciton is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_righttrimfraction">RightTrimFraction</code></td>
<td>
<p>The proportion of loci that are trimmed from the
upper end of the range of Fst before the likelihood funciton is applied
[default 0.05].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including calculations
from a locus [default 0.1].</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_numberofsamples">NumberOfSamples</code></td>
<td>
<p>The number of spatial locations included in the data
set.</p>
</td></tr>
<tr><td><code id="utils.outflank_+3A_qthreshold">qthreshold</code></td>
<td>
<p>The desired false discovery rate threshold for calculating
q-values [default 0.05].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method looks for Fst outliers from a list of Fst's for different loci.
It assumes that each locus has been genotyped in all populations with
approximately equal coverage.
</p>
<p>OutFLANK estimates the distribution of Fst based on a trimmed sample of Fst's.
It assumes that the majority of loci in the center of the distribution are
neutral and infers the shape of the distribution of neutral Fst using a
trimmed set of loci. Loci with the highest and lowest Fst's are trimmed from
the data set before this inference, and the distribution of Fst df/(mean Fst)
is assumed to'follow a chi-square distribution. Based on this inferred
distribution, each locus is given a q-value based on its quantile in the
inferred null'distribution.
</p>
<p>The main procedure is called OutFLANK &ndash; see comments in that function
immediately below for input and output formats. The other functions here are
necessary and must be uploaded, but are not necessarily needed by the user
directly.
</p>
<p>Steps:
</p>


<h3>Value</h3>

<p>The function returns a list with seven elements:
</p>

<ul>
<li><p> FSTbar: the mean FST inferred from loci not marked as outliers
</p>
</li>
<li><p> FSTNoCorrbar: the mean FST (not corrected for sample size -gives an
upwardly biased estimate of FST)
</p>
</li>
<li><p> dfInferred: the inferred number of degrees of freedom for the
chi-square distribution of neutral FST
</p>
</li>
<li><p> numberLowFstOutliers: Number of loci flagged as having a significantly
low FST (not reliable)
</p>
</li>
<li><p> numberHighFstOutliers: Number of loci identified as having
significantly high FST
</p>
</li>
<li><p> results: a data frame with a row for each locus. This data frame
includes all the original columns in the
data set, and six new ones:
</p>

<ul>
<li><p> $indexOrder (the original order of the input data set),
</p>
</li>
<li><p> $GoodH (Boolean variable which is TRUE if the expected
heterozygosity is greater than the Hemin set by input),
</p>
</li>
<li><p> $OutlierFlag (TRUE if the method identifies the locus as
an outlier, FALSE otherwise), and
</p>
</li>
<li><p> $q (the q-value for the test of neutrality for the locus)
</p>
</li>
<li><p> $pvalues (the p-value for the test of neutrality for the
locus)
</p>
</li>
<li><p> $pvaluesRightTail the one-sided (right tail) p-value for
a locus
</p>
</li></ul>

</li></ul>



<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Whitlock &amp; Lotterhos
</p>

<hr>
<h2 id='utils.outflank.MakeDiploidFSTMat'>Creates OutFLANK input file from individual genotype info.</h2><span id='topic+utils.outflank.MakeDiploidFSTMat'></span>

<h3>Description</h3>

<p>Creates OutFLANK input file from individual genotype info.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank.MakeDiploidFSTMat(SNPmat, locusNames, popNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_snpmat">SNPmat</code></td>
<td>
<p>This is an array of genotypes with a row for each individual.
There should be a column for each SNP, with the number of copies of the focal
allele (0, 1, or 2) for that individual. If that individual is missing data
for that SNP, there should be a 9, instead.</p>
</td></tr>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_locusnames">locusNames</code></td>
<td>
<p>A list of names for each SNP locus. There should be the
same number of locus names as there are columns in SNPmat.</p>
</td></tr>
<tr><td><code id="utils.outflank.MakeDiploidFSTMat_+3A_popnames">popNames</code></td>
<td>
<p>A list of population names to give location for each
individual. Typically multiple individuals will have the same popName. The
list popNames should have the same length as the number of rows in SNPmat.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame in the form needed for the main OutFLANK
function.
</p>

<hr>
<h2 id='utils.outflank.plotter'>Plotting functions for Fst distributions after OutFLANK</h2><span id='topic+utils.outflank.plotter'></span>

<h3>Description</h3>

<p>This function takes the output of OutFLANK as
input with the OFoutput parameter.  It plots a histogram of the FST (by
default, the uncorrected FSTs used by OutFLANK) of loci and overlays the
inferred null histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.outflank.plotter(
  OFoutput,
  withOutliers = TRUE,
  NoCorr = TRUE,
  Hmin = 0.1,
  binwidth = 0.005,
  Zoom = FALSE,
  RightZoomFraction = 0.05,
  titletext = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.outflank.plotter_+3A_ofoutput">OFoutput</code></td>
<td>
<p>The output of the function OutFLANK()</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_withoutliers">withOutliers</code></td>
<td>
<p>Determines whether the loci marked as outliers (with
$OutlierFlag) are included in the histogram.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_nocorr">NoCorr</code></td>
<td>
<p>Plots the distribution of FSTNoCorr when TRUE. Recommended,
because this is the data used by OutFLANK to infer the distribution.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_hmin">Hmin</code></td>
<td>
<p>The minimum heterozygosity required before including  a locus in
the plot.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_binwidth">binwidth</code></td>
<td>
<p>The width of bins in the histogram.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_zoom">Zoom</code></td>
<td>
<p>If Zoom is set to TRUE, then the graph will zoom in on the right
tail of the distirbution (based on argument RightZoomFraction)</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_rightzoomfraction">RightZoomFraction</code></td>
<td>
<p>Used when Zoom = TRUE. Defines the proportion of the
distribution to plot.</p>
</td></tr>
<tr><td><code id="utils.outflank.plotter_+3A_titletext">titletext</code></td>
<td>
<p>Allows a test string to be printed as a title on the graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>produces a histogram of the FST
</p>

<hr>
<h2 id='utils.read.dart'>Imports DarT data to R</h2><span id='topic+utils.read.dart'></span>

<h3>Description</h3>

<p>Internal function called by gl.read.dart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.read.dart(
  filename,
  nas = "-",
  topskip = NULL,
  lastmetric = "RepAvg",
  service_row = 1,
  plate_row = 3,
  verbose = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.read.dart_+3A_filename">filename</code></td>
<td>
<p>Path to file (csv file only currently) [required].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_nas">nas</code></td>
<td>
<p>A character specifying NAs [default '-'].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_topskip">topskip</code></td>
<td>
<p>A number specifying the number of rows to be skipped. If not
provided the number of rows to be skipped are 'guessed' by the number of rows
with '*' at the beginning [default NULL].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_lastmetric">lastmetric</code></td>
<td>
<p>Specifies the last non genetic column [default 'RepAvg'].
Be sure to check if that is true, otherwise the number of individuals will
not match. You can also specify the last column by a number.</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_service_row">service_row</code></td>
<td>
<p>The row number in which the information of the DArT
service is contained [default 1].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_plate_row">plate_row</code></td>
<td>
<p>The row number in which the information of the plate
location is contained [default 3].</p>
</td></tr>
<tr><td><code id="utils.read.dart_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length 5. #dart format (one or two rows) #individuals,
#snps, #non genetic metrics, #genetic data (still two line format, rows=snps,
columns=individuals)
</p>

<hr>
<h2 id='utils.recalc.avgpic'>A utility script to recalculate the OneRatioRef, OneRatioSnp, PICRef, PICSnp,
and AvgPIC by locus after some individuals or populations have been deleted.</h2><span id='topic+utils.recalc.avgpic'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT has OneRatioRef, OneRatioSnp, PICRef,
PICSnp, and AvgPIC included, but the allelic composition will change when
some individuals,or populations, are removed from the dataset and so the
initial statistics will no longer apply. This script recalculates these
statistics and places the recalculated values in the appropriate place in
the genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.avgpic(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.avgpic_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.avgpic_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the locus metadata OneRatioRef|Snp, PICRef|Snp and/or AvgPIC do not exist,
the script creates and populates them.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#out &lt;- utils.recalc.avgpic(testset.gl)
</code></pre>

<hr>
<h2 id='utils.recalc.callrate'>A utility script to recalculate the callrate by locus after some populations
have been deleted</h2><span id='topic+utils.recalc.callrate'></span>

<h3>Description</h3>

<p>SNP datasets generated by DArT have missing values primarily arising from
failure to call a SNP because of a mutation at one or both of the
restriction enzyme recognition sites. The locus metadata supplied by DArT has
callrate included, but the call rate will change when some individuals are
removed from the dataset. This script recalculates the callrate and places
these recalculated values in the appropriate place in the genlight object.
It sets the Call Rate flag to TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.callrate(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.callrate_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.callrate_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log ; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.avgpic</code> for recalculating avgPIC,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#out &lt;- utils.recalc.callrate(testset.gl)
</code></pre>

<hr>
<h2 id='utils.recalc.freqhets'>A utility script to recalculate the frequency of the heterozygous SNPs by
locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhets'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT has FreqHets included, but the frequency
of the heterozygotes will change when some individuals are removed from the
dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhets(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhets_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhets_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script recalculates the FreqHets and places these recalculated values in
the appropriate place in the genlight object.
</p>
<p>Note that the frequency of the homozygote reference SNPS is calculated from
the individuals that could be scored.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate,
<code>utils.recalc.AvgPIC</code> for recalculating RepAvg, <code>gl.recalc.maf</code> for
recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#out &lt;- utils.recalc.freqhets(testset.gl)
</code></pre>

<hr>
<h2 id='utils.recalc.freqhomref'>A utility script to recalculate the frequency of the homozygous reference
SNP by locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhomref'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT has FreqHomRef included, but the
frequency of the homozygous reference will change when some individuals are
removed from the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhomref(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhomref_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhomref_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script recalculates the FreqHomRef and places these recalculated values
in the appropriate place in the genlight object.
</p>
<p>Note that the frequency of the homozygote reference SNPS is calculated from
the individuals that could be scored.
</p>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.avgpic</code> for recalculating AvgPIC,
<code>utils.recalc.freqhomsnp</code> for recalculating frequency of homozygous
alternate, <code>utils.recalc.freqhet</code> for recalculating frequency of
heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#result &lt;- utils.recalc.freqhomref(testset.gl)
</code></pre>

<hr>
<h2 id='utils.recalc.freqhomsnp'>A utility script to recalculate the frequency of the homozygous alternate
SNP by locus after some populations have been deleted</h2><span id='topic+utils.recalc.freqhomsnp'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT has FreqHomSnp included, but the
frequency of the homozygous alternate will change when some individuals are
removed from the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.freqhomsnp(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.freqhomsnp_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.freqhomsnp_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This script recalculates the FreqHomSnp and places these recalculated values
in the appropriate place in the genlight object.
</p>
<p>Note that the frequency of the homozygote alternate SNPS is calculated from
the individuals that could be scored.
</p>


<h3>Value</h3>

<p>The modified genlight object.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.avgpic</code> for recalculating AvgPIC,
<code>utils.recalc.freqhet</code> for recalculating frequency of heterozygotes,
<code>gl.recalc.maf</code> for recalculating minor allele frequency,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#out &lt;- utils.recalc.freqhomsnp(testset.gl)
</code></pre>

<hr>
<h2 id='utils.recalc.maf'>A utility script to recalculate the minor allele frequency by locus,
typically after some populations have been deleted</h2><span id='topic+utils.recalc.maf'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT does not have MAF included, so it is
calculated and added to the locus.metadata by this script. The minimum allele
frequency will change when some individuals are removed from the dataset.
This script recalculates the MAF and places these recalculated values in the
appropriate place in the genlight object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.recalc.maf(x, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.recalc.maf_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data [required].</p>
</td></tr>
<tr><td><code id="utils.recalc.maf_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default 2].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified genlight dataset.
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.avgpic</code> for recalculating AvgPIC,
<code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#f &lt;- dartR::utils.recalc.maf(testset.gl)
</code></pre>

<hr>
<h2 id='utils.reset.flags'>A utility script to reset to FALSE (or TRUE) the locus metric flags after
some individuals or populations have been deleted.</h2><span id='topic+utils.reset.flags'></span>

<h3>Description</h3>

<p>The locus metadata supplied by DArT has OneRatioRef, OneRatioSnp, PICRef,
PICSnp, and AvgPIC included, but the allelic composition will change when
some individuals are removed from the dataset and so the initial statistics
will no longer apply. This applies also to some variable calculated by dartR
(e.g. maf). This script resets the locus metrics flags to FALSE to indicate
that these statistics in the genlight object are no longer current. The
verbosity default is also set, and in the case of SilcoDArT, the flags PIC
and OneRatio are also set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.reset.flags(x, set = FALSE, value = 2, verbose = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.reset.flags_+3A_x">x</code></td>
<td>
<p>Name of the genlight object containing the SNP data or
tag presence/absence data (SilicoDArT) [required].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_set">set</code></td>
<td>
<p>Set the flags to TRUE or FALSE [default FALSE].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_value">value</code></td>
<td>
<p>Set the default verbosity for all functions, where verbosity is
not specified [default 2].</p>
</td></tr>
<tr><td><code id="utils.reset.flags_+3A_verbose">verbose</code></td>
<td>
<p>Verbosity: 0, silent or fatal errors; 1, begin and end; 2,
progress log; 3, progress and results summary; 5, full report [default NULL].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the locus metrics do not exist then they are added to the genlight object
but not populated. If the locus metrics flags do not exist, then they are
added to the genlight object and set to FALSE (or TRUE).
</p>


<h3>Value</h3>

<p>The modified genlight object
</p>


<h3>Author(s)</h3>

<p>Custodian: Luis Mijangos (Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>)
</p>


<h3>See Also</h3>

<p><code>utils.recalc.metrics</code> for recalculating all metrics,
<code>utils.recalc.callrate</code> for recalculating CallRate,
<code>utils.recalc.freqhomref</code> for recalculating frequency of homozygous
reference, <code>utils.recalc.freqhomsnp</code> for recalculating frequency of
homozygous alternate, <code>utils.recalc.freqhet</code> for recalculating frequency
of heterozygotes, <code>gl.recalc.maf</code> for recalculating minor allele 
frequency, <code>gl.recalc.rdepth</code> for recalculating average read depth
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#result &lt;- utils.reset.flags(testset.gl)
</code></pre>

<hr>
<h2 id='utils.spautocor'>Spatial autocorrelation coefficient calculations</h2><span id='topic+utils.spautocor'></span>

<h3>Description</h3>

<p>Carries out calculation for spatial autocorrelation coefficient
starting from a genetic and geogaphic distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.spautocor(
  GD,
  GGD,
  permutation = FALSE,
  bootstrap = FALSE,
  bins = 10,
  reps
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.spautocor_+3A_gd">GD</code></td>
<td>
<p>Genetic distance matrix.</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_ggd">GGD</code></td>
<td>
<p>Geographic distance matrix.</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_permutation">permutation</code></td>
<td>
<p>Whether permutation calculations for the null hypothesis 
of no spatial structure should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_bootstrap">bootstrap</code></td>
<td>
<p>Whether bootstrap calculations to compute the 95% 
confidence intervals around r should be carried out [default TRUE].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_bins">bins</code></td>
<td>
<p>The number of bins for the distance classes
(i.e. <code>length(bins) == 1)</code> or a vectors with the break points. See 
details [default 5].</p>
</td></tr>
<tr><td><code id="utils.spautocor_+3A_reps">reps</code></td>
<td>
<p>The number to be used for permutation and bootstrap analyses
[default 100].</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code of this function is based one <code>spautocorr</code> from the package
<code>PopGenReport</code>, which has been modified to fix a few bugs (as of
<code>PopGenReport v 3.0.4</code> and allow calculations of bootstraps estimates.
</p>
<p>See details from <code>gl.spatial.autoCorr</code> for a detailed explanation.
</p>


<h3>Value</h3>

<p>Returns a data frame with the following columns:
</p>

<ol>
<li><p> Bin  The distance classes
</p>
</li>
<li><p> N The number of pairwise comparisons within each distance class
</p>
</li>
<li><p> r.uc The uncorrected autocorrelation coefficient
</p>
</li></ol>

<p>if both <code>bootstap</code> and <code>permutation</code> are <code>FALSE</code> otherwise only
<code>r</code> estimates are returned
</p>


<h3>Author(s)</h3>

<p>Carlo Pacioni &amp; Bernd Gruber
</p>


<h3>References</h3>


<ul>
<li><p> Smouse PE, Peakall R. 1999. Spatial autocorrelation analysis of
individual multiallele and multilocus genetic structure. Heredity 82:
561-573.
</p>
</li>
<li><p> Double, MC, et al. 2005. Dispersal, philopatry and infidelity: dissecting
local genetic structure in superb fairy-wrens (Malurus cyaneus). Evolution
59, 625-635.
</p>
</li>
<li><p> Peakall, R, et al. 2003. Spatial autocorrelation analysis offers new
insights into gene flow in the Australian bush rat, Rattus fuscipes.
Evolution 57, 1182-1195.
</p>
</li>
<li><p> Smouse, PE, et al. 2008. A heterogeneity test for fine-scale genetic
structure. Molecular Ecology 17, 3389-3400.
</p>
</li>
<li><p> Gonzales, E, et al. 2010. The impact of landscape disturbance on spatial
genetic structure in the Guanacaste tree, Enterolobium
cyclocarpum(Fabaceae). Journal of Heredity 101, 133-143.
</p>
</li>
<li><p> Beck, N, et al. 2008. Social constraint and an absence of sex-biased
dispersal drive fine-scale genetic structure in white-winged choughs.
Molecular Ecology 17, 4346-4358.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+gl.spatial.autoCorr">gl.spatial.autoCorr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See gl.spatial.autoCorr
</code></pre>

<hr>
<h2 id='utils.structure.evanno'>Util function for evanno plots</h2><span id='topic+utils.structure.evanno'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN 
(maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.evanno(sr, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.evanno_+3A_sr">sr</code></td>
<td>
<p>structure run object</p>
</td></tr>
<tr><td><code id="utils.structure.evanno_+3A_plot">plot</code></td>
<td>
<p>should the plots be returned</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='utils.structure.genind2gtypes'>structure util functions</h2><span id='topic+utils.structure.genind2gtypes'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN (maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.genind2gtypes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.genind2gtypes_+3A_x">x</code></td>
<td>
<p>a genind object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gtypes object
</p>


<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='utils.structure.run'>Utility function to run Structure</h2><span id='topic+utils.structure.run'></span>

<h3>Description</h3>

<p>These functions were copied from package strataG, which is no longer on CRAN
(maintained by Eric Archer)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utils.structure.run(
  g,
  k.range = NULL,
  num.k.rep = 1,
  label = NULL,
  delete.files = TRUE,
  exec = "structure",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utils.structure.run_+3A_g">g</code></td>
<td>
<p>a gtypes object [see <code>strataG</code>].</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_k.range">k.range</code></td>
<td>
<p>vector of values to for <code>maxpop</code> in multiple runs. If set
to <code>NULL</code>, a single STRUCTURE run is conducted with <code>maxpops</code>
groups. If specified, do not also specify <code>maxpops</code>.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_num.k.rep">num.k.rep</code></td>
<td>
<p>number of replicates for each value in <code>k.range</code>.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_label">label</code></td>
<td>
<p>label to use for input and output files</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_delete.files">delete.files</code></td>
<td>
<p>logical. Delete all files when STRUCTURE is finished?</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_exec">exec</code></td>
<td>
<p>name of executable for STRUCTURE. Defaults to &quot;structure&quot;.</p>
</td></tr>
<tr><td><code id="utils.structure.run_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <code>structureWrite</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

 <dl>
<dt><code>structureRun</code></dt><dd><p>a list where each element is a
list with results from <code>structureRead</code> and a vector of the filenames
used</p>
</dd> <dt><code>structureWrite</code></dt><dd><p>a vector of the filenames used by
STRUCTURE</p>
</dd> <dt><code>structureRead</code></dt><dd><p>a list containing: </p>

<dl>
<dt><code>summary</code></dt><dd><p>new locus name, which is a combination of loci in
group</p>
</dd> <dt><code>q.mat</code></dt><dd><p>data.frame of assignment probabilities for each
id</p>
</dd> <dt><code>prior.anc</code></dt><dd><p>list of prior ancestry estimates for each
individual where population priors were used</p>
</dd> <dt><code>files</code></dt><dd><p>vector of
input and output files used by STRUCTURE</p>
</dd> <dt><code>label</code></dt><dd><p>label for the
run</p>
</dd> </dl>
 </dd> </dl>



<h3>Author(s)</h3>

<p>Bernd Gruber (bugs? Post to
<a href="https://groups.google.com/d/forum/dartr">https://groups.google.com/d/forum/dartr</a>); original implementation of
Eric Archer <a href="https://github.com/EricArcher/strataG">https://github.com/EricArcher/strataG</a>
</p>

<hr>
<h2 id='zzz'>Setting up the package</h2><span id='topic+zzz'></span>

<h3>Description</h3>

<p>Setting theme, colors and verbosity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zzz
</code></pre>


<h3>Format</h3>

<p>An object of class <code>NULL</code> of length 0.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
