<!DOCTYPE html><html lang="en-US"><head><title>Help for package datana</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {datana}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#datana-package'>
<p>Datasets and Functions to Accompany Analisis De Datos Con R</p></a></li>
<li><a href='#aboutrsq'><p>About the R-Squared statistics: the Anscombe quartet dataset</p></a></li>
<li><a href='#aboutrsq2'><p>Sobre el estadístico R2: los datos del cuarteto de Anscombe</p></a></li>
<li><a href='#airnyc'><p>Airquality data in New York city.</p></a></li>
<li><a href='#airnyc2'><p>Calidad del aire en la ciudad de Nueva York.</p></a></li>
<li><a href='#annualppCities'><p>Time series of annual precipitations in cities of Chile.</p></a></li>
<li><a href='#annualppCities2'><p>Serie de tiempo de precipitaciones anuales en Chile.</p></a></li>
<li><a href='#araucaria'><p>Contains plot-level variables in Araucaria araucana forests in southern Chile.</p></a></li>
<li><a href='#araucaria2'><p>Variables a nivel de parcela para bosques de Araucaria araucana el sur de Chile.</p></a></li>
<li><a href='#baiTreelines'><p>Annual basal area increment for four tree species.</p></a></li>
<li><a href='#baiTreelines2'><p>Incremento anual en area basal de cuatro especies arboreas.</p></a></li>
<li><a href='#bears'><p>Age and physical measurement data for wild bears</p></a></li>
<li><a href='#bears2'><p>Edad y características biométricas de osos salvajes</p></a></li>
<li><a href='#bearsDepu'><p>Age and physical measurement data for wild bears (without missing values)</p></a></li>
<li><a href='#bearsDepu2'><p>Edad y características biométricas de osos salvajes (sin datos faltantes)</p></a></li>
<li><a href='#beetles'><p>Population density growth of beetles</p></a></li>
<li><a href='#beetles2'><p>Crecimiento poblacional de escarabajos</p></a></li>
<li><a href='#biomass'><p>Contains tree-level biomass data for several species in Canada.</p></a></li>
<li><a href='#biomass2'><p>Biomasa a nivel de árbol para especies arboreas de Canada.</p></a></li>
<li><a href='#carbohydrateTreelines'><p>Carbohydrates concentrations of tree species.</p></a></li>
<li><a href='#cdf'><p>Computes the cumulative distribution for a given random variable</p></a></li>
<li><a href='#chicksw'><p>Chicken growth data.</p></a></li>
<li><a href='#contrast'><p>Computes statistics for inference in a given contrast</p></a></li>
<li><a href='#corkoak'><p>Tree-level cork biomass data for Oak trees in Portugal</p></a></li>
<li><a href='#corkoak2'><p>Datos de biomasa de corcho en árboles de Encino en Portugal</p></a></li>
<li><a href='#crown'><p>Tree crown radii</p></a></li>
<li><a href='#crown2'><p>Radios de copa de árboles</p></a></li>
<li><a href='#deadForestCA'><p>Data contains  climatic, forest structure and forest mortality variable</p></a></li>
<li><a href='#deadForestCA2'><p>Los datos contienen variables climaticas, de estructura forestal y de mortalidad forestal.</p></a></li>
<li><a href='#deadLianas'><p>This dataset has 43 columns and 4247 rows. Each row corresponds to an epiphyte individual located on the</p>
reliable sections of the host trees</a></li>
<li><a href='#deadLianas2'><p>Este conjunto de datos tiene 43 columnas y 4247 filas. Cada fila corresponde a un individuo epifito ubicado en el</p>
secciones confiables de los árboles hospedantes</a></li>
<li><a href='#deleteRight'><p>Deletes the last n-characters of a string</p></a></li>
<li><a href='#demograph'><p>Contains information of demography of species.</p></a></li>
<li><a href='#descstat'><p>Creates a descriptive statistics table for continuous variables</p></a></li>
<li><a href='#election'><p>Presidential election data of Florida (USA) in 2000.</p></a></li>
<li><a href='#election2'><p>Elección presidencial en el estado de Florida (USA) en el 2000.</p></a></li>
<li><a href='#eucaleaf'><p>Leaf measurements for Eucalyptus nitens trees in Tasmania, Australia.</p></a></li>
<li><a href='#eucaleaf2'><p>Mediciones foliares para árboles de Eucalyptus nitens en Tasmania, Australia.</p></a></li>
<li><a href='#eucaleafAll'><p>Leaf measurements (all, n=744) for Eucalyptus nitens trees in Tasmania, Australia.</p></a></li>
<li><a href='#eucaleafAll2'><p>Mediciones foliares (todas, n=744) para árboles de Eucalyptus nitens en Tasmania, Australia.</p></a></li>
<li><a href='#eucaplot'><p>Data from a Eucalyptus globulus plantation near Gorbea, Region de La Araucania, Chile.</p></a></li>
<li><a href='#eucaplot2'><p>Árboles dentro de parcelas de muestreo en una plantación de Eucalyptus globulus, Chile.</p></a></li>
<li><a href='#fertiliza'><p>Fertilization experiment data.</p></a></li>
<li><a href='#fertiliza2'><p>Datos a nivel de parcela de un experimento de fertilización</p></a></li>
<li><a href='#ficdiamgr'><p>Diameter growth of trees</p></a></li>
<li><a href='#ficdiamgr2'><p>Crecimiento diametral de árboles</p></a></li>
<li><a href='#findColumn.byname'><p>Finds the position of a specific variable.</p></a></li>
<li><a href='#fishgrowth'><p>Data on fish growth.</p></a></li>
<li><a href='#fishgrowth2'><p>Crecimiento de peces</p></a></li>
<li><a href='#floraChile'><p>Flora of Chile.</p></a></li>
<li><a href='#floraChile2'><p>Flora de Chile.</p></a></li>
<li><a href='#football'><p>Anaerobic potential of soccer players.</p></a></li>
<li><a href='#football2'><p>Potencia anaerobica de jugadores de football.</p></a></li>
<li><a href='#forestFire'><p>Data of forest fire occurrence</p></a></li>
<li><a href='#forestFire2'><p>Datos de ocurrencia de incendios forestales</p></a></li>
<li><a href='#forestHawaii'><p>Contains information of forest plots across the Hawaiian archipelago.</p></a></li>
<li><a href='#gmean'><p>Function to compute the geometric mean of a vector</p></a></li>
<li><a href='#hawaii'><p>Diameter growth increments of a tropical tree species in Hawaii</p></a></li>
<li><a href='#hawaii2'><p>Incremento corriente anual en diámetro de una especie tropical en Hawaii</p></a></li>
<li><a href='#hgrdfir'><p>Tree height growth of Douglas-fir sample trees in the Northwest of the United States</p></a></li>
<li><a href='#hgrdfir2'><p>Crecimiento en altura de una muestra de árboles en los Estados Unidos</p></a></li>
<li><a href='#idahohd'><p>Tree height-diameter data from Idaho (USA)</p></a></li>
<li><a href='#idahohd2'><p>Altura-diámetro de árboles en el estado de Idaho (USA)</p></a></li>
<li><a href='#invasivesRCI'><p>Contains regeneration microsite data in Robinson Crusoe Island forest</p></a></li>
<li><a href='#kurto'><p>Computes the sample kurtosis of a distribution</p></a></li>
<li><a href='#landCoverSantiago'><p>Land-cover, environmental and sociodemographic data for the 34 municipalities composing the Greater Santiago area, Santiago, Chile.</p></a></li>
<li><a href='#landCoverSantiago2'><p>Cobertura territorial, ambiental y sociodemografica de los 34 municipios que componen el area del Gran Santiago, Santiago, Chile..</p></a></li>
<li><a href='#lleuque'><p>Contains species composition data of Prumnopitys andina (Lleuque) forests</p></a></li>
<li><a href='#lrt'><p>Computes a likelihood ratio test between a reduced model and a full model</p></a></li>
<li><a href='#moda'><p>Computes the mode</p></a></li>
<li><a href='#pinaster'><p>Tree volume for Pinus pinaster in the Baixo-Mino, Galicia, Spain.</p></a></li>
<li><a href='#pinaster2'><p>Volumen individual de árboles de Pinus pinaster en Galicia, España.</p></a></li>
<li><a href='#pinusContorta'><p>Contains spatial location of Pinus contorta trees in sample plots.</p></a></li>
<li><a href='#pinusContorta2'><p>Ubicación espacial de árboles de Pinus contorta en parcela de muestreo</p></a></li>
<li><a href='#pinusSpp'><p>Tree-level variables of several sample plots of invasive Pinus spp in Chile.</p></a></li>
<li><a href='#pinusSpp2'><p>Variables a nivel de árbol en parcelas de muestreo de Pinus spp en Chile.</p></a></li>
<li><a href='#plantsHawaii'><p>Maximum plant size in the Hawaiian archipelago.</p></a></li>
<li><a href='#presenceIce'><p>Presence or absence of sea ice from logbook records of annual cruises</p></a></li>
<li><a href='#presidentChile'><p>2021 presidential election in Chile.</p></a></li>
<li><a href='#presidentChile2'><p>Eleccion presidencial del 2021 en Chile.</p></a></li>
<li><a href='#primary'><p>2021 primary election for the president of Chile</p></a></li>
<li><a href='#primary2'><p>Elección primaria para la presidencia de Chile</p></a></li>
<li><a href='#pspLlancahue'><p>Tree locations for a sample plot in the Llancahue experimental forest</p></a></li>
<li><a href='#pspLlancahue2'><p>Ubicación cartesiana de árboles en el bosque de Llancahue</p></a></li>
<li><a href='#pspRuca'><p>Tree spatial coordinates in the Rucamanque forest</p></a></li>
<li><a href='#pspRuca2'><p>Ubicación espacial de árboles en el bosque de Rucamanque</p></a></li>
<li><a href='#ptaeda'><p>Height growth of Pinus taeda (Loblolly pine) trees</p></a></li>
<li><a href='#ptaeda2'><p>Crecimiento en altura de Pinus taeda</p></a></li>
<li><a href='#pvalt'><p>Obtain the P-value for a Standard t-distributed random variable</p></a></li>
<li><a href='#pvalz'><p>Obtain the P-value for a Standard Gaussian random variable</p></a></li>
<li><a href='#radiatapl'><p>Sampling plots data from a Pinus radiata plantation near Capitan Pastene, Region de La Araucania, Chile.</p></a></li>
<li><a href='#radiatapl2'><p>Datos a nivel de árbol de parcelas de muestreo en plantaciones de Pinus radiata</p></a></li>
<li><a href='#raulihg'><p>Height growth of Nothofagus alpina trees in Chile.</p></a></li>
<li><a href='#raulihg2'><p>Crecimiento en altura de árboles de Nothofagus alpina.</p></a></li>
<li><a href='#regNothofagus'><p>Contains information about regeneration of Nothofagus seedlings.</p></a></li>
<li><a href='#simula'><p>Simulated yield of forestry plantations of exotic species in Chile.</p></a></li>
<li><a href='#skew'><p>Computes the skewness of a numeric vector</p></a></li>
<li><a href='#slashpine'><p>Biomass dataset</p></a></li>
<li><a href='#slashpine2'><p>Biomasa</p></a></li>
<li><a href='#sludge'><p>Sludge data are at different cities, with a value of concentration zinc.</p></a></li>
<li><a href='#snaspeChile'><p>On the National System of State Protected Wild Areas (SNASPE) of Chile.</p></a></li>
<li><a href='#snaspeChile2'><p>Sistema nacional de areas protegidas del estado (SNASPE) de Chile</p></a></li>
<li><a href='#soiltreat'><p>Soil treatment experiment in tree seedlings</p></a></li>
<li><a href='#soiltreat2'><p>Tratamientos del suelo en el crecimiento de plantulas.</p></a></li>
<li><a href='#spatAustria'><p>Tree locations for several plots of Norway spruce in Austria</p></a></li>
<li><a href='#speciesList'><p>Names and other information of plant species (mainly trees)</p></a></li>
<li><a href='#sppAbundance'><p>Contains information of abundance of plant species in the central-southern Andes of Chile.</p></a></li>
<li><a href='#sppTraits'><p>Contains information of functional traits of species.</p></a></li>
<li><a href='#standLleuque'><p>Plot-level data with variables from Andean Prumnopitys forests</p></a></li>
<li><a href='#standLleuque2'><p>Datos con variables a nivel de parcela de bosques de Prumnopitys andina</p></a></li>
<li><a href='#timeserplot'><p>Produces a time series plot</p></a></li>
<li><a href='#trailCameraTrap'><p>Contains information of Camera trap data on medium to large terrestrial mammals</p>
collected at 54 camera stations in Ruaha National Park, southern Tanzania.</a></li>
<li><a href='#traits'><p>Functional traits of vegetative species in Chile.</p></a></li>
<li><a href='#traits2'><p>Rasgos funcionales para algunas especies vegetales de Chile.</p></a></li>
<li><a href='#treegr'><p>Diameter and height growth of Grand-fir (Abies grandis) sample trees</p></a></li>
<li><a href='#treegr2'><p>Crecimiento en diámetro y altura de árboles muestras de Grand-fir (Abies grandis)</p></a></li>
<li><a href='#treelistinve'><p>Tree-list data in a forest inventory.</p></a></li>
<li><a href='#treelistinve2'><p>Lista de árboles en un inventario forestal.</p></a></li>
<li><a href='#treevol'><p>Diameter, height and volume for Black Cherry Trees</p></a></li>
<li><a href='#treevol2'><p>Volumen, altura, y diámetro para árboles de Black Cherry</p></a></li>
<li><a href='#treevolroble'><p>Tree volume of roble (Nothofagus obliqua) in the Rucamanque forest</p></a></li>
<li><a href='#treevolroble2'><p>Volumen a nivel de árbol para roble (Nothofagus obliqua) especie en el bosque de Rucamanque</p></a></li>
<li><a href='#treevolruca'><p>Tree volume by species in the Rucamanque forest</p></a></li>
<li><a href='#treevolruca2'><p>Volumen a nivel de árbol en el bosque de Rucamanque</p></a></li>
<li><a href='#xyboxplot'><p>Creates a scatterplot with superposing boxplots</p></a></li>
<li><a href='#xyhist'><p>A scatterplot with marginal histograms</p></a></li>
<li><a href='#xymultiplot'><p>Figure of a matrix of scatterplots and histograms for several variables.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-02</td>
</tr>
<tr>
<td>Title:</td>
<td>Datasets and Functions to Accompany Analisis De Datos Con R</td>
</tr>
<tr>
<td>Description:</td>
<td>Datasets and functions to accompany the book 'Analisis de datos con el programa
    estadistico R: una introduccion aplicada' by Salas-Eljatib (2021, ISBN: 9789566086109).     
    The package helps carry out data management, exploratory analyses, and model fitting.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://eljatib.com/rlibro">https://eljatib.com/rlibro</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, stats, graphics, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>foreign, gdata, car, agricolae, multcomp</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>BuildResaveData:</td>
<td>best</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-07 15:41:28 UTC; christian</td>
</tr>
<tr>
<td>Author:</td>
<td>Christian Salas-Eljatib
    <a href="https://orcid.org/0000-0002-8468-0829"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Pino Nicolas [ctb] (up to 2020),
  Riquelme Joaquin [ctb] (up to 2020)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christian Salas-Eljatib &lt;cseljatib@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-07 16:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='datana-package'>
Datasets and Functions to Accompany Analisis De Datos Con R
</h2><span id='topic+datana-package'></span><span id='topic+datana'></span>

<h3>Description</h3>

<p>The datana package provides the datasets and functions that accompany the book &quot;Análisis de datos con el programa estadístico R: una introducción aplicada&quot; by Salas-Eljatib (2021, ISBN: 9789566086109). You can visit the book's website at <a href="https://eljatib.com/rlibro">https://eljatib.com/rlibro</a>.
</p>
<p>Notice that every dataframe has a counterpart but has column names in Spanish. For instance, the dataframe 'crown' has column names in English, but 'crown2' has column names in Spanish. Both data frames have the same data.
</p>


<h3>Details</h3>

<p>The package contains several datasets for exploratory data analysis in an array of disciplines. Furthermore, datana provides functions as tools for descriptive statistics and plotting. 
</p>
<p>To see the preferable citation of the package, type citation(&quot;datana&quot;).
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib [aut, cre]
    (&lt;https://orcid.org/0000-0002-8468-0829&gt;),
  Pino Nicolas [ctb] (up to 2020),
  Riquelme Joaquin [ctb] (up to 2020)
</p>
<p>Maintainer: Christian Salas-Eljatib &lt;cseljatib@gmail.com&gt;
</p>
<p>Christian Salas-Eljatib is also indebted to several people who have contributed to individual data frames and functions: see credits in help pages.
</p>


<h3>References</h3>

<p>Salas-Eljatib C. 2021. Análisis de datos con el programa estadístico R: una introducción aplicada. Santiago, Chile: Ediciones Universidad Mayor. ISBN: 9789566086109. <a href="https://www.buscalibre.cl/libro-analisis-de-datos-con-el-programa-estadistico-r/9789566086109/p/53775485">https://www.buscalibre.cl/libro-analisis-de-datos-con-el-programa-estadistico-r/9789566086109/p/53775485</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##scatter-plot and marginal histograms
data(treevolroble)
df &lt;- treevolroble
xyhist(x=df$dbh,y=df$toth, xlab="Variable X",  ylab="Variable Y")

##scatter-plot and box-plots 
data(fishgrowth)
df &lt;- fishgrowth
xyboxplot(x=df$length,y=df$scale)
</code></pre>

<hr>
<h2 id='aboutrsq'>About the R-Squared statistics: the Anscombe quartet dataset</h2><span id='topic+aboutrsq'></span>

<h3>Description</h3>

<p>A dataset that contains four pairs of columns with the same descriptive
statistics; however, there is a difference when representing the points through a graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(aboutrsq)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>X1</dt><dd><p>Integers values that represent X-axis for Y1, Y2 and Y3 column</p>
</dd>											
<dt>Y1</dt><dd><p>Float values that represent Y-axis for X1 column</p>
</dd>
<dt>Y2</dt><dd><p>Float values that represent Y-axis for X1 column</p>
</dd>
<dt>Y3</dt><dd><p>Float values that represent Y-axis for X1 column</p>
</dd>
<dt>X2</dt><dd><p>Integers values that represent X-axis for Y4 column</p>
</dd>
<dt>Y4</dt><dd><p>Float values that represent Y-axis for X2 column</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were assembled by Dr Christian Salas-Eljatib (Santiago, Chile).
</p>


<h3>References</h3>

<p>Anscombe, Francis J. (1973). Graphs in statistical analysis. The American Statistician, 27, 17-21. <a href="https://doi.org/10.2307/2682899">doi:10.2307/2682899</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(aboutrsq)    
head(aboutrsq) 
</code></pre>

<hr>
<h2 id='aboutrsq2'>Sobre el estadístico R2: los datos del cuarteto de Anscombe</h2><span id='topic+aboutrsq2'></span>

<h3>Description</h3>

<p>Dataset que contiene cuatro pares de columnas con la mismos estadísticos descriptivos,
sin embargo, si existe diferencia al representar los puntos mediante un gráfico.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(aboutrsq2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuación::
</p>

<dl>
<dt>X1</dt><dd><p>Valores enteros que representan el eje X para las columnas Y1, Y2 e Y3</p>
</dd>								
<dt>Y1</dt><dd><p>Valores flotantes que representan el eje Y para la columna X1</p>
</dd>
<dt>Y2</dt><dd><p>Valores flotantes que representan el eje Y para la columna X1</p>
</dd>
<dt>Y3</dt><dd><p>Valores flotantes que representan el eje Y para la columna X1</p>
</dd>
<dt>X2</dt><dd><p>Valores enteros que representan el eje X para las columnas Y4</p>
</dd>
<dt>Y4</dt><dd><p>Valores flotantes que representan el eje Y para la columna X2</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos fueron contribuidos por el Prof. Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>Anscombe FJ. 1973. Graphs in statistical analysis. The American Statistician, 27, 17-21. <a href="https://doi.org/10.2307/2682899">doi:10.2307/2682899</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(aboutrsq2)    
head(aboutrsq2) 
</code></pre>

<hr>
<h2 id='airnyc'>Airquality data in New York city.</h2><span id='topic+airnyc'></span>

<h3>Description</h3>

<p>Daily air quality measurements in New York, May to September 1973.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(airnyc)
</code></pre>


<h3>Format</h3>

<p>Contains 6 variables, as follows:
</p>

<dl>
<dt>ozone</dt><dd><p>numeric	Ozone (ppb).</p>
</dd>
<dt>solar</dt><dd><p>numeric	Solar R (lang).</p>
</dd>
<dt>wind</dt><dd><p>numeric	Wind (mph).</p>
</dd>
<dt>temp</dt><dd><p>numeric	Temperature (degrees F).</p>
</dd>
<dt>month</dt><dd><p>numeric	Month (1&ndash;12).</p>
</dd>
<dt>day</dt><dd><p>numeric	Day of month (1&ndash;31).</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the library 'datasets'.
</p>


<h3>References</h3>

<p>Chambers J, Cleveland W, Kleiner B, Tukey P. 1983. Graphical Methods for Data Analysis. Belmont. CA: Wadsworth.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(airnyc)    
head(airnyc) 
</code></pre>

<hr>
<h2 id='airnyc2'>Calidad del aire en la ciudad de Nueva York.</h2><span id='topic+airnyc2'></span>

<h3>Description</h3>

<p>Calidad del aire diario medido en New York, de Mayo a Septiembre de 1973.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(airnyc2)
</code></pre>


<h3>Format</h3>

<p>Contiene 6 variables:
</p>

<dl>
<dt>ozone</dt><dd><p>Ozono (ppb).</p>
</dd>
<dt>solar</dt><dd><p>Solar R (largo).</p>
</dd>
<dt>wind</dt><dd><p>Viento (mph).</p>
</dd>
<dt>temp</dt><dd><p>Temperatura (grados F).</p>
</dd>
<dt>month</dt><dd><p>Mes del año (1&ndash;12).</p>
</dd>
<dt>day</dt><dd><p>Dia del mes (1&ndash;31).</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron obtenidos desde la librería 'datasets'.
</p>


<h3>References</h3>

<p>Chambers J, Cleveland W, Kleiner B, Tukey P. 1983. Graphical Methods for Data Analysis. Belmont. CA: Wadsworth.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(airnyc2)    
head(airnyc2) 
</code></pre>

<hr>
<h2 id='annualppCities'>Time series of annual precipitations in cities of Chile.</h2><span id='topic+annualppCities'></span>

<h3>Description</h3>

<p>Data contains annual precipitations in six cities in Chile (Santiago, Talca, Chillan, Temuco, Valdivia, and Puerto Montt) at different years.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(annualppCities)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains three variables as follows:
</p>

<dl>
<dt>city</dt><dd><p>Name of city.</p>
</dd>
<dt>year</dt><dd><p>Year of registry.</p>
</dd>
<dt>annual</dt><dd><p>Value of the annual precipitation of a given year (mm).</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from <a href="https://explorador.cr2.cl/">https://explorador.cr2.cl/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(annualppCities)    
head(annualppCities) 
</code></pre>

<hr>
<h2 id='annualppCities2'>Serie de tiempo de precipitaciones anuales en Chile.</h2><span id='topic+annualppCities2'></span>

<h3>Description</h3>

<p>Data contains annual precipitations in six cities in Chile (Santiago, Talca, Chillan, Temuco, Valdivia, and Puerto Montt) at different years.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(annualppCities2)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains three variables as follows:
</p>

<dl>
<dt>ciudad</dt><dd><p>Name of city.</p>
</dd>
<dt>anho</dt><dd><p>Year of registry.</p>
</dd>
<dt>pp.anual</dt><dd><p>Value of the annual precipitation of a given year (mm).</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron obtenidos desde <a href="https://explorador.cr2.cl/">https://explorador.cr2.cl/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(annualppCities2)    
head(annualppCities2) 
</code></pre>

<hr>
<h2 id='araucaria'>Contains plot-level variables in Araucaria araucana forests in southern Chile.</h2><span id='topic+araucaria'></span>

<h3>Description</h3>

<p>These are plot-level measurement data from the Araucaria araucana forests 
in southern Chile, measured in 2009.  The data  was based on fixed-area plots of 1000 m<code class="reqn">^{2}</code>.  They are two forest stands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(araucaria)
</code></pre>


<h3>Format</h3>

<p>Contains plot-level variables as follows:
</p>

<dl>
<dt>stand</dt><dd><p>Stand number</p>
</dd>
<dt>plot.no</dt><dd><p>Plot sample identificator number</p>
</dd>
<dt>x.utm</dt><dd><p>UTM  coordinate in X-axis, in km</p>
</dd>
<dt>y.utm</dt><dd><p>UTM  coordinate in Y-axis, in km</p>
</dd>
<dt>slope</dt><dd><p>Slope, in %</p>
</dd>
<dt>aspect</dt><dd><p>Aspect, in degrees</p>
</dd>
<dt>eleva</dt><dd><p>Elevation, in msnm</p>
</dd>    
<dt>nha </dt><dd><p>Tree density, in trees/ha</p>
</dd>
<dt>gha </dt><dd><p>Basal area, in m<code class="reqn">^{2}</code>/ha</p>
</dd>
<dt>hdom</dt><dd><p>Dominant height, in m</p>
</dd>
<dt>vha </dt><dd><p>Gross stand volume, m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>dg</dt><dd><p>Diameter of the average basal area tree of the plot, in cm</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Dr Nelson Ojeda at Universidad de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Salas C, Ene L, Ojeda N, Soto H. 2010. Metodos estadísticos parametricos y no parametricos
para predecir variables de rodal basados en Landsat ETM+: una comparacion en un bosque
de Araucaria araucana en Chile [Parametric and non-parametric statistical methods for 
predicting plotwise variables based on Landsat ETM+: a comparison in an Araucaria araucana forest
in Chile]. Bosque 31(3): 179-194.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(araucaria)    
head(araucaria) 
</code></pre>

<hr>
<h2 id='araucaria2'>Variables a nivel de parcela para bosques de Araucaria araucana el sur de Chile.</h2><span id='topic+araucaria2'></span>

<h3>Description</h3>

<p>Estos son variables a nivel de parcela para bosques Araucaria araucana  
en el sur de Chile, medidos en 2009.  Estas variables se basan en mediciones
realizadas en parcelas de muestreo de superficie fija de 1000 m<code class="reqn">^{2}</code>.  Hay dos rodales.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(araucaria)
</code></pre>


<h3>Format</h3>

<p>Contains plot-level variables as follows:
</p>

<dl>
<dt>rodal</dt><dd><p>Stand number</p>
</dd>
<dt>parcela</dt><dd><p>Plot sample identificator number</p>
</dd>
<dt>x.utm</dt><dd><p>UTM  coordinate in X-axis, in km</p>
</dd>
<dt>y.utm</dt><dd><p>UTM  coordinate in Y-axis, in km</p>
</dd>
<dt>pendiente</dt><dd><p>Slope, in %</p>
</dd>
<dt>exposicion</dt><dd><p>Aspect, in degrees</p>
</dd>
<dt>altitud</dt><dd><p>Elevation, in msnm</p>
</dd>    
<dt>nha</dt><dd><p>Densidad, en arb/ha</p>
</dd>
<dt>gha</dt><dd><p>Area basal, en m<code class="reqn">^{2}</code>/ha</p>
</dd>
<dt>hdom</dt><dd><p>Altura dominante, en m</p>
</dd>
<dt>vha</dt><dd><p>Volumen bruto, en m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>dg</dt><dd><p>Diameter medio cuadratico, en cm</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos a nivel de árbol fueron cedidos por el Dr Nelson Ojeda de
la Universidad de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Salas C, Ene L, Ojeda N, Soto H. 2010. Metodos estadísticos parametricos y no parametricos
para predecir variables de rodal basados en Landsat ETM+: una comparacion en un bosque
de Araucaria araucana en Chile. Bosque 31(3): 179-194.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(araucaria2)    
head(araucaria2) 
</code></pre>

<hr>
<h2 id='baiTreelines'>Annual basal area increment for four tree species.</h2><span id='topic+baiTreelines'></span>

<h3>Description</h3>

<p>The dataset contains 157 observations of the last ten years in 6-8 adult trees of different species at three
elevations of altitudinal gradients sampled in four locations in Chile and two in Spain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(baiTreelines)
</code></pre>


<h3>Format</h3>

<p>Contains seven columns, as follows:
</p>

<dl>
<dt>climate</dt><dd><p>Climate of each location, mediterranean and temperate.</p>
</dd>
<dt>site</dt><dd><p>Name of Location of study (termmas:Termas de Chillan , antillanca:Antillanca area within Puyehue National Park,
castillo:Cerro Castillo Natural Reserve, farellones:Farellones in Central Chile, 
pyrenees: Sierra de Cutas area in Spanish Central Pyrenees,sierra:Sierra Nevada).</p>
</dd>
<dt>species</dt><dd><p>name species of study (lenga: Nothofagus pumilio, frangel: Kageneckia angustifolia,
uncinata: Pinus uncinata, sylvestris: Pinus sylvestris).</p>
</dd>
<dt>elevation</dt><dd><p>Type of elevation. &quot;Treeline&quot;, intermediate named as &quot;inter&quot;, and closed or
montane forest named as low.</p>
</dd>
<dt>tree</dt><dd><p>Id for tree.</p>
</dd>
<dt>bai</dt><dd><p>Value of annual basal area increment.</p>
</dd>
<dt>mean.bai</dt><dd><p>Mean of annual basal area increment.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.ks97h">doi:10.5061/dryad.ks97h</a>.
</p>


<h3>References</h3>

<p>Piper F, Vinegla B, Linares J, Camarero J, Cavieres L, Fajardo A. 2016. Mediterranean and temperate treelines are 
controlled by different environmental drivers. Journal of of Ecology. 104: 691-702.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(baiTreelines)    
head(baiTreelines) 
</code></pre>

<hr>
<h2 id='baiTreelines2'>Incremento anual en area basal de cuatro especies arboreas.</h2><span id='topic+baiTreelines2'></span>

<h3>Description</h3>

<p>Este set de datps contiene 157 observaciones, de los ultimos 10 años en 6-8 árboles adultos de cuatro especies 
en un gradiente altitudinal. Las muestras se distribuyeron en cuatro localidades o sitios de Chile y dos en España.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(baiTreelines2)
</code></pre>


<h3>Format</h3>

<p>Contains seven columns, as follows:
</p>

<dl>
<dt>clima</dt><dd><p>Climate of each location, mediterranean and temperate.</p>
</dd>
<dt>sitio</dt><dd><p>Name of Location of study (termmas:Termas de Chillan , antillanca:Antillanca area within Puyehue National Park,
castillo:Cerro Castillo Natural Reserve, farellones:Farellones in Central Chile, 
pyrenees: Sierra de Cutas area in Spanish Central Pyrenees,sierra:Sierra Nevada).</p>
</dd>
<dt>especie</dt><dd><p>name species of study (lenga: Nothofagus pumilio, frangel: Kageneckia angustifolia,
uncinata: Pinus uncinata, sylvestris: Pinus sylvestris).</p>
</dd>
<dt>tipo.altitud</dt><dd><p>Type of elevation. &quot;Treeline&quot;, intermediate named as &quot;inter&quot;, and closed or
montane forest named as low.</p>
</dd>
<dt>arbol</dt><dd><p>Id for tree.</p>
</dd>
<dt>bai</dt><dd><p>Value of annual basal area increment.</p>
</dd>
<dt>bai.medio</dt><dd><p>Mean of annual basal area increment.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.ks97h">doi:10.5061/dryad.ks97h</a>.
</p>


<h3>References</h3>

<p>Piper F, Vinegla B, Linares J, Camarero J, Cavieres L, Fajardo A. 2016. Mediterranean and temperate treelines are 
controlled by different environmental drivers. Journal of Ecology. 104: 691-702.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(baiTreelines2)    
head(baiTreelines2) 
</code></pre>

<hr>
<h2 id='bears'>Age and physical measurement data for wild bears</h2><span id='topic+bears'></span>

<h3>Description</h3>

<p>Wild bears were anesthetized, and their bodies were measured and weighed. One goal of the study
was to make a table (or perhaps a set of tables) for people interested in estimating the weight of a
bear based on other measurements. Notice that there are missing values for some of the variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bears)
</code></pre>


<h3>Format</h3>

<p>Contains individual-level variables, as follows:
</p>

<dl>
<dt>id</dt><dd><p>Bear id</p>
</dd>
<dt>age</dt><dd><p>Age in total number of months.</p>
</dd>
<dt>month</dt><dd><p>Month number within a given year.</p>
</dd>
<dt>sex</dt><dd><p>1 =male, 2 = female.</p>
</dd>
<dt>headL</dt><dd><p>Length of head, in cm.</p>
</dd>
<dt>headW</dt><dd><p>Width of head, in cm.</p>
</dd>
<dt>neckG</dt><dd><p>Girth of neck, in cm.</p>
</dd>
<dt>length</dt><dd><p>Body length, in cm.</p>
</dd>
<dt>chestG</dt><dd><p>Girth of chest, in cm.</p>
</dd>
<dt>weight</dt><dd><p>body weight, in kg.</p>
</dd>
<dt>obs</dt><dd><p>Temporal observation number for bear.</p>
</dd>
<dt>name</dt><dd><p>Name given to bear.</p>
</dd>
</dl>



<h3>Source</h3>

<p>According to Prof. Timothy Gregoire at Yale University (New Haven, CT, USA), the
data set was supplied by Gary Alt.
</p>


<h3>References</h3>

<p>Entertaining references are in Reader's Digest April, 1979, and Sports Afield September, 1981.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bears)    
head(bears) 
table(bears$sex)
boxplot(headL~sex, data=bears)
</code></pre>

<hr>
<h2 id='bears2'>Edad y características biométricas de osos salvajes</h2><span id='topic+bears2'></span>

<h3>Description</h3>

<p>Los osos salvajes fueron anestesiados y sus cuerpos medidos. Uno de los objetivos del estudio fue hacer una
tabla (o quizas un conjunto de tablas) para las personas interesadas en estimar el peso de un oso basandose en otras medidas. 
Observe que faltan valores para algunas de las variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bears2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables de nivel individual, como se describen a continuacion:
</p>

<dl>
<dt>id</dt><dd><p>Identificador del oso</p>
</dd>
<dt>edad</dt><dd><p>edad en meses</p>
</dd>
<dt>mes</dt><dd><p>Diámetro a la altura del pecho, en cm</p>
</dd>
<dt>sexo</dt><dd><p>1 = hombre, 2 = mujer</p>
</dd>
<dt>cabezaL</dt><dd><p>longitud de la cabeza, en cm</p>
</dd>
<dt>cabezaA</dt><dd><p>ancho de la cabeza, en cm</p>
</dd>
<dt>cuelloP</dt><dd><p>circunferencia del cuello, en cm</p>
</dd>
<dt>largo</dt><dd><p>longitud del cuerpo, en cm</p>
</dd>
<dt>pechoG</dt><dd><p>circunferencia del pecho, en cm</p>
</dd>
<dt>peso</dt><dd><p>peso corporal, en kg</p>
</dd>
<dt>obs</dt><dd><p>número de observación temporal para el oso</p>
</dd>
<dt>nombre</dt><dd><p>nombre dado al oso</p>
</dd>
</dl>



<h3>Source</h3>

<p>Segun el Prof. Timothy Gregoire de Yale University (New Haven, CT, USA), los datos fueron
cedidos por Gary Alt. Minitab, Inc. La descripcion de los datos fue dada por él.
</p>


<h3>References</h3>

<p>Algunas referencias generales estan en el Reader's Digest de Abril, 1979, y Sports Afield de Septiembre, 1981.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bears2)    
head(bears2) 
table(bears2$sexo)
boxplot(cabezaL~sexo, data=bears2)
</code></pre>

<hr>
<h2 id='bearsDepu'>Age and physical measurement data for wild bears (without missing values)</h2><span id='topic+bearsDepu'></span>

<h3>Description</h3>

<p>Wild bears were anesthetized, and their bodies were measured and weighed. One goal of the study
was to make a table (or perhaps a set of tables) for people interested in estimating the weight of a
bear based on other measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bearsDepu)
</code></pre>


<h3>Format</h3>

<p>Individual-level variables, as follows:
</p>

<dl>
<dt>id</dt><dd><p>Bear identificator.</p>
</dd>
<dt>age</dt><dd><p>Age in total number of months.</p>
</dd>
<dt>month</dt><dd><p>Month number within a given year.</p>
</dd>
<dt>sex</dt><dd><p>Sex code: 1 =male, 2 = female.</p>
</dd>
<dt>headL</dt><dd><p>Length of head, in cm.</p>
</dd>
<dt>headW</dt><dd><p>Width of head, in cm.</p>
</dd>
<dt>neckG</dt><dd><p>Girth of neck, in cm.</p>
</dd>
<dt>length</dt><dd><p>Body length, in cm.</p>
</dd>
<dt>chestG</dt><dd><p>Girth of chest, in cm.</p>
</dd>
<dt>weight</dt><dd><p>Body weight, in kg.</p>
</dd>
<dt>obs</dt><dd><p>Temporal observation number for bear.</p>
</dd>
<dt>name</dt><dd><p>name given to bear</p>
</dd>
</dl>



<h3>Source</h3>

<p>According to Prof. Timothy Gregoire at Yale University (New Haven, CT, USA), the
data set was supplied by Gary Alt.
</p>


<h3>References</h3>

<p>Entertaining references are in Reader's Digest April, 1979, and Sports Afield September, 1981.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bearsDepu)    
head(bearsDepu)
table(bearsDepu$sex)
boxplot(headL~sex, data=bearsDepu) 
</code></pre>

<hr>
<h2 id='bearsDepu2'>Edad y características biométricas de osos salvajes (sin datos faltantes)</h2><span id='topic+bearsDepu2'></span>

<h3>Description</h3>

<p>Los osos salvajes fueron anestesiados y sus cuerpos medidos. Uno de los objetivos del estudio fue hacer una 
tabla (o quizas un conjunto de tablas) para las personas interesadas en estimar el peso de un oso basandose en otras medidas.
Esta dataframe es igual que &quot;bears&quot; pero sin valores perdidos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bearsDepu2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables de nivel individual, como se describen a continuacion:
</p>

<dl>
<dt>id</dt><dd><p>Identificador del oso.</p>
</dd>
<dt>edad</dt><dd><p>edad en meses.</p>
</dd>
<dt>mes</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>sexo</dt><dd><p>1 = hombre, 2 = mujer.</p>
</dd>
<dt>cabezaL</dt><dd><p>longitud de la cabeza, en cm.</p>
</dd>
<dt>cabezaA</dt><dd><p>ancho de la cabeza, en cm.</p>
</dd>
<dt>cuelloP</dt><dd><p>circunferencia del cuello, en cm.</p>
</dd>
<dt>largo</dt><dd><p>longitud del cuerpo, en cm.</p>
</dd>
<dt>pechoG</dt><dd><p>circunferencia del pecho, en cm.</p>
</dd>
<dt>peso</dt><dd><p>peso corporal, en kg.</p>
</dd>
<dt>obs</dt><dd><p>número de observación temporal para el oso.</p>
</dd>
<dt>nombre</dt><dd><p>nombre dado al oso.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Segun el Prof. Timothy Gregoire de Yale University (New Haven, CT, USA), los datos fueron
cedidos por Gary Alt. Minitab, Inc. La descripcion de los datos fue dada por el .
</p>


<h3>References</h3>

<p>Algunas referencias generales estan en el Reader's Digest de Abril, 1979, y Sports Afield de Septiembre, 1981.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bearsDepu2)    
head(bearsDepu2)
table(bearsDepu2$sexo)
boxplot(cabezaL~sexo, data=bearsDepu2) 
</code></pre>

<hr>
<h2 id='beetles'>Population density growth of beetles</h2><span id='topic+beetles'></span>

<h3>Description</h3>

<p>Temporal measurements of density of beetles (Tribolium confusum) growing in different controlled environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beetles
</code></pre>


<h3>Format</h3>


<dl>
<dt>days</dt><dd><p>Number of days.</p>
</dd>
<dt>diet</dt><dd><p>The quantities of flour (in grams) of the environments where the beetles were growing.  Six levels of the factor Diet.</p>
</dd>
<dt>type</dt><dd><p>The various stage of beetles, i.e., eggs, larvae, pupae, and adults.</p>
</dd>
<dt>density</dt><dd><p>The number of insects per environment.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data from Table No. 1, page 116, of Chapman (1928). Series of experiments under controlled
conditions in which flour beetles (Tribolium confusum) are kept in environments of known size. 
The period from egg to adult is approximately forty days at 27C degrees. The data were entered by
Yamara Arancibia, a former student of Prof. Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>- Chapman RN. 1928. The Quantitative Analysis of Environmental Factors. Ecology 9(2):111-122.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(beetles)    
name.diet&lt;-unique(beetles$diet)
num.diet&lt;-length(name.diet)
##Time series plot
#first, some computation
alys&lt;-with(beetles,tapply(density, list(as.factor(days),as.factor(diet)),sum))
out&lt;-as.data.frame(alys)
out$time&lt;-row.names(out)
head(out)
#Figure 1 of the paper
matplot(out[,"time"], out[,1:num.diet], las=1, type=c("b"),pch=1,
        xlab="Time in days",ylab="Total individuals")
legend("topleft", legend = name.diet, title = "Diet (gr)",
       col = 1:6, lty = 1:6, pch = 1)
</code></pre>

<hr>
<h2 id='beetles2'>Crecimiento poblacional de escarabajos</h2><span id='topic+beetles2'></span>

<h3>Description</h3>

<p>Mediciones temporales de densidad de escarabajos (Tribolium confusum) creciendo en 
diferentes ambientes controlados.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beetles2
</code></pre>


<h3>Format</h3>


<dl>
<dt>dias</dt><dd><p>Número de dias.</p>
</dd>
<dt>dieta</dt><dd><p>La cantidad de harina (en gramos) de ambientes donde crecen los escarabajos.  Seis niveles del factor Dieta.</p>
</dd>
<dt>tipo</dt><dd><p>Estados de desarrollo de los escarabajos, i.e., huevos, larvas, pupas, y adultos.</p>
</dd>
<dt>densidad</dt><dd><p>Número total de individuos por ambiente de crecimiento.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos del Cuadro No. 1, page 116, de Chapman (1928). Serie de experimentos bajo condiciones controladas
donde escarabajos (Tribolium confusum) se mantienen en ambientes de tamaño conocido. 
El periodo desde huevo a adulto es de aproximadamente de cuarenta dias a 27 grados Celsius. Los datos
fueron digitados por Yamara Arancibia, una estudiante del Prof. Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>- Chapman RN. 1928. The Quantitative Analysis of Environmental Factors. Ecology 9(2):111-122.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(beetles2)    
nom.dieta&lt;-unique(beetles2$dieta)
num.dieta&lt;-length(nom.dieta)
##Grafico de serie de tiempo
#primero algunos calculos
alys&lt;-with(beetles2,tapply(densidad, list(as.factor(dias),as.factor(dieta)),sum))
out&lt;-as.data.frame(alys)
out$tiempo&lt;-row.names(out)
head(out)
##Figura 1 del paper
matplot(out[,"tiempo"], out[,1:num.dieta], las=1, type=c("b"),pch=1,
        xlab="Tiempo en dias",ylab="Densidad de individuos")
legend("topleft", legend = nom.dieta, title = "Dieta (gr)",
       col = 1:6, lty = 1:6, pch = 1)
</code></pre>

<hr>
<h2 id='biomass'>Contains tree-level biomass data for several species in Canada.</h2><span id='topic+biomass'></span>

<h3>Description</h3>

<p>These are tree-level variables for several species in Canada.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomass
</code></pre>


<h3>Format</h3>


<dl>
<dt>treenum</dt><dd><p>tree number.</p>
</dd>
<dt>spp</dt><dd><p>species common name.</p>
</dd>
<dt>dbh</dt><dd><p>diameter at breast height, in cm.</p>
</dd>
<dt>height</dt><dd><p>total height, in m.</p>
</dd>
<dt>totbiom</dt><dd><p>total biomass, in kg.</p>
</dd>
<dt>bolebiom</dt><dd><p>stem biomass, in kg.</p>
</dd>
<dt>branchbiom</dt><dd><p>branches biomass, in kg.</p>
</dd>
<dt>foliagebiom</dt><dd><p>foliage biomass, in kg.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Timothy Gregoire at
the School of Forestry and Environmental Studies at Yale University (New Haven, CT, USA).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(biomass)    
head(biomass) 
</code></pre>

<hr>
<h2 id='biomass2'>Biomasa a nivel de árbol para especies arboreas de Canada.</h2><span id='topic+biomass2'></span>

<h3>Description</h3>

<p>These are tree-level variables for several species in Canada.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomass2
</code></pre>


<h3>Format</h3>


<dl>
<dt>treenum</dt><dd><p>tree number.</p>
</dd>
<dt>spp</dt><dd><p>species common name.</p>
</dd>
<dt>dbh</dt><dd><p>diameter at breast height, in cm.</p>
</dd>
<dt>height</dt><dd><p>total height, in m.</p>
</dd>
<dt>totbiom</dt><dd><p>total biomass, in kg.</p>
</dd>
<dt>bolebiom</dt><dd><p>stem biomass, in kg.</p>
</dd>
<dt>branchbiom</dt><dd><p>branches biomass, in kg.</p>
</dd>
<dt>foliagebiom</dt><dd><p>foliage biomass, in kg.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Timothy Gregoire at
the School of Forestry and Environmental Studies at Yale University (New Haven, CT, USA).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(biomass2)    
head(biomass2) 
</code></pre>

<hr>
<h2 id='carbohydrateTreelines'>Carbohydrates concentrations of tree species.</h2><span id='topic+carbohydrateTreelines'></span>

<h3>Description</h3>

<p>Dataset contains 863 observations, about of  total soluble carbohydrate, starch, and non structural carbohydrates 
concentrations per mass unit and per volume unit, in three tissues in early summer and early autumn 6-8 adult trees
of different species at three elevations of altitudinal gradients sampled in four locations of Chile 
and Spain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(carbohydrateTreelines)
</code></pre>


<h3>Format</h3>

<p>Contains 16 variables, as follows:
</p>

<dl>
<dt>climate</dt><dd><p>Climate of each location, mediterranean and temperate.</p>
</dd>
<dt>site</dt><dd><p>Name of Location of study (termas:Termas de Chillan , antillanca:Antillanca area within Puyehue National Park,
castillo:Cerro Castillo Natural Reserve, farellones:Farellones in Central Chile, 
pyrenees: Sierra de Cutas area in Spanish Central Pyrenees,sierra:Sierra Nevada).</p>
</dd>
<dt>species</dt><dd><p>name species of study (lenga: Nothofagus pumilio, frangel: Kageneckia angustifolia,
uncinata: Pinus uncinata, sylvestris: Pinus sylvestris).</p>
</dd>
<dt>tissue</dt><dd><p>Type of tissue, new developing twings, stem sapwood and branches.</p>
</dd>
<dt>time</dt><dd><p>Meauserement season (spring or autumn).</p>
</dd>
<dt>elevation</dt><dd><p>Type of elevation. &quot;Treeline&quot;, intermediate named as &quot;mid&quot;, and closed or
montane forest named as &quot;low&quot;.</p>
</dd>
<dt>tree</dt><dd><p>Id for tree.</p>
</dd>
<dt>tree.site</dt><dd><p>Id site for each location of study.</p>
</dd>
<dt>tss</dt><dd><p>Value of concentrations soluble carbohydrate per mass unit.</p>
</dd>
<dt>st</dt><dd><p>Value of concentrations starch per mass unit.</p>
</dd>
<dt>nsc</dt><dd><p>Value of concentrations non structural carbohydrates per mass unit.</p>
</dd>
<dt>tss.nsc</dt><dd><p>.</p>
</dd>
<dt>wd</dt><dd><p>It might be 'wood density', but not sure.</p>
</dd>
<dt>tss.mv</dt><dd><p>Value of concentrations soluble carbohydrate per volume unit.</p>
</dd>
<dt>st.mv</dt><dd><p>Value of concentrations starch per volume unit.</p>
</dd>
<dt>nsc.mv</dt><dd><p>Value of concentrations non structural carbohydrates per volume unit.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.ks97h">doi:10.5061/dryad.ks97h</a>.
</p>


<h3>References</h3>

<p>Piper F, Vinegla B, Linares J, Camarero J, Cavieres L, Fajardo A. 2016. Mediterranean and temperate treelines are 
controlled by different environmental drivers. Journal of Ecology. 104: 691-702.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(carbohydrateTreelines)    
head(carbohydrateTreelines) 
</code></pre>

<hr>
<h2 id='cdf'>Computes the cumulative distribution for a given random variable</h2><span id='topic+cdf'></span>

<h3>Description</h3>

<p>Cumulative distribution for a given random variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cdf(y = y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cdf_+3A_y">y</code></td>
<td>
<p>a vector of a random variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The cumulative distribution of the random variable is build every 5
the data at hand, that is to say, from 0.05 (i.e., 5
</p>


<h3>Value</h3>

<p>The function returns a dataframe having two columns: the first contains 
the random variable values and the second the cumulative distribution for the variable.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y.var &lt;- rnorm(10)
cdf(y.var)
</code></pre>

<hr>
<h2 id='chicksw'>Chicken growth data.</h2><span id='topic+chicksw'></span>

<h3>Description</h3>

<p>The body weights of the chicks were measured at birth and every second day thereafter until day 20. They were 
also measured on day 21. There were four groups on chicks on different protein diets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chicksw)
</code></pre>


<h3>Format</h3>

<p>Contains four variables, as follows:
</p>

<dl>
<dt>weight</dt><dd><p>A numeric vector giving the body weight of the chick (gm).</p>
</dd>
<dt>time</dt><dd><p>A numeric vector giving the number of days since birth when the measurement was made.</p>
</dd>
<dt>chick</dt><dd><p>An ordered factor with levels different giving a unique identifier for the chick. The ordering of the
levels groups chicks on the same diet together and orders them according to their final weight 
(lightest to heaviest) within diet.</p>
</dd>
<dt>diet</dt><dd><p>A factor with levels 1,2,3 and 4 indicating which experimental diet the chick received.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the alr4 library.
</p>


<h3>References</h3>

<p>Crowder M, Hand D. 1990. Analysis of Repeated Measures. Chapman and Hall
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chicksw)    
head(chicksw) 
</code></pre>

<hr>
<h2 id='contrast'>Computes statistics for inference in a given contrast</h2><span id='topic+contrast'></span>

<h3>Description</h3>

<p>The function computes the statistics for inference in a given contrast, subject to
a given significance level. Those statistics are as follows: estimated contrast, 
standard error of the contrast, and confidence interval of the contrast.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrast(
  model = model,
  coef.cont = coef.cont,
  grp.m = grp.m,
  grp.n = grp.n,
  alpha = 0.05,
  full = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contrast_+3A_model">model</code></td>
<td>
<p>object containing the fitted model</p>
</td></tr>
<tr><td><code id="contrast_+3A_coef.cont">coef.cont</code></td>
<td>
<p>vector with the coefficients to establish the contrasts</p>
</td></tr>
<tr><td><code id="contrast_+3A_grp.m">grp.m</code></td>
<td>
<p>a vector having the sample mean per each group, or level of the factor under study.</p>
</td></tr>
<tr><td><code id="contrast_+3A_grp.n">grp.n</code></td>
<td>
<p>a vector having the sample size per each group, or level of the factor under study.</p>
</td></tr>
<tr><td><code id="contrast_+3A_alpha">alpha</code></td>
<td>
<p>is the significance level for building the confidence intervals. Default value is 0.05, which is 95% confidence level.</p>
</td></tr>
<tr><td><code id="contrast_+3A_full">full</code></td>
<td>
<p>FALSE if want short output, TRUE for longer (i.e. more details). Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The contrast is established based upon an already fitted statistical model that
describe the relationship among variables. The significance level (alpha) is defined
by the user, although by default has been set to 0.05, that is to say, a 95
statistical confidence.
</p>


<h3>Value</h3>

<p>This function returns the above described statistics for a given contrast.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>References</h3>

<p>Stage AR. 1963. A mathematical approach to polymorphic site index curves for 
Grand fir. Forest Science 9(2):167–180.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(fertiliza)
table(fertiliza$treat)
means.trt &lt;- tapply(fertiliza$volume,fertiliza$treat,mean);means.trt
sds.trt &lt;- tapply(fertiliza$volume,fertiliza$treat,sd);sds.trt
ns.trt &lt;- tapply(fertiliza$volume,fertiliza$treat,length);ns.trt
m1 &lt;- lm(volume ~ treat, data=fertiliza)
anova(m1)
## Coefficients to be used in the contrast
#c1: (tmoA1-A2) - (tmoA3-A4)
C1.coeff &lt;- c(0,1,1,-1,-1)
contrast(model=m1,C1.coeff,grp.m=means.trt,grp.n=ns.trt,alpha=0.1,full=TRUE)
contrast(model=m1,C1.coeff,grp.m=means.trt,grp.n=ns.trt,alpha=0.1,full=FALSE)
contrast(m1,C1.coeff,grp.m=means.trt,grp.n=ns.trt,alpha=0.05,full=TRUE)
contrast(m1,C1.coeff,grp.m=means.trt,grp.n=ns.trt)

</code></pre>

<hr>
<h2 id='corkoak'>Tree-level cork biomass data for Oak trees in Portugal</h2><span id='topic+corkoak'></span>

<h3>Description</h3>

<p>Measurements of cork weight in sample trees of Quercus suber (Oak) in Portugal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corkoak
</code></pre>


<h3>Format</h3>


<dl>
<dt>tree</dt><dd><p>A correlative number for each sample tree.</p>
</dd>
<dt>csc</dt><dd><p>is tree circumference at 1.3 m outside bark, in cm.</p>
</dd>
<dt>cbc</dt><dd><p>is tree circumference at 1.3 m under bark, in cm.</p>
</dd>
<dt>bt</dt><dd><p>bark thickness, in cm.</p>
</dd>
<dt>hdeb</dt><dd><p>is debarking height, in m.</p>
</dd>	
<dt>hblc</dt><dd><p>height to base of live crown, in m.</p>
</dd>	
<dt>nb</dt><dd><p>number of branches debarked</p>
</dd>
<dt>cr.diam</dt><dd><p>crown diameter, in m.</p>
</dd>
<dt>w</dt><dd><p>total green weight of the stripped cork, in kg</p>
</dd>
<dt>stratum</dt><dd><p>Stratum</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data supplied electronically to Prof. Timothy Gregoire (Yale University)
by authors accompanied by a note which said &quot;After the article was published we discovered a problem with 2 of the observations so Teresa	
and I decided it was best just to delete them.&quot;
</p>


<h3>References</h3>

<p>- Fonseca TJ, Parresol BR. 2001. A new model for cork weight estimation in northern Portugal with
methodology for construction of confidence intervals. Forest Ecology and Management 152(1):131–139.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(corkoak)    
head(corkoak) 
</code></pre>

<hr>
<h2 id='corkoak2'>Datos de biomasa de corcho en árboles de Encino en Portugal</h2><span id='topic+corkoak2'></span>

<h3>Description</h3>

<p>Mediciones de peso de corcho en árboles muestra de Quercus suber en Portugal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corkoak2
</code></pre>


<h3>Format</h3>


<dl>
<dt>arbol</dt><dd><p>A correlative number for each sample tree.</p>
</dd>
<dt>perimetro.cc</dt><dd><p>is tree circumference at 1.3 m outside bark, in cm.</p>
</dd>
<dt>perimetro.sc</dt><dd><p>is tree circumference at 1.3 m under bark, in cm.</p>
</dd>
<dt>e.corteza</dt><dd><p>bark thickness, in cm.</p>
</dd>
<dt>h.desc</dt><dd><p>is debarking height, in m.</p>
</dd>	
<dt>hcc</dt><dd><p>height to base of live crown, in m.</p>
</dd>	
<dt>num.ram</dt><dd><p>number of branches debarked</p>
</dd>
<dt>diam.copa</dt><dd><p>crown diameter, in m.</p>
</dd>
<dt>biomasa</dt><dd><p>total green weight of the stripped cork, in kg</p>
</dd>
<dt>estrato</dt><dd><p>Estrato</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos cedidos por Prof. Timothy Gregoire (Yale University)
y los autores originales mencionaron &quot;After the article was published we discovered a problem with 2 of the observations so Teresa	
and I decided it was best just to delete them.&quot;
</p>


<h3>References</h3>

<p>- Fonseca TJ, Parresol BR. 2001. A new model for cork weight estimation in northern Portugal with
methodology for construction of confidence intervals. Forest Ecology and Management 152(1):131–139.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(corkoak2)    
head(corkoak2) 
</code></pre>

<hr>
<h2 id='crown'>Tree crown radii</h2><span id='topic+crown'></span>

<h3>Description</h3>

<p>Crown radii measurements in cardinal directions for sample trees at the Rucamanque experimental forest, near Temuco, Chile.
Data were collected within a sample plot of 250 m<code class="reqn">^{2}</code>, located in a secondary forest stand
dominated by Nothofagus obliqua.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(crown)
</code></pre>


<h3>Format</h3>

<p>Contains of variables, as follows:
</p>

<dl>
<dt>spp</dt><dd><p>Species code. 'Ro' is Nothofagus obliqua (roble), 'Co' is Nothofagus dombeyi (Coigue) and 'Ol' is Olivillo.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>crad.n</dt><dd><p>Crown radii towards the north, in m.</p>
</dd>
<dt>crad.e</dt><dd><p>Crown radii towards the east, in m.</p>
</dd>
<dt>crad.s</dt><dd><p>Crown radii towards the south, in m.</p>
</dd>
<dt>crad.w</dt><dd><p>Crown radii towards the west, in m.</p>
</dd>
<dt>x.coord</dt><dd><p>Cardinal position at the X-axis, in m.</p>
</dd>
<dt>y.coord</dt><dd><p>Cardinal position at the Y-axis, in m.</p>
</dd>
<dt>cr.diam</dt><dd><p>Crown diameter, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Christian Salas-Eljatib, Universidad de Chile (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2001. Caracterización básica del relicto de Biodiversidad 
Rucamanque [Basic characterization of the biodiversity remnant Rucamanque]. Bosque Nativo, 29:3-9. <a href="https://eljatib.com/publication/2001-06-01_caracterizacion_basi/">https://eljatib.com/publication/2001-06-01_caracterizacion_basi/</a>
</p>
<p>- Salas C, and Garcia O. 2006. Modelling height development of mature Nothofagus obliqua. 
Forest Ecology and Management 229 (1-3): 1-6. <a href="https://doi.org/10.1016/j.foreco.2006.04.015">doi:10.1016/j.foreco.2006.04.015</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(crown)    
table(crown$spp) 
descstat(crown[,c("dbh","cr.diam")]) 
</code></pre>

<hr>
<h2 id='crown2'>Radios de copa de árboles</h2><span id='topic+crown2'></span>

<h3>Description</h3>

<p>Mediciones de radios de copa en direcciones cardinales para árboles muestra en Rucamanque, cerca de Temuco, Chile.
Los datos fueron colectados al interior de una parcela de muestreo de 250 m<code class="reqn">^{2}</code>, establecidad en un bosque
secundario dominado por Nothofagus obliqua.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(crown2)
</code></pre>


<h3>Format</h3>

<p>Contiene las siguientes columnas:
</p>

<dl>
<dt>espe</dt><dd><p>Código de especie, donde: 'Ro' es Nothofagus obliqua (Roble), 'Co' es Nothofagus dombeyi (Coigue) y 'Ol' es Olivillo.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>rc.n</dt><dd><p>Radio de copa hacia el Norte, en m.</p>
</dd>
<dt>rc.e</dt><dd><p>Radio de copa hacia el Este, en m.</p>
</dd>
<dt>rc.s</dt><dd><p>Radio de copa hacia el Sur, en m.</p>
</dd>
<dt>rc.w</dt><dd><p>Radio de copa hacia el Oeste, en m.</p>
</dd>
<dt>coord.x</dt><dd><p>Posición cartesiana en el eje-X, en m.</p>
</dd>
<dt>coord.y</dt><dd><p>Posición cartesiana en el eje-Y, en m.</p>
</dd>
<dt>dcopa</dt><dd><p>Diámetro de copa, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos cedidos por el Prof. Christian Salas-Eljatib, Universidad de Chile (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2001. Caracterización básica del relicto de Biodiversidad 
Rucamanque [Basic characterization of the biodiversity remnant Rucamanque]. Bosque Nativo, 29:3-9. <a href="https://eljatib.com/publication/2001-06-01_caracterizacion_basi/">https://eljatib.com/publication/2001-06-01_caracterizacion_basi/</a>
</p>
<p>- Salas C, and Garcia O. 2006. Modelling height development of mature Nothofagus obliqua. 
Forest Ecology and Management 229 (1-3): 1-6. <a href="https://doi.org/10.1016/j.foreco.2006.04.015">doi:10.1016/j.foreco.2006.04.015</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(crown2)    
table(crown2$espe) 
descstat(crown2[,c("dap","dcopa")]) 
</code></pre>

<hr>
<h2 id='deadForestCA'>Data contains  climatic, forest structure and forest mortality variable</h2><span id='topic+deadForestCA'></span>

<h3>Description</h3>

<p>The data file contains one row per unique 3.5km grid cell by year combination. The data frame covers 
all grid cells within the state of California where at least one Aerial Detection Survey (ADS) flight 
was taken between 2009 and 2015, so each grid cell position has between 1 and 7 years of data 
(reflected as 1 to 7 rows in the data file per grid cell position). The main response variables 
are mort.bin (presence of any mortality) and mort.tph (number of dead trees/ha within the given 
grid cell by year).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(deadForestCA)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>live.bah</dt><dd><p>Live basal area from the GNN dataset</p>
</dd>											
<dt>live.tph</dt><dd><p>Live trees per hectare from the GNN dataset</p>
</dd>
<dt>pos.x</dt><dd><p>rank-order x-position of the grid cell (position 1 is western-most)</p>
</dd>
<dt>pos.y</dt><dd><p>rank-order y-position of the grid cell (position 1 is northern-most)</p>
</dd>
<dt>alb.x</dt><dd><p>x-coordinate of the grid cell centroid in California Albers (EPSG 3310)</p>
</dd>
<dt>alb.y</dt><dd><p>y-coordinate of the grid cell centroid in California Albers (EPSG 3310)</p>
</dd>
<dt>mort.bin</dt><dd><p>1= dead trees observed in grid cell. 0= no dead trees observed</p>
</dd>
<dt>mort.tph</dt><dd><p>Dead trees per hectare from the aggregated ADS dataset</p>
</dd>
<dt>mort.tpa</dt><dd><p>Dead trees per acre from the aggregated ADS dataset</p>
</dd>
<dt>year</dt><dd><p>Year of the ADS flight. Most flights occurred from May-August.</p>
</dd>
<dt>Defnorm</dt><dd><p>Mean annual climatic water deficit for the grid cell, for Oct 1-Sept 31 water year, averaged from 1981-2015</p>
</dd>
<dt>Def0</dt><dd><p>Climatic water deficit for the grid cell during the Oct-Sept water year overlapping the summer ADS flight of the given year</p>
</dd>
<dt>Defz0</dt><dd><p>Z-score for climatic water deficit for the given grid cell/water year. Calculated as (Def0-Defnorm)/(standard deviation in deficit among all years 1981-2015 for the given grid cell)</p>
</dd>
<dt>Defz1</dt><dd><p>Z-score for climatic water deficit for the given grid cell in the preceeding water year.</p>
</dd>
<dt>Defz2</dt><dd><p>Z-score for climatic water deficit for the given grid cell two water years prior.</p>
</dd>
<dt>Tz0</dt><dd><p>Z-score for temperature for the given grid cell/year.</p>
</dd>
<dt>Pz0</dt><dd><p>Z-score for precipitation for the given grid cell/year.</p>
</dd>
<dt>Defquant</dt><dd><p>FDCI variable. Quantile of Defnorm of the given grid cell, relative to the Defnorm of all other grid cells with a basal area within 2.5 m<code class="reqn">^{2}</code>/ha of the given cell is basal area.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided from DRYAD repository.
</p>


<h3>References</h3>

<p>-Derek J. N. Young, Jens T. Stevens, J. Mason Earles, Jeffrey Moore, Adam Ellis, Amy L. Jirka, and Andrew M. Latimer.
Long-term climate and competition explain forest mortality patterns under extreme drought.Ecology Letters, 20(1):78-86, 2017.
</p>
<p>-C. Salas-Eljatib, Andres Fuentes-Ramirez, Timothy G. Gregoire, Adison Altamirano, and Valeska Yaitul. A study on the effects of unbalanced data when
fitting logistic regression models in ecology. Ecological Indicators, 85:502-508, 2018
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(deadForestCA)    
head(deadForestCA) 
</code></pre>

<hr>
<h2 id='deadForestCA2'>Los datos contienen variables climaticas, de estructura forestal y de mortalidad forestal.</h2><span id='topic+deadForestCA2'></span>

<h3>Description</h3>

<p>El archivo de datos contiene una fila por combinacion unica de celda de cuadricula de 3,5 km por año. 
El marco de datos cubre todas las celdas de la cuadricula dentro del estado de California donde se 
tomo al menos un vuelo de la Encuesta de deteccion aerea (ADS) entre 2009 y 2015, por lo que cada posicion
de celda de la cuadricula tiene entre 1 y 7 años de datos (reflejados como 1 a 7 filas en el archivo de datos
por posicion de celda de cuadricula). Las principales variables de respuesta son mort.bin (presencia de alguna mortalidad)
y mort.tph (número de árboles muertos /ha dentro de la celda de la cuadricula por año).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(deadForestCA2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuacion::
</p>

<dl>
<dt>live.bah</dt><dd><p>Area basal viva del conjunto de datos GNN</p>
</dd>											
<dt>live.tph</dt><dd><p>Árboles vivos por hectarea del conjunto de datos GNN</p>
</dd>
<dt>pos.x</dt><dd><p>posicion x del orden de clasificacion de la celda de la cuadricula (la posicion 1 es la mas occidental)</p>
</dd>
<dt>pos.y</dt><dd><p>posicion y del orden de clasificacion de la celda de la cuadricula (la posicion 1 es la mas al norte)</p>
</dd>
<dt>alb.x</dt><dd><p>coordenada x del centroide de la celda de la cuadricula en California Albers (EPSG 3310)</p>
</dd>
<dt>alb.y</dt><dd><p>coordenada y del centroide de la celda de la cuadricula en California Albers (EPSG 3310)</p>
</dd>
<dt>mort.bin</dt><dd><p>1 = árboles muertos observados en la celda de la cuadricula. 0 = no se observaron árboles muertos</p>
</dd>
<dt>mort.tph</dt><dd><p>Árboles muertos por hectarea del conjunto de datos ADS agregado</p>
</dd>
<dt>mort.tpa</dt><dd><p>Árboles muertos por acre del conjunto de datos ADS agregado</p>
</dd>
<dt>year</dt><dd><p>año del vuelo de ADS. La mayoría de los vuelos se realizaron entre mayo y agosto</p>
</dd>
<dt>Defnorm</dt><dd><p>Deficit hidrico climatico anual medio para la celda de la cuadricula, para el año hidrico del 1 de octubre al 31 de septiembre, promediado de 1981 a 2015</p>
</dd>
<dt>Def0</dt><dd><p>Deficit de agua climatica para la celda de la cuadricula durante el año hidrologico de octubre a septiembre que se superpone al vuelo ADS de verano del año dado</p>
</dd>
<dt>Defz0</dt><dd><p>Puntaje Z para el deficit hidrico climatico para la celda de cuadricula / año hidrico dado. Calculado como (Def0-Defnorm) / (desviacion estandar en el deficit entre todos los años 1981-2015 para la celda de la cuadricula dada</p>
</dd>
<dt>Defz1</dt><dd><p>Puntuacion Z para el deficit h?drico climatico para la celda de la cuadricula dada en el año hidrologico anterior.</p>
</dd>
<dt>Defz2</dt><dd><p>Puntuacion Z para el deficit hidrico climatico para la celda de la cuadricula dada dos años antes.</p>
</dd>
<dt>Tz0</dt><dd><p>Puntaje Z para la temperatura para la celda de cuadricula / año dado.</p>
</dd>
<dt>Pz0</dt><dd><p>Puntaje Z para la precipitacion para la celda / año de la cuadricula dado.</p>
</dd>
<dt>Defquant</dt><dd><p>Variable FDCI. Cuantil de Defnorm de la celda de la cuadricula dada, en relacion con la Defnorm de todas las demas celdas de la cuadricula con un area basal dentro de 2.5 m<code class="reqn">^{2}</code>/ha de la celda dada</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided from DRYAD repository.
</p>


<h3>References</h3>

<p>-Derek J. N. Young, Jens T. Stevens, J. Mason Earles, Jeffrey Moore, Adam Ellis, Amy L. Jirka, and Andrew M. Latimer.
Long-term climate and competition explain forest mortality patterns under extreme drought.Ecology Letters, 20(1):78-86, 2017.
</p>
<p>-Salas-Eljatib C, Fuentes-Ramirez A., Gregoire TG, Altamirano A., and Yaitul V. A study on the effects of unbalanced data when
fitting logistic regression models in ecology. Ecological Indicators, 85:502-508, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(deadForestCA2)    
head(deadForestCA2) 
</code></pre>

<hr>
<h2 id='deadLianas'>This dataset has 43 columns and 4247 rows. Each row corresponds to an epiphyte individual located on the 
reliable sections of the host trees</h2><span id='topic+deadLianas'></span>

<h3>Description</h3>

<p>This study is part of the project &quot;Diversity and dynamics of vascular epiphytes in Colombian Andes&quot; 
supported by COLCIENCIAS (contract 2115-2013). The data corresponds to the first large-scale 
assessment of vascular epiphyte mortality in the neotropics. Based on two consecutive annual surveys,
we followed the fate of 4247 epiphytes to estimate the epiphyte mortality rate on 116 host trees
at nine sites. Additional variables were taken from the area of study in order to find relationships
with epiphyte mortality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(deadLianas)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>PlotSite</dt><dd><p>Municipality name of the 9 study sites</p>
</dd>					
<dt>Y.Plot</dt><dd><p>Latitude of the plot in decimal degrees</p>
</dd>
<dt>X.Plot</dt><dd><p>Longitude of the plot in decimal degrees</p>
</dd>
<dt>PhoroNo</dt><dd><p>ID number of the sampled host trees in each site</p>
</dd>
<dt>EpiFam</dt><dd><p>Epiphyte taxonomic family</p>
</dd>
<dt>EpiGen</dt><dd><p>Epiphyte taxonomic genus</p>
</dd>
<dt>cf.aff</dt><dd><p>Abbreviations of Latin terms in the context of taxonomy. cf. &quot;confer&quot; meaning &quot;compare with&quot;. aff.: &quot;affinis&quot; meaning &quot;similar to&quot;.</p>
</dd>
<dt>Species</dt><dd><p>Epiphyte (morpho) species name</p>
</dd>
<dt>Author</dt><dd><p>Author of the scientific name</p>
</dd>
<dt>EpiAzi</dt><dd><p>Azimuth of the epiphyte individual on each host tree</p>
</dd>
<dt>BraAzi</dt><dd><p>Azimuth of the branch in which the epiphyte individual was found</p>
</dd>
<dt>EpiDisTru</dt><dd><p>Distance in meters from the trunk to the epiphyte attachment site on a branch</p>
</dd>
<dt>EpiSize</dt><dd><p>Estimated size of the epiphyte individual in centimetres</p>
</dd>
<dt>EpiAttHei</dt><dd><p>Epiphyte attachment height in meters</p>
</dd>
<dt>Date0</dt><dd><p>Date of the first census</p>
</dd>
<dt>Date1</dt><dd><p>Date of the final census</p>
</dd>
<dt>Location</dt><dd><p>Section (roots, trunks, branches) of the host tree in which theepiphyte individual was found</p>
</dd>
<dt>Mortality</dt><dd><p>Dichotomous variable. 0 if the epiphyte individual was dead in the final census and 1 if otherwise</p>
</dd>
<dt>MorCat</dt><dd><p>Mechanical or non-mechanical cause of mortality</p>
</dd>
<dt>Elevation</dt><dd><p>Elevation (m a.s.l.) of the plot</p>
</dd>
<dt>AP_bio12</dt><dd><p>Annual precipitation in the plot (mm yr-1)</p>
</dd>
<dt>PDM_bio14</dt><dd><p>Precipitation of driest month in the plot (mm)</p>
</dd>
<dt>PS_bio15</dt><dd><p>Precipitation seasonality in the plot (coefficient of variation)</p>
</dd>
<dt>MDT_bio2</dt><dd><p>Mean Diurnal Range (Mean of monthly (max temp - min temp)) in the plot (oC*10)</p>
</dd>
<dt>TS_bio4</dt><dd><p>Temperature seasonality in the plot (standard deviation*100)</p>
</dd>
<dt>ATR_bio7</dt><dd><p>Annual temperature range in the plot (10 celsius degrees)</p>
</dd>
<dt>AET</dt><dd><p>Actual evapotranspiration in the plot (mm yr-1)</p>
</dd>
<dt>BasAre</dt><dd><p>Basal area of trees with DBH major or equal to 5 cm (AB) in the plot (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>BasAre5_10</dt><dd><p>Basal area of trees with greater or equal than 5  DBH and  less than 10 cm in the plot (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>BasAre10</dt><dd><p>Basal area of trees with greater or equal than 10 cm DBH in the plot (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>Ind10</dt><dd><p>Number of canopy trees (with greater or equal than 10 cm DBH ) in the plot</p>
</dd>
<dt>Ind5</dt><dd><p>Number of understory trees (with greater or equal than 5  DBH and  less than 10 cm) in the plot</p>
</dd>
<dt>Ind5_10</dt><dd><p>Number of trees with greater or equal than 5  DBH and  less than 10 cm in the plot</p>
</dd>
<dt>Ind10_15</dt><dd><p>Number of trees with greater or equal than 10  DBH and  less than 15 cm in the plot</p>
</dd>
<dt>Ind15_20</dt><dd><p>Number of trees with greater or equal than 15  DBH and  less than 20 cm in the plot</p>
</dd>
<dt>Ind20_25</dt><dd><p>Number of trees with greater or equal than 20  DBH and  less than 25 cm in the plot</p>
</dd>
<dt>Ind25_30</dt><dd><p>Number of trees with greater or equal than 25  DBH and  less than 30 cm in the plot</p>
</dd>
<dt>Ind30</dt><dd><p>Number of trees with DBH major or equal to 30 cm in the plot</p>
</dd>
<dt>TreeHei</dt><dd><p>Total tree height in meters</p>
</dd>
<dt>MedHei</dt><dd><p>Median height of trees in each plot</p>
</dd>
<dt>MaxHei</dt><dd><p>Maximum height of trees in each plot</p>
</dd>
<dt>BranchNumb</dt><dd><p>Number of branches of the host tree</p>
</dd>
<dt>Obs</dt><dd><p>Observations and notes in Spanish</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were extracted from Zuleta, D., Benavides, A.M., Lopez-Ros, V. &amp; Duque, A. 2016. Local and regional determinants
of vascular epiphyte mortality in the Andean mountains of Colombia .
</p>


<h3>References</h3>

<p>Zuleta, D., Benavides, A.M., Lopez-Rios, V. &amp; Duque, A. 2016. Local and regional determinants
of vascular epiphyte mortality in the Andean mountains of Colombia.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(deadLianas)    
head(deadLianas) 
</code></pre>

<hr>
<h2 id='deadLianas2'>Este conjunto de datos tiene 43 columnas y 4247 filas. Cada fila corresponde a un individuo epifito ubicado en el
secciones confiables de los árboles hospedantes</h2><span id='topic+deadLianas2'></span>

<h3>Description</h3>

<p>Este estudio es parte del proyecto &quot;Diversidad y din?mica de ep?fitas vasculares en los Andes colombianos&quot;.
apoyado por COLCIENCIAS (contrato 2115-2013). Los datos corresponden a la primera gran escala
evaluaci?n de la mortalidad de ep?fitas vasculares en los neotr?picos. Basado en dos encuestas anuales consecutivas,
Seguimos el destino de 4247 ep?fitas para estimar la tasa de mortalidad de ep?fitas en 116 ?rboles hospedantes.
en nueve sitios. Se tomaron variables adicionales del area de estudio para encontrar relaciones
con mortalidad de epifitas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(deadLianas2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuacion::
</p>

<dl>
<dt>PlotSite</dt><dd><p>Nombre del municipio de los 9 sitios de estudio</p>
</dd>					
<dt>Y.Plot</dt><dd><p>Latitud del grafico en grados decimales</p>
</dd>
<dt>X.Plot</dt><dd><p>Longitud de la grafica en grados decimales</p>
</dd>
<dt>PhoroNo</dt><dd><p>número de identificacion de los árboles hospedantes muestreados en cada sitio</p>
</dd>
<dt>EpiFam</dt><dd><p>Familia taxonomica de epifitas</p>
</dd>
<dt>EpiGen</dt><dd><p>Genero taxonomico de epifitas</p>
</dd>
<dt>cf.aff</dt><dd><p>Abreviaturas de terminos latinos en el contexto de la taxonomia. cf. &quot;conferir&quot; que significa &quot;comparar con&quot;. aff .: &quot;affinis&quot; que significa &quot;similar a&quot;</p>
</dd>
<dt>Species</dt><dd><p>Nombre de la especie epifita (morfo)</p>
</dd>
<dt>Author</dt><dd><p>Autor del nombre cientifico</p>
</dd>
<dt>EpiAzi</dt><dd><p>Azimut del individuo epifito en cada árbol huesped</p>
</dd>
<dt>BraAzi</dt><dd><p>Azimut de la rama en la que se encontro el individuo epifito</p>
</dd>
<dt>EpiDisTru</dt><dd><p>Distancia en metros desde el tronco hasta el sitio de union de la epifita en una rama</p>
</dd>
<dt>EpiSize</dt><dd><p>Tamaño estimado del individuo epifito en centimetros</p>
</dd>
<dt>EpiAttHei</dt><dd><p>Altura del accesorio de la epifita en metros</p>
</dd>
<dt>Date0</dt><dd><p>Fecha del primer censo</p>
</dd>
<dt>Date1</dt><dd><p>Fecha del censo final</p>
</dd>
<dt>Location</dt><dd><p>Seccion (raices, troncos, ramas) del árbol anfitrion en el que se encontro el individuo epifito</p>
</dd>
<dt>Mortality</dt><dd><p>Variable dicotomica. 0 si el individuo epifito estaba muerto en el censo final y 1 si no</p>
</dd>
<dt>MorCat</dt><dd><p>Causa de mortalidad mecanica o no mecanica</p>
</dd>
<dt>Elevation</dt><dd><p>Elevacion (msnm) de la parcela</p>
</dd>
<dt>AP_bio12</dt><dd><p>Precipitacion anual en la parcela (mm año-1)</p>
</dd>
<dt>PDM_bio14</dt><dd><p>Precipitacion del mes mas seco en la parcela (mm)</p>
</dd>
<dt>PS_bio15</dt><dd><p>Estacionalidad de la precipitacion en la parcela (coeficiente de variacion)</p>
</dd>
<dt>MDT_bio2</dt><dd><p>Rango diurno medio (Media mensual (temperatura maxima - temperatura minima)) en la grafica (10 grados celsius)</p>
</dd>
<dt>TS_bio4</dt><dd><p>Estacionalidad de la temperatura en la grafica (desviacion estandar * 100)</p>
</dd>
<dt>ATR_bio7</dt><dd><p>Rango de temperatura anual en la parcela (10 grados centigrados)</p>
</dd>
<dt>AET</dt><dd><p>Evapotranspiracion real en la parcela (mm año-1)</p>
</dd>
<dt>BasAre</dt><dd><p>Area basal de árboles con DAP mayor o igual a 5 cm (AB) en la parcela (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>BasAre5_10</dt><dd><p>Area basal de árboles con DAP mayor o igual a 5 y menor a 10 cm en la parcela (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>BasAre10</dt><dd><p>Area basal de árboles con DAP mayor o igual a 10 cm en la parcela (m<code class="reqn">^{2}</code>/ha)</p>
</dd>
<dt>Ind10</dt><dd><p>Número de árboles del dosel (con un DAP superior o igual a 10 cm) en la parcela</p>
</dd>
<dt>Ind5</dt><dd><p>Número de árboles de sotobosque (con DAP mayor o igual a 5 y menor a 10 cm) en la parcela</p>
</dd>
<dt>Ind5_10</dt><dd><p>Número de árboles con un DAP mayor o igual a 5 y menos de 10 cm en la parcela</p>
</dd>
<dt>Ind10_15</dt><dd><p>Número de árboles con un DAP mayor o igual a 10 y menos de 15 cm en la parcela</p>
</dd>
<dt>Ind15_20</dt><dd><p>Número de árboles con un DAP mayor o igual a 15 y menos de 20 cm en la parcela</p>
</dd>
<dt>Ind20_25</dt><dd><p>Número de árboles con un DAP mayor o igual a 20 y menos de 25 cm en la parcela</p>
</dd>
<dt>Ind25_30</dt><dd><p>Número de árboles con un DAP mayor o igual a 25 y menos de 30 cm en la parcela</p>
</dd>
<dt>Ind30</dt><dd><p>Número de árboles con DAP mayor o igual a 30 cm en la parcela</p>
</dd>
<dt>TreeHei</dt><dd><p>Altura total del árbol en metros</p>
</dd>
<dt>MedHei</dt><dd><p>Altura media de los árboles en cada parcela</p>
</dd>
<dt>MaxHei</dt><dd><p>Altura maxima de los árboles en cada parcela</p>
</dd>
<dt>BranchNumb</dt><dd><p>Número de ramas del árbol anfitrion</p>
</dd>
<dt>Obs</dt><dd><p>Observaciones y notas en español</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data fue extraida desde Zuleta, D., Benavides, A.M., Lopez-Ros, V. &amp; Duque, A. 2016. Local and regional determinants
of vascular epiphyte mortality in the Andean mountains of Colombia .
</p>


<h3>References</h3>

<p>Zuleta, D., Benavides, A.M., Lopez-Rios, V. &amp; Duque, A. 2016. Local and regional determinants
of vascular epiphyte mortality in the Andean mountains of Colombia.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(deadLianas2)    
head(deadLianas2) 
</code></pre>

<hr>
<h2 id='deleteRight'>Deletes the last n-characters of a string</h2><span id='topic+deleteRight'></span>

<h3>Description</h3>

<p>Function to delete the last n-characters of a string from the right-hand side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deleteRight(fac, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deleteRight_+3A_fac">fac</code></td>
<td>
<p>is an object of class string or factor</p>
</td></tr>
<tr><td><code id="deleteRight_+3A_n">n</code></td>
<td>
<p>is the number of characters to be deleted of a the string given in 'fac'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is specially set to arrange data vector having alphanumeric format.
</p>


<h3>Value</h3>

<p>This function returns an object having n-less characters from the right-hand side.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
last.names.id &lt;- c("Stage-1924","Gregoire-1958","Robinson-1967")
deleteRight(last.names.id,5)
deleteRight(last.names.id,4)
</code></pre>

<hr>
<h2 id='demograph'>Contains information of demography of species.</h2><span id='topic+demograph'></span>

<h3>Description</h3>

<p>Dataset contains 61 observations about life histories values for each species and site,
as obtained from the parameterization carried out in studies that used the model SORTIE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(demograph)
</code></pre>


<h3>Format</h3>

<p>Contains 15 variables, as follows:
</p>

<dl>
<dt>sp</dt><dd><p>Name specie.</p>
</dd>
<dt>site</dt><dd><p>Name of site of study.</p>
</dd>
<dt>country</dt><dd><p>Name of country.</p>
</dd>
<dt>site.n</dt><dd><p>Code of site.</p>
</dd>
<dt>code</dt><dd><p>Code of specie.</p>
</dd>
<dt>genus</dt><dd><p>Genus of specie.</p>
</dd>
<dt>sps</dt><dd><p>Abbreviated name specie.</p>
</dd>
<dt>family</dt><dd><p>Family of specie.</p>
</dd>
<dt>phyl</dt><dd><p>Type of phylogeny.</p>
</dd>
<dt>l.hab</dt><dd><p>Type of leaf habit.</p>
</dd>
<dt>l.type</dt><dd><p>.</p>
</dd>
<dt>leaf</dt><dd><p>Type of leaf.</p>
</dd>
<dt>growth.l</dt><dd><p>Growth at full light (time in years).</p>
</dd>
<dt>growth.d</dt><dd><p>Growth in shade.</p>
</dd>
<dt>surv.d</dt><dd><p>Survival in shade.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository.
</p>


<h3>References</h3>

<p>- Ameztegui A, Paquette A, Shipley B, Heym M, Messier C, Gravel D. 2016 . Shade tolerance and 
the functional trait: demography relationship in temperate and boreal forests. Functional Ecology
31: 821-830.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(demograph)    
head(demograph) 
</code></pre>

<hr>
<h2 id='descstat'>Creates a descriptive statistics table for continuous variables</h2><span id='topic+descstat'></span>

<h3>Description</h3>

<p>Function to create a descriptive statistics table for continuous 
variables from a dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descstat(data = data, decnum = 4, full = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="descstat_+3A_data">data</code></td>
<td>
<p>a dataframe containing numeric variables as columns.</p>
</td></tr>
<tr><td><code id="descstat_+3A_decnum">decnum</code></td>
<td>
<p>the number of decimals to be used in the output.</p>
</td></tr>
<tr><td><code id="descstat_+3A_full">full</code></td>
<td>
<p>TRUE for a longer output (i.e. more descriptive statistics). The default is to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The resulting table offers the main central and dispersion statistics.
</p>


<h3>Value</h3>

<p>This function wraps descriptive statistics into a summarize table having the following
descriptive statistics: sample size, minimum, maximum, mean, median, SD, and coefficient of variation. 
If the full option is set to TRUE, the following statistics are added to the table: 25th and 75th 
percentiles, the interquartile range, skewness, and kurtosis.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib and Tomas Cayul.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#creating a ficticiuos dataframe
set.seed(1234)
df &lt;- as.data.frame(cbind(variable1=rnorm(5, 0), variable2=rnorm(5, 2)))
## adding one missing value
df[3,1] &lt;- NA
df
#' #using the function
descstat(data=df)
descstat(data=df,decnum=1)
descstat(df,2)
</code></pre>

<hr>
<h2 id='election'>Presidential election data of Florida (USA) in 2000.</h2><span id='topic+election'></span>

<h3>Description</h3>

<p>County-by-county vote for president in Florida in 2000 for Bush, Gore and Buchanan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(election)
</code></pre>


<h3>Format</h3>

<p>Contains three variables, as follows:
</p>

<dl>
<dt>gore</dt><dd><p>Vote for Gore.</p>
</dd>
<dt>bush</dt><dd><p>Vote for Bush.</p>
</dd>
<dt>buchanan</dt><dd><p>Vote for Pat Buchanan.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the 'alr4' library.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. Hoboken NJ: Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(election)    
head(election) 
</code></pre>

<hr>
<h2 id='election2'>Elección presidencial en el estado de Florida (USA) en el 2000.</h2><span id='topic+election2'></span>

<h3>Description</h3>

<p>Conteo de votos a nivel de condado en el estado de Florida, año 2000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(election2)
</code></pre>


<h3>Format</h3>

<p>Contiene las siguientes tres columnas:
</p>

<dl>
<dt>gore</dt><dd><p>Votos para Gore. Número de votos para Al Gore.</p>
</dd>
<dt>bush</dt><dd><p>Votos para Bush. Número de votos para George W. Bush.</p>
</dd>
<dt>buchanan</dt><dd><p>Votos para Buchaman. Número de votos para Pat Buchanan.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos se obtuvieron desde el paquete 'alr4' de R.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. Hoboken NJ: Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(election2)    
head(election2) 
</code></pre>

<hr>
<h2 id='eucaleaf'>Leaf measurements for Eucalyptus nitens trees in Tasmania, Australia.</h2><span id='topic+eucaleaf'></span>

<h3>Description</h3>

<p>The length, width, and area of Eucalyptus nitens leaves were measured.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaleaf)
</code></pre>


<h3>Format</h3>

<p>Contains leaf-level variables, as follows:
</p>

<dl>
<dt>time</dt><dd><p>Early or Late</p>
</dd>
<dt>tree</dt><dd><p>an identificator for a given sample tree</p>
</dd>
<dt>shoot</dt><dd><p>shoot description</p>
</dd>
<dt>l</dt><dd><p>length of the leaf, in mm</p>
</dd>
<dt>w</dt><dd><p>width of the leaf, in mm</p>
</dd>
<dt>la</dt><dd><p>leaf area, in cm<code class="reqn">^{2}</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Although the original source of the measurements is the Dissertation of Dr Candy (1999), 
the data file used here was courtesy of Prof. Timothy Gregoire at
Yale University (New Haven, CT, USA). Furthermore, these data were used by
Gregoire and Salas (2009).
</p>


<h3>References</h3>

<p>- Candy SG. 1999. Predictive models for integrated pest management of the
leaf beetle Chrysophtharta bimaculata in Eucalyptus nitens in Tasmania. Doctoral dissertation,
University of Tasmania, Hobart, Australia.
</p>
<p>- Gregoire TG, and Salas C. 2009. Ratio estimation with measurement
error in the auxiliary variate. Biometrics 65(2):590-598 <a href="https://doi.org/10.1111/j.1541-0420.2008.01110.x">doi:10.1111/j.1541-0420.2008.01110.x</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaleaf)    
head(eucaleaf) 
</code></pre>

<hr>
<h2 id='eucaleaf2'>Mediciones foliares para árboles de Eucalyptus nitens en Tasmania, Australia.</h2><span id='topic+eucaleaf2'></span>

<h3>Description</h3>

<p>Mediciones de largo, ancho y area de hojas de Eucalyptus nitens.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaleaf)
</code></pre>


<h3>Format</h3>

<p>Contiene variables a nivel de hoja, como sigue:
</p>

<dl>
<dt>tiempo</dt><dd><p>factor a dos niveles: Temprano o Tardio</p>
</dd>
<dt>arbol</dt><dd><p>un identificador del árbol muestra</p>
</dd>
<dt>meristema</dt><dd><p>descripcion del meristema</p>
</dd>
<dt>largo</dt><dd><p>largo de la hoja, en mm</p>
</dd>
<dt>ancho</dt><dd><p>ancho de la hoja, en mm</p>
</dd>
<dt>area</dt><dd><p>area foliar, en cm<code class="reqn">^{2}</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Aunque la fuente original de estas mediciones proviene de la tesis del Dr Candy (1999), 
el archivo de datos fue cortesia del Prof. Timothy Gregoire de
Yale University (New Haven, CT, USA). Ademas, estos datos fueron ocupados en
el estudio de Gregoire y Salas (2009).
</p>


<h3>References</h3>

<p>- Candy SG. 1999. Predictive models for integrated pest management of the
leaf beetle Chrysophtharta bimaculata in Eucalyptus nitens in Tasmania. Doctoral dissertation,
University of Tasmania, Hobart, Australia.
</p>
<p>- Gregoire TG, and Salas C. 2009. Ratio estimation with measurement
error in the auxiliary variate. Biometrics 65(2):590-598 <a href="https://doi.org/10.1111/j.1541-0420.2008.01110.x">doi:10.1111/j.1541-0420.2008.01110.x</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaleaf2)    
head(eucaleaf2) 
</code></pre>

<hr>
<h2 id='eucaleafAll'>Leaf measurements (all, n=744) for Eucalyptus nitens trees in Tasmania, Australia.</h2><span id='topic+eucaleafAll'></span>

<h3>Description</h3>

<p>The length, width, and area of Eucalyptus nitens leaves were measured for all the samples
of Candy (1999).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaleafAll)
</code></pre>


<h3>Format</h3>

<p>Contains leaf-level variables, as follows:
</p>

<dl>
<dt>time</dt><dd><p>Early or Late</p>
</dd>
<dt>tree</dt><dd><p>an identificator for a given sample tree</p>
</dd>
<dt>shoot</dt><dd><p>shoot description</p>
</dd>
<dt>l</dt><dd><p>length of the leaf, in mm</p>
</dd>
<dt>w</dt><dd><p>width of the leaf, in mm</p>
</dd>
<dt>la</dt><dd><p>leaf area, in cm<code class="reqn">^{2}</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Although the original source of the measurements is the Dissertation of Dr Candy (1999), 
the data file used here was courtesy of Prof. Timothy Gregoire at
Yale University (New Haven, CT, USA). Furthermore, these data were used by
Gregoire and Salas (2009).
</p>


<h3>References</h3>

<p>- Candy SG. 1999. Predictive models for integrated pest management of the
leaf beetle Chrysophtharta bimaculata in Eucalyptus nitens in Tasmania. Doctoral dissertation,
University of Tasmania, Hobart, Australia.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaleafAll)    
head(eucaleafAll) 
</code></pre>

<hr>
<h2 id='eucaleafAll2'>Mediciones foliares (todas, n=744) para árboles de Eucalyptus nitens en Tasmania, Australia.</h2><span id='topic+eucaleafAll2'></span>

<h3>Description</h3>

<p>Mediciones de largo, ancho y area de hojas de Eucalyptus nitens para toda la muestra de Candy (1999).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaleafAll2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables a nivel de hoja, como sigue:
</p>

<dl>
<dt>tiempo</dt><dd><p>factor a dos niveles: Temprano o Tardio</p>
</dd>
<dt>arbol</dt><dd><p>un identificador del árbol muestra</p>
</dd>
<dt>meristema</dt><dd><p>descripcion del meristema</p>
</dd>
<dt>largo</dt><dd><p>largo de la hoja, en mm</p>
</dd>
<dt>ancho</dt><dd><p>ancho de la hoja, en mm</p>
</dd>
<dt>area</dt><dd><p>area foliar, en cm<code class="reqn">^{2}</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Aunque la fuente original de estas mediciones proviene de la tesis del Dr. Candy (1999), 
el archivo de datos fue cortesia del Prof. Timothy Gregoire de
Yale University (New Haven, CT, USA).
</p>


<h3>References</h3>

<p>- Candy SG. 1999. Predictive models for integrated pest management of the
leaf beetle Chrysophtharta bimaculata in Eucalyptus nitens in Tasmania. Doctoral dissertation,
University of Tasmania, Hobart, Australia.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaleafAll2)    
head(eucaleafAll2) 
</code></pre>

<hr>
<h2 id='eucaplot'>Data from a Eucalyptus globulus plantation near Gorbea, Region de La Araucania, Chile.</h2><span id='topic+eucaplot'></span>

<h3>Description</h3>

<p>Tree-level data collected within a sample plot in a forestry plantation of Eucalyptus globulus near Gorbea, 
Southern Chile. The plot size is 500 m<code class="reqn">^{2}</code>. The plantation is 15 yr-old and
had been subject to three thinnings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaplot)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains four variables as follows:
</p>

<dl>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>health</dt><dd><p>health status (1: good, 2: medium, 3: bad).</p>
</dd>
<dt>shape</dt><dd><p>stem shape for timber purposes (1: good, 2: medium, 3: bad).</p>
</dd>
<dt>crown.class</dt><dd><p>Crown class (1: superior, 2: intermedium, 3: lower).</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided courtesy of Dr Christian Salas (Universidad de Chile, Santiago, Chile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaplot)    
head(eucaplot) 
</code></pre>

<hr>
<h2 id='eucaplot2'>Árboles dentro de parcelas de muestreo en una plantación de Eucalyptus globulus, Chile.</h2><span id='topic+eucaplot2'></span>

<h3>Description</h3>

<p>Datos a nivel de árbol colectados dentro de una parcela de muestreo en una plantación
forestal de Eucalyptus globulus cerca de Gorbea, Sur de Chile. La superficie de la parcela es de 500 m<code class="reqn">^{2}</code>.
La plantación tiene 15 años de edad y ha estado sujeta a tres raleos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eucaplot2)
</code></pre>


<h3>Format</h3>

<p>Los datos contienen las siguientes cuatro columnas:
</p>

<dl>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>sanidad</dt><dd><p>Evaluación cualitativa de la sanidad del árbol (1: buena, 2: media, 3: mala).</p>
</dd>
<dt>forma</dt><dd><p>Evaluación cualtitativa de la forma del fuste (1: buena, 2: media, 3: mala).</p>
</dd>
<dt>clase.copa</dt><dd><p>Clase de copa (1: superior, 2: intermedio, 3: inferior).</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Prof. Christian Salas (Universidad de Chile, Santiago, Chile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eucaplot2)    
head(eucaplot2) 
</code></pre>

<hr>
<h2 id='fertiliza'>Fertilization experiment data.</h2><span id='topic+fertiliza'></span>

<h3>Description</h3>

<p>Data contains volume data at plot-level for a fertilization
experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fertiliza)
</code></pre>


<h3>Format</h3>

<p>Contains three variables, as follows:
</p>

<dl>
<dt>treat</dt><dd><p>Treatment level.</p>
</dd>
<dt>volume</dt><dd><p>Plot-level volume, in m<code class="reqn">^{3}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided by Dr Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>not yet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fertiliza)    
head(fertiliza) 
levels(fertiliza$treat)
means.g &lt;- tapply(fertiliza$volume,fertiliza$treat,mean);means.g
sds.g &lt;- tapply(fertiliza$volume,fertiliza$treat,sd);sds.g
ns.g &lt;- tapply(fertiliza$volume,fertiliza$treat,length);ns.g

</code></pre>

<hr>
<h2 id='fertiliza2'>Datos a nivel de parcela de un experimento de fertilización</h2><span id='topic+fertiliza2'></span>

<h3>Description</h3>

<p>Datos a nivel de parcela de un experimento de fertilización con tratamientos
y replicas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fertiliza2)
</code></pre>


<h3>Format</h3>

<p>Contiene tres columnas como sigue:
</p>

<dl>
<dt>tmo</dt><dd><p>Tratamiento.Factor medido en diferentes niveles.</p>
</dd>
<dt>vol</dt><dd><p>Volumen de madera en la parcela experimental, en m<code class="reqn">^{3}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos cedidos por el Prof. Christian Salas.
</p>


<h3>References</h3>

<p>not yet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fertiliza2)    
head(fertiliza2) 
levels(fertiliza$tmo)
media.g &lt;- tapply(fertiliza2$vol,fertiliza2$tmo,mean);media.g
desvst.g &lt;- tapply(fertiliza2$vol,fertiliza2$tmo,sd);desvst.g
n.g &lt;- tapply(fertiliza2$vol,fertiliza2$tmo,length);n.g

</code></pre>

<hr>
<h2 id='ficdiamgr'>Diameter growth of trees</h2><span id='topic+ficdiamgr'></span>

<h3>Description</h3>

<p>The 'ficdiamgr' is a fictitious dataframe built to show the structure of 
longitudinal data. The dataframe has records of tree diameter growth of 
five sample trees, spanning three species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ficdiamgr)
</code></pre>


<h3>Format</h3>

<p>A time series data containing the following columns:
</p>

<dl>
<dt>tree.id</dt><dd><p>an ordered factor indicating the tree on which the measurement is made. The ordering is according to increasing maximum diameter.</p>
</dd>
<dt>time</dt><dd><p>a numeric vector giving the numbers of days since establishment.</p>
</dd>
<dt>dbh</dt><dd><p>a numeric vector of diameter at breast height, in cm.</p>
</dd>
<dt>site</dt><dd><p>a factor variable, representing site conditions with two levels.</p>
</dd>
<dt>spp</dt><dd><p>a factor variable, representing tree species with three levels.</p>
</dd>
</dl>



<h3>Source</h3>

<p>This dataframe was built from the 'Orange' data of the 'datasets' package,
by Christian Salas-Eljatib.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ficdiamgr)

coplot(dbh ~ time | tree, data = ficdiamgr, show.given = FALSE)
</code></pre>

<hr>
<h2 id='ficdiamgr2'>Crecimiento diametral de árboles</h2><span id='topic+ficdiamgr2'></span>

<h3>Description</h3>

<p>Los datos 'ficdiamgr2' son ficticios, y fue construida para mostrar la estructura 
de datos longitudinales. Los datos tienen registro de crecimiento en cinco 
árboles muestra, representando a tres especies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ficdiamgr2)
</code></pre>


<h3>Format</h3>

<p>Una serie de tiempo conteniendo las siguientes columnas:
</p>

<dl>
<dt>arbol</dt><dd><p>indica el identificador del árbol.</p>
</dd>
<dt>tiempo</dt><dd><p>número de dias desde el inicio de las mediciones.</p>
</dd>
<dt>dap</dt><dd><p>diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>sitio</dt><dd><p>un factor, representando condiciones de sitio, en dos niveles.</p>
</dd>
<dt>espe</dt><dd><p>un factor, representando especie del árbol, en tres niveles.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Estos datos fueron modificados desde la dataframe 'Orange' de la librería
'datasets', por Christian Salas-Eljatib.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ficdiamgr2)

coplot(dap ~ tiempo | arbol, data = ficdiamgr2, show.given = FALSE)
</code></pre>

<hr>
<h2 id='findColumn.byname'>Finds the position of a specific variable.</h2><span id='topic+findColumn.byname'></span>

<h3>Description</h3>

<p>Sometimes in data manipulation we face the task of locating the position
of a specific variable within a dataframe.  The function finds the position in 
which a column name is within an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findColumn.byname(data = data, col.name = col.name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findColumn.byname_+3A_data">data</code></td>
<td>
<p>is a dataframe</p>
</td></tr>
<tr><td><code id="findColumn.byname_+3A_col.name">col.name</code></td>
<td>
<p>is a string specifying the name of the variable</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Although the function finds the position of a specific variable, can also 
be used for more than one variable.
</p>


<h3>Value</h3>

<p>This function returns the number of a specific column-name.
</p>


<h3>Note</h3>

<p>It can be used for a vector of specified column-names as well.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
df &lt;- data.frame(varX=1:5, varY=letters[1:5], varZ=rep("a",5), varK=rep("b",5))
df
#using the function
findColumn.byname(df, c("varY","varZ"))
findColumn.byname(df, "varK")
#Creating an example vector
vector &lt;- letters
vector
findColumn.byname(vector, c("h","z"))
</code></pre>

<hr>
<h2 id='fishgrowth'>Data on fish growth.</h2><span id='topic+fishgrowth'></span>

<h3>Description</h3>

<p>Data on samples of small mouth bass collected in West Bearskin Lake, Minnesota, in 1991. The
file wblake includes only fish of ages 8 or younger.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fishgrowth)
</code></pre>


<h3>Format</h3>

<p>Contains 3 variables, as follows:
</p>

<dl>
<dt>years</dt><dd><p>Year at capture.</p>
</dd>
<dt>length</dt><dd><p>Length at capture (mm).</p>
</dd>
<dt>scale</dt><dd><p>radius of a key scale (mm).</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the alr4 library of R.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. Hoboken NJ: Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fishgrowth)    
head(fishgrowth) 
</code></pre>

<hr>
<h2 id='fishgrowth2'>Crecimiento de peces</h2><span id='topic+fishgrowth2'></span>

<h3>Description</h3>

<p>Data on samples of small mouth bass collected in West Bearskin Lake, Minnesota, in 1991. The
file wblake includes only fish of ages 8 or younger.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fishgrowth2)
</code></pre>


<h3>Format</h3>

<p>Contiene tres variables, como sigue:
</p>

<dl>
<dt>edad</dt><dd><p>Year at capture.</p>
</dd>
<dt>largo</dt><dd><p>Length at capture, en mm.</p>
</dd>
<dt>escala</dt><dd><p>radius of a key scale, en mm.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos obtenidos desde el paquete alr4 de R.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. Hoboken NJ: Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fishgrowth2)    
head(fishgrowth2) 
</code></pre>

<hr>
<h2 id='floraChile'>Flora of Chile.</h2><span id='topic+floraChile'></span>

<h3>Description</h3>

<p>Dataset contains taxonomic level information
segregatted by latitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(floraChile)
</code></pre>


<h3>Format</h3>

<p>Contains seven columns, as follows:
</p>

<dl>
<dt>family</dt><dd><p>.</p>
</dd>
<dt>genus</dt><dd><p>.</p>
</dd>
<dt>scientific.name</dt><dd><p>.</p>
</dd>
<dt>author</dt><dd><p>.</p>
</dd>
<dt>origin</dt><dd><p>.</p>
</dd>
<dt>life.form</dt><dd><p>.</p>
</dd>
<dt>lat...</dt><dd><p>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Dr Jan Bannister at
the Instituto Forestal (Chiloe, Chile).
</p>


<h3>References</h3>

<p>- Bannister JR, Vidal OJ, Teneb E, Sandoval V. 2012. Latitudinal patterns
and regionalization of plant diversity along a 4270‐km gradient in continental
Chile. Austral Ecology 37(4):500-509.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(floraChile)    
head(floraChile) 
</code></pre>

<hr>
<h2 id='floraChile2'>Flora de Chile.</h2><span id='topic+floraChile2'></span>

<h3>Description</h3>

<p>Contiene informacion taxonomica segregada por latitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(floraChile)
</code></pre>


<h3>Format</h3>

<p>Contains seven columns, as follows:
</p>

<dl>
<dt>family</dt><dd><p>.</p>
</dd>
<dt>genus</dt><dd><p>.</p>
</dd>
<dt>scientific.name</dt><dd><p>.</p>
</dd>
<dt>author</dt><dd><p>.</p>
</dd>
<dt>origin</dt><dd><p>.</p>
</dd>
<dt>life.form</dt><dd><p>.</p>
</dd>
<dt>lat...</dt><dd><p>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos cedidos por el  Dr Jan Bannister del Instituto Forestal (Chiloe, Chile).
</p>


<h3>References</h3>

<p>- Bannister JR, Vidal OJ, Teneb E, Sandoval V. 2012. Latitudinal patterns
and regionalization of plant diversity along a 4270‐km gradient in continental
Chile. Austral Ecology 37(4):500-509.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(floraChile2)    
head(floraChile2) 
</code></pre>

<hr>
<h2 id='football'>Anaerobic potential of soccer players.</h2><span id='topic+football'></span>

<h3>Description</h3>

<p>Data about anaerobic variables of football players.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(football)
</code></pre>


<h3>Format</h3>

<p>The data frame contains 13 variables as follows:
</p>

<dl>
<dt>WPM</dt><dd></dd>											
<dt>WPMk</dt><dd></dd>
<dt>WPm</dt><dd></dd>
<dt>WPmk</dt><dd></dd>
<dt>WTT</dt><dd></dd>
<dt>WTTk</dt><dd></dd>
<dt>WIF</dt><dd></dd>
<dt>W5</dt><dd></dd>
<dt>W10</dt><dd></dd>
<dt>W15</dt><dd></dd>
<dt>W20</dt><dd></dd>
<dt>W25</dt><dd></dd>
<dt>W30</dt><dd></dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Aquiles Yanez-Silva at Universidad Mayor (Santiago, Chile).
</p>


<h3>References</h3>

<p>Not yet.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(football)    
head(football) 
</code></pre>

<hr>
<h2 id='football2'>Potencia anaerobica de jugadores de football.</h2><span id='topic+football2'></span>

<h3>Description</h3>

<p>Datos sobre potencia anaerobica de jugadores de football.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(football2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables de nivel individual, como se describen a continuacion::
</p>

<dl>
<dt>WPM</dt><dd></dd>											
<dt>WPMk</dt><dd></dd>
<dt>WPm</dt><dd></dd>
<dt>WPmk</dt><dd></dd>
<dt>WTT</dt><dd></dd>
<dt>WTTk</dt><dd></dd>
<dt>WIF</dt><dd></dd>
<dt>W5</dt><dd></dd>
<dt>W10</dt><dd></dd>
<dt>W15</dt><dd></dd>
<dt>W20</dt><dd></dd>
<dt>W25</dt><dd></dd>
<dt>W30</dt><dd></dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Dr Aquiles Yanez-Silva de la Universidad Mayor (Santiago, Chile).
</p>


<h3>References</h3>

<p>Not yet.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(football2)    
head(football2) 
</code></pre>

<hr>
<h2 id='forestFire'>Data of forest fire occurrence</h2><span id='topic+forestFire'></span>

<h3>Description</h3>

<p>Data of forest fire occurrence from Altamirano et al. (2013) as our population, containing 7210 total observations (N), with only 
890 cases of fire occurrence (N 1 ) and 6320 cases of non occurrence (N 0 ). The binary variable (Y) is the occurrence of forest 
fire, where Y equal to 1 denotes occurrence and Y equal to 0  otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(forestFire)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>fire</dt><dd><p>Presence of forest fire (1 yes, 0 no)</p>
</dd>											
<dt>xcoord</dt><dd><p>Geographic coordinate x.utm</p>
</dd>
<dt>ycoord</dt><dd><p>Geographic coordinate y.utm</p>
</dd>
<dt>aspect</dt><dd><p>Exposure (degrees from north)</p>
</dd>
<dt>eleva</dt><dd><p>Elevation (m)</p>
</dd>
<dt>slope</dt><dd><p>Slope (degrees)</p>
</dd>
<dt>distr</dt><dd><p>Distance to dirt roads</p>
</dd>
<dt>distcity</dt><dd><p>Distance to cities</p>
</dd>
<dt>distriver</dt><dd><p>Distance to paved roads</p>
</dd>
<dt>covera</dt><dd><p>Land use classifications according to a polygon</p>
</dd>
<dt>coverb</dt><dd><p>Land use classifications according to a polygon</p>
</dd>
<dt>tempe</dt><dd><p>Minimum temperature of the coldest month</p>
</dd>
<dt>ppan</dt><dd><p>Annual precipitation</p>
</dd>
<dt>ndii</dt><dd><p>Normalized difference infrared index</p>
</dd>
<dt>nvdi</dt><dd><p>Normalized difference vegetation index</p>
</dd>
<dt>tempe2</dt><dd><p>Minimum temperature of the warmest month</p>
</dd>
<dt>ppan2</dt><dd><p>Precipitation of the driest month</p>
</dd>
<dt>frec.fire</dt><dd><p>Frequency of fires</p>
</dd>
<dt>perc.fire</dt><dd><p>Percentage of fire frequency</p>
</dd>
<dt>fireClass</dt><dd><p>Class for frecuency fire</p>
</dd>
<dt>asp.class</dt><dd><p>Class of variable exposure</p>
</dd>
<dt>eleva.class</dt><dd><p>Class of numerical variable elevation</p>
</dd>
<dt>slope.class</dt><dd><p>Class of numerical variable slope</p>
</dd>
<dt>ndii.class</dt><dd><p>Normalized difference infrared index class</p>
</dd>
<dt>nvdi.class</dt><dd><p>Normalized difference vegetation index class</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Adison Altamirano at the Universidad de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Altamirano A, Salas C, Yaitul V, Smith-Ramirez C, Avila A. Infuencia de la heterogeneidad 
del paisaje en la ocurrencia de incendios forestales en Chile Central. Revista de Geografia del Norte Grande, 55:157-170, 2013.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(forestFire)    
head(forestFire) 
</code></pre>

<hr>
<h2 id='forestFire2'>Datos de ocurrencia de incendios forestales</h2><span id='topic+forestFire2'></span>

<h3>Description</h3>

<p>Datos de ocurrencia de incendios forestales de Altamirano et al. (2013) como nuestra poblacion, que contiene 7210 observaciones
totales (N), con solo 890 casos de ocurrencia de incendios (N1) y 6320 casos de no ocurrencia (N0). La variable binaria (Y)
es la ocurrencia de un incendio forestal, donde Y igual a 1 denota ocurrencia e igual a 0 en caso contrario.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(forestFire2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuacion:
</p>

<dl>
<dt>fire</dt><dd><p>Presencia de incendio forestal (1 si, 0 no)</p>
</dd>											
<dt>xcoord</dt><dd><p>Coordenada geografica x.utm</p>
</dd>
<dt>ycoord</dt><dd><p>Coordenada geografica y.utm</p>
</dd>
<dt>aspect</dt><dd><p>Exposicion (grados desde el norte)</p>
</dd>
<dt>eleva</dt><dd><p>Elevacion (m)</p>
</dd>
<dt>slope</dt><dd><p>Pendiente (grados)</p>
</dd>
<dt>distr</dt><dd><p>Distancia a caminos de tierra</p>
</dd>
<dt>distcity</dt><dd><p>Distancia a ciudades</p>
</dd>
<dt>distriver</dt><dd><p>Distancia a caminos pavimentados</p>
</dd>
<dt>covera</dt><dd><p>Clasificaciones de uso del suelo segun un poligono</p>
</dd>
<dt>coverb</dt><dd><p>Clasificaciones de uso del suelo segun un poligono</p>
</dd>
<dt>tempe</dt><dd><p>Temperatura m?nima del mes m?s frio</p>
</dd>
<dt>ppan</dt><dd><p>Precipitacion anual</p>
</dd>
<dt>ndii</dt><dd><p>Indice infrarrojo de diferencia normalizado</p>
</dd>
<dt>nvdi</dt><dd><p>Indice de vegetacion de diferencia normalizado</p>
</dd>
<dt>tempe2</dt><dd><p>Temperatura m?nima del mes mas calido</p>
</dd>
<dt>ppan2</dt><dd><p>Precipitacion del mes mas seco</p>
</dd>
<dt>frec.fire</dt><dd><p>Frecuencia de incendios</p>
</dd>
<dt>perc.fire</dt><dd><p>Porcentajede la frecuencia de incendios</p>
</dd>
<dt>fireClass</dt><dd><p>Clase para variable frecuencia de incendio</p>
</dd>
<dt>asp.class</dt><dd><p>Clase de variable exposicion </p>
</dd>
<dt>eleva.class</dt><dd><p>Clase de variable numerica elevacion</p>
</dd>
<dt>slope.class</dt><dd><p>Clase de variable numerica pendiente</p>
</dd>
<dt>ndii.class</dt><dd><p>Clase de indice infrarrojo de diferencia normalizado</p>
</dd>
<dt>nvdi.class</dt><dd><p>Clase de indice de vegetacion de diferencia normalizado</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos fueron cedidos por el Dr Christian Salas-Eljatib (Santiago, Chile).
</p>


<h3>References</h3>

<p>Altamirano A, Salas C, Yaitul V, Smith-Ramirez C, and Avila A. Infuencia de la heterogeneidad 
del paisaje en la ocurrencia de incendios forestales en Chile Central. Revista de Geografia del Norte Grande, 55:157-170, 2013.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(forestFire2)    
head(forestFire2) 
</code></pre>

<hr>
<h2 id='forestHawaii'>Contains information of forest plots across the Hawaiian archipelago.</h2><span id='topic+forestHawaii'></span>

<h3>Description</h3>

<p>Diameter at breast height (or occurrence) of individual trees, shrubs and tree
ferns across 530 plots across the Hawaiian archipelago and includes native
status and cultivated status of the 185 species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(forestHawaii)
</code></pre>


<h3>Format</h3>

<p>Contains 18 variables, as follows:
</p>

<dl>
<dt>island</dt><dd><p>Island name.</p>
</dd>
<dt>plot.id</dt><dd><p>Unique numeric identifier for each plot.</p>
</dd>
<dt>study</dt><dd><p>Brief name of study.</p>
</dd>
<dt>plot.area</dt><dd><p>Plot area in m<code class="reqn">^{2}</code>.</p>
</dd>
<dt>longitude</dt><dd><p>Longitude of plot in decimal degrees; WGS84 coordinate system.</p>
</dd>
<dt>latitude</dt><dd><p>Latitude of plot in decimal degrees; WGS84 coordinate system.</p>
</dd>
<dt>year</dt><dd><p>Year in which plot data was collected.</p>
</dd>
<dt>census</dt><dd><p>Numeric identifier for each census.</p>
</dd>
<dt>tree.id</dt><dd><p>Unique numeric identifier for each individual.</p>
</dd>
<dt>scientific.name</dt><dd><p>Genus and species of each individual following TPL v. 1.1.</p>
</dd>
<dt>family</dt><dd><p>Family of each individual following TPL v. 1.1.</p>
</dd>
<dt>angiosperm</dt><dd><p>Binary variable (1 = yes, 0 = no) indicating whether an individual is classified as an angiosperm following APG III.</p>
</dd>
<dt>monocot</dt><dd><p>Binary variable (1 = yes, 0 = no) indicating whether an individual is classified as a monocot following APG III.</p>
</dd>
<dt>native.status</dt><dd><p>Categorical variable (&quot;native&quot;, &quot;alien&quot;, &quot;uncertain&quot;) indicating alien status of each individual following Wagner et al. (2005).</p>
</dd>
<dt>cultivated.status</dt><dd><p>Binary variable (1 = yes, 0 = no, NA = not applicable) indicating if species is cultivated following PIER.</p>
</dd>
<dt>abundance</dt><dd><p>Number of individuals (all = 1).</p>
</dd>
<dt>abundance.ha</dt><dd><p>Abundance of each individual on a per hectare basis.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at 1.3 m (in cm) for each individual; NA indicates that size was not measured, but was classified by size class.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.1kk02qr">doi:10.5061/dryad.1kk02qr</a>.
</p>


<h3>References</h3>

<p>- Craven D, Knight T, Barton K, Bialic-Murphy L, Cordell S, Giardina C, Gillespie T, Ostertag R, Sack L,Chase J. 2018. 
OpenNahele: the open Hawaiian forest plot database. Biodiversity Data Journal 6: e28406.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(forestHawaii)    
table(forestHawaii$plot.id) 
</code></pre>

<hr>
<h2 id='gmean'>Function to compute the geometric mean of a vector</h2><span id='topic+gmean'></span>

<h3>Description</h3>

<p>Computes the geometric mean of a numeric vector. It is the n-th root of the product of n numbers, as follows.
</p>
<p style="text-align: center;"><code class="reqn">y_g = \left(\prod_{i=1}^{n} y_i\right)^{1/n}</code>
</p>

<p>for <code class="reqn">y_i &gt; 0</code>. It can also be understood as
the average of the logarithmic values of a data set, converted back to a base 10 number. 
The geometric mean is a central position statistics of a random variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gmean(v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gmean_+3A_v">v</code></td>
<td>
<p>is a numeric vector</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notice that can only be computed for positive values. For negative values, there
are alternatives, but not covered here.
</p>


<h3>Value</h3>

<p>This function returns the geometric mean, a numeric scalar.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
y.var &lt;- runif(10, min=10, max=45)
gmean(y.var)
</code></pre>

<hr>
<h2 id='hawaii'>Diameter growth increments of a tropical tree species in Hawaii</h2><span id='topic+hawaii'></span>

<h3>Description</h3>

<p>Tree size, competition, and diameter growth increment of Metrosideros polymorpha trees
collected in the Kilauea Volcano, Hawaii. Data containing 64 observations at the current annual growth 
rate (defined as dbh increment within one calendar year) of each tree
was measured from 1986 to 1988 using band dendrometers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hawaii)
</code></pre>


<h3>Format</h3>

<p>The dataframe has the following columns:
</p>

<dl>
<dt>tree.code</dt><dd><p>Tree number identification. The first letter of the ID represents
a cohort. Six cohorts representing a chronosequence were sampled.</p>
</dd>
<dt>dbh</dt><dd><p>Initial stem diameter, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>crown.area</dt><dd><p>Crown outline area, in square meters.</p>
</dd>
<dt>comp.ind</dt><dd><p>Competition index (Basal area of nearest neighbor divided by square of distance to nearest neighbor plus
basal area of second nearest neighbor divided by square of distance to second nearest neighbor).</p>
</dd>
<dt>cai.1986</dt><dd><p>Current annual stem diameter increment during 1986, in mm.</p>
</dd>
<dt>cai.1987</dt><dd><p>Current annual stem diameter increment during 1987, in mm.</p>
</dd>
<dt>cai.1988</dt><dd><p>Current annual stem diameter increment during 1988, in mm.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from Gerrish and Mueller-Dombois (1999).
</p>


<h3>References</h3>

<p>Gerrish G, Mueller-Dombois D. 1999. Measuring stem growth rates for
determining age and cohort analysis of a tropical evergreen tree.
Pacific Science. 53(4): 418-429.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hawaii)    
head(hawaii) 
</code></pre>

<hr>
<h2 id='hawaii2'>Incremento corriente anual en diámetro de una especie tropical en Hawaii</h2><span id='topic+hawaii2'></span>

<h3>Description</h3>

<p>Tamaño del árbol, competencia, e incremento corriente anual de árboles de Metrosideros polymorpha 
colectado en el volcan Kilauea, en Hawaii. Los datos contienen 64 observaciones de incremento corriente anual 
(definido como el incremento en diámetro en un año calendario) de cada árbol. Estos incrementos
fueron medidos desde el año 1986 a 1988 usando bandas dendrometricas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hawaii)
</code></pre>


<h3>Format</h3>

<p>Estos datos contienen las siguientes columnas:
</p>

<dl>
<dt>arb.id</dt><dd><p>Codigo identificador del árbol. La primera letra del ID representa
un cohorte. Hay seis cohortes que representan una cronosecuencia.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>area.copa</dt><dd><p>Area de copa, en metros cuadrados.</p>
</dd>
<dt>ind.comp</dt><dd><p>Competition index (Basal area of nearest neighbor divided by square of distance to nearest neighbor plus
basal area of second nearest neighbor divided by square of distance to second nearest neighbor).</p>
</dd>
<dt>ica.1986</dt><dd><p>Incremento corriente anual durante el año 1986, en mm.</p>
</dd>
<dt>ica.1987</dt><dd><p>Incremento corriente anual durante el año 1987, en mm.</p>
</dd>
<dt>ica.1988</dt><dd><p>Incremento corriente anual durante el año 1988, en mm.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron obtenidos desde Gerrish and Mueller-Dombois (1999).
</p>


<h3>References</h3>

<p>Gerrish G, Mueller-Dombois D. 1999. Measuring stem growth rates for
determining age and cohort analysis of a tropical evergreen tree.
Pacific Science. 53(4): 418-429.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hawaii2)    
head(hawaii2) 
</code></pre>

<hr>
<h2 id='hgrdfir'>Tree height growth of Douglas-fir sample trees in the Northwest of the United States</h2><span id='topic+hgrdfir'></span>

<h3>Description</h3>

<p>Data contains 148 observations on the height growth of dominant trees of Pseudotsguga mensiezzi in the Northwest of the United States.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hgrdfir)
</code></pre>


<h3>Format</h3>

<p>The data frame contains seven variables as follows:
</p>

<dl>
<dt>natfor.id</dt><dd><p>Code identifier.</p>
</dd>
<dt>plot.code</dt><dd><p>Plot number identification</p>
</dd>
<dt>tree.code</dt><dd><p>Tree number identification.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height at sampling, in in.</p>
</dd>
<dt>toth</dt><dd><p>Total height at sa,pling, in ft.</p>
</dd>
<dt>age</dt><dd><p>Age of tree, yr.</p>
</dd>
<dt>height</dt><dd><p>Height at a given age, in ft.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided by Dr Christian Salas.
</p>


<h3>References</h3>

<p>- Monserud RA. 1984. Height growth and site index curves for Inland Douglas-fir based on 
stem analysis data and forest habitat type. Forest Science 30(4):943-965.
</p>
<p>- Salas C, Stage AR, and Robinson AP. 2008. Modeling effects of overstory density and competing 
vegetation on tree height growth. Forest Science 54(1):107-122. <a href="https://doi.org/10.1093/forestscience/54.1.107">doi:10.1093/forestscience/54.1.107</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hgrdfir)    
head(hgrdfir)
unique(hgrdfir$tree.code)
table(hgrdfir$plot.code,hgrdfir$tree.code)
tapply(hgrdfir$dbh, hgrdfir$tree.code, mean)
tapply(hgrdfir$dbh, hgrdfir$tree.code, mean) #dbh of each sample tree
tapply(hgrdfir$toth, hgrdfir$tree.code, mean) #toth of each sample tree
</code></pre>

<hr>
<h2 id='hgrdfir2'>Crecimiento en altura de una muestra de árboles en los Estados Unidos</h2><span id='topic+hgrdfir2'></span>

<h3>Description</h3>

<p>Data contiene 148 obserrvaciones sobre el crecimiento en altura de árboles dominantes de Pseudotsguga mensiezzi en el Nor-Oeste de los Estados Unidos
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hgrdfir2)
</code></pre>


<h3>Format</h3>

<p>La data frame contiene siete variables:
</p>

<dl>
<dt>bosque.id</dt><dd><p>Codigo identificador del bosque.</p>
</dd>
<dt>parcela</dt><dd><p>Codigo identificador de la parcela.</p>
</dd>
<dt>arbol</dt><dd><p>Número de identificacion árbol.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en  pulgadas.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en pies</p>
</dd>
<dt>edad</dt><dd><p>Edad, en os</p>
</dd>
<dt>altura</dt><dd><p>Altura para cada edad del árbol, en pies</p>
</dd>
</dl>



<h3>Source</h3>

<p>La data fue cedida por el Dr Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>Monserud RA. 1984. Height growth and site index curves for Inland Douglas-fir based on 
stem analysis data and forest habitat type. Forest Science 30(4):943-965.
</p>
<p>Salas C, Stage AR, and Robinson AP. 2008. Modeling effects of overstory density and competing 
vegetation on tree height growth. Forest Science 54(1):107-122. <a href="https://doi.org/10.1093/forestscience/54.1.107">doi:10.1093/forestscience/54.1.107</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hgrdfir2)    
head(hgrdfir2)
unique(hgrdfir2$arbol.id)
table(hgrdfir2$parcela,hgrdfir2$arbol.id)
tapply(hgrdfir2$dap, hgrdfir2$arbol.id, mean) #dap de cada arbol muestra
tapply(hgrdfir2$atot, hgrdfir2$arbol.id, mean) #atot de cada arbol muestra
</code></pre>

<hr>
<h2 id='idahohd'>Tree height-diameter data from Idaho (USA)</h2><span id='topic+idahohd'></span>

<h3>Description</h3>

<p>These data are forest inventory measures from the Upper Flat Creek stand of the University 
of Idaho Experimental Forest, dated 1991.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(idahohd)
</code></pre>


<h3>Format</h3>

<p>Contains five variables, as follows:
</p>

<dl>
<dt>plot</dt><dd><p>Plot number.</p>
</dd>
<dt>tree</dt><dd><p>Tree within plot.</p>
</dd>
<dt>species</dt><dd><p>A factor with levels DF = Douglas-fir, GF = Grand fir, SF = Subalpine fir, WL = Western larch,
WC = Western red cedar, WP = White pine.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter 137 cm perpendicular to the bole, cm.</p>
</dd>
<dt>toth</dt><dd><p>Height of the tree, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were assembled from the 'ufc' dataframe from the 'alr4' library.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. New York: Wiley.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(idahohd)    
head(idahohd) 
plot(toth~dbh, data=idahohd)
</code></pre>

<hr>
<h2 id='idahohd2'>Altura-diámetro de árboles en el estado de Idaho (USA)</h2><span id='topic+idahohd2'></span>

<h3>Description</h3>

<p>Estos datos provienen de un muestreo en el bosque experimental de la 
University of Idaho, en Upper Flat Creek, Idaho, USA. Medido en 1991.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(idahohd2)
</code></pre>


<h3>Format</h3>

<p>Contiene cinco variables detalladas a continuación:
</p>

<dl>
<dt>parce</dt><dd><p>Número de la parcela de muestreo.</p>
</dd>
<dt>arbol</dt><dd><p>Número del árbol dentro de la parcela.</p>
</dd>
<dt>spp</dt><dd><p>Especie del árbol, una variable factor  con niveles DF = Douglas-fir, GF = Grand fir, SF = Subalpine fir, WL = Western larch,
WC = Western red cedar, WP = White pine.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro del fuste a los 1.3 m sobre el suelo, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura del árbol, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron obtenidos desde la dataframe 'ufc' de la librería 'alr4'.
</p>


<h3>References</h3>

<p>Weisberg S. 2014. Applied Linear Regression. 4th edition. New York: Wiley.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(idahohd2)    
head(idahohd2)
plot(atot~dap, data=idahohd2)
</code></pre>

<hr>
<h2 id='invasivesRCI'>Contains regeneration microsite data in Robinson Crusoe Island forest</h2><span id='topic+invasivesRCI'></span>

<h3>Description</h3>

<p>These are plot-level measurement (2x2 m) data from the forests in the Robinson Crusoe Island, located
in the Pacific Ocean, 667 km from mainland Chile. Measurements were collected in transects 
of 100 to 240 meters in which, 398 squared plots (2x2 m) were set to include canopy gaps, 
gap borders and closed forest conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(invasivesRCI)
</code></pre>


<h3>Format</h3>

<p>Data has the following columns
</p>

<dl>
<dt>plot.id</dt><dd><p>Plot identification code</p>
</dd>
<dt>Gap.type</dt><dd><p>Canopy gap classified as invaded=Inv, non invaded= Nat or treated =Treat(considering the estimated cover of invasive plant species)</p>
</dd>
<dt>Forest.zone</dt><dd><p>Location of the plot (gap, border or forest)</p>
</dd>
<dt>Ferns</dt><dd><p>Estimated cover of fern species (in 2x2 plots)</p>
</dd>
<dt>Moss.liverw</dt><dd><p>Estimated cover of mosses and liverworts ( in 2x2 plots)</p>
</dd>
<dt>Cwd</dt><dd><p>Estimated cover of coarse woody debris &gt; 3 cm diameter ( in 2x2 plots)</p>
</dd>
<dt>Litter</dt><dd><p>Estimated cover of litter (in 2x2 plots)</p>
</dd>
<dt>Ms</dt><dd><p>Estimated cover of mineral soil ( in 2x2 plots)</p>
</dd>
<dt>Rock</dt><dd><p>Estimated cover of rocks (in 2x2 plots)</p>
</dd>
<dt>Est.age</dt><dd><p>Age category for the canopy gap associated to each plot</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Rodrigo Vargas-Gaete at Universidad
de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Vargas-Gaete R, Salas-Eljatib C, Gärtner SM, Vidal OJ, Bannister JR, Pauchard A. 2018. 
Invasive plant species thresholds in the forests of Robinson Crusoe Island, Chile. 
Plant Ecology &amp; Diversity, 11(2), 205-215.
</p>

<hr>
<h2 id='kurto'>Computes the sample kurtosis of a distribution</h2><span id='topic+kurto'></span>

<h3>Description</h3>

<p>The kurtosis is about the tailedness, or the degree of heaviness of the tails, in the frequency 
distribution. The function computes an estimator of the kurtosis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kurto(x, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kurto_+3A_x">x</code></td>
<td>
<p>a numeric vector of a random variable.</p>
</td></tr>
<tr><td><code id="kurto_+3A_na.rm">na.rm</code></td>
<td>
<p>logical operator to remove NA values. The default is set to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The kurtosis of a random variable is the fourth moment of the standardized variable.
There are several ways of parameterizing a kurtosis estimator, such as depending on the 
fourth moment and the standard deviation of the random variable.
</p>


<h3>Value</h3>

<p>An estimator of the kurtosis.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y.var&lt;-rnorm(100);x.var&lt;-rbeta(100,.2,2)
kurto(y.var)
kurto(x.var)

</code></pre>

<hr>
<h2 id='landCoverSantiago'>Land-cover, environmental and sociodemographic data for the 34 municipalities composing the Greater Santiago area, Santiago, Chile.</h2><span id='topic+landCoverSantiago'></span>

<h3>Description</h3>

<p>dataset contains 476 observations, 34 categorical and 442 numerical. Land-cover data was generated through remote 
sensing classification techniques using Sentinel-2 satellite images from year 2016. Temperatures were obtained from 
TIRS band 10 of Landsat 8 satellites images. Particulate matter concentrations were estimated using spatial modelling
techniques from 10 pollution stations distributed in the city. Altitude was generated from a Digital Elevation Model. 
Population and poverty were gathered from Casen 2017 survey.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(landCoverSantiago)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>Comuna</dt><dd><p>Name of Municipality</p>
</dd>											
<dt>p.Construido</dt><dd><p>Percentage of surface covered by built-up area</p>
</dd>
<dt>p.Vegetacion</dt><dd><p>Percentage of surface covered by vegetation</p>
</dd>
<dt>p.Desnudo</dt><dd><p>Percentage of surface covered by bare soil</p>
</dd>
<dt>p.Pasto</dt><dd><p>Percentage of surface covered by deciduous vegetation</p>
</dd>
<dt>p.Deciduo</dt><dd><p>Percentage of surface covered by evergreen vegetation</p>
</dd>
<dt>p.Siempreverde</dt><dd><p>Percentage of surface covered by evergreen vegetation</p>
</dd>
<dt>Temp Invierno</dt><dd><p>Land surface temperature in celsius degrees at 2pm on a winter 0% cloud day</p>
</dd>
<dt>Temp Verano</dt><dd><p>Land surface temperature in celsius degrees at 2pm on a summer 0% cloud day</p>
</dd>
<dt>PM10 Invierno</dt><dd><p>Average particulate matter 10 micron during winter months</p>
</dd>
<dt>PM10 Verano</dt><dd><p>Average particulate matter 10 micron during summer months</p>
</dd>
<dt>p.pobreza 2017</dt><dd><p>Percentage of people under poverty line year 2017</p>
</dd>
<dt>Altitud promedio</dt><dd><p>Average altitude of municipal area</p>
</dd>
<dt>Poblacion</dt><dd><p>Total population of municipality</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Ignacio Fernandez at Universidad Adolfo Ibanez (Santiago, Chile).
</p>


<h3>References</h3>

<p>Not yet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(landCoverSantiago)    
head(landCoverSantiago) 
</code></pre>

<hr>
<h2 id='landCoverSantiago2'>Cobertura territorial, ambiental y sociodemografica de los 34 municipios que componen el area del Gran Santiago, Santiago, Chile..</h2><span id='topic+landCoverSantiago2'></span>

<h3>Description</h3>

<p>El conjunto de datos contiene 476 observaciones, 34 categoricas y 442 numericas. Los datos de cobertura 
terrestre se generaron mediante tecnicas de clasificacion de teledeteccion utilizando imagenes de satelite 
Sentinel-2 del año 2016. Las temperaturas se obtuvieron de la banda TIRS 10 de las imagenes de los satelites 
Landsat 8. Las concentraciones de material particulado se estimaron mediante tecnicas de modelado espacial 
de 10 estaciones de contaminacion distribuidas en la ciudad. La altitud se genero a partir de un modelo de 
elevacion digital. La poblacion y la pobreza se obtuvieron de la encuesta Casen 2017.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(landCoverSantiago2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuacion:
</p>

<dl>
<dt>Comuna</dt><dd><p>Name of Municipality</p>
</dd>											
<dt>p.Construido</dt><dd><p>Porcentaje de superficie cubierta por area construida</p>
</dd>
<dt>p.Vegetacion</dt><dd><p>Porcentaje de superficie cubierta por vegetacion</p>
</dd>
<dt>p.Desnudo</dt><dd><p>Porcentaje de superficie cubierta por suelo desnudo</p>
</dd>
<dt>p.Pasto</dt><dd><p>Porcentaje de superficie cubierta por cesped</p>
</dd>
<dt>p.Deciduo</dt><dd><p>Porcentaje de superficie cubierta por vegetacion de hoja caduca</p>
</dd>
<dt>p.Siempreverde</dt><dd><p>Porcentaje de superficie cubierta por vegetacion siempre verde</p>
</dd>
<dt>Temp Invierno</dt><dd><p>Temperatura de la superficie terrestre en grados celsius a las 2 p.m.en un dia de invierno con 0% de nubes</p>
</dd>
<dt>Temp Verano</dt><dd><p>Temperatura de la superficie de la tierra en grados celsius a las 2 p.m.en un dia de verano con 0% de nubes</p>
</dd>
<dt>PM10 Invierno</dt><dd><p>Material particulado promedio de 10 micrones durante los meses de invierno</p>
</dd>
<dt>PM10 Verano</dt><dd><p>Material particulado promedio de 10 micrones durante los meses de verano</p>
</dd>
<dt>p.pobreza 2017</dt><dd><p>Porcentaje de personas por debajo de la linea de pobreza año 2017</p>
</dd>
<dt>Altitud promedio</dt><dd><p>Altitud media del termino municipal</p>
</dd>
<dt>Poblacion</dt><dd><p>poblacion total del municipio</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Dr Ignacio Fernandez de la Universidad Adolfo Ibanez (Santiago, Chile).
</p>


<h3>References</h3>

<p>Not yet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(landCoverSantiago2)    
head(landCoverSantiago2) 
</code></pre>

<hr>
<h2 id='lleuque'>Contains species composition data of Prumnopitys andina (Lleuque) forests</h2><span id='topic+lleuque'></span>

<h3>Description</h3>

<p>Contains species composition data for forests with presence of Lleuque (Prumnopitys andina)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lleuque
</code></pre>


<h3>Format</h3>

<p>The dataframe has the following columns
</p>

<dl>
<dt>stand</dt><dd><p>Stand number</p>
</dd>
<dt>plot.num</dt><dd><p>Plot number</p>
</dd>
<dt>Aus.chi</dt><dd><p>Tree density/ha of Austrocedrus chilensis</p>
</dd>
<dt>May.dis</dt><dd><p>Tree density/ha of Maytenus disticha</p>
</dd>
<dt>Not.obl</dt><dd><p>Tree density/ha of Nothofagus obliqua</p>
</dd>
<dt>Pru.and</dt><dd><p>Tree density/ha of Prumnopitys andina</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Rodrigo Vargas-Gaete at Universidad
de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Vargas-Gaete R, Salas-Eljatib C, Penneckamp D, Neira Z, Diez MC, Vargas-Picón, R. 2020. Estructura 
y regeneración de bosques de Prumnopitys andina en los Andes del sur de Chile. Gayana Botánica, 77(1), 48-58.
</p>

<hr>
<h2 id='lrt'>Computes a likelihood ratio test between a reduced model and a full model</h2><span id='topic+lrt'></span>

<h3>Description</h3>

<p>Computes a likelihood ratio test between a reduced model (modr) and a full model (modf). Both models must be
previously fitted by maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrt(modr, modf)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lrt_+3A_modr">modr</code></td>
<td>
<p>is a previously fitted model having less parameters than modf</p>
</td></tr>
<tr><td><code id="lrt_+3A_modf">modf</code></td>
<td>
<p>is a previously fitted model having more parameters than modr</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Double-check the order of the reduced and full model, before of using the model
</p>


<h3>Value</h3>

<p>This function returns an object having the following elements:
&quot;loglik.Modr&quot; maximized log-likelihood of modr; &quot;loglik.Modf&quot; maximized log-likelihood of modf;
&quot;dif.loglik&quot;  difference in log-likelihood between both models, and 
&quot;dif.df&quot; difference in degrees of freedong of both models, and 
&quot;p-value&quot; is the p-value for the LRT.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>Pinheiro JC, and Bates DM. 2000. Mixed-effects models in S
and Splus. Springer-Verlag, New York, NY. 528 p.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#not yet implemented
</code></pre>

<hr>
<h2 id='moda'>Computes the mode</h2><span id='topic+moda'></span>

<h3>Description</h3>

<p>Computes the mode of a random variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moda(y = y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moda_+3A_y">y</code></td>
<td>
<p>is a numeric vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mode is an statistics representing the most &quot;used&quot; value of
the random variable as a way of central position.
</p>


<h3>Value</h3>

<p>The function returns the mode, a numeric scalar.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1234)
variable &lt;- rnorm(10, mean=45,sd=6)
#using the function
moda(y=variable)
moda(variable)
</code></pre>

<hr>
<h2 id='pinaster'>Tree volume for Pinus pinaster in the Baixo-Mino, Galicia, Spain.</h2><span id='topic+pinaster'></span>

<h3>Description</h3>

<p>These are volume measurements data of sample trees in the Baixo-Mino region in Galicia, Spain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinaster)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>stand</dt><dd><p>stand number from the sample tree was selected.</p>
</dd>
<dt>si</dt><dd><p>Site index of the stand.</p>
</dd>
<dt>tree.no</dt><dd><p>tree number.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>d4</dt><dd><p>Upper-stem diameter at 4 m, in cm.</p>
</dd>
<dt>volwb</dt><dd><p>Tree gross volume, in m<code class="reqn">^{3}</code> with bark.</p>
</dd>
<dt>volwob</dt><dd><p>Tree gross volume, in m<code class="reqn">^{3}</code> without bark.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Dr Christian Salas-Eljatib at
the Universidad de Chile (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C, Nieto L, Irisarri A. 2005. Modelos de volumen para Pinus pinaster Ait. en la comarca del Baixo Mino,
Galicia, España. Quebracho 12: 11-22. <a href="https://eljatib.com/publication/2005-12-01_modelos_de_volumen_p/">https://eljatib.com/publication/2005-12-01_modelos_de_volumen_p/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinaster)    
head(pinaster) 
</code></pre>

<hr>
<h2 id='pinaster2'>Volumen individual de árboles de Pinus pinaster en Galicia, España.</h2><span id='topic+pinaster2'></span>

<h3>Description</h3>

<p>Variables de volumen y otras a nivel de árbol para una muestra de árboles
de Pinus pinaster en la comarca del Baixo-Mino en Galicia, España.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinaster2)
</code></pre>


<h3>Format</h3>

<p>Contiene las siguientes variables a nivel de árbol:
</p>

<dl>
<dt>rodal</dt><dd><p>Rodal desde donde el árbol fue muestreado</p>
</dd>
<dt>ind.sitio</dt><dd><p>Indice de sitio del rodal, en m.</p>
</dd>
<dt>arbol</dt><dd><p>Número del árbol.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>d4</dt><dd><p>Diámetro fustal a los 4 m, en cm.</p>
</dd>
<dt>vtcc</dt><dd><p>Volumen bruto total con corteza, en m<code class="reqn">^{3}</code>.</p>
</dd>
<dt>vtsc</dt><dd><p>Volumen bruto total sin corteza, en m<code class="reqn">^{3}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Dr Christian Salas (Chile).
</p>


<h3>References</h3>

<p>- Salas C, Nieto L, Irisarri A. 2005. Modelos de volumen para Pinus pinaster Ait. en la comarca del Baixo Mino,
Galicia, España. Quebracho 12: 11-22. <a href="https://eljatib.com/publication/2005-12-01_modelos_de_volumen_p/">https://eljatib.com/publication/2005-12-01_modelos_de_volumen_p/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinaster2)    
head(pinaster2) 
</code></pre>

<hr>
<h2 id='pinusContorta'>Contains spatial location of Pinus contorta trees in sample plots.</h2><span id='topic+pinusContorta'></span>

<h3>Description</h3>

<p>These are tree-level measurement data, with cartesian location of each tree, from Pinus contorta invasion in 
Patagonian steppe in Coyhaique in southhern Chile, measured in 2011. There are 3 plots, each
of 10.000 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinusContorta)
</code></pre>


<h3>Format</h3>

<p>Contains eight variables, as follows:
</p>

<dl>
<dt>plot.id</dt><dd><p>Plot sample ID.</p>
</dd>
<dt>tree.id</dt><dd><p>Tree identificator number in each plot. Same indv/id for multi-stem trees.</p>
</dd>
<dt>y.coord</dt><dd><p>coordinate of S latitude.</p>
</dd>
<dt>x.coord</dt><dd><p>coordinate of W longitude.</p>
</dd>
<dt>substrate</dt><dd><p>Ground cover in which each pine grow. Bare soil, Festuca pallescens, Baccharis magellanica, 
Oreopulus glacialis, Acaena integerrima and others species.</p>
</dd>
<dt>drc</dt><dd><p>Diameter at the root collar on trees, in mm.</p>
</dd>
<dt>h</dt><dd><p>Height of trees, in cm.</p>
</dd>
<dt>canopy.area</dt><dd><p>Proyection of canopy area of each tree, in square meters.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Drs Anibal Pauchard and Rafael Garcia at the Laboratorio de Invasiones Biologicas,
Universidad de Concepción (Concepción, Chile).
</p>


<h3>References</h3>

<p>Pauchard A, Escudero A, Garcia RA, de la Cruz M, Langdon B, Cavieres LA, Esquivel J. 2016.
Pine invasions in treeless environments: dispersal overruns microsite heterogeneity. 
Ecology and Evolution. 6(2): 447-459. <a href="https://doi.org/10.1002/ece3.1877">doi:10.1002/ece3.1877</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinusContorta)    
head(pinusContorta)
unique(pinusContorta$plot.id) 
</code></pre>

<hr>
<h2 id='pinusContorta2'>Ubicación espacial de árboles de Pinus contorta en parcela de muestreo</h2><span id='topic+pinusContorta2'></span>

<h3>Description</h3>

<p>Mediciones a nivel de árbol, con la ubicación cartesian de cada árbol de Pinus contorta, en parcelas
de muestreo para estudio de invasion en la estepa Patagonica en Coyhaique en el sur de Chile. Hay tres parcelas, cada
una de 10.000 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinusContorta2)
</code></pre>


<h3>Format</h3>

<p>Contiene ocho variables, como siguen:
</p>

<dl>
<dt>parcela</dt><dd><p>Parcela.</p>
</dd>
<dt>arbol</dt><dd><p>Número de árbol en cada parcela. Mismo árbol/id para árboles multifustales.</p>
</dd>
<dt>coord.y</dt><dd><p>coordinada de latitud W.</p>
</dd>
<dt>coord.x</dt><dd><p>coordinada de longitud W.</p>
</dd>
<dt>substrato</dt><dd><p>Cobertura del suelo donde cada pino crece. Bare soil, Festuca pallescens, Baccharis magellanica, 
Oreopulus glacialis, Acaena integerrima and others species.</p>
</dd>
<dt>h</dt><dd><p>Height of trees, in cm.</p>
</dd>
<dt>diam.cuello</dt><dd><p>Diámetro del cuello, en mm.</p>
</dd>
<dt>area.copa</dt><dd><p>Area de copa, en m<code class="reqn">^{2}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por los Drs. Anibal Pauchard y Rafael Garcia del Laboratorio de Invasiones Biologicas,
Universidad de Concepcion (Chile).
</p>


<h3>References</h3>

<p>Pauchard A, Escudero A, Garcia RA, de la Cruz M, Langdon B, Cavieres LA, Esquivel J. 2016.
Pine invasions in treeless environments: dispersal overruns microsite heterogeneity. 
Ecology and Evolution. 6(2): 447-459. <a href="https://doi.org/10.1002/ece3.1877">doi:10.1002/ece3.1877</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinusContorta2)    
head(pinusContorta2)
unique(pinusContorta2$plot.id) 
</code></pre>

<hr>
<h2 id='pinusSpp'>Tree-level variables of several sample plots of invasive Pinus spp in Chile.</h2><span id='topic+pinusSpp'></span>

<h3>Description</h3>

<p>These are tree-lavel measurement data from Pinus spp invasion in Araucaria-Nothofagus forests in 
the Malalcahuello National Reserve in La Araucania region in southhern Chile, measured in 2012. 
There are 26 plots and plot size is 100 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinusSpp)
</code></pre>


<h3>Format</h3>

<p>Contains eight variables, as follows:
</p>

<dl>
<dt>plot.id</dt><dd><p>Plot sample ID.</p>
</dd>
<dt>plot.size</dt><dd><p>Plot size, en m<code class="reqn">^{2}</code>.</p>
</dd>
<dt>lat.s</dt><dd><p>Decimal coordinate of S latitude.</p>
</dd>
<dt>long.w</dt><dd><p>Decimal coordinate of W longitude.</p>
</dd>
<dt>indv.id</dt><dd><p>Tree identificator number in each plot. Same indv/id for multi-stem trees.</p>
</dd>
<dt>stem.id</dt><dd><p>Stem identificator number in each plot.</p>
</dd>
<dt>spp</dt><dd><p>Specie.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast-height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>hcb</dt><dd><p>Height to crown base, in m.</p>
</dd>
<dt>crown.lenght</dt><dd><p>Crown lenght, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Drs Anibal Pauchard and Rafael García at the Laboratorio de Invasiones
Biológicas, Universidad de Concepcion (Concepción, Chile).
</p>


<h3>References</h3>

<p>Cobar-Carranza A, Garcia R, Pauchard A, Pena E. 2014. Effect of Pinus contorta invasion on forest fuel properties and
its potential implications on the fire regime of Araucaria araucana and Nothofagus antarctica forests.
Biological Invasions. 16(11): 2273 - 2291. <a href="https://doi.org/10.1007/s10530-014-0663-8">doi:10.1007/s10530-014-0663-8</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinusSpp)    
head(pinusSpp) 
length(unique(pinusSpp$plot.id)) 
boxplot(dbh~plot.id, data=pinusSpp) 
</code></pre>

<hr>
<h2 id='pinusSpp2'>Variables a nivel de árbol en parcelas de muestreo de Pinus spp en Chile.</h2><span id='topic+pinusSpp2'></span>

<h3>Description</h3>

<p>Mediciones a nivel de árbol para estudiar la invasion de Pinus spp en bosques de Araucaria-Nothofagus en 
la Reserva Nacional Malalcahuello en la region de la Araucania en el sur de Chile. 
Hay 26 parcelas, y la superficie de cada una es de 100 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pinusSpp2)
</code></pre>


<h3>Format</h3>

<p>Los datos contienen ocho columnas que se detallan a continuacion:
</p>

<dl>
<dt>parcela</dt><dd><p>Número de la parcela.</p>
</dd>
<dt>sup.parcela</dt><dd><p>Superficie de la parcela, en m<code class="reqn">^{2}</code>.</p>
</dd>
<dt>lat.s</dt><dd><p>Decimal coordinate of S latitude.</p>
</dd>
<dt>long.w</dt><dd><p>Decimal coordinate of W longitude.</p>
</dd>
<dt>indv.id</dt><dd><p>Identificador del árbol en la parcelaeach plot. Same indv/id for multi-stem trees.</p>
</dd>
<dt>fuste.id</dt><dd><p>Identificador del fuste.</p>
</dd>
<dt>espe</dt><dd><p>Especie.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>hcc</dt><dd><p>Altura comienzo de copa, en m.</p>
</dd>
<dt>largo.copa</dt><dd><p>Largo de copa, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por los Drs. Anibal Pauchard y Rafael García del Laboratorio de Invasiones
Biológicas, Universidad de Concepción (Concepción, Chile).
</p>


<h3>References</h3>

<p>Cobar-Carranza A, Garcia R, Pauchard A &amp; Pena E. 2014. Effect of Pinus contorta invasion on forest fuel properties and
its potential implications on the fire regime of Araucaria araucana and Nothofagus antarctica forests.
Biological Invasions. 16(11): 2273-2291. <a href="https://doi.org/10.1007/s10530-014-0663-8">doi:10.1007/s10530-014-0663-8</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pinusSpp2)    
head(pinusSpp2) 
length(unique(pinusSpp2$parce)) 
boxplot(dap~parce, data=pinusSpp2) 
</code></pre>

<hr>
<h2 id='plantsHawaii'>Maximum plant size in the Hawaiian archipelago.</h2><span id='topic+plantsHawaii'></span>

<h3>Description</h3>

<p>Maximum plant size of 58 tree, shrub and tree fern species that occur 
in 530 forest plots across the Hawaiian archipelago.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(plantsHawaii)
</code></pre>


<h3>Format</h3>

<p>Contains six variables, as follows:
</p>

<dl>
<dt>scientific.name</dt><dd><p>Genus and epithet of each individual following The Plant List v. 1.1 (2013).</p>
</dd>
<dt>family</dt><dd><p>Family of each individual following The Plant List v. 1.1 (2013).</p>
</dd>
<dt>native.status</dt><dd><p>Categorical variable ('native', 'alien', 'uncertain') indicating alien status of each individual following Wagner et al. (2005).</p>
</dd>
<dt>n</dt><dd><p>Number of individuals used to estimate maximum plant size.</p>
</dd>
<dt>d.95</dt><dd><p>Maximum plant size, estimated as D950.1 (King et al. 2006).</p>
</dd>
<dt>d.max.3</dt><dd><p>Maximum plant size, estimated as Dmax3 (King et al. 2006).</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.1kk02qr">doi:10.5061/dryad.1kk02qr</a>.
</p>


<h3>References</h3>

<p>- Craven D, Knight T, Barton K, Bialic-Murphy L, Cordell S, Giardina C, Gillespie T, Ostertag R, Sack L,Chase J.
2018. OpenNahele: the open Hawaiian forest plot database. Biodiversity Data Journal 6: e28406.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(plantsHawaii)    
head(plantsHawaii) 
</code></pre>

<hr>
<h2 id='presenceIce'>Presence or absence of sea ice from logbook records of annual cruises</h2><span id='topic+presenceIce'></span>

<h3>Description</h3>

<p>Data containing 52717 observations about presence of sea ice from logbook records of annual 
cruises to the B-C-B in an unbroken record between  years 1850 to 1910.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(presenceIce)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains the following columns:
</p>

<dl>
<dt>ship.id</dt><dd><p>The code number for ships.</p>
</dd>
<dt>move.type</dt><dd><p>Type of movement of ships. 0 indicates a sail-powered vessel and 1 indicates an auxiliary-powered vessel.</p>
</dd>
<dt>year</dt><dd><p>Year of registry.</p>
</dd>
<dt>month</dt><dd><p>Month of registry.</p>
</dd>
<dt>day</dt><dd><p>Day of registry.</p>
</dd>
<dt>lat.dec</dt><dd><p>Decimal latitude.</p>
</dd>
<dt>long.dec</dt><dd><p>Decimal longitude.</p>
</dd>
<dt>e.w</dt><dd><p>East or west of the Prime Meridian.</p>
</dd>
<dt>ice.cov</dt><dd><p>Sea Ice Observed. 0 no see (Not registered) and 1 presence sea ice (Registered).</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided from Sea Ice Group at the Geophysical Institute.
</p>


<h3>References</h3>

<p>Mahoney A, Bockstoce J, Botkin D, Eicken H, Nisbet R. 2011. Sea-Ice Distribution in the Bering and Chukchi Seas: Information from Historical Whaleships' Logbooks and Journals
ARCTIC. 64(4): 465-477.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(presenceIce)    
head(presenceIce) 
</code></pre>

<hr>
<h2 id='presidentChile'>2021 presidential election in Chile.</h2><span id='topic+presidentChile'></span>

<h3>Description</h3>

<p>Voter table-level data of the 2021 presidential election in Chile. The election was held on December 19, 2021.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(presidentChile)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains the following columns:
</p>

<dl>
<dt>region.no</dt><dd><p>Administrative region number of Chile.</p>
</dd>
<dt>region</dt><dd><p>Administrative region name.</p>
</dd>
<dt>province</dt><dd><p>Province.</p>
</dd>
<dt>senat.const</dt><dd><p>Senatorial constituency.</p>
</dd>
<dt>distrit</dt><dd><p>Distrit.</p>
</dd>
<dt>county</dt><dd><p>County.</p>
</dd>
<dt>elect.const</dt><dd><p>Electoral constituency.</p>
</dd>
<dt>location</dt><dd><p>Location.</p>
</dd>
<dt>table</dt><dd><p>Voter table.</p>
</dd>
<dt>table.type</dt><dd><p>Voter table type.</p>
</dd>
<dt>merged.tab</dt><dd><p>Merged voter tables.</p>
</dd>
<dt>electors</dt><dd><p>Electors.</p>
</dd>
<dt>num.in.vote</dt><dd><p>.</p>
</dd>
<dt>candidate</dt><dd><p>Candidate. Gabriel Boric and Jose A. Kast</p>
</dd>
<dt>total.votes</dt><dd><p>Total number of votes acording to the TRICEL.</p>
</dd>       
</dl>



<h3>Source</h3>

<p>The data were obtained from the electoral service of the Chilean Government (SERVEL) at
<a href="https://www.servel.cl">https://www.servel.cl</a>.
The datafile name was &quot;Resultados_mesa_presidencial_TRICEL_2v_2021-1.xlsx&quot;, and was downloaded
on October 24, 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(presidentChile)    
head(presidentChile) 
</code></pre>

<hr>
<h2 id='presidentChile2'>Eleccion presidencial del 2021 en Chile.</h2><span id='topic+presidentChile2'></span>

<h3>Description</h3>

<p>Datos de mesa de la eleccion presidencial del 2012 en Chile. La eleccion se llevo a cabo el 19 de Diciembre del 2021.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(presidentChile2)
</code></pre>


<h3>Format</h3>

<p>Los datos contienen las siguientes columnas:
</p>

<dl>
<dt>region.no</dt><dd><p>Número de la region adminsitrativa de Chile.</p>
</dd>
<dt>region</dt><dd><p>Nombre de la region administrativa de Chile</p>
</dd>
<dt>provincia</dt><dd><p>Provincia.</p>
</dd>
<dt>circu.senatorial</dt><dd><p>Circunscripcion senatorial.</p>
</dd>
<dt>distrito</dt><dd><p>Distrit.</p>
</dd>
<dt>comuna</dt><dd><p>County.</p>
</dd>
<dt>circu.elec</dt><dd><p>Circunscripcion electoral.</p>
</dd>
<dt>local</dt><dd><p>Local de votacion. Generalmente es un colegio.</p>
</dd>
<dt>no.mesa</dt><dd><p>Número de mesa.</p>
</dd>
<dt>tipo.mesa</dt><dd><p>Tipo de mesa de votacion.</p>
</dd>
<dt>mesas.fusionadas</dt><dd><p>Mesa de votacion fucionada.</p>
</dd>
<dt>electores</dt><dd><p>Electores.</p>
</dd>
<dt>nro.en.voto</dt><dd><p>.</p>
</dd>
<dt>candidato</dt><dd><p>Candidato, ya sea Gabriel Boric o Jose A. Kast</p>
</dd>
<dt>votos.tricel</dt><dd><p>Número total de votos segun el TRICEL (Tribunal calificador de elecciones).</p>
</dd>       
</dl>



<h3>Source</h3>

<p>Los datos fueron obtenidos desde el sitio web del Servicio Electoral del Gobierno de Chilean (SERVEL) en
<a href="https://www.servel.cl">https://www.servel.cl</a>.
El archivo de datos descargado el 24 de Octubre del 2022 tenia el nombre &quot;Resultados_mesa_presidencial_TRICEL_2v_2021-1.xlsx&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(presidentChile2)    
head(presidentChile2) 
</code></pre>

<hr>
<h2 id='primary'>2021 primary election for the president of Chile</h2><span id='topic+primary'></span>

<h3>Description</h3>

<p>Voter table-level data of the 2021 primary election for the president of Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(primary)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains the following columns:
</p>

<dl>
<dt>region.no</dt><dd><p>Administrative region number of Chile.</p>
</dd>
<dt>region</dt><dd><p>Administrative region name.</p>
</dd>   
<dt>province</dt><dd><p>Province.</p>
</dd>
<dt>distrit</dt><dd><p>Distrit.</p>
</dd>
<dt>county</dt><dd><p>County.</p>
</dd>
<dt>elect.const</dt><dd><p>Electoral constituency.</p>
</dd>
<dt>location</dt><dd><p>Location.</p>
</dd>
<dt>table.type</dt><dd><p>Voter table type.</p>
</dd>
<dt>table</dt><dd><p>Voter table.</p>
</dd>
<dt>merged.tab</dt><dd><p>Merged voter tables.</p>
</dd>
<dt>num.in.vote</dt><dd><p>.</p>
</dd>
<dt>list</dt><dd><p>Specific political list.</p>
</dd>
<dt>pact</dt><dd><p>Specific political pact.</p>
</dd>
<dt>party</dt><dd><p>Political party.</p>
</dd>
<dt>candidate</dt><dd><p>Candidate name</p>
</dd>
<dt>total.votes</dt><dd><p>Total number of votes.</p>
</dd>       
</dl>



<h3>Source</h3>

<p>The data were obtained from the electoral service of the Chilean Government (SERVEL) at
<a href="https://www.servel.cl">https://www.servel.cl</a>.
The datafile name was &quot;Resultados_Primarias_Presidenciales_2021_CHILE.xlsx&quot;, and was downloaded
on October 24, 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(primary)    
head(primary) 
</code></pre>

<hr>
<h2 id='primary2'>Elección primaria para la presidencia de Chile</h2><span id='topic+primary2'></span>

<h3>Description</h3>

<p>Datos a nivel de mesa para la votacion primaria de presidente de Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(primary2)
</code></pre>


<h3>Format</h3>

<p>The dataframe contains the following columns:
</p>

<dl>
<dt>region.no</dt><dd><p>Región administrativa de Chile.</p>
</dd>
<dt>region</dt><dd><p>Administrative region name.</p>
</dd>   
<dt>provincia</dt><dd><p>Provincia.</p>
</dd>
<dt>distrito</dt><dd><p>Distrito.</p>
</dd>
<dt>comuna</dt><dd><p>Comuna.</p>
</dd>
<dt>circu.elec</dt><dd><p>Circunscripcion electoral.</p>
</dd>
<dt>local</dt><dd><p>Local de votacion.</p>
</dd>
<dt>tipo.mesa</dt><dd><p>Voter table type.</p>
</dd>
<dt>mesa</dt><dd><p>Voter table.</p>
</dd>
<dt>mesas.fusionadas</dt><dd><p>Merged voter tables.</p>
</dd>
<dt>nro.voto</dt><dd><p>.</p>
</dd>
<dt>lista</dt><dd><p>Lista politica del candidato.</p>
</dd>
<dt>pacto</dt><dd><p>Pacto politico del candidato.</p>
</dd>
<dt>partido</dt><dd><p>Partido politica del candidato.</p>
</dd>
<dt>candidato</dt><dd><p>Nombre del candidato.</p>
</dd>
<dt>votos</dt><dd><p>Número total de votos.</p>
</dd>       
</dl>



<h3>Source</h3>

<p>The data were obtained from the electoral service of the Chilean Government (SERVEL) at
<a href="https://www.servel.cl">https://www.servel.cl</a>.
The datafile name was &quot;Resultados_Primarias_Presidenciales_2021_CHILE.xlsx&quot;, and was downloaded
on October 24, 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(primary2)    
head(primary2) 
</code></pre>

<hr>
<h2 id='pspLlancahue'>Tree locations for a sample plot in the Llancahue experimental forest</h2><span id='topic+pspLlancahue'></span>

<h3>Description</h3>

<p>The Cartesian position, species, and diameter of trees within a plot were measured. The sample plot is  rectangular of 130 m by 70 m. Further details can be #' reviewed in the reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pspLlancahue)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>tree.code</dt><dd><p>Tree identificator</p>
</dd>
<dt>spp</dt><dd><p>species abreviation as follows: AP= Aextocicon puncatatum, EC=Eucryphia cordifolia, GA=Gevuina avellana, 
LP= Laureliopsis philippiana, LS= Laurelia sempervirens, ND=Nothofagus dombeyi, Ot=Other, PS=Podocarpus saligna</p>
</dd>
<dt>dbh</dt><dd><p>diameter at breast height, in cm</p>
</dd>
<dt>x.coord</dt><dd><p>Cartesian position in the X-axis, in m</p>
</dd>
<dt>y.coord</dt><dd><p>Cartesian position in the Y-axis, in m</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Daniel Soto at Universidad
de Aysen (Coyhaique, Chile).
</p>


<h3>References</h3>

<p>- Soto DP, Salas C, Donoso PJ, Uteau D. 2010. Heterogeneidad estructural y espacial de un bosque mixto dominado por
Nothofagus dombeyi despues de un disturbio parcial. Revista Chilena de Historia Natural 83(3): 335-347.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pspLlancahue)    
head(pspLlancahue) 
descstat(pspLlancahue$dbh)
boxplot(dbh~spp, data=pspLlancahue)
</code></pre>

<hr>
<h2 id='pspLlancahue2'>Ubicación cartesiana de árboles en el bosque de Llancahue</h2><span id='topic+pspLlancahue2'></span>

<h3>Description</h3>

<p>Corresponde a la posición cartesiana, especie, y diámetro de árboles en una parcela de muestreo
en el bosque de Llancahue, cerca de Valdivia, Chile. La parcela es rectangular con
dimensiones de 130 m por 70 m. Mayores antecedentes aparecen en las referencias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pspLlancahue2)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>arb.id</dt><dd><p>Identificador del árbol.</p>
</dd>
<dt>spp</dt><dd><p>Codificación de la especie como sigue: AP= Aextocicon puncatatum, EC=Eucryphia cordifolia, GA=Gevuina avellana, 
LP= Laureliopsis philippiana, LS= Laurelia sempervirens, ND=Nothofagus dombeyi, Ot=Other, PS=Podocarpus saligna.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>coord.x</dt><dd><p>Posición cartesiana en el eje-X, en m.</p>
</dd>
<dt>coord.y</dt><dd><p>Posición cartesiana en el eje-Y, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Prof. Daniel Soto de Universidad
de Aysen (Coyhaique, Chile).
</p>


<h3>References</h3>

<p>- Soto DP, Salas C, Donoso PJ, Uteau D. 2010. Heterogeneidad estructural y espacial de un bosque mixto dominado por
Nothofagus dombeyi después de un disturbio parcial. Revista Chilena de Historia Natural 83(3): 335-347.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pspLlancahue2)    
head(pspLlancahue2) 
descstat(pspLlancahue2$dap)
boxplot(dap~spp, data=pspLlancahue2)
</code></pre>

<hr>
<h2 id='pspRuca'>Tree spatial coordinates in the Rucamanque forest</h2><span id='topic+pspRuca'></span>

<h3>Description</h3>

<p>Tree-level variables and spatial coordinates in a permanent sample plot of 1 ha (100 x 100m) in the Rucamanque 
experimental forest, near Temuco, Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pspRuca)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables for the standing-alive trees as follows:
</p>

<dl>
<dt>tree.no</dt><dd><p>tree number</p>
</dd>											
<dt>species</dt><dd><p>Species name, &quot;N. obliqua&quot; is Nothofagus obliqua, &quot;Ap&quot; is Aexitocicum puncatatum, etc.</p>
</dd>
<dt>crown.class</dt><dd><p>Crown class (1: superior, 2: intermediate, 3; inferior)</p>
</dd>
<dt>dbh</dt><dd><p>diameter at breast-height, in cm</p>
</dd>
<dt>x.coord</dt><dd><p>Cartesian position at the X-axis, in m</p>
</dd>
<dt>y.coord</dt><dd><p>Cartesian position at the Y-axis, in m</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>Salas C, LeMay V, Nunez P, Pacheco P, and Espinosa A. 2006. Spatial patterns in an old-growth Nothofagus obliqua forest in 
south-central Chile. Forest Ecology and Management 231(1-3): 38-46. <a href="https://doi.org/10.1016/j.foreco.2006.04.037">doi:10.1016/j.foreco.2006.04.037</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pspRuca)    
head(pspRuca) 
table(pspRuca$species) 
</code></pre>

<hr>
<h2 id='pspRuca2'>Ubicación espacial de árboles en el bosque de Rucamanque</h2><span id='topic+pspRuca2'></span>

<h3>Description</h3>

<p>Medidas a nivel de árbol y coordenadas espaciales en un parcela de muestreo permanente de 1 ha (100 x 100m) en el
bosque de Rucamanque, cerca de Temuco, Chile. Mayores antecedentes en las referencias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pspRuca2)
</code></pre>


<h3>Format</h3>

<p>Las columnas describen características de los árboles vivos en pie, como sigue:
</p>

<dl>
<dt>arbol</dt><dd><p>Número del árbol</p>
</dd>											
<dt>especie</dt><dd><p>Nombre de la especie, &quot;N. obliqua&quot; es Nothofagus obliqua, &quot;Ap&quot; es Aexitocicum puncatatum, etc.</p>
</dd>
<dt>clase.copa</dt><dd><p>Clase de copa (1: superior, 2: intermedio, 3; inferior)</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm</p>
</dd>
<dt>coord.x</dt><dd><p>Posicion cartesiana en el eje X, en m</p>
</dd>
<dt>coord.y</dt><dd><p>Posicion cartesiana en el eje Y, en m</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Dr Christian Salas-Eljatib (Santiago, Chile).
</p>


<h3>References</h3>

<p>Salas C, LeMay V, Nunez P, Pacheco P, and Espinosa A. 2006. Spatial patterns in an old-growth Nothofagus obliqua forest in 
south-central Chile. Forest Ecology and Management 231(1-3): 38-46. <a href="https://doi.org/10.1016/j.foreco.2006.04.037">doi:10.1016/j.foreco.2006.04.037</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pspRuca2)    
table(pspRuca2$especie) 
</code></pre>

<hr>
<h2 id='ptaeda'>Height growth of Pinus taeda (Loblolly pine) trees</h2><span id='topic+ptaeda'></span>

<h3>Description</h3>

<p>The Loblolly data frame has 84 rows and tree columns of records of the tree height growth of Loblolly pine trees. This dataframe
is a slight modification to the original dataframe &quot;Loblolly&quot; from the datasets R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ptaeda)
</code></pre>


<h3>Format</h3>

<p>An object of class c(&quot;nfnGroupedData&quot;, &quot;nfGroupedData&quot;, &quot;groupedData&quot;, &quot;data.frame&quot;) containing the following columns:
</p>

<dl>
<dt>seed.id</dt><dd><p>an ordered factor indicating the seed source for the tree. The ordering is according to increasing maximum height.</p>
</dd>
<dt>age</dt><dd><p>a numeric vector of tree ages, in yr.</p>
</dd>
<dt>toth</dt><dd><p>a numeric vector of tree heights, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Pinheiro, J. C. and Bates, D. M. (2000) Mixed-effects Models in S and S-PLUS. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ptaeda)
plot(toth ~ age, data = ptaeda, subset = seed.id == 329,
     xlab = "Age (yr)", las = 1,
     ylab = "Height (m)")
</code></pre>

<hr>
<h2 id='ptaeda2'>Crecimiento en altura de Pinus taeda</h2><span id='topic+ptaeda2'></span>

<h3>Description</h3>

<p>Esta dataframe contiene 84 folas y tres columnas de crecimiento en altura de 
árboles de Pinus taeda (Loblolly pine). Es una modificación de la dataframe &quot;Loblolly&quot; del paquete 'datasets' de R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ptaeda2)
</code></pre>


<h3>Format</h3>

<p>An object of class c(&quot;nfnGroupedData&quot;, &quot;nfGroupedData&quot;, &quot;groupedData&quot;, &quot;data.frame&quot;) containing the following columns:
</p>

<dl>
<dt>semilla.id</dt><dd><p>Un factor indicando el origen de la semilla del árbol.</p>
</dd>
<dt>edad</dt><dd><p>Edad del árbol, en años.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Pinheiro, J. C. and Bates, D. M. (2000) Mixed-effects Models in S and S-PLUS. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ptaeda2)
plot(atot ~ edad, data = ptaeda2, subset = semilla.id == 329,
     xlab = "Edad (años)", las = 1,
     ylab = "Altura (m)")
</code></pre>

<hr>
<h2 id='pvalt'>Obtain the P-value for a Standard t-distributed random variable</h2><span id='topic+pvalt'></span>

<h3>Description</h3>

<p>Function to compute the P-value for a Standard t-distributed random variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalt(t.value, df, decnum = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvalt_+3A_t.value">t.value</code></td>
<td>
<p>A numeric random variable following a t-student pdf distribution.</p>
</td></tr>
<tr><td><code id="pvalt_+3A_df">df</code></td>
<td>
<p>degrees of freedom of the random variable following a t-student pdf distribution.</p>
</td></tr>
<tr><td><code id="pvalt_+3A_decnum">decnum</code></td>
<td>
<p>the number of decimals to be used in the output. The default is set to 5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is suited to compute the P-value for any random variable following a Standard 
t probability density function. For instance, to obtain the p-value in a t-test.
</p>


<h3>Value</h3>

<p>The function returns the P-value or probability of getting a value as large as t.value.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load dataset
df &lt;- datana::araucaria
#
## Computes the t-test statistics (from the 'stats' package)
t.value &lt;- stats::t.test(df$hdom)
t.v &lt;- as.numeric(t.value$statistic)
deg.f &lt;- as.numeric(t.value$parameter)

## Obtaining the p value
pvalt(t.v,deg.f)

</code></pre>

<hr>
<h2 id='pvalz'>Obtain the P-value for a Standard Gaussian random variable</h2><span id='topic+pvalz'></span>

<h3>Description</h3>

<p>Function to computes the P-value for a Standard Gaussian random variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalz(zval, decnum = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvalz_+3A_zval">zval</code></td>
<td>
<p>A numeric random variable following a Standard Gaussian distribution.</p>
</td></tr>
<tr><td><code id="pvalz_+3A_decnum">decnum</code></td>
<td>
<p>the number of decimals to be used in the output. The default is set to 5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is suited to compute the P-value for any random variable following a Standard 
Gaussian probability density function.
</p>


<h3>Value</h3>

<p>This function returns the P-value or probability of getting a value as large as 'zval'.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
pvalz(1.96)

</code></pre>

<hr>
<h2 id='radiatapl'>Sampling plots data from a Pinus radiata plantation near Capitan Pastene, Region de La Araucania, Chile.</h2><span id='topic+radiatapl'></span>

<h3>Description</h3>

<p>Tree-level information collected within sample plots in a forestry plantation of Pinus radiata near Capitan Pastene, 
Southern Chile. Sample plots size is 150 150 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(radiatapl)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>plot</dt><dd><p>Plot number identification.</p>
</dd>
<dt>tree</dt><dd><p>Tree number identification.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>heigth</dt><dd><p>Total height, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Mr. Mauricio Lobos-Beneventi (Temuco, Chile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(radiatapl)    
head(radiatapl) 
</code></pre>

<hr>
<h2 id='radiatapl2'>Datos a nivel de árbol de parcelas de muestreo en plantaciones de Pinus radiata</h2><span id='topic+radiatapl2'></span>

<h3>Description</h3>

<p>Es un listado de árboles con características medidas dentro de unidades de muestreo en una plantacion de Pinus radiata cercana a Capitan Pastene, 
Region de la Araucania, Chile. Las parcelas de muestreo tienen 150 m<code class="reqn">^{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(radiatapl2)
</code></pre>


<h3>Format</h3>

<p>Los datos contienen las siguientes columnas
</p>

<dl>
<dt>parce</dt><dd><p>Número de identificación de la parcela de muestreo.</p>
</dd>
<dt>arbol</dt><dd><p>Número de identificación del árbol dentro de la parcela.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a los 1.3 m en el fuste, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m. Solo registrada para algunos árboles muestra.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos son cortesia del Ing. Forestal Mauricio Lobos-Beneventi (Temuco, Chile).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(radiatapl2)    
head(radiatapl2) 
</code></pre>

<hr>
<h2 id='raulihg'>Height growth of Nothofagus alpina trees in Chile.</h2><span id='topic+raulihg'></span>

<h3>Description</h3>

<p>Time series data of height for rauli (Nothofagus alpina) trees in south-central Chile. These sampled 
trees are part of the ones used in Salas-Eljatib (2021, Ecological Applications). The full citation
is provided below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(raulihg)
</code></pre>


<h3>Format</h3>

<p>The data frame contains four variables as follows:
</p>

<dl>
<dt>tree.code</dt><dd><p>tree id code</p>
</dd>											
<dt>spp</dt><dd><p>species common name</p>
</dd>
<dt>bha.t</dt><dd><p>breast-height age, in yrs.</p>
</dd>
<dt>h.t</dt><dd><p>total height, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Christian Salas-Eljatib (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas-Eljatib C. 2021. An approach to quantify climate-productivity relationships: an
example from a widespread Nothofagus forest. Ecological Applications 31(4): e02285.  <a href="https://doi.org/10.1002/eap.2285">doi:10.1002/eap.2285</a>
</p>
<p>- Salas-Eljatib, C. 2021. Time series height-data for Nothofagus alpina trees. <a href="https://doi.org/10.6084/m9.figshare.13521602.v5">doi:10.6084/m9.figshare.13521602.v5</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(raulihg)    
head(raulihg) 
</code></pre>

<hr>
<h2 id='raulihg2'>Crecimiento en altura de árboles de Nothofagus alpina.</h2><span id='topic+raulihg2'></span>

<h3>Description</h3>

<p>Datos de series de tiempo de altura para árboles muestreados de Nothofagus alpina (rauli) en el centro-sur de Chile. Estos árboles son parte
de los usados en Salas-Eljatib (2021, Ecological Applications). La cita completa se da en referencias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(raulihg2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables de nivel individual, como se describen a continuacion::
</p>

<dl>
<dt>tree.code</dt><dd><p>Codigo del árbol</p>
</dd>											
<dt>spp</dt><dd><p>Nombre comun especie</p>
</dd>
<dt>bha.t</dt><dd><p>Edad a la altura del pecho, en años.</p>
</dd>
<dt>h.t</dt><dd><p>Altura total, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos cedidos por el Prof. Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>- Salas-Eljatib C. 2021. An approach to quantify climate-productivity relationships: an
example from a widespread Nothofagus forest. Ecological Applications 31(4): e02285. <a href="https://doi.org/10.1002/eap.2285">doi:10.1002/eap.2285</a>
</p>
<p>- Salas-Eljatib C. 2021. Time series height-data for Nothofagus alpina trees. <a href="https://doi.org/10.6084/m9.figshare.13521602.v5">doi:10.6084/m9.figshare.13521602.v5</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(raulihg2)    
head(raulihg2) 
</code></pre>

<hr>
<h2 id='regNothofagus'>Contains information about regeneration of Nothofagus seedlings.</h2><span id='topic+regNothofagus'></span>

<h3>Description</h3>

<p>Dataset contains 442 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(regNothofagus)
</code></pre>


<h3>Format</h3>

<p>Contains 15 variables, as follows:
</p>

<dl>
<dt>site</dt><dd><p>Id site of study.</p>
</dd>
<dt>plot</dt><dd><p>Number of plot.</p>
</dd>
<dt>scar</dt><dd><p>Scarification in percentage of total area.</p>
</dd>
<dt>x.trans.total</dt><dd><p>Transmitted radiation in percentage.</p>
</dd>
<dt>kPa</dt><dd><p>Soil resistance to penetration.</p>
</dd>
<dt>SWC</dt><dd><p>Soil water content.</p>
</dd>
<dt>SM</dt><dd><p>Exposed mineral soil.</p>
</dd>
<dt>litter</dt><dd><p>Litter cover in percentage.</p>
</dd>
<dt>CWD</dt><dd><p>Ocular estimation in the regeneration plot in percentage.</p>
</dd>
<dt>MT</dt><dd><p>Microtopography. 1 plane, 2 convex, 3 concave, 4 mixed (convex and concave) in the regeneration plot.</p>
</dd>
<dt>S</dt><dd><p>Ground-layer vascular species richness in the regeneration plot..</p>
</dd>
<dt>LLES</dt><dd><p>Long-lived early-seral tree species (N. dombeyi , N. alpina , Nothofagus pumilio ).</p>
</dd>
<dt>SLES</dt><dd><p>Short-lived early-seral plants (Ribes spp. and Fuchsia sp).</p>
</dd>
<dt>LLLS</dt><dd><p>Long-lived late-seral tree species (L. philippiana and Dasyphyllum diacantaoides ).</p>
</dd>
<dt>log.bam</dt><dd><p>Logarithm of the cover of bamboo (%) in the regeneration plot.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were obtained from the DRYAD repository at <a href="https://doi.org/10.5061/dryad.3q977">doi:10.5061/dryad.3q977</a>
</p>


<h3>References</h3>

<p>Soto D, Puettmann K.2018. Topsoil removal through scarification improves natural regeneration
in high-graded Nothofagus old-growth forests. Journal Applied Ecology. 55: 967- 976.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(regNothofagus)    
head(regNothofagus) 
</code></pre>

<hr>
<h2 id='simula'>Simulated yield of forestry plantations of exotic species in Chile.</h2><span id='topic+simula'></span>

<h3>Description</h3>

<p>The yield tables of simulated plantations of Pinus radiata, Eucalyptus globulus, and Eucalyptus nitens
are obtained from the Radiata simulator and EucaSim simulator built in Chile. Several stand-level variables 
are part of the output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(simula)
</code></pre>


<h3>Format</h3>

<p>Contains stand-level variables, as follows:
</p>

<dl>
<dt>species</dt><dd><p>&quot;P. radiata&quot; is Pinus radiata, &quot;E. globulus&quot; is Eucalyptus globulus, and &quot;E. nitens&quot; is
Eucalyptus nitens.</p>
</dd>
<dt>age</dt><dd><p>plantation age, in years</p>
</dd>
<dt>tph </dt><dd><p>Tree density, in trees/ha</p>
</dd>
<dt>gha </dt><dd><p>Basal area, in m<code class="reqn">^{2}</code>/ha</p>
</dd>
<dt>toph</dt><dd><p>Dominant height, in m</p>
</dd>
<dt>qmd</dt><dd><p>quadratic mean diameter, in cm</p>
</dd>
<dt>totvol</dt><dd><p>gross stand volume, in m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>viu.10</dt><dd><p>stand volume below an utilizacion index of 10 cm, in m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>viu.15</dt><dd><p>stand volume below an utilizacion index of 15 cm, in m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>viu.20</dt><dd><p>stand volume below an utilizacion index of 20 cm, in m<code class="reqn">^{3}</code>/ha</p>
</dd> 
<dt>viu.25</dt><dd><p>stand volume below an utilizacion index of 25 cm, in m<code class="reqn">^{3}</code>/ha</p>
</dd> 
</dl>



<h3>Source</h3>

<p>The data were obtained as outputs for plantations without management in Chile. The academic
version of the simulator was used. You can visit mnssimulacion.cl
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simula)    
</code></pre>

<hr>
<h2 id='skew'>Computes the skewness of a numeric vector</h2><span id='topic+skew'></span>

<h3>Description</h3>

<p>The skewness is about the departure from symmetry of a frequency distribution. Therefore,
It is about asymmetry. One way to assess asymmetry of a random variable is to compute
an statistics representing its skewness. The current function an dimensionless statistics
of the skewness of given vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>skew(x, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="skew_+3A_x">x</code></td>
<td>
<p>A numeric vector representing a random variable.</p>
</td></tr>
<tr><td><code id="skew_+3A_na.rm">na.rm</code></td>
<td>
<p>Logical value to remove NA values. The default is set to TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The skewness of a random variable is the third moment of the standardized variable.
There are several ways of parameterizing an skewness estimator, such as depending on the 
third moment and the standard deviation of the random variable.
</p>


<h3>Value</h3>

<p>The value of the the skewness of given vector
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y.var&lt;-rnorm(100);x.var&lt;-rbeta(100,.2,2)
skew(y.var)
skew(x.var)

</code></pre>

<hr>
<h2 id='slashpine'>Biomass dataset</h2><span id='topic+slashpine'></span>

<h3>Description</h3>

<p>Dataset that contains nine pairs of columns with information about biomass of 40 samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(slashpine)
</code></pre>


<h3>Format</h3>

<p>The data frame contains nine variables as follows:
</p>

<dl>
<dt>tree_id</dt><dd><p>tree code</p>
</dd>											
<dt>dbh</dt><dd><p>diameter</p>
</dd>
<dt>h</dt><dd><p>heigth</p>
</dd>
<dt>lcl</dt><dd><p>live crown lenght</p>
</dd>
<dt>age</dt><dd><p>age tree</p>
</dd>
<dt>wood</dt><dd><p>wood biomass</p>
</dd>
<dt>bark</dt><dd><p>bark biomass</p>
</dd>
<dt>crown</dt><dd><p>crown biomass</p>
</dd>
<dt>tree</dt><dd><p>tree biomass</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data were provided by Dr Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>Parresol BR. 2001. Additivity of nonlinear biomass equations. Canadian Journal of For Research, 31:865-878.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(slashpine)    
head(slashpine) 
</code></pre>

<hr>
<h2 id='slashpine2'>Biomasa</h2><span id='topic+slashpine2'></span>

<h3>Description</h3>

<p>Dataset que contiene nueve pares de columnas con informacion sobre la biomasa de 40 árboles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(slashpine2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuación:
</p>

<dl>
<dt>tree_id</dt><dd><p>Identifcador del árbol</p>
</dd>											
<dt>dbh</dt><dd><p>diámetro</p>
</dd>
<dt>h</dt><dd><p>altura total</p>
</dd>
<dt>lcl</dt><dd><p>largo de copa</p>
</dd>
<dt>age</dt><dd><p>edad árbol</p>
</dd>
<dt>wood</dt><dd><p>biomasa madera</p>
</dd>
<dt>bark</dt><dd><p>biomasa corteza</p>
</dd>
<dt>crown</dt><dd><p>biomasa copa</p>
</dd>
<dt>tree</dt><dd><p>biomasa total</p>
</dd>
</dl>



<h3>Source</h3>

<p>Datos fueron contribuidos por el Dr Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>Parresol BR. 2001. Additivity of nonlinear biomass equations. Canadian Journal of For Research, 31:865-878.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(slashpine2)    
head(slashpine2) 
</code></pre>

<hr>
<h2 id='sludge'>Sludge data are at different cities, with a value of concentration zinc.</h2><span id='topic+sludge'></span>

<h3>Description</h3>

<p>Dataset contains 36 observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sludge)
</code></pre>


<h3>Format</h3>

<p>Contains four variables, as follows:
</p>

<dl>
<dt>city</dt><dd><p>Name of city.</p>
</dd>
<dt>rate</dt><dd><p>Concentration rate of sludge.</p>
</dd>
<dt>zinc</dt><dd><p>Value of concentration ( in ppm).</p>
</dd>
<dt>trt.comb</dt><dd><p>Combination between city and rate factors.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided from.
</p>


<h3>References</h3>

<p>not yet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sludge)    
head(sludge) 
</code></pre>

<hr>
<h2 id='snaspeChile'>On the National System of State Protected Wild Areas (SNASPE) of Chile.</h2><span id='topic+snaspeChile'></span>

<h3>Description</h3>

<p>Dataset contains the protected wild areas of Chile that are part of the National System of State Protected Wild Areas (SNASPE).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snaspeChile)
</code></pre>


<h3>Format</h3>

<p>Contains the following variables:
</p>

<dl>
<dt>unit.id</dt><dd><p>Number for the unit.</p>
</dd>
<dt>unit</dt><dd><p>Name of the protected area.</p>
</dd>
<dt>category</dt><dd><p>Category of the unit. It can be either a National Park, a National Reserve or a Natural Monument.</p>
</dd>
<dt>commune</dt><dd><p>Name of the commune (the smallest Chilean territorial division) where the unit is located.</p>
</dd>
<dt>province</dt><dd><p>Province where the comunne is located (one territorial division level above the commune).</p>
</dd>
<dt>region</dt><dd><p>Region where the province is located (one territorial division level above the province and the biggest Chilean territorial division).</p>
</dd>
<dt>perim.km</dt><dd><p>Perimeter, in km.</p>
</dd>
<dt>area.ha</dt><dd><p>Area, in hectares.</p>
</dd>
<dt>area.m2</dt><dd><p>Area, in m<code class="reqn">^{2}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>These data is freely available at
<a href="https://ide.minagri.gob.cl/geoweb/2019/11/21/medio-ambiente/">https://ide.minagri.gob.cl/geoweb/2019/11/21/medio-ambiente/</a>
</p>


<h3>References</h3>

<p>The SNASPE has been created and is currently managed by the National Forest Corporation (CONAF).
More information and documentation can be found at
<a href="https://www.conaf.cl">https://www.conaf.cl</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snaspeChile)    
head(snaspeChile) 
</code></pre>

<hr>
<h2 id='snaspeChile2'>Sistema nacional de areas protegidas del estado (SNASPE) de Chile</h2><span id='topic+snaspeChile2'></span>

<h3>Description</h3>

<p>Contiene variables general de las unidades del sistema de areas protegidas por el estado de 
Chile (SNASPE).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(snaspeChile2)
</code></pre>


<h3>Format</h3>

<p>Contiene las siguientes variables para cada unidad del SNASPE:
</p>

<dl>
<dt>uni.id</dt><dd><p>número indentificador de la unidad.</p>
</dd>
<dt>unidad</dt><dd><p>Nombre de la unidad.</p>
</dd>
<dt>categoria</dt><dd><p>Categoría de la unidad. It can be either a National Park, a National Reserve or a Natural Monument.</p>
</dd>
<dt>comuna</dt><dd><p>Name of the commune (the smallest Chilean territorial division) where the unit is located.</p>
</dd>
<dt>province</dt><dd><p>Province where the comunne is located (one territorial division level above the commune).</p>
</dd>
<dt>region</dt><dd><p>Region where the province is located (one territorial division level above the province and the biggest Chilean territorial division).</p>
</dd>
<dt>perim.km</dt><dd><p>Perimeter, in km.</p>
</dd>
<dt>area.ha</dt><dd><p>Area, in hectares.</p>
</dd>
<dt>area.m2</dt><dd><p>Area, in m<code class="reqn">^{2}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Estos datos fueron obtenidos desde 
<a href="https://ide.minagri.gob.cl/geoweb/2019/11/21/medio-ambiente/">https://ide.minagri.gob.cl/geoweb/2019/11/21/medio-ambiente/</a>
</p>


<h3>References</h3>

<p>EL SNASPE esta bajo la administracion de la Corporacion Nacional Forestal (CONAF) de Chile.
Mayor informacion y documentacion se puede encontrar en
<a href="https://www.conaf.cl">https://www.conaf.cl</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(snaspeChile2)    
head(snaspeChile2) 
</code></pre>

<hr>
<h2 id='soiltreat'>Soil treatment experiment in tree seedlings</h2><span id='topic+soiltreat'></span>

<h3>Description</h3>

<p>A test was made of the effect of three soil treatments on the height 
growth of 2-year-old seedlings. Treatments were assigned at random 
to the three plots within each of 11 blocks. Each plot was made 
up of 50 seedlings.  Average 5-year height growth was the
criterion for evaluating treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(soiltreat)
</code></pre>


<h3>Format</h3>

<p>Contains the four following columns, at the plot-level,
</p>

<dl>
<dt>block</dt><dd><p>Block unit.</p>
</dd>
<dt>treat</dt><dd><p>Treatment level.</p>
</dd>
<dt>ini.h</dt><dd><p>Initial height, in m.</p>
</dd>
<dt>inc.h</dt><dd><p>Increment in height during 5-year, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Table in page 71 of Freese (1967).  The data were entered by
Nayeli Ramirez, a former student of Prof. Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>- Freese, F 1967. Elementary statistical methods for foresters. Agriculture Handbook 3171, USDA 
Forest Service.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soiltreat)    
head(soiltreat) 
tapply(soiltreat$inc.h,soiltreat$treat,summary)
tapply(soiltreat$inc.h,soiltreat$treat,sd)
</code></pre>

<hr>
<h2 id='soiltreat2'>Tratamientos del suelo en el crecimiento de plantulas.</h2><span id='topic+soiltreat2'></span>

<h3>Description</h3>

<p>Un experimento sobre el efecto de tres tratamientos del suelo en el crecimiento 
en altura de plantulas de 2-años de edad. Los tratamientos fueron asignados aleatoriamente
a tres parcelas dentro de cada uno de 11 bloques. Cada parcela esta constituida por 
hasta 50 plantulas. El promedio del incremento en altura de los ultimos 5 años fue
la variable de interes para evaluar los tratamientos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(soiltreat2)
</code></pre>


<h3>Format</h3>

<p>Los datos, a nivel de parcela, tienen las siguientes columnas, 
</p>

<dl>
<dt>bloque</dt><dd><p>Bloque del experimento.</p>
</dd>
<dt>tmo</dt><dd><p>Factor tratamiento, medido en tres  nivels.</p>
</dd>
<dt>alt.ini</dt><dd><p>Altura initial, rn m.</p>
</dd>
<dt>alt.inc</dt><dd><p>Incremento en altura durante los últimos cinco años, en m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Cuadro de la página 71 de Freese (1967). Los datos fueron digitados por Nayeli Ramirez, una 
estudiante del Prof. Christian Salas-Eljatib.
</p>


<h3>References</h3>

<p>- Freese, F 1967. Elementary statistical methods for foresters. Agriculture Handbook 3171, USDA 
Forest Service.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soiltreat2)    
head(soiltreat2) 
tapply(soiltreat2$alt.inc,soiltreat2$tmo,summary)
tapply(soiltreat2$alt.inc,soiltreat2$tmo,sd)
</code></pre>

<hr>
<h2 id='spatAustria'>Tree locations for several plots of Norway spruce in Austria</h2><span id='topic+spatAustria'></span>

<h3>Description</h3>

<p>The Cartesian position, species, year, ID tree , and diameter of trees within a plot were measured.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(spatAustria)
</code></pre>


<h3>Format</h3>

<p>Contains cartesian position of trees, and covariates, in sample plots, as follows:
</p>

<dl>
<dt>plot.code</dt><dd><p>Plot identificator</p>
</dd>
<dt>tree.code</dt><dd><p>Tree identificator</p>
</dd>
<dt>spp.name</dt><dd><p>species abreviation as follows: PCAB=Picea abies, FASY= Fagus sylvatica, QCPE=Quercus petraea , PNSY= Pinus Sylvestris, LADC=Larix decidua</p>
</dd>
<dt>x.coord</dt><dd><p>Cartesian position in the X-axis, in m</p>
</dd>
<dt>y.coord</dt><dd><p>Cartesian position in the Y-axis, in m</p>
</dd>
<dt>year</dt><dd><p>Measurement year</p>
</dd>
<dt>dbh</dt><dd><p>diameter at breast-height, in cm</p>
</dd>
</dl>



<h3>References</h3>

<p>- Kindermann G. Kristofel F, Neumann M, Rossler G, Ledermann T &amp; Schueler. 2018. 109 years of forest growth measurements from
individual Norway spruce trees. Sci. Data 5:180077 <a href="https://doi.org/10.1038/sdata.2018.77">doi:10.1038/sdata.2018.77</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spatAustria)    
head(spatAustria)
pos&lt;-spatAustria
oldpar&lt;-par(mar=c(4,4,0,0))
bord&lt;-data.frame(x=c(min(pos$x.coord),max(pos$x.coord),min(pos$x.coord),max(pos$x.coord)),
               y=c(min(pos$y.coord),min(pos$y.coord),max(pos$y.coord),min(pos$y.coord)))
plot(bord,type="n", xlab="x (m)", ylab="y (m)", asp=1, bty='n')
points(pos$x.coord,pos$y.coord,col=pos$plot.code,cex=0.5) 
par(oldpar)
</code></pre>

<hr>
<h2 id='speciesList'>Names and other information of plant species (mainly trees)</h2><span id='topic+speciesList'></span>

<h3>Description</h3>

<p>This data set provides names (taxonomy), of plant species. Includes codes and name abbreviations used by the Biometrics group at the Forest Biometrics and Modelling Lab, Universidad de Chile, Santiago, Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(speciesList)
</code></pre>


<h3>Format</h3>

<p>A data frame with 63 observations on 31 variables
</p>

<dl>
<dt>nesp</dt><dd><p>Unique correlative specie number</p>
</dd>
<dt>spp.ci.name</dt><dd><p>Species scientific name</p>
</dd>
<dt>spp.ci.abb</dt><dd><p>Species scientific name abbreviation</p>
</dd>
<dt>common.name</dt><dd><p>Species common name. No blank spaces,  no special characters</p>
</dd>
<dt>common.nameBlank</dt><dd><p>Species common name. With blank spaces,  no special characters</p>
</dd>
<dt>esp</dt><dd><p>Species code: code given by CEM Biometrics to identify species for different processing routines</p>
</dd>
<dt>common.nameLatex</dt><dd><p>Species common name formatted for Latex</p>
</dd>
<dt>nTaxon</dt><dd><p>Unique number of the taxon (i.e., species)</p>
</dd>
<dt>kingdom</dt><dd><p>Taxonomic rank Kingdom. In this datase, all species belong to the Kingdom Plantae</p>
</dd>
<dt>division</dt><dd><p>Taxonomic rank division or phylum within the Kingdom</p>
</dd>
<dt>class</dt><dd><p>Taxonomic rank Class within the Kingdom</p>
</dd>
<dt>order</dt><dd><p>Taxonomic rank Order within the Class</p>
</dd>
<dt>family</dt><dd><p>Taxonomic rank Family within the Order</p>
</dd>
<dt>spp.ci.full</dt><dd><p>Full scientific name including author</p>
</dd>
<dt>genus</dt><dd><p>Taxonomic rank Genus within the Family</p>
</dd>
<dt>epithet</dt><dd><p>Specific epithet</p>
</dd>
<dt>sppAuthor</dt><dd><p>Species author</p>
</dd>
<dt>subSpp</dt><dd><p>Subespecies: one of two or more populations of a species varying from one another by morphological characteristics</p>
</dd>
<dt>subSppAuthor</dt><dd><p>Subespecies author</p>
</dd>
<dt>varSpp</dt><dd><p>Species variety or varietas</p>
</dd>
<dt>varSppAuthor</dt><dd><p>Variety author</p>
</dd>
<dt>formSpp</dt><dd><p>Form or forma</p>
</dd>
<dt>formSppAuthor</dt><dd><p>Form author</p>
</dd>
<dt>commonNamesList</dt><dd><p>List of common names per species, separated by commas</p>
</dd>
<dt>synonyms</dt><dd><p>Synonyms of the scientific name by which the species has been or is known</p>
</dd>
<dt>borCountries</dt><dd><p>Border countries given the species distribution range</p>
</dd>
<dt>habit</dt><dd><p>Habit. The general appearance, growth form, or architecture e.g., tree, shrub, grass</p>
</dd>
<dt>lifeCycle</dt><dd><p>Life cycle</p>
</dd>
<dt>statusOri</dt><dd><p>Status according to the species origin: Native or Endemic</p>
</dd>
<dt>regDist</dt><dd><p>Distribution range of the species, within Chile administrative regions</p>
</dd>
<dt>elevRange</dt><dd><p>Distribution range of the species, in terms of elevation. Meters above sea level</p>
</dd>
<dt>notes</dt><dd><p>Notes</p>
</dd>
</dl>



<h3>Source</h3>

<p>Data provided from <a href="https://investigacion.conaf.cl/repositorio/documento/ficha-repositorio.php?redo_id=1080946">https://investigacion.conaf.cl/repositorio/documento/ficha-repositorio.php?redo_id=1080946</a>
</p>


<h3>References</h3>

<p>Proyecto 004/2016 Lista sistematica actualizada de la flora vascular nativa de Chile, origen y distribucion geografica. VII Concurso del Fondo de Investigacion del Bosque Nativo
</p>

<hr>
<h2 id='sppAbundance'>Contains information of abundance of plant species in the central-southern Andes of Chile.</h2><span id='topic+sppAbundance'></span>

<h3>Description</h3>

<p>Abundance of plant species [50 total] (at parcel scale [100 m<code class="reqn">^{2}</code>]) in burned Araucaria-Nothofagus 
forests with different levels of fire severity (ie, unburned = unburned, low_sev = low severity, 
mid_sev = medium severity , high_sev = high severity) in the China Muerta National Reserve, 
Andes of central-southern Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sppAbundance)
</code></pre>


<h3>Format</h3>

<p>Contains 6 variables, as follows:
</p>

<dl>
<dt>sp.name</dt><dd><p>name of specie.</p>
</dd>
<dt>sp.code.name</dt><dd><p>code of specie</p>
</dd>
<dt>unburned</dt><dd><p>Abundance of plants unburned.</p>
</dd>
<dt>low.sev</dt><dd><p>Abundance of plants for low severity of burned.</p>
</dd>
<dt>mid.sev</dt><dd><p>Abundance of plants for middle severity of burned.</p>
</dd>
<dt>high.sev</dt><dd><p>Abundance of plants for high severity of burned.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Dr Andres Fuentes-Ramirez at the Universidad 
of La Frontera (Temuco, Chile)
</p>


<h3>References</h3>

<p>- Fuentes-Ramirez A, Salas-Eljatib C, Gonzalez M, Urrutia J, Arroyo P, Santibanez P. 2020. Initial response of understorey vegetation and tree
regeneration to a mixed-severity fire in old-growth Araucaria-Nothofagus forests.Applied 
Vegetation Science. 23:210-222.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sppAbundance)    
head(sppAbundance) 
</code></pre>

<hr>
<h2 id='sppTraits'>Contains information of functional traits of species.</h2><span id='topic+sppTraits'></span>

<h3>Description</h3>

<p>Dataset contains 48 observations about about functional trait values for each of the 48 study species, 
including 23 evergreen and 25 deciduous.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sppTraits)
</code></pre>


<h3>Format</h3>

<p>Contains 17 variables, as follows:
</p>

<dl>
<dt>sp</dt><dd><p>Abbreviated name of specie.</p>
</dd>
<dt>sp.name</dt><dd><p>Name of specie.</p>
</dd>
<dt>family</dt><dd><p>Family of specie.</p>
</dd>
<dt>genus</dt><dd><p>Genus of specie.</p>
</dd>
<dt>phyl</dt><dd><p>Type of phylogeny.</p>
</dd>
<dt>l.hab</dt><dd><p>Type of leaf habit.</p>
</dd>
<dt>leaf</dt><dd><p>Type of leaf.</p>
</dd>
<dt>lt</dt><dd><p>.</p>
</dd>
<dt>lma</dt><dd><p>Leaf mass area.</p>
</dd>
<dt>amass</dt><dd><p>Photosynthetic capacity per unit leaf mass.</p>
</dd>
<dt>n.mass</dt><dd><p>Leaf N content per unit mass.</p>
</dd>
<dt>pmass</dt><dd><p>Leaf P content per unit mass.</p>
</dd>
<dt>l.lifespan</dt><dd><p>Leaf life span.</p>
</dd>
<dt>l.length</dt><dd><p>Leaf length.</p>
</dd>
<dt>sem</dt><dd><p>Seed mass.</p>
</dd>
<dt>wd</dt><dd><p>Wood density.</p>
</dd>
<dt>max.h</dt><dd><p>Maximum height.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided from DRYAD repository
</p>


<h3>References</h3>

<p>- Ameztegui A, Paquette A, Shipley B, Heym M, Messier C, Gravel D. 2016 . Shade tolerance and 
the functional trait: demography relationship in temperate and boreal forests. Functional Ecology 
31: 821-830.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sppTraits)    
head(sppTraits) 
</code></pre>

<hr>
<h2 id='standLleuque'>Plot-level data with variables from Andean Prumnopitys forests</h2><span id='topic+standLleuque'></span>

<h3>Description</h3>

<p>Data on density, basal area, mean square diameter and other variables of 24 plots for Lleuque is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(standLleuque)
</code></pre>


<h3>Format</h3>

<p>The data frame contains seven variables as follows:
</p>

<dl>
<dt>rodal</dt><dd><p>number of stand</p>
</dd>											
<dt>plot.id</dt><dd><p>code of plot</p>
</dd>
<dt>nha</dt><dd><p>Density of plot</p>
</dd>
<dt>gha</dt><dd><p>Basal area of plot</p>
</dd>
<dt>qmd</dt><dd><p>Quadratic mean diameter of plot</p>
</dd>
<dt>toph</dt><dd><p>Dominant height of plot</p>
</dd>
<dt>structure</dt><dd><p>Forest structure level: open, secondary adult, pure</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Rodrigo Vargas-Gaete at Universidad
de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Vargas-Gaete R, Salas-Eljatib C, Penneckamp D, Neira Z, Diez MC, Vargas-Picón, R. 2020. Estructura 
y regeneración de bosques de Prumnopitys andina en los Andes del sur de Chile. Gayana Botánica, 77(1), 48-58.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(standLleuque)    
head(standLleuque) 
</code></pre>

<hr>
<h2 id='standLleuque2'>Datos con variables a nivel de parcela de bosques de Prumnopitys andina</h2><span id='topic+standLleuque2'></span>

<h3>Description</h3>

<p>Se proporciona informacion de densidad, area basal, diámetro medio cuadratico  y otras variables de 24 parcelas para Lleuque.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(standLleuque2)
</code></pre>


<h3>Format</h3>

<p>Variables se describen a continuacion::
</p>

<dl>
<dt>rodal</dt><dd><p>Número de rodal</p>
</dd>											
<dt>plot.id</dt><dd><p>Codigo de parcela</p>
</dd>
<dt>nha</dt><dd><p>Densidad de parcela</p>
</dd>
<dt>gha</dt><dd><p>Area basal de parcela</p>
</dd>
<dt>qmd</dt><dd><p>Diámetro medio cuadratico</p>
</dd>
<dt>toph</dt><dd><p>Altura dominante</p>
</dd>
<dt>estructura</dt><dd><p>Estructura del bosque. Abiero, secundario adulto o puro</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Rodrigo Vargas-Gaete at Universidad
de La Frontera (Temuco, Chile).
</p>


<h3>References</h3>

<p>Vargas-Gaete R, Salas-Eljatib C, Penneckamp D, Neira Z, Diez MC, Vargas-Picón, R. 2020. Estructura 
y regeneración de bosques de Prumnopitys andina en los Andes del sur de Chile. Gayana Botánica, 77(1), 48-58.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(standLleuque2)    
head(standLleuque2) 
</code></pre>

<hr>
<h2 id='timeserplot'>Produces a time series plot</h2><span id='topic+timeserplot'></span>

<h3>Description</h3>

<p>Produces a time series plot, of variable 'y' as a function of 'x' by an observational unit factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeserplot(
  data = data,
  y = y,
  x = x,
  obs.unit = obs.unit,
  factor1 = NA,
  factor2 = NA,
  only.lines = FALSE,
  ylab = NA,
  xlab = NA,
  linetype.lab = NA,
  factor2.line = TRUE,
  factor2.col = FALSE,
  max.y.all = NA,
  levels.i.want = FALSE,
  col.lev.i.want = FALSE,
  col.lines = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="timeserplot_+3A_data">data</code></td>
<td>
<p>a dataframe with at least tree columns representing the response variable (&quot;y&quot;),
the main predictor variable (&quot;x&quot;), and a variable indicating the observational unit (&quot;obs.unit&quot;).</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_y">y</code></td>
<td>
<p>a character giving the column name of the response variable or variable of interest.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_x">x</code></td>
<td>
<p>a character giving the column name of the main predictor variable. Generally this variable
is time.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_obs.unit">obs.unit</code></td>
<td>
<p>a character giving the column name containing the info of the observational unit.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_factor1">factor1</code></td>
<td>
<p>an optional character having the name of a column having a factor variable (e.g., treatment). The
detault value is set to NULL.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_factor2">factor2</code></td>
<td>
<p>an optional character having the name of a column having another factor variable (e.g., species). The
detault value is set to NULL.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_only.lines">only.lines</code></td>
<td>
<p>a logic value if only lines, but not including dots, are going to be drwan in the plot. The detault
value is set to FALSE.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_ylab">ylab</code></td>
<td>
<p>Label for the Y-axis</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_xlab">xlab</code></td>
<td>
<p>Label for the X-axis</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_linetype.lab">linetype.lab</code></td>
<td>
<p>is an optional string to be used as the title of the factor being represented by lines. It is only
needed if factor1 and factor2 are defined. See example.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_factor2.line">factor2.line</code></td>
<td>
<p>a logic value if the second factor, factor2, is going to be segregated according to the
type of lines. The detault  value is set to TRUE.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_factor2.col">factor2.col</code></td>
<td>
<p>a logic value if the second factor, factor2, is going to be segregated according to the
color of the lines only. The detault  value is set to FALSE.</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_max.y.all">max.y.all</code></td>
<td>
<p>A number representing the maximum level of Y-axis for all classes</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_levels.i.want">levels.i.want</code></td>
<td>
<p>A vector having the levels for the factor under study</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_col.lev.i.want">col.lev.i.want</code></td>
<td>
<p>A vector having the colors to be used for the factor under study</p>
</td></tr>
<tr><td><code id="timeserplot_+3A_col.lines">col.lines</code></td>
<td>
<p>A string specifying the single color to be used for the lines of the timeseries</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both 'y' and 'x' must be numeric variables, and the column representing the observational unit, must be
a factor. This factor identifies the longitudinal context of the data, for instance, a student
being measured on time. Besides, two more factors can be added to the plotting details, in order
to represent the potential variability among them.
</p>


<h3>Value</h3>

<p>This function returns a time series plot
</p>


<h3>Note</h3>

<p>Please, uses with caution, and run first the examples to understand it better.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#data(ficdiamgr)
# df &lt;- ficdiamgr
# 
# timeserplot(df, y="dbh", x="time", obs.unit = "tree")
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", only.lines = T)
# 
# # dos opciones de lo solicitado
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", col.lines = T, only.lines = T)
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", col.lines = T, only.lines = F)
# 
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", factor1="site")
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", factor1="site", factor2= "species")
# timeserplot(df, y="dbh", x="time", obs.unit = "tree", factor1="site", factor2= "species",
#             factor2.col = T, only.lines = T)
</code></pre>

<hr>
<h2 id='trailCameraTrap'>Contains information of Camera trap data on medium to large terrestrial mammals 
collected at 54 camera stations in Ruaha National Park, southern Tanzania.</h2><span id='topic+trailCameraTrap'></span>

<h3>Description</h3>

<p>Dataset contains 14604 observations and sampling was carried out for two months during the
dry season of 2013 and two months during the wet season of 2014. Each camera station is associated
with a randomly placed camera and a trail-based camer, with the aim of comparing communities 
resulting from the two camera trap placement strategies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(trailCameraTrap)
</code></pre>


<h3>Format</h3>

<p>Contains 6 variables, as follows:
</p>

<dl>
<dt>reference</dt><dd><p>Number of observation od datasets.</p>
</dd>
<dt>placement</dt><dd><p>Type of &quot;placement&quot; placed in each station (random or trail).</p>
</dd>
<dt>season</dt><dd><p>Season where were made the samplings.</p>
</dd>
<dt>station</dt><dd><p>Station where were collected the data.</p>
</dd>
<dt>specie</dt><dd><p>Name of specie medium to large terrestrial mammals.</p>
</dd>
<dt>date.time</dt><dd><p>The date and time of each photographic event is also given.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided by Dr Jeremy Cusack.
</p>


<h3>References</h3>

<p>- Cusack J, Dickman A, Rowcliffe M, Carbone C, Macdonald D, Coulson T. 2016 . Random versus
game trail-based camera trap placement strategy for monitoring terrestrial mammal communities. PLoS ONE 10(5): e0126373.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(trailCameraTrap)    
head(trailCameraTrap) 
</code></pre>

<hr>
<h2 id='traits'>Functional traits of vegetative species in Chile.</h2><span id='topic+traits'></span>

<h3>Description</h3>

<p>Functional traits of vegetative species in Chile. Includes column with codified name (esp)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(traits)
</code></pre>


<h3>Format</h3>


<dl>
<dt>esp</dt><dd><p>species codified name</p>
</dd>
<dt>shadeTolerance</dt><dd><p>indicates the species tolerance to shape. There are three main classes: shade-tolerant, shade-midtolerant and shade-intolerant</p>
</dd>
<dt>spp.ci.name</dt><dd><p>Scientific name.</p>
</dd>
<dt>spp.ci.abb.</dt><dd><p>.</p>
</dd>
<dt>wd</dt><dd><p>wood density in kg per cubic meters.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Some of the information on shade tolerance can be found in Soto et al 2010.
</p>


<h3>References</h3>

<p>- Soto DP, Salas C, Donoso PJ, Uteau D. 2010. Heterogeneidad estructural y espacial de un bosque mixto dominado por
Nothofagus dombeyi despues de un disturbio parcial. Revista Chilena de Historia Natural 83(3): 335-347.
</p>

<hr>
<h2 id='traits2'>Rasgos funcionales para algunas especies vegetales de Chile.</h2><span id='topic+traits2'></span>

<h3>Description</h3>

<p>Rasgos funcionales para algunas especies vegetales de Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(traits2)
</code></pre>


<h3>Format</h3>


<dl>
<dt>especie</dt><dd><p>Codigo alfanumerico para especie.</p>
</dd>
<dt>tolerancia.sombra</dt><dd><p>Tolerancia a la sombra de la especie.</p>
</dd>
<dt>nombre.cient</dt><dd><p>Nombre cientifico.</p>
</dd>
<dt>nom.cient.abre</dt><dd><p>Nombre cientifico abreviado.</p>
</dd>
<dt>den.madera</dt><dd><p>Densidad de la madera en kg/m<code class="reqn">^{3}</code>.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Parte de la informacion sobre tolerancia a la sombra se encuentra en Soto et al 2010
</p>


<h3>References</h3>

<p>- Soto DP, Salas C, Donoso PJ, Uteau D. 2010. Heterogeneidad estructural y espacial de un bosque mixto dominado por
Nothofagus dombeyi despues de un disturbio parcial. Revista Chilena de Historia Natural 83(3): 335-347.
</p>

<hr>
<h2 id='treegr'>Diameter and height growth of Grand-fir (Abies grandis) sample trees</h2><span id='topic+treegr'></span>

<h3>Description</h3>

<p>Diameter and height growth of 66 Grand-fir trees. Data derived from stem analysis sample
trees collected by Dr Albert Stage (US Forest Service, Moscow, ID, USA.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treegr)
</code></pre>


<h3>Format</h3>

<p>Contains seven column, as follows:
</p>

<dl>
<dt>tree.id</dt><dd><p>Tree number identificator. An unique number to each sample tree.</p>
</dd>
<dt>forest</dt><dd><p>Forest type.</p>
</dd>
<dt>habitat</dt><dd><p>Forest habitat type.</p>
</dd>
<dt>tree.code</dt><dd><p>A composite tree code representing the following columns: tree.id-forest-habitat</p>
</dd>
<dt>age</dt><dd><p>Age, in yr</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast-height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Originally, the data were provided by Dr Albert Stage (R.I.P) to Professor Andrew Robinson (University
of Idaho, USA), whom used them to explain the fitting of statistical models. Dr Christian Salas-Eljatib
was a former graduate student of Statistics of Prof Robinson at the Univ. of Idaho.
</p>


<h3>References</h3>

<p>Stage, A. R., 1963. A mathematical approach to polymorphic site index curves for 
Grand fir. Forest Science 9 (2), 167–180.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treegr)    
head(treegr) 
</code></pre>

<hr>
<h2 id='treegr2'>Crecimiento en diámetro y altura de árboles muestras de Grand-fir (Abies grandis)</h2><span id='topic+treegr2'></span>

<h3>Description</h3>

<p>Crecimiento en diámetro y altura de 66 árboles de Grand-fir. Los datos fueron derivados a partir
de árboles muestras de analisis fustal colectados por el Dr Albert Stage (US Forest Service, Moscow, ID, USA.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treegr2)
</code></pre>


<h3>Format</h3>

<p>Contiene las siguientes siete columnas:
</p>

<dl>
<dt>num.arb</dt><dd><p>Número identificador del árbol. Unico para cada árbol muestra.</p>
</dd>
<dt>bosque</dt><dd><p>Tipo forestal.</p>
</dd>
<dt>habitat</dt><dd><p>Clasificación de tipo de habitat.</p>
</dd>
<dt>cod.arb</dt><dd><p>Un código que combina a las siguientes columnas: num.arb-bosque-habitat</p>
</dd>
<dt>edad</dt><dd><p>Edad, en años.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm. Note que los decimales es debido a 
que esta variable originalmente fue medida en pulgadas.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, in m. Note que los decimales es debido a que esta variable
fue originalmente medida en pies.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Originalmente los datos fueron cedidos por el Dr Albert Stage (Q.E.P.D) al Profesor Andrew Robinson (University
of Idaho, USA), quien los usaba para explicar el ajuste de modelos estadísticos. El Dr Christian Salas-Eljatib
fue un estudiante de postgrado en estadística del Prof. Robinson en la Univ. of Idaho.
</p>


<h3>References</h3>

<p>Stage AR. 1963. A mathematical approach to polymorphic site index curves for 
Grand fir. Forest Science 9(2):167–180.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treegr2)    
head(treegr2) 
</code></pre>

<hr>
<h2 id='treelistinve'>Tree-list data in a forest inventory.</h2><span id='topic+treelistinve'></span>

<h3>Description</h3>

<p>Tree-level variables measured within three sample plots in a forest inventory. 
Notice that plots might have different areas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treelistinve)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>plot</dt><dd><p>Plot number.</p>
</dd>
<dt>plot.size</dt><dd><p>Plot size, in m<code class="reqn">^{2}</code>.</p>
</dd>
<dt>tree</dt><dd><p>Tree identificator</p>
</dd>
<dt>species</dt><dd><p>species common name as follows: Olivillo= Aextocicon puncatatum, Tepa= Laureliopsis philippiana, 
Lingue= Persea lingues, Coigue=Nothofagus dombeyi, Roble=Nothofagus obliqua, Other=Other</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast-height, in cm</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m. Only measured for some sample trees.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Prof. Christian Salas-Eljatib (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2001. Caracterización básica del relicto de Biodiversidad 
Rucamanque. Bosque Nativo, 29:3-9. <a href="https://eljatib.com/publication/2001-06-01_caracterizacion_basi/">https://eljatib.com/publication/2001-06-01_caracterizacion_basi/</a>
</p>
<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treelistinve)    
head(treelistinve) 
tapply(treelistinve$dbh,treelistinve$species,summary)
</code></pre>

<hr>
<h2 id='treelistinve2'>Lista de árboles en un inventario forestal.</h2><span id='topic+treelistinve2'></span>

<h3>Description</h3>

<p>Variables a nivel de árbol medidas en tres unidades de muestreo establecidas en un inventario forestal. 
Note que las parcelas pueden tener diferentes superficies. Las parcelas
fueron establecidas en un bosque secundario dominando por Nothofagus obliqua en las cercanias de Temuco.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treelistinve2)
</code></pre>


<h3>Format</h3>

<p>Contiene variables a nivel de árbol dentro de parcelas.
</p>

<dl>
<dt>parce</dt><dd><p>Número de la parcela de muestreo.</p>
</dd>
<dt>sup.parce</dt><dd><p>Superficie de la parcela, en m<code class="reqn">^{2}</code>.</p>
</dd>
<dt>arbol</dt><dd><p>Número identificador del árbol.</p>
</dd>
<dt>spp</dt><dd><p>Nombre comun de especies como sigue: Olivillo= Aextocicon puncatatum, Tepa= Laureliopsis philippiana, 
Lingue= Persea lingue, Coigue=Nothofagus dombeyi, Roble=Nothofagus obliqua, Other=Other</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m. Solo medida en algunas árboles muestra.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Prof. Christian Salas-Eljatib, Universidad de Chile (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2001. Caracterización básica del relicto de Biodiversidad 
Rucamanque. Bosque Nativo, 29:3-9. <a href="https://eljatib.com/publication/2001-06-01_caracterizacion_basi/">https://eljatib.com/publication/2001-06-01_caracterizacion_basi/</a>
</p>
<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treelistinve2)    
unique(treelistinve2$parce) 
table(treelistinve2$parce,treelistinve2$sup.parce)
tapply(treelistinve2$dap,treelistinve2$spp,summary)
</code></pre>

<hr>
<h2 id='treevol'>Diameter, height and volume for Black Cherry Trees</h2><span id='topic+treevol'></span>

<h3>Description</h3>

<p>This data set provides measurements of the diameter, height and volume of timber in 31 felled black cherry trees.
The records are a slight modification to the original dataframe &quot;trees&quot; from the datasets R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevol)
</code></pre>


<h3>Format</h3>

<p>A data frame with 31 observations and three variables
</p>

<dl>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>vtot</dt><dd><p>Timber volume, in cubic meters.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ryan TA, Joiner BL, and Ryan BF. 1976. The Minitab Student Handbook. Duxbury Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pairs(treevol, panel = panel.smooth, main = "treevol dataframe")
plot(vtot ~ dbh, data = treevol, log = "xy")
coplot(log(vtot) ~ log(dbh) | toth, data = treevol,
       panel = panel.smooth)
summary(m1 &lt;- lm(log(vtot) ~ log(dbh), data = treevol))
summary(m2 &lt;- update(m1, ~ . + log(toth), data = treevol))
anova(m1,m2)
</code></pre>

<hr>
<h2 id='treevol2'>Volumen, altura, y diámetro para árboles de Black Cherry</h2><span id='topic+treevol2'></span>

<h3>Description</h3>

<p>Estos datos provienen de mediciones de volumen, altura y diámetro en 31 árboles volteados de black cherry (Prunus serotina).
Son una modificacion la dataframe 'trees' del paquete datasets de R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevol2)
</code></pre>


<h3>Format</h3>

<p>Datos con 31 observaciones y tres variables
</p>

<dl>
<dt>dap</dt><dd><p>diámetro a la altura del pecho, en cm</p>
</dd>
<dt>atot</dt><dd><p>altural total, en m</p>
</dd>
<dt>vtot</dt><dd><p>volumen total, en m<code class="reqn">^{3}</code></p>
</dd>
</dl>



<h3>Source</h3>

<p>Ryan, T. A., Joiner, B. L. and Ryan, B. F. (1976) The Minitab Student Handbook. Duxbury Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pairs(treevol2, panel = panel.smooth, main = "treevol dataframe")
plot(vtot ~ dap, data = treevol2, log = "xy")
coplot(log(vtot) ~ log(dap) | atot, data = treevol2,
       panel = panel.smooth)
summary(m1 &lt;- lm(log(vtot) ~ log(dap), data = treevol2))
summary(m2 &lt;- update(m1, ~ . + log(atot), data = treevol2))
anova(m1,m2)
</code></pre>

<hr>
<h2 id='treevolroble'>Tree volume of roble (Nothofagus obliqua) in the Rucamanque forest</h2><span id='topic+treevolroble'></span>

<h3>Description</h3>

<p>These are tree-level measurement data of sample trees in the Rucamanque experimental forest, 
near Temuco, in the Araucania region in south-centralChile, measured in 1999.  The
data are the same as in the dataframe &quot;treevolruca&quot;, but only having observations for the 
species roble (Nothofagus obliqua).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevolroble)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>tree.no</dt><dd><p>Tree id</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>d6</dt><dd><p>Upper-stem diameter at 6 m, in cm</p>
</dd>
<dt>totv</dt><dd><p>Tree gross volume, in m<code class="reqn">^{3}</code> with bark.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data are provided courtesy of Dr Christian Salas at
the Universidad de Chile (Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treevolroble)    
head(treevolroble) 
</code></pre>

<hr>
<h2 id='treevolroble2'>Volumen a nivel de árbol para roble (Nothofagus obliqua) especie en el bosque de Rucamanque</h2><span id='topic+treevolroble2'></span>

<h3>Description</h3>

<p>Volumen, altura y diámetro, entre otras para árboles muestra de roble (Nothofagus obliqua) en 
el bosque de Rucamanque, cerca de Temuco, en la región de la Araucania, en el sur de Chile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevolroble2)
</code></pre>


<h3>Format</h3>

<p>Las siguientes columnas son parte de la dataframe:
</p>

<dl>
<dt>arbol</dt><dd><p>Número del árbol.</p>
</dd>
<dt>especie</dt><dd><p>Especie.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>d6</dt><dd><p>Diámetro fustal a los 6 m, en cm.</p>
</dd>
<dt>vtot</dt><dd><p>Volumen bruto total, en m<code class="reqn">^{3}</code> with bark.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos son proporcionados por el Prof. Christian Salas (Universidad de Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treevolroble2)    
head(treevolroble2) 
</code></pre>

<hr>
<h2 id='treevolruca'>Tree volume by species in the Rucamanque forest</h2><span id='topic+treevolruca'></span>

<h3>Description</h3>

<p>These are tree-level measurement data of sample trees in the Rucamanque experimental forest, 
near Temuco, in the Araucania region in south-centralChile, measured in 1999.  The following
species are part of the data: laurel (laurelia sempervirens), lingue (Persea lingue), olivillo (Aextocicon puncatum), 
roble (Nothofagus obliqua), tepa (Laureliosis philippiana), y tineo (Weinmannia trichosperma).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevolruca)
</code></pre>


<h3>Format</h3>

<p>Contains tree-level variables, as follows:
</p>

<dl>
<dt>tree.no</dt><dd><p>Tree id.</p>
</dd>
<dt>spp</dt><dd><p>Species.</p>
</dd>
<dt>dbh</dt><dd><p>Diameter at breast height, in cm.</p>
</dd>
<dt>toth</dt><dd><p>Total height, in m.</p>
</dd>
<dt>d6</dt><dd><p>Upper-stem diameter at 6 m, in cm.</p>
</dd>
<dt>totv</dt><dd><p>Tree gross volume, in m<code class="reqn">^{3}</code> with bark.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data were provided courtesy of Dr Christian Salas (Universidad de Chile, Santiago, Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treevolruca)    
head(treevolruca) 
</code></pre>

<hr>
<h2 id='treevolruca2'>Volumen a nivel de árbol en el bosque de Rucamanque</h2><span id='topic+treevolruca2'></span>

<h3>Description</h3>

<p>Volumen, altura y diámetro, entre otras para árboles muestra en el bosque de Rucamanque, cerca de Temuco, 
en la region de la Araucanía, en el sur de Chile.  Las siguientes especies son parte de los datos: laurel (laurelia sempervirens), lingue (Persea lingue), olivillo (Aextocicon puncatum), 
roble (Nothofagus obliqua), tepa (Laureliosis philippiana), y tineo (Weinmannia trichosperma).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(treevolruca2)
</code></pre>


<h3>Format</h3>

<p>Las siguientes columnas son parte de la dataframe:
</p>

<dl>
<dt>arbol</dt><dd><p>Número del árbol.</p>
</dd>
<dt>especie</dt><dd><p>Especie.</p>
</dd>
<dt>dap</dt><dd><p>Diámetro a la altura del pecho, en cm.</p>
</dd>
<dt>atot</dt><dd><p>Altura total, en m.</p>
</dd>
<dt>d6</dt><dd><p>Diámetro fustal a los 6 m, en cm.</p>
</dd>
<dt>vtot</dt><dd><p>Volumen bruto total, en m<code class="reqn">^{3}</code> with bark.</p>
</dd>
</dl>



<h3>Source</h3>

<p>Los datos fueron cedidos por el Dr Christian Salas-Eljatib (Chile).
</p>


<h3>References</h3>

<p>- Salas C. 2002. Ajuste y validación de ecuaciones de volumen para un relicto
del bosque de Roble-Laurel-Lingue. Bosque 23(2): 81-92. <a href="https://doi.org/10.4067/S0717-92002002000200009">doi:10.4067/S0717-92002002000200009</a> <a href="https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/">https://eljatib.com/publication/2002-07-01_ajuste_y_validacion_/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treevolruca2)    
head(treevolruca2) 
</code></pre>

<hr>
<h2 id='xyboxplot'>Creates a scatterplot with superposing boxplots</h2><span id='topic+xyboxplot'></span>

<h3>Description</h3>

<p>The function creates a scatterplot with superposing boxplots for the Y-axis variable. To a
simple scatterplot between the response variable 'y' and the predictor variable 'x', 
this function superposes boxplots of the response by groups of the predictor variable. 
The main aim of the above described graph is to get a sense of the distribution of the 
response variable depending upon the  predictor variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyboxplot(x = x, y = y, col.dots = "blue", xlab = NULL, ylab = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xyboxplot_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the time variable (X-axis).</p>
</td></tr>
<tr><td><code id="xyboxplot_+3A_y">y</code></td>
<td>
<p>A numeric vector representing the response variable (Y-axis).</p>
</td></tr>
<tr><td><code id="xyboxplot_+3A_col.dots">col.dots</code></td>
<td>
<p>(optional) A string specifying the dot colors. Default is &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="xyboxplot_+3A_xlab">xlab</code></td>
<td>
<p>(optional) A string specifying X-axis label.</p>
</td></tr>
<tr><td><code id="xyboxplot_+3A_ylab">ylab</code></td>
<td>
<p>(optional) A string specifying Y-axis label.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notice that the superposing boxplots for the Y-axis variable are computed
by grouping the X-axis variable in 10 classes. Those classes are set by computing the ten percentiles
of the X-axis variable, therefore each group has the same number of observations.
</p>


<h3>Value</h3>

<p>The function returns the above described graph.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>References</h3>

<p>- Salas-Eljatib C. 2021. Análisis de datos con el programa estadístico R: una introducción aplicada. 
Ediciones Universidad Mayor. Santiago, Chile. <a href="https://eljatib.com">https://eljatib.com</a>
</p>
<p>- Salas C, Stage AR, and Robinson AP. 2008. Modeling effects of overstory density and competing 
vegetation on tree height growth. Forest Science 54(1):107-122. <a href="https://doi.org/10.1093/forestscience/54.1.107">doi:10.1093/forestscience/54.1.107</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fishgrowth)
df &lt;- fishgrowth
xyboxplot(x=df$age,y=df$length)
xyboxplot(x=df$age,y=df$length)
</code></pre>

<hr>
<h2 id='xyhist'>A scatterplot with marginal histograms</h2><span id='topic+xyhist'></span>

<h3>Description</h3>

<p>The function produces a scatterplot between the 'y'-axis variable and
the 'x'-axis variable, but also adding the marginal histograms
for both variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyhist(
  x = x,
  y = y,
  col.x = "blue",
  col.y = "red",
  xlab = NULL,
  ylab = NULL,
  x.lim = NULL,
  y.lim = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xyhist_+3A_x">x</code></td>
<td>
<p>A numeric vector representing the X-axis variable</p>
</td></tr>
<tr><td><code id="xyhist_+3A_y">y</code></td>
<td>
<p>A numeric vector representing the Y-axis variable</p>
</td></tr>
<tr><td><code id="xyhist_+3A_col.x">col.x</code></td>
<td>
<p>(optional) A string specifying the color of the histogram of the X-variable. Default is &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="xyhist_+3A_col.y">col.y</code></td>
<td>
<p>(optional) A string specifying the color of the histogram of the Y-variable. Default is &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="xyhist_+3A_xlab">xlab</code></td>
<td>
<p>(optional) A string specifying X-axis label. Default is &quot;xvar&quot;.</p>
</td></tr>
<tr><td><code id="xyhist_+3A_ylab">ylab</code></td>
<td>
<p>(optional) A string specifying Y-axis label. Default is &quot;yvar&quot;.</p>
</td></tr>
<tr><td><code id="xyhist_+3A_x.lim">x.lim</code></td>
<td>
<p>(optional) A vector of two elements with the limits of the Y-axis. Default is the range of the X-variable.</p>
</td></tr>
<tr><td><code id="xyhist_+3A_y.lim">y.lim</code></td>
<td>
<p>(optional) A vector of two elements with the limits of the Y-axis. Default is the range of the Y-variable.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both the response variable (Y-axis) and the predictor variable (X-axis) must be
numeric.
</p>


<h3>Value</h3>

<p>The function returns the above described graph.
</p>


<h3>Author(s)</h3>

<p>Christian Salas-Eljatib
</p>


<h3>References</h3>

<p>- Salas-Eljatib C. 2021. Análisis de datos con el programa estadístico R: una introducción aplicada. 
Ediciones Universidad Mayor. Santiago, Chile. <a href="https://eljatib.com">https://eljatib.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(treevolroble)
df &lt;- treevolroble
head(df)
xyhist(x=df$dbh,y=df$toth) 
xyhist(x=df$dbh,y=df$toth, xlab="Variable X",  ylab="Variable Y") 
xyhist(x=df$dbh,y=df$toth, xlab="Variable X", ylab="Variable Y", 
  col.x = "gray",col.y="white")
</code></pre>

<hr>
<h2 id='xymultiplot'>Figure of a matrix of scatterplots and histograms for several variables.</h2><span id='topic+xymultiplot'></span>

<h3>Description</h3>

<p>The function produces a panel of multiple scatterplots and histograms, showing 
the correlation coefficient among all pairs of variables. Notice that 
the data must contain only numeric variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xymultiplot(
  x,
  smooth = TRUE,
  scale = FALSE,
  density = TRUE,
  digits = 2,
  method = "pearson",
  pch = 20,
  lm = FALSE,
  cor = TRUE,
  jiggle = FALSE,
  factor = 2,
  col.hist = "cyan",
  col.densi.curve = "black",
  show.points = TRUE,
  col.points = "gray",
  smoother = FALSE,
  col.smooth = "red",
  ellipses = FALSE,
  col.ellip = "blue",
  col.cent.point = "green",
  rug = TRUE,
  breaks = "Sturges",
  cex.cor = 1,
  ci = FALSE,
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xymultiplot_+3A_x">x</code></td>
<td>
<p>is a dataframe containing all the numeric variables to be used for drawing the panel plot</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_smooth">smooth</code></td>
<td>
<p>a logical value for drawing smooth curves. The default is set to TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_scale">scale</code></td>
<td>
<p>scales the correlation font by the size of the absolute correlation. The default is set to FALSE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_density">density</code></td>
<td>
<p>a logical value for drawing a density curve. The default is set to TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_digits">digits</code></td>
<td>
<p>an optional numeric value for the digits to be used for drawing the correlation coefficient in the panel.
Defaults is set to 2.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_method">method</code></td>
<td>
<p>a string giving the method to be used for computing the correlation coefficient. Default is set to &quot;pearson&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_pch">pch</code></td>
<td>
<p>The plot character (The default is 20, which looks like '.').</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_lm">lm</code></td>
<td>
<p>Plot the linear fit rather than the LOESS smoothed fits. The default is FALSE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_cor">cor</code></td>
<td>
<p>If plotting regressions, should correlations be reported? The default is TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_jiggle">jiggle</code></td>
<td>
<p>Should the points be jittered before plotting? The default is FALSE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_factor">factor</code></td>
<td>
<p>factor for jittering (1-5), therefore only needed if &quot;jiggle&quot; is set to TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.hist">col.hist</code></td>
<td>
<p>a string giving the color to be used for the histograms of the panel. Default is set to &quot;cyan&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.densi.curve">col.densi.curve</code></td>
<td>
<p>a string with the name of the color to be used for the density curve. The default is set to &quot;black&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_show.points">show.points</code></td>
<td>
<p>a logical value for drawing the points in the scatter-plots. Defauls is set to TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.points">col.points</code></td>
<td>
<p>a string giving the color to be used for the data points. Default is set to &quot;gray&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_smoother">smoother</code></td>
<td>
<p>If TRUE, then smooth.scatter the data points – slow but pretty with lots of subjects</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.smooth">col.smooth</code></td>
<td>
<p>a string giving the color to be used for the smoothed curve of the scatterplot. Default is set to &quot;red&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_ellipses">ellipses</code></td>
<td>
<p>an optional logical value for drawing an ellipse for the scatter-plots. The default is set to FALSE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.ellip">col.ellip</code></td>
<td>
<p>a string giving the color to be used for the ellipse of the scatterplot. The default is set to &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_col.cent.point">col.cent.point</code></td>
<td>
<p>a string giving the color to be used for the centroid point of the ellipse of the scatterplot. The default is set to &quot;blue&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_rug">rug</code></td>
<td>
<p>a logical value for drawing the rugs in the histograms. Defauls is set to TRUE.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_breaks">breaks</code></td>
<td>
<p>a string giving the method to be used for obtaining the breaks of the histogram. Defauls is set to &quot;Sturges&quot;.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_cex.cor">cex.cor</code></td>
<td>
<p>If this is specified, this will change the size of the text in the correlations. this 
allows one to also change the size of the points in the plot by specifying the normal cex values. If just
specifying cex, it will change the character size, if cex.cor is specified, then cex will function to change the point size.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_ci">ci</code></td>
<td>
<p>Draw confidence intervals for the linear model or for the loess fit, defaults to ci=FALSE. If confidence
intervals are not drawn, the fitting function is lowess.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_alpha">alpha</code></td>
<td>
<p>an optional numeric value for the significance level. Defauls is set to 0.05.</p>
</td></tr>
<tr><td><code id="xymultiplot_+3A_...">...</code></td>
<td>
<p>other <a href="graphics.html#topic+graphical+20parameters">graphical parameters</a> (see <code><a href="graphics.html#topic+par">par</a></code> and
section &lsquo;Details&rsquo; below).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generates a multipanel (matrix) of scatterplots and histograms to explore potential relationships among variables.
</p>


<h3>Value</h3>

<p>This function returns a multipanel of scatterplots and histograms
</p>


<h3>Author(s)</h3>

<p>A modification of Christian Salas-Eljatib of the function pairs.panels of the package &quot;psych&quot;.
</p>


<h3>References</h3>

<p>- Salas-Eljatib C. 2021. Análisis de datos con el programa estadístico R: una introducción aplicada. 
Ediciones Universidad Mayor. Santiago, Chile. <a href="https://eljatib.com">https://eljatib.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##First example
data(bears2)
head(bears2)
df &lt;- bears2[,c('peso','edad','cabezaL','cabezaA','largo','pechoP')]
descstat(df)
xymultiplot(df)
xymultiplot(df,ellipse=TRUE)
xymultiplot(df,ellipses=TRUE,col.cent.point = "yellow",
 col.densi.curve = "dark green",col.hist = "white")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
