<!DOCTYPE html><html><head><title>Help for package DataVisualizations</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DataVisualizations}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DataVisualizations-package'>
<p>Visualizations of High-Dimensional Data</p></a></li>
<li><a href='#ABCbarplot'>
<p>Barplot with Sorted Data Colored by ABCanalysis</p></a></li>
<li><a href='#AccountingInformation_PrimeStandard_Q3_2019'>
<p>Accounting Information in the Prime Standard in Q3 in 2019 (AI_PS_Q3_2019)</p></a></li>
<li><a href='#BimodalityAmplitude'><p>Bimodality Amplitude</p></a></li>
<li><a href='#categoricalVariable'>
<p>A categorical Feature.</p></a></li>
<li><a href='#CCDFplot'><p> plot Complementary Cumulative Distribution Function (CCDF) in Log/Log  uses ecdf, CCDF(x) = 1-cdf(x)</p></a></li>
<li><a href='#Choroplethmap'>
<p>Plots the Choropleth Map</p></a></li>
<li><a href='#ChoroplethPostalCodesAndAGS_Germany'>
<p>Postal Codes and AGS of Germany for a Choropleth Map</p></a></li>
<li><a href='#ClassBoxplot'>
<p>Creates Boxplot plot for all classes</p></a></li>
<li><a href='#ClassErrorbar'>
<p>ClassErrorbar</p></a></li>
<li><a href='#ClassMDplot'>
<p>Class MDplot for Data w.r.t. all classes</p></a></li>
<li><a href='#ClassPDEplot'>
<p>PDE Plot for all classes</p></a></li>
<li><a href='#ClassPDEplotMaxLikeli'>
<p>Create PDE plot for all classes with maximum likelihood</p></a></li>
<li><a href='#Classplot'>
<p>Classplot</p></a></li>
<li><a href='#CombineCols'>
<p>Combine vectors of various lengths</p></a></li>
<li><a href='#Crosstable'>
<p>Crosstable plot</p></a></li>
<li><a href='#DefaultColorSequence'>
<p>Default color sequence for plots</p></a></li>
<li><a href='#DensityContour'><p>Contour plot of densities</p></a></li>
<li><a href='#DensityScatter'>
<p>Scatter plot with densities</p></a></li>
<li><a href='#DualaxisClassplot'>
<p>Dualaxis Classplot</p></a></li>
<li><a href='#DualaxisLinechart'>
<p>DualaxisLinechart</p></a></li>
<li><a href='#estimateDensity2D'>
<p>estimateDensity2D</p></a></li>
<li><a href='#Fanplot'>
<p>The fan plot</p></a></li>
<li><a href='#FundamentalData_Q1_2018'>
<p>Fundamental Data of the 1st Quarter in 2018</p></a></li>
<li><a href='#GoogleMapsCoordinates'>
<p>Google Maps with marked coordinates</p></a></li>
<li><a href='#Heatmap'>
<p>Heatmap for Clustering</p></a></li>
<li><a href='#HeatmapColors'>
<p>Default color sequence for plots</p></a></li>
<li><a href='#InspectBoxplots'>
<p>Inspect Boxplots</p></a></li>
<li><a href='#InspectCorrelation'>
<p>Inspect the Correlation</p></a></li>
<li><a href='#InspectDistances'>
<p>Inspection of Distance-Distribution</p></a></li>
<li><a href='#InspectScatterplots'>
<p>Pairwise scatterplots and optimal histograms</p></a></li>
<li><a href='#InspectStandardization'>
<p>QQplot of Data versus Normalized Data</p></a></li>
<li><a href='#InspectVariable'><p>Visualization of Distribution of one variable</p></a></li>
<li><a href='#ITS'>
<p>Income Tax Share</p></a></li>
<li><a href='#JitterUniqueValues'>
<p>Jitters Unique Values</p></a></li>
<li><a href='#Lsun3D'>
<p>Lsun3D inspired by FCPS [Thrun/Ultsch, 2020] introduced in [Thrun, 2018]</p></a></li>
<li><a href='#MAplot'>
<p>Minus versus Add plot</p></a></li>
<li><a href='#MDplot'>
<p>Mirrored Density plot (MD-plot)</p></a></li>
<li><a href='#MDplot4multiplevectors'>
<p>Mirrored Density plot (MD-plot)for Multiple Vectors</p></a></li>
<li><a href='#MTY'>
<p>Muncipal Income Tax Yield</p></a></li>
<li><a href='#Multiplot'><p>Plot multiple ggplots objects in one panel</p></a></li>
<li><a href='#OptimalNoBins'><p>Optimal Number Of Bins</p></a></li>
<li><a href='#ParetoDensityEstimation'><p>Pareto Density Estimation V3</p></a></li>
<li><a href='#ParetoRadius'><p>ParetoRadius for distributions</p></a></li>
<li><a href='#PDEplot'><p>PDE plot</p></a></li>
<li><a href='#Piechart'>
<p>The pie chart</p></a></li>
<li><a href='#Pixelmatrix'><p>Plot of a Pixel Matrix</p></a></li>
<li><a href='#Plot3D'>
<p>3D plot of points</p></a></li>
<li><a href='#PlotGraph2D'><p>PlotGraph2D</p></a></li>
<li><a href='#PlotMissingvalues'><p>Plot of the Amount Of Missing Values</p></a></li>
<li><a href='#PlotProductratio'>
<p>Product-Ratio Plot</p></a></li>
<li><a href='#PmatrixColormap'><p>P-Matrix colors</p></a></li>
<li><a href='#QQplot'>
<p>QQplot with a Linear Fit</p></a></li>
<li><a href='#RobustNorm_BackTrafo'>
<p>Transforms the Robust Normalization back</p></a></li>
<li><a href='#RobustNormalization'>
<p>RobustNormalization</p></a></li>
<li><a href='#ROC'><p>ROC plot</p></a></li>
<li><a href='#ShepardDensityscatter'><p>Shepard PDE scatter</p></a></li>
<li><a href='#Sheparddiagram'>
<p>Draws a Shepard Diagram</p></a></li>
<li><a href='#SignedLog'>
<p>Signed Log</p></a></li>
<li><a href='#Silhouetteplot'>
<p>Silhouette plot of classified data.</p></a></li>
<li><a href='#Slopechart'>
<p>Slope Chart</p></a></li>
<li><a href='#stat_pde_density'>
<p>Calculate Pareto density estimation for ggplot2 plots</p></a></li>
<li><a href='#StatPDEdensity'>
<p>Pareto Density Estimation</p></a></li>
<li><a href='#world_country_polygons'>
<p>world_country_polygons</p></a></li>
<li><a href='#Worldmap'>
<p>plots a world map by country codes</p></a></li>
<li><a href='#zplot'>
<p>Plotting for 3 dimensional data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualizations of High-Dimensional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-09</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Thrun &lt;m.thrun@gmx.net&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Gives access to data visualisation methods that are relevant from the data scientist's point of view. The flagship idea of 'DataVisualizations' is the mirrored density plot (MD-plot) for either classified or non-classified multivariate data published in Thrun, M.C. et al.: "Analyzing the Fine Structure of Distributions" (2020), PLoS ONE, &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0238835">doi:10.1371/journal.pone.0238835</a>&gt;. The MD-plot outperforms the box-and-whisker diagram (box plot), violin plot and bean plot and geom_violin plot of ggplot2. Furthermore, a collection of various visualization methods for univariate data is provided. In the case of exploratory data analysis, 'DataVisualizations' makes it possible to inspect the distribution of each feature of a dataset visually through a combination of four methods. One of these methods is the Pareto density estimation (PDE) of the probability density function (pdf). Additionally, visualizations of the distribution of distances using PDE, the scatter-density plot using PDE for two variables as well as the Shepard density plot and the Bland-Altman plot are presented here. Pertaining to classified high-dimensional data, a number of visualizations are described, such as f.ex. the heat map and silhouette plot. A political map of the world or Germany can be visualized with the additional information defined by a classification of countries or regions. By extending the political map further, an uncomplicated function for a Choropleth map can be used which is useful for measurements across a geographic area. For categorical features, the Pie charts, slope charts and fan plots, improved by the ABC analysis, become usable. More detailed explanations are found in the book by Thrun, M.C.: "Projection-Based Clustering through Self-Organization and Swarm Intelligence" (2018) &lt;<a href="https://doi.org/10.1007%2F978-3-658-20540-9">doi:10.1007/978-3-658-20540-9</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.12), ggplot2, sp, pracma, reshape2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>plyr, MBA, ggmap, plotrix, rworldmap, rgl, ABCanalysis,
choroplethr, dplyr, R6, parallelDist, knitr (&ge; 1.12),
rmarkdown (&ge; 0.9), vioplot, ggExtra, plotly, htmlwidgets,
diptest, moments, signal, ggrepel, MASS, ROCit, ScatterDensity
(&ge; 0.0.3), colorspace, viridis</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.deepbionics.org/">https://www.deepbionics.org/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Mthrun/DataVisualizations/issues">https://github.com/Mthrun/DataVisualizations/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-10 07:17:09 UTC; MCT</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Thrun <a href="https://orcid.org/0000-0001-9542-5543"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Felix Pape [aut, rev],
  Onno Hansen-Goos [ctr, ctb],
  Quirin Stier <a href="https://orcid.org/0000-0002-7896-4737"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb, rev],
  Hamza Tayyab [ctr, ctb],
  Luca Brinkmann [ctr, ctb],
  Dirk Eddelbuettel [ctr],
  Craig Varrichio [ctr],
  Alfred Ultsch [dtc, ctb, ctr]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-10 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='DataVisualizations-package'>
Visualizations of High-Dimensional Data
</h2><span id='topic+DataVisualizations-package'></span><span id='topic+DataVisualizations'></span>

<h3>Description</h3>

<p>Gives access to data visualisation methods that are relevant from the data scientist's point of view. The flagship idea of 'DataVisualizations' is the mirrored density plot (MD-plot) for either classified or non-classified multivariate data published in Thrun, M.C. et al.: &quot;Analyzing the Fine Structure of Distributions&quot; (2020), PLoS ONE, &lt;DOI:10.1371/journal.pone.0238835&gt;. The MD-plot outperforms the box-and-whisker diagram (box plot), violin plot and bean plot and geom_violin plot of ggplot2. Furthermore, a collection of various visualization methods for univariate data is provided. In the case of exploratory data analysis, 'DataVisualizations' makes it possible to inspect the distribution of each feature of a dataset visually through a combination of four methods. One of these methods is the Pareto density estimation (PDE) of the probability density function (pdf). Additionally, visualizations of the distribution of distances using PDE, the scatter-density plot using PDE for two variables as well as the Shepard density plot and the Bland-Altman plot are presented here. Pertaining to classified high-dimensional data, a number of visualizations are described, such as f.ex. the heat map and silhouette plot. A political map of the world or Germany can be visualized with the additional information defined by a classification of countries or regions. By extending the political map further, an uncomplicated function for a Choropleth map can be used which is useful for measurements across a geographic area. For categorical features, the Pie charts, slope charts and fan plots, improved by the ABC analysis, become usable. More detailed explanations are found in the book by Thrun, M.C.: &quot;Projection-Based Clustering through Self-Organization and Swarm Intelligence&quot; (2018) &lt;DOI:10.1007/978-3-658-20540-9&gt;.
</p>


<h3>Details</h3>

<p>For a brief introduction to <span class="pkg">DataVisualizations</span> please see the vignette <a href="../doc/DataVisualizations.html">A Quick Tour in Data Visualizations</a>.
</p>
<p>Please see <a href="https://www.deepbionics.org/">https://www.deepbionics.org/</a>. 
Depending on the context please cite either [Thrun, 2018] regarding visualizations in the context of clustering or [Thrun/Ultsch, 2018] for other visualizations.
</p>
<p>For the Mirrored Density Plot (MD plot) please cite [Thrun et al., 2020] and see the extensive vignette in <a href="https://md-plot.readthedocs.io/en/latest/index.html">https://md-plot.readthedocs.io/en/latest/index.html</a>. The MD plot is also available in Python <a href="https://pypi.org/project/md-plot/">https://pypi.org/project/md-plot/</a>
</p>

<p>Index of help topics:
</p>
<pre>
ABCbarplot              Barplot with Sorted Data Colored by ABCanalysis
AccountingInformation_PrimeStandard_Q3_2019
                        Accounting Information in the Prime Standard in
                        Q3 in 2019 (AI_PS_Q3_2019)
BimodalityAmplitude     Bimodality Amplitude
CCDFplot                plot Complementary Cumulative Distribution
                        Function (CCDF) in Log/Log uses ecdf, CCDF(x) =
                        1-cdf(x)
ChoroplethPostalCodesAndAGS_Germany
                        Postal Codes and AGS of Germany for a
                        Choropleth Map
Choroplethmap           Plots the Choropleth Map
ClassBoxplot            Creates Boxplot plot for all classes
ClassErrorbar           ClassErrorbar
ClassMDplot             Class MDplot for Data w.r.t. all classes
ClassPDEplot            PDE Plot for all classes
ClassPDEplotMaxLikeli   Create PDE plot for all classes with maximum
                        likelihood
Classplot               Classplot
CombineCols             Combine vectors of various lengths
Crosstable              Crosstable plot
DataVisualizations-package
                        Visualizations of High-Dimensional Data
DefaultColorSequence    Default color sequence for plots
DensityContour          Contour plot of densities
DensityScatter          Scatter plot with densities
DualaxisClassplot       Dualaxis Classplot
DualaxisLinechart       DualaxisLinechart
Fanplot                 The fan plot
FundamentalData_Q1_2018
                        Fundamental Data of the 1st Quarter in 2018
GoogleMapsCoordinates   Google Maps with marked coordinates
Heatmap                 Heatmap for Clustering
HeatmapColors           Default color sequence for plots
ITS                     Income Tax Share
InspectBoxplots         Inspect Boxplots
InspectCorrelation      Inspect the Correlation
InspectDistances        Inspection of Distance-Distribution
InspectScatterplots     Pairwise scatterplots and optimal histograms
InspectStandardization
                        QQplot of Data versus Normalized Data
InspectVariable         Visualization of Distribution of one variable
JitterUniqueValues      Jitters Unique Values
Lsun3D                  Lsun3D inspired by FCPS [Thrun/Ultsch, 2020]
                        introduced in [Thrun, 2018]
MAplot                  Minus versus Add plot
MDplot                  Mirrored Density plot (MD-plot)
MDplot4multiplevectors
                        Mirrored Density plot (MD-plot)for Multiple
                        Vectors
MTY                     Muncipal Income Tax Yield
Multiplot               Plot multiple ggplots objects in one panel
OptimalNoBins           Optimal Number Of Bins
PDEplot                 PDE plot
ParetoDensityEstimation
                        Pareto Density Estimation V3
ParetoRadius            ParetoRadius for distributions
Piechart                The pie chart
Pixelmatrix             Plot of a Pixel Matrix
Plot3D                  3D plot of points
PlotGraph2D             PlotGraph2D
PlotMissingvalues       Plot of the Amount Of Missing Values
PlotProductratio        Product-Ratio Plot
PmatrixColormap         P-Matrix colors
QQplot                  QQplot with a Linear Fit
ROC                     ROC plot
RobustNorm_BackTrafo    Transforms the Robust Normalization back
RobustNormalization     RobustNormalization
ShepardDensityScatter   Shepard PDE scatter
Sheparddiagram          Draws a Shepard Diagram
SignedLog               Signed Log
Silhouetteplot          Silhouette plot of classified data.
Slopechart              Slope Chart
StatPDEdensity          Pareto Density Estimation
Worldmap                plots a world map by country codes
categoricalVariable     A categorical Feature.
estimateDensity2D       estimateDensity2D
stat_pde_density        Calculate Pareto density estimation for ggplot2
                        plots
world_country_polygons
                        world_country_polygons
zplot                   Plotting for 3 dimensional data
</pre>


<h3>Author(s)</h3>

<p>Michael Thrun, Felix Pape, Onno Hansen-Goos, Alfred Ultsch
</p>
<p>Maintainer: Michael Thrun &lt;m.thrun@gmx.net&gt;
</p>


<h3>References</h3>

<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, Heidelberg, ISBN: 978-3-658-20539-3, <a href="https://doi.org/10.1007/978-3-658-20540-9">doi:10.1007/978-3-658-20540-9</a>, 2018. 
</p>
<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Thrun et al., 2020]	Thrun, M. C., Gehlert, T. &amp; Ultsch, A.: Analyzing the Fine Structure of Distributions, PLoS ONE, Vol. 15(10), pp. 1-66, DOI 10.1371/journal.pone.0238835, 2020. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("Lsun3D")
Data=Lsun3D$Data

Pixelmatrix(Data)



InspectDistances(as.matrix(dist(Data)))


MAlist=MAplot(ITS,MTY)

data("Lsun3D")
Cls=Lsun3D$Cls
Data=Lsun3D$Data
#clear cluster structure
plot(Data[,1:2],col=Cls)
#However, the silhouette plot does not indicate a very good clustering in cluster 1 and 2

Silhouetteplot(Data,Cls = Cls)


Heatmap(as.matrix(dist(Data)),Cls = Cls)

</code></pre>

<hr>
<h2 id='ABCbarplot'>
Barplot with Sorted Data Colored by ABCanalysis
</h2><span id='topic+ABCbarplot'></span><span id='topic+ABC_screeplot'></span>

<h3>Description</h3>

<p>This plot can be read like a scree plot for PCA. It allowed to select the most important values visually.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ABCbarplot(Data,

Colors=DataVisualizations::DefaultColorSequence[1:3],

main,xlab,ylab="Value")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ABCbarplot_+3A_data">Data</code></td>
<td>
<p>[1:n] vector of Data, e.g. eigenvalues of PCA</p>
</td></tr>
<tr><td><code id="ABCbarplot_+3A_colors">Colors</code></td>
<td>
<p>three colors for A, B and C</p>
</td></tr>
<tr><td><code id="ABCbarplot_+3A_main">main</code></td>
<td>
<p>title of plot</p>
</td></tr>
<tr><td><code id="ABCbarplot_+3A_xlab">xlab</code></td>
<td>
<p>xlabel</p>
</td></tr>
<tr><td><code id="ABCbarplot_+3A_ylab">ylab</code></td>
<td>
<p>ylabel</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ABC analysis is explained in <span class="pkg">ABCanalysis</span>. The visualization is based on <span class="pkg">ggplot2</span>.
</p>


<h3>Value</h3>

<p>List V of
</p>
<table>
<tr><td><code>ABCanalysis</code></td>
<td>
<p>output of <span class="pkg">ABCanalysis</span></p>
</td></tr>
<tr><td><code>ggobject</code></td>
<td>
<p>object of <span class="pkg">ggplot2</span> plotted</p>
</td></tr>
<tr><td><code>DF</code></td>
<td>
<p>Data frame if another plot should be done manually</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Ultsch. A ., Lotsch J.: Computed ABC Analysis for Rational Selection of Most Informative Variables in Multivariate Data, PloS one, Vol. 10(6), pp. e0129767. doi 10.1371/journal.pone.0129767, 2015.
</p>


<h3>See Also</h3>

<p><code><a href="ABCanalysis.html#topic+ABCanalysis">ABCanalysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('FundamentalData_Q1_2018')
Data=as.matrix(FundamentalData_Q1_2018$Data)
Data[!is.finite(Data)]=0
results=prcomp(Data)
main="Scree plot with Class A of the Most-Important Eigenvalues"
plotlist = ABCbarplot(results$sdev,ylab='Eigenvalues',main=main)
plotlist$ggobject
</code></pre>

<hr>
<h2 id='AccountingInformation_PrimeStandard_Q3_2019'>
Accounting Information in the Prime Standard in Q3 in 2019 (AI_PS_Q3_2019)
</h2><span id='topic+AccountingInformation_PrimeStandard_Q3_2019'></span><span id='topic+AI_PS_Q3_2019'></span>

<h3>Description</h3>

<p>Accounting Information of 261 companies traded in the Frankfurt stock exchange in the German Prime standard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("AccountingInformation_PrimeStandard_Q3_2019")</code></pre>


<h3>Format</h3>

<p>A list with of three objects
</p>

<dl>
<dt><code>Key</code></dt><dd><p>[1:n] Key of the 261 obeservations </p>
</dd>
<dt><code>Data</code></dt><dd><p>[1:n,1:d] numeric matrix of 261 observations on the 45 variables describing the accounting information</p>
</dd>
<dt><code>Cls</code></dt><dd><p>[1:n] a numeric vector of k clusters of the clustering performend in [Thrun/Ultsch, 2019]</p>
</dd>
</dl>



<h3>Details</h3>

<p>Detailed data description can be found in [Thrun/Ultsch, 2019].
</p>


<h3>Source</h3>

<p>Yahoo Finance 
</p>


<h3>References</h3>

<p>[Thrun/Ultsch, 2019]  Thrun, M. C., &amp; Ultsch, A.: Stock Selection via Knowledge Discovery using Swarm Intelligence with Emergence, IEEE Intelligent Systems, Vol. under review, pp., 2019.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(AccountingInformation_PrimeStandard_Q3_2019)

str(AI_PS_Q3_2019)
dim(AI_PS_Q3_2019$Data)
</code></pre>

<hr>
<h2 id='BimodalityAmplitude'>Bimodality Amplitude</h2><span id='topic+BimodalityAmplitude'></span>

<h3>Description</h3>

<p>Computes the Bimodality Amplitude of [Zhang et al., 2003]</p>


<h3>Usage</h3>

<pre><code class='language-R'>BimodalityAmplitude(x, PlotIt=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BimodalityAmplitude_+3A_x">x</code></td>
<td>
<p>Data vector.</p>
</td></tr>
<tr><td><code id="BimodalityAmplitude_+3A_plotit">PlotIt</code></td>
<td>
<p>FALSE, TRUE if  a figure with the antimodes and peaks is plotted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the Bimodality Ampltiude of a data vector.
This is a measure of the proportion of bimodality and the existence
of bimodality. The value lies between zero and one (that is: [0,1])
where the value of zero implies that the data is unimodal and the 
value of one implies the data is two point masses.
</p>


<h3>Note</h3>

<p>function was rewritten after the flow of a function of Sathish Deevi because the original function was incorrect.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Zhang, C., Mapes, B., &amp; Soden, B.:  Bimodality in tropical water vapour, Quarterly Journal of the Royal Meteorological Society, Vol. 129(594), pp. 2847-2866, 2003.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example 1
data&lt;-c(rnorm(299,0,1),rnorm(299,5,1))
BimodalityAmplitude(data,TRUE)

#Example 2
dist1&lt;-rnorm(2100,5,2)
dist2&lt;-dist1+11
data&lt;-c(dist1,dist2)

BimodalityAmplitude(data,TRUE)

#Example 3
dist1&lt;-rnorm(210,-15,1)
dist2&lt;-rep(dist1,3)+30
data&lt;-c(dist1,dist2)

BimodalityAmplitude(data,TRUE)

#Example 4
data&lt;-runif(1000,-15,1)

BimodalityAmplitude(data,TRUE)
</code></pre>

<hr>
<h2 id='categoricalVariable'>
A categorical Feature.
</h2><span id='topic+categoricalVariable'></span>

<h3>Description</h3>

<p>Character vector of length 391029 with five different labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("categoricalVariable")</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>data(categoricalVariable)
unique(categoricalVariable)
</code></pre>

<hr>
<h2 id='CCDFplot'> plot Complementary Cumulative Distribution Function (CCDF) in Log/Log  uses ecdf, CCDF(x) = 1-cdf(x) </h2><span id='topic+CCDFplot'></span>

<h3>Description</h3>

<p>plot Complementary Cumulative Distribution Function (CCDF) in Log/Log  uses ecdf, CCDF(x) = 1-cdf(x) </p>


<h3>Arguments</h3>

<table>
<tr><td><code id="CCDFplot_+3A_feature">Feature</code></td>
<td>
<p>Vector of data to be plotted, or a matrix with given probability density function in column 2 and/or a cumulative density function in column 3</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_pch">pch</code></td>
<td>
<p>Optional, default:  pch=0 for Line, other numbers see documentation about pch of plot</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_plotit">PlotIt</code></td>
<td>
<p>Optional, if PlotIt==T (default) do a plot, otherwise return only values</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_loglogplot">LogLogPlot</code></td>
<td>
<p>Optional, if LogLogPlot==T (default) do a log/log plot</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_xlab">xlab</code></td>
<td>
<p>Optional,  xlab of plot</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_ylab">ylab</code></td>
<td>
<p>Optional,  ylab of plot</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_main">main</code></td>
<td>
<p>Optional,  main of plot</p>
</td></tr>
<tr><td><code id="CCDFplot_+3A_...">...</code></td>
<td>
<p>Optional, further arguments for plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>V$CCDFuniqX,V$CCDFuniqY    CCDFuniqY= 1-cdf(CCDFuniqX), such that plot(CCDFuniqX,CCDFuniqY)...)</p>


<h3>Author(s)</h3>

<p>Michael Thrun</p>

<hr>
<h2 id='Choroplethmap'>
Plots the Choropleth Map
</h2><span id='topic+plotChoroplethMap'></span><span id='topic+Choroplethmap'></span>

<h3>Description</h3>

<p>A thematic map with areas colored in proportion to the measurement of the statistical variable being displayed on the map. A political map geneated by this function was used in the conference talk of the publication [Thrun/Ultsch, 2018].</p>


<h3>Usage</h3>

<pre><code class='language-R'>Choroplethmap(Counts, PostalCodes, NumberOfBins = 0,

 Breaks4Intervals, percentiles = c(0.5, 0.95), 
 
 digits = 0, PostalCodesShapes, PlotIt = TRUE,

 DiscreteColors, HighColorContinuous = "red",

 LowColorContinuous = "deepskyblue1", NAcolor = "grey",

 ReferenceMap = FALSE, main = "Political Map of Germany",

 legend = "Range of values", Silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Choroplethmap_+3A_counts">Counts</code></td>
<td>

<p>vector [1:m], statistical variable being displayed
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_postalcodes">PostalCodes</code></td>
<td>

<p>vector[1:n], currently german postal codes (zip codes), if <code>PostalCodesShapes</code> is not changed manually, does not need to be unique
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_numberofbins">NumberOfBins</code></td>
<td>

<p>Default: 1; 1 or below continously changes the color as defined by the   package <code>choroplethr</code>.
A Number between 2 and 9 sets equally sized bins. Higher numbers are not allowed
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_breaks4intervals">Breaks4Intervals</code></td>
<td>

<p>If NumberOfBins&gt;1 you can set here the intervals of the bins manually
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_percentiles">percentiles</code></td>
<td>

<p>If NumberOfBins&gt;1 and Breaks4Intervals not set, then the percentiles of min and max bin can be set here. See also <code>quantile</code>.
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_digits">digits</code></td>
<td>

<p>number of digits for <code>round</code>
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_postalcodesshapes">PostalCodesShapes</code></td>
<td>

<p>Specially prepared shape file with postal codes and geographic boundaries.
If you set this object, then you can use non german zip codes.
You can see the required structure in map.df,  github trulia choroplethr blob master r chloropleth. 
The German PostalCodesShapes can be downloaded from <a href="https://github.com/Mthrun/DataVisualizations/tree/master/data">https://github.com/Mthrun/DataVisualizations/tree/master/data</a>.
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_plotit">PlotIt</code></td>
<td>

<p>Either Plot the map directly or change the object manually before plotting it
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_discretecolors">DiscreteColors</code></td>
<td>

<p>Set the discrete colors manually if NumberOfBins&gt;1, else it is ignored
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_highcolorcontinuous">HighColorContinuous</code></td>
<td>

<p>if NumberOfBins&lt;=1: color of highest continuous value, else it is ignored
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_lowcolorcontinuous">LowColorContinuous</code></td>
<td>

<p>if NumberOfBins&lt;=1: color of lowest continuous value, else it is ignored
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_nacolor">NAcolor</code></td>
<td>

<p>Color of NA values in the map (postal codes without any counts)
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_referencemap">ReferenceMap</code></td>
<td>

<p>TRUE: With Google map, FALSE: without Google map
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_main">main</code></td>
<td>

<p>title of plot
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_legend">legend</code></td>
<td>

<p>title of legend
</p>
</td></tr>
<tr><td><code id="Choroplethmap_+3A_silent">Silent</code></td>
<td>

<p>TRUE: disable warnings of <code>choroplethr</code> package
FALSE: enable  warnings of <code>choroplethr</code> package
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This wrapper for the <span class="pkg">choroplethr</span> enables to visualize a political map easily in the case of german zip codes based on given counts and postal codes. Other postal codes are in principle usable.
</p>


<h3>Value</h3>

<p>List of
</p>
<table>
<tr><td><code>chorR6obj</code></td>
<td>
<p>An R6 object of the package <code>choroplethr</code></p>
</td></tr>
<tr><td><code>DataFrame</code></td>
<td>
<p>Transformed PostalCodes and Counts in a way that they can be used in the package <code>choroplethr</code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>You could read <a href="https://www.r-bloggers.com/2016/05/case-study-mapping-german-zip-codes-in-r/">https://www.r-bloggers.com/2016/05/case-study-mapping-german-zip-codes-in-r/</a>, if you want to change the map (<code>PostalCodesShapes</code> shape object).
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>


<h3>See Also</h3>

<p>Google <code>choroplethr</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#If you download the package from CRAN
## Not run: 
# 1. Step: Downlaod the shape file from the website 
# https://github.com/Mthrun/DataVisualizations/blob/master/data/GermanPostalCodesShapes.rda
# 2. Step: load it from the local path od the downloaded file with
load(file='GermanPostalCodesShapes.rda')

## End(Not run)

# If you download the package from GitHub, you can omit the two steps above.
# Then, do not use the 'PostalCodesShapes' input parameter

#Many postal codes are required to see a structure
#Exemplary two postal codes in the upper left corner of the map

## Not run: 
out=Choroplethmap(c(4,8,5,4),

c('49838', '26817', '49838', '26817'),

NumberOfBins=2,PlotIt=FALSE,
 PostalCodesShapes=GermanPostalCodesShapes)

out$chorR6obj$render()

## End(Not run)
#bins are only presented in the map if the have values within
## Not run: 
out=Choroplethmap(c(4,8,5,4),c('49838', '26817',

 '49838', '26817'),NumberOfBins=5,
 
 Breaks4Intervals=c(1,2,3,5,10),PlotIt=FALSE,
 PostalCodesShapes=GermanPostalCodesShapes)


out$chorR6obj$render()

## End(Not run)
# Result of [Thrun/Ultsch, 2018]
# Slightly misuse the function for visualizing a political map 
# resulting out of a clustering

## Not run: 
data('ChoroplethPostalCodesAndAGS_Germany')
res=Choroplethmap(as.numeric(ChoroplethPostalCodesAndAGS_Germany$Cls)+1,

ChoroplethPostalCodesAndAGS_Germany$PLZ,NumberOfBins = 2,

Breaks4Intervals = c(0,1,2,3,4,5,6),digits = 1,ReferenceMap = F,

DiscreteColors = c('white','green','blue','red','magenta'),

main = 'Classification of German Postal Codes based on Income Tax Share and Yield',

legend = 'ITS vs MTY Classification in 2010',NAcolor = 'black',PlotIt=FALSE,
 PostalCodesShapes=GermanPostalCodesShapes)


#takes time to process
res$chorR6obj$render()

## End(Not run)
</code></pre>

<hr>
<h2 id='ChoroplethPostalCodesAndAGS_Germany'>
Postal Codes and AGS of Germany for a Choropleth Map
</h2><span id='topic+ChoroplethPostalCodesAndAGS_Germany'></span>

<h3>Description</h3>

<p>Zip Codes and Community Identification Number of Germany which can be used in a Choropleth Map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ChoroplethPostalCodesAndAGS_Germany")</code></pre>


<h3>Format</h3>

<p>A data frame with 8702 observations on the following 4 variables.
</p>

<dl>
<dt><code>PLZ</code></dt><dd><p>German postal codes/zip codes</p>
</dd>
<dt><code>Cls</code></dt><dd><p>Clustering aggregated of germany postal codes by <code>MTY</code> and <code>ITS</code> features</p>
</dd>
<dt><code>AGS</code></dt><dd><p>It is the 'Amtlicher Gemeindeschluessel' (Community Identification Number) of German municipalities </p>
</dd>
<dt><code>Names</code></dt><dd><p>Names of municipalities</p>
</dd>
</dl>



<h3>Details</h3>

<p>CLS are the the labels of a MTS versus ITS Bayesian classification showing two main groups of low quota ('1') and high quota ('2') municipalities. Additionally, outliers are manually classified into two separated groups called sponsors ('3') and promoted ('4').
In the Bayesian Classification non classified data have the label '0'. If a 'AGS' code of a 'PLZ' was unclear than the label is 'NaN'.
</p>

<table>
<tr>
 <td style="text-align: left;">
Class </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> low quota </td><td style="text-align: left;">  high quota </td><td style="text-align: left;"> sponsors </td><td style="text-align: left;"> promoted </td><td style="text-align: left;"> non classified  </td><td style="text-align: left;"> unclear  mapping </td>
</tr>
<tr>
 <td style="text-align: left;">
Labels </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> 5 </td><td style="text-align: left;"> NaN </td>
</tr>
<tr>
 <td style="text-align: left;">
CountPerClass </td><td style="text-align: left;"> 31 </td><td style="text-align: left;"> 1325 </td><td style="text-align: left;"> 7239 </td><td style="text-align: left;">  10 </td><td style="text-align: left;"> 95 </td><td style="text-align: left;"> 5 </td><td style="text-align: left;"> 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Source</h3>

<p>Generated for [Thrun/Ultsch, 2018] using the approach of [Ultsch/Behnisch, 2017].
</p>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Ultsch/Behnisch, 2017]  Ultsch, A., Behnisch, M.: Effects of the payout system of income taxes to municipalities in Germany, Applied Geography, Vol. 81, pp. 21-31, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ChoroplethPostalCodesAndAGS_Germany)
str(ChoroplethPostalCodesAndAGS_Germany)
</code></pre>

<hr>
<h2 id='ClassBoxplot'>
Creates Boxplot plot for all classes
</h2><span id='topic+ClassBoxplot'></span>

<h3>Description</h3>

<p>Boxplot the data for all classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassBoxplot(Data, Cls,  ColorSequence = DataVisualizations::DefaultColorSequence,

 ClassNames = NULL,All=FALSE, PlotLegend = TRUE,

 main = 'Boxplot per Class', xlab = 'Classes', ylab = 'Range of Data')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassBoxplot_+3A_data">Data</code></td>
<td>

<p>Vector of the data to be plotted
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_cls">Cls</code></td>
<td>

<p>Vector of class identifiers.
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_colorsequence">ColorSequence</code></td>
<td>

<p>Optional: The sequence of colors used, Default: DefaultColorSequence()
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_classnames">ClassNames</code></td>
<td>

<p>Optional: The names of the classes. Default: C1 - C(Number of Classes)
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_all">All</code></td>
<td>

<p>Optional: adds full data vector for comparison against classes
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_plotlegend">PlotLegend</code></td>
<td>

<p>Optional: Add a legent to plot. Default: TRUE)
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_main">main</code></td>
<td>

<p>Optional: Title of the plot. Default: &quot;ClassBoxPlot&quot;&quot;
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_xlab">xlab</code></td>
<td>

<p>Optional: Title of the x axis. Default: &quot;Classes&quot;
</p>
</td></tr>
<tr><td><code id="ClassBoxplot_+3A_ylab">ylab</code></td>
<td>

<p>Optional: Title of the y axis. Default: &quot;Data&quot;
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A List of
</p>
<table>
<tr><td><code>ClassData</code></td>
<td>
<p>The DataFrame used to plot</p>
</td></tr>
<tr><td><code>ggobject</code></td>
<td>
<p>The ggplot2 plot object</p>
</td></tr>
</table>
<p>in mode <code>invisible</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun, Felix Pape
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(ITS)
#please download package from cran
#model=AdaptGauss::AdaptGauss(ITS)
#Classification=AdaptGauss::ClassifyByDecisionBoundaries(ITS,

#DecisionBoundaries = AdaptGauss::BayesDecisionBoundaries(model$Means,model$SDs,model$Weights))

DataVisualizations::ClassBoxplot(ITS,Classification)$ggobject

</code></pre>

<hr>
<h2 id='ClassErrorbar'>
ClassErrorbar
</h2><span id='topic+ClassErrorbar'></span>

<h3>Description</h3>

<p>Plots ClassErrorbars at Xvalue positions for one or more than one classes with user means and defined whiskers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassErrorbar(Xvalues, Ymatrix, Cls, ClassNames, ClassCols, ClassShape, 

MeanFun = median, SDfun, JitterPosition = 0.5,

main = "Error bar plot", xlab, ylab, WhiskerWidth = 7, Whisker_lwd = 1, BW = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassErrorbar_+3A_xvalues">Xvalues</code></td>
<td>
<p>[1:m] Numerical or character vector, positions of error bars (see details) in on x-axis for the m variables</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_ymatrix">Ymatrix</code></td>
<td>
<p>[1:n,1:d] of n cases and d=m*k variables with for which the error-bar statistics defined by MeanFun and SDfun should be computed</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_cls">Cls</code></td>
<td>
<p>Optional, [1:d] numerical vector of k classes for the d variables. Each class is one method that will be shown as distinctive set of error bars in the plot</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_classnames">ClassNames</code></td>
<td>
<p>Optional, [1:k] character vector of k methods</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_classcols">ClassCols</code></td>
<td>
<p>Optional, [1:k] character vector of k colors</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_classshape">ClassShape</code></td>
<td>
<p>Optional, [1:k] numerical vector of k shapes, see pch in <code><a href="#topic+Classplot">Classplot</a></code>
for details</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_meanfun">MeanFun</code></td>
<td>
<p>Optional, error bar statstic of mean points, default=median</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_sdfun">SDfun</code></td>
<td>
<p>Optional, error bar statstic for the length of whiskers, default is the robust estimation of standard deviation</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_jitterposition">JitterPosition</code></td>
<td>
<p>Optional, how much in values of Xvalues should the error bars jitter around Xvalues to not overlap</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_main">main</code></td>
<td>
<p>Optional, title of plot</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_xlab">xlab</code></td>
<td>
<p>Optional, x-axis label</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_ylab">ylab</code></td>
<td>
<p>Optional, y-axis label</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_whiskerwidth">WhiskerWidth</code></td>
<td>
<p>Optional, scalar above zero defining the width of the end of the whiskers</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_whisker_lwd">Whisker_lwd</code></td>
<td>
<p>Optional, scalar obove zero defining the thickness of the whisker lines</p>
</td></tr>
<tr><td><code id="ClassErrorbar_+3A_bw">BW</code></td>
<td>
<p>Optional, FALSE: usual ggplot2 background and style which is good for screen visualizations. Default: TRUE: theme_bw() is used which is more appropriate for publications</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If k=1, e.g., one method is used, d=m and Cls=rep(1,m).  All vector [1:k] assume the occurance of the classes in Cls as ordered with increasing value.
</p>
<p>Statistics are provided in long table format with the column names Xvalues, Mean, SD and Method. 
The method column specifies the names of the k classes.
</p>
<p>If <code>Xvalues</code> is a character vector (see example), ggplot2 automatically sets the position on the x-axis.
Otherwise specific numeric positions can be set. This allowes also for plotting a smooth line over the average (see example).
</p>


<h3>Value</h3>

<p>List with
</p>
<table>
<tr><td><code>ggobj</code></td>
<td>
<p>The ggplot object of the ClassErrorbar</p>
</td></tr>
<tr><td><code>Statistics</code></td>
<td>
<p>[1:(d*k)1:4] data frame of statstics per class used for plotting</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('FundamentalData_Q1_2018')
Data=as.matrix(FundamentalData_Q1_2018$Data)
Cls = FundamentalData_Q1_2018$Cls
Class1Data = matrix(NA, nrow = nrow(Data), ncol = 2)
Class2Data = matrix(NA, nrow = nrow(Data), ncol = 2)
Class1Data[which(Cls==1), ] = Data[which(Cls==1), c("TotalAssets", "TotalLiabilities")]
Class2Data[which(Cls==2), ] = Data[which(Cls==2), c("TotalAssets", "TotalLiabilities")]
YMatrix = cbind(Class1Data, 
                Class2Data)

#Option 1: character vector
ClassErrorbar(c("TotalRevenue","GrossProfit"), 
         YMatrix, c
		 (1,1,2,2), 
         ClassNames=c("Class 1", "Class 2"), 
         main="ClassErrorbar of Q1 2018 for total revenue and gross profit",
         xlab="GrossProfit/TotalRevenue", 
         ylab="Median +- std", 
         WhiskerWidth = 1)
		 
#Option 2: numerical vector
ClassErrorbar(c(1,2), 
		 YMatrix,
		 c(1,1,2,2), 
         ClassNames=c("Class 1", "Class 2"), 
         main="ClassErrorbar of Q1 2018 for total revenue and gross profit",
         xlab="GrossProfit/TotalRevenue", 
         ylab="Median +- std", 
         WhiskerWidth = 1)

#Option 3: numerical vector + line
## Not run: 
#arbitrary data
Y_someOtherData=cbind(YMatrix,YMatrix,
YMatrix,YMatrix)
some_values=c(2,3,4,5,6,8,9,10)
ClassErrorbar(some_values, 
		 Y_someOtherData,
		 c(1,1,2,2), 
         ClassNames=c("Class 1", "Class 2"), 
         main="ClassErrorbar of Q1 2018 for total revenue and gross profit",
         xlab="GrossProfit/TotalRevenue", 
         ylab="Median +- std", 
         WhiskerWidth = 1)$ggobj+
geom_smooth(method="auto", se=F, fullrange=F, level=0.95)

## End(Not run)
</code></pre>

<hr>
<h2 id='ClassMDplot'>
Class MDplot for Data w.r.t. all classes
</h2><span id='topic+ClassMDplot'></span>

<h3>Description</h3>

<p>Creates a Mirrored-Density plot w.r.t. to each class of a numerical vector of data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassMDplot(Data, Cls, ColorSequence = DataVisualizations::DefaultColorSequence,

                         ClassNames = NULL, PlotLegend = TRUE,Ordering = "Columnwise",
                         
                         main = 'MDplot for each Class',
                         
                         xlab = 'Classes', ylab = 'PDE of Data per Class',
                         
                         Fill = 'darkblue', MinimalAmoutOfData=40,
                         
                         MinimalAmoutOfUniqueData=12,SampleSize=1e+05,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassMDplot_+3A_data">Data</code></td>
<td>

<p>[1:n] Vector of the data to be plotted
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_cls">Cls</code></td>
<td>

<p>[1:n] Vector of class identifiers of k clusters one number is the label of one cluster
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_colorsequence">ColorSequence</code></td>
<td>

<p>Optional: [1:k] vector, The sequence of colors used, Default: DataVisualizations::DefaultColorSequence
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_classnames">ClassNames</code></td>
<td>

<p>Optional: [1:k] named numerical vector, The names of the classes. Default: Class 1 - Class k with k beeing the number of classes
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_plotlegend">PlotLegend</code></td>
<td>

<p>Optional: Add a legent to plot. Default: TRUE)
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_ordering">Ordering</code></td>
<td>

<p>Optional: Ordering of Classes, please see <code>MDplot</code> for details)
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_main">main</code></td>
<td>

<p>Optional: Title of the plot. Default: MDplot for each Class
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_fill">Fill</code></td>
<td>

<p>Optional: [1:k] Vector with the colors, the MD's are to be colored with. If only one value is given, all MD's are colored in the same color.
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_xlab">xlab</code></td>
<td>

<p>Optional: Title of the x axis. Default: &quot;Classes&quot;
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_ylab">ylab</code></td>
<td>

<p>Optional: Title of the y axis. Default: &quot;Data&quot;
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_minimalamoutofdata">MinimalAmoutOfData</code></td>
<td>

<p>Optional: numeric value defining a threshold. Below this threshold no density estimation is performed and a Jitter plot with a median line is drawn. Please see <code><a href="#topic+MDplot">MDplot</a></code> for details.
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_minimalamoutofuniquedata">MinimalAmoutOfUniqueData</code></td>
<td>

<p>Optional: numeric value defining a threshold. Below this threshold no density estimation and statistical testing is performed and a Jitter plot is drawn. Only Data Science experts should change this value after they understand how the density is estimated (see [Ultsch, 2005]).
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_samplesize">SampleSize</code></td>
<td>

<p>Optional: numeric value defining a threshold. Above this thresholdclass-wise uniform sampling of finite cases is performed in order to shorten computation time. If required, <code>SampleSize=n</code> can be set to omit this procedure. 
</p>
</td></tr>
<tr><td><code id="ClassMDplot_+3A_...">...</code></td>
<td>

<p>Further arguments that are documented in <code>MDplot</code> except for <code>OnlyPlotOutput</code> which is always true.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Further examples for the ClassMDplot can be found in <a href="https://md-plot.readthedocs.io/en/latest/application/example_application.html">https://md-plot.readthedocs.io/en/latest/application/example_application.html</a>.
</p>
<p>The <code>Cls</code> vector is reordered from lowest to highest number.
The <code>ClassNames</code> vector and <code>ColorSequence</code> vectors are matched by this ordering of <code>Cls</code>, i.e. the lowest number gets the first color or class name.
</p>


<h3>Value</h3>

<p>A List of
</p>
<table>
<tr><td><code>ClassData</code></td>
<td>
<p>The matrix [1:m,1:NoOfClasses] used to plot with the reordered Cls, rows are filled partly with NaN, m is the length of the number of data in largest class.</p>
</td></tr>
<tr><td><code>ggobject</code></td>
<td>
<p>The ggplot2 plot object</p>
</td></tr>
</table>
<p>in mode <code>invisible</code>
</p>


<h3>Note</h3>

<p>Function is still experimental because <code>ColorSequence</code> does not work yet, because we are unable to specify the colors in ggplot2. If someone knows a solution, please mail the maintainer of the package. Similar issue for <code>PlotLegend</code>.</p>


<h3>Author(s)</h3>

<p>Michael Thrun, Felix Pape
</p>


<h3>References</h3>

<p>Thrun, M. C., Breuer, L., &amp; Ultsch, A. : Knowledge discovery from low-frequency stream nitrate concentrations: hydrology and biology contributions, Proc. European Conference on Data Analysis (ECDA), Paderborn, Germany, 2018. 
</p>


<h3>See Also</h3>

<p><a href="https://md-plot.readthedocs.io/en/latest/application/example_application.html">https://md-plot.readthedocs.io/en/latest/application/example_application.html</a>
</p>
<p><code><a href="#topic+MDplot">MDplot</a></code>
<a href="https://pypi.org/project/md-plot/">https://pypi.org/project/md-plot/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(ITS)

#shortcut for example if AdaptGauss not installed
Classification = kmeans(ITS, centers = 2)$cluster

#better approach
#please download package from cran
#model=AdaptGauss::AdaptGauss(ITS)
#Classification=AdaptGauss::ClassifyByDecisionBoundaries(ITS,

#DecisionBoundaries = AdaptGauss::BayesDecisionBoundaries(model$Means,model$SDs,model$Weights))
ClassNames=c(1,2)
names(ClassNames)=c("Insert name \n of Class 1","Insert name \n  of Class 2")
ClassMDplot(ITS,Classification,ClassNames = ClassNames)

</code></pre>

<hr>
<h2 id='ClassPDEplot'>
PDE Plot for all classes
</h2><span id='topic+ClassPDEplot'></span>

<h3>Description</h3>

<p>PDEplot the data for all classes, weights the pdf with priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassPDEplot(Data, Cls, ColorSequence,

 ColorSymbSequence, PlotLegend = 1,

 SameKernelsAndRadius = 0, xlim, ylim, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassPDEplot_+3A_data">Data</code></td>
<td>

<p>The Data to be plotted
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_cls">Cls</code></td>
<td>

<p>Vector of class identifiers. Can be integers or NaN's, need not be consecutive nor positive
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_colorsequence">ColorSequence</code></td>
<td>

<p>Optional: the sequence of colors used, Default: DefaultColorSequence
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_colorsymbsequence">ColorSymbSequence</code></td>
<td>

<p>Optional: the plot symbols used (theoretisch nicht notwendig, da erst wichtig, wenn mehr als 562 Cluster)
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_plotlegend">PlotLegend</code></td>
<td>

<p>Optional: add a legent to plot (default == 1)
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_samekernelsandradius">SameKernelsAndRadius</code></td>
<td>

<p>Optional: Use the same PDE kernels and radii for all distributions (default == 0)
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_xlim">xlim</code></td>
<td>

<p>Optional: range of the x axis
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_ylim">ylim</code></td>
<td>

<p>Optional: range of the y axis
</p>
</td></tr>
<tr><td><code id="ClassPDEplot_+3A_...">...</code></td>
<td>

<p>further arguments passed to plot  
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Kernels of the Pareto density estimation in mode <code>invisible</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(ITS)
#please download package from cran
#model=AdaptGauss::AdaptGauss(ITS)
#Classification=AdaptGauss::ClassifyByDecisionBoundaries(ITS,

#DecisionBoundaries = AdaptGauss::BayesDecisionBoundaries(model$Means,model$SDs,model$Weights))

DataVisualizations::ClassPDEplot(ITS,Classification)$ggobject

</code></pre>

<hr>
<h2 id='ClassPDEplotMaxLikeli'>
Create PDE plot for all classes with maximum likelihood
</h2><span id='topic+ClassPDEplotMaxLikeli'></span>

<h3>Description</h3>

<p>PDEplot the data for allclasses, weight the Plot with 1 (= maximum likelihood)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassPDEplotMaxLikeli(Data, Cls, ColorSequence = DataVisualizations::DefaultColorSequence,

 ClassNames, PlotLegend = TRUE, MinAnzKernels = 0,PlotNorm,

 main = "Pareto Density Estimation (PDE)",

 xlab = "Data", ylab = "ParetoDensity", xlim, ylim, lwd=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_data">Data</code></td>
<td>

<p>The Data to be plotted
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_cls">Cls</code></td>
<td>

<p>Vector of class identifiers. Can be integers or NaN's, need not be consecutive nor positive
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_colorsequence">ColorSequence</code></td>
<td>

<p>Optional: the sequence of colors used, Default: DefaultColorSequence
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_classnames">ClassNames</code></td>
<td>

<p>Optional: the names of the classes to be displayed in the legend
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_plotlegend">PlotLegend</code></td>
<td>

<p>Optional: add a legent to plot (default == 1)
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_minanzkernels">MinAnzKernels</code></td>
<td>

<p>Optional: Minimum number of kernels
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_plotnorm">PlotNorm</code></td>
<td>

<p>Optional:  ==1 =&gt; plot Normal distribuion on top , ==2 = plot robust normal distribution,; default:  PlotNorm= 0
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_main">main</code></td>
<td>

<p>Optional: Title of the plot
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_xlab">xlab</code></td>
<td>

<p>Optional: title of the x axis
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_ylab">ylab</code></td>
<td>

<p>Optional: title of the y axis
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_xlim">xlim</code></td>
<td>

<p>Optional: area of the x-axis to be plotted
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_lwd">lwd</code></td>
<td>

<p>Optional: area of the y-axis to be plotted
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_ylim">ylim</code></td>
<td>

<p>numerical scalar defining the width of the lines
</p>
</td></tr>
<tr><td><code id="ClassPDEplotMaxLikeli_+3A_...">...</code></td>
<td>

<p>further arguments passed to plot  
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Kernels</code></td>
<td>
<p>Kernels of the distributions</p>
</td></tr>
<tr><td><code>ClassParetoDensities</code></td>
<td>
<p>Pareto densities for classes</p>
</td></tr>
<tr><td><code>ggobject</code></td>
<td>
<p>ggplot2 plot object. This should be used to further modify the plot</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Felix Pape
</p>


<h3>References</h3>

<p>Aubert, A. H., Thrun, M. C., Breuer, L., &amp; Ultsch, A. : Knowledge discovery from high-frequency stream nitrate concentrations: hydrology and biology contributions, Scientific reports, Nature, Vol. 6(31536), pp. doi 10.1038/srep31536, 2016. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


data(ITS)
#model=AdaptGauss::AdaptGauss(ITS)
##please download package from cran
#Classification=AdaptGauss::ClassifyByDecisionBoundaries(ITS,

#DecisionBoundaries = AdaptGauss::BayesDecisionBoundaries(model$Means,model$SDs,model$Weights))

DataVisualizations::ClassPDEplotMaxLikeli(ITS,Classification)$ggobject

</code></pre>

<hr>
<h2 id='Classplot'>
Classplot
</h2><span id='topic+Classplot'></span>

<h3>Description</h3>

<p>Allows to plot one time series or feauture with a classification as a labeled scatter plot with a line.
The colors are the labels defined by the classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Classplot(X, Y, Cls, Plotter,Names = NULL, na.rm = FALSE, 

xlab = "X", ylab = "Y", main = "Class Plot", Colors = NULL,

Size = 8,PointBorderCol="black",

LineColor = NULL, LineWidth = 1, LineType  = NULL, 

Showgrid  = TRUE, pch,  AnnotateIt = FALSE, SaveIt = FALSE, 

Nudge_x_Names = 0, Nudge_y_Names = 0, Legend = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Classplot_+3A_x">X</code></td>
<td>

<p>[1:n] numeric vector or time
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_y">Y</code></td>
<td>

<p>[1:n] numeric vector of feature
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_cls">Cls</code></td>
<td>

<p>[1:n] numeric vector of k classes, if not set per default every point is in first class
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_names">Names</code></td>
<td>

<p>[1:n] character vector of k classes, if not set per default Cls is used, if set, names the legend and the points
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_na.rm">na.rm</code></td>
<td>
<p>Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE</p>
</td></tr>
<tr><td><code id="Classplot_+3A_xlab">xlab</code></td>
<td>

<p>Optional, string for xlabel
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_ylab">ylab</code></td>
<td>

<p>Optional, string for ylabel
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_main">main</code></td>
<td>

<p>Optional, string for title of plot
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_colors">Colors</code></td>
<td>

<p>Optional, [1;k] string defining the k colors, one per class
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_annotateit">AnnotateIt</code></td>
<td>

<p>Optional, in case of <code>Plotter==ggplot</code> and given <code>Names</code> annotates each point if TRUE
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_size">Size</code></td>
<td>

<p>Optional, size of points, beware: default is appropriate for  &quot;<code>plotly</code>&quot;, or &quot;<code>native</code>&quot; but should smaller for &quot;<code>ggplot</code>&quot;
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_pointbordercol">PointBorderCol</code></td>
<td>

<p>Optional, string, color of the dot outline for &quot;<code>plotly</code>&quot; for &quot;<code>ggplot</code>&quot;. If <code>FALSE</code> and <code>Plotter="ggplot"</code> or <code>Plotter="plotly"</code>, no borders for points which is useful if many points overlap.
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_linecolor">LineColor</code></td>
<td>

<p>Optional, name of color, in plotly then all points are connected by a curve, 
</p>
<p>in ggplot2 all points of one class ae connected by a curve of the color the class
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_linewidth">LineWidth</code></td>
<td>

<p>Optional, number defining the width of the curve (plotly only)
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_linetype">LineType</code></td>
<td>

<p>Optional, string defining the type of the curve in plotly only, &quot;<code>dot</code>&quot;, &quot;<code>dash</code>&quot;, &quot;<code>-</code>&quot;
</p>
<p>for ggplot2: just set =1 here and then the curve is plotted
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_showgrid">Showgrid</code></td>
<td>

<p>Optional, boolean (plotly only)
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_plotter">Plotter</code></td>
<td>

<p>Optional, either &quot;<code>ggplot</code>&quot; (default if <code>Names</code> given), &quot;<code>plotly</code>&quot; (default if no <code>Names</code> given), or &quot;<code>native</code>&quot;
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_pch">pch</code></td>
<td>
<p> [1:n] numeric vector of length n of the cases of Cls for the k classes. It defines the symbols to use, for native <code>Plotter</code> or <code>ggplot</code>, usally k can be in a range from zero to 25</p>
</td></tr>
<tr><td><code id="Classplot_+3A_saveit">SaveIt</code></td>
<td>

<p>Optional, boolean, if true saves plot as html (plotly) or png (ggplot2)
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_nudge_x_names">Nudge_x_Names</code></td>
<td>

<p>Optional, numerical scalar, for <code>Plotter</code> &quot;<code>ggplot</code>&quot; only, if <code>Names</code> are set, moves them consistently respective to x-axis within units of x-axis
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_nudge_y_names">Nudge_y_Names</code></td>
<td>

<p>Optional, numerical scalar, for <code>Plotter</code> &quot;<code>ggplot</code>&quot; only, if <code>Names</code> are set, moves them consistently respective to y-axis within units of y-axis
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_legend">Legend</code></td>
<td>

<p>Optional, if argument is not missing, character string defining the title of the legend which automatically enables the legend
</p>
</td></tr>
<tr><td><code id="Classplot_+3A_...">...</code></td>
<td>

<p>Further arguments for <code>ggplot2::ggplot</code>,or  <code>plotly::plot_ly</code>, or <code><a href="base.html#topic+plot">plot</a></code> (except &quot;<code>pch</code>&quot;&quot; and &quot;<code>type</code>&quot;) depending on <code>Plotter</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default is &quot;<code>plotly</code>&quot; if <code>Names</code> are <code>NULL</code>. However,  ggplot2 is preferable in case that <code>Names</code> parameter is used because overlapping text labels are avoided. In that case  the default is &quot;<code>ggplot</code>&quot;. Note that ggplot2 options are currently slightly restricted. 
</p>
<p>For example, the function is usefull to see if temporal clustering has time dependent variations and for Hidden Markov Models (see Mthrun/RHmm on GitHub).
</p>


<h3>Value</h3>

<p>plotly object or ggplot2 objected depending on <code>Plotter</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DualaxisClassplot">DualaxisClassplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lsun3D)
Classplot(Lsun3D$Data[,1],Lsun3D$Data[,2],Lsun3D$Cls)

#ggplot 2 with different symbols
  Classplot(
    Lsun3D$Data[, 1],
    Lsun3D$Data[, 2],
    Lsun3D$Cls,
    Plotter = "ggplot2",
    Size = 3,
    pch = Lsun3D$Cls + 5
  )

#plotly with line
data(Lsun3D)
Classplot(Lsun3D$Data[,1],Lsun3D$Data[,2],Lsun3D$Cls,
LineType="-",LineColor = "green")

#ggplot2 with annotations
data(Lsun3D)
ind=sample(1:nrow(Lsun3D$Data),20)
Classplot(Lsun3D$Data[ind,1],Lsun3D$Data[ind,2],Lsun3D$Cls[ind],
Names = rownames(Lsun3D$Data)[ind],Size =1,
Plotter = "ggplot2",AnnotateIt = TRUE)



#ggplot2 with labels and legend per class
data(Lsun3D)
Classplot(Lsun3D$Data[,1],Lsun3D$Data[,2],Lsun3D$Cls,
Names = paste0("C",Lsun3D$Cls),Size =2,Legend ="Classes")

</code></pre>

<hr>
<h2 id='CombineCols'>
Combine vectors of various lengths
</h2><span id='topic+CombineCols'></span>

<h3>Description</h3>

<p>Combine arbitrary vectors of data, filling in missing rows with NaN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CombineCols(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CombineCols_+3A_...">...</code></td>
<td>

<p>d vectors of arbitrary lengths, see example
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Robust alternative to <code><a href="base.html#topic+cbind">cbind</a></code> that fills missing values with nan instead of extending length of vector by duplicating elements
</p>


<h3>Value</h3>

<p>matrix of dimensionality of n x d with n beeing the length of the longest vector and d the number of vectors given as input
</p>


<h3>Note</h3>

<p>special application by MCT of rowr cbind.fill which is now not on CRAN anymore
</p>


<h3>Author(s)</h3>

<p>Craig Varrichio
</p>


<h3>Examples</h3>

<pre><code class='language-R'>CombineCols(c(1,2,3),c(1),c(2,3))
</code></pre>

<hr>
<h2 id='Crosstable'>
Crosstable plot
</h2><span id='topic+Crosstable'></span>

<h3>Description</h3>

<p>Presents a heatmap with values and a cross table of given Data matrix of two features and a bin width or percentualized values. In this approach the bin width is fixes. A more general way to approach this is the kernel density estimation plot of <code><a href="ScatterDensity.html#topic+PDEscatter">PDEscatter</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Crosstable(Data, xbins = seq(0, 100, 5), ybins = xbins, 

NormalizationFactor = 1, PlotIt = TRUE, main='Cross Table',

PlotText=TRUE,TextDigits=0,TextProbs=c(0.05,0.95))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Crosstable_+3A_data">Data</code></td>
<td>

<p>[1:n,1:2] matrix of two features from which the cross table should be generated from
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_xbins">xbins</code></td>
<td>

<p>[1:k] start of k bins as a vector generated with <code><a href="base.html#topic+seq">seq</a></code> of the first feature of data. Default setting assumes percentiled values between zero and 100.
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_ybins">ybins</code></td>
<td>

<p>[1:k] start of k bins as a vector generated with <code><a href="base.html#topic+seq">seq</a></code> of the second feature of data. Normally the same for both features, other settings are only possible if the length <code>k</code> is equal.
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_normalizationfactor">NormalizationFactor</code></td>
<td>

<p>Optional, Data feautures can be seen as regular time series, e.g. 1 measurement for a minute, in this case it is useful to normalize the output, e.g. to hours, then <code>NormalizationFactor=60</code>
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_plotit">PlotIt</code></td>
<td>

<p>Optional, Plots the heatmap if <code>TRUE</code>.
The first feature is on the x-axis (left to right) and the second on y-axis (bottom to top).
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_main">main</code></td>
<td>

<p>In case of for <code>PlotIt=TRUE</code>: title of plot, see <code><a href="graphics.html#topic+title">title</a></code>
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_plottext">PlotText</code></td>
<td>

<p>In case of for <code>PlotIt=TRUE</code>: Default TRUE: plots text in heatmap with the values of the crosstable
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_textdigits">TextDigits</code></td>
<td>

<p>In case of for <code>TextDigits=TRUE</code>: integer indicating the number of decimal places to use in <code><a href="base.html#topic+round">round</a></code>.
</p>
</td></tr>
<tr><td><code id="Crosstable_+3A_textprobs">TextProbs</code></td>
<td>

<p>In case of for <code>TextDigits=TRUE</code>: [1:2] numeric vector of two probabilities defining the thresholds for white text to grey text and grey text to black text, e.g. below the first threshold (Default 0.05) all values (5% of values) will be printed in white because the lowest values of the heatmap are blue. The second value of 0.95 works well if cross table  has many  zeros; uses <code><a href="stats.html#topic+quantile">quantile</a></code> internally.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The interval in each bin is closed to the left and opened to the right. The cross table can be seen as a two-dimensional histogram. The idea to add histograms to the table is taken from [Charpentier. 2014].
</p>


<h3>Value</h3>

<p>The cross table in <code>invisible</code> mode which depicts the number of values (frequency) in an specific range with regard to two features. 
</p>
<p>The first feature is on the x-axis (left to right), and the second on y-axis (top to bottom) contrary to the plot where it is bottom to top.
</p>


<h3>Note</h3>

<p>For non percentiled values the <code>PlotText</code> part does not seem always to work, but I currently dont know why the text does not always overlap with the heatmap. 
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Charpentier. 2014]  Charpentier, Arthur, ed. Computational actuarial science with R. CRC Press, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+table">table</a></code>, <code><a href="graphics.html#topic+image">image</a></code>, <code><a href="ScatterDensity.html#topic+PDEscatter">PDEscatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ITS)
data(MTY)
#simple but not a good transformation
Data=(cbind(ITS/max(ITS),MTY/max(MTY)))*100
#choice for bins could be better
Crosstable(Data)
</code></pre>

<hr>
<h2 id='DefaultColorSequence'>
Default color sequence for plots
</h2><span id='topic+DefaultColorSequence'></span>

<h3>Description</h3>

<p>Defines the default color sequence for plots made within the Projections package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("DefaultColorSequence")</code></pre>


<h3>Format</h3>

<p>A vector with 562 different strings describing colors for plots.
</p>

<hr>
<h2 id='DensityContour'>Contour plot of densities</h2><span id='topic+DensityContour'></span>

<h3>Description</h3>

<p>Density estimation (PDE) [Ultsch, 2005] or &quot;SDH&quot; [Eilers/Goeman, 2004] used for a density contour plot. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DensityContour(X,Y, DensityEstimation="SDH", 

SampleSize, na.rm=FALSE,PlotIt=TRUE,
                              
NrOfContourLines=20,Plotter='ggplot', DrawTopView = TRUE,
                              
xlab, ylab, main="DensityContour",
                              
xlim, ylim, Legendlab_ggplot="value",

AddString2lab="",NoBinsOrPareto=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DensityContour_+3A_x">X</code></td>
<td>
<p>Numeric vector [1:n], first feature (for x axis values)</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_y">Y</code></td>
<td>
<p>Numeric vector [1:n], second feature (for y axis values)</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_densityestimation">DensityEstimation</code></td>
<td>
<p><code>"SDH"</code> is very fast but maybe not correct, <code>"PDE"</code> is slow but proably more correct, third alternativ is the typical R density estimation with <code>"kde2d"</code> which is sensitive to parameters</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_samplesize">SampleSize</code></td>
<td>
<p>Numeric, positiv scalar, maximum size of the sample used for calculation. High values increase runtime significantly. The default is that no sample is drawn</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_na.rm">na.rm</code></td>
<td>
<p>Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_plotit">PlotIt</code></td>
<td>
<p><code>TRUE</code>: plots with function call
</p>
<p><code>FALSE</code>: Does not plot, plotting can be done using the list element <code>Handle</code>
</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_nrofcontourlines">NrOfContourLines</code></td>
<td>
<p>Numeric, number of contour lines to be drawn. 20 by default.</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_plotter">Plotter</code></td>
<td>
<p>String, name of the plotting backend to use. Possible values are: &quot;<code>ggplot</code>&quot;, &quot;<code>plotly</code>&quot;. Default: ggplot</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_drawtopview">DrawTopView</code></td>
<td>
<p>Boolean, True means contur is drawn, otherwise a 3D plot is drawn. Default: TRUE</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_xlab">xlab</code></td>
<td>
<p>String, title of the x axis. Default: &quot;X&quot;,   see <code>plot()</code> function</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_ylab">ylab</code></td>
<td>
<p>String, title of the y axis. Default: &quot;Y&quot;,   see <code>plot()</code> function</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_main">main</code></td>
<td>

<p>string, the same as &quot;main&quot; in <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_xlim">xlim</code></td>
<td>

<p>see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_ylim">ylim</code></td>
<td>

<p>see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_legendlab_ggplot">Legendlab_ggplot</code></td>
<td>
<p>String, in case of <code>Plotter="ggplot"</code> label for the legend. Default: &quot;value&quot;</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_addstring2lab">AddString2lab</code></td>
<td>
<p>adds the same string of information to x and y axis label, e.g. usefull for adding SI units</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_nobinsorpareto">NoBinsOrPareto</code></td>
<td>
<p>Density specifc parameters, for <code>PDEscatter(ParetoRadius)</code> or SDH (nbins)) or kde2d(bins)</p>
</td></tr>
<tr><td><code id="DensityContour_+3A_...">...</code></td>
<td>
<p>further plot arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>DensityContour</code> function generates the density of the xy data as a z coordinate. Afterwards xyz will be plotted either as a contour plot or a 3d plot. It assumens that the cases of x and y are mapped to each other meaning that a <code>cbind(x,y)</code> operation is allowed.
This function plots the Density on top of a scatterplot. Variances of x and y should not differ by extreme numbers, otherwise calculate the percentiles on both first. If <code>DrawTopView=FALSE</code> only the plotly option is currently available. If another option is chosen, the method switches automatically there.
</p>
<p><code>PlotIt=FALSE</code> is usefull if one likes to perform adjustements like axis scaling prior to plotting with <span class="pkg">ggplot2</span> or  <span class="pkg">plotly</span>.
</p>


<h3>Value</h3>

<p>List of:
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>Numeric vector [1:m],m&lt;=n, first feature used in the plot or the kernels used</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Numeric vector [1:m],m&lt;=n, second feature used in the plot or the kernels used</p>
</td></tr>
<tr><td><code>Densities</code></td>
<td>
<p>Number of points within the ParetoRadius of each point, i.e. density information</p>
</td></tr>
<tr><td><code>Handle</code></td>
<td>
<p>Handle of the plot object</p>
</td></tr>
</table>


<h3>Note</h3>

<p>MT contributed with several adjustments
</p>


<h3>Author(s)</h3>

<p>Felix Pape</p>


<h3>References</h3>

<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, (Ultsch, A. &amp; Huellermeier, E. Eds., 10.1007/978-3-658-20540-9), Doctoral dissertation, Heidelberg, Springer, ISBN: 978-3658205393, 2018.
</p>
<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Ultsch, 2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, In Baier, D. &amp; Werrnecke, K. D. (Eds.), Innovations in classification, data science, and information systems, (Vol. 27, pp. 91-100), Berlin, Germany, Springer, 2005.
</p>
<p>[Eilers/Goeman, 2004]  Eilers, P. H., &amp; Goeman, J. J.: Enhancing scatterplots with smoothed densities, Bioinformatics, Vol. 20(5), pp. 623-628. 2004.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#taken from [Thrun/Ultsch, 2018]
data("ITS")
data("MTY")
Inds=which(ITS&lt;900&amp;MTY&lt;8000)
plot(ITS[Inds],MTY[Inds],main='Bimodality is not visible in normal scatter plot')

DensityContour(ITS[Inds],MTY[Inds],DensityEstimation="SDH",xlab = 'ITS in EUR',

ylab ='MTY in EUR' ,main='Smoothed Densities histogram indicates Bimodality' )

DensityContour(ITS[Inds],MTY[Inds],DensityEstimation="PDE",xlab = 'ITS in EUR',

ylab ='MTY in EUR' ,main='PDE indicates Bimodality' )


</code></pre>

<hr>
<h2 id='DensityScatter'>
Scatter plot with densities
</h2><span id='topic+DensityScatter'></span>

<h3>Description</h3>

<p>Density estimation is performed by (PDE) [Ultsch, 2005] or &quot;SDH&quot; [Eilers/Goeman, 2004] and visualized in a  density scatter plot [Brinkmann et al., 2023] in which the points are colored by their density.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DensityScatter(X,Y,DensityEstimation="SDH",

Type="DDCAL", Plotter = "native",Marginals = FALSE,

SampleSize,na.rm=FALSE, xlab, ylab, 

main="DensityScatter", AddString2lab="",
                        
xlim, ylim,NoBinsOrPareto=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DensityScatter_+3A_x">X</code></td>
<td>

<p>Numeric vector [1:n], first feature (for x axis values)
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_y">Y</code></td>
<td>

<p>Numeric vector [1:n], second feature (for y axis values)
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_densityestimation">DensityEstimation</code></td>
<td>

<p>(Optional), <code>"SDH"</code> is very fast but maybe not correct, <code>"PDE"</code> is slow but proably more correct, third alternativ is the typical R density estimation with <code>"kde2d"</code> which is sensitive to parameters
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_type">Type</code></td>
<td>

<p>(Optional), <code>"DDCAL"</code> uses a new density to point color matching by DDCAL algorithm [Lux/Rinderle-Ma, 2023] , <code>"native"</code> uses a simple density to point color matching
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_plotter">Plotter</code></td>
<td>

<p>in case of <code>Type="DDCAL"</code>, (Optional) String, name of the plotting backend to use. Possible values are: &quot;<code>native</code>&quot;,&quot;<code>plotly</code>&quot;, or &quot;<code>ggplot2</code>&quot;
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_marginals">Marginals</code></td>
<td>

<p>(Optional) Boolean, if TRUE the marginal distributions of X and Y will be plotted together with the 2D density of X and Y. Default is FALSE
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_samplesize">SampleSize</code></td>
<td>

<p>(Optional), Numeric, positiv scalar, maximum size of the sample used for calculation. High values increase runtime significantly. The default is that no sample is drawn
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_na.rm">na.rm</code></td>
<td>

<p>(Optional), Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_xlab">xlab</code></td>
<td>

<p>(Optional), String, title of the x axis. Default: &quot;X&quot;,   see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_ylab">ylab</code></td>
<td>

<p>(Optional), String, title of the y axis. Default: &quot;Y&quot;,   see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_main">main</code></td>
<td>

<p>(Optional), string, the same as &quot;main&quot; in <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_addstring2lab">AddString2lab</code></td>
<td>

<p>(Optional), adds the same string of information to x and y axis label, e.g. usefull for adding SI units
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_xlim">xlim</code></td>
<td>

<p>(Optional), in case of <code>Type="natuive"</code>, see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_ylim">ylim</code></td>
<td>

<p>in case of <code>Type="natuive"</code>, see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_nobinsorpareto">NoBinsOrPareto</code></td>
<td>

<p>(Optional), in case of <code>Type="natuive"</code>, Density specifc parameters, for <code>PDEscatter(ParetoRadius)</code> or SDH (nbins)) or kde2d(bins)
</p>
</td></tr>
<tr><td><code id="DensityScatter_+3A_...">...</code></td>
<td>

<p>(Optional), further arguments either to ScatterDenstiy::DensityScatter.DDCAL or to plot()
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>DensityScatter</code> function generates the density of the xy data as a z coordinate. Afterwards xy points will be plotted as a scatter plot, where the z values defines the coloring of the xy points. It assumens that the cases of x and y are mapped to each other meaning that a <code>cbind(x,y)</code> operation is allowed.
This function plots the Density on top of a scatterplot. Variances of x and y should not differ by extreme numbers, otherwise calculate the percentiles on both first.
</p>


<h3>Value</h3>

<p>List of:
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>Numeric vector [1:m],m&lt;=n, first feature used in the plot or the kernels used</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Numeric vector [1:m],m&lt;=n, second feature used in the plot or the kernels used</p>
</td></tr>
<tr><td><code>Densities</code></td>
<td>
<p>Number of points within the ParetoRadius of each point, i.e. density information</p>
</td></tr>
</table>


<h3>Note</h3>

<p>MT contributed with several adjustments
</p>


<h3>Author(s)</h3>

<p>Felix Pape
</p>


<h3>References</h3>

<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, (Ultsch, A. &amp; Huellermeier, E. Eds., 10.1007/978-3-658-20540-9), Doctoral dissertation, Heidelberg, Springer, ISBN: 978-3658205393, 2018.
</p>
<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Ultsch, 2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, In Baier, D. &amp; Werrnecke, K. D. (Eds.), Innovations in classification, data science, and information systems, (Vol. 27, pp. 91-100), Berlin, Germany, Springer, 2005.
</p>
<p>[Eilers/Goeman, 2004]  Eilers, P. H., &amp; Goeman, J. J.: Enhancing scatterplots with smoothed densities, Bioinformatics, Vol. 20(5), pp. 623-628. 2004
</p>
<p>[Lux/Rinderle-Ma, 2023]  Lux, M. &amp; Rinderle-Ma, S.: DDCAL: Evenly Distributing Data into Low Variance Clusters Based on Iterative Feature Scaling, Journal of Classification vol. 40, pp. 106-144, 2023.
</p>
<p>[Brinkmann et al., 2023]  Brinkmann, L., Stier, Q., &amp; Thrun, M. C.: Computing Sensitive Color Transitions for the Identification of Two-Dimensional Structures, Proc. Data Science, Statistics &amp; Visualisation (DSSV) and the European Conference on Data Analysis (ECDA), p.109, Antwerp, Belgium, July 5-7, 2023. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#taken from [Thrun/Ultsch, 2018]
data("ITS")
data("MTY")
Inds=which(ITS&lt;900&amp;MTY&lt;8000)
plot(ITS[Inds],MTY[Inds],main='Bimodality is not visible in normal scatter plot')

DensityScatter(ITS[Inds],MTY[Inds],DensityEstimation="SDH",xlab = 'ITS in EUR',

ylab ='MTY in EUR' ,main='Smoothed Densities histogram indicates Bimodality' )

DensityScatter(ITS[Inds],MTY[Inds],DensityEstimation="PDE",xlab = 'ITS in EUR',

ylab ='MTY in EUR' ,main='PDE indicates Bimodality' )


</code></pre>

<hr>
<h2 id='DualaxisClassplot'>
Dualaxis Classplot
</h2><span id='topic+DualaxisClassplot'></span>

<h3>Description</h3>

<p>Allows to plot two time series or features with one or two classification(a) as labeled  scatter plots.
The colors are the labels defined by the classification.
Usefull to see if temporal clustering has time dependent variations and for Hidden Markov Models (see Mthrun/RHmm on GitHub).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DualaxisClassplot(X, Y1, Y2, Cls1,

Cls2, xlab = "X", y1lab = "Y1", y2lab = "Y2",

main = "Dual Axis Class Plot", Colors, Showgrid = TRUE, SaveIt = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DualaxisClassplot_+3A_x">X</code></td>
<td>

<p>[1:n] numeric vector or time
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_y1">Y1</code></td>
<td>

<p>[1:n] numeric vector of feauture
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_y2">Y2</code></td>
<td>

<p>[1:n] numeric vector of feauture
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_cls1">Cls1</code></td>
<td>

<p>[1:n] numeric vector defining a classification of k1 classes
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_cls2">Cls2</code></td>
<td>

<p>Optional, [1:n] numeric vector defining a classification of k2 classes for <code>Y2</code>
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_xlab">xlab</code></td>
<td>

<p>Optional, string
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_y1lab">y1lab</code></td>
<td>

<p>Optional, string
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_y2lab">y2lab</code></td>
<td>

<p>Optional, string
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_main">main</code></td>
<td>

<p>Optional, string
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_colors">Colors</code></td>
<td>

<p>[1:(k1+k2)] Colornames
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_showgrid">Showgrid</code></td>
<td>

<p>Optional, boolean
</p>
</td></tr>
<tr><td><code id="DualaxisClassplot_+3A_saveit">SaveIt</code></td>
<td>

<p>Optional, boolean
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotly object
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Classplot">Classplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##ToDo
</code></pre>

<hr>
<h2 id='DualaxisLinechart'>
DualaxisLinechart
</h2><span id='topic+DualaxisLinechart'></span>

<h3>Description</h3>

<p>A line chart with dual axisSS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DualaxisLinechart(X, Y1, Y2, xlab = "X", 

y1lab = "Y1", y2lab = "Y2", main = "Dual Axis Line Chart",

cols = c("black", "blue"),Overlaying="y", SaveIt = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DualaxisLinechart_+3A_x">X</code></td>
<td>
<p>[1:n] vector, both lines require the same xvalues, e.g. the time of the time series, <code>POSIXlt</code> or <code>POSIXct</code> are accepted</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_y1">Y1</code></td>
<td>

<p>[1:n] vector of first line
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_y2">Y2</code></td>
<td>

<p>[1:n] vector of second line
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_xlab">xlab</code></td>
<td>

<p>Optional, string for xlabel
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_y1lab">y1lab</code></td>
<td>

<p>Optional, string for first ylabel
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_y2lab">y2lab</code></td>
<td>

<p>Optional, string for second ylabel
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_main">main</code></td>
<td>

<p>Optional, title of plot
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_cols">cols</code></td>
<td>

<p>Optional, color of two lines
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_overlaying">Overlaying</code></td>
<td>

<p>Change only default in case of using <code><a href="plotly.html#topic+subplot">subplot</a></code>
</p>
</td></tr>
<tr><td><code id="DualaxisLinechart_+3A_saveit">SaveIt</code></td>
<td>

<p>Optional, default FALSE; TRUE if you want to save plot as html in <code>getwd()</code> directory
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>enables to visualize to lines in one plot overlaying them using ploty (e.g. two time series with two ranges of values)
</p>


<h3>Value</h3>

<p><code>plotly</code> object
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#subplot renames the numbering of subsequent plots
y1=runif(100,0,1)
y2=rnorm(100,m=5,s=1)
DualaxisLinechart(1:100, y1, y2,main="Random Time series")


y1=runif(100,0,1)
y2=(1:100*3+4)*runif(100,0,1)
p1=DualaxisLinechart(1:100, y1, y2,main="Random Time series",Overlaying="y2")

y3=1:100*(-2)+4
y4=rnorm(100,m=0,s=2)
p2=DualaxisLinechart(1:100, y3, y4,main="Random Time series",Overlaying="y4")
plotly::subplot(p1,p2)

</code></pre>

<hr>
<h2 id='estimateDensity2D'>
estimateDensity2D
</h2><span id='topic+estimateDensity2D'></span>

<h3>Description</h3>

<p>Estimates densities for two-dimensional data with the given estimation type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateDensity2D(X, Y, DensityEstimation = "SDH",

SampleSize, na.rm = FALSE, NoBinsOrPareto = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimateDensity2D_+3A_x">X</code></td>
<td>

<p>[1:n] numerical vector of first feature
</p>
</td></tr>
<tr><td><code id="estimateDensity2D_+3A_y">Y</code></td>
<td>

<p>[1:n] numerical vector of second feature
</p>
</td></tr>
<tr><td><code id="estimateDensity2D_+3A_densityestimation">DensityEstimation</code></td>
<td>

<p>Either &quot;PDE&quot;,&quot;SDH&quot; or &quot;kde2d&quot;
</p>
</td></tr>
<tr><td><code id="estimateDensity2D_+3A_samplesize">SampleSize</code></td>
<td>

<p>Sample Size in case of big data
</p>
</td></tr>
<tr><td><code id="estimateDensity2D_+3A_na.rm">na.rm</code></td>
<td>

<p>Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE
</p>
</td></tr>
<tr><td><code id="estimateDensity2D_+3A_nobinsorpareto">NoBinsOrPareto</code></td>
<td>

<p>Density specifc parameters, for PDEscatter(ParetoRadius) or SDH (nbins)) or kde2d(bins)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each two-dimensional data point is defined by its corresponding X and Y value.
</p>


<h3>Value</h3>

<p>List V with
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>[1:m] numerical vector of first feature, m&lt;=n depending if all values are finite an na.rm parameter</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>[1:m] numerical vector of second feature, m&lt;=n depending if all values are finite an na.rm parameter</p>
</td></tr>
<tr><td><code>Densities</code></td>
<td>
<p>the density of each two-dimensional data point</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Luca Brinkman and Michael Thrun
</p>


<h3>References</h3>

<p>[Ultsch, 2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, In Baier, D. &amp; Werrnecke, K. D. (Eds.), Innovations in classification, data science, and information systems, (Vol. 27, pp. 91-100), Berlin, Germany, Springer, 2005.
</p>
<p>[Eilers/Goeman, 2004]  Eilers, P. H., &amp; Goeman, J. J.: Enhancing scatterplots with smoothed densities, Bioinformatics, Vol. 20(5), pp. 623-628. 2004
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=runif(100)
Y=rnorm(100)
#V=estimateDensity2D(X,Y)
</code></pre>

<hr>
<h2 id='Fanplot'>
The fan plot
</h2><span id='topic+Fanplot'></span>

<h3>Description</h3>

<p>The better alternative to the pie chart represents amount of values given in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fanplot(Datavector,Names,Labels,MaxNumberOfSlices,main='',col,
MaxPercentage=FALSE,ShrinkPies=0.05,Rline=1.1, lwd=2,LabelCols="black",...)
</code></pre>


<h3>Arguments</h3>

 
<table>
<tr><td><code id="Fanplot_+3A_datavector">Datavector</code></td>
<td>

<p>[1:n] a vector of n non unique values
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_names">Names</code></td>
<td>
<p>Optional, 
[1:k] names to search for in Datavector, if not set <code>unique</code> of Datavector is calculated.
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_labels">Labels</code></td>
<td>
<p>Optional,
[1:k] Labels if they are specially named, if not Names are used.
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_maxnumberofslices">MaxNumberOfSlices</code></td>
<td>
<p>Default is k, integer value defining how many labels will be shown. Everything else will be summed up to <code>Other</code>.
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_main">main</code></td>
<td>

<p>Optional, title below the fan pie, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_col">col</code></td>
<td>
<p>Optional, the default are the first [1:k] colors of the default color sequence used in this package, otherwise a character vector of [1:k] specifying the colors analog to <code>plot</code>
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_maxpercentage">MaxPercentage</code></td>
<td>

<p>default FALSE; if true the biggest slice is 100 percent instead of the biggest procentual count
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_shrinkpies">ShrinkPies</code></td>
<td>

<p>Optional, distance between biggest and smallest slice of the pie
</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_rline">Rline</code></td>
<td>
<p>Optional, the distance between text and pie is defined here as the length of the line in numerical numbers</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_lwd">lwd</code></td>
<td>
<p>Optional, The line width, a positive number, defaut is 2</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_labelcols">LabelCols</code></td>
<td>
<p>Color of labels</p>
</td></tr>
<tr><td><code id="Fanplot_+3A_...">...</code></td>
<td>
<p>Further arguments to <code><a href="plotrix.html#topic+fan.plot">fan.plot</a></code> like circumferential positions for the labels <code>labelpos</code> or additional arguments passed to <code><a href="graphics.html#topic+polygon">polygon</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A normal pie plot is dificult to interpret for a human observer, because humans are not trained well to observe angles [Gohil, 2015, p. 102]. Therefore, the fan plot is used. As proposed in [Gohil 2015] the <code>fan.plot</code>() of the <code>plotrix</code> package is used to solve this problem.
If Number of Slices is higher than MaxNumberOfSlices then <code>ABCanalysis</code> is applied (see [Ultsch/Lotsch, 2015]) and group A chosen.
If Number of Slices in group A is higher than MaxNumberOfSlices, then the most important ones out of group A are chosen.
If MaxNumberOfSlices is higher than Slices in group A, additional slices are shown depending on the percentage (from high to low).
</p>
<p>Color sequence is automatically shortened to the MaxNumberOfSlices used in the fan plot. 
</p>


<h3>Value</h3>

<p>silent output by calling <code>invisible</code> of a list with
</p>
<table>
<tr><td><code>Percentages</code></td>
<td>

<p>[1:k] percent values visualized in fanplot
</p>
</td></tr>
<tr><td><code>Labels</code></td>
<td>

<p>[1:k] see input <code>Labels</code>, only relevant ones
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Gohil, 2015]  Gohil, Atmajitsinh. R data Visualization cookbook. Packt Publishing Ltd, 2015.
</p>
<p>[Ultsch/Lotsch, 2015]   Ultsch. A ., Lotsch J.: Computed ABC Analysis for Rational Selection of Most Informative Variables in Multivariate Data, PloS one, Vol. 10(6), pp. e0129767. doi 10.1371/journal.pone.0129767, 2015.
</p>


<h3>See Also</h3>

<p><code><a href="plotrix.html#topic+fan.plot">fan.plot</a></code>
<code><a href="#topic+Piechart">Piechart</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(categoricalVariable)
Fanplot(categoricalVariable)
</code></pre>

<hr>
<h2 id='FundamentalData_Q1_2018'>
Fundamental Data of the 1st Quarter in 2018
</h2><span id='topic+FundamentalData_Q1_2018'></span>

<h3>Description</h3>

<p>This dataset was extracted out of Yahoo finance and was investigated in [Thrun et al., 2019] and clustered in [Thrun, 2019].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("FundamentalData_Q1_2018")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 3
$ Data :'data.frame':	269 obs. of  45 variables:
..$ TotalRevenue                         : num [1:269] 3779000 78225 48220 63726 3084 ...
..$ CostofRevenue                        : num [1:269] 2348000 60835 26174 35203 882 ...
..$ GrossProfit                          : num [1:269] 1431000 17390 22046 28523 2202 ...
..$ SellingGeneralandAdministrative      : num [1:269] 459000 NaN 15162 17072 2005 ...
..$ Others                               : num [1:269] -3000 10272 -52 3131 1784 ...
..$ TotalOperatingExpenses               : num [1:269] 2872000 73833 41284 56787 5081 ...
..$ OperatingIncomeorLoss                : num [1:269] 907000 4392 6936 6939 -1997 ...
..$ TotalOtherIncomeDIVxpensesNet        : num [1:269] -28000 -344 1 -210 -240 ...
..$ EarningsBeforeInterestandTaxes       : num [1:269] 907000 4392 6936 6939 -1997 ...
..$ InterestExpense                      : num [1:269] -20000 -415 NaN -243 -238 ...
..$ IncomeBeforeTax                      : num [1:269] 879000 4048 6937 6729 -2237 ...
..$ IncomeTaxExpense                     : num [1:269] 233000 1365 2188 1896 7 ...
..$ NetIncomeFromContinuingOps           : num [1:269] 646000 2683 4749 4833 -2244 ...
..$ NetIncome_x                          : num [1:269] 644000 2817 4645 4833 -2244 ...
..$ NetIncome                            : num [1:269] 644000 2817 4645 4833 -2244 ...
..$ CashAndCashEquivalents               : num [1:269] 926000 29047 45911 94859 11217 ...
..$ NetReceivables                       : num [1:269] 2527000 46171 20774 151952 2774 ...
..$ Inventory                            : num [1:269] 2011000 471 NaN 10572 8924 ...
..$ TotalCurrentAssets                   : num [1:269] 5674000 80224 68061 267187 25989 ...
..$ LongTermInvestments                  : num [1:269] 234000 450 NaN 4155 872 ...
..$ PropertyPlantandEquipment            : num [1:269] 4216000 14561 3093 32247 7073 ...
..$ IntangibleAssets                     : num [1:269] 78000 40706 3975 6169 125 ...
..$ OtherAssets                          : num [1:269] 810000 8224 1091 2978 13310 ...
..$ DeferredLongTermAssetCharges         : num [1:269] 759000 684 1091 784 1405 ...
..$ TotalAssets                          : num [1:269] 11262000 167807 83155 351220 47369 ...
..$ AccountsPayable                      : num [1:269] 1442000 10567 1698 17316 1386 ...
..$ ShortDIVurrentLongTermDebt           : num [1:269] 1275000 30192 NaN 26668 917 ...
..$ OtherCurrentLiabilities              : num [1:269] 1064000 36942 22781 92297 2659 ...
..$ TotalCurrentLiabilities              : num [1:269] 2577000 54430 24479 114210 4299 ...
..$ OtherLiabilities                     : num [1:269] 1795000 19435 6876 29347 2018 ...
..$ TotalLiabilities                     : num [1:269] 5576000 97136 31355 165628 6980 ...
..$ CommonStock                          : num [1:269] 198000 14946 5198 15250 28644 ...
..$ RetainedEarnings                     : num [1:269] NaN 44030 34767 40374 -8965 ...
..$ TreasuryStock                        : num [1:269] 5455000 11686 NaN 129968 20710 ...
..$ OtherStockholderEquity               : num [1:269] 5455000 11686 NaN 129968 20710 ...
..$ TotalStockholderEquity               : num [1:269] 5653000 70662 51212 185592 40389 ...
..$ NetTangibleAssets                    : num [1:269] 5325000 6314 40302 140939 40264 ...
..$ Depreciation                         : num [1:269] 156000 2728 331 1381 410 ...
..$ AdjustmentsToNetIncome               : num [1:269] 216000 1911 116 2912 39 ...
..$ ChangesInOtherOperatingActivities    : num [1:269] -20000 -2174 -829 NaN 428 ...
..$ TotalCashFlowFromOperatingActivities : num [1:269] 452000 7349 4274 -8241 -1367 ...
..$ CapitalExpenditures                  : num [1:269] -88000 -966 -1778 -2067 -155 ...
..$ TotalCashFlowsFromInvestingActivities: num [1:269] 30000 -879 -1766 -2746 -484 ...
..$ TotalCashFlowsFromFinancingActivities: num [1:269] -789000 -6660 -21867 -961 -204 ...
..$ ChangeInCashandCashEquivalents       : num [1:269] -306000 -215 2508 -11842 -2062 ...
$ Names: chr [1:269, 1:6] &quot;1COV&quot; &quot;A1OS&quot; &quot;AAD&quot; &quot;AAG&quot; ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : NULL
.. ..$ : chr [1:6] &quot;Key&quot; &quot;ISIN&quot; &quot;Company&quot; &quot;Sector&quot; ...
$ Cls  : num [1:269] 1 1 1 1 2 1 1 1 3 1 ...
</p>


<h3>Details</h3>

<p>Stocks are selected by the German Prime standard accoridingly to the &quot;Names&quot; data frame.
Fundamental Data with missing values is stored in &quot;Data&quot;. The rownames of &quot;Data&quot; have the same Key as the first row of &quot;Names&quot; which is the trading symbol.
&quot;Cls&quot; provides the clustering as a numerical vector of 1:k classes performed by Databionic Swarm in [Thrun, 2019].
</p>


<h3>Source</h3>

<p>Yahoo finance
</p>


<h3>References</h3>

<p>Thrun, M. C., : Knowledge Discovery in Quarterly Financial Data of Stocks Based on the Prime Standard using a Hybrid of a Swarm with SOM, in Verleysen, M. (Ed.), European Symposium on Artificial Neural Networks, Computational Intelligence and Machine Learning (ESANN), Vol. 27, pp. 397-402, Ciaco, ISBN: 978-287-587-065-0, Bruges, Belgium, 2019. 
</p>
<p>[Thrun et al., 2019]	Thrun, M. C., Gehlert, Tino, &amp; Ultsch, A. : Analyzing the Fine Structure of Distributions, arXiv:1908.06081, 2019. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(FundamentalData_Q1_2018)
## maybe str(FundamentalData_Q1_2018) ; plot(FundamentalData_Q1_2018) ...
</code></pre>

<hr>
<h2 id='GoogleMapsCoordinates'>
Google Maps with marked coordinates 
</h2><span id='topic+GoogleMapsCoordinates'></span>

<h3>Description</h3>

<p>Google Maps with marked coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GoogleMapsCoordinates(Longitude,Latitude,Cls=rep(1,length(Longitude)),
zoom=3,location= c(mean(Longitude),mean(Latitude)),stroke=1.7,size=6,sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GoogleMapsCoordinates_+3A_longitude">Longitude</code></td>
<td>
<p> sphaerischer winkel der Kugeloberflaeche, coord 1
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_latitude">Latitude</code></td>
<td>
<p>sphaerischer winkel der Kugeloberflaeche,  coord 2
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_cls">Cls</code></td>
<td>
<p>Vorklassification/Clusterung
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_zoom">zoom</code></td>
<td>

<p>map zoom, an integer from 3 (continent) to 21 (building), default value 10 (city). 
openstreetmaps limits a zoom of 18, and the limit on stamen maps depends on the maptype. 
&quot;auto&quot; automatically determines the zoom for bounding box specifications, and is defaulted to 10
with center/zoom specifications. maps of the whole world currently not supported
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_location">location</code></td>
<td>
<p>Optional, 
default: c(mean(Longitude),mean(Latitude); an address, longitude/latitude pair (in that order), 
or left/bottom/right/top bounding box
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_stroke">stroke</code></td>
<td>
<p>Optional,  plotting parameter, dicke der linien der coordiantensymbole
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_size">size</code></td>
<td>
<p>Optional, plotting parameter, groesse der koordinatensymbole
</p>
</td></tr>
<tr><td><code id="GoogleMapsCoordinates_+3A_sequence">sequence</code></td>
<td>
<p>Optional, vector of length of number of clusers with numbers indicating the plotting symbols and colors to use
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plot was used in [Thrun, 2018, p. 135].
</p>


<h3>Value</h3>

<p>ggobject()
</p>


<h3>Note</h3>

<p>requires an Internet connection,
requires an API key of Google.
See <code>?ggmap::register_google</code> for details.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun, 2018] Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, ISBN: 978-3-658-20539-3, Heidelberg, 2018.
</p>

<hr>
<h2 id='Heatmap'>
Heatmap for Clustering
</h2><span id='topic+Heatmap'></span>

<h3>Description</h3>

<p>Heatmap of Distances of Data sorted by Cls. Clustering algorithms provide a Classifcation of data,  where the labels are defined as a numeric vector <code>Cls</code>. Then, a typical cluster-respectively group structure is displayed by the <code>Heatmap</code> function.
At the margin of the heatmap a dendrogram can be shown, if hierarchical cluster algorithms are used [Wilkinson,2009].
Here the dendrogram has to be shown separately and only the heatmap itself is displayed 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heatmap(DataOrDistances,Cls,method='euclidean',

LowLim=0,HiLim,LineWidth=0.5,Clabel="Cluster No.")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Heatmap_+3A_dataordistances">DataOrDistances</code></td>
<td>

<p>if not symmetric, then the function assumes a [1:n,1:d] numeric matrix of n data cases in rows amd d variables in columns. In this case, the distance metric specifed in <code>method</code> will be used.
</p>
<p>Otherwise,
</p>
<p>[1:n,1:n] distance matrix that is symmetric
</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_cls">Cls</code></td>
<td>

<p>[1:n] numerical vector of numbers defining the classification as the main output of the clustering algorithm. It has k unique numbers for k clusters that represent the arbitrary labels of the clustering, assuming a descending order of 1 to k. If not ordered please use <code><a href="FCPS.html#topic+ClusterRenameDescendingSize">ClusterRenameDescendingSize</a></code>. Otherwise x and y label will be incorrect.
</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_method">method</code></td>
<td>
<p>Optional, 
if <code>DataOrDistances</code> is a [1:n,1:d] not symmetric numerical matrix, please 
see <code><a href="parallelDist.html#topic+parDist">parDist</a></code> for accessible distance methods, default is Euclidean
</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_lowlim">LowLim</code></td>
<td>
<p> Optional: limits for the color axis </p>
</td></tr>
<tr><td><code id="Heatmap_+3A_hilim">HiLim</code></td>
<td>
<p> Optional: limits for the color axis </p>
</td></tr>
<tr><td><code id="Heatmap_+3A_linewidth">LineWidth</code></td>
<td>
<p>Width of lines seperating the clusters in the heatmap</p>
</td></tr>
<tr><td><code id="Heatmap_+3A_clabel">Clabel</code></td>
<td>
<p>Default &quot;<code>Cluster No.</code>&quot;, for large number of clusters abbrevations can be used like &quot;<code>Cls No.</code>&quot; or &quot;<code>C</code>&quot; in order to fit as the x and y axis labels</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&quot;Cluster heatmaps are commonly used in biology and related fields to reveal hierarchical clusters in data matrices. Heatmaps visualize a data matrix by drawing a rectangular grid corresponding to rows and columns in the matrix and coloring the cells by their values in the data matrix. In their most basic form, heatmaps have been used for over a century [Wilkinson, 2012]. In addition to coloring cells, cluster heatmaps reorder the rows and/or columns of the matrix based on the results of hierarchical clustering. (...) . Cluster heatmaps have high data density, allowing them to compact large amounts of information into a small space [Weinstein, 2008]&quot;, [Engle, 2017].
</p>
<p>The procedure can be adapted to distance matrices [Thrun, 2018]. Then, the color scale is chosen such that pixels of low distances have blue and teal colors, pixels of middle distances yellow colors, and pixels of high distances have orange and red colors [Thrun, 2018]. The distances are ordered by the clustering and the clusters are divided by black lines. A clustering is valid if the intra-cluster distances are distinctively smaller that inter-cluster distances in the heatmap [Thrun, 2018]. For another example, please see [Thrun, 2018] (Fig. 3.7, p. 31).
</p>


<h3>Value</h3>

<p>object of ggplot2
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Wilkinson,2009] Wilkinson, L., &amp; Friendly, M.: The history of the cluster heat map, The American Statistician, Vol. 63(2), pp. 179-184. 2009.
</p>
<p>[Engle et al., 2017]  Engle, S., Whalen, S., Joshi, A., &amp; Pollard, K. S.: Unboxing cluster heatmaps, BMC bioinformatics, Vol. 18(2), pp. 63. 2017.
</p>
<p>[Weinstein, 2008]  Weinstein, J. N.: A postgenomic visual icon, Science, Vol. 319(5871), pp. 1772-1773. 2008.
</p>
<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, Heidelberg, ISBN: 978-3-658-20539-3, <a href="https://doi.org/10.1007/978-3-658-20540-9">doi:10.1007/978-3-658-20540-9</a>, 2018. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Pixelmatrix">Pixelmatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Cls=Lsun3D$Cls
Data=Lsun3D$Data

#Data
Heatmap(Data,Cls = Cls)

#Distances
Heatmap(as.matrix(dist(Data)),Cls = Cls)


</code></pre>

<hr>
<h2 id='HeatmapColors'>
Default color sequence for plots
</h2><span id='topic+HeatmapColors'></span>

<h3>Description</h3>

<p>Defines the default color sequence for plots made with PixelMatrixPlot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("HeatmapColors")</code></pre>


<h3>Format</h3>

<p>A vector with different strings describing colors for this plot.
</p>

<hr>
<h2 id='InspectBoxplots'>
Inspect Boxplots
</h2><span id='topic+InspectBoxplots'></span>

<h3>Description</h3>

<p>Enables to inspect the boxplots for multiple variables in ggplot2 syntax. Each boxplot also has a point for the mean of the variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectBoxplots(Data, Names,Means=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectBoxplots_+3A_data">Data</code></td>
<td>

<p>Matrix containing the data. Each column is one variable.</p>
</td></tr>
<tr><td><code id="InspectBoxplots_+3A_names">Names</code></td>
<td>

<p>Optional: Names of the variables. If missing the columnnames of data are used.
</p>
</td></tr>
<tr><td><code id="InspectBoxplots_+3A_means">Means</code></td>
<td>

<p>Optional: TRUE: with mean, FALSE: Only median.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The ggplot object of the boxplots
</p>


<h3>Author(s)</h3>

<p>Felix Pape
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- cbind(A = rnorm(200, 1, 3), B = rnorm(100, -2, 5))
InspectBoxplots(x)
</code></pre>

<hr>
<h2 id='InspectCorrelation'>
Inspect the Correlation
</h2><span id='topic+InspectCorrelation'></span>

<h3>Description</h3>

<p>Inspects the correlation between two given features using density scatter plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectCorrelation(X, Y, DensityEstimation = "SDH",

CorMethod = "spearman", na.rm = TRUE,

SampleSize = round(sqrt(5e+08), -3),

NrOfContourLines = 20, Plotter = "native",

DrawTopView = T, xlab, ylab,

main = "Spearman correlation coef.:", xlim, ylim, 

Legendlab_ggplot = "value", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectCorrelation_+3A_x">X</code></td>
<td>
<p>Numeric vector [1:n], first feature (for x axis values)</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_y">Y</code></td>
<td>
<p>Numeric vector [1:n], second feature (for y axis values)</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_densityestimation">DensityEstimation</code></td>
<td>
<p>&quot;SDH&quot; is very fast but maybe not correct, &quot;PDE&quot; is slow but proably more correct.</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_cormethod">CorMethod</code></td>
<td>

<p>method of correlation of the cor function, One of &quot;pearson&quot; (default), &quot;kendall&quot;, or &quot;spearman
</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_samplesize">SampleSize</code></td>
<td>
<p>Numeric, positiv scalar, maximum size of the sample used for calculation. High values increase runtime significantly. The default is that no sample is drawn</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_na.rm">na.rm</code></td>
<td>
<p>Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_nrofcontourlines">NrOfContourLines</code></td>
<td>
<p>Numeric, number of contour lines to be drawn. 20 by default.</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_plotter">Plotter</code></td>
<td>
<p>String, name of the plotting backend to use. Possible values are: &quot;<code>native</code>&quot;, &quot;<code>ggplot</code>&quot;, &quot;<code>plotly</code>&quot;</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_drawtopview">DrawTopView</code></td>
<td>
<p>Boolean, True means contur is drawn, otherwise a 3D plot is drawn. Default: TRUE</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_xlab">xlab</code></td>
<td>
<p>String, title of the x axis. Default: &quot;X&quot;,   see <code>plot()</code> function</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_ylab">ylab</code></td>
<td>
<p>String, title of the y axis. Default: &quot;Y&quot;,   see <code>plot()</code> function</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_main">main</code></td>
<td>

<p>string, the same as &quot;main&quot; in <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_xlim">xlim</code></td>
<td>

<p>see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_ylim">ylim</code></td>
<td>

<p>see <code>plot()</code> function
</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_legendlab_ggplot">Legendlab_ggplot</code></td>
<td>
<p>String, in case of <code>Plotter="ggplot"</code> label for the legend. Default: &quot;value&quot;</p>
</td></tr>
<tr><td><code id="InspectCorrelation_+3A_...">...</code></td>
<td>
<p>Density specifc parameters, for <code>PDEscatter()</code> or SDH (nbins,lambda,Xkernels,Ykernel))</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Example shows that features with high correlation coefficient do not correlate because of bimodality.
</p>


<h3>Value</h3>

<p>plotting handler
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DensityScatter">DensityScatter</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ITS)
data(MTY)
Inds=which(ITS&lt;900&amp;MTY&lt;8000)

InspectCorrelation(ITS[Inds],MTY[Inds])

</code></pre>

<hr>
<h2 id='InspectDistances'>
Inspection of Distance-Distribution
</h2><span id='topic+InspectDistances'></span>

<h3>Description</h3>

<p>Visualizes the distances between objects in the data matrix 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectDistances(DataOrDistances,method= "euclidean",sampleSize = 50000,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectDistances_+3A_dataordistances">DataOrDistances</code></td>
<td>

<p>[1:n,1:d] data cases in rows, variables in columns, if not symmetric
</p>
<p>or
</p>
<p>[1:n,1:n] distance matrix, if symmetric
</p>
</td></tr>
<tr><td><code id="InspectDistances_+3A_method">method</code></td>
<td>
<p>Optional, 
if Data[1:n,1:d]
see <code>parallelDist::parDist</code> for distance method
</p>
</td></tr>
<tr><td><code id="InspectDistances_+3A_samplesize">sampleSize</code></td>
<td>
<p>double value defining the size of the sample for large distance matrizes, see <code>InspectVariable</code>
</p>
</td></tr>
<tr><td><code id="InspectDistances_+3A_...">...</code></td>
<td>
<p>further arguments passed on to <code>InspectVariable</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For an interpretation of the distribution analysis of the distance please read [Thrun, 2018, p. 27, 185].
</p>


<h3>Note</h3>

<p>uses <code>InspectVariable</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun, 2018] Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, ISBN: 978-3-658-20539-3, Heidelberg, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Data=Lsun3D$Data

InspectDistances(as.matrix(dist(Data)))


</code></pre>

<hr>
<h2 id='InspectScatterplots'>
Pairwise scatterplots and optimal histograms
</h2><span id='topic+InspectScatterplots'></span>

<h3>Description</h3>

<p>Pairwise scatterplots and optimal histograms of all features stored as columns of data are plotted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectScatterplots(Data,Names=colnames(Data))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectScatterplots_+3A_data">Data</code></td>
<td>
<p> [1:n,1:d] Data cases in rows (n), variables in columns (d) </p>
</td></tr>
<tr><td><code id="InspectScatterplots_+3A_names">Names</code></td>
<td>

<p>Optional: Names of the variables. If missing the columnnames of data are used.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For two features, <code>PDEscatter</code> function should be used to isnpect modalities [Thrun/Ultsch, 2018]. For many features the function takes too lang. In such a case this function can be used. See [Thrun/Ultsch, 2018] for optimal histogram description.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018]  Thrun, M. C., &amp; Ultsch, A.: Effects of the payout system of income taxes to municipalities in Germany, 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, Vol. accepted, Foundation of the Cracow University of Economics, Zakopane, Poland, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Data=cbind(rnorm(100, mean = 2, sd = 3  ),rnorm(100,mean = 0, sd = 1),rnorm(100,mean = 6, sd = 0.5))
#InspectScatterplots(Data)
</code></pre>

<hr>
<h2 id='InspectStandardization'>
QQplot of Data versus Normalized Data
</h2><span id='topic+InspectStandardization'></span>

<h3>Description</h3>

<p>Allows to inspect if standardization of data makes sense
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectStandardization(Data, TransData, xug = -3, xog = 3, xlab = "Normal", yDataLab =

"Data", yTransDataLab = "Trasformated Data", Symbol4Gerade = "red", main = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectStandardization_+3A_data">Data</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_transdata">TransData</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_xug">xug</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_xog">xog</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_xlab">xlab</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_ydatalab">yDataLab</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_ytransdatalab">yTransDataLab</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_symbol4gerade">Symbol4Gerade</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_main">main</code></td>
<td>

<p>...</p>
</td></tr>
<tr><td><code id="InspectStandardization_+3A_...">...</code></td>
<td>

<p>...
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>...
</p>


<h3>Value</h3>

<p>plot
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Michael, J. R.: The stabilized probability plot, Biometrika, Vol. 70(1), pp. 11-17, 1983.
</p>

<hr>
<h2 id='InspectVariable'>Visualization of Distribution of one variable</h2><span id='topic+InspectVariable'></span>

<h3>Description</h3>

<p>Enables distribution inspection by visualization as described in [Thrun, 2018] and for example used in 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InspectVariable(Feature, Name, i = 1, xlim, ylim,

 sampleSize =1e+05, main)
 </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InspectVariable_+3A_feature">Feature</code></td>
<td>
<p>[1:n] Variable/Vector of Data to be plotted</p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_name">Name</code></td>
<td>
<p>Optional, string, for x label</p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_i">i</code></td>
<td>
<p>Optional, No. of variable/feature, an integer of the for lope</p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_xlim">xlim</code></td>
<td>
<p>[2] Optional, range of x-axis for PDEplot and histogram </p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_ylim">ylim</code></td>
<td>
<p>[2] Optional, range of y-axis, only for PDEplot </p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_samplesize">sampleSize</code></td>
<td>
<p>Optional, default(100000), sample size, if datavector is to big</p>
</td></tr>
<tr><td><code id="InspectVariable_+3A_main">main</code></td>
<td>
<p>string for the title if other than what is desribed in <code>N</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun, 2018] Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, ISBN: 978-3-658-20539-3, Heidelberg, 2018.
</p>
<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("ITS")
InspectVariable(ITS,Name='Income in EUR',main='ITS')

</code></pre>

<hr>
<h2 id='ITS'>
Income Tax Share
</h2><span id='topic+ITS'></span>

<h3>Description</h3>

<p>Numerical vector of length 11194. details in [Ultsch/Behnisch, 2017; Thrun/Ultsch, 2018].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ITS")</code></pre>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Ultsch/Behnisch, 2017]	 Ultsch, A., Behnisch, M.: Effects of the payout system of income taxes to municipalities in Germany, Applied Geography, Vol. 81, pp. 21-31, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ITS)
str(ITS)
</code></pre>

<hr>
<h2 id='JitterUniqueValues'>
Jitters Unique Values
</h2><span id='topic+JitterUniqueValues'></span>

<h3>Description</h3>

<p>Jitters Unique Values for Visualizations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JitterUniqueValues(Data, Npoints = 20,

min = 0.99999, max = 1.00001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JitterUniqueValues_+3A_data">Data</code></td>
<td>

<p>[1:n] vector of data
</p>
</td></tr>
<tr><td><code id="JitterUniqueValues_+3A_npoints">Npoints</code></td>
<td>

<p>number of jittered points generated from the m unique values of the datavector Data
</p>
</td></tr>
<tr><td><code id="JitterUniqueValues_+3A_min">min</code></td>
<td>

<p>minimum value of jittering
</p>
</td></tr>
<tr><td><code id="JitterUniqueValues_+3A_max">max</code></td>
<td>

<p>maximum value of jittering
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>min and max are either multiplied or added to data depending on the range of values. If Npoints==2, then only two  values per unique of Data is jittered otherwise additional values are generated.Npoints==1 does not jitter the values but gives the unique values back.
</p>


<h3>Value</h3>

<p>vector of DataJitter[1:(m+Npoints-1)] jittered values
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>See Also</h3>

<p>used for example in <code><a href="#topic+MDplot">MDplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data=c(rep(1,10),rep(0,10),rep(100,10))

JitterUniqueValues(data,Npoints=1)

JitterUniqueValues(data,Npoints=2)

DataJitter=JitterUniqueValues(data,Npoints=20)
</code></pre>

<hr>
<h2 id='Lsun3D'>
Lsun3D inspired by FCPS [Thrun/Ultsch, 2020] introduced in [Thrun, 2018]
</h2><span id='topic+Lsun3D'></span>

<h3>Description</h3>

<p>Clearly defined clusters, different variances.  Detailed description of dataset and its clustering challenge is provided in [Thrun/Ultsch, 2020].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Lsun3D")</code></pre>


<h3>Details</h3>

<p>Size 404, Dimensions 3
</p>
<p>Dataset defines discontinuites, where the clusters have different variances.
Three main clusters, and four outliers (in cluster 4). For a more detailed description see [Thrun, 2018].
</p>


<h3>References</h3>

<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, Heidelberg, ISBN: 978-3-658-20539-3, <a href="https://doi.org/10.1007/978-3-658-20540-9">doi:10.1007/978-3-658-20540-9</a>, 2018. 
</p>
<p>[Thrun/Ultsch, 2020]  Thrun, M. C., &amp; Ultsch, A.: Clustering Benchmark Datasets Exploiting the Fundamental Clustering Problems, Data in Brief, Vol. 30(C), pp. 105501, <a href="https://doi.org/10.1016/j.dib.2020.105501">doi:10.1016/j.dib.2020.105501</a>, 2020.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lsun3D)
str(Lsun3D)
Cls=Lsun3D$Cls
Data=Lsun3D$Data
</code></pre>

<hr>
<h2 id='MAplot'>
Minus versus Add plot
</h2><span id='topic+MAplot'></span>

<h3>Description</h3>

<p>Bland-Altman plot [Altman/Bland, 1983].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAplot(X,Y,islog=TRUE,LoA=FALSE,CI=FALSE,

densityplot=FALSE,main,xlab,ylab,

Cls,lwd=2,ylim=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAplot_+3A_x">X</code></td>
<td>
<p>[1:n] numerical vector of a feature/variable
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_y">Y</code></td>
<td>
<p>[1:n] another numerical vector of a feature/variable
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_islog">islog</code></td>
<td>
<p>Optional, TRUE: MAplot, FALSE: M=x-y versus a=0.5(x+y)
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_loa">LoA</code></td>
<td>
<p>Optional, if TRUE: limits of agreement are plottet as lines if densityplot=FALSE
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_ci">CI</code></td>
<td>
<p>Optional, if TRUE: confidence intervals for LoA, see [Stockl et al., 2004], if  densityplot=FALSE
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_densityplot">densityplot</code></td>
<td>
<p>Optional, FALSE: Scatterplot using <code> <a href="#topic+Classplot">Classplot</a></code>, TRUE: density scatter plot with <code><a href="#topic+DensityScatter">DensityScatter</a></code>
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_main">main</code></td>
<td>
<p>Optional, see <code>plot</code>
</p>
</td></tr>  
<tr><td><code id="MAplot_+3A_xlab">xlab</code></td>
<td>
<p>Optional, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_ylab">ylab</code></td>
<td>

<p>Optional, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_cls">Cls</code></td>
<td>

<p>Optional, prior Classification as a numeric vector.
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_lwd">lwd</code></td>
<td>

<p>Optional, if <code>LoA=TRUE</code> or <code>CI=TRUE</code> the width of the lines, otherwise input argument is ignored
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_ylim">ylim</code></td>
<td>

<p>Optional, default <code>=NULL</code> sets this parameter automatically, otherwise see <code>Classplot</code>.
</p>
</td></tr>
<tr><td><code id="MAplot_+3A_...">...</code></td>
<td>

<p>for example, <code>ylim</code>, Please see either<code> <a href="#topic+Classplot">Classplot</a></code> in the mode <code>Plotter="native"</code>, or <code><a href="#topic+DensityScatter">DensityScatter</a></code> for further arguments depending on <code>densityplot</code>, see also details
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Bland-Altman plot [Altman/Bland, 1983] for visual representation of genomic data or in order to decorrelate data.
</p>
<p>&quot;The limits of agreement (LoA) are defined as the mean difference +- 1.96 SD of differences. If these limits do not exceed the maximum allowed difference between methods (the differences within mean +- 1.96 SD are not clinically important), the two methods are considered to be in agreement and may be used interchangeably.&quot; cited as in URL. Please note, that the underyling assumption is the normal distribution of the differences. Input argument <code>LoA=TRUE</code> shows the mean of the difference in blue and +- 1.96 SD in green. Input argument <code>CI=TRUE</code> shows the mean of the difference in blue and the confidence intervall as red dashed lines similar to the cited URL.
</p>
<p>In case of <code>densityplot=FALSE</code>, the function <code> <a href="#topic+Classplot">Classplot</a></code> is always called with <code>Plotter="native"</code>. Then, the input argument &quot;<code>Colors</code>&quot;&quot; of points can only be set in <code> <a href="#topic+Classplot">Classplot</a></code> if &quot;<code>Cls</code>&quot;&quot; is given in this function, otherwise the points are always black. The input argument &quot;<code>Size</code>&quot;&quot; sets the size of points in <code><a href="#topic+Classplot">Classplot</a></code>. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>MA</code></td>
<td>
<p>[1:n,2] Matrix of Minus component of two features and Add component of two features
</p>
</td></tr>
<tr><td><code>Handle</code></td>
<td>
<p>see <code><a href="#topic+DensityScatter">DensityScatter</a></code> for output options, if densityplot=TRUE, otherwise NULL
</p>
</td></tr>
<tr><td><code>Statistics</code></td>
<td>
<p>Named list of four element, each consisting of one value depending on input parameters <code>LoA</code> and <code>CI</code>, of this function. If not specifically set each list element is <code>NULL</code>. The elements are  <code>Mean_value</code> - mean allowed difference,  <code>SD_value</code> - standard deviation of difference,  <code>LoA_value</code> - Limits of agreement=1.96*SD,  <code>CI_value</code> - confidence intervall, i.e., maximum allowed difference
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Altman/Bland, 1983]  Altman D.G., Bland J.M.: Measurement in medicine: the analysis of method comparison studies, The Statistician, Vol. 32, p. 307-317, doi:10.2307/2987937, 1983.
</p>
<p><a href="https://www.medcalc.org/manual/bland-altman-plot.php">https://www.medcalc.org/manual/bland-altman-plot.php</a>
</p>
<p>[Stockl et al., 2004]  Stockl, D., Rodriguez Cabaleiro, D., Van Uytfanghe, K., &amp; Thienpont, L. M.: Interpreting method comparison studies by use of the Bland-Altman plot: reflecting the importance of sample size by incorporating confidence limits and predefined error limits in the graphic, Clinical chemistry, Vol. 50(11), pp. 2216-2218. 2004.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ITS")
data("MTY")
MAlist=MAplot(ITS,MTY)
</code></pre>

<hr>
<h2 id='MDplot'>
Mirrored Density plot (MD-plot)
</h2><span id='topic+MDplot'></span>

<h3>Description</h3>

<p>This function creates a MD-plot for each variable of the data matrix. 
The MD-plot is a visualization for a boxplot-like shape of the PDF published in [Thrun et al., 2020] with the default ordering by shape. It is an improvement of violin or so-called bean plots and posses advantages in comparison to the conventional well-known box plot [Thrun et al., 2020]. 
</p>
<p>A complete guide about the MDplot can be found in <a href="https://md-plot.readthedocs.io/en/latest/index.html">https://md-plot.readthedocs.io/en/latest/index.html</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MDplot(Data, Names, Ordering='Default', Scaling="None",

Fill='darkblue', RobustGaussian=TRUE, GaussianColor='magenta',

Gaussian_lwd=1.5, BoxPlot=FALSE,BoxColor='darkred', 

MDscaling='width', LineColor='black', LineSize=0.01,

QuantityThreshold=50, UniqueValuesThreshold=12,

SampleSize=5e+05,SizeOfJitteredPoints=1,OnlyPlotOutput=TRUE,

main="MD-plot",ylab="Range of values in which PDE is estimated",

BW=FALSE,ForceNames=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MDplot_+3A_data">Data</code></td>
<td>

<p>[1:n,1:d] Numerical Matrix containing the n cases of d variables. Each column is one variable. A data.frame is automatically transformed to a numerical matrix.</p>
</td></tr>
<tr><td><code id="MDplot_+3A_names">Names</code></td>
<td>

<p>Optional: [1:d] Names of the variables. If missing, the columnnames of data are used. If not missing, than the names can be cleaned or not (see <code>ForceNames</code>).
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_ordering">Ordering</code></td>
<td>

<p>Optional: string, either <code>Default</code>, <code>Columnwise</code> or <code>AsIs</code>, <code>Alphabetical</code>, <code>Average</code>, <code>Bimodal</code>, <code>Variance</code>  or <code>Statistics</code>. Please see details for explanation.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_scaling">Scaling</code></td>
<td>

<p>Optional, Default is <code>None</code>, <code>Percentalize</code>, <code>CompleteRobust</code>, <code>Robust</code> or <code>Log</code>, Please see details for explanation.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_fill">Fill</code></td>
<td>

<p>Optional: String or Vector, which gives the color(s) with which MDs are to be filled with.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_robustgaussian">RobustGaussian</code></td>
<td>

<p>Optional: If TRUE: each MDplot of a variable is overlayed with a roubustly estimated unimodal Gaussian distribution in the range of this variable,
if statistical testing does not yield a significant p.value. In this case the packages <span class="pkg">moments</span>, <span class="pkg">diptest</span> and <span class="pkg">signal</span> are required.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_gaussiancolor">GaussianColor</code></td>
<td>

<p>Optional: string, color of robustly estimated gaussian, only for <code>RobustGaussian=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_gaussian_lwd">Gaussian_lwd</code></td>
<td>

<p>Optional: numerical, line width of robustly estimated gaussian, only for <code>RobustGaussian=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_boxplot">BoxPlot</code></td>
<td>

<p>Optional: If TRUE: each MDplot is overlayed with a Box-Whisker Diagram.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_boxcolor">BoxColor</code></td>
<td>

<p>Optional: string, color of Boxplot, only for <code>BoxPlot=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_mdscaling">MDscaling</code></td>
<td>

<p>Optional: if &quot;area&quot;, all violins have the same area (before trimming the tails). If &quot;count&quot;, areas are scaled proportionally to  the number of observations. If &quot;width&quot; (default), all MDs have the same maximum width.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_linecolor">LineColor</code></td>
<td>

<p>Optional: string, color of line around the mirrored densities. <code>NA</code> disables this features which is usefull if ones wants to avoid vertical lines leading to outliers.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_linesize">LineSize</code></td>
<td>

<p>Optional: numerical, linewidth of line around the mirrored densities.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_quantitythreshold">QuantityThreshold</code></td>
<td>

<p>Optional: numeric value defining the threshold of the minimal amount of values in data. Below this threshold no density estimation is performed and a 1D scatter plot with jittered points is drawn. Only Data Science experts should change this value after they understand  how the density is estimated (see [Ultsch, 2005]).
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_uniquevaluesthreshold">UniqueValuesThreshold</code></td>
<td>

<p>Optional: numeric value defining the threshold of the minimal amount of unique values in data. Below this threshold no density estimation and statistical testing is performed and a 1D scatter plot with jittered points drawn. Only Data Science experts should change this value after they understand how the density is estimated (see [Ultsch, 2005]).
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_samplesize">SampleSize</code></td>
<td>

<p>Optional: numeric value defining a threshold. Above this threshold uniform sampling of finite cases is performed in order to shorten computation time.If <span class="pkg">rowr</span> is not installed, uniform sampling of all cases
is performed. If required, <code>SampleSize=n</code> can be set to omit this procedure. 
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_sizeofjitteredpoints">SizeOfJitteredPoints</code></td>
<td>

<p>Optional: scalar. If not enough unique values for density estimation are given, data points are jittered. This parameter defines the size of the points. 
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_onlyplotoutput">OnlyPlotOutput</code></td>
<td>

<p>Optional: Default TRUE only a ggplot object is given back, if FALSE: Additinally, scaled data and ordering are the output of this function in a <code>list</code>.
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_main">main</code></td>
<td>

<p>string defining the (centered) title of the plot
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_ylab">ylab</code></td>
<td>

<p>string defining the y label, PDE= pareto density estimation (see [Ultsch, 2005])
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_bw">BW</code></td>
<td>

<p>FALSE: usual ggplot2 background and style which is good for screen visualizations
</p>
<p>TRUE: theme_bw() is used which is more appropriate for publications
</p>
</td></tr>
<tr><td><code id="MDplot_+3A_forcenames">ForceNames</code></td>
<td>

<p>FALSE: Per Default column names are cleaned for propper plotting
</p>
<p>TRUE: forces to set the column names as given. Beware, this can result in plotting errors.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In short, the MD-plot can be described as a PDE optimized violin plot. The Pareto Density Estimation (PDE) is an approach to estimate the probability density function (pdf) [Ultsch, 2005].
</p>
<p>The MD-plot is in the process of beeing peer-reviewed [Thrun/Ultsch, 2019]. 
</p>
<p>Statistical testing is performed with <code><a href="diptest.html#topic+dip.test">dip.test</a></code> and <code><a href="moments.html#topic+agostino.test">agostino.test</a></code>.
</p>
<p>For the paramter <code>Ordering</code> the following options are possible:
</p>

<dl>
<dt><code>Default</code></dt><dd><p>Ordering of plots by convex/concav/unimodal/nonunimodal shapes using statistical criteria. In this case the <span class="pkg">signal</span> is required.</p>
</dd>
<dt><code>Columnwise</code></dt><dd><p>Ordering of plots by the order of columns of <code>Data</code>.</p>
</dd>
<dt><code>AsIs</code></dt><dd><p>Synonym of Columnwise: Ordering of plots by the order of columns of <code>Data</code>.</p>
</dd>
<dt><code>Alphabetical</code></dt><dd><p>Ordering of plots by the order of columns of <code>Data</code> sorted in alphabetical order by column names.</p>
</dd>
<dt><code>Average</code></dt><dd><p>Ordering of plots by the order of columns of <code>Data</code> sorted in order of increasing column-wise average</p>
</dd>
<dt><code>Bimodal</code></dt><dd><p>Ordering of plots by the order of columns of <code>Data</code> sorted in order of decreasing bimodality amplitude[Zhang et al., 2003]</p>
</dd>
<dt><code>Variance</code></dt><dd><p>Ordering of plots by the order of columns of <code>Data</code> sorted in order of increasing inter-quartile range</p>
</dd>
<dt><code>Statistics</code></dt><dd><p>Ordering of plots depending on the logarithm of the p-vlaues of statistical testing. In this case the packages <span class="pkg">moments</span>, <span class="pkg">diptest</span> and <span class="pkg">signal</span> are required.</p>
</dd>
</dl>

<p>For the paramter <code>Scaling</code> the following options are possible:
</p>

<dl>
<dt><code>None</code></dt><dd><p>No Scaling of data is done.</p>
</dd>
<dt><code>Percentalize</code></dt><dd><p>Data is scaled between zero and 100.</p>
</dd>
<dt><code>CompleteRobust</code></dt><dd><p>Data is first robustly scaled between zero and 1, then centered to zero and outliers are capped by a robustly formula described in <code><a href="#topic+RobustNormalization">RobustNormalization</a></code>. </p>
</dd>
<dt><code>Robust</code></dt><dd><p>Data is robustly scaled between zero and 1 by a formula described in the <code><a href="#topic+RobustNormalization">RobustNormalization</a></code>. </p>
</dd>
<dt><code>Log</code></dt><dd><p>Data is transformed with a sgined log allowing for negative values to be transformed with a logarithm of base 10, please see <code><a href="#topic+SignedLog">SignedLog</a></code> for details.</p>
</dd>
</dl>



<h3>Value</h3>

<p>In the default case of <code>OnlyPlotOutput==TRUE</code>: The ggplot object of the MD-plot. 
</p>
<p>Otherwise for <code>OnlyPlotOutput==FALSE</code>: A list of
</p>
<table>
<tr><td><code>ggplotObj</code></td>
<td>

<p>The ggplot object of the MD-plot.
</p>
</td></tr>
<tr><td><code>Ordering</code></td>
<td>

<p>The ordering of columns of data defined by  <code>Ordering</code>.
</p>
</td></tr>
<tr><td><code>DataOrdered</code></td>
<td>

<p>[1:n,1:d] matrix of ordered and scaled data defined by <code>Ordering</code> and <code>Scaling</code>.
</p>
</td></tr>
</table>
<p>Note that the package <span class="pkg">ggExtra</span> is not necessarily required but if given the feature names are automatically rotated. 
</p>


<h3>Note</h3>

<p>1.) One would assume that in the first of the two following cases ggplot2 only adjusts the plotting region but: 
</p>
<p><code>MDplot(MTY)+ylim(c(0,7000))</code> is equal to <code>MDplot(MTY[MTY&lt;7000])</code>. 
</p>
<p>This means in both cases the data is clipped and AFTERWARDS the density estimation is performed. 
</p>
<p>2.) Because of a (sometimes) strange behavior of either ggplot2 or reshape2, numerical column names are changed to character by adding 'C_' which can disabled using <code>ForceNames=TRUE</code>.
</p>
<p>3.) Columnnames will be automatically deblanked and cleaned. To force specific columnnames the input <code>Names</code> can be used in combination with <code>ForceNames=TRUE</code>. However, this can result in plotting errors or other strange behavior.
</p>
<p>4.) Overlaying MD-plots with robustly estimated gaussians seldomly will yield magenta  (or other <code>GaussianColor</code>) lines overlaying more than the violin plot they should overlay, because the width of the two plots is not the same (but I am unable to set it strictly in ggplot). In such a case just call the function again.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun, Felix Pape contributed with the idea to use ggplot2 as the basic framework.
</p>


<h3>References</h3>

<p>[Thrun et al., 2020]	Thrun, M. C., Gehlert, T. &amp; Ultsch, A.: Analyzing the Fine Structure of Distributions, PLoS ONE, Vol. 15(10), pp. 1-66, DOI 10.1371/journal.pone.0238835, 2020. 
</p>
<p>[Ultsch, 2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in  Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>
<p>[Zhang et al., 2003] Zhang, C., Mapes, B., &amp; Soden, B.: Bimodality in tropical water vapour, Quarterly Journalof the Royal Meteorological Society, 129(594), 2847-2866, 2003.
</p>


<h3>See Also</h3>

<p><a href="https://md-plot.readthedocs.io/en/latest/index.html">https://md-plot.readthedocs.io/en/latest/index.html</a>
</p>
<p><code><a href="#topic+ClassMDplot">ClassMDplot</a></code>
</p>
<p><a href="https://pypi.org/project/md-plot/">https://pypi.org/project/md-plot/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

x = cbind(
    A = runif(2000, 1, 5),
    B = c(rnorm(1000, 0, 1), rnorm(1000, 2.6, 1)),
    C = c(rnorm(2000, 2.5, 1)),
    D = rpois(2000, 5)
  )
MDplot(x)


</code></pre>

<hr>
<h2 id='MDplot4multiplevectors'>
Mirrored Density plot (MD-plot)for Multiple Vectors
</h2><span id='topic+MDplot4multiplevectors'></span>

<h3>Description</h3>

<p>This function creates a MD-plot for multiple numerical vectors of various lenghts.
The MD-plot is a visualization for a boxplot-like Shape of the PDF published in [Thrun et al., 2020].
It is an improvement of violin or so-called bean plots and posses advantages in comparison to the conventional well-known box plot [Thrun et al., 2020].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MDplot4multiplevectors(..., Names, Ordering = 'Columnwise',
Scaling = "None", Fill = 'darkblue', RobustGaussian = TRUE,

GaussianColor = 'magenta', Gaussian_lwd = 1.5, BoxPlot = FALSE,

BoxColor = 'darkred', MDscaling = 'width', LineSize = 0.01,

LineColor = 'black', QuantityThreshold = 40, UniqueValuesThreshold = 12,

SampleSize = 5e+05, SizeOfJitteredPoints = 1, OnlyPlotOutput = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MDplot4multiplevectors_+3A_...">...</code></td>
<td>
<p>Either d numerical vectors of different lengths or a list of length d where each element of the list is an vector of arbitrary length</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_names">Names</code></td>
<td>

<p>Optional: [1:d] Names of the variables. If missing, the columnnames of data are used.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_ordering">Ordering</code></td>
<td>

<p>Optional: string, either <code>Default</code>, <code>Columnwise</code>, <code>Alphabetical</code> or <code>Statistics</code>. Please see details for explanation.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_scaling">Scaling</code></td>
<td>

<p>Optional, Default is <code>None</code>, <code>Percentalize</code>, <code>CompleteRobust</code>, <code>Robust</code> or <code>Log</code>, Please see details for explanation.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_fill">Fill</code></td>
<td>

<p>Optional: string, color with which MDs are to be filled with.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_robustgaussian">RobustGaussian</code></td>
<td>

<p>Optional: If TRUE: each MDplot of a variable is overlayed with a roubustly estimated unimodal Gaussian distribution in the range of this variable,
if statistical testing does not yield a significant p.value. In this case the packages <span class="pkg">moments</span>, <span class="pkg">diptest</span> and <span class="pkg">signal</span> are required.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_gaussiancolor">GaussianColor</code></td>
<td>

<p>Optional: string, color of robustly estimated gaussian, only for <code>RobustGaussian=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_gaussian_lwd">Gaussian_lwd</code></td>
<td>

<p>Optional: numerical, line width of robustly estimated gaussian, only for <code>RobustGaussian=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_boxplot">BoxPlot</code></td>
<td>

<p>Optional: If TRUE: each MDplot is overlayed with a Box-Whisker Diagram.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_boxcolor">BoxColor</code></td>
<td>

<p>Optional: string, color of Boxplot, only for <code>BoxPlot=TRUE</code>.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_mdscaling">MDscaling</code></td>
<td>

<p>Optional: if &quot;area&quot;, all violins have the same area (before trimming the tails). If &quot;count&quot;, areas are scaled proportionally to  the number of observations. If &quot;width&quot; (default), all MDs have the same maximum width.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_linesize">LineSize</code></td>
<td>

<p>Optional: numerical, linewidth of line around the mirrored densities.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_linecolor">LineColor</code></td>
<td>

<p>Optional: string, color of line around the mirrored densities. <code>NA</code> disables this features which is usefull if ones wants to avoid vertical lines leading to outliers.
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_quantitythreshold">QuantityThreshold</code></td>
<td>

<p>Optional: numeric value defining a threshold. Below this threshold no density estimation is performed and a jitter plot with a median line is drawn. Only Data Science experts should change this value after they understand     how the density is estimated (see [Ultsch, 2005]).
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_uniquevaluesthreshold">UniqueValuesThreshold</code></td>
<td>

<p>Optional: numeric value defining a threshold. Below this threshold no density estimation and statistical testing is performed and a Jitter plot is drawn. Only Data Science experts should change this value after they understand how the density is estimated (see [Ultsch, 2005]).
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_samplesize">SampleSize</code></td>
<td>

<p>Optional: numeric value defining a threshold. Above this threshold uniform sampling of finite cases is performed in order to shorten computation time.If <span class="pkg">rowr</span> is not installed, uniform sampling of all cases
is performed. If required, <code>SampleSize=n</code> can be set to omit this procedure. 
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_sizeofjitteredpoints">SizeOfJitteredPoints</code></td>
<td>

<p>Optional: scalar. If Not enough unique values for density estimation are given, data points are jittered. This parameter defines the size of the points. 
</p>
</td></tr>
<tr><td><code id="MDplot4multiplevectors_+3A_onlyplotoutput">OnlyPlotOutput</code></td>
<td>

<p>Optional: Default TRUE only a ggplot object is given back, if FALSE: Additinally Scaled Data and ordering are the output of this function in a <code>list</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see <code><a href="#topic+MDplot">MDplot</a></code> for details.
</p>


<h3>Value</h3>

<p>In the default case of <code>OnlyPlotOutput==TRUE</code>: The ggplot object of the MD-plot. 
</p>
<p>Otherwise for <code>OnlyPlotOutput==FALSE</code>: A list of
</p>
<table>
<tr><td><code>ggplotObj</code></td>
<td>

<p>The ggplot object of the MD-plot.
</p>
</td></tr>
<tr><td><code>Ordering</code></td>
<td>

<p>The ordering of columns of data defined by  <code>Ordering</code>.
</p>
</td></tr>
<tr><td><code>DataOrdered</code></td>
<td>

<p>[1:n,1:d] matrix of ordered and scaled data defined by <code>Ordering</code> and <code>Scaling</code>.
</p>
</td></tr>
</table>
<p>Note that the package <span class="pkg">ggExtra</span> is not necessarily required but if given the feauture names are automatically rotated. 
</p>


<h3>Note</h3>

<p>cbind.fill is internally used from the depricated R package rowr of Craig Varrichio.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun.
</p>


<h3>References</h3>

<p>[Ultsch, 2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in  Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>
<p>[Thrun et al., 2020]	Thrun, M. C., Gehlert, T. &amp; Ultsch, A.: Analyzing the Fine Structure of Distributions, PLoS ONE, Vol. 15(10), pp. 1-66, DOI 10.1371/journal.pone.0238835, 2020. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ClassMDplot">ClassMDplot</a></code>
<code><a href="#topic+MDplot">MDplot</a></code>
<a href="https://pypi.org/project/md-plot/">https://pypi.org/project/md-plot/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


MDplot4multiplevectors(runif(20000, 1, 5),c(rnorm(20000,0,1),

rnorm(20000,2.6,1)),c(rnorm(2000,2.5,1)),rpois(25000,5),

Names=c('A','B','C','D'))

V=list(runif(20000, 1, 5),c(rnorm(20000,0,1),

rnorm(20000,2.6,1)),c(rnorm(2000,2.5,1)),rpois(25000,5))


MDplot4multiplevectors(V,Names=c('A','B','C','D'))


</code></pre>

<hr>
<h2 id='MTY'>
Muncipal Income Tax Yield
</h2><span id='topic+MTY'></span>

<h3>Description</h3>

<p>Numerical vector of length 11194. details in [Ultsch/Behnisch, 2017; Thrun/Ultsch, 2018].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("MTY")</code></pre>


<h3>References</h3>

<p>[Thrun/Ultsch, 2018] Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Ultsch/Behnisch, 2017]	 Ultsch, A., Behnisch, M.: Effects of the payout system of income taxes to municipalities in Germany, Applied Geography, Vol. 81, pp. 21-31, 2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MTY)
str(MTY)
</code></pre>

<hr>
<h2 id='Multiplot'>Plot multiple ggplots objects in one panel</h2><span id='topic+Multiplot'></span>

<h3>Description</h3>

<p>ggplot objects can be passed in ..., or to plotlist (as a list of ggplot objects)  <br />
For example, if the layout is specified as the matrix(c(1,2,3,3), nrow=2, byrow=TRUE), then plot 1 will go in the upper left, 2 will go in the upper right, and 3 will go all the way across the bottom.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Multiplot(..., Plotlist=NULL, ColNo=1, LayoutMat)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Multiplot_+3A_...">...</code></td>
<td>
<p> multiple ggplot objects to be plotted</p>
</td></tr>
<tr><td><code id="Multiplot_+3A_plotlist">Plotlist</code></td>
<td>
<p>Optional: list filled with ggplot objects to be plotted</p>
</td></tr>
<tr><td><code id="Multiplot_+3A_colno">ColNo</code></td>
<td>
<p>Number of columns in layout</p>
</td></tr>
<tr><td><code id="Multiplot_+3A_layoutmat">LayoutMat</code></td>
<td>
<p>A matrix specifying the layout. If present, 'ColNo' is ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with
Plotlist
</p>


<h3>Author(s)</h3>

<p>Winston Chang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lsun3D)
Data=Lsun3D$Data
Cls=Lsun3D$Cls
obj1=Classplot(Data[,1],Data[,2],Cls=Cls,Plotter="ggplot",Size=3,main="Top plot")
obj2=Classplot(Data[,2],Data[,3],Cls=Cls,Plotter="ggplot",Size=3,main="Middle plot")
obj3=Classplot(Data[,1],Data[,3],Cls=Cls,Plotter="ggplot",Size=3,main="Bottom plot")
V=Multiplot(obj1,obj2,obj3)
</code></pre>

<hr>
<h2 id='OptimalNoBins'>Optimal Number Of Bins</h2><span id='topic+OptimalNoBins'></span>

<h3>Description</h3>

<p>Optimal Number Of Bins is a kernel density estimation for fixed intervals.
</p>
<p>Calculation of the optimal number of bins for a histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OptimalNoBins(Data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OptimalNoBins_+3A_data">Data</code></td>
<td>
<p>Data</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The bin width ist defined with bw=3.49*stdrobust(1/(n)^1/3)
</p>


<h3>Value</h3>

<p>optNrOfBins The best possible number of bins. Not less than 10 though
</p>


<h3>Note</h3>

<p>This the second version of the function prior available in <span class="pkg">AdaptGauss</span>
</p>


<h3>Author(s)</h3>

<p>Alfred Ultsch, Michael Thrun
</p>


<h3>References</h3>

<p>David W. Scott Jerome P. Keating: A Primer on Density Estimation for the Great Home Run Race of 98, STATS 25, 1999, pp 16-22.
</p>
<p><strong>See Also</strong>
</p>
<p>ParetoRadius
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Data = c(rnorm(1000),rnorm(2000)+2,rnorm(1000)*2-1)

optNrOfBins = OptimalNoBins(Data)

minData = min(Data,na.rm = TRUE)

maxData = max(Data,na.rm = TRUE)

i = maxData-minData

optBreaks = seq(minData, maxData, i/optNrOfBins) # bins in fixed intervals 

hist(Data, breaks=optBreaks)

</code></pre>

<hr>
<h2 id='ParetoDensityEstimation'>Pareto Density Estimation V3</h2><span id='topic+ParetoDensityEstimation'></span>

<h3>Description</h3>

<p>This function estimates the Pareto Density for the distribution of one variable. In the default setting the functions estimates internally the appropriate number and position of kernels to estimate the density properly. However, the user can set the <code>kernels</code> manually. In this case density will only be estimated only around these values even if data exists outside the range of kernels or the internally estimated <code>paretoRadius</code> does not contain all datapoints between each kernel. See example for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParetoDensityEstimation(Data, paretoRadius, kernels = NULL,
  MinAnzKernels = 100,PlotIt=FALSE,Silent=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParetoDensityEstimation_+3A_data">Data</code></td>
<td>
<p>[1:n] numeric vector of data.</p>
</td></tr>
<tr><td><code id="ParetoDensityEstimation_+3A_paretoradius">paretoRadius</code></td>
<td>
<p>Optional scalar, numeric value, see <code><a href="#topic+ParetoRadius">ParetoRadius</a></code>.If not given it is estimated internally. Please do not set manually</p>
</td></tr>
<tr><td><code id="ParetoDensityEstimation_+3A_kernels">kernels</code></td>
<td>
<p>Optional,[1:m] numeric vector data values where pareto density is measured at. If 0 (by default) kernels will be computed.</p>
</td></tr>
<tr><td><code id="ParetoDensityEstimation_+3A_minanzkernels">MinAnzKernels</code></td>
<td>
<p>Optional, minimal number of kernels, default MinAnzKernels==100</p>
</td></tr>
<tr><td><code id="ParetoDensityEstimation_+3A_plotit">PlotIt</code></td>
<td>
<p>Optional, if TRUE: raw basic r plot of density estimation of debugging purposes. Usually please use <span class="pkg">ggplot2</span> interface via <code><a href="#topic+PDEplot">PDEplot</a></code> or <code><a href="#topic+MDplot">MDplot</a></code></p>
</td></tr>
<tr><td><code id="ParetoDensityEstimation_+3A_silent">Silent</code></td>
<td>
<p>Optional, if TRUE: disables all warnings</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pareto Density Estimation (PDE) is a method for the estimation of probability density functions using hyperspheres. The Pareto-radius of the hyperspheres is derived from the optimization of information for minimal set size. It is shown, that Pareto Density is the best estimate for clusters of Gaussian structure. The method is shown to be robust when cluster overlap and when the variances differ across clusters.
This is the best density estimation to judge Gaussian Mixtures of the data see [Ultsch 2003].
</p>
<p>If input argument <code>kernels</code> is set manually the output arguments <code>paretoDensity_internal</code> and <code>kernels_internal</code> provide the internally estimated density and kernels. Otherwise these arguments are NULL. The function provides a message if range of kernels and range of data does not overlap completly.
</p>
<p>Typically it is not advisable to set <code>paretoRadius</code> manually. However in specific cases, the function <code><a href="#topic+ParetoRadius">ParetoRadius</a></code> is used prior to calling this function. In such cases the input argument can use a priorly estimated <code>paretoRadius</code>.
</p>


<h3>Value</h3>

<p>List With 
</p>

<dl>
<dt>kernels</dt><dd><p>[1:m] numeric vector. data values at with Pareto Density is measured.</p>
</dd>
<dt>paretoDensity</dt><dd><p>[1:m] numeric vector containing the determined density by <code>paretoRadius</code>.</p>
</dd>
<dt>paretoRadius</dt><dd><p>numeric value of defining the radius</p>
</dd>
<dt>kernels_internal</dt><dd><p>Either NULL or internally estimated [1:p] numeric vector of kernels if input argument <code>kernels</code> was set by the user</p>
</dd>
<dt>paretoDensity_internal</dt><dd><p>Either NULL or internally estimated density if input argument <code>kernels</code> was set by the user</p>
</dd>
</dl>



<h3>Note</h3>

<p>This the second version of the function prior available in <span class="pkg">AdaptGauss</span>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ParetoRadius">ParetoRadius</a></code>
</p>
<p><code><a href="#topic+PDEplot">PDEplot</a></code>
</p>
<p><code><a href="#topic+MDplot">MDplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   
   #kernels are estimated internally
   data = c(rnorm(1000),rnorm(2000)+2,rnorm(1000)*2-1)
   pdeVal        &lt;- ParetoDensityEstimation(data)
   plot(pdeVal$kernels,pdeVal$paretoDensity,type='l',xaxs='i',
   yaxs='i',xlab='Data',ylab='PDE')
   
   ##data exist outside of the range kernels
   kernels=seq(from=-3,to=3,by=0.01) 
   pdeVal        &lt;- ParetoDensityEstimation(data,  kernels=kernels)
   plot(pdeVal$kernels,pdeVal$paretoDensity,type='l',xaxs='i',
   yaxs='i',xlab='Data',ylab='PDE')
   
   #data exists in-between kernels that is not measured
   pdeVal$paretoRadius#0.42
   kernels=seq(from=-8,to=8,by=1)
   pdeVal        &lt;- ParetoDensityEstimation(data,  kernels=kernels)
   plot(pdeVal$kernels,pdeVal$paretoDensity,type='l',xaxs='i',
   yaxs='i',xlab='Data',ylab='PDE')
   
   
</code></pre>

<hr>
<h2 id='ParetoRadius'>ParetoRadius for distributions</h2><span id='topic+ParetoRadius'></span>

<h3>Description</h3>

<p>Calculation of the ParetoRadius i.e. the 18 percentiles of all mutual Euclidian distances in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParetoRadius(Data, maximumNrSamples = 10000,
  plotDistancePercentiles = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParetoRadius_+3A_data">Data</code></td>
<td>
<p>numeric data vector</p>
</td></tr>
<tr><td><code id="ParetoRadius_+3A_maximumnrsamples">maximumNrSamples</code></td>
<td>
<p>Optional, numeric. Maximum number for which the distance calculation can be done. 1000 by default.</p>
</td></tr>
<tr><td><code id="ParetoRadius_+3A_plotdistancepercentiles">plotDistancePercentiles</code></td>
<td>
<p>Optional, logical. If TRUE, a plot of the percentiles of distances is produced. FALSE by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Pareto-radius of the hyperspheres is derived from the optimization of information for minimal set size.
ParetoRadius() is a kernel density estimation for variable intervals. It works only on Data without missing values (NA) or NaN. In other cases, please use ParetoDensityEstimation directly.
</p>


<h3>Value</h3>

<p>numeric value, the Pareto radius.
</p>


<h3>Note</h3>

<p>This the second version of the function prior available in <span class="pkg">AdaptGauss</span>.
</p>
<p>For larger datasets the quantile_c() function is used instead of quantile in R which was programmed by Dirk Eddelbuettel on Jun 6 and taken by the author from <a href="https://github.com/RcppCore/Rcpp/issues/967">https://github.com/RcppCore/Rcpp/issues/967</a>.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>
<p><strong>See Also</strong>
</p>
<p>ParetoDensityEstimation, OptimalNoBins
</p>

<hr>
<h2 id='PDEplot'>PDE plot</h2><span id='topic+PDEplot'></span>

<h3>Description</h3>

<p>This function plots the Pareto probability density estimation  (PDE),
uses PDEstimationForGauss and ParetoRadius.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDEplot(Data, paretoRadius = 0, weight = 1, kernels = NULL,

                 LogPlot = F, PlotIt = TRUE, title =
                 "ParetoDensityEstimation(PDE)", color = "blue",
				 
                 xpoints = FALSE, xlim, ylim, xlab, ylab =
                 "PDE", ggPlot = ggplot(), sampleSize = 2e+05, lwd = 2)
				 </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PDEplot_+3A_data">Data</code></td>
<td>
<p>[1:n] numeric vector of data to be plotted.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_paretoradius">paretoRadius</code></td>
<td>
<p>numeric, the Pareto Radius. If omitted, calculate by paretoRad.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_weight">weight</code></td>
<td>
<p>numeric,  Weight*ParetoDensity  is plotted. 1 by default.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_kernels">kernels</code></td>
<td>
<p>numeric vector of kernels. Optional</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_logplot">LogPlot</code></td>
<td>
<p>LogLog PDEplot if TRUE, xpoints has to be FALSE. Optional</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_plotit">PlotIt</code></td>
<td>
<p>logical, if plot. TRUE by default.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_title">title</code></td>
<td>
<p>character vector, title of plot.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_color">color</code></td>
<td>
<p>character vector, color of plot.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_xpoints">xpoints</code></td>
<td>
<p>logical, if TRUE only points are plotted. FALSE by default.</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_xlim">xlim</code></td>
<td>
<p>Arguments to be passed to the plot method. </p>
</td></tr>
<tr><td><code id="PDEplot_+3A_ylim">ylim</code></td>
<td>
<p>Arguments to be passed to the plot method. </p>
</td></tr>
<tr><td><code id="PDEplot_+3A_xlab">xlab</code></td>
<td>
<p>Arguments to be passed to the plot method. </p>
</td></tr>
<tr><td><code id="PDEplot_+3A_ylab">ylab</code></td>
<td>
<p>Arguments to be passed to the plot method. </p>
</td></tr>
<tr><td><code id="PDEplot_+3A_ggplot">ggPlot</code></td>
<td>
<p>ggplot2 object to be plotted upon. Insert an exisiting plot to add a new PDEPlot to it. Default: empty plot</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_samplesize">sampleSize</code></td>
<td>
<p>default(200000), sample size, if datavector is to big</p>
</td></tr>
<tr><td><code id="PDEplot_+3A_lwd">lwd</code></td>
<td>
<p>linewidth, see <code>plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>kernels</code></td>
<td>
<p>numeric vector. The x points of the PDE function.</p>
</td></tr>
<tr><td><code>paretoDensity</code></td>
<td>
<p>numeric vector,  the PDE(x).</p>
</td></tr>
<tr><td><code>paretoRadius</code></td>
<td>
<p>numeric value, the Pareto Radius used for the plot.</p>
</td></tr>
<tr><td><code>ggPlot</code></td>
<td>
<p>ggplot2 object. Can be used to further modify the plot or add other plots.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun</p>


<h3>References</h3>

<p>Ultsch, A.: Pareto Density Estimation: A Density Estimation for Knowledge Discovery, Baier D., Wernecke K.D. (Eds), In Innovations in Classification, Data Science, and Information Systems - Proceedings 27th Annual Conference of the German Classification Society (GfKL) 2003, Berlin, Heidelberg, Springer, pp, 91-100, 2005.</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(1000, mean = 0.5, sd = 0.5)
y &lt;- rnorm(750, mean = -0.5, sd = 0.75)
plt &lt;- PDEplot(x, color = "red")$ggPlot
plt &lt;- PDEplot(y, color = "blue", ggPlot = plt)$ggPlot

# Second Example
#  ggplotObj=ggplot()
#  for(i in 1:length(Variables))
#     ggplotObj=PDEplot(Data[,i],ggPlot = ggplotObj)$ggPlot
</code></pre>

<hr>
<h2 id='Piechart'>
The pie chart
</h2><span id='topic+Piechart'></span>

<h3>Description</h3>

<p>the pie chart represents amount of values given in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Piechart(Datavector,Names,Labels,MaxNumberOfSlices,
main='',col,Rline=1,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Piechart_+3A_datavector">Datavector</code></td>
<td>

<p>[1:n] a vector of n non unique values
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_names">Names</code></td>
<td>
<p>Optional, 
[1:k] names to search for in Datavector, if not set <code>unique</code> of Datavector is calculated.
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_labels">Labels</code></td>
<td>
<p>Optional,
[1:k] Labels if they are specially named, if not Names are used.
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_maxnumberofslices">MaxNumberOfSlices</code></td>
<td>
<p>Default is k, integer value defining how many labels will be shown. Everything else will be summed up to <code>Other</code>.
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_main">main</code></td>
<td>

<p>Optional, title below the fan pie, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_col">col</code></td>
<td>
<p>Optional, the default are the first [1:k] colors of the default color sequence used in this package, otherwise a character vector of [1:k] specifying the colors analog to <code>plot</code>
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_rline">Rline</code></td>
<td>
<p>Optional, the radius of the pie in numerical numbers
</p>
</td></tr>
<tr><td><code id="Piechart_+3A_...">...</code></td>
<td>
<p>Optional, further arguments passed on to <code><a href="base.html#topic+plot">plot</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Number of Slices is higher than MaxNumberOfSlices then <code>ABCanalysis</code> is applied (see [Ultsch/Lotsch, 2015]) and group A chosen.
If Number of Slices in group A is higher than MaxNumberOfSlices, then the most important ones out of group A are chosen.
If MaxNumberOfSlices is higher than Slices in group A, additional slices are shown depending on the percentage (from high to low).
Parameters of visualization a set as in [Schwabish, 2014] defined.
</p>
<p>Color sequence is automatically shortened to the MaxNumberOfSlices used in the pie chart.
</p>


<h3>Value</h3>

<p>silent output by calling <code>invisible</code> of a list with
</p>
<table>
<tr><td><code>Percentages</code></td>
<td>

<p>[1:k] percent values visualized in fanplot
</p>
</td></tr>
<tr><td><code>Labels</code></td>
<td>

<p>[1:k] see input <code>Labels</code>, only relevant ones
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>You see in the example below that a pie chart does not visualize such data well contrary to the <code>fanPlot</code>.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Schwabish, 2014]	Schwabish, Jonathan A. An Economist's Guide to Visualizing Data. Journal of Economic Perspectives, 28 (1): 209-34. DOI: 10.1257/jep.28.1.209, 2014.
</p>
<p>[Ultsch/Lotsch, 2015]   Ultsch. A ., Lotsch J.: Computed ABC Analysis for Rational Selection of Most Informative Variables in Multivariate Data, PloS one, Vol. 10(6), pp. e0129767. doi 10.1371/journal.pone.0129767, 2015.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(categoricalVariable)
Piechart(categoricalVariable)
</code></pre>

<hr>
<h2 id='Pixelmatrix'>Plot of a Pixel Matrix </h2><span id='topic+Pixelmatrix'></span><span id='topic+PlotPixMatrix'></span>

<h3>Description</h3>

<p>Plots Data matrix as a pixel coulour image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pixelmatrix(Data, XNames, LowLim, HiLim,

YNames, main,FillNotFiniteWithHighestValue=FALSE) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pixelmatrix_+3A_data">Data</code></td>
<td>
<p> [1:n,1:d] Data cases in rows (n), variables in columns (d) </p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_lowlim">LowLim</code></td>
<td>
<p> Optional: limits for the color axis </p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_hilim">HiLim</code></td>
<td>
<p> Optional: limits for the color axis </p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_xnames">XNames</code></td>
<td>
<p> Optional: Vector - names for the X-ticks, NULL: no ticks at all </p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_ynames">YNames</code></td>
<td>
<p> Optional: Vector - names for the Y-ticks, NULL: no ticks at all </p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_main">main</code></td>
<td>
<p>Optoinal: String - Title of the plot</p>
</td></tr>
<tr><td><code id="Pixelmatrix_+3A_fillnotfinitewithhighestvalue">FillNotFiniteWithHighestValue</code></td>
<td>
<p>Optional: TRUE: fills not finite values with same color as the highest value</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Low values are shown in blue and green, middle values in yellow and high values in orange and red.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun, Felix Pape</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Data=Lsun3D$Data

Pixelmatrix(Data)


</code></pre>

<hr>
<h2 id='Plot3D'>
3D plot of points
</h2><span id='topic+Plot3D'></span>

<h3>Description</h3>

<p>A wrapper for Data with systematic clustering colors for either a 2D (x,y) or 3D (x,y,z) plot combined with a classification
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot3D(Data,Cls,UniqueColors,

size=2,na.rm=FALSE,Plotter3D="rgl",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot3D_+3A_data">Data</code></td>
<td>

<p>[1:n,1:d] matrix with either <code>d=2</code> or <code>d=3</code>, if <code>d&gt;3</code> only the first 3 dimensions are taken
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_cls">Cls</code></td>
<td>

<p>[1:n] numeric vector of the classification of data with <code>k</code> classes
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_uniquecolors">UniqueColors</code></td>
<td>

<p>[1:k] character vector of colors, if not given DataVisualizations::DefaultColorSequence is used
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_size">size</code></td>
<td>

<p>size of points, for plotly additional a vector [1:n] of a mapping of sizes to Cls has to be given in the (...) argument with <code>sizes=</code>
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_na.rm">na.rm</code></td>
<td>

<p>if <code>na.rm=TRUE</code>, then missing values are removed
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_plotter3d">Plotter3D</code></td>
<td>

<p>in case of 3 dimensions, choose either &quot;plotly&quot; or &quot;rgl&quot;, 
</p>
<p>if one of this packages is not given, the other one is selected as a fallback method
</p>
</td></tr>
<tr><td><code id="Plot3D_+3A_...">...</code></td>
<td>

<p>further arguments to be processed by <code><a href="rgl.html#topic+plot3d">plot3d</a></code> or <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> or <code><a href="plotly.html#topic+plot_ly">plot_ly</a></code> of type &quot;scatter3d&quot;
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> only <code>size</code> and <code>na.rm</code> is available as further arguments.
</p>


<h3>Note</h3>

<p>Uses either <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> for 2D or <code><a href="rgl.html#topic+plot3d">plot3d</a></code> for 3D or <code><a href="plotly.html#topic+plot_ly">plot_ly</a></code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>RGL vignette in <a href="https://cran.r-project.org/package=rgl">https://cran.r-project.org/package=rgl</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Spin3D similar output

data(Lsun3D)
Plot3D(Lsun3D$Data,Lsun3D$Cls,type='s',radius=0.1,box=FALSE,aspect=TRUE)
rgl::grid3d(c("x", "y", "z"))


#Projected Points with Classification
Data=cbind(runif(500,min=-3,max=3),rnorm(500))

# Classification
Cls=ifelse(Data[,1]&gt;0,1,2)
Plot3D(Data,Cls,UniqueColors = DataVisualizations::DefaultColorSequence[c(1,3)],size=2)

## Not run: 
#Points with Non-Overlapping Labels
#require(ggrepel)
Data=cbind(runif(30,min=-1,max=1),rnorm(30,0,0.5))
Names=paste0('VeryLongName',1:30)
ggobj=Plot3D(Data)
ggobj +  geom_text_repel(aes(label=Names), size=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotGraph2D'>PlotGraph2D</h2><span id='topic+PlotGraph2D'></span>

<h3>Description</h3>

<p>plots a neighborhood graph in two dimensions given the 2D coordinates of the points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGraph2D(AdjacencyMatrix, Points, Cls, Colors, xlab = "X", ylab = "Y", xlim,
ylim, Plotter = "native", LineColor = "grey", pch = 20, lwd = 0.1, main = "",
mainSize)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotGraph2D_+3A_adjacencymatrix">AdjacencyMatrix</code></td>
<td>
<p>[1:n,1:n] numerical matrix consting of binary values. 1 indicates that two points have an edge, zero that they do not</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_points">Points</code></td>
<td>
<p>[1:n,1:2] numeric matrix of two feature</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_cls">Cls</code></td>
<td>
<p>[1:n] numeric vector of k classes, if not set per default every point is in first class</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_colors">Colors</code></td>
<td>
<p>Optional, string defining the k colors, one per class</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_xlab">xlab</code></td>
<td>
<p>Optional, string for xlabel</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_ylab">ylab</code></td>
<td>
<p>Optional, string for ylabel</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_xlim">xlim</code></td>
<td>
<p>Optional, [1:2] vector of x-axis limits</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_ylim">ylim</code></td>
<td>
<p>Optional, [1:2] vector of y-axis limits</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_plotter">Plotter</code></td>
<td>
<p>Optional, either <code>"native"</code> or <code>"plotly"</code></p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_linecolor">LineColor</code></td>
<td>
<p>Optional, color of edges</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_pch">pch</code></td>
<td>
<p>Optional, shape of point, usally can be in a range from zero to 25, see pch of plot for details</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_lwd">lwd</code></td>
<td>
<p>width of the lines</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_main">main</code></td>
<td>
<p>Optional, string for the title of plot</p>
</td></tr>
<tr><td><code id="PlotGraph2D_+3A_mainsize">mainSize</code></td>
<td>
<p>Optional, scalar for the size of the title of plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The points are the vertices of the graph. the adjacency matrix defines the edges. Via adjacency matrix various graphs, like from deldir package, can be used.
</p>


<h3>Value</h3>

<p>native plot or plotly object depending on input argument <code>Plotter</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Lecture of Knowledge Discovery II
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+pch">pch</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N=10
x=runif(N)
y=runif(N)
Euklid=as.matrix(dist(cbind(x,y)))
Radius=quantile(as.vector(Euklid),0.5)
RKugelGraphAdjMatrix = matrix(0, ncol = N, nrow = N)
for (i in 1:N) {
  RInd = which(Euklid[i, ] &lt;= Radius, arr.ind = TRUE)
  RKugelGraphAdjMatrix[i, RInd] = 1
}
PlotGraph2D(RKugelGraphAdjMatrix,cbind(x,y))
</code></pre>

<hr>
<h2 id='PlotMissingvalues'>Plot of the Amount Of Missing Values</h2><span id='topic+PlotMissingvalues'></span>

<h3>Description</h3>

<p>Percentage of missing values per feature are visualized as a bar plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotMissingvalues(Data,Names,

WhichDefineMissing=c('NA','NaN','DUMMY','.',' '),

PlotIt=TRUE,

xlab='Amount Of Missing Values in Percent',

xlim=c(0,100),...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotMissingvalues_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] data cases in rows, variables/features in columns</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_names">Names</code></td>
<td>
<p>[1:d] optional vector of string describing the names of the features</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_whichdefinemissing">WhichDefineMissing</code></td>
<td>
<p>[1:d] optional vector of string describing missing values, usefull for character features. Currently up to five different options are possible.</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_plotit">PlotIt</code></td>
<td>
<p>If FALES: Does not plot</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_xlab">xlab</code></td>
<td>
<p>x label of bar plot</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_xlim">xlim</code></td>
<td>
<p>x axis limits in percent</p>
</td></tr>
<tr><td><code id="PlotMissingvalues_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code>barplot</code>, such as <code>main</code> for title</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots not finite and missing values as a bar plot for each feature d
and returns with <code>invisible</code> the amount of missing values as a vector. Works even with character variables, but WhichDefineMissing cannot be changed at the current version. Please make a suggestion on GitHub how to improve this.
</p>


<h3>Note</h3>

<p>Does not work with the tibble format, in such a case please call <code>as.data.frame(as.matrix(Data))</code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("ITS")
data("MTY")

PlotMissingvalues(cbind(ITS,MTY),Names=c('ITS','MTY'))

</code></pre>

<hr>
<h2 id='PlotProductratio'>
Product-Ratio Plot
</h2><span id='topic+PlotProductratio'></span>

<h3>Description</h3>

<p>The product-ratio plot as defined in [Tukey, 1977, p. 594].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotProductratio(X, Y, na.rm = FALSE, 

main='Product Ratio Analysis',xlab = "Log of Ratio",ylab = "Root of Product", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotProductratio_+3A_x">X</code></td>
<td>

<p>[1:n] positive numerical vector, negativ values are removed automatically
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_y">Y</code></td>
<td>

<p>[1:n] positive numerical vector, negativ values are removed automatically
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_na.rm">na.rm</code></td>
<td>

<p>Function may not work with non finite values. If these cases should be automatically removed, set parameter TRUE
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_main">main</code></td>
<td>

<p>see <code>plot</code> 
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_ylab">ylab</code></td>
<td>

<p>see <code>plot</code> 
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_xlab">xlab</code></td>
<td>

<p>see <code>plot</code> 
</p>
</td></tr>
<tr><td><code id="PlotProductratio_+3A_...">...</code></td>
<td>

<p>further arguments passed on to <code>plot</code> 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the case where there are many instances of very small values, but a small number of very large ones, this plot is usefull [Tukey, 1977, p. 615].
</p>


<h3>Value</h3>

<p>matrix[1:n,2] with sqrt(x*y) and log(x/y) as the two columns
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Tukey, 1977]  Tukey, J. W.: Exploratory data analysis, United States Addison-Wesley Publishing Company, ISBN: 0-201-07616-0, 1977.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Beware: The data does no fit ne requirements for this approach
data('ITS')
data(MTY)
PlotProductratio(ITS,MTY)
</code></pre>

<hr>
<h2 id='PmatrixColormap'>P-Matrix colors</h2><span id='topic+PmatrixColormap'></span>

<h3>Description</h3>

<p>Defines the default color sequence for plots made with PDEscatter
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PmatrixColormap")</code></pre>


<h3>Format</h3>

<p>Returns the vectors for a (heat) colormap.
</p>

<hr>
<h2 id='QQplot'>
QQplot with a Linear Fit
</h2><span id='topic+QQplot'></span>

<h3>Description</h3>

<p>Qantile-quantile plot with a linear fit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQplot(X,Y,Type=8,NoQuantiles=10000,xlab, ylab,col="red",main='',
lwd=3,pch=20,subplot=FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QQplot_+3A_x">X</code></td>
<td>
<p>[1:n] numerical vector,
First Feature</p>
</td></tr>
<tr><td><code id="QQplot_+3A_y">Y</code></td>
<td>

<p>1:n] numerical vector, Second Feature to compare first feature with</p>
</td></tr>
<tr><td><code id="QQplot_+3A_type">Type</code></td>
<td>

<p>an integer between 1 and 9 selecting one of the nine quantile algorithms detailed in <code>quantile </code>
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_noquantiles">NoQuantiles</code></td>
<td>

<p>number of quantiles used in QQ-plot, if number is low and the data has outliers, there may be empty space visible in the plot</p>
</td></tr>
<tr><td><code id="QQplot_+3A_xlab">xlab</code></td>
<td>

<p>x label, see <code>plot</code>
...</p>
</td></tr>
<tr><td><code id="QQplot_+3A_ylab">ylab</code></td>
<td>

<p>y label, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_col">col</code></td>
<td>

<p>color of line, see <code>plot</code></p>
</td></tr>
<tr><td><code id="QQplot_+3A_main">main</code></td>
<td>

<p>title of plot, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_lwd">lwd</code></td>
<td>

<p>line width of plot, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_pch">pch</code></td>
<td>

<p>type of point, see <code>plot</code>
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_subplot">subplot</code></td>
<td>

<p>FALSE: par is set specifically, TRUE: assumption is the usage as a subfigure, par has to be set by the user, no checks are performed, labels have to be set by the user
</p>
</td></tr>
<tr><td><code id="QQplot_+3A_...">...</code></td>
<td>

<p>other parameters for <code>qqplot</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Output is the evaluation of a linear (regression) fit of <code>lm</code> called 'line' and a quantile quantile plot (QQplot). Per default 10.000 quantiles are chosen, but in the case of very large data vectors one can reduce the quantiles for faster computation.
The 100 percentiles used for the regression line are of darker blue than the quantiles chosen by the user.
</p>


<h3>Value</h3>

<p>List with
</p>
<table>
<tr><td><code>Quantiles</code></td>
<td>
<p>[1:NoQuantiles,1:2] quantiles in y and y</p>
</td></tr>
<tr><td><code>Residuals</code></td>
<td>
<p>Output of the Regression with <code>residuals.lm(line)</code></p>
</td></tr>
<tr><td><code>Summary</code></td>
<td>
<p>Output of the Regression with <code>summaryline)</code></p>
</td></tr>
<tr><td><code>Anova</code></td>
<td>
<p>Output of the Regression with <code>anova(line)</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Michael, J. R.: The stabilized probability plot, Biometrika, Vol. 70(1), pp. 11-17, 1983.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MTY)
NormalDistribution=rnorm(50000)
QQplot(NormalDistribution,MTY)
</code></pre>

<hr>
<h2 id='RobustNorm_BackTrafo'>
Transforms the Robust Normalization back
</h2><span id='topic+RobustNorm_BackTrafo'></span>

<h3>Description</h3>

<p>Transforms the Robust Normalization back if <code>Capped=FALSE</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RobustNorm_BackTrafo(TransformedData,

MinX,Denom,Center=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RobustNorm_BackTrafo_+3A_transformeddata">TransformedData</code></td>
<td>
<p>[1:n,1:d]  matrix</p>
</td></tr>
<tr><td><code id="RobustNorm_BackTrafo_+3A_minx">MinX</code></td>
<td>
<p>scalar</p>
</td></tr>
<tr><td><code id="RobustNorm_BackTrafo_+3A_denom">Denom</code></td>
<td>
<p>scalar</p>
</td></tr>
<tr><td><code id="RobustNorm_BackTrafo_+3A_center">Center</code></td>
<td>
<p>scalar</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For details see <code><a href="#topic+RobustNormalization">RobustNormalization</a></code>
</p>


<h3>Value</h3>

<p>[1:n,1:d] Data matrix</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RobustNormalization">RobustNormalization</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Lsun3D)
Data = Lsun3D$Data
TransList = RobustNormalization(Data, Centered = TRUE, WithBackTransformation = TRUE)

Lsun3DData = RobustNorm_BackTrafo(TransList$TransformedData,
                                 TransList$MinX,
                                 TransList$Denom,
                                 TransList$Center)

sum(Lsun3DData - Data) #&lt;e-15
</code></pre>

<hr>
<h2 id='RobustNormalization'>
RobustNormalization
</h2><span id='topic+RobustNormalization'></span>

<h3>Description</h3>

<p>RobustNormalization as described in [Milligan/Cooper, 1988].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RobustNormalization(Data,Centered=FALSE,Capped=FALSE,

na.rm=TRUE,WithBackTransformation=FALSE,

pmin=0.01,pmax=0.99) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RobustNormalization_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] data matrix of n cases and d features</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_centered">Centered</code></td>
<td>
<p> centered data around zero by median if TRUE</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_capped">Capped</code></td>
<td>
<p>TRUE: outliers are capped above 1 or below -1 and set to 1 or -1.</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_na.rm">na.rm</code></td>
<td>
<p>If TRUE, infinite vlaues are disregarded</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_withbacktransformation">WithBackTransformation</code></td>
<td>
<p>If in the case for forecasting with neural
networks a backtransformation is required, this parameter can be set to 'TRUE'.</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_pmin">pmin</code></td>
<td>
<p>defines outliers on the lower end of scale</p>
</td></tr>
<tr><td><code id="RobustNormalization_+3A_pmax">pmax</code></td>
<td>
<p>defines outliers on the higher end of scale</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Normalizes features either between -1 to 1 (Centered=TRUE) or 0-1
(Centered=TRUE) without changing the distribution of a feature itself. For a
more precise description please read [Thrun, 2018, p.17].
</p>
<p>&quot;[The] scaling of the inputs determines the effective scaling of the weights in
the last layer of a MLP with BP neural netowrk, it can have a large effect on
the quality of the final solution. At the outset it is best to standardize all
inputs to have mean zero and standard deviation 1
[(or at least the range under 1)]. This ensures all inputs are treated equally
in the regularization prozess, and allows to choose a meaningful range for the
random starting weights.&quot;[Friedman et al., 2012]
</p>


<h3>Value</h3>

<p>if <code>WithBackTransformation=FALSE</code>: TransformedData[1:n,1:d] i.e.,
normalized data matrix of n cases and d features
</p>
<p>if <code>WithBackTransformation=TRUE</code>: List with
</p>
<table>
<tr><td><code>TransformedData</code></td>
<td>
<p>[1:n,1:d]  normalized data matrix of n cases and d
features</p>
</td></tr>
<tr><td><code>MinX</code></td>
<td>
<p>[1:d] numerical vector used for manual back-transformation of each
feature</p>
</td></tr>
<tr><td><code>MaxX</code></td>
<td>
<p>[1:d] numerical vector used for manual back-transformation of each
feature</p>
</td></tr>
<tr><td><code>Denom</code></td>
<td>
<p>[1:d] numerical vector used for manual back-transformation of each
feature</p>
</td></tr>
<tr><td><code>Center</code></td>
<td>
<p>[1:d] numerical vector used for manual back-transformation of each
feature</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Milligan/Cooper, 1988]  Milligan, G. W., &amp; Cooper, M. C.: A study of
standardization of variables in cluster analysis, Journal of Classification,
Vol. 5(2), pp. 181-204. 1988.
</p>
<p>[Friedman et al., 2012]  Friedman, J., Hastie, T., &amp; Tibshirani, R.: The
Elements of Statistical Learning, (Second ed. Vol. 1), Springer series in
statistics New York, NY, USA:, ISBN, 2012.
</p>
<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through
Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer,
Heidelberg, ISBN: 978-3-658-20539-3, <a href="https://doi.org/10.1007/978-3-658-20540-9">doi:10.1007/978-3-658-20540-9</a>, 2018. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RobustNorm_BackTrafo">RobustNorm_BackTrafo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Scaled = RobustNormalization(rnorm(1000, 2, 100), Capped = TRUE)
hist(Scaled)

m = cbind(c(1, 2, 3), c(2, 6, 4))
List = RobustNormalization(m, FALSE, FALSE, FALSE, TRUE)
TransformedData = List$TransformedData

mback = RobustNorm_BackTrafo(TransformedData, List$MinX, List$Denom, List$Center)

sum(m - mback)
</code></pre>

<hr>
<h2 id='ROC'>ROC plot</h2><span id='topic+ROC'></span>

<h3>Description</h3>

<p>ROC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ROC(Data, Cls, Names, Colors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ROC_+3A_data">Data</code></td>
<td>
<p>[1:n, 1:d] numeric vector or matrix of scores to be evaluated with
ROC.</p>
</td></tr>
<tr><td><code id="ROC_+3A_cls">Cls</code></td>
<td>
<p>[1:n] numeric vector with true classes.</p>
</td></tr>
<tr><td><code id="ROC_+3A_names">Names</code></td>
<td>
<p>[1:d] character vector with names for scores.</p>
</td></tr>
<tr><td><code id="ROC_+3A_colors">Colors</code></td>
<td>
<p>[1:d] character vector with colores for scores.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>ROCit</code></td>
<td>
<p>List of ROCit results for each score column in Data.</p>
</td></tr>
<tr><td><code>Plot</code></td>
<td>
<p>Plotly object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Quirin Stier</p>


<h3>Examples</h3>

<pre><code class='language-R'>
Data = runif(1000,0,1)
Cls  = sample(c(0,1), 1000, replace = TRUE)
ROC(Data, Cls)

</code></pre>

<hr>
<h2 id='ShepardDensityscatter'>Shepard PDE scatter</h2><span id='topic+ShepardDensityScatter'></span><span id='topic+ShepardDensityPlot'></span>

<h3>Description</h3>

<p>Draws ein Shepard Diagram (scatterplot of distances) with an two-dimensional PDE density estimation .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
ShepardDensityScatter(InputDists, OutputDists, Plotter= "native", Type = "DDCAL",
DensityEstimation="SDH", Marginals = FALSE, xlab='Input Distances',
ylab='Output Distances',main='ProjectionMethod', sampleSize=500000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ShepardDensityscatter_+3A_inputdists">InputDists</code></td>
<td>
<p>[1:n,1:n] with n cases of data in d variables/features:
Matrix containing the distances of the inputspace.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_outputdists">OutputDists</code></td>
<td>
<p>[1:n,1:n] with n cases of data in d dimensionalites of the projection method variables/features: 
Matrix containing the distances of the outputspace.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_plotter">Plotter</code></td>
<td>

<p>Optional, either <code>"native"</code> or <code>"plotly"</code>
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_type">Type</code></td>
<td>

<p>Optional, either <code>"DDCAL"</code> which creates a special hard color transition sensitive to density-based structures or <code>"Standard"</code> which creates a standard continuous color transition which is proven to be not very sensitive for complex density-based structures.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_densityestimation">DensityEstimation</code></td>
<td>

<p>Optional, use either <code>"SDH"</code> or <code>"PDE"</code> for data density estimation.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_marginals">Marginals</code></td>
<td>

<p>Optional, either TRUE (draw Marginals) or FALSE (do not draw Marginals)
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_xlab">xlab</code></td>
<td>

<p>Label of the x axis in the resulting Plot.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_ylab">ylab</code></td>
<td>

<p>Label of the y axis in the resulting Plot.
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_main">main</code></td>
<td>

<p>Title of the Shepard diagram
</p>
</td></tr>
<tr><td><code id="ShepardDensityscatter_+3A_samplesize">sampleSize</code></td>
<td>
<p>Optional, default(500000),  reduces a.ount of data for density estimation, if too many distances given</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Introduced and described in [Thrun, 2018, p. 63] with examples in [Thrun, 2018, p. 71-72]</p>


<h3>Author(s)</h3>

<p>Michael Thrun</p>


<h3>References</h3>

<p>[Thrun, 2018]  Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, ISBN:  978-3-658-20540-9, Heidelberg, 2018.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Cls=Lsun3D$Cls
Data=Lsun3D$Data
InputDist=as.matrix(dist(Data))
res = stats::cmdscale(d = InputDist, k = 2, eig = TRUE, 
        add = FALSE, x.ret = FALSE)

ProjectedPoints = as.matrix(res$points)
ShepardDensityScatter(InputDist,as.matrix(dist(ProjectedPoints)),main = 'MDS')
ShepardDensityScatter(InputDist[1:100,1:100],

as.matrix(dist(ProjectedPoints))[1:100,1:100],main = 'MDS')

</code></pre>

<hr>
<h2 id='Sheparddiagram'>
Draws a Shepard Diagram
</h2><span id='topic+Sheparddiagram'></span>

<h3>Description</h3>

<p>This function plots a Shepard diagram  which is a scatter plot of InputDist and OutputDist
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sheparddiagram(InputDists, OutputDists, xlab = "Input Distances",

                 ylab= "Output Distances", fancy = F,

				 main = "ProjectionMethod", gPlot = ggplot())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sheparddiagram_+3A_inputdists">InputDists</code></td>
<td>
<p>[1:n,1:n] with n cases of data in d variables/features: 
Matrix containing the distances of the inputspace.
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_outputdists">OutputDists</code></td>
<td>
<p>[1:n,1:n] with n cases of data in d dimensionalites of the projection method variables/features: 
Matrix containing the distances of the outputspace.
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_xlab">xlab</code></td>
<td>

<p>Label of the x axis in the resulting Plot.
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_ylab">ylab</code></td>
<td>

<p>Label of the y axis in the resulting Plot.
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_fancy">fancy</code></td>
<td>

<p>Set FALSE for PC and TRUE for publication
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_main">main</code></td>
<td>

<p>Title of the Shepard diagram
</p>
</td></tr>
<tr><td><code id="Sheparddiagram_+3A_gplot">gPlot</code></td>
<td>

<p>ggplot2 object to plot upon.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object containing the plot.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Cls=Lsun3D$Cls
Data=Lsun3D$Data
InputDist=as.matrix(dist(Data))
res = stats::cmdscale(d = InputDist, k = 2, eig = TRUE, 
        add = FALSE, x.ret = FALSE)
ProjectedPoints = as.matrix(res$points)


Sheparddiagram(InputDist,as.matrix(dist(ProjectedPoints)),main = 'MDS')


</code></pre>

<hr>
<h2 id='SignedLog'>
Signed Log
</h2><span id='topic+SignedLog'></span>

<h3>Description</h3>

<p>Computes the Signed Log if Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SignedLog(Data,Base="Ten")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SignedLog_+3A_data">Data</code></td>
<td>

<p>[1:n,1:d] Data matrix with n cases and d variables
</p>
</td></tr>
<tr><td><code id="SignedLog_+3A_base">Base</code></td>
<td>

<p>Either &quot;Ten&quot;, &quot;Two&quot;, &quot;Zero&quot;, or any number.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A neat transformation for data, it it has a better representation on the log scale.
</p>


<h3>Value</h3>

<p>Transformed Data
</p>


<h3>Note</h3>

<p>Number Selections for <code>Base</code> for 2,10, &quot;Two&quot; or &quot;Ten&quot; add 1 to every datapoint as defined in the lectures.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Prof. Dr. habil. A. Ultsch, Lectures in Knowledge Discovery, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+log">log</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># sampling is done
# because otherwise the example takes too long
# in the CRAN check
data('ITS')
ind=sample(length(ITS),1000)

MDplot(SignedLog(cbind(ITS[ind],MTY[ind])*(-1),Base = "Ten"))
</code></pre>

<hr>
<h2 id='Silhouetteplot'>
Silhouette plot of classified data.
</h2><span id='topic+Silhouetteplot'></span>

<h3>Description</h3>

<p>Silhouette plot of cluster silhouettes for the n-by-d data
matrix Data or distance matrix where the clusters are defined in the vector Cls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Silhouetteplot(DataOrDistances, Cls, method='euclidean',

PlotIt=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Silhouetteplot_+3A_dataordistances">DataOrDistances</code></td>
<td>
<p>  [1:n,1:d] data cases in rows, variables in columns, if not symmetric
</p>
<p>or
</p>
<p>[1:n,1:n] distance matrix, if symmetric
</p>
</td></tr>
<tr><td><code id="Silhouetteplot_+3A_cls">Cls</code></td>
<td>
<p>   numeric vector, [1:n,1]  classified data  </p>
</td></tr>
<tr><td><code id="Silhouetteplot_+3A_method">method</code></td>
<td>
<p>Optional if Datamatrix is used, 
one of &quot;euclidean&quot;, &quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot; or &quot;minkowski&quot;. Any unambiguous substring can be given, see <code>dist</code>
</p>
</td></tr>
<tr><td><code id="Silhouetteplot_+3A_plotit">PlotIt</code></td>
<td>
<p>Optional, Default:TRUE, FALSE to supress the plot</p>
</td></tr>
<tr><td><code id="Silhouetteplot_+3A_...">...</code></td>
<td>
<p>If <code>PlotIt=TRUE</code>: Further arguements to <code><a href="graphics.html#topic+barplot">barplot</a></code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&quot;The Silhouette plot is a common unsupervised index for visual evaluation of a clustering [L. R. Kaufman/Rousseeuw, 2005] [introduced in  [Rousseeuw, 1987]]. A reasonable clustering is characterized by a silhouette width of greater than 0.5, and an average width below 0.2 should be interpreted as indicating a lack of any substantial cluster structure [Everitt et al., 2001, p. 105]. However, it is evident that silhouette scores assume clusters that are spherical or Gaussian in shape [Herrmann, 2011, pp. 91-92]&quot; [Thrun, 2018, p. 29].
</p>


<h3>Value</h3>

<table>
<tr><td><code>silh</code></td>
<td>
<p>Silhouette values in a N-by-1 vector</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Onno Hansen-Goos, Michael Thrun
</p>


<h3>References</h3>

<p>[Thrun, 2018] Thrun, M. C.: Projection Based Clustering through Self-Organization and Swarm Intelligence, doctoral dissertation 2017, Springer, ISBN: 978-3-658-20539-3, Heidelberg, 2018.
</p>
<p>[Rousseeuw, 1987]   Rousseeuw, Peter J.: Silhouettes: a Graphical Aid to the Interpretation and Validation of Cluster Analysis, Computational and Applied Mathematics, 20, p.53-65, 1987.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Lsun3D")
Cls=Lsun3D$Cls
Data=Lsun3D$Data
#clear cluster structure
plot(Data[,1:2],col=Cls)
#However, the silhouette plot does not indicate a very good clustering in cluster 1 and 2
Silhouetteplot(Data,Cls = Cls,main='Silhouetteplot')

</code></pre>

<hr>
<h2 id='Slopechart'>
Slope Chart
</h2><span id='topic+Slopechart'></span>

<h3>Description</h3>

<p>ABC analysis improved slope chart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Slopechart(FirstDatavector,

SecondDatavector,

Names,

Labels,

MaxNumberOfSlices,

TopLabels=c('FirstDatavector','SecondDatavector'),

main='Comparision of Descending Frequency')
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Slopechart_+3A_firstdatavector">FirstDatavector</code></td>
<td>

<p>[1:n] a vector of n non unique values - a features
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_seconddatavector">SecondDatavector</code></td>
<td>

<p>[1:m] a vector of n non unique values - a second feature
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_labels">Labels</code></td>
<td>
<p>Optional,
[1:k] Labels if they are specially named, if not Names are used.
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_names">Names</code></td>
<td>

<p>[1:k] names to search for in Datavector, if not set <code>unique</code> of Datavector is calculated.
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_maxnumberofslices">MaxNumberOfSlices</code></td>
<td>
<p>Default is k, integer value defining how many labels will be shown. Everything else will be summed up to <code>Other</code>.
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_toplabels">TopLabels</code></td>
<td>

<p>Labels of of feature names
</p>
</td></tr>
<tr><td><code id="Slopechart_+3A_main">main</code></td>
<td>

<p>title of the plot
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>still experimental.
</p>


<h3>Value</h3>

<p>silent output by calling <code>invisible</code> of a list with
</p>
<table>
<tr><td><code>Percentages</code></td>
<td>

<p>[1:k] percent values visualized in fanplot
</p>
</td></tr>
<tr><td><code>Labels</code></td>
<td>

<p>[1:k] see input <code>Labels</code>, only relevant ones
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>[Gohil, 2015]  Gohil, Atmajitsinh. R data Visualization cookbook. Packt Publishing Ltd, 2015.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Piechart">Piechart</a></code>, <code><a href="#topic+Fanplot">Fanplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## will follow
</code></pre>

<hr>
<h2 id='stat_pde_density'>
Calculate Pareto density estimation for ggplot2 plots
</h2><span id='topic+stat_pde_density'></span>

<h3>Description</h3>

<p>This function enables to replace the default density estimation for ggplot2 plots with the Pareto density estimation [Ultsch, 2005].
It is used for the PDE-Optimized violin plot published in [Thrun et al, 2018].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_pde_density(mapping = NULL,
                 data = NULL,
                 geom = "violin",
                 position = "dodge",
                 ...,
                 trim = TRUE,
                 scale = "area",
                 na.rm = FALSE,
                 show.legend = NA,
                 inherit.aes = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stat_pde_density_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code> or
<code><a href="ggplot2.html#topic+aes_">aes_()</a></code>. If specified and <code>inherit.aes = TRUE</code> (the
default), it is combined with the default mapping at the top level of the
plot. You must supply <code>mapping</code> if there is no plot mapping.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame.</code>, and
will be used as the layer data.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use display the data</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_position">position</code></td>
<td>
<p>Position adjustment, either as a string, or the result of
a call to a position adjustment function.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggplot2.html#topic+layer">layer()</a></code>. These are
often aesthetics, used to set an aesthetic to a fixed value, like
<code>color = "red"</code> or <code>size = 3</code>. They may also be parameters
to the paired geom/stat.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_trim">trim</code></td>
<td>
<p>This parameter only matters if you are displaying multiple
densities in one plot. If 'FALSE', the default, each density is
computed on the full range of the data. If 'TRUE', each density
is computed over the range of that group: this typically means the
estimated x values will not line-up, and hence you won't be able to
stack density values.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_scale">scale</code></td>
<td>
<p>When used with geom_violin: if &quot;area&quot; (default), all violins have the same area (before trimming
the tails). If &quot;count&quot;, areas are scaled proportionally to the number of
observations. If &quot;width&quot;, all violins have the same maximum width.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code> (the default), removes missing values with
a warning.  If <code>TRUE</code> silently removes missing values.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="stat_pde_density_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pareto Density Estimation (PDE) is a method for the estimation of probability density functions using hyperspheres. The Pareto-radius of the hyperspheres is derived from the optimization of information for minimal set size. It is shown, that Pareto Density is the best estimate for clusters of Gaussian structure. The method is shown to be robust when cluster overlap and when the variances differ across clusters.
</p>


<h3>Author(s)</h3>

<p>Felix Pape
</p>


<h3>References</h3>

<p>Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>
<p>[Thrun et al, 2018]	Thrun, M. C., Pape, F., &amp; Ultsch, A. : Benchmarking Cluster Analysis Methods using PDE-Optimized Violin Plots, Proc. European Conference on Data Analysis (ECDA), accepted, Paderborn, Germany, 2018. 
</p>


<h3>See Also</h3>

<p><code>[ggplot2]stat_density</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>miris &lt;- reshape2::melt(iris)

ggplot2::ggplot(miris, 

mapping = ggplot2::aes_string(y = 'value', x = 'variable')) +

ggplot2::geom_violin(stat = "PDEdensity")
</code></pre>

<hr>
<h2 id='StatPDEdensity'>
Pareto Density Estimation
</h2><span id='topic+StatPDEdensity'></span>

<h3>Description</h3>

<p>Density Estimation for ggplot with a clear model behind it.
</p>


<h3>Format</h3>

<p>The format is:
Classes 'StatPDEdensity', 'Stat', 'ggproto' &lt;ggproto object: Class StatPDEdensity, Stat&gt;
aesthetics: function
compute_group: function
compute_layer: function
compute_panel: function
default_aes: uneval
extra_params: na.rm
finish_layer: function
non_missing_aes: 
parameters: function
required_aes: x y
retransform: TRUE
setup_data: function
setup_params: function
super:  &lt;ggproto object: Class Stat&gt; 
</p>


<h3>Details</h3>

<p>PDE was published in [Ultsch, 2005], short explanation in [Thrun, Ultsch 2018] and the PDE optimized violin plot was published in [Thrun et al., 2018].
</p>


<h3>References</h3>

<p>[Ultsch,2005]  Ultsch, A.: Pareto density estimation: A density estimation for knowledge discovery, in  Baier, D.; Werrnecke, K. D., (Eds), Innovations in classification, data science, and information systems, Proc Gfkl 2003, pp 91-100, Springer, Berlin, 2005.
</p>
<p>[Thrun, Ultsch 2018]   Thrun, M. C., &amp; Ultsch, A. : Effects of the payout system of income taxes to municipalities in Germany, in Papiez, M. &amp; Smiech,, S. (eds.), Proc. 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, pp. 533-542, Cracow: Foundation of the Cracow University of Economics, Cracow, Poland, 2018. 
</p>
<p>[Thrun et al, 2018]	Thrun, M. C., Pape, F., &amp; Ultsch, A. : Benchmarking Cluster Analysis Methods using PDE-Optimized Violin Plots, Proc. European Conference on Data Analysis (ECDA), accepted, Paderborn, Germany, 2018. 
</p>

<hr>
<h2 id='world_country_polygons'>
world_country_polygons
</h2><span id='topic+world_country_polygons'></span>

<h3>Description</h3>

<p>world_country_polygons shapefile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("world_country_polygons")</code></pre>


<h3>Format</h3>

<p>world_country_polygons  stores data objects using classes defined
in the sp package or inheriting from those classes updated to sp Y= 1.4 and
rgdal &gt;= 1.5.
</p>
<p>Since DataVisualization Version 1.2.1 it stores now a CRS objects with a comment containing an WKT2 CRS representation, thanks to a suggestion of Roger Bivand.
</p>


<h3>Details</h3>

<p>Note that the rebuilt CRS object contains
a revised version of the input Proj4 string as well as the WKT2 string,
and may be used with both older and newer versions of sp. See maptools package for further details.
Also note that since sp &gt;= 2.0 maptools and rgdal were deprecated without change to the workflow. See terra for an alternative to maptools.
</p>


<h3>Author(s)</h3>

<p>Hamza Tayyab, Michael Thrun</p>


<h3>Source</h3>

<p>maptools package
</p>


<h3>References</h3>

<p>maptools package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(world_country_polygons)
str(world_country_polygons)

</code></pre>

<hr>
<h2 id='Worldmap'>
plots a world map by country codes
</h2><span id='topic+Worldmap'></span>

<h3>Description</h3>

<p>The Worldmap function is used in [Thrun, 2018].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Worldmap(CountryCodes, Cls, Colors, 

MissingCountryColor = grDevices::gray(0.8), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Worldmap_+3A_countrycodes">CountryCodes</code></td>
<td>

<p>[1:n] vector of characters identifying countries by ISO 3166 codes (2 or 3 letters)
</p>
</td></tr>
<tr><td><code id="Worldmap_+3A_cls">Cls</code></td>
<td>

<p>[1:n] numerical vector of classification
</p>
</td></tr>
<tr><td><code id="Worldmap_+3A_colors">Colors</code></td>
<td>

<p>optional, vector of charcters specifying the used colors
</p>
</td></tr>
<tr><td><code id="Worldmap_+3A_missingcountrycolor">MissingCountryColor</code></td>
<td>

<p>if not all countries are specified in <code>CountryCodes</code> then the color of non relevant countries can be changed here
</p>
</td></tr>
<tr><td><code id="Worldmap_+3A_...">...</code></td>
<td>

<p>Further arguments passed on to <code>plot</code>, see also <code>sp::SpatialPolygons-class</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of
</p>
<table>
<tr><td><code>Colors</code></td>
<td>
<p>[1:m] colors used in map, m&lt;=n</p>
</td></tr>
<tr><td><code>CountryCodeList</code></td>
<td>
<p> [1:m] countries found, m&lt;=n</p>
</td></tr>
<tr><td><code>world_country_polygons</code></td>
<td>
<p><code>SpatialPolygonsDataFrame</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michae Thrun
</p>


<h3>References</h3>

<p>Used in
</p>
<p>[Thrun, 2018] Thrun, M. C. : Cluster Analysis of the World Gross-Domestic Product Based on Emergent Self-Organization of a Swarm, 12th Professor Aleksander Zelias International Conference on Modelling and Forecasting of Socio-Economic Phenomena, Foundation of the Cracow University of Economics, Zakopane, Poland, accepted, 2018. 
</p>
<p>Source for shapefile: - package maptoops and 
</p>
<p>Originally 'mappinghacks.com/data/TM_WORLD_BORDERS_SIMPL-0.2.zip', now available from <a href="https://github.com/nasa/World-Wind-Java/tree/master/WorldWind/testData/shapefiles">https://github.com/nasa/World-Wind-Java/tree/master/WorldWind/testData/shapefiles</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data from [Thrun, 2018]
Cls=c(1L, 1L, 2L, 2L, 2L, 2L, 2L, 1L, 2L, 1L, 1L, 1L, 2L, 2L, 2L, 
2L, 2L, 1L, 2L, 2L, 2L, 1L, 2L, 1L, 2L, 1L, 2L, 2L, 1L, 1L, 1L, 
1L, 2L, 1L, 1L, 2L, 2L, 2L, 1L, 2L, 2L, 2L, 2L, 2L, 1L, 2L, 1L, 
2L, 2L, 2L, 1L, 2L, 2L, 2L, 1L, 1L, 1L, 1L, 3L, 2L, 2L, 2L, 1L, 
2L, 1L, 1L, 2L, 1L, 1L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 1L, 
1L, 2L, 2L, 2L, 1L, 2L, 1L, 2L, 1L, 1L, 2L, 2L, 1L, 1L, 1L, 2L, 
2L, 1L, 2L, 1L, 1L, 1L, 2L, 1L, 2L, 2L, 1L, 1L, 1L, 2L, 2L, 1L, 
2L, 2L, 1L, 2L, 2L, 1L, 2L, 1L, 2L, 2L, 2L, 1L, 2L, 1L, 1L, 1L, 
2L, 1L, 1L, 2L, 1L, 1L, 2L, 2L, 1L, 2L, 1L, 1L, 1L, 2L, 2L, 2L, 
2L, 2L, 2L, 1L, 1L, 2L, 2L, 2L, 2L, 1L, 2L, 2L, 2L, 1L, 1L, 1L
)
Codes=c("AFG", "AGO", "ALB", "ARG", "ATG", "AUS", "AUT", "BDI", "BEL", 
"BEN", "BFA", "BGD", "BGR", "BHR", "BHS", "BLZ", "BMU", "BOL", 
"BRA", "BRB", "BRN", "BTN", "BWA", "CAF", "CAN", "CH2", "CHE", 
"CHL", "CHN", "CIV", "CMR", "COG", "COL", "COM", "CPV", "CRI", 
"CUB", "CYP", "DJI", "DMA", "DNK", "DOM", "DZA", "ECU", "EGY", 
"ESP", "ETH", "FIN", "FJI", "FRA", "FSM", "GAB", "GBR", "GER", 
"GHA", "GIN", "GMB", "GNB", "GNQ", "GRC", "GRD", "GTM", "GUY", 
"HKG", "HND", "HTI", "HUN", "IDN", "IND", "IRL", "IRN", "IRQ", 
"ISL", "ISR", "ITA", "JAM", "JOR", "JPN", "KEN", "KHM", "KIR", 
"KNA", "KOR", "LAO", "LBN", "LBR", "LCA", "LKA", "LSO", "LUX", 
"MAC", "MAR", "MDG", "MDV", "MEX", "MHL", "MLI", "MLT", "MNG", 
"MOZ", "MRT", "MUS", "MWI", "MYS", "NAM", "NER", "NGA", "NIC", 
"NLD", "NOR", "NPL", "NZL", "OMN", "PAK", "PAN", "PER", "PHL", 
"PLW", "PNG", "POL", "PRI", "PRT", "PRY", "ROM", "RWA", "SDN", 
"SEN", "SGP", "SLB", "SLE", "SLV", "SOM", "STP", "SUR", "SWE", 
"SWZ", "SYC", "SYR", "TCD", "TGO", "THA", "TON", "TTO", "TUN", 
"TUR", "TWN", "TZA", "UGA", "URY", "USA", "VCT", "VEN", "VNM", 
"VUT", "WSM", "ZAF", "ZAR", "ZMB", "ZWE")
Worldmap(Codes,Cls)
</code></pre>

<hr>
<h2 id='zplot'>
Plotting for 3 dimensional data
</h2><span id='topic+zplot'></span>

<h3>Description</h3>

<p>Plots z above xy plane as 3D mountain or 2D contourlines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zplot(x, y, z, DrawTopView = TRUE, NrOfContourLines = 20,

                 TwoDplotter = "native", xlim, ylim)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zplot_+3A_x">x</code></td>
<td>

<p>Vector of x-coordinates of the data. If y and z are missing: Matrix containing 3 rows, one for each coordinate
</p>
</td></tr>
<tr><td><code id="zplot_+3A_y">y</code></td>
<td>

<p>Vector of y-coordinates of the data.
</p>
</td></tr>
<tr><td><code id="zplot_+3A_z">z</code></td>
<td>

<p>Vector of z-coordinates of the data.
</p>
</td></tr>
<tr><td><code id="zplot_+3A_drawtopview">DrawTopView</code></td>
<td>

<p>Optional: Boolean, if true plot contours otherwise a 3D plot. Default: True
</p>
</td></tr>
<tr><td><code id="zplot_+3A_nrofcontourlines">NrOfContourLines</code></td>
<td>

<p>Optional: Numeric. Only used when DrawTopView == True. Number of lines to be drawn in 2D contour plots. Default: 20
</p>
</td></tr>
<tr><td><code id="zplot_+3A_twodplotter">TwoDplotter</code></td>
<td>

<p>Optional: String indicating which backend to use for plotting. Possible Values: 'ggplot', 'native', 'plotly'
</p>
</td></tr>
<tr><td><code id="zplot_+3A_xlim">xlim</code></td>
<td>

<p>[1:2] scalar vector setting the limits of x-axis
</p>
</td></tr>
<tr><td><code id="zplot_+3A_ylim">ylim</code></td>
<td>

<p>[1:2] scalar vector setting the limits of y-axis
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the plotting backend does support it, this will return a handle for the generated plot. 
</p>


<h3>Author(s)</h3>

<p>Felix pape
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
