<!DOCTYPE html><html><head><title>Help for package dbparser</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dbparser}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dbparser'><p>dbparser: A package for reading and parsing <strong>DrugBank</strong> xml database.</p></a></li>
<li><a href='#add_drugbank_info'><p>add_drugbank_info</p>
Add passed DrugBank db metadata to passed dvobject</a></li>
<li><a href='#articles'><p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters articles element parser</p></a></li>
<li><a href='#attachments'><p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters attachments element parser</p></a></li>
<li><a href='#books'><p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters books element parser</p></a></li>
<li><a href='#cett_actions_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Actions parsers</p></a></li>
<li><a href='#cett_doc'><p>Carriers/ Enzymes/ Targets/ Transporters parsers</p></a></li>
<li><a href='#cett_ex_identity_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Polypeptide External Identifiers</p>
parsers</a></li>
<li><a href='#cett_go_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Polypeptide GO Classifier</p>
parsers</a></li>
<li><a href='#cett_nodes_options'><p>returns carriers, enzymes,targets and transporters node valid options.</p></a></li>
<li><a href='#cett_poly_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Polypeptide parsers</p></a></li>
<li><a href='#cett_poly_pfms_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Polypeptide PFAMS parsers</p></a></li>
<li><a href='#cett_poly_syn_doc'><p>Carriers/ Enzymes/ Targets/ Transporters Polypeptide Synonyms parsers</p></a></li>
<li><a href='#drug_affected_organisms'><p>Drug Affected Organism parser</p></a></li>
<li><a href='#drug_ahfs_codes'><p>Drug ahfs-codes parser</p></a></li>
<li><a href='#drug_atc_codes'><p>Drug ATC Codes element parser</p></a></li>
<li><a href='#drug_calc_prop'><p>Drug Calculated Properties parser</p></a></li>
<li><a href='#drug_categories'><p>Drug Categories parser</p></a></li>
<li><a href='#drug_classification'><p>Drug Classification parser</p></a></li>
<li><a href='#drug_dosages'><p>Drug Dosages parser</p></a></li>
<li><a href='#drug_ex_identity'><p>Drug External Identifiers parser</p></a></li>
<li><a href='#drug_exp_prop'><p>Drug Experimental Properties parser</p></a></li>
<li><a href='#drug_external_links'><p>Drug External Links parser</p></a></li>
<li><a href='#drug_food_interactions'><p>Drug Groups parser</p></a></li>
<li><a href='#drug_general_information'><p>Drugs General Information parser</p></a></li>
<li><a href='#drug_groups'><p>Drug Groups parser</p></a></li>
<li><a href='#drug_interactions'><p>Drug Interactions parser</p></a></li>
<li><a href='#drug_intern_brand'><p>Drug International Brands parser</p></a></li>
<li><a href='#drug_manufacturers'><p>Drug Manufacturers parser</p></a></li>
<li><a href='#drug_mixtures'><p>Drug Mixtures parser</p></a></li>
<li><a href='#drug_node_options'><p>returns drug node valid options.</p></a></li>
<li><a href='#drug_packagers'><p>Drug Packagers parser</p></a></li>
<li><a href='#drug_patents'><p>Drug Patents parser</p>
A property right issued by the United States Patent and Trademark
Office (USPTO) to an inventor for a limited time, in exchange for public
disclosure of the invention when the patent is granted. Drugs may be issued
multiple patents.</a></li>
<li><a href='#drug_pathway'><p>Drug Pathway parser</p></a></li>
<li><a href='#drug_pathway_drugs'><p>Drug Pathway Drugs parser</p></a></li>
<li><a href='#drug_pathway_enzyme'><p>Drug Pathway Enzymes parser</p></a></li>
<li><a href='#drug_pdb_entries'><p>Drug pdb-entries parser</p></a></li>
<li><a href='#drug_pharmacology'><p>Drug Pharmacology parser</p></a></li>
<li><a href='#drug_prices'><p>Drug Prices Parsers</p></a></li>
<li><a href='#drug_products'><p>Drug Products parser</p></a></li>
<li><a href='#drug_reactions'><p>Drug Reactions Parsers</p></a></li>
<li><a href='#drug_reactions_enzymes'><p>Drug Reactions Enzymes Parsers</p></a></li>
<li><a href='#drug_salts'><p>Drug Salts parser</p></a></li>
<li><a href='#drug_sequences'><p>Drug Sequences parser</p></a></li>
<li><a href='#drug_snp_adverse_reactions'><p>Drug SNP Adverse Drug Reactions parser</p></a></li>
<li><a href='#drug_snp_effects'><p>Drug SNP Effects parser</p></a></li>
<li><a href='#drug_syn'><p>Drug Synonyms parser</p></a></li>
<li><a href='#init_dvobject'><p>init_dvobject</p>
initialize dvobject</a></li>
<li><a href='#links'><p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters links element parser</p></a></li>
<li><a href='#parse_cett_node'><p>Run all CETT related parsers</p></a></li>
<li><a href='#parse_drug_nodes'><p>Run all drug  related parsers</p></a></li>
<li><a href='#parse_references_node'><p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters references element parser</p></a></li>
<li><a href='#parseDrugBank'><p>parseDrugBank</p></a></li>
<li><a href='#read_drugbank_xml_db'><p>Reads <strong>DrugBank</strong> xml database and load it into memory.</p></a></li>
<li><a href='#references_node_options'><p>returns references node valid options.</p></a></li>
<li><a href='#show_dvobject_metadata'><p>init_dvobject</p>
Returns data.frame with two columns (key, value) of dvobject attributes</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Drugs Databases Parser</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>This tool is for parsing public drug databases such as 'DrugBank' XML database <a href="https://go.drugbank.com/">https://go.drugbank.com/</a>.
    The parsed data are then returned in a proper 'R' object called 'dvobject'.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, progress, purrr, tibble, tools, XML</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>canvasXpress, knitr, rmarkdown, testthat, tidyr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/dbparser/">https://docs.ropensci.org/dbparser/</a>,
<a href="https://github.com/ropensci/dbparser">https://github.com/ropensci/dbparser</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/dbparser/issues">https://github.com/ropensci/dbparser/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-20 06:40:43 UTC; LEGION</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohammed Ali [aut, cre],
  Ali Ezzat [aut],
  Hao Zhu [rev],
  Emma Mendelsohn [rev]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohammed Ali &lt;moh_fcis@yahoo.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-20 06:52:52 UTC</td>
</tr>
</table>
<hr>
<h2 id='dbparser'>dbparser: A package for reading and parsing <strong>DrugBank</strong> xml database.</h2><span id='topic+dbparser'></span><span id='topic+dbparser-package'></span>

<h3>Description</h3>

<p>The main purpose of the 'dbparser' package is to parse
[DrugBank](https://go.drugbank.com/) database which is downloadable in XML format
from [this link](https://go.drugbank.com/releases/latest).
</p>


<h3>Details</h3>

<p>The parsed data can then be explored and analyzed.
</p>
<p>To achieve this purpose, 'dbparser&ldquo; package provides three main categories
of functions:
</p>
<p>- xml db reader,
</p>
<p>- <strong>DrugBank</strong> elements parsers,
</p>
<p>For more information kindly check the
reference/index (https://docs.ropensci.org/dbparser/reference/index.html)
</p>


<h3>xml db reader functions</h3>

<p>Reads <strong>DrugBank</strong> xml database and build drug elements full tree in
memory
</p>


<h3>parsers functions</h3>

<p>Each parser function is responsible of parsing certain drug element and
returning its tibble.
</p>
<p>Check this tutorial
(https://docs.ropensci.org/dbparser/articles/dbparser.html)
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Mohammed Ali <a href="mailto:moh_fcis@yahoo.com">moh_fcis@yahoo.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Ali Ezzat <a href="mailto:ali_ezzat85@yahoo.com">ali_ezzat85@yahoo.com</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Hao Zhu <a href="mailto:haozhu233@gmail.com">haozhu233@gmail.com</a> [reviewer]
</p>
</li>
<li><p> Emma Mendelsohn [reviewer]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://docs.ropensci.org/dbparser/">https://docs.ropensci.org/dbparser/</a>
</p>
</li>
<li> <p><a href="https://github.com/ropensci/dbparser">https://github.com/ropensci/dbparser</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ropensci/dbparser/issues">https://github.com/ropensci/dbparser/issues</a>
</p>
</li></ul>


<hr>
<h2 id='add_drugbank_info'>add_drugbank_info
Add passed DrugBank db metadata to passed dvobject</h2><span id='topic+add_drugbank_info'></span>

<h3>Description</h3>

<p>add_drugbank_info
Add passed DrugBank db metadata to passed dvobject
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_drugbank_info(dvobject)
</code></pre>


<h3>Value</h3>

<p>dvobject
</p>

<hr>
<h2 id='articles'>Drugs/ Carriers/ Enzymes/ Targets/ Transporters articles element parser</h2><span id='topic+articles'></span><span id='topic+drugs_articles'></span><span id='topic+carriers_articles'></span><span id='topic+enzymes_articles'></span><span id='topic+targets_articles'></span><span id='topic+transporters_articles'></span>

<h3>Description</h3>

<p>Return a list of articles that were used as references for drugs carriers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugs_articles()

carriers_articles()

enzymes_articles()

targets_articles()

transporters_articles()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>ref-id</dt><dd><p>Identifier for the article being referenced.
This is unique across all reference types (books, links, article,
attachments).</p>
</dd>
<dt>pubmed-id</dt><dd><p>The PubMed identifier for the article.</p>
</dd>
<dt>citation</dt><dd><p>Article citation in a standard format.</p>
</dd>
<dt><em>parent_id</em></dt><dd><p>drug/carrier/target/enzyme/transporter id</p>
</dd>
</dl>


<hr>
<h2 id='attachments'>Drugs/ Carriers/ Enzymes/ Targets/ Transporters attachments element parser</h2><span id='topic+attachments'></span><span id='topic+drugs_attachments'></span><span id='topic+carriers_attachments'></span><span id='topic+enzymes_attachments'></span><span id='topic+targets_attachments'></span><span id='topic+transporters_attachments'></span>

<h3>Description</h3>

<p>Return a list of attachment that were used as references for drugs carriers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugs_attachments()

carriers_attachments()

enzymes_attachments()

targets_attachments()

transporters_attachments()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>ref-id</dt><dd><p>Identifier for the article being referenced.
This is unique across all reference types (books, links, article,
attachments).</p>
</dd>
<dt>title</dt><dd><p>The title of the attachment.</p>
</dd>
<dt>url</dt><dd><p>The url to download the attachment from.</p>
</dd>
<dt><em>parent_id</em></dt><dd><p>drug/carrier/target/enzyme/transporter id</p>
</dd>
</dl>


<hr>
<h2 id='books'>Drugs/ Carriers/ Enzymes/ Targets/ Transporters books element parser</h2><span id='topic+books'></span><span id='topic+drugs_textbooks'></span><span id='topic+carriers_textbooks'></span><span id='topic+enzymes_textbooks'></span><span id='topic+targets_textbooks'></span><span id='topic+transporters_textbooks'></span>

<h3>Description</h3>

<p>Return a list of text books that were used as references for drugs, carriers,
enzymes, targets or transporters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugs_textbooks()

carriers_textbooks()

enzymes_textbooks()

targets_textbooks()

transporters_textbooks()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>ref-id</dt><dd><p>Identifier for the article being referenced.
This is unique across all reference types (books, links, article,
attachments).</p>
</dd>
<dt>isbn</dt><dd><p>ISBN identifying the textbook.</p>
</dd>
<dt>citation</dt><dd><p>A Textbook citation in a standard format.</p>
</dd>
<dt><em>parent_id</em></dt><dd><p>drug/ carrier/ target/ enzyme/ transporter id</p>
</dd>
</dl>


<hr>
<h2 id='cett_actions_doc'>Carriers/ Enzymes/ Targets/ Transporters Actions parsers</h2><span id='topic+cett_actions_doc'></span><span id='topic+carriers_actions'></span><span id='topic+enzymes_actions'></span><span id='topic+targets_actions'></span><span id='topic+transporters_actions'></span>

<h3>Description</h3>

<p>Collection of related actions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_actions()

enzymes_actions()

targets_actions()

transporters_actions()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>action</dt><dd><p>describe related action</p>
</dd>
<dt><em>parent_id</em></dt><dd><p>carrier/ target/ enzyme/ transporter id</p>
</dd>
</dl>


<hr>
<h2 id='cett_doc'>Carriers/ Enzymes/ Targets/ Transporters parsers</h2><span id='topic+cett_doc'></span><span id='topic+carriers'></span><span id='topic+enzymes'></span><span id='topic+targets'></span><span id='topic+transporters'></span>

<h3>Description</h3>

<p>Protein targets of drug action, enzymes that are inhibited/induced or
involved in metabolism, and carrier or transporter proteins involved in
movement of the drug across biological membranes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers()

enzymes()

targets()

transporters()
</code></pre>


<h3>Value</h3>

<p>a tibble with 6 variables (8 for enzymes):
</p>

<dl>
<dt>id</dt><dd><p>Universal Protein Resource (UniProt) Identifier for the record</p>
</dd>
<dt>name</dt><dd><p>related name</p>
</dd>
<dt>organism</dt><dd><p>Organism that the protein comes from.</p>
</dd>
<dt>known_action</dt><dd><p>Whether the pharmacological action of the drug is due
to this target interaction.</p>
</dd>
<dt>inhibition-strength</dt><dd><p>Whether the strength of enzyme inhibition is
strong, moderate, or unknown. <strong>Only applies to enzymes</strong></p>
</dd>
<dt>induction-strength</dt><dd><p>Whether the strength of enzyme induction is
strong or unknown. <strong>Only applies to enzymes</strong></p>
</dd>
<dt>position</dt><dd><p>related position</p>
</dd>
<dt>parent_id</dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='cett_ex_identity_doc'>Carriers/ Enzymes/ Targets/ Transporters Polypeptide External Identifiers
parsers</h2><span id='topic+cett_ex_identity_doc'></span><span id='topic+carriers_polypep_ex_ident'></span><span id='topic+enzymes_polypep_ex_ident'></span><span id='topic+targets_polypep_ex_ident'></span><span id='topic+transporters_polypep_ex_ident'></span>

<h3>Description</h3>

<p>Extract descriptions of identified polypeptide external identifiers for
targets, enzymes, carriers, or transporters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_polypep_ex_ident()

enzymes_polypep_ex_ident()

targets_polypep_ex_ident()

transporters_polypep_ex_ident()
</code></pre>


<h3>Value</h3>

<p>a tibble with 3 variables:
</p>

<dl>
<dt>resource</dt><dd><p>Name of the source database.</p>
</dd>
<dt>identifier</dt><dd><p>Identifier for this drug in the given resource.</p>
</dd>
<dt>parent_key</dt><dd><p>polypeptide id</p>
</dd>
</dl>


<hr>
<h2 id='cett_go_doc'>Carriers/ Enzymes/ Targets/ Transporters Polypeptide GO Classifier
parsers</h2><span id='topic+cett_go_doc'></span><span id='topic+carriers_polypeptides_go'></span><span id='topic+enzymes_polypeptides_go'></span><span id='topic+targets_polypeptides_go'></span><span id='topic+transporters_polypeptides_go'></span>

<h3>Description</h3>

<p>Extract descriptions of identified polypeptide go classifier for targets,
enzymes, carriers, or transporters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_polypeptides_go()

enzymes_polypeptides_go()

targets_polypeptides_go()

transporters_polypeptides_go()
</code></pre>


<h3>Value</h3>

<p>a tibble with 3 variables:
</p>

<dl>
<dt>category</dt><dd></dd>
<dt>description</dt><dd></dd>
<dt>parent_key</dt><dd><p>polypeptide id</p>
</dd>
</dl>


<hr>
<h2 id='cett_nodes_options'>returns carriers, enzymes,targets and transporters node valid options.</h2><span id='topic+cett_nodes_options'></span>

<h3>Description</h3>

<p>returns carriers, enzymes,targets and transporters node valid options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cett_nodes_options()
</code></pre>


<h3>Value</h3>

<p>list of CETT valid options
</p>


<h3>See Also</h3>

<p>Other parsers: 
<code><a href="#topic+drug_node_options">drug_node_options</a>()</code>,
<code><a href="#topic+parseDrugBank">parseDrugBank</a>()</code>,
<code><a href="#topic+references_node_options">references_node_options</a>()</code>
</p>

<hr>
<h2 id='cett_poly_doc'>Carriers/ Enzymes/ Targets/ Transporters Polypeptide parsers</h2><span id='topic+cett_poly_doc'></span><span id='topic+carriers_polypeptides'></span><span id='topic+enzymes_polypeptides'></span><span id='topic+targets_polypeptides'></span><span id='topic+transporters_polypeptides'></span>

<h3>Description</h3>

<p>Extract descriptions of identified polypeptide targets, enzymes, carriers,
or transporters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_polypeptides()

enzymes_polypeptides()

targets_polypeptides()

transporters_polypeptides()
</code></pre>


<h3>Value</h3>

<p>a tibble with 20 variables:
</p>

<dl>
<dt>id</dt><dd><p><a href="https://www.uniprot.org/">Universal Protein Resource
(UniProt) identifier</a></p>
</dd>
<dt>source</dt><dd><p>Specifies whether the identified polypeptide ID is
associated with any of the following UniProt knowledge bases:
Swiss-Prot, which is manually annotated and reviewed, or TrEMBL,
which is automatically annotated and not reviewed.</p>
</dd>
<dt>name</dt><dd></dd>
<dt>general_function</dt><dd><p>General summary of the physiological function of
the polypeptide</p>
</dd>
<dt>specific_function</dt><dd><p>A more specific description of the polypeptide’s
physiological function within the cell.</p>
</dd>
<dt>gene_name</dt><dd><p>The short name commonly associated with the associated
gene. Eg. PTGS1.</p>
</dd>
<dt>locus</dt><dd><p>The specific chromosomal location or position of the gene’s
sequence on a chromosome.</p>
</dd>
<dt>cellular_location</dt><dd><p>The cellular location of the polypeptide.</p>
</dd>
<dt>transmembrane_regions</dt><dd><p>Areas of the polypeptide sequence that span
a biological membrane.</p>
</dd>
<dt>signal_regions</dt><dd><p>Location of any signal peptides within the
polypeptide sequence.</p>
</dd>
<dt>theoretical_pi</dt><dd><p>Theoretical isoelectric point.</p>
</dd>
<dt>molecular_weight</dt><dd><p>The molecular weight of the polypeptide.</p>
</dd>
<dt>chromosome_location</dt><dd><p>The chromosomal location of the polypeptide
gene</p>
</dd>
<dt>organism</dt><dd><p>The organism in which this polypeptide functions.</p>
</dd>
<dt>organism_ncbi_taxonomy_id</dt><dd></dd>
<dt>amino_acid_sequence</dt><dd><p>The amino acid sequence of the polypeptide</p>
</dd>
<dt>amino_acid_format</dt><dd></dd>
<dt>gene_sequence</dt><dd><p>The sequence of the associated gene.</p>
</dd>
<dt>gene_format</dt><dd></dd>
<dt>parent_key</dt><dd><p>carrier/ target/ enzyme/ transporter id</p>
</dd>
</dl>


<hr>
<h2 id='cett_poly_pfms_doc'>Carriers/ Enzymes/ Targets/ Transporters Polypeptide PFAMS parsers</h2><span id='topic+cett_poly_pfms_doc'></span><span id='topic+carriers_polypeptides_pfams'></span><span id='topic+enzymes_polypeptides_pfams'></span><span id='topic+targets_polypeptides_pfams'></span><span id='topic+transporters_polypeptides_pfams'></span>

<h3>Description</h3>

<p>Extract descriptions of identified polypeptide PFAMS targets, enzymes,
carriers, or transporters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_polypeptides_pfams()

enzymes_polypeptides_pfams()

targets_polypeptides_pfams()

transporters_polypeptides_pfams()
</code></pre>


<h3>Value</h3>

<p>a tibble with 3 variables:
</p>

<dl>
<dt>name</dt><dd><p>The sequence of the associated gene.</p>
</dd>
<dt>identifier</dt><dd></dd>
<dt>parent_key</dt><dd><p>polypeptide id</p>
</dd>
</dl>


<hr>
<h2 id='cett_poly_syn_doc'>Carriers/ Enzymes/ Targets/ Transporters Polypeptide Synonyms parsers</h2><span id='topic+cett_poly_syn_doc'></span><span id='topic+carriers_polypeptides_syn'></span><span id='topic+enzymes_polypeptides_syn'></span><span id='topic+targets_polypeptides_syn'></span><span id='topic+transporters_polypeptides_syn'></span>

<h3>Description</h3>

<p>Extract descriptions of identified polypeptide synonyms for targets,
enzymes, carriers, or transporters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carriers_polypeptides_syn()

enzymes_polypeptides_syn()

targets_polypeptides_syn()

transporters_polypeptides_syn()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>synonym</dt><dd></dd>
<dt>parent_key</dt><dd><p>polypeptide id</p>
</dd>
</dl>


<hr>
<h2 id='drug_affected_organisms'>Drug Affected Organism parser</h2><span id='topic+drug_affected_organisms'></span>

<h3>Description</h3>

<p>Organisms in which the drug may display activity; activity may depend on
local susceptibility patterns and resistance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_affected_organisms()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>affected-organism</dt><dd><p>affected-organism name</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_ahfs_codes'>Drug ahfs-codes parser</h2><span id='topic+drug_ahfs_codes'></span>

<h3>Description</h3>

<p>The American Hospital Formulary Service (AHFS) identifier for this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_ahfs_codes()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>ahfs-code</dt><dd></dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_atc_codes'>Drug ATC Codes element parser</h2><span id='topic+drug_atc_codes'></span>

<h3>Description</h3>

<p>The Anatomical Therapeutic Classification (ATC) code for the drug assigned
by the World Health Organization Anatomical Chemical Classification System.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_atc_codes()
</code></pre>


<h3>Details</h3>

<p>Each 'atc-code&ldquo; row has one or more level. The atc-code and level&gt;
have a code  the code assigned by the World Health Organization Anatomical
Therapeutic Chemical Classification system.
</p>


<h3>Value</h3>

<p>a tibble with 10 variables
</p>

<hr>
<h2 id='drug_calc_prop'>Drug Calculated Properties parser</h2><span id='topic+drug_calc_prop'></span>

<h3>Description</h3>

<p>Drug properties that have been predicted by ChemAxon or ALOGPS based on the
imputed chemical structure. Associated links below will redirect to
descriptions of the specific term.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_calc_prop()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>kind</dt><dd><p>Name of the property.</p>
</dd>
<dt>value</dt><dd><p>Predicted physicochemical properties; obtained by the use of
prediction software such as ALGOPS and ChemAxon.</p>
</dd>
<dt>source</dt><dd><p>Name of the software used to calculate this property,
either ChemAxon or ALOGPS.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_categories'>Drug Categories parser</h2><span id='topic+drug_categories'></span>

<h3>Description</h3>

<p>General categorizations of the drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_categories()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>category</dt><dd><p>category name</p>
</dd>
<dt>mesh-id</dt><dd><p>The Medical Subjects Headings (MeSH) identifier for the
category.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_classification'>Drug Classification parser</h2><span id='topic+drug_classification'></span>

<h3>Description</h3>

<p>A description of the hierarchical chemical classification of the drug;
imported from ClassyFire.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_classification()
</code></pre>


<h3>Value</h3>

<p>a tibble with 9 variables:
</p>

<dl>
<dt>description</dt><dd></dd>
<dt>direct-parent</dt><dd></dd>
<dt>kingdom</dt><dd></dd>
<dt>superclass</dt><dd></dd>
<dt>class</dt><dd></dd>
<dt>subclass</dt><dd></dd>
<dt>alternative-parent</dt><dd><p>One or more alternative parents</p>
</dd>
<dt>substituent</dt><dd><p>One or more substituents</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_dosages'>Drug Dosages parser</h2><span id='topic+drug_dosages'></span>

<h3>Description</h3>

<p>A list of the commercially available dosages of the drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_dosages()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>form</dt><dd><p>The pharmaceutical formulation by which the drug is introduced
into the body</p>
</dd>
<dt>route</dt><dd><p>The path by which the drug or product is taken into the body.</p>
</dd>
<dt>strength</dt><dd><p>The amount of active drug ingredient provided in the dosage</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_ex_identity'>Drug External Identifiers parser</h2><span id='topic+drug_ex_identity'></span>

<h3>Description</h3>

<p>Identifiers used in other websites or databases providing information about
this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_ex_identity()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>resource</dt><dd><p>Name of the source database.</p>
</dd>
<dt>identifier</dt><dd><p>Identifier for this drug in the given resource.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_exp_prop'>Drug Experimental Properties parser</h2><span id='topic+drug_exp_prop'></span>

<h3>Description</h3>

<p>Drug properties that have been experimentally proven
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_exp_prop()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>kind</dt><dd><p>Name of the property.</p>
</dd>
<dt>value</dt><dd><p>Drug properties that have been experimentally proven.</p>
</dd>
<dt>source</dt><dd><p>Reference to the source of this experimental data.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>

<p>The following experimental properties are provided:
</p>

<dl>
<dt>Water Solubility</dt><dd><p>The experimentally determined aqueous solubility
of the molecule.</p>
</dd>
<dt>Molecular Formula</dt><dd><p>Protein formula of Biotech drugs</p>
</dd>
<dt>Molecular Weight</dt><dd><p>Protein weight of Biotech drugs.</p>
</dd>
<dt>Melting Point</dt><dd><p>The experimentally determined temperature at which the
drug molecule changes from solid to liquid at atmospheric temperature.</p>
</dd>
<dt>Boiling Point</dt><dd><p>The experimentally determined temperature at which the
drug molecule changes from liquid to gas at atmospheric temperature.</p>
</dd>
<dt>Hydrophobicity</dt><dd><p>The ability of a molecule to repel water rather than
absorb or dissolve water.</p>
</dd>
<dt>Isoelectric Point</dt><dd><p>The pH value at which the net electric charge of a
molecule is zero.</p>
</dd>
<dt>caco2 Permeability</dt><dd><p>A continuous line of heterogenous human epithelial
colorectal adenocarcinoma cells, CAC02 cells are employed as a model of
human intestinal absorption of various drugs and compounds. CAC02 cell
permeability is ultimately an assay to measure drug absorption.</p>
</dd>
<dt>pKa</dt><dd><p>The experimentally determined pka value of the molecule</p>
</dd>
<dt>logP</dt><dd><p>The experimentally determined partition coefficient (LogP)
based on the ratio of solubility of the molecule in 1-octanol compared to
water.</p>
</dd>
<dt>logS</dt><dd><p>The intrinsic solubility of a given compound is the
concentration in equilibrium with its solid phase that dissolves into
solution, given as the natural logarithm (LogS) of the concentration.</p>
</dd>
<dt>Radioactivity</dt><dd><p>The property to spontaneously emit particles
(alpha, beta, neutron) or radiation (gamma, K capture), or both at the same
time, from the decay of certain nuclides.</p>
</dd>
</dl>


<hr>
<h2 id='drug_external_links'>Drug External Links parser</h2><span id='topic+drug_external_links'></span>

<h3>Description</h3>

<p>Links to other websites or databases providing information about this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_external_links()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>resource</dt><dd><p>Name of the source website.</p>
</dd>
<dt>identifier</dt><dd><p>Identifier for this drug in the given resource</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_food_interactions'>Drug Groups parser</h2><span id='topic+drug_food_interactions'></span>

<h3>Description</h3>

<p>Food that may interact with this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_food_interactions()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>food-interaction</dt><dd></dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_general_information'>Drugs General Information parser</h2><span id='topic+drug_general_information'></span>

<h3>Description</h3>

<p>A description of the hierarchical chemical classification of the drug;
imported from ClassyFire.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_general_information()
</code></pre>


<h3>Value</h3>

<p>a tibble with 15 variables:
</p>

<dl>
<dt>primary_key</dt><dd><p>DrugBank id</p>
</dd>
<dt>other_keys</dt><dd><p>Other identifiers that may be associated with the drug</p>
</dd>
<dt>type</dt><dd><p>	Either small molecule, or biotech. Biotech is used for any
drug that is derived from living systems or organisms, usually composed of
high molecular weight mixtures of protein, while small molecule describes
a low molecular weight organic compound.</p>
</dd>
<dt>name</dt><dd></dd>
<dt>created</dt><dd><p>Date that this drug was first added to DrugBank.</p>
</dd>
<dt>updated</dt><dd><p>Denotes when this drug was last updated in DrugBank.</p>
</dd>
<dt>description</dt><dd><p>Descriptions of drug chemical properties,
history and regulatory status.</p>
</dd>
<dt>cas_number</dt><dd><p>The Chemical Abstracts Service (CAS) registry number
assigned to the drug.</p>
</dd>
<dt><em>unii</em></dt><dd><p>Unique Ingredient Identifier (UNII) of this drug.</p>
</dd>
<dt>average_mass</dt><dd><p>The weighted average of the isotopic masses of the
drug</p>
</dd>
<dt>state</dt><dd><p>One of solid, liquid, or gas</p>
</dd>
<dt>monoisotopic_mass</dt><dd><p>The mass of the most abundant isotope of the drug</p>
</dd>
<dt>synthesis_reference</dt><dd><p>Citation for synthesis of the drug molecule.</p>
</dd>
<dt>fda_label</dt><dd><p>Contains a URL for accessing the uploaded United States Food
and Drug Administration (FDA) Monograph for this drug.</p>
</dd>
<dt>msds</dt><dd><p>Contains a URL for accessing the Material Safety Data Sheet
(MSDS) for this drug.</p>
</dd>
</dl>


<hr>
<h2 id='drug_groups'>Drug Groups parser</h2><span id='topic+drug_groups'></span>

<h3>Description</h3>

<p>Groups that this drug belongs to. May include any of: approved, vet_approved,
nutraceutical, illicit, withdrawn, investigational, and experimental.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_groups()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>group</dt><dd></dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_interactions'>Drug Interactions parser</h2><span id='topic+drug_interactions'></span>

<h3>Description</h3>

<p>Drug-drug interactions detailing drugs that, when administered concomitantly
with the drug of interest, will affect its activity or result in adverse
effects. These interactions may be synergistic or antagonistic depending on
the physiological effects and mechanism of action of each drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_interactions()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>drugbank-id	</dt><dd><p>DrugBank ID of the interacting drug.</p>
</dd>
<dt>name</dt><dd><p>Name of the interacting drug.</p>
</dd>
<dt>description</dt><dd><p>Textual description of the physiological consequences
of the drug interaction</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>parent drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_intern_brand'>Drug International Brands parser</h2><span id='topic+drug_intern_brand'></span>

<h3>Description</h3>

<p>The proprietary names used by the manufacturers for commercially available
forms of the drug, focusing on brand names for products that are available
in countries other than Canada and the Unites States.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_intern_brand()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>brand</dt><dd><p>The proprietary, well-known name for given to this drug by a
manufacturer.</p>
</dd>
<dt>company</dt><dd><p>The company or manufacturer that uses this name.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_manufacturers'>Drug Manufacturers parser</h2><span id='topic+drug_manufacturers'></span>

<h3>Description</h3>

<p>A list of companies that are manufacturing the commercially available forms
of this drug that are available in Canada and the Unites States.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_manufacturers()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>generic</dt><dd><p>A list of companies that are manufacturing the generic
form of the drug.</p>
</dd>
<dt>url</dt><dd><p>A link to the companies that are manufacturing the drug.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_mixtures'>Drug Mixtures parser</h2><span id='topic+drug_mixtures'></span>

<h3>Description</h3>

<p>All commercially available products in which this drug is available in
combination with other drug molecules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_mixtures()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>name</dt><dd><p>The proprietary name provided by the manufacturer for this
combination product.</p>
</dd>
<dt>ingredients</dt><dd><p>A list of ingredients, separated by addition symbols</p>
</dd>
<dt>supplemental-ingredients</dt><dd><p>List of additional active ingredients which
are not clinically relevant to the main indication of the product,
separated by addition symbols.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_node_options'>returns drug node valid options.</h2><span id='topic+drug_node_options'></span>

<h3>Description</h3>

<p>returns drug node valid options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_node_options()
</code></pre>


<h3>Value</h3>

<p>list of drug valid options
</p>


<h3>See Also</h3>

<p>Other parsers: 
<code><a href="#topic+cett_nodes_options">cett_nodes_options</a>()</code>,
<code><a href="#topic+parseDrugBank">parseDrugBank</a>()</code>,
<code><a href="#topic+references_node_options">references_node_options</a>()</code>
</p>

<hr>
<h2 id='drug_packagers'>Drug Packagers parser</h2><span id='topic+drug_packagers'></span>

<h3>Description</h3>

<p>A list of companies that are packaging the drug for re-distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_packagers()
</code></pre>


<h3>Value</h3>

<p>a tibble with 2 variables:
</p>

<dl>
<dt>name</dt><dd></dd>
<dt>url</dt><dd><p>A link to any companies that are packaging the drug for
re-distribution.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_patents'>Drug Patents parser
A property right issued by the United States Patent and Trademark
Office (USPTO) to an inventor for a limited time, in exchange for public
disclosure of the invention when the patent is granted. Drugs may be issued
multiple patents.</h2><span id='topic+drug_patents'></span>

<h3>Description</h3>

<p>Drug Patents parser
A property right issued by the United States Patent and Trademark
Office (USPTO) to an inventor for a limited time, in exchange for public
disclosure of the invention when the patent is granted. Drugs may be issued
multiple patents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_patents()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>number</dt><dd><p>The patent number(s) associated with the drug.</p>
</dd>
<dt>country</dt><dd><p>The country that issued the patent rights.</p>
</dd>
<dt>approved</dt><dd><p>The date that the patent request was filed.</p>
</dd>
<dt>expires</dt><dd><p>The date that the patent rights expire.</p>
</dd>
<dt>pediatric-extension</dt><dd><p>Indicates whether or not a pediatric extension has been approved for
the patent. Granted pediatric extensions provide an additional 6 months of
market protection.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_pathway'>Drug Pathway parser</h2><span id='topic+drug_pathway'></span>

<h3>Description</h3>

<p>Metabolic, disease, and biological pathways that the drug is involved in, as
identified by the Small Molecule Protein Database (SMPDB).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_pathway()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>smpdb-id</dt><dd><p>Small Molecule Pathway Database identifier for this
pathway.</p>
</dd>
<dt>name</dt><dd><p>Pathway name</p>
</dd>
<dt>category</dt><dd><p>Pathway category</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_pathway_drugs'>Drug Pathway Drugs parser</h2><span id='topic+drug_pathway_drugs'></span>

<h3>Description</h3>

<p>Drugs involved in this pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_pathway_drugs()
</code></pre>


<h3>Value</h3>

<p>a tibble with pathway drugs properties
</p>

<hr>
<h2 id='drug_pathway_enzyme'>Drug Pathway Enzymes parser</h2><span id='topic+drug_pathway_enzyme'></span>

<h3>Description</h3>

<p>Enzymes involved in this pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_pathway_enzyme()
</code></pre>


<h3>Value</h3>

<p>a tibble with pathway properties
</p>

<hr>
<h2 id='drug_pdb_entries'>Drug pdb-entries parser</h2><span id='topic+drug_pdb_entries'></span>

<h3>Description</h3>

<p>Protein Data Bank (PDB) identifiers for this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_pdb_entries()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>pdb-entry</dt><dd></dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_pharmacology'>Drug Pharmacology parser</h2><span id='topic+drug_pharmacology'></span>

<h3>Description</h3>

<p>Describes the use, mechanism of action, pharmacokinetics, pharmacodynamics,
and physiological or biochemical effects in the body.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_pharmacology()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>indication</dt><dd><p>The approved conditions, diseases, or states for which a
drug can safely and effectively be used. An indication is considered to be
FDA-approved when it has any of the following designations: NDA, ANDA, BLA,
or OTC. May also include indications in other countries, such as Canada
(through Health Canada) or in Europe
(through the European Medicines Agency).</p>
</dd>
<dt>pharmacodynamics</dt><dd><p>A description of how the drug modifies or affects
the organism it is being used in. May include effects in the body that are
desired (enzyme or protein targets for example) and undesired
(also known as “side effects”). This is in contrast to pharmacokinetics,
which describes how the body modifies the drug being used.</p>
</dd>
<dt>mechanism_of_action</dt><dd><p>A component of pharmacodynamics that describes
the biochemical interaction through which a drug produces its intended
effect. May include the exact molecular protein or enzyme targets and/or
a description of the physiological effects produced.</p>
</dd>
<dt>toxicity</dt><dd><p>Any adverse reaction, or side effect, that may or may not
occur with use of the drug. May be attributed to a number of effects
including: an enhanced therapeutic effect, rare anaphylactic reactions,
interactions with other medications, or unanticipated binding of the
molecule at different sites within the body.</p>
</dd>
<dt>metabolism</dt><dd><p>A description of the chemical degradation of the drug
molecule within the body; most commonly by enzymes from the
Cytochrome P450 (CYP) system in the liver.</p>
</dd>
<dt>absorption</dt><dd><p>A description of the movement of the drug from the site
of administration into the bloodstream or target tissue. Common
pharmacokinetic metrics used to evaluate absorption include Area Under
the Curve (AUC), bioavailability (F), maximum concentration (Cmax), and
time to maximum concentration (Tmax).</p>
</dd>
<dt>half-life</dt><dd><p>The period of time it takes for the amount of drug in the
body to be reduced by one half. Provides a description of how quickly the
drug is being eliminated and how much is available in the bloodstream.</p>
</dd>
<dt>protein-binding	</dt><dd><p>A description of the drug’s affinity for plama
proteins and the proportion of the drug that is bound to them when in
circulation within the body.</p>
</dd>
<dt>route_of_elimination</dt><dd><p>A description of the pathway that is used to
excrete the drug from the body. Common pharmacokinetic parameters used to
evaluate excretion include elimination half life, renal clearance, and
tracking of radiolabelled compounds through the renal and GI system.</p>
</dd>
<dt>volume_of_distribution</dt><dd><p>The Vd of a drug represents the degree to
which it is distributed into body tissue compared to the plasma.</p>
</dd>
<dt>clearance</dt><dd><p>A pharmacokinetic measurement of the rate of removal of the
drug from plasma, expressed as mL/min; reflects the rate of elimination of
the drug.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_prices'>Drug Prices Parsers</h2><span id='topic+drug_prices'></span>

<h3>Description</h3>

<p>Unit drug prices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_prices()
</code></pre>


<h3>Value</h3>

<p>a tibble with 5 variables:
</p>

<dl>
<dt>description</dt><dd></dd>
<dt>cost</dt><dd><p>Drug price per unit</p>
</dd>
<dt>currency</dt><dd><p>Currency of price, example: US.</p>
</dd>
<dt>unit</dt><dd></dd>
<dt>parent_id</dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_products'>Drug Products parser</h2><span id='topic+drug_products'></span>

<h3>Description</h3>

<p>A list of commercially available products in Canada and the United States
that contain the drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_products()
</code></pre>


<h3>Value</h3>

<p>a tibble with 32 variables:
</p>

<dl>
<dt>name</dt><dd><p>The proprietary name(s) provided by the manufacturer for any
commercially available products containing this drug.</p>
</dd>
<dt>labeller</dt><dd><p>The corporation responsible for labelling this product.</p>
</dd>
<dt>ndc-id</dt><dd><p>The National Drug Code (NDC) identifier of the drug</p>
</dd>
<dt>ndc-product-code</dt><dd><p>The National Drug Code (NDC) product code from the
FDA National Drug Code directory.</p>
</dd>
<dt>dpd-id</dt><dd><p>Drug Product Database (DPD) identification number (a.k.a. DIN)
from the Canadian Drug Product Database. Only present for drugs that are
marketed in Canada</p>
</dd>
<dt>ema-product-code</dt><dd><p>EMA product code from the European Medicines Agency
Database. Only present for products that are authorized by central procedure
for marketing in the European Union.</p>
</dd>
<dt>ema-ma-number</dt><dd><p>EMA marketing authorization number from the European
Medicines Agency Database. Only present for products that are authorized by
central procedure for marketing in the European Union.</p>
</dd>
<dt>started-marketing-on</dt><dd><p>The starting date for market approval.</p>
</dd>
<dt>ended-marketing-on</dt><dd><p>The ending date for market approval.</p>
</dd>
<dt>dosage-form	</dt><dd><p>The pharmaceutical formulation by which the drug is
introduced into the body.</p>
</dd>
<dt>strength</dt><dd><p>The amount of active drug ingredient provided in the dosage</p>
</dd>
<dt>route</dt><dd><p>The path by which the drug or product is taken into the body</p>
</dd>
<dt>fda-application-number</dt><dd><p>The New Drug Application [NDA] number
assigned to this drug by the FDA.</p>
</dd>
<dt>over-the-counter</dt><dd><p>A list of Over The Counter (OTC) forms of the drug.</p>
</dd>
<dt>generic</dt><dd><p>Whether this product is a generic drug.</p>
</dd>
<dt>approved</dt><dd><p>Indicates whether this drug has been approved by the
regulating government.</p>
</dd>
<dt>country</dt><dd><p>The country where this commercially available drug has been
approved.</p>
</dd>
<dt>source</dt><dd><p>Source of this product information. For example, a value of
DPD indicates this information was retrieved from the Canadian Drug Product
Database.</p>
</dd>
<dt>standing</dt><dd><p>One of good, discordant, or deprecated. Distinguishes
products with up to date ingredient information (good) from products with
conflicting information (discordant) or products that have been removed from
an active label (deprecated).</p>
</dd>
<dt>standing-updated-on</dt><dd><p>The date on which the standing was last updated</p>
</dd>
<dt>standing-reason</dt><dd><p>Explains the non-good standing of the product.
One of: ingredient_change, code_duplication, invalid, or removed.</p>
</dd>
<dt>jurisdiction-marketing-category	</dt><dd><p>The marketing category of this
product in its jurisdiction</p>
</dd>
<dt>branded</dt><dd><p>Whether this product has a named brand</p>
</dd>
<dt>prescription</dt><dd><p>Whether this product is only available with
a prescription</p>
</dd>
<dt>unapproved</dt><dd><p>Whether this product is not approved in its jurisdiction</p>
</dd>
<dt>vaccine</dt><dd><p>Whether this product is a vaccine</p>
</dd>
<dt>allergenic</dt><dd><p>Whether this product is used in allergenic testing</p>
</dd>
<dt>cosmetic</dt><dd><p>Whether this product is a cosmetic, such as sunscreen</p>
</dd>
<dt>kit</dt><dd><p>Whether this product is a kit composed of multiple distinct
parts</p>
</dd>
<dt>solo</dt><dd><p>Whether this product has only a single active ingredient</p>
</dd>
<dt>available</dt><dd><p>Whether this product can be sold in its jurisdiction</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_reactions'>Drug Reactions Parsers</h2><span id='topic+drug_reactions'></span>

<h3>Description</h3>

<p>Extract the sequential representation of the metabolic reactions that this
drug molecule is involved in. Depending on available information, this may
include metabolizing enzymes, reaction type, substrates, products,
pharmacological activity of metabolites, and a structural representation of
the biochemical reactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_reactions()
</code></pre>


<h3>Value</h3>

<p>a tibble with 5 variables:
</p>

<dl>
<dt>sequence</dt><dd><p>	Reactions are displayed within a numerical sequence</p>
</dd>
<dt>left_drugbank_name</dt><dd><p>The substrate of the reaction. Maybe a drug or a
metabolite.</p>
</dd>
<dt>rightt_drugbank_name</dt><dd><p>	The product of the reaction. Maybe a drug or a
metabolite.</p>
</dd>
<dt>left_drugbank_id</dt><dd></dd>
<dt>right_drugbank_id</dt><dd></dd>
<dt>parent_id</dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_reactions_enzymes'>Drug Reactions Enzymes Parsers</h2><span id='topic+drug_reactions_enzymes'></span>

<h3>Description</h3>

<p>Enzymes involved in metabolizing this drug
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_reactions_enzymes()
</code></pre>


<h3>Value</h3>

<p>a tibble with 3 variables:
</p>

<dl>
<dt>name</dt><dd></dd>
<dt>uniprot-id</dt><dd></dd>
<dt>parent_id</dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_salts'>Drug Salts parser</h2><span id='topic+drug_salts'></span>

<h3>Description</h3>

<p>Available salt forms of the drug. Ions such as hydrochloride, sodium,
and sulfate are often added to the drug molecule to increase solubility,
dissolution, or absorption.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_salts()
</code></pre>


<h3>Value</h3>

<p>a tibble with 1 variables:
</p>

<dl>
<dt>drugbank-id</dt><dd><p>DrugBank identifiers of the available salt form(s).</p>
</dd>
<dt>name</dt><dd><p>Name of the available salt form(s)</p>
</dd>
<dt>unii</dt><dd><p>Unique Ingredient Identifier (UNII) of the available salt
form(s).</p>
</dd>
<dt>cas-number</dt><dd><p>Chemical Abstracts Service (CAS) registry number assigned
to the salt form(s) of the drug.</p>
</dd>
<dt>inchikey</dt><dd><p>IUPAC International Chemical Identifier (InChi) key
identifier for the available salt form(s).</p>
</dd>
<dt>average-mass</dt><dd><p>Average molecular mass: the weighted average of the
isotopic masses of the salt.</p>
</dd>
<dt>monoisotopic-mass</dt><dd><p>The mass of the most abundant isotope of the salt</p>
</dd>
<dt>smiles</dt><dd><p>The simplified molecular-input line-entry system (SMILES) is
a line notation used for describing the structure of chemical species using
short ASCII strings; calculated by ChemAxon.</p>
</dd>
<dt>inchi</dt><dd><p>A prediction of the IUPAC
International Chemical Identifier (InChI); imported by ChemAxon.</p>
</dd>
<dt>formula</dt><dd><p>Indicates the simple numbers of each type of atom within the
molecule; calculated by ChemAxon.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>parent drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_sequences'>Drug Sequences parser</h2><span id='topic+drug_sequences'></span>

<h3>Description</h3>

<p>The amino acid sequence; provided if the drug is a peptide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_sequences()
</code></pre>


<h3>Details</h3>

<p>Describes peptide sequences of biotech drugs
</p>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>sequence</dt><dd><p>a textual representation of the sequence</p>
</dd>
<dt>format</dt><dd><p>Currently, only the FASTA format is used</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_snp_adverse_reactions'>Drug SNP Adverse Drug Reactions parser</h2><span id='topic+drug_snp_adverse_reactions'></span>

<h3>Description</h3>

<p>The adverse drug reactions that may occur as a result of the listed single
nucleotide polymorphisms (SNPs)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_snp_adverse_reactions()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>protein-name</dt><dd><p>Proteins involved in this SNP.</p>
</dd>
<dt>gene-symbol</dt><dd><p>Genes involved in this SNP.</p>
</dd>
<dt>uniprot-id</dt><dd><p>Universal Protein Resource (UniProt) identifiers for
proteins involved in this pathway.</p>
</dd>
<dt>rs-id</dt><dd><p>The SNP Database identifier for this single nucleotide
polymorphism.</p>
</dd>
<dt>allele</dt><dd><p>The alleles associated with the identified SNP.</p>
</dd>
<dt>adverse-reaction</dt><dd></dd>
<dt>description</dt><dd></dd>
<dt>pubmed-id	</dt><dd><p>Reference to PubMed article.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_snp_effects'>Drug SNP Effects parser</h2><span id='topic+drug_snp_effects'></span>

<h3>Description</h3>

<p>A list of single nucleotide polymorphisms (SNPs) relevant to drug activity or
metabolism, and the effects these may have on pharmacological activity.
SNP effects in the patient may require close monitoring, an increase or
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_snp_effects()
</code></pre>


<h3>Value</h3>

<p>a tibble with the following variables:
</p>

<dl>
<dt>protein-name</dt><dd><p>Proteins involved in this SNP.</p>
</dd>
<dt>gene-symbol</dt><dd><p>Genes involved in this SNP.</p>
</dd>
<dt>uniprot-id</dt><dd><p>Universal Protein Resource (UniProt) identifiers for
proteins involved in this pathway.</p>
</dd>
<dt>rs-id</dt><dd><p>	The SNP Database identifier for this single nucleotide
polymorphism.</p>
</dd>
<dt>allele</dt><dd><p>The alleles associated with the identified SNP.</p>
</dd>
<dt>defining-change</dt><dd></dd>
<dt>description</dt><dd><p>A written description of the SNP effects.</p>
</dd>
<dt>pubmed-id	</dt><dd><p>Reference to PubMed article.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='drug_syn'>Drug Synonyms parser</h2><span id='topic+drug_syn'></span>

<h3>Description</h3>

<p>Other names or identifiers that are associated with this drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drug_syn()
</code></pre>


<h3>Value</h3>

<p>a tibble with 3 variables:
</p>

<dl>
<dt>language</dt><dd><p>Names of the drug in languages other than English.</p>
</dd>
<dt>coder</dt><dd><p>Organization or source providing the synonym. For example,
INN indicates the synonym is an International Nonproprietary Name,
while IUPAC indicates the synonym is the nomenclature designated by the
International Union of Pure and Applied Chemistry.</p>
</dd>
<dt><em>drugbank_id</em></dt><dd><p>drugbank id</p>
</dd>
</dl>


<hr>
<h2 id='init_dvobject'>init_dvobject
initialize dvobject</h2><span id='topic+init_dvobject'></span>

<h3>Description</h3>

<p>init_dvobject
initialize dvobject
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_dvobject()
</code></pre>


<h3>Value</h3>

<p>dvobject
</p>

<hr>
<h2 id='links'>Drugs/ Carriers/ Enzymes/ Targets/ Transporters links element parser</h2><span id='topic+links'></span><span id='topic+drugs_links'></span><span id='topic+carriers_links'></span><span id='topic+enzymes_links'></span><span id='topic+targets_links'></span><span id='topic+transporters_links'></span>

<h3>Description</h3>

<p>Return a list of websites that were used as references for
Drugs/ Carriers/ Enzymes/ Targets/ Transporters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugs_links()

carriers_links()

enzymes_links()

targets_links()

transporters_links()
</code></pre>


<h3>Value</h3>

<p>a tibble with 4 variables:
</p>

<dl>
<dt>ref-id</dt><dd><p>Name of the source website</p>
</dd>
<dt>title</dt><dd><p>Identifier for this drug in the given resource</p>
</dd>
<dt>url</dt><dd><p>The url of the website</p>
</dd>
<dt><em>parent_id</em></dt><dd><p>drug/ carrier/ target/ enzyme/ transporter id</p>
</dd>
</dl>


<hr>
<h2 id='parse_cett_node'>Run all CETT related parsers</h2><span id='topic+parse_cett_node'></span>

<h3>Description</h3>

<p>Run all parsers that retrieve carriers, enzymes, targets and transporters
related information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_cett_node(cett_options)
</code></pre>


<h3>Value</h3>

<p>a list of all drugs parsed tibbles
</p>

<hr>
<h2 id='parse_drug_nodes'>Run all drug  related parsers</h2><span id='topic+parse_drug_nodes'></span>

<h3>Description</h3>

<p>Run all parsers that retrieve drugs related information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_drug_nodes(drug_options)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_drug_nodes_+3A_drug_options">drug_options</code></td>
<td>
<p>- list of options to parse</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of all drugs parsed tibbles
</p>

<hr>
<h2 id='parse_references_node'>Drugs/ Carriers/ Enzymes/ Targets/ Transporters references element parser</h2><span id='topic+parse_references_node'></span>

<h3>Description</h3>

<p>Drugs/ Carriers/ Enzymes/ Targets/ Transporters references element parser
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_references_node(references_options)
</code></pre>


<h3>Value</h3>

<p>a list of all references for drugs, carriers, enzymes, targets or
transporters
</p>

<hr>
<h2 id='parseDrugBank'>parseDrugBank</h2><span id='topic+parseDrugBank'></span>

<h3>Description</h3>

<p>parses given DrugBank XML database into a dvobject. dvobject is a list of
data.frames in which each data.frame represents a part of parsed data (i.e
drugs, prices, carriers, ...)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseDrugBank(
  db_path,
  drug_options = NULL,
  parse_salts = FALSE,
  parse_products = FALSE,
  references_options = NULL,
  cett_options = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parseDrugBank_+3A_db_path">db_path</code></td>
<td>
<p><strong>string</strong>, full path for the <strong>DrugBank</strong> xml or
zip file.</p>
</td></tr>
<tr><td><code id="parseDrugBank_+3A_drug_options">drug_options</code></td>
<td>
<p><strong>character vector</strong>, list of sub drug related nodes
names options to parse (default = NULL). Check <code>drug_node_options()</code>
for all available options. If its value is 'NULL' ONLY 'drug_general_information'
will be placed in the returned dvobject.</p>
</td></tr>
<tr><td><code id="parseDrugBank_+3A_parse_salts">parse_salts</code></td>
<td>
<p><strong>boolean</strong>, parse salts info (default = FALSE)</p>
</td></tr>
<tr><td><code id="parseDrugBank_+3A_parse_products">parse_products</code></td>
<td>
<p><strong>boolean</strong>, parse products info (default = FALSE)</p>
</td></tr>
<tr><td><code id="parseDrugBank_+3A_references_options">references_options</code></td>
<td>
<p><strong>character vector</strong>, list of sub references
related nodes names options to parse (default = NULL).
Check <code>references_node_options()</code> for all available options.</p>
</td></tr>
<tr><td><code id="parseDrugBank_+3A_cett_options">cett_options</code></td>
<td>
<p><strong>character vector</strong>, list of sub cett related nodes
names options to parse (default = NULL). Check <code>cett_nodes_options()</code>
for all available options.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dvobject
</p>


<h3>See Also</h3>

<p>Other parsers: 
<code><a href="#topic+cett_nodes_options">cett_nodes_options</a>()</code>,
<code><a href="#topic+drug_node_options">drug_node_options</a>()</code>,
<code><a href="#topic+references_node_options">references_node_options</a>()</code>
</p>

<hr>
<h2 id='read_drugbank_xml_db'>Reads <strong>DrugBank</strong> xml database and load it into memory.</h2><span id='topic+read_drugbank_xml_db'></span>

<h3>Description</h3>

<p><code>read_drugbank_xml_db</code> loads <strong>DrugBank</strong> xml database full tree
into memory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_drugbank_xml_db(db_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_drugbank_xml_db_+3A_db_path">db_path</code></td>
<td>
<p><strong>string</strong>, full path for the
<strong>DrugBank</strong> xml or zip file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions reads <strong>DrugBank</strong> xml database and load it into memory
for later processing. Hence; this method <strong>must</strong> be called before any
other function in the package and it needs to be called one time only.
</p>
<p>It takes one single mandatory argument which is the location of DrugBank db.
</p>


<h3>Value</h3>

<p>loaded DB or NULL
</p>

<hr>
<h2 id='references_node_options'>returns references node valid options.</h2><span id='topic+references_node_options'></span>

<h3>Description</h3>

<p>returns references node valid options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>references_node_options()
</code></pre>


<h3>Value</h3>

<p>list of references valid options
</p>


<h3>See Also</h3>

<p>Other parsers: 
<code><a href="#topic+cett_nodes_options">cett_nodes_options</a>()</code>,
<code><a href="#topic+drug_node_options">drug_node_options</a>()</code>,
<code><a href="#topic+parseDrugBank">parseDrugBank</a>()</code>
</p>

<hr>
<h2 id='show_dvobject_metadata'>init_dvobject
Returns data.frame with two columns (key, value) of dvobject attributes</h2><span id='topic+show_dvobject_metadata'></span>

<h3>Description</h3>

<p>init_dvobject
Returns data.frame with two columns (key, value) of dvobject attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_dvobject_metadata(dvobject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_dvobject_metadata_+3A_dvobject">dvobject</code></td>
<td>
<p>- dvobject list to show related metadata</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
