<!DOCTYPE html><html lang="en"><head><title>Help for package dc3net</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dc3net}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dc3net'>
<p>Differential C3NET</p></a></li>
<li><a href='#dc3netcom'>
<p>Find Common Network</p></a></li>
<li><a href='#dc3netdif'>
<p>Find Differential Network</p></a></li>
<li><a href='#geneNames'>
<p>Gene Names</p></a></li>
<li><a href='#maxofMIM'>
<p>Find Max Partner For Each Gene</p></a></li>
<li><a href='#normalData'>
<p>Example Prostate Cancer Normal Gene Expression Dataset</p></a></li>
<li><a href='#orderList'>
<p>Order The List of Interactions</p></a></li>
<li><a href='#probSameFilt'>
<p>Filter Probes of The Same Gene</p></a></li>
<li><a href='#rankList'>
<p>Find Rank List</p></a></li>
<li><a href='#tumorData'>
<p>Example Prostate Cancer Tumor Gene Expression Dataset</p></a></li>
<li><a href='#uniqNet'>
<p>Unique Network</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inferring Condition-Specific Networks via Differential Network
Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-03-27</td>
</tr>
<tr>
<td>Author:</td>
<td>Gokmen Altay</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gokmen Altay &lt;altaylabs@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs differential network analysis to infer disease specific gene networks.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.3), c3net, igraph, RedeR</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>dc3net</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-03-27 19:46:18 UTC; altay</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-03-27 22:20:23</td>
</tr>
</table>
<hr>
<h2 id='dc3net'>
Differential C3NET
</h2><span id='topic+dc3net'></span>

<h3>Description</h3>

<p>The dc3net is an R package that infers direct physical interactions of differential gene networks from gene expression datasets of multiple conditions. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dc3net(dataT = c(), dataC = c(), probes = c(), genes = c(), method = "cutoff", 
methodValue = 0, itNum = 1, rankDif = 100, percentDif = 0.6, rankdCom = 10, 
percentCom = 0.85, probFiltered = FALSE, visualization = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dc3net_+3A_datat">dataT</code></td>
<td>

<p>Raw microarray gene expression data set or mutual information matrix, e.g. Tumor dataset or tumor mim.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_datac">dataC</code></td>
<td>

<p>Raw microarray gene expression data set or mutual information matrix, e.g. Normal dataset or normal mim.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_probes">probes</code></td>
<td>

<p>A vector of probe names. The size of the vector should be equal to the number of rows in your data sets. 
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_genes">genes</code></td>
<td>

<p>A vector of gene names. The size of the vector should be equal to the number of rows in your data sets. 
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_method">method</code></td>
<td>

<p>Significance test for elimination of nonsignificant edges. The available options for method parameter are &quot;cutoff&quot;, &quot;justp&quot;, &quot;rank&quot;, &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot; and &quot;BY&quot;. The last six parameter, starting from holm are multiple testing correction methods. methodValue and itnum parameters are dependent to method parameter.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_methodvalue">methodValue</code></td>
<td>

<p>If the method is &quot;cutoff&quot;, methodValue can be zero or predefined cut-off value. Zero means that mean of upper triangle will be taken as cutoff. If the method is &quot;justp&quot;, methodValue and itnum (iteration number) parameters are need to be adjusted. For the &quot;justp&quot; method, methodValue corresponds to alpha value, e.g. 0.05. If the method is &quot;rank&quot;, methodValue is the rank number. Rank corresponds to the number of interactions that will be taken as significant starting from the highest MI value. If the selected method is one of the MTC method, then methodValue and itnum parameters are need to be adjusted as it were in &quot;justp&quot; method. 
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_itnum">itNum</code></td>
<td>

<p>Iteration number for &quot;justp&quot; and MTC methods.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_rankdif">rankDif</code></td>
<td>

<p>Predefined rank cut-off parameter for inferring differential network.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_percentdif">percentDif</code></td>
<td>

<p>Predefined rate parameter for inferring differential network.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_rankdcom">rankdCom</code></td>
<td>

<p>Predefined rank cut-off parameter for inferring common network.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_percentcom">percentCom</code></td>
<td>

<p>Predefined rate parameter for inferring common network.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_probfiltered">probFiltered</code></td>
<td>

<p>If probes of the same gene are filtered or not. Setting this option FALSE eliminates the interactions between the probes of the same genes.
</p>
</td></tr>
<tr><td><code id="dc3net_+3A_visualization">visualization</code></td>
<td>

<p>Enables plotting the inferred networks. The visualization parameter takes three values, &quot;0&quot; for disabling the plot, &quot;1&quot; for plotting the differential network and &quot;2&quot; for plotting the common network. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The required inputs of the package are two different gene expression data sets, probe names and gene names. Users can also use pre-computed test and control mutual information (adjacency) matrices as input. Otherwise, the algorithm takes the two data sets and generates the matrices itself. The data sets need to be normalized together (e.g. using RMA) before using in dc3net. If the input data sets are precomputed mutual information matrices, then the algorithm skip this preprocessing step. The MI matrices are square adjacency matrices where the MI value corresponds to the weight of interaction for each gene pair. The diagonals are set to zero to ignore self-interactions. The next step is computing row wise ranked versions of these MI matrices in descending order. Here, rank 1 corresponds to the highest mutual information value in a row of the matrix. This ranked matrices will be used in comparing and filtering the networks at the comparison step. Then C3NET is applied to the test and control MI matrices to infer gene networks of direct physical interactions of test and control datasets independently. 
</p>
<p>In the comparison step of DC3NET, there are four conditions that all must be validated at the same time for an edge to be included in test difnet. Suppose that we check the potential interaction geneA to geneB to be included in difnet or not. As we stated above, we have been computed row wise ranked versions of the MI matrices in descending order. So we know the rank of interaction geneA to geneB in control MI matrix. The first parameter of DC3NET, rankdif, is the predefined cutoff parameter that checks the interaction between geneA and geneB is one of the top ranked interactions in control MI matrix or not. If the rank of geneA and geneB in the ranked control MI matrix is greater than the predefined cutoff parameter, rankdif, then the first condition becomes valid for deciding it as a difnet interaction. rankdif parameter can be adjusted to any value between 1 and number of rows of control MI matrix. If user wants a stricter difnet, then rankdif parameter needs to be adjusted to a greater value.  The second condition is the change in MI value of interaction from geneA to geneB in the control MI matrix. Here, algorithm uses MIdif value as the cutoff parameter. MIdif is defined as percentdif times the maximum MI value of the row of geneA in the control MI matrix. Default value for the percentdif parameter is 0.6. Depends on strictness of the differential network, user can increase or decrease the second cutoff parameter. The previous two conditions compared the interaction of geneA to geneB but we also need to compare the interaction of geneB to geneA. So the algorithm validates the first and second conditions also for the interaction of geneB to geneA. In this example, if four of the conditions are validated, then DC3NET infer this interaction as in test difnet and continue to perform same filtering process for all gene pairs in test network.
</p>
<p>Lets now start to describe the way that the algorithm infers the common network. Common network can be inferred by looking for all the same interactions between test and control network. However, this is a very strict way of inferring common network. So alternatively, one may consider the ranks and MI value decreases in the other data set. More broadly, users may follow the manner of the difnet process described above but change the comparison parameter, rankdif, from greater to less and for the percentdif from less to greater. Additionally, at this time, we only look at one of the two conditions, rather than all the four conditions together, from geneA to geneB or geneB to geneA. In the package, rankdcom, and percentcom parameters correspond to rank difference and MIdif for common network. Finally, DC3NET infers test difnet and common network and save it to the output folder of the package. If the visualization function is turned on, then the package plots the selected inferred network.
</p>


<h3>Value</h3>

<p><code>dc3net</code> returns an environment res that contains the new inferred networks and mutual information matrices. Users can access to output &quot;Differential Network&quot; and &quot;Common Network&quot; by using res$DifNet and res$CommonNet, respectively. Users can also access to computed mutual information matrices of test and control data by using res$mimT and res$mimC. Thus, users can use this precomputed mutual information matrices on the next run of the package to save time.
</p>
<p>Accessing Differential Network Table:
networks &lt;- dc3net(tumorData, normalData, probes, geneNames)
networks$DifNet
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017.
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
Altay,G., Altay,N., Neal,D. (2013) Global assessment of network inference algorithms based on available literature of gene/protein interactions. Turkish Journal of Biology. 37:547-555.
Castro MA, Wang X, Fletcher MN, Meyer KB and Markowetz F (2012). RedeR: R/Bioconductor package for representing modular structures, nested networks and multiple levels of hierarchical associations. Genome Biology, 13(4), pp. R29.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3netdif">dc3netdif</a></code>, <code><a href="#topic+dc3netcom">dc3netcom</a></code>, <code><a href="#topic+maxofMIM">maxofMIM</a></code>, <code><a href="#topic+orderList">orderList</a></code>, <code><a href="#topic+probSameFilt">probSameFilt</a></code>, <code><a href="#topic+rankList">rankList</a></code>, <code><a href="#topic+uniqNet">uniqNet</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  data(tumorData)
  
  data(normalData)

  data(geneNames)
  
  probes &lt;- rownames(tumorData)
  
  networks &lt;- dc3net(tumorData, normalData, probes, geneNames, visualization=0) 
  
  networks$DifNet ### Differential Network
  
  networks$CommonNet ### Common Network
  
</code></pre>

<hr>
<h2 id='dc3netcom'>
Find Common Network
</h2><span id='topic+dc3netcom'></span>

<h3>Description</h3>

<p><code>dc3netcom</code> finds common network list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dc3netcom(netTemp, vals, rankdif, percent)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dc3netcom_+3A_nettemp">netTemp</code></td>
<td>
<p>Test network, e.g. tumor network.</p>
</td></tr>
<tr><td><code id="dc3netcom_+3A_vals">vals</code></td>
<td>
<p>Statistics in the other (control) condition.</p>
</td></tr>
<tr><td><code id="dc3netcom_+3A_rankdif">rankdif</code></td>
<td>
<p>Predefined rank cut-off parameter.</p>
</td></tr>
<tr><td><code id="dc3netcom_+3A_percent">percent</code></td>
<td>
<p>Predefined rate parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>dc3netcom</code> finds differential network list. 
</p>


<h3>Value</h3>

<p><code>dc3dif</code> returns common network data matrix. The data matrix includes differential network list with gene names, mim indexes, probe names, control indexes, control mutual information rates and control ranks.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017.
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>
</p>

<hr>
<h2 id='dc3netdif'>
Find Differential Network
</h2><span id='topic+dc3netdif'></span>

<h3>Description</h3>

<p><code>dc3netdif</code> finds differential network list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dc3netdif(netTemp, vals, rankdif, percent)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dc3netdif_+3A_nettemp">netTemp</code></td>
<td>
<p>Test network, e.g. tumor network.</p>
</td></tr>
<tr><td><code id="dc3netdif_+3A_vals">vals</code></td>
<td>
<p>Statistics in the other (control) condition.</p>
</td></tr>
<tr><td><code id="dc3netdif_+3A_rankdif">rankdif</code></td>
<td>
<p>Predefined rank cut-off parameter.</p>
</td></tr>
<tr><td><code id="dc3netdif_+3A_percent">percent</code></td>
<td>
<p>Predefined rate parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>dc3netdif</code> finds differential network list. 
</p>


<h3>Value</h3>

<p><code>dc3netdif</code> returns differential network data matrix. The data matrix includes differential network list with gene names, mim indexes, probe names, control indexes, control mutual information rates and control ranks.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017. 
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>
</p>

<hr>
<h2 id='geneNames'>
Gene Names
</h2><span id='topic+geneNames'></span>

<h3>Description</h3>

<p>A vector of gene names. The size of the vector should be equal to the number of rows in your data sets. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("geneNames")</code></pre>


<h3>Format</h3>

<p>A data frame with 0 observations on the following 2 variables.
</p>

<dl>
<dt><code>x</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>y</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://www.broadinstitute.org/cgi-bin/cancer/datasets.cgi
</p>


<h3>References</h3>

<p>Singh,D., Febbo,P.G., Ross,K., et al. (2002) Gene expression correlates of clinical prostatecancer behavior. Cancer cell. 1:203-9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(geneNames)

</code></pre>

<hr>
<h2 id='maxofMIM'>
Find Max Partner For Each Gene
</h2><span id='topic+maxofMIM'></span>

<h3>Description</h3>

<p><code>maxofMIM</code> takes the mutual information matrix as input and find max partner gene with highest mutual information value for each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maxofMIM(mim)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maxofMIM_+3A_mim">mim</code></td>
<td>
<p>A symmetric square mutual information matrix, where the elements (i,j) correspond to the mutual information <code class="reqn">I(i,j)</code> or <code class="reqn">I(j,i)</code> between variables i and j.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>maxofMIM</code> scans each row of the mutual information matrix and find the column with the highest mutual information value for each row. Thus, it finds max partner gene with highest mutual information value for each gene.
</p>


<h3>Value</h3>

<p><code>maxofMIM</code> returns a data matrix that rows are probe/gene names and columns are mutual information values, row numbers of each gene and row numbers of partner gene with highest mutual information value, respectively.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017. 
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>, <code><a href="#topic+orderList">orderList</a></code>,
</p>

<hr>
<h2 id='normalData'>
Example Prostate Cancer Normal Gene Expression Dataset
</h2><span id='topic+normalData'></span>

<h3>Description</h3>

<p>Raw microarray gene expression data set. This data is randomly selected 500 gene subset of the original prostate cancer data set taken from Broad Institute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("normalData")</code></pre>


<h3>Details</h3>

<p>The example data setnis also given to provide a sample data set so that one can take it as reference while formatting the new data set as entry to dc3net. Data sets must be complete and no empty or non-numerical character allowed. 
</p>


<h3>Source</h3>

<p>This data was taken from http://www.broadinstitute.org/cgi-bin/cancer/datasets.cgi
</p>


<h3>References</h3>

<p>Singh,D., Febbo,P.G., Ross,K., et al. (2002) Gene expression correlates of clinical prostatecancer behavior. Cancer cell. 1:203-9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(normalData)

</code></pre>

<hr>
<h2 id='orderList'>
Order The List of Interactions
</h2><span id='topic+orderList'></span>

<h3>Description</h3>

<p><code>orderList</code> takes the data matrix generated by maxofMIM function and sort it according to mutual information values in increasing or decreasing order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orderList(mylist, colindx=1, decreasing=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="orderList_+3A_mylist">mylist</code></td>
<td>
<p>The data matrix generated by maxofMIM function that rows are probe/gene names and columns are mutual information values, row numbers of each gene and row numbers of partner gene with highest mutual information value, respectively. </p>
</td></tr>
<tr><td><code id="orderList_+3A_colindx">colindx</code></td>
<td>
<p>The number of column with mutual information values</p>
</td></tr>
<tr><td><code id="orderList_+3A_decreasing">decreasing</code></td>
<td>
<p>The order of sort. If the argument is set to FALSE, then the data matrix will be sort in increasing order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>orderList</code> takes the data matrix generated by maxofMIM function and sort it according to the column number stated in colindx argument. The direction of order is set by decreasing argument.
</p>


<h3>Value</h3>

<p><code>orderList</code> returns a sorted data matrix that rows are probe/gene names and columns are mutual information values, row numbers of each gene and row numbers of partner gene with highest mutual information value, respectively.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application  in LnCap prostate cancer&quot;, bioRxiv, 2017. 
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>, <code><a href="#topic+maxofMIM">maxofMIM</a></code>,
</p>

<hr>
<h2 id='probSameFilt'>
Filter Probes of The Same Gene
</h2><span id='topic+probSameFilt'></span>

<h3>Description</h3>

<p><code>probSameFilt</code> takes the mutual information matrix as input and set the mutual information values of same genes to zero. Thus, it eliminates the interactions between the probes of the same gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>probSameFilt(mim, genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="probSameFilt_+3A_mim">mim</code></td>
<td>
<p>A symmetric square mutual information matrix, where the elements (i,j) correspond to the mutual information <code class="reqn">I(i,j)</code> or <code class="reqn">I(j,i)</code> between variables i and j.
</p>
</td></tr>
<tr><td><code id="probSameFilt_+3A_genes">genes</code></td>
<td>

<p>A vector of gene names
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The use of <code>probSameFilt</code> is very important when working on probe level. If this function is not enabled, then the interactions between probes of the same gene, which should not be taken into account, will be included in differential or common network computation. This may result in difnet or commonnet tables which includes same gene interactions e.g. GPR50-GPR50.
</p>


<h3>Value</h3>

<p><code>probSameFilt</code> returns mutual information matrix that the interaction scores between probes of the same gene is updated to 0.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017. 
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(tumorData)

  data(geneNames)
  
  tumorData &lt;- copula(tumorData) 
      
  tumorData &lt;- makemim(tumorData)

  probSameFilt(tumorData, geneNames)

</code></pre>

<hr>
<h2 id='rankList'>
Find Rank List
</h2><span id='topic+rankList'></span>

<h3>Description</h3>

<p><code>rankList</code> finds the rank of test list in the mutual information matrix of control to see any rank difference using mutual information matrix indexes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankList(topTest, mimCont)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rankList_+3A_toptest">topTest</code></td>
<td>
<p>A data matrix that has columns 4 and 5 have mim indexes</p>
</td></tr>
<tr><td><code id="rankList_+3A_mimcont">mimCont</code></td>
<td>
<p>A mutual information matrix of control dataset</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>rankList</code> takes mim indexes from 4th and 5th columns of the data matrix and finds the rank of test list in the mutual information matrix of control to see any rank difference.
</p>


<h3>Value</h3>

<p><code>rankList</code> returns a data matrix that has ranks of test list in the control MI matrix
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>, <code><a href="#topic+maxofMIM">maxofMIM</a></code>,
</p>

<hr>
<h2 id='tumorData'>
Example Prostate Cancer Tumor Gene Expression Dataset
</h2><span id='topic+tumorData'></span>

<h3>Description</h3>

<p>Raw microarray gene expression data set. This data is randomly selected 500 gene subset of the original prostate cancer data set taken from Broad Institute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("tumorData")</code></pre>


<h3>Details</h3>

<p>The example data set is also given to provide a sample data set so that one can take it as reference while formatting the new data set as entry to dc3net. Data sets must be complete and no empty or non-numerical character allowed. 
</p>


<h3>Source</h3>

<p>This data was taken from http://www.broadinstitute.org/cgi-bin/cancer/datasets.cgi
</p>


<h3>References</h3>

<p>Singh,D., Febbo,P.G., Ross,K., et al. (2002) Gene expression correlates of clinical prostatecancer behavior. Cancer cell. 1:203-9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(tumorData)

</code></pre>

<hr>
<h2 id='uniqNet'>
Unique Network
</h2><span id='topic+uniqNet'></span>

<h3>Description</h3>

<p><code>uniqNet</code> makes the differential or common network list unique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniqNet(tops = c())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniqNet_+3A_tops">tops</code></td>
<td>
<p>A data matrix that contains list of interactions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>uniqNet</code> removes the redundant interactions in the generated differential or common network. For ex. it is possible that the generated network can include both geneA-to-geneB and geneB-to-geneA interactions. <code>uniqNet</code> eliminates one of this interaction so the list of final network is unique.
</p>


<h3>Value</h3>

<p><code>uniqNet</code> returns a unique data matrix that includes network list with gene names, mim indexes, probe names, control indexes, control mutual information rates and control ranks.
</p>


<h3>Author(s)</h3>

<p>Gokmen Altay
</p>


<h3>References</h3>

<p>Altay, G. et. al. &quot;Genome-wide differential gene network analysis software and an application in LnCap prostate cancer&quot;, bioRxiv, 2017. 
Altay, G., Asim, M., Markowetz, F., Neal, D.E. (2011) Differential C3NET reveals disease netwokrs of direct physical interactions. BMC Bioinformatics. 12: 296.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dc3net">dc3net</a></code>, <code><a href="#topic+maxofMIM">maxofMIM</a></code>, <code><a href="#topic+orderList">orderList</a></code>, <code><a href="#topic+rankList">rankList</a></code>, <code><a href="#topic+dc3netdif">dc3netdif</a></code>, <code><a href="#topic+dc3netcom">dc3netcom</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
