<!DOCTYPE html><html><head><title>Help for package DCLEAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DCLEAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_deletion'><p>add_deletion</p></a></li>
<li><a href='#add_dropout'><p>add_dropout</p></a></li>
<li><a href='#as_igraph'><p>Generic function for as_igraph</p></a></li>
<li><a href='#as_igraph+2Cdata.frame-method'><p>as_igraph</p></a></li>
<li><a href='#as_igraph+2Cphylo-method'><p>as_igraph</p></a></li>
<li><a href='#as_lineage_tree'><p>Generic function for as_lineage_tree</p></a></li>
<li><a href='#as_lineage_tree+2CphyDat+2Cphylo+2Clineage_tree_config-method'><p>as_lineage_tree</p></a></li>
<li><a href='#as_phylo'><p>Generic function for as_phylo</p></a></li>
<li><a href='#as_phylo+2Cigraph-method'><p>as_phylo</p></a></li>
<li><a href='#DCLEAR'><p>DCLEAR: A package for DCLEAR: Distance based Cell LinEAge Reconstruction</p></a></li>
<li><a href='#dist_kmer_replacement_inference'><p>Core function of computing kmer replacement distance</p></a></li>
<li><a href='#dist_replacement'><p>Generic function for dist_replacement</p></a></li>
<li><a href='#dist_replacement+2CphyDat+2Ckmer_summary+2Cinteger-method'><p>Compute the kmer replacement distance</p></a></li>
<li><a href='#dist_replacement+2CphyDat+2Cmissing+2Cinteger-method'><p>Compute the kmer replacement distance</p></a></li>
<li><a href='#dist_weighted_hamming'><p>Generic function for dist_weighted_hamming</p></a></li>
<li><a href='#dist_weighted_hamming+2CphyDat+2Cnumeric-method'><p>dist_weighted_hamming</p></a></li>
<li><a href='#downsample'><p>Generic function for downsample</p></a></li>
<li><a href='#downsample+2Cigraph-method'><p>downsample</p></a></li>
<li><a href='#downsample+2Clineage_tree-method'><p>downsample</p></a></li>
<li><a href='#get_distance_prior'><p>get_distance_prior</p></a></li>
<li><a href='#get_leaves'><p>Generic function for get_leaves</p></a></li>
<li><a href='#get_leaves+2Clineage_tree-method'><p>get_leaves</p></a></li>
<li><a href='#get_node_names'><p>get_node_names</p></a></li>
<li><a href='#get_replacement_probability'><p>get_replacement_probability</p></a></li>
<li><a href='#get_sequence'><p>get_sequence</p></a></li>
<li><a href='#get_transition_probability'><p>get_transition_probability</p></a></li>
<li><a href='#lineages'><p>Lineage data</p></a></li>
<li><a href='#positional_mutation_prob'><p>positional_mutation_prob</p></a></li>
<li><a href='#process_sequence'><p>Generic function for process_sequence</p></a></li>
<li><a href='#process_sequence+2CphyDat-method'><p>Process sequences</p></a></li>
<li><a href='#prune'><p>Generic function for prune</p></a></li>
<li><a href='#prune+2Cigraph-method'><p>prune</p></a></li>
<li><a href='#prune+2Clineage_tree-method'><p>prune</p></a></li>
<li><a href='#random_tree'><p>random_tree</p></a></li>
<li><a href='#rbind+2CphyDat-method'><p>rbind</p></a></li>
<li><a href='#sample_mutation_outcome'><p>sample_mutation_outcome</p></a></li>
<li><a href='#sample_mutation_site'><p>sample_mutation_site</p></a></li>
<li><a href='#sample_outcome_prob'><p>sample_outcome_prob</p></a></li>
<li><a href='#score_simulation'><p>score_simulation</p></a></li>
<li><a href='#sim_seqdata'><p>sim_seqdata</p></a></li>
<li><a href='#simulate'><p>Generic function for simulate</p></a></li>
<li><a href='#simulate_core'><p>simulate_core</p></a></li>
<li><a href='#simulate+2Clineage_tree_config+2Cmissing-method'><p>simulate</p></a></li>
<li><a href='#simulate+2Clineage_tree_config+2CphyDat-method'><p>simulate</p></a></li>
<li><a href='#substr_kmer'><p>Generic function for substr_kmer</p></a></li>
<li><a href='#substr_kmer+2Ckmer_summary-method'><p>Subseting a kmer_summary object</p></a></li>
<li><a href='#subtract'><p>Generic function for subtract</p></a></li>
<li><a href='#subtract+2Clineage_tree+2Clineage_tree-method'><p>subtract</p></a></li>
<li><a href='#subtree'><p>Generic function for subtree</p></a></li>
<li><a href='#subtree+2Clineage_tree-method'><p>subtree</p></a></li>
<li><a href='#subtree+2Cphylo-method'><p>subtree</p></a></li>
<li><a href='#summarize_kmer'><p>Generic function for summarize_kmer</p></a></li>
<li><a href='#summarize_kmer_core'><p>summarize_kmer_core</p></a></li>
<li><a href='#summarize_kmer+2CphyDat-method'><p>summarize_kmer</p></a></li>
<li><a href='#WH'><p>WH</p></a></li>
<li><a href='#WH_train'><p>Train weights for WH</p></a></li>
<li><a href='#WH_train_fit'><p>Train weights for WH, and output distance object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.13</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-9-14</td>
</tr>
<tr>
<td>Title:</td>
<td>Distance Based Cell Lineage Reconstruction</td>
</tr>
<tr>
<td>Author:</td>
<td>Il-Youp Kwak [aut, cre],
  Wuming Gong [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Il-Youp Kwak &lt;ikwak2@cau.ac.kr&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), tensorflow(&ge; 2.2.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocParallel, dplyr, Matrix, matrixStats, ape, phangorn, Rcpp,
igraph, methods, purrr, stringr, tidyr, rBayesianOptimization,
rlang, BiocGenerics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>Description:</td>
<td>R codes for distance based cell lineage reconstruction. Our methods won both sub-challenges 2 and 3 of the Allen Institute Cell Lineage Reconstruction DREAM Challenge in 2020. 
        References: Gong et al. (2021) &lt;<a href="https://doi.org/10.1016%2Fj.cels.2021.05.008">doi:10.1016/j.cels.2021.05.008</a>&gt;, Gong et al. (2022) &lt;<a href="https://doi.org/10.1186%2Fs12859-022-04633-x">doi:10.1186/s12859-022-04633-x</a>&gt;. </td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ikwak2/DCLEAR">https://github.com/ikwak2/DCLEAR</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-14 07:01:28 UTC; ikwak2</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-14 07:32:35 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_deletion'>add_deletion</h2><span id='topic+add_deletion'></span>

<h3>Description</h3>

<p>Add deletion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_deletion(x, tree, mutation_site, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_deletion_+3A_x">x</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="add_deletion_+3A_tree">tree</code></td>
<td>
<p>a matrix representing the lineage tree</p>
</td></tr>
<tr><td><code id="add_deletion_+3A_mutation_site">mutation_site</code></td>
<td>
<p>a binary matrix for mutation site</p>
</td></tr>
<tr><td><code id="add_deletion_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character matrix with deletions
</p>

<hr>
<h2 id='add_dropout'>add_dropout</h2><span id='topic+add_dropout'></span>

<h3>Description</h3>

<p>Add dropout events
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_dropout(x, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_dropout_+3A_x">x</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="add_dropout_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character matrix with dropout events
</p>

<hr>
<h2 id='as_igraph'>Generic function for as_igraph</h2><span id='topic+as_igraph'></span>

<h3>Description</h3>

<p>Generic function for as_igraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_igraph(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_igraph_+3A_x">x</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="as_igraph_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='as_igraph+2Cdata.frame-method'>as_igraph</h2><span id='topic+as_igraph+2Cdata.frame-method'></span>

<h3>Description</h3>

<p>Convert an phylo object to an igraph object, while keeping the weight (in contrast to igraph::as.igraph)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'data.frame'
as_igraph(x, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_igraph+2B2Cdata.frame-method_+3A_x">x</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="as_igraph+2B2Cdata.frame-method_+3A_config">config</code></td>
<td>
<p>a 'lineage_tree_config' object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an igraph object
</p>

<hr>
<h2 id='as_igraph+2Cphylo-method'>as_igraph</h2><span id='topic+as_igraph+2Cphylo-method'></span>

<h3>Description</h3>

<p>Convert an phylo object to an igraph object, while keeping the weight (in contrast to igraph::as.igraph)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phylo'
as_igraph(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_igraph+2B2Cphylo-method_+3A_x">x</code></td>
<td>
<p>a phylo object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an igraph object
</p>

<hr>
<h2 id='as_lineage_tree'>Generic function for as_lineage_tree</h2><span id='topic+as_lineage_tree'></span>

<h3>Description</h3>

<p>Generic function for as_lineage_tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_lineage_tree(x, y, config, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_lineage_tree_+3A_x">x</code></td>
<td>
<p>a phyDat object</p>
</td></tr>
<tr><td><code id="as_lineage_tree_+3A_y">y</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="as_lineage_tree_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="as_lineage_tree_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='as_lineage_tree+2CphyDat+2Cphylo+2Clineage_tree_config-method'>as_lineage_tree</h2><span id='topic+as_lineage_tree+2CphyDat+2Cphylo+2Clineage_tree_config-method'></span>

<h3>Description</h3>

<p>Convert a phylo object and a phyDat object to a lineage_tree object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat,phylo,lineage_tree_config'
as_lineage_tree(x, y, config, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_lineage_tree+2B2CphyDat+2B2Cphylo+2B2Clineage_tree_config-method_+3A_x">x</code></td>
<td>
<p>a phyDat object</p>
</td></tr>
<tr><td><code id="as_lineage_tree+2B2CphyDat+2B2Cphylo+2B2Clineage_tree_config-method_+3A_y">y</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="as_lineage_tree+2B2CphyDat+2B2Cphylo+2B2Clineage_tree_config-method_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="as_lineage_tree+2B2CphyDat+2B2Cphylo+2B2Clineage_tree_config-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>

<hr>
<h2 id='as_phylo'>Generic function for as_phylo</h2><span id='topic+as_phylo'></span>

<h3>Description</h3>

<p>Generic function for as_phylo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_phylo(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_phylo_+3A_x">x</code></td>
<td>
<p>a graph object</p>
</td></tr>
<tr><td><code id="as_phylo_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='as_phylo+2Cigraph-method'>as_phylo</h2><span id='topic+as_phylo+2Cigraph-method'></span>

<h3>Description</h3>

<p>Convert an igraph object to a phylo object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'igraph'
as_phylo(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_phylo+2B2Cigraph-method_+3A_x">x</code></td>
<td>
<p>an igraph object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a phylo object or a igraph object
</p>

<hr>
<h2 id='DCLEAR'>DCLEAR: A package for DCLEAR: Distance based Cell LinEAge Reconstruction</h2><span id='topic+DCLEAR'></span>

<h3>Description</h3>

<p>Distance based methods for inferring lineage treess from single cell data
</p>

<hr>
<h2 id='dist_kmer_replacement_inference'>Core function of computing kmer replacement distance</h2><span id='topic+dist_kmer_replacement_inference'></span>

<h3>Description</h3>

<p>Compute the sequence distance matrix using inferred kmer replacement matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_kmer_replacement_inference(x, kmer_summary, k = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_kmer_replacement_inference_+3A_x">x</code></td>
<td>
<p>input data in phyDat format</p>
</td></tr>
<tr><td><code id="dist_kmer_replacement_inference_+3A_kmer_summary">kmer_summary</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
<tr><td><code id="dist_kmer_replacement_inference_+3A_k">k</code></td>
<td>
<p>k-mers (default k=2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dist object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='dist_replacement'>Generic function for dist_replacement</h2><span id='topic+dist_replacement'></span>

<h3>Description</h3>

<p>Generic function for dist_replacement
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_replacement(x, kmer_summary, k, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_replacement_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="dist_replacement_+3A_kmer_summary">kmer_summary</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
<tr><td><code id="dist_replacement_+3A_k">k</code></td>
<td>
<p>k-mer length</p>
</td></tr>
<tr><td><code id="dist_replacement_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='dist_replacement+2CphyDat+2Ckmer_summary+2Cinteger-method'>Compute the kmer replacement distance</h2><span id='topic+dist_replacement+2CphyDat+2Ckmer_summary+2Cinteger-method'></span>

<h3>Description</h3>

<p>Compute the kmer replacement distance between sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat,kmer_summary,integer'
dist_replacement(x, kmer_summary, k = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Ckmer_summary+2B2Cinteger-method_+3A_x">x</code></td>
<td>
<p>input data in phyDat format</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Ckmer_summary+2B2Cinteger-method_+3A_kmer_summary">kmer_summary</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Ckmer_summary+2B2Cinteger-method_+3A_k">k</code></td>
<td>
<p>k-mer length</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Ckmer_summary+2B2Cinteger-method_+3A_...">...</code></td>
<td>
<p>other arguments passed to substr_kmer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dist object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='dist_replacement+2CphyDat+2Cmissing+2Cinteger-method'>Compute the kmer replacement distance</h2><span id='topic+dist_replacement+2CphyDat+2Cmissing+2Cinteger-method'></span>

<h3>Description</h3>

<p>Compute the kmer replacement distance between sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat,missing,integer'
dist_replacement(x, kmer_summary, k = 2L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Cmissing+2B2Cinteger-method_+3A_x">x</code></td>
<td>
<p>input data in phyDat format</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Cmissing+2B2Cinteger-method_+3A_kmer_summary">kmer_summary</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Cmissing+2B2Cinteger-method_+3A_k">k</code></td>
<td>
<p>k-mer length</p>
</td></tr>
<tr><td><code id="dist_replacement+2B2CphyDat+2B2Cmissing+2B2Cinteger-method_+3A_...">...</code></td>
<td>
<p>other arguments passed to substr_kmer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dist object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='dist_weighted_hamming'>Generic function for dist_weighted_hamming</h2><span id='topic+dist_weighted_hamming'></span>

<h3>Description</h3>

<p>Generic function for dist_weighted_hamming
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_weighted_hamming(x, wVec, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_weighted_hamming_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="dist_weighted_hamming_+3A_wvec">wVec</code></td>
<td>
<p>weight vector</p>
</td></tr>
<tr><td><code id="dist_weighted_hamming_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='dist_weighted_hamming+2CphyDat+2Cnumeric-method'>dist_weighted_hamming</h2><span id='topic+dist_weighted_hamming+2CphyDat+2Cnumeric-method'></span>

<h3>Description</h3>

<p>implementation of weighted hamming algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat,numeric'
dist_weighted_hamming(x, wVec, dropout = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_weighted_hamming+2B2CphyDat+2B2Cnumeric-method_+3A_x">x</code></td>
<td>
<p>Sequence object of 'phyDat' type.</p>
</td></tr>
<tr><td><code id="dist_weighted_hamming+2B2CphyDat+2B2Cnumeric-method_+3A_wvec">wVec</code></td>
<td>
<p>Weight vector for the calculation of weighted hamming distance</p>
</td></tr>
<tr><td><code id="dist_weighted_hamming+2B2CphyDat+2B2Cnumeric-method_+3A_dropout">dropout</code></td>
<td>
<p>Different weighting strategy is taken to consider interval dropout with dropout = 'TRUE'. Default is, dropout = 'FALSE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Calculated distance matrix of input sequences. The result is a 'dist' class object.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(DCLEAR)
library(phangorn)
library(ape)

set.seed(1)
mu_d1 = c( 30, 20, 10, 5, 5, 1, 0.01, 0.001)
mu_d1 = mu_d1/sum(mu_d1)
simn = 10 # number of cell samples
m = 10  ## number of targets
sD = sim_seqdata(sim_n = simn, m = m, mu_d = 0.03,
      d = 12, n_s = length(mu_d1), outcome_prob = mu_d1, p_d = 0.005 )
## RF score with hamming distance
D_hm = dist.hamming(sD$seqs)
tree_hm = NJ(D_hm)
RF.dist(tree_hm, sD$tree, normalize = TRUE)

## RF score with weighted hamming
InfoW = -log(mu_d1)
InfoW[1:2] = 1
InfoW[3:7] = 4.5
D_wh = dist_weighted_hamming(sD$seqs, InfoW, dropout = FALSE)
tree_wh = NJ(D_wh)
RF.dist(tree_wh, sD$tree, normalize = TRUE)

## RF score with weighted hamming, cosidering dropout situation
nfoW = -log(mu_d1)
InfoW[1] = 1
InfoW[2] = 12
InfoW[3:7] = 3
D_wh2 = dist_weighted_hamming(sD$seqs, InfoW, dropout = TRUE)
tree_wh2= NJ(D_wh2)
RF.dist(tree_wh2, sD$tree, normalize = TRUE)


</code></pre>

<hr>
<h2 id='downsample'>Generic function for downsample</h2><span id='topic+downsample'></span>

<h3>Description</h3>

<p>Generic function for downsample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downsample(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downsample_+3A_x">x</code></td>
<td>
<p>a data object</p>
</td></tr>
<tr><td><code id="downsample_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='downsample+2Cigraph-method'>downsample</h2><span id='topic+downsample+2Cigraph-method'></span>

<h3>Description</h3>

<p>Sample a lineage tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'igraph'
downsample(x, n = 10L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downsample+2B2Cigraph-method_+3A_x">x</code></td>
<td>
<p>a igraph object</p>
</td></tr>
<tr><td><code id="downsample+2B2Cigraph-method_+3A_n">n</code></td>
<td>
<p>number of leaves (tips) in the down-sampled tree</p>
</td></tr>
<tr><td><code id="downsample+2B2Cigraph-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a phylo object
</p>

<hr>
<h2 id='downsample+2Clineage_tree-method'>downsample</h2><span id='topic+downsample+2Clineage_tree-method'></span>

<h3>Description</h3>

<p>Sample a lineage tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree'
downsample(x, n = 10L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downsample+2B2Clineage_tree-method_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="downsample+2B2Clineage_tree-method_+3A_n">n</code></td>
<td>
<p>number of leaves (tips) in the down-sampled tree</p>
</td></tr>
<tr><td><code id="downsample+2B2Clineage_tree-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>

<hr>
<h2 id='get_distance_prior'>get_distance_prior</h2><span id='topic+get_distance_prior'></span>

<h3>Description</h3>

<p>prior distribution of distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_distance_prior(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_distance_prior_+3A_x">x</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a probabilistic vector of the distribution of nodal distances
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='get_leaves'>Generic function for get_leaves</h2><span id='topic+get_leaves'></span>

<h3>Description</h3>

<p>Generic function for get_leaves
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_leaves(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_leaves_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="get_leaves_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='get_leaves+2Clineage_tree-method'>get_leaves</h2><span id='topic+get_leaves+2Clineage_tree-method'></span>

<h3>Description</h3>

<p>Get the leaf sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree'
get_leaves(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_leaves+2B2Clineage_tree-method_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="get_leaves+2B2Clineage_tree-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a phyDat object
</p>

<hr>
<h2 id='get_node_names'>get_node_names</h2><span id='topic+get_node_names'></span>

<h3>Description</h3>

<p>Convenient function for get node names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_node_names(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_node_names_+3A_x">x</code></td>
<td>
<p>node id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>node names
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='get_replacement_probability'>get_replacement_probability</h2><span id='topic+get_replacement_probability'></span>

<h3>Description</h3>

<p>Compute p(A,B|d), the conditional probability of seeing a replacement of from kmer A to B or vice versa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_replacement_probability(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_replacement_probability_+3A_x">x</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an 3D probabilistic array (kmers by kmers by distances)
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='get_sequence'>get_sequence</h2><span id='topic+get_sequence'></span>

<h3>Description</h3>

<p>Get sequencees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sequence(x, tree, outcome, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_sequence_+3A_x">x</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_tree">tree</code></td>
<td>
<p>a matrix representing the lineage tree</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_outcome">outcome</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="get_sequence_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character matrix
</p>

<hr>
<h2 id='get_transition_probability'>get_transition_probability</h2><span id='topic+get_transition_probability'></span>

<h3>Description</h3>

<p>Compute p(A,X|B,Y,d), the conditional probability of seeing a replacement from A to B
given the previous replacement B from Y at nodal distance d
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_transition_probability(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_transition_probability_+3A_x">x</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an 3D probabilistic array (kmers by kmers by distances)
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='lineages'>Lineage data</h2><span id='topic+lineages'></span>

<h3>Description</h3>

<p>Lineage data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lineages)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 100.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lineages)
</code></pre>

<hr>
<h2 id='positional_mutation_prob'>positional_mutation_prob</h2><span id='topic+positional_mutation_prob'></span>

<h3>Description</h3>

<p>Convenient function for get node names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>positional_mutation_prob(x, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="positional_mutation_prob_+3A_x">x</code></td>
<td>
<p>a phyDat object</p>
</td></tr>
<tr><td><code id="positional_mutation_prob_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a positional mutation probability matrix
</p>

<hr>
<h2 id='process_sequence'>Generic function for process_sequence</h2><span id='topic+process_sequence'></span>

<h3>Description</h3>

<p>Generic function for process_sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_sequence(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_sequence_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="process_sequence_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='process_sequence+2CphyDat-method'>Process sequences</h2><span id='topic+process_sequence+2CphyDat-method'></span>

<h3>Description</h3>

<p>Process sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat'
process_sequence(
  x,
  division = 16L,
  dropout_character = "*",
  default_character = "0",
  deletion_character = "-"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_sequence+2B2CphyDat-method_+3A_x">x</code></td>
<td>
<p>input data in phyDat format</p>
</td></tr>
<tr><td><code id="process_sequence+2B2CphyDat-method_+3A_division">division</code></td>
<td>
<p>cell division</p>
</td></tr>
<tr><td><code id="process_sequence+2B2CphyDat-method_+3A_dropout_character">dropout_character</code></td>
<td>
<p>Dropout character (default: '*')</p>
</td></tr>
<tr><td><code id="process_sequence+2B2CphyDat-method_+3A_default_character">default_character</code></td>
<td>
<p>Default character (default: '0')</p>
</td></tr>
<tr><td><code id="process_sequence+2B2CphyDat-method_+3A_deletion_character">deletion_character</code></td>
<td>
<p>Deletion character (default: '-')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'lineage_tree_config' object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='prune'>Generic function for prune</h2><span id='topic+prune'></span>

<h3>Description</h3>

<p>Generic function for prune
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prune(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="prune_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='prune+2Cigraph-method'>prune</h2><span id='topic+prune+2Cigraph-method'></span>

<h3>Description</h3>

<p>Trim a full lineage tree into phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'igraph'
prune(x, weighted = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune+2B2Cigraph-method_+3A_x">x</code></td>
<td>
<p>an igraph object</p>
</td></tr>
<tr><td><code id="prune+2B2Cigraph-method_+3A_weighted">weighted</code></td>
<td>
<p>whether or not keep the edge weight (default: TRUE)</p>
</td></tr>
<tr><td><code id="prune+2B2Cigraph-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an igraph object
</p>

<hr>
<h2 id='prune+2Clineage_tree-method'>prune</h2><span id='topic+prune+2Clineage_tree-method'></span>

<h3>Description</h3>

<p>Trim a full lineage tree into phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree'
prune(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune+2B2Clineage_tree-method_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="prune+2B2Clineage_tree-method_+3A_...">...</code></td>
<td>
<p>additional parameters passed to as_phylo()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>

<hr>
<h2 id='random_tree'>random_tree</h2><span id='topic+random_tree'></span>

<h3>Description</h3>

<p>Simulate a random lineage tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_tree(n_samples, division = 16L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random_tree_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples to simulate</p>
</td></tr>
<tr><td><code id="random_tree_+3A_division">division</code></td>
<td>
<p>number of cell division</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='rbind+2CphyDat-method'>rbind</h2><span id='topic+rbind+2CphyDat-method'></span>

<h3>Description</h3>

<p>Concatenate multiple phyDat objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat'
rbind(..., deparse.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind+2B2CphyDat-method_+3A_...">...</code></td>
<td>
<p>a list of phyDat objects</p>
</td></tr>
<tr><td><code id="rbind+2B2CphyDat-method_+3A_deparse.level">deparse.level</code></td>
<td>
<p>see definition in generic rbind</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a phyDat object
</p>

<hr>
<h2 id='sample_mutation_outcome'>sample_mutation_outcome</h2><span id='topic+sample_mutation_outcome'></span>

<h3>Description</h3>

<p>Sample mutation outcome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_mutation_outcome(x, mp = NULL, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_mutation_outcome_+3A_x">x</code></td>
<td>
<p>an igraph object</p>
</td></tr>
<tr><td><code id="sample_mutation_outcome_+3A_mp">mp</code></td>
<td>
<p>a mutation site matrix</p>
</td></tr>
<tr><td><code id="sample_mutation_outcome_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a outcome matrix
</p>

<hr>
<h2 id='sample_mutation_site'>sample_mutation_site</h2><span id='topic+sample_mutation_site'></span>

<h3>Description</h3>

<p>Sample mutation site
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_mutation_site(tree, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_mutation_site_+3A_tree">tree</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="sample_mutation_site_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a mutation site matrix
</p>

<hr>
<h2 id='sample_outcome_prob'>sample_outcome_prob</h2><span id='topic+sample_outcome_prob'></span>

<h3>Description</h3>

<p>Sampling outcome probability based on a gamma distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_outcome_prob(config, num_states = 20L, shape = 0.1, scale = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_outcome_prob_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="sample_outcome_prob_+3A_num_states">num_states</code></td>
<td>
<p>number of states used in simulation.</p>
</td></tr>
<tr><td><code id="sample_outcome_prob_+3A_shape">shape</code></td>
<td>
<p>shape parameter in gamma distribution</p>
</td></tr>
<tr><td><code id="sample_outcome_prob_+3A_scale">scale</code></td>
<td>
<p>scale parameter in gamma distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a probability vector for each alphabet
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='score_simulation'>score_simulation</h2><span id='topic+score_simulation'></span>

<h3>Description</h3>

<p>Compare two sets of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_simulation(x, y, config)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_simulation_+3A_x">x</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="score_simulation_+3A_y">y</code></td>
<td>
<p>a character matrix</p>
</td></tr>
<tr><td><code id="score_simulation_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric scores
</p>

<hr>
<h2 id='sim_seqdata'>sim_seqdata</h2><span id='topic+sim_seqdata'></span>

<h3>Description</h3>

<p>Generate singe cell barcode data set with tree shaped lineage information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_seqdata(
  sim_n = 200,
  m = 200,
  mu_d = 0.03,
  d = 15,
  n_s = 23,
  outcome_prob = NULL,
  p_d = 0.003
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_seqdata_+3A_sim_n">sim_n</code></td>
<td>
<p>Number of cell samples to simulate.</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_m">m</code></td>
<td>
<p>Number of targets.</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_mu_d">mu_d</code></td>
<td>
<p>Mutation rate. (a scalar or a vector)</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_d">d</code></td>
<td>
<p>Number of cell divisions.</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_n_s">n_s</code></td>
<td>
<p>Number of possible outcome states</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_outcome_prob">outcome_prob</code></td>
<td>
<p>Outcome probability vector (default is NULL)</p>
</td></tr>
<tr><td><code id="sim_seqdata_+3A_p_d">p_d</code></td>
<td>
<p>Dropout probability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result is a list containing two objects, 'seqs' and 'tree'. The 'seqs' is 'phyDat' object of 'sim_n' number of simulated barcodes corresponding to each cell, and The 'tree' is a 'phylo' object, a ground truth tree structure for the simulated data.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(DCLEAR)
library(phangorn)
library(ape)

set.seed(1)
mu_d1 = c( 30, 20, 10, 5, 5, 1, 0.01, 0.001)
mu_d1 = mu_d1/sum(mu_d1)
simn = 10 # number of cell samples
m = 10  ## number of targets
sD = sim_seqdata(sim_n = simn, m = m, mu_d = 0.03,
        d = 12, n_s = length(mu_d1), outcome_prob = mu_d1, p_d = 0.005 )
## RF score with hamming distance
D_hm = dist.hamming(sD$seqs)
tree_hm = NJ(D_hm)
RF.dist(tree_hm, sD$tree, normalize = TRUE)

## RF score with weighted hamming
InfoW = -log(mu_d1)
InfoW[1:2] = 1
InfoW[3:7] = 4.5
D_wh = dist_weighted_hamming(sD$seqs, InfoW, dropout=FALSE)
tree_wh = NJ(D_wh)
RF.dist(tree_wh, sD$tree, normalize = TRUE)

## RF score with weighted hamming, cosidering dropout situation
nfoW = -log(mu_d1)
InfoW[1] = 1
InfoW[2] = 12
InfoW[3:7] = 3
D_wh2 = dist_weighted_hamming(sD$seqs, InfoW, dropout = TRUE)
tree_wh2= NJ(D_wh2)
RF.dist(tree_wh2, sD$tree, normalize = TRUE)

</code></pre>

<hr>
<h2 id='simulate'>Generic function for simulate</h2><span id='topic+simulate'></span>

<h3>Description</h3>

<p>Generic function for simulate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate(config, x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_+3A_config">config</code></td>
<td>
<p>a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="simulate_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="simulate_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='simulate_core'>simulate_core</h2><span id='topic+simulate_core'></span>

<h3>Description</h3>

<p>Simulate a cell lineage tree
Adoped from https://github.com/elifesciences-publications/CRISPR_recorders_sims/blob/master/MATLAB_sims/GESTALT_30hr_1x_simulation.m
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_core(config, tree, mutation_site, outcome)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_core_+3A_config">config</code></td>
<td>
<p>simulation configuration; a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="simulate_core_+3A_tree">tree</code></td>
<td>
<p>a matrix representing the lineage tree</p>
</td></tr>
<tr><td><code id="simulate_core_+3A_mutation_site">mutation_site</code></td>
<td>
<p>a binary matrix indicating the mutation sites</p>
</td></tr>
<tr><td><code id="simulate_core_+3A_outcome">outcome</code></td>
<td>
<p>a character matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'lineage_tree' object
</p>

<hr>
<h2 id='simulate+2Clineage_tree_config+2Cmissing-method'>simulate</h2><span id='topic+simulate+2Clineage_tree_config+2Cmissing-method'></span>

<h3>Description</h3>

<p>Simulate a cell lineage tree
Adoped from https://github.com/elifesciences-publications/CRISPR_recorders_sims/blob/master/MATLAB_sims/GESTALT_30hr_1x_simulation.m
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree_config,missing'
simulate(config, x, n_samples = 200, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2Cmissing-method_+3A_config">config</code></td>
<td>
<p>simulation configuration; a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2Cmissing-method_+3A_x">x</code></td>
<td>
<p>missing</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2Cmissing-method_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples to simulate</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2Cmissing-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='simulate+2Clineage_tree_config+2CphyDat-method'>simulate</h2><span id='topic+simulate+2Clineage_tree_config+2CphyDat-method'></span>

<h3>Description</h3>

<p>Simulate a cell lineage tree based on a set of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree_config,phyDat'
simulate(config, x, n_samples = 200L, k = 50, greedy = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_config">config</code></td>
<td>
<p>simulation configuration; a lineage_tree_config object</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples to simulate</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_k">k</code></td>
<td>
<p>Number of trials</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_greedy">greedy</code></td>
<td>
<p>Whether ot not use a greedy search</p>
</td></tr>
<tr><td><code id="simulate+2B2Clineage_tree_config+2B2CphyDat-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='substr_kmer'>Generic function for substr_kmer</h2><span id='topic+substr_kmer'></span>

<h3>Description</h3>

<p>Generic function for substr_kmer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>substr_kmer(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="substr_kmer_+3A_x">x</code></td>
<td>
<p>a kmer object</p>
</td></tr>
<tr><td><code id="substr_kmer_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='substr_kmer+2Ckmer_summary-method'>Subseting a kmer_summary object</h2><span id='topic+substr_kmer+2Ckmer_summary-method'></span>

<h3>Description</h3>

<p>Summarize the short k-mer summary from the long k-mer summary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'kmer_summary'
substr_kmer(x, k = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="substr_kmer+2B2Ckmer_summary-method_+3A_x">x</code></td>
<td>
<p>a kmer_summary object</p>
</td></tr>
<tr><td><code id="substr_kmer+2B2Ckmer_summary-method_+3A_k">k</code></td>
<td>
<p>k-mer length(default: 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new kmer_summary object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='subtract'>Generic function for subtract</h2><span id='topic+subtract'></span>

<h3>Description</h3>

<p>Generic function for subtract
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subtract(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtract_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtract_+3A_y">y</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtract_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='subtract+2Clineage_tree+2Clineage_tree-method'>subtract</h2><span id='topic+subtract+2Clineage_tree+2Clineage_tree-method'></span>

<h3>Description</h3>

<p>Subtract a subtree from a large tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree,lineage_tree'
subtract(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtract+2B2Clineage_tree+2B2Clineage_tree-method_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtract+2B2Clineage_tree+2B2Clineage_tree-method_+3A_y">y</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtract+2B2Clineage_tree+2B2Clineage_tree-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>

<hr>
<h2 id='subtree'>Generic function for subtree</h2><span id='topic+subtree'></span>

<h3>Description</h3>

<p>Generic function for subtree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subtree(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtree_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtree_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='subtree+2Clineage_tree-method'>subtree</h2><span id='topic+subtree+2Clineage_tree-method'></span>

<h3>Description</h3>

<p>Extract a subtree with specific leaves
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lineage_tree'
subtree(x, leaves = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtree+2B2Clineage_tree-method_+3A_x">x</code></td>
<td>
<p>a lineage_tree object</p>
</td></tr>
<tr><td><code id="subtree+2B2Clineage_tree-method_+3A_leaves">leaves</code></td>
<td>
<p>leaves of the extracted tree</p>
</td></tr>
<tr><td><code id="subtree+2B2Clineage_tree-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a lineage_tree object
</p>

<hr>
<h2 id='subtree+2Cphylo-method'>subtree</h2><span id='topic+subtree+2Cphylo-method'></span>

<h3>Description</h3>

<p>Extract a subtree with specific leaves
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phylo'
subtree(x, leaves = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtree+2B2Cphylo-method_+3A_x">x</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="subtree+2B2Cphylo-method_+3A_leaves">leaves</code></td>
<td>
<p>leaves of the extracted tree</p>
</td></tr>
<tr><td><code id="subtree+2B2Cphylo-method_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pylo object
</p>

<hr>
<h2 id='summarize_kmer'>Generic function for summarize_kmer</h2><span id='topic+summarize_kmer'></span>

<h3>Description</h3>

<p>Generic function for summarize_kmer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_kmer(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarize_kmer_+3A_x">x</code></td>
<td>
<p>a sequence object</p>
</td></tr>
<tr><td><code id="summarize_kmer_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>

<hr>
<h2 id='summarize_kmer_core'>summarize_kmer_core</h2><span id='topic+summarize_kmer_core'></span>

<h3>Description</h3>

<p>Summarize kmer distributions (core function)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_kmer_core(
  k = 2,
  reps = 20L,
  n_samples = 200L,
  n_nodes = 100L,
  config = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarize_kmer_core_+3A_k">k</code></td>
<td>
<p>k-mer (default = 2)</p>
</td></tr>
<tr><td><code id="summarize_kmer_core_+3A_reps">reps</code></td>
<td>
<p>number of simulated trees</p>
</td></tr>
<tr><td><code id="summarize_kmer_core_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples to simulate</p>
</td></tr>
<tr><td><code id="summarize_kmer_core_+3A_n_nodes">n_nodes</code></td>
<td>
<p>number of nodes to sample (including both leaves and internval nodes)</p>
</td></tr>
<tr><td><code id="summarize_kmer_core_+3A_config">config</code></td>
<td>
<p>lineage tree configuration (a lineage_tree_config object)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a kmer_summary object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='summarize_kmer+2CphyDat-method'>summarize_kmer</h2><span id='topic+summarize_kmer+2CphyDat-method'></span>

<h3>Description</h3>

<p>Summarize kmer distributions with input sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'phyDat'
summarize_kmer(
  x,
  division = 16L,
  k = 2,
  reps = 20L,
  n_samples = 200L,
  n_nodes = 100L,
  n_targets
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_x">x</code></td>
<td>
<p>input data as a phyDat object</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_division">division</code></td>
<td>
<p>number of cell division</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_k">k</code></td>
<td>
<p>k-mer (default = 2)</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_reps">reps</code></td>
<td>
<p>number of simulated trees</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_n_samples">n_samples</code></td>
<td>
<p>number of samples to simulate</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_n_nodes">n_nodes</code></td>
<td>
<p>number of nodes to sample (including both leaves and internval nodes)</p>
</td></tr>
<tr><td><code id="summarize_kmer+2B2CphyDat-method_+3A_n_targets">n_targets</code></td>
<td>
<p>sequence length. If this argument is missing, the length of the input sequences will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a kmer_summary object
</p>


<h3>Author(s)</h3>

<p>Wuming Gong (gongx030@umn.edu)
</p>

<hr>
<h2 id='WH'>WH</h2><span id='topic+WH'></span>

<h3>Description</h3>

<p>implementation of weighted hamming algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WH(x, InfoW, dropout = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WH_+3A_x">x</code></td>
<td>
<p>Sequence object of 'phyDat' type.</p>
</td></tr>
<tr><td><code id="WH_+3A_infow">InfoW</code></td>
<td>
<p>Weight vector for the calculation of weighted hamming distance</p>
</td></tr>
<tr><td><code id="WH_+3A_dropout">dropout</code></td>
<td>
<p>Different weighting strategy is taken to consider interval dropout with dropout = 'TRUE'. Default is, dropout = 'FALSE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Calculated distance matrix of input sequences. The result is a 'dist' class object.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)
library(phangorn)
mu_d1 = c( 30, 20, 10, 5, 5, 1, 0.01, 0.001)
mu_d1 = mu_d1/sum(mu_d1)
simn = 10 # number of cell samples
m = 10  ## number of targets
sD = sim_seqdata(sim_n = simn, m = m, mu_d = 0.03,
        d = 12, n_s = length(mu_d1), outcome_prob = mu_d1, p_d = 0.005 )

## RF score with hamming distance
D_h = dist.hamming(sD$seqs)
tree_h= NJ(D_h)
RF.dist(tree_h, sD$tree, normalize = TRUE)

## RF score with weighted hamming
InfoW = -log(mu_d1)
InfoW[1:2] = 1
InfoW[3:7] = 4.5

D_wh = WH(sD$seqs, InfoW)
tree_wh= NJ(D_wh)
RF.dist(tree_wh, sD$tree, normalize = TRUE)

## RF score with weighted hamming, cosidering dropout situation
nfoW = -log(mu_d1)
InfoW[1] = 1
InfoW[2] = 12
InfoW[3:7] = 3

D_wh2 = WH(sD$seqs, InfoW, dropout=TRUE)
tree_wh2= NJ(D_wh2)
RF.dist(tree_wh2, sD$tree, normalize = TRUE)

</code></pre>

<hr>
<h2 id='WH_train'>Train weights for WH</h2><span id='topic+WH_train'></span>

<h3>Description</h3>

<p>Train weights for WH and output weight vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WH_train(X, loc0 = 2, locDropout = 1, locMissing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WH_train_+3A_x">X</code></td>
<td>
<p>a list of k number of input data, X[[1]] ... X[[k]]. The ith data have sequence information as phyDat format in X[[i]][[1]], and tree information in X[[i]][[2]] as phylo format.</p>
</td></tr>
<tr><td><code id="WH_train_+3A_loc0">loc0</code></td>
<td>
<p>weight location of initial state</p>
</td></tr>
<tr><td><code id="WH_train_+3A_locdropout">locDropout</code></td>
<td>
<p>weight location of dropout state</p>
</td></tr>
<tr><td><code id="WH_train_+3A_locmissing">locMissing</code></td>
<td>
<p>weight location of missing state, FALSE if there is no missing values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a weight vector
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak (ikwak2@cau.ac.kr)
</p>

<hr>
<h2 id='WH_train_fit'>Train weights for WH, and output distance object</h2><span id='topic+WH_train_fit'></span>

<h3>Description</h3>

<p>Train weights for WH using the given data, and fit the distance matrix for a input sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WH_train_fit(x, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WH_train_fit_+3A_x">x</code></td>
<td>
<p>input data in phyDat format</p>
</td></tr>
<tr><td><code id="WH_train_fit_+3A_x">X</code></td>
<td>
<p>a list of k number of input data, X[[1]] ... X[[k]]. The ith data have sequence information as phyDat format in X[[i]][[1]], and tree information in X[[i]][[2]] as phylo format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dist object
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak (ikwak2@cau.ac.kr)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
