<!DOCTYPE html><html lang="en"><head><title>Help for package DCODE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DCODE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DCODE-package'>
<p>List Linear n-Peptide Constraints for Overlapping Protein Regions</p></a></li>
<li><a href='#build_data'>
<p>Function to build the elements <code>aaList</code>, <code>geneticCode</code>, <code>peptideList</code>, used by function <code>getConstraint</code>.</p></a></li>
<li><a href='#DCODE-internal'><p>Internal DCODE Functions</p></a></li>
<li><a href='#getConstraint'>
<p>Function giving the linear constraint associated with the chosen peptide and overlapping frame.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>List Linear n-Peptide Constraints for Overlapping Protein
Regions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Sophie Lebre</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sophie Lebre &lt;sophie.lebre@umontpellier.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), seqinr</td>
</tr>
<tr>
<td>Description:</td>
<td>Traversal graph algorithm for listing linear n-peptide constraints for overlapping protein regions. (Lebre and Gascuel, The combinatorics of overlapping genes, freely available from arXiv at : http://arxiv.org/abs/1602.04971). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>no</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-10 09:55:21 UTC; sophie</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-10 17:55:18</td>
</tr>
</table>
<hr>
<h2 id='DCODE-package'>
List Linear n-Peptide Constraints for Overlapping Protein Regions
</h2><span id='topic+DCODE-package'></span><span id='topic+DCODE'></span>

<h3>Description</h3>

<p>Traversal graph algorithm for listing linear n-peptide constraints for overlapping protein regions. 
</p>


<h3>Author(s)</h3>

<p>S. Lebre
</p>
<p>Maintainer: S. Lebre &lt;sophie.lebre@umontpellier.fr&gt;
</p>


<h3>References</h3>

<p>Lebre and Gascuel, The combinatorics of overlapping genes 
(freely available from arXiv at: 
</p>
<p>http://arxiv.org/abs/1602.04971). 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
# 1) Build peptideList, used by function getConstraint.
mydata &lt;- build_data()
aaList &lt;- mydata$aaList
geneticCode &lt;- mydata$geneticCode
peptideList &lt;- mydata$peptideList

# 2) Ask for the constraint induced on a chosen peptide in a chosen overlapping frame

## amino acid contraints 
getConstraint("C", frame = -2, code=geneticCode, pepList=peptideList) 
getConstraint("D", frame = 1, code=geneticCode, pepList=peptideList)

## 3-peptide contraints
getConstraint("CWC",frame = -2, code=geneticCode, pepList=peptideList)

## 5-peptide contraints
getConstraint("CWCCC",frame = -2, code=geneticCode, pepList=peptideList)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='build_data'>
Function to build the elements <code>aaList</code>, <code>geneticCode</code>, <code>peptideList</code>, used by function <code><a href="#topic+getConstraint">getConstraint</a></code>.
</h2><span id='topic+build_data'></span>

<h3>Description</h3>

<p>To be run before the first use of function <code><a href="#topic+getConstraint">getConstraint</a></code>.
This function generates the variables <code>aaList</code>, <code>geneticCode</code> and  <code>peptideList</code>, a list of all the peptides (up to length 5), used by function <code><a href="#topic+getConstraint">getConstraint</a></code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_data()
</code></pre>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>aaList</code></td>
<td>
<p>Vector of all amino acids.</p>
</td></tr>
<tr><td><code>geneticCode</code></td>
<td>
<p>A list describing the standard genetic code. </p>
</td></tr>
<tr><td><code>peptideList</code></td>
<td>
<p>A list of all the peptides (up to length 5).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sophie Lebre &lt;sophie.lebre@umontpellier.fr&gt;
</p>


<h3>References</h3>

<p>Lebre and Gascuel, The combinatorics of overlapping genes 
(freely available from arXiv at: 
</p>
<p>http://arxiv.org/abs/1602.04971). 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getConstraint">getConstraint</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
# 1) Build peptideList, used by function getConstraint.
mydata &lt;- build_data()
aaList &lt;- mydata$aaList
geneticCode &lt;- mydata$geneticCode
peptideList &lt;- mydata$peptideList

# 2) Ask for the constraint induced on a chosen peptide in a chosen overlapping frame

## amino acid contraints 
getConstraint("C", frame = -2, code=geneticCode, pepList=peptideList) 
getConstraint("D", frame = 1, code=geneticCode, pepList=peptideList)

## 3-peptide contraints
getConstraint("CWC",frame = -2, code=geneticCode, pepList=peptideList)

## 5-peptide contraints
getConstraint("CWCCC",frame = -2, code=geneticCode, pepList=peptideList)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='DCODE-internal'>Internal DCODE Functions</h2><span id='topic+getConstraint_rec1'></span><span id='topic+getConstraint_rec2'></span><span id='topic+posLecturef1'></span><span id='topic+posLecturef2'></span><span id='topic+compRev'></span><span id='topic+getAllSeqs'></span><span id='topic+deleteStops'></span><span id='topic+printConstraint'></span><span id='topic+pepList'></span><span id='topic+stopCodons'></span><span id='topic+toPep'></span>

<h3>Description</h3>

<p>Internal DCODE functions
</p>


<h3>Details</h3>

<p>These are not to be called by the user (or in some cases are just
waiting for proper documentation to be written).
</p>

<hr>
<h2 id='getConstraint'>
Function giving the linear constraint associated with the chosen peptide and overlapping frame. 
</h2><span id='topic+getConstraint'></span>

<h3>Description</h3>

<p>This function prints the linear constraint to which the chosen peptide in the reference frame <code>pep</code> is subject when condisering gene overlap with frameshift <code>frame</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getConstraint(pep, frame, code, pepList)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getConstraint_+3A_pep">pep</code></td>
<td>

<p>A chosen n-peptide in the reference frame (with length n&lt;=5 if <code>pepList</code> is generated by function <code><a href="#topic+build_data">build_data</a></code>).
</p>
</td></tr>
<tr><td><code id="getConstraint_+3A_frame">frame</code></td>
<td>

<p>The frameshift chosen  in {-2, -1, 0, 1, 2}.
</p>
</td></tr>
<tr><td><code id="getConstraint_+3A_code">code</code></td>
<td>

<p>The code used to translate codons into amino acids (which can be generated by function <code><a href="#topic+build_data">build_data</a></code>).
</p>
</td></tr>
<tr><td><code id="getConstraint_+3A_peplist">pepList</code></td>
<td>

<p>A list of all the peptide of length 1, 2, 3, ...required by the graph traversal algorithm (which can be generated by function <code><a href="#topic+build_data">build_data</a></code>).
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sophie Lebre &lt;sophie.lebre@umontpellier.fr&gt;
</p>


<h3>References</h3>

<p>Lebre and Gascuel, The combinatorics of overlapping genes 
(freely available from arXiv at: 
</p>
<p>http://arxiv.org/abs/1602.04971). 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+build_data">build_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
# 1) Build peptideList, used by function getConstraint.
mydata &lt;- build_data()
aaList &lt;- mydata$aaList
geneticCode &lt;- mydata$geneticCode
peptideList &lt;- mydata$peptideList

# 2) Ask for the constraint induced on a chosen peptide in a chosen overlapping frame

## amino acid contraints 
getConstraint("C", frame = -2, code=geneticCode, pepList=peptideList) 
getConstraint("D", frame = 1, code=geneticCode, pepList=peptideList)

## 3-peptide contraints
getConstraint("CWC",frame = -2, code=geneticCode, pepList=peptideList)

## 5-peptide contraints
getConstraint("CWCCC",frame = -2, code=geneticCode, pepList=peptideList)
	
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
