<!DOCTYPE html><html><head><title>Help for package DDHFm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DDHFm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cdna'><p>Example cDNA data</p></a></li>
<li><a href='#cDNAdata'><p>Samples from the cDNA data vector</p></a></li>
<li><a href='#DDHFm'><p>Data-driven Haar-Fisz for microarrays</p></a></li>
<li><a href='#ddhft.np.2'><p>Data-Driven Haar-Fisz transformation</p></a></li>
<li><a href='#ddhft.np.inv'><p>Inverse Data-Driven Haar-Fisz transformation</p></a></li>
<li><a href='#dhhrcomp'><p>Simulated genes, apply DDHFm then compute and return variance,</p>
skewness and kurtosis values</a></li>
<li><a href='#dhhrss'><p>Tabulates variance, skewness and kurtosis coefficients from</p>
the output of dhhrcomp</a></li>
<li><a href='#function.from.vector'><p>Function applied for the computation of the Data-Driven Haar-Fisz transform</p></a></li>
<li><a href='#genesimulator'><p>Gene means simulator</p></a></li>
<li><a href='#hftrialdatagen'><p>Gene intensities simulator and DDHFm tester</p></a></li>
<li><a href='#isotone'><p>Performs Isotone regression using &quot;pool-adjacent-violators&quot; algorithm</p></a></li>
<li><a href='#KURTOSIS'><p>Kurtosis Coefficient estimator</p></a></li>
<li><a href='#MOMENTS'><p>Moment estimator</p></a></li>
<li><a href='#ShipCreditFlow'><p>Example Shipping credit flow data</p></a></li>
<li><a href='#simdurbin'><p>Gene intensities simulator</p></a></li>
<li><a href='#simdurbin2'><p>Gene intensities simulator</p></a></li>
<li><a href='#SKEW'><p>Skewness coefficient estimator</p></a></li>
<li><a href='#which.min.diff'><p>Find index where two vectors are closest</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Variance Stabilization by Data-Driven Haar-Fisz (for
Microarrays)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-11-21</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains the normalizing and variance stabilizing
	Data-Driven Haar-Fisz algorithm. Also contains related algorithms
	for simulating from certain microarray gene intensity models and
	evaluation of certain transformations. Contains cDNA and shipping
	credit flow data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>lokern, wavethresh</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-22 21:04:31 UTC; guynason</td>
</tr>
<tr>
<td>Author:</td>
<td>Efthimios Motakis [aut],
  Guy Nason [aut, cre],
  Piotr Fryzlewicz [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guy Nason &lt;g.nason@imperial.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-25 23:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cdna'>Example cDNA data</h2><span id='topic+cdna'></span>

<h3>Description</h3>

<p>cDNA data from a one-colour experiment from the Stanford
Microarray Database. The data set is stored as an intensity matrix
with 42624 genes (rows) and 4 replicates (columns).
</p>
<p>The experiment numbers are 40430, 40571, 34905 and 34912.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data(cdna)
</code></pre>


<h3>Format</h3>

<p>A 42624x4 matrix</p>


<h3>Source</h3>

<p>Stanford Microarray Database</p>


<h3>References</h3>

<p>McCaffrey, R.L., Fawcett, P., O'Riordan, M. Lee, K., Havell, E.A. Brown, P.O. and Portnoy, D.A. (2004) A specific gene
expression program triggered by Gram-positive bacteria in the cytocol, Proc. Nat. Acad. of Science, 101, 11386-11391
</p>

<hr>
<h2 id='cDNAdata'>Samples from the cDNA data vector</h2><span id='topic+cDNAdata'></span>

<h3>Description</h3>

<p>Arranges a vector of intensities into a form amenable to analysis
by the DDHF and also can restrict the number of genes analysed and
also obtain a random sample</p>


<h3>Usage</h3>

<pre><code class='language-R'> cDNAdata(data.vect,cdnalength,datasize,ng,nrep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cDNAdata_+3A_data.vect">data.vect</code></td>
<td>
<p>The data vector of intensities</p>
</td></tr>
<tr><td><code id="cDNAdata_+3A_cdnalength">cdnalength</code></td>
<td>
<p>Only considers the first <code>cdnalength</code> genes for analysis</p>
</td></tr>
<tr><td><code id="cDNAdata_+3A_datasize">datasize</code></td>
<td>
<p>Needs to be a power of two. The number of genes that get
randomly sampled from the first <code>cdnalengthq</code> and then
subsequently transformed.</p>
</td></tr>
<tr><td><code id="cDNAdata_+3A_ng">ng</code></td>
<td>
<p>The total number of genes described by <code>data.vect</code></p>
</td></tr> 
<tr><td><code id="cDNAdata_+3A_nrep">nrep</code></td>
<td>
<p>The number of replicates (should be a power of two) contained
in <code>data.vect</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The </p>
<p style="text-align: center;"><code class="reqn">J = ng \times nrep</code>
</p>
<p> vector data.vect should contain first the
intensities of the first replicate of the <code>ng</code> genes',
then the second replicate of all of the <code>ng</code> genes in the
same order, and so on.
</p>
<p>This function first puts the one dimensional <code>data.vect</code> into
a matrix form with <code>ng</code> rows and <code>nrep</code> columns (so the
row number indices the gene and the column number the replicate).
</p>
<p>Then the first <code>cdnalength</code> rows are extracted and kept, the remaining
rows are discarded.
</p>
<p>Then <code>datasize</code> rows at random are extracted and kept and
the remaining rows are discarded.
</p>


<h3>Value</h3>

<table>
<tr><td><code>cDNAdata</code></td>
<td>
<p>The data vector in the proper format to perform Data-Driven Haar-Fisz algorithm</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Efthimios Motakis &lt;e.motakis@bris.ac.uk&gt;</p>

<hr>
<h2 id='DDHFm'>Data-driven Haar-Fisz for microarrays</h2><span id='topic+DDHFm'></span>

<h3>Description</h3>

<p>Takes a matrix containing the resuls of a (e.g.) microarray
experiment (rows are different genes, columns are replicates)
and performs a multiscale variance stabilization and Gaussianization
transform. Note, you have to have replicates to make this work.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DDHFm(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DDHFm_+3A_x">x</code></td>
<td>
<p>A matrix containing n rows (genes) and k columns (replicates)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data-drive Haar-Fisz transform is a multiscale variance
stabilization and Gaussianization transform.
The algorithm will run on a one-column matrix but the results
will NOT be good. For one column of data use some other
variance stabilizer (in which case why are you using one column,
eh?)
</p>
<p>However, if you have replicates then you can use this
powerful multiscale algorithm which will stabilize variance
and also draw the distribution of the intensities towards
Gaussian extremely effectively (so skew and kurtosis will look
good too).
</p>
<p>This function forms a single vector of intensities by the
following procedure. First, it computes the mean over replicates
for each gene (this is why you need replicates, really).
Then it orders the rows of the matrix into increasing mean
order. Then it converts that matrix into a vector (which is therefore
a sequence of intensities which are grouped according to gene
(according to increasing mean intensity within each group)).
Then this vector is padded up to the next power-of-two in length
with zeroes. Then the padded vector is subjected to the
<code><a href="#topic+ddhft.np.2">ddhft.np.2</a></code> function which actually does the
data-driven Haar-Fisz transform. The results of this are
unpadded and then rearranged into a matrix with an identical
structure to the input matrix. This final matrix contains
stabilized and Gaussianized (is this a word?) intensities.
</p>
<p>It is important that you have replicates (ie, the number of
columns of the matrix has to be greater than 1). Also, it
is scientifically important that the replicates be just that
(ie, although the rows might refer to the same genes the columns
must be results from repeats experiments held under identical
conditions). For example, you shouldn't have results for the
gene on one subject and then the same gene on another subject
unless I suppose you are considering them to be replicates in some way.
</p>


<h3>Value</h3>

<p>A single matrix with the same dimensions as the input matrix.
Each entry in the resultant matrix corresponds to a variance
stabilized and Gaussianized version of the entry in the input
matrix.
</p>


<h3>Author(s)</h3>

<p>Guy Nason</p>


<h3>References</h3>

<p>Motakis, E.S., Nason, G.P., Fryzlewicz, P. and Rutter, G.A.
(2005)
Variance stabilization and normalization for one-color
microarray data using a data-driven multiscale approach.
<em>Technical Report</em>, 05:16, Statistics Group,
Department of Mathematics, University of Bristol, UK
</p>
<p>Motakis, E.S., Nason, G.P., Fryzlewicz, P. and Rutter, G.A.
(2006)
Variance stabilization and normalization for one-color
microarray data using a data-driven multiscale approach.
<em>Bioinformatics</em>, <b>22</b>, 2547&ndash;2553.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ddhft.np.2">ddhft.np.2</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# First, let's make up some "true" intensities (these are gamma)
#
TrueInt &lt;- genesimulator(nreps=5, nps=100)
#
# Now let us simulate data from the Durbin and Rocke (2001) model conditioning
# on the TrueInt
#
MAInts &lt;- simdurbin2(TrueInt,alpha=24800, seta=0.227, seps=4800)
#
# Now put these intensities into the correct format for DDHFm. That is
# replicates across columns
#
MAm &lt;- matrix(MAInts, ncol=5, byrow=TRUE)
#
# Apply DDHFm transform
#
MAmDDHFm &lt;- DDHFm(MAm)
#
# Now look at the variances of each
#
diag(var(MAm))
#
#[1] 22443437 25065729 23430160 22470058 21023183
#
diag(var(MAm))/ mean(diag(var(MAm)))
#
#[1] 0.9806403 1.0952183 1.0237540 0.9818035 0.9185839
#
# Now for the DDHFm version
#
diag(var(MAmDDHFm))
#
# [1] 10938.71 11369.41 11169.50 11144.74 10959.94
#
diag(var(MAmDDHFm))/ mean(diag(var(MAmDDHFm)))
#
# [1] 0.9840099 1.0227549 1.0047711 1.0025441 0.9859200
#
# So stabilization is better with DDHFm 
#
</code></pre>

<hr>
<h2 id='ddhft.np.2'>Data-Driven Haar-Fisz transformation</h2><span id='topic+ddhft.np.2'></span>

<h3>Description</h3>

<p>Forward Data-Driven Haar-Fisz transform</p>


<h3>Usage</h3>

<pre><code class='language-R'>ddhft.np.2(data, Ccode=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ddhft.np.2_+3A_data">data</code></td>
<td>
<p>A vector of size $2^J$ containing the data to variance
stabilize and Gaussianize. </p>
</td></tr>
<tr><td><code id="ddhft.np.2_+3A_ccode">Ccode</code></td>
<td>
<p>If TRUE then fast C code is used, otherwise R code is used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs the data-driven Haar-Fisz transform on sequence data.
This consists of (i) the Haar wavelet transform of sequence;
(ii) estimation of mean-variance relationship between finest
level smoothing and detail wavelet coefficients using
isotonic regression (see <code><a href="#topic+isotone">isotone</a></code>); (iii)
divide wavelet detail coefficients by smooth ones subjected
to the estimated mean-variance relationship; (iv) perform
the inverse Haar wavelet transform of the modified coefficients.
</p>
<p>The aim is to variance stabilize and Gaussianize the sequence
data which is only assumed to be positive and possess an
underlying increasing mean-variance relationship.
</p>


<h3>Value</h3>

<p>A list containing the following components
</p>
<table>
<tr><td><code>hft</code></td>
<td>
<p>The data-driven Haar-Fisz transform of the input sequence</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>The mu's obtained from the input sequence used for
estimating the mean-variance relationship</p>
</td></tr> 
<tr><td><code>sigma</code></td>
<td>
<p>The estimated standard deviation as a function of
the mean, the result of the isotonic regression fit of
sigma2 on sigma</p>
</td></tr>
<tr><td><code>sigma2</code></td>
<td>
<p>The local multiscale standard deviations associated
with each mean</p>
</td></tr>
<tr><td><code>factors</code></td>
<td>
<p>The numbers that divide the detail coefficients
to standardize variance (obtained from the mean-variance
estimation)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Piotr Fryzlewicz &lt;p.fryzlewicz@imperial.ac.uk&gt;
</p>


<h3>References</h3>

<p>Fisz, M. (1955), The limiting distribution of a function of two independent random variables 
and its statistical application, Colloquium Mathematicum, 3, 138-146.
</p>
<p>Delouille, V., Fryzlewicz, P. and Nason, G.P. (2005), A data-driven Haar-Fisz transformation 
for multiscale variance stabilization. Technical Report, 05:06,
Statistics Group, Department of Mathematics, University of Bristol </p>


<h3>See Also</h3>

<p><code><a href="#topic+ddhft.np.inv">ddhft.np.inv</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Generate example Poisson data set.
#
# Intensity function is steps from 1 to 32 in steps of 4 with each intensity
# lasting for 128 observations. Then sample Poisson with these intensities
#
v &lt;- rpois(1024, lambda=rep(seq(from=1, to=32, by=4), rep(1024/8,8)))
#
# Let's take a look at this
#
## Not run: ts.plot(v)
#
# Ok. So mean of intensity clear increasing, but variance increasing too
#
# Now do data-driven Haar-Fisz
#
vhft &lt;- ddhft.np.2(v)
#
# Now plot the variance stabilized series
#
## Not run: ts.plot(vhft$hft)
#
# The variance of the observations is much closer to 1. For example, let's
# look a the variance of the original series and the transformed one
#
# For the first intensity of 1
#
var(v[1:128])
#[1] 0.6628322
#
var(vhft$hft[1:128])
#[1] 1.025151
#
#
# And for second intensity of 5
#
#
var(v[129:256])
#[1] 4.389518
var(vhft$hft[129:256])
#[1] 1.312953
#
# So both transformed variances near to 1
#
# Now plot the estimated variance-mean relationship
#
## Not run: plot(vhft$mu, vhft$sigma)
## Not run: lines(vhft$mu, sqrt(vhft$mu))
#
# This is an approximately square root function (because you expect the
# sd of Poisson to be the square root of the mean).
#
</code></pre>

<hr>
<h2 id='ddhft.np.inv'>Inverse Data-Driven Haar-Fisz transformation</h2><span id='topic+ddhft.np.inv'></span>

<h3>Description</h3>

<p>Inverts the Data-Driven Haar-Fisz transform to obtain the raw
(untransformed) data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ddhft.np.inv(hft.obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ddhft.np.inv_+3A_hft.obj">hft.obj</code></td>
<td>
<p>An object with the same structure as that returned
by the <code><a href="#topic+ddhft.np.2">ddhft.np.2</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Merely performs the inverse of the DDHF transform see
<code><a href="#topic+ddhft.np.2">ddhft.np.2</a></code>
</p>


<h3>Value</h3>

<p>The inverted transform</p>


<h3>Author(s)</h3>

<p>Piotr Fryzlewicz &lt;p.fryzlewicz@imperial.ac.uk&gt;
</p>


<h3>References</h3>

 
<p>Delouille, V., Fryzlewicz, P. and Nason, G.P. (2005), A data-driven Haar-Fisz transformation 
for multiscale variance stabilization. Technical Report 05:06,
Statistics Group, Department of Mathematics, University of Bristol.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ddhft.np.2">ddhft.np.2</a></code></p>

<hr>
<h2 id='dhhrcomp'>Simulated genes, apply DDHFm then compute and return variance,
skewness and kurtosis values</h2><span id='topic+dhhrcomp'></span>

<h3>Description</h3>

<p>An example gene intensity simulator with given mean values mu0
which then returns the variance, skew and kurtosis of the
DDHF transformed intensities. The gene intensity
simulation is performed using <code><a href="#topic+simdurbin2">simdurbin2</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dhhrcomp(nsims = 1024, nmu = 4,
	mu0 = c(0, 5, 10, 15, 20, 25, 30, 40, 50, 60,
	65, 70, 80, 100, 200, 300, 500, 1000) * 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dhhrcomp_+3A_nsims">nsims</code></td>
<td>
<p>Number of replicates for each mu</p>
</td></tr>
<tr><td><code id="dhhrcomp_+3A_nmu">nmu</code></td>
<td>
<p>Selects first <code>nmu</code> genes from <code>mu</code> for use</p>
</td></tr>
<tr><td><code id="dhhrcomp_+3A_mu0">mu0</code></td>
<td>
<p>The possible gene intensity means</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simulates some gene intensities. A list of
possible intensity means are supplied in <code>mu0</code>.
The first <code>nmu</code> of these are selected. Then for each
of the selected means <code>nsims</code> gene intensities are
generated.
</p>
<p>The intensities are them subjected to DDHF transformation.
</p>


<h3>Value</h3>

<table>
<tr><td><code>mu</code></td>
<td>
<p>A vector of length <code>nmu</code> of the mean intensities considered</p>
</td></tr>
<tr><td><code>v.hft</code></td>
<td>
<p>A vector of <code>nmu</code> variances, one for each mean in
<code>mu0</code>.
The variance is the variance of the DDHFm transformed simulated
intensity data.</p>
</td></tr>
<tr><td><code>s.hft</code></td>
<td>
<p>A vector of <code>nmu</code> skewnesses, one for each mean in
<code>mu0</code>.
The skewness is the skewness of the DDHFm transformed simulated
intensity data.</p>
</td></tr>
<tr><td><code>k.hft</code></td>
<td>
<p>A vector of <code>nmu</code> kurtoses, one for each mean in
<code>mu0</code>.
The kurtosis is the kurtosis of the DDHFm transformed simulated
intensity data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk.</p>


<h3>References</h3>

<p>Durbin, B.P., Hardin, J.S., Hawkins, D.M. and Rocke, D.M. (2002), A variance-stabilizing transformation 
for gene expression microarray data, Bioinformatics, 18, S105-S110</p>


<h3>See Also</h3>

<p><code><a href="#topic+simdurbin2">simdurbin2</a></code></p>

<hr>
<h2 id='dhhrss'>Tabulates variance, skewness and kurtosis coefficients from
the output of dhhrcomp</h2><span id='topic+dhhrss'></span>

<h3>Description</h3>

<p>Tabulates the variance, skewness and kurtosis coefficients from the
output of <code><a href="#topic+dhhrcomp">dhhrcomp</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dhhrss(dhhrobj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dhhrss_+3A_dhhrobj">dhhrobj</code></td>
<td>
<p>The results to be tabulated. The output of a call
to <code><a href="#topic+dhhrcomp">dhhrcomp</a></code> </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This routine merely runs summary statistics calculations
on the output results from <code><a href="#topic+dhhrcomp">dhhrcomp</a></code>
</p>


<h3>Value</h3>

<p>None, prints a table.
</p>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk.</p>

<hr>
<h2 id='function.from.vector'>Function applied for the computation of the Data-Driven Haar-Fisz transform</h2><span id='topic+function.from.vector'></span>

<h3>Description</h3>

<p>Applies the function which.min.diff to the argument.vect vector
with the argument x</p>


<h3>Usage</h3>

<pre><code class='language-R'>function.from.vector(x, y, argument.vect)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="function.from.vector_+3A_x">x</code></td>
<td>
<p>a vector of data in ascending order</p>
</td></tr>
<tr><td><code id="function.from.vector_+3A_y">y</code></td>
<td>
<p>a vector with length(y) = length(x)</p>
</td></tr>
<tr><td><code id="function.from.vector_+3A_argument.vect">argument.vect</code></td>
<td>
<p>can be a vector</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>function.from.vector</code></td>
<td>
<p>help function for data transformation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Piotr Fryzlewicz &lt;p.fryzlewicz@imperial.ac.uk&gt;</p>

<hr>
<h2 id='genesimulator'>Gene means simulator</h2><span id='topic+genesimulator'></span>

<h3>Description</h3>

<p>Simulates means of the gene intensities for &quot;nps&quot; genes, each
replicated &quot;nreps&quot; times. The mean is drawn from a gamma distribution
with shape parameter &quot;shape&quot; and scale parameter &quot;scale&quot; </p>


<h3>Usage</h3>

<pre><code class='language-R'>genesimulator(nreps = 3, nps = 100, shape = 4, scale = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genesimulator_+3A_nreps">nreps</code></td>
<td>
<p>Number of replicates</p>
</td></tr>
<tr><td><code id="genesimulator_+3A_nps">nps</code></td>
<td>
<p>Number of genes</p>
</td></tr>
<tr><td><code id="genesimulator_+3A_shape">shape</code></td>
<td>
<p>Shape parameter</p>
</td></tr>
<tr><td><code id="genesimulator_+3A_scale">scale</code></td>
<td>
<p>Scale parameter</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For many problems a set of reasonable gene mean intensities is
useful for testing algorithms that later draw actual gene intensities
with distributions that possess a gene mean intensity.
</p>


<h3>Value</h3>

<p>A matrix containing nreps*nps rows and 3
columns. The first col contains all the gene mean
intensities. The second and third col contain the gene
replicate number and gene number respectively.
</p>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk&gt;</p>

<hr>
<h2 id='hftrialdatagen'>Gene intensities simulator and DDHFm tester</h2><span id='topic+hftrialdatagen'></span>

<h3>Description</h3>

<p>Simulates gene intensities and also applies DDHFm to them</p>


<h3>Usage</h3>

<pre><code class='language-R'>hftrialdatagen(nreps = 4, nps = 128, plot.it = FALSE, uvp = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hftrialdatagen_+3A_nreps">nreps</code></td>
<td>
<p>Number of replicates</p>
</td></tr>
<tr><td><code id="hftrialdatagen_+3A_nps">nps</code></td>
<td>
<p>Number of genes</p>
</td></tr>
<tr><td><code id="hftrialdatagen_+3A_plot.it">plot.it</code></td>
<td>
<p>Takes TRUE to activate the command of the respective plot and FALSE to deactivate it</p>
</td></tr>
<tr><td><code id="hftrialdatagen_+3A_uvp">uvp</code></td>
<td>
<p>a parameter for the denoising</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code is well commented for further information.
</p>
<p>First, <code><a href="#topic+genesimulator">genesimulator</a></code> is called to obtain a vector of
mean gene intensities (for a number of genes and a number of replicates
for each gene.
</p>
<p>Then <code>link{simdurbin2}</code> simulates a series of gene intensities
using the (log-normal type) model as described in Durbin and Rocke  
(2001,2002).
</p>
<p>Then for each gene the mean of replicates for that gene is computed.
</p>
<p>Optionally, if <code>plot.it</code> is <code>TRUE</code> then the mean 
is plotted against its standard deviation (over
replicates).
</p>
<p>Then the intensities are sorted according to increasing replicate mean.
</p>
<p>Optionally, if <code>plot.it</code> is <code>TRUE</code> then a plot of the
intensities values as a vector (sorted according to increasing
replicate mean) is plotted in black, and then the true mean plotted
in colour 2 (on my screen this is red) and the computed replicate
mean plotted in green.
</p>
<p>The DDHF transform of the sorted intensities is computed.
</p>
<p>Optionally, if <code>plot.it</code> is <code>TRUE</code> then a plot of the
transformed means versus the transformed standard deviations is plotted.
Followed by a time series plot of the transformed sorted intensities.
These can be studied to see how well DDHF has done the transformation.
</p>
<p>Then two smoothing methods are applied the the DDHF transformed data.
One method is translation invariant, Haar wavelet universal thresholding.
The other method is the classical smoothing spline. If <code>plot.it</code> is
<code>TRUE</code> then these smoothed estimates are plotted in different
colours.
</p>
<p>Then the mean estimated intensity for each gene is  computed and this
is returned as the first column of a two-column matrix (<code>ansm</code>).
The second column is the true underlying mean. The object
<code>hftssq</code> contains a measure of error between the estimated
and true gene means. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>ansm</code></td>
<td>
<p>Two column matrix containing the estimated gene intensities
and the true ones</p>
</td></tr>
<tr><td><code>hftssq</code></td>
<td>
<p>Sum of squares between estimated means and true means</p>
</td></tr>
<tr><td><code>yhf</code></td>
<td>
<p>Simulated gene intensities</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk&gt;</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# First run hftrialdatagen
#
## Not run: v &lt;- hftrialdatagen()
#
# Now plot the Haar-Fisz transformed intensities.
#
## Not run: ts.plot(v$yhf)
#
# Now plot the denoised intensities 
#
# Note that above we have 128 genes and 4 replicates and so there are
# 4*128 = 512 intensities to plot.
#
# However, there are only 128 gene intensities, and estimates. So, for this
# plot we choose to plot the noisy intensities and then for each replicate
# group (which are colocated on the plot) plot the (necessarily constant)
# true and estimated intensities (ie we plot each true/estimated intensity
# 4 times, once for each replicate).
#
# First estimates...
#
## Not run: lines(1:512, rep(v$ansm[,1], rep(4,128)), col=2)
#
# Now plot the truth
#
## Not run: lines(1:512, rep(v$ansm[,2], rep(4,128)), col=3)
</code></pre>

<hr>
<h2 id='isotone'>Performs Isotone regression using &quot;pool-adjacent-violators&quot; algorithm</h2><span id='topic+isotone'></span>

<h3>Description</h3>

<p>It is applied in order to estimate the h() function in the Data-Driven Haar-Fisz algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isotone(x, wt = rep(1, length(x)), increasing = FALSE, Ccode=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isotone_+3A_x">x</code></td>
<td>
<p>the vector that will be fitted with the regression</p>
</td></tr>
<tr><td><code id="isotone_+3A_wt">wt</code></td>
<td>
<p>a vector of weights</p>
</td></tr>
<tr><td><code id="isotone_+3A_increasing">increasing</code></td>
<td>
<p>if TRUE the curve is set to be increasing, else FALSE</p>
</td></tr>
<tr><td><code id="isotone_+3A_ccode">Ccode</code></td>
<td>
<p>if TRUE then faster C code is used rather than R</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>isotone</code></td>
<td>
<p>the regression results</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Bernard Silverman, with C modifications by GPN</p>


<h3>References</h3>

<p>Johnstone, I.M. and Silverman, B.W. (2005), EbayesThresh: R and S-Plus programs for empirical Bayes 
thresholding, Journal of Statistical Software, to appear
</p>

<hr>
<h2 id='KURTOSIS'>Kurtosis Coefficient estimator</h2><span id='topic+KURTOSIS'></span>

<h3>Description</h3>

<p>Estimates the kurtosis coefficient of a data vector  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KURTOSIS(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KURTOSIS_+3A_x">x</code></td>
<td>
<p>the data vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimated kurtosis of x </p>


<h3>Author(s)</h3>

<p>Efthimios Motakis &lt;e.motakis@bris.ac.uk</p>

<hr>
<h2 id='MOMENTS'>Moment estimator</h2><span id='topic+MOMENTS'></span>

<h3>Description</h3>

<p>Computes the rth central moment of the data x
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MOMENTS(x, r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MOMENTS_+3A_x">x</code></td>
<td>
<p>the data vector</p>
</td></tr>
<tr><td><code id="MOMENTS_+3A_r">r</code></td>
<td>
<p>parameter that describes which moment will be produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The rth central moment of x</p>


<h3>Author(s)</h3>

<p>Efthimios Motakis &lt;e.motakis@bris.ac.uk.</p>

<hr>
<h2 id='ShipCreditFlow'>Example Shipping credit flow data</h2><span id='topic+ShipCreditFlow'></span>

<h3>Description</h3>

<p>Shipping credit flow data. A set of monthly credit
transaction flow series recorded by the Central Bank of Cyprus.
The time range of the data is January 2002 to December 2014 and
contains 156 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data(ShipCreditFlow)
</code></pre>


<h3>Format</h3>

<p>A vector of length 156</p>


<h3>Source</h3>

<p>Central Bank of Cyprus</p>


<h3>References</h3>

<p>Michis, A. A. and Nason, G.P. (2016)
Case Study: Shipping trend estimation and prediction via multiscale
variance stabilisation. <em>Journal of Applied Statistics</em> (to appear)
</p>

<hr>
<h2 id='simdurbin'>Gene intensities simulator</h2><span id='topic+simdurbin'></span>

<h3>Description</h3>

<p>Simulates gene intensities from the two components model as in
the papers of Durbin and Rocke.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simdurbin(n, alpha, mu, seta, seps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simdurbin_+3A_n">n</code></td>
<td>
<p>number of intensities to be simulated</p>
</td></tr>
<tr><td><code id="simdurbin_+3A_alpha">alpha</code></td>
<td>
<p>background mean of gene intensities</p>
</td></tr>
<tr><td><code id="simdurbin_+3A_mu">mu</code></td>
<td>
<p>the means where the gene intensities are simulated from</p>
</td></tr>
<tr><td><code id="simdurbin_+3A_seta">seta</code></td>
<td>
<p>standard deviation of high-level gene intensities</p>
</td></tr>
<tr><td><code id="simdurbin_+3A_seps">seps</code></td>
<td>
<p>standard deviation of low-level gene intensities</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates <code>n</code> intensities from the
Durbin and Rocke (2001) gene intensity model with parameters
alpha, seta and seps. 
</p>


<h3>Value</h3>

<p>A vector of the appropriate intensities.
</p>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk&gt;</p>


<h3>References</h3>

<p>Rocke, D.M. and Durbin, B.P. (2001), A model for measurement error for gene expression arrays, Journal of 
Computational Biology, 8, 557-569
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simdurbin2">simdurbin2</a></code></p>

<hr>
<h2 id='simdurbin2'>Gene intensities simulator</h2><span id='topic+simdurbin2'></span>

<h3>Description</h3>

<p>Simulates gene intensities from the two components model of
Durbin and Rocke.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simdurbin2(mu, alpha, seta, seps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simdurbin2_+3A_mu">mu</code></td>
<td>
<p>the means where the gene intensities are simulated from</p>
</td></tr>
<tr><td><code id="simdurbin2_+3A_alpha">alpha</code></td>
<td>
<p>background mean of gene intensities</p>
</td></tr>
<tr><td><code id="simdurbin2_+3A_seta">seta</code></td>
<td>
<p>standard deviation of high-level gene intensities</p>
</td></tr>
<tr><td><code id="simdurbin2_+3A_seps">seps</code></td>
<td>
<p>standard deviation of low-level gene intensities</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Like <code><a href="#topic+simdurbin">simdurbin</a></code> this function simulates gene
intensities from the Durbin and Rocke two component model (see
Durbin and Rocke, 2001). Unlike <code><a href="#topic+simdurbin">simdurbin</a></code> this function
generates each intensity with a mean specified by an entry in
<code>mu</code>. So, <code>length(mu)</code> intensities are generated with
intensity <code>i</code> having mean <code>mu[i]</code>. 
</p>


<h3>Author(s)</h3>

<p>Guy Nason &lt;g.p.nason@bris.ac.uk&gt;</p>


<h3>References</h3>

<p>Rocke, D.M. and Durbin, B.P. (2001), A model for measurement error for gene expression arrays, Journal of
Computational Biology, 8, 557-569
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simdurbin">simdurbin</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Generate data with mu equal to 80 and mu equal to 80000 with 10 replicates each
#
#
v &lt;- simdurbin2( c(rep(80, 10), rep(80000,10)), alpha=24800, seta=0.227, seps=4800)
#
# Let's look at the mean of the first 10 and the last 10
#
mean(v[1:10])
#[1] 27361.95
#
mean(v[11:20])
#[1] 92455.66
</code></pre>

<hr>
<h2 id='SKEW'>Skewness coefficient estimator</h2><span id='topic+SKEW'></span>

<h3>Description</h3>

<p>Estimates the skewness coefficient of the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SKEW(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SKEW_+3A_x">x</code></td>
<td>
<p>The data vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The skewness of x </p>


<h3>Author(s)</h3>

<p>Efthimios Motakis &lt;e.motakis@bris.ac.uk&gt;</p>

<hr>
<h2 id='which.min.diff'>Find index where two vectors are closest</h2><span id='topic+which.min.diff'></span>

<h3>Description</h3>

<p>Finds the index point where two index vectors are closest
in value</p>


<h3>Usage</h3>

<pre><code class='language-R'>which.min.diff(a, vect)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="which.min.diff_+3A_a">a</code></td>
<td>
<p>input vector</p>
</td></tr>
<tr><td><code id="which.min.diff_+3A_vect">vect</code></td>
<td>
<p>input vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The index where <code>abs(a-vect)</code> is smallest, ie where <code>a</code>
and <code>vect</code> are smallest.
</p>


<h3>Author(s)</h3>

<p>Piotr Fryzlewicz &lt;p.fryzlewicz@imperial.ac.uk&gt;</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
# Make up two vectors
#
a &lt;- c(1,2,3)
vect &lt;- c(3,2,1)
#
# Now see on which index are the two closest in value 
#
which.min.diff(a,vect)
#[1] 2
#
# ie its the second index where both vectors are actually 2
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
