<!DOCTYPE html><html><head><title>Help for package DDPNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DDPNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#anova_p'>
<p>anova_p</p></a></li>
<li><a href='#changedID'>
<p>changedID</p></a></li>
<li><a href='#Data_impute'>
<p>Data_impute</p></a></li>
<li><a href='#dataStatInf'>
<p>dataStatInf</p></a></li>
<li><a href='#DDPNA-package'>
<p>Disease-drived Differential Proteins And Proteomic Co-expression Network Associated Analysis</p></a></li>
<li><a href='#DEP_Mod_HeatMap'>
<p>DEP_Mod_HeatMap</p></a></li>
<li><a href='#DEP_Mod_net_plot'>
<p>DEP_Mod_net_plot</p></a></li>
<li><a href='#DEPsets'>
<p>DEPsets</p></a></li>
<li><a href='#Dforimpute'><p>a data used for Dataimpute function.</p></a></li>
<li><a href='#fc.pos'>
<p>fc.pos</p></a></li>
<li><a href='#FCSenrichplot'>
<p>FCSenrichplot</p></a></li>
<li><a href='#getmoduleHub'>
<p>getmoduleHub</p></a></li>
<li><a href='#groupmean'>
<p>groupmean</p></a></li>
<li><a href='#ID_match'>
<p>homolog protein Uniprot ID transformation</p></a></li>
<li><a href='#imputedData'><p>an imputed data used for downstream analysis</p></a></li>
<li><a href='#MaxQdataconvert'>
<p>one-step to extract 'Maxquant' quantification data and convert</p></a></li>
<li><a href='#MaxQprotein'>
<p>read proteomic quantification data and seperate the protein information and quantification information.</p></a></li>
<li><a href='#ME_inf'>
<p>module eigengenes information</p></a></li>
<li><a href='#modpcomp'>
<p>modpcomp</p></a></li>
<li><a href='#Module_Enrich'>
<p>Module_Enrich</p></a></li>
<li><a href='#Module_inf'>
<p>Module and protein information.</p></a></li>
<li><a href='#moduleID'>
<p>extract intersection ID between dataset and module</p></a></li>
<li><a href='#multi.t.test'>
<p>multi.t.test</p></a></li>
<li><a href='#net'><p>A network which is obtained by WGCNA-package blockwiseModule function.</p>
it can short the time of Module analysis example time.</a></li>
<li><a href='#P.G.extract'>
<p>Protein Groups information extract.</p></a></li>
<li><a href='#ProteomicData'><p>Small example proteomic quantification data extract from txt or csv file used to</p>
demonstrate data extract function.</a></li>
<li><a href='#rename_dupnewID'>
<p>rename_dupnewID</p></a></li>
<li><a href='#Sample_ID_data'><p>A small uniprot ID information used to do ID convert in example.</p></a></li>
<li><a href='#single_mod_enrichplot'>
<p>single_mod_enrichplot</p></a></li>
<li><a href='#SoftThresholdScaleGraph'>
<p>SoftThresholdScaleGraph</p></a></li>
<li><a href='#wgcnatest'>
<p>wgcnatest</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Disease-Drived Differential Proteins Co-Expression Network
Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-3-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Kefu Liu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kefu Liu &lt;liukefu19@163.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/liukf10/DDPNA">https://github.com/liukf10/DDPNA</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/liukf10/DDPNA/issues">https://github.com/liukf10/DDPNA/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Functions designed to connect disease-related differential proteins and 
  co-expression network. It provides the basic statics analysis included t test, ANOVA analysis. 
  The network construction is not offered by the package, you can used 'WGCNA' package which you 
  can learn in Peter et al. (2008) &lt;<a href="https://doi.org/10.1186%2F1471-2105-9-559">doi:10.1186/1471-2105-9-559</a>&gt;. It also provides module analysis 
  included PCA analysis, two enrichment analysis, Planner maximally filtered graph extraction and 
  hub analysis.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, ggplot2, ggalt, MEGENA, igraph, Hmisc, utils,
grDevices, plyr, scales, grid, VennDiagram</td>
</tr>
<tr>
<td>Suggests:</td>
<td>WGCNA, Biostrings, impute, ggfortify</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-14 01:40:58 UTC; liuke</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-14 03:20:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='anova_p'>
anova_p
</h2><span id='topic+anova_p'></span>

<h3>Description</h3>

<p>anova analysis in proteomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anova_p(data, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova_p_+3A_data">data</code></td>
<td>

<p>protein quantification data. column is sample. row is protein ID.
</p>
</td></tr>
<tr><td><code id="anova_p_+3A_group">group</code></td>
<td>

<p>sample group information
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+", "", colnames(logD))
anova_P &lt;- anova_p(logD[1:100,], group)

</code></pre>

<hr>
<h2 id='changedID'>
changedID
</h2><span id='topic+changedID'></span>

<h3>Description</h3>

<p>extract significant differential protein
</p>


<h3>Usage</h3>

<pre><code class='language-R'>changedID(relative_value, group, vs.set2, vs.set1 = "WT",
          rank = "none", anova = TRUE, anova.cutoff = 0.05,
          T.cutoff = 0.05, Padj = "fdr",
          cutoff = 1.5, datatype = c("none","log2"), fctype = "all",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="changedID_+3A_relative_value">relative_value</code></td>
<td>
<p>protein quantification data</p>
</td></tr>
<tr><td><code id="changedID_+3A_group">group</code></td>
<td>
<p>sample group information</p>
</td></tr>
<tr><td><code id="changedID_+3A_vs.set2">vs.set2</code></td>
<td>
<p>compared group 2 name</p>
</td></tr>
<tr><td><code id="changedID_+3A_vs.set1">vs.set1</code></td>
<td>
<p>compared group 1 name</p>
</td></tr>
<tr><td><code id="changedID_+3A_rank">rank</code></td>
<td>
<p>order by which type. This must be (an abbreviation of) one of the strings &quot;<code>none</code>&quot;,&quot;<code>foldchange</code>&quot;, &quot;<code>anova</code>&quot;,&quot;<code>t</code>&quot;</p>
</td></tr>
<tr><td><code id="changedID_+3A_anova">anova</code></td>
<td>
<p>a logical value indicating whether do anova analysis.</p>
</td></tr>
<tr><td><code id="changedID_+3A_anova.cutoff">anova.cutoff</code></td>
<td>
<p>a numberic value indicated that anova test p value upper limit.</p>
</td></tr>
<tr><td><code id="changedID_+3A_t.cutoff">T.cutoff</code></td>
<td>
<p>a numberic value indicated that t.test p value upper limit.</p>
</td></tr>
<tr><td><code id="changedID_+3A_padj">Padj</code></td>
<td>

<p>p adjust methods of multiple comparisons.
it can seen in <code>p.adjust.methods</code>.
</p>
</td></tr>
<tr><td><code id="changedID_+3A_cutoff">cutoff</code></td>
<td>
<p>a numberic value indicated that foldchange lower limit.</p>
</td></tr>
<tr><td><code id="changedID_+3A_datatype">datatype</code></td>
<td>
<p>The quantification data is normal data or log2 data.</p>
</td></tr>
<tr><td><code id="changedID_+3A_fctype">fctype</code></td>
<td>
<p>foldchange is ordered by up-regulated or down-regulated or changed</p>
</td></tr>
<tr><td><code id="changedID_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>extract significant differential protein ID based on foldchange, t.test p value, anova p value.
</p>


<h3>Value</h3>

<p>a vector of protein ID information.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
up &lt;- changedID(logD[201:260,], group, vs.set2 = "ad", vs.set1 = "ctl",
              rank = "foldchange",anova = FALSE, Padj = "none", cutoff = 1,
              datatype = "log2", fctype = "up")

</code></pre>

<hr>
<h2 id='Data_impute'>
Data_impute
</h2><span id='topic+Data_impute'></span>

<h3>Description</h3>

<p>data clean process: detect and remove outlier sample and impute missing value.
The process is following:
1. Remove some genes which the number of missing value larger than maxNAratio.
2. Outlier sample detect and remove these sample.
3. Repeat Steps 1-2 untile meet the iteration times or no outlier sample can be detected.
4. impute the missing value.
The function also can only do gene filter or remove outlier or impute missing value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Data_impute(data, inf = "inf", intensity = "LFQ", miss.value = NA,
            splNExt = TRUE, maxNAratio = 0.5,
            removeOutlier = TRUE,
            outlierdata = "intensity", iteration = NA, sdout = 2,
            distmethod = "manhattan", A.IAC = FALSE,
            dohclust = FALSE, treelabels = NA,
            plot = TRUE, filename = NULL,
            text.cex = 0.7, text.col = "red", text.pos = 1,
            text.labels = NA, abline.col = "red", abline.lwd = 2,
            impute = TRUE, verbose = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Data_impute_+3A_data">data</code></td>
<td>

<p>MaxQconvert data or a list Vector which contain two data.frame:ID information and quantification data
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_inf">inf</code></td>
<td>

<p>the data.frame name contain protein ID information
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_intensity">intensity</code></td>
<td>

<p>the data.frame name only contain quantification data
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_miss.value">miss.value</code></td>
<td>

<p>the type of miss.value showed in quantificaiton data.
The default value is <code>NA</code>. The miss.value usually can be <code>NA</code> or  <code>0</code>.</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_splnext">splNExt</code></td>
<td>

<p>a logical value whether extract sample name.(suited for MaxQuant quantification data)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_maxnaratio">maxNAratio</code></td>
<td>

<p>The maximum percent missing data allowed in any row (default 50%).For any rows with more than maxNAratio% missing will deleted.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_removeoutlier">removeOutlier</code></td>
<td>

<p>a logical value indicated whether remove outlier sample.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_outlierdata">outlierdata</code></td>
<td>

<p>The value is deprecated.
which data will be used to analysis outlier sample detect.This must be (an abbreviation of) one of the strings &quot;<code>intensity</code>&quot;,&quot;<code>relative_value</code>&quot;,&quot;<code>log2_value</code>&quot;.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_iteration">iteration</code></td>
<td>

<p>a numberic value indicating how many times it go through the outlier sample detect and remove loop.<code>NA</code> means do loops until no outlier sample.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_sdout">sdout</code></td>
<td>

<p>a numberic value indicating the threshold to judge the outlier sample. The default <code>2</code> means 0.95 confidence intervals
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_distmethod">distmethod</code></td>
<td>

<p>The distance measure to be used. This must be (an abbreviation of) one of the strings &quot;<code>manhattan</code>&quot;,&quot;<code>euclidean</code>&quot;, &quot;<code>canberra</code>&quot;,&quot;<code>correlation</code>&quot;,&quot;<code>bicor</code>&quot;
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_a.iac">A.IAC</code></td>
<td>

<p>a logical value indicated whether decreasing <code>correlation</code> variance.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_dohclust">dohclust</code></td>
<td>

<p>a logical value indicated whether doing hierarchical clustering and plot dendrograms.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_treelabels">treelabels</code></td>
<td>

<p>labels of dendrograms
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_plot">plot</code></td>
<td>

<p>a logical value indicated whether plot numbersd scatter diagrams.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_filename">filename</code></td>
<td>

<p>the filename of plot. The number and plot type information will added automatically. The default value is <code>NULL</code> which means no file saving.
all the plot will be saved to &quot;plot&quot; folder and saved in pdf format.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_text.cex">text.cex</code></td>
<td>

<p>outlier sample annotation text size(scatter diagrams parameters)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_text.col">text.col</code></td>
<td>

<p>outlier sample annotation color(scatter diagrams parameters)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_text.pos">text.pos</code></td>
<td>

<p>outlier sample annotation position(scatter diagrams parameters)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_text.labels">text.labels</code></td>
<td>

<p>outlier sample annotation (scatter diagrams parameters)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_abline.col">abline.col</code></td>
<td>

<p>the threshold line color (scatter diagrams parameters)</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_abline.lwd">abline.lwd</code></td>
<td>
<p>the threshold line width (scatter diagrams parameters)
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_impute">impute</code></td>
<td>

<p>a logical value indicated whether do knn imputation.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_verbose">verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, 1 means have some Diagnostic Messages.
</p>
</td></tr>
<tr><td><code id="Data_impute_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>detect and remove outlier sample and impute missing value.
</p>


<h3>Value</h3>

<p>a list of proteomic data.
</p>
<table>
<tr><td><code>inf</code></td>
<td>
<p>Portein information included protein IDs and other information.</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>Quantification informaton.</p>
</td></tr>
<tr><td><code>relative_value</code></td>
<td>
<p>intensity divided by geometric mean</p>
</td></tr>
<tr><td><code>log2_value</code></td>
<td>
<p>log2 of relative_value</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dforimpute)
data &lt;- Data_impute(Dforimpute,distmethod="manhattan")
</code></pre>

<hr>
<h2 id='dataStatInf'>
dataStatInf
</h2><span id='topic+dataStatInf'></span>

<h3>Description</h3>

<p>summrize the statistics information of data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataStatInf(prodata, group, intensity = "intensity",
            Egrp = NULL, Cgrp = "ctl",
            meanmethod = "mean", datatype = c("none", "log2"),
            anova = TRUE, T.test = c("pairwise", "two-sample", "none"),
            Aadj = "none", Tadj = "none", cutoff = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dataStatInf_+3A_prodata">prodata</code></td>
<td>

<p>proteome data. a list Vector which contain two data.frame: ID information and quantification data
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_intensity">intensity</code></td>
<td>

<p>the data.frame name only contain quantification data

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_group">group</code></td>
<td>

<p>sample group information
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_egrp">Egrp</code></td>
<td>

<p>experiment group name. It must be assigned when use Student T.test.

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_cgrp">Cgrp</code></td>
<td>

<p>control group name. It must be assigned. The default value is &quot;ctl&quot;.
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_meanmethod">meanmethod</code></td>
<td>

<p>Arithmetic mean of sample group or median of sample group.
This must be (an abbreviation of) one of the strings &quot;<code>mean</code>&quot;,&quot;<code>median</code>&quot;.
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_datatype">datatype</code></td>
<td>

<p>The quantification data is normal data or log2 data.

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_anova">anova</code></td>
<td>

<p>a logical value indicating whether do anova analysis.

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_t.test">T.test</code></td>
<td>

<p>T.test method. &quot;none&quot; means not running t.test.
&quot;pairwise&quot; means calculate pairwise comparisons between group levels with corrections for multiple testing
&quot;two-sample&quot; means student t test.
This must be (an abbreviation of) one of the strings &quot;<code>pairwise</code>&quot;,&quot;<code>two-sample</code> and &quot;<code>none</code>&quot;&quot;.
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_aadj">Aadj</code></td>
<td>

<p>anova P value adjust methods. it can seen in <code>p.adjust.methods</code>.

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_tadj">Tadj</code></td>
<td>

<p>T test P value adjust methods. it can seen in <code>p.adjust.methods</code>.

</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_cutoff">cutoff</code></td>
<td>

<p>a logical value or a numeric value.
The default value is FALSE, which means do not remove any P value.
If the value is TRUE, P value &gt; 0.05 will remove and showed as NA in result.
If the value is numeric, P value &gt; the number will remove and showed as NA in result.
</p>
</td></tr>
<tr><td><code id="dataStatInf_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame of protein ID and Statistics information.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
group &lt;- gsub("[0-9]+","", colnames(imputedData$intensity))
data &lt;- imputedData
data$inf &lt;- data$inf[1:100,]
data$intensity &lt;- data$intensity[1:100,]
stat &lt;- dataStatInf(data, group, meanmethod = "median",
                    T.test = "pairwise", Aadj = "fdr",
                    Tadj = "fdr", cutoff = FALSE)

</code></pre>

<hr>
<h2 id='DDPNA-package'>
Disease-drived Differential Proteins And Proteomic Co-expression Network Associated Analysis
</h2><span id='topic+DDPNA-package'></span><span id='topic+DDPNA'></span>

<h3>Description</h3>

<p>disease drived proteins associated network in different species crosstalk.
The package is used to analysis differential proteomics consensus network in two or more datasets.
The function <code>Data_impute</code> need <span class="pkg">impute</span> package from Bioconductor, the function <code>ID_match</code> and the function <code>MaxQdataconvert</code> need <span class="pkg">Biostrings</span> package from Bioconductor.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> DDPNA</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.3.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Creat Data: </td><td style="text-align: left;"> 2019-03-18</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-03-13</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package is used for construct core component sets associated co-expression network.
It offered a preprocess function and core component sets subnetwork construct.
Preprocess function can process proteomics data included data extract, outlier removal and impute. Some functions are fit for other omics data.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>
<p>Maintainer: Kefu Liu &lt;liukefu19@163.com&gt;
</p>

<hr>
<h2 id='DEP_Mod_HeatMap'>
DEP_Mod_HeatMap
</h2><span id='topic+DEP_Mod_HeatMap'></span>

<h3>Description</h3>

<p>get the DEP enrich fold in Module and plot a HeatMap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEP_Mod_HeatMap(DEP_Mod, xlab = "DEP", filter = c("p","p.adj"),
                cutoff = 0.05, filename = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DEP_Mod_HeatMap_+3A_dep_mod">DEP_Mod</code></td>
<td>

<p>a list of DEP_Mod enrichment information. data.frame in list is get from <code>Module_Enrich</code> function.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_HeatMap_+3A_xlab">xlab</code></td>
<td>

<p>it indicate x value in heatmap. it must be a value between &quot;<code>DEP</code>&quot; and &quot;<code>MOD</code>&quot;.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_HeatMap_+3A_filter">filter</code></td>
<td>

<p>p value or p.adjust value used to filter the enrich significant module.

</p>
</td></tr>
<tr><td><code id="DEP_Mod_HeatMap_+3A_cutoff">cutoff</code></td>
<td>

<p>a numeric value is the cutoff of p value. Larger than the value will remove to show in plot.

</p>
</td></tr>
<tr><td><code id="DEP_Mod_HeatMap_+3A_filename">filename</code></td>
<td>

<p>plot filename. If filename is null, it will print the plot.

</p>
</td></tr>
<tr><td><code id="DEP_Mod_HeatMap_+3A_...">...</code></td>
<td>

<p>other argument.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of enrich fold heatmap information.
</p>
<table>
<tr><td><code>enrichFold</code></td>
<td>
<p>enrichFold of DEP in Modules.</p>
</td></tr>
<tr><td><code>textMatrix</code></td>
<td>
<p>siginificant enrichment module information.</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
# define 2 DEP ID data: a and b
a &lt;- Module$ori.ID[1:100]
b &lt;- Module$ori.ID[50:100]
a &lt;- Module_Enrich(Module, a, coln="ori.ID", enrichtype = "ORA")
b &lt;- Module_Enrich(Module, b, coln="ori.ID", enrichtype = "ORA")
rowname &lt;- a$module.name;
a &lt;- data.frame(Counts = a$Counts, module.size = a$module.size,
                precent = a$precent, p = a$p, p.adj = a$p.adj,
                Z.score = a$Z.score, stringsAsFactors = FALSE)
rownames(a) &lt;- rowname;
rowname &lt;- b$module.name;
b &lt;- data.frame(Counts = b$Counts, module.size = b$module.size,
                precent = b$precent, p = b$p, p.adj = b$p.adj,
                Z.score = b$Z.score, stringsAsFactors = FALSE)
rownames(b) &lt;- rowname;
DEP_Mod &lt;- list(a = a , b = b)
heatMapInf &lt;- DEP_Mod_HeatMap(DEP_Mod)

</code></pre>

<hr>
<h2 id='DEP_Mod_net_plot'>
DEP_Mod_net_plot
</h2><span id='topic+DEP_Mod_net_plot'></span>

<h3>Description</h3>

<p>remove hubs which is not in the IDsets and replot the PFG network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEP_Mod_net_plot(ModNet, IDsets = NULL, data = NULL, module = NULL,
                 plot = TRUE, filename = NULL, filetype = "pdf",
                 OnlyPlotLast = TRUE, BranchCut = TRUE,
                 reconstructNet = TRUE,
                 iteration = Inf, label.hubs.only = TRUE,
                 node.default.color = "grey",
                 hubLabel.col = "black", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DEP_Mod_net_plot_+3A_modnet">ModNet</code></td>
<td>

<p>data contains network information which get from <code>getmoduleHub</code>
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_idsets">IDsets</code></td>
<td>

<p>ID sets information which get from <code>DEPsets</code>

</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_data">data</code></td>
<td>

<p>the value should be defined only when <code>reconstructNet</code> is TRUE.
The value is proteomic quantification data, which is same as the input in <code>getmoduleHub</code>.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_module">module</code></td>
<td>

<p>the value should be defined only when <code>reconstructNet</code> is TRUE.
The value is module information which is same as the input in <code>getmoduleHub</code>.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_plot">plot</code></td>
<td>

<p>a logical value whether plot a picture.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_filename">filename</code></td>
<td>

<p>the filename of plot. The default value is <code>NULL</code> which means no file saving. The function is use <code>ggsave</code> to achieve.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_filetype">filetype</code></td>
<td>

<p>the file type of plot.  the type should be one of &quot;eps&quot;, &quot;ps&quot;, &quot;tex&quot; (pictex), &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;png&quot;, &quot;bmp&quot;, &quot;svg&quot; or &quot;wmf&quot; (windows only).
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_onlyplotlast">OnlyPlotLast</code></td>
<td>

<p>a logical value whether plot the final network.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_branchcut">BranchCut</code></td>
<td>

<p>a logical value whether remove unhub proteins which have no connection to DEPs.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_reconstructnet">reconstructNet</code></td>
<td>

<p>a logical value whether reconstruct network.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_iteration">iteration</code></td>
<td>

<p>iteration times when reconstruct network.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_label.hubs.only">label.hubs.only</code></td>
<td>

<p>a logical value whether show labels for hubs only.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_node.default.color">node.default.color</code></td>
<td>

<p>Default node colors for those that do not intersect with signatures in gene.set.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_hublabel.col">hubLabel.col</code></td>
<td>

<p>Label color for hubs.
</p>
</td></tr>
<tr><td><code id="DEP_Mod_net_plot_+3A_...">...</code></td>
<td>

<p>additional <code>ggsave</code> parameter
</p>
</td></tr>
</table>


<h3>Value</h3>


<p>a list contains network information
</p>
<table>
<tr><td><code>netgene</code></td>
<td>
<p>all IDs in network.</p>
</td></tr>
<tr><td><code>hub</code></td>
<td>
<p>hub IDs</p>
</td></tr>
<tr><td><code>PMFG</code></td>
<td>
<p>PMFG graph data frame information</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(net)
data(imputedData)
Module &lt;- Module_inf(net, imputedData$inf)
group &lt;- gsub("[0-9]+","", colnames(imputedData$intensity))
data &lt;- imputedData
data$inf &lt;- data$inf[1:100,]
data$intensity &lt;- data$intensity[1:100,]
stat &lt;- dataStatInf(data, group, meanmethod = "median",
                    T.test = "pairwise", Aadj = "fdr",
                    Tadj = "fdr", cutoff = FALSE)
stat1 &lt;- stat$ori.ID[stat$ad &gt; 1]
stat2 &lt;- stat$ori.ID[stat$asym &gt; 1]
datalist &lt;- list(stat1 = stat1, stat2 = stat2)
sets &lt;- DEPsets(datalist)

logD &lt;- imputedData$log2_value
rownames(logD) &lt;- imputedData$inf$ori.ID
Mod3 &lt;- getmoduleHub(logD, Module, 3, coln = "ori.ID", adjustp = FALSE)

newnet &lt;- DEP_Mod_net_plot(Mod3, sets,
                           data = logD, module = Module,
                           plot = FALSE, filename = NULL, filetype = "pdf",
                           OnlyPlotLast = FALSE,reconstructNet = FALSE)


</code></pre>

<hr>
<h2 id='DEPsets'>
DEPsets
</h2><span id='topic+DEPsets'></span>

<h3>Description</h3>

<p>extract two or more IDsets interesection set and complementary set and define the colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEPsets(datalist, colors = c("red", "green", "blue"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DEPsets_+3A_datalist">datalist</code></td>
<td>

<p>a list contains more than two ID sets.
</p>
</td></tr>
<tr><td><code id="DEPsets_+3A_colors">colors</code></td>
<td>

<p>define each ID sets color.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list contains interesection set and complementary set information and colors.
</p>
<table>
<tr><td><code>gene.set</code></td>
<td>
<p>a list of each set ID information.</p>
</td></tr>
<tr><td><code>color.code</code></td>
<td>
<p>the colors of each set</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
Module &lt;- Module_inf(net, imputedData$inf)
group &lt;- gsub("[0-9]+","", colnames(imputedData$intensity))
data &lt;- imputedData
data$inf &lt;- data$inf[1:100,]
data$intensity &lt;- data$intensity[1:100,]
stat &lt;- dataStatInf(data, group, meanmethod = "median",
                    T.test = "pairwise", Aadj = "fdr",
                    Tadj = "fdr", cutoff = FALSE)
stat &lt;- rename_dupnewID(stat, Module, DEPfromMod = TRUE)
stat1 &lt;- stat$new.ID[stat$ad &gt; 1]
stat2 &lt;- stat$new.ID[stat$asym &gt; 1]
datalist &lt;- list(stat1 = stat1, stat2 = stat2)
sets &lt;- DEPsets(datalist)

</code></pre>

<hr>
<h2 id='Dforimpute'>a data used for Dataimpute function.</h2><span id='topic+Dforimpute'></span>

<h3>Description</h3>

<p>a data used for Dataimpute function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dforimpute
</code></pre>


<h3>Format</h3>

<p>A list containing proteomic protein information and quantification data.</p>

<hr>
<h2 id='fc.pos'>
fc.pos
</h2><span id='topic+fc.pos'></span>

<h3>Description</h3>

<p>Pick up proteins based on foldchange and return proteins position in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fc.pos(fc, vs.set2, vs.set1 = "WT",
       cutoff = 1, datatype = c("none", "log2"),
       fctype = "all", order = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fc.pos_+3A_fc">fc</code></td>
<td>

<p>proteomic data of mean value in groups.
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_vs.set2">vs.set2</code></td>
<td>

<p>compared group 2 name
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_vs.set1">vs.set1</code></td>
<td>

<p>compared group 1 name
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_cutoff">cutoff</code></td>
<td>

<p>a numberic value indicated foldchange threshold.
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_datatype">datatype</code></td>
<td>

<p>The quantification data is normal data or log2 data.
This must be (an abbreviation of) one of the strings &quot;<code>none</code>&quot;,&quot;<code>log2</code>&quot;.
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_fctype">fctype</code></td>
<td>

<p>foldchange is ordered by up-regulated or down-regulated or changed
</p>
</td></tr>
<tr><td><code id="fc.pos_+3A_order">order</code></td>
<td>

<p>a logical value indicated that whether ordered by foldchange.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data &lt;- imputedData
relative &lt;- data$relative_value
rownames(relative) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+", "", colnames(relative))
datamean &lt;- groupmean(relative, group, name = FALSE)
fc_1vs2 &lt;- fc.pos(datamean, vs.set2 = "ad", vs.set1 = "ctl",
                  cutoff = 1, datatype = "none",
                  fctype = "up", order = TRUE)
fc_ID &lt;- rownames(relative)[fc_1vs2]


</code></pre>

<hr>
<h2 id='FCSenrichplot'>
FCSenrichplot
</h2><span id='topic+FCSenrichplot'></span>

<h3>Description</h3>

<p>plot of FCS enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FCSenrichplot(FCSenrich, count = 1, p = 0.05, filter = "p",
              plot = TRUE, filename = NULL,filetype = "pdf", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FCSenrichplot_+3A_fcsenrich">FCSenrich</code></td>
<td>

<p>FCS enrichment information which is getted in <code>module_enrich</code> function.
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_count">count</code></td>
<td>

<p>a numeric value. Module will choosed when countnumber is larger than count value .
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_p">p</code></td>
<td>

<p>a numeric value. Module will choosed when any Fisher's extract test p value is less than count value .
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_filter">filter</code></td>
<td>

<p>filter methods.
This must be (an abbreviation of) one of the strings &quot;<code>p</code>&quot;,&quot;<code>p.adj</code>&quot;, &quot;<code>none</code>&quot;.
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_plot">plot</code></td>
<td>

<p>a logical value indicating whether draw enrichment variation trend plot.
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_filename">filename</code></td>
<td>

<p>the filename of plot. The default value is <code>NULL</code> which means no file saving.
The plot will be saved to &quot;plot&quot; folder.
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_filetype">filetype</code></td>
<td>

<p>the file type of plot.  the type should be one of &quot;eps&quot;, &quot;ps&quot;, &quot;tex&quot; (pictex), &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;png&quot;, &quot;bmp&quot;, &quot;svg&quot; or &quot;wmf&quot; (windows only).
</p>
</td></tr>
<tr><td><code id="FCSenrichplot_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data(net)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
pos&lt;-which(Module$moduleNum %in% c(11:13))
up &lt;- changedID(logD[pos,], group, vs.set2 = "ad",vs.set1 = "ctl",
              rank = "foldchange",anova = FALSE, Padj = "none",cutoff = 1,
              datatype = "log2",fctype = "up")
FCSenrich &lt;- Module_Enrich(Module[pos,], up, coln="ori.ID")
FCSenrich &lt;- FCSenrichplot(FCSenrich)

</code></pre>

<hr>
<h2 id='getmoduleHub'>
getmoduleHub
</h2><span id='topic+getmoduleHub'></span>

<h3>Description</h3>

<p>extract PMFG information and get Module hub proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getmoduleHub(data, module, mod_num, coln = "new.ID",
             cor.sig = 0.05, cor.r = 0, cor.adj="none",
             adjustp = TRUE, hub.p = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getmoduleHub_+3A_data">data</code></td>
<td>

<p>proteomic quantification data.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_module">module</code></td>
<td>

<p>module information which is getted in <code>Module_inf</code> function.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_mod_num">mod_num</code></td>
<td>

<p>the module name which module will be calculate.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_coln">coln</code></td>
<td>

<p>column name of module contains protein IDs. it could be matched with &quot;<code>classifiedID</code>&quot;
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_cor.sig">cor.sig</code></td>
<td>

<p>a numberic value indicated that correlation p value less than cor.sig will be picked.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_cor.r">cor.r</code></td>
<td>

<p>a numberic value indicated that correlation r value larger than cor.r will be picked.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_cor.adj">cor.adj</code></td>
<td>

<p>P value correction method. method information can see in <code>p.adjust.method</code>
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_adjustp">adjustp</code></td>
<td>

<p>a logical value indicating whether pick hub protein by FDR methods.
</p>
</td></tr>
<tr><td><code id="getmoduleHub_+3A_hub.p">hub.p</code></td>
<td>

<p>a numberic value indicated that hub proteins are p value less than hub.p.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list contains PMFG network information.
list(hub = hubgene, degreeStat = Stat, graph = g, PMFG = gg)
</p>
<table>
<tr><td><code>hub</code></td>
<td>
<p> hub information.</p>
</td></tr>
<tr><td><code>degreeStat</code></td>
<td>
<p>degree statistics information</p>
</td></tr>
<tr><td><code>graph</code></td>
<td>
<p>the original graph data frame  </p>
</td></tr>
<tr><td><code>PMFG</code></td>
<td>
<p>PMFG graph data frame </p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
Mod10 &lt;- getmoduleHub(logD, Module, 10, coln = "ori.ID", adjustp = FALSE)
if (requireNamespace("MEGENA", quietly = TRUE)) {
 library(MEGENA)
 plot_subgraph(module = Mod10$degreeStat$gene,
              hub = Mod10$hub,PFN = Mod10$PMFG,
              node.default.color = "black",
              gene.set = NULL,color.code = c("grey"),show.legend = TRUE,
              label.hubs.only = TRUE,hubLabel.col = "red",hubLabel.sizeProp = 0.5,
              show.topn.hubs = 10,node.sizeProp = 13,label.sizeProp = 13,
              label.scaleFactor = 10,layout = "kamada.kawai")
}
</code></pre>

<hr>
<h2 id='groupmean'>
groupmean
</h2><span id='topic+groupmean'></span>

<h3>Description</h3>

<p>mean of sample group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupmean(data, group, method = c("mean", "median"), name = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="groupmean_+3A_data">data</code></td>
<td>

<p>protein quantification data. column is sample. row is protein ID.
</p>
</td></tr>
<tr><td><code id="groupmean_+3A_group">group</code></td>
<td>

<p>sample group information
</p>
</td></tr>
<tr><td><code id="groupmean_+3A_method">method</code></td>
<td>

<p>Arithmetic mean of sample group or median of sample group.
This must be (an abbreviation of) one of the strings &quot;<code>mean</code>&quot;,&quot;<code>median</code>&quot;.
</p>
</td></tr>
<tr><td><code id="groupmean_+3A_name">name</code></td>
<td>

<p>a logical value indicated whether add &quot;mean&quot; or &quot;median&quot; in sample group name.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
group &lt;- gsub("[0-9]+","", colnames(logD))
datamean &lt;- groupmean(logD, group, name = FALSE)
</code></pre>

<hr>
<h2 id='ID_match'>
homolog protein Uniprot ID transformation
</h2><span id='topic+ID_match'></span>

<h3>Description</h3>

<p>homolog protein Uniprot ID match
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ID_match(data, db1.path = NULL, db2.path = NULL,out.folder = NULL,
         blast.path = NULL,evalue = 0.1, verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ID_match_+3A_data">data</code></td>
<td>

<p>dataset of protein information.Column Names should contain &quot;ori.ID&quot; and &quot;ENTRY.NAME&quot;. &quot;ori.ID&quot; is Uniprot ID
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_db1.path">db1.path</code></td>
<td>

<p>fasta file, database of transfered species
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_db2.path">db2.path</code></td>
<td>

<p>fasta file, database of original species
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_out.folder">out.folder</code></td>
<td>

<p>blast result output folder, the folder path should be the same with db1.path
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_blast.path">blast.path</code></td>
<td>

<p>blast+ software install path
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_evalue">evalue</code></td>
<td>

<p>blast threshold, the lower means more rigorous
</p>
</td></tr>
<tr><td><code id="ID_match_+3A_verbose">verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, 1 means have Diagnostic Messages.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>homolog protein Uniprot ID match is based on the ENTRY.NAME, gene name and sequence homophyly in two different species or different version of database.
</p>


<h3>Value</h3>

<p>a data.frame included 4 columns: ori.ID, ENTRY.NAME, new.ID, match.type.
</p>


<h3>Note</h3>

<p>This function should install 'blast+' software, Version 2.7.1.
'blast+' download website:https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/
If unstall 'blast+' software, it could use R function replaced, but it will take a lot of time.
db1.path, db2.path, out.folder are both need the complete path. Out.folder and db1.path should be in the same folder. Path should have no special character.
data should have colname: ori.ID, ENTRY.NAME.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'># suggested to install blast+ software

# it will take a long time without blast+ software
data(Sample_ID_data)
if(requireNamespace("Biostrings", quietly = TRUE)){
  out.folder = tempdir();
  write.table(Sample_ID_data$db1,file.path(out.folder,"db1.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  write.table(Sample_ID_data$db2,file.path(out.folder,"db2.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  data &lt;- ID_match(Sample_ID_data$ID_match_data,
                   db1.path = file.path(out.folder,"db1.fasta"),
                   db2.path = file.path(out.folder,"db2.fasta"),
                   out.folder = out.folder,
                   blast.path = NULL,
                   evalue = 0.1, verbose = 1)
 file.remove( file.path(out.folder,"db1.fasta"),
              file.path(out.folder,"db2.fasta"))
}

</code></pre>

<hr>
<h2 id='imputedData'>an imputed data used for downstream analysis</h2><span id='topic+imputedData'></span>

<h3>Description</h3>

<p>an imputed data used for downstream analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imputedData
</code></pre>


<h3>Format</h3>

<p>A list containing proteomic protein information and quantification data.</p>

<hr>
<h2 id='MaxQdataconvert'>
one-step to extract 'Maxquant' quantification data and convert
</h2><span id='topic+MaxQdataconvert'></span>

<h3>Description</h3>

<p>'Maxquant' quantification data extract and homolog protein Uniprot ID match.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MaxQdataconvert(pgfilename, IDname = "Majority.protein.IDs",
                IDtype = c("MaxQ","none"), CONremove = TRUE,
                justID = TRUE, status1 = TRUE, ENTRY1 = TRUE,
                db1.path = NULL, db2.path = NULL,
                out.folder = NULL, blast.path = NULL,
                savecsvpath = NULL, csvfilename = NULL,
                verbose = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MaxQdataconvert_+3A_pgfilename">pgfilename</code></td>
<td>

<p>'Maxquant' quantification file &quot;protein groups.txt&quot;
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_idname">IDname</code></td>
<td>

<p>The column name of uniprot ID. The default value is &quot;<code>Majority.protein.IDs</code>&quot;&quot; which is the column name in MaxQuant data.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_idtype">IDtype</code></td>
<td>

<p>&quot;<code>MaxQ</code>&quot; means <code>proteinGroups</code> is 'Maxquant' quantification data, &quot;<code>none</code>&quot; means other type data. This must be (an abbreviation of) one of the strings: &quot;<code>MaxQ</code>&quot;,&quot;<code>none</code>&quot;.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_conremove">CONremove</code></td>
<td>

<p>a logical value indicated whether remove contaminant IDs. When IDtype is &quot;none&quot;, it will remove unmatch ID compared with database2.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_justid">justID</code></td>
<td>

<p>a logical value indicated whether only extract ID when IDtype is &quot;MaxQ&quot;.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_status1">status1</code></td>
<td>

<p>a logical value indicated whether extract the first ID status when IDtype is &quot;MaxQ&quot;.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_entry1">ENTRY1</code></td>
<td>

<p>a logical value indicated whether extract the first ID ENTRY NAME when IDtype is &quot;MaxQ&quot;.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_db1.path">db1.path</code></td>
<td>

<p>fasta file, database of transfered species
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_db2.path">db2.path</code></td>
<td>

<p>fasta file, database of original species
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_out.folder">out.folder</code></td>
<td>

<p>blast result output folder, the folder path should be the same with db1.path
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_blast.path">blast.path</code></td>
<td>

<p>blast+ software install path
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_savecsvpath">savecsvpath</code></td>
<td>

<p>the information of csv file name output path. The default value means don't save csv file.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_csvfilename">csvfilename</code></td>
<td>

<p>the name of csv file which the data are to be output. The default value means don't save csv file.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_verbose">verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, higher values make the output progressively more and more verbose.
</p>
</td></tr>
<tr><td><code id="MaxQdataconvert_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>one-step to extract MaxQuant or other quantification data and convert.
The function contain <a href="#topic+ID_match">ID_match</a> function.
</p>


<h3>Value</h3>

<p>a list of proteomic information.
</p>
<table>
<tr><td><code>protein_IDs</code></td>
<td>
<p>Portein IDs which is <code>IDname</code> column information.</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>Quantification intensity informaton. When  <code>IDtype</code> is &quot;<code>none</code>&quot;, it is the  <code>QuanCol</code> columns information.</p>
</td></tr>
<tr><td><code>iBAQ</code></td>
<td>
<p>Quantification iBAQ intensity informaton.(only for <code>IDtype</code> is &quot;<code>MaxQ</code>&quot;)</p>
</td></tr>
<tr><td><code>LFQ</code></td>
<td>
<p>Quantification LFQ intensity informaton.(only for <code>IDtype</code> is &quot;<code>MaxQ</code>&quot;)</p>
</td></tr>

</table>


<h3>Note</h3>

<p>The function should install 'blast+' software, Version 2.7.1.
'blast+' download website:https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/
db1.path, db2.path, out.folder are both need the complete path. Out.folder and db1.path should be in the same folder. Path should have no special character.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>See Also</h3>

<p><a href="#topic+ID_match">ID_match</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># suggested to install blast+ software

# it will take a long time without blast+ software
data(Sample_ID_data)
if(requireNamespace("Biostrings", quietly = TRUE)){
  out.folder = tempdir();
  write.table(Sample_ID_data$db1,file.path(out.folder,"db1.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  write.table(Sample_ID_data$db2,file.path(out.folder,"db2.fasta"),
              quote = FALSE,row.names = FALSE, col.names = FALSE);
  write.table(Sample_ID_data$pginf,
              file = file.path(out.folder,"proteingroups.txt"),
              quote = FALSE,
              sep = "\t",dec = ".", row.names = FALSE, col.names = TRUE )
  Maxdata &lt;- MaxQdataconvert(file.path(out.folder,"proteingroups.txt"),
                             IDtype = "MaxQ",
                             db1.path = file.path(out.folder,"db1.fasta"),
                             db2.path = file.path(out.folder,"db2.fasta"),
                             out.folder = out.folder,
                             blast.path = NULL)
  file.remove( file.path(out.folder,"db1.fasta"),
               file.path(out.folder,"db2.fasta"),
               file.path(out.folder,"proteingroups.txt") )
  }

</code></pre>

<hr>
<h2 id='MaxQprotein'>
read proteomic quantification data and seperate the protein information and quantification information.
</h2><span id='topic+MaxQprotein'></span>

<h3>Description</h3>

<p>The function will seperate data into 4 parts: protein information, intensity, iBAQ and LFQ (iBAQ and LFQ only fit for 'MaxQuant' software result). For MaxQ data, it can remove the contaminant and reverse protein.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MaxQprotein(proteinGroups, IDname = "Majority.protein.IDs",
            IDtype = "MaxQ", remove = TRUE, QuanCol = NULL,
            verbose = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MaxQprotein_+3A_proteingroups">proteinGroups</code></td>
<td>

<p>the proteomic quantification data
</p>
</td></tr>
<tr><td><code id="MaxQprotein_+3A_idname">IDname</code></td>
<td>

<p>The column name of uniprot ID. The default value is &quot;<code>Majority.protein.IDs</code>&quot;&quot; which is the column name in MaxQuant data.
</p>
</td></tr>
<tr><td><code id="MaxQprotein_+3A_idtype">IDtype</code></td>
<td>

<p>&quot;<code>MaxQ</code>&quot; means <code>proteinGroups</code> is Maxquant quantification data, &quot;<code>none</code>&quot; means other type data. This must be (an abbreviation of) one of the strings: &quot;<code>MaxQ</code>&quot;,&quot;<code>none</code>&quot;.
</p>
</td></tr>
<tr><td><code id="MaxQprotein_+3A_remove">remove</code></td>
<td>

<p>a logical value indicated whether remove contaminant and reverse ID.
</p>
</td></tr>
<tr><td><code id="MaxQprotein_+3A_quancol">QuanCol</code></td>
<td>

<p>The quantification data columns. It's only needed when <code>IDtype</code> is &quot;<code>none</code>&quot;.
When <code>IDtype</code> is &quot;<code>none</code>&quot; and <code>QuanCol</code> is not given, the intensity will auto extract all columns except <code>IDname</code> as quantification data. It may have error in next analysis.
</p>
</td></tr>
<tr><td><code id="MaxQprotein_+3A_verbose">verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, 1 means have Diagnostic Messages.
</p>
</td></tr>
</table>


<h3>Value</h3>



<p>a list of proteomic information.
</p>
<table>
<tr><td><code>protein_IDs</code></td>
<td>
<p>Portein IDs which is <code>IDname</code> column information.</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>Quantification intensity informaton. When  <code>IDtype</code> is &quot;<code>none</code>&quot;, it is the  <code>QuanCol</code> columns information.</p>
</td></tr>
<tr><td><code>iBAQ</code></td>
<td>
<p>Quantification iBAQ intensity informaton.(only for <code>IDtype</code> is &quot;<code>MaxQ</code>&quot;)</p>
</td></tr>
<tr><td><code>LFQ</code></td>
<td>
<p>Quantification LFQ intensity informaton.(only for <code>IDtype</code> is &quot;<code>MaxQ</code>&quot;)</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ProteomicData)
# example for MaxQ Data
MaxQdata &lt;- MaxQprotein(ProteomicData$MaxQ)
# example for other type Data
otherdata &lt;- MaxQprotein(ProteomicData$none, IDname = "Protein",
                         IDtype = "none", QuanCol = 2:9)

</code></pre>

<hr>
<h2 id='ME_inf'>
module eigengenes information
</h2><span id='topic+ME_inf'></span>

<h3>Description</h3>

<p>put sample names as rownames in WGCNA module eigenvalue data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ME_inf(MEs, data, intensity.type = "LFQ", rowname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ME_inf_+3A_mes">MEs</code></td>
<td>

<p>module eigenvalue which is calculated in WGCNA package.
</p>
</td></tr>
<tr><td><code id="ME_inf_+3A_data">data</code></td>
<td>

<p>protein quantification data. column is sample. row is protein ID.
</p>
</td></tr>
<tr><td><code id="ME_inf_+3A_intensity.type">intensity.type</code></td>
<td>

<p>quantification data type, which can help extract sample name.
This must be (an abbreviation of) one of the strings &quot;<code>LFQ</code>&quot;,&quot;<code>intensity</code>&quot;,&quot;<code>iBAQ</code>&quot;,&quot;<code>none</code>&quot;.
</p>
</td></tr>
<tr><td><code id="ME_inf_+3A_rowname">rowname</code></td>
<td>

<p>sample names when &quot;<code>intensity.type</code>&quot; is &quot;<code>none</code>&quot;, rowname will be used.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
MEs &lt;- ME_inf(net$MEs, logD)
</code></pre>

<hr>
<h2 id='modpcomp'>
modpcomp
</h2><span id='topic+modpcomp'></span>

<h3>Description</h3>

<p>extract module pca component
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modpcomp(data, colors, nPC = 2,
         plot = FALSE, filename = NULL, group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modpcomp_+3A_data">data</code></td>
<td>

<p>protein quantification data. column is sample. row is protein ID.
</p>
</td></tr>
<tr><td><code id="modpcomp_+3A_colors">colors</code></td>
<td>

<p>protein and module information. which is calculated in WGCNA package.
</p>
</td></tr>
<tr><td><code id="modpcomp_+3A_npc">nPC</code></td>
<td>

<p>how many PCA component will saved.
</p>
</td></tr>
<tr><td><code id="modpcomp_+3A_plot">plot</code></td>
<td>

<p>a logical value indicating whether draw PCA plot. This function need load <span class="pkg">ggfortify</span> first.
</p>
</td></tr>
<tr><td><code id="modpcomp_+3A_filename">filename</code></td>
<td>

<p>The filename of plot. The default value is <code>NULL</code> which means no file saving.
The plot will be saved to &quot;plot&quot; folder and saved in pdf format.
</p>
</td></tr>
<tr><td><code id="modpcomp_+3A_group">group</code></td>
<td>

<p>sample group information.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
Module_PCA &lt;- modpcomp(logD, net$colors)

# if plot PCA and plot module 6 PCA
group &lt;- gsub("[0-9]+", "", colnames(logD))
pos &lt;- which(net$colors == 6)
if (requireNamespace("ggfortify", quietly = TRUE)){
require("ggfortify")
Module_PCA &lt;- modpcomp(logD[pos,], net$colors[pos], plot = TRUE, group = group)
}

</code></pre>

<hr>
<h2 id='Module_Enrich'>
Module_Enrich
</h2><span id='topic+Module_Enrich'></span>

<h3>Description</h3>

<p>Enrichment analysis of a sets of proteins in all modules. The function offered two enrichment methods:ORA and FCS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Module_Enrich(module, classifiedID, enrichtype = "FCS",
              coln = "new.ID", datainf = NULL, p.adj.method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Module_Enrich_+3A_module">module</code></td>
<td>

<p>module information which is getted in <code>Module_inf</code> function.
</p>
</td></tr>
<tr><td><code id="Module_Enrich_+3A_classifiedid">classifiedID</code></td>
<td>

<p>a sets of protein IDs which is ordered by change value/ p value and so on.
</p>
</td></tr>
<tr><td><code id="Module_Enrich_+3A_enrichtype">enrichtype</code></td>
<td>

<p>enrichment method.
This must be (an abbreviation of) one of the strings &quot;<code>FCS</code>&quot;,&quot;<code>ORA</code>&quot;.
&quot;<code>FCS</code>&quot; means  analyzes step-by-step a proteins list which is ordered by change ratio/ p value and so on.
&quot;<code>ORA</code>&quot; means  analyzes a proteins list by Fisher's extract test.
</p>
</td></tr>
<tr><td><code id="Module_Enrich_+3A_coln">coln</code></td>
<td>

<p>column name of module contains protein IDs. it could be matched with &quot;<code>classifiedID</code>&quot;
</p>
</td></tr>
<tr><td><code id="Module_Enrich_+3A_datainf">datainf</code></td>
<td>

<p>proteomic data protein ID information.
The default value is &quot;<code>NULL</code>&quot;. which is means that the &quot;<code>classifiedID</code>&quot; come from proteomic information is the same as the module construction proteomic information.
If they are different, proteomic data information should be given.
</p>
</td></tr>
<tr><td><code id="Module_Enrich_+3A_p.adj.method">p.adj.method</code></td>
<td>

<p>p adjust methods of multiple comparisons.
it can seen in <code>p.adjust.methods</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list contains classifiedID enrichment information.
</p>
<table>
<tr><td><code>Counts</code></td>
<td>
<p>the counts of classifiedID in module.</p>
</td></tr>
<tr><td><code>module.size</code></td>
<td>
<p>the number of module ID</p>
</td></tr>
<tr><td><code>module.name</code></td>
<td>
<p>module name</p>
</td></tr>
<tr><td><code>precent</code></td>
<td>
<p>counts divided module.size </p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>enrichment p value in each module</p>
</td></tr>
<tr><td><code>p.adj</code></td>
<td>
<p>enrichment p.adj value in each module</p>
</td></tr>
<tr><td><code>Z.score</code></td>
<td>
<p>Z score is -log2 P value.</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
up &lt;- changedID(logD, group, vs.set2 = "ad",vs.set1 = "ctl",
              rank = "foldchange",anova = FALSE, Padj = "none",cutoff = 1,
              datatype = "log2",fctype = "up")
FCSenrich &lt;- Module_Enrich(Module, up, coln="ori.ID")
</code></pre>

<hr>
<h2 id='Module_inf'>
Module and protein information.
</h2><span id='topic+Module_inf'></span>

<h3>Description</h3>

<p>module and protein information match
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Module_inf(net, inf, inftype = "Convert", IDname = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Module_inf_+3A_net">net</code></td>
<td>

<p>module network which is calculated in WGCNA package.
</p>
</td></tr>
<tr><td><code id="Module_inf_+3A_inf">inf</code></td>
<td>

<p>proteome quantification data information which contains protein IDs.
</p>
</td></tr>
<tr><td><code id="Module_inf_+3A_inftype">inftype</code></td>
<td>

<p>data information type.
This must be (an abbreviation of) one of the strings &quot;<code>Convert</code>&quot;,&quot;<code>MaxQ</code>&quot;,&quot;<code>none</code>&quot;.
&quot;<code>Convert</code>&quot; means protein ID is converted by <code>MaxquantDataconvert</code> function.
&quot;<code>MaxQ</code>&quot; means original Maxquant software quantification data information.
</p>
</td></tr>
<tr><td><code id="Module_inf_+3A_idname">IDname</code></td>
<td>

<p>IDname is &quot;<code>inf</code>&quot; column names which will extract.
</p>
</td></tr>
<tr><td><code id="Module_inf_+3A_...">...</code></td>
<td>

<p>other argument.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
Module &lt;- Module_inf(net, data$inf)
</code></pre>

<hr>
<h2 id='moduleID'>
extract intersection ID between dataset and module
</h2><span id='topic+moduleID'></span>

<h3>Description</h3>

<p>extract intersection ID between dataset and one of module
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moduleID(inf, module, num, coln = "new.ID")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moduleID_+3A_inf">inf</code></td>
<td>

<p>dataset protein ID information. a vector of protein IDs.
</p>
</td></tr>
<tr><td><code id="moduleID_+3A_module">module</code></td>
<td>

<p>module information which is getted in <code>Module_inf</code> function.
</p>
</td></tr>
<tr><td><code id="moduleID_+3A_num">num</code></td>
<td>

<p>module number which will extract to compared with dataset ID information.
</p>
</td></tr>
<tr><td><code id="moduleID_+3A_coln">coln</code></td>
<td>

<p>column names of module protein IDs.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>column <code>coln</code> information in <code>module</code> when module number is <code>num</code> intersect with inf.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
up &lt;- changedID(logD, group, vs.set2 = "ad",vs.set1 = "ctl",
              rank = "foldchange",anova = FALSE, Padj = "none",cutoff = 1,
              datatype = "log2",fctype = "up")
intersection &lt;- moduleID(up, Module, 5, coln = "ori.ID")
</code></pre>

<hr>
<h2 id='multi.t.test'>
multi.t.test
</h2><span id='topic+multi.t.test'></span>

<h3>Description</h3>

<p>multiple comparisons t test and choose significant proteins in proteomic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi.t.test(data, group,
             sig = 0.05, Adj.sig = TRUE,
             grpAdj = "bonferroni",
             geneAdj = "fdr", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi.t.test_+3A_data">data</code></td>
<td>

<p>protein quantification data. column is sample. row is protein ID.
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_group">group</code></td>
<td>

<p>sample group information
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_sig">sig</code></td>
<td>

<p>siginificant P value threshold. The default is 0.05.
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_adj.sig">Adj.sig</code></td>
<td>

<p>a logical value indicated that whether adjust P-values for multiple proteins comparisons in each two groups.
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_grpadj">grpAdj</code></td>
<td>

<p>adjust multiple groups comparisions P-value in each two groups. The default is <code>"bonferroni"</code>.
it can seen in <code>p.adjust.methods</code>.
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_geneadj">geneAdj</code></td>
<td>

<p>adjust multiple proteins comparisions P-value in each group. The default is <code>"fdr"</code>.
it can seen in <code>p.adjust.methods</code>.
</p>
</td></tr>
<tr><td><code id="multi.t.test_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+", "", colnames(logD))
Tsig_P &lt;- multi.t.test(logD[1:100,], group, Adj.sig = FALSE, geneAdj = "fdr")

</code></pre>

<hr>
<h2 id='net'>A network which is obtained by WGCNA-package blockwiseModule function.
it can short the time of Module analysis example time.</h2><span id='topic+net'></span>

<h3>Description</h3>

<p>A network which is obtained by WGCNA-package blockwiseModule function.
it can short the time of Module analysis example time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>net
</code></pre>


<h3>Format</h3>

<p>A list containing network module information, ME information and dendrograms information.</p>

<hr>
<h2 id='P.G.extract'>
Protein Groups information extract.
</h2><span id='topic+P.G.extract'></span>

<h3>Description</h3>

<p>uniprot ID, ENTRYNAME and status information extract.(only fit for 'MaxQuant' data.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>P.G.extract(inf, ncol = 4, justID = FALSE,
            status1 = TRUE, ENTRY1 = TRUE, ID1 = TRUE,
            sp1 = TRUE, onlysp = FALSE, verbose = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="P.G.extract_+3A_inf">inf</code></td>
<td>

<p>protein groups IDs information.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_ncol">ncol</code></td>
<td>

<p>column numbers of output result.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_justid">justID</code></td>
<td>

<p>a logical value indicated whether only extract uniprot ID.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_status1">status1</code></td>
<td>

<p>a logical value indicated whether extract the first ID status.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_entry1">ENTRY1</code></td>
<td>

<p>a logical value indicated whether extract the first ID ENTRY NAME.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_id1">ID1</code></td>
<td>

<p>a logical value indicated whether extract the first ID UNIPROT ACCESSION.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_sp1">sp1</code></td>
<td>

<p>a logical value indicated whether extract the first sp ID as the new first ID.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_onlysp">onlysp</code></td>
<td>

<p>a logical value indicated whether only keep sp ID in the second or later ID result.
</p>
</td></tr>
<tr><td><code id="P.G.extract_+3A_verbose">verbose</code></td>
<td>

<p>integer level of verbosity. Zero means silent, 1 means have Diagnostic Messages.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ProteomicData)
MaxQdata &lt;- MaxQprotein(ProteomicData$MaxQ)
inf &lt;- P.G.extract(MaxQdata$protein_IDs, justID = TRUE, status = TRUE, ENTRY = TRUE)

</code></pre>

<hr>
<h2 id='ProteomicData'>Small example proteomic quantification data extract from txt or csv file used to
demonstrate data extract function.</h2><span id='topic+ProteomicData'></span>

<h3>Description</h3>

<p>Small example proteomic quantification data extract from txt or csv file used to
demonstrate data extract function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProteomicData
</code></pre>


<h3>Format</h3>

<p>A list containing 2 data frame: one is MaxQ proteomic data; another is other type of proteomic data</p>

<hr>
<h2 id='rename_dupnewID'>
rename_dupnewID
</h2><span id='topic+rename_dupnewID'></span>

<h3>Description</h3>

<p>rename the duplicated newID in moduleinf and renew the ID in DEPstat
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_dupnewID(DEPstat, moduleinf, DEPfromMod = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_dupnewID_+3A_depstat">DEPstat</code></td>
<td>

<p>a dataframe contains columns:&quot;new.ID&quot; and &quot;ori.ID&quot;. it can get from <code>dataStatInf</code>.
</p>
</td></tr>
<tr><td><code id="rename_dupnewID_+3A_moduleinf">moduleinf</code></td>
<td>

<p>a dataframe contains columns:&quot;new.ID&quot; and &quot;ori.ID&quot;. it can get from <code>Module_inf</code>.
</p>
</td></tr>
<tr><td><code id="rename_dupnewID_+3A_depfrommod">DEPfromMod</code></td>
<td>

<p>a logical value indicated that whether DEPstat and moduleinf is getted from the same datasets.
The default value is FALSE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame contains DEPstat information and renewed the new.ID column.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
Module &lt;- Module_inf(net, imputedData$inf)
group &lt;- gsub("[0-9]+","", colnames(imputedData$intensity))
data &lt;- imputedData
data$inf &lt;- data$inf[1:100,]
data$intensity &lt;- data$intensity[1:100,]
stat &lt;- dataStatInf(data, group, meanmethod = "median",
                    T.test = "pairwise", Aadj = "fdr",
                    Tadj = "fdr", cutoff = FALSE)
stat &lt;- rename_dupnewID(stat, Module, DEPfromMod = TRUE)

</code></pre>

<hr>
<h2 id='Sample_ID_data'>A small uniprot ID information used to do ID convert in example.</h2><span id='topic+Sample_ID_data'></span>

<h3>Description</h3>

<p>A small uniprot ID information used to do ID convert in example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sample_ID_data
</code></pre>


<h3>Format</h3>

<p>A data frame corresponding uniprot ID information</p>

<hr>
<h2 id='single_mod_enrichplot'>
single_mod_enrichplot
</h2><span id='topic+single_mod_enrichplot'></span>

<h3>Description</h3>

<p>FCS enrichment analysis of a sets of proteins in one module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_mod_enrichplot(module, Mod_Nam, classifiedID,
                      coln = "new.ID", datainf = NULL,
                      plot = TRUE, filename = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single_mod_enrichplot_+3A_module">module</code></td>
<td>

<p>module information which is getted in <code>Module_inf</code> function.
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_mod_nam">Mod_Nam</code></td>
<td>

<p>the module name which module will be calculate.
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_classifiedid">classifiedID</code></td>
<td>

<p>a sets of protein IDs which is ordered by change value/ p value and so on.
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_coln">coln</code></td>
<td>

<p>column name of module contains protein IDs. it could be matched with &quot;<code>classifiedID</code>&quot;
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_datainf">datainf</code></td>
<td>

<p>proteomic data protein ID information.
The default value is &quot;<code>NULL</code>&quot;. which is means that the &quot;<code>classifiedID</code>&quot; come from proteomic information is the same as the module construction proteomic information.
If they are different, proteomic data information should be given.
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_plot">plot</code></td>
<td>

<p>a logical value indicating whether draw enrichment variation trend plot.
</p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_filename">filename</code></td>
<td>
<p>the filename of plot. The default value is <code>NULL</code> which means no file saving.
The plot will be saved to &quot;plot&quot; folder and saved in pdf format. </p>
</td></tr>
<tr><td><code id="single_mod_enrichplot_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net)
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
group &lt;- gsub("[0-9]+","", colnames(logD))
Module &lt;- Module_inf(net, data$inf)
up &lt;- changedID(logD, group, vs.set2 = "ad",vs.set1 = "ctl",
                rank = "foldchange",anova = FALSE, Padj = "none", cutoff = 1,
                datatype = "log2", fctype = "up")
m5enrich &lt;- single_mod_enrichplot(Module, 5, up, coln="ori.ID")

</code></pre>

<hr>
<h2 id='SoftThresholdScaleGraph'>
SoftThresholdScaleGraph
</h2><span id='topic+SoftThresholdScaleGraph'></span>

<h3>Description</h3>

<p>pick soft thresholding powers for WGCNA analysis and plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SoftThresholdScaleGraph(data,
                        xlab = "Soft Threshold (power)",
                        ylab = "Scale Free Topology Model Fit, signed R^2",
                        main = "Scale independence",
                        filename = NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SoftThresholdScaleGraph_+3A_data">data</code></td>
<td>

<p>protein quantification data. row is sample. column is protein ID.
</p>
</td></tr>
<tr><td><code id="SoftThresholdScaleGraph_+3A_xlab">xlab</code></td>
<td>

<p>x axis label
</p>
</td></tr>
<tr><td><code id="SoftThresholdScaleGraph_+3A_ylab">ylab</code></td>
<td>

<p>y axis label
</p>
</td></tr>
<tr><td><code id="SoftThresholdScaleGraph_+3A_main">main</code></td>
<td>

<p>plot title
</p>
</td></tr>
<tr><td><code id="SoftThresholdScaleGraph_+3A_filename">filename</code></td>
<td>

<p>the filename of plot. The default value is <code>NULL</code> which means no file saving.
The plot will be saved to &quot;plot&quot; folder and saved in pdf format.
</p>
</td></tr>
<tr><td><code id="SoftThresholdScaleGraph_+3A_...">...</code></td>
<td>

<p>Other arguments in function <code>pickSoftThreshold</code> in WGCNA package.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>pick soft thresholding powers for WGCNA analysis and plot. The function is also can replaced by &quot;<code>pickSoftThreshold</code>&quot; function in WGCNA package.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>powerEstimate</code></td>
<td>
<p>the lowest power fit for scale free topology.</p>
</td></tr>
<tr><td><code>fitIndices</code></td>
<td>
<p>a data frame containing the fit indices for scale free topology. </p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>See Also</h3>

<p><code>pickSoftThreshold</code> in WGCNA package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#it will take some times
data(imputedData)
data &lt;- imputedData
logD &lt;- data$log2_value
rownames(logD) &lt;- data$inf$ori.ID
if (requireNamespace("WGCNA", quietly = TRUE))
  sft &lt;- SoftThresholdScaleGraph(t(logD))


</code></pre>

<hr>
<h2 id='wgcnatest'>
wgcnatest
</h2><span id='topic+wgcnatest'></span>

<h3>Description</h3>

<p>The major parameter optimization in function <code>blockwiseModules</code> in WGCNA package. The function will do a series of network construction by change various parameter in <code>blockwiseModules</code> and record the result. (it will take a long time)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wgcnatest(data, power = NULL, maxBlockSize = 5000,
          corType = "pearson", networkType = "unsigned",
          TOMType = "unsigned",detectCutHeight = NULL,
          deepSplit = TRUE, minModSize = TRUE,
          minKMEtoStay = TRUE,minCoreKME = FALSE,
          reassignThreshold = FALSE,mergeCutHeight = FALSE,
          pamRespectsDendro = FALSE,
          maxModNum = 30,minModNum = 8,MaxMod0ratio = 0.3,
          ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wgcnatest_+3A_data">data</code></td>
<td>

<p>protein quantification data used in network construction. Row is sample. Column is protein ID.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_power">power</code></td>
<td>

<p>Soft-thresholding power for network construction. The default value is NULL. it will run <code>pickSoftThreshold</code> function in WGCNA package to pick the lowest appropriate power.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_maxblocksize">maxBlockSize</code></td>
<td>

<p>integer giving maximum block size for module detection.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_cortype">corType</code></td>
<td>

<p>one of &quot;pearson&quot; and &quot;bicor&quot;.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_networktype">networkType</code></td>
<td>

<p>one of &quot;signed hybrid&quot;, &quot;unsigned&quot;, &quot;signed&quot;.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_tomtype">TOMType</code></td>
<td>

<p>one of &quot;none&quot;, &quot;unsigned&quot;, &quot;signed&quot;,&quot;signed Nowick&quot;, &quot;unsigned 2&quot;, &quot;signed 2&quot; and &quot;signed Nowick 2&quot;..
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_detectcutheight">detectCutHeight</code></td>
<td>

<p>dendrogram cut height for module detection.
The default value is NULL, which means it will calculate the cutheight through correlation r when p value is 0.05. When the value is larger than 0.995, it will set to detectCutHeight or 0.995.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_deepsplit">deepSplit</code></td>
<td>

<p>The default value is TRUE, which means the function will test deepSplit from 0 to 4. If the value is FALSE, deepSplit is 2. You also can setting integer value between 0 and 4 by yourself.
integer value between 0 and 4.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_minmodsize">minModSize</code></td>
<td>

<p>minimum module size for module detection.
The default value is TRUE, which means the function will test 15, 20, 30, 50. If the value is FALSE, minModSize is 20. You also can setting integer value by yourself.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_minkmetostay">minKMEtoStay</code></td>
<td>

<p>The default value is TRUE, which means the function will test 0.1, 0.2, 0.3. If the value is FALSE, minKMEtoStay is 0.3. You also can setting value by yourself.
Value between 0 to 1.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_mincorekme">minCoreKME</code></td>
<td>

<p>The default value is FALSE, minCoreKME is 0.5. If the value is TRUE, which means the function will test 0.4 and 0.5. You also can setting value by yourself.
Value between 0 to 1.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_reassignthreshold">reassignThreshold</code></td>
<td>

<p>p-value ratio threshold for reassigning genes between modules.
The default value is FALSE, reassignThreshold is 1e-6. If the value is TRUE, which means the function will test 0.01 and 0.05. You also can setting value by yourself.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_mergecutheight">mergeCutHeight</code></td>
<td>

<p>dendrogram cut height for module merging.
The default value is FALSE, mergeCutHeight is 0.15. If the value is TRUE, which means the function will test 0.15, 0.3 and 0.45. You also can setting value by yourself.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_pamrespectsdendro">pamRespectsDendro</code></td>
<td>

<p>a logical value indicated that whether do pamStage or not.
More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_maxmodnum">maxModNum</code></td>
<td>

<p>The maximum module number. If network construction make more than maxModnum of modules. The result will not record.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_minmodnum">minModNum</code></td>
<td>

<p>The mininum module number. If network construction make less than minModNum of modules. The result will not record.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_maxmod0ratio">MaxMod0ratio</code></td>
<td>

<p>The maximum Mod0 protein numbers ratio in total proteins. If network construction make more than MaxMod0ratio in module 0. The result will not record.
</p>
</td></tr>
<tr><td><code id="wgcnatest_+3A_...">...</code></td>
<td>

<p>Other arguments from <code>blockwiseModules</code> in WGCNA packag.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>More information can get from <code>blockwiseModules</code> in WGCNA package.
</p>


<h3>Value</h3>

<p>a data.frame contains protein number in each module and the parameter information.
</p>


<h3>Author(s)</h3>

<p>Kefu Liu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(imputedData)
wgcnadata &lt;- t(imputedData$intensity)
sft &lt;- SoftThresholdScaleGraph(wgcnadata)
# It will take a lot of time
if (requireNamespace("WGCNA", quietly = TRUE)){
require("WGCNA")
WGCNAadjust &lt;- wgcnatest(wgcnadata, power = sft$powerEstimate)
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
