<!DOCTYPE html><html><head><title>Help for package deconstructSigs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {deconstructSigs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#changeColumnNames'><p>Re-formats column names</p></a></li>
<li><a href='#deconstructSigs'><p>deconstructSigs</p></a></li>
<li><a href='#example.output'><p>Example output of whichSignatures()</p></a></li>
<li><a href='#findSeed'><p>Seeds weight matrix</p></a></li>
<li><a href='#formatContexts'><p>Formats the data for use in plotSignatures()</p></a></li>
<li><a href='#getError'><p>Calculates error</p></a></li>
<li><a href='#getTriContextFraction'><p>Calculates trinucleotide context fraction</p></a></li>
<li><a href='#golden.section.search'><p>Golden section search</p></a></li>
<li><a href='#makePie'><p>Plots the weights from whichSignatures()</p></a></li>
<li><a href='#mut.to.sigs.input'><p>Converts mutation list to correct input format</p></a></li>
<li><a href='#norm.it'><p>Normalizes trinucleotide contexts</p></a></li>
<li><a href='#plotSignatures'><p>Plots the result from whichSignatures()</p></a></li>
<li><a href='#plotTumor'><p>Plots a tumor profile</p></a></li>
<li><a href='#randomly.generated.tumors'><p>Values for 100 randomly generated tumor samples</p></a></li>
<li><a href='#sample.mut.ref'><p>Example input to mut.to.sigs.input()</p></a></li>
<li><a href='#signatures.cosmic'><p>Published Signatures from Sanger COSMIC</p></a></li>
<li><a href='#signatures.nature2013'><p>Published Signatures from Alexandrov et al 2013</p></a></li>
<li><a href='#tri.counts.exome'><p>The counts of every trinuclotide frequency in an exome</p></a></li>
<li><a href='#tri.counts.genome'><p>The counts of every trinuclotide frequency in a genome</p></a></li>
<li><a href='#updateW_GR'><p>Updates the weights matrix</p></a></li>
<li><a href='#vcf.to.sigs.input'><p>Converts a VCF file to correct input format</p></a></li>
<li><a href='#whichSignatures'><p>Which signatures are present</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifies Signatures Present in a Tumor Sample</td>
</tr>
<tr>
<td>Version:</td>
<td>1.8.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-07-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Rachel Rosenthal</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rachel Rosenthal &lt;rachel.rosenthal.14@ucl.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Takes sample information in the form of the fraction of mutations
    in each of 96 trinucleotide contexts and identifies the weighted combination
    of published signatures that, when summed, most closely reconstructs the
    mutational profile.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, BSgenome, BSgenome.Hsapiens.UCSC.hg19, GenomeInfoDb,
grDevices, graphics, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>VariantAnnotation</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/raerose01/deconstructSigs">https://github.com/raerose01/deconstructSigs</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/raerose01/deconstructSigs/issues">https://github.com/raerose01/deconstructSigs/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-07-29 12:30:13 UTC; rachel</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-07-29 18:27:06</td>
</tr>
</table>
<hr>
<h2 id='changeColumnNames'>Re-formats column names</h2><span id='topic+changeColumnNames'></span>

<h3>Description</h3>

<p>Changes the way the column names are formatted from nxn.y to the n[x&gt;y]n format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>changeColumnNames(col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="changeColumnNames_+3A_col">col</code></td>
<td>
<p>Vector of column names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a newly formatted vector of column names
</p>

<hr>
<h2 id='deconstructSigs'>deconstructSigs</h2><span id='topic+deconstructSigs'></span><span id='topic+deconstructSigs-package'></span>

<h3>Description</h3>

<p>Takes sample information in the form of the fraction of mutations
in each of 96 trinucleotide contexts and identifies the weighted combination 
of published signatures that, when summed, most closely reconstructs the 
mutational profile.
</p>


<h3>Main functions</h3>

 
<ul>
<li> <p><code><a href="#topic+whichSignatures">whichSignatures</a></code> 
</p>
</li>
<li> <p><code><a href="#topic+mut.to.sigs.input">mut.to.sigs.input</a></code> 
</p>
</li>
<li> <p><code><a href="#topic+plotSignatures">plotSignatures</a></code> 
</p>
</li>
<li> <p><code><a href="#topic+makePie">makePie</a></code>
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Rachel Rosenthal rachel.rosenthal.14@ucl.ac.uk
</p>

<hr>
<h2 id='example.output'>Example output of whichSignatures()</h2><span id='topic+example.output'></span>

<h3>Description</h3>

<p>A list that was generated from running whichSignatures().  It contains the
following:
</p>


<h3>Format</h3>

<p>A list of four elements</p>


<h3>Details</h3>

 <ul>
<li><p> weights - weight of each signature found in the sample </p>
</li>
<li>
<p>tumor - tricontext fractions that were used as input </p>
</li>
<li><p> product - product
of the weights by the reference signatures used </p>
</li>
<li><p> diff - difference
between the tumor tricontexts fractions used as input and those generated as
the product </p>
</li>
<li><p> unknown - the fraction of the tumor profile that could not
be assigned to a signature </p>
</li></ul>


<hr>
<h2 id='findSeed'>Seeds weight matrix</h2><span id='topic+findSeed'></span>

<h3>Description</h3>

<p>Determines which single signature results in the lowest sum-squared error and
uses that as a seed to start from
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findSeed(tumor, signatures)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findSeed_+3A_tumor">tumor</code></td>
<td>
<p>Actual trinucleotide context fractions</p>
</td></tr>
<tr><td><code id="findSeed_+3A_signatures">signatures</code></td>
<td>
<p>Signatures matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the index corresponding to the signature that best describes
the input data
</p>

<hr>
<h2 id='formatContexts'>Formats the data for use in plotSignatures()</h2><span id='topic+formatContexts'></span>

<h3>Description</h3>

<p>Makes the contexts useable in plotSignatures()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatContexts(contexts)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatContexts_+3A_contexts">contexts</code></td>
<td>
<p>One of the entries in the output list from whichSignatures()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with sample.id, full_context, fraction, and 
mutation as column names
</p>

<hr>
<h2 id='getError'>Calculates error</h2><span id='topic+getError'></span>

<h3>Description</h3>

<p>Finds error between calculated trinucleotide context fractions and inputted
ones
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getError(tumor, signatures, w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getError_+3A_tumor">tumor</code></td>
<td>
<p>Actual trinucleotide context fractions</p>
</td></tr>
<tr><td><code id="getError_+3A_signatures">signatures</code></td>
<td>
<p>Signatures matrix</p>
</td></tr>
<tr><td><code id="getError_+3A_w">w</code></td>
<td>
<p>Weights matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the sum squared error between calculated and actual
trinucleotide context fractions
</p>

<hr>
<h2 id='getTriContextFraction'>Calculates trinucleotide context fraction</h2><span id='topic+getTriContextFraction'></span>

<h3>Description</h3>

<p>Determines trinucleotide context fraction from a mutation counts list. Given
a data frame or .txt file or mutation counts with columns as trincucleotide
contexts in the form A[C&gt;A]A and the rows as sample id's and a data frame or
.txt file of the times each tri nucleotide context is seen in the sequencing
region with the rows in the format ACA, the function returns a data frame of
the mutation counts normalized by trinucleotide frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTriContextFraction(mut.counts.ref, trimer.counts.method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTriContextFraction_+3A_mut.counts.ref">mut.counts.ref</code></td>
<td>
<p>data frame of counts of mutations in each trinucleotide
context for each sample or location where .txt file is found</p>
</td></tr>
<tr><td><code id="getTriContextFraction_+3A_trimer.counts.ref">trimer.counts.ref</code></td>
<td>
<p>data frame of counts of times each trinculeotide
context is seen in sequencing area or location where the .txt file is found</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the trinucleotide context fraction
</p>

<hr>
<h2 id='golden.section.search'>Golden section search</h2><span id='topic+golden.section.search'></span>

<h3>Description</h3>

<p>Uses golden section method to search given space for value that minimizes
function given
</p>


<h3>Usage</h3>

<pre><code class='language-R'>golden.section.search(f, lower, upper, tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="golden.section.search_+3A_f">f</code></td>
<td>
<p>function</p>
</td></tr>
<tr><td><code id="golden.section.search_+3A_lower">lower</code></td>
<td>
<p>Lower bound to search in</p>
</td></tr>
<tr><td><code id="golden.section.search_+3A_upper">upper</code></td>
<td>
<p>Upper bound to search in</p>
</td></tr>
<tr><td><code id="golden.section.search_+3A_tol">tol</code></td>
<td>
<p>How exact the answer must be</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a value that minimizes the function input
</p>

<hr>
<h2 id='makePie'>Plots the weights from whichSignatures()</h2><span id='topic+makePie'></span>

<h3>Description</h3>

<p>Uses the output from whichSignatures() and creates a pie chart of the weights
outputted
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makePie(sigs.output, sub = "", add.color = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makePie_+3A_sigs.output">sigs.output</code></td>
<td>
<p>The list output from whichSignatures()</p>
</td></tr>
<tr><td><code id="makePie_+3A_sub">sub</code></td>
<td>
<p>A character vector that specifies cancer subtype for plot title, 
if wanted</p>
</td></tr>
<tr><td><code id="makePie_+3A_add.color">add.color</code></td>
<td>
<p>Optional character vector to assign additional colors for
novel signatures</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a pie chart of the weights calculated in the given tumor sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>makePie(example.output)
</code></pre>

<hr>
<h2 id='mut.to.sigs.input'>Converts mutation list to correct input format</h2><span id='topic+mut.to.sigs.input'></span>

<h3>Description</h3>

<p>Given a mutation list, outputs a data frame with counts of how frequently a 
mutation is found within each trinucleotide context per sample ID.  Output 
can be used as input into getTriContextFraction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut.to.sigs.input(mut.ref, sample.id = "Sample", chr = "chr", pos = "pos",
  ref = "ref", alt = "alt", bsg = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mut.to.sigs.input_+3A_mut.ref">mut.ref</code></td>
<td>
<p>Location of the mutation file that is to be converted or name 
of data frame in environment</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_sample.id">sample.id</code></td>
<td>
<p>Column name in the mutation file corresponding to the Sample
ID</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_chr">chr</code></td>
<td>
<p>Column name in the mutation file corresponding to the chromosome</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_pos">pos</code></td>
<td>
<p>Column name in the mutation file corresponding to the mutation 
position</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_ref">ref</code></td>
<td>
<p>Column name in the mutation file corresponding to the reference 
base</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_alt">alt</code></td>
<td>
<p>Column name in the mutation file corresponding to the alternate 
base</p>
</td></tr>
<tr><td><code id="mut.to.sigs.input_+3A_bsg">bsg</code></td>
<td>
<p>Only set if another genome build is required. Must be a BSgenome
object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The context sequence is taken from the BSgenome.Hsapiens.UCSC.hgX::Hsapiens 
object. Therefore the coordinates must correspond to the human hgX assembly. 
Default is set to the UCSC hg19 assembly, which corresponds to the GRCh37 
assembly. If another assembly is required, it must already be present in the 
R workspace and fed as a parameter. This method will translate chromosome 
names from other versions of the assembly like NCBI or Ensembl. For instance,
the following transformation will be done: &quot;1&quot; -&gt; &quot;chr1&quot;; &quot;MT&quot; -&gt; &quot;chrM&quot;; 
&quot;GL000245.1&quot; -&gt; &quot;chrUn_gl000245&quot;; etc.
</p>


<h3>Value</h3>

<p>A data frame that contains sample IDs for the rows and trinucleotide 
contexts for the columns. Each entry is the count of how many times a 
mutation with that trinucleotide context is seen in the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sigs.input = mut.to.sigs.input(mut.ref = sample.mut.ref, sample.id = "Sample",
chr = "chr", pos = "pos", ref = "ref", alt = "alt", bsg = 
BSgenome.Hsapiens.UCSC.hg19)

## End(Not run)
</code></pre>

<hr>
<h2 id='norm.it'>Normalizes trinucleotide contexts</h2><span id='topic+norm.it'></span>

<h3>Description</h3>

<p>Normalizes the trinucleotide contexts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>norm.it(col, trimer.ratio)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="norm.it_+3A_col">col</code></td>
<td>
<p>column names</p>
</td></tr>
<tr><td><code id="norm.it_+3A_trimer.counts">trimer.counts</code></td>
<td>
<p>count of the number of times each trimer is found in the area sequenced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a normalized column based on the trimer counts
</p>

<hr>
<h2 id='plotSignatures'>Plots the result from whichSignatures()</h2><span id='topic+plotSignatures'></span>

<h3>Description</h3>

<p>Uses the output from whichSignatures() and creates a plot of the given tumor 
mutational spectrum and the calculated one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSignatures(sigs.output, sub = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSignatures_+3A_sigs.output">sigs.output</code></td>
<td>
<p>The list output from whichSignatures()</p>
</td></tr>
<tr><td><code id="plotSignatures_+3A_sub">sub</code></td>
<td>
<p>A character vector that specifies cancer subtype for plot title,
if wanted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots the trinucleotide frequency for the given tumor on the top 
panel, the calculated one on the middle panel, and the difference between 
the two on the bottom panel.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plotSignatures(example.output, sub = "example")
</code></pre>

<hr>
<h2 id='plotTumor'>Plots a tumor profile</h2><span id='topic+plotTumor'></span>

<h3>Description</h3>

<p>Uses the output from whichSignatures() and creates a plot of the given tumor 
mutational spectrum and the calculated one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTumor(tumor, sub = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTumor_+3A_tumor">tumor</code></td>
<td>
<p>A tumor matrix</p>
</td></tr>
<tr><td><code id="plotTumor_+3A_sub">sub</code></td>
<td>
<p>A character vector that specifies cancer subtype for plot title,
if wanted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots the trinucleotide frequency for the given tumor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plotTumor(example.output[['tumor']], sub = "example")
</code></pre>

<hr>
<h2 id='randomly.generated.tumors'>Values for 100 randomly generated tumor samples</h2><span id='topic+randomly.generated.tumors'></span>

<h3>Description</h3>

<p>A dataset containing the value for the 96-trinucleotide contexts of 100 
tumors generated by a random linear combination of the Nature 2013
signatures.
</p>


<h3>Format</h3>

<p>A data frame of 100 rows and 96 columns</p>

<hr>
<h2 id='sample.mut.ref'>Example input to mut.to.sigs.input()</h2><span id='topic+sample.mut.ref'></span>

<h3>Description</h3>

<p>A data frame containing example mutations that can be used as input to 
mut.to.sigs.input(). Data comes from two TCGA LUAD patients 
(http://cancergenome.nih.gov/) Contains the following columns:
</p>


<h3>Format</h3>

<p>A data frame of 657 rows and 5 column that contains example mutations 
which could be used in mut.to.sigs.input()</p>


<h3>Details</h3>

 <ul>
<li><p> Sample - sample name </p>
</li>
<li><p> chr - chromosome number </p>
</li>
<li><p> pos -
chromosome position </p>
</li>
<li><p> ref - reference base </p>
</li>
<li><p> alt - alternate base </p>
</li></ul>



<h3>Source</h3>

<p><a href="http://cancergenome.nih.gov/">http://cancergenome.nih.gov/</a>
</p>

<hr>
<h2 id='signatures.cosmic'>Published Signatures from Sanger COSMIC</h2><span id='topic+signatures.cosmic'></span>

<h3>Description</h3>

<p>A dataset containing the additional signatures identified read into R as a 
data frame. Can be used the 'signatures.ref' parameter in whichSignatures().
</p>


<h3>Format</h3>

<p>A data frame of 30 rows and 96 columns</p>


<h3>Source</h3>

<p><a href="http://cancer.sanger.ac.uk/cosmic/signatures">http://cancer.sanger.ac.uk/cosmic/signatures</a>
</p>

<hr>
<h2 id='signatures.nature2013'>Published Signatures from Alexandrov et al 2013</h2><span id='topic+signatures.nature2013'></span>

<h3>Description</h3>

<p>A dataset containing the published signatures from Alexandrov et al read into
R as a data frame. Can be used the 'signatures.ref' parameter in
whichSignatures(). Data obtained: Nature 2013 PMID:23945592
</p>


<h3>Format</h3>

<p>A data frame of 27 rows and 96 columns</p>


<h3>Source</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/23945592">http://www.ncbi.nlm.nih.gov/pubmed/23945592</a>
</p>

<hr>
<h2 id='tri.counts.exome'>The counts of every trinuclotide frequency in an exome</h2><span id='topic+tri.counts.exome'></span>

<h3>Description</h3>

<p>A datset containing the number of times each trinucleotide (ex: ACA) is found
in the exome region captured by sequencing. Can be used as the
'trimer.counts.loc' parameter in whichSignatures().
</p>


<h3>Format</h3>

<p>A data frame of 32 rows and 1 column that contains the counts</p>

<hr>
<h2 id='tri.counts.genome'>The counts of every trinuclotide frequency in a genome</h2><span id='topic+tri.counts.genome'></span>

<h3>Description</h3>

<p>A datset containing the number of times each trinucleotide (ex: ACA) is found
in the hg19 genome. Can be used as the
'trimer.counts.loc' parameter in whichSignatures().
</p>


<h3>Format</h3>

<p>A data frame of 32 rows and 1 column that contains the counts</p>

<hr>
<h2 id='updateW_GR'>Updates the weights matrix</h2><span id='topic+updateW_GR'></span>

<h3>Description</h3>

<p>Determines what proportion of what signature, when added to the current
weights matrix, results in the lowest sum-squared error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>updateW_GR(tumor, signatures, w, signatures.limit, bound = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="updateW_GR_+3A_tumor">tumor</code></td>
<td>
<p>Actual trinucleotide context fractions</p>
</td></tr>
<tr><td><code id="updateW_GR_+3A_signatures">signatures</code></td>
<td>
<p>Signatures matrix</p>
</td></tr>
<tr><td><code id="updateW_GR_+3A_w">w</code></td>
<td>
<p>Weights matrix</p>
</td></tr>
<tr><td><code id="updateW_GR_+3A_signatures.limit">signatures.limit</code></td>
<td>
<p>Number of signatures to limit the search to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an updated weights matrix
</p>

<hr>
<h2 id='vcf.to.sigs.input'>Converts a VCF file to correct input format</h2><span id='topic+vcf.to.sigs.input'></span>

<h3>Description</h3>

<p>Given a VCF file, outputs a data frame with counts of how frequently a
mutation is found within each trinucleotide context per sample ID.  Output
can be used as input into getTriContextFraction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf.to.sigs.input(vcf, bsg = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf.to.sigs.input_+3A_vcf">vcf</code></td>
<td>
<p>Location of the VCF file that is to be converted</p>
</td></tr>
<tr><td><code id="vcf.to.sigs.input_+3A_bsg">bsg</code></td>
<td>
<p>Only set if another genome build is required. Must be a BSgenome
object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The context sequence is taken from the BSgenome.Hsapiens.UCSC.hg19::Hsapiens
object, therefore the coordinates must correspond to the human hg19 assembly,
the UCSC version of the GRCh37 Homo sapiens assembly. This method will to
its best to translate chromosome names from other versions of the assembly
like NCBI or Ensembl. For instance, the following transformation will be
done: &quot;1&quot; -&gt; &quot;chr1&quot;; &quot;MT&quot; -&gt; &quot;chrM&quot;; &quot;GL000245.1&quot; -&gt; &quot;chrUn_gl000245&quot;; etc.
</p>
<p>This method relies on the VariantAnnotation package to read the VCF file.
</p>


<h3>Value</h3>

<p>A data frame that contains sample IDs for the rows and trinucleotide
contexts for the columns. Each entry is the count of how many times a
mutation with that trinucleotide context is seen in the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sigs.input = vcf.to.sigs.input(vcf = "variants.vcf")

## End(Not run)
</code></pre>

<hr>
<h2 id='whichSignatures'>Which signatures are present</h2><span id='topic+whichSignatures'></span>

<h3>Description</h3>

<p>Determines how much of each signature is present in the sample given
</p>


<h3>Usage</h3>

<pre><code class='language-R'>whichSignatures(tumor.ref = NA, sample.id,
  signatures.ref = signatures.nature2013, associated = c(),
  signatures.limit = NA, signature.cutoff = 0.06, contexts.needed = FALSE,
  tri.counts.method = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="whichSignatures_+3A_tumor.ref">tumor.ref</code></td>
<td>
<p>Either a data frame or location of input text file, where 
rows are samples, columns are trinucleotide contexts</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_sample.id">sample.id</code></td>
<td>
<p>Name of sample &ndash; should be rowname of tumor.ref. Optional
if the tumor.ref contains one single sample</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_signatures.ref">signatures.ref</code></td>
<td>
<p>Either a data frame or location of signature text file,
where rows are signatures, columns are trinucleotide contexts</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_associated">associated</code></td>
<td>
<p>Vector of associated signatures. If given, will narrow the 
signatures tested to only the ones listed.</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_signatures.limit">signatures.limit</code></td>
<td>
<p>Number of signatures to limit the search to</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_signature.cutoff">signature.cutoff</code></td>
<td>
<p>Discard any signature contributions with a weight 
less than this amount</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_contexts.needed">contexts.needed</code></td>
<td>
<p>FALSE if tumor.file is a context file, TRUE if it is 
only mutation counts</p>
</td></tr>
<tr><td><code id="whichSignatures_+3A_tri.counts.method">tri.counts.method</code></td>
<td>
<p>Set to either:
</p>

<ul>
<li><p> 'default' &ndash; no further normalization </p>
</li>
<li><p> 'exome' &ndash; normalized by
number of times each trinucleotide context is observed in the exome </p>
</li>
<li>
<p>'genome' &ndash; normalized by number of times each trinucleotide context is
observed in the genome </p>
</li>
<li><p> 'exome2genome' &ndash; multiplied by a ratio of that
trinucleotide's occurence in the genome to the trinucleotide's occurence in
the exome </p>
</li>
<li><p> 'genome2exome' &ndash; multiplied by a ratio of that
trinucleotide's occurence in the exome to the trinucleotide's occurence in
the genome </p>
</li>
<li><p> data frame containing user defined scaling factor &ndash; count
data for each trinucleotide context is multiplied by the corresponding value
given in the data frame </p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the weights for each signatures, the product when those are
multiplied on the signatures, the difference between the tumor sample and 
product, the tumor sample tricontext distribution given, and the unknown 
weight.
</p>


<h3>Normalization</h3>

<p>If the input data frame only contains the counts of
the mutations observed in each context, then the data frame must be
normalized. In these cases, the value of 'contexts.needed' should be TRUE.
<br /> The parameter, 'tri.counts.method', determines any additional
normalization performed. Any user provided data frames should match the
format of 'tri.counts.exome' and 'tri.counts.genome'. <br /> The method of
normalization chosen should match how the input signatures were normalized.
For exome data, the 'exome2genome' method is appropriate for the signatures
included in this package. For whole genome data, use the 'default' method
to obtain consistent results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test = whichSignatures(tumor.ref = randomly.generated.tumors,
                       sample.id = "2", 
                       contexts.needed = FALSE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
