<!DOCTYPE html><html><head><title>Help for package decode</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {decode}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getAssoGeneSetPValue'><p>Calculate the p-value between selected genes and functional gene set</p></a></li>
<li><a href='#getBestAssociatedGeneSet'><p>Get best associated functional gene sets for partitions of gene i</p></a></li>
<li><a href='#getBonferroniPValue'><p>Adjust p-value by Bonferroni correction</p></a></li>
<li><a href='#getDE_DC_OptimalThreshold'><p>Perform chi-square optimization</p></a></li>
<li><a href='#getFDR'><p>Adjust p-value by Benjamini and Hochberg method</p></a></li>
<li><a href='#getPartitionIndex'><p>Get gene index of 8 partitions for gene i</p></a></li>
<li><a href='#getPathway'><p>read functional gene sets</p></a></li>
<li><a href='#openFileToWrite'><p>Open file to write result</p></a></li>
<li><a href='#runDecode'><p>Differential Co-Expression and Differential Expression Analysis</p></a></li>
<li><a href='#sumResult_MinGain'><p>Summarize the functional gene set results into text file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Differential Co-Expression and Differential Expression Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Integrated differential expression (DE) and differential co-expression (DC) analysis on gene expression data based on DECODE (DifferEntial CO-expression and Differential Expression) algorithm.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-07-13 12:23:17 UTC; BB</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Lui [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Lui &lt;tlui27@yahoo.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-07-14 20:40:34</td>
</tr>
</table>
<hr>
<h2 id='getAssoGeneSetPValue'>Calculate the p-value between selected genes and functional gene set</h2><span id='topic+getAssoGeneSetPValue'></span>

<h3>Description</h3>

<p>Calculate the p-value between selected genes and functional gene set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAssoGeneSetPValue(geneList, geneSet, multipleTestCount, MaxGene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAssoGeneSetPValue_+3A_genelist">geneList</code></td>
<td>
<p>Selected genes</p>
</td></tr>
<tr><td><code id="getAssoGeneSetPValue_+3A_geneset">geneSet</code></td>
<td>
<p>Functional gene set</p>
</td></tr>
<tr><td><code id="getAssoGeneSetPValue_+3A_multipletestcount">multipleTestCount</code></td>
<td>
<p>Number of multiple testing</p>
</td></tr>
<tr><td><code id="getAssoGeneSetPValue_+3A_maxgene">MaxGene</code></td>
<td>
<p>Number of genes in expression data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted p-value for the associated gene set
</p>

<hr>
<h2 id='getBestAssociatedGeneSet'>Get best associated functional gene sets for partitions of gene i</h2><span id='topic+getBestAssociatedGeneSet'></span>

<h3>Description</h3>

<p>Get best associated functional gene sets for partitions of gene i
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBestAssociatedGeneSet(pathway, all8Partitions, onePartition, MaxGene,
  minSupport)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBestAssociatedGeneSet_+3A_pathway">pathway</code></td>
<td>
<p>All functional gene sets</p>
</td></tr>
<tr><td><code id="getBestAssociatedGeneSet_+3A_all8partitions">all8Partitions</code></td>
<td>
<p>All eight possible partitions for gene i</p>
</td></tr>
<tr><td><code id="getBestAssociatedGeneSet_+3A_onepartition">onePartition</code></td>
<td>
<p>The partition to be associated with the functional gene set</p>
</td></tr>
<tr><td><code id="getBestAssociatedGeneSet_+3A_maxgene">MaxGene</code></td>
<td>
<p>Number of genes in expression data</p>
</td></tr>
<tr><td><code id="getBestAssociatedGeneSet_+3A_minsupport">minSupport</code></td>
<td>
<p>Minimum support for functional gene set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The adjusted p-values for the best associated gene set of the input partition
</p>

<hr>
<h2 id='getBonferroniPValue'>Adjust p-value by Bonferroni correction</h2><span id='topic+getBonferroniPValue'></span>

<h3>Description</h3>

<p>Adjust p-value by Bonferroni correction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBonferroniPValue(pValues)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBonferroniPValue_+3A_pvalues">pValues</code></td>
<td>
<p>Unadjusted p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Adjusted p-values
</p>

<hr>
<h2 id='getDE_DC_OptimalThreshold'>Perform chi-square optimization</h2><span id='topic+getDE_DC_OptimalThreshold'></span>

<h3>Description</h3>

<p>Perform chi-square optimization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDE_DC_OptimalThreshold(t_result, MaxGene, d_r, minSupport)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDE_DC_OptimalThreshold_+3A_t_result">t_result</code></td>
<td>
<p>The t-statistics</p>
</td></tr>
<tr><td><code id="getDE_DC_OptimalThreshold_+3A_maxgene">MaxGene</code></td>
<td>
<p>Number of genes in expression data</p>
</td></tr>
<tr><td><code id="getDE_DC_OptimalThreshold_+3A_d_r">d_r</code></td>
<td>
<p>DC measures</p>
</td></tr>
<tr><td><code id="getDE_DC_OptimalThreshold_+3A_minsupport">minSupport</code></td>
<td>
<p>The minimum expected frequency in contingency table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The optimal threshold information
</p>

<hr>
<h2 id='getFDR'>Adjust p-value by Benjamini and Hochberg method</h2><span id='topic+getFDR'></span>

<h3>Description</h3>

<p>Adjust p-value by Benjamini and Hochberg method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFDR(pValues)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFDR_+3A_pvalues">pValues</code></td>
<td>
<p>Unadjusted p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Adjusted p-values
</p>

<hr>
<h2 id='getPartitionIndex'>Get gene index of 8 partitions for gene i</h2><span id='topic+getPartitionIndex'></span>

<h3>Description</h3>

<p>Get gene index of 8 partitions for gene i
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPartitionIndex(gene_i, t_result, optimalCutOff, abs_r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPartitionIndex_+3A_gene_i">gene_i</code></td>
<td>
<p>Gene i index</p>
</td></tr>
<tr><td><code id="getPartitionIndex_+3A_t_result">t_result</code></td>
<td>
<p>t-statistics</p>
</td></tr>
<tr><td><code id="getPartitionIndex_+3A_optimalcutoff">optimalCutOff</code></td>
<td>
<p>Optimal thresholds</p>
</td></tr>
<tr><td><code id="getPartitionIndex_+3A_abs_r">abs_r</code></td>
<td>
<p>Matrix consisting of absolute values of all differential co-expression measures</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The selected genes for each partition in index
</p>

<hr>
<h2 id='getPathway'>read functional gene sets</h2><span id='topic+getPathway'></span>

<h3>Description</h3>

<p>read functional gene sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPathway(inputFile, geneName, minSupport)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPathway_+3A_inputfile">inputFile</code></td>
<td>
<p>Input file name</p>
</td></tr>
<tr><td><code id="getPathway_+3A_genename">geneName</code></td>
<td>
<p>Gene name lists</p>
</td></tr>
<tr><td><code id="getPathway_+3A_minsupport">minSupport</code></td>
<td>
<p>Minimum support</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Functional gene set
</p>

<hr>
<h2 id='openFileToWrite'>Open file to write result</h2><span id='topic+openFileToWrite'></span>

<h3>Description</h3>

<p>Open file to write result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>openFileToWrite(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="openFileToWrite_+3A_filename">filename</code></td>
<td>
<p>file name
Output: Results in text file</p>
</td></tr>
</table>

<hr>
<h2 id='runDecode'>Differential Co-Expression and Differential Expression Analysis</h2><span id='topic+runDecode'></span>

<h3>Description</h3>

<p>Given a set of gene expression data and functional gene set data, the program will return a table summary for the selected gene sets with high differential co-expression and high differential expression (HDC-HDE). User need to specify the input paths for the gene expression data and functional gene set data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runDecode(geneSetInputFile, geneExpressionFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runDecode_+3A_genesetinputfile">geneSetInputFile</code></td>
<td>
<p>Path for functional gene set data</p>
</td></tr>
<tr><td><code id="runDecode_+3A_geneexpressionfile">geneExpressionFile</code></td>
<td>
<p>Path for gene expression data
</p>
<p>Input:
(1) gene expression data
</p>
<p>(2) functional gene set data
</p>
<p>Output: Table summary for the selected HDC-HDE gene sets, 'out_summary.txt'
</p>
<p>Data format for gene expression data (Columns are tab-separated):
</p>
<p>Column 1: Official gene symbol
</p>
<p>Column 2: Probe ID
</p>
<p>Starting from column 3: Expression for different samples
</p>
<p>Row 1 (starting from column 3): Sample class (&quot;1&quot; indicates control group; &quot;2&quot; indicates case group)
</p>
<p>Row 2: Sample id
</p>
<p>Starting from row 3: Expression for different genes
</p>
<p>Example:
</p>
<p>geneName   probeID         2       2        2        1         1         1
</p>
<p>-            -          Case1    Case2    Case3   Control1  Control2  Control3
</p>
<p>7A5      ILMN_1762337  5.12621  5.19419  5.06645  5.40649   5.51259   5.387
</p>
<p>A1BG     ILMN_2055271  5.63504  5.68533  5.66251  5.37466   5.43955   5.50973
</p>
<p>A1CF     ILMN_2383229  5.41543  5.58543  5.43239  5.49634   5.62685   5.36962
</p>
<p>A26C3    ILMN_1653355  5.56713  5.5547   5.59547  5.46895   5.49622   5.50094
</p>
<p>A2BP1    ILMN_1814316  5.23016  5.33808  5.31413  5.30586   5.40108   5.31855
</p>
<p>A2M      ILMN_1745607  7.65332  6.56431  8.20163  9.19837   9.04295   10.1448
</p>
<p>A2ML1    ILMN_2136495  5.53532  5.93801  5.33728  5.36676   5.79942   5.13974
</p>
<p>A3GALT2  ILMN_1668111  5.18578  5.35207  5.30554  5.26107   5.26536   5.28932
</p>
<p>Data format for functional gene set data (Columns are tab-separated):
</p>
<p>Column 1: Functional gene set name
</p>
<p>Column 2: Other description such as gene set id
</p>
<p>Starting from column 3: Official gene symbols for the functional gene set
</p>
<p>Example:
</p>
<p>B cell activation   GO\GO:0042113   AKAP17A   ZAP70   PFDN1 ...
</p>
<p>apoptotic signaling pathway   GO\GO:0097190   ITPR1   PTH   DNAJC10   HINT1 ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The main program for DECODE algorithm
</p>
<p>To run an example using expression data with 1400 genes.
</p>
<p>runDecode(&quot;\extdata\geneSet.txt&quot;,&quot;\extdata\Expression_data_1400genes.txt&quot;)
</p>
<p>or
</p>
<p>runDecode(&quot;/extdata/geneSet.txt&quot;,&quot;/extdata/Expression_data_1400genes.txt&quot;)
</p>
<p>The sample data with 1400 genes takes 16 minutes to complete. (Computer used: An Intel Core i7-4600 processor, 2.69 GHz, 8 GB RAM)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
path = system.file('extdata', package='decode')
geneSetInputFile = file.path(path, "geneSet.txt")
geneExpressionFile = file.path(path, "Expression_data_50genes.txt")
runDecode(geneSetInputFile, geneExpressionFile)

## End(Not run)
</code></pre>

<hr>
<h2 id='sumResult_MinGain'>Summarize the functional gene set results into text file</h2><span id='topic+sumResult_MinGain'></span>

<h3>Description</h3>

<p>Summarize the functional gene set results into text file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sumResult_MinGain()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
