<!DOCTYPE html><html><head><title>Help for package deepredeff</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {deepredeff}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#aas_to_df'><p>Convert AAString class to dataframe</p></a></li>
<li><a href='#aasset_to_df'><p>Convert AAStringset class to dataframe</p></a></li>
<li><a href='#deepredeff'><p><code>deepredeff</code> package</p></a></li>
<li><a href='#encode_integer'><p>Integer encoder</p></a></li>
<li><a href='#encode_one_hot'><p>One-hot encoder</p></a></li>
<li><a href='#ensemble_weighted'><p>Weighted ensemble</p></a></li>
<li><a href='#fasta_to_df'><p>Convert fasta format to dataframe</p></a></li>
<li><a href='#get_ensemble_method'><p>Get ensemble methods</p></a></li>
<li><a href='#install_tensorflow'><p>Install the TensorFlow backend</p></a></li>
<li><a href='#load_model'><p>Load model</p></a></li>
<li><a href='#package_glob'><p>Wildcard Expansion on File Paths</p></a></li>
<li><a href='#plot.tbl_deepredeff'><p>Plot the results of prediction</p></a></li>
<li><a href='#predict_effector'><p>Predict effector</p></a></li>
<li><a href='#prediction_mapper'><p>Prediction mapper helper</p></a></li>
<li><a href='#summary.tbl_deepredeff'><p>Get the summary of the prediction results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Deep Learning Prediction of Effectors</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A tool that contains trained deep learning models
    for predicting effector proteins. 'deepredeff' has been trained to
    identify effector proteins using a set of known experimentally
    validated effectors from either bacteria, fungi, or oomycetes.
    Documentation is available via several vignettes, and the paper by
    Kristianingsih and MacLean (2020) &lt;<a href="https://doi.org/10.1101%2F2020.07.08.193250">doi:10.1101/2020.07.08.193250</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ruthkr/deepredeff/">https://github.com/ruthkr/deepredeff/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ruthkr/deepredeff/issues/">https://github.com/ruthkr/deepredeff/issues/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, dplyr, ggplot2, ggthemes, keras, magrittr, purrr,
reticulate, rlang, seqinr, tensorflow</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, kableExtra, knitr, rmarkdown, stringr, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/reticulate:</td>
<td>list( packages = list( list(package = "tensorflow",
version = "2.0.0", pip = FALSE) ) )</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-16 09:12:16 UTC; kristiar</td>
</tr>
<tr>
<td>Author:</td>
<td>Ruth Kristianingsih
    <a href="https://orcid.org/0000-0003-1873-6203"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ruth Kristianingsih &lt;ruth.kristianingsih30@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-16 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='aas_to_df'>Convert AAString class to dataframe</h2><span id='topic+aas_to_df'></span>

<h3>Description</h3>

<p>Convert AAString class to dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aas_to_df(aas)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aas_to_df_+3A_aas">aas</code></td>
<td>
<p>AAString class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns data in data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input_fasta &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")
input_aas &lt;- Biostrings::readAAStringSet(input_fasta)
input_df &lt;- aasset_to_df(input_aas)
</code></pre>

<hr>
<h2 id='aasset_to_df'>Convert AAStringset class to dataframe</h2><span id='topic+aasset_to_df'></span>

<h3>Description</h3>

<p>Convert AAStringset class to dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aasset_to_df(aas)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aasset_to_df_+3A_aas">aas</code></td>
<td>
<p>AAStringset class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns data in data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- "MSHMTFNTWKAGLWRLAAAAVLSLLPVVARAAVPGITGPTFDLTAQPGRANQPDGASVYSWGYGCNPRTVPGFLPSVNPLAGQ"
input_aas &lt;- Biostrings::AAString(input)
input_df &lt;- aas_to_df(input_aas)
</code></pre>

<hr>
<h2 id='deepredeff'><code>deepredeff</code> package</h2><span id='topic+deepredeff'></span>

<h3>Description</h3>

<p>Effector protein predictor using Deep Learning models.
</p>


<h3>Details</h3>

<p>See the README on
<a href="https://github.com/ruthkr/deepredeff/">GitHub</a>
</p>

<hr>
<h2 id='encode_integer'>Integer encoder</h2><span id='topic+encode_integer'></span>

<h3>Description</h3>

<p><code>encode_integer</code> integer-encodes sequence in a string format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>encode_integer(sequence, max_length = 4034)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="encode_integer_+3A_sequence">sequence</code></td>
<td>
<p>Sequence in a string format.</p>
</td></tr>
<tr><td><code id="encode_integer_+3A_max_length">max_length</code></td>
<td>
<p>Maximum length of sequence to encode.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Integer encoded sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_seq &lt;- "MSHMTFNTWKAGLWRLAAAAVLSLLPVVARAAVPGITGPTFDLTAQPGRANQPDGASVYSWGYGCNPRTVPGFLPSVNPLAGQ"
encoded_seq &lt;- encode_integer(sample_seq)
</code></pre>

<hr>
<h2 id='encode_one_hot'>One-hot encoder</h2><span id='topic+encode_one_hot'></span>

<h3>Description</h3>

<p><code>encode_one_hot</code> one-hot-encodes sequence in a string format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>encode_one_hot(sequence, max_length = 4034)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="encode_one_hot_+3A_sequence">sequence</code></td>
<td>
<p>Sequence in a string format.</p>
</td></tr>
<tr><td><code id="encode_one_hot_+3A_max_length">max_length</code></td>
<td>
<p>Maximum length of sequence to encode.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One-hot encoded sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample_seq &lt;- "MSHMTFNTWKAGLWRLAAAAVLSLLPVVARAAVPGITGPTFDLTAQPGRANQPDGASVYSWGYGCNPRTVPGFLPSVNPLAGQ"
encoded_seq &lt;- encode_one_hot(sample_seq)
</code></pre>

<hr>
<h2 id='ensemble_weighted'>Weighted ensemble</h2><span id='topic+ensemble_weighted'></span>

<h3>Description</h3>

<p>Weighted ensemble
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemble_weighted(pred_list, weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ensemble_weighted_+3A_pred_list">pred_list</code></td>
<td>
<p>List of deep learning models.</p>
</td></tr>
<tr><td><code id="ensemble_weighted_+3A_weights">weights</code></td>
<td>
<p>Accuracy values from evaluation on the validation dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the prediction results from weighted ensemble.
</p>

<hr>
<h2 id='fasta_to_df'>Convert fasta format to dataframe</h2><span id='topic+fasta_to_df'></span>

<h3>Description</h3>

<p>Convert fasta format to dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasta_to_df(fasta_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasta_to_df_+3A_fasta_path">fasta_path</code></td>
<td>
<p>Path of FASTA file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns data in data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")
input_df &lt;- fasta_to_df(input)
</code></pre>

<hr>
<h2 id='get_ensemble_method'>Get ensemble methods</h2><span id='topic+get_ensemble_method'></span>

<h3>Description</h3>

<p>This function is used to get the ensemble methods used for each taxon group. If weights are needed for a particular ensemble, then the weights will automatically follow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ensemble_method(taxon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ensemble_method_+3A_taxon">taxon</code></td>
<td>
<p>taxon group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns ensemble method and weights.
</p>

<hr>
<h2 id='install_tensorflow'>Install the TensorFlow backend</h2><span id='topic+install_tensorflow'></span>

<h3>Description</h3>

<p>TensorFlow will be installed into an &quot;r-tensorflow&quot; virtual or conda
environment. Note that &quot;virtualenv&quot; is not available on Windows (as this isn't
supported by TensorFlow).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_tensorflow(
  method = c("conda", "virtualenv"),
  conda = "auto",
  version = "2.0.0",
  extra_packages = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_tensorflow_+3A_method">method</code></td>
<td>
<p>Installation method (&quot;conda&quot; or &quot;virtualenv&quot;).</p>
</td></tr>
<tr><td><code id="install_tensorflow_+3A_conda">conda</code></td>
<td>
<p>The path to a <code>conda</code> executable. Use <code>"auto"</code> to allow
<code>reticulate</code> to automatically find an appropriate <code>conda</code> binary. See
<strong>Finding Conda</strong> for more details.</p>
</td></tr>
<tr><td><code id="install_tensorflow_+3A_version">version</code></td>
<td>
<p>TensorFlow version to install ( by default, &quot;2.0.0&quot;).</p>
</td></tr>
<tr><td><code id="install_tensorflow_+3A_extra_packages">extra_packages</code></td>
<td>
<p>Additional PyPI packages to install along with TensorFlow.</p>
</td></tr>
<tr><td><code id="install_tensorflow_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="reticulate.html#topic+py_install">reticulate::py_install()</a></code>.</p>
</td></tr>
</table>


<h3>Custom Installation</h3>

<p>Custom installations of TensorFlow are not supported yet by deepredeff.
</p>


<h3>Additional Packages</h3>

<p>If you wish to add additional PyPI packages to your TensorFlow environment you
can either specify the packages in the <code>extra_packages</code> argument of <code>install_tensorflow()</code>,
or alternatively install them into an existing environment using the
<code><a href="reticulate.html#topic+py_install">reticulate::py_install()</a></code> function.
</p>
<p>Notice that this may have undesired side-effects on Windows installations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Default installation
library(deepredeff)
install_tensorflow()

## End(Not run)

</code></pre>

<hr>
<h2 id='load_model'>Load model</h2><span id='topic+load_model'></span>

<h3>Description</h3>

<p><code>load_model()</code> loads model saved in hdf5 format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_model(taxon = c("bacteria", "fungi", "oomycete"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_model_+3A_taxon">taxon</code></td>
<td>
<p>Name of taxon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns all of the hyperparamaters and parameters of particular model from specific taxon chosen.
</p>

<hr>
<h2 id='package_glob'>Wildcard Expansion on File Paths</h2><span id='topic+package_glob'></span>

<h3>Description</h3>

<p>Wildcard Expansion on File Paths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>package_glob(..., pattern)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="package_glob_+3A_...">...</code></td>
<td>
<p>Path</p>
</td></tr>
<tr><td><code id="package_glob_+3A_pattern">pattern</code></td>
<td>
<p>Pattern</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Glob
</p>

<hr>
<h2 id='plot.tbl_deepredeff'>Plot the results of prediction</h2><span id='topic+plot.tbl_deepredeff'></span>

<h3>Description</h3>

<p>Plot the results of prediction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tbl_deepredeff'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.tbl_deepredeff_+3A_x">x</code></td>
<td>
<p>tbl_deepredeff object</p>
</td></tr>
<tr><td><code id="plot.tbl_deepredeff_+3A_...">...</code></td>
<td>
<p>additional arguments ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>class distribution plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# FASTA input
input_fasta &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")

pred_result &lt;- deepredeff::predict_effector(
  input = input_fasta,
  taxon = "fungi"
)

plot(pred_result)


</code></pre>

<hr>
<h2 id='predict_effector'>Predict effector</h2><span id='topic+predict_effector'></span><span id='topic+predict_effector.character'></span><span id='topic+predict_effector.data.frame'></span><span id='topic+predict_effector.AAStringSet'></span><span id='topic+predict_effector.AAString'></span><span id='topic+predict_effector.default'></span>

<h3>Description</h3>

<p><code>predict_effector</code> is used to predict effector protein given amino acid protein sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_effector(input, taxon)

## S3 method for class 'character'
predict_effector(input, taxon)

## S3 method for class 'data.frame'
predict_effector(input, taxon)

## S3 method for class 'AAStringSet'
predict_effector(input, taxon)

## S3 method for class 'AAString'
predict_effector(input, taxon)

## Default S3 method:
predict_effector(input, taxon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_effector_+3A_input">input</code></td>
<td>
<p>Input data that contains amino acid sequence(s). It can be in fasta format, strings, AAString, AAStringset, and dataframe.</p>
</td></tr>
<tr><td><code id="predict_effector_+3A_taxon">taxon</code></td>
<td>
<p>Taxon group of input data. Available taxons are bacteria, fungi, and oomycete.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>predict_effector</code> returns an object of class &quot;tbl_deepredeff&quot; or for multiple responses of class c(&quot;tbl_deepredeff&quot;, &quot;data.frame&quot;).
</p>
<p>An object of class &quot;tbl_deepredeff&quot; is a data frame containing at least the following components:
</p>
<table>
<tr><td><code>sequence</code></td>
<td>
<p>the sequence(s) from the input data.</p>
</td></tr>
<tr><td><code>s_score</code></td>
<td>
<p>score obtained from sigmoid function showing how likely the sequences to be an effector.</p>
</td></tr>
<tr><td><code>prediction</code></td>
<td>
<p>class prediction for each sequence, obtained from s_score. If the value of <code>s_score</code> &gt;= 0.5, it will be classified as an effector. Otherwise, it will be classified as a non-effector.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# FASTA input
input_fasta &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")

pred_result &lt;- deepredeff::predict_effector(
  input = input_fasta,
  taxon = "fungi"
)


</code></pre>

<hr>
<h2 id='prediction_mapper'>Prediction mapper helper</h2><span id='topic+prediction_mapper'></span>

<h3>Description</h3>

<p>Prediction mapper helper
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prediction_mapper(sequence_list, model_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prediction_mapper_+3A_sequence_list">sequence_list</code></td>
<td>
<p>List of sequences input.</p>
</td></tr>
<tr><td><code id="prediction_mapper_+3A_model_list">model_list</code></td>
<td>
<p>List of models.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns list of prediction result of each sequence.
</p>

<hr>
<h2 id='summary.tbl_deepredeff'>Get the summary of the prediction results</h2><span id='topic+summary.tbl_deepredeff'></span>

<h3>Description</h3>

<p>Get the summary of the prediction results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tbl_deepredeff'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.tbl_deepredeff_+3A_object">object</code></td>
<td>
<p>Results of prediction from deepredeff::predict_effector().</p>
</td></tr>
<tr><td><code id="summary.tbl_deepredeff_+3A_...">...</code></td>
<td>
<p>Additional arguments ignored.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# FASTA input
input_fasta &lt;- system.file("extdata/example/fungi_sample.fasta", package = "deepredeff")

pred_result &lt;- deepredeff::predict_effector(
  input = input_fasta,
  taxon = "fungi"
)

summary(pred_result)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
