<!DOCTYPE html><html lang="en"><head><title>Help for package deeptime</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {deeptime}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#deeptime-package'><p>deeptime: Plotting Tools for Anyone Working in Deep Time</p></a></li>
<li><a href='#coord_geo'><p>Transformed coordinate system with geological timescale</p></a></li>
<li><a href='#coord_geo_polar'><p>Polar coordinate system with geological timescale</p></a></li>
<li><a href='#coord_geo_radial'><p>Enhanced polar coordinate system with geological timescale</p></a></li>
<li><a href='#coord_trans_flip'><p>Transformed and flipped Cartesian coordinate system</p></a></li>
<li><a href='#coord_trans_xy'><p>Transformed XY Cartesian coordinate system</p></a></li>
<li><a href='#disparity_through_time'><p>Disparity through time plot using lattice</p></a></li>
<li><a href='#eons'><p>Eon data from the International Commission on Stratigraphy (v2023/06)</p></a></li>
<li><a href='#epochs'><p>Epoch data from the International Commission on Stratigraphy (v2023/06)</p></a></li>
<li><a href='#eras'><p>Era data from the International Commission on Stratigraphy (v2023/06)</p></a></li>
<li><a href='#facet_grid_color'><p>Lay out panels in a grid with colored strips</p></a></li>
<li><a href='#facet_nested_color'><p>Layout panels in a grid with nested colored strips</p></a></li>
<li><a href='#facet_nested_wrap_color'><p>Ribbon of panels with nested colored strips</p></a></li>
<li><a href='#facet_wrap_color'><p>Wrap a 1d ribbon of panels into 2d with colored strips</p></a></li>
<li><a href='#geo_pattern'><p>Get a FGDC geologic plotting pattern</p></a></li>
<li><a href='#geom_phylomorpho'><p>Plot a 2-D phylomorphospace in ggplot2</p></a></li>
<li><a href='#geom_points_range'><p>Display points and their range</p></a></li>
<li><a href='#get_scale_data'><p>Get geological timescale data</p></a></li>
<li><a href='#getScaleData'><p>Get geological timescale data</p></a></li>
<li><a href='#ggarrange2'><p>Combine and arrange multiple ggplot-like objects</p></a></li>
<li><a href='#gggeo_scale'><p>Add a geologic scale below ggplots using <em>grid</em></p></a></li>
<li><a href='#gggeo_scale_old'><p>Add a geologic scale on top of ggplots</p></a></li>
<li><a href='#grid.pattern_geo'><p>Plot an individual FGDC pattern using grid</p></a></li>
<li><a href='#gtable_frame2'><p>Decompose a ggplot gtable</p></a></li>
<li><a href='#guide_geo'><p>Geological timescale axis guide</p></a></li>
<li><a href='#panel.disparity'><p>Combined wireframe and cloud panel</p></a></li>
<li><a href='#periods'><p>Period data from the International Commission on Stratigraphy (v2023/06)</p></a></li>
<li><a href='#scale_color_geo'><p>Geological Time Scale color scales</p></a></li>
<li><a href='#scale_fill_geopattern'><p>Geologic pattern fill scale</p></a></li>
<li><a href='#stages'><p>Stage data from the International Commission on Stratigraphy (v2023/06)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Plotting Tools for Anyone Working in Deep Time</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>William Gearty &lt;willgearty@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Extends the functionality of other plotting packages (notably 'ggplot2')
    to help facilitate the plotting of data over long time intervals, including,
    but not limited to, geological, evolutionary, and ecological data. The
    primary goal of 'deeptime' is to enable users to add highly customizable
    timescales to their visualizations. Other functions are also included to
    assist with other areas of deep time visualization.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://williamgearty.com/deeptime/">https://williamgearty.com/deeptime/</a>,
<a href="https://github.com/willgearty/deeptime">https://github.com/willgearty/deeptime</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/willgearty/deeptime/issues">https://github.com/willgearty/deeptime/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>deeptimedata, ggplot2 (&ge; 3.5.0), utils, ggforce, grid,
gridExtra, gtable, methods, stats, lattice, rlang (&ge; 1.1.0),
scales, ggfittext, curl, cli, lifecycle, grImport2, ggh4x</td>
</tr>
<tr>
<td>Suggests:</td>
<td>geomtextpath, phytools, dplyr, divDyn, gsloid, ape,
palaeoverse, paleotree, dispRity, ggtree (&ge; 3.6.1), testthat
(&ge; 3.0.0), vdiffr (&ge; 1.0.0), knitr, rmarkdown, withr,
ggpattern, ggrepel, rmacrostrat, svglite</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Config/Needs/revdeps:</td>
<td>revdepcheck</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-25 14:07:05 UTC; willg</td>
</tr>
<tr>
<td>Author:</td>
<td>William Gearty <a href="https://orcid.org/0000-0003-0076-3262"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-25 23:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='deeptime-package'>deeptime: Plotting Tools for Anyone Working in Deep Time</h2><span id='topic+deeptime'></span><span id='topic+deeptime-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Extends the functionality of other plotting packages (notably 'ggplot2') to help facilitate the plotting of data over long time intervals, including, but not limited to, geological, evolutionary, and ecological data. The primary goal of 'deeptime' is to enable users to add highly customizable timescales to their visualizations. Other functions are also included to assist with other areas of deep time visualization.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: William Gearty <a href="mailto:willgearty@gmail.com">willgearty@gmail.com</a> (<a href="https://orcid.org/0000-0003-0076-3262">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://williamgearty.com/deeptime/">https://williamgearty.com/deeptime/</a>
</p>
</li>
<li> <p><a href="https://github.com/willgearty/deeptime">https://github.com/willgearty/deeptime</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/willgearty/deeptime/issues">https://github.com/willgearty/deeptime/issues</a>
</p>
</li></ul>


<hr>
<h2 id='coord_geo'>Transformed coordinate system with geological timescale</h2><span id='topic+coord_geo'></span><span id='topic+CoordGeo'></span>

<h3>Description</h3>

<p><code>coord_geo</code> behaves similarly to <code><a href="ggplot2.html#topic+coord_trans">ggplot2::coord_trans()</a></code> in that it occurs
after statistical transformation and will affect the visual appearance of
geoms. The main difference is that it also adds a geological timescale to the
specified side(s) of the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_geo(
  pos = "bottom",
  dat = "periods",
  xlim = NULL,
  ylim = NULL,
  xtrans = identity_trans(),
  ytrans = identity_trans(),
  clip = "on",
  expand = FALSE,
  fill = NULL,
  alpha = 1,
  height = unit(2, "line"),
  bord = c("left", "right", "top", "bottom"),
  lwd = 0.25,
  color = "black",
  lab = TRUE,
  lab_color = NULL,
  rot = 0,
  family = "sans",
  fontface = "plain",
  size = 5,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  abbrv = TRUE,
  neg = FALSE,
  center_end_labels = FALSE,
  dat_is_discrete = FALSE,
  fittext_args = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_geo_+3A_pos">pos</code></td>
<td>
<p>Which side to add the scale to (left, right, top, or bottom).
First letter may also be used.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;),
B) a string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>), or C) a custom
data.frame of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_xlim">xlim</code>, <code id="coord_geo_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the x and y axes.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_xtrans">xtrans</code>, <code id="coord_geo_+3A_ytrans">ytrans</code></td>
<td>
<p>Transformers for the x and y axes. For more information
see <code><a href="ggplot2.html#topic+coord_trans">ggplot2::coord_trans()</a></code>.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. In most cases, the default of <code>"on"</code> should not be changed,
as setting <code>clip = "off"</code> can cause unexpected results. It allows
drawing of data points anywhere on the plot, including in the plot margins. If
limits are set via <code>xlim</code> and <code>ylim</code> and some data points fall outside those
limits, then those data points may show up in places such as the axes, the
legend, the plot title, or the plot margins.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_expand">expand</code></td>
<td>
<p>If <code>FALSE</code>, the default, limits are taken exactly from the data
or <code>xlim</code>/<code>ylim</code>. If <code>TRUE</code>, adds a small expansion factor to the limits to
ensure that data and axes don't overlap.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_fill">fill</code></td>
<td>
<p>The fill color of the boxes. The default is to use the <code>color</code>
column included in <code>dat</code>. If a custom dataset is provided with <code>dat</code>
without a <code>color</code> column and without fill, a greyscale will be used.
Custom fill colors can be provided with this option (overriding the <code>color</code>
column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_height">height</code></td>
<td>
<p>The height (or width if <code>pos</code> is <code>left</code> or <code>right</code>) of the
scale.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_bord">bord</code></td>
<td>
<p>A vector specifying on which sides of the scale to add borders
(same options as <code>pos</code>).</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_color">color</code></td>
<td>
<p>The outline color of the interval boxes.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_lab_color">lab_color</code></td>
<td>
<p>The color of the labels. The default is to use the
<code>lab_color</code> column included in <code>dat</code>. If a custom dataset is provided with
<code>dat</code> without a <code>lab_color</code> column and without fill, all labels will be
black. Custom label colors can be provided with this option (overriding the
<code>lab_color</code> column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_rot">rot</code></td>
<td>
<p>The amount of counter-clockwise rotation to add to the labels
(in degrees).</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_family">family</code></td>
<td>
<p>The font family to use for the labels. There are only three
fonts that are guaranteed to work everywhere: “sans” (the default),
“serif”, or “mono”.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_fontface">fontface</code></td>
<td>
<p>The font face to use for the labels. The standard options are
&quot;plain&quot; (default), &quot;bold&quot;, &quot;italic&quot;, and &quot;bold.italic&quot;.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_size">size</code></td>
<td>
<p>Label size. Either a number as you would specify in
<code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code> or <code>"auto"</code> to use <code><a href="ggfittext.html#topic+geom_fit_text">ggfittext::geom_fit_text()</a></code>.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled. If <code>abbrv</code> is <code>TRUE</code>, this can also include interval
abbreviations.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, should the labels be abbreviated? If
<code>TRUE</code>, the <code>abbr</code> column will be used for the labels. If <code>FALSE</code>, the
<code>name</code> column will be used for the labels. If <code>"auto"</code>, the <code><a href="base.html#topic+abbreviate">abbreviate()</a></code>
function will be used to abbreviate the values in the <code>name</code> column. Note
that the built-in data and data retrieved via <code><a href="#topic+get_scale_data">get_scale_data()</a></code> already
have built-in abbreviations. However, using the <code>"auto"</code> option here will
create new unique abbreviations based on only the intervals that are being
plotted. In many cases, this may result in abbreviations that are shorter
in length because there are fewer similar interval names to abbreviate.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_neg">neg</code></td>
<td>
<p>Set this to <code>TRUE</code> if your x-axis is using negative values.</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_center_end_labels">center_end_labels</code></td>
<td>
<p>Should labels be centered within the visible range
of intervals at the ends of the axis?</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_dat_is_discrete">dat_is_discrete</code></td>
<td>
<p>Are the ages in <code>dat</code> already converted for a discrete
scale?</p>
</td></tr>
<tr><td><code id="coord_geo_+3A_fittext_args">fittext_args</code></td>
<td>
<p>A list of named arguments to provide to
<code><a href="ggfittext.html#topic+geom_fit_text">ggfittext::geom_fit_text()</a></code>. Only used if <code>size</code> is set to <code>"auto"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Transforming the side with the scale is not currently implemented.
If a custom data.frame is provided (with <code>dat</code>), it should consist of at
least 3 columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>name</code> column lists the names of each time interval. These will
be used as labels if no abbreviations are provided.
</p>
</li>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is also optional and lists a
<a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a> for the background for each time interval.
</p>
</li>
<li><p> The <code>lab_color</code> column is also optional and lists a
<a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a> for the label for each time interval.
</p>
</li></ul>

<p>If the axis of the time scale is discrete, <code>max_age</code> and <code>min_age</code> will
automatically be converted to the discrete scale. In this case, the
categories of the discrete axis should match the values in the <code>name</code> column.
If the ages within <code>dat</code> are already discretized, you can set
<code>dat_is_discrete</code> to <code>TRUE</code> to prevent this automatic conversion. This can be
useful for adding a time scale where categories and time intervals are not
1:1.
</p>
<p><code>pos</code> may also be a <code>list</code> of sides (including duplicates) if multiple time
scales should be added to the plot. In this case, <code>dat</code>, <code>fill</code>, <code>alpha</code>,
<code>height</code>, <code>bord</code>, <code>lwd</code>, <code>color</code>, <code>lab</code>, <code>lab_color</code>, <code>rot</code>, <code>family</code>,
<code>fontface</code>, <code>size</code>, <code>skip</code>, <code>abbrv</code>, <code>neg</code>, <code>center_end_labels</code>, and
<code>dat_is_discrete</code> can also be <code>list</code>s. If these <code>list</code>s are not as long as
<code>pos</code>, the elements will be recycled.
If individual values (or vectors) are used for these parameters, they will be
applied to all time scales (and recycled as necessary).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
# single scale on bottom
ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 1000))) +
  scale_x_reverse() +
  coord_geo(xlim = c(1000, 0), ylim = c(0, 8)) +
  theme_classic()

# stack multiple scales
ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 100))) +
  scale_x_reverse() +
  coord_geo(
    xlim = c(100, 0), ylim = c(0, 8), pos = as.list(rep("bottom", 3)),
    dat = list("stages", "epochs", "periods"),
    height = list(unit(4, "lines"), unit(4, "lines"), unit(2, "line")),
    rot = list(90, 90, 0), size = list(2.5, 2.5, 5), abbrv = FALSE
  ) +
  theme_classic()
</code></pre>

<hr>
<h2 id='coord_geo_polar'>Polar coordinate system with geological timescale</h2><span id='topic+coord_geo_polar'></span><span id='topic+CoordGeoPolar'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><code>coord_geo_polar</code> behaves similarly to <code><a href="ggplot2.html#topic+coord_polar">ggplot2::coord_polar()</a></code> in that it
occurs after statistical transformation and will affect the visual appearance
of geoms. The main difference is that it also adds a geological timescale to
the background of the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_geo_polar(
  dat = "periods",
  theta = "y",
  start = -pi/2,
  direction = -1,
  clip = "on",
  fill = NULL,
  alpha = 1,
  lwd = 0.25,
  color = "grey80",
  lty = "solid",
  lab = FALSE,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  neg = TRUE,
  prop = 1,
  textpath_args = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_geo_polar_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;), B) a
string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>), or C) a custom
data.frame of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_theta">theta</code></td>
<td>
<p>variable to map angle to (<code>x</code> or <code>y</code>)</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_start">start</code></td>
<td>
<p>Offset of starting point from 12 o'clock in radians. Offset
is applied clockwise or anticlockwise depending on value of <code>direction</code>.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_direction">direction</code></td>
<td>
<p>1, clockwise; -1, anticlockwise</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. For details, please see <code><a href="ggplot2.html#topic+coord_cartesian">coord_cartesian()</a></code>.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_fill">fill</code></td>
<td>
<p>The fill color of the background. The default is to use the
<code>color</code> column included in <code>dat</code>. If a custom dataset is provided with
<code>dat</code> without a <code>color</code> column and without fill, a greyscale will be used.
Custom fill colors can be provided with this option (overriding the <code>color</code>
column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_lwd">lwd</code></td>
<td>
<p>Line width for lines between intervals. Set to <code>NULL</code> to remove
lines.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_color">color</code></td>
<td>
<p>The color of the lines between intervals.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_lty">lty</code></td>
<td>
<p>Line type for lines between intervals.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels. Requires the <code>geomtextpath</code> package.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, whether to use abbreviations instead of
full interval names.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled. If <code>abbrv</code> is <code>TRUE</code>, this can also include interval
abbreviations.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_neg">neg</code></td>
<td>
<p>Set this to true if your theta-axis is using negative values. This
is usually true if you are using <code>ggtree</code>.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_prop">prop</code></td>
<td>
<p>This is the rotational proportion of the background that the
scale takes up.</p>
</td></tr>
<tr><td><code id="coord_geo_polar_+3A_textpath_args">textpath_args</code></td>
<td>
<p>A list of named arguments to provide to
<code><a href="geomtextpath.html#topic+geom_textpath">geomtextpath::geom_textpath()</a></code>. Only used if <code>lab</code> is set to <code>TRUE</code>.
Useful arguments include <code>color</code> (font color), <code>family</code> (font family),
<code>fontface</code>, <code>hjust</code> (radial adjustment), and <code>size</code> (font size).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a custom data.frame is provided (with <code>dat</code>), it should consist of at
least 2 columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is optional and lists a <a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a>
for the background for each time interval.
</p>
</li></ul>

<p><code>dat</code> may also be a list of values and/or dataframes if multiple time scales
should be added to the background. Scales will be added sequentially starting
at <code>start</code> and going in the specified <code>direction</code>. By default the scales will
all be equal in circular/rotational proportion, but this can be overridden
with <code>prop</code>. If <code>dat</code> is a list, <code>fill</code>, <code>alpha</code>, <code>lwd</code>, <code>color</code>, <code>lty</code>,
<code>lab</code>, <code>abbrv</code>, <code>skip</code>, <code>neg</code>, <code>prop</code>, and <code>textpath_args</code> can also be lists
(N.B. <code>textpath_args</code> would be a list of lists). If these lists are not as
long as <code>dat</code>, the elements will be recycled. If individual values (or
vectors) are used for these parameters, they will be applied to all time
scales (and recycled as necessary).
</p>
<p>If the sum of the <code>prop</code> values is greater than 1, the proportions will be
scaled such that they sum to 1. However, the <code>prop</code> values may sum to less
than 1 if the user would like blank space in the background.
</p>
<p><code>coord_geo_polar</code> manually generates the <code>r</code> axis, meaning it does not
support changing the guide features of ggplot v. 2.5.0 or later. However, the
<code>deeptime.axis.line.r</code>, <code>deeptime.axis.text.r</code>, <code>deeptime.axis.ticks.r</code>, and
<code>deeptime.axis.ticks.length.r</code> ggplot2 <a href="ggplot2.html#topic+theme">theme elements</a> can
be modified just like their x and y counterparts to change the appearance of
the radius axis. The default settings work well for a horizontal axis
pointing towards the right, but these theme settings will need to be modified
for other orientations. The default value for <code>deeptime.axis.line.r</code> is
<code>element_line()</code>. The default value for <code>deeptime.axis.text.r</code> is
<code>element_text(size = 3.5, vjust = -2, hjust = NA)</code>. The default value for
<code>deeptime.axis.ticks.r</code> is <code>element_line()</code>. The default value for
<code>deeptime.axis.ticks.length.r</code> is <code>unit(1.5, "points")</code>. However, note that
the units for this element are meaningless and only the numeric value will be
used (but a <code>unit</code> must still be used).
</p>
<p>Care must be taken when adding labels to plots, as they are very likely to
overlap with the plot under the default settings. The <code>textpath_args</code>
argument can be used to adjust the settings for the plotting of the labels.
See <code><a href="geomtextpath.html#topic+geom_textpath">geomtextpath::geom_textpath()</a></code> for details about the available
arguments. Also note that the curvature of the labels may vary based on the
distance from the origin. This is why <code>abbrv</code> is set to <code>TRUE</code> by default.
</p>


<h3>Life cycle</h3>

<p>This function is soft-deprecated in favor of
<code><a href="#topic+coord_geo_radial">coord_geo_radial()</a></code> as of <strong>deeptime</strong> version 1.1.0. There is currently
no plan to remove this function, but users are strongly encouraged to
migrate to the new function for enhanced polar functionality.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

library(ggtree)
set.seed(1)
tree &lt;- rtree(100)
# single scale
revts(ggtree(tree)) +
  coord_geo_polar(dat = "stages")

# multiple scales
revts(ggtree(tree)) +
  coord_geo_polar(
    dat = list("stages", "periods"), alpha = .5,
    prop = list(0.75, .25), start = pi / 4, lty = "dashed"
  ) +
  scale_y_continuous(expand = expansion(mult = c(0.02, 0.02))) +
  theme(deeptime.axis.text.r = element_text(size = 3.5, hjust = .75,
                                            vjust = .75))


library(ggplot2)
library(paleotree)
data(RaiaCopesRule)
ggtree(ceratopsianTreeRaia,
       position = position_nudge(x = -ceratopsianTreeRaia$root.time)) +
  coord_geo_polar(dat = "stages")

</code></pre>

<hr>
<h2 id='coord_geo_radial'>Enhanced polar coordinate system with geological timescale</h2><span id='topic+coord_geo_radial'></span><span id='topic+CoordGeoRadial'></span>

<h3>Description</h3>

<p><code>coord_geo_radial</code> behaves similarly to <code><a href="ggplot2.html#topic+coord_polar">ggplot2::coord_radial()</a></code> in that it
occurs after statistical transformation and will affect the visual appearance
of geoms. The main difference is that it also adds a geological timescale to
the background of the plot. <code>coord_geo_radial</code> is similar to
<code><a href="#topic+coord_geo_polar">coord_geo_polar()</a></code> but has more options related to the polar coordinate
plotting that are inherited from <code><a href="ggplot2.html#topic+coord_polar">ggplot2::coord_radial()</a></code> (e.g., <code>end</code>,
<code>r_axis_inside</code>, <code>inner.radius</code>). Furthermore, unlike <code>coord_geo_polar</code>,
<code>coord_geo_radial</code> uses the ggplot2 internals to draw the <code>r</code> and <code>theta</code>
axes, gridlines, etc. This means that users can tweak the
<a href="ggplot2.html#topic+guides">guide</a> and <a href="ggplot2.html#topic+theme">theme</a> settings for these
features (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_geo_radial(
  dat = "periods",
  theta = "y",
  start = -0.5 * pi,
  end = 1.25 * pi,
  expand = TRUE,
  direction = 1,
  r_axis_inside = NULL,
  inner.radius = 0.05,
  fill = NULL,
  alpha = 1,
  lwd = 0.25,
  color = "grey80",
  lty = "solid",
  lab = FALSE,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  neg = TRUE,
  prop = 1,
  textpath_args = list(),
  clip = "off",
  rotate_angle = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_geo_radial_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;), B) a
string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>), or C) a custom
data.frame of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_theta">theta</code></td>
<td>
<p>variable to map angle to (<code>x</code> or <code>y</code>)</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_start">start</code></td>
<td>
<p>Offset of starting point from 12 o'clock in radians. Offset
is applied clockwise or anticlockwise depending on value of <code>direction</code>.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_end">end</code></td>
<td>
<p>Position from 12 o'clock in radians where plot ends, to allow
for partial polar coordinates. The default, <code>NULL</code>, is set to
<code>start + 2 * pi</code>.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_expand">expand</code></td>
<td>
<p>If <code>TRUE</code>, the default, adds a small expansion factor the
the limits to prevent overlap between data and axes. If <code>FALSE</code>, limits
are taken directly from the scale.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_direction">direction</code></td>
<td>
<p>1, clockwise; -1, anticlockwise</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_r_axis_inside">r_axis_inside</code>, <code id="coord_geo_radial_+3A_rotate_angle">rotate_angle</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a></p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_inner.radius">inner.radius</code></td>
<td>
<p>A <code>numeric</code> between 0 and 1 setting the size of a inner.radius hole.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_fill">fill</code></td>
<td>
<p>The fill color of the background. The default is to use the
<code>color</code> column included in <code>dat</code>. If a custom dataset is provided with
<code>dat</code> without a <code>color</code> column and without fill, a greyscale will be used.
Custom fill colors can be provided with this option (overriding the <code>color</code>
column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_lwd">lwd</code></td>
<td>
<p>Line width for lines between intervals. Set to <code>NULL</code> to remove
lines.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_color">color</code></td>
<td>
<p>The color of the lines between intervals.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_lty">lty</code></td>
<td>
<p>Line type for lines between intervals.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels. Requires the <code>geomtextpath</code> package.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, whether to use abbreviations instead of
full interval names.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled. If <code>abbrv</code> is <code>TRUE</code>, this can also include interval
abbreviations.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_neg">neg</code></td>
<td>
<p>Set this to true if your theta-axis is using negative values. This
is usually true if you are using <code>ggtree</code>.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_prop">prop</code></td>
<td>
<p>This is the rotational proportion of the background that the
scale takes up.</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_textpath_args">textpath_args</code></td>
<td>
<p>A list of named arguments to provide to
<code><a href="geomtextpath.html#topic+geom_textpath">geomtextpath::geom_textpath()</a></code>. Only used if <code>lab</code> is set to <code>TRUE</code>.
Useful arguments include <code>color</code> (font color), <code>family</code> (font family),
<code>fontface</code>, <code>hjust</code> (radial adjustment), and <code>size</code> (font size).</p>
</td></tr>
<tr><td><code id="coord_geo_radial_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. For details, please see <code><a href="ggplot2.html#topic+coord_cartesian">coord_cartesian()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a custom data.frame is provided (with <code>dat</code>), it should consist of at
least 2 columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is optional and lists a <a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a>
for the background for each time interval.
</p>
</li></ul>

<p><code>dat</code> may also be a list of values and/or dataframes if multiple time scales
should be added to the background. Scales will be added sequentially starting
at <code>start</code> and going in the specified <code>direction</code>. By default the scales will
all be equal in circular/rotational proportion, but this can be overridden
with <code>prop</code>. If <code>dat</code> is a list, <code>fill</code>, <code>alpha</code>, <code>lwd</code>, <code>color</code>, <code>lty</code>,
<code>lab</code>, <code>abbrv</code>, <code>skip</code>, <code>neg</code>, <code>prop</code>, and <code>textpath_args</code> can also be lists
(N.B. <code>textpath_args</code> would be a list of lists). If these lists are not as
long as <code>dat</code>, the elements will be recycled. If individual values (or
vectors) are used for these parameters, they will be applied to all time
scales (and recycled as necessary).
</p>
<p>If the sum of the <code>prop</code> values is greater than 1, the proportions will be
scaled such that they sum to 1. However, the <code>prop</code> values may sum to less
than 1 if the user would like blank space in the background.
</p>
<p>Care must be taken when adding labels to plots, as they are very likely to
overlap with the plot under the default settings. The <code>textpath_args</code>
argument can be used to adjust the settings for the plotting of the labels.
See <code><a href="geomtextpath.html#topic+geom_textpath">geomtextpath::geom_textpath()</a></code> for details about the available
arguments. Also note that the curvature of the labels may vary based on the
distance from the origin. This is why <code>abbrv</code> is set to <code>TRUE</code> by default.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

library(ggtree)
set.seed(1)
tree &lt;- rtree(100)
# single scale
revts(ggtree(tree)) +
  coord_geo_radial(dat = "stages") +
  scale_y_continuous(guide = "none", breaks = NULL) +
  theme_gray()

# multiple scales
revts(ggtree(tree)) +
  coord_geo_radial(
    dat = list("stages", "periods"), alpha = .5,
    prop = list(0.75, .25), start = pi / 4, end = 2 * pi, lty = "dashed"
  ) +
  scale_y_continuous(expand = expansion(mult = c(0.02, 0.02)),
                     guide = "none", breaks = NULL) +
  theme_gray()


library(ggplot2)
library(paleotree)
data(RaiaCopesRule)
ggtree(ceratopsianTreeRaia,
       position = position_nudge(x = -ceratopsianTreeRaia$root.time)) +
  coord_geo_radial(dat = "stages") +
  scale_y_continuous(guide = "none", breaks = NULL) +
  theme_classic()

</code></pre>

<hr>
<h2 id='coord_trans_flip'>Transformed and flipped Cartesian coordinate system</h2><span id='topic+coord_trans_flip'></span><span id='topic+CoordTransFlip'></span>

<h3>Description</h3>

<p><code>coord_trans_flip</code> behaves similarly to <code><a href="ggplot2.html#topic+coord_trans">ggplot2::coord_trans()</a></code> in that it
occurs after statistical transformation and will affect the visual appearance
of geoms. The main difference is that it also flips the x and y coordinates
like <code><a href="ggplot2.html#topic+coord_flip">ggplot2::coord_flip()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_trans_flip(
  x = "identity",
  y = "identity",
  xlim = NULL,
  ylim = NULL,
  clip = "on",
  expand = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_trans_flip_+3A_x">x</code>, <code id="coord_trans_flip_+3A_y">y</code></td>
<td>
<p>Transformers for x and y axes or their names.</p>
</td></tr>
<tr><td><code id="coord_trans_flip_+3A_xlim">xlim</code>, <code id="coord_trans_flip_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the x and y axes.</p>
</td></tr>
<tr><td><code id="coord_trans_flip_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. In most cases, the default of <code>"on"</code> should not be changed,
as setting <code>clip = "off"</code> can cause unexpected results. It allows
drawing of data points anywhere on the plot, including in the plot margins. If
limits are set via <code>xlim</code> and <code>ylim</code> and some data points fall outside those
limits, then those data points may show up in places such as the axes, the
legend, the plot title, or the plot margins.</p>
</td></tr>
<tr><td><code id="coord_trans_flip_+3A_expand">expand</code></td>
<td>
<p>If <code>TRUE</code>, the default, adds a small expansion factor to
the limits to ensure that data and axes don't overlap. If <code>FALSE</code>,
limits are taken exactly from the data or <code>xlim</code>/<code>ylim</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
ggplot(mtcars, aes(disp, wt)) +
  geom_point() +
  coord_trans_flip(x = "log10", y = "log10")
</code></pre>

<hr>
<h2 id='coord_trans_xy'>Transformed XY Cartesian coordinate system</h2><span id='topic+coord_trans_xy'></span><span id='topic+CoordTransXY'></span>

<h3>Description</h3>

<p><code>coord_trans_xy</code> behaves similarly to <code><a href="ggplot2.html#topic+coord_trans">ggplot2::coord_trans()</a></code> in that it
occurs after statistical transformation and will affect the visual appearance
of geoms. The main difference is that it takes a single transformer that is
applied to the x and y axes simultaneously. Any transformers produced by
<code><a href="ggforce.html#topic+linear_trans">ggforce::linear_trans()</a></code> that have x and y arguments should work, but any
other transformers produced using <code><a href="scales.html#topic+new_transform">scales::trans_new()</a></code> that take x and y
arguments should also work. Axis limits will be adjusted to account for
transformation unless limits are specified with <code>xlim</code> or <code>ylim</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_trans_xy(
  trans = NULL,
  xlim = NULL,
  ylim = NULL,
  expand = FALSE,
  default = FALSE,
  clip = "on"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coord_trans_xy_+3A_trans">trans</code></td>
<td>
<p>Transformer for x and y axes.</p>
</td></tr>
<tr><td><code id="coord_trans_xy_+3A_xlim">xlim</code>, <code id="coord_trans_xy_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the x and y axes.</p>
</td></tr>
<tr><td><code id="coord_trans_xy_+3A_expand">expand</code></td>
<td>
<p>If <code>TRUE</code>, the default, adds a small expansion factor to
the limits to ensure that data and axes don't overlap. If <code>FALSE</code>,
limits are taken exactly from the data or <code>xlim</code>/<code>ylim</code>.</p>
</td></tr>
<tr><td><code id="coord_trans_xy_+3A_default">default</code></td>
<td>
<p>Is this the default coordinate system? If <code>FALSE</code> (the default),
then replacing this coordinate system with another one creates a message alerting
the user that the coordinate system is being replaced. If <code>TRUE</code>, that warning
is suppressed.</p>
</td></tr>
<tr><td><code id="coord_trans_xy_+3A_clip">clip</code></td>
<td>
<p>Should drawing be clipped to the extent of the plot panel? A
setting of <code>"on"</code> (the default) means yes, and a setting of <code>"off"</code>
means no. In most cases, the default of <code>"on"</code> should not be changed,
as setting <code>clip = "off"</code> can cause unexpected results. It allows
drawing of data points anywhere on the plot, including in the plot margins. If
limits are set via <code>xlim</code> and <code>ylim</code> and some data points fall outside those
limits, then those data points may show up in places such as the axes, the
legend, the plot title, or the plot margins.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This coordinate system only works with geoms where all points are
defined with x and y coordinates (e.g., <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>,
<code><a href="ggplot2.html#topic+geom_polygon">ggplot2::geom_polygon()</a></code>). This does not currently work with geoms where
point coordinates are extrapolated (e.g., <code><a href="ggplot2.html#topic+geom_tile">ggplot2::geom_rect()</a></code>).
Furthermore, when used with ggplot2 3.5.0 and later, some transformation
edge cases may cause problems with rendering axis lines. This includes not
currently support &quot;capping&quot; axes. I hope to support all of these geoms,
edge cases, and features in the future.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># make transformer
library(ggforce)
trans &lt;- linear_trans(shear(2, 0), rotate(-pi / 3))

# set up data to be plotted
square &lt;- data.frame(x = c(0, 0, 4, 4), y = c(0, 1, 1, 0))
points &lt;- data.frame(x = runif(100, 0, 4), y = runif(100, 0, 1))

# plot data normally
library(ggplot2)
ggplot(data = points, aes(x = x, y = y)) +
  geom_polygon(data = square, fill = NA, color = "black") +
  geom_point(color = "black") +
  coord_cartesian(expand = FALSE) +
  theme_classic()

# plot data with transformation
ggplot(data = points, aes(x = x, y = y)) +
  geom_polygon(data = square, fill = NA, color = "black") +
  geom_point(color = "black") +
  coord_trans_xy(trans = trans, expand = FALSE) +
  theme_classic()
</code></pre>

<hr>
<h2 id='disparity_through_time'>Disparity through time plot using lattice</h2><span id='topic+disparity_through_time'></span>

<h3>Description</h3>

<p>Plots points on 2-D surfaces within a a 3-D framework. See
<code><a href="lattice.html#topic+cloud">lattice::wireframe()</a></code> and <code><a href="lattice.html#topic+panel.cloud">lattice::panel.cloud()</a></code> for customization
options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disparity_through_time(
  x,
  data,
  groups,
  pch = 16,
  col.point = c("blue"),
  scales = list(arrows = FALSE, distance = 1, col = "black", z = list(rot = 90)),
  colorkey = FALSE,
  screen = list(z = 90, x = 70, y = 180),
  aspect = c(1.5, 4),
  drape = TRUE,
  col.regions = c("white"),
  alpha.regions = c(1),
  perspective = FALSE,
  R.mat = matrix(c(1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1), 4, 4),
  par.settings = list(axis.line = list(col = "transparent"), layout.heights =
    list(top.padding = 0, main.key.padding = 0, key.axis.padding = 0, axis.xlab.padding =
    0, xlab.key.padding = 0, key.sub.padding = 0, bottom.padding = 0), layout.widths =
    list(left.padding = 0, key.ylab.padding = 0, ylab.axis.padding = 0, axis.key.padding
    = 0, right.padding = 0)),
  lattice.options = list(axis.padding = list(factor = 0)),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="disparity_through_time_+3A_x">x</code></td>
<td>
<p>a formula (most likely of the form <code>z ~ x * y</code>)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_data">data</code></td>
<td>
<p>a data frame in which variables in the formula are to be
evaluated</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_groups">groups</code></td>
<td>
<p>a variable in <code>data</code> to be used as a grouping variable (this is
probably the z variable)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_pch">pch</code></td>
<td>
<p>the point type</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_col.point">col.point</code></td>
<td>
<p>color(s) for points on surfaces</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_scales">scales</code></td>
<td>
<p>a list specifying how the axes are drawn (see
<code><a href="lattice.html#topic+xyplot">lattice::xyplot()</a></code> for details)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_colorkey">colorkey</code></td>
<td>
<p>logical, should a legend be drawn (or a list describing the
legend; see <code><a href="lattice.html#topic+levelplot">lattice::levelplot()</a></code> for details)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_screen">screen</code></td>
<td>
<p>a list of the rotations that should be applied to each axis</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_aspect">aspect</code></td>
<td>
<p>a numeric vector of length 2, giving the relative aspects of
the y-size/x-size and z-size/x-size of the enclosing cube</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_drape">drape</code></td>
<td>
<p>logical, whether the surfaces should be colored based on
<code>col.regions</code> and <code>alpha.regions</code></p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_col.regions">col.regions</code></td>
<td>
<p>color(s) for surfaces</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_alpha.regions">alpha.regions</code></td>
<td>
<p>alpha value(s) for surfaces</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_perspective">perspective</code></td>
<td>
<p>logical, whether to plot a perspective view</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_r.mat">R.mat</code></td>
<td>
<p>a transformational matrix that is applied to the orientation of
the axes</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_par.settings">par.settings</code></td>
<td>
<p>plotting settings (see <code><a href="lattice.html#topic+trellis.par.get">lattice::trellis.par.set()</a></code>)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_lattice.options">lattice.options</code></td>
<td>
<p>lattice settings (see <code><a href="lattice.html#topic+lattice.options">lattice::lattice.options()</a></code>)</p>
</td></tr>
<tr><td><code id="disparity_through_time_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="lattice.html#topic+cloud">lattice::wireframe()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"trellis"</code>, as output by <code><a href="lattice.html#topic+cloud">lattice::wireframe()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- data.frame(
  x = runif(100, 0, 60), y = runif(100, 0, 10),
  z = factor(rep(periods$name[1:5], each = 20),
    levels = periods$name[1:5]
  )
)
disparity_through_time(z ~ x * y,
  data = g, groups = z, aspect = c(1.5, 2),
  xlim = c(0, 60), ylim = c(0, 10), col.regions = "lightgreen",
  col.point = c("red", "blue")
)
</code></pre>

<hr>
<h2 id='eons'>Eon data from the International Commission on Stratigraphy (v2023/06)</h2><span id='topic+eons'></span>

<h3>Description</h3>

<p>A dataset containing the boundary ages, abbreviations, and colors for the
eons of the Geologic Time Scale. Based on The ICS International
Chronostratigraphic Chart (v2023/06), by Cohen, Finney, Gibbard, and Fan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eons
</code></pre>


<h3>Format</h3>

<p>A data frame with 3 rows and 5 variables:
</p>

<dl>
<dt>name</dt><dd><p>eon name</p>
</dd>
<dt>max_age</dt><dd><p>maximum age, in millions of years</p>
</dd>
<dt>min_age</dt><dd><p>minimum age, in millions of years</p>
</dd>
<dt>abbr</dt><dd><p>eon name abbreviations</p>
</dd>
<dt>color</dt><dd><p>the colors for each eon, according to the Commission for the
Geological Map of the World</p>
</dd>
<dt>lab_color</dt><dd><p>the label colors for each eon, either white or black,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://stratigraphy.org">https://stratigraphy.org</a> via <a href="https://macrostrat.org/api/v2/defs/intervals?timescale=international%20eons">https://macrostrat.org/api/v2/defs/intervals?timescale=international%20eons</a>
</p>


<h3>See Also</h3>

<p>Other built-in timescales: 
<code><a href="#topic+epochs">epochs</a></code>,
<code><a href="#topic+eras">eras</a></code>,
<code><a href="#topic+periods">periods</a></code>,
<code><a href="#topic+stages">stages</a></code>
</p>

<hr>
<h2 id='epochs'>Epoch data from the International Commission on Stratigraphy (v2023/06)</h2><span id='topic+epochs'></span>

<h3>Description</h3>

<p>A dataset containing the boundary ages, abbreviations, and colors for the
epochs of the Geologic Time Scale. Based on The ICS International
Chronostratigraphic Chart (v2023/06), by Cohen, Finney, Gibbard, and Fan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epochs
</code></pre>


<h3>Format</h3>

<p>A data frame with 34 rows and 5 variables:
</p>

<dl>
<dt>name</dt><dd><p>epoch name</p>
</dd>
<dt>max_age</dt><dd><p>maximum age, in millions of years</p>
</dd>
<dt>min_age</dt><dd><p>minimum age, in millions of years</p>
</dd>
<dt>abbr</dt><dd><p>epoch name abbreviations</p>
</dd>
<dt>color</dt><dd><p>the colors for each epoch, according to the Commission for the
Geological Map of the World</p>
</dd>
<dt>lab_color</dt><dd><p>the label colors for each epoch, either white or black,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://stratigraphy.org">https://stratigraphy.org</a> via <a href="https://macrostrat.org/api/v2/defs/intervals?timescale=international%20epochs">https://macrostrat.org/api/v2/defs/intervals?timescale=international%20epochs</a>
</p>


<h3>See Also</h3>

<p>Other built-in timescales: 
<code><a href="#topic+eons">eons</a></code>,
<code><a href="#topic+eras">eras</a></code>,
<code><a href="#topic+periods">periods</a></code>,
<code><a href="#topic+stages">stages</a></code>
</p>

<hr>
<h2 id='eras'>Era data from the International Commission on Stratigraphy (v2023/06)</h2><span id='topic+eras'></span>

<h3>Description</h3>

<p>A dataset containing the boundary ages, abbreviations, and colors for the
eras of the Geologic Time Scale. Based on The ICS International
Chronostratigraphic Chart (v2023/06), by Cohen, Finney, Gibbard, and Fan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eras
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 rows and 5 variables:
</p>

<dl>
<dt>name</dt><dd><p>era name</p>
</dd>
<dt>max_age</dt><dd><p>maximum age, in millions of years</p>
</dd>
<dt>min_age</dt><dd><p>minimum age, in millions of years</p>
</dd>
<dt>abbr</dt><dd><p>era name abbreviations</p>
</dd>
<dt>color</dt><dd><p>the colors for each era, according to the Commission for the
Geological Map of the World</p>
</dd>
<dt>lab_color</dt><dd><p>the label colors for each era, either white or black,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://stratigraphy.org">https://stratigraphy.org</a> via <a href="https://macrostrat.org/api/v2/defs/intervals?timescale=international%20eras">https://macrostrat.org/api/v2/defs/intervals?timescale=international%20eras</a>
</p>


<h3>See Also</h3>

<p>Other built-in timescales: 
<code><a href="#topic+eons">eons</a></code>,
<code><a href="#topic+epochs">epochs</a></code>,
<code><a href="#topic+periods">periods</a></code>,
<code><a href="#topic+stages">stages</a></code>
</p>

<hr>
<h2 id='facet_grid_color'>Lay out panels in a grid with colored strips</h2><span id='topic+facet_grid_color'></span><span id='topic+FacetGridColor'></span>

<h3>Description</h3>

<p><code>facet_grid_color</code> behaves similarly to <code><a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid()</a></code> in that it
forms a matrix of panels defined by row and column faceting variables. The
main difference is that it also allows the user to specify the background and
label colors of the individual facet strips using the <code>colors</code> and
<code>lab_colors</code> arguments. If you have only one variable with many levels, try
<code><a href="#topic+facet_wrap_color">facet_wrap_color()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_grid_color(
  rows = NULL,
  cols = NULL,
  scales = "fixed",
  space = "fixed",
  shrink = TRUE,
  labeller = "label_value",
  colors = stages,
  lab_colors = "auto",
  as.table = TRUE,
  switch = NULL,
  drop = TRUE,
  margins = FALSE,
  axes = "margins",
  axis.labels = "all"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_grid_color_+3A_rows">rows</code>, <code id="facet_grid_color_+3A_cols">cols</code></td>
<td>
<p>A set of variables or expressions quoted by
<code><a href="ggplot2.html#topic+vars">vars()</a></code> and defining faceting groups on the rows or columns
dimension. The variables can be named (the names are passed to
<code>labeller</code>).
</p>
<p>For compatibility with the classic interface, <code>rows</code> can also be
a formula with the rows (of the tabular display) on the LHS and
the columns (of the tabular display) on the RHS; the dot in the
formula is used to indicate there should be no faceting on this
dimension (either row or column).</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_scales">scales</code></td>
<td>
<p>Are scales shared across all facets (the default,
<code>"fixed"</code>), or do they vary across rows (<code>"free_x"</code>),
columns (<code>"free_y"</code>), or both rows and columns (<code>"free"</code>)?</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_space">space</code></td>
<td>
<p>If <code>"fixed"</code>, the default, all panels have the same size.
If <code>"free_y"</code> their height will be proportional to the length of the
y scale; if <code>"free_x"</code> their width will be proportional to the
length of the x scale; or if <code>"free"</code> both height and width will
vary.  This setting has no effect unless the appropriate scales also vary.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_shrink">shrink</code></td>
<td>
<p>If <code>TRUE</code>, will shrink scales to fit output of
statistics, not raw data. If <code>FALSE</code>, will be range of raw data
before statistical summary.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_labeller">labeller</code></td>
<td>
<p>A function that takes one data frame of labels and
returns a list or data frame of character vectors. Each input
column corresponds to one factor. Thus there will be more than
one with <code>vars(cyl, am)</code>. Each output
column gets displayed as one separate line in the strip
label. This function should inherit from the &quot;labeller&quot; S3 class
for compatibility with <code><a href="ggplot2.html#topic+labeller">labeller()</a></code>. You can use different labeling
functions for different kind of labels, for example use <code><a href="ggplot2.html#topic+label_parsed">label_parsed()</a></code> for
formatting facet labels. <code><a href="ggplot2.html#topic+label_value">label_value()</a></code> is used by default,
check it for more details and pointers to other options.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_colors">colors</code></td>
<td>
<p>Specifies which colors to use to replace the strip backgrounds.
Either A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, or D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;color&quot; (indicating desired colors). If
the function returns <code>NA</code>, the default background color will be used.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_lab_colors">lab_colors</code></td>
<td>
<p>Specifies which colors to use for the strip labels. Either
A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;lab_color&quot; (indicating desired colors),
or E) &quot;auto&quot; (the default), which set the labels to black or white,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a>.
If the function returns <code>NA</code>, the default label color will be used.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_as.table">as.table</code></td>
<td>
<p>If <code>TRUE</code>, the default, the facets are laid out like
a table with highest values at the bottom-right. If <code>FALSE</code>, the
facets are laid out like a plot with the highest value at the top-right.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_switch">switch</code></td>
<td>
<p>By default, the labels are displayed on the top and
right of the plot. If <code>"x"</code>, the top labels will be
displayed to the bottom. If <code>"y"</code>, the right-hand side
labels will be displayed to the left. Can also be set to
<code>"both"</code>.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_drop">drop</code></td>
<td>
<p>If <code>TRUE</code>, the default, all factor levels not used in the
data will automatically be dropped. If <code>FALSE</code>, all factor levels
will be shown, regardless of whether or not they appear in the data.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_margins">margins</code></td>
<td>
<p>Either a logical value or a character
vector. Margins are additional facets which contain all the data
for each of the possible values of the faceting variables. If
<code>FALSE</code>, no additional facets are included (the
default). If <code>TRUE</code>, margins are included for all faceting
variables. If specified as a character vector, it is the names of
variables for which margins are to be created.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_axes">axes</code></td>
<td>
<p>Determines which axes will be drawn. When <code>"margins"</code>
(default), axes will be drawn at the exterior margins. <code>"all_x"</code> and
<code>"all_y"</code> will draw the respective axes at the interior panels too, whereas
<code>"all"</code> will draw all axes at all panels.</p>
</td></tr>
<tr><td><code id="facet_grid_color_+3A_axis.labels">axis.labels</code></td>
<td>
<p>Determines whether to draw labels for interior axes when
the <code>axes</code> argument is not <code>"margins"</code>. When <code>"all"</code> (default), all
interior axes get labels. When <code>"margins"</code>, only the exterior axes get
labels and the interior axes get none. When <code>"all_x"</code> or <code>"all_y"</code>, only
draws the labels at the interior axes in the x- or y-direction
respectively.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other faceting functions: 
<code><a href="#topic+facet_nested_color">facet_nested_color</a>()</code>,
<code><a href="#topic+facet_nested_wrap_color">facet_nested_wrap_color</a>()</code>,
<code><a href="#topic+facet_wrap_color">facet_wrap_color</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(x = 1:10, y = 1:10, period = c("Permian", "Triassic"))
ggplot(df) +
  geom_point(aes(x, y)) +
  facet_grid_color(cols = vars(period), colors = periods)
</code></pre>

<hr>
<h2 id='facet_nested_color'>Layout panels in a grid with nested colored strips</h2><span id='topic+facet_nested_color'></span><span id='topic+FacetNestedColor'></span>

<h3>Description</h3>

<p><code>facet_nested_color</code> behaves similarly to <code><a href="ggh4x.html#topic+facet_nested">ggh4x::facet_nested()</a></code> in that it
forms a matrix of panels defined by row and column faceting variables and
nests grouped facets. The main difference is that it also allows the user to
specify the background and label colors of the individual facet strips using
the <code>colors</code> and <code>lab_colors</code> arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_nested_color(
  rows = NULL,
  cols = NULL,
  scales = "fixed",
  space = "fixed",
  axes = "margins",
  remove_labels = "none",
  independent = "none",
  shrink = TRUE,
  labeller = "label_value",
  colors = stages,
  lab_colors = "auto",
  as.table = TRUE,
  switch = NULL,
  drop = TRUE,
  margins = FALSE,
  nest_line = element_line(inherit.blank = TRUE),
  solo_line = FALSE,
  resect = unit(0, "mm"),
  render_empty = TRUE,
  strip = strip_nested(),
  bleed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_nested_color_+3A_rows">rows</code>, <code id="facet_nested_color_+3A_cols">cols</code></td>
<td>
<p>A set of variables or expressions quoted by
<code><a href="ggplot2.html#topic+vars">vars()</a></code> and defining faceting groups on the rows or columns
dimension. The variables can be named (the names are passed to
<code>labeller</code>).
</p>
<p>For compatibility with the classic interface, <code>rows</code> can also be
a formula with the rows (of the tabular display) on the LHS and
the columns (of the tabular display) on the RHS; the dot in the
formula is used to indicate there should be no faceting on this
dimension (either row or column).</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_scales">scales</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> whether scales are shared
across facets or allowed to vary. Interacts with the <code>independent</code>
argument. One of the following:
</p>

<dl>
<dt><code>"fixed"</code> or <code>FALSE</code></dt><dd><p>Scales are shared across all facets
(default).</p>
</dd>
<dt><code>"free_x"</code></dt><dd><p>x-scales are allowed to vary across rows.</p>
</dd>
<dt><code>"free_y"</code></dt><dd><p>y-scales are allowed to vary across columns.</p>
</dd>
<dt><code>"free"</code> or <code>TRUE</code></dt><dd><p>Scales can vary across rows and columns.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_space">space</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> determining whether the size of
panels are proportional to the length of the scales. When the <code>independent</code>
argument allows for free scales in a dimension, the panel sizes cannot be
proportional. Note that the <code>scales</code> argument must be free in the same
dimension as the <code>space</code> argument to have an effect.One of the following:
</p>

<dl>
<dt><code>"fixed"</code> or <code>FALSE</code></dt><dd><p>All panels have the same size (default).</p>
</dd>
<dt><code>"free_x"</code></dt><dd><p>Panel widths are proportional to the x-scales.</p>
</dd>
<dt><code>"free_y"</code></dt><dd><p>Panel heights are proportional to the y-scales.</p>
</dd>
<dt><code>"free"</code> or <code>TRUE</code></dt><dd><p>Both the widths and heights vary according to
scales.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_axes">axes</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> where axes should be drawn. One
of the following:
</p>

<dl>
<dt><code>"margins"</code> or <code>FALSE</code></dt><dd><p>Only draw axes at the outer margins
(default).</p>
</dd>
<dt><code>"x"</code></dt><dd><p>Draw axes at the outer margins and all inner x-axes too.</p>
</dd>
<dt><code>"y"</code></dt><dd><p>Draw axes at the outer margins and all inner y-axes too.</p>
</dd>
<dt><code>"all"</code> or <code>TRUE</code></dt><dd><p>Draw the axes for every panel.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_remove_labels">remove_labels</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> determining whether
axis text is displayed at inner panels. One of the following:
</p>

<dl>
<dt><code>"none"</code> or <code>FALSE</code></dt><dd><p>Display axis text at all axes (default).</p>
</dd>
<dt><code>"x"</code></dt><dd><p>Display axis text at outer margins and all inner y-axes.</p>
</dd>
<dt><code>"y"</code></dt><dd><p>Display axis text at outer margins and all inner x-axes.</p>
</dd>
<dt><code>"all"</code> or <code>TRUE</code></dt><dd><p>Only display axis text at the outer margins.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_independent">independent</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> determining whether
scales can vary within a row or column of panels, like they can be in
<a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap</a>. The <code>scales</code> argument must be free for the same
dimension before they can be set to independent. One of the following:
</p>

<dl>
<dt><code>"none"</code> or <code>FALSE</code></dt><dd><p>All y-scales should be fixed in a row and all
x-scales are fixed in a column (default).</p>
</dd>
<dt><code>"x"</code></dt><dd><p>x-scales are allowed to vary within a column.</p>
</dd>
<dt><code>"y"</code></dt><dd><p>y-scales are allowed to vary within a row.</p>
</dd>
<dt><code>"all"</code> or <code>TRUE</code></dt><dd><p>Both x- and y-scales are allowed to vary within
a column or row respectively.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_shrink">shrink</code></td>
<td>
<p>If <code>TRUE</code>, will shrink scales to fit output of
statistics, not raw data. If <code>FALSE</code>, will be range of raw data
before statistical summary.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_labeller">labeller</code></td>
<td>
<p>A function that takes one data frame of labels and
returns a list or data frame of character vectors. Each input
column corresponds to one factor. Thus there will be more than
one with <code>vars(cyl, am)</code>. Each output
column gets displayed as one separate line in the strip
label. This function should inherit from the &quot;labeller&quot; S3 class
for compatibility with <code><a href="ggplot2.html#topic+labeller">labeller()</a></code>. You can use different labeling
functions for different kind of labels, for example use <code><a href="ggplot2.html#topic+label_parsed">label_parsed()</a></code> for
formatting facet labels. <code><a href="ggplot2.html#topic+label_value">label_value()</a></code> is used by default,
check it for more details and pointers to other options.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_colors">colors</code></td>
<td>
<p>Specifies which colors to use to replace the strip backgrounds.
Either A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, or D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;color&quot; (indicating desired colors). If
the function returns <code>NA</code>, the default background color will be used.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_lab_colors">lab_colors</code></td>
<td>
<p>Specifies which colors to use for the strip labels. Either
A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;lab_color&quot; (indicating desired colors),
or E) &quot;auto&quot; (the default), which set the labels to black or white,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a>.
If the function returns <code>NA</code>, the default label color will be used.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_as.table">as.table</code></td>
<td>
<p>If <code>TRUE</code>, the default, the facets are laid out like
a table with highest values at the bottom-right. If <code>FALSE</code>, the
facets are laid out like a plot with the highest value at the top-right.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_switch">switch</code></td>
<td>
<p>By default, the labels are displayed on the top and
right of the plot. If <code>"x"</code>, the top labels will be
displayed to the bottom. If <code>"y"</code>, the right-hand side
labels will be displayed to the left. Can also be set to
<code>"both"</code>.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_drop">drop</code></td>
<td>
<p>If <code>TRUE</code>, the default, all factor levels not used in the
data will automatically be dropped. If <code>FALSE</code>, all factor levels
will be shown, regardless of whether or not they appear in the data.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_margins">margins</code></td>
<td>
<p>Either a logical value or a character
vector. Margins are additional facets which contain all the data
for each of the possible values of the faceting variables. If
<code>FALSE</code>, no additional facets are included (the
default). If <code>TRUE</code>, margins are included for all faceting
variables. If specified as a character vector, it is the names of
variables for which margins are to be created.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_nest_line">nest_line</code></td>
<td>
<p>a theme element, either <code>element_blank()</code> or inheriting
from <code><a href="ggplot2.html#topic+element">ggplot2::element_line()</a></code>. Lines are drawn between layers of strips
indicating hierarchy. The element inherits from the
<code><a href="ggh4x.html#topic+theme_extensions">ggh4x.facet.nestline</a></code> element in the theme.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_solo_line">solo_line</code></td>
<td>
<p>A <code>logical(1)</code> indicating whether parent strips with a
single child should be drawn with a <code>nest_line</code> (<code>TRUE</code>) or the line only
applies to parents with multiple children (<code>FALSE</code>, default). Only relevant
when <code>nest_line</code> is drawn.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_resect">resect</code></td>
<td>
<p>a <code>unit</code> vector of length 1, indicating how much the
nesting line should be shortened.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_render_empty">render_empty</code></td>
<td>
<p>A <code>logical(1)</code>: whether to draw panels without any data
(<code>TRUE</code>, default) or display these as blanks (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_strip">strip</code></td>
<td>
<p>An object created by a call to a strip function, such as
<code><a href="ggh4x.html#topic+strip_nested">strip_nested()</a></code>.</p>
</td></tr>
<tr><td><code id="facet_nested_color_+3A_bleed">bleed</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> the <code>bleed</code> argument has
moved to the <code>strip_nested()</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function inherits the capabilities of
<code><a href="ggh4x.html#topic+facet_grid2">facet_grid2()</a></code>.
</p>
<p>Unlike <code>facet_grid()</code>, this function only automatically expands
missing variables when they have no variables in that direction, to allow
for unnested variables. It still requires at least one layer to have all
faceting variables.
</p>
<p>Hierarchies are inferred from the order of variables supplied to
<code>rows</code> or <code>cols</code>. The first variable is interpreted to be the
outermost variable, while the last variable is interpreted to be the
innermost variable. They display order is always such that the outermost
variable is placed the furthest away from the panels. For more information
about the nesting of strips, please visit the documentation of
<code><a href="ggh4x.html#topic+strip_nested">strip_nested()</a></code>.
</p>


<h3>See Also</h3>

<p>Other faceting functions: 
<code><a href="#topic+facet_grid_color">facet_grid_color</a>()</code>,
<code><a href="#topic+facet_nested_wrap_color">facet_nested_wrap_color</a>()</code>,
<code><a href="#topic+facet_wrap_color">facet_wrap_color</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(x = 1:10, y = 1:10,
                 period = factor(c("Permian", "Triassic", "Jurassic",
                                   "Cretaceous", "Paleogene"),
                                 levels = c("Permian", "Triassic",
                                            "Jurassic", "Cretaceous",
                                            "Paleogene")),
                 era = factor(c("Paleozoic", "Mesozoic", "Mesozoic",
                                "Mesozoic", "Cenozoic"),
                                levels = c("Paleozoic", "Mesozoic",
                                           "Cenozoic")))
ggplot(df, aes(x, y)) +
  geom_point() +
  facet_nested_color(~ era + period, colors = rbind(periods, eras))
</code></pre>

<hr>
<h2 id='facet_nested_wrap_color'>Ribbon of panels with nested colored strips</h2><span id='topic+facet_nested_wrap_color'></span><span id='topic+FacetNestedWrapColor'></span>

<h3>Description</h3>

<p><code>facet_nested_wrap_color</code> behaves similarly to <code><a href="ggh4x.html#topic+facet_nested_wrap">ggh4x::facet_nested_wrap()</a></code>
in that it wraps a sequence of panels onto a two-dimensional layout, and
nests grouped facets where possible.. The main difference is that it also
allows the user to specify the background and label colors of the individual
facet strips using the <code>colors</code> and <code>lab_colors</code> arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_nested_wrap_color(
  facets,
  nrow = NULL,
  ncol = NULL,
  scales = "fixed",
  axes = "margins",
  remove_labels = "none",
  shrink = TRUE,
  labeller = "label_value",
  colors = stages,
  lab_colors = "auto",
  as.table = TRUE,
  drop = TRUE,
  dir = "h",
  strip.position = "top",
  nest_line = element_line(inherit.blank = TRUE),
  solo_line = FALSE,
  resect = unit(0, "mm"),
  trim_blank = TRUE,
  strip = strip_nested(),
  bleed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_nested_wrap_color_+3A_facets">facets</code></td>
<td>
<p>A set of variables or expressions quoted by <code><a href="ggplot2.html#topic+vars">vars()</a></code>
and defining faceting groups on the rows or columns dimension.
The variables can be named (the names are passed to <code>labeller</code>).
</p>
<p>For compatibility with the classic interface, can also be a
formula or character vector. Use either a one sided formula, <code>~a + b</code>,
or a character vector, <code>c("a", "b")</code>.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_nrow">nrow</code>, <code id="facet_nested_wrap_color_+3A_ncol">ncol</code></td>
<td>
<p>Number of rows and columns.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_scales">scales</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> whether scales are shared
across facets or allowed to vary. One of the following:
</p>

<dl>
<dt><code>"fixed"</code> or <code>FALSE</code></dt><dd><p>Scales are shared across all facets
(default).</p>
</dd>
<dt><code>"free_x"</code></dt><dd><p>x-scales are allowed to vary.</p>
</dd>
<dt><code>"free_y"</code></dt><dd><p>y-scales are allowed to vary.</p>
</dd>
<dt><code>"free"</code> or <code>TRUE</code></dt><dd><p>Both scales can vary</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_axes">axes</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> where axes should be drawn. One
of the following:
</p>

<dl>
<dt><code>"margins"</code> or <code>FALSE</code></dt><dd><p>Only draw axes at the outer margins
(default).</p>
</dd>
<dt><code>"x"</code></dt><dd><p>Draw axes at the outer margins and all inner x-axes too.</p>
</dd>
<dt><code>"y"</code></dt><dd><p>Draw axes at the outer margins and all inner y-axes too.</p>
</dd>
<dt><code>"all"</code> or <code>TRUE</code></dt><dd><p>Draw the axes for every panel.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_remove_labels">remove_labels</code></td>
<td>
<p>A <code>character(1)</code> or <code>logical(1)</code> determining whether
axis text is displayed at inner panels. One of the following:
</p>

<dl>
<dt><code>"none"</code> or <code>FALSE</code></dt><dd><p>Display axis text at all axes (default).</p>
</dd>
<dt><code>"x"</code></dt><dd><p>Display axis text at outer margins and all inner y-axes.</p>
</dd>
<dt><code>"y"</code></dt><dd><p>Display axis text at outer margins and all inner x-axes.</p>
</dd>
<dt><code>"all"</code> or <code>TRUE</code></dt><dd><p>Only display axis text at the outer margins.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_shrink">shrink</code></td>
<td>
<p>If <code>TRUE</code>, will shrink scales to fit output of
statistics, not raw data. If <code>FALSE</code>, will be range of raw data
before statistical summary.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_labeller">labeller</code></td>
<td>
<p>A function that takes one data frame of labels and
returns a list or data frame of character vectors. Each input
column corresponds to one factor. Thus there will be more than
one with <code>vars(cyl, am)</code>. Each output
column gets displayed as one separate line in the strip
label. This function should inherit from the &quot;labeller&quot; S3 class
for compatibility with <code><a href="ggplot2.html#topic+labeller">labeller()</a></code>. You can use different labeling
functions for different kind of labels, for example use <code><a href="ggplot2.html#topic+label_parsed">label_parsed()</a></code> for
formatting facet labels. <code><a href="ggplot2.html#topic+label_value">label_value()</a></code> is used by default,
check it for more details and pointers to other options.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_colors">colors</code></td>
<td>
<p>Specifies which colors to use to replace the strip backgrounds.
Either A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, or D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;color&quot; (indicating desired colors). If
the function returns <code>NA</code>, the default background color will be used.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_lab_colors">lab_colors</code></td>
<td>
<p>Specifies which colors to use for the strip labels. Either
A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;lab_color&quot; (indicating desired colors),
or E) &quot;auto&quot; (the default), which set the labels to black or white,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a>.
If the function returns <code>NA</code>, the default label color will be used.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_as.table">as.table</code></td>
<td>
<p>If <code>TRUE</code>, the default, the facets are laid out like
a table with highest values at the bottom-right. If <code>FALSE</code>, the
facets are laid out like a plot with the highest value at the top-right.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_drop">drop</code></td>
<td>
<p>If <code>TRUE</code>, the default, all factor levels not used in the
data will automatically be dropped. If <code>FALSE</code>, all factor levels
will be shown, regardless of whether or not they appear in the data.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_dir">dir</code></td>
<td>
<p>Direction: either <code>"h"</code> for horizontal, the default, or <code>"v"</code>,
for vertical.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_strip.position">strip.position</code></td>
<td>
<p>By default, the labels are displayed on the top of
the plot. Using <code>strip.position</code> it is possible to place the labels on
either of the four sides by setting <code>strip.position = c("top",
  "bottom", "left", "right")</code></p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_nest_line">nest_line</code></td>
<td>
<p>a theme element, either <code>element_blank()</code> or inheriting
from <code><a href="ggplot2.html#topic+element">ggplot2::element_line()</a></code>. Lines are drawn between layers of strips
indicating hierarchy. The element inherits from the
<code><a href="ggh4x.html#topic+theme_extensions">ggh4x.facet.nestline</a></code> element in the theme.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_solo_line">solo_line</code></td>
<td>
<p>A <code>logical(1)</code> indicating whether parent strips with a
single child should be drawn with a <code>nest_line</code> (<code>TRUE</code>) or the line only
applies to parents with multiple children (<code>FALSE</code>, default). Only relevant
when <code>nest_line</code> is drawn.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_resect">resect</code></td>
<td>
<p>a <code>unit</code> vector of length 1, indicating how much the
nesting line should be shortened.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_trim_blank">trim_blank</code></td>
<td>
<p>A <code>logical(1)</code>. When <code>TRUE</code> (default), does not draw rows
and columns containing no panels. When <code>FALSE</code>, the <code>nrow</code> and <code>ncol</code>
arguments are taken literally, even when there are more than needed to
fit all panels.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_strip">strip</code></td>
<td>
<p>An object created by a call to a strip function, such as
<code><a href="ggh4x.html#topic+strip_nested">strip_nested()</a></code>.</p>
</td></tr>
<tr><td><code id="facet_nested_wrap_color_+3A_bleed">bleed</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> the <code>bleed</code> argument has
moved to the <code>strip_nested()</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function inherits the capabilities of
<code><a href="ggh4x.html#topic+facet_wrap2">facet_wrap2()</a></code>.
</p>
<p>This function only merges strips in the same row or column as they appear
through regular <code>facet_wrap()</code> layout behaviour.
</p>
<p>Hierarchies are inferred from the order of variables supplied to
<code>facets</code>. The first variable is interpreted to be the outermost
variable, while the last variable is interpreted to be the innermost
variable. They display order is always such that the outermost
variable is placed the furthest away from the panels. For more information
about the nesting of strips, please visit the documentation of
<code><a href="ggh4x.html#topic+strip_nested">strip_nested()</a></code>.
</p>


<h3>See Also</h3>

<p>Other faceting functions: 
<code><a href="#topic+facet_grid_color">facet_grid_color</a>()</code>,
<code><a href="#topic+facet_nested_color">facet_nested_color</a>()</code>,
<code><a href="#topic+facet_wrap_color">facet_wrap_color</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(x = 1:10, y = 1:10,
                 period = factor(c("Permian", "Triassic", "Jurassic",
                                   "Cretaceous", "Paleogene"),
                                 levels = c("Permian", "Triassic",
                                            "Jurassic", "Cretaceous",
                                            "Paleogene")),
                 era = factor(c("Paleozoic", "Mesozoic", "Mesozoic",
                                "Mesozoic", "Cenozoic"),
                                levels = c("Paleozoic", "Mesozoic",
                                           "Cenozoic")))
ggplot(df, aes(x, y)) +
  geom_point() +
  facet_nested_wrap_color(~ era + period, colors = rbind(periods, eras))
</code></pre>

<hr>
<h2 id='facet_wrap_color'>Wrap a 1d ribbon of panels into 2d with colored strips</h2><span id='topic+facet_wrap_color'></span><span id='topic+FacetWrapColor'></span>

<h3>Description</h3>

<p><code>facet_wrap_color</code> behaves similarly to <code><a href="ggplot2.html#topic+facet_wrap">ggplot2::facet_wrap()</a></code> in that it
wraps a 1d sequence of panels into 2d. The main difference is that it also
allows the user to specify the background and label colors of the individual
facet strips using the <code>colors</code> and <code>lab_colors</code> arguments. This is generally
a better use of screen space than <code><a href="#topic+facet_grid_color">facet_grid_color()</a></code> because most displays
are roughly rectangular.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>facet_wrap_color(
  facets,
  nrow = NULL,
  ncol = NULL,
  scales = "fixed",
  shrink = TRUE,
  labeller = "label_value",
  colors = stages,
  lab_colors = "auto",
  as.table = TRUE,
  drop = TRUE,
  dir = "h",
  strip.position = "top",
  axes = "margins",
  axis.labels = "all"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="facet_wrap_color_+3A_facets">facets</code></td>
<td>
<p>A set of variables or expressions quoted by <code><a href="ggplot2.html#topic+vars">vars()</a></code>
and defining faceting groups on the rows or columns dimension.
The variables can be named (the names are passed to <code>labeller</code>).
</p>
<p>For compatibility with the classic interface, can also be a
formula or character vector. Use either a one sided formula, <code>~a + b</code>,
or a character vector, <code>c("a", "b")</code>.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_nrow">nrow</code>, <code id="facet_wrap_color_+3A_ncol">ncol</code></td>
<td>
<p>Number of rows and columns.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_scales">scales</code></td>
<td>
<p>Should scales be fixed (<code>"fixed"</code>, the default),
free (<code>"free"</code>), or free in one dimension (<code>"free_x"</code>,
<code>"free_y"</code>)?</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_shrink">shrink</code></td>
<td>
<p>If <code>TRUE</code>, will shrink scales to fit output of
statistics, not raw data. If <code>FALSE</code>, will be range of raw data
before statistical summary.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_labeller">labeller</code></td>
<td>
<p>A function that takes one data frame of labels and
returns a list or data frame of character vectors. Each input
column corresponds to one factor. Thus there will be more than
one with <code>vars(cyl, am)</code>. Each output
column gets displayed as one separate line in the strip
label. This function should inherit from the &quot;labeller&quot; S3 class
for compatibility with <code><a href="ggplot2.html#topic+labeller">labeller()</a></code>. You can use different labeling
functions for different kind of labels, for example use <code><a href="ggplot2.html#topic+label_parsed">label_parsed()</a></code> for
formatting facet labels. <code><a href="ggplot2.html#topic+label_value">label_value()</a></code> is used by default,
check it for more details and pointers to other options.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_colors">colors</code></td>
<td>
<p>Specifies which colors to use to replace the strip backgrounds.
Either A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, or D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;color&quot; (indicating desired colors). If
the function returns <code>NA</code>, the default background color will be used.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_lab_colors">lab_colors</code></td>
<td>
<p>Specifies which colors to use for the strip labels. Either
A) a function that returns a color for a given strip label, B) the
character name of a function that does the same, C) a named character
vector with names matching strip labels and values indicating the desired
colors, D) a data.frame representing a lookup table with columns named
&quot;name&quot; (matching strip labels) and &quot;lab_color&quot; (indicating desired colors),
or E) &quot;auto&quot; (the default), which set the labels to black or white,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a>.
If the function returns <code>NA</code>, the default label color will be used.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_as.table">as.table</code></td>
<td>
<p>If <code>TRUE</code>, the default, the facets are laid out like
a table with highest values at the bottom-right. If <code>FALSE</code>, the
facets are laid out like a plot with the highest value at the top-right.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_drop">drop</code></td>
<td>
<p>If <code>TRUE</code>, the default, all factor levels not used in the
data will automatically be dropped. If <code>FALSE</code>, all factor levels
will be shown, regardless of whether or not they appear in the data.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_dir">dir</code></td>
<td>
<p>Direction: either <code>"h"</code> for horizontal, the default, or <code>"v"</code>,
for vertical.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_strip.position">strip.position</code></td>
<td>
<p>By default, the labels are displayed on the top of
the plot. Using <code>strip.position</code> it is possible to place the labels on
either of the four sides by setting <code>strip.position = c("top",
  "bottom", "left", "right")</code></p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_axes">axes</code></td>
<td>
<p>Determines which axes will be drawn in case of fixed scales.
When <code>"margins"</code> (default), axes will be drawn at the exterior margins.
<code>"all_x"</code> and <code>"all_y"</code> will draw the respective axes at the interior
panels too, whereas <code>"all"</code> will draw all axes at all panels.</p>
</td></tr>
<tr><td><code id="facet_wrap_color_+3A_axis.labels">axis.labels</code></td>
<td>
<p>Determines whether to draw labels for interior axes when
the scale is fixed and the <code>axis</code> argument is not <code>"margins"</code>. When
<code>"all"</code> (default), all interior axes get labels. When <code>"margins"</code>, only
the exterior axes get labels, and the interior axes get none. When
<code>"all_x"</code> or <code>"all_y"</code>, only draws the labels at the interior axes in the
x- or y-direction respectively.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other faceting functions: 
<code><a href="#topic+facet_grid_color">facet_grid_color</a>()</code>,
<code><a href="#topic+facet_nested_color">facet_nested_color</a>()</code>,
<code><a href="#topic+facet_nested_wrap_color">facet_nested_wrap_color</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(x = 1:10, y = 1:10, period = c("Permian", "Triassic"))
ggplot(df) +
  geom_point(aes(x, y)) +
  facet_wrap_color(vars(period), colors = periods)
</code></pre>

<hr>
<h2 id='geo_pattern'>Get a FGDC geologic plotting pattern</h2><span id='topic+geo_pattern'></span><span id='topic+geo_grob'></span>

<h3>Description</h3>

<p>Retrieve a single geologic pattern as defined in the <a href="https://ngmdb.usgs.gov/fgdc_gds/geolsymstd.php">FGDC Digital Cartographic Standard for Geologic Map Symbolization</a> by the <a href="https://www.usgs.gov/">U.S. Geological Survey</a> and the <a href="https://ngmdb.usgs.gov/fgdc_gds/index.php">Geologic Data Subcommittee (GDS)</a> of the <a href="https://www.fgdc.gov/">Federal Geographic Data Committee (FGDC)</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geo_pattern(
  code,
  scale = 2,
  col = NULL,
  fill = NULL,
  alpha = NULL,
  bg = "white"
)

geo_grob(code, col = NULL, fill = NULL, alpha = NULL, bg = "white")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geo_pattern_+3A_code">code</code></td>
<td>
<p>The number corresponding to the pattern to return. Strings and
numbers are permitted.</p>
</td></tr>
<tr><td><code id="geo_pattern_+3A_scale">scale</code></td>
<td>
<p>The visual scale of the pattern (higher values mean the pattern
is more zoomed in).</p>
</td></tr>
<tr><td><code id="geo_pattern_+3A_col">col</code></td>
<td>
<p>The color to use for the lines of the pattern.</p>
</td></tr>
<tr><td><code id="geo_pattern_+3A_fill">fill</code></td>
<td>
<p>The color used to fill various closed shapes (e.g., circles) in
the pattern.</p>
</td></tr>
<tr><td><code id="geo_pattern_+3A_alpha">alpha</code></td>
<td>
<p>The transparency to use for the fill of the pattern.</p>
</td></tr>
<tr><td><code id="geo_pattern_+3A_bg">bg</code></td>
<td>
<p>The background color to use for the pattern.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For specific codes, see the &quot;pattern numbers&quot; in the <a href="https://ngmdb.usgs.gov/fgdc_gds/geolsymstd/fgdc-geolsym-patternchart.pdf">full pattern chart</a>
for valid <code>code</code> values. Daven Quinn has also assembled more accessible
documentation of the <a href="https://davenquinn.com/projects/geologic-patterns/#pattern-reference">map patterns/codes</a>
and <a href="https://davenquinn.com/projects/geologic-patterns/#series-600">lithology patterns/codes</a>.
<code><a href="rmacrostrat.html#topic+def_lithologies">rmacrostrat::def_lithologies()</a></code> can also be used to look up pattern codes
for various lithologies (see the &quot;fill&quot; column). Note that codes associated
with color variants (e.g., &quot;101-M&quot;) are supported but will result in the
default color variant instead (usually black and white, e.g., &quot;101-K&quot;).
</p>
<p>These patterns were originally processed and optimized by Daven Quinn and
are hosted on <a href="https://github.com/davenquinn/geologic-patterns/">GitHub</a>.
</p>


<h3>Value</h3>

<p><code>geo_grob()</code> returns a <a href="grid.html#topic+gTree">grob</a> object with a single
instance of the desired pattern. <code>geo_pattern()</code> returns a
<a href="grid.html#topic+patterns">GridPattern</a> object with a repeated instance of the
desired pattern.
</p>


<h3>See Also</h3>

<p>FGDC patterns:
<code><a href="#topic+grid.pattern_geo">grid.pattern_geo</a>()</code>,
<code><a href="#topic+scale_fill_geopattern">scale_fill_geopattern</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
# Get a generic igneous pattern
pattern1 &lt;- geo_pattern(code = "313-K")
# Get the pattern for a sandstone
pattern2 &lt;- geo_pattern(code = "607")

# plot the two patterns
grid.newpage()
grid.draw(rectGrob(gp = gpar(fill = pattern1)))
grid.newpage()
grid.draw(rectGrob(gp = gpar(fill = pattern2)))
</code></pre>

<hr>
<h2 id='geom_phylomorpho'>Plot a 2-D phylomorphospace in ggplot2</h2><span id='topic+geom_phylomorpho'></span>

<h3>Description</h3>

<p>This behaves similar to <code><a href="phytools.html#topic+phylomorphospace">phytools::phylomorphospace()</a></code>, but is for plotting a
2-D phylomorphospace with <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code>. This function works like any
other <code>ggplot2</code> geom; it can be combined with other geoms (see the example
below), and the output can be modified using scales, themes, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_phylomorpho(
  tree,
  mapping = NULL,
  data = NULL,
  position = "identity",
  ...,
  seg_args = list(),
  point_args = list(),
  arrow = NULL,
  arrow.fill = NULL,
  lineend = "butt",
  linejoin = "round",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_phylomorpho_+3A_tree">tree</code></td>
<td>
<p>An object of class &quot;phylo&quot;.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example, to
use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to both <code><a href="ggplot2.html#topic+geom_segment">ggplot2::geom_segment()</a></code> and
<code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_seg_args">seg_args</code></td>
<td>
<p>A list of arguments passed only to <code><a href="ggplot2.html#topic+geom_segment">ggplot2::geom_segment()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_point_args">point_args</code></td>
<td>
<p>A list of arguments passed only to <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_arrow">arrow</code></td>
<td>
<p>specification for arrow heads, as created by <code><a href="grid.html#topic+arrow">grid::arrow()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_arrow.fill">arrow.fill</code></td>
<td>
<p>fill colour to use for the arrow head (if closed). <code>NULL</code>
means use <code>colour</code> aesthetic.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_lineend">lineend</code></td>
<td>
<p>Line end style (round, butt, square).</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_linejoin">linejoin</code></td>
<td>
<p>Line join style (round, mitre, bevel).</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_phylomorpho_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ancestral states are estimated using <code><a href="phytools.html#topic+fastAnc">phytools::fastAnc()</a></code>. Note that
<code>phytools</code> is not necessarily installed with <code>deeptime</code>, but it is required
to use this function. Following the estimation of the ancestral states, the
nodes are connected using <code><a href="ggplot2.html#topic+geom_segment">ggplot2::geom_segment()</a></code>, while the tips are
indicated using <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>.
</p>
<p>The default expectation is that the order of the data is the same order as
the tip labels of the tree (<code>tree$tip.label</code>). However, if this is not the
case, you can map the optional <code>label</code> aesthetic to a column in the data that
contains the tip names (see example below).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

library(ape)
tr &lt;- rtree(10)
dat &lt;- data.frame(
  x = runif(10), y = runif(10), label = tr$tip.label,
  row.names = tr$tip.label
)
ggplot(dat, aes(x = x, y = y, label = label)) +
  geom_phylomorpho(tr) +
  geom_label(size = 5)

</code></pre>

<hr>
<h2 id='geom_points_range'>Display points and their range</h2><span id='topic+geom_points_range'></span><span id='topic+stat_points_range'></span>

<h3>Description</h3>

<p>This geom is like <code><a href="ggplot2.html#topic+geom_linerange">ggplot2::geom_pointrange()</a></code> in that it draws points and
lines. However, unlike <code><a href="ggplot2.html#topic+geom_linerange">ggplot2::geom_pointrange()</a></code>, this geom takes in sets
of x-y points and calculates the ranges/intervals based on those. It then
plots both the original points and the ranges using
<code><a href="ggplot2.html#topic+geom_linerange">ggplot2::geom_linerange()</a></code>. In cases where not all points are connected
(because of grouping due to aesthetics), the <code>background_line</code> argument can
be used to add lines that span the entire point range for each <code>x</code> or <code>y</code>
category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_points_range(
  mapping = NULL,
  data = NULL,
  stat = "points_range",
  position = "identity",
  ...,
  na.rm = FALSE,
  orientation = NA,
  background_line = NULL,
  show.legend = NA,
  inherit.aes = TRUE
)

stat_points_range(
  mapping = NULL,
  data = NULL,
  geom = "points_range",
  position = "identity",
  ...,
  na.rm = FALSE,
  orientation = NA,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geom_points_range_+3A_mapping">mapping</code></td>
<td>
<p>Set of aesthetic mappings created by <code><a href="ggplot2.html#topic+aes">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_data">data</code></td>
<td>
<p>The data to be displayed in this layer. There are three
options:
</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="ggplot2.html#topic+ggplot">ggplot()</a></code>.
</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="ggplot2.html#topic+fortify">fortify()</a></code> for which variables will be created.
</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_stat">stat</code></td>
<td>
<p>The statistical transformation to use on the data for this layer.
When using a <code style="white-space: pre;">&#8288;geom_*()&#8288;</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.
</p>
</li>
<li><p> A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code>stat_count()</code>,
give the stat as <code>"count"</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_points_range_+3A_position">position</code></td>
<td>
<p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:
</p>

<ul>
<li><p> The result of calling a position function, such as <code>position_jitter()</code>.
This method allows for passing extra arguments to the position.
</p>
</li>
<li><p> A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code>position_jitter()</code>, give the position as <code>"jitter"</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="geom_points_range_+3A_...">...</code></td>
<td>
<p>Arguments passed on to both <code><a href="ggplot2.html#topic+geom_linerange">ggplot2::geom_linerange()</a></code> and
<code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_na.rm">na.rm</code></td>
<td>
<p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_orientation">orientation</code></td>
<td>
<p>The orientation of the layer. The default (<code>NA</code>)
automatically determines the orientation from the aesthetic mapping. In the
rare event that this fails it can be given explicitly by setting <code>orientation</code>
to either <code>"x"</code> or <code>"y"</code>. See the <em>Orientation</em> section for more detail.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_background_line">background_line</code></td>
<td>
<p>A named list of aesthetic values to use for plotted
line segments that span the entire <code>y</code> or <code>x</code> range for each <code>x</code> or <code>y</code>
category. The default aesthetics will be used for any aesthetics that are
not specified in the list. This can be useful if the plotted groups of
points don't overlap but you want a continuous line connecting all points
for a given <code>x</code> or <code>y</code> category. If NULL (the default), no line segments
will be plotted.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_show.legend">show.legend</code></td>
<td>
<p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_inherit.aes">inherit.aes</code></td>
<td>
<p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="ggplot2.html#topic+borders">borders()</a></code>.</p>
</td></tr>
<tr><td><code id="geom_points_range_+3A_geom">geom</code></td>
<td>
<p>The geometric object to use to display the data for this layer.
When using a <code style="white-space: pre;">&#8288;stat_*()&#8288;</code> function to construct a layer, the <code>geom</code> argument
can be used to override the default coupling between stats and geoms. The
<code>geom</code> argument accepts the following:
</p>

<ul>
<li><p> A <code>Geom</code> ggproto subclass, for example <code>GeomPoint</code>.
</p>
</li>
<li><p> A string naming the geom. To give the geom as a string, strip the
function name of the <code>geom_</code> prefix. For example, to use <code>geom_point()</code>,
give the geom as <code>"point"</code>.
</p>
</li></ul>
</td></tr>
</table>


<h3>Aesthetics</h3>

<p><code>geom_points_range()</code> understands the following
aesthetics (required aesthetics are in bold):
</p>

<ul>
<li> <p><strong>x</strong>
</p>
</li>
<li> <p><strong>y</strong>
</p>
</li>
<li><p> size
</p>
</li>
<li><p> color/colour
</p>
</li>
<li><p> fill
</p>
</li>
<li><p> shape
</p>
</li>
<li><p> alpha
</p>
</li>
<li><p> group
</p>
</li>
<li><p> linetype
</p>
</li>
<li><p> linewidth
</p>
</li></ul>



<h3>Computed variables</h3>

<p>These are calculated by the 'stat' part of
layers and can be accessed with <a href="ggplot2.html#topic+aes_eval">delayed evaluation</a>.
<code>stat_points_range()</code> provides the following variables, some of which
depend on the orientation:
</p>

<ul>
<li> <p><code>after_stat(ymin)</code> <em>or</em> <code>after_stat(xmin)</code><br />
the minimum extent of the point range
</p>
</li>
<li> <p><code>after_stat(ymax)</code> <em>or</em> <code>after_stat(xmax)</code><br />
the maximum extent of the point range
</p>
</li></ul>



<h3>Orientation</h3>

<p>This geom treats each axis differently and, thus, can thus have two orientations. Often the orientation is easy to deduce from a combination of the given mappings and the types of positional scales in use. Thus, ggplot2 will by default try to guess which orientation the layer should have. Under rare circumstances, the orientation is ambiguous and guessing may fail. In that case the orientation can be specified directly using the <code>orientation</code> parameter, which can be either <code>"x"</code> or <code>"y"</code>. The value gives the axis that the geom should run along, <code>"x"</code> being the default orientation you would expect for the geom.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

library(palaeoverse)
data(tetrapods)
tetrapod_names &lt;- tetrapods$accepted_name[1:50]
beds_sampled &lt;- sample.int(n = 10, size = 50, replace = TRUE)
occdf &lt;- data.frame(taxon = tetrapod_names, bed = beds_sampled)
ggplot(occdf, aes(y = reorder(taxon, bed, min), x = bed)) +
  geom_points_range()

</code></pre>

<hr>
<h2 id='get_scale_data'>Get geological timescale data</h2><span id='topic+get_scale_data'></span>

<h3>Description</h3>

<p>This function takes a name of a geological timescale and returns data for the
timescale. Valid names include those of built-in <code>data.frames</code> (<code><a href="#topic+periods">periods()</a></code>,
<code><a href="#topic+epochs">epochs()</a></code>, <code><a href="#topic+stages">stages()</a></code>, <code><a href="#topic+eons">eons()</a></code>, or <code><a href="#topic+eras">eras()</a></code>), partial matches of those
names (e.g., &quot;per&quot; or &quot;age&quot;), and partial or exact matches to those hosted
by Macrostrat (see Details below). Note that the colors in the built-in
<code>data.frames</code> are according to the Commission for the Geological Map of the
World. If you would like to obtain custom Macrostrat colors that are better
for mapping, you should specify the full name of a timescale (e.g.,
&quot;international periods&quot;) and set <code>true_colors</code> to <code>FALSE</code>. Note that these
colors only vary for the Precambrian.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_scale_data(name, true_colors = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_scale_data_+3A_name">name</code></td>
<td>
<p>The name of the desired timescale.</p>
</td></tr>
<tr><td><code id="get_scale_data_+3A_true_colors">true_colors</code></td>
<td>
<p>Return original international time scale colors? (as
opposed to custom Macrostrat plotting colors)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following timescales are available from the Macrostrat API as of
2024-10-25:
</p>

<ul>
<li><p> international ages
</p>
</li>
<li><p> international epochs
</p>
</li>
<li><p> international periods
</p>
</li>
<li><p> calcareous nannoplankton zones
</p>
</li>
<li><p> New Zealand ages
</p>
</li>
<li><p> custom COSUNA
</p>
</li>
<li><p> North American land mammal ages
</p>
</li>
<li><p> international intervals
</p>
</li>
<li><p> COSUNA
</p>
</li>
<li><p> international eras
</p>
</li>
<li><p> international eons
</p>
</li>
<li><p> Trilobite Zonation - Laurentia
</p>
</li>
<li><p> Conodont Zonation
</p>
</li>
<li><p> North American Regional
</p>
</li>
<li><p> Ammonite Zonation - Boreal
</p>
</li>
<li><p> Ammonite Zonation - Western Interior
</p>
</li>
<li><p> international intervals covering all time
</p>
</li>
<li><p> Scotese Reconstruction
</p>
</li>
<li><p> Geomagnetic Polarity Chron
</p>
</li>
<li><p> Geomagnetic Polarity Subchron
</p>
</li>
<li><p> Planktic foraminiferal Primary Biozones
</p>
</li>
<li><p> Planktic foraminiferal Secondary Biozones
</p>
</li>
<li><p> Planktic foraminiferal datums
</p>
</li>
<li><p> Martian Periods
</p>
</li>
<li><p> Martian Epochs
</p>
</li>
<li><p> Cretaceous Planktic foraminifer zonations
</p>
</li>
<li><p> Low latitude radiolarian zonation
</p>
</li>
<li><p> Neogene North Pacific Diatom Biochronology
</p>
</li>
<li><p> Neogene North Pacific Diatom Biochronology Subzones
</p>
</li>
<li><p> Siberian Regional
</p>
</li>
<li><p> Australian Regional
</p>
</li>
<li><p> Western Europe Regional
</p>
</li>
<li><p> Russian Platform Regional Stages
</p>
</li>
<li><p> Russian Precambrian Eras
</p>
</li>
<li><p> Russian Precambrian Eons
</p>
</li>
<li><p> Russian Epochs
</p>
</li>
<li><p> Russian Stages
</p>
</li></ul>

<p>The most up-to-date list can be found via the Macrostrat API <a href="https://macrostrat.org/api/defs/timescales?all">here</a>.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with the following columns:
</p>
<table role = "presentation">
<tr><td><code>name</code></td>
<td>
<p>the names of the time intervals</p>
</td></tr>
<tr><td><code>max_age</code></td>
<td>
<p>the oldest boundaries of the time intervals, in millions of
years</p>
</td></tr>
<tr><td><code>min_age</code></td>
<td>
<p>the youngest boundaries of the time intervals, in millions
of years</p>
</td></tr>
<tr><td><code>abbr</code></td>
<td>
<p>either traditional abbreviations of the names of the time
intervals (if they exist) or custom abbreviations created with R</p>
</td></tr>
<tr><td><code>color</code></td>
<td>
<p>hex color codes associated with the time intervals (if
applicable)</p>
</td></tr>
<tr><td><code>lab_color</code></td>
<td>
<p>default label colors for the time interals, either white
or black, whichever has better contrast with the background color, based
on <a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</td></tr>
</table>

<hr>
<h2 id='getScaleData'>Get geological timescale data</h2><span id='topic+getScaleData'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p><code>getScaleData()</code> was renamed to <code><a href="#topic+get_scale_data">get_scale_data()</a></code> as of <strong>deeptime</strong> version
1.0.0 to create a more consistent API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getScaleData(name)
</code></pre>

<hr>
<h2 id='ggarrange2'>Combine and arrange multiple ggplot-like objects</h2><span id='topic+ggarrange2'></span>

<h3>Description</h3>

<p>Arrange multiple ggplot, grobified ggplot, or geo_scale objects on a page,
aligning the plot panels, axes, and axis titles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggarrange2(
  ...,
  plots = list(...),
  layout = NULL,
  nrow = NULL,
  ncol = NULL,
  widths = NULL,
  heights = NULL,
  byrow = TRUE,
  top = NULL,
  bottom = NULL,
  left = NULL,
  right = NULL,
  padding = unit(0.5, "line"),
  margin = unit(0.5, "line"),
  clip = "on",
  draw = TRUE,
  newpage = TRUE,
  debug = FALSE,
  labels = NULL,
  label.args = list(gp = gpar(font = 4, cex = 1.2))
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggarrange2_+3A_...">...</code></td>
<td>
<p>ggplot, grobified ggplot (gtable), or geo_scale objects</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_plots">plots</code></td>
<td>
<p>list of ggplot, gtable, or geo_scale objects</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_layout">layout</code></td>
<td>
<p>a matrix of integers specifying where each plot should go, like
<code>mat</code> in <code><a href="graphics.html#topic+layout">graphics::layout()</a></code>; <code>NA</code> or a value less than 0 or greater than
the number of plots indicates a blank plot; overrides nrow/ncol/byrow</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_nrow">nrow</code></td>
<td>
<p>number of rows</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_ncol">ncol</code></td>
<td>
<p>number of columns</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_widths">widths</code></td>
<td>
<p>list of requested widths</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_heights">heights</code></td>
<td>
<p>list of requested heights</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_byrow">byrow</code></td>
<td>
<p>logical, fill by rows</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_top">top</code></td>
<td>
<p>optional string, or grob</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_bottom">bottom</code></td>
<td>
<p>optional string, or grob</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_left">left</code></td>
<td>
<p>optional string, or grob</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_right">right</code></td>
<td>
<p>optional string, or grob</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_padding">padding</code></td>
<td>
<p>unit of length one, margin around annotations</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_margin">margin</code></td>
<td>
<p>vector of units of length 4: top, right, bottom, left (as in
<code><a href="gtable.html#topic+gtable_add_padding">gtable::gtable_add_padding()</a></code>)</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_clip">clip</code></td>
<td>
<p>argument of gtable</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_draw">draw</code></td>
<td>
<p>logical: draw or return a grob</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_newpage">newpage</code></td>
<td>
<p>logical: draw on a new page</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_debug">debug</code></td>
<td>
<p>logical, show layout with thin lines</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_labels">labels</code></td>
<td>
<p>character labels used for annotation of subfigures (should be
in the same order as <code>plots</code>)</p>
</td></tr>
<tr><td><code id="ggarrange2_+3A_label.args">label.args</code></td>
<td>
<p>label list of parameters for the formatting of labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>gtable of aligned plots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
p1 &lt;- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) +
  geom_point()
p2 &lt;- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) +
  geom_point() +
  facet_wrap(~cyl, ncol = 2, scales = "free") +
  guides(colour = "none") +
  theme()
ggarrange2(p1, p2, widths = c(2, 1), labels = c("a", "b"))

p3 &lt;- ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 1000))) +
  scale_x_reverse() +
  coord_geo(xlim = c(1000, 0), ylim = c(0, 8)) +
  theme_classic()
ggarrange2(ggarrange2(p1, p2, widths = c(2, 1), draw = FALSE), p3, nrow = 2)
</code></pre>

<hr>
<h2 id='gggeo_scale'>Add a geologic scale below ggplots using <em>grid</em></h2><span id='topic+gggeo_scale'></span><span id='topic+gggeo_scale.gtable'></span><span id='topic+gggeo_scale.ggplot'></span><span id='topic+gggeo_scale.geo_scale'></span><span id='topic+print.geo_scale'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p>This function takes a ggplot object and adds a geologic time scale at the
specified side using the grid and gtable packages.
</p>
<p>If custom data is provided (with <code>dat</code>), it should consist of at least 3
columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>name</code> column lists the names of each time interval. These will
be used as labels if no abbreviations are provided.
</p>
</li>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is also optional and lists a hex color code
(which can be obtained with <code>rgb()</code>) for each time interval.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>gggeo_scale(obj, ...)

## S3 method for class 'gtable'
gggeo_scale(
  obj,
  lims,
  dat = "periods",
  fill = NULL,
  color = "black",
  alpha = 1,
  height = unit(2, "line"),
  pos = "bottom",
  lab = TRUE,
  rot = 0,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  size = 5,
  lwd = 0.25,
  margin = unit(0, "line"),
  neg = FALSE,
  bord = c("left", "right", "top", "bottom"),
  center_end_labels = FALSE,
  ...
)

## S3 method for class 'ggplot'
gggeo_scale(
  obj,
  dat = "periods",
  fill = NULL,
  color = "black",
  alpha = 1,
  height = unit(2, "line"),
  pos = "bottom",
  lab = TRUE,
  rot = 0,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  size = 5,
  lwd = 0.25,
  margin = unit(0, "line"),
  neg = FALSE,
  bord = c("left", "right", "top", "bottom"),
  center_end_labels = FALSE,
  ...
)

## S3 method for class 'geo_scale'
gggeo_scale(
  obj,
  dat = "periods",
  fill = NULL,
  color = "black",
  alpha = 1,
  height = unit(2, "line"),
  pos = "bottom",
  lab = TRUE,
  rot = 0,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  size = 5,
  lwd = 0.25,
  margin = unit(0, "line"),
  neg = FALSE,
  bord = c("left", "right", "top", "bottom"),
  center_end_labels = FALSE,
  ...
)

## S3 method for class 'geo_scale'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gggeo_scale_+3A_obj">obj</code></td>
<td>
<p>An object of class <code>ggplot</code>, <code>gtable</code>, or <code>geo_scale</code> (as produced
by this function).</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_...">...</code></td>
<td>
<p>further arguments passed to <code><a href="grid.html#topic+grid.draw">grid::grid.draw()</a></code>.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_lims">lims</code></td>
<td>
<p>The limits of the axis of the desired side of the plot. Only
required if using a gtable object not created by this function.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;),
B) a string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>),
or C) a custom dataframe of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_fill">fill</code></td>
<td>
<p>The fill color of the boxes. The default is to use the colors
included in <code>dat</code>. If a custom dataset is provided with <code>dat</code> without color
and without fill, a greyscale will be used. Custom fill colors can be
provided with this option and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_color">color</code></td>
<td>
<p>The outline color of the interval boxes.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_height">height</code></td>
<td>
<p>The height (or width if <code>pos</code> is <code>left</code> or <code>right</code>) of the
scale.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_pos">pos</code></td>
<td>
<p>Which side to add the scale to (left, right, top, or bottom).
First letter may also be used.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_rot">rot</code></td>
<td>
<p>The amount of counter-clockwise rotation to add to the labels (in
degrees).</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, whether to use abbreviations instead of
full interval names.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_size">size</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_margin">margin</code></td>
<td>
<p>The width of the margin around the returned object (can be a
vector of length 4).</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_neg">neg</code></td>
<td>
<p>Set this to true if your x-axis is using negative values.</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_bord">bord</code></td>
<td>
<p>A vector specifying on Which sides of the scale to add borders
(same options as <code>pos</code>).</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_center_end_labels">center_end_labels</code></td>
<td>
<p>Should labels be centered within the visible range
of intervals at the ends of the axis?</p>
</td></tr>
<tr><td><code id="gggeo_scale_+3A_x">x</code></td>
<td>
<p>An object of class geo_scale.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A geo_scale object. Basically a gtable object but with the axis
limits included.
</p>


<h3>Life cycle</h3>

<p>This function is deprecated in favor of <code><a href="#topic+coord_geo">coord_geo()</a></code> as of <strong>deeptime</strong>
version 1.0.0. It will be removed in a future release.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
# bottom scale by default
p &lt;- ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 1000))) +
  scale_x_reverse() +
  coord_cartesian(xlim = c(1000, 0), ylim = c(0, 8), expand = FALSE) +
  theme_classic()
gggeo_scale(p)

# can specify any side of the plot
p &lt;- ggplot() +
  geom_point(aes(x = runif(1000, 0, 8), y = runif(1000, 0, 1000))) +
  scale_y_reverse() +
  coord_cartesian(xlim = c(0, 8), ylim = c(1000, 0), expand = FALSE) +
  theme_classic()
gggeo_scale(p, pos = "left", rot = 90)

# can add multiple scales
p &lt;- ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 100))) +
  scale_x_reverse() +
  coord_cartesian(xlim = c(100, 0), ylim = c(0, 8), expand = FALSE) +
  theme_classic()
p &lt;- gggeo_scale(p, abbrv = FALSE)
p &lt;- gggeo_scale(p, dat = "epochs", height = unit(4, "lines"), rot = 90,
                 size = 2.5, abbrv = FALSE)
gggeo_scale(p, dat = "stages", height = unit(4, "lines"), rot = 90,
            size = 2.5, abbrv = FALSE)
## Not run: 
# intervals on both sides for different timescales (ICS stages vs North
# American Land Mammal Ages)
p &lt;- ggplot() +
  geom_point(aes(x = runif(1000, 0, 10), y = runif(1000, 0, 65))) +
  scale_y_reverse() +
  coord_cartesian(xlim = c(0, 10), ylim = c(65, 0), expand = FALSE) +
  theme_classic()
p &lt;- gggeo_scale(p, dat = "stages", pos = "left", height = unit(4, "lines"),
  size = 2.5, abbrv = FALSE)
gggeo_scale(p, dat = "North American Land Mammal Ages",
            pos = "right", height = unit(4, "lines"),
            size = 2.5, abbrv = FALSE)

## End(Not run)

# can add scales to a faceted plot
# use gggeo_scale_old() if you have more than one column
df &lt;- data.frame(
  x = runif(1000, 0, 541), y = runif(1000, 0, 8),
  z = sample(c(1, 2, 3, 4), 1000, TRUE)
)
p &lt;- ggplot(df) +
  geom_point(aes(x, y)) +
  scale_x_reverse() +
  coord_cartesian(xlim = c(541, 0), ylim = c(0, 8), expand = FALSE) +
  theme_classic() +
  facet_wrap(~z, ncol = 1)
gggeo_scale(p)

# can even add a scale to a phylogeny (using ggtree)

library(phytools)
library(ggtree)
tree &lt;- pbtree(b = .03, d = .01, n = 100)
p &lt;- ggtree(tree) +
  coord_cartesian(xlim = c(-500, 0), ylim = c(-2, Ntip(tree)),
                  expand = FALSE) +
  scale_x_continuous(breaks = seq(-500, 0, 100),
                     labels = abs(seq(-500, 0, 100))) +
  theme_tree2()
p &lt;- revts(p)
gggeo_scale(p, neg = TRUE)

</code></pre>

<hr>
<h2 id='gggeo_scale_old'>Add a geologic scale on top of ggplots</h2><span id='topic+gggeo_scale_old'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
</p>
<p>This function takes a ggplot object and adds a geologic time scale at the
specified side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gggeo_scale_old(
  gg,
  dat = "periods",
  fill = NULL,
  color = "black",
  alpha = 1,
  height = 0.05,
  gap = 0,
  pos = "bottom",
  lab = TRUE,
  rot = 0,
  abbrv = TRUE,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  size = 5,
  neg = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gggeo_scale_old_+3A_gg">gg</code></td>
<td>
<p>The ggplot object.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;),
B) a string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>),
or C) a custom dataframe of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_fill">fill</code></td>
<td>
<p>The fill color of the boxes. The default is to use the colors
included in <code>dat</code>. If a custom dataset is provided with <code>dat</code> without color
and without fill, a greyscale will be used. Custom fill colors can be
provided with this option and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_color">color</code></td>
<td>
<p>The outline color of the interval boxes.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_height">height</code></td>
<td>
<p>The proportional height (or width if <code>pos</code> is <code>left</code> or
<code>right</code>) of the entire plot to use for the scale.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_gap">gap</code></td>
<td>
<p>The proportional height (or width) of the entire plot to use as a
gap between the axis and the scale.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_pos">pos</code></td>
<td>
<p>Which side to add the scale to (left, right, top, or bottom).
First letter may also be used.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_rot">rot</code></td>
<td>
<p>The amount of counter-clockwise rotation to add to the labels (in
degrees).</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, whether to use abbreviations instead of
full interval names.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_size">size</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="gggeo_scale_old_+3A_neg">neg</code></td>
<td>
<p>Set this to true if your x-axis is using negative values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If custom data is provided (with <code>dat</code>), it should consist of at least 3
columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>name</code> column lists the names of each time interval. These will
be used as labels if no abbreviations are provided.
</p>
</li>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is also optional and lists a hex color code (which
can be obtained with <code>rgb()</code>) for each time interval.
</p>
</li></ul>



<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Life cycle</h3>

<p>This function is fully deprecated in favor of <code><a href="#topic+coord_geo">coord_geo()</a></code> as of
<strong>deeptime</strong> version 1.2.0. It will be removed in a future version.
</p>

<hr>
<h2 id='grid.pattern_geo'>Plot an individual FGDC pattern using grid</h2><span id='topic+grid.pattern_geo'></span>

<h3>Description</h3>

<p>This function can be used to plot a single geologic pattern as defined in the
<a href="https://ngmdb.usgs.gov/fgdc_gds/geolsymstd.php">FGDC Digital Cartographic Standard for Geologic Map Symbolization</a> by the <a href="https://www.usgs.gov/">U.S. Geological Survey</a> and the <a href="https://ngmdb.usgs.gov/fgdc_gds/index.php">Geologic Data Subcommittee (GDS)</a> of the <a href="https://www.fgdc.gov/">Federal Geographic Data Committee (FGDC)</a>. The pattern is plotted on the
existing canvas (i.e., use <code><a href="grid.html#topic+grid.newpage">grid::grid.newpage()</a></code> to make a new canvas).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grid.pattern_geo(params, boundary_df, aspect_ratio, legend = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="grid.pattern_geo_+3A_params">params</code></td>
<td>
<p>A list of pattern parameters to customize the plotted pattern
(see &quot;Details&quot;).</p>
</td></tr>
<tr><td><code id="grid.pattern_geo_+3A_boundary_df">boundary_df</code></td>
<td>
<p>A <code>data.frame</code> consisting of three columns: &quot;x&quot;
(x-coordinates), &quot;y&quot; (y-coordinates), and &quot;id&quot; (polygon group ID). This
<code>data.frame</code> defines the boundary (as a closed polygon) of the plotted
pattern.</p>
</td></tr>
<tr><td><code id="grid.pattern_geo_+3A_aspect_ratio">aspect_ratio</code></td>
<td>
<p>Unused.</p>
</td></tr>
<tr><td><code id="grid.pattern_geo_+3A_legend">legend</code></td>
<td>
<p>Unused.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following <code>params</code> are accepted:
</p>

<dl>
<dt><strong><code>pattern_alpha</code></strong></dt><dd><p> Alpha transparency for pattern. default: 1</p>
</dd>
<dt><strong><code>pattern_colour</code></strong></dt><dd><p> Color used for strokes and points in
the pattern. default: 'black'</p>
</dd>
<dt><strong><code>pattern_fill</code></strong></dt><dd><p> Color used to fill various closed shapes
(e.g., circles) in the pattern. default: <code>NA</code></p>
</dd>
<dt><strong><code>pattern_scale</code></strong></dt><dd><p> Scale. default: 2</p>
</dd>
<dt><strong><code>pattern_type</code></strong></dt><dd><p> Code for the FGDC pattern to use. See
<code><a href="#topic+geo_pattern">geo_pattern()</a></code> for more details. default: &quot;101&quot; </p>
</dd>
<dt><strong><code>fill</code></strong></dt><dd><p> Color used for the background. default: &quot;white&quot; </p>
</dd>
</dl>



<h3>Warning</h3>

<p>Pattern fills are not supported on all graphics devices.
Not all devices are under active development, and such devices are unlikely
to add support for new features (such as pattern fills). The new features
have only been implemented on a subset of graphics devices so far:
<code><a href="grDevices.html#topic+cairo_pdf">cairo_pdf()</a></code>, <code><a href="grDevices.html#topic+cairo_ps">cairo_ps()</a></code>, <code><a href="grDevices.html#topic+x11">x11(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+png">png(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+jpeg">jpeg(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+tiff">tiff(type=&quot;cairo&quot;)</a></code>, <code><a href="grDevices.html#topic+svg">svg()</a></code>, and <code><a href="grDevices.html#topic+pdf">pdf()</a></code>. Although
there is no support yet for <code><a href="grDevices.html#topic+quartz">quartz()</a></code> or <code><a href="grDevices.html#topic+windows">windows()</a></code>, almost all of the
graphics devices above will work on all major platforms. Further, the
<a href="https://ragg.r-lib.org/">ragg</a> and
<a href="https://svglite.r-lib.org/index.html">svglite</a> packages contain graphics
devices that support patterns. When using a graphics device where patterns
are not supported, closed shapes will be rendered with a transparent fill.
Note that, at least on Windows machines, the default device in RStudio and
in the knitr package is <code><a href="grDevices.html#topic+png">png()</a></code>, which does not support patterns. In
RStudio, you can go to ‘Tools &gt; Global Options &gt; General &gt; Graphics’ and
choose the ‘Cairo PNG’ device from the dropdown menu to display patterns.
Similar issues may arise when using RStudio on other operating systems.
</p>


<h3>See Also</h3>

<p>FGDC patterns:
<code><a href="#topic+geo_pattern">geo_pattern</a>()</code>,
<code><a href="#topic+scale_fill_geopattern">scale_fill_geopattern</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># use the function directly to make a hexagon with the pattern
library(grid)
x &lt;- 0.5 + 0.5 * cos(seq(2 * pi / 4, by = 2 * pi / 6, length.out = 6))
y &lt;- 0.5 + 0.5 * sin(seq(2 * pi / 4, by = 2 * pi / 6, length.out = 6))
grid.newpage()
grid.pattern_geo(params = list(pattern_type = "633", pattern_scale = 4),
                 boundary_df = data.frame(x, y, id = 1))

# use the function via ggpattern by specifying `pattern = 'geo'`
library(ggplot2)
library(ggpattern)
df &lt;- data.frame(trt = c("a", "b", "c"), outcome = c(2.3, 1.9, 3.2))
ggplot(df, aes(trt, outcome)) +
  geom_col_pattern(aes(color = trt, pattern_type = trt), pattern = 'geo',
  pattern_color = "black", fill = "white", pattern_fill = "white") +
  scale_pattern_type_manual(values = c("101", "313", "634")) +
  scale_color_viridis_d() +
  theme(legend.key.size = unit(1.5, 'cm'))

</code></pre>

<hr>
<h2 id='gtable_frame2'>Decompose a ggplot gtable</h2><span id='topic+gtable_frame2'></span>

<h3>Description</h3>

<p>Reformat the gtable associated with a ggplot object into a 7x7 gtable where
the central cell corresponds to the plot panel(s), the rectangle of cells
around that corresponds to the axes, and the rectangle of cells around that
corresponds to the axis titles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_frame2(
  g,
  width = unit(1, "null"),
  height = unit(1, "null"),
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gtable_frame2_+3A_g">g</code></td>
<td>
<p>gtable</p>
</td></tr>
<tr><td><code id="gtable_frame2_+3A_width">width</code></td>
<td>
<p>requested width</p>
</td></tr>
<tr><td><code id="gtable_frame2_+3A_height">height</code></td>
<td>
<p>requested height</p>
</td></tr>
<tr><td><code id="gtable_frame2_+3A_debug">debug</code></td>
<td>
<p>logical draw gtable cells</p>
</td></tr>
</table>


<h3>Value</h3>

<p>7x7 gtable wrapping the plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(grid)
library(gridExtra)
library(ggplot2)
p1 &lt;- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) +
  geom_point()

p2 &lt;- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) +
  geom_point() +
  facet_wrap(~cyl, ncol = 2, scales = "free") +
  guides(colour = "none") +
  theme()

p3 &lt;- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) +
  geom_point() +
  facet_grid(. ~ cyl, scales = "free")

g1 &lt;- ggplotGrob(p1)
g2 &lt;- ggplotGrob(p2)
g3 &lt;- ggplotGrob(p3)
fg1 &lt;- gtable_frame2(g1)
fg2 &lt;- gtable_frame2(g2)
fg12 &lt;- gtable_frame2(gtable_rbind(fg1, fg2), width = unit(2, "null"),
                      height = unit(1, "null"))
fg3 &lt;- gtable_frame2(g3, width = unit(1, "null"), height = unit(1, "null"))
grid.newpage()
combined &lt;- gtable_cbind(fg12, fg3)
grid.draw(combined)
</code></pre>

<hr>
<h2 id='guide_geo'>Geological timescale axis guide</h2><span id='topic+guide_geo'></span><span id='topic+GuideGeo'></span>

<h3>Description</h3>

<p><code>guide_geo</code> behaves similarly to <code><a href="ggplot2.html#topic+guide_axis">ggplot2::guide_axis()</a></code> in that it modifies
the visual appearance of the axis. The main difference is that it adds a
geological timescale instead of an axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>guide_geo(
  dat = "periods",
  fill = NULL,
  alpha = 1,
  height = unit(2, "line"),
  bord = c("left", "right", "top", "bottom"),
  lwd = 0.25,
  color = "black",
  lab = TRUE,
  lab_color = NULL,
  rot = 0,
  family = "sans",
  fontface = "plain",
  size = 5,
  skip = c("Quaternary", "Holocene", "Late Pleistocene"),
  abbrv = TRUE,
  neg = FALSE,
  end_labels = "center",
  dat_is_discrete = FALSE,
  fittext_args = list(),
  theme = NULL,
  title = waiver(),
  order = 0,
  position = waiver()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="guide_geo_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;),
B) a string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>), or C) a custom
data.frame of time interval boundaries (see Details).</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_fill">fill</code></td>
<td>
<p>The fill color of the boxes. The default is to use the <code>color</code>
column included in <code>dat</code>. If a custom dataset is provided with <code>dat</code>
without a <code>color</code> column and without fill, a greyscale will be used.
Custom fill colors can be provided with this option (overriding the <code>color</code>
column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_alpha">alpha</code></td>
<td>
<p>The transparency of the fill colors.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_height">height</code></td>
<td>
<p>The height (or width if <code>pos</code> is <code>left</code> or <code>right</code>) of the
scale.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_bord">bord</code></td>
<td>
<p>A vector specifying on which sides of the scale to add borders
(same options as <code>pos</code>).</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_color">color</code></td>
<td>
<p>The outline color of the interval boxes.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_lab">lab</code></td>
<td>
<p>Whether to include labels.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_lab_color">lab_color</code></td>
<td>
<p>The color of the labels. The default is to use the
<code>lab_color</code> column included in <code>dat</code>. If a custom dataset is provided with
<code>dat</code> without a <code>lab_color</code> column and without fill, all labels will be
black. Custom label colors can be provided with this option (overriding the
<code>lab_color</code> column) and will be recycled if/as necessary.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_rot">rot</code></td>
<td>
<p>The amount of counter-clockwise rotation to add to the labels
(in degrees).</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_family">family</code></td>
<td>
<p>The font family to use for the labels. There are only three
fonts that are guaranteed to work everywhere: “sans” (the default),
“serif”, or “mono”.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_fontface">fontface</code></td>
<td>
<p>The font face to use for the labels. The standard options are
&quot;plain&quot; (default), &quot;bold&quot;, &quot;italic&quot;, and &quot;bold.italic&quot;.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_size">size</code></td>
<td>
<p>Label size. Either a number as you would specify in
<code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code> or <code>"auto"</code> to use <code><a href="ggfittext.html#topic+geom_fit_text">ggfittext::geom_fit_text()</a></code>.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_skip">skip</code></td>
<td>
<p>A vector of interval names indicating which intervals should not
be labeled. If <code>abbrv</code> is <code>TRUE</code>, this can also include interval
abbreviations.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_abbrv">abbrv</code></td>
<td>
<p>If including labels, should the labels be abbreviated? If
<code>TRUE</code>, the <code>abbr</code> column will be used for the labels. If <code>FALSE</code>, the
<code>name</code> column will be used for the labels. If <code>"auto"</code>, the <code><a href="base.html#topic+abbreviate">abbreviate()</a></code>
function will be used to abbreviate the values in the <code>name</code> column. Note
that the built-in data and data retrieved via <code><a href="#topic+get_scale_data">get_scale_data()</a></code> already
have built-in abbreviations. However, using the <code>"auto"</code> option here will
create new unique abbreviations based on only the intervals that are being
plotted. In many cases, this may result in abbreviations that are shorter
in length because there are fewer similar interval names to abbreviate.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_neg">neg</code></td>
<td>
<p>Set this to <code>TRUE</code> if your x-axis is using negative values.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_end_labels">end_labels</code></td>
<td>
<p>How should labels for intervals at the ends of the guide be
treated? &quot;center&quot;, the default, centers the labels within the visible part
of the label. &quot;clip&quot; removes the labels if their midpoint is beyond the
axis limits. &quot;keep&quot; plots the labels in the midpoint of the full interval.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_dat_is_discrete">dat_is_discrete</code></td>
<td>
<p>Are the ages in <code>dat</code> already converted for a discrete
scale?</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_fittext_args">fittext_args</code></td>
<td>
<p>A list of named arguments to provide to
<code><a href="ggfittext.html#topic+geom_fit_text">ggfittext::geom_fit_text()</a></code>. Only used if <code>size</code> is set to <code>"auto"</code>.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_theme">theme</code></td>
<td>
<p>A <code><a href="ggplot2.html#topic+theme">theme</a></code> object to style the guide individually or
differently from the plot's theme settings. The <code>theme</code> argument in the
guide overrides, and is combined with, the plot's theme.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_title">title</code></td>
<td>
<p>A character string or expression indicating a title of guide.
If <code>NULL</code>, the title is not shown. By default
(<code><a href="ggplot2.html#topic+waiver">waiver()</a></code>), the name of the scale object or the name
specified in <code><a href="ggplot2.html#topic+labs">labs()</a></code> is used for the title.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_order">order</code></td>
<td>
<p>A positive <code>integer</code> of length 1 that specifies the order of
this guide among multiple guides. This controls in which order guides are
merged if there are multiple guides for the same position. If 0 (default),
the order is determined by a secret algorithm.</p>
</td></tr>
<tr><td><code id="guide_geo_+3A_position">position</code></td>
<td>
<p>Where this guide should be drawn: one of top, bottom,
left, or right.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a custom data.frame is provided (with <code>dat</code>), it should consist of at
least 3 columns of data. See <code>data(periods)</code> for an example.
</p>

<ul>
<li><p> The <code>name</code> column lists the names of each time interval. These will
be used as labels if no abbreviations are provided.
</p>
</li>
<li><p> The <code>max_age</code> column lists the oldest boundary of each time interval.
</p>
</li>
<li><p> The <code>min_age</code> column lists the youngest boundary of each time
interval.
</p>
</li>
<li><p> The <code>abbr</code> column is optional and lists abbreviations that may be
used as labels.
</p>
</li>
<li><p> The <code>color</code> column is also optional and lists a
<a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a> for the background for each time interval.
</p>
</li>
<li><p> The <code>lab_color</code> column is also optional and lists a
<a href="ggplot2.html#topic+aes_colour_fill_alpha">color</a> for the label for each time interval.
</p>
</li></ul>

<p>If the axis of the time scale is discrete, <code>max_age</code> and <code>min_age</code> will
automatically be converted to the discrete scale. In this case, the
categories of the discrete axis should match the values in the <code>name</code> column.
If the ages within <code>dat</code> are already discretized, you can set
<code>dat_is_discrete</code> to <code>TRUE</code> to prevent this automatic conversion. This can be
useful for adding a time scale where categories and time intervals are not
1:1.
</p>
<p>Since this guide only plots the timescale and not ticks or an axis line, a
call to this function should almost always be combined with a call to
<code><a href="ggplot2.html#topic+guide_axis">ggplot2::guide_axis()</a></code> within a call to <code><a href="ggplot2.html#topic+guide_axis_stack">ggplot2::guide_axis_stack()</a></code> (see
Examples). Note that in most cases this has the same end result as a single
call to <code><a href="#topic+coord_geo">coord_geo()</a></code>; however, there are some use cases in which this may be
more useful or allow for more customization. For example, users may wish to
combine this guide in unique ways with other guides. Further, since
<code><a href="#topic+coord_geo">coord_geo()</a></code> doesn't work with radial/fan phylogenies (and
<code><a href="#topic+coord_geo_radial">coord_geo_radial()</a></code> is quite different visually), this guide can be used to
achieve the look of <code><a href="#topic+coord_geo">coord_geo()</a></code> on a radial/fan phylogeny.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
# reproduce the coord_geo() appearance
ggplot() +
  geom_point(aes(y = runif(1000, 0, 8), x = runif(1000, 0, 1000))) +
  scale_x_reverse(guide = guide_axis_stack(guide_geo(), "axis",
                  spacing = unit(0, "npc"))) +
  coord_cartesian(xlim = c(1000, 0), ylim = c(0, 8)) +
  theme_classic()

# the coord_geo() look on a radial phylogeny
library(ggtree)
library(paleotree)
data(RaiaCopesRule)
ggtree(ceratopsianTreeRaia,
       position = position_nudge(x = -ceratopsianTreeRaia$root.time)) +
  coord_geo_radial(dat = "stages", fill = c("grey80", "grey95"),
                   end = 1.25 * pi) +
  guides(
    r = guide_axis_stack(guide_geo(rot = -90, neg = TRUE,
                                   height = unit(0.5, "line")),
                         "axis", spacing = unit(0, "npc"))
  ) +
  scale_y_continuous(guide = "none", breaks = NULL) +
  theme_classic()

</code></pre>

<hr>
<h2 id='panel.disparity'>Combined wireframe and cloud panel</h2><span id='topic+panel.disparity'></span>

<h3>Description</h3>

<p>Plots the provided data on 2-D surfaces within a 3-D framework. See
<code><a href="#topic+disparity_through_time">disparity_through_time()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.disparity(x, y, z, groups, subscripts, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="panel.disparity_+3A_x">x</code>, <code id="panel.disparity_+3A_y">y</code>, <code id="panel.disparity_+3A_z">z</code>, <code id="panel.disparity_+3A_groups">groups</code>, <code id="panel.disparity_+3A_subscripts">subscripts</code>, <code id="panel.disparity_+3A_...">...</code></td>
<td>
<p>Same as for <code><a href="lattice.html#topic+panel.cloud">lattice::panel.cloud()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, plots the results of both <code><a href="lattice.html#topic+panel.cloud">lattice::panel.cloud()</a></code>
and <code><a href="lattice.html#topic+panel.cloud">lattice::panel.wireframe()</a></code>.
</p>

<hr>
<h2 id='periods'>Period data from the International Commission on Stratigraphy (v2023/06)</h2><span id='topic+periods'></span>

<h3>Description</h3>

<p>A dataset containing the boundary ages, abbreviations, and colors for the
periods of the Geologic Time Scale. Based on The ICS International
Chronostratigraphic Chart (v2023/06), by Cohen, Finney, Gibbard, and Fan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>periods
</code></pre>


<h3>Format</h3>

<p>A data frame with 22 rows and 5 variables:
</p>

<dl>
<dt>name</dt><dd><p>period name</p>
</dd>
<dt>max_age</dt><dd><p>maximum age, in millions of years</p>
</dd>
<dt>min_age</dt><dd><p>minimum age, in millions of years</p>
</dd>
<dt>abbr</dt><dd><p>period name abbreviations</p>
</dd>
<dt>color</dt><dd><p>the colors for each period, according to the Commission for
the Geological Map of the World</p>
</dd>
<dt>lab_color</dt><dd><p>the label colors for each period, either white or black,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://stratigraphy.org">https://stratigraphy.org</a> via <a href="https://macrostrat.org/api/v2/defs/intervals?timescale=international%20periods">https://macrostrat.org/api/v2/defs/intervals?timescale=international%20periods</a>
</p>


<h3>See Also</h3>

<p>Other built-in timescales: 
<code><a href="#topic+eons">eons</a></code>,
<code><a href="#topic+epochs">epochs</a></code>,
<code><a href="#topic+eras">eras</a></code>,
<code><a href="#topic+stages">stages</a></code>
</p>

<hr>
<h2 id='scale_color_geo'>Geological Time Scale color scales</h2><span id='topic+scale_color_geo'></span><span id='topic+scale_colour_geo'></span><span id='topic+scale_fill_geo'></span><span id='topic+scale_discrete_geo'></span>

<h3>Description</h3>

<p>Color scales using the colors in the Geological Time Scale graphics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_color_geo(dat, ...)

scale_fill_geo(dat, ...)

scale_discrete_geo(dat, aesthetics, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_color_geo_+3A_dat">dat</code></td>
<td>
<p>Either A) a string indicating a built-in dataframe with interval
data from the ICS (&quot;periods&quot;, &quot;epochs&quot;, &quot;stages&quot;, &quot;eons&quot;, or &quot;eras&quot;),
B) a string indicating a timescale from macrostrat (see list here:
<a href="https://macrostrat.org/api/defs/timescales?all">https://macrostrat.org/api/defs/timescales?all</a>),
or C) a custom data.frame of time interval boundaries
(see <code><a href="#topic+coord_geo">coord_geo()</a></code>).</p>
</td></tr>
<tr><td><code id="scale_color_geo_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="ggplot2.html#topic+discrete_scale">ggplot2::discrete_scale</a></code>
</p>

<dl>
<dt><code>scale_name</code></dt><dd><p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> The name of the scale
that should be used for error messages associated with this scale.</p>
</dd>
<dt><code>name</code></dt><dd><p>The name of the scale. Used as the axis or legend title. If
<code>waiver()</code>, the default, the name of the scale is taken from the first
mapping used for that aesthetic. If <code>NULL</code>, the legend title will be
omitted.</p>
</dd>
<dt><code>labels</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> for no labels
</p>
</li>
<li> <p><code>waiver()</code> for the default labels computed by the
transformation object
</p>
</li>
<li><p> A character vector giving labels (must be same length as <code>breaks</code>)
</p>
</li>
<li><p> An expression vector (must be the same length as breaks). See ?plotmath for details.
</p>
</li>
<li><p> A function that takes the breaks as input and returns labels
as output. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>limits</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> to use the default scale values
</p>
</li>
<li><p> A character vector that defines possible values of the scale and their
order
</p>
</li>
<li><p> A function that accepts the existing (automatic) values and returns
new ones. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>na.translate</code></dt><dd><p>Unlike continuous scales, discrete scales can easily show
missing values, and do so by default. If you want to remove missing values
from a discrete scale, specify <code>na.translate = FALSE</code>.</p>
</dd>
<dt><code>na.value</code></dt><dd><p>If <code>na.translate = TRUE</code>, what aesthetic value should the
missing values be displayed as? Does not apply to position scales
where <code>NA</code> is always placed at the far right.</p>
</dd>
<dt><code>drop</code></dt><dd><p>Should unused factor levels be omitted from the scale?
The default, <code>TRUE</code>, uses the levels that appear in the data;
<code>FALSE</code> includes the levels in the factor. Please note that to display
every level in a legend, the layer should use <code>show.legend = TRUE</code>.</p>
</dd>
<dt><code>guide</code></dt><dd><p>A function used to create a guide or its name. See
<code><a href="ggplot2.html#topic+guides">guides()</a></code> for more information.</p>
</dd>
<dt><code>call</code></dt><dd><p>The <code>call</code> used to construct the scale for reporting messages.</p>
</dd>
<dt><code>super</code></dt><dd><p>The super class to use for the constructed scale</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="scale_color_geo_+3A_aesthetics">aesthetics</code></td>
<td>
<p>Character string or vector of character strings listing the
name(s) of the aesthetic(s) that this scale works with. This can be useful, for
example, to apply colour settings to the <code>colour</code> and <code>fill</code> aesthetics at the
same time, via <code>aesthetics = c("colour", "fill")</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
df &lt;- data.frame(
  x = runif(1000, 0, 10), y = runif(1000, 0, 10),
  color = sample(periods$name, 1000, TRUE), shape = 21
)
ggplot(df) +
  geom_point(aes(x = x, y = y, fill = color), shape = 21) +
  scale_fill_geo("periods", name = "Period") +
  theme_classic()

# cut continuous variable into discrete
df &lt;- data.frame(x = runif(1000, 0, 1000), y = runif(1000, 0, 8))
df$color &lt;- cut(df$x, c(periods$min_age, periods$max_age[22]), periods$name)
ggplot(df) +
  geom_point(aes(x = x, y = y, color = color)) +
  scale_x_reverse() +
  scale_color_geo("periods", name = "Period") +
  coord_geo(xlim = c(1000, 0), ylim = c(0, 8)) +
  theme_classic()
</code></pre>

<hr>
<h2 id='scale_fill_geopattern'>Geologic pattern fill scale</h2><span id='topic+scale_fill_geopattern'></span>

<h3>Description</h3>

<p>Fill scale using the <a href="https://ngmdb.usgs.gov/fgdc_gds/geolsymstd.php">FGDC Digital Cartographic Standard for Geologic Map Symbolization</a>. Fill values
should correspond to specific pattern codes (see &quot;Details&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_fill_geopattern(na.value = "grey50", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_fill_geopattern_+3A_na.value">na.value</code></td>
<td>
<p>The aesthetic value to use for missing (NA) values. May be
either a color or a <a href="grid.html#topic+patterns">GridPattern</a> object (such as that
returned by <code><a href="#topic+geo_pattern">geo_pattern()</a></code>).</p>
</td></tr>
<tr><td><code id="scale_fill_geopattern_+3A_...">...</code></td>
<td>

<p>Arguments passed on to <code><a href="ggplot2.html#topic+discrete_scale">ggplot2::discrete_scale</a></code>
</p>

<dl>
<dt><code>scale_name</code></dt><dd><p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> The name of the scale
that should be used for error messages associated with this scale.</p>
</dd>
<dt><code>name</code></dt><dd><p>The name of the scale. Used as the axis or legend title. If
<code>waiver()</code>, the default, the name of the scale is taken from the first
mapping used for that aesthetic. If <code>NULL</code>, the legend title will be
omitted.</p>
</dd>
<dt><code>breaks</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> for no breaks
</p>
</li>
<li> <p><code>waiver()</code> for the default breaks (the scale limits)
</p>
</li>
<li><p> A character vector of breaks
</p>
</li>
<li><p> A function that takes the limits as input and returns breaks
as output. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>labels</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> for no labels
</p>
</li>
<li> <p><code>waiver()</code> for the default labels computed by the
transformation object
</p>
</li>
<li><p> A character vector giving labels (must be same length as <code>breaks</code>)
</p>
</li>
<li><p> An expression vector (must be the same length as breaks). See ?plotmath for details.
</p>
</li>
<li><p> A function that takes the breaks as input and returns labels
as output. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>limits</code></dt><dd><p>One of:
</p>

<ul>
<li> <p><code>NULL</code> to use the default scale values
</p>
</li>
<li><p> A character vector that defines possible values of the scale and their
order
</p>
</li>
<li><p> A function that accepts the existing (automatic) values and returns
new ones. Also accepts rlang <a href="rlang.html#topic+as_function">lambda</a> function
notation.
</p>
</li></ul>
</dd>
<dt><code>na.translate</code></dt><dd><p>Unlike continuous scales, discrete scales can easily show
missing values, and do so by default. If you want to remove missing values
from a discrete scale, specify <code>na.translate = FALSE</code>.</p>
</dd>
<dt><code>drop</code></dt><dd><p>Should unused factor levels be omitted from the scale?
The default, <code>TRUE</code>, uses the levels that appear in the data;
<code>FALSE</code> includes the levels in the factor. Please note that to display
every level in a legend, the layer should use <code>show.legend = TRUE</code>.</p>
</dd>
<dt><code>guide</code></dt><dd><p>A function used to create a guide or its name. See
<code><a href="ggplot2.html#topic+guides">guides()</a></code> for more information.</p>
</dd>
<dt><code>call</code></dt><dd><p>The <code>call</code> used to construct the scale for reporting messages.</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Details</h3>

<p>For specific codes, see the &quot;pattern numbers&quot; in the <a href="https://ngmdb.usgs.gov/fgdc_gds/geolsymstd/fgdc-geolsym-patternchart.pdf">full pattern chart</a>
for valid <code>code</code> values. Daven Quinn has also assembled more accessible
documentation of the <a href="https://davenquinn.com/projects/geologic-patterns/#pattern-reference">map patterns/codes</a>
and <a href="https://davenquinn.com/projects/geologic-patterns/#series-600">lithology patterns/codes</a>.
<code><a href="rmacrostrat.html#topic+def_lithologies">rmacrostrat::def_lithologies()</a></code> can also be used to look up pattern codes
for various lithologies (see the &quot;fill&quot; column). Note that codes associated
with color variants (e.g., &quot;101-M&quot;) are supported but will result in the
default color variant instead (usually black and white, e.g., &quot;101-K&quot;).
</p>
<p>These patterns were originally processed and optimized by Daven Quinn and
are hosted on <a href="https://github.com/davenquinn/geologic-patterns/">GitHub</a>.
</p>


<h3>Warning</h3>

<p>Pattern fills are not supported on all graphics devices.
Not all devices are under active development, and such devices are unlikely
to add support for new features (such as pattern fills). The new features
have only been implemented on a subset of graphics devices so far:
<code><a href="grDevices.html#topic+cairo_pdf">cairo_pdf()</a></code>, <code><a href="grDevices.html#topic+cairo_ps">cairo_ps()</a></code>, <code><a href="grDevices.html#topic+x11">x11(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+png">png(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+jpeg">jpeg(type=&quot;cairo&quot;)</a></code>,
<code><a href="grDevices.html#topic+tiff">tiff(type=&quot;cairo&quot;)</a></code>, <code><a href="grDevices.html#topic+svg">svg()</a></code>, and <code><a href="grDevices.html#topic+pdf">pdf()</a></code>. Although
there is no support yet for <code><a href="grDevices.html#topic+quartz">quartz()</a></code> or <code><a href="grDevices.html#topic+windows">windows()</a></code>, almost all of the
graphics devices above will work on all major platforms. Further, the
<a href="https://ragg.r-lib.org/">ragg</a> and
<a href="https://svglite.r-lib.org/index.html">svglite</a> packages contain graphics
devices that support patterns. When using a graphics device where patterns
are not supported, closed shapes will be rendered with a transparent fill.
Note that, at least on Windows machines, the default device in RStudio and
in the knitr package is <code><a href="grDevices.html#topic+png">png()</a></code>, which does not support patterns. In
RStudio, you can go to ‘Tools &gt; Global Options &gt; General &gt; Graphics’ and
choose the ‘Cairo PNG’ device from the dropdown menu to display patterns.
Similar issues may arise when using RStudio on other operating systems.
</p>


<h3>See Also</h3>

<p>FGDC patterns:
<code><a href="#topic+geo_pattern">geo_pattern</a>()</code>,
<code><a href="#topic+grid.pattern_geo">grid.pattern_geo</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
vals &lt;- c("101", "313", "603", "733")
ggplot(mpg, aes(factor(cyl), fill = vals[factor(cyl)])) +
  geom_bar() +
  scale_fill_geopattern(name = NULL)
</code></pre>

<hr>
<h2 id='stages'>Stage data from the International Commission on Stratigraphy (v2023/06)</h2><span id='topic+stages'></span>

<h3>Description</h3>

<p>A dataset containing the boundary ages, abbreviations, and colors for the
stages of the Geologic Time Scale. Based on The ICS International
Chronostratigraphic Chart (v2023/06), by Cohen, Finney, Gibbard, and Fan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stages
</code></pre>


<h3>Format</h3>

<p>A data frame with 102 rows and 5 variables:
</p>

<dl>
<dt>name</dt><dd><p>stage name</p>
</dd>
<dt>max_age</dt><dd><p>maximum age, in millions of years</p>
</dd>
<dt>min_age</dt><dd><p>minimum age, in millions of years</p>
</dd>
<dt>abbr</dt><dd><p>stage name abbreviations</p>
</dd>
<dt>color</dt><dd><p>the colors for each stage, according to the Commission for the
Geological Map of the World</p>
</dd>
<dt>lab_color</dt><dd><p>the label colors for each stage, either white or black,
whichever has better contrast with the background color, based on
<a href="https://www.itu.int/rec/R-REC-BT.601-7-201103-I/en">recommendations by the International Telecommunication Union</a></p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://stratigraphy.org">https://stratigraphy.org</a> via <a href="https://macrostrat.org/api/v2/defs/intervals?timescale=international%20ages">https://macrostrat.org/api/v2/defs/intervals?timescale=international%20ages</a>
</p>


<h3>See Also</h3>

<p>Other built-in timescales: 
<code><a href="#topic+eons">eons</a></code>,
<code><a href="#topic+epochs">epochs</a></code>,
<code><a href="#topic+eras">eras</a></code>,
<code><a href="#topic+periods">periods</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
