<!DOCTYPE html><html lang="en"><head><title>Help for package DEET</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DEET}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust_DE_cutoffs'><p>adjust_DE_cutoffs</p></a></li>
<li><a href='#DEET_data_download'><p>DEET_data_download</p></a></li>
<li><a href='#DEET_enrich'><p>DEET_enrich</p></a></li>
<li><a href='#DEET_enrich_genesonly'><p>DEET_enrich_genesonly</p></a></li>
<li><a href='#DEET_enrichment_plot'><p>DEET_enrichment_plot</p></a></li>
<li><a href='#DEET_example_data'><p>DEET_example_data</p></a></li>
<li><a href='#DEET_feature_extract'><p>DEET_feature_extract</p></a></li>
<li><a href='#DEET_feature_extract_example_matrix'><p>DEET_feature_extract_example_matrix</p></a></li>
<li><a href='#DEET_feature_extract_example_response'><p>DEET_feature_extract_example_response</p></a></li>
<li><a href='#DEET_Input_as_Reference'><p>DEET_Input_as_Reference</p></a></li>
<li><a href='#DEET_plot_correlation'><p>DEET_plot_correlation</p></a></li>
<li><a href='#example_DEET_enrich_input'><p>example_DEET_enrich_input</p></a></li>
<li><a href='#proccess_and_plot_DEET_enrich'><p>proccess_and_plot_DEET_enrich</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Differential Expression Enrichment Tool</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.12</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dustin Sokolowski &lt;dustin.sokolowski@sickkids.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Abstract of Manuscript. Differential gene expression analysis using RNA sequencing (RNA-seq) data is a standard approach for making biological discoveries. Ongoing large-scale efforts to process and normalize publicly available gene expression data enable rapid and systematic reanalysis. While several powerful tools systematically process RNA-seq data, enabling their reanalysis, few resources systematically recompute differentially expressed genes (DEGs) generated from individual studies. We developed a robust differential expression analysis pipeline to recompute 3162 human DEG lists from The Cancer Genome Atlas, Genotype-Tissue Expression Consortium, and 142 studies within the Sequence Read Archive. After measuring the accuracy of the recomputed DEG lists, we built the Differential Expression Enrichment Tool (DEET), which enables users to interact with the recomputed DEG lists. DEET, available through CRAN and RShiny, systematically queries which of the recomputed DEG lists share similar genes, pathways, and TF targets to their own gene lists. DEET identifies relevant studies based on shared results with the userâ€™s gene lists, aiding in hypothesis generation and data-driven literature review. Sokolowski, Dustin J., et al. "Differential Expression Enrichment Tool (DEET): an interactive atlas of human differential gene expression." Nucleic Acids Research Genomics and Bioinformatics (2023).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ActivePathways, pbapply, dplyr, ggplot2, glmnet, utils, stats,
ggrepel, downloader</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-26 03:56:49 UTC; djsok</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Dustin Sokolowski [aut, cre],
  Jedid Ahn [aut],
  Lauren Erdman [aut],
  Kai Ellis [aut],
  Huayun Hou [aut],
  Anna Goldenberg [aut],
  Michael Wilson [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-26 10:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjust_DE_cutoffs'>adjust_DE_cutoffs</h2><span id='topic+adjust_DE_cutoffs'></span>

<h3>Description</h3>

<p>Utility function to adjust mean expression, FDR, and log2FC cutoffs of the database of DEGs inputted into DEET.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust_DE_cutoffs(
  DEET_combined,
  redo_pathways_instructions = FALSE,
  baseMean = 1,
  abslog2FoldChange = 0,
  padj = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust_DE_cutoffs_+3A_deet_combined">DEET_combined</code></td>
<td>
<p>The databank of the differential expression enrichment tool.
Appropriate inputs here are &quot;DEET_example_data&quot; stored within DEET, the &quot;DEET_combined.rda&quot; file
from the DEET stable repositoy found at X, and the DEET database developmental repository found at Y.
The DEET_dataset is a named list where details of it's structure can be found ?DEET_example_data.</p>
</td></tr>
<tr><td><code id="adjust_DE_cutoffs_+3A_redo_pathways_instructions">redo_pathways_instructions</code></td>
<td>
<p>Boolean value specifying whether to print the instructions
required to update all pathway enrichments based on new DE cutoffs.</p>
</td></tr>
<tr><td><code id="adjust_DE_cutoffs_+3A_basemean">baseMean</code></td>
<td>
<p>Change the mean-expression cutoff.</p>
</td></tr>
<tr><td><code id="adjust_DE_cutoffs_+3A_abslog2foldchange">abslog2FoldChange</code></td>
<td>
<p>Change the log2 Fold-change cutoff.</p>
</td></tr>
<tr><td><code id="adjust_DE_cutoffs_+3A_padj">padj</code></td>
<td>
<p>Change the FDR-adjusted p-value cutoff.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The DEET_combined object but with the user-inputted 
expression, log2FC, and FDR-adjusted p-value cutoffs. DEET_gmt_DE
is also updated to the new cutoffs.
</p>


<h3>Author(s)</h3>

<p>Dustin Sokolowski
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("DEET_example_data")
DEET_cutoff &lt;- adjust_DE_cutoffs(DEET_example_data, abslog2FoldChange = 1, padj = 0.01)



</code></pre>

<hr>
<h2 id='DEET_data_download'>DEET_data_download</h2><span id='topic+DEET_data_download'></span>

<h3>Description</h3>

<p>Function to automatically download the files within
the DEET database that are required for the DEET_enrich
and DEET_feature_extractfunctions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_data_download(x = "enrich")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_data_download_+3A_x">x</code></td>
<td>
<p>categorical variable containing options &quot;ALL&quot;, &quot;enrich&quot;,
&quot;metadata&quot; or &quot;feature_matrix&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list with the neccesary data required to input into
DEET_feature_extract or DEET_enrich. The metadata within DEET
can also be downloaded.
</p>

<ul>
<li><p> feature_matrix - A gene by comparison matrix populated
with the log2FC of gene expression for all genes, regardless
of DE status.
</p>
</li>
<li><p> metadata - a comparison - by - explanatory piece of data dataframe
providing important details to contextualize each study.
For every pairwise comparison, the study name,
source (SRA, TCGA, GTEx and SRA-manual),
description from the DRA compendium,
the number of samples (total, up-condition,
and down-condition), samples (total ,up-condition, down-condition),
tissue (including tumour from TCGA),
number of DEs (total, up-condition, down-condition),
age (mean +- sd), sex, top 15 DEGs - up, top 15 DEGs - down,
top 5 enriched pathways, and top 5 enriched TFs.
PMID are also available for studies selected from SRA.
Lastly, each pairwise comparison was given an overall
category based on those decided in Crow et al., 2019.
</p>
</li>
<li><p> DEET_enrich - A named list of seven objects containing
the data frames summarizing the DEGs from comparisons within
DEET, GMT objects of comparisons within DEET for enrichment
through ActivePathways, GMT objects for basic pathway and TF
enrichment, and a dataframe for the metadata of each study.
For more detail on each element of the list, please consult
the vignette or &quot;?DEET_example_data&quot;, as it is a subset
of this object
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dustin Sokolowski, Jedid Ahn
</p>


<h3>References</h3>

<p>Engebretsen, S., &amp; Bohlin, J. (2019). Statistical predictions with glmnet.
Clinical epigenetics, 11(1), 1-3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# Download the metadata. Downloading other
# files within DEET are larger and take
# a bit more time.
downloaded &lt;- DEET_data_download(x = "metadata")

# extract metadata from the list
metadata &lt;- downloaded[["metadata"]]



</code></pre>

<hr>
<h2 id='DEET_enrich'>DEET_enrich</h2><span id='topic+DEET_enrich'></span>

<h3>Description</h3>

<p>Core function of DEET where an input weighted human gene list
will be queried to DEETs library of studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_enrich(
  DEG_list,
  DEET_dataset,
  ordered = FALSE,
  background = NULL,
  abs_cor = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_enrich_+3A_deg_list">DEG_list</code></td>
<td>
<p>Data frame or matrix of gene symbols with corresponding padj
and log2FC values (3 columns in total). Can also be a character vector of
gene symbols only. colnames of genes: c(&quot;gene_symbol&quot;, &quot;padj&quot;, &quot;coef&quot;)
The rownames of the dataframe are also the gene symbols.</p>
</td></tr>
<tr><td><code id="DEET_enrich_+3A_deet_dataset">DEET_dataset</code></td>
<td>
<p>The databank of the differential expression enrichment tool.
Appropriate inputs here are &quot;DEET_example_data&quot; stored within DEET, the &quot;DEET_combined.rda&quot; file
from the DEET stable repositoy found at X, and the DEET database developmental repository found at Y.
The DEET_dataset is a named list where details of it's structure can be found ?DEET_example_data.</p>
</td></tr>
<tr><td><code id="DEET_enrich_+3A_ordered">ordered</code></td>
<td>
<p>Boolean value specifying whether DEG_list is a character
vector of gene symbols that is ordered. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="DEET_enrich_+3A_background">background</code></td>
<td>
<p>Character vector of human gene symbols showing all
possible genes. Default value is NULL.</p>
</td></tr>
<tr><td><code id="DEET_enrich_+3A_abs_cor">abs_cor</code></td>
<td>
<p>Boolean value that forces log2FC's in DEET to be
their absolute value. Use when the directionality of the coefficient 
is unknown (or includes both up- down- directions). Default value
is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list where each element contains 6 objects. Each object will
contain the results (enrichment or correlation) and corresponding metadata.
</p>

<ul>
<li><p> AP_INPUT_BP_output - Enriched BPs of input gene list.
</p>
</li>
<li><p> AP_INPUT_TF_output - Enriched TFs of input gene list.
</p>
</li>
<li><p> AP_DEET_DE_output  - Enrichment of input gene list on DEETs studies.
</p>
</li>
<li><p> AP_DEET_BP_output  - Enrichment of BPs of input gene list on DEETs
BPs of studies.
</p>
</li>
<li><p> AP_DEET_TF_output  - Enrichment of TFs of input gene list on DEETs
TFs of studies.
</p>
</li>
<li><p> DE_correlations    - Correlation values of input gene list to DEETs
studies (both Pearson and Spearman).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dustin Sokolowski, Jedid Ahn
</p>


<h3>References</h3>

<p>Paczkowska M, Barenboim J, Sintupisut N, et al. Integrative pathway
enrichment analysis of multivariate omics data. Nat Commun. 2020;11(1):735.
doi:10.1038/s41467-019-13983-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("example_DEET_enrich_input")
data("DEET_example_data")
DEET_out &lt;- DEET_enrich(example_DEET_enrich_input, DEET_dataset = DEET_example_data)


</code></pre>

<hr>
<h2 id='DEET_enrich_genesonly'>DEET_enrich_genesonly</h2><span id='topic+DEET_enrich_genesonly'></span>

<h3>Description</h3>

<p>Altered version of the  function of DEET where 
an input weighted human gene list will be queried to DEETs 
library of studies. This version does not include pathway 
enrichments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_enrich_genesonly(
  DEG_list,
  DEET_dataset,
  ordered = FALSE,
  background = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_enrich_genesonly_+3A_deg_list">DEG_list</code></td>
<td>
<p>Data frame or matrix of gene symbols with corresponding padj
and log2FC values (3 columns in total). Can also be a character vector of
gene symbols only. colnames of genes: c(&quot;gene_symbol&quot;, &quot;padj&quot;, &quot;coef&quot;)
The rownames of the dataframe are also the gene symbols.</p>
</td></tr>
<tr><td><code id="DEET_enrich_genesonly_+3A_deet_dataset">DEET_dataset</code></td>
<td>
<p>The databank of the differential expression enrichment tool.
Appropriate inputs here are &quot;DEET_example_data&quot; stored within DEET, the &quot;DEET_combined.rda&quot; file
from the DEET stable repositoy found at X, and the DEET database developmental repository found at Y.
The DEET_dataset is a named list where details of it's structure can be found ?DEET_example_data.
Unlike in DEET_enrich, this dataset does not require the pathway-relevant elements
of the DEET_dataset list, namely &quot;gmt_BP&quot;, or &quot;gmt_TF&quot; &quot;DEET_gmt_BP&quot;,  &quot;DEET_gmt_TF&quot;.</p>
</td></tr>
<tr><td><code id="DEET_enrich_genesonly_+3A_ordered">ordered</code></td>
<td>
<p>Boolean value specifying whether DEG_list is a character
vector of gene symbols that is ordered. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="DEET_enrich_genesonly_+3A_background">background</code></td>
<td>
<p>Character vector of human gene symbols showing all
possible genes. Default value is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list where each element contains 2 objects. Each object will
contain the results (enrichment or correlation) and corresponding metadata.
</p>

<ul>
<li><p> AP_DEET_DE_output  - Enrichment of input gene list on DEETs studies.
</p>
</li>
<li><p> DE_correlations    - Correlation values of input gene list to DEETs
studies (both Pearson and Spearman).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dustin Sokolowski, Jedid Ahn
</p>


<h3>References</h3>

<p>Paczkowska M, Barenboim J, Sintupisut N, et al. Integrative pathway
enrichment analysis of multivariate omics data. Nat Commun. 2020;11(1):735.
doi:10.1038/s41467-019-13983-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("example_DEET_enrich_input")
data("DEET_example_data")
DEET_out &lt;- DEET_enrich_genesonly(example_DEET_enrich_input, DEET_dataset = DEET_example_data)


</code></pre>

<hr>
<h2 id='DEET_enrichment_plot'>DEET_enrichment_plot</h2><span id='topic+DEET_enrichment_plot'></span>

<h3>Description</h3>

<p>Generate barplots or dotplots from the output of DEET
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_enrichment_plot(
  enrich_list,
  outname,
  width = 8,
  text_angle = 0,
  horizontal = FALSE,
  topn = 5,
  ol_size = 1,
  exclude_domain = "",
  cluster_order = NULL,
  dot = FALSE,
  colors = "Set2",
  split_domain = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_enrichment_plot_+3A_enrich_list">enrich_list</code></td>
<td>
<p>A list of enrichments from DEET, with each element
post-processed with the barplot enrichment function.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_outname">outname</code></td>
<td>
<p>A character giving the title of the barplot or dotplot.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_width">width</code></td>
<td>
<p>The number of inches in the barplot or dotplot.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_text_angle">text_angle</code></td>
<td>
<p>The angle of the enriched studies.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_horizontal">horizontal</code></td>
<td>
<p>Whether the output barplot is vertical or horizontal</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_topn">topn</code></td>
<td>
<p>the top number of studies (by p-value) to be plotted.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_ol_size">ol_size</code></td>
<td>
<p>the minimum number of overlapping genes (or paths) in an enriched study.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_exclude_domain">exclude_domain</code></td>
<td>
<p>Exclude studies enriched based on DEGs, Paths, or TF if
the user happened to aggregate the results into a single DF, generally unused.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_cluster_order">cluster_order</code></td>
<td>
<p>Factor to group studies based on the researchers custom annotation.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_dot">dot</code></td>
<td>
<p>logical (T/F) of whether to produce a dotplot or a barplot</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_colors">colors</code></td>
<td>
<p>Type of color pallete to input into 'scale_fill_brewer' of ggplot.</p>
</td></tr>
<tr><td><code id="DEET_enrichment_plot_+3A_split_domain">split_domain</code></td>
<td>
<p>logical (T/F) of whether to plot the &quot;topn&quot; studies for
each &quot;domain&quot; (default is source) or to plot the topn pathwys regardless of domain.
default is set to FALSE, meaning it plots the topn pathways regardless of domain.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object (barplot or dotplot) of enrichment identified within DEET.
</p>


<h3>Author(s)</h3>

<p>Dustin Sokolowski, Hauyun Hou PhD
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("example_DEET_enrich_input")
data("DEET_example_data")
DEET_out &lt;- DEET_enrich(example_DEET_enrich_input, DEET_dataset = DEET_example_data)

# converting output to format compatible with DEET_enrichment plot
DE_example &lt;- DEET_out$AP_DEET_DE_output$results
DE_example$term.name &lt;- DEET_out$AP_DEET_DE_output$metadata$DEET.Name
DE_example$domain &lt;- "DE"
DE_example$overlap.size &lt;- lengths(DE_example$overlap)
DE_example$p.value &lt;- DE_example$adjusted_p_val

DE_example_plot &lt;- DEET_enrichment_plot(list(DE_example = DE_example), "DE_example")


</code></pre>

<hr>
<h2 id='DEET_example_data'>DEET_example_data</h2><span id='topic+DEET_example_data'></span>

<h3>Description</h3>

<p>Named list of gene-sets and representative metadata for studies associated with Alizada et al., 2021
This example data is the exact same as what is needed to run DEET enrich properly but
subsetted to have 13 studies that are enriched by 'example_DEET_enrich_input'.
This way, the example gives an output at all levels of enrichment and at the correlation level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DEET_example_data)
</code></pre>


<h3>Format</h3>

<p>A named list of seven objects containing the data frames summarizing the DEGs from comparisons within DEET,
GMT objects of comparisons within DEET for enrichment through ActivePathways, GMT objects for basic pathway and TF enrichment,
and a dataframe for the metadata of each study.
</p>
<p>#' </p>

<dl>
<dt>DEET_DE</dt><dd><p> A list of data frames containing the significant DE genes, mean expression, log2fold-change, and padj from DESeq (padj &lt; 0.05).</p>
</dd>
<dt>DEET_gmt_BP</dt><dd><p> A list of class GMT, which is a list of studies where each study is populated by comparison id (internal DEET identifier), comparison name (interpretable comparison name), and a gene set. In this case the gene-set is the pathways that are enriched within that study.</p>
</dd>
<dt>DEET_gmt_TF</dt><dd><p> A list of class GMT, which is a list of studies where each study is populated by comparison id (internal DEET identifier), comparison name (interpretable comparison name), and a gene set. In this case the gene-set is the TFs that are enriched within that study.</p>
</dd>
<dt>DEET_gmt_DE</dt><dd><p> A list of class GMT, which is a list of studies where each study is populated by comparison id (internal DEET identifier), comparison name (interpretable comparison name), and a gene set. In this case the gene-set is the DEGs that are enriched within that study.</p>
</dd>
<dt>gmt_BP</dt><dd><p> A list of class GMT, which is a list of gene ontology gene-sets acquired from the bader lab 'http://download.baderlab.org/EM_Genesets/'</p>
</dd></dl>
<p>#'
</p>
<dl>
<dt>gmt_TF</dt><dd><p> A list of class GMT, which is a list of Transcription Factor gene-sets acquired from the bader lab 'http://download.baderlab.org/EM_Genesets/'</p>
</dd>
<dt>DEET_metadata</dt><dd><p> For every pairwise comparison, the study name, source (SRA, TCGA, GTEx and SRA-manual), description from the DRA compendium, the number of samples (total, up-condition, and down-condition), samples (total ,up-condition, down-condition), tissue (including tumour from TCGA), number of DEs (total, up-condition, down-condition), age (mean +- sd), sex, top 15 DEGs - up, top 15 DEGs - down, top 5 enriched pathways, and top 5 enriched TFs. PMID are also available for studies selected from SRA. Lastly, each pairwise comparison was given an overall category based on those decided in Crow et al., 2019. </p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(DEET_example_data)
</code></pre>

<hr>
<h2 id='DEET_feature_extract'>DEET_feature_extract</h2><span id='topic+DEET_feature_extract'></span>

<h3>Description</h3>

<p>Identify which genes are associated with pieces of
metadata that a researcher queries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_feature_extract(mat, response, datatype, detection_cutoff = 0.7)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_feature_extract_+3A_mat">mat</code></td>
<td>
<p>A gene-by-study matrix populated by the coefficients of that study.
By default, the coefficient is the log2Fold-change of genes as long as they are
differentially expressed (cutoff = padj &lt; 0.05).</p>
</td></tr>
<tr><td><code id="DEET_feature_extract_+3A_response">response</code></td>
<td>
<p>A vector (binomial, categorical, or continuous) that is used to
associated the DEGs within the studies.</p>
</td></tr>
<tr><td><code id="DEET_feature_extract_+3A_datatype">datatype</code></td>
<td>
<p>indication of whether the response variable is binomial,
categorical, or continuous.</p>
</td></tr>
<tr><td><code id="DEET_feature_extract_+3A_detection_cutoff">detection_cutoff</code></td>
<td>
<p>Proportion of studies where the gene is detected (not
as DE but detected at all, designated with a FC != 0). Default value 0.7.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list given the elastic net coefficients and the eleastic net
regression between the response variable and the DEGs within DEET.
It also outputs the correlation, ANOVA, and wilcoxon test of every gene
against the response variable based on if it's continuous, categorical, or
binomial in nature.
</p>

<ul>
<li><p> elastic_net_coefficients - Association that a gene has with the
response variable based on the elastic net regression.
</p>
</li>
<li><p> elastic_net - Output of the elastic net regression
</p>
</li>
<li><p> - basic_features gives the output of the
correlation, ANOVA, and wilcoxon test of every gene against the
response variable.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dustin Sokolowski, Jedid Ahn
</p>


<h3>References</h3>

<p>Engebretsen, S., &amp; Bohlin, J. (2019). Statistical predictions with glmnet.
Clinical epigenetics, 11(1), 1-3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(DEET_feature_extract_example_matrix)
data(DEET_feature_extract_example_response)
single1 &lt;- DEET_feature_extract(DEET_feature_extract_example_matrix,
DEET_feature_extract_example_response,"categorical")

</code></pre>

<hr>
<h2 id='DEET_feature_extract_example_matrix'>DEET_feature_extract_example_matrix</h2><span id='topic+DEET_feature_extract_example_matrix'></span>

<h3>Description</h3>

<p>An object of class data.frame where rows are genes and columns are comparisons.
The matrix is populated by the log2Fold-change of each gene within each study.
If the gene is not detected within that study, it is populated with 0 instead of the log2Fold-change.
This object is inputted into the 'mat' input variable for the 'DEET_feature_extract' function.
This example takes 1000 random genes and 200 random studies (seed = 1234s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DEET_feature_extract_example_matrix)
</code></pre>


<h3>Format</h3>

<p>An object of class data.frame where rows are genes and columns are comparisons (1000 randomly selected genes and 200 randomly selected studies).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEET_feature_extract_example_matrix)
</code></pre>

<hr>
<h2 id='DEET_feature_extract_example_response'>DEET_feature_extract_example_response</h2><span id='topic+DEET_feature_extract_example_response'></span>

<h3>Description</h3>

<p>Character vector giving the source (TCGA SRA, GTEx, SRA-manual) of 200 comparisons within DEET.
Used as the input for the 'response' input of 'DEET_feature_extract' in the example.
For this response variable to work, the 'datatype' input variable would also need
to be set to &quot;categorical&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(DEET_feature_extract_example_response)
</code></pre>


<h3>Format</h3>

<p>Character vector giving the source (TCGA SRA, GTEx, SRA-manual) of 200 comparisons within DEET.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DEET_feature_extract_example_response)
</code></pre>

<hr>
<h2 id='DEET_Input_as_Reference'>DEET_Input_as_Reference</h2><span id='topic+DEET_Input_as_Reference'></span>

<h3>Description</h3>

<p>Alternative function to DEET enrich for when the inputted gene
list is unordered. Here, we can increase the statistical rigour of enrichment
by levaraging the p-values of the DEGs within DEET. Specifically, the inputted
DE list is used as the reference and we test each DE list against your reference.
Specifically. We convert your reference into a gmt file before inputting each pairwise
DE list into ActivePathways. This function does not complete correlations or pathway-level
analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_Input_as_Reference(genes, DEET_dataset, background = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_Input_as_Reference_+3A_genes">genes</code></td>
<td>
<p>A character vector of gene symbols within 'DEET_dataset'</p>
</td></tr>
<tr><td><code id="DEET_Input_as_Reference_+3A_deet_dataset">DEET_dataset</code></td>
<td>
<p>The databank of the differential expression enrichment tool.
Appropriate inputs here are &quot;DEET_example_data&quot; stored within DEET, the &quot;DEET_combined.rda&quot; file
from the DEET stable repositoy found at X, and the DEET database developmental repository found at Y.
The DEET_dataset is a named list where details of it's structure can be found ?DEET_example_data.
Unlike in DEET_enrich, this dataset does not require the pathway-relevant elements
of the DEET_dataset list, namely &quot;gmt_BP&quot;, or &quot;gmt_TF&quot; &quot;DEET_gmt_BP&quot;,  &quot;DEET_gmt_TF&quot;.
It also does not need DEET_gmt_DE.</p>
</td></tr>
<tr><td><code id="DEET_Input_as_Reference_+3A_background">background</code></td>
<td>
<p>Character vector of human gene symbols showing all
possible genes. Default value is NULL and the background is generated
as all detected DEGs across any comparison.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list containing the ActivePathways enrichment of each comparison
on the user's inputted gene list, as well as the associated metadata of 
each enriched comparison.
</p>


<h3>Author(s)</h3>

<p>Dustin Sokolowski
</p>


<h3>References</h3>

<p>Paczkowska M, Barenboim J, Sintupisut N, et al. Integrative pathway
enrichment analysis of multivariate omics data. Nat Commun. 2020;11(1):735.
doi:10.1038/s41467-019-13983-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("example_DEET_enrich_input")
genes &lt;- rownames(example_DEET_enrich_input)
data("DEET_example_data")
DEET_out_ref &lt;- DEET_Input_as_Reference(genes, DEET_dataset = DEET_example_data)


</code></pre>

<hr>
<h2 id='DEET_plot_correlation'>DEET_plot_correlation</h2><span id='topic+DEET_plot_correlation'></span>

<h3>Description</h3>

<p>Take significant correlation outputs and generate
scatterplots of the genes DE in one or the other.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEET_plot_correlation(correlation_input)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEET_plot_correlation_+3A_correlation_input">correlation_input</code></td>
<td>
<p>The &quot;DE_correlations&quot; element of the
output of the DEET_enrich function. This function only works if
there is at least one significantly correlated study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list of ggplot objects with the correlation between
the input study and the study within DEET
</p>


<h3>Author(s)</h3>

<p>Dustin Sokolowski, Jedid Ahn
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("example_DEET_enrich_input")
data("DEET_example_data")
DEET_out &lt;- DEET_enrich(example_DEET_enrich_input, DEET_dataset = DEET_example_data)
correlation_input &lt;- DEET_out$DE_correlations
correlation_plots &lt;- DEET_plot_correlation(correlation_input)



</code></pre>

<hr>
<h2 id='example_DEET_enrich_input'>example_DEET_enrich_input</h2><span id='topic+example_DEET_enrich_input'></span>

<h3>Description</h3>

<p>Exon-level DEGs of HAoEC after TNFa treatment for 45 mins
from Alizada et al., 2021. Object is a data.frame with columns
&quot;gene_symbol&quot; &quot;padj&quot; and &quot;coef&quot;, which in this case is the
log2Fold-change of differential expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(example_DEET_enrich_input)
</code></pre>


<h3>Format</h3>

<p>A data frame with three columns. Rows are genes and it's
populated by the gene symbol, padj of gene expression, and coef (log2Fold-change).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_DEET_enrich_input)
</code></pre>

<hr>
<h2 id='proccess_and_plot_DEET_enrich'>proccess_and_plot_DEET_enrich</h2><span id='topic+proccess_and_plot_DEET_enrich'></span>

<h3>Description</h3>

<p>Generates barplots and dotplots based on the output of
the DEET_enrich function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proccess_and_plot_DEET_enrich(
  DEET_output,
  colour_barplot = "Source",
  width = 8,
  text_angle = 0,
  horizontal = F,
  topn = 5,
  ol_size = 1,
  exclude_domain = "",
  cluster_order = NULL,
  colors = "Set2"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_deet_output">DEET_output</code></td>
<td>
<p>Direct output of the DEET_enrich function.
A list with all of the same names as DEET_output.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_colour_barplot">colour_barplot</code></td>
<td>
<p>Pick dotplot or barplot colours.
It can be NULL, in which all bars are the same or it can
be a (case sensitive) column within the metadata. Defaults
to &quot;source&quot;.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_width">width</code></td>
<td>
<p>The number of inches in the barplot or dotplot.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_text_angle">text_angle</code></td>
<td>
<p>The angle of the enriched studies.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_horizontal">horizontal</code></td>
<td>
<p>Whether the output barplot is vertical or horizontal</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_topn">topn</code></td>
<td>
<p>the top number of studies (by p-value) to be plotted.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_ol_size">ol_size</code></td>
<td>
<p>the minimum number of overlapping genes (or paths) in an enriched study.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_exclude_domain">exclude_domain</code></td>
<td>
<p>Exclude studies enriched based on DEGs, Paths, or TF if
the user happened to aggregate the results into a single DF, generally unused.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_cluster_order">cluster_order</code></td>
<td>
<p>Factor to group studies based on the researchers custom annotation.</p>
</td></tr>
<tr><td><code id="proccess_and_plot_DEET_enrich_+3A_colors">colors</code></td>
<td>
<p>Type of color pallete to input into 'scale_fill_brewer' of ggplot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list where each element is a ggplot object plotting the output of
the enrichment tests within DEET. The final element is the output of ActivePathways
(in DEET) that is directly compatible with the DEET_enrichment_barplot function.
</p>

<ul>
<li><p> DEET_DotPlot - ggplot object of Dotplot of enrichment of enrichment of DEET studies
based on DE, BP, and TF information. Only plotted if 2/3 levels contain
at least one significant study.
</p>
</li>
<li><p> Pathway_barplot - ggplot object of Barplot of standard gene set enrichment based on
gene ontology and TFs. Only plotted if there is at least one enriched
significant pathway/TF.
</p>
</li>
<li><p> individual_barplot - ggplot object of Barplot of the top enriched pathways or studies
(depending on the input list).Barplot is only generated if each list has at least one pathway (or study)
is enriched.
</p>
</li>
<li><p> DEET_output_forplotting - output of Activepathways with &quot;domain&quot;, &quot;overlap.size&quot;, and &quot;p.value&quot;
columns added to be compatible with the DEET_enrichment_barplot function.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dustin Sokolowski, Hauyun Hou PhD
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("example_DEET_enrich_input")
data("DEET_example_data")
DEET_out &lt;- DEET_enrich(example_DEET_enrich_input, DEET_dataset = DEET_example_data)
plotting_example &lt;- proccess_and_plot_DEET_enrich(DEET_out, text_angle = 45,
horizontal = TRUE, topn=4)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
