<!DOCTYPE html><html><head><title>Help for package DEGRE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DEGRE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BarGraphDEGRE'><p>Create a bar plot showing the number of downregulated and upregulated genes.</p></a></li>
<li><a href='#DEGRE'><p>Infer the Differentially Expressed Genes</p></a></li>
<li><a href='#DEGRE_package'><p>DEGRE package</p></a></li>
<li><a href='#VolcanoDEGRE'><p>Create a volcano plot to visualize the proportion of downregulated and upregulated genes by applying a log2FC cutoff.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Inferring Differentially Expressed Genes using Generalized
Linear Mixed Models</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Genes that are differentially expressed between two or more experimental conditions can be detected in RNA-Seq. A high biological variability may impact the discovery of these genes once it may be divergent between the fixed effects. However, this variability can be covered by the random effects. 'DEGRE' was designed to identify the differentially expressed genes considering fixed and random effects on individuals. These effects are identified earlier in the experimental design matrix. 'DEGRE' has the implementation of preprocessing procedures to clean the near zero gene reads in the count matrix, normalize by 'RLE' published in the 'DESeq2' package, 'Love et al. (2014)' &lt;<a href="https://doi.org/10.1186%2Fs13059-014-0550-8">doi:10.1186/s13059-014-0550-8</a>&gt; and it fits a regression for each gene using the Generalized Linear Mixed Model with the negative binomial distribution, followed by a Wald test to assess the regression coefficients.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, parglm, glmmTMB, foreach, tibble, ggplot2, ggpubr,
ggrepel, car, dplyr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-01 15:28:38 UTC; tavinbio</td>
</tr>
<tr>
<td>Author:</td>
<td>Douglas Terra Machado
    <a href="https://orcid.org/0000-0002-6580-7628"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Otávio José Bernardes Brustolini
    <a href="https://orcid.org/0000-0001-8132-9753"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Yasmmin Côrtes Martins
    <a href="https://orcid.org/0000-0002-6830-1948"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Marco Antonio Grivet Mattoso Maia
    <a href="https://orcid.org/0000-0001-5130-095X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Ana Tereza Ribeiro de Vasconcelos
    <a href="https://orcid.org/0000-0002-4632-2086"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Douglas Terra Machado &lt;dougterra@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-02 09:32:57 UTC</td>
</tr>
</table>
<hr>
<h2 id='BarGraphDEGRE'>Create a bar plot showing the number of downregulated and upregulated genes.</h2><span id='topic+BarGraphDEGRE'></span>

<h3>Description</h3>

<p>Create a bar plot showing the number of downregulated and upregulated genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BarGraphDEGRE(
  results,
  log2FC_cutoff = 1,
  downregulated_color = "coral2",
  upregulated_color = "cornflowerblue",
  xlab = "Regulation",
  ylab = "Number of genes",
  font.x = 10,
  font.y = 10,
  font.tickslab = 10,
  legend_position = "right",
  legend.title = "Regulation"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BarGraphDEGRE_+3A_results">results</code></td>
<td>
<p>a data.frame object. It receives the output of the DEGRE function, filtered or not, as input.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_log2fc_cutoff">log2FC_cutoff</code></td>
<td>
<p>it stores the cutoff of the log2FoldChange. The default is 1.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_downregulated_color">downregulated_color</code></td>
<td>
<p>the bar color related to the number of downregulated genes. The default is &quot;coral2&quot;.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_upregulated_color">upregulated_color</code></td>
<td>
<p>the bar color related to the number of upregulated genes. The default is &quot;cornflowerblue&quot;.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_xlab">xlab</code></td>
<td>
<p>the x lab text. The default is &quot;Regulation&quot;.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_ylab">ylab</code></td>
<td>
<p>the y lab text. The default is &quot;Number of genes&quot;.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_font.x">font.x</code></td>
<td>
<p>the font size of the x axis. The default is 10.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_font.y">font.y</code></td>
<td>
<p>the font size of the y axis. The default is 10.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_font.tickslab">font.tickslab</code></td>
<td>
<p>the font size of the ticks lab. The default is 10.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_legend_position">legend_position</code></td>
<td>
<p>you need to specify here the position of the legend. The default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="BarGraphDEGRE_+3A_legend.title">legend.title</code></td>
<td>
<p>the title of the legend. The default is &quot;Regulation&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Reading a CSV file for an example:
dir &lt;- system.file("extdata", package = "DEGRE")
results_DEGRE_example &lt;- read.csv(file.path(dir,"results_DEGRE_example.csv"))
# Running the BarGraphDEGRE function
BarGraphDEGRE(results = results_DEGRE_example,
           log2FC_cutoff = 1,
           downregulated_color = "coral2",
           upregulated_color = "cornflowerblue",
           xlab = "Regulation",
           ylab = "Number of genes",
           font.x = 10,
           font.y = 10,
           font.tickslab = 10,
           legend_position = "right",
           legend.title = "Regulation")

</code></pre>

<hr>
<h2 id='DEGRE'>Infer the Differentially Expressed Genes</h2><span id='topic+DEGRE'></span>

<h3>Description</h3>

<p>The DEGRE function acts inferring the differentially expressed genes fitting them using generalized linear mixed models and then applies the Wald test to the regression coefficients. It offers Benjamini-Hochberg or Bonferroni P-values techniques to adjust the P-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEGRE(count_matrix, p_value_adjustment = "BH", formula, design_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DEGRE_+3A_count_matrix">count_matrix</code></td>
<td>
<p>a data.frame object. It receives the raw matrix as input.</p>
</td></tr>
<tr><td><code id="DEGRE_+3A_p_value_adjustment">p_value_adjustment</code></td>
<td>
<p>a data.frame object. It receives the experimental design matrix. The sample names must be identified in the first column. This matrix can also have more columns with information for the fixed and the random effects for the samples.</p>
</td></tr>
<tr><td><code id="DEGRE_+3A_formula">formula</code></td>
<td>
<p>it receives fixed and random effects descriptions.</p>
</td></tr>
<tr><td><code id="DEGRE_+3A_design_matrix">design_matrix</code></td>
<td>
<p>All the P-values computed must be corrected and the DEGRE package offers two possibilities: &quot;BH&quot; (Benjamini-Hochberg) correction (default) and &quot;BON&quot; (Bonferroni) correction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Reading the count matrix and the design matrix for an example:
dir &lt;- system.file("extdata", package = "DEGRE")
tab &lt;- read.csv(file.path(dir,"count_matrix_for_example.csv"))
des &lt;- read.csv(file.path(dir,"design_matrix_for_example.csv"))
# Running DEGRE function:
results &lt;- DEGRE(count_matrix = tab,
                 p_value_adjustment = "BH",
                 design_matrix = des,
                 formula = "condition + (1|sex)")

</code></pre>

<hr>
<h2 id='DEGRE_package'>DEGRE package</h2><span id='topic+DEGRE_package'></span>

<h3>Description</h3>

<p>Genes that are differentially expressed (DEGs) between two or more experimental conditions can be found in investigations using RNA-Seq data. The gene reads matrix is a count matrix that is used to infer DEGs, and fixed effects are often utilized in this inference. A high biological variability may impact the discovery of DEGs once it may be divergent between the fixed effects. This variability can be covered by the random effects. The DEGRE R package was designed to identify DEGs in RNA-Seq experiments considering fixed and the random effects on individuals. These effects are identified earlier in the experimental design matrix. DEGRE has the implementation of preprocessing procedures to clean the count matrix of the gene reads, and it fits a regression for each gene using the Generalized Linear Mixed Model with the negative binomial distribution,followed by a Wald test to test to assess the regression coefficients.
</p>


<h3>Author(s)</h3>

<p>Douglas Terra Machado <a href="mailto:dougterra@gmail.com">dougterra@gmail.com</a>, Otávio José Bernardes Brustolini, Yasmmin Côrtes Martins, Marco Antonio Grivet Mattoso Maia, Ana Tereza Ribeiro de Vasconcelos
</p>

<hr>
<h2 id='VolcanoDEGRE'>Create a volcano plot to visualize the proportion of downregulated and upregulated genes by applying a log2FC cutoff.</h2><span id='topic+VolcanoDEGRE'></span>

<h3>Description</h3>

<p>Create a volcano plot to visualize the proportion of downregulated and upregulated genes by applying a log2FC cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VolcanoDEGRE(
  results,
  log2FC_cutoff = 1,
  padj = 0.05,
  font.x = 10,
  font.y = 10,
  font.tickslab = 10,
  downregulated_color = "coral2",
  upregulated_color = "cornflowerblue",
  xlab = "log2Foldchange",
  ylab = "-log10(P-value)",
  legend_position = "right",
  legend.title = "Regulation"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VolcanoDEGRE_+3A_results">results</code></td>
<td>
<p>a data.frame object. It receives the output of the DEGRE function, filtered or not, as input.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_log2fc_cutoff">log2FC_cutoff</code></td>
<td>
<p>it stores the cutoff of the log2FoldChange. The default is 1.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_padj">padj</code></td>
<td>
<p>it stores the cutoff of the P-adjusted value (Q-value). The default is 0.05.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_font.x">font.x</code></td>
<td>
<p>the font size of the x axis. The default is 10.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_font.y">font.y</code></td>
<td>
<p>the font size of the y axis. The default is 10.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_font.tickslab">font.tickslab</code></td>
<td>
<p>the font size of the ticks lab. The default is 10.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_downregulated_color">downregulated_color</code></td>
<td>
<p>the colors of the downregulated genes. The default is &quot;coral2&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_upregulated_color">upregulated_color</code></td>
<td>
<p>the colors of the upregulated genes. The default is &quot;cornflowerblue&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_xlab">xlab</code></td>
<td>
<p>the x lab text. The default is &quot;log2Foldchange&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_ylab">ylab</code></td>
<td>
<p>the y lab text. The default is &quot;-log10(P-value)&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_legend_position">legend_position</code></td>
<td>
<p>you need to specify here the position of the legend. The default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VolcanoDEGRE_+3A_legend.title">legend.title</code></td>
<td>
<p>the title of the legend. The default is &quot;Regulation&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dir &lt;- system.file("extdata", package = "DEGRE")
results_DEGRE_example &lt;- read.csv(file.path(dir,"results_DEGRE_example.csv"))
# Running the VolcanoDEGRE function
VolcanoDEGRE(results = results_DEGRE_example,
          log2FC_cutoff = 1,
          padj = 0.05,
          font.x = 10,
          font.y = 10,
          font.tickslab = 10,
          downregulated_color = "coral2",
          upregulated_color = "cornflowerblue",
          xlab = "log2Foldchange",
          ylab = "-log10(P-value)",
          legend_position = "right",
          legend.title = "Regulation")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
