<!DOCTYPE html><html lang="en"><head><title>Help for package DEHOGT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DEHOGT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dehogt_func'><p>Differentially Expressed Heterogeneous Overdispersion Genes Testing for Count Data</p>
This script implements the main function of the proposed method in the above paper</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differentially Expressed Heterogeneous Overdispersion Gene Test
for Count Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.99.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a generalized linear model approach for detecting
    differentially expressed genes across treatment groups in count data. The
    package supports both quasi-Poisson and negative binomial models to handle
    over-dispersion, ensuring robust identification of differential expression.
    It allows for the inclusion of treatment effects and gene-wise covariates, 
    as well as normalization factors for accurate scaling across samples.
    Additionally, it incorporates statistical significance testing with
    options for p-value adjustment and log2 fold range thresholds,
    making it suitable for RNA-seq analysis as described in by 
    Xu et al., (2024) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0300565">doi:10.1371/journal.pone.0300565</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>doParallel, foreach, MASS,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, BiocStyle</td>
</tr>
<tr>
<td>biocViews:</td>
<td>GeneExpression, DifferentialExpression, StatisticalMethod,
Regression, Normalization</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ahshen26/DEHOGT">https://github.com/ahshen26/DEHOGT</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ahshen26/DEHOGT/issues">https://github.com/ahshen26/DEHOGT/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-11 15:00:01 UTC; arlinashen</td>
</tr>
<tr>
<td>Author:</td>
<td>Qi Xu [aut],
  Arlina Shen <a href="https://orcid.org/0009-0008-5330-6659"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre],
  Yubai Yuan [ctb],
  Annie Qu [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arlina Shen &lt;ahshen24@berkeley.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-13 18:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='dehogt_func'>Differentially Expressed Heterogeneous Overdispersion Genes Testing for Count Data
This script implements the main function of the proposed method in the above paper</h2><span id='topic+dehogt_func'></span>

<h3>Description</h3>

<p>Differentially Expressed Heterogeneous Overdispersion Genes Testing for Count Data
This script implements the main function of the proposed method in the above paper
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dehogt_func(
  data,
  treatment,
  norm_factors = NULL,
  covariates = NULL,
  dist = "qpois",
  padj = TRUE,
  pval_thre = 0.05,
  l2fc = FALSE,
  l2fc_thre = 1,
  num_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dehogt_func_+3A_data">data</code></td>
<td>
<p>A matrix of gene expression data where rows represent genes and columns represent samples.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_treatment">treatment</code></td>
<td>
<p>A vector specifying the treatment conditions for each sample.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_norm_factors">norm_factors</code></td>
<td>
<p>An optional vector of normalization factors for each sample. Default is NULL, which assumes equal normalization factors.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_covariates">covariates</code></td>
<td>
<p>An optional matrix of gene-wise covariates. Default is NULL.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_dist">dist</code></td>
<td>
<p>The distribution family for the GLM. Can be &quot;qpois&quot; for quasi-Poisson or &quot;negbin&quot; for negative binomial. Default is &quot;qpois&quot;.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_padj">padj</code></td>
<td>
<p>Logical value indicating whether to adjust p-values using the Benjamini-Hochberg (BH) procedure. Default is TRUE.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_pval_thre">pval_thre</code></td>
<td>
<p>The threshold for identifying differentially expressed genes based on adjusted p-values. Default is 0.05.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_l2fc">l2fc</code></td>
<td>
<p>Logical value indicating whether to consider log2 fold change for identifying differentially expressed genes. Default is FALSE.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_l2fc_thre">l2fc_thre</code></td>
<td>
<p>The threshold for log2 fold change in identifying differentially expressed genes. Default is 1.</p>
</td></tr>
<tr><td><code id="dehogt_func_+3A_num_cores">num_cores</code></td>
<td>
<p>The number of CPU cores to use for parallel computing. Default is 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table role = "presentation">
<tr><td><code>DE_idx</code></td>
<td>
<p>A logical vector indicating differentially expressed genes.</p>
</td></tr>
<tr><td><code>pvals</code></td>
<td>
<p>A numeric vector of p-values for each gene.</p>
</td></tr>
<tr><td><code>log2fc</code></td>
<td>
<p>A numeric vector of log2 fold changes for each gene.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># simulate gene expression data
data &lt;- matrix(rpois(1000, 10), nrow = 100, ncol = 10)
# simulate random treatment assignments
treatment &lt;- sample(0:1, 10, replace = TRUE)
# Run main function with parallel computing using 2 cores
result &lt;- dehogt_func(data, treatment, num_cores = 2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
