<!DOCTYPE html><html><head><title>Help for package deltaccd</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {deltaccd}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calcCCD'><p>Calculate clock correlation distance (CCD).</p></a></li>
<li><a href='#calcDeltaCCD'><p>Calculate delta clock correlation distance.</p></a></li>
<li><a href='#getRefCor'><p>Retrieve the reference correlation matrix for circadian gene co-expression.</p></a></li>
<li><a href='#GSE19188'><p>Gene expression data for GSE19188.</p></a></li>
<li><a href='#plotHeatmap'><p>Visualize gene co-expression.</p></a></li>
<li><a href='#plotRefHeatmap'><p>Visualize the reference pattern of gene co-expression.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quantify Rhythmic Gene Co-Expression Relative to a Reference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Infer progression of circadian rhythms in transcriptome data in
  which samples are not labeled with time of day and coverage of the circadian
  cycle may be incomplete. See Shilts et al. (2018) &lt;<a href="https://doi.org/10.7717%2Fpeerj.4327">doi:10.7717/peerj.4327</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://deltaccd.hugheylab.org">https://deltaccd.hugheylab.org</a>,
<a href="https://github.com/hugheylab/deltaccd">https://github.com/hugheylab/deltaccd</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>gzip</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>Imports:</td>
<td>doRNG (&ge; 1.6.6), foreach (&ge; 1.4.3), ggplot2 (&ge; 2.2.1),
rlang (&ge; 0.4.11), scales (&ge; 0.5.0), statmod (&ge; 1.4.30),
data.table (&ge; 1.14.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>doParallel (&ge; 1.0.11), knitr, rmarkdown, testthat (&ge;
3.0.0), vdiffr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-11 14:58:18 UTC; joshuaschoenbachler</td>
</tr>
<tr>
<td>Author:</td>
<td>Jake Hughey [aut, cre],
  Elliot Outland [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jake Hughey &lt;jakejhughey@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-11 19:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='calcCCD'>Calculate clock correlation distance (CCD).</h2><span id='topic+calcCCD'></span>

<h3>Description</h3>

<p>Quantify the similarity of gene co-expression between a reference and a test
dataset. Statistical significance is calculated using permutation of the
genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcCCD(
  refCor,
  emat,
  groupVec = NULL,
  refEmat = NULL,
  nPerm = 1000,
  geneNames = NULL,
  dopar = FALSE,
  scale = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcCCD_+3A_refcor">refCor</code></td>
<td>
<p>Correlation matrix to be used as the reference, such as comes
from <code><a href="#topic+getRefCor">getRefCor()</a></code>. Should contain Spearman correlation values.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_emat">emat</code></td>
<td>
<p>Matrix of expression values, where each row corresponds to a
gene and each column corresponds to a sample. The rownames and colnames of
<code>refCor</code> should be present in the rownames of <code>emat</code>. For the p-value
calculation, it is important that <code>emat</code> include all measured genes, not
just those in <code>refCor</code>.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_groupvec">groupVec</code></td>
<td>
<p>Optional vector indicating the group to which group each
sample belongs. If not provided, the function assumes all samples belong to
the same group.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_refemat">refEmat</code></td>
<td>
<p>Optional expression matrix for calculating co-expression for
the reference, with the same organization as <code>emat</code>. Only used if <code>refCor</code>
is not provided.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_nperm">nPerm</code></td>
<td>
<p>Number of permutations for assessing statistical significance.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_genenames">geneNames</code></td>
<td>
<p>Optional vector indicating a subset of genes in <code>refCor</code>,
<code>emat</code>, and/or <code>refEmat</code> to use for calculating the CCD.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to process features in parallel. Make
sure to register a parallel backend first.</p>
</td></tr>
<tr><td><code id="calcCCD_+3A_scale">scale</code></td>
<td>
<p>Logical indicating whether to scale CCD by the number of gene
pairs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns for group name, CCD, and p-value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRefCor">getRefCor()</a></code>, <code><a href="#topic+calcDeltaCCD">calcDeltaCCD()</a></code>, <code><a href="#topic+plotHeatmap">plotHeatmap()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(35813)

refCor = getRefCor()
ccdResult = calcCCD(refCor, GSE19188$emat, GSE19188$groupVec, nPerm = 100)

</code></pre>

<hr>
<h2 id='calcDeltaCCD'>Calculate delta clock correlation distance.</h2><span id='topic+calcDeltaCCD'></span>

<h3>Description</h3>

<p>Calculate the difference between the clock correlation distances (CCDs),
relative to a reference, for two groups of samples. Statistical significance
is calculated using permutation of the samples that belong to either of those
two groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcDeltaCCD(
  refCor,
  emat,
  groupVec,
  groupNormal,
  refEmat = NULL,
  nPerm = 1000,
  geneNames = NULL,
  dopar = FALSE,
  scale = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcDeltaCCD_+3A_refcor">refCor</code></td>
<td>
<p>Correlation matrix to be used as the reference, such as comes
from <code><a href="#topic+getRefCor">getRefCor()</a></code>. Should contain Spearman correlation values.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_emat">emat</code></td>
<td>
<p>Matrix of expression values, where each row corresponds to a gene
and each column corresponds to a sample. The rownames and colnames of
<code>refCor</code> should be present in the rownames of <code>emat</code>. For the p-value
calculation, it is important that <code>emat</code> include all measured genes, not
just those in <code>refCor</code>.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_groupvec">groupVec</code></td>
<td>
<p>Vector indicating the group to which group each sample
belongs. It's ok for groupVec to have more than two groups.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_groupnormal">groupNormal</code></td>
<td>
<p>Value indicating the group in groupVec that corresponds to
normal or healthy. Other groups will be compared to this group.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_refemat">refEmat</code></td>
<td>
<p>Optional expression matrix for calculating co-expression for
the reference, with the same organization as <code>emat</code>. Only used if <code>refCor</code>
is not provided.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_nperm">nPerm</code></td>
<td>
<p>Number of permutations for assessing statistical significance.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_genenames">geneNames</code></td>
<td>
<p>Optional vector indicating a subset of genes in <code>refCor</code>,
<code>emat</code>, and/or <code>refEmat</code> to use for calculating the CCD.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to process features in parallel. Make
sure to register a parallel backend first.</p>
</td></tr>
<tr><td><code id="calcDeltaCCD_+3A_scale">scale</code></td>
<td>
<p>Logical indicating whether to use scaled CCDs to calculate
difference.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns for group 1, group 2, deltaCCD, and
p-value. In each row, the deltaCCD is the CCD of group 2 minus the CCD of
group 1, so group 1 corresponds to <code>groupNormal</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRefCor">getRefCor()</a></code>, <code><a href="#topic+calcCCD">calcCCD()</a></code>, <code><a href="#topic+plotHeatmap">plotHeatmap()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(35813)

refCor = getRefCor()
deltaCcdResult = calcDeltaCCD(
  refCor, GSE19188$emat, GSE19188$groupVec, 'healthy', nPerm = 100)

</code></pre>

<hr>
<h2 id='getRefCor'>Retrieve the reference correlation matrix for circadian gene co-expression.</h2><span id='topic+getRefCor'></span>

<h3>Description</h3>

<p>The pan-tissue reference matrix is based on a fixed-effects meta-analysis of
eight circadian transcriptome datasets from mice, as described in
Shilts et al. 2018(doi: <a href="https://doi.org/10.7717/peerj.4327">10.7717/peerj.4327</a>). The human blood
reference matrix is based an analysis of three microarray datasets
(manuscript in preparation).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRefCor(
  species = c("human", "mouse"),
  tissue = c("pan", "blood"),
  useEntrezGeneId = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRefCor_+3A_species">species</code></td>
<td>
<p>Currently either 'human' or 'mouse'. Only affects the row and
column names of the correlation matrix, not the actual values.</p>
</td></tr>
<tr><td><code id="getRefCor_+3A_tissue">tissue</code></td>
<td>
<p>One of either 'pan' or 'blood'.</p>
</td></tr>
<tr><td><code id="getRefCor_+3A_useentrezgeneid">useEntrezGeneId</code></td>
<td>
<p>If <code>FALSE</code>, row and column names of correlation matrix
will correspond to gene symbols (e.g., PER2).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of Spearman correlation values.
</p>


<h3>See Also</h3>

<p><a href="#topic+GSE19188">GSE19188</a>, <code><a href="#topic+plotRefHeatmap">plotRefHeatmap()</a></code>, <code><a href="#topic+calcCCD">calcCCD()</a></code>, <code><a href="#topic+calcDeltaCCD">calcDeltaCCD()</a></code>
</p>

<hr>
<h2 id='GSE19188'>Gene expression data for GSE19188.</h2><span id='topic+GSE19188'></span>

<h3>Description</h3>

<p>Data of gene expression measured by microarray for samples from human
non-small cell lung cancer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE19188
</code></pre>


<h3>Format</h3>

<p>A list with two objects:
</p>

<dl>
<dt>emat</dt><dd><p>Matrix of normalized expression values, where each row
corresponds to a gene (rownames are Entrez Gene IDs) and each column
corresponds to a sample. To save space, genes have been downsampled.</p>
</dd>
<dt>groupVec</dt><dd><p>Vector of condition (tumor or healthy) for each sample.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19188">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19188</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRefCor">getRefCor()</a></code>, <code><a href="#topic+calcCCD">calcCCD()</a></code>, <code><a href="#topic+calcDeltaCCD">calcDeltaCCD()</a></code>
</p>

<hr>
<h2 id='plotHeatmap'>Visualize gene co-expression.</h2><span id='topic+plotHeatmap'></span>

<h3>Description</h3>

<p>Make heatmaps of the co-expression (Spearman correlation) between pairs of
selected genes in a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHeatmap(geneNames, emat, groupVec = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHeatmap_+3A_genenames">geneNames</code></td>
<td>
<p>Vector indicating the subset of genes in the rownames of
<code>emat</code> for which to calculate the correlations in expression.</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_emat">emat</code></td>
<td>
<p>Matrix of expression values, where each row corresponds to a
gene and each column corresponds to a sample. The elements of <code>geneNames</code>
should be present in the rownames of <code>emat</code>.</p>
</td></tr>
<tr><td><code id="plotHeatmap_+3A_groupvec">groupVec</code></td>
<td>
<p>Optional vector indicating the group to which group each
sample belongs. If not provided, the function assumes all samples belong
to the same group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code> object, which can be saved using <code><a href="ggplot2.html#topic+ggsave">ggplot2::ggsave()</a></code>.
Heatmap colors will be directly comparable to any heatmaps created by this
function or by <code><a href="#topic+plotRefHeatmap">plotRefHeatmap()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calcCCD">calcCCD()</a></code>, <code><a href="#topic+calcDeltaCCD">calcDeltaCCD()</a></code>, <code><a href="#topic+plotRefHeatmap">plotRefHeatmap()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>refCor = getRefCor()
pRef = plotRefHeatmap(refCor)
pTest = plotHeatmap(rownames(refCor), GSE19188$emat, GSE19188$groupVec)

</code></pre>

<hr>
<h2 id='plotRefHeatmap'>Visualize the reference pattern of gene co-expression.</h2><span id='topic+plotRefHeatmap'></span>

<h3>Description</h3>

<p>Make a heatmap of the reference correlation matrix for gene co-expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRefHeatmap(refCor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRefHeatmap_+3A_refcor">refCor</code></td>
<td>
<p>Correlation matrix, such as comes from <code><a href="#topic+getRefCor">getRefCor()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code> object, which can be saved using <code><a href="ggplot2.html#topic+ggsave">ggplot2::ggsave()</a></code>.
Heatmap colors will be directly comparable to any heatmaps created by this
function or by <code><a href="#topic+plotHeatmap">plotHeatmap()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRefCor">getRefCor()</a></code>, <code><a href="#topic+plotHeatmap">plotHeatmap()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>refCor = getRefCor()
pRef = plotRefHeatmap(refCor)
pTest = plotHeatmap(rownames(refCor), GSE19188$emat, GSE19188$groupVec)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
