<!DOCTYPE html><html><head><title>Help for package denovolyzeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {denovolyzeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#autismDeNovos'><p>de novo variants found in 1,078 autism trios</p></a></li>
<li><a href='#denovolyze'><p>Evaluates burden of <em>de novo</em> variation against expectation</p></a></li>
<li><a href='#denovolyzeMultiHits'><p>Determine significance of genes with multiple <em>de novos</em></p></a></li>
<li><a href='#denovolyzeR'><p>A package for the analysis of <em>de novo</em> sequencing variants</p></a></li>
<li><a href='#fmrpGenes'><p>FMRP genes</p></a></li>
<li><a href='#parseInput'><p>Checks input for errors</p></a></li>
<li><a href='#PermuteMultiHits'><p>Permutes x variants across a genelist, and counts genes with multiple hits</p></a></li>
<li><a href='#viewProbabilityTable'><p>Displays underlying <em>de novo</em> probability tables</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Statistical Analyses of De Novo Genetic Variants</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-08-01</td>
</tr>
<tr>
<td>Description:</td>
<td>An integrated toolset for the analysis of de novo (sporadic)
    genetic sequence variants. denovolyzeR implements a mutational model that
    estimates the probability of a de novo genetic variant arising in each human
    gene, from which one can infer the expected number of de novo variants in a
    given population size. Observed variant frequencies can then be compared against
    expectation in a Poisson framework. denovolyzeR provides a suite of functions
    to implement these analyses for the interpretation of de novo variation in human
    disease.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 0.3), reshape2 (&ge; 1.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://denovolyzeR.org">http://denovolyzeR.org</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="http://github.com/jamesware/denovolyzeR/issues">http://github.com/jamesware/denovolyzeR/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-08-01 12:18:10 UTC; jamesware</td>
</tr>
<tr>
<td>Author:</td>
<td>James Ware [aut, cre],
  Jason Homsy [ctb],
  Kaitlin Samocha [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>James Ware &lt;j.ware@imperial.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-08-01 14:55:41</td>
</tr>
</table>
<hr>
<h2 id='autismDeNovos'>de novo variants found in 1,078 autism trios</h2><span id='topic+autismDeNovos'></span>

<h3>Description</h3>

<p>de novo variants found in 1,078 autism trios, published in Nature Genetics(http://www.nature.com/doifinder/10.1038/ng.3050)
</p>


<h3>Format</h3>

<p>A data frame with 1096 obs of 2 variables:
</p>

<dl>
<dt>gene</dt><dd><p>Gene symbol of gene containing de novo variant</p>
</dd>
<dt>class</dt><dd><p>Functional class of variant:
&quot;syn&quot; = synonymous,
&quot;mis&quot; = missense,
&quot;non&quot; = nonsense,
&quot;splice&quot; = canonical splice site,
&quot;frameshift&quot; = frameshift indel</p>
</dd>
</dl>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/</a>
</p>

<hr>
<h2 id='denovolyze'>Evaluates burden of <em>de novo</em> variation against expectation</h2><span id='topic+denovolyze'></span><span id='topic+denovolyzeByClass'></span><span id='topic+denovolyzeByGene'></span>

<h3>Description</h3>

<p>Determines whether the test population carry more <em>de novo</em> variants
than expected. Variants may be grouped by variant class (e.g. are there more
LOF variants than expected, across the whole dataset?), or by gene (are there
more variants of a given class in SCN2A?).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>denovolyze(genes, classes, nsamples, groupBy = "class",
  includeGenes = "all", includeClasses = c("syn", "mis", "misD", "non",
  "stoploss", "startgain", "splice", "frameshift", "lof", "prot", "protD",
  "all"), geneId = "geneName", signifP = 3, roundExpected = 1,
  probTable = NULL, misD = NULL)

denovolyzeByClass(genes, classes, nsamples, groupBy = "class",
  includeGenes = "all", includeClasses = c("syn", "mis", "lof", "prot",
  "all"), geneId = "geneName", signifP = 3, roundExpected = 1,
  probTable = NULL)

denovolyzeByGene(genes, classes, nsamples, groupBy = "gene",
  includeGenes = "all", includeClasses = c("lof", "prot"),
  geneId = "geneName", signifP = 3, roundExpected = 1, probTable = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="denovolyze_+3A_genes">genes</code></td>
<td>
<p>A vector of genes containing de novo variants.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_classes">classes</code></td>
<td>
<p>A vector of classes of de novo variants.  Standard supported
classes are &quot;syn&quot; (synonymous), &quot;mis&quot; (missense), &quot;non&quot; (nonsense),
&quot;splice&quot; (splice), &quot;frameshift&quot; (frameshift) and &quot;lof&quot; (loss of function =
non + splice + frameshift). Additional classes that are supported by the
code, but are not included in the built-in probability tables, are
&quot;stoploss&quot;,&quot;startloss&quot;, &quot;misD&quot; (damaging missense).  These labels may be
used for user-supplied probability tables. If &quot;misD&quot; is present, then &quot;mis&quot;
(in the input) implies non-damaging missense.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_nsamples">nsamples</code></td>
<td>
<p>Number of individuals considered in de novo analysis.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_groupby">groupBy</code></td>
<td>
<p>Results can be tabulated by &quot;gene&quot;, or by variant &quot;class&quot;</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_includegenes">includeGenes</code></td>
<td>
<p>Genes to include in analysis. &quot;all&quot; or a vector of gene
names.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_includeclasses">includeClasses</code></td>
<td>
<p>Determines which variant classes are tabulated in
output.  In addition to the input classes, summaries can be produced for
&quot;prot&quot; (protein-altering = mis + lof), &quot;all&quot;, and &quot;protD&quot; (protein damaging
= misD + lof, only available if misD included in user-specified probability
table).  If &quot;misD&quot; is present, then &quot;mis&quot; will return statistics for all
missense.  Non-damaging missense are not analysed separately.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_geneid">geneId</code></td>
<td>
<p>Gene identifier used. One of &quot;hgncID&quot;, &quot;hgncSymbol&quot;,
&quot;enstID&quot;, &quot;ensgID&quot; or &quot;geneName&quot; (default, equals ensembl &quot;external_gene_name&quot;)</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_signifp">signifP</code></td>
<td>
<p>Number of significant figures used to round p-values in
output.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_roundexpected">roundExpected</code></td>
<td>
<p>Number of decimal places used to round expected burdens
in output.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_probtable">probTable</code></td>
<td>
<p>Probability table. A user-defined table of probabilities can
be provided here, to replace the probability table included in the package.</p>
</td></tr>
<tr><td><code id="denovolyze_+3A_misd">misD</code></td>
<td>
<p>If the user-specified probability table contains probabilities
for a sub-category of missense variants (e.g. predicted to be damaging by
an in silico algorithm), this column should be called misD, or the
alternative name should be specified here.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Analyses can be restricted to a subset of genes, and/or a subset of variant
classes
</p>
<p>See vignette(&quot;denovolyzeR_intro&quot;) for more information.
</p>


<h3>Value</h3>

<p>Returns a data frame
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>denovolyzeByClass</code>: denovolyzeByClass
</p>
</li>
<li> <p><code>denovolyzeByGene</code>: denovolyzeByGene
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>
### denovolyze

denovolyze(genes=autismDeNovos$gene,
           classes=autismDeNovos$class,
           nsamples=1078)

### denovolyzeByClass

denovolyzeByClass(genes=autismDeNovos$gene,
                  classes=autismDeNovos$class,
                  nsamples=1078)

# this convenience function is identical to:

denovolyze(genes=autismDeNovos$gene,
           classes=autismDeNovos$class,
           nsamples=1078,
           groupBy="class",
           includeClasses=c("syn","mis","lof","prot","all"),
           includeGenes="all"
           )

### denovolyzeByGene

denovolyzeByGene(genes=autismDeNovos$gene,
                 classes=autismDeNovos$class,
                 nsamples=1078)

# this is identical to:

denovolyze(genes=autismDeNovos$gene,
           classes=autismDeNovos$class,
           nsamples=1078,
           groupBy="gene",
           includeClasses=c("lof","prot"),
           includeGenes="all"
           )

</code></pre>

<hr>
<h2 id='denovolyzeMultiHits'>Determine significance of genes with multiple <em>de novos</em></h2><span id='topic+denovolyzeMultiHits'></span>

<h3>Description</h3>

<p>Are there more genes containing &gt;1 <em>de novos</em> than expected?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>denovolyzeMultiHits(genes, classes, nsamples, nperms = 100,
  includeGenes = "all", includeClasses = c("syn", "mis", "lof", "prot",
  "all"), nVars = "actual", geneId = "geneName", probTable = NULL,
  misD = NULL, signifP = 3, roundExpected = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="denovolyzeMultiHits_+3A_genes">genes</code></td>
<td>
<p>A vector of genes containing de novo variants.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_classes">classes</code></td>
<td>
<p>A vector of classes of de novo variants.  Standard supported
classes are &quot;syn&quot; (synonymous), &quot;mis&quot; (missense), &quot;non&quot; (nonsense),
&quot;splice&quot; (splice), &quot;frameshift&quot; (frameshift) and &quot;lof&quot; (loss of function =
non + splice + frameshift). Additional classes that are supported by the
code, but are not included in the built-in probability tables, are
&quot;stoploss&quot;,&quot;startloss&quot;, &quot;misD&quot; (damaging missense).  These labels may be
used for user-supplied probability tables. If &quot;misD&quot; is present, then &quot;mis&quot;
(in the input) implies non-damaging missense.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_nsamples">nsamples</code></td>
<td>
<p>Number of individuals considered in de novo analysis.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_nperms">nperms</code></td>
<td>
<p>Number of permutations</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_includegenes">includeGenes</code></td>
<td>
<p>Genes to include in analysis. &quot;all&quot; or a vector of gene
names.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_includeclasses">includeClasses</code></td>
<td>
<p>Determines which variant classes are tabulated in
output.  In addition to the input classes, summaries can be produced for
&quot;prot&quot; (protein-altering = mis + lof), &quot;all&quot;, and &quot;protD&quot; (protein damaging
= misD + lof, only available if misD included in user-specified probability
table).  If &quot;misD&quot; is present, then &quot;mis&quot; will return statistics for all
missense.  Non-damaging missense are not analysed separately.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_nvars">nVars</code></td>
<td>
<p>Select whether expected number of multihits is determined
by &quot;expected&quot; total number of variants , or &quot;actual&quot; total.  Actual
(default) is more conservative.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_geneid">geneId</code></td>
<td>
<p>Gene identifier used. One of &quot;hgncID&quot;, &quot;hgncSymbol&quot;,
&quot;enstID&quot;, &quot;ensgID&quot; or &quot;geneName&quot; (default, equals ensembl &quot;external_gene_name&quot;)</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_probtable">probTable</code></td>
<td>
<p>Probability table. A user-defined table of probabilities can
be provided here, to replace the probability table included in the package.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_misd">misD</code></td>
<td>
<p>If the user-specified probability table contains probabilities
for a sub-category of missense variants (e.g. predicted to be damaging by
an in silico algorithm), this column should be called misD, or the
alternative name should be specified here.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_signifp">signifP</code></td>
<td>
<p>Number of significant figures used to round p-values in
output.</p>
</td></tr>
<tr><td><code id="denovolyzeMultiHits_+3A_roundexpected">roundExpected</code></td>
<td>
<p>Number of decimal places used to round expected burdens
in output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See vignette (denovostats_intro) for more information.
</p>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>denovolyzeMultiHits(genes=autismDeNovos$gene,
                    classes=autismDeNovos$class,
                    nsamples=1078)

</code></pre>

<hr>
<h2 id='denovolyzeR'>A package for the analysis of <em>de novo</em> sequencing variants</h2><span id='topic+denovolyzeR'></span><span id='topic+denovolyzeR-package'></span>

<h3>Description</h3>

<p>A package for the analysis of <em>de novo</em> sequencing variants
</p>


<h3>Author(s)</h3>

<p>James Ware <a href="mailto:j.ware@imperial.ac.uk">j.ware@imperial.ac.uk</a>
</p>


<h3>References</h3>

<p><a href="http://github.com/jamesware/denovolyzeR">http://github.com/jamesware/denovolyzeR</a>
</p>

<hr>
<h2 id='fmrpGenes'>FMRP genes</h2><span id='topic+fmrpGenes'></span>

<h3>Description</h3>

<p>837 genes found to interact with the fragile X mental retardation protein (FMRP)
</p>


<h3>Format</h3>

<p>A vector of gene symbols</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/</a>
</p>
<p><a href="http://dx.doi.org/10.1016/j.cell.2011.06.013">http://dx.doi.org/10.1016/j.cell.2011.06.013</a>
</p>

<hr>
<h2 id='parseInput'>Checks input for errors</h2><span id='topic+parseInput'></span>

<h3>Description</h3>

<p>An internal function to check inputs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseInput(genes = genes, classes = classes, nsamples = nsamples,
  groupBy = groupBy, includeGenes = includeGenes,
  includeClasses = includeClasses, geneId = geneId, signifP = signifP,
  roundExpected = roundExpected, probTable = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parseInput_+3A_genes">genes</code></td>
<td>
<p>A vector of genes containing de novo variants.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_classes">classes</code></td>
<td>
<p>A vector of classes of de novo variants.  Standard supported
classes are &quot;syn&quot; (synonymous), &quot;mis&quot; (missense), &quot;non&quot; (nonsense),
&quot;splice&quot; (splice), &quot;frameshift&quot; (frameshift) and &quot;lof&quot; (loss of function =
non + splice + frameshift). Additional classes that are supported by the
code, but are not included in the built-in probability tables, are
&quot;stoploss&quot;,&quot;startloss&quot;, &quot;misD&quot; (damaging missense).  These labels may be
used for user-supplied probability tables. If &quot;misD&quot; is present, then &quot;mis&quot;
(in the input) implies non-damaging missense.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_nsamples">nsamples</code></td>
<td>
<p>Number of individuals considered in de novo analysis.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_groupby">groupBy</code></td>
<td>
<p>Results can be tabulated by &quot;gene&quot;, or by variant &quot;class&quot;</p>
</td></tr>
<tr><td><code id="parseInput_+3A_includegenes">includeGenes</code></td>
<td>
<p>Genes to include in analysis. &quot;all&quot; or a vector of gene
names.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_includeclasses">includeClasses</code></td>
<td>
<p>Determines which variant classes are tabulated in
output.  In addition to the input classes, summaries can be produced for
&quot;prot&quot; (protein-altering = mis + lof), &quot;all&quot;, and &quot;protD&quot; (protein damaging
= misD + lof, only available if misD included in user-specified probability
table).  If &quot;misD&quot; is present, then &quot;mis&quot; will return statistics for all
missense.  Non-damaging missense are not analysed separately.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_geneid">geneId</code></td>
<td>
<p>Gene identifier used. One of &quot;hgncID&quot;, &quot;hgncSymbol&quot;,
&quot;enstID&quot;, &quot;ensgID&quot; or &quot;geneName&quot; (default, equals ensembl &quot;external_gene_name&quot;)</p>
</td></tr>
<tr><td><code id="parseInput_+3A_signifp">signifP</code></td>
<td>
<p>Number of significant figures used to round p-values in
output.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_roundexpected">roundExpected</code></td>
<td>
<p>Number of decimal places used to round expected burdens
in output.</p>
</td></tr>
<tr><td><code id="parseInput_+3A_probtable">probTable</code></td>
<td>
<p>Probability table. A user-defined table of probabilities can
be provided here, to replace the probability table included in the package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>warning or error if any invalid input, else assigns variables back to
parent function
</p>

<hr>
<h2 id='PermuteMultiHits'>Permutes x variants across a genelist, and counts genes with multiple hits</h2><span id='topic+PermuteMultiHits'></span>

<h3>Description</h3>

<p>An internal function called by denovolyzeMultiHits
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PermuteMultiHits(x, y, nperms = 100, class = "lof", geneId = "geneName",
  includeGenes = "all", probTable = pDNM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PermuteMultiHits_+3A_x">x</code></td>
<td>
<p>Total number of de novo variants observed in dataset</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_y">y</code></td>
<td>
<p>Number of genes with &gt;1 de novo variant (of class &quot;class&quot;) in the population</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_nperms">nperms</code></td>
<td>
<p>Number permutations</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_class">class</code></td>
<td>
<p>In c(&quot;lof&quot;,&quot;mis&quot;,&quot;syn&quot;,&quot;prot&quot;)</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_geneid">geneId</code></td>
<td>
<p>Gene identifier used. One of &quot;hgncID&quot;, &quot;hgncSymbol&quot;,
&quot;enstID&quot;, &quot;ensgID&quot; or &quot;geneName&quot; (default, equals ensembl &quot;external_gene_name&quot;)</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_includegenes">includeGenes</code></td>
<td>
<p>Genes to include in analysis. &quot;all&quot; or a vector of gene
names.</p>
</td></tr>
<tr><td><code id="PermuteMultiHits_+3A_probtable">probTable</code></td>
<td>
<p>Probability table. A user-defined table of probabilities can
be provided here, to replace the probability table included in the package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named vector of 5 values
</p>


<h3>See Also</h3>

<p><code><a href="#topic+denovolyzeMultiHits">denovolyzeMultiHits</a></code>
</p>

<hr>
<h2 id='viewProbabilityTable'>Displays underlying <em>de novo</em> probability tables</h2><span id='topic+viewProbabilityTable'></span>

<h3>Description</h3>

<p>Tabulates probability of <em>de novo</em> variant for each protein-coding variant class, for each gene.  Values are probability of a <em>de novo</em> variant per chromosome per generation.  i.e. expected number of de novos for a given gene/class = <code class="reqn">p * 2 * nsamples</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viewProbabilityTable(format = "wide")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="viewProbabilityTable_+3A_format">format</code></td>
<td>
<p>option to display table in wide format (default; one line per gene), or long format</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
