<!DOCTYPE html><html><head><title>Help for package denoiSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {denoiSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#denoiseq'><p>Differential expression analysis using a bottom-up model</p></a></li>
<li><a href='#ERCC'><p>ERCC dataset</p></a></li>
<li><a href='#getSamplesOf'><p>Get posterior samples  of a parameter</p></a></li>
<li><a href='#readsData'><p>An S4 class to represent summarised counts and the output of Bayesian</p>
inference.</a></li>
<li><a href='#results'><p>Compute the test statistic</p></a></li>
<li><a href='#setInitValues'><p>Generic  for altering the  initValues slot</p></a></li>
<li><a href='#setOutput'><p>Generic  for the altering output slot.</p></a></li>
<li><a href='#setReplicates'><p>Generic  for the altering setReplicates slot.</p></a></li>
<li><a href='#setStepSizes'><p>Generic  for altering the stepSizes slot.</p></a></li>
<li><a href='#simdat'><p>simulated data</p></a></li>
<li><a href='#tunedStepSize'><p>Get  values of the tuned step sizes.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Expression Analysis Using a Bottom-Up Model</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Given count data from two conditions, it determines which transcripts are differentially expressed across the two conditions using Bayesian inference of the parameters of  a bottom-up model for PCR amplification. This model is  developed in Ndifon Wilfred, Hilah Gal, Eric Shifrut, Rina Aharoni, Nissan Yissachar, Nir Waysbort, Shlomit Reich Zeliger, Ruth Arnon, and Nir Friedman (2012), <a href="http://www.pnas.org/content/109/39/15865.full">http://www.pnas.org/content/109/39/15865.full</a>, and results in a distribution for the counts that is a superposition of the binomial and negative binomial distribution.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-31 20:03:47 UTC; buri</td>
</tr>
<tr>
<td>Author:</td>
<td>Gershom Buri [aut, cre],
  Wilfred Ndifon [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gershom Buri &lt;buri@aims.edu.gh&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-31 20:26:31 UTC</td>
</tr>
</table>
<hr>
<h2 id='denoiseq'>Differential expression analysis using a bottom-up model</h2><span id='topic+denoiseq'></span>

<h3>Description</h3>

<p>The denoiseq function perfoms default analysis by first normalising the
counts and then estimating the model parameters using Bayesian inference.
Size factors are estimated from count matrix and used for the normalisation.
The  Gibb's sampling algorithm is then used to sample from the joint
posterior  distribution of the model parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>denoiseq(RDobject, steps, tuningSteps = floor(steps/3))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="denoiseq_+3A_rdobject">RDobject</code></td>
<td>
<p>A readsData object.</p>
</td></tr>
<tr><td><code id="denoiseq_+3A_steps">steps</code></td>
<td>
<p>An integer representing the number of iterations.</p>
</td></tr>
<tr><td><code id="denoiseq_+3A_tuningsteps">tuningSteps</code></td>
<td>
<p>An integer representing the number of iterations to be
used for tuning the step sizes. Defaulted to a third of steps.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The denoiSeq package is based on a bottom-up model for PCR sequencing
developed by Ndifon et al. (2012). The model generates, in a bottom-up
manner, a probability distribution for the final copy number of a gene, that
is  a superposition of the negative binomial and the binomial distributions.
The derived distribution has three main parameters, i.e <em>N, p</em> and
<em>f</em>, which represent the initial gene amount before amplification,
the amplification efficiency and the dilution rate, respectively.
</p>
<p>Bayesian inference is used to estimate the model parameters. The counts in
each column are used to estimate the size factors (Anders and Huber, 2010)
which are in turn used to normalise the counts. For an <code class="reqn">m</code>  by <code class="reqn">n</code>
matrix, inference aims at estimating the three sets of parameters, i.e
<code class="reqn">p, f</code> and <code class="reqn">N_i</code> â€™s (2m in total because we are considering 2
conditions with the same <em>m</em> genes in each). denoiseq  uses the rows in
each condition to estimate parameter <code class="reqn">N_i</code> for each gene in that
condition, and uses the entire dataset, combined from both conditions,
to estimate <code class="reqn">p</code> and <code class="reqn">f</code>.
</p>
<p>For differential expression analysis, the primary parameters of interest are
<code class="reqn">N_{iA}</code> and <code class="reqn">N_{iB}</code> (from conditions A and B respectively), for
each gene <code class="reqn">i</code>.
</p>


<h3>Value</h3>

<p>The same readsData object but with a filled output slot. The output
slot now contains  2 lists, i.e <strong>samples</strong> which contains
posterior  samples  for each of the parameters  <code class="reqn">N_i</code>, <code class="reqn">p</code> and
<code class="reqn">f</code>, and  <strong>stepsize</strong> which  contains the tuned step sizes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#pre -filtering to remove lowly expressed genes
ERCC &lt;- ERCC[rowSums(ERCC)&gt;0, ]
RD &lt;- new('readsData', counts = ERCC)
steps &lt;- 30
#30 steps are used for illustration here. Atleast 5000 steps are adequate.
BI &lt;- denoiseq(RD, steps)

</code></pre>

<hr>
<h2 id='ERCC'>ERCC dataset</h2><span id='topic+ERCC'></span>

<h3>Description</h3>

<p>RNA-seq data from biological replicates of 3 cell lines. This dataset
contains a mixture of spike-in synthetic oligonucleotides that
are mixed into samples A and B at four mixing ratios: 1/2, 2/3, 1 and 4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ERCC
</code></pre>


<h3>Format</h3>

<p>A matrix with 92 rows and 10 columns:
</p>

<dl>
<dt>Conditions</dt><dd><p>There are 5 columns for each of the conditions A and B.</p>
</dd>
<dt>Transcripts</dt><dd><p>There are 92 distinct transcripts distinguishable by
their names.</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://bitbucket.org/soccin/seqc/src/ccd0502ef25422e83b3f208f50f8e252f62f17a3/data/?at=master">https://bitbucket.org/soccin/seqc/src/ccd0502ef25422e83b3f208f50f8e252f62f17a3/data/?at=master</a>
</p>

<hr>
<h2 id='getSamplesOf'>Get posterior samples  of a parameter</h2><span id='topic+getSamplesOf'></span>

<h3>Description</h3>

<p>Extracts posterior samples of individual parameters contained in the  output
slot of the readsData object returned by denoiSeq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSamplesOf(RDobject, parm, steps, condition = "A")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSamplesOf_+3A_rdobject">RDobject</code></td>
<td>
<p>A readsData object with a filled output slot.</p>
</td></tr>
<tr><td><code id="getSamplesOf_+3A_parm">parm</code></td>
<td>
<p>A parameter name string i.e  p, f or gene name.</p>
</td></tr>
<tr><td><code id="getSamplesOf_+3A_steps">steps</code></td>
<td>
<p>An integer representing number of iterations  used while calling
denoiseq.</p>
</td></tr>
<tr><td><code id="getSamplesOf_+3A_condition">condition</code></td>
<td>
<p>A character (either  A or B) representing the two
experimental conditions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector  of parameter samples, of length equal to steps.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#pre-filtering to remove lowly expressed genes
ERCC &lt;- ERCC[rowSums(ERCC)&gt;0, ]
RD &lt;- new('readsData', counts = ERCC)
steps &lt;- 30
#30 steps are just for illustration here. Atleast 5000 steps are adequate.
BI &lt;- denoiseq(RD, steps)
samples &lt;- getSamplesOf(BI, 'ERCC-00051', steps)
plot(samples, type='l', main = 'History plot of ERCC-00051')

</code></pre>

<hr>
<h2 id='readsData'>An S4 class to represent summarised counts and the output of Bayesian
inference.</h2><span id='topic+readsData'></span>

<h3>Description</h3>

<p>An S4 class to represent summarised counts and the output of Bayesian
inference.
</p>


<h3>Slots</h3>


<dl>
<dt><code>counts</code></dt><dd><p>A positive integer matrix containing summarised counts for each
genomic event (genes, exons, transcripts, etc)  in the two conditions, A and
B.</p>
</dd>
<dt><code>replicates</code></dt><dd><p>A list containing the indices of the columns that belong to
each  of the two experimental conditions, A and B. It is defaulted to A =
1:(n/2), B = (n/2+1):n for an <code class="reqn">m</code> by <code class="reqn">n</code> matrix.</p>
</dd>
<dt><code>geneNames</code></dt><dd><p>A character vector containing the names of the genomic event.
It is appropriately defaulted to names of  the matrix.</p>
</dd>
<dt><code>initValues</code></dt><dd><p>A list containing initial values for each parameter.
Defaulted to <code>N_A</code> = rep(1, nrow(counts)), <code>N_B</code> = rep(1,
nrow(counts)), <code>p</code>= 0.0001, <code>f</code> = 0.01.</p>
</dd>
<dt><code>stepSizes</code></dt><dd><p>A list containing step sizes for sampling each parameter.
Defaulted to <code>stepsizeN_A</code> = rep(1, nrow(counts)), <code>stepsizeN_B</code> =
rep(1, nrow(counts)), <code>stepsize_p</code>= 1e3, <code>stepsize_f</code> = 5e7</p>
</dd>
<dt><code>output</code></dt><dd><p>A list containing the  samples for each parameter which are
generated by Bayesian inference. It can only be  filled
inside the results function.</p>
</dd>
</dl>

<hr>
<h2 id='results'>Compute the test statistic</h2><span id='topic+results'></span>

<h3>Description</h3>

<p>Extracts  posterior samples of the  parameters which are returned by denoiseq
function and  computes the summary and test statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>results(RDobject, steps, burnin = floor(steps/3), rope_limit = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="results_+3A_rdobject">RDobject</code></td>
<td>
<p>A readsData object with a filled output slot.</p>
</td></tr>
<tr><td><code id="results_+3A_steps">steps</code></td>
<td>
<p>An integer representing the number of iterations.</p>
</td></tr>
<tr><td><code id="results_+3A_burnin">burnin</code></td>
<td>
<p>An integer for the number of iterations to be considered as
burn in values. A default value equivalent to a third of steps is used.</p>
</td></tr>
<tr><td><code id="results_+3A_rope_limit">rope_limit</code></td>
<td>
<p>A float that delimits the range of the region of practical
equivalence, ROPE. A default value of 0.5 is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To calculate the  test statistic, this function  first log2 transforms the
posterior samples of the two relevant parameters i.e <code class="reqn">N_{iA}</code> and
<code class="reqn">N_{iB}</code>. It then randomly subtracts posterior samples of one of the
parameters  from the other and determines the proportion of this
distribution of differences that lies in the region of practical equivalence
(ROPE) (Kruschke, 2011). The genes can then be arranged in an ascending
order of  the ROPE_propn column and we can select the most differentially
expressed genes as those whose ROPE_propn is less than a particular
threshold value.
</p>
<p>Using both real and  simulated data, optimal values between 0.0007 and 0.4
were obtained for the threshold.
</p>


<h3>Value</h3>

<p>A dataframe with 3 columns namely; the log2 fold change (log2FC),
the standard error of the log2 fold change (lgfcSE) and the test static
(ROPE_propn).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#pre -filtering to remove lowly expressed genes
ERCC &lt;- ERCC[rowSums(ERCC) &gt; 0, ]
RD &lt;- new('readsData', counts = ERCC)
steps &lt;- 30
#30 steps are just for illustration here. At least 5000 steps are adequate.
BI &lt;- denoiseq(RD, steps)
rez &lt;- results(BI, steps)
head(rez)

#Re-ordering according to most differentially expressed
rez &lt;- rez[with(rez, order( ROPE_propn)), ]
head(rez, 10)

#Determine significant genes using a threshold of 0.38.
sgf &lt;- rez[rez$ROPE_propn&lt;0.38, ]
head(sgf)
dim(sgf)
</code></pre>

<hr>
<h2 id='setInitValues'>Generic  for altering the  initValues slot</h2><span id='topic+setInitValues'></span><span id='topic+setInitValues+2CreadsData-method'></span>

<h3>Description</h3>

<p>Updates the value of the initValues slot for the readsData object supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setInitValues(object, initval)

## S4 method for signature 'readsData'
setInitValues(object, initval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setInitValues_+3A_object">object</code></td>
<td>
<p>a readsData object</p>
</td></tr>
<tr><td><code id="setInitValues_+3A_initval">initval</code></td>
<td>
<p>A list of initial values for each of the  parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same readsData object with the initValues slot updated.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>readsData</code>: Alters the value of the initValues slot of
a readsData object.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>RD &lt;- new("readsData", counts = ERCC)
initvals &lt;- list(N_A = rep(2, 92), N_B = rep(1.5, 92), p = 0.0005, f = 0.03)
RD &lt;- setInitValues(RD, initvals)
RD@initValues
</code></pre>

<hr>
<h2 id='setOutput'>Generic  for the altering output slot.</h2><span id='topic+setOutput'></span><span id='topic+setOutput+2CreadsData-method'></span>

<h3>Description</h3>

<p>Updates the value of the output slot for the readsData object supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setOutput(object, outval)

## S4 method for signature 'readsData'
setOutput(object, outval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setOutput_+3A_object">object</code></td>
<td>
<p>a readsData object.</p>
</td></tr>
<tr><td><code id="setOutput_+3A_outval">outval</code></td>
<td>
<p>A list of the output from Bayesian inference.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same readsData object with the output slot updated.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>readsData</code>: Alters the value of the output slot of
the readsData object.
</p>
</li></ul>

<hr>
<h2 id='setReplicates'>Generic  for the altering setReplicates slot.</h2><span id='topic+setReplicates'></span><span id='topic+setReplicates+2CreadsData-method'></span>

<h3>Description</h3>

<p>Updates the value of the replicates slot for the readsData object supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setReplicates(object, repsval)

## S4 method for signature 'readsData'
setReplicates(object, repsval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setReplicates_+3A_object">object</code></td>
<td>
<p>a readsData object</p>
</td></tr>
<tr><td><code id="setReplicates_+3A_repsval">repsval</code></td>
<td>
<p>A list  of column indices for the samples in each condition.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same readsData object with the replicates slot updated.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>readsData</code>: Alters the value of the replicates slot of
a readsData object.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>RD &lt;- new("readsData", counts = ERCC)
reps &lt;- list(A = c(2,4,5,3,10),B = c(9,7,1,8,6))
RD &lt;- setReplicates(RD, reps)
RD@replicates

</code></pre>

<hr>
<h2 id='setStepSizes'>Generic  for altering the stepSizes slot.</h2><span id='topic+setStepSizes'></span><span id='topic+setStepSizes+2CreadsData-method'></span>

<h3>Description</h3>

<p>Updates the value of the stepSizes slot for the readsData object supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setStepSizes(object, stepSizesval)

## S4 method for signature 'readsData'
setStepSizes(object, stepSizesval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setStepSizes_+3A_object">object</code></td>
<td>
<p>a readsData object</p>
</td></tr>
<tr><td><code id="setStepSizes_+3A_stepsizesval">stepSizesval</code></td>
<td>
<p>A list of step sizes for each of  the  parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same readsData object with the stepSizes slot updated.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>readsData</code>: Alters the value of the stepSizes slot of
a readsData object.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>RD &lt;- new("readsData", counts = ERCC)
ss &lt;- list(N_A = rep(2,  92), N_B = rep(1.5, 92), p = 3e5, f = 3.5e7)
RD &lt;- setStepSizes(RD, ss)
RD@stepSizes
</code></pre>

<hr>
<h2 id='simdat'>simulated data</h2><span id='topic+simdat'></span>

<h3>Description</h3>

<p>A dataset containing simulated data based on parameter values N = 1,2,...,50
, p = 0.0017  and f =0.1,0.2,...,0.5. The values of N were repeated 15 times
to generate  750 genes.  This dataset  contains 750 observational genes
with 5 experimental samples for each condition,  summarised as a 750 by 10
integer matrix. The first 428 genes are not differentially expressed
between the two conditions whereas the last 322 genes are. The gene counts
were generated in accordance to the probability distribution derived in
Ndifon et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simdat
</code></pre>


<h3>Format</h3>

<p>A matrix with 750 rows and 10 columns:
</p>

<dl>
<dt>Conditions</dt><dd><p>There are 5 columns for each of the conditions A and B.</p>
</dd>
<dt>Transcripts</dt><dd><p>There are 750 distinct genes without  names.</p>
</dd>
</dl>

<hr>
<h2 id='tunedStepSize'>Get  values of the tuned step sizes.</h2><span id='topic+tunedStepSize'></span>

<h3>Description</h3>

<p>Extracts the tuned step sizes for sampling each parameter from  the return
value of  denoiseq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tunedStepSize(RDobject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tunedStepSize_+3A_rdobject">RDobject</code></td>
<td>
<p>A readsData object with a filled output slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the tuned step sizes for sampling each of the parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#pre -filtering to remove lowly expressed genes
ERCC &lt;- ERCC[rowSums(ERCC)&gt;0, ]
RD &lt;- new('readsData', counts = ERCC)
steps &lt;- 30
#30 steps are just for illustration here. Atleast 5000 steps are adequate.
BI &lt;- denoiseq(RD, steps)
tunedStepSize(BI)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
