<!DOCTYPE html><html lang="en"><head><title>Help for package DepLogo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DepLogo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Alphabet'><p>built alphabet</p></a></li>
<li><a href='#alphabet.dna'><p>DNA alphabet</p></a></li>
<li><a href='#alphabet.dna.gap'><p>DNA alphabet with gaps</p></a></li>
<li><a href='#alphabet.protein'><p>Amino acid alphabet</p></a></li>
<li><a href='#alphabet.protein.gap'><p>Amino acid alphabet with gaps</p></a></li>
<li><a href='#alphabet.rna'><p>RNA alphabet</p></a></li>
<li><a href='#alphabet.rna.gap'><p>RNA alphabet with gaps</p></a></li>
<li><a href='#colorchart'><p>Plots a colorchart representation of a set of sequences</p></a></li>
<li><a href='#deprects'><p>Rectangles of averaged colors</p></a></li>
<li><a href='#DLData'><p>Create <code>DLData</code> object</p></a></li>
<li><a href='#filter.by.conservation'><p>Filters columns (sequence positions) by conservation</p></a></li>
<li><a href='#filter.by.dependencies'><p>Filters columns (sequence positions) by dependency</p></a></li>
<li><a href='#filter.by.gaps'><p>Filters columns (sequence positions) by gaps</p></a></li>
<li><a href='#filterColumns'><p>Filters data columns by some filter function</p></a></li>
<li><a href='#getDeps'><p>Compute dependencies between positions</p></a></li>
<li><a href='#getPWM'><p>Position weight matrix from DLData object</p></a></li>
<li><a href='#logo'><p>Sequence logo</p></a></li>
<li><a href='#partition'><p>Paritions data by most inter-dependent positions</p></a></li>
<li><a href='#plotBlocks'><p>Plots blocks of data</p></a></li>
<li><a href='#plotDeparcs'><p>Plots a graph representation of dependency values</p></a></li>
<li><a href='#plotDeplogo'><p>Plot a dependency logo</p></a></li>
<li><a href='#plotDepmatrix'><p>Plots a matrix representation of dependency values</p></a></li>
<li><a href='#replaceColors'><p>Replaces colors in DLData object</p></a></li>
<li><a href='#revcom'><p>Reverse complement</p></a></li>
<li><a href='#subBoxes'><p>Plots weights as boxplots</p></a></li>
<li><a href='#subLines'><p>Plots weights as lines</p></a></li>
<li><a href='#suggestColors'><p>Suggests colors for symbols</p></a></li>
<li><a href='#summary.DLData'><p>Summarizing DLData objects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Dependency Logo</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jan Grau &lt;grau@informatik.uni-halle.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Plots dependency logos from a set of aligned input sequences.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-15 16:21:10 UTC; dev</td>
</tr>
<tr>
<td>Author:</td>
<td>Jan Grau [aut, cre],
  Jens Keilwagen [aut],
  Martin Nettling [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-15 17:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Alphabet'>built alphabet</h2><span id='topic+Alphabet'></span>

<h3>Description</h3>

<p>builts an object of class Alphabet from the given set of symbols and colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Alphabet(chars, cols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Alphabet_+3A_chars">chars</code></td>
<td>
<p>set of symbols</p>
</td></tr>
<tr><td><code id="Alphabet_+3A_cols">cols</code></td>
<td>
<p>set of colors; one for each symbol</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Alphabet object
</p>


<h3>Author(s)</h3>

<p>Martin Nettling
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DNA &lt;- Alphabet(c("A", "C", "G", "T"), c("green4", "blue", "orange", "red"))
</code></pre>

<hr>
<h2 id='alphabet.dna'>DNA alphabet</h2><span id='topic+alphabet.dna'></span>

<h3>Description</h3>

<p>DNA alphabet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.dna
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='alphabet.dna.gap'>DNA alphabet with gaps</h2><span id='topic+alphabet.dna.gap'></span>

<h3>Description</h3>

<p>DNA alphabet with gaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.dna.gap
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='alphabet.protein'>Amino acid alphabet</h2><span id='topic+alphabet.protein'></span>

<h3>Description</h3>

<p>Amino acid alphabet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.protein
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='alphabet.protein.gap'>Amino acid alphabet with gaps</h2><span id='topic+alphabet.protein.gap'></span>

<h3>Description</h3>

<p>Amino acid alphabet with gaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.protein.gap
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='alphabet.rna'>RNA alphabet</h2><span id='topic+alphabet.rna'></span>

<h3>Description</h3>

<p>RNA alphabet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.rna
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='alphabet.rna.gap'>RNA alphabet with gaps</h2><span id='topic+alphabet.rna.gap'></span>

<h3>Description</h3>

<p>RNA alphabet with gaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphabet.rna.gap
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 2.
</p>

<hr>
<h2 id='colorchart'>Plots a colorchart representation of a set of sequences</h2><span id='topic+colorchart'></span>

<h3>Description</h3>

<p>This function is a low-level plotting function (using <a href="graphics.html#topic+image">image</a> with <code>add=TRUE</code>, internally).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colorchart(part, yoff, ic.scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colorchart_+3A_part">part</code></td>
<td>
<p>the set of sequences as <a href="#topic+DLData">DLData</a> object</p>
</td></tr>
<tr><td><code id="colorchart_+3A_yoff">yoff</code></td>
<td>
<p>the offset in y-direction within the current plot</p>
</td></tr>
<tr><td><code id="colorchart_+3A_ic.scale">ic.scale</code></td>
<td>
<p>ignored for colorcharts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the vertical (y) offset after this plot
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[,1], weights = log1p(seqs[, 2]) )

# create high-level plot
plot(NULL, xlim = c(1, ncol(data$data) - 1), ylim = c(0, nrow(data$data)), 
    ylab = nrow(data$data), axes = FALSE)
# and add colorchart and axis
colorchart(data, yoff = nrow(data$data))
axis(1)
</code></pre>

<hr>
<h2 id='deprects'>Rectangles of averaged colors</h2><span id='topic+deprects'></span>

<h3>Description</h3>

<p>Plots a representation of a set of sequences by rectangles of (scaled) averaged color values of the symbols at each position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deprects(part, yoff, ic.scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deprects_+3A_part">part</code></td>
<td>
<p>the set of sequences as <a href="#topic+DLData">DLData</a> object</p>
</td></tr>
<tr><td><code id="deprects_+3A_yoff">yoff</code></td>
<td>
<p>the offset in y-direction within the current plot</p>
</td></tr>
<tr><td><code id="deprects_+3A_ic.scale">ic.scale</code></td>
<td>
<p>if <code>TRUE</code>, alpha values of colors will be assigned based on &quot;information content&quot; of the distribution at each position</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a low-level plotting function (using <a href="graphics.html#topic+rect">rect</a>, internally).
</p>


<h3>Value</h3>

<p>the vertical (y) offset after this plot
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"),  
   stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1],weights = log1p(seqs[, 2]) )

# create high-level plot
plot(NULL, xlim = c(1, ncol(data$data) - 1), ylim = c(0, nrow(data$data)), 
    ylab = nrow(data$data), axes = FALSE)
# and add deprects and axis
deprects(data, yoff = nrow(data$data))
axis(1)
</code></pre>

<hr>
<h2 id='DLData'>Create <code>DLData</code> object</h2><span id='topic+DLData'></span>

<h3>Description</h3>

<p>Creates a new <code>DLData</code> object from a set of input sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DLData(
  sequences,
  weights = NULL,
  symbols = alphabet.dna$alphabet,
  colors = alphabet.dna$colors,
  delim = "",
  sortByWeights = !is.null(weights),
  axis.labels = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DLData_+3A_sequences">sequences</code></td>
<td>
<p>the input sequences, may be provided as i) <code>character</code>
vector or ii) a <code>data.frame</code> with sequences organized in rows and one 
symbol per column</p>
</td></tr>
<tr><td><code id="DLData_+3A_weights">weights</code></td>
<td>
<p>weights associated with the sequences, numeric vector of the 
same length as <code>sequences</code> has sequences</p>
</td></tr>
<tr><td><code id="DLData_+3A_symbols">symbols</code></td>
<td>
<p>the symbols (alphabet) over which the sequences are defined</p>
</td></tr>
<tr><td><code id="DLData_+3A_colors">colors</code></td>
<td>
<p>colors for each of the <code>symbols</code>, not necessarily unique</p>
</td></tr>
<tr><td><code id="DLData_+3A_delim">delim</code></td>
<td>
<p>delimiter between the symbols in the input sequences, ignored if
<code>sequences</code> as a <code>data.frame</code></p>
</td></tr>
<tr><td><code id="DLData_+3A_sortbyweights">sortByWeights</code></td>
<td>
<p>if <code>TRUE</code>, <code>sequences</code> will be ordered by 
their <code>weight</code> in decreasing order</p>
</td></tr>
<tr><td><code id="DLData_+3A_axis.labels">axis.labels</code></td>
<td>
<p>the labels of the individual sequence positions; 
if <code>NULL</code>, indexes from 1 to to total number of positions will be used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sequences may either be provided as a <code>character</code> vector or as a
<code>data.frame</code>. All symbols occurring in these sequences need to be
defined and assigned to colors, which are used for plotting later. Colors do
not need to be unique, but symbols with identical colors may become
indistinguishable in subsequent plots (which might even be desired, for
instance, when visualizing protein properties instead of amino acids). 
Sequences may have an associated weight, which is used to order sequences,
e.g., for creating chunks/blocks of sequences in subsequent plots (see
<code>chunks</code> parameter of <code>plotDeplogo</code>).
</p>


<h3>Value</h3>

<p>the <code>DLData</code> object
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+plotDeplogo">plotDeplogo</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># creating a DLData object using default (DNA) alphabet and colors
# from a character vector with two entries
data &lt;- DLData(c("ACGT", "ATTA"))

# creating a DLData object using a custom, binary alphabet and custom colors
data2 &lt;- DLData(c("A,B,B,A,B", "A,B,B,A,A", "A,B,A,A,B"),
    symbols = c("A", "B"), colors = c("red","green"), delim = ",")

# creating a DLData object from a data frame 
# (created from a character vector, in this case)
vec &lt;- c("A,B,B,A,B", "A,B,B,A,A", "A,B,A,A,B")
df &lt;- as.data.frame(t(sapply(vec, function(a){strsplit(a, ",")[[1]]})))
data.df &lt;- DLData(df, symbols = c("A", "B"), colors = c("red", "green"))

# creating a DLData object from sequences and weights, read from a tabular file
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data3 &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )
</code></pre>

<hr>
<h2 id='filter.by.conservation'>Filters columns (sequence positions) by conservation</h2><span id='topic+filter.by.conservation'></span>

<h3>Description</h3>

<p>Filters columns based on the relative information content of each column
which is the standard information content normalized to the interval [0,1],
where 0 corresponds to uniform distribution and 1 to perfect conservation
of one nucleotide or amino acid, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter.by.conservation(relative.ic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter.by.conservation_+3A_relative.ic">relative.ic</code></td>
<td>
<p>the maximum relative information content allowed
to retain a position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>function that, given a <a href="#topic+DLData">DLData</a> object, returns <code>TRUE</code> 
for every column that does not exceed the specified relative information content
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fun &lt;- filter.by.conservation(relative.ic = 0.9)
</code></pre>

<hr>
<h2 id='filter.by.dependencies'>Filters columns (sequence positions) by dependency</h2><span id='topic+filter.by.dependencies'></span>

<h3>Description</h3>

<p>Filters columns based on the average or maximum mutual information of a column
to all other columns. Mutual information is normalized to to interval
[0,1], where 0 corresponds to independence and 1 to perfect dependence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter.by.dependencies(mi.threshold, use.max = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter.by.dependencies_+3A_mi.threshold">mi.threshold</code></td>
<td>
<p>the minimum average or maximum mutual information required</p>
</td></tr>
<tr><td><code id="filter.by.dependencies_+3A_use.max">use.max</code></td>
<td>
<p>if <code>TRUE</code>, the maximum and otherwise the average mutual
information will be considered</p>
</td></tr>
</table>


<h3>Value</h3>

<p>function that, given a <a href="#topic+DLData">DLData</a> object, returns <code>TRUE</code> 
for every column that does exceed the specified average mutual information
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fun &lt;- filter.by.dependencies(mi.threshold = 0.3)
</code></pre>

<hr>
<h2 id='filter.by.gaps'>Filters columns (sequence positions) by gaps</h2><span id='topic+filter.by.gaps'></span>

<h3>Description</h3>

<p>Filters columns (sequence positions) by gaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter.by.gaps(percent.gap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter.by.gaps_+3A_percent.gap">percent.gap</code></td>
<td>
<p>the maximum fraction of gaps allowed to retain a column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>function that, given a <a href="#topic+DLData">DLData</a> object, returns <code>TRUE</code> 
for every column that does not exceed the specified number of gaps
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fun &lt;- filter.by.gaps(percent.gap = 0.1)
</code></pre>

<hr>
<h2 id='filterColumns'>Filters data columns by some filter function</h2><span id='topic+filterColumns'></span>

<h3>Description</h3>

<p>Filters the columns of the input data, i.e., positions of input sequences,
by a filter function that, given a <a href="#topic+DLData">DLData</a> object, returns a list containing
i) as element <code>$selected</code> a vector with entries <code>TRUE</code> for every column 
that should be retained in the filtered data and ii) as element <code>$range</code> the
range of values obtained for the filtering criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterColumns(data, filter.fun)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterColumns_+3A_data">data</code></td>
<td>
<p>the data as <a href="#topic+DLData">DLData</a> object</p>
</td></tr>
<tr><td><code id="filterColumns_+3A_filter.fun">filter.fun</code></td>
<td>
<p>the filter function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+DLData">DLData</a> object containing the filtered columns and the indexes of the remaining in its <code>axis.labels</code> field
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+filter.by.gaps">filter.by.gaps</a>
</p>
<p><a href="#topic+filter.by.dependencies">filter.by.dependencies</a>
</p>
<p><a href="#topic+filter.by.conservation">filter.by.conservation</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

# create a filter function based on the percentage of gap symbols (at most 10%)
fun &lt;- filter.by.gaps(percent.gap = 0.1)
data2 &lt;- filterColumns(data, fun)
</code></pre>

<hr>
<h2 id='getDeps'>Compute dependencies between positions</h2><span id='topic+getDeps'></span><span id='topic+getDeps.DLData'></span><span id='topic+getDeps.data.frame'></span>

<h3>Description</h3>

<p>Computes the dependencies (as measures by mutual information) between all
positions (columns) of discrete data. Specifically, it returns for each pair
of positions (i,j) the mutual information I(X_i,X_j) multiplied by the number
N of sequences (rows), which may also be used for testing the statistical
significance of mutual information values, as for large N, 2*N*I(X_i,X_j) is
approximately chi squared.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDeps(data, ...)

## S3 method for class 'DLData'
getDeps(data, ...)

## S3 method for class 'data.frame'
getDeps(data, alphabet, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDeps_+3A_data">data</code></td>
<td>
<p>the data for computing mutual information. Either a DLData object
or a data.frame; In the latter case, the symbols of the alphabet must be
provided as a second parameter</p>
</td></tr>
<tr><td><code id="getDeps_+3A_...">...</code></td>
<td>
<p>the symbols of the alphabet as character vector, only if data is a
data.frame</p>
</td></tr>
<tr><td><code id="getDeps_+3A_alphabet">alphabet</code></td>
<td>
<p>only required when called on a data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of the mutual information values, where the diagonal is
fixed to zero
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- DLData(c("ACGT", "ATTA"))
deps &lt;- getDeps(data)


</code></pre>

<hr>
<h2 id='getPWM'>Position weight matrix from DLData object</h2><span id='topic+getPWM'></span><span id='topic+getPWM.DLData'></span>

<h3>Description</h3>

<p>Determines the position weight matrix from a DLData object as relative frequency of symbols
in each column of the data slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPWM(part)

## S3 method for class 'DLData'
getPWM(part)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPWM_+3A_part">part</code></td>
<td>
<p>the DLData object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the position weight matrix, where columns correspond to positions
(columns of the DLData$data slot) and rows to symbols
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- DLData(c("ACGT", "ATTA"))
getPWM(data)
</code></pre>

<hr>
<h2 id='logo'>Sequence logo</h2><span id='topic+logo'></span>

<h3>Description</h3>

<p>Plots a representation of a set of sequences as a sequence logo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logo(part, yoff, ic.scale = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logo_+3A_part">part</code></td>
<td>
<p>the set of sequences as <a href="#topic+DLData">DLData</a> object</p>
</td></tr>
<tr><td><code id="logo_+3A_yoff">yoff</code></td>
<td>
<p>the offset in y-direction within the current plot</p>
</td></tr>
<tr><td><code id="logo_+3A_ic.scale">ic.scale</code></td>
<td>
<p>if <code>TRUE</code>, symbols are scaled by &quot;information content&quot; of the distribution at each position</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a low-level plotting function (using <a href="graphics.html#topic+polygon">polygon</a>, internally).
</p>


<h3>Value</h3>

<p>the vertical (y) offset after this plot
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[,2]) )

# create high-level plot
plot(NULL, xlim = c(1, ncol(data$data) - 1), ylim = c(0, nrow(data$data)),  
   ylab = nrow(data$data), axes = FALSE)
# and add sequence logo and axis
logo(data, yoff = nrow(data$data))
axis(1)
</code></pre>

<hr>
<h2 id='partition'>Paritions data by most inter-dependent positions</h2><span id='topic+partition'></span><span id='topic+partition.DLData'></span>

<h3>Description</h3>

<p>Partitions <code>data</code> by the nucleotides at the most inter-dependent
positions as measures by pairwise mutual information. Paritioning is
performed recursively on the resulting subsets until i) the number of
sequences in a partition is less then <code>minElements</code>, ii) the average
pairwise dependency between the current position and <code>numBestForSorting</code>
other positions with the largest mutual information value drops below
<code>threshold</code>, or iii) <code>maxNum</code> recursive splits have already been
performed. If splitting results in smaller partitions than
<code>minElements</code>, these are added to the smallest partition with more than
<code>minElements</code> sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partition(
  data,
  minElements = 10,
  threshold = 0.1,
  numBestForSorting = 3,
  maxNum = 6,
  sortByWeights = NULL,
  partition.by = NULL
)

## S3 method for class 'DLData'
partition(
  data,
  minElements = 10,
  threshold = 0.1,
  numBestForSorting = 3,
  maxNum = 6,
  sortByWeights = NULL,
  partition.by = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="partition_+3A_data">data</code></td>
<td>
<p>the data as <a href="#topic+DLData">DLData</a> object</p>
</td></tr>
<tr><td><code id="partition_+3A_minelements">minElements</code></td>
<td>
<p>the minimum number of elements to perform a further split</p>
</td></tr>
<tr><td><code id="partition_+3A_threshold">threshold</code></td>
<td>
<p>the threshold on the average mutual information value</p>
</td></tr>
<tr><td><code id="partition_+3A_numbestforsorting">numBestForSorting</code></td>
<td>
<p>the number of dependencies to other positions
considered</p>
</td></tr>
<tr><td><code id="partition_+3A_maxnum">maxNum</code></td>
<td>
<p>the maximum number of recursive splits</p>
</td></tr>
<tr><td><code id="partition_+3A_sortbyweights">sortByWeights</code></td>
<td>
<p>if <code>TRUE</code>, partitions are ordered by their average
weight value, if <code>false</code> by frequency of symbols at the partitioning
position otherwise. If <code>NULL</code>, the <code>$sortByWeights</code> value of the
<a href="#topic+DLData">DLData</a> object is used</p>
</td></tr>
<tr><td><code id="partition_+3A_partition.by">partition.by</code></td>
<td>
<p>specify fixed positions to partition by</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the partitions as list of <a href="#topic+DLData">DLData</a> objects
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"),
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[,2]) )

# partition data using default parameters
partitions &lt;- partition(data)

# partition data using a threshold of 0.3 on the mutual 
# information value to the most dependent position, 
# sorting the resulting partitions by weight
partitions2 &lt;- partition(data = data, threshold = 0.3, numBestForSorting = 1, sortByWeights = TRUE)
</code></pre>

<hr>
<h2 id='plotBlocks'>Plots blocks of data</h2><span id='topic+plotBlocks'></span><span id='topic+plotBlocks.DLData'></span><span id='topic+plotBlocks.list'></span>

<h3>Description</h3>

<p>Plots the blocks of data in <code>data</code> by successive, vertically arranged sub-plots of the function provided as <code>block.fun</code>.
If <code>data</code> is a single <a href="#topic+DLData">DLData</a> object, one block is plotted. Further arguments are provided to <code>block.fun</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBlocks(
  data,
  show.number = TRUE,
  block.fun = deprects,
  ic.scale = TRUE,
  add = FALSE,
  ...
)

## S3 method for class 'DLData'
plotBlocks(
  data,
  show.number = TRUE,
  block.fun = deprects,
  ic.scale = TRUE,
  add = FALSE,
  ...
)

## S3 method for class 'list'
plotBlocks(
  data,
  show.number = TRUE,
  block.fun = deprects,
  ic.scale = TRUE,
  add = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotBlocks_+3A_data">data</code></td>
<td>
<p>the data, a single <a href="#topic+DLData">DLData</a> object or a list of <code>DLData</code> objects</p>
</td></tr>
<tr><td><code id="plotBlocks_+3A_show.number">show.number</code></td>
<td>
<p>if <code>true</code>, the number of sequences (in total) in data is displayed on the left side of the plot</p>
</td></tr>
<tr><td><code id="plotBlocks_+3A_block.fun">block.fun</code></td>
<td>
<p>the function called for each of the blocks</p>
</td></tr>
<tr><td><code id="plotBlocks_+3A_ic.scale">ic.scale</code></td>
<td>
<p>if <code>TRUE</code>, output of <code>block.fun</code> may be scaled by &quot;information content&quot;</p>
</td></tr>
<tr><td><code id="plotBlocks_+3A_add">add</code></td>
<td>
<p>if <code>TRUE</code>, the plot is added to an existing plot</p>
</td></tr>
<tr><td><code id="plotBlocks_+3A_...">...</code></td>
<td>
<p>if <code>add=FALSE</code> forwarded to the internal call to <a href="graphics.html#topic+plot">plot</a></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+deprects">deprects</a>
</p>
<p><a href="#topic+logo">logo</a>
</p>
<p><a href="#topic+colorchart">colorchart</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

# plot all data
plotBlocks(data)

# partition data
partitions &lt;- partition(data, threshold = 0.3)
# and plot partitions
plotBlocks(partitions)

# or plot partitions as sequence logos
plotBlocks(partitions, block.fun = logo)
</code></pre>

<hr>
<h2 id='plotDeparcs'>Plots a graph representation of dependency values</h2><span id='topic+plotDeparcs'></span>

<h3>Description</h3>

<p>Plots a representation of dependency values as arcs between the sequence
positions. Internally, dependency values are computed using <a href="#topic+getDeps">getDeps</a> on
the <code>data</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDeparcs(
  data,
  axis.at.bottom = TRUE,
  add.legend = TRUE,
  show.pvals = FALSE,
  axis.labels = NULL,
  threshold = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDeparcs_+3A_data">data</code></td>
<td>
<p>the <a href="#topic+DLData">DLData</a> object containing the data</p>
</td></tr>
<tr><td><code id="plotDeparcs_+3A_axis.at.bottom">axis.at.bottom</code></td>
<td>
<p>if <code>TRUE</code>, the x-axis is shown at the bottom
(side=1) of the plot, and at the top (side=3) otherwise</p>
</td></tr>
<tr><td><code id="plotDeparcs_+3A_add.legend">add.legend</code></td>
<td>
<p>if <code>TRUE</code> a legend of the color scale is added to the
plot</p>
</td></tr>
<tr><td><code id="plotDeparcs_+3A_show.pvals">show.pvals</code></td>
<td>
<p>if <code>TRUE</code>, -log10 p-values (computed by
<a href="stats.html#topic+pchisq">pchisq</a>) are shown instead of mutual information values</p>
</td></tr>
<tr><td><code id="plotDeparcs_+3A_axis.labels">axis.labels</code></td>
<td>
<p>the labels of the x-axis</p>
</td></tr>
<tr><td><code id="plotDeparcs_+3A_threshold">threshold</code></td>
<td>
<p>threshold in mutual information values, edges below this value are not shown; ignored in <code>show.pvals=TRUE</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[,1], weights = log1p(seqs[, 2]) )

# plot using default parameters
plotDeparcs(data)

# plot with axis at top, without a legend (color scale), and using p-values
plotDeparcs(data, axis.at.bottom = FALSE, add.legend = FALSE, show.pvals = TRUE)
</code></pre>

<hr>
<h2 id='plotDeplogo'>Plot a dependency logo</h2><span id='topic+plotDeplogo'></span><span id='topic+plotDeplogo.DLData'></span>

<h3>Description</h3>

<p>Plots a dependency logo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDeplogo(
  data,
  dep.fun = plotDeparcs,
  block.fun = deprects,
  summary.fun = logo,
  weight.fun = NULL,
  chunks = NULL,
  chunk.height = 800,
  summary.height = 100,
  minPercent = 0.03,
  threshold = 0.1,
  numBestForSorting = 3,
  maxNum = 6,
  sortByWeights = NULL,
  dep.fun.legend = TRUE,
  show.dependency.pvals = FALSE,
  axis.labels = NULL,
  weight.ratio = 5,
  partition.by = NULL,
  ...
)

## S3 method for class 'DLData'
plotDeplogo(
  data,
  dep.fun = plotDeparcs,
  block.fun = deprects,
  summary.fun = logo,
  weight.fun = NULL,
  chunks = NULL,
  chunk.height = 800,
  summary.height = 100,
  minPercent = 0.03,
  threshold = 0.1,
  numBestForSorting = 3,
  maxNum = 6,
  sortByWeights = NULL,
  dep.fun.legend = TRUE,
  show.dependency.pvals = FALSE,
  axis.labels = NULL,
  weight.ratio = 5,
  partition.by = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDeplogo_+3A_data">data</code></td>
<td>
<p>the data, currently implemented for <a href="#topic+DLData">DLData</a> objects</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_dep.fun">dep.fun</code></td>
<td>
<p>the function for plotting the representation of dependency
values (as computed by <a href="#topic+getDeps">getDeps</a>)</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_block.fun">block.fun</code></td>
<td>
<p>the function for plotting a representation of the individual
partitions of the data generated in dependency logos.</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_summary.fun">summary.fun</code></td>
<td>
<p>the function for plotting a representation of the summary
plot for (one chunk of) the data</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_weight.fun">weight.fun</code></td>
<td>
<p>the function for plotting a representation of the
<code>weights</code> values of the sequences within one partition</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_chunks">chunks</code></td>
<td>
<p>the size of chunks the data is split into. The sum of the chunk
sizes must not be greater than the number of data points in data; The
default value of NULL corresponds to one chunk containing all data points</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_chunk.height">chunk.height</code></td>
<td>
<p>the (relative) height of the parts of the plot
representing each of the chunks, one height for each chunk</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_summary.height">summary.height</code></td>
<td>
<p>the (relative) height of the block summaries in the
plot</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_minpercent">minPercent</code></td>
<td>
<p>the minimum percentage of the (sub) data set that may
constitute its own partition in the dependency logo</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_threshold">threshold</code></td>
<td>
<p>the threshold on the dependency value for further splits</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_numbestforsorting">numBestForSorting</code></td>
<td>
<p>the number of dependencies between position i and
all other positions when computing the dependency value of position i</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_maxnum">maxNum</code></td>
<td>
<p>the maximum number of splits allowed</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_sortbyweights">sortByWeights</code></td>
<td>
<p>are partitions sorted by their average weight
(descending)</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_dep.fun.legend">dep.fun.legend</code></td>
<td>
<p>if <code>TRUE</code>, a legend of the color scale used for
plotting the dependency values in <code>dep.fun</code> is added to the plot</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_show.dependency.pvals">show.dependency.pvals</code></td>
<td>
<p>is <code>TRUE</code>, p-values are used for plotting
dependency values in <code>dep.fun</code> instead of mutual information values</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_axis.labels">axis.labels</code></td>
<td>
<p>labels for the x-axis, vector of the same length as the
individual sequences</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_weight.ratio">weight.ratio</code></td>
<td>
<p>the factor by which the plotting width for the main plot is larger than for <code>weight.fun</code></p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_partition.by">partition.by</code></td>
<td>
<p>specify fixed positions to partition by</p>
</td></tr>
<tr><td><code id="plotDeplogo_+3A_...">...</code></td>
<td>
<p>forwarded to the high-level <code>plot</code> that contains the blocks
plotted by <code>block.fun</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>dep.fun</code> provided for plotting the representation of
dependencies is currently implemented in <a href="#topic+plotDeparcs">plotDeparcs</a> and
<a href="#topic+plotDepmatrix">plotDepmatrix</a>. Custom implementations must have the same signature as
these functions and create a single plot without using
<a href="graphics.html#topic+layout">layout</a> (or similar).
</p>
<p>The functions <code>block.fun</code> and <code>summary.fun</code> provided for plotting
the representation of individual partitions of the data generated in
dependency logos are currently implemented in <a href="#topic+deprects">deprects</a>,
<a href="#topic+colorchart">colorchart</a>, and <a href="#topic+logo">logo</a>. Custom implementations must have the same
signature as these functions and create a single plot without using
<a href="graphics.html#topic+layout">layout</a> (or similar).
</p>
<p>The function <code>weight.fun</code> for plotting a representation of the
<code>weights</code> values of the sequences within one partition is currently
implemented in <a href="#topic+subLines">subLines</a> and <a href="#topic+subBoxes">subBoxes</a>. Custom implementations
must have the same signature as these functions and create a single plot
without using <a href="graphics.html#topic+layout">layout</a> (or similar).
</p>


<h3>Value</h3>

<p>a list of <a href="#topic+DLData">DLData</a> objects with the partitions created for the
dependency logo
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1],weights = log1p(seqs[, 2]) )

# plot default dependency logo
plotDeplogo(data)

# refine threshold for clearer picture
plotDeplogo(data, threshold = 0.3)

# customize different parts of the plot
plotDeplogo(data, threshold = 0.3, dep.fun = plotDepmatrix, block.fun = colorchart)

# add plots of the weights
plotDeplogo(data, weight.fun = subBoxes)
</code></pre>

<hr>
<h2 id='plotDepmatrix'>Plots a matrix representation of dependency values</h2><span id='topic+plotDepmatrix'></span>

<h3>Description</h3>

<p>Plots a representation of dependency values as a triangular matrix rotated by
45 degrees. Internally, dependency values are computed using <a href="#topic+getDeps">getDeps</a>
on the <code>data</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDepmatrix(
  data,
  axis.at.bottom = TRUE,
  add.legend = TRUE,
  show.pvals = FALSE,
  axis.labels = NULL,
  threshold = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotDepmatrix_+3A_data">data</code></td>
<td>
<p>the <a href="#topic+DLData">DLData</a> object containing the data</p>
</td></tr>
<tr><td><code id="plotDepmatrix_+3A_axis.at.bottom">axis.at.bottom</code></td>
<td>
<p>if <code>TRUE</code>, the x-axis is shown at the bottom
(side=1) of the plot, and at the top (side=3) otherwise</p>
</td></tr>
<tr><td><code id="plotDepmatrix_+3A_add.legend">add.legend</code></td>
<td>
<p>if <code>TRUE</code> a legend of the color scale is added to the
plot</p>
</td></tr>
<tr><td><code id="plotDepmatrix_+3A_show.pvals">show.pvals</code></td>
<td>
<p>if <code>TRUE</code>, -log10 p-values (computed by
<a href="stats.html#topic+pchisq">pchisq</a>) are shown instead of mutual information values</p>
</td></tr>
<tr><td><code id="plotDepmatrix_+3A_axis.labels">axis.labels</code></td>
<td>
<p>the labels of the x-axis</p>
</td></tr>
<tr><td><code id="plotDepmatrix_+3A_threshold">threshold</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

# plot using default parameters
plotDepmatrix(data)

# plot with axis at top, without a legend (color scale), and using p-values
plotDepmatrix(data, axis.at.bottom = FALSE, add.legend = FALSE, show.pvals = TRUE)
</code></pre>

<hr>
<h2 id='replaceColors'>Replaces colors in <a href="#topic+DLData">DLData</a> object</h2><span id='topic+replaceColors'></span><span id='topic+replaceColors.DLData'></span>

<h3>Description</h3>

<p>Replaces colors in <a href="#topic+DLData">DLData</a> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replaceColors(data, colors)

## S3 method for class 'DLData'
replaceColors(data, colors)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replaceColors_+3A_data">data</code></td>
<td>
<p>the data</p>
</td></tr>
<tr><td><code id="replaceColors_+3A_colors">colors</code></td>
<td>
<p>the new colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the modified <a href="#topic+DLData">DLData</a> object
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+replaceColors">replaceColors</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

replaceColors(data, c("red", "green", "blue", "yellow"))
</code></pre>

<hr>
<h2 id='revcom'>Reverse complement</h2><span id='topic+revcom'></span><span id='topic+revcom.DLData'></span>

<h3>Description</h3>

<p>Determine the reverse complementary <code>DLData</code> object. Only works for DNA or RNA. Data may include gap symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>revcom(data)

## S3 method for class 'DLData'
revcom(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="revcom_+3A_data">data</code></td>
<td>
<p>the data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the reverse complement
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- DLData(c("ACGT", "ATTA"))
revcom(data)
</code></pre>

<hr>
<h2 id='subBoxes'>Plots weights as boxplots</h2><span id='topic+subBoxes'></span>

<h3>Description</h3>

<p>Plots a representation of the weights of a list of <a href="#topic+DLData">DLData</a> objects.
Each entry of the list is shown as an independent boxplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subBoxes(sub.parts, range, axis.above = TRUE, axis.below = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subBoxes_+3A_sub.parts">sub.parts</code></td>
<td>
<p>a list of <a href="#topic+DLData">DLData</a> objects</p>
</td></tr>
<tr><td><code id="subBoxes_+3A_range">range</code></td>
<td>
<p>the range of values shown in the plot (i.e., the <code>xlim</code>
value of the call to <a href="graphics.html#topic+plot">plot</a>)</p>
</td></tr>
<tr><td><code id="subBoxes_+3A_axis.above">axis.above</code></td>
<td>
<p>if <code>TRUE</code>, an axis at the top of the plot (side=3) is shown</p>
</td></tr>
<tr><td><code id="subBoxes_+3A_axis.below">axis.below</code></td>
<td>
<p>if <code>TRUE</code>, an axis at the bottom of the plot (side=1) is shown</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "nrsf.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

# create dependency logo with plotted weights
plotDeplogo(data, threshold = 0.03, weight.fun = subBoxes)
</code></pre>

<hr>
<h2 id='subLines'>Plots weights as lines</h2><span id='topic+subLines'></span>

<h3>Description</h3>

<p>Plots a representation of the weights of a list of <a href="#topic+DLData">DLData</a> objects.
Each entry of the list is shown as an independent line with the median value
shown as a red vertical line. Plots of list entries are separated by
horizontal grey lines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subLines(sub.parts, range, axis.above = TRUE, axis.below = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subLines_+3A_sub.parts">sub.parts</code></td>
<td>
<p>a list of <a href="#topic+DLData">DLData</a> objects</p>
</td></tr>
<tr><td><code id="subLines_+3A_range">range</code></td>
<td>
<p>the range of values shown in the plot (i.e., the <code>xlim</code>
value of the call to <a href="graphics.html#topic+plot">plot</a>)</p>
</td></tr>
<tr><td><code id="subLines_+3A_axis.above">axis.above</code></td>
<td>
<p>if <code>TRUE</code>, an axis at the top of the plot (side=3) is shown</p>
</td></tr>
<tr><td><code id="subLines_+3A_axis.below">axis.below</code></td>
<td>
<p>if <code>TRUE</code>, an axis at the bottom of the plot (side=1) is shown</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "nrsf.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )

# create dependency logo with plotted weights
plotDeplogo(data, threshold = 0.03, weight.fun = subLines)
</code></pre>

<hr>
<h2 id='suggestColors'>Suggests colors for symbols</h2><span id='topic+suggestColors'></span><span id='topic+suggestColors.DLData'></span>

<h3>Description</h3>

<p>Suggests colors for the symbols in <code>data</code> based on the co-occurrence of
symbols at common positions, weighted by the dependency values at those positions.
The idea is to assign similar colors only to symbols that either mostly occur at 
different positions or that are present at positions with low inter-dependencies 
to other positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>suggestColors(data)

## S3 method for class 'DLData'
suggestColors(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="suggestColors_+3A_data">data</code></td>
<td>
<p>the data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the colors
</p>


<h3>Author(s)</h3>

<p>Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+replaceColors">replaceColors</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read data and create DLData object
seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1] ,weights = log1p(seqs[, 2]) )

suggestColors(data)
</code></pre>

<hr>
<h2 id='summary.DLData'>Summarizing DLData objects</h2><span id='topic+summary.DLData'></span>

<h3>Description</h3>

<p><code>summary</code> method for class &quot;DLData&quot;.
The summary includes the number of sequences, the consensus sequence
and the number of sequences in <code>object</code> that match the consensus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DLData'
summary(object, delete.gaps = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.DLData_+3A_object">object</code></td>
<td>
<p>an object of class &quot;DLData&quot;</p>
</td></tr>
<tr><td><code id="summary.DLData_+3A_delete.gaps">delete.gaps</code></td>
<td>
<p>if gaps should be removed from the consensus</p>
</td></tr>
<tr><td><code id="summary.DLData_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>list</code> with elements <code>members</code> containing the number of sequences,
<code>consensus</code> containing the consensus sequences, and <code>equal.consensus</code> containing the
number of sequences in <code>object</code> that are identical to <code>consensus</code>
</p>


<h3>Author(s)</h3>

<p>Jens Keilwagen, Jan Grau &lt;grau@informatik.uni-halle.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seqs &lt;- read.table(system.file("extdata", "cjun.txt", package = "DepLogo"), 
    stringsAsFactors = FALSE)
data &lt;- DLData(sequences = seqs[, 1], weights = log1p(seqs[, 2]) )
summary(data)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
