<!DOCTYPE html><html lang="en"><head><title>Help for package DEploid.utils</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DEploid.utils}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DEploid.utils-package'><p>DEploid.utils: 'DEploid' Data Analysis and Results Interpretation</p></a></li>
<li><a href='#computeObsWSAF'><p>Compute observed WSAF</p></a></li>
<li><a href='#extractCoverageFromTxt'><p>Extract read counts from plain text file</p></a></li>
<li><a href='#extractCoverageFromVcf'><p>Extract VCF information</p></a></li>
<li><a href='#extractPLAF'><p>Extract PLAF</p></a></li>
<li><a href='#haplotypePainter'><p>Painting haplotype according the reference panel</p></a></li>
<li><a href='#histWSAF'><p>WSAF histogram</p></a></li>
<li><a href='#plotAltVsRef'><p>Plot coverage</p></a></li>
<li><a href='#plotObsExpWSAF'><p>Plot WSAF</p></a></li>
<li><a href='#plotProportions'><p>Plot proportions</p></a></li>
<li><a href='#plotWSAFvsPLAF'><p>Plot WSAF vs PLAF</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>'DEploid' Data Analysis and Results Interpretation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>'DEploid' (Zhu et.al. 2018 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx530">doi:10.1093/bioinformatics/btx530</a>&gt;)
    is designed for deconvoluting mixed genomes with unknown 
    proportions. Traditional phasing programs are limited to diploid organisms. 
    Our method modifies Li and Stephen’s algorithm with Markov chain Monte Carlo
    (MCMC) approaches, and builds a generic framework that allows haloptype
    searches in a multiple infection setting. This package provides R functions
    to support data analysis and results interpretation.  </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.2), scales (&ge; 0.4.0), magrittr (&ge; 1.5),
combinat</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown(&ge; 1.6), circlize, testthat (&ge; 0.9.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-11-18</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-18 12:02:05 UTC; rstudio</td>
</tr>
<tr>
<td>Author:</td>
<td>Joe Zhu <a href="https://orcid.org/0000-0001-7566-2787"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Jacob Almagro-Garcia [aut],
  Gil McVean [aut],
  University of Oxford [cph],
  Yinghan Liu [ctb],
  CodeCogs Zyba Ltd [com, cph],
  Deepak Bandyopadhyay [com, cph],
  Lutz Kettner [com, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Joe Zhu &lt;sha.joe.zhu@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-19 16:00:18 UTC</td>
</tr>
</table>
<hr>
<h2 id='DEploid.utils-package'>DEploid.utils: 'DEploid' Data Analysis and Results Interpretation</h2><span id='topic+DEploid.utils'></span><span id='topic+DEploid.utils-package'></span>

<h3>Description</h3>

<p>'DEploid' (Zhu et.al. 2018 <a href="https://doi.org/10.1093/bioinformatics/btx530">doi:10.1093/bioinformatics/btx530</a>) is designed for deconvoluting mixed genomes with unknown proportions. Traditional phasing programs are limited to diploid organisms. Our method modifies Li and Stephen’s algorithm with Markov chain Monte Carlo (MCMC) approaches, and builds a generic framework that allows haloptype searches in a multiple infection setting. This package provides R functions to support data analysis and results interpretation.
</p>
<p>Traditional phasing programs are limited to diploid organisms.
Our method modifies Li and Stephens algorithm with Markov chain Monte Carlo
(MCMC) approaches, and builds a generic framework that allows haplotype
searches in a multiple infection setting. This package is primarily developed
as part of #' the Pf3k project, which is a global collaboration using the
latest sequencing technologies to provide a high-resolution view of natural
variation in the malaria parasite Plasmodium falciparum. Parasite DNA are
extracted from patient blood sample, which often contains more than one
parasite strain, with unknown proportions. This package is used for
deconvoluting mixed haplotypes, #' and reporting the mixture proportions from
each sample.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Joe Zhu <a href="mailto:sha.joe.zhu@gmail.com">sha.joe.zhu@gmail.com</a> (<a href="https://orcid.org/0000-0001-7566-2787">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Jacob Almagro-Garcia
</p>
</li>
<li><p> Gil McVean
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> University of Oxford [copyright holder]
</p>
</li>
<li><p> Yinghan Liu [contributor]
</p>
</li>
<li><p> CodeCogs Zyba Ltd [compiler, copyright holder]
</p>
</li>
<li><p> Deepak Bandyopadhyay [compiler, copyright holder]
</p>
</li>
<li><p> Lutz Kettner [compiler, copyright holder]
</p>
</li></ul>

<p>Joe Zhu
</p>
<p>Maintainer: Joe Zhu <a href="mailto:sha.joe.zhu@gmail.com">sha.joe.zhu@gmail.com</a>
</p>

<hr>
<h2 id='computeObsWSAF'>Compute observed WSAF</h2><span id='topic+computeObsWSAF'></span>

<h3>Description</h3>

<p>Compute observed allele frequency within sample from the allele counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeObsWSAF(alt, ref)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="computeObsWSAF_+3A_alt">alt</code></td>
<td>
<p>Numeric array of alternative allele count.</p>
</td></tr>
<tr><td><code id="computeObsWSAF_+3A_ref">ref</code></td>
<td>
<p>Numeric array of reference allele count.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric array of observed allele frequency within sample.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+histWSAF">histWSAF</a></code> for histogram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid.utils")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid.utils")
PG0390CoverageTxt &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageTxt$altCount, PG0390CoverageTxt$refCount)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid.utils")
PG0390CoverageVcf &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageVcf$altCount, PG0390CoverageVcf$refCount)

</code></pre>

<hr>
<h2 id='extractCoverageFromTxt'>Extract read counts from plain text file</h2><span id='topic+extractCoverageFromTxt'></span>

<h3>Description</h3>

<p>Extract read counts from tab-delimited text files of a single sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCoverageFromTxt(refFileName, altFileName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractCoverageFromTxt_+3A_reffilename">refFileName</code></td>
<td>
<p>Path of the reference allele count file.</p>
</td></tr>
<tr><td><code id="extractCoverageFromTxt_+3A_altfilename">altFileName</code></td>
<td>
<p>Path of the alternative allele count file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame contains four columns: chromosomes, positions, reference allele count, alternative allele count.
</p>


<h3>Note</h3>

<p>The allele count files must be tab-delimited. The allele count files contain three columns: chromosomes, positions and allele count.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid.utils")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid.utils")
PG0390 &lt;- extractCoverageFromTxt(refFile, altFile)

</code></pre>

<hr>
<h2 id='extractCoverageFromVcf'>Extract VCF information</h2><span id='topic+extractCoverageFromVcf'></span>

<h3>Description</h3>

<p>Extract VCF information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractCoverageFromVcf(filename, samplename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractCoverageFromVcf_+3A_filename">filename</code></td>
<td>
<p>VCF file name.</p>
</td></tr>
<tr><td><code id="extractCoverageFromVcf_+3A_samplename">samplename</code></td>
<td>
<p>Sample name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe list with members of haplotypes, proportions and log likelihood of the MCMC chain.
</p>

<ul>
<li> <p><code>CHROM</code> SNP chromosomes.
</p>
</li>
<li> <p><code>POS</code> SNP positions.
</p>
</li>
<li> <p><code>refCount</code> reference allele count.
</p>
</li>
<li> <p><code>altCount</code> alternative allele count.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code>extractCoverageFromVcf</code>
</p>
</li>
<li> <p><code>extractCoverageFromTxt</code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid.utils")
vcf = extractCoverageFromVcf(vcfFile, "PG0390-C")

</code></pre>

<hr>
<h2 id='extractPLAF'>Extract PLAF</h2><span id='topic+extractPLAF'></span>

<h3>Description</h3>

<p>Extract population level allele frequency (PLAF) from text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractPLAF(plafFileName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractPLAF_+3A_plaffilename">plafFileName</code></td>
<td>
<p>Path of the PLAF text file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric array of PLAF
</p>


<h3>Note</h3>

<p>The text file must have header, and population level allele frequency recorded in the &quot;PLAF&quot; field.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt", package = "DEploid.utils")
plaf &lt;- extractPLAF(plafFile)

</code></pre>

<hr>
<h2 id='haplotypePainter'>Painting haplotype according the reference panel</h2><span id='topic+haplotypePainter'></span>

<h3>Description</h3>

<p>Plot the posterior probabilities of a haplotype given the refernece panel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotypePainter(
  posteriorProbabilities,
  title = "",
  labelScaling,
  numberOfInbreeding = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="haplotypePainter_+3A_posteriorprobabilities">posteriorProbabilities</code></td>
<td>
<p>Posterior probabilities matrix with the size of number of loci by the number of reference strain.</p>
</td></tr>
<tr><td><code id="haplotypePainter_+3A_title">title</code></td>
<td>
<p>Figure title.</p>
</td></tr>
<tr><td><code id="haplotypePainter_+3A_labelscaling">labelScaling</code></td>
<td>
<p>Scaling parameter for plotting.</p>
</td></tr>
<tr><td><code id="haplotypePainter_+3A_numberofinbreeding">numberOfInbreeding</code></td>
<td>
<p>Number of inbreading strains</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value called for side effects
</p>

<hr>
<h2 id='histWSAF'>WSAF histogram</h2><span id='topic+histWSAF'></span>

<h3>Description</h3>

<p>Produce histogram of the allele frequency within sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>histWSAF(
  obsWSAF,
  exclusive = TRUE,
  title = "Histogram 0&lt;WSAF&lt;1",
  cex.lab = 1,
  cex.main = 1,
  cex.axis = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="histWSAF_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Observed allele frequency within sample</p>
</td></tr>
<tr><td><code id="histWSAF_+3A_exclusive">exclusive</code></td>
<td>
<p>When TRUE 0 &lt; WSAF &lt; 1; otherwise 0 &lt;= WSAF &lt;= 1.</p>
</td></tr>
<tr><td><code id="histWSAF_+3A_title">title</code></td>
<td>
<p>Histogram title</p>
</td></tr>
<tr><td><code id="histWSAF_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="histWSAF_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="histWSAF_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>histogram
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid.utils")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid.utils")
PG0390CoverageTxt &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageTxt$altCount, PG0390CoverageTxt$refCount)
histWSAF(obsWSAF)
myhist &lt;- histWSAF(obsWSAF, FALSE)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid.utils")
PG0390CoverageVcf &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageVcf$altCount, PG0390CoverageVcf$refCount)
histWSAF(obsWSAF)
myhist &lt;- histWSAF(obsWSAF, FALSE)

</code></pre>

<hr>
<h2 id='plotAltVsRef'>Plot coverage</h2><span id='topic+plotAltVsRef'></span>

<h3>Description</h3>

<p>Plot alternative allele count vs reference allele count at each site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAltVsRef(
  ref,
  alt,
  title = "Alt vs Ref",
  exclude.ref = c(),
  exclude.alt = c(),
  potentialOutliers = c(),
  cex.lab = 1,
  cex.main = 1,
  cex.axis = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotAltVsRef_+3A_ref">ref</code></td>
<td>
<p>Numeric array of reference allele count.</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_alt">alt</code></td>
<td>
<p>Numeric array of alternative allele count.</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;Alt vs Ref&quot; by default</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_exclude.ref">exclude.ref</code></td>
<td>
<p>Numeric array of reference allele count at sites that are not deconvoluted.</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_exclude.alt">exclude.alt</code></td>
<td>
<p>Numeric array of alternative allele count at sites that are not deconvoluted</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_potentialoutliers">potentialOutliers</code></td>
<td>
<p>Potential outliers</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="plotAltVsRef_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid.utils")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid.utils")
PG0390CoverageTxt &lt;- extractCoverageFromTxt(refFile, altFile)
plotAltVsRef(PG0390CoverageTxt$refCount, PG0390CoverageTxt$altCount)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid.utils")
PG0390CoverageVcf &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
plotAltVsRef(PG0390CoverageVcf$refCount, PG0390CoverageVcf$altCount)

</code></pre>

<hr>
<h2 id='plotObsExpWSAF'>Plot WSAF</h2><span id='topic+plotObsExpWSAF'></span>

<h3>Description</h3>

<p>Plot observed alternative allele frequency within sample against expected WSAF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotObsExpWSAF(
  obsWSAF,
  expWSAF,
  title = "WSAF(observed vs expected)",
  cex.lab = 1,
  cex.main = 1,
  cex.axis = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotObsExpWSAF_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Numeric array of observed WSAF.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAF_+3A_expwsaf">expWSAF</code></td>
<td>
<p>Numeric array of expected WSAF.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAF_+3A_title">title</code></td>
<td>
<p>Figure title.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAF_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAF_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAF_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value called for side effects
</p>

<hr>
<h2 id='plotProportions'>Plot proportions</h2><span id='topic+plotProportions'></span>

<h3>Description</h3>

<p>Plot the MCMC samples of the proportion, indexed by the MCMC chain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProportions(
  proportions,
  title = "Components",
  cex.lab = 1,
  cex.main = 1,
  cex.axis = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotProportions_+3A_proportions">proportions</code></td>
<td>
<p>Matrix of the MCMC proportion samples. The matrix size is number of the MCMC samples by the number of strains.</p>
</td></tr>
<tr><td><code id="plotProportions_+3A_title">title</code></td>
<td>
<p>Figure title.</p>
</td></tr>
<tr><td><code id="plotProportions_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="plotProportions_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="plotProportions_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value called for side effects
</p>

<hr>
<h2 id='plotWSAFvsPLAF'>Plot WSAF vs PLAF</h2><span id='topic+plotWSAFvsPLAF'></span>

<h3>Description</h3>

<p>Plot allele frequencies within sample against population level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotWSAFvsPLAF(
  plaf,
  obsWSAF,
  expWSAF = c(),
  potentialOutliers = c(),
  title = "WSAF vs PLAF",
  cex.lab = 1,
  cex.main = 1,
  cex.axis = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotWSAFvsPLAF_+3A_plaf">plaf</code></td>
<td>
<p>Numeric array of population level allele frequency.</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Numeric array of observed altenative allele frequencies within sample.</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_expwsaf">expWSAF</code></td>
<td>
<p>Numeric array of expected WSAF from model.</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_potentialoutliers">potentialOutliers</code></td>
<td>
<p>Potential outliers</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;WSAF vs PLAF&quot; by default</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="plotWSAFvsPLAF_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid.utils")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid.utils")
PG0390CoverageTxt &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageTxt$altCount, PG0390CoverageTxt$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt", package = "DEploid.utils")
plaf &lt;- extractPLAF(plafFile)
plotWSAFvsPLAF(plaf, obsWSAF)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid.utils")
PG0390CoverageVcf &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageVcf$altCount, PG0390CoverageVcf$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt", package = "DEploid.utils")
plaf &lt;- extractPLAF(plafFile)
plotWSAFvsPLAF(plaf, obsWSAF)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
