<!DOCTYPE html><html lang="en"><head><title>Help for package DEploid</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DEploid}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DEploid-package'><p>DEploid: Deconvolute Mixed Genomes with Unknown Proportions</p></a></li>
<li><a href='#dEploid'><p>Deconvolute mixed haplotypes</p></a></li>
<li><a href='#plotAltVsRefPlotly'><p>Plot coverage</p></a></li>
<li><a href='#plotHistWSAFPlotly'><p>WSAF histogram</p></a></li>
<li><a href='#plotObsExpWSAFPlotly'><p>Plot WSAF</p></a></li>
<li><a href='#plotWSAFVsPLAFPlotly'><p>Plot WSAF vs PLAF</p></a></li>
<li><a href='#reexports'><p>Extract read counts from plain text file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Deconvolute Mixed Genomes with Unknown Proportions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.7</td>
</tr>
<tr>
<td>Description:</td>
<td>Traditional phasing programs are limited to diploid organisms.
 Our method modifies Li and Stephens algorithm with Markov chain Monte Carlo
 (MCMC) approaches, and builds a generic framework that allows haplotype searches
 in a multiple infection setting. This package is primarily developed as part of
 the Pf3k project, which is a global collaboration using the latest
 sequencing technologies to provide a high-resolution view of natural variation
 in the malaria parasite Plasmodium falciparum. Parasite DNA are extracted from
 patient blood sample, which often contains more than one parasite strain, with
 unknown proportions. This package is used for deconvoluting mixed haplotypes,
 and reporting the mixture proportions from each sample.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DEploid-dev/DEploid-r">https://github.com/DEploid-dev/DEploid-r</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/DEploid-dev/DEploid-r/issues">https://github.com/DEploid-dev/DEploid-r/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), DEploid.utils (&ge; 0.0.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.2), scales (&ge; 0.4.0), plotly (&ge; 4.7.1),
magrittr (&ge; 1.5), rmarkdown(&ge; 1.6), htmlwidgets (&ge; 1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat (&ge; 0.9.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-12-22</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-20 02:29:18 UTC; rstudio</td>
</tr>
<tr>
<td>Author:</td>
<td>Joe Zhu <a href="https://orcid.org/0000-0001-7566-2787"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Jacob Almagro-Garcia [aut],
  Gil McVean [aut],
  University of Oxford [cph],
  Yinghan Liu [ctb],
  CodeCogs Zyba Ltd [com, cph],
  Deepak Bandyopadhyay [com, cph],
  Lutz Kettner [com, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Joe Zhu &lt;sha.joe.zhu@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-20 03:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='DEploid-package'>DEploid: Deconvolute Mixed Genomes with Unknown Proportions</h2><span id='topic+DEploid'></span><span id='topic+DEploid-package'></span>

<h3>Description</h3>

<p>Traditional phasing programs are limited to diploid organisms. Our method modifies Li and Stephens algorithm with Markov chain Monte Carlo (MCMC) approaches, and builds a generic framework that allows haplotype searches in a multiple infection setting. This package is primarily developed as part of the Pf3k project, which is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. Parasite DNA are extracted from patient blood sample, which often contains more than one parasite strain, with unknown proportions. This package is used for deconvoluting mixed haplotypes, and reporting the mixture proportions from each sample.
</p>
<p>Traditional phasing programs are limited to diploid organisms.
Our method modifies Li and Stephens algorithm with Markov chain Monte Carlo
(MCMC) approaches, and builds a generic framework that allows haplotype
searches in a multiple infection setting. This package is primarily developed
as part of #' the Pf3k project, which is a global collaboration using the
latest sequencing technologies to provide a high-resolution view of natural
variation in the malaria parasite Plasmodium falciparum. Parasite DNA are
extracted from patient blood sample, which often contains more than one
parasite strain, with unknown proportions. This package is used for
deconvoluting mixed haplotypes, #' and reporting the mixture proportions from
each sample.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Joe Zhu <a href="mailto:sha.joe.zhu@gmail.com">sha.joe.zhu@gmail.com</a> (<a href="https://orcid.org/0000-0001-7566-2787">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Jacob Almagro-Garcia
</p>
</li>
<li><p> Gil McVean
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> University of Oxford [copyright holder]
</p>
</li>
<li><p> Yinghan Liu [contributor]
</p>
</li>
<li><p> CodeCogs Zyba Ltd [compiler, copyright holder]
</p>
</li>
<li><p> Deepak Bandyopadhyay [compiler, copyright holder]
</p>
</li>
<li><p> Lutz Kettner [compiler, copyright holder]
</p>
</li></ul>

<p>Zhu Sha
</p>
<p>Maintainer: Joe Zhu <a href="mailto:sha.joe.zhu@gmail.com">sha.joe.zhu@gmail.com</a>
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/DEploid-dev/DEploid-r">https://github.com/DEploid-dev/DEploid-r</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/DEploid-dev/DEploid-r/issues">https://github.com/DEploid-dev/DEploid-r/issues</a>
</p>
</li></ul>


<hr>
<h2 id='dEploid'>Deconvolute mixed haplotypes</h2><span id='topic+dEploid'></span>

<h3>Description</h3>

<p>Deconvolute mixed haplotypes, and reporting the mixture proportions from each sample
This function provieds an interface for calling <em>dEploid</em> from R.
The command line options are passed via the <code>args</code> argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dEploid(args)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dEploid_+3A_args">args</code></td>
<td>
<p>String of dEploid input.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with members of haplotypes, proportions and log likelihood of the MCMC chain.
</p>

<ul>
<li> <p><code>Haps</code> Haplotypes at the final iteration in plain text file.
</p>
</li>
<li> <p><code>Proportions</code> MCMC updates of the proportion estimates.
</p>
</li>
<li> <p><code>llks</code> Log likelihood of the MCMC chain.
</p>
</li></ul>



<h3>Seeding</h3>

<p>The R version of DEploid uses random number from R's random generator. Therefore, the '-seed'
argument of the command line version will be ignored, and no seed is given in the output. Use
the R function 'set.seed' prior to calling this function to ensure reproduciblity of results.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code>vignette('dEploid-Arguments')</code> for an overview of commandline arguments
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
plafFile = system.file("extdata", "labStrains.test.PLAF.txt", package = "DEploid")
set.seed(1234)
PG0390.deconv = dEploid(paste("-vcf", vcfFile, "-plaf", plafFile, "-noPanel"))

## End(Not run)

</code></pre>

<hr>
<h2 id='plotAltVsRefPlotly'>Plot coverage</h2><span id='topic+plotAltVsRefPlotly'></span>

<h3>Description</h3>

<p>Plot alternative allele count vs reference allele count at each
site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAltVsRefPlotly(ref, alt, title = "Alt vs Ref", potentialOutliers = c())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotAltVsRefPlotly_+3A_ref">ref</code></td>
<td>
<p>Numeric array of reference allele count.</p>
</td></tr>
<tr><td><code id="plotAltVsRefPlotly_+3A_alt">alt</code></td>
<td>
<p>Numeric array of alternative allele count.</p>
</td></tr>
<tr><td><code id="plotAltVsRefPlotly_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;Alt vs Ref&quot; by default</p>
</td></tr>
<tr><td><code id="plotAltVsRefPlotly_+3A_potentialoutliers">potentialOutliers</code></td>
<td>
<p>Index of potential outliers.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
plotAltVsRefPlotly(PG0390CoverageT$refCount, PG0390CoverageT$altCount)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
plotAltVsRefPlotly(PG0390CoverageV$refCount, PG0390CoverageV$altCount)

</code></pre>

<hr>
<h2 id='plotHistWSAFPlotly'>WSAF histogram</h2><span id='topic+plotHistWSAFPlotly'></span>

<h3>Description</h3>

<p>Produce histogram of the allele frequency within sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHistWSAFPlotly(obsWSAF, exclusive = TRUE, title = "Histogram 0&lt;WSAF&lt;1")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotHistWSAFPlotly_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Observed allele frequency within sample</p>
</td></tr>
<tr><td><code id="plotHistWSAFPlotly_+3A_exclusive">exclusive</code></td>
<td>
<p>When TRUE 0 &lt; WSAF &lt; 1; otherwise 0 &lt;= WSAF &lt;= 1.</p>
</td></tr>
<tr><td><code id="plotHistWSAFPlotly_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;Histogram 0&lt;WSAF&lt;1&quot; by default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>histogram
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390Coverage &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390Coverage$altCount, PG0390Coverage$refCount)
plotHistWSAFPlotly(obsWSAF)
myhist &lt;- plotHistWSAFPlotly(obsWSAF)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
plotHistWSAFPlotly(obsWSAF)
myhist &lt;- plotHistWSAFPlotly(obsWSAF)

</code></pre>

<hr>
<h2 id='plotObsExpWSAFPlotly'>Plot WSAF</h2><span id='topic+plotObsExpWSAFPlotly'></span>

<h3>Description</h3>

<p>Plot observed alternative allele frequency within sample against
expected WSAF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotObsExpWSAFPlotly(obsWSAF, expWSAF, title = "WSAF(observed vs expected)")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotObsExpWSAFPlotly_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Numeric array of observed WSAF.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAFPlotly_+3A_expwsaf">expWSAF</code></td>
<td>
<p>Numeric array of expected WSAF.</p>
</td></tr>
<tr><td><code id="plotObsExpWSAFPlotly_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;WSAF(observed vs expected)&quot; by default</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
PG0390CoverageV.deconv &lt;- dEploid(paste(
  "-vcf", vcfFile,
  "-plaf", plafFile, "-noPanel"
))

prop &lt;- PG0390CoverageV.deconv$Proportions[dim(PG0390CoverageV.deconv
$Proportions)[1], ]

expWSAF &lt;- t(PG0390CoverageV.deconv$Haps) %*% prop
plotObsExpWSAFPlotly(obsWSAF, expWSAF)

## End(Not run)

</code></pre>

<hr>
<h2 id='plotWSAFVsPLAFPlotly'>Plot WSAF vs PLAF</h2><span id='topic+plotWSAFVsPLAFPlotly'></span>

<h3>Description</h3>

<p>Plot allele frequencies within sample against population level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotWSAFVsPLAFPlotly(
  plaf,
  obsWSAF,
  ref,
  alt,
  title = "WSAF vs PLAF",
  potentialOutliers = c()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_plaf">plaf</code></td>
<td>
<p>Numeric array of population level allele frequency.</p>
</td></tr>
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Numeric array of observed altenative allele frequencies within
sample.</p>
</td></tr>
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_ref">ref</code></td>
<td>
<p>Numeric array of reference allele count.</p>
</td></tr>
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_alt">alt</code></td>
<td>
<p>Numeric array of alternative allele count.</p>
</td></tr>
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_title">title</code></td>
<td>
<p>Figure title, &quot;WSAF vs PLAF&quot; by default</p>
</td></tr>
<tr><td><code id="plotWSAFVsPLAFPlotly_+3A_potentialoutliers">potentialOutliers</code></td>
<td>
<p>Index of potential outliers.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageT$altCount, PG0390CoverageT$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
plaf &lt;- extractPLAF(plafFile)
plotWSAFVsPLAFPlotly(
  plaf, obsWSAF, PG0390CoverageT$refCount,
  PG0390CoverageT$altCount
)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
plaf &lt;- extractPLAF(plafFile)
plotWSAFVsPLAFPlotly(
  plaf, obsWSAF, PG0390CoverageV$refCount,
  PG0390CoverageV$altCount
)

</code></pre>

<hr>
<h2 id='reexports'>Extract read counts from plain text file</h2><span id='topic+reexports'></span><span id='topic+extractCoverageFromTxt'></span><span id='topic+extractCoverageFromVcf'></span><span id='topic+extractPLAF'></span><span id='topic+plotProportions'></span><span id='topic+plotAltVsRef'></span><span id='topic+histWSAF'></span><span id='topic+plotWSAFvsPLAF'></span><span id='topic+plotObsExpWSAF'></span><span id='topic+computeObsWSAF'></span><span id='topic+haplotypePainter'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>DEploid.utils</dt><dd><p><code><a href="DEploid.utils.html#topic+computeObsWSAF">computeObsWSAF</a></code>, <code><a href="DEploid.utils.html#topic+extractCoverageFromTxt">extractCoverageFromTxt</a></code>, <code><a href="DEploid.utils.html#topic+extractCoverageFromVcf">extractCoverageFromVcf</a></code>, <code><a href="DEploid.utils.html#topic+extractPLAF">extractPLAF</a></code>, <code><a href="DEploid.utils.html#topic+haplotypePainter">haplotypePainter</a></code>, <code><a href="DEploid.utils.html#topic+histWSAF">histWSAF</a></code>, <code><a href="DEploid.utils.html#topic+plotAltVsRef">plotAltVsRef</a></code>, <code><a href="DEploid.utils.html#topic+plotObsExpWSAF">plotObsExpWSAF</a></code>, <code><a href="DEploid.utils.html#topic+plotProportions">plotProportions</a></code>, <code><a href="DEploid.utils.html#topic+plotWSAFvsPLAF">plotWSAFvsPLAF</a></code></p>
</dd>
</dl>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reexports_+3A_reffilename">refFileName</code></td>
<td>
<p>Path of the reference allele count file.</p>
</td></tr>
<tr><td><code id="reexports_+3A_altfilename">altFileName</code></td>
<td>
<p>Path of the alternative allele count file.</p>
</td></tr>
<tr><td><code id="reexports_+3A_vcffilename">vcfFileName</code></td>
<td>
<p>Path of the VCF file.</p>
</td></tr>
<tr><td><code id="reexports_+3A_adfieldindex">ADFieldIndex</code></td>
<td>
<p>Index of the AD field of the sample field. For example,
if the format is &quot;GT:AD:DP:GQ:PL&quot;, the AD index is 2 (by default).</p>
</td></tr>
<tr><td><code id="reexports_+3A_plaffilename">plafFileName</code></td>
<td>
<p>Path of the PLAF text file.</p>
</td></tr>
<tr><td><code id="reexports_+3A_proportions">proportions</code></td>
<td>
<p>Matrix of the MCMC proportion samples. The matrix size is
number of the MCMC samples by the number of strains.</p>
</td></tr>
<tr><td><code id="reexports_+3A_exclude.ref">exclude.ref</code></td>
<td>
<p>Numeric array of reference allele count at sites that are
not deconvoluted.</p>
</td></tr>
<tr><td><code id="reexports_+3A_exclude.alt">exclude.alt</code></td>
<td>
<p>Numeric array of alternative allele count at sites that
are not deconvoluted</p>
</td></tr>
<tr><td><code id="reexports_+3A_exclusive">exclusive</code></td>
<td>
<p>When TRUE 0 &lt; WSAF &lt; 1; otherwise 0 &lt;= WSAF &lt;= 1.</p>
</td></tr>
<tr><td><code id="reexports_+3A_plaf">plaf</code></td>
<td>
<p>Numeric array of population level allele frequency.</p>
</td></tr>
<tr><td><code id="reexports_+3A_potentialoutliers">potentialOutliers</code></td>
<td>
<p>Index of potential outliers.</p>
</td></tr>
<tr><td><code id="reexports_+3A_obswsaf">obsWSAF</code></td>
<td>
<p>Numeric array of observed WSAF.</p>
</td></tr>
<tr><td><code id="reexports_+3A_expwsaf">expWSAF</code></td>
<td>
<p>Numeric array of expected WSAF.</p>
</td></tr>
<tr><td><code id="reexports_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Label size.</p>
</td></tr>
<tr><td><code id="reexports_+3A_cex.main">cex.main</code></td>
<td>
<p>Title size.</p>
</td></tr>
<tr><td><code id="reexports_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Axis text size.</p>
</td></tr>
<tr><td><code id="reexports_+3A_ref">ref</code></td>
<td>
<p>Numeric array of reference allele count.</p>
</td></tr>
<tr><td><code id="reexports_+3A_alt">alt</code></td>
<td>
<p>Numeric array of alternative allele count.</p>
</td></tr>
<tr><td><code id="reexports_+3A_posteriorprobabilities">posteriorProbabilities</code></td>
<td>
<p>Posterior probabilities matrix with the size of
number of loci by the number of reference strain.</p>
</td></tr>
<tr><td><code id="reexports_+3A_title">title</code></td>
<td>
<p>Figure title.</p>
</td></tr>
<tr><td><code id="reexports_+3A_labelscaling">labelScaling</code></td>
<td>
<p>Scaling parameter for plotting.</p>
</td></tr>
<tr><td><code id="reexports_+3A_numberofinbreeding">numberOfInbreeding</code></td>
<td>
<p>Number of inbreeding strains copying from.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame contains four columns: chromosomes, positions, reference
allele count, alternative allele count.
</p>
<p>A data.frame contains four columns: chromosomes, positions, reference
allele count, alternative allele count.
</p>
<p>A numeric array of PLAF
</p>
<p>histogram
</p>
<p>Numeric array of observed allele frequency within sample.
</p>


<h3>Note</h3>

<p>The allele count files must be tab-delimited. The allele count files
contain three columns: chromosomes, positions and allele count.
</p>
<p>The VCF file should only contain one sample. If more samples present in
the VCF, it only returns coverage for of the first sample.
</p>
<p>The text file must have header, and population level allele frequency
recorded in the &quot;PLAF&quot; field.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+histWSAF">histWSAF</a></code> for histogram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390 &lt;- extractCoverageFromTxt(refFile, altFile)

vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390 &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")

plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
plaf &lt;- extractPLAF(plafFile)
## Not run: 
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
panelFile &lt;- system.file("extdata", "labStrains.test.panel.txt",
  package = "DEploid"
)
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
PG0390Coverage.deconv &lt;- dEploid(paste(
  "-ref", refFile, "-alt", altFile,
  "-plaf", plafFile, "-noPanel"
))
plotProportions(PG0390Coverage.deconv$Proportions, "PG0390-C proportions")

## End(Not run)

# Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
plotAltVsRef(PG0390CoverageT$refCount, PG0390CoverageT$altCount)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
plotAltVsRef(PG0390CoverageV$refCount, PG0390CoverageV$altCount)

# Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390Coverage &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390Coverage$altCount, PG0390Coverage$refCount)
histWSAF(obsWSAF)
myhist &lt;- histWSAF(obsWSAF, FALSE)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
histWSAF(obsWSAF)
myhist &lt;- histWSAF(obsWSAF, FALSE)

# Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageT$altCount, PG0390CoverageT$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
plaf &lt;- extractPLAF(plafFile)
plotWSAFvsPLAF(plaf, obsWSAF)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
plaf &lt;- extractPLAF(plafFile)
plotWSAFvsPLAF(plaf, obsWSAF)


vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
plafFile &lt;- system.file("extdata", "labStrains.test.PLAF.txt",
  package = "DEploid"
)
PG0390.deconv &lt;- dEploid(paste(
  "-vcf", vcfFile,
  "-plaf", plafFile, "-noPanel"
))
prop &lt;- PG0390.deconv$Proportions[dim(PG0390.deconv$Proportions)[1], ]
expWSAF &lt;- t(PG0390.deconv$Haps) %*% prop
plotObsExpWSAF(obsWSAF, expWSAF)

# Example 1
refFile &lt;- system.file("extdata", "PG0390-C.test.ref", package = "DEploid")
altFile &lt;- system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
PG0390CoverageT &lt;- extractCoverageFromTxt(refFile, altFile)
obsWSAF &lt;- computeObsWSAF(PG0390CoverageT$altCount, PG0390CoverageT$refCount)

# Example 2
vcfFile &lt;- system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
PG0390CoverageV &lt;- extractCoverageFromVcf(vcfFile, "PG0390-C")
obsWSAF &lt;- computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
