<!DOCTYPE html><html lang="en-US"><head><title>Help for package depower</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {depower}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#depower-package'><p>depower: Power Analysis for Differential Expression Studies</p></a></li>
<li><a href='#glm_nb'><p>GLM for NB ratio of means</p></a></li>
<li><a href='#glmm_bnb'><p>GLMM for BNB ratio of means</p></a></li>
<li><a href='#glmm_poisson'><p>GLMM for Poisson ratio of means</p></a></li>
<li><a href='#lrt_bnb'><p>Likelihood ratio test for BNB ratio of means</p></a></li>
<li><a href='#lrt_nb'><p>Likelihood ratio test for NB ratio of means</p></a></li>
<li><a href='#mle_bnb'><p>MLE for BNB</p></a></li>
<li><a href='#mle_nb'><p>MLE for NB</p></a></li>
<li><a href='#nll_bnb'><p>Negative log-likelihood for BNB</p></a></li>
<li><a href='#nll_nb'><p>Negative log-likelihood for NB</p></a></li>
<li><a href='#plot.depower'><p>Plot power objects</p></a></li>
<li><a href='#power'><p>Simulated power</p></a></li>
<li><a href='#sim_bnb'><p>Simulate data from a BNB distribution</p></a></li>
<li><a href='#sim_log_lognormal'><p>Simulate data from a normal distribution</p></a></li>
<li><a href='#sim_nb'><p>Simulate data from a NB distribution</p></a></li>
<li><a href='#t_test_paired'><p>Paired and one-sample t-Tests</p></a></li>
<li><a href='#t_test_welch'><p>Welch's t-Test</p></a></li>
<li><a href='#wald_test_bnb'><p>Wald test for BNB ratio of means</p></a></li>
<li><a href='#wald_test_nb'><p>Wald test for NB ratio of means</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Power Analysis for Differential Expression Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>2025.1.20</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a convenient framework to simulate, test, power, and visualize 
    data for differential expression studies with lognormal or negative binomial 
    outcomes. Supported designs are two-sample comparisons of independent or 
    dependent outcomes. Power may be summarized in the context of controlling the 
    per-family error rate or family-wise error rate. Negative binomial methods are 
    described in Yu, Fernandez, and Brock (2017) &lt;<a href="https://doi.org/10.1186%2Fs12859-017-1648-2">doi:10.1186/s12859-017-1648-2</a>&gt; 
    and Yu, Fernandez, and Brock (2020) &lt;<a href="https://doi.org/10.1186%2Fs12859-020-3541-7">doi:10.1186/s12859-020-3541-7</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://brettklamer.com/work/depower/">https://brettklamer.com/work/depower/</a>,
<a href="https://bitbucket.org/bklamer/depower/">https://bitbucket.org/bklamer/depower/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rdpack, stats, mvnfast, glmmTMB, dplyr, multidplyr, ggplot2,
scales</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tinytest, rmarkdown</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-21 00:00:00 UTC; x</td>
</tr>
<tr>
<td>Author:</td>
<td>Brett Klamer <a href="https://orcid.org/0000-0002-8334-4831"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Lianbo Yu <a href="https://orcid.org/0000-0002-2025-2585"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Brett Klamer &lt;code@brettklamer.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-23 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='depower-package'>depower: Power Analysis for Differential Expression Studies</h2><span id='topic+depower'></span><span id='topic+depower-package'></span>

<h3>Description</h3>

<p>Provides a convenient framework to simulate, test, power, and visualize data for differential expression studies with lognormal or negative binomial outcomes. Supported designs are two-sample comparisons of independent or dependent outcomes. Power may be summarized in the context of controlling the per-family error rate or family-wise error rate. Negative binomial methods are described in Yu, Fernandez, and Brock (2017) <a href="https://doi.org/10.1186/s12859-017-1648-2">doi:10.1186/s12859-017-1648-2</a> and Yu, Fernandez, and Brock (2020) <a href="https://doi.org/10.1186/s12859-020-3541-7">doi:10.1186/s12859-020-3541-7</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Brett Klamer <a href="mailto:code@brettklamer.com">code@brettklamer.com</a> (<a href="https://orcid.org/0000-0002-8334-4831">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Lianbo Yu <a href="mailto:lianbo.yu@osumc.edu">lianbo.yu@osumc.edu</a> (<a href="https://orcid.org/0000-0002-2025-2585">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://brettklamer.com/work/depower/">https://brettklamer.com/work/depower/</a>
</p>
</li>
<li> <p><a href="https://bitbucket.org/bklamer/depower/">https://bitbucket.org/bklamer/depower/</a>
</p>
</li></ul>


<hr>
<h2 id='glm_nb'>GLM for NB ratio of means</h2><span id='topic+glm_nb'></span>

<h3>Description</h3>

<p>Generalized linear model for two independent negative binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glm_nb(data, equal_dispersion = FALSE, test = "wald", ci_level = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glm_nb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from group 1 and the second element is the vector of negative binomial
values from group 2.
<a href="base.html#topic+NA">NA</a>s are silently excluded. The default output from
<code><a href="#topic+sim_nb">sim_nb()</a></code>.</p>
</td></tr>
<tr><td><code id="glm_nb_+3A_equal_dispersion">equal_dispersion</code></td>
<td>
<p>(Scalar logical: <code>FALSE</code>)<br />
If <code>TRUE</code>, the model is fit assuming both groups have the same
population dispersion parameter. If <code>FALSE</code> (default), the model is
fit assuming different dispersions.</p>
</td></tr>
<tr><td><code id="glm_nb_+3A_test">test</code></td>
<td>
<p>(String: <code>"wald"</code>; <code style="white-space: pre;">&#8288;"c("wald", "lrt")&#8288;</code>)<br />
The statistical method used for the test results. <code>test = "wald"</code>
performs a Wald test and optionally the Wald confidence intervals.
<code>test = "lrt"</code> performs a likelihood ratio test and optionally
the profile likelihood confidence intervals.</p>
</td></tr>
<tr><td><code id="glm_nb_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.
Profile likelihood intervals are computationally intensive, so
intervals from <code>test = "lrt"</code> may be slow.</p>
</td></tr>
<tr><td><code id="glm_nb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code> in the form
</p>
<div class="sourceCode r"><pre>glmmTMB(
  formula = value ~ condition,
  data = data,
  dispformula = ~ condition,
  family = nbinom2
)
</pre></div>
<p>to model independent negative binomial outcomes
<code class="reqn">X_1 \sim \text{NB}(\mu, \theta_1)</code> and <code class="reqn">X_2 \sim \text{NB}(r\mu, \theta_2)</code>
where <code class="reqn">\mu</code> is the mean of group 1, <code class="reqn">r</code> is the ratio of the means of
group 2 with respect to group 1, <code class="reqn">\theta_1</code> is the dispersion parameter
of group 1, and <code class="reqn">\theta_2</code> is the dispersion parameter of group 2.
</p>
<p>The hypotheses for the LRT and Wald test of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: log(r) = 0 \\
H_{alt} &amp;: log(r) \neq 0
\end{aligned}
</code>
</p>

<p>where <code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of
arithmetic means for group 2 with respect to group 1 and
<code class="reqn">log(r_{null}) = 0</code> assumes the population means are identical.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>ratio</code>    </td><td style="text-align: left;"> Estimated ratio of means (group 2 / group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean1</code>    </td><td style="text-align: left;"> Estimated mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean2</code>    </td><td style="text-align: left;"> Estimated mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

7 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> Estimated dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code>    </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>       </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>       </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

8 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> Estimated dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code>    </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>       </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>       </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">

11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code>      </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>test</code>        </td><td style="text-align: left;"> Type of hypothesis test. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> The alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>equal_dispersion</code> </td><td style="text-align: left;"> Whether or not equal dispersions were assumed. </td>
</tr>
<tr>
 <td style="text-align: left;">
15 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ci_level</code>    </td><td style="text-align: left;"> Confidence level of the intervals. </td>
</tr>
<tr>
 <td style="text-align: left;">
16 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>hessian</code>     </td><td style="text-align: left;"> Information about the Hessian matrix. </td>
</tr>
<tr>
 <td style="text-align: left;">
17 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>convergence</code> </td><td style="text-align: left;"> Information about convergence.
</td>
</tr>

</table>



<h3>References</h3>

<p>Hilbe JM (2011).
<em>Negative Binomial Regression</em>, 2 edition.
Cambridge University Press.
ISBN 9780521198158 9780511973420, <a href="https://doi.org/10.1017/CBO9780511973420">doi:10.1017/CBO9780511973420</a>.
</p>
<p>Hilbe JM (2014).
<em>Modeling count data</em>.
Cambridge University Press, New York, NY.
ISBN 9781107028333 9781107611252, <a href="https://doi.org/10.1017/CBO9781139236065">doi:10.1017/CBO9781139236065</a>.
</p>


<h3>See Also</h3>

<p><code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# glm_nb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8
)

lrt &lt;- glm_nb(d, equal_dispersion = FALSE, test = "lrt", ci_level = 0.95)
lrt

wald &lt;- glm_nb(d, equal_dispersion = FALSE, test = "wald", ci_level = 0.95)
wald

#----------------------------------------------------------------------------
# Compare results to manual calculation of chi-square statistic
#----------------------------------------------------------------------------
# Use the same data, but as a data frame instead of list
set.seed(1234)
d &lt;- sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8,
  return_type = "data.frame"
)

mod_alt &lt;- glmmTMB::glmmTMB(
  formula = value ~ condition,
  data = d,
  dispformula = ~ condition,
  family = glmmTMB::nbinom2,
)
mod_null &lt;- glmmTMB::glmmTMB(
  formula = value ~ 1,
  data = d,
  dispformula = ~ condition,
  family = glmmTMB::nbinom2,
)

lrt_chisq &lt;- as.numeric(-2 * (logLik(mod_null) - logLik(mod_alt)))
lrt_chisq
wald_chisq &lt;- summary(mod_alt)$coefficients$cond["condition2", "z value"]^2
wald_chisq

anova(mod_null, mod_alt)

</code></pre>

<hr>
<h2 id='glmm_bnb'>GLMM for BNB ratio of means</h2><span id='topic+glmm_bnb'></span>

<h3>Description</h3>

<p>Generalized linear mixed model for bivariate negative binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm_bnb(data, test = "wald", ci_level = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glmm_bnb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from sample 1 and the second element is the vector of negative
binomial values from sample 2.
Each vector must be sorted by the subject/item index and must be the
same sample size. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default
output from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="glmm_bnb_+3A_test">test</code></td>
<td>
<p>(String: <code>"wald"</code>; <code style="white-space: pre;">&#8288;"c("wald", "lrt")&#8288;</code>)<br />
The statistical method used for the test results. <code>test = "wald"</code>
performs a Wald test and optionally the Wald confidence intervals.
<code>test = "lrt"</code> performs a likelihood ratio test and optionally
the profile likelihood confidence intervals (means and ratio). The
Wald interval is always used for the dispersion and standard deviation
of the item (subject) random intercept.</p>
</td></tr>
<tr><td><code id="glmm_bnb_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.
Profile likelihood intervals are computationally intensive, so
intervals from <code>test = "lrt"</code> may be slow.</p>
</td></tr>
<tr><td><code id="glmm_bnb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code> in the form
</p>
<div class="sourceCode r"><pre>glmmTMB(
  formula = value ~ condition + (1 | item),
  data = data,
  dispformula = ~ 1,
  family = nbinom2
)
</pre></div>
<p>to model dependent negative binomial outcomes
<code class="reqn">X_1, X_2 \sim \text{BNB}(\mu, r, \theta)</code> where <code class="reqn">\mu</code> is the mean of
sample 1, <code class="reqn">r</code> is the ratio of the means of sample 2 with respect to
sample 1, and <code class="reqn">\theta</code> is the dispersion parameter.
</p>
<p>The hypotheses for the LRT and Wald test of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: log(r) = 0 \\
H_{alt} &amp;: log(r) \neq 0
\end{aligned}
</code>
</p>

<p>where <code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of
arithmetic means for sample 2 with respect to sample 1 and
<code class="reqn">log(r_{null}) = 0</code> assumes the population means are identical.
</p>
<p>When simulating data from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>, the mean is a function of the
item (subject) random effect which in turn is a function of the dispersion
parameter. Thus, <code>glmm_bnb()</code> has biased mean and dispersion estimates. The
bias increases as the dispersion parameter gets smaller and decreases as
the dispersion parameter gets larger. However, estimates of the ratio and
standard deviation of the random intercept tend to be accurate. The p-value
for <code>glmm_bnb()</code> is generally overconservative compared to <code>glmm_poisson()</code>,
<code>wald_test_bnb()</code> and <code>lrt_bnb()</code>. In summary, the negative binomial
mixed-effects model fit by <code>glmm_bnb()</code> is not recommended for the BNB data
simulated by <code>sim_bnb()</code>. Instead, <code>wald_test_bnb()</code> or <code>lrt_bnb()</code> should
typically be used instead.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">

1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>ratio</code>    </td><td style="text-align: left;"> Estimated ratio of means (sample 2 / sample 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean1</code>    </td><td style="text-align: left;"> Estimated mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean2</code>    </td><td style="text-align: left;"> Estimated mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

7 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> Estimated dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code>   </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>      </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>      </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

8 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>item_sd</code>  </td><td style="text-align: left;"> Estimated standard deviation of the item
(subject) random intercept. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code>          </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code>          </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code>      </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>test</code>        </td><td style="text-align: left;"> Type of hypothesis test. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> The alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ci_level</code>    </td><td style="text-align: left;"> Confidence level of the interval. </td>
</tr>
<tr>
 <td style="text-align: left;">
15 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>hessian</code>     </td><td style="text-align: left;"> Information about the Hessian matrix. </td>
</tr>
<tr>
 <td style="text-align: left;">
16 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>convergence</code> </td><td style="text-align: left;"> Information about convergence.
</td>
</tr>

</table>



<h3>References</h3>

<p>Hilbe JM (2011).
<em>Negative Binomial Regression</em>, 2 edition.
Cambridge University Press.
ISBN 9780521198158 9780511973420, <a href="https://doi.org/10.1017/CBO9780511973420">doi:10.1017/CBO9780511973420</a>.
</p>
<p>Hilbe JM (2014).
<em>Modeling count data</em>.
Cambridge University Press, New York, NY.
ISBN 9781107028333 9781107611252, <a href="https://doi.org/10.1017/CBO9781139236065">doi:10.1017/CBO9781139236065</a>.
</p>


<h3>See Also</h3>

<p><code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# glmm_bnb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
)

lrt &lt;- glmm_bnb(d, test = "lrt")
lrt

wald &lt;- glmm_bnb(d, test = "wald", ci_level = 0.95)
wald

#----------------------------------------------------------------------------
# Compare results to manual calculation of chi-square statistic
#----------------------------------------------------------------------------
# Use the same data, but as a data frame instead of list
set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2,
  return_type = "data.frame"
)

mod_alt &lt;- glmmTMB::glmmTMB(
  formula = value ~ condition + (1 | item),
  data = d,
  dispformula = ~ 1,
  family = glmmTMB::nbinom2,
)
mod_null &lt;- glmmTMB::glmmTMB(
  formula = value ~ 1 + (1 | item),
  data = d,
  dispformula = ~ 1,
  family = glmmTMB::nbinom2,
)

lrt_chisq &lt;- as.numeric(-2 * (logLik(mod_null) - logLik(mod_alt)))
lrt_chisq
wald_chisq &lt;- summary(mod_alt)$coefficients$cond["condition2", "z value"]^2
wald_chisq

anova(mod_null, mod_alt)

</code></pre>

<hr>
<h2 id='glmm_poisson'>GLMM for Poisson ratio of means</h2><span id='topic+glmm_poisson'></span>

<h3>Description</h3>

<p>Generalized linear mixed model for two dependent Poisson outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm_poisson(data, test = "wald", ci_level = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glmm_poisson_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of Poisson values from sample
1 and the second element is the vector of Poisson values from sample 2.
Each vector must be sorted by the subject/item index and must be the
same sample size. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default
output from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="glmm_poisson_+3A_test">test</code></td>
<td>
<p>(String: <code>"wald"</code>; <code style="white-space: pre;">&#8288;"c("wald", "lrt")&#8288;</code>)<br />
The statistical method used for the test results. <code>test = "wald"</code>
performs a Wald test and optionally the Wald confidence intervals.
<code>test = "lrt"</code> performs a likelihood ratio test and optionally
the profile likelihood confidence intervals (means and ratio). The
Wald interval is always used for the standard deviation of the item
(subject) random intercept.</p>
</td></tr>
<tr><td><code id="glmm_poisson_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.
Profile likelihood intervals are computationally intensive, so
intervals from <code>test = "lrt"</code> may be slow.</p>
</td></tr>
<tr><td><code id="glmm_poisson_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses <code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code> in the form
</p>
<div class="sourceCode r"><pre>glmmTMB(
  formula = value ~ condition + (1 | item),
  data = data,
  family = stats::poisson
)
</pre></div>
<p>to model dependent Poisson outcomes <code class="reqn">X_1 \sim \text{Poisson}(\mu)</code> and
<code class="reqn">X_2 \sim \text{Poisson}(r \mu)</code> where <code class="reqn">\mu</code> is the mean of sample 1
and <code class="reqn">r</code> is the ratio of the means of sample 2 with respect to sample 1.
</p>
<p>The hypotheses for the LRT and Wald test of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: log(r) = 0 \\
H_{alt} &amp;: log(r) \neq 0
\end{aligned}
</code>
</p>

<p>where <code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of
arithmetic means for sample 2 with respect to sample 1 and
<code class="reqn">log(r_{null}) = 0</code> assumes the population means are identical.
</p>
<p>When simulating data from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>, the mean is a function of the
item (subject) random effect which in turn is a function of the dispersion
parameter. Thus, <code>glmm_poisson()</code> has biased mean estimates. The bias
increases as the dispersion parameter gets smaller and decreases as the
dispersion parameter gets larger. However, estimates of the ratio and
standard deviation of the random intercept tend to be accurate. In summary,
the Poisson mixed-effects model fit by <code>glmm_poisson()</code> is not recommended
for the BNB data simulated by <code>sim_bnb()</code>. Instead, <code>wald_test_bnb()</code> or
<code>lrt_bnb()</code> should typically be used instead.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">

1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>ratio</code>    </td><td style="text-align: left;"> Estimated ratio of means (sample 2 / sample 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean1</code>    </td><td style="text-align: left;"> Estimated mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean2</code>    </td><td style="text-align: left;"> Estimated mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

7 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>item_sd</code>  </td><td style="text-align: left;"> Estimated standard deviation of the item
(subject) random intercept. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

8 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code>           </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code>           </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
10  </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code>     </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>test</code>        </td><td style="text-align: left;"> Type of hypothesis test. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> The alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ci_level</code>    </td><td style="text-align: left;"> Confidence level of the interval. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>hessian</code>     </td><td style="text-align: left;"> Information about the Hessian matrix. </td>
</tr>
<tr>
 <td style="text-align: left;">
15 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>convergence</code> </td><td style="text-align: left;"> Information about convergence.
</td>
</tr>

</table>



<h3>References</h3>

<p>Hilbe JM (2011).
<em>Negative Binomial Regression</em>, 2 edition.
Cambridge University Press.
ISBN 9780521198158 9780511973420, <a href="https://doi.org/10.1017/CBO9780511973420">doi:10.1017/CBO9780511973420</a>.
</p>
<p>Hilbe JM (2014).
<em>Modeling count data</em>.
Cambridge University Press, New York, NY.
ISBN 9781107028333 9781107611252, <a href="https://doi.org/10.1017/CBO9781139236065">doi:10.1017/CBO9781139236065</a>.
</p>


<h3>See Also</h3>

<p><code><a href="glmmTMB.html#topic+glmmTMB">glmmTMB::glmmTMB()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# glmm_poisson() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
)

lrt &lt;- glmm_poisson(d, test = "lrt")
lrt

wald &lt;- glmm_poisson(d, test = "wald", ci_level = 0.95)
wald

#----------------------------------------------------------------------------
# Compare results to manual calculation of chi-square statistic
#----------------------------------------------------------------------------
# Use the same data, but as a data frame instead of list
set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2,
  return_type = "data.frame"
)

mod_alt &lt;- glmmTMB::glmmTMB(
  formula = value ~ condition + (1 | item),
  data = d,
  family = stats::poisson,
)
mod_null &lt;- glmmTMB::glmmTMB(
  formula = value ~ 1 + (1 | item),
  data = d,
  family = stats::poisson,
)

lrt_chisq &lt;- as.numeric(-2 * (logLik(mod_null) - logLik(mod_alt)))
lrt_chisq
wald_chisq &lt;- summary(mod_alt)$coefficients$cond["condition2", "z value"]^2
wald_chisq

anova(mod_null, mod_alt)

</code></pre>

<hr>
<h2 id='lrt_bnb'>Likelihood ratio test for BNB ratio of means</h2><span id='topic+lrt_bnb'></span>

<h3>Description</h3>

<p>Likelihood ratio test for the ratio of means from bivariate negative binomial
outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrt_bnb(data, ratio_null = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lrt_bnb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from sample 1 and the second element is the vector of negative
binomial values from sample 2.
Each vector must be sorted by the subject/item index and must be the
same sample size. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default
output from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="lrt_bnb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (sample 2 / sample 1).
Typically, <code>ratio_null = 1</code> (no difference). See 'Details' for
additional information.</p>
</td></tr>
<tr><td><code id="lrt_bnb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the MLE function <code><a href="#topic+mle_bnb">mle_bnb()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \mid G = g \sim \text{Poisson}(\mu g)</code> and
<code class="reqn">X_2 \mid G = g \sim \text{Poisson}(r \mu g)</code> where
<code class="reqn">G \sim \text{Gamma}(\theta, \theta^{-1})</code> is the random item (subject)
effect. Then <code class="reqn">X_1, X_2 \sim \text{BNB}(\mu, r, \theta)</code> is the joint
distribution where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are dependent (though conditionally
independent), <code class="reqn">X_1</code> is the count outcome for sample 1 of the items
(subjects), <code class="reqn">X_2</code> is the count outcome for sample 2 of the items (subjects),
<code class="reqn">\mu</code> is the conditional mean of sample 1, <code class="reqn">r</code> is the ratio of the
conditional means of sample 2 with respect to sample 1, and <code class="reqn">\theta</code> is
the gamma distribution shape parameter which controls the dispersion and the
correlation between sample 1 and 2.
</p>
<p>The hypotheses for the LRT of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: r = r_{null} \\
H_{alt} &amp;: r \neq r_{null}
\end{aligned}
</code>
</p>

<p>where <code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of
arithmetic means for sample 2 with respect to sample 1 and <code class="reqn">r_{null}</code> is
a constant for the assumed null population ratio of means (typically
<code class="reqn">r_{null} = 1</code>).
</p>
<p>The LRT statistic is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
\lambda &amp;= -2 \ln \frac{\text{sup}_{\Theta_{null}} L(r, \mu, \theta)}{\text{sup}_{\Theta} L(r, \mu, \theta)} \\
  &amp;= -2 \left[ \ln \text{sup}_{\Theta_{null}} L(r, \mu, \theta) - \ln \text{sup}_{\Theta} L(r, \mu, \theta) \right] \\
  &amp;= -2(l(r_{null}, \tilde{\mu}, \tilde{\theta}) - l(\hat{r}, \hat{\mu}, \hat{\theta}))
\end{aligned}
</code>
</p>

<p>Under <code class="reqn">H_{null}</code>, the LRT test statistic is asymptotically distributed
as <code class="reqn">\chi^2_1</code>. The approximate level <code class="reqn">\alpha</code> test rejects
<code class="reqn">H_{null}</code> if <code class="reqn">\lambda \geq \chi^2_1(1 - \alpha)</code>. Note that
the asymptotic critical value is known to underestimate the exact critical
value. Hence, the nominal significance level may not be achieved for small
sample sizes (possibly <code class="reqn">n \leq 10</code> or <code class="reqn">n \leq 50</code>).
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Estimated ratio of means (sample 2 / sample 1). </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> Point estimates under the alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Estimated mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Estimated mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> Estimated dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">

6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>null</code> </td><td style="text-align: left;"> Point estimates under the null hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Estimated mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Estimated mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> Estimated dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">

7 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> The sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> The sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Assumed population ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Information from the optimizer.
</td>
</tr>

</table>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# lrt_bnb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
) |&gt;
  lrt_bnb()

</code></pre>

<hr>
<h2 id='lrt_nb'>Likelihood ratio test for NB ratio of means</h2><span id='topic+lrt_nb'></span>

<h3>Description</h3>

<p>Likelihood ratio test for the ratio of means from two independent negative
binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lrt_nb(data, equal_dispersion = FALSE, ratio_null = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lrt_nb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from group 1 and the second element is the vector of negative binomial
values from group 2.
<a href="base.html#topic+NA">NA</a>s are silently excluded. The default output from
<code><a href="#topic+sim_nb">sim_nb()</a></code>.</p>
</td></tr>
<tr><td><code id="lrt_nb_+3A_equal_dispersion">equal_dispersion</code></td>
<td>
<p>(Scalar logical: <code>FALSE</code>)<br />
If <code>TRUE</code>, the LRT is calculated assuming both groups have the same
population dispersion parameter. If <code>FALSE</code> (default), the LRT is
calculated assuming different dispersions.</p>
</td></tr>
<tr><td><code id="lrt_nb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (group 2 / group 1).
Typically <code>ratio_null = 1</code> (no difference). See 'Details' for
additional information.</p>
</td></tr>
<tr><td><code id="lrt_nb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the MLE function <code><a href="#topic+mle_nb">mle_nb()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \sim NB(\mu, \theta_1)</code> and
<code class="reqn">X_2 \sim NB(r\mu, \theta_2)</code> where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are
independent, <code class="reqn">X_1</code> is the count outcome for items in group 1, <code class="reqn">X_2</code>
is the count outcome for items in group 2, <code class="reqn">\mu</code> is the arithmetic mean
count in group 1, <code class="reqn">r</code> is the ratio of arithmetic means for group 2 with
respect to group 1, <code class="reqn">\theta_1</code> is the dispersion parameter of group 1,
and <code class="reqn">\theta_2</code> is the dispersion parameter of group 2.
</p>
<p>The hypotheses for the LRT of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: r = r_{null} \\
H_{alt} &amp;: r \neq r_{null}
\end{aligned}
</code>
</p>

<p>where <code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of
arithmetic means for group 2 with respect to group 1 and <code class="reqn">r_{null}</code> is a
constant for the assumed null population ratio of means (typically
<code class="reqn">r_{null} = 1</code>).
</p>
<p>The LRT statistic is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
\lambda &amp;= -2 \ln \frac{\text{sup}_{\Theta_{null}} L(r, \mu, \theta_1, \theta_2)}{\text{sup}_{\Theta} L(r, \mu, \theta_1, \theta_2)} \\
  &amp;= -2 \left[ \ln \text{sup}_{\Theta_{null}} L(r, \mu, \theta_1, \theta_2) - \ln \text{sup}_{\Theta} L(r, \mu, \theta_1, \theta_2) \right] \\
  &amp;= -2(l(r_{null}, \tilde{\mu}, \tilde{\theta}_1, \tilde{\theta}_2) - l(\hat{r}, \hat{\mu}, \hat{\theta}_1, \hat{\theta}_2))
\end{aligned}
</code>
</p>

<p>Under <code class="reqn">H_{null}</code>, the LRT test statistic is asymptotically distributed
as <code class="reqn">\chi^2_1</code>. The approximate level <code class="reqn">\alpha</code> test rejects
<code class="reqn">H_{null}</code> if <code class="reqn">\lambda \geq \chi^2_1(1 - \alpha)</code>. Note that
the asymptotic critical value is known to underestimate the exact critical
value. Hence, the nominal significance level may not be achieved for small
sample sizes (possibly <code class="reqn">n \leq 10</code> or <code class="reqn">n \leq 50</code>).
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Estimated ratio of means (group 2 / group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> Point estimates under the alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>mean1</code>       </td><td style="text-align: left;"> Estimated mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>mean2</code>       </td><td style="text-align: left;"> Estimated mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> Estimated dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> Estimated dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">

6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>null</code> </td><td style="text-align: left;"> Point estimates under the null hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>mean1</code>       </td><td style="text-align: left;"> Estimated mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>mean2</code>       </td><td style="text-align: left;"> Estimated mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> Estimated dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> Estimated dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">

7 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>equal_dispersion</code> </td><td style="text-align: left;"> Whether or not equal dispersions were assumed. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Assumed population ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Information from the optimizer.
</td>
</tr>

</table>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# lrt_nb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8
) |&gt;
  lrt_nb()

</code></pre>

<hr>
<h2 id='mle_bnb'>MLE for BNB</h2><span id='topic+mle_bnb'></span><span id='topic+mle_bnb_null'></span><span id='topic+mle_bnb_alt'></span>

<h3>Description</h3>

<p>Maximum likelihood estimates (MLE) for bivariate negative binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_bnb_null(data, ratio_null = 1, method = "nlm_constrained", ...)

mle_bnb_alt(data, method = "nlm_constrained", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_bnb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from sample 1 and the second element is the vector of negative
binomial values from sample 2.
Each vector must be sorted by the subject/item index and must be the
same sample size. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default
output from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="mle_bnb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (sample 2 / sample 1).
Typically <code>ratio_null = 1</code> (no difference).</p>
</td></tr>
<tr><td><code id="mle_bnb_+3A_method">method</code></td>
<td>
<p>(string: <code>"nlm_constrained"</code>)<br />
The optimization method. Must choose one of <code>"nlm"</code>,
<code>"nlm_constrained"</code>, <code>"optim"</code>, or <code>"optim_constrained"</code>. The default
bounds for constrained optimization are <code style="white-space: pre;">&#8288;[1e-03, 1e06]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="mle_bnb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the optimization method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \mid G = g \sim \text{Poisson}(\mu g)</code> and
<code class="reqn">X_2 \mid G = g \sim \text{Poisson}(r \mu g)</code> where
<code class="reqn">G \sim \text{Gamma}(\theta, \theta^{-1})</code> is the random item (subject) effect.
Then <code class="reqn">X_1, X_2 \sim \text{BNB}(\mu, r, \theta)</code> is the joint distribution where
<code class="reqn">X_1</code> and <code class="reqn">X_2</code> are dependent (though conditionally independent),
<code class="reqn">X_1</code> is the count outcome for sample 1 of the items (subjects),
<code class="reqn">X_2</code> is the count outcome for sample 2 of the items (subjects),
<code class="reqn">\mu</code> is the conditional mean of sample 1, <code class="reqn">r</code> is the ratio of the
conditional means of sample 2 with respect to sample 1, and <code class="reqn">\theta</code> is
the gamma distribution shape parameter which controls the dispersion and the
correlation between sample 1 and 2.
</p>
<p>The MLEs of <code class="reqn">r</code> and <code class="reqn">\mu</code> are <code class="reqn">\hat{r} = \frac{\bar{x}_2}{\bar{x}_1}</code>
and <code class="reqn">\hat{\mu} = \bar{x}_1</code>. The MLE of <code class="reqn">\theta</code> is found by
maximizing the profile log-likelihood
<code class="reqn">l(\hat{r}, \hat{\mu}, \theta)</code> with respect to <code class="reqn">\theta</code>. When
<code class="reqn">r = r_{null}</code> is known, the MLE of <code class="reqn">\mu</code> is
<code class="reqn">\tilde{\mu} = \frac{\bar{x}_1 + \bar{x}_2}{1 + r_{null}}</code> and
<code class="reqn">\tilde{\theta}</code> is obtained by maximizing the profile log-likelihood
<code class="reqn">l(r_{null}, \tilde{\mu}, \theta)</code> with respect to <code class="reqn">\theta</code>.
</p>
<p>The backend method for numerical optimization is controlled by argument
<code>method</code> which refers to <code><a href="stats.html#topic+nlm">stats::nlm()</a></code>, <code><a href="stats.html#topic+nlminb">stats::nlminb()</a></code>, or
<code><a href="stats.html#topic+optim">stats::optim()</a></code>. If you would like to see warnings from the optimizer,
include argument <code>warnings = TRUE</code>.
</p>


<h3>Value</h3>


<ul>
<li><p> For <code>mle_bnb_alt</code>, a list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> MLE for mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> MLE for mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> MLE for ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> MLE for BNB dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>nll</code> </td><td style="text-align: left;"> Minimum of negative log-likelihood. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>nparams</code> </td><td style="text-align: left;"> Number of estimated parameters. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> <code>mle_method</code> </td><td style="text-align: left;"> Method used for optimization. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Additional information from the optimizer.
</td>
</tr>

</table>

</li>
<li><p> For <code>mle_bnb_null</code>, a list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> MLE for mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> MLE for mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Population ratio of means assumed for null hypothesis.
<code>mean2 = mean1 * ratio_null</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> MLE for BNB dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>nll</code> </td><td style="text-align: left;"> Minimum of negative log-likelihood. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>nparams</code> </td><td style="text-align: left;"> Number of estimated parameters. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> <code>mle_method</code> </td><td style="text-align: left;"> Method used for optimization. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Additional information from the optimizer.
</td>
</tr>

</table>

</li></ul>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_bnb">sim_bnb()</a></code>, <a href="#topic+nll_bnb">nll_bnb</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# mle_bnb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
)

mle_alt &lt;- d |&gt;
  mle_bnb_alt()

mle_null &lt;- d |&gt;
  mle_bnb_null()

mle_alt
mle_null

</code></pre>

<hr>
<h2 id='mle_nb'>MLE for NB</h2><span id='topic+mle_nb'></span><span id='topic+mle_nb_null'></span><span id='topic+mle_nb_alt'></span>

<h3>Description</h3>

<p>Maximum likelihood estimates (MLE) for two independent negative binomial
outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_nb_null(
  data,
  equal_dispersion = FALSE,
  ratio_null = 1,
  method = "nlm_constrained",
  ...
)

mle_nb_alt(data, equal_dispersion = FALSE, method = "nlm_constrained", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_nb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from group 1 and the second element is the vector of negative binomial
values from group 2.
<a href="base.html#topic+NA">NA</a>s are silently excluded. The default output from
<code><a href="#topic+sim_nb">sim_nb()</a></code>.</p>
</td></tr>
<tr><td><code id="mle_nb_+3A_equal_dispersion">equal_dispersion</code></td>
<td>
<p>(Scalar logical: <code>FALSE</code>)<br />
If <code>TRUE</code>, the MLEs are calculated assuming both groups have the same
population dispersion parameter. If <code>FALSE</code> (default), the MLEs are
calculated assuming different dispersions.</p>
</td></tr>
<tr><td><code id="mle_nb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (group 2 / group 1).
Typically <code>ratio_null = 1</code> (no difference).</p>
</td></tr>
<tr><td><code id="mle_nb_+3A_method">method</code></td>
<td>
<p>(string: <code>"nlm_constrained"</code>)<br />
The optimization method. Must choose one of <code>"nlm"</code>,
<code>"nlm_constrained"</code>, <code>"optim"</code>, or <code>"optim_constrained"</code>. The default
bounds for constrained optimization are <code style="white-space: pre;">&#8288;[1e-03, 1e06]&#8288;</code>.</p>
</td></tr>
<tr><td><code id="mle_nb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the optimization method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \sim \text{NB}(\mu, \theta_1)</code> and
<code class="reqn">X_2 \sim \text{NB}(r\mu, \theta_2)</code>, where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are
independent, <code class="reqn">X_1</code> is the count outcome for items in group 1, <code class="reqn">X_2</code>
is the count outcome for items in group 2, <code class="reqn">\mu</code> is the arithmetic mean
count in group 1, <code class="reqn">r</code> is the ratio of arithmetic means for group 2 with
respect to group 1, <code class="reqn">\theta_1</code> is the dispersion parameter of group 1,
and <code class="reqn">\theta_2</code> is the dispersion parameter of group 2.
</p>
<p>The MLEs of <code class="reqn">r</code> and <code class="reqn">\mu</code> are <code class="reqn">\hat{r} = \frac{\bar{x}_2}{\bar{x}_1}</code>
and <code class="reqn">\hat{\mu} = \bar{x}_1</code>. The MLEs of <code class="reqn">\theta_1</code> and <code class="reqn">\theta_2</code>
are found by maximizing the profile log-likelihood
<code class="reqn">l(\hat{r}, \hat{\mu}, \theta_1, \theta_2)</code> with respect to
<code class="reqn">\theta_1</code> and <code class="reqn">\theta_2</code>. When <code class="reqn">r = r_{null}</code> is known, the MLE
of <code class="reqn">\mu</code> is
<code class="reqn">\tilde{\mu} = \frac{n_1 \bar{x}_1 + n_2 \bar{x}_2}{n_1 + n_2}</code> and
<code class="reqn">\tilde{\theta}_1</code> and <code class="reqn">\tilde{\theta}_2</code> are obtained by maximizing
the profile log-likelihood <code class="reqn">l(r_{null}, \tilde{\mu}, \theta_1, \theta_2)</code>.
</p>
<p>The backend method for numerical optimization is controlled by argument
<code>method</code> which refers to <code><a href="stats.html#topic+nlm">stats::nlm()</a></code>, <code><a href="stats.html#topic+nlminb">stats::nlminb()</a></code>, or
<code><a href="stats.html#topic+optim">stats::optim()</a></code>. If you would like to see warnings from the optimizer,
include argument <code>warnings = TRUE</code>.
</p>


<h3>Value</h3>


<ul>
<li><p> For <code>mle_nb_alt()</code>, a list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> MLE for mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> MLE for mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> MLE for ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> MLE for dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> MLE for dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>equal_dispersion</code> </td><td style="text-align: left;"> Were equal dispersions assumed. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>nll</code> </td><td style="text-align: left;"> Minimum of negative log-likelihood. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> <code>nparams</code> </td><td style="text-align: left;"> Number of estimated parameters. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> <code>mle_method</code> </td><td style="text-align: left;"> Method used for optimization. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Additional information from the optimizer.
</td>
</tr>

</table>

</li>
<li><p> For <code>mle_nb_null()</code>, a list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> MLE for mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> MLE for mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Population ratio of means assumed for null hypothesis.
<code>mean2 = mean1 * ratio_null</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> MLE for dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> MLE for dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>equal_dispersion</code> </td><td style="text-align: left;"> Were equal dispersions assumed. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>nll</code> </td><td style="text-align: left;"> Minimum of negative log-likelihood. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> <code>nparams</code> </td><td style="text-align: left;"> Number of estimated parameters. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> <code>mle_method</code> </td><td style="text-align: left;"> Method used for optimization. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Additional information from the optimizer.
</td>
</tr>

</table>

</li></ul>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_nb">sim_nb()</a></code>, <a href="#topic+nll_nb">nll_nb</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# mle_nb() examples
#----------------------------------------------------------------------------
library(depower)

d &lt;- sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8
)

mle_alt &lt;- d |&gt;
  mle_nb_alt()

mle_null &lt;- d |&gt;
  mle_nb_null()

mle_alt
mle_null

</code></pre>

<hr>
<h2 id='nll_bnb'>Negative log-likelihood for BNB</h2><span id='topic+nll_bnb'></span><span id='topic+nll_bnb_null'></span><span id='topic+nll_bnb_alt'></span>

<h3>Description</h3>

<p>The negative log-likelihood for bivariate negative binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nll_bnb_null(param, value1, value2, ratio_null)

nll_bnb_alt(param, value1, value2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nll_bnb_+3A_param">param</code></td>
<td>
<p>(numeric)<br />
The vector of initial values for BNB parameters. Must be in the
following order for each scenario:
</p>

<ul>
<li><p> Null: <code>c(mean1, dispersion)</code>
</p>
</li>
<li><p> Alternative: <code>c(mean1, mean2, dispersion)</code>
</p>
</li></ul>

<p>for samples 1 and 2.</p>
</td></tr>
<tr><td><code id="nll_bnb_+3A_value1">value1</code></td>
<td>
<p>(numeric)<br />
The vector of BNB values from sample 1. Must be sorted by the
subject/item index. Must not contain <a href="base.html#topic+NA">NA</a>s.</p>
</td></tr>
<tr><td><code id="nll_bnb_+3A_value2">value2</code></td>
<td>
<p>(numeric)<br />
The vector of BNB values from sample 2. Must be sorted by the
subject/item index. Must not contain <a href="base.html#topic+NA">NA</a>s.</p>
</td></tr>
<tr><td><code id="nll_bnb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (sample 2 / sample 1).
Typically <code>ratio_null = 1</code> (no difference).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are primarily designed for speed in simulation. Arguments are
not checked.
</p>
<p>Suppose <code class="reqn">X_1 \mid G = g \sim \text{Poisson}(\mu g)</code> and
<code class="reqn">X_2 \mid G = g \sim \text{Poisson}(r \mu g)</code> where
<code class="reqn">G \sim \text{Gamma}(\theta, \theta^{-1})</code> is the random item (subject) effect.
Then <code class="reqn">X_1, X_2 \sim \text{BNB}(\mu, r, \theta)</code> is the joint distribution where
<code class="reqn">X_1</code> and <code class="reqn">X_2</code> are dependent (though conditionally independent),
<code class="reqn">X_1</code> is the count outcome for sample 1 of the items (subjects),
<code class="reqn">X_2</code> is the count outcome for sample 2 of the items (subjects),
<code class="reqn">\mu</code> is the conditional mean of sample 1, <code class="reqn">r</code> is the ratio of the
conditional means of sample 2 with respect to sample 1, and <code class="reqn">\theta</code> is
the gamma distribution shape parameter which controls the dispersion and the
correlation between sample 1 and 2.
</p>
<p>The likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
L(r, \mu, \theta \mid X_1, X_2) = &amp; \left( \frac{\theta^{\theta}}{\Gamma(\theta)} \right)^{n} \times \\
  &amp; \frac{\mu^{\sum{x_{1i}} + \sum{x_{2i}}}}{\prod_{i=1}^{n} x_{1i}!} \frac{r^{\sum{x_{2i}}}}{\prod_{i=1}^{n} x_{2i}!} \times \\
  &amp; \frac{\prod_{i = 1}^{n} \Gamma(x_{1i} + x_{2i} + \theta)}{(\mu + r \mu + \theta)^{\sum x_{1i} + x_{2i} + \theta}}
\end{aligned}
</code>
</p>

<p>and the parameter space is
<code class="reqn">\Theta = \left\{ (r, \mu, \theta) : r, \mu, \theta &gt; 0 \right\}</code>.
The log-likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
l(r, \mu, \theta) = \ &amp; n \left[ \theta \ln \theta - \ln \Gamma(\theta) \right] + \\
  &amp; n (\bar{x}_1 + \bar{x}_2) \ln(\mu) + n \bar{x}_2 \ln r + \\
  &amp; \sum_{i=1}^{n}{\ln \Gamma(x_{1i} + x_{2i} + \theta)} - \\
  &amp; n (\bar{x}_1 + \bar{x}_2 + \theta) \ln(\mu + r\mu + \theta) - \\
  &amp; \sum_{i = 1}^{n}{\ln x_{1i}!} - \sum_{i = 1}^{n}{\ln x_{2i}!}
\end{aligned}
</code>
</p>



<h3>Value</h3>

<p>Scalar numeric negative log-likelihood.
</p>


<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_nb">sim_nb()</a></code>, <code><a href="stats.html#topic+nlminb">stats::nlminb()</a></code>, <code><a href="stats.html#topic+nlm">stats::nlm()</a></code>,
<code><a href="stats.html#topic+optim">stats::optim()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# nll_bnb*() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
)

nll_bnb_alt(
  param = c(mean1 = 10, mean2 = 12, dispersion = 2),
  value1 = d[[1L]],
  value2 = d[[2L]]
)

nll_bnb_null(
  param = c(mean = 10, dispersion = 2),
  value1 = d[[1L]],
  value2 = d[[2L]],
  ratio_null = 1
)

</code></pre>

<hr>
<h2 id='nll_nb'>Negative log-likelihood for NB</h2><span id='topic+nll_nb'></span><span id='topic+nll_nb_null'></span><span id='topic+nll_nb_alt'></span>

<h3>Description</h3>

<p>The negative log-likelihood for two independent samples of negative binomial
distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nll_nb_null(param, value1, value2, equal_dispersion, ratio_null)

nll_nb_alt(param, value1, value2, equal_dispersion)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nll_nb_+3A_param">param</code></td>
<td>
<p>(numeric)<br />
The vector of initial values for NB parameters. Must be in the
following order for each scenario:
</p>

<ul>
<li><p> Null and unequal dispersion: <code>c(mean1, dispersion1, dispersion2)</code>
</p>
</li>
<li><p> Alternative and unequal dispersion: <code>c(mean1, mean2, dispersion1, dispersion2)</code>
</p>
</li>
<li><p> Null and equal dispersion: <code>c(mean1, dispersion)</code>
</p>
</li>
<li><p> Alternative and equal dispersion: <code>c(mean1, mean2, dispersion)</code>
</p>
</li></ul>

<p>for groups 1 and 2.</p>
</td></tr>
<tr><td><code id="nll_nb_+3A_value1">value1</code></td>
<td>
<p>(numeric)<br />
The vector of NB values from group 1. Must not contain
<a href="base.html#topic+NA">NA</a>s.</p>
</td></tr>
<tr><td><code id="nll_nb_+3A_value2">value2</code></td>
<td>
<p>(numeric)<br />
The vector of NB values from group 2. Must not contain
<a href="base.html#topic+NA">NA</a>s.</p>
</td></tr>
<tr><td><code id="nll_nb_+3A_equal_dispersion">equal_dispersion</code></td>
<td>
<p>(Scalar logical)<br />
If <code>TRUE</code>, the log-likelihood is calculated assuming both groups have
the same population dispersion parameter. If <code>FALSE</code> (default), the
log-likelihood is calculated assuming different dispersions.</p>
</td></tr>
<tr><td><code id="nll_nb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The ratio of means assumed under the null hypothesis (group 2 / group 1).
Typically <code>ratio_null = 1</code> (no difference).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are primarily designed for speed in simulation. Arguments are
not checked.
</p>
<p>Suppose <code class="reqn">X_1 \sim \text{NB}(\mu, \theta_1)</code> and
<code class="reqn">X_2 \sim \text{NB}(r\mu, \theta_2)</code> where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are
independent, <code class="reqn">X_1</code> is the count outcome for items in group 1, <code class="reqn">X_2</code>
is the count outcome for items in group 2, <code class="reqn">\mu</code> is the arithmetic mean
count in group 1, <code class="reqn">r</code> is the ratio of arithmetic means for group 2 with
respect to group 1, <code class="reqn">\theta_1</code> is the dispersion parameter of group 1,
and <code class="reqn">\theta_2</code> is the dispersion parameter of group 2.
</p>


<h4>Unequal dispersion parameters</h4>

<p>When the dispersion parameters are not equal, the likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
L(r, \mu, \theta_1, \theta_2 \mid X_1, X_2) = &amp; \left( \frac{\theta_1^{\theta_1}}{\Gamma(\theta_1)} \right)^{n_1} \frac{\mu^{\sum{x_{1i}}}}{(\mu + \theta_1)^{\sum{x_{1i} + n_1 \theta_1}}} \times \\
  &amp; \left( \frac{\theta_2^{\theta_2}}{\Gamma(\theta_2)} \right)^{n_2} \frac{(r \mu)^{\sum{x_{2j}}}}{(r \mu + \theta_2)^{\sum{x_{2j} + n_2 \theta_2}}} \times \\
  &amp; \prod_{i = 1}^{n_1}{\frac{\Gamma(x_{1i} + \theta_1)}{x_{1i}!}} \prod_{j = 1}^{n_2}{\frac{\Gamma(x_{2j} + \theta_2)}{x_{2j}!}}
\end{aligned}
</code>
</p>

<p>and the parameter space is
<code class="reqn">\Theta = \left\{ (r, \mu, \theta_1, \theta_2) : r, \mu, \theta_1, \theta_2 &gt; 0 \right\}</code>.
The log-likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
l(r, \mu, \theta_1, \theta_2) = \ &amp;n_1 \left[ \theta_1 \ln \theta_1 - \ln \Gamma(\theta_1) \right] + \\
  &amp;n_2 \left[ \theta_2 \ln \theta_2 - \ln \Gamma(\theta_2) \right] + \\
  &amp;(n_1 \bar{x}_1 + n_2 \bar{x}_2) \ln(\mu) - n_1 (\bar{x}_1 + \theta_1) \ln(\mu + \theta_1) + \\
  &amp;n_2 \bar{x}_2 \ln(r) - n_2 (\bar{x}_2 + \theta_2) \ln(r \mu + \theta_2) + \\
  &amp;\sum_{i = 1}^{n_1}{\left( \ln \Gamma(y_{1i} + \theta_1) - \ln(y_{1i}!) \right)} + \\
  &amp;\sum_{j = 1}^{n_2}{\left( \ln \Gamma(y_{2j} + \theta_2) - \ln(y_{2j}!) \right)}
\end{aligned}
</code>
</p>




<h4>Equal dispersion parameters</h4>

<p>When the dispersion parameters are equal, the likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
L(r, \mu, \theta \mid X_1, X_2) = &amp; \left( \frac{\theta^{\theta}}{\Gamma(\theta)} \right)^{n_1 + n_2} \times \\
  &amp; \frac{\mu^{\sum{x_{1i}}}}{(\mu + \theta)^{\sum{x_{1i} + n_1 \theta}}} \frac{(r \mu)^{\sum{x_{2j}}}}{(r \mu + \theta)^{\sum{x_{2j} + n_2 \theta}}} \times \\
  &amp; \prod_{i = 1}^{n_1}{\frac{\Gamma(x_{1i} + \theta)}{x_{1i}!}} \prod_{j = 1}^{n_2}{\frac{\Gamma(x_{2j} + \theta)}{x_{2j}!}}
\end{aligned}
</code>
</p>

<p>and the parameter space is
<code class="reqn">\Theta = \left\{ (r, \mu, \theta) : r, \mu, \theta &gt; 0 \right\}</code>.
The log-likelihood is
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
l(r, \mu, \theta) = \ &amp;(n_1 + n_2) \left[ \theta \ln \theta - \ln \Gamma(\theta) \right] + \\
  &amp;(n_1 \bar{x}_1 + n_2 \bar{x}_2) \ln(\mu) - n_1 (\bar{x}_1 + \theta) \ln(\mu + \theta) + \\
  &amp;n_2 \bar{x}_2 \ln(r) - n_2 (\bar{x}_2 + \theta) \ln(r \mu + \theta) + \\
  &amp;\sum_{i = 1}^{n_1}{\left( \ln \Gamma(y_{1i} + \theta) - \ln(y_{1i}!) \right)} + \\
  &amp;\sum_{j = 1}^{n_2}{\left( \ln \Gamma(y_{2j} + \theta) - \ln(y_{2j}!) \right)}
\end{aligned}
</code>
</p>




<h3>Value</h3>

<p>Scalar numeric negative log-likelihood.
</p>


<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_nb">sim_nb()</a></code>, <code><a href="stats.html#topic+nlminb">stats::nlminb()</a></code>, <code><a href="stats.html#topic+nlm">stats::nlm()</a></code>,
<code><a href="stats.html#topic+optim">stats::optim()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# nll_nb_*() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
d &lt;- sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8
)

nll_nb_alt(
  param = c(mean1 = 10, mean2 = 15, dispersion = 2, dispersion2 = 8),
  value1 = d[[1L]],
  value2 = d[[2L]],
  equal_dispersion = FALSE
)

nll_nb_null(
  param = c(mean = 10, dispersion = 2, dispersion2 = 8),
  value1 = d[[1L]],
  value2 = d[[2L]],
  equal_dispersion = FALSE,
  ratio_null = 1
)

</code></pre>

<hr>
<h2 id='plot.depower'>Plot power objects</h2><span id='topic+plot.depower'></span>

<h3>Description</h3>

<p>An automatic plot method for objects returned by <code><a href="#topic+power">power()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'depower'
plot(
  x,
  x_axis = NULL,
  y_axis = NULL,
  color = NULL,
  facet_row = NULL,
  facet_col = NULL,
  hline = NULL,
  caption = TRUE,
  caption_width = 70L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.depower_+3A_x">x</code></td>
<td>
<p>(depower)<br />
The data frame returned by <code><a href="#topic+power">power()</a></code>.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_x_axis">x_axis</code></td>
<td>
<p>(string: <code>NULL</code>; <code>names(x)</code>)<br />
The name of the column to be used for the x-axis. Automatically chosen
if <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_y_axis">y_axis</code></td>
<td>
<p>(string: <code>NULL</code>; <code>names(x)</code>)<br />
The name of the column to be used for the y-axis. Automatically chosen
if <code>NULL</code>. Generally, <code>"power"</code> (default) should be used for the y-axis.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_color">color</code></td>
<td>
<p>(string: <code>NULL</code>; <code>names(x)</code>)<br />
The name of the column to be used for the <code><a href="ggplot2.html#topic+aes">ggplot2::aes()</a></code> color
aesthetic. Automatically chosen if <code>NULL</code>. Use <code>NA</code> to turn off.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_facet_row">facet_row</code></td>
<td>
<p>(string: <code>NULL</code>; <code>names(x)</code>)<br />
The name of the column to be used for the <code><a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid()</a></code> row.
Automatically chosen if <code>NULL</code>. Use <code>NA</code> to turn off.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_facet_col">facet_col</code></td>
<td>
<p>(string: <code>NULL</code>; <code>names(x)</code>)<br />
The name of the column to be used for the <code><a href="ggplot2.html#topic+facet_grid">ggplot2::facet_grid()</a></code>
column. Automatically chosen if <code>NULL</code>. Use <code>NA</code> to turn off.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_hline">hline</code></td>
<td>
<p>(numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
The y-intercept at which to draw a horizontal line.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_caption">caption</code></td>
<td>
<p>(Scalar logical: <code>TRUE</code>)<br />
If <code>TRUE</code> (default), a caption is added to the plot. The caption
includes information on parameter values that were conditioned on to
generate the plot. If <code>FALSE</code>, the caption is not included.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_caption_width">caption_width</code></td>
<td>
<p>(Scalar integer: <code>70L</code>)<br />
The target column number for wrapping the caption text.</p>
</td></tr>
<tr><td><code id="plot.depower_+3A_...">...</code></td>
<td>
<p>Unused additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you are limited by the output from <code>plot.depower()</code>, keep in mind that the
object returned by <code><a href="#topic+power">power()</a></code> is a standard data frame. This allows
you to easily plot all results with standard plotting functions. In addition,
because <code>plot.depower()</code> uses ggplot2, you can modify the plot as you
normally would. For example:
</p>
<div class="sourceCode r"><pre>set.seed(1234)
sim_log_lognormal(
  n1 = c(10, 15),
  n2 = c(10, 15),
  ratio = c(1.3, 1.5),
  cv1 = c(0.3),
  cv2 = c(0.3, 0.5),
  nsims = 1000
) |&gt;
  power(alpha = 0.05) |&gt;
  plot(hline = 0.8, caption_width = 60) +
  ggplot2::theme_bw() +
  ggplot2::theme(plot.caption = ggplot2::element_text(hjust = 0)) +
  ggplot2::labs(title = "Power for the ratio of geometric means")
</pre></div>
<div style="display: flex; justify-content: center; padding-top: 10px; padding-bottom: 10px;">
  <img style="max-width: 100%; height: auto;" src="figures/plot_details-1.svg" alt="Example of extending plot() with ggplot2 functions" />
</div>



<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+power">power()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# plot() examples
#----------------------------------------------------------------------------
library(depower)

# Power for independent two-sample t-test
set.seed(1234)
sim_log_lognormal(
  n1 = c(10, 15),
  n2 = c(10, 15),
  ratio = c(1.3, 1.5),
  cv1 = c(0.3),
  cv2 = c(0.3, 0.5),
  nsims = 500
) |&gt;
  power(alpha = 0.05) |&gt;
  plot()

# Power for dependent two-sample t-test
set.seed(1234)
sim_log_lognormal(
  n1 = c(10, 15),
  n2 = c(10, 15),
  ratio = c(1.3, 1.5),
  cv1 = c(0.3, 0.5),
  cv2 = c(0.3, 0.5),
  cor = c(0.3),
  nsims = 500
) |&gt;
  power(alpha = 0.01) |&gt;
  plot()

# Power for two-sample independent AND two-sample dependent t-test
set.seed(1234)
sim_log_lognormal(
  n1 = c(10, 15),
  n2 = c(10, 15),
  ratio = c(1.3, 1.5),
  cv1 = c(0.3),
  cv2 = c(0.3),
  cor = c(0, 0.3, 0.6),
  nsims = 500
) |&gt;
  power(alpha = c(0.05, 0.01)) |&gt;
  plot(facet_row = "cor", color = "test")

# Power for one-sample t-test
set.seed(1234)
sim_log_lognormal(
  n1 = c(10, 15),
  ratio = c(1.2, 1.4),
  cv1 = c(0.3, 0.5),
  nsims = 500
) |&gt;
  power(alpha = c(0.05, 0.01)) |&gt;
  plot()


# Power for independent two-sample NB test
set.seed(1234)
sim_nb(
  n1 = c(10, 15),
  mean1 = 10,
  ratio = c(1.8, 2),
  dispersion1 = 10,
  dispersion2 = 3,
  nsims = 100
) |&gt;
  power(alpha = 0.01) |&gt;
  plot()

# Power for BNB test
set.seed(1234)
sim_bnb(
  n = c(10, 12),
  mean1 = 10,
  ratio = c(1.3, 1.5),
  dispersion = 5,
  nsims = 100
) |&gt;
  power(alpha = 0.01) |&gt;
  plot()


</code></pre>

<hr>
<h2 id='power'>Simulated power</h2><span id='topic+power'></span>

<h3>Description</h3>

<p>A method to calculate power for objects returned by <code><a href="#topic+sim_log_lognormal">sim_log_lognormal()</a></code>,
<code><a href="#topic+sim_nb">sim_nb()</a></code>, and <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power(data, ..., alpha = 0.05, list_column = FALSE, ncores = 1L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_+3A_data">data</code></td>
<td>
<p>(depower)<br />
The simulated data returned by <code><a href="#topic+sim_log_lognormal">sim_log_lognormal()</a></code>,
<code><a href="#topic+sim_nb">sim_nb()</a></code>, or <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="power_+3A_...">...</code></td>
<td>
<p>(functions)<br />
The function(s) used to perform the test. If empty, a default test
function will be selected based on <code>class(data)</code>. Names are used for
labeling and should be unique. If names are empty, the call coerced to
character will be used for name-value pairs. See 'Details'.</p>
</td></tr>
<tr><td><code id="power_+3A_alpha">alpha</code></td>
<td>
<p>(numeric: <code>0.05</code>; <code style="white-space: pre;">&#8288;(0,1)&#8288;</code>)<br />
The expected probability of rejecting a single null hypothesis when it
is actually true. See 'Details'.</p>
</td></tr>
<tr><td><code id="power_+3A_list_column">list_column</code></td>
<td>
<p>(Scalar logical: <code>FALSE</code>)<br />
If <code>TRUE</code>, the <code>data</code> and <code>result</code> list-columns are included in the
returned data frame. If <code>FALSE</code> (default), the <code>data</code> and <code>result</code>
list-columns are not included in the returned data frame.</p>
</td></tr>
<tr><td><code id="power_+3A_ncores">ncores</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The number of cores (number of worker processes) to use. Do not set
greater than the value returned by <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>. May be
helpful when the number of parameter combinations is large and <code>nsims</code>
is large.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Power is calculated as the proportion of hypothesis tests which result in a
p-value less than or equal to <code>alpha</code>. e.g.
</p>
<div class="sourceCode r"><pre>sum(p &lt;= alpha, na.rm = TRUE) / nsims
</pre></div>
<p>Power is defined as the expected probability of rejecting the null hypothesis
for a chosen value of the unknown effect. In a multiple comparisons scenario,
power is defined as the marginal power, which is the expected power of the
test for each individual null hypothesis assumed to be false.
</p>
<p>Other forms of power under the multiple comparisons scenario include
disjunctive or conjunctive power. Disjunctive power is defined as the
expected probability of correctly rejecting one or more null hypotheses.
Conjunctive power is defined as the expected probability of correctly
rejecting all null hypotheses. In the simplest case, and where all hypotheses
are independent, if the marginal power is defined as <code class="reqn">\pi</code> and <code class="reqn">m</code> is
the number of null hypotheses assumed to be false, then disjunctive power may
be calculated as <code class="reqn">1 - (1 - \pi)^m</code> and conjunctive power may be calculated
as <code class="reqn">\pi^m</code>. Disjunctive power tends to decrease with increasingly
correlated hypotheses and conjunctive power tends to increase with
increasingly correlated hypotheses.
</p>


<h4>Argument <code>...</code></h4>

<p><code>...</code> are the name-value pairs for the functions used to perform the tests.
If not named, the functions coerced to character will be used for the
name-value pairs. Typical in non-standard evaluation, <code>...</code> accepts bare
functions and converts them to a list of expressions. Each element in this
list will be validated as a <code>call</code> and then evaluated on the simulated data.
A <code><a href="base.html#topic+call">base::call()</a></code> is simply an unevaluated function. Below are some examples
of specifying <code>...</code> in <code>power()</code>.
</p>
<div class="sourceCode r"><pre># Examples of specifying ... in power()
data &lt;- sim_nb(
  n1 = 10,
  mean1 = 10,
  ratio = c(1.6, 2),
  dispersion1 = 2,
  dispersion2 = 2,
  nsims = 200
)

# ... is empty, so a an appropriate default function will be provided
power(data)

# This is equivalent to leaving ... empty
power(data, "NB Wald test" = wald_test_nb())

# If not named, "wald_test_nb()" will be used to label the function
power(data, wald_test_nb())

# You can specify any parameters in the call. The data argument
# will automatically be inserted or overwritten.
data |&gt;
  power("NB Wald test" = wald_test_nb(equal_dispersion=TRUE, link="log"))

# Multiple functions may be used.
data |&gt;
  power(
    wald_test_nb(link='log'),
    wald_test_nb(link='sqrt'),
    wald_test_nb(link='squared'),
    wald_test_nb(link='identity')
  )

# Just like functions in a pipe, the parentheses are required.
# This will error because wald_test_nb is missing parentheses.
try(power(data, wald_test_nb))
</pre></div>
<p>In most cases*, any user created test function may be utilized in <code>...</code> if the
following conditions are satisfied:
</p>

<ol>
<li><p> The function contains argument <code>data</code> which is defined as a list with the
first and second elements for simulated data.
</p>
</li>
<li><p> The return object is a list with element <code>p</code> for the p-value of the
hypothesis test.
</p>
</li></ol>

<p>Validate with test cases beforehand.
</p>
<p>*Simulated data of class <code>log_lognormal_mixed_two_sample</code> has both independent
and dependent data. To ensure the appropriate test function is used,
<code>power.log_lognormal_mixed_two_sample()</code> allows only
<code><a href="#topic+t_test_welch">t_test_welch()</a></code> and <code><a href="#topic+t_test_paired">t_test_paired()</a></code> in <code>...</code>. Each will
be evaluated on the simulated data according to column <code>data$cor</code>. If one or
both of these functions are not included in <code>...</code>, the corresponding default
function will be used automatically. If any other test function is included,
an error will be returned.
</p>



<h4>Argument <code>alpha</code></h4>

<p><code class="reqn">\alpha</code> is known as the type I assertion probability and is defined as
the expected probability of rejecting a null hypothesis when it was actually
true. <code class="reqn">\alpha</code> is compared with the p-value and used as the decision
boundary for rejecting or not rejecting the null hypothesis.
</p>
<p>The family-wise error rate is the expected probability of making one or more
type I assertions among a family of hypotheses. Using Bonferroni's method,
<code class="reqn">\alpha</code> is chosen for the family of hypotheses then divided by the
number of tests performed (<code class="reqn">m</code>). Each individual hypothesis is tested at
<code class="reqn">\frac{\alpha}{m}</code>. For example, if you plan to conduct 30 hypothesis
tests and want to control the family-wise error rate to no greater than
<code class="reqn">\alpha = 0.05</code>, you would set <code>alpha = 0.05/30</code>.
</p>
<p>The per-family error rate is the expected number of type I assertions among a
family of hypotheses. If you calculate power for the scenario where you
perform 1,000 hypotheses and want to control the per-family error rate to no
greater than 10 type I assertions, you would choose <code>alpha = 10/1000</code>. This
implicitly assumes all 1,000 hypotheses are truly null. Alternatively, if you
assume 800 of these hypotheses are truly null and 200 are not,
<code>alpha = 10/1000</code> would control the per-family error rate to no greater than
8 type I assertions. If it is acceptable to keep the per-family error rate as
10, setting <code>alpha = 10/800</code> would provide greater marginal power than the
previous scenario.
</p>
<p>These two methods assume that the distribution of p-values for the truly null
hypotheses are uniform(0,1), but remain valid under various other testing
scenarios (such as dependent tests). Other multiple comparison methods, such
as FDR control, are common in practice but don't directly fit into this
power simulation framework.
</p>



<h4>Column <code>nsims</code></h4>

<p>The final number of valid simulations per unique set of simulation parameters
may be less than the original number requested. This may occur when the test
results in a missing p-value. For <code>wald_test_bnb()</code>, pathological MLE
estimates, generally from small sample sizes and very small dispersions, may
result in a negative estimated standard deviation of the ratio. Thus the test
statistic and p-value would not be calculated. Note that simulated data from
<code>sim_nb()</code> and <code>sim_bnb()</code> may also reduce <code>nsims</code> during the data simulation
phase.
</p>
<p>The <code>nsims</code> column in the return data frame is the effective number of
simulations for power results.
</p>



<h3>Value</h3>

<p>A data frame with the following columns appended to the <code>data</code> object:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>alpha</code>  </td><td style="text-align: left;"> Type I assertion probability. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>test</code>   </td><td style="text-align: left;"> Name-value pair of the function and statistical test: <code style="white-space: pre;">&#8288;c(as.character(...) = names(...)&#8288;</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>data</code>   </td><td style="text-align: left;"> List-column of simulated data. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>result</code> </td><td style="text-align: left;"> List-column of test results. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>power</code>  </td><td style="text-align: left;"> Power of the test for corresponding parameters.
</td>
</tr>

</table>

<p>For <code>power(list_column = FALSE)</code>, columns <code>data</code>, and <code>result</code> are excluded from
the data frame.
</p>


<h3>References</h3>

<p>Yu L, Fernandez S, Brock G (2017).
&ldquo;Power analysis for RNA-Seq differential expression studies.&rdquo;
<em>BMC Bioinformatics</em>, <b>18</b>(1), 234.
ISSN 1471-2105, <a href="https://doi.org/10.1186/s12859-017-1648-2">doi:10.1186/s12859-017-1648-2</a>.
</p>
<p>Yu L, Fernandez S, Brock G (2020).
&ldquo;Power analysis for RNA-Seq differential expression studies using generalized linear mixed effects models.&rdquo;
<em>BMC Bioinformatics</em>, <b>21</b>(1), 198.
ISSN 1471-2105, <a href="https://doi.org/10.1186/s12859-020-3541-7">doi:10.1186/s12859-020-3541-7</a>.
</p>
<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>
<p>Julious SA (2004).
&ldquo;Sample sizes for clinical trials with Normal data.&rdquo;
<em>Statistics in Medicine</em>, <b>23</b>(12), 1921&ndash;1986.
<a href="https://doi.org/10.1002/sim.1783">doi:10.1002/sim.1783</a>.
</p>
<p>Vickerstaff V, Omar RZ, Ambler G (2019).
&ldquo;Methods to adjust for multiple comparisons in the analysis and sample size calculation of randomised controlled trials with multiple primary outcomes.&rdquo;
<em>BMC Medical Research Methodology</em>, <b>19</b>(1), 129.
ISSN 1471-2288, <a href="https://doi.org/10.1186/s12874-019-0754-4">doi:10.1186/s12874-019-0754-4</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.depower">plot.depower()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# power() examples
#----------------------------------------------------------------------------
library(depower)

# Power for independent two-sample t-Test
set.seed(1234)
data &lt;- sim_log_lognormal(
  n1 = 20,
  n2 = 20,
  ratio = c(1.2, 1.4),
  cv1 = 0.4,
  cv2 = 0.4,
  cor = 0,
  nsims = 1000
)

# Welch's t-test is used by default
power(data)

# But you can specify anything else that is needed
power(
  data = data,
  "Welch's t-Test" = t_test_welch(alternative = "greater"),
  alpha = 0.01
)

# Power for dependent two-sample t-Test
set.seed(1234)
sim_log_lognormal(
  n1 = 20,
  n2 = 20,
  ratio = c(1.2, 1.4),
  cv1 = 0.4,
  cv2 = 0.4,
  cor = 0.5,
  nsims = 1000
) |&gt;
  power()

# Power for mixed-type two-sample t-Test
set.seed(1234)
sim_log_lognormal(
  n1 = 20,
  n2 = 20,
  ratio = c(1.2, 1.4),
  cv1 = 0.4,
  cv2 = 0.4,
  cor = c(0, 0.5),
  nsims = 1000
) |&gt;
  power()

# Power for one-sample t-Test
set.seed(1234)
sim_log_lognormal(
  n1 = 20,
  ratio = c(1.2, 1.4),
  cv1 = 0.4,
  nsims = 1000
) |&gt;
  power()

# Power for independent two-sample NB test
set.seed(1234)
sim_nb(
  n1 = 10,
  mean1 = 10,
  ratio = c(1.6, 2),
  dispersion1 = 2,
  dispersion2 = 2,
  nsims = 200
) |&gt;
  power()

# Power for BNB test
set.seed(1234)
sim_bnb(
  n = 10,
  mean1 = 10,
  ratio = c(1.4, 1.6),
  dispersion = 10,
  nsims = 200
) |&gt;
  power()

</code></pre>

<hr>
<h2 id='sim_bnb'>Simulate data from a BNB distribution</h2><span id='topic+sim_bnb'></span>

<h3>Description</h3>

<p>Simulate data from the bivariate negative binomial (BNB) distribution. The
BNB distribution is used to simulate count data where the event counts are
jointly dependent (correlated). For independent data, see
<code><a href="#topic+sim_nb">sim_nb()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_bnb(
  n,
  mean1,
  mean2,
  ratio,
  dispersion,
  nsims = 1L,
  return_type = "list",
  max_zeros = 0.99,
  ncores = 1L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_bnb_+3A_n">n</code></td>
<td>
<p>(integer: <code style="white-space: pre;">&#8288;[2, Inf)&#8288;</code>)<br />
The number(s) of paired observations.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_mean1">mean1</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The mean(s) of sample 1 <code class="reqn">(\mu_1)</code>.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_mean2">mean2</code>, <code id="sim_bnb_+3A_ratio">ratio</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
Only specify one of these arguments.
</p>

<ul>
<li> <p><code>mean2</code>: The mean(s) of sample 2 <code class="reqn">(\mu_2)</code>.
</p>
</li>
<li> <p><code>ratio</code>: The ratio(s) of means for sample 2 with respect to sample 1
<code class="reqn">\left( r = \frac{\mu_2}{\mu_1} \right)</code>.
</p>
</li></ul>

<p><code>mean2 = ratio * mean1</code></p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_dispersion">dispersion</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The gamma distribution shape parameter(s) <code class="reqn">(\theta)</code> which control
the dispersion and the correlation between sample 1 and 2.
See 'Details' and 'Examples'.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_nsims">nsims</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1, Inf)&#8288;</code>)<br />
The expected number of simulated data sets. If <code>nsims &gt; 1</code>, the data
is returned in a list-column of a depower simulation data frame.
<code>nsims</code> may be reduced depending on <code>max_zeros</code>.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_return_type">return_type</code></td>
<td>
<p>(string: <code>"list"</code>; <code>c("list", "data.frame")</code>)<br />
The data structure of the simulated data. If <code>"list"</code> (default), a
list object is returned. If <code>"data.frame"</code> a data frame in tall format
is returned. The list object provides computational efficiency and the
data frame object is convenient for formulas. See 'Value'.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_max_zeros">max_zeros</code></td>
<td>
<p>(Scalar numeric: <code>0.99</code>; <code style="white-space: pre;">&#8288;[0, 1]&#8288;</code>)<br />
The maximum proportion of zeros each group in a simulated dataset is
allowed to have. If the proportion of zeros is greater than this
value, the corresponding data is excluded from the set of simulations.
This is most likely to occur when the sample size is small and the
dispersion parameter is small.</p>
</td></tr>
<tr><td><code id="sim_bnb_+3A_ncores">ncores</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The number of cores (number of worker processes) to use. Do not set
greater than the value returned by <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>. May be
helpful when the number of parameter combinations is large and <code>nsims</code>
is large.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The negative binomial distribution may be defined using a gamma-Poisson
mixture distribution. In this case, the Poisson parameter <code class="reqn">\lambda</code>
is a random variable with gamma distribution. Equivalence between different
parameterizations are demonstrated below:
</p>
<div class="sourceCode r"><pre># Define constants and their relationships
n &lt;- 10000
dispersion &lt;- 8
mu &lt;- 4
p &lt;- dispersion / (dispersion + mu)
q &lt;- mu / (mu + dispersion)
variance &lt;- mu + (mu^2 / dispersion)
rate &lt;- p / (1 - p)
scale &lt;- (1 - p) / p

# alternative formula for mu
mu_alt &lt;- (dispersion * (1 - p)) / p
stopifnot(isTRUE(all.equal(mu, mu_alt)))

set.seed(20240321)

# Using built-in rnbinom with dispersion and mean
w &lt;- rnbinom(n = n, size = dispersion, mu = mu)

# Using gamma-Poisson mixture with gamma rate parameter
x &lt;- rpois(
  n = n,
  lambda = rgamma(n = n, shape = dispersion, rate = rate)
)

# Using gamma-Poisson mixture with gamma scale parameter
y &lt;- rpois(
  n = n,
  lambda = rgamma(n = n, shape = dispersion, scale = scale)
)

# Using gamma-Poisson mixture with multiplicative mean and
# gamma scale parameter
z &lt;- rpois(
  n = n,
  lambda = mu * rgamma(n = n, shape = dispersion, scale = 1/dispersion)
)

# Compare CDFs
par(mar=c(4,4,1,1))
plot(
  x = sort(w),
  y = (1:n)/n,
  xlim = range(c(w,x,y,z)),
  ylim = c(0,1),
  col = 'green',
  lwd = 4,
  type = 'l',
  main = 'CDF'
)
lines(x = sort(x), y = (1:n)/n, col = 'red', lwd = 2)
lines(x = sort(y), y = (1:n)/n, col = 'yellow', lwd = 1.5)
lines(x = sort(z), y = (1:n)/n, col = 'black')
</pre></div>
<div style="display: flex; justify-content: center; padding-top: 10px; padding-bottom: 10px;">
  <img style="max-width: 100%; height: auto;" src="figures/gamma_poisson_mixture-1.svg" alt="Gamma-Poisson Mixture CDF" />
</div>
<br>

<p>The BNB distribution is implemented by compounding two conditionally
independent Poisson random variables <code class="reqn">X_1 \mid G = g \sim \text{Poisson}(\mu g)</code>
and <code class="reqn">X_2 \mid G = g \sim \text{Poisson}(r \mu g)</code> with a gamma random
variable <code class="reqn">G \sim \text{Gamma}(\theta, \theta^{-1})</code>. The probability mass
function for the joint distribution of <code class="reqn">X_1, X_2</code> is
</p>
<p style="text-align: center;"><code class="reqn">
P(X_1 = x_1, X_2 = x_2) = \frac{\Gamma(x_1 + x_2 + \theta)}{(\mu + r \mu + \theta)^{x_1 + x_2 + \theta}}
  \frac{\mu^{x_1}}{x_1!} \frac{(r \mu)^{x_2}}{x_2!}
  \frac{\theta^{\theta}}{\Gamma(\theta)}
</code>
</p>

<p>where <code class="reqn">x_1,x_2 \in \mathbb{N}^{\geq 0}</code> are specific values of the count
outcomes, <code class="reqn">\theta \in \mathbb{R}^{&gt; 0}</code> is the <code>dispersion</code> parameter
which controls the dispersion and level of correlation between the two
samples (otherwise known as the shape parameter of the gamma distribution),
<code class="reqn">\mu \in \mathbb{R}^{&gt; 0}</code> is the mean parameter, and
<code class="reqn">r = \frac{\mu_2}{\mu_1} \in \mathbb{R}^{&gt; 0}</code> is the <code>ratio</code> parameter
representing the multiplicative change in the mean of the second sample
relative to the first sample. <code class="reqn">G</code> denotes the random subject effect and
the gamma distribution scale parameter is assumed to be the inverse of the
dispersion parameter (<code class="reqn">\theta^{-1}</code>) for identifiability.
</p>
<p>Correlation decreases from 1 to 0 as the dispersion parameter increases from
0 to infinity. For a given dispersion, increasing means also increases the
correlation. See 'Examples' for a demonstration.
</p>
<p>See 'Details' in <code><a href="#topic+sim_nb">sim_nb()</a></code> for additional information on the
negative binomial distribution.
</p>


<h3>Value</h3>

<p>If <code>nsims = 1</code> and the number of unique parameter combinations is
one, the following objects are returned:
</p>

<ul>
<li><p> If <code>return_type = "list"</code>, a list:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated counts from sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated counts from sample 2.
</td>
</tr>

</table>

</li>
<li><p> If <code>return_type = "data.frame"</code>, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>item</code> </td><td style="text-align: left;"> Subject/item indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>condition</code> </td><td style="text-align: left;"> Sample/condition indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>value</code> </td><td style="text-align: left;"> Simulated counts.
</td>
</tr>

</table>

</li></ul>

<p>If <code>nsims &gt; 1</code> or the number of unique parameter combinations is greater than
one, each object described above is returned in a list-column named <code>data</code> in
a depower simulation data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Mean for sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Mean for sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Ratio of means (sample 2 / sample 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> Gamma distribution shape parameter (dispersion). </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>nsims</code> </td><td style="text-align: left;"> Number of valid simulation iterations. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>distribution</code> </td><td style="text-align: left;"> Distribution sampled from. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>data</code> </td><td style="text-align: left;"> List-column of simulated data.
</td>
</tr>

</table>



<h3>References</h3>

<p>Yu L, Fernandez S, Brock G (2020).
&ldquo;Power analysis for RNA-Seq differential expression studies using generalized linear mixed effects models.&rdquo;
<em>BMC Bioinformatics</em>, <b>21</b>(1), 198.
ISSN 1471-2105, <a href="https://doi.org/10.1186/s12859-020-3541-7">doi:10.1186/s12859-020-3541-7</a>.
</p>
<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_nb">sim_nb()</a></code>, <code><a href="stats.html#topic+Poisson">stats::rpois()</a></code>, <code><a href="stats.html#topic+GammaDist">stats::rgamma()</a></code>,
<code><a href="stats.html#topic+NegBinomial">stats::rnbinom()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# sim_bnb() examples
#----------------------------------------------------------------------------
library(depower)

# Paired two-sample data returned in a data frame
sim_bnb(
  n = 10,
  mean1 = 10,
  ratio = 1.6,
  dispersion = 3,
  nsims = 1,
  return_type = "data.frame"
)

# Paired two-sample data returned in a list
sim_bnb(
  n = 10,
  mean1 = 10,
  ratio = 1.6,
  dispersion = 3,
  nsims = 1,
  return_type = "list"
)

# Two simulations of paired two-sample data
# returned as a list of data frames
sim_bnb(
  n = 10,
  mean1 = 10,
  ratio = 1.6,
  dispersion = 3,
  nsims = 2,
  return_type = "data.frame"
)

# Two simulations of Paired two-sample data
# returned as a list of lists
sim_bnb(
  n = 10,
  mean1 = 10,
  ratio = 1.6,
  dispersion = 3,
  nsims = 2,
  return_type = "list"
)

#----------------------------------------------------------------------------
# Visualization of the BNB distribution as dispersion varies.
#----------------------------------------------------------------------------
set.seed(1234)
data &lt;- lapply(
  X = c(1, 10, 100, 1000),
  FUN = function(x) {
    d &lt;- sim_bnb(
      n = 1000,
      mean1 = 10,
      ratio = 1.5,
      dispersion = x,
      nsims = 1,
      return_type = "data.frame"
    )
    cor &lt;- cor(
      x = d[d$condition == "1", ]$value,
      y = d[d$condition == "2", ]$value
    )
    cbind(dispersion = x, correlation = cor, d)
  }
)

data &lt;- do.call(what = "rbind", args = data)

ggplot2::ggplot(
  data = data,
  mapping = ggplot2::aes(x = value, fill = condition)
) +
  ggplot2::facet_wrap(
    facets = ggplot2::vars(.data$dispersion),
    ncol = 2,
    labeller = ggplot2::labeller(.rows = ggplot2::label_both)
  ) +
  ggplot2::geom_density(alpha = 0.3) +
  ggplot2::coord_cartesian(xlim = c(0, 60)) +
  ggplot2::geom_text(
    mapping = ggplot2::aes(
      x = 30,
      y = 0.12,
      label = paste0("Correlation: ", round(correlation, 2))
    ),
    check_overlap = TRUE
  ) +
  ggplot2::labs(
    x = "Value",
    y = "Density",
    fill = "Condition",
    caption = "Mean1=10, Mean2=15, ratio=1.5"
  )

</code></pre>

<hr>
<h2 id='sim_log_lognormal'>Simulate data from a normal distribution</h2><span id='topic+sim_log_lognormal'></span>

<h3>Description</h3>

<p>Simulate data from the log transformed lognormal distribution (i.e. a normal
distribution). This function handles all three cases:
</p>

<ol>
<li><p> One-sample data
</p>
</li>
<li><p> Dependent two-sample data
</p>
</li>
<li><p> Independent two-sample data
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>sim_log_lognormal(
  n1,
  n2 = NULL,
  ratio,
  cv1,
  cv2 = NULL,
  cor = 0,
  nsims = 1L,
  return_type = "list",
  ncores = 1L,
  messages = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_log_lognormal_+3A_n1">n1</code></td>
<td>
<p>(integer: <code style="white-space: pre;">&#8288;[2, Inf)&#8288;</code>)<br />
The sample size(s) of sample 1.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_n2">n2</code></td>
<td>
<p>(integer: <code>NULL</code>; <code style="white-space: pre;">&#8288;[2, Inf)&#8288;</code>)<br />
The sample size(s) of sample 2. Set as <code>NULL</code> if you want to simulate
for the one-sample case.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_ratio">ratio</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The assumed population fold change(s) of sample 2 with respect to
sample 1.
</p>

<ul>
<li><p> For one-sample data, <code>ratio</code> is defined as the geometric mean (GM) of the
original lognormal population distribution.
</p>
</li>
<li><p> For dependent two-sample data, <code>ratio</code> is defined by
GM(sample 2 / sample 1) of the original lognormal population distributions.
</p>

<ul>
<li><p> e.g. <code>ratio = 2</code> assumes that the geometric mean of all paired
ratios (sample 2 / sample 1) is 2.
</p>
</li></ul>

</li>
<li><p> For independent two-sample data, the <code>ratio</code> is defined by
GM(group 2) / GM(group 1) of the original lognormal population distributions.
</p>

<ul>
<li><p> e.g. <code>ratio = 2</code> assumes that the geometric mean of sample 2 is 2
times larger than the geometric mean of sample 1.
</p>
</li></ul>

</li></ul>

<p>See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_cv1">cv1</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The coefficient of variation(s) of sample 1 in the original lognormal
data.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_cv2">cv2</code></td>
<td>
<p>(numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The coefficient of variation(s) of sample 2 in the original lognormal
data. Set as <code>NULL</code> if you want to simulate for the one-sample case.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_cor">cor</code></td>
<td>
<p>(numeric: <code>0</code>; <code style="white-space: pre;">&#8288;[-1, 1]&#8288;</code>)<br />
The correlation(s) between sample 1 and sample 2 in the original
lognormal data. Not used for the one-sample case.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_nsims">nsims</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The number of simulated data sets. If <code>nsims &gt; 1</code>, the data is
returned in a list-column of a depower simulation data frame.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_return_type">return_type</code></td>
<td>
<p>(string: <code>"list"</code>; <code>c("list", "data.frame")</code>)<br />
The data structure of the simulated data. If <code>"list"</code> (default), a
list object is returned. If <code>"data.frame"</code> a data frame in tall format
is returned. The list object provides computational efficiency and the
data frame object is convenient for formulas. See 'Value'.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_ncores">ncores</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The number of cores (number of worker processes) to use. Do not set
greater than the value returned by <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>. May be
helpful when the number of parameter combinations is large and <code>nsims</code>
is large.</p>
</td></tr>
<tr><td><code id="sim_log_lognormal_+3A_messages">messages</code></td>
<td>
<p>(Scalar logical: <code>TRUE</code>)<br />
Whether or not to display messages for pathological simulation cases.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on assumed characteristics of the original lognormal distribution, data
is simulated from the corresponding log-transformed (normal) distribution.
This simulated data is suitable for assessing power of a hypothesis for the
geometric mean or ratio of geometric means from the original lognormal data.
</p>
<p>This method can also be useful for other population distributions which are
positive and where it makes sense to describe the ratio of geometric means.
However, the lognormal distribution is theoretically correct in the sense
that you can log transform to a normal distribution, compute the summary
statistic, then apply the inverse transformation to summarize on the original
lognormal scale.
</p>
<p>Let <code class="reqn">GM(\cdot)</code> be the geometric mean and <code class="reqn">AM(\cdot)</code> be the
arithmetic mean. For independent lognormal samples <code class="reqn">X_1</code> and <code class="reqn">X_2</code>
</p>
<p style="text-align: center;"><code class="reqn">\text{Fold Change} = \frac{GM(X_2)}{GM(X_1)}</code>
</p>

<p>For dependent lognormal samples <code class="reqn">X_1</code> and <code class="reqn">X_2</code>
</p>
<p style="text-align: center;"><code class="reqn">\text{Fold Change} = GM\left( \frac{X_2}{X_1} \right)</code>
</p>

<p>Unlike ratios and the arithmetic mean, for equal sample sizes of
<code class="reqn">X_1</code> and <code class="reqn">X_2</code> it follows that
<code class="reqn">\frac{GM(X_2)}{GM(X_1)} = GM \left( \frac{X_2}{X_1} \right) =
e^{AM(\ln X_2) - AM(\ln X_1)} = e^{AM(\ln X_2 - \ln X_1)}</code>.
</p>
<p>The coefficient of variation (CV) for <code class="reqn">X</code> is defined as
</p>
<p style="text-align: center;"><code class="reqn">CV = \frac{SD(X)}{AM(X)}</code>
</p>

<p>The relationship between sample statistics for the original lognormal data
(<code class="reqn">X</code>) and the natural logged data (<code class="reqn">\ln{X}</code>) are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
AM(X)  &amp;= e^{AM(\ln{X}) + \frac{Var(\ln{X})}{2}} \\
GM(X)  &amp;= e^{AM(\ln{X})} \\
Var(X) &amp;= AM(X)^2 \left( e^{Var(\ln{X})} - 1 \right) \\
CV(X)  &amp;= \frac{\sqrt{AM(X)^2 \left( e^{Var(\ln{X})} - 1 \right)}}{AM(X)} \\
       &amp;= \sqrt{e^{Var(\ln{X})} - 1}
\end{aligned}
</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
AM(\ln{X})  &amp;= \ln \left( \frac{AM(X)}{\sqrt{CV(X)^2 + 1}} \right) \\
Var(\ln{X}) &amp;= \ln(CV(X)^2 + 1) \\
Cor(\ln{X_1}, \ln{X_2}) &amp;= \frac{\ln \left( Cor(X_1, X_2)CV(X_1)CV(X_2) + 1 \right)}{SD(\ln{X_1})SD(\ln{X_2})}
\end{aligned}
</code>
</p>

<p>Based on the properties of correlation and variance,
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
Var(X_2 - X_1) &amp;= Var(X_1) + Var(X_2) - 2Cov(X_1, X_2) \\
           &amp;= Var(X_1) + Var(X_2) - 2Cor(X_1, X_2)SD(X_1)SD(X_2) \\
SD(X_2 - X_1)  &amp;= \sqrt{Var(X_2 - X_1)}
\end{aligned}
</code>
</p>

<p>The standard deviation of the differences gets smaller the more positive
the correlation and conversely gets larger the more negative the correlation.
For the special case where the two samples are uncorrelated and each has the
same variance, it follows that
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
Var(X_2 - X_1) &amp;= \sigma^2 + \sigma^2 \\
SD(X_2 - X_1) &amp;= \sqrt{2}\sigma
\end{aligned}
</code>
</p>



<h3>Value</h3>

<p>If <code>nsims = 1</code> and the number of unique parameter combinations is
one, the following objects are returned:
</p>

<ul>
<li><p> If one-sample data with <code>return_type = "list"</code>, a list:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> One sample of simulated normal values.
</td>
</tr>

</table>

</li>
<li><p> If one-sample data with <code>return_type = "data.frame"</code>, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>item</code> </td><td style="text-align: left;"> Pair/subject/item indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>value</code> </td><td style="text-align: left;"> Simulated normal values.
</td>
</tr>

</table>

</li>
<li><p> If two-sample data with <code>return_type = "list"</code>, a list:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated normal values from sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated normal values from sample 2.
</td>
</tr>

</table>

</li>
<li><p> If two-sample data with <code>return_type = "data.frame"</code>, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>item</code> </td><td style="text-align: left;"> Pair/subject/item indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>condition</code> </td><td style="text-align: left;"> Time/group/condition indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>value</code> </td><td style="text-align: left;"> Simulated normal values.
</td>
</tr>

</table>

</li></ul>

<p>If <code>nsims &gt; 1</code> or the number of unique parameter combinations is greater than
one, each object described above is returned in data frame, located in a
list-column named <code>data</code>.
</p>

<ul>
<li><p> If one-sample data, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> The sample size. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Geometric mean [GM(sample 1)]. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>cv1</code> </td><td style="text-align: left;"> Coefficient of variation for sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>nsims</code> </td><td style="text-align: left;"> Number of data simulations. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>distribution</code> </td><td style="text-align: left;"> Distribution sampled from. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>data</code> </td><td style="text-align: left;"> List-column of simulated data.
</td>
</tr>

</table>

</li>
<li><p> If two-sample data, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Ratio of geometric means [GM(sample 2) / GM(sample 1)]
or geometric mean ratio [GM(sample 2 / sample 1)]. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>cv1</code> </td><td style="text-align: left;"> Coefficient of variation for sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>cv2</code> </td><td style="text-align: left;"> Coefficient of variation for sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>cor</code> </td><td style="text-align: left;"> Correlation between samples. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>nsims</code> </td><td style="text-align: left;"> Number of data simulations. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>distribution</code> </td><td style="text-align: left;"> Distribution sampled from. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>data</code> </td><td style="text-align: left;"> List-column of simulated data.
</td>
</tr>

</table>

</li></ul>



<h3>References</h3>

<p>Julious SA (2004).
&ldquo;Sample sizes for clinical trials with Normal data.&rdquo;
<em>Statistics in Medicine</em>, <b>23</b>(12), 1921&ndash;1986.
<a href="https://doi.org/10.1002/sim.1783">doi:10.1002/sim.1783</a>.
</p>
<p>Hauschke D, Steinijans VW, Diletti E, Burke M (1992).
&ldquo;Sample size determination for bioequivalence assessment using a multiplicative model.&rdquo;
<em>Journal of Pharmacokinetics and Biopharmaceutics</em>, <b>20</b>(5), 557&ndash;561.
ISSN 0090-466X, <a href="https://doi.org/10.1007/BF01061471">doi:10.1007/BF01061471</a>.
</p>
<p>Johnson NL, Kotz S, Balakrishnan N (1994).
<em>Continuous univariate distributions</em>,  Wiley series in probability and mathematical statistics, 2nd ed edition.
Wiley, New York.
ISBN 9780471584957 9780471584940.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+Normal">stats::rnorm()</a></code>, <code><a href="mvnfast.html#topic+rmvn">mvnfast::rmvn()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# sim_log_lognormal() examples
#----------------------------------------------------------------------------
library(depower)

# Independent two-sample data returned in a data frame
sim_log_lognormal(
  n1 = 10,
  n2 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0,
  nsims = 1,
  return_type = "data.frame"
)

# Independent two-sample data returned in a list
sim_log_lognormal(
  n1 = 10,
  n2 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0,
  nsims = 1,
  return_type = "list"
)

# Dependent two-sample data returned in a data frame
sim_log_lognormal(
  n1 = 10,
  n2 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0.4,
  nsims = 1,
  return_type = "data.frame"
)

# Dependent two-sample data returned in a list
sim_log_lognormal(
  n1 = 10,
  n2 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0.4,
  nsims = 1,
  return_type = "list"
)

# One-sample data returned in a data frame
sim_log_lognormal(
  n1 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  nsims = 1,
  return_type = "data.frame"
)

# One-sample data returned in a list
sim_log_lognormal(
  n1 = 10,
  ratio = 1.3,
  cv1 = 0.35,
  nsims = 1,
  return_type = "list"
)

# Independent two-sample data: two simulations for four parameter combinations.
# Returned as a list-column of lists within a data frame
sim_log_lognormal(
  n1 = c(10, 20),
  n2 = c(10, 20),
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0,
  nsims = 2,
  return_type = "list"
)

# Dependent two-sample data: two simulations for two parameter combinations.
# Returned as a list-column of lists within a data frame
sim_log_lognormal(
  n1 = c(10, 20),
  n2 = c(10, 20),
  ratio = 1.3,
  cv1 = 0.35,
  cv2 = 0.35,
  cor = 0.4,
  nsims = 2,
  return_type = "list"
)

# One-sample data: two simulations for two parameter combinations
# Returned as a list-column of lists within a data frame
sim_log_lognormal(
  n1 = c(10, 20),
  ratio = 1.3,
  cv1 = 0.35,
  nsims = 2,
  return_type = "list"
)

</code></pre>

<hr>
<h2 id='sim_nb'>Simulate data from a NB distribution</h2><span id='topic+sim_nb'></span>

<h3>Description</h3>

<p>Simulate data from two independent negative binomial (NB) distributions. For
paired data, see <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_nb(
  n1,
  n2 = n1,
  mean1,
  mean2,
  ratio,
  dispersion1,
  dispersion2 = dispersion1,
  nsims = 1L,
  return_type = "list",
  max_zeros = 0.99,
  ncores = 1L
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim_nb_+3A_n1">n1</code></td>
<td>
<p>(integer: <code style="white-space: pre;">&#8288;[2, Inf)&#8288;</code>)<br />
The sample size(s) of group 1.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_n2">n2</code></td>
<td>
<p>(integer: <code>n1</code>; <code style="white-space: pre;">&#8288;[2, Inf)&#8288;</code>)<br />
The sample size(s) of group 2.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_mean1">mean1</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The mean(s) of group 1 <code class="reqn">(\mu_1)</code>.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_mean2">mean2</code>, <code id="sim_nb_+3A_ratio">ratio</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
Only specify one of these arguments.
</p>

<ul>
<li> <p><code>mean2</code>: The mean(s) of group 2 <code class="reqn">(\mu_2)</code>.
</p>
</li>
<li> <p><code>ratio</code>: The ratio(s) of means for group 2 with respect to group 1
<code class="reqn">\left( r = \frac{\mu_2}{\mu_1} \right)</code>.
</p>
</li></ul>

<p><code>mean2 = ratio * mean1</code></p>
</td></tr>
<tr><td><code id="sim_nb_+3A_dispersion1">dispersion1</code></td>
<td>
<p>(numeric: <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The dispersion parameter(s) of group 1 <code class="reqn">(\theta_1)</code>. See 'Details'
and 'Examples'.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_dispersion2">dispersion2</code></td>
<td>
<p>(numeric: <code>dispersion1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The dispersion parameter(s) of group 2 <code class="reqn">(\theta_2)</code>. See 'Details'
and 'Examples'.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_nsims">nsims</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The expected number of simulated data sets. If <code>nsims &gt; 1</code>, the data
is returned in a list-column of a depower simulation data frame.
<code>nsims</code> may be reduced depending on <code>max_zeros</code>.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_return_type">return_type</code></td>
<td>
<p>(string: <code>"list"</code>; <code>c("list", "data.frame")</code>)<br />
The data structure of the simulated data. If <code>"list"</code> (default), a
list object is returned. If <code>"data.frame"</code> a data frame in tall format
is returned. The list object provides computational efficiency and the
data frame object is convenient for formulas. See 'Value'.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_max_zeros">max_zeros</code></td>
<td>
<p>(Scalar numeric: <code>0.99</code>; <code style="white-space: pre;">&#8288;[0, 1]&#8288;</code>)<br />
The maximum proportion of zeros each group in a simulated dataset is
allowed to have. If the proportion of zeros is greater than this
value, the corresponding data is excluded from the set of simulations.
This is most likely to occur when the sample size is small and the
dispersion parameter is small.</p>
</td></tr>
<tr><td><code id="sim_nb_+3A_ncores">ncores</code></td>
<td>
<p>(Scalar integer: <code>1L</code>; <code style="white-space: pre;">&#8288;[1,Inf)&#8288;</code>)<br />
The number of cores (number of worker processes) to use. Do not set
greater than the value returned by <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>. May be
helpful when the number of parameter combinations is large and <code>nsims</code>
is large.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The negative binomial distribution has many parameterizations. In the
regression modeling context, it is common to specify the distribution in
terms of its mean and dispersion. We use the following probability
mass function:
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
P(X = x) &amp;= \dbinom{x + \theta - 1}{x}
            \left( \frac{\theta}{\theta + \mu} \right)^{\theta}
            \left( \frac{\mu}{\mu + \theta} \right)^x \\
         &amp;= \frac{\Gamma(x + \theta)}{x! \Gamma(\theta)}
            \left( \frac{\theta}{\theta + \mu} \right)^{\theta}
            \left( \frac{\mu}{\mu + \theta} \right)^{x} \\
         &amp;= \frac{\Gamma(x + \theta)}{(\theta + \mu)^{\theta + x}}
            \frac{\theta^{\theta}}{\Gamma(\theta)} \frac{\mu^{x}}{x!}
\end{aligned}
</code>
</p>

<p>where <code class="reqn">x \in \mathbb{N}^{\geq 0}</code>, <code class="reqn">\theta \in \mathbb{R}^{&gt; 0}</code>
is the dispersion parameter, and <code class="reqn">\mu \in \mathbb{R}^{&gt; 0}</code> is the mean.
This is analogous to the typical formulation where <code class="reqn">X</code> is counting
<code class="reqn">x</code> failures given <code class="reqn">\theta</code> successes and
<code class="reqn">p = \frac{\theta}{\theta + \mu}</code> is the probability of success on each
trial. It follows that <code class="reqn">E(X) = \mu</code> and
<code class="reqn">Var(X) = \mu + \frac{\mu^2}{\theta}</code>. The <code class="reqn">\theta</code> parameter
describes the 'dispersion' among observations. Smaller values of <code class="reqn">\theta</code>
lead to overdispersion and larger values of <code class="reqn">\theta</code> decrease the
overdispersion, eventually converging to the Poisson distribution.
</p>
<p>Described above is the 'indirect quadratic parameterization' of the negative
binomial distribution, which is commonly found in the R ecosystem. However, it
is somewhat counterintuitive because the smaller <code class="reqn">\theta</code> gets, the larger
the overdispersion. The 'direct quadratic parameterization' of the negative
binomial distribution may be found in some R packages and other languages
such as SAS and Stata. The direct parameterization is defined by substituting
<code class="reqn">\alpha = \frac{1}{\theta}</code> (<code class="reqn">\alpha &gt; 0</code>) which results in
<code class="reqn">Var(X) = \mu + \alpha\mu^2</code>. In this case, the larger <code class="reqn">\alpha</code> gets
the larger the overdispersion, and the Poisson distribution is a special case
of the negative binomial distribution where <code class="reqn">\alpha = 0</code>.
</p>
<p>A general class of negative binomial models may be defined with mean
<code class="reqn">\mu</code> and variance <code class="reqn">\mu + \alpha\mu^{p}</code>. The 'linear
parameterization' is then found by setting <code class="reqn">p=1</code>, resulting in
<code class="reqn">Var(X) = \mu + \alpha\mu</code>. It's common to label the linear
parameterization as 'NB1' and the direct quadratic parameterization as 'NB2'.
</p>
<p>See 'Details' in <code><a href="#topic+sim_bnb">sim_bnb()</a></code> for additional information on the
gamma-Poisson mixture formulation of the negative binomial distribution.
</p>


<h3>Value</h3>

<p>If <code>nsims = 1</code> and the number of unique parameter combinations is
one, the following objects are returned:
</p>

<ul>
<li><p> If <code>return_type = "list"</code>, a list:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated counts from group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> Simulated counts from group 2.
</td>
</tr>

</table>

</li>
<li><p> If <code>return_type = "data.frame"</code>, a data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>item</code> </td><td style="text-align: left;"> Subject/item indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>condition</code> </td><td style="text-align: left;"> Group/condition indicator. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>value</code> </td><td style="text-align: left;"> Simulated counts.
</td>
</tr>

</table>

</li></ul>

<p>If <code>nsims &gt; 1</code> or the number of unique parameter combinations is greater than
one, each object described above is returned in a list-column named <code>data</code> in
a depower simulation data frame:
</p>

<table>
<tr>
 <td style="text-align: left;">
Column </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Mean for group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Mean for group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
5 </td><td style="text-align: left;"> <code>ratio</code> </td><td style="text-align: left;"> Ratio of means (group 2 / group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> Dispersion parameter for group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> Dispersion parameter for group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> <code>nsims</code> </td><td style="text-align: left;"> Number of valid simulation iterations. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> <code>distribution</code> </td><td style="text-align: left;"> Distribution sampled from. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> <code>data</code> </td><td style="text-align: left;"> List-column of simulated data.
</td>
</tr>

</table>



<h3>References</h3>

<p>Yu L, Fernandez S, Brock G (2017).
&ldquo;Power analysis for RNA-Seq differential expression studies.&rdquo;
<em>BMC Bioinformatics</em>, <b>18</b>(1), 234.
ISSN 1471-2105, <a href="https://doi.org/10.1186/s12859-017-1648-2">doi:10.1186/s12859-017-1648-2</a>.
</p>
<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>
<p>Hilbe JM (2011).
<em>Negative Binomial Regression</em>, 2 edition.
Cambridge University Press.
ISBN 9780521198158 9780511973420, <a href="https://doi.org/10.1017/CBO9780511973420">doi:10.1017/CBO9780511973420</a>.
</p>
<p>Hilbe JM (2014).
<em>Modeling count data</em>.
Cambridge University Press, New York, NY.
ISBN 9781107028333 9781107611252, <a href="https://doi.org/10.1017/CBO9781139236065">doi:10.1017/CBO9781139236065</a>.
</p>
<p>Cameron AC, Trivedi PK (2013).
<em>Regression Analysis of Count Data</em>,  Econometric Society Monographs, 2 edition.
Cambridge University Press.
<a href="https://doi.org/10.1017/CBO9781139013567">doi:10.1017/CBO9781139013567</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim_bnb">sim_bnb()</a></code>, <code><a href="stats.html#topic+NegBinomial">stats::rnbinom()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# sim_nb() examples
#----------------------------------------------------------------------------
library(depower)

# Independent two-sample NB data returned in a data frame
sim_nb(
  n1 = 10,
  mean1 = 5,
  ratio = 1.6,
  dispersion1 = 0.5,
  dispersion2 = 0.5,
  nsims = 1,
  return_type = "data.frame"
)

# Independent two-sample NB data returned in a list
sim_nb(
  n1 = 10,
  mean1 = 5,
  ratio = 1.6,
  dispersion1 = 0.5,
  dispersion2 = 0.5,
  nsims = 1,
  return_type = "list"
)

# Two simulations of independent two-sample data
# returned as a list of data frames
sim_nb(
  n1 = 10,
  mean1 = 5,
  ratio = 1.6,
  dispersion1 = 0.5,
  dispersion2 = 0.5,
  nsims = 2,
  return_type = "data.frame"
)

# Two simulations of independent two-sample data
# returned as a list of lists
sim_nb(
  n1 = 10,
  mean1 = 5,
  ratio = 1.6,
  dispersion1 = 0.5,
  dispersion2 = 0.5,
  nsims = 2,
  return_type = "list"
)

#----------------------------------------------------------------------------
# Visualization of the NB distribution as dispersion varies between groups.
#----------------------------------------------------------------------------
disp &lt;- expand.grid(c(1, 10, 100), c(1, 10, 100))
set.seed(1234)
data &lt;- mapply(
  FUN = function(disp1, disp2) {
    d &lt;- sim_nb(
      n1 = 1000,
      mean1 = 10,
      ratio = 1.5,
      dispersion1 = disp1,
      dispersion2 = disp2,
      nsims = 1,
      return_type = "data.frame"
    )
    cbind(dispersion1 = disp1, dispersion2 = disp2, d)
  },
  disp1 = disp[[1]],
  disp2 = disp[[2]],
  SIMPLIFY = FALSE
)

data &lt;- do.call(what = "rbind", args = data)

ggplot2::ggplot(
  data = data,
  mapping = ggplot2::aes(x = value, fill = condition)
) +
  ggplot2::facet_grid(
    rows = ggplot2::vars(.data$dispersion2),
    cols = ggplot2::vars(.data$dispersion1),
    labeller = ggplot2::labeller(
      .rows = ggplot2::label_both,
      .cols = ggplot2::label_both
    )
  ) +
  ggplot2::geom_density(alpha = 0.3) +
  ggplot2::coord_cartesian(xlim = c(0, 50)) +
  ggplot2::labs(
    x = "Value",
    y = "Density",
    fill = "Condition",
    caption = "Mean1=10, Mean2=15, ratio=1.5"
  )

</code></pre>

<hr>
<h2 id='t_test_paired'>Paired and one-sample t-Tests</h2><span id='topic+t_test_paired'></span>

<h3>Description</h3>

<p>Performs paired and one-sample t-Tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t_test_paired(data, alternative = "two.sided", ci_level = NULL, mean_null = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="t_test_paired_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of normal values from sample
1 and the second element is the vector of normal values from sample 2.
Both vectors must be the same sample size and sorted by the
subject/item index. If <code>length(data) == 1L</code>, the one-sample test is
used. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default output
from <code><a href="#topic+sim_log_lognormal">sim_log_lognormal()</a></code>.</p>
</td></tr>
<tr><td><code id="t_test_paired_+3A_alternative">alternative</code></td>
<td>
<p>(string: <code>"two.sided"</code>)<br />
The alternative hypothesis. Must be one of <code>"two.sided"</code>, <code>"greater"</code>,
or <code>"less"</code>. See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="t_test_paired_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.</p>
</td></tr>
<tr><td><code id="t_test_paired_+3A_mean_null">mean_null</code></td>
<td>
<p>(Scalar numeric: <code>0</code>; <code style="white-space: pre;">&#8288;(-Inf, Inf)&#8288;</code>)<br />
The mean or mean difference assumed under the null hypothesis. See
'Details' for additional information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>The one-sample test is used for both the true one-sample scenario and for the
paired differences from a dependent two-sample scenario. Below we use paired
difference language as that is the most common case. The hypotheses for the
paired t-test are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: \mu_{diff} = \mu_{null} \\
H_{alt} &amp;: \begin{cases}
  \mu_{diff} \neq \mu_{null} &amp; \text{two-sided}\\
  \mu_{diff} &gt; \mu_{null} &amp; \text{greater than}\\
  \mu_{diff} &lt; \mu_{null} &amp; \text{less than}
\end{cases}
\end{aligned}
</code>
</p>

<p>where <code class="reqn">\mu_{diff} = AM(X_2 - X_1)</code> is the arithmetic mean of the paired
differences (sample 2 - sample 1) and <code class="reqn">\mu_{null}</code> is a constant for the
assumed population mean difference (usually <code class="reqn">\mu_{null} = 0</code>).
</p>
<p>The test statistic is
</p>
<p style="text-align: center;"><code class="reqn">
T = \frac{\bar{x}_{diff} - \mu_{null}}{\sqrt{\frac{s^2}{n}}}
</code>
</p>

<p>where <code class="reqn">\bar{x}_{diff}</code> is the sample mean of the differences, <code class="reqn">\mu_{null}</code>
is the population mean difference assumed under the null hypothesis, <code class="reqn">n</code>
is the sample size of the differences, and <code class="reqn">s^2</code> is the sample variance.
</p>
<p>The critical value of the test statistic has degrees of freedom
</p>
<p style="text-align: center;"><code class="reqn">
df = n-1
</code>
</p>

<p>and the p-value is calculated as
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
p &amp;= \begin{cases}
  2 \text{min} \{P(T \geq t_{n-1} \mid H_{null}), P(T \leq t_{n-1} \mid H_{null})\} &amp; \text{two-sided}\\
  P(T \geq t_{n-1} \mid H_{null}) &amp; \text{greater than}\\
  P(T \leq t_{n-1} \mid H_{null}) &amp; \text{less than}
\end{cases}
\end{aligned}
</code>
</p>

<p>Let <code class="reqn">GM(\cdot)</code> be the geometric mean and <code class="reqn">AM(\cdot)</code> be the
arithmetic mean. For dependent lognormal samples <code class="reqn">X_1</code> and <code class="reqn">X_2</code> it
follows that <code class="reqn">\ln X_1</code> and <code class="reqn">\ln X_2</code> are dependent normally
distributed variables. Setting <code class="reqn">\mu_{diff} = AM(\ln X_2 - \ln X_1)</code>
we have
</p>
<p style="text-align: center;"><code class="reqn">
e^{\mu_{diff}} = GM\left( \frac{X_2}{X_1} \right)
</code>
</p>

<p>This forms the basis for making inference about the geometric mean ratio of
the original lognormal data using the mean difference of the log transformed
normal data.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>t</code>  </td><td style="text-align: left;"> Value of the t-statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>df</code> </td><td style="text-align: left;"> Degrees of freedom for the t-statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>p</code>  </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean_diff</code> </td><td style="text-align: left;"> Estimated mean or mean of the differences
(sample 2  sample 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code>  </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>     </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>     </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>n</code>           </td><td style="text-align: left;"> Number of paired observations. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>method</code>      </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> The alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>ci_level</code>    </td><td style="text-align: left;"> The confidence level. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean_null</code>   </td><td style="text-align: left;"> Assumed population mean of the differences
under the null hypothesis.
</td>
</tr>

</table>



<h3>References</h3>

<p>Julious SA (2004).
&ldquo;Sample sizes for clinical trials with Normal data.&rdquo;
<em>Statistics in Medicine</em>, <b>23</b>(12), 1921&ndash;1986.
<a href="https://doi.org/10.1002/sim.1783">doi:10.1002/sim.1783</a>.
</p>
<p>Hauschke D, Steinijans VW, Diletti E, Burke M (1992).
&ldquo;Sample size determination for bioequivalence assessment using a multiplicative model.&rdquo;
<em>Journal of Pharmacokinetics and Biopharmaceutics</em>, <b>20</b>(5), 557&ndash;561.
ISSN 0090-466X, <a href="https://doi.org/10.1007/BF01061471">doi:10.1007/BF01061471</a>.
</p>
<p>Johnson NL, Kotz S, Balakrishnan N (1994).
<em>Continuous univariate distributions</em>,  Wiley series in probability and mathematical statistics, 2nd ed edition.
Wiley, New York.
ISBN 9780471584957 9780471584940.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">stats::t.test()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# t_test_paired() examples
#----------------------------------------------------------------------------
library(depower)

# One-sample t-test
set.seed(1234)
t_test1 &lt;- sim_log_lognormal(
  n1 = 40,
  ratio = 1.5,
  cv1 = 0.4
) |&gt;
  t_test_paired(ci_level = 0.95)

t_test1

# Paired t-test using two dependent samples
set.seed(1234)
t_test2 &lt;- sim_log_lognormal(
  n1 = 40,
  n2 = 40,
  ratio = 1.5,
  cv1 = 0.4,
  cv2 = 0.2,
  cor = 0.3
) |&gt;
  t_test_paired(ci_level = 0.95)

t_test2

</code></pre>

<hr>
<h2 id='t_test_welch'>Welch's t-Test</h2><span id='topic+t_test_welch'></span>

<h3>Description</h3>

<p>Performs Welch's independent two-sample t-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t_test_welch(data, alternative = "two.sided", ci_level = NULL, mean_null = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="t_test_welch_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of normal values from group
1 and the second element is the vector of normal values from group 2.
<a href="base.html#topic+NA">NA</a>s are silently excluded. The default output from
<code><a href="#topic+sim_log_lognormal">sim_log_lognormal()</a></code>.</p>
</td></tr>
<tr><td><code id="t_test_welch_+3A_alternative">alternative</code></td>
<td>
<p>(string: <code>"two.sided"</code>)<br />
The alternative hypothesis. Must be one of <code>"two.sided"</code>, <code>"greater"</code>,
or <code>"less"</code>. See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="t_test_welch_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.</p>
</td></tr>
<tr><td><code id="t_test_welch_+3A_mean_null">mean_null</code></td>
<td>
<p>(Scalar numeric: <code>0</code>; <code style="white-space: pre;">&#8288;(-Inf, Inf)&#8288;</code>)<br />
The difference of means assumed under the null hypothesis. See
'Details' for additional information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>The hypotheses for Welch's independent two-sample t-test are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: \mu_2 - \mu_1 = \mu_{null} \\
H_{alt} &amp;: \begin{cases}
  \mu_2 - \mu_1 \neq \mu_{null} &amp; \text{two-sided}\\
  \mu_2 - \mu_1 &gt; \mu_{null} &amp; \text{greater than}\\
  \mu_2 - \mu_1 &lt; \mu_{null} &amp; \text{less than}
\end{cases}
\end{aligned}
</code>
</p>

<p>where <code class="reqn">\mu_1</code> is the population mean of group 1, <code class="reqn">\mu_2</code> is the
population mean of group 2, and <code class="reqn">\mu_{null}</code> is a constant for the assumed
difference of population means (usually <code class="reqn">\mu_{null} = 0</code>).
</p>
<p>The test statistic is
</p>
<p style="text-align: center;"><code class="reqn">
T = \frac{(\bar{x}_2 - \bar{x}_1) - \mu_{null}}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}}
</code>
</p>

<p>where <code class="reqn">\bar{x}_1</code> and <code class="reqn">\bar{x}_2</code> are the sample means, <code class="reqn">\mu_{null}</code>
is the difference of population means assumed under the null hypothesis,
<code class="reqn">n_1</code> and <code class="reqn">n_2</code> are the sample sizes, and <code class="reqn">s_1^2</code> and <code class="reqn">s_2^2</code>
are the sample variances.
</p>
<p>The critical value of the test statistic uses the WelchSatterthwaite degrees
of freedom
</p>
<p style="text-align: center;"><code class="reqn">
v = \frac{\left( \frac{s_1^2}{n_1} + \frac{s_2^2}{n_2} \right)^2}
         {(N_1 - 1)^{-1}\left( \frac{s_1^2}{n_1} \right)^2 +
         (N_2 - 1)^{-1}\left( \frac{s_2^2}{n_2} \right)^2}
</code>
</p>

<p>and the p-value is calculated as
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
p &amp;= \begin{cases}
  2 \text{min} \{P(T \geq t_{v} \mid H_{null}), P(T \leq t_{v} \mid H_{null})\} &amp; \text{two-sided}\\
  P(T \geq t_{v} \mid H_{null}) &amp; \text{greater than}\\
  P(T \leq t_{v} \mid H_{null}) &amp; \text{less than}
\end{cases}
\end{aligned}
</code>
</p>

<p>Let <code class="reqn">GM(\cdot)</code> be the geometric mean and <code class="reqn">AM(\cdot)</code> be the
arithmetic mean. For independent lognormal variables <code class="reqn">X_1</code> and <code class="reqn">X_2</code>
it follows that <code class="reqn">\ln X_1</code> and <code class="reqn">\ln X_2</code> are independent normally
distributed variables. Defining
<code class="reqn">\mu_{X_2} - \mu_{X_1} = AM(\ln X_2) - AM(\ln X_1)</code>
we have
</p>
<p style="text-align: center;"><code class="reqn">
e^{\mu_{X_2} - \mu_{X_1}} = \frac{GM(X_2)}{GM(X_1)}
</code>
</p>

<p>This forms the basis for making inference about the ratio of geometric means
of the original lognormal data using the difference of means of the log
transformed normal data.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>t</code>  </td><td style="text-align: left;"> Value of the t-statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>df</code> </td><td style="text-align: left;"> Degrees of freedom for the t-statistic. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>p</code>  </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>diff_mean</code> </td><td style="text-align: left;"> Estimated difference of means
(group 2  group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code>  </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>     </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>     </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Estimated mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Estimated mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9  </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>method</code>      </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>alternative</code> </td><td style="text-align: left;"> The alternative hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>ci_level</code>    </td><td style="text-align: left;"> The confidence level. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;">   </td><td style="text-align: left;"> <code>mean_null</code>   </td><td style="text-align: left;"> Assumed population difference of the means
under the null hypothesis.
</td>
</tr>

</table>



<h3>References</h3>

<p>Julious SA (2004).
&ldquo;Sample sizes for clinical trials with Normal data.&rdquo;
<em>Statistics in Medicine</em>, <b>23</b>(12), 1921&ndash;1986.
<a href="https://doi.org/10.1002/sim.1783">doi:10.1002/sim.1783</a>.
</p>
<p>Hauschke D, Steinijans VW, Diletti E, Burke M (1992).
&ldquo;Sample size determination for bioequivalence assessment using a multiplicative model.&rdquo;
<em>Journal of Pharmacokinetics and Biopharmaceutics</em>, <b>20</b>(5), 557&ndash;561.
ISSN 0090-466X, <a href="https://doi.org/10.1007/BF01061471">doi:10.1007/BF01061471</a>.
</p>
<p>Johnson NL, Kotz S, Balakrishnan N (1994).
<em>Continuous univariate distributions</em>,  Wiley series in probability and mathematical statistics, 2nd ed edition.
Wiley, New York.
ISBN 9780471584957 9780471584940.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">stats::t.test()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# t_test_welch() examples
#----------------------------------------------------------------------------
library(depower)

# Welch's t-test
set.seed(1234)
sim_log_lognormal(
  n1 = 40,
  n2 = 40,
  ratio = 1.5,
  cv1 = 0.4,
  cv2 = 0.4
) |&gt;
  t_test_welch(ci_level = 0.95)

</code></pre>

<hr>
<h2 id='wald_test_bnb'>Wald test for BNB ratio of means</h2><span id='topic+wald_test_bnb'></span>

<h3>Description</h3>

<p>Wald test for the ratio of means from bivariate negative binomial outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wald_test_bnb(data, ci_level = NULL, link = "log", ratio_null = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wald_test_bnb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from sample 1 and the second element is the vector of negative
binomial values from sample 2.
Each vector must be sorted by the subject/item index and must be the
same sample size. <a href="base.html#topic+NA">NA</a>s are silently excluded. The default
output from <code><a href="#topic+sim_bnb">sim_bnb()</a></code>.</p>
</td></tr>
<tr><td><code id="wald_test_bnb_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.</p>
</td></tr>
<tr><td><code id="wald_test_bnb_+3A_link">link</code></td>
<td>
<p>(Scalar string: <code>"log"</code>)<br />
The one-to-one link function for transformation of the ratio in the
test hypotheses. Must be one of <code>"log"</code> (default), <code>"sqrt"</code>,
<code>"squared"</code>, or &quot;<code style="white-space: pre;">&#8288;identity"&#8288;</code>. See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="wald_test_bnb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The (pre-transformation) ratio of means assumed under the null
hypothesis (sample 2 / sample 1). Typically <code>ratio_null = 1</code>
(no difference). See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="wald_test_bnb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the MLE function <code><a href="#topic+mle_bnb">mle_bnb()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \mid G = g \sim \text{Poisson}(\mu g)</code> and
<code class="reqn">X_2 \mid G = g \sim \text{Poisson}(r \mu g)</code> where
<code class="reqn">G \sim \text{Gamma}(\theta, \theta^{-1})</code> is the random item (subject)
effect. Then <code class="reqn">X_1, X_2 \sim \text{BNB}(\mu, r, \theta)</code> is the joint
distribution where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are dependent (though conditionally
independent), <code class="reqn">X_1</code> is the count outcome for sample 1 of the items
(subjects), <code class="reqn">X_2</code> is the count outcome for sample 2 of the items (subjects),
<code class="reqn">\mu</code> is the conditional mean of sample 1, <code class="reqn">r</code> is the ratio of the
conditional means of sample 2 with respect to sample 1, and <code class="reqn">\theta</code> is
the gamma distribution shape parameter which controls the dispersion and the
correlation between sample 1 and 2.
</p>
<p>The hypotheses for the Wald test of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: f(r) = f(r_{null}) \\
H_{alt} &amp;: f(r) \neq f(r_{null})
\end{aligned}
</code>
</p>

<p>where <code class="reqn">f(\cdot)</code> is a one-to-one link function with nonzero derivative,
<code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of arithmetic
means for sample 2 with respect to sample 1, and <code class="reqn">r_{null}</code> is a constant
for the assumed null population ratio of means (typically <code class="reqn">r_{null} = 1</code>).
</p>
<p>Rettiganti and Nagaraja (2012) found that <code class="reqn">f(r) = r^2</code>,
<code class="reqn">f(r) = r</code>, and <code class="reqn">f(r) = r^{0.5}</code> had greatest power when <code class="reqn">r &lt; 1</code>.
However, when <code class="reqn">r &gt; 1</code>, <code class="reqn">f(r) = \ln r</code>, the likelihood ratio test, and
<code class="reqn">f(r) = r^{0.5}</code> had greatest power. <code class="reqn">f(r) = r^2</code> was biased when
<code class="reqn">r &gt; 1</code>. Both <code class="reqn">f(r) = \ln r</code> and <code class="reqn">f(r) = r^{0.5}</code> produced
acceptable results for any <code class="reqn">r</code> value. These results depend on the use of
asymptotic vs. exact critical values.
</p>
<p>The Wald test statistic is
</p>
<p style="text-align: center;"><code class="reqn">
W(f(\hat{r})) = \left( \frac{f \left( \frac{\bar{x}_2}{\bar{x}_1} \right) - f(r_{null})}{f^{\prime}(\hat{r}) \hat{\sigma}_{\hat{r}}} \right)^2
</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">
\hat{\sigma}^{2}_{\hat{r}} = \frac{\hat{r} (1 + \hat{r}) (\hat{\mu} + \hat{r}\hat{\mu} + \hat{\theta})}{n \left[ \hat{\mu} (1 + \hat{r}) (\hat{\mu} + \hat{\theta}) - \hat{\theta}\hat{r} \right]}
</code>
</p>

<p>Under <code class="reqn">H_{null}</code>, the Wald test statistic is asymptotically distributed
as <code class="reqn">\chi^2_1</code>. The approximate level <code class="reqn">\alpha</code> test rejects
<code class="reqn">H_{null}</code> if <code class="reqn">W(f(\hat{r})) \geq \chi^2_1(1 - \alpha)</code>. Note that
the asymptotic critical value is known to underestimate the exact critical
value. Hence, the nominal significance level may not be achieved for small
sample sizes (possibly <code class="reqn">n \leq 10</code> or <code class="reqn">n \leq 50</code>). The level of
significance inflation also depends on <code class="reqn">f(\cdot)</code> and is most severe for
<code class="reqn">f(r) = r^2</code>, where only the exact critical value is recommended.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code> </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code> </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio</code>      </td><td style="text-align: left;"> Estimated ratio of means (group 2 / group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Estimated mean of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Estimated mean of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>dispersion</code> </td><td style="text-align: left;"> Estimated dispersion. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> The sample size of sample 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
9 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> The sample size of sample 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ci_level</code> </td><td style="text-align: left;"> The confidence level. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>link</code> </td><td style="text-align: left;"> Link function used to transform the ratio of means in the test hypotheses. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Assumed ratio of means under the null hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
15 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Information from the optimizer.
</td>
</tr>

</table>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# wald_test_bnb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
sim_bnb(
  n = 40,
  mean1 = 10,
  ratio = 1.2,
  dispersion = 2
) |&gt;
  wald_test_bnb()

</code></pre>

<hr>
<h2 id='wald_test_nb'>Wald test for NB ratio of means</h2><span id='topic+wald_test_nb'></span>

<h3>Description</h3>

<p>Wald test for the ratio of means from two independent negative binomial
outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wald_test_nb(
  data,
  equal_dispersion = FALSE,
  ci_level = NULL,
  link = "log",
  ratio_null = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wald_test_nb_+3A_data">data</code></td>
<td>
<p>(list)<br />
A list whose first element is the vector of negative binomial values
from group 1 and the second element is the vector of negative binomial
values from group 2.
<a href="base.html#topic+NA">NA</a>s are silently excluded. The default output from
<code><a href="#topic+sim_nb">sim_nb()</a></code>.</p>
</td></tr>
<tr><td><code id="wald_test_nb_+3A_equal_dispersion">equal_dispersion</code></td>
<td>
<p>(Scalar logical: <code>FALSE</code>)<br />
If <code>TRUE</code>, the Wald test is calculated assuming both groups have the
same population dispersion parameter. If <code>FALSE</code> (default), the Wald
test is calculated assuming different dispersions.</p>
</td></tr>
<tr><td><code id="wald_test_nb_+3A_ci_level">ci_level</code></td>
<td>
<p>(Scalar numeric: <code>NULL</code>; <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>)<br />
If <code>NULL</code>, confidence intervals are set as <code>NA</code>. If in <code style="white-space: pre;">&#8288;(0, 1)&#8288;</code>,
confidence intervals are calculated at the specified level.</p>
</td></tr>
<tr><td><code id="wald_test_nb_+3A_link">link</code></td>
<td>
<p>(Scalar string: <code>"log"</code>)<br />
The one-to-one link function for transformation of the ratio in the
test hypotheses. Must be one of <code>"log"</code> (default), <code>"sqrt"</code>,
<code>"squared"</code>, or &quot;<code style="white-space: pre;">&#8288;identity"&#8288;</code>.</p>
</td></tr>
<tr><td><code id="wald_test_nb_+3A_ratio_null">ratio_null</code></td>
<td>
<p>(Scalar numeric: <code>1</code>; <code style="white-space: pre;">&#8288;(0, Inf)&#8288;</code>)<br />
The (pre-transformation) ratio of means assumed under the null
hypothesis (group 2 / group 1). Typically <code>ratio_null = 1</code>
(no difference). See 'Details' for additional information.</p>
</td></tr>
<tr><td><code id="wald_test_nb_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the MLE function <code><a href="#topic+mle_nb">mle_nb()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is primarily designed for speed in simulation. Missing values
are silently excluded.
</p>
<p>Suppose <code class="reqn">X_1 \sim NB(\mu, \theta_1)</code> and
<code class="reqn">X_2 \sim NB(r\mu, \theta_2)</code> where <code class="reqn">X_1</code> and <code class="reqn">X_2</code> are
independent, <code class="reqn">X_1</code> is the count outcome for items in group 1, <code class="reqn">X_2</code>
is the count outcome for items in group 2, <code class="reqn">\mu</code> is the arithmetic mean
count in group 1, <code class="reqn">r</code> is the ratio of arithmetic means for group 2 with
respect to group 1, <code class="reqn">\theta_1</code> is the dispersion parameter of group 1,
and <code class="reqn">\theta_2</code> is the dispersion parameter of group 2.
</p>
<p>The hypotheses for the Wald test of <code class="reqn">r</code> are
</p>
<p style="text-align: center;"><code class="reqn">
\begin{aligned}
H_{null} &amp;: f(r) = f(r_{null}) \\
H_{alt} &amp;: f(r) \neq f(r_{null})
\end{aligned}
</code>
</p>

<p>where <code class="reqn">f(\cdot)</code> is a one-to-one link function with nonzero derivative,
<code class="reqn">r = \frac{\bar{X}_2}{\bar{X}_1}</code> is the population ratio of arithmetic
means for group 2 with respect to group 1, and <code class="reqn">r_{null}</code> is a constant
for the assumed null population ratio of means (typically <code class="reqn">r_{null} = 1</code>).
</p>
<p>Rettiganti and Nagaraja (2012) found that <code class="reqn">f(r) = r^2</code> and
<code class="reqn">f(r) = r</code> had greatest power when <code class="reqn">r &lt; 1</code>. However, when
<code class="reqn">r &gt; 1</code>, <code class="reqn">f(r) = \ln r</code>, the likelihood ratio test, and the Rao score
test have greatest power. Note that <code class="reqn">f(r) = \ln r</code>, LRT, and RST were
unbiased tests while the <code class="reqn">f(r) = r</code> and <code class="reqn">f(r) = r^2</code> tests were
biased when <code class="reqn">r &gt; 1</code>. The <code class="reqn">f(r) = \ln r</code>, LRT, and RST produced
acceptable results for any <code class="reqn">r</code> value. These results depend on the use of
asymptotic vs. exact critical values.
</p>
<p>The Wald test statistic is
</p>
<p style="text-align: center;"><code class="reqn">
W(f(\hat{r})) = \left( \frac{f \left( \frac{\bar{x}_2}{\bar{x}_1} \right) - f(r_{null})}{f^{\prime}(\hat{r}) \hat{\sigma}_{\hat{r}}} \right)^2
</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">
\hat{\sigma}^{2}_{\hat{r}} = \frac{\hat{r} \left[ n_1 \hat{\theta}_1 (\hat{r} \hat{\mu} + \hat{\theta}_2) + n_2 \hat{\theta}_2 \hat{r} (\hat{\mu} + \hat{\theta}_1) \right]}{n_1 n_2 \hat{\theta}_1 \hat{\theta}_2 \hat{\mu}}
</code>
</p>

<p>Under <code class="reqn">H_{null}</code>, the Wald test statistic is asymptotically distributed
as <code class="reqn">\chi^2_1</code>. The approximate level <code class="reqn">\alpha</code> test rejects
<code class="reqn">H_{null}</code> if <code class="reqn">W(f(\hat{r})) \geq \chi^2_1(1 - \alpha)</code>. Note that
the asymptotic critical value is known to underestimate the exact critical
value. Hence, the nominal significance level may not be achieved for small
sample sizes (possibly <code class="reqn">n \leq 10</code> or <code class="reqn">n \leq 50</code>). The level of
significance inflation also depends on <code class="reqn">f(\cdot)</code> and is most severe for
<code class="reqn">f(r) = r^2</code>, where only the exact critical value is recommended.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
Slot </td><td style="text-align: left;"> Subslot </td><td style="text-align: left;"> Name </td><td style="text-align: left;"> Description </td>
</tr>
<tr>
 <td style="text-align: left;">
1 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>chisq</code> </td><td style="text-align: left;"> <code class="reqn">\chi^2</code> test statistic for the ratio of means. </td>
</tr>
<tr>
 <td style="text-align: left;">
2 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>df</code>    </td><td style="text-align: left;"> Degrees of freedom. </td>
</tr>
<tr>
 <td style="text-align: left;">
3 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>p</code>     </td><td style="text-align: left;"> p-value. </td>
</tr>
<tr>
 <td style="text-align: left;">

4 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio</code>      </td><td style="text-align: left;"> Estimated ratio of means (group 2 / group 1). </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 1 </td><td style="text-align: left;"> <code>estimate</code> </td><td style="text-align: left;"> Point estimate. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> <code>lower</code>    </td><td style="text-align: left;"> Confidence interval lower bound. </td>
</tr>
<tr>
 <td style="text-align: left;">
4 </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> <code>upper</code>    </td><td style="text-align: left;"> Confidence interval upper bound. </td>
</tr>
<tr>
 <td style="text-align: left;">

5 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mean1</code> </td><td style="text-align: left;"> Estimated mean of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
6 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mean2</code> </td><td style="text-align: left;"> Estimated mean of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
7 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>dispersion1</code> </td><td style="text-align: left;"> Estimated dispersion of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
8 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>dispersion2</code> </td><td style="text-align: left;"> Estimated dispersion of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
9  </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n1</code> </td><td style="text-align: left;"> Sample size of group 1. </td>
</tr>
<tr>
 <td style="text-align: left;">
10 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>n2</code> </td><td style="text-align: left;"> Sample size of group 2. </td>
</tr>
<tr>
 <td style="text-align: left;">
11 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> Method used for the results. </td>
</tr>
<tr>
 <td style="text-align: left;">
12 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ci_level</code> </td><td style="text-align: left;"> The confidence level. </td>
</tr>
<tr>
 <td style="text-align: left;">
13 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>equal_dispersion</code> </td><td style="text-align: left;"> Whether or not equal dispersions were assumed. </td>
</tr>
<tr>
 <td style="text-align: left;">
14 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>link</code> </td><td style="text-align: left;"> Link function used to transform the ratio of means in the test hypotheses. </td>
</tr>
<tr>
 <td style="text-align: left;">
15 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>ratio_null</code> </td><td style="text-align: left;"> Assumed ratio of means under the null hypothesis. </td>
</tr>
<tr>
 <td style="text-align: left;">
16 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_code</code> </td><td style="text-align: left;"> Integer indicating why the optimization process terminated. </td>
</tr>
<tr>
 <td style="text-align: left;">
17 </td><td style="text-align: left;"> </td><td style="text-align: left;"> <code>mle_message</code> </td><td style="text-align: left;"> Information from the optimizer.
</td>
</tr>

</table>



<h3>References</h3>

<p>Rettiganti M, Nagaraja HN (2012).
&ldquo;Power Analyses for Negative Binomial Models with Application to Multiple Sclerosis Clinical Trials.&rdquo;
<em>Journal of Biopharmaceutical Statistics</em>, <b>22</b>(2), 237&ndash;259.
ISSN 1054-3406, 1520-5711, <a href="https://doi.org/10.1080/10543406.2010.528105">doi:10.1080/10543406.2010.528105</a>.
</p>
<p>Aban IB, Cutter GR, Mavinga N (2009).
&ldquo;Inferences and power analysis concerning two negative binomial distributions with an application to MRI lesion counts data.&rdquo;
<em>Computational Statistics &amp; Data Analysis</em>, <b>53</b>(3), 820&ndash;833.
ISSN 01679473, <a href="https://doi.org/10.1016/j.csda.2008.07.034">doi:10.1016/j.csda.2008.07.034</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#----------------------------------------------------------------------------
# wald_test_nb() examples
#----------------------------------------------------------------------------
library(depower)

set.seed(1234)
sim_nb(
  n1 = 60,
  n2 = 40,
  mean1 = 10,
  ratio = 1.5,
  dispersion1 = 2,
  dispersion2 = 8
) |&gt;
  wald_test_nb()

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
