<!DOCTYPE html><html lang="en"><head><title>Help for package detectRUNS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {detectRUNS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chromosomeLength'><p>Function to found max position for each chromosome</p></a></li>
<li><a href='#consecutiveRuns'><p>Function to detect consecutive runs in a vector (individual's genotypes)</p></a></li>
<li><a href='#consecutiveRUNS.run'><p>Main function to detect genomic RUNS (ROHom/ROHet) using the consecutive method</p></a></li>
<li><a href='#consecutiveRunsCpp'><p>Function to detect consecutive runs in a vector (individual's genotypes)</p></a></li>
<li><a href='#createRUNdf'><p>Function to create a dataframe of RUNS per individual animal</p>
Requires a map file (other filename to read or R object)
Parameters on maximum number of missing and opposite genotypes in the run (not the window) are implemented here</a></li>
<li><a href='#findOppositeAndMissing'><p>Function to calculate oppositeAndMissingGenotypes array</p></a></li>
<li><a href='#Froh_inbreeding'><p>Function to calculated Froh genome-wide or chromosome-wide</p></a></li>
<li><a href='#Froh_inbreedingClass'><p>Function to calculated Froh using a ROH-class</p></a></li>
<li><a href='#genoConvert'><p>Convert 0/1/2 genotypes to 0/1</p></a></li>
<li><a href='#genoConvertCpp'><p>Convert 0/1/2 genotypes to 0/1</p></a></li>
<li><a href='#heteroZygotTest'><p>Function to check whether a window is (loosely) heterozygous or not</p></a></li>
<li><a href='#heteroZygotTestCpp'><p>Function to check whether a window is (loosely) heterozygous or not</p></a></li>
<li><a href='#homoZygotTest'><p>Function to check whether a window is (loosely) homozygous or not</p></a></li>
<li><a href='#homoZygotTestCpp'><p>Function to check whether a window is (loosely) homozygous or not</p></a></li>
<li><a href='#pedConvertCpp'><p>Convert ped genotypes to 0/1</p></a></li>
<li><a href='#plot_DistributionRuns'><p>Plot Distribution of runs</p></a></li>
<li><a href='#plot_InbreedingChr'><p>Plot Froh-based inbreeding coefficients by group</p></a></li>
<li><a href='#plot_manhattanRuns'><p>Plot the proportion of times SNPs are inside runs - MANHATTAN PLOT</p></a></li>
<li><a href='#plot_PatternRuns'><p>Plot sum of run-lengths (or average run-lengths) against the number of runs per individual</p></a></li>
<li><a href='#plot_Runs'><p>Function to plot runs per individual</p></a></li>
<li><a href='#plot_SnpsInRuns'><p>Plot the number of times each SNP falls inside runs</p></a></li>
<li><a href='#plot_StackedRuns'><p>Plot stacked runs</p></a></li>
<li><a href='#plot_ViolinRuns'><p>Violin plot of run length per individual (either sum or mean)</p></a></li>
<li><a href='#readExternalRuns'><p>Read runs from external files</p></a></li>
<li><a href='#readPOPCpp'><p>Function to return a dataframe of population (POP, ID)</p></a></li>
<li><a href='#reorderDF'><p>Function to reorder data frames by CHROMOSOME</p></a></li>
<li><a href='#slidingRuns'><p>Function to detect runs using sliding window approach</p></a></li>
<li><a href='#slidingRUNS.run'><p>Main function to detect RUNS (ROHom/ROHet) using sliding windows (a la Plink)</p></a></li>
<li><a href='#slidingWindow'><p>Function to slide a window over a vector (individual's genotypes)</p></a></li>
<li><a href='#slidingWindowCpp'><p>Function to slide a window over a vector (individual's genotypes)</p></a></li>
<li><a href='#snpInRun'><p>Function to return a vector of T/F for whether a SNP is or not in a RUN</p></a></li>
<li><a href='#snpInRunCpp'><p>Function to return a vector of T/F for whether a SNP is or not in a RUN</p></a></li>
<li><a href='#snpInsideRuns'><p>Function to count number of times a SNP is in a RUN</p></a></li>
<li><a href='#snpInsideRunsCpp'><p>Function to count number of times a SNP is in a RUN</p></a></li>
<li><a href='#summaryRuns'><p>Summary statistics on detected runs</p></a></li>
<li><a href='#tableRuns'><p>Function to retrieve most common runs in the population</p></a></li>
<li><a href='#writeRUN'><p>Function to write out RUNS per individual animal</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detect Runs of Homozygosity and Runs of Heterozygosity in
Diploid Genomes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-10-24</td>
</tr>
<tr>
<td>Description:</td>
<td>Detection of runs of homozygosity and of heterozygosity
    in diploid genomes using two methods: sliding windows (Purcell et al (2007)
    &lt;<a href="https://doi.org/10.1086%2F519795">doi:10.1086/519795</a>&gt;) and consecutive runs (Marras et al (2015)
    &lt;<a href="https://doi.org/10.1111%2Fage.12259">doi:10.1111/age.12259</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>plyr, iterators, itertools, ggplot2, reshape2, Rcpp,
gridExtra, data.table</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, prettydoc</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/bioinformatics-ptp/detectRUNS/tree/master/detectRUNS">https://github.com/bioinformatics-ptp/detectRUNS/tree/master/detectRUNS</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bioinformatics-ptp/detectRUNS/issues">https://github.com/bioinformatics-ptp/detectRUNS/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-10-24 13:23:54 UTC; filippo</td>
</tr>
<tr>
<td>Author:</td>
<td>Filippo Biscarini [aut, cre],
  Paolo Cozzi [aut],
  Giustino Gaspa [aut],
  Gabriele Marras [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Filippo Biscarini &lt;filippo.biscarini@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-24 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='chromosomeLength'>Function to found max position for each chromosome</h2><span id='topic+chromosomeLength'></span>

<h3>Description</h3>

<p>Function to found max position for each chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromosomeLength(mapFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chromosomeLength_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Create a data frame with the max position in map file (plink format)
</p>


<h3>Value</h3>

<p>A data frame with the max position for chromosome
</p>

<hr>
<h2 id='consecutiveRuns'>Function to detect consecutive runs in a vector (individual's genotypes)</h2><span id='topic+consecutiveRuns'></span>

<h3>Description</h3>

<p>This is a core function. It implements the consecutive method for detection of runs in diploid genomes
(see Marras et al. 2015)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consecutiveRuns(indGeno, individual, mapFile, ROHet = TRUE, minSNP = 3,
  maxOppositeGenotype = 1, maxMiss = 1, minLengthBps = 1000,
  maxGap = 10^6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consecutiveRuns_+3A_indgeno">indGeno</code></td>
<td>
<p>vector of 0/1/NAs of individual genotypes (0: homozygote; 1: heterozygote)</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_individual">individual</code></td>
<td>
<p>list of group (breed, population, case/control etc.) and ID of individual sample</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_rohet">ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_minsnp">minSNP</code></td>
<td>
<p>minimum number of SNP in a run</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_maxoppositegenotype">maxOppositeGenotype</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max. n. of missing SNP</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_minlengthbps">minLengthBps</code></td>
<td>
<p>min length of a run in bps</p>
</td></tr>
<tr><td><code id="consecutiveRuns_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The consecutive method detect runs by consecutively scanning SNP loci along the genome.
No sliding windows are used. Checks on minimum n. of SNP, max n. of opposite and missing genotypes,
max gap between adjacent loci and minimum length of the run are implemented (as in the sliding window method).
Both runs of homozygosity (RoHom) and of heterozygosity (RoHet) can be search for (option ROHet: TRUE/FALSE)
</p>


<h3>Value</h3>

<p>A data frame of runs per individual sample
</p>

<hr>
<h2 id='consecutiveRUNS.run'>Main function to detect genomic RUNS (ROHom/ROHet) using the consecutive method</h2><span id='topic+consecutiveRUNS.run'></span>

<h3>Description</h3>

<p>This is the main detectRUNS function to scan the genome for runs (of homozygosity or heterozygosity)
using the consecutive method (Marras et al. 2015, Animal Genetics 46(2):110-121).
All parameters to detect runs (e.g. minimum n. of SNP, max n. of missing genotypes,
max n. of opposite genotypes etc.) are specified here.
Input data are in the ped/map
Plink format (https://www.cog-genomics.org/plink/1.9/input#ped)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consecutiveRUNS.run(genotypeFile, mapFile, ROHet = FALSE,
  maxOppRun = 0, maxMissRun = 0, minSNP = 15, minLengthBps = 1000,
  maxGap = 10^6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consecutiveRUNS.run_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file path</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_mapfile">mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_rohet">ROHet</code></td>
<td>
<p>should we look for ROHet or ROHom? (default = FALSE)</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_maxopprun">maxOppRun</code></td>
<td>
<p>max n. of opposite genotype SNPs in the run (default = 0)</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_maxmissrun">maxMissRun</code></td>
<td>
<p>max n. of missing SNPs in the run (default = 0)</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_minsnp">minSNP</code></td>
<td>
<p>minimum n. of SNP in a RUN (default = 15)</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_minlengthbps">minLengthBps</code></td>
<td>
<p>minimum length of run in bps (defaults to 1000 bps = 1 kbps)</p>
</td></tr>
<tr><td><code id="consecutiveRUNS.run_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run (defaults to 10^6)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function scans the genome (diploid) for runs using the consecutive method.
This is a wrapper function for many component functions that handle the input data (ped/map files), performs internal conversions,
accepts parameters specifications, selects the statistical method to detect runs (sliding windows, consecutive loci) and whether
runs of homozygosity (RoHom) or of heterozygosity (RoHet) are looked for.
</p>
<p>In the ped file, the groups samples belong to can be specified (first column). This is important if comparisons between
human ethnic groups or between animal breeds or plant varieties or biological populations are to be performed.
Also, if cases and controls are to be compared, this is the place where this information needs to be specified.
</p>
<p>This function returns a data frame with all runs detected in the dataset. This data frame can then be written out to a csv file.
The data frame is, in turn, the input for other functions of the detectRUNS package that create plots and produce statistics
of the results (see plot and statistic functions in this manual, and/or refer to the vignette of detectRUNS).
</p>


<h3>Value</h3>

<p>A dataframe with RUNs of Homozygosity or Heterozygosity in the analysed dataset.
The returned dataframe contains the following seven columns: &quot;group&quot;, &quot;id&quot;, &quot;chrom&quot;,
&quot;nSNP&quot;, &quot;from&quot;, &quot;to&quot;, &quot;lengthBps&quot; (group: population, breed, case/control etc.;
id: individual identifier; chrom: chromosome on which the run is located;
nSNP: number of SNPs in the run; from: starting position of the run, in bps;
to: end position of the run, in bps; lengthBps: size of the run)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")
# calculating runs with consecutive run approach
## Not run: 
# skipping runs calculation
runs &lt;- consecutiveRUNS.run(genotypeFile, mapFile, minSNP = 15, ROHet = FALSE,
maxOppRun = 0, maxMissRun = 0, maxGap=10^6,
minLengthBps = 100000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.consecutive.csv", package="detectRUNS")
colClasses &lt;- c(rep("character", 3), rep("numeric", 4)  )
runs &lt;- read.csv2(runsFile, header = TRUE, stringsAsFactors = FALSE,
colClasses = colClasses)

</code></pre>

<hr>
<h2 id='consecutiveRunsCpp'>Function to detect consecutive runs in a vector (individual's genotypes)</h2><span id='topic+consecutiveRunsCpp'></span>

<h3>Description</h3>

<p>This is a core function. It implements the consecutive method for detection of runs in diploid genomes
(see Marras et al. 2015)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consecutiveRunsCpp(indGeno, individual, mapFile, ROHet = TRUE,
  minSNP = 3L, maxOppositeGenotype = 1L, maxMiss = 1L,
  minLengthBps = 1000L, maxGap = 1000000L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="consecutiveRunsCpp_+3A_indgeno">indGeno</code></td>
<td>
<p>vector of 0/1/NAs of individual genotypes (0: homozygote; 1: heterozygote)</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_individual">individual</code></td>
<td>
<p>list of group (breed, population, case/control etc.) and ID of individual sample</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_rohet">ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_minsnp">minSNP</code></td>
<td>
<p>minimum number of SNP in a run</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_maxoppositegenotype">maxOppositeGenotype</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max. n. of missing SNP</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_minlengthbps">minLengthBps</code></td>
<td>
<p>min length of a run in bps</p>
</td></tr>
<tr><td><code id="consecutiveRunsCpp_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The consecutive method detect runs by consecutively scanning SNP loci along the genome.
No sliding windows are used. Checks on minimum n. of SNP, max n. of opposite and missing genotypes,
max gap between adjacent loci and minimum length of the run are implemented (as in the sliding window method).
Both runs of homozygosity (RoHom) and of heterozygosity (RoHet) can be search for (option ROHet: TRUE/FALSE)
</p>


<h3>Value</h3>

<p>A data frame of runs per individual sample
</p>

<hr>
<h2 id='createRUNdf'>Function to create a dataframe of RUNS per individual animal
Requires a map file (other filename to read or R object)
Parameters on maximum number of missing and opposite genotypes in the run (not the window) are implemented here</h2><span id='topic+createRUNdf'></span>

<h3>Description</h3>

<p>Function to create a dataframe of RUNS per individual animal
Requires a map file (other filename to read or R object)
Parameters on maximum number of missing and opposite genotypes in the run (not the window) are implemented here
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createRUNdf(snpRun, mapFile, minSNP = 3, minLengthBps = 1000,
  minDensity = 1/10, oppositeAndMissingSNP, maxOppRun = NULL,
  maxMissRun = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createRUNdf_+3A_snprun">snpRun</code></td>
<td>
<p>vector of TRUE/FALSE (is the SNP in a RUN?)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink-like map file (data.frame)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_minsnp">minSNP</code></td>
<td>
<p>minimum n. of SNP to call a RUN</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_minlengthbps">minLengthBps</code></td>
<td>
<p>minimum length of run in bps (defaults to 1000 bps = 1 kbps)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_mindensity">minDensity</code></td>
<td>
<p>minimum n. of SNP per kbps (defaults to 0.1 = 1 SNP every 10 kbps)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_oppositeandmissingsnp">oppositeAndMissingSNP</code></td>
<td>
<p>indexed array of missing and opposite genotypes (SNP order in the genome is the index)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_maxopprun">maxOppRun</code></td>
<td>
<p>max n. of opposite genotype SNPs in the run (not in the window!)</p>
</td></tr>
<tr><td><code id="createRUNdf_+3A_maxmissrun">maxMissRun</code></td>
<td>
<p>max n. of missing SNPs in the run (not in the window!)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with RUNS per animal
</p>

<hr>
<h2 id='findOppositeAndMissing'>Function to calculate oppositeAndMissingGenotypes array</h2><span id='topic+findOppositeAndMissing'></span>

<h3>Description</h3>

<p>This is an helper function, this will be called by another function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findOppositeAndMissing(data, ROHet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findOppositeAndMissing_+3A_data">data</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td></tr>
<tr><td><code id="findOppositeAndMissing_+3A_rohet">ROHet</code></td>
<td>
<p>TRUE in ROHet evaluation, FALSE for ROHom</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character array; names will be index in which opposite and missing
snps are found in data array
</p>

<hr>
<h2 id='Froh_inbreeding'>Function to calculated Froh genome-wide or chromosome-wide</h2><span id='topic+Froh_inbreeding'></span>

<h3>Description</h3>

<p>This function calculates the individual inbreeding coefficients based on runs of
homozygosity (ROH), either per-chromosome (chromosome-wide) or based on the
entire genome (genome-wide). See details of calculations below
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Froh_inbreeding(runs, mapFile, genome_wide = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Froh_inbreeding_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with results on runs</p>
</td></tr>
<tr><td><code id="Froh_inbreeding_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (to retrieve SNP position)</p>
</td></tr>
<tr><td><code id="Froh_inbreeding_+3A_genome_wide">genome_wide</code></td>
<td>
<p>vector of TRUE/FALSE (genome-wide or chromosome-wide;
defaults to TRUE/genome-wide)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Froh is calculated as:
</p>
<p><code class="reqn"> F_{ROH} = \frac{\sum ROH_{length}}{Length_{genome}} </code>
</p>
<p>Depending on whether genome-wide or chromosome-wide calculations are required,
the terms in the numerator and denominator will refer to the entire genome
or will be restricted to specific chromosomes.
</p>


<h3>Value</h3>

<p>A data frame with the inbreeding coefficients of each individual sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

Froh_inbreeding(runs = runsDF, mapFile = mapFile)
Froh_inbreeding(runs = runsDF, mapFile = mapFile, genome_wide=FALSE)

</code></pre>

<hr>
<h2 id='Froh_inbreedingClass'>Function to calculated Froh using a ROH-class</h2><span id='topic+Froh_inbreedingClass'></span>

<h3>Description</h3>

<p>This function calculates the individual inbreeding coefficients based on runs of
homozygosity (ROH) using only ROH of specific size classes.
The parameter <code>class</code> specify the size interval to split up calculations.
For example, if <code>class = 2</code> Froh based on ROH 0-2, 2-4, 4-8, 80-16, &gt;16 Mbps long
will be calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Froh_inbreedingClass(runs, mapFile, Class = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Froh_inbreedingClass_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with ROH results</p>
</td></tr>
<tr><td><code id="Froh_inbreedingClass_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="Froh_inbreedingClass_+3A_class">Class</code></td>
<td>
<p>base ROH-length interval (in Mbps) (default: 0-2, 2-4, 4-8, 8-16, &gt;16)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with individual inbreeding coefficients based on ROH-length of
specific size. The sum of ROH-length of specific size in each individual is
reported alongside
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

Froh_inbreedingClass(runs = runsDF, mapFile = mapFile, Class = 2)

</code></pre>

<hr>
<h2 id='genoConvert'>Convert 0/1/2 genotypes to 0/1</h2><span id='topic+genoConvert'></span>

<h3>Description</h3>

<p>This is a utility function, that convert 0/1/2 genotypes (AA/AB/BB) into 0/1
(either homozygous/heterozygous)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genoConvert(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genoConvert_+3A_x">x</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>converted vector of genotypes (0/1)
</p>

<hr>
<h2 id='genoConvertCpp'>Convert 0/1/2 genotypes to 0/1</h2><span id='topic+genoConvertCpp'></span>

<h3>Description</h3>

<p>This is a utility function, that convert 0/1/2 genotypes (AA/AB/BB) into 0/1
(either homozygous/heterozygous)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genoConvertCpp(genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genoConvertCpp_+3A_genotype">genotype</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>converted vector of genotypes (0/1)
</p>

<hr>
<h2 id='heteroZygotTest'>Function to check whether a window is (loosely) heterozygous or not</h2><span id='topic+heteroZygotTest'></span>

<h3>Description</h3>

<p>This is a core function within the sliding-window workflow. Parameters on how to consider a window heterozygous are here (maxHom, maxMiss)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heteroZygotTest(x, gaps, maxHom, maxMiss, maxGap, i, windowSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heteroZygotTest_+3A_x">x</code></td>
<td>
<p>vector of 0/1 genotypes (from genoConvert())</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_maxhom">maxHom</code></td>
<td>
<p>max n. of homozygous SNP in a heterozygous window</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max n. of missing in a window</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_i">i</code></td>
<td>
<p>index along the genome (genome-vector for each individual)</p>
</td></tr>
<tr><td><code id="heteroZygotTest_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list: i) TRUE/FALSE (whether a window is heterozygous or NOT); ii) indexes of &quot;opposite and missing&quot; genotype
</p>

<hr>
<h2 id='heteroZygotTestCpp'>Function to check whether a window is (loosely) heterozygous or not</h2><span id='topic+heteroZygotTestCpp'></span>

<h3>Description</h3>

<p>This is a core function. Parameters on how to consider a window heterozygous are here (maxHom, maxMiss)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heteroZygotTestCpp(x, gaps, maxHom, maxMiss, maxGap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="heteroZygotTestCpp_+3A_x">x</code></td>
<td>
<p>vector of 0/1 genotypes (from genoConvert())</p>
</td></tr>
<tr><td><code id="heteroZygotTestCpp_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="heteroZygotTestCpp_+3A_maxhom">maxHom</code></td>
<td>
<p>max n. of homozygous SNP in a heterozygous window</p>
</td></tr>
<tr><td><code id="heteroZygotTestCpp_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max n. of missing in a window</p>
</td></tr>
<tr><td><code id="heteroZygotTestCpp_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE (whether a window is heterozygous or NOT)
</p>

<hr>
<h2 id='homoZygotTest'>Function to check whether a window is (loosely) homozygous or not</h2><span id='topic+homoZygotTest'></span>

<h3>Description</h3>

<p>This is a core function. Parameters on how to consider a window homozygous are here (maxHet, maxMiss)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homoZygotTest(x, gaps, maxHet, maxMiss, maxGap, i, windowSize)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="homoZygotTest_+3A_x">x</code></td>
<td>
<p>vector of 0/1 genotypes (from genoConvert())</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_maxhet">maxHet</code></td>
<td>
<p>max n. of heterozygous SNP in a homozygous window</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max n. of missing in a window</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_i">i</code></td>
<td>
<p>index along the genome (genome-vector for each individual)</p>
</td></tr>
<tr><td><code id="homoZygotTest_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list: i) TRUE/FALSE (whether a window is heterozygous or NOT); ii) indexes of &quot;opposite and missing&quot; genotype
</p>

<hr>
<h2 id='homoZygotTestCpp'>Function to check whether a window is (loosely) homozygous or not</h2><span id='topic+homoZygotTestCpp'></span>

<h3>Description</h3>

<p>This is a core function. Parameters on how to consider a window homozygous are here (maxHet, maxMiss)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homoZygotTestCpp(x, gaps, maxHet, maxMiss, maxGap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="homoZygotTestCpp_+3A_x">x</code></td>
<td>
<p>vector of 0/1 genotypes (from genoConvert())</p>
</td></tr>
<tr><td><code id="homoZygotTestCpp_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="homoZygotTestCpp_+3A_maxhet">maxHet</code></td>
<td>
<p>max n. of heterozygous SNP in a homozygous window</p>
</td></tr>
<tr><td><code id="homoZygotTestCpp_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max n. of missing in a window</p>
</td></tr>
<tr><td><code id="homoZygotTestCpp_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE (whether a window is homozygous or NOT)
</p>

<hr>
<h2 id='pedConvertCpp'>Convert ped genotypes to 0/1</h2><span id='topic+pedConvertCpp'></span>

<h3>Description</h3>

<p>This is a utility function, that convert ped genotypes (AA/AB/BB) into 0/1
(either homozygous/heterozygous)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedConvertCpp(genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pedConvertCpp_+3A_genotype">genotype</code></td>
<td>
<p>vector of pair of genotypes (01, AA, AG)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>converted vector of genotypes (0/1)
</p>

<hr>
<h2 id='plot_DistributionRuns'>Plot Distribution of runs</h2><span id='topic+plot_DistributionRuns'></span>

<h3>Description</h3>

<p>This function the distribution of runs per group. The average run length per size-class,
the average run length per chromosome (and group), the percent distribution of runs
per size-class and group, and the proportion of runs per chromosome are plotted.
With <code>style="All"</code> all three plots are produced.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_DistributionRuns(runs, mapFile, groupSplit = TRUE,
  style = c("MeanClass", "MeanChr", "RunsPCT", "RunsPCT_Chr", "All"),
  savePlots = FALSE, outputName = NULL, plotTitle = NULL,
  Class = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_DistributionRuns_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_groupsplit">groupSplit</code></td>
<td>
<p>plots split by group (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_style">style</code></td>
<td>
<p>type of plot: MeanClass, MeanChr, RunsPCT, RunsPCT_Chr, All (all plots)</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out to files or plotted in the graphical terminal (default)?</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)#'</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_plottitle">plotTitle</code></td>
<td>
<p>title in plot (default NULL)</p>
</td></tr>
<tr><td><code id="plot_DistributionRuns_+3A_class">Class</code></td>
<td>
<p>group of length (in Mbps) by class (default: 0-2, 2-4, 4-8, 8-16, &gt;16)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot Distribution Runs
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

plot_InbreedingChr(runs = runsDF, mapFile = mapFile, style='All')

</code></pre>

<hr>
<h2 id='plot_InbreedingChr'>Plot Froh-based inbreeding coefficients by group</h2><span id='topic+plot_InbreedingChr'></span>

<h3>Description</h3>

<p>The function plots the distribution of inbreeding/consanguinity coefficients
per chromosome and/or group. Three types of plots can be produces: barplots, boxplots,
violin plots. With <code>style="All"</code> all three plots are produced.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_InbreedingChr(runs, mapFile, groupSplit = TRUE,
  style = c("ChrBarPlot", "ChrBoxPlot", "FrohBoxPlot", "All"),
  outputName = NULL, plotTitle = NULL, savePlots = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_InbreedingChr_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_groupsplit">groupSplit</code></td>
<td>
<p>plots split by group (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_style">style</code></td>
<td>
<p>type of plot: ChrBarPlot, ChrBoxPlot, FrohBoxPlot, All (all plots)</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_plottitle">plotTitle</code></td>
<td>
<p>title in plot (default NULL)</p>
</td></tr>
<tr><td><code id="plot_InbreedingChr_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out to files or plotted in the graphical terminal (default)?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots of the distribution of inbreeding by chromosome and group
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

plot_InbreedingChr(runs = runsDF, mapFile = mapFile, style='All')

</code></pre>

<hr>
<h2 id='plot_manhattanRuns'>Plot the proportion of times SNPs are inside runs - MANHATTAN PLOT</h2><span id='topic+plot_manhattanRuns'></span>

<h3>Description</h3>

<p>Function to plot the proportion of times/percentage each SNP in inside a run
(population-specific signals) against SNP position in all chromosomes together
Proportions on the y-axis, bps on the x-axis for all analysed chromosomes
This is similar to the familiar GWAS Manhattan plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_manhattanRuns(runs, genotypeFile, mapFile, savePlots = FALSE,
  outputName = NULL, plotTitle = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_manhattanRuns_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_manhattanRuns_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file path</p>
</td></tr>
<tr><td><code id="plot_manhattanRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td></tr>
<tr><td><code id="plot_manhattanRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out in files (default) or plotted in the graphical terminal?</p>
</td></tr>
<tr><td><code id="plot_manhattanRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
<tr><td><code id="plot_manhattanRuns_+3A_plottitle">plotTitle</code></td>
<td>
<p>title in plot (default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Manhattan plots of proportion of times SNPs are inside runs,
per population (pdf files)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

# plot runs per animal (interactive)
plot_manhattanRuns(runs = runsDF, genotypeFile = genotypeFile, mapFile = mapFile,
savePlots = FALSE, plotTitle = "ROHom")

</code></pre>

<hr>
<h2 id='plot_PatternRuns'>Plot sum of run-lengths (or average run-lengths) against the number of runs per individual</h2><span id='topic+plot_PatternRuns'></span>

<h3>Description</h3>

<p>Function to plot the sum of run lengths (or the average run length) per individual
against the average number of runs per individual. Points can be differentially
coloured by group/population. This plot can be useful to identify patterns in
the distribution of runs in different groups (e.g. few long runs vs many short runs)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_PatternRuns(runs, mapFile, method = c("sum", "mean"),
  outputName = NULL, savePlots = FALSE, plotTitle = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_PatternRuns_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_PatternRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td></tr>
<tr><td><code id="plot_PatternRuns_+3A_method">method</code></td>
<td>
<p>&quot;sum&quot; or &quot;mean&quot; of run lengths per individual sample</p>
</td></tr>
<tr><td><code id="plot_PatternRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)#'</p>
</td></tr>
<tr><td><code id="plot_PatternRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out to files or plotted in the graphical terminal (default)?</p>
</td></tr>
<tr><td><code id="plot_PatternRuns_+3A_plottitle">plotTitle</code></td>
<td>
<p>title in plot (default NULL)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot of number of runs vs run-length sum/mean per individual sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

plot_PatternRuns(runs = runsDF, mapFile = mapFile, method = 'sum')
plot_PatternRuns(runs = runsDF, mapFile = mapFile, method = 'mean')

</code></pre>

<hr>
<h2 id='plot_Runs'>Function to plot runs per individual</h2><span id='topic+plot_Runs'></span>

<h3>Description</h3>

<p>Function to plot runs per individual (see Williams et al. 2016, Animal Genetics,
for an example with animal data)
Individual IDs on the y-axis, bps on the x-axis (position along the chromosome)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Runs(runs, suppressInds = FALSE, savePlots = FALSE,
  separatePlots = FALSE, outputName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Runs_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_Runs_+3A_suppressinds">suppressInds</code></td>
<td>
<p>shall we suppress individual IDs on the y-axis? (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="plot_Runs_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out to files (one pdf file for all chromosomes)
or plotted in the graphical terminal (default)?</p>
</td></tr>
<tr><td><code id="plot_Runs_+3A_separateplots">separatePlots</code></td>
<td>
<p>should plots for each chromosome be saved out to separate files?</p>
</td></tr>
<tr><td><code id="plot_Runs_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot of runs by chromosome
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

# plot runs per animal (interactive)
plot_Runs(runs = runsDF, suppressInds = FALSE, savePlots = FALSE, outputName = "ROHom")

</code></pre>

<hr>
<h2 id='plot_SnpsInRuns'>Plot the number of times each SNP falls inside runs</h2><span id='topic+plot_SnpsInRuns'></span>

<h3>Description</h3>

<p>Function to plot the number of times/percentage each SNP is inside a run
(population-specific signals) against the SNP positions in the genome.
Proportions on the y-axis, bps on the x-axis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SnpsInRuns(runs, genotypeFile, mapFile, savePlots = FALSE,
  separatePlots = FALSE, outputName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SnpsInRuns_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_SnpsInRuns_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file path</p>
</td></tr>
<tr><td><code id="plot_SnpsInRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td></tr>
<tr><td><code id="plot_SnpsInRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out in files (default) or plotted in
the graphical terminal?</p>
</td></tr>
<tr><td><code id="plot_SnpsInRuns_+3A_separateplots">separatePlots</code></td>
<td>
<p>should plots for each chromosome be saved out to separate files?</p>
</td></tr>
<tr><td><code id="plot_SnpsInRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot number of times a SNP is in a run by chromosome and population (pdf files)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
# skipping runs calculation
## Not run: 
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

# plot runs per animal (interactive)
plot_SnpsInRuns(runs = runsDF, genotypeFile = genotypeFile, mapFile = mapFile,
savePlots = FALSE, outputName = "ROHom")

</code></pre>

<hr>
<h2 id='plot_StackedRuns'>Plot stacked runs</h2><span id='topic+plot_StackedRuns'></span>

<h3>Description</h3>

<p>Function to plot stacked runs along the chromosome (signaling presence of large numbers of runs
in specific regions of a chromosome)
Counts on the y-axis, bps on the x-axis (position along the chromosome)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_StackedRuns(runs, savePlots = FALSE, separatePlots = FALSE,
  outputName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_StackedRuns_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_StackedRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out in files (default) or plotted in
the graphical terminal?</p>
</td></tr>
<tr><td><code id="plot_StackedRuns_+3A_separateplots">separatePlots</code></td>
<td>
<p>should plots for chromosomes be saved out to separate files?</p>
</td></tr>
<tr><td><code id="plot_StackedRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot of stacked runs by population and by chromosome (pdf files)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

# plot runs per animal (interactive)
plot_StackedRuns(runs = runsDF, savePlots = FALSE, outputName = "ROHom")

</code></pre>

<hr>
<h2 id='plot_ViolinRuns'>Violin plot of run length per individual (either sum or mean)</h2><span id='topic+plot_ViolinRuns'></span>

<h3>Description</h3>

<p>Function to produce violin plots of the distribution of runs lengths per group
The sum of run lengths, or its average, per individual sample is used to
characterize the distribution of runs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ViolinRuns(runs, method = c("sum", "mean"), outputName = NULL,
  plotTitle = NULL, savePlots = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ViolinRuns_+3A_runs">runs</code></td>
<td>
<p>a data.frame with runs per individual (group, id, chrom, nSNP, start, end, length)</p>
</td></tr>
<tr><td><code id="plot_ViolinRuns_+3A_method">method</code></td>
<td>
<p>&quot;sum&quot; or &quot;mean&quot; of run lengths per individual samples</p>
</td></tr>
<tr><td><code id="plot_ViolinRuns_+3A_outputname">outputName</code></td>
<td>
<p>title prefix (the base name of graph, if savePlots is TRUE)</p>
</td></tr>
<tr><td><code id="plot_ViolinRuns_+3A_plottitle">plotTitle</code></td>
<td>
<p>title in plot (default NULL)</p>
</td></tr>
<tr><td><code id="plot_ViolinRuns_+3A_saveplots">savePlots</code></td>
<td>
<p>should plots be saved out to files or plotted in the graphical terminal (default)?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Violin plot of the distribution of runs-lengths (sum or mean)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

plot_ViolinRuns(runs = runsDF, method = "sum" , savePlots = FALSE)
plot_ViolinRuns(runs = runsDF, method = "mean" , savePlots = FALSE)

</code></pre>

<hr>
<h2 id='readExternalRuns'>Read runs from external files</h2><span id='topic+readExternalRuns'></span>

<h3>Description</h3>

<p>Function to read in, from external files, the output of software for ROH:
</p>

<ol>
<li> <p><code>detectRUNS</code>: output saved out to a file (e.g. write.table)
</p>
</li>
<li> <p><code>Plink</code>: output from the <code>--homozyg</code> option (<code>.hom</code> files)
</p>
</li>
<li> <p><code>BCFtools</code>: output from the <code>roh</code> option
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>readExternalRuns(inputFile = NULL, program = c("plink", "BCFtools",
  "detectRUNS"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readExternalRuns_+3A_inputfile">inputFile</code></td>
<td>
<p>name of (path to) external file</p>
</td></tr>
<tr><td><code id="readExternalRuns_+3A_program">program</code></td>
<td>
<p>source program that produced the ROH file (one of <code>detectRUNS</code>,
<code>Plink</code>, <code>BCFtools</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe in the correct format to be used with plots and statistics functions from <code>detectRUNS</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
## Not run: 
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxMissRun = 1, maxMissWindow = 1,  minLengthBps = 100000,  minDensity = 1/10000)

write.table(x= runs,file = 'RunsFileTest.txt', quote=F, row.names = F)
newData=readRunsFromFile(runsFile = 'RunsFileTest.txt', program = 'detectRUNS')

## End(Not run)

</code></pre>

<hr>
<h2 id='readPOPCpp'>Function to return a dataframe of population (POP, ID)</h2><span id='topic+readPOPCpp'></span>

<h3>Description</h3>

<p>This is a core function. Read PED file and returns a data.frame with the first two
columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readPOPCpp(genotypeFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readPOPCpp_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of POP, ID
</p>

<hr>
<h2 id='reorderDF'>Function to reorder data frames by CHROMOSOME</h2><span id='topic+reorderDF'></span>

<h3>Description</h3>

<p>The data frame will be reordered according to chromosome:
from 1 to n, then X, Y, XY, MT
The data frame needs to have a column with name &quot;CHROMOSOME&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorderDF(dfx)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reorderDF_+3A_dfx">dfx</code></td>
<td>
<p>data frame to be reordered (with column &quot;CHROMOSOME&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reorder results based on chromosome
</p>


<h3>Value</h3>

<p>A reordered data frame by chromosome
</p>

<hr>
<h2 id='slidingRuns'>Function to detect runs using sliding window approach</h2><span id='topic+slidingRuns'></span>

<h3>Description</h3>

<p>This is a core function not intended to be exported
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slidingRuns(indGeno, individual, mapFile, gaps, parameters, cpp = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slidingRuns_+3A_indgeno">indGeno</code></td>
<td>
<p>vector of 0/1/NAs of individual genotypes (0: homozygote; 1: heterozygote)</p>
</td></tr>
<tr><td><code id="slidingRuns_+3A_individual">individual</code></td>
<td>
<p>list of group (breed, population, case/control etc.) and ID of individual sample</p>
</td></tr>
<tr><td><code id="slidingRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="slidingRuns_+3A_gaps">gaps</code></td>
<td>
<p>distance between SNPs</p>
</td></tr>
<tr><td><code id="slidingRuns_+3A_parameters">parameters</code></td>
<td>
<p>list of parameters</p>
</td></tr>
<tr><td><code id="slidingRuns_+3A_cpp">cpp</code></td>
<td>
<p>use cpp functions or not (DEBUG)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method uses sliding windows to detect RUNs. Checks on minimum n. of SNP, max n. of opposite and missing genotypes,
max gap between adjacent loci and minimum length of the run are implemented (as in the sliding window method).
Both runs of homozygosity (RoHom) and of heterozygosity (RoHet) can be search for (option ROHet: TRUE/FALSE)
NOTE: this methods is intended to not be exported
</p>


<h3>Value</h3>

<p>A data frame of runs per individual sample
</p>

<hr>
<h2 id='slidingRUNS.run'>Main function to detect RUNS (ROHom/ROHet) using sliding windows (a la Plink)</h2><span id='topic+slidingRUNS.run'></span>

<h3>Description</h3>

<p>This is one of the main function of detectRUNS and is used to detect runs
(of homozygosity or heterozygosity)
in the genome (diploid) with the sliding-window method.
All parameters to detect runs (e.g. minimum n. of SNP, max n. of missing genotypes,
max n. of opposite genotypes etc.) are specified here.
Input data are in the ped/map
Plink format (https://www.cog-genomics.org/plink/1.9/input#ped)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slidingRUNS.run(genotypeFile, mapFile, windowSize = 15,
  threshold = 0.05, minSNP = 3, ROHet = FALSE, maxOppWindow = 1,
  maxMissWindow = 1, maxGap = 10^6, minLengthBps = 1000,
  minDensity = 1/1000, maxOppRun = NULL, maxMissRun = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slidingRUNS.run_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file path</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_mapfile">mapFile</code></td>
<td>
<p>map file (.map) file path</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_windowsize">windowSize</code></td>
<td>
<p>the size of sliding window (number of SNP loci) (default = 15)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_threshold">threshold</code></td>
<td>
<p>the threshold of overlapping windows of the same state
(homozygous/heterozygous) to call a SNP in a RUN (default = 0.05)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_minsnp">minSNP</code></td>
<td>
<p>minimum n. of SNP in a RUN (default = 3)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_rohet">ROHet</code></td>
<td>
<p>should we look for ROHet or ROHom? (default = FALSE)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_maxoppwindow">maxOppWindow</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP in the
sliding window (default = 1)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_maxmisswindow">maxMissWindow</code></td>
<td>
<p>max. n. of missing SNP in the sliding window (default = 1)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP to be still considered a
potential run (default = 10^6 bps)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_minlengthbps">minLengthBps</code></td>
<td>
<p>minimum length of run in bps (defaults to 1000 bps = 1 kbps)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_mindensity">minDensity</code></td>
<td>
<p>minimum n. of SNP per kbps (defaults to 0.1 = 1 SNP every 10 kbps)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_maxopprun">maxOppRun</code></td>
<td>
<p>max n. of opposite genotype SNPs in the run (optional)</p>
</td></tr>
<tr><td><code id="slidingRUNS.run_+3A_maxmissrun">maxMissRun</code></td>
<td>
<p>max n. of missing SNPs in the run (optional)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function scans the genome (diploid) for runs using the sliding-window method.
This is a wrapper function for many component functions that handle the input data
(ped/map files), perform internal conversions, accept parameters specifications,
select whether runs of homozygosity (RoHom) or of heterozygosity (RoHet)
are looked for.
</p>
<p>In the ped file, the groups samples belong to can be specified (first column).
This is important if comparisons between human ethnic groups or between animal breeds
or plant varieties or biological populations are to be performed.
Also, if cases and controls are to be compared, this is the place where this
information needs to be specified.
</p>
<p>This function returns a data frame with all runs detected in the dataset.
This data frame can then be written out to a csv file.
The data frame is, in turn, the input for other functions of the detectRUNS package
that create plots and produce statistics from the results
(see plots and statistics functions in this manual,
and/or refer to the detectRUNS vignette).
</p>


<h3>Value</h3>

<p>A dataframe with RUNs of Homozygosity or Heterozygosity in the analysed dataset.
The returned dataframe contains the following seven columns: &quot;group&quot;, &quot;id&quot;, &quot;chrom&quot;,
&quot;nSNP&quot;, &quot;from&quot;, &quot;to&quot;, &quot;lengthBps&quot; (group: population, breed, case/control etc.;
id: individual identifier; chrom: chromosome on which the run is located;
nSNP: number of SNPs in the run; from: starting position of the run, in bps;
to: end position of the run, in bps; lengthBps: size of the run)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")
# calculating runs with sliding window approach
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,
minSNP = 15, ROHet = FALSE,  maxOppWindow = 1, maxMissWindow = 1, maxGap=10^6,
minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
colClasses &lt;- c(rep("character", 3), rep("numeric", 4)  )
runs &lt;- read.csv2(runsFile, header = TRUE, stringsAsFactors = FALSE,
colClasses = colClasses)

</code></pre>

<hr>
<h2 id='slidingWindow'>Function to slide a window over a vector (individual's genotypes)</h2><span id='topic+slidingWindow'></span>

<h3>Description</h3>

<p>This is a core function. The functions to detect RUNS are slid over the genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slidingWindow(data, gaps, windowSize, step, maxGap, ROHet = TRUE,
  maxOppositeGenotype = 1, maxMiss = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slidingWindow_+3A_data">data</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_step">step</code></td>
<td>
<p>by which (how many SNP) is the window slid</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_rohet">ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_maxoppositegenotype">maxOppositeGenotype</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP</p>
</td></tr>
<tr><td><code id="slidingWindow_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max. n. of missing SNP</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of TRUE/FALSE (whether a window is homozygous or NOT)
</p>

<hr>
<h2 id='slidingWindowCpp'>Function to slide a window over a vector (individual's genotypes)</h2><span id='topic+slidingWindowCpp'></span>

<h3>Description</h3>

<p>This is a core function. The functions to detect RUNS are slid over the genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slidingWindowCpp(data, gaps, windowSize, step, maxGap, ROHet = TRUE,
  maxOppositeGenotype = 1L, maxMiss = 1L)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slidingWindowCpp_+3A_data">data</code></td>
<td>
<p>vector of 0/1/2 genotypes</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_gaps">gaps</code></td>
<td>
<p>vector of differences between consecutive positions (gaps) in bps</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_step">step</code></td>
<td>
<p>by which (how many SNP) is the window slid</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_maxgap">maxGap</code></td>
<td>
<p>max distance between consecutive SNP in a window to be still considered a potential run</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_rohet">ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_maxoppositegenotype">maxOppositeGenotype</code></td>
<td>
<p>max n. of homozygous/heterozygous SNP</p>
</td></tr>
<tr><td><code id="slidingWindowCpp_+3A_maxmiss">maxMiss</code></td>
<td>
<p>max. n. of missing SNP</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of TRUE/FALSE (whether a window is homozygous or NOT)
</p>

<hr>
<h2 id='snpInRun'>Function to return a vector of T/F for whether a SNP is or not in a RUN</h2><span id='topic+snpInRun'></span>

<h3>Description</h3>

<p>This is a core function. The function to determine whether a SNP is or not in a RUN.
The ratio between homozygous/heterozygous windows and total n. of windows is computed here
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpInRun(RunVector, windowSize, threshold)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpInRun_+3A_runvector">RunVector</code></td>
<td>
<p>vector of TRUE/FALSE (is a window homozygous/heterozygous?)</p>
</td></tr>
<tr><td><code id="snpInRun_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
<tr><td><code id="snpInRun_+3A_threshold">threshold</code></td>
<td>
<p>threshold to call a SNP in a RUN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of TRUE/FALSE (whether a SNP is in a RUN or NOT)
</p>

<hr>
<h2 id='snpInRunCpp'>Function to return a vector of T/F for whether a SNP is or not in a RUN</h2><span id='topic+snpInRunCpp'></span>

<h3>Description</h3>

<p>This is a core function. The function to determine whether a SNP is or not in a RUN.
The ratio between homozygous/heterozygous windows and total n. of windows is computed here
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpInRunCpp(RunVector, windowSize, threshold)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpInRunCpp_+3A_runvector">RunVector</code></td>
<td>
<p>vector of TRUE/FALSE (is a window homozygous/heterozygous?)</p>
</td></tr>
<tr><td><code id="snpInRunCpp_+3A_windowsize">windowSize</code></td>
<td>
<p>size of window (n. of SNP)</p>
</td></tr>
<tr><td><code id="snpInRunCpp_+3A_threshold">threshold</code></td>
<td>
<p>threshold to call a SNP in a RUN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of TRUE/FALSE (whether a SNP is in a RUN or NOT)
</p>

<hr>
<h2 id='snpInsideRuns'>Function to count number of times a SNP is in a RUN</h2><span id='topic+snpInsideRuns'></span>

<h3>Description</h3>

<p>Function to count number of times a SNP is in a RUN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpInsideRuns(runsChrom, mapChrom, genotypeFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpInsideRuns_+3A_runschrom">runsChrom</code></td>
<td>
<p>R object (dataframe) with results per chromosome (column names:&quot;POPULATION&quot;,&quot;IND&quot;,&quot;CHROMOSOME&quot;,&quot;COUNT&quot;,&quot;START&quot;,&quot;END&quot;,&quot;LENGTH&quot;)</p>
</td></tr>
<tr><td><code id="snpInsideRuns_+3A_mapchrom">mapChrom</code></td>
<td>
<p>R object (dataframe) with SNP name and position per chromosome (map file) (column names: &quot;CHR&quot;,&quot;SNP_NAME&quot;,&quot;x&quot;,&quot;POSITION&quot;)</p>
</td></tr>
<tr><td><code id="snpInsideRuns_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with counts per SNP in runs (per population)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# defining mapChrom
mappa &lt;- data.table::fread(mapFile, header = FALSE)
names(mappa) &lt;- c("CHR","SNP_NAME","x","POSITION")
mappa$x &lt;- NULL
chrom &lt;- "24"
mapChrom &lt;- mappa[mappa$CHR==chrom, ]

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
colClasses &lt;- c(rep("character", 3), rep("numeric", 4)  )
runs &lt;- read.csv2(runsFile, header = TRUE, stringsAsFactors = FALSE,
colClasses = colClasses)

# fix column names and define runsChrom
names(runs) &lt;- c("POPULATION","IND","CHROMOSOME","COUNT","START","END","LENGTH")
runsChrom &lt;- runs[runs$CHROMOSOME==chrom, ]

snpInsideRuns(runsChrom, mapChrom, genotypeFile)

</code></pre>

<hr>
<h2 id='snpInsideRunsCpp'>Function to count number of times a SNP is in a RUN</h2><span id='topic+snpInsideRunsCpp'></span>

<h3>Description</h3>

<p>Function to count number of times a SNP is in a RUN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpInsideRunsCpp(runsChrom, mapChrom, genotypeFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snpInsideRunsCpp_+3A_runschrom">runsChrom</code></td>
<td>
<p>R object (dataframe) with results per chromosome</p>
</td></tr>
<tr><td><code id="snpInsideRunsCpp_+3A_mapchrom">mapChrom</code></td>
<td>
<p>R map object with SNP per chromosome</p>
</td></tr>
<tr><td><code id="snpInsideRunsCpp_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>genotype (.ped) file location</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with counts per SNP in runs (per population)
</p>

<hr>
<h2 id='summaryRuns'>Summary statistics on detected runs</h2><span id='topic+summaryRuns'></span>

<h3>Description</h3>

<p>This function processes the results from <code>slidingRUNS.run</code> and
<code>consecutiveRUNS.run</code> and produces a number of interesting descriptives
statistics on results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryRuns(runs, mapFile, genotypeFile, Class = 2, snpInRuns = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summaryRuns_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td></tr>
<tr><td><code id="summaryRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="summaryRuns_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>Plink ped file (for SNP position)</p>
</td></tr>
<tr><td><code id="summaryRuns_+3A_class">Class</code></td>
<td>
<p>group of length (in Mbps) by class (default: 0-2, 2-4, 4-8, 8-16, &gt;16)</p>
</td></tr>
<tr><td><code id="summaryRuns_+3A_snpinruns">snpInRuns</code></td>
<td>
<p>TRUE/FALSE (default): should the function <code>snpInsideRuns</code> be
called to compute the proportion of times each SNP falls inside a run in the
group/population?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>summaryRuns</code> calculates: i) the number of runs per chromosome and group/population;
ii) the percent distribution of runs per chromosome and group; iii) the number of
runs per size-class and group; iv) the percent distribution of runs per size-class
and group; v) the mean length of runs per chromosome and group; vi) the mean
length of runs per size-class and group; vii) individual inbreeding coefficient
estimated from ROH; viii) individual inbreeding coefficient estimated from ROH
per chromosome; ix) individual inbreeding coefficient estimated from ROH per
size-class
</p>


<h3>Value</h3>

<p>A list of dataframes containing the most relevant descriptives
statistics on detected runs. The list conveniently contains 9 dataframes that can
be used for further processing and visualization, or can be written out to text files
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF &lt;- readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

summaryRuns(runs = runsDF, mapFile = mapFile, genotypeFile = genotypeFile, Class = 2,
snpInRuns = FALSE)

</code></pre>

<hr>
<h2 id='tableRuns'>Function to retrieve most common runs in the population</h2><span id='topic+tableRuns'></span>

<h3>Description</h3>

<p>This function takes in input either the run results or the output from
the function <code>snpInsideRuns</code> (proportion of times a SNP is inside a run)
in the population/group, and returns a subset of the runs most commonly
found in the group/population. The parameter <code>threshold</code> controls the definition
of most common (e.g. in at least 50%, 70% etc. of the sampled individuals)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tableRuns(runs = NULL, SnpInRuns = NULL, genotypeFile, mapFile,
  threshold = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tableRuns_+3A_runs">runs</code></td>
<td>
<p>R object (dataframe) with results on detected runs</p>
</td></tr>
<tr><td><code id="tableRuns_+3A_snpinruns">SnpInRuns</code></td>
<td>
<p>dataframe with the proportion of times each SNP falls inside a
run in the population (output from <code>snpInsideRuns</code>)</p>
</td></tr>
<tr><td><code id="tableRuns_+3A_genotypefile">genotypeFile</code></td>
<td>
<p>Plink ped file (for SNP position)</p>
</td></tr>
<tr><td><code id="tableRuns_+3A_mapfile">mapFile</code></td>
<td>
<p>Plink map file (for SNP position)</p>
</td></tr>
<tr><td><code id="tableRuns_+3A_threshold">threshold</code></td>
<td>
<p>value from 0 to 1 (default 0.7) that controls the desired
proportion of individuals carrying that run (e.g. 70%)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with the most common runs detected in the sampled individuals
(the group/population, start and end position of the run, chromosome and number of SNP
included in the run are reported in the output dataframe)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># getting map and ped paths
genotypeFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.ped", package = "detectRUNS")
mapFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.map", package = "detectRUNS")

# calculating runs of Homozygosity
## Not run: 
# skipping runs calculation
runs &lt;- slidingRUNS.run(genotypeFile, mapFile, windowSize = 15, threshold = 0.1,  minSNP = 15,
ROHet = FALSE,  maxOppositeGenotype = 1, maxMiss = 1,  minLengthBps = 100000,  minDensity = 1/10000)

## End(Not run)
# loading pre-calculated data
runsFile &lt;- system.file("extdata", "Kijas2016_Sheep_subset.sliding.csv", package="detectRUNS")
runsDF = readExternalRuns(inputFile = runsFile, program = 'detectRUNS')

tableRuns(runs = runsDF, genotypeFile = genotypeFile, mapFile = mapFile, threshold = 0.5)

</code></pre>

<hr>
<h2 id='writeRUN'>Function to write out RUNS per individual animal</h2><span id='topic+writeRUN'></span>

<h3>Description</h3>

<p>Function to write out RUNS per individual animal
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeRUN(ind, dRUN, ROHet = TRUE, group, outputName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeRUN_+3A_ind">ind</code></td>
<td>
<p>ID of animals</p>
</td></tr>
<tr><td><code id="writeRUN_+3A_drun">dRUN</code></td>
<td>
<p>data.frame with RUNS per animal</p>
</td></tr>
<tr><td><code id="writeRUN_+3A_rohet">ROHet</code></td>
<td>
<p>shall we detect ROHet or ROHom?</p>
</td></tr>
<tr><td><code id="writeRUN_+3A_group">group</code></td>
<td>
<p>group (factor): population, breed, ethnicity, case/control etc.</p>
</td></tr>
<tr><td><code id="writeRUN_+3A_outputname">outputName</code></td>
<td>
<p>output filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE if RUNS are written out or not
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
