<!DOCTYPE html><html><head><title>Help for package deTS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {deTS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#deTS-package'><p>Tissue-Specific Enrichment Analysis</p>
Tissue-Specific Enrichment Analysis</a></li>
<li><a href='#correction_factor'><p>Gene average expression level and standard deviation in GTEx data</p></a></li>
<li><a href='#ENCODE_z_score'><p>ENCODE z-score to define tissue-specific genes</p></a></li>
<li><a href='#GTEx_t_score'><p>GTEx t-score to define tissue-specific genes</p></a></li>
<li><a href='#GWAS_gene'><p>Gene symbol query data for single sample</p></a></li>
<li><a href='#GWAS_gene_multiple'><p>Gene symbol query data for multiple samples</p></a></li>
<li><a href='#query_ENCODE'><p>ENCODE raw query data</p></a></li>
<li><a href='#query_GTEx'><p>GTEx raw query data</p></a></li>
<li><a href='#tsea.analysis'><p>Tissue-specific enrichment analysis for query gene list</p></a></li>
<li><a href='#tsea.analysis.multiple'><p>Tissue-specific enrichment analysis for multi query gene lists</p></a></li>
<li><a href='#tsea.expression.decode'><p>Tissue-specific enrichment analysis for RNA-Seq expression profiles</p></a></li>
<li><a href='#tsea.expression.normalization'><p>RNA-Seq expression profiles normalization</p></a></li>
<li><a href='#tsea.plot'><p>Tissue-specific enrichment analysis result heatmap plot</p></a></li>
<li><a href='#tsea.summary'><p>Tissue-specific enrichment analysis result summary</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tissue-Specific Enrichment Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-02-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Guangsheng Pei</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guangsheng Pei &lt;peiguangsheng@gmail.com&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>pheatmap, RColorBrewer</td>
</tr>
<tr>
<td>Description:</td>
<td>Tissue-specific enrichment analysis to assess lists of candidate genes or RNA-Seq expression profiles. Pei G., Dai Y., Zhao Z. Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-12 16:32:49 UTC; gpei</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-22 13:30:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='deTS-package'>Tissue-Specific Enrichment Analysis
Tissue-Specific Enrichment Analysis
</h2><span id='topic+deTS-package'></span><span id='topic+deTS'></span>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis to assess lists of candidate genes and tissue-specific expression decode analysis for RNA-seq data to decode RNA expression matrices tissue heterogeneity.
</p>


<h3>Details</h3>

<p>Since disease and physiological condition are often associated with a specific tissue, understanding the tissue-specific genes (TSG) expression patterns will substantially reduce false discoveries in biomedical research. However, due to cell complexity in human system, heterogeneous tissues are frequently collected. Making it difficult to distinguish gene expression variability and mislead result interpretation. Here, we present deTS, an R package that conducts Tissue-Specific Enrichment Analysis (TSEA) using two built-in reference panels: the Genotype-Tissue Expression (GTEx) data and the ENCyclopedia Of DNA Elements (ENCODE) data. We implemented two major functions in TSEA to assess lists of candidate genes or expression matrices.
</p>
<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> deTS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Tissue-Specific Enrichment Analysis</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-02-06</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Guangsheng Pei</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Guangsheng Pei &lt;peiguangsheng@gmail.com&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> pheatmap, RColorBrewer</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Tissue-specific enrichment analysis to assess lists of candidate genes or RNA-Seq expression profiles. Pei G., Dai Y., Zhao Z. Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Index of help topics:
</p>
<pre>
ENCODE_z_score          ENCODE z-score to define tissue-specific genes
GTEx_t_score            GTEx t-score to define tissue-specific genes
GWAS_gene               Gene symbol query data for single sample
GWAS_gene_multiple      Gene symbol query data for multiple samples
correction_factor       Gene average expression level and standard
                        deviation in GTEx data
deTS-package            Tissue-Specific Enrichment Analysis
                        Tissue-Specific Enrichment Analysis
query_ENCODE            ENCODE raw query data
query_GTEx              GTEx raw query data
tsea.analysis           Tissue-specific enrichment analysis for query
                        gene list
tsea.analysis.multiple
                        Tissue-specific enrichment analysis for multi
                        query gene lists
tsea.expression.decode
                        Tissue-specific enrichment analysis for RNA-Seq
                        expression profiles
tsea.expression.normalization
                        RNA-Seq expression profiles normalization
tsea.plot               Tissue-specific enrichment analysis result
                        heatmap plot
tsea.summary            Tissue-specific enrichment analysis result
                        summary
</pre>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>
<p>Maintainer: Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GTEx_t_score)
data(ENCODE_z_score)
library(pheatmap)

data(GWAS_gene)
query_gene_list = GWAS_gene
tsea_t = tsea.analysis(query_gene_list, GTEx_t_score, 0.05,
	p.adjust.method = "bonferroni")
tsea_t_summary = tsea.summary(tsea_t)

data(GWAS_gene_multiple)
query_gene_list = GWAS_gene_multiple[,1:2]
tsea_t_multi = tsea.analysis.multiple(query_gene_list, 
	GTEx_t_score, 0.05, p.adjust.method = "BH")

data(query_GTEx)
query_matrix = query_GTEx[,1:2]
data(correction_factor)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")
tseaed_in_ENCODE_summary = tsea.summary(tseaed_in_ENCODE)
</code></pre>

<hr>
<h2 id='correction_factor'>Gene average expression level and standard deviation in GTEx data
</h2><span id='topic+correction_factor'></span>

<h3>Description</h3>

<p>Gene average expression level and standard deviation in GTEx data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("correction_factor")</code></pre>


<h3>Format</h3>

<p>A data frame with 14725 observations on the following 2 variables.
</p>

<dl>
<dt><code>avg.all</code></dt><dd><p>a factor with gene average expression level </p>
</dd>
<dt><code>sd.all</code></dt><dd><p>a factor with gene standard deviation of expression level </p>
</dd>
</dl>



<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) Tissue-Specific Enrichment Analysis deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(correction_factor)
</code></pre>

<hr>
<h2 id='ENCODE_z_score'>ENCODE z-score to define tissue-specific genes
</h2><span id='topic+ENCODE_z_score'></span>

<h3>Description</h3>

<p>ENCODE z-score matrix to define tissue-specific genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ENCODE_z_score")</code></pre>


<h3>Format</h3>

<p>A data frame with z-score of 14031 genes in 44 ENCODE tissues.
</p>
<p>Row is genes symbol and column is tissue names.
</p>
<p>Adrenal Gland Body of Pancreas Breast Epithelium Camera-type Eye Cerebellum
</p>
<p>C1orf112        -0.674           -0.440            -0.246           3.892      1.333
</p>
<p>FGR             -0.078           -0.345             0.159          -0.354     -0.407
</p>
<p>CFH             -0.093           -0.365            -0.134          -0.133     -0.160
</p>
<p>FUCA2            3.028            1.467             0.040           0.228     -0.601
</p>
<p>NFYA            -0.637           -0.872             0.053           2.364      0.619
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ENCODE_z_score)
</code></pre>

<hr>
<h2 id='GTEx_t_score'>GTEx t-score to define tissue-specific genes
</h2><span id='topic+GTEx_t_score'></span>

<h3>Description</h3>

<p>GTEx t-score matrix to define tissue-specific genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GTEx_t_score")</code></pre>


<h3>Format</h3>

<p>A data frame with t-score of 14725 genes in 47 GTEx tissues.
</p>
<p>Row is genes symbol and column is tissue names.
</p>
<p>Adipose - Subcutaneous Adipose - Visceral (Omentum) Adrenal Gland Artery - Aorta Artery - Coronary
</p>
<p>OR4F5                    -0.524                       -0.597         0.134         -1.109            -0.588
</p>
<p>SAMD11                   -9.921                       -1.734         3.633          3.595             0.017
</p>
<p>KLHL17                   -6.812                       -4.553        -3.084         -0.744             0.494
</p>
<p>PLEKHN1                  -7.785                       -6.882        -3.915         -6.570            -4.892
</p>
<p>C1orf170                 -7.113                       -6.257        -4.465         -5.897            -4.004
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GTEx_t_score)
</code></pre>

<hr>
<h2 id='GWAS_gene'>Gene symbol query data for single sample
</h2><span id='topic+GWAS_gene'></span>

<h3>Description</h3>

<p>An example of input gene symbol query data for single sample tissue-specific enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GWAS_gene")</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>&quot;A1BG&quot; &quot;A1BG-AS1&quot; &quot;A1CF&quot; &quot;A2M&quot; &quot;A2M-AS1&quot; &quot;A2ML1&quot; &quot;A2MP1&quot; &quot;A3GALT2&quot; &quot;A4GALT&quot; &quot;A4GNT&quot; &quot;AA06&quot; &quot;AAAS&quot; &quot;AACS&quot; &quot;AACSP1&quot; &quot;AADAC&quot; ...
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS_gene)
</code></pre>

<hr>
<h2 id='GWAS_gene_multiple'>Gene symbol query data for multiple samples 
</h2><span id='topic+GWAS_gene_multiple'></span>

<h3>Description</h3>

<p>An example of input gene symbol query data for multiple samples tissue-specific enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GWAS_gene_multiple")</code></pre>


<h3>Format</h3>

<p>A data frame with 22134 genes if associated with following 5 neuropsychiatric disorders GWAS traits.
</p>
<p>Row is genes symbol and column is sample names.
</p>
<p>ADHD ASD BD MDD SCZ
</p>
<p>A1BG       0   0  0   0   0
</p>
<p>A1BG-AS1   0   0  0   0   0
</p>
<p>A1CF       0   1  0   0   0
</p>
<p>A2M        0   0  0   0   0
</p>
<p>A2M-AS1    0   0  0   0   0
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS_gene_multiple)
</code></pre>

<hr>
<h2 id='query_ENCODE'>ENCODE raw query data
</h2><span id='topic+query_ENCODE'></span>

<h3>Description</h3>

<p>An example of RNA-Seq query data from ENCODE data for tissue-specific enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("query_ENCODE")</code></pre>


<h3>Format</h3>

<p>A data frame with average expression level of 18661 genes in 44 ENCODE tissues.
</p>
<p>Row is genes symbol and column is sample names.
</p>
<p>Adrenal Gland Body of Pancreas Breast Epithelium Camera-type Eye Cerebellum
</p>
<p>TSPAN6           11.64            5.390             11.04           24.65     13.238
</p>
<p>TNMD              0.01            0.147              2.24           12.43      0.090
</p>
<p>DPM1             18.82            9.812             14.21           24.02     10.505
</p>
<p>SCYL3             3.81            2.593              5.63           10.50      3.783
</p>
<p>C1orf112          1.64            2.308              2.86           14.61      7.345
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(query_ENCODE)
</code></pre>

<hr>
<h2 id='query_GTEx'>GTEx raw query data
</h2><span id='topic+query_GTEx'></span>

<h3>Description</h3>

<p>An example of RNA-Seq query data from GTEx data for tissue-specific enrichment analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("query_GTEx")</code></pre>


<h3>Format</h3>

<p>A data frame with average expression level of 18067 gene in 47 GTEx tissues.
</p>
<p>Row is genes symbol and column is sample names.
</p>
<p>Adipose - Subcutaneous Adipose - Visceral (Omentum) Adrenal Gland Artery - Aorta Artery - Coronary
</p>
<p>OR4F5                    0.0317                       0.0284        0.0469         0.0133            0.0225
</p>
<p>SAMD11                   0.4451                       2.3056        3.8928         3.5822            2.5632
</p>
<p>NOC2L                   21.9084                      21.0439       19.4613        19.4929           19.8367
</p>
<p>KLHL17                   4.1406                       4.4075        4.4227         5.0840            5.3749
</p>
<p>PLEKHN1                  0.4531                       0.3452        1.1795         0.3081            0.3722
</p>


<h3>Details</h3>

<p>nothing
</p>


<h3>Source</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(query_GTEx)
</code></pre>

<hr>
<h2 id='tsea.analysis'>Tissue-specific enrichment analysis for query gene list
</h2><span id='topic+tsea.analysis'></span>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis by Fisher's Exact Test for given gene list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.analysis(query_gene_list, score, ratio = 0.05,
p.adjust.method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.analysis_+3A_query_gene_list">query_gene_list</code></td>
<td>
<p>a gene symbol list object.
</p>
</td></tr>
<tr><td><code id="tsea.analysis_+3A_score">score</code></td>
<td>
<p>a gene tissue-specific score matrix, c(&quot;GTEx_t_score&quot; or &quot;ENCODE_z_score&quot;), can be loaded by data(GTEx) or data(ENCODE), the default value is recommended &quot;GTEx_t_score&quot;.
</p>
</td></tr>
<tr><td><code id="tsea.analysis_+3A_ratio">ratio</code></td>
<td>
<p>the threshold to define tissue-specific genes (with top t-score or z-score), the default value is 0.05.
</p>
</td></tr>
<tr><td><code id="tsea.analysis_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p.adjust.method, c(&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tissue-specific enrichment analysis by Fisher's Exact Test for given gene list.
</p>


<h3>Value</h3>

<p>A data frame with p-value of tissue-specific enrichment result.
</p>
<p>Rows stand for tissue names and columns stand for sample names.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei	
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS_gene)
data(GTEx_t_score)
query_gene_list = GWAS_gene
tsea_t = tsea.analysis(query_gene_list, GTEx_t_score, 0.05,
	p.adjust.method = "bonferroni")
</code></pre>

<hr>
<h2 id='tsea.analysis.multiple'>Tissue-specific enrichment analysis for multi query gene lists
</h2><span id='topic+tsea.analysis.multiple'></span>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis by Fisher's Exact Test for multiple gene list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.analysis.multiple(query_gene_list, score, ratio = 0.05, 
p.adjust.method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.analysis.multiple_+3A_query_gene_list">query_gene_list</code></td>
<td>
<p>a 0~1 gene~sample table object, row should be gene symbol, column should be sample name. In the table, gene labeled with 1 indicated it is target gene for a given sample, while 0 indicated it is not target in a given sample.
</p>
</td></tr>
<tr><td><code id="tsea.analysis.multiple_+3A_score">score</code></td>
<td>
<p>a gene tissue-specific score matrix, c(&quot;GTEx_t_score&quot; or &quot;ENCODE_z_score&quot;), can be loaded by data(GTEx) or data(ENCODE), the default value is recommended &quot;GTEx_t_score&quot;.
</p>
</td></tr>
<tr><td><code id="tsea.analysis.multiple_+3A_ratio">ratio</code></td>
<td>
<p>the threshold to define tissue-specific genes (with top t-score or z-score), the default value is 0.05.
</p>
</td></tr>
<tr><td><code id="tsea.analysis.multiple_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p.adjust.method, c(&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tissue-specific enrichment analysis by Fisher's Exact Test for multiple gene list.
</p>


<h3>Value</h3>

<p>A data frame with p-value of tissue-specific enrichment result.
</p>
<p>Rows stand for tissue names and columns stand for sample names.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z. Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS_gene_multiple)
data(GTEx_t_score)
query_gene_list = GWAS_gene_multiple
tsea_t_multi = tsea.analysis.multiple(query_gene_list, 
	GTEx_t_score, 0.05, p.adjust.method = "BH")
</code></pre>

<hr>
<h2 id='tsea.expression.decode'>Tissue-specific enrichment analysis for RNA-Seq expression profiles
</h2><span id='topic+tsea.expression.decode'></span>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis to decode whether a given RNA-seq sample (RPKM) with potential confounding effects based on expression profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.expression.decode(query_mat_normalized_score, score, 
ratio = 0.05, p.adjust.method = "BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.expression.decode_+3A_query_mat_normalized_score">query_mat_normalized_score</code></td>
<td>
<p>a normalized RNA-seq RPKM object, which produced by &quot;tsea.expression.normalization&quot;.
</p>
</td></tr>
<tr><td><code id="tsea.expression.decode_+3A_score">score</code></td>
<td>
<p>a gene tissue-specific score matrix, c(&quot;GTEx_t_score&quot; or &quot;ENCODE_z_score&quot;), can be loaded by data(GTEx) or data(ENCODE), the default value is recommended &quot;GTEx_t_score&quot;.
</p>
</td></tr>
<tr><td><code id="tsea.expression.decode_+3A_ratio">ratio</code></td>
<td>
<p>the threshold to define tissue-specific genes (with top t-score or z-score), the default value is 0.05.
</p>
</td></tr>
<tr><td><code id="tsea.expression.decode_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p.adjust.method, c(&quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tissue-specific enrichment analysis for RNA-Seq expression profiles.
</p>


<h3>Value</h3>

<p>A data frame with p-value of tissue-specific enrichment result for RNA-Seq expression profiles.
</p>
<p>Rows stand for tissue names and columns stand for sample names.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(query_GTEx)
query_matrix = query_GTEx[,1:2]
data(correction_factor)
data(ENCODE_z_score)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")
</code></pre>

<hr>
<h2 id='tsea.expression.normalization'>RNA-Seq expression profiles normalization
</h2><span id='topic+tsea.expression.normalization'></span>

<h3>Description</h3>

<p>To avoid the data bias and adapt better data heterogeneity, before tsea.expression.decode() analysis, the raw discrete RPKM value have to normalized to continuous variable meet the normal distribution before t-test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.expression.normalization(query_mat, 
correction_factor, normalization = "abundance")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.expression.normalization_+3A_query_mat">query_mat</code></td>
<td>
<p>a RNA-seq RPKM object, row name should be gene symbol, and column name should be sample name.
</p>
</td></tr>
<tr><td><code id="tsea.expression.normalization_+3A_correction_factor">correction_factor</code></td>
<td>
<p>correction_factor, a gene table object contain genes average expression level and standard variance in GTEx database, can be loaded by data(correction_factor).
</p>
</td></tr>
<tr><td><code id="tsea.expression.normalization_+3A_normalization">normalization</code></td>
<td>
<p>normalization methods, c(&quot;z-score&quot;, &quot;abundance&quot;)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As RNA-Seq samples are often heterogeneous, before in-depth analysis, it is necessary to decode tissue heterogeneity to avoid samples with confounding effects. However, the raw discrete RPKM value have to normalized to continuous variable meet the normal distribution before t-test.
</p>


<h3>Value</h3>

<p>A data frame with normalized RNA-Seq expression profiles.
</p>
<p>Rows stand for tissue names and columns stand for sample names.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(query_GTEx)
query_matrix = query_GTEx[,1:2]
data(correction_factor)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
</code></pre>

<hr>
<h2 id='tsea.plot'>Tissue-specific enrichment analysis result heatmap plot
</h2><span id='topic+tsea.plot'></span>

<h3>Description</h3>

<p>Heat map plot for tissue-specific enrichment analysis result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.plot(tsea_result, threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.plot_+3A_tsea_result">tsea_result</code></td>
<td>
<p>the result of tissue-specific enrichment analysis, which produced by &quot;tsea.analysis&quot;, &quot;tsea.analysis.multiple&quot; or &quot;tsea.expression.decode&quot;.
</p>
</td></tr>
<tr><td><code id="tsea.plot_+3A_threshold">threshold</code></td>
<td>
<p>the p-value threshold to define if the gene list or RNA-seq data enriched in a given tissue, p-value greater than threshold will not be labeled in the plot. The default value is 0.05.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Heat map plot for tissue-specific enrichment analysis result 
</p>


<h3>Value</h3>

<p>Heatmap plot
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GWAS_gene_multiple)
data(GTEx_t_score)
query_gene_list = GWAS_gene_multiple
tsea_t_multi = tsea.analysis.multiple(query_gene_list, 
	GTEx_t_score, 0.05, p.adjust.method = "BH")
tsea.plot(tsea_t_multi, 0.05)
</code></pre>

<hr>
<h2 id='tsea.summary'>Tissue-specific enrichment analysis result summary
</h2><span id='topic+tsea.summary'></span>

<h3>Description</h3>

<p>Tissue-specific enrichment analysis result summary (list the top 3 most enriched tissues) from the given gene list or RNA-seq expression profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsea.summary(tsea_result)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsea.summary_+3A_tsea_result">tsea_result</code></td>
<td>
<p>the result of tissue-specific enrichment analysis, which produced by &quot;tsea.analysis&quot;, &quot;tsea.analysis.multiple&quot; or &quot;tsea.expression.decode&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tissue-specific enrichment analysis result summary
</p>


<h3>Value</h3>

  
<p>A data frame with summary result of top 3 most enriched tissues.
</p>
<p>Rows stand for sample names and columns stand for top 3 most enriched tissues (with p-value).
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Guangsheng Pei
</p>


<h3>References</h3>

<p>Pei G., Dai Y., Zhao Z., Jia P. (2019) deTS: Tissue-Specific Enrichment Analysis to decode tissue specificity. Bioinformatics, In submission.
</p>


<h3>See Also</h3>

<p>https://github.com/bsml320/deTS
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(query_GTEx)
query_matrix = query_GTEx
data(correction_factor)
data(ENCODE_z_score)
query_mat_zscore_nor = tsea.expression.normalization(query_matrix, 
	correction_factor, normalization = "z-score")
tseaed_in_ENCODE = tsea.expression.decode(query_mat_zscore_nor, 
	ENCODE_z_score, 0.05, p.adjust.method = "BH")	
tseaed_in_ENCODE_summary = tsea.summary(tseaed_in_ENCODE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
