<!DOCTYPE html><html lang="en"><head><title>Help for package DFD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DFD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convert_id'><p>Convert Gene Symbols to ENTREZ IDs</p></a></li>
<li><a href='#filter_drugs'><p>Re-rank drugs based on the number of targets</p></a></li>
<li><a href='#get_drugs'><p>Get Drugs associated with the differential expression profile</p></a></li>
<li><a href='#get_pert_by_type'><p>Filter the drug based on the tested cell-line</p></a></li>
<li><a href='#parse_perts'><p>Parse the perturbagens to find out their frequency of occurrence.</p></a></li>
<li><a href='#plot_activities'><p>Visualise the number of perturbagens and the top activities</p></a></li>
<li><a href='#prepare_ids'><p>Prepare IDs for CMAP Search</p></a></li>
<li><a href='#read_id'><p>Read Gene Symbols from CSV file into up and down regulated lists</p></a></li>
<li><a href='#run_pipeline'><p>Run the main pipeline for getting drugs from differentail expression profile</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Extract Drugs from Differential Expression Data from LINCS
Database</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohamed Soudy &lt;MohmedSoudy2009@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Get Drug information from given differential expression profile. The package search for the bioactive compounds from reference databases such as LINCS containing the genome-wide gene expression signature (GES) from tens of thousands of drug and genetic perturbations (Subramanian et al. (2017) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2017.10.049">doi:10.1016/j.cell.2017.10.049</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringr, gprofiler2, scales, ggplot2, gridExtra, ggpubr</td>
</tr>
<tr>
<td>Depends:</td>
<td>signatureSearch, signatureSearchData</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MohmedSoudy/DFD">https://github.com/MohmedSoudy/DFD</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MohmedSoudy/DFD/issues">https://github.com/MohmedSoudy/DFD/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-11 13:23:00 UTC; mohamed.soudy</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohamed Soudy [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-11 13:50:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='convert_id'>Convert Gene Symbols to ENTREZ IDs</h2><span id='topic+convert_id'></span>

<h3>Description</h3>

<p>The function is used to convert gene symbols to entrez ids and map the genes to human orthologs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_id(gene_symbols)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_id_+3A_gene_symbols">gene_symbols</code></td>
<td>
<p>gene symbols</p>
</td></tr>
</table>


<h3>Value</h3>

<p>IDs that are converted from gene symbols to ENTREZ gene ids
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>convert_id(c("TP53", "A2M"))

</code></pre>

<hr>
<h2 id='filter_drugs'>Re-rank drugs based on the number of targets</h2><span id='topic+filter_drugs'></span>

<h3>Description</h3>

<p>The function is used to re-rank drugs based on their targets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_drugs(drug_frame)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_drugs_+3A_drug_frame">drug_frame</code></td>
<td>
<p>drugs data frame returned by 'get_drugs' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>re-ranked drug data frame based on their number of targets
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='get_drugs'>Get Drugs associated with the differential expression profile</h2><span id='topic+get_drugs'></span>

<h3>Description</h3>

<p>The function is used to get list of drugs that are associated with differential expression profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_drugs(up_regulated, down_regulated)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_drugs_+3A_up_regulated">up_regulated</code></td>
<td>
<p>up-regulated genes returned by 'prepare_ids' function</p>
</td></tr>
<tr><td><code id="get_drugs_+3A_down_regulated">down_regulated</code></td>
<td>
<p>down-regulated genes returned by 'prepare_ids' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>significant drugs data frame that contains list of drugs with their targets
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='get_pert_by_type'>Filter the drug based on the tested cell-line</h2><span id='topic+get_pert_by_type'></span>

<h3>Description</h3>

<p>The function is used to re-rank drugs based on their targets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pert_by_type(perts, perts_type = c("cancer", "Normal"), high_targets = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_pert_by_type_+3A_perts">perts</code></td>
<td>
<p>drugs data frame returned by 'get_drugs' function</p>
</td></tr>
<tr><td><code id="get_pert_by_type_+3A_perts_type">perts_type</code></td>
<td>
<p>type of drug, whether it's from a normal or cancerous cell line</p>
</td></tr>
<tr><td><code id="get_pert_by_type_+3A_high_targets">high_targets</code></td>
<td>
<p>Boolean paramter to rank their drugs per the number of targets</p>
</td></tr>
</table>


<h3>Value</h3>

<p>re-ranked drug data frame based on their number of targets and cell line
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='parse_perts'>Parse the perturbagens to find out their frequency of occurrence.</h2><span id='topic+parse_perts'></span>

<h3>Description</h3>

<p>The function is used to parse drugs based on their targets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_perts(perts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_perts_+3A_perts">perts</code></td>
<td>
<p>drugs data frame returned by 'get_drugs' function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>re-ranked drug data frame based on their number of targets and cell line
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='plot_activities'>Visualise the number of perturbagens and the top activities</h2><span id='topic+plot_activities'></span>

<h3>Description</h3>

<p>The function is used to provide a handy visualization of the top activities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_activities(perts, top = 10, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_activities_+3A_perts">perts</code></td>
<td>
<p>drugs data frame returned by 'get_drugs' function or 'get_pert_by_type' function</p>
</td></tr>
<tr><td><code id="plot_activities_+3A_top">top</code></td>
<td>
<p>Number of activities to be visualised (N) The default value is set to 10.</p>
</td></tr>
<tr><td><code id="plot_activities_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save the output figure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>re-ranked drug data frame based on their number of targets and cell line
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='prepare_ids'>Prepare IDs for CMAP Search</h2><span id='topic+prepare_ids'></span>

<h3>Description</h3>

<p>The function is used to prepare the ids for the CMAP search
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_ids(up_regulated, down_regulated)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_ids_+3A_up_regulated">up_regulated</code></td>
<td>
<p>up regulated gene symbols</p>
</td></tr>
<tr><td><code id="prepare_ids_+3A_down_regulated">down_regulated</code></td>
<td>
<p>down regulated gene symbols</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing up_regulated and down_regulated genes ENTREZ gene ids
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='read_id'>Read Gene Symbols from CSV file into up and down regulated lists</h2><span id='topic+read_id'></span>

<h3>Description</h3>

<p>The function is used to read symbols from a CSV file. The gene symbols should be in the first column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_id(csv_path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_id_+3A_csv_path">csv_path</code></td>
<td>
<p>absolute path of CSV file containing gene symbols and sign</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing up_regulated and down_regulated genes symbols
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

<hr>
<h2 id='run_pipeline'>Run the main pipeline for getting drugs from differentail expression profile</h2><span id='topic+run_pipeline'></span>

<h3>Description</h3>

<p>The function is used to run the main pipeline by extracting the drug list given differential expressed genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pipeline(degs_path, output_path = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_pipeline_+3A_degs_path">degs_path</code></td>
<td>
<p>path to csv file containing degs see example file at https://raw.githubusercontent.com/MohmedSoudy/datasharing/master/differential-expression.csv</p>
</td></tr>
<tr><td><code id="run_pipeline_+3A_output_path">output_path</code></td>
<td>
<p>absolute path to output directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>significant drug data frame after the re-ranking step
</p>


<h3>Author(s)</h3>

<p>Mohamed Soudy <a href="mailto:Mohmedsoudy2009@gmail.com">Mohmedsoudy2009@gmail.com</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
