<!DOCTYPE html><html><head><title>Help for package dGAselID</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dGAselID}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AnalyzeResults'><p>AnalyzeResults</p></a></li>
<li><a href='#Crossover'><p>Crossover</p></a></li>
<li><a href='#dGAselID'><p>dGAselID</p></a></li>
<li><a href='#Elitism'><p>Elitism</p></a></li>
<li><a href='#EmbryonicSelection'><p>EmbryonicSelection</p></a></li>
<li><a href='#EvaluationFunction'><p>EvaluationFunction</p></a></li>
<li><a href='#frameShiftMutation'><p>frameShiftMutation</p></a></li>
<li><a href='#Individuals'><p>Individuals</p></a></li>
<li><a href='#InitialPopulation'><p>InitialPopulation</p></a></li>
<li><a href='#largeSegmentDeletion'><p>largeSegmentDeletion</p></a></li>
<li><a href='#nonSenseMutation'><p>nonSenseMutation</p></a></li>
<li><a href='#PlotGenAlg'><p>PlotGenAlg</p></a></li>
<li><a href='#pointMutation'><p>pointMutation</p></a></li>
<li><a href='#RandomAssortment'><p>RandomAssortment</p></a></li>
<li><a href='#RandomizePop'><p>RandomizePop</p></a></li>
<li><a href='#splitChromosomes'><p>splitChromosomes</p></a></li>
<li><a href='#transposon'><p>transposon</p></a></li>
<li><a href='#wholeChromosomeDeletion'><p>wholeChromosomeDeletion</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetic Algorithm with Incomplete Dominance for Feature
Selection</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Nicolae Teodor Melita</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nicolae Teodor Melita &lt;nt_melita@yahoo.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Feature selection from high dimensional data using a diploid
    genetic algorithm with Incomplete Dominance for genotype to phenotype mapping
    and Random Assortment of chromosomes approach to recombination.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1), Biobase, MLInterfaces</td>
</tr>
<tr>
<td>Imports:</td>
<td>genefilter, ALL, grDevices, graphics, stats, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-07-10 03:52:32 UTC; Dr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-07-10 05:02:55 UTC</td>
</tr>
</table>
<hr>
<h2 id='AnalyzeResults'>AnalyzeResults</h2><span id='topic+AnalyzeResults'></span>

<h3>Description</h3>

<p>Ranks individuals according to their fitness and records the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnalyzeResults(individuals, results, randomAssortment = TRUE, chrConf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AnalyzeResults_+3A_individuals">individuals</code></td>
<td>
<p>Population of individuals with diploid genotypes.</p>
</td></tr>
<tr><td><code id="AnalyzeResults_+3A_results">results</code></td>
<td>
<p>Results returned by EvaluationFunction().</p>
</td></tr>
<tr><td><code id="AnalyzeResults_+3A_randomassortment">randomAssortment</code></td>
<td>
<p>Random Assortment of Chromosomes for recombinations. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="AnalyzeResults_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)
 set.seed(1357)

 population0&lt;-InitialPopulation(smallALL, 14, 10, FALSE)
 individuals0&lt;-Individuals(population0)
 results0&lt;-EvaluationFunction(smallALL, individuals0, response="mol.biol",
             method=knn.cvI(k=3, l=2), trainTest="LOG")
 chrConf0&lt;-splitChromosomes(smallALL)
 iterRes0&lt;-AnalyzeResults(individuals0, results0, randomAssortment=TRUE, chrConf0)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='Crossover'>Crossover</h2><span id='topic+Crossover'></span>

<h3>Description</h3>

<p>Two-point crossover operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Crossover(c1, c2, chrConf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Crossover_+3A_c1">c1</code></td>
<td>
<p>Set of chromosomes.</p>
</td></tr>
<tr><td><code id="Crossover_+3A_c2">c2</code></td>
<td>
<p>Set of chromosomes.</p>
</td></tr>
<tr><td><code id="Crossover_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]
set.seed(1357)
population02&lt;-InitialPopulation(demoALL, 2, 4, FALSE)
chrConf02&lt;-splitChromosomes(demoALL, 2)
chrConf02
population02[1:2,]
Crossover(population02[1,], population02[2,], chrConf02)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='dGAselID'>dGAselID</h2><span id='topic+dGAselID'></span>

<h3>Description</h3>

<p>Initializes and starts the search with the genetic algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dGAselID(x, response, method = knn.cvI(k = 3, l = 2), trainTest = "LOG",
  startGenes, populationSize, iterations, noChr = 22, elitism = NA,
  ID = "ID1", pMutationChance = 0, nSMutationChance = 0,
  fSMutationChance = 0, lSDeletionChance = 0, wChrDeletionChance = 0,
  transposonChance = 0, randomAssortment = TRUE, embryonicSelection = NA,
  EveryGeneInInitialPopulation = TRUE, nnetSize = NA, nnetDecay = NA,
  rdaAlpha = NA, rdaDelta = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dGAselID_+3A_x">x</code></td>
<td>
<p>Dataset in ExpressionSet format.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_response">response</code></td>
<td>
<p>Response variable</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_method">method</code></td>
<td>
<p>Supervised classifier for fitness evaluation. Most of the supervised classifiers in MLInterfaces are acceptable. The default is knn.cvI(k=3, l=2).</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_traintest">trainTest</code></td>
<td>
<p>Cross-validation method. The default is &quot;LOG&quot;.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_startgenes">startGenes</code></td>
<td>
<p>Genes in the genotypes at initialization.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_populationsize">populationSize</code></td>
<td>
<p>Number of genotypes in initial population.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_nochr">noChr</code></td>
<td>
<p>Number of chromosomes. The default value is 22.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_elitism">elitism</code></td>
<td>
<p>Elite population in percentages.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_id">ID</code></td>
<td>
<p>Dominance. The default value is &quot;ID1&quot;. Use &quot;ID2&quot; for Incomplete Dominance.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_pmutationchance">pMutationChance</code></td>
<td>
<p>Chance for a Point Mutation to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_nsmutationchance">nSMutationChance</code></td>
<td>
<p>Chance for a Non-sense Mutation to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_fsmutationchance">fSMutationChance</code></td>
<td>
<p>Chance for a Frameshift Mutation to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_lsdeletionchance">lSDeletionChance</code></td>
<td>
<p>Chance for a Large Segment Deletion to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_wchrdeletionchance">wChrDeletionChance</code></td>
<td>
<p>Chance for a Whole Chromosome Deletion to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_transposonchance">transposonChance</code></td>
<td>
<p>Chance for a Transposon Mutation to occur. The default value is 0.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_randomassortment">randomAssortment</code></td>
<td>
<p>Random Assortment of Chromosomes for recombinations. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_embryonicselection">embryonicSelection</code></td>
<td>
<p>Remove chromosomes with fitness &lt; specified value. The default value is NA.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_everygeneininitialpopulation">EveryGeneInInitialPopulation</code></td>
<td>
<p>Request for every gene to be present in the initial population. The default value is TRUE.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_nnetsize">nnetSize</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_nnetdecay">nnetDecay</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_rdaalpha">rdaAlpha</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_rdadelta">rdaDelta</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td></tr>
<tr><td><code id="dGAselID_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a list containing 5 named vectors, records of the evolution:
</p>
<table>
<tr><td><code>DGenes</code></td>
<td>
<p>The occurrences in selected genotypes for every gene,</p>
</td></tr>
<tr><td><code>dGenes</code></td>
<td>
<p>The occurrences in discarded genotypes for every gene,</p>
</td></tr>
<tr><td><code>MaximumAccuracy</code></td>
<td>
<p>Maximum accuracy in every generation,</p>
</td></tr>
<tr><td><code>MeanAccuracy</code></td>
<td>
<p>Average accuracy in every generation,</p>
</td></tr>
<tr><td><code>MinAccuracy</code></td>
<td>
<p>Minimum accuracy in every generation,</p>
</td></tr>
<tr><td><code>BestIndividuals</code></td>
<td>
<p>Best individual in every generation.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)

 set.seed(149)
 res&lt;-dGAselID(smallALL, "mol.biol", trainTest=1:79, startGenes=12, populationSize=200,
               iterations=150, noChr=5, pMutationChance=0.0075, elitism=4)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='Elitism'>Elitism</h2><span id='topic+Elitism'></span>

<h3>Description</h3>

<p>Operator for elitism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Elitism(results, elitism, ID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Elitism_+3A_results">results</code></td>
<td>
<p>Results returned by EvaluationFunction().</p>
</td></tr>
<tr><td><code id="Elitism_+3A_elitism">elitism</code></td>
<td>
<p>Elite population in percentages.</p>
</td></tr>
<tr><td><code id="Elitism_+3A_id">ID</code></td>
<td>
<p>Dominance. The default value is &quot;ID1&quot;. Use &quot;ID2&quot; for Incomplete Dominance.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)
 set.seed(1357)

 population0&lt;-InitialPopulation(smallALL, 14, 8, FALSE)
 individuals0&lt;-Individuals(population0)
 results0&lt;-EvaluationFunction(smallALL, individuals0, response="mol.biol",
             method=knn.cvI(k=3, l=2), trainTest="LOG")
 Elitism(results0, 25, ID="ID1")
 Elitism(results0, 25, ID="ID2")
 
## End(Not run)
</code></pre>

<hr>
<h2 id='EmbryonicSelection'>EmbryonicSelection</h2><span id='topic+EmbryonicSelection'></span>

<h3>Description</h3>

<p>Function for deleting individuals with a fitness below a specified threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EmbryonicSelection(population, results, embryonicSelection)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EmbryonicSelection_+3A_population">population</code></td>
<td>
<p>Population of individuals with diploid genotypes.</p>
</td></tr>
<tr><td><code id="EmbryonicSelection_+3A_results">results</code></td>
<td>
<p>Results returned by EvaluationFunction().</p>
</td></tr>
<tr><td><code id="EmbryonicSelection_+3A_embryonicselection">embryonicSelection</code></td>
<td>
<p>Threshold value. The default value is NA.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)
 set.seed(1357)

 population0&lt;-InitialPopulation(smallALL, 14, 8, FALSE)
 individuals0&lt;-Individuals(population0)
 results0&lt;-EvaluationFunction(smallALL, individuals0, response="mol.biol",
             method=knn.cvI(k=3, l=2), trainTest="LOG")
 EmbryonicSelection(individuals0, results0, 0.5)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='EvaluationFunction'>EvaluationFunction</h2><span id='topic+EvaluationFunction'></span>

<h3>Description</h3>

<p>Evaluates the individuals' fitnesses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EvaluationFunction(x, individuals, response, method, trainTest, nnetSize = NA,
  nnetDecay = NA, rdaAlpha = NA, rdaDelta = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EvaluationFunction_+3A_x">x</code></td>
<td>
<p>Dataset in ExpressionSet format.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_individuals">individuals</code></td>
<td>
<p>Population of individuals with diploid genotypes.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_response">response</code></td>
<td>
<p>Response variable.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_method">method</code></td>
<td>
<p>Supervised classifier for fitness evaluation. Most of the supervised classifiers in MLInterfaces are acceptable. The default is knn.cvI(k=3, l=2).</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_traintest">trainTest</code></td>
<td>
<p>Cross-validation method. The default is &quot;LOG&quot;.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_nnetsize">nnetSize</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_nnetdecay">nnetDecay</code></td>
<td>
<p>for nnetI. The default value is NA.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_rdaalpha">rdaAlpha</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_rdadelta">rdaDelta</code></td>
<td>
<p>for rdaI. The default value is NA.</p>
</td></tr>
<tr><td><code id="EvaluationFunction_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 library(genefilter)
 library(ALL)
 data(ALL)
 bALL = ALL[, substr(ALL$BT,1,1) == "B"]
 smallALL = bALL[, bALL$mol.biol %in% c("BCR/ABL", "NEG")]
 smallALL$mol.biol = factor(smallALL$mol.biol)
 smallALL$BT = factor(smallALL$BT)
 f1 &lt;- pOverA(0.25, log2(100))
 f2 &lt;- function(x) (IQR(x) &gt; 0.5)
 f3 &lt;- ttest(smallALL$mol.biol, p=0.1)
 ff &lt;- filterfun(f1, f2, f3)
 selectedsmallALL &lt;- genefilter(exprs(smallALL), ff)
 smallALL = smallALL[selectedsmallALL, ]
 rm(f1)
 rm(f2)
 rm(f3)
 rm(ff)
 rm(bALL)
 sum(selectedsmallALL)
 set.seed(1357)

 population0&lt;-InitialPopulation(smallALL, 14, 8, FALSE)
 individuals0&lt;-Individuals(population0)
 results&lt;-EvaluationFunction(smallALL, individuals0, response="mol.biol",
             method=knn.cvI(k=3, l=2), trainTest="LOG")
 
## End(Not run)
</code></pre>

<hr>
<h2 id='frameShiftMutation'>frameShiftMutation</h2><span id='topic+frameShiftMutation'></span>

<h3>Description</h3>

<p>Operator for the frameshift mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frameShiftMutation(individuals, chrConf, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="frameShiftMutation_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="frameShiftMutation_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
<tr><td><code id="frameShiftMutation_+3A_mutationchance">mutationChance</code></td>
<td>
<p>Chance for a frameshift mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

chrConf&lt;-splitChromosomes(demoALL, 2)
chrConf
individuals

set.seed(123)
frameShiftMutation(individuals, chrConf, 20)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='Individuals'>Individuals</h2><span id='topic+Individuals'></span>

<h3>Description</h3>

<p>Generates individuals with diploid genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Individuals(population)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Individuals_+3A_population">population</code></td>
<td>
<p>Population of haploid genotypes.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

population02&lt;-InitialPopulation(demoALL, 20, 4, FALSE)
individuals02&lt;-Individuals(population02)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='InitialPopulation'>InitialPopulation</h2><span id='topic+InitialPopulation'></span>

<h3>Description</h3>

<p>Generates an initial randomly generated population of haploid genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InitialPopulation(x, populationSize, startGenes,
  EveryGeneInInitialPopulation = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InitialPopulation_+3A_x">x</code></td>
<td>
<p>Dataset in ExpressionSet format.</p>
</td></tr>
<tr><td><code id="InitialPopulation_+3A_populationsize">populationSize</code></td>
<td>
<p>Number of genotypes in initial population.</p>
</td></tr>
<tr><td><code id="InitialPopulation_+3A_startgenes">startGenes</code></td>
<td>
<p>Genes in the genotypes at initialization.</p>
</td></tr>
<tr><td><code id="InitialPopulation_+3A_everygeneininitialpopulation">EveryGeneInInitialPopulation</code></td>
<td>
<p>Request for every gene to be present in the initial population. The default value is TRUE.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

population01&lt;-InitialPopulation(demoALL, 4, 4)
population02&lt;-InitialPopulation(demoALL, 20, 4, FALSE)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='largeSegmentDeletion'>largeSegmentDeletion</h2><span id='topic+largeSegmentDeletion'></span>

<h3>Description</h3>

<p>Operator for the large segment deletion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>largeSegmentDeletion(individuals, chrConf, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="largeSegmentDeletion_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="largeSegmentDeletion_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
<tr><td><code id="largeSegmentDeletion_+3A_mutationchance">mutationChance</code></td>
<td>
<p>Chance for a large segment deletion mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

chrConf&lt;-splitChromosomes(demoALL, 2)
chrConf
individuals

set.seed(123)
largeSegmentDeletion(individuals, chrConf, 20)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='nonSenseMutation'>nonSenseMutation</h2><span id='topic+nonSenseMutation'></span>

<h3>Description</h3>

<p>Operator for the nonsense mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nonSenseMutation(individuals, chrConf, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nonSenseMutation_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="nonSenseMutation_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
<tr><td><code id="nonSenseMutation_+3A_mutationchance">mutationChance</code></td>
<td>
<p>Chance for a nonsense mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

chrConf&lt;-splitChromosomes(demoALL, 2)
chrConf
individuals

set.seed(123)
nonSenseMutation(individuals, chrConf, 20)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='PlotGenAlg'>PlotGenAlg</h2><span id='topic+PlotGenAlg'></span>

<h3>Description</h3>

<p>Function for graphically representing the evolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGenAlg(DGenes, dGenes, maxEval, meanEval)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotGenAlg_+3A_dgenes">DGenes</code></td>
<td>
<p>Occurences of genes as dominant.</p>
</td></tr>
<tr><td><code id="PlotGenAlg_+3A_dgenes">dGenes</code></td>
<td>
<p>Occurences of genes as recessive. For future developments.</p>
</td></tr>
<tr><td><code id="PlotGenAlg_+3A_maxeval">maxEval</code></td>
<td>
<p>Maximum fitness.</p>
</td></tr>
<tr><td><code id="PlotGenAlg_+3A_meaneval">meanEval</code></td>
<td>
<p>Average fitness.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
		#Graphical representation of the evolution after each generation.
		#Intended to be used by dGAselID() only.
		#Please refer to the example for dGAselID().
 
## End(Not run)
</code></pre>

<hr>
<h2 id='pointMutation'>pointMutation</h2><span id='topic+pointMutation'></span>

<h3>Description</h3>

<p>Operator for the point mutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pointMutation(individuals, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pointMutation_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="pointMutation_+3A_mutationchance">mutationChance</code></td>
<td>
<p>chance for a point mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

individuals
set.seed(123)
pointMutation(individuals, 4)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='RandomAssortment'>RandomAssortment</h2><span id='topic+RandomAssortment'></span>

<h3>Description</h3>

<p>Random assortment of chromosomes operator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RandomAssortment(newChrs, chrConf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RandomAssortment_+3A_newchrs">newChrs</code></td>
<td>
<p>Set of chromosomes.</p>
</td></tr>
<tr><td><code id="RandomAssortment_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

population02&lt;-InitialPopulation(demoALL, 2, 4, FALSE)
chrConf02&lt;-splitChromosomes(demoALL, 4)

set.seed(1357)
cr1&lt;-Crossover(population02[1,], population02[2,], chrConf02)
RandomAssortment(cr1, chrConf02)
cr1
chrConf02
 
## End(Not run)
</code></pre>

<hr>
<h2 id='RandomizePop'>RandomizePop</h2><span id='topic+RandomizePop'></span>

<h3>Description</h3>

<p>Generates a random population for the next generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RandomizePop(population)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RandomizePop_+3A_population">population</code></td>
<td>
<p>Population of chromosome sets in current generation.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

population01&lt;-InitialPopulation(demoALL, 4, 4)
population01
RandomizePop(population01)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='splitChromosomes'>splitChromosomes</h2><span id='topic+splitChromosomes'></span>

<h3>Description</h3>

<p>Divides the genotypes into sets with a desired number of chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitChromosomes(x, noChr = 22)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitChromosomes_+3A_x">x</code></td>
<td>
<p>Dataset in ExpressionSet format.</p>
</td></tr>
<tr><td><code id="splitChromosomes_+3A_nochr">noChr</code></td>
<td>
<p>Desired number of chromosomes. The default value is 22.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

splitChromosomes(demoALL, 3)
splitChromosomes(demoALL)

 
## End(Not run)
</code></pre>

<hr>
<h2 id='transposon'>transposon</h2><span id='topic+transposon'></span>

<h3>Description</h3>

<p>Operator for transposons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transposon(individuals, chrConf, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transposon_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="transposon_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
<tr><td><code id="transposon_+3A_mutationchance">mutationChance</code></td>
<td>
<p>Chance for a transposon mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

chrConf&lt;-splitChromosomes(demoALL, 2)
chrConf
individuals

set.seed(123)
transposon(individuals, chrConf, 20)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='wholeChromosomeDeletion'>wholeChromosomeDeletion</h2><span id='topic+wholeChromosomeDeletion'></span>

<h3>Description</h3>

<p>Operator for the deletion of a whole chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wholeChromosomeDeletion(individuals, chrConf, mutationChance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wholeChromosomeDeletion_+3A_individuals">individuals</code></td>
<td>
<p>dataset returned by Individuals().</p>
</td></tr>
<tr><td><code id="wholeChromosomeDeletion_+3A_chrconf">chrConf</code></td>
<td>
<p>Configuration of chromosomes returned by splitChromosomes().</p>
</td></tr>
<tr><td><code id="wholeChromosomeDeletion_+3A_mutationchance">mutationChance</code></td>
<td>
<p>Chance for a deletion of a whole chromosome mutation to occur.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ALL)
data(ALL)

demoALL&lt;-ALL[1:12,1:8]

set.seed(1234)
population&lt;-InitialPopulation(demoALL, 4, 9)
individuals&lt;-Individuals(population)

chrConf&lt;-splitChromosomes(demoALL, 2)
chrConf
individuals

set.seed(123)
wholeChromosomeDeletion(individuals, chrConf, 20)
 
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
