<!DOCTYPE html><html><head><title>Help for package DGEAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DGEAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DGEAR'><p>Differential Gene Expression Analysis with R</p></a></li>
<li><a href='#gene_exp_data'><p>A dataset containing gene expression data</p></a></li>
<li><a href='#perform_anova'><p>Function for ANOVA One-Way Test</p></a></li>
<li><a href='#perform_dunnett_test'><p>Function for Dunnett's Test</p></a></li>
<li><a href='#perform_h_test'><p>Function for Half's-T-Test Analysis</p></a></li>
<li><a href='#perform_t_test'><p>Function for t-Test Analysis</p></a></li>
<li><a href='#perform_wilcox_test'><p>Function for Wilcoxon-Mann-Whitney U-Test</p></a></li>
<li><a href='#read_and_preprocess_data'><p>Function to read data and perform initial pre-processing</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Gene Expression Analysis with R</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Koushik Bardhan <a href="https://orcid.org/0009-0002-8846-8347"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, ctb],
  Chiranjib Sarkar <a href="https://orcid.org/0000-0003-1536-7449"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Koushik Bardhan &lt;koushikbardhan2000@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analyses gene expression data derived from experiments to detect differentially expressed genes by employing the concept of majority voting with five different statistical models. It includes functions for differential expression analysis, significance testing, etc. It simplifies the process of uncovering meaningful patterns and trends within gene expression data, aiding researchers in downstream analysis. Boyer, R.S., Moore, J.S. (1991) &lt;<a href="https://doi.org/10.1007%2F978-94-011-3488-0_5">doi:10.1007/978-94-011-3488-0_5</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>DescTools</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-30 15:57:18 UTC; INFINIX</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-30 19:30:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='DGEAR'>Differential Gene Expression Analysis with R</h2><span id='topic+DGEAR'></span>

<h3>Description</h3>

<p>Main function which incorporates results from five statistical models and detects DEGs through majority voting
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DGEAR(dataframe, con1, con2, exp1, exp2, alpha, votting_cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DGEAR_+3A_dataframe">dataframe</code></td>
<td>
<p>A matrix containing the gene expression data</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_con1">con1</code></td>
<td>
<p>Starting column of the control of the expression data</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_con2">con2</code></td>
<td>
<p>Ending column of the control of the expression data</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_exp1">exp1</code></td>
<td>
<p>Starting column of the experiment of the expression data</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_exp2">exp2</code></td>
<td>
<p>Ending column of the experiment of the expression data</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (0.05 states 5 % significance)(Default = 0.05).</p>
</td></tr>
<tr><td><code id="DGEAR_+3A_votting_cutoff">votting_cutoff</code></td>
<td>
<p>A numeric value serves as Majority voting (Default = 2)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To use this tool the necessary parameters are con1 = Control start column, con2 = Control end column, exp1 = Experiment start column, exp2 = Experiment end column, alpha = Value of significance level, voting_cutoff = Majority voting value (not more than 5, since there are 5 statistical methods which take part in the majority voting)
</p>


<h3>Value</h3>

<p>A matrix containing Differentially Expressed Genes(DEGs) detected
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
DGEAR(dataframe = gene_exp_data, con1 = 1, con2 = 10,
  exp1 = 11, exp2 = 20, alpha = 0.05, votting_cutoff = 2)
</code></pre>

<hr>
<h2 id='gene_exp_data'>A dataset containing gene expression data</h2><span id='topic+gene_exp_data'></span>

<h3>Description</h3>

<p>This dataset contains simulated gene expression data for analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_exp_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 20 columns, where columns 1 to 10 being control and 11 to 20 being experiment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Data will be loaded with lazy loading and can be accessible when needed.
data("gene_exp_data")
head(gene_exp_data)
</code></pre>

<hr>
<h2 id='perform_anova'>Function for ANOVA One-Way Test</h2><span id='topic+perform_anova'></span>

<h3>Description</h3>

<p>Function for ANOVA One-Way Test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_anova(datafile, con, exp, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_anova_+3A_datafile">datafile</code></td>
<td>
<p>A matrix containing the gene expression data</p>
</td></tr>
<tr><td><code id="perform_anova_+3A_con">con</code></td>
<td>
<p>A data frame or matrix containing the expression values for the control.</p>
</td></tr>
<tr><td><code id="perform_anova_+3A_exp">exp</code></td>
<td>
<p>A data frame or matrix containing the expression values for the experiment.</p>
</td></tr>
<tr><td><code id="perform_anova_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (default = 0.05 states 5 % significance).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing values for statistic score, p-values etc for each gene being tested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
data = read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
perform_anova(datafile = data$datafile, con= data$con, exp= data$exp)
</code></pre>

<hr>
<h2 id='perform_dunnett_test'>Function for Dunnett's Test</h2><span id='topic+perform_dunnett_test'></span>

<h3>Description</h3>

<p>Function for Dunnett's Test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_dunnett_test(datafile, con, exp, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_dunnett_test_+3A_datafile">datafile</code></td>
<td>
<p>A matrix containing the gene expression data</p>
</td></tr>
<tr><td><code id="perform_dunnett_test_+3A_con">con</code></td>
<td>
<p>A data frame or matrix containing the expression values for the control.</p>
</td></tr>
<tr><td><code id="perform_dunnett_test_+3A_exp">exp</code></td>
<td>
<p>A data frame or matrix containing the expression values for the experiment.</p>
</td></tr>
<tr><td><code id="perform_dunnett_test_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (default = 0.05 states 5 % significance).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing values for statistic score, p-values etc for each gene being tested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
data = read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
perform_dunnett_test(datafile = data$datafile, con= data$con, exp= data$exp)
</code></pre>

<hr>
<h2 id='perform_h_test'>Function for Half's-T-Test Analysis</h2><span id='topic+perform_h_test'></span>

<h3>Description</h3>

<p>Function for Half's-T-Test Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_h_test(con, exp, alpha = 0.05, FC)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_h_test_+3A_con">con</code></td>
<td>
<p>A data frame or matrix containing the expression values for the control.</p>
</td></tr>
<tr><td><code id="perform_h_test_+3A_exp">exp</code></td>
<td>
<p>A data frame or matrix containing the expression values for the experiment.</p>
</td></tr>
<tr><td><code id="perform_h_test_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (default = 0.05 states 5 % significance).</p>
</td></tr>
<tr><td><code id="perform_h_test_+3A_fc">FC</code></td>
<td>
<p>An array  or list containing fold change values for each gene, calculated by</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing values for statistic score, p-values etc for each gene being tested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
data = read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
perform_h_test(con= data$con, exp= data$exp, FC = data$FC)
</code></pre>

<hr>
<h2 id='perform_t_test'>Function for t-Test Analysis</h2><span id='topic+perform_t_test'></span>

<h3>Description</h3>

<p>Function for t-Test Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_t_test(con, exp, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_t_test_+3A_con">con</code></td>
<td>
<p>A data frame or matrix containing the expression values for the control.</p>
</td></tr>
<tr><td><code id="perform_t_test_+3A_exp">exp</code></td>
<td>
<p>A data frame or matrix containing the expression values for the experiment.</p>
</td></tr>
<tr><td><code id="perform_t_test_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (default = 0.05 states 5 % significance).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing values for statistic score, p-values etc for each gene being tested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
data = read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
perform_t_test(con= data$con, exp= data$exp)
</code></pre>

<hr>
<h2 id='perform_wilcox_test'>Function for Wilcoxon-Mann-Whitney U-Test</h2><span id='topic+perform_wilcox_test'></span>

<h3>Description</h3>

<p>Function for Wilcoxon-Mann-Whitney U-Test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_wilcox_test(con, exp, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_wilcox_test_+3A_con">con</code></td>
<td>
<p>A data frame or matrix containing the expression values for the control.</p>
</td></tr>
<tr><td><code id="perform_wilcox_test_+3A_exp">exp</code></td>
<td>
<p>A data frame or matrix containing the expression values for the experiment.</p>
</td></tr>
<tr><td><code id="perform_wilcox_test_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (default = 0.05 states 5 % significance).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing values for statistic score, p-values etc for each gene being tested.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(DGEAR)
data("gene_exp_data")
data = read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
perform_wilcox_test(con= data$con, exp= data$exp)
</code></pre>

<hr>
<h2 id='read_and_preprocess_data'>Function to read data and perform initial pre-processing</h2><span id='topic+read_and_preprocess_data'></span>

<h3>Description</h3>

<p>Function to read data and perform initial pre-processing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_and_preprocess_data(
  datafile,
  con1,
  con2,
  exp1,
  exp2,
  alpha = 0.05,
  votting_cutoff = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_and_preprocess_data_+3A_datafile">datafile</code></td>
<td>
<p>A matrix or data frame containing gene expression data</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_con1">con1</code></td>
<td>
<p>Starting column of the control of the expression data</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_con2">con2</code></td>
<td>
<p>Ending column of the control of the expression data</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_exp1">exp1</code></td>
<td>
<p>Starting column of the experiment of the expression data</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_exp2">exp2</code></td>
<td>
<p>Ending column of the experiment of the expression data</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_alpha">alpha</code></td>
<td>
<p>Value of significance level ranging from 0 to 1 (0.05 states 5 % significance)(Default = 0.05).</p>
</td></tr>
<tr><td><code id="read_and_preprocess_data_+3A_votting_cutoff">votting_cutoff</code></td>
<td>
<p>A numeric value serves as Majority voting (Default = 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A large list containing the data file and the input values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("gene_exp_data")
read_and_preprocess_data(datafile = gene_exp_data, con1=1,con2=10,exp1=11,exp2=20)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
