<!DOCTYPE html><html><head><title>Help for package DGEobj</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DGEobj}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DGEobj-package'><p>DGEobj Package Overview</p></a></li>
<li><a href='#[.DGEobj'><p>Subset with square brackets</p></a></li>
<li><a href='#addItem'><p>Add a data item</p></a></li>
<li><a href='#addItems'><p>Add multiple data items</p></a></li>
<li><a href='#annotateDGEobj'><p>Add annotations</p></a></li>
<li><a href='#as.list.DGEobj'><p>Cast as a simple list</p></a></li>
<li><a href='#baseType'><p>Get the baseType of an internal data item</p></a></li>
<li><a href='#baseTypes'><p>Get a list of the available baseTypes</p></a></li>
<li><a href='#dim.DGEobj'><p>Get the &quot;assay&quot; dimensions (row/genes by col/samples)</p></a></li>
<li><a href='#dimnames.DGEobj'><p>Get the &quot;assay&quot; names (row/genes by col/samples)</p></a></li>
<li><a href='#getAttributes'><p>Get all attributes</p></a></li>
<li><a href='#getBaseType'><p>Retrieve data items by baseType</p></a></li>
<li><a href='#getItem'><p>Retrieve a data item by name</p></a></li>
<li><a href='#getItems'><p>Retrieve multiple data items by name</p></a></li>
<li><a href='#getType'><p>Retrieve data items by type</p></a></li>
<li><a href='#initDGEobj'><p>Initialize with base data (primaryAssayData, row annotations, col annotations)</p></a></li>
<li><a href='#initDGEobjDef'><p>Instantiate a class DGEobjDef object.</p></a></li>
<li><a href='#inventory'><p>Retrieve the object inventory</p></a></li>
<li><a href='#newType'><p>Add a new type definition to a DGEobj</p></a></li>
<li><a href='#print.DGEobj'><p>Print the Inventory</p></a></li>
<li><a href='#resetDGEobj'><p>Reset to original data</p></a></li>
<li><a href='#rmItem'><p>Removes a named data item</p></a></li>
<li><a href='#setAttributes'><p>Set attributes</p></a></li>
<li><a href='#showAttributes'><p>Print attributes</p></a></li>
<li><a href='#showMeta'><p>Retrieve the Key/Value metadata attributes that have a character value and length of 1</p></a></li>
<li><a href='#showTypes'><p>Returns and prints the list of all defined types</p></a></li>
<li><a href='#subset.DGEobj'><p>Subset internal row or column data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Gene Expression (DGE) Analysis Results Data Object</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Provides a flexible container to manage and annotate Differential Gene 
    Expression (DGE) analysis results (Smythe et. al (2015) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv007">doi:10.1093/nar/gkv007</a>&gt;).  
    The DGEobj has data slots for row (gene), col (samples), assays (matrix n-rows 
    by m-samples dimensions) and metadata (not keyed to row, col, or assays).  
    A set of accessory functions to deposit, query and retrieve subsets of a data 
    workflow has been provided.  Attributes are used to capture metadata such as 
    species and gene model, including reproducibility information such that a 3rd 
    party can access a DGEobj history to see how each data object was created or 
    modified.  Since the DGEobj is customizable and extensible it is not limited 
    to RNA-seq analysis types of workflows &ndash; it can accommodate nearly any data 
    analysis workflow that starts from a matrix of assays (rows) by samples (columns).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, magrittr, stringr, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>biomaRt, conflicted, dplyr, edgeR, GenomicRanges, glue,
knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-13 19:51:14 UTC; conniebrett</td>
</tr>
<tr>
<td>Author:</td>
<td>John Thompson [aut],
  Connie Brett [aut, cre],
  Isaac Neuhaus [aut],
  Ryan Thompson [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Connie Brett &lt;connie@aggregate-genius.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-16 07:10:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='DGEobj-package'>DGEobj Package Overview</h2><span id='topic+DGEobj-package'></span>

<h3>Description</h3>

<p>DGEobj is an S3 data class that provides a flexible container for Differential Gene
Expression (DGE) analysis results.  The DGEobj class is designed to be extensible
allowing definition of new data types as needed. A set of accessory functions to
deposit, query and retrieve subsets of a data workflow has been provided.  Attributes
are used to capture metadata such as species and gene model, including reproducibility
information such that a 3rd party can access a DGEobj history to see how each data
object was created or modified.
</p>


<h3>Details</h3>

<p>Operationally, the DGEobj is influenced by the RangedSummarizedExperiment (RSE).
The DGEobj has data slots for row (gene), col (samples), assays (anything with n-rows
by m-samples dimensions) and metadata (anything that can't be keyed to row, col or
assay).  The key motivation for creating the DGEobj data structure is that the RSE
only allows one data item each in the row and col slots and thus is unsuitable for
capturing the plethora of data objects created during a typical DGE workflow.   The
DGEobj data structure can hold any number of row and col data objects and thus is
suitable for capturing the multiple steps of a downstream analysis.
</p>
<p>Certain object types, primarily the count matrix and associated row and column info,
are defined as unique which means only one instance of that type may be added to
the DGEobj.
</p>
<p>When multiple objects of one type are included in a DGEobj (e.g. two different fits),
the concept of parent attributes is used to associate downstream data objects (e.g.
contrasts) with the appropriate data object they are derived from.
</p>


<h3>More Information</h3>

<p><code>browseVignettes(package = 'DGEobj')</code>
</p>

<hr>
<h2 id='+5B.DGEobj'>Subset with square brackets</h2><span id='topic++5B.DGEobj'></span>

<h3>Description</h3>

<p>Subset with square brackets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
x[...]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.DGEobj_+3A_x">x</code></td>
<td>
<p>A DGEobj</p>
</td></tr>
<tr><td><code id="+2B5B.DGEobj_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>

<hr>
<h2 id='addItem'>Add a data item</h2><span id='topic+addItem'></span>

<h3>Description</h3>

<p>Add a data item
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addItem(
  dgeObj,
  item,
  itemName,
  itemType,
  funArgs = match.call(),
  itemAttr,
  parent = "",
  overwrite = FALSE,
  init = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addItem_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="addItem_+3A_item">item</code></td>
<td>
<p>The data item to be deposited in the DGEobj</p>
</td></tr>
<tr><td><code id="addItem_+3A_itemname">itemName</code></td>
<td>
<p>The user-assigned name for this data item</p>
</td></tr>
<tr><td><code id="addItem_+3A_itemtype">itemType</code></td>
<td>
<p>The type attribute.  See showTypes() to see the predefined types &ndash; types are extensible with the newType() function.</p>
</td></tr>
<tr><td><code id="addItem_+3A_funargs">funArgs</code></td>
<td>
<p>(optional) A text field to annotate how the data object was created. If the result of match.call() is passed as this argument, the name and arguments used in the current function are captured</p>
</td></tr>
<tr><td><code id="addItem_+3A_itemattr">itemAttr</code></td>
<td>
<p>(optional) A named list of attributes to add directly to the item</p>
</td></tr>
<tr><td><code id="addItem_+3A_parent">parent</code></td>
<td>
<p>(optional) itemName of the parent of this item</p>
</td></tr>
<tr><td><code id="addItem_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite a matching data object stored in the itemName slot (default = FALSE)</p>
</td></tr>
<tr><td><code id="addItem_+3A_init">init</code></td>
<td>
<p>Internal Use (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   myFunArgs &lt;- match.call()  #  Capture calling function and arguments

   myDGEobj &lt;- addItem(myDGEobj, item     = MyCounts,
                                 itemName = "counts",
                                 itemType = "counts",
                                 funArgs  = myFunArgs)

## End(Not run)

</code></pre>

<hr>
<h2 id='addItems'>Add multiple data items</h2><span id='topic+addItems'></span>

<h3>Description</h3>

<p>Add multiple data items
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addItems(dgeObj, itemList, itemTypes, parents, itemAttr, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addItems_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="addItems_+3A_itemlist">itemList</code></td>
<td>
<p>A named list of data items to add to DGEobj</p>
</td></tr>
<tr><td><code id="addItems_+3A_itemtypes">itemTypes</code></td>
<td>
<p>A list of type values for each item on itemList</p>
</td></tr>
<tr><td><code id="addItems_+3A_parents">parents</code></td>
<td>
<p>(optional) A list of parent values for each item on itemList
(optional, but highly recommended)</p>
</td></tr>
<tr><td><code id="addItems_+3A_itemattr">itemAttr</code></td>
<td>
<p>(optional) An named list of attributes to add to each item.
These attributes will be attached to all items in the call.</p>
</td></tr>
<tr><td><code id="addItems_+3A_overwrite">overwrite</code></td>
<td>
<p>Whether to overwrite a matching data object stored in the
itemName slot (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   # NOTE: Requires the edgeR package

   # Add normalized counts and log2CPM as additional "assay" items in the DGEobj
   dgeObj  &lt;- readRDS(system.file("exampleObj.RDS", package = "DGEobj"))
   dgeList &lt;- edgeR::calcNormFactors(edgeR::DGEList(dgeObj$counts), method="TMM")
   log2cpm &lt;- edgeR::cpm(dgeList, log = TRUE)

   dgeObj &lt;- addItems(dgeObj,
                      itemList = list(newDgelist = dgeList, Log2CPM = log2cpm),
                      itemTypes = list("assay", "assay"),
                      parents = list("counts", "newDgelist")
   )
   inventory(dgeObj)

## End(Not run)

</code></pre>

<hr>
<h2 id='annotateDGEobj'>Add annotations</h2><span id='topic+annotateDGEobj'></span>

<h3>Description</h3>

<p>Reads an annotation file containing key/value pairs or a named list and attaches them
attributes to a DGEobj. If a file is used, it should be a text file
containing key/value pairs separated by an equals sign. The keys argument
specifies which keys we want to capture as attributes on the DGEobj.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotateDGEobj(dgeObj, annotations, keys = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotateDGEobj_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A object of class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="annotateDGEobj_+3A_annotations">annotations</code></td>
<td>
<p>Either a character string path to a file with annotations
given as key/value pairs separated by an equal sign, or a named list of
key/value pairs</p>
</td></tr>
<tr><td><code id="annotateDGEobj_+3A_keys">keys</code></td>
<td>
<p>By default (value = NULL), all keys are read in and applied as
DGEobj attributes.  Use the keys argument to specify a specific list of
keys to read from the file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   MyDgeObj &lt;- system.file("exampleObj.RDS", package = "DGEobj")

## Not run: 
   #using a text file file of key=value pairs
   annotationFile &lt;- "/location/to/myAnnotations.txt"
   MyDgeObj &lt;- annotateDGEobj(MyDgeObj, annotationFile)

   #using a named list of key/values
   annotations &lt;- list(Title     = "Rat Liver Slices from Bile Duct Ligation animals",
                       Organism  = "Rat",
                       GeneModel = "Ensembl.R89")
   MyDgeObj &lt;- annotateDGEobj(MyDgeObj, annotations)

## End(Not run)

</code></pre>

<hr>
<h2 id='as.list.DGEobj'>Cast as a simple list</h2><span id='topic+as.list.DGEobj'></span>

<h3>Description</h3>

<p>Cast as a simple list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
as.list(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list.DGEobj_+3A_x">x</code></td>
<td>
<p>A DGEobj</p>
</td></tr>
<tr><td><code id="as.list.DGEobj_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    mylist &lt;- as.list(exObj)

</code></pre>

<hr>
<h2 id='baseType'>Get the baseType of an internal data item</h2><span id='topic+baseType'></span>

<h3>Description</h3>

<p>Get the baseType of an internal data item
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseType(dgeObj, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="baseType_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="baseType_+3A_type">type</code></td>
<td>
<p>An item type for which you want the baseType</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character string
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    baseType(exObj, type = "DGEList")

</code></pre>

<hr>
<h2 id='baseTypes'>Get a list of the available baseTypes</h2><span id='topic+baseTypes'></span>

<h3>Description</h3>

<p>Get a list of the available baseTypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>baseTypes(dgeObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="baseTypes_+3A_dgeobj">dgeObj</code></td>
<td>
<p>(optional) A class DGEobj created by function initDGEobj()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of baseTypes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Global definition of baseTypes
    baseTypes()

    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    # Basetypes from a specific DGEobj
    baseTypes(exObj)

</code></pre>

<hr>
<h2 id='dim.DGEobj'>Get the &quot;assay&quot; dimensions (row/genes by col/samples)</h2><span id='topic+dim.DGEobj'></span>

<h3>Description</h3>

<p>Returns the dimensions of the assay data (baseType)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
dim(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim.DGEobj_+3A_x">x</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer vector [r,c] with a length of 2.
</p>

<hr>
<h2 id='dimnames.DGEobj'>Get the &quot;assay&quot; names (row/genes by col/samples)</h2><span id='topic+dimnames.DGEobj'></span>

<h3>Description</h3>

<p>Returns a list of length 2 containing the the assay data names (baseType)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
dimnames(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimnames.DGEobj_+3A_x">x</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length 2 containing rownames and colnames of the DGEobj
</p>

<hr>
<h2 id='getAttributes'>Get all attributes</h2><span id='topic+getAttributes'></span>

<h3>Description</h3>

<p>Get all user-defined attributes from a DGEobj except for any listed in the
excludeList argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAttributes(
  dgeObj,
  excludeList = list("dim", "dimnames", "names", "row.names", "class")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAttributes_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A DGEobj</p>
</td></tr>
<tr><td><code id="getAttributes_+3A_excludelist">excludeList</code></td>
<td>
<p>A list of attribute names to exclude from the output (default = list(&quot;dim&quot;, &quot;dimnames&quot;, &quot;names&quot;, &quot;row.names&quot;))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    getAttributes(exObj)

    # Get the formula attribute from the design (if set)
    attr(exObj$design, "formula")

</code></pre>

<hr>
<h2 id='getBaseType'>Retrieve data items by baseType</h2><span id='topic+getBaseType'></span>

<h3>Description</h3>

<p>Retrieve data items by baseType
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBaseType(dgeObj, baseType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBaseType_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="getBaseType_+3A_basetype">baseType</code></td>
<td>
<p>One or more of: [&quot;row&quot;, &quot;col&quot;, &quot;assay&quot;, &quot;meta&quot;]</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data items
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    Assays &lt;- getBaseType(exObj, baseType = "assay")
    AssaysAndMeta &lt;- getBaseType(exObj, c("assay", "meta"))

</code></pre>

<hr>
<h2 id='getItem'>Retrieve a data item by name</h2><span id='topic+getItem'></span>

<h3>Description</h3>

<p>Retrieve a data item by name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getItem(dgeObj, itemName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getItem_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="getItem_+3A_itemname">itemName</code></td>
<td>
<p>Name of item to retrieve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The requested data item
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    MyCounts &lt;- getItem(exObj, "counts")

</code></pre>

<hr>
<h2 id='getItems'>Retrieve multiple data items by name</h2><span id='topic+getItems'></span>

<h3>Description</h3>

<p>Retrieve multiple data items by name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getItems(dgeObj, itemNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getItems_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="getItems_+3A_itemnames">itemNames</code></td>
<td>
<p>A character string, character vector, or list names to retrieve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    myList &lt;- getItems(exObj, list("counts", "geneData"))
    names(myList)

</code></pre>

<hr>
<h2 id='getType'>Retrieve data items by type</h2><span id='topic+getType'></span>

<h3>Description</h3>

<p>Retrieve data items by type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getType(dgeObj, type, parent)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getType_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="getType_+3A_type">type</code></td>
<td>
<p>A type or list of types to retrieve</p>
</td></tr>
<tr><td><code id="getType_+3A_parent">parent</code></td>
<td>
<p>(optional) Filter return list for common parent (e.g. useful
to select one set of contrast results when multiple fits have been performed)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data items
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    MyRawData      &lt;- getType(exObj, type = list("counts", "design", "geneData"))

</code></pre>

<hr>
<h2 id='initDGEobj'>Initialize with base data (primaryAssayData, row annotations, col annotations)</h2><span id='topic+initDGEobj'></span>

<h3>Description</h3>

<p>Initialize with base data (primaryAssayData, row annotations, col annotations)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initDGEobj(
  primaryAssayData,
  rowData,
  colData,
  level,
  customAttr,
  allowShortSampleIDs = FALSE,
  DGEobjDef = initDGEobjDef()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initDGEobj_+3A_primaryassaydata">primaryAssayData</code></td>
<td>
<p>A numeric matrix or dataframe with row and
colnames. Each column represents a sample.  Each row represents and assay.
This is typically the counts matrix in a DGE RNA-Seq experiment.</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_rowdata">rowData</code></td>
<td>
<p>Gene, exon, isoform or protein level annotation. Rownames
must match the rownames in primaryAssayData</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_coldata">colData</code></td>
<td>
<p>A dataframe describing the experiment design.
Rownames much match the colnames(primaryAssayData)</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_level">level</code></td>
<td>
<p>One of &quot;gene&quot;, &quot;exon&quot;, &quot;isoform&quot; or &quot;protein&quot;</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_customattr">customAttr</code></td>
<td>
<p>(optional) Named list of attributes</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_allowshortsampleids">allowShortSampleIDs</code></td>
<td>
<p>Using sequential integer rownames (even if typed
as character) is discouraged and by default will abort the DGEobj creation.
If you have a legitimate need to have short sample names composed of
numeric characters, you can set this argument to TRUE (default = FALSE)</p>
</td></tr>
<tr><td><code id="initDGEobj_+3A_dgeobjdef">DGEobjDef</code></td>
<td>
<p>An object definition. Defaults to the global
DGEobj definition (initDGEobjDef()) and you usually shouldn't change this unless
you're customizing the object for new data types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   dgeObj   &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))
   MyCounts &lt;- dgeObj$counts
   geneinfo &lt;- dgeObj$geneData
   sampinfo &lt;- dgeObj$design

  myDgeObj &lt;- initDGEobj(primaryAssayData = MyCounts,
                         rowData          = geneinfo,
                         colData          = sampinfo,
                         level            = "gene",
                         customAttr       = list (Genome = "Rat.B6.0",
                                               GeneModel = "Ensembl.R89"))

</code></pre>

<hr>
<h2 id='initDGEobjDef'>Instantiate a class DGEobjDef object.</h2><span id='topic+initDGEobjDef'></span>

<h3>Description</h3>

<p>Instantiate a class DGEobjDef object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initDGEobjDef(levels, primaryAssayNames, types, uniqueTypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initDGEobjDef_+3A_levels">levels</code></td>
<td>
<p>A character string or vector providing names for new levels</p>
</td></tr>
<tr><td><code id="initDGEobjDef_+3A_primaryassaynames">primaryAssayNames</code></td>
<td>
<p>A character string or vector, must be the same length as levels
This argument supplies the primaryAssayNames for the corresponding levels.</p>
</td></tr>
<tr><td><code id="initDGEobjDef_+3A_types">types</code></td>
<td>
<p>A named character vector of new types where the
values indicate the basetype for each named type (optional)</p>
</td></tr>
<tr><td><code id="initDGEobjDef_+3A_uniquetypes">uniqueTypes</code></td>
<td>
<p>A name or vector of names to add to the uniqueType list (optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A class DGEobjDef object suitable for use with initDGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # return the default DGEobj definition
    myDGEobjDef &lt;- initDGEobjDef()

    # Optionally add some new types and levels for metabolomics data
     myDGEobjDef &lt;- initDGEobjDef(levels = "metabolomics",
                                  primaryAssayNames = "intensity",
                                  types &lt;- c(normalizedIntensity = "assay"))

    # When a new level is defined, the itemNames and types for the
    # rowData and colData are automatically established.  The
    # types argument is only needed to define downstream workflow objects.

</code></pre>

<hr>
<h2 id='inventory'>Retrieve the object inventory</h2><span id='topic+inventory'></span>

<h3>Description</h3>

<p>Retrieve the object inventory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inventory(dgeObj, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inventory_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="inventory_+3A_verbose">verbose</code></td>
<td>
<p>Include funArgs column in the output (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame summarizing the data contained in the DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    inventory(exObj)

</code></pre>

<hr>
<h2 id='newType'>Add a new type definition to a DGEobj</h2><span id='topic+newType'></span>

<h3>Description</h3>

<p>Add a new type definition to a DGEobj
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newType(dgeObj, itemType, baseType, uniqueItem = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newType_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="newType_+3A_itemtype">itemType</code></td>
<td>
<p>The name of the new type to create</p>
</td></tr>
<tr><td><code id="newType_+3A_basetype">baseType</code></td>
<td>
<p>The baseType of the new item. One of [row, col, assay, meta]</p>
</td></tr>
<tr><td><code id="newType_+3A_uniqueitem">uniqueItem</code></td>
<td>
<p>If set to TRUE, only one instance of the new type is allowed in a DGEobj</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    exObj &lt;- newType(exObj,
                     itemType   = "AffyRMA",
                     baseType   = "assay",
                     uniqueItem = TRUE)

</code></pre>

<hr>
<h2 id='print.DGEobj'>Print the Inventory</h2><span id='topic+print.DGEobj'></span>

<h3>Description</h3>

<p>Print the Inventory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
print(x, ..., verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.DGEobj_+3A_x">x</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="print.DGEobj_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
<tr><td><code id="print.DGEobj_+3A_verbose">verbose</code></td>
<td>
<p>Include funArgs column in the output (default = FALSE)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    print(exObj)

</code></pre>

<hr>
<h2 id='resetDGEobj'>Reset to original data</h2><span id='topic+resetDGEobj'></span>

<h3>Description</h3>

<p>During a workflow, a DGEobj typically gets filtered down to remove samples
that fail QC or non-expressed genes. The resetDGEobj() function produces a new DGEobj with
the original unfiltered data. Resetting an object does not restore changes to attributes or class,
but does revert changes made with addItem() and rmItem(). Reset requires that *_orig data is still
in the DGEobj.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resetDGEobj(dgeObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resetDGEobj_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    #example object
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    # subset to first 10 rows to show reset functionality
    exObj &lt;- exObj[c(1:10), ]

    exObj &lt;- resetDGEobj(exObj)
    dim(exObj)

</code></pre>

<hr>
<h2 id='rmItem'>Removes a named data item</h2><span id='topic+rmItem'></span>

<h3>Description</h3>

<p>Removes a named data item
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmItem(dgeObj, itemName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmItem_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="rmItem_+3A_itemname">itemName</code></td>
<td>
<p>Name of the item to remove</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    exObj &lt;- rmItem(exObj, "design")

</code></pre>

<hr>
<h2 id='setAttributes'>Set attributes</h2><span id='topic+setAttributes'></span>

<h3>Description</h3>

<p>Set one or more attributes on a DGEobj or on a specific item within a DGEobj.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setAttributes(dgeObj, attribs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setAttributes_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A DGEobj</p>
</td></tr>
<tr><td><code id="setAttributes_+3A_attribs">attribs</code></td>
<td>
<p>A named list of attribute/value pairs</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function adds attributes without deleting the attributes that are already
present. Any named attribute that already exists in the object will be updated.
To remove an attribute from an object pass NULL as the attribute value.
</p>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    # Assign attributes to a DGEobj
    MyAttributes &lt;- list(Platform       = "RNA-Seq",
                         Instrument     = "HiSeq",
                         Vendor         = "Unknown",
                         readType       = "PE",
                         readLength     = 75,
                         strandSpecific = TRUE)
    exObj &lt;- setAttributes(exObj, MyAttributes)

    # Set attributes on an item inside a DGEobj
    MyAttributes &lt;- list(normalized   = FALSE,
                         LowIntFilter = "FPK &gt;5 in &gt;= 1 group")
    exObj[["counts"]] &lt;- setAttributes(exObj[["counts"]], MyAttributes)

</code></pre>

<hr>
<h2 id='showAttributes'>Print attributes</h2><span id='topic+showAttributes'></span>

<h3>Description</h3>

<p>This function prints all attributes regardless of the class of the attribute value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showAttributes(
  dgeObj,
  skipList = c("dim", "dimnames", "rownames", "colnames", "listData", "objDef")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showAttributes_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A DGEobj</p>
</td></tr>
<tr><td><code id="showAttributes_+3A_skiplist">skipList</code></td>
<td>
<p>A character vector of attributes to skip. Use this to avoid
printing certain lengthy attributes like rownames.  Defaults to c(&quot;dim&quot;,
&quot;dimnames&quot;, &quot;rownames&quot;, &quot;colnames&quot;, &quot;listData&quot;, &quot;objDef&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>*Note* Use showMeta() to only retrieve attributes that are key/value pairs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

   showAttributes(exObj)

</code></pre>

<hr>
<h2 id='showMeta'>Retrieve the Key/Value metadata attributes that have a character value and length of 1</h2><span id='topic+showMeta'></span>

<h3>Description</h3>

<p>Retrieve the Key/Value metadata attributes that have a character value and length of 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showMeta(dgeObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showMeta_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A DGEobj with attributes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with &quot;Attribute&quot; and &quot;Value&quot; columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    showMeta(exObj)

</code></pre>

<hr>
<h2 id='showTypes'>Returns and prints the list of all defined types</h2><span id='topic+showTypes'></span>

<h3>Description</h3>

<p>Returns and prints the list of all defined types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showTypes(dgeObj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showTypes_+3A_dgeobj">dgeObj</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    showTypes(exObj)

</code></pre>

<hr>
<h2 id='subset.DGEobj'>Subset internal row or column data</h2><span id='topic+subset.DGEobj'></span>

<h3>Description</h3>

<p>Subset internal row or column data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEobj'
subset(x, ..., row, col, drop = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.DGEobj_+3A_x">x</code></td>
<td>
<p>A class DGEobj created by function initDGEobj()</p>
</td></tr>
<tr><td><code id="subset.DGEobj_+3A_...">...</code></td>
<td>
<p>Additional parameters</p>
</td></tr>
<tr><td><code id="subset.DGEobj_+3A_row">row</code></td>
<td>
<p>Row index for the subset</p>
</td></tr>
<tr><td><code id="subset.DGEobj_+3A_col">col</code></td>
<td>
<p>Col index for the subset</p>
</td></tr>
<tr><td><code id="subset.DGEobj_+3A_drop">drop</code></td>
<td>
<p>Included for compatibility only</p>
</td></tr>
<tr><td><code id="subset.DGEobj_+3A_debug">debug</code></td>
<td>
<p>(default = FALSE) Set to TRUE to get additional information on the console if subsetting a DGEobj fails with a dimension error.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEobj
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # example DGEobj
    exObj &lt;- readRDS(system.file("miniObj.RDS", package = "DGEobj"))

    exObj &lt;- subset(exObj, 1:10, 5:50)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
