<!DOCTYPE html><html lang="en"><head><title>Help for package DGP4LCF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DGP4LCF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DGP4LCF-package'><p>DGP4LCF: Dependent Gaussian Processes for Longitudinal Correlated Factors</p></a></li>
<li><a href='#factor_loading_heatmap'><p>Displaying significant factor loadings in the heatmap.</p></a></li>
<li><a href='#factor_score_trajectory'><p>Plotting figures for factor score trajectory.</p></a></li>
<li><a href='#gibbs_after_mcem_algorithm'><p>Generating posterior samples for parameters (other than DGP parameters) in the model and predicted gene expression for one chain.</p></a></li>
<li><a href='#gibbs_after_mcem_combine_chains'><p>Combining from all chains the posterior samples for parameters in the model and predicted gene expressions.</p></a></li>
<li><a href='#gibbs_after_mcem_diff_initials'><p>Generating different initials for multiple chains.</p></a></li>
<li><a href='#gibbs_after_mcem_load_chains'><p>Loading the saved posterior samples for parameters in the model and predicted gene expressions.</p></a></li>
<li><a href='#mcem_algorithm'><p>Monte Carlo Expectation Maximization (MCEM) algorithm to return the Maximum Likelihood Estimate (MLE) of DGP Parameters.</p></a></li>
<li><a href='#mcem_cov_plot'><p>Visualizing cross-correlations among factors.</p></a></li>
<li><a href='#mcem_parameter_setup'><p>Parameters' setup and initial value assignment for the Monte Carlo Expectation Maximization (MCEM) algorithm.</p></a></li>
<li><a href='#numerics_summary_do_not_need_alignment'><p>Numerical summary for important continuous variables that do not need alignment.</p></a></li>
<li><a href='#numerics_summary_need_alignment'><p>Numerical summary for factor loadings and factor scores, which need alignment.</p></a></li>
<li><a href='#sim_fcs_init'><p>Initials values.</p></a></li>
<li><a href='#sim_fcs_results_irregular_6_8'><p>Results when people have irregularly observed time points (some 6 while others 8).</p></a></li>
<li><a href='#sim_fcs_results_regular_8'><p>Results when people are observed at common 8 time points.</p></a></li>
<li><a href='#sim_fcs_truth'><p>Truth of simulated data.</p></a></li>
<li><a href='#subject_specific_objects'><p>Constructing subject-specific objects required for Gibbs sampler (for subjects with incomplete observations only).</p></a></li>
<li><a href='#table_generator'><p>Generating a table listing all possible combinations of the binary variables for one gene.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Dependent Gaussian Processes for Longitudinal Correlated Factors</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiachen Cai &lt;jiachen.cai@mrc-bsu.cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functionalities for analyzing high-dimensional and longitudinal biomarker data to facilitate precision medicine, using a joint model of Bayesian sparse factor analysis and dependent Gaussian processes. This paper illustrates the method in detail: J Cai, RJB Goudie, C Starr, BDM Tom (2023) &lt;<a href="https://doi.org/10.48550%2FarXiv.2307.02781">doi:10.48550/arXiv.2307.02781</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>GPFDA, Rcpp, factor.switching, mvtnorm, combinat, coda,
corrplot, pheatmap, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-08 07:11:11 UTC; hornik</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiachen Cai [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-08 08:58:39 UTC</td>
</tr>
</table>
<hr>
<h2 id='DGP4LCF-package'>DGP4LCF: Dependent Gaussian Processes for Longitudinal Correlated Factors</h2><span id='topic+DGP4LCF'></span><span id='topic+DGP4LCF-package'></span>

<h3>Description</h3>

<p>This package implements a novel methodology to model high-dimensional gene expression trajectories.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: 
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Jiachen Cai <a href="mailto:jiachen.cai@mrc-bsu.cam.ac.uk">jiachen.cai@mrc-bsu.cam.ac.uk</a>
</p>
</li></ul>


<hr>
<h2 id='factor_loading_heatmap'>Displaying significant factor loadings in the heatmap.</h2><span id='topic+factor_loading_heatmap'></span>

<h3>Description</h3>

<p>This function is used to visualize results of estimates of factor loadings (in heatmaps).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>factor_loading_heatmap(factor_loading_matrix, heatmap_title)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="factor_loading_heatmap_+3A_factor_loading_matrix">factor_loading_matrix</code></td>
<td>
<p>A matrix of dimension (p, k), which stores results for factor loadings.</p>
</td></tr>
<tr><td><code id="factor_loading_heatmap_+3A_heatmap_title">heatmap_title</code></td>
<td>
<p>A character. Title for the heatmap.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heatmap presenting posterior median estimates of factor loadings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='factor_score_trajectory'>Plotting figures for factor score trajectory.</h2><span id='topic+factor_score_trajectory'></span>

<h3>Description</h3>

<p>This function is used to visualize results of factor score trajectories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>factor_score_trajectory(
  factor_score_matrix,
  factor_index,
  person_index,
  trajectory_title,
  cex_main = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="factor_score_trajectory_+3A_factor_score_matrix">factor_score_matrix</code></td>
<td>
<p>A matrix of dimension (q, k, n), used to store results for factor scores.</p>
</td></tr>
<tr><td><code id="factor_score_trajectory_+3A_factor_index">factor_index</code></td>
<td>
<p>A numeric scalar. Index of the factor of interest.</p>
</td></tr>
<tr><td><code id="factor_score_trajectory_+3A_person_index">person_index</code></td>
<td>
<p>A numeric scalar. Index of the person of interest.</p>
</td></tr>
<tr><td><code id="factor_score_trajectory_+3A_trajectory_title">trajectory_title</code></td>
<td>
<p>A character. Title for the factor trajectory plot.</p>
</td></tr>
<tr><td><code id="factor_score_trajectory_+3A_cex_main">cex_main</code></td>
<td>
<p>A numeric scalar. Text size of the title.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Trajectory of the designated person-factor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='gibbs_after_mcem_algorithm'>Generating posterior samples for parameters (other than DGP parameters) in the model and predicted gene expression for one chain.</h2><span id='topic+gibbs_after_mcem_algorithm'></span>

<h3>Description</h3>

<p>Generating posterior samples for parameters (other than DGP parameters) in the model and predicted gene expression for one chain.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gibbs_after_mcem_algorithm(
  chain_index,
  mc_num,
  burnin,
  thin_step,
  pathname,
  pred_indicator = FALSE,
  pred_time_index = NULL,
  x,
  mcem_parameter_setup_result,
  mcem_algorithm_result,
  gibbs_after_mcem_diff_initials_result
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_chain_index">chain_index</code></td>
<td>
<p>A numeric scalar. Index of the chain.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_mc_num">mc_num</code></td>
<td>
<p>A numeric scalar. Number of iterations in the Gibbs sampler.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_burnin">burnin</code></td>
<td>
<p>A numeric scalar. Number of iterations to be discarded as 'burn-in'.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_thin_step">thin_step</code></td>
<td>
<p>A numeric scalar. This function will only save every 'thin_step'th iteration results in the specified directory to reduce storage space needed. Note that this number can be different from that used in the function 'mcem_algorithm'.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_pathname">pathname</code></td>
<td>
<p>A character. The directory where the saved Gibbs samplers are stored.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_pred_indicator">pred_indicator</code></td>
<td>
<p>A logical value. pred_indicator = TRUE denotes the need to predict gene expression at new time points. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_pred_time_index">pred_time_index</code></td>
<td>
<p>Only needed if pred_indicator = TRUE. Index of the new time points in the full time vector.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_x">x</code></td>
<td>
<p>A list of n elements. Each element is a matrix of dimension (p, q_i), storing the gene expression observed at q_i time points for the ith subject.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_mcem_parameter_setup_result">mcem_parameter_setup_result</code></td>
<td>
<p>A list of objects returned from the function 'mcem_parameter_setup'.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_mcem_algorithm_result">mcem_algorithm_result</code></td>
<td>
<p>A list of objects returned from the function 'mcem_algorithm'.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_algorithm_+3A_gibbs_after_mcem_diff_initials_result">gibbs_after_mcem_diff_initials_result</code></td>
<td>
<p>A list of objects returned from the function 'gibbs_after_mcem_diff_initials'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function corresponds to Algorithm 2: Step 1 in the main manuscript; therefore reader can consult the paper for more explanations.
</p>


<h3>Value</h3>

<p>Posterior samples for parameters (other than DGP parameters) in the model and predicted gene expression for one chain.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='gibbs_after_mcem_combine_chains'>Combining from all chains the posterior samples for parameters in the model and predicted gene expressions.</h2><span id='topic+gibbs_after_mcem_combine_chains'></span>

<h3>Description</h3>

<p>Combining from all chains the posterior samples for parameters in the model and predicted gene expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gibbs_after_mcem_combine_chains(tot_chain, gibbs_after_mcem_algorithm_result)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gibbs_after_mcem_combine_chains_+3A_tot_chain">tot_chain</code></td>
<td>
<p>A numeric scalar. Total number of chains.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_combine_chains_+3A_gibbs_after_mcem_algorithm_result">gibbs_after_mcem_algorithm_result</code></td>
<td>
<p>A list of objects storing model constants. Should be the same as that input to the 'function gibbs_after_mcem_load_chains'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>All saved posterior samples for parameters in the model and predicted gene expressions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='gibbs_after_mcem_diff_initials'>Generating different initials for multiple chains.</h2><span id='topic+gibbs_after_mcem_diff_initials'></span>

<h3>Description</h3>

<p>Generating different initials for multiple chains.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gibbs_after_mcem_diff_initials(
  ind_x = TRUE,
  tot_chain = 5,
  mcem_parameter_setup_result,
  mcem_algorithm_result
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gibbs_after_mcem_diff_initials_+3A_ind_x">ind_x</code></td>
<td>
<p>A logical value. ind_x = TRUE uses the model including the intercept term for subject-gene mean in after-MCEM-Gibbs sampler; otherwise uses the model without the intercept term.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_diff_initials_+3A_tot_chain">tot_chain</code></td>
<td>
<p>A numeric scalar. Number of parallel chains.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_diff_initials_+3A_mcem_parameter_setup_result">mcem_parameter_setup_result</code></td>
<td>
<p>A list of objects returned from the function 'mcem_parameter_setup'.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_diff_initials_+3A_mcem_algorithm_result">mcem_algorithm_result</code></td>
<td>
<p>A list of objects returned from the function 'mcem_algorithm'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Different initials for multiple chains.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='gibbs_after_mcem_load_chains'>Loading the saved posterior samples for parameters in the model and predicted gene expressions.</h2><span id='topic+gibbs_after_mcem_load_chains'></span>

<h3>Description</h3>

<p>Loading the saved posterior samples for parameters in the model and predicted gene expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gibbs_after_mcem_load_chains(chain_index, gibbs_after_mcem_algorithm_result)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gibbs_after_mcem_load_chains_+3A_chain_index">chain_index</code></td>
<td>
<p>A numeric scalar. Index of the chain.</p>
</td></tr>
<tr><td><code id="gibbs_after_mcem_load_chains_+3A_gibbs_after_mcem_algorithm_result">gibbs_after_mcem_algorithm_result</code></td>
<td>
<p>A list of objects storing model constants.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>All saved posterior samples for parameters in the model and predicted gene expressions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example",  package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='mcem_algorithm'>Monte Carlo Expectation Maximization (MCEM) algorithm to return the Maximum Likelihood Estimate (MLE) of DGP Parameters.</h2><span id='topic+mcem_algorithm'></span>

<h3>Description</h3>

<p>This function is used to return the MLE of DGP parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcem_algorithm(
  ind_x,
  ig_parameter = 10^-2,
  increasing_rate = 0.5,
  prob_conf_interval = 0.9,
  iter_count_num = 5,
  x,
  mcem_parameter_setup_result,
  ipt_x = FALSE,
  missing_list = NULL,
  missing_num = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcem_algorithm_+3A_ind_x">ind_x</code></td>
<td>
<p>A logical value. ind_x = TRUE uses the model including the intercept term for subject-gene mean in within-MCEM-Gibbs sampler; otherwise uses the model without the intercept term.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_ig_parameter">ig_parameter</code></td>
<td>
<p>A numeric scalar. Hyper-parameters for the prior Inverse-Gamma distribution.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_increasing_rate">increasing_rate</code></td>
<td>
<p>A numeric scalar. Rate of increasing the sample size.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_prob_conf_interval">prob_conf_interval</code></td>
<td>
<p>A numeric scalar. The probability that the true change in the Q-function is larger than the lower bound.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_iter_count_num">iter_count_num</code></td>
<td>
<p>A numeric scalar. Maximum number of increasing the sample size; a larger number than this would end the algorithm.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_x">x</code></td>
<td>
<p>A list of n elements. Each element is a matrix of dimension (p, q_i), storing the gene expression observed at q_i time points for the ith subject.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_mcem_parameter_setup_result">mcem_parameter_setup_result</code></td>
<td>
<p>A list of objects returned from the function 'mcem_parameter_setup'.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_ipt_x">ipt_x</code></td>
<td>
<p>A logical value. ind_x = TRUE denotes the need to impute for NAs of gene expression. The default value is ind_x = FALSE.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_missing_list">missing_list</code></td>
<td>
<p>A list of n elements. Each element is a matrix of dimension (missing_num, 2): each row corresponds to the position of one NA that needs imputation; first and second columns denote the row and column indexes, respectively, of the NA in the corresponding person's matrix of gene expression.</p>
</td></tr>
<tr><td><code id="mcem_algorithm_+3A_missing_num">missing_num</code></td>
<td>
<p>A vector of n elements. Each element corresponds to a single person's number of NAs that needs imputation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MLE of DGP parameters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='mcem_cov_plot'>Visualizing cross-correlations among factors.</h2><span id='topic+mcem_cov_plot'></span>

<h3>Description</h3>

<p>Visualizing cross-correlations among factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcem_cov_plot(k, q, cov_input, title)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcem_cov_plot_+3A_k">k</code></td>
<td>
<p>A numeric scalar. Number of latent factors.</p>
</td></tr>
<tr><td><code id="mcem_cov_plot_+3A_q">q</code></td>
<td>
<p>A numeric scalar. Number of time points in the covariance matrix of factors.</p>
</td></tr>
<tr><td><code id="mcem_cov_plot_+3A_cov_input">cov_input</code></td>
<td>
<p>A matrix of dimension (kq, kq). The covariance matrix of the vector obtained from vectorizing the matrix of latent factor scores.</p>
</td></tr>
<tr><td><code id="mcem_cov_plot_+3A_title">title</code></td>
<td>
<p>A character. Title for the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Visualization of cross-correlations among factors.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='mcem_parameter_setup'>Parameters' setup and initial value assignment for the Monte Carlo Expectation Maximization (MCEM) algorithm.</h2><span id='topic+mcem_parameter_setup'></span>

<h3>Description</h3>

<p>This function is used to create R objects storing parameters in the desired format, and assign initial values so that they are ready to use in the MCEM algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcem_parameter_setup(
  p,
  k,
  n,
  q,
  ind_num = 10,
  burn_in_prop = 0.2,
  thin_step = 5,
  prior_sparsity = 0.1,
  em_num = 50,
  obs_time_num,
  obs_time_index,
  a_person,
  col_person_index,
  y_init,
  a_init,
  z_init,
  phi_init,
  a_full,
  train_index,
  x,
  model_dgp = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcem_parameter_setup_+3A_p">p</code></td>
<td>
<p>A numeric scalar. Number of genes.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_k">k</code></td>
<td>
<p>A numeric scalar. Number of latent factors.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_n">n</code></td>
<td>
<p>A numeric scalar. Number of subjects.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_q">q</code></td>
<td>
<p>A numeric scalar. Complete number of time points in the training data.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_ind_num">ind_num</code></td>
<td>
<p>A numeric scalar. Starting size of approximately independent samples for MCEM.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_burn_in_prop">burn_in_prop</code></td>
<td>
<p>A numeric scalar. Proportion of burnin, which be used to calculate size of Monte Carlo samples needed in the Gibbs sampler. Must be the same as that in the function 'mcem_algorithm_irregular_time'.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_thin_step">thin_step</code></td>
<td>
<p>A numeric scalar. Thinning step, which be used to calculate size of Monte Carlo samples needed in the Gibbs sampler. Must be the same as that in the function 'mcem_algorithm_irregular_time'.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_prior_sparsity">prior_sparsity</code></td>
<td>
<p>A numeric scalar. Prior expected proportion of genes involved within each pathway.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_em_num">em_num</code></td>
<td>
<p>A numeric scalar. Maximum iterations of the expectation maximization (EM) algorithm allowed.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_obs_time_num">obs_time_num</code></td>
<td>
<p>A n-dimensional vector. One element represents one person's observed number of time points in the training data.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_obs_time_index">obs_time_index</code></td>
<td>
<p>A list of n elements. One element is a vector of observed time indexes for one person in the training data, sorted from early to late.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_a_person">a_person</code></td>
<td>
<p>A list of n elements. One element is a vector of observed time for one subject in the training data, sorted from early to late.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_col_person_index">col_person_index</code></td>
<td>
<p>A list of n elements. One element is a vector of column indexes for one subject in y_init.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_y_init">y_init</code></td>
<td>
<p>A matrix of dimension (k, sum(obs_time_num)). Initial values of the latent factor score. Can be obtained using BFRM software.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_a_init">a_init</code></td>
<td>
<p>A matrix of dimension (p, k). Initial values of the regression coefficients of factor loadings. Can be obtained using BFRM software.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_z_init">z_init</code></td>
<td>
<p>A matrix of dimension (p, k). Initials values of the binary variables of factor loadings. Can be obtained using BFRM software.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_phi_init">phi_init</code></td>
<td>
<p>A p-dimensional column vector. Initials values of the variance for residuals when modeling gene expressions, corresponding to <code class="reqn">\frac{1}{\phi^2}</code> in the manuscript. Can be obtained using BFRM software.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_a_full">a_full</code></td>
<td>
<p>A numeric vector. Complete time observed, sorted from early to late.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_train_index">train_index</code></td>
<td>
<p>A q-dimensional column vector. Index of time points used in the training data.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_x">x</code></td>
<td>
<p>A list of n elements. Each element is a matrix of dimension (p, q_i), storing the gene expressions for the ith subject.</p>
</td></tr>
<tr><td><code id="mcem_parameter_setup_+3A_model_dgp">model_dgp</code></td>
<td>
<p>A logical value. model_dgp = TRUE (default setting) uses the Dependent Gaussian Process to model latent factor trajectories, otherwise the Independent Gaussian Process is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following parameters are worth particular attention, and users should tune these parameters according to the specific data.
</p>
<p>'burn_in_prop' and 'thin_step' co-control the number of Gibbs samples needed in order to generate approximately 'ind_num' independent samples.
The ultimate purpose of tuning these two parameters is to generate high-quality posterior samples for latent factor scores.
Therefore: if initials of the Gibbs sampler are not good, readers may need to increase 'burn_in_prop' to discard more burn-in samples; if high-correlation is a potential concern, 'thin_step' may need to be larger.
</p>


<h3>Value</h3>

<p>A list of R objects required in the MCEM algorithm.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='numerics_summary_do_not_need_alignment'>Numerical summary for important continuous variables that do not need alignment.</h2><span id='topic+numerics_summary_do_not_need_alignment'></span>

<h3>Description</h3>

<p>Numerical summary for important continuous variables that do not need alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numerics_summary_do_not_need_alignment(
  burnin = 0,
  thin_step = 1,
  pred_x_truth_indicator = FALSE,
  pred_x_truth = NULL,
  gibbs_after_mcem_combine_chains_result
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="numerics_summary_do_not_need_alignment_+3A_burnin">burnin</code></td>
<td>
<p>A numeric scalar. The saved samples are already after burnin; therefore the default value for this parameter here is 0. Can discard further samples if needed.</p>
</td></tr>
<tr><td><code id="numerics_summary_do_not_need_alignment_+3A_thin_step">thin_step</code></td>
<td>
<p>A numeric scalar. The saved samples are already after thinning; therefore the default value for this parameter here is 1. Can be further thinned if needed.</p>
</td></tr>
<tr><td><code id="numerics_summary_do_not_need_alignment_+3A_pred_x_truth_indicator">pred_x_truth_indicator</code></td>
<td>
<p>A logical value. pred_x_truth_indicator = TRUE means that truth of predicted gene expressions are available. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="numerics_summary_do_not_need_alignment_+3A_pred_x_truth">pred_x_truth</code></td>
<td>
<p>Only needed if pred_x_truth_inidcator = TRUE. An array of dimension (n, p, num_time_test), storing true gene expressions in the testing data.</p>
</td></tr>
<tr><td><code id="numerics_summary_do_not_need_alignment_+3A_gibbs_after_mcem_combine_chains_result">gibbs_after_mcem_combine_chains_result</code></td>
<td>
<p>A list of objects returned from the function 'gibbs_after_mcem_combine_chains'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function corresponds to Algorithm 2: Steps 3 and 4 in the main manuscript; therefore reader can consult the paper for more explanations.
</p>


<h3>Value</h3>

<p>Convergence assessment for important continuous variables that do not need alignment, and posterior summary for predicted gene expressions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example",  package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='numerics_summary_need_alignment'>Numerical summary for factor loadings and factor scores, which need alignment.</h2><span id='topic+numerics_summary_need_alignment'></span>

<h3>Description</h3>

<p>Numerical summary for factor loadings and factor scores, which need alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numerics_summary_need_alignment(
  burnin = 0,
  thin_step = 1,
  gibbs_after_mcem_combine_chains_result
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="numerics_summary_need_alignment_+3A_burnin">burnin</code></td>
<td>
<p>A numeric scalar. The saved samples are already after burnin; therefore the default value for this parameter here is 0. Can discard further samples if needed.</p>
</td></tr>
<tr><td><code id="numerics_summary_need_alignment_+3A_thin_step">thin_step</code></td>
<td>
<p>A numeric scalar. The saved samples are already after thinning; therefore the default value for this parameter here is 1. Can be further thinned if needed.</p>
</td></tr>
<tr><td><code id="numerics_summary_need_alignment_+3A_gibbs_after_mcem_combine_chains_result">gibbs_after_mcem_combine_chains_result</code></td>
<td>
<p>A list of objects returned from the function 'gibbs_after_mcem_combine_chains'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function corresponds to Algorithm 2: Steps 2, 3 and 4 in the main manuscript; therefore reader can consult the paper for more explanations.
</p>


<h3>Value</h3>

<p>Reordered posterior samples, convergence assessment, and summarized posterior results for factor loadings and factor scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='sim_fcs_init'>Initials values.</h2><span id='topic+sim_fcs_init'></span>

<h3>Description</h3>

<p>Initial values provided by the two-step approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_fcs_init
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 14.
</p>

<hr>
<h2 id='sim_fcs_results_irregular_6_8'>Results when people have irregularly observed time points (some 6 while others 8).</h2><span id='topic+sim_fcs_results_irregular_6_8'></span>

<h3>Description</h3>

<p>Results when people have irregularly observed time points (some 6 while others 8).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_fcs_results_irregular_6_8
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 3.
</p>

<hr>
<h2 id='sim_fcs_results_regular_8'>Results when people are observed at common 8 time points.</h2><span id='topic+sim_fcs_results_regular_8'></span>

<h3>Description</h3>

<p>Results when people are observed at common 8 time points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_fcs_results_regular_8
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 3.
</p>

<hr>
<h2 id='sim_fcs_truth'>Truth of simulated data.</h2><span id='topic+sim_fcs_truth'></span>

<h3>Description</h3>

<p>Simulated data under the scenario where factors are correlated and have small variability (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_fcs_truth
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 19.
</p>

<hr>
<h2 id='subject_specific_objects'>Constructing subject-specific objects required for Gibbs sampler (for subjects with incomplete observations only).</h2><span id='topic+subject_specific_objects'></span>

<h3>Description</h3>

<p>Constructing subject-specific objects required for Gibbs sampler (for subjects with incomplete observations only).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subject_specific_objects(k, q, a_full, a_avail, cor_all)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subject_specific_objects_+3A_k">k</code></td>
<td>
<p>A numeric scalar. Number of latent factors.</p>
</td></tr>
<tr><td><code id="subject_specific_objects_+3A_q">q</code></td>
<td>
<p>A numeric scalar. Number of time points in the complete factor covariance matrix.</p>
</td></tr>
<tr><td><code id="subject_specific_objects_+3A_a_full">a_full</code></td>
<td>
<p>A q-dimensional numeric vector. Complete time sorted from early to late.</p>
</td></tr>
<tr><td><code id="subject_specific_objects_+3A_a_avail">a_avail</code></td>
<td>
<p>A vector of time when gene expressions are available, sorted from early to late.</p>
</td></tr>
<tr><td><code id="subject_specific_objects_+3A_cor_all">cor_all</code></td>
<td>
<p>A matrix of dimension (kq, kq). Correlation matrix of latent factor scores.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to extract subject-specific factor covariance matrix from the complete factor covariance matrix, through constructing subject-specific indicator matrix, which indicates time indexes when gene expression are available.
</p>


<h3>Value</h3>

<p>Subject-specific objects needed for Gibbs sampler.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

<hr>
<h2 id='table_generator'>Generating a table listing all possible combinations of the binary variables for one gene.</h2><span id='topic+table_generator'></span>

<h3>Description</h3>

<p>Generating a table listing all possible combinations of the binary variables for one gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table_generator(k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="table_generator_+3A_k">k</code></td>
<td>
<p>A numeric scalar. Number of latent factors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table listing all possible combinations of the binary variables for one gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples in vignette
vignette("bsfadgp_regular_data_example", package = "DGP4LCF")
vignette("bsfadgp_irregular_data_example", package = "DGP4LCF")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
