<!DOCTYPE html><html><head><title>Help for package diemr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {diemr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CheckDiemFormat'><p>diem input file checker</p></a></li>
<li><a href='#diem'><p>Diagnostic Index Expectation Maximisation</p></a></li>
<li><a href='#emPolarise'><p>Polarises a marker</p></a></li>
<li><a href='#importPolarized'><p>Imports genomic data polarized according to the specification</p></a></li>
<li><a href='#ModelOfDiagnostic'><p>Model of Diagnostic Marker Based on All Individual State Counts</p></a></li>
<li><a href='#myotis'><p>Dataset of modified genotypes of bats</p></a></li>
<li><a href='#pHetErrOnStateCount'><p>Hybrid index, heterozygosity, error rate</p></a></li>
<li><a href='#plotPolarized'><p>Plots polarized genotypes</p></a></li>
<li><a href='#sStateCount'><p>Count states in a vector</p></a></li>
<li><a href='#vcf2diem'><p>Convert vcf files to diem format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Diagnostic Index Expectation Maximisation in R</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Likelihood-based genome polarisation finds which alleles of genomic markers 
    belong to which side of the barrier. 
	Co-estimates which individuals belong to either side of the barrier and barrier strength.
	Uses expectation maximisation in likelihood framework. The method is described in 
	Baird et al. (2023) &lt;<a href="https://doi.org/10.1111%2F2041-210X.14010">doi:10.1111/2041-210X.14010</a>&gt;.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/StuartJEBaird/diem/issues">https://github.com/StuartJEBaird/diem/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>zoo, vcfR</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-11 06:31:39 UTC; martinkova</td>
</tr>
<tr>
<td>Author:</td>
<td>Natalia Martinkova
    <a href="https://orcid.org/0000-0003-4556-4363"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Stuart Baird <a href="https://orcid.org/0000-0002-7144-9919"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Natalia Martinkova &lt;martinkova@ivb.cz&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-11 12:00:20 UTC</td>
</tr>
</table>
<hr>
<h2 id='CheckDiemFormat'>diem input file checker</h2><span id='topic+CheckDiemFormat'></span>

<h3>Description</h3>

<p>Checks format of files with genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckDiemFormat(files, ChosenInds, ploidy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckDiemFormat_+3A_files">files</code></td>
<td>
<p>character vector with paths to files with genotypes.</p>
</td></tr>
<tr><td><code id="CheckDiemFormat_+3A_choseninds">ChosenInds</code></td>
<td>
<p>numeric vector of indices of individuals to be included in the analysis.</p>
</td></tr>
<tr><td><code id="CheckDiemFormat_+3A_ploidy">ploidy</code></td>
<td>
<p>list of length equal to length of <code>files</code>. Each element of the list
contains a numeric vector with ploidy numbers for all individuals specified in
the <code>ChosenInds</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input file must have genotypes of one marker for all individuals on one
line. The line must start with a letter &quot;S&quot; and contain only characters
&quot;_&quot; or &quot;U&quot; for unknown genotypes or a third/fourth allele, &quot;0&quot; for homozygots for
allele 1, &quot;1&quot; for heterozygots, and &quot;2&quot; for homozygots for allele 2. Check the
vignette with <code>browseVignettes(package = "diemr")</code> for the example of the
input format.
</p>
<p>Ploidies must be given as a list with each element corresponding to a genomic
compartment (aka a file). For each compartment, the numeric vector specifying
ploidies of all individuals chosen for the specific analysis must be given.
</p>


<h3>Value</h3>

<p>Returns invisible <code>TRUE</code> if all files are executable by <code>diem</code>. Exits
with informative error messages otherwise, specifying file names and lines with
potential problems. When too many lines contain problems, the first six are given.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set up input genotypes file names, ploidies and selection of individual samples
inputFile &lt;- system.file("extdata", "data6x3.txt", package = "diemr")
ploidies &lt;- list(c(2, 1, 2, 2, 2, 1))
inds &lt;- 1:6

# check input data
CheckDiemFormat(files = inputFile, ploidy = ploidies, ChosenInds = inds)
#  File check passed: TRUE
#  Ploidy check passed: TRUE
</code></pre>

<hr>
<h2 id='diem'>Diagnostic Index Expectation Maximisation</h2><span id='topic+diem'></span>

<h3>Description</h3>

<p>Estimates how to assign alleles in a genome to maximise the distinction between two
unknown groups of individuals. Using expectation maximisation (EM) in likelihood
framework, <code>diem</code> provides marker
polarities for importing data, their likelihood-based diagnostic index and its support
for all markers, and hybrid indices for all individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diem(
  files,
  ploidy = list(2),
  markerPolarity = FALSE,
  ChosenInds,
  epsilon = 0.99999,
  verbose = FALSE,
  nCores = parallel::detectCores() - 1,
  maxIterations = 50,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diem_+3A_files">files</code></td>
<td>
<p>character vector with paths to files with genotypes.</p>
</td></tr>
<tr><td><code id="diem_+3A_ploidy">ploidy</code></td>
<td>
<p>list of length equal to length of <code>files</code>. Each element of the list
contains a numeric vector with ploidy numbers for all individuals specified in
the <code>ChosenInds</code>.</p>
</td></tr>
<tr><td><code id="diem_+3A_markerpolarity">markerPolarity</code></td>
<td>
<p><code>FALSE</code> or list of logical vectors.</p>
</td></tr>
<tr><td><code id="diem_+3A_choseninds">ChosenInds</code></td>
<td>
<p>numeric vector of indices of individuals to be included in the analysis.</p>
</td></tr>
<tr><td><code id="diem_+3A_epsilon">epsilon</code></td>
<td>
<p>numeric, specifying how much the hypothetical diagnostic markers should
contribute to the likelihood calculations. Must be in <code>[0,1)</code>, keeping
tolerance setting of the <code>R</code> session in mind.</p>
</td></tr>
<tr><td><code id="diem_+3A_verbose">verbose</code></td>
<td>
<p>logical or character with path to directory where run diagnostics will
be saved.</p>
</td></tr>
<tr><td><code id="diem_+3A_ncores">nCores</code></td>
<td>
<p>numeric. Number of cores to be used for parallelisation. Must be
<code>nCores = 1</code> on Windows.</p>
</td></tr>
<tr><td><code id="diem_+3A_maxiterations">maxIterations</code></td>
<td>
<p>numeric.</p>
</td></tr>
<tr><td><code id="diem_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given two alleles of a marker, one allele can belong to one side of a barrier
to geneflow and the other to the other side. Which allele belongs where is a non-trivial
matter. A marker state in an individual can be encoded as 0 if the individual is
homozygous for the first allele, and 2 if the individual is homozygous for the second
allele. Marker polarity determines how the marker will be imported. Marker polarity
equal to <code>FALSE</code> means that the marker will be imported as-is. A marker with
polarity equal to <code>TRUE</code> will be imported with states 0 mapped as 2 and states 2
mapped as 0, in effect switching which allele belongs to which side of a barrier to
geneflow.
</p>
<p>When <code>markerPolarity = FALSE</code>, <code>diem</code> uses random null polarities to
initiate the EM algorithm. To fix the null polarities, <code>markerPolarity</code> must be
a list of length equal to the length of the <code>files</code> argument, where each element
in the list is a logical vector of length equal to the number of markers (rows) in
the specific file.
</p>
<p>Ploidy needs to be given for each compartment and for each individual. For example,
for a dataset of three diploid mammal males consisting of an autosomal
compartment, an X chromosome
compartment and a Y chromosome compartment, the ploidy list would be
<code>ploidy = list(rep(2, 3), rep(1, 3), rep(1, 3)</code>. If the dataset consisted of
one male and two females,
ploidy for the sex chromosomes should be vectors reflecting that females have two X
chromosomes, but males only one, and females have no Y chromosomes:
<code>ploidy = list(rep(2, 3), c(1, 2, 2), c(1, 0, 0))</code>.
</p>


<h3>Value</h3>

<p>A list including suggested marker polarities, diagnostic indices and support for all
markers, four genomic state counts matrix for all individuals, and polarity changes
for the EM iterations.
</p>


<h3>Note</h3>

<p>To ensure that the data input format of the genotype files, ploidies and individual
selection are readable for <code>diem</code>, first use <code><a href="#topic+CheckDiemFormat">CheckDiemFormat</a></code>.
Fix all errors, and run <code>diem</code> only once the checks all passed.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CheckDiemFormat">CheckDiemFormat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set up input genotypes file names, ploidies and selection of individual samples
inputFile &lt;- system.file("extdata", "data6x3.txt", package = "diemr")
ploidies &lt;- list(c(2, 1, 2, 2, 2, 1))
inds &lt;- 1:6

# check input data
CheckDiemFormat(files = inputFile, ploidy = ploidies, ChosenInds = inds)
#  File check passed: TRUE
#  Ploidy check passed: TRUE

# run diem
## Not run: 
# diem will store temporal files during EM iterations
# prior to running diem, set the working directory to a location with write permission
fit &lt;- diem(files = inputFile, ChosenInds = inds, ploidy = ploidies, nCores = 1)

# run diem with fixed null polarities
fit2 &lt;- diem(
  files = inputFile, ChosenInds = inds, ploidy = ploidies, nCores = 1,
  markerPolarity = list(c(TRUE, FALSE, TRUE))
)

## End(Not run)
</code></pre>

<hr>
<h2 id='emPolarise'>Polarises a marker</h2><span id='topic+emPolarise'></span>

<h3>Description</h3>

<p>Changes encodings of genomic markers according to user specification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>emPolarise(origM, changePolarity = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="emPolarise_+3A_origm">origM</code></td>
<td>
<p>character vector of genotypes comprising of _012 encodings.</p>
</td></tr>
<tr><td><code id="emPolarise_+3A_changepolarity">changePolarity</code></td>
<td>
<p>logical scalar, indicating whether to leave the marker as is
(<code>FALSE</code>) or whether to change its polarity (<code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector with polarised markers.
</p>


<h3>Note</h3>

<p>Note that <code><a href="#topic+diem">diem</a></code> and <code><a href="#topic+importPolarized">importPolarized</a></code> accept also a <code>U</code>
encoding for an unknown or third allele, but <code>emPolarise</code> requires all <code>U</code> to
be replaced with <code style="white-space: pre;">&#8288;_&#8288;</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diem">diem</a></code> for determining appropriate marker polarity with
respect to a barrier to geneflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>emPolarise(c("0", "0", "1", "2", "2"), TRUE)
# [1] "2" "2" "1" "0" "0"

emPolarise(c("0", "_", "2", "2", "1"), FALSE)
# [1] "0" "_" "2" "2" "1"
</code></pre>

<hr>
<h2 id='importPolarized'>Imports genomic data polarized according to the specification</h2><span id='topic+importPolarized'></span>

<h3>Description</h3>

<p>Reads genotypes from a file and changes marker polarity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importPolarized(file, changePolarity, ChosenInds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importPolarized_+3A_file">file</code></td>
<td>
<p>character vector with a single path to a file with genotypes.</p>
</td></tr>
<tr><td><code id="importPolarized_+3A_changepolarity">changePolarity</code></td>
<td>
<p>logical vector with length equal to the number of markers.</p>
</td></tr>
<tr><td><code id="importPolarized_+3A_choseninds">ChosenInds</code></td>
<td>
<p>numeric vector of indices of individuals to be included in the analysis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For details on the input data format, check the <code>file</code> with
<code>CheckDiemFormat</code>.
</p>
<p>The <code>changePolarity</code> argument influences how each marker is imported. Value
<code>FALSE</code> means that the marker will be imported as it is saved in the <code>file</code>. Value
<code>TRUE</code> means that the genotypes encoded as <code>0</code> will be imported as <code>2</code>, and genotypes
encoded in the <code>file</code> as <code>2</code> will be imported as <code>0</code>.
</p>


<h3>Value</h3>

<p>Returns a character matrix with rows containing individual genotypes and columns
containing markers.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diem">diem</a></code> for determining appropriate marker polarity with
respect to a barrier to geneflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- importPolarized(
  file = system.file("extdata", "data6x3.txt", package = "diemr"),
  changePolarity = c(FALSE, TRUE, TRUE),
  ChosenInds = 1:6
)
dat
#    m1  m2  m3
# 1 "0" "1" "2"
# 2 "0" "0" "0"
# 3 "1" "1" "0"
# 4 "1" "2" "0"
# 5 "2" "2" "1"
# 6 "2" "2" "_"
</code></pre>

<hr>
<h2 id='ModelOfDiagnostic'>Model of Diagnostic Marker Based on All Individual State Counts</h2><span id='topic+ModelOfDiagnostic'></span>

<h3>Description</h3>

<p>Estimates a diagnostic marker for the state counts of all genomic markers for all
individuals. Using the hypothetical, diagnostic marker, calculates individual state
counts with respect to their weighted similarity to the diagnostic marker states.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ModelOfDiagnostic(I4, OriginalHI, epsilon = 0.99, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ModelOfDiagnostic_+3A_i4">I4</code></td>
<td>
<p>a matrix or data.frame with 4 numeric columns
representing character state
counts for missing data, homozygots for allele 1, heterozygots, and
homozygots for allele 2. Individuals in rows.</p>
</td></tr>
<tr><td><code id="ModelOfDiagnostic_+3A_originalhi">OriginalHI</code></td>
<td>
<p>numeric vector of length equal to number of rows in <code>I4</code>,
representing hybrid indices of individuals.</p>
</td></tr>
<tr><td><code id="ModelOfDiagnostic_+3A_epsilon">epsilon</code></td>
<td>
<p>numeric, specifying how much the hypothetical diagnostic markers should
contribute to the likelihood calculations. Must be in <code>[0,1)</code>, keeping
tolerance setting of the <code>R</code> session in mind.</p>
</td></tr>
<tr><td><code id="ModelOfDiagnostic_+3A_verbose">verbose</code></td>
<td>
<p>logical or character with path to directory where run diagnostics will
be saved.</p>
</td></tr>
<tr><td><code id="ModelOfDiagnostic_+3A_...">...</code></td>
<td>
<p>parameters to be passed to other functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>OriginalHI</code> can be calculated with <code><a href="#topic+pHetErrOnStateCount">pHetErrOnStateCount</a></code>.
</p>


<h3>Value</h3>

<p>Matrix with dimensions of I4.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diem">diem</a></code> for utilising the model to determine appropriate marker
polarisation in estimating barriers to geneflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># state count matrix
dat &lt;- matrix(c(0, 0, 1, 3, 1, 2, 2, 0, 2, 1, 4, 1), ncol = 4)

# hybrid index calculation, assuming diploid markers
HI &lt;- apply(dat * 2, MARGIN = 1, FUN = pHetErrOnStateCount)[1, ]

# run model of diagnostics, with the weight of the ideal diagnostic marker being 0.8
ModelOfDiagnostic(I4 = dat, OriginalHI = HI, epsilon = 0.8)
#      [,1] [,2] [,3] [,4]
# [1,]  0.0  5.4  0.4  0.2
# [2,]  0.0  0.2  0.0  5.6
# [3,]  0.4  4.4  0.8  0.4
</code></pre>

<hr>
<h2 id='myotis'>Dataset of modified genotypes of bats</h2><span id='topic+myotis'></span>

<h3>Description</h3>

<p>A subset of single nucleotide polymorphisms in <em>Myotis myotis</em> from Harazim et al.
(2021). The genotypes were modified for testing purposes in such a way that markers
15 and 17 now include additional indel and substitution alleles. Eight markers
used in the dataset are monomorphic.
</p>


<h3>Format</h3>

<p>vcf file with 14 individuals and 20 markers.
</p>


<h3>Details</h3>

<p>The data is used to test conversion of genotype data from vcf to diem format with
function <code>vcf2diem</code>.
</p>


<h3>Source</h3>

<p>Harazim M., Pialek L., Pikula J., Seidlova V., Zukal J., Bachorec E.,
Bartonicka T., Kokurewicz T., Martinkova N. (2021) Associating physiological
functions with genomic variability in hibernating bats.
<em>Evolutionary Ecology</em>, 35, 291-308, doi: 10.1007/s10682-020-10096-4.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filename &lt;- system.file("extdata", "myotis.vcf", package = "diemr")
</code></pre>

<hr>
<h2 id='pHetErrOnStateCount'>Hybrid index, heterozygosity, error rate</h2><span id='topic+pHetErrOnStateCount'></span>

<h3>Description</h3>

<p>Using genotype allele counts, calculates
the hybrid index,
heterozygosity and error rate in a single individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pHetErrOnStateCount(sCount)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pHetErrOnStateCount_+3A_scount">sCount</code></td>
<td>
<p>a numeric vector of length 4 with allele counts for missing data,
homozygots for allele 1, heterozygots, and homozygots for allele 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Allele counts are genomic state counts multiplied by ploidy. As different
compartments might have different ploidies (e.g. autosomal markers, sex chromosomes,
mitochondrial markers), allele counts should be calculated per compartment and then
summarised to obtain the correct genomic allele counts.
When all individuals in each compartmenst have the same ploidy, state counts do
not need to be corrected.
</p>


<h3>Value</h3>

<p>Returns a named numeric vector with three values: p - hybrid index,
Het - heterozygosity,
Err - error rate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pHetErrOnStateCount(sCount = c(2, 4, 2, 6))
#         p       Het       Err
# 0.5833333 0.1666667 0.1428571
</code></pre>

<hr>
<h2 id='plotPolarized'>Plots polarized genotypes</h2><span id='topic+plotPolarized'></span>

<h3>Description</h3>

<p>Plots genotypes that can be optionally polarized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPolarized(
  genotypes,
  HI,
  cols = c("#FFFFFF", "#800080", "#FFE500", "#008080"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPolarized_+3A_genotypes">genotypes</code></td>
<td>
<p>character matrix comprising of _012 encodings.</p>
</td></tr>
<tr><td><code id="plotPolarized_+3A_hi">HI</code></td>
<td>
<p>numeric vector of individual hybrid indices with length equal to number
of rows in <code>genotypes</code>.</p>
</td></tr>
<tr><td><code id="plotPolarized_+3A_cols">cols</code></td>
<td>
<p>vector of four colours, representing missing data, homozygots for
genotype 0, heterozygots and homozygots for genotype 2.</p>
</td></tr>
<tr><td><code id="plotPolarized_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To import and polarize genotypes, use the function <code><a href="#topic+importPolarized">importPolarized</a></code>.
</p>
<p>When using <code><a href="#topic+diem">diem</a></code> with argument <code>verbose = TRUE</code>, hybrid indices,
<code>HI</code>, can be found in file 'HIwithOptimalPolarities.txt' in folder 'diagnostics'
in the working directory.
</p>


<h3>Value</h3>

<p>No return value, called for side effects. In the default plot, purple and green
represent side of the barrier to geneflow encoded as <code>0</code> and <code>2</code>, respectively,
yellow shows heterozygots and white missing or undetermined genotypes. Individuals
are ordered according to the <code>HI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gen &lt;- importPolarized(
  file = system.file("extdata", "data7x10.txt", package = "diemr"),
  changePolarity = c(TRUE, FALSE, TRUE, TRUE, FALSE, FALSE, TRUE, FALSE, FALSE, TRUE),
  ChosenInds = 1:7
)

h &lt;- c(0.625, 0.5, 0.455, 0.455, 0.227, 0.818, 0.292)

plotPolarized(genotypes = gen, HI = h)
</code></pre>

<hr>
<h2 id='sStateCount'>Count states in a vector</h2><span id='topic+sStateCount'></span>

<h3>Description</h3>

<p>Counts genomic states in one sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sStateCount(s)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sStateCount_+3A_s">s</code></td>
<td>
<p>character vector with elements &quot;_&quot;, &quot;0&quot;, &quot;1&quot;, &quot;2&quot; representing
missing data, homozygots for allele 1, heterozygots, and
homozygots for allele 2. The vector should represent a single individual.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes the number of markers that are fixed for an allele in the genome of
one individual. This is used to construct the I4 matrix in <code><a href="#topic+diem">diem</a></code>.
</p>


<h3>Value</h3>

<p>Numeric vector of length 4 with counts of &quot;_&quot;, &quot;0&quot;, &quot;1&quot;, &quot;2&quot; respectively.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+emPolarise">emPolarise</a></code> for changing marker polarity.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype &lt;- c("0", "0", "_", "2", "1", "0", "1")
sStateCount(genotype)
# [1] 1 3 2 1

# calculate state counts for a polarised genotype
sStateCount(emPolarise(genotype, TRUE))
# [1] 1 1 2 3
</code></pre>

<hr>
<h2 id='vcf2diem'>Convert vcf files to diem format</h2><span id='topic+vcf2diem'></span>

<h3>Description</h3>

<p>Reads vcf files and writes genotypes of the most frequent alleles based on
chromosome positions to diem format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf2diem(SNP, filename, chunk = 1L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf2diem_+3A_snp">SNP</code></td>
<td>
<p>character vector with a path to the '.vcf' or '.vcf.gz' file, or an <code>vcfR</code>
object. Diploid data are currently supported.</p>
</td></tr>
<tr><td><code id="vcf2diem_+3A_filename">filename</code></td>
<td>
<p>character vector with a path where to save the converted genotypes.</p>
</td></tr>
<tr><td><code id="vcf2diem_+3A_chunk">chunk</code></td>
<td>
<p>numeric indicating by how many markers should the result be split into
separate files. <code>chunk = 1</code> saves all markers into one file.</p>
</td></tr>
<tr><td><code id="vcf2diem_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Importing vcf files larger than 1GB is not recommended. The path to the
vcf file in <code>SNP</code> reads the file line by line, and might be a solution for
very large genomic datasets.
</p>
<div class="sourceCode"><pre> The number of files \code{vcf2diem} creates depends on the \code{chunk} argument
 and class of the \code{SNP} object. 
 
  * When \code{chunk = 1}, one output file will be created.
  * Values of \code{chunk &lt; 100} are interpreted as the number of files into which to
 split data in \code{SNP}. For \code{SNP} object of class \code{vcfR}, the number
 of markers per file is calculated from the dimensions of \code{SNP}. When class
 of \code{SNP} is \code{character}, the number of markers per file is approximated
 from a model with a message. If this number is inappropriate for the expected
 output, provide the intended number of markers per file in \code{chunk} greater
 than 100. \code{vcf2diem} will scan the whole input \code{SNP} file, creating
 additional output files until the last line in \code{SNP} is reached.
  * Values of \code{chunk &gt;= 100} mean that each output file
 in diem format will contain \code{chunk} number of lines with the data in \code{SNP}.

 When the vcf file contains markers non-informative for genome polarisation, those 
 those are removed and listed in a file *omittedLoci.txt* in the working directory. 
 The omitted loci are identified by their information in the CHROM and POS columns.
</pre></div>


<h3>Value</h3>

<p>No value returned, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# vcf2diem will write files to a working directory or a specified folder
# make sure the working directory or the folder are at a location with write permission
  myofile &lt;- system.file("extdata", "myotis.vcf", package = "diemr")
  myovcf &lt;- vcfR::read.vcfR(myofile)

  vcf2diem(SNP = myofile, filename = "test1")
  vcf2diem(SNP = myofile, filename = "test2", chunk = 3)
  vcf2diem(SNP = myovcf, filename = "test3")
  vcf2diem(SNP = myovcf, filename = "test4", chunk = 3)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
