<!DOCTYPE html><html><head><title>Help for package difconet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {difconet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#difconet.build.controlled.dataset'><p>GENERATES A DATASET CONTROLLING FOR NOISE AND GENES CONNECTED IN NETWORKS</p></a></li>
<li><a href='#difconet.noise.inspection'><p>PLOT ESTIMATED CORRELATION DISTRIBUTION AFTER ADDING NOISE</p></a></li>
<li><a href='#difconet.plot.gene.correlations'><p>PLOTS THE CORRELATIONS OF A SPECIFIC GENE</p></a></li>
<li><a href='#difconet.plot.histograms.heatmap2'><p>PLOT A HEATMAP REPRESENTATION OF THE DISTRIBUTION OF CORRELATIONS OF MANY GENES</p></a></li>
<li><a href='#difconet.run'><p>RUNS A DIFCONET ANALYSIS</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0-4</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-05-26</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Coexpressed Networks</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0), gplots</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, stringr, data.table, mvtnorm, graphics,
grDevices</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimation of DIFferential COexpressed NETworks using diverse and user metrics.
			 This package is basically used for three functions related to the estimation
			 of differential coexpression. 
			 First, to estimate differential coexpression where
			 the coexpression is estimated, by default, by Spearman correlation. For this,
			 a metric to compare two correlation distributions is needed. The package includes
			 6 metrics. Some of them needs a threshold. A new metric can also be specified as
			 a user function with specific parameters (see difconet.run). The significance is
			 be estimated by permutations.
			 Second, to generate datasets with controlled differential correlation data. This 
			 is done by either adding noise, or adding specific correlation structure.
			 Third, to show the results of differential correlation analyses. Please see
			 <a href="http://bioinformatica.mty.itesm.mx/difconet">http://bioinformatica.mty.itesm.mx/difconet</a> for further information.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://bioinformatica.mty.itesm.mx/difconet">http://bioinformatica.mty.itesm.mx/difconet</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-05-30 20:26:59 UTC; victortrevino</td>
</tr>
<tr>
<td>Author:</td>
<td>Elpidio-Emmanuel Gonzalez-Valbuena [aut],
  Victor Trevino [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Victor Trevino &lt;vtrevino@itesm.mx&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-05-31 06:24:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='difconet.build.controlled.dataset'>GENERATES A DATASET CONTROLLING FOR NOISE AND GENES CONNECTED IN NETWORKS</h2><span id='topic+difconet.build.controlled.dataset'></span>

<h3>Description</h3>

<p>This function takes a normal dataset and generate simulated tumor stages by adding progressive levels of noise. It may add artificial networks of genes connected at given correlations that can progressively increase or decrease their level of correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difconet.build.controlled.dataset(data,
    noise.genes = round(nrow(data)*0.1),
    noise.sigma = c(0.0, 0.1, 0.2), 
    nonoise.sigma = c(0.0, 0.01, 0.01), 
    netcov = matrix(c(
      0.90, 0.90, 0.75, 0.75, 0.60, 0.60, 0.45, 0.45, 0.30, 0.30, 
      0.15, 0.15, 0.30, 0.30, 0.45, 0.45, 0.60, 0.60, 0.75, 0.75,
      0.95, 0.95, 0.80, 0.80, 0.65, 0.65, 0.50, 0.50, 0.35, 0.35, 
      0.10, 0.10, 0.25, 0.25, 0.40, 0.40, 0.55, 0.55, 0.70, 0.70,
      1.00, 1.00, 0.85, 0.85, 0.70, 0.70, 0.55, 0.55, 0.40, 0.40, 
      0.05, 0.05, 0.20, 0.20, 0.35, 0.35, 0.50, 0.50, 0.65, 0.65
      ), ncol=3),
    genes.nets = 10,
    corfunc=function(a,b) cor(a,b,method="spearman"),
    verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difconet.build.controlled.dataset_+3A_data">data</code></td>
<td>
<p>data.frame or matrix representing the normal dataset. Rows are genes and columns are samples.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_noise.genes">noise.genes</code></td>
<td>
<p>the number of genes from data that will noised.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_noise.sigma">noise.sigma</code></td>
<td>
<p>Levels of gaussian noise to be added (at zero mean) expressed in a cumulative manner. </p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_nonoise.sigma">nonoise.sigma</code></td>
<td>
<p>Levels of gaussian noise to be added (at zero mean) for the rest of the genes.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_netcov">netcov</code></td>
<td>
<p>numeric matrix of correlation levels for networks, rows represent networks and columns represent stages.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_genes.nets">genes.nets</code></td>
<td>
<p>The number of genes in each generated network.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_corfunc">corfunc</code></td>
<td>
<p>Correlation method used.</p>
</td></tr>
<tr><td><code id="difconet.build.controlled.dataset_+3A_verbose">verbose</code></td>
<td>
<p>Print progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a simulated tumor progression dataset based on normal data. The progression is done by stages. The number of stages is given by the length of noise.sigma. Each stage will have the same dimensions than data (plus the networks). The stages will be N, T1, T2, and so on. The N is meant to be the data itself with no noise but for generality, the first element of noise.sigma specifies the level of noise for N (default to 0). The next values of noise.sigma will be used to generate T1, T2, and so on. Thus the returned data will be estimated by N=data+noise.sigma[1], T1=N+noise.sigma[2], T2=T1+noise.sigma[3], and so on. Note that noise.sigma will be added only to a specific number of rows given by noise.genes. The value returned is a list of the generated matrices. In top of that, the nonoise.sigma specify the level of noise added to those genes not selected to be noised. This is meant to be lower levels of noise than noise.sigma to avoid that data in stages is just a copy of previous data. This function also adds full connected networks of genes connected at netcov levels. The data added has mean=0 and sd=1. The number of rows represent the networks added. The columns represent the stages. 
</p>


<h3>Value</h3>

<p>List of stages.
</p>


<h3>Author(s)</h3>

<p>Elpidio Gonzalez and Victor Trevino <a href="mailto:vtrevino@itesm.mx">vtrevino@itesm.mx</a></p>


<h3>References</h3>

<p>Gonzalez-Valbuena and Trevino 2017 Metrics to Estimate Differential Co-Expression Networks <em>Journal Pending</em> <b>volume</b> 00&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+difconet.noise.inspection">difconet.noise.inspection</a></code>.
<code><a href="#topic+difconet.run">difconet.run</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: difconet.noise.inspection(normaldata, tumordata, sigma=0:15/10)

</code></pre>

<hr>
<h2 id='difconet.noise.inspection'>PLOT ESTIMATED CORRELATION DISTRIBUTION AFTER ADDING NOISE</h2><span id='topic+difconet.noise.inspection'></span>

<h3>Description</h3>

<p>Plots the estimated correlation distribution of a normal dataset after adding different levels of gaussian noise. It is used to estimate the level of noise needed to be added to a normal dataset to match the correlation distribution of a tumor dataset. This assumes that the correlation distribution of the tumor dataset is sharper around zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difconet.noise.inspection(ndata, tdata, sigma=c(0.5, 0.75, 1.25), maxgenes=5000, 
  corfunc=function(a,b) cor(a,b,method="spearman"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difconet.noise.inspection_+3A_ndata">ndata</code></td>
<td>
<p>The normal dataset. Rows are genes and columns are samples.</p>
</td></tr>
<tr><td><code id="difconet.noise.inspection_+3A_tdata">tdata</code></td>
<td>
<p>The tumor dataset. Rows are genes and columns are samples. Rows of tumor and normal datasets should be the same.</p>
</td></tr>
<tr><td><code id="difconet.noise.inspection_+3A_sigma">sigma</code></td>
<td>
<p>Levels of gaussian noise to be added (at zero mean).</p>
</td></tr>
<tr><td><code id="difconet.noise.inspection_+3A_maxgenes">maxgenes</code></td>
<td>
<p>Number of genes used to estimate the correlation distribution. If the number of rows in normal/tumor datasets are larger than maxgenes, maxgenes random genes are used for the estimation.</p>
</td></tr>
<tr><td><code id="difconet.noise.inspection_+3A_corfunc">corfunc</code></td>
<td>
<p>Correlation method used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the estimated density of correlation distributions of normal, tumor, and normal after adding sigma levels of noise.
</p>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Elpidio Gonzalez and Victor Trevino <a href="mailto:vtrevino@itesm.mx">vtrevino@itesm.mx</a></p>


<h3>References</h3>

<p>Gonzalez-Valbuena and Trevino 2017 Metrics to Estimate Differential Co-Expression Networks <em>Journal Pending</em> <b>volume</b> 00&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+difconet.build.controlled.dataset">difconet.build.controlled.dataset</a></code>.
<code><a href="#topic+difconet.run">difconet.run</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: difconet.noise.inspection(normaldata, tumordata, sigma=0:15/10)

</code></pre>

<hr>
<h2 id='difconet.plot.gene.correlations'>PLOTS THE CORRELATIONS OF A SPECIFIC GENE</h2><span id='topic+difconet.plot.gene.correlations'></span>

<h3>Description</h3>

<p>Draw scatter plots of the correlations of a specific gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difconet.plot.gene.correlations(dObj, gene, 
  stages=1:length(dObj$stages.data), type=c("density","scatter")[1], 
  main=rownames(dObj$stages.data[[1]])[gene], 
  legends=TRUE, plot=TRUE, ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difconet.plot.gene.correlations_+3A_dobj">dObj</code></td>
<td>
<p>The difconet object.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_gene">gene</code></td>
<td>
<p>Numeric or character. The gene index/rowname whose correlations will be drawn.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_stages">stages</code></td>
<td>
<p>Numeric or character. The stages to be included. If <strong>type=&quot;scatter&quot;</strong> and more than two stages, a call to pairs is used instead of plot.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_type">type</code></td>
<td>
<p>Character. The type of plot <strong>density</strong> or <strong>scatter</strong>.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_main">main</code></td>
<td>
<p>Character. The main title passed to plot.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_legends">legends</code></td>
<td>
<p>Logical. Specifies whether the legends are drawn when <strong>type=&quot;density&quot;</strong>.</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_plot">plot</code></td>
<td>
<p>Logical. Specifies whether the plots are actually drawn (to get the correlations).</p>
</td></tr>
<tr><td><code id="difconet.plot.gene.correlations_+3A_...">...</code></td>
<td>
<p>Further parameters passed to plot/pairs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Run the whole process of estimation differences in correlations for a given dataset. The estimations are done for all <strong>metric</strong> values, all <strong>cutoff</strong> values across all <strong>comparisons</strong>.
</p>


<h3>Value</h3>

<p>The correlations of the gene across stages (invisible).
</p>


<h3>Author(s)</h3>

<p>Elpidio Gonzalez and Victor Trevino <a href="mailto:vtrevino@itesm.mx">vtrevino@itesm.mx</a></p>


<h3>References</h3>

<p>Gonzalez-Valbuena and Trevino 2017 Metrics to Estimate Differential Co-Expression Networks <em>Journal Pending</em> <b>volume</b> 00&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+difconet.run">difconet.run</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xdata &lt;- matrix(rnorm(1000), ncol=100)
xpredictor &lt;- sample(c("A","B","C","D"),100,replace=TRUE)
dObj &lt;- difconet.run(xdata, xpredictor, metric = 4, num_perms = 10,              
  comparisons = list(c("A","D"), c("A","B"), c("B","D")),
  perm_mode = "columns")

#Top highest metric in first comparison but showing correlations in only 3 stages
difconet.plot.gene.correlations(dObj, order(dObj$combstats[[1]][,"M4.dist"], 
  decreasing=TRUE)[1], type="s", stages=1:3)
#Bottom lowest metric in second comparison showing all stages
difconet.plot.gene.correlations(dObj, order(dObj$combstats[[2]][,"M4.dist"], 
  decreasing=TRUE)[1], type="d")
#Another specific gene (3), showing densities of correlations
difconet.plot.gene.correlations(dObj, 3, type="d")

</code></pre>

<hr>
<h2 id='difconet.plot.histograms.heatmap2'>PLOT A HEATMAP REPRESENTATION OF THE DISTRIBUTION OF CORRELATIONS OF MANY GENES</h2><span id='topic+difconet.plot.histograms.heatmap2'></span>

<h3>Description</h3>

<p>Draw a heatmap whose rows are genes and columns are segments of the histogram of the distribution of correlations per gene. The height/density of the histogram is shown in colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difconet.plot.histograms.heatmap2(dObj, 
  genes=1:10, 
  stages=1:length(dObj$stages.data), 
  qprobs=c(0,.50,.975,.995), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difconet.plot.histograms.heatmap2_+3A_dobj">dObj</code></td>
<td>
<p>The difconet object.</p>
</td></tr>
<tr><td><code id="difconet.plot.histograms.heatmap2_+3A_genes">genes</code></td>
<td>
<p>Numeric or character. The gene indexes/rownames included.</p>
</td></tr>
<tr><td><code id="difconet.plot.histograms.heatmap2_+3A_stages">stages</code></td>
<td>
<p>Numeric or character. The stages to be included.</p>
</td></tr>
<tr><td><code id="difconet.plot.histograms.heatmap2_+3A_qprobs">qprobs</code></td>
<td>
<p>The quantiles used to draw the heatmap. Should be 4 points. Each has specific color codes. </p>
</td></tr>
<tr><td><code id="difconet.plot.histograms.heatmap2_+3A_...">...</code></td>
<td>
<p>Further parameters passed to plot/pairs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A heatmap is draw representing the distribution of correlations of several genes across stages.
</p>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Elpidio Gonzalez and Victor Trevino <a href="mailto:vtrevino@itesm.mx">vtrevino@itesm.mx</a></p>


<h3>References</h3>

<p>Gonzalez-Valbuena and Trevino 2017 Metrics to Estimate Differential Co-Expression Networks <em>Journal Pending</em> <b>volume</b> 00&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+difconet.run">difconet.run</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xdata &lt;- matrix(rnorm(1000), ncol=100)
xpredictor &lt;- sample(c("A","B","C","D"),100,replace=TRUE)
dObj &lt;- difconet.run(xdata, xpredictor, metric = 4, num_perms = 10,              
  comparisons = list(c("A","D"), c("A","B"), c("B","D")),
  perm_mode = "columns")

  #Top highest metric in first comparison but showing correlations in only 3 stages
  difconet.plot.gene.correlations(dObj, order(dObj$combstats[[1]][,"M4.dist"], 
    decreasing=TRUE)[1], type="s", stages=1:3)
  #Bottom lowest metric in second comparison showing all stages
  difconet.plot.gene.correlations(dObj, order(dObj$combstats[[2]][,"M4.dist"], 
    decreasing=TRUE)[1], type="d")
  #Another specific gene (1), showing densities of correlations
  difconet.plot.gene.correlations(dObj, 1, type="d")
</code></pre>

<hr>
<h2 id='difconet.run'>RUNS A DIFCONET ANALYSIS</h2><span id='topic+difconet.run'></span><span id='topic+difconet'></span>

<h3>Description</h3>

<p>Estimates the DIFferential COrrelation NETworks analysis from a given dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difconet.run(data, predictor, metric=c(1,2,3,4,5,6), cutoff=0.3, blocs=5000, 
  num_perms=10, comparisons="all", perm_mode="columns", use_all_perm = TRUE,
  save_perm=FALSE, speedup=0, verbose=TRUE, metricfunc=NULL, 
  corfunc=function(a,b) cor(a,b,method="spearman") )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difconet.run_+3A_data">data</code></td>
<td>
<p>data.frame or matrix represent the dataset. Genes in rows, samples in columns. </p>
</td></tr>
<tr><td><code id="difconet.run_+3A_predictor">predictor</code></td>
<td>
<p>Factor or numeric vector representing the classes of each column in data. The correlations will be estimated for each class separately.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_metric">metric</code></td>
<td>
<p>The metrics needed to be calculated. Valid values are 1 to 6 and 8. 1 to 6 are already implemented and shown in details. 8 specifies a user-defined metric specified in metricfunc. </p>
</td></tr>
<tr><td><code id="difconet.run_+3A_cutoff">cutoff</code></td>
<td>
<p>Cut off values used for metric 1 and/or 3.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_blocs">blocs</code></td>
<td>
<p>Number of rows per block. Because of memory issues, the correlations are estimated by blocks of genes. This value represent the size of the block. Larger values requires more memory if needed. Lower values requiere more cycles and therefore it is slower but makes it computable depending on database size and memory.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_num_perms">num_perms</code></td>
<td>
<p>Number of permutations.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_comparisons">comparisons</code></td>
<td>
<p>Character or list. If character, it could be &quot;all&quot; to specify all possible combinations of classes. If set to &quot;seq&quot;, classes are taken in order and comparisons are done by first versus second, second versus third, and so on. If this is a list containing vectors of two elements, the estimations are done for the specific comparisons included (numeric or character).</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_perm_mode">perm_mode</code></td>
<td>
<p>Character. It determines the how the permutated data is generated. It can be permutated by &quot;columns&quot;,  permutated by &quot;rows&quot; (all classes/stages), or permutated by rows within each class separately using &quot;rows.class&quot;, or &quot;all&quot; in which all data is shuffled. </p>
</td></tr>
<tr><td><code id="difconet.run_+3A_use_all_perm">use_all_perm</code></td>
<td>
<p>Logical. If TRUE, it uses all permutated data to estimate the p-value, otherwise it uses only the same row permutations to estimate the p-value (it requires a lot more permutations).</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_save_perm">save_perm</code></td>
<td>
<p>Logical. If TRUE, it save all permutated data. It may require more memory.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_speedup">speedup</code></td>
<td>
<p>Numeric. Determines whether the calculation will be sped up. This is experimental. The value specify which metric will be used to speed up. This is done by modeling the dependency of the metric and p-value using 1 percent of the rows.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Determines if printing progress information.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_metricfunc">metricfunc</code></td>
<td>
<p>Function. Specify the function to be used if a metric==8 is included. The function should receive dObj, a, and b which correspond to the difconet object and the a and b vectors of correlations needed to estimate the value of the metric. It is assumed a distance-like measure (non-negative) and values close to 0 means no difference whereas larger values represent more dissimilar correlations.</p>
</td></tr>
<tr><td><code id="difconet.run_+3A_corfunc">corfunc</code></td>
<td>
<p>Function. Specify the function that estimates the correlations, similar to the cor function. The default uses cor and spearman coefficients.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Run the whole process of estimation differences in correlations for a given dataset. The estimations are done for all <strong>metric</strong> values, all <strong>cutoff</strong> values across all <strong>comparisons</strong>.
</p>


<h3>Value</h3>

<p>A difconet object represented as a list. The items are the followings:
</p>
<table>
<tr><td><code>stage</code></td>
<td>
<p>Vector. A copy of <strong>predictor</strong> (classes).</p>
</td></tr>
<tr><td><code>labels</code></td>
<td>
<p>Vector. The levels or values of the different classes. </p>
</td></tr>
<tr><td><code>comparisons</code></td>
<td>
<p>The specified <strong>comparisons</strong> parameter.</p>
</td></tr>
<tr><td><code>num_perms</code></td>
<td>
<p>The specified number of permutations <strong>num_perms</strong> parameter.</p>
</td></tr>
<tr><td><code>perm_mode</code></td>
<td>
<p>The specified number of permutations <strong>perm_mode</strong> parameter.</p>
</td></tr>
<tr><td><code>use_all_perm</code></td>
<td>
<p>The specified number of permutations <strong>use_all_perm</strong> parameter.</p>
</td></tr>
<tr><td><code>speedup</code></td>
<td>
<p>The specified <strong>speedup</strong> parameter.</p>
</td></tr>
<tr><td><code>verbose</code></td>
<td>
<p>The specified <strong>verbose</strong> parameter.</p>
</td></tr>
<tr><td><code>metricfunc</code></td>
<td>
<p>The specified <strong>metricfunc</strong> parameter.</p>
</td></tr>
<tr><td><code>combinations</code></td>
<td>
<p>A data.frame of the combinations that were compared.</p>
</td></tr>
<tr><td><code>stages.data</code></td>
<td>
<p>A list of datasets. This is only the original <strong>data</strong> split by classes.</p>
</td></tr>
<tr><td><code>combstats</code></td>
<td>
<p>A list of all comparisons made. Each element contains a matrix whose rows represent the genes and columns represent the results of all metrics (metric.dist : metric value, metric.p : p-value, metric.q : q-value, metric.expr.p : p-value of differential expression for comparison purposes, metric.expr.q : q-value of differential expression.)</p>
</td></tr>
<tr><td><code>combdens</code></td>
<td>
<p>A list of the densities of the metric for observed data and permutations. This can be used to compare the estimated metric statistics.</p>
</td></tr>
<tr><td><code>permutations</code></td>
<td>
<p>List. If <strong>save_perm==TRUE</strong>, it saves all permutated data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Elpidio Gonzalez and Victor Trevino <a href="mailto:vtrevino@itesm.mx">vtrevino@itesm.mx</a></p>


<h3>References</h3>

<p>Gonzalez-Valbuena and Trevino 2017 Metrics to Estimate Differential Co-Expression Networks <em>Journal Pending</em> <b>volume</b> 00&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+difconet.build.controlled.dataset">difconet.build.controlled.dataset</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xdata &lt;- matrix(rnorm(1000), ncol=100)
xpredictor &lt;- sample(c("A","B","C","D"),100,replace=TRUE)
dObj &lt;- difconet.run(xdata, xpredictor, metric = 4, num_perms = 10,              
  comparisons = list(c("A","D"), c("A","B"), c("B","D")),
  perm_mode = "columns")

## Not run: 
  #xpredictor contains A, B, C, and D.
  #xdata contains the data matrix
  dObj &lt;- difconet.run(xdata, xpredictor,
  metric = c(1,2,4),
  cutoff = 0.6,
  blocs = 7000,
  num_perms = 10,              
  comparisons = list(c("A","D"), c("A","B"), c("B","D")),          
  perm_mode = "columns")

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
