<!DOCTYPE html><html><head><title>Help for package diffEnrich</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {diffEnrich}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.api_pull'><p>.api_pull</p></a></li>
<li><a href='#.combineEnrich'><p>.combineEnrich</p></a></li>
<li><a href='#.data_read'><p>.data_read</p></a></li>
<li><a href='#diffEnrich'><p>diffEnrich</p></a></li>
<li><a href='#geneLists'><p>geneLists</p></a></li>
<li><a href='#get_kegg'><p>get_kegg</p></a></li>
<li><a href='#kegg'><p>kegg</p></a></li>
<li><a href='#pathEnrich'><p>pathEnrich</p></a></li>
<li><a href='#plotFoldEnrichment'><p>plotFoldEnrichment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Given a List of Gene Symbols, Performs Differential Enrichment
Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Compare functional enrichment between two experimentally-derived groups of genes or proteins (Peterson, DR., et al.(2018)) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0198139">doi:10.1371/journal.pone.0198139</a>&gt;. Given a list of gene symbols, 'diffEnrich'  will
  perform differential enrichment analysis using the Kyoto Encyclopedia of Genes
  and Genomes (KEGG) REST API. This package provides a number of functions that are 
  intended to be used in a pipeline. Briefly, the user provides a KEGG formatted species id for either human, mouse or rat, and the package will
  download and clean species specific ENTREZ gene IDs and map them to their respective
  KEGG pathways by accessing KEGG's REST API. KEGG's API is used to guarantee the most up-to-date pathway data from KEGG. Next, the user will identify significantly
  enriched pathways from two gene sets, and finally, the user will identify 
  pathways that are differentially enriched between the two gene sets. In addition to 
  the analysis pipeline, this package also provides a plotting function. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SabaLab/diffEnrich">https://github.com/SabaLab/diffEnrich</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SabaLab/diffEnrich/issues">https://github.com/SabaLab/diffEnrich/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, kableExtra, diagram</td>
</tr>
<tr>
<td>Depends:</td>
<td>dplyr, ggplot2, R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>here, stats, rlang, stringr, reshape2, ggnewscale</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-27 17:31:13 UTC; smithh</td>
</tr>
<tr>
<td>Author:</td>
<td>Harry Smith [aut, cre],
  Laura Saba [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Harry Smith &lt;harry.smith@ucdenver.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-27 18:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='.api_pull'>.api_pull</h2><span id='topic+.api_pull'></span>

<h3>Description</h3>

<p>This function connects to the KEGG API, downloads, and cleans
ncbi gene ID data, KEGG pathway descriptions, and species specific data.
Currently, this function supports Human, Mouse, and Rat. Files will be
written to the working directory unless otherwise specified by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.api_pull(species, path = path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".api_pull_+3A_species">species</code></td>
<td>
<p>character. The species to use in kegg data pull</p>
</td></tr>
<tr><td><code id=".api_pull_+3A_path">path</code></td>
<td>
<p>character. A character string describing the path to write out KEGG
API data sets. If not provided, defaults to current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>kegg_out: A named list of the data pulled from kegg api when the
function was run. This may be different if the function is run at
different times. For reproducible results, use text files generated
by function that include the date they were pulled.
</p>

<hr>
<h2 id='.combineEnrich'>.combineEnrich</h2><span id='topic+.combineEnrich'></span>

<h3>Description</h3>

<p>This is a helper function for <code>diffEnrich</code>. This function takes the objects generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
If performing a dfferential enrichment analysis, the user will have 2 objects. There
will be one for list1 and one for list2(see example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>).
This function then merges the two data frames using the following columns that should be present
in both objects (<code>by = c("KEGG_PATHWAY_ID", "KEGG_PATHWAY_description", "KEGG_PATHWAY_cnt", "KEGG_DATABASE_cnt")</code>). This merged data frame
will be used as the input for the differential enrichment function. Any pathways that do not contain any genes from either gene list will be removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.combineEnrich(list1_pe, list2_pe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".combineEnrich_+3A_list1_pe">list1_pe</code></td>
<td>
<p>object of class <code>pathEnrich</code> generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
See example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>.</p>
</td></tr>
<tr><td><code id=".combineEnrich_+3A_list2_pe">list2_pe</code></td>
<td>
<p>object of class <code>pathEnrich</code> generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
See example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>combined_enrich: An object of class data.frame that is the result of merging
<code>list1_pe</code> and <code>list2_pe</code>, using the default joining implemented in the base
<code><a href="base.html#topic+merge">merge</a></code> function.
</p>

<hr>
<h2 id='.data_read'>.data_read</h2><span id='topic+.data_read'></span>

<h3>Description</h3>

<p>This function reads in the text files generated from a previous
<code>get_kegg</code> call and saves them as a names list formatted for down
stream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.data_read(path = path, date = date, release = release)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".data_read_+3A_path">path</code></td>
<td>
<p>character. A character string describing the path to write out KEGG
API data sets. If not provided, defaults to current working directory.</p>
</td></tr>
<tr><td><code id=".data_read_+3A_date">date</code></td>
<td>
<p>character. A character string describing the date that was used to
time stamp files from previous call. Must be formatted like YYYY-MM-DD.</p>
</td></tr>
<tr><td><code id=".data_read_+3A_release">release</code></td>
<td>
<p>character. A character string describing the KEGG release that was used to
time stamp files from previous call (e.g. &quot;90&quot; or &quot;92&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>kegg_out: A named list of the data pulled from kegg api when the
function was run. This may be different if the function is run at
different times. For reproducible results, use text files generated
by function that include the date they were pulled.
</p>

<hr>
<h2 id='diffEnrich'>diffEnrich</h2><span id='topic+diffEnrich'></span><span id='topic+print.diffEnrich'></span><span id='topic+summary.diffEnrich'></span>

<h3>Description</h3>

<p>This function takes the objects generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
If performing a dfferential enrichment analysis, the user will have 2 objects. There
will be one for the genes of interest in gene list 1 and one for the genes of interest in gene list 2 (see example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>).
This function then uses a Fisher's Exact test to identify differentially enriched
pathways between the terms enriched in the gene-of-interest lists. <code>diffEnrich</code>
will remove KEGG pathways that do not contain any genes from either gene list as these
cannot be tested, and will print a warning message telling the user how many pathways
were removed.
<code>diffEnrich</code> returns a dataframe containing differentially enriched
pathways with their associated estimated odds ratio, unadjusted p-value, and fdr adjusted
p-value. S3 generic functions for <code>print</code> and <code>summary</code> are
provided. The <code>print</code> function prints the results table as a <code>tibble</code>, and the
<code>summary</code> function returns the number of pathways that reached statistical significance
as well as their descriptions, the number of genes used from the KEGG data base, the KEGG species,
the number of pathways that were shared (and therefore tested) between the gene lists and the
method used for multiple testing correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diffEnrich(list1_pe, list2_pe, method = "BH", cutoff = 0.05)

## S3 method for class 'diffEnrich'
print(x, ...)

## S3 method for class 'diffEnrich'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diffEnrich_+3A_list1_pe">list1_pe</code></td>
<td>
<p>object of class <code>pathEnrich</code> generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
See example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>.</p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_list2_pe">list2_pe</code></td>
<td>
<p>object of class <code>pathEnrich</code> generated from <code><a href="#topic+pathEnrich">pathEnrich</a></code>.
See example for <code><a href="#topic+pathEnrich">pathEnrich</a></code>.</p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_method">method</code></td>
<td>
<p>character. Character string telling <code>diffEnrich</code> which method to
use for multiple testing correction. Available methods are thos provided by
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, and the default is &quot;BH&quot;, or False Discovery Rate (FDR).</p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric. The p-value threshold to be used as the cutoff when determining statistical significance, and used to filter list of significant pathways.</p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_x">x</code></td>
<td>
<p>object of class <code>diffEnrich</code></p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_...">...</code></td>
<td>
<p>Unused</p>
</td></tr>
<tr><td><code id="diffEnrich_+3A_object">object</code></td>
<td>
<p>object of class <code>diffEnrich</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of class <code>diffEnrich</code> that contains 5 items:
</p>

<dl>
<dt>species</dt><dd><p>The species used in enrichment</p>
</dd>
<dt>padj</dt><dd><p>The method used to correct for multiple testing for the differential enrichment</p>
</dd>
<dt>sig_paths</dt><dd><p>The KEGG pathways the reached statistical significance after multiple testing correction.</p>
</dd>
<dt>path_intersect</dt><dd><p>the number of pathways that were shared (and therefore tested) between the gene lists.</p>
</dd>
<dt>de_table</dt><dd><p>A data frame that summarizes the results of the differential enrichment analysis and contains the following variables:</p>
</dd>
</dl>


<dl>
<dt>KEGG_PATHWAY_ID</dt><dd><p>KEGG Pathway Identifier</p>
</dd>
<dt>KEGG_PATHWAY_description</dt><dd><p>Description of KEGG Pathway (provided by KEGG)</p>
</dd>
<dt>KEGG_PATHWAY_cnt</dt><dd><p>Number of Genes in KEGG Pathway</p>
</dd>
<dt>KEGG_DATABASE_cnt</dt><dd><p>Number of Genes in KEGG Database</p>
</dd>
<dt>KEGG_PATHWAY_in_list1</dt><dd><p>Number of Genes from gene list 1 in KEGG Pathway</p>
</dd>
<dt>KEGG_DATABASE_in_list1</dt><dd><p>Number of Genes from gene list 1 in KEGG Database</p>
</dd>
<dt>expected_list1</dt><dd><p>Expected number of genes from list 1 to be in KEGG pathway by chance (i.e., not enriched)</p>
</dd>
<dt>enrich_p_list1</dt><dd><p>P-value for enrichment of list 1 genes related to KEGG pathway</p>
</dd>
<dt>p_adj_list1</dt><dd><p>Multiple testing adjustment of enrich_p_list1 (default = False Discovery Rate (Benjamini and Hochberg))</p>
</dd>
<dt>fold_enrichment_list1</dt><dd><p>KEGG_PATHWAY_in_list1/expected_list1</p>
</dd>
<dt>KEGG_PATHWAY_in_list2</dt><dd><p>Number of Genes from gene list 2 in KEGG Pathway</p>
</dd>
<dt>KEGG_DATABASE_in_list2</dt><dd><p>Number of Genes from gene list 2 in KEGG Database</p>
</dd>
<dt>expected_list2</dt><dd><p>Expected number of genes from list 2 to be in KEGG pathway by chance (i.e., not enriched)</p>
</dd>
<dt>enrich_p_list2</dt><dd><p>P-value for enrichment of list 2 genes related to KEGG pathway</p>
</dd>
<dt>p_adj_list2</dt><dd><p>Multiple testing adjustment of enrich_p_list2 (default = False Discovery Rate (Benjamini and Hochberg))</p>
</dd>
<dt>fold_enrichment_list2</dt><dd><p>KEGG_PATHWAY_in_list2/expected_list2</p>
</dd>
<dt>odd_ratio</dt><dd><p>Odds of a gene from list 2 being from this KEGG pathway / Odds of a gene from list 1 being from this KEGG pathway</p>
</dd>
<dt>diff_enrich_p</dt><dd><p>P-value for differential enrichment of this KEGG pathway between list 1 and list 2</p>
</dd>
<dt>diff_enrich_adjusted</dt><dd><p>Multiple testing adjustment of diff_enrich_p (default = False Discovery Rate (Benjamini and Hochberg))</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Generate individual enrichment reults
list1_pe &lt;- pathEnrich(gk_obj = kegg, gene_list = geneLists$list1)
list2_pe &lt;- pathEnrich(gk_obj = kegg, gene_list = geneLists$list2)

## Perform differential enrichment
dif_enrich &lt;- diffEnrich(list1_pe = list1_pe, list2_pe = list2_pe, method = 'none', cutoff = 0.05)

</code></pre>

<hr>
<h2 id='geneLists'>geneLists</h2><span id='topic+geneLists'></span>

<h3>Description</h3>

<p>This is a <code>list</code> object that contains the list background
genes and significant genes used in pathway enrichment. This object is
mostly meant for running examples and vignettes. The data provided is
for the rat, and is loaded from org.Rn.eg.db version 3.7.0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneLists
</code></pre>


<h3>Format</h3>

<p>A <code>list</code> with two names items which are:
</p>

<dl>
<dt>background</dt><dd><p>List of ENTREZ gene IDs that will considered background </p>
</dd>
<dt>sigGenes</dt><dd><p> List of ENTREZ gene IDs that were significant</p>
</dd>
</dl>

<hr>
<h2 id='get_kegg'>get_kegg</h2><span id='topic+get_kegg'></span>

<h3>Description</h3>

<p>This function calls an internal helper function that connects to the KEGG API, downloads, and stores
ncbi gene ID data, KEGG pathway descriptions, and species specific data.
Currently, this function supports Human, Mouse, and Rat. Files will be
written to the working directory unless otherwise specified by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_kegg(species, read = FALSE, path = NULL, date, release)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_kegg_+3A_species">species</code></td>
<td>
<p>character. The species to use in kegg data pull</p>
</td></tr>
<tr><td><code id="get_kegg_+3A_read">read</code></td>
<td>
<p>logical. Should <code>get_kegg</code> read in files from previous call.
If TRUE, all 3 files generated by <code>get_kegg</code> must be in the same directory and
the user must provide a file path that points to that directory.</p>
</td></tr>
<tr><td><code id="get_kegg_+3A_path">path</code></td>
<td>
<p>character. A character string describing the path to write out KEGG
API data sets. If not provided, defaults to current working directory.</p>
</td></tr>
<tr><td><code id="get_kegg_+3A_date">date</code></td>
<td>
<p>character. A character string describing the date that was used to
time stamp files from previous call. Must be formatted like YYYY-MM-DD.</p>
</td></tr>
<tr><td><code id="get_kegg_+3A_release">release</code></td>
<td>
<p>character. A character string describing the KEGG release that was used to
time stamp files from previous call (e.g. &quot;90&quot; or &quot;92&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>the <code>get_kegg</code> function is used to connect to the KEGG REST API
and download the data sets required to perform downstream analysis.
Currently, this function supports three species, and recognizes the KEGG code
for Homo sapiens (‘hsa’), Mus musculus (‘mmu’), and Rattus norvegicus (‘rno’).
For a given species, three data sets are generated: 1) Because the user must
provide their own gene lists in downstream analysis using ENTREZ gene IDs,
the data set maps NCBI/ENTREZ gene IDs to KEGG gene IDs, 2) a data set that
maps KEGG gene IDs to their respective KEGG pathway IDs, and 3) a data set that
maps KEGG pathway IDs to their respective pathway descriptions.  This function
allows the user save versioned (based on KEGG release) and time-stamped text
files of the three data sets described above. In addition to these flat files,
<code>get_kegg()</code> will also create a named list with the three relevant KEGG
data sets. The names of this list will describe the data set.
</p>
<p><strong>Table 1.</strong> Description of <code>get_kegg</code> list object
</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>get_kegg_list_object</strong> </td><td style="text-align: left;"> <strong>Object_description</strong>                              </td>
</tr>
<tr>
 <td style="text-align: left;">
   ncbi_to_kegg         </td><td style="text-align: left;"> ncbi gene ID &lt;-- mapped to --&gt; KEGG gene ID                       </td>
</tr>
<tr>
 <td style="text-align: left;">
   kegg_to_pathway      </td><td style="text-align: left;"> KEGG gene ID &lt;-- mapped to --&gt; KEGG pathway ID                    </td>
</tr>
<tr>
 <td style="text-align: left;">
   pathway_to_species   </td><td style="text-align: left;"> KEGG pathway ID &lt;-- mapped to --&gt; KEGG pathway species description
</td>
</tr>

</table>



<h3>Value</h3>

<p>kegg_out: A named list of the data pulled from kegg api when the
function was run. This may be different if the function is run at
different times. For reproducible results, use text files generated
by function that include the date they were pulled.
</p>

<dl>
<dt>ncbi_to_kegg</dt><dd><p>ncbi_to_kegg mappings as class data.frame</p>
</dd>
<dt>kegg_to_pathway</dt><dd><p>kegg_to_pathway mappings as class data.frame</p>
</dd>
<dt>pathway_to_species</dt><dd><p>pathway_to_species mappings as class data.frame</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
kegg &lt;- get_kegg(species = "rno")

## End(Not run)
## Not run: 
kegg &lt;- get_kegg(species = "mmu", path = "usr/data/out/")
kegg &lt;- get_kegg(species = "mmu", path = "usr/data/out/",
read = TRUE,
date = "2019-09-30",
release = "92")

## End(Not run)

</code></pre>

<hr>
<h2 id='kegg'>kegg</h2><span id='topic+kegg'></span>

<h3>Description</h3>

<p>This is a <code>list</code> object that contains the output generated
from the <code>get_kegg</code> function. This object is
mostly meant for running examples and vignettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg
</code></pre>


<h3>Format</h3>

<p>A <code>list</code> with three names items which are:
</p>

<dl>
<dt>kegg_to_pathway</dt><dd><p>List of kegg IDs mapped to pathway IDs </p>
</dd>
<dt>ncbi_to_kegg</dt><dd><p> List of ENTREZ gene IDs that map to kegg IDs</p>
</dd>
<dt>pathway_to_species</dt><dd><p> List of pathways IDs that map to rat pathways</p>
</dd>
</dl>

<hr>
<h2 id='pathEnrich'>pathEnrich</h2><span id='topic+pathEnrich'></span><span id='topic+print.pathEnrich'></span><span id='topic+summary.pathEnrich'></span>

<h3>Description</h3>

<p>This function takes the list generated in <code><a href="#topic+get_kegg">get_kegg</a></code> as well as a vector
of NCBI (ENTREZ) geneIDs, and identifies significantly enriched KEGG pathways using
a Fisher's Exact Test. Unadjusted p-values as well as FDR corrected p-values are
calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathEnrich(gk_obj, gene_list, method = "BH", cutoff = 0.05, N = 2)

## S3 method for class 'pathEnrich'
print(x, ...)

## S3 method for class 'pathEnrich'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathEnrich_+3A_gk_obj">gk_obj</code></td>
<td>
<p>list. Object genrated from <code>get_kegg</code>, or a list containing the
output generated from a past <code>get_kegg</code> call. Names of the list must match those
defined in <code>get_kegg</code>. If the user wishes to use an older version of data
generated by <code>get_kegg</code>, they must first load that data and put it in a named
list that matches the names given in the list generated by <code>get_kegg</code>.</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_gene_list">gene_list</code></td>
<td>
<p>Vector. Vector of NCBI (ENTREZ) geneIDs.</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_method">method</code></td>
<td>
<p>Character. Character string telling <code>diffEnrich</code> which method to
use for multiple testing correction. Available methods are those provided by
<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, and the default is &quot;BH&quot;, or False Discovery Rate (FDR).</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric. The p-value threshold to be used as the cutoff when determining statistical significance, and used to filter list of significant pathways.</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_n">N</code></td>
<td>
<p>Numeric. The number of genes from the gene list that must be present in a KEGG pathway in order for that pathway to be retained and tested.</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_x">x</code></td>
<td>
<p>object of class <code>pathEnrich</code></p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_...">...</code></td>
<td>
<p>Unused</p>
</td></tr>
<tr><td><code id="pathEnrich_+3A_object">object</code></td>
<td>
<p>object of class <code>pathEnrich</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function may not always use the complete list of genes provided by the user.
Specifically, it will only use the genes from the list provided that are also in
the most current species list pulled from the KEGG REST API, or from the older data KEGG
loaded by the user. The 'cutoff' only filters the list of pathways provided in the 'sig_paths'
list item. It is not used to filter the 'enrich_table' list object. S3 generic functions for <code>print</code> and <code>summary</code> are
provided. The <code>print</code> function prints the results table as a <code>tibble</code>, and the
<code>summary</code> function returns the number of pathways that reached statistical significance,
as well as their descriptions, the number of genes used from the KEGG data base, the KEGG species, and the
method used for multiple testing correction, and the p-value cutoff required for reaching statistical significance.
</p>


<h3>Value</h3>

<p>A list object of class <code>pathEnrich</code> that contains 6 items:
</p>

<dl>
<dt>species</dt><dd><p>The species used in enrichment</p>
</dd>
<dt>padj</dt><dd><p>The method used to correct for multiple testing</p>
</dd>
<dt>sig_paths</dt><dd><p>The KEGG pathways the reached statistical significance after multiple testing correction.</p>
</dd>
<dt>cutoff</dt><dd><p>The p-value threshold to be used as the cutoff when determining statistical significance, and used to filter final results data set.</p>
</dd>
<dt>N</dt><dd><p>The number of genes from the gene list that must be present in a KEGG pathway in order for that pathway to be retained and tested.</p>
</dd>
<dt>enrich_table</dt><dd><p>A data frame that summarizes the results of the pathway analysis and contains the following variables:</p>
</dd>
</dl>


<dl>
<dt>KEGG_PATHWAY_ID</dt><dd><p>KEGG Pathway Identifier</p>
</dd>
<dt>KEGG_PATHWAY_description</dt><dd><p>Description of KEGG Pathway (provided by KEGG)</p>
</dd>
<dt>KEGG_PATHWAY_cnt</dt><dd><p>Number of Genes in KEGG Pathway</p>
</dd>
<dt>KEGG_PATHWAY_in_list</dt><dd><p>Number of Genes from gene list in KEGG Pathway</p>
</dd>
<dt>KEGG_DATABASE_cnt</dt><dd><p>Number of Genes in KEGG Database</p>
</dd>
<dt>KEGG_DATABASE_in_list</dt><dd><p>Number of Genes from gene list in KEGG Database</p>
</dd>
<dt>expected</dt><dd><p>Expected number of genes from list to be in KEGG pathway by chance (i.e., not enriched)</p>
</dd>
<dt>enrich_p</dt><dd><p>P-value for enrichment of list genes related to KEGG pathway</p>
</dd>
<dt>p_adj</dt><dd><p>False Discovery Rate (Benjamini and Hochberg) to account for multiple testing across KEGG pathways</p>
</dd>
<dt>fold_enrichment</dt><dd><p>KEGG_PATHWAY_in_list/expected</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>
list1_pe &lt;- pathEnrich(gk_obj = kegg, gene_list = geneLists$list1)
## Not run: 
list2_pe &lt;- pathEnrich(gk_obj = kegg, gene_list = geneLists$list2, method = 'none', N = 4)

## End(Not run)

</code></pre>

<hr>
<h2 id='plotFoldEnrichment'>plotFoldEnrichment</h2><span id='topic+plotFoldEnrichment'></span>

<h3>Description</h3>

<p>This function uses the results generated using <code><a href="#topic+diffEnrich">diffEnrich</a></code>
to generate a bar plot describing the fold enrichment of a set of given KEGG
pathways stratified by their enrichment in list 1 or list 2. Users can plot
all pathways based on the adjusted p-value threshold used in <code><a href="#topic+diffEnrich">diffEnrich</a></code>
and the top N pathways sorted by the adjusted p-value threshold used in <code><a href="#topic+diffEnrich">diffEnrich</a></code>.
<code>plotFoldEnrich</code> returns a ggplot2 object so users can add additional
customizations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotFoldEnrichment(de_res, pval, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotFoldEnrichment_+3A_de_res">de_res</code></td>
<td>
<p>Dataframe. Generated using <code><a href="#topic+diffEnrich">diffEnrich</a></code></p>
</td></tr>
<tr><td><code id="plotFoldEnrichment_+3A_pval">pval</code></td>
<td>
<p>Numeric. Threshold for filtering pathways based on adjusted pvalue in de_res</p>
</td></tr>
<tr><td><code id="plotFoldEnrichment_+3A_n">N</code></td>
<td>
<p>Numeric. Number of top pathways to plot after filtering based on pval</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a grouped bar plot using ggplot2 and the
ggnewscale package. KEGG pathways are plotted on the y-axis and fold
enrichment is plotted on the x-axis. each KEGG pathway has a bar plotting
its fold enrichment in list 1 (red) and its fold enrichment in list 2 (blue).
The transparency of the bars correspond to the adjusted p-value for the
pathway's enrichment in the given list. The p-value presented as text to the
right of each pair of bars is the adjusted p-value associated with the
differential enrichment of the pathway between the two lists, and the pathways
are ordered from top to bottom by this p-value (i.e. smallest p-value on top
of plot, and largest p-value on bottom of plot).
</p>


<h3>Value</h3>

<p>ggplot object. If the user calls <code>plotFoldEnrich</code> and
assigns it to an object (see example) then no plot will print in viewer,
but if <code>plotFoldEnrich</code> is called without being assigned to an
object the plot will print to the viewer. Users can edit the ggplot object
as they would any other ggplot object (e.g. add title, theme, etc.).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plot &lt;- plotFoldEnrichment(de_res = diff_enrich, pval = 0.05, N = 5)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
