<!DOCTYPE html><html lang="en"><head><title>Help for package DiNAMIC.Duo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DiNAMIC.Duo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cyclicNullR'><p>Create a cyclic shift-based null distribution for one or two copy number matrices</p></a></li>
<li><a href='#cyclicShiftColR'><p>Perform the cyclic shift procedure on the columns of a matrix</p></a></li>
<li><a href='#dataPrep'><p>Prepare copy number data for downstream analysis</p></a></li>
<li><a href='#genomeChrPlot'><p>A Function for Plotting Mean Copy Number Values and Differences Across Multiple Chromosomes</p></a></li>
<li><a href='#genomePlot'><p>A Function for Plotting Mean Copy Number Values and Differences Across the Genome</p></a></li>
<li><a href='#luadSubset'><p>DNA copy number data for lung adenocarcinoma.</p></a></li>
<li><a href='#luscSubset'><p>DNA copy number data for lung squamous cell carcinoma.</p></a></li>
<li><a href='#pD'><p>Prepped data for DiNAMIC.Duo.</p></a></li>
<li><a href='#peelingOne'><p>A Function to Apply the Peeling Algorithm in a Single Copy Number Matrix</p></a></li>
<li><a href='#peelingOneIterate'><p>A Function to Apply the Peeling Algorithm in a Single Copy Number Matrix</p></a></li>
<li><a href='#peelingTwo'><p>A Function to Apply the Peeling Algorithm in a Two Copy Number Matrices</p></a></li>
<li><a href='#peelingTwoIterate'><p>A Function to Apply the Peeling Algorithm for Two Copy Number Matrices</p></a></li>
<li><a href='#resultsProcess'><p>Processing peeling results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Finding Recurrent DNA Copy Number Alterations and Differences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>In tumor tissue, underlying genomic instability can lead to DNA copy number alterations,
	e.g., copy number gains or losses. Sporadic copy number alterations occur randomly throughout the
	genome, whereas recurrent alterations are observed in the same genomic region across multiple
	independent samples, perhaps because they provide a selective growth advantage. Here we use
	cyclic shift permutations to identify recurrent copy number alterations in a single cohort or
	recurrent copy number differences in two cohorts based on a common set of genomic markers.
	Additional functionality is provided to perform downstream analyses, including the creation of
	summary files and graphics. DiNAMIC.Duo builds upon the original DiNAMIC package of Walter et al.
	(2011) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtq717">doi:10.1093/bioinformatics/btq717</a>&gt; and leverages the theory developed in Walter et al.
	(2015) &lt;<a href="https://doi.org/10.1093%2Fbiomet%2Fasv046">doi:10.1093/biomet/asv046</a>&gt;. An article describing DiNAMIC.Duo by Walter et al. (2022)
	can be found at &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtac542">doi:10.1093/bioinformatics/btac542</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>biocViews:</td>
<td>biomaRt</td>
</tr>
<tr>
<td>Imports:</td>
<td>biomaRt, dinamic, plyr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Python, NumPy Python library</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-02 15:03:17 UTC; vwalter1</td>
</tr>
<tr>
<td>Author:</td>
<td>Vonn Walter <a href="https://orcid.org/0000-0001-6114-6714"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Hyo Young Choi <a href="https://orcid.org/0000-0002-7627-8493"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Xiaobei Zhao <a href="https://orcid.org/0000-0002-5277-0846"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Yan Gao <a href="https://orcid.org/0000-0002-6852-2038"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  David Neil Hayes <a href="https://orcid.org/0000-0001-6203-7771"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vonn Walter &lt;vwalter1@pennstatehealth.psu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-02 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cyclicNullR'>Create a cyclic shift-based null distribution for one or two copy number matrices</h2><span id='topic+cyclicNullR'></span>

<h3>Description</h3>

<p>Create a cyclic shift-based null distribution for one or two copy number matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cyclicNullR(X, Y = NULL, numPerms = 100, randomSeed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cyclicNullR_+3A_x">X</code></td>
<td>
<p>a matrix or a data frame of copy number data. The rows and columns
of X correspond to genes and subjects, respectively.</p>
</td></tr>
<tr><td><code id="cyclicNullR_+3A_y">Y</code></td>
<td>
<p>a matrix or a data frame of copy number data. The rows and columns
of X correspond to genes and subjects, respectively. It is assumed that
the rows of X and Y are indexed by the same set of genes that appear in
genomic order.</p>
</td></tr>
<tr><td><code id="cyclicNullR_+3A_numperms">numPerms</code></td>
<td>
<p>the number of cyclic shifts used to create the null distribution.
Default = 1e2.</p>
</td></tr>
<tr><td><code id="cyclicNullR_+3A_randomseed">randomSeed</code></td>
<td>
<p>a random seed.  Default = NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function iteratively calls <code><a href="#topic+cyclicShiftColR">cyclicShiftColR</a></code> to create an empirical
permutation-based null distribution to assess the statistical significance of either
(i) the maximum and minimum difference row means of X - row means of Y, or (ii) the
maximum and minimum row means of X, depending on whether two or one copy number matrices
are being analyzed. The application of cyclic shift permutations to DNA copy number
matrices was originally described by Walter et al. (Bioinformatics, 2011;27(5):678–685).
</p>


<h3>Value</h3>

<p>a matrix with two columns.  The first column, maxNull, is an empirical
permutation-based null distribution for the maximum difference of row means
of X - row means of Y based on cyclic shift permutations of the columns of each
matrix; the second column, minNull, is an empirical distribution of the minimum
difference of the row means of X - the row means of Y based on the same permutations.
If Y = NULL, the null distributions apply to the maximum and minimum row means of X.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DiNAMIC.Duo)
output = cyclicNullR(X = pD[["X"]], Y = pD[["Y"]], numPerms = 25, randomSeed = NULL)

</code></pre>

<hr>
<h2 id='cyclicShiftColR'>Perform the cyclic shift procedure on the columns of a matrix</h2><span id='topic+cyclicShiftColR'></span>

<h3>Description</h3>

<p>Perform the cyclic shift procedure on the columns of a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cyclicShiftColR(X, randomSeed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cyclicShiftColR_+3A_x">X</code></td>
<td>
<p>a matrix or a data frame of copy number data. The rows and columns
of X correspond to genes and subjects, respectively.</p>
</td></tr>
<tr><td><code id="cyclicShiftColR_+3A_randomseed">randomSeed</code></td>
<td>
<p>a random seed.  Default = NULL.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Many algorithms for identifying recurrent DNA copy number alterations,
e.g., amplifications and deletions, assess statistical significance using
permutation-based null distributions.  Like many genomic data types, DNA copy
number data has an underlying spatial correlation structure.  Randomly permuting
the values within a given subject ignores this structure, and this can impact
type I error rates.  In contrast, cyclic permutation largely preserves the
local correlation structure.
</p>


<h3>Value</h3>

<p>a matrix Z whose dimensions are the same as X.  Each column of Z
is obtained by perform a cyclic shift of the corresponding column of X.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test = matrix(c(1:50), 10, 5)
cyclicShiftColR(test, randomSeed = NULL)

</code></pre>

<hr>
<h2 id='dataPrep'>Prepare copy number data for downstream analysis</h2><span id='topic+dataPrep'></span>

<h3>Description</h3>

<p>Prepare copy number data for downstream analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataPrep(X, Y = NULL, species = c("human", "mouse"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dataPrep_+3A_x">X</code></td>
<td>
<p>a matrix or a data frame of copy number data. The rows and columns
of X correspond to genes and subjects, respectively. X must have gene
symbols as its row names.</p>
</td></tr>
<tr><td><code id="dataPrep_+3A_y">Y</code></td>
<td>
<p>an optional matrix or data frame of copy number data to compare with
X. As is the case for X, the rows and columns of Y correspond to genes and
subjects, respectively. Y must have gene symbols as its row names.</p>
</td></tr>
<tr><td><code id="dataPrep_+3A_species">species</code></td>
<td>
<p>a value specifying species for ensembl database to be used;
Default is &quot;human&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This helper function is designed to prepare input matrices or data frames
X and Y containing DNA copy number data for analysis. The downstream functions
that X and Y are indexed by a common set of genes that appear in genomic order.
In addition, the peelingOne and peelingTwo functions require information about
the cytoband for each gene, and the resultsProcess function uses information
about gene position. The dataPrep function queries biomaRt so users are not
required to provide this information.
</p>


<h3>Value</h3>

<p>a list of processed X and Y with a data frame containing gene
annotation information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#This runtime for this code slightly exceeds the limits imposed by CRAN.
data(DiNAMIC.Duo)
output = dataPrep(X=luadSubset,Y=NULL)


</code></pre>

<hr>
<h2 id='genomeChrPlot'>A Function for Plotting Mean Copy Number Values and Differences Across Multiple Chromosomes</h2><span id='topic+genomeChrPlot'></span>

<h3>Description</h3>

<p>This function plots mean copy number values from one or two cohorts at a common set of markers across multiple chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomeChrPlot(
  inputList,
  plottingChrs = NULL,
  lwdVec = rep(1, 3),
  ltyVec = c(1:3),
  lineColorVec = c("red", "blue", "black"),
  ylimLow = -1,
  ylimHigh = 1,
  chrLabel = TRUE,
  xaxisLabel = "Chromosome",
  yaxisLabel = NULL,
  mainLabel = NULL,
  axisCex = 1,
  labelCex = 1,
  xaxisLine = 2.5,
  yaxisLine = 2.5,
  mainLine = 0,
  marginVec = c(4, 4, 3, 3),
  legendText = NULL,
  highThreshold = NULL,
  lowThreshold = NULL,
  showLegend = FALSE,
  legendXQuantile = 0.55,
  legendYCoord = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomeChrPlot_+3A_inputlist">inputList</code></td>
<td>
<p>A list produced by dataPrep.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_plottingchrs">plottingChrs</code></td>
<td>
<p>A numeric list of chromosomes to be plotted.  A separate plot is produced for each chromosome.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_lwdvec">lwdVec</code></td>
<td>
<p>A vector of line widths.  Default = rep(1, 3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_ltyvec">ltyVec</code></td>
<td>
<p>A vector of line types.  Default = c(1:3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_linecolorvec">lineColorVec</code></td>
<td>
<p>A vector of line colors.  Default = c(&quot;red&quot;, &quot;blue&quot;, &quot;black&quot;).</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_ylimlow">ylimLow</code></td>
<td>
<p>The lower limit of the y-values in the plot.  Default = -1.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_ylimhigh">ylimHigh</code></td>
<td>
<p>The upper limit of the y-values in the plot.  Default = 1.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_chrlabel">chrLabel</code></td>
<td>
<p>Binary value determining whether or not chromosomes are labeled.  Default = TRUE.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_xaxislabel">xaxisLabel</code></td>
<td>
<p>Label for the x-axis in the plot.  Default = &quot;Chromosome&quot;.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_yaxislabel">yaxisLabel</code></td>
<td>
<p>Label for the y-axis in the plot.  Default = NULL.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_mainlabel">mainLabel</code></td>
<td>
<p>Main label in the plot.  Default = NULL.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_axiscex">axisCex</code></td>
<td>
<p>Point size for the scale on the axis.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_labelcex">labelCex</code></td>
<td>
<p>Point size for the axis label.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_xaxisline">xaxisLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the x-axis label.  Default = 2.5.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_yaxisline">yaxisLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the y-axis label.  Default = 2.5.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_mainline">mainLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the main.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_marginvec">marginVec</code></td>
<td>
<p>Numerical vector specifying margin sizes.  Default = c(4, 4, 3, 3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_legendtext">legendText</code></td>
<td>
<p>Character vector used in the legend.  Only shown if showLegend = TRUE.  Default = NULL.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_highthreshold">highThreshold</code></td>
<td>
<p>Numerical value representing the position of the upper horizontal line, e.g., a threshold for
assessing statistical significance.  Default = NULL.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_lowthreshold">lowThreshold</code></td>
<td>
<p>Numerical value representing the position of the lower horizontal line, e.g., a threshold for
assessing statistical significance.  Default = NULL.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_showlegend">showLegend</code></td>
<td>
<p>Binary value determining whether or not the legend is shown.  Default = FALSE.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_legendxquantile">legendXQuantile</code></td>
<td>
<p>Quantile to specify the &quot;x&quot; location of the legend.  Only relevant if showLegend = TRUE  Default = 0.55.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomeChrPlot_+3A_legendycoord">legendYCoord</code></td>
<td>
<p>Numerical value to specify the &quot;y&quot; location of the legend.  Only relevant if showLegend = TRUE.  Default = 1.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Although <code><a href="#topic+genomePlot">genomePlot</a></code> can be used to visualize copy number values and
copy number alterations across the genome, the scale makes it difficult to see events
that affect small genomic regions.  These events are easier to see if the viewing window
is restricted to individual chromosomes, as is done here.  If Y = NULL in the input list,
then the plot shows a single line corresponding to the mean DNA copy number values based
on the entries in X.  If both X and Y are specified, the plot shows three lines corresponding
to the mean DNA copy number values in X, the mean DNA copy number values in Y, and the
difference of the mean DNA copy number values.
</p>


<h3>Value</h3>

<p>Creates a multi-page plot of mean copy number values and differences by chromosome.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genomeChrPlot(inputList = pD, ylimLow = -1.4, ylimHigh = 1.4)

</code></pre>

<hr>
<h2 id='genomePlot'>A Function for Plotting Mean Copy Number Values and Differences Across the Genome</h2><span id='topic+genomePlot'></span>

<h3>Description</h3>

<p>This function plots mean copy number values from one or two cohorts at a common set of markers across the genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genomePlot(
  inputList,
  lwdVec = rep(1, 3),
  ltyVec = c(1:3),
  lineColorVec = c("red", "blue", "black"),
  ylimLow = -1,
  ylimHigh = 1,
  chrLabel = TRUE,
  xaxisLabel = "Chromosome",
  yaxisLabel = NULL,
  mainLabel = NULL,
  rectColors = c("light gray", "gray"),
  axisCex = 1,
  labelCex = 1,
  xaxisLine = 2.5,
  yaxisLine = 2.5,
  mainLine = 0,
  marginVec = c(4, 4, 3, 3),
  legendText = NULL,
  highThreshold = NULL,
  lowThreshold = NULL,
  showLegend = FALSE,
  legendXQuantile = 0.55,
  legendYCoord = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomePlot_+3A_inputlist">inputList</code></td>
<td>
<p>A list produced by dataPrep.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_lwdvec">lwdVec</code></td>
<td>
<p>A vector of line widths.  Default = rep(1, 3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_ltyvec">ltyVec</code></td>
<td>
<p>A vector of line types.  Default = c(1:3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_linecolorvec">lineColorVec</code></td>
<td>
<p>A vector of line colors.  Default = c(&quot;red&quot;, &quot;blue&quot;, &quot;black&quot;).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_ylimlow">ylimLow</code></td>
<td>
<p>The lower limit of the y-values in the plot.  Default = -1.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_ylimhigh">ylimHigh</code></td>
<td>
<p>The upper limit of the y-values in the plot.  Default = 1.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_chrlabel">chrLabel</code></td>
<td>
<p>Binary value determining whether or not chromosomes are labeled.  Default = TRUE.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_xaxislabel">xaxisLabel</code></td>
<td>
<p>Label for the x-axis in the plot.  Default = &quot;Chromosome&quot;.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_yaxislabel">yaxisLabel</code></td>
<td>
<p>Label for the y-axis in the plot.  Default = NULL.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_mainlabel">mainLabel</code></td>
<td>
<p>Main label in the plot.  Default = NULL.  See <code><a href="base.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_rectcolors">rectColors</code></td>
<td>
<p>Background colors for different chromosomes.  Default = c(&quot;light gray&quot;, &quot;gray&quot;).</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_axiscex">axisCex</code></td>
<td>
<p>Point size for the scale on the axis.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_labelcex">labelCex</code></td>
<td>
<p>Point size for the axis label.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_xaxisline">xaxisLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the x-axis label.  Default = 2.5.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_yaxisline">yaxisLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the y-axis label.  Default = 2.5.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_mainline">mainLine</code></td>
<td>
<p>Numerical value used to specify the location (line) of the main.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_marginvec">marginVec</code></td>
<td>
<p>Numerical vector specifying margin sizes.  Default = c(4, 4, 3, 3).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_legendtext">legendText</code></td>
<td>
<p>Character vector used to legend.  Only shown if showLegend = TRUE.  Default = NULL.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_highthreshold">highThreshold</code></td>
<td>
<p>Numerical value representing the position of the upper horizontal line, e.g., a threshold for
assessing statistical significance.  Default = NULL.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_lowthreshold">lowThreshold</code></td>
<td>
<p>Numerical value representing the position of the lower horizontal line, e.g., a threshold for
assessing statistical significance.  Default = NULL.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_showlegend">showLegend</code></td>
<td>
<p>Binary value determining whether or not the legend is shown.  Default = FALSE.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_legendxquantile">legendXQuantile</code></td>
<td>
<p>Quantile to specify the &quot;x&quot; location of the legend.  Only relevant if showLegend = TRUE  Default = 0.55.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="genomePlot_+3A_legendycoord">legendYCoord</code></td>
<td>
<p>Numerical value to specify the &quot;y&quot;location of the legend.  Only relevant if showLegend = TRUE.  Default = 1.  See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to visualize copy number values and copy number alterations
across the genome.  If Y = NULL in the input list, then the plot shows a single line
corresponding to the mean DNA copy number values based on the entries in X.
If both X and Y are specified, the plot shows three lines corresponding to the mean
DNA copy number values in X, the mean DNA copy number values in Y, and the difference
of the mean DNA copy number values.
</p>


<h3>Value</h3>

<p>Creates a genomewide plot of mean copy number values and differences.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genomeChrPlot(inputList = pD, ylimLow = -1.4, ylimHigh = 1.4)

</code></pre>

<hr>
<h2 id='luadSubset'>DNA copy number data for lung adenocarcinoma.</h2><span id='topic+luadSubset'></span>

<h3>Description</h3>

<p>A subset of the DNA copy number data from the TCGA lung
lung adenocarcinoma cohort.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>luadSubset
</code></pre>


<h3>Format</h3>

<p>A numeric matrix of quantitative DNA copy number values with 3475 rows and 65 columns.
Rows are indexed by genes, and columns are indexed by samples.  The entries are on the log
ratio scale and were produced by the GISTIC pipeline.
</p>


<h3>Details</h3>

<p>The original data set downloaded from <a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>
contained gene-level segmented DNA copy number values for 24776 genes and 516 samples
that were produced by the GISTIC pipeline.  Because of the size limitations on data
in R packages, a random subset of genes and samples was selected.  By construction,
the number of genes and samples in the lung adenocarcinoma data is different than the
number of genes and samples in the lung squamous cell carcinoma data.
</p>


<h3>Source</h3>

<p><a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>
</p>

<hr>
<h2 id='luscSubset'>DNA copy number data for lung squamous cell carcinoma.</h2><span id='topic+luscSubset'></span>

<h3>Description</h3>

<p>A subset of the DNA copy number data from the TCGA lung
lung squamous cell carcinoma cohort.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>luscSubset
</code></pre>


<h3>Format</h3>

<p>A numeric matrix of quantitative DNA copy number values with 3480 rows and 60 columns.
Rows are indexed by genes, and columns are indexed by samples.  The entries are on the log
ratio scale and were produced by the GISTIC pipeline.
</p>


<h3>Details</h3>

<p>The original data set downloaded from <a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>
contained gene-level segmented DNA copy number values for 24776 genes and 501 samples
that were produced by the GISTIC pipeline.  Because of the size limitations on data
in R packages, a random subset of genes and samples was selected.  By construction,
the number of genes and samples in the lung squamous cell carcinoma data is different
than the number of genes and samples in the lung adenocarcinoma data.
</p>


<h3>Source</h3>

<p><a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>
</p>

<hr>
<h2 id='pD'>Prepped data for DiNAMIC.Duo.</h2><span id='topic+pD'></span>

<h3>Description</h3>

<p>A list produced by the dataPrep() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pD
</code></pre>


<h3>Format</h3>

<p>A list containing three components for a common set of genes.
</p>


<h3>Details</h3>

<p>The components of the list are named X, Y, and posDT.  They contain the DNA
copy number data for lung adenocarcinoma, the DNA copy number data for lung squamous
cell carcinoma, and gene position data, respectively.  This data object was produced
by applying <code><a href="#topic+dataPrep">dataPrep</a></code> using the <code><a href="#topic+luadSubset">luadSubset</a></code> and <code><a href="#topic+luscSubset">luscSubset</a></code>
data sets.  This reduces run time when the package is compiled by CRAN, thus eliminating run
time errors.
</p>

<hr>
<h2 id='peelingOne'>A Function to Apply the Peeling Algorithm in a Single Copy Number Matrix</h2><span id='topic+peelingOne'></span>

<h3>Description</h3>

<p>This function applies the peeling algorithm described in Walter et al. (Bioinformatics, 2011;27(5):678–685)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peelingOne(X, posDT, k, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peelingOne_+3A_x">X</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingOne_+3A_posdt">posDT</code></td>
<td>
<p>A data frame containing genomic position information for the genes in X.</p>
</td></tr>
<tr><td><code id="peelingOne_+3A_k">k</code></td>
<td>
<p>The location (row of X) containing the peak that will be peeled.</p>
</td></tr>
<tr><td><code id="peelingOne_+3A_threshold">threshold</code></td>
<td>
<p>A tuning parameter that controls the size of the peeled region.  Rows of X
</p>
<p>with mean copy number less than threshold will not be peeled.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>to remove a peak from a copy number data set and define a genomic interval of interest around
</p>
<p>the peak.
</p>
<p>Tumor genomes often contain multiple DNA copy number alterations, e.g., amplifications
or deletions. The locus that harbors the most extreme alteration, k, as evidenced by the maximum
or minimum column mean, provides a point estimate for the location of an underlying driver gene.
Also, loci near k may be affected by the same underlying genomic event. The peeling procedure is
applied to &quot;nullify&quot; entries in X that contribute to the alteration at k, thus making it possible
to identify altered regions elsewhere in the genome. This function is called by
<code><a href="#topic+peelingOneIterate">peelingOneIterate</a></code>.
</p>


<h3>Value</h3>

<p>A list containing two elements: X and interval.  X is an updated version of the input
copy number matrix in which the peak at k has been removed, and interval is genomic region
containing k.  By construction, interval cannot extend beyond the chromosome arm containing k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lusc=pD[["X"]]

 posDT=pD[["posDT"]]

 kLusc=which.max(rowMeans(lusc))

 peeledLusc=peelingOne(X=lusc,posDT=posDT,k=kLusc,threshold=NULL)   

</code></pre>

<hr>
<h2 id='peelingOneIterate'>A Function to Apply the Peeling Algorithm in a Single Copy Number Matrix</h2><span id='topic+peelingOneIterate'></span>

<h3>Description</h3>

<p>This function iteratively applies the peelingOne function, thereby identifying multiple
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peelingOneIterate(
  X,
  posDT,
  gain = TRUE,
  nullDist = NULL,
  threshold = NULL,
  numIters = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peelingOneIterate_+3A_x">X</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingOneIterate_+3A_posdt">posDT</code></td>
<td>
<p>A data frame containing genomic position information for the genes in X.</p>
</td></tr>
<tr><td><code id="peelingOneIterate_+3A_gain">gain</code></td>
<td>
<p>A logical value indicating whether gains (TRUE) or losses (FALSE) will be peeled.
</p>
<p>Default = TRUE.</p>
</td></tr>
<tr><td><code id="peelingOneIterate_+3A_nulldist">nullDist</code></td>
<td>
<p>An empirical null distribution produced by the cyclic shift algorithm.
</p>
<p>Default = NULL.</p>
</td></tr>
<tr><td><code id="peelingOneIterate_+3A_threshold">threshold</code></td>
<td>
<p>A tuning parameter that controls the size of the peeled region.  Rows of X
</p>
<p>with mean copy number less than threshold will not be peeled.  Default = NULL.</p>
</td></tr>
<tr><td><code id="peelingOneIterate_+3A_numiters">numIters</code></td>
<td>
<p>The number of times peelingOne will be iterated.  Default = 5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>peaks across the genome in a single cohort.  Gains and losses should be analyzed separately.
</p>
<p>The <code><a href="#topic+peelingOne">peelingOne</a></code> function applies the peeling algorithm described by
Walter et al. (Bioinformatics, 2011;27(5):678–685) at a given marker k. Because tumor genomes
may contain multiple regions of copy number gain or loss, it important to apply the peeling
process iteratively, thereby identifying multiple markers and surrounding genomic regions.
</p>


<h3>Value</h3>

<p>A list containing two elements: X and interval.  X is an updated version of the input
copy number matrix in which the peak at k has been removed, and interval is genomic region
containing k.  By construction, interval cannot extend beyond the chromosome arm containing k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lusc=pD[["X"]]

 posDT=pD[["posDT"]]

 gain = TRUE

 nullDist = NULL

 threshold = NULL

 numIters = 3

 peeledLusc=peelingOneIterate(X=lusc,posDT=posDT,gain=TRUE,nullDist=NULL,threshold=NULL,numIters=3)   

</code></pre>

<hr>
<h2 id='peelingTwo'>A Function to Apply the Peeling Algorithm in a Two Copy Number Matrices</h2><span id='topic+peelingTwo'></span>

<h3>Description</h3>

<p>This function applies a modified version of the peeling algorithm originally described in Walter et al.,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peelingTwo(X, Y, posDT, k, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peelingTwo_+3A_x">X</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingTwo_+3A_y">Y</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingTwo_+3A_posdt">posDT</code></td>
<td>
<p>A data frame containing genomic position information for the genes in X.</p>
</td></tr>
<tr><td><code id="peelingTwo_+3A_k">k</code></td>
<td>
<p>The location (row of X and Y) containing the peak that will be peeled.</p>
</td></tr>
<tr><td><code id="peelingTwo_+3A_threshold">threshold</code></td>
<td>
<p>A tuning parameter that controls the size of the peeled region.  Rows in which
</p>
<p>rowMeans(X) - rowMeans(Y) are less than threshold will not be peeled.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>(PMID  21183584) to remove a peak from the copy number differences and define a genomic interval of interest
</p>
<p>around the peak.
</p>
<p>When tumor genomes from two cohorts are compared, there may be multiple regions that harbor
copy number differences.  For example, gains or losses may be present in only one of the two cohorts,
and this could give rise to copy number differences.  Alternatively, the same region of the genome
may exhibit gain or loss in both cohorts.  If the magnitudes of the common gain or loss are distinct,
then this also gives rise to copy number differences.  The locus that harbors the most extreme difference,
k, provides a point estimate for the underlying driver gene that gives rise to the difference.  Loci
near k may also be affected by the underlying difference in copy number.   The peeling procedure for
two cohorts is applied to &quot;nullify&quot; entries of both X and Y that contribute to the alteration at k,
thus making it possible to identify other regions of the genome that harbor copy number differences.
This function is called by <code><a href="#topic+peelingTwoIterate">peelingTwoIterate</a></code>.
</p>


<h3>Value</h3>

<p>A list containing three elements: X, Y, and interval.  X and Y are updated versions of the
input copy number matrices X and Y in which the peak at k has been removed, and interval is genomic region
containing k.  By construction, interval cannot extend beyond the chromosome arm containing k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>luad=pD[["X"]]

 lusc=pD[["Y"]]

 posDT=pD[["posDT"]]

 kDiff=which.max(rowMeans(luad)-rowMeans(lusc))

 peeledDiff=peelingTwo(X=luad,Y=lusc,posDT=posDT,k=kDiff,threshold=NULL)

</code></pre>

<hr>
<h2 id='peelingTwoIterate'>A Function to Apply the Peeling Algorithm for Two Copy Number Matrices</h2><span id='topic+peelingTwoIterate'></span>

<h3>Description</h3>

<p>This function iteratively applies the peelingTwo function, thereby identifying multiple
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peelingTwoIterate(
  X,
  Y,
  posDT,
  gain = TRUE,
  nullDist = NULL,
  threshold = NULL,
  numIters = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="peelingTwoIterate_+3A_x">X</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_y">Y</code></td>
<td>
<p>A matrix of normalized gene-level copy number data (rows = genes, columns = subjects).</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_posdt">posDT</code></td>
<td>
<p>A data frame containing genomic position information for the genes in X.</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_gain">gain</code></td>
<td>
<p>A logical value indicating whether gains (TRUE) or losses (FALSE) will be peeled.
</p>
<p>Default = TRUE.</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_nulldist">nullDist</code></td>
<td>
<p>An empirical null distribution produced by the cyclic shift algorithm.
</p>
<p>Default = NULL.</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_threshold">threshold</code></td>
<td>
<p>A tuning parameter that controls the size of the peeled region.  Rows of X
</p>
<p>and Y with mean copy number differences less than threshold will not be peeled.  Default = NULL.</p>
</td></tr>
<tr><td><code id="peelingTwoIterate_+3A_numiters">numIters</code></td>
<td>
<p>The number of times peelingTwo will be iterated.  Default = 5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>differences across the genome between a two cohorts.  Gains and losses should be analyzed separately.
</p>
<p>The <code><a href="#topic+peelingTwo">peelingTwo</a></code> function applies the peeling procedure for two cohorts to &quot;nullify&quot;
entries in two copy number matrices X and Y that give rise to the most significant copy number
difference.  Because tumor genomes may contain multiple regions that harbor copy number differences,
it is important to apply the peeling procedure for two cohorts iteratively, thereby identifying
multiple markers and surrounding genomic regions.
</p>


<h3>Value</h3>

<p>A list containing two elements: X, Y, and interval.  X and Y are updated versions of the input
copy number matrices in which the peak difference at k has been removed, and interval is genomic region
containing k.  By construction, interval cannot extend beyond the chromosome arm containing k.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>luad=pD[["X"]]

 lusc=pD[["Y"]]

 posDT=pD[["posDT"]]

 gain = TRUE

 nullDist = NULL

 threshold = NULL

 numIters = 3

 out=peelingTwoIterate(X=luad,Y=lusc,posDT=posDT,gain=TRUE,nullDist=NULL,threshold=NULL,numIters=3)   

</code></pre>

<hr>
<h2 id='resultsProcess'>Processing peeling results</h2><span id='topic+resultsProcess'></span>

<h3>Description</h3>

<p>Processing peeling results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resultsProcess(peel.results, posDT)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resultsProcess_+3A_peel.results">peel.results</code></td>
<td>
<p>peeling results</p>
</td></tr>
<tr><td><code id="resultsProcess_+3A_posdt">posDT</code></td>
<td>
<p>a data frame containing gene annotation information; a list
component created by <code>dataPrep</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+peelingOneIterate">peelingOneIterate</a></code> function identifies (i) multiple loci across the genome
where copy number gains are losses, and (ii) regions around those loci that also exhibit copy number
changes.  Similarly, <code><a href="#topic+peelingTwoIterate">peelingTwoIterate</a></code> identifies loci and surrounding regions that
harbor copy number differences.  The output of either <code><a href="#topic+peelingOneIterate">peelingOneIterate</a></code> or
<code><a href="#topic+peelingTwoIterate">peelingTwoIterate</a></code> is processed to produce a tab-delimited text file that provides
a summary of the peeling results.  Each column corresponds to a peeled locus, and the column contains
the genomic location of the locus, the start and end positions of the surrounding peeled region,
the mean copy number value (one matrix X) or difference of mean copy number values (two matrices X and Y),
the cyclic shift-based p-value, and the names of the genes in the peeled region (in alphabetical order).
</p>


<h3>Value</h3>

<p>processed peeling results with a list of genes corresponding to each
peeled region
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dataPrep">dataPrep</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
