<!DOCTYPE html><html><head><title>Help for package dinamic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dinamic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#annot.file'><p>Cytoband annotation data frame</p></a></li>
<li><a href='#detailedLook'><p>Assessing the Significance of Recurrent DNA Copy Number Aberrations</p></a></li>
<li><a href='#makeCytoband'><p>Find the chromosome arm for each marker</p></a></li>
<li><a href='#peeling'><p>Apply the peeling procedure at a given marker</p></a></li>
<li><a href='#quickLook'><p>Find DiNAMIC's null distribution</p></a></li>
<li><a href='#recodeBinary'><p>Recode binary vectors</p></a></li>
<li><a href='#wilms.data'><p>DNA copy number data from  Wilms' tumor</p></a></li>
<li><a href='#wilms.markers'><p>Array comparative genomic hybridization marker data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Method to Analyze Recurrent DNA Copy Number Aberrations in
Tumors</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>In tumor tissue, underlying genomic instability can lead to DNA copy number alterations,
	e.g., copy number gains or losses. Sporadic copy number alterations occur randomly throughout the
	genome, whereas recurrent alterations are observed in the same genomic region across multiple
	independent samples, perhaps because they provide a selective growth advantage.
	This package implements the DiNAMIC procedure for assessing the statistical significance of 
	recurrent DNA copy number aberrations (Bioinformatics (2011) 27(5) 678 - 685).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-15 18:31:15 UTC; vwalter1</td>
</tr>
<tr>
<td>Author:</td>
<td>Vonn Walter <a href="https://orcid.org/0000-0001-6114-6714"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Andrew B. Nobel [aut],
  Fred A. Wright [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vonn Walter &lt;vwalter1@pennstatehealth.psu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-15 21:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='annot.file'>Cytoband annotation data frame</h2><span id='topic+annot.file'></span>

<h3>Description</h3>

<p>Cytoband annotation information from the hg19 genome build
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot.file
</code></pre>


<h3>Format</h3>

<p>This four-column data frame contains cytoband annotation data that is used by the <code><a href="#topic+makeCytoband">makeCytoband</a></code>
function.  Each row corresponds to a distinct cytoband, and column 1 contains the chromosome number, column 2
contains the start position (in base pairs), column 3 contains the end position (in base pairs), and column 4
contains the cytoband name (e.g. p21.3).  Additional columns may be present, but they are not used.
</p>


<h3>Details</h3>

<p>The file cytoBand.txt.gz for the hg19 build can be downloaded from the UCSC Genome Browser at
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/</a>.  The format of cytoBand.txt differs from that
of annot.file, but it can be used by the function <code><a href="#topic+makeCytoband">makeCytoband</a></code> if <code>reformat.cytoband = TRUE</code>.
</p>


<h3>Source</h3>

<p><a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/</a>
</p>

<hr>
<h2 id='detailedLook'>Assessing the Significance of Recurrent DNA Copy Number Aberrations</h2><span id='topic+detailedLook'></span>

<h3>Description</h3>

<p>Assessing the Significance of Recurrent DNA Copy Number Aberrations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detailedLook(
  x,
  marker.data,
  annot.file,
  num.perms,
  num.iters,
  gain.loss = "gain",
  reformat.annot = FALSE,
  random.seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detailedLook_+3A_x">x</code></td>
<td>
<p>An n by m numeric matrix containing DNA copy number data from n subjects at m markers.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_marker.data">marker.data</code></td>
<td>
<p>A dataframe containing marker position data for markers in the autosomes.
Column 1 contains the chromosome number for each marker, and column 2 contains the position
(in base pairs) each markers.  Additional columns, if present, represent information about
the markers (e.g. probe names).</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_annot.file">annot.file</code></td>
<td>
<p>A cytoband annotation dataframe.  Each row corresponds to a distinct cytoband,
and column 1 contains the chromosome number, column 2 contains the start position (in base pairs),
column 3 contains the end position (in base pairs), and column 4 contains the cytoband name
(e.g. p21.3).  Additional columns may be present, but they are not used.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_num.perms">num.perms</code></td>
<td>
<p>A positive integer that represents the number of cyclic shifts used to create the
empirical null distribution.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_num.iters">num.iters</code></td>
<td>
<p>A positive integer that represents the number of distinct gain (loss) loci that
will be assessed.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_gain.loss">gain.loss</code></td>
<td>
<p>A character string that indicates whether recurrent gains (<code>gain.loss = "gain"</code>)
or recurrent losses (<code>gain.loss = "loss"</code>) are assessed.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_reformat.annot">reformat.annot</code></td>
<td>
<p>A logical value that indicates whether annot.file needs to be reformatted
(default = FALSE).  See the &quot;note&quot; section of <code><a href="#topic+makeCytoband">makeCytoband</a></code> for additional information.</p>
</td></tr>
<tr><td><code id="detailedLook_+3A_random.seed">random.seed</code></td>
<td>
<p>An optional random seed (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function applies the <em>Detailed Look</em> version of DiNAMIC's cyclic shift procedure to assess
the statistical significance of recurrent DNA copy number aberrations.  Either recurrent gains
(<code>gain.loss = "gain"</code>) or recurrent losses (<code>gain.loss = "loss"</code>) are assessed using a null
distribution based on <code>num.perms</code> cyclic shifts of <code>x</code>.  Iterative calls to DiNAMIC's
<em>peeling</em> procedure (implemented here in the <code><a href="#topic+peeling">peeling</a></code> function) allow users to assess
the statistical significance of num.iters distinct gains (losses).  As noted in Bioinformatics (2011) 27(5)
678 - 685, the Detailed Look procedure recalculates the null distribution after each iteration of the peeling
procedure.  While this approach is more computationally intensive, simulations suggest that it provides more
power to detect recurrent gains (losses).
</p>


<h3>Value</h3>

<p>A matrix with <code>num.iters</code> rows.  The entries of each row correspond to the marker that is
being assessed.  More specifically, the entries are (1) the chromosome number, (2) the marker position
(in base pairs), (3) additional marker information present in <code>marker.data</code>, (4) the marker number,
and (5) the p-value obtained from the null distribution, (6) the endpoints of the peak interval (in base
pairs), as described in Bioinformatics (2011) 27(5) 678 - 685.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>detailedLook(wilms.data, wilms.markers, annot.file, 100, 3)   

</code></pre>

<hr>
<h2 id='makeCytoband'>Find the chromosome arm for each marker</h2><span id='topic+makeCytoband'></span>

<h3>Description</h3>

<p>Find the chromosome arm for each marker
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeCytoband(marker.data, annot.file, reformat.annot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeCytoband_+3A_marker.data">marker.data</code></td>
<td>
<p>A two-column numeric matrix of marker position data for markers in the autosomes.
Column 1 contains the chromosome number for each marker, and column 2 contains the position (in base
pairs) for each marker.  This is a submatrix of the marker position matrix used by <code><a href="#topic+quickLook">quickLook</a></code>
and <code><a href="#topic+detailedLook">detailedLook</a></code>.</p>
</td></tr>
<tr><td><code id="makeCytoband_+3A_annot.file">annot.file</code></td>
<td>
<p>A dataframe containing cytoband annotation for the autosomes.  Each row corresponds to a
distinct cytoband, and column 1 contains the chromosome number, column 2 contains the start position (in
base pairs), column 3 contains the end position (in base pairs), and column 4 contains the cytoband name
(e.g. p21.3).  Additional columns may be present, but they are not used.</p>
</td></tr>
<tr><td><code id="makeCytoband_+3A_reformat.annot">reformat.annot</code></td>
<td>
<p>A logical value that indicates whether <code>annot.file</code> needs to be reformatted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DiNAMIC's peeling procedure is detailed in Bioinformatics (2011) 27(5) 678 - 685, and it is performed
by the <code><a href="#topic+peeling">peeling</a></code> function.  By construction, the peeling procedure only affects markers in a given
chromosome arm.  This function is used internally by the <code><a href="#topic+peeling">peeling</a></code> function to restrict the peeling
procedure to the chromosome arm containing the marker that corresponds to <code>max(colSums(x))</code>.
</p>


<h3>Value</h3>

<p>A character vector of length m, where m is the number of markers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>wilms.pq = makeCytoband(wilms.markers, annot.file)
#A character vector of length 3288, and each entry is either
#"p" or "q", depending on the chromosome arm of the given marker.
table(wilms.pq)
#Produces the following output:
#wilms.pq
#   p    q 
#1147 2141 

</code></pre>

<hr>
<h2 id='peeling'>Apply the peeling procedure at a given marker</h2><span id='topic+peeling'></span>

<h3>Description</h3>

<p>Apply the peeling procedure at a given marker
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peeling(x, marker.data, cytoband, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peeling_+3A_x">x</code></td>
<td>
<p>An n by m numeric matrix containing DNA copy number data from n subjects at m markers.</p>
</td></tr>
<tr><td><code id="peeling_+3A_marker.data">marker.data</code></td>
<td>
<p>marker.data A two-column numeric matrix of marker position data for markers in the
autosomes.  Column 1 contains the chromosome number for each marker, and column 2 contains the position
(in base pairs) for each markers.  This is a submatrix of the marker position matrix used by
<code><a href="#topic+quickLook">quickLook</a></code> and <code><a href="#topic+detailedLook">detailedLook</a></code>.</p>
</td></tr>
<tr><td><code id="peeling_+3A_cytoband">cytoband</code></td>
<td>
<p>A character vector of length m that contains the chromosome arm (p or q) for each
marker.  This is produced by the <code><a href="#topic+makeCytoband">makeCytoband</a></code> function.</p>
</td></tr>
<tr><td><code id="peeling_+3A_k">k</code></td>
<td>
<p>A positive integer between 1 and m that represents the most aberrant marker.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The peeling procedure is detailed in Algorithm 2 of Bioinformatics (2011) 27(5) 678 - 685, but here
we provide a brief overview.  By construction, marker <code>k</code> represents the most aberrant gain (loss).
The peeling procedure rescales all copy number values in <code>x</code> that contribute to making marker <code>k</code>
aberrant, so that after applying the peeling procedure marker <code>k</code> is &quot;null.&quot;  By construction, the
rescaling procedure is restricted to entries in <code>x</code> that correspond to markers in the same chromosome
arm as <code>k</code>.  This allows users to assess the statistical significance of multiple gains (losses) throughout
the genome.
</p>


<h3>Value</h3>

<p>A list containing two components:  (1) the n by m matrix produced by applying the peeling algorithm
to the matrix <code>x</code> at marker <code>k</code>, and (2) the peak interval around marker <code>k</code>, as described
in Bioinformatics (2011) 27(5) 678 - 685.
</p>

<hr>
<h2 id='quickLook'>Find DiNAMIC's null distribution</h2><span id='topic+quickLook'></span>

<h3>Description</h3>

<p>Find DiNAMIC's null distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quickLook(
  x,
  marker.data,
  annot.file,
  num.perms,
  num.iters,
  gain.loss = "gain",
  reformat.annot = FALSE,
  random.seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quickLook_+3A_x">x</code></td>
<td>
<p>An n by m numeric matrix containing DNA copy number data from n subjects at m markers.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_marker.data">marker.data</code></td>
<td>
<p>A dataframe containing marker position data for markers in the autosomes.
Column 1 contains the chromosome number for each marker, and column 2 contains the position
(in base pairs) for each markers.  Additional columns, if present, represent information about
the markers (e.g. probe names).</p>
</td></tr>
<tr><td><code id="quickLook_+3A_annot.file">annot.file</code></td>
<td>
<p>A cytoband annotation dataframe.  Each row corresponds to a distinct cytoband,
and column 1 contains the chromosome number, column 2 contains the start position (in base pairs),
column 3 contains the end position (in base pairs), and column 4 contains the cytoband name (e.g.
p21.3).  Additional columns may be present, but they are not used.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_num.perms">num.perms</code></td>
<td>
<p>A positive integer that represents the number of cyclic shifts used to create the
empirical distribution.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_num.iters">num.iters</code></td>
<td>
<p>A positive integer that represents the number of distinct gain (loss) loci that
will be assessed.  See &quot;Details&quot; for more information.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_gain.loss">gain.loss</code></td>
<td>
<p>A character string that indicates whether recurrent gains (<code>gain.loss = "gain"</code>)
or recurrent losses (<code>gain.loss = "loss"</code>) are assessed.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_reformat.annot">reformat.annot</code></td>
<td>
<p>A logical value that indicates whether annot.file needs to be reformatted (default = FALSE).
See the &quot;Note&quot; section of <code><a href="#topic+makeCytoband">makeCytoband</a></code> for additional information.</p>
</td></tr>
<tr><td><code id="quickLook_+3A_random.seed">random.seed</code></td>
<td>
<p>An optional random seed (default = NULL).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function applies the &quot;Quick Look&quot; version of DiNAMIC's cyclic shift procedure to assess the statistical
significance of recurrent DNA copy number aberrations.  Either recurrent gains (<code>gain.loss = "gain"</code>) or
recurrent losses (<code>gain.loss = "loss"</code>) are assessed using a null distribution based on num.perms cyclic shifts
of <code>x</code>.  Iterative calls to DiNAMIC's peeling procedure (implemented here in the <code><a href="#topic+peeling">peeling</a></code> function)
allow users to assess the statistical significance of num.iters distinct gains (losses).  As noted in Bioinformatics
(2011) 27(5) 678 - 685, the &quot;Quick Look&quot; procedure calculates the null distribution once, and the same distribution
is used to assess the statistical significance of the most aberrant gain or loss after each iteration of the peeling
procedure.  This approach is less computationally intensive than &quot;Detailed Look&quot; because the null distribution is
only computed once, but simulations suggest that it provides less power to detect recurrent gains (losses).  The
resulting p-values are corrected for multiple comparisons because the null distribution is based on computing
<code>max(colSums(x))</code> or <code>min(colSums(x))</code>.
</p>


<h3>Value</h3>

<p>A matrix with <code>num.iters</code> rows.  The entries of each row correspond to the marker that is
being assessed.  More specifically, the entries are (1) the chromosome number, (2) the marker position
(in base pairs), (3) additional marker information present in marker.data, (4) the marker number, and
(5) the p-value obtained from the null distribution, (6) the endpoints of the peak interval (in base pairs),
as described in Bioinformatics (2011) 27(5) 678 - 685.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>quickLook(wilms.data, wilms.markers, annot.file, 100, 3)

</code></pre>

<hr>
<h2 id='recodeBinary'>Recode binary vectors</h2><span id='topic+recodeBinary'></span>

<h3>Description</h3>

<p>Recode binary vectors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recodeBinary(binary.vec, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recodeBinary_+3A_binary.vec">binary.vec</code></td>
<td>
<p>A binary vector of length <code>m</code> (&gt;= 1) whose <code>k</code>th entry is 1.</p>
</td></tr>
<tr><td><code id="recodeBinary_+3A_k">k</code></td>
<td>
<p>A positive integer.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called internally by <code><a href="#topic+peeling">peeling</a></code>.
</p>


<h3>Value</h3>

<p>A binary vector of length m that contains a single contiguous string of 1's,
namely the string that contains the 1 in the kth position of binary.vec.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test = c(1, 0, 0, 1, 1, 0, 0, 1, 0)
recodeBinary(test, 5)   
#Returns (0, 0, 0, 1, 1, 0, 0, 0, 0)

</code></pre>

<hr>
<h2 id='wilms.data'>DNA copy number data from  Wilms' tumor</h2><span id='topic+wilms.data'></span>

<h3>Description</h3>

<p>Probe-level DNA copy number data from Wilms' tumor (Natrajan et al., 2006)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilms.data
</code></pre>


<h3>Format</h3>

<p>A 97 by 3288 numeric matrix containing DNA copy number data, as described below.
</p>


<h3>Details</h3>

<p>Natrajan et al. (J. Pathology (2006) 210:  49 - 58) used array comparative genomic
hybridization to obtain genome-wide DNA copy number data from 97 Wilms' tumor samples at
3288 markers.  This matrix contains the DNA copy number data after applying the bias-correction
procedure outlined in Bioinformatics (2011) 27(5) 678 - 685.  Each row corresponds to DNA copy
number from one subject at 3288 markers, while each column contains DNA copy number data for
97 subjects at one marker.
</p>


<h3>Source</h3>

<p><a href="https://www.ebi.ac.uk/biostudies/arrayexpress">https://www.ebi.ac.uk/biostudies/arrayexpress</a> accession number E-TABM-10.
</p>

<hr>
<h2 id='wilms.markers'>Array comparative genomic hybridization marker data</h2><span id='topic+wilms.markers'></span>

<h3>Description</h3>

<p>Array comparative genomic hybridization marker data from Natrajan et al. (2006)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilms.markers
</code></pre>


<h3>Format</h3>

<p>A data frame with 3288 observations on the following 3 variables.
</p>

<dl>
<dt><code>Chromosome</code></dt><dd><p>The chromosome for the given marker</p>
</dd>
<dt><code>Position</code></dt><dd><p>The position (in bp) for the given marker</p>
</dd>
<dt><code>Name</code></dt><dd><p>The name of the marker (e.g., R:A-MEXP-192:RP11-465B22)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Natrajan et al. (J. Pathology (2006) 210:  49 - 58) used array comparative genomic
hybridization to obtain genome-wide DNA copy number data from 97 Wilms' tumor samples at
3288 markers.  This data frame contains genomic position data for the probes in the array.
</p>


<h3>Source</h3>

<p><a href="https://www.ebi.ac.uk/biostudies/arrayexpress">https://www.ebi.ac.uk/biostudies/arrayexpress</a> accession number E-TABM-10.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
