<!DOCTYPE html><html><head><title>Help for package DiPALM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DiPALM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AdjustPvalue'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Determine p-values based on permutation test results</a></li>
<li><a href='#BlankPlot'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Generate a blank plot window</a></li>
<li><a href='#BuildLimmaLM'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Fit linear models to time-course expression patterns using the limma package</a></li>
<li><a href='#BuildModMembership'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Calculate module membership (kME) for all genes in a time-course data set</a></li>
<li><a href='#DiPALM-package'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

DiPALM - Differential Pattern Analysis via Linear Modeling</a></li>
<li><a href='#exampleData'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Example Data: Data for use with the DiPALM vignette</a></li>
<li><a href='#ggPlotMultiDensities'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Plot multiple density distributions</a></li>
<li><a href='#PlotTCs'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Plot multiple time-course expression patterns for one gene</a></li>
<li><a href='#PlotTCsRibbon'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Plot multiple averaged time-course expression patterns for a group of genes</a></li>
<li><a href='#testData'>
<p><img src="../help/figures/DiPalm.png" alt="" /></p>

Test Data: Data for function testing</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Pattern Analysis via Linear Modeling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-24</td>
</tr>
<tr>
<td>Author:</td>
<td>Ryan C. Sartor &lt;sartorry@gmail.com&gt;, Kathleen Greenham &lt;greenham@umn.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ryan C. Sartor &lt;sartorry@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Individual gene expression patterns are encoded into a series of eigenvector patterns ('WGCNA' package). Using the framework of linear model-based differential expression comparisons ('limma' package), time-course expression patterns for genes in different conditions are compared and analyzed for significant pattern changes. For reference, see: Greenham K, Sartor RC, Zorich S, Lou P, Mockler TC and McClung CR. eLife. 2020 Sep 30;9(4). &lt;<a href="https://doi.org/10.7554%2FeLife.58993">doi:10.7554/eLife.58993</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>limma, WGCNA, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>edgeR</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-26 02:56:58 UTC; ryan</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-27 23:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='AdjustPvalue'>
<img src="../help/figures/DiPalm.png" alt="" />

Determine p-values based on permutation test results
</h2><span id='topic+AdjustPvalue'></span>

<h3>Description</h3>

<p>Used to calculate the false discovery rate associated with a specified cutoff. It uses a vector of actual test scores and a vector of permuted scores. The result can be considered a corrected p-value or q-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AdjustPvalue(tVal, tVec, pVec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AdjustPvalue_+3A_tval">tVal</code></td>
<td>

<p>The threshold or cutoff value to use (given this value, the function returns the FDR associated with using any score above this one as a positive result).
</p>
</td></tr>
<tr><td><code id="AdjustPvalue_+3A_tvec">tVec</code></td>
<td>

<p>A numeric vector of actual test result values (used to determine true positives).
</p>
</td></tr>
<tr><td><code id="AdjustPvalue_+3A_pvec">pVec</code></td>
<td>

<p>A numeric vector of permuted test result values (used to determine false positives).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function assumes all test result values above the given threshold are true positives (TP) and all permuted result values above the threshold are false positives (FP). It calculates FDR as (FP)/(TP).
When applied to individual test results, this works to estimate an adjusted p-value (or q-value) for that result.
</p>


<h3>Value</h3>

<p>A numeric value representing the FDR associated with the given cutoff (tVal).
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Returns a vector of adjusted p-values from a vector of test results
data("testData")
AdjPval_kMEs&lt;-sapply(testData$testResults[1:100],function(x) 
AdjustPvalue(tVal = x, tVec = testData$testResults, pVec = testData$permutedResults))
</code></pre>

<hr>
<h2 id='BlankPlot'>
<img src="../help/figures/DiPalm.png" alt="" />

Generate a blank plot window
</h2><span id='topic+BlankPlot'></span>

<h3>Description</h3>

<p>Creates a blank plotting window of desired size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BlankPlot(xrng = c(0, 1), yrng = c(0, 1), Main = "", xlab = "", ylab = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BlankPlot_+3A_xrng">xrng</code></td>
<td>

<p>The range of the x-axis.
</p>
</td></tr>
<tr><td><code id="BlankPlot_+3A_yrng">yrng</code></td>
<td>

<p>The range of the y-axis.
</p>
</td></tr>
<tr><td><code id="BlankPlot_+3A_main">Main</code></td>
<td>

<p>The main plot title.
</p>
</td></tr>
<tr><td><code id="BlankPlot_+3A_xlab">xlab</code></td>
<td>

<p>X-axis label.
</p>
</td></tr>
<tr><td><code id="BlankPlot_+3A_ylab">ylab</code></td>
<td>

<p>Y-axis label.
</p>
</td></tr>
<tr><td><code id="BlankPlot_+3A_...">...</code></td>
<td>

<p>Additional arguments passed to the <code><a href="graphics.html#topic+plot">plot</a></code> function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to initiate a blank plotting window in R. xrng and yrng are used to initiate the size of the plotting region.
</p>


<h3>Value</h3>

<p>Generates a blank plot
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+plot">plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>BlankPlot(xrng=c(0,10),yrng=c(0,10),Main="Test",xlab="xlab",ylab="ylab")
</code></pre>

<hr>
<h2 id='BuildLimmaLM'>
<img src="../help/figures/DiPalm.png" alt="" />

Fit linear models to time-course expression patterns using the limma package
</h2><span id='topic+BuildLimmaLM'></span>

<h3>Description</h3>

<p>This function fits a linear model (using <code><a href="limma.html#topic+lmFit">lmFit</a></code> form the limma package) to each set of kME values returned from <code><a href="#topic+BuildModMembership">BuildModMembership</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildLimmaLM(dataMat, designMat, contrastStr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildLimmaLM_+3A_datamat">dataMat</code></td>
<td>

<p>a matrix containing module membership values (a single element of the list returned by <code><a href="#topic+BuildModMembership">BuildModMembership</a></code>). Module membership is a Pearson correlation value for a single gene compared to a module eigengene.
</p>
</td></tr>
<tr><td><code id="BuildLimmaLM_+3A_designmat">designMat</code></td>
<td>

<p>a linear model design matrix containing binary values (0 and 1) with rows as samples and columns as treatments. Use the <code><a href="stats.html#topic+model.matrix">model.matrix</a></code> function in the stats package.
</p>
</td></tr>
<tr><td><code id="BuildLimmaLM_+3A_contraststr">contrastStr</code></td>
<td>

<p>a linear model contrast specifying the desired differential comparison to be made using the linear model (see the <code><a href="limma.html#topic+limma">limma</a></code> package documentation for more details).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the limma package functionality to carry out an analysis of differential patterning. Asking if a gene displays a significantly different expression pattern across time when looking at one condition compared to another.
The concept of differential patterning is similar to differential expression. However, the expression level of most genes fluctuates significantly throughout the day. Most of this is due to diurnal or circadian rhythms. These regular changes in expression severely complicate differential expression analysis and give rise to the need for examining differential patterning.
By &quot;encoding&quot; the expression pattern of each gene into a vector of module membership (kME) values using the <code><a href="#topic+BuildModMembership">BuildModMembership</a></code> function, the DiPALM analysis provides a way to use the limma method to examine differential patterning.
</p>


<h3>Value</h3>

<p>This function returns an <code><a href="limma.html#topic+MArrayLM-class">MArrayLM-class</a></code> object. This is a custom class from the <code><a href="limma.html#topic+limma">limma</a></code> package. It is an object that contains all information and results of individual linear models built for each gene in the input matrix.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BuildLimmaLM">BuildLimmaLM</a></code> , <code><a href="limma.html#topic+lmFit">lmFit</a></code> , <code><a href="limma.html#topic+makeContrasts">makeContrasts</a></code> , <code><a href="limma.html#topic+contrasts.fit">contrasts.fit</a></code> , <code><a href="limma.html#topic+eBayes">eBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("testData")
# First calculate the module memberships
kMEsList&lt;-BuildModMembership(MeMat=testData$moduleEigengenes, TCsLst=testData$timeCourseList)

# Then build the limma models
LimmaMods&lt;-lapply(kMEsList, function(x) 
BuildLimmaLM(dataMat = x, designMat = testData$modelDesign, contrastStr = testData$modelContrast))
</code></pre>

<hr>
<h2 id='BuildModMembership'>
<img src="../help/figures/DiPalm.png" alt="" />

Calculate module membership (kME) for all genes in a time-course data set
</h2><span id='topic+BuildModMembership'></span>

<h3>Description</h3>

<p>This function calculates module membership (kME) between each gene's expression vector throughout a time-course and the module eigengenes of the data set (from the WGCNA package, using <code><a href="WGCNA.html#topic+blockwiseModules">blockwiseModules</a></code>). Module membership is defined as the Pearson correlation between the expression of a single gene and the expression of a single eigengene. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BuildModMembership(MeMat, TCsLst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BuildModMembership_+3A_memat">MeMat</code></td>
<td>

<p>A data frame of eigengenes with each column being a different eigengene vector.
</p>
</td></tr>
<tr><td><code id="BuildModMembership_+3A_tcslst">TCsLst</code></td>
<td>

<p>A named list of time-course matrices. Each element of the list is a time-course expression matrix (genes as rows and time-points as columns).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The matrix of eigengene vectors, <code>MeMat</code> (see <code><a href="WGCNA.html#topic+blockwiseModules">blockwiseModules</a></code> from the WGCNA package) is used as a convenient way to summarize all the unique expression patterns (across time) that are present in a time-course data set. 
Any large gene expression data set from a complex organism will contain tens of thousands of genes. However, extensive coordinated regulation is present in all biological systems. Because of this co-regulation, usually only dozens of distinct expression patterns exist and most genes can be categorized by one (or a few) of them.
By using a relatively small set of descriptive expression vectors (eigengenes), we can &quot;encode&quot; the pattern of expression for any single gene into a length N vector where N is the number of eigengenes and the values of the vector are the Pearson correlation of the gene to each eigengene (kME).
These kME values can then be compared using the <code><a href="limma.html#topic+limma">limma</a></code> package to determine differential patterning similar to how expression values are compared to determine differential expression.
This function is the first step to calculate all kMEs for each gene in a list of time-course matrices relative to the set of eigengenes for the whole data set.
</p>


<h3>Value</h3>

<p>This function returns a list of kME matrices. The list contains a separate element for each eigengene. Each element is a matrix with a row for each gene and a column for each member of the input <code>TCsLst</code>. 
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>See Also</h3>

<p><code><a href="WGCNA.html#topic+blockwiseModules">blockwiseModules</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("testData")
kMEsList&lt;-BuildModMembership(MeMat=testData$moduleEigengenes, TCsLst=testData$timeCourseList)
</code></pre>

<hr>
<h2 id='DiPALM-package'>
<img src="../help/figures/DiPalm.png" alt="" />

DiPALM - Differential Pattern Analysis via Linear Modeling
</h2><span id='topic+DiPALM-package'></span><span id='topic+DiPALM'></span>

<h3>Description</h3>

<p>Individual gene expression patterns are encoded into a series of eigenvector patterns (WGCNA package). Using the framework of linear model-based differential expression comparisons (limma package), time-course expression patterns for genes in different conditions are compared and analyzed for significant pattern changes.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor &amp; Kathleen Greenham
</p>
<p>Maintainer: Ryan C. Sartor &lt;sartorry@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="limma.html#topic+limma">limma</a> package</code>
<code><a href="WGCNA.html#topic+blockwiseModules">blockwiseModules</a></code> from the WGCNA package
</p>

<hr>
<h2 id='exampleData'>
<img src="../help/figures/DiPalm.png" alt="" />

Example Data: Data for use with the DiPALM vignette
</h2><span id='topic+exampleData'></span>

<h3>Description</h3>

<p>A list of data objects that are used to run through the DiPALM example vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("exampleData")</code></pre>


<h3>Format</h3>

<p>A List of 3 objects:
</p>

<ol>
<li><p>$rawCounts - A data frame [42383 rows X 48 columns] containing raw counts for mRNA expression data on <em>Brassica rapa</em> R500.
</p>

<ul>
<li><p>rows : Each row represents the expression of one gene across many different samples.
</p>
</li>
<li><p>columns: Each column represents a different sample. Both drought-treated and properly watered samples are present, with two replicates of each time-course. Tissue was sampled at 4-hour intervals and column names are encoded as follows:
S[replicate number] [D-Drought / W-Watered] [Time - sample]
Example: S2D10 is part of the second replcate of the drought data and is the 10th time-point (ZT13, 13th hour after dawn on day two).
</p>
</li></ul>

</li>
<li><p>$moduleEigengenes - A dataframe [12 rows and 90 columns] representing 90 different expression vectors that describe descrete expression patterns found in the whole dataset ($timeCourseList).
</p>

<ul>
<li><p>Rows: each row represents a single time-point.
</p>
</li>
<li><p>Columns: each column represents a distinct expression pattern seen repeatedly in the full data.
</p>
</li></ul>

</li>
<li><p>$geneAnnotations - A data frame [42383 rows and 5 columns] giving the locations within the <em>Brassica rapa</em> R500 genome where each gene is found. This is in SAF format which is used by the Linux Subread software package.
</p>

<ul>
<li><p>rows: Each row represents a single gene.
</p>
</li>
<li><p>$GeneID: (character string) Unique gene accession.  
</p>
</li>
<li><p>$Chr: (character string) Chromosome number.  
</p>
</li>
<li><p>$Start: (integer) The position where the gene starts (first basepair).  
</p>
</li>
<li><p>$End: (integer) The position where the gene ends (last basepair).  
</p>
</li>
<li><p>$Strand: (character) is the gene on the positive [+] or negative [-] strand.  
</p>
</li></ul>

</li></ol>



<h3>Details</h3>

<p>Raw count data was summarized by mapping RNA-seq counts to the R500 <em>Brassica rapa</em> genome using the hisat2 aligner. Mapped reads were counted using the featureCounts function from the Linux Subread software package.
</p>


<h3>Source</h3>

<p>Data is from: <a href="https://elifesciences.org/articles/29655">Greenham et al., eLife 2017;6:e29655</a>
</p>

<hr>
<h2 id='ggPlotMultiDensities'>
<img src="../help/figures/DiPalm.png" alt="" />

Plot multiple density distributions
</h2><span id='topic+ggPlotMultiDensities'></span>

<h3>Description</h3>

<p>This function generates one or more smoothed density distributions using <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code> functionality
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggPlotMultiDensities(denslist, main = "", xlab = "", ylab = "Normalized Frequency",
 scale = T, cols = c("red", "blue", "grey50", "black", "skyblue", "orange"),
 ledgx = "topright", ltype = NULL, lwidth = NULL, xrng = NULL, yrng = NULL, Ledge = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggPlotMultiDensities_+3A_denslist">denslist</code></td>
<td>

<p>A named list of populations. Each element is a vector of numbers. A separate density plot will be generated for each population.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_main">main</code></td>
<td>

<p>The main plot title.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_xlab">xlab</code></td>
<td>

<p>The x-axis label.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_ylab">ylab</code></td>
<td>

<p>The y-axis label.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_scale">scale</code></td>
<td>

<p>When multiple densities are plotted, should they be scaled to represent the relative number of samples in each population.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_cols">cols</code></td>
<td>

<p>A vector of colors (one for each population in <code>denslist</code>).
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_ledgx">ledgx</code></td>
<td>

<p>The x value from <code><a href="graphics.html#topic+legend">legend</a></code> function. Determines where the legend is plotted.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_ltype">ltype</code></td>
<td>

<p>A vector of integers specifying the line types to be used for each population (repeated if necessary for multiple populations).
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_lwidth">lwidth</code></td>
<td>

<p>The line width to be used in the plot.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_xrng">xrng</code></td>
<td>

<p>A numeric vector of 2, the x-axis range to be plotted.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_yrng">yrng</code></td>
<td>

<p>A numeric vector of 2, the y-axis range to be plotted.
</p>
</td></tr>
<tr><td><code id="ggPlotMultiDensities_+3A_ledge">Ledge</code></td>
<td>

<p>A logical: should the legend be plotted?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes in a list of one or more numeric vectors. Each vector is considered a separate population and a smoothed density plot (similar to a histogram) will be generated for each and plotted together.
The actual numeric y-axis values are meaningless. A density plot is normalized so that the total area under the curve is equal to 1. 
When scale = T, the total area under the density plots may not equal one, but instead the total area is scaled to the relative proportions of the total population sizes of each population.
</p>


<h3>Value</h3>

<p>Generates multiple density plots on the same graph.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>See Also</h3>

<p><code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(testData)
require(ggplot2)

# Two populations of the same size, unscaled
ggPlotMultiDensities(denslist = list(Test=testData$testResults,Permuted=testData$permutedResults),
main = "Pattern Change Scores", xlab = "Differential Pattern Score",lwidth = 1)

# Two populations of different sizes, scaled
ggPlotMultiDensities(denslist = list(Test=testData$testResults[1:3000],
Permuted=testData$permutedResults), scale=TRUE , main = "Pattern Change Scores (Scaled)",
xlab = "Differential Pattern Score",lwidth = 1)

</code></pre>

<hr>
<h2 id='PlotTCs'>
<img src="../help/figures/DiPalm.png" alt="" />

Plot multiple time-course expression patterns for one gene
</h2><span id='topic+PlotTCs'></span>

<h3>Description</h3>

<p>A plotting function that generates a line plot for a single gene in multiple time-course expression sets. Each time-course is overlaid on the same plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotTCs(TClst, tgene, main = "", scale = TRUE, xlab = "Time", 
    ylab = "", xAxsLabs = colnames(TClst[[1]]), ledgeX = "top", 
    colAdj = 0.4, tcols = c("red", "red", "blue", "blue"), tltys = c(1, 1, 2, 2))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotTCs_+3A_tclst">TClst</code></td>
<td>

<p>A named list of time-course matrices. Each element of the list is a time-course expression matrix (Genes as rows and time points as columns).
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_tgene">tgene</code></td>
<td>

<p>A string of a single gene accession (found in the row names of one element of TClst).
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_main">main</code></td>
<td>

<p>Main figure title.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_scale">scale</code></td>
<td>

<p>Logical: should the expression values of each time-course be scaled and centered?
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_xlab">xlab</code></td>
<td>

<p>The X-axis label.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_ylab">ylab</code></td>
<td>

<p>The Y-axis label.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_xaxslabs">xAxsLabs</code></td>
<td>

<p>A vector of names to be associated with each x-axis point.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_ledgex">ledgeX</code></td>
<td>

<p>The x value from <code><a href="graphics.html#topic+legend">legend</a></code> function. Determines where the legend is plotted.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_coladj">colAdj</code></td>
<td>

<p>Controls the alpha channel for the line colors using <code><a href="grDevices.html#topic+adjustcolor">adjustcolor</a></code>.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_tcols">tcols</code></td>
<td>

<p>A vector of colors controlling the colors of the lines. Must list one color for each time-course in TClst.
</p>
</td></tr>
<tr><td><code id="PlotTCs_+3A_tltys">tltys</code></td>
<td>

<p>A vector of integers controlling the line type (lty) of the lines. Must list one lty for each time-course in TClst.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to view the expression pattern of a single gene across multiple time-course samples.
</p>


<h3>Value</h3>

<p>The function generates a plot with multiple lines.
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(testData)
PlotTCs(TClst = testData$timeCourseList, tgene = "BraA05g36370R" ,
scale = TRUE,tcols = c("red","red","blue","blue"), tltys = c(1,2,1,2), ledgeX = "topleft")
</code></pre>

<hr>
<h2 id='PlotTCsRibbon'>
<img src="../help/figures/DiPalm.png" alt="" />

Plot multiple averaged time-course expression patterns for a group of genes
</h2><span id='topic+PlotTCsRibbon'></span>

<h3>Description</h3>

<p>A plotting function that generates a ribbon plot representing the summarized expression of a set of genes. A separate ribbon is generated for each time-course. The ribbon represents + &amp; - 1 standard deviation from the mean at each point.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotTCsRibbon(TClst, tgenes, main = "", xlab = "Time", ylab = "",
 xAxsLabs = colnames(TClst[[1]]),scale = TRUE, alpha = 0.1, tcols, tltys,
 splits = rep(1, length(TClst)))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotTCsRibbon_+3A_tclst">TClst</code></td>
<td>

<p>A named list of time-course matrices. Each element of the list is a time-course expression matrix (Genes as rows and time points as columns).
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_tgenes">tgenes</code></td>
<td>

<p>A vector of gene accessions (found in the row names of one element of TClst).
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_main">main</code></td>
<td>

<p>The main title of the plot, if splits are defined, multiple plots will be generated and a vector of titles may be defined here.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_xlab">xlab</code></td>
<td>

<p>The X-axis label.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_ylab">ylab</code></td>
<td>

<p>The Y-axis label.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_xaxslabs">xAxsLabs</code></td>
<td>

<p>The time point labels for the x-axis, by default they will be the column names of the first time-course in TClst.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_scale">scale</code></td>
<td>

<p>Logical: should the expression values of each time-course be scaled and centered?
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_alpha">alpha</code></td>
<td>

<p>The alpha channel to be used for the ribbons (number from 0 - 1 defining transparency). See <code><a href="grDevices.html#topic+adjustcolor">adjustcolor</a></code>.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_tcols">tcols</code></td>
<td>

<p>A vector of colors controlling the colors of the lines. Must list one color for each time-course in TClst.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_tltys">tltys</code></td>
<td>

<p>A vector of integers controlling the line type (lty) of the lines. Must list one lty for each time-course in TClst.
</p>
</td></tr>
<tr><td><code id="PlotTCsRibbon_+3A_splits">splits</code></td>
<td>

<p>A vector of integers defining groups to be plotted on separate plots if desired (see examples).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can be used to plot the summarized expression pattern of an entire cluster of genes and compare between multiple time courses. Line plots are generated representing the mean expression value at each time point for the group. Ribbons are plotted along the lines in semi-transparent colors. The ribbons represent + and - one standard deviation from the mean expression level of the group.
# 
</p>


<h3>Value</h3>

<p>This function generates multiple line plots representing the average expression of a group of genes with ribbons representing standard deviation. 
</p>


<h3>Author(s)</h3>

<p>Ryan C. Sartor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("testData")

# Calculate the module memberships
kMEsList&lt;-BuildModMembership(MeMat=testData$moduleEigengenes, TCsLst=testData$timeCourseList)

# Build the limma models
LimmaMods&lt;-lapply(kMEsList, function(x) 
BuildLimmaLM(dataMat = x, designMat = testData$modelDesign, contrastStr = testData$modelContrast))

# Get the adjusted Pvalues for each gene
AdjPval_kMEs&lt;-sapply(testData$testResults[rownames(LimmaMods[[1]])],function(x) 
AdjustPvalue(tVal = x, tVec = testData$testResults, pVec = testData$permutedResults))

# Cluster the genes base on the limma results
LimmaSums&lt;-do.call(cbind,lapply(LimmaMods,function(x) x$t))
LimmaModsSig&lt;-LimmaSums[names(AdjPval_kMEs)[which(AdjPval_kMEs&lt;0.01)],]
patternCor&lt;-cor(t(LimmaModsSig))
patternTree&lt;-hclust(as.dist(1-patternCor),method = "complete")
patternClusters&lt;-cutree(tree = patternTree, k = 25)
patternClusters&lt;-tapply(rownames(patternCor),INDEX = patternClusters,function(x) x)

#Replicates are grouped together (same color + lty values)
PlotTCsRibbon(TClst = testData$timeCourseList, main="Drought Time-Course", 
xAxsLabs = c(seq(1,23,4),seq(1,23,4)), xlab="ZT Time (hours)", tgenes = patternClusters[[15]],
scale = TRUE, tcols = c("red","red","blue","blue"), tltys = c(1,1,1,1))

#Replicates are in different groups (same color but different lty values)
PlotTCsRibbon(TClst = testData$timeCourseList, main="Drought Time-Course",
xAxsLabs = c(seq(1,23,4),seq(1,23,4)), xlab="ZT Time (hours)", tgenes = patternClusters[[15]],
scale = TRUE, tcols = c("red","red","blue","blue"), tltys = c(1,2,1,2))

#Replicates are represented on differnt plots (using the splits argument)
PlotTCsRibbon(TClst = testData$timeCourseList, main=c("Rep1","Rep2"), 
xAxsLabs = c(seq(1,23,4),seq(1,23,4)), xlab="ZT Time (hours)", tgenes = patternClusters[[15]],
scale = TRUE, tcols = c("red","red","blue","blue"), tltys = c(1,2,1,2), splits = c(1,2,1,2))
</code></pre>

<hr>
<h2 id='testData'>
<img src="../help/figures/DiPalm.png" alt="" />

Test Data: Data for function testing
</h2><span id='topic+testData'></span>

<h3>Description</h3>

<p>A list of objects that can be used to test individual DiPALM functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("testData")</code></pre>


<h3>Format</h3>

<p>A List of 6 objects:
</p>

<ol>
<li><p>$timeCourseList - A List of 4 matrices containing mRNA expression data on <em>Brassica rapa</em> R500.
</p>

<ul>
<li><p>$Drought.R1 (Plants exposed to drought, Replicate 1)
A matrix of time-course gene expression data [18428 Genes X  12 Time-points]
</p>
</li>
<li><p>$Drought.R2 (Plants exposed to drought, Replicate 2)
A matrix of time-course gene expression data [18428 Genes X  12 Time-points]
</p>
</li>
<li><p>$Watered.R1 (Properly watered plants, Replicate 1)
A matrix of time-course gene expression data [18428 Genes X  12 Time-points]
</p>
</li>
<li><p>$Watered.R2 (Properly watered plants, Replicate 2)
A matrix of time-course gene expression data [18428 Genes X  12 Time-points]
</p>
</li></ul>

</li>
<li><p>$moduleEigengenes - A dataframe [12 rows and 90 columns] representing 90 different expression vectors that describe descrete expression patterns found in the whole dataset ($timeCourseList).
</p>

<ul>
<li><p>Rows: each row represents a single time-point
</p>
</li>
<li><p>Columns: each column represents a distinct expression pattern seen repeatedly in the full data
</p>
</li></ul>

</li>
<li><p>$modelDesign - A matrix [4 rows and 2 columns] used as a design matrix for a simple linear model. This matrix provides a mapping between the input instances (timeCourseList) and the two fixed effects fitted in the mode (Drought and Watered). This matrix was created with <code><a href="stats.html#topic+model.matrix">model.matrix</a></code>.
</p>
</li>
<li><p>$modelContrast - A character string specifying the contrast to be evaluated using the fitted model for each gene. This example is used to compare watered to drought conditions.
</p>
</li>
<li><p>$testResults - A named numeric vector with 18428 values from DiPALM test results on actual data.
</p>

<ul>
<li><p>names: Gene accessions
</p>
</li>
<li><p>values: The summed absolute values of t-values generated from limma model contrasts related to each eigengene pattern. This can be thought of as a DiPALM score for differential patterning.
</p>
</li></ul>

</li>
<li><p>$permutedResults- A named numeric vector with 18428 values from DiPALM test results on permuted data. This vector acts as a null distribution in order to estimate significance of actual test results.
</p>

<ul>
<li><p>names: Gene accessions
</p>
</li>
<li><p>values: The summed absolute values of t-values generated from limma model contrasts related to each eigengene pattern using permuted expression data. This can be thought of as a null distribution of DiPALM scores for differential patterning.
</p>
</li></ul>

</li></ol>



<h3>Source</h3>

<p>Data is from: <a href="https://elifesciences.org/articles/29655">Greenham et al., eLife 2017;6:e29655</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
