<!DOCTYPE html><html><head><title>Help for package DiPhiSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DiPhiSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#diphiseq'><p>Main function. For most users, this function is all what they</p>
need for the analysis.</a></li>
<li><a href='#example_data'><p>Numeric count matrix of example data.</p></a></li>
<li><a href='#outprop'><p>Give a rough estimate of the proportion of outliers in the data based on the results of DiPhiSeq.</p></a></li>
<li><a href='#robnb'><p>Calculates the estimate and standard error of beta and phi.</p>
It takes as input counts from one group of samples for a single gene.
This function is the core underlining function of the whole package.
A significant part of the code is edited based on William H. Aeberhard's glmrob.nb R function;
we appreciate them very much for sharing their code online.
This function also implement Algorithm 1 of our submitted paper about DiPhiSeq.
This function is called by robtest.
Most users don't need to call this function directly.</a></li>
<li><a href='#robtest'><p>Calls the robnb function to estimate the coefficients,</p>
and then construct the statistical tests for DD and DE.
It works for a single gene. y1 and y2 are count vectors for a single gene.
diphiseq calls this function to do the calculation for each gene.
Normal users often don't need to use this function directly.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Tests for Differential Dispersion and Differential
Expression in RNA-Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the algorithm described in Jun Li and Alicia T. Lamere, 
    "DiPhiSeq: Robust comparison of expression levels on RNA-Seq data with large sample sizes"  (Unpublished). 
    Detects not only genes that show different 
    average expressions ("differential expression", DE), but also genes that show different 
    diversities of expressions in different groups ("differentially dispersed", DD). DD genes 
    can be important clinical markers. 'DiPhiSeq' uses a redescending penalty on the 
    quasi-likelihood function, and thus has superior robustness against outliers and other noise.
    Updates from version 0.1.0: (1) Added the option of using adaptive initial value for phi.
    (2) Added a function for estimating the proportion of outliers in the data.
    (3) Modified the input parameter names for clarity, and modified the output format for the main function.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-10-24 22:30:47 UTC; jun</td>
</tr>
<tr>
<td>Author:</td>
<td>Jun Li [aut, cre],
  Alicia T. Lamere [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Li &lt;jun.li@nd.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-10-24 22:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='diphiseq'>Main function. For most users, this function is all what they
need for the analysis.</h2><span id='topic+diphiseq'></span>

<h3>Description</h3>

<p>Main function. For most users, this function is all what they
need for the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diphiseq(countmat, classlab, depth = NULL, c.tukey.beta = 4,
  c.tukey.phi = 4, phi.ini = "adaptive")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diphiseq_+3A_countmat">countmat</code></td>
<td>
<p>A count matrix. Rows are genes, and columns are samples. Each element/count is
the number of reads mapped to a gene in a sample.</p>
</td></tr>
<tr><td><code id="diphiseq_+3A_classlab">classlab</code></td>
<td>
<p>The class labels. A vector whose elements are of value 1 or 2.</p>
</td></tr>
<tr><td><code id="diphiseq_+3A_depth">depth</code></td>
<td>
<p>Sequencing depth. A vector of sequencing depth. Users are encouraged to
provide estimated values from edgeR, DESeq, PoissonSeq, or other software that they prefer.
If no values are provided, depth estimated by total counts will be used.</p>
</td></tr>
<tr><td><code id="diphiseq_+3A_c.tukey.beta">c.tukey.beta</code></td>
<td>
<p>The c value for beta in Tukey's biweight function. The default value, 4, is typically
regarded as appropriate and should work for most datasets.</p>
</td></tr>
<tr><td><code id="diphiseq_+3A_c.tukey.phi">c.tukey.phi</code></td>
<td>
<p>The c value for phi in Tukey's biweight function. The default value, 4, is typically
regarded as appropriate and should work for most datasets.</p>
</td></tr>
<tr><td><code id="diphiseq_+3A_phi.ini">phi.ini</code></td>
<td>
<p>The initial value for phi to start search. If phi.in == 'adaptive' (the default value), 
the algorithm will adaptively
choose a value (check Algorithm 1 for details). Otherwise, a positive numeric value (such as 0.5)
should be given.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A List that contains the following elements:
<code>tab</code>: This is a data frame that contains the main results of this package. It has the following columns:
<code>phi1</code>: the estimated dispersion of sample group 1.
<code>phi2</code>: the estimated dispersion of sample group 2.
<code>beta1</code>: the estimated (log) expression of sample group 1.
<code>beta2</code>: the estimated (log) expression of sample group 2.
<code>statistic.phi</code>: the z statistic for DD.
<code>statistic.beta</code>: the z statistic for DE.
<code>p.value.phi</code>: the p value for DD.
<code>p.value.beta</code>: the p value for DE.
<code>fdr.phi</code>: the FDR for DD.
<code>fdr.beta</code>: the FDR for DE.
<code>log.depth</code>: A vector of the log sequencing depths.
<code>countmat</code>: This is the count matrix that the user provided.
<code>classlab</code>: This is the vecgtor of class labels that the user provided.
<code>phi.ini</code>: The initial searching value of the dispersion parameter.
<code>mumat</code>: This is the (estimated) mu (expected expression) matrix, of the same size as countmat.
In another word, E(countmat)=mumat.
<code>phimat</code>: This is the (estimated) phi matrix, of the same size as countmat.
In another word, counmat ~ negative binomial(mumat, phimat).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>countmat &lt;- matrix(rnbinom(100, size=1, mu=50), nrow=4, ncol=25)
classlab &lt;- c(rep(1, 10), rep(2, 15))
res &lt;- diphiseq(countmat, classlab)

countmat &lt;- matrix(rnbinom(100, size=1, mu=50), nrow=4, ncol=25)
classlab &lt;- c(rep(1, 10), rep(2, 15))
res &lt;- diphiseq(countmat, classlab, phi.ini=0.5)
</code></pre>

<hr>
<h2 id='example_data'>Numeric count matrix of example data.</h2><span id='topic+example_data'></span>

<h3>Description</h3>

<p>The first several rows of a dataset from the Cancer Genome Atlas Kidney Renal Clear Cell Carcinoma
data collection, consisting of 5 genes from a dataset of raw RNA-Seq counts from 35 tumor and 35 normal samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_data
</code></pre>


<h3>Format</h3>

<p>A numeric matrix with five genes observed across 70 experiments. The first 35 are tumor samples,
the second 35 are normal samples.</p>


<h3>Source</h3>

<p>Akin, O. and Elnajjar, P. and Heller, M. and Jarosz, R. and Erickson, B. J. and Kirk,
S. and ... Filippini, J. (2016) Data from The Cancer Genome Atlas Kidney Renal Clear Cell Carcinoma
(TCGA-KIRC) collection. The Cancer Imaging Archive.
</p>

<hr>
<h2 id='outprop'>Give a rough estimate of the proportion of outliers in the data based on the results of DiPhiSeq.</h2><span id='topic+outprop'></span>

<h3>Description</h3>

<p>Give a rough estimate of the proportion of outliers in the data based on the results of DiPhiSeq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outprop(diphiseq.res, fdr.cutoff = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="outprop_+3A_diphiseq.res">diphiseq.res</code></td>
<td>
<p>The results given by running diphiseq.</p>
</td></tr>
<tr><td><code id="outprop_+3A_fdr.cutoff">fdr.cutoff</code></td>
<td>
<p>The cutoff for FDR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric value. The estimated proportion of outliers under the FDR cutoff in the data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>countmat &lt;- matrix(rnbinom(100, size=1, mu=50), nrow=4, ncol=25)
classlab &lt;- c(rep(1, 10), rep(2, 15))
res &lt;- diphiseq(countmat, classlab)
outlier.proportion &lt;- outprop(res)
</code></pre>

<hr>
<h2 id='robnb'>Calculates the estimate and standard error of beta and phi.
It takes as input counts from one group of samples for a single gene.
This function is the core underlining function of the whole package.
A significant part of the code is edited based on William H. Aeberhard's glmrob.nb R function;
we appreciate them very much for sharing their code online.
This function also implement Algorithm 1 of our submitted paper about DiPhiSeq.
This function is called by robtest.
Most users don't need to call this function directly.</h2><span id='topic+robnb'></span>

<h3>Description</h3>

<p>Calculates the estimate and standard error of beta and phi.
It takes as input counts from one group of samples for a single gene.
This function is the core underlining function of the whole package.
A significant part of the code is edited based on William H. Aeberhard's glmrob.nb R function;
we appreciate them very much for sharing their code online.
This function also implement Algorithm 1 of our submitted paper about DiPhiSeq.
This function is called by robtest.
Most users don't need to call this function directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robnb(y, log.depth, c.tukey.beta = 4, c.tukey.phi = 4, phi.ini = 0.5,
  alpha = 0.2, minphi = 0.01, maxphi = 5, maxit = 30, maxit.beta = 30,
  maxit.phi = 30, tol.beta = 0.01, tol.phi = 0.005)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="robnb_+3A_y">y</code></td>
<td>
<p>A count vector.</p>
</td></tr>
<tr><td><code id="robnb_+3A_log.depth">log.depth</code></td>
<td>
<p>Vector of log(sequencing depths).</p>
</td></tr>
<tr><td><code id="robnb_+3A_c.tukey.beta">c.tukey.beta</code></td>
<td>
<p>The c value for beta in Huber function. The default value should be appropriate
for most datasets.</p>
</td></tr>
<tr><td><code id="robnb_+3A_c.tukey.phi">c.tukey.phi</code></td>
<td>
<p>The c value for phi in Huber function. The default value should be appropriate
for most datasets.</p>
</td></tr>
<tr><td><code id="robnb_+3A_phi.ini">phi.ini</code></td>
<td>
<p>The initial value of phi.</p>
</td></tr>
<tr><td><code id="robnb_+3A_alpha">alpha</code></td>
<td>
<p>A positive value for setting initial values. The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_minphi">minphi</code></td>
<td>
<p>A searching parameter for Algorithm 1 (check the algorithm for details.)
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_maxphi">maxphi</code></td>
<td>
<p>A searching parameter for Algorithm 1 (check the algorithm for details.)
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations for the outer loop.
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_maxit.beta">maxit.beta</code></td>
<td>
<p>Maximum number of iterations for the inner loop of solving beta.
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_maxit.phi">maxit.phi</code></td>
<td>
<p>Maximum number of iterations for the inner loop of solving phi.
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_tol.beta">tol.beta</code></td>
<td>
<p>The numerical tolerance of solving beta.
The default value is usually appropriate.</p>
</td></tr>
<tr><td><code id="robnb_+3A_tol.phi">tol.phi</code></td>
<td>
<p>The numerical tolerance of solving phi.
The default value is usually appropriate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains the elements:
<code>beta</code>: the estimated (log) expression.
<code>phi</code>: the estimated dispersion.
<code>fconv</code>: flag of the convergence of the search.
<code>vars</code>: the variance-covariance matrix of the estimates.
<code>sd.beta</code>: the standard error of beta.
<code>sd.phi</code>: the standard error of phi.
<code>y</code>: the input y value.
<code>log.depth</code>: log(sequencing depth).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d &lt;- runif(10, min=1, max=2)
y &lt;- rnbinom(10, size=1, mu=d*50)
res &lt;- robnb(y, log(d))
</code></pre>

<hr>
<h2 id='robtest'>Calls the robnb function to estimate the coefficients,
and then construct the statistical tests for DD and DE.
It works for a single gene. y1 and y2 are count vectors for a single gene.
diphiseq calls this function to do the calculation for each gene.
Normal users often don't need to use this function directly.</h2><span id='topic+robtest'></span>

<h3>Description</h3>

<p>Calls the robnb function to estimate the coefficients,
and then construct the statistical tests for DD and DE.
It works for a single gene. y1 and y2 are count vectors for a single gene.
diphiseq calls this function to do the calculation for each gene.
Normal users often don't need to use this function directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robtest(y1, log.depth1, y2, log.depth2, c.tukey.beta = 4, c.tukey.phi = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="robtest_+3A_y1">y1</code></td>
<td>
<p>counts from group 1. a vector.</p>
</td></tr>
<tr><td><code id="robtest_+3A_log.depth1">log.depth1</code></td>
<td>
<p>log(sequencing depths) for samples in group 1. a vector.</p>
</td></tr>
<tr><td><code id="robtest_+3A_y2">y2</code></td>
<td>
<p>counts from group 2. a vector.</p>
</td></tr>
<tr><td><code id="robtest_+3A_log.depth2">log.depth2</code></td>
<td>
<p>log(sequencing depths) for samples in group 2. a vector.</p>
</td></tr>
<tr><td><code id="robtest_+3A_c.tukey.beta">c.tukey.beta</code></td>
<td>
<p>The c value for beta in Huber function. The default value, 4, is typically
regarded as appropriate and should work for most datasets.</p>
</td></tr>
<tr><td><code id="robtest_+3A_c.tukey.phi">c.tukey.phi</code></td>
<td>
<p>The c value for phi in Huber function. The default value, 4, is typically
regarded as appropriate and should work for most datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector that contains the elements:
<code>phi1</code>: the estimated dispersion of sample group 1.
<code>phi2</code>: the estimated dispersion of sample group 2.
<code>beta1</code>: the estimated (log) expression of sample group 1.
<code>beta2</code>: the estimated (log) expression of sample group 2.
<code>statistic.phi</code>: the z statistic for DD.
<code>statistic.beta</code>: the z statistic for DE.
<code>p.value.phi</code>: the p value for DD.
<code>p.value.beta</code>: the p value for DE.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d1 &lt;- runif(10, min=1, max=2)
d2 &lt;- runif(15, min=1, max=2)
y1 &lt;- rnbinom(10, size=1, mu=d1*50)
y2 &lt;- rnbinom(15, size=1, mu=d2*50)
res &lt;- robtest(y1, log(d1), y2, log(d2))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
