<!DOCTYPE html><html lang="en"><head><title>Help for package directPA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {directPA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#directPA-package'>
<p>Direction Pathway Analysis Package</p></a></li>
<li><a href='#directExplorer2d'><p>Batch Direction Analysis in 2-dimentional space</p></a></li>
<li><a href='#directPA'><p>Direction Analysis for Pathways</p></a></li>
<li><a href='#geneStats'><p>Molecule Level Statistics</p></a></li>
<li><a href='#HEK'><p>Phosphoproteomics of HEK-293E</p></a></li>
<li><a href='#kinasePA'><p>Direction Analysis for Kinases</p></a></li>
<li><a href='#Pathways.KEGG'><p>KEGG pathway annotations</p></a></li>
<li><a href='#Pathways.reactome'><p>Reactome pathway annotations</p></a></li>
<li><a href='#pathwayStats'><p>Pathway Level Statistics</p></a></li>
<li><a href='#perturbPlot2d'><p>Perturbation Plot</p></a></li>
<li><a href='#perturbPlot3d'><p>Perturbation Plot 3D</p></a></li>
<li><a href='#PhosphoELM.human'><p>PhosphoELM annotations for human</p></a></li>
<li><a href='#PhosphoELM.mouse'><p>PhosphoELM annotations for mouse</p></a></li>
<li><a href='#PhosphoSite.human'><p>PhosphoSitePlus annotations for human</p></a></li>
<li><a href='#PhosphoSite.mouse'><p>PhosphoSitePlus annotations for mouse</p></a></li>
<li><a href='#PM'><p>Plasma Membrame Protoemics Data</p></a></li>
<li><a href='#rotate2d'><p>Polar Coordinates Rotation</p></a></li>
<li><a href='#rotate3d'><p>Spherical Coordinates Rotation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Direction Analysis for Pathways and Kinases</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Pengyi Yang &amp; Ellis Patrick</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pengyi Yang &lt;yangpy7@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Direction analysis is a set of tools designed to identify
    combinatorial effects of multiple treatments/conditions on pathways
    and kinases profiled by microarray, RNA-seq, proteomics, or phosphoproteomics
    data. See Yang P et al (2014) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtt616">doi:10.1093/bioinformatics/btt616</a>&gt;; and Yang P et al. (2016) &lt;<a href="https://doi.org/10.1002%2Fpmic.201600068">doi:10.1002/pmic.201600068</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.10.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, graphics, stats, plotly, calibrate</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-16 03:31:20 UTC; pyan6380</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-16 20:30:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
</table>
<hr>
<h2 id='directPA-package'>
Direction Pathway Analysis Package
</h2><span id='topic+directPA-package'></span>

<h3>Description</h3>

<p>The directPA-package is designed for analysing pathways in experiments with multiple perturbations. The combination effects of
different treatments are tested by rotating polar coordinates in two-dimentional space when the experiment contains two perturbations
and corresponding controls, or spherical coordinates in three-dimensional space when the experiment contains three perturbations and 
corresponding controls.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> directPA</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.5.1</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Pengyi Yang &lt;yangpy7@gmail.com&gt; &amp; Ellis Patrick &lt;epatrick87@gmail.com&gt;
</p>


<h3>References</h3>

<p>Pengyi Yang, Ellis Patrick, Shi-Xiong Tan, Daniel J. Fazakerley, James Burchfield, Christopher Gribben, Matthew J. Prior, 
David E. James, Yee Hwa Yang, Direction pathway analysis of large-scale proteomics data reveals novel features of the insulin action pathway, Bioinformatics, 30(6), 808-814, 2014.
</p>

<hr>
<h2 id='directExplorer2d'>Batch Direction Analysis in 2-dimentional space</h2><span id='topic+directExplorer2d'></span>

<h3>Description</h3>

<p>Rotate to the direction of interest in polar coordinates by degree (e.g. pi/4).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>directExplorer2d(Tc, annotation=NULL, gene.method="OSP", 
path.method="Stouffer", top=10, nd=8, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="directExplorer2d_+3A_tc">Tc</code></td>
<td>
<p>a numeric matrix with 2 columns. The rows are genes or phosphorylation sites and the columns 
are treatments vs control statistics.</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_annotation">annotation</code></td>
<td>
<p>a list with names correspond to pathways or kinases and elements correspond to 
genes or substrates belong to each pathway or kinase, respectively.</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_gene.method">gene.method</code></td>
<td>
<p>the method to be used for integrating statistics across treatments for each gene
or phosphorylation site.  Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is OSP.</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_path.method">path.method</code></td>
<td>
<p>the method to be used for integrating statistics of all genes or phosphorylation
sites that belongs to a pathway or kinase. Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is Stouffer.</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_top">top</code></td>
<td>
<p>the number of entries to be highlighted in the plot.</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_nd">nd</code></td>
<td>
<p>the number of directions to plot (4 or 8)</p>
</td></tr>
<tr><td><code id="directExplorer2d_+3A_...">...</code></td>
<td>
<p>parameters for controlling the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The the list of enrichment analysis in tables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the phosphoproteomics dataset
data(HEK)

# load the kinase-substrate annoations
data(PhosphoSite)

# test enrichment on 8 directions in polar coordinate system.
bda &lt;- directExplorer2d(Tc=HEK, annotation=PhosphoSite.mouse)

# the direction are denoted as follow for the two treatments vs control:
# ++: up-regulated in both treatments
# +*: up-regulated in the first treatment and unchanged in the second treatment
# +-: up-regulated in the first treatment and down-regulated in the second treatment
# *-: unchanged in the first treatment and down-regulated in the second treatment
# --: down-regulated in both treatments
# -*: down-regulated in the first treatment and unchanged in the second treatment
# -+: down-regulated in the first treatment and up-regulated in the second treatment
# *+: unchanged in the first treatment and up-regulated in the second treatment

# sort the most enriched phosphorylation sites and kinases on down-regulaiton from both 
# treatments (i.e. "--") and displa the top-10 entries
bda$gene.tab[order(bda$gene.tab[,"--"]),][1:10,]
bda$path.tab[order(bda$path.tab[,"--"]),][1:10,]

</code></pre>

<hr>
<h2 id='directPA'>Direction Analysis for Pathways</h2><span id='topic+directPA'></span>

<h3>Description</h3>

<p>The main function of direction Analysis. This function takes in a matrix of test statistics with 
two (2-dimensional space) or three (3-dimensional space) columns, the direction of interests, and 
the annotation list such as pathway annotation, and test for enrichment of pathways on the specified 
direction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>directPA(Tc, direction, annotation, minSize=5, gene.method="OSP", 
path.method="Stouffer", visualize=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="directPA_+3A_tc">Tc</code></td>
<td>
<p>a numeric matrix. Rows are genes and columns are treatments vs control statistics.</p>
</td></tr>
<tr><td><code id="directPA_+3A_direction">direction</code></td>
<td>
<p>the direction to be tested for enrichment. Either specified as a degree for 
two-dimensional analysis or as contrast (in a triplet) for three-dimensional analysis.</p>
</td></tr>
<tr><td><code id="directPA_+3A_annotation">annotation</code></td>
<td>
<p>a list with names correspond to pathways and elements correspond to genes belong to 
each pathway, respectively.</p>
</td></tr>
<tr><td><code id="directPA_+3A_minsize">minSize</code></td>
<td>
<p>the size of annotation groups to be considered for calculating enrichment. Groups 
that are smaller than the minSize will be removed from the analysis.</p>
</td></tr>
<tr><td><code id="directPA_+3A_gene.method">gene.method</code></td>
<td>
<p>the method to be used for integrating statistics across treatments for each gene. 
Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is OSP.</p>
</td></tr>
<tr><td><code id="directPA_+3A_path.method">path.method</code></td>
<td>
<p>the method to be used for integrating statistics of all genes that belongs to a 
pathway. Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is Stouffer.</p>
</td></tr>
<tr><td><code id="directPA_+3A_visualize">visualize</code></td>
<td>
<p>whether to visualize the plot.</p>
</td></tr>
<tr><td><code id="directPA_+3A_...">...</code></td>
<td>
<p>other visualization parameters to pass on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list that contains directional p-values for each gene and directional enrichment for each pathway.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load the proteomics dataset
data(PM)

# load pathway annotations
data(Pathways)

# display reactome pathways. Could be replaced by any other pathway databases
Pathways.reactome[1:5]

# direction pathway analysis in 3-dimensional space. Implemnted as rotating by contrast
# (1) test combined effect of all 3 treatments (stimulation and inhibitions) vs control (basal) 
# on the original direction.
dPA &lt;- directPA(Tc=PM, direction=c(1,1,1), annotation=Pathways.reactome)
dPA$gst[order(unlist(dPA$gst[,1])),][1:20,]
# rank substrates on the direciton of interest
sort(dPA$gene.pvalues)[1:20]

# (2) test combined effect of all 3 treatments vs controls on direction c(1,-1, 0)
# this rotates Ins by 0 degree, Wmn by 90 degree, and MK by 45 degree.
dPA &lt;- directPA(Tc=PM, direction=c(1,-1,0), annotation=Pathways.reactome)
dPA$gst[order(unlist(dPA$gst[,1])),][1:20,]

# (3) test combined effect of all 3 perturbations vs controls on direction c(1,-1, 1)
# this rotates Ins by 0 degree, Wmn by 90 degree, and MK by 0 degree.
dPA &lt;- directPA(Tc=PM, direction=c(1,-1,1), annotation=Pathways.reactome)
dPA$gst[order(unlist(dPA$gst[,1])),][1:20,]

</code></pre>

<hr>
<h2 id='geneStats'>Molecule Level Statistics</h2><span id='topic+geneStats'></span>

<h3>Description</h3>

<p>Takes a vector of statistics with each element corresponds to a treatment vs control comparison, 
and calculates a combined statistics accross multiple treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneStats(T, method="OSP")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geneStats_+3A_t">T</code></td>
<td>
<p>a vector of statistics (z-scores converted) with each element correspond to a treatment vs control 
comparison.</p>
</td></tr>
<tr><td><code id="geneStats_+3A_method">method</code></td>
<td>
<p>the p-value integration method for combining accross multiple treatments. Available methods 
are Stouffer, OSP, Fisher, and maxP. The default method is OSP.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a p-value after integration across treatments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the example data
data(PM)

# convert statistics into z-scores
PM.zscores &lt;- apply(PM, 2, function(x){qnorm(rank(x)/(nrow(PM)+1))})

# Rotate the matrix by contrast 1, -1, -1 (i.e. up-regulation, down-regulation, dow-regulation).
PM.rotated &lt;- rotate3d(PM.zscores, contrast = c(1, -1, -1))

# combine rotated statistics across treatments
gene.pvalues &lt;- apply(PM.rotated, 1, geneStats)

</code></pre>

<hr>
<h2 id='HEK'>Phosphoproteomics of HEK-293E</h2><span id='topic+HEK'></span>

<h3>Description</h3>

<p>The data object contains all quantified phosphoryaltion sites in HEK-293E cells with Rapamycin or Torin1 treatment vs insulin stimulation. See
Hsu et al. Science, 332(6035), 1317-1322, 2011, for more details.
</p>

<hr>
<h2 id='kinasePA'>Direction Analysis for Kinases</h2><span id='topic+kinasePA'></span>

<h3>Description</h3>

<p>This is a wrapper for runing directPA for kinase perturbation analysis (kinasePA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinasePA(Tc, direction, annotation, minSize=5, substrate.method="OSP", 
kinase.method="Stouffer", visualize=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kinasePA_+3A_tc">Tc</code></td>
<td>
<p>a numeric matrix. The columns are phosphorylation sites and the columns are treatments vs 
control statistics.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_direction">direction</code></td>
<td>
<p>the direction to be tested for enrichment. Either specified as a degree for 
two-dimensional analysis or as contrast (in a triplet) for three-dimensional analysis.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_annotation">annotation</code></td>
<td>
<p>a list with names correspond to kinases and elements correspond to substrates belong 
to each kinase, respectively.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_minsize">minSize</code></td>
<td>
<p>the size of annotation groups to be considered for calculating enrichment. Groups 
that are smaller than the minSize will be removed from the analysis.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_substrate.method">substrate.method</code></td>
<td>
<p>the method to be used for integrating statistics across treatments for each 
substrate (phosphorylation site). Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is OSP.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_kinase.method">kinase.method</code></td>
<td>
<p>the method to be used for integrating statistics of all phosphorylation
sites that belongs to a kinase. Available methods are Stouffer, OSP, Fisher, and maxP. 
Default method is Stouffer.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_visualize">visualize</code></td>
<td>
<p>whether to visualize the plot.</p>
</td></tr>
<tr><td><code id="kinasePA_+3A_...">...</code></td>
<td>
<p>other visualization parameters to pass on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list that contains directional p-values for each substrate and directional enrichment for 
each kinase.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load the phosphoproteomics dataset
data(HEK)

# load the kinase-substrate annoations
data(PhosphoSite)

# direction pathway analysis in 2-dimensional space. Implemented as rotating by degree 
# (1) test combined effect of Torin1 and Rapamycin vs insul both on "down-regulation"
# (180 degree to original direction)
kPA &lt;- kinasePA(Tc=HEK, direction=pi, annotation=PhosphoSite.mouse)
kPA$kinase[order(unlist(kPA$kinase[,1])),][1:20,]
# rank substrates on the direciton of interest
sort(kPA$substrate.pvalues)[1:20]

# (2) test combined effect of Torin1 and Rapamycin vs insul on "no change and down-regulation"
# (135 degree to the original direction) 
kPA &lt;- kinasePA(Tc=HEK, direction=pi*3/4, annotation=PhosphoSite.mouse)
kPA$kinase[order(unlist(kPA$kinase[,1])),][1:20,]

# (3) test combined effect of Torin1 and Rapamycin vs insul on "down-regulation and no change"
# (225 degree to the original direction) 
kPA &lt;- kinasePA(Tc=HEK, direction=pi*5/4, annotation=PhosphoSite.mouse)
kPA$kinase[order(unlist(kPA$kinase[,1])),][1:20,]

</code></pre>

<hr>
<h2 id='Pathways.KEGG'>KEGG pathway annotations</h2><span id='topic+Pathways.KEGG'></span>

<h3>Description</h3>

<p>The data object contains the annotations of KEGG pathways.
</p>

<hr>
<h2 id='Pathways.reactome'>Reactome pathway annotations</h2><span id='topic+Pathways.reactome'></span>

<h3>Description</h3>

<p>The data object contains the annotations of reactome pathways.
</p>

<hr>
<h2 id='pathwayStats'>Pathway Level Statistics</h2><span id='topic+pathwayStats'></span>

<h3>Description</h3>

<p>Takes a vector of statistics with each element corresponds to a gene or phosphorylation site, 
and calculates a combined statistics for those that belong to the same pathway or kinase.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathwayStats(PGs, T, minSize=5, method="Stouffer")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathwayStats_+3A_pgs">PGs</code></td>
<td>
<p>an array of names indicating genes or substrates that belong to a given pathway or kinase.</p>
</td></tr>
<tr><td><code id="pathwayStats_+3A_t">T</code></td>
<td>
<p>a vector of statistics (z-scores converted) with each element correspond to a gene or 
phosphorylation site that belong to the same pathway or kinase.</p>
</td></tr>
<tr><td><code id="pathwayStats_+3A_minsize">minSize</code></td>
<td>
<p>the size of annotation groups to be considered for calculating enrichment. Groups 
that are smaller than the minSize will be removed from the analysis.</p>
</td></tr>
<tr><td><code id="pathwayStats_+3A_method">method</code></td>
<td>
<p>the p-value integration method for combining accross multiple treatments. Available methods 
are Stouffer, OSP, Fisher, and maxP. The default method is Stouffer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a doublet corresponding to the enrichment after integration across all genes or substrates 
that belong to the same pathway or kinase, and the size of the mapped genes or substrates to that pathway
or kinase.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the example data
data(PM)

# load pathway annotations
data(Pathways)

# convert statistics into z-scores
PM.zscores &lt;- apply(PM, 2, function(x){qnorm(rank(x)/(nrow(PM)+1))})

# Rotate the matrix by contrast 1, -1, -1 (i.e. up-regulation, down-regulation, dow-regulation).
PM.rotated &lt;- rotate3d(PM.zscores, contrast = c(1, -1, -1))

# combine rotated statistics across treatments
gene.pvalues &lt;- apply(PM.rotated, 1, geneStats)

# compute statistics for all reactome pathways
gene.zscores &lt;- qnorm(gene.pvalues, lower.tail = FALSE)
gst &lt;- t(sapply(Pathways.reactome, pathwayStats, gene.zscores))

</code></pre>

<hr>
<h2 id='perturbPlot2d'>Perturbation Plot</h2><span id='topic+perturbPlot2d'></span>

<h3>Description</h3>

<p>This function takes in a matrix of test statistics with two columns (2-dimensional space) and the 
annotation list such as pathway annotation or kinase-substrate annotation, and visualize the enrichment
of pathways or kinases in direction specific manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perturbPlot2d(Tc, annotation, minSize=5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perturbPlot2d_+3A_tc">Tc</code></td>
<td>
<p>a numeric matrix. The columns are genes or phosphorylation sites and the columns are treatments 
vs control statistics.</p>
</td></tr>
<tr><td><code id="perturbPlot2d_+3A_annotation">annotation</code></td>
<td>
<p>a list with names correspond to pathways or kinases and elements correspond to genes or
substrates belong to each pathway or kinase, respectively.</p>
</td></tr>
<tr><td><code id="perturbPlot2d_+3A_minsize">minSize</code></td>
<td>
<p>the size of annotation groups to be considered for calculating enrichment. Groups 
that are smaller than the minSize will be removed from the analysis.</p>
</td></tr>
<tr><td><code id="perturbPlot2d_+3A_...">...</code></td>
<td>
<p>parameters for controling the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of coordinates for pathways or kinases
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the phosphoproteomics dataset
data(HEK)

# load the kinase-substrate annoations
data(PhosphoSite)

perturbPlot2d(Tc=HEK, annotation=PhosphoSite.mouse, cex=3)

</code></pre>

<hr>
<h2 id='perturbPlot3d'>Perturbation Plot 3D</h2><span id='topic+perturbPlot3d'></span>

<h3>Description</h3>

<p>This function takes in a matrix of test statistics with two columns (3-dimensional space) and the 
annotation list such as pathway annotation or kinase-substrate annotation, and visualize the enrichment
of pathways or kinases in direction specific manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perturbPlot3d(Tc, annotation, minSize=5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perturbPlot3d_+3A_tc">Tc</code></td>
<td>
<p>a numeric matrix. The columns are genes or phosphorylation sites and the columns are treatments 
vs control statistics.</p>
</td></tr>
<tr><td><code id="perturbPlot3d_+3A_annotation">annotation</code></td>
<td>
<p>a list with names correspond to pathways or kinases and elements correspond to genes or
substrates belong to each pathway or kinase, respectively.</p>
</td></tr>
<tr><td><code id="perturbPlot3d_+3A_minsize">minSize</code></td>
<td>
<p>the size of annotation groups to be considered for calculating enrichment. Groups 
that are smaller than the minSize will be removed from the analysis.</p>
</td></tr>
<tr><td><code id="perturbPlot3d_+3A_...">...</code></td>
<td>
<p>parameters for controling the plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of coordinates for pathways or kinases
</p>

<hr>
<h2 id='PhosphoELM.human'>PhosphoELM annotations for human</h2><span id='topic+PhosphoELM.human'></span>

<h3>Description</h3>

<p>The data object contains the annotations of kinases and their conrresponding substrates as phosphorylation sites in human. It is extracted
from the PhosphoELM database. For details of PhosphoELM, please refer to the article: Phospho.ELM: a database of phosphorylation sites&ndash;update 2008.
Diella, F., Gould, C.M., Chica, C., Via, A. &amp; Gibson, T.J. Nucleic Acids Res. 2008 Jan;36(Database issue):D240-4.
</p>

<hr>
<h2 id='PhosphoELM.mouse'>PhosphoELM annotations for mouse</h2><span id='topic+PhosphoELM.mouse'></span>

<h3>Description</h3>

<p>The data object contains the annotations of kinases and their conrresponding substrates as phosphorylation sites in mouse. It is extracted
from the PhosphoELM database. For details of PhosphoELM, please refer to the article: Phospho.ELM: a database of phosphorylation sites&ndash;update 2008.
Diella, F., Gould, C.M., Chica, C., Via, A. &amp; Gibson, T.J. Nucleic Acids Res. 2008 Jan;36(Database issue):D240-4.
</p>

<hr>
<h2 id='PhosphoSite.human'>PhosphoSitePlus annotations for human</h2><span id='topic+PhosphoSite.human'></span>

<h3>Description</h3>

<p>The data object contains the annotations of kinases and their conrresponding substrates as phosphorylation sites in human. It is extracted
from the PhosphoSitePlus database. For details of PhosphoSitePlus, please refer to the article: Hornbeck et al. Nucleic Acids Res., 40:D261-70, 2012.
</p>

<hr>
<h2 id='PhosphoSite.mouse'>PhosphoSitePlus annotations for mouse</h2><span id='topic+PhosphoSite.mouse'></span>

<h3>Description</h3>

<p>The data object contains the annotations of kinases and their conrresponding substrates as phosphorylation sites in mouse. It is extracted
from the PhosphoSitePlus database. For details of PhosphoSitePlus, please refer to the article: Hornbeck et al. Nucleic Acids Res., 40:D261-70, 2012.
</p>

<hr>
<h2 id='PM'>Plasma Membrame Protoemics Data</h2><span id='topic+PM'></span>

<h3>Description</h3>

<p>The data object contains all quantified plasma membrame proteins with insulin simulation with and without prior treatments of MK or Wmn vs basal 
condition in mouse 3T3L1 cell lines. Please refer to the article: Yang et al. Bioinformatics, 30(6):808-14, 2014.
</p>

<hr>
<h2 id='rotate2d'>Polar Coordinates Rotation</h2><span id='topic+rotate2d'></span>

<h3>Description</h3>

<p>Rotate to the direction of interest in polar coordinates by degree (e.g. pi/4).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotate2d(T, degree = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rotate2d_+3A_t">T</code></td>
<td>
<p>a numeric matrix (normally z-score converted) with 2 columns. The rows are genes or phosphorylation
sites and the columns are treatments vs control statistics.</p>
</td></tr>
<tr><td><code id="rotate2d_+3A_degree">degree</code></td>
<td>
<p>the direction to be tested for enrichment. Specified as degree from to the original direction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A rotated matrix with respect to the direction of interest.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the phosphoproteomics dataset
data(HEK)

# convert statistics into z-scores
HEK.zscores &lt;- apply(HEK, 2, function(x){qnorm(rank(x)/(nrow(HEK)+1))})

# Rotate the matrix by 1/2 pi (i.e. down-regulation, dow-regulation).
HEK.rotated &lt;- rotate2d(HEK.zscores, degree = pi/2)

</code></pre>

<hr>
<h2 id='rotate3d'>Spherical Coordinates Rotation</h2><span id='topic+rotate3d'></span>

<h3>Description</h3>

<p>Rotate to the direction of interest in spherical coordinates by contrasts (e.g. 1, -1, -1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotate3d(T, contrast)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rotate3d_+3A_t">T</code></td>
<td>
<p>a numeric matrix (normally z-score converted) with 3 columns. The rows are genes or phosphorylation
sites and the columns are treatments vs control statistics.</p>
</td></tr>
<tr><td><code id="rotate3d_+3A_contrast">contrast</code></td>
<td>
<p>the direction to be tested for enrichment. Specified as contrast in a triplet (see example).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A rotated matrix with respect to the direction of interest.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load the example data
data(PM)

# convert statistics into z-scores
PM.zscores &lt;- apply(PM, 2, function(x){qnorm(rank(x)/(nrow(PM)+1))})

# Rotate the matrix by contrast 1, -1, -1 (i.e. up-regulation, down-regulation, dow-regulation).
PM.rotated &lt;- rotate3d(PM.zscores, contrast = c(1, -1, -1))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
