<!DOCTYPE html><html><head><title>Help for package disclapmix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {disclapmix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clusterdist'><p>Calculate distance between clusters</p></a></li>
<li><a href='#clusterprob'><p>Cluster origin probabilities for haplotypes</p></a></li>
<li><a href='#contributor_pairs'><p>Contributor pairs from a 2 person mixture</p></a></li>
<li><a href='#danes'><p>Y-STR haplotypes</p></a></li>
<li><a href='#disclapmix'><p>Discrete Laplace mixture inference using the EM algorithm</p></a></li>
<li><a href='#disclapmix_adaptive'><p>Adaptive fitting</p></a></li>
<li><a href='#disclapmix_robust'><p>Robust fitting</p></a></li>
<li><a href='#disclapmix-package'><p>disclapmix</p></a></li>
<li><a href='#generate_mixture'><p>Generate a mixture</p></a></li>
<li><a href='#get_rank'><p>Get rank of pair</p></a></li>
<li><a href='#haplotype_diversity'><p>Calculate haplotype diversity from a disclapmixfit</p></a></li>
<li><a href='#plot.disclapmixfit'><p>Plot a disclapmixfit</p></a></li>
<li><a href='#plot.ranked_contrib_pairs'><p>Plot ranked contributor pairs</p></a></li>
<li><a href='#predict.disclapmixfit'><p>Predict from a disclapmixfit</p></a></li>
<li><a href='#print.contrib_pairs'><p>Print contributor pairs</p></a></li>
<li><a href='#print.disclapmixfit'><p>Print a disclapmixfit</p></a></li>
<li><a href='#print.ranked_contrib_pairs'><p>Print ranked contributor pairs</p></a></li>
<li><a href='#rank_contributor_pairs'><p>Separate a 2 person mixture</p></a></li>
<li><a href='#simulate.disclapmixfit'><p>Simulate from a disclapmixfit</p></a></li>
<li><a href='#summary.disclapmixfit'><p>Summary of a disclapmixfit</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Discrete Laplace Mixture Inference using the EM Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>1.7.4</td>
</tr>
<tr>
<td>Author:</td>
<td>Mikkel Meyer Andersen [aut, cre],
  Poul Svante Eriksen [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Make inference in a mixture of discrete Laplace distributions using the EM algorithm. This can e.g. be used for modelling the distribution of Y chromosomal haplotypes as described in [1, 2] (refer to the URL section).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | file LICENSE</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppProgress</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11), disclap (&ge; 1.4), cluster (&ge; 1.14.4), MASS,
stats, graphics, methods, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, ggplot2, gridExtra, ggdendro, scales, seriation,
fwsim, testthat, rmarkdown</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mikldk/disclapmix/issues">https://github.com/mikldk/disclapmix/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://dx.doi.org/10.1016/j.jtbi.2013.03.009">http://dx.doi.org/10.1016/j.jtbi.2013.03.009</a>
<a href="https://arxiv.org/abs/1304.2129">https://arxiv.org/abs/1304.2129</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-29 13:57:15 UTC; mikl</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-29 14:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='clusterdist'>Calculate distance between clusters</h2><span id='topic+clusterdist'></span>

<h3>Description</h3>

<p><code>clusterdist</code> calculates the distance between each pair of clusters.
The distance measure is based on a symmetric Kullback-Leibler divergence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterdist(fit, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterdist_+3A_fit">fit</code></td>
<td>
<p>A <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.</p>
</td></tr>
<tr><td><code id="clusterdist_+3A_...">...</code></td>
<td>
<p>Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance matrix
</p>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix-package">disclapmix-package</a></code> <code><a href="#topic+disclapmix">disclapmix</a></code>
<code><a href="#topic+disclapmixfit">disclapmixfit</a></code> <code><a href="#topic+clusterprob">clusterprob</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code> <code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code>

<code><a href="disclap.html#topic+disclap-package">disclap</a></code>
</p>

<hr>
<h2 id='clusterprob'>Cluster origin probabilities for haplotypes</h2><span id='topic+clusterprob'></span>

<h3>Description</h3>

<p><code>clusterprob</code> calculates the cluster origin probabilities for
haplotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clusterprob(fit, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clusterprob_+3A_fit">fit</code></td>
<td>
<p>A <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.</p>
</td></tr>
<tr><td><code id="clusterprob_+3A_newdata">newdata</code></td>
<td>
<p>The haplotypes to predict the cluster origin probabilities
for.</p>
</td></tr>
<tr><td><code id="clusterprob_+3A_...">...</code></td>
<td>
<p>Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix where the rows correspond to the rows in <code>newdata</code> and
the sum of each row is 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix-package">disclapmix-package</a></code> <code><a href="#topic+disclapmix">disclapmix</a></code>
<code><a href="#topic+disclapmixfit">disclapmixfit</a></code> <code><a href="#topic+clusterdist">clusterdist</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code> <code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code>

<code><a href="disclap.html#topic+disclap-package">disclap</a></code>
</p>

<hr>
<h2 id='contributor_pairs'>Contributor pairs from a 2 person mixture</h2><span id='topic+contributor_pairs'></span>

<h3>Description</h3>

<p>Get all possible contributor pairs from a 2 person mixture
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contributor_pairs(mixture)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contributor_pairs_+3A_mixture">mixture</code></td>
<td>
<p>A list of integer vectors. The k'th element in the list is an
integer vector with the alleles in the mixture at locus k.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>contrib_pairs</code> object that is a unordered list of pairs.
Note, that contributor order is disregarded so that each contributor pair is
only present once (and not twice as would be the case if taking order into
consideration). See example usage at <code><a href="#topic+rank_contributor_pairs">rank_contributor_pairs</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rank_contributor_pairs">rank_contributor_pairs</a></code> <code><a href="#topic+generate_mixture">generate_mixture</a></code>
<code><a href="#topic+disclapmix-package">disclapmix-package</a></code> <code><a href="#topic+disclapmix">disclapmix</a></code>
<code><a href="#topic+disclapmixfit">disclapmixfit</a></code> <code><a href="#topic+clusterprob">clusterprob</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code> <code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code>

<code><a href="disclap.html#topic+disclap-package">disclap</a></code>
</p>

<hr>
<h2 id='danes'>Y-STR haplotypes</h2><span id='topic+danes'></span>

<h3>Description</h3>

<p>185 Y-STR 10 loci haplotypes
</p>


<h3>Format</h3>

<p>A data frame with 185 observations on the following 10 loci
(<code>n</code> is the number of times each haplotype has been observed)
</p>
 <dl>
<dt>DYS19</dt><dd></dd> <dt>DYS389I</dt><dd></dd>
<dt>DYS389II</dt><dd></dd> <dt>DYS390</dt><dd></dd> <dt>DYS391</dt><dd></dd>
<dt>DYS392</dt><dd></dd> <dt>DYS393</dt><dd></dd> <dt>DYS437</dt><dd></dd>
<dt>DYS438</dt><dd></dd> <dt>DYS439</dt><dd></dd> <dt>n</dt><dd></dd> </dl>



<h3>Source</h3>

<p>&rdquo;Y-chromosome STR haplotypes Danes&rdquo; by Hallenberg et al (2005),
http://www.sciencedirect.com/science/article/pii/S0379073805000095
</p>

<hr>
<h2 id='disclapmix'>Discrete Laplace mixture inference using the EM algorithm</h2><span id='topic+disclapmix'></span><span id='topic+disclapmixfit'></span>

<h3>Description</h3>

<p><code>disclapmix</code> makes inference in a mixture of Discrete Laplace
distributions using the EM algorithm. After the EM algorithm has converged,
the centers are moved if the marginal likelihood increases by doing so. And
then the EM algorithm is run again. This continues until the centers are not
moved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disclapmix(
  x,
  clusters,
  init_y = NULL,
  iterations = 100L,
  eps = 0.001,
  verbose = 0L,
  glm_method = "internal_coef",
  glm_control_maxit = 50L,
  glm_control_eps = 1e-06,
  init_y_method = "pam",
  init_v = NULL,
  ret_x = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="disclapmix_+3A_x">x</code></td>
<td>
<p>Dataset.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_clusters">clusters</code></td>
<td>
<p>The number of clusters/components to fit the model for.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_init_y">init_y</code></td>
<td>
<p>Initial central haplotypes, if NULL, these will be estimated
as described under the <code>init_y_method</code> argument.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_iterations">iterations</code></td>
<td>
<p>Maximum number of iterations in the EM-algorithm.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_eps">eps</code></td>
<td>
<p>Convergence stop criteria in the EM algorithm which is compared
to <code class="reqn">\frac{\max \{ v_{new} - v_{old} \}}{\max \{ v_{old} \}}</code>, where <code>v</code> is a matrix of each
observation's probability of belonging to a certain center.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 2 (both including): 0 for silent, 2 for extra
verbose.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_glm_method">glm_method</code></td>
<td>
<p><code>internal_coef</code>, <code>internal_dev</code> or
<code>glm.fit</code>. Please see details.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_glm_control_maxit">glm_control_maxit</code></td>
<td>
<p>Integer giving the maximal number of IWLS
iterations.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_glm_control_eps">glm_control_eps</code></td>
<td>
<p>Positive convergence tolerance epsilon; the
iterations converge when <code>|x - x_{old}|/(|x| + 0.1) &lt; epsilon</code>, where
<code>x = beta_correction</code> for <code>internal_coef</code> and <code>x = deviance</code>
otherwise.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_init_y_method">init_y_method</code></td>
<td>
<p>Which cluster method to use for finding initial central
haplotypes, y: <code>pam</code> (recommended) or <code>clara</code>. Ignored if
<code>init_y</code> is supplied.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_init_v">init_v</code></td>
<td>
<p>Matrix with 'nrow(x)' rows and 'clusters' columns specifying 
initial posterior probabilities to get EM started, if
none specified, then 'matrix(1/clusters, nrow = nrow(x), ncol = clusters)' is used.</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_ret_x">ret_x</code></td>
<td>
<p>Return data 'x'</p>
</td></tr>
<tr><td><code id="disclapmix_+3A_...">...</code></td>
<td>
<p>Used to detect obsolete usage (when using parameters
<code>centers</code>, <code>use.parallel</code>, <code>calculate.logLs</code> or
<code>plots.prefix</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>glm_method</code>: <code>internal_coef</code> is the fastest as it uses the
relative changes in the coefficients as a stopping criterium, hence it does
not need to compute the deviance until the very end. In normal situations,
it would not be a problem to use this method. <code>internal_dev</code> is the
reasonably fast method that uses the deviance as a stopping criterium (like
<code>glm.fit</code>). <code>glm.fit</code> to use the traditional <code>glm.fit</code> IWLS
implementation and is slow compared to the other two methods.
</p>
<p><code>init_y_method</code>: For <code>init_y_method = 'clara'</code>, the sampling
parameters are: <code>samples = 100</code>, <code>sampsize =
min(ceiling(nrow(x)/2), 100 + 2*clusters)</code> and the random number generator
in R is used.
</p>


<h3>Value</h3>

<p>A <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object: </p>

<dl>
<dt>list(&quot;glm_method&quot;)</dt><dd><p>The supplied GLM method.</p>
</dd>
<dt>list(&quot;init_y&quot;)</dt><dd><p>The supplied initial central haplotypes,
<code>init_y</code>.</p>
</dd> 
<dt>list(&quot;init_y_method&quot;)</dt><dd><p>The supplied method for
choosing initial central haplotypes (only used if <code>init_y</code> is
<code>NULL</code>).</p>
</dd>
<dt>list(&quot;converged&quot;)</dt><dd><p>Whether the estimation converged or not.</p>
</dd>
<dt>list(&quot;x&quot;)</dt><dd><p>Dataset used to fit the model if 'ret_x' is 'TRUE', else 'NULL'.</p>
</dd> 
<dt>list(&quot;y&quot;)</dt><dd><p>The
central haplotypes, <code>y</code>.</p>
</dd> <dt>list(&quot;tau&quot;)</dt><dd><p>The prior probabilities of
belonging to a cluster, <code>tau</code>.</p>
</dd> <dt>list(&quot;v_matrix&quot;)</dt><dd><p>The matrix
<code>v</code> of each observation's probability of belonging to a certain
cluster. The rows are in the same order as the observations in <code>x</code> used
to generate this fit.</p>
</dd> <dt>list(&quot;disclap_parameters&quot;)</dt><dd><p>A matrix with the
estimated dicrete Laplace parameters.</p>
</dd> <dt>list(&quot;glm_coef&quot;)</dt><dd><p>The
coefficients from the last GLM fit (used to calculate
<code>disclap_parameters</code>).</p>
</dd>
<dt>list(&quot;model_observations&quot;)</dt><dd><p>Number of observations.</p>
</dd>
<dt>list(&quot;model_parameters&quot;)</dt><dd><p>Number of parameters in the model.</p>
</dd>
<dt>list(&quot;iterations&quot;)</dt><dd><p>Number of iterations performed in total (including
moving centers and re-estimating using the EM algorithm).</p>
</dd>
<dt>list(&quot;logL_full&quot;)</dt><dd><p>Full log likelihood of the final model.</p>
</dd>
<dt>list(&quot;logL_marginal&quot;)</dt><dd><p>Marginal log likelihood of the final model.</p>
</dd>
<dt>list(&quot;BIC_full&quot;)</dt><dd><p>BIC based on the full log likelihood of the final
model.</p>
</dd> 
<dt>list(&quot;BIC_marginal&quot;)</dt><dd><p>BIC based on the marginal log likelihood
of the final model.</p>
</dd>
<dt>list(&quot;v_gain_iterations&quot;)</dt><dd><p>The gain <code class="reqn">\frac{\max \{ v_{new} -
v_{old} \}}{\max \{ v_{old} \}}</code>,
where <code>v</code> is <code>vic_matrix</code> mentioned above, during the iterations.</p>
</dd>
<dt>list(&quot;tau_iterations&quot;)</dt><dd><p>The prior probability of belonging to the
centers during the iterations.</p>
</dd>
<dt>list(&quot;logL_full_iterations&quot;)</dt><dd><p>Full log likelihood of the models during
the iterations (only calculated when <code>verbose = 2L</code>).</p>
</dd>
<dt>list(&quot;logL_marginal_iterations&quot;)</dt><dd><p>Marginal log likelihood of the
models during the iterations (only calculated when <code>verbose = 2L</code>).</p>
</dd>
<dt>list(&quot;BIC_full_iterations&quot;)</dt><dd><p>BIC based on full log likelihood of the
models during the iterations (only calculated when <code>verbose = 2L</code>).</p>
</dd>
<dt>list(&quot;BIC_marginal_iterations&quot;)</dt><dd><p>BIC based on marginal log likelihood
of the models during the iterations (only calculated when <code>verbose =
2L</code>).</p>
</dd> </dl>



<h3>See Also</h3>

<p><code><a href="#topic+disclapmix-package">disclapmix-package</a></code> <code><a href="#topic+disclapmix">disclapmix</a></code>
<code><a href="#topic+disclapmixfit">disclapmixfit</a></code> <code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code>
<code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> 
<code><a href="#topic+clusterdist">clusterdist</a></code> <code><a href="#topic+clusterprob">clusterprob</a></code> <code><a href="stats.html#topic+glm.fit">glm.fit</a></code>
<code><a href="disclap.html#topic+disclap">disclap</a></code> <code><a href="cluster.html#topic+pam">pam</a></code>
<code><a href="cluster.html#topic+clara">clara</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate sample database
db &lt;- matrix(disclap::rdisclap(1000, 0.3), nrow = 250, ncol = 4)

# Add location parameters
db &lt;- sapply(1:ncol(db), function(i) as.integer(db[, i]+13+i))

head(db)

fit1 &lt;- disclapmix(db, clusters = 1L, verbose = 1L, glm_method = "glm.fit")
fit1$disclap_parameters
fit1$y

fit1b &lt;- disclapmix(db, clusters = 1L, verbose = 1L, glm_method = "internal_coef")
fit1b$disclap_parameters
fit1b$y

max(abs(fit1$disclap_parameters - fit1b$disclap_parameters))

# Generate another type of database
db2 &lt;- matrix(disclap::rdisclap(2000, 0.1), nrow = 500, ncol = 4)
db2 &lt;- sapply(1:ncol(db2), function(i) as.integer(db2[, i]+14+i))
fit2 &lt;- disclapmix(rbind(db, db2), clusters = 2L, verbose = 1L)
fit2$disclap_parameters
fit2$y
fit2$tau

</code></pre>

<hr>
<h2 id='disclapmix_adaptive'>Adaptive fitting</h2><span id='topic+disclapmix_adaptive'></span>

<h3>Description</h3>

<p>A wrapper around 'disclapmix_robust()' that instead of fitting one model 
for a given number of clusters, fits models until the best model (lowest marginal BIC) 
is in the interior (with margin 'M') of all number of clusters tried.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disclapmix_adaptive(x, margin = 5L, criteria = "BIC_marginal", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="disclapmix_adaptive_+3A_x">x</code></td>
<td>
<p>Dataset.</p>
</td></tr>
<tr><td><code id="disclapmix_adaptive_+3A_margin">margin</code></td>
<td>
<p>Fit models until there is at least this margin</p>
</td></tr>
<tr><td><code id="disclapmix_adaptive_+3A_criteria">criteria</code></td>
<td>
<p>The slot to chose the best model from (small values indicate better model)</p>
</td></tr>
<tr><td><code id="disclapmix_adaptive_+3A_...">...</code></td>
<td>
<p>Passed on to 'disclapmix_robust()' (and further to 'disclapmix()')</p>
</td></tr>
</table>


<h3>Details</h3>

<p>E.g., the best model has 3 clusters and the margin 'M = 5', then 
this function ensures that models with 1, 2, ..., 3+5 = 8 clusters 
are fitted. If e.g. then 7 is better than 3, then it continues such that 
also models with up to 7+5 = 12 clusters are fitted. 
</p>
<p>Note that models with 1-5 clusters are always fitted.
</p>


<h3>Value</h3>

<p>A list of all 'disclapmix' fits
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(danes)
db &lt;- as.matrix(danes[rep(1:nrow(danes), danes$n), 1:(ncol(danes)-1)])
fits &lt;- disclapmix_adaptive(db, margin = 5L)
fits
BICs &lt;- sapply(fits, function(x) x$BIC_marginal)
BICs
ks &lt;- sapply(fits, function(x) nrow(x$y)) # Always same as seq_along(fits)
ks
max_k &lt;- max(ks)
best_k &lt;- which.min(BICs)
max_k
best_k
max_k - best_k # = margin = 5

</code></pre>

<hr>
<h2 id='disclapmix_robust'>Robust fitting</h2><span id='topic+disclapmix_robust'></span>

<h3>Description</h3>

<p>A wrapper around 'disclapmix()' that tries to avoid errors. 
Can sometimes avoid errors with SVD problems happening with 
&lsquo;glm_method = &rsquo;internal_coef'&lsquo; and 'glm_method = &rsquo;internal_dev''.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disclapmix_robust(x, clusters, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="disclapmix_robust_+3A_x">x</code></td>
<td>
<p>Dataset.</p>
</td></tr>
<tr><td><code id="disclapmix_robust_+3A_clusters">clusters</code></td>
<td>
<p>The number of clusters/components to fit the model for.</p>
</td></tr>
<tr><td><code id="disclapmix_robust_+3A_...">...</code></td>
<td>
<p>Passed on to 'disclapmix()'</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(danes)
db &lt;- as.matrix(danes[rep(1:nrow(danes), danes$n), 1:(ncol(danes)-1)])
fit &lt;- disclapmix_robust(db, 3L)
fit

</code></pre>

<hr>
<h2 id='disclapmix-package'>disclapmix</h2><span id='topic+disclapmix-package'></span>

<h3>Description</h3>

<p>Discrete Laplace Mixture Inference using the EM Algorithm. 
A central function is <code><a href="#topic+disclapmix_adaptive">disclapmix_adaptive</a></code> (and the underlying <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>).
</p>


<h3>Author(s)</h3>

<p>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt; and 
Poul Svante Eriksen &lt;svante@math.aau.dk&gt;
</p>

<hr>
<h2 id='generate_mixture'>Generate a mixture</h2><span id='topic+generate_mixture'></span>

<h3>Description</h3>

<p>This function can generate a mixture given a list of contributors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_mixture(profiles)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_mixture_+3A_profiles">profiles</code></td>
<td>
<p>A list with profiles to mix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list, e.g. for use with <code><a href="#topic+contributor_pairs">contributor_pairs</a></code>. See
example usage at <code><a href="#topic+rank_contributor_pairs">rank_contributor_pairs</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contributor_pairs">contributor_pairs</a></code>
<code><a href="#topic+rank_contributor_pairs">rank_contributor_pairs</a></code> <code><a href="#topic+disclapmix-package">disclapmix-package</a></code>
<code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+clusterprob">clusterprob</a></code> <code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code>
<code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> 
<code><a href="disclap.html#topic+disclap-package">disclap</a></code>
</p>

<hr>
<h2 id='get_rank'>Get rank of pair</h2><span id='topic+get_rank'></span>

<h3>Description</h3>

<p>Get rank of pair
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_rank(x, haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_rank_+3A_x">x</code></td>
<td>
<p>A <code>ranked_contrib_pairs</code> object.</p>
</td></tr>
<tr><td><code id="get_rank_+3A_haplotype">haplotype</code></td>
<td>
<p>A haplotype.</p>
</td></tr>
</table>

<hr>
<h2 id='haplotype_diversity'>Calculate haplotype diversity from a disclapmixfit</h2><span id='topic+haplotype_diversity'></span>

<h3>Description</h3>

<p>Calculate haplotype diversity from a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object. The
method is based on simulating a huge database that approximates the
population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotype_diversity(object, nsim = 10000L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotype_diversity_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object, usually from a result of
a call to <code>disclapmix</code>.</p>
</td></tr>
<tr><td><code id="haplotype_diversity_+3A_nsim">nsim</code></td>
<td>
<p>number of haplotypes to generate for calculating the haplotype
diversity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The calculated haplotype diversity.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code> <code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code>

</p>

<hr>
<h2 id='plot.disclapmixfit'>Plot a disclapmixfit</h2><span id='topic+plot.disclapmixfit'></span>

<h3>Description</h3>

<p>Plot a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'disclapmixfit'
plot(x, which = 1L, clusdist = clusterdist(x), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.disclapmixfit_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object, usually from a result of a
call to <code>disclapmix</code>.</p>
</td></tr>
<tr><td><code id="plot.disclapmixfit_+3A_which">which</code></td>
<td>
<p>What plot to make. 1L = clusters and their distances.</p>
</td></tr>
<tr><td><code id="plot.disclapmixfit_+3A_clusdist">clusdist</code></td>
<td>
<p>To use previously computed cluster distances to avoid doing
the same computations twice.</p>
</td></tr>
<tr><td><code id="plot.disclapmixfit_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with discrete Laplace distributions for each cluster
and locus. Side effect: A plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>

</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(danes)
db &lt;- as.matrix(danes[rep(1:nrow(danes), danes$n), 1:(ncol(danes)-1)])
fit &lt;- disclapmix(db, clusters = 4L)
plot(fit)

</code></pre>

<hr>
<h2 id='plot.ranked_contrib_pairs'>Plot ranked contributor pairs</h2><span id='topic+plot.ranked_contrib_pairs'></span>

<h3>Description</h3>

<p>Plot ranked contributor pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ranked_contrib_pairs'
plot(x, top = NULL, ..., xlab = "Rank", ylab = "P(H1)P(H2)")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.ranked_contrib_pairs_+3A_x">x</code></td>
<td>
<p>A <code>ranked_contrib_pairs</code> object.</p>
</td></tr>
<tr><td><code id="plot.ranked_contrib_pairs_+3A_top">top</code></td>
<td>
<p>The top ranked number of pairs to print. <code>NULL</code> for
all.</p>
</td></tr>
<tr><td><code id="plot.ranked_contrib_pairs_+3A_...">...</code></td>
<td>
<p>Delegated to the generic <code><a href="graphics.html#topic+plot">plot</a></code> function.</p>
</td></tr>
<tr><td><code id="plot.ranked_contrib_pairs_+3A_xlab">xlab</code></td>
<td>
<p>Graphical parameter.</p>
</td></tr>
<tr><td><code id="plot.ranked_contrib_pairs_+3A_ylab">ylab</code></td>
<td>
<p>Graphical parameter.</p>
</td></tr>
</table>

<hr>
<h2 id='predict.disclapmixfit'>Predict from a disclapmixfit</h2><span id='topic+predict.disclapmixfit'></span>

<h3>Description</h3>

<p>Is able to predict haplotype frequencies using a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'disclapmixfit'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.disclapmixfit_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object</p>
</td></tr>
<tr><td><code id="predict.disclapmixfit_+3A_newdata">newdata</code></td>
<td>
<p>the haplotypes in matrix format to estimate haplotype
probabilities for</p>
</td></tr>
<tr><td><code id="predict.disclapmixfit_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> <code><a href="#topic+plot.disclapmixfit">plot.disclapmixfit</a></code>

<code><a href="#topic+clusterprob">clusterprob</a></code>
</p>

<hr>
<h2 id='print.contrib_pairs'>Print contributor pairs</h2><span id='topic+print.contrib_pairs'></span>

<h3>Description</h3>

<p>Print contributor pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'contrib_pairs'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.contrib_pairs_+3A_x">x</code></td>
<td>
<p>A <code>contrib_pairs</code> object.</p>
</td></tr>
<tr><td><code id="print.contrib_pairs_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>

<hr>
<h2 id='print.disclapmixfit'>Print a disclapmixfit</h2><span id='topic+print.disclapmixfit'></span>

<h3>Description</h3>

<p>Prints a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'disclapmixfit'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.disclapmixfit_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object, usually from a result of a
call to <code>disclapmix</code>.</p>
</td></tr>
<tr><td><code id="print.disclapmixfit_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> <code><a href="#topic+plot.disclapmixfit">plot.disclapmixfit</a></code>

</p>

<hr>
<h2 id='print.ranked_contrib_pairs'>Print ranked contributor pairs</h2><span id='topic+print.ranked_contrib_pairs'></span>

<h3>Description</h3>

<p>Print ranked contributor pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ranked_contrib_pairs'
print(x, top = 5L, hide_non_varying_loci = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ranked_contrib_pairs_+3A_x">x</code></td>
<td>
<p>A <code>ranked_contrib_pairs</code> object.</p>
</td></tr>
<tr><td><code id="print.ranked_contrib_pairs_+3A_top">top</code></td>
<td>
<p>The top ranked number of pairs to print/plot. <code>NULL</code> for
all.</p>
</td></tr>
<tr><td><code id="print.ranked_contrib_pairs_+3A_hide_non_varying_loci">hide_non_varying_loci</code></td>
<td>
<p>Whether to hide alleles on loci that do not
vary.</p>
</td></tr>
<tr><td><code id="print.ranked_contrib_pairs_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>

<hr>
<h2 id='rank_contributor_pairs'>Separate a 2 person mixture</h2><span id='topic+rank_contributor_pairs'></span>

<h3>Description</h3>

<p>Separate a 2 person mixture by ranking the possible contributor pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rank_contributor_pairs(contrib_pairs, fit, max_rank = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rank_contributor_pairs_+3A_contrib_pairs">contrib_pairs</code></td>
<td>
<p>A <code>contrib_pairs</code> object obtained from
<code><a href="#topic+contributor_pairs">contributor_pairs</a></code>.</p>
</td></tr>
<tr><td><code id="rank_contributor_pairs_+3A_fit">fit</code></td>
<td>
<p>A <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.</p>
</td></tr>
<tr><td><code id="rank_contributor_pairs_+3A_max_rank">max_rank</code></td>
<td>
<p>Not used. Reserved for future use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ranked_contrib_pairs</code> object that is basically an order
vector and the probabilities for each pair (in the same order as given in
<code>contrib_pairs</code>), found by using <code>fit</code>. Note, that contributor
order is disregarded so that each contributor pair is only present once (and
not twice as would be the case if taking order into consideration).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contributor_pairs">contributor_pairs</a></code> <code><a href="#topic+generate_mixture">generate_mixture</a></code>
<code><a href="#topic+disclapmix-package">disclapmix-package</a></code> <code><a href="#topic+disclapmix">disclapmix</a></code>
<code><a href="#topic+disclapmixfit">disclapmixfit</a></code> <code><a href="#topic+clusterprob">clusterprob</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code> <code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code>

<code><a href="disclap.html#topic+disclap-package">disclap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(danes)
db &lt;- as.matrix(danes[rep(1L:nrow(danes), danes$n), 1L:(ncol(danes) - 1L)])

set.seed(1)
true_contribs &lt;- sample(1L:nrow(db), 2L)
h1 &lt;- db[true_contribs[1L], ]
h2 &lt;- db[true_contribs[2L], ]
db_ref &lt;- db[-true_contribs, ]

h1h2 &lt;- c(paste(h1, collapse = ";"), paste(h2, collapse = ";"))
tab_db &lt;- table(apply(db, 1, paste, collapse = ";"))
tab_db_ref &lt;- table(apply(db_ref, 1, paste, collapse = ";"))
tab_db[h1h2]
tab_db_ref[h1h2]

rm(db) # To avoid use by accident

mixture &lt;- generate_mixture(list(h1, h2))

possible_contributors &lt;- contributor_pairs(mixture)
possible_contributors

fits &lt;- lapply(1L:5L, function(clus) disclapmix(db_ref, clusters = clus))

best_fit_BIC &lt;- fits[[which.min(sapply(fits, function(fit) fit$BIC_marginal))]]
best_fit_BIC

ranked_contributors_BIC &lt;- rank_contributor_pairs(possible_contributors, best_fit_BIC)
ranked_contributors_BIC

plot(ranked_contributors_BIC, top = 10L, type = "b")

get_rank(ranked_contributors_BIC, h1)

</code></pre>

<hr>
<h2 id='simulate.disclapmixfit'>Simulate from a disclapmixfit</h2><span id='topic+simulate.disclapmixfit'></span>

<h3>Description</h3>

<p>Simulate from a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'disclapmixfit'
simulate(object, nsim = 1L, seed = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate.disclapmixfit_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object, usually from a result of
a call to <code>disclapmix</code>.</p>
</td></tr>
<tr><td><code id="simulate.disclapmixfit_+3A_nsim">nsim</code></td>
<td>
<p>number of haplotypes to generate.</p>
</td></tr>
<tr><td><code id="simulate.disclapmixfit_+3A_seed">seed</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="simulate.disclapmixfit_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix where the rows correspond to the simulated haplotypes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+plot.disclapmixfit">plot.disclapmixfit</a></code> <code><a href="#topic+summary.disclapmixfit">summary.disclapmixfit</a></code>

</p>

<hr>
<h2 id='summary.disclapmixfit'>Summary of a disclapmixfit</h2><span id='topic+summary.disclapmixfit'></span>

<h3>Description</h3>

<p>Summary of a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'disclapmixfit'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.disclapmixfit_+3A_object">object</code></td>
<td>
<p>a <code><a href="#topic+disclapmixfit">disclapmixfit</a></code> object, usually from a result of
a call to <code>disclapmix</code>.</p>
</td></tr>
<tr><td><code id="summary.disclapmixfit_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+disclapmix">disclapmix</a></code> <code><a href="#topic+disclapmixfit">disclapmixfit</a></code>
<code><a href="#topic+predict.disclapmixfit">predict.disclapmixfit</a></code> <code><a href="#topic+print.disclapmixfit">print.disclapmixfit</a></code>
<code><a href="#topic+simulate.disclapmixfit">simulate.disclapmixfit</a></code> 
<code><a href="#topic+clusterdist">clusterdist</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
