<!DOCTYPE html><html lang="en"><head><title>Help for package DiscreteDatasets</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DiscreteDatasets}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DiscreteDatasets-package'><p>DiscreteDatasets</p></a></li>
<li><a href='#airway'><p>Airway smooth muscle cells</p></a></li>
<li><a href='#amnesia'><p>Amnesia and other drug reactions in the MHRA pharmacovigilance spontaneous</p>
reporting system</a></li>
<li><a href='#disorderdetection'><p>Disorder Detection data</p></a></li>
<li><a href='#federalist'><p>The Federalist Papers word count data</p></a></li>
<li><a href='#hiv'><p>HIV data</p></a></li>
<li><a href='#listerdata'><p>Lister data</p></a></li>
<li><a href='#reconstruct_four'><p>Reconstruct a set of reformatted four-fold tables</p></a></li>
<li><a href='#reconstruct_two'><p>Reconstruct a set of four-fold tables from rows or columns</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Example Data Sets for Use with Discrete Statistical Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-01-24</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides several data sets for use with discrete statistical tests
  and discrete multiple testing procedures. Some of them are also available as a
  four-column version, so that each row represents a 2x2 table.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>checkmate</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DISOhda/DiscreteDatasets">https://github.com/DISOhda/DiscreteDatasets</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/DISOhda/DiscreteDatasets/issues">https://github.com/DISOhda/DiscreteDatasets/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-24 20:22:07 UTC; Florian Junge</td>
</tr>
<tr>
<td>Author:</td>
<td>Christina Kihn [aut],
  Sebastian Döhler [aut],
  Florian Junge [cre, aut],
  Lukas Klein [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Florian Junge &lt;diso.fbmn@h-da.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-24 21:20:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='DiscreteDatasets-package'>DiscreteDatasets</h2><span id='topic+DiscreteDatasets'></span><span id='topic+DiscreteDatasets-package'></span>

<h3>Description</h3>

<p>This package contains example datasets for use with discrete statistical
tests and discrete multiple testing procedures. Some of them are also
available as a four-column version, so that each row represents a 2x2 table.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Florian Junge <a href="mailto:diso.fbmn@h-da.de">diso.fbmn@h-da.de</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Christina Kihn
</p>
</li>
<li><p> Sebastian Döhler
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Lukas Klein [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/DISOhda/DiscreteDatasets">https://github.com/DISOhda/DiscreteDatasets</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/DISOhda/DiscreteDatasets/issues">https://github.com/DISOhda/DiscreteDatasets/issues</a>
</p>
</li></ul>


<hr>
<h2 id='airway'>Airway smooth muscle cells</h2><span id='topic+airway'></span><span id='topic+airway_treat'></span><span id='topic+airway_four_columns'></span>

<h3>Description</h3>

<p>Read counts per gene for airway smooth muscle cell lines RNA-Seq experiment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("airway")

data("airway_treat")

data("airway_four_columns")
</code></pre>


<h3>Format</h3>

<p><code>airway</code> is a <code>data.frame</code> with 63,677 rows and 8 columns. Each row
corresponds to a specific gene and each column to a labeled sample.
</p>
<p><code>airway_treat</code> is a <code>data.frame</code> with 63,677 rows representing
genes with the following two columns:
</p>

<dl>
<dt>Treatment</dt><dd><p>Number of reads for the specific gene in all treated
samples.</p>
</dd>
<dt>NoTreatment</dt><dd><p>Number of reads for the specific gene in all untreated
samples.</p>
</dd>
</dl>

<p>Thus, each line describes a 2x2 table, e.g.:
</p>

<table>
<tr>
 <td style="text-align: right;">
  ENSG00000000003 </td><td style="text-align: center;"> This gene      </td><td style="text-align: center;"> All other genes</td>
</tr>
<tr>
 <td style="text-align: right;">
  Treatment       </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> 89,561,179 - <code class="reqn">X_{i, 1}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  No Treatment    </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> 85,955,244 - <code class="reqn">X_{i, 2}</code>
</td>
</tr>

</table>

<p><code>airway_four_columns</code> is a <code>data.frame</code> with 63,677 rows
representing genes with the following four columns:
</p>

<dl>
<dt>Treatment.ThisGene</dt><dd><p>Number of reads for the specific gene in all
treated samples.</p>
</dd>
<dt>NoTreatment.ThisGene</dt><dd><p>Number of reads for the specific gene in all
untreated samples.</p>
</dd>
<dt>Treatment.AllOtherGenes</dt><dd><p>Number of reads for all other genes in all
treated samples.</p>
</dd>
<dt>NoTreatment.AllOtherGenes</dt><dd><p>Number of reads for tall other genes in
all untreated samples.</p>
</dd>
</dl>

<p>Thus, each line describes a 2x2 table, e.g.:
</p>

<table>
<tr>
 <td style="text-align: right;">
  ENSG00000000003 </td><td style="text-align: center;"> This gene      </td><td style="text-align: center;"> All other genes</td>
</tr>
<tr>
 <td style="text-align: right;">
  Treatment       </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 3}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  No Treatment    </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 4}</code>
</td>
</tr>

</table>



<h3>Details</h3>

<p>The cell lines of the even-numbered samples were treated with dexamethasone,
whereas the cell lines of the odd-numbered samples were not. There were
89,561,179 reads for all treated samples and 85,955,244 for the untreated
ones.
</p>


<h3>Note</h3>

<p>The original <code>airway</code> dataset has been taken from the
<code><a href="airway.html#topic+airway">airway</a></code> BioConductor package. Since the original
data would require other BioConductor packages to access it, it has been
reformatted to a standard data frame (with <code>assay(airway)</code>) which only
contains the raw numeric data.
</p>


<h3>Source</h3>

<p>FASTQ files from SRA, phenotypic data from GEO
</p>


<h3>References</h3>

<p>Himes, B. E., Jiang, X., Wagner, P., Hu, R., Wang, Q., Klanderman, B.,
Whitaker, R. M., Duan, Q., Lasky-Su, J., Nikolos, C., Jester, W.,
Johnson, M., Panettieri, R. Jr., Tantisira, K. G., Weiss, S. T., Lu, Q.
(2014). RNA-Seq Transcriptome Profiling Identifies CRISPLD2 as a
Glucocorticoid Responsive Gene that Modulates Cytokine Function in Airway
Smooth Muscle Cells. <em>PLoS One</em> <strong>9</strong>(6).
<a href="https://doi.org/10.1371/journal.pone.0099625">doi:10.1371/journal.pone.0099625</a>
</p>

<hr>
<h2 id='amnesia'>Amnesia and other drug reactions in the MHRA pharmacovigilance spontaneous
reporting system</h2><span id='topic+amnesia'></span><span id='topic+amnesia_four_columns'></span>

<h3>Description</h3>

<p>For each of 2,446 drugs in the MHRA database (column 1), the number of cases
with amnesia as an adverse event (column 2), and the number of cases with
other adverse event for this drug (column 3). In total, 684,652 adverse drug
reactions were reported, among them 2,044 cases of amnesia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("amnesia")

data("amnesia_four_columns")
</code></pre>


<h3>Format</h3>

<p><code>amnesia</code> is a <code>data.frame</code> with 2,446 rows representing drugs
with the following two columns:
</p>

<dl>
<dt>AmnesiaCases</dt><dd><p>Number of the amnesia cases reported for the drug.</p>
</dd>
<dt>OtherAdverseCases</dt><dd><p>Number of other adverse drug reactions reported
for the drug.</p>
</dd>
</dl>

<p>Thus, each line describes a 2x2 table, e.g.:
</p>

<table>
<tr>
 <td style="text-align: right;">
  1-ANDROSTENEDIOL    </td><td style="text-align: center;"> This drug      </td><td style="text-align: center;"> All other drugs</td>
</tr>
<tr>
 <td style="text-align: right;">
  Amnesia cases       </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> 2,044   - <code class="reqn">X_{i, 1}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Other adverse cases </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> 682,648 - <code class="reqn">X_{i, 2}</code>
</td>
</tr>

</table>

<p><code>amnesia_four_columns</code> is a <code>data.frame</code> with 2,446 rows
representing drugs with the following four columns:
</p>

<dl>
<dt>AmnesiaCases.ThisDrug</dt><dd><p>Number of the amnesia cases reported for the
drug.</p>
</dd>
<dt>AmnesiaCases.AllOtherDrugs</dt><dd><p>Number of the amnesia cases reported for
all other drugs.</p>
</dd>
<dt>OtherAdverseCases.ThisDrug</dt><dd><p>Number of other adverse drug reactions
reported for the drug.</p>
</dd>
<dt>OtherAdverseCases.AllOtherDrugs</dt><dd><p>Number of other adverse drug
reactions reported for all other
drugs.</p>
</dd>
</dl>

<p>Thus, each line describes a 2x2 table:
</p>

<table>
<tr>
 <td style="text-align: right;">
  1-ANDROSTENEDIOL    </td><td style="text-align: center;"> This drug      </td><td style="text-align: center;"> All other drugs</td>
</tr>
<tr>
 <td style="text-align: right;">
  Amnesia cases       </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 3}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Other adverse cases </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 4}</code>
</td>
</tr>

</table>



<h3>Details</h3>

<p>The data was collected from the Drug Analysis Prints published
by the Medicines and Healthcare products Regulatory Agency (MHRA),
by Heller &amp; Gur. See references for more details.
</p>


<h3>Note</h3>

<p>The original <code>amnesia</code> dataset has been taken from the
<code>discreteMTP</code> package, which is no longer available on CRAN. It has been
reformatted such that the names in first column are now row descriptions;
this way, the actual contents of the table are purely numeric.
</p>


<h3>Source</h3>

<p><a href="https://yellowcard.mhra.gov.uk/idaps">Drug Analysis Prints on MHRA site</a>
</p>


<h3>References</h3>

<p>R. Heller and H. Gur (2011). False discovery rate controlling procedures for
discrete tests. arXiv pre-print arXiv:1112.4627v2
<a href="https://arxiv.org/abs/1112.4627v2">link</a>.
</p>

<hr>
<h2 id='disorderdetection'>Disorder Detection data</h2><span id='topic+disorderdetection'></span>

<h3>Description</h3>

<p>For earlier recognition of diseases, multiple variations of the human base
sequence get studied. The so-called coverage of each base is calculated to
detect duplicates, deletions and insertions in the base sequence. To find
these variations a hypothesis-test gets performed for each base in the tested
area. The null-hypothesis being that the coverage of the base is as expected
under the null-hypothesis (expected coverage Cb can be calculated using a
given formula, following a poisson distribution). If the observed coverage is
exceptionally high or low the null-hypothesis gets rejected. For each type of
variation there is a different formula to calculate the expected coverages.
The expected coverages in this data set were calculated using the formula for
a local test without GC-correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("disorderdetection")
</code></pre>


<h3>Format</h3>

<p>A data frame with 315 rows representing a base sequence with the
following 2 columns:
</p>

<dl>
<dt><code>observed frequencies</code></dt><dd><p>Observed coverage of each base</p>
</dd>
<dt><code>expected frequencies</code></dt><dd><p>Expected coverage of each base</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data was collected from the &quot;Goodness-of-fit tests for disorder detection
in NGS experiments&quot; published by the Biometrical Journal , by Jiménez-Otero,
de Uña-Álvarez and Pardo-Fernández. See references for more details.
</p>


<h3>References</h3>

<p>Jiménez-Otero N, de Uña-Álvarez J, Pardo-Fernández JC (2019). Goodness-of-fit
tests for disorder detection in NGS experiments.
<em>Biometrical Journal</em>, <strong>61</strong>(2), pp. 424-441.
<a href="https://doi.org/10.1002/bimj.201700284">doi:10.1002/bimj.201700284</a>.
</p>

<hr>
<h2 id='federalist'>The Federalist Papers word count data</h2><span id='topic+federalist'></span>

<h3>Description</h3>

<p>Author assignments and counts of the 1,500 most common
words from &quot;The Federalist&quot; articles.
</p>
<p>&quot;The Federalist Papers&quot; are a set of 85 articles written under the pseudonym
&quot;Publius&quot; to promote the ratification of the US constitution by Alexander
Hamilton, James Madison and John Jay in 1787 and 1788. There are multiple
sources which attribute the articles to their real
authors. We use the attributions by the Project Gutenberg
and the correction by the authors of the <code>syllogi</code> package.
This task has been a popular problem in natural language
processing. One of the most prominent examples is the work by Mosteller
and Wallace (1964) who used the word frequencies to attribute the
disputed articles to their authors.
</p>
<p>The data provided in this package was prepared with the following steps
by employing the <code>tm</code> package:
</p>

<ol>
<li><p>Load the texts from the <code>syllogi</code> package,
</p>
</li>
<li><p>Lowercase,
</p>
</li>
<li><p>Remove punctuation,
</p>
</li>
<li><p>Strip whitespace,
</p>
</li>
<li><p>Remove the texts by Jay, one text coauthored by
Madison and Hamilton together, and the redundant
version of article 70,
</p>
</li>
<li><p>Find the 1,500 most common words for each author,
</p>
</li>
<li><p>Count the occurrences of these words in the texts.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>data("federalist")
</code></pre>


<h3>Format</h3>

<p><code>federalist</code> is a <code>data.frame</code> with 77 rows and 1,984
columns:
</p>

<dl>
<dt>doc_no</dt><dd><p>Article number</p>
</dd>
<dt>doc_author</dt><dd><p>Author of the article (according to Project
Gutenberg)</p>
</dd>
<dt>...</dt><dd><p>The remaining 1,982 columns are the word counts</p>
</dd>
</dl>



<h3>References</h3>

<p>Watson, G. S. (1966). Review: Frederick Mosteller, David L. Wallace, 
Inference and Disputed Authorship: The Federalist.
<em>The Annals of Mathematical Statistics</em>, <strong>37</strong>(1), 308-312.
<a href="https://doi.org/10.1214/aoms/1177699628">doi:10.1214/aoms/1177699628</a>
</p>
<p>Donoho, D. L., &amp; Kipnis, A. (2022). Higher criticism to compare two large
frequency tables, with sensitivity to possible rare and weak differences.
<em>Annals of Statistics</em>, <strong>50</strong>(3), 1447-1472.
<a href="https://doi.org/10.1214/21-AOS2158">doi:10.1214/21-AOS2158</a>
</p>
<p>Feinerer, I., &amp; Hornik, K. (2024). tm: Text Mining Package. R package
version 0.7-15. <em>CRAN</em>. <a href="https://CRAN.R-project.org/package=tm">https://CRAN.R-project.org/package=tm</a>
</p>
<p>Studyvin, J. (2024). syllogi: Collection of Data Sets for Teaching Purposes.
R package version 1.0.3. <em>CRAN</em>.
<a href="https://CRAN.R-project.org/package=syllogi">https://CRAN.R-project.org/package=syllogi</a>
</p>

<hr>
<h2 id='hiv'>HIV data</h2><span id='topic+hiv'></span><span id='topic+hiv_four_columns'></span>

<h3>Description</h3>

<p>This data set has been analyzed and provided by the listed reference.
Examined were two groups with different types of HIV (Type B and  Type C),
each consisting of 73 participants. Within both groups the number of
amino-acid mutations at each position was determined.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("hiv")

data("hiv_four_columns")
</code></pre>


<h3>Format</h3>

<p><code>hiv</code> is a <code>data.frame</code> with 118 rows and the following two
columns:
</p>

<dl>
<dt>TypeC</dt><dd><p>Number of test subjects with HIV type C and mutated i-th amino
acid.</p>
</dd>
<dt>TypeB</dt><dd><p>Number of test subjects with HIV type B and mutated i-th amino
acid.</p>
</dd>
</dl>

<p>Thus, each row describes a 2x2 table:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Subject 1 </td><td style="text-align: center;"> Mutation       </td><td style="text-align: center;"> No mutation</td>
</tr>
<tr>
 <td style="text-align: right;">
  Type C    </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> 73 - <code class="reqn">X_{i, 1}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Type B    </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> 73 - <code class="reqn">X_{i, 2}</code>
</td>
</tr>

</table>

<p><code>hiv_four_columns</code> is a <code>data.frame</code> with 118 rows and the
following four columns:
</p>

<dl>
<dt>TypeC.Mutation</dt><dd><p>Number of test subjects with HIV type C and mutated
i-th amino acid.</p>
</dd>
<dt>TypeB.Mutation</dt><dd><p>Number of test subjects with HIV type B and mutated
i-th amino acid.</p>
</dd>
<dt>TypeC.NoMutation</dt><dd><p>Number of test subjects with HIV type C and
non-mutated i-th amino acid.</p>
</dd>
<dt>TypeB.NoMutation</dt><dd><p>Number of test subjects with HIV type B and
non-mutated i-th amino acid.</p>
</dd>
</dl>

<p>Thus, each row describes a 2x2 table:
</p>

<table>
<tr>
 <td style="text-align: right;">
  Subject 1 </td><td style="text-align: center;"> mutation       </td><td style="text-align: center;"> no mutation</td>
</tr>
<tr>
 <td style="text-align: right;">
  Type C    </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 3}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Type B    </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 4}</code>
</td>
</tr>

</table>



<h3>Note</h3>

<p>The original <code>hiv</code> dataset has been taken from the
<code><a href="fdrDiscreteNull.html#topic+hivdata">fdrDiscreteNull</a></code> package, where it is
named <code>hivdata</code>.
</p>


<h3>References</h3>

<p>Gilbert, P. B. (2005). A modified false discovery rate multiple-comparisons
procedure for discrete data, applied to human immunodeficiency virus
genetics. <em>Journal of the Royal Statistical Society</em>, <strong>54</strong>(1),
pp. 143-158. <a href="https://doi.org/10.1111/j.1467-9876.2005.00475.x">doi:10.1111/j.1467-9876.2005.00475.x</a>
</p>

<hr>
<h2 id='listerdata'>Lister data</h2><span id='topic+listerdata'></span><span id='topic+listerdata_four_columns'></span>

<h3>Description</h3>

<p>This dataset has been analyzed and provided by the listed reference. There
are around 22,000 cytosines, each of which is under two conditions. For each
cytosine under each condition, there is only one replicate. The discrete
count for each replicate can be modeled by binomial distribution, and
Fisher's exact test can be applied to assess if a cytosine is differentially
methylated. The filtered data <code>lister</code> contains cytosines whose
total counts for both lines are greater than 5 and whose count for each line
does not exceed 25.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("listerdata")

data("listerdata_four_columns")
</code></pre>


<h3>Format</h3>

<p><code>listerdata</code> is a <code>data.frame</code> with 3,525 rows and the following two
columns:
</p>

<dl>
<dt>Col0_Counts</dt><dd><p>Degree of methylation of the i-th cytosine in reference
genome.</p>
</dd>
<dt>Met13_Counts</dt><dd><p>Degree of methylation of the i-th cytosine in mutated
genome.</p>
</dd>
</dl>

<p>Thus, each row describes a 2x2 table:
</p>

<table>
<tr>
 <td style="text-align: right;">
  AT1G01070.1  </td><td style="text-align: center;"> This cytosine  </td><td style="text-align: center;"> All other cytosines</td>
</tr>
<tr>
 <td style="text-align: right;">
  Col0 counts  </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> 34,244 - <code class="reqn">X_{i, 1}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Met13 counts </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> 39,342 - <code class="reqn">X_{i, 2}</code>
</td>
</tr>

</table>

<p><code>listerdata_four_columns</code> is a <code>data.frame</code> with 3,525 rows and the
following four columns:
</p>

<dl>
<dt>Col0_Counts.ThisCyto</dt><dd><p>Degree of methylation of the i-th cytosine in
reference genome.</p>
</dd>
<dt>Met13_Counts.ThisCyto</dt><dd><p>Degree of methylation of the i-th cytosine in
mutated genome.</p>
</dd>
<dt>Col0_Counts.AllOtherCytos</dt><dd><p>Degree of methylation of all other
cytosines in reference genome.</p>
</dd>
<dt>Met13_Counts.AllOtherCytos</dt><dd><p>Degree of methylation of all other
cytosines in mutated genome.</p>
</dd>
</dl>


<table>
<tr>
 <td style="text-align: right;">
  AT1G01070.1  </td><td style="text-align: center;"> This cytosine  </td><td style="text-align: center;"> All other cytosines</td>
</tr>
<tr>
 <td style="text-align: right;">
  Col0 counts  </td><td style="text-align: center;"> <code class="reqn">X_{i, 1}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 3}</code></td>
</tr>
<tr>
 <td style="text-align: right;">
  Met13 counts </td><td style="text-align: center;"> <code class="reqn">X_{i, 2}</code> </td><td style="text-align: center;"> <code class="reqn">X_{i, 4}</code>
</td>
</tr>

</table>



<h3>Note</h3>

<p>The original <code>listerdata</code> dataset has been taken from the
<code><a href="fdrDiscreteNull.html#topic+listerdata">fdrDiscreteNull</a></code> package.
</p>


<h3>References</h3>

<p>Lister, R., O'Malley, R., Tonti-Filippini, J., Gregory, B. D., Berry, C. C.,
Millar, A. H. &amp; Ecker, J. R. (2008). Highly integrated single-base
resolution maps of the epigenome in arabidopsis, <em>Cell</em>
<strong>133</strong>(3), pp. 523-536.
<a href="https://doi.org/10.1016/j.cell.2008.03.029">doi:10.1016/j.cell.2008.03.029</a>
</p>

<hr>
<h2 id='reconstruct_four'>Reconstruct a set of reformatted four-fold tables</h2><span id='topic+reconstruct_four'></span>

<h3>Description</h3>

<p>Sometimes, fourfold tables are reformatted by replacing rows or columns by
marginal totals. This makes it impossible to use them straight away for
statistical tests like Fisher's exact test. But with that knowledge, the
missing values can easily be restored. The <code>reconstruct_four</code>
function uses a set of such reduced tables, stored row-wise in a matrix or a
data frame, and rebuilds the two reformatted cells when they were replaced by
marginal totals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconstruct_four(dat, idx_marginals = NULL, colnames_add = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reconstruct_four_+3A_dat">dat</code></td>
<td>
<p>integer matrix or data frame with exactly two
columns; each row represents the first column of a
2x2 matrix for which the other two values are to
be computed and appended to <code>dat</code> as two new
columns; real numbers will be coerced to integer.</p>
</td></tr>
<tr><td><code id="reconstruct_four_+3A_idx_marginals">idx_marginals</code></td>
<td>
<p>integer vector of exactly two values or
<code>NULL</code> (the default) indicating the columns
of <code>dat</code> that contain the marginal totals;
if <code>NULL</code>, the last two columns are used.</p>
</td></tr>
<tr><td><code id="reconstruct_four_+3A_colnames_add">colnames_add</code></td>
<td>
<p>character vector of exactly two unique character
strings or <code>NULL</code> (the default), which
contains the desired headers of the new
(reconstructed) columns of the input; if
<code>NULL</code>, the headers of the marginal totals
are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer data frame with four columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X1 &lt;- c(4, 2, 2, 14, 6, 9, 4, 0, 1)
X2 &lt;- c(0, 0, 1, 3, 2, 1, 2, 2, 2)
N1 &lt;- rep(148, 9)
N2 &lt;- rep(132, 9)

df1 &lt;- data.frame(X1, X2, N1, N2)
reconstruct_four(df1, colnames_add = c("Y1", "Y2"))
# same as reconstruct_four(df1, c(3, 4), c("Y1", "Y2"))

df2 &lt;- data.frame(X1, N1, X2, N2)
reconstruct_four(df2, c(2, 4), c("Y1", "Y2"))

</code></pre>

<hr>
<h2 id='reconstruct_two'>Reconstruct a set of four-fold tables from rows or columns</h2><span id='topic+reconstruct_two'></span>

<h3>Description</h3>

<p>In some situations, fourfold tables are reduced to two elements, which makes
it impossible to use them straight away for statistical tests like Fisher's
exact test. But sometimes, when all tables had the same known marginal sums,
the missing values can be restored using that additional information. The
<code>reconstruct_two</code> function uses a set of such reduced tables, stored
row-wise in a matrix or a data frame, and rebuilds the two missing columns
from automatically computed or given marginal totals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconstruct_two(
  dat,
  totals = NULL,
  insert_at = NULL,
  colnames_add = NULL,
  colnames_prepend = NULL,
  colnames_append = NULL,
  colnames_sep = "_"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reconstruct_two_+3A_dat">dat</code></td>
<td>
<p>integer matrix or data frame with exactly two
columns; each row represents the first column of a
2x2 matrix for which the other two values are to
be computed and appended to <code>dat</code> as two new
columns; real numbers will be coerced to integer.</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_totals">totals</code></td>
<td>
<p>integer vector of exactly one or two values or
<code>NULL</code> (the default); the new columns will be
derived by subtracting the existing column values
from <code>totals</code>; if <code>NULL</code>, the sums of
the two existing columns of <code>dat</code> are used.</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_insert_at">insert_at</code></td>
<td>
<p>integer vector of exactly two values between 1 and
4 or <code>NULL</code> (the default) indicating the
indices at which the values are to be inserted; if
<code>NULL</code>, the new values are appended at the
end, i.e. at positions 3 and 4.</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_colnames_add">colnames_add</code></td>
<td>
<p>character vector of exactly two unique character
strings or <code>NULL</code> (the default), which
contains the desired headers of the new
(reconstructed) columns of the input; if
<code>NULL</code>, the headers of <code>dat</code> are used
(with appended strings; see below).</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_colnames_prepend">colnames_prepend</code></td>
<td>
<p>character vector of exactly two unique character
strings (<code>NA</code>s are allowed) or <code>NULL</code>
(the default); the first string will be prepended
to the headers of the original headers of
<code>dat</code>, while the second is used in
the same manner for the reconstructed columns.</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_colnames_append">colnames_append</code></td>
<td>
<p>character vector of exactly two unique character
strings (<code>NA</code>s are allowed) or <code>NULL</code>
(the default); the first string will be appended
to the headers of the original headers of
<code>dat</code>, while the second is used in
the same manner for the reconstructed columns; if
<code>colnames_add = NULL</code> and
<code>colnames_append = NULL</code>, <code>c("A", "B")</code>
will be used.</p>
</td></tr>
<tr><td><code id="reconstruct_two_+3A_colnames_sep">colnames_sep</code></td>
<td>
<p>a single character or <code>NULL</code> (the default)
giving the separator for combining
<code>colnames_prepend</code> and <code>colnames_append</code>
with the column names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer data frame with four columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(amnesia)
amnesia_four_columns &lt;- reconstruct_two(
  amnesia,
  NULL,
  NULL,
  NULL,
  NULL,
  c("ThisDrug", "AllOtherDrugs"),
  "."
)
head(amnesia_four_columns)

data(hiv)
hiv_four_columns &lt;- reconstruct_two(
  hiv,
  73,
  NULL,
  NULL,
  NULL,
  c("Mutation", "NoMutation"),
  "."
)
head(hiv_four_columns)

data(listerdata)
listerdata_four_columns &lt;- reconstruct_two(
  listerdata,
  c(34244, 39342),
  NULL,
  NULL,
  NULL,
  c("This_Cyto", "All_Other_Cytos"),
  "_"
)
head(listerdata_four_columns)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
