<!DOCTYPE html><html><head><title>Help for package DisHet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DisHet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DisHet'><p>Heterogeneity Dissection</p></a></li>
<li><a href='#eTME_signatures'><p>Empirically-defined Immune Signature Genes in RCC Bulk Tumor</p></a></li>
<li><a href='#StromaExp'><p>Stroma (Immune) Component Gene Expression Estimation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate the Gene Expression Levels and Component Proportions of
the Normal, Stroma (Immune) and Tumor Components of Bulk Tumor
Samples</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-05-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Tao Wang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tao Wang &lt;tao.wang@utsouthwestern.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0), gtools, matrixStats</td>
</tr>
<tr>
<td>Description:</td>
<td>Model cell type heterogeneity of bulk renal cell carcinoma. The observed gene expression in bulk tumor sample is modeled by a log-normal distribution with the location parameter structured as a linear combination of the component-specific gene expressions. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-25 20:54:44 UTC; s168319</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-25 21:40:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='DisHet'>Heterogeneity Dissection</h2><span id='topic+DisHet'></span>

<h3>Description</h3>

<p>This function performs dissection of bulk sample gene expression using matched normal and tumorgraft RNA-seq data. It outputs the final proportion estiamtes of the three components for all patients. 
</p>
<p>The patient-specific dissection proportion estimates are saved in a 3-by-k matrix named &quot;rho&quot;, where k is the number of patients. The 3 rows of &quot;rho&quot; matrix correspond to the tumor, normal, stroma components in order. That is, the proportion estimate of tumor component for patient i is stored in rho[1,i]; the normal component proportion estimate of this patient is stored in rho[2,i];and stroma component proportion in rho[3,i]. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DisHet(exp_T,exp_N,exp_G, save=TRUE, MCMC_folder, 
      n_cycle=10000, save_last=500, mean_last=200, dirichlet_c=1, S_c=1, rho_small=1e-2, 
      initial_rho_S=0.02,initial_rho_G=0.96,initial_rho_N=0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DisHet_+3A_exp_t">exp_T</code></td>
<td>

<p>Gene expression in bulk RNA-seq samples. The rows correspond to different genes. The columns correspond to different patients.
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_exp_n">exp_N</code></td>
<td>

<p>Gene expression in the corresponding normal samples. The rows list the same set of genes as in exp_G. The columns correspond to patients matched with exp_T.
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_exp_g">exp_G</code></td>
<td>

<p>Gene expression in the corresponding tumor samples. The rows list the same set of genes as in exp_G. The columns correspond to patients matched with exp_T.
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_save">save</code></td>
<td>

<p>When save==TRUE, as in default, all component proportion estimates during MCMC iterations can be saved into a user-specified directory using the &quot;MCMC_folder&quot; argument. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_mcmc_folder">MCMC_folder</code></td>
<td>

<p>Directory for saving the estimated mixture proportion matrix updates during MCMC iterations. The default setting is to create a &quot;DisHet&quot; folder under the current working directory. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_n_cycle">n_cycle</code></td>
<td>

<p>Number of MCMC iterations(chain length). The default value is 10,000. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_save_last">save_last</code></td>
<td>

<p>Save the rho matrix updates for the last &quot;save_last&quot; Number of MCMC iterations. The default value is 500. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_mean_last">mean_last</code></td>
<td>

<p>Calculate the final proportion estiamte matrix using the last &quot;mean_last&quot; number of MCMC iterations. The default value is 200.  
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_dirichlet_c">dirichlet_c</code></td>
<td>

<p>Stride scale in sampling rho. Larger value leads to smaller steps in sampling rho. The default value is 1. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_s_c">S_c</code></td>
<td>

<p>Stride scale in sampling Sij. Larger value leads to larger steps in sampling Sij. The default value is 1. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_rho_small">rho_small</code></td>
<td>

<p>The smallest rho updates allowed during MCMC. The default is 1e-2. This threshold is set to help improve numerical stability of the algorithm. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_initial_rho_s">initial_rho_S</code></td>
<td>

<p>Initial value of the proportion estimate for the stroma component. The default value is 0.02. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_initial_rho_g">initial_rho_G</code></td>
<td>

<p>Initial value of the proportion estimate for the tumor component. The default value is 0.96. 
</p>
</td></tr>
<tr><td><code id="DisHet_+3A_initial_rho_n">initial_rho_N</code></td>
<td>

<p>Initial value of the proportion estimate for the normal component. The default value is 0.02. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Un-logged expression values should be used in exp_N/T/G matrices, and their rows and columns must match each other corresponding to the same set of genes and patients. 
</p>
<p>The values specified for &quot;initial_rho_S&quot;, &quot;initial_rho_G&quot;, and &quot;initial_rho_S&quot; all have to be positive. If the three proportion initials are not summing to 1, normalization is performed automatically to force the sum to be 1. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  load(system.file("example/example_data.RData",package="DisHet"))
  exp_T &lt;- exp_T[1:200,]
  exp_N &lt;- exp_N[1:200,]
  exp_G &lt;- exp_G[1:200,]
  
  rho &lt;- DisHet(exp_T, exp_N, exp_G, save=FALSE, n_cycle=200, mean_last=50)
</code></pre>

<hr>
<h2 id='eTME_signatures'>Empirically-defined Immune Signature Genes in RCC Bulk Tumor
</h2><span id='topic+eTME_signatures'></span>

<h3>Description</h3>

<p>Based on DisHet analysis of 35 RCC trio RNA-Seq data, we defined immune-specific genes with empirical evidence, named eTME, for empirically-defined immune signature of tumor. Using eTME, we refined previously published Immunome signatures. We also assigned other eTME genes to specific immune cell types using the BLUEPRINT project data. These two sets of refined gene signatures were consolidated and documented in the DisHet R package as the &quot;eTME&quot; immune cell gene signatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("eTME_signatures")</code></pre>


<h3>Format</h3>

<p>A List contains 2 lists of signatures:the 1st list &quot;signatures_gene&quot; contains the signatures on gene level, and the 2nd list &quot;signatures_mRNA&quot; contains the signatures on mRNA level. Both lists contain 25 items/vectors.
</p>

<dl>
<dt>'M2 macrophages'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in M2 macrophages</p>
</dd>
<dt>'M1 macrophages'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in M1 macrophages</p>
</dd>
<dt>Macrophages</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Macrophages</p>
</dd>
<dt>Monocytes</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Monocytes</p>
</dd>
<dt>'B cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in B cells</p>
</dd>
<dt>'CD8 T cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in CD8 T cells</p>
</dd>
<dt>'T cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in T cells</p>
</dd>
<dt>'Dendritic cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Dendritic cells</p>
</dd>
<dt>'CD56dim NK cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in CD56dim NK cells</p>
</dd>
<dt>'CD56bright NK cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in CD56bright NK cells</p>
</dd>
<dt>'NK cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in NK cells</p>
</dd>
<dt>'Endothelial cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Endothelial cells</p>
</dd>
<dt>Eosinophils</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Eosinophils</p>
</dd>
<dt>Neutrophils</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Neutrophils</p>
</dd>
<dt>'Treg cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Treg cells</p>
</dd>
<dt>'Th1 cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Th1 cells</p>
</dd>
<dt>'Th2 cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Th2 cells</p>
</dd>
<dt>'Tfh cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Tfh cells</p>
</dd>
<dt>'Th cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Th cells</p>
</dd>
<dt>aDCs</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in aDCs</p>
</dd>
<dt>iDCs</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in iDCs</p>
</dd>
<dt>pDCs</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in pDCs</p>
</dd>
<dt>'Mast cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Mast cells</p>
</dd>
<dt>'Tm cells'</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Tm cells</p>
</dd>
<dt>Pericytes</dt><dd><p>:   a vector of genes/mRNA that are abundantly expressed in Pericytes</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(eTME_signatures)
eTME_signatures$signatures_gene$Macrophages
eTME_signatures$signatures_mRNA$Macrophages
eTME_signatures$signatures_gene$`T cells`
eTME_signatures$signatures_mRNA$`T cells`
</code></pre>

<hr>
<h2 id='StromaExp'>Stroma (Immune) Component Gene Expression Estimation</h2><span id='topic+StromaExp'></span>

<h3>Description</h3>

<p>This function estimate the stroma component gene expression profiles of all patients, using the proportion estimates obtained from function <code>DisHet</code>. The estimates are stored in a p-by-k matrix, where p is the number of genes and k is the number of patients.The order of genes and the order of patients are the same as in the input bulk sample expression matrix.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StromaExp(exp_T,exp_N,exp_G, rho)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StromaExp_+3A_exp_t">exp_T</code></td>
<td>

<p>Gene expression in bulk RNA-seq samples. The rows correspond to different genes. The columns correspond to different patients.
</p>
</td></tr>
<tr><td><code id="StromaExp_+3A_exp_n">exp_N</code></td>
<td>

<p>Gene expression in the corresponding normal samples. The rows list the same set of genes as in exp_G. The columns correspond to patients matched with exp_T.
</p>
</td></tr>
<tr><td><code id="StromaExp_+3A_exp_g">exp_G</code></td>
<td>

<p>Gene expression in the corresponding tumor samples. The rows list the same set of genes as in exp_G. The columns correspond to patients matched with exp_T.
</p>
</td></tr>
<tr><td><code id="StromaExp_+3A_rho">rho</code></td>
<td>

<p>Output from function <code>DisHet</code>: the patient-specific proportion estimates corresponding to tumor, normal, stroma components in order.  
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>  load(system.file("example/example_data.RData",package="DisHet"))
  exp_T &lt;- exp_T[1:200,]
  exp_N &lt;- exp_N[1:200,]
  exp_G &lt;- exp_G[1:200,]
  
  rho &lt;- DisHet(exp_T, exp_N, exp_G, save=FALSE, n_cycle=200, mean_last=50)
  S &lt;- StromaExp(exp_T,exp_N,exp_G, rho)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
