<!DOCTYPE html><html lang="en"><head><title>Help for package distanceHD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {distanceHD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#distanceHD-package'>
<p>Distance Metrics for High-Dimensional Clustering</p></a></li>
<li><a href='#dist_cen'>
<p>Centroid distance between two groups</p></a></li>
<li><a href='#dist_mah'>
<p>ridge Mahalanobis distance between two groups</p></a></li>
<li><a href='#dist_mdp'>
<p>Maximal data piling (MDP) distance between two groups</p></a></li>
<li><a href='#leukemia'>
<p>Gene expression data from Golub et al. (1999)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Distance Metrics for High-Dimensional Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-31 16:50:11 UTC</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jung Ae Lee &lt;jungaeleeb@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We provide three distance metrics for measuring the separation between two clusters in high-dimensional spaces. The first metric is the centroid distance, which calculates the Euclidean distance between the centers of the two groups. The second is a ridge Mahalanobis distance, which incorporates a ridge correction constant, alpha, to ensure that the covariance matrix is invertible. The third metric is the maximal data piling distance, which computes the orthogonal distance between the affine spaces spanned by each class. These three distances are asymptotically interconnected and are applicable in tasks such as discrimination, clustering, and outlier detection in high-dimensional settings.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), MASS</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-31 16:28:44 UTC; jungae</td>
</tr>
<tr>
<td>Author:</td>
<td>Jung Ae Lee [aut, cre],
  Jeongyoun Ahn [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
</table>
<hr>
<h2 id='distanceHD-package'>
Distance Metrics for High-Dimensional Clustering
</h2><span id='topic+distanceHD-package'></span><span id='topic+distanceHD'></span>

<h3>Description</h3>

<p>We provide three distance metrics for measuring the separation between two clusters in high-dimensional spaces. The first metric is the centroid distance, which calculates the Euclidean distance between the centers of the two groups. The second is a ridge Mahalanobis distance, which incorporates a ridge correction constant, alpha, to ensure that the covariance matrix is invertible. The third metric is the maximal data piling distance, which computes the orthogonal distance between the affine spaces spanned by each class. These three distances are asymptotically interconnected and are applicable in tasks such as discrimination, clustering, and outlier detection in high-dimensional settings.
</p>


<h3>Author(s)</h3>

<p>Jung Ae Lee &lt;jungae.lee@umassmed.edu&gt;; Jeongyoun Ahn &lt;jyahn@kaist.ac.kr&gt;
</p>
<p>Maintainer: Jung Ae Lee &lt;jungaeleeb@gmail.com&gt;
</p>


<h3>References</h3>

<p>1. Ahn J, Marron JS (2010). The maximal data piling direction for discrimination. Biometrika, 97(1):254-259.
</p>
<p>2. Ahn J, Lee MH, Yoon YJ (2012). Clustering high dimension, low sample size data using the maximal data piling distance. Statistica Sinica, 22(2):443-464.
</p>
<p>3. Ahn J, Lee MH, Lee JA (2019). Distance-based outlier detection for high dimension, low sample size data. Journal of Applied Statistics.46(1):13-29.
</p>

<hr>
<h2 id='dist_cen'>
Centroid distance between two groups
</h2><span id='topic+dist_cen'></span><span id='topic+dist_centroid'></span>

<h3>Description</h3>

<p>Calculate the Centroid (Euclidean) distance between two groups in a high-dimensional space (d &gt; n), with support for a single-member cluster. It also works in low-dimensional settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_cen(x, group)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dist_cen_+3A_x">x</code></td>
<td>

<p><code>x</code> is n by d matrix
</p>
</td></tr>
<tr><td><code id="dist_cen_+3A_group">group</code></td>
<td>

<p><code>group</code> is a binary group label with the length of n1 and n2
</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A numeric value of distance
</p>


<h3>Author(s)</h3>

<p>Jung Ae Lee &lt;jungae.lee@umassmed.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
group = leukemia$Y      # 38 patients status with a value of 1 or 2
x = leukemia$X          # 38 by 3051 genes

# apply the function
dist_cen(x, group)  # 25.4
</code></pre>

<hr>
<h2 id='dist_mah'>
ridge Mahalanobis distance between two groups
</h2><span id='topic+dist_mah'></span><span id='topic+dist_ma'></span>

<h3>Description</h3>

<p>Calculate the Mahalanobis distance between two groups in a high-dimensional space (d &gt; n), using a ridge correction on the covariance matrix to ensure invertibility. The method also supports a single-member cluster and works in low-dimensional settings without a ridge correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_mah(x, group, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dist_mah_+3A_x">x</code></td>
<td>

<p><code>x</code> is n by d matrix
</p>
</td></tr>
<tr><td><code id="dist_mah_+3A_group">group</code></td>
<td>

<p><code>group</code> is a binary group label with the length of n1 and n2
</p>
</td></tr>
<tr><td><code id="dist_mah_+3A_alpha">alpha</code></td>
<td>

<p><code>alpha</code> is a positive numeric value representing the ridge correction constant. If not specified, the default value is set to <code class="reqn">\sqrt{\log(d)/n}</code>.
</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A numeric value of distance
</p>


<h3>Author(s)</h3>

<p>Jung Ae Lee &lt;jungae.lee@umassmed.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
group = leukemia$Y      # 38 patients status with a value of 1 or 2
x = leukemia$X          # 38 x 3051 genes

# apply the function
dist_mah(x, group)   # 26.8 

# default alpha
d = 3051; n = 38
alpha = sqrt(log(d)/n)
dist_mah(x, group, alpha)   # 26.8
</code></pre>

<hr>
<h2 id='dist_mdp'>
Maximal data piling (MDP) distance between two groups
</h2><span id='topic+dist_mdp'></span><span id='topic+dist_md'></span>

<h3>Description</h3>

<p>Calculate the MDP (maximal data piling) distance between two groups in a high-dimensional space (d &gt; n), with support for a single-member cluster. It also works in low-dimensional settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_mdp(x, group)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dist_mdp_+3A_x">x</code></td>
<td>

<p><code>x</code> is n by d matrix
</p>
</td></tr>
<tr><td><code id="dist_mdp_+3A_group">group</code></td>
<td>

<p><code>group</code> is a binary group label with the length of n1 and n2
</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A numeric value of distance
</p>


<h3>Author(s)</h3>

<p>Jeongyoun Ahn &lt;jyahn@kaist.ac.kr&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)
group = leukemia$Y      # 38 patients status with a value of 1 or 2
x = leukemia$X          # 38 x 3051 genes

# apply the function
dist_mah(x, group)   # 26.8 
</code></pre>

<hr>
<h2 id='leukemia'>
Gene expression data from Golub et al. (1999)
</h2><span id='topic+leukemia'></span>

<h3>Description</h3>

<p>Gene expression data (3051 genes and 38 tumor mRNA samples) from the leukemia microarray study of Golub et al. (1999). )
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(leukemia)</code></pre>


<h3>Format</h3>

<p>A list with the following elements:
</p>

<ul>
<li> <p><code>X</code>: a (38 by 3051) matrix giving the expression levels of 3051 genes for 38 leukemia patients. Each row corresponds to a patient, each column to a gene.
</p>
</li>
<li> <p><code>Y</code>: a numeric vector of length 38 giving the cancer class of each patient.
</p>
</li>
<li> <p><code>gene.names</code>: a matrix containing the names of the 3051 genes for the gene expression matrix X. The three columns correspond to the gene index, ID, and Name, respectively.
</p>
</li></ul>



<h3>Source</h3>

<p>The data are described in Golub et al. (1999) and can be freely downloaded from http://www.broadinstitute.org/cgi-bin/cancer/publications/pub_paper.cgi?paper_id=43. 
</p>


<h3>References</h3>

<p>S. Dudoit, J. Fridlyand and T. P. Speed (2002). Comparison of discrimination methods for the classification of tumors using gene expression data, Journal of the American Statistical Association 97, 77-87.
</p>
<p>Golub et al. (1999). Molecular classification of cancer: class discovery and class prediction by gene expression monitoring, Science 286, 531-537.
</p>


<h3>See Also</h3>

<p>plsgenomics::leukemia</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(leukemia)

# how many samples and genes?
dim(leukemia$X)

# how many samples of class 1 and 2, respectively?
table(leukemia$Y)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
