<!DOCTYPE html><html><head><title>Help for package DNAmotif</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DNAmotif}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DNAmotifs'><p>Generation of motifs from DNA sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>DNA Sequence Motifs</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Motifs within biological sequences show a significant role. This package utilizes a user-defined threshold value (window size and similarity) to create consensus segments or motifs through local alignment of dynamic  programming with gap and it calculates the frequency of each identified motif, offering a detailed view of their prevalence within the dataset. It allows for thorough exploration and understanding of sequence patterns and their biological importance.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, Biostrings, Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-03 12:16:18 UTC; searc</td>
</tr>
<tr>
<td>Author:</td>
<td>Subham Ghosh [aut, cre],
  UB Angadi [aut],
  Md Yeasin [aut],
  Dipro Sinha [aut],
  Saikath Das [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Subham Ghosh &lt;search4aghosh@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-04 09:44:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='DNAmotifs'>Generation of motifs from DNA sequences</h2><span id='topic+DNAmotifs'></span>

<h3>Description</h3>

<p>Using a fasta file as input, the motifs and its corresponding frequencies are generated, considering threshold values (window size and similarity), by making consensus segments via local alignment with gap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAmotifs(fasta_file, ws, cut_off)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DNAmotifs_+3A_fasta_file">fasta_file</code></td>
<td>
<p>Sequence file path (.fasta format)</p>
</td></tr>
<tr><td><code id="DNAmotifs_+3A_ws">ws</code></td>
<td>
<p>Window size</p>
</td></tr>
<tr><td><code id="DNAmotifs_+3A_cut_off">cut_off</code></td>
<td>
<p>Minimum similarity percentage between the motifs for generating a consensus motif</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>final_results</code> A dataframe of motifs and their corresponding frequncies
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_fasta = system.file("exdata/sample.fasta", package = "DNAmotif")
DNAmotifs(fasta_file = example_fasta, ws = 15, cut_off = 0.75)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
