<!DOCTYPE html><html lang="en"><head><title>Help for package dnapath</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dnapath}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+5B.dnapath'><p>Extract results of a single pathway from a 'dnapath' object.</p></a></li>
<li><a href='#+5B.dnapath_list'><p>Extract parts of a 'dnapath_list' object.</p></a></li>
<li><a href='#+5B+5B.dnapath'><p>Extract results of a single pathway from a 'dnapath' object.</p></a></li>
<li><a href='#+5B+5B.dnapath_list'><p>Extract results of a single pathway from a 'dnapath_list' object.</p></a></li>
<li><a href='#+5B+26lt+3B-.dnapath'><p>Replace parts of a 'dnapath' object.</p></a></li>
<li><a href='#+5B+26lt+3B-.dnapath_list'><p>Replace parts of a 'dnapath_list' object.</p></a></li>
<li><a href='#biomart_hsapiens'><p>Default mapping for entrezgene IDs and HGNC gene symbols</p></a></li>
<li><a href='#c.dnapath'><p>Combine two 'dnapath' objects.</p></a></li>
<li><a href='#c.dnapath_list'><p>Combine two 'dnapath_list' objects.</p></a></li>
<li><a href='#combine_overlapping_pathways'><p>Modify a pathway list to combine overlapping pathways.</p></a></li>
<li><a href='#d_edgesC'><p>C++ implementation of d_edges</p></a></li>
<li><a href='#d_genesC'><p>C++ implementation of d_genes</p></a></li>
<li><a href='#d_pathwayC'><p>C++ implementation of d_pathway</p></a></li>
<li><a href='#dna_pathway'><p>Differential network analysis on a single pathway</p></a></li>
<li><a href='#dnapath'><p>Differential Network Analysis Using Gene Pathways</p></a></li>
<li><a href='#entrez_to_symbol'><p>Obtain gene symbols for entrezgene IDs</p></a></li>
<li><a href='#filter_pathways'><p>Remove pathways with non-significant DC scores.</p></a></li>
<li><a href='#format_species_name'><p>Format sepcies name input.</p></a></li>
<li><a href='#get_biomart_mapping'><p>Internal function for obtaining biomaRt mapping</p></a></li>
<li><a href='#get_genes'><p>Get the gene names from a differential network analysis</p></a></li>
<li><a href='#get_mean_expr_mat'><p>Get mean expression of pathway genes in both groups</p></a></li>
<li><a href='#get_min_alpha'><p>Get the minimum alpha level for the permutation test</p></a></li>
<li><a href='#get_networks'><p>Get the two association networks</p></a></li>
<li><a href='#get_reactome_pathways'><p>Obtain Reactome pathways</p></a></li>
<li><a href='#head.dnapath_list'><p>Return the first part of the dnapath results.</p></a></li>
<li><a href='#init_mart'><p>Initialize biomaRt for a given species</p></a></li>
<li><a href='#length.dnapath_list'><p>The number of pathways in a 'dnapath_list' object.</p></a></li>
<li><a href='#meso'><p>Gene expression dataset for two groups</p></a></li>
<li><a href='#names.dnapath'><p>The pathway names in a 'dnapath' object.</p></a></li>
<li><a href='#names.dnapath_list'><p>The pathway names in a 'dnapath_list' object.</p></a></li>
<li><a href='#p53_pathways'><p>Reactome pathway list for Homo sapiens</p></a></li>
<li><a href='#plot_pair'><p>Plot the expression values of two genes</p></a></li>
<li><a href='#plot.dnapath'><p>Plot function for 'dnapath' object.</p></a></li>
<li><a href='#print.dnapath'><p>Print function for 'dnapath' object.</p></a></li>
<li><a href='#print.dnapath_list'><p>Print function for 'dnapath_list' object.</p></a></li>
<li><a href='#rename_genes'><p>Rename genes in the differential network analysis</p></a></li>
<li><a href='#rev.dnapath_list'><p>Reverse the order of pathways in a 'dnapath_list' object.</p></a></li>
<li><a href='#run_aracne'><p>Wrapper for ARACNE method</p></a></li>
<li><a href='#run_bc3net'><p>Wrapper for BC3Net method</p></a></li>
<li><a href='#run_c3net'><p>Wrapper for C3Net method</p></a></li>
<li><a href='#run_clr'><p>Wrapper for CLR method</p></a></li>
<li><a href='#run_corr'><p>Wrapper for correlation co-expression</p></a></li>
<li><a href='#run_genie3'><p>Wrapper for GENIE3 method</p></a></li>
<li><a href='#run_glasso'><p>Wrapper for glasso method</p></a></li>
<li><a href='#run_mrnet'><p>Wrapper for MRNET method</p></a></li>
<li><a href='#run_pcor'><p>Wrapper for partial correlations from corpcor</p></a></li>
<li><a href='#run_pcor_fdr'><p>Wrapper for partial correlations with Empirical Bayes FDR correction</p></a></li>
<li><a href='#run_silencer'><p>Wrapper for silencer method</p></a></li>
<li><a href='#sort.dnapath_list'><p>Sort function for 'dnapath_list' object.</p></a></li>
<li><a href='#subset.dnapath_list'><p>Subset function for 'dnapath_list' object.</p></a></li>
<li><a href='#summarize_edges'><p>Summarize differential connections for a pathway</p></a></li>
<li><a href='#summarize_genes'><p>Summarize the differential connectivity of genes over all pathways.</p></a></li>
<li><a href='#summarize_genes_for_pathway'><p>Summarize differential connectivity of genes in a pathway</p></a></li>
<li><a href='#summarize_pathways'><p>Summarize the differential connectivity of pathways.</p></a></li>
<li><a href='#summary.dnapath'><p>Summary function for 'dnapath' object.</p></a></li>
<li><a href='#summary.dnapath_list'><p>Summary function for 'dnapath_list' object.</p></a></li>
<li><a href='#symbol_to_entrez'><p>Obtain entrezgene IDs for gene symbols</p></a></li>
<li><a href='#tail.dnapath_list'><p>Return the last part of the dnapath results.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Network Analysis using Gene Pathways</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7.6</td>
</tr>
<tr>
<td>Description:</td>
<td>Integrates pathway information into the differential network analysis of two gene expression datasets as described in Grimes, Potter, and Datta (2019) &lt;<a href="https://doi.org/10.1038%2Fs41598-019-41918-3">doi:10.1038/s41598-019-41918-3</a>&gt;. Provides summary functions to break down the results at the pathway, gene, or individual connection level. The differential networks for each pathway of interest can be plotted, and the visualization will highlight any differentially expressed genes and all of the gene-gene associations that are significantly differentially connected.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.1), corpcor, igraph, ggplot2, SeqNet, Rdpack,
dplyr, tibble, stats, methods, curl, graphics, grDevices,
gtools, utils, wCorr, parallel</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0), knitr, rmarkdown, markdown, xml2, readr,
bc3net, huge, GENIE3, minet, AnnotationDbi, reactome.db,
biomaRt</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-21 01:57:02 UTC; gregory</td>
</tr>
<tr>
<td>Author:</td>
<td>Tyler Grimes [aut, cre],
  Somnath Datta [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tyler Grimes &lt;tylerlukegrimes@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-21 19:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B.dnapath'>Extract results of a single pathway from a 'dnapath' object.</h2><span id='topic++5B.dnapath'></span>

<h3>Description</h3>

<p>Extract results of a single pathway from a 'dnapath' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
x[i, ...]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object.</p>
</td></tr>
<tr><td><code id="+2B5B.dnapath_+3A_i">i</code></td>
<td>
<p>The index specifying which pathway to extract.</p>
</td></tr>
<tr><td><code id="+2B5B.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The 'dnapath' object unmodified
</p>


<h3>Note</h3>

<p>In the current implementation, there is nothing to subset on for
individual pathway results, so the original object is returned unmodified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways[[1]],
                   group_labels = meso$groups, n_perm = 10)
results[1]
</code></pre>

<hr>
<h2 id='+5B.dnapath_list'>Extract parts of a 'dnapath_list' object.</h2><span id='topic++5B.dnapath_list'></span>

<h3>Description</h3>

<p>Extract parts of a 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
x[i, ...]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B.dnapath_list_+3A_i">i</code></td>
<td>
<p>The indices of pathways to extract.</p>
</td></tr>
<tr><td><code id="+2B5B.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'dnapath_list' object containing pathways indexed by 'i'.
</p>

<hr>
<h2 id='+5B+5B.dnapath'>Extract results of a single pathway from a 'dnapath' object.</h2><span id='topic++5B+5B.dnapath'></span>

<h3>Description</h3>

<p>Extract results of a single pathway from a 'dnapath' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
x[[i, ...]]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B+2B5B.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.dnapath_+3A_i">i</code></td>
<td>
<p>The index specifying which pathway to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The 'dnapath' object unmodified
</p>


<h3>Note</h3>

<p>In the current implementation, there is nothing to subset on for
individual pathway results, so the original object is returned unmodified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways[[1]],
                   group_labels = meso$groups, n_perm = 10)
results[[1]]
</code></pre>

<hr>
<h2 id='+5B+5B.dnapath_list'>Extract results of a single pathway from a 'dnapath_list' object.</h2><span id='topic++5B+5B.dnapath_list'></span>

<h3>Description</h3>

<p>Extract results of a single pathway from a 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
x[[i, ...]]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B+2B5B.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.dnapath_list_+3A_i">i</code></td>
<td>
<p>The index specifying which pathway to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'dnapath' object containing a single pathway result.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
results[[1]]
</code></pre>

<hr>
<h2 id='+5B+26lt+3B-.dnapath'>Replace parts of a 'dnapath' object.</h2><span id='topic++5B+3C-.dnapath'></span>

<h3>Description</h3>

<p>This functionality is not implemented and will return an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 replacement method for class 'dnapath'
x[...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_+3A_value">value</code></td>
<td>
<p>A 'dnapath' object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Replacement is not defined; an error is generated.
</p>

<hr>
<h2 id='+5B+26lt+3B-.dnapath_list'>Replace parts of a 'dnapath_list' object.</h2><span id='topic++5B+3C-.dnapath_list'></span>

<h3>Description</h3>

<p>This functionality is not implemented and will return an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 replacement method for class 'dnapath_list'
x[...] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
<tr><td><code id="+2B5B+2B26lt+2B3B-.dnapath_list_+3A_value">value</code></td>
<td>
<p>A 'dnapath_list' object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Replacement is not defined; an error is generated.
</p>

<hr>
<h2 id='biomart_hsapiens'>Default mapping for entrezgene IDs and HGNC gene symbols</h2><span id='topic+biomart_hsapiens'></span>

<h3>Description</h3>

<p>This dataset is used by default if the connection to biomaRt fails. It
is highly recommended to retry the function call that attempted to connect to
biomaRt. Using this dataset in general may not produce the correct
results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomart_hsapiens
</code></pre>


<h3>Format</h3>

<p>A data.frame containing a mapping between entrezgene IDs and
HGNC gene symbols.
</p>

<hr>
<h2 id='c.dnapath'>Combine two 'dnapath' objects.</h2><span id='topic+c.dnapath'></span>

<h3>Description</h3>

<p>This functionality is not implemented and will return an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
c(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c.dnapath_+3A_...">...</code></td>
<td>
<p>'dnapath' objects to be concatenated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Concatenation is not defined; an error is generated.
</p>

<hr>
<h2 id='c.dnapath_list'>Combine two 'dnapath_list' objects.</h2><span id='topic+c.dnapath_list'></span>

<h3>Description</h3>

<p>This functionality is not implemented and will return an error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
c(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="c.dnapath_list_+3A_...">...</code></td>
<td>
<p>'dnapath_list' objects to be concatenated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Concatenation is not defined; an error is generated.
</p>

<hr>
<h2 id='combine_overlapping_pathways'>Modify a pathway list to combine overlapping pathways.</h2><span id='topic+combine_overlapping_pathways'></span>

<h3>Description</h3>

<p>Modify a pathway list to combine overlapping pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_overlapping_pathways(pathway_list, overlap_limit = 0.9)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine_overlapping_pathways_+3A_pathway_list">pathway_list</code></td>
<td>
<p>A list of pathways obtained from
<code><a href="#topic+get_reactome_pathways">get_reactome_pathways</a></code>.</p>
</td></tr>
<tr><td><code id="combine_overlapping_pathways_+3A_overlap_limit">overlap_limit</code></td>
<td>
<p>A percentage between 0 and 1. If two pathways
overlap by more than this amount, they are combined into one pathway.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A modified list with overlapping pathways combined together.
</p>

<hr>
<h2 id='d_edgesC'>C++ implementation of d_edges</h2><span id='topic+d_edgesC'></span>

<h3>Description</h3>

<p>Calculates differential network score for each edge in a network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>d_edgesC(nw1, nw2, lp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="d_edgesC_+3A_nw1">nw1</code></td>
<td>
<p>The association scores for network 1</p>
</td></tr>
<tr><td><code id="d_edgesC_+3A_nw2">nw2</code></td>
<td>
<p>The association scores for network 2</p>
</td></tr>
<tr><td><code id="d_edgesC_+3A_lp">lp</code></td>
<td>
<p>The lp value to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of differential network scores for the edges.
</p>

<hr>
<h2 id='d_genesC'>C++ implementation of d_genes</h2><span id='topic+d_genesC'></span>

<h3>Description</h3>

<p>Calculates differential network score for a set of genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>d_genesC(nw1, nw2, lp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="d_genesC_+3A_nw1">nw1</code></td>
<td>
<p>The association scores for network 1</p>
</td></tr>
<tr><td><code id="d_genesC_+3A_nw2">nw2</code></td>
<td>
<p>The association scores for network 2</p>
</td></tr>
<tr><td><code id="d_genesC_+3A_lp">lp</code></td>
<td>
<p>The lp value to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of differential network scores for the genes.
</p>

<hr>
<h2 id='d_pathwayC'>C++ implementation of d_pathway</h2><span id='topic+d_pathwayC'></span>

<h3>Description</h3>

<p>Calculates differential network score for an entire pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>d_pathwayC(nw1, nw2, lp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="d_pathwayC_+3A_nw1">nw1</code></td>
<td>
<p>The association scores for network 1</p>
</td></tr>
<tr><td><code id="d_pathwayC_+3A_nw2">nw2</code></td>
<td>
<p>The association scores for network 2</p>
</td></tr>
<tr><td><code id="d_pathwayC_+3A_lp">lp</code></td>
<td>
<p>The lp value to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The differential network score for the pathway.
</p>

<hr>
<h2 id='dna_pathway'>Differential network analysis on a single pathway</h2><span id='topic+dna_pathway'></span>

<h3>Description</h3>

<p>This is an internal function used by <code><a href="#topic+dnapath">dnapath</a></code> to conduct
the differential network analysis on a single pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dna_pathway(
  x,
  pathway,
  group_prob,
  network_inference,
  permutations,
  lp = 2,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dna_pathway_+3A_x">x</code></td>
<td>
<p>The gene expression data to be analyzed. Assumed to already be
processed by <code><a href="#topic+dnapath">dnapath</a></code></p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_pathway">pathway</code></td>
<td>
<p>A single vector containing gene names.</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_group_prob">group_prob</code></td>
<td>
<p>A matrix with number of rows equal to the
number of rows in <code>x</code> and two columns, each containing the probability that
a given observation belongs to each group.</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_network_inference">network_inference</code></td>
<td>
<p>A function used to infer the pathway network.</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_permutations">permutations</code></td>
<td>
<p>A matrix containing the permutations to perform during
permutation testing. If <code>NULL</code>, the permutation tests are not performed.</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_lp">lp</code></td>
<td>
<p>The lp value used to compute differential connectivity
scores. (Note: If a vector is provided, then the results are returned as
a list of <code>dnapath_list</code> objects, one result for each value of <code>lp</code>. This
option is available so that network inference methods only need to be run
once for each pathway when multple values of <code>lp</code> are being considered. This
may be useful when conducting simulation studies).</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_verbose">verbose</code></td>
<td>
<p>Set to TRUE to turn on messages.</p>
</td></tr>
<tr><td><code id="dna_pathway_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed into the network inference function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the results of the differential network analysis
for a single pathway.
</p>

<hr>
<h2 id='dnapath'>Differential Network Analysis Using Gene Pathways</h2><span id='topic+dnapath'></span>

<h3>Description</h3>

<p>Integrates pathways into the differential network analysis of gene expression
data (Grimes et al. 2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dnapath(
  x,
  pathway_list,
  group_labels = NULL,
  network_inference = run_pcor,
  n_perm = 100,
  lp = 2,
  seed = NULL,
  verbose = FALSE,
  mc.cores = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dnapath_+3A_x">x</code></td>
<td>
<p>The gene expression data to be analyzed. This can be either (1) a
list of two matrices or data frames that contain the gene expression profile
from each of two populations (groups) &ndash; with rows corresponding to samples
and columns to genes &ndash; or (2) a single matrix or data frame
that contains the expression profiles for both groups. For case (2), the
<code>group_labels</code> argument must be specified to identify which rows belong to which
group.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_pathway_list">pathway_list</code></td>
<td>
<p>A single vector or list of vectors containing gene names
to indicate pathway membership. The vectors are used to subset the columns
of the matrices in <code>x</code>. A pathway list can be obtained using
<code><a href="#topic+get_reactome_pathways">get_reactome_pathways</a></code>. If NULL, then the entire expression
dataset is analyzed as a single network (this approach is not recommended
unless there are only a small number of genes).</p>
</td></tr>
<tr><td><code id="dnapath_+3A_group_labels">group_labels</code></td>
<td>
<p>If <code>x</code> is a single matrix or data frame, <code>group_labels</code> must
be specified to label each row. <code>group_labels</code> is a matrix each row
corresponding to a in <code>x</code>. This matrix may either (1) have a single column
containing the group label for each observation, or (2) individual columns
representing each group with values in <code style="white-space: pre;">&#8288;[0, 1]&#8288;</code> representing the probability
that the patient in that row is in each group. In the latter case, if the
rows do not sum to 1, then each entry will be divided by its row sum.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_network_inference">network_inference</code></td>
<td>
<p>A function used to infer the pathway network. It
should take in an n by p matrix and return a p by p matrix of association
scores. (Built-in options include: <code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>.)
Defaults to <code><a href="#topic+run_pcor">run_pcor</a></code> for partial correlations.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_n_perm">n_perm</code></td>
<td>
<p>The number of random permutations to perform during
permutation testing. If <code>n_perm == 1</code>, the permutation tests are not performed.
If <code>n_perm</code> is larger than the number of possible permutations, <code>n_perm</code> will
be set to this value with a warning message.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_lp">lp</code></td>
<td>
<p>The lp value used to compute differential connectivity
scores. (Note: If a vector is provided, then the results are returned as
a list of <code>dnapath_list</code> objects, one result for each value of <code>lp</code>. This
option is available so that network inference methods only need to be run
once for each pathway when multple values of <code>lp</code> are being considered. This
may be useful when conducting simulation studies).</p>
</td></tr>
<tr><td><code id="dnapath_+3A_seed">seed</code></td>
<td>
<p>(Optional) Used to set.seed prior to permutation test for
each pathway. This allows results for individual pathways to be easily
reproduced.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_verbose">verbose</code></td>
<td>
<p>Set to TRUE to turn on messages.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_mc.cores">mc.cores</code></td>
<td>
<p>Used in <code><a href="parallel.html#topic+mclapply">mclapply</a></code> to run the differential
network analysis in parallel across pathways. Must be set to 1 if on a Windows
machine.</p>
</td></tr>
<tr><td><code id="dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed into the network inference function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'dnapath_list' or 'dnapath' object containing results for each
pathway in <code>pathway_list</code>.
</p>


<h3>References</h3>

<p>Grimes T, Potter SS, Datta S (2019).
&ldquo;Integrating Gene Regulatory Pathways into Differential Network Analysis of Gene Expression Data.&rdquo;
<em>Scientific reports</em>, <b>9</b>(1), 5479.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+filter_pathways">filter_pathways</a></code>, <code><a href="#topic+summary.dnapath_list">summary.dnapath_list</a></code>
<code><a href="#topic+subset.dnapath_list">subset.dnapath_list</a></code>, <code><a href="#topic+sort.dnapath_list">sort.dnapath_list</a></code>,
<code><a href="#topic+plot.dnapath">plot.dnapath</a></code>, <code><a href="#topic+rename_genes">rename_genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
results
summary(results) # Summary over all pathways in the pathway list.
# Remove results for pathways with p-values above 0.2.
top_results &lt;- filter_pathways(results, 0.2)
# Sort the top results by the pathway DC score.
top_results &lt;- sort(top_results, by = "dc_score")
top_results
summary(top_results[[1]])  # Summary of pathway 1.
plot(results[[1]]) # Plot of the differential network for pathway 1.

# Use ... to adjust arguments in the network inference function.
# For example, using run_corr() with method = "spearman":
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10,
                   network_inference = run_corr,
                   method = "spearman")
results
</code></pre>

<hr>
<h2 id='entrez_to_symbol'>Obtain gene symbols for entrezgene IDs</h2><span id='topic+entrez_to_symbol'></span>

<h3>Description</h3>

<p>Uses <code>biomaRt</code> (Durinck et al. 2009) to map entrezgene IDs to gene
symbols for a given species. Obtains MGI symbols for mouse species and
HGNC symbols for other species.
(Note: this mapping may not work for all species.)
The output of this function can be used in <code><a href="#topic+rename_genes">rename_genes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>entrez_to_symbol(
  x,
  species,
  symbol_name = NULL,
  dir_save = tempdir(),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="entrez_to_symbol_+3A_x">x</code></td>
<td>
<p>A vector of entrezgene IDs.</p>
</td></tr>
<tr><td><code id="entrez_to_symbol_+3A_species">species</code></td>
<td>
<p>The species used to obtain the entrezgene IDs. For example:
&quot;Homo sapiens&quot;, &quot;m musculus&quot;, &quot;C. elegans&quot;, or &quot;S cerevisiae&quot;.
&quot;Human&quot; and &quot;mouse&quot; can also be used and will be converted to the
correct species name.</p>
</td></tr>
<tr><td><code id="entrez_to_symbol_+3A_symbol_name">symbol_name</code></td>
<td>
<p>The type of gene symbol to use. If NULL, then &quot;hgnc_symbol&quot;
is used for HGNC symbols, unless <code>species</code> is &quot;mmusculus&quot;, in which case</p>
</td></tr>
<tr><td><code id="entrez_to_symbol_+3A_dir_save">dir_save</code></td>
<td>
<p>The directory to store annotation reference. Future
calls to this function will use the stored annotations. This speeds up the
operation and allows for reproducibility in the event that the <code>biomaRt</code>
database is updated. Set to NULL to disable. By default, it uses a
temporary directory to store files during the R session.
&quot;mgi_symbol&quot; is used.</p>
</td></tr>
<tr><td><code id="entrez_to_symbol_+3A_verbose">verbose</code></td>
<td>
<p>Set to FALSE to avoid messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If entrezgene IDs are used in a <code>dnapath_list</code> or <code>dnapath</code>
object, or a pathway list, then <code><a href="#topic+get_genes">get_genes</a></code> can be used to
extract them and used for the <code>x</code> argument here.
</p>


<h3>Value</h3>

<p>A data frame with two columns: the first contains the original
entrezgene IDs, and the second contains the corresponding gene symbols.
MGI symbols are returned when <code>species = "Mus musculus"</code> and HGNC symbols
are returned otherwise.
</p>


<h3>Note</h3>

<p>Internet connection is required to connect to biomaRt. If unavailable, the
default biomart and default species contained in the package is used, but
this may not match the desired species.
</p>


<h3>References</h3>

<p>Durinck S, Spellman PT, Birney E, Huber W (2009).
&ldquo;Mapping Identifiers for the Integration of Genomic Datasets with the R/Bioconductor Package biomaRt.&rdquo;
<em>Nature Protocols</em>, <b>4</b>, 1184&ndash;1191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+symbol_to_entrez">symbol_to_entrez</a></code>, <code><a href="#topic+get_genes">get_genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(meso)
# The meso gene expression data contains entrezgene IDs. 
# These can be converted to gene symbols.
gene_mat &lt;- entrez_to_symbol(colnames(meso$gene_expression), species = "human")

</code></pre>

<hr>
<h2 id='filter_pathways'>Remove pathways with non-significant DC scores.</h2><span id='topic+filter_pathways'></span>

<h3>Description</h3>

<p>Remove pathways with non-significant DC scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_pathways(x, alpha_pathway = NULL, monotonized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_pathways_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="filter_pathways_+3A_alpha_pathway">alpha_pathway</code></td>
<td>
<p>Threshold for pathway p-values to determine significance.
If NULL, defaults to 0.05 or the minimum possible threshold (based on the
number of permutatiosn that were run).</p>
</td></tr>
<tr><td><code id="filter_pathways_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'dnapath_list' object containing only those pathways with differential
connectivity p-values below <code>alpha</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
results_sig &lt;- filter_pathways(results)
</code></pre>

<hr>
<h2 id='format_species_name'>Format sepcies name input.</h2><span id='topic+format_species_name'></span>

<h3>Description</h3>

<p>Internal function used to format species names to be used with biomart.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_species_name(species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="format_species_name_+3A_species">species</code></td>
<td>
<p>The species to obtain a biomart dataset for. For example:
&quot;Homo sapiens&quot;, &quot;m musculus&quot;, &quot;C. elegans&quot;, or &quot;S cerevisiae&quot;, and
&quot;Human&quot; and &quot;mouse&quot; will be converted to the format required in biomart.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string containing the formatted species name.
</p>

<hr>
<h2 id='get_biomart_mapping'>Internal function for obtaining biomaRt mapping</h2><span id='topic+get_biomart_mapping'></span>

<h3>Description</h3>

<p>Internal function for obtaining biomaRt mapping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biomart_mapping(species, symbol_name, dir_save, verbose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_biomart_mapping_+3A_species">species</code></td>
<td>
<p>The species to obtain a biomart dataset for. This should be
an output from <code><a href="#topic+format_species_name">format_species_name</a></code>.</p>
</td></tr>
<tr><td><code id="get_biomart_mapping_+3A_symbol_name">symbol_name</code></td>
<td>
<p>The type of gene symbol to use. HGNC symbols are used by
default (hgnc_symbol), unless <code>species</code> is &quot;mmusculus&quot;, in which case MGI
symbols are used (mgi_symbol).</p>
</td></tr>
<tr><td><code id="get_biomart_mapping_+3A_dir_save">dir_save</code></td>
<td>
<p>The directory to store annotation reference. Future
calls to this function will use the stored annotations. This speeds up the
operation and allows for reproducibility in the event that the <code>biomaRt</code>
database is updated. Set to NULL to disable. By default, it uses a
temporary directory to store files during the R session.</p>
</td></tr>
<tr><td><code id="get_biomart_mapping_+3A_verbose">verbose</code></td>
<td>
<p>Set to FALSE to avoid messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing a mapping between entrezgene IDs and gene
symbols.
</p>

<hr>
<h2 id='get_genes'>Get the gene names from a differential network analysis</h2><span id='topic+get_genes'></span>

<h3>Description</h3>

<p>Get the gene names from a differential network analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_genes(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_genes_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' or 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>,
or a pathway list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector containing all the genes in <code>x</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rename_genes">rename_genes</a></code>, <code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code>,
<code><a href="#topic+symbol_to_entrez">symbol_to_entrez</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
genes &lt;- get_genes(results)
</code></pre>

<hr>
<h2 id='get_mean_expr_mat'>Get mean expression of pathway genes in both groups</h2><span id='topic+get_mean_expr_mat'></span>

<h3>Description</h3>

<p>Get mean expression of pathway genes in both groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mean_expr_mat(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mean_expr_mat_+3A_x">x</code></td>
<td>
<p>A 'dnapath' or 'dnapath_list' object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of two rows whose columns contain the mean expression of
each gene in the pathway.
</p>

<hr>
<h2 id='get_min_alpha'>Get the minimum alpha level for the permutation test</h2><span id='topic+get_min_alpha'></span>

<h3>Description</h3>

<p>This method is used internally by several methods to determine the
minimum significance threshold (alpha value) that can be applied to the
permutation p-values obtained in the differential network analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_min_alpha(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_min_alpha_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' or 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The minimum alpha level that can be used based on the number
of permutations performed in the analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 5)
get_min_alpha(results) # 1 / (5 + 1) = 0.167
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
get_min_alpha(results) # 1 / (10 + 1) = 0.091
</code></pre>

<hr>
<h2 id='get_networks'>Get the two association networks</h2><span id='topic+get_networks'></span>

<h3>Description</h3>

<p>Extracts the estimated association network for each group from the
differential network analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_networks(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_networks_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two association matrices.
</p>


<h3>Note</h3>

<p>The two matrices can be plotted using the
<code><a href="SeqNet.html#topic+plot_network">plot_network</a></code> function from the SeqNet package, as
illustrated in the examples below.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
# Extract the two estimated association networks for the first pathway
nw &lt;- get_networks(results[[1]])
# Plot the networks using the SeqNet::plot_network function.
# Note that the `compare_graph` argument is used so that the same node layout
# is used across all of the plots.
# Plot the two networks (in separate plots)
g &lt;- SeqNet::plot_network(nw[[1]])
SeqNet::plot_network(nw[[1]], compare_graph = g)
# Plot of the differential network for pathway 1.
# Again, the `compare_graph` argument is used to maintain the same layout.
plot(results[[1]], compare_graph = g) 
# We see that genes 51230 and 7311 show strong differential connectivity.
# The plot_pair() function can be used to investigate these two genes further.
plot_pair(results[[1]], "51230", "7311")
</code></pre>

<hr>
<h2 id='get_reactome_pathways'>Obtain Reactome pathways</h2><span id='topic+get_reactome_pathways'></span>

<h3>Description</h3>

<p>Connects to <code>reactome.db</code> (Ligtenberg 2019)
to obtain a list of pathways for a given species.
The pathway list is processed by combining any two pathways that have
substantial overlap (default is over 90% overlap). This output if this
function can be used for the <code>pathway_list</code> argument in
<code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_reactome_pathways(
  species,
  overlap_limit = 0.9,
  min_size = 10,
  max_size = 50,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_reactome_pathways_+3A_species">species</code></td>
<td>
<p>A string, for example &quot;Homo sapiens&quot; or &quot;Mus musculus&quot;,
indicating the species to use.</p>
</td></tr>
<tr><td><code id="get_reactome_pathways_+3A_overlap_limit">overlap_limit</code></td>
<td>
<p>(Optional) Any pathways that have an overlap
greater than overlap_limit are combined. Set to NULL to disable
this option.</p>
</td></tr>
<tr><td><code id="get_reactome_pathways_+3A_min_size">min_size</code></td>
<td>
<p>The minimum pathway size. Any Reactome pathways with fewer
than <code>min_size</code> genes are removed from the list. Defaults to 10.</p>
</td></tr>
<tr><td><code id="get_reactome_pathways_+3A_max_size">max_size</code></td>
<td>
<p>The maximum pathway size. Any Reactome pathways with more
than <code>max_size</code> genes are removed from the list. Defaults to 50.</p>
</td></tr>
<tr><td><code id="get_reactome_pathways_+3A_verbose">verbose</code></td>
<td>
<p>Set to FALSE to turn off messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list of vectors. Each vector corresponds to a Reactome pathway
and contains the entrezgene IDs of the genes in that pathway.
</p>


<h3>References</h3>

<p>Ligtenberg W (2019).
<em>reactome.db: A Set of Annotation Maps for Reactome</em>.
R package version 1.68.0.
</p>


<h3>See Also</h3>

<p>The genes in the Reactome pathways use entrezgene IDs. These can be converted
to gene symbols, if desired, using the <code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code> and
<code><a href="#topic+rename_genes">rename_genes</a></code> functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtaining a pathway list for human (Homo sapiens).
# In this example, overlapping pathways are not combined (this is
# specified by setting overlap_limit to NULL).
pathway_list &lt;- get_reactome_pathways("Homo sapiens", overlap_limit = NULL,
                                      min_size = 10, max_size = 20)
</code></pre>

<hr>
<h2 id='head.dnapath_list'>Return the first part of the dnapath results.</h2><span id='topic+head.dnapath_list'></span>

<h3>Description</h3>

<p>Return the first part of the dnapath results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
head(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="head.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object.</p>
</td></tr>
<tr><td><code id="head.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional paramters are passed into
<code><a href="#topic+summary.dnapath_list">summary.dnapath_list</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the first five rows of the summary table of the
'dnapath_list' object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
head(results)
</code></pre>

<hr>
<h2 id='init_mart'>Initialize biomaRt for a given species</h2><span id='topic+init_mart'></span>

<h3>Description</h3>

<p>Initialize biomaRt for a given species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>init_mart(species)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="init_mart_+3A_species">species</code></td>
<td>
<p>The species to obtain a biomart dataset for. This should be
an output from <code><a href="#topic+format_species_name">format_species_name</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A biomaRt object.
</p>


<h3>Note</h3>

<p>Requires internet connection. This function is called by
<code><a href="#topic+get_biomart_mapping">get_biomart_mapping</a></code>, which checks for internet connection
prior to calling this function.
</p>

<hr>
<h2 id='length.dnapath_list'>The number of pathways in a 'dnapath_list' object.</h2><span id='topic+length.dnapath_list'></span>

<h3>Description</h3>

<p>The number of pathways in a 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
length(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="length.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
length(results)
</code></pre>

<hr>
<h2 id='meso'>Gene expression dataset for two groups</h2><span id='topic+meso'></span>

<h3>Description</h3>

<p><code>meso</code> is a list containing gene expression data from Mesothelioma tumors
generated by The Cancer Genome Atlas (TCGA) and obtained using the
LinkedOmics portal. The first element in the list, named &quot;gene_expression&quot;,
contains 32 samples (rows) with 150 genes (columns). The second element,
named &quot;groups&quot;, is a vector of length 32 indicating which group (stage ii
or stage iv) each gene expression sample belongs to. See the &quot;Package data&quot;
vignette for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meso
</code></pre>


<h3>Format</h3>

<p>A list containing two items:
</p>

<dl>
<dt>$gene_expression</dt><dd><p>A 32 by 150 matrix of gene expression values</p>
</dd>
<dt>$groups</dt><dd><p>A vector of length 32 indicating which group (stageii or
stageiv) each of the rows in the gene expression data belong to.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://www.linkedomics.org/data_download/TCGA-GBMLGG/">http://www.linkedomics.org/data_download/TCGA-GBMLGG/</a>
</p>

<hr>
<h2 id='names.dnapath'>The pathway names in a 'dnapath' object.</h2><span id='topic+names.dnapath'></span>

<h3>Description</h3>

<p>The pathway names in a 'dnapath' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
names(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code> or from subsetting
a 'dnapath_list'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The pathway's name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
names(results[[1]])
</code></pre>

<hr>
<h2 id='names.dnapath_list'>The pathway names in a 'dnapath_list' object.</h2><span id='topic+names.dnapath_list'></span>

<h3>Description</h3>

<p>The pathway names in a 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
names(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The pathway names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
names(results)
</code></pre>

<hr>
<h2 id='p53_pathways'>Reactome pathway list for Homo sapiens</h2><span id='topic+p53_pathways'></span>

<h3>Description</h3>

<p>This is a pathway list obtained from <code><a href="#topic+get_reactome_pathways">get_reactome_pathways</a></code>
with <code>species = "human"</code> (used <code>reactome.db</code> version 1.68.0).
Only pathways with &quot;p53&quot; in their name are retained (to subset on some
cancer-related pathways). The list contains 13 total pathways. See the
&quot;Package data&quot; vignette for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p53_pathways
</code></pre>


<h3>Format</h3>

<p>A list of 13 vectors each containing a set of entregene IDs.
</p>

<hr>
<h2 id='plot_pair'>Plot the expression values of two genes</h2><span id='topic+plot_pair'></span>

<h3>Description</h3>

<p>Inspired by the <code>plotCors</code> function from the DGCA package,
this function is used to plot the expression values of two genes contained
in the differential network analysis results. This is useful for comparing
the marginal relationship between two genes. Note, however, that this
visualization is not able to show conditional associations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pair(
  x,
  gene_A,
  gene_B,
  method = "loess",
  alpha = 0.5,
  se_alpha = 0.1,
  use_facet = FALSE,
  scales = "fixed",
  colors = c(rgb(0.31, 0.58, 0.8, 0.9), rgb(1, 0.19, 0.19, 0.9))
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pair_+3A_x">x</code></td>
<td>
<p>A 'dnapath' or 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_gene_a">gene_A</code></td>
<td>
<p>The name of the first gene to plot. Must be one of the names
in <code><a href="#topic+get_genes">get_genes</a></code>(x).</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_gene_b">gene_B</code></td>
<td>
<p>The name of the second gene to plot. Must be one of the names
in <code><a href="#topic+get_genes">get_genes</a></code>(x).</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_method">method</code></td>
<td>
<p>A charater string, either &quot;lm&quot; or &quot;loess&quot; (the default)
used by <code><a href="ggplot2.html#topic+geom_smooth">geom_smooth</a></code> to summarize the marginal
gene-gene association. For no line, set method = NULL.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_alpha">alpha</code></td>
<td>
<p>Sets the transparancy of the points, used to set alpha in
<code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_se_alpha">se_alpha</code></td>
<td>
<p>Sets the transparancy of the confidence band around
the association trend line. Set to 0 to remove the band.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_use_facet">use_facet</code></td>
<td>
<p>If TRUE, the groups are plotted in separate graphs
using the <code>link[ggplot2]{facet_wrap}</code> method.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_scales">scales</code></td>
<td>
<p>Only used if do_facet_wrap is TRUE. See
<code>link[ggplot2]{facet_wrap}</code> for details.</p>
</td></tr>
<tr><td><code id="plot_pair_+3A_colors">colors</code></td>
<td>
<p>A vector of length two containing the rgb colors used for the
two groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots the differential network and returns the ggplot object.
Additional modifications can be applied to this object just
like any other ggplot.
</p>


<h3>References</h3>

<p>Grimes T, Datta S (2019).
<em>SeqNet: Generate RNA-Seq Data from Gene-Gene Association Networks</em>.
R package version 1.1.0, <a href="https://CRAN.R-project.org/package=SeqNet">https://CRAN.R-project.org/package=SeqNet</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
# Plot of the marginal association between the first two genes.
genes &lt;- get_genes(results)[1:2]
g &lt;- plot_pair(results, genes[1], genes[2])
# The ggplot object, g, can be further modified.
# Here we move the legend and use a log scale for the expression values
# (the log scale doesn't help with these data but is shown for demonstration).
g &lt;- g +
  ggplot2::theme(legend.position = "bottom") +
  ggplot2::scale_x_log10() +
  ggplot2::scale_y_log10()
g
</code></pre>

<hr>
<h2 id='plot.dnapath'>Plot function for 'dnapath' object.</h2><span id='topic+plot.dnapath'></span>

<h3>Description</h3>

<p>Uses the plotting functions for networks from the <code>SeqNet</code> package
(Grimes and Datta 2019)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
plot(
  x,
  alpha = NULL,
  monotonized = FALSE,
  only_dc = FALSE,
  require_dc_genes = FALSE,
  scale_edges = 1,
  scale_nodes = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values to infer differentially connected edges.
If NULL (the default) then no edges are removed from the plot.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized (i.e. step-down) p-values from the
permutation test will be used.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_only_dc">only_dc</code></td>
<td>
<p>If TRUE, only differentially connected edges will be shown;
any edges that are present in both groups are hidden. If FALSE, the edges
shared by both groups are shown. If a non-sparse estimator for network
edges is used, then the graph may be dense and setting this argument to TRUE
will be useful for highlighting the DC edges.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_require_dc_genes">require_dc_genes</code></td>
<td>
<p>If TRUE, the gene-level differential connectivity
p-value of the two genes for a given edge are also considered when deciding
whether an edge is differentially connected. If neither gene is significantly
differentially connected, then the edge between them will not be either.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_scale_edges">scale_edges</code></td>
<td>
<p>(Optional) multiplier for edge widths.</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_scale_nodes">scale_nodes</code></td>
<td>
<p>(Optional) multiplier for node radius</p>
</td></tr>
<tr><td><code id="plot.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed into the plotting function
<code><a href="SeqNet.html#topic+plot_network">plot_network</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots the differential network and returns the graph object.
See <code><a href="SeqNet.html#topic+plot_network">plot_network</a></code> for details.
</p>


<h3>References</h3>

<p>Grimes T, Datta S (2019).
<em>SeqNet: Generate RNA-Seq Data from Gene-Gene Association Networks</em>.
R package version 1.1.0, <a href="https://CRAN.R-project.org/package=SeqNet">https://CRAN.R-project.org/package=SeqNet</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
# Plot of the differential network for pathway 1.
plot(results[[1]]) 
# Plot of the differential network for pathway 1; remove any edges from
# the plot that have p-values above 0.1.
plot(results[[1]], alpha = 0.1) 
</code></pre>

<hr>
<h2 id='print.dnapath'>Print function for 'dnapath' object.</h2><span id='topic+print.dnapath'></span>

<h3>Description</h3>

<p>Print function for 'dnapath' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.dnapath_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="print.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a summary of the module.
</p>

<hr>
<h2 id='print.dnapath_list'>Print function for 'dnapath_list' object.</h2><span id='topic+print.dnapath_list'></span>

<h3>Description</h3>

<p>Print function for 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="print.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a summary of the module.
</p>

<hr>
<h2 id='rename_genes'>Rename genes in the differential network analysis</h2><span id='topic+rename_genes'></span>

<h3>Description</h3>

<p>Rename genes in the differential network analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_genes(x, gene_mat = NULL, to = NULL, species = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rename_genes_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' or 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>,
a pathway list, or a vector of gene names.</p>
</td></tr>
<tr><td><code id="rename_genes_+3A_gene_mat">gene_mat</code></td>
<td>
<p>(Optional) A matrix of key value pairs. The first column should contain
current gene names, and the second column the new names. Any genes that are
not in this matrix will retain their current names. This can be any
user-defined mapping, or the mapping obtained using
<code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code> or <code><a href="#topic+symbol_to_entrez">symbol_to_entrez</a></code>.</p>
</td></tr>
<tr><td><code id="rename_genes_+3A_to">to</code></td>
<td>
<p>(Optional) Setting <code>to = "symbol"</code> will rename entrezgene IDs to
gene symbols; this will automatically call the <code>entrez_to_symbol()</code> function
to obtain the mapping for <code>gene_mat</code>. The <code>species</code> argument must also be
specified when <code>to</code> is used.</p>
</td></tr>
<tr><td><code id="rename_genes_+3A_species">species</code></td>
<td>
<p>(Optional) Must be specified when setting <code>to = "symbol"</code>. This
argument is passed into <code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code>.</p>
</td></tr>
<tr><td><code id="rename_genes_+3A_...">...</code></td>
<td>
<p>Additional arugments are passed into <code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code>
in the case that <code>to</code> and <code>species</code> are specified. This may be useful to
specify the <code>dir_save</code> argument to save the mapping obtained from
biomaRt for offline use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>x</code> with all gene names updated according to <code>gene_mat</code>.
</p>


<h3>Note</h3>

<p>Internet connection is required to connect to use
<code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code> or <code><a href="#topic+symbol_to_entrez">symbol_to_entrez</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code>, <code><a href="#topic+symbol_to_entrez">symbol_to_entrez</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summary(results[[1]])  # Summary of pathway 1; note that it uses entrezgene IDs.


# Rename the entrezgene IDs into gene symbols.
results_sym &lt;- rename_genes(results, to = "symbol", species = "human")
summary(results_sym[[1]]) # Now the summary uses gene symbols.

</code></pre>

<hr>
<h2 id='rev.dnapath_list'>Reverse the order of pathways in a 'dnapath_list' object.</h2><span id='topic+rev.dnapath_list'></span>

<h3>Description</h3>

<p>Reverse the order of pathways in a 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
rev(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rev.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="rev.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'dnapath_list' object containing the pathways in 'x' in reverse order.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
# Filter out pathways that have p-values above 0.2.
results &lt;- filter_pathways(results, 0.2) 
results &lt;- sort(results, by = "dc_score") # Sort by the pathway DC score.
results &lt;- rev(results) # Reverse the ordering.
</code></pre>

<hr>
<h2 id='run_aracne'>Wrapper for ARACNE method</h2><span id='topic+run_aracne'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using ARACNE (Margolin et al. 2006).
Uses the implementation from the <code>minet</code> package (Meyer et al. 2008).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_aracne(
  x,
  weights = NULL,
  estimator = "spearman",
  disc = "none",
  nbins = NULL,
  eps = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_aracne_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_estimator">estimator</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+build.mim">build.mim</a></code>.</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_disc">disc</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+build.mim">build.mim</a></code>.</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_nbins">nbins</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+build.mim">build.mim</a></code>.</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_eps">eps</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+aracne">aracne</a></code>.</p>
</td></tr>
<tr><td><code id="run_aracne_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Margolin AA, Nemenman I, Basso K, Wiggins C, Stolovitzky G, Dalla Favera R, Califano A (2006).
&ldquo;ARACNE: An Algorithm for the Reconstruction of Gene Regulatory Networks in a Mammalian Cellular Context.&rdquo;
In <em>BMC Bioinformatics</em>, volume 7(1), S7.
BioMed Central.
</p>
<p>Meyer PE, Lafitte F, Bontempi G (2008).
&ldquo;<span class="pkg">minet</span>: A R/Bioconductor Package for Inferring Large Transcriptional Networks using Mutual Information.&rdquo;
<em>BMC Bioinformatics</em>, <b>9</b>(1), 461.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 5 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 5

# Use this method to perform differential network analysis.
# The parameters in run_aracne() can be adjusted using the ... argument.
# For example, the 'estimator' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_aracne,
                   estimator = "spearman")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_bc3net'>Wrapper for BC3Net method</h2><span id='topic+run_bc3net'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using BC3Net (Matos Simoes and Emmert-Streib 2012).
Uses the implementation from the <code>bc3net</code> package (de Matos Simoes and Emmert-Streib 2016).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_bc3net(
  x,
  weights = NULL,
  boot = 100,
  estimator = "spearman",
  disc = "equalwidth",
  mtc1 = TRUE,
  adj1 = "bonferroni",
  alpha1 = 0.05,
  mtc2 = TRUE,
  adj2 = "bonferroni",
  alpha2 = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_bc3net_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_boot">boot</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_estimator">estimator</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_disc">disc</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_mtc1">mtc1</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_adj1">adj1</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_alpha1">alpha1</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_mtc2">mtc2</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_adj2">adj2</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_alpha2">alpha2</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+bc3net">bc3net</a></code>.</p>
</td></tr>
<tr><td><code id="run_bc3net_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Matos Simoes Rd, Emmert-Streib F (2012).
&ldquo;Bagging Statistical Network Inference from Large-Scale Gene Expression Data.&rdquo;
<em>PloS ONE</em>, <b>7</b>(3), e33624.
</p>
<p>de Matos Simoes R, Emmert-Streib F (2016).
<em>bc3net: Gene Regulatory Network Inference with Bc3net</em>.
R package version 1.0.4, <a href="https://CRAN.R-project.org/package=bc3net">https://CRAN.R-project.org/package=bc3net</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on one pathway from the p53 pathway list,
# and will only run 1 permutation for significance testing.
pathway_list &lt;- p53_pathways[13]
n_perm &lt;- 1

# Use this method to perform differential network analysis.
# The parameters in run_bc3net() can be adjusted using the ... argument.
# For example, the 'estimator' and 'boot' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_bc3net,
                   boot = 10,
                   estimator = "pearson",
                   mtc1 = FALSE,
                   mtc2 = FALSE)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_c3net'>Wrapper for C3Net method</h2><span id='topic+run_c3net'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using C3Net (Altay and Emmert-Streib 2010).
Uses the implementation from the <code>bc3net</code> package (de Matos Simoes and Emmert-Streib 2016).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_c3net(
  x,
  weights = NULL,
  estimator = "spearman",
  disc = "equalwidth",
  mtc = TRUE,
  adj = "bonferroni",
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_c3net_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_estimator">estimator</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+c3mtc">c3mtc</a></code>.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_disc">disc</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+c3mtc">c3mtc</a></code>.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_mtc">mtc</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+c3mtc">c3mtc</a></code>.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_adj">adj</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+c3mtc">c3mtc</a></code>.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_alpha">alpha</code></td>
<td>
<p>Argument is passed into <code><a href="bc3net.html#topic+c3mtc">c3mtc</a></code>.</p>
</td></tr>
<tr><td><code id="run_c3net_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Altay G, Emmert-Streib F (2010).
&ldquo;Inferring the Conservative Causal Core of Gene Regulatory Networks.&rdquo;
<em>BMC Systems Biology</em>, <b>4</b>(1), 132.
</p>
<p>de Matos Simoes R, Emmert-Streib F (2016).
<em>bc3net: Gene Regulatory Network Inference with Bc3net</em>.
R package version 1.0.4, <a href="https://CRAN.R-project.org/package=bc3net">https://CRAN.R-project.org/package=bc3net</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on one pathway from the p53 pathway list,
# and will only run 1 permutation for significance testing.
pathway_list &lt;- p53_pathways[13]
n_perm &lt;- 1

# Use this method to perform differential network analysis.
# The parameters in run_c3net() can be adjusted using the ... argument.
# For example, the 'estimator' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_c3net,
                   estimator = "pearson",
                   mtc = FALSE)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results) # Get networks for the pathway.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_clr'>Wrapper for CLR method</h2><span id='topic+run_clr'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using CLR (Faith et al. 2007).
Uses the implementation from the <code>minet</code> package (Meyer et al. 2008).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_clr(x, weights = NULL, estimator = "spearman", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_clr_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_clr_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_clr_+3A_estimator">estimator</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+build.mim">build.mim</a></code>.</p>
</td></tr>
<tr><td><code id="run_clr_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Faith JJ, Hayete B, Thaden JT, Mogno I, Wierzbowski J, Cottarel G, Kasif S, Collins JJ, Gardner TS (2007).
&ldquo;Large-Scale Mapping and Validation of Escherichia Coli Transcriptional Regulation from a Compendium of Expression Profiles.&rdquo;
<em>PLoS Biology</em>, <b>5</b>(1), e8.
</p>
<p>Meyer PE, Lafitte F, Bontempi G (2008).
&ldquo;<span class="pkg">minet</span>: A R/Bioconductor Package for Inferring Large Transcriptional Networks using Mutual Information.&rdquo;
<em>BMC Bioinformatics</em>, <b>9</b>(1), 461.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>,
<code><a href="#topic+run_genie3">run_genie3</a></code>, <code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 5 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 5

# Use this method to perform differential network analysis.
# The parameters in run_clr() can be adjusted using the ... argument.
# For example, the 'estimator' paramter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_clr,
                   estimator = "spearman")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_corr'>Wrapper for correlation co-expression</h2><span id='topic+run_corr'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using correlation for association measure.
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_corr(
  x,
  weights = NULL,
  threshold = NULL,
  method = c("pearson", "spearman"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_corr_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_corr_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_corr_+3A_threshold">threshold</code></td>
<td>
<p>Cutoff for significant associations. If NULL, all correlations
are returned. Otherwise, correlations of magnitude at or below this threshold are
set to zero.</p>
</td></tr>
<tr><td><code id="run_corr_+3A_method">method</code></td>
<td>
<p>Argument is passed into <code><a href="stats.html#topic+cor">cor</a></code>. Should be one
of &quot;pearson&quot; or &quot;spearman&quot;.</p>
</td></tr>
<tr><td><code id="run_corr_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 5 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 5

# Use this method to perform differential network analysis.
# The parameters in run_corr() can be adjusted using the ... argument.
# For example, the 'method' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_corr,
                   method = "spearman")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_genie3'>Wrapper for GENIE3 method</h2><span id='topic+run_genie3'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using GENIE3 (Huynh-Thu et al. 2010).
Uses the implementation from the <code>GENIE3</code> package.
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_genie3(x, weights = NULL, nTrees = 200, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_genie3_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_genie3_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_genie3_+3A_ntrees">nTrees</code></td>
<td>
<p>Argument is passed into <code><a href="GENIE3.html#topic+GENIE3">GENIE3</a></code>.</p>
</td></tr>
<tr><td><code id="run_genie3_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Huynh-Thu VA, Irrthum A, Wehenkel L, Geurts P (2010).
&ldquo;Inferring Regulatory Networks from Expression Data using Tree-Based Methods.&rdquo;
<em>PloS ONE</em>, <b>5</b>(9), e12776.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(!requireNamespace("GENIE3", quietly = TRUE)) {
data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 5 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 5

# Use this method to perform differential network analysis.
# The parameters in run_genie3() can be adjusted using the ... argument.
# For example, the 'nTrees' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_genie3,
                   nTrees = 100)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.

# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])
}
</code></pre>

<hr>
<h2 id='run_glasso'>Wrapper for glasso method</h2><span id='topic+run_glasso'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using glasso (Friedman et al. 2018).
Uses the implementation from the <code>huge</code> package (Jiang et al. 2019).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_glasso(
  x,
  method = c("glasso", "mb", "ct"),
  criterion = c("ric", "stars"),
  verbose = FALSE,
  weights = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_glasso_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_glasso_+3A_method">method</code></td>
<td>
<p>Argument is passed into <code><a href="huge.html#topic+huge">huge</a></code>.</p>
</td></tr>
<tr><td><code id="run_glasso_+3A_criterion">criterion</code></td>
<td>
<p>Argument is passed into <code><a href="huge.html#topic+huge.select">huge.select</a></code>.</p>
</td></tr>
<tr><td><code id="run_glasso_+3A_verbose">verbose</code></td>
<td>
<p>Argument is passed into <code><a href="huge.html#topic+huge">huge</a></code> and
<code><a href="huge.html#topic+huge.select">huge.select</a></code></p>
</td></tr>
<tr><td><code id="run_glasso_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_glasso_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Friedman J, Hastie T, Tibshirani R (2018).
<em><span class="pkg">glasso</span>: Graphical Lasso: Estimation of Gaussian Graphical Models</em>.
R package version 1.10.
</p>
<p>Jiang H, Fei X, Liu H, Roeder K, Lafferty J, Wasserman L, Li X, Zhao T (2019).
<em>huge: High-Dimensional Undirected Graph Estimation</em>.
R package version 1.3.3, <a href="https://CRAN.R-project.org/package=huge">https://CRAN.R-project.org/package=huge</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>,
<code><a href="#topic+run_genie3">run_genie3</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on one pathway from the p53 pathway list,
# and will only run 1 permutation for significance testing.
pathway_list &lt;- p53_pathways[13]
n_perm &lt;- 1

# Use this method to perform differential network analysis.
# The parameters in run_glasso() can be adjusted using the ... argument.
# For example, the 'criterion' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_glasso,
                   criterion = "ric")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_mrnet'>Wrapper for MRNET method</h2><span id='topic+run_mrnet'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using MRNET (Meyer et al. 2007).
Uses the implementation from the <code>minet</code> package (Meyer et al. 2008).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_mrnet(x, estimator = "spearman", weights = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_mrnet_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_mrnet_+3A_estimator">estimator</code></td>
<td>
<p>Argument is passed into <code><a href="minet.html#topic+build.mim">build.mim</a></code>.</p>
</td></tr>
<tr><td><code id="run_mrnet_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_mrnet_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Meyer PE, Kontos K, Lafitte F, Bontempi G (2007).
&ldquo;Information-Theoretic Inference of Large Transcriptional Regulatory Networks.&rdquo;
<em>EURASIP Journal on Bioinformatics and Systems Biology</em>, <b>2007</b>, 8&ndash;8.
</p>
<p>Meyer PE, Lafitte F, Bontempi G (2008).
&ldquo;<span class="pkg">minet</span>: A R/Bioconductor Package for Inferring Large Transcriptional Networks using Mutual Information.&rdquo;
<em>BMC Bioinformatics</em>, <b>9</b>(1), 461.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>,
<code><a href="#topic+run_genie3">run_genie3</a></code>, <code><a href="#topic+run_glasso">run_glasso</a></code>,
<code><a href="#topic+run_pcor">run_pcor</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 3 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 3

# Use this method to perform differential network analysis.
# The parameters in run_mrnet() can be adjusted using the ... argument.
# For example, the 'estimator' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_mrnet,
                   estimator = "spearman")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_pcor'>Wrapper for partial correlations from corpcor</h2><span id='topic+run_pcor'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using full partial correlations; these are
computed using the shrinkage approach for covariance estimation
(Schfer and Strimmer 2005) from the
<code>corpcor</code> package (Schafer et al. 2017).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pcor(x, weights = NULL, ranks = FALSE, verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_pcor_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_pcor_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_pcor_+3A_ranks">ranks</code></td>
<td>
<p>If TRUE, the gene expression values will be converted to ranks
(across samples) prior to covariance estimation.</p>
</td></tr>
<tr><td><code id="run_pcor_+3A_verbose">verbose</code></td>
<td>
<p>Argument is passed into <code><a href="corpcor.html#topic+pcor.shrink">pcor.shrink</a></code>.</p>
</td></tr>
<tr><td><code id="run_pcor_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Schfer J, Strimmer K (2005).
&ldquo;A Shrinkage Approach to Large-Scale Covariance Matrix Estimation and Implications for Functional Genomics.&rdquo;
<em>Statistical Applications in Genetics and Molecular Biology</em>, <b>4</b>(1), Article 32.
</p>
<p>Schafer J, Opgen-Rhein R, Zuber V, Ahdesmaki M, Silva APD, Strimmer. K (2017).
<em>corpcor: Efficient Estimation of Covariance and (Partial) Correlation</em>.
R package version 1.6.9, <a href="https://CRAN.R-project.org/package=corpcor">https://CRAN.R-project.org/package=corpcor</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 3 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 3

# Use this method to perform differential network analysis.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_pcor)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_pcor_fdr'>Wrapper for partial correlations with Empirical Bayes FDR correction</h2><span id='topic+run_pcor_fdr'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using full partial correlations; these are
computed using the shrinkage approach for covariance estimation
(Schfer and Strimmer 2005) from the
<code>corpcor</code> package (Schafer et al. 2017).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
This method will use Empirical Bayes FDR to set some estimates to zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pcor_fdr(
  x,
  weights = NULL,
  ranks = TRUE,
  thrsh = 1.5,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_pcor_fdr_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_pcor_fdr_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_pcor_fdr_+3A_ranks">ranks</code></td>
<td>
<p>If TRUE, the gene expression values will be converted to ranks
(across samples) prior to covariance estimation.</p>
</td></tr>
<tr><td><code id="run_pcor_fdr_+3A_thrsh">thrsh</code></td>
<td>
<p>A positive value (defaults to 1.5). This is used as
the cutoff for the likelihood ratio of the estimate local FDR.</p>
</td></tr>
<tr><td><code id="run_pcor_fdr_+3A_verbose">verbose</code></td>
<td>
<p>Argument is passed into <code><a href="corpcor.html#topic+pcor.shrink">pcor.shrink</a></code>.</p>
</td></tr>
<tr><td><code id="run_pcor_fdr_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Schfer J, Strimmer K (2005).
&ldquo;A Shrinkage Approach to Large-Scale Covariance Matrix Estimation and Implications for Functional Genomics.&rdquo;
<em>Statistical Applications in Genetics and Molecular Biology</em>, <b>4</b>(1), Article 32.
</p>
<p>Schafer J, Opgen-Rhein R, Zuber V, Ahdesmaki M, Silva APD, Strimmer. K (2017).
<em>corpcor: Efficient Estimation of Covariance and (Partial) Correlation</em>.
R package version 1.6.9, <a href="https://CRAN.R-project.org/package=corpcor">https://CRAN.R-project.org/package=corpcor</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>, and <code><a href="#topic+run_silencer">run_silencer</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 3 permutations for significance testing.
pathway_list &lt;- p53_pathways[c(8, 13)]
n_perm &lt;- 3

# Use this method to perform differential network analysis.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_pcor)
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results[[1]]) # Get networks for pathway 1.


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results[[1]]) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='run_silencer'>Wrapper for silencer method</h2><span id='topic+run_silencer'></span>

<h3>Description</h3>

<p>Conducts co-expression analysis using the matrix silencer method
(Barzel and Barabsi 2013).
Can be used for the <code>network_inference</code> argument in <code><a href="#topic+dnapath">dnapath</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_silencer(x, weights = NULL, method = "spearman", verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_silencer_+3A_x">x</code></td>
<td>
<p>A n by p matrix of gene expression data (n samples and p genes).</p>
</td></tr>
<tr><td><code id="run_silencer_+3A_weights">weights</code></td>
<td>
<p>An optional vector of weights. This is used by <code>dnapath()</code> to
apply the probabilistic group labels to each observation when estimating the
group-specific network.</p>
</td></tr>
<tr><td><code id="run_silencer_+3A_method">method</code></td>
<td>
<p>Argument is passed into <code><a href="stats.html#topic+cor">cor</a></code>.</p>
</td></tr>
<tr><td><code id="run_silencer_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, updates are printed during the estimation process.</p>
</td></tr>
<tr><td><code id="run_silencer_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A p by p matrix of association scores.
</p>


<h3>References</h3>

<p>Barzel B, Barabsi A (2013).
&ldquo;Network Link Prediction by Global Silencing of Indirect Correlations.&rdquo;
<em>Nature Biotechnology</em>, <b>31</b>(8), 720.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run_aracne">run_aracne</a></code>,
<code><a href="#topic+run_bc3net">run_bc3net</a></code>, <code><a href="#topic+run_c3net">run_c3net</a></code>,
<code><a href="#topic+run_clr">run_clr</a></code>, <code><a href="#topic+run_corr">run_corr</a></code>, <code><a href="#topic+run_genie3">run_genie3</a></code>,
<code><a href="#topic+run_glasso">run_glasso</a></code>, <code><a href="#topic+run_mrnet">run_mrnet</a></code>, and
<code><a href="#topic+run_pcor">run_pcor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)

# To create a short example, we subset on two pathways from the p53 pathway list,
# and will only run 1 permutation for significance testing.
pathway_list &lt;- p53_pathways[13]
n_perm &lt;- 1

# Use this method to perform differential network analysis.
# The parameters in run_silencer() can be adjusted using the ... argument.
# For example, the 'method' parameter can be specified as shown here.
results &lt;- dnapath(x = meso$gene_expression,
                   pathway_list = pathway_list,
                   group_labels = meso$groups,
                   n_perm = n_perm,
                   network_inference = run_silencer,
                   method = "spearman")
summary(results)

# The group-specific association matrices can be extracted using get_networks().
nw_list &lt;- get_networks(results) # Get networks for the pathway


# nw_list has length 2 and contains the inferred networks for the two groups.
# The gene names are the Entrezgene IDs from the original expression dataset.
# Renaming the genes in the dnapath results to rename those in the networks.
# NOTE: The temporary directory, tempdir(), is used in this example. In practice,
#       this argument can be removed or changed to an existing directory
results &lt;- rename_genes(results, to = "symbol", species = "human",
                        dir_save = tempdir())
nw_list &lt;- get_networks(results) # The genes (columns) will have new names.

# (Optional) Plot the network using SeqNet package (based on igraph plotting).
# First rename entrezgene IDs into gene symbols.
SeqNet::plot_network(nw_list[[1]])

</code></pre>

<hr>
<h2 id='sort.dnapath_list'>Sort function for 'dnapath_list' object.</h2><span id='topic+sort.dnapath_list'></span>

<h3>Description</h3>

<p>Sort function for 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
sort(x, decreasing = TRUE, by = "dc_score", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="sort.dnapath_list_+3A_decreasing">decreasing</code></td>
<td>
<p>Logical. If TRUE (the default), results are sorted in
decreasing order.</p>
</td></tr>
<tr><td><code id="sort.dnapath_list_+3A_by">by</code></td>
<td>
<p>The variable to sort the results by. Must be one of: &quot;mean_expr&quot;,
the mean expression of each pathway across both groups; &quot;mean_expr1&quot;
or &quot;mean_expr2&quot;, the mean expression of each pathway in group
1 or 2, respectively; &quot;dc_score&quot;, the differential connectivity score
of the pathway; &quot;p_value&quot;, the p-value of the dc score; &quot;n_genes&quot;,
the number of genes in each pathway; &quot;pathway&quot;, the pathway names; or
&quot;n_dc&quot; the number of significantly differentially conncted genes in each
pathway.</p>
</td></tr>
<tr><td><code id="sort.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The differential network analysis results ordered by DC pathway score.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
# Filter out pathways that have p-values above 0.2.
results_sig &lt;- filter_pathways(results, 0.2) 
sort(results_sig, by = "dc_score") # Sort by the pathway DC score.
sort(results_sig, by = "n_genes") # Sort by the pathway size.
sort(results_sig, by = "mean_expr") # Sort by the mean expression.
</code></pre>

<hr>
<h2 id='subset.dnapath_list'>Subset function for 'dnapath_list' object.</h2><span id='topic+subset.dnapath_list'></span>

<h3>Description</h3>

<p>Subset function for 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
subset(x, pathways = NULL, genes = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="subset.dnapath_list_+3A_pathways">pathways</code></td>
<td>
<p>A set of pathways to index on. This can be (1) a vector of
character strings, corresponding to
pathway names or regular expressions used to find pathways, (2) a vector of
indices to select pathways, (3) a vector of
negative indices indicating pathways to remove, or (4) a logical (boolean)
vector  that is the same length of current number of pathways in <code>x</code>.</p>
</td></tr>
<tr><td><code id="subset.dnapath_list_+3A_genes">genes</code></td>
<td>
<p>A set of gene names to index on; exact matching is used. Only
pathways containing these genes are retained.</p>
</td></tr>
<tr><td><code id="subset.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A subset of the differential network analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
# Obtain a pathway list for this short example:
pathway_list &lt;- get_reactome_pathways("human", overlap_limit = NULL,
                                      min_size = 13, max_size = 19)
# Run the differential network analysis.
results &lt;- dnapath(x = meso$gene_expression, pathway_list = pathway_list,
                   group_labels = meso$groups, n_perm = 5, seed = 0)
summary(results) # Summary over all pathways in the pathway list.

# Subset on pathways that contain "cell cycle" in its name.
cell_cycle_pathways &lt;- subset(results, pathways = "cell cycle")
summary(cell_cycle_pathways)
# Subset on pathways that contain the gene 1026 (Entrezgene ID).
pathways_with_1026 &lt;- subset(results, genes = "1026")
summary(pathways_with_1026)

# Multiple pathways and/or genes can also be specified. 
# Specifying both acts as an "OR" operation. For example, the following subset 
# will contain pathways containing the words "acetylation" or "methylation" 
# OR pathways that contain the genes "1108" or "11200". 
results_OR &lt;- subset(results, 
                     pathways = c("acetylation", "methylation"),
                     genes = c("1108", "11200"))
summary(results_OR)
# To subset on pathways that have both a specific pathway name AND
# certain genes, call the subset function twice: once specifying the
# `pathways` argument, then pass those results back into subset() with the
# `genes` argument specified. For example:
results_AND &lt;- subset(results, 
                      pathways = c("acetylation", "methylation"))
results_AND &lt;- subset(results_AND,
                      genes = c("1108", "11200"))
summary(results_AND)
</code></pre>

<hr>
<h2 id='summarize_edges'>Summarize differential connections for a pathway</h2><span id='topic+summarize_edges'></span>

<h3>Description</h3>

<p>Summarize differential connections for a pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_edges(x, alpha = 0.1, monotonized = FALSE, require_dc_genes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_edges_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="summarize_edges_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values of edge DC scores.
Defaults to 0.1 or the minimum possible threshold for the number
of permutations performed, whichever is greater.</p>
</td></tr>
<tr><td><code id="summarize_edges_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
<tr><td><code id="summarize_edges_+3A_require_dc_genes">require_dc_genes</code></td>
<td>
<p>If TRUE, the gene-level differential connectivity
p-value of the two genes for a given edge are also considered when deciding
whether an edge is differentially connected. If neither gene is significantly
differentially connected, then the edge between them will not be either.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble summarizing the differential connections in
the pathway.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_pathways">summarize_pathways</a></code>, <code><a href="#topic+summarize_genes">summarize_genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summarize_edges(results[[1]])
</code></pre>

<hr>
<h2 id='summarize_genes'>Summarize the differential connectivity of genes over all pathways.</h2><span id='topic+summarize_genes'></span>

<h3>Description</h3>

<p>Summarize the differential connectivity of genes over all pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_genes(x, alpha = 0.1, monotonized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_genes_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="summarize_genes_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values of gene DC scores. Used to determine
the number of pathways that each gene is differentially connected in.
Defaults to 0.1 or the minimum possible threshold for the number
of permutations performed, whichever is greater.</p>
</td></tr>
<tr><td><code id="summarize_genes_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble summarizing the differential connectivity of genes across
all pathways.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_pathways">summarize_pathways</a></code>, <code><a href="#topic+summarize_edges">summarize_edges</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summarize_genes(results) # Summary of genes across all pathways.
summarize_genes(results[[1]]) # Summary of genes within the first pathway.
</code></pre>

<hr>
<h2 id='summarize_genes_for_pathway'>Summarize differential connectivity of genes in a pathway</h2><span id='topic+summarize_genes_for_pathway'></span>

<h3>Description</h3>

<p>Summarize differential connectivity of genes in a pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_genes_for_pathway(x, alpha = NULL, monotonized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_genes_for_pathway_+3A_x">x</code></td>
<td>
<p>A 'dnapath' object of a single pathway.</p>
</td></tr>
<tr><td><code id="summarize_genes_for_pathway_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values of gene DC scores, used to
subset the x. If NULL (or 1), x for all genes are shown.</p>
</td></tr>
<tr><td><code id="summarize_genes_for_pathway_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble summarizing the differential connectivity of genes in
the pathway.
</p>

<hr>
<h2 id='summarize_pathways'>Summarize the differential connectivity of pathways.</h2><span id='topic+summarize_pathways'></span>

<h3>Description</h3>

<p>Summarize the differential connectivity of pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_pathways(x, alpha = 1, alpha_gene = 0.1, monotonized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_pathways_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="summarize_pathways_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values of pathway DC scores.
Defaults to 1, which leads to results for all pathways being shown.</p>
</td></tr>
<tr><td><code id="summarize_pathways_+3A_alpha_gene">alpha_gene</code></td>
<td>
<p>Threshold for p-values of gene DC scores. Used to determine
the number of genes that are differentially connected within each pathway.
Defaults to 0.1 or the minimum possible threshold for the number
of permutations performed, whichever is greater.</p>
</td></tr>
<tr><td><code id="summarize_pathways_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble summarizing the differential connectivity of genes in
the pathway.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_genes">summarize_genes</a></code>, <code><a href="#topic+summarize_edges">summarize_edges</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summarize_pathways(results)
</code></pre>

<hr>
<h2 id='summary.dnapath'>Summary function for 'dnapath' object.</h2><span id='topic+summary.dnapath'></span>

<h3>Description</h3>

<p>Summary function for 'dnapath' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath'
summary(object, by_gene = TRUE, alpha = 1, monotonized = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.dnapath_+3A_object">object</code></td>
<td>
<p>A 'dnapath' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="summary.dnapath_+3A_by_gene">by_gene</code></td>
<td>
<p>If TRUE, summarizes the differential network analysis by genes;
otherwise, summarizes by gene-gene interactions.</p>
</td></tr>
<tr><td><code id="summary.dnapath_+3A_alpha">alpha</code></td>
<td>
<p>Threshold for p-values to determine significance; defaults
to 1 and returns all results. If 'by_gene'
is FALSE, then 'alpha' is used to filter edges. If 'by_gene' is TRUE,
then 'alpha' is used to filter genes.</p>
</td></tr>
<tr><td><code id="summary.dnapath_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
<tr><td><code id="summary.dnapath_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Summarizes the differential network analysis result.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_genes">summarize_genes</a></code>, <code><a href="#topic+summarize_edges">summarize_edges</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summary(results[[1]]) # Summary of the first pathway in the results.
</code></pre>

<hr>
<h2 id='summary.dnapath_list'>Summary function for 'dnapath_list' object.</h2><span id='topic+summary.dnapath_list'></span>

<h3>Description</h3>

<p>Summary function for 'dnapath_list' object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
summary(
  object,
  by_gene = FALSE,
  alpha_pathway = 1,
  alpha_gene = 0.1,
  monotonized = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.dnapath_list_+3A_object">object</code></td>
<td>
<p>A 'dnapath_list' object from <code><a href="#topic+dnapath">dnapath</a></code>.</p>
</td></tr>
<tr><td><code id="summary.dnapath_list_+3A_by_gene">by_gene</code></td>
<td>
<p>If TRUE, summarizes the differential network analysis by genes
instead of by pathways.</p>
</td></tr>
<tr><td><code id="summary.dnapath_list_+3A_alpha_pathway">alpha_pathway</code></td>
<td>
<p>Threshold for p-values of pathway DC scores; used
to subset the results. If NULL (or 1), results for all pathways are shown.</p>
</td></tr>
<tr><td><code id="summary.dnapath_list_+3A_alpha_gene">alpha_gene</code></td>
<td>
<p>Threshold for p-values of gene DC scores. Used to determine
the number of genes that are differentially connected within each pathway.
Defaults to 0.1 or the minimum possible threshold for the number
of permutations performed, whichever is greater.</p>
</td></tr>
<tr><td><code id="summary.dnapath_list_+3A_monotonized">monotonized</code></td>
<td>
<p>If TRUE, monotonized p-values are used.</p>
</td></tr>
<tr><td><code id="summary.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional arguments are ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Summarizes the differential network analysis results.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_pathways">summarize_pathways</a></code>, <code><a href="#topic+summarize_genes">summarize_genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
summary(results) # Summary across all pathways in the analysis.
</code></pre>

<hr>
<h2 id='symbol_to_entrez'>Obtain entrezgene IDs for gene symbols</h2><span id='topic+symbol_to_entrez'></span>

<h3>Description</h3>

<p>Uses <code>biomaRt</code> (Durinck et al. 2009)
to map entrezgene IDs to gene symbols for a given species. The output of
this function can be used in <code><a href="#topic+rename_genes">rename_genes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbol_to_entrez(
  x,
  species,
  symbol_name = NULL,
  dir_save = tempdir(),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="symbol_to_entrez_+3A_x">x</code></td>
<td>
<p>A vector of gene symbols.</p>
</td></tr>
<tr><td><code id="symbol_to_entrez_+3A_species">species</code></td>
<td>
<p>The species used to obtain the entrezgene IDs. For example:
&quot;Homo sapiens&quot;, &quot;m musculus&quot;, &quot;C. elegans&quot;, or &quot;S cerevisiae&quot;.
&quot;Human&quot; and &quot;mouse&quot; can also be used and will be converted to the
correct species name.</p>
</td></tr>
<tr><td><code id="symbol_to_entrez_+3A_symbol_name">symbol_name</code></td>
<td>
<p>The type of gene symbol to use. If NULL, then &quot;hgnc_symbol&quot;
is used for HGNC symbols, unless <code>species</code> is &quot;mmusculus&quot;, in which case
&quot;mgi_symbol&quot; is used.</p>
</td></tr>
<tr><td><code id="symbol_to_entrez_+3A_dir_save">dir_save</code></td>
<td>
<p>The directory to store annotation reference. Future
calls to this function will use the stored annotations. This speeds up the
operation and allows for reproducibility in the event that the <code>biomaRt</code>
database is updated. Set to NULL to disable. By default, it uses a
temporary directory to store files during the R session.</p>
</td></tr>
<tr><td><code id="symbol_to_entrez_+3A_verbose">verbose</code></td>
<td>
<p>Set to FALSE to avoid messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If entrezgene IDs are used in a <code>dnapath_list</code> or <code>dnapath</code>
object, or a pathway list, then <code><a href="#topic+get_genes">get_genes</a></code> can be used to
extract them and used for the <code>x</code> argument here.
</p>


<h3>Value</h3>

<p>A data frame with two columns: the first contains the original
gene symbols, and the second contains a corresponding entrezgene ID. If a
gene symbol is not mapped to an entrezgene ID, the entrezgene ID is set to -1.
</p>


<h3>Note</h3>

<p>Internet connection is required to connect to biomaRt. If unavailable, the
default biomart and default species contained in the package is used, but
this may not match the desired species.
</p>
<p>It is assumed that <code>x</code> contains MGI symbols when the biomart species is
&quot;Mus musculus&quot; and HGNC symbols otherwise.
</p>


<h3>References</h3>

<p>Durinck S, Spellman PT, Birney E, Huber W (2009).
&ldquo;Mapping Identifiers for the Integration of Genomic Datasets with the R/Bioconductor Package biomaRt.&rdquo;
<em>Nature Protocols</em>, <b>4</b>, 1184&ndash;1191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+entrez_to_symbol">entrez_to_symbol</a></code>, <code><a href="#topic+get_genes">get_genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Convert a set of gene symbols to entrezgene IDs.
# Note that not all may have mapping (such as "MSX" in this example).
gene_mat &lt;- symbol_to_entrez(c("SOX2", "SEMA3E", "COL11A1", "UBB", "MSX"),
                             species = "human")

</code></pre>

<hr>
<h2 id='tail.dnapath_list'>Return the last part of the dnapath results.</h2><span id='topic+tail.dnapath_list'></span>

<h3>Description</h3>

<p>Return the last part of the dnapath results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dnapath_list'
tail(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tail.dnapath_list_+3A_x">x</code></td>
<td>
<p>A 'dnapath_list' object.</p>
</td></tr>
<tr><td><code id="tail.dnapath_list_+3A_...">...</code></td>
<td>
<p>Additional paramters are passed into
<code><a href="#topic+summary.dnapath_list">summary.dnapath_list</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the last five rows of the summary table of the
'dnapath_list' object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(meso)
data(p53_pathways)
set.seed(0)
results &lt;- dnapath(x = meso$gene_expression, pathway_list = p53_pathways,
                   group_labels = meso$groups, n_perm = 10)
tail(results)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
