<!DOCTYPE html><html><head><title>Help for package DNAseqtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DNAseqtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DNAseqtest-package'><p>Generating and Testing DNA Sequences</p></a></li>
<li><a href='#artomat'><p>Transforming <code class="reqn">4^K</code> Array to <code class="reqn">m \times K</code> Matrix</p></a></li>
<li><a href='#Distance'><p>Paralinear Distances</p></a></li>
<li><a href='#Fmatrix'><p>Joint Distribution for Two Matched Sequences</p></a></li>
<li><a href='#gn'><p>Joint Distribution for K Matched Sequences</p></a></li>
<li><a href='#gn2'><p>Joint Distribution for K Matched Sequences (2)</p></a></li>
<li><a href='#gn3sim'><p>Generating Random DNA Samples Using the Rambaut and Grassly Method</p></a></li>
<li><a href='#likelihood'><p>Negative Log Likelihood Ratio</p></a></li>
<li><a href='#Ntml'><p>Generating Samples from a Multinomial Distribution</p></a></li>
<li><a href='#Pt'><p>Transition Probability Function</p></a></li>
<li><a href='#simapp'><p>Generating Random DNA Samples Using Approximation Method</p></a></li>
<li><a href='#simemb'><p>Generating Random DNA Samples Using Embedded Markov Chain</p></a></li>
<li><a href='#Smatrix'><p>Symmetric Matrix S</p></a></li>
<li><a href='#TEST2'><p>Test for Symmetry of Matched DNA Sequences</p></a></li>
<li><a href='#TEST3'><p>Overall Test for Marginal Symmetry</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Generating and Testing DNA Sequences</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Faisal Ababneh, John Robinson, Lars S Jermiin and Hasinur Rahaman Khan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hasinur Rahaman Khan &lt;hasinurkhan@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates DNA sequences based on Markov model techniques for matched sequences. This can be generalized to several sequences. The sequences (taxa) are then arranged in an evolutionary tree (phylogenetic tree) depicting how taxa diverge from their common ancestors. This gives the tests and estimation methods for the parameters of different models. Standard phylogenetic methods assume stationarity, homogeneity and reversibility for the Markov processes, and  often impose further restrictions on the parameters.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-24 18:58:52 UTC; hasinur</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-24 23:32:09</td>
</tr>
</table>
<hr>
<h2 id='DNAseqtest-package'>Generating and Testing DNA Sequences</h2><span id='topic+DNAseqtest-package'></span><span id='topic+DNAseqtest'></span>

<h3>Description</h3>

<p>Generates DNA sequences based on Markov model techniques for matched sequences. This can be generalized to several sequences. The sequences (taxa) are then arranged in an evolutionary tree (phylogenetic tree) depicting how taxa diverge from their common ancestors. This gives the tests and estimation methods for the parameters of different models. Standard phylogenetic methods assume stationarity, homogeneity and reversibility for the Markov processes, and  often impose further restrictions on the parameters.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> DNAseqtest</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-03-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Faisal Ababneh, John Robinson, Lars S Jermiin and Hasinur Rahaman Khan
Maintainer: Hasinur Rahaman Khan &lt;hasinurkhan@gmail.com&gt;
</p>


<h3>References</h3>

<p>Lars Sommer Jermiin, Vivek Jayaswal, Faisal Ababneh, John Robinson (2008). Phylogenetic model evaluation. Bioinformatics, Volume 452 of the series Methods in Molecular Biology, 331-364.
</p>
<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>
<p>Faisal Ababneh, Lars S Jermiin, John Robinson (2006). Generation of the Exact Distribution and Simulation of Matched Nucleotide Sequences on a Phylogenetic Tree. Journal of mathematical modelling and algorithms, 5(3), 291-308.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To generate a 4^5 gene array
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3),4,2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47), 3,.1,.5,.8)
gn.sec&lt;-gn(theta, merge2)
gn.sec
</code></pre>

<hr>
<h2 id='artomat'>Transforming <code class="reqn">4^K</code> Array to <code class="reqn">m \times K</code> Matrix</h2><span id='topic+artomat'></span>

<h3>Description</h3>

<p>This function transfers any array to a matrix.</p>


<h3>Usage</h3>

<pre><code class='language-R'>artomat(fobs)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="artomat_+3A_fobs">fobs</code></td>
<td>
<p>a <code class="reqn">4^K</code> array, containing the observed divergent frequencies for K matched sequences</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function transfers any <code class="reqn">4^K</code> array containing the observed divergent frequencies of K matched sequences to an m x K matrix, where m is the sum of the frequencies in the <code class="reqn">4^K</code> observed divergence array.</p>


<h3>Value</h3>

<p>An m x K matrix</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn2, gn, Fmatrix</p>

<hr>
<h2 id='Distance'>Paralinear Distances</h2><span id='topic+Distance'></span>

<h3>Description</h3>

<p>This function calculates the paralinear distance between K matched DNA sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Distance(F4)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Distance_+3A_f4">F4</code></td>
<td>
<p>a <code class="reqn">4^K</code> array containing the joint distribution array <code class="reqn">F(t)</code> or the observed array N</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the paralinear distances between K matched DNA sequences, depending on the joint distribution array for these K sequences or on the observed divergence array N.</p>


<h3>Value</h3>

<p>A K x K symmetric matrix distances between the K sequences</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn2, gn, Fmatrix, Ntml</p>


<h3>Examples</h3>

<pre><code class='language-R'>merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47), 3,.1,.5,.8)
F1&lt;-gn(theta,merge2)
dn&lt;-Distance(F1)
dn
</code></pre>

<hr>
<h2 id='Fmatrix'>Joint Distribution for Two Matched Sequences</h2><span id='topic+Fmatrix'></span>

<h3>Description</h3>

<p>This function calculates the joint distribution function for two edge tree.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fmatrix(t1, t2, f0, Sx2, Sy2, Pix, Piy)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fmatrix_+3A_t1">t1</code></td>
<td>
<p>represents the length from the tree root to the first node</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_t2">t2</code></td>
<td>
<p>represents the length from the tree root to the second node</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_f0">f0</code></td>
<td>
<p>the initial distribution for the four nucleotides</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_sx2">Sx2</code></td>
<td>
<p>a 4 x 4 symmetric matrix related to the first edge</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_sy2">Sy2</code></td>
<td>
<p>a 4 x 4 symmetric matrix related to the second edge</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_pix">Pix</code></td>
<td>
<p>a diagonal matrix for the stationary distribution of the first edge</p>
</td></tr>
<tr><td><code id="Fmatrix_+3A_piy">Piy</code></td>
<td>
<p>a diagonal matrix for the stationary distribution of the second edge</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the joint distribution function for a two edge tree with different edge lengths, stationary distributions and differentS matrices.</p>


<h3>Value</h3>

<p>A 4 x 4 matrix containing the joint edges</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn, Smatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'>f0&lt;-c(.25,.25,.25,.25)
Pi1&lt;-diag(c(.2,.2,.2,.4))
Pi2&lt;-diag(c(.1,.1,.1,.7))
S1&lt;-Smatrix(c(.2,.2,.2,.2,.2,.2),diag(Pi1))
S2&lt;-Smatrix(c(.3,.3,.3,.3,.3,.3),diag(Pi2))
fm&lt;-Fmatrix(1, .5, f0, S1, S2, Pi1, Pi2)
fm
</code></pre>

<hr>
<h2 id='gn'>Joint Distribution for K Matched Sequences</h2><span id='topic+gn'></span>

<h3>Description</h3>

<p>This function calculates the joint distribution array for K matched sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gn(theta, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gn_+3A_theta">theta</code></td>
<td>
<p>a vector of variables containing the following parameters in this order&ndash;1. the first three parameters from <code class="reqn">\pi_X</code> vector, 2. the first three parameters from <code class="reqn">\pi_Y</code> vector, 3. the first three parameters from <code class="reqn">f_0</code> vector, 4. the six off-diagonal free parameters in the S matrix, 5. a scalar <code class="reqn">\rho</code>, 6. a vector of lengths containing K-2 values</p>
</td></tr>
<tr><td><code id="gn_+3A_merge2">merge2</code></td>
<td>
<p>(K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the joint distribution array for a tree with K matched sequences. it uses the following functions&ndash; Pt, Fmatrix and Smatrix.</p>


<h3>Value</h3>

<p>A <code class="reqn">4^K</code> array containing the joint distribution for the K edges</p>


<h3>References</h3>

<p>Lars Sommer Jermiin, Vivek Jayaswal, Faisal Ababneh, John Robinson (2008). Phylogenetic model evaluation. Bioinformatics, Volume 452 of the series Methods in Molecular Biology, 331-364.
</p>
<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Fmatrix, Pt, Smatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To generate a 4^5 gene array
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3),4,2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47), 3,.1,.5,.8)
gn.sec&lt;-gn(theta, merge2)
gn.sec
</code></pre>

<hr>
<h2 id='gn2'>Joint Distribution for K Matched Sequences (2)</h2><span id='topic+gn2'></span>

<h3>Description</h3>

<p>This function calculates the joint distribution array for K matched sequences (second option).</p>


<h3>Usage</h3>

<pre><code class='language-R'>gn2(theta, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gn2_+3A_theta">theta</code></td>
<td>
<p>a vector of variables containing the following parameters in this order&ndash;1. the first three parameters from <code class="reqn">\pi_X</code> vector, 2. the first three parameters from <code class="reqn">\pi_Y</code> vector, 3. the first three parameters from <code class="reqn">f_0</code> vector, 4. the six off-diagonal free parameters in the S matrix, 5. a scalar <code class="reqn">\rho</code>, 6. a vector of lengths containing K-2 values</p>
</td></tr>
<tr><td><code id="gn2_+3A_merge2">merge2</code></td>
<td>
<p>(K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the joint distribution array for a tree with K matched sequences. it uses the following functions&ndash; Pt, Fmatrix and Smatrix.</p>


<h3>Value</h3>

<p>A <code class="reqn">4^K</code> array containing the joint distribution for the K edges</p>


<h3>References</h3>

<p>Lars Sommer Jermiin, Vivek Jayaswal, Faisal Ababneh, John Robinson (2008). Phylogenetic model evaluation. Bioinformatics, Volume 452 of the series Methods in Molecular Biology, 331-364.
</p>
<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Fmatrix, Pt, Smatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'>#To generate a 4^5 gene array
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
rho2&lt;-matrix(c(.3,.5,.3,.2,.3,.5,.8,2.7),4,2)
theta&lt;-c(rep(.25,3), rep(.25,3),rep(.25,3), c(.2,.35,.79,.01,.93,.47),rho2)
gn2&lt;-gn2(theta, merge2)
gn2
</code></pre>

<hr>
<h2 id='gn3sim'>Generating Random DNA Samples Using the Rambaut and Grassly Method</h2><span id='topic+gn3sim'></span>

<h3>Description</h3>

<p>This function generates random DNA samples using Rambaut and Grassly method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gn3sim(theta, seqLength, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gn3sim_+3A_theta">theta</code></td>
<td>
<p>a vector of variables containing the following parameters in this order&ndash;1. the first three parameters from <code class="reqn">\pi_X</code> vector, 2. the first three parameters from <code class="reqn">\pi_Y</code> vector, 3. the first three parameters from <code class="reqn">f_0</code> vector, 4. the six off-diagonal free parameters in the S matrix, 5. a scalar <code class="reqn">\rho</code>, 6. a vector of lengths containing K-2 values</p>
</td></tr>
<tr><td><code id="gn3sim_+3A_seqlength">seqLength</code></td>
<td>
<p>the length of sequences we need to generate</p>
</td></tr>
<tr><td><code id="gn3sim_+3A_merge2">merge2</code></td>
<td>
<p>(K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a <code class="reqn">4^K</code> DNA array using Rambaut and Grassly,
(1997) method. It depends on a set of variables theta, the sequence
length and a merge matrix describing the tree topology.</p>


<h3>Value</h3>

<p>A n x K observed divergence matrix</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Ntml, simapp, simemb, gn, gn2, Fmatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'># This will give 4^5 observed divergence array
theta&lt;-(c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),
3,.1,.5,.8))
n&lt;-1000
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
gn3&lt;-gn3sim(theta, n, merge2)
gn3
</code></pre>

<hr>
<h2 id='likelihood'>Negative Log Likelihood Ratio</h2><span id='topic+likelihood'></span>

<h3>Description</h3>

<p>This function calculates log likelihood ratio value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>likelihood(thetast, fobs, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="likelihood_+3A_thetast">thetast</code></td>
<td>
<p>a starting values for the parameter we need to estimate</p>
</td></tr>
<tr><td><code id="likelihood_+3A_fobs">fobs</code></td>
<td>
<p>the <code class="reqn">4^K</code> joint distribution array for K edge tree</p>
</td></tr>
<tr><td><code id="likelihood_+3A_merge2">merge2</code></td>
<td>
<p>a (K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the log likelihood ratio value for F(t). It needs
a vector of starting values for the parameters estimate, <code class="reqn">4^K</code> observed divergence array and merge matrix describing the tree topology.</p>


<h3>Value</h3>

<p>The value of the log likelihood ratio</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>gn, gn2</p>


<h3>Examples</h3>

<pre><code class='language-R'>merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),3,.1,.5,.8)
F1&lt;-gn(theta, merge2)
lh&lt;-likelihood(theta, F1, merge2)
lh
</code></pre>

<hr>
<h2 id='Ntml'>Generating Samples from a Multinomial Distribution</h2><span id='topic+Ntml'></span>

<h3>Description</h3>

<p>Generating random DNA samples from a multinomial distribution.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ntml(N, Ft)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ntml_+3A_n">N</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="Ntml_+3A_ft">Ft</code></td>
<td>
<p>a <code class="reqn">4^K</code> array, containing the joint distribution probabilities for K matched sequences.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a <code class="reqn">4^K</code> DNA array from a multinomial distribution. It depends on the sample size we need to generate and the <code class="reqn">4^K</code> joint distribution array of K matched sequences.</p>


<h3>Value</h3>

<p>A <code class="reqn">4^K</code> observed divergence array</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>simemb, simapp, gn3sim, gn, gn2, Fmatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'>#This will give a 4^K observed divergence array
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),
3,.1,.5,.8)
F1&lt;-gn(theta,merge2)
Nt&lt;-Ntml(1000, F1)
Nt
</code></pre>

<hr>
<h2 id='Pt'>Transition Probability Function</h2><span id='topic+Pt'></span>

<h3>Description</h3>

<p>This function calculates the transition probability function for a process
during a period of time.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pt(S, Pi, t)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Pt_+3A_s">S</code></td>
<td>
<p>a 4 x 4 symmetric matrix</p>
</td></tr>
<tr><td><code id="Pt_+3A_pi">Pi</code></td>
<td>
<p>a diagonal matrix containing the stationary distribution for the
process</p>
</td></tr>
<tr><td><code id="Pt_+3A_t">t</code></td>
<td>
<p>a period of time describing the length of the process</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function needs the 4 x 4 symmetric matrix S, II and the process
length t in order to find the transition probability over that process,
where <code class="reqn">P_{ij}(t)</code> is the probability that the ith nucleotide changes to the
j-th nucleotide during the period of t.</p>


<h3>Value</h3>

<p>A 4 x 4 matrix containing the transition probabilities for a process.</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Smatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'>Pi&lt;-diag(c(.1,.1,.1,.7)) 
S&lt;-Smatrix(c(.3,.3,.3,.3,.3,.3),diag(Pi))
t&lt;-1
p&lt;-Pt(S, Pi, t)
p
</code></pre>

<hr>
<h2 id='simapp'>Generating Random DNA Samples Using Approximation Method</h2><span id='topic+simapp'></span>

<h3>Description</h3>

<p>This function generates random DNA samples using an approximation method</p>


<h3>Usage</h3>

<pre><code class='language-R'>simapp(theta, seqLength, merge1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simapp_+3A_theta">theta</code></td>
<td>
<p>a vector of variables containing the following parameters in this order&ndash;1. the first three parameters from <code class="reqn">\pi_X</code> vector, 2. the first three parameters from <code class="reqn">\pi_Y</code> vector, 3. the first three parameters from <code class="reqn">f_0</code> vector, 4. the six off-diagonal free parameters in the S matrix, 5. a scalar <code class="reqn">\rho</code>, 6. a vector of lengths containing K-2 values</p>
</td></tr>
<tr><td><code id="simapp_+3A_seqlength">seqLength</code></td>
<td>
<p>the length of sequences we need to generate</p>
</td></tr>
<tr><td><code id="simapp_+3A_merge1">merge1</code></td>
<td>
<p>(K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a <code class="reqn">4^K</code> DNA array using an approximation method. It depends on a set of variables theta, the sequence length and a merge matrix describing the tree topology.</p>


<h3>Value</h3>

<p>A n x K observed divergence matrix</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Ntml, simemb, gn3sim, gn, gn2, Fmatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'># This will give 4^5 observed divergence array
theta&lt;-(c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.2,.2,.2,.2,.2),
3,.1,.5,.8))
n&lt;-1000
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
sa&lt;-simapp(theta, n, merge2)
sa
</code></pre>

<hr>
<h2 id='simemb'>Generating Random DNA Samples Using Embedded Markov Chain</h2><span id='topic+simemb'></span>

<h3>Description</h3>

<p>This function generates random DNA samples using embedded chain.</p>


<h3>Usage</h3>

<pre><code class='language-R'>simemb(theta, seqLength, merge2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simemb_+3A_theta">theta</code></td>
<td>
<p>a vector of variables containing the following parameters in this order&ndash;1. the first three parameters from <code class="reqn">\pi_X</code> vector, 2. the first three parameters from <code class="reqn">\pi_Y</code> vector, 3. the first three parameters from <code class="reqn">f_0</code> vector, 4. the six off-diagonal free parameters in the S matrix, 5. a scalar <code class="reqn">\rho</code>, 6. a vector of lengths containing K-2 values</p>
</td></tr>
<tr><td><code id="simemb_+3A_seqlength">seqLength</code></td>
<td>
<p>the length of sequences we need to generate</p>
</td></tr>
<tr><td><code id="simemb_+3A_merge2">merge2</code></td>
<td>
<p>(K-1) x 2 matrix describing the tree topology</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates <code class="reqn">4^K</code> DNA array using embedded Markov chain.
It depends on a set of variables theta, the sequence length and a merge
matrix describing the tree topology.</p>


<h3>Value</h3>

<p>A n x K observed divergence matrix</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Ntml, simapp, gn3sim, gn, gn2, Fmatrix</p>


<h3>Examples</h3>

<pre><code class='language-R'># This will give 4^5 observed divergence array
theta&lt;-(c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),
3,.1,.5,.8))
n&lt;-1000
merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
sm&lt;-simemb(theta, n, merge2)
sm
</code></pre>

<hr>
<h2 id='Smatrix'>Symmetric Matrix S</h2><span id='topic+Smatrix'></span>

<h3>Description</h3>

<p>This function calculates the symmetric matrix S.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Smatrix(s, pix)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Smatrix_+3A_s">s</code></td>
<td>
<p>a vector of variables containing the six free parameters in the S
matrix</p>
</td></tr>
<tr><td><code id="Smatrix_+3A_pix">pix</code></td>
<td>
<p>a vector giving the stationary probabilities for the four nucleotides
A, C, G and T</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the matrix S, which we used to calculate the
rate matrix R.</p>


<h3>Value</h3>

<p>A 4 x 4 symmetric matrix</p>


<h3>References</h3>

<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Pt, Fmatrix, gn ,gn2</p>


<h3>Examples</h3>

<pre><code class='language-R'>s&lt;-c(.1,.2,.3,.4,.5,.6)
pi&lt;-c(.1,.1,.1,.7)
sm&lt;-Smatrix(s, pi)
sm
</code></pre>

<hr>
<h2 id='TEST2'>Test for Symmetry of Matched DNA Sequences</h2><span id='topic+TEST2'></span>

<h3>Description</h3>

<p>This function tests for symmetry between all the pairs of K matched
DNA sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>TEST2(f)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TEST2_+3A_f">f</code></td>
<td>
<p>a <code class="reqn">4^K</code> array containing the observed divergence array N</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates Bowker's test for symmetry, Stuart's test for
marginal symmetry and the test for internal symmetry. It depends on the <code class="reqn">4^K</code> observed divergence array N.
</p>


<h3>Value</h3>

<p>A list of three lower triangle matrices
</p>
<table>
<tr><td><code>first</code></td>
<td>
<p>the lower triangle of the matrix contains (K-1) x (K-1) values
shows Bowker's test between all the possible pairs of the K sequences</p>
</td></tr>
<tr><td><code>second</code></td>
<td>
<p>the lower triangle of the matrix contains (K-1) x (K-1) values
shows Stuart's test between all the possible pairs of the K sequences</p>
</td></tr>
<tr><td><code>third</code></td>
<td>
<p>the lower triangle of the matrix contains (K-1) x (K-1) values
shows the internal test between all the possible pairs of the K sequences</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lars Sommer Jermiin, Vivek Jayaswal, Faisal Ababneh, John Robinson (2008). Phylogenetic model evaluation. Bioinformatics, Volume 452 of the series Methods in Molecular Biology, 331-364.
</p>
<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Ntml, simapp, simemb, TEST3</p>


<h3>Examples</h3>

<pre><code class='language-R'>merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),
3,.1,.5,.8)
F1&lt;-gn(theta,merge2)
N1&lt;-Ntml(1000,F1)
t2&lt;-TEST2(N1)
t2
</code></pre>

<hr>
<h2 id='TEST3'>Overall Test for Marginal Symmetry</h2><span id='topic+TEST3'></span>

<h3>Description</h3>

<p>This function tests for symmetry between K matched DNA sequences.</p>


<h3>Usage</h3>

<pre><code class='language-R'>TEST3(Farray)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TEST3_+3A_farray">Farray</code></td>
<td>
<p>a <code class="reqn">4^K</code> array containing the observed divergence array N</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates overall test for marginal symmetry. It depends
on the <code class="reqn">4^K</code> observed divergence array N.</p>


<h3>Value</h3>

<p>A single value gives the overall test for marginal symmetry between K
matched sequences</p>


<h3>References</h3>

<p>Lars Sommer Jermiin, Vivek Jayaswal, Faisal Ababneh, John Robinson (2008). Phylogenetic model evaluation. Bioinformatics, Volume 452 of the series Methods in Molecular Biology, 331-364.
</p>
<p>Faisal Ababneh, Lars S Jermiin, Chunsheng Ma, John Robinson (2006). Matched-pairs tests of homogeneity with applications to homologous nucleotide sequences. Bioinformatics, 22(10), 1225-1231.
</p>


<h3>See Also</h3>

<p>Ntml, simapp, simemb, TEST2</p>


<h3>Examples</h3>

<pre><code class='language-R'>merge2&lt;-matrix(c(-1,-4,-3,2,-2,-5,1,3), 4, 2)
theta&lt;-c(rep(.25,3), rep(.25,3), rep(.25,3), c(.2,.35,.79,.01,.93,.47),
3,.1,.5,.8)
F1&lt;-gn(theta,merge2)
N1&lt;-Ntml(1000,F1)
t3&lt;-TEST3(N1)
t3
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
