<!DOCTYPE html><html lang="en"><head><title>Help for package DNMF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DNMF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DNMF'><p>Discriminant Non-Negative Matrix Factorization.</p></a></li>
<li><a href='#ndNMF'><p>a new discriminant Non-Negative Matrix Factorization (dNMF)</p></a></li>
<li><a href='#NMFpval'><p>P value for discriminant Non-Negative Matrix Factorization</p></a></li>
<li><a href='#rnk'><p>write rnk to a file from matrix W.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-09</td>
</tr>
<tr>
<td>Title:</td>
<td>Discriminant Non-Negative Matrix Factorization</td>
</tr>
<tr>
<td>Description:</td>
<td>Discriminant Non-Negative Matrix Factorization aims to extend the Non-negative Matrix Factorization algorithm in order to extract features that enforce not only the spatial locality, but also the separability between classes in a discriminant manner. It refers to three article, Zafeiriou, Stefanos, et al. "Exploiting discriminant information in nonnegative matrix factorization with application to frontal face verification." Neural Networks, IEEE Transactions on 17.3 (2006): 683-695. Kim, Bo-Kyeong, and Soo-Young Lee. "Spectral Feature Extraction Using dNMF for Emotion Recognition in Vowel Sounds." Neural Information Processing. Springer Berlin Heidelberg, 2013. and Lee, Soo-Young, Hyun-Ah Song, and Shun-ichi Amari. "A new discriminant NMF algorithm and its application to the extraction of subtle emotional differences in speech." Cognitive neurodynamics 6.6 (2012): 525-535.</td>
</tr>
<tr>
<td>Depends:</td>
<td>foreach</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, gplots, parallel, doParallel</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/zhilongjia/DNMF">https://github.com/zhilongjia/DNMF</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/zhilongjia/DNMF/issues">https://github.com/zhilongjia/DNMF/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-10 08:51:46 UTC; zjia</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhilong Jia [aut, cre],
  Xiang Zhang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhilong Jia &lt;zhilongjia@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-10 09:10:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='DNMF'>Discriminant Non-Negative Matrix Factorization.</h2><span id='topic+DNMF'></span>

<h3>Description</h3>

<p>Discriminant Non-Negative Matrix Factorization, DNMF, is to extend the Non-negative Matrix Factorization algorithm in 
order to extract features that enforce not only the spatial locality, but
also the separability between classes in a discriminant manner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNMF(
  data,
  trainlabel,
  r = 2,
  gamma = 0.1,
  delta = 1e-04,
  maxIter = 1000,
  tol = 1e-07,
  log = TRUE,
  plotit = FALSE,
  checkH = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DNMF_+3A_data">data</code></td>
<td>
<p>a matrix, like expression profilings of some samples. the columns are samples and the rows are gene's expression.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_trainlabel">trainlabel</code></td>
<td>
<p>a numeric vector of sample type of all the samples, this vector should ONLY contain 1 and 2 so far and length of it should equal the column (sample) size of data.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_r">r</code></td>
<td>
<p>the dimension of expected reduction dimension, with the default value 2.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_gamma">gamma</code></td>
<td>
<p>the tradeoff value for the within scatter matrix, with the default value 0.1.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_delta">delta</code></td>
<td>
<p>the tradeoff value for the between scatter matrix, with the default value 1e-4.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_maxiter">maxIter</code></td>
<td>
<p>the maximum iteration of update rules, with the default value 1000.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_tol">tol</code></td>
<td>
<p>the toleration of coverange, with the default value 1e-7.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_log">log</code></td>
<td>
<p>log2 data. Default is TRUE.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_plotit">plotit</code></td>
<td>
<p>whether plot H (V=WH). Default: FALSE.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_checkh">checkH</code></td>
<td>
<p>whether or not check H. Default: TRUE. This parameter aims to 
check whether or not the H safisfy the discriminant metagenes. Usually, this
should be TRUE.</p>
</td></tr>
<tr><td><code id="DNMF_+3A_...">...</code></td>
<td>
<p>to gplots::heatmap.2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The main algorithm is based on 
<a href="https://pubmed.ncbi.nlm.nih.gov/16722172/">Zafeiriou, S., et al. 
(2006) Exploiting discriminant information in 
nonnegative matrix factorization with application to frontal face 
verification, IEEE transactions on neural networks, 17, 683-695</a>, 
with some <strong>CORRECTIONs</strong>.
</p>


<h3>Author(s)</h3>

<p>Zhilong Jia and Xiang Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- rbind(matrix(c(rep(3, 16), rep(8, 24)), ncol=5), 
matrix(c(rep(5, 16), rep(5, 24)), ncol=5), 
matrix(c(rep(18, 16), rep(7, 24)), ncol=5)) + 
matrix(runif(120,-1,1), ncol=5)
trainlabel &lt;- c(1,1,2,2,2)

DNMF_result &lt;- DNMF(dat, trainlabel, r=2)


## Not run: 
# Gene ranking. dat is the raw read count maatrix with sample in column.

#normalising dat
Sizefactors &lt;- DESeq::estimateSizeFactorsForMatrix(dat)
dat = sweep(dat, 2, Sizefactors, `/`)

res &lt;- DNMF(dat, trainlabel, r=2)
rnk &lt;- res$rnk

#The end of gene ranking exmaples

#Other exmaples
DNMF_result &lt;- DNMF(dat, trainlabel, r=2, gamma=0.1, delta=0.0001, plotit=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='ndNMF'>a new discriminant Non-Negative Matrix Factorization (dNMF)</h2><span id='topic+ndNMF'></span>

<h3>Description</h3>

<p>The ndNMF algorithm with the additional Fisher criterion on the cost 
function of conventional NMF was designed to increase class-related
discriminating power.
</p>
<p>This algorithm is based on articles.
</p>

<ol>
<li><p> Kim, Bo-Kyeong, and Soo-Young Lee. &quot;Spectral Feature Extraction Using dNMF for Emotion Recognition in Vowel Sounds.&quot; Neural Information Processing. Springer Berlin Heidelberg, 2013.
</p>
</li>
<li><p> Lee, Soo-Young, Hyun-Ah Song, and Shun-ichi Amari. &quot;A new discriminant NMF algorithm and its application to the extraction of subtle emotional differences in speech.&quot; Cognitive neurodynamics 6.6 (2012): 525-535.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>ndNMF(
  dat,
  trainlabel,
  r = 2,
  lambada = 0.1,
  maxIter = 1000,
  tol = 1e-07,
  log = TRUE,
  plotit = FALSE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ndNMF_+3A_dat">dat</code></td>
<td>
<p>a matrix with gene in row and sample in column</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_trainlabel">trainlabel</code></td>
<td>
<p>the label of sample, like c(1,1,2,2,2)</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_r">r</code></td>
<td>
<p>the dimension of expected reduction dimension, with the default value 2</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_lambada">lambada</code></td>
<td>
<p>a relative weighting factor for the discriminant. Default 0.1</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_maxiter">maxIter</code></td>
<td>
<p>the maximum iteration of update rules, with the default value 1000</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_tol">tol</code></td>
<td>
<p>the toleration of coverange, with the default value 1e-7</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_log">log</code></td>
<td>
<p>log2 data. Default is TRUE.</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_plotit">plotit</code></td>
<td>
<p>whether plot H (V=WH). Default: FALSE.</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_verbose">verbose</code></td>
<td>
<p>TRUE</p>
</td></tr>
<tr><td><code id="ndNMF_+3A_...">...</code></td>
<td>
<p>to gplots::heatmap.2</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhilong Jia and Xiang Zhang
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- rbind(matrix(c(rep(3, 16), rep(8, 24)), ncol=5), 
matrix(c(rep(5, 16), rep(5, 24)), ncol=5), 
matrix(c(rep(18, 16), rep(7, 24)), ncol=5)) + 
matrix(runif(120,-1,1), ncol=5)
trainlabel &lt;- c(1,1,2,2,2)

res &lt;- ndNMF(dat, trainlabel, r=2, lambada = 0.1)
res$H
res$rnk

</code></pre>

<hr>
<h2 id='NMFpval'>P value for discriminant Non-Negative Matrix Factorization</h2><span id='topic+NMFpval'></span>

<h3>Description</h3>

<p>Estimate the significance of differentially expressed genes in parallel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NMFpval(
  nmf_res,
  np = 100,
  ncores = parallel::detectCores(),
  fdr = FALSE,
  top = 1000,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NMFpval_+3A_nmf_res">nmf_res</code></td>
<td>
<p>result from DNMF or dNMF</p>
</td></tr>
<tr><td><code id="NMFpval_+3A_np">np</code></td>
<td>
<p>number of permutations</p>
</td></tr>
<tr><td><code id="NMFpval_+3A_ncores">ncores</code></td>
<td>
<p>cores used. Default is all the availiable cores</p>
</td></tr>
<tr><td><code id="NMFpval_+3A_fdr">fdr</code></td>
<td>
<p>false discovery rate. Default is FALSE</p>
</td></tr>
<tr><td><code id="NMFpval_+3A_top">top</code></td>
<td>
<p>only include top ranked genes. Default is 1000</p>
</td></tr>
<tr><td><code id="NMFpval_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Details</h3>

<p>P value is caculated based on aatricle, Wang, Hong-Qiang, Chun-Hou Zheng, and Xing-Ming Zhao. 
&quot;jNMFMA: a joint non-negative matrix factorization meta-analysis of
transcriptomics data.&quot; Bioinformatics (2014): btu679.
</p>


<h3>Value</h3>

<p>a matrix with columns rnk, p (and fdr)
</p>


<h3>Author(s)</h3>

<p>Zhilong Jia
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- rbind(matrix(c(rep(3, 16), rep(8, 24)), ncol=5), 
matrix(c(rep(5, 16), rep(5, 24)), ncol=5), 
matrix(c(rep(18, 16), rep(7, 24)), ncol=5)) + 
matrix(runif(120,-1,1), ncol=5)
trainlabel &lt;- c(1,1,2,2,2)

nmf_res &lt;- ndNMF(dat, trainlabel, r=2, lambada = 0.1)
pMat &lt;- NMFpval(nmf_res, np=10, ncores=2, top=4)

</code></pre>

<hr>
<h2 id='rnk'>write rnk to a file from matrix W.</h2><span id='topic+rnk'></span>

<h3>Description</h3>

<p>write a rnk file from matrix W in a returned object of function <code>DNMF</code>.
The rnk format is referred <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats">RNK</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rnk(object, fn = "./tmp.rnk", type = "o2m")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rnk_+3A_object">object</code></td>
<td>
<p>a returned object of function <code>DNMF</code></p>
</td></tr>
<tr><td><code id="rnk_+3A_fn">fn</code></td>
<td>
<p>the output filename. Default is &quot;./tmp.rnk&quot;</p>
</td></tr>
<tr><td><code id="rnk_+3A_type">type</code></td>
<td>
<p>type o2m (Default) or o2o. to compare with multi sample labels. 
o2m means one Vs others, while o2o means one Vs another one.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
rnk(dnmf_result, fn="tmp.rnk")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
