<!DOCTYPE html><html lang="en"><head><title>Help for package DoEstRare</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DoEstRare}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DoEstRare'><p>DoEstRare</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Rare Variant Association Test Based on Position Density
Estimation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-07-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Elodie Persyn</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Elodie Persyn &lt;elodie.persyn@univ-nantes.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15)</td>
</tr>
<tr>
<td>Description:</td>
<td>Rare variant association test integrating variant position information. It aims to identify the presence of clusters of disease-risk variants in specific gene regions. For more details, please read the publication from Persyn et al. (2017)  &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0179364">doi:10.1371/journal.pone.0179364</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-01 13:00:25 UTC; E14E934Z</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-01 13:24:30 UTC</td>
</tr>
</table>
<hr>
<h2 id='DoEstRare'>DoEstRare</h2><span id='topic+DoEstRare'></span>

<h3>Description</h3>

<p>Rare variant association test comparing position density functions and mutation counts between cases and controls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DoEstRare(pheno, geno, position, genome.size,
          perm=NULL, alpha=NULL, c=NULL,
          autosomal=TRUE, gender=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DoEstRare_+3A_pheno">pheno</code></td>
<td>
<p>a numeric vector of phenotypes. Affected individuals are coded 1 and unaffected individuals are coded 0.</p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_geno">geno</code></td>
<td>
<p>a numeric matrix of genotypes (row: individual, column: variant). Genotypes are coded 0,1 or 2 corresponding to the number of minor alleles. </p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_position">position</code></td>
<td>
<p>a numeric vector of variant positions. </p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_genome.size">genome.size</code></td>
<td>
<p>a numeric value corresponding to the length of the analyzed region. </p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_perm">perm</code></td>
<td>
<p>number of permutations. If not NULL,a &quot;standard permutation procedure&quot; is performed to compute the significance. See Details. </p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_alpha">alpha</code></td>
<td>
<p>error level. If not NULL,an &quot;adaptive permutation procedure&quot; is performed to compute the significance. See Details.</p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_c">c</code></td>
<td>
<p>precision of the p-value. If not NULL,an &quot;adaptive permutation procedure&quot; is performed to compute the significance. See Details.</p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_autosomal">autosomal</code></td>
<td>
<p>boolean. If TRUE, autosomal chromosome; FALSE, X chromosome.</p>
</td></tr>
<tr><td><code id="DoEstRare_+3A_gender">gender</code></td>
<td>
<p>numeric vector. 1=male; 2=female.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two types of permutations procedures can be defined in the function: the standard permutation procedure and the adaptive permutation procedure.
</p>
<p>In the standard permutation procedure, the user specifies, in the argument &quot;perm&quot;, the number of permutations to be done. The p-value will be <code class="reqn">(R+1)(B+1)</code>. With <code class="reqn">R</code> the number of permutation statistics superior to the observed statistic and <code class="reqn">B</code> the number of permutations.
</p>
<p>In the adaptive permutation procedure, the user specifies, in the argument &quot;alpha&quot;, the significance to achieve after multiple testing correction. In the argument &quot;c&quot;, the estimation precision of the p-value. In function of these two paremeters, the maximal number of permutations and the maximal number of success to achieve will be computed. If the maximal number of success is reached, the p-value will be <code class="reqn">R/B</code>. If not, the maximal number of permutations will be used to compute the p-value <code class="reqn">(R+1)(B+1)</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>
<p>the p-value obtained by the phenotype permutation procedure. </p>
</td></tr>
<tr><td><code>stat</code></td>
<td>
<p>the test statistic. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Elodie Persyn, <a href="mailto:elodie.persyn@univ-nantes.fr">elodie.persyn@univ-nantes.fr</a></p>


<h3>References</h3>

<p>Persyn E, Karakachoff M, Le Scouarnec S, Le Cl??zio C, Campion D, French Exome Consortium, et al. DoEstRare: A statistical test to identify local enrichments in rare genomic variants associated with disease. Wang K, editor. PLOS ONE. 2017 Jul 24;12(7):e0179364.
</p>
<p>Che R, Jack JR, Motsinger-Reif AA, Brown CC. An adaptive permutation approach for genome-wide association study: evaluation and recommendations for use. BioData Min. 2014;7:9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pheno=rep(c(0,1), 500)
geno=matrix(sample(c(0,1),prob=c(0.7,0.3) ,1000*30, replace=TRUE), ncol=30)
position=sample(1:500,30)
genome.size=500
perm=200

#Autosomal gene
#standard phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, perm)
#adaptive phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, alpha=0.05, c=0.2)

#X gene
gender=rep(c(1,2), each=500)
#standard phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, perm, autosomal=FALSE, gender=gender)
#adaptive phenotype permutation procedure
DoEstRare(pheno, geno, position, genome.size, alpha=0.05, c=0.2, autosomal=FALSE, gender=gender)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
