<!DOCTYPE html><html><head><title>Help for package dotgen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dotgen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cst'><p>Correlation among association test statistics</p></a></li>
<li><a href='#dot'><p>Decorrelation by Orthogonal Transformation (DOT)</p></a></li>
<li><a href='#dot_sst'><p>Methods for combining decorrelated summary statistics</p></a></li>
<li><a href='#imp'><p>Impute missing genotypes</p></a></li>
<li><a href='#zsc'><p>Calculate Z-scores from P-values and estimated effects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene-Set Analysis via Decorrelation by Orthogonal Transformation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-03</td>
</tr>
<tr>
<td>Description:</td>
<td>Decorrelates a set of summary statistics (i.e., Z-scores or P-values per SNP) via Decorrelation by Orthogonal Transformation (DOT) approach and performs gene-set analyses by combining transformed statistic values; operations are performed with algorithms that rely only on the association summary results and the linkage disequilibrium (LD). For more details on DOT and its power, see Olga (2020) &lt;<a href="https://doi.org/10.1371%2Fjournal.pcbi.1007819">doi:10.1371/journal.pcbi.1007819</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), mvtnorm</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/xiaoran831213/dotgen">https://github.com/xiaoran831213/dotgen</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Olga Vsevolozhskaya
    <a href="https://orcid.org/0000-0001-9376-3645"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Dmitri Zaykin <a href="https://orcid.org/0000-0002-9444-2859"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Xiaoran Tong <a href="https://orcid.org/0000-0002-4648-3330"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaoran Tong &lt;tongxia1@msu.edu&gt;</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-11 05:21:17 UTC; xt</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-16 09:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cst'>Correlation among association test statistics</h2><span id='topic+cst'></span>

<h3>Description</h3>

<p>Calculates the correlation among genetic association test statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cst(g, x = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cst_+3A_g">g</code></td>
<td>
<p>matrix of  genotype, one row per  sample, one column  per variant,
missing values allowed.</p>
</td></tr>
<tr><td><code id="cst_+3A_x">x</code></td>
<td>
<p>matrix of covariates, one row per sample, no missing values allowed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When no covariates are present in per-variant association analyses, that is,
<code>x==NULL</code>, correlation among test statistics  is the same as the correlation
among variants, <code>cor(g)</code>.
</p>
<p>With  covariates, correlation  among  test  statistics is  not  the same  as
<code>cor(g)</code>. In this case, <code><a href="#topic+cst">cst()</a></code> takes  the generalized inverse of the entire
correlation  matrix, <code>corr(cbind(g,  x))</code>, and  then inverts  back only  the
submtarix containing genotype variables, <code>g</code>.
</p>
<p>If Z-scores were calculated based on genotypes with some missing values, the
correlation among test statistics will be  reduced by the amount that can be
theoretically derived. It can be shown  that this reduced correlation can be
calculated by imputing  the missing values with the  averages of non-missing
values. Therefore, by default, <code><a href="#topic+cst">cst()</a></code>  fills missing values in each variant
with  the  average  of  non-missing  values  in  that  same  variant  (i.e.,
imputation  by  average, <code><a href="#topic+imp_avg">imp_avg()</a></code>).  Other  imputation  methods are  also
available (see topic <a href="#topic+imp">imp</a> for other techniques that may improve power), but
note that  techniques other than the  imputation by average requires  one to
re-run  the  association  analyses  with  imputed  variants  to  ensure  the
correlation among new statistics (i.e.,  Z-scores) and the correlation among
imputed variants are identical. Otherwise, Type  I error may be inflated for
decorrelation-based methods.
</p>


<h3>Value</h3>

<p>Correlation matrix among association test statistics.
</p>


<h3>See Also</h3>

<p><a href="#topic+imp">imp</a>, <code><a href="#topic+imp_avg">imp_avg()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## get genotype and covariate matrices
gno &lt;- readRDS(system.file("extdata", 'rs208294_gno.rds', package="dotgen"))
cvr &lt;- readRDS(system.file("extdata", 'rs208294_cvr.rds', package="dotgen"))

## correlation among association statistics, covariates involved
res &lt;- cst(gno, cvr)
print(res[1:4, 1:4])

## genotype matrix with 2% randomly missing data
g02 &lt;- readRDS(system.file("extdata", 'rs208294_g02.rds', package="dotgen"))
cvr &lt;- readRDS(system.file("extdata", 'rs208294_cvr.rds', package="dotgen"))
res &lt;- cst(g02, cvr)
print(res[1:4, 1:4])

</code></pre>

<hr>
<h2 id='dot'>Decorrelation by Orthogonal Transformation (DOT)</h2><span id='topic+dot'></span>

<h3>Description</h3>

<p><code><a href="#topic+dot">dot()</a></code>  decorrelates   genetic  association  test  statistics   by  special
symmetric orthogonal transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dot(Z, C, tol.cor = NULL, tol.egv = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dot_+3A_z">Z</code></td>
<td>
<p>vector of association test statistics (i.e., Z-scores).</p>
</td></tr>
<tr><td><code id="dot_+3A_c">C</code></td>
<td>
<p>correlation matrix among the association test statistics, as
obtained by <code><a href="#topic+cst">cst()</a></code>.</p>
</td></tr>
<tr><td><code id="dot_+3A_tol.cor">tol.cor</code></td>
<td>
<p>tolerance threshold for the largest correlation absolute value.</p>
</td></tr>
<tr><td><code id="dot_+3A_tol.egv">tol.egv</code></td>
<td>
<p>tolerance threshold for the smallest eigenvalue.</p>
</td></tr>
<tr><td><code id="dot_+3A_...">...</code></td>
<td>
<p>additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genetic  association studies  typically provide  per-variant test  statistics
that can be  converted to asymptotically normal, signed  Z-scores. Once those
Z-scores are transformed to independent random variables, various methods can
be applied to combine them and obtain SNP-set overall association.
</p>
<p><code><a href="#topic+dot">dot()</a></code>  uses  per-variant  genetic  association  test  statistics  and  the
correlation among them to decorrelate Z-scores.
</p>
<p>To estimate the  correlation among genetic association  test statistics, use
<code><a href="#topic+cst">cst()</a></code>.  If  P-values and  estimated effects  (i.e, beta  coefficients) are
given instead  of test statistics, <code><a href="#topic+zsc">zsc()</a></code>  can be used to  recover the test
statistics (i.e., Z-scores).
</p>
<p><code>tol.cor</code>: variants  with correlation too close  to 1 in absolute  value are
considered to be collinear  and only one of them will  be retained to ensure
that  the  LD  matrix  is   full-rank.   The  maximum  value  for  tolerable
correlation  is  1   -  <code>tol.cor</code>.  The  default  value   for  <code>tol.cor</code>  is
<code>sqrt(.Machine$double.eps)</code>.
</p>
<p><code>tol.egv</code>: negative and close to  zero eigenvalues are truncated from matrix
<code>D</code> in <code style="white-space: pre;">&#8288;H = EDE'&#8288;</code>. The corresponding  columns of <code>E</code> are also deleted. Note
the  the dimention  of the  square matrix  <code>H</code> does  not change  after this
truncation. See DOT publication in the  reference below for more details on
definitions  of <code>E</code>  and <code>D</code>  matrices.  The  default eigenvalue  tolerance
value is <code>sqrt(.Machine$double.eps)</code>.
</p>
<p>A number of methods are available for combining de-correlated P-values,
see <a href="#topic+dot_sst">dot_sst</a> for details.
</p>


<h3>Value</h3>

<p>a  list with
</p>

<ul>
<li><p><code>X</code>: association test statistics, de-correlated.
</p>
</li>
<li><p><code>H</code>: orthogonal transformation, such that <code>X = H %*% Z</code>.
</p>
</li>
<li><p><code>M</code>: effective number of variants after de-correlation.
</p>
</li>
<li><p><code>L</code>: effective number of eigenvalues after truncation.
</p>
</li></ul>



<h3>References</h3>

<p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007819">
Vsevolozhskaya, O. A., Shi, M., Hu, F., &amp; Zaykin, D. V. (2020). DOT: Gene-set
analysis by combining decorrelated association statistics. PLOS Computational
Biology, 16(4), e1007819.</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cst">cst()</a></code>, <code><a href="#topic+zsc">zsc()</a></code>, <a href="#topic+dot_sst">dot_sst</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## get genotype and covariate matrices
gno &lt;- readRDS(system.file("extdata", 'rs208294_gno.rds', package="dotgen"))
cvr &lt;- readRDS(system.file("extdata", 'rs208294_cvr.rds', package="dotgen"))

## estimate the correlation among association test statistics
sgm &lt;- cst(gno, cvr)

## get the result of genetic association analysis (P-values and effects)
res &lt;- readRDS(system.file("extdata", 'rs208294_res.rds', package="dotgen"))

## recover Z-score statistics
stt &lt;- with(res, zsc(P, BETA))

## decorrelate Z-scores by DOT
result &lt;- dot(stt, sgm)
print(result$X)          # decorrelated statistics
print(result$H)          # orthogonal transformation

## sum of squares of decorrelated statistics is a chi-square
ssq &lt;- sum(result$X^2)
pvl &lt;- 1 - pchisq(ssq, df=result$L)

print(ssq)            # sum of squares = 35.76306
print(pvl)            # chisq P-value =  0.001132132
</code></pre>

<hr>
<h2 id='dot_sst'>Methods for combining decorrelated summary statistics</h2><span id='topic+dot_sst'></span><span id='topic+dot_chisq'></span><span id='topic+dot_fisher'></span><span id='topic+dot_art'></span><span id='topic+dot_arta'></span><span id='topic+dot_rtp'></span><span id='topic+dot_tpm'></span>

<h3>Description</h3>

<p>Decorrelates and combines per-variant genetic association test statistics, <code>Z</code>,
given the correlation matrix among them, <code>C</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dot_chisq(Z, C, ...)

dot_fisher(Z, C, ...)

dot_art(Z, C, k = NULL, ...)

dot_arta(Z, C, k = NULL, w = NULL, ...)

dot_rtp(Z, C, k = NULL, ...)

dot_tpm(Z, C, tau = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dot_sst_+3A_z">Z</code></td>
<td>
<p>vector of association test statistics (i.e., Z-scores).</p>
</td></tr>
<tr><td><code id="dot_sst_+3A_c">C</code></td>
<td>
<p>matrix of  correlation among the  test statistics, as  obtained by
<code><a href="#topic+cst">cst()</a></code>.</p>
</td></tr>
<tr><td><code id="dot_sst_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
<tr><td><code id="dot_sst_+3A_k">k</code></td>
<td>
<p>combine <code>k</code> smallest (decorrelated) P-values.</p>
</td></tr>
<tr><td><code id="dot_sst_+3A_w">w</code></td>
<td>
<p>weight assigned to partial sums in ARTA implementation; default is 1.</p>
</td></tr>
<tr><td><code id="dot_sst_+3A_tau">tau</code></td>
<td>
<p>combine (decorrelated) P-values no large than tau; default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions  first call <code><a href="#topic+dot">dot()</a></code>  to decorrelate the  genetic association
test  statistics and  then provide  various options  to combine  independent
statistics or corresponding P-values into the overall statistic and P-value.
</p>
<p>The two  rank truncated  tests (i.e.,  <code><a href="#topic+dot_art">dot_art()</a></code>, <code><a href="#topic+dot_rtp">dot_rtp()</a></code>)  require an
additional parameter <code>k</code> that specifes the number of smallest (decorrelated)
P-values to combine. By default, <code>k</code>  equals half of the number of variants.
The adaptive  rank truncation  method, <code><a href="#topic+dot_arta">dot_arta()</a></code>, determines  the optimal
truncation value between 1 and <code>k</code>.
</p>
<p>The truncated  product method,  <code><a href="#topic+dot_tpm">dot_tpm()</a></code>, combines  P-values at  least as
small as <code>tau</code> (0.05 by default).  If  <code>tau</code> is equal to 1, then <code><a href="#topic+dot_tpm">dot_tpm()</a></code>
provides  the  same result  as  <code><a href="#topic+dot_fisher">dot_fisher()</a></code>  (i.e., Fisher's  method  for
combining  P-values). Similarly,  if <code>k</code>  is equal  to the  total number  of
tests, the results  of <code><a href="#topic+dot_art">dot_art()</a></code> and <code><a href="#topic+dot_rtp">dot_rtp()</a></code> will be  the same as that
of <code><a href="#topic+dot_fisher">dot_fisher()</a></code>.
</p>
<p>Reference  (<strong>a</strong>)  below  details  how to  combine  decorrelated  test
statistics  or  P-values  via  <code><a href="#topic+dot_art">dot_art()</a></code>,  <code><a href="#topic+dot_rtp">dot_rtp()</a></code>  and  <code><a href="#topic+dot_arta">dot_arta()</a></code>;
reference (<strong>b</strong>) details <code><a href="#topic+dot_tpm">dot_tpm()</a></code> method.
</p>


<h3>Value</h3>

<p>a list of
</p>

<ul>
<li><p><code>X</code>: decorrelated  association statistics.
</p>
</li>
<li><p><code>H</code>: orthogonal transformation, such that <code>X = H %*% Z</code>.
</p>
</li>
<li><p><code>Y</code>: the overall combined statistic.
</p>
</li>
<li><p><code>P</code>: the P-value corresponding to <code>Y</code>.
</p>
</li></ul>

<p>for Augmented Rank Truncated Adaptive (ARTA) test,
</p>

<ul>
<li><p>k: the number of decorrelated P-values that were adaptively picked.</p>
</li></ul>

<p>for Truncated Product Method (TPM),
</p>

<ul>
<li><p>k: the number of decorrelated P-values <code class="reqn">\le</code> <code>tau</code>.</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>dot_chisq</code>: decorrelation followed by a Chi-square test.
</p>
</li>
<li> <p><code>dot_fisher</code>: decorrelated Fisher's combined P-value test.
</p>
</li>
<li> <p><code>dot_art</code>: decorrelated Augmented Rank Truncated (ART) test.
</p>
</li>
<li> <p><code>dot_arta</code>: decorrelated Augmented Rank Truncated Adaptive (ARTA) test.
</p>
</li>
<li> <p><code>dot_rtp</code>: decorrelated Rank Truncated Product (RTP) test.
</p>
</li>
<li> <p><code>dot_tpm</code>: decorrelated Truncated Product Method (TPM) test.
</p>
</li></ul>


<h3>References</h3>

<p>(a) <a href="https://www.frontiersin.org/articles/10.3389/fgene.2019.01051/full">
Vsevolozhskaya, O.   A., Hu, F., &amp;  Zaykin, D.  V. (2019).   <em>Detecting weak
signals  by  combining  small  P-values  in  genetic  association  studies.</em>
Frontiers in genetics, 10, 1051.</a>
</p>
<p>(b) <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/gepi.0042">Zaykin,
D.    V.,   Zhivotovsky,    L.     A.,   Westfall,    P.    H.,   &amp;    Weir,
B. S.  (2002). <em>Truncated  product method  for combining  P-values.</em> Genetic
Epidemiology, 22(2), 170-185.</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dot">dot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## get the test statistics and pre-calculated LD matrix
stt &lt;- readRDS(system.file("extdata", 'art_zsc.rds', package="dotgen"))
sgm &lt;- readRDS(system.file("extdata", 'art_ldm.rds', package="dotgen"))


## decorrelated chi-square test
result &lt;- dot_chisq(stt, sgm)
print(result$Y)  # 37.2854
print(result$P)  # 0.0003736988

## decorrelated Fisher's combined P-value chi-square test
result &lt;- dot_fisher(stt, sgm)
print(result$Y)  # 58.44147
print(result$P)  # 0.0002706851

## decorrelated augmented rank truncated (ART) test.
result &lt;- dot_art(stt, sgm, k=6)
print(result$Y)  # 22.50976
print(result$P)  # 0.0006704994

## decorrelated Augmented Rank Truncated Adaptive (ARTA) test
result &lt;- dot_arta(stt, sgm, k=6)
print(result$Y)  # -1.738662
print(result$k)  #  5 smallest P-values are retained
print(result$P)  #  0.003165 (varies)

## decorrelated Rank Truncated Product (RTP)
result &lt;- dot_rtp(stt, sgm, k=6)
print(result$Y)  # 22.6757
print(result$P)  # 0.0007275518

## decorrelated Truncated Product Method (TPM)
result &lt;- dot_tpm(stt, sgm, tau=0.05)
print(result$Y)  #  1.510581e-08
print(result$k)  #  6 P-values &lt;= tau
print(result$P)  #  0.0007954961

</code></pre>

<hr>
<h2 id='imp'>Impute missing genotypes</h2><span id='topic+imp'></span><span id='topic+imp_avg'></span><span id='topic+imp_cnd'></span>

<h3>Description</h3>

<p>Impute missing genotype calls with values inferred from non-missing ones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imp_avg(g, ...)

imp_cnd(g, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imp_+3A_g">g</code></td>
<td>
<p>genotype matrix, one row per sample, and one column per variant.</p>
</td></tr>
<tr><td><code id="imp_+3A_...">...</code></td>
<td>
<p>additional parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A seemingly naive way to impute a missing value is to use the average of all
non-missing  values  per  variant, <code><a href="#topic+imp_avg">imp_avg()</a></code>.   Besides  simplicity,  this
imputation by  average has  the advantage  of approximating  the correlation
among  test  statistics  (i.e.,  Z-scores)  when  the  original  association
analyses  were performed  with missing  values unfilled,  which is  a common
practice.  This naive approach is the defualt for the correlation calculator
<code><a href="#topic+cst">cst()</a></code>.
</p>
<p>An  advanced imputation  approach is  based on  the conditional  expectation
method,  <code><a href="#topic+imp_cnd">imp_cnd()</a></code>, that  explores the  relationship between  variants and
borrows  information from  variants other  than the  target one  when making
guesses.  The sample  correlation among variants imputed this  way is closer
to the  true LD, and may  improve power.  However, after  this imupation one
must  re-run  the  association  analyses  with  imputed  variants  to  avoid
inflation in Type I error rates.
</p>


<h3>Value</h3>

<p>imputed genotype matrix without any missing values.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>imp_avg</code>: imputation by average.
</p>
</li>
<li> <p><code>imp_cnd</code>: imputation by conditional expectation
</p>
</li></ul>

<hr>
<h2 id='zsc'>Calculate Z-scores from P-values and estimated effects</h2><span id='topic+zsc'></span>

<h3>Description</h3>

<p><code><a href="#topic+zsc">zsc()</a></code> recovers Z-scores from  P-values and corresponding effect directions
(or beta coefficients) reported by a genetic association analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zsc(P, BETA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zsc_+3A_p">P</code></td>
<td>
<p>vector of P-values.</p>
</td></tr>
<tr><td><code id="zsc_+3A_beta">BETA</code></td>
<td>
<p>vector of effect directions or beta coefficients.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For any  genetic variant,  its two-sided  P-value (<code class="reqn">p</code>)  and the  sign of
estimated effect (<code class="reqn">\beta</code>) is used to recover the Z-score (<code class="reqn">z</code>), that
is, <code class="reqn">z = sign(\beta) \Phi^{-1}(p/2)</code>.
</p>


<h3>Value</h3>

<p>A vector of Z-scores.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dot">dot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## result of per-variant analysis (P-values and estimated effects)
res &lt;- readRDS(system.file("extdata", 'rs208294_res.rds', package="dotgen"))

## recover Z-score statistics
stt &lt;- with(res, zsc(P, BETA))

## checking
stopifnot(all.equal(pnorm(abs(stt), lower.tail = FALSE) * 2, res$P))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
