<!DOCTYPE html><html><head><title>Help for package DoubleExpSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DoubleExpSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#counts'>
<p>Exon Inclusion Counts</p>
</a></li>
<li><a href='#DB.MAPlot'>
<p>Plots Log-Fold Change versus Log-Concentration for Inclusion/Exclusion Data</p>
</a></li>
<li><a href='#DBGLM1'>

<p>Double Binomial Generalized Linear Model with Shrinkage of the Dispersion Parameter</p></a></li>
<li><a href='#DoubleExpSeq-package'>
<p>DoubleExpSeq is a package with application to RNA-Seq experiments that tests for differential exon usage.</p></a></li>
<li><a href='#EstimateDEBDisp'>
<p>DEB-Seq: Empirical Bayes Estimates of Dispersion for a Double Binomial Distribution</p>
</a></li>
<li><a href='#EstimateWEBDisp'>
<p>WEB-Seq: Weighted Likelihood Empirical Bayes Estimates of Dispersion for a Double Binomial Distribution</p>
</a></li>
<li><a href='#exon'>
<p>Toy Exon Inclusion and Total Counts Used for Examples</p>
</a></li>
<li><a href='#groups'>
<p>Group Structure of the Toy Data Set</p>
</a></li>
<li><a href='#offsets'>
<p>Exon Total Counts</p>
</a></li>
<li><a href='#optimPlot'>
<p>Plot the WEB-Seq Maximum Likelihood Solution for the Weight Parameter in the Weighted Likelihood</p>
</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Differential Exon Usage Test for RNA-Seq Data via Empirical
Bayes Shrinkage of the Dispersion Parameter</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-09-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Sean Ruddy</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sean Ruddy &lt;s.ruddy@yahoo.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Differential exon usage test for RNA-Seq data via an empirical Bayes shrinkage method for the dispersion parameter the utilizes inclusion-exclusion data to analyze the propensity to skip an exon across groups. The input data consists of two matrices where each row represents an exon and the columns represent the biological samples. The first matrix is the count of the number of reads expressing the exon for each sample. The second matrix is the count of the number of reads that either express the exon or explicitly skip the exon across the samples, a.k.a. the total count matrix. Dividing the two matrices yields proportions representing the propensity to express the exon versus skipping the exon for each sample.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>numDeriv, datasets, grDevices, graphics, stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-09-03 23:11:01 UTC; exon</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-09-04 09:25:45</td>
</tr>
</table>
<hr>
<h2 id='counts'>
Exon Inclusion Counts

</h2><span id='topic+counts'></span>

<h3>Description</h3>

<p>A toy data set of inclusion counts consisting of 3 groups each with 5 samples.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exon)
</code></pre>


<h3>Format</h3>

<p>numeric matrix
</p>

<hr>
<h2 id='DB.MAPlot'>
Plots Log-Fold Change versus Log-Concentration for Inclusion/Exclusion Data

</h2><span id='topic+DB.MAPlot'></span>

<h3>Description</h3>

<p>M-A Plot

</p>


<h3>Usage</h3>

<pre><code class='language-R'>DB.MAPlot( y, m, groups, contrast=c(1,2), de.tags=NULL,
 col="lightgrey", deCol="red", deCex=0.2,
 xlab="Average Over Groups of log2 Mean Total Count",
 ylab="logFC of Odds Ratio", pch=19, cex=0.2,
 panel.last=grid(col = "red", lwd = 0.2) , ylim = c(-15, 15), ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DB.MAPlot_+3A_y">y</code></td>
<td>

<p>numeric matrix of inclusion counts.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_m">m</code></td>
<td>

<p>numeric matrix of total counts: inclusion + exclusion.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_groups">groups</code></td>
<td>

<p>vector or factor giving the experimental group/condition for each sample/library.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_contrast">contrast</code></td>
<td>

<p>numeric vector of length 2 specifying which levels of the &quot;groups&quot; factor should be compared.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_de.tags">de.tags</code></td>
<td>

<p>rownames for events identified as being differentially expressed.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_col">col</code></td>
<td>

<p>color given to the points.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_decol">deCol</code></td>
<td>

<p>color for the events given in &quot;de.tags&quot;.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_decex">deCex</code></td>
<td>

<p>cex for the events given in &quot;de.tags&quot;.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_xlab">xlab</code></td>
<td>

<p>x-label of plot

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_ylab">ylab</code></td>
<td>

<p>y-label of plot

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_pch">pch</code></td>
<td>

<p>pch given to the points.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_cex">cex</code></td>
<td>

<p>cex given to the points.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_panel.last">panel.last</code></td>
<td>

<p>an expression to be evaluated after plotting; the default grid() draws a background grid to aid interpretation of the plot.

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_ylim">ylim</code></td>
<td>

<p>y-limits for the plot

</p>
</td></tr>
<tr><td><code id="DB.MAPlot_+3A_...">...</code></td>
<td>

<p>further arguments passed to plot().

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The total counts are used to determine A, and the log-fold change of the odds ratio is used to determine M. In the case where a group has proportions all 1 or all 0, resulting in an infinite value for M, these points are plotted in orange and away from the main plot. Significant calls made for such events are still colored in red.

</p>


<h3>Value</h3>

<p>A plot to the current device





</p>


<h3>Author(s)</h3>

<p>Sean Ruddy

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exon)
results.G1G2 &lt;- DBGLM1( counts, offsets, groups)
de.tags.G1G2 &lt;- rownames(results.G1G2$Sig)
DB.MAPlot(counts, offsets, groups, contrast=c(1,2), de.tags=de.tags.G1G2)
</code></pre>

<hr>
<h2 id='DBGLM1'>

Double Binomial Generalized Linear Model with Shrinkage of the Dispersion Parameter
</h2><span id='topic+DBGLM1'></span>

<h3>Description</h3>


<p>Fits a double binomial GLM with the normalization constant set to 1 and uses shrinkage to obtain estimates of dispersion used for p-value calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DBGLM1( y, m, groups, shrink.method=c("WEB","DEB"),
        contrast=c(1,2), fdr.level=0.05, use.all.groups=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DBGLM1_+3A_y">y</code></td>
<td>

<p>numeric matrix of inclusion counts.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_m">m</code></td>
<td>

<p>numeric matrix of total counts: inclusion + exclusion.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_groups">groups</code></td>
<td>

<p>vector or factor giving the experimental group/condition for each sample/library.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_shrink.method">shrink.method</code></td>
<td>

<p>for shinkage estimation of the dispersion parameter. &quot;WEB&quot; implements the WEB-Seq method. &quot;DEB&quot; implements the DEB-Seq method. Default is &quot;WEB&quot;.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_contrast">contrast</code></td>
<td>

<p>numeric vector of length 2 specifying which levels of the &quot;groups&quot; factor should be compared.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_fdr.level">fdr.level</code></td>
<td>

<p>a numeric constant. The FDR level to determine the list of significant events. Default is 0.05.

</p>
</td></tr>
<tr><td><code id="DBGLM1_+3A_use.all.groups">use.all.groups</code></td>
<td>

<p>logical. If TRUE, all data in &quot;y&quot; is used to estimate dispersions. If FALSE, only the 2 groups given in &quot;contrasts&quot; are used to estimate dispersions. Only makes a difference if &quot;y&quot; contains more than 2 groups. Default is TRUE.

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function tests for group differences for a two group comparison via a double binomial GLM with the normalization constant set to 1, and utilizes shrinkage estimates of the dispersion parameter for p-value calcuation which is done using a likelihood ratio test. The shrinkage estimates of the dispersion are found according to the selection of &quot;shrink.method&quot;. &quot;DEB&quot; implements the DEB-Seq method which uses an empirical bayes strategy to obtain shrunken estimates of the dipersion parameter. &quot;WEB&quot; implements the WEB-Seq method which reparameterizes the empirical bayes strategy in terms of the weight parameter in the weighted liklelihood formulation. An emprical bayes estimate of the weight parameter is found and plugged into the weighted likelihood which is then maximized to obtain shrunken estimates of the dipsersion parameter. DEB-Seq has shown to be more powerful than WEB-Seq; however, WEB-Seq is more conservative thus being more robust against departures from assumptions and therefore maintains the required FDR better in moderate to larger sample sizes.
</p>
<p>In the case when &quot;groups&quot; specifies more than two groups the default procedure is to use all groups to calculate the shrunken dispersion estimates. The argument &quot;constrast&quot; is used to specify a particular comparison of two of the groups. If &quot;use.all.groups&quot; is FALSE, only the data for the groups specified in &quot;contrasts&quot; are used to estimate the dispersions.

</p>


<h3>Value</h3>

<table>
<tr><td><code>Sig</code></td>
<td>
<p>a matrix consisting of the significant events at the specified FDR level. The matrix contains the proportion estimates, unadjusted and adjusted p-values, the effective sample size, mean total count and dispersion estimates. </p>
</td></tr>
<tr><td><code>All</code></td>
<td>
<p>the same matrix  as above but consisting of all events.</p>
</td></tr>





</table>


<h3>Author(s)</h3>

<p>Sean Ruddy

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Toy exon data: "counts"=inclusion counts, "offsets"=total counts, "groups"=experiment design
  data(exon)

  ## Default will compare "G1" &amp; "G2" using the WEB-Seq method
  ## and uses ALL groups to estimate dispersion
    results.G1G2.WEB &lt;- DBGLM1( counts, offsets, groups)

    # Compare G1 &amp; G3
    results.G1G3.WEB &lt;- DBGLM1( counts, offsets, groups, contrast=c(1,3))

    # Compare G1 &amp; G3. Does not use G2 for dispersion estimation.
    results.G1G3.noG2.WEB &lt;- DBGLM1(counts, offsets, groups, contrast=c(1,3), use.all.groups=FALSE)

    # Global minimum check of the weight parameter estimate in the WEB-Seq method.
    optimPlot( counts, offsets, groups, contrast=c(1,3), use.all.groups=FALSE)

## The DEB-Seq method. Less conservative, more powerful. Very similar ranks to WEB-Seq.
  results.G1G2.DEB &lt;- DBGLM1( counts, offsets, groups, shrink.method="DEB")

## M-A Plot
  WEB.sig &lt;- rownames(results.G1G2.WEB$Sig)
  DB.MAPlot( counts, offsets, groups, de.tags=WEB.sig, main="WEB-Seq")
</code></pre>

<hr>
<h2 id='DoubleExpSeq-package'>
DoubleExpSeq is a package with application to RNA-Seq experiments that tests for differential exon usage.
</h2><span id='topic+DoubleExpSeq-package'></span><span id='topic+DoubleExpSeq'></span>

<h3>Description</h3>

<p>The functions take inclusion and total counts. Inclusion counts are counts that express the exon. Exclusion counts are counts that explicitly skip the exon. The Total count is the sum of inclusion and exclusion.
The package provides 2 methods for the analysis of differential exon usage in RNA-Seq data: 1) DEB-Seq and 2) WEB-Seq. Each of these assume a double-binomial distribution with the normalization constant equal to 1. A simple empirical bayes strategy is used to shrink the dispersion parameter toward a common consensus from all exons.
DEB-Seq implements a 2-parameter empirical bayes strategy to estimate shrunken dispersion estimates. WEB-Seq implements a weighted likelihood approach and estimates the weight parameter via the empirical bayes strategy after reparameterizing the posterior predictive distribution in terms of only the weight parameter.
Once the dispersion estimates are found, the count data is fit with a double binomial GLM (ignoring the normalization constant) and the dispersion estimates used within the test statistic calculation, but have no bearing on the estimates of the proportion.
The main function returns p-values, adjusted p-values using the Benjamini-Hochberg procedure for multiple testing, and proportion estimates along with other relevant information for each exon.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> DoubleExpSeq</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2014-05-13</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Sean Ruddy
Maintainer: Sean Ruddy &lt;s.ruddy@yahoo.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Toy exon data: "counts"=inclusion counts, "offsets"=total counts, "groups"=experiment design
  data(exon)
  
  ## Default will compare "G1" &amp; "G2" using the WEB-Seq method
  ## and uses ALL groups to estimate dispersion
    results.G1G2.WEB &lt;- DBGLM1( counts, offsets, groups)
    
    # Compare G1 &amp; G3
    results.G1G3.WEB &lt;- DBGLM1( counts, offsets, groups, contrast=c(1,3))
	
    # Compare G1 &amp; G3. Does not use G2 for dispersion estimation.
    results.G1G3.noG2.WEB &lt;- DBGLM1(counts, offsets, groups, contrast=c(1,3), use.all.groups=FALSE)
	    
    # Global minimum check of the weight parameter estimate in the WEB-Seq method.
    optimPlot( counts, offsets, groups, contrast=c(1,3), use.all.groups=FALSE)
		
## The DEB-Seq method. Less conservative, more powerful. Very similar ranks to WEB-Seq.
  results.G1G2.DEB &lt;- DBGLM1( counts, offsets, groups, shrink.method="DEB")
		  
## M-A Plot
  WEB.sig &lt;- rownames(results.G1G2.WEB$Sig)
  DB.MAPlot( counts, offsets, groups, de.tags=WEB.sig, main="WEB-Seq")
</code></pre>

<hr>
<h2 id='EstimateDEBDisp'>
DEB-Seq: Empirical Bayes Estimates of Dispersion for a Double Binomial Distribution

</h2><span id='topic+EstimateDEBDisp'></span>

<h3>Description</h3>

<p>Calculation of shrunken dispersion estimates via a 2-parameter empirical bayes method.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>EstimateDEBDisp(y,m,groups=NULL,neff=NULL,S=NULL,optim.method=c("BFGS","Nelder-Mead"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstimateDEBDisp_+3A_y">y</code></td>
<td>

<p>numeric matrix of inclusion counts.

</p>
</td></tr>
<tr><td><code id="EstimateDEBDisp_+3A_m">m</code></td>
<td>

<p>numeric matrix of total counts: inclusion + exclusion.

</p>
</td></tr>
<tr><td><code id="EstimateDEBDisp_+3A_groups">groups</code></td>
<td>

<p>vector or factor giving the experimental group/condition for each sample/library. Default NULL.

</p>
</td></tr>
<tr><td><code id="EstimateDEBDisp_+3A_neff">neff</code></td>
<td>

<p>numeric vector of length equal to the number of rows of &quot;y&quot; where each value is the effective sample size for the event. Default is NULL in which case the effective sample size is calculated within the function.

</p>
</td></tr>
<tr><td><code id="EstimateDEBDisp_+3A_s">S</code></td>
<td>

<p>numeric vector of length equal to the number of rows of &quot;y&quot; where each value is the random variable for each event whose distribution across exons is gamma. Default is NULL in which case the vector is calculated internally.

</p>
</td></tr>
<tr><td><code id="EstimateDEBDisp_+3A_optim.method">optim.method</code></td>
<td>

<p>character string determining which optimization routine to use for estimating the parameters of the prior distribution. Default is &quot;BFGS&quot;.

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The shrunken dispersion estimates are a function of 2 parameters of the generalized beta prime distribution which are estimated via maximum likelihood resulting in empricaly bayes shinkage of the dispersion parameter.

</p>


<h3>Value</h3>

<p>vector of length equal to the number of rows of &quot;y&quot; where each value is the estimate of dispersion.





</p>


<h3>Author(s)</h3>

<p>Sean Ruddy

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exon)
dispersions &lt;- EstimateDEBDisp( counts, offsets, groups)
</code></pre>

<hr>
<h2 id='EstimateWEBDisp'>
WEB-Seq: Weighted Likelihood Empirical Bayes Estimates of Dispersion for a Double Binomial Distribution

</h2><span id='topic+EstimateWEBDisp'></span>

<h3>Description</h3>

<p>Calculation of shrunken dispersion estimates via weighted likelihood where the weight parameter is estimated using an empirical bayes strategy.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>EstimateWEBDisp(y, m, groups, neff = NULL, S = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EstimateWEBDisp_+3A_y">y</code></td>
<td>

<p>numeric matrix of inclusion counts.

</p>
</td></tr>
<tr><td><code id="EstimateWEBDisp_+3A_m">m</code></td>
<td>

<p>numeric matrix of total counts: inclusion + exclusion.

</p>
</td></tr>
<tr><td><code id="EstimateWEBDisp_+3A_groups">groups</code></td>
<td>

<p>vector or factor giving the experimental group/condition for each sample/library.

</p>
</td></tr>
<tr><td><code id="EstimateWEBDisp_+3A_neff">neff</code></td>
<td>

<p>numeric vector of length equal to the number of rows of &quot;y&quot; where each value is the effective sample size for the event. Default is NULL in which case the effective sample size is calculated within the function.

</p>
</td></tr>
<tr><td><code id="EstimateWEBDisp_+3A_s">S</code></td>
<td>

<p>numeric vector of length equal to the number of rows of &quot;y&quot; where each value is the random variable for each event whose distribution across exons is gamma. Default is NULL in which case the vector is calculated internally.

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Shrunken dispersion estimates are obtained by maximizing the weighted sum of the likelihood for a given event and the sum of likelihoods for all events, the common likelihood. The weight given to the common likelihood is estimated via empirical bayes.

</p>


<h3>Value</h3>

<p>vector of length equal to the number of rows of &quot;y&quot; where each value is the estimate of dispersion.





</p>


<h3>Author(s)</h3>

<p>Sean Ruddy

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exon)
dispersions &lt;- EstimateWEBDisp( counts, offsets, groups)
</code></pre>

<hr>
<h2 id='exon'>
Toy Exon Inclusion and Total Counts Used for Examples

</h2><span id='topic+exon'></span>

<h3>Description</h3>

<p>A toy RNA-Seq count data set consisting of 3 groups of 5 samples each. Goal is to detect differential exon usage between groups.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exon)
</code></pre>

<hr>
<h2 id='groups'>
Group Structure of the Toy Data Set

</h2><span id='topic+groups'></span>

<h3>Description</h3>

<p>A vector signifying which samples of the toy exon data set belong to which of the 3 groups.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exon)
</code></pre>


<h3>Format</h3>

<p>character vector
</p>

<hr>
<h2 id='offsets'>
Exon Total Counts

</h2><span id='topic+offsets'></span>

<h3>Description</h3>

<p>A toy data set of Total (inclusion+exclusion) counts consisting of 3 groups each with 5 samples.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exon)
</code></pre>


<h3>Format</h3>

<p>numeric matrix
</p>

<hr>
<h2 id='optimPlot'>
Plot the WEB-Seq Maximum Likelihood Solution for the Weight Parameter in the Weighted Likelihood

</h2><span id='topic+optimPlot'></span>

<h3>Description</h3>

<p>The MLE solution is signified on a plot as the minimum of the negative log likelihood of the generalized beta prime distribution, parameterized in terms of the weight parameter.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimPlot(y, m, groups, contrast=c(1,2), use.all.groups=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimPlot_+3A_y">y</code></td>
<td>

<p>numeric matrix of inclusion counts.

</p>
</td></tr>
<tr><td><code id="optimPlot_+3A_m">m</code></td>
<td>

<p>numeric matrix of total counts: inclusion + exclusion.

</p>
</td></tr>
<tr><td><code id="optimPlot_+3A_groups">groups</code></td>
<td>

<p>vector or factor giving the experimental group/condition for each sample/library.

</p>
</td></tr>
<tr><td><code id="optimPlot_+3A_contrast">contrast</code></td>
<td>

<p>numeric vector of length 2 specifying which levels of the &quot;groups&quot; factor should be compared. This is only relevant if &quot;use.all.groups&quot; is FALSE.

</p>
</td></tr>
<tr><td><code id="optimPlot_+3A_use.all.groups">use.all.groups</code></td>
<td>

<p>logical. If TRUE, all data in &quot;y&quot; is used to estimate dispersions. If FALSE, only the 2 groups given in &quot;contrasts&quot; are used to estimate dispersions. Only makes a difference if &quot;y&quot; contains more than 2 groups. Default is TRUE.

</p>
</td></tr>
<tr><td><code id="optimPlot_+3A_...">...</code></td>
<td>

<p>further arguments passes to plot()
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The MLE estimate of the (transformed) weight parameter in the WEB-Seq method is checked to be a true global minimum of the negative log likelihood of the generalized beta prime distribution. The weight parameter is transformed from an infinite range to the (0,1) range before optimization and this is the range on which the estimate is checked.

</p>


<h3>Value</h3>

<p>A plot to the current device





</p>


<h3>Author(s)</h3>

<p>Sean Ruddy

</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exon)
# If all groups were used to estimate dispersions
  optimPlot(counts, offsets, groups) 
# If only the 2 groups being compared were used to estimate dispersions
  optimPlot(counts, offsets, groups, contrast=c(1,3), use.all.groups=FALSE) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
