<!DOCTYPE html><html><head><title>Help for package DPBBM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DPBBM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bbm_data_generate'>
<p>bbm_data_generate</p></a></li>
<li><a href='#BCubed_metric'>
<p>BCubed_metric</p></a></li>
<li><a href='#dpbbm_mc_iterations'>
<p>dpbbm_mc_iterations</p></a></li>
<li><a href='#DPBBM-package'>
<p>DPBBM</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Dirichlet Process Beta-Binomial Mixture</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-09-21</td>
</tr>
<tr>
<td>Author:</td>
<td>Lin Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lin Zhang &lt;lin.zhang@cumt.edu.cn&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>tmvtnorm, VGAM, gplots, CEoptim</td>
</tr>
<tr>
<td>Description:</td>
<td>Beta-binomial Mixture Model is used to infer the pattern from count data.
		It can be used for clustering of RNA methylation sequencing data. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-28 15:21:06 UTC; lin.zhang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-29 08:02:30</td>
</tr>
</table>
<hr>
<h2 id='bbm_data_generate'>
bbm_data_generate
</h2><span id='topic+bbm_data_generate'></span>

<h3>Description</h3>

<p>This is to generate the simulation data based on Beta-bionomial mixture model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbm_data_generate(S=3, G=50, K=3, prob=rep(1,times=3),
                              alpha_band=c(2,6),
                              beta_band=c(2,6),
                              nb_mu=100,nb_size=0.2, plotf = FALSE, 
                              max_cor=0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbm_data_generate_+3A_s">S</code></td>
<td>
<p> Number of samples in the simulated data

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_g">G</code></td>
<td>
<p> Number of sites in the simulated data

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_k">K</code></td>
<td>
<p> Number of clusters that exist in the simulated data

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_prob">prob</code></td>
<td>
<p> the cluster weight for each cluster

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_alpha_band">alpha_band</code></td>
<td>
<p> the region used to generate the parameter of beta distribution alpha

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_beta_band">beta_band</code></td>
<td>
<p> the region used to generate the parameter of beta distribution beta

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_nb_mu">nb_mu</code></td>
<td>
<p> alternative parametrization via mean for Negative Binomial distribution

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_nb_size">nb_size</code></td>
<td>
<p> target for number of successful trials, or dispersion parameter (the shape parameter of the gamma mixing distribution) for Negative binomial distrition. Must be strictly positive, need not be integer.

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_plotf">plotf</code></td>
<td>
<p> option for whether plot the generated data according to clusters or not

</p>
</td></tr>
<tr><td><code id="bbm_data_generate_+3A_max_cor">max_cor</code></td>
<td>
<p> The maximized correlation allowed for the simulated data, which used to guarantee the data is not highly correlated.

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Dirichlet Process based beta-binomial mixture model clustering
</p>


<h3>Value</h3>

<p>The function returns simulation data generated based on beta binomial mixture model
</p>


<h3>Author(s)</h3>

<p> Lin Zhang, PhD  &lt;lin.zhang@cumt.edu.cn&gt;

</p>


<h3>References</h3>

<p>Reference coming soon!
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123455)
S &lt;- 4
G &lt;- 100
K &lt;- 3
nb_mu &lt;- 100
nb_size &lt;- 0.8
prob &lt;- c(1,1,1)
mat &lt;- bbm_data_generate(S=S,G=G,K=K,prob=prob,alpha_band=c(2,6),beta_band=c(2,6),
                         nb_mu=nb_mu,nb_size=nb_size, plotf = TRUE, max_cor=0.5) 
table(mat$gamma)
pie(mat$gamma)
id &lt;- order(mat$gamma);
c &lt;- mat$gamma[id]
mat_ratio &lt;- (mat$k+1)/(mat$n+1);
heatmap(mat_ratio[id,], Rowv = NA, Colv = NA, scale="none", RowSideColors=as.character(c), 
        xlab = "4 samples", ylab="100 RNA methylation sites")
</code></pre>

<hr>
<h2 id='BCubed_metric'>
BCubed_metric
</h2><span id='topic+BCubed_metric'></span>

<h3>Description</h3>

<p>This is to evaluate the clustering method with Bcubed F sore.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BCubed_metric(L, C, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BCubed_metric_+3A_l">L</code></td>
<td>
<p> real label of classes 

</p>
</td></tr>
<tr><td><code id="BCubed_metric_+3A_c">C</code></td>
<td>
<p> classification label of classes drawn by clustering method

</p>
</td></tr>
<tr><td><code id="BCubed_metric_+3A_alpha">alpha</code></td>
<td>
<p> F metric parameter which used to average precision and recall

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The clustering evaluation method based on Bcubed F metric.
</p>


<h3>Value</h3>

<p>The function returns Bcubed F score of the clustering method. The higher the value is, the better performance the clustering method can get.
</p>


<h3>Author(s)</h3>

<p> Lin Zhang, PhD  &lt;lin.zhang@cumt.edu.cn&gt;

</p>


<h3>References</h3>

<p>Reference coming soon!
</p>


<h3>Examples</h3>

<pre><code class='language-R'>L &lt;- c(1,1,2,1,1,2,2)
C &lt;- c(2,2,1,2,2,1,1)
alpha &lt;- 0.5
Bcubed_score &lt;- BCubed_metric(L, C, alpha)
Bcubed_score
</code></pre>

<hr>
<h2 id='dpbbm_mc_iterations'>
dpbbm_mc_iterations
</h2><span id='topic+dpbbm_mc_iterations'></span>

<h3>Description</h3>

<p>This is the Markov Chain Monte Carlo iterations for DPBBM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dpbbm_mc_iterations(x, size.x, m = 1, max_iter = 2000, 
			a = 0.1, b = 1, tau = 1, 
			sig_alpha = 25/9, sig_beta = 25/9, 
			tau.method = "auto", debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dpbbm_mc_iterations_+3A_x">x</code></td>
<td>
<p> a matrix of k for clustering, referring to IP reads in m6A seq data 

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_size.x">size.x</code></td>
<td>
<p> a matrix of n for clustering, referring to the summation of IP reads and input reads in m6A seq data 

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_m">m</code></td>
<td>
<p> a value indicating the auxiliary clusters used in DPBBM

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_max_iter">max_iter</code></td>
<td>
<p> maximized iterations in DPBBM

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_a">a</code></td>
<td>
<p> Hyperparameter a for tau

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_b">b</code></td>
<td>
<p> Hyperparameter b for tau

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_tau">tau</code></td>
<td>
<p> Prior for tau

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_sig_alpha">sig_alpha</code></td>
<td>
<p> variation for parameter alpha of beta distribution

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_sig_beta">sig_beta</code></td>
<td>
<p> variation for parameter beta of beta distribution

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_tau.method">tau.method</code></td>
<td>
<p> tau.method should be set to &quot;auto&quot; or &quot;stable&quot;, refer to tau for detail description.

</p>
</td></tr>
<tr><td><code id="dpbbm_mc_iterations_+3A_debug">debug</code></td>
<td>
<p> whether DPBBM print the debug info or not. Default: FALSE

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Dirichlet Process based beta-binomial mixture model clustering
</p>


<h3>Value</h3>

<p>The function returns the cluster label withdrawn by DPBBM
</p>


<h3>Author(s)</h3>

<p> Lin Zhang, PhD  &lt;lin.zhang@cumt.edu.cn&gt;

</p>


<h3>References</h3>

<p>Reference coming soon!
</p>


<h3>Examples</h3>

<pre><code class='language-R'># generate a simulated dataset
set.seed(123455)
S &lt;- 4
G &lt;- 100
K &lt;- 3
nb_mu &lt;- 100
nb_size &lt;- 0.8
prob &lt;- c(1,1,1)
mat &lt;- bbm_data_generate(S=S,G=G,K=K,prob=prob,alpha_band=c(2,6),beta_band=c(2,6),
                     nb_mu=nb_mu,nb_size=nb_size, plotf = FALSE, max_cor=0.5) 
# check generated data
id &lt;- order(mat$gamma);
c &lt;- mat$gamma[id]
mat_ratio &lt;- (mat$k+1)/(mat$n+1);
heatmap(mat_ratio[id,], Rowv = NA, Colv = NA, scale="none", RowSideColors=as.character(c), 
        xlab = "4 samples", ylab="100 RNA methylation sites")

## Run the DPBBM result. This step takes a really long time.
## You are suggested to check the pre-prepared example for a quick start
F=system.file("extdata", "DPBBM_example.html", package="DPBBM")
browseURL(url=F)

## Alternatively
# cluster_label &lt;- dpbbm_mc_iterations(mat$k, mat$n)
# # Show the clustering result. 
# table(cluster_label)
# pie(table(mat$gamma))
# 
# # Compare the clustering result with the true clustering IDs.
# id &lt;- order(mat$gamma);
# c &lt;- cluster_label
# r &lt;- rainbow(3, start = 0, end = 0.3)
# mat_ratio &lt;- (mat$k+1)/(mat$n+1);
# heatmap(mat_ratio[id,], Rowv = NA, Colv = NA, scale="none",
#         RowSideColors = as.character(cluster_label[id]), 
#         margins = c(3,25))
</code></pre>

<hr>
<h2 id='DPBBM-package'>
DPBBM
</h2><span id='topic+DPBBM-package'></span><span id='topic+DPBBM'></span>

<h3>Description</h3>

<p>This package is developed for the beta-binomial mixture model based clustering
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> DPBBM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-06-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>References</h3>

<p>Coming soon!
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Please check the main function of the package
?dpbbm_mc_iterations 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
