<!DOCTYPE html><html><head><title>Help for package dQTG.seq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dQTG.seq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BSA'>
<p>BSA data</p></a></li>
<li><a href='#Dodata'><p>To perform method</p></a></li>
<li><a href='#dQTG.seq'><p>Title The function of dQTG.seq</p></a></li>
<li><a href='#Readdata1'><p>Title readdata function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A BSA Software for Detecting All Types of QTLs in BC, DH, RIL
and F2</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuan-Ming Zhang &lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Contact:</td>
<td>Yuan-Ming Zhang &lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The new (dQTG.seq1 and dQTG.seq2) and existing (SmoothLOD, G', deltaSNP and ED) bulked segregant analysis methods are used to identify various types of quantitative trait loci for complex traits via extreme phenotype individuals in bi-parental segregation populations (F2, backcross, doubled haploid and recombinant inbred line). The numbers of marker alleles in extreme low and high pools are used in existing methods to identify trait-related genes, while the numbers of marker alleles and genotypes in extreme low and high pools are used in the new methods to construct a new statistic Gw for identifying trait-related genes. dQTG-seq2 is feasible to identify extremely over-dominant and small-effect genes in F2. Li P, Li G, Zhang YW, Zuo JF, Liu JY, Zhang YM (2022, &lt;<a href="https://doi.org/10.1016%2Fj.xplc.2022.100319">doi:10.1016/j.xplc.2022.100319</a>&gt;).      </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>BB, data.table, doParallel, openxlsx, qtl, stringr,
writexl,vroom, parallel,foreach</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-21 09:51:30 UTC; Administrator</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Pei Li [aut],
  Yuan-Ming Zhang <a href="https://orcid.org/0000-0003-2317-2190"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-22 09:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='BSA'>
BSA data 
</h2><span id='topic+BSA'></span>

<h3>Description</h3>

<p>BSA format of F2 dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(BSA)
</code></pre>


<h3>Details</h3>

<p>Input file for dQTG.seq	function. 
</p>

<hr>
<h2 id='Dodata'>To perform method</h2><span id='topic+Dodata'></span>

<h3>Description</h3>

<p>To perform method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dodata(dir, calculatedata, chr, color1, CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dodata_+3A_dir">dir</code></td>
<td>
<p>the path of the output</p>
</td></tr>
<tr><td><code id="Dodata_+3A_calculatedata">calculatedata</code></td>
<td>
<p>the inputdata</p>
</td></tr>
<tr><td><code id="Dodata_+3A_chr">chr</code></td>
<td>
<p>chromosome</p>
</td></tr>
<tr><td><code id="Dodata_+3A_color1">color1</code></td>
<td>
<p>the color of the chromosome</p>
</td></tr>
<tr><td><code id="Dodata_+3A_clo">CLO</code></td>
<td>
<p>the numbers of CPU</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BSA)
dir&lt;- tempdir()
data.calculatedata&lt;-Readdata1(BSA)
Dodata(dir,calculatedata=data.calculatedata,chr="all",color1="blue",CLO=1)
</code></pre>

<hr>
<h2 id='dQTG.seq'>Title The function of dQTG.seq</h2><span id='topic+dQTG.seq'></span>

<h3>Description</h3>

<p>Title The function of dQTG.seq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dQTG.seq(dir, filegen, chr, color, CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dQTG.seq_+3A_dir">dir</code></td>
<td>
<p>the path of the output</p>
</td></tr>
<tr><td><code id="dQTG.seq_+3A_filegen">filegen</code></td>
<td>
<p>the input data</p>
</td></tr>
<tr><td><code id="dQTG.seq_+3A_chr">chr</code></td>
<td>
<p>the chromosome</p>
</td></tr>
<tr><td><code id="dQTG.seq_+3A_color">color</code></td>
<td>
<p>the color</p>
</td></tr>
<tr><td><code id="dQTG.seq_+3A_clo">CLO</code></td>
<td>
<p>the numbers of CPU</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BSA)
dQTG.seq(dir=tempdir(),filegen=BSA,chr="all",color="blue",CLO=1)
</code></pre>

<hr>
<h2 id='Readdata1'>Title readdata function</h2><span id='topic+Readdata1'></span>

<h3>Description</h3>

<p>Title readdata function
</p>
<p>Title readdata function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Readdata1(File)

Readdata1(File)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Readdata1_+3A_file">File</code></td>
<td>
<p>the input file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>
<p>list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BSA)
Readdata1(BSA) 
data(BSA)
Readdata1(BSA)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
