<!DOCTYPE html><html lang="en"><head><title>Help for package DR.SC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DR.SC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dlpfc151510'><p>A human dorsolateral prefrontal cortex data</p></a></li>
<li><a href='#DR.SC'><p>Joint dimension reduction and spatial clustering</p></a></li>
<li><a href='#DR.SC_fit'><p>Joint dimension reduction and spatial clustering</p></a></li>
<li><a href='#drscPlot'><p>tNSE or UMAP plot visualization</p></a></li>
<li><a href='#FindSVGs'><p>Find spatially variable genes</p></a></li>
<li><a href='#gendata_RNAExp'><p>Generate simulated data</p></a></li>
<li><a href='#getAdj'><p>Calculate the adjacency matrix given the spatial coordinates</p></a></li>
<li><a href='#getAdj_auto'><p>Calculate adjacency matrix by automatically choosing radius</p></a></li>
<li><a href='#getAdj_manual'><p>Calculate adjacency matrix by user-specified radius</p></a></li>
<li><a href='#getneighborhood_fast'><p>getneighborhood_fast</p></a></li>
<li><a href='#mbicPlot'><p>MBIC plot visualization</p></a></li>
<li><a href='#read10XVisium'><p>Read the spatial transcriptomics data measured on 10X Visium platform</p></a></li>
<li><a href='#readscRNAseq'><p>Read the scRNAseq data measured on scRNA sequencing platform</p></a></li>
<li><a href='#RunWPCA'><p>Run Weighted Principal Component Analysis</p></a></li>
<li><a href='#selectModel'><p>Select the number of clusters</p></a></li>
<li><a href='#sp_means_Rcpp'><p>Calculate column-wise or row-wise mean</p></a></li>
<li><a href='#sp_sums_Rcpp'><p>Calculate column-wise or row-wise sum</p></a></li>
<li><a href='#spatialPlotClusters'><p>Spatial coordinates plot visualization</p></a></li>
<li><a href='#topSVGs'><p>Return the top n SVGs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Joint Dimension Reduction and Spatial Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>3.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Liu [aut, cre],
  Yi Yang [aut],
  Jin Liu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Liu &lt;liuweideng@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Joint dimension reduction and spatial clustering is conducted for
    Single-cell RNA sequencing and spatial transcriptomics data, and more details can be referred to 
    Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi and Jin Liu. (2022) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkac219">doi:10.1093/nar/gkac219</a>&gt;. It is not only computationally efficient and scalable to the sample size increment, but also is capable of choosing the smoothness parameter and the number of clusters as well.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>parallel, spatstat.geom, R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm, irlba, cowplot, ggplot2, GiRaF, MASS, Matrix,
mclust, methods, purrr, S4Vectors, RColorBrewer, Rcpp (&ge;
1.0.5), Seurat, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/feiyoung/DR.SC">https://github.com/feiyoung/DR.SC</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/feiyoung/DR.SC/issues">https://github.com/feiyoung/DR.SC/issues</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-19 01:44:22 UTC; Liuxianju</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-19 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='dlpfc151510'>A human dorsolateral prefrontal cortex data</h2><span id='topic+dlpfc151510'></span>

<h3>Description</h3>

<p>A human dorsolateral prefrontal cortex dataset measured on the  Visium platform, which invcludes 4634 spots and 500 genes, a subset of raw dataset at https://github.com/LieberInstitute/spatialLIBD.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("dlpfc151510")
</code></pre>

<hr>
<h2 id='DR.SC'>Joint dimension reduction and spatial clustering</h2><span id='topic+DR.SC'></span><span id='topic+DR.SC.Seurat'></span>

<h3>Description</h3>

<p>Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
## S3 method for class 'Seurat'
DR.SC(seu, K, q=15,  platform= c('Visium', "ST", "Other_SRT", "scRNAseq"),...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DR.SC_+3A_seu">seu</code></td>
<td>
<p>an object of class &quot;Seurat&quot;. The details of this object are given under 'Details'.</p>
</td></tr>
<tr><td><code id="DR.SC_+3A_q">q</code></td>
<td>
<p>a positive integer, specify the number of latent features to be extracted, default as 15.</p>
</td></tr>
<tr><td><code id="DR.SC_+3A_k">K</code></td>
<td>
<p>a positive  integer or integer vector, specify the number of clusters. When <code>K</code> is a vector, it is automatically selected by MBIC criteria. Users can also use BIC and AIC  to select K or refine model with MBIC by argument <code>pen.const</code> in the function <a href="#topic+selectModel">selectModel</a>.</p>
</td></tr>
<tr><td><code id="DR.SC_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Visium&quot;. There are more platforms to be chosen, including (&quot;Visuim&quot;, &quot;ST&quot;, &quot;Other_SRT&quot;) and &quot;scRNAseq&quot;, where the first group means there are spatial coordinates information in the metadata of seu, named &quot;row&quot; and &quot;col&quot; and a Hidden Markov random field is used to model the unobserved class label using spatial coordinates (&quot;Other_SRT&quot; represents the other SRT platforms except for 'Visium' and 'ST'), and the other group &quot;scRNAseq&quot; means there is no spatial information in object seu and a multinomial model is used to model the unobserved class labels. The platform helps to calculate the adjacency matrix.</p>
</td></tr>
<tr><td><code id="DR.SC_+3A_...">...</code></td>
<td>
<p>Other arguments to pass into <a href="#topic+DR.SC_fit">DR.SC_fit</a> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>seu is an object named <code>Seurat</code>, thich can easily created by R package Seurat.
DR-SC  model can be applied to analyze both single cell RNA sequencing data and spatially resoved transcriptomics (SRT) data. 
</p>
<p>If the data are collected by the single cell RNA sequencing techonologies which means there is no spatial information in object seu then a multinomial model is used to model the unobserved class labels. 
</p>
<p>If the data is collected by the spatial transcriptomics technologies, then there are spatial coordinates information in the metadata of seu, named &quot;row&quot; and &quot;col&quot;. DR-SC model uses a Hidden Markov random field to model the spatial coordinates.  <code>DR.SC</code> supports different platforms of SRT data, such as 'Visium', 'ST' and any other platforms 'Other_SRT'.
</p>
<p>For lattice grids in ST platform (ST), the interior spot has four neighbors (left, right, up and down),the boundary spot has three neighbors, and the spot in the corner has two neighbors. For hexagon grids, such as spatial coordinate in 10X Visium platform (Visium), the interior spot has six neighbors. For the irregular coordinates in other platforms (Other_SRT), Euclidean distance is adopted to decide whether a spot is an neighbor of another spot. For example, if the Euclidean distance between spot A and B is less than a radius, then A is taken as the neighbourhood of B. See function <a href="#topic+getAdj">getAdj</a> for more details.
</p>


<h3>Value</h3>

<p>DR.SC returns a revised  <code>Seurat</code> object. There are two revisions in the seu: 1. the metadata is added a new column named <code>spatial.drsc.cluster</code> that represents the clustering results from DR-SC model, and the <code>Idents(seu)</code> is assigned with <code>spatial.drsc.cluster</code>. 2. a DimReduc object named <code>dr-sc</code> is added in the slot <code>reductions</code>, which represents the features extracted by DR-SC model.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4,platform="ST")
library(Seurat)
seu &lt;- NormalizeData(seu, verbose=FALSE)
# choose 100 highly variable features
# seu &lt;- FindVariableFeatures(seu, nfeatures = 100)
# maxIter = 2 is only used for illustration, and user can use default.
# seu1 &lt;- DR.SC(seu, K=4, platform = 'ST', maxIter=2,verbose=FALSE)

# choose spatially variable features (SVGs)
seu &lt;- FindSVGs(seu, nfeatures = 40, verbose=FALSE)
# use SVGs to fit DR.SC model
# maxIter = 2 is only used for illustration, and user can use default.
seu1 &lt;- DR.SC(seu, K=4,platform = 'ST', maxIter=2, verbose=TRUE)

</code></pre>

<hr>
<h2 id='DR.SC_fit'>Joint dimension reduction and spatial clustering</h2><span id='topic+DR.SC_fit'></span>

<h3>Description</h3>

<p>Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  DR.SC_fit(X, K, Adj_sp=NULL, q=15,
             error.heter= TRUE, beta_grid=seq(0.5, 5, by=0.5),
             maxIter=25, epsLogLik=1e-5, verbose=FALSE, maxIter_ICM=6,
             wpca.int=FALSE, int.model="EEE", approxPCA=FALSE, coreNum = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DR.SC_fit_+3A_x">X</code></td>
<td>
<p>a  sparse matrix with class <code>dgCMatrix</code> or <code>matrix</code>, specify the log-normalization gene expression matrix used for DR-SC model.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_k">K</code></td>
<td>
<p>a positive  integer allowing scalar or vector, specify the number of clusters in model fitting.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_adj_sp">Adj_sp</code></td>
<td>
<p>an optional sparse matrix with class <code>dgCMatrix</code>, specify the adjoint matrix used for DR-SC model. We provide this interface for those users  who would like to define the adjacency matrix by their own.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_q">q</code></td>
<td>
<p>a positive integer, specify the number of latent features to be extracted, default as 15. Usually, the choice of q is a trade-off between model complexity and fit to the data,  and depends on the goals of the analysis and the structure of the data. A higher value will result in a more complex model with a higher number of parameters, which may lead to overfitting and poor generalization performance. On the other hand, a lower value will result in a simpler model with fewer parameters, but may also lead to underfitting and a poorer fit to the data.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_error.heter">error.heter</code></td>
<td>
<p>an optional logical value, whether use the heterogenous error for DR-SC model, default as <code>TRUE</code>. If <code>error.heter=FALSE</code>, then the homogenuous error is used for probabilistic PCA model in DR-SC.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_beta_grid">beta_grid</code></td>
<td>
<p>an optional vector of positive value, the candidate set of the smoothing parameter to be searched by the grid-search optimization approach.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_maxiter">maxIter</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of EM.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_epsloglik">epsLogLik</code></td>
<td>
<p>an optional positive vlaue, tolerance vlaue of relative variation rate of the observed pseudo log-loglikelihood value, defualt as '1e-5'.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_verbose">verbose</code></td>
<td>
<p>an optional logical value, whether output the information of the ICM-EM algorithm.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_maxiter_icm">maxIter_ICM</code></td>
<td>
<p>an optional positive value, represents the maximum iterations of ICM.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_wpca.int">wpca.int</code></td>
<td>
<p>an optional logical value, means whether use the weighted PCA to obtain the initial values of loadings and other paramters, default as <code>FALSE</code> which means the ordinary PCA  is used.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_int.model">int.model</code></td>
<td>
<p>an optional string, specify which Gaussian mixture model is used in evaluting the initial values for DR-SC, default as &quot;EEE&quot;; and see Mclust for more models' names.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_approxpca">approxPCA</code></td>
<td>
<p>an optional logical value, whether use approximated PCA to speed up the computation for initial values.</p>
</td></tr>
<tr><td><code id="DR.SC_fit_+3A_corenum">coreNum</code></td>
<td>
<p>an optional positive integer, means the number of thread used in parallel computating, default as 5. If the length of K is one, then coreNum will be set as 1 automatically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>DR.SC_fit returns a <a href="base.html#topic+list">list</a> with class &quot;drscObject&quot; with the following three components:
</p>
<table role = "presentation">
<tr><td><code>Objdrsc</code></td>
<td>
<p>a list including the model fitting results, in which the number of elements is same as the length of K.</p>
</td></tr>
<tr><td><code>out_param</code></td>
<td>
<p>a numeric matrix used for model selection in MBIC.</p>
</td></tr>
<tr><td><code>K_set</code></td>
<td>
<p>a scalar or vector equal to input argument K.</p>
</td></tr>
</table>
<p>In addition, each element of &quot;Objdrsc&quot; is a list with the following comoponents:
</p>
<table role = "presentation">
<tr><td><code>cluster</code></td>
<td>
<p>inferred class labels</p>
</td></tr>
<tr><td><code>hZ</code></td>
<td>
<p>extracted latent features.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>estimated smoothing parameter</p>
</td></tr>
<tr><td><code>Mu</code></td>
<td>
<p>mean vectors of mixtures components.</p>
</td></tr>
<tr><td><code>Sigma</code></td>
<td>
<p>covariance matrix of mixtures components.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>estimated loading matrix</p>
</td></tr>
<tr><td><code>Lam_vec</code></td>
<td>
<p>estimated variance of errors in probabilistic PCA model</p>
</td></tr>
<tr><td><code>loglik</code></td>
<td>
<p>pseudo observed log-likelihood.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4)
library(Seurat)
seu &lt;- NormalizeData(seu, verbose=FALSE)
# choose 40 highly variable features using FindVariableFeatures in Seurat
# seu &lt;- FindVariableFeatures(seu, nfeatures = 40)
# or choose 40 spatailly variable features using FindSVGs in DR.SC
seu &lt;- FindSVGs(seu, nfeatures = 40, verbose=FALSE)
# users define the adjacency matrix
Adj_sp &lt;- getAdj(seu, platform = 'ST')
if(class(seu@assays$RNA)=="Assay5"){
 var.features &lt;- seu@assays$RNA@meta.data$var.features
 var.features  &lt;- var.features[!is.na(var.features )]
 dat &lt;- GetAssayData(seu, assay = "RNA", slot='data')
 X &lt;- Matrix::t(dat[var.features,])
}else{
 var.features &lt;- seu@assays$RNA@var.features
 X &lt;- Matrix::t(seu[["RNA"]]@data[var.features,])
}


# maxIter = 2 is only used for illustration, and user can use default.
drscList &lt;- DR.SC_fit(X,Adj_sp=Adj_sp, K=4, maxIter=2, verbose=TRUE)

</code></pre>

<hr>
<h2 id='drscPlot'>tNSE or UMAP plot visualization</h2><span id='topic+drscPlot'></span>

<h3>Description</h3>

<p>Intuitive way of visualizing how cell types changes across the embeddings obatined by DR-SC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  drscPlot(seu, dims=1:5, visu.method='tSNE',...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="drscPlot_+3A_seu">seu</code></td>
<td>
<p>an object of class &quot;Seurat&quot; obtained by <a href="#topic+DR.SC">DR.SC</a>.</p>
</td></tr>
<tr><td><code id="drscPlot_+3A_dims">dims</code></td>
<td>
<p>a positive integer to specify the number of latent features for visualiztion.</p>
</td></tr>
<tr><td><code id="drscPlot_+3A_visu.method">visu.method</code></td>
<td>
<p>a string including 'tSNE' or &quot;UMAP&quot;.</p>
</td></tr>
<tr><td><code id="drscPlot_+3A_...">...</code></td>
<td>
<p>Other arguments passing to DimPlot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4)
library(Seurat)
seu &lt;- NormalizeData(seu)
# choose spatially variable features
seu &lt;- FindSVGs(seu)

# use SVGs to fit DR.SC model
# maxIter = 2 is only used for illustration, and user can use default.
seu1 &lt;- DR.SC(seu, K=4,platform = 'ST', maxIter = 2,verbose=FALSE)
drscPlot(seu1)
</code></pre>

<hr>
<h2 id='FindSVGs'>Find spatially variable genes</h2><span id='topic+FindSVGs'></span>

<h3>Description</h3>

<p>Identifies features that have spatially variation along spots using SPARK-X.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  FindSVGs(seu, nfeatures=2000, covariates=NULL,
             preHVGs=5000,num_core=1, verbose=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FindSVGs_+3A_seu">seu</code></td>
<td>
<p>an object of class &quot;Seurat&quot;.</p>
</td></tr>
<tr><td><code id="FindSVGs_+3A_nfeatures">nfeatures</code></td>
<td>
<p>a positive integer, means how many spatially variable genes to be chosen. If there are less than 2000 features in seu, then all features are identified.</p>
</td></tr>
<tr><td><code id="FindSVGs_+3A_covariates">covariates</code></td>
<td>
<p>a covariate matrix named control variable matrix whose number of rows is equal to the number of columns of seu.</p>
</td></tr>
<tr><td><code id="FindSVGs_+3A_prehvgs">preHVGs</code></td>
<td>
<p>a positive integer, the number of highly variable genes selected for speeding up computation of SPARK-X  in selecting spatially variable features. </p>
</td></tr>
<tr><td><code id="FindSVGs_+3A_num_core">num_core</code></td>
<td>
<p>an optional positive integer, specify the cores used for identifying the SVGs in parallel.</p>
</td></tr>
<tr><td><code id="FindSVGs_+3A_verbose">verbose</code></td>
<td>
<p>an optional logical value, whether output the related information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a revised Seurat object by adding three columns named &quot;is.SVGs&quot;, &quot;order.SVGs&quot; and &quot;adjusted.pval.SVGs&quot; in the meta.features  of default Assay.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>References</h3>

<p>Zhu, J., Sun, S., Zhou, X.: Spark-x: non-parametric modeling enables scalable and robust detection of spatialexpression patterns for large spatial transcriptomic studies. Genome Biology 22(1), 1-25 (2021)
</p>


<h3>See Also</h3>

<p><a href="#topic+topSVGs">topSVGs</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  seu&lt;-gendata_RNAExp(height=20, width=20,p=200, K=4)
  seu&lt;-FindSVGs(seu, nfeatures=100)
  topSVGs(seu)

</code></pre>

<hr>
<h2 id='gendata_RNAExp'>Generate simulated data</h2><span id='topic+gendata_RNAExp'></span>

<h3>Description</h3>

<p>Generate simulated spatial transcriptomics data or scRNAseq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  gendata_RNAExp(height=30, width=30, platform="ST", p =100, q=10, K=7, 
                            G=4,sigma2=1, tau=8, seed=1, view=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gendata_RNAExp_+3A_height">height</code>, <code id="gendata_RNAExp_+3A_width">width</code></td>
<td>
<p>Height and width of lattice grids for generating spatial coordinates. n=height * width cells for scRNAseq data</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_platform">platform</code></td>
<td>
<p>set the platform for the simulated data, only support 'ST' and 'scRNAseq'.</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_p">p</code></td>
<td>
<p>number of genes to generate.</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_q">q</code></td>
<td>
<p>number of true latent features to generate gene expression</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_k">K</code></td>
<td>
<p>number of clusters (cell types).</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_seed">seed</code></td>
<td>
<p>random seed for generate data</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_g">G</code></td>
<td>
<p>the number of neighbors. The latter must be one of G = 4 or G = 8, which respectively correspond to a first order and a second order dependency structure. By default, G = 4.</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_sigma2">sigma2</code></td>
<td>
<p>Variance of error term in probabilitic PCA model.</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_tau">tau</code></td>
<td>
<p>a positive factor of mixture mean values.</p>
</td></tr>
<tr><td><code id="gendata_RNAExp_+3A_view">view</code></td>
<td>
<p>Logical value indicating whether the draw should be printed. Do not display the optional borders.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a &quot;Seurat&quot; object. If <code>platform="ST"</code>, then the metadata of this Seurat object will include two columns with names &quot;row&quot; and &quot;col&quot; which are the spatial coordinates; If <code>platform="scRNAseq"</code>, then the metadata of this Seurat object will not have them.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
seu &lt;- gendata_RNAExp(height=20, width=20,p=200, K=4)
seu

## generate scRNAseq data
seu &lt;- gendata_RNAExp(height=20, width=20, platform="scRNAseq", p=100, K=4)
seu
</code></pre>

<hr>
<h2 id='getAdj'>Calculate the adjacency matrix given the spatial coordinates</h2><span id='topic+getAdj'></span><span id='topic+getAdj.Seurat'></span>

<h3>Description</h3>

<p>Calculate the adjacency matrix for the spatial transcriptomics data measured on 10X Visium, ST or other platforms as a Seurat object.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
  ## S3 method for class 'Seurat'
getAdj(obj, platform = c('Visium', "ST", "Other_SRT"), ...) 
  
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAdj_+3A_obj">obj</code></td>
<td>
<p>an object with class &quot;Seurat&quot;, there are spatial coordinates information in the metadata of obj, named &quot;row&quot; and &quot;col&quot;, where first column is x-axis coordinate, the second column is y-axis coordinate. <a href="#topic+getAdj_manual">getAdj_manual</a> and <a href="#topic+getAdj_auto">getAdj_auto</a>  supports multi-dimensional spatial coordinates with a matrix as input.</p>
</td></tr>
<tr><td><code id="getAdj_+3A_platform">platform</code></td>
<td>
<p>a string, specify the platform of the provided data, default as &quot;Visium&quot;. There are more platforms to be chosen, including (&quot;Visuim&quot;, &quot;ST&quot;, &quot;Other_SRT&quot;), which means there are spatial coordinates information in the metadata of obj, named &quot;row&quot; and &quot;col&quot;. The platform helps to calculate the adjacency matrix by defining the neighborhoods.</p>
</td></tr>
<tr><td><code id="getAdj_+3A_...">...</code></td>
<td>
<p>Other arguments to pass into <a href="#topic+getAdj_auto">getAdj_auto</a> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For lattice grids in ST platform (ST), the interior spot has four neighbors (left, right, up and down),the boundary spot has three neighbors, and the spot in the corner has two neighbors. For hexagon grids, such as spatial coordinate in 10X Visium platform (Visium), the interior spot has six neighbors. For the irregular coordinates in other platforms (Other_SRT), Euclidean distance is adopted to decide whether a spot is an neighbor of another spot. For example, if the Euclidean distance between spot A and B is less than a radius, then A is taken as the neighbourhood of B. See functions <a href="#topic+getAdj_auto">getAdj_auto</a> and <a href="#topic+getAdj_manual">getAdj_manual</a> for more details.
</p>


<h3>Value</h3>

<p>Return a <code>dgCMatrix</code> object recording the information of neighborhoods about each spot.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+getAdj_auto">getAdj_auto</a>, <a href="#topic+getAdj_manual">getAdj_manual</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## S3 method for class "Seurat"
  seu &lt;- gendata_RNAExp(height=20, width=20,p=200, K=4)
  Adj_sp &lt;- getAdj(seu, platform = 'ST')
</code></pre>

<hr>
<h2 id='getAdj_auto'>Calculate adjacency matrix by automatically choosing radius</h2><span id='topic+getAdj_auto'></span>

<h3>Description</h3>

<p>an efficient function to find the radius by bi-section method ,  then find neighbors based on the matrix of position, which ensures that each spot has approximately lower.med~upper.med neighbors in the sense of median.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAdj_auto(pos, lower.med=4, upper.med=6, radius.upper= NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAdj_auto_+3A_pos">pos</code></td>
<td>
<p>a n-by-2 matrix of position.</p>
</td></tr>
<tr><td><code id="getAdj_auto_+3A_lower.med">lower.med</code></td>
<td>
<p>an integer, the lower bound of median number of neighbors among all spots.</p>
</td></tr>
<tr><td><code id="getAdj_auto_+3A_upper.med">upper.med</code></td>
<td>
<p>an integer, the upper bound of median number of neighbors among all spots.</p>
</td></tr>
<tr><td><code id="getAdj_auto_+3A_radius.upper">radius.upper</code></td>
<td>
<p>a real, the upper bound of radius, default as NULL. If <code>radius.upper= NULL</code>, the upper bound is automatically determined by algorithm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse adjacency matrix containing the neighbourhood.
</p>


<h3>See Also</h3>

<p><a href="#topic+getAdj">getAdj</a>, <a href="#topic+getAdj_manual">getAdj_manual</a>.
</p>

<hr>
<h2 id='getAdj_manual'>Calculate adjacency matrix by user-specified radius</h2><span id='topic+getAdj_manual'></span>

<h3>Description</h3>

<p>an efficient function to find the neighbors based on the matrix of position and a pre-defined radius.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAdj_manual(pos, radius)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getAdj_manual_+3A_pos">pos</code></td>
<td>
<p>is a n-by-d matrix of position, where n is the number of spots, and d is the dimension of coordinates.</p>
</td></tr>
<tr><td><code id="getAdj_manual_+3A_radius">radius</code></td>
<td>
<p>is a threashold of Euclidean distance to decide whether a spot is an neighborhood of another spot. For example, if the Euclidean distance between spot A and B is less than radius, then A is taken as the neighbourhood of B.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse matrix containing the neighbourhood
</p>


<h3>See Also</h3>

<p><a href="#topic+getAdj_auto">getAdj_auto</a>, <a href="#topic+getAdj">getAdj</a>.
</p>

<hr>
<h2 id='getneighborhood_fast'>getneighborhood_fast</h2><span id='topic+getneighborhood_fast'></span>

<h3>Description</h3>

<p>an efficient function to find the neighborhood based on the matrix of position and a pre-defined cutoff
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getneighborhood_fast(x, radius)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getneighborhood_fast_+3A_x">x</code></td>
<td>
<p>is a n-by-2 matrix of position.</p>
</td></tr>
<tr><td><code id="getneighborhood_fast_+3A_radius">radius</code></td>
<td>
<p>is a threashold of Euclidean distance to decide whether a spot is an neighborhood of another spot. For example, if the Euclidean distance between spot A and B is less than cutoff, then A is taken as the neighbourhood of B.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sparse matrix containing the neighbourhood
</p>

<hr>
<h2 id='mbicPlot'>MBIC plot visualization</h2><span id='topic+mbicPlot'></span>

<h3>Description</h3>

<p>Intuitive way of visualizing how modified BIC values changes across different number of clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  mbicPlot(seu, criteria="MBIC")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mbicPlot_+3A_seu">seu</code></td>
<td>
<p>an object of class <code>Seurat</code> revised by <a href="#topic+DR.SC">DR.SC</a> with argument <code>K=NULL</code>.</p>
</td></tr>
<tr><td><code id="mbicPlot_+3A_criteria">criteria</code></td>
<td>
<p>a string specifying the information criteria such as AIC, BIC and MBIC to be plotted, default as MBIC.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  ## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
  seu &lt;- gendata_RNAExp(height=20, width=20,p=100, K=4)
  library(Seurat)
  seu &lt;- NormalizeData(seu)
  # choose spatially variable features
  seu &lt;- FindSVGs(seu)
  ## Just for illustrating the usage of mbicPlot
  seu[["RNA"]]@misc[['icMat']] &lt;- data.frame(K=2:5, MBIC=c(105, 101, 99, 108))
  mbicPlot(seu)

</code></pre>

<hr>
<h2 id='read10XVisium'>Read the spatial transcriptomics data measured on 10X Visium platform</h2><span id='topic+read10XVisium'></span>

<h3>Description</h3>

<p>Read the spatial transcriptomics data measured on 10X Visium platform  as a Seurat object, where the spatial coordinates are saved  in the metadata, named &quot;row&quot; and &quot;col&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  read10XVisium(dirname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read10XVisium_+3A_dirname">dirname</code></td>
<td>
<p>A string, the dictory of Visium datasets</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a Seurat object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  ## set your file directory, then read it.
  data_name &lt;- "D/HCC"
  HCC1 &lt;- read10XVisium(data_name)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='readscRNAseq'>Read the scRNAseq data measured on scRNA sequencing platform</h2><span id='topic+readscRNAseq'></span>

<h3>Description</h3>

<p>Read the single cell RNA sequencing data measured on scRNA sequencing platform  as a Seurat object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  readscRNAseq(mtx, cells, features, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readscRNAseq_+3A_mtx">mtx</code></td>
<td>
<p>a string, ame or remote URL of the mtx file</p>
</td></tr>
<tr><td><code id="readscRNAseq_+3A_cells">cells</code></td>
<td>
<p>a string, Name or remote URL of the cells/barcodes file</p>
</td></tr>
<tr><td><code id="readscRNAseq_+3A_features">features</code></td>
<td>
<p>a string, Name or remote URL of the features/genes file</p>
</td></tr>
<tr><td><code id="readscRNAseq_+3A_...">...</code></td>
<td>
<p>the arguments passing to ReadMtx</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a Seurat object including expression matrix.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  ### set the file directory, then read it.
  seu &lt;- readscRNAseq(mtx="GSM3755564_16_Liver_Treg_matrix.mtx.gz",
                       features='GSM3755564_16_Liver_Treg_genes.tsv.gz',
                       cells='GSM3755564_16_Liver_Treg_barcodes.tsv.gz' )
  seu
  
## End(Not run)
</code></pre>

<hr>
<h2 id='RunWPCA'>Run Weighted Principal Component Analysis</h2><span id='topic+RunWPCA'></span>

<h3>Description</h3>

<p>Run a weighted PCA dimensionality reduction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  RunWPCA(object, q=15)
  ### S3 method for class "Seurat"
  ## RunWPCA(object, q=15)
  
  ### S3 method for class "matrix"
  ## RunWPCA(object, q=15)
  
  ### S3 method for class "dgCMatrix"
  ## RunWPCA(object, q=15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RunWPCA_+3A_object">object</code></td>
<td>
<p>an object named &quot;Seurat&quot;, &quot;maxtrix&quot; or &quot;dgCMatrix&quot;. The object of class &quot;Seurat&quot; must include slot &quot;scale.data&quot;.</p>
</td></tr>
<tr><td><code id="RunWPCA_+3A_q">q</code></td>
<td>
<p>an optional positive integer, specify the number of features to be extracted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>For Seurat object, return a Seurat object. For objcet &quot;matrix&quot; and &quot;dgCMatrix&quot;, return a object &quot;matrix&quot; with rownames same as the colnames of <code>X</code>, and colnames &quot;WPCA1&quot; to &quot;WPCAq&quot;.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>Bai, J. and Liao, Y. (2017). Inferences in panel data with interactive effects using large
covariance matrices. Journal of Econometrics, 200(1):59–78.
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  library(Seurat)
  seu &lt;- gendata_RNAExp(height=20, width=20,p=100, K=4)
  ## log-normalization
  seu &lt;- NormalizeData(seu)
  ##
  seu &lt;- FindVariableFeatures(seu, nfeatures=80)
  ## Scale
  seu &lt;- ScaleData(seu)
  ## Run WPCA
  seu &lt;- RunWPCA(seu)
  seu
  ## Run tSNE based on wpca
  seu &lt;- RunTSNE(seu, reduction='wpca')
  seu
  ## Find SVGs
  seu &lt;- FindSVGs(seu, nfeatures=80)
  (genes &lt;- topSVGs(seu, ntop=10))
  Idents(seu) &lt;- factor(paste0("cluster", seu$true_clusters), levels=paste0("cluster",1:4))
  RidgePlot(seu, features = genes[1:2], ncol = 2)
  FeaturePlot(seu, features = genes[1:2], reduction = 'tsne' ,ncol=2)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='selectModel'>Select the number of clusters</h2><span id='topic+selectModel'></span><span id='topic+selectModel.drscObject'></span><span id='topic+selectModel.Seurat'></span>

<h3>Description</h3>

<p>Select the number of clusters by specified criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  
  selectModel(obj, criteria = 'MBIC', pen.const=1)
  ## S3 method for class 'drscObject'
selectModel(obj, criteria = 'MBIC', pen.const=1)
  ## S3 method for class 'Seurat'
selectModel(obj, criteria = 'MBIC', pen.const=1)
  
  
</code></pre>


<h3>Arguments</h3>

<p>S
</p>
<table role = "presentation">
<tr><td><code id="selectModel_+3A_obj">obj</code></td>
<td>
<p>an object with class <code>Seurat</code> by <a href="#topic+DR.SC">DR.SC</a> or class <code>drscObject</code> by <a href="#topic+DR.SC_fit">DR.SC_fit</a>.</p>
</td></tr>
<tr><td><code id="selectModel_+3A_criteria">criteria</code></td>
<td>
<p>a string, specify the criteria used for selecting the number of clusters, supporting &quot;MBIC&quot;, &quot;BIC&quot; and &quot;AIC&quot;.</p>
</td></tr>
<tr><td><code id="selectModel_+3A_pen.const">pen.const</code></td>
<td>
<p>an optional positive value, the adjusted constant used in the MBIC criteria. It usually takes value between 0.1 to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For S3 method of <code>Seurat</code>, it return a  revised &quot;Seurat&quot; object with updated <code>Idents(seu)</code>, <code>spatial.drsc.cluster</code> in the metadata and DimReduc object named <code>dr-sc</code> in the slot <code>reductions</code>. For S3 method of <code>drscObject</code>, it returns a list with the following components:
</p>
<table role = "presentation">
<tr><td><code>bestK</code></td>
<td>
<p>the selected number of clusters.</p>
</td></tr>
<tr><td><code>cluster</code></td>
<td>
<p>inferred class labels</p>
</td></tr>
<tr><td><code>hZ</code></td>
<td>
<p>extracted latent features.</p>
</td></tr>
<tr><td><code>icMat</code></td>
<td>
<p>a numeric matrix including the criteria value for each number of clusters K.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+DR.SC">DR.SC</a>, <a href="#topic+DR.SC_fit">DR.SC_fit</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4)
library(Seurat)
seu &lt;- NormalizeData(seu, verbose=FALSE)
# or choose 40 spatailly variable features using FindSVGs in DR.SC
seu &lt;- FindSVGs(seu, nfeatures = 40, verbose=FALSE)
# users define the adjacency matrix
Adj_sp &lt;- getAdj(seu, platform = 'ST')
dat &lt;- GetAssayData(seu, assay = "RNA", slot='data')
X &lt;- Matrix::t(dat)
# maxIter = 2 is only used for illustration, and user can use default.
drscList &lt;- DR.SC_fit(X,Adj_sp=Adj_sp ,K=4, maxIter=2, verbose=TRUE)
drsc1 &lt;- selectModel(drscList)
str(drsc1)
</code></pre>

<hr>
<h2 id='sp_means_Rcpp'>Calculate column-wise or row-wise mean</h2><span id='topic+sp_means_Rcpp'></span>

<h3>Description</h3>

<p>Calculate column-wise or row-wise mean
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp_means_Rcpp(sp_data, rowMeans = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sp_means_Rcpp_+3A_sp_data">sp_data</code></td>
<td>
<p>A sparse matrix</p>
</td></tr>
<tr><td><code id="sp_means_Rcpp_+3A_rowmeans">rowMeans</code></td>
<td>
<p>A boolean value, whether to calculate row-wise mean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x 1 or p x 1 matrix
</p>

<hr>
<h2 id='sp_sums_Rcpp'>Calculate column-wise or row-wise sum</h2><span id='topic+sp_sums_Rcpp'></span>

<h3>Description</h3>

<p>Calculate column-wise or row-wise sum
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sp_sums_Rcpp(sp_data, rowSums = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sp_sums_Rcpp_+3A_sp_data">sp_data</code></td>
<td>
<p>A sparse matrix</p>
</td></tr>
<tr><td><code id="sp_sums_Rcpp_+3A_rowsums">rowSums</code></td>
<td>
<p>A boolean value, whether to calculate row-wise sum</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A n x 1 or p x 1 matrix
</p>

<hr>
<h2 id='spatialPlotClusters'>Spatial coordinates plot visualization</h2><span id='topic+spatialPlotClusters'></span>

<h3>Description</h3>

<p>Intuitive way of visualizing how cell types changes across the spatial locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  spatialPlotClusters(seu)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spatialPlotClusters_+3A_seu">seu</code></td>
<td>
<p>an object of class &quot;Seurat&quot; obtained by <a href="#topic+DR.SC">DR.SC</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a ggplot2 object.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
    seu &lt;- gendata_RNAExp(height=10, width=10,p=50, K=4)
    library(Seurat)
    seu &lt;- NormalizeData(seu)
    # choose spatially variable features using Seurat
    seu &lt;- FindSVGs(seu)
    # use SVGs to fit DR.SC model
    # maxIter = 2 is only used for illustration, and user can use default.
    seu1 &lt;- DR.SC(seu, K=4,platform = 'ST', maxIter=2,verbose=FALSE)
    spatialPlotClusters(seu1)
</code></pre>

<hr>
<h2 id='topSVGs'>Return the top n SVGs</h2><span id='topic+topSVGs'></span>

<h3>Description</h3>

<p>Return top n spatially variable genes given a Seurat object performed by <a href="#topic+FindSVGs">FindSVGs</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  topSVGs(seu, ntop=5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="topSVGs_+3A_seu">seu</code></td>
<td>
<p>an object of class &quot;Seurat&quot;.</p>
</td></tr>
<tr><td><code id="topSVGs_+3A_ntop">ntop</code></td>
<td>
<p>an optional positive integer, means how many spatially variable genes to access. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>return a <a href="base.html#topic+character">character</a> vector including the names of SVGs.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p>None
</p>


<h3>See Also</h3>

<p><a href="#topic+topSVGs">topSVGs</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  seu &lt;- gendata_RNAExp(height=20, width=20,p=200, K=4)
  seu &lt;- FindSVGs(seu, nfeatures=100, verbose=FALSE)
  (genes &lt;- topSVGs(seu, ntop=10))
  
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
