<!DOCTYPE html><html lang="en"><head><title>Help for package DrImpute</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DrImpute}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DrImpute'><p>Imputing dropout events in single-cell RNA-sequencing data.</p></a></li>
<li><a href='#exdata'><p>Usoskin data</p></a></li>
<li><a href='#getCls'><p>get base clustering results using SC3 based clustering methods.</p></a></li>
<li><a href='#preprocessSC'><p>A function for preprocessing gene expression matrix.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-7-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Imputing Dropout Events in Single-Cell RNA-Sequencing Data</td>
</tr>
<tr>
<td>Description:</td>
<td>R codes for imputing dropout events. Many statistical methods in cell type identification, visualization and lineage reconstruction do not account for dropout events ('PCAreduce', 'SC3', 'PCA', 't-SNE', 'Monocle', 'TSCAN', etc). 'DrImpute' can improve the performance of such software by imputing dropout events.</td>
</tr>
<tr>
<td>Author:</td>
<td>Il-Youp Kwak with contributions from Wuming Gong</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Il-Youp Kwak &lt;ilyoup.kwak@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, devtools, roxygen2, irlba</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ikwak2/DrImpute">https://github.com/ikwak2/DrImpute</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-07-15 12:47:22 UTC; ikwak</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-07-15 21:00:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='DrImpute'>Imputing dropout events in single-cell RNA-sequencing data.</h2><span id='topic+DrImpute'></span>

<h3>Description</h3>

<p>Imputing dropout events in single-cell RNA-sequencing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DrImpute(X, ks = 10:15, dists = c("spearman", "pearson"), method = "mean",
  cls = NULL, seed = 1, zerop = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DrImpute_+3A_x">X</code></td>
<td>
<p>Gene expression matrix (gene by cell).</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_ks">ks</code></td>
<td>
<p>Number of cell clustering groups. Default set to ks = 10:15.</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_dists">dists</code></td>
<td>
<p>Distribution matrices to use. Default is set to c(&quot;spearman&quot;, &quot;pearson&quot;). &quot;eucleadian&quot; can be added as well.</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_method">method</code></td>
<td>
<p>Use &quot;mean&quot; for mean imputation, &quot;med&quot; for median imputation.</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_cls">cls</code></td>
<td>
<p>User can manually provide clustering information. Using different base clusterings. each row represent different clusterings. each column represent each cell.</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_seed">seed</code></td>
<td>
<p>User can provide a seed.</p>
</td></tr>
<tr><td><code id="DrImpute_+3A_zerop">zerop</code></td>
<td>
<p>zero percentage of resulting imputation is at least zerop.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Imputed Gene expression matrix (gene by cell).
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdata)
exdata &lt;- preprocessSC(exdata)
exdata &lt;- exdata[1:3000, 1:80]
logdat &lt;- log(exdata+1)
cls &lt;- getCls(logdat)
logdat_imp &lt;- DrImpute(logdat, cls = cls)

</code></pre>

<hr>
<h2 id='exdata'>Usoskin data</h2><span id='topic+exdata'></span>

<h3>Description</h3>

<p>This data set is subset from Usoskin et al. (2015). Original data is
RNA-seq data on 799 cells dissected from the mouse lumbar dorsal root
ganglion distributed over a total of nine 96-well plates. We randomly
selected 150 cells from the data. 
</p>
<p>Column names indicate four different cell types, NF, NP, TH, and PEP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exdata)
</code></pre>


<h3>References</h3>

<p>Usoskin D et al. Unbiased classification of sensory neuron types by large-scale
single-cell RNA sequencing. Nature Neuroscience. Nature Research,2015;18:145-53.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exdata)
exdata &lt;- preprocessSC(exdata)
</code></pre>

<hr>
<h2 id='getCls'>get base clustering results using SC3 based clustering methods.</h2><span id='topic+getCls'></span>

<h3>Description</h3>

<p>Similarity matrix constructed using &quot;pearson&quot;, &quot;spearman&quot; or &quot;euclidean&quot;. K-means clustering is performed on first few number of principal components of similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCls(X, ks = 10:15, dists = c("spearman", "pearson"),
  dim.reduc.prop = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCls_+3A_x">X</code></td>
<td>
<p>Log transformed gene expression matrix (Gene by Cell).</p>
</td></tr>
<tr><td><code id="getCls_+3A_ks">ks</code></td>
<td>
<p>Number of cell clustering groups. Default set to ks = 10:15.</p>
</td></tr>
<tr><td><code id="getCls_+3A_dists">dists</code></td>
<td>
<p>Distribution matrices to use. Default is set to c(&quot;spearman&quot;, &quot;pearson&quot;). &quot;euclidean&quot; can be added as well.</p>
</td></tr>
<tr><td><code id="getCls_+3A_dim.reduc.prop">dim.reduc.prop</code></td>
<td>
<p>Proportion of principal components to use for K-means clustering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix object, Each row represent different clustering results.
</p>


<h3>Author(s)</h3>

<p>Il-Youp Kwak
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DrImpute">DrImpute</a></code>  <code><a href="#topic+preprocessSC">preprocessSC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdata)
exdata &lt;- preprocessSC(exdata)
exdata &lt;- exdata[1:3000, 1:80]
logdat &lt;- log(exdata+1)
cls &lt;- getCls(logdat)

</code></pre>

<hr>
<h2 id='preprocessSC'>A function for preprocessing gene expression matrix.</h2><span id='topic+preprocessSC'></span>

<h3>Description</h3>

<p>Preprocess gene expression data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocessSC(X, min.expressed.gene = 0, min.expressed.cell = 2,
  max.expressed.ratio = 1, normalize.by.size.effect = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocessSC_+3A_x">X</code></td>
<td>
<p>Gene expression matrix (Gene by Cell).</p>
</td></tr>
<tr><td><code id="preprocessSC_+3A_min.expressed.gene">min.expressed.gene</code></td>
<td>
<p>Cell level filtering criteria. For a given cell, if the number of expressed genes are less than min.expressed.gene, we filter it out.</p>
</td></tr>
<tr><td><code id="preprocessSC_+3A_min.expressed.cell">min.expressed.cell</code></td>
<td>
<p>Gene level filtering criteria. For a given gene, if the number of expressed cells are less than min.expressed.cell, we filter it out.</p>
</td></tr>
<tr><td><code id="preprocessSC_+3A_max.expressed.ratio">max.expressed.ratio</code></td>
<td>
<p>Gene level filtering criteria. For a given gene, if the ratio of expressed cells are larger than max.expressed.ratio, we filter it out.</p>
</td></tr>
<tr><td><code id="preprocessSC_+3A_normalize.by.size.effect">normalize.by.size.effect</code></td>
<td>
<p>Normaize using size factor.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Filtered gene expression matrix
</p>


<h3>Author(s)</h3>

<p>Wuming Gong
</p>


<h3>References</h3>

<p>Il-Youp Kwak, Wuming Gong, Kaoko Koyano-Nakagawa and Daniel J. Garry (2017+)
DrImpute: Imputing dropout eveents in single cell RNA sequencing data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DrImpute">DrImpute</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(exdata)
exdata &lt;- preprocessSC(exdata)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
