<!DOCTYPE html><html><head><title>Help for package driveR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {driveR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#driveR'><p>driveR: An R Package for Prioritizing Cancer Driver Genes Using Genomics Data</p></a></li>
<li><a href='#create_features_df'><p>Create Data Frame of Features for Driver Gene Prioritization</p></a></li>
<li><a href='#create_gene_level_scna_df'><p>Create Gene-level SCNA Data Frame</p></a></li>
<li><a href='#create_noncoding_impact_score_df'><p>Create Non-coding Impact Score Data Frame</p></a></li>
<li><a href='#create_SCNA_score_df'><p>Create SCNA Score Data Frame</p></a></li>
<li><a href='#determine_double_hit_genes'><p>Determine Double-Hit Genes</p></a></li>
<li><a href='#determine_hotspot_genes'><p>Determine Hotspot Containing Genes</p></a></li>
<li><a href='#example_cohort_features_table'><p>Example Cohort-level Features Table for Driver Prioritization</p></a></li>
<li><a href='#example_cohort_scna_table'><p>Example Cohort-level Somatic Copy Number Alteration Table</p></a></li>
<li><a href='#example_features_table'><p>Example Features Table for Driver Prioritization</p></a></li>
<li><a href='#example_scna_table'><p>Example Somatic Copy Number Alteration Table</p></a></li>
<li><a href='#KEGG_cancer_pathways_descriptions'><p>KEGG &quot;Pathways in cancer&quot;-related Pathways - Descriptions</p></a></li>
<li><a href='#MTL_submodel_descriptions'><p>MTL Sub-model Descriptions</p></a></li>
<li><a href='#predict_coding_impact'><p>Create Coding Impact Meta-prediction Score Data Frame</p></a></li>
<li><a href='#prioritize_driver_genes'><p>Prioritize Cancer Driver Genes</p></a></li>
<li><a href='#specific_thresholds'><p>Tumor type specific probability thresholds</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Prioritizing Cancer Driver Genes Using Genomics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ege Ulgen &lt;egeulgen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Cancer genomes contain large numbers of somatic alterations but few
    genes drive tumor development. Identifying cancer driver genes is critical 
    for precision oncology. Most of current approaches either identify driver 
    genes based on mutational recurrence or using estimated scores predicting 
    the functional consequences of mutations. 'driveR' is a tool for 
    personalized or batch analysis of genomic data for driver gene prioritization 
    by combining genomic information and prior biological knowledge. As features, 
    'driveR' uses coding impact metaprediction scores, non-coding impact scores, 
    somatic copy number alteration scores, hotspot gene/double-hit gene 
    condition, 'phenolyzer' gene scores and memberships to cancer-related KEGG 
    pathways. It uses these features to estimate cancer-type-specific 
    probability for each gene of being a cancer driver using the related task of 
    a multi-task learning classification model. The method is described in detail 
    in Ulgen E, Sezerman OU. 2021. driveR: driveR: a novel method for 
    prioritizing cancer driver genes using somatic genomics data. BMC 
    Bioinformatics &lt;<a href="https://doi.org/10.1186%2Fs12859-021-04203-7">doi:10.1186/s12859-021-04203-7</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://egeulgen.github.io/driveR/">https://egeulgen.github.io/driveR/</a>,
<a href="https://github.com/egeulgen/driveR/">https://github.com/egeulgen/driveR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/egeulgen/driveR/issues">https://github.com/egeulgen/driveR/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, randomForest, GenomicRanges, GenomeInfoDb,
GenomicFeatures, TxDb.Hsapiens.UCSC.hg19.knownGene,
TxDb.Hsapiens.UCSC.hg38.knownGene, S4Vectors, org.Hs.eg.db,
rlang,</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, covr, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-19 13:44:30 UTC; egeulgen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ege Ulgen <a href="https://orcid.org/0000-0003-2090-3621"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-19 14:02:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='driveR'>driveR: An R Package for Prioritizing Cancer Driver Genes Using Genomics Data</h2><span id='topic+driveR'></span><span id='topic+driveR-package'></span>

<h3>Description</h3>

<p>Cancer genomes contain large numbers of somatic alterations but few genes
drive tumor development. Identifying cancer driver genes is critical
for precision oncology. Most of current approaches either identify driver
genes based on mutational recurrence or using estimated scores predicting
the functional consequences of mutations.
</p>


<h3>Details</h3>

<p>driveR is a tool for personalized or batch analysis of genomic data for
driver gene prioritization by combining genomic information and prior
biological knowledge. As features, driveR uses coding impact metaprediction
scores, non-coding impact scores, somatic copy number alteration scores,
hotspot gene/double-hit gene condition, 'phenolyzer' gene scores and
memberships to cancer-related KEGG pathways. It uses these features to
estimate cancer-type-specific probabilities for each gene of being a cancer
driver using the related task of a multi-task learning classification model.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ege Ulgen <a href="mailto:egeulgen@gmail.com">egeulgen@gmail.com</a> (<a href="https://orcid.org/0000-0003-2090-3621">ORCID</a>) [copyright holder]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+predict_coding_impact">predict_coding_impact</a></code> for metaprediction of impact of
coding variants.
<code><a href="#topic+create_features_df">create_features_df</a></code> for creating the features table to
be used to prioritize cancer driver genes.
See <code><a href="#topic+prioritize_driver_genes">prioritize_driver_genes</a></code> for prioritizing cancer driver genes
</p>

<hr>
<h2 id='create_features_df'>Create Data Frame of Features for Driver Gene Prioritization</h2><span id='topic+create_features_df'></span>

<h3>Description</h3>

<p>Create Data Frame of Features for Driver Gene Prioritization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_features_df(
  annovar_csv_path,
  scna_df,
  phenolyzer_annotated_gene_list_path,
  batch_analysis = FALSE,
  prep_phenolyzer_input = FALSE,
  build = "GRCh37",
  log2_ratio_threshold = 0.25,
  gene_overlap_threshold = 25,
  MCR_overlap_threshold = 25,
  hotspot_threshold = 5L,
  log2_hom_loss_threshold = -1,
  verbose = TRUE,
  na.string = "."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_features_df_+3A_annovar_csv_path">annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_scna_df">scna_df</code></td>
<td>
<p>the SCNA segments data frame. Must contain: </p>

<dl>
<dt>chr</dt><dd><p>chromosome the segment is located in</p>
</dd>
<dt>start</dt><dd><p>start position of the segment</p>
</dd>
<dt>end</dt><dd><p>end position of the segment</p>
</dd>
<dt>log2ratio</dt><dd>log<sub>2</sub><p> ratio of the segment</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="create_features_df_+3A_phenolyzer_annotated_gene_list_path">phenolyzer_annotated_gene_list_path</code></td>
<td>
<p>path to 'phenolyzer'
&quot;annotated_gene_list&quot; file</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_batch_analysis">batch_analysis</code></td>
<td>
<p>boolean to indicate whether to perform batch analysis
(<code>TRUE</code>, default) or personalized analysis (<code>FALSE</code>). If <code>TRUE</code>,
a column named 'tumor_id' should be present in both the ANNOVAR csv and the SCNA
table.</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_prep_phenolyzer_input">prep_phenolyzer_input</code></td>
<td>
<p>boolean to indicate whether or not to create
a vector of genes for use as the input of 'phenolyzer' (default = <code>FALSE</code>).
If <code>TRUE</code>, the features data frame is not created and instead the vector
of gene symbols (union of all genes for which scores are available) is
returned.</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_build">build</code></td>
<td>
<p>genome build for the SCNA segments data frame (default = &quot;GRCh37&quot;)</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_log2_ratio_threshold">log2_ratio_threshold</code></td>
<td>
<p>the log<sub>2</sub>
ratio threshold for keeping high-confidence SCNA events (default = 0.25)</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_gene_overlap_threshold">gene_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a segment and a transcript (default = 25). This means that if only a segment
overlaps a transcript more than this threshold, the transcript is assigned
the segment's SCNA event.</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_mcr_overlap_threshold">MCR_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a gene and an MCR region (default = 25). This means that if only a gene
overlaps an MCR region more than this threshold, the gene is assigned the
SCNA density of the MCR</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_hotspot_threshold">hotspot_threshold</code></td>
<td>
<p>to determine hotspot genes, the (integer) threshold
for the minimum number of cases with certain mutation in COSMIC (default = 5)</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_log2_hom_loss_threshold">log2_hom_loss_threshold</code></td>
<td>
<p>to determine double-hit events, the
log<sub>2</sub> threshold for identifying
homozygous loss events (default = -1).</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_verbose">verbose</code></td>
<td>
<p>boolean controlling verbosity (default = <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="create_features_df_+3A_na.string">na.string</code></td>
<td>
<p>string that was used to indicate when a score is not available
during annotation with ANNOVAR (default = &quot;.&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>prep_phenolyzer_input=FALSE</code> (default), a data frame of
features for prioritizing cancer driver genes (<code>gene_symbol</code> as
the first column and 26 other columns containing features). If
<code>prep_phenolyzer_input=TRUE</code>, the functions returns a vector gene symbols
(union of all gene symbols for which scores are available) to be used as the
input for performing 'phenolyzer' analysis.
</p>
<p>The features data frame contains the following columns:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt><dd><p>the maximum metapredictor (coding) impact score for the gene</p>
</dd>
<dt>noncoding_score</dt><dd><p>the maximum non-coding PHRED-scaled CADD score for the gene</p>
</dd>
<dt>scna_score</dt><dd><p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located</p>
</dd>
<dt>hotspot_double_hit</dt><dd><p>boolean indicating whether the gene is a hotspot gene (indication of oncogenes) or subject to double-hit (indication of tumor-suppressor genes)</p>
</dd>
<dt>phenolyzer_score</dt><dd><p>'phenolyzer' score for the gene</p>
</dd>
<dt>hsa03320</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04010</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04020</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04024</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04060</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04066</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04110</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04115</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04150</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04151</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04210</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04310</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04330</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04340</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04350</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04370</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04510</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04512</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04520</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04630</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04915</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+prioritize_driver_genes">prioritize_driver_genes</a></code> for prioritizing cancer driver genes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path2annovar_csv &lt;- system.file("extdata/example.hg19_multianno.csv",
                                package = "driveR")
path2phenolyzer_out &lt;- system.file("extdata/example.annotated_gene_list",
                                   package = "driveR")
features_df &lt;- create_features_df(annovar_csv_path = path2annovar_csv,
                                  scna_df = example_scna_table,
                                  phenolyzer_annotated_gene_list_path = path2phenolyzer_out)


</code></pre>

<hr>
<h2 id='create_gene_level_scna_df'>Create Gene-level SCNA Data Frame</h2><span id='topic+create_gene_level_scna_df'></span>

<h3>Description</h3>

<p>Create Gene-level SCNA Data Frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_gene_level_scna_df(
  scna_df,
  build = "GRCh37",
  gene_overlap_threshold = 25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_gene_level_scna_df_+3A_scna_df">scna_df</code></td>
<td>
<p>the SCNA segments data frame. Must contain: </p>

<dl>
<dt>chr</dt><dd><p>chromosome the segment is located in</p>
</dd>
<dt>start</dt><dd><p>start position of the segment</p>
</dd>
<dt>end</dt><dd><p>end position of the segment</p>
</dd>
<dt>log2ratio</dt><dd>log<sub>2</sub><p> ratio of the segment</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="create_gene_level_scna_df_+3A_build">build</code></td>
<td>
<p>genome build for the SCNA segments data frame (default = &quot;GRCh37&quot;)</p>
</td></tr>
<tr><td><code id="create_gene_level_scna_df_+3A_gene_overlap_threshold">gene_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a segment and a transcript (default = 25). This means that if only a segment
overlaps a transcript more than this threshold, the transcript is assigned
the segment's SCNA event.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame of gene-level SCNA events, i.e. table of genes overlapped
by SCNA segments.
</p>

<hr>
<h2 id='create_noncoding_impact_score_df'>Create Non-coding Impact Score Data Frame</h2><span id='topic+create_noncoding_impact_score_df'></span>

<h3>Description</h3>

<p>Create Non-coding Impact Score Data Frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_noncoding_impact_score_df(annovar_csv_path, na.string = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_noncoding_impact_score_df_+3A_annovar_csv_path">annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td></tr>
<tr><td><code id="create_noncoding_impact_score_df_+3A_na.string">na.string</code></td>
<td>
<p>string that was used to indicate when a score is not available
during annotation with ANNOVAR (default = &quot;.&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame of meta-prediction scores containing 2 columns: </p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>CADD_phred</dt><dd><p>PHRED-scaled CADD score</p>
</dd>
</dl>


<hr>
<h2 id='create_SCNA_score_df'>Create SCNA Score Data Frame</h2><span id='topic+create_SCNA_score_df'></span>

<h3>Description</h3>

<p>Create SCNA Score Data Frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_SCNA_score_df(
  gene_SCNA_df,
  build = "GRCh37",
  log2_ratio_threshold = 0.25,
  MCR_overlap_threshold = 25
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_SCNA_score_df_+3A_gene_scna_df">gene_SCNA_df</code></td>
<td>
<p>data frame of gene-level SCNAs (output of <code><a href="#topic+create_gene_level_scna_df">create_gene_level_scna_df</a></code>)</p>
</td></tr>
<tr><td><code id="create_SCNA_score_df_+3A_build">build</code></td>
<td>
<p>genome build for the SCNA segments data frame (default = &quot;GRCh37&quot;)</p>
</td></tr>
<tr><td><code id="create_SCNA_score_df_+3A_log2_ratio_threshold">log2_ratio_threshold</code></td>
<td>
<p>the log<sub>2</sub>
ratio threshold for keeping high-confidence SCNA events (default = 0.25)</p>
</td></tr>
<tr><td><code id="create_SCNA_score_df_+3A_mcr_overlap_threshold">MCR_overlap_threshold</code></td>
<td>
<p>the percentage threshold for the overlap between
a gene and an MCR region (default = 25). This means that if only a gene
overlaps an MCR region more than this threshold, the gene is assigned the
SCNA density of the MCR</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first aggregates SCNA log<sub>2</sub> ratio
on gene-level (by keeping the ratio with the maximal |log<sub>2</sub>|
ratio over all the SCNA segments overlapping a gene). Next, it identifies the
minimal common regions (MCRs) that the genes overlap and finally assigns the
SCNA density (SCNA/Mb) values as proxy SCNA scores.
</p>


<h3>Value</h3>

<p>data frame of SCNA proxy scores containing 2 columns: </p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>SCNA_density</dt><dd><p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located.</p>
</dd>
</dl>


<hr>
<h2 id='determine_double_hit_genes'>Determine Double-Hit Genes</h2><span id='topic+determine_double_hit_genes'></span>

<h3>Description</h3>

<p>Determine Double-Hit Genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>determine_double_hit_genes(
  annovar_csv_path,
  gene_SCNA_df,
  log2_hom_loss_threshold = -1,
  batch_analysis = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="determine_double_hit_genes_+3A_annovar_csv_path">annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td></tr>
<tr><td><code id="determine_double_hit_genes_+3A_gene_scna_df">gene_SCNA_df</code></td>
<td>
<p>data frame of gene-level SCNAs (output of <code><a href="#topic+create_gene_level_scna_df">create_gene_level_scna_df</a></code>)</p>
</td></tr>
<tr><td><code id="determine_double_hit_genes_+3A_log2_hom_loss_threshold">log2_hom_loss_threshold</code></td>
<td>
<p>to determine double-hit events, the
log<sub>2</sub> threshold for identifying
homozygous loss events (default = -1).</p>
</td></tr>
<tr><td><code id="determine_double_hit_genes_+3A_batch_analysis">batch_analysis</code></td>
<td>
<p>boolean to indicate whether to perform batch analysis
(<code>TRUE</code>, default) or personalized analysis (<code>FALSE</code>). If <code>TRUE</code>,
a column named 'tumor_id' should be present in both the ANNOVAR csv and the SCNA
table.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of gene symbols that are subject to double-hit event(s), i.e.
non-synonymous mutation + homozygous copy-number loss
</p>

<hr>
<h2 id='determine_hotspot_genes'>Determine Hotspot Containing Genes</h2><span id='topic+determine_hotspot_genes'></span>

<h3>Description</h3>

<p>Determine Hotspot Containing Genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>determine_hotspot_genes(annovar_csv_path, hotspot_threshold = 5L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="determine_hotspot_genes_+3A_annovar_csv_path">annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td></tr>
<tr><td><code id="determine_hotspot_genes_+3A_hotspot_threshold">hotspot_threshold</code></td>
<td>
<p>to determine hotspot genes, the (integer) threshold
for the minimum number of cases with certain mutation in COSMIC (default = 5)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of gene symbols of genes containing hotspot mutation(s)
</p>

<hr>
<h2 id='example_cohort_features_table'>Example Cohort-level Features Table for Driver Prioritization</h2><span id='topic+example_cohort_features_table'></span>

<h3>Description</h3>

<p>The example dataset containing features for prioritizing cancer driver genes for 10 randomly
selected samples from TCGA's LAML (Acute Myeloid Leukemia) cohort
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_cohort_features_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 349 rows and 27 variables:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt><dd><p>the maximum metapredictor (coding) impact score for the gene</p>
</dd>
<dt>noncoding_score</dt><dd><p>the maximum non-coding PHRED-scaled CADD score for the gene</p>
</dd>
<dt>scna_score</dt><dd><p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located</p>
</dd>
<dt>hotspot_double_hit</dt><dd><p>boolean indicating whether the gene is a hotspot gene (indication of oncogenes) or subject to double-hit (indication of tumor-suppressor genes)</p>
</dd>
<dt>phenolyzer_score</dt><dd><p>'phenolyzer' score for the gene</p>
</dd>
<dt>hsa03320</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04010</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04020</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04024</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04060</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04066</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04110</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04115</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04150</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04151</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04210</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04310</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04330</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04340</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04350</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04370</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04510</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04512</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04520</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04630</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04915</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+KEGG_cancer_pathways_descriptions">KEGG_cancer_pathways_descriptions</a></code> for descriptions of
KEGG &quot;Pathways in cancer&quot;-related pathways.
</p>

<hr>
<h2 id='example_cohort_scna_table'>Example Cohort-level Somatic Copy Number Alteration Table</h2><span id='topic+example_cohort_scna_table'></span>

<h3>Description</h3>

<p>A data set containing the somatic copy number alteration data for 10 randomly
selected samples from TCGA's LAML (Acute Myeloid Leukemia) cohort
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_cohort_scna_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 126147 rows and 5 variables:
</p>

<dl>
<dt>chr</dt><dd><p>chromosome the segment is located in</p>
</dd>
<dt>start</dt><dd><p>start position of the segment</p>
</dd>
<dt>end</dt><dd><p>end position of the segment</p>
</dd>
<dt>log2ratio</dt><dd>log<sub>2</sub><p> ratio of
the segment</p>
</dd>
<dt>tumor_id</dt><dd><p>ID for the tumor containing the SCNA segment</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://dcc.icgc.org/releases/release_28">https://dcc.icgc.org/releases/release_28</a>
</p>

<hr>
<h2 id='example_features_table'>Example Features Table for Driver Prioritization</h2><span id='topic+example_features_table'></span>

<h3>Description</h3>

<p>The example dataset containing features for prioritizing cancer driver genes for
the lung adenocarcinoma patient studied in Imielinski M, Greulich H, Kaplan B, et al.
Oncogenic and sorafenib-sensitive ARAF mutations in lung adenocarcinoma.
J Clin Invest. 2014;124(4):1582-6.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_features_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 4901 rows and 27 variables:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt><dd><p>the maximum metapredictor (coding) impact score for the gene</p>
</dd>
<dt>noncoding_score</dt><dd><p>the maximum non-coding PHRED-scaled CADD score for the gene</p>
</dd>
<dt>scna_score</dt><dd><p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located</p>
</dd>
<dt>hotspot_double_hit</dt><dd><p>boolean indicating whether the gene is a hotspot gene (indication of oncogenes) or subject to double-hit (indication of tumor-suppressor genes)</p>
</dd>
<dt>phenolyzer_score</dt><dd><p>'phenolyzer' score for the gene</p>
</dd>
<dt>hsa03320</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04010</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04020</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04024</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04060</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04066</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04110</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04115</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04150</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04151</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04210</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04310</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04330</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04340</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04350</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04370</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04510</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04512</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04520</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04630</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04915</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+KEGG_cancer_pathways_descriptions">KEGG_cancer_pathways_descriptions</a></code> for descriptions of
KEGG &quot;Pathways in cancer&quot;-related pathways.
</p>

<hr>
<h2 id='example_scna_table'>Example Somatic Copy Number Alteration Table</h2><span id='topic+example_scna_table'></span>

<h3>Description</h3>

<p>A data set containing the somatic copy number alteration data for the lung
adenocarcinoma patient studied in Imielinski M, Greulich H, Kaplan B, et al.
Oncogenic and sorafenib-sensitive ARAF mutations in lung adenocarcinoma.
J Clin Invest. 2014;124(4):1582-6.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_scna_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 3160 rows and 4 variables:
</p>

<dl>
<dt>chr</dt><dd><p>chromosome the segment is located in</p>
</dd>
<dt>start</dt><dd><p>start position of the segment</p>
</dd>
<dt>end</dt><dd><p>end position of the segment</p>
</dd>
<dt>log2ratio</dt><dd>log<sub>2</sub><p> ratio of
the segment</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/24569458/">https://pubmed.ncbi.nlm.nih.gov/24569458/</a>
</p>

<hr>
<h2 id='KEGG_cancer_pathways_descriptions'>KEGG &quot;Pathways in cancer&quot;-related Pathways - Descriptions</h2><span id='topic+KEGG_cancer_pathways_descriptions'></span>

<h3>Description</h3>

<p>A data frame containing descriptions for KEGG &quot;Pathways in cancer&quot;
(hsa05200)-related pathways.
<em>Generated on Nov 17, 2020.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KEGG_cancer_pathways_descriptions
</code></pre>


<h3>Format</h3>

<p>A data frame with 21 rows and 2 variables:
</p>

<dl>
<dt>id</dt><dd><p>KEGG pathway ID</p>
</dd>
<dt>description</dt><dd><p>KEGG pathway description</p>
</dd>
</dl>


<hr>
<h2 id='MTL_submodel_descriptions'>MTL Sub-model Descriptions</h2><span id='topic+MTL_submodel_descriptions'></span>

<h3>Description</h3>

<p>A data frame containing descriptions for all sub-models of the MTL model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTL_submodel_descriptions
</code></pre>


<h3>Format</h3>

<p>A data frame with 21 rows and 2 variables:
</p>

<dl>
<dt>short_name</dt><dd><p>short name for the cancer type</p>
</dd>
<dt>description</dt><dd><p>description of the cancer type</p>
</dd>
</dl>


<hr>
<h2 id='predict_coding_impact'>Create Coding Impact Meta-prediction Score Data Frame</h2><span id='topic+predict_coding_impact'></span>

<h3>Description</h3>

<p>Create Coding Impact Meta-prediction Score Data Frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_coding_impact(
  annovar_csv_path,
  keep_highest_score = TRUE,
  keep_single_symbol = TRUE,
  na.string = "."
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_coding_impact_+3A_annovar_csv_path">annovar_csv_path</code></td>
<td>
<p>path to 'ANNOVAR' csv output file</p>
</td></tr>
<tr><td><code id="predict_coding_impact_+3A_keep_highest_score">keep_highest_score</code></td>
<td>
<p>boolean to indicate whether to keep only the maximal
impact score per gene (default = <code>TRUE</code>). If <code>FALSE</code>, all scores
per each gene are returned</p>
</td></tr>
<tr><td><code id="predict_coding_impact_+3A_keep_single_symbol">keep_single_symbol</code></td>
<td>
<p>in ANNOVAR outputs, a variant may be annotated as
exonic in multiple genes. This boolean argument controls whether or not to
keep only the first encountered symbol for a variant (default = <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="predict_coding_impact_+3A_na.string">na.string</code></td>
<td>
<p>string that was used to indicate when a score is not available
during annotation with ANNOVAR (default = &quot;.&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame of meta-prediction scores containing 2 columns: </p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt><dd><p>metapredictor impact score</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>path2annovar_csv &lt;- system.file("extdata/example.hg19_multianno.csv",
                                package = "driveR")
metapred_df &lt;- predict_coding_impact(path2annovar_csv)
</code></pre>

<hr>
<h2 id='prioritize_driver_genes'>Prioritize Cancer Driver Genes</h2><span id='topic+prioritize_driver_genes'></span>

<h3>Description</h3>

<p>Prioritize Cancer Driver Genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prioritize_driver_genes(features_df, cancer_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prioritize_driver_genes_+3A_features_df">features_df</code></td>
<td>
<p>the features data frame for all genes, containing the following columns:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>metaprediction_score</dt><dd><p>the maximum metapredictor (coding) impact score for the gene</p>
</dd>
<dt>noncoding_score</dt><dd><p>the maximum non-coding PHRED-scaled CADD score for the gene</p>
</dd>
<dt>scna_score</dt><dd><p>SCNA proxy score. SCNA density (SCNA/Mb) of the minimal common region (MCR) in which the gene is located</p>
</dd>
<dt>hotspot_double_hit</dt><dd><p>boolean indicating whether the gene is a hotspot gene (indication of oncogenes) or subject to double-hit (indication of tumor-suppressor genes)</p>
</dd>
<dt>phenolyzer_score</dt><dd><p>'phenolyzer' score for the gene</p>
</dd>
<dt>hsa03320</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04010</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04020</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04024</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04060</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04066</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04110</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04115</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04150</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04151</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04210</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04310</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04330</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04340</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04350</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04370</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04510</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04512</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04520</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04630</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
<dt>hsa04915</dt><dd><p>boolean indicating whether or not the gene takes part in this KEGG pathway</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="prioritize_driver_genes_+3A_cancer_type">cancer_type</code></td>
<td>
<p>short name of the cancer type. All available cancer types
are listed in <code><a href="#topic+MTL_submodel_descriptions">MTL_submodel_descriptions</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with 3 columns:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>driverness_prob</dt><dd><p>estimated probability for each gene in <code>features_df</code> of being a
cancer driver. The probabilities are calculated using the selected (via
<code>cancer_type</code>) cancer type's sub-model.</p>
</dd>
<dt>prediction</dt><dd><p>prediction based on the cancer-type-specific threshold (either &quot;driver&quot; or &quot;non-driver&quot;)</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+create_features_df">create_features_df</a></code> for creating the features table.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>drivers_df &lt;- prioritize_driver_genes(example_features_table, "LUAD")

</code></pre>

<hr>
<h2 id='specific_thresholds'>Tumor type specific probability thresholds</h2><span id='topic+specific_thresholds'></span>

<h3>Description</h3>

<p>Driver gene probability thresholds for all 21 cancer types (submodels).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>specific_thresholds
</code></pre>


<h3>Format</h3>

<p>vector with 21 elements
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
