<!DOCTYPE html><html><head><title>Help for package DRquality</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DRquality}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ClassificationError'><p>Classification Error (rate)</p></a></li>
<li><a href='#Cmeasure'><p>C-Measure subtypes</p></a></li>
<li><a href='#GabrielClassificationError'>
<p>Gabriel Classification Error (GCE)</p></a></li>
<li><a href='#KendallsTau'>
<p>Statistical correlation by Kendall</p></a></li>
<li><a href='#MeasureTandD'><p>Trustworthiness and Discontinuity.</p></a></li>
<li><a href='#plotMeasureRAAR'><p>Computes Rescaled Average Agreement Rate</p></a></li>
<li><a href='#plotMeasureTandD'><p>Computes rank-based smoothed precision and recall</p></a></li>
<li><a href='#PrecisionAndRecall'><p>Precision and Recall.</p></a></li>
<li><a href='#RAAR'><p>Rescaled average agreement rate</p></a></li>
<li><a href='#SpearmanError'><p>Calculates the error of a projection with spearman's rank correlation coefficient.</p></a></li>
<li><a href='#SpearmansRho'><p>Calculates the error of a projection with spearman's rank correlation coefficient</p></a></li>
<li><a href='#TopologicalCorrelation'><p>Topological Correlation</p></a></li>
<li><a href='#ZrehenMeasure4All'><p>ZrehenMeasure4All</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quality Measurements for Dimensionality Reduction</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-10</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Thrun &lt;m.thrun@gmx.net&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Several quality measurements for investigating the performance of dimensionality reduction methods are provided here. In addition a new quality measurement called Gabriel classification error is made accessible, which was published in Thrun, M. C., Märte, J., &amp; Stier, Q: "Analyzing Quality Measurements for Dimensionality Reduction" (2023), Machine Learning and Knowledge Extraction (MAKE), &lt;<a href="https://doi.org/10.3390%2Fmake5030056">doi:10.3390/make5030056</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>DatabionicSwarm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>plotly, geometry, deldir, FCPS, ProjectionBasedClustering (&ge;
1.2.1), DataVisualizations, FastKNN, ggplot2, pcaPP,pracma,
spdep, grid, igraph, cccd, sf</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-12 07:30:23 UTC; MCT</td>
</tr>
<tr>
<td>Author:</td>
<td>Quirin Stier [aut],
  Florian Lerch [ctb],
  Julian Märte [aut],
  Hermann Tafo [ctb],
  Laukert Schlichting [ctb],
  Michael Thrun <a href="https://orcid.org/0000-0001-9542-5543"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-12 09:40:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='ClassificationError'>Classification Error (rate)</h2><span id='topic+ClassificationError'></span>

<h3>Description</h3>

<p>Compares projected points to a given prior classification using knn classifier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClassificationError(OutputDistances,Cls,k=5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClassificationError_+3A_outputdistances">OutputDistances</code></td>
<td>
<p>[1:n,1:n] numeric matrix with distance matrix of
projected data.</p>
</td></tr>
<tr><td><code id="ClassificationError_+3A_cls">Cls</code></td>
<td>
<p>[1:n] Numeric vector containing class information.</p>
</td></tr>
<tr><td><code id="ClassificationError_+3A_k">k</code></td>
<td>
<p>number of k nearest neighbors, in Venna 2010 set to 5 (here default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Projected points are evaluated by k-nearest neighbor classification accuracy (with k = 5), that is, each sample in the
visualization is classified by majority vote of its k nearest neighbors in the visualization, and the
classification is compared to the ground truth label. [Venna 2010].
</p>


<h3>Value</h3>

<p>List with three entries:
</p>
<table>
<tr><td><code>Error</code></td>
<td>
<p>Classification Error: 1-Accuracy[1]</p>
</td></tr>
<tr><td><code>Accuracy</code></td>
<td>
<p>Accuracy </p>
</td></tr>
<tr><td><code>KNNCls</code></td>
<td>
<p>[1:n]] cls of knn classifier</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Here, the Outputdistances of the Projected points are used.
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Venna, J., Peltonen, J., Nybo, K., Aidos, H., and Kaski, S. Information retrieval perspective to nonlinear dimensionality reduction for data visualization. The Journal of Machine Learning Research, 11, 451-490.  (2010)
</p>
<p>Gracia, A., Gonzalez, S., Robles, V., and Menasalvas, E. A methodology to compare Dimensionality Reduction algorithms in terms of loss of quality. Information Sciences, 270, 1-27.  (2014)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
ClassificationError(as.matrix(dist(projection)),Hepta$Cls)
}


</code></pre>

<hr>
<h2 id='Cmeasure'>C-Measure subtypes</h2><span id='topic+Cmeasure'></span>

<h3>Description</h3>

<p>Calculate the C-Measure subtypes of minimal path length and minimal
wiring</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cmeasure(Data, Projection, k = 1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cmeasure_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] numerical matrix of points in input space.</p>
</td></tr>
<tr><td><code id="Cmeasure_+3A_projection">Projection</code></td>
<td>
<p>[1:n,1:2] numerical matrix of points in output space.</p>
</td></tr>
<tr><td><code id="Cmeasure_+3A_k">k</code></td>
<td>
<p>Number of nearest neighbors, both measures set it always to k=1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>[1:2] Numerical vector of MinimalPathlength and MinimalWiring values.</p>


<h3>Author(s)</h3>

<p>Michael Thrun</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
Cmeasure(Hepta$Data,projection)
}


</code></pre>

<hr>
<h2 id='GabrielClassificationError'>
Gabriel Classification Error (GCE)
</h2><span id='topic+GabrielClassificationError'></span>

<h3>Description</h3>

<p>GCE searches for the k-nearest neighbors of the first gabriel neighbors
weighted by the Euclidean Distances of the Inputspace [Thrun et al, 2023]. GCE evaluates these
neighbors in the Output space. A low value indicates a better two-dimensional
projection of the high-dimensional Input space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GabrielClassificationError(Data,ProjectedPoints,Cls,LC,
PlotIt=FALSE,Plotter = "native", Colors = NULL,LineColor= 'grey',
main = "Name of Projection", mainSize = 24,xlab = "X", ylab = "Y", xlim, ylim,
pch,lwd,Margin=list(t=50,r=0,l=0,b=0))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GabrielClassificationError_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] Numeric matrix with n cases and d variables</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_projectedpoints">ProjectedPoints</code></td>
<td>
<p>[1:n,1:2] Numeric matrix with 2D points in cartesian coordinates</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_cls">Cls</code></td>
<td>
<p>[1:n] Numeric vector with class labels</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_lc">LC</code></td>
<td>
<p>Optional, Numeric vector of two values determining grid size of the underlying projection</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_plotit">PlotIt</code></td>
<td>
<p>Optional, Boolean: TRUE/FALSE =&gt; Plot/Do not plot (Default: FALSE)</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_plotter">Plotter</code></td>
<td>
<p>Optional, Character with plot technique (native or plotly)</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_colors">Colors</code></td>
<td>
<p>Optional, Character vector of class colors for points</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_linecolor">LineColor</code></td>
<td>
<p>Optional, Character of line color used for edges of graph</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_main">main</code></td>
<td>
<p>Optional, Character plot title</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_mainsize">mainSize</code></td>
<td>
<p>Optional, Numeric size of plot title</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_xlab">xlab</code></td>
<td>
<p>Optional, Character name of x ax</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_ylab">ylab</code></td>
<td>
<p>Optional, Character name of y ax</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_xlim">xlim</code></td>
<td>
<p>Optional, Numeric vector with two values defining x ax range</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_ylim">ylim</code></td>
<td>
<p>Optional, Numeric vector with two values defining y ax range</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_pch">pch</code></td>
<td>
<p>Optional, Numeric of point size (graphic parameter)</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_lwd">lwd</code></td>
<td>
<p>Optional, Numeric of linewidth (graphic parameter)</p>
</td></tr>
<tr><td><code id="GabrielClassificationError_+3A_margin">Margin</code></td>
<td>
<p>Optional, Margin of plotly plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gabriel Classification Error (GCE) makes an unbiased evaluation of distance- and
density-based structures which might be even non-linear separable. First, GCE
utilizes the information provided by a prior classification to assess projected
structures. Second, GCE applies the insights drawn from graph theory. Details
are described in [Thrun et al, 2023].</p>


<h3>Value</h3>

<p>list of several entries containing first the GCE itself as main result followed
by further entries which contain potential important information
</p>
<table>
<tr><td><code>GCE</code></td>
<td>
<p>Numeric: the 'Gabriel Classification Error'</p>
</td></tr>
<tr><td><code>GCEperPoint</code></td>
<td>
<p>[1:n] unnormalized GCE of each point: GCE =
mean(GCEperPoint)</p>
</td></tr>
<tr><td><code>nn</code></td>
<td>
<p>the number of points in a relevant neghborhood:
0.5 * 85percentile(AnzNN)</p>
</td></tr>
<tr><td><code>AnzNN</code></td>
<td>
<p>[1:n] the number of points with a Gabriel graph neighborhood</p>
</td></tr>
<tr><td><code>NNdists</code></td>
<td>
<p>[1:n,1:nn] the distances within the relevant neighborhood, 1 for
inter cluster distances and 0 for inner cluster distances</p>
</td></tr>
<tr><td><code>HD</code></td>
<td>
<p>[1:nn] HD = HarmonicDecay(nn) i.e weight function for the NNdists:
GCEperPoint = HD*NNdists</p>
</td></tr>
<tr><td><code>IsInterDistance</code></td>
<td>
<p>Distances to the nn closest neighbors.</p>
</td></tr>
<tr><td><code>GabrielDists</code></td>
<td>
<p>Distance matrix implied by high dimensional distances and the underlying gabriel (Gabriel) graph </p>
</td></tr>
<tr><td><code>ProjectionGraphError</code></td>
<td>
<p>Plotly object in case, plotly is chosen.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun, Quirin Stier, Julian Märte
</p>


<h3>References</h3>

<p>[Thrun et al, 2023] Thrun, M.C, Märte, J., Stier, Q.: Analyzing Quality Measurements for Dimensionality Reduction, Machine Learning and Knowledge Extraction (MAKE), Vol 5., accepted, 2023.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
GabrielClassificationError(Hepta$Data,projection,Hepta$Cls)$GCE
}


if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
GabrielClassificationError(Hepta$Data,projection,Hepta$Cls)$GCE
}


</code></pre>

<hr>
<h2 id='KendallsTau'>
Statistical correlation by Kendall
</h2><span id='topic+KendallsTau'></span>

<h3>Description</h3>

<p>Calculates the statistical correlation by Kendall. Basically a wrapper to
pcaPP::cor.fk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KendallsTau(InputDists, OutputDists)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KendallsTau_+3A_inputdists">InputDists</code></td>
<td>
<p>Matrix containing the distances of the first dataset.</p>
</td></tr>
<tr><td><code id="KendallsTau_+3A_outputdists">OutputDists</code></td>
<td>
<p>Matrix containing the distances of the second dataset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Equivalent to <code><a href="pcaPP.html#topic+cor.fk">cor.fk</a></code>
</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
InputDist=dist(Hepta$Data)
projection=cmdscale(InputDist, k=2)
KendallsTau(as.matrix(InputDist),as.matrix(dist(projection)))
}


</code></pre>

<hr>
<h2 id='MeasureTandD'>Trustworthiness and Discontinuity.</h2><span id='topic+MeasureTandD'></span>

<h3>Description</h3>

<p>In a trustworthy projection the visualized proximities hold in the original data
as well, whereas a continuous projection visualizes all proximities of the
original data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>MeasureTandD(Data, pData, NeighborhoodSize)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MeasureTandD_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] points in input room with d attributes</p>
</td></tr>
<tr><td><code id="MeasureTandD_+3A_pdata">pData</code></td>
<td>
<p>[1:n,1:2] projected points in output room, with index,x,y or index,line,column</p>
</td></tr>
<tr><td><code id="MeasureTandD_+3A_neighborhoodsize">NeighborhoodSize</code></td>
<td>
<p>Integer - sets the maximum number of neighbors to
calculate trustworthiness and continuity for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix [1:NeighborhoodSize,1:2] containing the trustworthiness values
in the first column and the discontinuity values in the second column.
</p>


<h3>Author(s)</h3>

<p>Julian Märte</p>


<h3>References</h3>

<p>Venna, J., &amp; Kaski, S. (2005, September). Local multidimensional scaling with controlled tradeoff between trustworthiness and continuity. In Proceedings of 5th Workshop on Self-Organizing Maps (pp. 695-702).
</p>
<p>Kaski, S., Nikkilä, J., Oja, M., Venna, J., Törönen, P., &amp; Castrén, E. (2003). Trustworthiness and metrics in visualizing similarity of gene expression. BMC bioinformatics, 4(1), 1-13.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
MeasureTandD(Hepta$Data,projection, 2)
}


</code></pre>

<hr>
<h2 id='plotMeasureRAAR'>Computes Rescaled Average Agreement Rate</h2><span id='topic+plotMeasureRAAR'></span>

<h3>Description</h3>

<p>Rescaled average agreement rate deduced by the co-ranking matrix
from LCMC for various different sizes of the neighborhood.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMeasureRAAR(Raar, label = 'ProjectionMethod',
gPlotList = list(RAARplot = ggplot2::ggplot()), LineType="solid", Shape = 16,
PointsPerE = 10, fancy = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMeasureRAAR_+3A_raar">Raar</code></td>
<td>
<p>Output of RAAR() applied for a projection method.</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_label">label</code></td>
<td>
<p>Title of plot.</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_gplotlist">gPlotList</code></td>
<td>
<p>Settings for ggplot.</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_linetype">LineType</code></td>
<td>
<p>Character - graphic parameter: Line type of ggplot.</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_shape">Shape</code></td>
<td>
<p>Integer: type of point</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_pointspere">PointsPerE</code></td>
<td>
<p>Numeric graphic parameter: Distance between markers on plot
line</p>
</td></tr>
<tr><td><code id="plotMeasureRAAR_+3A_fancy">fancy</code></td>
<td>
<p>Boolean graphic parameter: Some automatic settings for a more
appealing plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object</p>


<h3>Author(s)</h3>

<p>Michael Thrun</p>

<hr>
<h2 id='plotMeasureTandD'>Computes rank-based smoothed precision and recall</h2><span id='topic+plotMeasureTandD'></span>

<h3>Description</h3>

<p>Compares the projection in pData with the original data in Data and
calculates trustworthiness and continuity of the projection for neighborhood
sizes ranging from 1 to the size of the neighborhood.</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMeasureTandD(TDmatrix, label = 'ProjectionMethod',
gPlotList = list(TW = ggplot2::ggplot(), DC = ggplot2::ggplot()), LineType = "solid",
Shape = 16, PointsPerE = 16)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMeasureTandD_+3A_tdmatrix">TDmatrix</code></td>
<td>
<p>Output of MeasureTundD() applied for a projection method.</p>
</td></tr>
<tr><td><code id="plotMeasureTandD_+3A_label">label</code></td>
<td>
<p>Title of plot.</p>
</td></tr>
<tr><td><code id="plotMeasureTandD_+3A_gplotlist">gPlotList</code></td>
<td>
<p>Settings for ggplot.</p>
</td></tr>
<tr><td><code id="plotMeasureTandD_+3A_linetype">LineType</code></td>
<td>
<p>Character - graphic parameter: Line type of ggplot.</p>
</td></tr>
<tr><td><code id="plotMeasureTandD_+3A_shape">Shape</code></td>
<td>
<p>Integer: type of point</p>
</td></tr>
<tr><td><code id="plotMeasureTandD_+3A_pointspere">PointsPerE</code></td>
<td>
<p>Numeric graphic parameter: Distance between markers on plot
line</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object</p>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>

<hr>
<h2 id='PrecisionAndRecall'>Precision and Recall.</h2><span id='topic+PrecisionAndRecall'></span>

<h3>Description</h3>

<p>Trade-off between missing similar points versus retrieving dissimilar points.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrecisionAndRecall(Data, pData, NeighborhoodSize = 20)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrecisionAndRecall_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] points in input room with d attributes</p>
</td></tr>
<tr><td><code id="PrecisionAndRecall_+3A_pdata">pData</code></td>
<td>
<p>[1:n,1:2] projected points in output room, with index,x,y or index,line,column</p>
</td></tr>
<tr><td><code id="PrecisionAndRecall_+3A_neighborhoodsize">NeighborhoodSize</code></td>
<td>
<p>Sets the 'effective number of neighbors' used to control
the width of the Gaussian, NeRV paper Seite 463 setzt Default auf 20</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix [1:NeighborhoodSize, 1:2] containing the precision values in the
first column and the recall values in the second column of the matrix.
</p>


<h3>Author(s)</h3>

<p>Felix Pape</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
PrecisionAndRecall(Hepta$Data,projection)
}


</code></pre>

<hr>
<h2 id='RAAR'>Rescaled average agreement rate</h2><span id='topic+RAAR'></span>

<h3>Description</h3>

<p>Rescaled average agreement rate deduced by the co-ranking matrix from LCMC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RAAR(Data, ProjectedPoints, kmax = nrow(Data) - 2, PlotIt = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RAAR_+3A_data">Data</code></td>
<td>

<p>Matrix containing n cases in rows, d variables in columns or a distance matrix which in this case has to be symmetric
</p>
</td></tr>
<tr><td><code id="RAAR_+3A_projectedpoints">ProjectedPoints</code></td>
<td>

<p>n by OutputDimension matrix containing coordinates of the Projection
</p>
</td></tr>
<tr><td><code id="RAAR_+3A_kmax">kmax</code></td>
<td>

<p>maximum of intervall 1:kmax of k nearest neighbors
</p>
</td></tr>
<tr><td><code id="RAAR_+3A_plotit">PlotIt</code></td>
<td>

<p>Optional: Should the output be plottet. Default: TRUE
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr><td><code>Raar</code></td>
<td>
<p>Rescaled average agreement rate</p>
</td></tr>
<tr><td><code>Aar</code></td>
<td>
<p>Average agreement rate</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Thrun
</p>


<h3>References</h3>

<p>Lee, J. A., Peluffo-Ordonez, D. H., &amp; Verleysen, M. Multiscale stochastic neighbor embedding: Towards parameter-free dimensionality reduction. Paper presented at the Proceedings of 22st European Symposium on Artificial Neural Networks, Computational Intelligence And Machine Learning (ESANN) (2014).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
RAAR(Hepta$Data,projection,kmax=nrow(Hepta$Data)-2,PlotIt=TRUE)
}


</code></pre>

<hr>
<h2 id='SpearmanError'>Calculates the error of a projection with spearman's rank correlation coefficient.</h2><span id='topic+SpearmanError'></span>

<h3>Description</h3>

<p>Calculates the error of a projection with spearman's rank correlation
coefficient.</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="SpearmanError_+3A_vectorofinputdists+281+3An2+29">VectorOfInputDists(1:n2)</code></td>
<td>
<p>dissimilarities in Input Space between the n data points in vector form as produced by squareform(Dists(1:n,1:n))</p>
</td></tr>
<tr><td><code id="SpearmanError_+3A_vectorofoutputdists+281+3An2+29">VectorOfOutputDists(1:n2)</code></td>
<td>
<p>dissimilarities in Output Space between the n data points in vector form as produced by squareform(Dists(1:n,1:n))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rho rank correlation coefficient</p>


<h3>Author(s)</h3>

<p>Florian Lerch</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
SpearmanError(as.matrix(dist(Hepta$Data)),as.matrix(dist(projection)))
}


</code></pre>

<hr>
<h2 id='SpearmansRho'>Calculates the error of a projection with spearman's rank correlation coefficient</h2><span id='topic+SpearmansRho'></span>

<h3>Description</h3>

<p>Calculates the error of a projection with spearman's rank
correlation coefficient</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpearmansRho(InputDists, OutputDists)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SpearmansRho_+3A_inputdists">InputDists</code></td>
<td>
<p>[1:d,1:d] numeric matrix with input distances</p>
</td></tr>
<tr><td><code id="SpearmansRho_+3A_outputdists">OutputDists</code></td>
<td>
<p>[1:d,1:d] numeric matrix with output distances</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rho</p>


<h3>Author(s)</h3>

<p>Julian Märte</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
SpearmansRho(as.matrix(dist(Hepta$Data)),as.matrix(dist(projection)))
}


</code></pre>

<hr>
<h2 id='TopologicalCorrelation'>Topological Correlation</h2><span id='topic+TopologicalCorrelation'></span>

<h3>Description</h3>

<p>Calculates the Topological Correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TopologicalCorrelation(Data,ProjectedPoints,type='norm',method,Kn=0)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TopologicalCorrelation_+3A_data">Data</code></td>
<td>
<p>[1:n, 1:d] a numeric matrix of the given n-dim. points: the rows represent the points and the columns represent the coordinates in the d-dim. space.</p>
</td></tr>
<tr><td><code id="TopologicalCorrelation_+3A_projectedpoints">ProjectedPoints</code></td>
<td>
<p>[1:n, 1:2] numeric matrix of Projected Points, if missing, method should be set!</p>
</td></tr>
<tr><td><code id="TopologicalCorrelation_+3A_method">method</code></td>
<td>
<p>Determines whether the selected projections method for a given set of d-dim. points is a good choice. Therefor, a result of 1 means the selected projection method is good, and a result value of 0 means that the Visualization of the given Data in the two-dim. space doesnt fit the problem.</p>
</td></tr>
<tr><td><code id="TopologicalCorrelation_+3A_type">type</code></td>
<td>
<p>How the paths in the adjacency matrix should be weighted. 'norm' representes path lenthgs of 1 and eucldidean represents the distance in the euclidean metric.</p>
</td></tr>
<tr><td><code id="TopologicalCorrelation_+3A_kn">Kn</code></td>
<td>
<p>k nearest neighbours in the graph. only needed in method is isomap and LocallyLinearEmbedding</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TC value</p>


<h3>Author(s)</h3>

<p>Hermann Tafo, Laukert Schlichting 07/2015</p>


<h3>Examples</h3>

<pre><code class='language-R'>#requires DatabionicSwarm v2.2.1


if(requireNamespace("FCPS")){
#data(Hepta,package="FCPS")
#projection=cmdscale(dist(Hepta$Data), k=2)
#TopologicalCorrelation(Hepta$Data,projection)
}


</code></pre>

<hr>
<h2 id='ZrehenMeasure4All'>ZrehenMeasure4All</h2><span id='topic+ZrehenMeasure4All'></span>

<h3>Description</h3>

<p>A generalized version of the Zrehen-measure which defines the neighbourhood with
a Gabriel Graph and is therefore not restricted to grid-based projections.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZrehenMeasure4All(Data, Projection, width, height, isToroid = FALSE,
isGrid = TRUE, plotGabriel = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZrehenMeasure4All_+3A_data">Data</code></td>
<td>
<p>[1:n,1:d] points in input room with d attributes</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_projection">Projection</code></td>
<td>
<p>[1:n,1:2] projected points in output room, with index,x,y or index,line,column</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_width">width</code></td>
<td>
<p>Numeric: only necessary if toroid</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_height">height</code></td>
<td>
<p>Numeric: only necessary if toroid</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_istoroid">isToroid</code></td>
<td>
<p>Boolean: are the points toroid?</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_isgrid">isGrid</code></td>
<td>
<p>Boolean: is the grid a toroid?</p>
</td></tr>
<tr><td><code id="ZrehenMeasure4All_+3A_plotgabriel">plotGabriel</code></td>
<td>
<p>Boolean: plot the generated GabrielGraph (TRUE) or not
(FALSE). Default: plotGabriel=FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with
</p>
<table>
<tr><td><code>V$zrehen</code></td>
<td>
<p>the raw zrehen measure</p>
</td></tr>
<tr><td><code>V$normedzrehen</code></td>
<td>
<p>the zrehen measure normed by the number of neighbours</p>
</td></tr>
<tr><td><code>v$neighbourcounter</code></td>
<td>
<p>the number of possible neighbours by which the zrehen measure is normed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Lerch 07/2015</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("FCPS")){
data(Hepta,package="FCPS")
projection=cmdscale(dist(Hepta$Data), k=2)
ZrehenMeasure4All(Hepta$Data,projection)$zrehen
}


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
