<!DOCTYPE html><html><head><title>Help for package DrugSim2DR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DrugSim2DR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#datasummary'><p>datasummary: Custom Data Summaries</p></a></li>
<li><a href='#CalDEscore'><p>CalDEscore</p></a></li>
<li><a href='#DrugReposition'><p>DrugReposition</p></a></li>
<li><a href='#DrugSimscore'><p>DrugSimscore</p></a></li>
<li><a href='#Gettest'><p>Gettest</p></a></li>
<li><a href='#myenv'><p>An environment variable which includes some example data</p></a></li>
<li><a href='#plotDruglink'><p>plotDruglink</p></a></li>
<li><a href='#plotDrugstructure'><p>plotDrugstructure</p></a></li>
<li><a href='#plotTargetheatmap'><p>plotTargetheatmap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predict Drug Functional Similarity to Drug Repurposing</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic biology tool was developed to repurpose drugs via a drug-drug functional similarity network. 'DrugSim2DR' first predict drug-drug functional similarity  in the context of specific disease, and then using the similarity constructed a weighted drug similarity network. Finally, it used a network propagation algorithm on the network to identify drugs with significant target abnormalities as candidate drugs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, stats, pheatmap, ChemmineR, rvest, base, sp, tidyr,
reshape2, fastmatch</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-10 03:24:09 UTC; Administrator</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Ji Li [aut],
  Jiashuo Wu [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-10 04:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='datasummary'>datasummary: Custom Data Summaries</h2><span id='topic+DrugSim2DR'></span><span id='topic+DrugSim2DR-package'></span>

<h3>Description</h3>

<p>Easily generate custom data frame summaries
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Junwei Han <a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a> [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Ji Li
</p>
</li>
<li><p> Jiashuo Wu
</p>
</li></ul>


<hr>
<h2 id='CalDEscore'>CalDEscore</h2><span id='topic+CalDEscore'></span>

<h3>Description</h3>

<p>Function &quot;CalDEscore&quot; uses gene expression to calculate differential expression level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalDEscore(exp, Label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalDEscore_+3A_exp">exp</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="CalDEscore_+3A_label">Label</code></td>
<td>
<p>A character vector consist of &quot;0&quot; and &quot;1&quot; which represent sample class in the gene expression profile. &quot;0&quot; means normal sample and &quot;1&quot; means disease sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with one column of zscore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-Gettest("GEP")
label&lt;-Gettest("label")
# Run the function
DEscore&lt;-CalDEscore(GEP,label)
</code></pre>

<hr>
<h2 id='DrugReposition'>DrugReposition</h2><span id='topic+DrugReposition'></span>

<h3>Description</h3>

<p>The function &quot;DrugReposition&quot; is used in drug repositioning by calculating the eigenvector centrality of drugs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DrugReposition(DE,nperm = 1000,r = 0.9,p = 10^-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DrugReposition_+3A_de">DE</code></td>
<td>
<p>A matrix with one column of zscore.</p>
</td></tr>
<tr><td><code id="DrugReposition_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 1000).</p>
</td></tr>
<tr><td><code id="DrugReposition_+3A_r">r</code></td>
<td>
<p>Restart the probability of the random-walk algorithm (default: 0.9).</p>
</td></tr>
<tr><td><code id="DrugReposition_+3A_p">p</code></td>
<td>
<p>For each node, if the difference in centrality score between iterations changes less than this value, the algorithm considers the calculation complete (default: 10^-10).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with seven columns those are drugbankid, centralscore, p.value,fdr,number of targets, drug targets,drugname.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("igraph")
# Obtain the example data
GEP&lt;-Gettest("GEP")
label&lt;-Gettest("label")
# Run the function
DEscore&lt;-CalDEscore(GEP,label)
# Run the function
drug_centrality&lt;-DrugReposition(DE=DEscore,nperm = 1000,r = 0.9,p = 10^-10)
</code></pre>

<hr>
<h2 id='DrugSimscore'>DrugSimscore</h2><span id='topic+DrugSimscore'></span>

<h3>Description</h3>

<p>The function &quot;DrugSimscore&quot; is used in calculating the drug functional similarity score.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DrugSimscore(DE,nperm = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DrugSimscore_+3A_de">DE</code></td>
<td>
<p>A matrix with one column of zscore.</p>
</td></tr>
<tr><td><code id="DrugSimscore_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with four columns those are drug1, drug2, drug1 name, drug2 name, functional similarity score and FDR.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-Gettest("GEP")
label&lt;-Gettest("label")
# Run the function
DEscore&lt;-CalDEscore(GEP,label)
# Run the function
drug_drug&lt;-DrugSimscore(DE=DEscore,nperm = 0)
</code></pre>

<hr>
<h2 id='Gettest'>Gettest</h2><span id='topic+Gettest'></span>

<h3>Description</h3>

<p>Get the example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gettest(exampleData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Gettest_+3A_exampledata">exampleData</code></td>
<td>
<p>A character,should be one of&quot;Jaccard&quot;,&quot;commongenes&quot;,&quot;GO_MF&quot;,&quot;Drugs&quot;,&quot;Drugbankid_CID&quot;,&quot;drugname&quot;,&quot;GEP&quot;,&quot;label&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data
</p>

<hr>
<h2 id='myenv'>An environment variable which includes some example data</h2><span id='topic+myenv'></span>

<h3>Description</h3>

<p>An environment variable which includes some example data.
Jaccard:A matrix of Jaccard score between drugs and GOMF.
commongenes:A matrix consisting of genes shared by drug targets and GOMF.
GO_MF:GO terms of molecular functions.
Drugs:Drugs and corresponding targets.
GEP:An example gene expression profile.
label:A vector representing the label of the sample of GEP, where &quot;1&quot; is the disease sample and &quot;0&quot; is the normal sample.
Drugbankid_CID:A dataframe including three columns which are drugbankid, ChembleID, and drugname.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myenv
</code></pre>


<h3>Format</h3>

<p>An environment variable
</p>

<hr>
<h2 id='plotDruglink'>plotDruglink</h2><span id='topic+plotDruglink'></span>

<h3>Description</h3>

<p>The function &quot;plotDruglink&quot; is used to plot a bipartite network of drugs and shared molecular functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDruglink(drug1,drug2,i = 5,color_MF = "#43AAEF",color_drug = "#F7525B",
layout_type = "circle")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDruglink_+3A_drug1">drug1</code></td>
<td>
<p>The drugbank ID of drug1.</p>
</td></tr>
<tr><td><code id="plotDruglink_+3A_drug2">drug2</code></td>
<td>
<p>The drugbank ID of drug2.</p>
</td></tr>
<tr><td><code id="plotDruglink_+3A_i">i</code></td>
<td>
<p>Specifies the number of outputs molecular functions, which is 5 by default.</p>
</td></tr>
<tr><td><code id="plotDruglink_+3A_color_mf">color_MF</code></td>
<td>
<p>Defines the color of MF nodes in the network.</p>
</td></tr>
<tr><td><code id="plotDruglink_+3A_color_drug">color_drug</code></td>
<td>
<p>Defines the color of drug nodes in the network.</p>
</td></tr>
<tr><td><code id="plotDruglink_+3A_layout_type">layout_type</code></td>
<td>
<p>layout_type used to set the appropriate arrangement, there is an option to choose from &quot;circle&quot;,&quot;dh&quot;,and &quot;sugiyama&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A bipartite network of drugs and shared molecular functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Set drug1
drug1&lt;-"DB02721"
# Set drug2
drug2&lt;-"DB01213"
# Run the function
library(igraph)
plotDruglink(drug1,drug2,i = 5)
</code></pre>

<hr>
<h2 id='plotDrugstructure'>plotDrugstructure</h2><span id='topic+plotDrugstructure'></span>

<h3>Description</h3>

<p>The function &quot;plotDrugstructure&quot; can plot the chemical structure of a drug.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDrugstructure(drugid = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDrugstructure_+3A_drugid">drugid</code></td>
<td>
<p>A drugbank ID.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A chemical structure of specific drug
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load depend package
library(ChemmineR)
library(rvest)
# Obtain molecular formula and visualize it.
plotDrugstructure(drugid ="DB00780")
</code></pre>

<hr>
<h2 id='plotTargetheatmap'>plotTargetheatmap</h2><span id='topic+plotTargetheatmap'></span>

<h3>Description</h3>

<p>The function &quot;plotTargetheatmap&quot; is used to plot a heat map of drug targets expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTargetheatmap(drugid,ExpData,label,significance=FALSE,
cluster.rows=FALSE,cluster.cols=FALSE,bk=c(-2.4,2.3),show.rownames=TRUE,
show.colnames=FALSE,ann_colors=c("#FFAA2C","#2CBADA"),col=c("#2A95FF","#FF1C1C"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTargetheatmap_+3A_drugid">drugid</code></td>
<td>
<p>The drugbank ID of a drug.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_expdata">ExpData</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_label">label</code></td>
<td>
<p>A character vector consists of &quot;0&quot; and &quot;1&quot; which represent sample class in the gene expression profile. &quot;0&quot; means normal sample and &quot;1&quot; means disease sample.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_significance">significance</code></td>
<td>
<p>This parameter controls whether the p-value of differential expression is displayed.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_cluster.rows">cluster.rows</code></td>
<td>
<p>Logical value that represents whether row clustering is used.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_cluster.cols">cluster.cols</code></td>
<td>
<p>Logical value that represents whether col clustering is used.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_bk">bk</code></td>
<td>
<p>This parameter adjusts the range of values displayed by the color bar.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_show.rownames">show.rownames</code></td>
<td>
<p>This parameter controls whether row names are displayed.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_show.colnames">show.colnames</code></td>
<td>
<p>This parameter controls whether column names are displayed.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_ann_colors">ann_colors</code></td>
<td>
<p>Vector of colors used to define groups.</p>
</td></tr>
<tr><td><code id="plotTargetheatmap_+3A_col">col</code></td>
<td>
<p>Vector of colors used in the heatmap.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heat map of drug targets expression.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-Gettest("GEP")
label&lt;-Gettest("label")
# Run the function
plotTargetheatmap("DB00780",GEP,label)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
