<!DOCTYPE html><html lang="en"><head><title>Help for package dslabs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dslabs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#admissions'><p>Gender bias among graduate school admissions to UC Berkeley.</p></a></li>
<li><a href='#brca'><p>Breast Cancer Wisconsin Diagnostic Dataset from UCI Machine Learning Repository</p></a></li>
<li><a href='#brexit_polls'><p>Brexit Poll Data</p></a></li>
<li><a href='#death_prob'><p>2015 US Period Life Table</p></a></li>
<li><a href='#divorce_margarine'><p>Divorce rate and margarine consumption data</p></a></li>
<li><a href='#ds_theme_set'><p>dslabs theme set</p></a></li>
<li><a href='#gapminder'><p>Gapminder Data</p></a></li>
<li><a href='#greenhouse_gases'><p>Greenhouse gas concentrations over 2000 years</p></a></li>
<li><a href='#heights'><p>Self-reported Heights in Inches</p></a></li>
<li><a href='#historic_co2'><p>Atmospheric carbon dioxide concentration over 800,000 years</p></a></li>
<li><a href='#mice_weights'><p>Mice weights</p></a></li>
<li><a href='#mnist_127'><p>Useful example for illustrating generative models based on MNIST data</p></a></li>
<li><a href='#mnist_27'><p>Useful example for illustrating machine learning algorithms based on MNIST data</p></a></li>
<li><a href='#movielens'><p>Movie ratings</p></a></li>
<li><a href='#murders'><p>US gun murders by state for 2010</p></a></li>
<li><a href='#na_example'><p>Count data with some missing values</p></a></li>
<li><a href='#nyc_regents_scores'><p>NYC Regents exams scores 2010</p></a></li>
<li><a href='#olive'><p>Italian olive</p></a></li>
<li><a href='#outlier_example'><p>Adult male heights in feet with outliers</p></a></li>
<li><a href='#polls_2008'><p>Poll data for popular vote in 2008 presidential election</p></a></li>
<li><a href='#polls_us_election_2016'><p>Fivethirtyeight 2016 Poll Data</p></a></li>
<li><a href='#pr_death_counts'><p>Puerto Rico daily mortality</p></a></li>
<li><a href='#read_mnist'><p>Download and read the mnist dataset</p></a></li>
<li><a href='#reported_heights'><p>Self-reported Heights</p></a></li>
<li><a href='#research_funding_rates'><p>Gender bias in research funding in the Netherlands</p></a></li>
<li><a href='#rfalling_object'><p>Simulate falling object data</p></a></li>
<li><a href='#stars'><p>Physical Properties of Stars</p></a></li>
<li><a href='#take_poll'><p>Models results from taking a poll</p></a></li>
<li><a href='#temp_carbon'><p>Global temperature anomaly and carbon emissions, 1751-2018</p></a></li>
<li><a href='#tissue_gene_expression'><p>Gene expression profiles for 189 biological samples taken from seven different tissue types.</p></a></li>
<li><a href='#trump_tweets'><p>Trump Tweets from2009 to 2017</p></a></li>
<li><a href='#us_contagious_diseases'><p>Contagious disease data for US states</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Data Science Labs</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Datasets and functions that can be used for data analysis practice, homework and projects in data science courses and workshops. 26 datasets are available for case studies in data visualization, statistical inference, modeling, linear regression, data wrangling and machine learning.</td>
</tr>
<tr>
<td>Author:</td>
<td>Rafael A. Irizarry, Amy Gill</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rafael A. Irizarry &lt;rafael_irizarry@dfci.harvard.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-01 15:20:18 UTC; rafa</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-01 16:42:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='admissions'>Gender bias among graduate school admissions to UC Berkeley.</h2><span id='topic+admissions'></span>

<h3>Description</h3>

<p>The admission data for six majors for the fall of 1973; often used as an example of Simpson's paradox
</p>


<h3>Usage</h3>

<pre><code class='language-R'>admissions
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> major. The major or university department.
</p>
</li>
<li><p> gender. Men or women.
</p>
</li>
<li><p> admitted. Percent of students admitted.
</p>
</li>
<li><p> applicants. Total number of applicants.
</p>
</li></ul>



<h3>Source</h3>

<p>PJ Bickel, EA Hammel, and JW O'Connell. Science (1975)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>admissions
</code></pre>

<hr>
<h2 id='brca'>Breast Cancer Wisconsin Diagnostic Dataset from UCI Machine Learning Repository</h2><span id='topic+brca'></span>

<h3>Description</h3>

<p>Biopsy features for classification of 569 malignant (cancer) and benign (not cancer) breast masses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brca
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code>.
</p>


<h3>Details</h3>

<p>Features were computationally extracted from digital images of fine needle aspirate
biopsy slides. Features correspond to properties of cell nuclei, such as size, shape and regularity.
The mean, standard error, and worst value of each of 10 nuclear parameters is reported for a total of 30 features.
</p>
<p>This is a classic dataset for training and benchmarking machine learning algorithms.
</p>

<ul>
<li><p> y. The outcomes. A factor with two levels denoting whether a mass is malignant (&quot;M&quot;) or benign (&quot;B&quot;).
</p>
</li>
<li><p> x. The predictors. A matrix with the mean, standard error and worst value of each of 10 nuclear measurements on the slide, for 30 total features per biopsy:
</p>

<ul>
<li><p> radius. Nucleus radius (mean of distances from center to points on perimeter).
</p>
</li>
<li><p> texture. Nucleus texture (standard deviation of grayscale values).
</p>
</li>
<li><p> perimeter. Nucleus perimeter.
</p>
</li>
<li><p> area. Nucleus area.
</p>
</li>
<li><p> smoothness. Nucleus smoothness (local variation in radius lengths).
</p>
</li>
<li><p> compactness. Nucleus compactness (perimeter^2/area - 1).
</p>
</li>
<li><p> concavity, Nucleus concavity (severity of concave portions of the contour).
</p>
</li>
<li><p> concave_pts. Number of concave portions of the nucleus contour.
</p>
</li>
<li><p> symmetry. Nucleus symmetry.
</p>
</li>
<li><p> fractal_dim. Nucleus fractal dimension (&quot;coastline approximation&quot; -1).
</p>
</li></ul>

</li></ul>



<h3>Source</h3>

<p><a href="https://archive.ics.uci.edu/ml/datasets/">UCI Machine Learning Repository</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(brca$y)
dim(brca$x)
head(brca$x)
</code></pre>

<hr>
<h2 id='brexit_polls'>Brexit Poll Data</h2><span id='topic+brexit_polls'></span>

<h3>Description</h3>

<p>Brexit (EU referendum) poll outcomes for 127 polls from January 2016 to the referendum date on June 23, 2016.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brexit_polls
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> startdate. Start date of poll.
</p>
</li>
<li><p> enddate. End date of poll.
</p>
</li>
<li><p> pollster. Pollster conducting the poll.
</p>
</li>
<li><p> poll_type. Online or telephone poll.
</p>
</li>
<li><p> samplesize. Sample size of poll.
</p>
</li>
<li><p> remain. Proportion voting Remain.
</p>
</li>
<li><p> leave. Proportion voting Leave.
</p>
</li>
<li><p> undecided. Proportion of undecided voters.
</p>
</li>
<li><p> spread. Spread calculated as remain - leave.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://en.wikipedia.org/w/index.php?title=Opinion_polling_for_the_United_Kingdom_European_Union_membership_referendum&amp;oldid=896735054/">Wikipedia</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(brexit_polls)
</code></pre>

<hr>
<h2 id='death_prob'>2015 US Period Life Table</h2><span id='topic+death_prob'></span>

<h3>Description</h3>

<p>Probability of death within 1 year by age and sex in the United States in 2015.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>death_prob
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> age. Age strata, with each year a different stratum.
</p>
</li>
<li><p> sex. Male or Female.
</p>
</li>
<li><p> prob. Probability of death within 1 year given exact age and sex.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://www.ssa.gov/oact/STATS/table4c6.html">Social Security Administraton</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(death_prob)
</code></pre>

<hr>
<h2 id='divorce_margarine'>Divorce rate and margarine consumption data</h2><span id='topic+divorce_margarine'></span>

<h3>Description</h3>

<p>Divorce rates in Maine and per capita consumption of margarine in US data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divorce_margarine
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> divorce_rate_maine. Divorce per 1000 in Maine.
</p>
</li>
<li><p> margarine_consumption_per_capita. US per capita consumption of margarine in pounds.
</p>
</li>
<li><p> year. Year.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://www.tylervigen.com/spurious-correlations">Spurious Correlations</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(divorce_margarine, plot(margarine_consumption_per_capita, divorce_rate_maine))
</code></pre>

<hr>
<h2 id='ds_theme_set'>dslabs theme set</h2><span id='topic+ds_theme_set'></span>

<h3>Description</h3>

<p>This function sets a ggplot2 theme used throughout the data science labs. It can be called without arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ds_theme_set(
  new = "theme_bw",
  args = NULL,
  base_size = 11,
  bold_title = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ds_theme_set_+3A_new">new</code></td>
<td>
<p>a prebuilt ggplot2 theme. Defaults to <code>"theme_minimal"</code></p>
</td></tr>
<tr><td><code id="ds_theme_set_+3A_args">args</code></td>
<td>
<p>the arguments to be passed along to the ggplot2 theme function. Defaults to <code>"NULL"</code>.</p>
</td></tr>
<tr><td><code id="ds_theme_set_+3A_base_size">base_size</code></td>
<td>
<p>if <code>"args"</code> is <code>"NULL"</code>, base_size is one of the arguments passed to the theme function. It defaults to 11.</p>
</td></tr>
<tr><td><code id="ds_theme_set_+3A_bold_title">bold_title</code></td>
<td>
<p>if TRUE, sets titles to be bold</p>
</td></tr>
<tr><td><code id="ds_theme_set_+3A_...">...</code></td>
<td>
<p>additional arguments to be used by theme</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
ds_theme_set()
qplot(hp, mpg, data=mtcars, color=am, facets=gear~cyl, 
main="Scatterplots of MPG vs. Horsepower", 
xlab="Horsepower", ylab="Miles per Gallon")

</code></pre>

<hr>
<h2 id='gapminder'>Gapminder Data</h2><span id='topic+gapminder'></span><span id='topic+oecd'></span><span id='topic+opec'></span>

<h3>Description</h3>

<p>Health and income outcomes for 184 countries from 1960 to 2016. Also includes two character vectors, <code>oecd</code> 
and <code>opec</code>, with the names of OECD and OPEC countries from 2016.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gapminder
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> country.  
</p>
</li>
<li><p> year.
</p>
</li>
<li><p> infant_mortality. Infant deaths per 1000.
</p>
</li>
<li><p> life_expectancy. Life expectancy in years.
</p>
</li>
<li><p> fertility. Average number of children per woman. 
</p>
</li>
<li><p> population. Country population.
</p>
</li>
<li><p> gpd. GDP according to World Bankdev.
</p>
</li>
<li><p> continent. 
</p>
</li>
<li><p> region. Geographical region.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>head(gapminder)
print(oecd)
print(opec)
</code></pre>

<hr>
<h2 id='greenhouse_gases'>Greenhouse gas concentrations over 2000 years</h2><span id='topic+greenhouse_gases'></span>

<h3>Description</h3>

<p>Concentrations of the three main greenhouse gases carbon dioxide, methane and nitrous oxide. Measurements are from the Law Dome Ice Core in Antarctica. Selected measurements are provided every 20 years from 1-2000 CE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>greenhouse_gases
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> year. Year (CE).
</p>
</li>
<li><p> gas. Gas being measured: carbon dioxide ('CO2'), methane ('CH4') or nitrous oxide ('N2O'). 
</p>
</li>
<li><p> concentration. Gas concentration in ppm by volume ('CO2') or ppb by volume ('CH4', 'N2O').
</p>
</li></ul>



<h3>Source</h3>

<p>MacFarling Meure et al. 2006 via <a href="https://www.ncei.noaa.gov/access/paleo-search/study/9959/">NOAA</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(greenhouse_gases)
</code></pre>

<hr>
<h2 id='heights'>Self-reported Heights in Inches</h2><span id='topic+heights'></span>

<h3>Description</h3>

<p>Self-reported heights and sex. The heights were converted to inches from the original data included in <code><a href="#topic+reported_heights">reported_heights</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heights
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> sex. A factor with the self-reported sex. 
</p>
</li>
<li><p> height. A numeric vector with self-reported heights in inches. 
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+reported_heights">reported_heights</a></code> for the original data source.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(heights)
</code></pre>

<hr>
<h2 id='historic_co2'>Atmospheric carbon dioxide concentration over 800,000 years</h2><span id='topic+historic_co2'></span>

<h3>Description</h3>

<p>Concentration of carbon dioxide in ppm by volume from direct measurements at Mauna Loa (1959-2018 CE) and indirect measurements from a series of Antarctic ice cores (approx. -800,000-2001 CE).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>historic_co2
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> year. Year (CE).
</p>
</li>
<li><p> co2. Carbon dioxide concentration in ppm by volume.
</p>
</li>
<li><p> source. Source of carbon dioxide measurement: direct CO2 annual mean concentrations from Mauna Loa ('Mauna Loa') or indirect CO2 concentrations from air trapped in ice cores ('Ice Cores').
</p>
</li></ul>



<h3>Source</h3>

<p>Mauna Loa data from <a href="https://gml.noaa.gov/ccgg/trends/data.html">NOAA</a>. Ice core data from Bereiter et al. 2015 via <a href="https://gml.noaa.gov/ccgg/trends/data.html">NOAA</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(historic_co2)
</code></pre>

<hr>
<h2 id='mice_weights'>Mice weights</h2><span id='topic+mice_weights'></span>

<h3>Description</h3>

<p>Body weights, bone density, and percent fat for mice under two diets: chow and high fat. Data provided by Karen Svenson from Jackson Laboratories. Funding to generate these data came from NIH grant P50 GM070683 awarded to Gary Churchill.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mice_weights
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> body_weight. Body weight in grams at 19 weeks.
</p>
</li>
<li><p> bone_density. Body density.
</p>
</li>
<li><p> percent_fat. Percent fat.
</p>
</li>
<li><p> sex. The sex of the mice.
</p>
</li>
<li><p> diet. The diet of the mice: chow or high fat.
</p>
</li>
<li><p> gen. These are outbread mice. This variable denotes the generation.
</p>
</li>
<li><p> litter. Which of two litters mice belong to.
</p>
</li></ul>



<h3>Source</h3>

<p>Karen Svenson, Daniel M. Gatti, and Gary Churchill from Jackson Laboratories.
</p>


<h3>References</h3>

<p>Daniel M. Gatti, Petr Simecek, Lisa Somes, Clifton T. Jeffrey, Matthew J. Vincent, Kwangbom Choi, Xingyao Chen, Gary A. Churchill, and Karen L. Svenson. &quot;The Effects of Sex and Diet on Physiology and Liver Gene Expression in Diversity Outbred Mice&quot;. bioRxiv 098657; <a href="https://doi.org/10.1101/098657">doi:10.1101/098657</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mice_weights |&gt; head()
with(mice_weights, table(sex, diet))
</code></pre>

<hr>
<h2 id='mnist_127'>Useful example for illustrating generative models based on MNIST data</h2><span id='topic+mnist_127'></span>

<h3>Description</h3>

<p>We only include a randomly selected set of 1s, 2s and 7s along with the two predictors based on the 
proportion of dark pixels in the upper left and lower right quadrants respectively. The dataset is divided into 
training and test sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mnist_127
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> train. A data frame containing  training data: labels and predictors.
</p>
</li>
<li><p> test. A data frame containing  test data: labels and predictors.
</p>
</li>
<li><p> index_train. The index of the original mnist training data used for the training set.
</p>
</li>
<li><p> index_test. The index of the original mnist test data used for the test set.
</p>
</li></ul>



<h3>References</h3>

<p>Y. LeCun, L. Bottou, Y. Bengio, and P. Haffner. &quot;Gradient-based learning applied to document recognition.&quot; Proceedings of the IEEE, 86(11):2278-2324, November 1998.
</p>


<h3>See Also</h3>

<p>[read_mnist(), mnist_27]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(mnist_127$train, plot(x_1, x_2, col = as.numeric(y)))

</code></pre>

<hr>
<h2 id='mnist_27'>Useful example for illustrating machine learning algorithms based on MNIST data</h2><span id='topic+mnist_27'></span>

<h3>Description</h3>

<p>We only include a randomly selected set of 2s and 7s along with the two predictors based on the 
proportion of dark pixels in the upper left and lower right quadrants respectively. The dataset is divided into 
training and test sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mnist_27
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> train. A data frame containing  training data: labels and predictors.
</p>
</li>
<li><p> test. A data frame containing  test data: labels and predictors.
</p>
</li>
<li><p> index_train. The index of the original mnist training data used for the training set.
</p>
</li>
<li><p> index_test. The index of the original mnist test data used for the test set.
</p>
</li>
<li><p> true_p. A <code>data.frame</code> containing the 
two predictors <code>x_1</code> and <code>x_2</code> and the conditional probability of being a 7 
for <code>x_1</code>, <code>x_2</code>.
</p>
</li></ul>



<h3>References</h3>

<p>Y. LeCun, L. Bottou, Y. Bengio, and P. Haffner. &quot;Gradient-based learning applied to document recognition.&quot; Proceedings of the IEEE, 86(11):2278-2324, November 1998.
</p>


<h3>See Also</h3>

<p>[read_mnist()]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(mnist_27$train, plot(x_1, x_2, col = as.numeric(y)))

</code></pre>

<hr>
<h2 id='movielens'>Movie ratings</h2><span id='topic+movielens'></span>

<h3>Description</h3>

<p>MovieLens Latest Dataset (Small)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>movielens
</code></pre>


<h3>Format</h3>

<p>Two object of class <code>data.frame</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> movieId. Unique ID for the movie.
</p>
</li>
<li><p> title. Movie title (not unique). 
</p>
</li>
<li><p> year. Year the movie was released.
</p>
</li>
<li><p> genres. Genres associated with the movie.
</p>
</li>
<li><p> userId. Unique ID for the user. 
</p>
</li>
<li><p> rating. A rating between 0 and 5 for the movie.
</p>
</li>
<li><p> timestamp. Date and time the rating was given.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://files.grouplens.org/datasets/movielens/ml-latest-small.zip">https://files.grouplens.org/datasets/movielens/ml-latest-small.zip</a>
</p>


<h3>References</h3>

<p>F. Maxwell Harper and Joseph A. Konstan. 2015. The MovieLens Datasets: History and Context. ACM Transactions on Interactive Intelligent Systems (TiiS) 5, 4, Article 19 (December 2015), 19 pages. DOI=https://dx.doi.org/10.1145/2827872
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(movielens)
</code></pre>

<hr>
<h2 id='murders'>US gun murders by state for 2010</h2><span id='topic+murders'></span>

<h3>Description</h3>

<p>Gun murder data from FBI reports. Also contains the population of each state.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>murders
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> state. US state
</p>
</li>
<li><p> abb. Abbreviation of US state
</p>
</li>
<li><p> region. Geographical US region
</p>
</li>
<li><p> population. State population (2010)
</p>
</li>
<li><p> total. Number of gun murders in state (2010)
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://en.wikipedia.org/wiki/Gun_violence_in_the_United_States_by_state">Wikipedia</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(murders)
</code></pre>

<hr>
<h2 id='na_example'>Count data with some missing values</h2><span id='topic+na_example'></span>

<h3>Description</h3>

<p>This dataset was randomly generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>na_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"integer"</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(sum(is.na(na_example)))
</code></pre>

<hr>
<h2 id='nyc_regents_scores'>NYC Regents exams scores 2010</h2><span id='topic+nyc_regents_scores'></span>

<h3>Description</h3>

<p>Distribution of scores for New York City Regents algebra, global history, biology, English, and U.S. history exams. 
These data were used to make <a href="http://graphics8.nytimes.com/images/2011/02/19/nyregion/19schoolsch/19schoolsch-popup.gif">this New York Times plot</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nyc_regents_scores
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> score. Test score from 0 to 100.
</p>
</li>
<li><p> integrated_algebra. Score frequency on Algebra exam.
</p>
</li>
<li><p> global_history. Score frequency on global history exam.
</p>
</li>
<li><p> living_environment.  Score frequency on biology exam.
</p>
</li>
<li><p> english. Score frequency on English exam.
</p>
</li>
<li><p> us_history. Score frequency on  U.S. history exam.
</p>
</li></ul>



<h3>Source</h3>

<p>New York City Department of Education via Amanda Cox.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(nyc_regents_scores)
</code></pre>

<hr>
<h2 id='olive'>Italian olive</h2><span id='topic+olive'></span>

<h3>Description</h3>

<p>Composition in percentage of eight fatty acids found in the lipid fraction of 572 Italian olive oils
</p>


<h3>Usage</h3>

<pre><code class='language-R'>olive
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> region. General region of Italy.
</p>
</li>
<li><p> area. Area of Italy. 
</p>
</li>
<li><p> palmitic. Percent palmitic acid of sample. 
</p>
</li>
<li><p> palmitoleic. Percent palmitoleic of sample. 
</p>
</li>
<li><p> stearic. Percent stearic acid of sample.  
</p>
</li>
<li><p> oleic. Percent oleic acid of sample.  
</p>
</li>
<li><p> linoleic. Percent linoleic acid of sample.  
</p>
</li>
<li><p> linolenic. Percent linolenic acid of sample.  
</p>
</li>
<li><p> arachidic. Percent arachidic acid of sample.  
</p>
</li>
<li><p> eicosenoic. Percent eicosenoic acid of sample.  
</p>
</li></ul>



<h3>Source</h3>

<p>J. Zupan, and J. Gasteiger.  Neural Networks in Chemistry and Drug Design.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(olive)
</code></pre>

<hr>
<h2 id='outlier_example'>Adult male heights in feet with outliers</h2><span id='topic+outlier_example'></span>

<h3>Description</h3>

<p>This dataset was randomly generated with a normal distribution (average: 5 feet 9 inches, standard deviation: 3 inches). One value was changed to be mistakenly reported in centimeters rather than feet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outlier_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"numeric"</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mean(outlier_example)
median(outlier_example)
</code></pre>

<hr>
<h2 id='polls_2008'>Poll data for popular vote in 2008 presidential election</h2><span id='topic+polls_2008'></span>

<h3>Description</h3>

<p>Data from different pollsters for the popular vote between Obama and McCain 
in the 2008 presidential election.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polls_2008
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> day. Days until election day. Negative numbers are reported so that days can increase up to 0, which is election day.
</p>
</li>
<li><p> margin. Average difference between Obama and McCain for that day.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://web.archive.org/web/20161108190914/http://www.pollster.com/08USPresGEMvO-2.html">https://web.archive.org/web/20161108190914/http://www.pollster.com/08USPresGEMvO-2.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>with(polls_2008, plot(day, margin))

</code></pre>

<hr>
<h2 id='polls_us_election_2016'>Fivethirtyeight 2016 Poll Data</h2><span id='topic+polls_us_election_2016'></span><span id='topic+results_us_election_2016'></span>

<h3>Description</h3>

<p>Poll results from US 2016 presidential elections aggregated from HuffPost Pollster, RealClearPolitics, polling firms, and news reports. 
The dataset also includes election results (popular vote) and electoral college votes in <code>results_us_election_2016</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polls_us_election_2016
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> state. State in which poll was taken. 'U.S' is for national polls.
</p>
</li>
<li><p> startdate. Poll's start date.
</p>
</li>
<li><p> enddate. Poll's end date.
</p>
</li>
<li><p> pollster. Pollster conducting the poll.
</p>
</li>
<li><p> grade. Grade assigned by fivethirtyeight to pollster.
</p>
</li>
<li><p> samplesize. Sample size. 
</p>
</li>
<li><p> population. Type of population being polled.
</p>
</li>
<li><p> rawpoll_clinton. Percentage for Hillary Clinton.
</p>
</li>
<li><p> rawpoll_trump. Percentage for Donald Trump
</p>
</li>
<li><p> rawpoll_johnson. Percentage for Gary Johnson
</p>
</li>
<li><p> rawpoll_mcmullin. Percentage for Evan McMullin.
</p>
</li>
<li><p> adjpoll_clinton. Fivethirtyeight adjusted percentage for Hillary Clinton.
</p>
</li>
<li><p> ajdpoll_trump. Fivethirtyeight adjusted percentage for Donald Trump
</p>
</li>
<li><p> adjpoll_johnson. Fivethirtyeight adjusted percentage for Gary Johnson
</p>
</li>
<li><p> adjpoll_mcmullin. Fivethirtyeight adjusted percentage for Evan McMullin.
</p>
</li></ul>



<h3>Source</h3>

<p>The original csv file used to create <code>polls_us_election_2016</code> is here: <a href="https://projects.fivethirtyeight.com/general-model/president_general_polls_2016.csv">https://projects.fivethirtyeight.com/general-model/president_general_polls_2016.csv</a> 
</p>
<p>The data for <code>results_us_election_2016</code> is from Ballotpedia and can be found here:
<a href="https://docs.google.com/spreadsheets/d/1zxyOQDjNOJS_UkzerorUCf2OAdcMcIQEwRciKuYBIZ4/pubhtml?widget=true&amp;headers=false#gid=658726802/">https://docs.google.com/spreadsheets/d/1zxyOQDjNOJS_UkzerorUCf2OAdcMcIQEwRciKuYBIZ4/pubhtml?widget=true&amp;headers=false#gid=658726802/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(polls_us_election_2016)
</code></pre>

<hr>
<h2 id='pr_death_counts'>Puerto Rico daily mortality</h2><span id='topic+pr_death_counts'></span>

<h3>Description</h3>

<p>A data frame with Puerto Rico daily mortality counts 2015 to May 2018.
This includes the day hurricanes Maria made 2017-09-20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pr_death_counts
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> date. Date of the count.
</p>
</li>
<li><p> deaths. Number of deaths reported that day.
</p>
</li></ul>



<h3>Source</h3>

<p>Puerto Rico Demographic Registry. Data was extracted from PDF provided in 'system.file(&quot;extdata&quot;, &quot;RD-Mortality-Report_2015-18-180531.pdf&quot;, package = &quot;dslabs&quot;)'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
with(pr_death_counts, plot(date, deaths))

</code></pre>

<hr>
<h2 id='read_mnist'>Download and read the mnist dataset</h2><span id='topic+read_mnist'></span>

<h3>Description</h3>

<p>This function downloads the mnist training and test data available here http://yann.lecun.com/exdb/mnist/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_mnist(
  path = NULL,
  download = FALSE,
  destdir = tempdir(),
  url = "https://www2.harvardx.harvard.edu/courses/IDS_08_v2_03/",
  keep.files = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_mnist_+3A_path">path</code></td>
<td>
<p>A character giving the full path of the directory to look for files. It assumes the filenames are the same as the originals. If path is <code>NULL</code> a download or direct read of the files is attempted.</p>
</td></tr>
<tr><td><code id="read_mnist_+3A_download">download</code></td>
<td>
<p>If <code>TRUE</code> the files will be downloaded and saved in detsdir.</p>
</td></tr>
<tr><td><code id="read_mnist_+3A_destdir">destdir</code></td>
<td>
<p>A character giving the full path of the directory in which to save the downloaded files. The default is to use a temporary directory.</p>
</td></tr>
<tr><td><code id="read_mnist_+3A_url">url</code></td>
<td>
<p>A character giving the URL from which to download files. Currently a copy of the data is available at https://www2.harvardx.harvard.edu/courses/IDS_08_v2_03/, the current default URL.</p>
</td></tr>
<tr><td><code id="read_mnist_+3A_keep.files">keep.files</code></td>
<td>
<p>A logical. If <code>TRUE</code> the downloaded files will be saved in <code>destdir</code>. If <code>FALSE</code> the entire directory is erased. This argument is ignored if <code>download</code> is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two components: train and test. Each of these is a list with two components: images and labels. The images component is a
matrix with each column representing one of the 28*28 = 784 pixels. The values are integers between 0 and 255 representing grey scale. 
The labels components is a vector representing the digit shown in the image.
</p>
<p>Note that the data is over 10MB, so the download may take several seconds depending on internet speed. If you plan to load the data more than once
we recommend you download the data once and read it from disk in the future. See examples.
</p>


<h3>Author(s)</h3>

<p>Samuela Pollack
</p>
<p>Rafael A. Irizarry, <a href="mailto:rafael_irizarry@dfci.harvard.edu">rafael_irizarry@dfci.harvard.edu</a>
</p>


<h3>References</h3>

<p>Y. LeCun, L. Bottou, Y. Bengio, and P. Haffner. &quot;Gradient-based learning applied to document recognition.&quot; Proceedings of the IEEE, 86(11):2278-2324, November 1998.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># this can take several seconds, depending on internet speed.

## Not run: 
mnist &lt;- read_mnist()
i &lt;- 5
image(1:28, 1:28, matrix(mnist$test$images[i,], nrow=28)[ , 28:1], 
    col = gray(seq(0, 1, 0.05)), xlab = "", ylab="")
## the labels for this image is: 
mnist$test$labels[i]

## End(Not run)

# You can download and save the data to a directory like this:
## Not run: 
mnist &lt;- read_mnist(download = TRUE, destdir = "~/Downloads")

# and then, going forward, read from disk 
mnist &lt;- read_mnist("~/Downloads")

## End(Not run)
</code></pre>

<hr>
<h2 id='reported_heights'>Self-reported Heights</h2><span id='topic+reported_heights'></span>

<h3>Description</h3>

<p>Students were asked to report their height (in inches) and sex in an anonymous online form. 
This table includes the results from combining data from four courses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reported_heights
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> time_stamp. Time and date of the entry.
</p>
</li>
<li><p> sex. Sex as reported by the students. 
</p>
</li>
<li><p> height. Height as reported by student by filling in a text free box. 
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>head(reported_heights)
</code></pre>

<hr>
<h2 id='research_funding_rates'>Gender bias in research funding in the Netherlands</h2><span id='topic+research_funding_rates'></span><span id='topic+raw_data_research_funding_rates'></span>

<h3>Description</h3>

<p>Table S1 from paper title &quot;Gender contributes to personal research funding success in The Netherlands&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>research_funding_rates
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> discipline. Research area discipline.
</p>
</li>
<li><p> applications_total. Total applications.
</p>
</li>
<li><p> applications_men. Total applications by men.
</p>
</li>
<li><p> applications_women. Total applications by women.
</p>
</li>
<li><p> awards_total. Total awards.
</p>
</li>
<li><p> awards_men. Total awards received by men.
</p>
</li>
<li><p> awards_women. Total awards received by women.
</p>
</li>
<li><p> success_rates_total. Overall success rate.
</p>
</li>
<li><p> success_rates_men. Success rate for men.
</p>
</li>
<li><p> success_rates_women. Success rate for women.
</p>
</li></ul>



<h3>References</h3>

<p>van der Lee R, Ellemers N. Gender contributes to personal research funding success in The Netherlands. Proc Natl Acad Sci U S A. 2015 Oct 6;112(40):12349-53. doi: 10.1073/pnas.1510159112. Epub 2015 Sep 21. PMID: 26392544; PMCID: PMC4603485.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>research_funding_rates
# The raw data for this table is available from
invisible(raw_data_research_funding_rates)

</code></pre>

<hr>
<h2 id='rfalling_object'>Simulate falling object data</h2><span id='topic+rfalling_object'></span>

<h3>Description</h3>

<p>The function simulates a falling object's position. Default parameters are for dropping a 
weight from the tower of Pisa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfalling_object(
  n = 14,
  d_0 = 55.86,
  v_0 = 0,
  g = -9.8,
  scale = 1,
  time = seq(0, 3.25, length.out = n),
  error_distribution = c("rnorm", "rt"),
  df = 3
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rfalling_object_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_d_0">d_0</code></td>
<td>
<p>Height from which object will fall in meters.</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_v_0">v_0</code></td>
<td>
<p>Initial velocity with which object will fall in meters per second.</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_g">g</code></td>
<td>
<p>Gravitational constant, 9.8 meters per second per seonnd</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_scale">scale</code></td>
<td>
<p>The measurement errors will be multiplied by this constant.</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_time">time</code></td>
<td>
<p>Numeric vector of times, in seconds, at which measurements were taken.</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_error_distribution">error_distribution</code></td>
<td>
<p>Character. Either <code>rnorm</code> for normal or <code>rt</code> for t-distribution.</p>
</td></tr>
<tr><td><code id="rfalling_object_+3A_df">df</code></td>
<td>
<p>If using t-distribution, the degrees of freedom.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the time, the distance travelled, and the observed distance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dat &lt;- rfalling_object()
with(dat, plot(time, observed_distance))
with(dat, lines(time, distance, col = "blue"))

</code></pre>

<hr>
<h2 id='stars'>Physical Properties of Stars</h2><span id='topic+stars'></span>

<h3>Description</h3>

<p>Physical properties of selected stars, including luminosity, temperature, and spectral class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stars
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> star. Name of star.
</p>
</li>
<li><p> magnitude. Absolute magnitude of the star, which is a function of the star's luminosity and distance to the star.
</p>
</li>
<li><p> temp. Surface temperature in degrees Kelvin (K).
</p>
</li>
<li><p> type. Spectral class of star in the OBAFGKM system.
</p>
</li></ul>



<h3>Source</h3>

<p>Compiled from multiple open-access references on <a href="http://vizier.u-strasbg.fr/viz-bin/VizieR">VizieR</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(stars)
</code></pre>

<hr>
<h2 id='take_poll'>Models results from taking a poll</h2><span id='topic+take_poll'></span>

<h3>Description</h3>

<p>The function shows a plot of a random sample drawn from an urn with blue and red beads. The sample is taken with replacement. The proportion of blue beads is not shown so that students can try to estimate it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>take_poll(n, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="take_poll_+3A_n">n</code></td>
<td>
<p>Sample size</p>
</td></tr>
<tr><td><code id="take_poll_+3A_...">...</code></td>
<td>
<p>additional arguments to be used by the function <code>sample</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>take_poll(25)

</code></pre>

<hr>
<h2 id='temp_carbon'>Global temperature anomaly and carbon emissions, 1751-2018</h2><span id='topic+temp_carbon'></span>

<h3>Description</h3>

<p>Annual mean global temperature anomaly on land, sea and combined, 1880-2018. Annual global carbon emissions, 1751-2014.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>temp_carbon
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> year. Year (CE).
</p>
</li>
<li><p> temp_anomaly. Global annual mean temperature anomaly in degrees Celsius relative to the 20th century mean temperature. 1880-2018. 
</p>
</li>
<li><p> land_anomaly. Annual mean temperature anomaly on land in degrees Celsius relative to the 20th century mean temperature. 1880-2018.
</p>
</li>
<li><p> ocean_anomaly. Annual mean temperature anomaly over ocean in degrees Celsius relative to the 20th century mean temperature. 1880-2018.
</p>
</li>
<li><p> carbon_emissions. Annual carbon emissions in millions of metric tons of carbon. 1751-2014.
</p>
</li></ul>



<h3>Source</h3>

<p>NOAA and Boden, T.A., G. Marland, and R.J. Andres (2017) via CDIAC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(temp_carbon)
</code></pre>

<hr>
<h2 id='tissue_gene_expression'>Gene expression profiles for 189 biological samples taken from seven different tissue types.</h2><span id='topic+tissue_gene_expression'></span>

<h3>Description</h3>

<p>This is a subset of the data provided by the <code>tissuesGeneExpression</code>
package available from the <code>genomicsclass</code> GitHub repository. 
The predictors are gene expression measurements from 500 genes that
are a random subset of the original 22,215.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue_gene_expression
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code>.
</p>


<h3>Details</h3>

<p>The example dataset is recommended for illustrating
clustering and machine learning techniques.
</p>

<ul>
<li><p> x. The predictors composed of 500 genes. Each row is a gene expression profile and each column is different gene. The column names are the gene symbols.
</p>
</li>
<li><p> y. The outcomes. A character vector representing the tissue. One of seven tissue types.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://github.com/genomicsclass/tissuesGeneExpression">https://github.com/genomicsclass/tissuesGeneExpression</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(tissue_gene_expression$y)
dim(tissue_gene_expression$x)
</code></pre>

<hr>
<h2 id='trump_tweets'>Trump Tweets from2009 to 2017</h2><span id='topic+trump_tweets'></span><span id='topic+sentiment_counts'></span>

<h3>Description</h3>

<p>This dataset contains all tweets from Donald Trump's Twitter account from 2009 to 2017. Additionally, the results of a sentiment analysis, conducted on tweets from the campaign period (2015-06-17 to 2016-11-08), are included in <code>sentiment_counts</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trump_tweets
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> source. Device or service used to compose tweet.
</p>
</li>
<li><p> id_str. Tweet ID.
</p>
</li>
<li><p> text. Tweet.
</p>
</li>
<li><p> created_at. Data and time tweet was tweeted.
</p>
</li>
<li><p> retweet_count. How many times tweet had been retweeted at time dataset was created.
</p>
</li>
<li><p> in_reply_to_user_id_str. If a reply, the user id of person being replied to.
</p>
</li>
<li><p> favorite_count. Number of times tweet had been favored at time dataset was created.
</p>
</li>
<li><p> is_retweet. A logical telling us if it is a retweet or not.
</p>
</li></ul>



<h3>Source</h3>

<p>The Trump Twitter Archive: <a href="https://www.thetrumparchive.com/">https://www.thetrumparchive.com/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(trump_tweets)

</code></pre>

<hr>
<h2 id='us_contagious_diseases'>Contagious disease data for US states</h2><span id='topic+us_contagious_diseases'></span>

<h3>Description</h3>

<p>Yearly counts for Hepatitis A, Measles, Mumps, Pertussis, Polio,  Rubella, and Smallpox for US states. Original data courtesy of Tycho Project (http://www.tycho.pitt.edu/).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>us_contagious_diseases
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>.
</p>


<h3>Details</h3>


<ul>
<li><p> disease. A factor containing disease names.
</p>
</li>
<li><p> state. A factor containing state names.
</p>
</li>
<li><p> year. 
</p>
</li>
<li><p> weeks_reporting. Number of weeks counts were reported that year.
</p>
</li>
<li><p> count. Total number of reported cases. 
</p>
</li>
<li><p> population. State population, interpolated for non-census years.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="http://www.tycho.pitt.edu/">Tycho Project</a>
</p>


<h3>References</h3>

<p>Willem G. van Panhuis, John Grefenstette, Su Yon Jung, Nian Shong Chok, Anne Cross, Heather Eng, Bruce Y Lee, Vladimir Zadorozhny, Shawn Brown, Derek Cummings, Donald S. Burke. Contagious Diseases in the United States from 1888 to the present. NEJM 2013; 369(22): 2152-2158.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(us_contagious_diseases)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
