<!DOCTYPE html><html lang="en-US"><head><title>Help for package DWLS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DWLS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#buildSignatureMatrixMAST'><p>Signature Matrix Using MAST</p></a></li>
<li><a href='#buildSignatureMatrixUsingSeurat'><p>Signature Matrix Using Seurat</p></a></li>
<li><a href='#dataSC_3'><p>Single cell data</p></a></li>
<li><a href='#DEAnalysisMAST'><p>DEAnalysisMAST</p></a></li>
<li><a href='#DEAnalysisSeurat'><p>Differential Expression without Idents (Seurat)</p></a></li>
<li><a href='#DEAnalysisSeuratIdents'><p>Differential Expression with Idents (Seurat)</p></a></li>
<li><a href='#findDampeningConstant'><p>findDampeningConstant</p></a></li>
<li><a href='#m.auc'><p>m.auc</p></a></li>
<li><a href='#MASTSignatureMatrixGivenDE'><p>Build Signature matrix using MAST given a differential expression</p>
matrix</a></li>
<li><a href='#Mean.in.log2space'><p>Mean.in.log2space</p></a></li>
<li><a href='#solveDampenedWLS'><p>solveDampenedWLS</p></a></li>
<li><a href='#solveDampenedWLSj'><p>solveDampenedWLSj</p></a></li>
<li><a href='#solveOLS'><p>solveOLS</p></a></li>
<li><a href='#solveOLSInternal'><p>solveOLSInternal</p></a></li>
<li><a href='#solveSVR'><p>solveSVR</p></a></li>
<li><a href='#stat.log2'><p>stat.log2</p></a></li>
<li><a href='#trimData'><p>trimData</p></a></li>
<li><a href='#v.auc'><p>v.auc</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene Expression Deconvolution Using Dampened Weighted Least
Squares</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Adriana Sistig &lt;adriana.sistig@icahn.mssm.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The rapid development of single-cell transcriptomic technologies 
    has helped uncover the cellular heterogeneity within cell populations. 
    However, bulk RNA-seq continues to be the main workhorse for quantifying 
    gene expression levels due to technical simplicity and low cost. To most 
    effectively extract information from bulk data given the new knowledge 
    gained from single-cell methods, we have developed a novel algorithm to 
    estimate the cell-type composition of bulk data from a single-cell 
    RNA-seq-derived cell-type signature. Comparison with existing methods using 
    various real RNA-seq data sets indicates that our new approach is more 
    accurate and comprehensive than previous methods, especially for the 
    estimation of rare cell types. More importantly,our method can detect 
    cell-type composition changes in response to external perturbations, 
    thereby providing a valuable, cost-effective method for dissecting the 
    cell-type-specific effects of drug treatments or condition changes. 
    As such, our method is applicable to a wide range of biological and 
    clinical investigations. Dampened weighted least squares ('DWLS') is an 
    estimation method for gene expression deconvolution, in which the cell-type 
    composition of a bulk RNA-seq data set is computationally inferred. 
    This method corrects common biases towards cell types that are 
    characterized by highly expressed genes and/or are highly prevalent, to 
    provide accurate detection across diverse cell types. See: 
    <a href="https://www.nature.com/articles/s41467-019-10802-z.pdf">https://www.nature.com/articles/s41467-019-10802-z.pdf</a> for more 
    information about the development of 'DWLS' and the methods behind our 
    functions. </td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sistia01/DWLS">https://github.com/sistia01/DWLS</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sistia01/DWLS/issues">https://github.com/sistia01/DWLS/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>quadprog, reshape, Seurat, ROCR, varhandle, dplyr, stats,
utils, e1071, MAST, SummarizedExperiment</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), Matrix (&ge; 1.3.3)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-19 13:37:17 UTC; adrianasistig</td>
</tr>
<tr>
<td>Author:</td>
<td>Daphne Tsoucas [aut],
  Adriana Sistig [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-24 09:20:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='buildSignatureMatrixMAST'>Signature Matrix Using MAST</h2><span id='topic+buildSignatureMatrixMAST'></span>

<h3>Description</h3>

<p>This function builds a signature matrix using genes identified
by the DEAnalysisMAST() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildSignatureMatrixMAST(
  scdata,
  id,
  path,
  diff.cutoff = 0.5,
  pval.cutoff = 0.01,
  f = 200
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildSignatureMatrixMAST_+3A_scdata">scdata</code></td>
<td>
<p>The data</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixMAST_+3A_id">id</code></td>
<td>
<p>The identities of the genes</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixMAST_+3A_path">path</code></td>
<td>
<p>The path to the file results</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixMAST_+3A_diff.cutoff">diff.cutoff</code></td>
<td>
<p>This is automatically set to 0.5</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixMAST_+3A_pval.cutoff">pval.cutoff</code></td>
<td>
<p>This is automatically set to 0.01</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixMAST_+3A_f">f</code></td>
<td>
<p>The maximum number of genes (when creating the signature matrix,
need to reduce number of genes, between 50:f number of significant genes are
chosen). If not set, this number is automatically set to 200.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Signature Matrix built using the MAST algorithm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

labels&lt;-trueLabels

#Change to real labels
newcat&lt;-c("NonCycISC","CycISC","TA","Ent","PreEnt","Goblet","Paneth","Tuft",
"EE")
for (i in 1:length(newcat)){
  labels[which(labels==(i-1))]&lt;-newcat[i]
  }

#Run on local w/ inst/extdata/results folder
#Signature &lt;-
#buildSignatureMatrixMAST(
#dataSC,labels,"inst/extdata/results",diff.cutoff = 0.5,pval.cutoff = 0.01)

</code></pre>

<hr>
<h2 id='buildSignatureMatrixUsingSeurat'>Signature Matrix Using Seurat</h2><span id='topic+buildSignatureMatrixUsingSeurat'></span>

<h3>Description</h3>

<p>This function builds a signature matrix using genes identified
by the DEAnalysis() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildSignatureMatrixUsingSeurat(
  scdata,
  id,
  path,
  diff.cutoff = 0.5,
  pval.cutoff = 0.01,
  f = 200
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_scdata">scdata</code></td>
<td>
<p>The data</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_id">id</code></td>
<td>
<p>The identities of the genes</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_path">path</code></td>
<td>
<p>The path to the file results</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_diff.cutoff">diff.cutoff</code></td>
<td>
<p>This is automatically set to 0.5</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_pval.cutoff">pval.cutoff</code></td>
<td>
<p>The p-value cutoff. This is automatically set to 0.01</p>
</td></tr>
<tr><td><code id="buildSignatureMatrixUsingSeurat_+3A_f">f</code></td>
<td>
<p>The maximum number of genes (when creating the signature matrix,
need to reduce number of genes, between 50:f number of significant genes are
chosen). If not set, this number is automatically set to 200.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Signature Matrix built using the Seurat algorithm
</p>

<hr>
<h2 id='dataSC_3'>Single cell data</h2><span id='topic+dataSC_3'></span>

<h3>Description</h3>

<p>A subset of the dataSC dataset in inst/extdata/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataSC_3
</code></pre>


<h3>Format</h3>

<p>Data
</p>


<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/28467820/">https://pubmed.ncbi.nlm.nih.gov/28467820/</a>
</p>

<hr>
<h2 id='DEAnalysisMAST'>DEAnalysisMAST</h2><span id='topic+DEAnalysisMAST'></span>

<h3>Description</h3>

<p>Perform DE analysis using MAST.
Dampened weighted least squares (DLWS) is an estimation
method for gene expression deconvolution, in which the cell-type
composition of a bulk RNA-seq data set is computationally inferred.
This method corrects common biases towards cell types that are
characterized by highly expressed genes and/or are highly prevalent,
to provide accurate detection across diverse cell types. To begin,
the user must input a bulk RNA-seq data set, along with a labeled
representative single-cell RNA-seq data set that will serve to generate
cell-type-specific gene expression profiles. Ideally, the single-cell
data set will contain cells from all cell types that may be found in the
bulk data. DWLS will return the cell-type composition of the bulk data.
First, solve OLS then use the solution to find a starting point for the
weights. Next, the dampened weighted least squares is performed. The weights
are iterated until convergence then the dampening constant for weights is
found using cross-validation (with decreasing step size for convergence).
</p>
<p>DWLS captures ISC composition changes across conditions.
One of the most important applications of deconvolution methods is in
the identification of cell-type composition variations across conditions.
</p>
<p>Note: The function uses solveDampenedWLSj() and findDampeningConstant().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEAnalysisMAST(scdata, id, path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEAnalysisMAST_+3A_scdata">scdata</code></td>
<td>
<p>The gene expression datafarme</p>
</td></tr>
<tr><td><code id="DEAnalysisMAST_+3A_id">id</code></td>
<td>
<p>The unique identities within the data</p>
</td></tr>
<tr><td><code id="DEAnalysisMAST_+3A_path">path</code></td>
<td>
<p>The path for the RData results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix
The resulting matrix is a gene by cell-type signature matrix.
The cell-type signature matrix is constructed using a representative
single-cell data set, such that all cell types expected in the bulk
data are also represented in the single-cell data (the converse need not be
true). The single-cell data is first clustered to reveal
its constituent cell types.The function will return 3 different files:
an RData file, an rds file, and a csv file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
dest &lt;- "data/dataBulk.RData"
load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

labels&lt;-trueLabels

#Old Method
#load("data/dataBulk.RData") #read in bulk data for WT1 (control condition #1)
#load("data/labels.RData") #read in single-cell labels from clustering
#data('dataSC_3', package = "DWLS")
#dataSC &lt;- dataSC_3

labels&lt;-trueLabels

#Old Method
#load("data/dataBulk.RData") #read in bulk data for WT1 (control condition #1)
#load("data/labels.RData") #read in single-cell labels from clustering
#data('dataSC_3', package = "DWLS")
#dataSC &lt;- dataSC_3

labels&lt;-trueLabels

#Change to real labels
newcat&lt;-c("NonCycISC","CycISC","TA","Ent","PreEnt","Goblet","Paneth","Tuft",
 "EE")
for (i in 1:length(newcat)){
  labels[which(labels==(i-1))]&lt;-newcat[i]
  }

#Run deconvolution
#Results are in inst/extdata/results folder -- run on local
#Example code below
Mast_test &lt;- DEAnalysisMAST(dataSC, labels, "inst/extdata/results")


</code></pre>

<hr>
<h2 id='DEAnalysisSeurat'>Differential Expression without Idents (Seurat)</h2><span id='topic+DEAnalysisSeurat'></span>

<h3>Description</h3>

<p>This function calculates the differential expression values
along with identifying the Idents (through Seurat). The output is
saved in an RData file for each unique identity (id).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEAnalysisSeurat(scdata, id, path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEAnalysisSeurat_+3A_scdata">scdata</code></td>
<td>
<p>The gene expression datafarme</p>
</td></tr>
<tr><td><code id="DEAnalysisSeurat_+3A_id">id</code></td>
<td>
<p>The unique identities within the data</p>
</td></tr>
<tr><td><code id="DEAnalysisSeurat_+3A_path">path</code></td>
<td>
<p>The path for the RData results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An RData and rds file with the differential expression analysis
results for each unique id.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#download.file(url, dest)
#load("data/dataSC.RData")
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

labels&lt;-trueLabels

#Old Method
#load("data/dataBulk.RData") #read in bulk data for WT1 (control condition #1)
#load("data/labels.RData") #read in single-cell labels from clustering
#data('dataSC_3', package = "DWLS")
#dataSC &lt;- dataSC_3

labels&lt;-trueLabels
#Change to real labels
newcat&lt;-c("NonCycISC","CycISC","TA","Ent","PreEnt","Goblet","Paneth","Tuft","EE")
for (i in 1:length(newcat)){
  labels[which(labels==(i-1))]&lt;-newcat[i]
  }
#Run deconvolution -- run on local
#Results in inst/extdata/results
Seurat_DE &lt;- DEAnalysisSeurat(dataSC, labels, "inst/extdata/results")


</code></pre>

<hr>
<h2 id='DEAnalysisSeuratIdents'>Differential Expression with Idents (Seurat)</h2><span id='topic+DEAnalysisSeuratIdents'></span>

<h3>Description</h3>

<p>This function calculates the differential expression values
along with identifying the Idents (through Seurat). The output is
saved in an RData file for each unique identity (id).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DEAnalysisSeuratIdents(scdata, id, path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DEAnalysisSeuratIdents_+3A_scdata">scdata</code></td>
<td>
<p>The gene expression datafarme</p>
</td></tr>
<tr><td><code id="DEAnalysisSeuratIdents_+3A_id">id</code></td>
<td>
<p>The unique identities within the data</p>
</td></tr>
<tr><td><code id="DEAnalysisSeuratIdents_+3A_path">path</code></td>
<td>
<p>The path for the RData results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An RData file with the differential expression analysis results
for each unique id.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

#Old Method
#load("data/dataSC_3.RData")
#load("data/trueLabels.RData")
#load("data/dataBulk.RData") #read in bulk data for WT1 (control condition #1)
#load("data/labels.RData") #read in single-cell labels from clustering

labels&lt;-trueLabels
#Change to real labels
newcat&lt;-c("NonCycISC","CycISC","TA","Ent","PreEnt","Goblet","Paneth",
"Tuft","EE")
for (i in 1:length(newcat)){
  labels[which(labels==(i-1))]&lt;-newcat[i]
  }
#Run deconvolution
#Seurat_test2 &lt;- DEAnalysisSeuratIdents(dataSC, labels, "results")


</code></pre>

<hr>
<h2 id='findDampeningConstant'>findDampeningConstant</h2><span id='topic+findDampeningConstant'></span>

<h3>Description</h3>

<p>Finds a dampening constant for the weights using cross-validation.
The goldStandard is used to define the weights. Multiple values of the
dampening constant (multiplier) are tried. For each attempt,
the variance of the dampened weighted solution for a subset of genes is
calculated (on a randomly selected half of the genes).
Note that infinite weights are ignored.The dampening constant that
results in least cross-validation variance is chosen. It functions
in a nondeterministic manner. The dampening constant defines the maximum
value that any weight can take on.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findDampeningConstant(S, B, goldStandard)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findDampeningConstant_+3A_s">S</code></td>
<td>
<p>List output from trimData$Sig (S)</p>
</td></tr>
<tr><td><code id="findDampeningConstant_+3A_b">B</code></td>
<td>
<p>List output from trimData$dataBulk (B)</p>
</td></tr>
<tr><td><code id="findDampeningConstant_+3A_goldstandard">goldStandard</code></td>
<td>
<p>Starting point for the weights, determined by solving OLS</p>
</td></tr>
</table>


<h3>Value</h3>

<p>value (dampening constant value)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solution &lt;- solveOLSInternal(S,B)
findDampeningConstant(S, B, solution)


</code></pre>

<hr>
<h2 id='m.auc'>m.auc</h2><span id='topic+m.auc'></span>

<h3>Description</h3>

<p>Calculates the AUC of a dataset. The function mainly serves
to support the DWLS function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>m.auc(dataset, grouping)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="m.auc_+3A_dataset">dataset</code></td>
<td>
<p>Data</p>
</td></tr>
<tr><td><code id="m.auc_+3A_grouping">grouping</code></td>
<td>
<p>Data subdivision</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of standardized output of AUC calculation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

pseudo.count = 0.1
data.used.log2 &lt;- log2(dataSC+pseudo.count)
colnames(data.used.log2)&lt;-make.unique(colnames(data.used.log2))
diff.cutoff=0.5
id = labels
for (i in unique(id)){
  cells.symbol.list2 = colnames(data.used.log2)[which(id==i)]
  cells.coord.list2 = match(cells.symbol.list2, colnames(data.used.log2))
  cells.symbol.list1 = colnames(data.used.log2)[which(id != i)]
  cells.coord.list1= match(cells.symbol.list1, colnames(data.used.log2))
  data.used.log2.ordered = cbind(data.used.log2[,cells.coord.list1],
                                         data.used.log2[,cells.coord.list2])
  group.v &lt;- c(rep(0,length(cells.coord.list1)),
                               rep(1, length(cells.coord.list2)))
  #ouput
  log2.stat.result &lt;- stat.log2(data.used.log2.ordered,
                                    group.v, pseudo.count)
  Auc &lt;- m.auc(data.used.log2.ordered, group.v)}


</code></pre>

<hr>
<h2 id='MASTSignatureMatrixGivenDE'>Build Signature matrix using MAST given a differential expression
matrix</h2><span id='topic+MASTSignatureMatrixGivenDE'></span>

<h3>Description</h3>

<p>This function builds a signature matrix using a pre-created
differential expression matrix. The input matrix must have the same format
as the DEAnalysisMAST() function and must be saved as an RData file
ending with _MIST. The file must be named identity_MIST.RData.
See exampledata_MIST.RData for more information (inst/man).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MASTSignatureMatrixGivenDE(
  scdata,
  id,
  path,
  diff.cutoff = 0.5,
  pval.cutoff = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MASTSignatureMatrixGivenDE_+3A_scdata">scdata</code></td>
<td>
<p>The data</p>
</td></tr>
<tr><td><code id="MASTSignatureMatrixGivenDE_+3A_id">id</code></td>
<td>
<p>The identities of the genes</p>
</td></tr>
<tr><td><code id="MASTSignatureMatrixGivenDE_+3A_path">path</code></td>
<td>
<p>The path to the file results</p>
</td></tr>
<tr><td><code id="MASTSignatureMatrixGivenDE_+3A_diff.cutoff">diff.cutoff</code></td>
<td>
<p>This is automatically set to 0.5</p>
</td></tr>
<tr><td><code id="MASTSignatureMatrixGivenDE_+3A_pval.cutoff">pval.cutoff</code></td>
<td>
<p>This is automatically set to 0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Signature Matrix built using the MAST algorithm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

labels&lt;-trueLabels

#Change to real labels
newcat&lt;-c("NonCycISC","CycISC","TA","Ent","PreEnt","Goblet",
"Paneth","Tuft","EE")
for (i in 1:length(newcat)){
  labels[which(labels==(i-1))]&lt;-newcat[i]
  }
#Results in inst/extdata/results -- run on local
#Signature&lt;-buildSignatureMatrixMAST(dataSC,labels,"results",
# diff.cutoff=0.5,pval.cutoff=0.01)


</code></pre>

<hr>
<h2 id='Mean.in.log2space'>Mean.in.log2space</h2><span id='topic+Mean.in.log2space'></span>

<h3>Description</h3>

<p>Applies the log2 to the mean of ((2^x - pseudo count) + pseudo
count).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mean.in.log2space(x, pseudo.count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Mean.in.log2space_+3A_x">x</code></td>
<td>
<p>Data</p>
</td></tr>
<tr><td><code id="Mean.in.log2space_+3A_pseudo.count">pseudo.count</code></td>
<td>
<p>A pseudocount value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

Mean.in.log2space(dataBulk, 0.1)


</code></pre>

<hr>
<h2 id='solveDampenedWLS'>solveDampenedWLS</h2><span id='topic+solveDampenedWLS'></span>

<h3>Description</h3>

<p>Dampened weighted least squares (DLWS) is an estimation
method for gene expression deconvolution, in which the cell-type
composition of a bulk RNA-seq data set is computationally inferred.
This method corrects common biases towards cell types that are
characterized by highly expressed genes and/or are highly prevalent,
to provide accurate detection across diverse cell types. To begin,
the user must input a bulk RNA-seq data set, along with a labeled
representative single-cell RNA-seq data set that will serve to generate
cell-type-specific gene expression profiles. Ideally, the single-cell
data set will contain cells from all cell types that may be found in the
bulk data. DWLS will return the cell-type composition of the bulk data.
First, solve OLS then use the solution to find a starting point for the
weights. Next, the dampened weighted least squares is performed. The weights
are iterated until convergence then the dampening constant for weights is
found using cross-validation (with decreasing step size for convergence).
</p>
<p>Note: The function uses solveDampenedWLSj() and findDampeningConstant().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solveDampenedWLS(S, B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solveDampenedWLS_+3A_s">S</code></td>
<td>
<p>List output from trimData</p>
</td></tr>
<tr><td><code id="solveDampenedWLS_+3A_b">B</code></td>
<td>
<p>List output from trimData</p>
</td></tr>
</table>


<h3>Value</h3>

<p>value (Dampened weighted least squares estimation values)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solveDampenedWLS(S, B)

</code></pre>

<hr>
<h2 id='solveDampenedWLSj'>solveDampenedWLSj</h2><span id='topic+solveDampenedWLSj'></span>

<h3>Description</h3>

<p>Solve dampened weighted least squares given a dampening constant.
</p>
<p>Note: The function uses solveDampenedWLS() and findDampeningConstant().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solveDampenedWLSj(S, B, goldStandard, j)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solveDampenedWLSj_+3A_s">S</code></td>
<td>
<p>List output from trimData$sig (S)</p>
</td></tr>
<tr><td><code id="solveDampenedWLSj_+3A_b">B</code></td>
<td>
<p>List output from trimData$bulk (B)</p>
</td></tr>
<tr><td><code id="solveDampenedWLSj_+3A_goldstandard">goldStandard</code></td>
<td>
<p>Starting point for the weights, this can be determined
using solveOLSInternal(S,B)</p>
</td></tr>
<tr><td><code id="solveDampenedWLSj_+3A_j">j</code></td>
<td>
<p>The dampening constant, this can be determined using
findDampeningConstant(S,B,goldStandard)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>value (Dampened weighted least squares estimation values)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solution &lt;- solveOLSInternal(S,B)
j &lt;- findDampeningConstant(S,B,solution)
goldStandard &lt;- solveOLSInternal(S,B)
solveDampenedWLSj(S,B,goldStandard,j)

</code></pre>

<hr>
<h2 id='solveOLS'>solveOLS</h2><span id='topic+solveOLS'></span>

<h3>Description</h3>

<p>This function solves or the unknown parameters
using ordinary least squares (OLS). It is constrained such
that cell type numbers are greater than 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solveOLS(S, B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solveOLS_+3A_s">S</code></td>
<td>
<p>List output from trimData$sig (S)</p>
</td></tr>
<tr><td><code id="solveOLS_+3A_b">B</code></td>
<td>
<p>List output from trimData$bulk (B)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell-type proportion
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solveOLS(S, B)

</code></pre>

<hr>
<h2 id='solveOLSInternal'>solveOLSInternal</h2><span id='topic+solveOLSInternal'></span>

<h3>Description</h3>

<p>This function solves or the unknown parameters
using ordinary least squares (OLS) without printing the output.
It returns the cell numbers, not the proportions (see solveOLS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solveOLSInternal(S, B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solveOLSInternal_+3A_s">S</code></td>
<td>
<p>List output from trimData$sig (S)</p>
</td></tr>
<tr><td><code id="solveOLSInternal_+3A_b">B</code></td>
<td>
<p>List output from trimData$bulk (B)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell numbers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solveOLSInternal(S, B)

</code></pre>

<hr>
<h2 id='solveSVR'>solveSVR</h2><span id='topic+solveSVR'></span>

<h3>Description</h3>

<p>Performs a support vector regression (SVR). First, the data is
scaled then it solves for the SVR. An svm model is used with the following
specifications nu=0.5,scale = TRUE, type = &quot;nu-regression&quot;,
kernel =&quot;linear&quot;,cost = 1.
</p>
<p>Nu-support vector regression was performed using the svm function in the
e1071 package in R. Parameters were set to nu = 0.5, type = “nu-regression”,
kernel = “linear”, cost = 1, and all others to the default values.
Bulk data and signature matrices were scaled to -1, 1. These parameter
and scaling choices match those specified in Schelker et al. in their
MATLAB code, accessed through https://figshare.com/s/865e694ad06d5857db4b.
As in Newman et al., model coefficients are extracted from the svm model
using t(model$coefs) model$SV, and any negative coefficients are set
to zero. The coefficients are then scaled by the sum of the coefficients,
such that the scaled coefficients will sum to one.
</p>
<p>Citations:
Newman, A. M. et al. Robust enumeration of cell subsets from tissue
expression profiles. Nat. Methods 12, 453–457 (2015).
</p>
<p>Schelker, M. et al. Estimation of immune cell content in tumor
tissue using single-cell RNA-seq data. Nat. Commun. 8, 2032 (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>solveSVR(S, B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="solveSVR_+3A_s">S</code></td>
<td>
<p>List output from trimData$sig (S)</p>
</td></tr>
<tr><td><code id="solveSVR_+3A_b">B</code></td>
<td>
<p>List output from trimData$bulk (B)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value (SVR)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimmed &lt;- trimData(Sig, dataBulk)
S &lt;- trimmed$sig
B &lt;- trimmed$bulk
solveSVR(S, B)

</code></pre>

<hr>
<h2 id='stat.log2'>stat.log2</h2><span id='topic+stat.log2'></span>

<h3>Description</h3>

<p>One of the functions required for the  differential expression
analysis using MAST (DEAnalysisMast()) function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat.log2(data.m, group.v, pseudo.count)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stat.log2_+3A_data.m">data.m</code></td>
<td>
<p>Data</p>
</td></tr>
<tr><td><code id="stat.log2_+3A_group.v">group.v</code></td>
<td>
<p>Groupings</p>
</td></tr>
<tr><td><code id="stat.log2_+3A_pseudo.count">pseudo.count</code></td>
<td>
<p>A pseudocount value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of the log2 applied results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

pseudo.count = 0.1
data.used.log2 &lt;- log2(dataSC+pseudo.count)
colnames(data.used.log2)&lt;-make.unique(colnames(data.used.log2))
diff.cutoff=0.5
id = labels
for (i in unique(id)){
  cells.symbol.list2 = colnames(data.used.log2)[which(id==i)]
  cells.coord.list2 = match(cells.symbol.list2, colnames(data.used.log2))
  cells.symbol.list1 = colnames(data.used.log2)[which(id != i)]
  cells.coord.list1 = match(cells.symbol.list1, colnames(data.used.log2))
  data.used.log2.ordered = cbind(data.used.log2[,cells.coord.list1],
                                        data.used.log2[,cells.coord.list2])}
group.v &lt;- c(rep(0,length(cells.coord.list1)),
                                         rep(1, length(cells.coord.list2)))

</code></pre>

<hr>
<h2 id='trimData'>trimData</h2><span id='topic+trimData'></span>

<h3>Description</h3>

<p>This function trims bulk and single-cell data to
contain the same genes. The result is a list of the intersecting genes
within the two datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trimData(Signature_Matrix, bulkdata)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trimData_+3A_signature_matrix">Signature_Matrix</code></td>
<td>
<p>A single-cell signature matrix</p>
</td></tr>
<tr><td><code id="trimData_+3A_bulkdata">bulkdata</code></td>
<td>
<p>A bulk dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of trimmed bulk and single-cell data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
#Sig
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/Sig.RData"
#dest &lt;- "data/Sig.RData"
#download.file(url, dest)
#load("data/Sig.RData")
load(system.file("extdata", "Sig.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#download.file(url, dest)
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

trimData(Signature_Matrix = Sig, bulkdata = dataBulk)

</code></pre>

<hr>
<h2 id='v.auc'>v.auc</h2><span id='topic+v.auc'></span>

<h3>Description</h3>

<p>Uses the prediction() function in order to create standardized
output from the data in order to perform an AUC calculation. The calculation
results are rounded to the third decimal place. This function serves mainly
to support the DWLS function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>v.auc(data.v, group.v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="v.auc_+3A_data.v">data.v</code></td>
<td>
<p>Data</p>
</td></tr>
<tr><td><code id="v.auc_+3A_group.v">group.v</code></td>
<td>
<p>Data subdivision</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of standardized output of AUC calculation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

#dataSC
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataSC.RData"
#dest &lt;- "data/dataSC.RData"
#load(download.file(url, tempfile(data/dataSC.RData))
#load("dataSC.RData")
#SOLUTION
load(system.file("extdata", "dataSC.RData", package = "DWLS"))

#dataBulk
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/dataBulk.RData"
#dest &lt;- "data/dataBulk.RData"
#load(download.file(url, tempfile(dest)))
#load("data/dataBulk.RData")
load(system.file("extdata", "dataBulk.RData", package = "DWLS"))

#labels
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/labels.RData"
#dest &lt;- "data/labels.RData"
#download.file(url, dest)
#load("data/labels.RData")
load(system.file("extdata", "labels.RData", package = "DWLS"))

#data('trueLabels', package = "DWLS")
#url &lt;- "https://github.com/sistia01/DWLS/raw/main/inst/extdata/trueLabels.RData"
#dest &lt;- "data/trueLabels.RData"
#download.file(url, dest)
#load("data/trueLabels.RData")
load(system.file("extdata", "trueLabels.RData", package = "DWLS"))

pseudo.count = 0.1
data.used.log2   &lt;- log2(dataSC+pseudo.count)
colnames(data.used.log2)&lt;-make.unique(colnames(data.used.log2))
diff.cutoff=0.5
id = labels
for (i in unique(id)){
  cells.symbol.list2 = colnames(data.used.log2)[which(id==i)]
  cells.coord.list2 = match(cells.symbol.list2, colnames(data.used.log2))
  cells.symbol.list1 = colnames(data.used.log2)[which(id != i)]
  cells.coord.list1= match(cells.symbol.list1, colnames(data.used.log2))
  data.used.log2.ordered = cbind(data.used.log2[,cells.coord.list1],
                                         data.used.log2[,cells.coord.list2])
  group.v &lt;- c(rep(0,length(cells.coord.list1)),
                               rep(1, length(cells.coord.list2)))
  #ouput
  log2.stat.result &lt;- stat.log2(data.used.log2.ordered,
                                    group.v, pseudo.count)
m.auc=function(data.used.log2.ordered,group.v)
{AUC=apply(data.used.log2.ordered, 1, function(x) v.auc(x,group.v))
AUC[is.na(AUC)]=0.5
return(AUC)}

}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
