<!DOCTYPE html><html lang="en"><head><title>Help for package DYNATE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {DYNATE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DYNATE'><p>DYNATE</p>
Function to conduct hierarchical mutiple testing based on the leaf p-values</a></li>
<li><a href='#p_leaf'><p>p_leaf</p></a></li>
<li><a href='#snp_dat'><p>snp_dat</p></a></li>
<li><a href='#Test_Leaf'><p>Test_Leaf</p>
The function is used to generate Leaf P-values for case-control study.
Users can input the leaf information through argument struct_map.
If there is not leaf information e.g. struct_map=NULL, Test_Leaf will automatically construct leaf.
Argument thresh_val specifies the leaf size constructed from the function.
When the argument Gmat_case and Gmat_ctrl is null, Test_Leaf will automatically generate those matrices.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Dynamic Aggregation Testing</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A multiple testing procedure aims to find the rare-variant association regions. When variants are rare, the single variant association test approach suffers from low power. To improve testing power, the procedure dynamically and hierarchically aggregates smaller genome regions to larger ones and performs multiple testing for disease associations with a controlled node-level false discovery rate. This method are members of the family of ancillary information assisted recursive testing introduced in Pura, Li, Chan and Xie (2021) &lt;<a href="https://doi.org/10.48550/arXiv.1906.07757">doi:10.48550/arXiv.1906.07757</a>&gt; and Li, Sung and Xie (2021) &lt;<a href="https://doi.org/10.48550/arXiv.2103.11085">doi:10.48550/arXiv.2103.11085</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2.9000</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, tidyverse, Matrix, reshape2, stats, methods,
tibble, dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-14 01:08:56 UTC; xuechanli</td>
</tr>
<tr>
<td>Author:</td>
<td>Xuechan Li [aut, cre],
  Jichun Xie [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuechan Li &lt;xuechanli0530@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-14 10:05:21 UTC</td>
</tr>
</table>
<hr>
<h2 id='DYNATE'>DYNATE
Function to conduct hierarchical mutiple testing based on the leaf p-values</h2><span id='topic+DYNATE'></span>

<h3>Description</h3>

<p>DYNATE
Function to conduct hierarchical mutiple testing based on the leaf p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DYNATE(struct_map, L = 5, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DYNATE_+3A_struct_map">struct_map</code></td>
<td>
<p>a data frame with both leaf information and P-value information.</p>
</td></tr>
<tr><td><code id="DYNATE_+3A_l">L</code></td>
<td>
<p>maximum number of layers</p>
</td></tr>
<tr><td><code id="DYNATE_+3A_alpha">alpha</code></td>
<td>
<p>desired FDR</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with testing results.
</p>


<h3>References</h3>

<p>Li, Xuechan, Anthony Sung, and Jichun Xie. &quot;Distance Assisted Recursive Testing.&quot; arXiv preprint arXiv:2103.11085 (2021).
Pura, John, et al. &quot;TEAM: A Multiple Testing Algorithm on the Aggregation Tree for Flow Cytometry Analysis.&quot; arXiv preprint arXiv:1906.07757 (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("p_leaf")

# Set tuning parameters
L &lt;- 3 # layer number
alpha &lt;- 0.05 # desired FDR

# conduct dynamic and hierarchical testing based on the leaf level p values.
out &lt;- DYNATE(struct_map=p_leaf,L=L,alpha=alpha)
summary(out)
</code></pre>

<hr>
<h2 id='p_leaf'>p_leaf</h2><span id='topic+p_leaf'></span>

<h3>Description</h3>

<p>p_leaf
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_leaf
</code></pre>


<h3>Format</h3>

<p>A data frame with 16281 rows and 5 variables. Each row links to a SNP that belongs to a leaf with testing p-value&lt;1.
</p>

<hr>
<h2 id='snp_dat'>snp_dat</h2><span id='topic+snp_dat'></span>

<h3>Description</h3>

<p>snp_dat
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp_dat
</code></pre>


<h3>Format</h3>

<p>A data frame with 210454 rows and 6 variables.
</p>

<hr>
<h2 id='Test_Leaf'>Test_Leaf
The function is used to generate Leaf P-values for case-control study.
Users can input the leaf information through argument struct_map.
If there is not leaf information e.g. struct_map=NULL, Test_Leaf will automatically construct leaf.
Argument thresh_val specifies the leaf size constructed from the function.
When the argument Gmat_case and Gmat_ctrl is null, Test_Leaf will automatically generate those matrices.</h2><span id='topic+Test_Leaf'></span>

<h3>Description</h3>

<p>Test_Leaf
The function is used to generate Leaf P-values for case-control study.
Users can input the leaf information through argument struct_map.
If there is not leaf information e.g. struct_map=NULL, Test_Leaf will automatically construct leaf.
Argument thresh_val specifies the leaf size constructed from the function.
When the argument Gmat_case and Gmat_ctrl is null, Test_Leaf will automatically generate those matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Test_Leaf(snp_dat = NULL, thresh_val = 10, covars = NULL, teststat = "FET")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Test_Leaf_+3A_snp_dat">snp_dat</code></td>
<td>
<p>an optional data frame containing patients mutation information. If <code>snp_dat=NULL</code>, the mutation information should be taken from <code>Gmat_case</code>, <code>Gmat_ctrl</code> and <code>glm_input</code>. See vignettes for detail.</p>
</td></tr>
<tr><td><code id="Test_Leaf_+3A_thresh_val">thresh_val</code></td>
<td>
<p>a positive integer for leaf size.</p>
</td></tr>
<tr><td><code id="Test_Leaf_+3A_covars">covars</code></td>
<td>
<p>an optional vector about the name of covariates to be considered in the fitting process. Should be NULL (default) or a character vector.</p>
</td></tr>
<tr><td><code id="Test_Leaf_+3A_teststat">teststat</code></td>
<td>
<p>the statistic used to derive p-value. Must be one of &quot;FET&quot; (default) or &quot;score&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of rejected leafs with snp information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("snp_dat")

# Set leaf size M
M &lt;- 5

#Construct leaves and generate leaf p-value.
p.leaf &lt;- Test_Leaf(snp_dat=snp_dat,thresh_val=M)
summary(p.leaf)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
