<!DOCTYPE html><html><head><title>Help for package dynwrap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {dynwrap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.method_process_definition'><p>Method process definition</p></a></li>
<li><a href='#add_attraction'><p>Add attraction of cells using RNA velocity</p></a></li>
<li><a href='#add_branch_trajectory'><p>Construct a trajectory given its branch network and the pseudotime of the cells on one of the branches.</p></a></li>
<li><a href='#add_cell_graph'><p>Constructs a trajectory using a graph between cells, by mapping cells onto a set of backbone cells.</p></a></li>
<li><a href='#add_cell_waypoints'><p>Add or select waypoint cells of a trajectory</p></a></li>
<li><a href='#add_cluster_graph'><p>Constructs a trajectory using a cell grouping and a network between groups. Will use an existing grouping if it is present in the dataset.</p></a></li>
<li><a href='#add_cyclic_trajectory'><p>Constructs a circular trajectory using the pseudotime values of each cell.</p></a></li>
<li><a href='#add_dimred'><p>Add or create a dimensionality reduction</p></a></li>
<li><a href='#add_dimred_projection'><p>Constructs a trajectory by projecting cells within a dimensionality reduction</p></a></li>
<li><a href='#add_end_state_probabilities'><p>Constructs a multifurcating trajectory using end state probabilities</p></a></li>
<li><a href='#add_expression'><p>Add count and normalised expression values to a dataset</p></a></li>
<li><a href='#add_feature_importance'><p>Add a feature importance to a dataset</p></a></li>
<li><a href='#add_grouping'><p>Add a cell grouping to a dataset</p></a></li>
<li><a href='#add_linear_trajectory'><p>Constructs a linear trajectory using pseudotime values</p></a></li>
<li><a href='#add_prior_information'><p>Add or compute prior information for a trajectory</p></a></li>
<li><a href='#add_regulatory_network'><p>Add a GRN to a dynwrap object</p></a></li>
<li><a href='#add_root'><p>Root the trajectory</p></a></li>
<li><a href='#add_tde_overall'><p>Add information on overall differentially expressed features</p></a></li>
<li><a href='#add_timings'><p>Add timings to a trajectory</p></a></li>
<li><a href='#add_trajectory'><p>Construct a trajectory given its milestone network and milestone percentages or progressions</p></a></li>
<li><a href='#allowed_inputs'><p>All allowed inputs for a TI method</p></a></li>
<li><a href='#allowed_outputs'><p>All allowed outputs for a TI method</p></a></li>
<li><a href='#calculate_attraction'><p>Calculate the attraction of cells to other cells using velocity</p></a></li>
<li><a href='#calculate_average_by_group'><p>Calculate average values of a matrix</p></a></li>
<li><a href='#calculate_geodesic_distances'><p>Calculate geodesic distances between cells in a trajectory</p></a></li>
<li><a href='#calculate_pseudotime'><p>Add or calculate pseudotime as distance from the root</p></a></li>
<li><a href='#calculate_trajectory_dimred'><p>Layout the trajectory and its cells in 2 dimensions using a graph layout</p></a></li>
<li><a href='#classify_milestone_network'><p>Classify a milestone network</p></a></li>
<li><a href='#common_param'><p>Common param</p></a></li>
<li><a href='#convert_definition'><p>Convert a definition loaded in from a yaml</p></a></li>
<li><a href='#convert_milestone_percentages_to_progressions'><p>Conversion between milestone percentages and progressions</p></a></li>
<li><a href='#convert_progressions_to_milestone_percentages'><p>Conversion between milestone percentages and progressions</p></a></li>
<li><a href='#create_ti_method_container'><p>Create a TI method from a docker / singularity container</p></a></li>
<li><a href='#create_ti_method_definition'><p>Create a TI method from a local method definition file</p></a></li>
<li><a href='#create_ti_method_r'><p>Create a TI method from an R function wrapper</p></a></li>
<li><a href='#def_author'><p>Meta information on an author</p></a></li>
<li><a href='#def_container'><p>Meta information on the container in which the wrapper resides</p></a></li>
<li><a href='#def_manuscript'><p>Meta information on the manuscript</p></a></li>
<li><a href='#def_method'><p>Define meta information on the TI method.</p></a></li>
<li><a href='#def_package'><p>Meta information on the package in which the TI function resides</p></a></li>
<li><a href='#def_parameters'><p>Meta information on the parameters of the TI method</p></a></li>
<li><a href='#def_wrapper'><p>Meta information on the wrapper</p></a></li>
<li><a href='#definition'><p>Create a definition</p></a></li>
<li><a href='#dynwrap'><p>Inferring and adapting single-cell trajectories</p></a></li>
<li><a href='#example_dataset'><p>Example dataset</p></a></li>
<li><a href='#example_trajectory'><p>Example trajectory</p></a></li>
<li><a href='#flip_edges'><p>Flip a set of edges of the milestone network</p></a></li>
<li><a href='#gather_cells_at_milestones'><p>Gather cells to their closest milestones</p></a></li>
<li><a href='#generate_parameter_documentation'><p>Generate the parameter documentation of a method, use with <code style="white-space: pre;">&#8288;@eval&#8288;</code></p></a></li>
<li><a href='#get_default_parameters'><p>Get the default parameters of a method</p></a></li>
<li><a href='#get_ti_methods'><p>Return all TI that are installed in one or more packages</p></a></li>
<li><a href='#group_from_trajectory'><p>Create a grouping from a trajectory</p></a></li>
<li><a href='#infer_trajectories'><p>Infer one or more trajectories from a single-cell dataset</p></a></li>
<li><a href='#label_milestones'><p>Label milestones either manually (<code>label_milestones</code>) or using marker genes (<code>label_milestones_markers</code>)</p></a></li>
<li><a href='#prior_usages'><p>Metadata on prior usages</p></a></li>
<li><a href='#priors'><p>Metadata on priors</p></a></li>
<li><a href='#project_trajectory'><p>Project a trajectory onto a dimensionality reduction</p></a></li>
<li><a href='#project_waypoints'><p>Project waypoints of a trajectory (e.g. milestones) into a space defined by cells (e.g. expression or a dimensionality reduction)</p></a></li>
<li><a href='#random_seed'><p>Generate a random seed</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#select_waypoints'><p>Add or create waypoints to a trajectory</p></a></li>
<li><a href='#simplify_igraph_network'><p>Simplify an igraph network such that consecutive linear edges are removed</p></a></li>
<li><a href='#simplify_trajectory'><p>Simplify a trajectory by removing transient milestones</p></a></li>
<li><a href='#trajectory_type_dag'><p>A DAG connecting different trajectory types</p></a></li>
<li><a href='#trajectory_types'><p>Metadata on the trajectory types</p></a></li>
<li><a href='#wrap_data'><p>A data wrapper for datasets and trajectories</p></a></li>
<li><a href='#wrap_expression'><p>Create a wrapper object with expression and counts</p></a></li>
<li><a href='#wrapper_types'><p>Metadata on wrapper types</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Representing and Inferring Single-Cell Trajectories</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functionality to infer trajectories from single-cell data,
  represent them into a common format, and adapt them. Other biological information
  can also be added, such as cellular grouping, RNA velocity and annotation.
  Saelens et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41587-019-0071-9">doi:10.1038/s41587-019-0071-9</a>&gt;.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dynverse/dynwrap">https://github.com/dynverse/dynwrap</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dynverse/dynwrap/issues">https://github.com/dynverse/dynwrap/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, babelwhale, crayon, dplyr, dynutils (&ge; 1.0.3),
dynparam, igraph, glue, magrittr, Matrix, methods, purrr,
processx, readr, stringr, reshape2, tibble, tidyr, yaml</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, dyndimred, hdf5r, knitr, pkgload, ranger, rmarkdown,
testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-18 10:01:02 UTC; rcannood</td>
</tr>
<tr>
<td>Author:</td>
<td>Robrecht Cannoodt <a href="https://orcid.org/0000-0003-3641-729X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre] (rcannood),
  Wouter Saelens <a href="https://orcid.org/0000-0002-7114-6248"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut] (zouter)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Robrecht Cannoodt &lt;rcannood@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-19 09:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.method_process_definition'>Method process definition</h2><span id='topic+.method_process_definition'></span>

<h3>Description</h3>

<p>Method process definition
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.method_process_definition(definition, return_function)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".method_process_definition_+3A_definition">definition</code></td>
<td>
<p>A definition, see <code><a href="#topic+definition">definition()</a></code></p>
</td></tr>
<tr><td><code id=".method_process_definition_+3A_return_function">return_function</code></td>
<td>
<p>Whether to return a function that allows you to override the default parameters, or just return the method meta data as is.</p>
</td></tr>
</table>

<hr>
<h2 id='add_attraction'>Add attraction of cells using RNA velocity</h2><span id='topic+add_attraction'></span>

<h3>Description</h3>

<p>Add attraction of cells using RNA velocity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_attraction(dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_attraction_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the attraction added.
</p>

<hr>
<h2 id='add_branch_trajectory'>Construct a trajectory given its branch network and the pseudotime of the cells on one of the branches.</h2><span id='topic+add_branch_trajectory'></span>

<h3>Description</h3>

<p>The branch network is converted to a milestone network by giving each branch a start and end milestone. If two branches are connected in the branch network, the end milestone of branch 1 and start milestone of branch 2 will be merged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_branch_trajectory(
  dataset,
  branch_network,
  branches,
  branch_progressions,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_branch_trajectory_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_branch_trajectory_+3A_branch_network">branch_network</code></td>
<td>
<p>The network between branches, a dataframe with a <em>from</em> and <em>to</em> branch identifier</p>
</td></tr>
<tr><td><code id="add_branch_trajectory_+3A_branches">branches</code></td>
<td>
<p>The length and directedness of the branches, a dataframe with the branch identifier (<em>branch_id</em>), the length of the branch (<em>length</em>) and whether it is <em>directed</em></p>
</td></tr>
<tr><td><code id="add_branch_trajectory_+3A_branch_progressions">branch_progressions</code></td>
<td>
<p>Specifies the progression of a cell along a transition in the branch network. A dataframe containing the <em>cell_id</em>, <em>branch_id</em> and its progression along the edge (<em>percentage</em>, between 0 and 1)</p>
</td></tr>
<tr><td><code id="add_branch_trajectory_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the trajectory</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The resulting trajectory will always be directed.
</p>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- wrap_data(cell_ids = letters)

branch_network &lt;- tibble::tibble(from = c("A", "A"), to = c("B", "C"))
branch_network
branches &lt;- tibble::tibble(branch_id = c("A", "B", "C"), length = 1, directed = TRUE)
branches
branch_progressions &lt;- tibble::tibble(
  cell_id = dataset$cell_ids,
  branch_id = sample(branches$branch_id, length(dataset$cell_ids), replace = TRUE),
  percentage = runif(length(dataset$cell_ids))
)
branch_progressions

trajectory &lt;- add_branch_trajectory(
  dataset,
  branch_network,
  branches,
  branch_progressions
)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_cell_graph'>Constructs a trajectory using a graph between cells, by mapping cells onto a set of backbone cells.</h2><span id='topic+add_cell_graph'></span>

<h3>Description</h3>

<p>The cells that are part of the backbone will form the trajectory. All other cells are moved towards the nearest cell that is part of the backbone.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_cell_graph(
  dataset,
  cell_graph,
  to_keep,
  milestone_prefix = "milestone_",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_cell_graph_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_cell_graph_+3A_cell_graph">cell_graph</code></td>
<td>
<p>The edges between cells, a dataframe containing the <em>from</em> and <em>to</em> cells, the *length, and whether this edge is <em>directed</em></p>
</td></tr>
<tr><td><code id="add_cell_graph_+3A_to_keep">to_keep</code></td>
<td>
<p>Whether a cells is part of the backbone. May be a character vector with the identifiers of the backbone cells, or a named boolean vector whether a cell is from the backbone</p>
</td></tr>
<tr><td><code id="add_cell_graph_+3A_milestone_prefix">milestone_prefix</code></td>
<td>
<p>A prefix to add to the id of the cell ids when they are used as milestones, in order to avoid any naming conflicts,</p>
</td></tr>
<tr><td><code id="add_cell_graph_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the wrapper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)
dataset &lt;- wrap_data(cell_ids = letters)

backbone_cell_graph &lt;- tibble::tibble(
  from = letters[1:10],
  to = letters[2:11],
  length = 1,
  directed = TRUE
)
leaves_cell_graph &lt;- tibble::tibble(
  from = letters[12:26],
  to = sample(letters[1:11], 15, replace = TRUE),
  length = 1,
  directed = TRUE
)
cell_graph &lt;- bind_rows(backbone_cell_graph, leaves_cell_graph)
cell_graph
to_keep &lt;- letters[1:11]
to_keep

trajectory &lt;- add_cell_graph(dataset, cell_graph, to_keep)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_cell_waypoints'>Add or select waypoint cells of a trajectory</h2><span id='topic+add_cell_waypoints'></span><span id='topic+is_wrapper_with_waypoint_cells'></span><span id='topic+determine_cell_trajectory_positions'></span><span id='topic+select_waypoint_cells'></span>

<h3>Description</h3>

<p>Waypoint cells are cells spread across all of the trajectory such that there is no other cell
that has a large geodesic distance to any of the waypoint cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_cell_waypoints(trajectory, num_cells_selected = 100)

is_wrapper_with_waypoint_cells(trajectory)

determine_cell_trajectory_positions(
  milestone_ids,
  milestone_network,
  milestone_percentages,
  progressions,
  divergence_regions
)

select_waypoint_cells(
  milestone_ids,
  milestone_network,
  milestone_percentages,
  progressions,
  divergence_regions,
  num_cells_selected = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_cell_waypoints_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_num_cells_selected">num_cells_selected</code></td>
<td>
<p>About the number of cells selected as waypoints</p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_milestone_percentages">milestone_percentages</code></td>
<td>
<p>A data frame specifying what percentage milestone each cell
consists of.
Type: Data frame(cell_id = character, milestone_id = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_progressions">progressions</code></td>
<td>
<p>Specifies the progression of a cell along a transition in the milestone_network.
Type: Data frame(cell_id = character, from = character, to = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_cell_waypoints_+3A_divergence_regions">divergence_regions</code></td>
<td>
<p>A data frame specifying the divergence
regions between milestones (e.g. a bifurcation).
Type: Data frame(divergence_id = character, milestone_id = character, is_start = logical).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>add_cell_waypoints</code></strong> returns a trajectory with <em>waypoint_cells</em>, a character vector containing the cell ids of the waypoint cells
</p>
<p><strong><code>select_waypoint_cells</code></strong> returns a character vector containing the cell ids of the waypoint cells
</p>
<p>A dynwrap object with the waypoint cells added.
</p>

<hr>
<h2 id='add_cluster_graph'>Constructs a trajectory using a cell grouping and a network between groups. Will use an existing grouping if it is present in the dataset.</h2><span id='topic+add_cluster_graph'></span>

<h3>Description</h3>

<p>A trajectory in this form will rarely be useful, given that cells are only placed at the milestones themselves, but not on the edges between milestones. A better alternative might be to project the cells using a dimensionality reduction, see <code><a href="#topic+add_dimred_projection">add_dimred_projection()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_cluster_graph(
  dataset,
  milestone_network,
  grouping = NULL,
  explicit_splits = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_cluster_graph_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_cluster_graph_+3A_milestone_network">milestone_network</code></td>
<td>
<p>A network of milestones.</p>
</td></tr>
<tr><td><code id="add_cluster_graph_+3A_grouping">grouping</code></td>
<td>
<p>A grouping of the cells, can be a named vector or a dataframe with <em>group_id</em> and <em>cell_id</em></p>
</td></tr>
<tr><td><code id="add_cluster_graph_+3A_explicit_splits">explicit_splits</code></td>
<td>
<p>Whether to make splits specific by adding a starting node. For example: A-&gt;B, A-&gt;C becomes A-&gt;X, X-&gt;B, X-&gt;C</p>
</td></tr>
<tr><td><code id="add_cluster_graph_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the wrapper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(tibble)
dataset &lt;- wrap_data(cell_ids = letters)

milestone_network &lt;- tibble::tibble(
  from = c("A", "B", "B"),
  to = c("B", "C", "D"),
  directed = TRUE,
  length = 1
)
milestone_network
grouping &lt;- sample(c("A", "B", "C", "D"), length(dataset$cell_ids), replace = TRUE)
grouping
trajectory &lt;- add_cluster_graph(dataset, milestone_network, grouping)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_cyclic_trajectory'>Constructs a circular trajectory using the pseudotime values of each cell.</h2><span id='topic+add_cyclic_trajectory'></span>

<h3>Description</h3>

<p>The pseudotime is divided into three equally sized segments, and are placed within a trajectory in the form A -&gt; B -&gt; C -&gt; A
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_cyclic_trajectory(
  dataset,
  pseudotime,
  directed = FALSE,
  do_scale_minmax = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_cyclic_trajectory_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_cyclic_trajectory_+3A_pseudotime">pseudotime</code></td>
<td>
<p>A named vector of pseudo times.</p>
</td></tr>
<tr><td><code id="add_cyclic_trajectory_+3A_directed">directed</code></td>
<td>
<p>Whether or not the directionality of the pseudotime is predicted.</p>
</td></tr>
<tr><td><code id="add_cyclic_trajectory_+3A_do_scale_minmax">do_scale_minmax</code></td>
<td>
<p>Whether or not to scale the pseudotime between 0 and 1.
Otherwise, will assume the values are already within that range.</p>
</td></tr>
<tr><td><code id="add_cyclic_trajectory_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the wrapper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(tibble)
dataset &lt;- wrap_data(cell_ids = letters)

pseudotime &lt;- tibble(cell_id = dataset$cell_ids, pseudotime = runif(length(dataset$cell_ids)))
pseudotime
trajectory &lt;- add_cyclic_trajectory(dataset, pseudotime)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_dimred'>Add or create a dimensionality reduction</h2><span id='topic+add_dimred'></span><span id='topic+is_wrapper_with_dimred'></span><span id='topic+get_dimred'></span>

<h3>Description</h3>

<p>This can also perform dimensionality reduction of
</p>

<ul>
<li><p> The projected expression state with RNA velocity, only if <code>dimred</code> is a function and  <code>pair_with_velocity=TRUE</code>
</p>
</li>
<li><p> The trajectory, by projecting the milestones and some &quot;waypoints&quot; to the reduced space, only if <code>dataset</code> contains a trajectory
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>add_dimred(
  dataset,
  dimred,
  dimred_milestones = NULL,
  dimred_segment_progressions = NULL,
  dimred_segment_points = NULL,
  project_trajectory = TRUE,
  connect_segments = FALSE,
  pair_with_velocity = !is.null(dataset$expression_future),
  expression_source = "expression",
  ...
)

is_wrapper_with_dimred(dataset)

get_dimred(
  dataset,
  dimred = NULL,
  expression_source = "expression",
  return_other_dimreds = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_dimred_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_dimred_+3A_dimred">dimred</code></td>
<td>
<p>Can be
</p>

<ul>
<li><p> A function which will perform the dimensionality reduction, see <code><a href="dyndimred.html#topic+dimred">dyndimred::list_dimred_methods()</a></code>
</p>
</li>
<li><p> A matrix with the dimensionality reduction, with cells in rows and dimensions (<em>comp_1</em>, <em>comp_2</em>, ...) in columns
</p>
</li></ul>
</td></tr>
<tr><td><code id="add_dimred_+3A_dimred_milestones">dimred_milestones</code></td>
<td>
<p>An optional dimensionality reduction of the milestones. A matrix with milestones in rows and components (<em>comp_1</em>, <em>comp_2</em>, ...) in columns
</p>
<p>This will be automatically calculated if <code>project_trajectory = TRUE</code></p>
</td></tr>
<tr><td><code id="add_dimred_+3A_dimred_segment_progressions">dimred_segment_progressions</code>, <code id="add_dimred_+3A_dimred_segment_points">dimred_segment_points</code></td>
<td>
<p>An optional set of points along the trajectory with their dimensionality reduction. <code>dimred_segment_progressions</code> is a dataframe containing the <em>from</em> and <em>to</em> milestones, and their <em>progression</em>. <code>dimred_segment_points</code> is a matrix with points (the same number as in <code>dimred_segment_progressions</code>) in rows and components (<em>comp_1</em>, <em>comp_2</em>, ...) in columns. Both objects have the same number of rows.
</p>
<p>These will be automatically calculated if <code>project_trajectory = TRUE</code></p>
</td></tr>
<tr><td><code id="add_dimred_+3A_project_trajectory">project_trajectory</code></td>
<td>
<p>Whether to also project the trajectory. Only relevant if dataset contains a trajectory, and dimred_segment_progressions and dimred_segment_points are not provided</p>
</td></tr>
<tr><td><code id="add_dimred_+3A_connect_segments">connect_segments</code></td>
<td>
<p>Whether to connect segments between edges</p>
</td></tr>
<tr><td><code id="add_dimred_+3A_pair_with_velocity">pair_with_velocity</code></td>
<td>
<p>Can perform dimensionality reduction if <code>dimred</code> is a function.</p>
</td></tr>
<tr><td><code id="add_dimred_+3A_expression_source">expression_source</code></td>
<td>
<p>The source of expression, can be &quot;counts&quot;, &quot;expression&quot;, an expression matrix, or another dataset which contains expression</p>
</td></tr>
<tr><td><code id="add_dimred_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the wrapper</p>
</td></tr>
<tr><td><code id="add_dimred_+3A_return_other_dimreds">return_other_dimreds</code></td>
<td>
<p>Whether or not to return also the milestone dimreds and the segment dimreds, if available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset object with <em>dimred</em>, which is a numeric matrix with cells in rows and the different components in columns.
</p>

<ul>
<li><p> If the dataset contained a trajectory, and <code>project_trajectory=TRUE</code> (default), <em>dimred_milestones</em>, <em>dimred_segment_progressions</em> and <em>dimred_segment_points</em> will also be present. These are described in <code><a href="#topic+project_trajectory">project_trajectory()</a></code>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code>dyndimred::list_dimred_methods()</code>, <code><a href="#topic+project_trajectory">project_trajectory()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("dyndimred", quietly = TRUE)) {
  dataset &lt;- example_dataset
  dataset &lt;- add_dimred(
    dataset,
    dyndimred::dimred_landmark_mds
  )
  head(dataset$dimred)
}

</code></pre>

<hr>
<h2 id='add_dimred_projection'>Constructs a trajectory by projecting cells within a dimensionality reduction</h2><span id='topic+add_dimred_projection'></span>

<h3>Description</h3>

<p>A dimensionality reduction of cells and milestones is used, along with the milestone network, to project cells onto the nearest edge. Optionally, a cell grouping can be given which will restrict the edges on which a cell can be projected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_dimred_projection(
  dataset,
  milestone_ids = NULL,
  milestone_network,
  dimred,
  dimred_milestones,
  grouping = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_dimred_projection_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_dimred">dimred</code></td>
<td>
<p>Can be
</p>

<ul>
<li><p> A function which will perform the dimensionality reduction, see <code><a href="dyndimred.html#topic+dimred">dyndimred::list_dimred_methods()</a></code>
</p>
</li>
<li><p> A matrix with the dimensionality reduction, with cells in rows and dimensions (<em>comp_1</em>, <em>comp_2</em>, ...) in columns
</p>
</li></ul>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_dimred_milestones">dimred_milestones</code></td>
<td>
<p>An optional dimensionality reduction of the milestones. A matrix with milestones in rows and components (<em>comp_1</em>, <em>comp_2</em>, ...) in columns
</p>
<p>This will be automatically calculated if <code>project_trajectory = TRUE</code></p>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_grouping">grouping</code></td>
<td>
<p>A grouping of the cells, can be a named vector or a dataframe with <em>group_id</em> and <em>cell_id</em></p>
</td></tr>
<tr><td><code id="add_dimred_projection_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the wrapper.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(tibble)
dataset &lt;- wrap_data(cell_ids = letters)

milestone_network &lt;- tibble::tibble(
  from = c("A", "B", "B"),
  to = c("B", "C", "D"),
  directed = TRUE,
  length = 1
)
milestone_network
dimred &lt;- matrix(
  runif(length(dataset$cell_ids) * 2),
  ncol = 2,
  dimnames = list(dataset$cell_ids, c("comp_1", "comp_2"))
)
dimred
dimred_milestones &lt;- matrix(
  runif(2*4),
  ncol = 2,
  dimnames = list(c("A", "B", "C", "D"), c("comp_1", "comp_2"))
)
dimred_milestones
trajectory &lt;- add_dimred_projection(
  dataset,
  milestone_network = milestone_network,
  dimred = dimred,
  dimred_milestones = dimred_milestones
)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_end_state_probabilities'>Constructs a multifurcating trajectory using end state probabilities</h2><span id='topic+add_end_state_probabilities'></span>

<h3>Description</h3>

<p>Constructs a multifurcating trajectory using the pseudotime values of each cell and their end state probabilities.
If pseudotime values are not given, will use pseudotime already present in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_end_state_probabilities(
  dataset,
  end_state_probabilities,
  pseudotime = NULL,
  do_scale_minmax = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_end_state_probabilities_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_end_state_probabilities_+3A_end_state_probabilities">end_state_probabilities</code></td>
<td>
<p>A dataframe containing the <em>cell_id</em> and additional numeric columns containing the probability for every end milestone. If the tibble contains only a cell_id column, the data will be processed using <code>add_linear_trajectory</code></p>
</td></tr>
<tr><td><code id="add_end_state_probabilities_+3A_pseudotime">pseudotime</code></td>
<td>
<p>A named vector of pseudo times.</p>
</td></tr>
<tr><td><code id="add_end_state_probabilities_+3A_do_scale_minmax">do_scale_minmax</code></td>
<td>
<p>Whether or not to scale the pseudotime between 0 and 1.
Otherwise, will assume the values are already within that range.</p>
</td></tr>
<tr><td><code id="add_end_state_probabilities_+3A_...">...</code></td>
<td>
<p>Extras to be added to the trajectory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- wrap_data(cell_ids = letters)

pseudotime &lt;- runif(length(dataset$cell_ids))
names(pseudotime) &lt;- dataset$cell_ids
pseudotime
end_state_probabilities &lt;- tibble::tibble(
  cell_id = dataset$cell_ids,
  A = runif(length(dataset$cell_ids)),
  B = 1-A
)
end_state_probabilities
trajectory &lt;- add_end_state_probabilities(dataset, end_state_probabilities, pseudotime)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='add_expression'>Add count and normalised expression values to a dataset</h2><span id='topic+add_expression'></span><span id='topic+is_wrapper_with_expression'></span><span id='topic+get_expression'></span>

<h3>Description</h3>

<p>Add count and normalised expression values to a dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_expression(
  dataset,
  counts,
  expression,
  feature_info = NULL,
  expression_future = NULL,
  ...
)

is_wrapper_with_expression(dataset)

get_expression(dataset, expression_source = "expression")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_expression_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_expression_+3A_counts">counts</code></td>
<td>
<p>The counts values of genes (columns) within cells (rows). This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="add_expression_+3A_expression">expression</code></td>
<td>
<p>The normalised expression values of genes (columns) within cells (rows). This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="add_expression_+3A_feature_info">feature_info</code></td>
<td>
<p>Optional meta-information of the features, a dataframe with at least <em>feature_id</em> as column</p>
</td></tr>
<tr><td><code id="add_expression_+3A_expression_future">expression_future</code></td>
<td>
<p>Projected expression using RNA velocity of genes (columns) within cells (rows).  This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="add_expression_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the dataset</p>
</td></tr>
<tr><td><code id="add_expression_+3A_expression_source">expression_source</code></td>
<td>
<p>The source of expression, can be &quot;counts&quot;, &quot;expression&quot;, an expression matrix, or another dataset which contains expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the expression added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cell_ids &lt;- c("A", "B", "C")
counts &lt;- matrix(sample(0:10, 3*10, replace = TRUE), nrow = 3)
rownames(counts) &lt;- cell_ids
colnames(counts) &lt;- letters[1:10]
expression &lt;- log2(counts + 1)

dataset &lt;- wrap_data(id = "my_awesome_dataset", cell_ids = cell_ids)
dataset &lt;- add_expression(dataset, counts = counts, expression = expression)

str(dataset$expression)
str(dataset$counts)

</code></pre>

<hr>
<h2 id='add_feature_importance'>Add a feature importance to a dataset</h2><span id='topic+add_feature_importance'></span><span id='topic+is_wrapper_with_feature_importance'></span>

<h3>Description</h3>

<p>Add a feature importance to a dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_feature_importance(dataset, feature_importance, ...)

is_wrapper_with_feature_importance(dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_feature_importance_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_feature_importance_+3A_feature_importance">feature_importance</code></td>
<td>
<p>The impotances of the features, can be a named vector or a dataframe with columns <em>feature_id</em> and <em>importance</em></p>
</td></tr>
<tr><td><code id="add_feature_importance_+3A_...">...</code></td>
<td>
<p>Extra information to be stored in the dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the feature importance added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- example_dataset

feature_importance &lt;- runif(nrow(dataset$feature_info))
names(feature_importance) &lt;- dataset$feature_info$feature_id

dataset &lt;- add_feature_importance(dataset, feature_importance)
head(dataset$feature_importance)

</code></pre>

<hr>
<h2 id='add_grouping'>Add a cell grouping to a dataset</h2><span id='topic+add_grouping'></span><span id='topic+is_wrapper_with_grouping'></span><span id='topic+get_grouping'></span>

<h3>Description</h3>

<p>Add a cell grouping to a dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_grouping(dataset, grouping, group_ids = NULL, ...)

is_wrapper_with_grouping(dataset)

get_grouping(dataset, grouping = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_grouping_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_grouping_+3A_grouping">grouping</code></td>
<td>
<p>A grouping of the cells, can be a named vector or a dataframe with <em>group_id</em> and <em>cell_id</em></p>
</td></tr>
<tr><td><code id="add_grouping_+3A_group_ids">group_ids</code></td>
<td>
<p>All group identifiers, optional</p>
</td></tr>
<tr><td><code id="add_grouping_+3A_...">...</code></td>
<td>
<p>Extra information to be stored in the dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the grouping added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- example_dataset

grouping &lt;- sample(c("A", "B", "C"), length(dataset$cell_ids), replace = TRUE)
names(grouping) &lt;- dataset$cell_ids

dataset &lt;- add_grouping(dataset, grouping)
head(dataset$grouping)

</code></pre>

<hr>
<h2 id='add_linear_trajectory'>Constructs a linear trajectory using pseudotime values</h2><span id='topic+add_linear_trajectory'></span>

<h3>Description</h3>

<p>Constructs a linear trajectory using pseudotime values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_linear_trajectory(
  dataset,
  pseudotime,
  directed = FALSE,
  do_scale_minmax = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_linear_trajectory_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_linear_trajectory_+3A_pseudotime">pseudotime</code></td>
<td>
<p>A named vector of pseudo times.</p>
</td></tr>
<tr><td><code id="add_linear_trajectory_+3A_directed">directed</code></td>
<td>
<p>Whether the trajectory will be directed.</p>
</td></tr>
<tr><td><code id="add_linear_trajectory_+3A_do_scale_minmax">do_scale_minmax</code></td>
<td>
<p>Whether or not to scale the pseudotime between 0 and 1.
Otherwise, will assume the values are already within that range.</p>
</td></tr>
<tr><td><code id="add_linear_trajectory_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the trajectory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(tibble)
dataset &lt;- wrap_data(cell_ids = letters)

pseudotime &lt;- tibble(
  cell_id = dataset$cell_ids,
  pseudotime = runif(length(dataset$cell_ids))
)

trajectory &lt;- add_linear_trajectory(dataset, pseudotime)
</code></pre>

<hr>
<h2 id='add_prior_information'>Add or compute prior information for a trajectory</h2><span id='topic+add_prior_information'></span><span id='topic+is_wrapper_with_prior_information'></span><span id='topic+generate_prior_information'></span>

<h3>Description</h3>

<p>If you specify
</p>
<p>For example, what are the start cells, the end cells, to which milestone does each cell belong to, ...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_prior_information(
  dataset,
  start_id = NULL,
  end_id = NULL,
  groups_id = NULL,
  groups_network = NULL,
  features_id = NULL,
  groups_n = NULL,
  start_n = NULL,
  end_n = NULL,
  leaves_n = NULL,
  timecourse_continuous = NULL,
  timecourse_discrete = NULL,
  dimred = NULL,
  verbose = TRUE
)

is_wrapper_with_prior_information(dataset)

generate_prior_information(
  cell_ids,
  milestone_ids,
  milestone_network,
  milestone_percentages,
  progressions,
  divergence_regions,
  expression,
  feature_info = NULL,
  cell_info = NULL,
  marker_fdr = 0.005,
  given = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_prior_information_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_start_id">start_id</code></td>
<td>
<p>The start cells</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_end_id">end_id</code></td>
<td>
<p>The end cells</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_groups_id">groups_id</code></td>
<td>
<p>The grouping of cells, a dataframe with cell_id and group_id</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_groups_network">groups_network</code></td>
<td>
<p>The network between groups, a dataframe with from and to</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_features_id">features_id</code></td>
<td>
<p>The features (genes) important for the trajectory</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_groups_n">groups_n</code></td>
<td>
<p>Number of branches</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_start_n">start_n</code></td>
<td>
<p>Number of start states</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_end_n">end_n</code></td>
<td>
<p>Number of end states</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_leaves_n">leaves_n</code></td>
<td>
<p>Number of leaves</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_timecourse_continuous">timecourse_continuous</code></td>
<td>
<p>The time for every cell</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_timecourse_discrete">timecourse_discrete</code></td>
<td>
<p>The time for every cell in groups</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_dimred">dimred</code></td>
<td>
<p>A dimensionality reduction of the cells (see <code><a href="#topic+add_dimred">add_dimred()</a></code>)</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to print informative messages</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_cell_ids">cell_ids</code></td>
<td>
<p>The identifiers of the cells.</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_milestone_percentages">milestone_percentages</code></td>
<td>
<p>A data frame specifying what percentage milestone each cell
consists of.
Type: Data frame(cell_id = character, milestone_id = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_progressions">progressions</code></td>
<td>
<p>Specifies the progression of a cell along a transition in the milestone_network.
Type: Data frame(cell_id = character, from = character, to = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_divergence_regions">divergence_regions</code></td>
<td>
<p>A data frame specifying the divergence
regions between milestones (e.g. a bifurcation).
Type: Data frame(divergence_id = character, milestone_id = character, is_start = logical).</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_expression">expression</code></td>
<td>
<p>The normalised expression values of genes (columns) within cells (rows). This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_feature_info">feature_info</code></td>
<td>
<p>Optional meta-information pertaining the features.</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_cell_info">cell_info</code></td>
<td>
<p>Optional meta-information pertaining the cells.</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_marker_fdr">marker_fdr</code></td>
<td>
<p>Maximal FDR value for a gene to be considered a marker</p>
</td></tr>
<tr><td><code id="add_prior_information_+3A_given">given</code></td>
<td>
<p>Prior information already calculated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the dataset contains a trajectory (see <code><a href="#topic+add_trajectory">add_trajectory()</a></code>) and expression data, this function will compute and add prior information using <code><a href="#topic+generate_prior_information">generate_prior_information()</a></code>
</p>
<p>The dataset has to contain a trajectory for this to work
</p>


<h3>Value</h3>

<p>A dynwrap object with the prior information added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># add some prior information manually
dataset &lt;- example_dataset
dataset &lt;- add_prior_information(dataset, start_id = "Cell1")
dataset$prior_information$start_id

# compute prior information from a trajectory
trajectory &lt;- example_trajectory
trajectory &lt;- add_prior_information(trajectory)
trajectory$prior_information$end_id

</code></pre>

<hr>
<h2 id='add_regulatory_network'>Add a GRN to a dynwrap object</h2><span id='topic+add_regulatory_network'></span>

<h3>Description</h3>

<p>Add a GRN to a dynwrap object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_regulatory_network(
  dataset,
  regulatory_network,
  regulatory_network_sc = NULL,
  regulators = NULL,
  targets = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_regulatory_network_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_regulatory_network_+3A_regulatory_network">regulatory_network</code></td>
<td>
<p>A data frame consisting of three columns: <code>"regulator"</code>, <code>"target"</code>, <code>"strength"</code>.</p>
</td></tr>
<tr><td><code id="add_regulatory_network_+3A_regulatory_network_sc">regulatory_network_sc</code></td>
<td>
<p>A data frame consisting of four columns: <code>"cell_id"</code>, <code>"regulator"</code>, <code>"target"</code>, <code>"strength"</code>.</p>
</td></tr>
<tr><td><code id="add_regulatory_network_+3A_regulators">regulators</code></td>
<td>
<p>The feature ids of the regulators.</p>
</td></tr>
<tr><td><code id="add_regulatory_network_+3A_targets">targets</code></td>
<td>
<p>The feature ids of the targets.</p>
</td></tr>
<tr><td><code id="add_regulatory_network_+3A_...">...</code></td>
<td>
<p>Extra arguments to be saved in the model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the regulatory network added.
</p>

<hr>
<h2 id='add_root'>Root the trajectory</h2><span id='topic+add_root'></span><span id='topic+add_root_using_expression'></span><span id='topic+is_rooted'></span><span id='topic+remove_root'></span>

<h3>Description</h3>

<p>Designates a milestone as root, and changes the direction of any edges so that they move away from the specified root (if <code>flip_edges=TRUE</code>, default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_root(
  trajectory,
  root_cell_id = trajectory$root_cell_id,
  root_milestone_id = trajectory$root_milestone_id,
  flip_edges = TRUE
)

add_root_using_expression(
  trajectory,
  features_oi,
  expression_source = "expression"
)

is_rooted(trajectory)

remove_root(trajectory)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_root_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="add_root_+3A_root_cell_id">root_cell_id</code></td>
<td>
<p>The root cell id, not required if root_milestone_id is given</p>
</td></tr>
<tr><td><code id="add_root_+3A_root_milestone_id">root_milestone_id</code></td>
<td>
<p>The root milestone id, not required if root_cell_id is given</p>
</td></tr>
<tr><td><code id="add_root_+3A_flip_edges">flip_edges</code></td>
<td>
<p>Whether to flip edges which are going in the other direction compared to the root</p>
</td></tr>
<tr><td><code id="add_root_+3A_features_oi">features_oi</code></td>
<td>
<p>The feature ids which will be used to root</p>
</td></tr>
<tr><td><code id="add_root_+3A_expression_source">expression_source</code></td>
<td>
<p>Source of the expression, either a string or a matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A <code>root_cell_id</code> can also be specified, and the root milestone will be determined as the milestone with the closest geodesic distance to this cell.
</p>


<h3>Value</h3>

<p>A trajectory, with a <em>root_milestone_id</em> and with adapted <em>milestone_network</em> and <em>progressions</em> based on the rooting.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># add a root using a root cell
trajectory &lt;- example_trajectory
trajectory &lt;- add_root(
  trajectory,
  root_cell_id = sample(trajectory$cell_ids, 1)
)
trajectory$root_milestone_id

# add a root using a root milestone id
trajectory &lt;- add_root(
  trajectory,
  root_milestone_id = "milestone_end"
)
trajectory$root_milestone_id
trajectory$milestone_network

</code></pre>

<hr>
<h2 id='add_tde_overall'>Add information on overall differentially expressed features</h2><span id='topic+add_tde_overall'></span>

<h3>Description</h3>

<p>To calculate differential expression within trajectories, check out the <a href="https://github.com/dynverse/dynfeature">dynfeature</a> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_tde_overall(trajectory, tde_overall)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_tde_overall_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="add_tde_overall_+3A_tde_overall">tde_overall</code></td>
<td>
<p>A dataframe containing the <em>feature_id</em>, and some other columns including whether it is differentially expressed (<em>differentially_expressed</em>), the rank of differential expression among all other features (<em>rank</em>), the p-value (<em>pval</em>) or corrected value (<em>qval</em>), and the log-fold change (<em>lfc</em>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory containing <em>tde_overall</em>, a dataframe containing the <em>feature_id</em>, and some other columns including whether it is differentially expressed (<em>differentially_expressed</em>), the rank of differential expression among all other features (<em>rank</em>), the p-value (<em>pval</em>) or corrected value (<em>qval</em>), and the log-fold change (<em>lfc</em>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory &lt;- example_trajectory
tde_overall &lt;- tibble::tibble(
  feature_id = trajectory$feature_info$feature_id,
  differentially_expressed = sample(c(TRUE, FALSE), length(feature_id), replace = TRUE)
)
trajectory &lt;- add_tde_overall(trajectory, tde_overall)
trajectory$tde_overall

</code></pre>

<hr>
<h2 id='add_timings'>Add timings to a trajectory</h2><span id='topic+add_timings'></span><span id='topic+is_wrapper_with_timings'></span><span id='topic+add_timing_checkpoint'></span>

<h3>Description</h3>

<p>Add timings to a trajectory
</p>
<p>Helper function for storing timings information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_timings(trajectory, timings)

is_wrapper_with_timings(trajectory)

add_timing_checkpoint(timings, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_timings_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="add_timings_+3A_timings">timings</code></td>
<td>
<p>A list of timings.</p>
</td></tr>
<tr><td><code id="add_timings_+3A_name">name</code></td>
<td>
<p>The name of the timings checkpoint.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dynwrap object with the timings added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory &lt;- example_trajectory
trajectory &lt;- add_timings(
  trajectory,
  list(start = 0, end = 1)
)

</code></pre>

<hr>
<h2 id='add_trajectory'>Construct a trajectory given its milestone network and milestone percentages or progressions</h2><span id='topic+add_trajectory'></span><span id='topic+is_wrapper_with_trajectory'></span>

<h3>Description</h3>

<p>Construct a trajectory given its milestone network and milestone percentages or progressions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_trajectory(
  dataset,
  milestone_ids = NULL,
  milestone_network,
  divergence_regions = NULL,
  milestone_percentages = NULL,
  progressions = NULL,
  allow_self_loops = FALSE,
  ...
)

is_wrapper_with_trajectory(trajectory)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_trajectory_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_divergence_regions">divergence_regions</code></td>
<td>
<p>A data frame specifying the divergence
regions between milestones (e.g. a bifurcation).
Type: Data frame(divergence_id = character, milestone_id = character, is_start = logical).</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_milestone_percentages">milestone_percentages</code></td>
<td>
<p>A data frame specifying what percentage milestone each cell
consists of.
Type: Data frame(cell_id = character, milestone_id = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_progressions">progressions</code></td>
<td>
<p>Specifies the progression of a cell along a transition in the milestone_network.
Type: Data frame(cell_id = character, from = character, to = character, percentage = numeric).</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_allow_self_loops">allow_self_loops</code></td>
<td>
<p>Whether to allow self loops
Type: Logical</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the dataset</p>
</td></tr>
<tr><td><code id="add_trajectory_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The dataset object with trajectory information, including:
</p>

<ul>
<li> <p><em>milestone_ids</em>: The names of the milestones, a character vector.
</p>
</li>
<li> <p><em>milestone_network</em>: The network between the milestones, a dataframe with the <em>from</em> milestone, <em>to</em> milestone, <em>length</em> of the edge, and whether it is <em>directed</em>.
</p>
</li>
<li> <p><em>divergence_regions</em>: The regions between three or more milestones where cells are diverging, a dataframe with the divergence id (<em>divergence_id</em>), the milestone id (<em>milestone_id</em>) and whether this milestone is the start of the divergence (<em>is_start</em>)
</p>
</li>
<li> <p><em>milestone_percentages</em>: For each cell its closeness to a particular milestone, a dataframe with the cell id (<em>cell_id</em>), the milestone id (<em>milestone_id</em>), and its <em>percentage</em> (a number between 0 and 1 where higher values indicate that a cell is close to the milestone).
</p>
</li>
<li> <p><em>progressions</em>: For each cell its progression along a particular edge of the <em>milestone_network</em>. Contains the same information as <em>milestone_percentages</em>. A dataframe with cell id (<em>cell_id</em>), <em>from</em> milestone, <em>to</em> milestone, and its <em>percentage</em> (a number between 0 and 1 where higher values indicate that a cell is close to the <em>to</em> milestone and far from the <em>from</em> milestone).
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library(dplyr)
library(tibble)

dataset &lt;- wrap_data(cell_ids = letters)

milestone_network &lt;- tribble(
  ~from, ~to, ~length, ~directed,
  "A", "B", 1, FALSE,
  "B", "C", 2, FALSE,
  "B", "D", 1, FALSE,
)
milestone_network
progressions &lt;- milestone_network %&gt;%
  sample_n(length(dataset$cell_ids), replace = TRUE, weight = length) %&gt;%
  mutate(
    cell_id = dataset$cell_ids,
    percentage = runif(n())
  ) %&gt;%
  select(cell_id, from, to, percentage)
progressions
divergence_regions &lt;- tribble(
  ~divergence_id, ~milestone_id, ~is_start,
  "1", "A", TRUE,
  "1", "B", FALSE,
  "1", "C", FALSE
)
divergence_regions

trajectory &lt;- add_trajectory(
  dataset,
  milestone_network = milestone_network,
  divergence_regions = divergence_regions,
  progressions = progressions
)

# for plotting the result, install dynplot
#- dynplot::plot_graph(trajectory)
</code></pre>

<hr>
<h2 id='allowed_inputs'>All allowed inputs for a TI method</h2><span id='topic+allowed_inputs'></span>

<h3>Description</h3>

<p>All allowed inputs for a TI method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allowed_inputs
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 16 rows and 2 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>allowed_inputs

</code></pre>

<hr>
<h2 id='allowed_outputs'>All allowed outputs for a TI method</h2><span id='topic+allowed_outputs'></span>

<h3>Description</h3>

<p>All allowed outputs for a TI method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>allowed_outputs
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 14 rows and 5 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>allowed_outputs

</code></pre>

<hr>
<h2 id='calculate_attraction'>Calculate the attraction of cells to other cells using velocity</h2><span id='topic+calculate_attraction'></span>

<h3>Description</h3>

<p>Calculate the attraction of cells to other cells using velocity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_attraction(
  current,
  projected,
  cells = colnames(projected),
  n_waypoints = 50,
  k = 50
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_attraction_+3A_current">current</code></td>
<td>
<p>Current expression</p>
</td></tr>
<tr><td><code id="calculate_attraction_+3A_projected">projected</code></td>
<td>
<p>Projected expression based on RNA velocity</p>
</td></tr>
<tr><td><code id="calculate_attraction_+3A_cells">cells</code></td>
<td>
<p>Which cells to use</p>
</td></tr>
<tr><td><code id="calculate_attraction_+3A_n_waypoints">n_waypoints</code></td>
<td>
<p>Number of waypoints to use</p>
</td></tr>
<tr><td><code id="calculate_attraction_+3A_k">k</code></td>
<td>
<p>K knns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix containing the attraction ([-1, 1]) of each cell to the waypoint cells
</p>

<hr>
<h2 id='calculate_average_by_group'>Calculate average values of a matrix</h2><span id='topic+calculate_average_by_group'></span>

<h3>Description</h3>

<p><code>calculate_average_by_group</code> will calculate an average value per group, given a matrix with cells in the rows and some features in the columns (e.g. expression matrix)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_average_by_group(x, cell_grouping)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_average_by_group_+3A_x">x</code></td>
<td>
<p>A matrix. One row for every cell; one column for every feature. The rows must be named.</p>
</td></tr>
<tr><td><code id="calculate_average_by_group_+3A_cell_grouping">cell_grouping</code></td>
<td>
<p>A data frame denoting the grouping of the cells.
Format: <code>tibble(cell_id = character(), group_id = character())</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing for each feature (column) the average
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calculate_average_by_group(
  x = example_trajectory$expression,
  cell_grouping = example_trajectory$prior_information$groups_id
)

</code></pre>

<hr>
<h2 id='calculate_geodesic_distances'>Calculate geodesic distances between cells in a trajectory</h2><span id='topic+calculate_geodesic_distances'></span><span id='topic+compute_tented_geodesic_distances'></span>

<h3>Description</h3>

<p>Will calculate geodesic distances between cells within a trajectory. To speed things up, only the distances with a set of waypoint cells are calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_geodesic_distances(
  trajectory,
  waypoint_cells = NULL,
  waypoint_milestone_percentages = NULL,
  directed = FALSE
)

compute_tented_geodesic_distances(
  trajectory,
  waypoint_cells = NULL,
  waypoint_milestone_percentages = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_geodesic_distances_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="calculate_geodesic_distances_+3A_waypoint_cells">waypoint_cells</code></td>
<td>
<p>A vector of waypoint cells. Only the geodesic distances between waypoint cells and all other cells will be calculated.</p>
</td></tr>
<tr><td><code id="calculate_geodesic_distances_+3A_waypoint_milestone_percentages">waypoint_milestone_percentages</code></td>
<td>
<p>The milestone percentages of non-cell waypoints, containing waypoint_id, milestone_id and percentage columns</p>
</td></tr>
<tr><td><code id="calculate_geodesic_distances_+3A_directed">directed</code></td>
<td>
<p>Take into account the directions of the milestone edges. The cells that cannot be reached from a particular waypoint will have distance infinity. You can also give a character, if it is &quot;forward&quot; it will look forward, if it is &quot;reverse&quot; it will look in the reversed direction</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The geodesic distance takes into account the length of an edge regions of delayed commitment.
</p>


<h3>Value</h3>

<p>A matrix containing geodesic distances between each waypoint cell (rows) and cell (columns)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geodesic_distances &lt;- calculate_geodesic_distances(example_trajectory)
geodesic_distances[1:10, 1:10]

</code></pre>

<hr>
<h2 id='calculate_pseudotime'>Add or calculate pseudotime as distance from the root</h2><span id='topic+calculate_pseudotime'></span><span id='topic+add_pseudotime'></span>

<h3>Description</h3>

<p>When calculating the pseudotime, the trajectory is expected to be rooted (see <code><a href="#topic+add_root">add_root()</a></code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_pseudotime(trajectory)

add_pseudotime(trajectory, pseudotime = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_pseudotime_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="calculate_pseudotime_+3A_pseudotime">pseudotime</code></td>
<td>
<p>Named vector containing the pseudotime for every cell. If not given, the pseudotime will be calculated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The trajectory with <em>pseudotime</em> added, which is a named vector containing the pseudotime values for every cell.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_root">add_root()</a></code>, <code><a href="#topic+add_linear_trajectory">add_linear_trajectory()</a></code>
</p>

<hr>
<h2 id='calculate_trajectory_dimred'>Layout the trajectory and its cells in 2 dimensions using a graph layout</h2><span id='topic+calculate_trajectory_dimred'></span>

<h3>Description</h3>

<p>Layout the trajectory and its cells in 2 dimensions using a graph layout
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_trajectory_dimred(trajectory, adjust_weights = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_trajectory_dimred_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="calculate_trajectory_dimred_+3A_adjust_weights">adjust_weights</code></td>
<td>
<p>Whether or not to rescale the milestone network weights</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containg
</p>

<ul>
<li> <p><em>milestone_positions</em>: A dataframe containing the <em>milestone_id</em>
and the location of each milestone (<em>comp_1</em> and <em>comp_2</em>)
</p>
</li>
<li> <p><em>edge_positions</em>: A dataframe containing for each edge
(<em>from</em>, <em>to</em>, <em>length</em> and <em>directed</em> columns) the position
of the from milestone (<em>comp_1_from</em> and <em>comp_2_from</em>) and to
milestone (<em>comp_1_to</em> and <em>comp_2_to</em>).
</p>
</li>
<li> <p><em>cell_positions</em>: A dataframe containing the <em>cell_id</em> and the
location of each cell (<em>comp_1</em> and <em>comp_2</em>)
</p>
</li>
<li> <p><em>divergence_edge_positions</em>: A dataframe as <em>edge_positions</em>
but for each edge within a divergence
</p>
</li>
<li> <p><em>divergence_polygon_positions</em>: A dataframe containing the <em>triangle_id</em>
and the location of the milestone within a divergence (<em>comp_1</em> and <em>comp_2</em>)
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+wrap_data">wrap_data()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory_dimred &lt;- calculate_trajectory_dimred(example_trajectory)
head(trajectory_dimred$milestone_positions)
head(trajectory_dimred$edge_positions)
head(trajectory_dimred$cell_positions)

</code></pre>

<hr>
<h2 id='classify_milestone_network'>Classify a milestone network</h2><span id='topic+classify_milestone_network'></span>

<h3>Description</h3>

<p>Classify a milestone network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify_milestone_network(milestone_network)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify_milestone_network_+3A_milestone_network">milestone_network</code></td>
<td>
<p>A milestone network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>

<ul>
<li> <p><em>network_type</em>: The network type (also known as the trajectory_type). See <code>dynwrap::trajectory_types</code> for an overview.
</p>
</li>
<li> <p><em>directed</em>: Whether the trajectory is directed
</p>
</li>
<li> <p><em>properties</em>: Different properties of the trajectory, including:
</p>

<ul>
<li> <p><em>is_directed</em>: Whether the trajectory is directed
</p>
</li>
<li> <p><em>max_degree</em>: The maximal degree
</p>
</li>
<li> <p><em>num_branch_nodes</em>: The number of branching nodes
</p>
</li>
<li> <p><em>num_outer_nodes</em>: Number of leaf (outer) nodes
</p>
</li>
<li> <p><em>is_self_loop</em>: Whether it contains self-loops
</p>
</li>
<li> <p><em>has_cycles</em>: Whether it has cycles
</p>
</li>
<li> <p><em>num_components</em>: The number of independent components
</p>
</li></ul>

</li></ul>



<h3>See Also</h3>

<p><code>dynwrap::trajectory_types</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>milestone_network &lt;- tibble::tibble(
  from = c("A", "B", "C"),
  to = c("B", "C", "A"),
  length = 1,
  directed = TRUE
)
classification &lt;- classify_milestone_network(milestone_network)
classification$network_type
classification$directed

milestone_network &lt;- tibble::tibble(
  from = c("A", "B", "B", "C", "C"),
  to = c("B", "C", "D", "E", "F"),
  length = 2,
  directed = FALSE
)
classification &lt;- classify_milestone_network(milestone_network)
classification$network_type
classification$directed
classification$props

</code></pre>

<hr>
<h2 id='common_param'>Common param</h2><span id='topic+common_param'></span>

<h3>Description</h3>

<p>These parameters are commonly used in dynwrap.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="common_param_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="common_param_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Internal documentation
</p>

<hr>
<h2 id='convert_definition'>Convert a definition loaded in from a yaml</h2><span id='topic+convert_definition'></span>

<h3>Description</h3>

<p>Convert a definition loaded in from a yaml
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_definition(definition_raw)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_definition_+3A_definition_raw">definition_raw</code></td>
<td>
<p>The raw definition loaded from the yaml</p>
</td></tr>
</table>

<hr>
<h2 id='convert_milestone_percentages_to_progressions'>Conversion between milestone percentages and progressions</h2><span id='topic+convert_milestone_percentages_to_progressions'></span>

<h3>Description</h3>

<p>Conversion between milestone percentages and progressions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_milestone_percentages_to_progressions(
  cell_ids,
  milestone_ids,
  milestone_network,
  milestone_percentages
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_milestone_percentages_to_progressions_+3A_cell_ids">cell_ids</code></td>
<td>
<p>The identifiers of the cells.</p>
</td></tr>
<tr><td><code id="convert_milestone_percentages_to_progressions_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="convert_milestone_percentages_to_progressions_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="convert_milestone_percentages_to_progressions_+3A_milestone_percentages">milestone_percentages</code></td>
<td>
<p>A data frame specifying what percentage milestone each cell
consists of.
Type: Data frame(cell_id = character, milestone_id = character, percentage = numeric).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns <code>cell_id</code>, <code>from</code>, <code>to</code>, <code>percentage</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_trajectory">add_trajectory()</a></code>, <a href="#topic+convert_progressions_to_milestone_percentages">convert_progressions_to_milestone_percentages</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>progressions &lt;- convert_milestone_percentages_to_progressions(
  cell_ids = example_trajectory$cell_ids,
  milestone_ids = example_trajectory$milestone_ids,
  milestone_network = example_trajectory$milestone_network,
  milestone_percentages = example_trajectory$milestone_percentages
)
head(progressions)

</code></pre>

<hr>
<h2 id='convert_progressions_to_milestone_percentages'>Conversion between milestone percentages and progressions</h2><span id='topic+convert_progressions_to_milestone_percentages'></span>

<h3>Description</h3>

<p>Conversion between milestone percentages and progressions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_progressions_to_milestone_percentages(
  cell_ids,
  milestone_ids,
  milestone_network,
  progressions
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_progressions_to_milestone_percentages_+3A_cell_ids">cell_ids</code></td>
<td>
<p>The identifiers of the cells.</p>
</td></tr>
<tr><td><code id="convert_progressions_to_milestone_percentages_+3A_milestone_ids">milestone_ids</code></td>
<td>
<p>The ids of the milestones in the trajectory. Type: Character vector.</p>
</td></tr>
<tr><td><code id="convert_progressions_to_milestone_percentages_+3A_milestone_network">milestone_network</code></td>
<td>
<p>The network of the milestones.
Type: Data frame(from = character, to = character, length = numeric, directed = logical).</p>
</td></tr>
<tr><td><code id="convert_progressions_to_milestone_percentages_+3A_progressions">progressions</code></td>
<td>
<p>Specifies the progression of a cell along a transition in the milestone_network.
Type: Data frame(cell_id = character, from = character, to = character, percentage = numeric).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns <code>cell_id</code>, <code>milestone_id</code>, and <code>percentage</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_trajectory">add_trajectory()</a></code>, <code><a href="#topic+convert_milestone_percentages_to_progressions">convert_milestone_percentages_to_progressions()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>milestone_percentages &lt;- convert_progressions_to_milestone_percentages(
  cell_ids = example_trajectory$cell_ids,
  milestone_ids = example_trajectory$milestone_ids,
  milestone_network = example_trajectory$milestone_network,
  progressions = example_trajectory$progressions
)
head(milestone_percentages)

</code></pre>

<hr>
<h2 id='create_ti_method_container'>Create a TI method from a docker / singularity container</h2><span id='topic+create_ti_method_container'></span>

<h3>Description</h3>

<p>These functions create a TI method from a container using <code>babelwhale</code>. Supports both docker and singularity as a backend. See <a href="../doc/create_ti_method_container.html"><code>vignette("create_ti_method_container", "dynwrap")</code></a> for a tutorial on how to create a containerized TI method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ti_method_container(
  container_id,
  pull_if_needed = TRUE,
  return_function = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ti_method_container_+3A_container_id">container_id</code></td>
<td>
<p>The name of the container repository (e.g. <code>"dynverse/ti_angle"</code>).</p>
</td></tr>
<tr><td><code id="create_ti_method_container_+3A_pull_if_needed">pull_if_needed</code></td>
<td>
<p>Pull the container if not yet available.</p>
</td></tr>
<tr><td><code id="create_ti_method_container_+3A_return_function">return_function</code></td>
<td>
<p>Whether to return a function that allows you to override the default parameters, or just return the method meta data as is.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A function that can be used to adapt the parameters of the method. This functions returns a list containing all metadata of the method, and can be used to <a href="#topic+infer_trajectory">infer a trajectory</a>
</p>


<h3>See Also</h3>

<p>vignette(&quot;create_ti_method_container&quot;, &quot;dynwrap&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(babelwhale)

# only run if docker works on this platform
if (test_docker_installation()) {
  method &lt;- create_ti_method_container("dynverse/ti_angle")
  trajectory &lt;- infer_trajectory(example_dataset, method())
}

</code></pre>

<hr>
<h2 id='create_ti_method_definition'>Create a TI method from a local method definition file</h2><span id='topic+create_ti_method_definition'></span>

<h3>Description</h3>

<p>The local method definition file describes a method that is runnable on the local system.
See <a href="../doc/create_ti_method_definition.html"><code>vignette("create_ti_method_definition", "dynwrap")</code></a> for a tutorial on how to create a containerized TI method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ti_method_definition(definition, script, return_function = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ti_method_definition_+3A_definition">definition</code></td>
<td>
<p>A definition, see <code><a href="#topic+definition">definition()</a></code></p>
</td></tr>
<tr><td><code id="create_ti_method_definition_+3A_script">script</code></td>
<td>
<p>Location of the script that will be executed. Has to contain a #!</p>
</td></tr>
<tr><td><code id="create_ti_method_definition_+3A_return_function">return_function</code></td>
<td>
<p>Whether to return a function that allows you to override the default parameters, or just return the method meta data as is.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A method definition
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# See the vignette "create_ti_method_definition" to get a good idea on how
# to use this function.

# create a definition.yaml file and a run.R/py script.
# method &lt;- create_ti_method_definition("definition.yml", "run.R")
# trajectory &lt;- infer_trajectory(example_dataset, method(), verbose = TRUE)

</code></pre>

<hr>
<h2 id='create_ti_method_r'>Create a TI method from an R function wrapper</h2><span id='topic+create_ti_method_r'></span>

<h3>Description</h3>

<p>Create a TI method from an R function wrapper
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ti_method_r(
  definition,
  run_fun,
  package_required = character(),
  package_loaded = character(),
  remotes_package = character(),
  return_function = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ti_method_r_+3A_definition">definition</code></td>
<td>
<p>A definition, see <code><a href="#topic+definition">definition()</a></code></p>
</td></tr>
<tr><td><code id="create_ti_method_r_+3A_run_fun">run_fun</code></td>
<td>
<p>A function to infer a trajectory, with parameters counts/expression, parameters, priors, verbose and seed</p>
</td></tr>
<tr><td><code id="create_ti_method_r_+3A_package_required">package_required</code></td>
<td>
<p>The packages that need to be installed before executing the method.</p>
</td></tr>
<tr><td><code id="create_ti_method_r_+3A_package_loaded">package_loaded</code></td>
<td>
<p>The packages that need to be loaded before executing the method.</p>
</td></tr>
<tr><td><code id="create_ti_method_r_+3A_remotes_package">remotes_package</code></td>
<td>
<p>Package from which the remote locations of dependencies have to be extracted, eg. <code>dynmethods</code>.</p>
</td></tr>
<tr><td><code id="create_ti_method_r_+3A_return_function">return_function</code></td>
<td>
<p>Whether to return a function that allows you to override the default parameters, or just return the method meta data as is.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A method definition
</p>


<h3>Examples</h3>

<pre><code class='language-R'># define the parameters and other metadata
definition &lt;- definition(
  method = def_method(
    id = "comp1"
  ),
  parameters = def_parameters(
    dynparam::integer_parameter(
      id = "component",
      default = 1,
      distribution = dynparam::uniform_distribution(1, 10),
      description = "The nth component to use"
    )
  ),
  wrapper = def_wrapper(
    input_required = "expression",
    input_optional = "start_id"
  )
)

# define a wrapper function
run_fun &lt;- function(expression, priors, parameters, seed, verbose) {
  pca &lt;- prcomp(expression)

  pseudotime &lt;- pca$x[, parameters$component]

  # flip pseudotimes using start_id
  if (!is.null(priors$start_id)) {
    if(mean(pseudotime[start_id]) &gt; 0.5) {
     pseudotime &lt;- 1-pseudotime
    }
  }

  wrap_data(cell_ids = rownames(expression)) %&gt;%
    add_linear_trajectory(pseudotime = pseudotime)
}

method &lt;- create_ti_method_r(definition, run_fun, package_loaded = "dplyr")
trajectory &lt;- infer_trajectory(example_dataset, method())

</code></pre>

<hr>
<h2 id='def_author'>Meta information on an author</h2><span id='topic+def_author'></span>

<h3>Description</h3>

<p>Meta information on an author
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_author(given, family, email = NULL, github = NULL, orcid = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_author_+3A_given">given</code></td>
<td>
<p>The given name</p>
</td></tr>
<tr><td><code id="def_author_+3A_family">family</code></td>
<td>
<p>The family name</p>
</td></tr>
<tr><td><code id="def_author_+3A_email">email</code></td>
<td>
<p>The email address</p>
</td></tr>
<tr><td><code id="def_author_+3A_github">github</code></td>
<td>
<p>The github handle</p>
</td></tr>
<tr><td><code id="def_author_+3A_orcid">orcid</code></td>
<td>
<p>The orcid id</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_author(
  given = "Bob",
  family = "Dylan",
  email = "bob@dylan.com",
  github = "bobdylan",
  orcid = "0000-0003-1234-5678"
)

</code></pre>

<hr>
<h2 id='def_container'>Meta information on the container in which the wrapper resides</h2><span id='topic+def_container'></span>

<h3>Description</h3>

<p>Meta information on the container in which the wrapper resides
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_container(docker, url = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_container_+3A_docker">docker</code></td>
<td>
<p>The handle of the docker container</p>
</td></tr>
<tr><td><code id="def_container_+3A_url">url</code></td>
<td>
<p>An url of where the docker codebase resides (containing definition.yml, Dockerfile, ...)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_container(
  docker = "bobdylan/ti_some_method",
  url = "https://github.com/bobdylan/ti_some_method"
)
</code></pre>

<hr>
<h2 id='def_manuscript'>Meta information on the manuscript</h2><span id='topic+def_manuscript'></span>

<h3>Description</h3>

<p>Meta information on the manuscript
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_manuscript(
  doi = NULL,
  google_scholar_cluster_id = NULL,
  preprint_date = NULL,
  publication_date = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_manuscript_+3A_doi">doi</code></td>
<td>
<p>A doi identifier (not an url)</p>
</td></tr>
<tr><td><code id="def_manuscript_+3A_google_scholar_cluster_id">google_scholar_cluster_id</code></td>
<td>
<p>The google cluster id. Finding this id is a bit tricky;
you need to find the manuscript on one of the author pages, and hover over the 'All X versions' button.
Example: <a href="https://goo.gl/Y9uLFs">google scholar page</a>, <a href="https://i.imgur.com/03eLCaO.png">screenshot</a>.</p>
</td></tr>
<tr><td><code id="def_manuscript_+3A_preprint_date">preprint_date</code></td>
<td>
<p>Date of publication of the preprint (format: YYYY-MM-DD).</p>
</td></tr>
<tr><td><code id="def_manuscript_+3A_publication_date">publication_date</code></td>
<td>
<p>Date of publication of the peer-reviewed manuscript (format: YYYY-MM-DD).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_manuscript(
  doi = "101010101/1101010101",
  google_scholar_cluster_id = "1010001010101111211",
  preprint_date = "1970-01-30",
  publication_date = "1970-01-31"
)
</code></pre>

<hr>
<h2 id='def_method'>Define meta information on the TI method.</h2><span id='topic+def_method'></span>

<h3>Description</h3>

<p>Define meta information on the TI method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_method(
  id,
  name = id,
  source = "tool",
  tool_id = NULL,
  platform = NULL,
  url = NULL,
  license = NULL,
  authors = list(),
  description = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_method_+3A_id">id</code></td>
<td>
<p>An id by which to identify a method. Should only contain lowercase letters or underscores.</p>
</td></tr>
<tr><td><code id="def_method_+3A_name">name</code></td>
<td>
<p>The name of the method.</p>
</td></tr>
<tr><td><code id="def_method_+3A_source">source</code></td>
<td>
<p>The type of TI method. Options are :
</p>

<ul>
<li> <p><code>"tool"</code>: a published TI method (peer-reviewed or preprint) (default),
</p>
</li>
<li> <p><code>"adaptation"</code>: an adaptation of a published method,
</p>
</li>
<li> <p><code>"offtheshelf"</code>: a method constructed from off-the-shelf algorithms,
</p>
</li>
<li> <p><code>"control"</code>: a control TI method (so not actually a TI method).
</p>
</li></ul>
</td></tr>
<tr><td><code id="def_method_+3A_tool_id">tool_id</code></td>
<td>
<p>If there are multiple TI methods from the same toolkit, the name of the toolkit can be specified here.</p>
</td></tr>
<tr><td><code id="def_method_+3A_platform">platform</code></td>
<td>
<p>The platform the TI method uses (e.g. R, Python, C++, ...).</p>
</td></tr>
<tr><td><code id="def_method_+3A_url">url</code></td>
<td>
<p>An URL to the codebase of the method.</p>
</td></tr>
<tr><td><code id="def_method_+3A_license">license</code></td>
<td>
<p>The software license the method uses (e.g. GPL-3, BSD-3, Artistic-2.0, MIT).</p>
</td></tr>
<tr><td><code id="def_method_+3A_authors">authors</code></td>
<td>
<p>A list of authors (see example).</p>
</td></tr>
<tr><td><code id="def_method_+3A_description">description</code></td>
<td>
<p>Additional information on the method</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_method(
  id = "some_method",
  name = "Some method &lt;3",
  source = "tool",
  tool_id = "bobstoolkit",
  platform = "VBA",
  url = "https://github.com/bobdylan/singlecellvba",
  license = "GPL-3",
  authors = list(
    def_author(
      given = "Bob",
      family = "Dylan",
      email = "bob@dylan.com",
      github = "bobdylan",
      orcid = "0000-0003-1234-5678"
    )
  ),
  description = "I love trajectories!!"
)
</code></pre>

<hr>
<h2 id='def_package'>Meta information on the package in which the TI function resides</h2><span id='topic+def_package'></span>

<h3>Description</h3>

<p>Meta information on the package in which the TI function resides
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_package(remote, name, function_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_package_+3A_remote">remote</code></td>
<td>
<p>The github repository handle</p>
</td></tr>
<tr><td><code id="def_package_+3A_name">name</code></td>
<td>
<p>The name of the package</p>
</td></tr>
<tr><td><code id="def_package_+3A_function_name">function_name</code></td>
<td>
<p>The name of the function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_package(
  remote = "rcannood/SCORPIUS",
  name = "SCORPIUS",
  function_name = "ti_scorpius"
)
</code></pre>

<hr>
<h2 id='def_parameters'>Meta information on the parameters of the TI method</h2><span id='topic+def_parameters'></span>

<h3>Description</h3>

<p>Parameters can be defined using <code><a href="dynparam.html#topic+dynparam">dynparam::dynparam()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_parameters(..., parameters = NULL, forbidden = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_parameters_+3A_...">...</code></td>
<td>
<p>Parameters to wrap in a parameter set.</p>
</td></tr>
<tr><td><code id="def_parameters_+3A_parameters">parameters</code></td>
<td>
<p>A list of parameters to wrap in a parameter set.</p>
</td></tr>
<tr><td><code id="def_parameters_+3A_forbidden">forbidden</code></td>
<td>
<p>States forbidden region of parameter via a character vector, which will be turned into an expression.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(dynparam)
def_parameters(
  character_parameter(id = "method", default = "one", values = c("one", "two", "three")),
  integer_parameter(
    id = "ndim",
    default = 3L,
    distribution = uniform_distribution(lower = 2L, upper = 20L)
  ),
  numeric_parameter(
    id = "beta",
    default = 0.005,
    distribution = expuniform_distribution(lower = 1e-10, upper = 1)
  )
)
</code></pre>

<hr>
<h2 id='def_wrapper'>Meta information on the wrapper</h2><span id='topic+def_wrapper'></span>

<h3>Description</h3>

<p>Meta information on the wrapper
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_wrapper(
  input_required,
  input_optional = character(),
  type = "trajectory",
  topology_inference = NULL,
  trajectory_types = character()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_wrapper_+3A_input_required">input_required</code></td>
<td>
<p>The required inputs for this method. See <code>dynwrap::allowed_inputs()</code>.</p>
</td></tr>
<tr><td><code id="def_wrapper_+3A_input_optional">input_optional</code></td>
<td>
<p>Optional inputs for this method. See <code>dynwrap::allowed_inputs()</code>.</p>
</td></tr>
<tr><td><code id="def_wrapper_+3A_type">type</code></td>
<td>
<p>Which type of trajectory post-processing is used. Possible values:
<code>"trajectory"</code> (default), <code>"linear_trajectory"</code>, <code>"cyclic_trajectory"</code>, <code>"branch_trajectory"</code>,
<code>"cluster_graph"</code>, <code>"dimred_projection"</code>, <code>"end_state_probabilities"</code>, <code>"cell_graph"</code>.</p>
</td></tr>
<tr><td><code id="def_wrapper_+3A_topology_inference">topology_inference</code></td>
<td>
<p>Whether the topology is fixed (<code>"fixed"</code>), free (<code>"free"</code>),
or fixed by a parameter provided to the algorithm (<code>"param"</code>).</p>
</td></tr>
<tr><td><code id="def_wrapper_+3A_trajectory_types">trajectory_types</code></td>
<td>
<p>The possible trajectory types this method can return. Must be a subset of
<code>c("cyclic", "linear", "bifurcation", "convergence", "multifurcation", "tree", "graph", "acyclic_graph", "disconnected_graph")</code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>def_wrapper(
  input_required = c("expression", "start_id"),
  input_optional = "groups_n",
  type = "dimred_projection",
  trajectory_types = c("linear", "cyclic"),
  topology_inference = "free"
)
</code></pre>

<hr>
<h2 id='definition'>Create a definition</h2><span id='topic+definition'></span><span id='topic+is_ti_method'></span>

<h3>Description</h3>

<p>A definition contains meta information on a TI method and various aspects thereof.
For brevity, the example only contains a minimum example, check the documentation
of the <code style="white-space: pre;">&#8288;def_*&#8288;</code> helper functions for more extensive examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>definition(
  method,
  wrapper,
  manuscript = NULL,
  container = NULL,
  package = NULL,
  parameters = parameter_set()
)

is_ti_method(method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="definition_+3A_method">method</code></td>
<td>
<p>Meta information on the TI method (see <code><a href="#topic+def_method">def_method()</a></code>).</p>
</td></tr>
<tr><td><code id="definition_+3A_wrapper">wrapper</code></td>
<td>
<p>Meta information on the wrapper itself (see <code><a href="#topic+def_wrapper">def_wrapper()</a></code>).</p>
</td></tr>
<tr><td><code id="definition_+3A_manuscript">manuscript</code></td>
<td>
<p>Meta information on the manuscript, if applicable (see <code><a href="#topic+def_manuscript">def_manuscript()</a></code>).</p>
</td></tr>
<tr><td><code id="definition_+3A_container">container</code></td>
<td>
<p>Meta information on the container in which the wrapper resides, if applicable (see <code><a href="#topic+def_container">def_container()</a></code>).</p>
</td></tr>
<tr><td><code id="definition_+3A_package">package</code></td>
<td>
<p>Meta information on the package in which the wrapper resides, if applicable (see <code><a href="#topic+def_package">def_package()</a></code>).</p>
</td></tr>
<tr><td><code id="definition_+3A_parameters">parameters</code></td>
<td>
<p>Meta information on the parameters of the TI method (see <code><a href="#topic+def_parameters">def_parameters()</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A method definition
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(dynparam)
definition(
  method = def_method(id = "some_method"),
  wrapper = def_wrapper(input_required = "expression"),
  parameters = parameter_set(
    integer_parameter(id = "k", default = 5L, distribution = uniform_distribution(3L, 20L))
  )
)
</code></pre>

<hr>
<h2 id='dynwrap'>Inferring and adapting single-cell trajectories</h2><span id='topic+dynwrap'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" alt="logo.png" />
</p>

<hr>
<h2 id='example_dataset'>Example dataset</h2><span id='topic+example_dataset'></span>

<h3>Description</h3>

<p>Example dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_dataset
</code></pre>


<h3>Format</h3>

<p>An object of class <code>dynwrap::with_dimred</code> (inherits from <code>dynwrap::with_expression</code>, <code>dynwrap::data_wrapper</code>, <code>list</code>) of length 11.
</p>

<hr>
<h2 id='example_trajectory'>Example trajectory</h2><span id='topic+example_trajectory'></span>

<h3>Description</h3>

<p>Example trajectory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_trajectory
</code></pre>


<h3>Format</h3>

<p>An object of class <code>dynwrap::with_dimred</code> (inherits from <code>dynwrap::with_cell_waypoints</code>, <code>dynwrap::with_prior</code>, <code>dynwrap::with_trajectory</code>, <code>dynwrap::with_dimred</code>, <code>dynwrap::with_expression</code>, <code>dynwrap::data_wrapper</code>, <code>list</code>) of length 21.
</p>

<hr>
<h2 id='flip_edges'>Flip a set of edges of the milestone network</h2><span id='topic+flip_edges'></span>

<h3>Description</h3>

<p>Note that this will remove associated roots, reroot the trajectory using <code><a href="#topic+add_root">add_root()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flip_edges(trajectory, milestone_network_toflip)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flip_edges_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="flip_edges_+3A_milestone_network_toflip">milestone_network_toflip</code></td>
<td>
<p>A dataframe with a from and to column, containing the subset of the milestone network #'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory object
</p>

<hr>
<h2 id='gather_cells_at_milestones'>Gather cells to their closest milestones</h2><span id='topic+gather_cells_at_milestones'></span>

<h3>Description</h3>

<p>Cells will be moved to their closest milestones.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gather_cells_at_milestones(trajectory)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gather_cells_at_milestones_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A trajectory where cells where moved to the closest milestone, the milestone_percentages and progressions will be adapated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory &lt;- example_trajectory
trajectory &lt;- gather_cells_at_milestones(trajectory)
head(trajectory$milestone_percentages)

</code></pre>

<hr>
<h2 id='generate_parameter_documentation'>Generate the parameter documentation of a method, use with <code style="white-space: pre;">&#8288;@eval&#8288;</code></h2><span id='topic+generate_parameter_documentation'></span>

<h3>Description</h3>

<p>Generate the parameter documentation of a method, use with <code style="white-space: pre;">&#8288;@eval&#8288;</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_parameter_documentation(definition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_parameter_documentation_+3A_definition">definition</code></td>
<td>
<p>The definition which contain the parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the roxygen tags
</p>

<hr>
<h2 id='get_default_parameters'>Get the default parameters of a method</h2><span id='topic+get_default_parameters'></span>

<h3>Description</h3>

<p>Get the default parameters of a method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_parameters(definition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_default_parameters_+3A_definition">definition</code></td>
<td>
<p>A TI method description</p>
</td></tr>
</table>

<hr>
<h2 id='get_ti_methods'>Return all TI that are installed in one or more packages</h2><span id='topic+get_ti_methods'></span>

<h3>Description</h3>

<p>Return all TI that are installed in one or more packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ti_methods(
  method_ids = NULL,
  as_tibble = TRUE,
  ti_packages = ifelse(is_dynmethods_installed(), "dynmethods", "dynwrap"),
  evaluate = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ti_methods_+3A_method_ids">method_ids</code></td>
<td>
<p>The method identifiers. NULL if listing all methods</p>
</td></tr>
<tr><td><code id="get_ti_methods_+3A_as_tibble">as_tibble</code></td>
<td>
<p>Whether or not to return the ti_methods as a tibble</p>
</td></tr>
<tr><td><code id="get_ti_methods_+3A_ti_packages">ti_packages</code></td>
<td>
<p>In which packages to look for TI methods. This will by default look into dynmethods if it is installed, otherwise in dynwrap.</p>
</td></tr>
<tr><td><code id="get_ti_methods_+3A_evaluate">evaluate</code></td>
<td>
<p>Whether to evaluate the functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe (or list if <code>as_tibble = FALSE</code>) containing the name (<em>id</em>) of the TI method and the function (<em>fun</em>) to load in the method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(get_ti_methods())

</code></pre>

<hr>
<h2 id='group_from_trajectory'>Create a grouping from a trajectory</h2><span id='topic+group_from_trajectory'></span><span id='topic+group_onto_trajectory_edges'></span><span id='topic+group_onto_nearest_milestones'></span>

<h3>Description</h3>

<p>Grouping cells onto their edges, or grouping cells onto their nearest milestones
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_onto_trajectory_edges(trajectory, group_template = "{from}-&gt;{to}")

group_onto_nearest_milestones(trajectory)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="group_from_trajectory_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="group_from_trajectory_+3A_group_template">group_template</code></td>
<td>
<p>Processed by glue::glue to name the group</p>
</td></tr>
</table>

<hr>
<h2 id='infer_trajectories'>Infer one or more trajectories from a single-cell dataset</h2><span id='topic+infer_trajectories'></span><span id='topic+infer_trajectory'></span>

<h3>Description</h3>

<p>Infer one or more trajectories from a single-cell dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_trajectories(
  dataset,
  method,
  parameters = NULL,
  give_priors = NULL,
  seed = random_seed(),
  verbose = FALSE,
  return_verbose = FALSE,
  debug = FALSE,
  map_fun = map
)

infer_trajectory(
  dataset,
  method,
  parameters = NULL,
  give_priors = NULL,
  seed = random_seed(),
  verbose = FALSE,
  return_verbose = FALSE,
  debug = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_trajectories_+3A_dataset">dataset</code></td>
<td>
<p>One or more datasets as created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code>. Prior information can be added using <code><a href="#topic+add_prior_information">add_prior_information()</a></code>.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_method">method</code></td>
<td>
<p>One or more methods. Must be one of:
</p>

<ul>
<li><p>an object or list of ti_... objects (e.g. <code>dynmethods::ti_comp1</code>,
</p>
</li>
<li><p>a character vector containing the names of methods to execute (e.g. <code>"scorpius"</code>),
</p>
</li>
<li><p>a character vector containing dockerhub repositories (e.g. <code>dynverse/paga</code>), or
</p>
</li>
<li><p>a dynguidelines data frame.
</p>
</li></ul>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_parameters">parameters</code></td>
<td>
<p>A set of parameters to be used during trajectory inference.
A parameter set must be a named list of parameters.
If multiple methods were provided in the <code>method</code> parameter,
<code>parameters</code> must be an unnamed list of the same length.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_give_priors">give_priors</code></td>
<td>
<p>All the priors a method is allowed to receive.
Must be a subset of all available priors (<a href="#topic+priors">priors</a>).</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_seed">seed</code></td>
<td>
<p>A seed to be passed to the TI method.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to print information output.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_return_verbose">return_verbose</code></td>
<td>
<p>Whether to store and return messages printed by the method.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_debug">debug</code></td>
<td>
<p>Used for debugging containers methods.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_map_fun">map_fun</code></td>
<td>
<p>A map function to use when inferring trajectories with multiple datasets or methods.
Allows to parallellise the execution in an arbitrary way.</p>
</td></tr>
<tr><td><code id="infer_trajectories_+3A_...">...</code></td>
<td>
<p>Any additional parameters given to the method, will be concatenated to the parameters argument</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>infer_trajectory</code></strong>: A trajectory object, which is a list containing
</p>

<ul>
<li> <p><em>milestone_ids</em>: The names of the milestones, a character vector.
</p>
</li>
<li> <p><em>milestone_network</em>: The network between the milestones, a dataframe with the from milestone, to milestone, length of the edge, and whether it is directed.
</p>
</li>
<li> <p><em>divergence_regions</em>: The regions between three or more milestones where cells are diverging, a dataframe with the divergence id, the milestone id and whether this milestone is the start of the divergence
</p>
</li>
<li> <p><em>milestone_percentages</em>: For each cell its closeness to a particular milestone, a dataframe with the cell id, the milestone id, and its percentage (a number between 0 and 1 where higher values indicate that a cell is close to the milestone).
</p>
</li>
<li> <p><em>progressions</em>: For each cell its progression along a particular edge of the <em>milestone_network</em>. Contains the same information as <em>milestone_percentages</em>. A dataframe with cell id, from milestone, to milestone, and its percentage (a number between 0 and 1 where higher values indicate that a cell is close to the 'to' milestone and far from the 'from' milestone).
</p>
</li>
<li> <p><em>cell_ids</em>: The names of the cells
</p>
</li></ul>

<p>Some methods will include additional information in the output, such as
</p>

<ul>
<li><p> A dimensionality reduction (<em>dimred</em>), the location of the trajectory milestones and edges in this dimensionality reduction (<em>dimred_milestones</em>, <em>dimred_segment_progressions</em> and <em>dimred_segment_points</em>). See <code><a href="#topic+add_dimred">add_dimred()</a></code> for more information on these objects.
</p>
</li>
<li><p> A cell grouping (<em>grouping</em>). See <code><a href="#topic+add_grouping">add_grouping()</a></code> for more information on this object.
</p>
</li></ul>

<p><strong><code>infer_trajectories</code></strong>: A tibble containing the dataset and method identifiers (<em>dataset_id</em> and <em>method_id</em>), the trajectory model as described above (<em>model</em>), and a <em>summary</em> containing the execution times, output and error if appropriate
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- example_dataset
method &lt;- get_ti_methods(as_tibble = FALSE)[[1]]$fun

trajectory &lt;- infer_trajectory(dataset, method())

head(trajectory$milestone_network)
head(trajectory$progressions)

</code></pre>

<hr>
<h2 id='label_milestones'>Label milestones either manually (<code>label_milestones</code>) or using marker genes (<code>label_milestones_markers</code>)</h2><span id='topic+label_milestones'></span><span id='topic+label_milestones_markers'></span><span id='topic+is_wrapper_with_milestone_labelling'></span><span id='topic+get_milestone_labelling'></span>

<h3>Description</h3>

<p><code>label_milestones</code> can be used to manually assign labels to a milestone using their identifiers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>label_milestones(trajectory, labelling)

label_milestones_markers(
  trajectory,
  markers,
  expression_source = "expression",
  n_nearest_cells = 20
)

is_wrapper_with_milestone_labelling(trajectory)

get_milestone_labelling(trajectory, label_milestones = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="label_milestones_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="label_milestones_+3A_labelling">labelling</code></td>
<td>
<p>Named character vector containing for a milestone a new label</p>
</td></tr>
<tr><td><code id="label_milestones_+3A_markers">markers</code></td>
<td>
<p>List containing for each label a list of marker genes</p>
</td></tr>
<tr><td><code id="label_milestones_+3A_expression_source">expression_source</code></td>
<td>
<p>The expression source</p>
</td></tr>
<tr><td><code id="label_milestones_+3A_n_nearest_cells">n_nearest_cells</code></td>
<td>
<p>The number of nearest cells to use for extracting milestone expression</p>
</td></tr>
<tr><td><code id="label_milestones_+3A_label_milestones">label_milestones</code></td>
<td>
<p>How to label the milestones. Can be TRUE (in which case the labels within the trajectory will be used), &quot;all&quot; (in which case both given labels and milestone_ids will be used), a named character vector, or FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>label_milestones_markers</code> will assign a label to a milestone if its marker profile most closely resembles a given profile
</p>


<h3>Value</h3>

<p><strong><code>label_milestones</code></strong>:  A trajectory object with <em>milestone_labelling</em>, a named vector where milestone identifiers are mapped to their labels
</p>
<p><strong><code>get_milestone_labelling</code></strong>: A named vector giving a mapping between milestones and their labels. If certain milestones were not given a label, this vector will give the identifiers themselves.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory &lt;- example_trajectory

# manual labelling
trajectory &lt;- label_milestones(
  trajectory,
  labelling = c("milestone_begin" = "Let's go")
)
get_milestone_labelling(trajectory)

# marker gene labelling
trajectory &lt;- label_milestones_markers(
  trajectory,
  markers = list(A_high = "A")
)
get_milestone_labelling(trajectory)

is_wrapper_with_milestone_labelling(trajectory)

</code></pre>

<hr>
<h2 id='prior_usages'>Metadata on prior usages</h2><span id='topic+prior_usages'></span>

<h3>Description</h3>

<p>Metadata on prior usages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prior_usages
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 3 rows and 2 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prior_usages

</code></pre>

<hr>
<h2 id='priors'>Metadata on priors</h2><span id='topic+priors'></span>

<h3>Description</h3>

<p>Metadata on priors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>priors
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 13 rows and 6 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>priors

</code></pre>

<hr>
<h2 id='project_trajectory'>Project a trajectory onto a dimensionality reduction</h2><span id='topic+project_trajectory'></span><span id='topic+project_milestones'></span>

<h3>Description</h3>

<p>Project a trajectory onto a dimensionality reduction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>project_trajectory(
  trajectory,
  dimred,
  waypoints = select_waypoints(trajectory),
  trajectory_projection_sd = sum(trajectory$milestone_network$length) * 0.05
)

project_milestones(
  trajectory,
  dimred,
  trajectory_projection_sd = sum(trajectory$milestone_network$length) * 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="project_trajectory_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="project_trajectory_+3A_dimred">dimred</code></td>
<td>
<p>The dimensionality reduction of the cells. A matrix with the positions of cells (rows) in the dimensions (columns)</p>
</td></tr>
<tr><td><code id="project_trajectory_+3A_waypoints">waypoints</code></td>
<td>
<p>A set of waypoints, which can be created by <code><a href="#topic+add_waypoints">select_waypoints()</a></code>. It is a list containing:
</p>

<ul>
<li> <p><code>waypoints</code>: a dataframe containing in the very least the waypoint_id
</p>
</li>
<li> <p><code>milestone_percentages</code>: the positions of waypoints withing the trajectory
</p>
</li>
<li> <p><code>geodesic_distances</code>: matrix with precalculated geodesic distances between waypoints (rows) and cells (columns), optional
</p>
</li></ul>
</td></tr>
<tr><td><code id="project_trajectory_+3A_trajectory_projection_sd">trajectory_projection_sd</code></td>
<td>
<p>The standard deviation of the gaussian kernel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>

<ul>
<li> <p><em>dimred_segment_points</em>: The dimensionality reduction of a set of points along the trajectory. A matrix with the position of points (rows) in the dimensions (columns)
</p>
</li>
<li> <p><em>dimred_segment_progressions</em> The progressions of the points. A dataframe containing the <em>from</em> and <em>to</em> milestones, and their <em>progression</em>. Has the same number of rows as <em>dimred_segment_points</em>
</p>
</li>
<li> <p><em>dimred_milestones</em>: The dimensionality reduction of the milestones. A matrix with the position of milestones (rows) in the dimensions (columns)
</p>
</li></ul>

<p>These objects can be given to <code><a href="#topic+add_dimred">add_dimred()</a></code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_dimred">add_dimred()</a></code>
</p>

<hr>
<h2 id='project_waypoints'>Project waypoints of a trajectory (e.g. milestones) into a space defined by cells (e.g. expression or a dimensionality reduction)</h2><span id='topic+project_waypoints'></span>

<h3>Description</h3>

<p>This will first calculate the geodesic distance of each cell to the waypoint. This distance is used as a weight
</p>


<h3>Usage</h3>

<pre><code class='language-R'>project_waypoints(
  trajectory,
  space,
  waypoints = select_waypoints(trajectory),
  trajectory_projection_sd = sum(trajectory$milestone_network$length) * 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="project_waypoints_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="project_waypoints_+3A_space">space</code></td>
<td>
<p>A matrix with cells in rows and different dimensions in the columns. This is typically an expression matrix or a dimensionality reduction</p>
</td></tr>
<tr><td><code id="project_waypoints_+3A_waypoints">waypoints</code></td>
<td>
<p>A set of waypoints, which can be created by <code><a href="#topic+add_waypoints">select_waypoints()</a></code>. It is a list containing:
</p>

<ul>
<li> <p><code>waypoints</code>: a dataframe containing in the very least the waypoint_id
</p>
</li>
<li> <p><code>milestone_percentages</code>: the positions of waypoints withing the trajectory
</p>
</li>
<li> <p><code>geodesic_distances</code>: matrix with precalculated geodesic distances between waypoints (rows) and cells (columns), optional
</p>
</li></ul>
</td></tr>
<tr><td><code id="project_waypoints_+3A_trajectory_projection_sd">trajectory_projection_sd</code></td>
<td>
<p>The standard deviation of the gaussian kernel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix in which the waypoints (rows) were projected into a new space defined by the same number of dimensions (columns) as in the <code>space</code> argument
</p>

<hr>
<h2 id='random_seed'>Generate a random seed</h2><span id='topic+random_seed'></span>

<h3>Description</h3>

<p>From the current seed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_seed()
</code></pre>


<h3>Value</h3>

<p>A random seed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>random_seed()

</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+test_docker_installation'></span><span id='topic+test_singularity_installation'></span><span id='topic+create_config'></span><span id='topic+create_docker_config'></span><span id='topic+create_singularity_config'></span><span id='topic+get_default_config'></span><span id='topic+set_default_config'></span><span id='topic+test_h5_installation'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>babelwhale</dt><dd><p><code><a href="babelwhale.html#topic+create_config">create_config</a></code>, <code><a href="babelwhale.html#topic+create_config">create_docker_config</a></code>, <code><a href="babelwhale.html#topic+create_config">create_singularity_config</a></code>, <code><a href="babelwhale.html#topic+create_config">get_default_config</a></code>, <code><a href="babelwhale.html#topic+create_config">set_default_config</a></code>, <code><a href="babelwhale.html#topic+test_docker_installation">test_docker_installation</a></code>, <code><a href="babelwhale.html#topic+test_singularity_installation">test_singularity_installation</a></code></p>
</dd>
<dt>dynutils</dt><dd><p><code><a href="dynutils.html#topic+test_h5_installation">test_h5_installation</a></code></p>
</dd>
</dl>

<hr>
<h2 id='select_waypoints'>Add or create waypoints to a trajectory</h2><span id='topic+select_waypoints'></span><span id='topic+add_waypoints'></span><span id='topic+is_wrapper_with_waypoints'></span>

<h3>Description</h3>

<p>Waypoints are points along the trajectory, which do not necessarily correspond to cells. They are selected in such a way that all parts of the trajectory are covered
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_waypoints(
  trajectory,
  n_waypoints = 200,
  trafo = sqrt,
  resolution = sum(trafo(trajectory$milestone_network$length))/n_waypoints,
  recompute = FALSE
)

add_waypoints(
  trajectory,
  n_waypoints = 200,
  trafo = sqrt,
  resolution = sum(trafo(trajectory$milestone_network$length))/n_waypoints,
  recompute = FALSE
)

is_wrapper_with_waypoints(trajectory)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_waypoints_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="select_waypoints_+3A_n_waypoints">n_waypoints</code></td>
<td>
<p>The number of waypoints</p>
</td></tr>
<tr><td><code id="select_waypoints_+3A_trafo">trafo</code></td>
<td>
<p>Transformation function of the edge lengths</p>
</td></tr>
<tr><td><code id="select_waypoints_+3A_resolution">resolution</code></td>
<td>
<p>The resolution of the waypoints, measured in the same units as the lengths of the milestone network edges, will be automatically computed using n_waypoints</p>
</td></tr>
<tr><td><code id="select_waypoints_+3A_recompute">recompute</code></td>
<td>
<p>Force recompute</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>add_waypoints</code></strong> returns the trajectory with <em>waypoints</em> added, which is a list containing:
</p>

<ul>
<li> <p><em>milestone_percentages</em> and <em>progressions</em>: The milestone percentages and progressions of each waypoint, in the same format as the cell equivalents (see <code><a href="#topic+add_trajectory">add_trajectory()</a></code>) but with a <em>waypoint_id</em> column instead of a <em>cell_id</em> column
</p>
</li>
<li> <p><em>geodesic_distances</em>: a matrix with the geodesic distance of each waypoint (rows) to every cell (columns)
</p>
</li>
<li> <p><em>waypoint_network</em>: a dataframe containing the network between consecutive waypoints, it contains information on the connected waypoints (<em>from</em> and <em>to</em>) and the edge on which they reside (<em>from_milestone_id</em> and <em>to_milestone_id</em>)
</p>
</li>
<li> <p><em>waypoints</em>: the waypoint identifiers
</p>
</li></ul>

<p>**<code>select_waypoints</code> returns the list as mentioned in <code>add_waypoints</code>
</p>

<hr>
<h2 id='simplify_igraph_network'>Simplify an igraph network such that consecutive linear edges are removed</h2><span id='topic+simplify_igraph_network'></span>

<h3>Description</h3>


<ul>
<li><p> Nodes with degree 2 (or indegree 1 and outdegree 1) are removed: A -&gt; B -&gt; C becomes A -&gt; C
</p>
</li>
<li><p> Cycles contain at least 3 nodes, ie. A -&gt; B -&gt; A becomes A -&gt; B -&gt; C -&gt; A
</p>
</li>
<li><p> Loops are converted to a cycle, unless <code>allow_self_loops = TRUE</code>
</p>
</li>
<li><p> Duplicated edges are removed, unless <code>allow_duplcated_edges = FALSE</code>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>simplify_igraph_network(
  gr,
  allow_duplicated_edges = TRUE,
  allow_self_loops = TRUE,
  force_keep = NULL,
  edge_points = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simplify_igraph_network_+3A_gr">gr</code></td>
<td>
<p>An igraph object, see <code><a href="igraph.html#topic+make_graph">igraph::graph()</a></code></p>
</td></tr>
<tr><td><code id="simplify_igraph_network_+3A_allow_duplicated_edges">allow_duplicated_edges</code></td>
<td>
<p>Whether or not to allow duplicated edges between nodes.</p>
</td></tr>
<tr><td><code id="simplify_igraph_network_+3A_allow_self_loops">allow_self_loops</code></td>
<td>
<p>Whether or not to allow self loops.</p>
</td></tr>
<tr><td><code id="simplify_igraph_network_+3A_force_keep">force_keep</code></td>
<td>
<p>Nodes that will not be removed under any condition</p>
</td></tr>
<tr><td><code id="simplify_igraph_network_+3A_edge_points">edge_points</code></td>
<td>
<p>Points that are on edges</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An igraph object, or a list with an igraph object and a data frame with edge points
</p>


<h3>Examples</h3>

<pre><code class='language-R'>net &lt;- data.frame(
  from = 1:2,
  to = 2:3,
  length = 1,
  directed = TRUE,
  stringsAsFactors = F
)
gr &lt;- igraph::graph_from_data_frame(net)
simplify_igraph_network(gr)

net &lt;- data.frame(
  from = c(1, 2, 3, 1),
   to = c(2, 3, 1, 4),
    length = 1,
    directed = TRUE,
    stringsAsFactors = F
)
gr &lt;- igraph::graph_from_data_frame(net)
simplify_igraph_network(gr)

net &lt;- data.frame(
  from = c(1, 2, 3, 4),
   to = c(2, 3, 1, 5),
    length = 1,
    directed = TRUE,
    stringsAsFactors = F
)
gr &lt;- igraph::graph_from_data_frame(net)
simplify_igraph_network(gr)
</code></pre>

<hr>
<h2 id='simplify_trajectory'>Simplify a trajectory by removing transient milestones</h2><span id='topic+simplify_trajectory'></span>

<h3>Description</h3>


<ul>
<li><p> Milestones that are not a leaf or a branching point are removed: A -&gt; B -&gt; C becomes A -&gt; C
</p>
</li>
<li><p> Cycles contain at least 3 nodes, ie. A -&gt; B -&gt; A becomes A -&gt; B -&gt; C -&gt; A
</p>
</li>
<li><p> Loops are converted to a cycle, unless <code>allow_self_loops = TRUE</code>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>simplify_trajectory(trajectory, allow_self_loops = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simplify_trajectory_+3A_trajectory">trajectory</code></td>
<td>
<p>The trajectory as created by <code><a href="#topic+infer_trajectory">infer_trajectory()</a></code> or <code><a href="#topic+add_trajectory">add_trajectory()</a></code></p>
</td></tr>
<tr><td><code id="simplify_trajectory_+3A_allow_self_loops">allow_self_loops</code></td>
<td>
<p>Whether or not to allow self loops.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The positions of the cells within the trajectory remain the same.
</p>


<h3>Value</h3>

<p>A trajectory object
</p>

<hr>
<h2 id='trajectory_type_dag'>A DAG connecting different trajectory types</h2><span id='topic+trajectory_type_dag'></span>

<h3>Description</h3>

<p>A DAG connecting different trajectory types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trajectory_type_dag
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_graph</code> (inherits from <code>igraph</code>) of length 9.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory_type_dag

</code></pre>

<hr>
<h2 id='trajectory_types'>Metadata on the trajectory types</h2><span id='topic+trajectory_types'></span>

<h3>Description</h3>

<p>Metadata on the trajectory types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trajectory_types
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 9 rows and 6 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trajectory_types

</code></pre>

<hr>
<h2 id='wrap_data'>A data wrapper for datasets and trajectories</h2><span id='topic+wrap_data'></span><span id='topic+is_data_wrapper'></span>

<h3>Description</h3>

<p>A data wrapper for datasets and trajectories
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrap_data(
  id = NULL,
  cell_ids,
  cell_info = NULL,
  feature_ids = NULL,
  feature_info = NULL,
  ...
)

is_data_wrapper(dataset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wrap_data_+3A_id">id</code></td>
<td>
<p>A unique identifier for the data. If <code>NULL</code>, a random string will be generated.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_cell_ids">cell_ids</code></td>
<td>
<p>The identifiers of the cells.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_cell_info">cell_info</code></td>
<td>
<p>Optional meta-information pertaining the cells.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_feature_ids">feature_ids</code></td>
<td>
<p>The identifiers of the features.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_feature_info">feature_info</code></td>
<td>
<p>Optional meta-information pertaining the features.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_...">...</code></td>
<td>
<p>Extra information to be stored in the wrapper.</p>
</td></tr>
<tr><td><code id="wrap_data_+3A_dataset">dataset</code></td>
<td>
<p>A dataset created by <code><a href="#topic+wrap_data">wrap_data()</a></code> or <code><a href="#topic+wrap_expression">wrap_expression()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <em>id</em>, <em>cell_ids</em> and <em>cell_info</em> (if specified)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- wrap_data(
  cell_ids = c("A", "B", "C")
)
dataset$cell_ids
</code></pre>

<hr>
<h2 id='wrap_expression'>Create a wrapper object with expression and counts</h2><span id='topic+wrap_expression'></span>

<h3>Description</h3>

<p>Projected expression based on RNA velocity can also be added to the wrapper through the <code>expression_future</code> argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrap_expression(
  id = NULL,
  expression,
  counts,
  cell_info = NULL,
  feature_info = NULL,
  expression_future = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wrap_expression_+3A_id">id</code></td>
<td>
<p>A unique identifier for the data. If <code>NULL</code>, a random string will be generated.</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_expression">expression</code></td>
<td>
<p>The normalised expression values of genes (columns) within cells (rows). This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_counts">counts</code></td>
<td>
<p>The counts values of genes (columns) within cells (rows). This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_cell_info">cell_info</code></td>
<td>
<p>Optional meta-information pertaining the cells.</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_feature_info">feature_info</code></td>
<td>
<p>Optional meta-information of the features, a dataframe with at least <em>feature_id</em> as column</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_expression_future">expression_future</code></td>
<td>
<p>Projected expression using RNA velocity of genes (columns) within cells (rows).  This can be both a dense and sparse matrix.</p>
</td></tr>
<tr><td><code id="wrap_expression_+3A_...">...</code></td>
<td>
<p>extra information to be stored in the dataset</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Information about the cells and/or features can be added through <code>cell_info</code> and <code>feature_info</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dataset &lt;- wrap_expression(
  counts = example_dataset$counts,
  expression = example_dataset$expression,
  expression_future = example_dataset$expression_future
)

dataset$counts[1:10, 1:3]
dataset$expression[1:10, 1:3]
dataset$expression_future[1:10, 1:3]

</code></pre>

<hr>
<h2 id='wrapper_types'>Metadata on wrapper types</h2><span id='topic+wrapper_types'></span>

<h3>Description</h3>

<p>Metadata on wrapper types
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrapper_types
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 7 rows and 4 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>wrapper_types

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
