<!DOCTYPE html><html><head><title>Help for package Eagle</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Eagle}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Eagle-package'>
<p>Eagle for Genome-wide Association Mapping</p></a></li>
<li><a href='#AM'><p>multiple-locus Association Mapping</p></a></li>
<li><a href='#FPR4AM'><p>Set the false positive rate for <code>AM</code></p></a></li>
<li><a href='#OpenGUI'><p>Browser-based Graphical User Interface</p></a></li>
<li><a href='#PlotAM'><p>Visualisation of multiple locus association mapping results</p></a></li>
<li><a href='#ReadMap'><p>Read map file</p></a></li>
<li><a href='#ReadMarker'><p>Read  marker data.</p></a></li>
<li><a href='#ReadPheno'><p>Read phenotype file</p></a></li>
<li><a href='#ReadZmat'><p>Read Z matrix</p></a></li>
<li><a href='#SummaryAM'><p>Summary of multiple locus association mapping results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multiple Locus Association Mapping on a Genome-Wide Scale</td>
</tr>
<tr>
<td>Version:</td>
<td>2.5</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andrew George  &lt;geo047@gmail.com&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Andrew George [aut, cre],
  Joshua Bowden [ctb],
  Ryan Stephenson [ctb],
  Hyun Kang [ctb],
  Noah Zaitlen [ctb],
  Claire Wade [ctb],
  Andrew Kirby [ctb],
  David Heckerman [ctb],
  Mark Daly [ctb],
  Eleazar Eskin [ctb]</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of multiple-locus association mapping on a genome-wide scale. 'Eagle' can handle inbred and  outbred study populations, populations of arbitrary unknown complexity, and data larger than the memory capacity of the computer. Since 'Eagle' is based on linear mixed models, it is best suited to the analysis of data on continuous traits. However, it can tolerate non-normal data. 'Eagle' reports, as its findings, the best set of snp in strongest association with a trait. For users unfamiliar with R, to perform an analysis, run 'OpenGUI()'. This  opens a web browser to the menu-driven user interface for the input of data, and for performing genome-wide analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), shinyFiles, shinyBS, ggplot2, ggthemes, plotly</td>
</tr>
<tr>
<td>Imports:</td>
<td>R.utils, mmap, shiny, shinythemes, shinyjs, fontawesome,
stats, utils, parallel, data.table</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>RcppEigen, Rcpp</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>TRUE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://eagle.r-forge.r-project.org">http://eagle.r-forge.r-project.org</a></td>
</tr>
<tr>
<td>Contact:</td>
<td>geo047@gmail.com</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-28 09:20:06 UTC; andrew</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-30 05:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Eagle-package'>
Eagle for Genome-wide Association Mapping 
</h2><span id='topic+Eagle-package'></span><span id='topic+Eagle'></span>

<h3>Description</h3>

<p>An implementation of multiple-locus association mapping on a genome-wide scale. 'Eagle' can handle inbred and  outbred study populations, populations of arbitrary unknown complexity, and data larger than the memory capacity of the computer. Since 'Eagle' is based on linear mixed models, it is best suited to the analysis of data on continuous traits. However, it can tolerate non-normal data. 'Eagle' reports, as its findings, the best set of snp in strongest association with a trait. For users unfamiliar with R, to perform an analysis, run 'OpenGUI()'. This  opens a web browser to the menu-driven user interface for the input of data, and for performing genome-wide analysis.
</p>


<h3>Details</h3>

<p><em>Motivation:</em>  Data from genome-wide association studies are analyzed, commonly, with single-locus 
models. That is,  analyzes are performed on a locus-by-locus basis. Multiple-locus approaches 
that model the association between a trait and multiple loci simultaneously are more powerful. However, 
these methods do not scale well with study size and many of the packages that implement these methods are not easy 
to use. 
Eagle was specifically designed to make 
genome-wide association mapping with multiple-locus models simple and practical.  
</p>
<p><em>Assumptions</em>
</p>

<ol>
<li><p> Individuals are diploid but they can be inbred or outbred.
</p>
</li>
<li><p> The marker and phenotype data are in separate files.
</p>
</li>
<li><p>  Marker loci are snps. Dominant and multi-allelic loci will need to be converted into biallelic (snp-like) loci. 
</p>
</li>
<li><p> The trait is continuous and normally distributed. Eagle can handle non-normally distributed trait data but 
there may be a loss of power
to detect marker-trait associations. 
</p>
</li></ol>

<p><em>Important Functions:</em>
</p>

<ol>
<li> <p><code><a href="#topic+ReadMarker">ReadMarker</a></code> for reading in the snp data.
</p>
</li>
<li> <p><code><a href="#topic+ReadPheno">ReadPheno</a></code> for reading in the phenotypic data (traits and features/covariates)
</p>
</li>
<li> <p><code><a href="#topic+ReadMap">ReadMap</a></code> for reading in the marker map.
</p>
</li>
<li> <p><code><a href="#topic+FPR4AM">FPR4AM</a></code> for calculating the value of the <code>lambda</code> parameter to be used by <code><a href="#topic+AM">AM</a></code> that will give a desired false positive rate for detecting SNP-trait associations. 
</p>
</li>
<li> <p><code><a href="#topic+AM">AM</a></code> for performing association mapping on the data.
</p>
</li>
<li> <p><code><a href="#topic+OpenGUI">OpenGUI</a></code> which opens the GUI. 
</p>
</li></ol>

<p><em>Output:</em> 
The key output from <code><a href="#topic+AM">AM</a></code> is a list of snp. Each snp identifies a separate genomic region of interest, housing 
genes that are affecting the trait. Additional summary information such as the size of the snp effects, their statistical significance, and 
how much phenotypic variation they explain can be obtained by running <code><a href="#topic+SummaryAM">SummaryAM</a></code>.
</p>
<p><em>Where to get help: </em>
A variety of different help options are available. 
</p>

<ul>
<li><p> At the R prompt, type  </p>
<pre>library(, "Eagle")</pre><p> for an overview of the package and its functions. 
</p>
</li>
<li><p> For detailed help on a function called &quot;foo&quot; say, type  </p>
<pre>help("foo")</pre> 
</li>
<li><p> Visit the Eagle website at <a href="http://eagle.r-forge.r-project.org/">http://eagle.r-forge.r-project.org/</a> where 
you can find a quick start guide, instructions on getting the most out of Eagle, 
video tutorials, and other useful information. 
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Andrew W. George (Data61, CSIRO)  with a lot of support from Joshua Bowden (IM&amp;T, CSIRO)
</p>
<p>Maintainer: Andrew W. George  &lt;geo047@gmail.com&gt;
</p>

<hr>
<h2 id='AM'>multiple-locus Association Mapping</h2><span id='topic+AM'></span>

<h3>Description</h3>

<p><code>AM</code> performs  association mapping within a multiple-locus linear mixed model framework. 
<code>AM</code>  finds the best set of 
marker loci in strongest association with a trait while simultaneously accounting for any fixed effects and the genetic background.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AM(
  trait = NULL,
  fformula = NULL,
  geno = NULL,
  pheno = NULL,
  map = NULL,
  Zmat = NULL,
  ncpu = detectCores(),
  ngpu = 0,
  quiet = TRUE,
  maxit = 40,
  fixit = FALSE,
  lambda = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AM_+3A_trait">trait</code></td>
<td>
<p>the name of the column in the phenotype data file that contains the trait data. The name is case sensitive and must match exactly the column name in the phenotype data file.</p>
</td></tr>
<tr><td><code id="AM_+3A_fformula">fformula</code></td>
<td>
<p>the right hand side formula for the fixed effects.   See below for details. 
If
not specified, only an overall mean will be fitted.</p>
</td></tr>
<tr><td><code id="AM_+3A_geno">geno</code></td>
<td>
<p>the R  object obtained from running <code><a href="#topic+ReadMarker">ReadMarker</a></code>. This must be specified.</p>
</td></tr>
<tr><td><code id="AM_+3A_pheno">pheno</code></td>
<td>
<p>the R  object  obtained  from running <code><a href="#topic+ReadPheno">ReadPheno</a></code>. This must be specified.</p>
</td></tr>
<tr><td><code id="AM_+3A_map">map</code></td>
<td>
<p>the R object obtained from running <code><a href="#topic+ReadMap">ReadMap</a></code>. If not specified, a generic map will 
be assumed.</p>
</td></tr>
<tr><td><code id="AM_+3A_zmat">Zmat</code></td>
<td>
<p>the R object obtained from running <code><a href="#topic+ReadZmat">ReadZmat</a></code>. If not specified, an identity matrix will be assumed.</p>
</td></tr>
<tr><td><code id="AM_+3A_ncpu">ncpu</code></td>
<td>
<p>a integer  value for the number of CPU that are available for distributed computing.  The default is to determine the number of CPU automatically.</p>
</td></tr>
<tr><td><code id="AM_+3A_ngpu">ngpu</code></td>
<td>
<p>a integer value for the number of gpu available for computation.  The default
is to assume there are no gpu available.  This option has not yet been implemented.</p>
</td></tr>
<tr><td><code id="AM_+3A_quiet">quiet</code></td>
<td>
<p>a logical value. If set to <code>FALSE</code>, additional runtime output is printed. 
This is useful for error checking and monitoring the progress of a large analysis.</p>
</td></tr>
<tr><td><code id="AM_+3A_maxit">maxit</code></td>
<td>
<p>an integer value for the maximum number of forward steps to be performed.  This will rarely need adjusting.</p>
</td></tr>
<tr><td><code id="AM_+3A_fixit">fixit</code></td>
<td>
<p>a boolean value. If TRUE, then <code>maxit</code> iterations are performed, regardless of the value of the model fit value extBIC. If FALSE, 
then the model building process is stopped when extBIC increases in value.</p>
</td></tr>
<tr><td><code id="AM_+3A_lambda">lambda</code></td>
<td>
<p>a value between 0 and 1 for the regularization parameter for the extBIC. Values close to 0 lead to an anti-conservative test. Values close to 1 lead to a
more conservative test. If this value is left unspecified, a default value of 1 is assumed. See <code><a href="#topic+FPR4AM">FPR4AM</a></code> for an empirical approach for setting the  lambda value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to perform genome-wide association mapping. The phenotypic and SNP data should already be read in prior to running this function 
(see below for examples).  <code>AM</code> builds the linear mixed model iteratively, via forward selection. It is through this model building process that we 
identify the SNP-trait associations.  We use the extended BIC (extBIC) to decide on the 'best' model and when to stop looking for a better model. 
The conservativeness of extBIC can be adjusted.  If the <code>lambda</code> parameter is left at is default setting, then <code>AM</code> is run in its most 
conservative state (i.e. false positives are minimized but this also decreases the chance of true positives). 
</p>
<p>When interested in running  <code>AM</code>  at a certain false positive rate, use <code><a href="#topic+FPR4AM">FPR4AM</a></code>. This function uses permutation to 
find the lambda value for a desired false positive rate for <code>AM</code>. 
</p>
<p>Below are some examples of how to use <code>AM</code> for genome-wide association mapping of data. 
</p>


<h4>How to perform a basic AM analysis</h4>

<p>Suppose, 
</p>

<ul>
<li><p>the snp data are contained in the file geno.txt which is a plain space separated
text file with no column headings. The file is located in the current working directory. 
It contains numeric genotype values 0, 1, and 2 for snp genotypes
AA, AB, and BB, respectively. It also contains the  value X for a missing genotype. 
</p>
</li>
<li><p>the phenotype data is contained in the file pheno.txt which is a plain space
separated text file containing a single column with the trait data. The first row of the file 
has the column heading 'y'. This file does not contain any missing data.  
The file is located in the current working directory.
</p>
</li>
<li><p>there is no map data.
</p>
</li></ul>

<p>To analyse these data, we would use the following three functions (the parameters can be specified in any order, as well as the 
functions as long as AM is run last):
</p>
<pre>
  geno_obj &lt;-  ReadMarker(filename='geno.txt', AA=0, AB=1, BB=2, type="text", missing='X')
  
  pheno_obj &lt;- ReadPheno(filename='pheno.txt')

 # since lambda is not specified, this will run AM conservatively (where the false positive rate is lowest).
  res &lt;- AM(trait='y', geno=geno_obj, pheno=pheno_obj)
</pre>
<p>A table of results is printed to the screen and saved in the R object <code>res</code>. 
</p>



<h4>How to perform a more complicated AM analysis where the false positive rate is 5%</h4>

<p>Suppose, 
</p>

<ul>
<li><p>the snp data are contained in the file geno.ped which is a 'PLINK' ped file. See
<code><a href="#topic+ReadMarker">ReadMarker</a></code> for details. The file is located in /my/dir. Let's assume 
the file is large, say 50 gigabytes,   and our computer only has 32 gigabytes of RAM.
</p>
</li>
<li><p>the phenotype data is contained in the file pheno.txt which is a plain space
separated text file with  six columns. The first row of the file contains the column headings. 
The first column is a trait and is labeled y1.
The second column is another trait and is labeled y2. The third and fourth columns 
are nuisance variables and are labeled cov1 and cov2. The fifth and sixth columns
are the first two principal components to account for population substructure and are 
labeled pc1 and pc2. The file contains missing data that are coded as 99. 
The file is located in /my/dir.
</p>
</li>
<li><p>the map data is contained in the file map.txt, is also located in 
/my/dir, and the first row has the column headings.
</p>
</li>
<li><p>An 'AM' analysis is performed where the trait of interest is y2, and
the fixed effects part of the model is cov1 + cov2 + pc1 + pc2, 
</p>
</li></ul>
 
<p>To analyse these data, we would run the following:
</p>
<pre>
  geno_obj &lt;-  ReadMarker(filename='/my/dir/geno.ped', type='PLINK', availmemGb=32)
  
  pheno_obj &lt;- ReadPheno(filename='/my/dir/pheno.txt', missing=99)

  map_obj   &lt;- ReadMap(filename='/my/dir/map.txt')

  # FPR4AM calculates the lambda value corresponding to a desired false positive rate of 5%
  ans &lt;- FPR4AM(falseposrate=0.05, numreps=200, trait='y2', fformula=c('cov1 + cov2 + pc1 + pc2'), 
            geno=geno_obj, pheno=pheno_obj, map=map_obj)

  # performs association mapping with a 5% false positive rate
  res &lt;- AM(trait='y2', fformula=c('cov1 + cov2 + pc1 + pc2'), 
            geno=geno_obj, pheno=pheno_obj, map=map_obj,  lambda=ans$setlambda)
</pre>
<p>A table of results is printed to the screen and saved in the R object <code>res</code>. 
</p>



<h4>How to perform an analysis where individuals have multiple observations</h4>

<p>Suppose, 
</p>

<ul>
<li><p>the snp data are contained in the file geno.ped which is a 'PLINK' ped file. See
<code><a href="#topic+ReadMarker">ReadMarker</a></code> for details. The file is located in /my/dir. Let's assume 
the file is large, say 50 gigabytes,   and our computer only has 32 gigabytes of RAM.
</p>
</li>
<li><p>the phenotype data is contained in the file pheno.txt which is a plain space
separated text file with  six columns. The first row of the file contains the column headings. 
The first column is a trait and is labeled y1.
The second column is another trait and is labeled y2. The third and fourth columns 
are nuisance variables and are labeled cov1 and cov2. The fifth and sixth columns
are the first two principal components to account for population substructure and are 
labeled pc1 and pc2. The file contains missing data that are coded as 99. 
The file is located in /my/dir.
</p>
</li>
<li><p>the Z matrix data are contained in the file Zmatrix.txt. The file is located in /my/dir.This file is
a design matrix that only contains zeros and ones where each row must contain only a single one in the column that matches 
the individual's trait value to their corresponding genotype.
</p>
</li>
<li><p>the map data is contained in the file map.txt, is also located in 
/my/dir, and the first row has the column headings.
</p>
</li>
<li><p>An 'AM' analysis is performed where the trait of interest is y2,  and
the fixed effects part of the model is cov1 + cov2 + pc1 + pc2.
</p>
</li></ul>
 
<p>To analyse these data, we would run the following:
</p>
<pre>
  geno_obj &lt;-  ReadMarker(filename='/my/dir/geno.ped', type='PLINK', availmemGb=32)
  
  pheno_obj &lt;- ReadPheno(filename='/my/dir/pheno.txt', missing=99)

  map_obj   &lt;- ReadMap(filename='/my/dir/map.txt')

  Zmat_obj  &lt;- ReadZmat(filename='/my/dir/Zmatrix.txt')

  res &lt;- AM(trait='y2', fformula=c('cov1 + cov2 + pc1 + pc2'), 
            geno=geno_obj, pheno=pheno_obj, map=map_obj, Zmat=Zmat_obj )
</pre>
<p>A table of results is printed to the screen and saved in the R object <code>res</code>. 
</p>



<h4>Dealing with missing marker data</h4>

<p><code>AM</code> can tolerate some missing marker data. However, ideally, 
a specialized genotype imputation program such as  'BEAGLE'  or 'PHASE2', should be 
used to impute the missing marker data before being read into 'Eagle'.  
</p>



<h4>Dealing with missing trait data</h4>

<p><code>AM</code> deals automatically with individuals with missing trait data. 
These individuals are removed  from the analysis and a warning message is generated.
</p>



<h4>Dealing with missing explanatory variable values</h4>

<p><code>AM</code> deals automatically with individuals with missing explanatory variable values. 
These individuals are removed from the analysis and a warning message is generated
</p>



<h4>Error Checking</h4>

<p>Most errors occur when reading in the data. However, as an extra precaution, if <code>quiet=FALSE</code>, then additional 
output is printed during the running of <code>AM</code>. If <code>AM</code> is failing, then this output can be useful for diagnosing 
the problem. 
</p>



<h3>Value</h3>

<p>A list with the following components:
</p>

<dl>
<dt>trait:</dt><dd><p>column name of the trait being used by 'AM'.</p>
</dd>
<dt>fformula:</dt><dd><p>the fixed effects part of the linear mixed model.</p>
</dd>
<dt>indxNA_pheno:</dt><dd><p>a vector containing the row indexes of the phenotyic data  that have been removed from the analysis.</p>
</dd>
<dt>indxNA_geno:</dt><dd><p>a vector containing the row indexes of those genotypes that have been removed from the analysis 
due to missing data. </p>
</dd>
<dt>Mrk:</dt><dd><p>a vector with the names of the snp in strongest and significant association with the trait.If no loci are found to be 
significant, then this component is <code>NA</code>.</p>
</dd>
<dt>Chr:</dt><dd><p>the chromosomes on which the identified snp lie.</p>
</dd>
<dt>Pos:</dt><dd><p>the map positions for the identified snp.</p>
</dd>
<dt>Indx:</dt><dd><p>the column indexes in the marker file of the identified snp.</p>
</dd> 
<dt>ncpu:</dt><dd><p>number of cpu used for the calculations.</p>
</dd>
<dt>availmemGb:</dt><dd><p>amount of RAM in gigabytes that has been set by the user.</p>
</dd>
<dt>quiet:</dt><dd><p> boolean value of the parameter.</p>
</dd>
<dt>extBIC:</dt><dd><p>numeric vector with the extended BIC values for the loci  found to be in  significant association with the trait.</p>
</dd>
<dt>lambda</dt><dd><p>the numeric value of the parameter.</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+FPR4AM">FPR4AM</a></code> , <code><a href="#topic+ReadMarker">ReadMarker</a></code>, <code><a href="#topic+ReadPheno">ReadPheno</a></code>,  <code><a href="#topic+ReadZmat">ReadZmat</a></code>, and <code><a href="#topic+ReadMap">ReadMap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run:  
  # Since the following code takes longer than 5 seconds to run, it has been tagged as dontrun. 
  # However, the code can be run by the user. 
  #

  #-------------------------
  #  Example  
  #------------------------

  # read the map 
  #~~~~~~~~~~~~~~
  
  # File is a plain space separated text file with the first row 
  # the column headings
  complete.name &lt;- system.file('extdata', 'map.txt', 
                                   package='Eagle')
  map_obj &lt;- ReadMap(filename=complete.name) 

  # read marker data
  #~~~~~~~~~~~~~~~~~~~~
  # Reading in a PLINK ped file 
  # and setting the available memory on the machine for the reading of the data to 8  gigabytes
  complete.name &lt;- system.file('extdata', 'geno.ped', 
                                     package='Eagle')
  geno_obj &lt;- ReadMarker(filename=complete.name,  type='PLINK', availmemGb=8) 
 
  # read phenotype data
  #~~~~~~~~~~~~~~~~~~~~~~~

  # Read in a plain text file with data on a single trait and two covariates
  # The first row of the text file contains the column names y, cov1, and cov2. 
  complete.name &lt;- system.file('extdata', 'pheno.txt', package='Eagle')
  
  pheno_obj &lt;- ReadPheno(filename=complete.name)
           

 # Performing multiple-locus genome-wide association mapping with a model 
 #    with fixed effects cov1 and cov2 and an intercept. The intercept 
 #    need not be specified as it is assumed. 
 #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
  res &lt;- AM(trait = 'y',
                           fformula=c('cov1+cov2'),
                           map = map_obj,
                           pheno = pheno_obj,
                           geno = geno_obj )

## End(Not run)

</code></pre>

<hr>
<h2 id='FPR4AM'>Set the false positive rate for <code>AM</code></h2><span id='topic+FPR4AM'></span>

<h3>Description</h3>

<p>The lambda parameter in <code>AM</code> controls the false positive rate of the model 
building process. This function uses permutation to  find the lambda value for a desired false positive rate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FPR4AM(
  falseposrate = 0.05,
  trait = trait,
  numreps = 200,
  fformula = NULL,
  numlambdas = 50,
  geno = NULL,
  pheno = NULL,
  map = NULL,
  Zmat = NULL,
  ncpu = detectCores(),
  ngpu = 0,
  seed = 101
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FPR4AM_+3A_falseposrate">falseposrate</code></td>
<td>
<p>the desired false positive rate.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_trait">trait</code></td>
<td>
<p>the name of the column in the phenotype data file that contains the trait data. The name is case sensitive and must match exactly the column name in the phenotype data file.  This parameter must be specified.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_numreps">numreps</code></td>
<td>
<p>the number of replicates upon which to base the calculation of the false 
positive rate. We have found 200 replicates to be sufficient but more is better.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_fformula">fformula</code></td>
<td>
<p>the right hand side formula for the fixed effects part of the model.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_numlambdas">numlambdas</code></td>
<td>
<p>the number of equidistant lambda values from 0 to 1 for which to calculate the false positive rate of the model building process. This should not need 
adjusting.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_geno">geno</code></td>
<td>
<p>the R  object obtained from running <code><a href="#topic+ReadMarker">ReadMarker</a></code>. This must be specified.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_pheno">pheno</code></td>
<td>
<p>the R  object  obtained  from running <code><a href="#topic+ReadPheno">ReadPheno</a></code>. This must be specified.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_map">map</code></td>
<td>
<p>the R object obtained from running <code><a href="#topic+ReadMap">ReadMap</a></code>. If not specified, a generic map will 
be assumed.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_zmat">Zmat</code></td>
<td>
<p>the R object obtained from running <code><a href="#topic+ReadZmat">ReadZmat</a></code>. If not specified, an identity matrix will be assumed.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_ncpu">ncpu</code></td>
<td>
<p>a integer  value for the number of CPU that are available for distributed computing.  The default is to determine the number of CPU automatically.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_ngpu">ngpu</code></td>
<td>
<p>a integer value for the number of gpu available for computation.  The default
is to assume there are no gpu available.  
This option has not yet been implemented.</p>
</td></tr>
<tr><td><code id="FPR4AM_+3A_seed">seed</code></td>
<td>
<p>a integer value for the starting seed for the permutations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The false positive rate for <code><a href="#topic+AM">AM</a></code> is controlled by its lambda parameter. Values close to 
1 (0) decreases (increases) the false positive rate of detecting SNP-trait associations. There is no 
analytical way of setting lambda for a specified false positive rate. So we are using permutation to do this empirically. 
</p>
<p>By setting <code>falseposrate</code> to the desired false positive rate, this function will find the corresponding lambda value for <code><a href="#topic+AM">AM</a></code>. 
</p>
<p>A table of other lambda values for a range of false positive rates is also given. 
</p>
<p>To increase the precision of the lambda estimates, increase <code>numreps</code>.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>

<dl>
<dt>numreps:</dt><dd><p>the number of permutations performed.</p>
</dd>
<dt>lambda:</dt><dd><p>the vector of lambda values.</p>
</dd>
<dt>falsepos:</dt><dd><p>the false positive rates for the lambda values.</p>
</dd>
<dt>setlambda:</dt><dd><p>the lambda value that gives a false positive rate of <code>falseposrate</code> </p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+AM">AM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run:  
  # Since the following code takes longer than 5 seconds to run, it has been tagged as dontrun. 
  # However, the code can be run by the user. 
  #

  #-------------------------
  #  Example  
  #------------------------

  # read the map 
  #~~~~~~~~~~~~~~
  
  # File is a plain space separated text file with the first row 
  # the column headings
  complete.name &lt;- system.file('extdata', 'map.txt', 
                                   package='Lion')
  map_obj &lt;- ReadMap(filename=complete.name) 

  # read marker data
  #~~~~~~~~~~~~~~~~~~~~
  # Reading in a PLINK ped file 
  # and setting the available memory on the machine for the reading of the data to 8  gigabytes
  complete.name &lt;- system.file('extdata', 'geno.ped', 
                                     package='Lion')
  geno_obj &lt;- ReadMarker(filename=complete.name,  type='PLINK', availmemGb=8) 
 
  # read phenotype data
  #~~~~~~~~~~~~~~~~~~~~~~~

  # Read in a plain text file with data on a single trait and two covariates
  # The first row of the text file contains the column names y, cov1, and cov2. 
  complete.name &lt;- system.file('extdata', 'pheno.txt', package='Lion')
  
  pheno_obj &lt;- ReadPheno(filename=complete.name)
           

 #  Suppose we want to perform the AM analysis at a 5% false positive rate. 
 #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
  ans &lt;- FPR4AM(falseposrate = 0.05,
                trait = 'y',
                fformula=c('cov1+cov2'),
                map = map_obj,
                pheno = pheno_obj,
                geno = geno_obj) 
 

  res &lt;- AM(trait =  'y',
                fformula=c('cov1+cov2'),
                map = map_obj,
                pheno = pheno_obj,
                geno = geno_obj,
                lambda = ans$setlambda)



## End(Not run)


</code></pre>

<hr>
<h2 id='OpenGUI'>Browser-based Graphical User Interface</h2><span id='topic+OpenGUI'></span>

<h3>Description</h3>

<p>Opens a web browser to act as a user-friendly interface to 'Eagle'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OpenGUI()
</code></pre>


<h3>Details</h3>

<p>Once <code>OPenGUI</code> is run, your default web-browser will open automatically the Eagle GUI. We have written our GUI in html code. 
The GUI allows users to input data files, analyse data, summarise findings, and view results via interactive plots. We 
have designed the GUI for users whom may be unfamilar with R and wishing to avoid having to write R code.  
</p>
<p>Note, that even though a web browser is being used as the user interface, everything remains local to the computer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# opens a web browser 
OpenGUI()

## End(Not run)

</code></pre>

<hr>
<h2 id='PlotAM'>Visualisation of multiple locus association mapping results</h2><span id='topic+PlotAM'></span>

<h3>Description</h3>

<p>A interactive plotting function that provides additional information on the significant 
marker-trait associations found by <code><a href="#topic+AM">AM</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotAM(
  AMobj = NULL,
  itnum = 1,
  chr = "All",
  type = "Manhattan",
  interactive = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotAM_+3A_amobj">AMobj</code></td>
<td>
<p>the (list) object obtained from running <code><a href="#topic+AM">AM</a></code>.</p>
</td></tr>
<tr><td><code id="PlotAM_+3A_itnum">itnum</code></td>
<td>
<p>the iteration number of the model building process whose results are to be viewed.</p>
</td></tr>
<tr><td><code id="PlotAM_+3A_chr">chr</code></td>
<td>
<p>either &quot;All&quot; for all chromosomes or the label of a specific chromosome.</p>
</td></tr>
<tr><td><code id="PlotAM_+3A_type">type</code></td>
<td>
<p>either &quot;Manhattan&quot; or  &quot;Score&quot;  for a manhattan plot or a plot of the 
score statistics across SNPs.</p>
</td></tr>
<tr><td><code id="PlotAM_+3A_interactive">interactive</code></td>
<td>
<p>boolean parameter. When <code>TRUE</code>, an interactive plot is generated. When 
<code>FALSE</code>, a ggplot object is returned which can be saved as an image to file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function useful for viewing the strength of association across the whole genome and 
how this association changes as the model is built. 
</p>
<p>The  score statistics (<code>type="Score"</code>) or p-values  of the score statistics
(<code>type="Manhattan"</code>) are plotted against the location of the SNPs.  The orange
vertical  lines 
denote the location of the SNPs already found by <code><a href="#topic+AM">AM</a></code>. 
The red vertical line is the location of the SNP in strongest association with the trait 
at that iteration number. 
</p>
<p>The vertical lines are numbered according to the order in which the snp-trait associations were found by the model. 
</p>
<p>A single chromosome or all (<code>chr="All"</code>) chromosomes can be viewed.  
</p>
<p>By setting <code>itnum</code> to different values, how the score statistics or p-values 
increase/decrease 
over the model building process can be observed.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AM">AM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
  # Since the following code takes longer than 5 seconds to run, it has been tagged as dontrun. 
  # However, the code can be run by the user. 
  #

  #---------------
  # read the map 
  #---------------
  #
  # File is a plain space separated text file with the first row 
  # the column headings
  complete.name &lt;- system.file('extdata', 'map.txt', 
                                   package='Eagle')
  map_obj &lt;- ReadMap(filename=complete.name) 

 # to look at the first few rows of the map file
 head(map_obj)

  # read marker data
  complete.name &lt;- system.file('extdata', 'geno.ped', 
                                     package='Eagle')
  geno_obj &lt;- ReadMarker(filename=complete.name,  type='PLINK', availmemGb=8) 
 
  # read phenotype data
  complete.name &lt;- system.file('extdata', 'pheno.txt', package='Eagle')
  
  pheno_obj &lt;- ReadPheno(filename=complete.name)
           
  # Perform multiple-locus genome-wide association mapping 
  res &lt;- AM(trait = 'y',
                           fformula=c("cov1 + cov2"),
                           map = map_obj,
                           pheno = pheno_obj,
                           geno = geno_obj)

 # Plotting the p-values from the first iteration of the module building process. 
 # You can see why Eagle has identified the SNP that is has. 
  PlotAM(AMobj=res, itnum=1)


 # Plotting the results from the final step of the model building process
 # By accounting for the effect of SNP in strong association with the trait, the 
 # strength of association changes across the genome. 
  PlotAM(AMobj=res, itnum=3)


 # Suppose you want to save the above plot to a jpeg file called myplot.jpg
 jpeg("./myplot.jpg", width=1200, height=800)
 PlotAM(AMobj=res, itnum=3, interactive=FALSE)
 dev.off()


 
## End(Not run)



</code></pre>

<hr>
<h2 id='ReadMap'>Read map file</h2><span id='topic+ReadMap'></span>

<h3>Description</h3>

<p>Read in the marker map  data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadMap(filename = NULL, header = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadMap_+3A_filename">filename</code></td>
<td>
<p>contains the name of the map file. The file name needs to be in quotes. If the file is not in the working directory, then the full path 
to the file is required.</p>
</td></tr>
<tr><td><code id="ReadMap_+3A_header">header</code></td>
<td>
<p>a logical value. When <code>TRUE</code>, the first row of the file contains the column headings.</p>
</td></tr>
<tr><td><code id="ReadMap_+3A_...">...</code></td>
<td>
<p>arguments to be passed to read.table such as <code>skip</code>, <code>sep</code>. See <code><a href="utils.html#topic+read.table">read.table</a></code> so the list 
of arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Association mapping, unlike classical linkage mapping, 
does not require a map to find marker-trait associations. 
So, reading in a map file is optional. 
If a map file is supplied, then the marker names from this file are used when reporting the findings from <code><a href="#topic+AM">AM</a></code>. 
If a map file is not supplied, then generic names M1, M2, ..., are assigned to the 
marker loci where the number refers to the  column number in the marker file. 
</p>
<p>A space separated text file with column headings is assumed as the default input. The map file can have three or four 
columns. If the map file has three columns, then it is assumed that the three columns are the marker locus names, 
the chromosome number, and the map position (in any units). If the map file has four columns as with a 'PLINK map file, 
then the columns are assumed to be the marker locus names, the chromosome number, the map position in centimorgans, 
and the map position in base pairs. 
</p>
<p>Missing values are allowed but not in the first column of the file (i.e. the marker labels are not allowed to be missing). 
</p>
<p>The order of the marker loci in this file is assumed to be  the same order as the loci in the marker data file.  
</p>
<p>The first column of the map file is assumed to contain the marker names.
</p>


<h3>Value</h3>

<p>a data frame is returned of the map data.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReadMarker">ReadMarker</a></code> and <code><a href="#topic+ReadPheno">ReadPheno</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read in  example map data from ./extdata/

# find the full location of the map data 
complete.name &lt;- system.file('extdata', 'map.txt', package='Eagle')
  
# read in map data 
map_obj &lt;- ReadMap(filename=complete.name) 
                               
# look at first few rows of the map file
head(map_obj)


</code></pre>

<hr>
<h2 id='ReadMarker'>Read  marker data.</h2><span id='topic+ReadMarker'></span>

<h3>Description</h3>

<p>A function for reading in different types of snp marker data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadMarker(
  filename = NULL,
  type = "text",
  missing = NULL,
  AA = NULL,
  AB = NULL,
  BB = NULL,
  availmemGb = 16,
  quiet = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadMarker_+3A_filename">filename</code></td>
<td>
<p>contains the name of the marker  file. The file name needs to be in quotes. If the file is not in the working directory, then the full path 
to the file is required.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_type">type</code></td>
<td>
<p>specify the type of file. Choices are 'text' (the default) , PLINK, and vcf.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_missing">missing</code></td>
<td>
<p>the number or character for a missing genotype in the text file. There is no need to specify this for a vcf or PLINK ped file. Missing 
allele values in a vcf file are coded as &quot;.&quot; and missing 
allele values in a PLINK file must be coded as '0' or '-'.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_aa">AA</code></td>
<td>
<p>the character or number corresponding to the 'AA' snp genotype in the marker genotype file. 
This need only be specified if the file type is 'text'.  If a character then it must be in quotes.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_ab">AB</code></td>
<td>
<p>the  character or number  corresponding to the 'AB' snp genotype in the marker genotype file. 
This need only be specified if the file type is 'text'.
This can be left unspecified 
if there are no heterozygous genotypes (i.e. the individuals are inbred). Only a single 
heterozygous genotype is allowed ('Eagle' does not distinguish between 'AB' and 'BA').  
If specified and a character, it must be in quotes.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_bb">BB</code></td>
<td>
<p>the character or number corresponding to the 'BB' snp genotype in the marker genotype file. 
This need only be  specified if the file type is 'text'.  If a character, then it must be in quotes.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_availmemgb">availmemGb</code></td>
<td>
<p>a numeric value. It specifies the amount of available memory (in Gigabytes). 
This should be set to be as large as possible for best performance.</p>
</td></tr>
<tr><td><code id="ReadMarker_+3A_quiet">quiet</code></td>
<td>
<p>a logical value. If set to <code>TRUE</code>, additional runtime output is printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ReadMarker</code> can handle three different types of marker data; namely,
genotype data in a plain text file, PLINK ped files, and vcf files.  
</p>


<h4><strong>Reading in a plain text file containing the marker genotypes</strong></h4>

<p>To load a text file that contains snp genotypes, run <code>ReadMarker</code> with <code>filename</code> set to the name of the file, 
and <code>AA</code>, <code>AB</code>, <code>BB</code> set to the corresponding genotype values. 
The genotype values in the text file can be numeric, character, or  a mix of both.
</p>
<p>We make the following assumptions
</p>

<ul>
<li><p>The text file does not contain row or column headings
</p>
</li>
<li><p>The file is allowed to contain missing genotypes that have been coded according to <code>missing</code>
</p>
</li>
<li><p>Individuals are diploid
</p>
</li>
<li><p>The rows of the text file are the individuals and the columns are the marker loci
</p>
</li>
<li><p>The file is space separated
</p>
</li>
<li><p>The mapping of the observed genotypes in the marker file to <code>AA</code>, <code>AB</code>, and <code>BB</code>, remains the same for all loci
</p>
</li>
<li><p>Individuals are outbred when <code>AA</code>, <code>AB</code>, and <code>BB</code> are specified and 
inbred when only <code>AA</code>, and <code>BB</code> are specified
</p>
</li>
<li><p>For a text file, the same alphanumeric value is used for all missing marker genotypes. For a PLINK ped file, the missing allele is allowed to 
be '0' or '-'. 
</p>
</li></ul>

<p>For example, suppose we have a space separated text file with marker genotype data collected from five snp loci on three individuals
where the snp genotype AA has been coded 0, the snp genotype AB has been coded 1, the snp genotype BB has been coded 2, and missing genotypes are coded 
as 99
</p>

<table>
<tr>
 <td style="text-align: center;">
 0 </td><td style="text-align: center;">  1  </td><td style="text-align: center;">  2 </td><td style="text-align: center;">  0</td><td style="text-align: center;">   2 </td>
</tr>
<tr>
 <td style="text-align: center;">
 1 </td><td style="text-align: center;">  1  </td><td style="text-align: center;">  0 </td><td style="text-align: center;">  2 </td><td style="text-align: center;">  0 </td>
</tr>
<tr>
 <td style="text-align: center;">
 2 </td><td style="text-align: center;">  2  </td><td style="text-align: center;">  1 </td><td style="text-align: center;">  1 </td><td style="text-align: center;">  99
</td>
</tr>

</table>

<p>The file is called geno.txt and is located in the directory /my/dir/. 
</p>
<p>To load these data, we would use the command
</p>
<pre>geno_obj &lt;- ReadMarker(filename='/my/dir/geno.txt', AA=0, AB=1, BB=2, type='text', missing=99)</pre>
<p>where the results from running the function are placed in <code>geno_obj</code>.
</p>
<p>As another example, suppose we have a space separated text file with marker genotype data collected from five snp loci on three individuals 
where the snp genotype AA has been coded a/a, the snp genotype AB has been coded a/b, and the snp genotype BB has been coded b/b
</p>

<table>
<tr>
 <td style="text-align: center;">
 a/a a/b b/b a/a b/b </td>
</tr>
<tr>
 <td style="text-align: center;">
 a/b a/b a/a b/b a/a </td>
</tr>
<tr>
 <td style="text-align: center;">
 b/b b/b a/b a/b NA 
</td>
</tr>

</table>

<p>The file is called geno.txt and is located in the same directory from which R is being run (i.e. the working directory). 
</p>
<p>To load these data, we would use the command 
</p>
<pre>geno_obj &lt;- ReadMarker(filename='geno.txt', AA='a/a', AB='a/b', BB='b/b', 
                                       type='text', missing = 'NA')</pre>
<p>where the results from running the function are placed in <code>geno_obj</code>.
</p>



<h4><strong>Reading in a PLINK ped file</strong></h4>

<p>PLINK is a well known toolkit for the analysis of genome-wide association data. See  
<a href="https://www.cog-genomics.org/plink2">https://www.cog-genomics.org/plink2</a>
for details. 
</p>
<p>Full details of PLINK ped files can be found <a href="https://www.cog-genomics.org/plink/1.9/formats#ped">https://www.cog-genomics.org/plink/1.9/formats#ped</a>. Briefly, 
the PED file is a space delimited file (tabs are not allowed): the first six columns are mandatory:
</p>

<table>
<tr>
 <td style="text-align: left;">
    Family ID  </td>
</tr>
<tr>
 <td style="text-align: left;">
    Individual ID </td>
</tr>
<tr>
 <td style="text-align: left;">
    Paternal ID </td>
</tr>
<tr>
 <td style="text-align: left;">
    Maternal ID </td>
</tr>
<tr>
 <td style="text-align: left;">
    Sex (1=male; 2=female; other=unknown) </td>
</tr>
<tr>
 <td style="text-align: left;">
    Phenotype
</td>
</tr>

</table>

<p>Here, these columns can be any values since <code>ReadMarker</code> ignores these columns.  
</p>
<p>Genotypes (column 7 onwards) can be any character 
(e.g. 1,2,3,4 or A,C,G,T or anything else) except 0 which is, by default, the missing genotype character. All markers should be biallelic. 
All snps must have two alleles specified.  Missing alleles (i.e 0 or -) are allowed.
No column headings should be given. 
</p>
<p>As an example, suppose we have data on three individuals  genotyped for four snp loci 
</p>

<table>
<tr>
 <td style="text-align: center;">
    FAM001 </td><td style="text-align: center;"> 101  </td><td style="text-align: center;">  0    </td><td style="text-align: center;"> 0   </td><td style="text-align: center;">   1  </td><td style="text-align: center;"> 0 </td><td style="text-align: center;">  A  </td><td style="text-align: center;">  G  </td><td style="text-align: center;"> C </td><td style="text-align: center;"> C </td><td style="text-align: center;"> C </td><td style="text-align: center;"> G </td><td style="text-align: center;"> A </td><td style="text-align: center;"> A </td>
</tr>
<tr>
 <td style="text-align: center;">
    FAM001 </td><td style="text-align: center;"> 201  </td><td style="text-align: center;">  0    </td><td style="text-align: center;"> 0   </td><td style="text-align: center;">   2  </td><td style="text-align: center;"> 0 </td><td style="text-align: center;">  A  </td><td style="text-align: center;">  A  </td><td style="text-align: center;"> C </td><td style="text-align: center;"> T </td><td style="text-align: center;"> G </td><td style="text-align: center;"> G </td><td style="text-align: center;"> T </td><td style="text-align: center;"> A </td>
</tr>
<tr>
 <td style="text-align: center;"> 
    FAM001 </td><td style="text-align: center;"> 300  </td><td style="text-align: center;">  101  </td><td style="text-align: center;"> 201 </td><td style="text-align: center;">   2  </td><td style="text-align: center;"> 0 </td><td style="text-align: center;">  G  </td><td style="text-align: center;">  A  </td><td style="text-align: center;"> T </td><td style="text-align: center;"> T </td><td style="text-align: center;"> C </td><td style="text-align: center;"> G </td><td style="text-align: center;"> A </td><td style="text-align: center;"> T 
</td>
</tr>

</table>

<p>Then to load these data, we would use the command 
</p>
<pre>geno_obj &lt;- ReadMarker(filename='PLINK.ped', type='PLINK')</pre>
<p>where <code>geno_obj</code> is used by <code><a href="#topic+AM">AM</a></code>, and the file PLINK.ped is located in the working directory (i.e. the directory from which R 
is being run). 
</p>



<h4><strong>Reading in a vcf file</strong></h4>

<p>VCF is a tab separated text file containing  meta-information lines, a header line, and data 
lines. The data lines  contain information about a position in the genome.
</p>
<p>It is assumed that genotype information has been recorded on samples for each position. 
</p>
<p>Loci with more than two alleles will be removed automatically. 
</p>
<p>Eagle will only accept a single (uncompressed) vcf file. If chromosomal information has been recorded in separate 
vcf files, these files need to be merged into a single vcf file.  This can be done by using the 
BCFtools utility set with command line &quot;bcftools concat&quot;.
</p>



<h3>Value</h3>

<p>To allow Eagle to handle data larger than the memory capacity of a machine, <code>ReadMarker</code> doesn't load 
the marker data into memory. Instead, it 
writes a reformatted version of the marker data, and its transpose, to the harddrive. These two files
are only temporary, being removed at the end of the R session. 
The object returned by
<code>ReadMarker</code> is a list object with the elements <code>tmpM</code> , <code>tmpMt</code>, and <code>dim_of_M</code>  
which is the full file name (name and path)  of the reformatted file for the marker  data,  the full file name of the reformatted file 
for the transpose of the marker  data,  and a 2 element vector with the first element the number of individuals and the second 
element the number of marker loci.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  #--------------------------------
  #  Example 1
  #-------------------------------
  #
  # Read in the genotype data contained in the text file geno.txt
  #
  # The function system.file() gives the full file name (name + full path).
  complete.name &lt;- system.file('extdata', 'geno.txt', package='Eagle')
  # 
  # The full path and name of the file is
  print(complete.name)
  
  # Here, 0 values are being treated as genotype AA,
  # 1 values are being treated as genotype AB, 
  # and 2 values are being treated as genotype BB. 
  # 4 gigabytes of memory has been specified. 
  # The file is space separated with the rows the individuals
  # and the columns the snp loci.
  geno_obj &lt;- ReadMarker(filename=complete.name, type='text', AA=0, AB=1, BB=2, availmemGb=4) 
   
  # view list contents of geno_obj
  print(geno_obj)

  #--------------------------------
  #  Example 2
  #-------------------------------
  #
  # Read in the allelic data contained in the PLINK ped file geno.ped
  #
  # The function system.file() gives the full file name (name + full path).
  complete.name &lt;- system.file('extdata', 'geno.ped', package='Eagle')

  # 
  # The full path and name of the file is
  print(complete.name)
  
  # Here,  the first 6 columns are being ignored and the allelic 
  # information in columns 7 -  10002 is being converted into a reformatted file. 
  # 4 gigabytes of memory has been specified. 
  # The file is space separated with the rows the individuals
  # and the columns the snp loci.
  geno_obj &lt;- ReadMarker(filename=complete.name, type='PLINK', availmemGb=4) 
   
  # view list contents of geno_obj
  print(geno_obj)



  #--------------------------------
  #  Example 3
  #-------------------------------
  #
  #
  # Read in the genotype data contained in the vcf file geno.vcf
  #
  # The function system.file() gives the full file name (name + full path).
  complete.name &lt;- system.file('extdata', 'geno.vcf', package='Eagle')
  # 
  # The full path and name of the file is
  print(complete.name)
  
  # The file contains 5 marker loci recorded on 3 individuals
  # Two of the loci contain multiple alleles and are removed. 
  # A summary of the file is printed once the file has been read.
  geno_obj &lt;- ReadMarker(filename=complete.name, type="vcf", availmemGb=4) 
   
  # view list contents of geno_obj
  print(geno_obj)

</code></pre>

<hr>
<h2 id='ReadPheno'>Read phenotype file</h2><span id='topic+ReadPheno'></span>

<h3>Description</h3>

<p>Read in the phenotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadPheno(filename = NULL, header = TRUE, csv = FALSE, missing = "NA", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadPheno_+3A_filename">filename</code></td>
<td>
<p>contains the name of the phenotype  file. The file name needs to be in quotes. If the file is not in the working directory, then the full path 
to the file is required.</p>
</td></tr>
<tr><td><code id="ReadPheno_+3A_header">header</code></td>
<td>
<p>a logical value. When <code>TRUE</code>, the first row of the file contains the names of the columns.  Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="ReadPheno_+3A_csv">csv</code></td>
<td>
<p>a logical value. When <code>TRUE</code>, a csv file format is assumed. When <code>FALSE</code>, a space separated format is assumed. Default
is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="ReadPheno_+3A_missing">missing</code></td>
<td>
<p>the number or character for a missing phenotype value.</p>
</td></tr>
<tr><td><code id="ReadPheno_+3A_...">...</code></td>
<td>
<p>arguments to be passed to read.table such as <code>skip</code>, <code>sep</code>. See <code><a href="utils.html#topic+read.table">read.table</a></code> so the list 
of arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ReadPheno</code> reads in the phenotype data 
which are data measured on traits and any fixed effects (or predictors/features/explanatory variables). 
A space separated plain text file is assumed. Each row in this file 
corresponds to an individual. The number of rows in the phenotype file must be the same as the number of rows in 
the marker data file. Also, the ordering of the individuals must be the same in the two files.  A space separated file with 
column headings is the default but can be changed with the <code>header</code> and <code>csv</code> options. 
</p>
<p>The phenotype file may contain multiple traits and fixed effects variables. 
</p>
<p>Missing values are allowed. Eagle is told which value should be treated as missing by setting the <code>missing</code> 
parameter to the value. 
</p>
<p>For example, suppose we have three individuals for which we have collected data on two quantitative traits (y1 and y2), and 
four explanatory variables (age, weight, height, and sex). The data looks like  
</p>

<table>
<tr>
 <td style="text-align: center;">
    y1      </td><td style="text-align: center;"> y2      </td><td style="text-align: center;">  age  </td><td style="text-align: center;"> weight </td><td style="text-align: center;">   height  </td><td style="text-align: center;"> sex </td>
</tr>
<tr>
 <td style="text-align: center;">
    112.02  </td><td style="text-align: center;"> -3.123  </td><td style="text-align: center;">  26    </td><td style="text-align: center;"> 75   </td><td style="text-align: center;">   168.5     </td><td style="text-align: center;"> M </td>
</tr>
<tr>
 <td style="text-align: center;">
    156.44  </td><td style="text-align: center;"> 1.2     </td><td style="text-align: center;">  45    </td><td style="text-align: center;"> 102   </td><td style="text-align: center;">   NA     </td><td style="text-align: center;"> NA  </td>
</tr>
<tr>
 <td style="text-align: center;">
    10.3    </td><td style="text-align: center;"> NA   </td><td style="text-align: center;">  28     </td><td style="text-align: center;"> 98   </td><td style="text-align: center;">   189.4     </td><td style="text-align: center;"> F 
</td>
</tr>

</table>

<p>where the first row has the column headings and the next three rows contain the observed data on three 
individuals. 
</p>
<p>To load these data, we would use the command 
</p>
<pre>pheno_obj &lt;- ReadPheno(filename='pheno.dat', missing='NA')</pre>
<p>where pheno.dat is the name of the phenotype file, and <code>pheno_obj</code> is the R object that contains the 
results from reading in the phenotype data.    The file is located in the working directory so there is no need to specify the full path, just the file name is suffice. 
</p>


<h4>Dealing with missing trait data</h4>

<p><code>AM</code> deals automatically with individuals with missing trait data. 
These individuals are removed  from the analysis and a warning message is generated.
</p>



<h4>Dealing with missing fixed effects values</h4>

<p><code>AM</code> deals automatically with individuals with missing fixed effects values. 
These individuals are removed from the analysis and a warning message is generated
</p>



<h3>Value</h3>

<p>a data frame is returned of the phenotype data. If <code>header</code> is true, the 
names of the columns will be as specified by the first row of the phenotype file. If <code>header</code> is <code>FALSE</code>, 
generic names are supplied by R in the form of V1, V2, etc.  If no column headings are given, these 
generic names will need to be used in the <code>trait</code> and <code>fformula</code> parameters in 
<code><a href="#topic+AM">AM</a></code>.  You can print out the column names of the data frame by using
</p>
<pre>names(pheno_obj)</pre>
<p>The column names are also printed along with other summary information when <code>ReadPheno</code> is run.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReadMarker">ReadMarker</a></code> for reading in marker data, <code><a href="#topic+AM">AM</a></code> for performing association mapping.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read in  phenotype data from ./extdata/

# find the full location of the phenotype data 
complete.name &lt;- system.file('extdata', 'pheno.txt', package='Eagle')

pheno_obj &lt;- ReadPheno(filename=complete.name)
  
 ## print a couple of lines of the data file
 head(pheno_obj)

</code></pre>

<hr>
<h2 id='ReadZmat'>Read Z matrix</h2><span id='topic+ReadZmat'></span>

<h3>Description</h3>

<p>Read in the Z matrix that assigns groups/strains/lines to their trait measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadZmat(filename = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadZmat_+3A_filename">filename</code></td>
<td>
<p>contains the name of the Z matrix file. The file name needs to be in quotes. If the file is not in the working directory, then the full path 
to the file is required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The underlying linear mixed model is of the form
</p>
<p style="text-align: center;"><code class="reqn">Y = X \beta  + Z u_g + e</code>
</p>

<p>where Z is a (n x <code class="reqn">n_g</code>) matrix that contains ones and zeros, n is the number of trait measurements, and <code class="reqn">n_g</code> 
is the number of groups/strains/lines. If n and <code class="reqn">n_g</code> are the same, then there is no need to specify Z.   
However, if a group/strain/line has multiple trait measurements (i.e. <code class="reqn">n &gt; n_g</code>) then the Z matrix is needed to tell Eagle which 
trait measurements belong to which groups/strains/lines. 
</p>
<p>A space separated text file is assumed.  Each row of the matrix contains multiple zeroes but only a  single one. 
The file cannot contain column or row headings.  The file also cannot contain a row of only zeroes. Here,  
n must be larger than <code class="reqn">n_g</code> otherwise an error will be issued.
</p>


<h3>Value</h3>

<p>a data matrix is returned of the Z matrix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReadMarker">ReadMarker</a></code> and <code><a href="#topic+ReadPheno">ReadPheno</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read in  example Z matrix from ./extdata/

# find the full location of the Z matrix data 
complete.name &lt;- system.file('extdata', 'Z.txt', package='Eagle')
  
# read in Z matrix data 
Z_obj &lt;- ReadZmat(filename=complete.name) 
                               
# look at first few rows of the Z matrix file
head(Z_obj)


</code></pre>

<hr>
<h2 id='SummaryAM'>Summary of multiple locus association mapping results</h2><span id='topic+SummaryAM'></span>

<h3>Description</h3>

<p>A summary function that provides additional information on the significant 
marker-trait associations found by <code><a href="#topic+AM">AM</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SummaryAM(AMobj = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SummaryAM_+3A_amobj">AMobj</code></td>
<td>
<p>the (list) object obtained from running <code><a href="#topic+AM">AM</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>SummaryAM</code> produces three  tables, an overall summary table, a table of the 
SNP names and positions, and a table  of results with 
the  p-value for each 
fixed effect in the final model.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AM">AM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
  # Since the following code takes longer than 5 seconds to run, it has been tagged as dontrun. 
  # However, the code can be run by the user. 
  #

  #---------------
  # read the map 
  #---------------
  #
  # File is a plain space separated text file with the first row 
  # the column headings
  complete.name &lt;- system.file('extdata', 'map.txt', 
                                   package='Eagle')
  map_obj &lt;- ReadMap(filename=complete.name) 

 # to look at the first few rows of the map file
 head(map_obj)

  #------------------
  # read marker data
  #------------------
  # Reading in a PLINK ped file 
  # and setting the available memory on the machine for the reading of the data to 8 gigabytes
  complete.name &lt;- system.file('extdata', 'geno.ped', 
                                     package='Eagle')
  geno_obj &lt;- ReadMarker(filename=complete.name,  type='PLINK', availmemGb=8) 
 
  #----------------------
  # read phenotype data
  #-----------------------

  # Read in a plain text file with data on a single trait and two fixed effects
  # The first row of the text file contains the column names y, cov1, and cov2. 
  complete.name &lt;- system.file('extdata', 'pheno.txt', package='Eagle')
  
  pheno_obj &lt;- ReadPheno(filename=complete.name)
           
  #-------------------------------------------------------
  # Perform multiple-locus genome-wide association mapping 
  #-------------------------------------------------------                   
  res &lt;- AM(trait = 'y',
                           fformula=c("cov1 + cov2"),
                           map = map_obj,
                           pheno = pheno_obj,
                           geno = geno_obj)

  #-----------------------------------------
  # Produce additional summary information 
  #------------------------------------------

  SummaryAM(AMobj=res)
 
## End(Not run)



</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
