<!DOCTYPE html><html lang="en-US"><head><title>Help for package easybio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {easybio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Artist'><p>Visualization Artist for Custom Plots</p></a></li>
<li><a href='#available_tissue_class'><p>Retrieve Available Tissue Classes for a Given Species</p></a></li>
<li><a href='#available_tissue_type'><p>Retrieve Available Tissue Types for a Given Species</p></a></li>
<li><a href='#check_marker'><p>Verify Markers for Specific Clusters Using matchCellMarker</p></a></li>
<li><a href='#CHOL_DEGs'><p>Example DEGs data from Limma-Voom workflow for TCGA-CHOL project</p></a></li>
<li><a href='#dgeList'><p>Construct a DGEList Object</p></a></li>
<li><a href='#dprocess_dgeList'><p>Filter Low-Expressed Genes and Normalize DGEList Data</p></a></li>
<li><a href='#finsert'><p>Insert Specific Values into a Character Vector at Defined Positions</p></a></li>
<li><a href='#get_attr'><p>Retrieve Attributes from an R Object</p></a></li>
<li><a href='#get_marker'><p>Retrieve Markers for Specific Cells from cellMarker2</p></a></li>
<li><a href='#groupStat'><p>Perform Summary Analysis by Group Using Regular Expressions</p></a></li>
<li><a href='#groupStatI'><p>Perform Summary Analysis by Group Using an column Index</p></a></li>
<li><a href='#limmaFit'><p>Fit a Linear Model for RNA-seq data using limma</p></a></li>
<li><a href='#list2dt'><p>Convert a List with Vector Values to a Long Data.table</p></a></li>
<li><a href='#list2graph'><p>Convert a Named List into a Graph Based on Overlap</p></a></li>
<li><a href='#matchCellMarker2'><p>Match Markers with cellMarker2 Dataset</p></a></li>
<li><a href='#pbmc.markers'><p>Example marker data from Seurat::FindAllMarkers()</p></a></li>
<li><a href='#plotEnrichment2'><p>Plot Enrichment for a Specific Pathway in fgsea</p></a></li>
<li><a href='#plotGSEA'><p>Visualization of GSEA Result from <code>fgsea::fgsea()</code></p></a></li>
<li><a href='#plotMarkerDistribution'><p>Plot Distribution of a Marker Across Tissues and Cell Types</p></a></li>
<li><a href='#plotORA'><p>Visualization of ORA Test Results</p></a></li>
<li><a href='#plotPossibleCell'><p>Plot Possible Cell Distribution Based on matchCellMarker2() Results</p></a></li>
<li><a href='#plotRank'><p>Visualization of GSEA Rank Statistics</p></a></li>
<li><a href='#plotSeuratDot'><p>Create Dot Plots for Markers from check_marker</p></a></li>
<li><a href='#plotVolcano'><p>Plot Volcano Plot for Differentially Expressed Genes</p></a></li>
<li><a href='#prepare_geo'><p>Download and Process GEO Data</p></a></li>
<li><a href='#prepare_tcga'><p>Prepare TCGA Data for Analysis</p></a></li>
<li><a href='#setcolnames'><p>Rename Column Names of a Data Frame or Matrix</p></a></li>
<li><a href='#setrownames'><p>Rename Row Names of a Data Frame or Matrix</p></a></li>
<li><a href='#setSavedir'><p>Set a Directory for Saving Files</p></a></li>
<li><a href='#split_matrix'><p>Split a Matrix into Smaller Sub-matrices by Column or Row</p></a></li>
<li><a href='#theme_publication'><p>Custom ggplot2 Theme for Academic Publications</p></a></li>
<li><a href='#tuneParameters'><p>Optimize Resolution and Gene Number Parameters for Cell Type Annotation</p></a></li>
<li><a href='#uniprot_id_map'><p>Map UniProt IDs to Other Identifiers</p></a></li>
<li><a href='#workIn'><p>Perform Operations in a Specified Directory and Return to the Original Directory</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Comprehensive Single-Cell Annotation and Transcriptomic Analysis
Toolkit</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a comprehensive toolkit for single-cell annotation with the 'CellMarker2.0' database (see Xia Li, Peng Wang, Yunpeng Zhang (2023) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkac947">doi:10.1093/nar/gkac947</a>&gt;). Streamlines biological label assignment in single-cell RNA-seq data and facilitates transcriptomic analysis, including preparation of TCGA<a href="https://portal.gdc.cancer.gov/">https://portal.gdc.cancer.gov/</a> and GEO<a href="https://www.ncbi.nlm.nih.gov/geo/">https://www.ncbi.nlm.nih.gov/geo/</a> datasets, differential expression analysis and visualization of enrichment analysis results. Additional utility functions support various bioinformatics workflows. See Wei Cui (2024) &lt;<a href="https://doi.org/10.1101%2F2024.09.14.609619">doi:10.1101/2024.09.14.609619</a>&gt; for more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/person-c/easybio">https://github.com/person-c/easybio</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/person-c/easybio/issues">https://github.com/person-c/easybio/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, ggplot2, httr2, R6, xml2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, patchwork, rmarkdown, ggrepel, Seurat, limma,
GEOquery, fgsea, edgeR, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>biocViews:</td>
<td>limma, GEOquery, edgeR, fgsea</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-12 04:39:47 UTC; HUAWEI</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Cui <a href="https://orcid.org/0009-0004-8315-5899"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Cui &lt;m2c.w@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-12 04:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Artist'>Visualization Artist for Custom Plots</h2><span id='topic+Artist'></span>

<h3>Description</h3>

<p>The <code>Artist</code> class offers a suite of methods designed to create a variety of plots using <code>ggplot2</code> for
data exploration. Any methods prefixed with <code>plot_</code> or <code>test_</code> will log the command history along with
their results, allowing you to review all outcomes later via the <code>get_all_results()</code> method.
Notably, methods starting with <code>plot_</code> will check if the result of the preceding command is of the
<code>htest</code> class. If so, it will incorporate the previous command and its p-value as the title and subtitle,
respectively. This class encompasses methods for crafting dumbbell plots, bubble plots, divergence bar charts,
lollipop plots, contour plots, scatter plots with ellipses, donut plots, and pie charts.
Each method is tailored to map data to specific visual aesthetics and to apply additional customizations as needed.
</p>


<h3>Value</h3>

<p>The <code>R6</code> class <a href="#topic+Artist">Artist</a>.
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>data</code></dt><dd><p>Stores the dataset used for plotting.</p>
</dd>
<dt><code>command</code></dt><dd><p>recode the command.</p>
</dd>
<dt><code>result</code></dt><dd><p>record the plot.</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-Artist-new"><code>Artist$new()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-get_all_result"><code>Artist$get_all_result()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-test_wilcox"><code>Artist$test_wilcox()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-test_t"><code>Artist$test_t()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-plot_scatter"><code>Artist$plot_scatter()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-plot_box"><code>Artist$plot_box()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-dumbbbell"><code>Artist$dumbbbell()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-bubble"><code>Artist$bubble()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-barchart_divergence"><code>Artist$barchart_divergence()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-lollipop"><code>Artist$lollipop()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-contour"><code>Artist$contour()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-scatter_ellipses"><code>Artist$scatter_ellipses()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-donut"><code>Artist$donut()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-pie"><code>Artist$pie()</code></a>
</p>
</li>
<li> <p><a href="#method-Artist-clone"><code>Artist$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-Artist-new"></a>



<h4>Method <code>new()</code></h4>

<p>Initializes the Artist class with an optional dataset.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$new(data = NULL)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame  containing the dataset to be used for plotting. Default is <code>NULL</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>An instance of the Artist class.
</p>


<hr>
<a id="method-Artist-get_all_result"></a>



<h4>Method <code>get_all_result()</code></h4>

<p>Get all history result
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$get_all_result()</pre></div>



<h5>Returns</h5>

<p>a data.table object
</p>


<hr>
<a id="method-Artist-test_wilcox"></a>



<h4>Method <code>test_wilcox()</code></h4>

<p>Conduct wilcox.test
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$test_wilcox(formula, data = self$data, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>formula</code></dt><dd><p><code><a href="stats.html#topic+wilcox.test">wilcox.test()</a></code> formula arguments</p>
</dd>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted. Default is <code>self$data</code>.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings passed to <code><a href="stats.html#topic+wilcox.test">wilcox.test()</a></code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 scatter plot.
</p>


<hr>
<a id="method-Artist-test_t"></a>



<h4>Method <code>test_t()</code></h4>

<p>Conduct wilcox.test
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$test_t(formula, data = self$data, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>formula</code></dt><dd><p><code><a href="stats.html#topic+t.test">t.test()</a></code> formula arguments</p>
</dd>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted. Default is <code>self$data</code>.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings passed to <code><a href="stats.html#topic+t.test">t.test()</a></code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 scatter plot.
</p>


<hr>
<a id="method-Artist-plot_scatter"></a>



<h4>Method <code>plot_scatter()</code></h4>

<p>Creates a scatter plot.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$plot_scatter(
  data = self$data,
  fun = function(x) x,
  x,
  y,
  ...,
  add = private$is_htest()
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted. Default is <code>self$data</code>.</p>
</dd>
<dt><code>fun</code></dt><dd><p>function to process the <code>self$data</code>.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column name for the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column name for the y-axis.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings passed to <code>aes()</code>.</p>
</dd>
<dt><code>add</code></dt><dd><p>whether to add the test result.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 scatter plot.
</p>


<hr>
<a id="method-Artist-plot_box"></a>



<h4>Method <code>plot_box()</code></h4>

<p>Creates a box plot.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$plot_box(
  data = self$data,
  fun = function(x) x,
  x,
  ...,
  add = private$is_htest()
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame or tibble containing the data to be plotted. Default is <code>self$data</code>.</p>
</dd>
<dt><code>fun</code></dt><dd><p>function to process the <code>self$data</code>.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column name for the x-axis.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings passed to <code>aes()</code>.</p>
</dd>
<dt><code>add</code></dt><dd><p>whether to add the test result.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot2 box plot.
</p>


<hr>
<a id="method-Artist-dumbbbell"></a>



<h4>Method <code>dumbbbell()</code></h4>

<p>Create a dumbbell plot
</p>
<p>This method generates a dumbbell plot using the provided data, mapping the specified columns
to the x-axis, y-axis, and color aesthetic.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$dumbbbell(data = self$data, x, y, col, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>col</code></dt><dd><p>The column in <code>data</code> to map to the color aesthetic.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the dumbbell plot.
</p>


<hr>
<a id="method-Artist-bubble"></a>



<h4>Method <code>bubble()</code></h4>

<p>Create a bubble plot
</p>
<p>This method generates a bubble plot where points are mapped to the x and y axes, with their
size and color representing additional variables.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$bubble(data = self$data, x, y, size, col, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>size</code></dt><dd><p>The column in <code>data</code> to map to the size of the points.</p>
</dd>
<dt><code>col</code></dt><dd><p>The column in <code>data</code> to map to the color of the points.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the bubble plot.
</p>


<hr>
<a id="method-Artist-barchart_divergence"></a>



<h4>Method <code>barchart_divergence()</code></h4>

<p>Create a divergence bar chart
</p>
<p>This method generates a divergence bar chart where bars are colored based on their
positive or negative value.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$barchart_divergence(data = self$data, group, y, fill, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>group</code></dt><dd><p>The column in <code>data</code> representing the grouping variable.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>fill</code></dt><dd><p>The column in <code>data</code> to map to the fill color of the bars.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the divergence bar chart.
</p>


<hr>
<a id="method-Artist-lollipop"></a>



<h4>Method <code>lollipop()</code></h4>

<p>Create a lollipop plot
</p>
<p>This method generates a lollipop plot, where points are connected to a baseline by vertical
segments, with customizable colors and labels.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$lollipop(data = self$data, x, y, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the lollipop plot.
</p>


<hr>
<a id="method-Artist-contour"></a>



<h4>Method <code>contour()</code></h4>

<p>Create a contour plot
</p>
<p>This method generates a contour plot that includes filled and outlined density contours,
with data points overlaid.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$contour(data = self$data, x, y, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the contour plot.
</p>


<hr>
<a id="method-Artist-scatter_ellipses"></a>



<h4>Method <code>scatter_ellipses()</code></h4>

<p>Create a scatter plot with ellipses
</p>
<p>This method generates a scatter plot where data points are colored by group, with ellipses
representing the confidence intervals for each group.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$scatter_ellipses(data = self$data, x, y, col, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>col</code></dt><dd><p>The column in <code>data</code> to map to the color aesthetic.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the scatter plot with ellipses.
</p>


<hr>
<a id="method-Artist-donut"></a>



<h4>Method <code>donut()</code></h4>

<p>Create a donut plot
</p>
<p>This method generates a donut plot, which is a variation of a pie chart with a hole in the center.
The sections of the donut represent the proportion of categories in the data.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$donut(data = self$data, x, y, fill, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>x</code></dt><dd><p>The column in <code>data</code> to map to the x-axis.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>fill</code></dt><dd><p>The column in <code>data</code> to map to the fill color of the sections.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the donut plot.
</p>


<hr>
<a id="method-Artist-pie"></a>



<h4>Method <code>pie()</code></h4>

<p>Create a pie chart
</p>
<p>This method generates a pie chart where sections represent the proportion of categories in the data.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$pie(data = self$data, y, fill, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A data frame containing the data to be plotted.</p>
</dd>
<dt><code>y</code></dt><dd><p>The column in <code>data</code> to map to the y-axis.</p>
</dd>
<dt><code>fill</code></dt><dd><p>The column in <code>data</code> to map to the fill color of the sections.</p>
</dd>
<dt><code>...</code></dt><dd><p>Additional aesthetic mappings or other arguments passed to <code>ggplot</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A ggplot object representing the pie chart.
</p>


<hr>
<a id="method-Artist-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>Artist$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>




<h3>Examples</h3>

<pre><code class='language-R'>library(data.table)
air &lt;- subset(airquality, Month %in% c(5, 6))
setDT(air)
cying &lt;- Artist$new(data = air)
cying$plot_scatter(x = Wind, y = Temp)
cying$test_wilcox(
  formula = Ozone ~ Month,
)
cying$plot_scatter(x = Wind, y = Temp)
cying$plot_scatter(f = \(x) x[, z := Wind * Temp], x = Wind, y = z)

</code></pre>

<hr>
<h2 id='available_tissue_class'>Retrieve Available Tissue Classes for a Given Species</h2><span id='topic+available_tissue_class'></span>

<h3>Description</h3>

<p>This function extracts and returns a unique list of available tissue classes
from the CellMarker2.0 database for a specified species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_tissue_class(spc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="available_tissue_class_+3A_spc">spc</code></td>
<td>
<p>A character string specifying the species (e.g., &quot;Human&quot; or &quot;Mouse&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of unique tissue classes available for the given species.
If no tissue classes are found, an empty vector is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>available_tissue_class("Human")
available_tissue_class("Mouse")

</code></pre>

<hr>
<h2 id='available_tissue_type'>Retrieve Available Tissue Types for a Given Species</h2><span id='topic+available_tissue_type'></span>

<h3>Description</h3>

<p>This function extracts and returns a unique list of available tissue types
from the CellMarker2.0 database for a specified species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>available_tissue_type(spc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="available_tissue_type_+3A_spc">spc</code></td>
<td>
<p>A character string specifying the species (e.g., &quot;Human&quot; or &quot;Mouse&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of unique tissue types available for the given species.
If no tissue types are found, an empty vector is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>available_tissue_type("Human")
available_tissue_type("Mouse")

</code></pre>

<hr>
<h2 id='check_marker'>Verify Markers for Specific Clusters Using matchCellMarker</h2><span id='topic+check_marker'></span>

<h3>Description</h3>

<p>This function checks the markers for specified clusters returned by the
<code>matchCellMarker2</code> function. It allows users to filter by species, cluster,
and to specify whether to consider cis or trans interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_marker(
  marker,
  n,
  spc,
  tissueClass = available_tissue_class(spc),
  tissueType = available_tissue_type(spc),
  cl = c(),
  topcellN = 2,
  cis = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_marker_+3A_marker">marker</code></td>
<td>
<p>A data frame of markers obtained from <code>Seurat::FindAllMarkers</code>.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_n">n</code></td>
<td>
<p>An integer specifying the top number of genes to match from the
input markers.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_spc">spc</code></td>
<td>
<p>A character string specifying the species, which can be either
'Human' or 'Mouse'.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_tissueclass">tissueClass</code></td>
<td>
<p>A character specifying the tissue classes, default <code>available_tissue_class(spc)</code>.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_tissuetype">tissueType</code></td>
<td>
<p>A character specifying the tissue types, default <code>available_tissue_type(spc)</code>.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_cl">cl</code></td>
<td>
<p>An integer or vector of integers specifying the clusters to check.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_topcelln">topcellN</code></td>
<td>
<p>An integer specifying the number of top cells to check for
each cluster.</p>
</td></tr>
<tr><td><code id="check_marker_+3A_cis">cis</code></td>
<td>
<p>A logical value indicating whether to check marker directly from the top symbol of
<code>matchCellMarker2</code> or re-search marker for top cell in cellMarker2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where each name corresponds to a cell type and each
element is a vector of marker names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
# Check the top 50 markers for clusters 1, 4, and 7 in the Human species.
## Not run: 
library(easybio)
data(pbmc.markers)
check_marker(pbmc.markers, n = 50, spc = "Human", cl = c(1, 4, 7))

## End(Not run)
</code></pre>

<hr>
<h2 id='CHOL_DEGs'>Example DEGs data from Limma-Voom workflow for TCGA-CHOL project</h2><span id='topic+CHOL_DEGs'></span>

<h3>Description</h3>

<p>The data were obtained by the limma-voom workflow
</p>

<hr>
<h2 id='dgeList'>Construct a DGEList Object</h2><span id='topic+dgeList'></span>

<h3>Description</h3>

<p>This function creates a <code>DGEList</code> object from a count matrix, sample
information, and feature information. It is designed to facilitate the
analysis of differential gene expression using the edgeR package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dgeList(count, sample.info, feature.info)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dgeList_+3A_count">count</code></td>
<td>
<p>A numeric matrix where rows represent features (e.g., genes) and
columns represent samples. Row names should correspond to feature identifiers,
and column names should correspond to sample identifiers.</p>
</td></tr>
<tr><td><code id="dgeList_+3A_sample.info">sample.info</code></td>
<td>
<p>A data frame containing information about the samples. The
number of rows should match the number of columns in the <code>count</code> matrix.</p>
</td></tr>
<tr><td><code id="dgeList_+3A_feature.info">feature.info</code></td>
<td>
<p>A data frame containing information about the features. The
number of rows should match the number of rows in the <code>count</code> matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>DGEList</code> object as defined by the edgeR package, which includes the
count data, sample information, and feature information.
</p>

<hr>
<h2 id='dprocess_dgeList'>Filter Low-Expressed Genes and Normalize DGEList Data</h2><span id='topic+dprocess_dgeList'></span>

<h3>Description</h3>

<p>This function filters out low-expressed genes from a <code>DGEList</code> object and
normalizes the count data. It also provides diagnostic plots for raw and
filtered data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dprocess_dgeList(x, group.column, min.count = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dprocess_dgeList_+3A_x">x</code></td>
<td>
<p>A <code>DGEList</code> object containing raw count data.</p>
</td></tr>
<tr><td><code id="dprocess_dgeList_+3A_group.column">group.column</code></td>
<td>
<p>The name of the column in <code>x$samples</code> that contains the
grouping information for the samples.</p>
</td></tr>
<tr><td><code id="dprocess_dgeList_+3A_min.count">min.count</code></td>
<td>
<p>The minimum number of counts required for a gene to be
considered expressed. Genes with counts below this threshold in any group
will be filtered out. Defaults to 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a <code>DGEList</code> object with low-expressed genes
filtered out and normalization factors calculated.
</p>

<hr>
<h2 id='finsert'>Insert Specific Values into a Character Vector at Defined Positions</h2><span id='topic+finsert'></span>

<h3>Description</h3>

<p>This function constructs a character vector of a specified length, inserting
given values at positions determined by numeric indices. It is designed for
single cell annotation tasks, where specific annotations need to be placed
at certain positions in a vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>finsert(
  x = expression(c(0, 1, 3) == "Neutrophil", c(2, 4, 8) == "Macrophage"),
  len = integer(),
  setname = TRUE,
  na = "Unknown"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="finsert_+3A_x">x</code></td>
<td>
<p>An expression defining the value to insert and the positions at which
to insert them. The expression should be a list of logical comparisons,
where the left side is a numeric vector of positions and the right side is
the corresponding character value to insert.</p>
</td></tr>
<tr><td><code id="finsert_+3A_len">len</code></td>
<td>
<p>The desired length of the output character vector. If the specified
positions exceed this length, the vector will be padded with the <code>na</code> value.</p>
</td></tr>
<tr><td><code id="finsert_+3A_setname">setname</code></td>
<td>
<p>A logical value indicating whether to set names for the elements
of the vector. If <code>TRUE</code>, names are set as character representations of the
positions from 0 to the length of the vector minus one.</p>
</td></tr>
<tr><td><code id="finsert_+3A_na">na</code></td>
<td>
<p>The default value to use for positions not specified in <code>x</code>. This
value is also used to pad the vector if its length exceeds the positions
specified in <code>x</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector with the specified values inserted at given
positions and padded with the <code>na</code> value if necessary.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
# Insert "Neutrophil" at positions 0, 1, 3 and "Macrophage" at positions 2, 4, 8
# in a vector of length 10, with "Unknown" as the default value.
library(easybio)
annotated_vector &lt;- finsert(
  x = expression(
    c(0, 1, 3) == "Neutrophil",
    c(2, 4, 8) == "Macrophage"
  ),
  len = 10,
  na = "Unknown"
)
print(annotated_vector)
</code></pre>

<hr>
<h2 id='get_attr'>Retrieve Attributes from an R Object</h2><span id='topic+get_attr'></span>

<h3>Description</h3>

<p>This function extracts a specified attribute from an R object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_attr(x, attr_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_attr_+3A_x">x</code></td>
<td>
<p>An R object that has attributes.</p>
</td></tr>
<tr><td><code id="get_attr_+3A_attr_name">attr_name</code></td>
<td>
<p>The name of the attribute to retrieve.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The value of the attribute with the given name.
</p>

<hr>
<h2 id='get_marker'>Retrieve Markers for Specific Cells from cellMarker2</h2><span id='topic+get_marker'></span>

<h3>Description</h3>

<p>This function extracts a list of markers for one or more cell types from the
<code>cellMarker2</code> dataset. It allows filtering by species, cell type, the number
of markers to retrieve, and a minimum count threshold for marker occurrences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_marker(
  spc,
  cell = character(),
  tissueClass = available_tissue_class(spc),
  tissueType = available_tissue_type(spc),
  number = 5,
  min.count = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_marker_+3A_spc">spc</code></td>
<td>
<p>A character string specifying the species, which can be either
'Human' or 'Mouse'.</p>
</td></tr>
<tr><td><code id="get_marker_+3A_cell">cell</code></td>
<td>
<p>A character vector of cell types for which to retrieve markers.</p>
</td></tr>
<tr><td><code id="get_marker_+3A_tissueclass">tissueClass</code></td>
<td>
<p>A character specifying the tissue classes, default <code>available_tissue_class(spc)</code>.</p>
</td></tr>
<tr><td><code id="get_marker_+3A_tissuetype">tissueType</code></td>
<td>
<p>A character specifying the tissue types, default <code>available_tissue_type(spc)</code>.</p>
</td></tr>
<tr><td><code id="get_marker_+3A_number">number</code></td>
<td>
<p>An integer specifying the number of top markers to return for
each cell type.</p>
</td></tr>
<tr><td><code id="get_marker_+3A_min.count">min.count</code></td>
<td>
<p>An integer representing the minimum number of times a marker
must have been reported to be included in the results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list where each name corresponds to a cell type and each
element is a vector of marker names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
# Retrieve the top 5 markers for 'Macrophage' and 'Monocyte' cell types in humans,
# with a minimum count of 1.
library(easybio)
markers &lt;- get_marker(spc = "Human", cell = c("Macrophage", "Monocyte"))
print(markers)
</code></pre>

<hr>
<h2 id='groupStat'>Perform Summary Analysis by Group Using Regular Expressions</h2><span id='topic+groupStat'></span>

<h3>Description</h3>

<p>This function applies a specified function to each group defined by a regular expression
pattern applied to the names of a data object. It is useful for summarizing data when
groups are defined by a pattern in the names rather than a specific column or index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupStat(f, x, xname = colnames(x), patterns)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="groupStat_+3A_f">f</code></td>
<td>
<p>A function that takes a single argument and returns a summary of the data.</p>
</td></tr>
<tr><td><code id="groupStat_+3A_x">x</code></td>
<td>
<p>A data frame or matrix containing the data to be summarized.</p>
</td></tr>
<tr><td><code id="groupStat_+3A_xname">xname</code></td>
<td>
<p>A character vector containing the names of the variables in <code>x</code>.</p>
</td></tr>
<tr><td><code id="groupStat_+3A_patterns">patterns</code></td>
<td>
<p>A list of regular expressions that define the groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the summary statistics for each group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(easybio)
groupStat(f = \(x) x + 1, x = mtcars, patterns = list("mp", "t"))
</code></pre>

<hr>
<h2 id='groupStatI'>Perform Summary Analysis by Group Using an column Index</h2><span id='topic+groupStatI'></span>

<h3>Description</h3>

<p>This function applies a specified function to each group defined by an column index,
and returns a summary of the results. It is useful for summarizing data by
group when the groups are defined by an  column index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupStatI(f, x, idx)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="groupStatI_+3A_f">f</code></td>
<td>
<p>A function that takes a single argument and returns a summary of the data.</p>
</td></tr>
<tr><td><code id="groupStatI_+3A_x">x</code></td>
<td>
<p>A data frame or matrix containing the data to be summarized.</p>
</td></tr>
<tr><td><code id="groupStatI_+3A_idx">idx</code></td>
<td>
<p>A list of indices or group names that define the column groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the summary statistics for each group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(easybio)
groupStatI(f = \(x) x + 1, x = mtcars, idx = list(c(1, 10), 2))
</code></pre>

<hr>
<h2 id='limmaFit'>Fit a Linear Model for RNA-seq data using limma</h2><span id='topic+limmaFit'></span>

<h3>Description</h3>

<p>This function fits a linear model to processed <code>DGEList</code> data using the
<code>limma</code> package. It defines contrasts between groups and performs
differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>limmaFit(x, group.column)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limmaFit_+3A_x">x</code></td>
<td>
<p>A processed <code>DGEList</code> object containing normalized count data.</p>
</td></tr>
<tr><td><code id="limmaFit_+3A_group.column">group.column</code></td>
<td>
<p>The name of the column in <code>x$samples</code> that contains
the grouping information for the samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>eBayes</code> object containing the fitted linear model and
results of the differential expression analysis.
</p>

<hr>
<h2 id='list2dt'>Convert a List with Vector Values to a Long Data.table</h2><span id='topic+list2dt'></span>

<h3>Description</h3>

<p>This function converts a named list with vector values in each element to a
long data.table. The list is first flattened into a single vector, and then
the data.table is created with two columns: one for the name of the original
list element and another for the value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list2dt(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list2dt_+3A_x">x</code></td>
<td>
<p>A named list where each element contains a vector of values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A long data.table with two columns: 'name' and 'value'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(easybio)
list2dt(list(a = c(1, 1), b = c(2, 2)))
</code></pre>

<hr>
<h2 id='list2graph'>Convert a Named List into a Graph Based on Overlap</h2><span id='topic+list2graph'></span>

<h3>Description</h3>

<p>This function creates a graph from a named list, where the edges are determined
by the overlap between the elements of the list. Each node in the graph represents
an element of the list, and the weight of the edge between two nodes is the number
of overlapping elements between the two corresponding lists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list2graph(nodes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list2graph_+3A_nodes">nodes</code></td>
<td>
<p>A named list where each element is a vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table representing the graph, with columns for the node names
(<code>node_1</code> and <code>node_2</code>) and the weight of the edge (<code>interWeight</code>).
</p>

<hr>
<h2 id='matchCellMarker2'>Match Markers with cellMarker2 Dataset</h2><span id='topic+matchCellMarker2'></span>

<h3>Description</h3>

<p>This function matches markers from the <code>FindAllMarkers</code> output with the
<code>cellMarker2</code> dataset, filtering by species and selecting the top genes based
on their average log2 fold change and adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchCellMarker2(
  marker,
  n,
  spc,
  tissueClass = available_tissue_class(spc),
  tissueType = available_tissue_type(spc)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matchCellMarker2_+3A_marker">marker</code></td>
<td>
<p>A data frame of markers obtained from the <code>FindAllMarkers</code>
function, expected to contain columns such as <code>avg_log2FC</code>, <code>p_val_adj</code>,
and <code>gene</code>.</p>
</td></tr>
<tr><td><code id="matchCellMarker2_+3A_n">n</code></td>
<td>
<p>An integer specifying the top number of genes to match from the
input markers.</p>
</td></tr>
<tr><td><code id="matchCellMarker2_+3A_spc">spc</code></td>
<td>
<p>A character string specifying the species, which can be either
'Human' or 'Mouse'.</p>
</td></tr>
<tr><td><code id="matchCellMarker2_+3A_tissueclass">tissueClass</code></td>
<td>
<p>A character specifying the tissue classes, default <code>available_tissue_class(spc)</code>.</p>
</td></tr>
<tr><td><code id="matchCellMarker2_+3A_tissuetype">tissueType</code></td>
<td>
<p>A character specifying the tissue types, default <code>available_tissue_type(spc)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing matched markers from the <code>cellMarker2</code>
dataset, with additional columns indicating the number of matches and
ordered symbols.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example usage:
# Match the top 50 differential genes from the pbmc.markers dataset with the Human
# species in the cellMarker2 dataset.
## Not run: 
library(easybio)
data(pbmc.markers)
matchCellMarker2(pbmc.markers, n = 50, spc = "Human")[]

## End(Not run)
</code></pre>

<hr>
<h2 id='pbmc.markers'>Example marker data from Seurat::FindAllMarkers()</h2><span id='topic+pbmc.markers'></span>

<h3>Description</h3>

<p>The data were obtained by the seurat PBMC workflow.
exact script for this data is available as system.file(&quot;example-single-cell.R&quot;, package=&quot;easybio&quot;)
</p>

<hr>
<h2 id='plotEnrichment2'>Plot Enrichment for a Specific Pathway in fgsea</h2><span id='topic+plotEnrichment2'></span>

<h3>Description</h3>

<p>This function creates a plot of enrichment scores for a specified pathway.
It provides a visual representation of the enrichment
score (ES) along with the ranks and ticks indicating the GSEA walk length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEnrichment2(pathways, pwayname, stats, gseaParam = 1, ticksSize = 0.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotEnrichment2_+3A_pathways">pathways</code></td>
<td>
<p>A list of pathways.</p>
</td></tr>
<tr><td><code id="plotEnrichment2_+3A_pwayname">pwayname</code></td>
<td>
<p>The name of the pathway for which to plot enrichment.</p>
</td></tr>
<tr><td><code id="plotEnrichment2_+3A_stats">stats</code></td>
<td>
<p>A rank vector obtained from the 'fgsea' package.</p>
</td></tr>
<tr><td><code id="plotEnrichment2_+3A_gseaparam">gseaParam</code></td>
<td>
<p>The GSEA walk length parameter. Default is 1.</p>
</td></tr>
<tr><td><code id="plotEnrichment2_+3A_tickssize">ticksSize</code></td>
<td>
<p>The size of the tick marks. Default is 0.2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object representing the enrichment plot.
</p>

<hr>
<h2 id='plotGSEA'>Visualization of GSEA Result from <code><a href="fgsea.html#topic+fgsea">fgsea::fgsea()</a></code></h2><span id='topic+plotGSEA'></span>

<h3>Description</h3>

<p>The <code>plotGSEA</code> function visualizes the results of a GSEA (Gene Set Enrichment Analysis) using data from
the <code>fgsea</code> package. It generates a composite plot that includes an enrichment plot and a ranked metric plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGSEA(fgseaRes, pathways, pwayname, stats, save = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotGSEA_+3A_fgseares">fgseaRes</code></td>
<td>
<p>A data table containing the GSEA results from the <code>fgsea</code> package.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_pathways">pathways</code></td>
<td>
<p>A list of all pathways used in the GSEA analysis.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_pwayname">pwayname</code></td>
<td>
<p>The name of the pathway to visualize.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_stats">stats</code></td>
<td>
<p>A numeric vector representing the ranked statistics.</p>
</td></tr>
<tr><td><code id="plotGSEA_+3A_save">save</code></td>
<td>
<p>A logical value indicating whether to save the plot as a PDF file. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object.
</p>

<hr>
<h2 id='plotMarkerDistribution'>Plot Distribution of a Marker Across Tissues and Cell Types</h2><span id='topic+plotMarkerDistribution'></span>

<h3>Description</h3>

<p>This function creates a dot plot displaying the distribution of a specified marker across
different tissues and cell types, based on data from the CellMarker2.0 database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMarkerDistribution(mkr = character())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMarkerDistribution_+3A_mkr">mkr</code></td>
<td>
<p>character, the name of the marker to be plotted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object representing the distribution of the marker.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotMarkerDistribution("CD14")

## End(Not run)
</code></pre>

<hr>
<h2 id='plotORA'>Visualization of ORA Test Results</h2><span id='topic+plotORA'></span>

<h3>Description</h3>

<p>The <code>plotORA</code> function visualizes the results of an ORA (Over-Representation Analysis) test.
It generates a plot with customizable aesthetics for x, y, point size, and fill, with an option to flip the axes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotORA(data, x, y, size, fill, flip = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotORA_+3A_data">data</code></td>
<td>
<p>A data frame containing the ORA results to be visualized.</p>
</td></tr>
<tr><td><code id="plotORA_+3A_x">x</code></td>
<td>
<p>The column in <code>data</code> to map to the x-axis.</p>
</td></tr>
<tr><td><code id="plotORA_+3A_y">y</code></td>
<td>
<p>The column in <code>data</code> to map to the y-axis.</p>
</td></tr>
<tr><td><code id="plotORA_+3A_size">size</code></td>
<td>
<p>The column in <code>data</code> to map to the size of the points.</p>
</td></tr>
<tr><td><code id="plotORA_+3A_fill">fill</code></td>
<td>
<p>The column in <code>data</code> to map to the fill color of the bars or points.
Use a constant value for a single category.</p>
</td></tr>
<tr><td><code id="plotORA_+3A_flip">flip</code></td>
<td>
<p>A logical value indicating whether to flip the axes of the plot. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object.
</p>

<hr>
<h2 id='plotPossibleCell'>Plot Possible Cell Distribution Based on matchCellMarker2() Results</h2><span id='topic+plotPossibleCell'></span>

<h3>Description</h3>

<p>This function creates a dot plot to visualize the distribution of possible cell types
based on the results from the <code>matchCellMarker2()</code> function, utilizing data from the CellMarker2.0 database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPossibleCell(marker, min.uniqueN = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPossibleCell_+3A_marker">marker</code></td>
<td>
<p>data.table, the result from the <code>matchCellMarker2()</code> function.</p>
</td></tr>
<tr><td><code id="plotPossibleCell_+3A_min.uniquen">min.uniqueN</code></td>
<td>
<p>integer, the minimum number of unique marker genes that must be matched for a cell type to be included in the plot. Default is 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object representing the distribution of possible cell types.
</p>

<hr>
<h2 id='plotRank'>Visualization of GSEA Rank Statistics</h2><span id='topic+plotRank'></span>

<h3>Description</h3>

<p>The <code>plotRank</code> function visualizes the ranked statistics of a GSEA (Gene Set Enrichment Analysis) analysis.
The function creates a plot where the x-axis represents the rank of each gene, and the y-axis shows
the corresponding ranked list metric.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRank(stats)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotRank_+3A_stats">stats</code></td>
<td>
<p>A numeric vector containing the ranked statistics from a GSEA analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>

<hr>
<h2 id='plotSeuratDot'>Create Dot Plots for Markers from check_marker</h2><span id='topic+plotSeuratDot'></span>

<h3>Description</h3>

<p>This function generates dot plots for the markers obtained from the
<code>check_marker</code> function for specified cluster groups within a Seurat object.
The plots are saved to a temporary directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSeuratDot(srt, cls, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotSeuratDot_+3A_srt">srt</code></td>
<td>
<p>A Seurat object containing the single-cell data.</p>
</td></tr>
<tr><td><code id="plotSeuratDot_+3A_cls">cls</code></td>
<td>
<p>A list containing cluster groups to check. Each element of the list
should correspond to a cluster or a group of clusters for which to generate
dot plots.</p>
</td></tr>
<tr><td><code id="plotSeuratDot_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass to the <code>check_marker</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing multiple DotPlot.
</p>

<hr>
<h2 id='plotVolcano'>Plot Volcano Plot for Differentially Expressed Genes</h2><span id='topic+plotVolcano'></span>

<h3>Description</h3>

<p>This function generates a volcano plot for differentially expressed genes
(DEGs) using <code>ggplot2</code>. It allows for customization of the plot with
different aesthetic parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotVolcano(data, data.text, x, y, color, label)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotVolcano_+3A_data">data</code></td>
<td>
<p>A data frame containing the DEGs result.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_data.text">data.text</code></td>
<td>
<p>A data frame containing labeled data for text annotation.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_x">x</code></td>
<td>
<p>variable representing the aesthetic for the x-axis.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_y">y</code></td>
<td>
<p>variable representing the aesthetic for the y-axis.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_color">color</code></td>
<td>
<p>variable representing the column name for the color aesthetic.</p>
</td></tr>
<tr><td><code id="plotVolcano_+3A_label">label</code></td>
<td>
<p>variable representing the column name for the text label aesthetic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>ggplot</code> object representing the volcano plot.
</p>

<hr>
<h2 id='prepare_geo'>Download and Process GEO Data</h2><span id='topic+prepare_geo'></span>

<h3>Description</h3>

<p>This function downloads gene expression data from the Gene Expression Omnibus (GEO) database.
It retrieves either the expression matrix or the supplementary tabular data if the expression data is not available.
The function also allows for the conversion of probe identifiers to gene symbols and can combine multiple probes into a single symbol.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_geo(geo, dir = ".", combine = TRUE, method = "max")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_geo_+3A_geo">geo</code></td>
<td>
<p>A character string specifying the GEO Series ID (e.g., &quot;GSE12345&quot;).</p>
</td></tr>
<tr><td><code id="prepare_geo_+3A_dir">dir</code></td>
<td>
<p>A character string specifying the directory where files should be downloaded. Default is the current working directory (<code>"."</code>).</p>
</td></tr>
<tr><td><code id="prepare_geo_+3A_combine">combine</code></td>
<td>
<p>A logical value indicating whether to combine multiple probes into a single gene symbol. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="prepare_geo_+3A_method">method</code></td>
<td>
<p>A character string specifying the method to use for combining probes into a single gene symbol. Options are <code>"max"</code> (take the maximum value) or <code>"mean"</code> (compute the average). Default is <code>"max"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing:
</p>
<table role = "presentation">
<tr><td><code>data</code></td>
<td>
<p>A data frame of the expression matrix.</p>
</td></tr>
<tr><td><code>sample</code></td>
<td>
<p>A data frame of the sample metadata.</p>
</td></tr>
<tr><td><code>feature</code></td>
<td>
<p>A data frame of the feature metadata, which includes gene symbols if combining probes.</p>
</td></tr>
</table>

<hr>
<h2 id='prepare_tcga'>Prepare TCGA Data for Analysis</h2><span id='topic+prepare_tcga'></span>

<h3>Description</h3>

<p>This function prepares TCGA data for downstream analyses such as differential expression analysis with <code>limma</code> or survival analysis.
It extracts and processes the necessary information from the TCGA data object, separating tumor and non-tumor samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_tcga(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_tcga_+3A_data">data</code></td>
<td>
<p>A <code>SummarizedExperiment</code> object containing TCGA data, typically obtained from R package <code>TCGABiolinks</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list.
</p>

<hr>
<h2 id='setcolnames'>Rename Column Names of a Data Frame or Matrix</h2><span id='topic+setcolnames'></span>

<h3>Description</h3>

<p>This function renames the column names of a data frame or matrix to the
specified names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setcolnames(object, nm)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setcolnames_+3A_object">object</code></td>
<td>
<p>A data frame or matrix whose column names will be renamed.</p>
</td></tr>
<tr><td><code id="setcolnames_+3A_nm">nm</code></td>
<td>
<p>A character vector containing the new names for the columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame or matrix with the new column names.
</p>

<hr>
<h2 id='setrownames'>Rename Row Names of a Data Frame or Matrix</h2><span id='topic+setrownames'></span>

<h3>Description</h3>

<p>This function renames the row names of a data frame or matrix to the
specified names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setrownames(object, nm)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setrownames_+3A_object">object</code></td>
<td>
<p>A data frame or matrix whose row names will be renamed.</p>
</td></tr>
<tr><td><code id="setrownames_+3A_nm">nm</code></td>
<td>
<p>A character vector containing the new names for the rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame or matrix with the new row names.
</p>

<hr>
<h2 id='setSavedir'>Set a Directory for Saving Files</h2><span id='topic+setSavedir'></span>

<h3>Description</h3>

<p>This function sets a directory path for saving files, creating the directory if it
does not already exist. The directory path is created with the given arguments, which
are passed directly to <code>file.path()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setSavedir(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setSavedir_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to <code>file.path()</code> to construct the directory path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path to the newly created or existing directory.
</p>

<hr>
<h2 id='split_matrix'>Split a Matrix into Smaller Sub-matrices by Column or Row</h2><span id='topic+split_matrix'></span>

<h3>Description</h3>

<p>This function splits a matrix into multiple smaller matrices by column or row.
It is useful for processing large matrices in chunks, such as when performing
analysis on a single computer with limited memory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_matrix(matrix, chunk_size, column = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="split_matrix_+3A_matrix">matrix</code></td>
<td>
<p>A numeric or logical matrix to be split.</p>
</td></tr>
<tr><td><code id="split_matrix_+3A_chunk_size">chunk_size</code></td>
<td>
<p>The number of columns or rows to include in each smaller matrix.</p>
</td></tr>
<tr><td><code id="split_matrix_+3A_column">column</code></td>
<td>
<p>Divided by column(default is <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of smaller matrices, each with <code>chunk_size</code> columns or rows.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(easybio)
split_matrix(mtcars, chunk_size = 2)
split_matrix(mtcars, chunk_size = 5, column = FALSE)
</code></pre>

<hr>
<h2 id='theme_publication'>Custom ggplot2 Theme for Academic Publications</h2><span id='topic+theme_publication'></span>

<h3>Description</h3>

<p><code>theme_publication</code> creates a custom ggplot2 theme designed for academic publications, ensuring clarity, readability, and a professional appearance.
It is based on <code>theme_classic()</code> and includes additional refinements to axis lines, text, and other plot elements to meet the standards of high-quality academic figures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_publication(base_size = 12, base_family = "sans")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="theme_publication_+3A_base_size">base_size</code></td>
<td>
<p>numeric, the base font size. Default is 12.</p>
</td></tr>
<tr><td><code id="theme_publication_+3A_base_family">base_family</code></td>
<td>
<p>character, the base font family. Default is &quot;sans&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 theme object that can be applied to ggplot2 plots.
</p>
<p>ggplot2 theme.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)
p &lt;- ggplot(mtcars, aes(mpg, wt)) +
  geom_point() +
  theme_publication()
print(p)
</code></pre>

<hr>
<h2 id='tuneParameters'>Optimize Resolution and Gene Number Parameters for Cell Type Annotation</h2><span id='topic+tuneParameters'></span>

<h3>Description</h3>

<p>This function tunes the <code>resolution</code> parameter in <code>Seurat::FindClusters()</code> and the number of top differential genes (<code>N</code>) to obtain different cell type annotation results. The function generates UMAP plots for each parameter combination, allowing for a comparison of how different settings affect the clustering and annotation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tuneParameters(srt, resolution = numeric(), N = integer(), spc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tuneParameters_+3A_srt">srt</code></td>
<td>
<p>Seurat object, the input data object to be analyzed.</p>
</td></tr>
<tr><td><code id="tuneParameters_+3A_resolution">resolution</code></td>
<td>
<p>numeric vector, a vector of resolution values to be tested in <code>Seurat::FindClusters()</code>.</p>
</td></tr>
<tr><td><code id="tuneParameters_+3A_n">N</code></td>
<td>
<p>integer vector, a vector of values indicating the number of top differential genes to be used for matching in <code>matchCellMarker2()</code>.</p>
</td></tr>
<tr><td><code id="tuneParameters_+3A_spc">spc</code></td>
<td>
<p>character, the species parameter for the <code>matchCellMarker2()</code> function, specifying the organism.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ggplot2 objects, each representing a UMAP plot generated with a different combination of resolution and N parameters.
</p>

<hr>
<h2 id='uniprot_id_map'>Map UniProt IDs to Other Identifiers</h2><span id='topic+uniprot_id_map'></span>

<h3>Description</h3>

<p>This function maps UniProt IDs to other identifiers using UniProt's ID mapping service.
It sends a request to the UniProt API to perform the mapping and retrieves the results in a tabular format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uniprot_id_map(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uniprot_id_map_+3A_...">...</code></td>
<td>
<p>Parameters to be passed in the request body.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> containing the mapped identifiers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
uniprot_id_map(
  ids = "P21802,P12345",
  from = "UniProtKB_AC-ID",
  to = "UniRef90"
)

## End(Not run)
</code></pre>

<hr>
<h2 id='workIn'>Perform Operations in a Specified Directory and Return to the Original Directory</h2><span id='topic+workIn'></span>

<h3>Description</h3>

<p>This function allows you to perform operations in a specified directory and then
return to the original directory. It is useful when you need to work with files or
directories that are located in a specific location, but you want to return to the
original working directory after the operation is complete.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>workIn(dir, expr)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="workIn_+3A_dir">dir</code></td>
<td>
<p>The directory path in which to operate. If the directory does not exist,
it will be created recursively.</p>
</td></tr>
<tr><td><code id="workIn_+3A_expr">expr</code></td>
<td>
<p>An R expression to be evaluated within the specified directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of evaluating the expression within the specified directory.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
