<!DOCTYPE html><html lang="en-gb"><head><title>Help for package easylabel</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {easylabel}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_labels'><p>Add labels to a plotly scatter plot</p></a></li>
<li><a href='#arial'><p>Change plotly font to Arial</p></a></li>
<li><a href='#easylabel'><p>Interactive scatter plot labels</p></a></li>
<li><a href='#easyManhattan'><p>Interactive Manhattan plot labels</p></a></li>
<li><a href='#easyMAplot'><p>Interactive MA plot labels</p></a></li>
<li><a href='#easyVolcano'><p>Interactive volcano plot labels</p></a></li>
<li><a href='#gg_qqplot'><p>Log QQ p-value plot (ggplot2)</p></a></li>
<li><a href='#loadlabel'><p>Load easylabel saved state</p></a></li>
<li><a href='#qqplot'><p>Log QQ p-value plot</p></a></li>
<li><a href='#qqplot2'><p>Log QQ p-value plot (2nd stage)</p></a></li>
<li><a href='#summary.easylab'><p>Summarise easylabel saved state object</p></a></li>
<li><a href='#volc1'><p>Example volcano data for vignette</p></a></li>
<li><a href='#volc2'><p>Example volcano data for vignette</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Scatter Plot and Volcano Plot Labels</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Myles Lewis &lt;myles.lewis@qmul.ac.uk&gt;</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/myles-lewis/easylabel/issues">https://github.com/myles-lewis/easylabel/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/myles-lewis/easylabel">https://github.com/myles-lewis/easylabel</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Interactive labelling of scatter plots, volcano plots and 
    Manhattan plots using a 'shiny' and 'plotly' interface. Users can hover 
    over points to see where specific points are located and click points 
    on/off to easily label them. Labels can be dragged around the plot to place 
    them optimally. Plots can be exported directly to PDF for publication. For 
    plots with large numbers of points, points can optionally be rasterized as a 
    bitmap, while all other elements (axes, text, labels &amp; lines) are preserved 
    as vector objects. This can dramatically reduce file size for plots with 
    millions of points such as Manhattan plots, and is ideal for publication.</td>
</tr>
<tr>
<td>Language:</td>
<td>en-gb</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>DT, ggplot2, gtools, memoise, plotly (&ge; 4.10.0),
RColorBrewer, rlang, shiny, shinycssloaders, shinybusy</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi, knitr, magick, org.Hs.eg.db, qvalue, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-06 20:03:40 UTC; myles</td>
</tr>
<tr>
<td>Author:</td>
<td>Myles Lewis <a href="https://orcid.org/0000-0001-9365-5345"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Katriona Goldmann <a href="https://orcid.org/0000-0002-9073-6323"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Cankut Cubuk <a href="https://orcid.org/0000-0003-4646-0849"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-06 20:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_labels'>Add labels to a plotly scatter plot</h2><span id='topic+add_labels'></span>

<h3>Description</h3>

<p>Adds labels to a plotly 2d or 3d scatter plot. The labels can be dragged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_labels(p, labs, plotGlPixelRatio = 8)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_labels_+3A_p">p</code></td>
<td>
<p>A plotly scatter plot object</p>
</td></tr>
<tr><td><code id="add_labels_+3A_labs">labs</code></td>
<td>
<p>Character vector of labels to match</p>
</td></tr>
<tr><td><code id="add_labels_+3A_plotglpixelratio">plotGlPixelRatio</code></td>
<td>
<p>Integer passed to plotly.js config which controls
pixel resolution of webGl rendering. Larger values increase resolution of
points as well as file size.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Labels are identified by searching the rownames of the embedded marker data
within the plotly object. The plotly mode bar camera icon is set to export to
svg by default.
</p>


<h3>Value</h3>

<p>A plotly plot with added labels
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(plotly)
data(mtcars)

p &lt;- plot_ly(mtcars, x = ~mpg, y = ~wt, color = ~cyl,
             type = 'scatter', mode = 'markers')
p %&gt;% add_labels(c("Mazda RX4", "Fiat 128"))
</code></pre>

<hr>
<h2 id='arial'>Change plotly font to Arial</h2><span id='topic+arial'></span>

<h3>Description</h3>

<p>Change plotly font to Arial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arial(p, size = 14)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="arial_+3A_p">p</code></td>
<td>
<p>A plotly object</p>
</td></tr>
<tr><td><code id="arial_+3A_size">size</code></td>
<td>
<p>Font size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plotly object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(plotly)
data(mtcars)

p &lt;- plot_ly(mtcars, x = ~mpg, y = ~wt, color = ~cyl,
             type = 'scatter', mode = 'markers')
p %&gt;% arial
p %&gt;% arial(18)
</code></pre>

<hr>
<h2 id='easylabel'>Interactive scatter plot labels</h2><span id='topic+easylabel'></span>

<h3>Description</h3>

<p>Interactive labelling of scatter plots using shiny/plotly interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>easylabel(
  data,
  x,
  y,
  labs = NULL,
  startLabels = NULL,
  start_xy = NULL,
  cex.text = 0.72,
  col = NULL,
  colScheme = NULL,
  alpha = 1,
  shape = NULL,
  shapeScheme = 21,
  size = 8,
  sizeRange = c(4, 80),
  xlab = x,
  ylab = y,
  xlim = NULL,
  ylim = NULL,
  xticks = NULL,
  yticks = NULL,
  showOutliers = TRUE,
  outlier_shape = 5,
  outline_col = "white",
  outline_lwd = 0.5,
  plotly_filter = NULL,
  width = 800,
  height = 600,
  showgrid = FALSE,
  zeroline = TRUE,
  hline = NULL,
  vline = NULL,
  mgp = c(1.8, 0.5, 0),
  Ltitle = "",
  Rtitle = "",
  LRtitle_side = 1,
  labelDir = "radial",
  labCentre = NULL,
  lineLength = 75,
  text_col = "black",
  line_col = "black",
  rectangles = FALSE,
  rect_col = "white",
  border_col = "black",
  padding = 3,
  border_radius = 5,
  showLegend = TRUE,
  legendxy = c(1.02, 1),
  filename = NULL,
  panel.last = NULL,
  fullGeneNames = FALSE,
  AnnotationDb = NULL,
  custom_annotation = NULL,
  output_shiny = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="easylabel_+3A_data">data</code></td>
<td>
<p>Dataset (data.frame or data.table) to use for plot.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_x">x</code></td>
<td>
<p>Specifies column of x coordinates in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_y">y</code></td>
<td>
<p>Specifies column of y coordinates in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_labs">labs</code></td>
<td>
<p>Specifies the column in <code>data</code> with label names for points.
Label names do not have to be unique. If <code>NULL</code> defaults to <code>rownames(data)</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_startlabels">startLabels</code></td>
<td>
<p>Vector of initial labels. With a character vector, labels
are identified in the column specified by <code>labs</code>. With a numeric vector,
points to be labelled are referred to by row number.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_start_xy">start_xy</code></td>
<td>
<p>List containing label annotation starting coordinates.
Invoked by <code><a href="#topic+loadlabel">loadlabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for labels. Default 0.72 to match plotly font size.
See <code><a href="graphics.html#topic+text">text()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_col">col</code></td>
<td>
<p>Specifies which column in <code>data</code> affects point colour. Must be
categorical. If it is not a factor, it will be coerced to a factor.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_colscheme">colScheme</code></td>
<td>
<p>A single colour or a vector of colours for points.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_alpha">alpha</code></td>
<td>
<p>Alpha value for transparency of points.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_shape">shape</code></td>
<td>
<p>Specifies which column in <code>data</code> controls point shapes. If not a
factor, will be coerced to a factor.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_shapescheme">shapeScheme</code></td>
<td>
<p>A single symbol for points or a vector of symbols.
See <code>pch</code> in <code><a href="graphics.html#topic+points">points()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_size">size</code></td>
<td>
<p>Either a single value for size of points (default 8), or
specifies which column in <code>data</code> affects point size for bubble charts.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_sizerange">sizeRange</code></td>
<td>
<p>Range of size of points for bubble charts.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_xlab">xlab</code></td>
<td>
<p>x axis title. Accepts expressions when exporting base graphics.
Set <code>cex.lab</code> to alter the font size of the axis titles (default 1).
Set <code>cex.axis</code> to alter the font size of the axis numbering (default 1).</p>
</td></tr>
<tr><td><code id="easylabel_+3A_ylab">ylab</code></td>
<td>
<p>y axis title. Accepts expressions when exporting base graphics.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (x1, x2) of the plot.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_ylim">ylim</code></td>
<td>
<p>The y limits of the plot.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_xticks">xticks</code></td>
<td>
<p>List of custom x axis ticks and labels specified as a list of
two named vectors <code>at = ...</code> and <code>labels = ...</code>. Another method is to use
<code>xaxp</code> as a vector of the form <code>c(x1, x2, n)</code> giving the coordinates of the
extreme tick marks and the number of intervals between tick-marks.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_yticks">yticks</code></td>
<td>
<p>List of custom y axis ticks and labels specified as a list of
two named vectors <code>at = ...</code> and <code>labels = ...</code>. Another method is to use
<code>yaxp</code> as a vector of the form <code>c(y1, y2, n)</code> giving the coordinates of the
extreme tick marks and the number of intervals between tick-marks.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_showoutliers">showOutliers</code></td>
<td>
<p>Logical whether to show outliers on the margins of the
plot.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_outlier_shape">outlier_shape</code></td>
<td>
<p>Symbol for outliers.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_outline_col">outline_col</code></td>
<td>
<p>Colour of symbol outlines. Set to <code>NA</code> for no outlines.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_outline_lwd">outline_lwd</code></td>
<td>
<p>Line width of symbol outlines.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_plotly_filter">plotly_filter</code></td>
<td>
<p>Refers to a column of logical values in <code>data</code> used to
filter rows to reduce the number of points shown by plotly. We recommend
using this for datasets with &gt;100,000 rows. When saving to pdf, the full
original dataset is still plotted. This is useful for plots with millions of
points such as Manhattan plots where a subset of points to be labelled is
already known.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_width">width</code></td>
<td>
<p>Width of the plot in pixels. Saving to pdf scales 100 pixels to
1 inch.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_height">height</code></td>
<td>
<p>Height of the plot in pixels.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_showgrid">showgrid</code></td>
<td>
<p>Either logical whether to show gridlines, or a character
value where &quot;x&quot; means showing x axis gridlines and &quot;y&quot; means showing y axis
gridlines.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_zeroline">zeroline</code></td>
<td>
<p>Logical whether to show lines at x = 0 and y = 0.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_hline">hline</code></td>
<td>
<p>Adds horizontal lines at values of y.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_vline">vline</code></td>
<td>
<p>Adds vertical lines at values of x.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_mgp">mgp</code></td>
<td>
<p>The margin line for the axis title, axis labels and axis line.
See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_ltitle">Ltitle</code></td>
<td>
<p>A character or expression (see <a href="grDevices.html#topic+plotmath">plotmath</a>) value specifying
text for left side title. Size of font can be changed using <code>cex.lab</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_rtitle">Rtitle</code></td>
<td>
<p>A character or expression value specifying text for right side
title. Size of font can be changed using <code>cex.lab</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_lrtitle_side">LRtitle_side</code></td>
<td>
<p>On which side of the plot for <code>Ltitle</code> and <code>Rtitle</code>
(1 = bottom, 3 = top). See <code><a href="graphics.html#topic+mtext">mtext()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_labeldir">labelDir</code></td>
<td>
<p>Initial label line directions. Options include 'radial'
(default) for radial lines around the centre of the plot,
'origin' for radial lines around the origin,
'horiz' for horizontal and 'vert' for vertical,
'xellipse' and 'yellipse' for near-horizontal and near-vertical lines
arranged in an elliptical way around the centre,
'rect' for rectilinear lines (a mix of horizontal and vertical),
'x' for diagonal lines,
'oct' for lines in 8 directions around the centre.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_labcentre">labCentre</code></td>
<td>
<p>Coordinates in x/y units of the central point towards which
radial labels converge. Defaults to the centre of the plot.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_linelength">lineLength</code></td>
<td>
<p>Initial length of label lines in pixels.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_text_col">text_col</code></td>
<td>
<p>Colour of label text. If set to
<code>"match"</code> label text will match the colour of each point.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_line_col">line_col</code></td>
<td>
<p>Colour of label lines. If set to
<code>"match"</code> label line will match the colour of each point.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_rectangles">rectangles</code></td>
<td>
<p>Logical whether to show rectangles around labels
(not supported by plotly).</p>
</td></tr>
<tr><td><code id="easylabel_+3A_rect_col">rect_col</code></td>
<td>
<p>Colour for filling rectangles (not supported by plotly). If
set to <code>"match"</code> rectangle fill colour will match the colour of each point.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_border_col">border_col</code></td>
<td>
<p>Colour of rectangle borders (not supported by plotly).
Use <code>border_col = NA</code> to omit borders. If set to <code>"match"</code> rectangle border
colour will match the colour of each point.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_padding">padding</code></td>
<td>
<p>Amount of padding in pixels around label text.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_border_radius">border_radius</code></td>
<td>
<p>Amount of roundedness in pixels to apply to label
rectangles (not supported by plotly).</p>
</td></tr>
<tr><td><code id="easylabel_+3A_showlegend">showLegend</code></td>
<td>
<p>Logical whether to show or hide the legend.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_legendxy">legendxy</code></td>
<td>
<p>Vector of coordinates for the position of the legend.
Coordinates are in plotly paper reference with <code>c(0, 0)</code> being the bottom
left corner and <code>c(1, 1)</code> being the top right corner of the plot window.
Plotly has unusual behaviour in that the x coordinate always aligns the left
side of the legend. However, the y coordinate aligns the top, middle or
bottom of the legend dependent on whether it is in the top, middle or bottom
1/3 of the plot window. So <code>c(1, 0)</code> positions the legend in the bottom right
corner outside the right margin of the plot, while <code>c(1, 0.5)</code> centre aligns
the legend around the centre of y axis.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_filename">filename</code></td>
<td>
<p>Filename for saving plots to pdf in a browser. Rstudio opens
its own pdf file.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_panel.last">panel.last</code></td>
<td>
<p>An expression to be evaluated after plotting has taken
place but before the axes, title and box are added. This can be useful for
adding extra titles, legends or trend lines. Currently only works when saving
plots using base graphics and does not work with plotly. See <a href="graphics.html#topic+plot.default">plot.default</a></p>
</td></tr>
<tr><td><code id="easylabel_+3A_fullgenenames">fullGeneNames</code></td>
<td>
<p>Logical whether to expand gene symbols using
Bioconductor AnnotationDbi package. With multiple matches, returns first
value only.
See <code><a href="AnnotationDbi.html#topic+AnnotationDb-class">AnnotationDbi::mapIds()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_annotationdb">AnnotationDb</code></td>
<td>
<p>Annotation database to use when expanding gene symbols.
Defaults to human gene database <code>AnnotationDb = org.Hs.eg.db</code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_custom_annotation">custom_annotation</code></td>
<td>
<p>List of annotations to be added via
<code><a href="plotly.html#topic+layout">plotly::layout()</a></code>.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_output_shiny">output_shiny</code></td>
<td>
<p>Logical whether to output a shiny app. If <code>FALSE</code> a
plotly figure will be returned.</p>
</td></tr>
<tr><td><code id="easylabel_+3A_...">...</code></td>
<td>
<p>Further graphical parameters passed to <code>plot()</code> when saving via
base graphics. The most useful for most users are likely to be <code>cex.lab</code>
which alters axis title font size (default 1, see <code><a href="graphics.html#topic+par">par()</a></code>), <code>cex.axis</code> which
alters axis numbering font size (default 1), and <code>panel.last</code> which allows
additional plotting functions to be called after the main plot has been
plotted but before the labels and label lines are drawn, which will allow the
addition of trend lines, extra titles or legends for example (see
<code><a href="graphics.html#topic+plot.default">plot.default()</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Instructions:
</p>

<ul>
<li><p> Hover over and click on/off genes which you want to label.
</p>
</li>
<li><p> When you have selected all your chosen genes, then drag gene names to
move label positions.
</p>
</li>
<li><p> Click the save button to export a PDF or other file types (SVG, jpeg, tiff,
png) produced in base graphics. Files are saved to the browser downloads
folder by default.
</p>
</li>
<li><p> The Table tab shows a table view of the dataset to help with annotation.
</p>
</li></ul>

<p>The 'raster points' checkbox converts only the points to a raster which is
embedded in pdf or svg exports, while the rest of the plot (axis/label
lines/text etc) remain vectorised. This only applies when saving pdf or svg.
For other file types the whole plot is rasterized as normal by base graphics.
</p>
<p>For efficiency, the raster of the points is cached using the <code>memoise</code>
package. So if users change the labels or change from pdf to svg, the export
is much faster. If the resolution is changed the points will need to be
re-rasterized.
</p>
<p>In the shiny interface the &quot;Save state&quot; button saves an rds file containing
the label positions and state of the plot as an 'easylab' S3 class object to
the current working directory. This can be reloaded in future sessions using
<code><a href="#topic+loadlabel">loadlabel()</a></code>.
</p>


<h3>Value</h3>

<p>By default no return value. If <code>output_shiny = FALSE</code> or the shiny
button 'Export plotly &amp; exit' is pressed, a plotly figure is returned.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+easyVolcano">easyVolcano()</a></code>, <code><a href="#topic+easyMAplot">easyMAplot()</a></code>, <code><a href="#topic+easyManhattan">easyManhattan()</a></code>, <code><a href="#topic+loadlabel">loadlabel()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simple example using mtcars dataset
data(mtcars)
# Launch easylabel Shiny app: only run this example in interactive R sessions
if (interactive()) {
easylabel(mtcars, x = 'mpg', y = 'wt', col = 'cyl')
}
</code></pre>

<hr>
<h2 id='easyManhattan'>Interactive Manhattan plot labels</h2><span id='topic+easyManhattan'></span>

<h3>Description</h3>

<p>Interactive labelling of Manhattan plots using 'shiny' and 'plotly'
interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>easyManhattan(
  data,
  chrom = "chrom",
  pos = "pos",
  p = "p",
  labs = "rsid",
  startLabels = NULL,
  pcutoff = 5e-08,
  chromGap = NULL,
  chromCols = c("royalblue", "skyblue"),
  sigCol = "red",
  alpha = 0.7,
  labelDir = "horiz",
  xlab = "Chromosome position",
  ylab = expression("-log"[10] ~ "P"),
  xlim = NULL,
  ylim = NULL,
  outline_col = NA,
  shapeScheme = 16,
  size = 6,
  width = ifelse(transpose, 600, 1000),
  height = ifelse(transpose, 800, 600),
  lineLength = 60,
  npoints = max(c(nrow(data)/5, 1e+06)),
  nplotly = 1e+05,
  npeaks = NULL,
  span = 2e+07,
  transpose = FALSE,
  filename = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="easyManhattan_+3A_data">data</code></td>
<td>
<p>The dataset (data.frame or data.table) for the plot.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_chrom">chrom</code></td>
<td>
<p>The column of chomosome values in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_pos">pos</code></td>
<td>
<p>The column of SNP positions in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_p">p</code></td>
<td>
<p>The column of p values in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_labs">labs</code></td>
<td>
<p>The column of labels in <code>data</code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_startlabels">startLabels</code></td>
<td>
<p>Vector of initial labels. With a character vector, labels
are identified in the column specified by <code>labs</code>. With a numeric vector,
points to be labelled are referred to by row number.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to 5E-08.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_chromgap">chromGap</code></td>
<td>
<p>Size of gap between chromosomes along the x axis in base
pairs. If <code>NULL</code> this is automatically calculated dependent on the size of
the genome. Default is around 3E07 for a human genome, and smaller for
smaller genomes.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_chromcols">chromCols</code></td>
<td>
<p>A vector of colours for points by chromosome. Colours are
recycled dependent on the length of the vector.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_sigcol">sigCol</code></td>
<td>
<p>Colour for statistically significant points. Ignored if set to
<code>NA</code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_alpha">alpha</code></td>
<td>
<p>Transparency for points.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_labeldir">labelDir</code></td>
<td>
<p>Option for label lines. See <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_xlab">xlab</code></td>
<td>
<p>x axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_ylab">ylab</code></td>
<td>
<p>y axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (x1, x2) of the plot.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_ylim">ylim</code></td>
<td>
<p>The y limits of the plot.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_outline_col">outline_col</code></td>
<td>
<p>Colour of symbol outlines. Passed to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_shapescheme">shapeScheme</code></td>
<td>
<p>A single symbol for points or a vector of symbols. Passed
to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_size">size</code></td>
<td>
<p>Specifies point size. Passed to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_width">width</code></td>
<td>
<p>Width of the plot in pixels. Saving to pdf scales 100 pixels to
1 inch.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_height">height</code></td>
<td>
<p>Height of the plot in pixels.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_linelength">lineLength</code></td>
<td>
<p>Initial length of label lines in pixels.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_npoints">npoints</code></td>
<td>
<p>Maximum number of points to plot when saving the final plot to
pdf. By default plots with &gt;1 million points are thinned to speed up
plotting. Setting a value of <code>NA</code> will plot all points.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_nplotly">nplotly</code></td>
<td>
<p>Maximum number of points to display via plotly. We recommend
the default setting of 100,000 points (or fewer).</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_npeaks">npeaks</code></td>
<td>
<p>Number of peaks to label initially.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_span">span</code></td>
<td>
<p>a peak is defined as the most significant SNP within a window of
width span centred at that SNP.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_transpose">transpose</code></td>
<td>
<p>Logical whether to transpose the plot.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_filename">filename</code></td>
<td>
<p>Filename for saving to pdf.</p>
</td></tr>
<tr><td><code id="easyManhattan_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default no return value. If <code>output_shiny = FALSE</code> or the shiny
button 'Export plotly &amp; exit' is pressed, a plotly figure is returned.
See <code><a href="#topic+easylabel">easylabel()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+easylabel">easylabel()</a></code> <code><a href="#topic+easyVolcano">easyVolcano()</a></code>
</p>

<hr>
<h2 id='easyMAplot'>Interactive MA plot labels</h2><span id='topic+easyMAplot'></span>

<h3>Description</h3>

<p>Interactive labelling of MA plots using shiny/plotly interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>easyMAplot(
  data,
  x = NULL,
  y = NULL,
  padj = NULL,
  fdrcutoff = 0.05,
  colScheme = c("darkgrey", "blue", "red"),
  hline = 0,
  labelDir = "yellipse",
  xlab = expression("log"[2] ~ " mean expression"),
  ylab = expression("log"[2] ~ " fold change"),
  filename = NULL,
  showCounts = TRUE,
  useQ = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="easyMAplot_+3A_data">data</code></td>
<td>
<p>The dataset for the plot. Automatically attempts to recognises
DESeq2 and limma objects.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_x">x</code></td>
<td>
<p>Name of the column containing mean expression. For DESeq2 and limma
objects this is automatically set.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_y">y</code></td>
<td>
<p>Name of the column containing log fold change. For DESeq2 and limma
objects this is automatically set.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_padj">padj</code></td>
<td>
<p>Name of the column containing adjusted p values (optional). For DESeq2 and
limma objects this is automatically set. If <code>y</code> is specified and <code>padj</code> is
left blank or equal to <code>y</code>, nominal unadjusted p values are used for cut-off
for significance.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_fdrcutoff">fdrcutoff</code></td>
<td>
<p>Cut-off for FDR significance. Defaults to FDR &lt; 0.05. Can
be vector with multiple cut-offs. To use nominal P values instead of adjusted
p values, set <code>y</code> but leave <code>padj</code> blank.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_colscheme">colScheme</code></td>
<td>
<p>Colour colScheme. Length must match either length(fdrcutoff) + 1
to allow for non-significant genes, or match length(fdrcutoff) * 2 + 1 to
accommodates asymmetric colour colSchemes for positive &amp; negative fold change.
(see examples).</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_hline">hline</code></td>
<td>
<p>Vector of horizontal lines (default is y = 0).</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_labeldir">labelDir</code></td>
<td>
<p>Option for label lines. See <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_xlab">xlab</code></td>
<td>
<p>x axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_ylab">ylab</code></td>
<td>
<p>y axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_filename">filename</code></td>
<td>
<p>Filename for saving to pdf.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_showcounts">showCounts</code></td>
<td>
<p>Logical whether to show legend with number of
differentially expressed genes.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_useq">useQ</code></td>
<td>
<p>Logical whether to convert nominal P values to q values.
Requires the qvalue Bioconductor package.</p>
</td></tr>
<tr><td><code id="easyMAplot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default no return value. If <code>output_shiny = FALSE</code> or the shiny
button 'Export plotly &amp; exit' is pressed, a plotly figure is returned.
See <code><a href="#topic+easylabel">easylabel()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+easylabel">easylabel()</a></code> <code><a href="#topic+easyVolcano">easyVolcano()</a></code>
</p>

<hr>
<h2 id='easyVolcano'>Interactive volcano plot labels</h2><span id='topic+easyVolcano'></span>

<h3>Description</h3>

<p>Interactive labelling of volcano plots using shiny/plotly interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>easyVolcano(
  data,
  x = NULL,
  y = NULL,
  padj = y,
  fdrcutoff = 0.05,
  fccut = NULL,
  colScheme = c("darkgrey", "blue", "red"),
  xlab = expression("log"[2] ~ " fold change"),
  ylab = expression("-log"[10] ~ " P"),
  filename = NULL,
  showCounts = TRUE,
  useQ = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="easyVolcano_+3A_data">data</code></td>
<td>
<p>The dataset for the plot. Automatically attempts to recognises
DESeq2 and limma objects.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_x">x</code></td>
<td>
<p>Name of the column containing log fold change. For DESeq2 and limma
objects this is automatically set.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_y">y</code></td>
<td>
<p>Name of the column containing p values. For DESeq2 and limma objects
this is automatically set.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_padj">padj</code></td>
<td>
<p>Name of the column containing adjusted p values (optional). If
<code>y</code> is specified and <code>padj</code> is left blank or equal to <code>y</code>, nominal unadjusted
p values are used for cut-off for significance instead of adjusted p values.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_fdrcutoff">fdrcutoff</code></td>
<td>
<p>Cut-off for FDR significance. Defaults to FDR &lt; 0.05. If <code>y</code>
is specified manually and <code>padj</code> is left blank then this refers to the
cut-off for significant points using nominal unadjusted p values.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_fccut">fccut</code></td>
<td>
<p>Optional vector of log fold change cut-offs.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_colscheme">colScheme</code></td>
<td>
<p>Colour scheme. If no fold change cut-off is set, 2 colours
need to be specified. With a single fold change cut-off, 3 or 5 colours are
required, depending on whether the colours are symmetrical about x = 0.
Accommodates asymmetric colour schemes with multiple fold change cut-offs
(see examples).</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_xlab">xlab</code></td>
<td>
<p>x axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_ylab">ylab</code></td>
<td>
<p>y axis title. Accepts expressions.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_filename">filename</code></td>
<td>
<p>Filename for saving to pdf.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_showcounts">showCounts</code></td>
<td>
<p>Logical whether to show legend with number of
differentially expressed genes.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_useq">useQ</code></td>
<td>
<p>Logical whether to convert nominal P values to q values.
Requires the qvalue Bioconductor package.</p>
</td></tr>
<tr><td><code id="easyVolcano_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default no return value. If <code>output_shiny = FALSE</code> or the shiny
button 'Export plotly &amp; exit' is pressed, a plotly figure is returned.
See <code><a href="#topic+easylabel">easylabel()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+easylabel">easylabel()</a></code> <code><a href="#topic+easyMAplot">easyMAplot()</a></code>
</p>

<hr>
<h2 id='gg_qqplot'>Log QQ p-value plot (ggplot2)</h2><span id='topic+gg_qqplot'></span>

<h3>Description</h3>

<p>Produces a QQ plot via ggplot2. Requires a dataframe generated by
<code><a href="#topic+qqplot">qqplot()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_qqplot(df, scheme = c("darkgrey", "royalblue"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gg_qqplot_+3A_df">df</code></td>
<td>
<p>A dataframe generated by <code><a href="#topic+qqplot">qqplot()</a></code></p>
</td></tr>
<tr><td><code id="gg_qqplot_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 2 colours for plotting non-significant and
significant SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 graphics plot object
</p>

<hr>
<h2 id='loadlabel'>Load easylabel saved state</h2><span id='topic+loadlabel'></span>

<h3>Description</h3>

<p>Loads a file or easylab object containing saved state with specified labels
and their positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadlabel(object, data = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="loadlabel_+3A_object">object</code></td>
<td>
<p>Either a character value specifying an rds file to be loaded,
or an object of class 'easylab'.</p>
</td></tr>
<tr><td><code id="loadlabel_+3A_data">data</code></td>
<td>
<p>Dataset (data.frame or data.table) to use for plot. If not
specified, the function will first use embedded data included in <code>object</code>;
if no data is embedded it will attempt to load the original data object as
it was specified in the original call to <code><a href="#topic+easylabel">easylabel()</a></code>.</p>
</td></tr>
<tr><td><code id="loadlabel_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+easylabel">easylabel()</a></code>. Can be used to
overwrite original arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default no return value. Calls <code><a href="#topic+easylabel">easylabel()</a></code> which will open a
shiny interactive session based on previous settings stored in an 'easylab'
class object or rds file of such an object.
</p>

<hr>
<h2 id='qqplot'>Log QQ p-value plot</h2><span id='topic+qqplot'></span>

<h3>Description</h3>

<p>Fast function for generating a log quantile-quantile (QQ) p-value plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqplot(
  pval,
  fdr = NULL,
  fdr_cutoff = 0.05,
  scheme = c("darkgrey", "royalblue"),
  npoints = 5e+05,
  show_plot = TRUE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqplot_+3A_pval">pval</code></td>
<td>
<p>A vector of p-values</p>
</td></tr>
<tr><td><code id="qqplot_+3A_fdr">fdr</code></td>
<td>
<p>An optional vector of FDR values to save time if previously
computed. If not supplied, these will be calculated using <code><a href="stats.html#topic+p.adjust">p.adjust()</a></code>
using the Benjamini-Hochberg method.</p>
</td></tr>
<tr><td><code id="qqplot_+3A_fdr_cutoff">fdr_cutoff</code></td>
<td>
<p>Cutoff for FDR significance</p>
</td></tr>
<tr><td><code id="qqplot_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 2 colours for plotting non-significant and
significant SNPs</p>
</td></tr>
<tr><td><code id="qqplot_+3A_npoints">npoints</code></td>
<td>
<p>Limits the number of non-significant points being plotted to
speed up plotting. See details. Set to <code>NULL</code> to plot all points.</p>
</td></tr>
<tr><td><code id="qqplot_+3A_show_plot">show_plot</code></td>
<td>
<p>Logical whether to produce a plot via base graphics or just
return dataframe ready for plotting.</p>
</td></tr>
<tr><td><code id="qqplot_+3A_verbose">verbose</code></td>
<td>
<p>Whether to show messages</p>
</td></tr>
<tr><td><code id="qqplot_+3A_...">...</code></td>
<td>
<p>Optional plotting arguments passed to <code><a href="#topic+qqplot2">qqplot2()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a fast QQ plot. Particularly useful for analyses with very large
numbers of p-values (such as eQTL analysis) which can be slow to plot. The
function looks first for all comparisons which reached FDR at the designated
cut-off and ensures all of these points are plotted. Additional points which
typically overlap substantially near the origin are thinned by random
sampling. In this way the plot can be reduced from millions of points to
500,000 points with a plot which is indistinguishable from one with all
points plotted. For comparison, set <code>npoints</code> to <code>NULL</code> to plot all points as
usual.
</p>
<p>Calling <code><a href="#topic+qqplot">qqplot()</a></code> will result in a base graphics plot. The plotting
dataframe is returned invisibly, so users can save time when refining plots
by saving the dataframe produced by <code><a href="#topic+qqplot">qqplot()</a></code> and then invoking <code><a href="#topic+qqplot2">qqplot2()</a></code>
to simply plot the points. Users who prefer ggplot2 can also pass the
dataframe generated by <code><a href="#topic+qqplot">qqplot()</a></code> to <code><a href="#topic+gg_qqplot">gg_qqplot()</a></code>.
</p>


<h3>Value</h3>

<p>Generates a plot using base graphics. Also returns a dataframe
invisibly which can be used for downstream plotting via either <code><a href="#topic+qqplot2">qqplot2()</a></code>
or <code><a href="#topic+gg_qqplot">gg_qqplot()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+qqplot2">qqplot2()</a></code> <code><a href="#topic+gg_qqplot">gg_qqplot()</a></code>
</p>

<hr>
<h2 id='qqplot2'>Log QQ p-value plot (2nd stage)</h2><span id='topic+qqplot2'></span>

<h3>Description</h3>

<p>Second stage plotting function which accepts dataframe generated by
<code><a href="#topic+qqplot">qqplot()</a></code>. This can be used to avoid repeating computation of the QQ plot
values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqplot2(df, scheme = c("darkgrey", "royalblue"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqplot2_+3A_df">df</code></td>
<td>
<p>A dataframe generated by <code><a href="#topic+qqplot">qqplot()</a></code></p>
</td></tr>
<tr><td><code id="qqplot2_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 2 colours for plotting non-significant and
significant SNPs</p>
</td></tr>
<tr><td><code id="qqplot2_+3A_...">...</code></td>
<td>
<p>Optional plotting arguments passed to <code><a href="base.html#topic+plot">plot()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Produces a base graphics plot.
</p>

<hr>
<h2 id='summary.easylab'>Summarise easylabel saved state object</h2><span id='topic+summary.easylab'></span>

<h3>Description</h3>

<p>Prints summary information on an easylabel saved state object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'easylab'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.easylab_+3A_object">object</code></td>
<td>
<p>Object of class 'easylab'</p>
</td></tr>
<tr><td><code id="summary.easylab_+3A_...">...</code></td>
<td>
<p>Optional arguments for compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Prints information on the original easylabel call
and saved labels.
</p>

<hr>
<h2 id='volc1'>Example volcano data for vignette</h2><span id='topic+volc1'></span>

<h3>Description</h3>

<p>Example DESeq2 volcano data for vignette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volc1
</code></pre>


<h3>Format</h3>

<p>Data frame with 6 rows and 6 variables
</p>

<hr>
<h2 id='volc2'>Example volcano data for vignette</h2><span id='topic+volc2'></span>

<h3>Description</h3>

<p>Example limma volcano data for vignette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volc2
</code></pre>


<h3>Format</h3>

<p>Data frame with 6 rows and 6 variables
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
