<!DOCTYPE html><html><head><title>Help for package easyDifferentialGeneCoexpression</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {easyDifferentialGeneCoexpression}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#easyDifferentialGeneCoexpression'><p>Function that computes the differential coexpression of a list of probesets in a specific dataset and returns the most significant pairs</p></a></li>
<li><a href='#fromProbesetToGeneSymbol'><p>Function that associates a gene symbol to a probeset for some Affymetrix platforms</p></a></li>
<li><a href='#geoDataDownload'><p>Function that downloads gene expression data from GEO, after checking the connection</p></a></li>
<li><a href='#geoPlatformAnnotationsDownload'><p>Function that downloads the annotations of a GEO platform</p></a></li>
<li><a href='#probesetRetrieval'><p>Function that reads a CSV file of probesets or gene symbols and, in the latter case, it retrieves the original probesets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Title:</td>
<td>Easily Performs Differential Coexpression Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>A function that reads in the GEO code of a list of probesets or gene symbols, a gene expression dataset GEO accession code, the name of the dataset feature discriminating the two conditions for the differential coexpression, and the values of the two different conditions for the differential coexpression, and returns the significant pairs of genes/probesets with highest differential coexpression (p-value &lt; 0.005). If the input gene list is made of gene symbols, this package associates the probesets to these gene symbols, if found.  Platforms available: GPL80, GPL8300, GPL80, GPL96, GPL570, GPL571, GPL20115, GPL1293,  GPL6102, GPL6104, GPL6883, GPL6884, GPL13497, GPL14550, GPL17077, GPL6480. GEO: Gene Expression Omnibus. ID: identifier code. The GEO datasets are downloaded from the URL <a href="https://ftp.ncbi.nlm.nih.gov/geo/series/">https://ftp.ncbi.nlm.nih.gov/geo/series/</a>. </td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/davidechicco/easyDifferentialGeneCoexpression/issues">https://github.com/davidechicco/easyDifferentialGeneCoexpression/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/davidechicco/easyDifferentialGeneCoexpression">https://github.com/davidechicco/easyDifferentialGeneCoexpression</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>annotate, Biobase, diffcoexp, geneExpressionFromGEO, GEOquery,
jetset, limma, magrittr, xml2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-01 14:43:13 UTC; davide</td>
</tr>
<tr>
<td>Author:</td>
<td>Davide Chicco <a href="https://orcid.org/0000-0001-9655-7142"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Davide Chicco &lt;davidechicco@davidechicco.it&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-05 11:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='easyDifferentialGeneCoexpression'>Function that computes the differential coexpression of a list of probesets in a specific dataset and returns the most significant pairs</h2><span id='topic+easyDifferentialGeneCoexpression'></span>

<h3>Description</h3>

<p>Function that computes the differential coexpression of a list of probesets in a specific dataset and returns the most significant pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>easyDifferentialGeneCoexpression(
  list_of_probesets_to_select,
  GSE_code,
  featureNameToDiscriminateConditions,
  firstConditionName,
  secondConditionName,
  batchCorrection = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_list_of_probesets_to_select">list_of_probesets_to_select</code></td>
<td>
<p>list of probesets for which the differential coexpression should be computed</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_gse_code">GSE_code</code></td>
<td>
<p>GEO accession code of the dataset to analyze</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_featurenametodiscriminateconditions">featureNameToDiscriminateConditions</code></td>
<td>
<p>name of the feature of the dataset that contains the two conditions to investigate</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_firstconditionname">firstConditionName</code></td>
<td>
<p>name of the first condition in the feature to discriminate (for example, &quot;healthy&quot;)</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_secondconditionname">secondConditionName</code></td>
<td>
<p>name of the second condition in the feature to discriminate (for example, &quot;cancer&quot;)</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_batchcorrection">batchCorrection</code></td>
<td>
<p>says if the script should perform the batch correction with limma::removeBatchEffect() or not</p>
</td></tr>
<tr><td><code id="easyDifferentialGeneCoexpression_+3A_verbose">verbose</code></td>
<td>
<p>prints all the intermediate message to standard output or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing the significantly differentially co-expressed pairs of genes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
probesetList &lt;- c("200738_s_at", "217356_s_at", "206686_at")
verboseFlag &lt;- "TRUE"
batchCorrection &lt;- "TRUE"
signDiffCoexpressGenePairs &lt;- easyDifferentialGeneCoexpression(probesetList, 
"GSE3268", "description", "Normal", "Tumor", verboseFlag)
</code></pre>

<hr>
<h2 id='fromProbesetToGeneSymbol'>Function that associates a gene symbol to a probeset for some Affymetrix platforms</h2><span id='topic+fromProbesetToGeneSymbol'></span>

<h3>Description</h3>

<p>Function that associates a gene symbol to a probeset for some Affymetrix platforms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fromProbesetToGeneSymbol(
  thisProbeset,
  thisPlatform,
  this_platform_ann_df,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fromProbesetToGeneSymbol_+3A_thisprobeset">thisProbeset</code></td>
<td>
<p>probeset in input</p>
</td></tr>
<tr><td><code id="fromProbesetToGeneSymbol_+3A_thisplatform">thisPlatform</code></td>
<td>
<p>GEO platform accession code</p>
</td></tr>
<tr><td><code id="fromProbesetToGeneSymbol_+3A_this_platform_ann_df">this_platform_ann_df</code></td>
<td>
<p>annotation dataframe of the platform</p>
</td></tr>
<tr><td><code id="fromProbesetToGeneSymbol_+3A_verbose">verbose</code></td>
<td>
<p>prints all the intermediate message to standard output or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gene symbol as string
</p>

<hr>
<h2 id='geoDataDownload'>Function that downloads gene expression data from GEO, after checking the connection</h2><span id='topic+geoDataDownload'></span>

<h3>Description</h3>

<p>Function that downloads gene expression data from GEO, after checking the connection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geoDataDownload(GSE_code, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geoDataDownload_+3A_gse_code">GSE_code</code></td>
<td>
<p>GEO code dataset</p>
</td></tr>
<tr><td><code id="geoDataDownload_+3A_verbose">verbose</code></td>
<td>
<p>prints all the intermediate message to standard output or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a gene set gene expression AnnotationDataFrame
</p>

<hr>
<h2 id='geoPlatformAnnotationsDownload'>Function that downloads the annotations of a GEO platform</h2><span id='topic+geoPlatformAnnotationsDownload'></span>

<h3>Description</h3>

<p>Function that downloads the annotations of a GEO platform
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geoPlatformAnnotationsDownload(platformID, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geoPlatformAnnotationsDownload_+3A_platformid">platformID</code></td>
<td>
<p>GEO platform ID</p>
</td></tr>
<tr><td><code id="geoPlatformAnnotationsDownload_+3A_verbose">verbose</code></td>
<td>
<p>prints all the intermediate message to standard output or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe containing the annotations of the GEO platform
</p>

<hr>
<h2 id='probesetRetrieval'>Function that reads a CSV file of probesets or gene symbols and, in the latter case, it retrieves the original probesets</h2><span id='topic+probesetRetrieval'></span>

<h3>Description</h3>

<p>Function that reads a CSV file of probesets or gene symbols and, in the latter case, it retrieves the original probesets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>probesetRetrieval(
  probesets_or_gene_symbols,
  csv_file_name,
  platformCode,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="probesetRetrieval_+3A_probesets_or_gene_symbols">probesets_or_gene_symbols</code></td>
<td>
<p>flag saying if we're reading probesets or gene symbols</p>
</td></tr>
<tr><td><code id="probesetRetrieval_+3A_csv_file_name">csv_file_name</code></td>
<td>
<p>complete name of CSV file containing the probesets or the gene symbols</p>
</td></tr>
<tr><td><code id="probesetRetrieval_+3A_platformcode">platformCode</code></td>
<td>
<p>code of the microarray platform for which the probeset-gene symbol mapping should be done</p>
</td></tr>
<tr><td><code id="probesetRetrieval_+3A_verbose">verbose</code></td>
<td>
<p>prints all the intermediate message to standard output or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of probesets
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
