<!DOCTYPE html><html><head><title>Help for package EBEN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EBEN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BASIS'><p>An Example Data File for the Gauss Model</p></a></li>
<li><a href='#BASISbinomial'><p>An Example Data File for the Binomial Model</p></a></li>
<li><a href='#EBelasticNet.Binomial'><p>The EB Elastic Net Algorithm for Binomial Model with Normal-Gamma(NG) Prior Distribution</p></a></li>
<li><a href='#EBelasticNet.BinomialCV'><p> Cross Validation (CV) Function to Determine Hyperparameter of the EB_Elastic Net Algorithm for Binomial Model with Normal-Gamma (NG) Prior Distribution</p></a></li>
<li><a href='#EBelasticNet.Gaussian'><p>The EB Elastic Net Algorithm for Gaussian Model</p></a></li>
<li><a href='#EBelasticNet.GaussianCV'><p>Cross Validation (CV) Function to Determine Hyperparameters of</p>
the EBEN Algorithm for Gaussian Model</a></li>
<li><a href='#EBEN-internal'><p>Internal EBEN functions</p></a></li>
<li><a href='#EBEN-package'><p>Empirical Bayesian Elastic Net (EBEN)</p></a></li>
<li><a href='#EBlassoNEG.Binomial'><p>The EBlasso Algorithm for Binomial Model with Normal-Exponential-Gamma (NEG) Prior Distribution</p></a></li>
<li><a href='#EBlassoNEG.BinomialCV'><p>Cross Validation (CV) Function to Determine Hyperparameters of the EBlasso Algorithm for Binomial Model with Normal-Exponential-Gamma (NEG) Prior Distribution</p></a></li>
<li><a href='#EBlassoNEG.Gaussian'>
<p>The EBlasso Algorithm for Gaussian Model with Normal-Exponential-Gamma (NEG) Prior Distribution</p></a></li>
<li><a href='#EBlassoNEG.GaussianCV'><p>Cross Validation (CV) Function to Determine Hyperparameters of</p>
the EBlasso Algorithm for Gaussian Model with Normal-Exponential-Gamma (NEG) Prior Distribution</a></li>
<li><a href='#y'><p>Sample Response Data for Gaussian Model</p></a></li>
<li><a href='#yBinomial'><p>Sample Variable Data for Binomial Model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Empirical Bayesian Elastic Net</td>
</tr>
<tr>
<td>Version:</td>
<td>5.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-5-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Anhui Huang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anhui Huang &lt;anhuihuang@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides the Empirical Bayesian Elastic Net for handling multicollinearity in generalized linear regression models.  As a special case of the 'EBglmnet'
  package (also available on CRAN), this package encourages a grouping effects to select relevant variables and estimate the corresponding non-zero effects. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-29 02:52:26 UTC; anhui</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-29 16:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BASIS'>An Example Data File for the Gauss Model</h2><span id='topic+BASIS'></span>

<h3>Description</h3>

<p>This is a 1000x481 sample feature matrix</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BASIS)</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:1000, 1:481] 0 -1 0 0 1 0 1 0 1 0 ...
</p>


<h3>Details</h3>

<p>The data was simulated on a 2400cM chromosome, each column corresponded to an even spaced QTL</p>


<h3>Source</h3>

<p> 	Huang, A., Xu, S., and Cai, X. (2014). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. Heredity 10.1038/hdy.2014.79 <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BASIS)
</code></pre>

<hr>
<h2 id='BASISbinomial'>An Example Data File for the Binomial Model</h2><span id='topic+BASISbinomial'></span>

<h3>Description</h3>

<p>This is a 500x481 sample feature matrix</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(BASISbinomial)</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:500, 1:481] 0 -1 0 0 0 0 -1 -1 0 1 ...
</p>


<h3>Details</h3>

<p>The data was simulated on a 2400cM chromosome, each column corresponded to an even spaced QTL</p>


<h3>Source</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BASISbinomial)
</code></pre>

<hr>
<h2 id='EBelasticNet.Binomial'>The EB Elastic Net Algorithm for Binomial Model with Normal-Gamma(NG) Prior Distribution</h2><span id='topic+EBelasticNet.Binomial'></span>

<h3>Description</h3>

<p>Generalized linear regression, normal-Gxponential (NG) hierarchical prior for regression coefficients</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBelasticNet.Binomial(BASIS, Target, lambda, alpha,Epis = FALSE,verbose = 0)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBelasticNet.Binomial_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBelasticNet.Binomial_+3A_target">Target</code></td>
<td>
<p>Class label of each individual, TAKES VALUES OF 0 OR 1</p>
</td></tr>
<tr><td><code id="EBelasticNet.Binomial_+3A_lambda">lambda</code></td>
<td>
<p>Hyperparameter controls degree of shrinkage; can be obtained via Cross Validation; lambda&gt;0</p>
</td></tr>
<tr><td><code id="EBelasticNet.Binomial_+3A_alpha">alpha</code></td>
<td>
<p>Hyperparameter controls degree of shrinkage; can be obtained via Cross Validation; 0&lt;alpha&lt;1</p>
</td></tr>
<tr><td><code id="EBelasticNet.Binomial_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBelasticNet.Binomial_+3A_verbose">verbose</code></td>
<td>
<p>0 or 1; 1: display message; 0 no message</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction of K*(K-1)/2 more columns to BASIS</p>


<h3>Value</h3>

<table>
<tr><td><code>weight</code></td>
<td>
<p>the none-zero regression coefficients: <br />
col1,col2 are the indices of the bases(main if equal); <br />
col3: coefficent value; <br />
col4: posterior variance;<br />
col5: t-value;<br />
col6: p-value</p>
</td></tr>
<tr><td><code>logLikelihood</code></td>
<td>
<p>log likelihood from the final regression coefficients</p>
</td></tr>
<tr><td><code>WaldScore</code></td>
<td>
<p>Wald Score</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>Intercept</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the hyperparameter; same as input lambda</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the hyperparameter; same as input alpha</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASISbinomial)
data(yBinomial)
#reduce sample size to speed up the running time
n = 50;
k = 100;
N = length(yBinomial);
set  = sample(N,n);
BASIS = BASISbinomial[set,1:k];
y  = yBinomial[set];
output = EBelasticNet.Binomial(BASIS, y,lambda = 0.1,alpha = 0.5, Epis = FALSE,verbose = 5)
</code></pre>

<hr>
<h2 id='EBelasticNet.BinomialCV'> Cross Validation (CV) Function to Determine Hyperparameter of the EB_Elastic Net Algorithm for Binomial Model with Normal-Gamma (NG) Prior Distribution</h2><span id='topic+EBelasticNet.BinomialCV'></span>

<h3>Description</h3>

<p>Hyperparameter controls degree of shrinkage, and is obtained via Cross Validation (CV). This program calculates the maximum lambda
that allows one non-zero basis; and performs a search down to 0.001*lambda_max at even steps. (20 steps)</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBelasticNet.BinomialCV(BASIS, Target, nFolds,foldId, Epis = FALSE, verbose = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_target">Target</code></td>
<td>
<p>Class label of each individual, TAKES VALUES OF 0 OR 1</p>
</td></tr>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_nfolds">nFolds</code></td>
<td>
<p>number of n-fold cv</p>
</td></tr>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_foldid">foldId</code></td>
<td>
<p>random assign samples to different folds</p>
</td></tr>
<tr><td><code id="EBelasticNet.BinomialCV_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 5; larger verbose displays more messages</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS</p>


<h3>Value</h3>

<table>
<tr><td><code>CrossValidation</code></td>
<td>
<p>col1: hyperparameter; col2: loglikelihood mean; standard ERROR of nfold mean log likelihood</p>
</td></tr>
<tr><td><code>Lmabda_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
<tr><td><code>Alpha_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## not run
library(EBEN)
data(BASISbinomial)
data(yBinomial)
#reduce sample size to speed up the running time
n = 50;
k = 100;
N = length(yBinomial);
set.seed(1)
set  = sample(N,n);
BASIS = BASISbinomial[set,1:k];
y  = yBinomial[set];
nFolds = 3
## Not run: 
CV = EBelasticNet.BinomialCV(BASIS, y, nFolds = 3,Epis = FALSE)

## End(Not run)


</code></pre>

<hr>
<h2 id='EBelasticNet.Gaussian'>The EB Elastic Net Algorithm for Gaussian Model</h2><span id='topic+EBelasticNet.Gaussian'></span>

<h3>Description</h3>

<p>General linear regression, normal-Gamma (NG) hierarchical prior for regression coefficients</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBelasticNet.Gaussian(BASIS, Target, lambda, alpha,Epis = FALSE,verbose = 0)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBelasticNet.Gaussian_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBelasticNet.Gaussian_+3A_target">Target</code></td>
<td>
<p>Response each individual</p>
</td></tr>
<tr><td><code id="EBelasticNet.Gaussian_+3A_lambda">lambda</code></td>
<td>
<p>Hyperparameter controls degree of shrinkage; can be obtained via Cross Validation; lambda&gt;0</p>
</td></tr>
<tr><td><code id="EBelasticNet.Gaussian_+3A_alpha">alpha</code></td>
<td>
<p>Hyperparameter controls degree of shrinkage; can be obtained via Cross Validation; 0&lt;alpha&lt;1</p>
</td></tr>
<tr><td><code id="EBelasticNet.Gaussian_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBelasticNet.Gaussian_+3A_verbose">verbose</code></td>
<td>
<p>0 or 1; 1: display message; 0 no message</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction of K*(K-1)/2 more columns to BASIS</p>


<h3>Value</h3>

<table>
<tr><td><code>weight</code></td>
<td>
<p>the none-zero regression coefficients: <br />
col1,col2 are the indices of the bases(main if equal); <br />
col3: coefficent value; <br />
col4: posterior variance;<br />
col5: t-value;<br />
col6: p-value</p>
</td></tr>
<tr><td><code>WaldScore</code></td>
<td>
<p>Wald Score</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>Intercept</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the hyperparameter; same as input lambda</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the hyperparameter; same as input alpha</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>	Huang, A., Xu, S., and Cai, X. (2014). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. Heredity 10.1038/hdy.2014.79 <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASIS)
data(y)
n = 50;
k = 100;
BASIS = BASIS[1:n,1:k];
y  = y[1:n];
Blup = EBelasticNet.Gaussian(BASIS, y,lambda = 0.0072,alpha = 0.95, Epis = FALSE,verbose = 0)
betas 			= Blup$weight
betas
</code></pre>

<hr>
<h2 id='EBelasticNet.GaussianCV'>Cross Validation (CV) Function to Determine Hyperparameters of
the EBEN Algorithm for Gaussian Model</h2><span id='topic+EBelasticNet.GaussianCV'></span>

<h3>Description</h3>

<p>Hyperparameter controls degree of shrinkage, and is obtained via Cross Validation (CV). This program calculates the maximum lambda
that allows one non-zero basis; and performs a search down to 0.0001*lambda_max at even steps. (20 steps)</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBelasticNet.GaussianCV(BASIS, Target, nFolds,foldId, Epis = FALSE, verbose = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_target">Target</code></td>
<td>
<p>Response each individual</p>
</td></tr>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_nfolds">nFolds</code></td>
<td>
<p>number of n-fold cv</p>
</td></tr>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_foldid">foldId</code></td>
<td>
<p>random assign samples to different folds</p>
</td></tr>
<tr><td><code id="EBelasticNet.GaussianCV_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 5; larger verbose displays more messages</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS</p>


<h3>Value</h3>

<table>
<tr><td><code>CrossValidation</code></td>
<td>
<p>col1: hyperparameter; col2: loglikelihood mean; standard ERROR of nfold mean log likelihood</p>
</td></tr>
<tr><td><code>Lmabda_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
<tr><td><code>Alpha_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang, A., Xu, S., and Cai, X. (2013). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASIS)
data(y)
#reduce sample size to speed up the running time
n = 50;
k = 100;
BASIS = BASIS[1:n,1:k];
y  = y[1:n];
## Not run: 
CV = EBelasticNet.GaussianCV(BASIS, y, nFolds = 3,Epis = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='EBEN-internal'>Internal EBEN functions</h2><span id='topic+ijIndex'></span><span id='topic+CVonePair'></span><span id='topic+lambdaMax'></span>

<h3>Description</h3>

<p>Internal EBEN functions</p>


<h3>Usage</h3>

<pre><code class='language-R'>	ijIndex(trueLoc,K)
	CVonePair(X,y,nFolds,foldId,hyperpara,Epis,prior,family,verbose,group)
	lambdaMax(X,y,Epis)
</code></pre>


<h3>Details</h3>

<p>These are not intended for use by users.
<code>ijIndex</code>Function for looking at the pair of interaction terms.
<code>CVonePair</code>Function performs nFolds CV for the given one pair of hyperparameter.
<code>lambdaMax</code>Function calculate the maximum lambda for EBlasso-NE and EBEN in CV.
</p>


<h3>Author(s)</h3>

<p>Anhui Huang and Dianting Liu <br /> Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>

<hr>
<h2 id='EBEN-package'>Empirical Bayesian Elastic Net (EBEN)</h2><span id='topic+EBEN-package'></span><span id='topic+EBEN'></span>

<h3>Description</h3>

<p> Fast EBEN algorithms. <br />
EBEN implements a normal and generalized gamma hierearchical priors. <br />
(    **    )	Two parameters (alpha, lambda) are equivalent with elastic net priors. 	<br />
(    **    )	When parameter alpha = 1, it is equivalent with EBlasso-NE (normal + exponential) <br />
Two models are available for both methods: 			<br />
(    **    )	General linear regression model.			<br />
(    **    )	Logistic regression model.			<br />
Multi-collinearity: 	<br />
(    **    )	for group of high correlated or collinear variables: EBEN identifies the group of variables estimates their effects together. <br />
(    **    )	group of variables can be selected together. <br />
*Epistasis (two-way interactions) can be included for all models/priors <br />
*model implemented with memory efficient c code. <br />
*LAPACK/BLAS are used for most linear algebra computations.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> EBEN</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 5.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-10-06</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> gpl</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Anhui Huang
</p>


<h3>References</h3>

<p>key algorithms: <br />
Cai, X., Huang, A., and Xu, S. (2011). Fast empirical Bayesian LASSO for multiple quantitative trait locus mapping. BMC Bioinformatics 12, 211.<br />
Huang A, Xu S, Cai X. (2013). Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics  14(1):5. <br />
Huang, A., Xu, S., and Cai, X. (2014). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. Heredity 10.1038/hdy.2014.79 <br />
Other publications: <br />
Huang, A., E. Martin, et al. (2014). &quot;Detecting genetic interactions in pathway-based genome-wide association studies.&quot; Genet Epidemiol 38(4): 300-309.<br />
Huang, A., S. Xu, et al. (2014). &quot;Whole-genome quantitative trait locus mapping reveals major role of epistasis on yield of rice.&quot; PLoS ONE 9(1): e87330.<br />
Huang, A. (2014). &quot;Sparse model learning for inferring genotype and phenotype associations.&quot; Ph.D Dissertation. University of Miami(1186).
</p>

<hr>
<h2 id='EBlassoNEG.Binomial'>The EBlasso Algorithm for Binomial Model with Normal-Exponential-Gamma (NEG) Prior Distribution</h2><span id='topic+EBlassoNEG.Binomial'></span>

<h3>Description</h3>

<p>Generalized linear regression, normal-exponential-gamma (NEG) hierarchical prior for regression coefficients</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBlassoNEG.Binomial(BASIS, Target, a_gamma, b_gamma, Epis,verbose,group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBlassoNEG.Binomial_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_target">Target</code></td>
<td>
<p>Class label of each individual, TAKES VALUES OF 0 OR 1</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_a_gamma">a_gamma</code></td>
<td>
<p>Hyperparameters control degree of shrinkage; can be obtained via Cross Validation; a_gamma&gt;=-1</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_b_gamma">b_gamma</code></td>
<td>
<p>Hyperparameters control degree of shrinkage; can be obtained via Cross Validation; b_gamma&gt;0</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_verbose">verbose</code></td>
<td>
<p>0 or 1; 1: display message; 0 no message</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Binomial_+3A_group">group</code></td>
<td>
<p>0 or 1; 0: No group effect; 1 two-way interaction grouped. Only valid when Epis = TRUE </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS</p>


<h3>Value</h3>

<table>
<tr><td><code>weight</code></td>
<td>
<p>the none-zero regression coefficients: <br />
col1,col2 are the indices of the bases(main if equal); <br />
col3: coefficent value; <br />
col4: posterior variance;<br />
col5: t-value;<br />
col6: p-value</p>
</td></tr>
<tr><td><code>logLikelihood</code></td>
<td>
<p>log likelihood with the final regression coefficients</p>
</td></tr>
<tr><td><code>WaldScore</code></td>
<td>
<p>Wald Score</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>Intercept</p>
</td></tr>
<tr><td><code>a_gamma</code></td>
<td>
<p>the hyperparameter; same as input</p>
</td></tr>
<tr><td><code>b_gamma</code></td>
<td>
<p>the hyperparameter; same as input</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang, A., Xu, S., and Cai, X.(2012). Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC Genetics. Submitted</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASISbinomial)
data(yBinomial)
#reduce sample size to speed up the running time
n = 50;
k = 100;
BASIS = BASISbinomial[1:n,1:k];
y  = yBinomial[1:n];
output = EBlassoNEG.Binomial(BASIS,y,0.1,0.1,Epis = FALSE)
</code></pre>

<hr>
<h2 id='EBlassoNEG.BinomialCV'>Cross Validation (CV) Function to Determine Hyperparameters of the EBlasso Algorithm for Binomial Model with Normal-Exponential-Gamma (NEG) Prior Distribution</h2><span id='topic+EBlassoNEG.BinomialCV'></span>

<h3>Description</h3>

<p>Hyperparameters control degree of shrinkage, and are obtained via Cross Validation. This program performs three steps of CV.<br />
1st: a = b = 0.001, 0.01, 0.1, 1;<br />
2nd: fix b= b1; a=[-0.5, -0.4, -0.3, -0.2, -0.1, -0.01, 0.01, 0.05, 0.1, 0.5, 1];<br />
3rd: fix a = a2; b= 0.01 to 10 with a step size of one for b &gt; 1 and a step size of one on the logarithmic scale for b &lt; 1<br />
In the 2nd step, a can take value from -1 and values in [-1, -0.5] can be added to the set in line 13 of this function (The smaller a is, the less shrinkage.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBlassoNEG.BinomialCV(BASIS, Target, nFolds,foldId, Epis,verbose, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_target">Target</code></td>
<td>
<p>Class label of each individual, TAKES VALUES OF 0 OR 1</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_nfolds">nFolds</code></td>
<td>
<p>number of n-fold cv</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_foldid">foldId</code></td>
<td>
<p>random assign samples to different folds</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 5; larger verbose displays more messages</p>
</td></tr>
<tr><td><code id="EBlassoNEG.BinomialCV_+3A_group">group</code></td>
<td>
<p>TRUE or FALSE; FALSE: No group effect; TRUE two-way interaction grouped. Only valid when Epis = TRUE </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TRUE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS <br />
Note: Given the fact that degree of shrinkage is a monotonic function of (a,b), <br />
The function implemented a 3-step search as described in Huang, A. 2014, for full <br />
grid search, user needs to modify the function accordingly.</p>


<h3>Value</h3>

<table>
<tr><td><code>CrossValidation</code></td>
<td>
<p>col1: hyperparameters; col2: loglikelihood mean; standard ERROR of nfold mean log likelihood</p>
</td></tr>
<tr><td><code>a_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
<tr><td><code>b_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5. <br />
Huang, A., S. Xu, et al. Whole-genome quantitative trait locus mapping reveals major role of epistasis on yield of rice. PLoS ONE 2014, 9(1): e87330.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASISbinomial)
data(yBinomial)
#reduce sample size to speed up the running time
n = 50;
k = 100;
BASIS = BASISbinomial[1:n,1:k];
y  = yBinomial[1:n];
## Not run: 
CV = EBlassoNEG.BinomialCV(BASIS, y, nFolds = 3,Epis = FALSE, verbose = 0)

## End(Not run)
</code></pre>

<hr>
<h2 id='EBlassoNEG.Gaussian'>
The EBlasso Algorithm for Gaussian Model with Normal-Exponential-Gamma (NEG) Prior Distribution
</h2><span id='topic+EBlassoNEG.Gaussian'></span>

<h3>Description</h3>

<p>General linear regression, normal-exponential-gamma (NEG) hierarchical prior for regression coefficients
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBlassoNEG.Gaussian(BASIS, Target, a_gamma, b_gamma, Epis, verbose, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_target">Target</code></td>
<td>
<p>Response each individual</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_a_gamma">a_gamma</code></td>
<td>
<p>Hyperparameters control degree of shrinkage; can be obtained via Cross Validation</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_b_gamma">b_gamma</code></td>
<td>
<p>Hyperparameters control degree of shrinkage; can be obtained via Cross Validation</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 5; larger verbose displays more messages</p>
</td></tr>
<tr><td><code id="EBlassoNEG.Gaussian_+3A_group">group</code></td>
<td>
<p>0 or 1; 0: No group effect; 1 two-way interaction grouped. Only valid when Epis = TRUE </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis=TURE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS<br />
for memory efficient, the function pass n_effect to C. n_effect &gt; n_true effects, which is <br />
a rough guess on how many variables will be selected by the function <br />
by providing a relative 'small' n_effect, the function will not allocate <br />
a large trunck of memory during computation.
</p>


<h3>Value</h3>

<table>
<tr><td><code>weight</code></td>
<td>
<p>the none-zero regression coefficients: <br />
col1,col2 are the indices of the bases(main if equal); <br />
col3: coefficent value; <br />
col4: posterior variance;<br />
col5: t-value;<br />
col6: p-value</p>
</td></tr>
<tr><td><code>WaldScore</code></td>
<td>
<p>Wald Score</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>Intercept</p>
</td></tr>
<tr><td><code>residVar</code></td>
<td>
<p>residual variance</p>
</td></tr>
<tr><td><code>a_gamma</code></td>
<td>
<p>the hyperparameter; same as input</p>
</td></tr>
<tr><td><code>b_gamma</code></td>
<td>
<p>the hyperparameter; same as input</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Cai, X., Huang, A., and Xu, S. (2011). Fast empirical Bayesian LASSO for multiple quantitative trait locus mapping. BMC Bioinformatics 12, 211.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASIS)
data(y)
n = 50;
k = 100;
BASIS = BASIS[1:n,1:k];
y  = y[1:n];
output = EBlassoNEG.Gaussian(BASIS, y, a_gamma = 0.1, b_gamma = 0.1)

</code></pre>

<hr>
<h2 id='EBlassoNEG.GaussianCV'>Cross Validation (CV) Function to Determine Hyperparameters of
the EBlasso Algorithm for Gaussian Model with Normal-Exponential-Gamma (NEG) Prior Distribution</h2><span id='topic+EBlassoNEG.GaussianCV'></span>

<h3>Description</h3>

<p>Hyperparameters control degree of shrinkage, and are obtained
via Cross Validation. This program performs three steps of CV.<br />
1st: a = b = 0.001, 0.01, 0.1, 1;<br />
2nd: fix b= b1; a=[-0.5, -0.4, -0.3, -0.2, -0.1, -0.01, 0.01, 0.05, 0.1, 0.5, 1];<br />
3rd: fix a = a2; b= 0.01 to 10 with a step size of one for b &gt; 1 and a step size of one on the logarithmic scale for b &lt; 1<br />
In the 2nd step, a can take value from -1 and values in [-1, -0.5] can be added to the set in line 13 of this function (The smaller a is, the less shrinkage.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBlassoNEG.GaussianCV(BASIS, Target, nFolds, foldId, Epis,verbose, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_basis">BASIS</code></td>
<td>
<p>sample matrix; rows correspond to samples, columns correspond to features</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_target">Target</code></td>
<td>
<p>Class label of each individual, TAKES VALUES OF 0 OR 1</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_nfolds">nFolds</code></td>
<td>
<p>number of n-fold cv</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_foldid">foldId</code></td>
<td>
<p>random assign samples to different folds</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_epis">Epis</code></td>
<td>
<p>TRUE or FALSE for including two-way interactions</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_verbose">verbose</code></td>
<td>
<p>from 0 to 5; larger verbose displays more messages</p>
</td></tr>
<tr><td><code id="EBlassoNEG.GaussianCV_+3A_group">group</code></td>
<td>
<p>TRUE or FALSE; FALSE: No group effect; TRUE two-way interaction grouped. Only valid when Epis = TRUE </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Epis= TRUE, the program adds two-way interaction K*(K-1)/2 more columns to BASIS <br />
Note: Given the fact that degree of shrinkage is a monotonic function of (a,b), <br />
The function implemented a 3-step search as described in Huang, A. 2014, for full <br />
grid search, user needs to modify the function accordingly.</p>


<h3>Value</h3>

<table>
<tr><td><code>CrossValidation</code></td>
<td>
<p>col1: hyperparameters; col2: loglikelihood mean; standard ERROR of nfold mean log likelihood</p>
</td></tr>
<tr><td><code>a_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
<tr><td><code>b_optimal</code></td>
<td>
<p>the optimal hyperparameter as computed</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Anhui Huang; Dept of Electrical and Computer Engineering, Univ of Miami, Coral Gables, FL</p>


<h3>References</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5. <br />
Huang, A., S. Xu, et al. Whole-genome quantitative trait locus mapping reveals major role of epistasis on yield of rice. PLoS ONE 2014, 9(1): e87330.</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(EBEN)
data(BASIS)
data(y)
#reduce sample size to speed up the running time
n = 50;
k = 100;
BASIS = BASIS[1:n,1:k];
y  = y[1:n];
## Not run: 
CV = EBlassoNEG.GaussianCV(BASIS, y, nFolds = 3,Epis = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='y'>Sample Response Data for Gaussian Model</h2><span id='topic+y'></span>

<h3>Description</h3>

<p>Corresponding to the response of BASIS</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(y)</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:1000, 1] 113.5 97.1 116.6 96.7 105.5 ...
</p>


<h3>Source</h3>

<p> Huang, A., Xu, S., and Cai, X. (2014). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. Heredity 10.1038/hdy.2014.79 <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(y)
</code></pre>

<hr>
<h2 id='yBinomial'>Sample Variable Data for Binomial Model</h2><span id='topic+yBinomial'></span>

<h3>Description</h3>

<p>Corresponding to the class label of BASISbinomial</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yBinomial)</code></pre>


<h3>Format</h3>

<p>The format is:
int [1:500, 1] 1 1 1 1 1 1 1 1 1 1 ...
</p>


<h3>Source</h3>

<p>Huang A, Xu S, Cai X: Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics 2013, 14(1):5.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(BASISbinomial)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
