<!DOCTYPE html><html><head><title>Help for package ebGenotyping</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ebGenotyping}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ebGenotyping-package'>
<p>Genotyping and SNP Detection using Next Generation Sequencing Data</p></a></li>
<li><a href='#ecm'>
<p>Genotyping and SNP Detection using Next Generation Sequencing Data</p></a></li>
<li><a href='#estep'>
<p>E step</p></a></li>
<li><a href='#logit'>
<p>Logit Transformation</p></a></li>
<li><a href='#mstep'>
<p>CM steps</p></a></li>
<li><a href='#my.bisec'>
<p>Bisection method to find the root</p></a></li>
<li><a href='#rlogit'>
<p>Inverse Logit Transformation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genotyping and SNP Detection using Next Generation Sequencing
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-04-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gongyi Huang&lt;53hgy@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genotyping the population using next generation sequencing data is essentially important for the rare variant detection. In order to distinguish the genomic structural variation from sequencing error, we propose a statistical model which involves the genotype effect through a latent variable to depict the distribution of non-reference allele frequency data among different samples and different genome loci, while decomposing the sequencing error into sample effect and positional effect. An ECM algorithm is implemented to estimate the model parameters, and then the genotypes and SNPs are inferred based on the empirical Bayes method.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-12 14:20:14 UTC; huanggongyi</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-13 09:28:12</td>
</tr>
</table>
<hr>
<h2 id='ebGenotyping-package'>
Genotyping and SNP Detection using Next Generation Sequencing Data
</h2><span id='topic+ebGenotyping-package'></span><span id='topic+ebGenotyping'></span>

<h3>Description</h3>

<p>Genotyping the population using next generation sequencing data is essentially important for the rare variant detection. In order to distinguish the genomic structural variation from sequencing error, we propose a statistical model which involves the genotype effect through a latent variable to depict the distribution of non-reference allele frequency data among different samples and different genome loci, while decomposing the sequencing error into sample effect and positional effect. An ECM algorithm is implemented to estimate the model parameters, and then the genotypes and SNPs are inferred based on the empirical Bayes method.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ebGenotyping</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-04-07</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The most important function is ecm, which is used to establish the model described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data' to do genotyping and SNP detection using NGS data.
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>

<hr>
<h2 id='ecm'>
Genotyping and SNP Detection using Next Generation Sequencing Data
</h2><span id='topic+ecm'></span>

<h3>Description</h3>

<p>This function implements the method described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecm(dat,cvg,eps=1e-6,max.steps=500,eps.bisec=1e-6,ini.m=-7,ini.d=-7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ecm_+3A_dat">dat</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the non-reference counts of ith sample at jth position. 
</p>
</td></tr>
<tr><td><code id="ecm_+3A_cvg">cvg</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the depth of ith sample at jth position. 
</p>
</td></tr>
<tr><td><code id="ecm_+3A_eps">eps</code></td>
<td>

<p>a single value: a threshold to control the convergence criterion. The default is 1e-06.
</p>
</td></tr>
<tr><td><code id="ecm_+3A_max.steps">max.steps</code></td>
<td>

<p>a single value: the maximum steps to run iterative algorithm to estimate parameters. The default is 500.(Adjustment is needed according to the number of parameters to estimate and the initial value of them.) 
</p>
</td></tr>
<tr><td><code id="ecm_+3A_eps.bisec">eps.bisec</code></td>
<td>

<p>a single value: a threshold to control the convergence criterion of bisection criterion. The default is 1e-06.
</p>
</td></tr>
<tr><td><code id="ecm_+3A_ini.m">ini.m</code></td>
<td>

<p>the initial value of each element of mu. We suggest users to use default -7.
</p>
</td></tr>
<tr><td><code id="ecm_+3A_ini.d">ini.d</code></td>
<td>

<p>the initial value of each element of delta. We suggest users to use default -7.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the method described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data'. 
According to the paper, users can do genotyping with the estimated genotypes(&quot;geno.est&quot;), 
and do SNP detection with the posterior probabilities of RR(&quot;post.probs$zRR&quot;), based on the non-reference counts(&quot;dat&quot;)and depth(&quot;cvg&quot;).
</p>


<h3>Value</h3>

<table>
<tr><td><code>par.est</code></td>
<td>
<p>a list including the estimate of position effect(mu), sample effect(delta), and the probability of RR and RV.</p>
</td></tr>
<tr><td><code>post.probs</code></td>
<td>
<p>3 matrix: the estimate of the posterior probabilities of 3 genotypes for n samples at m positions.</p>
</td></tr>
<tr><td><code>steps</code></td>
<td>
<p>the total steps to run iterative algorithm.</p>
</td></tr>
<tr><td><code>geno.est</code></td>
<td>
<p>a n*m matrix: the estimated genotypes(0 for RR, 1 for RV and 2 for VV) of n samples at m positions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#---------generate simulation data-----------
#start:generate simulation data#
set.seed(2016)
m &lt;- 100
m0 &lt;- m*0.95
m1 &lt;- m-m0
n &lt;- 30
Q &lt;- 0.8
z &lt;- cbind(matrix(0,n,m0),matrix(rbinom(n*m1,1,Q),n,m1))
b &lt;- which(z==1)
R &lt;- 0.8 # proportion of homozygous SNP
w &lt;- rbinom(length(which(z==1)),1,R)
# z are genotypes
z[b[which(w==0)]] &lt;- 1
z[b[which(w==1)]] &lt;- 2
mu &lt;- rep(-3,m)# stands for no effect
delta &lt;- rep(-3,n)# stands for no effect
er.p &lt;- -abs(outer(delta,mu,"+"))
p &lt;- rlogit(er.p)
p[which(z==1)] &lt;- 1/2
p[which(z==2)] &lt;- 1-p[which(z==2)]
cvg &lt;- matrix(rbinom(m*n,50,0.5),n,m)
dat &lt;- matrix(sapply(1:(m*n),function(i) rbinom(1,cvg[i],p[i])),n,m)
#end:generate simulation data-#
#-----genotyping and SNP detection----------------
res &lt;- ecm(dat=dat,cvg=cvg)
mean(z!=res$geno.est)#genotyping error
#----------call SNP---------
#start:call snp#
# define a function to calculate power, typeI error and FDR.
cutsnp &lt;- function(fdr,alpha,true){
wh &lt;- (true!=0)
tp &lt;- sum((wh)&amp;(fdr&lt;alpha));
tn &lt;- sum((!wh)&amp;(fdr&gt;=alpha));
fp &lt;- sum((!wh)&amp;(fdr&lt;alpha));
fn &lt;- sum((wh)&amp;(fdr&gt;=alpha));
pw  &lt;- tp/(tp+fn);
t1 &lt;- fp/(fp+tn);
fdr &lt;- fp/(fp+tp);
return(c(TP=tp,TN=tn,FP=fp,FN=fn,power=pw,typeI=t1,FDR=fdr))
}
cutsnp(fdr=res$post.probs$zRR,alpha=0.001,true=z)
cutsnp(fdr=res$post.probs$zRR,alpha=0.01,true=z)
cutsnp(fdr=res$post.probs$zRR,alpha=0.05,true=z)
#end:call snp#
</code></pre>

<hr>
<h2 id='estep'>
E step 
</h2><span id='topic+estep'></span>

<h3>Description</h3>

<p>This function calculates the E step of ECM algorithm for the model described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estep(mu, delta, pm1, p0, dat, cvg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estep_+3A_mu">mu</code></td>
<td>

<p>a vetor of the same length as number of positions: the position effect.
</p>
</td></tr>
<tr><td><code id="estep_+3A_delta">delta</code></td>
<td>

<p>a vetor of the same length as number of samples: the sample effect.
</p>
</td></tr>
<tr><td><code id="estep_+3A_pm1">pm1</code></td>
<td>

<p>a single value,which is larger than 0 and less than 1: the probability of RR. 
</p>
</td></tr>
<tr><td><code id="estep_+3A_p0">p0</code></td>
<td>

<p>a single value,which is larger than 0 and less than 1: the probability of RV. 
</p>
</td></tr>
<tr><td><code id="estep_+3A_dat">dat</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the non-reference counts of ith sample at jth position. 
</p>
</td></tr>
<tr><td><code id="estep_+3A_cvg">cvg</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the depth of ith sample at jth position. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The value of mu and delta must satisfy that each element of outer(delta,mu,&quot;+&quot;) must less than zero. This is the requirement of the model 
described in paper &quot;Genotyping for Rare Variant Detection Using Next-generation Sequencing Data.&quot;
</p>


<h3>Value</h3>

<table>
<tr><td><code>zRR</code></td>
<td>
<p>a n*m matrix: the posterior probabilities of genotype RR for n samples at m positions</p>
</td></tr>
<tr><td><code>zRV</code></td>
<td>
<p>a n*m matrix: the posterior probabilities of genotype RV for n samples at m positions</p>
</td></tr>
<tr><td><code>zVV</code></td>
<td>
<p>a n*m matrix: the posterior probabilities of genotype VV for n samples at m positions</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The most important function in this package is &quot;ecm&quot;. &quot;estep&quot; is a function called by &quot;ecm&quot; to realize one E step in the whole process of iteration in &quot;ecm&quot;.
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>

<hr>
<h2 id='logit'>
Logit Transformation
</h2><span id='topic+logit'></span>

<h3>Description</h3>

<p>This function is for calculating the logit transformation:ln(x/(1-x))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logit(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logit_+3A_x">x</code></td>
<td>

<p>A numeric vector, whose elements are all greater than 0 and less than 1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ln(x/(1-x))
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logit(0.5)
</code></pre>

<hr>
<h2 id='mstep'>
CM steps
</h2><span id='topic+mstep'></span>

<h3>Description</h3>

<p>This function calculates the CM steps of ECM algorithm for the model described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mstep(mu0, delta0, zm1, z0, zp1, dat, cvg, eps = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mstep_+3A_mu0">mu0</code></td>
<td>

<p>a vetor of the same length as number of positions: the initial value of position effect mu.
</p>
</td></tr>
<tr><td><code id="mstep_+3A_delta0">delta0</code></td>
<td>

<p>a vetor of the same length as number of samples: the initial value of sample effect delta.
</p>
</td></tr>
<tr><td><code id="mstep_+3A_zm1">zm1</code></td>
<td>

<p>the output of estep: the posterior probabilities of genotype RR for n samples at m positions
</p>
</td></tr>
<tr><td><code id="mstep_+3A_z0">z0</code></td>
<td>

<p>the output of estep: the posterior probabilities of genotype RV for n samples at m positions
</p>
</td></tr>
<tr><td><code id="mstep_+3A_zp1">zp1</code></td>
<td>

<p>the output of estep: the posterior probabilities of genotype VV for n samples at m positions
</p>
</td></tr>
<tr><td><code id="mstep_+3A_dat">dat</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the non-reference counts of ith sample at jth position. 
</p>
</td></tr>
<tr><td><code id="mstep_+3A_cvg">cvg</code></td>
<td>

<p>a n*m matrix: the ith row, jth column of the matrix represents the depth of ith sample at jth position. 
</p>
</td></tr>
<tr><td><code id="mstep_+3A_eps">eps</code></td>
<td>

<p>a single value: a threshold to control the convergence criterion. The default is 1e-06.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The value of mu and delta must satisfy that each element of outer(delta,mu,&quot;+&quot;) must less than zero. This is the requirement of the model 
described in 'An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data'.
</p>


<h3>Value</h3>

<table>
<tr><td><code>mu</code></td>
<td>
<p>the optimal value of mu in current CM steps</p>
</td></tr>
<tr><td><code>delta</code></td>
<td>
<p>the optimal value of delta in current CM steps</p>
</td></tr>
<tr><td><code>pRR</code></td>
<td>
<p>the optimal value of the probability of RR in current CM steps</p>
</td></tr>
<tr><td><code>pRV</code></td>
<td>
<p>the optimal value of the probability of RV in current CM steps</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The most important function in this package is &quot;ecm&quot;. &quot;mstep&quot; is a function called by &quot;ecm&quot; to realize one M step(several CM steps) in the whole process of iteration in &quot;ecm&quot;.
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>

<hr>
<h2 id='my.bisec'>
Bisection method to find the root
</h2><span id='topic+my.bisec'></span>

<h3>Description</h3>

<p>This function is to apply bisection method to find the root of a function f.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>my.bisec(f, int.l, int.u, eps = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="my.bisec_+3A_f">f</code></td>
<td>

<p>the function for which the root is sought.
</p>
</td></tr>
<tr><td><code id="my.bisec_+3A_int.l">int.l</code></td>
<td>

<p>a vector containing the lower bound of the interval to be searched for the root. The length of the vector is the same as that of the input of function f.
</p>
</td></tr>
<tr><td><code id="my.bisec_+3A_int.u">int.u</code></td>
<td>

<p>a vector containing the upper bound of the interval to be searched for the root. The length of the vector is the same as that of the input of function f.
</p>
</td></tr>
<tr><td><code id="my.bisec_+3A_eps">eps</code></td>
<td>

<p>a single value: a threshold to control the convergence criterion. The default is 1e-06.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both int.l and int.u must be specified: the upper bound must be strictly larger than the lower bound. 
</p>
<p>The function f must be well defined without invalid output(NA, nan, Inf, etc). 
</p>
<p>The length of the input of function f, the output of function f, int.l and int.u must be the same.
</p>


<h3>Value</h3>

<p>a vector containing the root of the function. If there is no root in the interval (int.l, int.u), lower bound of the interval will be returned.
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- function(x){
a &lt;- 1:10
return(x-a)
}
my.bisec(f=f, int.l=rep(-1,10), int.u=rep(11,10), eps = 1e-08)
</code></pre>

<hr>
<h2 id='rlogit'>
Inverse Logit Transformation
</h2><span id='topic+rlogit'></span>

<h3>Description</h3>

<p>This function is for calculating the inverse logit transformation:exp(x)/(1+exp(x))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rlogit(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rlogit_+3A_x">x</code></td>
<td>

<p>A numeric vector
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to avoid overflow, we define the function like this: rlogit &lt;- function(x) return(ifelse(x&gt;100,1,exp(x)/(1+exp(x))))
</p>


<h3>Value</h3>

<p>exp(x)/(1+exp(x))
</p>


<h3>Author(s)</h3>

<p>Na You &lt;youn@mail.sysu.edu.cn&gt; and Gongyi Huang&lt;53hgy@163.com&gt;
</p>


<h3>References</h3>

<p>Na You and Gongyi Huang.(2016) An Empirical Bayes Method for Genotyping and SNP detection Using Multi-sample Next-generation Sequencing Data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rlogit(-3)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
