<!DOCTYPE html><html lang="en"><head><title>Help for package EBPRS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EBPRS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#EBPRS'><p>Main function</p></a></li>
<li><a href='#EBPRSpackage'><p>Description of the package</p></a></li>
<li><a href='#read_plink'><p>Read plink bfiles to R and reformat</p></a></li>
<li><a href='#traindat'><p>Example data for training set</p></a></li>
<li><a href='#validate'><p>Validate the performance of EBPRS</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Derive Polygenic Risk Score Based on Emprical Bayes Theory</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Shuang Song [aut, cre], Wei Jiang [aut], Lin Hou [aut] and Hongyu Zhao [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shuang Song &lt;song-s19@mails.tsinghua.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>EB-PRS is a novel method that leverages information for effect sizes across all the markers to improve the prediction accuracy.  No parameter tuning is needed in the method, and no external information is needed. This R-package provides the calculation of polygenic risk scores from the given training summary statistics and testing data. We can use EB-PRS to extract main information, estimate Empirical Bayes parameters, derive polygenic risk scores for  each individual in testing data, and evaluate the PRS according to AUC and predictive r2. See Song et al. (2020) &lt;<a href="https://doi.org/10.1371%2Fjournal.pcbi.1007565">doi:10.1371/journal.pcbi.1007565</a>&gt; for a detailed presentation of the method.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), ROCR, methods, BEDMatrix, data.table</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-25 08:01:54 UTC; lenovo</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-26 05:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='EBPRS'>Main function</h2><span id='topic+EBPRS'></span>

<h3>Description</h3>

<p>Clean the dataset, extract information from raw data and calculate effect sizes.
(Please notice that there are some requirements for the training and testing datasets.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBPRS(train, test, N1, N0, robust = T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="EBPRS_+3A_train">train</code></td>
<td>
<p>training dataset</p>
</td></tr>
<tr><td><code id="EBPRS_+3A_test">test</code></td>
<td>
<p>testing dataset (list) including fam, bed, bim, which can be generated from function read_plink in our package. If missing(test)=T, the function will use all SNPs in training dataset by default.</p>
</td></tr>
<tr><td><code id="EBPRS_+3A_n1">N1</code></td>
<td>
<p>case number</p>
</td></tr>
<tr><td><code id="EBPRS_+3A_n0">N0</code></td>
<td>
<p>control number</p>
</td></tr>
<tr><td><code id="EBPRS_+3A_robust">robust</code></td>
<td>
<p>T/F, indicator that whether robust estimation is needed. The function will run faster when robust is set to F. The default is T.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The raw training data should be a
data.fame including A1, A2, OR, P, SNP (NOTE that the colnames should be exactly consistent with the above).
</p>
<p>The SNP column (rsid) is used for indexing.
</p>
<p>An example training dataset can be acquired using data(&quot;traindat&quot;)
</p>
<p>&quot;test&quot; file can be generated from read_plink(&quot;path_to_test_plink_bfile&quot;)
</p>
<p>test is a list, which has test$fam (6 columns with information on samples), test$bim (6 columns with information on SNPs), test$bed (genotypes matrix 0, 1, 2)
</p>
<p>Note that in real data, we usually use beta0 = m/20 as the default setting for the EM algorithm,
which is accurate enough in most cases and will have little influence on the prediction performance.
If more accurate parameter estimation is required, we provide a robust estimation (by setting robust=T),
integrating our data-driven bootstrap-based parameter tuning method. This can
derive the best parameter for robust estimation, while more time is needed.
</p>


<h3>Value</h3>

<p>A list containing
data.frame (result): combining the summary statistics and estimated effect sizes (eff)
</p>
<p>estimated effect sizes (eff)
</p>
<p>estimated mu (muHat)
</p>
<p>estimated sigma2 (sigmaHat2)
</p>
<p>estimated proportion of non-associated SNPs (pi0)
</p>
<p>estimated variance of effect sizes of associated SNPs (sigma02)
</p>
<p>If the test file is provided the results also include:
</p>
<p>predictive r2 (r2)
</p>
<p>AUC (AUC)
</p>
<p>estimated polygenic risk score (S)
</p>


<h3>Author(s)</h3>

<p>Shuang Song, Wei Jiang, Lin Hou and Hongyu Zhao
</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>

<hr>
<h2 id='EBPRSpackage'>Description of the package</h2><span id='topic+EBPRSpackage'></span>

<h3>Description</h3>

<p>Description of the package. This is the 2.0.3 version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EBPRSpackage()
</code></pre>


<h3>Details</h3>

<p>EB-PRS is a novel method that leverages information for effect sizes across all the markers to improve the prediction accuracy.  No parameter tuning is needed in the method, and no external information is needed. This R-package provides the calculation of polygenic risk scores from the given training summary statistics and test data. We can use EB-PRS to extract main information, estimate Empirical Bayes parameters, derive polygenic risk scores for  each individual in test data, and evaluate the PRS according to AUC and predictive r2.
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> EBPRS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-12</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The package contains three main functions for users,<code>read_plink</code>, <code>EBPRS</code>, and <code>validate</code>.
</p>
<p>1. <code>read_plink</code>. Thie function is used to read plink bfiles into R and reformat to suit the input of function <code>EBPRS()</code>.
</p>
<p>2. <code>EBPRS</code>. This function integrate three parts: (1) merge the train and test (if have)
data, (2) estimate effectsize (3) generate polygenic risk scores (if test data provided.)
</p>
<p>There is a strict requirement for the format of imput, which is
detailedly illustrated in details in function <code>EBPRS()</code>. The training summary statistics are necessary.
The test data can either
be included in the input or not. If test data are provided. The function will first
merge the data, as well as generate scores for each person in the result.
Users could first use the function <code>read_plink()</code> implemented in our package to read plink files into R.
</p>
<p>3. <code>validate</code>. We use this to validate the performance of the PRS.
</p>
<p>4. <code>data("traindat")</code> for the example training dataset.
</p>
<p>A complete pipeline can be:
</p>
<p>train &lt;- fread('trainpath')  (pay attention to the format, detailed in <code>EBPRS()</code>)
</p>
<p>test &lt;- read_plink('testpath')   (path to the plink bfile without extensions)
</p>
<p>result &lt;- EBPRS(train=traindat, test=plinkfile, N1, N0)
</p>
<p>validate(result$S, truey)
</p>
<p>or
</p>
<p>train &lt;- fread('trainpath')  (pay attention to the format)
</p>
<p>result &lt;- EBPRS(train=traindat, N1, N0)  (will only provide estimated effect sizes)
</p>


<h3>Author(s)</h3>

<p>Shuang Song, Wei Jiang, Lin Hou and Hongyu Zhao
</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EBPRS">EBPRS</a></code>,
<code><a href="#topic+validate">validate</a></code>,
</p>

<hr>
<h2 id='read_plink'>Read plink bfiles to R and reformat</h2><span id='topic+read_plink'></span>

<h3>Description</h3>

<p>To read plink files into R and transfer the files to the format we use in the EBPRS() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_plink(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_plink_+3A_path">path</code></td>
<td>
<p>path to the test files (plink bfiles, without extension)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input should not include the extension. For example, the test files are AA.bed,
AA.bim and AA.fam, then the input should be 'AA' instead of 'AA.bed'.
</p>


<h3>Author(s)</h3>

<p>Shuang Song, Wei Jiang, Lin Hou and Hongyu Zhao
</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EBPRS">EBPRS</a></code>
</p>

<hr>
<h2 id='traindat'>Example data for training set</h2><span id='topic+traindat'></span>

<h3>Description</h3>

<p>Summary statistics simulated in the manuscript Leveraging effect size distributions to improve polygenic risk scores derived from genome-wide association studies. Data from a QTL experiment on gravitropism in
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("traindat")
</code></pre>


<h3>Format</h3>

<p>data.frame</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("traindat")
 ## Not run: 
 result=EBPRS(train=traindat, N1=364, N0=2063)
## End(Not run)
</code></pre>

<hr>
<h2 id='validate'>Validate the performance of EBPRS</h2><span id='topic+validate'></span>

<h3>Description</h3>

<p>Provide the performance evaluated by predictive r2 and AUC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate(score, truey)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate_+3A_score">score</code></td>
<td>
<p>polygenic score generated by 'EBPRS'</p>
</td></tr>
<tr><td><code id="validate_+3A_truey">truey</code></td>
<td>
<p>the true phenotype (the 6th column in the fam file)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Shuang Song, Wei Jiang, Lin Hou and Hongyu Zhao
</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EBPRS">EBPRS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>validate(score=rnorm(20,0,1), truey=sample(0:1,20,replace=TRUE))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
