<!DOCTYPE html><html><head><title>Help for package echo.find</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {echo.find}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#echo.find'><p>echo.find: Provides a function (echo_find) designed to find rhythms from data using extended harmonic oscillators.</p></a></li>
<li><a href='#echo_find'><p>Function to calculate the results for all genes using the extended circadian harmonic oscillator (ECHO) method.</p></a></li>
<li><a href='#expressions'><p>Synthetic expression data for 12 genes.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Finding Rhythms Using Extended Circadian Harmonic Oscillators
(ECHO)</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a function (echo_find()) designed to find rhythms 
    from data using extended harmonic oscillators. For more information,
    see H. De los Santos et al. (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz617">doi:10.1093/bioinformatics/btz617</a>&gt; .</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>minpack.lm (&ge; 1.2.1), boot (&ge; 1.3-22)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/delosh653/ECHO">https://github.com/delosh653/ECHO</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, ggplot2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-10 13:45:25 UTC; delosh</td>
</tr>
<tr>
<td>Author:</td>
<td>Hannah De los Santos [aut],
  Emily Collins [aut],
  Kristin Bennett [aut],
  Jennifer Hurley [aut, cre],
  R Development Core Team [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jennifer Hurley &lt;hurlej2@rpi.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-10 19:50:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='echo.find'>echo.find: Provides a function (echo_find) designed to find rhythms from data using extended harmonic oscillators.</h2><span id='topic+echo.find'></span><span id='topic+echo.find-package'></span>

<h3>Description</h3>

<p>To read more about our inital work on this project and cite us, see Circadian Rhythms in Neurospora Exhibit Biologically Relevant Driven and Damped Harmonic Oscillations by H. De los Santos et al. (2017)
</p>

<hr>
<h2 id='echo_find'>Function to calculate the results for all genes using the extended circadian harmonic oscillator (ECHO) method.</h2><span id='topic+echo_find'></span>

<h3>Description</h3>

<p>Function to calculate the results for all genes using the extended circadian harmonic oscillator (ECHO) method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>echo_find(genes, begin, end, resol, num_reps, low = 1, high = 2,
  run_all_per, paired, rem_unexpr, rem_unexpr_amt = 70,
  rem_unexpr_amt_below = 0, is_normal, is_de_linear_trend, is_smooth,
  run_conf = F, which_conf = "Bootstrap", harm_cut = 0.03,
  over_cut = 0.15, seed = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="echo_find_+3A_genes">genes</code></td>
<td>
<p>data frame of genes with the following specifications: first row is column labels, first column has gene labels/names, and all other columns have expression data. This expression data must be ordered by time point then by replicate, and must have evenly spaced time points. Any missing data must have cells left blank.</p>
</td></tr>
<tr><td><code id="echo_find_+3A_begin">begin</code></td>
<td>
<p>first time point for dataset</p>
</td></tr>
<tr><td><code id="echo_find_+3A_end">end</code></td>
<td>
<p>last time point for dataset</p>
</td></tr>
<tr><td><code id="echo_find_+3A_resol">resol</code></td>
<td>
<p>resolution of time points</p>
</td></tr>
<tr><td><code id="echo_find_+3A_num_reps">num_reps</code></td>
<td>
<p>number of replicates</p>
</td></tr>
<tr><td><code id="echo_find_+3A_low">low</code></td>
<td>
<p>lower limit when looking for rhythms, in hours. May be unused if finding rhythms of any length within timecouse (run_all_per is TRUE).</p>
</td></tr>
<tr><td><code id="echo_find_+3A_high">high</code></td>
<td>
<p>upper limit when looking for rhythms, in hours. May be unused if finding rhythms of any length within timecouse (run_all_per is TRUE).</p>
</td></tr>
<tr><td><code id="echo_find_+3A_run_all_per">run_all_per</code></td>
<td>
<p>boolean which indicates whether or not rhythms of any length within timecourse should be searched for.</p>
</td></tr>
<tr><td><code id="echo_find_+3A_paired">paired</code></td>
<td>
<p>if replicate data, whether the replicates are related (paired) or not (unpaired)</p>
</td></tr>
<tr><td><code id="echo_find_+3A_rem_unexpr">rem_unexpr</code></td>
<td>
<p>boolean indicating whether genes with less than rem_unexpr_amt percent expression should not be considered</p>
</td></tr>
<tr><td><code id="echo_find_+3A_rem_unexpr_amt">rem_unexpr_amt</code></td>
<td>
<p>percentage of expression for which genes should not be considered if rem_unexpr is TRUE</p>
</td></tr>
<tr><td><code id="echo_find_+3A_rem_unexpr_amt_below">rem_unexpr_amt_below</code></td>
<td>
<p>cutoff for expression</p>
</td></tr>
<tr><td><code id="echo_find_+3A_is_normal">is_normal</code></td>
<td>
<p>boolean that indicates whether data should be normalized or not</p>
</td></tr>
<tr><td><code id="echo_find_+3A_is_de_linear_trend">is_de_linear_trend</code></td>
<td>
<p>boolean that indicates whether linear trends should be removed from data or not</p>
</td></tr>
<tr><td><code id="echo_find_+3A_is_smooth">is_smooth</code></td>
<td>
<p>boolean that indicates whether data should be smoothed or not</p>
</td></tr>
<tr><td><code id="echo_find_+3A_run_conf">run_conf</code></td>
<td>
<p>boolean of whether or not to run confidence intervals</p>
</td></tr>
<tr><td><code id="echo_find_+3A_which_conf">which_conf</code></td>
<td>
<p>string of which type of confidence interval to compute (&quot;Bootstrap&quot; or &quot;Jackknife&quot;)</p>
</td></tr>
<tr><td><code id="echo_find_+3A_harm_cut">harm_cut</code></td>
<td>
<p>postive number indicating the cutoff for a gene to be considered harmonic</p>
</td></tr>
<tr><td><code id="echo_find_+3A_over_cut">over_cut</code></td>
<td>
<p>postive number indicating the cutoff for a gene to be considered repressed/overexpressed</p>
</td></tr>
<tr><td><code id="echo_find_+3A_seed">seed</code></td>
<td>
<p>number for random seed to fix for bootstrapping for confidence intervals</p>
</td></tr>
</table>


<h3>Value</h3>

<p>results, a data frame which contains:
</p>
<table>
<tr><td><code>Gene Name</code></td>
<td>
<p>gene name</p>
</td></tr>
<tr><td><code>Convergence</code></td>
<td>
<p>depreciated result, always 0, will be removed in future versions</p>
</td></tr>
<tr><td><code>Iterations</code></td>
<td>
<p>depreciated result, always 0, will be removed in future versions</p>
</td></tr>
<tr><td><code>Amplitude.Change.Coefficient</code></td>
<td>
<p>Amplitude change coefficient value for fit</p>
</td></tr>
<tr><td><code>Oscillation Type</code></td>
<td>
<p>Type of oscillation (damped, driven, etc.)</p>
</td></tr>
<tr><td><code>Initial.Amplitude</code></td>
<td>
<p>Initial amplitude value for fit</p>
</td></tr>
<tr><td><code>Radian.Frequency</code></td>
<td>
<p>Radian frequency for fit</p>
</td></tr>
<tr><td><code>Period</code></td>
<td>
<p>Period for fit (in time units)</p>
</td></tr>
<tr><td><code>Phase Shift</code></td>
<td>
<p>Phase shift for fit (radians)</p>
</td></tr>
<tr><td><code>Hours Shifted</code></td>
<td>
<p>Phase shift for fit (hours)</p>
</td></tr>
<tr><td><code>Equilibrium Value</code></td>
<td>
<p>Equilibrium shift for fit</p>
</td></tr>
<tr><td><code>Slope</code></td>
<td>
<p>Slope value of original data, if linear baseline is removed</p>
</td></tr>
<tr><td><code>Tau</code></td>
<td>
<p>Kendall's tau between original and fitted values</p>
</td></tr>
<tr><td><code>P-value</code></td>
<td>
<p>P-value calculated based on Kendall's tau</p>
</td></tr>
<tr><td><code>BH Adj P-Value</code></td>
<td>
<p>Benjamini-Hochberg adjusted p-values</p>
</td></tr>
<tr><td><code>BY Adj P-Value</code></td>
<td>
<p>Benjamini-Yekutieli adjusted p-values</p>
</td></tr>
<tr><td><code>CI.PARAM.Low</code></td>
<td>
<p>Lower confidence interval bound for all parameters, if calculated</p>
</td></tr>
<tr><td><code>CI.PARAM.High</code></td>
<td>
<p>Higher confidence interval bound for all parameters, if calculated</p>
</td></tr>
<tr><td><code>Original TPX.Y</code></td>
<td>
<p>Processed values for gene expression at time point X, replicate Y</p>
</td></tr>
<tr><td><code>Fitted TPX</code></td>
<td>
<p>Fitted values for gene expression at time point X</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># for more elaboration, please see the vignette
# "expressions" is the example echo.find data frame
 # long example - commented out
echo_find(genes = expressions, begin = 2, end = 48, resol = 2,
  num_reps = 3, low = 20, high = 26, run_all_per = FALSE,
  paired = FALSE, rem_unexpr = FALSE, rem_unexpr_amt = 70, rem_unexpr_amt_below=0,
  is_normal = FALSE, is_de_linear_trend = FALSE, is_smooth = FALSE)

</code></pre>

<hr>
<h2 id='expressions'>Synthetic expression data for 12 genes.</h2><span id='topic+expressions'></span>

<h3>Description</h3>

<p>A dataset containing the names and expression values for 12 synthetically
generated samples. This example data has time points from 2 to 48 hours with
2 hour resolution and 3 replicates. Random missing data is also included.
Synthetic data was created by randomly selecting parameters for the
extended harmonic oscillator equation (see journal paper link in vignette
for the equation), then adding random uniform noise to each expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expressions
</code></pre>


<h3>Format</h3>

<p>A data frame with 12 rows and 73 variables (column 1: sample labels, columns to 2 to 73: numerical values for gene expression in the forsmat CTX.Y (time point X, replicate Y)).</p>


<h3>Details</h3>

<p>Note the data format: its first column first column has gene labels/names, and
all other columns have expression data. This expression data is ordered by
time point then by replicate, and has evenly spaced time points. Any missing
data has cells left blank.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
