<!DOCTYPE html><html><head><title>Help for package EcoIndR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EcoIndR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DER'><p>DIFFERENTIATING SAMPLES USING RARITY, HETEROGENEITY, EVENNESS, TAXONOMIC, PHYLOGENETIC AND FUNCTIONAL DIVERSITY INDICES</p></a></li>
<li><a href='#EnVarIP'><p>ENVIRONMENTAL VARIABLES IN THE IBERIAN PENINSULA</p></a></li>
<li><a href='#Rotifers'><p>ABUNDANCE OF ROTIFERS IN PONDS</p></a></li>
<li><a href='#SINENVAP'><p>SPATIAL INTERPOLATION OF VARIABLES WITHIN POLYGONS</p></a></li>
<li><a href='#VirtualSpecies'><p>PROBABILITY OF A VIRTUAL SPECIES</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Ecological Indicators</td>
</tr>
<tr>
<td>Author:</td>
<td>Castor Guisande Gonzalez</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Castor Guisande Gonzalez &lt;castor@uvigo.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculates several indices, such as of diversity, fluctuation, etc., and they are used to estimate ecological indicators.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1), picante, Rarity, vegan, splancs, FD</td>
</tr>
<tr>
<td>Suggests:</td>
<td>shape, plotrix, sp, geoR, mgcv, raster, alphahull, sf,
methods</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-06 20:06:15 UTC; castor</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-08 00:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='DER'>DIFFERENTIATING SAMPLES USING RARITY, HETEROGENEITY, EVENNESS, TAXONOMIC, PHYLOGENETIC AND FUNCTIONAL DIVERSITY INDICES</h2><span id='topic+DER'></span>

<h3>Description</h3>

<p>An algorithm for differentiating samples on the basis of the rarity, heterogeneity, evenness, taxonomic/phylogenetic and functional diversity indices that better reflect the differences among assemblages.</p>


<h3>Usage</h3>

<pre><code class='language-R'>DER(data, Samples, Species, Taxon, TaxonFunc=NULL, TaxonPhyl=NULL,
pos=NULL, varSize="Richness", varColor="Rarity.G", Index=NULL, corr="sqrt",
palette= "heat.colors", size=c(1,5),digitsS=1, digitsC=2, ncolor=100,
transparency=1, references=TRUE, a=1.5, q=2.5, ResetPAR=TRUE, PAR=NULL,
dbFD=NULL, LEGENDS=NULL, TEXT=NULL, COLOR=c("#EEC591FF", "black", "grey50"),
file1="Diversity indices.csv", file2="Polar coordinates.csv",
file3="Indices and area of the polygon.csv", na="NA", dec=",", row.names=FALSE, save=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DER_+3A_data">data</code></td>
<td>
<p>Data file with the taxonomy, abundance of the species and functional traits (optional). The format of the file must be: an optional column with the position of labels' samples in the DER plot (blue column) in the same order as the variables with the species' abundance in the samples (red columns), the columns with the taxonomy of the species (as many as needed, green columns), the columns with the abundance of the species in each sample (red columns) and optionally the columns with the functional traits of the species. Each row is an unique species, genus, family, etc.
</p>

<table>
<tr>
 <td style="text-align: right;">
<img src="../help/figures/EcoIndRF1.jpg" width="50" alt="EcoIndRF1.jpg" />

</td>
</tr>

</table>

</td></tr>
<tr><td><code id="DER_+3A_samples">Samples</code></td>
<td>
<p>Variables with the abundance of the species in each sample: sampling sites, dates, etc.</p>
</td></tr>
<tr><td><code id="DER_+3A_species">Species</code></td>
<td>
<p>Variable with the name of the species (without including the genus). It may be other node of the phylogenetic tree, such as the genus, family, etc., for genus level phylogenies, family level phylogenies, etc.</p>
</td></tr>
<tr><td><code id="DER_+3A_taxon">Taxon</code></td>
<td>
<p>Variables with the taxonomy of the species (taxonomic diversity), as many levels as needed but without including the variable with the node of the argument <em>Species</em>.</p>
</td></tr>
<tr><td><code id="DER_+3A_taxonfunc">TaxonFunc</code></td>
<td>
<p>Optionally variables with the functional traits (functional diversity).</p>
</td></tr>
<tr><td><code id="DER_+3A_taxonphyl">TaxonPhyl</code></td>
<td>
<p>Optionally the name of the RData file of the class phylo with the phylogeny. The file must be in the working directory.</p>
</td></tr>
<tr><td><code id="DER_+3A_pos">pos</code></td>
<td>
<p>Optionally it is possible to indicate a column with the position of labels' samples in the DER plot. It must be as many as the number of samples and in the same order than the variables described in the argument <em>Samples</em>. Values of 1, 2, 3 and 4, respectively indicate positions below, to the left of, above and to the right of the specified coordinates.</p>
</td></tr>
<tr><td><code id="DER_+3A_varsize">varSize</code></td>
<td>
<p>This variable defines the size of the bubble in the DER plot.</p>
</td></tr>
<tr><td><code id="DER_+3A_varcolor">varColor</code></td>
<td>
<p>This variable defines the color gradient of the bubbles in the DER plot.</p>
</td></tr>
<tr><td><code id="DER_+3A_index">Index</code></td>
<td>
<p>The four/five indices used in the DER algorithm. If it is NULL the algorithm select one, index of rarity, one of heterogeneity, one of evenness one of taxonomy and one of the functional group (if functional traits are provided in the argument <em>TaxonFunc</em>) that achieve a higher dispersion among samples in a polar coordinates system.</p>
</td></tr>
<tr><td><code id="DER_+3A_corr">corr</code></td>
<td>
<p>Character string specifying the correction method to use, in the function <a href="FD.html#topic+dbFD">dbFD</a>, when the species-by-species distance matrix cannot be represented in a Euclidean space. Options are &quot;sqrt&quot; (default), &quot;cailliez&quot;, &quot;lingoes&quot; or &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="DER_+3A_palette">palette</code></td>
<td>
<p>The color gradient of the bubbles may be one of these palettes: &quot;heat.colors&quot;, &quot;terrain.colors&quot;, &quot;gray.colors&quot;, &quot;topo.colors&quot; or &quot;cm.colors&quot;, or any other option defined by the user.</p>
</td></tr>
<tr><td><code id="DER_+3A_size">size</code></td>
<td>
<p>Range of size of the bubbles. Two values: minimum and maximum size.</p>
</td></tr>
<tr><td><code id="DER_+3A_digitss">digitsS</code></td>
<td>
<p>Number of digits of the bubble size legend.</p>
</td></tr>
<tr><td><code id="DER_+3A_digitsc">digitsC</code></td>
<td>
<p>Number of digits of the color legend.</p>
</td></tr>
<tr><td><code id="DER_+3A_ncolor">ncolor</code></td>
<td>
<p>Gradient color of the color legend.</p>
</td></tr>
<tr><td><code id="DER_+3A_transparency">transparency</code></td>
<td>
<p>Transparency of the color gradient, from 0 to 1.</p>
</td></tr>
<tr><td><code id="DER_+3A_references">references</code></td>
<td>
<p>If it is TRUE the reference points are depicted on the DER plot.</p>
</td></tr>
<tr><td><code id="DER_+3A_a">a</code></td>
<td>
<p>Scale of Renyi diversity.</p>
</td></tr>
<tr><td><code id="DER_+3A_q">q</code></td>
<td>
<p>Scale of Tsallis diversity.</p>
</td></tr>
<tr><td><code id="DER_+3A_resetpar">ResetPAR</code></td>
<td>
<p>If it is FALSE, the default condition of the function PAR of the package StatR is not placed and maintained those defined by the user in previous graphics.</p>
</td></tr>
<tr><td><code id="DER_+3A_par">PAR</code></td>
<td>
<p>It accesses the function PAR of the package StatR that allows to modify many different aspects of the graph.</p>
</td></tr>
<tr><td><code id="DER_+3A_dbfd">dbFD</code></td>
<td>
<p>It accesses the function <a href="FD.html#topic+dbFD">dbFD</a> which allows to specify the arguments that calculates the functional diversity indices.</p>
</td></tr>
<tr><td><code id="DER_+3A_legends">LEGENDS</code></td>
<td>
<p>It allows to modify the legend of the bubble size.</p>
</td></tr>
<tr><td><code id="DER_+3A_text">TEXT</code></td>
<td>
<p>It allows to modify the text of the labels in the bubbles.</p>
</td></tr>
<tr><td><code id="DER_+3A_color">COLOR</code></td>
<td>
<p>A vector with three values: color of the ellipse, color of the points in the legend of the size of the bubbles and color of the references points in the ellipse, respectively.</p>
</td></tr>
<tr><td><code id="DER_+3A_file1">file1</code></td>
<td>
<p>CSV FILES. Filename with values of total abundance, richness and the rarity, heterogeneity, evenness, taxonomic, phylogenetic and functional diversity indices of each sample.</p>
</td></tr>
<tr><td><code id="DER_+3A_file2">file2</code></td>
<td>
<p>CSV FILES. Filename with the polar coordinates of all samples considering the four/five selected indices.</p>
</td></tr>
<tr><td><code id="DER_+3A_file3">file3</code></td>
<td>
<p>CSV FILES. Filename with the area of the convex hull (alpha=6) and Euclidean distance obtained in the polar coordinates system for all combinations of the indices.</p>
</td></tr>
<tr><td><code id="DER_+3A_na">na</code></td>
<td>
<p>CSV FILE. Text that is used in the cells without data.</p>
</td></tr>
<tr><td><code id="DER_+3A_dec">dec</code></td>
<td>
<p>CSV FILE. It defines if the comma &quot;,&quot; is used as decimal separator or the dot &quot;.&quot;.</p>
</td></tr>
<tr><td><code id="DER_+3A_row.names">row.names</code></td>
<td>
<p>CSV FILE. Logical value that defines if identifiers are put in rows or a vector with a text for each of the rows.</p>
</td></tr>
<tr><td><code id="DER_+3A_save">save</code></td>
<td>
<p>If it is TRUE, the CSV files are saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>DER algorithm</strong>
</p>
<p>The steps of DER algorithm are described below:
</p>
<p><strong>1.</strong> The function DER calculates the most often used indices (see below): a total of 39 indices that includes 2 of rarity, 14 of heterogeneity, 7 of evenness, 2 of taxonomic diversity, 8 of phylogenetic diversity and 6 of functional diversity. It is important to mention that the indices included in the groups of phylogenetic diversity and functional diversity, each explores a different facet of phylogenetic diversity (Kembel et al., 2010) and functional diversity (Laliberte et al., 2010), respectively.
</p>
<p><em>Rarity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species (species richness) in the sample, <em>s</em> is the number of samples, <code class="reqn">p_{ij}</code> is the presence or absence (0 or 1) of the species <em>i</em> in the sample <em>j</em>, <code class="reqn">r_{ij}</code> is the number of records or abundance of the species <em>i</em> in the sample <em>j</em>, <code class="reqn">R_{i}</code> is the total number of records or total abundance of the species <em>i</em> considering all samples, <code class="reqn">Q_i</code> is the occurrence of species <em>i</em>, <code class="reqn">Q_{min}</code> and <code class="reqn">Q_{max}</code> are respectively the minimum and maximum occurrences in the species pool, <em>r</em> is the chosen rarity cutoff point (as a percentage of occurrence), <code class="reqn">w_i</code> is the weight of the <em>ith</em> species in the assemblage, <code class="reqn">w_{min}</code> and <code class="reqn">w_{max}</code> the minimum and maximum weights respectively.
</p>
<p><em>Leroy</em> (Leroy et al., 2012; 2013)
</p>
<p style="text-align: center;"><code class="reqn">w_i=\frac{1}{e^{(\frac{Q_i-Q_{min}}{rQ_{max}Q_{min}}*0.97+1.05)^2}}</code>
</p>

<p style="text-align: center;"><code class="reqn">I_{RR}=\frac{\frac{\displaystyle\sum_{i=1}^{S}w_i}{S}-w_{min}}{w_{max}-w_{min}}</code>
</p>

<p><em>Geographical rarity</em> This index is a novel contribution of this package.
</p>
<p style="text-align: center;"><code class="reqn">GR=1-\frac{\displaystyle\sum_{i=1}^{S}\frac{\displaystyle\sum_{j=1}^{s}p_{ij}}{s}}{S}</code>
</p>

<p><em>Occurrential rarity</em> This index is a novel contribution of this package.
</p>
<p style="text-align: center;"><code class="reqn">OR=1-\frac{\displaystyle\sum_{i=1}^{S}\frac{\displaystyle\sum_{j=1}^{s}r_{ij}}{R_{i}}}{S}</code>
</p>

<p><em>Heterogeneity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species (species richness) in the sample, <code class="reqn">p_i</code> is the abundace's proportion of species <em>i</em>, <em>N</em> is the total number of individuals in the sample, <code class="reqn">n_i</code> is the number of individuals of the species <em>i</em>, <code class="reqn">n_{max}</code> is the number of individuals of the most abundant species, <em>a</em> and <em>q</em> are the orders of Renyi and Tsallis indices respectively and, finally, in Fisher's alpha the index is the <code class="reqn">\alpha</code> parameter.
</p>
<p><em>log Shannon-Wiener (S.W.LOG2)</em> and <em>ln Shannon-Wiener (S.W)</em> (Wiener, 1939; 1948; 1949; Shannon, 1948; Shannon &amp; Weaver, 1949). See Spellerger &amp; Fedor (2013) for an explanation of the dual use of the terms Shannon-Wiener and Shannon-Weaver to refer to this diversity index.
</p>
<p style="text-align: center;"><code class="reqn">H=-\displaystyle\sum_{i=1}^{S}p_ilog_2p_i</code>
</p>

<p style="text-align: center;"><code class="reqn">H'=-\displaystyle\sum_{i=1}^{S}p_ilnp_i</code>
</p>

<p><em>Fisher's alpha</em> (Fisher et al., 1943)
</p>
<p style="text-align: center;"><code class="reqn">{\alpha}x,\frac{{\alpha}x^2}{2},\frac{{\alpha}x3}{3},.....,\frac{{\alpha}x^n}{n}</code>
</p>

<p><em>Simpson</em> (Simpson, 1949) 
</p>
<p style="text-align: center;"><code class="reqn">D_1=1-\displaystyle\sum_{i=1}^{S}p_i^2</code>
</p>

<p><em>Inverse Simpson (InvSimpson)</em> Williams (1964)
</p>
<p style="text-align: center;"><code class="reqn">D_2=\frac{1}{\displaystyle\sum_{i=1}^{S}p_i^2}</code>
</p>

<p><em>Brillouin</em> (Brillouin, 1956)
</p>
<p style="text-align: center;"><code class="reqn">H_B=\frac{lnN!-\displaystyle\sum_{i=1}^{S}lnn_i!}{N}</code>
</p>

<p><em>Margalef</em> (Margalef, 1959)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mg}=\frac{(S-1)}{lnN}</code>
</p>

<p><em>Renyi entropy</em> (Renyi, 1961) 
</p>
<p style="text-align: center;"><code class="reqn">H_a=\frac{1}{(1-a)}log\displaystyle\sum_{i=1}^{S}p_i^a</code>
</p>
 
<p><em>Menhinick</em> (Menhinick, 1964)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mn}=\frac{S}{\sqrt{N}}</code>
</p>

<p><em>McIntosh</em> (McIntosh, 1967)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mc}=\frac{N-\sqrt{\displaystyle\sum_{i=1}^{S}n_i^2}}{N-\sqrt{N}}</code>
</p>

<p><em>Inverse Berger-Parker (InvB.P)</em> (Berger &amp; Parker, 1970)
</p>
<p style="text-align: center;"><code class="reqn">D_{BP}=\frac{1}{\frac{n_{max}}{N}}</code>
</p>

<p><em>Hill numbers</em> (Hill, 1973)
</p>
<p>Hill-Renyi
</p>
<p style="text-align: center;"><code class="reqn">N_a=e^{H_a}</code>
</p>

<p>Hill-Tsallis
</p>
<p style="text-align: center;"><code class="reqn">N_q=(1-(q-1)H_q)^\frac{1}{(1-q)}</code>
</p>

<p>where
</p>
<p>a or q = 0 is species richness
</p>
<p>a or q = 1 is Shannon's index (H')
</p>
<p>a or q = 2 is Inverse Simpson's index (<code class="reqn">D_2</code>)
</p>
<p>a or q = Inf is Inverse Berger-Parker index (<code class="reqn">D_{BP}</code>)
</p>
<p><em>Tsallis entropy</em> (Patil &amp; Taillie, 1982; Tsallis, 1988)
</p>
<p style="text-align: center;"><code class="reqn">H_q=\frac{1}{(q-1)}(1-\displaystyle\sum_{i=1}^{S}p_i^q)</code>
</p>

<p><em>Evenness indices</em>
</p>
<p>Annotations of the equations as mentioned for heterogeneity indices.
</p>
<p><em>Simpson evenness (SimpsonE)</em> (Simpson, 1949) 
</p>
<p style="text-align: center;"><code class="reqn">E_{D1}=\frac{D_1}{S}</code>
</p>

<p><em>Pielou (PielouE)</em> (Pielou, 1966)
</p>
<p style="text-align: center;"><code class="reqn">J'=\frac{H'}{lnS}</code>
</p>

<p><em>McIntosh evenness (McIntoshE)</em> (McIntosh, 1967)
</p>
<p style="text-align: center;"><code class="reqn">E_{Mc}=\frac{N-\sqrt{\displaystyle\sum_{i=1}^{S}n_i^2}}{N-\frac{N}{\sqrt{S}}}</code>
</p>

<p><em>Hill evenness (HillE)</em> (Hill, 1973)
</p>
<p>It is used Hill-Renyi numbers where in <code class="reqn">N_2</code> the value of a = 2 and in <code class="reqn">N_2</code> the value of a = 1
</p>
<p style="text-align: center;"><code class="reqn">E_{2,1}=\frac{N_2}{N_1}</code>
</p>

<p><em>Heip evenness (HeipE)</em> (Heip, 1974)
</p>
<p style="text-align: center;"><code class="reqn">E_{Heip}=\frac{e^{H'}-1}{S-1}</code>
</p>

<p><em>Camargo (CamargoE)</em> (Camargo, 1992)
</p>
<p style="text-align: center;"><code class="reqn">E_{Camargo}=1-\displaystyle\sum_{i=1}^{S}\displaystyle\sum_{j=i'+1}^{S}\frac{|p_i-p_j|}{s}</code>
</p>

<p><em>Smith and Wilson's Index (Evar)</em> (Smith and Wilson, 1996)
</p>
<p style="text-align: center;"><code class="reqn">E_{var}=1-\left(\frac{2}{\pi}\right)\left(arctan\left(\frac{\displaystyle\sum_{i=1}^{S}{\left(lnn_i-\frac{\displaystyle\sum_{j=1}^{S}n_j}{S}\right)}^2}{S}\right)\right)</code>
</p>

<p><em>Taxonomic diversity indices</em>
</p>
<p>In the following equations summation goes over species <em>i</em> and <em>j</em>, <code class="reqn">\omega</code> are the taxonomic distances among taxa, <em>x</em> are species abundances, and <em>n</em> is the total abundance for a site.
</p>
<p><em>Taxonomic diversity (D)</em> (Clarke, 1995; 1998; 2001):
</p>
<p style="text-align: center;"><code class="reqn">\Delta=\frac{\sum\sum_{i&lt;j}\omega_{ij}x_ix_j}{n(n-1)/2}</code>
</p>

<p><em>Taxonomic distinctness (Dstar)</em> (Clarke and Warwick, 1998):
</p>
<p style="text-align: center;"><code class="reqn">\Delta^*=\frac{\sum\sum_{i&lt;j}\omega_{ij}x_ix_j}{\sum\sum_{i&lt;j}x_ix_j}</code>
</p>

<p><em>Phylogenetic diversity indices</em>
</p>

<table>
<tr>
 <td style="text-align: left;">
Faith's phylogenetic diversity </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Faith (1992)</td>
</tr>
<tr>
 <td style="text-align: left;">
Mean pairwise phylogenetic distance </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Webb et al. (2008)</td>
</tr>
<tr>
 <td style="text-align: left;">
Mean nearest taxon distance </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Webb et al. (2008)</td>
</tr>
<tr>
 <td style="text-align: left;">
Phylogenetic species richness </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
 <td style="text-align: left;">
Phylogenetic species variability </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
 <td style="text-align: left;">
Phylogenetic species evenness </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
 <td style="text-align: left;">
Phylogenetic species clustering </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
 <td style="text-align: left;">
Quadratic entropy </td><td style="text-align: left;">  </td><td style="text-align: left;">   </td><td style="text-align: left;"> Rao (1982)
</td>
</tr>

</table>

<p><em>Functional diversity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species, where <code class="reqn">d_{ij}</code> is the difference between the <em>i-th</em> and <em>j-th</em> species, <code class="reqn">p_i</code> is the abundance's proportion of the species <em>i</em>, <code class="reqn">p_j</code> is the abundance's proportion of the species <code class="reqn">j</code>, <em>EW</em> is weighted evenness, dist(<em>i</em>, <em>j</em>) is the Euclidean distance between species <em>i</em> and <em>j</em>, the species involved is branch <em>l</em>, <code class="reqn">w_i</code> is the relative abundance of the species <em>i</em>, <code class="reqn">PEW_l</code> is the partial weighted evenness, <code class="reqn">g_k</code> are the coordinates of the center of gravity of the <em>V</em> species forming the vertices of the convex hull, <code class="reqn">x_{ik}</code> is the coordinate of species <em>i</em> on trait <em>k</em>, <code class="reqn">dG_i</code> is the Euclidean distance of the center of gravity, <code class="reqn">\bar{dG}</code> is the mean distance of the <em>S</em> species to the center of gravity, <code class="reqn">\Delta</code><em>d</em> is the sum of abundance-weighted deviances and <code class="reqn">\Delta</code><em>|d|</em> absolute abundance-weighted deviances.
</p>
<p><em>Rao's quadratic entropy</em> (Rao, 1982; Botta-Dukat, 2005):
</p>
<p style="text-align: center;"><code class="reqn">Q=\displaystyle\sum_{i=1}^{S-1}\displaystyle\sum_{j=i+1}^{S-1}d_{ij}p_ip_j</code>
</p>

<p style="text-align: center;"><code class="reqn">d_{ij}=\displaystyle\sum_{k=1}^{n}\displaystyle\sum_{l=1}^{n}w_{kl}(X_{ik}-X_{jk})(X_{il}-X_{jl})</code>
</p>

<p><em>Functional group richness (FGR)</em> Petchey and Gaston (2006)
</p>
<p><em>Functional richness (FRic)</em> is measured as the amount of functional space (convex hull volume) filled by the community (Villeger et al., 2008).
</p>
<p><em>Functional evenness (FEve)</em> (Villeger et al., 2008):
</p>
<p style="text-align: center;"><code class="reqn">FEve=\frac{\displaystyle\sum_{l=1}^{S-1}min(PEW_l,\frac{1}{S-1})-\frac{1}{S-1}}{1-\frac{1}{S-1}}</code>
</p>


<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">PEW_l=\frac{EW_l}{\displaystyle\sum_{l=1}^{S}EW_l}</code> </td><td style="text-align: left;"> <code class="reqn">EW_l=\frac{dis(i,j)}{w_i+w_j}</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><em>Functional divergence (FDiv)</em> (Villeger et al., 2008):
</p>
<p style="text-align: center;"><code class="reqn">FDiv=\frac{\Delta{d}+\bar{dG}}{\Delta{|d|}+\bar{dG}}</code>
</p>


<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">\Delta{|d|}=\displaystyle\sum_{i=1}^{S}w_i|dG_i-\bar{dG}|</code> </td><td style="text-align: left;"> <code class="reqn">\Delta{d}=\displaystyle\sum_{i=1}^{S}w_i(dG_i-\bar{dG})</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">\bar{dG}=\frac{1}{S}\displaystyle\sum_{i=1}^{S}dG_i</code> </td><td style="text-align: left;"> <code class="reqn">dG_i=\sqrt{\displaystyle\sum_{k=1}^{T}(x_{ik}-g_k)^2}</code> </td><td style="text-align: left;"> <code class="reqn">g_k=\frac{1}{V}\displaystyle\sum_{i=1}^{V}x_{ik}</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><em>Functional dispersion (FDis)</em> (Laliberte and Legendre, 2010):
</p>

<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">FDisp=\frac{\sum{a_jz_j}}{\sum{a_j}}</code> </td><td style="text-align: left;"> <code class="reqn">c=\frac{\sum{a_jx_{ij}}}{\sum{a_j}}</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>2.</strong> Each index is transformed to a scale range between 0 and 1 for all samples with the following equation:
</p>
<p style="text-align: center;"><code class="reqn">\frac{(index of the sample-min)}{(max-min)}</code>
</p>

<p>where <em>min</em> and <em>max</em> are the minimum and maximum values of the index considering all samples, respectively.
</p>
<p><strong>3.</strong> With the standardized values of the indices, the algorithm calculates the polar coordinates of all samples with all possible combinations among all groups of indices. Therefore, in each combination an index of each group of rarity, heterogeneity (species richness is included in this group), evenness, taxonomic/ phylogenetic diversity and functional diversity (if it is included functional traits in the analysis) is used for calculating the polar coordinates of all samples. In the group of taxonomic/phylogenetic diversity the user must use either taxonomy or a phylogenetic tree, so either taxonomic diversity or phylogenetic diversity indices are used in the algorithm. The X and Y polar coordinates for each sample are estimated using the following equations:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code class="reqn">X=\displaystyle\sum_{i=1}^{4}|z_j|cos(\alpha)</code> </td><td style="text-align: left;"> <code class="reqn">Y=\displaystyle\sum_{i=1}^{4}|z_j|sin(\alpha)</code>
</td>
</tr>

</table>

<p>where <em>z</em> is the standardized value of the index <em>j</em> of the four groups considered.
</p>
<p>Each index is assigned an angle (<code class="reqn">\alpha</code>). Degrees to radians angle conversion is carried out assuming that 1 degree = 0.0174532925 radians.
</p>
<p><strong>4.</strong> With the polar coordinates of the samples obtained for each combination, it is calculated the convex hull (alpha = 6) and the mean Euclidean distance, and the values are saved in a file.
</p>
<p><strong>5.</strong> The algorithm selects the combination of indices with the highest value of the mean between convex hull and mean Euclidean distance among samples, therefore priority is given to maximize dispersion among samples (see Fig. 1). The polar coordinates of the selected combination are depicted on a diagram, where it is possible to see the differences in rarity, heterogeneity, evenness and taxonomic/phylogenetic  diversity and/or functional diversity (if it is included) among assemblages.
</p>
<p><strong>6.</strong> Finally, DER function allows the user to select the four/five indices to be used in the diagram, so the algorithm of selecting the combination with the maximum dispersion among samples is not applied.
</p>
<p><strong>FUNCTIONS</strong>
</p>
<p>The index Fisher alpha was estimated with the function <a href="vegan.html#topic+fisher.alpha">fisher.alpha</a>, the index Renyi with the function <a href="vegan.html#topic+renyi">renyi</a>, the index Tsallis with the function <a href="vegan.html#topic+tsallis">tsallis</a>, the taxonomic diversity and taxonomic distinctness with the functions <a href="vegan.html#topic+taxa2dist">taxa2dist</a> and <a href="vegan.html#topic+taxondive">taxondive</a>, all of them of the package vegan (Oksanen et al., 2016). The ellipse is depicted with the function <a href="shape.html#topic+plotellipse">plotellipse</a> of the package shape (Soetaert, 2016). The convex hull (alpha=6) was calculated with the function <a href="splancs.html#topic+areapl">areapl</a> of the package splancs (Bivand et al., 2016). The color legend of DER plot was depicted with the function <a href="plotrix.html#topic+color.legend">color.legend</a> of the package plotrix (Lemon et al., 2016). The rarity index of Leroy was calculated with the functions <a href="Rarity.html#topic+rWeights">rWeights</a> and <a href="Rarity.html#topic+Irr">Irr</a>, both of the package Rarity (Leroy et al., 2012; 2103; Leroy, 2016). The functional diversity indices were calculated with the function <a href="FD.html#topic+dbFD">dbFD</a> of the package FD (Laliberte et al., 2015). The phylogenetic indices were calculated with the functions psv, psr, pse, psc, raoD, mntd, mpd and pd of the package picante (Kembel et al. 2010 2016)
</p>
<p><strong>EXAMPLE</strong>
</p>
<p>The example without functional diversity is a dataset with the abundance of rotifers species in ponds (see table 1 in Mazuelos et al., 1993). In the argument <em>Index</em> were selected Rarity, Menhinick, McIntoshE and Dstar, which are the indices selected by the algorithm when <em>Index=NULL</em> (default option). The sample G3.1 had the lowest values of the indices of rarity, heterogeneity, evenness and taxonomic diversity and the pond I3.1 the highest values for all indices.
</p>

<table>
<tr>
 <td style="text-align: center;">
<img src="../help/figures/EcoIndRF2.jpg" width="50" alt="EcoIndRF2.jpg" />

</td>
</tr>

</table>



<h3>Value</h3>

<p>It is depicted a plot of polar coordinates estimated with the rarity, heterogeneity, evenness, taxonomic/phylogenetic and functional diversity indices, CSV files are saved with all the indices, the polar coordinates estimated with the indices specified in the argument <em>Index</em> or estimated by the algorithm, and the area of the convex hull and mean Euclidean distance obtained in the polar coordinates system for all combinations of the indices.</p>


<h3>References</h3>

<p>Berger, W.H., Parker, F.L. (1970) Diversity of planktonic Foramenifera in deep sea sediments. <em>Science</em>, 168: 1345-1347.
</p>
<p>Bivand, R., Rowlingson, B., Diggle, P., Petris, G., Eglen, S. (2016) Spatial and Space-Time Point Pattern Analysis. R Package Version 2.01-39. <a href="https://CRAN.R-project.org/package=splancs">https://CRAN.R-project.org/package=splancs</a>
</p>
<p>Botta-Dukat, Z. (2005) Rao's quadratic entropy as a measure of functional diversity based on multiple traits. <em>Journal of Vegetation Science</em>, 16: 533-540.
</p>
<p>Brillouin, L. (1956) <em>Science and information theory</em>. New York: Academic Press.
</p>
<p>Camargo, J.A. (1992) New diversity index for assessing structural alterations in aquatic communities. <em>Bulletin of Environmental Contamination and Toxicology</em>, 48, 428-434.
</p>
<p>Clarke, K.R. &amp; Warwick, R.M. (1998). A taxonomic distinctness index and its statistical properties. <em>Journal of Applied Ecology</em>, 35: 523-531.
</p>
<p>Faith, D.P. (1992) Conservation evaluation and phylogenetic diversity. <em>Biological Conservation</em>, 61: 1-10.
</p>
<p>Fisher, R.A., Corbet, A.S. &amp; Williams, C.B. (1943) The relation between the number of species and the number of individuals in a random sample of animal population. <em>Journal of Animal Ecology</em>, 12: 42-58.
</p>
<p>Heip, C. 1974. A new index measuring evenness. <em>Journal of the Marine Biological Association of the United Kingdom</em>, 54: 555-557.
</p>
<p>Helmus, M.R., Bland, T.J., Williams, C.J., Ives, A.R. (2007) Phylogenetic measures of biodiversity. <em>The American Naturalist</em>, 169: E68-E83.
</p>
<p>Hill, M.O. (1973) Diversity and evenness: a unifying notation and its consequences. <em>Ecology</em>, 54: 427-432.
</p>
<p>Hurlbert, S.H. 1971. The nonconcept of species diversity: a critique and alternative parameters. <em>Ecology</em>, 52: 577-586.
</p>
<p>Kembel, S.W., Cowan, P.D., Helmus, M.R., Cornwell, W.K., Morlon, H., Ackerly, D.D., Blomberg, S.P.,Webb, C.O. (2010) Picante: R tools for integrating phylogenies and ecology. <em>Bioinformatics</em>, 26: 1463-1464.
</p>
<p>Kembel, S.W., Ackerly, D.D., Blomberg, S.P., Cornwell, W.K., Cowan, P.D., Helmus, M.R., Morlon, H. &amp; Webb, C.O. (2016) R tools for integrating phylogenies and ecology. R Package Version 1.6-2. <a href="https://CRAN.R-project.org/package=picante">https://CRAN.R-project.org/package=picante</a>
</p>
<p>Laliberte, E. &amp; Legendre, P. (2010) A distance-based framework for measuring functional diversity from multiple traits. <em>Ecology</em>, 91: 299-305.
</p>
<p>Laliberte, E., Legendre, P. &amp; Shipley, B. (2015) Measuring functional diversity (FD) from multiple traits, and other tools for functional ecology. R package version 1.0-12. Available at: <a href="https://CRAN.R-project.org/package=FD">https://CRAN.R-project.org/package=FD</a>.
</p>
<p>Lemon, J., Bolker, B., Oom, S., Klein, E., Rowlingson, B.,Wickham, H., Tyagi, A., Eterradossi, O., Grothendieck, G., Toews, M., Kane, J., Turner, R., Witthoft, C., Stander, J., Petzoldt, T., Duursma, R., Biancotto, E., Levy, O., Dutang, C., Solymos, P., Engelmann, R., Hecker, M., Steinbeck, F., Borchers, H., Singmann, H., Toal, T. &amp; Ogle, D. (2016) Various plotting functions. R package version 3.6-3. Available at: <a href="https://CRAN.R-project.org/package=plotrix">https://CRAN.R-project.org/package=plotrix</a>.
</p>
<p>Leroy, B., Petillon, J., Gallon, R., Canard, A. &amp; Ysnel, F. (2012) Improving occurrence-based rarity metrics in conservation studies by including multiple rarity cut-off points. <em>Insect Conservation and Diversity</em>, 5, 159-168.
</p>
<p>Leroy, B., Canard, A. &amp; Ysnel, F. (2013) Integrating multiple scales in rarity assessments of invertebrate taxa. <em>Diversity and Distributions</em>, 19, 794-803.
</p>
<p>Leroy, B. (2016) Calculation of Rarity Indices for Species and Assemblages of Species. R package version 1.3-4. Available at: <a href="https://CRAN.R-project.org/package=Rarity">https://CRAN.R-project.org/package=Rarity</a>.
</p>
<p>Oksanen, J., Blanchet, F.G., Kindt, R., Legendre, P., Minchin, P.R., O'Hara, R.B., Simpson, G.L., Solymos, P., Henry, M., Stevens, H. &amp; Wagner, H. (2016) Community Ecology Package. R package version 2.4-0. Available at: <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.
</p>
<p>Patil, G.P. &amp; Taillie, C. (1982) Diversity as a concept and its measurement. <em>Journal of the Acoustical Society of America</em>, 77: 548-561.
</p>
<p>Pielou, E.C. (1966) The measurement of diversity in different types of biological collections. <em>Journal Theoretical Biology</em>, 13: 131-144.
</p>
<p>Petchey, O.L. &amp; Gaston, K.J. (2002) Functional diversity (FD), species richness and community composition. <em>Ecology Letters</em>, 5: 402-411.
</p>
<p>Margalef, R. (1958) Information theory in ecology. <em>General Systems</em>, 3: 36-71.
</p>
<p>Mazuelos, N., Toja, J. &amp; Guisande, C. (1993) Rotifers in ephemeral ponds of Doñana National Park. <em>Hydrobiologia</em>, 255/256: 429-434.
</p>
<p>McIntosh, R.P. (1967) An index of diversity and the relation of certain concepts to diversity. <em>Ecology</em>, 48: 392-404.
</p>
<p>Menhinick, E.P. (1964) A comparison of some species-individual diversity indices applies to samples of field insects. <em>Ecology</em>, 45: 859-861.
</p>
<p>Renyi, A. (1961) On measures of information and entropy. <em>Proceedings of the fourth Berkeley Symposium on Mathematics, Statistics and Probability Vol. 1</em>: 547-561.
</p>
<p>Rao, C.R. (1982) Diversity and dissimilarity coefficients: a unified approach. <em>Theoretical Population Biology</em>, 21: 24-43.
</p>
<p>Shannon, C.E. (1948) A Mathematical Theory of Communication. <em>Bell System Technical Journal</em>, 27 (4): 379-423, 623-656.
</p>
<p>Shannon, C.E. &amp; Weaver, W. (1949) <em>The Mathematical Theory of Communication</em>. The University of Illinois Press, Illinois. 
</p>
<p>Simpson, E.H. (1949) Measurement of diversity. <em>Nature</em>, 163: 688.
</p>
<p>Smith, B. &amp; J.B. Wilson (1996) A consumer's guide to evenness indices. <em>Oikos</em>, 76: 70-82.
</p>
<p>Soetaert, K. 2016. Functions for plotting graphical shapes, colors. R package version 1.4.2. Available at: <a href="https://CRAN.R-project.org/package=shape">https://CRAN.R-project.org/package=shape</a>.
</p>
<p>Spellerger, I.F. &amp; Fedor P.J. (2013) A tribute to Claude Shannon (1916-2001) and a plea for more rigorous use of species richness, species diversity and the &quot;Shannon-Wiener&quot; Index. <em>Global Ecology and Biogeography</em>, 12: 177-179.
</p>
<p>Tsallis, C. (1988) Possible generalization of Boltzmann-Gibbs statistics. <em>Journal of Statistical Physics</em>, 52: 479-487.
</p>
<p>Villeger, S., Mason, N.W.H. &amp; Mouillot, D. (2008) New multidimensional functional diversity indices for a multifaceted framework in functional ecology. <em>Ecology</em> 89: 2290-2301.
</p>
<p>Webb, C.O., Ackerly, D.D. &amp; Kembel, W. (2008) Phylocom: software for the analysis of phylogenetic community structure and trait evolution. <em>Bioinformatics</em>, 24: 2098-2100.
</p>
<p>Wiener, N. (1939) The ergodic theorem. <em>Duke Mathematical Journal</em>, 5: 1-18.
</p>
<p>Wiener, N. (1948) <em>Cybernetics</em>. Wiley, New York.
</p>
<p>Wiener, N. (1949) <em>The interpolation, extrapolation, and smoothing of stationary time series</em>. Wiley, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#An example without functional diversity

data(Rotifers)

DER(data=Rotifers, Samples=c("J1.1","K4.1","G3.1","F2.1","K2.2","F8.2","F8.1",
"F1.1","F4.1","J2.1","E5.1","H5.1","K3.2","E4.2","I6.1","K2.1","J5.1","I3.1",
"K3.3","G5.1","E6.1","J1.2","J6.1","G7.1","G6.1","G4.1","E3.1","E4.3","E2.1",
"H6.2","F7.1","J6.2"), Species="Species", Taxon=c("Class","Subclass",
"Superorder","Order","Family","Genus"), pos="Pos", Index=c("Rarity.G","Menhinick",
"McIntoshE", "Dstar"), save=FALSE)



</code></pre>

<hr>
<h2 id='EnVarIP'>ENVIRONMENTAL VARIABLES IN THE IBERIAN PENINSULA</h2><span id='topic+EnVarIP'></span>

<h3>Description</h3>

<p>Geographic coordinates with values of temperature and precipitation in the Iberian Peninsula.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(EnVarIP)</code></pre>


<h3>Format</h3>

<p>Data frame with longitude, latitude and values of temperature, rainfall and altitude, and the geographic coordinates of the polygons of Spain and Portugal in the Iberian Peninsula.</p>

<hr>
<h2 id='Rotifers'>ABUNDANCE OF ROTIFERS IN PONDS</h2><span id='topic+Rotifers'></span>

<h3>Description</h3>

<p>Abundance of species of rotifers in ponds of Doñana National Park (Spain), which were obtained from Table 1 of Mazuelos et al. (1993).</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Rotifers)</code></pre>


<h3>Format</h3>

<p>An data frame with 40 columns: position of the label of the pond in the DER plot, 7 columns with the taxonomy of the species and the rest of columns are the abundance of the species of rotifers in the ponds.</p>


<h3>References</h3>

<p>Mazuelos, N., Toja, J., &amp; Guisande, C. (1993) Rotifers in ephemeral ponds of Doñana National Park. <em>Hidrobiologia</em>, 255/256: 429-434.
</p>

<hr>
<h2 id='SINENVAP'>SPATIAL INTERPOLATION OF VARIABLES WITHIN POLYGONS</h2><span id='topic+SINENVAP'></span>

<h3>Description</h3>

<p>An algorithm for finding an optimal spatial interpolation model for variables within polygons using kriging.</p>


<h3>Usage</h3>

<pre><code class='language-R'>SINENVAP(data=NULL, var=NULL, dataLat=NULL, dataLon=NULL, polyLat=NULL,
polyLon=NULL, zonedata=NULL, zonepoly=NULL, convex=FALSE, alpha=0.07, ASC=NULL,
shape=NULL, shapenames=NULL, Area=NULL, validation=30, type.krige="OK",
trend.d="cte", trend.l="cte", model="AUTO", minimisation="optim", weights="npairs",
maxdist=NULL, nugget=NULL, sill=NULL, range=NULL, kappa=NULL, beta=NULL,
jitter="jitter", maxjitter=0.00001, direction=c(0,45,90,135), inside=TRUE,
error=FALSE, ResetPAR=TRUE, PAR=NULL, BOXPLOT=NULL, OUTLINE=FALSE, XLABP=NULL,
YLABP=NULL, XLAB=NULL, YLAB=NULL, XLABB="Model", YLABB="Accuracy measures",
MAIN="", XLIM=NULL, YLIM=NULL, ZLIM=NULL, COLOR="rev(heat.colors(100))",
COLORC="black", COLORB=NULL, COLORM="transparent", NLEVELS=10, LABCEX=0.6,
contour=TRUE, breaks=10, ndigits=0, xl=0, xr=0, pro=TRUE, cell=NULL,
file1="Predictions data.csv", file2="Predictions polygon.csv",
file3="Accuracy measures.csv", file4="Semivariogram.csv",
file5="Standard errors.csv", file6="Model selected.txt", na="NA",
dec=",", row.names=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SINENVAP_+3A_data">data</code></td>
<td>
<p>Data file (CSV, RData or Excel) with the latitudes and longitudes and the values of the environmental variables. This file may also include the latitudes and longitudes of the polygons. Each polygon must be separated by a blank row.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_var">var</code></td>
<td>
<p>Variable with the values of the environmental variable.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_datalat">dataLat</code></td>
<td>
<p>Variable with the latitudes of the environmental variable.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_datalon">dataLon</code></td>
<td>
<p>Variable with the longitudes of the environmental variable.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_polylat">polyLat</code></td>
<td>
<p>If the geographic coordinates of the polygons are in the same file than the data of the variable, here it is indicated the variable with the latitudes of the polygons.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_polylon">polyLon</code></td>
<td>
<p>If the geographic coordinates of the polygons are in the same file than the data of the variable, here it is indicated the variable with the latitudes of the polygons.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_zonedata">zonedata</code></td>
<td>
<p>If the latitude and longitude of the data are in UTM, it is necessary to specify the variable with the zone of each pair of coordinates in this argument.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_zonepoly">zonepoly</code></td>
<td>
<p>If the latitude and longitude of the polygons are in UTM, it is necessary to specify the variable with the zone of each pair of coordinates in this argument.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_convex">convex</code></td>
<td>
<p>If it is TRUE, it is considered as polygon the alpha shape of the distribution of the variable. This option is useful if the variable is for instance the abundance of a species, so the spatial interpolation is performed considering the limits of the distribution of the species.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_alpha">alpha</code></td>
<td>
<p>Alpha value of the alpha shape.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_asc">ASC</code></td>
<td>
<p>ASC file with the values of the variable. It is not necessary to specify the latitude and longitude of the variable, but it is mandatory to specify the polygon (in the argument <em>data</em>, <em>shape</em> or <em>Area</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_shape">shape</code></td>
<td>
<p>It is possible to use a shape file for importing the coordinates of the polygons. In this case, it is not necessary to specify the latitude and longitude of the polygons in the arguments <em>polyLat</em> and <em>polyLon</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_shapenames">shapenames</code></td>
<td>
<p>Variable in the shapefile with the names of the polygons.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_area">Area</code></td>
<td>
<p>Only if using RWizard. It is also possible to use the polygons available in RWizard of administrative areas and river basins. A character with the name of the administrative area or a vector with several administrative areas (countries, regions, etc.) or river basins. In this case, it is not necessary to specify the latitude and longitude of the polygons in the arguments <em>polyLat</em> and <em>polyLon</em>. If it is &quot;World&quot; the entire world is plotted.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_validation">validation</code></td>
<td>
<p>Percentage of cases used from original data for validation. These data are not used for the estimation of the model and they are just used for evaluating the accuracy of the model (see details). If it is zero, all data are used for estimating the accuracy measures. If there are many data, a way for shortening the running time is to increase the number of data for validation, so reducing the number of data used for estimating the models.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_type.krige">type.krige</code></td>
<td>
<p>Type of kriging to be performed. Options are simple &quot;SK&quot; or ordinary kriging &quot;OK&quot;. Kriging with external trend and universal kriging can be defined setting type.krige=&quot;OK&quot; and specifying the trend model using the arguments <em>trend.d</em> and <em>trend.l</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_trend.d">trend.d</code></td>
<td>
<p>It specifies the trend (covariate) values at the data locations (see function <a href="geoR.html#topic+krige.conv">krige.conv</a> of the package geoR).</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_trend.l">trend.l</code></td>
<td>
<p>It specifies the trend (covariate) values at prediction locations. It must be of the same type as for trend.d. Only used if prediction locations are provided in the
argument locations (see function <a href="geoR.html#topic+krige.conv">krige.conv</a> of the package geoR).</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_model">model</code></td>
<td>
<p>If it is &quot;AUTO&quot;, the algorithm tries to find the model with the highest accuracy measures (see details). It is also possible to select one or several of the following models: &quot;exponential&quot;, &quot;matern&quot;, &quot;gaussian&quot;, &quot;spherical&quot;, &quot;circular&quot;, &quot;cubic&quot;, &quot;wave&quot;, &quot;power&quot;, &quot;linear&quot;, &quot;cauchy&quot;, &quot;gneiting&quot;, &quot;powered.exponential&quot;, and/or &quot;pure.nugget&quot;.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_minimisation">minimisation</code></td>
<td>
<p>Minimization function used to estimate the parameters of the model fitted to the semivariogram. The options are &quot;optim&quot;, &quot;nlm&quot; or &quot;nls&quot; (see function <a href="geoR.html#topic+variofit">variofit</a> of the package geoR).</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_weights">weights</code></td>
<td>
<p>Type weights used in the loss function when fitting the model to the semivariogram. The options are &quot;npairs&quot;, &quot;cressie&quot; or &quot;equal&quot; (see the function <a href="geoR.html#topic+variofit">variofit</a> of the package geoR).</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_maxdist">maxdist</code></td>
<td>
<p>Maximum distance in the semivariogram. If it is NULL, it is half of the maximum distance of the semivariogram.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_nugget">nugget</code>, <code id="SINENVAP_+3A_sill">sill</code>, <code id="SINENVAP_+3A_range">range</code></td>
<td>
<p>The value of the nugget variance parameter <code class="reqn">\tau^2</code>, fixed value of the sill parameter (<code class="reqn">\sigma^2</code>) and fixed value of the range parameter (<code class="reqn">\phi</code>). If they are NULL (default) the algorithm tries to find the optimal values for each of the models specified in the argument <em>model</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_kappa">kappa</code></td>
<td>
<p>One numerical value required for the following models: &quot;matern&quot;, &quot;cauchy&quot;, &quot;gneiting.matern&quot; and &quot;powered.exponential&quot;, and two values for the model &quot;gencauchy&quot;. If they are NULL (default) the algorithm tries to find the optimal values for each of the models specified in the argument <em>model</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_beta">beta</code></td>
<td>
<p>Numerical value of the mean (vector) parameter. Only used if type.krige=&quot;SK&quot;. If it is NULL, it is automatically estimated by the algorithm.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_jitter">jitter</code></td>
<td>
<p>It may be one of these three options: &quot;NO&quot; means no action, &quot;jitter&quot; means that jitters duplicated coordinates of the environmental variable, and &quot;mean&quot; means that the mean of the environmental variable is estimated for those duplicated coordinates.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_maxjitter">maxjitter</code></td>
<td>
<p>Maximum jittering distance in decimal degrees.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_direction">direction</code></td>
<td>
<p>A vector with values of 4 angles, indicating the directions for which the variograms will be computed. Default corresponds to c(0,45,90,135 (degrees).</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_inside">inside</code></td>
<td>
<p>If it is TRUE only those geographic coordinates of the environmental variables inside the polygons are considered for the estimation of the model.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_error">error</code></td>
<td>
<p>If it is TRUE, a contour map with the standard errors is depicted.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_resetpar">ResetPAR</code></td>
<td>
<p>If it is FALSE, the default condition of the function PAR is not placed and maintained those defined by the user in previous graphics.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_par">PAR</code></td>
<td>
<p>It accesses the function PAR that allows to modify many different aspects of the graph.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_boxplot">BOXPLOT</code></td>
<td>
<p>It allows to specify the characteristics of the function <a href="graphics.html#topic+boxplot">boxplot</a>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_outline">OUTLINE</code></td>
<td>
<p>If it is TRUE, the outliers are shown in the boxplot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_xlabp">XLABP</code>, <code id="SINENVAP_+3A_ylabp">YLABP</code></td>
<td>
<p>Legends of <em>X</em> and <em>Y</em> axes of the plot with the relationship between  observed and predicted values of the model.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_xlab">XLAB</code>, <code id="SINENVAP_+3A_ylab">YLAB</code></td>
<td>
<p>Legends of <em>X</em> and <em>Y</em> axes of the contour plot with the spatial interpolation.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_xlabb">XLABB</code>, <code id="SINENVAP_+3A_ylabb">YLABB</code></td>
<td>
<p>Legends of <em>X</em> and <em>Y</em> axes of the boxplot .</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_main">MAIN</code></td>
<td>
<p>Main title of the contour plot with the spatial interpolation.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_xlim">XLIM</code>, <code id="SINENVAP_+3A_ylim">YLIM</code>, <code id="SINENVAP_+3A_zlim">ZLIM</code></td>
<td>
<p>Limits of the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_color">COLOR</code></td>
<td>
<p>Palette of colours or a vector with the colours of the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_colorc">COLORC</code></td>
<td>
<p>Colour of the lines in the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_colorb">COLORB</code></td>
<td>
<p>Vector with the colours of the models or just one colour for all models of the boxplot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_colorm">COLORM</code></td>
<td>
<p>Colour of the administrative areas and river basins, if any area has been specified in the argument <em>Area</em> and <em>convexhull=TRUE</em>.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_nlevels">NLEVELS</code></td>
<td>
<p>Numeric vector of levels at which to draw contour lines.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_labcex">LABCEX</code></td>
<td>
<p>Size of the text in the contour lines.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_contour">contour</code></td>
<td>
<p>If it is TRUE, the contour lines are depicted in the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_breaks">breaks</code></td>
<td>
<p>Number of breakpoints of the colour legend in the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_ndigits">ndigits</code></td>
<td>
<p>Number of decimals in the legend of the colour scale in the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_xl">xl</code>, <code id="SINENVAP_+3A_xr">xr</code></td>
<td>
<p>The left and right limits of the colour legend considering the <em>X</em> axis of the contour plot.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_pro">pro</code></td>
<td>
<p>If it is TRUE, an automatic calculation is made in order to correct the aspect ratio y/x along latitude.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_cell">cell</code></td>
<td>
<p>Cell size in decimal degrees of the grid inside the polygons with the predictions of the model. If it is NULL, it is automatically estimated according to the limits of the polygons. To select an appropriate cell size according to the polygon size is important for shortening the running time.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file1">file1</code></td>
<td>
<p>CSV FILES. Filename with the predictions of the models.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file2">file2</code></td>
<td>
<p>CSV FILES. Filename with the predictions inside the polygons.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file3">file3</code></td>
<td>
<p>CSV FILES. Filename with accuracy measures of the models.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file4">file4</code></td>
<td>
<p>CSV FILES. Filename with values of the semivariogram.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file5">file5</code></td>
<td>
<p>CSV FILES. Filename with standard errors of the predictions.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_file6">file6</code></td>
<td>
<p>TXT FILE. Model selected with indication of the accuracy measures.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_na">na</code></td>
<td>
<p>CSV FILE. Text that is used in the cells without data.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_dec">dec</code></td>
<td>
<p>CSV FILE. It defines if the comma &quot;,&quot; is used as decimal separator or the dot &quot;.&quot;.</p>
</td></tr>
<tr><td><code id="SINENVAP_+3A_row.names">row.names</code></td>
<td>
<p>CSV FILE. Logical value that defines if identifiers are put in rows or a vector with a text for each of the rows.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>SINENVAP algorithm</strong>
</p>
<p>The aim of this algorithm is to select a model, from a set of different models, with the nugget variance parameter <code class="reqn">\tau^2</code>, fixed value of the sill parameter (<code class="reqn">\sigma^2</code>) and fixed value of the range parameter (<code class="reqn">\phi</code>), as close as possible to those values that generates an optimal spatial interpolation, and to validate the predictions obtained. The model and parameters selected by the algorithm may be utilized by users as references, or to make modifications for spatial interpolation prediction improvement.
</p>
<p>The algorithm uses the package geoR (Ribeiro and Diggle, 2001; 2018) to estimate simple, ordinary, and universal kriging. The corresponding algorithm is detailed below.
</p>
<p><strong>1. Data with variable and polygon coordinates.</strong>
</p>
<p>The algorithm must be supplied with variable values (argument <em>var</em>), latitude and longitude for each datum (arguments <em>dataLat</em> and <em>dataLon</em>), and polygon coordinates (arguments <em>polyLat</em> and <em>polyLon</em>), for the estimation of spatial interpolation.
</p>
<p>Variable and polygon latitudes and longitudes may be in either decimal or UTM form. If they are in UTM form, a column with the zone of each coordinate in the <em>zonedata</em> and <em>zonepoly</em> arguments must be added for variable and polygon coordinates, respectively. Polygon variables and coordinates are not required to be in the same units. Therefore, variables may be in decimal form, and polygons in UTM form, vice versa, or both may be in the same units.
</p>
<p>Variable and polygon information data may be in CSV, EXCEL, or RData files. ASC files (argument ASC) may also be used for the variable, but in this case, it is not necessary to provide latitude or longitude information for each datum in the <em>dataLat</em> and <em>dataLon</em> arguments. Polygon coordinates may be in the same file as variable data in CSV, EXCEL, or RData files.
</p>
<p>If RWizard is used, any of the administrative areas available in RWizard may be chosen as polygons in the <em>Area</em> argument: countries, departments, provinces, etc. River basins may also be used as polygons, in the database available in RWizard (Gonzalez-Vilas et al., 2015). 
</p>
<p>Shape files may be used to import polygon coordinates with the <em>shape</em> argument. If a shape file or the polygons available in RWizard are used, coordinate specification is unnecessary in <em>polyLat</em> and <em>polyLon</em> arguments.
</p>
<p>Finally, if the argument <em>convex=TRUE</em>, the alpha shape distribution is considered to be a polygon. This last option is useful when the variable is, for instance, the abundance of a species, such that spatial interpolation is performed considering the limits of species distribution. If <em>convex=TRUE</em>, the specification of information in <em>polyLat</em> and <em>polyLon</em> arguments is unnecessary.
</p>
<p><strong>2. Algorithm design.</strong>
</p>
<p>Algorithm steps are described as follows:
</p>
<p>1. If the argument <em>inside=TRUE</em> (default option), based on all variable data available, only those data inside the polygons are used to estimate spatial interpolation. If the argument <em>inside=FALSE</em>, all available variable data are used.
</p>
<p>2. Duplicated coordinates may be treated in two ways: with the application of a jitter function (default) or with estimation of the mean value of the variable for duplicated coordinates.
</p>
<p>3. Spatial interpolation estimates the values on a grid. Therefore, it is necessary to first create a grid with a fixed cell size, in which the spatial interpolation will be estimated. If the argument <em>cell=NULL</em> (default), the algorithm estimates the optimal grid cell size, in accordance with polygon size. The output TXT provides information about the cell size chosen. The user may specify different cell sizes. Appropriate cell size selection, in accordance with polygon size, shortens running time. 
</p>
<p>4. Only those points in polygons are selected from the grid created, for the spatial interpolation prediction.
</p>
<p>5. By the default <em>validation=30</em>, which means that 30% of variable data are not used in the spatial interpolation model. However, they are employed to test the model. If <em>validation=0</em>, all data are used in the model, and model validation is performed with all data. If validation is higher than zero, because validation data are randomly selected, spatial interpolation values may vary each time the script is implemented.
</p>
<p>6. In order to test the model, the grid coordinates nearest to data coordinates are selected. Cross validation is performed by comparing variable data reserved for validation (see step 5) to spatial interpolation predictions on the grid, so as to verify which coordinates are nearest to the data reserved for validation.
</p>
<p>7.Next, the semivariogram is depicted (Fig. 1). If the argument <em>maxdist=NULL</em> (default), it is considered to be half of the maximum semivariogram distance for performing the models.
</p>
<p>8. Any of the following models may be implemented: cauchy, circular, cubic, exponential, gaussian, gneiting, linear, matern, power, powered.exponential, pure.nugget, spherical, or wave. If <em>model=&quot;AUTO&quot;</em>, all models are tested, with the exception of matern. The <em>variofit</em> function in the geoR package is used to find the nugget, range, and sill for each model. The data reserved for validation are compared to model predictions, and seven accuracy measures, described in the following section, are used to decide which model made the best prediction. The model that made the best prediction is chosen by the algorithm to depict variable spatial interpolation in the selected polygons.
</p>
<p><strong>3. Accuracy measures.</strong>
</p>
<p>The following accuracy measures are used in the algorithm, so as to compare and evaluate model predictions, where <em>n</em> is the number of observations, and <em>P</em> and <em>O</em> are the predicted and observed values for each <em>i</em> datum, respectively.
</p>
<p>The measures catalogued as &quot;normalized&quot; are adapted to reflect values of 1 when the model is most efficient (i.e., predictions are the same as observed values). Thus, in all accuracy measures used in the algorithm, the maximum value is 1. This indicates a model with maximum predictive power. Various measures were utilized to obtain improved evaluation framework (i.e. consideration of a group of skill scores that show different result characteristics) (see Li and Head, 2011).
</p>
<p><em>3.1. r-squared (<code class="reqn">r^2</code>)</em>. This is the square of the Pearson correlation coefficient (<em>r</em>) between the predictions of the interpolation model and the observed values. It ranges from 0 to 1.
</p>
<p style="text-align: center;"><code class="reqn">r=\frac{\displaystyle\sum_{i=1}^{n}[(O_i-\bar{O})(P_i-\bar{P})]}{\sqrt{\displaystyle\sum_{i=1}^{n}{(O_i-\bar{O})}^2\displaystyle\sum_{i=1}^{n}{(P_i-\bar{P})}^2}}</code>
</p>

<p><em>3.2. Normalized mean absolute error (NMAE)</em>. This is a measure of the average error between predictions and observed values. It ranges from <code class="reqn">-\infty</code> to 1.
</p>
<p style="text-align: center;"><code class="reqn">NMAE=1-\frac{\displaystyle\sum_{i=1}^{n}|P_i-O_i|}{\displaystyle\sum_{i=1}^{n}O_i}</code>
</p>

<p><em>3.3. Normalized root mean square error (NRMSE)</em>. This measure shows the distribution error variability. It ranges from <code class="reqn">-\infty</code> to 1.
</p>
<p style="text-align: center;"><code class="reqn">NRMSE=1-\sqrt{\frac{\displaystyle\sum_{i=1}^{n}(P_i-O_i)^2}{\displaystyle\sum_{i=1}^{n}O_i}}</code>
</p>

<p><em>3.4. Nash-Sutcliffe coefficient (E)</em>. Nash-Sutcliffe efficiencies may range from <code class="reqn">-\infty</code> to 1 (Nash &amp; Sutcliffe, 1970). An efficiency of 1 (E = 1) corresponds to a perfect match between model and observations. An efficiency of 0 indicates that the model predictions are as accurate as the mean of the observed data, whereas an efficiency less than zero (<code class="reqn">-\infty</code> &lt; E &lt; 0) occurs when the observed mean is a better predictor than the model.
</p>
<p style="text-align: center;"><code class="reqn">E=1-\frac{\displaystyle\sum_{i=1}^{n}(O_i-P_i)^2}{\displaystyle\sum_{i=1}^{n}(O_i-\bar{O})^2}</code>
</p>

<p><em>3.5. Index of agreement (d)</em>. It was developed by Willmott (1981) as a standardized measure of the degree of model prediction error and varies between 0 and 1.
</p>
<p style="text-align: center;"><code class="reqn">d=1-\frac{\displaystyle\sum_{i=1}^{n}(O_i-P_i)^2}{\displaystyle\sum_{i=1}^{n}(|P_i-\bar{O}|+|O_i-\bar{O}|)^2}</code>
</p>

<p><em>3.6. Normalized relative mean absolute error (NRMAE)</em>. This is a modification of a measure developed by Li and Head (2011), whose maximum value is 1. According to the authors, this measure removes the effect of unit/scale and is not sensitive to changes in unit/scale. It ranges from <code class="reqn">-\infty</code> to 1.
</p>
<p style="text-align: center;"><code class="reqn">NRMAE=1-\frac{\displaystyle\sum_{i=1}^{n}\frac{|P_i-O_i|}{O_i}*100}{\displaystyle\sum_{i=1}^{n}O_i}</code>
</p>

<p><em>3.7. Normalized relative root mean square error (NRRMSE)</em>. This is a modification of a measure developed by Li and Head (2011), whose maximum value is 1. It ranges from 0 to 1.
</p>
<p style="text-align: center;"><code class="reqn">NRRMSE=1-\left[\frac{\displaystyle\sum_{i=1}^{n}\left(\frac{[P_i-O_i]}{O_i}\right)^2}{\displaystyle\sum_{i=1}^{n}O_i}\right]^\frac{1}{2}*100</code>
</p>

<p><strong>FUNCTIONS</strong>
</p>
<p>Spatial interpolation, using simple, ordinary, and universal kriging, is performed with as.geodata, variofit, krige.conv and krige.control functions, as well as the jitter of points with the jitterDupCoords function, all of which are from the geoR package (Ribeiro and Diggle, 2001; 2018).
</p>
<p>The ASC files are loaded with the raster function from the raster package (Hijmans et al., 2018).
</p>
<p>Points inside polygons are estimated with the in.out function from the mgcv package (Wood, 2018).
</p>
<p>The sp package (Pebesma and Bivand, 2005; Pebesma et al., 2018) is used to process shape files.
</p>
<p>The BreuschPagan test is performed with the bptest function from the lmtest package (Zeileis and Hothorn, 2002; Hothorn et al., 2018).
</p>
<p>The color scale is depicted with the color.legend function from the plotrix package (Lemon, 2006; Lemon et al., 2018).
</p>
<p><strong>EXAMPLE</strong>
The figure shows the contour map of the spatial interpolation of the example 1, the rainfall in the Iberian Peninsula.
</p>

<table>
<tr>
 <td style="text-align: center;">
<img src="../help/figures/EcoIndRF3.jpg" width="50" alt="EcoIndRF3.jpg" />

</td>
</tr>

</table>



<h3>Value</h3>

<p>It is obtained:
</p>
<p>1. A CSV file, called &quot;Predictions data.CSV&quot; by default, contains model data and predictions. If <em>validation=0</em>, the observed data are from the original dataset. Predicted values are those points inside the polygon, which are, spatially, the nearest neighbors to the observed data.
</p>
<p>2. A CSV file, &quot;Predictions polygon.CSV&quot; by default, contains predictions for inside polygons.
</p>
<p>3. A CSV file, called &quot;Accuracy measures.CSV&quot; by default, contains the values of the seven accuracy measures shown above, for all models.
</p>
<p>4. A CSV file, called &quot;Semivariogram.CSV&quot; by default, contains semivariogram values.
</p>
<p>5. A CSV file, called &quot;Standard errors.CSV&quot; by default, contains the standard prediction errors.
</p>
<p>6. A TXT file called &quot;Model selected.TXT&quot; by default, contains the full details of the model selected by the algorithm.
</p>
<p>7. A plot of the semivariogram, with the values used in the models, is depicted in green. Application of the maximum distance specified in the <em>maxdist</em> argument, and the points in red, yield a semivariogram without distance limitations.
</p>
<p>8. The directional variogram in four directions.
</p>
<p>9. A plot with the relationship between observed and predicted values. As mentioned above, the observed values may be either randomly selected values or those from the original dataset. If <em>validation=0</em>, the observed data are from the original dataset. Predicted values are those points inside the polygon, which are, spatially, the nearest neighbors to the observed data.
</p>
<p>10. If the argument <em>model</em> is &quot;AUTO&quot;, or there is more than one model, a boxplot is depicted with the median value of the seven accuracy measures from each model.
</p>
<p>11. The contour plot with the spatial interpolation predictions of the selected model, i.e., the model with the highest accuracy measures mean.
</p>
<p>12. If the argument <em>error=TRUE</em>, the contour plot is depicted with the selected model's standard errors.
</p>


<h3>References</h3>

<p>Hijmans, R.J., van Etten, J., Cheng, J., Sumner, M., Mattiuzzi, M., Greenberg, J.A., Lamigueiro, O.P., Bevan, A., Bivand, R., Busetto, L., Canty, M., Forrest, D., Ghosh, A., Golicher, D., Gray, J., Hiemstra, P., Karney, C., Mosher, S., Nowosad, J., Pebesma, E., Racine, E.B., Rowlingson, B., Shortridge, A., Venables, B., Wueest, R. (2018) Geographic Data Analysis and Modeling. R package version 2.8-4. Available at: <a href="https://CRAN.R-project.org/package=raster">https://CRAN.R-project.org/package=raster</a>.
</p>
<p>Gonzalez-Vilas, L., Guisande, C., Vari, R. P., Pelayo-Villamil, P., Manjarres-Hernandez, A., Garcia-Rosello, E., Gonzalez-Dacosta, J., Heine, J., Perez-Costas, E., Granado-Lorencio, C., Palau-Ibars, A., Lobo, J. M. (2016) Geospatial data of freshwater habitats for macroecological studies: an example with freshwater fishes. <em>Journal of Geographical Information Science</em>, 30: 126-141.
</p>
<p>Lemon, J. (2006) Plotrix: a package in the red light district of R. <em>R-News</em>, 6: 8-12.
</p>
<p>Lemon, J., Bolker, B., Oom, S., Klein, E., Rowlingson, B.,Wickham, H., Tyagi, A., Eterradossi, O., Grothendieck, G., Toews, M., Kane, J., Turner, R., Witthoft, C., Stander, J., Petzoldt, T., Duursma, R., Biancotto, E., Levy, O., Dutang, C., Solymos, P., Engelmann, R., Hecker, M., Steinbeck, F., Borchers, H., Singmann, H., Toal, T. &amp; Ogle, D. (2018). Various plotting functions. R package version 3.7-4. Available at: <a href="https://CRAN.R-project.org/package=plotrix">https://CRAN.R-project.org/package=plotrix</a>.
</p>
<p>Li, J. &amp; Heap, A.D. (2011) A review of comparative studies of spatial interpolation methods in environmental sciences: Performance and impact factors. <em>Ecological Informatics</em>, 6: 248-251.
</p>
<p>Nash, J.E. &amp; Sutcliffe, J.V. (1970) River flow forecasting through conceptual models part I - A discussion of principles. <em>Journal of Hydrology</em>, 10, 282-290.
</p>
<p>Pebesma, E., Bivand, R.S., Rowlingson, B., Gomez-Rubio, V., Hijmans, R., Sumner, M., MacQueen, D., Lemon, J. &amp; O'Brien, J. (2018) Classes and Methods for Spatial Data. R package version 1.3-1. Available at: <a href="https://CRAN.R-project.org/package=sp">https://CRAN.R-project.org/package=sp</a>.
</p>
<p>Pebesma, E.J. &amp; Bivand, R.S. (2005) Classes and methods for spatial data in R. <em>R News</em>, 5: 9-13.
</p>
<p>Ribeiro, P.A. &amp; Diggle, P.J. (2001) geoR: a package for geostatistical analysis. R-News 1, 14-18.
</p>
<p>Ribeiro, P.J. &amp; Diggle, P.J. (2018) Analysis of Geostatistical Data. R package version 1.7-5.2.1. Available at: <a href="https://CRAN.R-project.org/package=geoR">https://CRAN.R-project.org/package=geoR</a>.
</p>
<p>Willmott, C.J. (1981) On the validation of models. <em>Physical Geography</em>, 2, 184-194.
</p>
<p>Wood, S. (2018) Mixed GAM Computation Vehicle with Automatic Smoothness. R package version 1.8-26. Available at: <a href="https://CRAN.R-project.org/package=mgcv">https://CRAN.R-project.org/package=mgcv</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

#Example 1.
#An example with the geographic coordinates of the polygons
#in the same file than the environmental variable

data(EnVarIP)

SINENVAP(data=EnVarIP, var="Rainfall", dataLat="dataLat",
dataLon="dataLon", polyLat="polyLat", polyLon="polyLon",
model=c("cubic", "spherical"), MAIN="Rainfall", dec=".")

#Example 2. Only to be used with RWizard
#An example using the administrative areas available in RWizard.

data(EnVarIP)
@_Build_AdWorld_
SINENVAP(data=EnVarIP, var="Temperature", dataLat="dataLat", dataLon="dataLon",
Area = c("Galicia&gt;A CoruÃ±a", "Galicia&gt;Lugo", "Galicia&gt;Ourense",
"Galicia&gt;Pontevedra"), model=c("spherical"), MAIN="Temperature", ndigits=1, dec=".")

#Example 3. Only to be used with RWizard
#An example with a virtual species using as polygon the alpha
#shape of the species (argument convex=TRUE).

data(VirtualSpecies)
@_Build_AdWorld_
SINENVAP(data=VirtualSpecies, var="Probability", dataLat="Lat",
dataLon="Lon", model=c("circular", "exponential"), convex=TRUE,
Area=c("France"), validation=90, COLORM="#DEDEDE64", ndigits=2, dec=".")


## End(Not run)
</code></pre>

<hr>
<h2 id='VirtualSpecies'>PROBABILITY OF A VIRTUAL SPECIES</h2><span id='topic+VirtualSpecies'></span>

<h3>Description</h3>

<p>Probability of a virtual species in France.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(VirtualSpecies)</code></pre>


<h3>Format</h3>

<p>The 3 bioclimatic variables of WorldClim at a resolution of five arc-minutes (Hijmans et al., 2005) BIO8 (Mean Temperature of Wettest Quarter), BIO12 (Annual Precipitation) and BIO15 (Precipitation Seasonality, Coefficient of Variation) were used to generate the distribution of the virtual species by using the virtualspecies R package (Leroy et al. 2016; 2018). The limits were from -5º to 6º in latitude, and from 44º to 49º. Subsequently, 50% of the data were randomly selected, in order to simulate a random sampling of the species. The final continuous suitability maps obtained in this way were scaled between 0 and 1.
</p>


<h3>References</h3>

<p>Hijmans, R. J., Cameron, S. E., Parra, J. L., Jones, P. G. &amp; Jarvis, A. (2005) Very high resolution interpolated climate surfaces for global land areas. <em>International Journal of Climatology</em>, 25: 1965-1978.
</p>
<p>Leroy, B., Meynard, C. N., Bellard, C., Courchamp, F. (2016) Virtualspecies, an R package to generate virtual species distributions. <em>Ecography</em>, 39: 599-607.
</p>
<p>Leroy, B., Meynard, C.N., Bellard, C., Courchamp, F., Delsol, R. &amp; Gaul, W. (2018) Generation of Virtual Species Distributions. R package version 1.4-4. Available at: <a href="https://CRAN.R-project.org/package=virtualspecies">https://CRAN.R-project.org/package=virtualspecies</a>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
