<!DOCTYPE html><html lang="en"><head><title>Help for package econullnetr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {econullnetr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bipartite_stats'><p>Test for significant differences in a range of network metrics between the</p>
observed and null bipartite networks</a></li>
<li><a href='#Broadstone'><p>Part of the highly-resolved food web from Broadstone Stream, UK</p></a></li>
<li><a href='#Broadstone.fl'><p>'Forbidden' links to accompany the <code>Broadstone</code> data set</p></a></li>
<li><a href='#Broadstone.nodes'><p>Abundance and body mass data to accompany the <code>Broadstone</code> data set</p></a></li>
<li><a href='#Broadstone.prey'><p>Macroinvertebrate abundance data to accompany the <code>Broadstone</code> data frame</p></a></li>
<li><a href='#econullnetr'><p>econullnetr: tools for using null models to analyse the structure of ecological</p>
networks and identify resource selection</a></li>
<li><a href='#expand_matrix'><p>Expand a summarised interaction matrix</p></a></li>
<li><a href='#generate_edgelist'><p>Export null modelling results</p></a></li>
<li><a href='#generate_null_net'><p>Null models for ecological networks</p></a></li>
<li><a href='#plot_bipartite'><p>Plot a bipartite network, colour coding individual links according to</p>
whether they are stronger or weaker than expected under the null model</a></li>
<li><a href='#plot_preferences'><p>Plot the resource preferences of a consumer</p></a></li>
<li><a href='#Silene'><p>Flower visitation network from an arable field in the UK</p></a></li>
<li><a href='#Silene.plants'><p>Flower abundance data to accompany the <em>Silene</em> visitation network</p></a></li>
<li><a href='#test_interactions'><p>Compare observed interaction strengths in a network to those estimated from</p>
a null model</a></li>
<li><a href='#WelshStreams'><p>Part of the food web from upland streams in south Wales, UK</p></a></li>
<li><a href='#WelshStreams.fl'><p>'Forbidden' links to accompany the upland streams food web</p></a></li>
<li><a href='#WelshStreams.order'><p>Additional data to assist with plotting outputs from <code>WelshStreams</code></p></a></li>
<li><a href='#WelshStreams.prey'><p>Macroinvertebrate abundance to accompany part of the food web from upland</p>
streams in south Wales, UK</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Null Model Analysis for Ecological Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, bipartite, gtools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>igraph, knitr, readr, rmarkdown, testthat, vdiffr</td>
</tr>
<tr>
<td>Author:</td>
<td>Ian Vaughan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ian Vaughan &lt;vaughanip@cardiff.ac.uk&gt;</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ivaughan/econullnetr/issues">https://github.com/ivaughan/econullnetr/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for using null models to analyse ecological
    networks (e.g. food webs, flower-visitation networks, seed-dispersal
    networks) and detect resource preferences or non-random interactions among
    network nodes. Tools are provided to run null models, test for and plot
    preferences, plot and analyse bipartite networks, and export null model
    results in a form compatible with other network analysis packages. The 
    underlying null model was developed by Agusti et al. (2003) Molecular 
    Ecology &lt;<a href="https://doi.org/10.1046%2Fj.1365-294X.2003.02014.x">doi:10.1046/j.1365-294X.2003.02014.x</a>&gt; and the full application to 
    ecological networks by Vaughan et al. (2018) econullnetr: an R package 
    using null models to analyse the structure of ecological networks and 
    identify resource selection. Methods in Ecology &amp; Evolution, 
    &lt;<a href="https://doi.org/10.1111%2F2041-210X.12907">doi:10.1111/2041-210X.12907</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-01 11:58:18 UTC; sbiipv</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-02 08:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='bipartite_stats'>Test for significant differences in a range of network metrics between the
observed and null bipartite networks</h2><span id='topic+bipartite_stats'></span>

<h3>Description</h3>

<p>Acts as a wrapper for the <code>bipartite</code> package's
<code><a href="bipartite.html#topic+networklevel">networklevel</a></code>,
<code><a href="bipartite.html#topic+grouplevel">grouplevel</a></code> and
<code><a href="bipartite.html#topic+specieslevel">specieslevel</a></code> functions, allowing a wide range
of measures to be calculated (Dormann <em>et al</em>., 2008, 2009; Dorman
2011).  These are calculated both for the observed network and across the
iterations of the null model, allowing for simple tests of whether the
observed values differ from those expected by chance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bipartite_stats(
  nullnet,
  signif.level = 0.95,
  index.type,
  indices,
  prog.count = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bipartite_stats_+3A_nullnet">nullnet</code></td>
<td>
<p>An object of class 'nullnet' from <code>generate_null_net</code></p>
</td></tr>
<tr><td><code id="bipartite_stats_+3A_signif.level">signif.level</code></td>
<td>
<p>An optional value specifying the threshold used for
testing for 'significant' deviations from the null model.  Defaults to 0.95</p>
</td></tr>
<tr><td><code id="bipartite_stats_+3A_index.type">index.type</code></td>
<td>
<p>String specifying which function to call from the
<code>bipartite</code> package: <code>specieslevel</code>, <code>grouplevel</code> or
<code>networklevel</code>, according to the level at which the statistics are
required.  For <code>specieslevel</code>, statistics are calculated for both
'higher' and 'lower' levels i.e. <code>level = "both"</code> and cannot be
changed: adding a <code>level</code> argument will produce an error:
statistics from higher and lower levels are easily separated in the output
from <code>bipartite_stats</code>.</p>
</td></tr>
<tr><td><code id="bipartite_stats_+3A_indices">indices</code></td>
<td>
<p>Vector listing the bipartite network statistics to calculate.
All indices are currently supported, with the exception of the dependence
matrix: if <code>ALL</code> is specified, it will default to <code>ALLBUTD</code>.</p>
</td></tr>
<tr><td><code id="bipartite_stats_+3A_prog.count">prog.count</code></td>
<td>
<p>A logical value specifying whether the progress count
should be shown. Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="bipartite_stats_+3A_...">...</code></td>
<td>
<p>Other arguments that may be supplied to <code>bipartite</code>'s
<code>specieslevel</code>, <code>grouplevel</code> or <code>networklevel</code> functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Allows most of the network metrics in the <code>bipartite</code> package
to be calculated for an observed bipartite network and compared to the
distribution of those network metrics across the iterations of the null
model.  This indicates whether the observed network differs from the
structure of the network that could be expected if consumers simply used
resources in proportion to their relative abundance.
</p>
<p>The user sets the significance level (default = 95% confidence limits),
and the metrics selected are classified into those that are higher,
lower or consistent with the null model at that significance level.
Significance is determined by comparing the observed value of the statistic
to the 1-alpha/2 percentiles from the frequency distribution, with
'significant' values falling outside the confidence interval (Manly 2006).
</p>


<h3>Value</h3>

<p>Returns one or more data frames according to the level at which the
statistics are calculated (<code>specieslevel, grouplevel</code> or
<code>networklevel</code>).  If <code>index.type = "networklevel"</code> or
<code>index.type = "grouplevel"</code> a single data frame is returned, listing
the chosen network statistics and with the following column headings:
</p>

<dl>
<dt><code>Observed</code></dt><dd><p>Value of the statistic for the observed network</p>
</dd>
<dt><code>Null</code></dt><dd><p>Mean value of the statistic across the iterations of
the null model</p>
</dd>
<dt><code>Lower.CL</code></dt><dd><p>Lower confidence limit for the metric</p>
</dd>
<dt><code>Upper.CL</code></dt><dd><p>Upper confidence limit for the network metric</p>
</dd>
<dt><code>Test</code></dt><dd><p>Whether the value of the statistic with the observed
network is significantly <strong>higher</strong> than expected under the
null model, <strong>lower</strong> or consistent with the null model
(<strong>ns</strong>)</p>
</dd>
<dt><code>SES</code></dt><dd><p>The standardised effect size for the difference between
the observed network and the null model (see Gotelli &amp; McCabe 2002
for details)</p>
</dd>
</dl>

<p>If <code>index.type = "specieslevel"</code>, a list comprising two data frames for
each statistic, representing the <code>higher</code> and <code>lower</code> levels in
the network. Each data frame has the same format as for
<code>networklevel</code> except that the rows are individual nodes (species)
in the network. See examples for how to call the individual data frames.
</p>


<h3>References</h3>

<p>Dormann, C.F., Gruber B. &amp; Frund, J. (2008). Introducing the
bipartite package: analysing ecological networks. <em>R news</em>,
<strong>8</strong>, 8&ndash;11.
</p>
<p>Dormann, C.F., Frund, J., Bluthgen, N. &amp; Gruber, B. (2009). Indices, graphs
and null models: analyzing bipartite ecological networks. <em>Open
Ecology Journal</em> <strong>2</strong>, 7&ndash;24.
</p>
<p>Dormann, C.F. (2011) How to be a specialist? Quantifying specialisation
in pollination networks. <em>Network Biology</em>, <strong>1</strong>, 1-20.
</p>
<p>Gotelli, N.J. &amp; McCabe, D.J. (2002) Species co-occurrence: a meta-analysis
of J.M. Diamond's assembly rules model. <em>Ecology</em>, <strong>83</strong>,
2091&ndash;2096.
</p>
<p>Manly, B.F.J. (2006) <em>Randomization, Bootstrap and Monte Carlo Methods
in Biology</em> (3rd edn). Chapman &amp; Hall, Boca Raton.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generate_null_net">generate_null_net</a></code>, <code><a href="#topic+plot_bipartite">plot_bipartite</a></code>,
<code><a href="bipartite.html#topic+networklevel">networklevel</a></code>, <code><a href="bipartite.html#topic+grouplevel">grouplevel</a></code>,
<code><a href="bipartite.html#topic+specieslevel">specieslevel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
sil.null &lt;- generate_null_net(Silene[, 2:7], Silene.plants[, 2:6], sims = 10,
                              c.samples = Silene[, 1],
                              r.samples = Silene.plants[, 1])
# Network-level analysis
net.stats &lt;- bipartite_stats(sil.null, index.type = "networklevel",
                             indices = c("linkage density",
                             "weighted connectance", "weighted nestedness",
                             "interaction evenness"), intereven = "sum")
net.stats

# Group-level analysis
grp.stats &lt;- bipartite_stats(sil.null, index.type = "grouplevel",
                             indices = c("generality",
                             "vulnerability", "partner diversity"),
                             logbase = 2)
grp.stats

# Species-level statistics
spp.stats &lt;- bipartite_stats(sil.null, index.type = "specieslevel",
                             indices = c("degree", "normalised degree",
                             "partner diversity"), logbase = exp(1))

spp.stats # Show all data frames of results
spp.stats$normalised.degree # Select one statistic
spp.stats$normalised.degree$lower # Select one statistic at one level

</code></pre>

<hr>
<h2 id='Broadstone'>Part of the highly-resolved food web from Broadstone Stream, UK</h2><span id='topic+Broadstone'></span>

<h3>Description</h3>

<p>Part of the highly-resolved macroinvertebrate food web from Broadstone Stream
in south-east England (see Woodward <em>et al</em>., 2005 for full details of
the web). This data frame contains data collected in August 1996, with 19
macroinvertebrate taxa. Predation was determined via visual gut contents
analysis, with identification of hard parts allowing counts of the number of
individual prey organisms each individual predator consumed.
There are three accompanying data sets:
</p>

<ol>
<li> <p><code>Broadstone.prey</code> Gives the total abundance of each prey taxon
from 30 Surber samples collected at the same time as the interaction data.
</p>
</li>
<li> <p><code>Broadstone.fl</code> Specifies 'forbidden' links which are not
allowed in the null model.
</p>
</li>
<li> <p><code>Broadstone.nodes</code> The abundance and mean body mass for each
taxon in the food web in a format for use alongside a
<code>generate_null_model</code> object with the <code>cheddar</code> package
(Hudson <em>et al</em>., 2013). Body size is taken from cheddar's
'BroadstoneStream' data set.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>Broadstone
</code></pre>


<h3>Format</h3>

<p>A data frame with 319 rows and 20 columns. Each row represents the gut
contents of an individual predator. There are 20 columns in total, with the
first column (<code>Predator</code>) indicating the predator taxon (seven taxa in
total) and the remaining 19 columns indicating the number of individuals eaten
of each of the 19 potential prey taxa.
</p>


<h3>Source</h3>

<p>Woodward, G., Speirs, D.C. &amp; Hildrew, A.G. (2005) Quantification and
resolution of a complex, size-structured food web. <em>Advances in
Ecological Research</em>, <strong>36</strong>, 84&ndash;135.
</p>

<hr>
<h2 id='Broadstone.fl'>'Forbidden' links to accompany the <code>Broadstone</code> data set</h2><span id='topic+Broadstone.fl'></span>

<h3>Description</h3>

<p>Networks often contain 'forbidden links', where interactions are not possible
between two taxa due to factors such as large differences in body size,
spatio-temporal mismatches or specialised flower morphology in flower visitation
networks.  <code>Broadstone.fl</code> summarises the forbidden links in the
<code><a href="#topic+Broadstone">Broadstone</a></code> food web, estimated from the interactions observed
from a complete year's sampling (Table 3 in Woodward <em>et al</em>. 2005). This
table is illustrative, and should not be considered a definitive list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Broadstone.fl
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 columns and seven rows, one row for each of the
seven predator taxa.  The first column (<code>Predator</code>) lists the seven taxa,
whilst the remaining 19 columns are the invertebrate taxa in the food web:
each element indicates whether the predator can consume that taxon (1) or
whether the link is 'forbidden' (0).
</p>


<h3>Source</h3>

<p>Woodward, G., Speirs, D.C. &amp; Hildrew, A.G. (2005) Quantification and
resolution of a complex, size-structured food web. <em>Advances in
Ecological Research</em>, <strong>36</strong>, 84&ndash;135.
</p>

<hr>
<h2 id='Broadstone.nodes'>Abundance and body mass data to accompany the <code>Broadstone</code> data set</h2><span id='topic+Broadstone.nodes'></span>

<h3>Description</h3>

<p>Provides additional information about each macroinvertebrate taxon in the
<code><a href="#topic+Broadstone">Broadstone</a></code> food web for use alongside null model results in the
foodweb package <code>cheddar</code> (Hudson <em>et al</em>. 2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Broadstone.nodes
</code></pre>


<h3>Format</h3>

<p>A data frame with three columns and 19 rows:
</p>

<dl>
<dt>node</dt><dd><p>The name of the macroinvertebrate taxon (node) in the network</p>
</dd>
<dt>M</dt><dd><p>The mean body mass of an individual of the taxon, from
<code>cheddar</code>'s <code>BroadstoneStream</code> object</p>
</dd>
<dt>N</dt><dd><p>The total abundance of each taxon from the 30 Surber samples (see
<code><a href="#topic+Broadstone.prey">Broadstone.prey</a></code> for details)</p>
</dd>
</dl>


<hr>
<h2 id='Broadstone.prey'>Macroinvertebrate abundance data to accompany the <code>Broadstone</code> data frame</h2><span id='topic+Broadstone.prey'></span>

<h3>Description</h3>

<p>The abundance of the 19 macroinvertebrate taxa present in the Broadstone food
web in August 1996 (see Woodward <em>et al</em>. 2005 for full details).
The values are the total number of individuals collected from 30 Surber
samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Broadstone.prey
</code></pre>


<h3>Format</h3>

<p>A data frame with 19 columns and one row, with each column name
representing one taxon. Names match those in <code>Broadstone</code>.
</p>


<h3>Source</h3>

<p>Woodward, G., Speirs, D.C. &amp; Hildrew, A.G. (2005) Quantification and
resolution of a complex, size-structured food web. <em>Advances in
Ecological Research</em>, <strong>36</strong>, 84&ndash;135.
</p>

<hr>
<h2 id='econullnetr'>econullnetr: tools for using null models to analyse the structure of ecological
networks and identify resource selection</h2><span id='topic+econullnetr'></span>

<h3>Description</h3>

<p>econullnetr provides a set of functions for running null models, analysing and
plotting the results, and exporting the outputs for use with other network
packages in R.
</p>


<h3>References</h3>

<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E.,
Woodward, G. &amp; Symondson, W.O.C. (2018) econullnetr: an R package using null
models to analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>

<hr>
<h2 id='expand_matrix'>Expand a summarised interaction matrix</h2><span id='topic+expand_matrix'></span>

<h3>Description</h3>

<p>A simple function for converting interaction matrices that are summarised
at (typically) species-level to individual-level matrices, ready for
use with <code>generate_null_net</code>. This is only applicable to the
special (but common) case where one individual = one interaction
(e.g. many pollination networks, ant-seed networks).
Data can be stored either with consumers as columns and resources as
rows or vice versa. Taxon names for each row in the matrix could either
be stored as the row names of the matrix or data frame (as used, for
example, by the <code>bipartite</code> package), or as a column containing the
names in a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expand_matrix(X, r.names = rownames(X), MARGIN = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expand_matrix_+3A_x">X</code></td>
<td>
<p>A matrix or data frame representing the interaction matrix. This
should only include the interaction data (i.e. counts of different
interactions) and no additional columns for taxon names, covariates, etc</p>
</td></tr>
<tr><td><code id="expand_matrix_+3A_r.names">r.names</code></td>
<td>
<p>An optional object of identical length to the number of
rows in <code>X</code> listing the taxon names. In many situations these may
be the row names of <code>X</code> (the default). Alternatively
<code>r.names</code> can be use to specify a column in a data frame
containing the names or a separate vector.</p>
</td></tr>
<tr><td><code id="expand_matrix_+3A_margin">MARGIN</code></td>
<td>
<p>Similar to <code>apply</code>, an integer value indicating
whether the data are arranged in rows or columns. <code>MARGIN = 1</code>
(the default) indicates that each column relates to one consumer taxon
(the format typically used for bipartite networks), whilst
<code>MARGIN = 2</code> indicates that each row is one consumer taxon, with
column names being the resources.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame where each row represents the interaction observed
between an individual consumer and one resource species. The first
column is named <code>Consumer</code> and records which taxon each indidual
belongs to. The remaining columns represent the resources: one column for each
taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Toy example representing a typical bipartite format.
  bp.inter &lt;- matrix(c(1, 2, 2, 0, 5, 3, 3, 0, 2), nrow = 3, byrow = FALSE,
                     dimnames = list(c("A", "B", "C"),
                                     c("sp1", "sp2", "sp3")))
 bp.inter
 expand_matrix(bp.inter)

# Use a simplified version of the Silene data set, pooling data
# across the 11 visits.
  int.summ &lt;- aggregate(Silene[, 3:7], by = list(Silene$Insect), sum)
  colnames(int.summ)[1] &lt;- "taxon"
  expand_matrix(int.summ[, -1], r.names = int.summ$taxon, MARGIN = 2)

</code></pre>

<hr>
<h2 id='generate_edgelist'>Export null modelling results</h2><span id='topic+generate_edgelist'></span>

<h3>Description</h3>

<p>Exports the observed network alongside the mean interaction strengths
calculated from the null model and the significance test results in a
standard format that can be imported into other network analysis packages
e.g. <code>igraph</code> (Csardi &amp; Nepusz 2006) or <code>cheddar</code> (Hudson
<em>et al</em>. 2013).  This provides access to a wide range of plotting and
analysis tools, especially for non-bipartite networks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_edgelist(
  nullnet,
  signif.level = 0.95,
  export.null = FALSE,
  edge.cols = c("#67A9CF", "#F7F7F7", "#EF8A62")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_edgelist_+3A_nullnet">nullnet</code></td>
<td>
<p>An object of class nullnet from <code>generate_null_net</code></p>
</td></tr>
<tr><td><code id="generate_edgelist_+3A_signif.level">signif.level</code></td>
<td>
<p>An optional value specifying the threshold used for
testing for 'significant' deviations from the null model.  Defaults to 0.95</p>
</td></tr>
<tr><td><code id="generate_edgelist_+3A_export.null">export.null</code></td>
<td>
<p>An optional logical value specifying whether export should
be limited to those interactions that were present in the observed network
or should include any additional interactions present across iterations
of the null network.  Depending upon the data and any forbidden links
specified, additional interactions may be present in the modelled networks.
Defaults to FALSE (only observed interactions are exported).</p>
</td></tr>
<tr><td><code id="generate_edgelist_+3A_edge.cols">edge.cols</code></td>
<td>
<p>An optional character vector of length three specifying
potential colours for network links when used with a suitable plotting
function: in sequence, these should represent i) interactions that are
weaker than expected, ii) consistent with the null model and iii) stronger
than expected. The default is a colourblind friendly blue, white and red
scheme, using Colorbrewer's Red-Blue colour scheme (Brewer 2017).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame where each row represents the interaction observed
between a pair of consumer and resource species, and with the following
column headings:
</p>

<dl>
<dt><code>Consumer</code></dt><dd><p>Name of the consumer species (node)</p>
</dd>
<dt><code>Resource</code></dt><dd><p>Name of the resource species (node)</p>
</dd>
<dt><code>Observed</code></dt><dd><p>Strength of the observed interaction</p>
</dd>
<dt><code>Null</code></dt><dd><p>Mean strength of the interaction across the iterations
of the null model</p>
</dd>
<dt><code>SES</code></dt><dd><p>The standardised effect size for the interaction</p>
</dd>
<dt><code>Test</code></dt><dd><p>Whether the observed interaction is significantly
<strong>stronger</strong> than expected under the null model, <strong>weaker</strong>
or consistent with the null model <strong>ns</strong></p>
</dd>
</dl>



<h3>References</h3>

<p>Brewer, C.A. (2017) <a href="https://colorbrewer2.org/">https://colorbrewer2.org/</a>
</p>
<p>Csardi, G. &amp; Nepusz, T. (2006) The igraph software package for complex
network research. <em>InterJournal, Complex Systems</em>, <strong>1695</strong>.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generate_null_net">generate_null_net</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
bs.null &lt;- generate_null_net(Broadstone, Broadstone.prey,
                             data.type = "counts", sims = 10,
                             r.weights = Broadstone.fl)
BS.export &lt;- generate_edgelist(bs.null, signif.level = 0.95,
                               edge.cols = c("#67A9CF", "black", "#EF8A62"))

if (requireNamespace("igraph", quietly = TRUE)) {
 net.1 &lt;- igraph::graph_from_edgelist(as.matrix(BS.export[, c("Resource",
                                                              "Consumer")]),
                                      directed = TRUE)

 # Add in the null modelling results
 igraph::E(net.1)$obs.str &lt;- BS.export$Observed
 igraph::E(net.1)$test.res &lt;- BS.export$Test
 igraph::E(net.1)$edge.cols &lt;- BS.export$edge.col

 igraph::plot.igraph(net.1, layout = igraph::layout_in_circle,
                     edge.color = igraph::E(net.1)$edge.cols,
                     edge.width = sqrt(igraph::E(net.1)$obs.str),
                     edge.arrow.size = .4)
}

</code></pre>

<hr>
<h2 id='generate_null_net'>Null models for ecological networks</h2><span id='topic+generate_null_net'></span>

<h3>Description</h3>

<p>Uses the algorithm of Agusti <em>et al</em>. (2003) to specify and run a null
model for an ecological network based upon interaction data and
independent estimates of resource abundance (see Vaughan <em>et al</em>.,
2018 for full details).  Typically, network nodes represent species,
and the following documentation uses the term species in place of node,
but this need not be the case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_null_net(
  consumers,
  resources,
  sims = 100,
  data.type = "names",
  maintain.d = NULL,
  summary.type = "sum",
  c.samples = NULL,
  r.samples = NULL,
  r.weights = NULL,
  prog.count = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_null_net_+3A_consumers">consumers</code></td>
<td>
<p>A matrix or data frame containing the interaction data. The
first column should contain the name of the consumer species, with the
remaining column names listing the resources (the names must be identical
and in the same order as in <code>resources</code>).  Each row represents one
individual, with the elements in the matrix indicating whether a resource
was used or how much was used (see Details).</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_resources">resources</code></td>
<td>
<p>A matrix or data frame containing the relative abundances of
the different resources (e.g. density of different prey species or abundance
of different flower species).  May either have one row, if all data came
from the same location and/or time, or have the same number of rows as
there are sampling stratum codes in <code>r.samples</code> and <code>c.samples</code>
(e.g. the set of time points or plot names), if the data are subdivided
across sampling times/sites. Resource names (column names) in
<code>resources</code> and <code>consumers</code> must be identical and in the same
order.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_sims">sims</code></td>
<td>
<p>Number of iterations of the null model.  The default value is 100,
but this should generally be increased to estimate meaningful confidence
limits.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_data.type">data.type</code></td>
<td>
<p>An optional string specifying the type of interaction data.
One of three options: <code>"names"</code> (the default), <code>"counts"</code> or
<code>"quantities"</code>. See Details for a full explanation and examples.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_maintain.d">maintain.d</code></td>
<td>
<p>When <code>data.type = "counts"</code> or <code>"quantities"</code>,
a string indicating whether the degree of each individual consumer (i.e.
the number of different resource species it interacted with) should be
maintained when allocating individual resources/quantities. Default is FALSE.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_summary.type">summary.type</code></td>
<td>
<p>An optional string indicating how the interactions should
be summarised at the species level: one of <code>"sum"</code> (the default),
indicating that the interactions between a consumer and resource species
will be summed across all individuals, or <code>"mean"</code>, indicating that
the mean value per individual of a consumer species will be calculated.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_c.samples">c.samples</code></td>
<td>
<p>An optional vector that specifies names for subdivisions of the
interaction data when data on interactions and resource abundance were
collected across a series of subdivisions (e.g. at different sites or time
points). If suddivisions were present, they should be specified for both
<code>c.samples</code> and <code>r.samples</code>.  The sample names must be identical
in <code>c.samples</code> and <code>r.samples</code> (although <code>c.samples</code> is likely to
be much longer than <code>r.samples</code> due to &gt;1 individual consumer per
subdivision)</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_r.samples">r.samples</code></td>
<td>
<p>An optional vector specifying the sample names for the
resource abundance data (<code>resources</code>), and corresponding to the names
used in <code>c.samples</code>. Must have the name number of entries as there are
rows in <code>resources</code>. Not needed when <code>resources</code> contains a single
row/<code>c.samples</code> is absent.</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_r.weights">r.weights</code></td>
<td>
<p>An optional data frame or matrix specifying weights to be
applied to rows in <code>resources</code>: all entries must be in the range 0&ndash;1.
The first column should contain the consumer species names, one entry for
each species, within the remaining columns listing all species present in
<code>resources</code> (names must be identical and in the same order as in
<code>resources</code>).  Mainly used to specify forbidden links by specifying
zero values for forbidden links and ones for the remaining entries.  Only
one table is supplied, which then applies across all subdivisions of the
data (if present).</p>
</td></tr>
<tr><td><code id="generate_null_net_+3A_prog.count">prog.count</code></td>
<td>
<p>A logical value specifying whether the progress count
should be shown. Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A basic call to <code>generate_null_net</code> only needs two arguments:
<code>consumers</code> and <code>resources</code>, but it is recommended that
<code>sims</code> is also specified to run a larger number of iteration of the
null model (e.g. 1000).  It is important to ensure that species names are
consistent throughout and that the resource species used as column names in
<code>resources</code>, <code>consumers</code> and (optionally) <code>r.weights</code> are in
the same order (both <code>consumers</code> and <code>r.weights</code> should be one
column wider than <code>resources</code>, because they include an extra column
at the start to list the consumer species).
</p>
<p>The same species can appear as both a consumer and a resource e.g. when a
species is both predator and prey in a food web.  Interactions can be
excluded from the modelled networks by specifying forbidden links with
<code>r.weights</code>, based either on existing data/literature or hypothesed
choices, generating the network that would be created if those choices were
made.  Placing limits on the feasible range of resources with which a
consumer interacts should lead to more realistic networks: great white
sharks not feeding on zooplankton in a marine food web, for example.
</p>
<p>The interactions between individual consumers and the resources may be
recorded in a range of different ways depending upon the empirical data
that are available, and these differences are handled by using the
optional <code>data.type</code> argument.  Three types of data can be specified:
</p>

<ol>
<li> <p><code>data.type = "names"</code> (the default value). This is the most
common type of data, recording one or more resource species with which
an interaction occurred, but without any attempt to quantify the
strength of the interaction.  For data of this type, <code>resources</code>
should simply comprise 1s and 0s, indicating whether an interaction was
present or absent respectively: row sums may equal one or be &gt;1 if
individual consumers can interact with multiple resources.
</p>
</li>
<li> <p><code>data.type = "counts"</code>.  These data record the number of times
an individual consumer interacted with different resource species e.g.
the number of times different flower species were visited by an
individual pollinator during a 5 minute observation period.  When
modelling count data there is a choice of whether the total number of
interactions (an individual's row sum) can be distributed across all
potential resources, which may result in interactions with a different
number of resource species (i.e. a change in an individual's degree) or
whether the degree is held constant for each individual.  This is
determined by the additional <code>maintain.d</code> argument:
<code>maintain.d = FALSE</code> (the default) does not constrain an
individual's degree, whereas <code>maintain = TRUE</code> does.
</p>
</li>
<li> <p><code>data.type = "quantities"</code>. Quantitative data are obtained
from each individual, such as the proportion of a predator's gut
contents derived from different prey.  The data in <code>consumers</code> can
be represented by either proportions (so that the row sum for an
individual = 1) or in the native units (so the total &lt;&gt; 1). As for count
data, the degree can be free or fixed at the individual consumer level
by using the <code>maintain.d</code> argument.
</p>
</li></ol>

<p>For data types 2 and 3, it does not matter what units are used (e.g. each
row does not need to add up to one).  The row total will be maintained,
so the results are returned in the same units as the original data.
</p>
<p>One problem that may arise in specifying the null model is in situations
where an interaction is recorded with a particular resource, but that
resource was not actually detected in the abundance data (i.e. abundance =
0).  This may occur, for example, when a predator eats a rare species that
was missed during concomitant sampling of prey abundance: in effect, the
predator's 'sampling' was more comprehensive.  The implication of this is
that the resource species will not be sampled in the null model &ndash; a
potential source of bias.  <code>generate_null_net</code> will issue a warning if
it detects this situation.  Possible corrective actions include removing
that resource species altogether or replacing its zero abundance with a
small constant.
</p>


<h3>Value</h3>

<p>Returns an object of class &quot;nullnet&quot;, which is a list containing
the following components:
</p>

<dl>
<dt><code>rand.data</code></dt><dd><p>Data frame containing the results from all of the
iterations of the null model</p>
</dd>
<dt><code>obs.interactions</code></dt><dd><p>Interaction matrix summarising the observed
interactions (from <code>consumers</code>)</p>
</dd>
<dt><code>n.iterations</code></dt><dd><p>The value of <code>sims</code> i.e. the number of
interations of the null model</p>
</dd>
</dl>



<h3>References</h3>

<p>Agusti, N., Shayler, S.P., Harwood, J.D., Vaughan, I.P.,
Sunderland, K.D. &amp; Symondson, W.O.C. (2003) Collembola as alternative prey
sustaining spiders in arable ecosystems: prey detection within predators
using molecular markers. <em>Molecular Ecology</em>, <strong>12</strong>, 3467&ndash;3475.
</p>
<p>Davey, J.S., Vaughan, I.P., King, R.A., Bell, J.R., Bohan, D.A.,
Bruford, M.W., Holland, J.M. &amp; Symondson, W.O.C. (2013) Intraguild
predation in winter wheat: prey choice by a common epigeal carabid
consuming spiders. <em>Journal of Applied Ecology</em>, <strong>50</strong>,
271&ndash;279.
</p>
<p>King, R.A, Vaughan, I.P., Bell, J.R., Bohan, D.A, &amp; Symondson, W.O.C.
(2010) Prey choice by carabid beetles feeding on an earthworm community
analysed using species- and lineage-specific PCR primers. <em>Molecular
Ecology</em>, <strong>19</strong>, 1721&ndash;1732.
</p>
<p>Pearson, C.E., Symondson, W.O.C., Clare, E.L., Ormerod, S.J.,
Iparraguirre Bolanos, E. &amp; Vaughan, I.P. (2018) The effects of
pastoral intensification on the feeding interactions of generalist
predators in streams. <em>Molecular Ecology</em>, <strong>27</strong>, 590-602.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+test_interactions">test_interactions</a></code>, <code><a href="#topic+plot_preferences">plot_preferences</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>null.1 &lt;- generate_null_net(Silene[, 2:7], Silene.plants[, 2:6], sims = 10,
  data.type = "names", summary.type = "sum", c.samples = Silene[, 1],
  r.samples = Silene.plants[, 1])

</code></pre>

<hr>
<h2 id='plot_bipartite'>Plot a bipartite network, colour coding individual links according to
whether they are stronger or weaker than expected under the null model</h2><span id='topic+plot_bipartite'></span>

<h3>Description</h3>

<p>Acts as a wrapper for the <code>bipartite</code> package's
<code><a href="bipartite.html#topic+plotweb">plotweb</a></code> function (Dormann <em>et al</em>., 2008),
colour coding the links in the familiar bipartite plots according to
whether they are stronger, weaker or consistent with the null model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bipartite(
  nullnet,
  signif.level = 0.95,
  edge.cols = c("#67A9CF", "#F7F7F7", "#EF8A62"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_bipartite_+3A_nullnet">nullnet</code></td>
<td>
<p>An object of class &quot;nullnet&quot; from <code>generate_null_net</code></p>
</td></tr>
<tr><td><code id="plot_bipartite_+3A_signif.level">signif.level</code></td>
<td>
<p>An optional value specifying the threshold used for
testing for 'significant' deviations from the null model.  Defaults to 0.95</p>
</td></tr>
<tr><td><code id="plot_bipartite_+3A_edge.cols">edge.cols</code></td>
<td>
<p>An optional character vector of length three specifying
the colours for links in the bipartite plot: they should represent
interactions that are weaker than expected, consistent with the null model
and stronger than expected in that order.  The default is a colourblind
friendly blue, white and red scheme, using colorbrewer's Red-Blue
colour scheme (Brewer 2017).</p>
</td></tr>
<tr><td><code id="plot_bipartite_+3A_...">...</code></td>
<td>
<p>Other arguments to be supplied to <code>bipartite</code>'s
<code>plotweb</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extensive options can be passed to <code><a href="bipartite.html#topic+plotweb">plotweb</a></code>
to customise the network plot beyond the colour coding of the links. See
the appropriate help file in the bipartite package for details.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>References</h3>

<p>Brewer, C.A. (2017) <a href="https://colorbrewer2.org/">https://colorbrewer2.org/</a>
</p>
<p>Dormann, C.F., Gruber B. &amp; Frund, J. (2008). Introducing the bipartite
package: analysing ecological networks. <em>R news</em>, <strong>8</strong>, 8-11.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generate_null_net">generate_null_net</a></code>, <code><a href="#topic+bipartite_stats">bipartite_stats</a></code>,
<code><a href="bipartite.html#topic+plotweb">plotweb</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Run the null model
set.seed(1234)
sil.null &lt;- generate_null_net(Silene[, 2:7], Silene.plants[, 2:6], sims = 10,
                              c.samples = Silene[, 1],
                              r.samples = Silene.plants[, 1])

# Basic plot
plot_bipartite(sil.null)

# With alternative colour scheme and nodes width in the lower level proportional
#  to mean floral abundance
mean.abunds &lt;- colMeans(Silene.plants[, 2:6])
plot_bipartite(sil.null, signif.level = 0.95, edge.cols = c("#67a9cf",
               "#F7F7F7", "#ef8a62"), low.abun = mean.abunds,
               abuns.type = "independent", low.abun.col = "black",
               high.abun.col = "black", high.lablength = 0, low.lablength = 0)

</code></pre>

<hr>
<h2 id='plot_preferences'>Plot the resource preferences of a consumer</h2><span id='topic+plot_preferences'></span>

<h3>Description</h3>

<p>Takes a 'nullnet' object from running a null model with
<code>generate_null_net</code> and plots the observed and expected link
strengths for every resource for a selected consumer species (node in
the network).  There are two styles of plot: the default is a dot plot
based on <code><a href="graphics.html#topic+dotchart">dotchart</a></code>, whilst the alternative is a
bar plot based on <code><a href="graphics.html#topic+barplot">barplot</a></code>.  There are arguments in
<code>plot_preferences</code> to set some of the graphical parameters, but see
the respective help files for <code><a href="graphics.html#topic+dotchart">dotchart</a></code> and
<code><a href="graphics.html#topic+barplot">barplot</a></code> for further options to customise the plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_preferences(
  nullnet,
  node,
  signif.level = 0.95,
  style = "dots",
  type = "counts",
  res.col = c("#67A9CF", "#F7F7F7", "#EF8A62"),
  res.order = NULL,
  l.cex = 1,
  p.cex = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_preferences_+3A_nullnet">nullnet</code></td>
<td>
<p>An object of class nullnet from <code>generate_null_net</code></p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_node">node</code></td>
<td>
<p>A string specifying the consumer node (species) whose
preferences will be plotted</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_signif.level">signif.level</code></td>
<td>
<p>An optional value specifying the threshold used for
testing for 'significant' deviations from the null model.  Defaults to 0.95</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_style">style</code></td>
<td>
<p>An optional string to set whether a dotchart or bar chart is
plotted. The default (<code>style = "dots"</code>) is a Cleveland dot plot,
whilst the alternative (<code>style = "bars"</code>) is a bar chart of the type
used by King <em>et al.</em> (2010) and Davey <em>et al.</em> (2013).</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_type">type</code></td>
<td>
<p>Optional string to specify how preferences are displayed. The
default (<code>type = "counts"</code>) is the total number of interactions or
mean interaction strength, depending upon which one was requested in the
original call to <code><a href="#topic+generate_null_net">generate_null_net</a></code> using the
<code>summary.type</code> argument.  The alternative is the standardised effect
size (<code>type = "SES"</code>).  With <code>type = "counts"</code>,
confidence limits will be drawn, whilst for the standardised effect
size, dashed lines are added at +2 and -2, which is approximately equivalent
to a 5% significance level: fill colours for <code>type = "SES"</code> are
based on the estimated confidence limits, rather than the +2/-2 thresholds.</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_res.col">res.col</code></td>
<td>
<p>An optional character vector of length three specifying
the colours with which to fill the dots or bars representing interactions
that are weaker than expected, consistent with the null model and stronger
than expected (in that order). The default is a red-blue colour scheme.</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_res.order">res.order</code></td>
<td>
<p>An optional data frame used to set the order in which the
resource species are plotted.  Should have two columns: the first listing
the resource species (names must be identical to those used by
<code>generate_null_net</code>) and the second the plotting order (bars will be
plotted in ascending order)</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_l.cex">l.cex</code></td>
<td>
<p>An optional numeric value to set the size of the labels when
using <code>style = "dots"</code>. Ignored when <code>style = "bars"</code>, as
<code>barplot</code>'s <code>cex.names</code> argument can be used.</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_p.cex">p.cex</code></td>
<td>
<p>An optional numeric value to set the size of the points when
when using <code>style = "dots"</code>.</p>
</td></tr>
<tr><td><code id="plot_preferences_+3A_...">...</code></td>
<td>
<p>Other arguments to control basic plotting functions in R, such as
<code>lwd</code>, <code>xlab</code>, <code>ylab</code>, <code>main</code>, <code>xlim</code> and
<code>ylim</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the preferences for individual consumer species. The bar plot
format follows the basic style of King <em>et al</em>. (2010; Figure 3) and
Davey <em>et al.</em> (2013; Figure 3).
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>References</h3>

<p>Brewer, C.A. (2017) <a href="https://colorbrewer2.org/">https://colorbrewer2.org/</a>
</p>
<p>Davey, J.S., Vaughan, I.P., King, R.A., Bell, J.R., Bohan, D.A., Bruford,
M.W., Holland, J.M. &amp; Symondson, W.O.C. (2013) Intraguild predation in
winter wheat: prey choice by a common epigeal carabid consuming spiders.
<em>Journal of Applied Ecology</em>, <strong>50</strong>, 271&ndash;279.
</p>
<p>King, R.A, Vaughan, I.P., Bell, J.R., Bohan, D.A, &amp; Symondson, W.O.C.
(2010) Prey choice by carabid beetles feeding on an earthworm community
analysed using species- and lineage-specific PCR primers. <em>Molecular
Ecology</em>, <strong>19</strong>, 1721&ndash;1732.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generate_null_net">generate_null_net</a></code>, <code><a href="#topic+plot_bipartite">plot_bipartite</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>null.1 &lt;- generate_null_net(WelshStreams[, 2:18], WelshStreams.prey[, 2:17],
                            sims = 10, c.samples = WelshStreams[, 1],
                            r.samples = WelshStreams.prey[, 1])

# Basic plots, showing the dot and bar plot styles. Increased lower margin
#   on the bar plot so that names fit
plot_preferences(null.1, "Dinocras", signif.level = 0.95, type = "counts",
                 xlab = "Num. of visits", p.cex = 1.2, lwd = 2)

op &lt;- par(mar = c(9, 4, 4, 2) + 0.1)
plot_preferences(null.1, "Dinocras", style = "bars", signif.level = 0.95,
                 type = "counts", ylab = "Num. of visits")
par(op)

# Same results, this time showing the standardised effect sizes
plot_preferences(null.1, "Rhyacophila", signif.level = 0.95,
                 type = "SES", xlab = "SES")

op &lt;- par(mar = c(9, 4, 4, 2) + 0.1)
plot_preferences(null.1, "Rhyacophila", signif.level = 0.95, style = "bars",
                 type = "SES", ylab = "SES")
par(op)

</code></pre>

<hr>
<h2 id='Silene'>Flower visitation network from an arable field in the UK</h2><span id='topic+Silene'></span>

<h3>Description</h3>

<p>A flower visitation network from an arable field site in the UK, notable for
the presence of small-flowered catchfly <em>Silene gallica</em>, a rare arable
weed species in the UK.  Flower visitors and floral abundance were sampled on
11 occasions in the summer of 2002, with all insect flower visitors (31 taxa)
recorded on the flower species where they were observed. Gibson <em>et al</em>.
(2006) provide full details of the data and analysis of the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Silene
</code></pre>


<h3>Format</h3>

<p>A data frame with 128 rows and seven columns, one row for each individual
pollinator. The first column (<code>Visit</code>) represents the sampling occasion
on which the pollinator was collected (V1&ndash;V11), the second column
(<code>Insect</code>) represents the flower visitor taxon, and the remaining five
columns are the five flower species, which were either visited (1) or not (0)
by each insect.
</p>


<h3>Source</h3>

<p>Gibson, R.H., Nelson, I.L., Hopkins, G.W., Hamlett, B.J. &amp;
Memmott, J. (2006) Pollinator webs, plant communities and the conservation
of rare plants: arable weeds as a case study. <em>Journal of Applied
Ecology</em>, <strong>43</strong>, 246&ndash;257.
</p>

<hr>
<h2 id='Silene.plants'>Flower abundance data to accompany the <em>Silene</em> visitation network</h2><span id='topic+Silene.plants'></span>

<h3>Description</h3>

<p>Floral abundance data for the five flower species recorded in the Silene flower
visitation network (see <code><a href="#topic+Silene">Silene</a></code>) for full details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Silene.plants
</code></pre>


<h3>Format</h3>

<p>A data frame with 11 rows and six columns. Each row represents the
floral abundance on one sampling occasion. The first column (<code>Visit</code>)
lists the codes for the sampling visit (V1&ndash;V11), whilst the remaining five
variables represent the abundance of each of the flower species.
</p>


<h3>Source</h3>

<p>Gibson, R.H., Nelson, I.L., Hopkins, G.W., Hamlett, B.J. &amp;
Memmott, J. (2006) Pollinator webs, plant communities and the conservation
of rare plants: arable weeds as a case study. <em>Journal of Applied
Ecology</em>, <strong>43</strong>, 246&ndash;257.
</p>

<hr>
<h2 id='test_interactions'>Compare observed interaction strengths in a network to those estimated from
a null model</h2><span id='topic+test_interactions'></span>

<h3>Description</h3>

<p>Takes the result of running a null model with <code>generate_null_net</code> and
tests whether the observed interactions between consumer species and
resource species differ those expected under the null model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_interactions(nullnet, signif.level = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="test_interactions_+3A_nullnet">nullnet</code></td>
<td>
<p>An object of class &quot;nullnet&quot; from <code>generate_null_net</code></p>
</td></tr>
<tr><td><code id="test_interactions_+3A_signif.level">signif.level</code></td>
<td>
<p>An optional value specifying the threshold used for
testing for 'significant' deviations from the null model.  Defaults to 0.95</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Statistical significance is determined for each consumer-resource
interaction according to whether the observed interaction strength falls
outside the confidence limits calculated across the iterations of the null
model.  Confidence limits are calculated as the 1 &ndash; alpha/2 percentiles from
the frequency distribution (Manly 2006).
</p>
<p>The observed and expected interactions strengths are also compared by
calculating the standardised effect size  (Gotelli &amp; McCabe 2002):
</p>
<p style="text-align: center;"><code class="reqn">(observed link strength - expected link strength) / standard deviation
  of the link strength across the iterations of the null model</code>
</p>

<p><code>test_interactions</code> will issue warnings when:
</p>

<ol>
<li><p>The number of iterations of the null model was small &lt;100, as
the confidence intervals are unlikely to be reliable
</p>
</li>
<li><p>The number of tests &gt;50, due to the increasing risk of Type I
errors (incorrectly denoting an interaction as significantly different
from the null model).  Many networks will contain many more than 100
potential interactions, so the significance of individual interactions
should be treated with caution.  Some form of false discovery rate
correction may be valuable (e.g. the local false discovery rate;
Gotelli &amp; Ulrich 2010).
</p>
</li></ol>



<h3>Value</h3>

<p>Returns a data frame listing all possible consumer and resource
species combinations with the following column headings:
</p>

<dl>
<dt><code>Consumer</code></dt><dd><p>The name of the consumer species</p>
</dd>
<dt><code>Resource</code></dt><dd><p>The name of the resource species</p>
</dd>
<dt><code>Observed</code></dt><dd><p>The 'strength' of the observed interaction (e.g. total number
of interactions summed across the individual consumers)</p>
</dd>
<dt><code>Null</code></dt><dd><p>The mean strength of the interaction across the iterations of the
null model</p>
</dd>
<dt><code>Lower</code></dt><dd><p>Lower confidence limit for the interaction strength</p>
</dd>
<dt><code>Upper</code></dt><dd><p>Upper confidence limit for the interaction strength</p>
</dd>
<dt><code>Test</code></dt><dd><p>Whether the observed interaction is significantly
<strong>stronger</strong> than expected under the null model, <strong>weaker</strong>
or consistent with the null model (<strong>ns</strong>)</p>
</dd>
<dt><code>SES</code></dt><dd><p>The standardised effect size for the interaction</p>
</dd>
</dl>



<h3>References</h3>

<p>Gotelli, N.J. &amp; McCabe, D.J. (2002) Species co-occurrence:
a meta-analysis of J. M. Diamond's assembly rules model. <em>Ecology</em>,
<strong>83</strong>, 2091&ndash;2096.
</p>
<p>Gotelli, N.J. &amp; Ulrich, W. (2010) The empirical Bayes approach as a tool
to identify non-random species associations. <em>Oecologia</em>,
<strong>162</strong>, 463&ndash;477.
</p>
<p>Manly, B.F.J. (2006) <em>Randomization, Bootstrap and Monte Carlo
Methods in Biology</em> (3rd edn). Chapman &amp; Hall, Boca Raton.
</p>
<p>Vaughan, I.P., Gotelli, N.J., Memmott, J., Pearson, C.E., Woodward, G. &amp;
Symondson, W.O.C. (2018) econullnetr: an R package using null models to
analyse the structure of ecological networks and identify resource
selection. <em>Methods in Ecology and Evolution</em>, <strong>9</strong>, 728&ndash;733.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+generate_null_net">generate_null_net</a></code>, <code><a href="#topic+plot_preferences">plot_preferences</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>null.1 &lt;- generate_null_net(WelshStreams[, 2:18], WelshStreams.prey[, 2:17],
                            sims = 10, c.samples = WelshStreams[, 1],
                            r.samples = WelshStreams.prey[, 1])
test_interactions(null.1, 0.95)

</code></pre>

<hr>
<h2 id='WelshStreams'>Part of the food web from upland streams in south Wales, UK</h2><span id='topic+WelshStreams'></span>

<h3>Description</h3>

<p>Part of a food web from upland streams in south Wales, UK, comprising 85
individuals of two macroinvertebrate predator species (the caddisfly
<em>Rhyacophila dorsalis</em> and the stonefly <em> Dinocras cephalotes</em>).
Data were collected in Dec 2013 from six streams spread across an
agricultural intensity gradient as part of a wider study (Pearson
<em>et al</em>. 2018). The data comprise presence or absence of predation
by each predator on 16 potential prey taxa determined using next
generation sequencing of predator gut contents.
There are three accompanying data sets:
</p>

<ol>
<li> <p><code><a href="#topic+WelshStreams.prey">WelshStreams.prey</a></code>, which gives the mean abundance of each
prey taxon in each of the six streams (counts of individuals from 3&ndash;min
kick samples)
</p>
</li>
<li> <p><code><a href="#topic+WelshStreams.fl">WelshStreams.fl</a></code> which specifies forbidden links (one
forbidden link for each predator species, ruling out cannibalism)
</p>
</li>
<li> <p><code><a href="#topic+WelshStreams.order">WelshStreams.order</a></code> which ranks the 16 prey taxa in
taxonomic order according to the Centre for Ecology and Hydrology's
Coded Macroinvertebrate List. This is used for plotting the results.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>WelshStreams
</code></pre>


<h3>Format</h3>

<p>A data frame with 85 rows and 18 variables. Each row represents the gut
contents of an individual predator. The first column <code>Stream</code> indicates
which of the six streams each predator was collected from, whilst the second
column indicates which genus the predator belonged to (<em>Dinocras</em> or
<em>Rhyacophila</em>). The remaining 16 columns represent the potential prey
taxa, which were either predated (1) or not (0) by each individual predator.
</p>


<h3>Source</h3>

<p>Pearson, C.E., Symondson, W.O.C., Clare, E.L., Ormerod, S.J.,
Iparraguirre Bolanos, E. &amp; Vaughan, I.P. (2018) The effects of pastoral
intensification on the feeding interactions of generalist predators in
streams. <em>Molecular Ecology</em>, <strong>27</strong>, 590-602.
</p>

<hr>
<h2 id='WelshStreams.fl'>'Forbidden' links to accompany the upland streams food web</h2><span id='topic+WelshStreams.fl'></span>

<h3>Description</h3>

<p>A table specifying trophic links that are allowed when modelling part of the
food web from upland Welsh stream containing the two macroinvetebrate
predators <em>Rhyacophila dorsalis</em> and <em>Dinocras cephalotes</em>
(see <code><a href="#topic+WelshStreams">WelshStreams</a></code>).  There is one forbidden link for each
predator species, preventing cannibalism from being included in the null
model: this is because the method used to screen predator gut contents for
prey (next generation sequencing) cannot distinguish between predator and
prey of the same species.  Both predators are generalists, so all other
trophic links are permitted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WelshStreams.fl
</code></pre>


<h3>Format</h3>

<p>A data frame with 2 rows and 17 columns, each row representing one of the
two predator genera. The first column (<code>Predator</code>) indicates the predator
genus, whilst columns 2&ndash;17 indicate which of the 16 potential prey taxa can be
consumed (1) or are forbidden (0).
</p>


<h3>Source</h3>

<p>Pearson, C.E., Symondson, W.O.C., Clare, E.L., Ormerod, S.J.,
Iparraguirre Bolanos, E. &amp; Vaughan, I.P. (2018) The effects of pastoral
intensification on the feeding interactions of generalist predators in
streams. <em>Molecular Ecology</em>, <strong>27</strong>, 590-602.
</p>

<hr>
<h2 id='WelshStreams.order'>Additional data to assist with plotting outputs from <code><a href="#topic+WelshStreams">WelshStreams</a></code></h2><span id='topic+WelshStreams.order'></span>

<h3>Description</h3>

<p>The rank order of 16 prey taxa in <code><a href="#topic+WelshStreams">WelshStreams</a></code> on the Centre for
Ecology and Hydrology's Coded Macroinvertebrate List
<a href="https://www.ceh.ac.uk/services/coded-macroinvertebrates-list">https://www.ceh.ac.uk/services/coded-macroinvertebrates-list</a>. This is
used primarily for controlling the order in which prey taxa are plotted
with <code><a href="#topic+plot_preferences">plot_preferences</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WelshStreams.order
</code></pre>


<h3>Format</h3>

<p>A data frame with 16 rows and two columns:
</p>

<dl>
<dt>Taxon</dt><dd><p>The names of each prey taxon</p>
</dd>
<dt>CEH</dt><dd><p>The rank order of the taxa based on the Centre for Ecology and
Hydrology's Coded Macroinvertebrate List</p>
</dd>
</dl>



<h3>Source</h3>

<p>Pearson, C.E., Symondson, W.O.C., Clare, E.L., Ormerod, S.J.,
Iparraguirre Bolanos, E. &amp; Vaughan, I.P. (2018) The effects of pastoral
intensification on the feeding interactions of generalist predators in
streams. <em>Molecular Ecology</em>, <strong>27</strong>, 590-602.
</p>

<hr>
<h2 id='WelshStreams.prey'>Macroinvertebrate abundance to accompany part of the food web from upland
streams in south Wales, UK</h2><span id='topic+WelshStreams.prey'></span>

<h3>Description</h3>

<p>The abundance of 16 potential macroinvertebrate prey taxa to accompany the
food web data in <code><a href="#topic+WelshStreams">WelshStreams</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WelshStreams.prey
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 17 variables. Each row represents the
abundance of individual macroinvertebrate taxa in each of the six streams.
Column 1 contains the stream codes, whilst columns 2&ndash;17 represent the
abundance of each macroinvertebrate taxon summed across three 1-minute
kick-samples in each stream.
</p>


<h3>Source</h3>

<p>Pearson, C.E., Symondson, W.O.C., Clare, E.L., Ormerod, S.J.,
Iparraguirre Bolanos, E. &amp; Vaughan, I.P. (2018) The effects of pastoral
intensification on the feeding interactions of generalist predators in
streams. <em>Molecular Ecology</em>, <strong>27</strong>, 590-602.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
