<!DOCTYPE html><html><head><title>Help for package ecospace</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ecospace}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ecospace-package'><p>ecospace: Simulating Community Assembly and Ecological Diversification Using</p>
Ecospace Frameworks</a></li>
<li><a href='#calc_metrics'><p>Calculate Ecological Disparity (Functional Diversity) Dynamics as Function of</p>
Sample Size.</a></li>
<li><a href='#create_ecospace'><p>Create Ecospace Framework.</p></a></li>
<li><a href='#expansion'><p>Use Expansion Rule to Simulate Ecological Diversification of a Biota.</p></a></li>
<li><a href='#KWTraits'><p>Species-by-Trait Matrix for Late Ordovician Marine Fossils.</p></a></li>
<li><a href='#neutral'><p>Use Neutral Rule to Simulate Ecological Diversification of a Biota.</p></a></li>
<li><a href='#partitioning'><p>Use Partitioning Rule to Simulate Ecological Diversification of a Biota.</p></a></li>
<li><a href='#prep_data'><p>Internal Ecospace Functions.</p></a></li>
<li><a href='#rbind_listdf'><p>Combine Large Lists of Data Frames.</p></a></li>
<li><a href='#redundancy'><p>Use Redundancy Rule to Simulate Ecological Diversification of a Biota.</p></a></li>
<li><a href='#sample2'><p>Internal Ecospace Functions.</p></a></li>
<li><a href='#unique2'><p>Internal Ecospace Functions.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulating Community Assembly and Ecological Diversification
Using Ecospace Frameworks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-06-12</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>FD (&ge; 1.0-12)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>vegan, knitr, rmarkdown, data.table</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Phil Novack-Gottshall &lt;pnovack-gottshall@ben.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements stochastic simulations of community assembly (ecological
    diversification) using customizable ecospace frameworks (functional trait
    spaces). Provides a wrapper to calculate common ecological disparity and
    functional ecology statistical dynamics as a function of species richness.
    Functions are written so they will work in a parallel-computing environment.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/publicdomain/zero/1.0/legalcode">CC0</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/pnovack-gottshall/ecospace">https://github.com/pnovack-gottshall/ecospace</a>,
<a href="http://www.ben.edu/faculty/pnovack-gottshall/index.html">http://www.ben.edu/faculty/pnovack-gottshall/index.html</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/pnovack-gottshall/ecospace/issues">https://github.com/pnovack-gottshall/ecospace/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-06-13 17:31:43 UTC; pnovack-gottshall</td>
</tr>
<tr>
<td>Author:</td>
<td>Phil Novack-Gottshall [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-06-13 17:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ecospace-package'>ecospace: Simulating Community Assembly and Ecological Diversification Using
Ecospace Frameworks</h2><span id='topic+ecospace-package'></span><span id='topic+ecospace'></span>

<h3>Description</h3>

<p>ecospace is an R package that implements stochastic simulations of community
assembly (ecological diversification) using customizable ecospace frameworks
(functional trait spaces). Simulations model the 'neutral', 'redundancy',
'partitioning', and 'expansion' models of Bush and Novack-Gottshall (2012)
and Novack-Gottshall (2016a,b). It provides a wrapper to calculate common
ecological disparity and functional ecology statistical dynamics as a
function of species richness. Functions are written so they will work in a
parallel-computing environment.
</p>


<h3>Details</h3>

<p>The package also contains a sample data set, functional traits for Late
Ordovician (Type Cincinnatian) fossil species from the Kope and Waynesville
formations, from Novack-Gottshall (2016b).
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bush, A. and P.M. Novack-Gottshall. 2012. Modelling the
ecological-functional diversification of marine Metazoa on geological time
scales. <em>Biology Letters</em> 8: 151-155.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>See Also</h3>

<p>The 'calc_metrics' function relies extensively on the functional
diversity package <code><a href="FD.html#topic+FD-package">FD-package</a></code>, and hence lists this package as a
depends, so it is loaded simultaneously.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get package version, citation, updates, and vignette
packageVersion("ecospace")
citation("ecospace")
news(package = "ecospace")
vignette("ecospace")

# Create an ecospace framework (functional-trait space) with 15 characters
#    (functional traits) of mixed types
nchar &lt;- 15
ecospace &lt;- create_ecospace(nchar = nchar, char.state = rep(3, nchar),
  char.type = rep(c("factor", "ord.fac", "ord.num"), nchar / 3))

# Use to assemble a stochastic "neutral" sample of 20 species (from
#    initial seeding by 5 species)
x &lt;- neutral(Sseed = 5, Smax = 20, ecospace = ecospace)
head(x, 10)

# Calculate ecological disparity (functional diversity) dynamics as a
#    function of species richness
# Statistic 'V' [total variance] not calculated because there are factors
#    in the sample
metrics &lt;- calc_metrics(samples = x, Smax = 20, Model = "Neutral", Param = "NA")
metrics

# Plot statistical dynamics as function of species richness
op &lt;- par()
par(mfrow = c(2,4), mar = c(4, 4, 1, .3))
attach(metrics)
plot(S, H, type = "l", cex = .5)
plot(S, D, type = "l", cex = .5)
plot(S, M, type = "l", cex = .5)
plot(S, V, type = "l", cex = .5)
plot(S, FRic, type = "l", cex = .5)
plot(S, FEve, type = "l", cex = .5)
plot(S, FDiv, type = "l", cex = .5)
plot(S, FDis, type = "l", cex = .5)

par(op)

</code></pre>

<hr>
<h2 id='calc_metrics'>Calculate Ecological Disparity (Functional Diversity) Dynamics as Function of
Sample Size.</h2><span id='topic+calc_metrics'></span>

<h3>Description</h3>

<p>Wrapper to <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> that calculates common ecological
disparity and functional diversity statistics. When used with species-wise
simulations of community assembly or ecological diversification (and default
<code>increm = 'TRUE'</code>), calculates statistical dynamics incrementally as a
function of species richness. Avoids file-sharing errors so that can be used
in 'embarrasingly parallel' implementations in a high-performance computing
environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_metrics(
  nreps = 1,
  samples = NA,
  Smax = NA,
  Model = "",
  Param = "",
  m = 3,
  corr = "lingoes",
  method = "Euclidean",
  increm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_metrics_+3A_nreps">nreps</code></td>
<td>
<p>Sample number to calculate statistics for. Default is the first
sample <code>nreps = 1</code>, but statistics can be calculated for other samples
(i.e., second sample if <code>nreps = 2</code>), or multiple samples if assigned a
vector (sequence) of integers and function is applied within <code>lapply</code>
or related function.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_samples">samples</code></td>
<td>
<p>Data frame (if <code>nreps = 1</code>) or list of data frames (if
<code>nreps = seq()</code> or <code>nreps! = 1</code>), with each data frame a
species-by-trait matrix with species as rows and traits as columns. Traits
can be binary, numeric, ordered numeric, factor, or ordered factor types.
Each sample is converted to a distance metric (see <code>method</code> below)
before calculating statistics.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of <code>samples</code> rows (species) to include in
calculations, incremented starting with first row. Default (<code>NA</code>) is to
increment to the maximum number of <code>samples</code> rows (calculated
separately for each data frame sample, if a list of data frames). If
<code>Smax</code> is greater than the size of a sample, then calculation stops
after calculating the sample statistics and issues a warning.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_model">Model</code></td>
<td>
<p>Optional character string or numeric value naming simulation
model. A warning issues if left blank.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_param">Param</code></td>
<td>
<p>Optional numeric value or character string naming strength
parameter used in simulation. A warning issues if left blank.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_m">m</code></td>
<td>
<p>The number of PCoA axes to keep as 'traits' for calculating FRic and
FDiv in <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code>. Default <code>m = 3</code> is justified below,
but any integer value greater than 1 is possible. See 'details' for more
information.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_corr">corr</code></td>
<td>
<p>Character string specifying the correction method to use in
<code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> when the species-by-species distance matrix
cannot be represented in a Euclidean space. Default <code>corr = 'lingoes'</code> is
justified below, but see <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> for other possible
values.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_method">method</code></td>
<td>
<p>Distance measure to use when calculating functional distances
between species. Default is <code>method = 'Euclidean'</code> using
<code>stats::<a href="stats.html#topic+dist">dist</a></code>. <code>method = 'Gower'</code> or any other value
uses Gower distance (using <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code>). Presence of factor
or ordered factor character types forces use of Gower distance, triggering a
warning to notify user when changed internally.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_increm">increm</code></td>
<td>
<p>Default <code>increm = 'TRUE'</code> calculates statistics incrementally
as a function of species richness. <code>increm = 'FALSE'</code> only calculates a
single set of statistics for the entire sample.</p>
</td></tr>
<tr><td><code id="calc_metrics_+3A_...">...</code></td>
<td>
<p>Additional parameters for controlling <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code>.
Common uses include setting <code>calc.FRic = FALSE</code> or <code>calc.FDiv = FALSE</code>
to exclude calculation of FRic and FDiv. Note that the arguments <code>m</code>,
<code>corr</code>, and <code>method</code> above have different defaults than used in
<code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code>, and <code>w.abun = FALSE</code> and
<code>messages = FALSE</code> are also internally changed to different defaults.
These and others can be controlled here.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The primary goal of this function is to describe the statistical
dynamics of common ecological disparity (functional diversity) metrics as a
function of species richness (sample size). Statistics are calculated
incrementally within samples, first for the first row (species), second for
the first and second rows, ..., ending with the entire sample (by default,
or terminating with <code>Smax</code> total species). The function assumes that
supplied samples are ecologically or evolutionary cohesive assemblages in
which there is a logical order to the rows (such that the sixth row is the
sixth species added to the assemblage) and that such incremental
calculations are sensible. See Novack-Gottshall (2016a,b) for additional
context. Samples must have species as rows and traits as columns (of many
allowed character types), and have <code>class(data.frame)</code> or a list of
such data frames, with each data frame a separate sample.
</p>
<p>Statistics calculated include four widely used in ecological disparity
studies (adapted from studies of morphological disparity) and four used in
functional diversity studies. See Foote (1993), Ciampaglio et al. (2001),
and Wills (2001) for definitions and details on morphological disparity
measures and Novack-Gottshall (2007; 2016a,b) for implementation as measures
of ecological disparity. See Mason et al. (2005), Anderson et al. (2006),
Villeger et al. (2008), Laliberte and Legendre (2010), Mouchet et al.
(2010), Mouillot et al. (2013) for definitions and details on functional
diversity statistics. For computation details of functional diversity
metrics, see Laliberte and Shipley (2014) package FD, and especially
<code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code>, which this function wraps around to calculate
the functional diversity statistics.
</p>
<p>Statistic (<code>S</code>) is species (taxonomic) richness, or sample size.
</p>
<p>When <code>increm = 'FALSE'</code>, the function calculates statistics for the
entire sample(s) instead of doing so incrementally. In this case, the
implementation is essentially the same as <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> with
default arguments (e.g., <code>m, corr</code>) that reduce common calculation
errors, plus inclusion of common morphological disparity statistics.
</p>
<p><strong>Statistics that measure diversity (unique number of life habits /
trait combinations within ecospace / functional-trait space):</strong> </p>

<dl>
<dt>H</dt><dd><p>Life habit richness, the number of functionally unique trait
combinations.</p>
</dd>
</dl>
<p><strong>Statistics that measure disparity (or dispersion of species within
ecospace / functional-trait space) (Note these statistics are sensitive to
outliers and sample size):</strong> </p>
<dl>
<dt>M</dt><dd><p>Maximum pairwise distance between
species in functional-trait space, measured using the distance <code>method</code>
specified above.</p>
</dd> <dt>FRic</dt><dd><p>Functional richness, the minimal convex-hull
volume in multidimensional principal coordinates analysis (PCoA) trait-space
ordination.</p>
</dd><dt>FDiv</dt><dd><p>Functional divergence, the mean distance of species
from the PCoA trait-space centroid.</p>
</dd>  </dl>

<p><strong>Statistics that measure internal structure (i.e., clumping or
inhomogeneities within the trait-space):</strong> </p>
 <dl>
<dt>D</dt><dd><p>Mean pairwise
distance between species in functional-trait space, measured using the
distance <code>method</code> specified above.</p>
</dd> <dt>V</dt><dd><p>Total variance, the sum of
variances for each functional trait across species; when using factor or
ordered factor character types, this statistic cannot be measured and is
left blank, with a warning.</p>
</dd> <dt>FDis</dt><dd><p>Functional dispersion, the total
deviance of species from the circle with radius equal to mean distance from
PCoA trait-space centroid.</p>
</dd> </dl>

<p><strong>Statistics that measure spacing among species within the
trait-space:</strong> </p>
 <dl>
<dt>FEve</dt><dd><p>Functional evenness, the evenness of
minimum-spanning-tree lengths between species in PCoA trait-space.</p>
</dd> </dl>

<p>The default number of PCoA axes used in calculating of FRic and FDiv equals
<code>m = 3</code>. Because their calculation requires more species than traits
(here the <code>m = 3</code> PCoA axes), the four functional diversity statistics
are only calculated when a calculated sample contains a minimum of <code>m</code>
species (S) or unique life habits (H). <code>qual.FRic</code> is appended to the
output to record the proportion ('quality') of PCoA space retained by this
loss of dimensionality. Although including more PCoA axes allows greater
statistical power (Villeger et al. 2011, Maire et al. 2015), the use of
<code>m = 3</code> here is computationally manageable, ecologically meaningful, and
allows standardized measurement of statistical dynamics across the wide
range of sample sizes typically involved in simulations of
ecological/evolutionary assemblages, especially when functionally redundant
data occur. Other integers greater than 1 can also be specified. See the
help file for <code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> for additional information.
</p>
<p>Lingoes correction <code>corr = 'lingoes'</code>, as recommended by Legendre and
Anderson (1999), is called when the species-by-species distance matrix
cannot be represented in a Euclidean space. See the help file for
<code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> for additional information.
</p>
<p>Note that the ecological disparity statistics are calculated on the raw
(unstandardized) distance matrix. The functional diversity statistics are
calculated on standardized data using standardizations in
<code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code>. If all traits are numeric, they by default are
standardized to mean 0 and unit variance. If not all traits are numeric,
Gower's (1971) standardization by the range is automatically used.
</p>


<h3>Value</h3>

<p>Returns a data frame (if <code>nreps</code> is a single integer or
<code>samples</code> is a single data frame) or a list of data frames. Each
returned data frame has <code>Smax</code> rows corresponding to incremental
species richness (sample size) and 12 columns, corresponding to:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>(optional) <code>Model</code> name</p>
</td></tr> <tr><td><code>Param</code></td>
<td>
<p>(optional) strength
parameter</p>
</td></tr> <tr><td><code>S</code></td>
<td>
<p>Species richness (sample size)</p>
</td></tr> <tr><td><code>H</code></td>
<td>
<p>Number of
functionally unique life habits</p>
</td></tr> <tr><td><code>D</code></td>
<td>
<p>Mean pairwise distance</p>
</td></tr>
<tr><td><code>M</code></td>
<td>
<p>Maximum pairwise distance</p>
</td></tr> <tr><td><code>V</code></td>
<td>
<p>Total variance</p>
</td></tr>
<tr><td><code>FRic</code></td>
<td>
<p>Functional richness</p>
</td></tr> <tr><td><code>FEve</code></td>
<td>
<p>Functional evenness</p>
</td></tr>
<tr><td><code>FDiv</code></td>
<td>
<p>Functional divergence</p>
</td></tr> <tr><td><code>FDis</code></td>
<td>
<p>Functional dispersion</p>
</td></tr>
<tr><td><code>qual.FRic</code></td>
<td>
<p>proportion ('quality') of total PCoA trait-space used when
calculating FRic and FDiv</p>
</td></tr>
</table>


<h3>Note</h3>

<p>A bug exists within <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code> where nearest-neighbor
distances can not be calculated when certain characters (especially numeric
characters with values other than 0 and 1) share identical traits across
species. The nature of the bug is under investigation, but the current
implementation is reliable under most uses. If you run into problems because
of this bug, a work-around is to manually change the function to call
<code>cluster::<a href="cluster.html#topic+daisy">daisy</a></code> using <code>metric = "gower"</code> instead.
</p>
<p>If calculating statistics for more than several hundred samples, it is
recommended to use a parallel-computing environment. The function has been
written to allow usage (using <code><a href="base.html#topic+lapply">lapply</a></code> or some other list-apply
function) in 'embarrassingly parallel' implementations in such HPC
environments. Most importantly, overwriting errors during calculation of
convex hull volume in FRic are avoided by creating CPU-specific temporarily
stored vertices files.
</p>
<p>See Novack-Gottshall (2016b) for recommendations for using random forest
classification trees to conduct multi-model inference.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Anderson, M. J., K. E. Ellingsen, and B. H. McArdle. 2006.
Multivariate dispersion as a measure of beta diversity. <em>Ecology
Letters</em> 9(6):683-693.
</p>
<p>Ciampaglio, C. N., M. Kemp, and D. W. McShea. 2001. Detecting
changes in morphospace occupation patterns in the fossil record:
characterization and analysis of measures of disparity. <em>Paleobiology</em>
27(4):695-715.
</p>
<p>Foote, M. 1993. Discordance and concordance between morphological
and taxonomic diversity. <em>Paleobiology</em> 19:185-204.
</p>
<p>Gower, J. C. 1971. A general coefficient of similarity and some of
its properties. <em>Biometrics</em> 27:857-871.
</p>
<p>Laliberte, E., and P. Legendre. 2010. A distance-based framework
for measuring functional diversity from multiple traits. <em>Ecology</em>
91(1):299-305.
</p>
<p>Legendre, P., and M. J. Anderson. 1999. Distance-based redundancy
analysis: testing multispecies responses in multifactorial ecological
experiments. <em>Ecological Monographs</em> 69(1):1-24.
</p>
<p>Maire, E., G. Grenouillet, S. Brosse, and S. Villeger. 2015. How
many dimensions are needed to accurately assess functional diversity? A
pragmatic approach for assessing the quality of functional spaces.
<em>Global Ecology and Biogeography</em> 24(6):728-740.
</p>
<p>Mason, N. W. H., D. Mouillot, W. G. Lee, and J. B. Wilson. 2005.
Functional richness, functional evenness and functional divergence: the
primary components of functional diversity. <em>Oikos</em> 111(1):112-118.
</p>
<p>Mouchet, M. A., S. Villeger, N. W. H. Mason, and D. Mouillot.
2010. Functional diversity measures: an overview of their redundancy and
their ability to discriminate community assembly rules. <em>Functional
Ecology</em> 24(4):867-876.
</p>
<p>Mouillot, D., N. A. J. Graham, S. Villeger, N. W. H. Mason, and D.
R. Bellwood. 2013. A functional approach reveals community responses to
disturbances. <em>Trends in Ecology and Evolution</em> 28(3):167-177.
</p>
<p>Novack-Gottshall, P.M. 2007. Using a theoretical ecospace to
quantify the ecological diversity of Paleozoic and modern marine biotas.
<em>Paleobiology</em> 33: 274-295.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>
<p>Villeger, S., N. W. H. Mason, and D. Mouillot. 2008. New
multidimensional functional diversity indices for a multifaceted framework
in functional ecology. <em>Ecology</em> 89(8):2290-2301.
</p>
<p>Villeger, S., P. M. Novack-Gottshall, and D. Mouillot. 2011. The
multidimensionality of the niche reveals functional diversity changes in
benthic marine biotas across geological time. <em>Ecology Letters</em>
14(6):561-568.
</p>
<p>Wills, M. A. 2001. Morphological disparity: a primer. Pp. 55-143.
<em>In</em> J. M. Adrain, G. D. Edgecombe, and B. S. Lieberman, eds.
<em>Fossils, phylogeny, and form: an analytical approach.</em> Kluwer
Academic/Plenum Publishers, New York.
</p>
<p>Laliberte, E., and B. Shipley. 2014. <em>FD: Measuring
functional diversity from multiple traits, and other tools for functional
ecology</em>, Version 1.0-12.
</p>


<h3>See Also</h3>

<p><code>FD::<a href="FD.html#topic+dbFD">dbFD</a></code> for details on the core function wrapped
here for calculating functional diversity statistics. <code><a href="#topic+neutral">neutral</a></code>,
<code><a href="#topic+redundancy">redundancy</a></code>, <code><a href="#topic+partitioning">partitioning</a></code>,
<code><a href="#topic+expansion">expansion</a></code> for building samples using simulations.
<code><a href="#topic+rbind_listdf">rbind_listdf</a></code> for efficient way to combine lists of data frames
for subsequent analyses.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Build ecospace framework and a random 50-species sample using neutral rule:
ecospace &lt;- create_ecospace(nchar = 15, char.state = rep(3, 15), char.type = rep("numeric", 15))
sample &lt;- neutral(Sseed = 5, Smax = 50, ecospace = ecospace)
# Using Smax = 10 here for fast example
metrics &lt;- calc_metrics(samples = sample, Smax = 10, Model = "Neutral", Param = "NA")
metrics

# Plot statistical dynamics as function of species richness
op &lt;- par()
par(mfrow = c(2,4), mar = c(4, 4, 1, .3))
attach(metrics)
plot(S, H, type = "l", cex = .5)
plot(S, D, type = "l", cex = .5)
plot(S, M, type = "l", cex = .5)
plot(S, V, type = "l", cex = .5)
plot(S, FRic, type = "l", cex = .5)
plot(S, FEve, type = "l", cex = .5)
plot(S, FDiv, type = "l", cex = .5)
plot(S, FDis, type = "l", cex = .5)

par(op)

# Argument 'increm' switches between incremental and entire-sample calculation
metrics2 &lt;- calc_metrics(samples = sample, Smax = 10, Model = "Neutral",
                         Param = "NA", increm = FALSE)
metrics2
identical(tail(metrics, 1), metrics2) # These are identical

# ... can further control 'FD::dbFD', here turning off calculation of FRic and FDiv
metrics3 &lt;- calc_metrics(samples = sample, Smax = 10, Model = "Neutral",
                         Param = "NA", calc.FRic = FALSE, calc.FDiv = FALSE)
metrics3
rbind(metrics[10, ], metrics3[10, ])

## Not run: 
# Can take a few minutes to run to completion
# Calculate for 5 samples
nreps &lt;- 1:5
samples &lt;- lapply(X = nreps, FUN = neutral, Sseed = 5, Smax = 50, ecospace)
metrics &lt;- lapply(X = nreps, FUN = calc_metrics, samples = samples,
                  Model = "Neutral", Param = "NA")
alarm()
str(metrics)

## End(Not run)
</code></pre>

<hr>
<h2 id='create_ecospace'>Create Ecospace Framework.</h2><span id='topic+create_ecospace'></span>

<h3>Description</h3>

<p>Create ecospace frameworks (functional trait spaces) of specified structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ecospace(
  nchar,
  char.state,
  char.type,
  char.names = NA,
  state.names = NA,
  constraint = Inf,
  weight.file = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ecospace_+3A_nchar">nchar</code></td>
<td>
<p>Number of life habit characters (functional traits).</p>
</td></tr>
<tr><td><code id="create_ecospace_+3A_char.state">char.state</code></td>
<td>
<p>Numeric vector of number of character states in each
character.</p>
</td></tr>
<tr><td><code id="create_ecospace_+3A_char.type">char.type</code></td>
<td>
<p>Character string listing type for each character. See
'Details' for explanation. Allowed types include: </p>
 <ul>
<li>
<p><code>numeric</code> for numeric and binary characters, </p>
</li>
<li> <p><code>ord.num</code> for
ordered numeric characters, </p>
</li>
<li> <p><code>ord.fac</code> for ordered factor
characters, or </p>
</li>
<li> <p><code>factor</code> for factor characters. </p>
</li></ul>
</td></tr>
<tr><td><code id="create_ecospace_+3A_char.names">char.names</code></td>
<td>
<p>Optional character string listing character names.</p>
</td></tr>
<tr><td><code id="create_ecospace_+3A_state.names">state.names</code></td>
<td>
<p>Optional character string listing character state names.</p>
</td></tr>
<tr><td><code id="create_ecospace_+3A_constraint">constraint</code></td>
<td>
<p>Positive integer specifying the maximum number of &quot;multiple
presences&quot; to allow if using multistate binary/numeric character types. The
default <code>Inf</code> allows all possible permutations (except for &quot;all
absences&quot;). See 'Details' for additional explanation.</p>
</td></tr>
<tr><td><code id="create_ecospace_+3A_weight.file">weight.file</code></td>
<td>
<p>Data frame (species X trait matrix) or a vector (of mode
numeric, integer, or array) of relative weights for ecospace
character-state probabilities. Default action omits such probabilities and
creates equal weighting among states. If a data frame is supplied, the
first three columns must be (1) class [or similar taxonomic identifier],
(2) genus, and (3) species names (or three dummy columns that will be
ignored by algorithm).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function specifies the data structure for a theoretical
ecospace framework used in Monte Carlo simulations of ecological
diversification. An ecospace framework (functional trait space) is a
multidimensional data structure describing how organisms interact with
their environments, often summarized by a list of the individual life habit
characters (functional traits) inhabited by organisms. Commonly used
characters describe diet and foraging habit, mobility, microhabitat, among
others, with the individual diets, modes of locomotions, and microhabitats
as possible character states.  When any combination of character states is
allowed, the framework serves as a theoretical ecospace; actually occurring
life-habit combinations circumscribe the realized ecospace.
</p>
<p>Arguments <code>nchar, char.state, char.type</code> specify the number and types
of characters and their states. Character names and state names are
optional, and assigned using numeric names (i.e., character 1, character 2,
etc.) if not provided. The function returns an error if the number of
states and names is different than numbers specified in provided arguments.
</p>
<p>Allowed character types include the following: </p>
 <ul>
<li>
<p><code>numeric</code> for numeric and binary characters, whether present/absent or
multistate. See below for examples and more discussion on these
implementations. </p>
</li>
<li> <p><code>ord.num</code> for ordered numeric values, whether
discrete or continuous. Examples include body size, metabolic rate, or
temperature tolerance. States are pulled as sorted unique levels from
<code>weight.file</code>, if provided. </p>
</li>
<li> <p><code>ord.fac</code> for ordered factor
characters (factors with a specified order). An example is mobility:
habitual &gt; intermittent &gt; facultative &gt; passive &gt; sedentary. (If wish to
specify relative distances between these ordered factors, it is best to use
an ordered numeric character type instead). </p>
</li>
<li> <p><code>factor</code> for
discrete, unordered factors (e.g., diet can have states of autotrophic,
carnivorous, herbivorous, or microbivorous).</p>
</li></ul>

<p>Binary characters can be treated individually (e.g., states of
present = 1/absent = 0) or can be treated as multiple binary character states.
For example, the character 'reproduction' could be treated as including two
states [sexual, asexual] with exclusively sexual habits coded as [0,1],
exclusively asexual as [1,0], and hermaphrodites as [1,1]. The
<code>constraint</code> argument allows additional control of such combinations.
Setting <code>constraint = 2</code> only allows a maximum of &quot;two-presence&quot;
combinations (e.g., [1,0,0], [0,1,0], [0,0,1], [1,1,0], [1,0,1], and
[0,1,1]) as state combinations, but excludes [1,1,1]; setting
<code>constraint = 1</code> only allows the first three of these combinations; the
default behavior (<code>Inf</code>) allows all of these combinations. In all
cases, the nonsensical &quot;all-absence&quot; state combination [0,0,0] is
disallowed.
</p>
<p>Character states can be weighted using the optional <code>weight.file</code>.
This is useful so that random draws of life habits (functional-trait
combinations) from the ecospace framework are biased in specified ways. If
not provided, the default action assigns equal weighting among states. If a
vector of mode array, integer, or numeric is provided, character states (or
character-state combinations, if multistate binary) are assigned the
specified relative weight. The function returns an error if the supplied
vector has a length different that the number of states allowed.
</p>
<p>If a data frame is supplied for the weight file (such as a species-by-trait
matrix, with species as rows and traits as columns, describing a regional
species pool), weights are calculated according to the observed relative
frequency of states in this pool. If such a data frame is supplied, the
first three columns must be (1) class [or similar taxonomic identifier],
(2) genus, and (3) species names, although these can be left blank. In all
cases, character state probabilities are calculated separately within each
character (including only those allowed by the specified
<code>constraint</code>).
</p>


<h3>Value</h3>

<p>Returns a list of class <code>ecospace</code> describing the structure of
the theoretical ecospace framework needed for running simulations. The list
has a length equal to <code>nchar + 1</code>, with one list component for each
character, plus a final list component recording constraints used in
producing allowable character states.
</p>
<p>Each character component has the following list components:</p>

<dl>
<dt><code>char</code></dt><dd><p>character name.</p>
</dd><dt><code>type</code></dt><dd><p>character type.</p>
</dd>
<dt><code>char.space</code></dt><dd><p>data frame listing each allowable state
combination in each row, the calculated proportional weight (<code>pro</code>),
frequency (<code>n</code>) of observed species with such state combination in
species pool (<code>weight.file</code>, if supplied).</p>
</dd>
<dt><code>allowed.combos</code></dt><dd><p>allowed character state combinations allowed
by <code>constraint</code> and <code>weight.file</code>, if supplied.</p>
</dd></dl>

<p>The last component lists the following components:</p>

<dl>
<dt><code>constraint</code></dt><dd><p><code>constraint</code> argument
used.</p>
</dd><dt><code>wts</code></dt><dd><p>vector of character-state weights used.</p>
</dd>
<dt><code>pool</code></dt><dd><p>species by trait matrix used in assigning
character-state weights, if supplied. Note that this matrix may differ from
that supplied as <code>weight.file</code> when, for example, the supplied file
includes character-state combinations not allowed by <code>constraint]</code>. It
also excludes taxonomic identifiers (class, genus, species).</p>
</dd></dl>



<h3>Note</h3>

<p>If you have trouble matching the characters with <code>char.state</code> and
<code>char.type</code>, see <code>data.frame</code> in first example for easy way to
trouble-shoot. If you have trouble supplying correct length of
<code>char.name, state.name</code> and <code>weight.file</code>, consider producing an
ecospace framework with defaults first, then using these to supply custom
names and weights.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bambach, R. K. 1983. Ecospace utilization and guilds in marine
communities through the Phanerozoic. Pp. 719-746. <em>In</em> M. J. S.
Tevesz, and P. L. McCall, eds. <em>Biotic Interactions in Recent and
Fossil Benthic Communities</em>. Plenum, New York.
</p>
<p>Bambach, R. K. 1985. Classes and adaptive variety: the ecology of
diversification in marine faunas through the Phanerozoic. Pp. 191-253.
<em>In</em> J. W. Valentine, ed. <em>Phanerozoic Diversity Patterns:
Profiles in Macroevolution</em>. Princeton University Press, Princeton, NJ.
</p>
<p>Bambach, R. K., A. M. Bush, and D. H. Erwin. 2007. Autecology and
the filling of ecospace: key metazoan radiations. <em>Palaeontology</em>
50(1):1-22.
</p>
<p>Bush, A. M. and R. K. Bambach. 2011. Paleoecologic megatrends in
marine Metazoa. <em>Annual Review of Earth and Planetary Sciences</em>
39:241-269.
</p>
<p>Bush, A. M., R. K. Bambach, and G. M. Daley. 2007. Changes in
theoretical ecospace utilization in marine fossil assemblages between the
mid-Paleozoic and late Cenozoic. <em>Paleobiology</em> 33(1):76-97.
</p>
<p>Bush, A. M., R. K. Bambach, and D. H. Erwin. 2011. Ecospace
utilization during the Ediacaran radiation and the Cambrian eco-explosion.
Pp. 111-134. <em>In</em> M. Laflamme, J. D. Schiffbauer, and S. Q. Dornbos,
eds. <em>Quantifying the Evolution of Early Life: Numerical Approaches to
the Evaluation of Fossils and Ancient Ecosystems</em>. Springer, New York.
</p>
<p>Novack-Gottshall, P.M. 2007. Using a theoretical ecospace to
quantify the ecological diversity of Paleozoic and modern marine biotas.
<em>Paleobiology</em> 33: 274-295.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create random ecospace framework with all character types
set.seed(88)
nchar &lt;- 10
char.state &lt;- rpois(nchar, 1) + 2
char.type &lt;- replace(char.state, char.state &lt;= 3, "numeric")
char.type &lt;- replace(char.type, char.state == 4, "ord.num")
char.type &lt;- replace(char.type, char.state == 5, "ord.fac")
char.type &lt;- replace(char.type, char.state &gt; 5, "factor")
# Good practice to confirm everything matches expectations:
data.frame(char = seq(nchar), char.state, char.type)
ecospace &lt;- create_ecospace(nchar, char.state, char.type, constraint = Inf)
ecospace

# How many life habits in this ecospace are theoretically possible?
seq &lt;- seq(nchar)
prod(sapply(seq, function(seq) length(ecospace[[seq]]$allowed.combos)))
# ~12 million

# Observe effect of constraint for binary characters
create_ecospace(1, 4, "numeric", constraint = Inf)[[1]]$char.space
create_ecospace(1, 4, "numeric", constraint = 2)[[1]]$char.space
create_ecospace(1, 4, "numeric", constraint = 1)[[1]]$char.space
try(create_ecospace(1, 4, "numeric", constraint = 1.5)[[1]]$char.space) # ERROR!
try(create_ecospace(1, 4, "numeric", constraint = 0)[[1]]$char.space) # ERROR!

# Using custom-weighting for traits (singletons weighted twice as frequent
#   as other state combinations)
weight.file &lt;- c(rep(2, 3), rep(1, 3), 2, 2, 1, rep(1, 4), rep(2, 3), rep(1, 3),
rep(1, 14), 2, 2, 1, rep(1, 4), rep(2, 3), rep(1, 3), rep(1, 5))
create_ecospace(nchar, char.state, char.type, constraint = 2,
  weight.file = weight.file)

# Bambach's (1983, 1985) classic ecospace framework
# 3 characters, all factors with variable states
nchar &lt;- 3
char.state &lt;- c(3, 4, 4)
char.type &lt;- c("ord.fac", "factor", "factor")
char.names &lt;- c("Tier", "Diet", "Activity")
state.names &lt;- c("Pelag", "Epif", "Inf", "SuspFeed", "Herb", "Carn", "DepFeed",
  "Mobile/ShallowActive", "AttachLow/ShallowPassive", "AttachHigh/DeepActive",
  "Recline/DeepPassive")
ecospace &lt;- create_ecospace(nchar, char.state, char.type, char.names, state.names)
ecospace
seq &lt;- seq(nchar)
prod(sapply(seq, function(seq) length(ecospace[[seq]]$allowed.combos)))
# 48 possible life habits

# Bush and Bambach's (Bambach et al. 2007, bush et al. 2007) updated ecospace
#   framework, with Bush et al. (2011) and Bush and Bambach (2011) addition of
#   osmotrophy as a possible diet category
#   3 characters, all factors with variable states
nchar &lt;- 3
char.state &lt;- c(6, 6, 7)
char.type &lt;- c("ord.fac", "ord.fac", "factor")
char.names &lt;- c("Tier", "Motility", "Diet")
state.names &lt;- c("Pelag", "Erect", "Surfic", "Semi-inf", "ShallowInf", "DeepInf",
  "FastMotile", "SlowMotile ", "UnattachFacMot", "AttachFacMot", "UnattachNonmot",
  "AttachNonmot", "SuspFeed", "SurfDepFeed", "Mining", "Grazing", "Predation",
  "Absorpt/Osmotr", "Other")
ecospace &lt;- create_ecospace(nchar, char.state, char.type, char.names, state.names)
ecospace
seq &lt;- seq(nchar)
prod(sapply(seq, function(seq) length(ecospace[[seq]]$allowed.combos)))
# 252 possible life habits

# Novack-Gottshall (2007) ecospace framework, updated in Novack-Gottshall (2016b)
#   Fossil species pool from Late Ordovician (Type Cincinnatian) Kope and
#   Waynesville Formations, with functional-trait characters coded according
#   to Novack-Gottshall (2007, 2016b)
data(KWTraits)
head(KWTraits)
nchar &lt;- 18
char.state &lt;- c(2, 7, 3, 3, 2, 2, 5, 5, 2, 5, 2, 2, 5, 2, 5, 5, 3, 3)
char.type &lt;- c("numeric", "ord.num", "numeric", "numeric", "numeric", "numeric",
  "ord.num", "ord.num", "numeric", "ord.num", "numeric", "numeric", "ord.num",
  "numeric", "ord.num", "numeric", "numeric", "numeric")
char.names &lt;- c("Reproduction", "Size", "Substrate composition", "Substrate
  consistency", "Supported", "Attached", "Mobility", "Absolute tier", "Absolute
  microhabitat", "Relative tier", "Relative microhabitat", "Absolute food
  microhabitat", "Absolute food tier", "Relative food microhabitat", "Relative
  food tier", "Feeding habit", "Diet", "Food condition")
state.names &lt;- c("SEXL", "ASEX", "BVOL", "BIOT", "LITH", "FLUD", "HARD", "SOFT",
  "INSB", "SPRT", "SSUP", "ATTD", "FRLV", "MOBL", "ABST", "AABS", "IABS", "RLST",
  "AREL", "IREL", "FAAB", "FIAB", "FAST", "FARL", "FIRL", "FRST", "AMBT", "FILT",
  "ATTF", "MASS", "RAPT", "AUTO", "MICR", "CARN", "INCP", "PART", "BULK")
ecospace &lt;- create_ecospace(nchar, char.state, char.type, char.names, state.names,
  constraint = 2, weight.file = KWTraits)
ecospace
seq &lt;- seq(nchar)
prod(sapply(seq, function(seq) length(ecospace[[seq]]$allowed.combos)))
# ~57 billion life habits

ecospace &lt;- create_ecospace(nchar, char.state, char.type, char.names, state.names,
  constraint = Inf)
ecospace
seq &lt;- seq(nchar)
prod(sapply(seq, function(seq) length(ecospace[[seq]]$allowed.combos)))
# ~3.6 trillion life habits

</code></pre>

<hr>
<h2 id='expansion'>Use Expansion Rule to Simulate Ecological Diversification of a Biota.</h2><span id='topic+expansion'></span>

<h3>Description</h3>

<p>Implement Monte Carlo simulation of a biota undergoing ecological
diversification using the expansion rule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expansion(nreps = 1, Sseed, Smax, ecospace, method = "Euclidean", strength = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expansion_+3A_nreps">nreps</code></td>
<td>
<p>Vector of integers (such as a sequence) specifying sample number
produced. Only used when function is applied within <code>lapply</code> or
related function. Default <code>nreps=1</code> or any other integer produces a
single sample.</p>
</td></tr>
<tr><td><code id="expansion_+3A_sseed">Sseed</code></td>
<td>
<p>Integer giving number of species (or other taxa) to use at start
of simulation.</p>
</td></tr>
<tr><td><code id="expansion_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of species (or other taxa) to include in
simulation.</p>
</td></tr>
<tr><td><code id="expansion_+3A_ecospace">ecospace</code></td>
<td>
<p>An ecospace framework (functional trait space) of class
<code>ecospace</code>.</p>
</td></tr>
<tr><td><code id="expansion_+3A_method">method</code></td>
<td>
<p>Distance measure to use when calculating functional distances
between species. Default is <code>Euclidean</code> using
<code>stats::<a href="stats.html#topic+dist">dist</a></code>. <code>Gower</code> or any other value uses
Gower distance (using <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code>). Presence of factor or
ordered factor character types forces use of Gower distance.</p>
</td></tr>
<tr><td><code id="expansion_+3A_strength">strength</code></td>
<td>
<p>Strength parameter controlling probability that expansion
rule is followed during simulation. Values must range between
<code>strength = 1</code> (default, rules always implemented) and <code>strength = 0</code>
(rules never implemented).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulations are implemented as Monte Carlo processes in which
species are added iteratively to assemblages, with all added species having
their character states specified by the model rules, here the 'expansion'
rule. Simulations begin with the seeding of <code>Sseed</code> number of species,
chosen at random (with replacement) from either the species pool (if
provided in the <code>weight.file</code> when building the ecospace framework
using <code>create_ecospace</code>) or following the neutral-rule algorithm (if a
pool is not provided). Once seeded, the simulations proceed iteratively
(character-by-character, species-by-species) by following the appropriate
algorithm, as explained below, until terminated at <code>Smax</code>.
</p>
<p><strong>Expansion rule algorithm:</strong>  Measure distances between all pairs of
species, using <code>Euclidean</code> or <code>Gower</code> distance method specified
by <code>method</code> argument. Identify species pair that is maximally distant.
If multiple pairs are equally maximally distant, one pair is chosen at
random. The newly added species has traits that are equal to or more
extreme than those in this species pair, with probability of following the
expansion rule determined by the <code>strength</code> parameter. Default
<code>strength = 1</code> always implements the rule, whereas <code>strength = 0</code>
never implements it (essentially making the simulation follow the
<code><a href="#topic+neutral">neutral</a></code> rule.)
</p>
<p>Each newly assigned character is compared with the ecospace framework
(<code>ecospace</code>) to confirm that it is an allowed state combination before
proceeding to the next character. If the newly built character is
disallowed from the ecospace framework (i.e., because it has &quot;dual
absences&quot; [0,0], has been excluded based on the species pool
[<code>weight.file</code> in <code>create_ecospace</code>], or is not allowed by the
ecospace <code>constraint</code> parameter), then the character-selection
algorithm is repeated until an allowable character is selected. When
simulations proceed to very large sample sizes (&gt;100), this confirmatory
process can require long computational times, and produce &quot;new&quot; species
that are functionally identical to pre-existing species. This can occur,
for example, when no life habits, or perhaps only one, exist that forms an
allowable novelty between the selected neighbors.
</p>
<p>Expansion rules tend to produce ecospaces that progressively expand into
more novel regions. Additional details on the expansion simulation are
provided in Novack-Gottshall (2016a,b), including sensitivity to
ecospace framework (functional trait space) structure, recommendations for
model selection, and basis in ecological and evolutionary theory.
</p>


<h3>Value</h3>

<p>Returns a data frame with <code>Smax</code> rows (representing species) and
as many columns as specified by number of characters/states (functional
traits) in the ecospace framework. Columns will have the same data type
(numeric, factor, ordered numeric, or ordered factor) as specified in the
ecospace framework.
</p>


<h3>Note</h3>

<p>A bug exists within <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code> where nearest-neighbor
distances can not be calculated when certain characters (especially numeric
characters with values other than 0 and 1) share identical traits across
species. The nature of the bug is under investigation, but the current
implementation is reliable under most uses. If you run into problems
because of this bug, a work-around is to manually change the function to
call <code>cluster::<a href="cluster.html#topic+daisy">daisy</a></code> using <code>metric = "gower"</code>
instead.
</p>
<p>The function has been written to allow usage (using <code><a href="base.html#topic+lapply">lapply</a></code> or
some other list-apply function) in 'embarrassingly parallel' implementations
in a high-performance computing environment.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bush, A. and P.M. Novack-Gottshall. 2012. Modelling the
ecological-functional diversification of marine Metazoa on geological time
scales. <em>Biology Letters</em> 8: 151-155.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_ecospace">create_ecospace</a></code>, <code><a href="#topic+neutral">neutral</a></code>,
<code><a href="#topic+redundancy">redundancy</a></code>, <code><a href="#topic+partitioning">partitioning</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an ecospace framework with 15 3-state factor characters
# Can also accept following character types: "numeric", "ord.num", "ord.fac"
nchar &lt;- 15
ecospace &lt;- create_ecospace(nchar = nchar, char.state = rep(3, nchar),
  char.type = rep("factor", nchar))

# Single (default) sample produced by expansion function (with strength = 1):
Sseed &lt;- 5
Smax &lt;- 40
x &lt;- expansion(Sseed = Sseed, Smax = Smax, ecospace = ecospace)
head(x, 10)

# Plot results, showing order of assembly
# (Seed species in red, next 5 in black, remainder in gray)
# Notice that new life habits progressively expand outward into previously
#   unoccupied portions of ecospace
seq &lt;- seq(nchar)
types &lt;- sapply(seq, function(seq) ecospace[[seq]]$type)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Expansion model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Change strength parameter so rules followed 95% of time:
x &lt;- expansion(Sseed = Sseed, Smax = Smax, ecospace = ecospace, strength = 0.95)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Expansion model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Create 4 samples using multiple nreps and lapply (can be slow)
nreps &lt;- 1:4
samples &lt;- lapply(X = nreps, FUN = expansion, Sseed = 5, Smax = 40, ecospace)
str(samples)

</code></pre>

<hr>
<h2 id='KWTraits'>Species-by-Trait Matrix for Late Ordovician Marine Fossils.</h2><span id='topic+KWTraits'></span>

<h3>Description</h3>

<p>Sample data set of life habit codings (functional traits) for fossil taxa
from the Late Ordovician (Type Cincinnatian) Kope and Waynesville Formations
from Ohio, Indiana, and Kentucky (U.S.A.). The faunal list was compiled from
the <a href="https://paleobiodb.org/">Paleobiology Database</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KWTraits
</code></pre>


<h3>Format</h3>

<p>A data frame with 237 rows (taxa) and 40 columns (3 taxonomic
identifiers and 37 functional traits): </p>
<dl>
<dt>Class</dt><dd><p>Taxonomic
class(character)</p>
</dd> <dt>Genus</dt><dd><p>Taxonomic genus (character)</p>
</dd>
<dt>sp.</dt><dd><p>Taxonomic species (character)</p>
</dd> <dt>SEXL</dt><dd><p>Sexual reproduction
(binary)</p>
</dd> <dt>ASEX</dt><dd><p>Asexual reproduction (binary)</p>
</dd> <dt>BVOL</dt><dd><p>Skeletal
body volume of typical adult (ordered numeric with 7 bins). Estimated using
methods of Novack-Gottshall (2008).</p>
 <ul>
<li><p> 1.000: &gt;= 100 cm^3</p>
</li>
<li>
<p>0.833: 100-10 cm^3 </p>
</li>
<li><p> 0.667: 10-1 cm^3 </p>
</li>
<li><p> 0.500: 1-0.1 cm^3 </p>
</li>
<li>
<p>0.333: 0.1-0.01 cm^3 </p>
</li>
<li><p> 0.167: 0.01-0.001 cm^3 </p>
</li>
<li><p> 0: &lt; 0.001 cm^3</p>
</li></ul>
</dd>
<dt>BIOT</dt><dd><p>Biotic substrate composition (binary)</p>
</dd> <dt>LITH</dt><dd><p>Lithic
substrate composition (binary)</p>
</dd> <dt>FLUD</dt><dd><p>Fluidic medium (binary)</p>
</dd>
<dt>HARD</dt><dd><p>Hard substrate consistency (binary)</p>
</dd> <dt>SOFT</dt><dd><p>Soft substrate
consistency (binary)</p>
</dd> <dt>INSB</dt><dd><p>Insubstantial medium consistency
(binary)</p>
</dd> <dt>SPRT</dt><dd><p>Supported on other object (binary)</p>
</dd>
<dt>SSUP</dt><dd><p>Self-supported (binary)</p>
</dd> <dt>ATTD</dt><dd><p>Attached to substrate
(binary)</p>
</dd> <dt>FRLV</dt><dd><p>Free-living (binary)</p>
</dd> <dt>MOBL</dt><dd><p>Mobility (ordered
numeric with 5 bins): </p>
 <ul>
<li><p> 1: habitually mobile </p>
</li>
<li><p> 0.75:
intermittently mobile </p>
</li>
<li><p> 0.50: facultatively mobile </p>
</li>
<li><p> 0.25:
passively mobile (i.e., planktonic drifting) </p>
</li>
<li><p> 0: sedentary
(immobile)</p>
</li></ul>
</dd> <dt>ABST</dt><dd><p>Primary microhabitat stratification: absolute
distance from seafloor (ordered numeric with 5 bins): </p>
 <ul>
<li><p> 1: &gt;=
100 cm </p>
</li>
<li><p> 0.75: 100-10 cm: </p>
</li>
<li><p> 0.50: 10-1 cm</p>
</li>
<li><p> 0.25: 1-0.1 cm </p>
</li>
<li>
<p>0: &lt;0.1 cm</p>
</li></ul>
</dd> <dt>AABS</dt><dd><p>Primary microhabitat is above seafloor (i.e.,
epifaunal)</p>
</dd> <dt>IABS</dt><dd><p>Primary microhabitat is within seafloor (i.e.,
infaunal)</p>
</dd> <dt>RLST</dt><dd><p>Immediately surrounding microhabitat stratification:
relative distance from substrate (ordered numeric with 5 bins): </p>

<ul>
<li><p> 1: &gt;= 100 cm </p>
</li>
<li><p> 0.75: 100-10 cm: </p>
</li>
<li><p> 0.50: 10-1 cm </p>
</li>
<li><p> 0.25:
1-0.1 cm </p>
</li>
<li><p> 0: &lt;0.1 cm</p>
</li></ul>
</dd> <dt>AREL</dt><dd><p>Lives above immediate substrate</p>
</dd>
<dt>IREL</dt><dd><p>Lives within immediate substrate</p>
</dd> <dt>FAAB</dt><dd><p>Food is above
seafloor</p>
</dd> <dt>FIAB</dt><dd><p>Food is within seafloor</p>
</dd> <dt>FAST</dt><dd><p>Primary feeding
microhabitat stratification: absolute distance of food from seafloor
(ordered numeric with 5 bins): </p>
 <ul>
<li><p> 1: &gt;= 100 cm </p>
</li>
<li><p> 0.75:
100-10 cm: </p>
</li>
<li><p> 0.50: 10-1 cm </p>
</li>
<li><p> 0.25: 1-0.1 cm </p>
</li>
<li><p> 0: &lt;0.1 cm</p>
</li></ul>
</dd>
<dt>FARL</dt><dd><p>Food is above immediate substrate</p>
</dd> <dt>FIRL</dt><dd><p>Food is within
immediate substrate</p>
</dd> <dt>FRST</dt><dd><p>Immediately surrounding feeding
microhabitat stratification: relative distance of food from substrate
(ordered numeric with 5 bins): </p>
 <ul>
<li><p> 1: &gt;= 100 cm </p>
</li>
<li><p> 0.75:
100-10 cm: </p>
</li>
<li><p> 0.50: 10-1 cm</p>
</li>
<li><p> 0.25: 1-0.1 cm </p>
</li>
<li><p> 0: &lt;0.1 cm</p>
</li></ul>
</dd>
<dt>AMBT</dt><dd><p>Ambient foraging habit</p>
</dd> <dt>FILT</dt><dd><p>Filter-feeding foraging
habit</p>
</dd> <dt>ATTF</dt><dd><p>Attachment-feeding foraging habit</p>
</dd>
<dt>MASS</dt><dd><p>Mass-feeding foraging habit</p>
</dd> <dt>RAPT</dt><dd><p>Raptorial foraging
habit</p>
</dd> <dt>AUTO</dt><dd><p>Autotrophic diet</p>
</dd> <dt>MICR</dt><dd><p>Microbivorous (bacteria,
protists, algae) diet</p>
</dd> <dt>CARN</dt><dd><p>Carnivorous diet</p>
</dd> <dt>INCP</dt><dd><p>Food has
incorporeal physical condition</p>
</dd> <dt>PART</dt><dd><p>Food consumed as particulate
matter</p>
</dd> <dt>BULK</dt><dd><p>Food consumed as bulk matter</p>
</dd></dl>



<h3>Details</h3>

<p>Binary traits are coded with 0 = absent and 1 = present. Five
ordered numeric traits (body volume, mobility, distance from seafloor
[stratification]) were rescaled to range from 0 to 1 with discrete bins at
equally spaced intermediate values.
</p>
<p>See Novack-Gottshall (2007: especially online Supplementary Appendix A at
<a href="https://www.researchgate.net/publication/237375733_Using_a_theoretical_ecospace_to_quantify_the_ecological_diversity_of_Paleozoic_and_modern_marine_biotas">for
reprint</a>) for definition each functional trait, justifications,
explanations, and examples. Novack-Gottshall (2007: Supplementary Appendix
B; 2016: Supplementary Appendix A) provides examples of how traits were
coded using inferences derived from functional morphology, body size,
ichnology, <em>in situ</em> preservation, biotic associations recording
direct interactions, and interpretation of geographic and depositional
environment patterns.
</p>
<p>Indeterminate taxa (e.g., trepostome bryozoan indet. or
<em>Platystrophia</em> sp.) that occurred within individual samples within
these formations were excluded from the aggregate species pool unless their
occurrence was the sole member of that taxon. Such indeterminate taxa and
genera lacking a species identification were coded for a particular state
only when all other members of that taxon within the Kope-Waynesville
species pool unanimously shared that common state; otherwise, the state was
listed as NA (missing).
</p>


<h3>Source</h3>

<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>References</h3>

<p>Novack-Gottshall, P.M. 2007. Using a theoretical ecospace to
quantify the ecological diversity of Paleozoic and modern marine biotas.
<em>Paleobiology</em> 33: 274-295.
</p>
<p>Novack-Gottshall, P.M. 2008. Using simple body-size metrics to
estimate fossil body volume: empirical validation using diverse Paleozoic
invertebrates. <em>PALAIOS</em> 23(3):163-173.
</p>
<p>Novack-Gottshall, P.M. 2016. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>
<p>Villeger, S., P. M. Novack-Gottshall, and D. Mouillot. 2011. The
multidimensionality of the niche reveals functional diversity changes in
benthic marine biotas across geological time. <em>Ecology Letters</em>
14(6):561-568.
</p>

<hr>
<h2 id='neutral'>Use Neutral Rule to Simulate Ecological Diversification of a Biota.</h2><span id='topic+neutral'></span>

<h3>Description</h3>

<p>Implement Monte Carlo simulation of a biota undergoing ecological
diversification using the neutral rule. Can be used as a simple permutation
test (draw species at random with replacement from provided species pool) if
set <code>Sseed</code> equal to <code>Smax</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neutral(nreps = 1, Sseed, Smax, ecospace)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neutral_+3A_nreps">nreps</code></td>
<td>
<p>Vector of integers (such as a sequence) specifying sample number
produced. Only used when function is applied within <code>lapply</code> or
related function. Default <code>nreps = 1</code> or any other integer produces a
single sample.</p>
</td></tr>
<tr><td><code id="neutral_+3A_sseed">Sseed</code></td>
<td>
<p>Integer giving number of species (or other taxa) to use at start
of simulation.</p>
</td></tr>
<tr><td><code id="neutral_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of species (or other taxa) to include in
simulation.</p>
</td></tr>
<tr><td><code id="neutral_+3A_ecospace">ecospace</code></td>
<td>
<p>An ecospace framework (functional trait space) of class
<code>ecospace</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulations are implemented as Monte Carlo processes in which
species are added iteratively to assemblages, with all added species having
their character states specified by the model rules, here the 'neutral'
rule. Simulations begin with the seeding of <code>Sseed</code> number of species,
chosen at random (with replacement) from either the species pool (if
provided in the <code>weight.file</code> when building the ecospace framework
using <code>create_ecospace</code>) or following the neutral-rule algorithm (if a
pool is not provided). Once seeded, the simulations proceed iteratively
(character-by-character, species-by-species) by following the appropriate
algorithm, as explained below, until terminated at <code>Smax</code>.
</p>
<p><strong>Neutral rule algorithm:</strong> Choose remaining species (or seed species,
if no pool) as random multinomial draws from theoretical ecospace framework
(using whatever constraints and structure was provided by the ecospace
framework in <code>create_ecospace</code>). Thus, if relative weighting was
provided to character states (functional traits), simulated species will
mimic these weights, on average. If state combinations were constrained (by
setting <code>constraint</code> in <code>create_ecospace</code>), then unallowed state
combinations will not be allowed in simulated species.
</p>
<p>Note that this simulation is not a simple permutation test of a species
pool (if provided). The life habit of each new species is built
character-by-character from the realm of theoretically possible states
allowed by the ecospace framework. Simulated species can occupy
combinations of character states that did not occur in the species pool (if
provided). This is an important feature of the simulations, allowing the
entire theoretical ecospace to be explored by the neutral model. However,
the simulation can be used as a simple permutation test (draw species at
random with replacement from provided species pool) if set <code>Sseed</code>
equal to <code>Smax</code> and a species pool is supplied when building the
ecospace framework.
</p>
<p>This rule has also been termed the diffusional, null, and passive model
(Bush and Novack-Gottshall 2012). Additional details on the neutral
simulation are provided in Novack-Gottshall (2016a,b), including
sensitivity to ecospace framework (functional trait space) structure,
recommendations for model selection, and basis in ecological and
evolutionary theory.
</p>


<h3>Value</h3>

<p>Returns a data frame with <code>Smax</code> rows (representing species) and
as many columns as specified by number of characters/states (functional
traits) in the ecospace framework. Columns will have the same data type
(numeric, factor, ordered numeric, or ordered factor) as specified in the
ecospace framework.
</p>


<h3>Note</h3>

<p>The function has been written to allow usage (using
<code><a href="base.html#topic+lapply">lapply</a></code> or some other list-apply function) in 'embarrassingly
parallel' implementations in a high-performance computing environment.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bush, A. and P.M. Novack-Gottshall. 2012. Modelling the
ecological-functional diversification of marine Metazoa on geological time
scales. <em>Biology Letters</em> 8: 151-155.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_ecospace">create_ecospace</a></code>, <code><a href="#topic+redundancy">redundancy</a></code>,
<code><a href="#topic+partitioning">partitioning</a></code>, <code><a href="#topic+expansion">expansion</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an ecospace framework with 15 3-state factor characters
# Can also accept following character types: "numeric", "ord.num", "ord.fac"
nchar &lt;- 15
ecospace &lt;- create_ecospace(nchar = nchar, char.state = rep(3, nchar),
  char.type = rep("factor", nchar))

# Single (default) sample produced by neutral function:
Sseed &lt;- 5
Smax &lt;- 50
x &lt;- neutral(Sseed = Sseed, Smax = Smax, ecospace = ecospace)
head(x, 10)

# Plot results, showing order of assembly
# (Seed species in red, next 5 in black, remainder in gray)
# Notice the neutral model fills the entire ecospace with life habits
seq &lt;- seq(nchar)
types &lt;- sapply(seq, function(seq) ecospace[[seq]]$type)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Neutral model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Create 5 samples using multiple nreps and lapply
nreps &lt;- 1:5
samples &lt;- lapply(X = nreps, FUN = neutral, Sseed = 5, Smax = 50, ecospace)
str(samples)

# Implement as simple permutation test by setting Sseed = Smax and providing species pool)
nchar &lt;- 18
char.state &lt;- c(2, 7, 3, 3, 2, 2, 5, 5, 2, 5, 2, 2, 5, 2, 5, 5, 3, 3)
char.type &lt;- c("numeric", "ord.num", "numeric", "numeric", "numeric", "numeric",
  "ord.num", "ord.num", "numeric", "ord.num", "numeric", "numeric", "ord.num",
  "numeric", "ord.num", "numeric", "numeric", "numeric")
data(KWTraits)
ecospace &lt;- create_ecospace(nchar, char.state, char.type, constraint = 2,
  weight.file = KWTraits)

x &lt;- neutral(Sseed = 100, Smax = 100, ecospace = ecospace)
mean(dist(x))

# Note ecological disparity (functional diversity) is less when perform permutation
x &lt;- neutral(Sseed = 5, Smax = 100, ecospace = ecospace)
mean(dist(x))

# Simulated character states (functional traits) proportionally mimic those in species pool
x &lt;- neutral(Sseed = 5, Smax = 234, ecospace = ecospace)
table(x[,1:2])
table(KWTraits$SEXL, KWTraits$ASEX)

</code></pre>

<hr>
<h2 id='partitioning'>Use Partitioning Rule to Simulate Ecological Diversification of a Biota.</h2><span id='topic+partitioning'></span>

<h3>Description</h3>

<p>Implement Monte Carlo simulation of a biota undergoing ecological
diversification using the partitioning rule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partitioning(
  nreps = 1,
  Sseed,
  Smax,
  ecospace,
  method = "Euclidean",
  rule = "strict",
  strength = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="partitioning_+3A_nreps">nreps</code></td>
<td>
<p>Vector of integers (such as a sequence) specifying sample number
produced. Only used when function is applied within <code>lapply</code> or
related function. Default <code>nreps=1</code> or any other integer produces a
single sample.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_sseed">Sseed</code></td>
<td>
<p>Integer giving number of species (or other taxa) to use at start
of simulation.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of species (or other taxa) to include in
simulation.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_ecospace">ecospace</code></td>
<td>
<p>An ecospace framework (functional trait space) of class
<code>ecospace</code>.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_method">method</code></td>
<td>
<p>Distance measure to use when calculating functional distances
between species. Default is <code>Euclidean</code> using
<code>stats::<a href="stats.html#topic+dist">dist</a></code>. <code>Gower</code> or any other value uses
Gower distance (using <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code>). Presence of factor or
ordered factor character types forces use of Gower distance.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_rule">rule</code></td>
<td>
<p>The partitioning implementation to use in the simulation. Default
<code>strict</code> implements the 'minimum distant neighbor' rule;
<code>relaxed</code> implements the &quot;maximum nearest neighbor&quot; rule. See
'Details' below for further explanation.</p>
</td></tr>
<tr><td><code id="partitioning_+3A_strength">strength</code></td>
<td>
<p>Strength parameter controlling probability that partitioning
rule is followed during simulation. Values must range between
<code>strength = 1</code> (default, rules always implemented) and <code>strength = 0</code>
(rules never implemented).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulations are implemented as Monte Carlo processes in which
species are added iteratively to assemblages, with all added species having
their character states specified by the model rules, here the
'partitioning' rule. Simulations begin with the seeding of <code>Sseed</code>
number of species, chosen at random (with replacement) from either the
species pool (if provided in the <code>weight.file</code> when building the
ecospace framework using <code>create_ecospace</code>) or following the
neutral-rule algorithm (if a pool is not provided). Once seeded, the
simulations proceed iteratively (character-by-character,
species-by-species) by following the appropriate algorithm, as explained
below, until terminated at <code>Smax</code>.
</p>
<p><strong>Partitioning rule algorithm:</strong> Measure distances between all pairs
of species, using <code>Euclidean</code> or <code>Gower</code> distance method
specified by <code>method</code> argument. Use either of the following rules to
identify the position of each additional species. </p>

<dl>
<dt><code>strict</code> (minimum distant neighbor) rule</dt><dd><p>Identify the maximum
distances between all pairs of species (the most-distant neighbors); the
space to be partitioned is the minimum of these distances. This
implementation progressively fills in the largest parts of the ecospace
that are least occupied between neighboring species, and eventually
partitions the ecospace in straight-line gradients between seed species.</p>
</dd>
<dt><code>relaxed</code> (maximum nearest neighbor) rule</dt><dd><p>Identify
nearest-neighbor distances between all pairs of species; the space to be
partitioned is the maximum of these distances. This implementation places
new species in the most unoccupied portion of the ecospace that is within
the cluster of pre-existing species, often the centroid.</p>
</dd></dl>
<p> In both rules,
each new species is created as a resampled combination of the character
states of the identified neighbors. If multiple pairs meet the specific
criteria, one of these pairs is chosen at random. Ordered, multistate
character partitioning (such as ordered factors or order numeric character
types) can include any state equal to or between the observed states of
existing species. The probability of following the partitioning rule is
determined by the <code>strength</code> parameter. Default <code>strength = 1</code>
always implements the rule, whereas <code>strength = 0</code> never implements it
(essentially making the simulation follow the <code><a href="#topic+neutral">neutral</a></code> rule.)
</p>
<p>Each newly assigned character is compared with the ecospace framework
(<code>ecospace</code>) to confirm that it is an allowed state combination before
proceeding to the next character. If the newly built character is
disallowed from the ecospace framework (i.e., because it has &quot;dual
absences&quot; [0,0], has been excluded based on the species pool
[<code>weight.file</code> in <code>create_ecospace</code>], or is not allowed by the
ecospace <code>constraint</code> parameter), then the character-selection
algorithm is repeated until an allowable character is selected. When
simulations proceed to very large sample sizes (&gt;100), this confirmatory
process can require long computational times, and produce &quot;new&quot;
intermediate species that are functionally identical to pre-existing
species. This can occur, for example, when no life habits, or perhaps only
one, exist that forms an allowable intermediate between the selected
neighbors.
</p>
<p>Partitioning rules tend to produce ecospaces displaying linear gradients
between seed species (in the <code>strict</code> implementation) or concentration
of life habits near the functional centroid (in the <code>relaxed</code>
implementation). Additional details on the partitioning simulation are
provided in Novack-Gottshall (2016a,b), including sensitivity to
ecospace framework (functional trait space) structure, recommendations for
model selection, and basis in ecological and evolutionary theory.
</p>


<h3>Value</h3>

<p>Returns a data frame with <code>Smax</code> rows (representing species) and
as many columns as specified by number of characters/states (functional
traits) in the ecospace framework. Columns will have the same data type
(numeric, factor, ordered numeric, or ordered factor) as specified in the
ecospace framework.
</p>


<h3>Note</h3>

<p>A bug exists within <code>FD::<a href="FD.html#topic+gowdis">gowdis</a></code> where nearest-neighbor
distances can not be calculated when certain characters (especially numeric
characters with values other than 0 and 1) share identical traits across
species. The nature of the bug is under investigation, but the current
implementation is reliable under most uses. If you run into problems
because of this bug, a work-around is to manually change the function to
call <code>cluster::<a href="cluster.html#topic+daisy">daisy</a></code> using <code>metric = "gower"</code>
instead.
</p>
<p>The function has been written to allow usage (using <code><a href="base.html#topic+lapply">lapply</a></code> or
some other list-apply function) in 'embarrassingly parallel' implementations
in a high-performance computing environment.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bush, A. and P.M. Novack-Gottshall. 2012. Modelling the
ecological-functional diversification of marine Metazoa on geological time
scales. <em>Biology Letters</em> 8: 151-155.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_ecospace">create_ecospace</a></code>, <code><a href="#topic+neutral">neutral</a></code>,
<code><a href="#topic+redundancy">redundancy</a></code>, <code><a href="#topic+expansion">expansion</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an ecospace framework with 15 3-state factor characters
# Can also accept following character types: "numeric", "ord.num", "ord.fac"
nchar &lt;- 15
ecospace &lt;- create_ecospace(nchar = nchar, char.state = rep(3, nchar),
  char.type = rep("factor", nchar))

# Single (default) sample produced by partitioning function (with strength = 1 and
#   "strict" partitioning rules):
Sseed &lt;- 5
Smax &lt;- 40
x &lt;- partitioning(Sseed = Sseed, Smax = Smax, ecospace = ecospace, rule = "strict")
head(x, 10)

# Plot results, showing order of assembly
# (Seed species in red, next 5 in black, remainder in gray)
# Notice the 'strict' partitioning model produces an ecospace with life-habit gradients
#   between seed species
seq &lt;- seq(nchar)
types &lt;- sapply(seq, function(seq) ecospace[[seq]]$type)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Partitioning model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Same, but following "relaxed" partitioning rules:
# Notice the 'relaxed' partitioning model only fills in the ecospace between seed species
x &lt;- partitioning(Sseed = Sseed, Smax = Smax, ecospace = ecospace, rule = "relaxed")
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Partitioning model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Change strength parameter so rules followed 95% of time:
x &lt;- partitioning(Sseed = Sseed, Smax = Smax, ecospace = ecospace, strength = 0.95, rule = "strict")
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Partitioning model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Create 5 samples using multiple nreps and lapply (can be slow)
nreps &lt;- 1:5
samples &lt;- lapply(X = nreps, FUN = partitioning, Sseed = 5, Smax = 40, ecospace)
str(samples)

</code></pre>

<hr>
<h2 id='prep_data'>Internal Ecospace Functions.</h2><span id='topic+prep_data'></span>

<h3>Description</h3>

<p>Internal functions not intended to be called directly by users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_data(ecospace, Smax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep_data_+3A_ecospace">ecospace</code></td>
<td>
<p>An ecospace framework (functional trait space) of class
<code>ecospace</code>.</p>
</td></tr>
<tr><td><code id="prep_data_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of species (or other taxa) to include in
simulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pre-allocate the basic data frame structure so classes (and factor
levels, if used) are properly set. Bit clunky but efficient solution:
adding first placeholder column allows <code>cbind</code> to work properly with
data frame later, but requires deleting that placeholder at end.
</p>


<h3>Value</h3>

<p>Returns an empty data frame with pre-defined number of species and
functional traits of proper class type.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_ecospace">create_ecospace</a></code> for how to create an ecospace
framework (functional trait space).
</p>

<hr>
<h2 id='rbind_listdf'>Combine Large Lists of Data Frames.</h2><span id='topic+rbind_listdf'></span>

<h3>Description</h3>

<p>Quickly combine large lists of dataframes into a single data frame by
combining them first into smaller data frames. This is only time- and
memory-efficient when dealing with large (&gt;100) lists of data frames.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbind_listdf(lists = NULL, seq.by = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind_listdf_+3A_lists">lists</code></td>
<td>
<p>List in which each component is a data frame. Each data frame
must have the same number and types of columns, although the data frames
can have different numbers of rows.</p>
</td></tr>
<tr><td><code id="rbind_listdf_+3A_seq.by">seq.by</code></td>
<td>
<p>Length of small data frames to work with at a time. Default is
100, which timing tests confirm is generally most efficient.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Rather than combine all list data frames into a single data frame,
this function builds smaller subsets of data frames, and then combines them
together at the end. This process is more time- and memory-efficient.
Timing tests confirm that seq.by = 100 is the optimal break-point size. See
examples for confirmation. Function can break large lists into up to
456,976 data frame subparts, giving a warning if requires more subparts.
Only time- and memory-efficient when dealing with large (&gt;100) lists of
data frames.
</p>


<h3>Value</h3>

<p>Single data frame with number of rows equal to the sum of all data
frames in the list, and columns the same as those of individual list data
frames.
</p>


<h3>Note</h3>

<p>When called within <code>lapply</code>, the simulation functions
<code>neutral</code>, <code>redundancy</code>, <code>partitioning</code>, <code>expansion</code>,
and <code>calc_metrics</code>, which calculates common ecological disparity
(functional diversity) statistics on simulated biotas, produce lists of
data frames. This function is useful for combining these separate lists
into a single data frame for subsequent analyses. This is especially useful
when using the functions within a high-performance computing environment
when submitted as 'embarrassingly parallel' implementations.
<code><a href="data.table.html#topic+rbindlist">rbindlist</a></code> is a more efficient version of this
function.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neutral">neutral</a></code>, <code><a href="#topic+redundancy">redundancy</a></code>,
<code><a href="#topic+partitioning">partitioning</a></code>, <code><a href="#topic+expansion">expansion</a></code>,
<code><a href="#topic+calc_metrics">calc_metrics</a></code>, and <code><a href="data.table.html#topic+rbindlist">rbindlist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nl &lt;- 500     # List length
lists &lt;- vector("list", length = nl)
for(i in 1:nl) lists[[i]] &lt;- list(x = rnorm(100), y = rnorm(100))
str(lists)
object.size(lists)
all &lt;- rbind_listdf(lists)
str(all)
object.size(all)       # Also smaller object size

# Build blank ecospace framework to use in simulations
ecospace &lt;- create_ecospace(nchar = 15, char.state = rep(3, 15),
                            char.type = rep("numeric", 15))

# Build 5 samples for neutral model:
nreps &lt;- 1:5
Smax &lt;- 10
n.samples &lt;- lapply(X = nreps, FUN = neutral, Sseed = 3, Smax = Smax, ecospace)

# Calculate functional diversity metrics for simulated samples
n.metrics &lt;- lapply(X = nreps, FUN = calc_metrics, samples = n.samples,
                    Model = "neutral", Param = "NA")
alarm()
str(n.metrics)

# rbind lists together into a single dataframe
all &lt;- rbind_listdf(n.metrics)

# Calculate mean dynamics
means &lt;- n.metrics[[1]]
for(n in 1:Smax) means[n,4:11] &lt;- apply(all[which(all$S == means$S[n]),4:11],
                                        2, mean, na.rm = TRUE)
means

# Plot statistics as function of species richness, overlaying mean dynamics
op &lt;- par()
par(mfrow = c(2,4), mar = c(4, 4, 1, .3))
attach(all)

plot(S, H, type = "p", cex = .75, col = "gray")
lines(means$S, means$H, type = "l", lwd = 2)
plot(S, D, type = "p", cex = .75, col = "gray")
lines(means$S, means$D, type = "l", lwd = 2)
plot(S, M, type = "p", cex = .75, col = "gray")
lines(means$S, means$M, type = "l", lwd = 2)
plot(S, V, type = "p", cex = .75, col = "gray")
lines(means$S, means$V, type = "l", lwd = 2)
plot(S, FRic, type = "p", cex = .75, col = "gray")
lines(means$S, means$FRic, type = "l", lwd = 2)
plot(S, FEve, type = "p", cex = .75, col = "gray")
lines(means$S, means$FEve, type = "l", lwd = 2)
plot(S, FDiv, type = "p", cex = .75, col = "gray")
lines(means$S, means$FDiv, type = "l", lwd = 2)
plot(S, FDis, type = "p", cex = .75, col = "gray")
lines(means$S, means$FDis, type = "l", lwd = 2)

par(op)

## Not run: 
# Note each of following can take a few seconds to run
# Compare timings:
t0 &lt;- Sys.time()
all &lt;- rbind_listdf(lists)
(Sys.time() - t0)

t0 &lt;- Sys.time()
all &lt;- rbind_listdf(lists, seq.by = 20)
(Sys.time() - t0)

t0 &lt;- Sys.time()
all &lt;- rbind_listdf(lists, seq.by = 500)
(Sys.time() - t0)

# Compare to non-function version
all2 &lt;- data.frame()
t0 &lt;- Sys.time()
for(i in 1:nl) all2 &lt;- rbind(all2, lists[[i]])
(Sys.time() - t0)

## End(Not run)

# Compare to data.table's 'rbindlist' version
library(data.table)
t0 &lt;- Sys.time()
all &lt;- rbindlist(lists)
(Sys.time() - t0)

</code></pre>

<hr>
<h2 id='redundancy'>Use Redundancy Rule to Simulate Ecological Diversification of a Biota.</h2><span id='topic+redundancy'></span>

<h3>Description</h3>

<p>Implement Monte Carlo simulation of a biota undergoing ecological
diversification using the redundancy rule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>redundancy(nreps = 1, Sseed, Smax, ecospace, strength = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="redundancy_+3A_nreps">nreps</code></td>
<td>
<p>Vector of integers (such as a sequence) specifying sample number
produced. Only used when function is applied within <code>lapply</code> or
related function. Default <code>nreps = 1</code> or any other integer produces a
single sample.</p>
</td></tr>
<tr><td><code id="redundancy_+3A_sseed">Sseed</code></td>
<td>
<p>Integer giving number of species (or other taxa) to use at start
of simulation.</p>
</td></tr>
<tr><td><code id="redundancy_+3A_smax">Smax</code></td>
<td>
<p>Maximum number of species (or other taxa) to include in
simulation.</p>
</td></tr>
<tr><td><code id="redundancy_+3A_ecospace">ecospace</code></td>
<td>
<p>An ecospace framework (functional trait space) of class
<code>ecospace</code>.</p>
</td></tr>
<tr><td><code id="redundancy_+3A_strength">strength</code></td>
<td>
<p>Strength parameter controlling probability that redundancy
rule is followed during simulation. Values must range between
<code>strength = 1</code> (default, rules always implemented) and <code>strength = 0</code>
(rules never implemented).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulations are implemented as Monte Carlo processes in which
species are added iteratively to assemblages, with all added species having
their character states specified by the model rules, here the 'redundancy'
rule. Simulations begin with the seeding of <code>Sseed</code> number of species,
chosen at random (with replacement) from either the species pool (if
provided in the <code>weight.file</code> when building the ecospace framework
using <code>create_ecospace</code>) or following the neutral-rule algorithm (if a
pool is not provided). Once seeded, the simulations proceed iteratively
(character-by-character, species-by-species) by following the appropriate
algorithm, as explained below, until terminated at <code>Smax</code>.
</p>
<p><strong>Redundancy rule algorithm:</strong> Pick one existing species at random and
create a new species using that species' characters as a template. A
character is modified (using a random multinomial draw from the ecospace
framework) according to the <code>strength</code> parameter. Default
<code>strength = 1</code> always implements the redundancy rule, whereas
<code>strength = 0</code> never implements it (essentially making the simulation
follow the <code><a href="#topic+neutral">neutral</a></code> rule.) Because new character states can be
any allowed by the ecospace framework, there is the possibility of
obtaining redundancy greater than that specified by a strength parameter
less than 1 (if, for example, the new randomly chosen character states are
identical to those of the template species).
</p>
<p>Redundancy rules tend to produce ecospaces with discrete clusters of
functionally similar species. Additional details on the redundancy
simulation are provided in Novack-Gottshall (2016a,b), including
sensitivity to ecospace framework (functional trait space) structure,
recommendations for model selection, and basis in ecological and
evolutionary theory.
</p>


<h3>Value</h3>

<p>Returns a data frame with <code>Smax</code> rows (representing species) and
as many columns as specified by number of characters/states (functional
traits) in the ecospace framework. Columns will have the same data type
(numeric, factor, ordered numeric, or ordered factor) as specified in the
ecospace framework.
</p>


<h3>Note</h3>

<p>The function has been written to allow usage (using
<code><a href="base.html#topic+lapply">lapply</a></code> or some other list-apply function) in 'embarrassingly
parallel' implementations in a high-performance computing environment.
</p>


<h3>Author(s)</h3>

<p>Phil Novack-Gottshall <a href="mailto:pnovack-gottshall@ben.edu">pnovack-gottshall@ben.edu</a>
</p>


<h3>References</h3>

<p>Bush, A. and P.M. Novack-Gottshall. 2012. Modelling the
ecological-functional diversification of marine Metazoa on geological time
scales. <em>Biology Letters</em> 8: 151-155.
</p>
<p>Novack-Gottshall, P.M. 2016a. General models of ecological
diversification. I. Conceptual synthesis. <em>Paleobiology</em> 42: 185-208.
</p>
<p>Novack-Gottshall, P.M. 2016b. General models of ecological
diversification. II. Simulations and empirical applications.
<em>Paleobiology</em> 42: 209-239.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_ecospace">create_ecospace</a></code>, <code><a href="#topic+neutral">neutral</a></code>,
<code><a href="#topic+partitioning">partitioning</a></code>, <code><a href="#topic+expansion">expansion</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create an ecospace framework with 15 3-state factor characters
# Can also accept following character types: "numeric", "ord.num", "ord.fac"
nchar &lt;- 15
ecospace &lt;- create_ecospace(nchar = nchar, char.state = rep(3, nchar),
  char.type = rep("factor", nchar))

# Single (default) sample produced by redundancy function (with strength = 1):
Sseed &lt;- 5
Smax &lt;- 50
x &lt;- redundancy(Sseed = Sseed, Smax = Smax, ecospace = ecospace)
head(x, 10)

# Plot results, showing order of assembly
# (Seed species in red, next 5 in black, remainder in gray)
# Notice the redundancy model produces an ecospace with discrete clusters of life habits
seq &lt;- seq(nchar)
types &lt;- sapply(seq, function(seq) ecospace[[seq]]$type)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Redundancy model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Change strength parameter so new species are 95% identical:
x &lt;- redundancy(Sseed = Sseed, Smax = Smax, ecospace = ecospace, strength = 0.95)
if(any(types == "ord.fac" | types == "factor")) pc &lt;- prcomp(FD::gowdis(x)) else
  pc &lt;- prcomp(x)
plot(pc$x, type = "n", main = paste("Redundancy model,\n", Smax, "species"))
text(pc$x[,1], pc$x[,2], labels = seq(Smax), col = c(rep("red", Sseed), rep("black", 5),
  rep("slategray", (Smax - Sseed - 5))), pch = c(rep(19, Sseed), rep(21, (Smax - Sseed))),
  cex = .8)

# Create 5 samples using multiple nreps and lapply (can be slow)
nreps &lt;- 1:5
samples &lt;- lapply(X = nreps, FUN = redundancy, Sseed = 5, Smax = 50, ecospace)
str(samples)

</code></pre>

<hr>
<h2 id='sample2'>Internal Ecospace Functions.</h2><span id='topic+sample2'></span>

<h3>Description</h3>

<p>Internal functions not intended to be called directly by users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample2(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample2_+3A_x">x</code></td>
<td>
<p>Either a vector of one or more elements from which to choose, or a positive integer.</p>
</td></tr>
<tr><td><code id="sample2_+3A_...">...</code></td>
<td>
<p>arguments for particular methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Corrects behavior when object sampled has unit length. This is the same
function proposed in help file of <code>sample</code> as <code>resample</code>.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+sample">sample</a></code>
</p>

<hr>
<h2 id='unique2'>Internal Ecospace Functions.</h2><span id='topic+unique2'></span>

<h3>Description</h3>

<p>Internal functions not intended to be called directly by users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unique2(x, length = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unique2_+3A_x">x</code></td>
<td>
<p>a vector or a data frame or an array or NULL.</p>
</td></tr>
<tr><td><code id="unique2_+3A_length">length</code></td>
<td>
<p>number of times (default = 1) to repeat the dimensionally shorter
state.</p>
</td></tr>
<tr><td><code id="unique2_+3A_...">...</code></td>
<td>
<p>arguments for particular methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Modified <code>unique</code> function that returns proper number of items for
filling of matrix when called within <code>apply</code>.
</p>


<h3>Note</h3>

<p>When dealing with ecospace frameworks with multistate binary (numeric)
character types and characters weighted by supplied species pools,
sometimes all species will share the same state value for one of several
states. (For example, perhaps all species are capable of sexual
reproduction, but there is variation in whether some are exclusively sexual
and some are hermaphroditic.) When this occurs, calling <code>apply</code> when
choosing possible ecospace states will 'break the dimensionality' of the
character matrix and convert it into a list. This function maintains matrix
dimensionality, by repeating the dimensionally shorter unique state
sufficient times to maintain equal length as found in other states.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+unique">unique</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
