<!DOCTYPE html><html lang="en"><head><title>Help for package effectR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {effectR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#effectR'><p>effectR: An R Package to Call Oomycete Effectors</p></a></li>
<li><a href='#candidate.rxlr'><p>An example of 'effector.summary' output</p></a></li>
<li><a href='#effector.summary'><p>Returns non-redundant sequences from hmm.search or regex.search and generates a motif table</p></a></li>
<li><a href='#fasta_to_stockholm'><p>Converts FASTA to STOCKHOLM</p></a></li>
<li><a href='#get_hmmer_path'><p>Check if HMMER is installed and return path</p></a></li>
<li><a href='#get_mafft_path'><p>Check if MAFFT is installed and return path</p></a></li>
<li><a href='#hmm.logo'><p>Plots the relative frequencies of each position for <code>hmmsearch</code> table.</p></a></li>
<li><a href='#hmm.search'><p>Searching for motifs using HMM searches</p></a></li>
<li><a href='#regex.search'><p>Searching for motifs using regular expressions (REGEX)</p></a></li>
<li><a href='#shiny.effectR'><p>Function to run the effectR package using a shiny app</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Predicts Oomycete Effectors</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Predicts cytoplasmic effector proteins using genomic data by searching for motifs of interest using regular expression searches and hidden Markov models (HMM) based in Haas et al. (2009) &lt;<a href="https://doi.org/10.1038%2Fnature08358">doi:10.1038/nature08358</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>seqinr (&ge; 3.3.6), ggplot2 (&ge; 2.2.1), shiny, reshape2, utils,
rmarkdown, viridis</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-30 20:56:05 UTC; tabimaj</td>
</tr>
<tr>
<td>Author:</td>
<td>Javier Tabima [aut, cre],
  Niklaus J. Grunwald [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Javier Tabima &lt;tabimaj@oregonstate.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-09-30 21:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='effectR'>effectR: An R Package to Call Oomycete Effectors</h2><span id='topic+effectR'></span><span id='topic+effectR-package'></span>

<h3>Description</h3>

<p>An R Package to Call Oomycete Effectors.
</p>


<h3>Author(s)</h3>

<p>Javier F. Tabima <a href="mailto:tabimaj@oregonstate.edu">tabimaj@oregonstate.edu</a>
</p>

<hr>
<h2 id='candidate.rxlr'>An example of 'effector.summary' output</h2><span id='topic+candidate.rxlr'></span>

<h3>Description</h3>

<p>'candidate.rxlr' is an example result of the <code><a href="#topic+effector.summary">effector.summary</a></code> function
for new users that have not installed the additional software used by the
effectR package.
</p>


<h3>Author(s)</h3>

<p>Javier F. Tabima <a href="mailto:tabimaj@science.oregonstate.edu">tabimaj@science.oregonstate.edu</a>
</p>

<hr>
<h2 id='effector.summary'>Returns non-redundant sequences from hmm.search or regex.search and generates a motif table</h2><span id='topic+effector.summary'></span>

<h3>Description</h3>

<p>This function summarize the results from <code><a href="#topic+regex.search">regex.search</a></code> or <code><a href="#topic+hmm.search">hmm.search</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effector.summary(hmm.result, motif = "RxLR", reg.pat = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effector.summary_+3A_hmm.result">hmm.result</code></td>
<td>
<p>A list of <code>SeqFastadna</code> objects obtained from <code><a href="#topic+regex.search">regex.search</a></code> or <code><a href="#topic+hmm.search">hmm.search</a></code></p>
</td></tr>
<tr><td><code id="effector.summary_+3A_motif">motif</code></td>
<td>
<p>A character string indicating the motif of interest. Motifs for two cytoplasmic effectors are added to the function: <code>RxLR</code> or <code>CRN</code> effectors. Each of these motifs are associated with a by-default REGEX (<code>reg.pat</code>). A third option, <code>custom</code>, allows for the search of custom motifs. The <code>custom</code> option requires the specification of the motif REGEX pattern in the <code>reg.pat</code> option, in a <code><a href="base.html#topic+regex">regex</a></code> format. Default is RxLR</p>
</td></tr>
<tr><td><code id="effector.summary_+3A_reg.pat">reg.pat</code></td>
<td>
<p>A character string indicating the REGEX pattern for the <code>custom</code> motif. The specification of the REGEX pattern in must be in <code><a href="base.html#topic+regex">regex</a></code> format. Required for <code>custom</code> option of <code>motif</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two objects: Summary motif table and non-redundant sequences (only with results of <code><a href="#topic+hmm.search">hmm.search</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

fasta.file &lt;- system.file("extdata", "test_infestans.fasta", package = "effectR")
ORF &lt;- seqinr::read.fasta(fasta.file)
REGEX &lt;- regex.search(ORF, motif='RxLR')
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq=REGEX, num.threads = 2)
effector.summary(candidate.rxlr)
# Custom motifs
reg.pat &lt;- "^\\w{50,60}[w,v]"
REGEX &lt;- regex.search(sequence = ORF, motif = "custom", reg.pat = reg.pat)
candidate.custom &lt;- hmm.search(original.seq = fasta.file, regex.seq = REGEX)
effector.summary(candidate.custom, motif = "custom", reg.pat = reg.pat)


## End(Not run)
</code></pre>

<hr>
<h2 id='fasta_to_stockholm'>Converts FASTA to STOCKHOLM</h2><span id='topic+fasta_to_stockholm'></span>

<h3>Description</h3>

<p>The HMMER binary version for windows does not recognize FASTA files to build the
hmm profile. This function creates a STOCKHOLM file readable by HMMER 3.0
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasta_to_stockholm(fasta.file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fasta_to_stockholm_+3A_fasta.file">fasta.file</code></td>
<td>
<p>FASTA object to be converted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE
</p>

<hr>
<h2 id='get_hmmer_path'>Check if HMMER is installed and return path</h2><span id='topic+get_hmmer_path'></span>

<h3>Description</h3>

<p>Check if HMMER is installed and return path to an executable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hmmer_path(command, hmmer.path = NULL, error = TRUE,
  verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_hmmer_path_+3A_command">command</code></td>
<td>
<p>Which command to get the path for. For example &quot;hmmsearch&quot;.</p>
</td></tr>
<tr><td><code id="get_hmmer_path_+3A_error">error</code></td>
<td>
<p>If <code>TRUE</code>, throw an error if HMMER is not installed.</p>
</td></tr>
<tr><td><code id="get_hmmer_path_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, print progress reports.</p>
</td></tr>
<tr><td><code id="get_hmmer_path_+3A_path">path</code></td>
<td>
<p>Where to look for HMMER. By default it will look on the search
path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE
</p>

<hr>
<h2 id='get_mafft_path'>Check if MAFFT is installed and return path</h2><span id='topic+get_mafft_path'></span>

<h3>Description</h3>

<p>Check if MAFFT is installed and return path to executable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mafft_path(mafft.path = NULL, error = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mafft_path_+3A_error">error</code></td>
<td>
<p>If <code>TRUE</code>, throw an error if MAFFT is not installed.</p>
</td></tr>
<tr><td><code id="get_mafft_path_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code>, print progress reports.</p>
</td></tr>
<tr><td><code id="get_mafft_path_+3A_path">path</code></td>
<td>
<p>Where to look for MAFFT. By default it will look on the search
path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE
</p>

<hr>
<h2 id='hmm.logo'>Plots the relative frequencies of each position for <code>hmmsearch</code> table.</h2><span id='topic+hmm.logo'></span>

<h3>Description</h3>

<p>This function plots the results from <code><a href="#topic+hmm.search">hmm.search</a></code> as a barplot with amino acids in the x axis and the relative frequency of each amino acid in the y axis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmm.logo(hmm.table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hmm.logo_+3A_hmm.table">hmm.table</code></td>
<td>
<p>The HMM profile table resulting from <code><a href="#topic+hmm.search">hmm.search</a></code></p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

fasta.file &lt;- system.file("extdata", "test_infestans.fasta", package = "effectR")
ORF &lt;- seqinr::read.fasta(fasta.file)
REGEX &lt;- regex.search(ORF, motif='RxLR')
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq=REGEX, num.threads = 2)
hmm.logo(candidate.rxlr$HMM_Table)

## End(Not run)
</code></pre>

<hr>
<h2 id='hmm.search'>Searching for motifs using HMM searches</h2><span id='topic+hmm.search'></span>

<h3>Description</h3>

<p>This function uses MAFFT and HMMER to search for sequences with RxLR or CRN motifs using hidden markov models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmm.search(original.seq, regex.seq, alignment.file = NULL,
  save.alignment = FALSE, mafft.path = NULL, num.threads = 2,
  hmm.path = NULL, seed = 12345)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hmm.search_+3A_original.seq">original.seq</code></td>
<td>
<p>The absolute path for the original six-frame translation FASTA file</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_regex.seq">regex.seq</code></td>
<td>
<p>A list of <code>SeqFastadna</code> objects resulting from <code><a href="#topic+regex.search">regex.search</a></code>. The HMM profile will be constructed using these sequences</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_alignment.file">alignment.file</code></td>
<td>
<p>(Optional) The absolute path for an alignment file of the sequences to build the hmmer profile from. It's recommended that the alignment file cointains the same sequences than the &lsquo;regex.seq' files. If the user provides the absolute path, *effectR* won&rsquo;t use MAFFT to align the sequences and will use the alignment for the HMMER searches. If no alignment file is provided, *effectR* will use MAFFT to align the sequences from 'regex.seq' and run HMMER.</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_save.alignment">save.alignment</code></td>
<td>
<p>(Optional) Save the alignment in the returning object. The MAFFT alignment will be saved as the first element of the returned object.</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_mafft.path">mafft.path</code></td>
<td>
<p>(Optional) Local path of folder containing the MAFFT binary executable file or the executable file itself. If not specified, then MAFFT must be in the program search path.</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_num.threads">num.threads</code></td>
<td>
<p>(Optional) Number of threads to be used by MAFFT</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_hmm.path">hmm.path</code></td>
<td>
<p>(Optional) Local path of  folder containing the HMMER binaries.  If not specified, then HMMER executables must be in the program search path.</p>
</td></tr>
<tr><td><code id="hmm.search_+3A_seed">seed</code></td>
<td>
<p>(Optional) The seed to used with HMMER commands. Set this to get the same output each time</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>hmm.search</code> uses the results from <code><a href="#topic+regex.search">regex.search</a></code> to search for motifs of interest using Hidden Markov Models after aligning the sequences with MAFFT.
After the multiple sequence alignment is complete, the function constructs a HMM profile using the alignment data. The HMM profile is in the original list of <code>SeqFastadna</code> objects to obtain the best HMM results with sequences with RxLR or CRN motifs.
</p>


<h3>Value</h3>

<p>A list of three elements: REGEX candidate effectors, HMM candidate effectors, and HMM results table.
</p>


<h3>Note</h3>

<p>If MAFFT/HMMER are not the program search path, the user has to specify the path for the MAFFT and the HMMER executable binaries and specify them in the <code>mafft.path</code> and <code>hmm.path</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

fasta.file &lt;- system.file("extdata", "test_infestans.fasta", package = "effectR")
ORF &lt;- seqinr::read.fasta(fasta.file)
REGEX &lt;- regex.search(ORF, motif="RxLR")
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq = REGEX,
 alignment.file=NULL, save.alignment=T)

# To save the alignment:
library(seqinr)
write.fasta(sequences = getSequence(candidate.rxlr$Alignment),
 names = getName(candidate.rxlr$Alignment),
 file.out = "Alin_infestans.fasta", nbchar = 10000)

# To use an alignment file
ALIN &lt;- system.file("extdata", "Alin_infestans.fasta", package = "effectR")
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq = REGEX,
 alignment.file = ALIN)

## End(Not run)
</code></pre>

<hr>
<h2 id='regex.search'>Searching for motifs using regular expressions (REGEX)</h2><span id='topic+regex.search'></span>

<h3>Description</h3>

<p>This function uses searches a list of <code>SeqFastadna</code> objects for sequences with RxLR or CRN motifs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>regex.search(sequence, motif = "RxLR", reg.pat = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="regex.search_+3A_sequence">sequence</code></td>
<td>
<p>A list of <code>SeqFastadna</code> objects from <code>seqinr</code> <code>read.fasta</code>. The <code>SeqFastadna</code> object must be comprised by amino acid sequences, not DNA sequences</p>
</td></tr>
<tr><td><code id="regex.search_+3A_motif">motif</code></td>
<td>
<p>A character string indicating the motif to be searched. Motifs for two cytoplasmic effectors are added to the function: <code>RxLR</code> or <code>CRN</code> effectors. Each of these motifs are associated with a by-default REGEX (<code>reg.pat</code>).These motifs are adapted from Haas et al. (2009).
</p>
<p>A third option, <code>custom</code>, allows for the search of custom motifs. The <code>custom</code> option requires the specification of the motif REGEX pattern in the <code>reg.pat</code> option, in a <code><a href="base.html#topic+regex">regex</a></code> format.
</p>
<p>Default <code>motif</code> is RxLR</p>
</td></tr>
<tr><td><code id="regex.search_+3A_reg.pat">reg.pat</code></td>
<td>
<p>A character string indicating the REGEX pattern for the <code>custom</code> motif. The specification of the REGEX pattern in must be in <code><a href="base.html#topic+regex">regex</a></code> format. Required for <code>custom</code> option of <code>motif</code></p>
</td></tr>
</table>


<h3>References</h3>

<p>Haas, B.J., Kamoun, S., Zody, M.C., Jiang, R.H., Handsaker, R.E., Cano, L.M., Grabherr, M.,
Kodira, C.D., Raffaele, S., Torto-Alalibo, T. and Bozkurt, T.O., 2009.
Genome sequence and analysis of the Irish potato famine pathogen Phytophthora infestans.
Nature, 461(7262), p.393.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fasta.file &lt;- system.file("extdata", "test_infestans.fasta", package = "effectR")
ORF &lt;- seqinr::read.fasta(fasta.file)
rxlr.cand &lt;- regex.search(ORF)
custom.cand &lt;- regex.search(ORF, motif = "custom", reg.pat ="^\\w{12,60}r\\wlr\\w{6,10}eer")

</code></pre>

<hr>
<h2 id='shiny.effectR'>Function to run the effectR package using a shiny app</h2><span id='topic+shiny.effectR'></span>

<h3>Description</h3>

<p>This function will launch an interactive web interface to run the effectR package functions to obtain effectors. It runs using the <span class="pkg">shiny</span> R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiny.effectR(mafft.path = NULL, hmm.path = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="shiny.effectR_+3A_mafft.path">mafft.path</code></td>
<td>
<p>Local path of folder containing the MAFFT binary executable file or the executable file itself. If not specified, then MAFFT must be in the program search path.</p>
</td></tr>
<tr><td><code id="shiny.effectR_+3A_hmm.path">hmm.path</code></td>
<td>
<p>Local path of  folder containing the HMMER binaries.  If not specified, then HMMER executables must be in the program search path.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To successfully run this function the user will need to set the
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
