<!DOCTYPE html><html lang="en"><head><title>Help for package eGST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {eGST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eGST-package'><p>eGST (eQTL-based Genetic Sub-Typer): Leveraging eQTLs to identify</p>
individual-level tissue of interest for a complex trait.</a></li>
<li><a href='#eGST'><p>Run eGST.</p></a></li>
<li><a href='#ExampleEQTLgenoData'><p>An example of tissue-specific eQTLs genotype data.</p></a></li>
<li><a href='#ExamplePhenoData'><p>An example of phenotype data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Leveraging eQTLs to Identify Individual-Level Tissue of Interest
for a Complex Trait</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-30</td>
</tr>
<tr>
<td>Description:</td>
<td>Genetic predisposition for complex traits is often manifested through multiple tissues of interest at different time points in the development. As an example, the genetic predisposition for obesity could be manifested through inherited variants that control metabolism through regulation of genes expressed in the brain and/or through the control of fat storage in the adipose tissue by dysregulation of genes expressed in adipose tissue. We present a method eGST (eQTL-based genetic subtyper) that integrates tissue-specific eQTLs with GWAS data for a complex trait to probabilistically assign a tissue of interest to the phenotype of each individual in the study. eGST estimates the posterior probability that an individual's phenotype can be assigned to a tissue based on individual-level genotype data of tissue-specific eQTLs and marginal phenotype data in a genome-wide association study (GWAS) cohort. Under a Bayesian framework of mixture model, eGST employs a maximum a posteriori (MAP) expectation-maximization (EM) algorithm to estimate the tissue-specific posterior probability across individuals. Methodology is available from: A Majumdar, C Giambartolomei, N Cai, MK Freund, T Haldar, T Schwarz, J Flint, B Pasaniuc (2019) &lt;<a href="https://doi.org/10.1101%2F674226">doi:10.1101/674226</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ArunabhaCodes/eGST">https://github.com/ArunabhaCodes/eGST</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ArunabhaCodes/eGST/issues">https://github.com/ArunabhaCodes/eGST/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>purrr, mvtnorm, MASS, utils, stats, matrixStats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-07-01 06:55:20 UTC; Arunabha</td>
</tr>
<tr>
<td>Author:</td>
<td>Arunabha Majumdar [aut, cre],
  Tanushree Haldar [aut],
  Bogdan Pasaniuc [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arunabha Majumdar &lt;statgen.arunabha@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-07-02 15:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='eGST-package'>eGST (eQTL-based Genetic Sub-Typer): Leveraging eQTLs to identify
individual-level tissue of interest for a complex trait.</h2><span id='topic+eGST-package'></span><span id='topic+_PACKAGE'></span>

<h3>Description</h3>

<p>Genetic predisposition for complex traits is often manifested through multiple tissues
of interest at different time points in the development. As an example, the genetic predisposition
for obesity could be manifested through inherited variants that control metabolism through regulation
of genes expressed in the brain and/or through the control of fat storage in the adipose tissue by
dysregulation of genes expressed in adipose tissue. We present a method eGST that integrates
tissue-specific eQTLs with GWAS data for a complex trait to probabilistically assign a
tissue of interest to the phenotype of each individual in the study. eGST estimates
the posterior probability that an individual's phenotype can be assigned to a
tissue based on individual-level genotype data of tissue-specific eQTLs and marginal
phenotype data in a GWAS cohort. Under a Bayesian framework of mixture model,
eGST employs a maximum a posteriori (MAP) expectation-maximization (EM) algorithm
to estimate the tissue-specific posterior probability across individuals.
</p>


<h3>Functions</h3>


<dl>
<dt><code><a href="#topic+eGST">eGST</a></code></dt><dd><p>It estimates the posterior probability that the genetic susceptibility
of the phenotype of an individual in the study is mediated through eQTLs specific to a tissue
of interest. The phenotype across individuals can be classified into tissues under consideration
based on the estimated tissue-specific posterior probability across individuals.</p>
</dd></dl>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Arunabha Majumdar <a href="mailto:statgen.arunabha@gmail.com">statgen.arunabha@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Tanushree Haldar <a href="mailto:tanushree.haldar@gmail.com">tanushree.haldar@gmail.com</a>
</p>
</li>
<li><p> Bogdan Pasaniuc <a href="mailto:pasaniuc@ucla.edu">pasaniuc@ucla.edu</a>
</p>
</li></ul>



<h3>References</h3>

<p>Majumdar A, Giambartolomei C, Cai N, Freund MK, Haldar T, J Flint, Pasaniuc B (2019) Leveraging eQTLs to identify
tissue-specific genetic subtype of complex trait. bioRxiv.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ArunabhaCodes/eGST">https://github.com/ArunabhaCodes/eGST</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ArunabhaCodes/eGST/issues">https://github.com/ArunabhaCodes/eGST/issues</a>
</p>
</li></ul>


<hr>
<h2 id='eGST'>Run eGST.</h2><span id='topic+eGST'></span>

<h3>Description</h3>

<p>Run eGST to estimate the posterior probability that the genetic
susceptibility of the phenotype of
an individual in the study is mediated through eQTLs specific to a tissue of interest.
To create sets of tissue-specific eQTLs in your context, please see our manuscript:
Majumdar A, Giambartolomei C, Cai N, Freund MK, Haldar T, J Flint, Pasaniuc B (2019)
Leveraging eQTLs to identify tissue-specific genetic subtype of complex trait, bioRxiv.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eGST(pheno, geno, tissues, logLimprovement = 5 * 10^(-8),
  seed_choice = sample(1:1000, size = 1), nIter = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="eGST_+3A_pheno">pheno</code></td>
<td>
<p>A numeric vector of length N where N is the number of individuals. It
contains the GWAS phenotype values of individuals. No default.</p>
</td></tr>
<tr><td><code id="eGST_+3A_geno">geno</code></td>
<td>
<p>A list with K elements where K is the number of tissues. Each element of
geno is the genotype matrix of the eQTLs specific to a tissue in the GWAS cohort.
So j-th element of geno is N by Mj matrix
containing the genotype data of N individuals (rows) at Mj eQTLs (columns) specific to j-th tissue.
Each eQTL is a bi-allelic SNP with minor allele frequency &gt; 0.01.
Genotypes at each eQTL must be normalized across N individuals.
If 0/1/2 valued genotype matrix is provided, it is internally normalized. No default.</p>
</td></tr>
<tr><td><code id="eGST_+3A_tissues">tissues</code></td>
<td>
<p>A character vector of length K. It contains the names of tissues of interest in the analysis.
The order of tissues in this vector must match the order of tissues in the previous argument 'geno'. No default.</p>
</td></tr>
<tr><td><code id="eGST_+3A_loglimprovement">logLimprovement</code></td>
<td>
<p>A positive real number specifying the minimum possible improvement
of data log-likelihood in MAP-EM stopping criterion. Default <code class="reqn">5*10^(-8)</code>.</p>
</td></tr>
<tr><td><code id="eGST_+3A_seed_choice">seed_choice</code></td>
<td>
<p>An integer providing the choice of random seed for initialization in MAP-EM algorithm.
Default is an integer randomly selected in (1,...,1000).</p>
</td></tr>
<tr><td><code id="eGST_+3A_niter">nIter</code></td>
<td>
<p>An integer providing the maximum number of iterations allowed in the MAP-EM algorithm. Default is 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by <code><a href="#topic+eGST">eGST</a></code> is a list which consists of various components.
</p>
<table role = "presentation">
<tr><td><code>gamma</code></td>
<td>
<p>A N by K matrix providing the tissue-specific posterior probability of N individuals across K tissues.</p>
</td></tr>
<tr><td><code>alfa</code></td>
<td>
<p>Baseline tissue-specific intercepts/means of the trait.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Tissue-specific eQTLs' genetic effect on the trait.</p>
</td></tr>
<tr><td><code>sigma_g</code></td>
<td>
<p>Square root of the variance of tissue-specific per-eQTL genetic effect on the trait.</p>
</td></tr>
<tr><td><code>sigma_e</code></td>
<td>
<p>Square root of the error variance of tissue-specific subtype of the trait
which remains unexplained by the tissue-specific eQTLs.</p>
</td></tr>
<tr><td><code>m</code></td>
<td>
<p>Number of tissue-specific eQTLs.</p>
</td></tr>
<tr><td><code>logL</code></td>
<td>
<p>log-likelihood of the data.</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Majumdar, C Giambartolomei, N Cai, MK Freund, T Haldar, T Schwarz, J Flint, B Pasaniuc (2019) Leveraging eQTLs to identify
tissue-specific genetic subtype of complex trait. bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExamplePhenoData)
pheno &lt;- ExamplePhenoData
head(pheno)
data(ExampleEQTLgenoData)
geno &lt;- ExampleEQTLgenoData
geno[[1]][1:5,1:5]
geno[[2]][1:5,1:5]
tissues &lt;- paste("tissue", 1:2, sep = "")
result &lt;- eGST(pheno, geno, tissues)
str(result)

</code></pre>

<hr>
<h2 id='ExampleEQTLgenoData'>An example of tissue-specific eQTLs genotype data.</h2><span id='topic+ExampleEQTLgenoData'></span>

<h3>Description</h3>

<p>ExampleEQTLgenoData is a list with two elements each containing a 1000 by 100 ordered genotype matrix.
Each matrix provides the genotype data of the 1000 individuals at 100 tissue-specific eQTLs for each tissue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleEQTLgenoData)
</code></pre>


<h3>Format</h3>

<p>A list of two numeric matrix each having 1000 rows (individuals) and 100 columns (eQTLs):</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleEQTLgenoData)
geno &lt;- ExampleEQTLgenoData
</code></pre>

<hr>
<h2 id='ExamplePhenoData'>An example of phenotype data.</h2><span id='topic+ExamplePhenoData'></span>

<h3>Description</h3>

<p>ExamplePhenoData is a simulated vector of phenotype for 1000 individuals.
In this simulated example dataset, we have considered two tissues and corresponding
sets of 100 tissue-specific eQTLs each.
First half of 1000 individuals' phenotypes were simulated to have genetic effect from the first tissue
specific eQTLs, but no effect from the second tissue-specific eQTLs. Hence first 500 individuals
were assigned to the first tissue. Similarly, second half of
the 1000 individuals were simulated to have genetic effect from the second-tissue specific eQTLs and hence
they were assigned to the second tissue.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExamplePhenoData)
</code></pre>


<h3>Format</h3>

<p>A numeric vector of length 1000 containing the phenotype data of 1000 individuals.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExamplePhenoData)
pheno &lt;- ExamplePhenoData
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
