<!DOCTYPE html><html lang="en"><head><title>Help for package EHR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EHR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#EHR-package'><p>Electronic Health Record (EHR) Data Processing and Analysis Tool</p></a></li>
<li><a href='#addLastDose'><p>Add Lastdose Data</p></a></li>
<li><a href='#analysisPheWAS'><p>Statistical Analysis for PheWAS</p></a></li>
<li><a href='#build_lastdose'><p>PK oral functions</p></a></li>
<li><a href='#build-internal'><p>Internal functions for buildDose process</p></a></li>
<li><a href='#buildDose'><p>Combine Dose Data</p></a></li>
<li><a href='#collapse-internal'><p>Internal functions for collapseDose process</p></a></li>
<li><a href='#collapseDose'><p>Collapse Dose Data</p></a></li>
<li><a href='#dataTransformation'><p>Data Transformation</p></a></li>
<li><a href='#dd'><p>dd</p></a></li>
<li><a href='#dd.baseline'><p>dd.baseline</p></a></li>
<li><a href='#dd.baseline.small'><p>dd.baseline.small</p></a></li>
<li><a href='#dd.small'><p>dd.small</p></a></li>
<li><a href='#extractMed'><p>Extract medication information from clinical notes</p></a></li>
<li><a href='#extractMed-internal'><p>Internal functions for Extract-Med module</p></a></li>
<li><a href='#freqNum'><p>Convert Character Frequency to Numeric</p></a></li>
<li><a href='#idCrosswalk'><p>Create ID Crosswalk</p></a></li>
<li><a href='#lam_metadata'><p>Example of Metadata for Lamotrigine Data</p></a></li>
<li><a href='#lam_mxr_parsed'><p>Example of Lamotrigine Output from 'parseMedExtractR'</p></a></li>
<li><a href='#lastdose-internal'><p>Internal functions for lastdose</p></a></li>
<li><a href='#Logistf'><p>Firth's penalized-likelihood logistic regression with more decimal places of</p>
p-value than <code>logistf</code> function in the R package &lsquo;logistf&rsquo;</a></li>
<li><a href='#makeDose'><p>Make Dose Data</p></a></li>
<li><a href='#mod-internal'><p>Internal functions for EHR modules</p></a></li>
<li><a href='#parse-internal'><p>Internal functions for parse process</p></a></li>
<li><a href='#parseCLAMP'><p>Parse CLAMP NLP Output</p></a></li>
<li><a href='#parseMedEx'><p>Parse MedEx NLP Output</p></a></li>
<li><a href='#parseMedExtractR'><p>Parse medExtractR NLP Output</p></a></li>
<li><a href='#parseMedXN'><p>Parse MedXN NLP Output</p></a></li>
<li><a href='#pk-internal'><p>Internal functions for pk analysis</p></a></li>
<li><a href='#processLastDose'><p>Process and standardize extracted last dose times</p></a></li>
<li><a href='#pullFakeId'><p>Pull Fake/Mod ID</p></a></li>
<li><a href='#pullRealId'><p>Pull Real ID</p></a></li>
<li><a href='#readTransform'><p>Read and Transform</p></a></li>
<li><a href='#run_Build_PK_IV'><p>Build-PK-IV Module</p></a></li>
<li><a href='#run_Build_PK_Oral'><p>Build-PK-Oral Module</p></a></li>
<li><a href='#run_Demo'><p>Run Demographic Data</p></a></li>
<li><a href='#run_DrugLevel'><p>Run Drug Level Data</p></a></li>
<li><a href='#run_Labs'><p>Run Lab Data</p></a></li>
<li><a href='#run_MedStrI'><p>Run Str Data I</p></a></li>
<li><a href='#run_MedStrII'><p>Run Structured E-Prescription Data</p></a></li>
<li><a href='#stdzDose'><p>Standardize Dose Entity</p></a></li>
<li><a href='#stdzDoseChange'><p>Standardize Dose Change Entity</p></a></li>
<li><a href='#stdzDoseSchedule'><p>Standardize Dose Schedule Entity</p></a></li>
<li><a href='#stdzDuration'><p>Standardize Duration Entity</p></a></li>
<li><a href='#stdzFreq'><p>Standardize Frequency Entity</p></a></li>
<li><a href='#stdzRoute'><p>Standardize Route Entity</p></a></li>
<li><a href='#stdzStrength'><p>Standardize Strength Entity</p></a></li>
<li><a href='#tac_lab'><p>Example of Lab Time Data for Tacrolimus</p></a></li>
<li><a href='#tac_metadata'><p>Example of Metadata for Tacrolimus Data</p></a></li>
<li><a href='#tac_mxr_parsed'><p>Example of Tacrolimus Output from 'parseMedExtractR'</p></a></li>
<li><a href='#writeCheckData'><p>Write Check File as CSV</p></a></li>
<li><a href='#zeroOneTable'><p>Make Zero One Contingency Tables</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.4-11</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-12-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Electronic Health Record (EHR) Data Processing and Analysis Tool</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Leena Choi &lt;leena.choi@vanderbilt.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Process and analyze electronic health record (EHR) data. The 'EHR'
    package provides modules to perform diverse medication-related studies using
    data from EHR databases. Especially, the package includes modules to perform
    pharmacokinetic/pharmacodynamic (PK/PD) analyses using EHRs, as outlined in
    Choi, Beck, McNeer, Weeks, Williams, James, Niu, Abou-Khalil, Birdwell, Roden, 
    Stein, Bejan, Denny, and Van Driest (2020) &lt;<a href="https://doi.org/10.1002%2Fcpt.1787">doi:10.1002/cpt.1787</a>&gt;. Additional 
    modules will be added in future. In addition, this package provides various 
    functions useful to perform Phenome Wide Association Study (PheWAS) to explore 
    associations between drug exposure and phenotypes obtained from EHR data, as 
    outlined in Choi, Carroll, Beck, Mosley, Roden, Denny, and Van Driest (2018) 
    &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty306">doi:10.1093/bioinformatics/bty306</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://choileena.github.io/">https://choileena.github.io/</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, data.table, methods, lubridate, pkdata</td>
</tr>
<tr>
<td>Suggests:</td>
<td>glmnet, logistf, medExtractR (&ge; 0.4.1), knitr, rmarkdown,
ggplot2, markdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-19 16:07:42 UTC; beckca</td>
</tr>
<tr>
<td>Author:</td>
<td>Leena Choi <a href="https://orcid.org/0000-0002-2544-7090"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Cole Beck <a href="https://orcid.org/0000-0002-6849-6255"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Hannah Weeks <a href="https://orcid.org/0000-0002-0262-6790"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Elizabeth McNeer [aut],
  Nathan James <a href="https://orcid.org/0000-0001-7079-9151"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Michael Williams [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-19 18:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='EHR-package'>Electronic Health Record (EHR) Data Processing and Analysis Tool</h2><span id='topic+EHR'></span><span id='topic+EHR-package'></span>

<h3>Description</h3>

<p>The 'EHR' package provides modules to perform diverse medication-related
studies using data from EHR databases.
</p>


<h3>Details</h3>

<p>Package functionality:
</p>

<ul>
<li><p> Process and analyze Electronic Health Record (EHR) data.
</p>
</li>
<li><p> Implement modules to perform diverse medication-related studies using
data from EHR databases. Especially, the package includes modules to perform
pharmacokinetic/pharmacodynamic (PK/PD) analyses using EHRs, as outlined in
Choi et al. (2020).
</p>
</li>
<li><p> Implement three statistical methods for Phenome Wide Association Study
(PheWAS). Contingency tables for many binary outcomes (e.g., phenotypes) and
a binary covariate (e.g., drug exposure) can be efficiently generated by
<code><a href="#topic+zeroOneTable">zeroOneTable</a></code>, and three commonly used statistical methods to
analyze data for PheWAS are implement by <code><a href="#topic+analysisPheWAS">analysisPheWAS</a></code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Leena Choi <a href="mailto:leena.choi@vanderbilt.edu">leena.choi@vanderbilt.edu</a> (<a href="https://orcid.org/0000-0002-2544-7090">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Cole Beck <a href="mailto:cole.beck@vumc.org">cole.beck@vumc.org</a> (<a href="https://orcid.org/0000-0002-6849-6255">ORCID</a>)
</p>
</li>
<li><p> Hannah Weeks <a href="mailto:hannah.l.weeks@vanderbilt.edu">hannah.l.weeks@vanderbilt.edu</a> (<a href="https://orcid.org/0000-0002-0262-6790">ORCID</a>)
</p>
</li>
<li><p> Elizabeth McNeer <a href="mailto:elizabeth.mcneer@vumc.org">elizabeth.mcneer@vumc.org</a>
</p>
</li>
<li><p> Nathan James <a href="mailto:nathan.t.james@vanderbilt.edu">nathan.t.james@vanderbilt.edu</a> (<a href="https://orcid.org/0000-0001-7079-9151">ORCID</a>)
</p>
</li>
<li><p> Michael Williams <a href="mailto:michael.l.williams@vanderbilt.edu">michael.l.williams@vanderbilt.edu</a>
</p>
</li></ul>



<h3>References</h3>


<ol>
<li><p> Development of a system for postmarketing population pharmacokinetic
and pharmacodynamic studies using real-world data from electronic health
records.<br />
Choi L, Beck C, McNeer E, Weeks HL, Williams ML, James NT, Niu X,
Abou-Khalil BW, Birdwell KA, Roden DM, Stein CM, Bejan CA, Denny JC,
Van Driest SL.<br />
Clin Pharmacol Ther. 2020 Apr;107(4):934-943. doi: 10.1002/cpt.1787.
</p>
</li>
<li><p> Evaluating statistical approaches to leverage large clinical datasets
for uncovering therapeutic and adverse medication effects.<br />
Choi L, Carroll RJ, Beck C, Mosley JD, Roden DM, Denny JC, Van Driest SL.<br />
Bioinformatics. 2018 Sep 1;34(17):2988-2996. doi: 10.1093/bioinformatics/bty306.
</p>
</li></ol>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://choileena.github.io/">https://choileena.github.io/</a>
</p>
</li></ul>


<hr>
<h2 id='addLastDose'>Add Lastdose Data</h2><span id='topic+addLastDose'></span>

<h3>Description</h3>

<p>Add lastdose data to data set from the <code><a href="#topic+buildDose">buildDose</a></code> process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addLastDose(buildData, lastdoseData)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addLastDose_+3A_builddata">buildData</code></td>
<td>
<p>data.frame, output of <code><a href="#topic+buildDose">buildDose</a></code> function.</p>
</td></tr>
<tr><td><code id="addLastDose_+3A_lastdosedata">lastdoseData</code></td>
<td>
<p>data.frame with columns filename, ld_start, lastdose, raw_time, time_type</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Lastdose is a datetime string associated with dose data. Information on time
of last dose can be extracted within the <code><a href="#topic+extractMed">extractMed</a></code> function
(i.e., <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code>) using the argument
<code>lastdose=TRUE</code>. Raw extracted times should first be processed using the
<code><a href="#topic+processLastDose">processLastDose</a></code> function to convert to datetime format before providing
to <code>addLastDose</code>. This function then combines the processed last dose times with output
from the <code><a href="#topic+buildDose">buildDose</a></code> process by file name to pair last dose times with dosing regimens based on position.
Alternatively, the user can provide their own table of lastdose data. In this case, with position
information absent, the lastdose data should be restricted to one unique last dose time per
unique patient ID-date identifier.
</p>
<p>In the case where <code>lastdoseData</code> is output from <code><a href="#topic+processLastDose">processLastDose</a></code>, it is possible to
have more than one extracted last dose time. In this case, rules are applied to determine which
time should be kept. First, we give preference to an explicit time expression (e.g., &quot;10:30pm&quot;)
over a duration expression (e.g., &quot;14 hour level&quot;). Then, we pair last dose times with drug regimens
based on minimum distance between last dose time start position and drug name start position.
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP for details.
</p>


<h3>Value</h3>

<p>a data.frame with the &lsquo;lastdose&rsquo; column added.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get build data
data(tac_mxr_parsed)
# don't combine lastdose at this stage
tac_build &lt;- buildDose(tac_mxr_parsed, preserve = 'lastdose')
# Get processed last dose data
tac_mxr &lt;- read.csv(system.file("examples", "tac_mxr.csv", package = "EHR"))
data(tac_metadata)
data(tac_lab)
ld_data &lt;- processLastDose(tac_mxr, tac_metadata, tac_lab)

addLastDose(tac_build, ld_data)

</code></pre>

<hr>
<h2 id='analysisPheWAS'>Statistical Analysis for PheWAS</h2><span id='topic+analysisPheWAS'></span>

<h3>Description</h3>

<p>Implement three commonly used statistical methods to analyze data for
Phenome Wide Association Study (PheWAS)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analysisPheWAS(
  method = c("firth", "glm", "lr"),
  adjust = c("PS", "demo", "PS.demo", "none"),
  Exposure,
  PS,
  demographics,
  phenotypes,
  data
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analysisPheWAS_+3A_method">method</code></td>
<td>
<p>define the statistical analysis method from 'firth', 'glm', and
'lr'. 'firth': Firth's penalized-likelihood logistic regression; 'glm':
logistic regression with Wald test, 'lr': logistic regression with likelihood
ratio test.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_adjust">adjust</code></td>
<td>
<p>define the adjustment method from 'PS','demo','PS.demo', and
'none'. 'PS': adjustment of PS only; 'demo': adjustment of demographics only;
'PS.demo': adjustment of PS and demographics; 'none': no adjustment.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_exposure">Exposure</code></td>
<td>
<p>define the variable name of exposure variable.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_ps">PS</code></td>
<td>
<p>define the variable name of propensity score.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_demographics">demographics</code></td>
<td>
<p>define the list of demographic variables.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_phenotypes">phenotypes</code></td>
<td>
<p>define the list of phenotypes that need to be analyzed.</p>
</td></tr>
<tr><td><code id="analysisPheWAS_+3A_data">data</code></td>
<td>
<p>define the data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implements three commonly used statistical methods to analyze the associations 
between exposure (e.g., drug exposure, genotypes) and various phenotypes in 
PheWAS. Firth's penalized-likelihood logistic regression is the default method to 
avoid the problem of separation in logistic regression, which is often a problem 
when analyzing sparse binary outcomes and exposure. Logistic regression with 
likelihood ratio test and conventional logistic regression with Wald test can be 
also performed.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>estimate</code></td>
<td>
<p>the estimate of log odds ratio.</p>
</td></tr>
<tr><td><code>stdError</code></td>
<td>
<p>the standard error.</p>
</td></tr>
<tr><td><code>statistic</code></td>
<td>
<p>the test statistic.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>the p-value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Leena Choi <a href="mailto:leena.choi@vanderbilt.edu">leena.choi@vanderbilt.edu</a> and
Cole Beck <a href="mailto:cole.beck@vumc.org">cole.beck@vumc.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## use small datasets to run this example
data(dataPheWASsmall)
## make dd.base with subset of covariates from baseline data (dd.baseline.small)
## or select covariates with upper code as shown below
upper.code.list &lt;- unique(sub("[.][^.]*(.).*", "", colnames(dd.baseline.small)) )
upper.code.list &lt;- intersect(upper.code.list, colnames(dd.baseline.small))
dd.base &lt;- dd.baseline.small[, upper.code.list]
## perform regularized logistic regression to obtain propensity score (PS) 
## to adjust for potential confounders at baseline
phenos &lt;- setdiff(colnames(dd.base), c('id', 'exposure'))
data.x &lt;- as.matrix(dd.base[, phenos])
glmnet.fit &lt;- glmnet::cv.glmnet(x=data.x, y=dd.base[,'exposure'],
                                family="binomial", standardize=TRUE,
                                alpha=0.1)
dd.base$PS &lt;- c(predict(glmnet.fit, data.x, s='lambda.min'))
data.ps &lt;- dd.base[,c('id', 'PS')]
dd.all.ps &lt;- merge(data.ps, dd.small, by='id')  
demographics &lt;- c('age', 'race', 'gender')
phenotypeList &lt;- setdiff(colnames(dd.small), c('id','exposure','age','race','gender'))
## run with a subset of phenotypeList to get quicker results
phenotypeList.sub &lt;- sample(phenotypeList, 5)
results.sub &lt;- analysisPheWAS(method='firth', adjust='PS', Exposure='exposure',
                              PS='PS', demographics=demographics, 
                              phenotypes=phenotypeList.sub, data=dd.all.ps)
## run with the full list of phenotype outcomes (i.e., phenotypeList)

        results &lt;- analysisPheWAS(method='firth', adjust='PS',Exposure='exposure',
                          PS='PS', demographics=demographics,
                          phenotypes=phenotypeList, data=dd.all.ps) 
         
</code></pre>

<hr>
<h2 id='build_lastdose'>PK oral functions</h2><span id='topic+build_lastdose'></span>

<h3>Description</h3>

<p>codebuild_lastdose: build PK oral with lastdose information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_lastdose(x, first_interval_hours = 336, ldCol = NULL)
</code></pre>

<hr>
<h2 id='build-internal'>Internal functions for buildDose process</h2><span id='topic+build-internal'></span><span id='topic+tstrsplit2'></span><span id='topic+convert'></span><span id='topic+entVal'></span><span id='topic+entStart'></span><span id='topic+entStop'></span><span id='topic+remainder'></span><span id='topic+getCombosIx'></span><span id='topic+getPaths'></span><span id='topic+getOpt'></span><span id='topic+getCosts'></span><span id='topic+findOptPath'></span><span id='topic+stdRoute'></span><span id='topic+grp'></span><span id='topic+anchorByStrength'></span><span id='topic+combineGroups'></span><span id='topic+makeCombos'></span><span id='topic+findRareValues'></span>

<h3>Description</h3>

<p>These internal functions aid the <code><a href="#topic+buildDose">buildDose</a></code> function and process.
</p>


<h3>Details</h3>

<p><code>tstrsplit2</code>: a wrapper for the <code><a href="data.table.html#topic+tstrsplit">tstrsplit</a></code> function in <span class="pkg">data.table</span> 
with some additional formatting 
</p>
<p><code>convert</code>: formats final results that are returned in <code><a href="#topic+buildDose">buildDose</a></code>, including 
separating the extracted expression from its start and stop position
</p>
<p><code>entVal</code>: isolates the extracted expression from a string of the format 
&quot;extracted expression::start position::stop position&quot;
</p>
<p><code>entStart</code>: isolates the start position from a string of the format 
&quot;extracted expression::start position::stop position&quot;
</p>
<p><code>entStop</code>: isolates the stop position from a string of the format 
&quot;extracted expression::start position::stop position&quot;
</p>
<p><code>remainder</code>: performs some formatting for paths containing NA values
</p>
<p><code>getCombosIx</code>: forms the individual paths within the <code>getPaths</code> function
</p>
<p><code>getPaths</code>: builds paths from entity pairings using other helper function <code>getCombosIx</code>
</p>
<p><code>getOpt</code>: takes in the possible paths as well as the cost computations from <code>getCosts</code>, 
combines costs at the path level using the helper functions <code>remainder</code> and <code>getPaths</code>, 
and returns the path with the minimum cost
</p>
<p><code>getCosts</code>: computes the distance for each path based on the distance method specified in <code>buildDose</code> 
</p>
<p><code>findOptPath</code>: uses the helper functions <code>getCosts</code> and <code>getOpt</code> to 
determine the optimal path based on the selected distance method
</p>
<p><code>stdRoute</code>: standardize the &quot;route&quot; entity. This function is also used in the 
<code>collapseDose</code> function and process.
</p>
<p><code>grp</code>: groups drug entities based on a particular anchor point (e.g. strength 
or dose)
</p>
<p><code>anchorByStrength</code>: forms subgroups of entities by anchoring to strength. This is done within existing groups 
which are anchored by drug name
</p>
<p><code>combineGroups</code>: determines how to form groups of the entities based on the complexity of the extraction (i.e., 
number of extractions for each entity)
</p>
<p><code>makeCombos</code>: takes all extracted entities and puts them into groups based on position and minimum cost of paths. 
This is the highest level function within <code><a href="#topic+buildDose">buildDose</a></code>; it calls other helper functions including <code>combineGroups</code> 
and <code>anchorByStrength</code>.
</p>
<p><code>findRareValues</code>: find rare values for each column in a data.frame; rare defaults to less than two percent.
</p>

<hr>
<h2 id='buildDose'>Combine Dose Data</h2><span id='topic+buildDose'></span>

<h3>Description</h3>

<p>Output from parse process is taken and converted into a wide format, grouping drug entity
information together based on various steps and rules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildDose(
  dat,
  dn = NULL,
  preserve = NULL,
  dist_method,
  na_penalty,
  neg_penalty,
  greedy_threshold,
  checkForRare = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildDose_+3A_dat">dat</code></td>
<td>
<p>data.table object from the output of <code><a href="#topic+parseMedExtractR">parseMedExtractR</a></code>,
<code><a href="#topic+parseMedXN">parseMedXN</a></code>, <code><a href="#topic+parseMedEx">parseMedEx</a></code>, or <code><a href="#topic+parseCLAMP">parseCLAMP</a></code></p>
</td></tr>
<tr><td><code id="buildDose_+3A_dn">dn</code></td>
<td>
<p>Regular expression specifying drug name(s) of interest.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_preserve">preserve</code></td>
<td>
<p>Column names to include in output, whose values should not be combined with other rows.
If present, dosechange is always preserved.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_dist_method">dist_method</code></td>
<td>
<p>Distance method to use for calculating distance of various paths. Alternatively set the
&lsquo;ehr.dist_method&rsquo; option, which defaults to &lsquo;minEntEnd&rsquo;.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_na_penalty">na_penalty</code></td>
<td>
<p>Penalty for matching extracted entities with NA. Alternatively set the<br />
&lsquo;ehr.na_penalty&rsquo; option, which defaults to 32.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_neg_penalty">neg_penalty</code></td>
<td>
<p>Penalty for negative distances between frequency/intake time and dose amounts. Alternatively
set the &lsquo;ehr.neg_penalty&rsquo; option, which defaults to 0.5.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_greedy_threshold">greedy_threshold</code></td>
<td>
<p>Threshold to use greedy matching; increasing this value too high could lead to the
algorithm taking a long time to finish. Alternatively set the<br />
&lsquo;ehr.greedy_threshold&rsquo; option, which defaults to 1e8.</p>
</td></tr>
<tr><td><code id="buildDose_+3A_checkforrare">checkForRare</code></td>
<td>
<p>Indicate if rare values for each entity should be found and displayed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>buildDose</code> function takes as its main input (<code>dat</code>), a data.table object that
is the output of a parse process function (<code><a href="#topic+parseMedExtractR">parseMedExtractR</a></code>, <code><a href="#topic+parseMedXN">parseMedXN</a></code>,
<code><a href="#topic+parseMedEx">parseMedEx</a></code>, or <code><a href="#topic+parseCLAMP">parseCLAMP</a></code>). Broadly, the parsed extractions are grouped
together to form wide, more complete drug regimen information. This reformatting facilitates
calculation of dose given intake and daily dose in the <code><a href="#topic+collapseDose">collapseDose</a></code> process.
</p>
<p>The process of creating this output is broken down into multiple steps:
</p>

<ol>
<li><p> Removing rows for any drugs not of interest. Drugs of interest are specified with the <code>dn</code> argument.
</p>
</li>
<li><p> Determining whether extractions are &quot;simple&quot; (only one drug mention and at most one extraction per entity) or
complex. Complex cases can be more straightforward if they contain at most one extraction per entity, or require a
pairing algorithm to determine the best pairing if there are multiple extractions for one or more entities.
</p>
</li>
<li><p> Drug entities are anchored by drug name mention within the parse process. For complex cases, drug entities are
further grouped together anchored at each strength (and dose with medExtractR) extraction.
</p>
</li>
<li><p> For strength groups with multiple extractions for at least one entity, these groups go through a path searching
algorithm, which computes the cost for each path (based on a chosen distance method) and chooses the path with the
lowest cost.
</p>
</li>
<li><p> The chosen paths for each strength group are returned as the final pairings. If route is unique within a strength
group, it is standardized and added to all entries for that strength group.
</p>
</li></ol>

<p>The user can specify additional arguments including:
</p>

<ul>
<li> <p><code>dist_method</code>: The distance method is the metric used to determine which entity path is the most likely to
be correct based on minimum cost.
</p>
</li>
<li> <p><code>na_penalty</code>: NA penalties are incurred when extractions are paired with nothing (i.e., an NA), requiring that
entities be sufficiently far apart from one another before being left unpaired.
</p>
</li>
<li> <p><code>neg_penalty</code>: When working with dose amount (DA) and frequency/intake time (FIT), it is much more common
for the ordering to be DA followed by FIT. Thus, when we observe FIT followed by DA, we apply a negative penalty to make such pairings
less likely.
</p>
</li>
<li> <p><code>greedy threshold</code>: When there are many extractions from a clinical note, the number of possible combinations for paths
can get exponentially large, particularly when the medication extraction natural language processing system is incorrect. The greedy
threshold puts an upper bound on the number of entity pairings to prevent the function from stalling in such cases.
</p>
</li></ul>

<p>If none of the optional arguments are specified, then the <code>buildDose</code> process uses the default option values specified in the EHR
package documentation. 
See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details. For additional details, see McNeer, et al. 2020.
</p>


<h3>Value</h3>

<p>A data.frame object that contains columns for filename (of the clinical note, inherited from the
parse output object <code>dat</code>), drugname, strength, dose, route, freq, duration, and drugname_start.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lam_mxr_parsed)

buildDose(lam_mxr_parsed)
</code></pre>

<hr>
<h2 id='collapse-internal'>Internal functions for collapseDose process</h2><span id='topic+collapse-internal'></span><span id='topic+qrbind'></span><span id='topic+nowarnnum'></span><span id='topic+most'></span><span id='topic+pairDay'></span><span id='topic+mergeAdjacent'></span><span id='topic+borrowFreqDoseSeq'></span><span id='topic+borrowVal'></span><span id='topic+reOrder'></span><span id='topic+borrowWithinDoseSeq'></span><span id='topic+calcDailyDose'></span><span id='topic+daySeqDups'></span><span id='topic+rmDuplicates'></span>

<h3>Description</h3>

<p>These internal functions aid the main functions used in the collapse
process (<code><a href="#topic+collapseDose">collapseDose</a></code>, <code><a href="#topic+makeDose">makeDose</a></code>).
</p>


<h3>Details</h3>

<p><code>qrbind</code>: fast version of rbind.data.frame
</p>
<p><code>nowarnnum</code>: converts a variable to numeric and suppresses warnings
</p>
<p><code>most</code>: returns the mode
</p>
<p><code>pairDay</code>: calculates daily dose for dose sequences
</p>
<p><code>mergeAdjacent</code>: collapses adjacent rows that have a drug name change
</p>
<p><code>borrowFreqDoseSeq</code>: imputes a frequency of 'pm' if the preceding row has a frequency of 'am' or 'noon' and there is a 'pm' elsewhere in the note
</p>
<p><code>borrowVal</code>: borrow a unique value within a drug mention
</p>
<p><code>reOrder</code>: order rows
</p>
<p><code>borrowWithinDoseSeq</code>: borrow values within a dose sequence
</p>
<p><code>calcDailyDose</code>: calculates daily dose
</p>
<p><code>daySeqDups</code>: look for duplicate dose sequence 
</p>
<p><code>rmDuplicates</code>: removes redundant rows
</p>

<hr>
<h2 id='collapseDose'>Collapse Dose Data</h2><span id='topic+collapseDose'></span>

<h3>Description</h3>

<p>Splits drug data and calls <code><a href="#topic+makeDose">makeDose</a></code> to collapse at the note and date level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collapseDose(x, noteMetaData, naFreq = "most", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="collapseDose_+3A_x">x</code></td>
<td>
<p>data.frame containing the output of <code><a href="#topic+buildDose">buildDose</a></code>, or the output of
<code><a href="#topic+addLastDose">addLastDose</a></code> if last dose information is being incorporated.</p>
</td></tr>
<tr><td><code id="collapseDose_+3A_notemetadata">noteMetaData</code></td>
<td>
<p>data.frame containing identifying meta data for each
note, including patient ID, date of the note, and note ID. Column names
should be set to &lsquo;filename&rsquo;, &lsquo;pid&rsquo;, &lsquo;date&rsquo;,
&lsquo;note&rsquo;. Date should have format YYYY-MM-DD.</p>
</td></tr>
<tr><td><code id="collapseDose_+3A_nafreq">naFreq</code></td>
<td>
<p>Expression used to replace missing frequencies with, or by default use the most
common.</p>
</td></tr>
<tr><td><code id="collapseDose_+3A_...">...</code></td>
<td>
<p>drug formulations to split by</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If different formulations of the drug (e.g., extended release) exist, they can be
separated using a regular expression (e.g., &lsquo;xr|er&rsquo;). This function will call
<code><a href="#topic+makeDose">makeDose</a></code> on parsed and paired medication data to calculate dose intake
and daily dose and remove redundancies at the note and date level.
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details.
</p>


<h3>Value</h3>

<p>A list containing two dataframes, one with the note level and one with the date level collapsed data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lam_mxr_parsed)
data(lam_metadata)

lam_build_out &lt;- buildDose(lam_mxr_parsed)

lam_collapsed &lt;- collapseDose(lam_build_out, lam_metadata, naFreq = 'most', 'xr|er')
lam_collapsed$note # Note level collapsing
lam_collapsed$date # Date level collapsing
</code></pre>

<hr>
<h2 id='dataTransformation'>Data Transformation</h2><span id='topic+dataTransformation'></span>

<h3>Description</h3>

<p>Convenience function for making small modifications to a data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataTransformation(x, select, rename, modify)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dataTransformation_+3A_x">x</code></td>
<td>
<p>a data.frame</p>
</td></tr>
<tr><td><code id="dataTransformation_+3A_select">select</code></td>
<td>
<p>columns to select</p>
</td></tr>
<tr><td><code id="dataTransformation_+3A_rename">rename</code></td>
<td>
<p>character vector with names for all columns</p>
</td></tr>
<tr><td><code id="dataTransformation_+3A_modify">modify</code></td>
<td>
<p>list of expressions used to transform data set</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified data.frame
</p>

<hr>
<h2 id='dd'>dd</h2><span id='topic+dd'></span>

<h3>Description</h3>

<p>Simulated outcome data example from Phenome Wide Association Study (PheWAS) 
that examines associations between drug exposure and various phenotypes at follow-up 
after the drug exposure. The dataset includes 1505 variables: subject identification 
number ('id'), drug exposure ('exposure'), 3 demographic variables 
('age', 'race', 'gender'), and 1500 phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dataPheWAS, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 10000 observations on 1505 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataPheWAS)
</code></pre>

<hr>
<h2 id='dd.baseline'>dd.baseline</h2><span id='topic+dd.baseline'></span>

<h3>Description</h3>

<p>Simulated baseline data example from a Phenome Wide Association Study (PheWAS) 
obtained at baseline before drug exposure. The dataset includes 1505 variables: 
subject identification number ('id'), drug exposure ('exposure'), 3 demographic 
variables ('age', 'race', 'gender'), and 1500 phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dataPheWAS, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 10000 observations on 1505 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataPheWAS)
</code></pre>

<hr>
<h2 id='dd.baseline.small'>dd.baseline.small</h2><span id='topic+dd.baseline.small'></span>

<h3>Description</h3>

<p>A smaller subset of baseline data example, dd.baseline. The dataset includes 55
variables: subject identification number ('id'), drug exposure ('exposure'),
3 demographic variables ('age', 'race', 'gender'), and 50 phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dataPheWASsmall, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 observations on 55 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataPheWASsmall)
</code></pre>

<hr>
<h2 id='dd.small'>dd.small</h2><span id='topic+dd.small'></span>

<h3>Description</h3>

<p>A smaller subset of outcome data example, 'dd'. The dataset includes 55
variables: subject identification number ('id'), drug exposure ('exposure'),
3 demographic variables ('age', 'race', 'gender'), and 50 phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dataPheWASsmall, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 observations on 55 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataPheWASsmall)
</code></pre>

<hr>
<h2 id='extractMed'>Extract medication information from clinical notes</h2><span id='topic+extractMed'></span>

<h3>Description</h3>

<p>This function is an interface to the <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code> function
within the <span class="pkg">medExtractR</span> package, and allows drug dosing information to be extracted
from free-text sources, e.g., clinical notes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractMed(note_fn, drugnames, drgunit, windowlength, max_edit_dist = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractMed_+3A_note_fn">note_fn</code></td>
<td>
<p>File name(s) for the text file(s) containing the clinical notes. Can be
a character string for an individual note, or a vector or list of file names for
multiple notes.</p>
</td></tr>
<tr><td><code id="extractMed_+3A_drugnames">drugnames</code></td>
<td>
<p>Vector of drug names for which dosing information should be extracted.
Can include various forms (e.g., generic, brand name) as well as abbreviations.</p>
</td></tr>
<tr><td><code id="extractMed_+3A_drgunit">drgunit</code></td>
<td>
<p>Unit of the drug being extracted, e.g., 'mg'</p>
</td></tr>
<tr><td><code id="extractMed_+3A_windowlength">windowlength</code></td>
<td>
<p>Length of the search window (in characters) around the drug name in
which to search for dosing entities</p>
</td></tr>
<tr><td><code id="extractMed_+3A_max_edit_dist">max_edit_dist</code></td>
<td>
<p>Maximum edit distance allowed when attempting to extract <code>drugnames</code>.
Allows for capturing misspelled drug name information.</p>
</td></tr>
<tr><td><code id="extractMed_+3A_...">...</code></td>
<td>
<p>Additional arguments to <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code>, for example
<code>lastdose=TRUE</code> to extract time of last dose (see <span class="pkg">medExtractR</span> package documentation for details)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Medication information, including dosing data, is often stored in free-text sources such as
clinical notes. The <code>extractMed</code> function serves as a convenient wrapper for the
<span class="pkg">medExtractR</span> package, a natural language processing system written in R for extracting
medication data. Within <code>extractMed</code>, the <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code> function
identifies dosing data for drug(s) of interest, specified by the <code>drugnames</code> argument,
using rule-based and dictionary-based approaches. Relevant dosing entities include medication
strength (identified using the <code>unit</code> argument), dose amount, dose given intake, intake
time or frequency of dose, dose change keywords (e.g., 'increase' or 'decrease'), and time of
last dose. After applying <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code>
to extract drug dosing information, <code>extractMed</code> appends the file name to results to ensure
they are appropriately labeled. 
</p>
<p>See EHR Vignette for for Extract-Med and Pro-Med-NLP. For more details, see Weeks, et al. 2020.
</p>


<h3>Value</h3>

<p>A data.frame with the extracted dosing information, labeled with file name as an identifier <br />
Sample output:<br />
</p>

<table>
<tr>
 <td style="text-align: right;">
filename </td><td style="text-align: right;"> entity    </td><td style="text-align: right;">  expr   </td><td style="text-align: right;">    pos</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> DoseChange</td><td style="text-align: right;">  decrease </td><td style="text-align: right;">  66:74</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> DrugName   </td><td style="text-align: right;"> Prograf </td><td style="text-align: right;">   78:85</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> Strength  </td><td style="text-align: right;">  2 mg   </td><td style="text-align: right;">    86:90</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> DoseAmt   </td><td style="text-align: right;">  1     </td><td style="text-align: right;">     91:92</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> Frequency </td><td style="text-align: right;">  bid    </td><td style="text-align: right;">    101:104</td>
</tr>
<tr>
 <td style="text-align: right;">
note_file1.txt </td><td style="text-align: right;"> LastDose  </td><td style="text-align: right;">  2100    </td><td style="text-align: right;">   121:125</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
tac_fn &lt;- list(system.file("examples", "tacpid1_2008-06-26_note1_1.txt", package = "EHR"),
               system.file("examples", "tacpid1_2008-06-26_note2_1.txt", package = "EHR"),
               system.file("examples", "tacpid1_2008-12-16_note3_1.txt", package = "EHR"))

extractMed(tac_fn,
           drugnames = c("tacrolimus", "prograf", "tac", "tacro", "fk", "fk506"),
           drgunit = "mg",
           windowlength = 60,
           max_edit_dist = 2,
           lastdose=TRUE)

</code></pre>

<hr>
<h2 id='extractMed-internal'>Internal functions for Extract-Med module</h2><span id='topic+extractMed-internal'></span><span id='topic+getNote'></span><span id='topic+getDose'></span>

<h3>Description</h3>

<p>These internal functions aid the main function used in the Extract-Med module (<code><a href="#topic+extractMed">extractMed</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNote(note)
</code></pre>


<h3>Details</h3>

<p><code>getNote</code>: Takes the file name and reads in the note as a character vector
</p>
<p><code>getDose</code>: Runs <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code> on a single clinical note (file) and labels 
the results output with file name
</p>

<hr>
<h2 id='freqNum'>Convert Character Frequency to Numeric</h2><span id='topic+freqNum'></span>

<h3>Description</h3>

<p>This function converts the frequency entity to numeric.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freqNum(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="freqNum_+3A_x">x</code></td>
<td>
<p>character vector of extracted frequency values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- stdzFreq(c('in the morning', 'four times a day', 'with meals'))
freqNum(f)
</code></pre>

<hr>
<h2 id='idCrosswalk'>Create ID Crosswalk</h2><span id='topic+idCrosswalk'></span>

<h3>Description</h3>

<p>Link ID columns from multiple data sets. De-identified columns
are created to make a crosswalk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idCrosswalk(data, idcols, visit.id = "subject_id", uniq.id = "subject_uid")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="idCrosswalk_+3A_data">data</code></td>
<td>
<p>list of data.frames</p>
</td></tr>
<tr><td><code id="idCrosswalk_+3A_idcols">idcols</code></td>
<td>
<p>list of character vectors, indicating ID columns
found in each data set given in &lsquo;data&rsquo;</p>
</td></tr>
<tr><td><code id="idCrosswalk_+3A_visit.id">visit.id</code></td>
<td>
<p>character sting indicating visit-level ID variable (default is &quot;subject_id&quot;)</p>
</td></tr>
<tr><td><code id="idCrosswalk_+3A_uniq.id">uniq.id</code></td>
<td>
<p>character sting indicating subject-level ID variable (default is &quot;subject_uid&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&lsquo;visit.id&rsquo; and &lsquo;uniq.id&rsquo; may occur multiple
times, but should have a one-to-one linkage defined by at least
one of the input data sets. A new visit number is generated for
each repeated &lsquo;uniq.id&rsquo;.
</p>


<h3>Value</h3>

<p>crosswalk of ID columns and their de-identified versions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_data &lt;- data.frame(subj_id=c(4.1,4.2,5.1,6.1),
                        pat_id=c(14872,14872,24308,37143),
                        gender=c(1,1,0,1),
                        weight=c(34,42,28,63),
                        height=c(142,148,120,167))

conc_data &lt;- data.frame(subj_id=rep((4:6)+0.1,each=5),
                        event=rep(1:5,times=3),
                        conc.level=15*exp(-1*rep(1:5,times=3))+rnorm(15,0,0.1))

data &lt;- list(demo_data, conc_data)
idcols &lt;- list(c('subj_id', 'pat_id'), 'subj_id')
idCrosswalk(data, idcols, visit.id='subj_id', uniq.id='pat_id')

</code></pre>

<hr>
<h2 id='lam_metadata'>Example of Metadata for Lamotrigine Data</h2><span id='topic+lam_metadata'></span>

<h3>Description</h3>

<p>An example of the metadata needed for the <code><a href="#topic+processLastDose">processLastDose</a></code>,
<code><a href="#topic+makeDose">makeDose</a></code>, and <code><a href="#topic+collapseDose">collapseDose</a></code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lam_metadata, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 observations on the following variables.
</p>

<dl>
<dt>filename</dt><dd><p>A character vector, filename for the clinical note</p>
</dd>
<dt>pid</dt><dd><p>A character vector, patient ID associated with the filename</p>
</dd>
<dt>date</dt><dd><p>A character vector, date associated with the filename</p>
</dd>
<dt>note</dt><dd><p>A character vector, note ID associated with the filename</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(lam_metadata)
</code></pre>

<hr>
<h2 id='lam_mxr_parsed'>Example of Lamotrigine Output from 'parseMedExtractR'</h2><span id='topic+lam_mxr_parsed'></span>

<h3>Description</h3>

<p>The output after running <code><a href="#topic+parseMedExtractR">parseMedExtractR</a></code> on 4 example clinical notes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lam_mxr_parsed, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations on the following variables.
</p>

<dl>
<dt>filename</dt><dd><p>A character vector, filename for the clinical note</p>
</dd>
<dt>drugname</dt><dd><p>A character vector, drug name extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>strength</dt><dd><p>A character vector, strengths extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dose</dt><dd><p>A character vector, dose amounts extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>route</dt><dd><p>A character vector, routes extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>freq</dt><dd><p>A character vector, frequencies extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dosestr</dt><dd><p>A character vector, dose intakes extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dosechange</dt><dd><p>A character vector, dose change keywords extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>lastdose</dt><dd><p>A character vector, last dose times extracted from the clinical note along with start and stop positions</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(lam_mxr_parsed)
</code></pre>

<hr>
<h2 id='lastdose-internal'>Internal functions for lastdose</h2><span id='topic+lastdose-internal'></span><span id='topic+matchLastDose'></span><span id='topic+mergeLastDose'></span><span id='topic+checkTime'></span><span id='topic+checkDuration'></span><span id='topic+standardizeTime'></span><span id='topic+convertMilitary'></span><span id='topic+convertMornNight'></span><span id='topic+convertAMPM'></span><span id='topic+getDuration'></span>

<h3>Description</h3>

<p>These internal functions incorporate lastdose.
</p>


<h3>Details</h3>

<p><code>matchLastDose</code>: match lastdose data to built data
</p>
<p><code>mergeLastDose</code>: merge lastdose data into built data
</p>
<p><code>getDuration</code>: Duration - don't convert just extract number
</p>
<p><code>convertAMPM</code>: convert am/pm times
</p>
<p><code>convertMornNight</code>: if night/morning used, convert to am or pm
</p>
<p><code>convertMilitary</code>: convert military format into 12h
</p>
<p><code>standardizeTime</code>: create standardized form of time
</p>
<p><code>checkTime</code>: check if lastdose expression is a time expression in the format HH:MM:SS
</p>
<p><code>checkDuration</code>: check if lastdose expression is a duration expression (number that is
not in the HH:MM:SS format)
</p>

<hr>
<h2 id='Logistf'>Firth's penalized-likelihood logistic regression with more decimal places of
p-value than <code>logistf</code> function in the R package &lsquo;logistf&rsquo;</h2><span id='topic+Logistf'></span>

<h3>Description</h3>

<p>Adapted from <code>logistf</code> in the R package &lsquo;logistf&rsquo;, this is 
the same as <code>logistf</code> except that it provides more decimal places 
of p-value that would be useful for Genome-Wide Association Study (GWAS) 
or Phenome Wide Association Study (PheWAS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Logistf(
  formula = attr(data, "formula"),
  data = sys.parent(),
  pl = TRUE,
  alpha = 0.05,
  control,
  plcontrol,
  firth = TRUE,
  init,
  weights,
  plconf = NULL,
  dataout = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Logistf_+3A_formula">formula</code></td>
<td>
<p>a formula object, with the response on the left of the
operator, and the model terms on the right. The response must be a vector
with 0 and 1 or FALSE and TRUE for the  outcome, where the higher value (1 or
TRUE) is modeled. It is possible to include contrasts, interactions, nested
effects, cubic or polynomial splines and all S features as well, e.g.
<code>Y ~ X1*X2 + ns(X3, df=4)</code>. From version 1.10, you may also include
offset() terms.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_data">data</code></td>
<td>
<p>a data.frame where the variables named in the formula can be
found, i. e. the variables containing the binary response and the covariates.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_pl">pl</code></td>
<td>
<p>specifies if confidence intervals and tests should be based on the
profile penalized log likelihood (<code>pl=TRUE</code>, the default) or on the Wald
method (<code>pl=FALSE</code>).</p>
</td></tr>
<tr><td><code id="Logistf_+3A_alpha">alpha</code></td>
<td>
<p>the significance level (1-<code class="reqn">\alpha</code> the confidence level,
0.05 as default).</p>
</td></tr>
<tr><td><code id="Logistf_+3A_control">control</code></td>
<td>
<p>Controls Newton-Raphson iteration. Default is <br />
<code>control=logistf.control(maxstep, maxit, maxhs, lconv, gconv, xconv</code>)</p>
</td></tr>
<tr><td><code id="Logistf_+3A_plcontrol">plcontrol</code></td>
<td>
<p>Controls Newton-Raphson iteration for the estimation of the
profile likelihood confidence intervals. Default is <br />
<code>plcontrol=logistpl.control(maxstep, maxit,</code>
<code>maxhs, lconv, xconv, ortho, pr)</code></p>
</td></tr>
<tr><td><code id="Logistf_+3A_firth">firth</code></td>
<td>
<p>use of Firth's penalized maximum likelihood (<code>firth=TRUE</code>,
default) or the standard maximum likelihood method (<code>firth=FALSE</code>) for
the logistic regression. Note that by specifying <code>pl=TRUE</code> and
<code>firth=FALSE</code> (and probably a lower number of iterations) one obtains
profile likelihood confidence intervals for maximum likelihood logistic
regression parameters.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_init">init</code></td>
<td>
<p>specifies the initial values of the coefficients for the fitting
algorithm.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_weights">weights</code></td>
<td>
<p>specifies case weights. Each line of the input data set is
multiplied by the corresponding element of <code>weights</code>.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_plconf">plconf</code></td>
<td>
<p>specifies the variables (as vector of their indices) for which
profile likelihood confidence intervals should be computed. Default is to
compute for all variables.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_dataout">dataout</code></td>
<td>
<p>If TRUE, copies the <code>data</code> set to the output object.</p>
</td></tr>
<tr><td><code id="Logistf_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to logistf.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>same as <code>logistf</code> except for providing more decimal places of p-value.
</p>


<h3>Author(s)</h3>

<p>Leena Choi <a href="mailto:leena.choi@vanderbilt.edu">leena.choi@vanderbilt.edu</a> and
Cole Beck <a href="mailto:cole.beck@vumc.org">cole.beck@vumc.org</a>
</p>


<h3>References</h3>

<p>same as those provided in the R package &lsquo;logistf&rsquo;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataPheWAS)
fit &lt;- Logistf(X264.3 ~ exposure + age + race + gender, data=dd)
summary(fit)
</code></pre>

<hr>
<h2 id='makeDose'>Make Dose Data</h2><span id='topic+makeDose'></span>

<h3>Description</h3>

<p>Takes parsed and paired medication data, calculates dose intake and daily dose, and 
removes redundant information at the note and date level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeDose(x, noteMetaData, naFreq = "most")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeDose_+3A_x">x</code></td>
<td>
<p>data.frame containing the output of <code><a href="#topic+buildDose">buildDose</a></code>, or the output
of <code><a href="#topic+addLastDose">addLastDose</a></code> if last dose information is being incorporated.</p>
</td></tr>
<tr><td><code id="makeDose_+3A_notemetadata">noteMetaData</code></td>
<td>
<p>data.frame containing identifying meta data for each
note, including patient ID, date of the note, and note ID. Column names
should be set to &lsquo;filename&rsquo;, &lsquo;pid&rsquo;, &lsquo;date&rsquo;,
&lsquo;note&rsquo;. Date should have format YYYY-MM-DD.</p>
</td></tr>
<tr><td><code id="makeDose_+3A_nafreq">naFreq</code></td>
<td>
<p>Replacing missing frequencies with this value, or by default the most common value across
the entire set in <code>x</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function standardizes frequency, route, and duration entities. Dose amount, strength,
and frequency entities are converted to numeric. Rows with only drug name and/or route are
removed. If there are drug name changes in adjacent rows (e.g., from a generic to brand name),
these rows are collapsed into one row if there are no conflicts. Missing strengths, dose
amounts, frequencies, and routes are borrowed or imputed using various rules (see McNeer et al.,
2020 for details). Dose given intake and daily dose are calculated. Redundancies are removed at
the date and note level. If time of last dose is being used and it is unique within the level of
collapsing, it is borrowed across all rows.
</p>


<h3>Value</h3>

<p>A list containing two dataframes, one with the note level and one with the date level
collapsed data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lam_mxr_parsed)
data(lam_metadata)

lam_build_out &lt;- buildDose(lam_mxr_parsed)

lam_collapsed &lt;- makeDose(lam_build_out, lam_metadata)
lam_collapsed[[1]] # Note level collapsing
lam_collapsed[[2]] # Date level collapsing
</code></pre>

<hr>
<h2 id='mod-internal'>Internal functions for EHR modules</h2><span id='topic+mod-internal'></span><span id='topic+colflow_mod'></span><span id='topic+findMedFlowRow_mod'></span><span id='topic+flowData_mod'></span><span id='topic+concData_mod'></span><span id='topic+infusionData_mod'></span><span id='topic+resolveDoseDups_mod'></span><span id='topic+processErx'></span><span id='topic+processErxAddl'></span><span id='topic+validateColumns'></span><span id='topic+read'></span><span id='topic+setDoseMar'></span>

<h3>Description</h3>

<p>codecolflow_mod: build a comparison of potentially duplicate flow data
</p>


<h3>Details</h3>

<p>codefindMedFlowRow_mod: find row numbers for matching ID and time
</p>
<p>codeflowData_mod: correct flow data
</p>
<p>codeconcData_mod: correct concentration data
</p>
<p>codeinfusionData_mod: combine flow and MAR data; standardize hourly rate,
which may or may not include weight
</p>
<p>coderesolveDoseDups_mod: correct duplicated infusion and bolus doses
</p>
<p>codewriteCheckData
</p>
<p>codeprocessErx: prepare e-prescription data
</p>
<p>codeprocessErxAddl: additional processing, checking for connected observations
</p>
<p>codevalidateColumns: check columns are in place
</p>
<p>coderead: generic read function
</p>
<p>codesetDoseMar: set unit/rate/date columns
</p>

<hr>
<h2 id='parse-internal'>Internal functions for parse process</h2><span id='topic+parse-internal'></span><span id='topic+medxnColonFormat'></span><span id='topic+entorder'></span><span id='topic+medxnEntityFormat'></span>

<h3>Description</h3>

<p>These internal functions aid the main functions used in the parsing process 
(<code><a href="#topic+parseMedExtractR">parseMedExtractR</a></code>, <code><a href="#topic+parseMedXN">parseMedXN</a></code>,
<code><a href="#topic+parseMedEx">parseMedEx</a></code>, <code><a href="#topic+parseCLAMP">parseCLAMP</a></code>).
</p>


<h3>Details</h3>

<p><code>medxnColonFormat</code>: converts entity information into the 
form &quot;extracted expression::start position::stop position&quot;, similar 
to how output is formatted by default in MedXN output.  
</p>
<p><code>entorder</code>: a helper function that orders the entities by start position  
</p>
<p><code>medxnEntityFormat</code>: a helper function that applies both <code>medxnColonFormat</code> 
to convert entities into the &quot;extraction::start::stop&quot; format and <code>entorder</code> to 
sort them. It then collapses the entities and separates the extractions with backticks
</p>

<hr>
<h2 id='parseCLAMP'>Parse CLAMP NLP Output</h2><span id='topic+parseCLAMP'></span>

<h3>Description</h3>

<p>Takes files with the raw medication extraction output generated by the CLAMP
natural language processing system and converts it into a standardized format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseCLAMP(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseCLAMP_+3A_filename">filename</code></td>
<td>
<p>File name for a single file containing CLAMP output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Output from different medication extraction systems is formatted in different ways.
In order to be able to process the extracted information, we first need to convert
the output from different systems into a standardized format. Extracted expressions
for various drug entities (e.g., drug name, strength, frequency, etc.) each receive
their own column formatted as &quot;extracted expression::start position::stop position&quot;.
If multiple expressions are extracted for the same entity, they will be separated by
backticks.
</p>
<p>CLAMP output files anchor extractions to a specific drug name extraction through
semantic relations.
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details.
</p>


<h3>Value</h3>

<p>A data.table object with columns for filename, drugname, strength, dose, route,
and freq. The filename contains the file name corresponding to the clinical
note. Each of the entity columns are of the format
&quot;extracted expression::start position::stop position&quot;.
</p>

<hr>
<h2 id='parseMedEx'>Parse MedEx NLP Output</h2><span id='topic+parseMedEx'></span>

<h3>Description</h3>

<p>Takes files with the raw medication extraction output generated by the MedEx
natural language processing system and converts it into a standardized format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseMedEx(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseMedEx_+3A_filename">filename</code></td>
<td>
<p>File name for a single file containing MedEx output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Output from different medication extraction systems is formatted in different ways.
In order to be able to process the extracted information, we first need to convert
the output from different systems into a standardized format. Extracted expressions
for various drug entities (e.g., drug name, strength, frequency, etc.) each receive
their own column formatted as &quot;extracted expression::start position::stop position&quot;.
If multiple expressions are extracted for the same entity, they will be separated by
backticks.
</p>
<p>MedEx output files anchor extractions to a specific drug name extraction.
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details.
</p>


<h3>Value</h3>

<p>A data.table object with columns for filename, drugname, strength, dose, route,
and freq. The filename contains the file name corresponding to the clinical
note. Each of the entity columns are of the format
&quot;extracted expression::start position::stop position&quot;.
</p>

<hr>
<h2 id='parseMedExtractR'>Parse medExtractR NLP Output</h2><span id='topic+parseMedExtractR'></span>

<h3>Description</h3>

<p>Takes files with the raw medication extraction output generated by the medExtractR
natural language processing system and converts it into a standardized format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseMedExtractR(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseMedExtractR_+3A_filename">filename</code></td>
<td>
<p>File name for a single file containing medExtractR output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Output from different medication extraction systems is formatted in different ways.
In order to be able to process the extracted information, we first need to convert
the output from different systems into a standardized format. Extracted expressions
for various drug entities (e.g., drug name, strength, frequency, etc.) each receive
their own column formatted as &quot;extracted expression::start position::stop position&quot;.
If multiple expressions are extracted for the same entity, they will be separated by
backticks.
</p>
<p>The medExtractR system returns extractions in a long table format, indicating the
entity, extracted expression, and start:stop position of the extraction. To perform
this initial parsing, entities are paired with the closest preceding drug name. The
one exception to this is the dose change entity, which can occur before the drug name
(see Weeks, et al. 2020 for details). 
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details.
</p>


<h3>Value</h3>

<p>A data.table object with columns for filename, drugname, strength, dose, route,
freq, dosestr, dosechange and lastdose. The filename contains the file name
corresponding to the clinical note. Each of the entity columns are of the format
&quot;extracted expression::start position::stop position&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mxr_output &lt;- system.file("examples", "lam_mxr.csv", package = "EHR")
mxr_parsed &lt;- parseMedExtractR(mxr_output)
mxr_parsed
</code></pre>

<hr>
<h2 id='parseMedXN'>Parse MedXN NLP Output</h2><span id='topic+parseMedXN'></span>

<h3>Description</h3>

<p>Takes files with the raw medication extraction output generated by the MedXN
natural language processing system and converts it into a standardized format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseMedXN(filename, begText = "^[R0-9]+_[0-9-]+_[0-9]+_")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseMedXN_+3A_filename">filename</code></td>
<td>
<p>File name for single file containing MedXN output.</p>
</td></tr>
<tr><td><code id="parseMedXN_+3A_begtext">begText</code></td>
<td>
<p>A regular expression that would indicate the beginning of a new
observation (i.e., extracted clinical note).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Output from different medication extraction systems is formatted in different ways.
In order to be able to process the extracted information, we first need to convert
the output from different systems into a standardized format. Extracted expressions
for various drug entities (e.g., drug name, strength, frequency, etc.) each receive
their own column formatted as &quot;extracted expression::start position::stop position&quot;.
If multiple expressions are extracted for the same entity, they will be separated by
backticks.
</p>
<p>MedXN output files anchor extractions to a specific drug name extraction.
</p>
<p>In MedXN output files, the results from multiple clinical notes can be combined into
a single output file. The beginning of some lines of the output file can indicate
when output for a new observation (or new clinical note) begins. The user should specify
the argument <code>begText</code> to be a regular expression used to identify the lines where output
for a new clinical note begins.
</p>
<p>See EHR Vignette for Extract-Med and Pro-Med-NLP as well as Dose Building Using Example Vanderbilt EHR Data for details.
</p>


<h3>Value</h3>

<p>A data.table object with columns for filename, drugname, strength, dose, route,
freq, and duration. The filename contains the file name corresponding to the clinical
note. Each of the entity columns are of the format
&quot;extracted expression::start position::stop position&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mxn_output &lt;- system.file("examples", "lam_medxn.csv", package = "EHR")
mxn_parsed &lt;- parseMedXN(mxn_output, begText = "^ID[0-9]+_[0-9-]+_")
mxn_parsed
</code></pre>

<hr>
<h2 id='pk-internal'>Internal functions for pk analysis</h2><span id='topic+pk-internal'></span><span id='topic+takeClosest'></span><span id='topic+takeClosestTime'></span><span id='topic+daysDiff'></span><span id='topic+minDiff'></span><span id='topic+fixDates'></span><span id='topic+joinFlowMar'></span><span id='topic+updateInterval_mod'></span><span id='topic+fix_serum_time'></span><span id='topic+merge_by_time'></span><span id='topic+merge_inf_bolus'></span><span id='topic+addZeroDose'></span><span id='topic+pkdata'></span>

<h3>Description</h3>

<p>codetakeClosest: take value at nearest time
</p>


<h3>Details</h3>

<p>codetakeClosestTime: find nearest time
</p>
<p>codedaysDiff: number of days between two dates
</p>
<p>codeminDiff: difference (in minutes) of time vector
</p>
<p>codefixDates: add leading zeroes to date
</p>
<p>codejoinFlowMar: rbind two data sets
</p>
<p>codeupdateInterval_mod: replace maxint with difference between surgery time
and infusion time (if smaller than maxint)
</p>
<p>codefix_serum_time: create time variable
</p>
<p>codemerge_by_time: impute in x by closest time in y
</p>
<p>codemerge_inf_bolus: combine infusion and bolus into one hourly dose file
</p>
<p>codeaddZeroDose: add rows of zero dose, after a gap
</p>
<p>codepkdata
</p>

<hr>
<h2 id='processLastDose'>Process and standardize extracted last dose times</h2><span id='topic+processLastDose'></span>

<h3>Description</h3>

<p>This function takes last dose times extracted using the <span class="pkg">medExtractR</span> system and
processes the times into standardized datetime objects using recorded lab data where necessary.
The raw output from <code><a href="#topic+extractMed">extractMed</a></code> is filtered to just the LastDose extractions. Time expressions
are standardized into HH:MM:SS format based on what category they fall into (e.g., a time represented
with AM/PM, 24-hour military time, etc.). When the last dose time is after 12pm, it is assumed to have
been taken one day previous to the note's date. For any duration extractions (e.g. &quot;14 hour level&quot;),
the last dose time is calculated from the labtime by extracting the appropriate number of hours. The
final dataset is returned with last dose time formatted into a POSIXct variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processLastDose(mxrData, noteMetaData, labData)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="processLastDose_+3A_mxrdata">mxrData</code></td>
<td>
<p>data.frame containing output from the <code><a href="medExtractR.html#topic+medExtractR">medExtractR</a></code> system</p>
</td></tr>
<tr><td><code id="processLastDose_+3A_notemetadata">noteMetaData</code></td>
<td>
<p>data.frame with meta data (<code>pid</code> (patient ID) and <code>date</code>) for
the file names contained within <code>mxrData</code></p>
</td></tr>
<tr><td><code id="processLastDose_+3A_labdata">labData</code></td>
<td>
<p>data.frame that contains lab dates and times associated with the file names
within <code>mxrData</code>. Must contain columns <code>pid</code> and <code>date</code>, as well as <code>labtime</code>.
The <code>date</code> column must be in the same format as <code>date</code> in <code>noteMetaData</code>, and
<code>labtime</code> must be a POSIXct</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Extract-Med and Pro-Med-NLP for details.
</p>


<h3>Value</h3>

<p>data.frame with identifying information (e.g., filename, etc) as well as processed
and standardized last dose times as a POSIXct column
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tac_mxr &lt;- read.csv(system.file("examples", "tac_mxr.csv", package = "EHR"))
data(tac_metadata)
data(tac_lab)

processLastDose(mxrData = tac_mxr, noteMetaData = tac_metadata, labData = tac_lab)

</code></pre>

<hr>
<h2 id='pullFakeId'>Pull Fake/Mod ID</h2><span id='topic+pullFakeId'></span>

<h3>Description</h3>

<p>Replace IDs with de-identified version pulled from a crosswalk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pullFakeId(
  dat,
  xwalk,
  firstCols = NULL,
  orderBy = NULL,
  uniq.id = "subject_uid"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pullFakeId_+3A_dat">dat</code></td>
<td>
<p>a data.frame</p>
</td></tr>
<tr><td><code id="pullFakeId_+3A_xwalk">xwalk</code></td>
<td>
<p>a data.frame providing linkage for each ID, e.g. output from <code><a href="#topic+idCrosswalk">idCrosswalk</a></code></p>
</td></tr>
<tr><td><code id="pullFakeId_+3A_firstcols">firstCols</code></td>
<td>
<p>name of columns to put at front of output data set</p>
</td></tr>
<tr><td><code id="pullFakeId_+3A_orderby">orderBy</code></td>
<td>
<p>name of columns used to reorder output data set</p>
</td></tr>
<tr><td><code id="pullFakeId_+3A_uniq.id">uniq.id</code></td>
<td>
<p>character string indicating subject-level id variable (default is &quot;subject_uid&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_data &lt;- data.frame(subj_id=c(4.1,4.2,5.1,6.1),
                        pat_id=c(14872,14872,24308,37143),
                        gender=c(1,1,0,1),
                        weight=c(34,42,28,63),
                        height=c(142,148,120,167))

# crosswalk w/ same format as idCrosswalk() output
xwalk &lt;- data.frame(subj_id=c(4.1,4.2,5.1,6.1),
                    pat_id=c(14872,14872,24308,37143),
                    mod_visit=c(1,2,1,1),
                    mod_id=c(1,1,2,3),
                    mod_id_visit=c(1.1,1.2,2.1,3.1))

demo_data_deident &lt;- pullFakeId(demo_data, xwalk, 
                                firstCols = c('mod_id','mod_id_visit','mod_visit'),
                                uniq.id='pat_id')

</code></pre>

<hr>
<h2 id='pullRealId'>Pull Real ID</h2><span id='topic+pullRealId'></span>

<h3>Description</h3>

<p>Replace de-identified IDs with identified version pulled from a crosswalk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pullRealId(dat, xwalk = NULL, remove.mod.id = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pullRealId_+3A_dat">dat</code></td>
<td>
<p>a data.frame</p>
</td></tr>
<tr><td><code id="pullRealId_+3A_xwalk">xwalk</code></td>
<td>
<p>a data.frame providing linkage for each ID, e.g. output from <code><a href="#topic+idCrosswalk">idCrosswalk</a></code>; 
if NULL, the crosswalk will be pulled from the &lsquo;pkxwalk&rsquo; option, or otherwise the
unmodified data.frame.</p>
</td></tr>
<tr><td><code id="pullRealId_+3A_remove.mod.id">remove.mod.id</code></td>
<td>
<p>logical, should the de-identified IDs &ndash; mod_id, mod_visit, mod_id_visit &ndash; 
be removed (default=FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_data_deident &lt;- data.frame(mod_id=c(1,1,2,3),
                                mod_id_visit=c(1.1,1.2,2.1,3.1),
                                mod_visit=c(1,2,1,1),
                                gender=c(1,1,0,1),
                                weight=c(34,42,28,63),
                                height=c(142,148,120,167))

# crosswalk w/ same format as idCrosswalk() output
xwalk &lt;- data.frame(subj_id=c(4.1,4.2,5.1,6.1),
                    pat_id=c(14872,14872,24308,37143),
                    mod_visit=c(1,2,1,1),
                    mod_id=c(1,1,2,3),
                    mod_id_visit=c(1.1,1.2,2.1,3.1))

pullRealId(demo_data_deident, xwalk)
pullRealId(demo_data_deident, xwalk, remove.mod.id=TRUE)

</code></pre>

<hr>
<h2 id='readTransform'>Read and Transform</h2><span id='topic+readTransform'></span>

<h3>Description</h3>

<p>Convenience function for reading in a CSV file, and making small
modifications to a data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readTransform(file, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="readTransform_+3A_file">file</code></td>
<td>
<p>filename of a CSV file</p>
</td></tr>
<tr><td><code id="readTransform_+3A_...">...</code></td>
<td>
<p>additional information passed to <code><a href="#topic+dataTransformation">dataTransformation</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code><a href="utils.html#topic+read.csv">read.csv</a></code> needs additional arguments (or the file is in a
different format), the user should load the data first, then directly call
<code><a href="#topic+dataTransformation">dataTransformation</a></code>.
</p>


<h3>Value</h3>

<p>The modified data.frame
</p>

<hr>
<h2 id='run_Build_PK_IV'>Build-PK-IV Module</h2><span id='topic+run_Build_PK_IV'></span>

<h3>Description</h3>

<p>This module builds PK data for intravenously (IV) administered medications.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_Build_PK_IV(
  conc,
  conc.columns = list(),
  dose,
  dose.columns = list(),
  censor = NULL,
  censor.columns = list(),
  demo.list = NULL,
  demo.columns = list(),
  lab.list = NULL,
  lab.columns = list(),
  dosePriorWindow = 7,
  labPriorWindow = 7,
  postWindow = NA,
  pk.vars = NULL,
  drugname = NULL,
  check.path = NULL,
  missdemo_fn = "-missing-demo",
  faildupbol_fn = "DuplicateBolus-",
  date.format = "%m/%d/%y %H:%M:%S",
  date.tz = "America/Chicago",
  isStrict = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_Build_PK_IV_+3A_conc">conc</code></td>
<td>
<p>concentration data, the output of <code><a href="#topic+run_DrugLevel">run_DrugLevel</a></code>, 
a filename (CSV, RData, RDS), or a correctly formatted data.frame</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_conc.columns">conc.columns</code></td>
<td>
<p>a named list that should specify columns in concentration
data; &lsquo;id&rsquo;, &lsquo;datetime&rsquo;, &lsquo;druglevel&rsquo; are required.
&lsquo;idvisit&rsquo; may also be specified; &lsquo;idvisit&rsquo; can be used when there are multiple visits 
(i.e., several occasions) for the same subject. &lsquo;datetime&rsquo; is date and time for
concentration measurement, which can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time
separately (e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_dose">dose</code></td>
<td>
<p>dose data, the output of <code><a href="#topic+run_MedStrI">run_MedStrI</a></code>, a filename
(CSV, RData, RDS), or a correctly formatted data.frame</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_dose.columns">dose.columns</code></td>
<td>
<p>a named list that should specify columns in dose data;
&lsquo;id&rsquo; is required. &lsquo;infuseDatetime&rsquo; and &lsquo;infuseDose&rsquo;
should be set if infusion dose data is present. &lsquo;infuseTimeExact&rsquo;
may also be specified for infusion data &ndash; this variable represents an
precise time, if for example the &lsquo;infuseDatetime&rsquo; variable is rounded.
&lsquo;bolusDatetime&rsquo; and &lsquo;bolusDose&rsquo; should be set if bolus dose
data is present. A generic &lsquo;date&rsquo; variable may be provided, agnostic
to either infusion or bolus dosing. &lsquo;gap&rsquo; and &lsquo;weight&rsquo; column
names may also be set. Any of the date-time variables can be specified as a
single date-time variable (infuseDatetime = &lsquo;date_time&rsquo;) or two variables
holding date and time separately (e.g., infuseDatetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_censor">censor</code></td>
<td>
<p>censoring information, if available; this will censor concentration
and dose data for dates occuring after the censor datetime variable.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_censor.columns">censor.columns</code></td>
<td>
<p>a named list that should specify columns in censoring data; &lsquo;id&rsquo;,
and &lsquo;datetime&rsquo; are required. &lsquo;datetime&rsquo; is the date and time when
data should be censored. This can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time separately
(e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_demo.list">demo.list</code></td>
<td>
<p>demographic information, if available; the output from 
<code><a href="#topic+run_Demo">run_Demo</a></code> or a correctly formatted data.frame</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_demo.columns">demo.columns</code></td>
<td>
<p>a named list that should specify columns in demographic data;
&lsquo;id&rsquo; is required. &lsquo;weight&rsquo; and &lsquo;idvisit&rsquo;
may also be used to specify columns for weight or the unique idvisit. Any other columns
present in the demographic data are treated as covariates.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_lab.list">lab.list</code></td>
<td>
<p>lab data, if available; the output from <code><a href="#topic+run_Labs">run_Labs</a></code> or 
a correctly formatted list</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_lab.columns">lab.columns</code></td>
<td>
<p>a named list that should specify columns in lab data; &lsquo;id&rsquo;,
and &lsquo;datetime&rsquo; are required. &lsquo;datetime&rsquo; is the date and time when
the lab data was obtained, which can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time separately
(e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)). Any other columns present in lab
data are treated as lab values.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_dosepriorwindow">dosePriorWindow</code></td>
<td>
<p>Dose data is merged with drug level data. This value sets the
time frame window with the number of days prior to the first drug level data; defaults to 7.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_labpriorwindow">labPriorWindow</code></td>
<td>
<p>Lab data is merged with drug level data. This value sets the
time frame window with the number of days prior to the first drug level data; defaults to 7.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_postwindow">postWindow</code></td>
<td>
<p>Data is merged with drug level data. This postWindow can set the end time
for the drug level data, being the number of days after the first drug level data. The
default (NA) will use the date of the last drug level data.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_pk.vars">pk.vars</code></td>
<td>
<p>variables to include in the returned PK data. The variable &lsquo;date&rsquo;
is a special case; when included, it maps the &lsquo;time&rsquo; offset to its original date-time.
Other named variables will be merged from the concentration data set. For example,
rather than being separate data sets, labs or demographics may already be present in
the concentration data. These columns should be named here.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_drugname">drugname</code></td>
<td>
<p>drug of interest, included in filename of check files. The default (NULL)
will produce filenames without drugname included.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_check.path">check.path</code></td>
<td>
<p>path to &lsquo;check&rsquo; directory, where check files are
created. The default (NULL) will not produce any check files.</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_missdemo_fn">missdemo_fn</code></td>
<td>
<p>filename for checking NA frequency among demographic data</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_faildupbol_fn">faildupbol_fn</code></td>
<td>
<p>filename for duplicate bolus data</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_date.format">date.format</code></td>
<td>
<p>output format for &lsquo;date&rsquo; variable</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_date.tz">date.tz</code></td>
<td>
<p>output time zone for &lsquo;date&rsquo; variable</p>
</td></tr>
<tr><td><code id="run_Build_PK_IV_+3A_isstrict">isStrict</code></td>
<td>
<p>logical; when TRUE dose amount totals are strictly summed rather than repeated
hourly until stopped</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>
<p>Regarding the &lsquo;gap&rsquo; variable in the dose dataset, if &lsquo;gap&rsquo; is specified in &lsquo;dose.columns&rsquo;, 
it allows a continuous infusion given when there are missing records 
between infusion dosing records. For example, suppose that &lsquo;gap&rsquo; = 60 is defined 
(which is typical gap size when infusion dosing is supposed to be recorded hourly for inpatients) 
and time between two records (i.e., gap) are greater than 1 hour (i.e., missing records). If the gap 
between the two records is less or equal to twice of the gap (i.e., 2*60 = 120 min), a continuous infusion 
is assumed until the 2nd dose record; otherwise, the first infusion is assumed to be stopped 
(i.e., add zero doses) after 60 min (i.e., equal to the gap size)  and a new infusion (the 2nd record) starts at its recorded time.
</p>


<h3>Value</h3>

<p>PK data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># make fake data
set.seed(6543)

build_date &lt;- function(x) format(seq(x, length.out=5, by="1 hour"), "%Y-%m-%d %H:%M")
dates &lt;- unlist(lapply(rep(Sys.time(),3), build_date))

plconc &lt;- data.frame(mod_id = rep(1:3,each=5),
                   mod_id_visit = rep(1:3,each=5)+0.1,
                   event = rep(1:5,times=3),
                   conc.level = 15*exp(-1*rep(1:5,times=3))+rnorm(15,0,0.1),
                   date.time = as.POSIXct(dates))

ivdose &lt;- data.frame(mod_id = 1:3,
                     date.dose = substr(dates[seq(1,15,by=5)],1,10),
                     infuse.time.real = NA, infuse.time = NA, infuse.dose = NA,
                     bolus.time = as.POSIXct(dates[seq(1,15,by=5)])-300,
                     bolus.dose = 90,
                     maxint = 0L,
                     weight = 45)


run_Build_PK_IV(conc = plconc,
                conc.columns = list(id = 'mod_id', datetime = 'date.time',
                  druglevel = 'conc.level', idvisit = 'mod_id_visit'),
                dose = ivdose,
                dose.columns = list(id = 'mod_id', date = 'date.dose',
                  bolusDatetime = 'bolus.time', bolusDose = 'bolus.dose',
                  gap = 'maxint', weight = 'weight'),
                pk.vars = 'date')

</code></pre>

<hr>
<h2 id='run_Build_PK_Oral'>Build-PK-Oral Module</h2><span id='topic+run_Build_PK_Oral'></span>

<h3>Description</h3>

<p>This module builds PK data for orally administered medications.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_Build_PK_Oral(
  x,
  idCol = "id",
  dtCol = "dt",
  doseCol = "dose",
  concCol = "conc",
  ldCol = NULL,
  first_interval_hours = 336,
  imputeClosest = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_Build_PK_Oral_+3A_x">x</code></td>
<td>
<p>a data.frame or file saved as either CSV, RData, or RDS</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_idcol">idCol</code></td>
<td>
<p>data.frame id column name</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_dtcol">dtCol</code></td>
<td>
<p>data.frame date column name</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_dosecol">doseCol</code></td>
<td>
<p>dose column name</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_conccol">concCol</code></td>
<td>
<p>concentration column name</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_ldcol">ldCol</code></td>
<td>
<p>last-dose time column name</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_first_interval_hours">first_interval_hours</code></td>
<td>
<p>number of hours before the first concentration to start time=0; the default is 336 hours = 14 days</p>
</td></tr>
<tr><td><code id="run_Build_PK_Oral_+3A_imputeclosest">imputeClosest</code></td>
<td>
<p>columns to impute missing data with next observation propagated backward; this is in addition to
all covariates receving imputation using last observation carried forward</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Build-PK-Oral.
</p>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Data Generating Function
mkdat &lt;- function() {
  npat &lt;- 3
  visits &lt;- floor(runif(npat, min=2, max=6))
  id &lt;- rep(1:npat, visits)
  dt_samp &lt;- as.Date(sort(sample(700, sum(visits))), origin = '2019-01-01')
  tm_samp &lt;- as.POSIXct(paste(dt_samp, '10:00:00'), tz = 'UTC')
  dt &lt;- tm_samp + rnorm(sum(visits), 0, 1*60*60)
  dose_morn &lt;- sample(c(2.5,5,7.5,10), sum(visits), replace = TRUE)
  conc &lt;- round(rnorm(sum(visits), 1.5*dose_morn, 1),1)
  ld &lt;- dt - sample(10:16, sum(visits), replace = TRUE) * 3600
  ld[rnorm(sum(visits)) &lt; .3] &lt;- NA
  age &lt;- rep(sample(40:75, npat), visits)
  gender &lt;- rep(sample(0:1, npat, replace=TRUE), visits)
  weight &lt;- rep(round(rnorm(npat, 180, 20)),visits)
  hgb &lt;- rep(rnorm(npat, 10, 2), visits)
  data.frame(id, dt, dose_morn, conc, ld, age, gender, weight, hgb)
}

# Make raw data
set.seed(30)
dat &lt;- mkdat()

#Process data without last-dose times
run_Build_PK_Oral(x = dat,
                  idCol = "id",
                  dtCol = "dt",
                  doseCol = "dose_morn",
                  concCol = "conc",
                  ldCol = NULL,
                  first_interval_hours = 336,
                  imputeClosest = NULL)

#Process data with last-dose times
run_Build_PK_Oral(x = dat, doseCol = "dose_morn", ldCol = "ld")

</code></pre>

<hr>
<h2 id='run_Demo'>Run Demographic Data</h2><span id='topic+run_Demo'></span>

<h3>Description</h3>

<p>This module will load and modify demographic data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_Demo(demo.path, demo.columns = list(), toexclude, demo.mod.list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_Demo_+3A_demo.path">demo.path</code></td>
<td>
<p>filename of demographic file (CSV, RData, RDS) or data.frame</p>
</td></tr>
<tr><td><code id="run_Demo_+3A_demo.columns">demo.columns</code></td>
<td>
<p>a named list that should specify columns in demo data; &lsquo;id&rsquo;,
is required.</p>
</td></tr>
<tr><td><code id="run_Demo_+3A_toexclude">toexclude</code></td>
<td>
<p>expression that should evaluate to a logical, indicating if
the observation should be excluded</p>
</td></tr>
<tr><td><code id="run_Demo_+3A_demo.mod.list">demo.mod.list</code></td>
<td>
<p>list of expressions, giving modifications to make</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>


<h3>Value</h3>

<p>list with two components
</p>
<table role = "presentation">
<tr><td><code>demo</code></td>
<td>
<p>demographic data</p>
</td></tr>
<tr><td><code>exclude</code></td>
<td>
<p>vector of excluded visit IDs</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2525)
dateSeq &lt;- seq(as.Date('2019/01/01'), as.Date('2020/01/01'), by="day")
demo &lt;- data.frame(mod_id_visit = 1:10,
                   weight.lbs = rnorm(10,160,20),
                   age = rnorm(10, 50, 10),
                   enroll.date = sample(dateSeq, 10))
tmpfile &lt;- paste0(tempfile(), '.rds')
saveRDS(demo, file = tmpfile)

# exclusion functions
exclude_wt &lt;- function(x) x &lt; 150
exclude_age &lt;- function(x) x &gt; 60
ind.risk &lt;- function(wt, age) wt&gt;170 &amp; age&gt;55
exclude_enroll &lt;- function(x) x &lt; as.Date('2019/04/01')

# make demographic data that:
# (1) excludes ids with weight.lbs &lt; 150, age &gt; 60, or enroll.date before 2019/04/01
# (2) creates new 'highrisk' variable for subjects with weight.lbs&gt;170 and age&gt;55
out &lt;- run_Demo(demo.path = tmpfile, demo.columns = list(id = 'mod_id_visit'),
               toexclude = expression(
                 exclude_wt(weight.lbs)|exclude_age(age)|exclude_enroll(enroll.date)
               ),
               demo.mod.list = list(highrisk = expression(ind.risk(weight.lbs, age))))

out


</code></pre>

<hr>
<h2 id='run_DrugLevel'>Run Drug Level Data</h2><span id='topic+run_DrugLevel'></span>

<h3>Description</h3>

<p>This module will load and modify drug-level data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_DrugLevel(
  conc.path,
  conc.columns = list(),
  conc.select,
  conc.rename,
  conc.mod.list = NULL,
  samp.path = NULL,
  samp.columns = list(),
  samp.mod.list = NULL,
  check.path = NULL,
  failmiss_fn = "MissingConcDate-",
  multsets_fn = "multipleSetsConc-",
  faildup_fn = "DuplicateConc-",
  drugname = NULL,
  LLOQ = NA,
  demo.list = NULL,
  demo.columns = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_DrugLevel_+3A_conc.path">conc.path</code></td>
<td>
<p>filename of concentration data (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_conc.columns">conc.columns</code></td>
<td>
<p>a named list that should specify columns in concentration data.
&lsquo;id&rsquo; and &lsquo;conc&rsquo; are required. &lsquo;idvisit&rsquo; may also be specified.
If linking with sampling data, &lsquo;samplinkid&rsquo; is required. Otherwise
&lsquo;datetime&rsquo; is required. This is the date and time when blood samples were
obtained. This can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time
separately (e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_conc.select">conc.select</code></td>
<td>
<p>columns to select from concentration data</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_conc.rename">conc.rename</code></td>
<td>
<p>new column names for concentration data</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_conc.mod.list">conc.mod.list</code></td>
<td>
<p>list of expressions, giving modifications to make</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_samp.path">samp.path</code></td>
<td>
<p>filename of data with sampling time (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_samp.columns">samp.columns</code></td>
<td>
<p>a named list that should specify columns in sampling data.
&lsquo;conclinkid&rsquo; and &lsquo;datetime&rsquo; are required to link sampling data to
concentration data. &lsquo;conclinkid&rsquo; should match the id variable provided as
&lsquo;samplinkid&rsquo; in the &lsquo;conc.columns&rsquo; argument. &lsquo;datetime&rsquo; is the
date and time when blood samples were obtained. This can refer to a single date-time
variable (datetime = &lsquo;date_time&rsquo;) or two variables holding date and time
separately (e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_samp.mod.list">samp.mod.list</code></td>
<td>
<p>list of expressions, giving modifications to make</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_check.path">check.path</code></td>
<td>
<p>path to &lsquo;check&rsquo; directory, where check files are
created. The default (NULL) will not produce any check files.</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_failmiss_fn">failmiss_fn</code></td>
<td>
<p>filename for data missing concentration date</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_multsets_fn">multsets_fn</code></td>
<td>
<p>filename for data with multiple concentration sets</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_faildup_fn">faildup_fn</code></td>
<td>
<p>filename for data with duplicate concentration observations</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_drugname">drugname</code></td>
<td>
<p>drug of interest, included in filename of check files. The default (NULL)
will produce filenames without drugname included.</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_lloq">LLOQ</code></td>
<td>
<p>lower limit of concentration values; values below this are invalid</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_demo.list">demo.list</code></td>
<td>
<p>demographic information; if available, concentration records
must have a valid demo record</p>
</td></tr>
<tr><td><code id="run_DrugLevel_+3A_demo.columns">demo.columns</code></td>
<td>
<p>a named list that should specify columns in demographic data; &lsquo;id&rsquo;,
is required. If &lsquo;idvisit&rsquo; is present in the concentration data, then it is required
here too.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>


<h3>Value</h3>

<p>drug-level data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># concentrations
conc_data &lt;- data.frame(mod_id = rep(1:3,each=4),
                       mod_visit = rep(c(2,1,1),each=4),
                       mod_id_visit = as.numeric(paste(rep(1:3,each=4),
                                                      rep(c(2,1,1),each=4), sep=".")),
                       samp = rep(1:4,times=3),
                       drug_calc_conc=15*exp(-1*rep(1:4,times=3))+rnorm(12,0,0.1))

# sample times
build_date &lt;- function(x) format(seq(x, length.out=4, by="1 hour"), "%Y-%m-%d %H:%M")
dates &lt;- unlist(lapply(rep(Sys.time(),3), build_date))

samp_data &lt;- data.frame(mod_id = rep(1:3,each=4),
                       mod_visit = rep(c(2,1,1),each=4),
                       mod_id_visit = as.numeric(paste(rep(1:3,each=4),
                                                       rep(c(2,1,1),each=4), sep=".")),
                       samp = rep(1:4,times=3),
                       Sample.Collection.Date.and.Time = dates)

run_DrugLevel(
  conc.path = conc_data,
  conc.columns = list(
    id = 'mod_id', idvisit = 'mod_id_visit', samplinkid = 'mod_id_event', conc = 'conc.level'
  ),
  conc.select = c('mod_id','mod_id_visit','samp','drug_calc_conc'),
  conc.rename = c(drug_calc_conc= 'conc.level', samp='event'),
  conc.mod.list = list(mod_id_event = expression(paste(mod_id_visit, event, sep = "_"))),
  samp.path = samp_data,
  samp.columns = list(conclinkid = 'mod_id_event', datetime = 'Sample.Collection.Date.and.Time'),
  samp.mod.list = list(mod_id_event = expression(paste(mod_id_visit, samp, sep = "_"))),
  drugname = 'drugnm',
  LLOQ = 0.05
)

# minimal example with data in required format
conc_data &lt;- conc_data[,c('mod_id','mod_id_visit','samp','drug_calc_conc')]
conc_data[,'mod_id_event'] &lt;- paste(conc_data[,'mod_id_visit'], conc_data[,'samp'], sep = "_")
names(conc_data)[3:4] &lt;- c('event','conc.level')
samp_data[,'mod_id_event'] &lt;- paste(samp_data[,'mod_id_visit'], samp_data[,'samp'], sep = "_")
conc_samp_link &lt;- match(conc_data[,'mod_id_event'], samp_data[,'mod_id_event'])
conc_date &lt;- samp_data[conc_samp_link, 'Sample.Collection.Date.and.Time']
conc_data[,'date.time'] &lt;- as.POSIXct(conc_date)
run_DrugLevel(conc_data, conc.columns = list(
  id = 'mod_id', idvisit = 'mod_id_visit', datetime = 'date.time', conc = 'conc.level'
))

</code></pre>

<hr>
<h2 id='run_Labs'>Run Lab Data</h2><span id='topic+run_Labs'></span>

<h3>Description</h3>

<p>This module will load and modify laboratory data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_Labs(lab.path, lab.select, lab.mod.list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_Labs_+3A_lab.path">lab.path</code></td>
<td>
<p>filename of a lab file (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_Labs_+3A_lab.select">lab.select</code></td>
<td>
<p>columns to select</p>
</td></tr>
<tr><td><code id="run_Labs_+3A_lab.mod.list">lab.mod.list</code></td>
<td>
<p>list of expressions giving modifications to make;
passed to <code><a href="#topic+dataTransformation">dataTransformation</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>


<h3>Value</h3>

<p>lab data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lab_data &lt;- data.frame(mod_id=rep(1:3,each=3),
                       date=rep(c("01/12/17","05/05/18","11/28/16"),each=3),
                       time=rep(c("1:30","2:30","3:30"),3),
                       creat=rnorm(9,0.5,0.05))
run_Labs(lab_data, lab.mod.list=list(log_creat=expression(log(creat))))

</code></pre>

<hr>
<h2 id='run_MedStrI'>Run Str Data I</h2><span id='topic+run_MedStrI'></span>

<h3>Description</h3>

<p>This module will load and modify structured intravenous (IV) infusion and 
bolus medication data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_MedStrI(
  mar.path,
  mar.columns = list(),
  medGivenReq = FALSE,
  flow.path = NULL,
  flow.columns = list(),
  medchk.path = NULL,
  demo.list = NULL,
  demo.columns = list(),
  missing.wgt.path = NULL,
  wgt.columns = list(),
  check.path = NULL,
  failflow_fn = "FailFlow",
  failnounit_fn = "NoUnit",
  failunit_fn = "Unit",
  failnowgt_fn = "NoWgt",
  censor_date_fn = "CensorTime",
  infusion.unit = "mcg/kg/hr",
  bolus.unit = "mcg",
  bol.rate.thresh = Inf,
  rateunit = "mcg/hr",
  ratewgtunit = "mcg/kg/hr",
  weightunit = "kg",
  drugname = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_MedStrI_+3A_mar.path">mar.path</code></td>
<td>
<p>filename of MAR data (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_mar.columns">mar.columns</code></td>
<td>
<p>a named list that should specify columns in MAR data; &lsquo;id&rsquo;,
&lsquo;datetime&rsquo; and &lsquo;dose&rsquo; are required. &lsquo;drug&rsquo;, &lsquo;weight&rsquo;,
&lsquo;given&rsquo; may also be specified. &lsquo;datetime&rsquo; is date and time for data
measurement, which can refer to a single date-time variable (datetime = &lsquo;date_time&rsquo;)
or two variables holding date and time separately (e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).
&lsquo;dose&rsquo; can also be given as a single variable or two variables. If given as a single column,
the column's values should contain dose and units such as &lsquo;25 mcg&rsquo;. If given as
two column names, the dose column should come before the unit column
(e.g., dose = c(&lsquo;doseamt&rsquo;, &lsquo;unit&rsquo;)). &lsquo;drug&rsquo; can provide list of acceptable drug names.
If &lsquo;drug&rsquo; is present, the &lsquo;medchk.path&rsquo; argument should also be provided.
The &lsquo;given&rsquo; is a variable that flags whether the medication (inpatient) was given. When it is given, 
values shoule be &ldquo;Given&rdquo;; should be used in conjunction with the &lsquo;medGivenReq&rsquo; argument.</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_medgivenreq">medGivenReq</code></td>
<td>
<p>if TRUE, values in &lsquo;given&rsquo; column in MAR data should equal &ldquo;Given&rdquo;; 
if this is FALSE (the default), NA values are also acceptable.</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_flow.path">flow.path</code></td>
<td>
<p>filename of flow data (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_flow.columns">flow.columns</code></td>
<td>
<p>a named list that should specify columns in flow data; &lsquo;id&rsquo;,
&lsquo;datetime&rsquo;, &lsquo;finalunits&rsquo;, &lsquo;unit&rsquo;, &lsquo;rate&rsquo;, &lsquo;weight&rsquo;
are required. &lsquo;idvisit&rsquo; may also be specified. &lsquo;datetime&rsquo; is date and time
for data measurement, which can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time separately
(e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_medchk.path">medchk.path</code></td>
<td>
<p>filename containing data set (CSV, RData, RDS), or data.frame; should have
the column &lsquo;medname&rsquo; with list of acceptable drug names (e.g., brand and generic name, abbreviations)
to subset drugs of interest using &lsquo;drug&rsquo; column in MAR data. This argument can be used when MAR data 
contains different drugs that should be excluded.</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_demo.list">demo.list</code></td>
<td>
<p>demographic information; if available, the output from 'run_Demo' or 
a correctly formatted data.frame, which can be used to impute weight when missing</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_demo.columns">demo.columns</code></td>
<td>
<p>a named list that should specify columns in demographic data; &lsquo;id&rsquo;,
&lsquo;datetime&rsquo;, and &lsquo;weight&rsquo; are required. &lsquo;datetime&rsquo; is the date
and time when the demographic data were obtained, which can refer to a single date-time
variable (datetime = &lsquo;date_time&rsquo;) or two variables holding date and time separately
(e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_missing.wgt.path">missing.wgt.path</code></td>
<td>
<p>filename containing additional weight data (CSV, RData, RDS), or data.frame. 
The variables in this file should be defined in the &lsquo;wgt.columns&rsquo; argument.</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_wgt.columns">wgt.columns</code></td>
<td>
<p>a named list that should specify columns in additional weight data; &lsquo;id&rsquo;,
&lsquo;datetime&rsquo;, and &lsquo;weight&rsquo; are required. &lsquo;datetime&rsquo; is date
and time for weight measurement, which can refer to a single date-time variable
(datetime = &lsquo;date_time&rsquo;) or two variables holding date and time separately
(e.g., datetime = c(&lsquo;Date&rsquo;, &lsquo;Time&rsquo;)).</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_check.path">check.path</code></td>
<td>
<p>path to &lsquo;check&rsquo; directory, where check files are
created. The default (NULL) will not produce any check files.</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_failflow_fn">failflow_fn</code></td>
<td>
<p>filename for duplicate flow data with rate zero</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_failnounit_fn">failnounit_fn</code></td>
<td>
<p>filename for MAR data with missing unit</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_failunit_fn">failunit_fn</code></td>
<td>
<p>filename for MAR data with invalid unit</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_failnowgt_fn">failnowgt_fn</code></td>
<td>
<p>filename for infusion data with missing weight where unit
indicates weight is required</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_censor_date_fn">censor_date_fn</code></td>
<td>
<p>filename containing censor times created with invalid dose data</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_infusion.unit">infusion.unit</code></td>
<td>
<p>acceptable unit for infusion data</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_bolus.unit">bolus.unit</code></td>
<td>
<p>acceptable unit for bolus data</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_bol.rate.thresh">bol.rate.thresh</code></td>
<td>
<p>upper limit for bolus rate; values above this are invalid</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_rateunit">rateunit</code></td>
<td>
<p>acceptable unit for hourly rate; defaults to &lsquo;mcg/hr&rsquo;</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_ratewgtunit">ratewgtunit</code></td>
<td>
<p>acceptable unit for hourly rate by weight; defaults to &lsquo;mcg/kg/hr&rsquo;</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_weightunit">weightunit</code></td>
<td>
<p>acceptable unit for weight; defaults to &lsquo;kg&rsquo;</p>
</td></tr>
<tr><td><code id="run_MedStrI_+3A_drugname">drugname</code></td>
<td>
<p>drug of interest, included in filename of check files. The default (NULL)
will produce filenames without drugname included.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>


<h3>Value</h3>

<p>structured data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'># flow data for 'Fakedrug1'
flow &lt;- data.frame(mod_id=c(1,1,2,2,2),
                   mod_id_visit=c(46723,46723,84935,84935,84935),
                   record.date=c("07/05/2019 5:25","07/05/2019 6:01",
                                 "09/04/2020 3:21", "09/04/2020 4:39",
                                 "09/04/2020 5:32"),
                   Final.Weight=c(6.75,6.75,4.5,4.5,4.5),
                   Final.Rate=c(rep("1 mcg/kg/hr",2),
                                rep("0.5 mcg/kg/hr",3)),
                   Final.Units=c("3.375","6.5",
                                 "2.25","2.25","2.25"))
flow[,'Perform.Date'] &lt;- pkdata::parse_dates(flow[,'record.date'])
flow[,'unit'] &lt;- sub('.*[ ]', '', flow[,'Final.Rate'])
flow[,'rate'] &lt;- as.numeric(sub('([0-9.]+).*', '\\1', flow[,'Final.Rate']))

# mar data for 4 fake drugs
mar &lt;- data.frame(mod_id=rep(1,5),
                  Date=rep("2019-07-05",5),
                  Time=c("07:12","07:31","08:47","09:16","10:22"),
                  `med:mDrug`=c("Fakedrug2","Fakedrug1","Fakedrug2",
                                "Fakedrug3","Fakedrug4"),
                  `med:dosage`=c("30 mg","0.5 mcg","1 mg",
                                 "20 mg","3 mcg/kg/min"),
                  `med:route`=rep("IV",5),
                  `med:given`=rep("Given",5),
                  check.names=FALSE)

# medcheck file for drug of interest ('Fakedrug1')
medcheck &lt;- data.frame(medname="Fakedrug1",freq=4672)

run_MedStrI(mar.path = mar,
            mar.columns = list(id = 'mod_id', datetime = c('Date','Time'),
                               dose = 'med:dosage', drug = 'med:mDrug', given = 'med:given'),
            flow.path = flow,
            flow.columns = list(id = 'mod_id', datetime = 'Perform.Date',
                                finalunits = 'Final.Units', unit = 'unit',
                                rate = 'rate', weight = 'Final.Weight'),
            medchk.path = medcheck,
            check.path = tempdir(),
            drugname = 'fakedrg1')

</code></pre>

<hr>
<h2 id='run_MedStrII'>Run Structured E-Prescription Data</h2><span id='topic+run_MedStrII'></span>

<h3>Description</h3>

<p>This module will load and modify structured e-prescription data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_MedStrII(file, dat.columns = list())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_MedStrII_+3A_file">file</code></td>
<td>
<p>filename of prescription data (CSV, RData, RDS), or data.frame</p>
</td></tr>
<tr><td><code id="run_MedStrII_+3A_dat.columns">dat.columns</code></td>
<td>
<p>a named list that should specify columns in data; &lsquo;id&rsquo;,
&lsquo;dose&rsquo;, &lsquo;freq&rsquo;, &lsquo;date&rsquo;, and &lsquo;str&rsquo; are required.
&lsquo;desc&rsquo; may also be specified.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See EHR Vignette for Structured Data.
</p>


<h3>Value</h3>

<p>str data set
</p>


<h3>Examples</h3>

<pre><code class='language-R'>erx_data &lt;- data.frame(GRID=paste0("ID",c(1,1,2,2,2,2)),
                       MED_NAME=c("fakedrug","fakedrug","fakedrug",
                                  "Brandname","fakedrug","fakedrug"),
                       RX_DOSE=c(1,2,1,'2 tabs',1,'1+1.5+1'),
                       FREQUENCY=c(rep("bid",3),"qam","bid",
                                   "brkfst,lunch,dinner"),
                       ENTRY_DATE=c("2018-02-15","2018-03-14","2017-07-01",
                                    "2017-07-01","2017-09-15","2017-11-01"),
                       STRENGTH_AMOUNT=c("100","100","200",
                                         "100mg","100","100"),
                       DESCRIPTION=c("fakedrug 100 mg tablet","fakedrug 100 mg tablet",
                                     "fakedrug 200 mg tablet (also known as brandname)",
                                     "Brandname 100mg tablet", "fakedrug 100 mg tablet",
                                     "fakedrug 100 mg tablet"))

run_MedStrII(erx_data, list(id = 'GRID', dose = 'RX_DOSE', freq = 'FREQUENCY',
             date = 'ENTRY_DATE', str = 'STRENGTH_AMOUNT', desc = 'DESCRIPTION'))

</code></pre>

<hr>
<h2 id='stdzDose'>Standardize Dose Entity</h2><span id='topic+stdzDose'></span>

<h3>Description</h3>

<p>This function standardizes the dose entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzDose(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzDose_+3A_x">x</code></td>
<td>
<p>character vector of extracted dose values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some dose strings may include multiple values and additional interpretation
may be needed. For example &lsquo;2-1&rsquo; likely indicates a dose of 2 followed
by a dose of 1. Currently it would be converted to the average of 1.5.
</p>


<h3>Value</h3>

<p>numeric vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzDose(c('one tablet', '1/2 pill', '1-3 tabs'))
</code></pre>

<hr>
<h2 id='stdzDoseChange'>Standardize Dose Change Entity</h2><span id='topic+stdzDoseChange'></span>

<h3>Description</h3>

<p>This function standardizes the dose change entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzDoseChange(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzDoseChange_+3A_x">x</code></td>
<td>
<p>character vector of extracted dose change values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzDoseChange(c('decreasing','dropped','increased'))
</code></pre>

<hr>
<h2 id='stdzDoseSchedule'>Standardize Dose Schedule Entity</h2><span id='topic+stdzDoseSchedule'></span>

<h3>Description</h3>

<p>This function standardizes the dose schedule entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzDoseSchedule(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzDoseSchedule_+3A_x">x</code></td>
<td>
<p>character vector of extracted dose schedule values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzDoseSchedule(c('tapered','weaned','TAPER'))
</code></pre>

<hr>
<h2 id='stdzDuration'>Standardize Duration Entity</h2><span id='topic+stdzDuration'></span>

<h3>Description</h3>

<p>This function standardizes the duration entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzDuration(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzDuration_+3A_x">x</code></td>
<td>
<p>character vector of extracted duration values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzDuration(c('1 month', 'three days', 'two-weeks'))
</code></pre>

<hr>
<h2 id='stdzFreq'>Standardize Frequency Entity</h2><span id='topic+stdzFreq'></span>

<h3>Description</h3>

<p>This function standardizes the frequency entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzFreq(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzFreq_+3A_x">x</code></td>
<td>
<p>character vector of extracted frequency values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzFreq(c('in the morning', 'four times a day', 'with meals'))
</code></pre>

<hr>
<h2 id='stdzRoute'>Standardize Route Entity</h2><span id='topic+stdzRoute'></span>

<h3>Description</h3>

<p>This function standardizes the route entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzRoute(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzRoute_+3A_x">x</code></td>
<td>
<p>character vector of extracted route values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzRoute(c('oral', 'po', 'subcut'))
</code></pre>

<hr>
<h2 id='stdzStrength'>Standardize Strength Entity</h2><span id='topic+stdzStrength'></span>

<h3>Description</h3>

<p>This function standardizes the strength entity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stdzStrength(str, freq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stdzStrength_+3A_str">str</code></td>
<td>
<p>character vector of extracted strength values</p>
</td></tr>
<tr><td><code id="stdzStrength_+3A_freq">freq</code></td>
<td>
<p>character vector of extracted frequency values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some strength strings may include multiple values and additional
interpretation may be needed. For example &lsquo;2-1&rsquo; likely indicates
a strength of 2 followed by a strength of 1. Thus a single element may need
to be standarized into two elements. This can only happen if the frequency
entity is missing or in agreement (&lsquo;bid&rsquo; for example). See the
&lsquo;addl_data&rsquo; attribute of the returned vector.
</p>


<h3>Value</h3>

<p>numeric vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stdzStrength(c('1.5', '1/2', '1/1/1'))
stdzStrength(c('1.5', '1/2', '1/1/1'), c('am', 'daily', NA))
stdzStrength(c('1.5', '1/2', '1/1/1'), FALSE)
</code></pre>

<hr>
<h2 id='tac_lab'>Example of Lab Time Data for Tacrolimus</h2><span id='topic+tac_lab'></span>

<h3>Description</h3>

<p>An example dataset used in <code><a href="#topic+processLastDose">processLastDose</a></code> that contains lab time data. This dataset should
have one row per patient ID-date pair, and contain the time a lab was performed as a datetime variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tac_lab, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 2 observations on the following variables.
</p>

<dl>
<dt>pid</dt><dd><p>A character vector, patient ID associated with the lab value</p>
</dd>
<dt>date</dt><dd><p>A character vector, date associated with the lab value</p>
</dd>
<dt>labtime</dt><dd><p>A POSIXct vector, datetime at which the lab was performed formatted as YYYY-MM-DD HH:MM:SS</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(tac_lab)
</code></pre>

<hr>
<h2 id='tac_metadata'>Example of Metadata for Tacrolimus Data</h2><span id='topic+tac_metadata'></span>

<h3>Description</h3>

<p>An example of the metadata needed for the <code><a href="#topic+processLastDose">processLastDose</a></code>,
<code><a href="#topic+makeDose">makeDose</a></code>, and <code><a href="#topic+collapseDose">collapseDose</a></code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tac_metadata, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 observations on the following variables.
</p>

<dl>
<dt>filename</dt><dd><p>A character vector, filename for the clinical note</p>
</dd>
<dt>pid</dt><dd><p>A character vector, patient ID associated with the filename</p>
</dd>
<dt>date</dt><dd><p>A character vector, date associated with the filename</p>
</dd>
<dt>note</dt><dd><p>A character vector, note ID associated with the filename</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(tac_metadata)
</code></pre>

<hr>
<h2 id='tac_mxr_parsed'>Example of Tacrolimus Output from 'parseMedExtractR'</h2><span id='topic+tac_mxr_parsed'></span>

<h3>Description</h3>

<p>The output after running <code><a href="#topic+parseMedExtractR">parseMedExtractR</a></code> on 3 example clinical notes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tac_mxr_parsed, package = 'EHR')
</code></pre>


<h3>Format</h3>

<p>A data frame with 7 observations on the following variables.
</p>

<dl>
<dt>filename</dt><dd><p>A character vector, filename for the clinical note</p>
</dd>
<dt>drugname</dt><dd><p>A character vector, drug name extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>strength</dt><dd><p>A character vector, strengths extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dose</dt><dd><p>A character vector, dose amounts extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>route</dt><dd><p>A character vector, routes extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>freq</dt><dd><p>A character vector, frequencies extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dosestr</dt><dd><p>A character vector, dose intakes extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>dosechange</dt><dd><p>A character vector, dose change keywords extracted from the clinical note along with start and stop positions</p>
</dd>
<dt>lastdose</dt><dd><p>A character vector, last dose times extracted from the clinical note along with start and stop positions</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(tac_mxr_parsed)
</code></pre>

<hr>
<h2 id='writeCheckData'>Write Check File as CSV</h2><span id='topic+writeCheckData'></span>

<h3>Description</h3>

<p>A check file contains IDs and data that need to be manually verified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeCheckData(dat, filename, msg, decor = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeCheckData_+3A_dat">dat</code></td>
<td>
<p>a data.frame</p>
</td></tr>
<tr><td><code id="writeCheckData_+3A_filename">filename</code></td>
<td>
<p>name of output file</p>
</td></tr>
<tr><td><code id="writeCheckData_+3A_msg">msg</code></td>
<td>
<p>character text to display</p>
</td></tr>
<tr><td><code id="writeCheckData_+3A_decor">decor</code></td>
<td>
<p>logical value to indicate decoration before and after the msg</p>
</td></tr>
<tr><td><code id="writeCheckData_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to <code><a href="utils.html#topic+write.csv">write.csv</a></code></p>
</td></tr>
</table>

<hr>
<h2 id='zeroOneTable'>Make Zero One Contingency Tables</h2><span id='topic+zeroOneTable'></span>

<h3>Description</h3>

<p>Make contingency tables for many binary outcomes and a binary covariate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zeroOneTable(EXPOSURE, phenotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="zeroOneTable_+3A_exposure">EXPOSURE</code></td>
<td>
<p>binary covariate (e.g., exposure).</p>
</td></tr>
<tr><td><code id="zeroOneTable_+3A_phenotype">phenotype</code></td>
<td>
<p>binary outcome (e.g., phenotype).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generates frequency and contingency tables for many binary outcomes (e.g.,
large number of phenotypes) and a binary covariate (e.g., drug exposure,
genotypes) more efficiently.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>t00</code></td>
<td>
<p>frequency for non-exposed group and non-case outcome.</p>
</td></tr>
<tr><td><code>t01</code></td>
<td>
<p>frequency for non-exposed group and case outcome.</p>
</td></tr>
<tr><td><code>t10</code></td>
<td>
<p>frequency for exposed group and non-case outcome.</p>
</td></tr>
<tr><td><code>t11</code></td>
<td>
<p>frequency for exposed group and case outcome.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Leena Choi <a href="mailto:leena.choi@vanderbilt.edu">leena.choi@vanderbilt.edu</a> and
Cole Beck <a href="mailto:cole.beck@vumc.org">cole.beck@vumc.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## full example data
data(dataPheWAS)
demo.covariates &lt;- c('id','exposure','age','race','gender')
phenotypeList &lt;- setdiff(colnames(dd), demo.covariates)
tablePhenotype &lt;- matrix(NA, ncol=4, nrow=length(phenotypeList), 
dimnames=list(phenotypeList, c("n.nocase.nonexp", "n.case.nonexp", 
"n.nocase.exp", "n.case.exp")))
for(i in seq_along(phenotypeList)) {
    tablePhenotype[i, ] &lt;- zeroOneTable(dd[, 'exposure'], dd[, phenotypeList[i]])
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
