<!DOCTYPE html><html><head><title>Help for package eLNNpairedCov</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {eLNNpairedCov}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eLNNpairedCov'>
<p>Model-Based Clustering for Paired Data Adjusting for Covariates</p></a></li>
<li><a href='#eLNNpairedCovSEM'>
<p>Model-Based Clustering for Paired Data Adjusting for Covariates Using Simulated Annealing Modified EM</p></a></li>
<li><a href='#esDiff'>
<p>An ExpressionSet Object Storing a Simulated Data</p></a></li>
<li><a href='#genSimDat'>
<p>Generate Simulated Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Model-Based Gene Selection for Paired Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-12-22</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), Biobase</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, graphics, stats, limma, methods, parallel</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bioinformatics, DifferentialExpression</td>
</tr>
<tr>
<td>Description:</td>
<td>Model-based clustering for paired data based on the regression of a mixture of Bayesian hierarchical models on covariates. Zhang et al. (2023) &lt;<a href="https://doi.org/10.1186%2Fs12859-023-05556-x">doi:10.1186/s12859-023-05556-x</a>&gt;.                      </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-10 03:28:34 UTC; zhyl1</td>
</tr>
<tr>
<td>Author:</td>
<td>Yixin Zhang [aut, cre],
  Wei Liu [aut, ctb],
  Weiliang Qiu [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yixin Zhang &lt;zhyl133@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-11 09:30:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='eLNNpairedCov'>
Model-Based Clustering for Paired Data Adjusting for Covariates
</h2><span id='topic+eLNNpairedCov'></span>

<h3>Description</h3>

<p>Model-based clustering based on extended log-normal normal model for paired data adjusting for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eLNNpairedCov(
  EsetDiff,
  fmla = ~Age + Sex,
  probeID.var = "probeid",
  gene.var = "gene",
  chr.var = "chr",
  scaleFlag = TRUE,
  Maxiter =10,
  maxIT = 10,
  b=c(2,2,2),
  converge_threshold = 1e-3,
  optimMethod = "L-BFGS-B",
  bound.alpha = c(0.001, 6),
  bound.beta = c(0.001, 6),
  bound.k = c(0.001, 0.9999),
  bound.eta = c(-10, 10),
  mc.cores = 1,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eLNNpairedCov_+3A_esetdiff">EsetDiff</code></td>
<td>

<p>An ExpressionSet object storing the log2 difference
between post-treatment and pre-treatment.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_fmla">fmla</code></td>
<td>

<p>A formula without outcome variable.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_probeid.var">probeID.var</code></td>
<td>

<p>character. Indicates the probe id.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_gene.var">gene.var</code></td>
<td>

<p>character. Indicates the gene symbol.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_chr.var">chr.var</code></td>
<td>

<p>character. Indicates the chromosome.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_scaleflag">scaleFlag</code></td>
<td>

<p>logical. Indicating if rows (probes) need to be scaled (but not centered).
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_maxiter">Maxiter</code></td>
<td>

<p>integer. 
The max allowed number of iterations for EM algorithm. Default value is maxRT = 100. 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_maxit">maxIT</code></td>
<td>

<p>integer. The max allowed number of iterations in R built-in function optim. Default value is maxIT = 100.
<code>maxIT</code>. 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_b">b</code></td>
<td>

<p>numeric. 
A vector of concentration parameters used in Dirichlet distribution. Default value is b = c(2,2,2). 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_converge_threshold">converge_threshold</code></td>
<td>

<p>numeric.
One of the two termination criteria of iteration. 
The smaller this value is set, the harder the optimization procedure in eLNNpaired will be considered to be converged. Default value is converge_threshold <code>= 1e-6</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_optimmethod">optimMethod</code></td>
<td>

<p>character. Indicates the method for optimization. <code><a href="stats.html#topic+optim">optim</a></code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_bound.alpha">bound.alpha</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">\alpha_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_bound.beta">bound.beta</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">\beta_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_bound.k">bound.k</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">k_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_bound.eta">bound.eta</code></td>
<td>

<p>numeric. A vector of <code>p+1</code> positive numbers to specify lower and upper bound of estimate of <code class="reqn">\eta_c</code>, <code>c="OE", "UE", or "NE"</code>,
where <code>p</code> is the number of covariates.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_mc.cores">mc.cores</code></td>
<td>

<p>integer. A positive integer specifying number of computer cores to be used by parallel computing.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCov_+3A_verbose">verbose</code></td>
<td>

<p>logic.
An indicator variable telling if print out intermediate results: FALSE for not printing out, TRUE for printing out. Default value is verbose = False.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A gene will be assigned to cluster &ldquo;NE&rdquo; if its posterior probability for non-differentially expressed gene cluster is the largest. 
A gene will be assigned to cluster &ldquo;OE&rdquo; if its posterior probability for over-expressed gene cluster is the largest. A gene will be assigned to cluster &ldquo;UE&rdquo; if its responsibility for under-expressed gene cluster is the largest.
</p>


<h3>Value</h3>

<p>A list of 9 elementes: 
</p>
<table>
<tr><td><code>par.ini</code></td>
<td>
<p>initial estimate of parameter</p>
</td></tr>
<tr><td><code>par.final</code></td>
<td>
<p>A vector of the estimated model parameters in original scale.</p>
</td></tr>
<tr><td><code>memGenes</code></td>
<td>
<p>probe cluster membership based on eLNNpairedCov algorithm.</p>
</td></tr>
<tr><td><code>memGenes2</code></td>
<td>
<p>probe cluster membership based on eLNNpairedCov algorithm. 2-categories: &quot;DE&quot; indicates differentially expressed; &quot;NE&quot; indicates non-differentially expressed.</p>
</td></tr>
<tr><td><code>memGenes.limma</code></td>
<td>
<p>probe cluster membership based on limma.</p>
</td></tr>
<tr><td><code>res.ini</code></td>
<td>
<p>results of limma analysis</p>
</td></tr>
<tr><td><code>update_info</code></td>
<td>
<p>object returned by <code><a href="stats.html#topic+optim">optim</a></code> function</p>
</td></tr>
<tr><td><code>wmat</code></td>
<td>
<p>matrix of responsibilities</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>number of EM iterations.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yixin Zhang <a href="mailto:zhyl133@gmail.com">zhyl133@gmail.com</a>,
Wei Liu <a href="mailto:liuwei@mathstat.yorku.ca">liuwei@mathstat.yorku.ca</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@sanofi.com">weiliang.qiu@sanofi.com</a>
</p>


<h3>References</h3>

<p>Zhang Y, Liu W, Qiu W.
A model-based clustering via mixture of hierarchical models with covariate adjustment for detecting differentially expressed genes from paired design.
<em>BMC Bioinformatics</em> 24, 423 (2023)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esDiff)

res = eLNNpairedCov(EsetDiff = esDiff, 
		    fmla = ~Age + Sex, 
		    probeID.var = "probeid", 
		    gene.var = "gene", 
		    chr.var = "chr",
		    scaleFlag = FALSE,
		    mc.cores = 1,
        verbose = TRUE)

# true probe cluster membership
memGenes.true = fData(esDiff)$memGenes.true
print(table(memGenes.true))

# probe cluster membership
memGenes.limma = res$memGenes.limma
print(table(memGenes.limma))

# final probe cluster membership
memGenes = res$memGenes
print(table(memGenes))

# cross tables
print(table(memGenes.true, memGenes.limma))
print(table(memGenes.true, memGenes))

# accuracies
print(mean(memGenes.true == memGenes.limma))
print(mean(memGenes.true == memGenes))

</code></pre>

<hr>
<h2 id='eLNNpairedCovSEM'>
Model-Based Clustering for Paired Data Adjusting for Covariates Using Simulated Annealing Modified EM
</h2><span id='topic+eLNNpairedCovSEM'></span>

<h3>Description</h3>

<p>Model-based clustering based on extended log-normal normal model for paired data adjusting for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eLNNpairedCovSEM(
  EsetDiff,
  fmla = ~Age + Sex,
  probeID.var = "probeid",
  gene.var = "gene",
  chr.var = "chr",
  scaleFlag = TRUE,
  Maxiter =10,
  maxIT = 10,
  b=c(2,2,2),
  converge_threshold = 1e-3,
  optimMethod = "L-BFGS-B",
  bound.alpha = c(0.001, 6),
  bound.beta = c(0.001, 6),
  bound.k = c(0.001, 0.9999),
  bound.eta = c(-10, 10),
  mc.cores = 1,
  temp0 = 2, 
  r_cool=0.9,
  verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eLNNpairedCovSEM_+3A_esetdiff">EsetDiff</code></td>
<td>

<p>An ExpressionSet object storing the log2 difference
between post-treatment and pre-treatment.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_fmla">fmla</code></td>
<td>

<p>A formula without outcome variable.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_probeid.var">probeID.var</code></td>
<td>

<p>character. Indicates the probe id.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_gene.var">gene.var</code></td>
<td>

<p>character. Indicates the gene symbol.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_chr.var">chr.var</code></td>
<td>

<p>character. Indicates the chromosome.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_scaleflag">scaleFlag</code></td>
<td>

<p>logical. Indicating if rows (probes) need to be scaled (but not centered).
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_maxiter">Maxiter</code></td>
<td>

<p>integer. 
The max allowed number of iterations for EM algorithm. Default value is maxRT = 100. 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_maxit">maxIT</code></td>
<td>

<p>integer. The max allowed number of iterations in R built-in function optim. Default value is maxIT = 100.
<code>maxIT</code>. 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_b">b</code></td>
<td>

<p>numeric. 
A vector of concentration parameters used in Dirichlet distribution. Default value is b = c(2,2,2). 
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_converge_threshold">converge_threshold</code></td>
<td>

<p>numeric.
One of the two termination criteria of iteration. 
The smaller this value is set, the harder the optimization procedure in eLNNpaired will be considered to be converged. Default value is converge_threshold <code>= 1e-6</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_optimmethod">optimMethod</code></td>
<td>

<p>character. Indicates the method for optimization. <code><a href="stats.html#topic+optim">optim</a></code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_bound.alpha">bound.alpha</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">\alpha_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_bound.beta">bound.beta</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">\beta_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_bound.k">bound.k</code></td>
<td>

<p>numeric. A vector of 2 positive numbers to specify lower and upper bound of estimate of <code class="reqn">k_c</code>, <code>c="OE", "UE", or "NE"</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_bound.eta">bound.eta</code></td>
<td>

<p>numeric. A vector of <code>p+1</code> positive numbers to specify lower and upper bound of estimate of <code class="reqn">\eta_c</code>, <code>c="OE", "UE", or "NE"</code>,
where <code>p</code> is the number of covariates.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_mc.cores">mc.cores</code></td>
<td>

<p>integer. A positive integer specifying number of computer cores to be used by parallel computing.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_temp0">temp0</code></td>
<td>

<p>numeric. Initial temperature in simulated-annealing modified EM.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_r_cool">r_cool</code></td>
<td>

<p>numeric. Cooling rate in simulated-annealing modified EM, which is inside interval <code class="reqn">(0, 1)</code>.
</p>
</td></tr>
<tr><td><code id="eLNNpairedCovSEM_+3A_verbose">verbose</code></td>
<td>

<p>logic.
An indicator variable telling if print out intermediate results: FALSE for not printing out, TRUE for printing out. Default value is verbose = False.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A gene will be assigned to cluster &ldquo;NE&rdquo; if its posterior probability for non-differentially expressed gene cluster is the largest. 
A gene will be assigned to cluster &ldquo;OE&rdquo; if its posterior probability for over-expressed gene cluster is the largest. A gene will be assigned to cluster &ldquo;UE&rdquo; if its responsibility for under-expressed gene cluster is the largest.
</p>


<h3>Value</h3>

<p>A list of 9 elementes: 
</p>
<table>
<tr><td><code>par.ini</code></td>
<td>
<p>initial estimate of parameter</p>
</td></tr>
<tr><td><code>par.final</code></td>
<td>
<p>A vector of the estimated model parameters in original scale.</p>
</td></tr>
<tr><td><code>memGenes</code></td>
<td>
<p>probe cluster membership based on eLNNpairedCovSEM algorithm.</p>
</td></tr>
<tr><td><code>memGenes2</code></td>
<td>
<p>probe cluster membership based on eLNNpairedCovSEM algorithm. 2-categories: &quot;DE&quot; indicates differentially expressed; &quot;NE&quot; indicates non-differentially expressed.</p>
</td></tr>
<tr><td><code>memGenes.limma</code></td>
<td>
<p>probe cluster membership based on limma.</p>
</td></tr>
<tr><td><code>res.ini</code></td>
<td>
<p>results of limma analysis</p>
</td></tr>
<tr><td><code>update_info</code></td>
<td>
<p>object returned by <code><a href="stats.html#topic+optim">optim</a></code> function</p>
</td></tr>
<tr><td><code>wmat</code></td>
<td>
<p>matrix of responsibilities</p>
</td></tr>
<tr><td><code>iter.EM</code></td>
<td>
<p>number of EM iterations.</p>
</td></tr>
<tr><td><code>tempFinal</code></td>
<td>
<p>final temperature in simulated-annealing modification EM</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yixin Zhang <a href="mailto:zhyl133@gmail.com">zhyl133@gmail.com</a>,
Wei Liu <a href="mailto:liuwei@mathstat.yorku.ca">liuwei@mathstat.yorku.ca</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@sanofi.com">weiliang.qiu@sanofi.com</a>
</p>


<h3>References</h3>

<p>Zhang Y, Liu W, Qiu W.
A model-based clustering via mixture of hierarchical models with covariate adjustment for detecting differentially expressed genes from paired design.
<em>BMC Bioinformatics</em> 24, 423 (2023)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esDiff)

res.SEM = eLNNpairedCovSEM(EsetDiff = esDiff, 
		    fmla = ~Age + Sex, 
		    probeID.var = "probeid", 
		    gene.var = "gene", 
		    chr.var = "chr",
		    scaleFlag = FALSE,
		    mc.cores = 1,
        verbose = TRUE)

# true probe cluster membership
memGenes.true = fData(esDiff)$memGenes.true
print(table(memGenes.true))

# probe cluster membership
memGenes.limma = res.SEM$memGenes.limma
print(table(memGenes.limma))

# final probe cluster membership
memGenes.SEM = res.SEM$memGenes
print(table(memGenes.SEM))

# cross tables
print(table(memGenes.true, memGenes.limma))
print(table(memGenes.true, memGenes.SEM))

# accuracies
print(mean(memGenes.true == memGenes.limma))
print(mean(memGenes.true == memGenes.SEM))

</code></pre>

<hr>
<h2 id='esDiff'>
An ExpressionSet Object Storing a Simulated Data
</h2><span id='topic+esDiff'></span>

<h3>Description</h3>

<p>An ExpressionSet object storing a simulated data of log2 difference of expression levels with 1000 probes, 20 subjects, and 2 covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("esDiff")</code></pre>


<h3>Details</h3>

<p>This dataset was generated from the mixture of 3-component
Bayesian hierarchical models. For true parameters, please
refer to the manual for the R function <code><a href="#topic+genSimDat">genSimDat</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(esDiff)

print(esDiff)

</code></pre>

<hr>
<h2 id='genSimDat'>
Generate Simulated Data
</h2><span id='topic+genSimDat'></span>

<h3>Description</h3>

<p>Generate a simulated dataset from a mixture of Bayesian hierarchical models
with two covariates: age and sex.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genSimDat(G, n, psi, t_pi, m.age = 50, sd.age = 5, p.female = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genSimDat_+3A_g">G</code></td>
<td>

<p>integer. Number of probes.
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_n">n</code></td>
<td>

<p>integer. Number of samples.
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_psi">psi</code></td>
<td>

<p>numeric. A vector of model hyper-parameters with elements
<code class="reqn">\alpha_1</code>, <code class="reqn">\beta_1</code>, <code class="reqn">k_1</code>, 
<code class="reqn">\eta_{1,intercept}</code>, <code class="reqn">\eta_{1,age}</code>, <code class="reqn">\eta_{1,sex}</code>,
<code class="reqn">\alpha_2</code>, <code class="reqn">\beta_2</code>, <code class="reqn">k_2</code>, 
<code class="reqn">\eta_{2,intercept}</code>, <code class="reqn">\eta_{2,age}</code>, <code class="reqn">\eta_{2,sex}</code>,
<code class="reqn">\alpha_3</code>, <code class="reqn">\beta_3</code>, <code class="reqn">k_3</code>, 
<code class="reqn">\eta_{3,intercept}</code>, <code class="reqn">\eta_{3,age}</code>, <code class="reqn">\eta_{3,sex}</code>.
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_t_pi">t_pi</code></td>
<td>

<p>numeric. A vector of mixture proportions: <code class="reqn">\pi_1</code> (proportion
for probes over-expressed in cases); 
<code class="reqn">\pi_2</code> (proportion for probes under-expressed in cases).
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_m.age">m.age</code></td>
<td>

<p>numeric. mean age.
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_sd.age">sd.age</code></td>
<td>

<p>numeric. standard deviation of age.
</p>
</td></tr>
<tr><td><code id="genSimDat_+3A_p.female">p.female</code></td>
<td>

<p>numeric. proportion of females.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ExpressionSet object.
</p>


<h3>Note</h3>

<p>Age will be mean-centered and scaled so that it will
have mean zero and variance one.
</p>


<h3>Author(s)</h3>

<p>Yixin Zhang <a href="mailto:zhyl133@gmail.com">zhyl133@gmail.com</a>,
Wei Liu <a href="mailto:liuwei@mathstat.yorku.ca">liuwei@mathstat.yorku.ca</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@sanofi.com">weiliang.qiu@sanofi.com</a>
</p>


<h3>References</h3>

<p>Zhang Y, Liu W, Qiu W.
A model-based clustering via mixture of hierarchical models with covariate adjustment for detecting differentially expressed genes from paired design.
<em>BMC Bioinformatics</em> 24, 423 (2023)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1234567)

true.psi = c(2, 1, 0.8,
             0.1, -0.01, -0.1,
             2, 1, 0.8,
             -0.1, -0.01, -0.1,
             2, 1, 0.8,
             -0.01, -0.1)
names(true.psi)=c("alpha1", "beta1", "k1",
                  "eta1.intercept", "eta1.Age", "eta1.Sex",
                  "alpha2", "beta2", "k2",
                  "eta2.intercept", "eta2.Age", "eta2.Sex",
                  "alpha3", "beta3", "k3",
                  "eta3.Age", "eta3.Sex")
true.pi=c(0.1, 0.1)
names(true.pi)=c("pi.OE", "pi.UE")
par.true=c(true.pi, true.psi)

esDiff = genSimDat(G = 1000, 
	       n = 20, 
	       psi = true.psi, 
	       t_pi = true.pi, 
	       m.age = 0, # scaled age 
	       sd.age = 1, # scaled age
	       p.female = 0.5)

print(esDiff)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
