<!DOCTYPE html><html><head><title>Help for package EMAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EMAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data.m'><p>A data for 221 participants</p></a></li>
<li><a href='#E.result'><p>An <code>Emas</code> results data.</p></a></li>
<li><a href='#Emas'><p>Epigenome-Wide Mediation Analysis Study</p></a></li>
<li><a href='#EMAS.manhattan'><p>Plotting the manhattan plot from the EMAS results</p></a></li>
<li><a href='#Emas.parallel'><p>Epigenome-Wide Mediation Analysis Study: Parallel multiple mediation model</p></a></li>
<li><a href='#EMAS.volcano'><p>Plotting the volcano plot from the EMAS results</p></a></li>
<li><a href='#Mvalue'><p>A M-value matrix for 221 participants</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Epigenome-Wide Mediation Analysis Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-08-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiuquan Nie [aut, cph, cre],
  Mengyi Wang [ctb, rev],
  Weihong Chen [fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiuquan Nie &lt;niexiuquan1995@foxmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>mediation, parallel, multilevel, minfi, ggplot2, qqman,
lavaan, IlluminaHumanMethylationEPICanno.ilm10b4.hg19,
IlluminaHumanMethylation450kanno.ilmn12.hg19</td>
</tr>
<tr>
<td>Description:</td>
<td>DNA methylation is essential for human, and environment can change the DNA methylation
    and affect body status. Epigenome-Wide Mediation Analysis Study (EMAS) can find
    potential mediator CpG sites between exposure (x) and outcome (y) in epigenome-wide.
    For more information on the methods we used, please see the following references:
    Tingley, D. (2014) &lt;<a href="https://doi.org/10.18637%2Fjss.v059.i05">doi:10.18637/jss.v059.i05</a>&gt;,
    Turner, S. D. (2018) &lt;<a href="https://doi.org/10.21105%2Fjoss.00731">doi:10.21105/joss.00731</a>&gt;,
    Rosseel, D. (2012) &lt;<a href="https://doi.org/10.18637%2Fjss.v048.i02">doi:10.18637/jss.v048.i02</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-10 11:52:57 UTC; niexi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-11 13:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='data.m'>A data for 221 participants</h2><span id='topic+data.m'></span>

<h3>Description</h3>

<p>A data for 221 participants. The data were fabricated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data.m
</code></pre>


<h3>Format</h3>

<p>A data.frame contains 221 obs of 7 variables. The variables are:
</p>

<dl>
<dt>ID</dt><dd><p>The ID of these 221 people.</p>
</dd>
<dt>age</dt><dd><p>A numeric.</p>
</dd>
<dt>gender</dt><dd><p>A two level factor: &quot;1&quot; or &quot;2&quot;</p>
</dd>
<dt>CD8T</dt><dd><p>A numeric.</p>
</dd>
<dt>CD4T</dt><dd><p>A numeric.</p>
</dd>
<dt>x</dt><dd><p>A numeric.</p>
</dd>
<dt>y</dt><dd><p>A numeric.</p>
</dd>
</dl>


<hr>
<h2 id='E.result'>An <code>Emas</code> results data.</h2><span id='topic+E.result'></span>

<h3>Description</h3>

<p>An <code>Emas</code> results data for 2000 CpGs. The data were fabricated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>E.result
</code></pre>


<h3>Format</h3>

<p>A data.frame produced by <code>Emas</code> contains 2000 obs of 13 variables.
</p>

<hr>
<h2 id='Emas'>Epigenome-Wide Mediation Analysis Study</h2><span id='topic+Emas'></span>

<h3>Description</h3>

<p>This function can perform the Epigenome-Wide Mediation Analysis Study (EMAS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Emas(
  data,
  M.matrix,
  id = "",
  x = "",
  y = "",
  x.cov = c(),
  y.cov = c(),
  m.cov = c(),
  mem.sav = FALSE,
  p.th = 0.1,
  ini.sims = 100,
  boot = FALSE,
  cl.n = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Emas_+3A_data">data</code></td>
<td>
<p>A data.frame included id, x, y, x.cov, y.cov, m.cov.</p>
</td></tr>
<tr><td><code id="Emas_+3A_m.matrix">M.matrix</code></td>
<td>
<p>A matrix with the epigenome-wide CpG information, maybe a M-value matrix or a beta value matrix.</p>
</td></tr>
<tr><td><code id="Emas_+3A_id">id</code></td>
<td>
<p>Variable name of the id.</p>
</td></tr>
<tr><td><code id="Emas_+3A_x">x</code>, <code id="Emas_+3A_y">y</code></td>
<td>
<p>Variable name of exposure(x) and outcome(y).</p>
</td></tr>
<tr><td><code id="Emas_+3A_x.cov">x.cov</code></td>
<td>
<p>Variable names of covariates related to exposure(x).</p>
</td></tr>
<tr><td><code id="Emas_+3A_y.cov">y.cov</code></td>
<td>
<p>Variable names of covariates related to outcome(y).</p>
</td></tr>
<tr><td><code id="Emas_+3A_m.cov">m.cov</code></td>
<td>
<p>Variable names of covariates related to mediator(m).</p>
</td></tr>
<tr><td><code id="Emas_+3A_mem.sav">mem.sav</code></td>
<td>
<p>A logical value. If 'TRUE', the memory required for the function will decrease, but the speed will also decrease.</p>
</td></tr>
<tr><td><code id="Emas_+3A_p.th">p.th</code></td>
<td>
<p>Sobel indirect effects P-value threshold for subsequent nonparametric bootstrap or quasi-Bayesian approximation mediation analyses.</p>
</td></tr>
<tr><td><code id="Emas_+3A_ini.sims">ini.sims</code></td>
<td>
<p>Initial number of Monte Carlo draws for nonparametric bootstrap or quasi-Bayesian approximation.</p>
</td></tr>
<tr><td><code id="Emas_+3A_boot">boot</code></td>
<td>
<p>A logical value. If 'FALSE' a quasi-Bayesian approximation is used for confidence intervals; if 'TRUE' nonparametric bootstrap will be used. Default is 'FALSE'.</p>
</td></tr>
<tr><td><code id="Emas_+3A_cl.n">cl.n</code></td>
<td>
<p>Number of cores used for parallel computing.</p>
</td></tr>
<tr><td><code id="Emas_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="parallel.html#topic+makeCluster">makeCluster</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can perform the Epigenome-Wide Mediation Analysis Study (EMAS) to explore the potential mediating CpG sites of exposure variables affecting outcome variables within the epigenome-wide.
</p>


<h3>Value</h3>

<p><code>Emas</code> returns a data.frame with the average mediation effects(AME), 
average direct effects(ADE), total effects, mediation proportion.
</p>

<ul>
<li><p>AMEEst: Point estimates for average mediation effects under the exposure conditions.
</p>
</li>
<li><p>AMElow95, AMEupp95: 95 percentage confidence intervals for average mediation effects.
</p>
</li>
<li><p>AME.P: Two-sided p-values for average mediation effects.
</p>
</li>
<li><p>ADEEst: Point estimates for average direct effect under the exposure conditions.
</p>
</li>
<li><p>ADElow95, ADEupp95: 95 percentage confidence intervals for average direct effects.
</p>
</li>
<li><p>ADE.P: Two-sided p-values for average direct effects.
</p>
</li>
<li><p>TotEst: Point estimate for total effect.
</p>
</li>
<li><p>Totlow95, Totupp95: 95 percentage confidence interval for total effect.
</p>
</li>
<li><p>Tot.P: Two-sided p-values for total effect.
</p>
</li>
<li><p>PropEst: The &quot;proportions mediated&quot;, or the size of the average mediation effects relative to the total effect.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Xiuquan Nie, niexiuquan1995@foxmail.com
</p>


<h3>References</h3>

<p>Tingley, D., Yamamoto, T., Hirose, K., Imai, K. and Keele, L. (2014). mediation: R package for Causal Mediation Analysis.
<em>Journal of Statistical Software</em>, 59(5), 1–38. doi: <a href="https://doi.org/10.18637/jss.v059.i05">10.18637/jss.v059.i05</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.m)
data(Mvalue)
E.result &lt;- Emas(data.m, Mvalue, id = "ID", x = "x", y = "y",
                 x.cov = c("age", "gender"),
                 y.cov = c("age", "gender"),
                 m.cov = c("age", "gender", "CD8T", "CD4T"), 
                 p.th = 0.1, ini.sims = 100, boot = FALSE, cl.n = 1)
</code></pre>

<hr>
<h2 id='EMAS.manhattan'>Plotting the manhattan plot from the EMAS results</h2><span id='topic+EMAS.manhattan'></span>

<h3>Description</h3>

<p>Function to plot a manhattan plot from the <code>Emas</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EMAS.manhattan(E.result, type = "EPIC", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EMAS.manhattan_+3A_e.result">E.result</code></td>
<td>
<p>A data.frame produced by <code>Emas</code>.</p>
</td></tr>
<tr><td><code id="EMAS.manhattan_+3A_type">type</code></td>
<td>
<p>A character string indicating the type of annotation, only &quot;EPIC&quot; and &quot;450k&quot; are available.</p>
</td></tr>
<tr><td><code id="EMAS.manhattan_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="qqman.html#topic+manhattan">manhattan</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can plot a manhattan plot from the <code>Emas</code> results according to the annotation from 450k or EPIC.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Xiuquan Nie, niexiuquan1995@foxmail.com
</p>


<h3>References</h3>

<p>Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots.
<em>Journal of Open Source Software</em>, 3(25), 731. doi: <a href="https://doi.org/10.21105/joss.00731">10.21105/joss.00731</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(E.result)
EMAS.manhattan(E.result, type = "EPIC",
               genomewideline = -log10(0.05/2000),
               suggestiveline = -log10(1/100), ylim=c(0,5))
</code></pre>

<hr>
<h2 id='Emas.parallel'>Epigenome-Wide Mediation Analysis Study: Parallel multiple mediation model</h2><span id='topic+Emas.parallel'></span>

<h3>Description</h3>

<p>This function can perform the parallel multiple mediation model after the Epigenome-Wide Mediation Analysis Study (EMAS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Emas.parallel(
  data,
  M.matrix,
  id = "",
  x = "",
  y = "",
  x.cov = c(),
  y.cov = c(),
  m.cov = c(),
  m.cor = TRUE,
  boot = FALSE,
  lavaan = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Emas.parallel_+3A_data">data</code></td>
<td>
<p>A data.frame included id, x, y, x.cov, y.cov, m.cov.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_m.matrix">M.matrix</code></td>
<td>
<p>A matrix with the CpG information screened from EMAS., maybe a M-value matrix or a beta value matrix.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_id">id</code></td>
<td>
<p>Variable name of the id.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_x">x</code>, <code id="Emas.parallel_+3A_y">y</code></td>
<td>
<p>Variable name of exposure(x) and outcome(y).</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_x.cov">x.cov</code></td>
<td>
<p>Variable names of covariates related to exposure(x).</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_y.cov">y.cov</code></td>
<td>
<p>Variable names of covariates related to outcome(y).</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_m.cov">m.cov</code></td>
<td>
<p>Variable names of covariates related to mediator(m).</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_m.cor">m.cor</code></td>
<td>
<p>A logical value. If 'TRUE', the mediators in the parallel multiple mediation model are set to correlate with each other.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_boot">boot</code></td>
<td>
<p>A logical value or a numeric value. If a numeric value, the number for bootstrap.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_lavaan">lavaan</code></td>
<td>
<p>A logical value. If 'TRUE', a lavaan object will be given.</p>
</td></tr>
<tr><td><code id="Emas.parallel_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="lavaan.html#topic+sem">sem</a></code> from <code><a href="lavaan.html#topic+lavaan">lavaan</a></code> package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can perform the parallel multiple mediation model after the Epigenome-Wide Mediation Analysis Study (EMAS) to further explore the potential parallel mediating CpG sites of exposure variables affecting outcome variables.
</p>


<h3>Value</h3>

<p><code>Emas.parallel</code> returns a data.frame with the average mediation effects(AME), 
average direct effects(ADE), and total effects(Tot). If <code>lavaan</code> is 'TRUE', a lavaan object will be given.
</p>


<h3>Author(s)</h3>

<p>Xiuquan Nie, niexiuquan1995@foxmail.com
</p>


<h3>References</h3>

<p>Rosseel, Y. (2012). lavaan: An R Package for Structural Equation Modeling.
<em>Journal of Statistical Software</em>, 48(2), 1–36. doi: <a href="https://doi.org/10.18637/jss.v048.i02">10.18637/jss.v048.i02</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.m)
data(Mvalue)
EP.result &lt;- Emas.parallel(data.m, Mvalue, 
                           id = "ID", x = "x", y = "y", 
                           x.cov = c("age", "gender"), 
                           y.cov = c("age", "gender"), 
                           m.cov = c("age", "gender", "CD8T", "CD4T"), 
                           m.cor = TRUE, boot = FALSE, lavaan = FALSE)
</code></pre>

<hr>
<h2 id='EMAS.volcano'>Plotting the volcano plot from the EMAS results</h2><span id='topic+EMAS.volcano'></span>

<h3>Description</h3>

<p>Function to plot a volcano plot from the <code>Emas</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EMAS.volcano(
  E.result,
  epiwideline = -log10(1e-07),
  suggestiveline = -log10(1e-05)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EMAS.volcano_+3A_e.result">E.result</code></td>
<td>
<p>A data.frame produced by <code>Emas</code>.</p>
</td></tr>
<tr><td><code id="EMAS.volcano_+3A_epiwideline">epiwideline</code></td>
<td>
<p>Where to draw a &quot;epigenome-wide sigificant&quot; line. Default -log10(1.0e-7).</p>
</td></tr>
<tr><td><code id="EMAS.volcano_+3A_suggestiveline">suggestiveline</code></td>
<td>
<p>Where to draw a &quot;suggestive&quot; line. Default -log10(1.0e-5). Set to FALSE to disable.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can plot a volcano plot from the <code>Emas</code> results.
</p>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Author(s)</h3>

<p>Xiuquan Nie, niexiuquan1995@foxmail.com
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(E.result)
EMAS.volcano(E.result, 
             epiwideline = -log10(0.05/2000),
             suggestiveline = -log10(1/100))
</code></pre>

<hr>
<h2 id='Mvalue'>A M-value matrix for 221 participants</h2><span id='topic+Mvalue'></span>

<h3>Description</h3>

<p>A M-value matrix for 221 participants. The data were fabricated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mvalue
</code></pre>


<h3>Format</h3>

<p>A matrix contains 221 obs of 10 CpGs.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
