<!DOCTYPE html><html><head><title>Help for package emBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {emBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#emBayes-package'><p>Robust Bayesian Variable Selection via Expectation-Maximization</p></a></li>
<li><a href='#cv.emBayes'><p>k-folds cross-validation for 'emBayes'</p></a></li>
<li><a href='#data'><p>simulated gene expression example data</p></a></li>
<li><a href='#emBayes'><p>fit a model with given tuning parameters</p></a></li>
<li><a href='#print.cv.emBayes'><p>print an cv.emBayes result</p></a></li>
<li><a href='#print.emBayes'><p>print an emBayes result</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Bayesian Variable Selection via Expectation-Maximization</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-27</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuwen Liu &lt;yuwen@ksu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Variable selection methods have been extensively developed for analyzing high-dimensional omics data within both the frequentist and Bayesian frameworks. This package implemented the spike-and-slab quantile LASSO which has been developed along the line of Bayesian hierarchical model but deeply rooted in the frequentist regularization methods by utilizing the Expectation–Maximization (EM) algorithm. Therefore, the proposed method borrows strength from both the frequentist and Bayesian frameworks while overcoming their respective limitations. The spike-and-slab quantile LASSO can handle data irregularity in terms of skewness and outliers in the disease trait, compared to its nonrobust alternative, the spike-and-slab LASSO, which has also been implemented in the package. The core module of this package is developed in 'C++'.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, glmnet</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-27 13:30:38 UTC; 26057</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuwen Liu [aut, cre],
  Cen Wu [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-29 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='emBayes-package'>Robust Bayesian Variable Selection via Expectation-Maximization</h2><span id='topic+emBayes-package'></span>

<h3>Description</h3>

<p>This package provides the implementation of the spike-and-slab quantile LASSO (ssQLASSO) which combines the strength of Bayesian robust variable selection and the Expectation-Maximization (EM) coordinate descent approach. The alternative method spike-and-slab LASSO (ssLASSO) is also included in the package.
</p>


<h3>Details</h3>

<p>Two user friendly, integrated interface <strong>cv.emBayes()</strong> and <strong>emBayes()</strong> allows users to flexibly choose the variable selection method by specifying the following parameter:
</p>

<table>
<tr>
 <td style="text-align: right;">
quant: </td><td style="text-align: left;"> to specify different quantiles when using robust methods.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
func: </td><td style="text-align: left;"> the model to perform variable selection. Two choices are available: </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> "ssLASSO" and "ssQLASSO". </td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
error: </td><td style="text-align: left;"> to specify the difference between expectations of likelihood of two  </td>
</tr>
<tr>
 <td style="text-align: right;"> </td><td style="text-align: left;"> consecutive iterations. It can be used to determine convergence.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
maxiter: </td><td style="text-align: left;"> to specify the maximum number of iterations.
</td>
</tr>

</table>

<p>Function cv.emBayes() returns cross-validation errors based on the check loss, least squares loss and Schwarz Information Criterion along with the corresponding optimal tuning parameters.
Function emBayes() returns the estimated intercept, clinical coefficients, beta coefficients, scale parameter, probability parameter, number of iterations and expectation of likelihood at each iteration.
</p>


<h3>References</h3>

<p>Liu, Y. and Wu, C. (2023). Spike-and-Slab Quantile LASSO. (to be submitted)
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y., and Wu, C. (2022). Robust Bayesian variable selection for gene–environment interactions.
<em>Biometrics.</em> <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang,Y. and Wu, C. (2019). Robust network-based regularization
and variable selection for high dimensional genomics data in cancer prognosis.
<em>Genet. Epidemiol.</em>, 43:276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Wu, C., Zhang, Q., Jiang,Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi)genetic measurements.
<em>J Multivar Anal.</em>, 168:119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Tang, Z., Shen, Y., Zhang, X., and Yi, N. (2017). The spike-and-slab lasso generalized linear models for prediction and associated genes detection. 
<em>Genetics</em>, 205(1), 77-88 <a href="https://doi.org/10.1534/genetics.116.192195">doi:10.1534/genetics.116.192195</a>
</p>
<p>Tang, Z., Shen, Y., Zhang, X., and Yi, N. (2017). The spike-and-slab lasso Cox model for survival prediction and associated genes detection. 
<em>Bioinformatics</em>, 33(18), 2799-2807 <a href="https://doi.org/10.1093/bioinformatics/btx300">doi:10.1093/bioinformatics/btx300</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a> 
</p>
<p>Zhou, Y. H., Ni, Z. X., and Li, Y. (2014). Quantile regression via the EM algorithm. 
<em>Communications in Statistics-Simulation and Computation</em>, 43(10), 2162-2172 <a href="https://doi.org/10.1080/03610918.2012.746980">doi:10.1080/03610918.2012.746980</a>
</p>
<p>Ročková, V., and George, E. I. (2014). EMVS: The EM approach to Bayesian variable selection. 
<em>Journal of the American Statistical Association</em>, 109(506), 828-846 <a href="https://doi.org/10.1080/01621459.2013.869223">doi:10.1080/01621459.2013.869223</a>
</p>
<p>Li, Q., Lin, N., and Xi, R. (2010). Bayesian regularized quantile regression.
<em>Bayesian Analysis</em>, 5(3), 533-556 <a href="https://doi.org/10.1214/10-BA521">doi:10.1214/10-BA521</a>
</p>
<p>George, E. I., and McCulloch, R. E. (1993). Variable selection via Gibbs sampling. 
<em>Journal of the American Statistical Association</em>, 88(423), 881-889 <a href="https://doi.org/10.1080/01621459.1993.10476353">doi:10.1080/01621459.1993.10476353</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cv.emBayes">cv.emBayes</a></code> <code><a href="#topic+emBayes">emBayes</a></code>
</p>

<hr>
<h2 id='cv.emBayes'>k-folds cross-validation for 'emBayes'</h2><span id='topic+cv.emBayes'></span>

<h3>Description</h3>

<p>This function performs cross-validation and returns the optimal values of the tuning parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.emBayes(
  y,
  clin = NULL,
  X,
  quant,
  t0,
  t1,
  k,
  func,
  error = 0.01,
  maxiter = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cv.emBayes_+3A_y">y</code></td>
<td>
<p>a vector of response variable.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical factors. It has default value NULL.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_x">X</code></td>
<td>
<p>a matrix of genetic factors.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_quant">quant</code></td>
<td>
<p>value of quantile.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_t0">t0</code></td>
<td>
<p>a user-supplied sequence of the spike scale <code class="reqn">s_{0}</code>.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_t1">t1</code></td>
<td>
<p>a user-supplied sequence of the slab scale <code class="reqn">s_{1}</code>.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_k">k</code></td>
<td>
<p>number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_func">func</code></td>
<td>
<p>methods to perform variable selection. Two choices are available: &quot;ssLASSO&quot; and &quot;ssQLASSO&quot;.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_error">error</code></td>
<td>
<p>cutoff value for determining convergence. The algorithm reaches convergence if the difference in the expected log-likelihood of two iterations is less than the value of error. The default value is 0.01.</p>
</td></tr>
<tr><td><code id="cv.emBayes_+3A_maxiter">maxiter</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm. The default value is 200.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When performing cross-validation for emBayes, function cv.emBayes returns two sets of optimal tuning parameters and their corresponding cross-validation error matrices. 
The spike scale parameter <code class="reqn">CL.s0</code> and the slab scale parameter <code class="reqn">CL.s1</code> are obtained based on the quantile check loss. 
The spike scale parameter <code class="reqn">SL.s0</code> and the slab scale parameter <code class="reqn">SL.s1</code> are obtained based on the least squares loss. 
The spike scale parameter <code class="reqn">SIC.s0</code> and the slab scale parameter <code class="reqn">SIC.s1</code> are obtained based on the Schwarz Information Criterion (SIC).
Corresponding error matrices <code class="reqn">CL.CV</code>, <code class="reqn">SL.CV</code> and <code class="reqn">SIC.CV</code> can also be obtained from the output.
</p>
<p>Schwarz Information Criterion has the following form:
</p>
<p style="text-align: center;"><code class="reqn">SIC=\log\sum_{i=1}^nL(y_i-\hat{y_i})+\frac{\log n}{2n}edf</code>
</p>

<p>where <code class="reqn">L(\cdot)</code> is the check loss and <code class="reqn">edf</code> is the number of close to zero residuals <code class="reqn">(\leq 0.001)</code>. For non-robust method “ssLASSO”, one should use least squares loss for tuning selection. For robust method “ssQLASSO”, one can either use quantile check loss or SIC for tuning selection. We suggest using SIC, since it has been extensively utilized for tuning selection in high-dimensional quantile regression, as documented in numerous literature sources.
</p>


<h3>Value</h3>

<p>A list with components:
</p>
<table>
<tr><td><code>CL.s0</code></td>
<td>
<p>the optimal spike scale under check loss.</p>
</td></tr>
<tr><td><code>CL.s1</code></td>
<td>
<p>the optimal slab scale under check loss.</p>
</td></tr>
<tr><td><code>SL.s0</code></td>
<td>
<p>the optimal slab scale under least squares loss.</p>
</td></tr>
<tr><td><code>SL.s1</code></td>
<td>
<p>the optimal slab scale under least squares loss.</p>
</td></tr>
<tr><td><code>SIC.s0</code></td>
<td>
<p>the optimal slab scale under SIC.</p>
</td></tr>
<tr><td><code>SIC.s1</code></td>
<td>
<p>the optimal slab scale under SIC.</p>
</td></tr>
<tr><td><code>CL.CV</code></td>
<td>
<p>cross-validation error matrix under check loss.</p>
</td></tr>
<tr><td><code>SL.CV</code></td>
<td>
<p>cross-validation error matrix under least squares loss.</p>
</td></tr>
<tr><td><code>SIC.CV</code></td>
<td>
<p>cross-validation error matrix under SIC.</p>
</td></tr>
</table>

<hr>
<h2 id='data'>simulated gene expression example data</h2><span id='topic+data'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the usage of emBayes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data)
</code></pre>


<h3>Format</h3>

<p>The data file consists of five components: y, clin, X, quant, coef and clin.coe. The coefficients and clinical coefficients are the true values of parameters used for generating response y. They can be used for performance evaluation.
</p>


<h3>Details</h3>

<p><strong>The data model for generating response</strong>
</p>
<p>Let <code class="reqn">y_{i}</code> be the response of the <code class="reqn">i</code>-th subject (1<code class="reqn">\leq</code> i<code class="reqn">\leq</code> n). We have <code class="reqn">z_{i}=(1,z_{i1},\dots,z_{iq})^{\top}</code> being a <code class="reqn">(q+1)</code>-dimensional vector of which the last <code class="reqn">q</code> components indicate clinical factors and <code class="reqn">x_{i}=(x_{i1},\dots,x_{ip})^{\top}</code> denoting a <code class="reqn">p</code>-dimensional vector of genetic factors. The linear quantile regression model for the <code class="reqn">\tau</code>-th quantile <code class="reqn">(0&lt;\tau&lt;1)</code> is:
</p>
<p style="text-align: center;"><code class="reqn">y_i=z_i^\top\alpha+x_i^\top\beta+\epsilon_i</code>
</p>

<p>where <code class="reqn">\alpha=(\alpha_0,\cdots,\alpha_q)^\top</code> contains the intercept and the regression coefficients for the clinical covariates. <code class="reqn">\beta=(\beta_1,\cdots,\beta_p)^\top</code> are the regression coefficients and random error <code class="reqn">\epsilon_{i}=(\epsilon_{1},...,\epsilon_{n})^\top</code> is set to follow a T2 distribution and has value <code class="reqn">0</code> at its <code class="reqn">\tau</code>-th quantile.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+emBayes">emBayes</a></code>
</p>

<hr>
<h2 id='emBayes'>fit a model with given tuning parameters</h2><span id='topic+emBayes'></span>

<h3>Description</h3>

<p>This function performs penalized variable selection based on spike-and-slab quantile LASSO (ssQLASSO) or spike-and-slab LASSO (ssLASSO).
Typical usage is to first obtain the optimal spike scale and slab scale using cross-validation, then specify them in the 'emBayes' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>emBayes(y, clin = NULL, X, quant, s0, s1, func, error = 0.01, maxiter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="emBayes_+3A_y">y</code></td>
<td>
<p>a vector of response variable.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical factors. It has default value NULL.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_x">X</code></td>
<td>
<p>a matrix of genetic factors.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_quant">quant</code></td>
<td>
<p>value of quantile.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_s0">s0</code></td>
<td>
<p>value of the spike scale <code class="reqn">s_{0}</code>.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_s1">s1</code></td>
<td>
<p>value of the slab scale <code class="reqn">s_{1}</code>.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_func">func</code></td>
<td>
<p>methods to perform variable selection. Two choices are available: &quot;ssLASSO&quot; and &quot;ssQLASSO&quot;.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_error">error</code></td>
<td>
<p>cutoff value for determining convergence. The algorithm reaches convergence if the difference in the expected log-likelihood of two iterations is less than the value of error. The default value is 0.01.</p>
</td></tr>
<tr><td><code id="emBayes_+3A_maxiter">maxiter</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm. The default value is 200.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The current version of emBayes supports two types of methods: &quot;ssLASSO&quot; and &quot;ssQLASSO&quot;.
</p>

<ul>
<li> <p><strong>ssLASSO:</strong> spike-and-slab LASSO fits a Bayesian linear regression through the EM algorithm.  
</p>
</li>
<li> <p><strong>ssQLASSO:</strong> spike-and-slab quantile LASSO fits a Bayesian quantile regression (based on asymmetric Laplace distribution) through the EM algorithm.
</p>
</li></ul>

<p>Users can choose the desired method by setting func=&quot;ssLASSO&quot; or &quot;ssQLASSO&quot;.
</p>


<h3>Value</h3>

<p>A list with components:
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>a vector containing the estimated intercept and clinical coefficients.</p>
</td></tr>
<tr><td><code>intercept</code></td>
<td>
<p>value of the estimated intercept.</p>
</td></tr>
<tr><td><code>clin.coe</code></td>
<td>
<p>a vector of estimated clinical coefficients.</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>a vector of estimated beta coefficients.</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>value of estimated asymmetric Laplace distribution scale parameter <code class="reqn">\sigma</code>.</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>value of estimated probability parameter <code class="reqn">\theta</code>.</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>value of number of iterations.</p>
</td></tr>
<tr><td><code>ll</code></td>
<td>
<p>a vector of expectation of likelihood at each iteration.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
##load the clinical factors, genetic factors, response and quantile data
clin=data$clin
X=data$X
y=data$y
quant=data$quant

##generate tuning vectors of desired range 
t0 &lt;- seq(0.01,0.015,length.out=2)
t1 &lt;- seq(0.1,0.5,length.out=2)

##perform cross-validation and obtain tuning parameters based on check loss
CV &lt;- cv.emBayes(y,clin,X,quant,t0,t1,k=5,func="ssQLASSO",error=0.01,maxiter=200)
s0 &lt;- CV$CL.s0
s1 &lt;- CV$CL.s1

##perform BQLSS under optimal tuning and calculate value of TP and FP for selecting beta 
EM &lt;- emBayes(y,clin,X,quant,s0,s1,func="ssQLASSO",error=0.01,maxiter=200)
fit &lt;- EM$beta
coef &lt;- data$coef
tp &lt;- sum(fit[coef!=0]!=0)
fp &lt;- sum(fit[coef==0]!=0)
list(tp=tp,fp=fp)

</code></pre>

<hr>
<h2 id='print.cv.emBayes'>print an cv.emBayes result</h2><span id='topic+print.cv.emBayes'></span>

<h3>Description</h3>

<p>Print a summary of an 'cv.emBayes' result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cv.emBayes'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.cv.emBayes_+3A_x">x</code></td>
<td>
<p>cv.emBayes result</p>
</td></tr>
<tr><td><code id="print.cv.emBayes_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.cv.emBayes_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Print a list of output from a cv.emBayes object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cv.emBayes">cv.emBayes</a></code>
</p>

<hr>
<h2 id='print.emBayes'>print an emBayes result</h2><span id='topic+print.emBayes'></span>

<h3>Description</h3>

<p>Print a summary of an 'emBayes' result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'emBayes'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.emBayes_+3A_x">x</code></td>
<td>
<p>emBayes result</p>
</td></tr>
<tr><td><code id="print.emBayes_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.emBayes_+3A_...">...</code></td>
<td>
<p>other print arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Print a list of output from a emBayes object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+emBayes">emBayes</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
