<!DOCTYPE html><html><head><title>Help for package EMMIXgene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EMMIXgene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#all_cluster_tissues'><p>Clusters tissues using all group means</p></a></li>
<li><a href='#alon_data'><p>Normalized gene expression values from Alon et al. (1999).</p></a></li>
<li><a href='#cluster_genes'><p>Clusters genes using mixtures of normal distributions</p></a></li>
<li><a href='#cluster_tissues'><p>Clusters tissues</p></a></li>
<li><a href='#EMMIXgene'><p>EMMIXgene:</p></a></li>
<li><a href='#golub_data'><p>Normalized gene expression values from Golub et al. (1999).</p></a></li>
<li><a href='#heat_maps'><p>Heat maps</p></a></li>
<li><a href='#plot_single_gene'><p>Plot a single gene expression histogram with best</p>
fitted mixture of t-distributions.</a></li>
<li><a href='#select_genes'><p>Selects genes using the EMMIXgene algorithm.</p></a></li>
<li><a href='#top_genes_cluster_tissues'><p>Cluster tissues</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Title:</td>
<td>A Mixture Model-Based Approach to the Clustering of Microarray
Expression Data</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides unsupervised selection and clustering of microarray data
    using mixture models. Following the methods described in McLachlan, Bean and
    Peel (2002) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2F18.3.413">doi:10.1093/bioinformatics/18.3.413</a>&gt; a subset of genes are selected
    based one the likelihood ratio statistic for the test of one versus two
    components when fitting mixtures of t-distributions to the expression data
    for each gene. The dimensionality of this gene subset is further reduced through
    the use of mixtures of factor analyzers, allowing the tissue samples to be
    clustered by fitting mixtures of normal distributions.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andrew Thomas Jones &lt;andrewthomasjones@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.3.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, BH</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, stats, mclust, reshape, ggplot2, scales, tools</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-21 01:30:04 UTC; uqajon14</td>
</tr>
<tr>
<td>Author:</td>
<td>Andrew Thomas Jones [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-21 11:22:56 UTC</td>
</tr>
</table>
<hr>
<h2 id='all_cluster_tissues'>Clusters tissues using all group means</h2><span id='topic+all_cluster_tissues'></span>

<h3>Description</h3>

<p>Clusters tissues using all group means
</p>


<h3>Usage</h3>

<pre><code class='language-R'>all_cluster_tissues(gen, clusters, q = 6, G = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="all_cluster_tissues_+3A_gen">gen</code></td>
<td>
<p>EMMIXgene object</p>
</td></tr>
<tr><td><code id="all_cluster_tissues_+3A_clusters">clusters</code></td>
<td>
<p>mclust object</p>
</td></tr>
<tr><td><code id="all_cluster_tissues_+3A_q">q</code></td>
<td>
<p>number of factors if using mfa</p>
</td></tr>
<tr><td><code id="all_cluster_tissues_+3A_g">G</code></td>
<td>
<p>number of components if using mfa</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a clustering for each sample (columns) by each group(rows)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
example &lt;- plot_single_gene(alon_data,1) 
#only run on first 100 genes for speed
alon_sel &lt;- select_genes(alon_data[seq_len(100), ]) 
alon_clust&lt;- cluster_genes(alon_sel , 2)
alon_tissue_all&lt;-all_cluster_tissues(alon_sel, alon_clust, q=1, G=2)
</code></pre>

<hr>
<h2 id='alon_data'>Normalized gene expression values from Alon et al. (1999).</h2><span id='topic+alon_data'></span>

<h3>Description</h3>

<p>A dataset containing centred and normalized values of the 
logged expression values of a subset of 2000 genes taken from 
Alon, Uri, et al. &quot;Broad patterns of gene expression revealed by clustering
analysis of tumor and normal colon tissues probed by oligonucleotide arrays.&quot;
Proceedings of the National Academy of Sciences 96.12 (1999): 6745-6750.
The method of subset selection was described in G. J. McLachlan, R. W. Bean, 
D. Peel; A mixture model-based approach to the clustering of microarray 
expression data ,
Bioinformatics, Volume 18, Issue 3, 1 March 2002, Pages 413–422.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alon_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 rows (genes) and 62 variables (samples).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dim(alon_data)
</code></pre>

<hr>
<h2 id='cluster_genes'>Clusters genes using mixtures of normal distributions</h2><span id='topic+cluster_genes'></span>

<h3>Description</h3>

<p>Sorts genes into clusters using mixtures of normal distributions with
covariance matrices restricted to be multiples of the identity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_genes(gen, g = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_genes_+3A_gen">gen</code></td>
<td>
<p>an EMMIXgene object produced by select_genes().</p>
</td></tr>
<tr><td><code id="cluster_genes_+3A_g">g</code></td>
<td>
<p>The desired number of gene clusters. If not specified will be 
selected automatically on the basis of BIC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An array containing the clustering.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#only run on first 100 genes for speed
alon_sel &lt;- select_genes(alon_data[seq_len(100), ]) 
alon_clust&lt;- cluster_genes(alon_sel , 2)
</code></pre>

<hr>
<h2 id='cluster_tissues'>Clusters tissues</h2><span id='topic+cluster_tissues'></span>

<h3>Description</h3>

<p>Clusters tissues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_tissues(gen, clusters, method = "t", q = 6, G = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_tissues_+3A_gen">gen</code></td>
<td>
<p>EMMIXgene object</p>
</td></tr>
<tr><td><code id="cluster_tissues_+3A_clusters">clusters</code></td>
<td>
<p>mclust object</p>
</td></tr>
<tr><td><code id="cluster_tissues_+3A_method">method</code></td>
<td>
<p>Method for separating tissue classes. Can be either 't' for a 
univariate mixture of t-distributions on gene cluster means, or 'mfa'
for a mixture of factor analyzers.</p>
</td></tr>
<tr><td><code id="cluster_tissues_+3A_q">q</code></td>
<td>
<p>number of factors if using mfa</p>
</td></tr>
<tr><td><code id="cluster_tissues_+3A_g">G</code></td>
<td>
<p>number of components if using mfa</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a clustering for each sample (columns) by each group(rows)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#only run on first 100 genes for speed
alon_sel &lt;- select_genes(alon_data[seq_len(100), ]) 
alon_clust&lt;- cluster_genes(alon_sel,2)
alon_tissue_t&lt;-
   cluster_tissues(alon_sel,alon_clust,method='t')
alon_tissue_mfa&lt;-
   cluster_tissues(alon_sel, alon_clust,method='mfa',q=2,G=2) 
</code></pre>

<hr>
<h2 id='EMMIXgene'>EMMIXgene:</h2><span id='topic+EMMIXgene'></span>

<h3>Description</h3>

<p>Selects genes using the EMMIXgene algorithm, following the methodology of 
G. J. McLachlan, R. W. Bean, D. Peel; A mixture model-based approach to the 
clustering of microarray expression data , Bioinformatics, Volume 18, 
Issue 3, 1 March 2002, Pages 413–422,
https://doi.org/10.1093/bioinformatics/18.3.413
</p>


<h3>Functions</h3>

<p><code><a href="#topic+select_genes">select_genes</a></code>: Selects the most differentially expressed genes.
</p>
<p><code><a href="#topic+cluster_genes">cluster_genes</a></code>: Clusters the genes using a mixture model
approach.
</p>
<p><code><a href="#topic+cluster_tissues">cluster_tissues</a></code>: Clusters the tissues based on the differences 
between the tissue samples among the gene groups.
</p>
<p>See <code>vignette('The-EMMIXgene-Workflow')</code> for more details.
</p>

<hr>
<h2 id='golub_data'>Normalized gene expression values from Golub et al. (1999).</h2><span id='topic+golub_data'></span>

<h3>Description</h3>

<p>A dataset containing the centred and normalized values of the
logged expression values of a subset of 3731 genes taken from Golub,
Todd R., et al. &quot;Molecular classification of cancer: class discovery
and class prediction by gene expression monitoring.&quot; 
Science 286.5439 (1999): 531-537.
The method of subset selection was described in G. J. McLachlan, R. W. Bean,
D. Peel; A mixture model-based approach to the clustering of microarray
expression data ,
Bioinformatics, Volume 18, Issue 3, 1 March 2002, Pages 413–422.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(golub_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 3731 rows (genes) and 72 variables (samples).
#'@examples
dim(golub_data)
</p>

<hr>
<h2 id='heat_maps'>Heat maps</h2><span id='topic+heat_maps'></span>

<h3>Description</h3>

<p>Plot heat maps of gene expression data. Optionally sort the x-axis 
according to a predetermined clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_maps(dat, clustering = NULL, y_lab = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heat_maps_+3A_dat">dat</code></td>
<td>
<p>matrix of gene expression data.</p>
</td></tr>
<tr><td><code id="heat_maps_+3A_clustering">clustering</code></td>
<td>
<p>a vector of sample classifications.
Must be same length as the number of columns in dat.</p>
</td></tr>
<tr><td><code id="heat_maps_+3A_y_lab">y_lab</code></td>
<td>
<p>optional label for y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 heat map.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
example &lt;- heat_maps(alon_data[seq_len(100), ])


</code></pre>

<hr>
<h2 id='plot_single_gene'>Plot a single gene expression histogram with best 
fitted mixture of t-distributions.</h2><span id='topic+plot_single_gene'></span>

<h3>Description</h3>

<p>Plot a single gene expression histogram with best 
fitted mixture of t-distributions according to the EMMIX-gene algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_single_gene(
  dat,
  gene_id,
  g = NULL,
  random_starts = 8,
  max_it = 100,
  ll_thresh = 8,
  min_clust_size = 8,
  tol = 1e-04,
  start_method = "both",
  three = TRUE,
  min = -4,
  max = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_single_gene_+3A_dat">dat</code></td>
<td>
<p>matrix of gene expression data.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_gene_id">gene_id</code></td>
<td>
<p>row number of gene to be plotted.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_g">g</code></td>
<td>
<p>force number of components, default = NULL</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_random_starts">random_starts</code></td>
<td>
<p>The number of random initializations used 
per gene when fitting mixtures of t-distributions. 
Initialization uses k-means by default.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_max_it">max_it</code></td>
<td>
<p>The maximum number of iterations per mixture fit.
Default value is 100.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_ll_thresh">ll_thresh</code></td>
<td>
<p>The difference in -2 log lambda used as a 
threshold for selecting between g=1 and g=2 for each gene. 
Default value is 8, which was chosen arbitrarily in the original paper.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_min_clust_size">min_clust_size</code></td>
<td>
<p>The minimum number of observations per cluster 
used when fitting mixtures of t-distributions for each gene.
Default value is 8.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_tol">tol</code></td>
<td>
<p>Tolerance value used for detecting convergence of EMMIX fits.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_start_method">start_method</code></td>
<td>
<p>Default value is &quot;both&quot;. 
Can also choose &quot;random&quot; for purely random starts.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_three">three</code></td>
<td>
<p>Also test g=2 vs g=3 where appropriate. Defaults to TRUE.</p>
</td></tr>
<tr><td><code id="plot_single_gene_+3A_min">min</code>, <code id="plot_single_gene_+3A_max">max</code></td>
<td>
<p>Minimum and maximum x-axis values for the plot window.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 histogram with fitted t-distributions overlayed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
example &lt;- plot_single_gene(alon_data,1) 
#plot(example)

</code></pre>

<hr>
<h2 id='select_genes'>Selects genes using the EMMIXgene algorithm.</h2><span id='topic+select_genes'></span>

<h3>Description</h3>

<p>Follows the gene selection methodology of 
G. J. McLachlan, R. W. Bean, D. Peel; A mixture model-based approach to the 
clustering of microarray expression data , Bioinformatics, Volume 18, 
Issue 3, 1 March 2002, Pages 413–422,
https://doi.org/10.1093/bioinformatics/18.3.413
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_genes(
  dat,
  filename,
  random_starts = 4,
  max_it = 100,
  ll_thresh = 8,
  min_clust_size = 8,
  tol = 1e-04,
  start_method = "both",
  three = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_genes_+3A_dat">dat</code></td>
<td>
<p>A matrix or dataframe containing gene expression data.
Rows are genes and columns are samples. Must supply one of filename and dat.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_filename">filename</code></td>
<td>
<p>Name of file containing gene data. Can be either .csv 
or space separated .dat. Rows are genes and columns are samples. 
Must supply one of filename and dat.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_random_starts">random_starts</code></td>
<td>
<p>The number of random initializations used per gene when
fitting mixtures of t-distributions. Initialization uses k-means by default.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_max_it">max_it</code></td>
<td>
<p>The maximum number of iterations per mixture fit. 
Default value is 100.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_ll_thresh">ll_thresh</code></td>
<td>
<p>The difference in -2 log lambda used as a threshold 
for selecting between g=1 and g=2 for each gene. Default value is 8,
which was chosen arbitrarily in the original paper.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_min_clust_size">min_clust_size</code></td>
<td>
<p>The minimum number of observations per cluster
used when fitting mixtures of t-distributions for each gene.
Default value is 8.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_tol">tol</code></td>
<td>
<p>Tolerance value used for detecting convergence of EMMIX fits.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_start_method">start_method</code></td>
<td>
<p>Default value is &quot;both&quot;. 
Can also choose &quot;random&quot; for purely random starts.</p>
</td></tr>
<tr><td><code id="select_genes_+3A_three">three</code></td>
<td>
<p>Also test g=2 vs g=3 where appropriate. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An EMMIXgene object containing:
</p>
<table>
<tr><td><code>stat</code></td>
<td>
<p>The difference in log-likelihood for g=1 
and g=2 for each gene (or for g=2 and g=3 where relevant).</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>The selected number of components for each gene.</p>
</td></tr>
<tr><td><code>it</code></td>
<td>
<p>The number of iterations for each genes selected fit.</p>
</td></tr> 
<tr><td><code>selected</code></td>
<td>
<p>An indicator for each genes selected status</p>
</td></tr>
<tr><td><code>ranks</code></td>
<td>
<p>selected gene ids ranked by stat</p>
</td></tr>
<tr><td><code>genes</code></td>
<td>
<p>A dataframe of selected genes.</p>
</td></tr>
<tr><td><code>all_genes</code></td>
<td>
<p>Returns dat or contents of filename.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#only run on first 100 genes for speed
alon_sel &lt;- select_genes(alon_data[seq_len(100), ]) 

</code></pre>

<hr>
<h2 id='top_genes_cluster_tissues'>Cluster tissues</h2><span id='topic+top_genes_cluster_tissues'></span>

<h3>Description</h3>

<p>Cluster tissues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_genes_cluster_tissues(gen, n_top = 100, method = "mfa", q = 2, g = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="top_genes_cluster_tissues_+3A_gen">gen</code></td>
<td>
<p>An EMMIXgene object produced by select_genes().</p>
</td></tr>
<tr><td><code id="top_genes_cluster_tissues_+3A_n_top">n_top</code></td>
<td>
<p>number of top genes (as ranked by likelihood) to be selected</p>
</td></tr>
<tr><td><code id="top_genes_cluster_tissues_+3A_method">method</code></td>
<td>
<p>Method for separating tissue classes. Can be either 't' for a 
univariate mixture of t-distributions on gene cluster means, 
or 'mfa' for a mixture of factor analysers.</p>
</td></tr>
<tr><td><code id="top_genes_cluster_tissues_+3A_q">q</code></td>
<td>
<p>number of factors if using mfa</p>
</td></tr>
<tr><td><code id="top_genes_cluster_tissues_+3A_g">g</code></td>
<td>
<p>number of components if using mfa</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An EMMIXgene object containing:
</p>
<table>
<tr><td><code>stat</code></td>
<td>
<p>A matrix containing clustering (0 or 1) 
for each sample (columns) by each group(rows).</p>
</td></tr>
<tr><td><code>top_gene</code></td>
<td>
<p>The row numbers of the top genes.</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>The fit object used to determine the clustering.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>alon_sel &lt;- select_genes(alon_data[seq_len(100), ])
alon_top_10&lt;-top_genes_cluster_tissues(alon_sel, 10, method='mfa', q=3, g=2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
