<!DOCTYPE html><html lang="en"><head><title>Help for package EncDNA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EncDNA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#APR.Feature'>
<p>Adjacent position relationship feature.</p></a></li>
<li><a href='#Bayes.Feature'>
<p>Projecting nucleotide sequences into numeric feature vectors using Bayes kernel encoding approach.</p></a></li>
<li><a href='#Density.Feature'>
<p>Nucleotide sequence encoding with the distribution of trinucleotides.</p></a></li>
<li><a href='#droso'>
<p>An example dataset consisting of true and false donor splice sites of Drosophila melanogaster.</p></a></li>
<li><a href='#Maldoss.Feature'>
<p>Encoding of nucleic acid sequences using di-nucleotide frequency difference between positive and negative class datasets.</p></a></li>
<li><a href='#MM1.Feature'>
<p>Transforming nucleotide sequences into numeric vectors using first order nucleotide dependency.</p></a></li>
<li><a href='#MM2.Feature'>
<p>Mapping nucleotide sequences onto numeric feature vectors based on second order nucleotide dependencies.</p></a></li>
<li><a href='#MN.Fdtf.Feature'>
<p>Sequence encoding with nucleotide frequency difference between two classes of sequence datasets.</p></a></li>
<li><a href='#PN.Fdtf.Feature'>
<p>Conversion of nucleotide sequences into numeric feature vectors based on the difference of dinucleotide frequency.</p></a></li>
<li><a href='#POS.Feature'>
<p>Transformation of nucleic acid sequences into numeric vectors using position-wise frequency of nucleotides.</p></a></li>
<li><a href='#Predoss.Feature'>
<p>Encoding nucleotide sequences using all possible di-nucleotide dependencies.</p></a></li>
<li><a href='#SAE.Feature'>
<p>Encoding of nucleotide sequences based on sum of absolute error (SAE) of each sequence.</p></a></li>
<li><a href='#Sparse.Feature'>
<p>Nucleotide sequence encoding with 0 and 1.</p></a></li>
<li><a href='#Trint.Dist.Feature'>
<p>Tri-nucleotide distribution-based encoding of nucleotide sequences.</p></a></li>
<li><a href='#WAM.Feature'>
<p>Nucleic acid sequence encoding based on weighted array model.</p></a></li>
<li><a href='#WMM.Feature'>
<p>Weighted matrix model based mapping of nucleotide sequences into vectors of numeric observations.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Encoding of Nucleotide Sequences into Numeric Feature Vectors</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Prabina Kumar Meher</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prabina Kumar Meher &lt;meherprabin@yahoo.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Description:</td>
<td>We describe fifteen different splice site sequence encoding schemes that have been used in earlier studies for mapping of splice site sequences into numeric feature vectors. These encoding schemes will also be helpful for transforming other nucleotide sequences into numeric forms, provided they are of equal length. These encoding schemes will help the computational biologist working in the field of classification (binary or multiclass) or prediction involving nucleic acid sequences of equal length.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-28 05:28:21 UTC; USER</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-28 06:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='APR.Feature'>
Adjacent position relationship feature.
</h2><span id='topic+APR.Feature'></span>

<h3>Description</h3>

<p>This feature was proposed by Li <em>et al</em>.(2012). In fact this is similar to the PN.FDTF encoding scheme (Huang <em>et al</em>., 2006). In this encoding, correlation between adjacent nucleotides  are taken into account.  For any nucleotide sequence with <code class="reqn">n</code> nucleotides, every two consecutive positions between 1 and <code class="reqn">n</code>, i.e., <code class="reqn">(1, 2), (2, 3)... (n-1, n)</code> constitute an APR feature set. For each pair of positions, frequencies of 16 dinucleotides are first computed for both positive and negative dataset, and then the diffence matrix is obtained by substracting the <code class="reqn">16*(n-1)</code> dinucleotide frequency matrix of positive set from that of negative set. The difference matrix is then be used for encoding of nucleotide sequences. In this encoding procedure each sequence with <code class="reqn">n</code> nucleotides can be encoded into a vector of <code class="reqn">(n-1)</code> numeric observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>APR.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="APR.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Nucleotide sequence dataset of positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="APR.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Nucleotide sequence dataset of negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="APR.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Nucleotide sequences to be encoded into numeric feature vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The class DNAStringSet can be obtained by using the function <em>readDNAStringSet</em> avialble in <strong>Biostrings</strong> package Bioconductor. Here, the sequences must be supplied in FASTA format. Both positive and negative datasets are required for this encoding scheme.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-1)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>

<p>Li, J.L., Wang, L.F., Wang, H.Y., Bai, L.Y. and Yuan, Z.M. (2012). High-accuracy splice sites prediction based on sequence component and position features. <em>Genetics and Molecular Research</em>, 11(3): 3432-3451.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PN.Fdtf.Feature">PN.Fdtf.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- APR.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='Bayes.Feature'>
Projecting nucleotide sequences into numeric feature vectors using Bayes kernel encoding approach.
</h2><span id='topic+Bayes.Feature'></span>

<h3>Description</h3>

<p>This sequence encoding technique was introduced by Zhang <em>et al</em>. (2006) for prediction of splice sites. In this encoding technique, positional frequencies of nucleotides are computed for both positive and negative datasets, which are then used for encoding of any nucleotide sequence of same length. Each sequence of length <em>L</em> can be encoded into a numeric feature vector of length <em>2L</em>. Both positive and negative classes of sequences are required for sequence encoding.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bayes.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Bayes.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Nucleotide sequence dataset of positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Bayes.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Nucleotide sequence dataset of negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Bayes.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Nucleotide sequences to be encoded into numeric feature vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The class DNAStringSet can be obtained by using the function <em>readDNAStringSet</em> avialble in <strong>Biostrings</strong> package of Bioconductor. Here, the sequences must be supplied in FASTA format.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*2n</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the sequence length.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>

<p>Zhang, Y., Chu, C., Chen, Y., Zha, H. and Ji, X. (2006). Splice site prediction using support vector machines with a Bayes kernel. <em>Expert Systems with Applications</em>, 30: 73-81.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+POS.Feature">POS.Feature</a></code>, <code><a href="#topic+MN.Fdtf.Feature">MN.Fdtf.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- Bayes.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='Density.Feature'>
Nucleotide sequence encoding with the distribution of trinucleotides.
</h2><span id='topic+Density.Feature'></span>

<h3>Description</h3>

<p>Each nucleotide sequence is encoded into a numeric vector of same length based on the distribution of nucleotides over the sequence. Here, two classes of dataset are not required for encoding, and each sequence is independently encoded instead. This encoding seheme was introduced by Wei <em>et al</em>. (2013) for prediction of donor and acceptor human splice sites along with the <code>MM1.Feature</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Density.Feature(test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Density.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequence dataset to be encoded, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The class DNAStringSet can be obtained by reading FASTA sequences using the function <em>readDNAStringSet</em> avialble in <strong>Biostrings</strong> package of Bioconductor.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*n</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>

<p>Bari, A.T.M.G., Reaz, M.R. and Jeong, B.S. (2014). Effective DNA encoding for splice site prediction using SVM. <em>MATCH Commun. Math. Comput. Chem.</em>, 71: 241-258.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
test &lt;- droso$test
tst &lt;- test[1:5]
enc &lt;- Density.Feature(test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='droso'>
An example dataset consisting of true and false donor splice sites of Drosophila melanogaster. 
</h2><span id='topic+droso'></span>

<h3>Description</h3>

<p>The dataset contains 400 true donor sites, 400 false donor sites and 50 false sites as test set. Each sequence is of 40 nucleotides long. The conserved di-nucleotides GT are present at $21^st$ and $22^nd$ positions respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("droso")</code></pre>


<h3>Format</h3>

<p>The dataset belongs to the class <code>DNAStringSet</code>.
</p>


<h3>Source</h3>

<p>This is a sample dataset which was collected from the dataset originally developed by Reese <em>et al</em>. (1997). The dataset can also be accessed at <a href="http://www.fruitfly.org/sequence/drosophila-datasets.html">http://www.fruitfly.org/sequence/drosophila-datasets.html</a>.
</p>


<h3>References</h3>

<p>Reese, M. G., Eeckman, F. H., Kulp, D. and Haussler, D. (1997). Improved splice site detection in Genie. <em>Journal of Computational Biology</em>, 4(3): 311-323.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
</code></pre>

<hr>
<h2 id='Maldoss.Feature'>
Encoding of nucleic acid sequences using di-nucleotide frequency difference between positive and negative class datasets. 
</h2><span id='topic+Maldoss.Feature'></span>

<h3>Description</h3>

<p>In Maldoss (Meher <em>et al</em>., 2016), the authors propose three encoding approaches namely P1, P2 and P3. Out of these three encoding schemes, the accuracies were reported to be higher for P1 as compared to the other two encoding procedures. Here, we describe the sequence encodng based on P1 only. This P1 encoding approach has similarity with that of PN-FDTF encoding (Huang <em>et al</em>., 2006) approach. The difference is only with respect to the logarithmic transformation in case of <code>Maldoss.Feature</code>. In this encoding procedure, both positive and negative class sequences are required for transformation of nucleotide sequences into numeric vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Maldoss.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Maldoss.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Maldoss.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Maldoss.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For getting an object of class <code>DNAStringSet</code>, the FASTA sequence dataset must be read in R through the function <em>raedDNAStringSet</em> available in <strong>Biostrings</strong> package of Bioconductor (<a href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html">https://bioconductor.org/packages/release/bioc/html/Biostrings.html</a> ).    
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-1)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Meher, P.K., Sahu, T.K. and  Rao, A.R. (2016). Prediction of donor splice sites using random forest with a new sequence encoding approach. <em>BioData Mining</em>, 9.
</p>
</li>
<li><p>Huang, J., Li, T., Chen, K. and Wu, J. (2006). An approach of encoding for prediction of splice sites using SVM. <em>Biochimie</em>, 88(7): 923-929.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+PN.Fdtf.Feature">PN.Fdtf.Feature</a></code>, <code><a href="#topic+MM1.Feature">MM1.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- Maldoss.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='MM1.Feature'>
Transforming nucleotide sequences into numeric vectors using first order nucleotide dependency.
</h2><span id='topic+MM1.Feature'></span>

<h3>Description</h3>

<p>The concept of sequence encoding using Markov model (<code class="reqn">1^{st}</code> order) was introduced by Ho and Rajapakse (2005) for prediction of splice sites. However, this encoding scheme has been comprehensively used by Baten <em>et al</em>. (2006) for prediction of splice sites. In this encoding procedure, first order dependencies between nucleotides in nucleotide sequence are accounted. Only the postive class dataset is used for estimation of dependencies in terms of probabilities, which are then used for encoding. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MM1.Feature(positive_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MM1.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="MM1.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The FASTA sequences should be read into R using the function <em>readDNAStringSet</em> available in <strong>Biostrings</strong> package. This encoding is similar to PN.FDTF feature, as far as the dependency among nucleotides in a sequence is concerned. The only difference is the use of positive class only in stead of both positive and negative classes in PN.FDTF. This encoding approach has similarity with WAM features (Meher <em>et al</em>. 2016) in which the dinucleotide dependencies are considered.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-1)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li> <p>Rajapakse, J. and Ho, L.S. (2005). Markov encoding for detecting signals in genomic sequences. <em>IEEE/ACM Trans Comput Biol Bioinf</em>., 2(2): 131-142.
</p>
</li>
<li> <p>Baten, A., Chang, B., Halgamuge, S. and Li, J. (2006) Splice site identification using probabilistic parameters and SVM classification. <em>BMC Bioinformatics</em>, 7(Suppl 5): S15.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). Identification of donor splice sites using support vector machine: a computational approach based on positional, compositional and dependency features. <em>Algorithms for Molecular Biology</em>, 11(1), 16.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+PN.Fdtf.Feature">PN.Fdtf.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
test &lt;- droso$test
pos &lt;- positive[1:200]
tst &lt;- test
enc &lt;- MM1.Feature(positive_class=pos, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='MM2.Feature'>
Mapping nucleotide sequences onto numeric feature vectors based on second order nucleotide dependencies.
</h2><span id='topic+MM2.Feature'></span>

<h3>Description</h3>

<p>This encoding procedure is similar to the MM1 encoding. The only difference is consideration of second order dependencies unlike first order in <code>MM2.Feature</code>. This technique was first conceptualized by Rajapakse and Ho (2005), and adopted by Maji and Garg (2014). The number of parameters to be estimated in MM2 is 64, which is higher than that of MM1 i.e., 16. Further, only the positive class dataset is used for encoding of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MM2.Feature(positive_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MM2.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="MM2.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For getting an object of class <code>DNAStringSet</code>, the FASTA sequences should be read using the function <em>readDNAStringSet</em> avialble in the <strong>Biostrings</strong> package.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-2)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li> <p>Rajapakse, J. and Ho, L.S. (2005). Markov encoding for detecting signals in genomic sequences. <em>IEEE/ACM Trans Comput Biol Bioinf</em>., 2(2): 131-142.
</p>
</li>
<li> <p>Maji, S. and Garg, D. (2014). Hybrid approach using SVM and MM2 in splice site junction identification. <em>Current Bioinformatics</em>, 9(1): 76-85.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+MM1.Feature">MM1.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
test &lt;- droso$test
pos &lt;- positive[1:200]
tst &lt;- test
enc &lt;- MM2.Feature(positive_class=pos, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='MN.Fdtf.Feature'>
Sequence encoding with nucleotide frequency difference between two classes of sequence datasets.
</h2><span id='topic+MN.Fdtf.Feature'></span>

<h3>Description</h3>

<p>In this encoding procedure, at first, frequency of each nucleotide at each position is computed for both positive and negative classes datasets. Then, the frequency matrix of the positive set is substracted from that of negative set. The sequences are then encoded into numeric vectors after passing them through this difference matrix. So, both positive and negative datasets are necessary for encoding of sequences. This concept was introduced by Huang <em>et al</em>. (2006), and was also used by Pashaei <em>et al</em>. (2016) to generate features for prediction of splice sites along with other features. This has similarity with Bayes kernel encoding (Zhang <em>et al</em>., 2006), where both frequency matrices are used for encoding instead of the difference matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MN.Fdtf.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MN.Fdtf.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="MN.Fdtf.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="MN.Fdtf.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For getting an object of class <code>DNAStringSet</code>, the sequence dataset must be read in FASTA format through the function <em>readDNAStringSet</em> available in the <strong>Biostrings</strong> package of Bioconductor (<a href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html">https://bioconductor.org/packages/release/bioc/html/Biostrings.html</a> ).
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*n</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the sequence length.
</p>


<h3>Note</h3>

<p>This feature does not take into consideration the dependencies among nucleotides in the sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li> <p>Zhang, Y., Chu, C., Chen, Y., Zha, H. and Ji, X. (2006). Splice site prediction using support vector machines with a Bayes kernel. <em>Expert Systems with Applications</em>, 30: 73-81.
</p>
</li>
<li><p>Huang, J., Li, T., Chen, K. and Wu, J. (2006). An approach of encoding for prediction of splice sites using SVM. <em>Biochimie</em>, 88(7): 923-929.
</p>
</li>
<li><p>Pashaei, E., Yilmaz, A., Ozen, M. and Aydin, N. (2016). Prediction of splice site using AdaBoost with a new sequence encoding approach. <em>In Systems, Man, and Cybernetics (SMC), IEEE International Conference</em>, pp 3853-3858.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+WMM.Feature">WMM.Feature</a></code>, <code><a href="#topic+Bayes.Feature">Bayes.Feature</a></code>, <code><a href="#topic+PN.Fdtf.Feature">PN.Fdtf.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- MN.Fdtf.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='PN.Fdtf.Feature'>
Conversion of nucleotide sequences into numeric feature vectors based on the difference of dinucleotide frequency.
</h2><span id='topic+PN.Fdtf.Feature'></span>

<h3>Description</h3>

<p>Dinucleotide frequency matrix is first computed for both positive and negative classes. Then, frequency matrix of the positive class is substracted from that of negative class. The sequences are then passed through this difference matrix to encode them into numeric feature vectors. Similar to the <code>MN.Fdtf feature</code>, both positive and negative classes are necessary for encoding of nucleotide sequences. This was also conceptualized by Huang <em>et al</em>. (2006). This has also been used by Pashaei <em>et al</em>. (2016) as one of the features for prediction of splice sites along with the other features. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PN.Fdtf.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PN.Fdtf.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="PN.Fdtf.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="PN.Fdtf.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For getting an object of class <code>DNAStringSet</code>, the sequence dataset must be read in FASTA format through the function <em>readDNAStringSet</em> available in <strong>Biostrings</strong> package of Bioconductor (<a href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html">https://bioconductor.org/packages/release/bioc/html/Biostrings.html</a> ).
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-1)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the sequence length.
</p>


<h3>Note</h3>

<p>Both positive and negative classes datasets are essential for the encoding. This feature has similarity with that of <code>MM1.Feature</code> and <code>WAM.Feature</code> with respect to the first order dependency. Unlike <code>MN.Fdtf.Feature</code>, this feature takes into account the first order dependencies of nucleotides in the sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Huang, J., Li, T., Chen, K. and Wu, J. (2006). An approach of encoding for prediction of splice sites using SVM. <em>Biochimie</em>, 88(7): 923-929.
</p>
</li>
<li><p>Pashaei, E., Yilmaz, A., Ozen, M. and Aydin, N. (2016). Prediction of splice site using AdaBoost with a new sequence encoding approach. <em>In Systems, Man, and Cybernetics (SMC), IEEE International Conference</em>, pp 3853-3858.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+MN.Fdtf.Feature">MN.Fdtf.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code>, <code><a href="#topic+MM1.Feature">MM1.Feature</a></code>,  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- PN.Fdtf.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='POS.Feature'>
Transformation of nucleic acid sequences into numeric vectors using position-wise frequency of nucleotides.
</h2><span id='topic+POS.Feature'></span>

<h3>Description</h3>

<p>This encoding scheme was devised by Li <em>et al</em>. (2012). Frequencies of 4 nucleotides are first computed at each position for both positive and negative datasets, resulting in two <code class="reqn">4*L</code> probability tables for the two classes for sequence length <code class="reqn">L</code>. A <code class="reqn">4*L</code> statistical difference table is obtained by elementwise substraction of the two probability distribution tables, which is then used for encoding of sequences. Further, as per sparse encoding, the nucleotides A, T, G and C can be encoded as (1,0,0,0), (0,1,0,0), (0,0,1,0) and (0,0,0,1) respectively. The value 1 of sparse encoding is then replaced with the difference values obtained from the difference table for encoding nucleotide at each postion. Thus, it can be said that POS feature encoding is a blending of MN-FDTF (Huang <em>et al</em>., 2006) and Sparse encoding (Meher <em>et al</em>., 2016) technique. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>POS.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="POS.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="POS.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="POS.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>DNAstringSet</code> object can be obtained by reading the sequences in FASTA format using the function <em>readDNAStringSet</em>available in the <strong>Biostrings</strong> package of Bioconductor.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*4n</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Note</h3>

<p>In this encoding procedure, dependencies of nucleotides are not taken into consideration. Both positive and negative datasets are required for encoding of nucleotide sequences. Each sequence of length <code class="reqn">L</code> can be transformed into a numeric vector of length <code class="reqn">4*L</code> with this encoding technique.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Huang, J., Li, T., Chen, K. and Wu, J. (2006). An approach of encoding for prediction of splice sites using SVM. <em>Biochimie</em>, 88(7): 923-929.
</p>
</li>
<li><p>Li, J.L., Wang, L.F., Wang, H.Y., Bai, L.Y., Yuan, Z.M. (2012). High-accuracy splice sites prediction based on sequence component and position features. <em>Genetics and Molecular Research</em>, 11(3): 3432-3451.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). A computational approach for prediction of donor splice sites with improved accuracy. <em>Journal of Theoretical Biology</em>, 404: 285-294.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+MN.Fdtf.Feature">MN.Fdtf.Feature</a></code>, <code><a href="#topic+Bayes.Feature">Bayes.Feature</a></code>, <code><a href="#topic+WMM.Feature">WMM.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- POS.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='Predoss.Feature'>
Encoding nucleotide sequences using all possible di-nucleotide dependencies.
</h2><span id='topic+Predoss.Feature'></span>

<h3>Description</h3>

<p>In this encoding, not only the adjecent dependencies are considered, but also the association that exists among non-adjacent nucleotides. In MM1, PN.FDTF features, only the dependencies between adjacent nucleotides are taken into account. Though all possible pair-wise dependencies are first introduced by Meher <em>et al</em>. (2014) for predicting splice sites through probablistic approach, the same authors further used this association to encode the splice site dataset for prediction using machine learning classifiers (Meher <em>et al</em>., 2016).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Predoss.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Predoss.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Predoss.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="Predoss.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be of an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This encoding approach will be helpful for transformation of nucleotide sequences into numeric feature vectors, which can subsequently be used as input in several supervised learning models for classification.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*n^{2}</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Note</h3>

<p>Dimension of the feature space will increase geometrically with increase in the length of the sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2014). A statistical approach for 5' splice site prediction using short sequence motifs and without encoding sequence data. <em>BMC Bioinformatics</em>, 15(1), 362.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). A computational approach for prediction of donor splice sites with improved accuracy. <em>Journal of Theoretical Biology</em>, 404: 285-294.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- Predoss.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='SAE.Feature'>
Encoding of nucleotide sequences based on sum of absolute error (SAE) of each sequence.
</h2><span id='topic+SAE.Feature'></span>

<h3>Description</h3>

<p>The sum of absolute error (SAE) concept was introduced by Meher <em>et al</em>. (2014) for prediction of donor splice sites, and was subsequently used by the same authors (Meher <em>et al</em>., 2016) for encoding of splice site motif for prediction using supervised learning model. In this encoding technique also all possible pair-wise nucleotide dependencies are considered.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SAE.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SAE.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="SAE.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="SAE.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this encoding approach a vector of two observations will be obtained for each sequence. This two values correspond to the values obtained, when only positive class and both positive &amp; neagtive datasets are used for encoding. This encoding scheme is invariant to the length of the sequence. Thus, both positive and negative classes datasets are required for encoding of sequence.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*2</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code>. 
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2014). A statistical approach for 5' splice site prediction using short sequence motifs and without encoding sequence data. <em>BMC Bioinformatics</em>, 15(1), 362.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). Identification of donor splice sites using support vector machine: a computational approach based on positional, compositional and dependency features. <em>Algorithms for Molecular Biology</em>, 11(1), 16.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+MM1.Feature">MM1.Feature</a></code>, <code><a href="#topic+WAM.Feature">WAM.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- SAE.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='Sparse.Feature'>
Nucleotide sequence encoding with 0 and 1.
</h2><span id='topic+Sparse.Feature'></span>

<h3>Description</h3>

<p>In this encoding approach A, T, G and C are encoded as (1,1,1), (1,0,0), (0,1,0) and (0,0,1). This was introduced by Golam Bari <em>et al</em>. (2014). Besides, each nucleotide can also be encoded with four bits i.e., A as (1,0,0,0), T as (0,1,0,0), G as (0,0,1,0) and C as (0,0,0,1) as followed in Meher <em>et al</em>. (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sparse.Feature(test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Sparse.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequence dataset to be encoded into numeric vector containing 0 and 1, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each sequence is encoded independently, without the need of positive and negative classes datasets. 
</p>


<h3>Value</h3>

<p>A vector of length <code class="reqn">4*n</code> for sequence of <code class="reqn">n</code> nucleotides long in <code>test_seq</code>.
</p>


<h3>Note</h3>

<p>For larger sequence length, high dimensional feature vector will be generated.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Bari, A.T.M.G., Reaz, M.R. and Jeong, B.S. (2014). Effective DNA encoding for splice site prediction using SVM. <em>MATCH Commun. Math. Comput. Chem</em>., 71: 241-258.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). A computational approach for prediction of donor splice sites with improved accuracy. <em>Journal of Theoretical Biology</em>, 404: 285-294.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
test &lt;- droso$test
tst &lt;- test
enc &lt;- Sparse.Feature(test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='Trint.Dist.Feature'>
Tri-nucleotide distribution-based encoding of nucleotide sequences.
</h2><span id='topic+Trint.Dist.Feature'></span>

<h3>Description</h3>

<p>This encoding scheme was first time adopted by Wei <em>et al</em>. (2013) for prediction of splice sites along with MM1 features. In this encoding technique, distribution of trinucleotides are taken into consideration independently for the exon and intron regions of splice site motifs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Trint.Dist.Feature(test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Trint.Dist.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequence dataset to be transformed into numeric feature vectors. There should be atleat two sequences, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This encoding scheme is independent of positive and negative datasets. In other words, each sequence can be encoded independently. Further, nucleotide sequence of any length will be transformed into a numeric vector of 64 observations corresponding to 64 combinations of trinucleotides.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*64</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code>.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>

<p>Wei, D., Zhang, H., Wei, Y. and Jiang, Q. (2013). A novel splice site prediction method using support vector machine. <em>J Comput Inform Syst</em>., 920: 8053-8060.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
test &lt;- droso$test
tst &lt;- test
enc &lt;- Trint.Dist.Feature(test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='WAM.Feature'>
Nucleic acid sequence encoding based on weighted array model.
</h2><span id='topic+WAM.Feature'></span>

<h3>Description</h3>

<p>Unlike weighted matrix method (WMM), first order nucleotide dependencies are accounted in weighted array model (WAM). The WAM was introduced by Zhang and Marr (1993) for locating splicing signal on nuclotide sequences. The WAM was employed by Meher <em>et al</em>. (2016) for encoding of splice site motifs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WAM.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WAM.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="WAM.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="WAM.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this encoding approach, a vector of two observations will be obtained for each sequence, corresponds to the situation when only positive class and both positive &amp; neagtive datasets are used for encoding. This encoding scheme is also invariant to the length of the sequence.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*2</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code>. 
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Zhang, M. and Marr, T. (1993). A weight array method for splicing signal analysis. <em>Comput Appl Biosci</em>., 9(5): 499-509.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). Identification of donor splice sites using support vector machine: a computational approach based on positional, compositional and dependency features. <em>Algorithms for Molecular Biology</em>, 11(1): 16. 
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+MM1.Feature">MM1.Feature</a></code>, <code><a href="#topic+PN.Fdtf.Feature">PN.Fdtf.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- WAM.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

<hr>
<h2 id='WMM.Feature'> 
Weighted matrix model based mapping of nucleotide sequences into vectors of numeric observations.
</h2><span id='topic+WMM.Feature'></span>

<h3>Description</h3>

<p>The weighted matrix model (WMM) was developed by Staden (1984) for prediction of splice sites. In this technique, position weight matrix are computed from the aligned positive dataset which is then used for computing the liklihood of any nucleotide sequence being a positive or negative class. The position weight matrix was later on used by Meher et al. (2016) for encoding of splice site motifs for prediction using supervised learning models. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WMM.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WMM.Feature_+3A_positive_class">positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="WMM.Feature_+3A_negative_class">negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
<tr><td><code id="WMM.Feature_+3A_test_seq">test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vector, must be an object of class <code>DNAStringSet</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this encoding approach, a vector of two observations will be obtained for each sequence, corresponds to the situation when only positive class and both positive and neagtive datasets are used for encoding. This encoding scheme is also invariant to the length of the sequence.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*2</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code>. 
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li><p>Staden, R. (1984). Computer methods to locate signals in nucleic acid sequences. <em>Nucleic Acids Research</em>, 12: 505-519.
</p>
</li>
<li><p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). Identification of donor splice sites using support vector machine: a computational approach based on positional, compositional and dependency features. <em>Algorithms for Molecular Biology</em>, 11(1), 16. 
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+Bayes.Feature">Bayes.Feature</a></code>, <code><a href="#topic+MN.Fdtf.Feature">MN.Fdtf.Feature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- WMM.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
