<!DOCTYPE html><html lang="en"><head><title>Help for package EnsCat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EnsCat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alphadata'><p>Alphaherpesvirinae virus genome sequence data</p></a></li>
<li><a href='#Benhc'><p>Performs bootstrap ensemble hierarchical clustering for categorical data.</p></a></li>
<li><a href='#cancer'><p>Primary tumor domain (cancer) data</p></a></li>
<li><a href='#CTN'><p>convert genetic data (nucleotides) to numerical values</p></a></li>
<li><a href='#ebola'><p>Ebolavirus genome sequence data</p></a></li>
<li><a href='#enhcHi'><p>Performs ensemble hierarchical clustering for high dimensional categorical data</p></a></li>
<li><a href='#EnsCat'>
<p>This package includes several methods that can be used to cluster categorical data.</p></a></li>
<li><a href='#ggdplot'>
<p>Nice plots of hierarchical clustering results via ggdendrogram</p></a></li>
<li><a href='#hammingD'><p>Calculate the hamming distance between data points.</p></a></li>
<li><a href='#kmodes'><p>Run Kmodes</p></a></li>
<li><a href='#lympho'><p>Lymphography domian (lympho) data</p></a></li>
<li><a href='#mush'><p>Mushroom data</p></a></li>
<li><a href='#rhabdodata'><p>Rhabdoviridae virus genome sequence data</p></a></li>
<li><a href='#soybean'><p>Soybean (small) data</p></a></li>
<li><a href='#tangle'>
<p>Generate a tanglegram from two hierarchical clusterings of a data set</p></a></li>
<li><a href='#USFlag'><p>United States Flag Privately-Owned Merchant Fleet Data</p></a></li>
<li><a href='#zoo'><p>zoo data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Clustering of Categorical Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-1-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Saeid Amiri, Bertrand Clarke and Jennifer Clarke.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Saeid Amiri &lt;saeid.amiri1@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>stats, utils, graphics, dendextend, ggplot2, ggdendro, seqinr,
R (&ge; 3.3.2)</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of the clustering methods of categorical data
    discussed in Amiri, S., Clarke, B., and Clarke, J. (2015). Clustering categorical 
    data via ensembling dissimilarity matrices.  Preprint &lt;<a href="https://doi.org/10.48550/arXiv.1506.07930">doi:10.48550/arXiv.1506.07930</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/jlp2duke/EnsCat/wiki/How-To-with-Examples">https://github.com/jlp2duke/EnsCat/wiki/How-To-with-Examples</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-02-01 00:37:12 UTC; saeidamiri</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-02-01 01:39:44</td>
</tr>
</table>
<hr>
<h2 id='alphadata'>Alphaherpesvirinae virus genome sequence data
</h2><span id='topic+alphadata'></span>

<h3>Description</h3>

<p>A dataset consisting of whole genome sequences for viruses from the family Alphaherpesvirinae
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("alphadata")</code></pre>


<h3>Format</h3>

<p>This dataset is a matrix of dimension 98x359883 that represents the sequences of 98 viral genomes from the subfamily Alphaherpesvirinae.
</p>


<h3>Details</h3>

<p>This data includes whole genome sequences of viruses belonging to the subfamily Alphaherpesvirinae. Alphaherpesvirinae is  a subfamily of the Herpesviridae family of viruses that cause diseases in humans and animals. The data is downloaded from ViPR, <a href="http://www.viprbrc.org">http://www.viprbrc.org</a>, and are aligned using &quot;MAFFT&quot;, see Katoh et a. (2013), and saved in &quot;alphadata&quot;. Alphaherpesvirinae has five genera: Iltovirus (IIt), Mardivirus (Mar), Scutavirus, Simplexvirus (Sim), and Vari-cellovirus (Var). The viruses were collected from different hosts, namely, human, monkey, chicken, turkey, duck, cow, bat, equidae, boar, cat, amazona oratrix (denoted hum, mon, chi, tur, duc, cow, bat, equ, boa, cat, aor). The codes in the example show the labels.
</p>


<h3>References</h3>

<p>Katoh, K., and D.M. Standley (2013). MAFFT multiple sequence alignment software version 7: improvements in performance and usability. Molecular biology and evolution, 30(4), 772-780.
</p>
<p>Pickett, BE et al. (2012). ViPR: an open bioinformatics database and analysis resource for virology research. Nucleic Acids Research 40: D593-8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### load Alphaherpesvirinae data
#data("alphadata")
### the following codes define the labels of the data by genera and host.
#xlab1&lt;-NULL
#xlab1[1:8]&lt;-"Var-boa";xlab1[9:13]&lt;-"Var-hum";xlab1[14]&lt;-"Var-cat"
#xlab1[15:32]&lt;-"Var-equ";xlab1[33]&lt;-"Var-mon";xlab1[34:40]&lt;-"Var-cow"
#xlab1[41:45]&lt;-"Sim-mon";xlab1[46:47]&lt;-"Sim-mon";xlab1[48:58]&lt;-"Sim-hum"
#xlab1[59]&lt;-"Sim-bat";xlab1[60]&lt;-"Sim-mon";xlab1[61]&lt;-"Mar-tur"
#xlab1[62:71]&lt;-"Mar-chi";xlab1[72:78]&lt;-"Mar-duc";xlab1[79]&lt;-"Ilt-ora";xlab1[80:98]&lt;-"Ilt-chi"
</code></pre>

<hr>
<h2 id='Benhc'>Performs bootstrap ensemble hierarchical clustering for categorical data.
</h2><span id='topic+Benhc'></span>

<h3>Description</h3>

<p>This function performs a bootstrap ensemble hierarchical clustering of categorical data,
as described in details below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Benhc(x, En)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Benhc_+3A_x">x</code></td>
<td>
<p>A nxp data matrix or data frame; n is the number of observations and p is the number of dimensions.</p>
</td></tr>
<tr><td><code id="Benhc_+3A_en">En</code></td>
<td>
<p>Number of clusterings to include in the ensemble, i.e., cardinality of the ensemble.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function 'Benhc' generates a dissimilarity matrix via the bootstrap ensemble. The
ensembled dissimilarity matrix is generated using the same procedure as described for
the function &lsquo;enhc&rsquo; except that each clustering is based on a bootstrap sample of the data.
The number of clusters for each clustering is selected randomly from {2,...,sqrt(n)}.
</p>


<h3>References</h3>

<p>Amiri, S., Clarke, B., and Clarke, J. (2015). Clustering categorical data via ensembling dissimilarity matrices. arXiv preprint arXiv:1506.07930.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data('zoo')
### zoo includes the zoo data downloaded from UCI
### Machine Learning Repository
### Calculate ensemble dissimilarities with 150 ensemble members
#disten&lt;-Benhc(zoo$obs,En=150)
### This function performs a hierarchical cluster analysis using
### dissimilarities obtained by the ensembling procedure in Benhc
#en&lt;-hclust(disten,method='average')
### A plot of the dendrogram can be generated by
#plot(en,label=zoo$lab)
</code></pre>

<hr>
<h2 id='cancer'>Primary tumor domain (cancer) data
</h2><span id='topic+cancer'></span>

<h3>Description</h3>

<p>Classification data set from the UCI Machine Learning Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("cancer")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;cancer$obs&quot; includes the observations that are stored as numerical values.
&quot;cancer$lab&quot; contains the labels of the data.
</p>


<h3>Details</h3>

<p>A simple classification data set containing 16 attributes with 339
observations. Since the true labels are known, this data can be used to evaluate clustering methods.
</p>


<h3>Source</h3>

<p>Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Primary+Tumor">https://archive.ics.uci.edu/ml/datasets/Primary+Tumor</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(cancer)
</code></pre>

<hr>
<h2 id='CTN'>convert genetic data (nucleotides) to numerical values
</h2><span id='topic+CTN'></span>

<h3>Description</h3>

<p>This function converts genetic data (nucleotides) to numeric data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CTN(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CTN_+3A_x">x</code></td>
<td>
<p>x should be a dataset in fasta format
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>R is more efficient with numerical data and storage of data via numerical values takes
less memory. Genetic data consists of nucleotide data A,T,C,G and are usually saved
in Fasta format. After downloading the data from one of the bioinformatics repositories
and importing it to R, this function converts the data to numerical values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### import fasta data to R.
##x.dna0 &lt;- read.fasta("dna.fasta")
###  convert data to numerical values
##x.dna&lt;-CTN(x.dna0)
</code></pre>

<hr>
<h2 id='ebola'>Ebolavirus genome sequence data
</h2><span id='topic+ebola'></span>

<h3>Description</h3>

<p>A dataset consisting of whole genome sequences for Ebolavirus  from the family Filovirdae.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ebola")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;ebola$obs&quot; includes the observations that are stored as numerical values.
&quot;ebola$lab&quot; contains the labels of the data.
</p>
<p>This ebola$obs is a matrix of dimension 103x26445 that represents the sequences of 103 viral genomes from the Ebolavirus.
</p>


<h3>Details</h3>

<p>This data includes whole genome sequences of viruses belonging to the Ebolavirus. Ebolavirus
is  a subfamily of the Filovirdae family of viruses. The data is downloaded from ViPR, <a href="http://www.viprbrc.org">http://www.viprbrc.org</a>, and are aligned using &quot;MAFFT&quot;, see Katoh et a. (2013), and saved in &quot;ebola&quot;. Ebolavirus subdivides into five species: Bundibugyo virus (Bun), Reston ebolavirus (Res), Sudan ebolavirus (Sud), Tai Forest ebolavirus (Tai), and Zaire ebolavirus (Zai).
The hosts are human, monkey, swine, guinea pig, mouse, and bat (denoted hum, mon, swi, gpi, mou, bat, respectively,  in our dataset).  The ebola$lab in the example show the labels, the combination of species and host.
</p>


<h3>References</h3>

<p>Katoh, K., and D.M. Standley (2013). MAFFT multiple sequence alignment software version 7: improvements in performance and usability. Molecular biology and evolution, 30(4), 772-780.
</p>
<p>Pickett, BE et al. (2012). ViPR: an open bioinformatics database and analysis resource for virology research. Nucleic Acids Research 40: D593-8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### load Ebolavirus data
#data("ebola")
</code></pre>

<hr>
<h2 id='enhcHi'>Performs ensemble hierarchical clustering for high dimensional categorical data
</h2><span id='topic+enhcHi'></span>

<h3>Description</h3>

<p>This function performs an ensemble hierarchical clustering of high dimensional categorical
data (p &gt;&gt; n).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enhcHi(data, En=100, len=c(2,10), type=2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enhcHi_+3A_data">data</code></td>
<td>
<p>A nxp data matrix of data frame; n is the number of observations and p is the number of features or dimensions.</p>
</td></tr>
<tr><td><code id="enhcHi_+3A_en">En</code></td>
<td>
<p>Number of clusterings to include in the ensemble, i.e., cardinality of the ensemble.
</p>
</td></tr>
<tr><td><code id="enhcHi_+3A_len">len</code></td>
<td>
<p>Range of sizes of clusterings (i.e., number of clusters) to run and ensemble.</p>
</td></tr>
<tr><td><code id="enhcHi_+3A_type">type</code></td>
<td>
<p>Numeric indicator of single bootstrap (type=1) or double bootstrap (type=2)
for selecting subsets of variables to include in each clustering within the ensemble. The
default is type=2</p>
</td></tr>
</table>


<h3>References</h3>

<p>Amiri, S., Clarke, B., and Clarke, J. (2015). Clustering categorical data via ensembling dissimilarity matrices. arXiv preprint arXiv:1506.07930.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data("rhabdodata")
### The following code generates the dissimilary matrix of sequence data stored in alphadata
### The ensemble has 100 member clusterings, and the number of clusters in each clustering
### is generated randomly from a discrete uniform on (2,10). A double bootstrap procedure is
### used to select a subset of variables for each clustering.
#ens&lt;-enhcHi(rhabdodata$dat,En=100,len=c(2,10), type=2)
### Calculate the hamming distance
#dis0&lt;-hammingD(ens)
### Save as distance format
#REDIST&lt;-as.dist(dis0)
#hc0 &lt;- hclust(REDIST,method = "average")
#plot(hc0,label=rhabdodata$lab,hang =-1)
</code></pre>

<hr>
<h2 id='EnsCat'>
This package includes several methods that can be used to cluster categorical data.
</h2><span id='topic+EnsCat'></span>

<h3>Description</h3>

<p>EnsCat implements several methods for clustering of categorical data.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> EnsCat</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-01-29</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> &gt;=GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> https://github.com/jlp2duke/EnsCat/wiki/How-To-with-Examples
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Maintainer: Saeid Amiri &lt;saeid.amiri1@gmail.com&gt;
</p>


<h3>References</h3>

<p>Amiri, S., Clarke, B., and Clarke, J. (2015). Clustering categorical data via ensembling dissimilarity matrices. arXiv preprint arXiv:1506.07930.
</p>

<hr>
<h2 id='ggdplot'>
Nice plots of hierarchical clustering results via ggdendrogram
</h2><span id='topic+ggdplot'></span>

<h3>Description</h3>

<p>This function provides two different plotting options for either a dendro object or an
object that can be coerced to class dendro, using the function ggdendrogram from the
package <code>ggdendro</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggdplot(hc, lab = NULL, ptype = 1, title = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggdplot_+3A_hc">hc</code></td>
<td>
<p>Either a dendro object or an object that can be coerced to class dendro using
the <code><a href="ggdendro.html#topic+dendro_data">dendro_data</a></code> function, i.e. objects of class dendrogram, hclust or tree</p>
</td></tr>
<tr><td><code id="ggdplot_+3A_lab">lab</code></td>
<td>
<p>A character vector of labels for the leaves of the tree. By default labels
in hc are used.</p>
</td></tr>
<tr><td><code id="ggdplot_+3A_ptype">ptype</code></td>
<td>
<p>A numeric indicator of the type of plot desired. If <code>ptype==1</code>, the plot is
rotated by 90 degrees, the theme is blank, and the title is plotted. If <code>ptype!=1</code>, the
plot is rotated by 90 degrees, and the background is a grayscale grid.</p>
</td></tr>

<tr><td><code id="ggdplot_+3A_title">title</code></td>
<td>
<p>A character label for the title of the plot. Only used if <code>ptype!=1</code>.</p>
</td></tr>
<tr><td><code id="ggdplot_+3A_...">...</code></td>
<td>
<p>other parameters passed to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given either a dendro object or an object that can be coerced to class dendro, this is a
convenience function for plotting. For an object of type dendro, if <code>ptype==1</code>, the function
executes the equivalent of
</p>
<p><code>ggdendrogram(hcdata, rotate=TRUE, size=2) + labs(title="Dendrogram in ggplot2")</code>
</p>
<p>If <code>ptype!=1</code>, the function executes the equivalent of
</p>
<p><code>ggdendrogram(hcdata, rotate = TRUE, theme_dendro = FALSE)</code>
</p>
<p>Objects that are not of class dendro are coerced to class dendro prior to plotting.
</p>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ggplot2)
hc &lt;- hclust(dist(USArrests), "ave")
p&lt;-ggdplot(hc, ptype=2)
</code></pre>

<hr>
<h2 id='hammingD'>Calculate the hamming distance between data points.
</h2><span id='topic+hammingD'></span>

<h3>Description</h3>

<p>Hamming distance is defined on categorical vectors.  It counts the number of
times the coordinates in two data vectors differ, or the number of substitutions required
to convert one data vector into the other. Here the Hamming distance is normalized, so the
result is the number of coordinates that differ divided by the vector length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hammingD(dat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hammingD_+3A_dat">dat</code></td>
<td>
<p>dat should be a matrix or data frame of data. n is the number of observations (rows)
and p is the number of dimensions (columns).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the Hamming distance (normalized) between rows of the input data.
</p>


<h3>Value</h3>

<p>The result is a nxn matrix whose (i,j) element is the Hamming distance between rows i and j
</p>


<h3>See Also</h3>

<p>See Also as <a href="#topic+alphadata">alphadata</a>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### The running is time consuming
### Run hamming distance
#dis0&lt;-hammingD(alphadata)
### Save as distance format
#REDIST&lt;-as.dist(dis0)
### Run a hierarchical clustering using average linkage
#hc0 &lt;- hclust(REDIST,method = "average")
### plot the dendrogram
#plot(hc0,label=xlab1,hang =-1)
</code></pre>

<hr>
<h2 id='kmodes'>Run Kmodes
</h2><span id='topic+kmodes'></span>

<h3>Description</h3>

<p>This function runs Kmodes.  The user must choose the number of clusters and the initial modes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kmodes(data, k, k2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kmodes_+3A_data">data</code></td>
<td>
<p> data should be a matrix or data frame, columns include the variables.
</p>
</td></tr>
<tr><td><code id="kmodes_+3A_k">k</code></td>
<td>
<p> number of clusters
</p>
</td></tr>
<tr><td><code id="kmodes_+3A_k2">k2</code></td>
<td>
<p> set of initial modes; indices of data points
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function clusters the rows of the data.
</p>


<h3>References</h3>

<p>Huang, Z. (1998). Extensions to the v-means Algorithm for Clustering Large Data Sets with Categorical Values, Data Mining and Knowledge Discovery, 2, 283-304.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("zoo")
### Run Kmodes on zoo data with 7 clusters and the first seventh observations as initial modes
kmodes(zoo$obs,k=7,1:7)
### Run Kmodes with seven random initial modes selected from data points
kmodes(zoo$obs,k=7,sort(sample(dim(zoo$obs)[1],7)))
</code></pre>

<hr>
<h2 id='lympho'>Lymphography domian (lympho) data
</h2><span id='topic+lympho'></span>

<h3>Description</h3>

<p>Classification data set from the UCI Machine Learning Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("cancer")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;lympho$obs&quot; includes the observations that are stored as numerical values.
&quot;lympho$lab&quot; contains the labels of the data.
</p>


<h3>Details</h3>

<p>A simple classification data set containing 18 attributes with 148
observations. Since the true labels are known, this data can be used to evaluate clustering methods.
</p>


<h3>Source</h3>

<p>Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Lymphography">https://archive.ics.uci.edu/ml/datasets/Lymphography</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lympho)
</code></pre>

<hr>
<h2 id='mush'>Mushroom data
</h2><span id='topic+mush'></span>

<h3>Description</h3>

<p>Classification data set from the UCI Machine Learning Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mush")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;mush$obs&quot; includes the observations that are stored as numerical values.
&quot;mush$lab&quot; contains the labels of the data.
</p>


<h3>Details</h3>

<p>A simple classification data set containing 22 attributes with 8124
observations, because the dataset is large, we only used the last 400 observations in our analysis. Since the true labels are known, this data can be used to evaluate clustering methods.
</p>


<h3>Source</h3>

<p>Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Mushroom">https://archive.ics.uci.edu/ml/datasets/Mushroom</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mush)
</code></pre>

<hr>
<h2 id='rhabdodata'>Rhabdoviridae virus genome sequence data
</h2><span id='topic+rhabdodata'></span>

<h3>Description</h3>

<p>This dataset consists of whole genome sequences of viruses from the Family Rhabdoviridae.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("rhabdodata")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $dat and $lab. The components dat includes
a matrix of dimension 53x26035 that represents the sequences of 53 viral genomes from the
family Rhabdoviridae. The component lab includes labels for each sample that include
abbreviations of the relevant genus and viral host.
</p>


<h3>Details</h3>

<p>This data includes whole genome sequences of viruses belonging to the subfamily Rhabdoviridae.
Rhabdoviridae is a family of viruses with single-stranded RNA genomes that are able to
infect a wide variety of hosts, both plants and animals. cause diseases in humans and
animals. The data is downloaded from ViPR, <a href="http://www.viprbrc.org">http://www.viprbrc.org</a>, and are aligned
using &quot;MAFFT&quot;, see Katoh et a. (2013), and saved in &quot;rhabdodata&quot;. Rhabdoviridae has twelve
genera of which nine are repesented here: Cytorhabdovirus, Ephemerovirus, Novirhabdovirus,
Nucleorhabdovirus, Perhabdovirus, Sigmavirus, Sprivivirus, Tibrovirus, and Tupavirus.
The viruses were collected from different hosts, namely, Alfalfa, Cattle, Drosophila, Eel,
Fish, Garlic, Midge, Mosquito, Eggplant, Taro, Trout, and Unknown.
</p>


<h3>References</h3>

<p>Katoh, K., and D.M. Standley (2013). MAFFT multiple sequence alignment software version 7: improvements in performance and usability. Molecular biology and evolution, 30(4), 772-780.
</p>
<p>Pickett, BE et al. (2012). ViPR: an open bioinformatics database and analysis resource for virology research. Nucleic Acids Research 40: D593-8.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### load Rhabdoviridae data
data("rhabdodata")
### the following codes define the labels of the data by genera and host.
dim(rhabdodata$dat)
#[1]	53	26035

</code></pre>

<hr>
<h2 id='soybean'>Soybean (small) data
</h2><span id='topic+soybean'></span>

<h3>Description</h3>

<p>Classification data set from the UCI Machine Learning Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("soybean")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;soybean$obs&quot; includes the observations that are stored as numerical values.
&quot;soybean$lab&quot; contains the labels of the data.
</p>


<h3>Details</h3>

<p>A simple classification data set containing 35 attributes with 47
observations. Since the true labels are known, this data can be used to evaluate clustering methods.
</p>


<h3>Source</h3>

<p>Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Soybean+(Small)">https://archive.ics.uci.edu/ml/datasets/Soybean+(Small)</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(soybean)
</code></pre>

<hr>
<h2 id='tangle'>
Generate a tanglegram from two hierarchical clusterings of a data set
</h2><span id='topic+tangle'></span>

<h3>Description</h3>

<p>This function generates a tanglegram of two different hierarchical clusterings of the same
dataset. This is essentially a convenience wrapper for the function <code><a href="dendextend.html#topic+tanglegram">tanglegram</a></code> in
the package <code>dendextend</code>; see Galili (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tangle(hc0, hc1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tangle_+3A_hc0">hc0</code></td>
<td>

<p>An object that can be coerced to a dendrogram, e.g., an object from <code><a href="stats.html#topic+hclust">hclust</a></code>. This
object will be plotted on the left side of the tanglegram.
</p>
</td></tr>
<tr><td><code id="tangle_+3A_hc1">hc1</code></td>
<td>

<p>An object that can be coerced to a dendrogram, e.g., an object from <code><a href="stats.html#topic+hclust">hclust</a></code>. This
object will be plotted on the right side of the tanglegram.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a convenience wrapper for the function <code><a href="dendextend.html#topic+tanglegram">tanglegram</a></code> in the R package
<code>dendextend</code>; see <a href="http://cran.at.r-project.org/web/packages/dendextend/">http://cran.at.r-project.org/web/packages/dendextend/</a>.
A tanglegram is used to visualize the similarities and differences between two different
hierarchical clusterings of the same dataset.
</p>


<h3>Value</h3>

<p>An invisible <code><a href="dendextend.html#topic+dendlist">dendlist</a></code>, with two trees after being modified during the creation of the tanglegram
</p>


<h3>References</h3>

<p><a href="https://cran.r-project.org/package=dendextend">https://cran.r-project.org/package=dendextend</a>, <a href="https://github.com/talgalili/dendextend/">https://github.com/talgalili/dendextend/</a>,
<a href="http://www.r-statistics.com/tag/dendextend/">http://www.r-statistics.com/tag/dendextend/</a>, <a href="http://bioinformatics.oxfordjournals.org/content/31/22/3718">http://bioinformatics.oxfordjournals.org/content/31/22/3718</a>
al Galili (2015). dendextend: an R package for visualizing, adjusting, and comparing
trees of hierarchical clustering. Bioinformatics. doi:10.1093/bioinformatics/btv428
</p>


<h3>See Also</h3>

<p><code><a href="dendextend.html#topic+tanglegram">tanglegram</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
function (hc0, hc1)
{
    hcd0 &lt;- as.dendrogram(hc0)
    hcd1 &lt;- as.dendrogram(hc1)
    hcd0 &lt;- match_order_by_labels(hcd0, hcd1)
    dends_0_1 &lt;- dendlist(hcd0, hcd1)
    t &lt;- tanglegram(dends_0_1)
    t
  }
</code></pre>

<hr>
<h2 id='USFlag'>United States Flag Privately-Owned Merchant Fleet Data
</h2><span id='topic+USFlag'></span>

<h3>Description</h3>

<p>This dataset includes 10 categorical variables that describe U.S. flag privately owned merchant
fleet vessels, based on data provided by the United Stated Department of Transportation
Maritime Administration (MARAD).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("USFlag")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $lab and $obs. The component $lab contains
a categorical indicator of Ship Type (see below). The component $obs includes
a matrix of dimension 170x10 that contains categorical data on 170 United States flag privately
owned merchant fleet vessels. The columns are as follows:
</p>
<p>Ship.Type
a categorical variable with 5 levels, Containership [1], Dry Bulk [2], General Cargo [3],
Ro-Ro [4], and Tanker [5], indicating the ship type. This is identical to $lab.
</p>
<p>Gross.Tonnage
a categorical variable with 6 levels, &lt;20000GT [1], 20000-40000GT [2], 40000-60000GT [3],
60000-80000GT [4],80000-100000GT [5], &gt;100000GT [6], indicating the ship gross tonnage
</p>
<p>Deadweight
a categorical variable with 6 levels, &lt;20000DWT [1], 20000-40000DWT [2], 40000-60000DWT [3],
60000-100000DWT [4], 100000-140000DWT [5], &gt;140000DWT [6], indicating the ship deadweight
</p>
<p>Year.Built
a categorical variable with 6 levels, &lt;1960 [1], 1961-1980 [2], 1981-1990 [3], 1991-2000 [4],
2001-2010 [5], and &gt;2010 [6], indicating the year of completion of ship construction
</p>
<p>Operator
a categorical variable with 49 levels indicating the operator of the ship
</p>
<p>MSP
a binary variable indicating whether the ship is [1] or is not [0] part of the maritime
security program
</p>
<p>VISA
a binary variable indicating whether the ship is [1] or is not [0] part of the
Voluntary Intermodal Sealift Agreement
</p>
<p>VTA
a binary variable indicating whether the ship is [1] or is not [0] part of the
Voluntary Tanker Agreement
</p>
<p>Jones.Act.Eligible
a binary variable indicating whether the ship is [1] or is not [0] Jones Act Eligible.
These vessels are eligible to participate in domestic trade. Jones Act eligible vessels
are built in the United States, owned by United States citizens and crewed by U.S. Mariners
</p>
<p>Militarily.Useful
a binary variable indicating whether the ship is [1] or is not [0] considered a militarily
useful sealift vessel
</p>
<p>For more information on these definitions please see IHS Maritime, Sea-Web. www.sea-web.com
</p>


<h3>Details</h3>

<p>This data includes categorical variables that describe U.S. flag privately owned merchant
fleet vessels. Information is provided only for oceangoing, self-propelled,
cargo-carrying vessels of 1,000 gross tons and above. These data are based on information
from the U.S. Department of Transportation Maritime Administration (MARAD) as of 3/3/2015,
obtained from the MARAD Open Data Portal
(http://www.marad.dot.gov/resources/data-statistics/).
</p>


<h3>Source</h3>

<p>United States Maritime Administration (MARAD) Open Data Portal. http://www.marad.dot.gov/resources/data-statistics/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### load USFlag maritime data
data("USFlag")
### the following codes define the labels of the data by genera and host.
dim(USFlag$obs)
#[1] 170  10
length(USFlag$lab)
#[1] 170

</code></pre>

<hr>
<h2 id='zoo'>zoo data
</h2><span id='topic+zoo'></span>

<h3>Description</h3>

<p>Classification data set from the UCI Machine Learning Repository
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("zoo")</code></pre>


<h3>Format</h3>

<p>The format of the data is a list with components $obs and
$lab.
&quot;zoo$obs&quot; includes the observations that are stored as numerical values.
&quot;zoo$lab&quot; contains the labels of the data.
</p>


<h3>Details</h3>

<p>A simple classification data set containing 17 Boolean-valued attributes with 101
observations. Since the true labels are known, this data can be used to evaluate clustering methods.
</p>


<h3>Source</h3>

<p>Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
</p>
<p><a href="https://archive.ics.uci.edu/ml/datasets/Zoo">https://archive.ics.uci.edu/ml/datasets/Zoo</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(zoo)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
