<!DOCTYPE html><html lang="en"><head><title>Help for package ensembleTax</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ensembleTax}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assign.ensembleTax'><p>Computes ensemble taxonomic assignments for each ASV in an amplicon data set</p></a></li>
<li><a href='#bayes.sample'><p>Example output of dada2 assignTaxonomy function</p></a></li>
<li><a href='#bayestax2df'><p>Converts the output of DADA2's assignTaxonomy, which implements a naive</p>
bayesian classifier, into a dataframe compatible with the algorithms used in
ensembleTax</a></li>
<li><a href='#gg_13_8_train_set_97'><p>All unique taxonomic assignments from the GreenGenes v13.8 clusted at 97%</p></a></li>
<li><a href='#idtax.pr2.sample'><p>Example output of DECIPHER idtaxa function with pr2 taxonomy</p></a></li>
<li><a href='#idtax.silva.sample'><p>Example output of DECIPHER idtaxa function with silva taxonomy</p></a></li>
<li><a href='#idtax2df'><p>Converts outputs of DECIPHER's idtaxa algorithm into a dataframe compatible</p>
with the algorithms used in ensembleTax.</a></li>
<li><a href='#pr2v4.12.0'><p>All unique taxonomic assignments from the pr2 reference database v4.12.0</p></a></li>
<li><a href='#rdp_train_set_16'><p>All unique taxonomic assignments from the RDP Train Set 16</p></a></li>
<li><a href='#rubric.sample'><p>Example rubric with ASV-identifying data</p></a></li>
<li><a href='#silva.nr.v138'><p>All unique taxonomic assignments from the Silva SSU nr database v138</p></a></li>
<li><a href='#sort_my_taxtab'><p>Sorts taxonomy table by ASV-identifying columns.</p></a></li>
<li><a href='#synonyms_v2'><p>Taxonomic synonyms searched by the taxmapper algorithm</p></a></li>
<li><a href='#taxmapper'><p>Maps an input taxonomy table onto a different taxonomic nomenclature.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Ensemble Taxonomic Assignments of Amplicon Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Dylan Catlett [aut, cre], Kevin Son [ctb], Connie Liang [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dylan Catlett &lt;dcat4444@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Creates ensemble taxonomic assignments of amplicon sequencing data 
  in R using outputs of multiple taxonomic assignment algorithms and/or
	reference databases. Includes flexible algorithms for mapping taxonomic 
	nomenclatures onto one another and for computing ensemble taxonomic 
	assignments. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, Biostrings, DECIPHER, stringr, ggplot2, reshape2,
usethis</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-21 15:28:14 UTC; dylancatlett</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-21 16:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='assign.ensembleTax'>Computes ensemble taxonomic assignments for each ASV in an amplicon data set</h2><span id='topic+assign.ensembleTax'></span>

<h3>Description</h3>

<p>Computes ensemble taxonomic assignments for each ASV in an amplicon data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign.ensembleTax(
  x,
  tablenames = names(x),
  ranknames = c("kingdom", "supergroup", "division", "class", "order", "family",
    "genus", "species"),
  weights = rep(1, length(x)),
  tiebreakz = NULL,
  count.na = TRUE,
  assign.threshold = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="assign.ensembleTax_+3A_x">x</code></td>
<td>
<p>A list of dataframes of type character or list (no factors) that
contain an arbitrary number of meta-data columns (e.g. ASV sequences or
numbers), and other columns named according to ranknames that include
taxonomic assignments for each ASV in the data set</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_tablenames">tablenames</code></td>
<td>
<p>A character vector of the names of each taxonomy table
provided in x. Default is names(x)</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_ranknames">ranknames</code></td>
<td>
<p>The names of ranks (columns) of the taxonomy tables included
in x. These are used to track ASV-identifying data through the ensemble
calculations.</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_weights">weights</code></td>
<td>
<p>A numeric vector with length = length(x) that specifies
relative weights to the taxonomic assignments in the corresponding element of
x. Default is a vector with all elements =1 to specify equal weighting of
all taxonomy tables assignments. All values must be integers.</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_tiebreakz">tiebreakz</code></td>
<td>
<p>NULL is the default. Alternatively, a character vector
containing the tablenames in order of priority to be used as a tie-breaker
in the event that multiple taxonomic names are found at equal (weighted)
highest frequencies (above assign.threshold).</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_count.na">count.na</code></td>
<td>
<p>TRUE or FALSE indicating whether you would like NA
assignments considered in the ensemble calculation. TRUE considers NA
assignments, FALSE does not consider NA assignments. assign.threshold is
implemented differently depending on whether this is TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="assign.ensembleTax_+3A_assign.threshold">assign.threshold</code></td>
<td>
<p>A number between 0 and 1 that indicates the
(weighted) proportion at which a particular taxonomic name must be assigned
in the input taxonomy tables in order to be assigned to the ensemble
taxonomic assignment. When count.na=FALSE, proportions are calculated only
relative to the number of tables with no NA assignments. When count.na=TRUE,
proportions are calculated relative to the sum of the weights argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algorithm takes as input a list of taxonomy tables (dataframes of type
character or list; no factors) and assumes rows correspond to ASVs/OTUs and
columns correspond to taxonomic assignments at ranks listed in descending
order in the input ranknames. All taxonomy tables should follow the same
taxonomic nomenclature (naming and ranking conventions), should include
ASV/OTU-identifying columns (e.g. ASV sequences or a column of asv numbers,
etc), and each row of each taxonomy table should represent the same ASV/OTU.
Use of the functions bayestax2df, idtax2df, and/or taxmapper will ensure your
taxonomy tables meet these requirements. Be advised that rownames of each
taxonomy table are set to NULL by assign.ensembleTax.
</p>
<p>Ensemble taxonomic assignments are computed by finding the highest-frequency
taxonomic assignment for each ASV across all input taxonomy tables. Several
parameters can be controlled by the user to weight the assignments of
specific taxonomy tables more highly than others (weights), to favor
assignments by a specific table in the event that multiple assignments are
found at the same (weighted) highest frequency (tiebreakz), to set a
(weighted) frequency threshold above which a taxonomic assignment must be
found to be assigned in the ensemble (assign.threshold), and finally to
ignore non-assignments signalled by NA in the frequency and assignment
computations (count.na).
</p>
<p>The output is a dataframe of ASVs and corresponding ensemble taxonomic
assignments.
</p>


<h3>Value</h3>

<p>a dataframe containing ensemble taxonomic assignments
</p>


<h3>Author(s)</h3>

<p>Dylan Catlett
</p>
<p>Kevin Son
</p>


<h3>See Also</h3>

<p>idtax2df, bayestax2df, taxmapper
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fake1.pr2 &lt;- data.frame(ASV = c("AAAA", "ATCG", "GCGC", "TATA", "TCGA"),
         kingdom = c("Eukaryota", "Eukaryota", "Eukaryota", "Eukaryota",
         "Eukaryota"),
         supergroup = c(NA, "Stramenopiles", "Rhizaria", "Stramenopiles",
         "Alveolata"),
         division = c(NA, "Ochrophyta", "Radiolaria", "Opalozoa",
         "Dinoflagellata"),
         class = c(NA, "Bacillariophyta", "Polycystinea", "MAST-12",
         "Syndiniales"),
         order = c(NA, "Bacillariophyta_X", "Collodaria", "MAST-12A", NA),
         family = c(NA, "Polar-centric-Mediophyceae", "Collophidiidae", NA,
         NA),
         genus = c(NA, NA, "Collophidium", NA, NA),
         species = as.character(c(NA, NA, NA, NA, NA)),
         stringsAsFactors = FALSE)
fake2.pr2 &lt;- data.frame(ASV = c("AAAA", "ATCG", "GCGC", "TATA", "TCGA"),
         kingdom = c("Eukaryota", "Eukaryota", "Eukaryota", "Eukaryota",
         "Eukaryota"),
         supergroup = c(NA, "Stramenopiles", "Rhizaria", "Stramenopiles",
         "Alveolata"),
         division = c(NA, "Opalozoa", "Radiolaria", "Opalozoa",
         "Dinoflagellata"),
         class = c(NA, NA, "Polycystinea", NA, "Dinophycese"),
         order = c(NA, NA, "Collodaria", NA, NA),
         family = c(NA, NA, "Collophidiidae", NA, NA),
         genus = c(NA, NA, "Collophidium", NA, NA),
         species = as.character(c(NA, NA, NA, NA, NA)),
         stringsAsFactors = FALSE)
head(fake1.pr2)
head(fake2.pr2)
xx &lt;- list(fake1.pr2, fake2.pr2)
names(xx) &lt;- c("fake1", "fake2")
xx
eTax &lt;- assign.ensembleTax(xx,
           tablenames = names(xx),
           ranknames = c("kingdom", "supergroup", "division","class","order",
           "family","genus","species"),
           tiebreakz = NULL,
           count.na=TRUE,
           assign.threshold = 0.5,
           weights=rep(1,length(xx)))
head(eTax)
eTax &lt;- assign.ensembleTax(xx,
                    tablenames = names(xx),
                    ranknames = c("kingdom", "supergroup", "division",
                    "class","order","family","genus","species"),
                    tiebreakz = NULL,
                    count.na=FALSE,
                    assign.threshold = 0.5,
                    weights=c(2,1))
head(eTax)

</code></pre>

<hr>
<h2 id='bayes.sample'>Example output of dada2 assignTaxonomy function</h2><span id='topic+bayes.sample'></span>

<h3>Description</h3>

<p>Example output of dada2 assignTaxonomy function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bayes.sample
</code></pre>


<h3>Format</h3>

<p>^ list with 2 elements
</p>

<dl>
<dt>tax</dt><dd><p>taxonomic assignments</p>
</dd>
<dt>boot</dt><dd><p>bootstrap confidence estimates</p>
</dd>
</dl>


<hr>
<h2 id='bayestax2df'>Converts the output of DADA2's assignTaxonomy, which implements a naive
bayesian classifier, into a dataframe compatible with the algorithms used in
ensembleTax</h2><span id='topic+bayestax2df'></span>

<h3>Description</h3>

<p>Converts the output of DADA2's assignTaxonomy, which implements a naive
bayesian classifier, into a dataframe compatible with the algorithms used in
ensembleTax
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bayestax2df(
  tt,
  db = "pr2",
  ranks = NULL,
  boot = 0,
  rubric = NULL,
  return.conf = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bayestax2df_+3A_tt">tt</code></td>
<td>
<p>The taxonomy table output by DADA2's assignTaxonomy function.</p>
</td></tr>
<tr><td><code id="bayestax2df_+3A_db">db</code></td>
<td>
<p>The database you ran assignTaxonomy against. Either &quot;pr2&quot;, &quot;silva&quot;,
&quot;rdp&quot;, or &quot;gg&quot; are supported. You may set to NULL and include a character
vector of rank (column) names for other databases.</p>
</td></tr>
<tr><td><code id="bayestax2df_+3A_ranks">ranks</code></td>
<td>
<p>NULL, or a character vector of column names if db is set to NULL</p>
</td></tr>
<tr><td><code id="bayestax2df_+3A_boot">boot</code></td>
<td>
<p>The bootstrap threshold below which taxonomic assignments should
be set to NA. This can also be done with DADA2's assignTaxonomy but is
included here for convenience.</p>
</td></tr>
<tr><td><code id="bayestax2df_+3A_rubric">rubric</code></td>
<td>
<p>NULL, or a DNAStringSet (see Biostrings package) with ASV
sequences named by your preferred ASV identifier. Both the ASV sequence and
identifier will be merged with the output dataframe. If NULL, ASV-identifying
data are excluded in the output dataframe.</p>
</td></tr>
<tr><td><code id="bayestax2df_+3A_return.conf">return.conf</code></td>
<td>
<p>If TRUE, returns a list where the first element is your
formatted taxonomy table and the second element is a dataframe of bootstrap
confidence values. If FALSE, your formatted taxonomy table is returned as a
dataframe.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For consistency with dada2's assignTaxonomy function, when used with
Silva, RDP, or GreenGenes it subsamples the ranks c(&quot;domain&quot;, &quot;phylum&quot;,
&quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;). Set db = NULL and supply ranks for
databases that aren't directly supported. If a rubric is supplied with
ASV-identifying meta-data (this is highly recommended), the output taxonomy
table is sorted by the (first returned column of) ASV-identifying data.
</p>


<h3>Value</h3>

<p>a dataframe formatted for use with taxmapper and/or ensembleTax
</p>


<h3>Author(s)</h3>

<p>Dylan Catlett
</p>
<p>Connie Liang
</p>


<h3>See Also</h3>

<p>idtax2df, ensembleTax, taxmapper
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("bayes.sample")
data("rubric.sample")
head(bayes.sample)
head(rubric.sample)
df &lt;- bayestax2df(tt = bayes.sample, db = "pr2", boot = 0, rubric = NULL,
return.conf = FALSE)
head(df)
df &lt;- bayestax2df(tt = bayes.sample, db = "pr2", boot = 0,
rubric = rubric.sample, return.conf = FALSE)
head(df)
df &lt;- bayestax2df(tt = bayes.sample, db = "pr2", boot = 60,
rubric = rubric.sample, return.conf = FALSE)
head(df)
df &lt;- bayestax2df(tt = bayes.sample, db = "pr2", boot = 60,
rubric = rubric.sample, return.conf = TRUE)
head(df)

</code></pre>

<hr>
<h2 id='gg_13_8_train_set_97'>All unique taxonomic assignments from the GreenGenes v13.8 clusted at 97%</h2><span id='topic+gg_13_8_train_set_97'></span>

<h3>Description</h3>

<p>All unique taxonomic assignments from the GreenGenes v13.8 clusted at 97%
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_13_8_train_set_97
</code></pre>


<h3>Format</h3>

<p>^ dataframe with 4163 rows and 7 columns
</p>

<dl>
<dt>domain</dt><dd><p>domain assignments</p>
</dd>
<dt>phylum</dt><dd><p>phylum assignments</p>
</dd>
<dt>class</dt><dd><p>class assignments</p>
</dd>
<dt>order</dt><dd><p>order assignments</p>
</dd>
<dt>family</dt><dd><p>family assignments</p>
</dd>
<dt>genus</dt><dd><p>genus assignments</p>
</dd>
<dt>species</dt><dd><p>genus assignments</p>
</dd>
</dl>


<hr>
<h2 id='idtax.pr2.sample'>Example output of DECIPHER idtaxa function with pr2 taxonomy</h2><span id='topic+idtax.pr2.sample'></span>

<h3>Description</h3>

<p>Example output of DECIPHER idtaxa function with pr2 taxonomy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idtax.pr2.sample
</code></pre>


<h3>Format</h3>

<p>^ list with 5 elements
</p>

<dl>
<dt>1</dt><dd><p>tax data for ASV 1</p>
</dd>
<dt>2</dt><dd><p>tax data for ASV 2</p>
</dd>
<dt>3</dt><dd><p>tax data for ASV 3</p>
</dd>
<dt>4</dt><dd><p>tax data for ASV 4</p>
</dd>
<dt>5</dt><dd><p>tax data for ASV 5</p>
</dd>
</dl>


<hr>
<h2 id='idtax.silva.sample'>Example output of DECIPHER idtaxa function with silva taxonomy</h2><span id='topic+idtax.silva.sample'></span>

<h3>Description</h3>

<p>Example output of DECIPHER idtaxa function with silva taxonomy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idtax.silva.sample
</code></pre>


<h3>Format</h3>

<p>^ list with 5 elements
</p>

<dl>
<dt>1</dt><dd><p>tax data for ASV 1</p>
</dd>
<dt>2</dt><dd><p>tax data for ASV 2</p>
</dd>
<dt>3</dt><dd><p>tax data for ASV 3</p>
</dd>
<dt>4</dt><dd><p>tax data for ASV 4</p>
</dd>
<dt>5</dt><dd><p>tax data for ASV 5</p>
</dd>
</dl>


<hr>
<h2 id='idtax2df'>Converts outputs of DECIPHER's idtaxa algorithm into a dataframe compatible
with the algorithms used in ensembleTax.</h2><span id='topic+idtax2df'></span>

<h3>Description</h3>

<p>Converts outputs of DECIPHER's idtaxa algorithm into a dataframe compatible
with the algorithms used in ensembleTax.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idtax2df(
  tt,
  db = "pr2",
  ranks = NULL,
  boot = 0,
  rubric = NULL,
  return.conf = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="idtax2df_+3A_tt">tt</code></td>
<td>
<p>The taxonomy table output by DECIPHER's idtaxa algorithm</p>
</td></tr>
<tr><td><code id="idtax2df_+3A_db">db</code></td>
<td>
<p>The database you ran idtaxa against. Either &quot;pr2&quot;, &quot;silva&quot;, &quot;rdp&quot;,
or &quot;gg&quot; are supported.</p>
</td></tr>
<tr><td><code id="idtax2df_+3A_ranks">ranks</code></td>
<td>
<p>NULL, or a character vector of column names if db is set to NULL</p>
</td></tr>
<tr><td><code id="idtax2df_+3A_boot">boot</code></td>
<td>
<p>The bootstrap threshold below which taxonomic assignments should
be set to NA. This can also be done with DECIPHER's idtaxa but is included
here for convenience.</p>
</td></tr>
<tr><td><code id="idtax2df_+3A_rubric">rubric</code></td>
<td>
<p>a DNAStringSet (see Biostrings package) with ASV sequences
named by your preferred ASV identifier. Both the ASV sequence and identifier
will be merged with the output dataframe. If NULL, ASV-identifying data is
not included in the output dataframe.</p>
</td></tr>
<tr><td><code id="idtax2df_+3A_return.conf">return.conf</code></td>
<td>
<p>If TRUE, returns a list where the first element is your
formatted taxonomy table and the second element is a dataframe of bootstrap
confidence values. If FALSE, your formatted taxonomy table is returned as a
dataframe.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For consistency with DADA2's assignTaxonomy function, when used with
Silva, RDP, or GreenGenes it subsamples the ranks c(&quot;domain&quot;, &quot;phylum&quot;,
&quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;). Set db = NULL and supply ranks for
databases that aren't directly supported. The output taxonomy table is sorted
by the ASV-identifying data supplied in the rubric.
</p>
<p>CAUTION: the idtaxa algorithm does not return any ASV-identifying data in its
output &quot;taxon&quot; object. The elements of tt should thus be supplied in the same
order as the elements in rubric. This will typically be the case so long as
there is no tampering with the rubric or taxon object in between implementing
idtaxa and their use here.
</p>


<h3>Value</h3>

<p>a dataframe formatted for use with taxmapper and/or ensembleTax
</p>


<h3>Author(s)</h3>

<p>Dylan Catlett
</p>
<p>Connie Liang
</p>


<h3>See Also</h3>

<p>bayestax2df, ensembleTax, taxmapper
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("idtax.pr2.sample")
data("rubric.sample")
head(idtax.pr2.sample)
head(rubric.sample)
df &lt;- idtax2df(tt = idtax.pr2.sample, db = "pr2", ranks = NULL, boot = 0,
rubric = NULL, return.conf = FALSE)
head(df)
df &lt;- idtax2df(tt = idtax.pr2.sample, db = "pr2", ranks = NULL, boot = 0,
rubric = rubric.sample, return.conf = FALSE)
head(df)
df &lt;- idtax2df(tt = idtax.pr2.sample, db = "pr2", ranks = NULL, boot = 60,
rubric = rubric.sample, return.conf = FALSE)
head(df)
df &lt;- idtax2df(tt = idtax.pr2.sample, db = "pr2", ranks = NULL, boot = 60,
rubric = rubric.sample, return.conf = TRUE)
head(df)

</code></pre>

<hr>
<h2 id='pr2v4.12.0'>All unique taxonomic assignments from the pr2 reference database v4.12.0</h2><span id='topic+pr2v4.12.0'></span>

<h3>Description</h3>

<p>All unique taxonomic assignments from the pr2 reference database v4.12.0
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pr2v4.12.0
</code></pre>


<h3>Format</h3>

<p>^ dataframe with 45352 rows and 8 columns
</p>

<dl>
<dt>kingdom</dt><dd><p>kingdom assignments</p>
</dd>
<dt>supergroup</dt><dd><p>supergroup assignments</p>
</dd>
<dt>division</dt><dd><p>division assignments</p>
</dd>
<dt>class</dt><dd><p>class assignments</p>
</dd>
<dt>order</dt><dd><p>order assignments</p>
</dd>
<dt>family</dt><dd><p>family assignments</p>
</dd>
<dt>genus</dt><dd><p>genus assignments</p>
</dd>
<dt>species</dt><dd><p>species assignments</p>
</dd>
</dl>


<hr>
<h2 id='rdp_train_set_16'>All unique taxonomic assignments from the RDP Train Set 16</h2><span id='topic+rdp_train_set_16'></span>

<h3>Description</h3>

<p>All unique taxonomic assignments from the RDP Train Set 16
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdp_train_set_16
</code></pre>


<h3>Format</h3>

<p>^ dataframe with 2472 rows and 6 columns
</p>

<dl>
<dt>domain</dt><dd><p>domain assignments</p>
</dd>
<dt>phylum</dt><dd><p>phylum assignments</p>
</dd>
<dt>class</dt><dd><p>class assignments</p>
</dd>
<dt>order</dt><dd><p>order assignments</p>
</dd>
<dt>family</dt><dd><p>family assignments</p>
</dd>
<dt>genus</dt><dd><p>genus assignments</p>
</dd>
</dl>


<hr>
<h2 id='rubric.sample'>Example rubric with ASV-identifying data</h2><span id='topic+rubric.sample'></span>

<h3>Description</h3>

<p>Example rubric with ASV-identifying data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rubric.sample
</code></pre>


<h3>Format</h3>

<p>^ DNAStringSet with 5 elements
</p>

<dl>
<dt>sv1</dt><dd><p>sample ASV 1</p>
</dd>
<dt>sv2</dt><dd><p>sample ASV 2</p>
</dd>
<dt>sv3</dt><dd><p>sample ASV 3</p>
</dd>
<dt>sv4</dt><dd><p>sample ASV 4</p>
</dd>
<dt>sv5</dt><dd><p>sample ASV 5</p>
</dd>
</dl>


<hr>
<h2 id='silva.nr.v138'>All unique taxonomic assignments from the Silva SSU nr database v138</h2><span id='topic+silva.nr.v138'></span>

<h3>Description</h3>

<p>All unique taxonomic assignments from the Silva SSU nr database v138
</p>


<h3>Usage</h3>

<pre><code class='language-R'>silva.nr.v138
</code></pre>


<h3>Format</h3>

<p>^ dataframe with 6011 rows and 6 columns
</p>

<dl>
<dt>domain</dt><dd><p>domain assignments</p>
</dd>
<dt>phylum</dt><dd><p>phylum assignments</p>
</dd>
<dt>class</dt><dd><p>class assignments</p>
</dd>
<dt>order</dt><dd><p>order assignments</p>
</dd>
<dt>family</dt><dd><p>family assignments</p>
</dd>
<dt>genus</dt><dd><p>genus assignments</p>
</dd>
</dl>


<hr>
<h2 id='sort_my_taxtab'>Sorts taxonomy table by ASV-identifying columns.</h2><span id='topic+sort_my_taxtab'></span>

<h3>Description</h3>

<p>Sorts taxonomy table by ASV-identifying columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_my_taxtab(tt, ranknames)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort_my_taxtab_+3A_tt">tt</code></td>
<td>
<p>A taxonomy table supplied as a dataframe (no factors)</p>
</td></tr>
<tr><td><code id="sort_my_taxtab_+3A_ranknames">ranknames</code></td>
<td>
<p>A character vector of the names of columns of tt that
contain taxonomic assignments. tt is sorted by columns not included in
ranknames.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A helper function for the ...2df family of pre-processing functions.
If multiple columns are available to sort, it uses the left-most column.
</p>


<h3>Value</h3>

<p>a dataframe sorted by the columns specified in ranknames
</p>


<h3>Author(s)</h3>

<p>Dylan Catlett
</p>


<h3>See Also</h3>

<p>bayestax2df, idtax2df
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("bayes.sample")
data("rubric.sample")
bayes.pretty &lt;- bayestax2df(bayes.sample, rubric = rubric.sample)
sort_my_taxtab(bayes.pretty,
ranknames = c("kingdom", "supergroup", "division", "class", "order",
"family", "genus", "species"))

</code></pre>

<hr>
<h2 id='synonyms_v2'>Taxonomic synonyms searched by the taxmapper algorithm</h2><span id='topic+synonyms_v2'></span>

<h3>Description</h3>

<p>Taxonomic synonyms searched by the taxmapper algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synonyms_v2
</code></pre>


<h3>Format</h3>

<p>^ dataframe with 174 rows and 11 columns
</p>

<dl>
<dt>Name_1</dt><dd><p>first synonym</p>
</dd>
<dt>Name_2</dt><dd><p>second synonym</p>
</dd>
<dt>Name_3</dt><dd><p>third synonym</p>
</dd>
<dt>Name_4</dt><dd><p>fourth synonym</p>
</dd>
<dt>Name_5</dt><dd><p>fifth synonym</p>
</dd>
<dt>Name_6</dt><dd><p>sixth synonym</p>
</dd>
<dt>Name_7</dt><dd><p>seventh synonym</p>
</dd>
<dt>References</dt><dd><p>Reference for some synonyms</p>
</dd>
<dt>Notes.References</dt><dd><p>Notes from references</p>
</dd>
<dt>X</dt><dd><p>Additional references for some synonyms</p>
</dd>
<dt>X.1</dt><dd><p>Additional references for some synonyms</p>
</dd>
</dl>


<hr>
<h2 id='taxmapper'>Maps an input taxonomy table onto a different taxonomic nomenclature.</h2><span id='topic+taxmapper'></span>

<h3>Description</h3>

<p>Maps an input taxonomy table onto a different taxonomic nomenclature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxmapper(
  tt,
  tt.ranks = colnames(tt),
  tax2map2 = "pr2",
  exceptions = c("Archaea", "Bacteria"),
  ignore.format = FALSE,
  synonym.file = "default",
  streamline = TRUE,
  outfilez = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxmapper_+3A_tt">tt</code></td>
<td>
<p>The input taxonomy table you would like to map onto a new
taxonomic nomenclature. Should be a dataframe of type char or list (no
factors).</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_tt.ranks">tt.ranks</code></td>
<td>
<p>A character vector of the column names where taxonomic
names are found in tt. Supply them heirarchically (e.g. kingdom &ndash;&gt; species)</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_tax2map2">tax2map2</code></td>
<td>
<p>The taxonomic nomenclature you would like to map onto. pr2
v4.12.0, Silva SSU v138 nr, GreenGenes v13.8 clustered at 97% similarity, and
the RDP train set v16 are included in the ensembleTax package. You can map to
these by specifying &quot;pr2&quot;, &quot;Silva&quot;, &quot;gg&quot;, or &quot;rdp&quot;. Otherwise should be a
dataframe of type character or list (no factors) with each column
corresponding to a taxonomic rank.</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_exceptions">exceptions</code></td>
<td>
<p>A character vector of taxonomic names at the basal/root
rank of tt that will be propagated onto the mapped taxonomy. ASVs assigned
to these names will retain these names at their basal/root rank in the mapped
taxonomy. All other ranks are assigned NA.</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_ignore.format">ignore.format</code></td>
<td>
<p>If TRUE, the algorithm modifies taxonomic names in tt to
account for common variations in taxonomic name syntax and/or formatting
commonly encountered in reference databases (e.g. Pseudo-nitzschia will map
to Pseudonitzschia). If FALSE, formatting issues may preclude mapping of
synonymous taxonomic names (e.g. Pseudo-nitzschia will NOT map to
Pseudonitzschia). An exhaustive list of formatting details is included in
Details. Note that formatting variants are only generated for the names in
tt. This can cause some issues for mapping in the other direction (e.g.
Pseudonitzschia in tt will NOT map to Pseudo-nitzschia in tax2map2 whether or
not ignore.format is TRUE).</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_synonym.file">synonym.file</code></td>
<td>
<p>If &quot;default&quot;, taxmapper uses taxonomic synonyms included
with the ensembleTax package. If a custom taxonomic synonym file is
preferred, a string corresponding to the name of the csv file should be
supplied. Taxonomic synonyms are searched when exact name matches are not
found in tax2map2. ignore.format applies to synonyms if TRUE. Specify NULL if
you wish to forego synonym searches.</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_streamline">streamline</code></td>
<td>
<p>If TRUE, only the mapped version of tt is returned as a
dataframe. If FALSE, a 3-element list is returned where element 1 is the
mapping key returned as a dataframe, element 2 is a character vector of all
names that could not be mapped (no exact matches found in tax2map2), and
element 3 is the mapped version of tt (a dataframe).</p>
</td></tr>
<tr><td><code id="taxmapper_+3A_outfilez">outfilez</code></td>
<td>
<p>If NULL, mapping files are not saved to the current working
directory. Otherwise should be a 3-element character vector including, in
this order, the name of the file to store the taxonomic mapping key, the name
of the file to store the names that could not be mapped, and the name of the
file to store the ASVs supplied with tt with their mapped taxonomic
assignments. Each element of the vector should end in csv (only csv files
may be saved)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Exceptions should be used when the user knows a particular taxonomic
group is not found in tax2map2. The user is responsible for supplying valid
taxonomic names as these must be found in tt and will be propagated as
given to all ASVs that are assigned this name in tt. This should only be
used for high-level taxonomic groups that are not found in a database (e.g.
for retaining Eukaryota when mapping onto a prokaryote-only taxonomic
nomenclature).
</p>
<p>When ignore.format = TRUE, names for which taxmapper cannot find exact
matches in tax2map2 are altered in case an exact match was not found due to
formatting issues. To do this taxmapper first removes square brackets (&quot;[]&quot;).
It then checks for hyphens &quot;-&quot;, underscores &quot;_&quot;, and single spaces &quot; &quot;. If
these are found, variants of the name with the hyphen/underscore/spaces
replaced by each of the other two, as well as all subnames spearated by these
characters, and all subnames pasted together with none of these special
characters, are searched against tax2map2 for exact matches. It also creates
all-lower and all-upper case versions of these elements and again searches
for exact name matches in tax2map2. Words generated by this process that are
2 characters or less are not searched for matches in tax2map2. All
alternative names created when ignore.format = TRUE are also searched for
synonyms in synonym.file if specified.
</p>
<p>To prevent matching of arbitrary names often used in reference databases (eg,
&quot;Clade_X&quot;), and after creating all of the above alternative names if
ignore.format = TRUE, those that BEGIN with any of the words below are
are not use in exact name matching. Instead, the lowest assigned
non-ambiguous name is determined (any name that begins with a word NOT
included in the list below) and is appended to the ambiguous name separated
by a hyphen. The words taxmapper flags as ambiguous are: &quot;Clade&quot;, &quot;CLADE&quot;,
&quot;clade&quot;, &quot;Group&quot;, &quot;GROUP&quot;, &quot;group&quot;, &quot;Class&quot;, &quot;CLASS&quot;, &quot;class&quot;,
&quot;Subclass&quot;, &quot;SubClass&quot;, &quot;SUBCLASS&quot;, &quot;subclass&quot;, &quot;Subclade&quot;, &quot;SubClade&quot;,
&quot;SUBCLADE&quot;, &quot;subclade&quot;, &quot;Subgroup&quot;, &quot;SubGroup&quot;, &quot;SUBGROUP&quot;, &quot;subgroup&quot;,
&quot;Sub group&quot;, &quot;Sub Group&quot;, &quot;SUB GROUP&quot;, &quot;sub group&quot;, &quot;Sub clade&quot;, &quot;Sub Clade&quot;,
&quot;SUB CLADE&quot;, &quot;sub clade&quot;, &quot;Sub class&quot;, &quot;Sub Class&quot;, &quot;SUB CLASS&quot;, &quot;sub class&quot;,
&quot;Sub_group&quot;, &quot;Sub_Group&quot;, &quot;SUB_GROUP&quot;, &quot;sub_group&quot;, &quot;Sub_clade&quot;, &quot;Sub_Clade&quot;,
&quot;SUB_CLADE&quot;, &quot;sub_clade&quot;, &quot;Sub_class&quot;, &quot;Sub_Class&quot;, &quot;SUB_CLASS&quot;, &quot;sub_class&quot;,
&quot;Sub-group&quot;, &quot;Sub-Group&quot;, &quot;SUB-GROUP&quot;, &quot;sub-group&quot;, &quot;Sub-clade&quot;, &quot;Sub-Clade&quot;,
&quot;SUB-CLADE&quot;, &quot;sub-clade&quot;, &quot;Sub-class&quot;, &quot;Sub-Class&quot;, &quot;SUB-CLASS&quot;, &quot;sub-class&quot;,
&quot;incertae sedis&quot;, &quot;INCERTAE SEDIS&quot;, &quot;Incertae sedis&quot;, &quot;Incertae Sedis&quot;,
&quot;incertae-sedis&quot;, &quot;INCERTAE-SEDIS&quot;, &quot;Incertae-sedis&quot;, &quot;Incertae-Sedis&quot;,
&quot;incertae_sedis&quot;, &quot;INCERTAE_-SEDIS&quot;, &quot;Incertae_sedis&quot;, &quot;Incertae_Sedis&quot;,
&quot;incertaesedis&quot;, &quot;INCERTAESEDIS&quot;, &quot;Incertaesedis&quot;, &quot;IncertaeSedis&quot;,
&quot;unclassified&quot;, &quot;UNCLASSIFIED&quot;, &quot;Unclassified&quot;, &quot;Novel&quot;, &quot;novel&quot;, &quot;NOVEL&quot;,
&quot;sp&quot;, &quot;sp.&quot;, &quot;spp&quot;, &quot;spp.&quot;, &quot;lineage&quot;, &quot;Lineage&quot;, &quot;LINEAGE&quot;
</p>
<p>For high-throughput implementation of taxmapper, it's recommended to set
streamline = TRUE.
</p>


<h3>Value</h3>

<p>If streamline = TRUE, a dataframe formatted for use with ensembleTax
that contains mapped taxonomic assignments for each ASV/OTU in the data set.
</p>
<p>If streamline = FALSE, a 3-element list where the first element is a
dataframe that contains all unique input taxonomic assignments and their
corresponding mapped outputs, the second element is a character vector that
contains all taxonomic names that could not be mapped, and the third element
contains mapped taxonomic assignments for each ASV in the data set.
</p>
<p>If is.null(outfilez) = FALSE, three csv files are saved in the current
working directory containing each of the three list elements above.
</p>


<h3>Author(s)</h3>

<p>Dylan Catlett
</p>
<p>Kevin Son
</p>


<h3>See Also</h3>

<p>idtax2df, bayestax2df, ensembleTax
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fake.silva &lt;- data.frame(ASV = c("AAAA", "ATCG", "GCGC", "TATA", "TCGA"),
domain = c("Bacteria", "Eukaryota", "Eukaryota", "Eukaryota", "Eukaryota"),
phylum = c("Firmicutes", "Diatomea", "Retaria", "MAST-12", "Diatomea"),
class = c(NA, "Coscinodiscophytina_cl", "Polycystinea", "MAST-12A",
"Mediophyceae"),
order = c(NA, "Fragilariales", "Collodaria", NA, NA),
family = c(NA, "Fragilariales_fa", "Collodaria_fa", NA, NA),
genus = c(NA, "Podocystis", "Collophidium", NA, NA),
stringsAsFactors = FALSE)
head(fake.silva)
mapped.silva &lt;- taxmapper(fake.silva,
                          tt.ranks = colnames(fake.silva)[2:ncol(fake.silva)],
                          tax2map2 = "pr2",
                          exceptions = c("Archaea", "Bacteria"),
                          ignore.format = FALSE,
                          synonym.file = "default",
                          streamline = TRUE,
                          outfilez = NULL)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
