<!DOCTYPE html><html><head><title>Help for package enshuman</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {enshuman}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#enshuman'><p>enshuman</p></a></li>
<li><a href='#hg37'><p>Ensembl genes build GRCh37.87</p></a></li>
<li><a href='#hg38'><p>Ensembl genes build GRCh38.111</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Human Gene Annotation Data from 'Ensembl'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene information from 'Ensembl' genome builds 'GRCh38.p14' and 'GRCh37.p13' to use with the 'topr' package. The datasets were originally downloaded from <a href="https://ftp.ensembl.org/pub/current/gtf/homo_sapiens/Homo_sapiens.GRCh38.111.gtf.gz">https://ftp.ensembl.org/pub/current/gtf/homo_sapiens/Homo_sapiens.GRCh38.111.gtf.gz</a> and <a href="https://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz">https://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz</a> and converted into the format required by the 'topr' package. See <a href="https://github.com/totajuliusd/topr?tab=readme-ov-file#how-to-use-topr-with-other-species-than-human">https://github.com/totajuliusd/topr?tab=readme-ov-file#how-to-use-topr-with-other-species-than-human</a> to see the required format.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown, markdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-31 10:59:17 UTC; tota</td>
</tr>
<tr>
<td>Author:</td>
<td>Thorhildur Juliusdottir [cph, aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thorhildur Juliusdottir &lt;totajuliusd@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-02 12:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='enshuman'>enshuman</h2><span id='topic+enshuman'></span>

<h3>Description</h3>

<p>Human gene annotation datasets to use with the 'topr' package
</p>


<h3>datasets</h3>

<p>The two datasets are:
</p>

<ul>
<li> <p><code><a href="#topic+hg38">hg38</a></code> Gene information from genome build GRCh38.p14
</p>
</li>
<li> <p><code><a href="#topic+hg37">hg37</a></code> Gene information from genome build GRCh37.p13
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>library(enshuman)
head(hg38)
head(hg37)
</code></pre>

<hr>
<h2 id='hg37'>Ensembl genes build GRCh37.87</h2><span id='topic+hg37'></span>

<h3>Description</h3>

<p>wget https://ftp.ensembl.org/pub/grch37/current/gtf/homo_sapiens/Homo_sapiens.GRCh37.87.gtf.gz
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg37
</code></pre>


<h3>Format</h3>

<p>A data frame  with 55,882 rows and 7 variables:
</p>

<dl>
<dt>chrom</dt><dd><p>chromosome</p>
</dd>
<dt>gene_start</dt><dd><p>genetic position of gene start</p>
</dd>
<dt>gene_end</dt><dd><p>genetic position of gene end</p>
</dd>
<dt>gene_symbol</dt><dd><p>The name of the gene</p>
</dd>
<dt>biotype</dt><dd><p>the biotype of the gene</p>
</dd>
<dt>exon_chromstart</dt><dd><p>genetic positions of exon start</p>
</dd>
<dt>exon_chromend</dt><dd><p>genetic position of exon end</p>
</dd>
</dl>


<hr>
<h2 id='hg38'>Ensembl genes build GRCh38.111</h2><span id='topic+hg38'></span>

<h3>Description</h3>

<p>wget https://ftp.ensembl.org/pub/current/gtf/homo_sapiens/Homo_sapiens.GRCh38.111.gtf.gz
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hg38
</code></pre>


<h3>Format</h3>

<p>A data frame  with 41,016 rows and 7 variables:
</p>

<dl>
<dt>chrom</dt><dd><p>chromosome</p>
</dd>
<dt>gene_start</dt><dd><p>genetic position of gene start</p>
</dd>
<dt>gene_end</dt><dd><p>genetic position of gene end</p>
</dd>
<dt>gene_symbol</dt><dd><p>the name of the gene</p>
</dd>
<dt>biotype</dt><dd><p>the biotype of the gene</p>
</dd>
<dt>exon_chromstart</dt><dd><p>genetic positions of exon start</p>
</dd>
<dt>exon_chromend</dt><dd><p>genetic position of exon end</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
