<!DOCTYPE html><html lang="en"><head><title>Help for package enveomics.R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {enveomics.R}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+24+2Cenve.GrowthCurve-method'><p>Attribute accessor</p></a></li>
<li><a href='#+24+2Cenve.RecPlot2-method'><p>Attribute accessor</p></a></li>
<li><a href='#+24+2Cenve.RecPlot2.Peak-method'><p>Attribute accessor</p></a></li>
<li><a href='#enve.__prune.iter'><p>Enveomics: Prune Iter (Internal Function)</p></a></li>
<li><a href='#enve.__prune.reduce'><p>Enveomics: Prune Reduce (Internal Function)</p></a></li>
<li><a href='#enve.__tribs'><p>Enveomics: TRIBS - Internal Ancillary Function</p></a></li>
<li><a href='#enve.barplot'><p>Enveomics: Barplot</p></a></li>
<li><a href='#enve.cliopts'><p>Enveomics: Cliopts</p></a></li>
<li><a href='#enve.col.alpha'><p>Enveomics: Color Alpha</p></a></li>
<li><a href='#enve.col2alpha'><p>Enveomics: Color to Alpha (deprecated)</p></a></li>
<li><a href='#enve.df2dist'><p>Enveomics: Data Frame to Dist</p></a></li>
<li><a href='#enve.df2dist.group'><p>Enveomics: Data Frame to Dist (Group)</p></a></li>
<li><a href='#enve.df2dist.list'><p>Enveomics: Data Frame to Dist (List)</p></a></li>
<li><a href='#enve.growthcurve'><p>Enveomics: Growth Curve</p></a></li>
<li><a href='#enve.GrowthCurve-class'><p>Enveomics: Growth Curve S4 Class</p></a></li>
<li><a href='#enve.prefscore'><p>Enveomics: Pref Score</p></a></li>
<li><a href='#enve.prune.dist'><p>Enveomics: Prune Dist</p></a></li>
<li><a href='#enve.recplot'><p>Enveomics: Recruitment Plots</p></a></li>
<li><a href='#enve.recplot2'><p>Enveomics: Recruitment Plot (2)</p></a></li>
<li><a href='#enve.RecPlot2-class'><p>Enveomics: Recruitment Plot (2) - S4 Class</p></a></li>
<li><a href='#enve.recplot2.__counts'><p>Enveomics: Recruitment Plot (2) Internal Ancillary Function</p></a></li>
<li><a href='#enve.recplot2.__peakHist'><p>Enveomics: Recruitment Plot (2) Peak S4 Class - Internal Ancillary Function</p></a></li>
<li><a href='#enve.recplot2.__whichClosestPeak'><p>Enveomics: Recruitment Plot (2) Peak Finder - Internal Ancillary Function</p></a></li>
<li><a href='#enve.recplot2.ANIr'><p>Enveomics: Recruitment Plot (2) ANI Estimate</p></a></li>
<li><a href='#enve.recplot2.changeCutoff'><p>Enveomics: Recruitment Plot (2) Change Cutoff</p></a></li>
<li><a href='#enve.recplot2.compareIdentities'><p>Enveomics: Recruitment Plot (2) Compare Identities</p></a></li>
<li><a href='#enve.recplot2.coordinates'><p>Enveomics: Recruitment Plot (2) Coordinates</p></a></li>
<li><a href='#enve.recplot2.corePeak'><p>Enveomics: Recruitment Plot (2) Core Peak Finder</p></a></li>
<li><a href='#enve.recplot2.extractWindows'><p>Enveomics: Recruitment Plot (2) Extract Windows</p></a></li>
<li><a href='#enve.recplot2.findPeaks'><p>Enveomics: Recruitment Plot (2) Peak Finder</p></a></li>
<li><a href='#enve.recplot2.findPeaks.__em_e'><p>Enveomics: Recruitment Plot (2) EM Peak Finder - Internal Ancillary Function Expectation</p></a></li>
<li><a href='#enve.recplot2.findPeaks.__em_m'><p>Enveomics: Recruitment Plot (2) Em Peak Finder - Internal Ancillary Function Maximization</p></a></li>
<li><a href='#enve.recplot2.findPeaks.__emauto_one'><p>Enveomics: Recruitment Plot (2) EMauto Peak Finder - Internal Ancillary Function</p></a></li>
<li><a href='#enve.recplot2.findPeaks.__mow_one'><p>Enveomics: Recruitment Plot (2) Mowing Peak Finder - Internal Ancillary Function 1</p></a></li>
<li><a href='#enve.recplot2.findPeaks.__mower'><p>Enveomics: Recruitment Plot (2) Mowing Peak Finder - Internal Ancillary Function 2</p></a></li>
<li><a href='#enve.recplot2.findPeaks.em'><p>Enveomics: Recruitment Plot (2) Em Peak Finder</p></a></li>
<li><a href='#enve.recplot2.findPeaks.emauto'><p>Enveomics: Recruitment Plot (2) Emauto Peak Finder</p></a></li>
<li><a href='#enve.recplot2.findPeaks.mower'><p>Enveomics: Recruitment Plot (2) Mowing Peak Finder</p></a></li>
<li><a href='#enve.RecPlot2.Peak-class'><p>Enveomics: Recruitment Plot (2) Peak - S4 Class</p></a></li>
<li><a href='#enve.recplot2.seqdepth'><p>Enveomics: Recruitment Plot (2) Sequencing Depth</p></a></li>
<li><a href='#enve.recplot2.windowDepthThreshold'><p>Enveomics: Recruitment Plot (2) Window Depth Threshold</p></a></li>
<li><a href='#enve.selvector'><p>Enveomics: Selection vector</p></a></li>
<li><a href='#enve.tribs'><p>Enveomics: TRIBS</p></a></li>
<li><a href='#enve.TRIBS-class'><p>Enveomics: TRIBS S4 Class</p></a></li>
<li><a href='#enve.TRIBS.merge'><p>Enveomics: TRIBS Merge</p></a></li>
<li><a href='#enve.tribs.test'><p>Enveomics: TRIBS Test</p></a></li>
<li><a href='#enve.TRIBStest-class'><p>Enveomics: TRIBS Test S4 Class</p></a></li>
<li><a href='#enve.truncate'><p>Enveomics: Truncate</p></a></li>
<li><a href='#growth.curves'><p>Bacterial growth curves for three Escherichia coli mutants</p></a></li>
<li><a href='#phyla.counts'><p>Counts of microbial phyla in four sites</p></a></li>
<li><a href='#plot.enve.GrowthCurve'><p>Enveomics: Plot of Growth Curve</p></a></li>
<li><a href='#plot.enve.RecPlot2'><p>Enveomics: Recruitment Plot (2)</p></a></li>
<li><a href='#plot.enve.TRIBS'><p>Enveomics: TRIBS Plot</p></a></li>
<li><a href='#plot.enve.TRIBStest'><p>Enveomics: TRIBS Plot Test</p></a></li>
<li><a href='#summary.enve.GrowthCurve'><p>Enveomics: Summary of Growth Curve</p></a></li>
<li><a href='#summary.enve.TRIBS'><p>Enveomics: TRIBS Summary</p></a></li>
<li><a href='#summary.enve.TRIBStest'><p>Enveomics: TRIBS Summary Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.10.0</td>
</tr>
<tr>
<td>Title:</td>
<td>Various Utilities for Microbial Genomics and Metagenomics</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of functions for microbial ecology and other
   applications of genomics and metagenomics. Companion package for the
   Enveomics Collection (Rodriguez-R, L.M. and Konstantinidis, K.T., 2016
   &lt;<a href="https://doi.org/10.7287%2Fpeerj.preprints.1900v1">doi:10.7287/peerj.preprints.1900v1</a>&gt;).</td>
</tr>
<tr>
<td>Author:</td>
<td>Luis M. Rodriguez-R [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luis M. Rodriguez-R &lt;lmrodriguezr@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://enve-omics.ce.gatech.edu/enveomics/">http://enve-omics.ce.gatech.edu/enveomics/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.9), stats, methods, parallel, fitdistrplus, sn,
investr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tools, vegan, ape, picante, gplots, optparse</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-11 17:38:43 UTC; miguel</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-11 17:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='+24+2Cenve.GrowthCurve-method'>Attribute accessor</h2><span id='topic++24+2Cenve.GrowthCurve-method'></span>

<h3>Description</h3>

<p>Attribute accessor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'enve.GrowthCurve'
x$name
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B24+2B2Cenve.GrowthCurve-method_+3A_x">x</code></td>
<td>
<p>Object</p>
</td></tr>
<tr><td><code id="+2B24+2B2Cenve.GrowthCurve-method_+3A_name">name</code></td>
<td>
<p>Attribute name</p>
</td></tr>
</table>

<hr>
<h2 id='+24+2Cenve.RecPlot2-method'>Attribute accessor</h2><span id='topic++24+2Cenve.RecPlot2-method'></span>

<h3>Description</h3>

<p>Attribute accessor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'enve.RecPlot2'
x$name
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B24+2B2Cenve.RecPlot2-method_+3A_x">x</code></td>
<td>
<p>Object</p>
</td></tr>
<tr><td><code id="+2B24+2B2Cenve.RecPlot2-method_+3A_name">name</code></td>
<td>
<p>Attribute name</p>
</td></tr>
</table>

<hr>
<h2 id='+24+2Cenve.RecPlot2.Peak-method'>Attribute accessor</h2><span id='topic++24+2Cenve.RecPlot2.Peak-method'></span>

<h3>Description</h3>

<p>Attribute accessor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'enve.RecPlot2.Peak'
x$name
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B24+2B2Cenve.RecPlot2.Peak-method_+3A_x">x</code></td>
<td>
<p>Object</p>
</td></tr>
<tr><td><code id="+2B24+2B2Cenve.RecPlot2.Peak-method_+3A_name">name</code></td>
<td>
<p>Attribute name</p>
</td></tr>
</table>

<hr>
<h2 id='enve.__prune.iter'>Enveomics: Prune Iter (Internal Function)</h2><span id='topic+enve.__prune.iter'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+enve.prune.dist">enve.prune.dist</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.__prune.iter(t, dist, min_dist, quiet)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.__prune.iter_+3A_t">t</code></td>
<td>
<p>A <strong>phylo</strong> object.</p>
</td></tr>
<tr><td><code id="enve.__prune.iter_+3A_dist">dist</code></td>
<td>
<p>Cophenetic distance matrix.</p>
</td></tr>
<tr><td><code id="enve.__prune.iter_+3A_min_dist">min_dist</code></td>
<td>
<p>Minimum distance.</p>
</td></tr>
<tr><td><code id="enve.__prune.iter_+3A_quiet">quiet</code></td>
<td>
<p>If running quietly.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <strong>phylo</strong> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.__prune.reduce'>Enveomics: Prune Reduce (Internal Function)</h2><span id='topic+enve.__prune.reduce'></span>

<h3>Description</h3>

<p>Internal function for <code><a href="#topic+enve.prune.dist">enve.prune.dist</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.__prune.reduce(t, nodes, min_dist, quiet)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.__prune.reduce_+3A_t">t</code></td>
<td>
<p>A <strong>phylo</strong> object.</p>
</td></tr>
<tr><td><code id="enve.__prune.reduce_+3A_nodes">nodes</code></td>
<td>
<p>Vector of nodes.</p>
</td></tr>
<tr><td><code id="enve.__prune.reduce_+3A_min_dist">min_dist</code></td>
<td>
<p>Minimum distance.</p>
</td></tr>
<tr><td><code id="enve.__prune.reduce_+3A_quiet">quiet</code></td>
<td>
<p>If running quietly.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <strong>phylo</strong> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.__tribs'>Enveomics: TRIBS - Internal Ancillary Function</h2><span id='topic+enve.__tribs'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.tribs">enve.tribs</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.__tribs(
  rep,
  frx,
  selection,
  dimensions,
  dots,
  dist.method,
  summary.fx,
  dist
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.__tribs_+3A_rep">rep</code></td>
<td>
<p>Replicates</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_frx">frx</code></td>
<td>
<p>Fraction</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_selection">selection</code></td>
<td>
<p>Selection</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_dimensions">dimensions</code></td>
<td>
<p>Dimensions</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_dots">dots</code></td>
<td>
<p>Sampling points</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_dist.method">dist.method</code></td>
<td>
<p>Distance method</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_summary.fx">summary.fx</code></td>
<td>
<p>Summary function</p>
</td></tr>
<tr><td><code id="enve.__tribs_+3A_dist">dist</code></td>
<td>
<p>Distance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric indicating the <code>summary.fx</code> value applied to the
distance matrix subset
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.barplot'>Enveomics: Barplot</h2><span id='topic+enve.barplot'></span>

<h3>Description</h3>

<p>Creates nice barplots from tab-delimited tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.barplot(
  x,
  sizes,
  top = 25,
  colors.per.group = 9,
  bars.width = 4,
  legend.ncol = 1,
  other.col = "#000000",
  add.trend = FALSE,
  organic.trend = FALSE,
  sort.by = median,
  min.report = 101,
  order = NULL,
  col,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.barplot_+3A_x">x</code></td>
<td>
<p>Can be either the input data or the path to the file containing 
the table. 
</p>

<ul>
<li><p>If it contains the data, it must be a data frame or an
object coercible to a data frame. 
</p>
</li>
<li><p>If it is a path, it must point to a
tab-delimited file containing a header (first row) and row names 
(first column).
</p>
</li></ul>
</td></tr>
<tr><td><code id="enve.barplot_+3A_sizes">sizes</code></td>
<td>
<p>A numeric vector containing the real size of the samples
(columns) in the same order of the input table. If set, the values are
assumed to be 100%. Otherwise, the sum of the columns is used.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_top">top</code></td>
<td>
<p>Maximum number of categories to display. Any additional
categories will be listed as &quot;Others&quot;.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_colors.per.group">colors.per.group</code></td>
<td>
<p>Number of categories in the first two saturation 
groups of colors. The third group contains the remaining categories if 
needed.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_bars.width">bars.width</code></td>
<td>
<p>Width of the barplot with respect to the legend.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_legend.ncol">legend.ncol</code></td>
<td>
<p>Number of columns in the legend.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_other.col">other.col</code></td>
<td>
<p>Color of the &quot;Others&quot; category.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_add.trend">add.trend</code></td>
<td>
<p>Controls if semi-transparent areas are to be plotted
between the bars to connect the regions (trend regions).</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_organic.trend">organic.trend</code></td>
<td>
<p>Controls if the trend regions are to be smoothed 
(curves). By default, trend regions have straight edges. If <code>TRUE</code>, 
forces <code>add.trend=TRUE</code>.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_sort.by">sort.by</code></td>
<td>
<p>Any function that takes a numeric vector and returns a
numeric scalar. This function is applied to each row, and the resulting
values are used to sort the rows (decreasingly). Good options include: 
<code>sd, min, max, mean, median</code>.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_min.report">min.report</code></td>
<td>
<p>Minimum percentage to report the value in the plot.
Any value above 100 indicates that no values are to be reported.</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_order">order</code></td>
<td>
<p>Controls how the rows should be ordered. 
</p>

<ul>
<li>
<p>If <code>NULL</code> (default), <code>sort.by</code> is applied per row and the
results are sorted decreasingly.

</p>
</li>
<li>
<p>If <code>NA</code>, no sorting is performed, i.e., the original order is
respected.

</p>
</li>
<li>
<p>If a vector is provided, it is assumed to be the custom order to be used
(either by numeric index or by row names).

</p>
</li></ul>
</td></tr>
<tr><td><code id="enve.barplot_+3A_col">col</code></td>
<td>
<p>Colors to use. If provided, overrides the variables <code>top</code>
and <code>colors.per.group</code>, but <code>other.col</code> is still used if the 
vector is insufficient for all the rows. An additional palette is available
with <code>col='coto'</code> (contributed by Luis (Coto) Orellana).</p>
</td></tr>
<tr><td><code id="enve.barplot_+3A_...">...</code></td>
<td>
<p>Any additional parameters to be passed to barplot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("phyla.counts", package = "enveomics.R", envir = environment())
# Create a barplot sorted by variance with organic trends
enve.barplot(
  phyla.counts, # Counts of phyla in four sites
  sizes = c(250,100,75,200), # Total sizes of the datasets of each site
  bars.width = 2, # Decrease from default, so the names are fully displayed
  organic.trend = TRUE, # Nice curvy background
  sort.by = var # Sort by variance across sites
)

</code></pre>

<hr>
<h2 id='enve.cliopts'>Enveomics: Cliopts</h2><span id='topic+enve.cliopts'></span>

<h3>Description</h3>

<p>Generates nicely formatted command-line interfaces for functions 
(<strong>closures</strong> only).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.cliopts(
  fx,
  rd_file,
  positional_arguments,
  usage,
  mandatory = c(),
  vectorize = c(),
  ignore = c(),
  number = c(),
  defaults = list(),
  o_desc = list(),
  p_desc = ""
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.cliopts_+3A_fx">fx</code></td>
<td>
<p>Function for which the interface should be generated.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_rd_file">rd_file</code></td>
<td>
<p>(Optional) .Rd file with the standard documentation of 
the function.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_positional_arguments">positional_arguments</code></td>
<td>
<p>(Optional) Number of <strong>positional</strong> 
arguments passed to <code><a href="optparse.html#topic+parse_args">parse_args</a></code> 
(package: <span class="pkg">optparse</span>).</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_usage">usage</code></td>
<td>
<p>(Optional) Usage passed to <code><a href="optparse.html#topic+OptionParser">OptionParser</a></code> 
(package: <span class="pkg">optparse</span>).</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_mandatory">mandatory</code></td>
<td>
<p>Mandatory arguments.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_vectorize">vectorize</code></td>
<td>
<p>Arguments of the function to vectorize (comma-delimited).
If numeric, use also <code>number</code>.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_ignore">ignore</code></td>
<td>
<p>Arguments of the function to ignore.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_number">number</code></td>
<td>
<p>Force these arguments as numerics. Useful for numeric
vectors (see <code>vectorize</code>) or arguments with no defaults.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_defaults">defaults</code></td>
<td>
<p>Defaults to use instead of the ones provided by the 
formals.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_o_desc">o_desc</code></td>
<td>
<p>Descriptions of the options. Help from <code>rd</code> is ignored
for arguments present in this list.</p>
</td></tr>
<tr><td><code id="enve.cliopts_+3A_p_desc">p_desc</code></td>
<td>
<p>Description Description of the function. Help from <code>rd</code> 
is ignored for the function description unless this value is an empty string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with keys: 
</p>

<ul>
<li><p><code>options</code>, a named list with the values for the function's 
arguments 
</p>
</li>
<li><p><code>args</code>, a vector with zero or more strings containing the
positional arguments</p>
</li></ul>



<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.col.alpha'>Enveomics: Color Alpha</h2><span id='topic+enve.col.alpha'></span>

<h3>Description</h3>

<p>Modify alpha in a color (or vector of colors).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.col.alpha(col, alpha = 1/2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.col.alpha_+3A_col">col</code></td>
<td>
<p>Color or vector of colors. It can be any value supported by 
<code><a href="grDevices.html#topic+col2rgb">col2rgb</a></code>, such as <code>darkred</code> or <code>#009988</code>.</p>
</td></tr>
<tr><td><code id="enve.col.alpha_+3A_alpha">alpha</code></td>
<td>
<p>Alpha value to add to the color, from 0 to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a color or a vector of colors in <em>hex</em> notation, 
including <code>alpha</code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Hexcode for a color by hexcode
enve.col.alpha("#009988", 3/4) # "#009988BF"

# Hexcode for a color by name
enve.col.alpha("white", 1/4) # "#FFFFFF3F"

# Hexcode for a color from other functions
enve.col.alpha(rainbow(3)) # "#FF00007F" "#00FF007F" "#0000FF7F"
</code></pre>

<hr>
<h2 id='enve.col2alpha'>Enveomics: Color to Alpha (deprecated)</h2><span id='topic+enve.col2alpha'></span>

<h3>Description</h3>

<p>Takes a vector of colors and sets the alpha.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.col2alpha(x, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.col2alpha_+3A_x">x</code></td>
<td>
<p>A vector of any value base colors.</p>
</td></tr>
<tr><td><code id="enve.col2alpha_+3A_alpha">alpha</code></td>
<td>
<p>Alpha level to set, in the [0, 1] range.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>DEPRECATED: Use instead <code><a href="#topic+enve.col.alpha">enve.col.alpha</a></code>.
</p>


<h3>Value</h3>

<p>A vector of colors with alpha set.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.df2dist'>Enveomics: Data Frame to Dist</h2><span id='topic+enve.df2dist'></span>

<h3>Description</h3>

<p>Transform a dataframe (or coercible object, like a table) into a
<strong>dist</strong> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.df2dist(
  x,
  obj1.index = 1,
  obj2.index = 2,
  dist.index = 3,
  default.d = NA,
  max.sim = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.df2dist_+3A_x">x</code></td>
<td>
<p>A dataframe (or coercible object) with at least three columns:
</p>

<ol>
<li><p> ID of the object 1, 
</p>
</li>
<li><p> ID of the object 2, and 
</p>
</li>
<li><p> distance between the two objects.</p>
</li></ol>
</td></tr>
<tr><td><code id="enve.df2dist_+3A_obj1.index">obj1.index</code></td>
<td>
<p>Index of the column containing the ID of the object 1.</p>
</td></tr>
<tr><td><code id="enve.df2dist_+3A_obj2.index">obj2.index</code></td>
<td>
<p>Index of the column containing the ID of the object 2.</p>
</td></tr>
<tr><td><code id="enve.df2dist_+3A_dist.index">dist.index</code></td>
<td>
<p>Index of the column containing the distance.</p>
</td></tr>
<tr><td><code id="enve.df2dist_+3A_default.d">default.d</code></td>
<td>
<p>Default value (for missing values).</p>
</td></tr>
<tr><td><code id="enve.df2dist_+3A_max.sim">max.sim</code></td>
<td>
<p>If not zero, assumes that the values are similarity
(not distance) and this is the maximum similarity (corresponding to 
distance 0). Applies transformation: 
<code class="reqn">distance = (max.sim - values)/max.sim.</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <strong>dist</strong> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A sparse matrix representation of similarities as data frame.
# The column "extra_data" is meaningless, only included to illustrate
# the use of the obj*.index parameters
sim &lt;- data.frame(
  extra_data = c(0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.5),
  query      = c("A", "A", "A", "B", "C", "C", "D"),
  subject    = c("A", "B", "C", "B", "C", "B", "A"),
  similarity = c(100,  90,  60, 100, 100,  70,  10)
)
dist &lt;- enve.df2dist(sim, "query", "subject", "similarity", max.sim = 100)
print(dist)

</code></pre>

<hr>
<h2 id='enve.df2dist.group'>Enveomics: Data Frame to Dist (Group)</h2><span id='topic+enve.df2dist.group'></span>

<h3>Description</h3>

<p>Transform a dataframe (or coercible object, like a table) into a 
<strong>dist</strong> object, where there are 1 or more distances between each pair 
of objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.df2dist.group(
  x,
  obj1.index = 1,
  obj2.index = 2,
  dist.index = 3,
  summary = median,
  empty.rm = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.df2dist.group_+3A_x">x</code></td>
<td>
<p>A dataframe (or coercible object) with at least three columns:
</p>

<ol>
<li><p> ID of the object 1, 
</p>
</li>
<li><p> ID of the object 2, and 
</p>
</li>
<li><p> distance between the two objects.
</p>
</li></ol>
</td></tr>
<tr><td><code id="enve.df2dist.group_+3A_obj1.index">obj1.index</code></td>
<td>
<p>Index of the column containing the ID of the object 1.</p>
</td></tr>
<tr><td><code id="enve.df2dist.group_+3A_obj2.index">obj2.index</code></td>
<td>
<p>Index of the column containing the ID of the object 2.</p>
</td></tr>
<tr><td><code id="enve.df2dist.group_+3A_dist.index">dist.index</code></td>
<td>
<p>Index of the column containing the distance.</p>
</td></tr>
<tr><td><code id="enve.df2dist.group_+3A_summary">summary</code></td>
<td>
<p>Function summarizing the different distances between the 
two objects.</p>
</td></tr>
<tr><td><code id="enve.df2dist.group_+3A_empty.rm">empty.rm</code></td>
<td>
<p>Remove rows with empty or <code>NA</code> groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <strong>dist</strong> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A sparse matrix representation of distances as data frame.
# Note that some pairs are repeated.
dist.df &lt;- data.frame(
  query    = c("A", "A", "A", "B", "C", "C", "B", "B", "B"),
  subject  = c("A", "B", "C", "B", "C", "B", "A", "C", "C"),
  distance = c(  0, 0.1, 0.4,   0,   0, 0.4, 0.2, 0.2, 0.1)
)
dist &lt;- enve.df2dist.group(dist.df)
print(dist)

# Use the mean of all repeated occurrences instead of the median.
dist &lt;- enve.df2dist.group(dist.df, summary = mean)

# Simply use the first occurrence for any given pair.
dist &lt;- enve.df2dist.group(dist.df, summary = function(x) head(x, n = 1))

</code></pre>

<hr>
<h2 id='enve.df2dist.list'>Enveomics: Data Frame to Dist (List)</h2><span id='topic+enve.df2dist.list'></span>

<h3>Description</h3>

<p>Transform a dataframe (or coercible object, like a table)
into a <strong>list</strong> of <strong>dist</strong> objects, one per group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.df2dist.list(
  x,
  groups,
  obj1.index = 1,
  obj2.index = 2,
  dist.index = 3,
  empty.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.df2dist.list_+3A_x">x</code></td>
<td>
<p>A dataframe (or coercible object) with at least three columns:
</p>

<ol>
<li><p> ID of the object 1, 
</p>
</li>
<li><p> ID of the object 2, and 
</p>
</li>
<li><p> distance between the two objects.</p>
</li></ol>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_groups">groups</code></td>
<td>
<p>Named array where the IDs correspond to the object IDs,
and the values correspond to the group.</p>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_obj1.index">obj1.index</code></td>
<td>
<p>Index of the column containing the ID of the object 1.</p>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_obj2.index">obj2.index</code></td>
<td>
<p>Index of the column containing the ID of the object 2.</p>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_dist.index">dist.index</code></td>
<td>
<p>Index of the column containing the distance.</p>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_empty.rm">empty.rm</code></td>
<td>
<p>Remove incomplete matrices.</p>
</td></tr>
<tr><td><code id="enve.df2dist.list_+3A_...">...</code></td>
<td>
<p>Any other parameters supported by <code><a href="#topic+enve.df2dist">enve.df2dist</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <strong>list</strong> of <strong>dist</strong> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.growthcurve'>Enveomics: Growth Curve</h2><span id='topic+enve.growthcurve'></span>

<h3>Description</h3>

<p>Calculates growth curves using the logistic growth function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.growthcurve(
  x,
  times = 1:nrow(x),
  triplicates = FALSE,
  design,
  new.times = seq(min(times), max(times), length.out = length(times) * 10),
  level = 0.95,
  interval = c("confidence", "prediction"),
  plot = TRUE,
  FUN = function(t, K, r, P0) K * P0 * exp(r * t)/(K + P0 * (exp(r * t) - 1)),
  nls.opt = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.growthcurve_+3A_x">x</code></td>
<td>
<p>Data frame (or coercible) containing the observed growth data
(e.g., O.D. values). Each column is an independent growth curve and each
row is a time point. <code>NA</code>'s are allowed.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_times">times</code></td>
<td>
<p>Vector with the times at which each row was taken. By default,
all rows are assumed to be part of constantly periodic measurements.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_triplicates">triplicates</code></td>
<td>
<p>If <code>TRUE</code>, the columns are assumed to be sorted by
sample with three replicates by sample. It requires a number of columns 
multiple of 3.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_design">design</code></td>
<td>
<p>Experimental design of the data. An <strong>array</strong> of mode list
with sample names as index and the list of column names in each sample as 
the values. By default, each column is assumed to be an independent sample
if <code>triplicates</code> is <code>FALSE</code>, or every three columns are assumed 
to be a sample if <code>triplicates</code> is <code>TRUE</code>. In the latter case, 
samples are simply numbered.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_new.times">new.times</code></td>
<td>
<p>Values of time for the fitted curve.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_level">level</code></td>
<td>
<p>Confidence (or prediction) interval in the fitted curve.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_interval">interval</code></td>
<td>
<p>Type of interval to be calculated for the fitted curve.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_plot">plot</code></td>
<td>
<p>Should the growth curve be plotted?</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_fun">FUN</code></td>
<td>
<p>Function to fit. By default: logistic growth with paramenters
<code>K</code>: carrying capacity, 
<code>r</code>: intrinsic growth rate, and
<code>P0</code>: Initial population.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_nls.opt">nls.opt</code></td>
<td>
<p>Any additional options passed to <code>nls</code>.</p>
</td></tr>
<tr><td><code id="enve.growthcurve_+3A_...">...</code></td>
<td>
<p>Any additional parameters to be passed to 
<code>plot.enve.GrowthCurve</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+enve.GrowthCurve">enve.GrowthCurve</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data("growth.curves", package = "enveomics.R", envir = environment())

# Generate growth curves with different colors
g &lt;- enve.growthcurve(growth.curves[, -1], growth.curves[, 1],
                      triplicates = TRUE)

# Generate black-and-white growth curves with different symbols
plot(g, pch=15:17, col="black", band.density=45, band.angle=c(-45,45,0))

</code></pre>

<hr>
<h2 id='enve.GrowthCurve-class'>Enveomics: Growth Curve S4 Class</h2><span id='topic+enve.GrowthCurve-class'></span><span id='topic+enve.GrowthCurve'></span>

<h3>Description</h3>

<p>Enve-omics representation of fitted growth curves.
</p>


<h3>Slots</h3>


<dl>
<dt><code>design</code></dt><dd><p><code>(array)</code> Experimental design of the experiment.</p>
</dd>
<dt><code>models</code></dt><dd><p><code>(list)</code>  Fitted growth curve models.</p>
</dd>
<dt><code>predict</code></dt><dd><p><code>(list)</code>  Fitted growth curve values.</p>
</dd>
<dt><code>call</code></dt><dd><p><code>(call)</code>  Call producing this object.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.prefscore'>Enveomics: Pref Score</h2><span id='topic+enve.prefscore'></span>

<h3>Description</h3>

<p>Estimate preference score of species based on occupancy in biased sample sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.prefscore(
  x,
  set,
  ignore = NULL,
  signif.thr,
  plot = TRUE,
  col.above = rgb(148, 17, 0, maxColorValue = 255),
  col.equal = rgb(189, 189, 189, maxColorValue = 255),
  col.below = rgb(47, 84, 150, maxColorValue = 255),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.prefscore_+3A_x">x</code></td>
<td>
<p>Occupancy matrix (logical or numeric binary) with species as rows and samples
as columns</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_set">set</code></td>
<td>
<p>Vector indicating samples in the test set. It can be any selection vector:
boolean (same length as the number of columns in <code>x</code>), or numeric or
character vector with indexes of the <code>x</code> columns.</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_ignore">ignore</code></td>
<td>
<p>Vector indicating species to ignore. It can be any selection vector with
respect to the rows in <code>x</code> (see <code>set</code>).</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_signif.thr">signif.thr</code></td>
<td>
<p>Absolute value of the significance threshold</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_plot">plot</code></td>
<td>
<p>Indicates if a plot should be generated</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_col.above">col.above</code></td>
<td>
<p>Color for points significantly above zero</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_col.equal">col.equal</code></td>
<td>
<p>Color for points not significantly different from zero</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_col.below">col.below</code></td>
<td>
<p>Color for points significantly below zero</p>
</td></tr>
<tr><td><code id="enve.prefscore_+3A_...">...</code></td>
<td>
<p>Any additional parameters supported by <code>plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named vector of preference scores.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.prune.dist'>Enveomics: Prune Dist</h2><span id='topic+enve.prune.dist'></span>

<h3>Description</h3>

<p>Automatically prunes a tree, to keep representatives of each clade.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.prune.dist(
  t,
  dist.quantile = 0.25,
  min_dist,
  quiet = FALSE,
  max_iters = 100,
  min_nodes_random = 40000,
  random_nodes_frx = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.prune.dist_+3A_t">t</code></td>
<td>
<p>A <strong>phylo</strong> object or a path to the Newick file.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_dist.quantile">dist.quantile</code></td>
<td>
<p>The quantile of edge lengths.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_min_dist">min_dist</code></td>
<td>
<p>The minimum distance to allow between two tips.
If not set, <code>dist.quantile</code> is used instead to calculate it.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_quiet">quiet</code></td>
<td>
<p>Boolean indicating if the function must run without output.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_max_iters">max_iters</code></td>
<td>
<p>Maximum number of iterations.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_min_nodes_random">min_nodes_random</code></td>
<td>
<p>Minimum number of nodes to trigger <em>tip-pairs</em> nodes sampling. 
This sampling is less reproducible and more computationally expensive,
but it's the only solution if the cophenetic matrix exceeds <code>2^31-1</code> 
entries; above that, it cannot be represented in R.</p>
</td></tr>
<tr><td><code id="enve.prune.dist_+3A_random_nodes_frx">random_nodes_frx</code></td>
<td>
<p>Fraction of the nodes to be sampled if more than <code>min_nodes_random</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a pruned <strong>phylo</strong> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot'>Enveomics: Recruitment Plots</h2><span id='topic+enve.recplot'></span>

<h3>Description</h3>

<p>Produces recruitment plots provided that BlastTab.catsbj.pl has
been previously executed. Requires the <span class="pkg">gplots</span> library.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot(
  prefix,
  id.min = NULL,
  id.max = NULL,
  id.binsize = NULL,
  id.splines = 0,
  id.metric = "id",
  id.summary = "sum",
  pos.min = 1,
  pos.max = NULL,
  pos.binsize = 1000,
  pos.splines = 0,
  rec.col1 = "white",
  rec.col2 = "black",
  main = NULL,
  contig.col = grey(0.85),
  ret.recplot = FALSE,
  ret.hist = FALSE,
  ret.mode = FALSE,
  id.cutoff = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot_+3A_prefix">prefix</code></td>
<td>
<p>Path to the prefix of the BlastTab.catsbj.pl output files. At
least the files <strong>.rec</strong> and <strong>.lim</strong> must exist with this prefix.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.min">id.min</code></td>
<td>
<p>Minimum identity to be considered. By default, the minimum detected
identity. This value is a percentage.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.max">id.max</code></td>
<td>
<p>Maximum identity to be considered. By default, 100%.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.binsize">id.binsize</code></td>
<td>
<p>Size of the identity bins (vertical histograms). By default, 0.1 for
identity metrics and 5 for bit score.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.splines">id.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the identity histogram. Zero (0) for
no splines. A generally good value is 1/2. If non-zero, requires the
<span class="pkg">stats</span> package.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.metric">id.metric</code></td>
<td>
<p>Metric of identity to be used (Y-axis). It can be any unambiguous prefix of:
</p>

<ul>
<li><p> &quot;identity&quot;
</p>
</li>
<li><p> &quot;corrected identity&quot; 
</p>
</li>
<li><p> &quot;bit score&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.summary">id.summary</code></td>
<td>
<p>Method used to build the identity histogram (Horizontal axis of the right
panel). It can be any unambiguous prefix of: 
</p>

<ul>
<li><p> &quot;sum&quot;
</p>
</li>
<li><p> &quot;average&quot; 
</p>
</li>
<li><p> &quot;median&quot;
</p>
</li>
<li><p> &quot;90% lower bound&quot;
</p>
</li>
<li><p> &quot;90% upper bound&quot;
</p>
</li>
<li><p> &quot;95% lower bound&quot;
</p>
</li>
<li><p> &quot;95% upper bound&quot;
</p>
</li></ul>

<p>The last four options
correspond to the upper and lower boundaries of the 90% and 95% empirical
confidence intervals.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_pos.min">pos.min</code></td>
<td>
<p>Minimum (leftmost) position in the reference (concatenated) genome (in bp).</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_pos.max">pos.max</code></td>
<td>
<p>Maximum (rightmost) position in the reference (concatenated) genome (in bp).
By default: Length of the genome.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_pos.binsize">pos.binsize</code></td>
<td>
<p>Size of the position bins (horizontal histograms) in bp.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_pos.splines">pos.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the position histogram. Zero (0) for
no splines. If non-zero, requires the stats package.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_rec.col1">rec.col1</code></td>
<td>
<p>Lightest color in the recruitment plot.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_rec.col2">rec.col2</code></td>
<td>
<p>Darkest color in the recruitment plot.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_main">main</code></td>
<td>
<p>Title of the plot.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_contig.col">contig.col</code></td>
<td>
<p>Color of the Contig boundaries. Set to <code>NA</code> to ignore Contig boundaries.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_ret.recplot">ret.recplot</code></td>
<td>
<p>Indicates if the matrix of the recruitment plot is to be returned.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_ret.hist">ret.hist</code></td>
<td>
<p>Ignored, for backwards compatibility.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_ret.mode">ret.mode</code></td>
<td>
<p>Indicates if the mode of the identity is to be computed. It requires the 
<span class="pkg">modeest</span> package.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_id.cutoff">id.cutoff</code></td>
<td>
<p>Minimum identity to consider an alignment as &quot;top&quot;. By default, it is 0.95
for the identity metrics and 95% of the best scoring alignment for bit
score.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_verbose">verbose</code></td>
<td>
<p>Indicates if the function should report the advance.</p>
</td></tr>
<tr><td><code id="enve.recplot_+3A_...">...</code></td>
<td>
<p>Any additional graphic parameters to be passed to plot for all panels except
the recruitment plot (lower-left).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the following elements:
</p>

<dl>
<dt><code>pos.marks</code></dt><dd><p>Midpoints of the position histogram.</p>
</dd>
<dt><code>id.matrix</code></dt><dd><p>Midpoints of the identity histogram.</p>
</dd>
<dt><code>recplot</code></dt><dd><p>Matrix containing the recruitment plot values 
(if <code>ret.recplot=TRUE</code>).</p>
</dd>
<dt><code>id.mean</code></dt><dd><p>Mean identity.</p>
</dd>
<dt><code>id.median</code></dt><dd><p>Median identity.</p>
</dd>
<dt><code>id.mode</code></dt><dd><p>Mode of the identity (if <code>ret.mode=TRUE</code>).
Deprecated.</p>
</dd>
<dt><code>id.hist</code></dt><dd><p>Values of the identity histogram
(if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>pos.hist.low</code></dt><dd><p>Values of the position histogram (depth) with
&quot;low&quot; identity (i.e., below id.cutoff) (if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>pos.hist.top</code></dt><dd><p>Values of the position histogram (depth) with
&quot;top&quot; identity (i.e., above id.cutoff) (if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>id.max</code></dt><dd><p>Value of <code>id.max</code>. This is returned because 
<code>id.max=NULL</code> may vary.</p>
</dd>
<dt><code>id.cutoff</code></dt><dd><p>Value of <code>id.cutoff</code>. 
This is returned because <code>id.cutoff=NULL</code> may vary.</p>
</dd>
<dt><code>seqdepth.mean.top</code></dt><dd><p>Average sequencing depth with identity above 
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.mean.low</code></dt><dd><p>Average sequencing depth with identity below 
<code>id.cutoff</code>.</p>
</dd> 
<dt><code>seqdepth.mean.all</code></dt><dd><p>Average sequencing depth without identity
filtering.</p>
</dd>
<dt><code>seqdepth.median.top</code></dt><dd><p>Median sequencing depth with identity above
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.median.low</code></dt><dd><p>Median sequencing depth with identity below
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.median.all</code></dt><dd><p>Median sequencing depth without identity
filtering.</p>
</dd>
<dt><code>id.metric</code></dt><dd><p>Full name of the used identity metric.</p>
</dd>
<dt><code>id.summary</code></dt><dd><p>Full name of the summary method used to build the
identity plot.</p>
</dd></dl>



<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2'>Enveomics: Recruitment Plot (2)</h2><span id='topic+enve.recplot2'></span>

<h3>Description</h3>

<p>Produces recruitment plots provided that <code>BlastTab.catsbj.pl</code> has
been previously executed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2(
  prefix,
  plot = TRUE,
  pos.breaks = 1000,
  pos.breaks.tsv = NA,
  id.breaks = 60,
  id.free.range = FALSE,
  id.metric = c("identity", "corrected identity", "bit score"),
  id.summary = sum,
  id.cutoff = 95,
  threads = 2,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2_+3A_prefix">prefix</code></td>
<td>
<p>Path to the prefix of the <code>BlastTab.catsbj.pl</code> output files. At
least the files .rec and .lim must exist with this prefix.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_plot">plot</code></td>
<td>
<p>Should the object be plotted?</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_pos.breaks">pos.breaks</code></td>
<td>
<p>Breaks in the positions histogram. It can also be a vector of break
points, and values outside the range are ignored. If zero (0), it
uses the sequence breaks as defined in the .lim file, which means
one bin per contig (or gene, if the mapping is agains genes). Ignored
if 'pos.breaks.tsv' is passed.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_pos.breaks.tsv">pos.breaks.tsv</code></td>
<td>
<p>Path to a list of (absolute) coordinates to use as position breaks.
This tab-delimited file can be produced by <code>GFF.catsbj.pl</code>, and it
must contain at least one column: coordinates of the break positions of
each position bin. If it has a second column, this is used as the name
of the position bin that ends at the given coordinate (the first row is
ignored). Any additional columns are currently ignored. If <code>NA</code>,
position bins are determined by <code>pos.breaks</code>.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_id.breaks">id.breaks</code></td>
<td>
<p>Breaks in the identity histogram. It can also be a vector of break
points, and values outside the range are ignored.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_id.free.range">id.free.range</code></td>
<td>
<p>Indicates that the range should be freely set from the observed
values. Otherwise, 70-100% is included in the identity histogram
(default).</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_id.metric">id.metric</code></td>
<td>
<p>Metric of identity to be used (Y-axis). Corrected identity is only
supported if the original BLAST file included sequence lengths.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_id.summary">id.summary</code></td>
<td>
<p>Function summarizing the identity bins. Other recommended options
include: <code>median</code> to estimate the median instead of total bins, and
<code>function(x) mlv(x,method='parzen')$M</code> to estimate the mode.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_id.cutoff">id.cutoff</code></td>
<td>
<p>Cutoff of identity metric above which the hits are considered
<code>in-group</code>. The 95% identity corresponds to the expectation of
ANI&lt;95% within species.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_threads">threads</code></td>
<td>
<p>Number of threads to use.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_verbose">verbose</code></td>
<td>
<p>Indicates if the function should report the advance.</p>
</td></tr>
<tr><td><code id="enve.recplot2_+3A_...">...</code></td>
<td>
<p>Any additional parameters supported by <code><a href="#topic+plot.enve.RecPlot2">plot.enve.RecPlot2</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>
<p>Kenji Gerhardt [aut]
</p>

<hr>
<h2 id='enve.RecPlot2-class'>Enveomics: Recruitment Plot (2) - S4 Class</h2><span id='topic+enve.RecPlot2-class'></span><span id='topic+enve.RecPlot2'></span>

<h3>Description</h3>

<p>Enve-omics representation of Recruitment plots. This object can
be produced by <code><a href="#topic+enve.recplot2">enve.recplot2</a></code> and supports S4 method plot.
</p>


<h3>Slots</h3>


<dl>
<dt><code>counts</code></dt><dd><p><code>(matrix)</code> Counts as a two-dimensional histogram.</p>
</dd>
<dt><code>pos.counts.in</code></dt><dd><p><code>(numeric)</code> Counts of in-group hits per position bin.</p>
</dd>
<dt><code>pos.counts.out</code></dt><dd><p><code>(numeric)</code> Counts of out-group hits per position bin.</p>
</dd>
<dt><code>id.counts</code></dt><dd><p><code>(numeric)</code> Counts per ID bin.</p>
</dd>
<dt><code>id.breaks</code></dt><dd><p><code>(numeric)</code> Breaks of identity bins.</p>
</dd>
<dt><code>pos.breaks</code></dt><dd><p><code>(numeric)</code> Breaks of position bins.</p>
</dd>
<dt><code>pos.names</code></dt><dd><p><code>(character)</code> Names of the position bins.</p>
</dd>
<dt><code>seq.breaks</code></dt><dd><p><code>(numeric)</code> Breaks of input sequences.</p>
</dd>
<dt><code>peaks</code></dt><dd><p><code>(list)</code> Peaks identified in the recplot.
Limits of the subject sequences after concatenation.</p>
</dd>
<dt><code>seq.names</code></dt><dd><p><code>(character</code>) Names of the subject sequences.</p>
</dd>
<dt><code>id.metric</code></dt><dd><p><code>(character</code>) Metric used as 'identity'.</p>
</dd>
<dt><code>id.ingroup</code></dt><dd><p><code>(logical</code>) Identity bins considered in-group.</p>
</dd>
<dt><code>call</code></dt><dd><p><code>(call)</code> Call producing this object.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.__counts'>Enveomics: Recruitment Plot (2) Internal Ancillary Function</h2><span id='topic+enve.recplot2.__counts'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.recplot2">enve.recplot2</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.__counts(x, pos.breaks, id.breaks, rec.idcol)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.__counts_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object</p>
</td></tr>
<tr><td><code id="enve.recplot2.__counts_+3A_pos.breaks">pos.breaks</code></td>
<td>
<p>Position breaks</p>
</td></tr>
<tr><td><code id="enve.recplot2.__counts_+3A_id.breaks">id.breaks</code></td>
<td>
<p>Identity breaks</p>
</td></tr>
<tr><td><code id="enve.recplot2.__counts_+3A_rec.idcol">rec.idcol</code></td>
<td>
<p>Identity column to use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>2-dimensional matrix of counts per identity and position bins.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>
<p>Kenji Gerhardt [aut]
</p>

<hr>
<h2 id='enve.recplot2.__peakHist'>Enveomics: Recruitment Plot (2) Peak S4 Class - Internal Ancillary Function</h2><span id='topic+enve.recplot2.__peakHist'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.__peakHist(x, mids, counts = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.__peakHist_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> object</p>
</td></tr>
<tr><td><code id="enve.recplot2.__peakHist_+3A_mids">mids</code></td>
<td>
<p>Midpoints</p>
</td></tr>
<tr><td><code id="enve.recplot2.__peakHist_+3A_counts">counts</code></td>
<td>
<p>Counts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of counts (histogram)
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.__whichClosestPeak'>Enveomics: Recruitment Plot (2) Peak Finder - Internal Ancillary Function</h2><span id='topic+enve.recplot2.__whichClosestPeak'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.__whichClosestPeak(peak, peaks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.__whichClosestPeak_+3A_peak">peak</code></td>
<td>
<p>Query <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> object</p>
</td></tr>
<tr><td><code id="enve.recplot2.__whichClosestPeak_+3A_peaks">peaks</code></td>
<td>
<p>list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric index out of <code>peaks</code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.ANIr'>Enveomics: Recruitment Plot (2) ANI Estimate</h2><span id='topic+enve.recplot2.ANIr'></span>

<h3>Description</h3>

<p>Estimate the Average Nucleotide Identity from reads (ANIr) from a
recruitment plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.ANIr(x, range = c(0, Inf))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.ANIr_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.ANIr_+3A_range">range</code></td>
<td>
<p>Range of identities to be considered. By default, the full range
is used (note that the upper boundary is <code>Inf</code> and not 100 because
recruitment plots can also be built with bit-scores). To use only
intra-population matches (with identities), use <code>c(95, 100)</code>. To use
only inter-population values, use <code>c(0, 95)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value indicating the ANIr (as percentage).
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.changeCutoff'>Enveomics: Recruitment Plot (2) Change Cutoff</h2><span id='topic+enve.recplot2.changeCutoff'></span>

<h3>Description</h3>

<p>Change the intra-species cutoff of an existing recruitment plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.changeCutoff(rp, new.cutoff = 98)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.changeCutoff_+3A_rp">rp</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.changeCutoff_+3A_new.cutoff">new.cutoff</code></td>
<td>
<p>New cutoff to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The modified <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.compareIdentities'>Enveomics: Recruitment Plot (2) Compare Identities</h2><span id='topic+enve.recplot2.compareIdentities'></span>

<h3>Description</h3>

<p>Compare the distribution of identities between two
<code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.compareIdentities(
  x,
  y,
  method = "hellinger",
  smooth.par = NULL,
  pseudocounts = 0,
  max.deviation = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.compareIdentities_+3A_x">x</code></td>
<td>
<p>First <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.compareIdentities_+3A_y">y</code></td>
<td>
<p>Second <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.compareIdentities_+3A_method">method</code></td>
<td>
<p>Distance method to use. This should be (an unambiguous abbreviation of)
one of:
</p>

<ul>
<li><p>&quot;hellinger&quot;
(<em>Hellinger, 1090, doi:10.1515/crll.1909.136.210</em>),
</p>
</li>
<li><p>&quot;bhattacharyya&quot;
(<em>Bhattacharyya, 1943, Bull. Calcutta Math. Soc. 35</em>),
</p>
</li>
<li><p>&quot;kl&quot; or &quot;kullback-leibler&quot;
(<em>Kullback &amp; Leibler, 1951, doi:10.1214/aoms/1177729694</em>), or
</p>
</li>
<li><p>&quot;euclidean&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="enve.recplot2.compareIdentities_+3A_smooth.par">smooth.par</code></td>
<td>
<p>Smoothing parameter for cubic spline smoothing. Use 0 for no smoothing.
Use <code>NULL</code> to automatically determine this value using leave-one-out
cross-validation (see <code>smooth.spline</code> parameter <code>spar</code>).</p>
</td></tr>
<tr><td><code id="enve.recplot2.compareIdentities_+3A_pseudocounts">pseudocounts</code></td>
<td>
<p>Smoothing parameter for Laplace smoothing. Use 0 for no smoothing, or
1 for add-one smoothing.</p>
</td></tr>
<tr><td><code id="enve.recplot2.compareIdentities_+3A_max.deviation">max.deviation</code></td>
<td>
<p>Maximum mean deviation between identity breaks tolerated (as percent
identity). Difference in number of <code>id.breaks</code> is never tolerated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <strong>numeric</strong> indicating the distance between the objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.coordinates'>Enveomics: Recruitment Plot (2) Coordinates</h2><span id='topic+enve.recplot2.coordinates'></span>

<h3>Description</h3>

<p>Returns the sequence name and coordinates of the requested position bins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.coordinates(x, bins)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.coordinates_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.coordinates_+3A_bins">bins</code></td>
<td>
<p>Vector of selected bins to return. It can be a vector of logical values
with the same length as <code>x$pos.breaks-1</code> or a vector of integers. If
missing, returns the coordinates of all windows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with five columns: <code>name.from</code> (character),
<code>pos.from</code> (numeric), <code>name.to</code> (character), <code>pos.to</code>
(numeric), and <code>seq.name</code> (character).
The first two correspond to sequence and position of the start point of the
bin. The next two correspond to the sequence and position of the end point of
the bin. The last one indicates the name of the sequence (if defined).
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.corePeak'>Enveomics: Recruitment Plot (2) Core Peak Finder</h2><span id='topic+enve.recplot2.corePeak'></span>

<h3>Description</h3>

<p>Finds the peak in a list of peaks that is most likely to represent the
&quot;core genome&quot; of a population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.corePeak(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.corePeak_+3A_x">x</code></td>
<td>
<p><code>list</code> of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.extractWindows'>Enveomics: Recruitment Plot (2) Extract Windows</h2><span id='topic+enve.recplot2.extractWindows'></span>

<h3>Description</h3>

<p>Extract windows significantly below (or above) the peak in sequencing
depth.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.extractWindows(
  rp,
  peak,
  lower.tail = TRUE,
  significance = 0.05,
  seq.names = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.extractWindows_+3A_rp">rp</code></td>
<td>
<p>Recruitment plot, a <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.extractWindows_+3A_peak">peak</code></td>
<td>
<p>Peak, an <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> object. If list, it is assumed to
be a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects, in which case the core
peak is used (see <code><a href="#topic+enve.recplot2.corePeak">enve.recplot2.corePeak</a></code>).</p>
</td></tr>
<tr><td><code id="enve.recplot2.extractWindows_+3A_lower.tail">lower.tail</code></td>
<td>
<p>If <code>FALSE</code>, it returns windows significantly above the peak in
sequencing depth.</p>
</td></tr>
<tr><td><code id="enve.recplot2.extractWindows_+3A_significance">significance</code></td>
<td>
<p>Significance threshold (alpha) to select windows.</p>
</td></tr>
<tr><td><code id="enve.recplot2.extractWindows_+3A_seq.names">seq.names</code></td>
<td>
<p>Returns subject sequence names instead of a vector of Booleans. If
the recruitment plot was generated with named position bins (e.g, using
<code>pos.breaks=0</code> or a two-column <code>pos.breaks.tsv</code>), it returns a
vector of characters (the sequence identifiers), otherwise it returns a
data.frame with a name column and two columns of coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector of logicals if <code>seq.names = FALSE</code>.
If <code>seq.names = TRUE</code>, it returns a data.frame with five columns:
<code>name.from</code>, <code>name.to</code>, <code>pos.from</code>, <code>pos.to</code>, and
<code>seq.name</code> (see <code><a href="#topic+enve.recplot2.coordinates">enve.recplot2.coordinates</a></code>).
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks'>Enveomics: Recruitment Plot (2) Peak Finder</h2><span id='topic+enve.recplot2.findPeaks'></span>

<h3>Description</h3>

<p>Identifies peaks in the population histogram potentially indicating
sub-population mixtures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks(x, method = "emauto", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks_+3A_method">method</code></td>
<td>
<p>Peak-finder method. This should be one of:
</p>

<ul>
<li> <p><strong>emauto</strong>
(Expectation-Maximization with auto-selection of components)
</p>
</li>
<li> <p><strong>em</strong>
(Expectation-Maximization)
</p>
</li>
<li> <p><strong>mower</strong>
(Custom distribution-mowing method)
</p>
</li></ul>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks_+3A_...">...</code></td>
<td>
<p>Any additional parameters supported by
<code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.__em_e'>Enveomics: Recruitment Plot (2) EM Peak Finder - Internal Ancillary Function Expectation</h2><span id='topic+enve.recplot2.findPeaks.__em_e'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.recplot2.findPeaks.em">enve.recplot2.findPeaks.em</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.__em_e(x, theta)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.__em_e_+3A_x">x</code></td>
<td>
<p>Vector of log-transformed sequencing depths</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__em_e_+3A_theta">theta</code></td>
<td>
<p>Parameters list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with components <code>ll</code> (numeric) the log-likelihood, and
<code>posterior</code> (numeric) the posterior probability.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.__em_m'>Enveomics: Recruitment Plot (2) Em Peak Finder - Internal Ancillary Function Maximization</h2><span id='topic+enve.recplot2.findPeaks.__em_m'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.recplot2.findPeaks.em">enve.recplot2.findPeaks.em</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.__em_m(x, posterior)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.__em_m_+3A_x">x</code></td>
<td>
<p>Vector of log-transformed sequencing depths</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__em_m_+3A_posterior">posterior</code></td>
<td>
<p>Posterior probability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with components <code>mu</code> (numeric) the estimated mean,
<code>sd</code> (numeric) the estimated standard deviation, and <code>alpha</code>
(numeric) the estimated alpha parameter.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.__emauto_one'>Enveomics: Recruitment Plot (2) EMauto Peak Finder - Internal Ancillary Function</h2><span id='topic+enve.recplot2.findPeaks.__emauto_one'></span>

<h3>Description</h3>

<p>Internal ancillary function (see
<code><a href="#topic+enve.recplot2.findPeaks.emauto">enve.recplot2.findPeaks.emauto</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.__emauto_one(x, comp, do_crit, best, verbose, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_comp">comp</code></td>
<td>
<p>Components.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_do_crit">do_crit</code></td>
<td>
<p>Function estimating the criterion.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_best">best</code></td>
<td>
<p>Best solution thus far.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_verbose">verbose</code></td>
<td>
<p>If verbose.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__emauto_one_+3A_...">...</code></td>
<td>
<p>Additional parameters for <code><a href="#topic+enve.recplot2.findPeaks.em">enve.recplot2.findPeaks.em</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated solution with the same structure as <code>best</code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.__mow_one'>Enveomics: Recruitment Plot (2) Mowing Peak Finder - Internal Ancillary Function 1</h2><span id='topic+enve.recplot2.findPeaks.__mow_one'></span>

<h3>Description</h3>

<p>Internal ancillary function (see
<code><a href="#topic+enve.recplot2.findPeaks.mower">enve.recplot2.findPeaks.mower</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.__mow_one(
  lsd1,
  min.points,
  quant.est,
  mlv.opts,
  fitdist.opts,
  with.skewness,
  optim.rounds,
  optim.epsilon,
  n.total,
  merge.logdist,
  verbose,
  log
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_lsd1">lsd1</code></td>
<td>
<p>Vector of log-transformed sequencing depths</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_min.points">min.points</code></td>
<td>
<p>Minimum number of points</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_quant.est">quant.est</code></td>
<td>
<p>Quantile estimate</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_mlv.opts">mlv.opts</code></td>
<td>
<p>List of options for <code>mlv</code></p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_fitdist.opts">fitdist.opts</code></td>
<td>
<p>List of options for <code>fitdist</code></p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_with.skewness">with.skewness</code></td>
<td>
<p>If skewed-normal should be used</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_optim.rounds">optim.rounds</code></td>
<td>
<p>Maximum number of optimization rounds</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_optim.epsilon">optim.epsilon</code></td>
<td>
<p>Minimum difference considered negligible</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_n.total">n.total</code></td>
<td>
<p>Global number of windows</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_merge.logdist">merge.logdist</code></td>
<td>
<p>Attempted <code>merge.logdist</code> parameter</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_verbose">verbose</code></td>
<td>
<p>If verbose</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.__mow_one_+3A_log">log</code></td>
<td>
<p>If log-transformed depths</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return an <code>enve.RecPlot2.Peak</code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.__mower'>Enveomics: Recruitment Plot (2) Mowing Peak Finder - Internal Ancillary Function 2</h2><span id='topic+enve.recplot2.findPeaks.__mower'></span>

<h3>Description</h3>

<p>Internal ancillary function (see <code><a href="#topic+enve.recplot2.findPeaks.mower">enve.recplot2.findPeaks.mower</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.__mower(peaks.opts)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.__mower_+3A_peaks.opts">peaks.opts</code></td>
<td>
<p>List of options for <code><a href="#topic+enve.recplot2.findPeaks.__mow_one">enve.recplot2.findPeaks.__mow_one</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of <code>enve.RecPlot2.Peak</code> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.em'>Enveomics: Recruitment Plot (2) Em Peak Finder</h2><span id='topic+enve.recplot2.findPeaks.em'></span>

<h3>Description</h3>

<p>Identifies peaks in the population histogram using a Gaussian Mixture
Model Expectation Maximization (GMM-EM) method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.em(
  x,
  max.iter = 1000,
  ll.diff.res = 1e-08,
  components = 2,
  rm.top = 0.05,
  verbose = FALSE,
  init,
  log = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_max.iter">max.iter</code></td>
<td>
<p>Maximum number of EM iterations.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_ll.diff.res">ll.diff.res</code></td>
<td>
<p>Maximum Log-Likelihood difference to be considered as convergent.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_components">components</code></td>
<td>
<p>Number of distributions assumed in the mixture.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_rm.top">rm.top</code></td>
<td>
<p>Top-values to remove before finding peaks, as a quantile probability.
This step is useful to remove highly conserved regions, but can be
turned off by setting <code>rm.top=0</code>. The quantile is determined
<strong>after</strong> removing zero-coverage windows.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_verbose">verbose</code></td>
<td>
<p>Display (mostly debugging) information.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_init">init</code></td>
<td>
<p>Initialization parameters. By default, these are derived from k-means
clustering. A named list with vectors for <code>mu</code>, <code>sd</code>, and
<code>alpha</code>, each of length <code>components</code>.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.em_+3A_log">log</code></td>
<td>
<p>Logical value indicating if the estimations should be performed in
natural logarithm units. Do not change unless you know what you're
doing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.emauto'>Enveomics: Recruitment Plot (2) Emauto Peak Finder</h2><span id='topic+enve.recplot2.findPeaks.emauto'></span>

<h3>Description</h3>

<p>Identifies peaks in the population histogram using a Gaussian Mixture
Model Expectation Maximization (GMM-EM) method with number of components
automatically detected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.emauto(
  x,
  components = seq(1, 5),
  criterion = "aic",
  merge.tol = 2L,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_components">components</code></td>
<td>
<p>A vector of number of components to evaluate.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_criterion">criterion</code></td>
<td>
<p>Criterion to use for components selection. Must be one of:
<code>aic</code> (Akaike Information Criterion), <code>bic</code> or <code>sbc</code>
(Bayesian Information Criterion or Schwarz Criterion).</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_merge.tol">merge.tol</code></td>
<td>
<p>When attempting to merge peaks with very similar sequencing depth, use
this number of significant digits (in log-scale).</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_verbose">verbose</code></td>
<td>
<p>Display (mostly debugging) information.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.emauto_+3A_...">...</code></td>
<td>
<p>Any additional parameters supported by
<code><a href="#topic+enve.recplot2.findPeaks.em">enve.recplot2.findPeaks.em</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.findPeaks.mower'>Enveomics: Recruitment Plot (2) Mowing Peak Finder</h2><span id='topic+enve.recplot2.findPeaks.mower'></span>

<h3>Description</h3>

<p>Identifies peaks in the population histogram potentially indicating
sub-population mixtures, using a custom distribution-mowing method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.findPeaks.mower(
  x,
  min.points = 10,
  quant.est = c(0.002, 0.998),
  mlv.opts = list(method = "parzen"),
  fitdist.opts.sn = list(distr = "sn", method = "qme", probs = c(0.1, 0.5, 0.8), start =
    list(omega = 1, alpha = -1), lower = c(0, -Inf, -Inf)),
  fitdist.opts.norm = list(distr = "norm", method = "qme", probs = c(0.4, 0.6), start =
    list(sd = 1), lower = c(0, -Inf)),
  rm.top = 0.05,
  with.skewness = TRUE,
  optim.rounds = 200,
  optim.epsilon = 1e-04,
  merge.logdist = log(1.75),
  verbose = FALSE,
  log = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_min.points">min.points</code></td>
<td>
<p>Minimum number of points in the quantile-estimation-range
<code>(quant.est)</code> to estimate a peak.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_quant.est">quant.est</code></td>
<td>
<p>Range of quantiles to be used in the estimation of a peak's
parameters.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_mlv.opts">mlv.opts</code></td>
<td>
<p>Ignored. For backwards compatibility.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_fitdist.opts.sn">fitdist.opts.sn</code></td>
<td>
<p>Options passed to <code>fitdist</code> to estimate the standard deviation if
<code>with.skewness=TRUE</code>. Note that the <code>start</code> parameter will be
ammended with <code>xi=estimated</code> mode for each peak.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_fitdist.opts.norm">fitdist.opts.norm</code></td>
<td>
<p>Options passed to <code>fitdist</code> to estimate the standard deviation if
<code>with.skewness=FALSE</code>. Note that the <code>start</code> parameter will be
ammended with <code>mean=estimated</code> mode for each peak.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_rm.top">rm.top</code></td>
<td>
<p>Top-values to remove before finding peaks, as a quantile probability.
This step is useful to remove highly conserved regions, but can be
turned off by setting <code>rm.top=0</code>. The quantile is determined
<strong>after</strong> removing zero-coverage windows.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_with.skewness">with.skewness</code></td>
<td>
<p>Allow skewness correction of the peaks. Typically, the
sequencing-depth distribution for a single peak is left-skewed, due
partly (but not exclusively) to fragmentation and mapping sensitivity.
See <em>Lindner et al 2013, Bioinformatics 29(10):1260-7</em> for an
alternative solution for the first problem (fragmentation) called
&quot;tail distribution&quot;.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_optim.rounds">optim.rounds</code></td>
<td>
<p>Maximum rounds of peak optimization.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_optim.epsilon">optim.epsilon</code></td>
<td>
<p>Trace change at which optimization stops (unless <code>optim.rounds</code> is
reached first). The trace change is estimated as the sum of square
differences between parameters in one round and those from two rounds
earlier (to avoid infinite loops from approximation).</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_merge.logdist">merge.logdist</code></td>
<td>
<p>Maximum value of <code>|log-ratio|</code> between centrality parameters in peaks
to attempt merging. The default of ~0.22 corresponds to a maximum
difference of 25%.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_verbose">verbose</code></td>
<td>
<p>Display (mostly debugging) information.</p>
</td></tr>
<tr><td><code id="enve.recplot2.findPeaks.mower_+3A_log">log</code></td>
<td>
<p>Logical value indicating if the estimations should be performed in
natural logarithm units. Do not change unless you know what you're
doing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.RecPlot2.Peak-class'>Enveomics: Recruitment Plot (2) Peak - S4 Class</h2><span id='topic+enve.RecPlot2.Peak-class'></span><span id='topic+enve.RecPlot2.Peak'></span>

<h3>Description</h3>

<p>Enve-omics representation of a peak in the sequencing depth histogram
of a Recruitment plot (see <code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>).
</p>


<h3>Slots</h3>


<dl>
<dt><code>dist</code></dt><dd><p><code>(character)</code>
Distribution of the peak. Currently supported: <code>norm</code> (normal) and <code>sn</code>
(skew-normal).</p>
</dd>
<dt><code>values</code></dt><dd><p><code>(numeric)</code>
Sequencing depth values predicted to conform the peak.</p>
</dd>
<dt><code>values.res</code></dt><dd><p><code>(numeric)</code>
Sequencing depth values not explained by this or previously identified
peaks.</p>
</dd>
<dt><code>mode</code></dt><dd><p><code>(numeric)</code>
Seed-value of mode anchoring the peak.</p>
</dd>
<dt><code>param.hat</code></dt><dd><p><code>(list)</code>
Parameters of the distribution. A list of two values if dist=<code>norm</code> (sd
and mean), or three values if dist=<code>sn</code>(omega=scale, alpha=shape, and
xi=location). Note that the &quot;dispersion&quot; parameter is always first and
the &quot;location&quot; parameter is always last.</p>
</dd>
<dt><code>n.hat</code></dt><dd><p><code>(numeric)</code>
Number of bins estimated to be explained by this peak. This should
ideally be equal to the length of  <code>values</code>, but it's not an integer.</p>
</dd>
<dt><code>n.total</code></dt><dd><p><code>(numeric)</code>
Total number of bins from which the peak was extracted. I.e., total
number of position bins with non-zero sequencing depth in the recruitment
plot (regardless of peak count).</p>
</dd>
<dt><code>err.res</code></dt><dd><p><code>(numeric)</code>
Error left after adding the peak (mower) or log-likelihood (em or emauto).</p>
</dd>
<dt><code>merge.logdist</code></dt><dd><p><code>(numeric)</code>
Attempted <code>merge.logdist</code> parameter.</p>
</dd>
<dt><code>seq.depth</code></dt><dd><p><code>(numeric)</code>
Best estimate available for the sequencing depth of the peak (centrality).</p>
</dd>
<dt><code>log</code></dt><dd><p><code>(logical)</code>
Indicates if the estimation was performed in natural logarithm space.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.seqdepth'>Enveomics: Recruitment Plot (2) Sequencing Depth</h2><span id='topic+enve.recplot2.seqdepth'></span>

<h3>Description</h3>

<p>Calculate the sequencing depth of the given window(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.seqdepth(x, sel, low.identity = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.seqdepth_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.seqdepth_+3A_sel">sel</code></td>
<td>
<p>Window(s) for which the sequencing depth is to be calculated. If not
passed, it returns the sequencing depth of all windows.</p>
</td></tr>
<tr><td><code id="enve.recplot2.seqdepth_+3A_low.identity">low.identity</code></td>
<td>
<p>A logical indicating if the sequencing depth is to be estimated only
with low-identity matches. By default, only high-identity matches are
used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector of sequencing depths (in bp/bp).
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.recplot2.windowDepthThreshold'>Enveomics: Recruitment Plot (2) Window Depth Threshold</h2><span id='topic+enve.recplot2.windowDepthThreshold'></span>

<h3>Description</h3>

<p>Identifies the threshold below which windows should be identified as
variable or absent.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.recplot2.windowDepthThreshold(
  rp,
  peak,
  lower.tail = TRUE,
  significance = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.recplot2.windowDepthThreshold_+3A_rp">rp</code></td>
<td>
<p>Recruitment plot, an <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.recplot2.windowDepthThreshold_+3A_peak">peak</code></td>
<td>
<p>Peak, an <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> object. If list, it is assumed to
be a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects, in which case the core
peak is used (see <code><a href="#topic+enve.recplot2.corePeak">enve.recplot2.corePeak</a></code>).</p>
</td></tr>
<tr><td><code id="enve.recplot2.windowDepthThreshold_+3A_lower.tail">lower.tail</code></td>
<td>
<p>If <code>FALSE</code>, it returns windows significantly above the peak in
sequencing depth.</p>
</td></tr>
<tr><td><code id="enve.recplot2.windowDepthThreshold_+3A_significance">significance</code></td>
<td>
<p>Significance threshold (alpha) to select windows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a float. The units are depth if the peaks were estimated in
linear scale, or log-depth otherwise (<code>peak$log</code>).
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.selvector'>Enveomics: Selection vector</h2><span id='topic+enve.selvector'></span>

<h3>Description</h3>

<p>Normalizes a selection vector <code>sel</code> to a logical vector with indexes
from <code>dim.names</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.selvector(sel, dim.names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.selvector_+3A_sel">sel</code></td>
<td>
<p>A vector of numbers, characters, or booleans.</p>
</td></tr>
<tr><td><code id="enve.selvector_+3A_dim.names">dim.names</code></td>
<td>
<p>A vector of names from which to select.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a logical vector with the same length as <code>dim.name</code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.tribs'>Enveomics: TRIBS</h2><span id='topic+enve.tribs'></span>

<h3>Description</h3>

<p>Subsample any objects in &quot;distance space&quot; to reduce the effect of
sample-clustering. This function was originally designed to subsample
genomes in &quot;phylogenetic distance space&quot;, a clear case of strong
clustering bias in sampling, by Luis M. Rodriguez-R and Michael R
Weigand.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.tribs(
  dist,
  selection = labels(dist),
  replicates = 1000,
  summary.fx = median,
  dist.method = "euclidean",
  subsamples = seq(0, 1, by = 0.01),
  dimensions = ceiling(length(selection) * 0.05),
  metaMDS.opts = list(),
  threads = 2,
  verbosity = 1,
  points,
  pre.tribs
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.tribs_+3A_dist">dist</code></td>
<td>
<p>Distances as a <code>dist</code> object.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_selection">selection</code></td>
<td>
<p>Objects to include in the subsample. By default, all objects are
selected.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_replicates">replicates</code></td>
<td>
<p>Number of replications per point.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_summary.fx">summary.fx</code></td>
<td>
<p>Function to summarize the distance distributions in a given replicate. By
default, the median distance is estimated.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_dist.method">dist.method</code></td>
<td>
<p>Distance method between random points and samples in the transformed
space. See <code>dist</code>.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_subsamples">subsamples</code></td>
<td>
<p>Subsampling fractions.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_dimensions">dimensions</code></td>
<td>
<p>Dimensions to use in the NMDS. By default, 5% of the selection length.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_metamds.opts">metaMDS.opts</code></td>
<td>
<p>Any additional options to pass to metaMDS, as <code>list</code>.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_threads">threads</code></td>
<td>
<p>Number of threads to use.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_verbosity">verbosity</code></td>
<td>
<p>Verbosity. Use 0 to run quietly, increase for additional information.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_points">points</code></td>
<td>
<p>Optional. If passed, the MDS step is skipped and this object is used
instead.  It can be the <code>$points</code> slot of class <code>metaMDS</code> 
(from <code>vegan</code>).
It must be a matrix or matrix-coercible object, with samples as rows and
dimensions as columns.</p>
</td></tr>
<tr><td><code id="enve.tribs_+3A_pre.tribs">pre.tribs</code></td>
<td>
<p>Optional. If passed, the points are recovered from this object (except if
<code>points</code> is also passed. This should be an <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code>
object estimated on the same objects (the selection is unimportant).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.TRIBS-class'>Enveomics: TRIBS S4 Class</h2><span id='topic+enve.TRIBS-class'></span><span id='topic+enve.TRIBS'></span>

<h3>Description</h3>

<p>Enve-omics representation of &quot;Transformed-space Resampling In Biased Sets
(TRIBS)&quot;. This object represents sets of distances between objects,
sampled nearly-uniformly at random in &quot;distance space&quot;. Subsampling
without selection is trivial, since both the distances space and the
selection occur in the same transformed space. However, it's useful to
compare randomly subsampled sets against a selected set of objects. This
is intended to identify overdispersion or overclustering (see
<code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code>) of a subset against the entire collection of
objects with minimum impact of sampling biases. This object can be produced
by <code><a href="#topic+enve.tribs">enve.tribs</a></code> and supports S4 methods <code>plot</code> and
<code>summary</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>distance</code></dt><dd><p><code>(numeric)</code> Centrality measurement of the distances 
between the selected objects (without subsampling).</p>
</dd>
<dt><code>points</code></dt><dd><p><code>(matrix)</code>    Position of the different objects in distance
space.</p>
</dd>
<dt><code>distances</code></dt><dd><p><code>(matrix)</code> Subsampled distances, where the rows are 
replicates and the columns are subsampling levels.</p>
</dd>
<dt><code>spaceSize</code></dt><dd><p><code>(numeric)</code> Number of objects.</p>
</dd>
<dt><code>selSize</code></dt><dd><p><code>(numeric)</code> Number of selected objects.</p>
</dd>
<dt><code>dimensions</code></dt><dd><p><code>(numeric)</code> Number of dimensions in the distance space.</p>
</dd>
<dt><code>subsamples</code></dt><dd><p><code>(numeric)</code> Subsampling levels (as fractions, from 
0 to 1).</p>
</dd>
<dt><code>call</code></dt><dd><p><code>(call)</code> Call producing this object.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.TRIBS.merge'>Enveomics: TRIBS Merge</h2><span id='topic+enve.TRIBS.merge'></span>

<h3>Description</h3>

<p>Merges two <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> objects generated from the same objects
at different subsampling levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.TRIBS.merge(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.TRIBS.merge_+3A_x">x</code></td>
<td>
<p>First <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.</p>
</td></tr>
<tr><td><code id="enve.TRIBS.merge_+3A_y">y</code></td>
<td>
<p>Second <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.tribs.test'>Enveomics: TRIBS Test</h2><span id='topic+enve.tribs.test'></span>

<h3>Description</h3>

<p>Estimates the empirical difference between all the distances in a set of
objects and a subset, together with its statistical significance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.tribs.test(dist, selection, bins = 50, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.tribs.test_+3A_dist">dist</code></td>
<td>
<p>Distances as <code>dist</code> object.</p>
</td></tr>
<tr><td><code id="enve.tribs.test_+3A_selection">selection</code></td>
<td>
<p>Selection defining the subset.</p>
</td></tr>
<tr><td><code id="enve.tribs.test_+3A_bins">bins</code></td>
<td>
<p>Number of bins to evaluate in the range of distances.</p>
</td></tr>
<tr><td><code id="enve.tribs.test_+3A_...">...</code></td>
<td>
<p>Any other parameters supported by <code><a href="#topic+enve.tribs">enve.tribs</a></code>, 
except <code>subsamples</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.TRIBStest-class'>Enveomics: TRIBS Test S4 Class</h2><span id='topic+enve.TRIBStest-class'></span><span id='topic+enve.TRIBStest'></span>

<h3>Description</h3>

<p>Test of significance of overclustering or overdispersion in a selected
set of objects with respect to the entire set (see <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code>).
This object can be produced by <code><a href="#topic+enve.tribs.test">enve.tribs.test</a></code> and supports S4
methods <code>plot</code> and <code>summary</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>pval.gt</code></dt><dd><p><code>(numeric)</code>
P-value for the overdispersion test.</p>
</dd>
<dt><code>pval.lt</code></dt><dd><p><code>(numeric)</code>
P-value for the overclustering test.</p>
</dd>
<dt><code>all.dist</code></dt><dd><p><code>(numeric)</code>
Empiric PDF of distances for the entire dataset (subsampled at selection
size).</p>
</dd>
<dt><code>sel.dist</code></dt><dd><p><code>(numeric)</code>
Empiric PDF of distances for the selected objects (without subsampling).</p>
</dd>
<dt><code>diff.dist</code></dt><dd><p><code>(numeric)</code>
Empiric PDF of the difference between <code>all.dist</code> and <code>sel.dist</code>. 
The p-values are estimating by comparing areas in this PDF greater than and
lesser than zero.</p>
</dd>
<dt><code>dist.mids</code></dt><dd><p><code>(numeric)</code>
Midpoints of the empiric PDFs of distances.</p>
</dd>
<dt><code>diff.mids</code></dt><dd><p><code>(numeric)</code>
Midpoints of the empiric PDF of difference of distances.</p>
</dd>
<dt><code>call</code></dt><dd><p><code>(call)</code>
Call producing this object.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='enve.truncate'>Enveomics: Truncate</h2><span id='topic+enve.truncate'></span>

<h3>Description</h3>

<p>Removes the <code>n</code> highest and lowest values from a vector, and applies
summary function. The value of <code>n</code> is determined such that the central
range is used, corresponding to the <code>f</code> fraction of values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enve.truncate(x, f = 0.95, FUN = mean)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="enve.truncate_+3A_x">x</code></td>
<td>
<p>A vector of numbers.</p>
</td></tr>
<tr><td><code id="enve.truncate_+3A_f">f</code></td>
<td>
<p>The fraction of values to retain.</p>
</td></tr>
<tr><td><code id="enve.truncate_+3A_fun">FUN</code></td>
<td>
<p>Summary function to apply to the vectors. To obtain the 
truncated vector itself, use <code>c</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the summary <code>(FUN)</code> of the truncated vector.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='growth.curves'>Bacterial growth curves for three Escherichia coli mutants</h2><span id='topic+growth.curves'></span>

<h3>Description</h3>

<p>This data set provides time (first column) and three triplicated growth
curves as optical density at 600nm (OD_600nm) for different mutants of E.
coli.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>growth.curves</code></pre>


<h3>Format</h3>

<p>A data frame with 16 rows (times) and 10 rows (times and OD_600nm).</p>

<hr>
<h2 id='phyla.counts'>Counts of microbial phyla in four sites</h2><span id='topic+phyla.counts'></span>

<h3>Description</h3>

<p>This data set gives the counts of phyla in three different
sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyla.counts</code></pre>


<h3>Format</h3>

<p>A data frame with 9 rows (phyla) and 4 rows (sites).</p>

<hr>
<h2 id='plot.enve.GrowthCurve'>Enveomics: Plot of Growth Curve</h2><span id='topic+plot.enve.GrowthCurve'></span>

<h3>Description</h3>

<p>Plots an <code><a href="#topic+enve.GrowthCurve">enve.GrowthCurve</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.GrowthCurve'
plot(
  x,
  col,
  samples,
  pt.alpha = 0.9,
  ln.alpha = 1,
  ln.lwd = 1,
  ln.lty = 1,
  band.alpha = 0.4,
  band.density = NULL,
  band.angle = 45,
  xp.alpha = 0.5,
  xp.lwd = 1,
  xp.lty = 1,
  pch = 19,
  new = TRUE,
  legend = new,
  add.params = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.enve.GrowthCurve_+3A_x">x</code></td>
<td>
<p>An <code><a href="#topic+enve.GrowthCurve">enve.GrowthCurve</a></code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_col">col</code></td>
<td>
<p>Base colors to use for the different samples. Can be recycled.
By default, grey for one sample or rainbow colors for more than one.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_samples">samples</code></td>
<td>
<p>Vector of sample names to plot. By default: plot all samples.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_pt.alpha">pt.alpha</code></td>
<td>
<p>Color alpha for the observed data points, using <code>col</code>
as a base.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_ln.alpha">ln.alpha</code></td>
<td>
<p>Color alpha for the fitted growth curve, using <code>col</code>
as a base.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_ln.lwd">ln.lwd</code></td>
<td>
<p>Line width for the fitted curve.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_ln.lty">ln.lty</code></td>
<td>
<p>Line type for the fitted curve.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_band.alpha">band.alpha</code></td>
<td>
<p>Color alpha for the confidence interval band of the 
fitted growth curve, using <code>col</code> as a base.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_band.density">band.density</code></td>
<td>
<p>Density of the filling pattern in the interval band.
If <code>NULL</code>, a solid color is used.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_band.angle">band.angle</code></td>
<td>
<p>Angle of the density filling pattern in the interval 
band. Ignored if <code>band.density</code> is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_xp.alpha">xp.alpha</code></td>
<td>
<p>Color alpha for the line connecting individual experiments,
using <code>col</code> as a base.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_xp.lwd">xp.lwd</code></td>
<td>
<p>Width of line for the experiments.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_xp.lty">xp.lty</code></td>
<td>
<p>Type of line for the experiments.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_pch">pch</code></td>
<td>
<p>Point character for observed data points.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_new">new</code></td>
<td>
<p>Should a new plot be generated? If <code>FALSE</code>, the existing
canvas is used.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_legend">legend</code></td>
<td>
<p>Should the plot include a legend? If <code>FALSE</code>, no legend
is added. If <code>TRUE</code>, a legend is added in the bottom-right corner.
Otherwise, a legend is added in the position specified as <code>xy.coords</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_add.params">add.params</code></td>
<td>
<p>Should the legend include the parameters of the fitted 
model?</p>
</td></tr>
<tr><td><code id="plot.enve.GrowthCurve_+3A_...">...</code></td>
<td>
<p>Any other graphic parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='plot.enve.RecPlot2'>Enveomics: Recruitment Plot (2)</h2><span id='topic+plot.enve.RecPlot2'></span>

<h3>Description</h3>

<p>Plots an <code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.RecPlot2'
plot(
  x,
  layout = matrix(c(5, 5, 2, 1, 4, 3), nrow = 2),
  panel.fun = list(),
  widths = c(1, 7, 2),
  heights = c(1, 2),
  palette = grey((100:0)/100),
  underlay.group = TRUE,
  peaks.col = "darkred",
  use.peaks,
  id.lim = range(x$id.breaks),
  pos.lim = range(x$pos.breaks),
  pos.units = c("Mbp", "Kbp", "bp"),
  mar = list(`1` = c(5, 4, 1, 1) + 0.1, `2` = c(ifelse(any(layout == 1), 1, 5), 4, 4, 1)
    + 0.1, `3` = c(5, ifelse(any(layout == 1), 1, 4), 1, 2) + 0.1, `4` =
    c(ifelse(any(layout == 1), 1, 5), ifelse(any(layout == 2), 1, 4), 4, 2) + 0.1, `5` =
    c(5, 3, 4, 1) + 0.1, `6` = c(5, 4, 4, 2) + 0.1),
  pos.splines = 0,
  id.splines = 1/2,
  in.lwd = ifelse(is.null(pos.splines) || pos.splines &gt; 0, 1/2, 2),
  out.lwd = ifelse(is.null(pos.splines) || pos.splines &gt; 0, 1/2, 2),
  id.lwd = ifelse(is.null(id.splines) || id.splines &gt; 0, 1/2, 2),
  in.col = "darkblue",
  out.col = "lightblue",
  id.col = "black",
  breaks.col = "#AAAAAA40",
  peaks.opts = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.enve.RecPlot2_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.RecPlot2">enve.RecPlot2</a></code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_layout">layout</code></td>
<td>
<p>Matrix indicating the position of the different panels in the layout,
where:
</p>

<ul>
<li><p> 0: Empty space
</p>
</li>
<li><p> 1: Counts matrix
</p>
</li>
<li><p> 2: position histogram (sequencing depth)
</p>
</li>
<li><p> 3: identity histogram
</p>
</li>
<li><p> 4: Populations histogram (histogram of sequencing depths)
</p>
</li>
<li><p> 5: Color scale for the counts matrix (vertical)
</p>
</li>
<li><p> 6: Color scale of the counts matrix (horizontal)
</p>
</li></ul>

<p>Only panels indicated here will be plotted. To plot only one panel
simply set this to the number of the panel you want to plot.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_panel.fun">panel.fun</code></td>
<td>
<p>List of functions to be executed after drawing each panel. Use the
indices in <code>layout</code> (as characters) as keys. Functions for indices
missing in <code>layout</code> are ignored. For example, to add a vertical line
at the 3Mbp mark in both the position histogram and the counts matrix:
<code>list('1'=function() abline(v=3), '2'=function() abline(v=3))</code>.
Note that the X-axis in both panels is in Mbp by default. To change
this behavior, set <code>pos.units</code> accordingly.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_widths">widths</code></td>
<td>
<p>Relative widths of the columns of <code>layout</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_heights">heights</code></td>
<td>
<p>Relative heights of the rows of <code>layout</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_palette">palette</code></td>
<td>
<p>Colors to be used to represent the counts matrix, sorted from no hits
to the maximum sequencing depth.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_underlay.group">underlay.group</code></td>
<td>
<p>If TRUE, it indicates the in-group and out-group areas couloured based
on <code>in.col</code> and <code>out.col</code>. Requires support for semi-transparency.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_peaks.col">peaks.col</code></td>
<td>
<p>If not <code>NA</code>, it attempts to represent peaks in the population histogram
in the specified color. Set to <code>NA</code> to avoid peak-finding.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_use.peaks">use.peaks</code></td>
<td>
<p>A list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects, as returned by
<code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>. If passed, <code>peaks.opts</code> is ignored.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_id.lim">id.lim</code></td>
<td>
<p>Limits of identities to represent.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_pos.lim">pos.lim</code></td>
<td>
<p>Limits of positions to represent (in bp, regardless of <code>pos.units</code>).</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_pos.units">pos.units</code></td>
<td>
<p>Units in which the positions should be represented (powers of 1,000
base pairs).</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_mar">mar</code></td>
<td>
<p>Margins of the panels as a list, with the character representation of
the number of the panel as index (see <code>layout</code>).</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_pos.splines">pos.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the position histogram. Zero
(0) for no splines. Use <code>NULL</code> to automatically detect by leave-one-out
cross-validation.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_id.splines">id.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the identity histogram. Zero
(0) for no splines. Use <code>NULL</code> to automatically detect by leave-one-out
cross-validation.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_in.lwd">in.lwd</code></td>
<td>
<p>Line width for the sequencing depth of in-group matches.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_out.lwd">out.lwd</code></td>
<td>
<p>Line width for the sequencing depth of out-group matches.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_id.lwd">id.lwd</code></td>
<td>
<p>Line width for the identity histogram.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_in.col">in.col</code></td>
<td>
<p>Color associated to in-group matches.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_out.col">out.col</code></td>
<td>
<p>Color associated to out-group matches.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_id.col">id.col</code></td>
<td>
<p>Color for the identity histogram.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_breaks.col">breaks.col</code></td>
<td>
<p>Color of the vertical lines indicating sequence breaks.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_peaks.opts">peaks.opts</code></td>
<td>
<p>Options passed to <code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>,
if <code>peaks.col</code> is not <code>NA</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.RecPlot2_+3A_...">...</code></td>
<td>
<p>Any other graphic parameters (currently ignored).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of <code><a href="#topic+enve.RecPlot2.Peak">enve.RecPlot2.Peak</a></code> objects (see
<code><a href="#topic+enve.recplot2.findPeaks">enve.recplot2.findPeaks</a></code>). If <code>peaks.col=NA</code> or
<code>layout</code> doesn't include 4, returns <code>NA</code>.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='plot.enve.TRIBS'>Enveomics: TRIBS Plot</h2><span id='topic+plot.enve.TRIBS'></span>

<h3>Description</h3>

<p>Plot an <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.TRIBS'
plot(
  x,
  new = TRUE,
  type = c("boxplot", "points"),
  col = "#00000044",
  pt.cex = 1/2,
  pt.pch = 19,
  pt.col = col,
  ln.col = col,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.enve.TRIBS_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_new">new</code></td>
<td>
<p>Should a new canvas be drawn?</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_type">type</code></td>
<td>
<p>Type of plot. The <strong>points</strong> plot shows all the replicates, the 
<strong>boxplot</strong> plot represents the values found by
<code><a href="grDevices.html#topic+boxplot.stats">boxplot.stats</a></code>.
as areas, and plots the outliers as points.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_col">col</code></td>
<td>
<p>Color of the areas and/or the points.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_pt.cex">pt.cex</code></td>
<td>
<p>Size of the points.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_pt.pch">pt.pch</code></td>
<td>
<p>Points character.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_pt.col">pt.col</code></td>
<td>
<p>Color of the points.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_ln.col">ln.col</code></td>
<td>
<p>Color of the lines.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBS_+3A_...">...</code></td>
<td>
<p>Any additional parameters supported by <code>plot</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='plot.enve.TRIBStest'>Enveomics: TRIBS Plot Test</h2><span id='topic+plot.enve.TRIBStest'></span>

<h3>Description</h3>

<p>Plots an <code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.TRIBStest'
plot(
  x,
  type = c("overlap", "difference"),
  col = "#00000044",
  col1 = col,
  col2 = "#44001144",
  ylab = "Probability",
  xlim = range(attr(x, "dist.mids")),
  ylim = c(0, max(c(attr(x, "all.dist"), attr(x, "sel.dist")))),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.enve.TRIBStest_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_type">type</code></td>
<td>
<p>What to plot. <code>overlap</code> generates a plot of the two contrasting
empirical PDFs (to compare against each other), <code>difference</code> produces a
plot of the differences between the empirical PDFs (to compare against zero).</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_col">col</code></td>
<td>
<p>Main color of the plot if type=<code>difference</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_col1">col1</code></td>
<td>
<p>First color of the plot if type=<code>overlap</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_col2">col2</code></td>
<td>
<p>Second color of the plot if type=<code>overlap</code>.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_xlim">xlim</code></td>
<td>
<p>X-axis limits.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_ylim">ylim</code></td>
<td>
<p>Y-axis limits.</p>
</td></tr>
<tr><td><code id="plot.enve.TRIBStest_+3A_...">...</code></td>
<td>
<p>Any other graphical arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='summary.enve.GrowthCurve'>Enveomics: Summary of Growth Curve</h2><span id='topic+summary.enve.GrowthCurve'></span>

<h3>Description</h3>

<p>Summary of an <code><a href="#topic+enve.GrowthCurve">enve.GrowthCurve</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.GrowthCurve'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.enve.GrowthCurve_+3A_object">object</code></td>
<td>
<p>An <code><a href="#topic+enve.GrowthCurve">enve.GrowthCurve</a></code> object.</p>
</td></tr>
<tr><td><code id="summary.enve.GrowthCurve_+3A_...">...</code></td>
<td>
<p>No additional parameters are currently supported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='summary.enve.TRIBS'>Enveomics: TRIBS Summary</h2><span id='topic+summary.enve.TRIBS'></span>

<h3>Description</h3>

<p>Summary of an <code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.TRIBS'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.enve.TRIBS_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+enve.TRIBS">enve.TRIBS</a></code> object.</p>
</td></tr>
<tr><td><code id="summary.enve.TRIBS_+3A_...">...</code></td>
<td>
<p>No additional parameters are currently supported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

<hr>
<h2 id='summary.enve.TRIBStest'>Enveomics: TRIBS Summary Test</h2><span id='topic+summary.enve.TRIBStest'></span>

<h3>Description</h3>

<p>Summary of an <code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'enve.TRIBStest'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.enve.TRIBStest_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+enve.TRIBStest">enve.TRIBStest</a></code> object.</p>
</td></tr>
<tr><td><code id="summary.enve.TRIBStest_+3A_...">...</code></td>
<td>
<p>No additional parameters are currently supported.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
