<!DOCTYPE html><html><head><title>Help for package epigraphdb</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {epigraphdb}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#epigraphdb-package'><p>epigraphdb: Interface Package for the 'EpiGraphDB' Platform</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#api_get_request'><p>Wrapper of httr::GET that handles status errors and custom headers</p></a></li>
<li><a href='#api_post_request'><p>Wrapper of httr::POST that handles status errors and custom headers</p></a></li>
<li><a href='#api_request'><p>The very general wrapper from EpiGraphDB endpoint request</p></a></li>
<li><a href='#confounder'><p>MR evidence on confounding traits between exposure and outcome</p></a></li>
<li><a href='#cypher'><p>Send a query in Cypher to EpiGraphDB</p></a></li>
<li><a href='#drugs_risk_factors'><p>Drugs for risk factors</p></a></li>
<li><a href='#flatten_response'><p>Flatten the &quot;results&quot; field from an API response to a tibble df</p></a></li>
<li><a href='#genetic_cor'><p>Genetic correlations between traits</p></a></li>
<li><a href='#http_condition'><p>Modified httr::http_condition</p></a></li>
<li><a href='#literature_gwas'><p>Literature evidence regarding a GWAS trait</p></a></li>
<li><a href='#mappings_gene_to_protein'><p>Return protein uniprot_id from associated genes</p></a></li>
<li><a href='#meta_nodes_list'><p>List meta nodes (e.g. Gwas, Gene, etc.)</p></a></li>
<li><a href='#meta_nodes_list_node'><p>List nodes under a meta node</p></a></li>
<li><a href='#meta_nodes_search_node'><p>Search a node by its id field, or its name field</p></a></li>
<li><a href='#meta_rels_list'><p>List meta rels (e.g. MR, etc.)</p></a></li>
<li><a href='#meta_rels_list_rel'><p>List relationships under a meta relationship</p></a></li>
<li><a href='#mr'><p>Return information related to Mendelian Randomisation</p></a></li>
<li><a href='#obs_cor'><p>Observational correlations between traits</p></a></li>
<li><a href='#ontology_gwas_efo'><p>Ontology association between EFO term and Gwas</p></a></li>
<li><a href='#pathway'><p>Pathway evidence</p></a></li>
<li><a href='#pqtl'><p>Return information related to the pQTL analysis</p></a></li>
<li><a href='#pqtl_list'><p>Return a list of all proteins/exposures or traits/outcomes</p>
available in the database</a></li>
<li><a href='#pqtl_pleio'><p>Return information related to the pleiotropy of SNPs</p></a></li>
<li><a href='#protein_in_pathway'><p>For the list of proteins, returns their associated pathway data</p></a></li>
<li><a href='#query_epigraphdb'><p>Send data request to an EpiGraphDB API endpoint</p></a></li>
<li><a href='#stop_for_status'><p>Catch error for an HTTP response</p></a></li>
<li><a href='#xqtl_multi_snp_mr'><p>Multi SNP QTL MR evidence</p></a></li>
<li><a href='#xqtl_single_snp_mr'><p>Single SNP QTL MR evidence</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interface Package for the 'EpiGraphDB' Platform</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Description:</td>
<td>The interface package to access data from the
    'EpiGraphDB' <a href="https://epigraphdb.org">https://epigraphdb.org</a> platform.
    It provides easy access to the 'EpiGraphDB' platform with functions that
    query the corresponding REST endpoints on the API <a href="https://api.epigraphdb.org">https://api.epigraphdb.org</a>
    and return the response data in the 'tibble' data frame format.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mrcieu.github.io/epigraphdb-r/">https://mrcieu.github.io/epigraphdb-r/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MRCIEU/epigraphdb-r/issues">https://github.com/MRCIEU/epigraphdb-r/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, roxygen2, knitr, rmarkdown, spelling, devtools,
usethis, pkgdown, styler, lintr, covr, igraph, gtools, stringr,
dplyr, ggplot2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, tibble, httr, glue, purrr, jsonlite</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-14 19:42:08 UTC; root</td>
</tr>
<tr>
<td>Author:</td>
<td>Yi Liu [cre, aut],
  Valeriia Haberland [aut],
  Marina Vabistsevits [aut],
  Tom Gaunt [aut],
  MRC IEU [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi Liu &lt;yi6240.liu@bristol.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-14 19:52:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='epigraphdb-package'>epigraphdb: Interface Package for the 'EpiGraphDB' Platform</h2><span id='topic+epigraphdb'></span><span id='topic+epigraphdb-package'></span>

<h3>Description</h3>

<p>The interface package to access data from the 'EpiGraphDB' &lt;https://epigraphdb.org&gt; platform. It provides easy access to the 'EpiGraphDB' platform with functions that query the corresponding REST endpoints on the API &lt;https://api.epigraphdb.org&gt; and return the response data in the 'tibble' data frame format.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yi Liu <a href="mailto:yi6240.liu@bristol.ac.uk">yi6240.liu@bristol.ac.uk</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Valeriia Haberland <a href="mailto:valeriia.haberland@bristol.ac.uk">valeriia.haberland@bristol.ac.uk</a>
</p>
</li>
<li><p> Marina Vabistsevits <a href="mailto:ny19205@bristol.ac.uk">ny19205@bristol.ac.uk</a>
</p>
</li>
<li><p> Tom Gaunt <a href="mailto:Tom.Gaunt@bristol.ac.uk">Tom.Gaunt@bristol.ac.uk</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> MRC IEU [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://mrcieu.github.io/epigraphdb-r/">https://mrcieu.github.io/epigraphdb-r/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/MRCIEU/epigraphdb-r/issues">https://github.com/MRCIEU/epigraphdb-r/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='api_get_request'>Wrapper of httr::GET that handles status errors and custom headers</h2><span id='topic+api_get_request'></span>

<h3>Description</h3>

<p>Wrapper of httr::GET that handles status errors and custom headers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>api_get_request(route, params, retry_times, retry_pause_min)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="api_get_request_+3A_route">route</code></td>
<td>
<p>An EpiGraphDB API endpoint route, e.g. &quot;/mr&quot;</p>
</td></tr>
<tr><td><code id="api_get_request_+3A_params">params</code></td>
<td>
<p>GET request params</p>
</td></tr>
</table>

<hr>
<h2 id='api_post_request'>Wrapper of httr::POST that handles status errors and custom headers</h2><span id='topic+api_post_request'></span>

<h3>Description</h3>

<p>Wrapper of httr::POST that handles status errors and custom headers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>api_post_request(route, params, retry_times, retry_pause_min)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="api_post_request_+3A_route">route</code></td>
<td>
<p>An EpiGraphDB API endpoint route, e.g. &quot;/mr&quot;</p>
</td></tr>
<tr><td><code id="api_post_request_+3A_params">params</code></td>
<td>
<p>POST request payload</p>
</td></tr>
</table>

<hr>
<h2 id='api_request'>The very general wrapper from EpiGraphDB endpoint request</h2><span id='topic+api_request'></span>

<h3>Description</h3>

<p>The very general wrapper from EpiGraphDB endpoint request
</p>


<h3>Usage</h3>

<pre><code class='language-R'>api_request(
  route,
  params,
  mode = c("table", "raw"),
  method = api_get_request,
  retry_times,
  retry_pause_min
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="api_request_+3A_route">route</code></td>
<td>
<p>An EpiGraphDB API endpoint route, e.g. &quot;/mr&quot;</p>
</td></tr>
<tr><td><code id="api_request_+3A_params">params</code></td>
<td>
<p>A list of parameters to send</p>
</td></tr>
<tr><td><code id="api_request_+3A_mode">mode</code></td>
<td>
<p>Either <code>"table"</code> (returns tibble) or
<code>"raw"</code> (returns raw response parsed from json to R list).</p>
</td></tr>
<tr><td><code id="api_request_+3A_method">method</code></td>
<td>
<p>A specific request handler, e.g. <code>epi_get_request</code></p>
</td></tr>
</table>

<hr>
<h2 id='confounder'>MR evidence on confounding traits between exposure and outcome</h2><span id='topic+confounder'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-confounder"><code>GET /confounder</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>confounder(
  exposure_trait = NULL,
  outcome_trait = NULL,
  type = c("confounder", "intermediate", "reverse_intermediate", "collider"),
  pval_threshold = 1e-05,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="confounder_+3A_exposure_trait">exposure_trait</code></td>
<td>
<p>A trait name, e.g. &quot;Body mass index&quot;,
leaving <code>exposure_trait</code> as <code>NULL</code> will return MR information
related to a specific <code>outcome</code>.
<strong>NOTE</strong>: <code>exposure_trait</code> and <code>outcome_trait</code> cannot be both <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="confounder_+3A_outcome_trait">outcome_trait</code></td>
<td>
<p>A trait name, e.g. &quot;Coronary heart disease&quot;,
leaving <code>outcome_trait</code> as <code>NULL</code> will return MR information
related to a specific <code>exposure_trait</code>.
<strong>NOTE</strong>: <code>exposure_trait</code> and <code>outcome_trait</code> cannot be both <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="confounder_+3A_type">type</code></td>
<td>
<p>One in <code style="white-space: pre;">&#8288;["confounder", "intermediate", "reverse_intermediate", "collider"]&#8288;</code>
Refer to <a href="https://epigraphdb.org/confounder"> the confounder view in web application </a>
for details</p>
</td></tr>
<tr><td><code id="confounder_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="confounder_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /confounder</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
confounder(exposure_trait = "Body mass index", outcome_trait = "Coronary heart disease")

## End(Not run)
</code></pre>

<hr>
<h2 id='cypher'>Send a query in Cypher to EpiGraphDB</h2><span id='topic+cypher'></span>

<h3>Description</h3>

<p>NOTE: this function is intended for advanced uses.
Regular users are recommended to use standard query functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cypher(query, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cypher_+3A_query">query</code></td>
<td>
<p>A Cypher query.</p>
</td></tr>
<tr><td><code id="cypher_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cypher("MATCH (n:Gwas) RETURN n LIMIT 2")

## End(Not run)
</code></pre>

<hr>
<h2 id='drugs_risk_factors'>Drugs for risk factors</h2><span id='topic+drugs_risk_factors'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-drugsrisk-factors"><code>GET /drugs/risk-factors</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drugs_risk_factors(trait, pval_threshold = 1e-08, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drugs_risk_factors_+3A_trait">trait</code></td>
<td>
<p>A trait name</p>
</td></tr>
<tr><td><code id="drugs_risk_factors_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="drugs_risk_factors_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /drugs/risk-factors</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
drugs_risk_factors(trait = "Body mass index")

## End(Not run)
</code></pre>

<hr>
<h2 id='flatten_response'>Flatten the &quot;results&quot; field from an API response to a tibble df</h2><span id='topic+flatten_response'></span>

<h3>Description</h3>

<p>The general tibble flattener for EpiGraphDB endpoints
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flatten_response(response, field = "results")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flatten_response_+3A_response">response</code></td>
<td>
<p>An httr response</p>
</td></tr>
<tr><td><code id="flatten_response_+3A_field">field</code></td>
<td>
<p>Default to the &quot;results&quot; field</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble df
</p>

<hr>
<h2 id='genetic_cor'>Genetic correlations between traits</h2><span id='topic+genetic_cor'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-genetic-cor"><code>GET /genetic-cor</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetic_cor(trait, cor_coef_threshold = 0.8, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetic_cor_+3A_trait">trait</code></td>
<td>
<p>name of the trait, e.g. &quot;body mass index&quot;</p>
</td></tr>
<tr><td><code id="genetic_cor_+3A_cor_coef_threshold">cor_coef_threshold</code></td>
<td>
<p>correlation coefficient threshold</p>
</td></tr>
<tr><td><code id="genetic_cor_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /genetic_cor</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
genetic_cor(trait = "Body mass index") %&gt;%
  dplyr::glimpse()

## End(Not run)

# Use a different threshold
## Not run: 
genetic_cor(trait = "Body mass index", cor_coef_threshold = 0.4) %&gt;%
  dplyr::glimpse()

## End(Not run)
</code></pre>

<hr>
<h2 id='http_condition'>Modified httr::http_condition</h2><span id='topic+http_condition'></span>

<h3>Description</h3>

<p>Modified httr::http_condition
</p>


<h3>Usage</h3>

<pre><code class='language-R'>http_condition(response, context)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="http_condition_+3A_response">response</code></td>
<td>
<p>An httr response</p>
</td></tr>
<tr><td><code id="http_condition_+3A_context">context</code></td>
<td>
<p>A list on the url and params for the request</p>
</td></tr>
</table>

<hr>
<h2 id='literature_gwas'>Literature evidence regarding a GWAS trait</h2><span id='topic+literature_gwas'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-literaturegwas"><code>GET /literature/gwas</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>literature_gwas(trait, semmed_predicate = NULL, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="literature_gwas_+3A_trait">trait</code></td>
<td>
<p>A trait name</p>
</td></tr>
<tr><td><code id="literature_gwas_+3A_semmed_predicate">semmed_predicate</code></td>
<td>
<p>Either NULL which returns entries from
all predicates, or a SemMed predicate e.g. &quot;DIAGNOSES&quot; or &quot;ASSOCIATED_WITH&quot;</p>
</td></tr>
<tr><td><code id="literature_gwas_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /literature/gwas</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
literature_gwas(trait = "Body mass index")

## End(Not run)
</code></pre>

<hr>
<h2 id='mappings_gene_to_protein'>Return protein uniprot_id from associated genes</h2><span id='topic+mappings_gene_to_protein'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#post-mappingsgene-to-protein"><code>POST /mappings/gene-to-protein</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mappings_gene_to_protein(
  gene_name_list = NULL,
  gene_id_list = NULL,
  by_gene_id = FALSE,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mappings_gene_to_protein_+3A_gene_name_list">gene_name_list</code></td>
<td>
<p>List of HGNC symbols of the genes (default)</p>
</td></tr>
<tr><td><code id="mappings_gene_to_protein_+3A_gene_id_list">gene_id_list</code></td>
<td>
<p>List of Ensembl gene IDs (when <code>by_gene_id == TRUE</code>)</p>
</td></tr>
<tr><td><code id="mappings_gene_to_protein_+3A_by_gene_id">by_gene_id</code></td>
<td>
<p>Search for gene ids (Ensembl gene IDs) instead of gene names (HGNC symbols)</p>
</td></tr>
<tr><td><code id="mappings_gene_to_protein_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>POST /mappings/gene-to-protein</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># By HGNC symbols
## Not run: 
mappings_gene_to_protein(gene_name_list = c("GCH1", "MYOF"))

## End(Not run)

# By Enselbl Ids
## Not run: 
mappings_gene_to_protein(gene_id_list = c("ENSG00000162594", "ENSG00000113302"), by_gene_id = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='meta_nodes_list'>List meta nodes (e.g. Gwas, Gene, etc.)</h2><span id='topic+meta_nodes_list'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodeslist"><code>GET /meta/nodes/list</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_nodes_list(mode = c("raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta_nodes_list_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /meta/nodes/list</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
meta_nodes_list()

## End(Not run)
</code></pre>

<hr>
<h2 id='meta_nodes_list_node'>List nodes under a meta node</h2><span id='topic+meta_nodes_list_node'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodesmeta_nodelist"><code>GET /meta/nodes/{meta_node}/list</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_nodes_list_node(
  meta_node,
  full_data = TRUE,
  limit = 10,
  offset = 0,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta_nodes_list_node_+3A_meta_node">meta_node</code></td>
<td>
<p>Name of a meta node (e.g. Gwas). Use <code>meta_nodes_list</code> to get the full list of meta nodes.</p>
</td></tr>
<tr><td><code id="meta_nodes_list_node_+3A_full_data">full_data</code></td>
<td>
<p>When False, only return the id and name fields (their specific names differ in specific nodes) for a node.
This is useful if you want your queries to return results faster with smaller amount of data requested.</p>
</td></tr>
<tr><td><code id="meta_nodes_list_node_+3A_limit">limit</code></td>
<td>
<p>Max number of items to retrieve.</p>
</td></tr>
<tr><td><code id="meta_nodes_list_node_+3A_offset">offset</code></td>
<td>
<p>Number of items to skip. Use <code>limit</code> and <code>offset</code> in combination to do pagination.</p>
</td></tr>
<tr><td><code id="meta_nodes_list_node_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /meta/nodes/{meta_node}/list</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List the first 5 Gwas nodes, with only id and name fields
## Not run: 
meta_nodes_list_node(meta_node = "Gwas", full_data = FALSE, limit = 5)

## End(Not run)

# List the 6th - 10th Disease nodes, with full properties
## Not run: 
meta_nodes_list_node(meta_node = "Disease", full_data = TRUE, limit = 5, offset = 0)

## End(Not run)
</code></pre>

<hr>
<h2 id='meta_nodes_search_node'>Search a node by its id field, or its name field</h2><span id='topic+meta_nodes_search_node'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodesmeta_nodesearch"><code>GET /meta/nodes/{meta_node}/search</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_nodes_search_node(
  meta_node,
  id = NULL,
  name = NULL,
  limit = 10,
  full_data = TRUE,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta_nodes_search_node_+3A_meta_node">meta_node</code></td>
<td>
<p>Name of a meta node (e.g. Gwas). Use <code>meta_nodes_list</code> to get the full list of meta nodes.</p>
</td></tr>
<tr><td><code id="meta_nodes_search_node_+3A_id">id</code></td>
<td>
<p>The id field of a node (e.g. &quot;ieu-a-2&quot; for a Gwas).
Use EpiGraphDB web UI to get a sense of what those ids are for entities.</p>
</td></tr>
<tr><td><code id="meta_nodes_search_node_+3A_name">name</code></td>
<td>
<p>The name field of a node (e.g. &quot;body mass index&quot; for a Gwas).
Use EpiGraphDB web UI to get a sense of what those names are for entities.</p>
</td></tr>
<tr><td><code id="meta_nodes_search_node_+3A_limit">limit</code></td>
<td>
<p>Max number of items to retrieve.</p>
</td></tr>
<tr><td><code id="meta_nodes_search_node_+3A_full_data">full_data</code></td>
<td>
<p>When False, only return the id and name fields (their specific names differ in specific nodes) for a node.
This is useful if you want your queries to return results faster with smaller amount of data requested.</p>
</td></tr>
<tr><td><code id="meta_nodes_search_node_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /meta/nodes/{meta_node}/search</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Search Gwas nodes
## Not run: 
meta_nodes_search_node(meta_node = "Gwas", name = "body mass index")

## End(Not run)
</code></pre>

<hr>
<h2 id='meta_rels_list'>List meta rels (e.g. MR, etc.)</h2><span id='topic+meta_rels_list'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metarelslist"><code>GET /meta/rels/list</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_rels_list(mode = c("raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta_rels_list_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /meta/rels/list</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
meta_rels_list()

## End(Not run)
</code></pre>

<hr>
<h2 id='meta_rels_list_rel'>List relationships under a meta relationship</h2><span id='topic+meta_rels_list_rel'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metarelsmeta_rellist"><code>GET /meta/rels/{meta_rel}/list</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_rels_list_rel(meta_rel, limit = 10, offset = 0, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta_rels_list_rel_+3A_meta_rel">meta_rel</code></td>
<td>
<p>Name of a meta relationship (e.g. MR). Use <code>meta_rels_list</code> to get the full list of meta relationships.</p>
</td></tr>
<tr><td><code id="meta_rels_list_rel_+3A_limit">limit</code></td>
<td>
<p>Max number of items to retrieve.</p>
</td></tr>
<tr><td><code id="meta_rels_list_rel_+3A_offset">offset</code></td>
<td>
<p>Number of items to skip. Use <code>limit</code> and <code>offset</code> in combination to do pagination.</p>
</td></tr>
<tr><td><code id="meta_rels_list_rel_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /meta/rels/{meta_rel}/list</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List the first 5 MR relationships
## Not run: 
meta_rels_list_rel(meta_rel = "MR_EVE_MR", limit = 5)

## End(Not run)
</code></pre>

<hr>
<h2 id='mr'>Return information related to Mendelian Randomisation</h2><span id='topic+mr'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-mr"><code>GET /mr</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr(
  exposure_trait = NULL,
  outcome_trait = NULL,
  pval_threshold = 1e-05,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_+3A_exposure_trait">exposure_trait</code></td>
<td>
<p>A trait name, e.g. &quot;Body mass index&quot;,
leaving <code>exposure_trait</code> as <code>NULL</code> will return MR information
related to a specific <code>outcome</code>.
<strong>NOTE</strong>: <code>exposure_trait</code> and <code>outcome_trait</code> cannot be both <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="mr_+3A_outcome_trait">outcome_trait</code></td>
<td>
<p>A trait name, e.g. &quot;Coronary heart disease&quot;,
leaving <code>outcome_trait</code> as <code>NULL</code> will return MR information
related to a specific <code>exposure_trait</code>.
<strong>NOTE</strong>: <code>exposure_trait</code> and <code>outcome_trait</code> cannot be both <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="mr_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="mr_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /mr</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Returns a data frame
## Not run: 
mr(exposure_trait = "Body mass index", outcome_trait = "Coronary heart disease")

## End(Not run)

# Returns raw response
## Not run: 
mr(
  exposure_trait = "Body mass index", outcome_trait = "Coronary heart disease",
  mode = "raw"
) %&gt;% str()

## End(Not run)

# Use a different threshold
## Not run: 
mr(exposure_trait = "Body mass index", pval_threshold = 1e-8)

## End(Not run)
</code></pre>

<hr>
<h2 id='obs_cor'>Observational correlations between traits</h2><span id='topic+obs_cor'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-obs-cor"><code>GET /obs-cor</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obs_cor(trait, cor_coef_threshold = 0.8, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="obs_cor_+3A_trait">trait</code></td>
<td>
<p>name of the trait, e.g. &quot;body mass index&quot;</p>
</td></tr>
<tr><td><code id="obs_cor_+3A_cor_coef_threshold">cor_coef_threshold</code></td>
<td>
<p>correlation coefficient threshold</p>
</td></tr>
<tr><td><code id="obs_cor_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /obs-cor</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
obs_cor(trait = "Body mass index (BMI)") %&gt;%
  dplyr::glimpse()

## End(Not run)

# Use a different threshold
## Not run: 
obs_cor(trait = "Body mass index (BMI)", cor_coef_threshold = 0.8) %&gt;%
  dplyr::glimpse()

## End(Not run)
</code></pre>

<hr>
<h2 id='ontology_gwas_efo'>Ontology association between EFO term and Gwas</h2><span id='topic+ontology_gwas_efo'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-ontologygwas-efo"><code>GET /ontology/gwas-efo</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontology_gwas_efo(
  trait = NULL,
  efo_term = NULL,
  fuzzy = TRUE,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ontology_gwas_efo_+3A_trait">trait</code></td>
<td>
<p>trait name, e.g. &quot;body mass&quot;</p>
</td></tr>
<tr><td><code id="ontology_gwas_efo_+3A_efo_term">efo_term</code></td>
<td>
<p>EFO term, e.g. &quot;systolic blood pressure&quot;</p>
</td></tr>
<tr><td><code id="ontology_gwas_efo_+3A_fuzzy">fuzzy</code></td>
<td>
<p>whether query with exact matching (FALSE) or fuzzy matching (default, TRUE)</p>
</td></tr>
<tr><td><code id="ontology_gwas_efo_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /ontology/gwas-efo</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ontology_gwas_efo(trait = "blood", fuzzy = FALSE)

## End(Not run)

## Not run: 
ontology_gwas_efo(efo_term = "blood pressure", fuzzy = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='pathway'>Pathway evidence</h2><span id='topic+pathway'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pathway"><code>GET /pathway</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway(trait, pval_threshold = 1e-05, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_+3A_trait">trait</code></td>
<td>
<p>A trait name</p>
</td></tr>
<tr><td><code id="pathway_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="pathway_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /pathway</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pathway(trait = "Body mass index")

## End(Not run)
</code></pre>

<hr>
<h2 id='pqtl'>Return information related to the pQTL analysis</h2><span id='topic+pqtl'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtl"><code style="white-space: pre;">&#8288;GET /pqtl/&#8288;</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqtl(
  query,
  rtype = c("mrres", "simple", "sglmr", "inst", "sense"),
  pvalue = 0.05,
  searchflag = c("traits", "proteins"),
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pqtl_+3A_query">query</code></td>
<td>
<p>(Required) A protein coding gene name or a trait name,
e.g. &quot;ADAM19&quot; or &quot;Inflammatory bowel disease&quot;
which cannot be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="pqtl_+3A_rtype">rtype</code></td>
<td>
<p>(Optional) A type of data to be extracted, which can be one of these options:
</p>

<ol>
<li> <p><code>simple</code>: Basic summary
</p>
</li>
<li> <p><code>mrres</code>: MR results (DEFAULT)
</p>
</li>
<li> <p><code>sglmr</code>: Single SNP MR results
</p>
</li>
<li> <p><code>inst</code>: SNP information
</p>
</li>
<li> <p><code>sense</code>: Sensitivity analysis
<strong>NOTE</strong>: <code>mrres</code> is a DEFAULT option.
</p>
</li></ol>
</td></tr>
<tr><td><code id="pqtl_+3A_pvalue">pvalue</code></td>
<td>
<p>(Optional) A pvalue threshold for MR results with the DEFAULT set to 0.05.
<strong>NOTE</strong>: this threshold applies to any <code>rtype</code> chosen.</p>
</td></tr>
<tr><td><code id="pqtl_+3A_searchflag">searchflag</code></td>
<td>
<p>(Required) A flag to indicate whether you are searching for proteins or
traits which cannot be <code>NULL</code>.
If <code>query</code> is a protein name, then this flag should be &quot;proteins&quot;;
if <code>query</code> is a trait, this flag should be &quot;traits&quot;.
<strong>NOTE</strong>: if the wrong flag is chosen for <code>query</code>, there will be no result
returned.</p>
</td></tr>
<tr><td><code id="pqtl_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code style="white-space: pre;">&#8288;GET /pqtl/&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Returns a data frame of MR results, while searching for proteins
## Not run: 
pqtl(query = "ADAM19", searchflag = "proteins")

## End(Not run)

# Returns a data frame with SNP information, while searching for traits
## Not run: 
pqtl(
  query = "Inflammatory bowel disease",
  rtype = "inst",
  searchflag = "traits"
)

## End(Not run)

# Change a pvalue threshold (the default is 0.05)
## Not run: 
pqtl(
  query = "Inflammatory bowel disease",
  rtype = "inst",
  pvalue = 1.0,
  searchflag = "traits"
)

## End(Not run)

# Returns raw response if mode="raw"
## Not run: 
pqtl(
  query = "ADAM19", searchflag = "proteins",
  mode = "raw"
) %&gt;% str()

## End(Not run)
</code></pre>

<hr>
<h2 id='pqtl_list'>Return a list of all proteins/exposures or traits/outcomes
available in the database</h2><span id='topic+pqtl_list'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtllist"><code style="white-space: pre;">&#8288;GET /pqtl/list/&#8288;</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqtl_list(flag = c("exposures", "outcomes"), mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pqtl_list_+3A_flag">flag</code></td>
<td>
<p>(Optional) A flag which indicates whether the list of
exposures (if &quot;exposures&quot;) or outcomes (if &quot;outcomes&quot;)
should be returned. The DEFAULT is &quot;exposures&quot;.</p>
</td></tr>
<tr><td><code id="pqtl_list_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code style="white-space: pre;">&#8288;GET /pqtl/list/&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Returns a list of available proteins (exposures)
## Not run: 
pqtl_list()

## End(Not run)

# Returns a list of available traits (outcomes)
## Not run: 
pqtl_list(flag = "outcomes")

## End(Not run)
</code></pre>

<hr>
<h2 id='pqtl_pleio'>Return information related to the pleiotropy of SNPs</h2><span id='topic+pqtl_pleio'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtlpleio"><code style="white-space: pre;">&#8288;GET /pqtl/pleio/&#8288;</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pqtl_pleio(
  rsid = NULL,
  prflag = c("proteins", "count"),
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pqtl_pleio_+3A_rsid">rsid</code></td>
<td>
<p>(Required) A SNP identified by rsID which cannot be <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="pqtl_pleio_+3A_prflag">prflag</code></td>
<td>
<p>(Optional) A flag which determines whether the number (if &quot;count&quot;)
or names (if &quot;proteins&quot;) of the associated proteins should be returned.
The DEFAULT value is &quot;proteins&quot;.</p>
</td></tr>
<tr><td><code id="pqtl_pleio_+3A_mode">mode</code></td>
<td>
<p>(Optional) If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code style="white-space: pre;">&#8288;GET /pqtl/pleio/&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Returns a data frame of associated proteins
## Not run: 
pqtl_pleio(rsid = "rs1260326")

## End(Not run)

# Returns a number of associated proteins
## Not run: 
pqtl_pleio(rsid = "rs1260326", prflag = "count")

## End(Not run)
</code></pre>

<hr>
<h2 id='protein_in_pathway'>For the list of proteins, returns their associated pathway data</h2><span id='topic+protein_in_pathway'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#post-proteinin-pathway"><code style="white-space: pre;">&#8288;POST /protein/in-pathway&#8288;</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protein_in_pathway(uniprot_id_list, mode = c("table", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protein_in_pathway_+3A_uniprot_id_list">uniprot_id_list</code></td>
<td>
<p>A list of protein UniProt IDs</p>
</td></tr>
<tr><td><code id="protein_in_pathway_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code style="white-space: pre;">&#8288;POST /protein/in-pathway&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
protein_in_pathway(uniprot_id_list = c("014933", "060674", "P32455"))

## End(Not run)
</code></pre>

<hr>
<h2 id='query_epigraphdb'>Send data request to an EpiGraphDB API endpoint</h2><span id='topic+query_epigraphdb'></span>

<h3>Description</h3>

<p>This is a general purpose function to send data request
which can be used when there has not been an R equivalent package function
to an API endpoint.
Underneath this is a wrapper around <code>httr</code> functions with better handling of
returned status.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_epigraphdb(
  route,
  params = NULL,
  mode = c("raw", "table"),
  method = c("GET", "POST"),
  retry_times = 3,
  retry_pause_min = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_epigraphdb_+3A_route">route</code></td>
<td>
<p>An EpiGraphDB API endpoint route, e.g. <code>"/mr"</code> or <code>"/confounder"</code>.
Consult the <a href="https://api.epigraphdb.org">EpiGraphDB API documentation</a>.</p>
</td></tr>
<tr><td><code id="query_epigraphdb_+3A_params">params</code></td>
<td>
<p>A list of parameters associated with the query endpoint.</p>
</td></tr>
<tr><td><code id="query_epigraphdb_+3A_mode">mode</code></td>
<td>
<p><code>c("raw", "table")</code>, if <code>"table"</code> then the query handler will try
to convert the returned data to a tibble dataframe.
NOTE: The default mode is &quot;raw&quot; which will NOT convert the returned response to
a dataframe.
This is different to functions that query topic endpoints which
default to return a dataframe.
Explicitly specify <code>mode = "table"</code> when needed.</p>
</td></tr>
<tr><td><code id="query_epigraphdb_+3A_method">method</code></td>
<td>
<p>Type of HTTP (GET, POST, PUT, etc.) method.
</p>
<p>NOTE: When sending a POST request where a specific parameter is specified as a list on the API,
and if the equivalent in R is a vector of length 1, you should wrap this parameter
in <code>I()</code>, e.g. I(c(&quot;APOE&quot;)) to avoid auto unboxing.
For details, please refer to <a href="https://httr.r-lib.org/reference/POST.html"><code>httr::POST</code></a></p>
</td></tr>
<tr><td><code id="query_epigraphdb_+3A_retry_times">retry_times</code></td>
<td>
<p>Number of times the function will retry the request to the API.</p>
</td></tr>
<tr><td><code id="query_epigraphdb_+3A_retry_pause_min">retry_pause_min</code></td>
<td>
<p>Minimum number of seconds to wait for the next retry.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from an EpiGraphDB API endpoint.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># GET /mr
# equivalent to `mr(exposure_trait = "Body mass index", outcome_trait = "Coronary heart disease")`
## Not run: 
query_epigraphdb(
  route = "/mr",
  params = list(
    exposure_trait = "Body mass index",
    outcome_trait = "Coronary heart disease"
  ),
  mode = "table"
)

## End(Not run)

# GET /meta/nodes/Gwas/list
## Not run: 
query_epigraphdb(
  route = "/meta/nodes/Gwas/list",
  params = list(
    limit = 5,
    offset = 0
  )
) %&gt;% str(1)

## End(Not run)

# POST /protein/ppi
## Not run: 
query_epigraphdb(
  route = "/protein/ppi",
  params = list(
    uniprot_id_list = c("P30793", "Q9NZM1", "O95236")
  ),
  method = "POST"
)

## End(Not run)

# error handling
## Not run: 
tryCatch(
  query_epigraphdb(
    route = "/mr",
    params = list(
      exposure_trait = NULL,
      outcome_trait = NULL
    ),
    retry_times = 0
  ),
  error = function(e) {
    message(e)
  }
)

## End(Not run)
</code></pre>

<hr>
<h2 id='stop_for_status'>Catch error for an HTTP response</h2><span id='topic+stop_for_status'></span>

<h3>Description</h3>

<p>Modifies from httr::stop_for_status
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stop_for_status(response, context)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stop_for_status_+3A_response">response</code></td>
<td>
<p>An httr response</p>
</td></tr>
<tr><td><code id="stop_for_status_+3A_context">context</code></td>
<td>
<p>A list on the url and params for the request</p>
</td></tr>
</table>

<hr>
<h2 id='xqtl_multi_snp_mr'>Multi SNP QTL MR evidence</h2><span id='topic+xqtl_multi_snp_mr'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlmulti-snp-mr"><code>GET /xqtl/multi-snp-mr</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xqtl_multi_snp_mr(
  exposure_gene = NULL,
  outcome_trait = NULL,
  mr_method = c("IVW", "Egger"),
  qtl_type = c("eQTL", "pQTL"),
  pval_threshold = 1e-05,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xqtl_multi_snp_mr_+3A_exposure_gene">exposure_gene</code></td>
<td>
<p>Name of the exposure gene</p>
</td></tr>
<tr><td><code id="xqtl_multi_snp_mr_+3A_outcome_trait">outcome_trait</code></td>
<td>
<p>Name of the outcome trait</p>
</td></tr>
<tr><td><code id="xqtl_multi_snp_mr_+3A_mr_method">mr_method</code></td>
<td>
<p>&quot;IVW&quot; or &quot;Egger&quot;</p>
</td></tr>
<tr><td><code id="xqtl_multi_snp_mr_+3A_qtl_type">qtl_type</code></td>
<td>
<p>&quot;eQTL&quot; or &quot;pQTL&quot;</p>
</td></tr>
<tr><td><code id="xqtl_multi_snp_mr_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="xqtl_multi_snp_mr_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GET /xqtl/multi-snp-mr</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xqtl_multi_snp_mr(outcome_trait = "Coronary heart disease")

## End(Not run)
</code></pre>

<hr>
<h2 id='xqtl_single_snp_mr'>Single SNP QTL MR evidence</h2><span id='topic+xqtl_single_snp_mr'></span>

<h3>Description</h3>

<p><a href="https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlsingle-snp-mr"><code>GET /xqtl/single-snp-mr</code></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xqtl_single_snp_mr(
  exposure_gene = NULL,
  outcome_trait = NULL,
  snp = NULL,
  qtl_type = c("eQTL", "pQTL"),
  pval_threshold = 1e-05,
  mode = c("table", "raw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xqtl_single_snp_mr_+3A_exposure_gene">exposure_gene</code></td>
<td>
<p>Name of the exposure gene</p>
</td></tr>
<tr><td><code id="xqtl_single_snp_mr_+3A_outcome_trait">outcome_trait</code></td>
<td>
<p>Name of the outcome trait</p>
</td></tr>
<tr><td><code id="xqtl_single_snp_mr_+3A_snp">snp</code></td>
<td>
<p>SNP rsid</p>
</td></tr>
<tr><td><code id="xqtl_single_snp_mr_+3A_qtl_type">qtl_type</code></td>
<td>
<p>&quot;eQTL&quot; or &quot;pQTL&quot;</p>
</td></tr>
<tr><td><code id="xqtl_single_snp_mr_+3A_pval_threshold">pval_threshold</code></td>
<td>
<p>P-value threshold</p>
</td></tr>
<tr><td><code id="xqtl_single_snp_mr_+3A_mode">mode</code></td>
<td>
<p>If <code>mode = "table"</code>, returns a data frame
(a <a href="https://tibble.tidyverse.org/"><code>tibble</code></a> as per
<a href="https://style.tidyverse.org/"><code>tidyverse</code></a> convention).
If <code>mode = "raw"</code>, returns a raw response from EpiGraphDB API
with minimal parsing done by <a href="https://httr.r-lib.org/"><code>httr</code></a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data from <code>GEET /xqtl/single-snp-mr</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xqtl_single_snp_mr(outcome_trait = "Coronary heart disease")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
