<!DOCTYPE html><html lang="en"><head><title>Help for package epilogi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {epilogi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#epilogi-package'>
<p>The 'epilogi' Variable Selection Algorithm for Continuous Data.</p></a></li>
<li><a href='#epilogi'>
<p>The epilogi Variable Selection Algorithm for Continuous Data.</p></a></li>
<li><a href='#pcor.equiv'>
<p>Equivalence test using partial correlation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The 'epilogi' Variable Selection Algorithm for Continuous Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-12-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Michail Tsagris [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michail Tsagris &lt;mtsagris@uoc.gr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rfast, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Rfast2</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'epilogi' variable selection algorithm is implemented for the case of continuous response and predictor variables. The relevant paper is: Lakiotaki K., Papadovasilakis Z., Lagani V., Fafalios S., Charonyktakis P., Tsagris M. and Tsamardinos I. (2023). "Automated machine learning for Genome Wide Association Studies". Bioinformatics, 39(9): btad545. &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtad545">doi:10.1093/bioinformatics/btad545</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-20 15:18:24 UTC; mtsag</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-23 14:10:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='epilogi-package'>
The 'epilogi' Variable Selection Algorithm for Continuous Data.
</h2><span id='topic+epilogi-package'></span>

<h3>Description</h3>

<p>The '<code class="reqn">\epsilon</code>pilogi' Variable Selection Algorithm for Continuous Data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> epilogi</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-12-20</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Maintainers</h3>

<p>Michail Tsagris <a href="mailto:mtsagris@uoc.gr">mtsagris@uoc.gr</a>.
</p>


<h3>Author(s)</h3>

<p>Michail Tsagris <a href="mailto:mtsagris@uoc.gr">mtsagris@uoc.gr</a>.
</p>


<h3>References</h3>

<p>Lakiotaki K., Papadovasilakis Z., Lagani V., Fafalios S., Charonyktakis P., Tsagris M. and Tsamardinos I. (2023). Automated machine learning for Genome Wide Association Studies. Bioinformatics, 39(9): btad545.
</p>
<p>Tsagris M., Papadovasilakis Z., Lakiotaki K. and Tsamardinos I. (2022). The <code class="reqn">\gamma</code>-OMP algorithm for feature selection with application to gene expression data. IEEE/ACM Transactions on Computational Biology and Bioinformatics, 19(2): 1214&ndash;1224.
</p>

<hr>
<h2 id='epilogi'>
The epilogi Variable Selection Algorithm for Continuous Data.
</h2><span id='topic+epilogi'></span>

<h3>Description</h3>

<p>The <code class="reqn">\epsilon</code>pilogi Variable Selection Algorithm for Continuous Data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epilogi(y, x, tol = 0.01, alpha = 0.05, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epilogi_+3A_y">y</code></td>
<td>

<p>A vector with the continuous response variable.
</p>
</td></tr>
<tr><td><code id="epilogi_+3A_x">x</code></td>
<td>

<p>A matrix with the continuous predictor variables.
</p>
</td></tr>
<tr><td><code id="epilogi_+3A_tol">tol</code></td>
<td>

<p>The tolerance value for the algortihm to terminate. This takes values greater than 0 and it refers to the change between two successive <code class="reqn">R^2</code>-adjusted values.
</p>
</td></tr>
<tr><td><code id="epilogi_+3A_alpha">alpha</code></td>
<td>

<p>The significance level to deem a predictor variable is statistically equivalent to a selected variable.
</p>
</td></tr>
<tr><td><code id="epilogi_+3A_parallel">parallel</code></td>
<td>

<p>If set to TRUE, some of the computations take place in parallel (in C++).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code class="reqn">\epsilon</code>pilogi variable selection algorithm (Lakiotaki et al., 2023) is a generalisation of the <code class="reqn">\gamma</code>-OMP algorithm (Tsagris et al. 2022). It applies the aforementioned algorithm with the addition that it returns possible statistically equivalent predictor(s) for each selected predictor. Once a variable is selected the algorithm searches for possible equivalent predictors using the partial correlation between the residuals.
</p>
<p>The heuristic method to consider two predictors R and C informationally equivalent given the current selected predictor S is determined as follows: first, the residuals r of the model using S are computed. Then, if the following two conditions hold R and C are considered equivalent: Ind(R; r | C) and Ind(r ; C | R), where Ind(R; r | C) denotes the conditional independence of R with r given C. When linearity is assumed, the test can be implemented by testing for significance the corresponding partial correlation. The tests Ind return a p-value and independence is accepted when it is larger than a threshold (significance value, argument alpha). Intuitively, R and C are heuristically considered equivalent, if C is known, then R provides no additional information for the residuals r, and if R is known, then C provides no additional information for r.
</p>


<h3>Value</h3>

<p>A list including:
</p>
<table role = "presentation">
<tr><td><code>runtime</code></td>
<td>
<p>The runtime of the algorithm.
</p>
</td></tr>
<tr><td><code>result</code></td>
<td>

<p>A matrix with two columns. The selected predictor(s) and the adjusted <code class="reqn">R^2</code>-values.
</p>
</td></tr>
<tr><td><code>equiv</code></td>
<td>

<p>A list with the equivalent predictors (if any) corresponding to each selected predictor.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michail Tsagris.
</p>
<p>R implementation and documentation: Michail Tsagris <a href="mailto:mtsagris@uoc.gr">mtsagris@uoc.gr</a>.
</p>


<h3>References</h3>

<p>Lakiotaki K., Papadovasilakis Z., Lagani V., Fafalios S., Charonyktakis P., Tsagris M. and Tsamardinos I. (2023). Automated machine learning for Genome Wide Association Studies. Bioinformatics, 39(9): btad545
</p>
<p>Tsagris M., Papadovasilakis Z., Lakiotaki K. and Tsamardinos I. (2022). The <code class="reqn">\gamma</code>-OMP algorithm for feature selection with application to gene expression data. IEEE/ACM Transactions on Computational Biology and Bioinformatics, 19(2): 1214&ndash;1224.
</p>


<h3>See Also</h3>

<p><code> <a href="#topic+pcor.equiv">pcor.equiv</a>
</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#simulate a dataset with continuous data
set.seed(1234)
n &lt;- 500
x &lt;- matrix( rnorm(n * 50, 0, 30), ncol = 50 )

#define a simulated class variable
y &lt;- 2 * x[, 1] - 1.5 * x[, 2] + x[, 3] + rnorm(n, 0, 15)

# define some simulated equivalences
x[, 4] &lt;- x[, 1] + rnorm(n, 0, 1)
x[, 5] &lt;- x[, 2] + rnorm(n, 0, 1)

epilogi(y, x, tol = 0.05)
</code></pre>

<hr>
<h2 id='pcor.equiv'>
Equivalence test using partial correlation
</h2><span id='topic+pcor.equiv'></span>

<h3>Description</h3>

<p>Equivalence test using partial correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcor.equiv(res, y, x, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcor.equiv_+3A_res">res</code></td>
<td>

<p>A vector with the residuals of the linear model.
</p>
</td></tr>
<tr><td><code id="pcor.equiv_+3A_y">y</code></td>
<td>

<p>A vector with a selected predictor.
</p>
</td></tr>
<tr><td><code id="pcor.equiv_+3A_x">x</code></td>
<td>

<p>A matrix with other predictors.
</p>
</td></tr>
<tr><td><code id="pcor.equiv_+3A_alpha">alpha</code></td>
<td>

<p>The significance level to check for predictors from x that are equivalent to y.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with 0s and 1s. 0s indicate that the predictors are not equivalent, while 1s indicate the equivalent predictors.
</p>


<h3>Author(s)</h3>

<p>Michail Tsagris.
</p>
<p>R implementation and documentation: Michail Tsagris <a href="mailto:mtsagris@uoc.gr">mtsagris@uoc.gr</a>.
</p>


<h3>See Also</h3>

<p><code> <a href="#topic+epilogi">epilogi</a>
</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#simulate a dataset with continuous data
set.seed(1234)
n &lt;- 500
x &lt;- matrix( rnorm(n * 50, 0, 30), ncol = 50 )

#define a simulated class variable
y &lt;- 2 * x[, 1] - 1.5 * x[, 2] + x[, 3] + rnorm(n, 0, 15)

# define some simulated equivalences
x[, 4] &lt;- x[, 1] + rnorm(n, 0, 1)
x[, 5] &lt;- x[, 2] + rnorm(n, 0, 1)


b &lt;- epilogi(y, x, tol = 0.05)
sel &lt;- b$result[2, 1]
## standardise the y and x first
y &lt;- (y - mean(y)) / Rfast::Var(y, std = TRUE)
x &lt;- Rfast::standardise(x)

res &lt;- resid( lm(y ~ x[, sel] ) )
sela &lt;- b$result[2:3, 1]
pcor.equiv(res, x[, sela[2]], x[, -sela] )
## bear in mind that this gives the third variable after removing the first two,
## so this is essentially the 5th variable in the "x" matrix.
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
