<!DOCTYPE html><html><head><title>Help for package episcan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {episcan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkchunksize'><p>Check chunk size</p></a></li>
<li><a href='#epiblaster1geno'><p>Parallelized calculation of the difference of correlation coefficients and compute <code class="reqn">Z</code> test with one genotype input</p></a></li>
<li><a href='#epiblaster2genos'><p>Parallelized calculattion of the difference of correlation coefficients and compute Z test with two genotype inputs</p></a></li>
<li><a href='#epiHSIC'><p>Calculate HSIC values</p></a></li>
<li><a href='#epiHSIC1geno'><p>Calculate epistasis using HSIC with one genotype input</p></a></li>
<li><a href='#epiHSIC2genos'><p>Calculate epistasis using HSIC with two genotype inputs</p></a></li>
<li><a href='#episcan'><p>Scan pairwise epistasis</p></a></li>
<li><a href='#getcor'><p>Get correlation matrix</p></a></li>
<li><a href='#ithChunk'><p>index set for idx-th chunk of size chunk for n elements</p></a></li>
<li><a href='#WriteSnpPairs'><p>Write out epistasis result (normal matrix)</p></a></li>
<li><a href='#WriteSnpPairs_sym'><p>Write out epistasis result (symmetric matrix)</p></a></li>
<li><a href='#ZtoP'><p>Convert Z-score to correponding p-value</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Scan Pairwise Epistasis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Beibei Jiang &lt;beibei_jiang@psych.mpg.de&gt; and Benno Pütz &lt;puetz@psych.mpg.de&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Beibei Jiang &lt;beibei_jiang@psych.mpg.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Searching genomic interactions with linear/logistic regression in a high-dimensional dataset is a time-consuming task. This package provides some efficient ways to scan epistasis in genome-wide interaction studies (GWIS). Both case-control status (binary outcome) and quantitative phenotype (continuous outcome) are supported (the main references: 1. Kam-Thong, T., D. Czamara, K. Tsuda, K. Borgwardt, C. M. Lewis, A. Erhardt-Lehmann, B. Hemmer, et al. (2011). &lt;<a href="https://doi.org/10.1038%2Fejhg.2010.196">doi:10.1038/ejhg.2010.196</a>&gt;. 2. Kam-Thong, T., B. Pütz, N. Karbalai, B. Müller-Myhsok, and K. Borgwardt. (2011).  &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtr218">doi:10.1093/bioinformatics/btr218</a>&gt;.)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-10 12:13:44 UTC; beibei_jiang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-09-14 23:02:19 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkchunksize'>Check chunk size</h2><span id='topic+checkchunksize'></span>

<h3>Description</h3>

<p>Check the chunk size whether it is over the given number of variables(vaiants) in genotype data. 
If yes, reset the chunk size equal to the number of variables(vaiants).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkchunksize(c, m, n = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkchunksize_+3A_c">c</code></td>
<td>
<p>an integer indicating the set chunk size.</p>
</td></tr>
<tr><td><code id="checkchunksize_+3A_m">m</code></td>
<td>
<p>an integer indicating the number of variables(vaiants) in <code>geno1</code> if there is only one genotype input.</p>
</td></tr>
<tr><td><code id="checkchunksize_+3A_n">n</code></td>
<td>
<p>an integer indicating the number of variables(vaiants) in <code>geno2</code> if there are two genotype inputs.
The default is NULL.</p>
</td></tr>
<tr><td><code id="checkchunksize_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an integer indicating the chunk size
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), 
ncol = 10)
geno2 &lt;- matrix(sample(0:2, size = 2000, replace = TRUE, prob = c(0.4, 0.3, 0.3)), 
ncol = 20)

# if chunk size is smaller, there is no problem
chunksize &lt;- 10
checkchunksize(chunksize, ncol(geno1))

# if chunk size is bigger than the number of columns in genotype input, 
# set chunk size equal to the number of columns in genotype input
chuksize &lt;- 12
checkchunksize(chunksize, ncol(geno1))

# if chunk size is bigger than the number of columns of geno1 and geno2, 
# set chunk size equal to the minima nunumber of columns of geno1 and geno2
chunksize &lt;- 50
checkchunksize(chunksize, ncol(geno1), ncol(geno2))
</code></pre>

<hr>
<h2 id='epiblaster1geno'>Parallelized calculation of the difference of correlation coefficients and compute <code class="reqn">Z</code> test with one genotype input</h2><span id='topic+epiblaster1geno'></span>

<h3>Description</h3>

<p>Calculate the difference of correlation coefficents between cases and controls,
conduct <code class="reqn">Z</code> test for the differences (values) and choose variant pairs with the significance below the given threshold for output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiblaster1geno(geno, pheno, chunk = 1000, zpthres = 1e-05,
  outfile = "NONE", suffix = ".txt", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiblaster1geno_+3A_geno">geno</code></td>
<td>
<p>is the normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object (from <span class="pkg">bigmemory</span>. 
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_pheno">pheno</code></td>
<td>
<p>a vector containing the binary phenotype information (case/control). The values are either 0 (control) or 1 (case).</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_chunk">chunk</code></td>
<td>
<p>is the number of variants in each chunk. Default: 1000.</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_zpthres">zpthres</code></td>
<td>
<p>is the significance threshold to select variant pairs for output. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_outfile">outfile</code></td>
<td>
<p>is the base of out filename. Default: 'NONE'.</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_suffix">suffix</code></td>
<td>
<p>is the suffix of out filename. Default: '.txt'.</p>
</td></tr>
<tr><td><code id="epiblaster1geno_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), ncol = 10)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), col = paste0("rs", 1:ncol(geno1)))
p1 &lt;- c(rep(0, 60), rep(1, 40))

# normalized data
geno1 &lt;- scale(geno1)

# one genotype with case-control phenotype
epiblaster1geno(geno = geno1, 
pheno = p1,
outfile = "episcan_1geno_cc", 
suffix = ".txt", 
zpthres = 0.9, 
chunk = 10)

# take a look at the result
res &lt;- read.table("episcan_1geno_cc.txt", 
header = TRUE, 
stringsAsFactors = FALSE)
head(res)
</code></pre>

<hr>
<h2 id='epiblaster2genos'>Parallelized calculattion of the difference of correlation coefficients and compute Z test with two genotype inputs</h2><span id='topic+epiblaster2genos'></span>

<h3>Description</h3>

<p>Calculate the difference of correlation coeficents between cases and controls,
conduct Z test for the differences (values) and choose variant pairs with the significance below the given threshold for output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiblaster2genos(geno1, geno2, pheno, chunk = 1000, zpthres = 1e-05,
  outfile = "NONE", suffix = ".txt", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiblaster2genos_+3A_geno1">geno1</code></td>
<td>
<p>is the first normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object from <span class="pkg">bigmemory</span>. 
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_geno2">geno2</code></td>
<td>
<p>is the second normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object from <span class="pkg">bigmemory</span>. 
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_pheno">pheno</code></td>
<td>
<p>a vector containing the binary phenotype information (case/control). The values are either 0 (control) or 1 (case).</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_chunk">chunk</code></td>
<td>
<p>is the number of variants in each chunk.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_zpthres">zpthres</code></td>
<td>
<p>is the significance threshold to select variant pairs for output. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_outfile">outfile</code></td>
<td>
<p>is the prefix of out filename.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_suffix">suffix</code></td>
<td>
<p>is the suffix of out filename.</p>
</td></tr>
<tr><td><code id="epiblaster2genos_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), ncol = 10)
geno2 &lt;- matrix(sample(0:2, size = 2000, replace = TRUE, prob = c(0.4, 0.3, 0.3)), ncol = 20)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), col = paste0("rs", 1:ncol(geno1)))
dimnames(geno2) &lt;- list(row = paste0("IND", 1:nrow(geno2)), col = paste0("exm", 1:ncol(geno2)))
p1 &lt;- c(rep(0, 60), rep(1, 40))

# normalized data
geno1 &lt;- scale(geno1)
geno2 &lt;- scale(geno2)

# two genotypes with quantitative phenotype
epiblaster2genos(geno1 = geno1, 
geno2 = geno2, 
pheno = p1, outfile = "episcan_2geno_cc", 
suffix = ".txt", 
zpthres = 0.9, 
chunk = 10)

# take a look at the result
res &lt;- read.table("episcan_2geno_cc.txt", 
header = TRUE, 
stringsAsFactors = FALSE)
head(res)
</code></pre>

<hr>
<h2 id='epiHSIC'>Calculate HSIC values</h2><span id='topic+epiHSIC'></span>

<h3>Description</h3>

<p>Calculate HSIC values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiHSIC(A = NULL, B = NULL, P = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiHSIC_+3A_a">A</code></td>
<td>
<p>is one matrix.</p>
</td></tr>
<tr><td><code id="epiHSIC_+3A_b">B</code></td>
<td>
<p>is one matrix.</p>
</td></tr>
<tr><td><code id="epiHSIC_+3A_p">P</code></td>
<td>
<p>is &quot;phenoype&quot;, a vector.</p>
</td></tr>
<tr><td><code id="epiHSIC_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
# simulate some data
set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), ncol = 10)
geno2 &lt;- matrix(sample(0:2, size = 2000, replace = TRUE, prob = c(0.4, 0.3, 0.3)), ncol = 20)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), col = paste0("rs", 1:ncol(geno1)))
dimnames(geno2) &lt;- list(row = paste0("IND", 1:nrow(geno2)), col = paste0("exm", 1:ncol(geno2)))

epiHSIC(A = scale(geno1),
B = scale(geno2),
P = rnorm(100))
</code></pre>

<hr>
<h2 id='epiHSIC1geno'>Calculate epistasis using HSIC with one genotype input</h2><span id='topic+epiHSIC1geno'></span>

<h3>Description</h3>

<p>Calculate the significance of epistasis according the definition of HSIC, conduct <code class="reqn">Z</code> test for HSIC values and 
choose variant pairs with the significance below the given threshold for output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiHSIC1geno(geno = NULL, pheno, chunk = 1000, zpthres = 1e-05,
  outfile = "NONE", suffix = ".txt", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiHSIC1geno_+3A_geno">geno</code></td>
<td>
<p>is the normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object from <span class="pkg">bigmemory</span>.
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_pheno">pheno</code></td>
<td>
<p>is a vector containing the normalized phenotype information.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_chunk">chunk</code></td>
<td>
<p>is the number of variants in each chunk.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_zpthres">zpthres</code></td>
<td>
<p>is is the significance threshold to select variant pairs for output. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_outfile">outfile</code></td>
<td>
<p>is the basename of out filename.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_suffix">suffix</code></td>
<td>
<p>is the suffix of out filename.</p>
</td></tr>
<tr><td><code id="epiHSIC1geno_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), ncol = 10)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), col = paste0("rs", 1:ncol(geno1)))
p2 &lt;- rnorm(100, mean = 5, sd = 10)

# normalized data
geno1 &lt;- scale(geno1)
p2 &lt;- as.vector(unlist(scale(p2)))

# one genotypes with quantitative phenotype
epiHSIC1geno(geno = geno1,
pheno = p2, 
outfile = "episcan_1geno_quant", 
suffix = ".txt", 
zpthres = 0.9, 
chunk = 10)

# take a look at the result
res &lt;- read.table("episcan_1geno_quant.txt", 
header = TRUE, 
stringsAsFactors = FALSE)
head(res)
</code></pre>

<hr>
<h2 id='epiHSIC2genos'>Calculate epistasis using HSIC with two genotype inputs</h2><span id='topic+epiHSIC2genos'></span>

<h3>Description</h3>

<p>Calculate the significance of epistasis according the definition of HSIC, conduct Z test for HSIC values and 
choose variant pairs with the significance below the given threshold for output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiHSIC2genos(geno1 = NULL, geno2 = NULL, pheno = NULL,
  chunk = 1000, zpthres = 1e-05, outfile = "NONE", suffix = ".txt",
  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiHSIC2genos_+3A_geno1">geno1</code></td>
<td>
<p>is the first normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object from <span class="pkg">bigmemory</span>. 
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_geno2">geno2</code></td>
<td>
<p>is the second normalized genotype data. It can be a matrix or a dataframe, or a big.matrix object from <span class="pkg">bigmemory</span>.
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_pheno">pheno</code></td>
<td>
<p>is a vector containing the normalized phenotype information.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_chunk">chunk</code></td>
<td>
<p>is the number of variants in each chunk.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_zpthres">zpthres</code></td>
<td>
<p>is the significance threshold for cut-off output of the variant pairs.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_outfile">outfile</code></td>
<td>
<p>is the basename of out filename.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_suffix">suffix</code></td>
<td>
<p>is the suffix of out filename.</p>
</td></tr>
<tr><td><code id="epiHSIC2genos_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
set.seed(123)
n1 &lt;- 10; n2 &lt;- 15; rows &lt;- 10
geno1 &lt;- matrix(sample(0:2, size = n1*rows, replace = TRUE, prob = c(0.5, 0.3, 0.2)), ncol = n1)
geno2 &lt;- matrix(sample(0:2, size = n2*rows, replace = TRUE, prob = c(0.4, 0.3, 0.3)), ncol = n2)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), col = paste0("rs", 1:ncol(geno1)))
dimnames(geno2) &lt;- list(row = paste0("IND", 1:nrow(geno2)), col = paste0("exm", 1:ncol(geno2)))
p2 &lt;- rnorm(rows, mean = 5, sd = 10)

# normalized data
geno1 &lt;- scale(geno1)
geno2 &lt;- scale(geno2)
p2 &lt;- as.vector(unlist(scale(p2)))

# two genotypes with quantitative phenotype
epiHSIC2genos(geno1 = geno1, 
geno2 = geno2, 
pheno = p2, 
outfile = "episcan_2geno_quant", 
suffix = ".txt", 
zpthres = 0.9, 
chunk = 10)

# take a look at the result
res &lt;- read.table("episcan_2geno_quant.txt", 
header = TRUE, 
stringsAsFactors = FALSE)
head(res)
</code></pre>

<hr>
<h2 id='episcan'>Scan pairwise epistasis</h2><span id='topic+episcan'></span>

<h3>Description</h3>

<p>Genomic interaction analysis with EPIBLASTER or epistasis-oriented Hilbert–Schmidt Independence Criterion (HSIC).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>episcan(geno1, geno2 = NULL, pheno = NULL,
  phetype = c("case-control", "quantitative"), outfile = "episcan",
  suffix = ".txt", zpthres = 1e-06, chunksize = 1000, scale = TRUE,
  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="episcan_+3A_geno1">geno1</code></td>
<td>
<p>a data.frame or matrix of the first genotype data. <code>big.matrix</code> object from <span class="pkg">bigmemory</span> also works.
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="episcan_+3A_geno2">geno2</code></td>
<td>
<p>optional. A data.frame or matrix of the second genotype data. <code>big.matrix</code> object from <span class="pkg">bigmemory</span> also works.
The columns contain the information of variables and the rows contain the information of samples.</p>
</td></tr>
<tr><td><code id="episcan_+3A_pheno">pheno</code></td>
<td>
<p>a vector (named or not). If not provided, the value of <code>geno2</code> will be used if it is a vector. The values is either case-control phenotype (0, 1) or quantitative phenotype.</p>
</td></tr>
<tr><td><code id="episcan_+3A_phetype">phetype</code></td>
<td>
<p>character string. Either &quot;case-control&quot; or &quot;quantitative&quot;.</p>
</td></tr>
<tr><td><code id="episcan_+3A_outfile">outfile</code></td>
<td>
<p>output file name. Default is &quot;episcan&quot;.</p>
</td></tr>
<tr><td><code id="episcan_+3A_suffix">suffix</code></td>
<td>
<p>suffix for output file. Default is &quot;.txt&quot;. The final result will be stored in <code>outfile</code><code>suffix</code>.</p>
</td></tr>
<tr><td><code id="episcan_+3A_zpthres">zpthres</code></td>
<td>
<p>is the significance threshold to select variant pairs for output. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="episcan_+3A_chunksize">chunksize</code></td>
<td>
<p>the number of variants in each chunk.</p>
</td></tr>
<tr><td><code id="episcan_+3A_scale">scale</code></td>
<td>
<p>a logical value to define wheter the input data needs to be normalized. Default is TRUE which means, by default, 
all the genotype data will be normalized and if the phetype is &quot;quantitative&quot;, the phenotype will also be normalized.</p>
</td></tr>
<tr><td><code id="episcan_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>


<h3>References</h3>

<p>Kam-Thong, T., D. Czamara, K. Tsuda, K. Borgwardt, C. M. Lewis, A. Erhardt-Lehmann, B. Hemmer, et al. 2011. 
&quot;EPIBLASTER-Fast Exhaustive Two-Locus Epistasis Detection Strategy Using Graphical Processing Units.&quot; Journal Article. 
European Journal of Human Genetics 19 (4): 465–71. https://doi.org/10.1038/ejhg.2010.196. 
</p>
<p>Kam-Thong, T., B. Pütz, N. Karbalai, B. Müller-Myhsok, and K. Borgwardt. 2011. &quot;Epistasis Detection on Quantitative 
Phenotypes by Exhaustive Enumeration Using GPUs.&quot; Journal Article. Bioinformatics 27 (13): i214–21. https://doi.org/10.1093/bioinformatics/btr218.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate some data
set.seed(123)
geno1 &lt;- matrix(sample(0:2, size = 1000, replace = TRUE, prob = c(0.5, 0.3, 0.2)), 
ncol = 10)
geno2 &lt;- matrix(sample(0:2, size = 2000, replace = TRUE, prob = c(0.4, 0.3, 0.3)), 
ncol = 20)
dimnames(geno1) &lt;- list(row = paste0("IND", 1:nrow(geno1)), 
col = paste0("rs", 1:ncol(geno1)))
dimnames(geno2) &lt;- list(row = paste0("IND", 1:nrow(geno2)), 
col = paste0("exm", 1:ncol(geno2)))
p1 &lt;- c(rep(0, 60), rep(1, 40))
p2 &lt;- rnorm(100)

# one genotype with case-control phenotype
episcan(geno1 = geno1, 
geno2 = NULL, 
pheno = p1, 
phetype = "case-control",
outfile = "episcan_1geno_cc", 
suffix = ".txt", 
zpthres = 0.9, 
chunksize = 10, 
scale = TRUE)

# take a look at the result
res &lt;- read.table("episcan_1geno_cc.txt", 
header = TRUE, 
stringsAsFactors = FALSE)
head(res)

# two genotypes with quantitative phenotype
episcan(geno1 = geno1, 
geno2 = geno2, 
pheno = p2, 
phetype = "quantitative",
outfile = "episcan_2geno_quant", 
suffix = ".txt", 
zpthres = 0.9, 
chunksize = 10, 
scale = TRUE)
</code></pre>

<hr>
<h2 id='getcor'>Get correlation matrix</h2><span id='topic+getcor'></span>

<h3>Description</h3>

<p>Fast calculation of correlation matrix on CPU
(the idea is from <span class="pkg">WGCNA</span> fast function for pearson correlations)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getcor(A = NULL, B = NULL, method = "pearson", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getcor_+3A_a">A</code></td>
<td>
<p>is a matrix or data.frame.</p>
</td></tr>
<tr><td><code id="getcor_+3A_b">B</code></td>
<td>
<p>is a matrix or data.frame.</p>
</td></tr>
<tr><td><code id="getcor_+3A_method">method</code></td>
<td>
<p>a character string indicating which correlation coefficient is to be computed.
Current version only supports &quot;pearson&quot; correlation.</p>
</td></tr>
<tr><td><code id="getcor_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>correlation matrix
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
A &lt;- matrix(rnorm(100, mean = 5, sd = 10), ncol = 10)
B &lt;- matrix(rnorm(200, mean = 10, sd = 100), ncol = 20)
C &lt;- getcor(A, B)
</code></pre>

<hr>
<h2 id='ithChunk'>index set for idx-th chunk of size chunk for n elements</h2><span id='topic+ithChunk'></span>

<h3>Description</h3>

<p>For proper use of this function it will return the set of variant indices
corresponding to the <code>idx</code>-th chunk of size <code>chunk</code> in <code>n</code> variants, taking
care of the case that the last chunk might have less than <code>n</code> elements.
If used with an <code>idx</code>-value outside the possible chunks (i.e., negative or
larger than <code>ceiling(n/chunk)</code>) an empty vector (<code>numeric(0)</code>) is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ithChunk(idx, n, chunk = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ithChunk_+3A_idx">idx</code></td>
<td>
<p>chunk index (which chunk, first is 1)</p>
</td></tr>
<tr><td><code id="ithChunk_+3A_n">n</code></td>
<td>
<p>total number of variants</p>
</td></tr>
<tr><td><code id="ithChunk_+3A_chunk">chunk</code></td>
<td>
<p>desired chunksize</p>
</td></tr>
</table>


<h3>Value</h3>

<p>index range into variants for chunk <code>idx</code> (see details)
</p>

<hr>
<h2 id='WriteSnpPairs'>Write out epistasis result (normal matrix)</h2><span id='topic+WriteSnpPairs'></span>

<h3>Description</h3>

<p>Write out the result of epistasis analysis. Z score matrix is not a symmetric matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WriteSnpPairs(Zmatrix, indexArr, outfile = "NONE", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WriteSnpPairs_+3A_zmatrix">Zmatrix</code></td>
<td>
<p>is the Z score matrix (non-symmetric matrix).</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_+3A_indexarr">indexArr</code></td>
<td>
<p>is the index of Zmarix whose z score is over the given <code>zpthres</code>.</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_+3A_outfile">outfile</code></td>
<td>
<p>is the SNP pairs file for the second stage.</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>

<hr>
<h2 id='WriteSnpPairs_sym'>Write out epistasis result (symmetric matrix)</h2><span id='topic+WriteSnpPairs_sym'></span>

<h3>Description</h3>

<p>Write out the result of epistasis analysis. Z score matrix is a symmetric matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WriteSnpPairs_sym(Zmatrix, indexArr, outfile = "NONE", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WriteSnpPairs_sym_+3A_zmatrix">Zmatrix</code></td>
<td>
<p>is the Z score matrix (symmetric matrix).</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_sym_+3A_indexarr">indexArr</code></td>
<td>
<p>is the index of Zmarix whose z score is over the given <code>zpthres</code>.</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_sym_+3A_outfile">outfile</code></td>
<td>
<p>is the SNP pairs file for the second stage.</p>
</td></tr>
<tr><td><code id="WriteSnpPairs_sym_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>null
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a>
</p>

<hr>
<h2 id='ZtoP'>Convert Z-score to correponding p-value</h2><span id='topic+ZtoP'></span>

<h3>Description</h3>

<p>Convert Z score to correponding p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZtoP(z.score, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZtoP_+3A_z.score">z.score</code></td>
<td>
<p>Z-score(s) (either scalar or vector).</p>
</td></tr>
<tr><td><code id="ZtoP_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>corresponding <code class="reqn">p</code>-value(s).
</p>


<h3>Note</h3>

<p>Due to the IEEE number limits of representing doubles,
any <code class="reqn">Z</code> score over 37.51929999999999765 will be converted to a <code class="reqn">p</code>-value of 1e-309.
</p>


<h3>Author(s)</h3>

<p>Beibei Jiang <a href="mailto:beibei_jiang@psych.mpg.de">beibei_jiang@psych.mpg.de</a> and Benno Pütz <a href="mailto:puetz@psych.mpg.de">puetz@psych.mpg.de</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
