<!DOCTYPE html><html><head><title>Help for package EpiSemble</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EpiSemble}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#epiPred'><p>Epigenetic Modification Prediction</p></a></li>
<li><a href='#ImpFeatures'><p>Important Features</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Ensemble Based Machine Learning Approach for Predicting
Methylation States</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Dipro Sinha [aut, cre],
  Sunil Archak [aut],
  Dwijesh Chandra Mishra [aut],
  Tanwy Dasmandal [aut],
  Md Yeasin [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dipro Sinha &lt;diprosinha@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>DNA methylation (6mA) is a major epigenetic process by which alteration in gene expression took place without changing the DNA sequence. Predicting these sites in-vitro is laborious, time consuming as well as costly. This 'EpiSemble' package is an in-silico pipeline for predicting DNA sequences containing the 6mA sites. It uses an ensemble-based machine learning approach by combining Support Vector Machine (SVM), Random Forest (RF) and Gradient Boosting approach to predict the sequences with 6mA sites in it. This package has been developed by using the concept of Chen et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz015">doi:10.1093/bioinformatics/btz015</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, devtools, tidyverse, seqinr, Biostrings,
splitstackshape, entropy, party, stringr, tibble, doParallel,
parallel, e1071, caret, randomForest, gbm, foreach, ftrCOOL,
iterators</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-02 16:06:43 UTC; YEASIN</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-04 14:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='epiPred'>Epigenetic Modification Prediction</h2><span id='topic+epiPred'></span>

<h3>Description</h3>

<p>Predicting sequences with 6mA sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epiPred(FastaData,Species)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epiPred_+3A_fastadata">FastaData</code></td>
<td>
<p>Sequence file (.fasta format)</p>
</td></tr>
<tr><td><code id="epiPred_+3A_species">Species</code></td>
<td>
<p>Model organism</p>
</td></tr>
</table>


<h3>Value</h3>

<p>MethStatus: Sequences with their methylation state (methylated or non-methylated)
</p>


<h3>References</h3>

<p>Chen, W., Lv, H., Nie, F., &amp; Lin, H. (2019). i6mA-Pred: identifying DNA N6-methyladenine 	sites in the rice genome. Bioinformatics, 35(16), 2796-2800.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(EpiSemble)
data&lt;-system.file("exdata/test.fasta", package = "EpiSemble")
pred&lt;-epiPred(FastaData=data, Species="Rice")

</code></pre>

<hr>
<h2 id='ImpFeatures'>Important Features</h2><span id='topic+ImpFeatures'></span>

<h3>Description</h3>

<p>Find out the most suitable features for predicting sequences with 6mA sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ImpFeatures(Fastafile,Species)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ImpFeatures_+3A_fastafile">Fastafile</code></td>
<td>
<p>Sequence file in .fasta format</p>
</td></tr>
<tr><td><code id="ImpFeatures_+3A_species">Species</code></td>
<td>
<p>Model organism</p>
</td></tr>
</table>


<h3>Value</h3>

<p>test_data_input: A matrix containing important features for prediction
</p>


<h3>References</h3>

<p>Chen, W., Lv, H., Nie, F., &amp; Lin, H. (2019). i6mA-Pred: identifying DNA N6-methyladenine 	sites in the rice genome. Bioinformatics, 35(16), 2796-2800.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(EpiSemble)
data&lt;-system.file("exdata/test.fasta", package = "EpiSemble")
imp&lt;-ImpFeatures(Fastafile=data, Species="Rice")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
