<!DOCTYPE html><html><head><title>Help for package EpiTest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EpiTest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#Ancestry.list'><p>Allele ancestries for the American maize NAM dataset</p></a></li>
<li><a href='#EpiTest.fit'><p>Inference and Test</p></a></li>
<li><a href='#EpiTest.plot'><p>Display variance component or directional epistasis graphs</p></a></li>
<li><a href='#EpiTest.plot.mean'><p>Display directional epistasis plot</p></a></li>
<li><a href='#EpiTest.plot.var'><p>Display variance component graphs</p></a></li>
<li><a href='#Parents'><p>Phenotypes for parental lines of the American maize NAM dataset</p></a></li>
<li><a href='#pchisqmix'><p>Chi-Squared Mixtures Distribution Function</p></a></li>
<li><a href='#Pheno.list'><p>Phenotypes for the American maize NAM dataset</p></a></li>
<li><a href='#VarList.comp'><p>Compute covariance matrices</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Test for Gene x Gene Interactions in Bi-Parental Populations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to test for gene x gene interactions in a
    bi-parental population of inbred lines. The data are fitted with the
    mixed linear model described in Rio et al. (2022) &lt;<a href="https://doi.org/10.1101%2F2022.12.18.520958">doi:10.1101/2022.12.18.520958</a>&gt;, 
    that accounts for gene x gene interactions
    at both the fixed effect and variance levels. The package also
    provides graphical tools to display the gene x gene interaction trend at
    the mean level and the variance component analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, ggrepel, magrittr, MM4LMM, purrr, stringr,
utils</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-20 06:48:51 UTC; rio</td>
</tr>
<tr>
<td>Author:</td>
<td>Simon Rio <a href="https://orcid.org/0000-0001-7014-8789"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Tristan Mary-Huard
    <a href="https://orcid.org/0000-0002-3839-9067"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Simon Rio &lt;simon.rio@cirad.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-20 12:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='Ancestry.list'>Allele ancestries for the American maize NAM dataset</h2><span id='topic+Ancestry.list'></span>

<h3>Description</h3>

<p><code>Ancestry.list</code> is a list of numeric matrices corresponding to three populations
of the American maize NAM experiment. Each matrix has 0/1 entries representing
the ancestry of alleles (0: homozygous for B73 alleles and 1: homozygous
for alternative parent alleles).
Each matrix includes 191 to 196 recombinant inbred lines characterized at 1,106 markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ancestry.list
</code></pre>


<h3>Format</h3>

<p>A list of three numeric matrices
</p>

<hr>
<h2 id='EpiTest.fit'>Inference and Test</h2><span id='topic+EpiTest.fit'></span>

<h3>Description</h3>

<p>This function fits the EpiTest model on a bi-parental population
dataset, then outputs the estimates of the fixed and random effects, along
with their associated test results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EpiTest.fit(Ancestry, Pheno, Trait, ParentName, Parents = NULL, Weight = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EpiTest.fit_+3A_ancestry">Ancestry</code></td>
<td>
<p>a numeric matrix with 0/1 entries representing the allele ancestries
of a bi-parental population at a set of markers (0: homozygous for parent A alleles and
1: homozygous for parent B alleles)</p>
</td></tr>
<tr><td><code id="EpiTest.fit_+3A_pheno">Pheno</code></td>
<td>
<p>a data.frame with a <code>Genotype</code> character column indicating the names
of individuals, a numeric column for each trait to be analyzed, and possibly
a <code>Weight</code> numeric column indicating the number of observation for each
individual. Missing values are discarded if present for a trait</p>
</td></tr>
<tr><td><code id="EpiTest.fit_+3A_trait">Trait</code></td>
<td>
<p>a string indicating the name of the phenotypic trait</p>
</td></tr>
<tr><td><code id="EpiTest.fit_+3A_parentname">ParentName</code></td>
<td>
<p>a character vector of length 2 providing the parent names where the
first element should be named P0 (parent whose allele ancestries are coded 0) and
the second element should be named P1 (parent whose allele ancestries are coded 1)</p>
</td></tr>
<tr><td><code id="EpiTest.fit_+3A_parents">Parents</code></td>
<td>
<p>(optional) a data.frame with a <code>Genotype</code> character column indicating
the names of parents, a <code>Family</code> character column indicating the names of
families/populations, a numeric column for each trait to be analyzed, and possibly
a <code>Weight</code> numeric column indicating the number of observation for each parent
(see Details)</p>
</td></tr>
<tr><td><code id="EpiTest.fit_+3A_weight">Weight</code></td>
<td>
<p>a boolean indicating whether weights indicating the number of
observations should be used for the inference. If TRUE, a column <code>Weight</code>
should be found in Pheno and Parents (see Details)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Depending on cases, the data may be organized in two different ways.
</p>
<p>In Case 1, the <code>Ancestry</code> matrix and the <code>Pheno</code> data.frame respectively provide
the whole ancestry and phenotypic data, including the parental data.
</p>
<p>In Case 2, the phenotypic data are split into two parts, the parental data being included
in a separate <code>Parents</code> data.frame.
</p>
<p>In both cases, one may provide a set of weights associated to each individual by including
a <code>Weight</code> column in the <code>Pheno</code> and <code>Parents</code> data.frames. These weights
correspond to the number of observations that where used to compute the parental BLUPS/mean values.
These weights must be provided as a <code>Weight</code> column in the <code>Parents</code> and <code>Pheno</code>
data.frame.
</p>
<p>The fitted EpiTest model includes three variance components: the segregation variance,
the (segregation <code>x</code> segregation) variance and the error variance.
For each genetic variance a likelihood ratio test is performed.
</p>
<p>The fitted EpiTest model includes three fixed parameters: the intercept,
the linear regression coefficient (beta) on parent proportions and the quadratic
regression coefficient (delta) on parent proportions  that only involves epistatic effects.
Each fixed parameter is tested using a Wald test.
</p>
<p>Additionally, epistasis can be tested by testing that both the (segregation <code>x</code> segregation) variance component
and the quadratic mean component (delta) are null through a likelihood ratio test. Note that in this
case the LRT is based on the full (i.e. unrestricted) likelihoods.
</p>
<p>The function outputs a list of five items: Beta is the vector of estimated
fixed effects, Sigma2 is the vector of estimated variances, Test_full, Test_fixed,
Text_random are 3 data.table that provide the results of the tests for no epistasis (jointly tested on fixed and variance
components), and for the fixed and
random effects, respectively.
</p>


<h3>Value</h3>

<p>A list of four items: Beta, Sigma2, Test_fixed, Test_random
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## One bi-parental population, no weighting and no parental phenotypes
data(Pheno.list)
data(Ancestry.list)
Ancestry &lt;- Ancestry.list[[1]]
Pheno &lt;- Pheno.list[[1]]
ParentName &lt;- c(P0 = 'B73',P1 = 'B97')
ETest.1 &lt;- EpiTest.fit(Ancestry = Ancestry,
                       Pheno = Pheno,
                       ParentName = ParentName,
                       Trait = "GDD_DTA")

## One bi-parental population, with weights and parental phenotypes
data(Parents)
ETest.2 &lt;- EpiTest.fit(Ancestry = Ancestry,
                       Pheno = Pheno,
                       ParentName = ParentName,
                       Trait = "GDD_DTA",
                       Parents = Parents,
                       Weight=TRUE)

## Full NAM analysis, with weights and parental phenotypes
Parent.list &lt;- Parents$Genotype[-1]
names(Parent.list) &lt;- Parents$Family[-1]
ETest.nam &lt;- purrr::imap(Parent.list, ~ EpiTest.fit(Ancestry = Ancestry.list[[.y]],
                                                    Pheno = Pheno.list[[.y]],
                                                    ParentName=c(P0 = 'B73',P1 = .x),
                                                    Parents = Parents,
                                                    Trait = 'GDD_DTA',
                                                    Weight = TRUE))
</code></pre>

<hr>
<h2 id='EpiTest.plot'>Display variance component or directional epistasis graphs</h2><span id='topic+EpiTest.plot'></span>

<h3>Description</h3>

<p>Display variance component or directional epistasis graphs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EpiTest.plot(
  ETest,
  Title = NULL,
  Family_names = NULL,
  Colors = NULL,
  Type = "mean",
  Alpha = 5/100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EpiTest.plot_+3A_etest">ETest</code></td>
<td>
<p>a list as obtained from the EpiTest.fit function</p>
</td></tr>
<tr><td><code id="EpiTest.plot_+3A_title">Title</code></td>
<td>
<p>(optional) a character string to be used as a title for the graph</p>
</td></tr>
<tr><td><code id="EpiTest.plot_+3A_family_names">Family_names</code></td>
<td>
<p>(optional) a character vector with family names</p>
</td></tr>
<tr><td><code id="EpiTest.plot_+3A_colors">Colors</code></td>
<td>
<p>(optional) a character vector with three colors for the
graphical display (for variance graphs only)</p>
</td></tr>
<tr><td><code id="EpiTest.plot_+3A_type">Type</code></td>
<td>
<p>a string with &quot;mean&quot; to display the directional epistasis plot or
&quot;var&quot; to display the variance component graph</p>
</td></tr>
<tr><td><code id="EpiTest.plot_+3A_alpha">Alpha</code></td>
<td>
<p>type I error for the test, with 5% default value
(for directional epistasis plot only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot graph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pheno.list)
data(Ancestry.list)

## Full NAM analysis
Parent.list &lt;- Parents$Genotype[-1]
names(Parent.list) &lt;- Parents$Family[-1]
ETest.nam &lt;- purrr::imap(Parent.list, ~ EpiTest.fit(Ancestry = Ancestry.list[[.y]],
                                                    Pheno = Pheno.list[[.y]],
                                                    ParentName = c(P0 = 'B73',P1 = .x),
                                                    Parents = Parents,
                                                    Trait = 'GDD_DTA',
                                                    Weight = TRUE))
## Variance component plot
EpiTest.plot(ETest.nam,Title = 'Days to anthesis',Type = "var")

## Directional epistasis plot
EpiTest.plot(ETest.nam,Title = 'Days to anthesis',Type = "mean",Alpha = 5/100)
</code></pre>

<hr>
<h2 id='EpiTest.plot.mean'>Display directional epistasis plot</h2><span id='topic+EpiTest.plot.mean'></span>

<h3>Description</h3>

<p>Display directional epistasis plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EpiTest.plot.mean(ETest, Title = NULL, Family_names = NULL, Alpha = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EpiTest.plot.mean_+3A_etest">ETest</code></td>
<td>
<p>a list as obtained from the EpiTest.fit function</p>
</td></tr>
<tr><td><code id="EpiTest.plot.mean_+3A_title">Title</code></td>
<td>
<p>(optional) a character string to be used as a title for the graph</p>
</td></tr>
<tr><td><code id="EpiTest.plot.mean_+3A_family_names">Family_names</code></td>
<td>
<p>(optional) a character vector with family names</p>
</td></tr>
<tr><td><code id="EpiTest.plot.mean_+3A_alpha">Alpha</code></td>
<td>
<p>type I error for the test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot graph
</p>

<hr>
<h2 id='EpiTest.plot.var'>Display variance component graphs</h2><span id='topic+EpiTest.plot.var'></span>

<h3>Description</h3>

<p>Display variance component graphs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EpiTest.plot.var(ETest, Title = NULL, Family_names = NULL, Colors = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EpiTest.plot.var_+3A_etest">ETest</code></td>
<td>
<p>a list as obtained from the EpiTest.fit function</p>
</td></tr>
<tr><td><code id="EpiTest.plot.var_+3A_title">Title</code></td>
<td>
<p>(optional) a character string to be used as a title for the graph</p>
</td></tr>
<tr><td><code id="EpiTest.plot.var_+3A_family_names">Family_names</code></td>
<td>
<p>(optional) a character vector with family names</p>
</td></tr>
<tr><td><code id="EpiTest.plot.var_+3A_colors">Colors</code></td>
<td>
<p>(optional) a character vector with three colors for the
graphical display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot graph
</p>

<hr>
<h2 id='Parents'>Phenotypes for parental lines of the American maize NAM dataset</h2><span id='topic+Parents'></span>

<h3>Description</h3>

<p><code>Parents</code> is a data.frame with four rows. The first row corresponds to the
parental  inbred line common to all bi-parental populations of the American maize NAM experiment
(i.e., B73), whereas the three other rows correspond to alternative inbred parental lines used to
generate other bi-parental populations. For each parental inbred line, the phenotypic values
for five traits are provided, along with two extra columns:
Genotype (providing the line ID) and Weight (providing the weight associated
with the individual, see function EpiTest.fit for details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Parents
</code></pre>


<h3>Format</h3>

<p>A data.frame
</p>

<hr>
<h2 id='pchisqmix'>Chi-Squared Mixtures Distribution Function</h2><span id='topic+pchisqmix'></span>

<h3>Description</h3>

<p>The approximate null distribution of a likelihood ratio for 2 nested mixed models,
where both fixed and random effects are tested simultaneously, is a very specific mixture
of chi-square distributions. It depends on both the number of random effects
and the number of fixed effects to be tested simultaneously. Note that this function is a copy
of the <code>pchisqmix</code> function of the <code>TcGSA</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pchisqmix(quant, s, q, lower.tail = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pchisqmix_+3A_quant">quant</code></td>
<td>
<p>a quantile</p>
</td></tr>
<tr><td><code id="pchisqmix_+3A_s">s</code></td>
<td>
<p>number of fixed effects to be tested</p>
</td></tr>
<tr><td><code id="pchisqmix_+3A_q">q</code></td>
<td>
<p>number of random effects to be tested</p>
</td></tr>
<tr><td><code id="pchisqmix_+3A_lower.tail">lower.tail</code></td>
<td>
<p>logical. if TRUE (default), probabilities are <code class="reqn">P[X&lt;=x]</code>, otherwise <code class="reqn">P[X&gt;x]</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a probability.
</p>

<hr>
<h2 id='Pheno.list'>Phenotypes for the American maize NAM dataset</h2><span id='topic+Pheno.list'></span>

<h3>Description</h3>

<p><code>Pheno.list</code> is a list of three data.frames corresponding to three
populations of the American maize NAM experiment.
Each data.frame contains the phenotypic values of 191 to 196 recombinant
inbred lines (RILs) evaluated for five different traits, along with
two extra columns:
Genotype (providing the RIL ID) and Weight (providing the weight associated
with the RIL, see function EpiTest.fit for details).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pheno.list
</code></pre>


<h3>Format</h3>

<p>A list of three data.frames
</p>

<hr>
<h2 id='VarList.comp'>Compute covariance matrices</h2><span id='topic+VarList.comp'></span>

<h3>Description</h3>

<p>This function computes the covariance matrices associated with
the variance components of the EpiTest model: the segregation variance,
the (segregation <code>x</code> segregation) variance and the error variance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VarList.comp(Ancestry)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VarList.comp_+3A_ancestry">Ancestry</code></td>
<td>
<p>a matrix with 0/1 entries representing the allele ancestries
of a bi-parental population at a set of markers  (0: homozygous for parent A
alleles and 1: homozygous for parent B alleles)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three covariance matrices
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## One bi-parental population, no weighting
data(Ancestry.list)
Ancestry &lt;- Ancestry.list[[1]]
VarList &lt;- VarList.comp(Ancestry = Ancestry)
purrr::map(VarList,~.x[1:5,1:5])
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
