<!DOCTYPE html><html><head><title>Help for package EstMix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EstMix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_1d'><p>Return mixture estimation of a normal and a tumor</p>
Takes BAF, LRR, chr, x, gt, seg_raw</a></li>
<li><a href='#calc_1d_wgs'><p>Return mixture estimation for percentage of normal cells and tumor (1 normal + 1 tumor) with wgs data</p>
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code></a></li>
<li><a href='#calc_2d'><p>Return mixture estimation of a normal and 2 tumors</p>
Takes BAF, LRR, chr, x, gt, seg_raw</a></li>
<li><a href='#calc_2d_wgs'><p>Return mixture estimation for percentage of normal cells and tumor (1 normal + 2 tumors) with wgs data</p>
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code></a></li>
<li><a href='#calc_3d'><p>Return mixture estimation of a normal and 3 tumors</p>
Takes BAF, LRR, chr, x, gt, seg_raw</a></li>
<li><a href='#calc_n'><p>return sqrt(n)</p></a></li>
<li><a href='#calcll_1d_baf'><p>calculate baf_1d for 1 normal + 1 tumor case</p></a></li>
<li><a href='#calcll_baf'><p>calculate baf for 1 normal + 2 tumors case</p></a></li>
<li><a href='#calcll_cpp'><p>calculate likelihood for 1 normal + 2 tumors case</p></a></li>
<li><a href='#calcll_p1_cpp'><p>calculate likelihood for 1 normal + 1 tumor case</p></a></li>
<li><a href='#calcll_p3'><p>calculate likelihood for 1 normal + 3 tumors case</p></a></li>
<li><a href='#combine_close_seg'><p>combine close segmentation</p></a></li>
<li><a href='#est_mixture'><p>It is a function that takes the LRR obtained from SNP array data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2 or 3. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.</p></a></li>
<li><a href='#est_mixture_wgs'><p>It is a function that takes the count data obtained from whole genome sequencing (WGS) data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.</p></a></li>
<li><a href='#example_data'><p>ExampleData</p></a></li>
<li><a href='#f_baf'><p>f_baf</p></a></li>
<li><a href='#get_segmentation'><p>calculate segmentation</p></a></li>
<li><a href='#get_var_tcn_baf'><p>get_var_tcn_baf</p></a></li>
<li><a href='#mean2'><p>trancate data and take mean</p></a></li>
<li><a href='#preprocessing'><p>Preprocessing data</p></a></li>
<li><a href='#seg_eg'><p>Segmentation</p></a></li>
<li><a href='#sel_scale_1d'><p>select scale for 1 normal + 1 tumor case</p></a></li>
<li><a href='#sel_scale_1d_wgs'><p>select scale for 1 normal + 1 tumor case for wgs data</p></a></li>
<li><a href='#sel_scale_2d'><p>select scale for 1 normal + 2 tumors case</p></a></li>
<li><a href='#sel_scale_2d_wgs'><p>select scale for 1 normal + 2 tumors case for wgs data</p></a></li>
<li><a href='#wgs_eg'><p>wgsData</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tumor Clones Percentage Estimations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Xuan You &lt;youxuan90@gmail.com&gt;, Yichen Cheng &lt;ycheng11@gsu.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuan You &lt;youxuan90@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Includes R functions for the estimation of tumor clones percentages for both snp data and (whole) genome sequencing data. See Cheng, Y., Dai, J. Y., Paulson, T. G., Wang, X., Li, X., Reid, B. J., &amp; Kooperberg, C. (2017). Quantification of multiple tumor clones using gene array and sequencing data. The Annals of Applied Statistics, 11(2), 967-991, &lt;<a href="https://doi.org/10.1214%2F17-AOAS1026">doi:10.1214/17-AOAS1026</a>&gt; for more details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.12), PSCBS</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-13 00:31:16 UTC; xuan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-09-13 04:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_1d'>Return mixture estimation of a normal and a tumor
Takes BAF, LRR, chr, x, gt, seg_raw</h2><span id='topic+calc_1d'></span>

<h3>Description</h3>

<p>Return mixture estimation of a normal and a tumor
Takes BAF, LRR, chr, x, gt, seg_raw
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_1d(BAF, LRR, chr, x, GT, seg_raw)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_1d_+3A_baf">BAF</code></td>
<td>
<p>vector containing B allen frequency (BAF)</p>
</td></tr>
<tr><td><code id="calc_1d_+3A_lrr">LRR</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_1d_+3A_chr">chr</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_1d_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_1d_+3A_gt">GT</code></td>
<td>
<p>vector of factors containing genotype</p>
</td></tr>
<tr><td><code id="calc_1d_+3A_seg_raw">seg_raw</code></td>
<td>
<p>dataframe about segmentation</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1</code></td>
<td>
<p>percentage of tumor for optimal solution 1</p>
</td></tr> <tr><td><code>sol2</code></td>
<td>
<p>percentage of tumor 1 for optimal solution 2</p>
</td></tr>
</table>

<hr>
<h2 id='calc_1d_wgs'>Return mixture estimation for percentage of normal cells and tumor (1 normal + 1 tumor) with wgs data
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code></h2><span id='topic+calc_1d_wgs'></span>

<h3>Description</h3>

<p>Return mixture estimation for percentage of normal cells and tumor (1 normal + 1 tumor) with wgs data
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_1d_wgs(baf, lrr, n_baf, nrc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_1d_wgs_+3A_baf">baf</code></td>
<td>
<p>a numeric vector. Each element is the mean adjusted B allele frequency for that segment, calculated as the mean of baf_tumor/baf_normal/2 for that segment</p>
</td></tr>
<tr><td><code id="calc_1d_wgs_+3A_lrr">lrr</code></td>
<td>
<p>a numeric vector. Each element is the log ratio of the tumor read count and the normal read count for a segment, defined as log(tumorCount/normalCount)</p>
</td></tr>
<tr><td><code id="calc_1d_wgs_+3A_n_baf">n_baf</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="calc_1d_wgs_+3A_nrc">nrc</code></td>
<td>
<p>a numeric vector. Each element is the normal read count of the segment divided by two</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1</code></td>
<td>
<p>a numeric number. It provides the estimated percentages of normal from the best solution. The number is the percentage of the estimated normal percentage.</p>
</td></tr>
<tr><td><code>sol2</code></td>
<td>
<p>a numeric number. It provides the estimated percentages of normal from the second best solution. The number is the percentage of the estimated normal percentage.</p>
</td></tr>
</table>

<hr>
<h2 id='calc_2d'>Return mixture estimation of a normal and 2 tumors
Takes BAF, LRR, chr, x, gt, seg_raw</h2><span id='topic+calc_2d'></span>

<h3>Description</h3>

<p>Return mixture estimation of a normal and 2 tumors
Takes BAF, LRR, chr, x, gt, seg_raw
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_2d(BAF, LRR, chr, x, GT, seg_raw)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_2d_+3A_baf">BAF</code></td>
<td>
<p>vector containing B allen frequency (BAF)</p>
</td></tr>
<tr><td><code id="calc_2d_+3A_lrr">LRR</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_2d_+3A_chr">chr</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_2d_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_2d_+3A_gt">GT</code></td>
<td>
<p>vectors of factors containing genotype</p>
</td></tr>
<tr><td><code id="calc_2d_+3A_seg_raw">seg_raw</code></td>
<td>
<p>dataframe about segmentation</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1</code></td>
<td>
<p>a numeric vector of length 2. It provides the estimated percentages of normal and tumor from the best solution. The first number is the percentage of the estimated normal percentage. The second number-1 is the percentage of the estimated tumor 1 percentage</p>
</td></tr> <tr><td><code>sol2</code></td>
<td>
<p>a numeric vector of length 2. It provides the estimated percentages of normal and tumor from the second best solution. The first number is the percentage of the estimated normal percentage. The second number-1 is the percentage of the estimated tumor 1 percentage</p>
</td></tr>
</table>

<hr>
<h2 id='calc_2d_wgs'>Return mixture estimation for percentage of normal cells and tumor (1 normal + 2 tumors) with wgs data
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code></h2><span id='topic+calc_2d_wgs'></span>

<h3>Description</h3>

<p>Return mixture estimation for percentage of normal cells and tumor (1 normal + 2 tumors) with wgs data
Takes <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_2d_wgs(baf, lrr, n_baf, nrc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_2d_wgs_+3A_baf">baf</code></td>
<td>
<p>a numeric vector. Each element is the mean adjusted B allele frequency for that segment, calculated as the mean of baf_tumor/baf_normal/2 for that segment</p>
</td></tr>
<tr><td><code id="calc_2d_wgs_+3A_lrr">lrr</code></td>
<td>
<p>a numeric vector. Each element is the log ratio of the tumor read count and the normal read count for a segment, defined as log(tumorCount/normalCount)</p>
</td></tr>
<tr><td><code id="calc_2d_wgs_+3A_n_baf">n_baf</code></td>
<td>
<p>a numeric vector</p>
</td></tr>
<tr><td><code id="calc_2d_wgs_+3A_nrc">nrc</code></td>
<td>
<p>a numeric vector. Each element is the normal read count of the segment divided by two</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1</code></td>
<td>
<p>a numeric vector of length 2. It provides the estimated percentages of normal and tumor from the best solution. The first number is the percentage of the estimated normal percentage. The second number is the percentage of the estimated tumor 1 percentage</p>
</td></tr>
<tr><td><code>sol2</code></td>
<td>
<p>a numeric vector of length 2. It provides the estimated percentages of normal and tumor from the second best solution. The first number is the percentage of the estimated normal percentage. The second number is the percentage of the estimated tumor 1 percentage</p>
</td></tr>
</table>

<hr>
<h2 id='calc_3d'>Return mixture estimation of a normal and 3 tumors
Takes BAF, LRR, chr, x, gt, seg_raw</h2><span id='topic+calc_3d'></span>

<h3>Description</h3>

<p>Return mixture estimation of a normal and 3 tumors
Takes BAF, LRR, chr, x, gt, seg_raw
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_3d(BAF, LRR, chr, x, GT, seg_raw)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_3d_+3A_baf">BAF</code></td>
<td>
<p>vector containing B allen frequency (BAF)</p>
</td></tr>
<tr><td><code id="calc_3d_+3A_lrr">LRR</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_3d_+3A_chr">chr</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_3d_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="calc_3d_+3A_gt">GT</code></td>
<td>
<p>vector of factors containing genotype</p>
</td></tr>
<tr><td><code id="calc_3d_+3A_seg_raw">seg_raw</code></td>
<td>
<p>dataframe about segmentation</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1</code></td>
<td>
<p>percentage of tumor for optimal solution 1</p>
</td></tr> <tr><td><code>sol2</code></td>
<td>
<p>percentage of tumor 1 for optimal solution 2</p>
</td></tr>
</table>

<hr>
<h2 id='calc_n'>return sqrt(n)</h2><span id='topic+calc_n'></span>

<h3>Description</h3>

<p>return sqrt(n)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_n(n)
</code></pre>

<hr>
<h2 id='calcll_1d_baf'>calculate baf_1d for 1 normal + 1 tumor case</h2><span id='topic+calcll_1d_baf'></span>

<h3>Description</h3>

<p>calculate baf_1d for 1 normal + 1 tumor case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcll_1d_baf(lrr, nrc, baf, n_baf, lprior_f_2d, rlprior_f_2d, scale, pscnMax,
  MaxCn)
</code></pre>

<hr>
<h2 id='calcll_baf'>calculate baf for 1 normal + 2 tumors case</h2><span id='topic+calcll_baf'></span>

<h3>Description</h3>

<p>calculate baf for 1 normal + 2 tumors case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcll_baf(lrr, nrc, baf, n_baf, lprior_f_2d, rlprior_f_2d, scale, pscnMax,
  MaxCn)
</code></pre>

<hr>
<h2 id='calcll_cpp'>calculate likelihood for 1 normal + 2 tumors case</h2><span id='topic+calcll_cpp'></span>

<h3>Description</h3>

<p>calculate likelihood for 1 normal + 2 tumors case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcll_cpp(IT_new, B_new, lp, rlp, var_baf, var_tcn, scale, pscnMax, cnMax)
</code></pre>

<hr>
<h2 id='calcll_p1_cpp'>calculate likelihood for 1 normal + 1 tumor case</h2><span id='topic+calcll_p1_cpp'></span>

<h3>Description</h3>

<p>calculate likelihood for 1 normal + 1 tumor case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcll_p1_cpp(IT_new, B_new, lp, rlp, var_baf, var_tcn, scale, pscnMax, cnMax)
</code></pre>

<hr>
<h2 id='calcll_p3'>calculate likelihood for 1 normal + 3 tumors case</h2><span id='topic+calcll_p3'></span>

<h3>Description</h3>

<p>calculate likelihood for 1 normal + 3 tumors case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcll_p3(IT_new, B_new, var_baf, var_tcn, scale, pscnMax, cnMax)
</code></pre>

<hr>
<h2 id='combine_close_seg'>combine close segmentation</h2><span id='topic+combine_close_seg'></span>

<h3>Description</h3>

<p>combine close segmentation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_close_seg(seg_raw, var_baf, data, delta)
</code></pre>

<hr>
<h2 id='est_mixture'>It is a function that takes the LRR obtained from SNP array data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2 or 3. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.</h2><span id='topic+est_mixture'></span>

<h3>Description</h3>

<p>It is a function that takes the LRR obtained from SNP array data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2 or 3. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_mixture(BAF, LRR, chr, x, GT, seg_raw = "NA", num_tumor = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="est_mixture_+3A_baf">BAF</code></td>
<td>
<p>a numeric vector containing the B Allele Frequency for the sample, corresponding to the location (chr, x).</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_lrr">LRR</code></td>
<td>
<p>a numveric vector containing the Log R ratio for the sample, corresponding to the location (chr, x). In practice, the LRR values you include should be the raw LRR output devided by 0.55.</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_chr">chr</code></td>
<td>
<p>a factor vector containing the chromosome.</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_x">x</code></td>
<td>
<p>a numeric vector containing the location on the chromosome, measured by base pair.</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_gt">GT</code></td>
<td>
<p>a factor vector containing the genotype. Possible values are &quot;AA&quot;, &quot;AB&quot;, &quot;BB&quot; and NA.</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_seg_raw">seg_raw</code></td>
<td>
<p>Optional. A dataframe containing the segmentaiton results. If not supplied, function <code>segmentByPairedPSCBS</code> from package PSCBS will be used to obtain the segmentation. You can also use the <code>segmentByPairedPSCBS</code> function to preprocess your data set and obtain the segmentation results and use that and the input. (On examples about how to obtain the segmentation results beforehand, please see the examples section below.)</p>
</td></tr>
<tr><td><code id="est_mixture_+3A_num_tumor">num_tumor</code></td>
<td>
<p>1 or 2 or 3, indicating the number of tumor clones. 1 indicates a mixture for a normal and one tumor clone. 2 indicates a mixture for a normal and 2 tumors and so on. Default value is set to be 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1_pct</code></td>
<td>
<p>the estimated percentages for all tumor clones for optimal solution 1. Each value is between 0 and 100.</p>
</td></tr>
<tr><td><code>sol1_scale</code></td>
<td>
<p>a scaler that provide the normalization constant for LRR for optimal solution 1. That is 2*2^LRR/scale will be on the same scale as the copy number.</p>
</td></tr>
<tr><td><code>sol1_cn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 1 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol1_cn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 2 for the optimal solution. </p>
</td></tr>
<tr><td><code>sol1_pscn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 1 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol1_pscn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 2 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pct</code></td>
<td>
<p>the estimated percentages for all tumor clones for optimal solution 2. Each value is between 0 and 100.</p>
</td></tr>
<tr><td><code>sol2_scale</code></td>
<td>
<p>a scaler that provide the normalization constant for LRR for optimal solution 2. That is 2*2^LRR/scale will be on the same scale as the copy number.</p>
</td></tr>
<tr><td><code>sol2_cn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 1 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_cn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 2 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pscn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 1 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pscn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 2 for the second optimal solution.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>##########################################################
##
## short example
##
#########################################################
## first load the data
BAF &lt;- example_data$BAF
LRR &lt;- example_data$LRR ## In practice, the orignal LRR should be devided by 0.55
chr &lt;- example_data$chr
loc &lt;- example_data$x
GT &lt;- example_data$GT
gt = (GT=='BB')*2+(GT=='AB')*1.5+(GT=='AA')-1;gt[gt==(-1)]=NA

## then perform segmentation
gaps = PSCBS::findLargeGaps(x=loc,minLength=5e6,chromosome=chr)
if(!is.null(gaps)) knownSegments = PSCBS::gapsToSegments(gaps)
p &lt;- 0.0001
fit &lt;- PSCBS::segmentByPairedPSCBS(CT=2*2^LRR,betaT=BAF,muN=gt,chrom=chr,
knownSegments=knownSegments,tbn=FALSE,x=loc,seed=1, alphaTCN=p*.9,alphaDH=p*.1)
seg_eg = fit$output

## then perform tumor mixture estimation by assuming 1 tumor clones
out = est_mixture(BAF, LRR, chr, loc, GT, num_tumor = 1, seg_raw = seg_eg)
out$sol1_pct
out$sol1_scale
## References: Quantification of multiple tumor clones using gene array and sequencing data.
## Y Cheng, JY Dai, TG Paulson, X Wang, X Li, BJ Reid, C Kooperberg.
## Annals of Applied Statistics 11 (2), 967-991
## Segmentation-based detection of allelic imbalance and loss-of-heterozygosity
## in cancer cells using whole genome SNP arrays.
## J Staaf, D Lindgren, J Vallon-Christersson, A Isaksson, H Goransson, G Juliusson,
## R Rosenquist, M H, A Borg, and M Ringner
</code></pre>

<hr>
<h2 id='est_mixture_wgs'>It is a function that takes the count data obtained from whole genome sequencing (WGS) data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.</h2><span id='topic+est_mixture_wgs'></span>

<h3>Description</h3>

<p>It is a function that takes the count data obtained from whole genome sequencing (WGS) data and returns the estimated tumor and normal proportions. Currently, the function can performs the proportion estimations by assuming the number of tumor clones to be 1 or 2. The normalization step is not required and the normalization constant will be returned by this function. The function will output two sets of solutions corresponding to the top 2 optimal solutions based on the posterior distribution. You can choose according to your expertise the one that is more reasonable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_mixture_wgs(exp_data, normal_snp, tumor_snp, f_path, num_tumor = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="est_mixture_wgs_+3A_exp_data">exp_data</code></td>
<td>
<p>a string. It provides the file name of interval. exp_data.intervals should be the name of the interval file. For the format of this file, please see the example section. The file should contain 6 and only 6 columns with each column corresponds to &quot;ID&quot;,&quot;chrm&quot;,&quot;start&quot;,&quot;end&quot;,&quot;tumorCount&quot; and &quot;normalCount&quot;. It is very important to keep the order of the columns the same as listed.</p>
</td></tr>
<tr><td><code id="est_mixture_wgs_+3A_normal_snp">normal_snp</code></td>
<td>
<p>a string. It provides the file name of WGS count data for a normal sample or a control sample.</p>
</td></tr>
<tr><td><code id="est_mixture_wgs_+3A_tumor_snp">tumor_snp</code></td>
<td>
<p>a string. It provides the file name of WGS count data for the tumor sample.</p>
</td></tr>
<tr><td><code id="est_mixture_wgs_+3A_f_path">f_path</code></td>
<td>
<p>a string. It provides the absolute path of the folder that contains the files above.</p>
</td></tr>
<tr><td><code id="est_mixture_wgs_+3A_num_tumor">num_tumor</code></td>
<td>
<p>1 or 2, indicating the number of tumor clones. 1 indicates a mixture for a normal and one tumor clone. 2 indicates a mixture for a normal and 2 tumors and so on. Default value is set to be 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>sol1_pct</code></td>
<td>
<p>the estimated percentages for all tumor clones for optimal solution 1. Each value is between 0 and 100.</p>
</td></tr>
<tr><td><code>sol1_scale</code></td>
<td>
<p>sol1_scale: a scaler that provide the normalization constant for LRR for optimal solution 1. That is 2*tumor_count/normal_count will be on the same scale as the copy number.</p>
</td></tr>
<tr><td><code>sol1_cn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 1 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol1_cn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 2 for the optimal solution. </p>
</td></tr>
<tr><td><code>sol1_pscn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 1 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol1_pscn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 2 for the optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pct</code></td>
<td>
<p>the estimated percentages for all tumor clones for optimal solution 2. Each value is between 0 and 100.</p>
</td></tr>
<tr><td><code>sol2_scale</code></td>
<td>
<p>sol1_scale: a scaler that provide the normalization constant for LRR for optimal solution 2. That is 2*tumor_count/normal_count will be on the same scale as the copy number.</p>
</td></tr>
<tr><td><code>sol2_cn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 1 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_cn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated copy number for tumor 2 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pscn1</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 1 for the second optimal solution.</p>
</td></tr>
<tr><td><code>sol2_pscn2</code></td>
<td>
<p>a vector of length S, where S is the number of segments. It is the estimated parent specifit copy number for tumor 2 for the second optimal solution.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>exp_data = "data_exp_eg" ## exp_data.intervals should be the file name of the segments.
## For the format of the input files, you can use the example code below.
normal_snp = "snp_norm_eg" ## snp_norm_eg.txt should be the count file name for the normal sample.
tumor_snp = "snp_tum_eg" ## snp_tum_eg.txt should be the count file name for the tumor sample.
f_path = system.file("extdata",package="EstMix")
## f_path should be the absolute path of folder that contains the txt and interval files.
out_wgs = est_mixture_wgs(exp_data, normal_snp, tumor_snp,f_path,num_tumor = 1)
out_wgs$sol1_pct
out_wgs$sol1_scale

## for the format of the input files, please see the following code
data_exp_path = file.path(f_path, paste("/", exp_data, ".intervals", sep=""))
snp_norm_path = file.path(f_path, paste("/",normal_snp, ".txt", sep=""))
snp_tumor_path = file.path(f_path, paste("/",tumor_snp, ".txt", sep=""))
data_exp = read.table(data_exp_path);
colnames(data_exp) = c("ID","chrm","start","end","tumorCount","normalCount")
snp_norm = read.table(snp_norm_path)
snp_tum = read.table(snp_tumor_path)

## References: Quantification of multiple tumor clones using gene array and sequencing data.
## Y Cheng, JY Dai, TG Paulson, X Wang, X Li, BJ Reid, C Kooperberg.
## Annals of Applied Statistics 11 (2), 967-991
</code></pre>

<hr>
<h2 id='example_data'>ExampleData</h2><span id='topic+example_data'></span>

<h3>Description</h3>

<p>ExampleData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 5 variables: <code>BAF</code>, <code>chr</code>, <code>GT</code>, <code>LRR</code> and <code>x</code>, with chromosome number = 22</p>

<hr>
<h2 id='f_baf'>f_baf</h2><span id='topic+f_baf'></span>

<h3>Description</h3>

<p>f_baf
</p>


<h3>Usage</h3>

<pre><code class='language-R'>f_baf(x, beta, var = 8e-04)
</code></pre>

<hr>
<h2 id='get_segmentation'>calculate segmentation</h2><span id='topic+get_segmentation'></span>

<h3>Description</h3>

<p>calculate segmentation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segmentation(BAF, LRR, chr, x, GT)
</code></pre>

<hr>
<h2 id='get_var_tcn_baf'>get_var_tcn_baf</h2><span id='topic+get_var_tcn_baf'></span>

<h3>Description</h3>

<p>get_var_tcn_baf
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_var_tcn_baf(LRR_raw, BAF_raw, gt, sl = 1000)
</code></pre>

<hr>
<h2 id='mean2'>trancate data and take mean</h2><span id='topic+mean2'></span>

<h3>Description</h3>

<p>trancate data and take mean
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mean2(x)
</code></pre>

<hr>
<h2 id='preprocessing'>Preprocessing data</h2><span id='topic+preprocessing'></span>

<h3>Description</h3>

<p>Takes <code>exp_data</code>, <code>normal_snp</code>, and <code>tumor_snp</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocessing(exp_data, normal_snp, tumor_snp, f_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preprocessing_+3A_exp_data">exp_data</code></td>
<td>
<p>a string, file name of xxx</p>
</td></tr>
<tr><td><code id="preprocessing_+3A_normal_snp">normal_snp</code></td>
<td>
<p>a string, file name of xxx</p>
</td></tr>
<tr><td><code id="preprocessing_+3A_tumor_snp">tumor_snp</code></td>
<td>
<p>a string, file name of xxx</p>
</td></tr>
<tr><td><code id="preprocessing_+3A_f_path">f_path</code></td>
<td>
<p>a string, file path of the files above</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>df</code></td>
<td>
<p>a dataframe containing lrr, nrc, baf and n_baf </p>
</td></tr>
</table>

<hr>
<h2 id='seg_eg'>Segmentation</h2><span id='topic+seg_eg'></span>

<h3>Description</h3>

<p>Segmentation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seg_eg
</code></pre>


<h3>Format</h3>

<p>A data frame with segmentation info, with chromosome = 22</p>

<hr>
<h2 id='sel_scale_1d'>select scale for 1 normal + 1 tumor case</h2><span id='topic+sel_scale_1d'></span>

<h3>Description</h3>

<p>select scale for 1 normal + 1 tumor case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_scale_1d(cnMax = 6, pscnMax = 6, ngrid = 100, nslaves = 50, B_new,
  IT_new, temp)
</code></pre>

<hr>
<h2 id='sel_scale_1d_wgs'>select scale for 1 normal + 1 tumor case for wgs data</h2><span id='topic+sel_scale_1d_wgs'></span>

<h3>Description</h3>

<p>select scale for 1 normal + 1 tumor case for wgs data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_scale_1d_wgs(cnMax = 6, pscnMax = 6, ngrid = 100, nslaves = 50,
  temp)
</code></pre>

<hr>
<h2 id='sel_scale_2d'>select scale for 1 normal + 2 tumors case</h2><span id='topic+sel_scale_2d'></span>

<h3>Description</h3>

<p>select scale for 1 normal + 2 tumors case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_scale_2d(cnMax = 6, pscnMax = 6, ngrid = 100, nslaves = 50, B_new,
  IT_new, temp)
</code></pre>

<hr>
<h2 id='sel_scale_2d_wgs'>select scale for 1 normal + 2 tumors case for wgs data</h2><span id='topic+sel_scale_2d_wgs'></span>

<h3>Description</h3>

<p>select scale for 1 normal + 2 tumors case for wgs data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sel_scale_2d_wgs(cnMax = 6, pscnMax = 6, ngrid = 100, nslaves = 50,
  temp)
</code></pre>

<hr>
<h2 id='wgs_eg'>wgsData</h2><span id='topic+wgs_eg'></span>

<h3>Description</h3>

<p>wgsData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wgs_eg
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 variables: <code>baf</code>, <code>lrr</code>, <code>n_baf</code> and <code>nrc</code></p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
