<!DOCTYPE html><html><head><title>Help for package EthSEQ</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EthSEQ}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ethseq.Analysis'><p>Ancestry analysis from whole-exome and targeted sequencing data</p></a></li>
<li><a href='#ethseq.RM'><p>Create Reference Model for Ancestry Analysis</p></a></li>
<li><a href='#getModelsList'><p>List the models available</p></a></li>
<li><a href='#getSamplesInfo'><p>List the samples annotation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Ethnicity Annotation from Whole-Exome and Targeted Sequencing
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>3.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Reliable and rapid ethnicity annotation from whole exome and targeted sequencing data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, utils, parallel, grDevices, MASS (&ge; 7.3-47),
geometry (&ge; 0.3-6), data.table (&ge; 1.10.0), SNPRelate (&ge;
1.8.0), gdsfmt (&ge; 1.10.1), plot3D (&ge; 1.1), Rcpp (&ge; 0.11.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-18 07:55:29 UTC; aromanel</td>
</tr>
<tr>
<td>Author:</td>
<td>Alessandro Romanel [aut, cre],
  Davide Dalfovo [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alessandro Romanel &lt;alessandro.romanel@unitn.it&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-18 22:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ethseq.Analysis'>Ancestry analysis from whole-exome and targeted sequencing data</h2><span id='topic+ethseq.Analysis'></span>

<h3>Description</h3>

<p>This function performs ancestry analysis of a set of samples ad reports the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ethseq.Analysis(
  target.vcf = NA,
  target.gds = NA,
  bam.list = NA,
  out.dir = tempdir(),
  model.gds = NA,
  model.available = NA,
  model.assembly = "hg38",
  model.pop = "All",
  model.folder = tempdir(),
  run.genotype = FALSE,
  aseq.path = tempdir(),
  mbq = 20,
  mrq = 20,
  mdc = 10,
  cores = 1,
  verbose = TRUE,
  composite.model.call.rate = 1,
  refinement.analysis = NA,
  space = "2D",
  bam.chr.encoding = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ethseq.Analysis_+3A_target.vcf">target.vcf</code></td>
<td>
<p>Path to the sample's genotypes in VCF format</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_target.gds">target.gds</code></td>
<td>
<p>Path to the sample's genotypes in GDS format</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_bam.list">bam.list</code></td>
<td>
<p>Path to a file containing a list of BAM files paths</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_out.dir">out.dir</code></td>
<td>
<p>Path to the folder where the output of the analysis is saved</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_model.gds">model.gds</code></td>
<td>
<p>Path to a GDS file specifying the reference model</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_model.available">model.available</code></td>
<td>
<p>String specifying the pre-computed reference model to use</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_model.assembly">model.assembly</code></td>
<td>
<p>String value indicating the assembly version to download for the pre-build models</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_model.pop">model.pop</code></td>
<td>
<p>String value indicating the population to download for the pre-build models</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_model.folder">model.folder</code></td>
<td>
<p>Path to the folder where reference models are already present or downloaded when needed</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_run.genotype">run.genotype</code></td>
<td>
<p>Logical values indicating whether the ASEQ genotype should be run</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_aseq.path">aseq.path</code></td>
<td>
<p>Path to the folder where ASEQ binary is available or is downloaded when needed</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_mbq">mbq</code></td>
<td>
<p>Minmum base quality used in the pileup by ASEQ</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_mrq">mrq</code></td>
<td>
<p>Minimum read quality used in the piluep by ASEQ</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_mdc">mdc</code></td>
<td>
<p>Minimum read count acceptable  for genotype inference by ASEQ</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_cores">cores</code></td>
<td>
<p>Number of parallel cores used for the analysis</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_verbose">verbose</code></td>
<td>
<p>Print detailed information</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_composite.model.call.rate">composite.model.call.rate</code></td>
<td>
<p>SNP call rate used to run Principal Component Analysis (PCA)</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_refinement.analysis">refinement.analysis</code></td>
<td>
<p>Matrix specifying a tree of ancestry sets</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_space">space</code></td>
<td>
<p>Dimensions of PCA space used to infer ancestry (2D or 3D)</p>
</td></tr>
<tr><td><code id="ethseq.Analysis_+3A_bam.chr.encoding">bam.chr.encoding</code></td>
<td>
<p>Logical value indicating whether input BAM files have chromosomes encoded with &quot;chr&quot; prefix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical value indicating the success of the analysis
</p>

<hr>
<h2 id='ethseq.RM'>Create Reference Model for Ancestry Analysis</h2><span id='topic+ethseq.RM'></span>

<h3>Description</h3>

<p>This function creates a GDS reference model that can be used to performe EthSEQ ancestry analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ethseq.RM(
  vcf.fn,
  annotations,
  out.dir = "./",
  model.name = "Reference.Model",
  bed.fn = NA,
  verbose = TRUE,
  call.rate = 1,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ethseq.RM_+3A_vcf.fn">vcf.fn</code></td>
<td>
<p>vector of paths to genotype files in VCF format</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_annotations">annotations</code></td>
<td>
<p>data.frame with mapping of all samples names, ancestries and gender</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_out.dir">out.dir</code></td>
<td>
<p>Path to output folder</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_model.name">model.name</code></td>
<td>
<p>Name of the output model</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_bed.fn">bed.fn</code></td>
<td>
<p>path to BED file with regions of interest</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_verbose">verbose</code></td>
<td>
<p>Print detailed information</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_call.rate">call.rate</code></td>
<td>
<p>SNPs call rate cutoff for inclusion in the final reference model</p>
</td></tr>
<tr><td><code id="ethseq.RM_+3A_cores">cores</code></td>
<td>
<p>How many parallel cores to use in the reference model generation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical value indicating the success of the analysis
</p>

<hr>
<h2 id='getModelsList'>List the models available</h2><span id='topic+getModelsList'></span>

<h3>Description</h3>

<p>This function prints the list of all available models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getModelsList()
</code></pre>


<h3>Value</h3>

<p>data.frame of all available models to use with specified assembly and population
</p>

<hr>
<h2 id='getSamplesInfo'>List the samples annotation</h2><span id='topic+getSamplesInfo'></span>

<h3>Description</h3>

<p>This function prints the list of all 1,000 Genomes Project samples used to build the reference models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSamplesInfo()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
