<!DOCTYPE html><html lang="en"><head><title>Help for package europepmc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {europepmc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#europepmc'><p>europepmc - an R client for the Europe PMC RESTful article API</p></a></li>
<li><a href='#epmc_annotations_by_id'><p>Get annotations by article</p></a></li>
<li><a href='#epmc_citations'><p>Get citations for a given publication</p></a></li>
<li><a href='#epmc_db'><p>Retrieve external database entities referenced in a given publication</p></a></li>
<li><a href='#epmc_db_count'><p>Retrieve the number of database links from Europe PMC publication database</p></a></li>
<li><a href='#epmc_details'><p>Get details for individual records</p></a></li>
<li><a href='#epmc_ftxt'><p>Fetch Europe PMC full texts</p></a></li>
<li><a href='#epmc_ftxt_book'><p>Fetch Europe PMC books</p></a></li>
<li><a href='#epmc_hits'><p>Get search result count</p></a></li>
<li><a href='#epmc_hits_trend'><p>Get the yearly number of hits for a query and the total</p>
yearly number of hits for a given period</a></li>
<li><a href='#epmc_lablinks'><p>Get links to external sources</p></a></li>
<li><a href='#epmc_lablinks_count'><p>Summarise links to external sources</p></a></li>
<li><a href='#epmc_profile'><p>Obtain a summary of hit counts</p></a></li>
<li><a href='#epmc_refs'><p>Get references for a given publication</p></a></li>
<li><a href='#epmc_search'><p>Search Europe PMC publication database</p></a></li>
<li><a href='#epmc_search_'><p>Get one page of results when searching Europe PubMed Central</p></a></li>
<li><a href='#epmc_search_by_doi'><p>Search Europe PMC by DOIs</p></a></li>
<li><a href='#epmc_search_by_doi_'><p>Search Europe PMC by a DOI name</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>R Interface to the Europe PubMed Central RESTful Web Service</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.3</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-20</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/europepmc/">https://docs.ropensci.org/europepmc/</a>,
<a href="https://github.com/ropensci/europepmc/">https://github.com/ropensci/europepmc/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/europepmc/issues">https://github.com/ropensci/europepmc/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>An R Client for the Europe PubMed Central RESTful Web Service
    (see <a href="https://europepmc.org/RestfulWebService">https://europepmc.org/RestfulWebService</a> for more information). It
    gives access to both metadata on life science literature and open access
    full texts. Europe PMC indexes all PubMed content and other literature
    sources including Agricola, a bibliographic database of citations to the
    agricultural literature, or Biological Patents. In addition to bibliographic
    metadata, the client allows users to fetch citations and reference lists.
    Links between life-science literature and other EBI databases, including
    ENA, PDB or ChEMBL are also accessible. No registration or API key is
    required. See the vignettes for usage examples.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.00)</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr, jsonlite, plyr, dplyr, progress, urltools, purrr, xml2,
tibble, tidyr, rlang</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, ggplot2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-20 15:49:31 UTC; najkojahn</td>
</tr>
<tr>
<td>Author:</td>
<td>Najko Jahn [aut, cre, cph],
  Maëlle Salmon [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Najko Jahn &lt;najko.jahn@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-20 18:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='europepmc'>europepmc - an R client for the Europe PMC RESTful article API</h2><span id='topic+europepmc'></span><span id='topic+europepmc-package'></span>

<h3>Description</h3>

<p>What is europepmc?:
</p>
<p>europepmc facilitates access to Europe PMC RESTful Web Service. Europe
PMC covers life science literature and gives access to open access full
texts. Coverage is not only restricted to Europe, but articles and
abstracts are indexed from all over the world. Europe PMC ingests all PubMed
content and extends its index with other sources, including Agricola, a
bibliographic database of citations to the agricultural literature, or
Biological Patents.
</p>
<p>Besides searching abstracts and full text, europepmc can be used to
retrieve reference sections and citations, text-mined terms or cross-links
to other databases hosted by the European Bioinformatics Institute (EBI).
</p>
<p>For more information about Europe PMC, see their current paper:
Ferguson, C., Araújo, D., Faulk, L., Gou, Y., Hamelers, A., Huang, Z., 
Ide-Smith, M., Levchenko, M., Marinos, N., Nambiar, R., Nassar, M., Parkin, M.,
Pi, X., Rahman, F., Rogers, F., Roochun, Y., Saha, S., Selim, M., Shafique, Z.,
… McEntyre, J. (2020). Europe PMC in 2020. Nucleic Acids Research, 49(D1), 
D1507–D1514. <a href="https://doi.org/10.1093/nar/gkaa994">doi:10.1093/nar/gkaa994</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Najko Jahn <a href="mailto:najko.jahn@gmail.com">najko.jahn@gmail.com</a> [copyright holder]
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Maëlle Salmon [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://docs.ropensci.org/europepmc/">https://docs.ropensci.org/europepmc/</a>
</p>
</li>
<li> <p><a href="https://github.com/ropensci/europepmc/">https://github.com/ropensci/europepmc/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ropensci/europepmc/issues">https://github.com/ropensci/europepmc/issues</a>
</p>
</li></ul>


<hr>
<h2 id='epmc_annotations_by_id'>Get annotations by article</h2><span id='topic+epmc_annotations_by_id'></span>

<h3>Description</h3>

<p>Retrieve text-mined annotations contained in abstracts and open access
full-text articles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_annotations_by_id(ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_annotations_by_id_+3A_ids">ids</code></td>
<td>
<p>character vector with publication identifiers
following the structure &quot;source:ext_id&quot;, e.g. '&quot;MED:28585529&quot;'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns text-mined annotations in a tidy format with the following
variables
</p>

<dl>
<dt>source</dt><dd><p>Publication data source</p>
</dd>
<dt>ext_id</dt><dd><p>Article Identifier</p>
</dd>
<dt>pmcid</dt><dd><p>PMCID that locates full-text in Pubmed Central</p>
</dd>
<dt>prefix</dt><dd><p>Text snipped found before the annotation</p>
</dd>
<dt>exact</dt><dd><p>Annotated entity</p>
</dd>
<dt>postfix</dt><dd><p>Text snipped found after the annotation</p>
</dd>
<dt>name</dt><dd><p>Targeted entity</p>
</dd>
<dt>uri</dt><dd><p>Uniform link dictionary entry for targeted entity</p>
</dd>
<dt>id</dt><dd><p>URL to full-text occurence of the annotation</p>
</dd>
<dt>type</dt><dd><p>Type of annotation like Chemicals</p>
</dd>
<dt>section</dt><dd><p>Article section mentioning the annotation like Methods</p>
</dd>
<dt>provider</dt><dd><p>Annotation data provider</p>
</dd>
<dt>subtype</dt><dd><p>Sub-data provider</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  annotations_by_id("MED:28585529")
  # multiple ids
  annotations_by_id(c("MED:28585529", "PMC:PMC1664601"))

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_citations'>Get citations for a given publication</h2><span id='topic+epmc_citations'></span>

<h3>Description</h3>

<p>Finds works that cite a given publication.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_citations(ext_id = NULL, data_src = "med", limit = 100, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_citations_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td></tr>
<tr><td><code id="epmc_citations_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.
</p>
<p>The following three letter codes represent the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="epmc_citations_+3A_limit">limit</code></td>
<td>
<p>integer, number of results. By default, this function
returns 100 records.</p>
</td></tr>
<tr><td><code id="epmc_citations_+3A_verbose">verbose</code></td>
<td>
<p>logical, print some information on what is going on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Metadata of citing documents as data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
epmc_citations("PMC3166943", data_src = "pmc")
epmc_citations("9338777")

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_db'>Retrieve external database entities referenced in a given publication</h2><span id='topic+epmc_db'></span>

<h3>Description</h3>

<p>This function returns EBI database entities referenced in a publication from
Europe PMC RESTful Web Service.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_db(
  ext_id = NULL,
  data_src = "med",
  db = NULL,
  limit = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_db_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td></tr>
<tr><td><code id="epmc_db_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.
</p>
<p>The following three letter codes represent the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="epmc_db_+3A_db">db</code></td>
<td>
<p>character, specify database:
</p>

<dl>
<dt>'ARXPR'</dt><dd><p>Array Express, a database of functional genomics experiments</p>
</dd>
<dt>'CHEBI'</dt><dd><p>a database and ontology of chemical entities of biological
interest</p>
</dd>
<dt>'CHEMBL'</dt><dd><p>a database of bioactive drug-like small molecules</p>
</dd>
<dt>'EMBL'</dt><dd><p>now ENA, provides a comprehensive record of the world's
nucleotide sequencing information</p>
</dd>
<dt>'INTACT'</dt><dd><p>provides a freely available, open
source database system and analysis tools for molecular interaction data</p>
</dd>
<dt>'INTERPRO'</dt><dd><p>provides functional analysis of proteins by classifying
them into families and predicting domains and important sites</p>
</dd>
<dt>'OMIM'</dt><dd><p>a comprehensive and authoritative compendium of human genes and
genetic phenotypes</p>
</dd>
<dt>'PDB'</dt><dd><p>European resource for the collection,
organisation and dissemination of data on biological macromolecular
structures</p>
</dd>
<dt>'UNIPROT'</dt><dd><p>comprehensive and freely accessible
resource of protein sequence and functional information</p>
</dd>
<dt>'PRIDE'</dt><dd><p>PRIDE Archive - proteomics data repository</p>
</dd></dl>
</td></tr>
<tr><td><code id="epmc_db_+3A_limit">limit</code></td>
<td>
<p>integer, number of results. By default, this function
returns 100 records.</p>
</td></tr>
<tr><td><code id="epmc_db_+3A_verbose">verbose</code></td>
<td>
<p>logical, print some information on what is going on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cross-references as data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  epmc_db("12368864", db = "uniprot", limit = 150)
  epmc_db("25249410", db = "embl")
  epmc_db("14756321", db = "uniprot")
  epmc_db("11805837", db = "pride")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_db_count'>Retrieve the number of database links from Europe PMC publication database</h2><span id='topic+epmc_db_count'></span>

<h3>Description</h3>

<p>This function returns the number of EBI database links associated with a
publication.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_db_count(ext_id = NULL, data_src = "med")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_db_count_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td></tr>
<tr><td><code id="epmc_db_count_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Europe PMC supports cross-references between literature and the
following databases:
</p>

<dl>
<dt>'ARXPR'</dt><dd><p>Array Express, a database of functional genomics experiments</p>
</dd>
<dt>'CHEBI'</dt><dd><p>a database and ontology of chemical entities of biological
interest</p>
</dd>
<dt>'CHEMBL'</dt><dd><p>a database of bioactive drug-like small molecules</p>
</dd>
<dt>'EMBL'</dt><dd><p>now ENA, provides a comprehensive record of the world's
nucleotide sequencing information</p>
</dd>
<dt>'INTACT'</dt><dd><p>provides a freely available, open
source database system and analysis tools for molecular interaction data</p>
</dd>
<dt>'INTERPRO'</dt><dd><p>provides functional analysis of proteins by classifying
them into families and predicting domains and important sites</p>
</dd>
<dt>'OMIM'</dt><dd><p>a comprehensive and authoritative compendium of human genes and
genetic phenotypes</p>
</dd>
<dt>'PDB'</dt><dd><p>European resource for the collection,
organisation and dissemination of data on biological macromolecular
structures</p>
</dd>
<dt>'UNIPROT'</dt><dd><p>comprehensive and freely accessible
resource of protein sequence and functional information</p>
</dd>
<dt>'PRIDE'</dt><dd><p>PRIDE Archive - proteomics data repository</p>
</dd></dl>



<h3>Value</h3>

<p>data.frame with counts for each database
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  epmc_db_count(ext_id = "10779411")
  epmc_db_count(ext_id = "PMC3245140", data_src = "PMC")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_details'>Get details for individual records</h2><span id='topic+epmc_details'></span>

<h3>Description</h3>

<p>This function returns parsed metadata for a given publication ID
including abstract, full text links, author details including ORCID and affiliation,
MeSH terms, chemicals, grants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_details(ext_id = NULL, data_src = "med")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_details_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td></tr>
<tr><td><code id="epmc_details_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.
Other sources Europe PubMed Central supports are:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
<dt>ppr</dt><dd><p>Preprint records</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>list of data frames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
epmc_details(ext_id = "26980001")
epmc_details(ext_id = "24270414")

# PMC record
epmc_details(ext_id = "PMC4747116", data_src = "pmc")

# Other sources:
# Agricolo
epmc_details("IND43783977", data_src = "agr")
# Biological Patents
epmc_details("EP2412369", data_src = "pat")
# Chinese Biological Abstracts
epmc_details("583843", data_src = "cba")
# CiteXplore
epmc_details("C6802", data_src = "ctx")
# NHS Evidence
epmc_details("338638", data_src = "hir")
# Theses
epmc_details("409323", data_src = "eth")
# Preprint
epmc_details("PPR158112", data_src = "ppr")

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_ftxt'>Fetch Europe PMC full texts</h2><span id='topic+epmc_ftxt'></span>

<h3>Description</h3>

<p>This function loads full texts into R. Full texts are in XML format and are
only provided for the Open Access subset of Europe PMC.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_ftxt(ext_id = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_ftxt_+3A_ext_id">ext_id</code></td>
<td>
<p>character, PMCID. 
All full text publications have external IDs starting 'PMC_'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>xml_document
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  epmc_ftxt("PMC3257301")
  epmc_ftxt("PMC3639880")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_ftxt_book'>Fetch Europe PMC books</h2><span id='topic+epmc_ftxt_book'></span>

<h3>Description</h3>

<p>Use this function to retrieve book XML formatted full text for the Open
Access subset of the Europe PMC bookshelf.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_ftxt_book(ext_id = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_ftxt_book_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier. All book full texts are accessible
either by the PMID or the 'NBK' book number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>xml_document
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  epmc_ftxt_book("NBK32884")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_hits'>Get search result count</h2><span id='topic+epmc_hits'></span>

<h3>Description</h3>

<p>Search over Europe PMC and retrieve the number of results found
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_hits(query = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_hits_+3A_query">query</code></td>
<td>
<p>query in the Europe PMC syntax</p>
</td></tr>
<tr><td><code id="epmc_hits_+3A_...">...</code></td>
<td>
<p>add query parameters from 'epmc_search()', e.g. synonym=true</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+epmc_search">epmc_search</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> ## Not run: 
 epmc_hits('abstract:"burkholderia pseudomallei"')
 epmc_hits('AUTHORID:"0000-0002-7635-3473"')
 
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_hits_trend'>Get the yearly number of hits for a query and the total
yearly number of hits for a given period</h2><span id='topic+epmc_hits_trend'></span>

<h3>Description</h3>

<p>Get the yearly number of hits for a query and the total
yearly number of hits for a given period
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_hits_trend(query, synonym = TRUE, data_src = "med", period = 1975:2016)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_hits_trend_+3A_query">query</code></td>
<td>
<p>query in the Europe PMC syntax</p>
</td></tr>
<tr><td><code id="epmc_hits_trend_+3A_synonym">synonym</code></td>
<td>
<p>logical, synonym search. If TRUE, synonym terms from MeSH
terminology and the UniProt synonym list are queried, too. Disabled by
default.</p>
</td></tr>
<tr><td><code id="epmc_hits_trend_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index  (<code>med</code>)
will be searched.
The following three letter codes represent the sources, which are currently
supported
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
<dt>ppr</dt><dd><p>Preprint records</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="epmc_hits_trend_+3A_period">period</code></td>
<td>
<p>a vector of years (numeric) over which to perform the search</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A similar function was used in
<a href="https://masalmon.eu/2017/05/14/evergreenreviewgraph/">https://masalmon.eu/2017/05/14/evergreenreviewgraph/</a> where
it was advised to not plot no. of hits over time for a query,
but to normalize it by the total no. of hits.
</p>


<h3>Value</h3>

<p>a data.frame (dplyr tbl_df) with year, total number of hits
(all_hits) and number of hits for the query (query_hits)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# aspirin as query
epmc_hits_trend('aspirin', period = 2006:2016, synonym = FALSE)
# link to cran packages in reference lists
epmc_hits_trend('REF:"cran.r-project.org*"', period = 2006:2016, synonym = FALSE)
# more complex with publication type review
epmc_hits_trend('(REF:"cran.r-project.org*") AND (PUB_TYPE:"Review" OR PUB_TYPE:"review-article")',
period = 2006:2016, synonym = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_lablinks'>Get links to external sources</h2><span id='topic+epmc_lablinks'></span>

<h3>Description</h3>

<p>With the External Link services, Europe PMC allows third parties to publish
links from Europe PMC to other webpages or tools. Current External Link
providers, which can be selected through Europe PMC's advanced search,
include Wikipedia, Dryad Digital Repository or other open services.
For more information, see
<a href="https://europepmc.org/labslink">https://europepmc.org/labslink</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_lablinks(
  ext_id = NULL,
  data_src = "med",
  lab_id = NULL,
  limit = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_lablinks_+3A_ext_id">ext_id</code></td>
<td>
<p>publication identifier</p>
</td></tr>
<tr><td><code id="epmc_lablinks_+3A_data_src">data_src</code></td>
<td>
<p>data source, by default Pubmed/MedLine index will be searched.
The following three letter codes represents the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="epmc_lablinks_+3A_lab_id">lab_id</code></td>
<td>
<p>character vector, identifiers of the external link service.
Use Europe PMC's advanced search form to find ids.</p>
</td></tr>
<tr><td><code id="epmc_lablinks_+3A_limit">limit</code></td>
<td>
<p>Number of records to be returned. By default, this function
returns 100 records.</p>
</td></tr>
<tr><td><code id="epmc_lablinks_+3A_verbose">verbose</code></td>
<td>
<p>print information about what's going on</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Links found as nested data_frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Not run: 
  # Fetch links
  epmc_lablinks("24007304")
  # Link to Altmetric (lab_id = "1562")
  epmc_lablinks("25389392", lab_id = "1562")

  # Links to Wikipedia
  epmc_lablinks("24007304", lab_id = "1507")

  # Link to full text copy archived through the institutional repo of
  Bielefeld University
  epmc_lablinks("12736239", lab_id = "1056")
  
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_lablinks_count'>Summarise links to external sources</h2><span id='topic+epmc_lablinks_count'></span>

<h3>Description</h3>

<p>With the External Link services, Europe PMC allows third parties to publish
links from Europe PMC to other webpages or tools. Current External Link
providers, which can be selected through Europe PMC's advanced search,
include Wikipedia, Dryad Digital Repository or the institutional repo of
Bielefeld University. For more information, see
<a href="https://europepmc.org/labslink">https://europepmc.org/labslink</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_lablinks_count(ext_id = NULL, data_src = "med")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_lablinks_count_+3A_ext_id">ext_id</code></td>
<td>
<p>publication identifier</p>
</td></tr>
<tr><td><code id="epmc_lablinks_count_+3A_data_src">data_src</code></td>
<td>
<p>data source, by default Pubmed/MedLine index will be searched.
The following three letter codes represents the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
</dl>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with counts for each database
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   ## Not run: 
   epmc_lablinks_count("24023770")
   epmc_lablinks_count("PMC3986813", data_src = "pmc")
   
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_profile'>Obtain a summary of hit counts</h2><span id='topic+epmc_profile'></span>

<h3>Description</h3>

<p>This functions returns the number of results found for your query,
and breaks it down to the various publication types, data sources, and
subsets Europe PMC provides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_profile(query = NULL, synonym = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_profile_+3A_query">query</code></td>
<td>
<p>character, search query. For more information on how to
build a search query, see <a href="https://europepmc.org/Help">https://europepmc.org/Help</a></p>
</td></tr>
<tr><td><code id="epmc_profile_+3A_synonym">synonym</code></td>
<td>
<p>logical, synonym search. If TRUE, synonym terms from MeSH
terminology and the UniProt synonym list are queried, too. Enabled by
default.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  epmc_profile('malaria')
  # use field search, e.g. query materials and reference section for
  # mentions of "ropensci"
  epmc_profile('(METHODS:"ropensci")')
 
## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_refs'>Get references for a given publication</h2><span id='topic+epmc_refs'></span>

<h3>Description</h3>

<p>This function retrieves all the works listed in the bibliography of a given
article.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_refs(ext_id = NULL, data_src = "med", limit = 100, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_refs_+3A_ext_id">ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td></tr>
<tr><td><code id="epmc_refs_+3A_data_src">data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.
</p>
<p>The following three letter codes represent the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt><dd><p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt><dd><p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt><dd><p>CiteXplore</p>
</dd>
<dt>eth</dt><dd><p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt><dd><p>NHS Evidence</p>
</dd>
<dt>med</dt><dd><p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt><dd><p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt><dd><p>Biological Patents</p>
</dd>
<dt>pmc</dt><dd><p>PubMed Central</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="epmc_refs_+3A_limit">limit</code></td>
<td>
<p>integer, number of results. By default, this function
returns 100 records.</p>
</td></tr>
<tr><td><code id="epmc_refs_+3A_verbose">verbose</code></td>
<td>
<p>logical, print some information on what is going on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns reference section as tibble
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
epmc_refs("PMC3166943", data_src = "pmc")
epmc_refs("25378340")
epmc_refs("21753913")

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_search'>Search Europe PMC publication database</h2><span id='topic+epmc_search'></span>

<h3>Description</h3>

<p>This is the main function to search Europe PMC RESTful Web
Service (<a href="https://europepmc.org/RestfulWebService">https://europepmc.org/RestfulWebService</a>). It fully supports
the comprehensive Europe PMC query language. Simply copy &amp; paste your query
terms to R. To get familiar with the Europe PMC query syntax, check the
Advanced Search Query Builder <a href="https://europepmc.org/advancesearch">https://europepmc.org/advancesearch</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_search(
  query = NULL,
  output = "parsed",
  synonym = TRUE,
  verbose = TRUE,
  limit = 100,
  sort = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_search_+3A_query">query</code></td>
<td>
<p>character, search query. For more information on how to build a
search query, see <a href="https://europepmc.org/Help">https://europepmc.org/Help</a></p>
</td></tr>
<tr><td><code id="epmc_search_+3A_output">output</code></td>
<td>
<p>character, what kind of output should be returned. One of
'parsed', 'id_list' or 'raw' As default, parsed key metadata will be
returned as data.frame. 'id_list' returns a list of IDs and sources. Use
'raw' to get full metadata as list. Please be aware that these lists can
become very large.</p>
</td></tr>
<tr><td><code id="epmc_search_+3A_synonym">synonym</code></td>
<td>
<p>logical, synonym search. If TRUE, synonym terms from MeSH
terminology and the UniProt synonym list are queried, too.
In order to replicate results from the website, with the Rest API
you need to turn synonyms ON!</p>
</td></tr>
<tr><td><code id="epmc_search_+3A_verbose">verbose</code></td>
<td>
<p>logical, print progress bar. Activated by default.</p>
</td></tr>
<tr><td><code id="epmc_search_+3A_limit">limit</code></td>
<td>
<p>integer, limit the number of records you wish to retrieve. By
default, 100 are returned.</p>
</td></tr>
<tr><td><code id="epmc_search_+3A_sort">sort</code></td>
<td>
<p>character, relevance ranking is used by default. Use
<code>sort = 'cited'</code> for sorting by the number of citations, or
<code>sort = 'date'</code> by the most recent publications.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble
</p>


<h3>See Also</h3>

<p><a href="https://europepmc.org/Help">https://europepmc.org/Help</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#Search articles for 'Gabi-Kat'
my.data &lt;- epmc_search(query='Gabi-Kat')

#Get article metadata by DOI
my.data &lt;- epmc_search(query = 'DOI:10.1007/bf00197367')

#Get article metadata by PubMed ID (PMID)
my.data &lt;- epmc_search(query = 'EXT_ID:22246381')

#Get only PLOS Genetics article with EMBL database references
my.data &lt;- epmc_search(query = 'ISSN:1553-7404 HAS_EMBL:y')
#Limit search to 250 PLOS Genetics articles
my.data &lt;- epmc_search(query = 'ISSN:1553-7404', limit = 250)

# exclude MeSH synonyms in search
my.data &lt;- epmc_search(query = 'aspirin', synonym = FALSE)

# get 100 most cited atricles from PLOS ONE publsihed in 2014
epmc_search(query = '(ISSN:1932-6203) AND FIRST_PDATE:2014', sort = 'cited')

# print number of records found
attr(my.data, "hit_count")

# change output


## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_search_'>Get one page of results when searching Europe PubMed Central</h2><span id='topic+epmc_search_'></span>

<h3>Description</h3>

<p>In general, use <code><a href="#topic+epmc_search">epmc_search</a></code> instead. It calls this function, calling all
pages within the defined limit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_search_(
  query = NULL,
  limit = 100,
  output = "parsed",
  page_token = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_search__+3A_query">query</code></td>
<td>
<p>character, search query. For more information on how to
build a search query, see <a href="https://europepmc.org/Help">https://europepmc.org/Help</a></p>
</td></tr>
<tr><td><code id="epmc_search__+3A_limit">limit</code></td>
<td>
<p>integer, limit the number of records you wish to retrieve.
By default, 25 are returned.</p>
</td></tr>
<tr><td><code id="epmc_search__+3A_output">output</code></td>
<td>
<p>character, what kind of output should be returned. One of 'parsed', 'id_list'
or 'raw' As default, parsed key metadata will be returned as data.frame.
'id_list returns a list of IDs and sources.
Use 'raw' to get full metadata as list. Please be aware that these lists
can become very large.</p>
</td></tr>
<tr><td><code id="epmc_search__+3A_page_token">page_token</code></td>
<td>
<p>cursor marking the page</p>
</td></tr>
<tr><td><code id="epmc_search__+3A_...">...</code></td>
<td>
<p>further params from <code><a href="#topic+epmc_search">epmc_search</a></code></p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+epmc_search">epmc_search</a>
</p>

<hr>
<h2 id='epmc_search_by_doi'>Search Europe PMC by DOIs</h2><span id='topic+epmc_search_by_doi'></span>

<h3>Description</h3>

<p>Look up DOIs indexed in Europe PMC and get metadata back.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_search_by_doi(doi = NULL, output = "parsed")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_search_by_doi_+3A_doi">doi</code></td>
<td>
<p>character vector containing DOI names.</p>
</td></tr>
<tr><td><code id="epmc_search_by_doi_+3A_output">output</code></td>
<td>
<p>character, what kind of output should be returned. One of
'parsed', 'id_list' or 'raw' As default, parsed key metadata will be
returned as data.frame. 'id_list' returns a list of IDs and sources. Use
'raw' to get full metadata as list. Please be aware that these lists can
become very large.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# single DOI name
epmc_search_by_doi(doi = "10.1161/strokeaha.117.018077")
# multiple DOIname in a vector
my_dois &lt;- c(
  "10.1159/000479962",
  "10.1002/sctm.17-0081",
  "10.1161/strokeaha.117.018077",
  "10.1007/s12017-017-8447-9")
epmc_search_by_doi(doi = my_dois)
# full metadata
epmc_search_by_doi(doi = my_dois, output = "raw")

## End(Not run)
</code></pre>

<hr>
<h2 id='epmc_search_by_doi_'>Search Europe PMC by a DOI name</h2><span id='topic+epmc_search_by_doi_'></span>

<h3>Description</h3>

<p>Please use <code><a href="#topic+epmc_search_by_doi">epmc_search_by_doi</a></code> instead. It calls this
method, returning open access status information from all your requests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epmc_search_by_doi_(doi, .pb = NULL, output = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="epmc_search_by_doi__+3A_doi">doi</code></td>
<td>
<p>character vector containing DOI names.</p>
</td></tr>
<tr><td><code id="epmc_search_by_doi__+3A_.pb">.pb</code></td>
<td>
<p>progress bar object</p>
</td></tr>
<tr><td><code id="epmc_search_by_doi__+3A_output">output</code></td>
<td>
<p>character, what kind of output should be returned. One of
'parsed', 'id_list' or 'raw' As default, parsed key metadata will be
returned as data.frame. 'id_list' returns a list of IDs and sources. Use
'raw' to get full metadata as list. Please be aware that these lists can
become very large.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  epmc_search_by_doi_("10.1159/000479962")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
