<!DOCTYPE html><html><head><title>Help for package EvaluateCore</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<script type="text/javascript" src="mathjax-config.js"></script>
<script type="text/javascript" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {EvaluateCore}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bar.evaluate.core'><p>Bar Plots</p></a></li>
<li><a href='#box.evaluate.core'><p>Box Plots</p></a></li>
<li><a href='#cassava_CC'><p>IITA Cassava Germplasm Data - Core Collection</p></a></li>
<li><a href='#cassava_EC'><p>IITA Cassava Germplasm Data - Entire Collection</p></a></li>
<li><a href='#checks.evaluate.core'><p>Common checks for all functions</p></a></li>
<li><a href='#chisquare.evaluate.core'><p>Chi-squared Test for Homogeneity</p></a></li>
<li><a href='#corr.evaluate.core'><p>Phenotypic Correlations</p></a></li>
<li><a href='#coverage.evaluate.core'><p>Class Coverage</p></a></li>
<li><a href='#cr.evaluate.core'><p>Coincidence Rate of Range</p></a></li>
<li><a href='#dist.evaluate.core'><p>Distance Measures</p></a></li>
<li><a href='#diversity.evaluate.core'><p>Diversity Indices</p></a></li>
<li><a href='#EvaluateCore-deprecated'><p>Deprecated functions in package <span class="pkg">EvaluateCore</span>.</p></a></li>
<li><a href='#freqdist.evaluate.core'><p>Frequency Distribution Histogram</p></a></li>
<li><a href='#iqr.evaluate.core'><p>Interquartile Range</p></a></li>
<li><a href='#levene.evaluate.core'><p>Levene's Test</p></a></li>
<li><a href='#pca.evaluate.core'><p>Principal Component Analysis</p></a></li>
<li><a href='#pdfdist.evaluate.core'><p>Distance Between Probability Distributions</p></a></li>
<li><a href='#percentdiff.evaluate.core'><p>Percentage Difference of Means and Variances</p></a></li>
<li><a href='#qq.evaluate.core'><p>Quantile-Quantile Plots</p></a></li>
<li><a href='#shannon.evaluate.core-deprecated'><p>Shannon-Weaver Diversity Index</p></a></li>
<li><a href='#signtest.evaluate.core'><p>Sign Test</p></a></li>
<li><a href='#snk.evaluate.core'><p>Student-Newman-Keuls Test</p></a></li>
<li><a href='#ttest.evaluate.core'><p>Student's t Test</p></a></li>
<li><a href='#vr.evaluate.core'><p>Variable Rate of Coefficient of Variation</p></a></li>
<li><a href='#wilcox.evaluate.core'><p>Wilcoxon Rank Sum Test</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Quality Evaluation of Core Collections</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements various quality evaluation statistics to assess the
    value of plant germplasm core collections using qualitative and 
    quantitative phenotypic trait data according to Odong et al. (2015) 
    &lt;<a href="https://doi.org/10.1007%2Fs00122-012-1971-y">doi:10.1007/s00122-012-1971-y</a>&gt;.</td>
</tr>
<tr>
<td>Copyright:</td>
<td>2018-2022, ICAR-NBPGR</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/aravind-j/EvaluateCore">https://github.com/aravind-j/EvaluateCore</a>
<a href="https://CRAN.R-project.org/package=EvaluateCore">https://CRAN.R-project.org/package=EvaluateCore</a>
<a href="https://aravind-j.github.io/EvaluateCore/">https://aravind-j.github.io/EvaluateCore/</a>
<a href="https://doi.org/10.5281/zenodo.3875930">https://doi.org/10.5281/zenodo.3875930</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/aravind-j/EvaluateCore/issues">https://github.com/aravind-j/EvaluateCore/issues</a></td>
</tr>
<tr>
<td>RdMacros:</td>
<td>mathjaxr, Rdpack</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>agricolae, boot, car, cluster, dplyr, entropy, ggcorrplot,
ggplot2, grDevices, gridExtra, kSamples, mathjaxr, psych,
reshape2, Rdpack, stats, vegan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>corehunter, pander, rJava (&ge; 0.9-8)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-30 12:06:32 UTC; J. Aravind</td>
</tr>
<tr>
<td>Author:</td>
<td>J. Aravind <a href="https://orcid.org/0000-0002-4791-442X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Vikender Kaur <a href="https://orcid.org/0000-0001-9698-7329"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Dhammaprakash Pandhari Wankhede
    <a href="https://orcid.org/0000-0001-6384-8664"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Joghee Nanjundan <a href="https://orcid.org/0000-0001-6254-7341"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  ICAR-NBGPR [cph] (www.nbpgr.ernet.in)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>J. Aravind &lt;j.aravind@icar.gov.in&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-03 12:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bar.evaluate.core'>Bar Plots</h2><span id='topic+bar.evaluate.core'></span>

<h3>Description</h3>

<p>Plot Bar plots to graphically compare the frequency distributions of
qualitative traits between entire collection (EC) and core set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bar.evaluate.core(data, names, qualitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bar.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="bar.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="bar.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="bar.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the <code>ggplot</code> objects of relative frequency bar plots
of CS and EC for each trait specified as <code>qualitative</code>.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+barplot">barplot</a></code>, <code><a href="ggplot2.html#topic+geom_bar">geom_bar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

bar.evaluate.core(data = ec, names = "genotypes",
                  qualitative = qual, selected = core)

</code></pre>

<hr>
<h2 id='box.evaluate.core'>Box Plots</h2><span id='topic+box.evaluate.core'></span>

<h3>Description</h3>

<p>Plot Box-and-Whisker plots
(Tukey 1970; McGill et al. 1978) to
graphically compare the probability distributions of quantitative traits
between entire collection (EC) and core set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>box.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="box.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="box.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="box.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="box.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the <code>ggplot</code> objects of box plots of CS and EC for
each trait specified as <code>quantitative</code>.
</p>


<h3>References</h3>

<p>McGill R, Tukey JW, Larsen WA (1978).
&ldquo;Variations of box plots.&rdquo;
<em>The American Statistician</em>, <b>32</b>(1), 12.<br /><br /> Tukey JW (1970).
<em>Exploratory Data Analysis. Preliminary edition</em>.
Addison-Wesley.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+boxplot">boxplot</a></code>, <code><a href="ggplot2.html#topic+geom_boxplot">geom_boxplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

box.evaluate.core(data = ec, names = "genotypes",
                  quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='cassava_CC'>IITA Cassava Germplasm Data - Core Collection</h2><span id='topic+cassava_CC'></span>

<h3>Description</h3>

<p>An example germplasm characterisation data of a core collection generated
from 1591 accessions of IITA Cassava collection
(International Institute of Tropical Agriculture et al. 2019) using 10 quantitative and 48
qualitative trait data with CoreHunter3
(<code><a href="corehunter.html#topic+corehunter">corehunter</a></code>). The core set was generated using
distance based measures giving equal weightage to Average
entry-to-nearest-entry distance (EN) and Average accession-to-nearest-entry
distance (AN). Includes data on 26 descriptors for 168 (10 % of
<code><a href="#topic+cassava_EC">cassava_EC</a></code>) accessions. It is used to demonstrate
the various functions of <code>EvaluateCore</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cassava_CC
</code></pre>


<h3>Format</h3>

<p>A data frame with 58 columns: </p>
 <dl>
<dt>CUAL</dt><dd><p>Colour of
unexpanded apical leaves</p>
</dd> <dt>LNGS</dt><dd><p>Length of stipules</p>
</dd>
<dt>PTLC</dt><dd><p>Petiole colour</p>
</dd> <dt>DSTA</dt><dd><p>Distribution of anthocyanin</p>
</dd>
<dt>LFRT</dt><dd><p>Leaf retention</p>
</dd> <dt>LBTEF</dt><dd><p>Level of branching at the end of
flowering</p>
</dd> <dt>CBTR</dt><dd><p>Colour of boiled tuberous root</p>
</dd> <dt>NMLB</dt><dd><p>Number
of levels of branching</p>
</dd> <dt>ANGB</dt><dd><p>Angle of branching</p>
</dd>
<dt>CUAL9M</dt><dd><p>Colours of unexpanded apical leaves at 9 months</p>
</dd>
<dt>LVC9M</dt><dd><p>Leaf vein colour at 9 months</p>
</dd> <dt>TNPR9M</dt><dd><p>Total number of
plants remaining per accession at 9 months</p>
</dd> <dt>PL9M</dt><dd><p>Petiole length at 9
months</p>
</dd> <dt>STRP</dt><dd><p>Storage root peduncle</p>
</dd> <dt>STRC</dt><dd><p>Storage root
constrictions</p>
</dd> <dt>PSTR</dt><dd><p>Position of root</p>
</dd> <dt>NMSR</dt><dd><p>Number of storage
root per plant</p>
</dd> <dt>TTRN</dt><dd><p>Total root number per plant</p>
</dd> <dt>TFWSR</dt><dd><p>Total
fresh weight of storage root per plant</p>
</dd> <dt>TTRW</dt><dd><p>Total root weight per
plant</p>
</dd> <dt>TFWSS</dt><dd><p>Total fresh weight of storage shoot per plant</p>
</dd>
<dt>TTSW</dt><dd><p>Total shoot weight per plant</p>
</dd> <dt>TTPW</dt><dd><p>Total plant weight</p>
</dd>
<dt>AVPW</dt><dd><p>Average plant weight</p>
</dd> <dt>ARSR</dt><dd><p>Amount of rotted storage root
per plant</p>
</dd> <dt>SRDM</dt><dd><p>Storage root dry matter</p>
</dd> </dl>



<h3>Details</h3>

<p>Further details on how the example dataset was built from the original data
is available
<a href="https://aravind-j.github.io/EvaluateCore/articles/additional/Example%20Core%20Data.html">online</a>.
</p>


<h3>References</h3>

<p>International Institute of Tropical Agriculture, Benjamin F, Marimagne T (2019).
&ldquo;Cassava morphological characterization. Version 2018.1.&rdquo;
<em>www.genesys-pgr.org</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cassava_CC)
summary(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

lapply(seq_along(cassava_CC[, qual]),
       function(i) barplot(table(cassava_CC[, qual][, i]),
                           xlab = names(cassava_CC[, qual])[i]))

lapply(seq_along(cassava_CC[, quant]),
       function(i) hist(table(cassava_CC[, quant][, i]),
                        xlab = names(cassava_CC[, quant])[i],
                        main = ""))

</code></pre>

<hr>
<h2 id='cassava_EC'>IITA Cassava Germplasm Data - Entire Collection</h2><span id='topic+cassava_EC'></span>

<h3>Description</h3>

<p>An example germplasm characterisation data of a subset of IITA Cassava
collection (International Institute of Tropical Agriculture et al. 2019). Includes data on
26 (out of 62) descriptors for 1684 (out of 2170) accessions. It is used to
demonstrate the various functions of <code>EvaluateCore</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cassava_EC
</code></pre>


<h3>Format</h3>

<p>A data frame with 58 columns: </p>
 <dl>
<dt>CUAL</dt><dd><p>Colour of
unexpanded apical leaves</p>
</dd> <dt>LNGS</dt><dd><p>Length of stipules</p>
</dd>
<dt>PTLC</dt><dd><p>Petiole colour</p>
</dd> <dt>DSTA</dt><dd><p>Distribution of anthocyanin</p>
</dd>
<dt>LFRT</dt><dd><p>Leaf retention</p>
</dd> <dt>LBTEF</dt><dd><p>Level of branching at the end of
flowering</p>
</dd> <dt>CBTR</dt><dd><p>Colour of boiled tuberous root</p>
</dd> <dt>NMLB</dt><dd><p>Number
of levels of branching</p>
</dd> <dt>ANGB</dt><dd><p>Angle of branching</p>
</dd>
<dt>CUAL9M</dt><dd><p>Colours of unexpanded apical leaves at 9 months</p>
</dd>
<dt>LVC9M</dt><dd><p>Leaf vein colour at 9 months</p>
</dd> <dt>TNPR9M</dt><dd><p>Total number of
plants remaining per accession at 9 months</p>
</dd> <dt>PL9M</dt><dd><p>Petiole length at 9
months</p>
</dd> <dt>STRP</dt><dd><p>Storage root peduncle</p>
</dd> <dt>STRC</dt><dd><p>Storage root
constrictions</p>
</dd> <dt>PSTR</dt><dd><p>Position of root</p>
</dd> <dt>NMSR</dt><dd><p>Number of storage
root per plant</p>
</dd> <dt>TTRN</dt><dd><p>Total root number per plant</p>
</dd> <dt>TFWSR</dt><dd><p>Total
fresh weight of storage root per plant</p>
</dd> <dt>TTRW</dt><dd><p>Total root weight per
plant</p>
</dd> <dt>TFWSS</dt><dd><p>Total fresh weight of storage shoot per plant</p>
</dd>
<dt>TTSW</dt><dd><p>Total shoot weight per plant</p>
</dd> <dt>TTPW</dt><dd><p>Total plant weight</p>
</dd>
<dt>AVPW</dt><dd><p>Average plant weight</p>
</dd> <dt>ARSR</dt><dd><p>Amount of rotted storage root
per plant</p>
</dd> <dt>SRDM</dt><dd><p>Storage root dry matter</p>
</dd> </dl>



<h3>Details</h3>

<p>Further details on how the example dataset was built from the original data
is available
<a href="https://aravind-j.github.io/EvaluateCore/articles/additional/Example%20Core%20Data.html">online</a>.
</p>


<h3>References</h3>

<p>International Institute of Tropical Agriculture, Benjamin F, Marimagne T (2019).
&ldquo;Cassava morphological characterization. Version 2018.1.&rdquo;
<em>www.genesys-pgr.org</em>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(cassava_EC)
summary(cassava_EC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

lapply(seq_along(cassava_EC[, qual]),
       function(i) barplot(table(cassava_EC[, qual][, i]),
                           xlab = names(cassava_EC[, qual])[i]))

lapply(seq_along(cassava_EC[, quant]),
       function(i) hist(table(cassava_EC[, quant][, i]),
                        xlab = names(cassava_EC[, quant])[i],
                        main = ""))

</code></pre>

<hr>
<h2 id='checks.evaluate.core'>Common checks for all functions</h2><span id='topic+checks.evaluate.core'></span>

<h3>Description</h3>

<p>Not exported. Strictly internal
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checks.evaluate.core(
  data,
  names,
  quantitative = NULL,
  qualitative = NULL,
  selected
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checks.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="checks.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="checks.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="checks.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="checks.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>

<hr>
<h2 id='chisquare.evaluate.core'>Chi-squared Test for Homogeneity</h2><span id='topic+chisquare.evaluate.core'></span>

<h3>Description</h3>

<p>Compare the distribution frequencies of qualitative traits between entire
collection (EC) and core set (CS) by Chi-squared test for homogeneity
(Pearson 1900; Snedecor and Irwin 1933).
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chisquare.evaluate.core(data, names, qualitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chisquare.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="chisquare.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="chisquare.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="chisquare.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A a data frame with the following columns. </p>
<table>
<tr><td><code>Trait</code></td>
<td>
<p>The
qualitative trait.</p>
</td></tr> <tr><td><code>EC_No.Classes</code></td>
<td>
<p>The number of classes in the trait
for EC.</p>
</td></tr> <tr><td><code>EC_Classes</code></td>
<td>
<p>The frequency of the classes in the trait for
EC.</p>
</td></tr> <tr><td><code>CS_No.Classes</code></td>
<td>
<p>The number of classes in the trait for CS.</p>
</td></tr>
<tr><td><code>CS_Classes</code></td>
<td>
<p>The frequency of the classes in the trait for CS.</p>
</td></tr>
<tr><td><code>chisq_statistic</code></td>
<td>
<p>The \(\chi^{2}\) test statistic.</p>
</td></tr>
<tr><td><code>chisq_pvalue</code></td>
<td>
<p>The p value for the test statistic.</p>
</td></tr>
<tr><td><code>chisq_significance</code></td>
<td>
<p>The significance of the test statistic (*: p
\(\leq\) 0.01; **: p \(\leq\) 0.05; ns: p \( > \) 0.05).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Pearson K (1900).
&ldquo;X. On the criterion that a given system of deviations from the probable in the case of a correlated system of variables is such that it can be reasonably supposed to have arisen from random sampling.&rdquo;
<em>The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science</em>, <b>50</b>(302), 157&ndash;175.<br /><br /> Snedecor G, Irwin MR (1933).
&ldquo;On the chi-square test for homogeneity.&rdquo;
<em>Iowa State College Journal of Science</em>, <b>8</b>, 75&ndash;81.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+chisq.test">chisq.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

chisquare.evaluate.core(data = ec, names = "genotypes",
                        qualitative = qual, selected = core)

</code></pre>

<hr>
<h2 id='corr.evaluate.core'>Phenotypic Correlations</h2><span id='topic+corr.evaluate.core'></span>

<h3>Description</h3>

<p>Compute phenotypic correlations (Pearson 1895)
between traits, plot correlation matrices as correlograms
(Friendly 2002) and calculate mantel
correlation (Legendre and Legendre 2012) between
them to compare entire collection (EC) and core set (CS). <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr.evaluate.core(data, names, quantitative, qualitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="corr.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="corr.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="corr.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="corr.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components. </p>
<table>
<tr><td><code>Correlation Matrix</code></td>
<td>
<p>The
matrix with phenotypic correlations between traits in EC (below diagonal)
and CS (above diagonal).</p>
</td></tr> <tr><td><code>Correologram</code></td>
<td>
<p>A correlogram of phenotypic
correlations between traits in EC (below diagonal) and CS (above diagonal)
as a <code>ggplot</code> object.</p>
</td></tr> <tr><td><code>Mantel Correlation</code></td>
<td>
<p>A data frame with
Mantel correlation coefficient (\(r\)) between EC and CS phenotypic
correlation matrices, it's p value and significance (*: p \(\leq\)
0.01; **: p \(\leq\) 0.05; ns: p \( > \) 0.05).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Friendly M (2002).
&ldquo;Corrgrams.&rdquo;
<em>The American Statistician</em>, <b>56</b>(4), 316&ndash;324.<br /><br /> Legendre P, Legendre L (2012).
&ldquo;Interpretation of ecological structures.&rdquo;
In <em>Developments in Environmental Modelling</em>, volume 24, 521&ndash;624.
Elsevier.<br /><br /> Pearson K (1895).
&ldquo;Note on regression and inheritance in the case of two parents.&rdquo;
<em>Proceedings of the Royal Society of London</em>, <b>58</b>, 240&ndash;242.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code>,
<code><a href="ggcorrplot.html#topic+ggcorrplot">cor_pmat</a></code>
<code><a href="ggcorrplot.html#topic+ggcorrplot">ggcorrplot</a></code>, <code><a href="vegan.html#topic+mantel">mantel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

corr.evaluate.core(data = ec, names = "genotypes", quantitative = quant,
                   qualitative = qual, selected = core)

</code></pre>

<hr>
<h2 id='coverage.evaluate.core'>Class Coverage</h2><span id='topic+coverage.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the Class Coverage (Kim et al. 2007) to
compare the distribution frequencies of qualitative traits between entire
collection (EC) and core set (CS). <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coverage.evaluate.core(data, names, qualitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coverage.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="coverage.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="coverage.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="coverage.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Class Coverage (Kim et al. 2007) is computed as
follows.
</p>
\[Class\, Coverage = \left ( \frac{1}{n} \sum_{i=1}^{n}
\frac{A_{CS_{i}}}{A_{EC_{i}}} \right ) \times 100\]
<p>Where, \(A_{CS_{i}}\) is the sets of categories in the CS for the
\(i\)th trait, \(A_{EC_{i}}\) is the sets of categories in the EC
for the \(i\)th trait and \(n\) is the total number of traits.
</p>


<h3>Value</h3>

<p>The Class Coverage value.
</p>


<h3>References</h3>

<p>Kim K, Chung H, Cho G, Ma K, Chandrabalan D, Gwag J, Kim T, Cho E, Park Y (2007).
&ldquo;PowerCore: A program applying the advanced M strategy with a heuristic search for establishing core sets.&rdquo;
<em>Bioinformatics</em>, <b>23</b>(16), 2155&ndash;2162.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

coverage.evaluate.core(data = ec, names = "genotypes",
                       qualitative = qual, selected = core)

</code></pre>

<hr>
<h2 id='cr.evaluate.core'>Coincidence Rate of Range</h2><span id='topic+cr.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the Coincidence Rate of Range (\(CR\))
(Hu et al. 2000) (originally described by
(Diwan et al. 1995) as Mean range ratio) to compare
quantitative traits of the entire collection (EC) and core set (CS).
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cr.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cr.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="cr.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="cr.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="cr.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Coincidence Rate of Range (\(CR\)) is computed as follows.
</p>
\[CR = \left ( \frac{1}{n} \sum_{i=1}^{n}
\frac{R_{CS_{i}}}{R_{EC_{i}}} \right ) \times 100\]
<p>Where, \(R_{CS_{i}}\) is the range of the \(i\)th trait in the CS,
\(R_{EC_{i}}\) is the range of the \(i\)th trait in the EC and
\(n\) is the total number of traits.
</p>
<p>A representative CS should have a \(CR\) value no less than 70%
(Diwan et al. 1995) or 80%
(Hu et al. 2000).
</p>


<h3>Value</h3>

<p>The \(CR\) value.
</p>


<h3>References</h3>

<p>Diwan N, McIntosh MS, Bauchan GR (1995).
&ldquo;Methods of developing a core collection of annual <em>Medicago</em> species.&rdquo;
<em>Theoretical and Applied Genetics</em>, <b>90</b>(6), 755&ndash;761.<br /><br /> Hu J, Zhu J, Xu HM (2000).
&ldquo;Methods of constructing core collections by stepwise clustering with three sampling strategies based on the genotypic values of crops.&rdquo;
<em>Theoretical and Applied Genetics</em>, <b>101</b>(1), 264&ndash;268.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

cr.evaluate.core(data = ec, names = "genotypes",
                 quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='dist.evaluate.core'>Distance Measures</h2><span id='topic+dist.evaluate.core'></span>

<h3>Description</h3>

<p>Compute average Entry-to-nearest-entry distance
(\(E\text{-}EN\)),
Accession-to-nearest-entry distance
(\(E\text{-}EN\)) and
Entry-to-entry distance (\(E\text{-}EN\))
(Odong et al. 2013)
to evaluate a core set (CS) selected from an entire collection (EC).
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist.evaluate.core(data, names, quantitative, qualitative, selected, d = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="dist.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="dist.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="dist.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="dist.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
<tr><td><code id="dist.evaluate.core_+3A_d">d</code></td>
<td>
<p>A distance matrix of class &quot;<code>dist</code>&quot; with individual names in
the <code>names</code> column in data as labels. If <code>NULL</code> (default),
then a distance matrix is computed using Gower's metric.
(Gower 1971).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the average values of
\(E\text{-}EN\),
\(E\text{-}EN\) and
\(E\text{-}EN\).
</p>


<h3>References</h3>

<p>Gower JC (1971).
&ldquo;A general coefficient of similarity and some of its properties.&rdquo;
<em>Biometrics</em>, <b>27</b>(4), 857&ndash;871.<br /><br /> Odong TL, Jansen J, van Eeuwijk FA, van Hintum TJL (2013).
&ldquo;Quality of core collections for effective utilisation of genetic resources review, discussion and interpretation.&rdquo;
<em>Theoretical and Applied Genetics</em>, <b>126</b>(2), 289&ndash;305.
</p>


<h3>See Also</h3>

<p><code><a href="corehunter.html#topic+evaluateCore">evaluateCore</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

dist.evaluate.core(data = ec, names = "genotypes", quantitative = quant,
                   qualitative = qual, selected = core)


####################################
# Compare with corehunter
####################################

library(corehunter)
# Prepare phenotype dataset
dtype &lt;- c(rep("RD", length(quant)),
           rep("NS", length(qual)))
rownames(ec) &lt;- ec[, "genotypes"]
ecdata &lt;- corehunter::phenotypes(data = ec[, c(quant, qual)],
                                 types = dtype)

# Compute average distances
EN &lt;- evaluateCore(core = rownames(cassava_CC), data = ecdata,
                   objective = objective("EN", "GD"))
AN &lt;- evaluateCore(core = rownames(cassava_CC), data = ecdata,
                   objective = objective("AN", "GD"))
EE &lt;- evaluateCore(core = rownames(cassava_CC), data = ecdata,
                   objective = objective("EE", "GD"))
EN
AN
EE


</code></pre>

<hr>
<h2 id='diversity.evaluate.core'>Diversity Indices</h2><span id='topic+diversity.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the following diversity indices and perform corresponding statistical
tests to compare the phenotypic diversity for qualitative traits between
entire collection (EC) and core set (CS). </p>
 <ul>
<li><p>Simpson's and
related indices </p>
 <ul>
<li><p>Simpson's Index (\(d\))
(Simpson 1949; Peet 1974)
</p>
</li>
<li><p>Simpson's Index of Diversity or Gini's Diversity Index or Gini-Simpson
Index or Nei's Diversity Index or Nei's Variation Index (\(D\))
(Gini 1912, 1912; Greenberg 1956; Berger and Parker 1970; Nei 1973; Peet 1974)
</p>
</li>
<li><p>Maximum Simpson's Index of Diversity or Maximum Nei's
Diversity/Variation Index (\(D_{max}\))
(Hennink and Zeven 1990) </p>
</li>
<li><p>Simpson's
Reciprocal Index or Hill's \(N_{2}\) (\(D_{R}\))
(Williams 1964; Hill 1973)
</p>
</li>
<li><p>Relative Simpson's Index of Diversity or Relative Nei's
Diversity/Variation Index (\(D'\))
(Hennink and Zeven 1990) </p>
</li></ul>

</li>
<li><p>Shannon-Weaver and related indices </p>
 <ul>
<li><p>Shannon or
Shannon-Weaver or Shannon-Weiner Diversity Index (\(H\))
(Shannon and Weaver 1949; Peet 1974)
</p>
</li>
<li><p>Maximum Shannon-Weaver Diversity Index (\(H_{max}\))
(Hennink and Zeven 1990) </p>
</li>
<li><p>Relative
Shannon-Weaver Diversity Index or Shannon Equitability Index (\(H'\))
(Hennink and Zeven 1990) </p>
</li></ul>
 </li>
<li><p>McIntosh
Diversity Index </p>
 <ul>
<li><p>McIntosh Diversity Index (\(D_{Mc}\))
(McIntosh 1967; Peet 1974) </p>
</li></ul>
 </li></ul>

<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>


<h3>Usage</h3>

<pre><code class='language-R'>diversity.evaluate.core(data, names, qualitative, selected, base = 2, R = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diversity.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="diversity.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="diversity.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="diversity.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
<tr><td><code id="diversity.evaluate.core_+3A_base">base</code></td>
<td>
<p>The logarithm base to be used for computation of Shannon-Weaver
Diversity Index (\(I\)). Default is 2.</p>
</td></tr>
<tr><td><code id="diversity.evaluate.core_+3A_r">R</code></td>
<td>
<p>The number of bootstrap replicates. Default is 1000.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three data frames as follows. </p>
<table>
<tr><td><code>simpson</code></td>
<td>

<dl>
<dt>Trait</dt><dd><p>The qualitative trait.</p>
</dd>
<dt>EC_No.Classes</dt><dd><p>The number of classes in the trait for EC.</p>
</dd>
<dt>CS_No.Classes</dt><dd><p>The number of classes in the trait for CS.</p>
</dd>
<dt>EC_d</dt><dd><p>The Simpson's Index (\(d\)) for EC.</p>
</dd>
<dt>EC_D</dt><dd><p>The Simpson's Index of Diversity (\(D\)) for EC.</p>
</dd>
<dt>EC_D.max</dt><dd><p>The Maximum Simpson's Index of Diversity
(\(D_{max}\)) for EC.</p>
</dd> <dt>EC_D.inv</dt><dd><p>The Simpson's Reciprocal
Index (\(D_{R}\)) for EC.</p>
</dd> <dt>EC_D.rel</dt><dd><p>The Relative
Reciprocal Index (\(D'\)) for EC.</p>
</dd> <dt>EC_d.V</dt><dd><p>The variance
of \(d\) for EC according to
(Simpson 1949).</p>
</dd>
<dt>EC_d.boot.V</dt><dd><p>The bootstrap variance of \(d\) for EC.</p>
</dd>
<dt>CS_d</dt><dd><p>The Simpson's Index (\(d\)) for CS.</p>
</dd>
<dt>CS_D</dt><dd><p>The Simpson's Index of Diversity (\(D\)) for CS.</p>
</dd>
<dt>CS_D.max</dt><dd><p>The Maximum Simpson's Index of Diversity
(\(D_{max}\)) for CS.</p>
</dd> <dt>CS_D.inv</dt><dd><p>The Simpson's Reciprocal
Index (\(D_{R}\)) for CS.</p>
</dd> <dt>CS_D.rel</dt><dd><p>The Relative
Reciprocal Index (\(D'\)) for CS.</p>
</dd> <dt>CS_d.V</dt><dd><p>The variance
of \(d\) for CS according to
(Simpson 1949).</p>
</dd>
<dt>CS_d.boot.V</dt><dd><p>The bootstrap variance of \(d\) for CS.</p>
</dd>
<dt>d.t.df</dt><dd><p>The degrees of freedom for t test.</p>
</dd>
<dt>d.t.stat</dt><dd><p>The t statistic.</p>
</dd> <dt>d.t.pvalue</dt><dd><p>The p
value for t test.</p>
</dd> <dt>d.t.significance</dt><dd><p>The  significance of t
test for t-test</p>
</dd> <dt>d.boot.z.df</dt><dd><p>The degrees of freedom for
bootstrap z score.</p>
</dd> <dt>d.boot.z.stat</dt><dd><p>The bootstrap z score.</p>
</dd>
<dt>d.boot.z.pvalue</dt><dd><p>The p value of z score.</p>
</dd>
<dt>d.boot.z.significance</dt><dd><p>The significance of z score.</p>
</dd></dl>
 </td></tr>
<tr><td><code>shannon</code></td>
<td>
 <dl>
<dt>Trait</dt><dd><p>The qualitative
trait.</p>
</dd> <dt>EC_No.Classes</dt><dd><p>The number of classes in the trait for
EC.</p>
</dd> <dt>CS_No.Classes</dt><dd><p>The number of classes in the trait for
CS.</p>
</dd> <dt>EC_I</dt><dd><p>The Shannon-Weaver Diversity Index (\(I\)) for
EC.</p>
</dd> <dt>EC_I.max</dt><dd><p>The Maximum Shannon-Weaver Diversity Index
(\(I_{max}\)) for EC.</p>
</dd> <dt>EC_I.rel</dt><dd><p>The Relative
Shannon-Weaver Diversity Index (\(I'\)) for EC.</p>
</dd>
<dt>EC_I.V</dt><dd><p>The variance of \(I\) for EC according to
(Hutcheson 1970).</p>
</dd>
<dt>EC_I.boot.V</dt><dd><p>The bootstrap variance of \(I\) for EC.</p>
</dd>
<dt>CS_I</dt><dd><p>The Shannon-Weaver Diversity Index (\(I\)) for CS.</p>
</dd>
<dt>CS_I.max</dt><dd><p>The Maximum Shannon-Weaver Diversity Index
(\(I_{max}\)) for CS.</p>
</dd> <dt>CS_I.rel</dt><dd><p>The Relative
Shannon-Weaver Diversity Index (\(I'\)) for CS.</p>
</dd>
<dt>CS_I.V</dt><dd><p>The variance of \(I\) for CS according to
(Hutcheson 1970).</p>
</dd>
<dt>CS_I.boot.V</dt><dd><p>The bootstrap variance of \(I\) for CS.</p>
</dd>
<dt>I.t.stat</dt><dd><p>The t statistic.</p>
</dd> <dt>I.t.df</dt><dd><p>The degrees
of freedom for t test.</p>
</dd> <dt>I.t.pvalue</dt><dd><p>The p value for t test.</p>
</dd>
<dt>I.t.significance</dt><dd><p>The  significance of t test for t-test</p>
</dd>
<dt>I.boot.z.df</dt><dd><p>The degrees of freedom for bootstrap z score.</p>
</dd>
<dt>I.boot.z.stat</dt><dd><p>The bootstrap z score.</p>
</dd>
<dt>I.boot.z.pvalue</dt><dd><p>The p value of z score.</p>
</dd>
<dt>I.boot.z.significance</dt><dd><p>The significance of z score.</p>
</dd></dl>
 </td></tr>
<tr><td><code>mcintosh</code></td>
<td>
 <dl>
<dt>EC_No.Classes</dt><dd><p>The number of
classes in the trait for EC.</p>
</dd> <dt>CS_No.Classes</dt><dd><p>The number of
classes in the trait for CS.</p>
</dd> <dt>EC_D.Mc</dt><dd><p>The McIntosh Index
(\(D_{Mc}\)) for EC.</p>
</dd> <dt>CS_D.Mc</dt><dd><p>The McIntosh Index
(\(D_{Mc}\)) for CS.</p>
</dd> <dt>M.boot.z.stat</dt><dd><p>The bootstrap z
score.</p>
</dd> <dt>M.boot.z.df</dt><dd><p>The degrees of freedom for bootstrap z
score.</p>
</dd> <dt>M.boot.z.pvalue</dt><dd><p>The p value of z score.</p>
</dd>
<dt>M.boot.z.significance</dt><dd><p>The significance of z score.</p>
</dd></dl>
 </td></tr>
</table>


<h3>Details</h3>

<p>The diversity indices and the corresponding statistical
tests implemented in <code>diversity.evaluate.core</code> are as follows.
</p>


<h4>Simpson's and related indices</h4>

<p>Simpson's index (\(d\))
which estimates the probability that two accessions randomly selected will
belong to the same phenotypic class of a trait, is computed as follows
(Simpson 1949; Peet 1974).
</p>
\[d = \sum_{i = 1}^{k}p_{i}^{2}\]
<p>Where, \(p_{i}\) denotes the proportion/fraction/frequency of
accessions in the \(i\)th phenotypic class for a trait and \(k\)
is the number of phenotypic classes for the trait.
</p>
<p>The value of \(d\) can range from 0 to 1 with 0 representing maximum
diversity and 1, no diversity.
</p>
<p>\(d\) is subtracted from 1 to give Simpson's index of diversity
(\(D\))
(Greenberg 1956; Berger and Parker 1970; Peet 1974; Hennink and Zeven 1990)
originally suggested by
Gini (1912, 1912)
and described in literature as Gini's diversity index or Gini-Simpson
index. It is the same as Nei's diversity index or Nei's variation index
(Nei 1973; Hennink and Zeven 1990).
Greater the value of \(D\), greater the diversity with a range from 0
to 1.
</p>
\[D = 1 - d\]
<p>The maximum value of \(D\), \(D_{max}\) occurs when accessions
are uniformly distributed across the phenotypic classes and is computed as
follows (Hennink and Zeven 1990).
</p>
\[D_{max} = 1 - \frac{1}{k}\]
<p>Reciprocal of \(d\) gives the Simpson's reciprocal index
(\(D_{R}\))
(Williams 1964; Hennink and Zeven 1990)
and can range from 1 to \(k\). This was also described in
Hill (1973) as (\(N_{2}\)).
</p>
\[D_{R} = \frac{1}{d}\]
<p>Relative Simpson's index of diversity or Relative Nei's diversity/variation
index (\(H'\)) (Hennink and Zeven 1990)
is defined as follows (Peet 1974).
</p>
\[D' = \frac{D}{D_{max}}\]
<p>Differences in Simpson's diversity index for qualitative traits of EC and
CS can be tested by a t-test using the associated variance estimate
described in Simpson (1949)
(Lyons and Hutcheson 1978).
</p>
<p>The t statistic is computed as follows.
</p>
\[t = \frac{d_{EC} - d_{CS}}{\sqrt{V_{d_{EC}} + V_{d_{CS}}}}\]
<p>Where, the variance of \(d\) (\(V_{d}\)) is,
</p>
\[V_{d} = \frac{4N(N-1)(N-2)\sum_{i=1}^{k}(p_{i})^{3} +
  2N(N-1)\sum_{i=1}^{k}(p_{i})^{2} - 2N(N-1)(2N-3) \left(
  \sum_{i=1}^{k}(p_{i})^{2} \right)^{2}}{[N(N-1)]^{2}}\]
<p>The associated degrees of freedom is computed as follows.
</p>
\[df = (k_{EC} - 1) + (k_{CS} - 1)\]
<p>Where, \(k_{EC}\) and \(k_{CS}\) are the number of phenotypic
classes in the trait for EC and CS respectively.
</p>



<h4>Shannon-Weaver and related indices</h4>

<p>An index of information
\(H\), was described by
Shannon and Weaver (1949) as follows.
</p>
\[H = -\sum_{i=1}^{k}p_{i} \log_{2}(p_{i})\]
<p>\(H\) is described as Shannon or Shannon-Weaver or Shannon-Weiner
diversity index in literature.
</p>
<p>Alternatively, \(H\) is also computed using natural logarithm instead
of logarithm to base 2.
</p>
\[H = -\sum_{i=1}^{k}p_{i} \ln(p_{i})\]
<p>The maximum value of \(H\) (\(H_{max}\)) is \(\ln(k)\). This
value occurs when each phenotypic class for a trait has the same proportion
of accessions.
</p>
\[H_{max} = \log_{2}(k)\;\; \textrm{OR} \;\; H_{max} = \ln(k)\]
<p>The relative Shannon-Weaver diversity index or Shannon equitability index
(\(H'\)) is the Shannon diversity index (\(I\)) divided by the
maximum diversity (\(H_{max}\)).
</p>
\[H' = \frac{H}{H_{max}}\]
<p>Differences in Shannon-Weaver diversity index for qualitative traits of EC
and CS can be tested by Hutcheson t-test
(Hutcheson 1970).
</p>
<p>The Hutcheson t statistic is computed as follows.
</p>
\[t = \frac{H_{EC} - H_{CS}}{\sqrt{V_{H_{EC}} + V_{H_{CS}}}}\]
<p>Where, the variance of \(H\) (\(V_{H}\)) is,
</p>
\[V_{H} = \frac{\sum_{i=1}^{k}n_{i}(\log_{2}{n_{i}})^{2}
  \frac{(\sum_{i=1}^{k}\log_{2}{n_{i}})^2}{N}}{N^{2}}\]
\[\textrm{OR}\]
\[V_{H} = \frac{\sum_{i=1}^{k}n_{i}(\ln{n_{i}})^{2}
  \frac{(\sum_{i=1}^{k}\ln{n_{i}})^2}{N}}{N^{2}}\]
<p>The associated degrees of freedom is approximated as follows.
</p>
\[df = \frac{(V_{H_{EC}} +
  V_{H_{CS}})^{2}}{\frac{V_{H_{EC}}^{2}}{N_{EC}} +
  \frac{V_{H_{CS}}^{2}}{N_{CS}}}\]



<h4>McIntosh Diversity Index</h4>

<p>A similar index of diversity was
described by McIntosh (1967) as
follows (\(D_{Mc}\)) (Peet 1974).
</p>
\[D_{Mc} = \frac{N - \sqrt{\sum_{i=1}^{k}n_{i}^2}}{N - \sqrt{N}}\]
<p>Where, \(n_{i}\) denotes the number of accessions in the \(i\)th
phenotypic class for a trait and \(N\) is the total number of
accessions so that \(p_{i} = {n_{i}}/{N}\).</p>



<h4>Testing for difference with bootstrapping</h4>

<p>Bootstrap statistics
are employed to test the difference between the Simpson, Shannon-Weaver and
McIntosh indices for  qualitative traits of EC and CS
(Solow 1993).
</p>
<p>If \(I_{EC}\) and \(I_{CS}\) are the diversity indices with the
original number of accessions, then random samples of the same size as the
original are repeatedly generated (with replacement) \(R\) times and
the corresponding diversity index is computed for each sample.
</p>
\[I_{EC}^{*} = \lbrace H_{EC_{1}}, H_{EC_{}}, \cdots, H_{EC_{R}}
  \rbrace\]
\[I_{CS}^{*} = \lbrace H_{CS_{1}}, H_{CS_{}}, \cdots, H_{CS_{R}}
  \rbrace\]
<p>Then the bootstrap null sample \(I_{0}\) is computed as follows.
</p>
\[\Delta^{*} = I_{EC}^{*} - I_{CS}^{*}\]
\[I_{0} = \Delta^{*} - \overline{\Delta^{*}}\]
<p>Where, \(\overline{\Delta^{*}}\) is the mean of \(\Delta^{*}\).
</p>
<p>Now the original difference in diversity indices (\(\Delta_{0} =
  I_{EC} - I_{CS}\)) is tested against mean of bootstrap null sample
(\(I_{0}\)) by a z test. The z score test statistic is computed as
follows.
</p>
\[z = \frac{\Delta_{0} - \overline{H_{0}}}{\sqrt{V_{H_{0}}}}\]
<p>Where,  \(\overline{H_{0}}\) and \(V_{H_{0}}\) are the mean and
variance of the bootstrap null sample \(H_{0}\).
</p>
<p>The corresponding degrees of freedom is estimated as follows.
</p>
\[df = (k_{EC} - 1) + (k_{CS} - 1)\]



<h3>References</h3>

<p>Berger WH, Parker FL (1970).
&ldquo;Diversity of planktonic foraminifera in deep-sea sediments.&rdquo;
<em>Science</em>, <b>168</b>(3937), 1345&ndash;1347.<br /><br /> Gini C (1912).
<em>Variabilita e Mutabilita. Contributo allo Studio delle Distribuzioni e delle Relazioni Statistiche. [Fasc. I.]</em>.
Tipogr. di P. Cuppini, Bologna.<br /><br /> Gini C (1912).
&ldquo;Variabilita e mutabilita.&rdquo;
In Pizetti E, Salvemini T (eds.), <em>Memorie di Metodologica Statistica</em>.
Liberia Eredi Virgilio Veschi, Roma, Italy.<br /><br /> Greenberg JH (1956).
&ldquo;The measurement of linguistic diversity.&rdquo;
<em>Language</em>, <b>32</b>(1), 109.<br /><br /> Hennink S, Zeven AC (1990).
&ldquo;The interpretation of Nei and Shannon-Weaver within population variation indices.&rdquo;
<em>Euphytica</em>, <b>51</b>(3), 235&ndash;240.<br /><br /> Hill MO (1973).
&ldquo;Diversity and evenness: A unifying notation and its consequences.&rdquo;
<em>Ecology</em>, <b>54</b>(2), 427&ndash;432.<br /><br /> Hutcheson K (1970).
&ldquo;A test for comparing diversities based on the Shannon formula.&rdquo;
<em>Journal of Theoretical Biology</em>, <b>29</b>(1), 151&ndash;154.<br /><br /> Lyons NI, Hutcheson K (1978).
&ldquo;C20. Comparing diversities: Gini's index.&rdquo;
<em>Journal of Statistical Computation and Simulation</em>, <b>8</b>(1), 75&ndash;78.<br /><br /> McIntosh RP (1967).
&ldquo;An index of diversity and the relation of certain concepts to diversity.&rdquo;
<em>Ecology</em>, <b>48</b>(3), 392&ndash;404.<br /><br /> Nei M (1973).
&ldquo;Analysis of gene diversity in subdivided populations.&rdquo;
<em>Proceedings of the National Academy of Sciences</em>, <b>70</b>(12), 3321&ndash;3323.<br /><br /> Peet RK (1974).
&ldquo;The measurement of species diversity.&rdquo;
<em>Annual Review of Ecology and Systematics</em>, <b>5</b>(1), 285&ndash;307.<br /><br /> Shannon CE, Weaver W (1949).
<em>The Mathematical Theory of Communication</em>,  number v. 2 in The Mathematical Theory of Communication.
University of Illinois Press.<br /><br /> Simpson EH (1949).
&ldquo;Measurement of diversity.&rdquo;
<em>Nature</em>, <b>163</b>(4148), 688&ndash;688.<br /><br /> Solow AR (1993).
&ldquo;A simple test for change in community structure.&rdquo;
<em>The Journal of Animal Ecology</em>, <b>62</b>(1), 191.<br /><br /> Williams CB (1964).
<em>Patterns in the Balance of Nature and Related Problems in Quantitative Ecology</em>.
Academic Press.
</p>


<h3>See Also</h3>

<p><code><a href="psych.html#topic+misc">shannon</a></code>, <code><a href="vegan.html#topic+diversity">diversity</a></code>,
<code><a href="boot.html#topic+boot">boot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))


diversity.evaluate.core(data = ec, names = "genotypes",
                        qualitative = qual, selected = core)


</code></pre>

<hr>
<h2 id='EvaluateCore-deprecated'>Deprecated functions in package <span class="pkg">EvaluateCore</span>.</h2><span id='topic+EvaluateCore-deprecated'></span><span id='topic+shannon.evaluate.core'></span>

<h3>Description</h3>

<p>The functions listed below are deprecated and will be defunct in
the near future. When possible, alternative functions with similar
functionality are also mentioned. Help pages for deprecated functions are
available at <code>help("&lt;function&gt;-deprecated")</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shannon.evaluate.core(data, names, qualitative, selected)
</code></pre>


<h3><code>shannon.evaluate.core</code></h3>

<p>For <code>shannon.evaluate.core</code>, use <code><a href="#topic+diversity.evaluate.core">diversity.evaluate.core</a></code>.
</p>

<hr>
<h2 id='freqdist.evaluate.core'>Frequency Distribution Histogram</h2><span id='topic+freqdist.evaluate.core'></span>

<h3>Description</h3>

<p>Plot stacked frequency distribution histogram to graphically compare the
probability distributions of traits between entire collection (EC) and core
set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freqdist.evaluate.core(
  data,
  names,
  quantitative,
  qualitative,
  selected,
  highlight = NULL,
  include.highlight = TRUE,
  highlight.se = NULL,
  highlight.col = "red"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freqdist.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_qualitative">qualitative</code></td>
<td>
<p>Name of columns with the qualitative traits as a character
vector.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_highlight">highlight</code></td>
<td>
<p>Individual names to be highlighted as a character vector.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_include.highlight">include.highlight</code></td>
<td>
<p>If <code>TRUE</code>, the highlighted individuals are
included in the frequency distribution histogram. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_highlight.se">highlight.se</code></td>
<td>
<p>Optional data frame of standard errors for the
individuals specified in <code>highlight</code>. It should have the same column
names as in <code>data</code>.</p>
</td></tr>
<tr><td><code id="freqdist.evaluate.core_+3A_highlight.col">highlight.col</code></td>
<td>
<p>The colour(s) to be used to highlighting individuals in
the plot as a character vector of the same length as <code>highlight</code>. Must
be valid colour values in R (named colours, hexadecimal representation,
index of colours [<code>1:8</code>] in default R <code>palette()</code> etc.).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the <code>ggplot</code> objects of stacked frequency
distribution histograms plots for each trait specified as
<code>quantitative</code> and <code>qualitative</code>.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+hist">hist</a></code>, <code><a href="ggplot2.html#topic+geom_histogram">geom_histogram</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))


freqdist.evaluate.core(data = ec, names = "genotypes",
                       quantitative = quant, qualitative = qual,
                       selected = core)

checks &lt;- c("TMe-1199", "TMe-1957", "TMe-3596", "TMe-3392")

freqdist.evaluate.core(data = ec, names = "genotypes",
                       quantitative = quant, qualitative = qual,
                       selected = core,
                       highlight = checks, highlight.col = "red")

quant.se &lt;- data.frame(genotypes = checks,
                       NMSR = c(0.107, 0.099, 0.106, 0.062),
                       TTRN = c(0.081, 0.072, 0.057, 0.049),
                       TFWSR = c(0.089, 0.031, 0.092, 0.097),
                       TTRW = c(0.064, 0.031, 0.071, 0.071),
                       TFWSS = c(0.106, 0.071, 0.121, 0.066),
                       TTSW = c(0.084, 0.045, 0.066, 0.054),
                       TTPW = c(0.098, 0.052, 0.111, 0.082),
                       AVPW = c(0.074, 0.038, 0.054, 0.061),
                       ARSR = c(0.104, 0.019, 0.204, 0.044),
                       SRDM = c(0.078, 0.138, 0.076, 0.079))

freqdist.evaluate.core(data = ec, names = "genotypes",
                       quantitative = quant,
                       selected = core,
                       highlight = checks, highlight.col = "red",
                       highlight.se = quant.se)


</code></pre>

<hr>
<h2 id='iqr.evaluate.core'>Interquartile Range</h2><span id='topic+iqr.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the Interquartile Range (IQR)
(Upton and Cook 1996) to compare quantitative traits
of the entire collection (EC) and core set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iqr.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iqr.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="iqr.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="iqr.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="iqr.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the IQR values of the EC and CS for the traits
specified as <code>quantitative</code>.
</p>


<h3>References</h3>

<p>Upton G, Cook I (1996).
&ldquo;General summary statistics.&rdquo;
In <em>Understanding statistics</em>.
Oxford University Press.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+IQR">IQR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

iqr.evaluate.core(data = ec, names = "genotypes",
                  quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='levene.evaluate.core'>Levene's Test</h2><span id='topic+levene.evaluate.core'></span>

<h3>Description</h3>

<p>Test for  of variances of the entire collection (EC) and core set (CS) for
quantitative traits by Levene's test
(Levene 1960). <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>levene.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="levene.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="levene.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="levene.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="levene.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following columns </p>
<table>
<tr><td><code>Trait</code></td>
<td>
<p>The quantitative
trait.</p>
</td></tr> <tr><td><code>EC_V</code></td>
<td>
<p>The variance of the EC.</p>
</td></tr> <tr><td><code>CS_V</code></td>
<td>
<p>The variance of
the CS.</p>
</td></tr> <tr><td><code>EC_CV</code></td>
<td>
<p>The coefficient of variance of the EC.</p>
</td></tr>
<tr><td><code>CS_CV</code></td>
<td>
<p>The coefficient of variance of the CS.</p>
</td></tr>
<tr><td><code>Levene_Fvalue</code></td>
<td>
<p>The test statistic.</p>
</td></tr> <tr><td><code>Levene_pvalue</code></td>
<td>
<p>The p value
for the test statistic.</p>
</td></tr> <tr><td><code>Levene_significance</code></td>
<td>
<p>The significance of the
test statistic (*: p \(\leq\) 0.01; **: p \(\leq\) 0.05; ns: p
\( > \) 0.05).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Levene H (1960).
&ldquo;Robust tests for equality of variances.&rdquo;
In Olkin I, Ghurye SG, Hoeffding W, Madow WG, Mann HB (eds.), <em>Contribution to Probability and Statistics: Essays in Honor of Harold Hotelling</em>, 278&ndash;292.
Stanford University Press, Palo Alto, CA.
</p>


<h3>See Also</h3>

<p><code><a href="car.html#topic+leveneTest">leveneTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

levene.evaluate.core(data = ec, names = "genotypes",
                     quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='pca.evaluate.core'>Principal Component Analysis</h2><span id='topic+pca.evaluate.core'></span>

<h3>Description</h3>

<p>Compute Principal Component Analysis Statistics
(Mardia et al. 1979) to compare the
probability distributions of quantitative traits between entire collection
(EC) and core set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca.evaluate.core(
  data,
  names,
  quantitative,
  selected,
  center = TRUE,
  scale = TRUE,
  npc.plot = 6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_center">center</code></td>
<td>
<p>either a logical value or numeric-alike vector of length
equal to the number of columns of <code>x</code>, where
&lsquo;numeric-alike&rsquo; means that <code><a href="base.html#topic+as.numeric">as.numeric</a>(.)</code> will
be applied successfully if <code><a href="base.html#topic+is.numeric">is.numeric</a>(.)</code> is not true.</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_scale">scale</code></td>
<td>
<p>either a logical value or a numeric-alike vector of length
equal to the number of columns of <code>x</code>.</p>
</td></tr>
<tr><td><code id="pca.evaluate.core_+3A_npc.plot">npc.plot</code></td>
<td>
<p>The number of principal components for which eigen values are
to be plotted. The default value is 6.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components. </p>
<table>
<tr><td><code>EC PC Importance</code></td>
<td>
<p>A data
frame of importance of principal components for EC</p>
</td></tr> <tr><td><code>EC PC Loadings</code></td>
<td>
<p>A
data frame with eigen vectors of principal components for EC</p>
</td></tr> <tr><td><code>CS PC
  Importance</code></td>
<td>
<p>A data frame of importance of principal components for CS</p>
</td></tr>
<tr><td><code>CS PC Loadings</code></td>
<td>
<p>A data frame with eigen vectors of principal
components for CS</p>
</td></tr> <tr><td><code>Scree Plot</code></td>
<td>
<p>The scree plot of principal components
for EC and CS as a <code>ggplot</code> object.</p>
</td></tr> <tr><td><code>PC Loadings Plot</code></td>
<td>
<p>A plot of
the eigen vector values of principal components for EC and CS as specified
by <code>npc.plot</code> as a <code>ggplot2</code> object.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mardia KV, Kent JT, Bibby JM (1979).
<em>Multivariate analysis</em>.
Academic Press, London; New York.
ISBN 0-12-471250-9 978-0-12-471250-8 0-12-471252-5 978-0-12-471252-2.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+prcomp">prcomp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

pca.evaluate.core(data = ec, names = "genotypes",
                  quantitative = quant, selected = core,
                  center = TRUE, scale = TRUE, npc.plot = 4)

</code></pre>

<hr>
<h2 id='pdfdist.evaluate.core'>Distance Between Probability Distributions</h2><span id='topic+pdfdist.evaluate.core'></span>

<h3>Description</h3>

<p>Compute Kullback-Leibler
(Kullback and Leibler 1951), Kolmogorov-Smirnov
(Kolmogorov 1933; Smirnov 1948) and
Anderson-Darling distances
(Anderson and Darling 1952) between the probability
distributions of collection (EC) and core set (CS) for quantitative traits.
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdfdist.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pdfdist.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="pdfdist.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="pdfdist.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="pdfdist.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following columns. </p>
<table>
<tr><td><code>Trait</code></td>
<td>
<p>The
quantitative trait.</p>
</td></tr> <tr><td><code>KL_Distance</code></td>
<td>
<p>The Kullback-Leibler distance
(Kullback and Leibler 1951) between EC and CS.</p>
</td></tr>
<tr><td><code>KS_Distance</code></td>
<td>
<p>The Kolmogorov-Smirnov distance
(Kolmogorov 1933; Smirnov 1948) between
EC and CS.</p>
</td></tr> <tr><td><code>KS_pvalue</code></td>
<td>
<p>The p value of the Kolmogorov-Smirnov
distance.</p>
</td></tr> <tr><td><code>AD_Distance</code></td>
<td>
<p>Anderson-Darling distance
(Anderson and Darling 1952) between EC and CS.</p>
</td></tr>
<tr><td><code>AD_pvalue</code></td>
<td>
<p>The p value of the Anderson-Darling distance.</p>
</td></tr>
<tr><td><code>KS_significance</code></td>
<td>
<p>The significance of the Kolmogorov-Smirnov distance
(*: p \(\leq\) 0.01; **: p \(\leq\) 0.05; ns: p \(>\)
0.05).</p>
</td></tr> <tr><td><code>AD_pvalue</code></td>
<td>
<p>The significance of the Anderson-Darling distance
(*: p \(\leq\) 0.01; **: p \(\leq\) 0.05; ns: p \(>\)
0.05).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="entropy.html#topic+KL.plugin">KL.plugin</a></code>, <code><a href="stats.html#topic+ks.test">ks.test</a></code>,
<code><a href="kSamples.html#topic+ad.test">ad.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

pdfdist.evaluate.core(data = ec, names = "genotypes",
                      quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='percentdiff.evaluate.core'>Percentage Difference of Means and Variances</h2><span id='topic+percentdiff.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the following differences between the entire collection (EC) and core
set (CS). </p>
 <ul>
<li><p> Percentage of significant differences of mean
(\(MD\%_{Hu}\))
(Hu et al. 2000) </p>
</li>
<li><p> Percentage of significant
differences of variance (\(VD\%_{Hu}\))
(Hu et al. 2000) </p>
</li>
<li><p> Average of absolute
differences between means
(\(MD\%_{Kim}\))
(Kim et al. 2007) </p>
</li>
<li><p> Average of absolute
differences between variances
(\(VD\%_{Kim}\))
(Kim et al. 2007) </p>
</li>
<li><p> Percentage difference
between the mean squared Euclidean distance among accessions
(\(\overline{d}D\%\))
(Studnicki et al. 2013) </p>
</li></ul>
 <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>


<h3>Usage</h3>

<pre><code class='language-R'>percentdiff.evaluate.core(data, names, quantitative, selected, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="percentdiff.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="percentdiff.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="percentdiff.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="percentdiff.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
<tr><td><code id="percentdiff.evaluate.core_+3A_alpha">alpha</code></td>
<td>
<p>Type I error probability (Significance level) of difference.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The differences are computed as follows.
</p>
\[MD\%_{Hu} = \left ( \frac{S_{t}}{n} \right ) \times 100\]
<p>Where, \(S_{t}\) is the number of traits with a significant difference
between the means of the EC and the CS and \(n\) is the total number of
traits. A representative core should have
\(MD\%_{Hu}\) &lt; 20 % and \(CR\) &gt; 80
% (Hu et al. 2000).
</p>
\[VD\%_{Hu} = \left ( \frac{S_{F}}{n} \right ) \times 100\]
<p>Where, \(S_{F}\) is the number of traits with a significant difference
between the variances of the EC and the CS and \(n\) is the total number
of traits. Larger \(VD\%_{Hu}\) value
indicates a more diverse core set.
</p>
\[MD\%_{Kim} = \left ( \frac{1}{n}\sum_{i=1}^{n} \frac{\left |
M_{EC_{i}}-M_{CS_{i}} \right |}{M_{CS_{i}}} \right ) \times 100\]
<p>Where, \(M_{EC_{i}}\) is the mean of the EC for the \(i\)th trait,
\(M_{CS_{i}}\) is the mean of the CS for the \(i\)th trait and
\(n\) is the total number of traits.
</p>
\[VD\%_{Kim} = \left ( \frac{1}{n}\sum_{i=1}^{n} \frac{\left |
V_{EC_{i}}-V_{CS_{i}} \right |}{V_{CS_{i}}} \right ) \times 100\]
<p>Where, \(V_{EC_{i}}\) is the variance of the EC for the \(i\)th
trait, \(V_{CS_{i}}\) is the variance of the CS for the \(i\)th
trait and \(n\) is the total number of traits.
</p>
\[\overline{d}D\% =
\frac{\overline{d}_{CS}-\overline{d}_{EC}}{\overline{d}_{EC}} \times
100\]
<p>Where, \(\overline{d}_{CS}\) is the mean squared Euclidean distance
among accessions in the CS and \(\overline{d}_{EC}\) is the mean squared
Euclidean distance among accessions in the EC.
</p>


<h3>Value</h3>

<p>A data frame with the values of
\(MD\%_{Hu}\),
\(VD\%_{Hu}\),
\(MD\%_{Kim}\),
\(VD\%_{Kim}\) and
\(\overline{d}D\%\).
</p>


<h3>References</h3>

<p>Hu J, Zhu J, Xu HM (2000).
&ldquo;Methods of constructing core collections by stepwise clustering with three sampling strategies based on the genotypic values of crops.&rdquo;
<em>Theoretical and Applied Genetics</em>, <b>101</b>(1), 264&ndash;268.<br /><br /> Kim K, Chung H, Cho G, Ma K, Chandrabalan D, Gwag J, Kim T, Cho E, Park Y (2007).
&ldquo;PowerCore: A program applying the advanced M strategy with a heuristic search for establishing core sets.&rdquo;
<em>Bioinformatics</em>, <b>23</b>(16), 2155&ndash;2162.<br /><br /> Studnicki M, Madry W, Schmidt J (2013).
&ldquo;Comparing the efficiency of sampling strategies to establish a representative in the phenotypic-based genetic diversity core collection of orchardgrass (<em>Dactylis glomerata</em> L.).&rdquo;
<em>Czech Journal of Genetics and Plant Breeding</em>, <b>49</b>(1), 36&ndash;47.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snk.evaluate.core">snk.evaluate.core</a></code>,
<code><a href="#topic+snk.evaluate.core">snk.evaluate.core</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

percentdiff.evaluate.core(data = ec, names = "genotypes",
                          quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='qq.evaluate.core'>Quantile-Quantile Plots</h2><span id='topic+qq.evaluate.core'></span>

<h3>Description</h3>

<p>Plot Quantile-Quantile (QQ) plots
(Wilk and Gnanadesikan 1968) to graphically compare the
probability distributions of quantitative traits between entire collection
(EC) and core set (CS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qq.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qq.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="qq.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="qq.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="qq.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the <code>ggplot</code> objects of QQ plots of CS vs EC for
each trait specified as <code>quantitative</code>.
</p>


<h3>References</h3>

<p>Wilk MB, Gnanadesikan R (1968).
&ldquo;Probability plotting methods for the analysis for the analysis of data.&rdquo;
<em>Biometrika</em>, <b>55</b>(1), 1&ndash;17.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+qqnorm">qqplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

qq.evaluate.core(data = ec, names = "genotypes",
                 quantitative = quant, selected = core)


</code></pre>

<hr>
<h2 id='shannon.evaluate.core-deprecated'>Shannon-Weaver Diversity Index</h2><span id='topic+shannon.evaluate.core-deprecated'></span>

<h3>Description</h3>

<p>Compute the Shannon-Weaver Diversity Index (\(H'\)), Maximum diversity
(\(H'_{max}\)) and Shannon Equitability Index (\(E_{H}\))
(Shannon and Weaver 1949) to compare the
phenotypic diversity for qualitative traits between entire collection (EC)
and core set (CS). <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shannon.evaluate.core(data, names, qualitative, selected)
</code></pre>


<h3>Details</h3>

<p>Shannon-Weaver Diversity Index (\(H'\)) is computed as follows.
</p>
\[H' = -\sum_{i=1}^{k}p_{i} \ln(p_{i})\]
<p>Where \(p_{i}\) denotes the proportion in the group \(k\).
</p>
<p>The maximum value of the index (\(H'_{max}\)) is \(\ln(k)\). This
value occurs when each group has the same frequency.
</p>
<p>The Shannon equitability index (\(E_{H}\)) is the Shannon diversity
index divided by the maximum diversity.
</p>
\[E_{H} = \frac{H'}{\ln{(k)}}\]


<h3>Value</h3>

<p>A data frame with the following columns. </p>
<table>
<tr><td><code>Trait</code></td>
<td>
<p>The qualitative
trait.</p>
</td></tr> <tr><td><code>EC_H</code></td>
<td>
<p>The Shannon-Weaver Diversity Index (\(H'\)) for
EC.</p>
</td></tr> <tr><td><code>EC_H</code></td>
<td>
<p>The Shannon-Weaver Diversity Index (\(H'\)) for CS.</p>
</td></tr>
<tr><td><code>EC_Hmax</code></td>
<td>
<p>The Maximum diversity value (\(H'_{max}\)) for EC.</p>
</td></tr>
<tr><td><code>CS_Hmax</code></td>
<td>
<p>The Maximum diversity value (\(H'_{max}\)) for CS.</p>
</td></tr>
<tr><td><code>EC_EH</code></td>
<td>
<p>The Shannon Equitability Index (\(E_{H}\)) for EC.</p>
</td></tr>
<tr><td><code>CS_EH</code></td>
<td>
<p>The Shannon Equitability Index (\(E_{H}\)) for CS.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Shannon CE, Weaver W (1949).
<em>The Mathematical Theory of Communication</em>,  number v. 2 in The Mathematical Theory of Communication.
University of Illinois Press.
</p>


<h3>See Also</h3>

<p><code><a href="psych.html#topic+misc">shannon</a></code>
</p>
<p><code><a href="#topic+EvaluateCore-deprecated">EvaluateCore-deprecated</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

shannon.evaluate.core(data = ec, names = "genotypes",
                      qualitative = qual, selected = core)

</code></pre>

<hr>
<h2 id='signtest.evaluate.core'>Sign Test</h2><span id='topic+signtest.evaluate.core'></span>

<h3>Description</h3>

<p>Test difference between means and variances of entire collection (EC) and
core set (CS) for quantitative traits by Sign test (\(+\) versus
\(-\))
(Basigalup et al. 1995; Tai and Miller 2001).
<script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>signtest.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="signtest.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="signtest.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="signtest.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="signtest.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The test statistic for Sign test (\(\chi^{2}\)) is computed as follows.
</p>
\[\chi^{2} = \frac{(N_{1}-N_{2})^{2}}{N_{1}+N_{2}}\]
<p>Where, where \(N_{1}\) is the number of variables for which the mean or
variance of the CS is greater than the mean or variance of the EC (number of
\(+\) signs); \(N_{2}\) is the number of variables for which the
mean or variance of the CS is less than the mean or variance of the EC
(number of \(-\) signs). The value of \(\chi^{2}\) is compared with
a Chi-square distribution with 1 degree of freedom.
</p>


<h3>Value</h3>

<p>A data frame with the following components. </p>
<table>
<tr><td><code>Comparison</code></td>
<td>
<p>The
comparison measure.</p>
</td></tr> <tr><td><code>ChiSq</code></td>
<td>
<p>The test statistic (\(\chi^{2}\)).</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p value for the test statistic.</p>
</td></tr> <tr><td><code>significance</code></td>
<td>
<p>The
significance of the test statistic (*: p \(\leq\) 0.01; **: p
\(\leq\) 0.05; ns: p \( > \) 0.05).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Basigalup DH, Barnes DK, Stucker RE (1995).
&ldquo;Development of a core collection for perennial <em>Medicago</em> plant introductions.&rdquo;
<em>Crop Science</em>, <b>35</b>(4), 1163&ndash;1168.<br /><br /> Tai PYP, Miller JD (2001).
&ldquo;A Core Collection for <em>Saccharum spontaneum</em> L. from the World Collection of Sugarcane.&rdquo;
<em>Crop Science</em>, <b>41</b>(3), 879&ndash;885.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

signtest.evaluate.core(data = ec, names = "genotypes",
                       quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='snk.evaluate.core'>Student-Newman-Keuls Test</h2><span id='topic+snk.evaluate.core'></span>

<h3>Description</h3>

<p>Test difference between means of entire collection (EC) and core set (CS) for
quantitative traits by Newman-Keuls or Student-Newman-Keuls test
(Newman 1939; Keuls 1952).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snk.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snk.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="snk.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="snk.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="snk.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following components. </p>
<table>
<tr><td><code>Trait</code></td>
<td>
<p>The
quantitative trait.</p>
</td></tr> <tr><td><code>EC_Min</code></td>
<td>
<p>The minimum value of the trait in EC.</p>
</td></tr>
<tr><td><code>EC_Max</code></td>
<td>
<p>The maximum value of the trait in EC.</p>
</td></tr> <tr><td><code>EC_Mean</code></td>
<td>
<p>The
mean value of the trait in EC.</p>
</td></tr> <tr><td><code>EC_SE</code></td>
<td>
<p>The standard error of the
trait in EC.</p>
</td></tr> <tr><td><code>CS_Min</code></td>
<td>
<p>The minimum value of the trait in CS.</p>
</td></tr>
<tr><td><code>CS_Max</code></td>
<td>
<p>The maximum value of the trait in CS.</p>
</td></tr> <tr><td><code>CS_Mean</code></td>
<td>
<p>The
mean value of the trait in CS.</p>
</td></tr> <tr><td><code>CS_SE</code></td>
<td>
<p>The standard error of the
trait in CS.</p>
</td></tr> <tr><td><code>SNK_pvalue</code></td>
<td>
<p>The p value of the Student-Newman-Keuls
test for equality of means of EC and CS.</p>
</td></tr> <tr><td><code>SNK_significance</code></td>
<td>
<p>The
significance of the Student-Newman-Keuls test for equality of means of EC
and CS.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Keuls M (1952).
&ldquo;The use of the ,,studentized range&quot; in connection with an analysis of variance.&rdquo;
<em>Euphytica</em>, <b>1</b>(2), 112&ndash;122.<br /><br /> Newman D (1939).
&ldquo;The distribution of range in samples from a normal population, expressed in terms of an independent estimate of standard deviation.&rdquo;
<em>Biometrika</em>, <b>31</b>(1-2), 20&ndash;30.
</p>


<h3>See Also</h3>

<p><code><a href="agricolae.html#topic+SNK.test">SNK.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

snk.evaluate.core(data = ec, names = "genotypes",
                  quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='ttest.evaluate.core'>Student's t Test</h2><span id='topic+ttest.evaluate.core'></span>

<h3>Description</h3>

<p>Test difference between means of entire collection (EC) and core set (CS) for
quantitative traits by Student's t test
(Student 1908).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ttest.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ttest.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="ttest.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="ttest.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="ttest.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Trait</code></td>
<td>
<p>The
quantitative trait.</p>
</td></tr> <tr><td><code>EC_Min</code></td>
<td>
<p>The minimum value of the trait in EC.</p>
</td></tr>
<tr><td><code>EC_Max</code></td>
<td>
<p>The maximum value of the trait in EC.</p>
</td></tr> <tr><td><code>EC_Mean</code></td>
<td>
<p>The
mean value of the trait in EC.</p>
</td></tr> <tr><td><code>EC_SE</code></td>
<td>
<p>The standard error of the
trait in EC.</p>
</td></tr> <tr><td><code>CS_Min</code></td>
<td>
<p>The minimum value of the trait in CS.</p>
</td></tr>
<tr><td><code>CS_Max</code></td>
<td>
<p>The maximum value of the trait in CS.</p>
</td></tr> <tr><td><code>CS_Mean</code></td>
<td>
<p>The
mean value of the trait in CS.</p>
</td></tr> <tr><td><code>CS_SE</code></td>
<td>
<p>The standard error of the
trait in CS.</p>
</td></tr> <tr><td><code>ttest_pvalue</code></td>
<td>
<p>The p value of the Student's t
test for equality of means of EC and CS.</p>
</td></tr> <tr><td><code>ttest_significance</code></td>
<td>
<p>The
significance of the Student's t test for equality of means of EC
and CS.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Student (1908).
&ldquo;The probable error of a mean.&rdquo;
<em>Biometrika</em>, <b>6</b>(1), 1&ndash;25.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">t.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

ttest.evaluate.core(data = ec, names = "genotypes",
                    quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='vr.evaluate.core'>Variable Rate of Coefficient of Variation</h2><span id='topic+vr.evaluate.core'></span>

<h3>Description</h3>

<p>Compute the Variable Rate of Coefficient of Variation (\(VR\))
(Hu et al. 2000) to compare quantitative traits of
the entire collection (EC) and core set (CS). <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vr.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vr.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="vr.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="vr.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="vr.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Variable Rate of Coefficient of Variation (\(VR\)) is computed as
follows.
</p>
\[VR = \left ( \frac{1}{n} \sum_{i=1}^{n}
\frac{CV_{CS_{i}}}{CV_{EC_{i}}} \right ) \times 100\]
<p>Where, \(CV_{CS_{i}}\) is the coefficients of variation for the
\(i\)th trait in the CS, \(CV_{EC_{i}}\) is the coefficients of
variation for the \(i\)th trait in the EC and \(n\) is the total
number of traits
</p>


<h3>Value</h3>

<p>The \(VR\) value.
</p>


<h3>References</h3>

<p>Hu J, Zhu J, Xu HM (2000).
&ldquo;Methods of constructing core collections by stepwise clustering with three sampling strategies based on the genotypic values of crops.&rdquo;
<em>Theoretical and Applied Genetics</em>, <b>101</b>(1), 264&ndash;268.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

vr.evaluate.core(data = ec, names = "genotypes",
                 quantitative = quant, selected = core)

</code></pre>

<hr>
<h2 id='wilcox.evaluate.core'>Wilcoxon Rank Sum Test</h2><span id='topic+wilcox.evaluate.core'></span>

<h3>Description</h3>

<p>Compare the medians of quantitative traits between entire collection (EC) and
core set (CS) by Wilcoxon rank sum test or Mann-Whitney-Wilcoxon test or
Mann-Whitney U test
(Wilcoxon 1945; Mann and Whitney 1947).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcox.evaluate.core(data, names, quantitative, selected)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wilcox.evaluate.core_+3A_data">data</code></td>
<td>
<p>The data as a data frame object. The data frame should possess
one row per individual and columns with the individual names and multiple
trait/character data.</p>
</td></tr>
<tr><td><code id="wilcox.evaluate.core_+3A_names">names</code></td>
<td>
<p>Name of column with the individual names as a character string</p>
</td></tr>
<tr><td><code id="wilcox.evaluate.core_+3A_quantitative">quantitative</code></td>
<td>
<p>Name of columns with the quantitative traits as a
character vector.</p>
</td></tr>
<tr><td><code id="wilcox.evaluate.core_+3A_selected">selected</code></td>
<td>
<p>Character vector with the names of individuals selected in
core collection and present in the <code>names</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Trait</code></td>
<td>
<p>The quantitative trait.</p>
</td></tr> <tr><td><code>EC_Med</code></td>
<td>
<p>The median value
of the trait in EC.</p>
</td></tr> <tr><td><code>CS_Med</code></td>
<td>
<p>The median value of the trait in CS.</p>
</td></tr>
<tr><td><code>Wilcox_pvalue</code></td>
<td>
<p>The p value of the Wilcoxon test for equality of
medians of EC and CS.</p>
</td></tr> <tr><td><code>Wilcox_significance</code></td>
<td>
<p>The significance of the
Wilcoxon test for equality of medians of EC and CS.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mann HB, Whitney DR (1947).
&ldquo;On a test of whether one of two random variables is stochastically larger than the other.&rdquo;
<em>The Annals of Mathematical Statistics</em>, <b>18</b>(1), 50&ndash;60.<br /><br /> Wilcoxon F (1945).
&ldquo;Individual comparisons by ranking methods.&rdquo;
<em>Biometrics Bulletin</em>, <b>1</b>(6), 80.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("cassava_CC")
data("cassava_EC")

ec &lt;- cbind(genotypes = rownames(cassava_EC), cassava_EC)
ec$genotypes &lt;- as.character(ec$genotypes)
rownames(ec) &lt;- NULL

core &lt;- rownames(cassava_CC)

quant &lt;- c("NMSR", "TTRN", "TFWSR", "TTRW", "TFWSS", "TTSW", "TTPW", "AVPW",
           "ARSR", "SRDM")
qual &lt;- c("CUAL", "LNGS", "PTLC", "DSTA", "LFRT", "LBTEF", "CBTR", "NMLB",
          "ANGB", "CUAL9M", "LVC9M", "TNPR9M", "PL9M", "STRP", "STRC",
          "PSTR")

ec[, qual] &lt;- lapply(ec[, qual],
                     function(x) factor(as.factor(x)))

wilcox.evaluate.core(data = ec, names = "genotypes",
                     quantitative = quant, selected = core)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
