<!DOCTYPE html><html lang="en"><head><title>Help for package evian</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {evian}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjustModel'><p>Genotype coding adjusement</p></a></li>
<li><a href='#calculateEvianMLE'><p>Profile likelihood calculation using regression models</p></a></li>
<li><a href='#calculateGLR'><p>Profile likelihood calculation using regression models</p></a></li>
<li><a href='#densityPlot'><p>Plot profile likelihood density for a single SNP.</p></a></li>
<li><a href='#evian'><p>Evidential analysis for genetic data using regression models</p></a></li>
<li><a href='#evian_binary_bim'><p>Example map data for evian_binary_raw.</p></a></li>
<li><a href='#evian_binary_raw'><p>Example dataset with a binary outcome.</p></a></li>
<li><a href='#evian_linear_bim'><p>Example map data for evian_linear_raw.</p></a></li>
<li><a href='#evian_linear_raw'><p>Example dataset with a quantative outcome.</p></a></li>
<li><a href='#expandBound'><p>A recursive function that expands the grid search for MLE.</p></a></li>
<li><a href='#getGridBound'><p>Obtain the range of the grid where MLE will be searched at</p></a></li>
<li><a href='#glr'><p>Generalized Likelihood Ratio test</p></a></li>
<li><a href='#glr_plot'><p>Plot methods for multiple glr result in a genomic region.</p></a></li>
<li><a href='#multiLine_plot'><p>Plot methods for multiple likelihood intervals in a genomic region.</p></a></li>
<li><a href='#robust_forCluster'><p>Robust adjustment function</p></a></li>
<li><a href='#subsetData'><p>interior subsetting function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Evidential Analysis of Genetic Association Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.5), ProfileLikelihood, sandwich, foreach, doParallel</td>
</tr>
<tr>
<td>Author:</td>
<td>Dr. Lisa J Strug &lt;lisa.strug@utoronto.ca&gt;; 
        Dr. Zeynep Baskurt &lt;zeynep.baskurt@sickkids.ca&gt;; 
        Bowei Xiao &lt;bowei.xiao91@gmail.com&gt;; 
        Ted Chiang &lt;theodorechiang@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiafen Gong &lt;jiafen.gong@sickkids.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Evidential regression analysis for dichotomous and quantitative outcome data. The following references described the methods in this package:
        Strug, L. J., Hodge, S. E., Chiang, T., Pal, D. K., Corey, P. N., &amp; Rohde, C. (2010) &lt;<a href="https://doi.org/10.1038%2Fejhg.2010.47">doi:10.1038/ejhg.2010.47</a>&gt;.
        Strug, L. J., &amp; Hodge, S. E. (2006) &lt;<a href="https://doi.org/10.1159%2F000094709">doi:10.1159/000094709</a>&gt;.
        Royall, R. (1997) &lt;ISBN:0-412-04411-0&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-23 14:38:58 UTC; bowei</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-23 16:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjustModel'>Genotype coding adjusement</h2><span id='topic+adjustModel'></span>

<h3>Description</h3>

<p>This is a helper function that adjusts the genotye coding scheme based on the genetic model specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  adjustModel(data_nomiss,model)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjustModel_+3A_data_nomiss">data_nomiss</code></td>
<td>
<p>a data frame that contains the phenotype, covariates and genotype columns. Genotypes need to be coded as 0/1/2.</p>
</td></tr>
<tr><td><code id="adjustModel_+3A_model">model</code></td>
<td>
<p>The model specified. Must be a string from one of the following: <code>additive</code>, <code>dominant</code>, <code>recessive</code>, or <code>overdominance</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>adjustModel</code> is an interior function that adjusts the genotype coding based on the genetic model specified. The coding scheme for different genetic models can be found in <code><a href="#topic+calculateEvianMLE">calculateEvianMLE</a></code>.
</p>


<h3>Value</h3>

<p>This function returns the data frame with the same columns but changed genotype coding based on the genetic model specified.
</p>

<hr>
<h2 id='calculateEvianMLE'>Profile likelihood calculation using regression models</h2><span id='topic+calculateEvianMLE'></span>

<h3>Description</h3>

<p>This is the function that calculates profileLikelihood for a single SNP. The main function <code><a href="#topic+evian">evian</a></code> calls this function repeatedly to obtain results for multiple SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  calculateEvianMLE(snp, formula_tofit, model, data, bim, lolim, hilim,
                    m, bse, k, robust, family, plinkCC)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculateEvianMLE_+3A_snp">snp</code></td>
<td>
<p>a string specifying the SNP of interests to be calculated.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_formula_tofit">formula_tofit</code></td>
<td>
<p>a formula object of the genetic model. The model should be formatted as <code>y~nuisance parameters</code>. The parameter of interest should not be included here.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_model">model</code></td>
<td>
<p>a string specifying the mode of inheritance parameterization: <br /> <code>additive, dominant, recessive, or overdominance</code>. See details.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_data">data</code></td>
<td>
<p>data frame; read from the argument <code>data</code> in the main function <code><a href="#topic+evian">evian</a></code>. It should contain the SNP ID specified in the <code>snp</code> argument as a column name.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_bim">bim</code></td>
<td>
<p>data frame; read from from the argument <code>bim</code> in the main function <code><a href="#topic+evian">evian</a></code>. Provides allele information (base pair, effect/reference alleles) for the SNP of interest.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_lolim">lolim</code></td>
<td>
<p>numeric; the lower limit for the grid or the minimum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood function. </p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_hilim">hilim</code></td>
<td>
<p>numeric; the upper limit for the grid or the maximum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood funciton. </p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_m">m</code></td>
<td>
<p>numeric; the density of the grid at which to compute the standardized likelihood function. A beta grid is defined as the grid of values for the SNP parameter used to evaluate the likelihood function.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_bse">bse</code></td>
<td>
<p>numeric; the number of beta standard errors to utilize in constraining the beta grid limits. Beta grid is evaluated at <code class="reqn">\beta</code> +/- <code>bse</code>*s.e.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_k">k</code></td>
<td>
<p>numeric or numeric vector; The strength of evidence criterion k. Reads from the input of <code>kcutoff</code> from the main <code><a href="#topic+evian">evian</a></code> function</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_robust">robust</code></td>
<td>
<p>logical; if <code>TRUE</code>, then a robust adjustment is applied to the likelihood function to account for the cluster nature in the data. See <a href="#topic+robust_forCluster">robust_forCluster</a>.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_family">family</code></td>
<td>
<p>the link function for <code>glm</code>.</p>
</td></tr>
<tr><td><code id="calculateEvianMLE_+3A_plinkcc">plinkCC</code></td>
<td>
<p>A boolean type that specifies how case/control are coded. case/control were coded 1/0 if it is FALSE, and were coded 2/1 if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>calculateEvianMLE</code> calculates the profile likelihood for a single SNP. A proper grid range is first established for <code class="reqn">\beta</code> then the standardized profile likelihood is evaluated at each of the <code>m</code> cuts uniformly spread across the grid. Based on the standardized profile likelihood, the MLE for <code class="reqn">\beta</code> is computed as well as the likelihood intervals for each value of <code>k</code> provided. <br />
For different genetic models, their coding schemes are shown as below:
</p>
<pre>
   Additive
 AA  0
 AB  1
 BB  2

   Dominant
 AA  0
 AB  1
 BB  1

   Recessive
 AA  0
 AB  0
 BB  1

 Overdominance model
        A  D
    AA  0  0
    AB  1  1
    BB  2  0
</pre>
<p>Specifically for the overdominance model, the column of interest is the D column.
</p>


<h3>Value</h3>

<p>This function outputs a list containg 4 elements that can be directly accessed using '<code>$</code>' operator.
</p>
<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p>numeric vector; It stores all <code>m</code> <code class="reqn">\beta</code> values that used to estimate the standardized profile likelihood.</p>
</td></tr>
<tr><td><code>profile.lik.norm</code></td>
<td>
<p>numeric vector; the corresponding <code>m</code> standardized profile likelihood value at each of the <code class="reqn">\beta</code> values in <code>theta</code>. If <code>robust</code>=<code>TRUE</code>, then the values will be adjusted by the robust factor.</p>
</td></tr>
<tr><td><code>k_cutoff</code></td>
<td>
<p>numeric vector; It specifies which k-cutoff had been used in the calculation, ordered from the smallest k to the largest k.</p>
</td></tr>
<tr><td><code>SummaryStats</code></td>
<td>
<p>data frame; contains the summary statistics of the profile likelihood calculation. It contains the following columns:
</p>

<ul>
<li> <p><code>mle</code>: the estimates for SNP effect with respect to the effective allele
</p>
</li>
<li> <p><code>maxlr</code>: maximum likelihood ratio in the beta grid defined by <code>lolim</code> and <code>hilim</code>
</p>
</li>
<li> <p><code>AF</code>: allele frequency for the effective allele
</p>
</li>
<li> <p><code>SNP</code>: SNP ID
</p>
</li>
<li> <p><code>bp</code>: base pair position from the <code>bim</code> input
</p>
</li>
<li> <p><code>effect</code>, <code>ref</code>: the effective allele and the other allele from the <code>bim</code> input
</p>
</li>
<li> <p><code>robustFactor</code>: robust factor calculated, set to 1 if <code>robust</code>=<code>FALSE</code>.
</p>
</li>
<li> <p><code>lo_1</code>, <code>hi_1</code>, <code>lo_2</code>, <code>hi_2</code>...: the lower and upper bound of the likelihood intervals for the kth cut-off in <code>k_cutoff</code>.
</p>
</li></ul>

</td></tr>
</table>


<h3>Note</h3>

<p>When <code>lolim</code> or <code>hilim</code> are NOT defined, then the boundaries of the beta grid will be determined by the default <code>bse=5</code>, or a <code>bse</code> defined by the user. Otherwise, the user can define the exact beta grid boundaries using <code>lolim</code> and <code>hilim</code>.
</p>
<p>In some cases the beta grid (using <code>bse</code> or <code>lolim</code>,<code>hilim</code>) may need to be increased substantially (<code>bse</code> as large as 15) if covariates are present in the formula. This is automatically dealt by the current function, but contributes to longer computation time to find the appropriate ranges. Estimation may become inaccurate with large number of correlated covariates, which is a known limitation of profile likelihoods.
</p>


<h3>References</h3>

<p>Strug, L. J., Hodge, S. E., Chiang, T., Pal, D. K., Corey, P. N., &amp; Rohde, C. (2010). A pure likelihood approach to the analysis of genetic association data: an alternative to Bayesian and frequentist analysis. Eur J Hum Genet, 18(8), 933-941. doi:10.1038/ejhg.2010.47
</p>
<p>Strug, L. J., &amp; Hodge, S. E. (2006). An alternative foundation for the planning and evaluation of linkage analysis. I. Decoupling &quot;error probabilities&quot; from &quot;measures of evidence&quot;. Hum Hered, 61(3), 166-188. doi:10.1159/000094709
</p>
<p>Royall, R. (1997). Statistical Evidence: A Likelihood Paradigm. London, Chapman and Hall.
</p>

<hr>
<h2 id='calculateGLR'>Profile likelihood calculation using regression models</h2><span id='topic+calculateGLR'></span>

<h3>Description</h3>

<p>This is the function that calculates profileLikelihood for a single SNP. The main function <code><a href="#topic+evian">evian</a></code> calls this function repeatedly to obtain results for multiple SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  calculateGLR(snp, formula_tofit, model, data, bim, lolim, hilim, m, bse,
                     family, c, plinkCC)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calculateGLR_+3A_snp">snp</code></td>
<td>
<p>a string specifying the SNP of interests to be calculated.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_formula_tofit">formula_tofit</code></td>
<td>
<p>a formula object of the genetic model. The model should be formatted as <code>y~nuisance parameters</code>. The parameter of interest should not be included here.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_model">model</code></td>
<td>
<p>a string specifying the mode of inheritance parameterization: <br /> <code>additive, dominant, recessive, or overdominance</code>. See details.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_data">data</code></td>
<td>
<p>data frame; read from the argument <code>data</code> in the main function <code><a href="#topic+evian">evian</a></code>. It should contain the SNP ID specified in the <code>snp</code> argument as a column name.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_bim">bim</code></td>
<td>
<p>data frame; read from from the argument <code>bim</code> in the main function <code><a href="#topic+evian">evian</a></code>. Provides allele information (base pair, effect/reference alleles) for the SNP of interest.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_lolim">lolim</code></td>
<td>
<p>numeric; the lower limit for the grid or the minimum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood function. </p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_hilim">hilim</code></td>
<td>
<p>numeric; the upper limit for the grid or the maximum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood funciton. </p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_m">m</code></td>
<td>
<p>numeric; the density of the grid at which to compute the standardized likelihood function. A beta grid is defined as the grid of values for the SNP parameter used to evaluate the likelihood function.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_bse">bse</code></td>
<td>
<p>numeric; the number of beta standard errors to utilize in constraining the beta grid limits. Beta grid is evaluated at <code class="reqn">\beta</code> +/- <code>bse</code>*s.e.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_family">family</code></td>
<td>
<p>the link function for <code>glm</code>.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_c">c</code></td>
<td>
<p>numeric; interval of Null Hypothesis to be tested.</p>
</td></tr>
<tr><td><code id="calculateGLR_+3A_plinkcc">plinkCC</code></td>
<td>
<p>A boolean type that specifies how case/control are coded. case/control were coded 1/0 if it is FALSE, and were coded 2/1 if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>calculateGLR</code> conducts a likelihood ratio test for testing the SNP of interest. It uses the same numerical approach as the main function <a href="#topic+calculateEvianMLE">calculateEvianMLE</a> to construct the likelihood function and it is then testing whether the effect of the SNP falls in an interval (-c, c) instead of testing whether the effect is 0 as in the <a href="#topic+calculateEvianMLE">calculateEvianMLE</a>.
</p>


<h3>Value</h3>

<p>This function outputs a dataframe that contains the summary statistics of the profile likelihood calculation. It contains the following columns:
</p>

<ul>
<li> <p><code>GLR</code>: the estimated generalized Likelihood ratio, a value smaller than 1 indicating in favor of the null hypothesis whereas a value greater than 1 indicating in favor of the alternative hypothesis.
</p>
</li>
<li> <p><code>boundary</code>: the boundary where null hypothesis is defined. i.e. the value c in (-c, c)
</p>
</li>
<li> <p><code>AF</code>: allele frequency for the effective allele
</p>
</li>
<li> <p><code>SNP</code>: SNP ID
</p>
</li>
<li> <p><code>bp</code>: base pair position from the <code>bim</code> input
</p>
</li>
<li> <p><code>effect</code>, <code>ref</code>: the effective allele and the other allele from the <code>bim</code> input
</p>
</li></ul>



<h3>Note</h3>

<p>When <code>lolim</code> or <code>hilim</code> are NOT defined, then the boundaries of the beta grid will be determined by the default <code>bse=5</code>, or a <code>bse</code> defined by the user. Otherwise, the user can define the exact beta grid boundaries using <code>lolim</code> and <code>hilim</code>.
</p>
<p>In some cases the beta grid (using <code>bse</code> or <code>lolim</code>,<code>hilim</code>) may need to be increased substantially (<code>bse</code> as large as 15) if covariates are present in the formula. This is automatically dealt by the current function, but contributes to longer computation time to find the appropriate ranges. Estimation may become inaccurate with large number of correlated covariates, which is a known limitation of profile likelihoods.
</p>


<h3>Author(s)</h3>

<p>Dr. Lisa J Strug <a href="mailto:lisa.strug@utoronto.ca">lisa.strug@utoronto.ca</a>
</p>


<h3>References</h3>

<p>Bickel, D. R. (2012). &ldquo;The strength of statistical evidence for composite hypotheses: Inference to the best explanation.&rdquo; Statistica Sinica, 22, 1147-1198.
</p>
<p>Zhang, Z., \&amp; Zhang, B. (2013). &ldquo;A likelihood paradigm forclinical trials. Journal of Statistical Theory and Practice&rdquo;, 7, 157-177.
</p>

<hr>
<h2 id='densityPlot'>Plot profile likelihood density for a single SNP.</h2><span id='topic+densityPlot'></span>

<h3>Description</h3>

<p>This function plots the density distribution for a single SNP calculated from the <code><a href="#topic+evian">evian</a></code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  densityPlot(dList, snpName, kcut = NULL, pl = 'linear', xlim = NULL,
  color = c('red','orange','green','blue'), round = 2,legend_cex=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="densityPlot_+3A_dlist">dList</code></td>
<td>
<p>a row-combined list, output from <code><a href="#topic+evian">evian</a></code>.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_snpname">snpName</code></td>
<td>
<p>a string specifying the SNP to be plotted.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_kcut">kcut</code></td>
<td>
<p>numeric; the cut-off to be plotted. If <code>kcut</code>=<code>NULL</code>, all values of k in the <code>kcutoff</code> will be plotted.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_pl">pl</code></td>
<td>
<p>a string specifying the y-axis for the plot. The y-axis will be plotted as 'Odds Ratio' if <code>pl</code> is specified as <code>logit</code>, 'Beta' otherwise.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_xlim">xlim</code></td>
<td>
<p>graphical parameter used in <code>plot</code> function</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_color">color</code></td>
<td>
<p>color of the likelihood interval lines from smallest to largest. For instance, <code>c('red','green')</code> for LIs of <code>k=c(8,32)</code> means that the 1/8 interval will be plotted as red, and 1/32 will be plotted as green.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_round">round</code></td>
<td>
<p>numeric; number of digits displayed on the plot.</p>
</td></tr>
<tr><td><code id="densityPlot_+3A_legend_cex">legend_cex</code></td>
<td>
<p>numeric; control the size of the legend, default 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes output from <code><a href="#topic+evian">evian</a></code> as input. It will plot the density of the estimated standardized profile likelihood for the SNP of interest. Some basic summary statistics will be included on the plot too.
</p>


<h3>References</h3>

<p>Strug, L. J., Hodge, S. E., Chiang, T., Pal, D. K., Corey, P. N., &amp; Rohde, C. (2010). A pure likelihood approach to the analysis of genetic association data: an alternative to Bayesian and frequentist analysis. Eur J Hum Genet, 18(8), 933-941. doi:10.1038/ejhg.2010.47
</p>
<p>Royall, R. (1997). Statistical Evidence: A Likelihood Paradigm. London, Chapman and Hall.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(evian_linear_raw)
data(evian_linear_bim)
rst1=evian(data=evian_linear_raw, bim=evian_linear_bim, xcols=10:ncol(evian_linear_raw),
      ycol=6, covariateCol=c(5,7:9), robust=FALSE, model="additive", m=200, lolim=-0.4,
      hilim=0.4, kcutoff = c(32,100), multiThread=1,family='gaussian',plinkCC=FALSE)

# Plot the density for rs912
densityPlot(dList=rst1,snpName='rs912')


</code></pre>

<hr>
<h2 id='evian'>Evidential analysis for genetic data using regression models</h2><span id='topic+evian'></span>

<h3>Description</h3>

<p>Calculates the likelihood intervals for genetic association in a genomic region of interest. Covariates can be accommodated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evian(data, bim, xcols = NULL, ycol = NULL, covariateCol = NULL,
      formula = NULL, robust = FALSE, model='additive', m=200,
      bse = 5, lolim = NULL, hilim = NULL, kcutoff = c(8,32,100,1000),
      multiThread = 1, family='gaussian',plinkCC=F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evian_+3A_data">data</code></td>
<td>
<p>a data frame includes a column for the response variable, one or multiple columns of genotype data (coded as <code>0, 1, 2,</code> or <code>NA</code>), and optionally columns for covariates. Headers are assumed. If the data is from related individuals, an additional column named <code>'FID'</code> needs to be included to specify the related structure. Using the PLINK toolkit with option <code>--recodeA</code> can produce the file in the required format and is recommended.</p>
</td></tr>
<tr><td><code id="evian_+3A_bim">bim</code></td>
<td>
<p>a data frame with six columns representing chromosome, SNP ID, physical distance, base pair position, effective allele, and reference allele. i.e. data from a file in PLINK binary format (bim). No header is assumed, but the ordering of the columns must follow the standard bim file format.</p>
</td></tr>
<tr><td><code id="evian_+3A_ycol">ycol</code></td>
<td>
<p>numeric; column index in the <code>data</code> data frame for the column representing the response variable.</p>
</td></tr>
<tr><td><code id="evian_+3A_xcols">xcols</code></td>
<td>
<p>numeric vector; the column range in the <code>data</code> where genotype information is stored. Note that although a range of X is required, only one SNP at a time is calculated.</p>
</td></tr>
<tr><td><code id="evian_+3A_covariatecol">covariateCol</code></td>
<td>
<p>numeric or numeric vector; optional argument specifying which columns represent covariates. If left as <code>NULL</code>, no covariates will be included and the model <code>Y~snp</code> will be used.</p>
</td></tr>
<tr><td><code id="evian_+3A_formula">formula</code></td>
<td>
<p>string; this is an alternative way of specifying model rather than using <code>xcols</code> and <code>ycol</code> arguments. This model follows the same format as the <code>glm</code> function (e.g. <code>Y~snp1+age+sex</code>). Note that in the case where multiple SNPs are included, only one SNP will be considered (e.g. given <code>Y~snp1+snp2</code>, the function will consider snp1 as the parameter of interests). The function can automatically identify SNPs with rsID as proper Xs, and would treat all other predictors as covariates.</p>
</td></tr>
<tr><td><code id="evian_+3A_robust">robust</code></td>
<td>
<p>logical; default <code>FALSE</code>. If <code>TRUE</code>, then a robust adjustment is applied to the likelihood function to account for clustering in the data; See <a href="#topic+robust_forCluster">robust_forCluster</a>.</p>
</td></tr>
<tr><td><code id="evian_+3A_model">model</code></td>
<td>
<p>a string that specifies the mode of inheritance parameterization: <br /> <code>additive, dominant, recessive,</code> or <code>overdominance</code>. Default <code>additive</code>.</p>
</td></tr>
<tr><td><code id="evian_+3A_m">m</code></td>
<td>
<p>numeric; the density of the grid at which to compute the standardized likelihood function. A beta grid is defined as the grid of values for the SNP parameter used to
evaluate the likelihood function.</p>
</td></tr>
<tr><td><code id="evian_+3A_bse">bse</code></td>
<td>
<p>numeric; the number of beta standard errors to utilize in constraining the beta grid limits. Beta grid is evaluated at <code class="reqn">\beta</code> +/- <code>bse</code>*s.e.</p>
</td></tr>
<tr><td><code id="evian_+3A_lolim">lolim</code></td>
<td>
<p>numeric; the lower limit for the grid or the minimum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood function. </p>
</td></tr>
<tr><td><code id="evian_+3A_hilim">hilim</code></td>
<td>
<p>numeric; the upper limit for the grid or the maximum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood funciton. </p>
</td></tr>
<tr><td><code id="evian_+3A_kcutoff">kcutoff</code></td>
<td>
<p>numeric or numeric vector; default = <code>c(8,32,100,1000)</code>. The strength of evidence criterion k. The function will calculate the <code>1/k</code> standardized likelihood intervals for each value provided.</p>
</td></tr>
<tr><td><code id="evian_+3A_multithread">multiThread</code></td>
<td>
<p>numeric; number of threads to use for parallel computing.</p>
</td></tr>
<tr><td><code id="evian_+3A_family">family</code></td>
<td>
<p>the link function for <code>glm</code>.</p>
</td></tr>
<tr><td><code id="evian_+3A_plinkcc">plinkCC</code></td>
<td>
<p>A boolean type that specifies how case/control are coded. case/control were coded 1/0 if it is FALSE, and were coded 2/1 if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>evian</code> is the main function called to calculate the <code>1/k</code> likelihood intervals for the additive, dominant, recessive, or overdominance genotypic models. This function calls <code>calculateEvianMLE</code> in parallel to calculate the likelihood for each SNP. The calculation details can be found in <code>calculateEvianMLE</code>.
</p>
<p>The input for the <code>data</code> and <code>bim</code> arguments can be obtained from the PLINK files; <code>data</code> is expected to follow PLINK format when run with the <code>--recodeA</code> option and <code>bim</code> can be obtained directly from a PLINK binary format file. Note if covariates are to be included, it is expected that the covariates are appended to the <code>data</code> file with a header for each covariate.
</p>
<p>The statistical model can be specified in two ways. Column index can be provided through the <code>xcols</code>, <code>ycol</code>, and <code>covariateCol</code> arguments or through the <code>formula</code> argument, which can accept a formula specified as the <code>formula</code> argument in the R <code>glm</code> function. We recommend using <code>xcols</code>, <code>ycol</code>, and <code>covariateCol</code> arguments in most scenarios as this is relatively easier to input and it works for all the cases that we have considered so far. The alternative <code>formula</code> argument is not able to detect non-rsID variants as parameters of interests, and is only suggested in the scenario where only a single variant is of interest and that its rsID is known in advance. Since the profileLikelihood can only accomendate scalar parameter and thus if multiple rsID variants are inputted through <code>formula</code> option, it will only assume the first one to be parameter of interests.s
</p>
<p>Parallel computing is avaliable through the use of the <code>multiThread</code> argument. This parallelization uses the <code>foreach</code> and <code>doMC</code> packages and will typically reduce computation time significantly. Due to this dependency, parallelization is not available on Windows OS as <code>foreach</code> and <code>doMC</code> are not supported on Windows.
</p>


<h3>Value</h3>

<p>This function outputs the row-combined the results from <code>calculateEvianMLE</code> for each of the SNPs included in the data/bim files. The exact output for each SNP can be found in the <code>calculateEvianMLE</code> documentation.
</p>


<h3>Note</h3>

<p>When <code>lolim/hilim</code> is NOT defined, then the boundaries of the beta grid will be determined by the default <code>bse=5</code>, or a <code>bse</code> defined by the user. Otherwise, the user can define the exact beta grid boundaries using <code>lolim/hilim</code>.
</p>
<p>In some cases the beta grid (using <code>bse</code> or <code>lolim/hilim</code>) may need to be increased substantially (<code>bse</code> as large as 15) if covariates are present in the formula. This is automatically dealt by the current function, but contribute to longer computation time to find the appropriate ranges. Estimation may become inaccurate with large number of correlated covariates, which is a known limitation of profile likelihoods.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateEvianMLE">calculateEvianMLE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(evian_linear_raw)
data(evian_linear_bim)
rst1=evian(data=evian_linear_raw, bim=evian_linear_bim, xcols=10:ncol(evian_linear_raw),
ycol=6, covariateCol=c(5,7:9), robust=FALSE, model="additive", m=200, lolim=-0.4,
hilim=0.4, kcutoff = c(32,100), multiThread=1,family='gaussian',plinkCC=FALSE)


</code></pre>

<hr>
<h2 id='evian_binary_bim'>Example map data for evian_binary_raw.</h2><span id='topic+evian_binary_bim'></span>

<h3>Description</h3>

<p>This dataset stored the corresponding SNP information from <code><a href="#topic+evian_binary_raw">evian_binary_raw</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(evian_binary_bim)
</code></pre>


<h3>Details</h3>

<p>This is the corresponding map file for <code><a href="#topic+evian_binary_raw">evian_binary_raw</a></code>. Specifically it stores the chromosome, base pair, and two alleles for the 30 SNPs listed in <code><a href="#topic+evian_binary_raw">evian_binary_raw</a></code> in the same order.
</p>

<hr>
<h2 id='evian_binary_raw'>Example dataset with a binary outcome.</h2><span id='topic+evian_binary_raw'></span>

<h3>Description</h3>

<p>This dataset included the genotypic and phenotypic information of 250 individuals in proper formats. This dataset is used together with <code><a href="#topic+evian_binary_bim">evian_binary_bim</a></code> to illustrate how to use the main function <code><a href="#topic+evian">evian</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(evian_binary_raw)
</code></pre>


<h3>Details</h3>

<p>This is an example dataset for <code><a href="#topic+evian">evian</a></code> function. It contained 250 individuals, and for each of the individuals, their genotype at 30 SNPs and a binary outcome (PHENOTYPE; coded as 0/1) were stored. Three additional covariates (age, weight, city) were provided as well. Specifically, our function can incorporate with related individuals, some of these individuals in the dataset are correlated with others, and are specified through the <code>FID</code> column. This is usually what a plink .raw file looks like except for the case/control status not coded as 1/2.
</p>

<hr>
<h2 id='evian_linear_bim'>Example map data for evian_linear_raw.</h2><span id='topic+evian_linear_bim'></span>

<h3>Description</h3>

<p>This dataset stored the corresponding SNP information from <code><a href="#topic+evian_linear_raw">evian_linear_raw</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(evian_linear_bim)
</code></pre>


<h3>Details</h3>

<p>This is the corresponding map file for <code><a href="#topic+evian_linear_raw">evian_linear_raw</a></code>. Specifically it stores the chromosome, base pair, and two alleles for the 10 SNPs listed in <code><a href="#topic+evian_linear_raw">evian_linear_raw</a></code> in the same order.
</p>

<hr>
<h2 id='evian_linear_raw'>Example dataset with a quantative outcome.</h2><span id='topic+evian_linear_raw'></span>

<h3>Description</h3>

<p>This dataset included the genotypic and phenotypic information of 781 individuals in proper formats. This dataset is used together with <code><a href="#topic+evian_linear_bim">evian_linear_bim</a></code> to illustrate how to use the main function <code><a href="#topic+evian">evian</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(evian_linear_raw)
</code></pre>


<h3>Details</h3>

<p>This is an example dataset for <code><a href="#topic+evian">evian</a></code> function. It contained 781 individuals, and for each of the individuals, their genotype at 10 SNPs and a continuous outcome (Y_norma) were stored. Three additional covariates (Fev, BMI_group, Age_group) were provided as well. Specifically, our function can incorporate with related individuals, some of these individuals in the dataset are correlated with others, and are specified through the <code>FID</code> column. This is usually what a plink .raw file looks like.
</p>

<hr>
<h2 id='expandBound'>A recursive function that expands the grid search for MLE.</h2><span id='topic+expandBound'></span>

<h3>Description</h3>

<p>This is an internal function that finds the proper boundary of the grid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  expandBound(data,bse,parameters,formula,m,k,family)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expandBound_+3A_data">data</code></td>
<td>
<p>a data frame inputted from the main function.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_bse">bse</code></td>
<td>
<p>numeric. The number of beta standard errors to utilize in constraining the beta grid limits. Passed down from argument <code>bse</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_parameters">parameters</code></td>
<td>
<p>a numeric vector of length 3 providing the starting values for the search. This is obtained from the <code><a href="#topic+getGridBound">getGridBound</a></code> function. The three numeric values in the vector should represent the beta estimates, s.e., and the correction factor respectively. Details can be found in <code><a href="#topic+getGridBound">getGridBound</a></code>.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_formula">formula</code></td>
<td>
<p>a formula specifying the response and possible covariates to keep in the output dataframe. This is directly obtained from <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_k">k</code></td>
<td>
<p>numeric vector. The strength of evidence criterion k. Passed down from argument <code>kcutoff</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_m">m</code></td>
<td>
<p>numeric. The density of the grid at which to compute the standardized likelihood function. Passed down from argument <code>m</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="expandBound_+3A_family">family</code></td>
<td>
<p>a string representing the link function for <code>ProfileLikelihood::ProfileLikelihood.glm</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Even though the initial grid bound calculated from <code>getGridBound</code> works for most of the data, there can be cases where <code>bse</code> needs to be increased in order to observe all the Likelihood Intervals (LIs) specified from the main function in the range <code>kcutoff</code> calculated. In this case, our approach is to check whether the current grid range includes the largest LIs. The function will expand the grid range by increasing <code>bse</code> by 1 if it is not included. This step will be running recursively until the largest LIs are included in the current grid range.
</p>


<h3>Value</h3>

<p>This function returns a numeric vector of length two representing the optimal lower and upper bounds for the grid on which the later functions will search for MLE.
</p>

<hr>
<h2 id='getGridBound'>Obtain the range of the grid where MLE will be searched at</h2><span id='topic+getGridBound'></span>

<h3>Description</h3>

<p>This is an internal function that provides the range where the profileLikelihood function would search for MLE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  getGridBound(formula,data,bse,k,m,family,robust)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getGridBound_+3A_formula">formula</code></td>
<td>
<p>a formula specifying the response and possible covariates to keep in the output dataframe. This is directly obtained from <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_data">data</code></td>
<td>
<p>a data frame inputted from the output of <code><a href="#topic+subsetData">subsetData</a></code>.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_bse">bse</code></td>
<td>
<p>numeric. The number of beta standard errors to utilize in constraining the beta grid limits. Passed down from argument <code>bse</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_k">k</code></td>
<td>
<p>numeric vector. The strength of evidence criterion k. Passed down from argument <code>kcutoff</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_m">m</code></td>
<td>
<p>numeric. The density of the grid at which to compute the standardized likelihood function. Passed down from argument <code>m</code> in the main <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_family">family</code></td>
<td>
<p>a string representing the link function for <code>ProfileLikelihood::ProfileLikelihood.glm</code>.</p>
</td></tr>
<tr><td><code id="getGridBound_+3A_robust">robust</code></td>
<td>
<p>A numeric value, robust correction factor.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>getGridBound</code> is an interior function that searches for the proper grid range that would be used to search for MLE. This is done through two steps: First, it finds a starting grid range by fitting a (generalized) linear model to obtain the estimate and s.e. of the beta. Then the starting grid range can be defined as mean +- bse*s.e. . In the case where robust correction is needed, the grid will be defined as mean +- bse*s.e./correction factor. Then the function determines an optimal grid range by using <code><a href="#topic+expandBound">expandBound</a></code> function.
</p>


<h3>Value</h3>

<p>This function returns a numeric vector of length 2 that represents the lower and upper bounds of the grid for the MLE search.
</p>

<hr>
<h2 id='glr'>Generalized Likelihood Ratio test</h2><span id='topic+glr'></span>

<h3>Description</h3>

<p>Conducts a generalized likelihood ratio test testing whether <code class="reqn">\beta \in</code>  (-<code>c</code>,<code>c</code>). Covariates can be accommodated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  glr(data, bim, xcols = NULL, ycol = NULL, covariateCol = NULL,
      formula = NULL, c, robust = FALSE, model = 'additive',
      m = 200, bse = 5, lolim = NULL, hilim = NULL, multiThread = 1,
      family='gaussian',plinkCC=F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glr_+3A_data">data</code></td>
<td>
<p>a data frame includes a column for the response variable, one or multiple columns of genotype data (coded as <code>0, 1, 2,</code> or <code>NA</code>), and optionally columns for covariates. Headers are assumed. If the data is from related individuals, an additional column named <code>'FID'</code> needs to be included to specify the related structure. Using the PLINK toolkit with option <code>--recodeA</code> can produce the file in the required format and is recommended.</p>
</td></tr>
<tr><td><code id="glr_+3A_bim">bim</code></td>
<td>
<p>a data frame with six columns representing chromosome, SNP ID, physical distance, base pair position, effective allele, and reference allele. i.e. data from a file in PLINK binary format (bim). No header is assumed, but the ordering of the columns must follow the standard bim file format.</p>
</td></tr>
<tr><td><code id="glr_+3A_ycol">ycol</code></td>
<td>
<p>numeric; column index in the <code>data</code> data frame for the column representing the response variable.</p>
</td></tr>
<tr><td><code id="glr_+3A_xcols">xcols</code></td>
<td>
<p>numeric vector; the column range in the <code>data</code> where genotype information is stored. Note that although a range of X is required, only one SNP at a time is calculated.</p>
</td></tr>
<tr><td><code id="glr_+3A_covariatecol">covariateCol</code></td>
<td>
<p>numeric or numeric vector; optional argument specifying which columns represent covariates. If left as <code>NULL</code>, no covariates will be included and the model <code>Y~snp</code> will be used.</p>
</td></tr>
<tr><td><code id="glr_+3A_formula">formula</code></td>
<td>
<p>string; this is an alternative way of specifying model rather than using <code>xcols</code> and <code>ycol</code> arguments. This model follows the same format as the <code>glm</code> function (e.g. <code>Y~snp1+age+sex</code>). Note that in the case where multiple SNPs are included, only the first SNP will be considered (e.g. given <code>Y~snp1+snp2</code>, the function will consider snp1 as the parameter of interests). The function can automatically identify SNPs with rsID as proper Xs, and would treat all other predictors as covariates.</p>
</td></tr>
<tr><td><code id="glr_+3A_c">c</code></td>
<td>
<p>numeric; interval of Null Hypothesis to be tested.</p>
</td></tr>
<tr><td><code id="glr_+3A_robust">robust</code></td>
<td>
<p>logical; default <code>FALSE</code>. If <code>TRUE</code>, then a robust adjustment is applied to the likelihood function to account for clustering in the data; See <a href="#topic+robust_forCluster">robust_forCluster</a>.</p>
</td></tr>
<tr><td><code id="glr_+3A_model">model</code></td>
<td>
<p>a string that specifies the mode of inheritance parameterization: <br /> <code>additive, dominant, recessive,</code> or <code>overdominance</code>. Default <code>additive</code>.</p>
</td></tr>
<tr><td><code id="glr_+3A_m">m</code></td>
<td>
<p>numeric; the density of the grid at which to compute the standardized likelihood function. A beta grid is defined as the grid of values for the SNP parameter used to
evaluate the likelihood function.</p>
</td></tr>
<tr><td><code id="glr_+3A_bse">bse</code></td>
<td>
<p>numeric; the number of beta standard errors to utilize in constraining the beta grid limits. Beta grid is evaluated at <code class="reqn">\beta</code> +/- <code>bse</code>*s.e.</p>
</td></tr>
<tr><td><code id="glr_+3A_lolim">lolim</code></td>
<td>
<p>numeric; the lower limit for the grid or the minimum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood function. </p>
</td></tr>
<tr><td><code id="glr_+3A_hilim">hilim</code></td>
<td>
<p>numeric; the upper limit for the grid or the maximum value of the regression parameter <code class="reqn">\beta</code> used to calculate the likelihood funciton. </p>
</td></tr>
<tr><td><code id="glr_+3A_multithread">multiThread</code></td>
<td>
<p>numeric; number of threads to use for parallel computing.</p>
</td></tr>
<tr><td><code id="glr_+3A_family">family</code></td>
<td>
<p>the link function for <code>glm</code>.</p>
</td></tr>
<tr><td><code id="glr_+3A_plinkcc">plinkCC</code></td>
<td>
<p>A boolean type that specifies how case/control are coded. case/control were coded 1/0 if it is FALSE, and were coded 2/1 if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a similar function to the main function <code>evian</code>. Instead of testing H0: <code class="reqn">\beta</code>=0, it tests for whether H0:<code class="reqn">\beta \in</code> (-<code>c</code>,<code>c</code>).
</p>


<h3>Value</h3>

<p>This function outputs the row-combined the results from <code>calculateGLR</code> for each of the SNPs included in the data/bim files. The exact output for each SNP can be found in the <code>calculateGLR</code> documentation.
</p>


<h3>Note</h3>

<p>When <code>lolim/hilim</code> is NOT defined, then the boundaries of the beta grid will be determined by the default <code>bse=5</code>, or a <code>bse</code> defined by the user. Otherwise, the user can define the exact beta grid boundaries using <code>lolim/hilim</code>.
</p>
<p>In some cases the beta grid (using <code>bse</code> or <code>lolim/hilim</code>) may need to be increased substantially (<code>bse</code> as large as 15) if covariates are present in the formula. This is automatically dealt by the current function, but contribute to longer computation time to find the appropriate ranges. Estimation may become inaccurate with large number of correlated covariates, which is a known limitation of profile likelihoods.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calculateGLR">calculateGLR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(evian_linear_raw)
  data(evian_linear_bim)
  rst2=glr(data=evian_linear_raw, bim=evian_linear_bim, xcols=10:ncol(evian_linear_raw),
             ycol=6, covariateCol=c(5,7:9), c=0.025,robust=F, model="additive",
             m=200, lolim=-0.6, hilim=0.6, multiThread=1,family='gaussian',plinkCC=FALSE)


</code></pre>

<hr>
<h2 id='glr_plot'>Plot methods for multiple glr result in a genomic region.</h2><span id='topic+glr_plot'></span>

<h3>Description</h3>

<p>This function plots the log10 GLR value in the region calculated from the <code><a href="#topic+glr">glr</a></code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> glr_plot(glr, snpName,kcut,col,legend_cex=1,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glr_plot_+3A_glr">glr</code></td>
<td>
<p>a dataframe, output from <code><a href="#topic+glr">glr</a></code>.</p>
</td></tr>
<tr><td><code id="glr_plot_+3A_snpname">snpName</code></td>
<td>
<p>a string specifying the SNP to be marked.</p>
</td></tr>
<tr><td><code id="glr_plot_+3A_kcut">kcut</code></td>
<td>
<p>numeric; the cut-off to be plotted. Can have value &lt;=1.</p>
</td></tr>
<tr><td><code id="glr_plot_+3A_col">col</code></td>
<td>
<p>color of the likelihood interval lines from smallest to largest. For instance, <code>c('red','green')</code> for LIs of <code>k=c(8,32)</code> means that the 1/8 interval will be plotted as red, and 1/32 will be plotted as green.</p>
</td></tr>
<tr><td><code id="glr_plot_+3A_legend_cex">legend_cex</code></td>
<td>
<p>numeric; control the size of the legend, default 1.</p>
</td></tr>
<tr><td><code id="glr_plot_+3A_...">...</code></td>
<td>
<p>Argument passed to the <code>plot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes output from <code><a href="#topic+glr">glr</a></code> as input. It will plot the GLR value on a log-scale and marked the SNPs of interests (defaultly the SNP with the maximum GLR values).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(evian_linear_raw)
  data(evian_linear_bim)
  rst2=glr(data=evian_linear_raw, bim=evian_linear_bim, xcols=10:ncol(evian_linear_raw),
             ycol=6, covariateCol=c(5,7:9), c=0.025,robust=F, model="additive",
             m=200, lolim=-0.6, hilim=0.6, multiThread=1,family='gaussian',plinkCC=FALSE)

  # Plot the density for rs912
  glr_plot(glr=rst2,ylim=c(-0.5,2))


</code></pre>

<hr>
<h2 id='multiLine_plot'>Plot methods for multiple likelihood intervals in a genomic region.</h2><span id='topic+multiLine_plot'></span>

<h3>Description</h3>

<p>This function plots the likelihood intervals (LIs) for all SNPs calculated using <code>evian</code> .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  multiLine_plot(bpstart = 0, bpend = 1000000000, dList, title = NULL,
  showmaxlr = 3, kcut = NULL, pl = 'linear', ylim = c(-0.5,10),
  color = c('violet','green','red','blue'), markSNP = NULL, round = 2,legend_cex=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multiLine_plot_+3A_bpstart">bpstart</code>, <code id="multiLine_plot_+3A_bpend">bpend</code></td>
<td>
<p>numeric; indicating the range of base pairs to be plotted. From <code>bpstart</code> to <code>bpend</code>.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_dlist">dList</code></td>
<td>
<p>a row-combined list, output from <code>evian</code>.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_title">title</code></td>
<td>
<p>string; title of plot</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_showmaxlr">showmaxlr</code></td>
<td>
<p>numeric; number of top SNPs to display on the graph. Default = 3. SNPs are chosen by their maximum likelihood ratio values.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_kcut">kcut</code></td>
<td>
<p>numeric; the cut-off to be plotted. If <code>kcut</code>=<code>NULL</code>, all intervals will be plotted.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_pl">pl</code></td>
<td>
<p>a string specifying the y-axis for the plot. The y-axis will be plotted as 'Odds Ratio' if <code>pl</code> is specified as <code>logit</code>, 'Beta' otherwise.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_marksnp">markSNP</code></td>
<td>
<p>vector of strings; indicates which SNPs to be marked on the plot. By default it will mark all SNPs that are significant at the smallest cut-off.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_round">round</code></td>
<td>
<p>numeric; number of digits displayed on the plot.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_ylim">ylim</code></td>
<td>
<p>graphical parameter used in <code>plot</code> function</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_color">color</code></td>
<td>
<p>color of the likelihood interval lines from smallest to largest. For instance, <code>c('red','green')</code> for LIs of <code>k=c(8,32)</code> means that the 1/8 interval will be plotted as red, and 1/32 will be plotted as green.</p>
</td></tr>
<tr><td><code id="multiLine_plot_+3A_legend_cex">legend_cex</code></td>
<td>
<p>numeric; control the size of the legend, default 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes output from <code>evian</code> as input. It will plot the likelihood intervals for each of the SNPs analyzed. If 1/k interval is significant then it will be colored by the specified color and will remain grey if the interval is not significant.
</p>


<h3>Author(s)</h3>

<p>Dr. Lisa J Strug <a href="mailto:lisa.strug@utoronto.ca">lisa.strug@utoronto.ca</a>
</p>


<h3>References</h3>

<p>Strug, L. J., Hodge, S. E., Chiang, T., Pal, D. K., Corey, P. N., &amp; Rohde, C. (2010). A pure likelihood approach to the analysis of genetic association data: an alternative to Bayesian and frequentist analysis. Eur J Hum Genet, 18(8), 933-941. doi:10.1038/ejhg.2010.47
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(evian_linear_raw)
data(evian_linear_bim)
rst1=evian(data=evian_linear_raw, bim=evian_linear_bim, xcols=10:ncol(evian_linear_raw),
ycol=6, covariateCol=c(5,7:9), robust=FALSE, model="additive", m=200, lolim=-0.4,
hilim=0.4, kcutoff = c(32,100), multiThread=1,family='gaussian',plinkCC=FALSE)

# Plot the LIs for all 3 SNPs
multiLine_plot(dList=rst1)


</code></pre>

<hr>
<h2 id='robust_forCluster'>Robust adjustment function</h2><span id='topic+robust_forCluster'></span>

<h3>Description</h3>

<p>The robust function computes an adjustment that is applied to the likelihood function to account for the cluster nature of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robust_forCluster(formula, data, family)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="robust_forCluster_+3A_formula">formula</code></td>
<td>
<p>a formula specifying the response and possible covariates to keep in the output data frame. This is directly obtained from the <code>evian</code>function.</p>
</td></tr>
<tr><td><code id="robust_forCluster_+3A_data">data</code></td>
<td>
<p>data frame; from the output of <code>subsetData</code>.</p>
</td></tr>
<tr><td><code id="robust_forCluster_+3A_family">family</code></td>
<td>
<p>string; the link function for <code>glm</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The robust function is called from within <code>evian</code> functions. It computes a robust adjustment factor that is
applied to the likelihood function to account for the cluster nature of the data. The family ID column (FID) specifies the clusters. The robust adjustment factor is the ratio of the regular variance estimator of the maximum likelihood estimate (MLE) to the sandwich variance estimator of the MLE, where the &lsquo;meat&rsquo; of the sandwich variance estimator is corrected for clustering in the data (Blume et.al, 2007).
If the data is not clustered (i.e. the observations are independent) then the adjustment factor can still be applied to make the working model robust to possible model misspecifications (Royall and Tsou, 2003).
</p>


<h3>Value</h3>

<p>A numeric constant.
</p>


<h3>Author(s)</h3>

<p> Zeynep Baskurt <a href="mailto:zeynep.baskurt@sickkids.ca">zeynep.baskurt@sickkids.ca</a> </p>


<h3>References</h3>

<p>Blume, J. D., Su, L., Remigio, M. O., &amp; McGarvey, S. T. (2007). Statistical evidence for GLM regression parameters: A robust likelihood approach. Statistics in Medicine, 26, 2919-2936.
</p>
<p>Royall, R. , Tsou, T. S. (2003). Interpreting statistical evidence by using imperfect models: robust adjusted likelihood functions. J Roy Stat Soc B; 65: 391-404.
</p>

<hr>
<h2 id='subsetData'>interior subsetting function</h2><span id='topic+subsetData'></span>

<h3>Description</h3>

<p>This is an internal function that subsets the SNP column with matching name and removes rows with missing observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  subsetData(snp,formula_tofit,data,plinkCC)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subsetData_+3A_snp">snp</code></td>
<td>
<p>a string specifying the SNP of interests. The SNP ID must exist in <code>data</code>.</p>
</td></tr>
<tr><td><code id="subsetData_+3A_formula_tofit">formula_tofit</code></td>
<td>
<p>a formula object of the genetic model. This is directly obtained from <code><a href="#topic+evian">evian</a></code> function.</p>
</td></tr>
<tr><td><code id="subsetData_+3A_data">data</code></td>
<td>
<p>a data frame inputted from the main function. Should contain the SNP ID <code>snp</code> as one of the column names.</p>
</td></tr>
<tr><td><code id="subsetData_+3A_plinkcc">plinkCC</code></td>
<td>
<p>A boolean type that specifies how case/control are coded. case/control were coded 1/0 if it is FALSE, and were coded 2/1 if TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>subsetData</code> is an interior function that subsets the full dataset into a smaller set containing only one specific SNP by the <code>snp</code> option. It will then remove any rows with missing values.
</p>


<h3>Value</h3>

<p>This function returned a dataframe containing phenotype, covariates in their original column names as in the full dataset, and a column called <code>X</code> representing the genotype information for the SNP chosen. The column names are essetial.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
