<!DOCTYPE html><html><head><title>Help for package evobiR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {evobiR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#evobiR-package'><p>evobiR: Evolutionary Biology in R</p></a></li>
<li><a href='#1.fasta'><p>simulated SNP data</p></a></li>
<li><a href='#AICc'>
<p>Computes an AICc score</p></a></li>
<li><a href='#AncCond'><p>Calculate the mean of a continuous character at the origin of derived state of a binary charachter</p></a></li>
<li><a href='#CalcD'><p>Calculate Patterson's D-statistic</p></a></li>
<li><a href='#Even'>
<p>Tests whether a number is even</p></a></li>
<li><a href='#FuzzyMatch'><p>Find Close Matches in a tree and dataset</p></a></li>
<li><a href='#horn.beetle.csv'><p>Gnatocerus measurements</p></a></li>
<li><a href='#hym.tree'><p>Phylogenetic tree</p></a></li>
<li><a href='#mcmc2'><p>mcmc log file</p></a></li>
<li><a href='#mite.trait'><p>phenotype data for mites</p></a></li>
<li><a href='#Mode'>
<p>Calcualtes the mode of a numeric vector</p></a></li>
<li><a href='#PPSDiscrete'><p>Create Simulated Datasets via PPS</p></a></li>
<li><a href='#ReorderData'>
<p>Reorders trait data to match the order of tips in a tree</p></a></li>
<li><a href='#ResSel'><p>Selection on Residuals</p></a></li>
<li><a href='#SampleTrees'><p>Select a random sample of trees</p></a></li>
<li><a href='#SlidingWindow'>
<p>Sliding window analysis</p></a></li>
<li><a href='#SuperMatrix'>
<p>creates a supermatrix from multiple gene alignments</p></a></li>
<li><a href='#trees'><p>10 Phylogenetic trees</p></a></li>
<li><a href='#trees.mite'><p>10 Phylogenetic trees</p></a></li>
<li><a href='#trees.nex'><p>100 Phylogenetic trees</p></a></li>
<li><a href='#ViewEvo'><p>Learning Resources</p></a></li>
<li><a href='#WinCalcD'><p>Calculate Patterson's D-statistic in sliding windows</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Comparative and Population Genetic Analyses</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-8-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Heath Blackmon and Richard H. Adams</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Heath Blackmon &lt;coleoguy@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.uta.edu/karyodb/evobiR/">http://www.uta.edu/karyodb/evobiR/</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Comparative analysis of continuous traits influencing discrete states, and utility tools to facilitate comparative analyses.  Implementations of ABBA/BABA type statistics to test for introgression in genomic data. Wright-Fisher, phylogenetic tree, and statistical distribution Shiny interactive simulations for use in teaching.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>seqinr, ape, geiger, shiny, phytools</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-09-06 15:55:34 UTC; hblackmo</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-09-06 19:30:55</td>
</tr>
</table>
<hr>
<h2 id='evobiR-package'>evobiR: Evolutionary Biology in R
</h2><span id='topic+evobiR-package'></span><span id='topic+evobiR'></span>

<h3>Description</h3>

<p><span class="pkg">evobiR</span> is a collection of tools for use in evolutionary biology.  Some of the functions manipulate data in a way not implemented by other functions while others calculate sequence statistics or perform simulations, either of data across trees or genetic and genomic simulations. </p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> evobiR</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2013-10-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>More information on <span class="pkg">evobiR</span> is available at <a href="http://coleoguy.github.io/software.html">http://coleoguy.github.io/software.html</a>
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>
<p>Maintainer: Heath Blackmon &lt;coleoguy@gmail.com&gt;
</p>

<hr>
<h2 id='1.fasta'>simulated SNP data</h2><span id='topic+1.fasta'></span><span id='topic+2.fasta'></span><span id='topic+3.fasta'></span>

<h3>Description</h3>

<p>This file contains simulated SNP data</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p>http://coleoguy.github.io/</p>

<hr>
<h2 id='AICc'>
Computes an AICc score</h2><span id='topic+AICc'></span>

<h3>Description</h3>

<p>Supplied with a log likelihood, the number of model parameters, and sample size calculates the small sample size version of the AIC score.</p>


<h3>Usage</h3>

<pre><code class='language-R'>AICc(loglik, K, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AICc_+3A_loglik">loglik</code></td>
<td>
<p>log likelihood.</p>
</td></tr>
<tr><td><code id="AICc_+3A_k">K</code></td>
<td>
<p>the number of parameters in the model</p>
</td></tr>
<tr><td><code id="AICc_+3A_n">N</code></td>
<td>
<p>the sample size.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns an AICc score.</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>AICc(-32, 3, 100)
</code></pre>

<hr>
<h2 id='AncCond'>Calculate the mean of a continuous character at the origin of derived state of a binary charachter</h2><span id='topic+AncCond'></span>

<h3>Description</h3>

<p>This function uses stochastic mapping and ancestral state reconstruction to determine if the derived state of a binary trait originates when a continuous trait has an extreme value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>AncCond(trees, data, derived.state, iterations=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AncCond_+3A_trees">trees</code></td>
<td>
<p>tree(s) of class phylo or multiPhylo</p>
</td></tr>
<tr><td><code id="AncCond_+3A_data">data</code></td>
<td>
<p>a dataframe with 3 columns. The first should match the taxa names in the tree, the second should have the continuous trait values and the third the states for the binary character</p>
</td></tr>
<tr><td><code id="AncCond_+3A_derived.state">derived.state</code></td>
<td>
<p>the derived condition for the binary trait</p>
</td></tr>
<tr><td><code id="AncCond_+3A_iterations">iterations</code></td>
<td>
<p>the number of iterations to be used in estimating significance</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses stochastic mapping and ancestral state reconstruction as implemented in phytools to determine if the derived state of a binary trait originates when a continuous trait has an extreme value.  This test assumes that the derived state of the binary character may lead to correlated selection in the continuous trait.  Because of this the ancestral state reconstruction of the continuous trait is based only on data from species that remain in the ancestral condition for the binary trait</p>


<h3>Value</h3>

<p>Returns a plot of the null distribution and the observed data as well as empirical p-value for the observed data.
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon and Richard H. Adams</p>


<h3>References</h3>

<p>http://coleoguy.github.io/</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(mite.trait)
data(trees.mite)
AncCond(trees, mite.trait, derived.state = "haplodiploidy", iterations=100) 

## End(Not run)
</code></pre>

<hr>
<h2 id='CalcD'>Calculate Patterson's D-statistic</h2><span id='topic+CalcD'></span><span id='topic+CalcPopD'></span>

<h3>Description</h3>

<p>These functions calculate Patterson's D-statistic to compare the frequencies of discordant SNP genealogies.  These tests assume equal substitution rates and unlinked loci, D-statistics significantly different from 0 suggest that introgression has occurred.</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalcD(alignment = "alignment.fasta", sig.test = "N", block.size = 1000, replicate = 1000)

CalcPopD(alignment = "alignment.fasta")

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalcD_+3A_alignment">alignment</code></td>
<td>
<p>This is an alignment in fasta format.  Sequences should be in the order: P1, P2, P3, Outgroup.</p>
</td></tr>
<tr><td><code id="CalcD_+3A_sig.test">sig.test</code></td>
<td>
<p>This indicates whether or if to test for significance. Options are &quot;B&quot; bootstrap, &quot;J&quot; jackknife, or &quot;N&quot; none.</p>
</td></tr>
<tr><td><code id="CalcD_+3A_block.size">block.size</code></td>
<td>
<p>The number of sites to be dropped in the jackknife approach</p>
</td></tr>
<tr><td><code id="CalcD_+3A_replicate">replicate</code></td>
<td>
<p>Number of replicates to be used in estimating variance</p>
</td></tr>
</table>


<h3>Details</h3>

<p> The functions CalcD and CalcPopD are implementations of the algorithm described in Durand et al. 2011.  Significance of the D-stat can be calculated either through bootstrapping or jackknifing.  Bootstrapping is appropriate for datasets where SNPs are unlinked for instance unmapped RADSeq data.  Jackknifing is the appropriate approach when SNPs are potentially in linkage for instance gene alignments or genome alignments.</p>


<h3>Value</h3>

<p>Returns the number of each type of site, Z scores and p-values
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p>http://coleoguy.github.io/
</p>
<p>Durand, Eric Y., et al. Testing for ancient admixture between closely related populations. Molecular biology and evolution 28.8 (2011): 2239-2252.
</p>
<p>Eaton, D. A. R., and R. H. Ree. 2013. Inferring phylogeny and introgression using RADseq data: An example from flowering plants (Pedicularis: Orobanchaceae). Syst. Biol. 62:689-706</p>


<h3>Examples</h3>

<pre><code class='language-R'>CalcD(alignment = system.file("1.fasta", package = "evobiR"), sig.test = "N")

CalcPopD(alignment = system.file("3.fasta", package = "evobiR"))
  </code></pre>

<hr>
<h2 id='Even'>
Tests whether a number is even</h2><span id='topic+Even'></span>

<h3>Description</h3>

<p>Just a simple function that returns True if a number is even and False otherwise.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Even(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Even_+3A_x">x</code></td>
<td>
<p>a numerical vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns a vector of logical values of the same length as the input vector. If the input value is not a number it will return an error message.</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>Even(c(1,2,3,4,5,6,2,5))
</code></pre>

<hr>
<h2 id='FuzzyMatch'>Find Close Matches in a tree and dataset</h2><span id='topic+FuzzyMatch'></span>

<h3>Description</h3>

<p>When assembling data from different sources typos can sometimes cause a loss of perfect matches between trees and datasets.  This function helps you find these close matches that can be hand curated to keep as many species as possible in your analysis.</p>


<h3>Usage</h3>

<pre><code class='language-R'>FuzzyMatch(tree, data, max.dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FuzzyMatch_+3A_tree">tree</code></td>
<td>
<p> a phylogenetic tree of the class &quot;phylo&quot;.</p>
</td></tr>
<tr><td><code id="FuzzyMatch_+3A_data">data</code></td>
<td>
<p> character vector with the names from your dataset.</p>
</td></tr>
<tr><td><code id="FuzzyMatch_+3A_max.dist">max.dist</code></td>
<td>
<p> This is the maximum number of characters that can differ between your tree and data and still be recognized as a close match.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with the following rows:
</p>
<p>Name in data<br />
Name in tree<br />
Number of differences<br />
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p>http://coleoguy.github.io/</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hym.tree)
names &lt;- c("Pepsis_elegans", "Plagiolepis_alluaudi", "Pheidele_lucreti",
           "Meliturgula_scriptifronsi", "Andrena_afimbriat")
FuzzyMatch(tree = hym.tree, data = names, max.dist=3)
</code></pre>

<hr>
<h2 id='horn.beetle.csv'>Gnatocerus measurements</h2><span id='topic+horn.beetle.csv'></span>

<h3>Description</h3>

<p>A csv file containing measurements of horn and body size for the beetle Gnatocerus cornutus.</p>

<hr>
<h2 id='hym.tree'>Phylogenetic tree</h2><span id='topic+hym.tree'></span>

<h3>Description</h3>

<p>This is a phylogenetic tree with 5 species of hymenoptera.</p>

<hr>
<h2 id='mcmc2'>mcmc log file</h2><span id='topic+mcmc2'></span><span id='topic+mcmc3'></span>

<h3>Description</h3>

<p>an mcmc log file.  The first column is the tree used during the iteration the remaining columns are the rate parameters of the Q matrix listed by column order.</p>

<hr>
<h2 id='mite.trait'>phenotype data for mites</h2><span id='topic+mite.trait'></span>

<h3>Description</h3>

<p>dataframe of sexual system and chromosome number data for mites</p>

<hr>
<h2 id='Mode'>
Calcualtes the mode of a numeric vector</h2><span id='topic+Mode'></span>

<h3>Description</h3>

<p>R's base package function mode returns the type of object 'numeric', 'character' etc.  This give the option of an easy  to remember work around for that.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mode(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Mode_+3A_x">x</code></td>
<td>
<p>a numerical vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns the most frequently occuring value in a vector. In the case of a tie it will return the mode which has the earliest initial occurence in the vector</p>


<h3>Value</h3>

<p>returns the most frequently occuring value in a series of numbers
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>Mode(c(1,2,3,4,5,6,2,5))
</code></pre>

<hr>
<h2 id='PPSDiscrete'>Create Simulated Datasets via PPS
</h2><span id='topic+PPSDiscrete'></span>

<h3>Description</h3>

<p>This function performs posterior predictive simulations of discrete traits.  The function is written to work with the output of bayesian programs that produce a collection of rate matrix parameter estimates based on either one or a collection of trees.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PPSDiscrete(trees, MCMC, states, N = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PPSDiscrete_+3A_trees">trees</code></td>
<td>
<p>an object of class &quot;multiPhylo&quot; or &quot;phylo&quot; containing the trees used in generting the rate estimates</p>
</td></tr>
<tr><td><code id="PPSDiscrete_+3A_mcmc">MCMC</code></td>
<td>
<p>this will normally be a log file that is brought into R with read.csv the columns for a three state character should be: tree, qAA, qBA, qCA, qAB, qBB, qCB, qAC, qBC, qCC.  If your analysis involves only a single tree then the tree column should be excluded.</p>
</td></tr>
<tr><td><code id="PPSDiscrete_+3A_states">states</code></td>
<td>
<p>a vector of root probabilities</p>
</td></tr>
<tr><td><code id="PPSDiscrete_+3A_n">N</code></td>
<td>
<p>the number of PPS datasets desired</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix is returned with the rownames being the species names from the tree and each column containing a result of a single PPS.
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(trees)
data(mcmc2)
data(mcmc3)
# 1 tree 100 q-mats 3 states
PPSDiscrete(trees[[1]], MCMC=mcmc3[,2:10], states=c(.5,.2,.3), N=2)
# 10 trees 100 q-mats 3 states
PPSDiscrete(trees, MCMC=mcmc3, states=c(.5,.2,.3), N=10)
# 10 trees 100 q-mats 2 states
PPSDiscrete(trees, MCMC=mcmc2, states=c(.5,.5), N=10)
</code></pre>

<hr>
<h2 id='ReorderData'>
Reorders trait data to match the order of tips in a tree</h2><span id='topic+ReorderData'></span>

<h3>Description</h3>

<p>This function takes a vector, matrix, or dataframe and reorders the data to match the order of tips in a phylo object.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReorderData(tree, data, taxa.names="row names")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReorderData_+3A_tree">tree</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="ReorderData_+3A_data">data</code></td>
<td>
<p>a vector, matrix, dataframe set of taxa names as present in the tree and data must match.  If data is a vector it should be a named vector.  If the data is a matrix or dataframe the taxa names may be row names or present in a column.</p>
</td></tr>
<tr><td><code id="ReorderData_+3A_taxa.names">taxa.names</code></td>
<td>
<p>If taxa names are present in a column the column number should be supplied.  If taxa names are the row names the argument can be set to &quot;row names&quot; (default setting).  If the data is being supplied in a vector this argument is not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns data in the same format as supplied but reordered to match the order of tip labels.</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>

<hr>
<h2 id='ResSel'>Selection on Residuals</h2><span id='topic+ResSel'></span>

<h3>Description</h3>

<p>This function takes measurements of multiple traits and performs a linear regression and identifies those records with the largest and smallest residual.  Originally it was written to perform a regression of horn size on body size allowing for high and low selection lines.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ResSel(data, traits, percent = 10, identifier = 1, model = "linear")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ResSel_+3A_data">data</code></td>
<td>
<p>this is a dataframe with subject identifiers and phenotypic trait values</p>
</td></tr>
<tr><td><code id="ResSel_+3A_traits">traits</code></td>
<td>
<p>a numeric vector indicating the column containing the predictor and response variables in that order</p>
</td></tr>
<tr><td><code id="ResSel_+3A_percent">percent</code></td>
<td>
<p>the percentage of highest and lowest residuals that should be identified</p>
</td></tr>
<tr><td><code id="ResSel_+3A_identifier">identifier</code></td>
<td>
<p>the column which contains the record numbers to identify individuals</p>
</td></tr>
<tr><td><code id="ResSel_+3A_model">model</code></td>
<td>
<p>currently this is not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a list
</p>
<table>
<tr><td><code>high line</code></td>
<td>
<p> the ID numbers for the individuals selected for the high line</p>
</td></tr>
<tr><td><code>low line</code></td>
<td>
<p> the ID numbers of the individuals selected for the low line</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- read.csv(file = system.file("horn.beetle.csv", package = "evobiR"))
ResSel(data = data, traits = c(2,3), percent = 15, identifier = 1, model = "linear")
</code></pre>

<hr>
<h2 id='SampleTrees'>Select a random sample of trees</h2><span id='topic+SampleTrees'></span>

<h3>Description</h3>

<p>This function takes as its input a large collection of trees from a program like MrBayes or Beast and allows the user to select the number of randomly drawn trees they wish to retrieve</p>


<h3>Usage</h3>

<pre><code class='language-R'>SampleTrees(trees, burnin, final.number, format, prefix)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SampleTrees_+3A_trees">trees</code></td>
<td>
<p>a nexus format file containing trees that the user wants to sample from</p>
</td></tr>
<tr><td><code id="SampleTrees_+3A_burnin">burnin</code></td>
<td>
<p>the proportion of trees to remove as burnin</p>
</td></tr>
<tr><td><code id="SampleTrees_+3A_final.number">final.number</code></td>
<td>
<p>the number of trees desired</p>
</td></tr>
<tr><td><code id="SampleTrees_+3A_format">format</code></td>
<td>
<p>options are &quot;new&quot; or &quot;nex&quot; indicating to save the trees in newick format or nexus format</p>
</td></tr>
<tr><td><code id="SampleTrees_+3A_prefix">prefix</code></td>
<td>
<p>a text string to assing to the new treefile name
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of the class &quot;multiPhylo&quot; is returned
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>SampleTrees(trees = system.file("trees.nex", package = "evobiR"), 
            burnin = .1, final.number = 20, format = 'new', prefix = 'sample')</code></pre>

<hr>
<h2 id='SlidingWindow'>
Sliding window analysis</h2><span id='topic+SlidingWindow'></span>

<h3>Description</h3>

<p>Applies a function within a sliding window of a numeric vector.  Both the step size and the window size can be set by the user.</p>


<h3>Usage</h3>

<pre><code class='language-R'>SlidingWindow(FUN, data, window, step)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SlidingWindow_+3A_fun">FUN</code></td>
<td>
<p>a function to be applied within each window.</p>
</td></tr>
<tr><td><code id="SlidingWindow_+3A_data">data</code></td>
<td>
<p>a numerical vector.</p>
</td></tr>
<tr><td><code id="SlidingWindow_+3A_window">window</code></td>
<td>
<p>an integer setting the size of the window.</p>
</td></tr>
<tr><td><code id="SlidingWindow_+3A_step">step</code></td>
<td>
<p>an integer setting the size of step between windows.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns a vector of numeric values representing the applying the selected function within each window.  The length will be unequal to the original data and will be determined primarily by the step size.</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- c(1,2,1,2,10,2,1,2,1,2,3,4,5,6,2,5)
SlidingWindow("mean", data, 3, 1)
</code></pre>

<hr>
<h2 id='SuperMatrix'>
creates a supermatrix from multiple gene alignments</h2><span id='topic+SuperMatrix'></span>

<h3>Description</h3>

<p>combines all alignments in a folder into a single supermatrix</p>


<h3>Usage</h3>

<pre><code class='language-R'>SuperMatrix(missing = "-", prefix = "concatenated", save = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SuperMatrix_+3A_missing">missing</code></td>
<td>
<p>the character to use when no data is available for a taxa</p>
</td></tr>
<tr><td><code id="SuperMatrix_+3A_prefix">prefix</code></td>
<td>
<p>prefix for the resulting supermatrix</p>
</td></tr>
<tr><td><code id="SuperMatrix_+3A_save">save</code></td>
<td>
<p>if True then supermatrix and partition file will be saved</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reads all fasta format alignments in the working directory and constructs a single supermatrix that includes all taxa present in any of the fasta files and inserts missing symbols for taxa that are missing sequences for some loci.</p>


<h3>Value</h3>

<p>A list with two elements is returned. The first element contains partition data that records the alignment positions of each input fasta file in the combined supermatrix.  The second element is a dataframe version of the supermatrix. If the argument save is set to True then both of these files are also saved to the working directory.
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
SuperMatrix(missing = "N", prefix = "DATASET2", save = T)

## End(Not run)
</code></pre>

<hr>
<h2 id='trees'>10 Phylogenetic trees</h2><span id='topic+trees'></span>

<h3>Description</h3>

<p>This is a collection of 10 simulated phylogenetic trees with 200 tips each.</p>

<hr>
<h2 id='trees.mite'>10 Phylogenetic trees</h2><span id='topic+trees.mite'></span>

<h3>Description</h3>

<p>These are trees from a previously published work on mite sexual system evolution.</p>

<hr>
<h2 id='trees.nex'>100 Phylogenetic trees</h2><span id='topic+trees.nex'></span>

<h3>Description</h3>

<p>This is a collection of 100 simulated phylogenetic trees with 10 tips each.</p>

<hr>
<h2 id='ViewEvo'>Learning Resources</h2><span id='topic+ViewEvo'></span>

<h3>Description</h3>

<p>This uses the shiny app to produce interactive pages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ViewEvo(simulation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ViewEvo_+3A_simulation">simulation</code></td>
<td>
<p>Text string indicating the application to run.  Currently options are &quot;wf.model&quot;, &quot;bd.model&quot;, &quot;dist.model&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>  The wf.model was implemented to illustrate to students the effects of genetic drift.  In particular the high likelihood of losing a beneficial allele when population size is finite.  The bd.model will plot 4 phylogenetic trees based on a birth death model with a single set of parameters.  This application was developed to illustrate the high variability of a birth death process as a generating model for phylogenies and the inherint difficulty in detecting differential diversification rates.  Finally the dist.model was developed to help illustrate the relationship between common statistical distributions often used as priors and the way that parameters effect the density distribution.</p>


<h3>Value</h3>

<p>This function returns an interactive webpage.</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a>
</p>
<p>Wright-Fisher Simulator: <a href="https://evobir.shinyapps.io/wf_model/">https://evobir.shinyapps.io/wf_model/</a>
</p>
<p>Birth-death Simulator: <a href="https://evobir.shinyapps.io/bd_model">https://evobir.shinyapps.io/bd_model</a>
</p>
<p>Statistical Distribution: <a href="https://evobir.shinyapps.io/dist_model">https://evobir.shinyapps.io/dist_model</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ViewEvo("wf.model")
ViewEvo("bd.model")
ViewEvo("dist.model")

## End(Not run)
</code></pre>

<hr>
<h2 id='WinCalcD'>Calculate Patterson's D-statistic in sliding windows</h2><span id='topic+WinCalcD'></span>

<h3>Description</h3>

<p>This functions calculate Patterson's D-statistic in windows.</p>


<h3>Usage</h3>

<pre><code class='language-R'>WinCalcD(alignment = "alignment.fasta", win.size = 100, step.size=50,
boot = F, replicate = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WinCalcD_+3A_alignment">alignment</code></td>
<td>
<p>This is an alignment in fasta format</p>
</td></tr>
<tr><td><code id="WinCalcD_+3A_win.size">win.size</code></td>
<td>
<p>This is the size of the window used</p>
</td></tr>
<tr><td><code id="WinCalcD_+3A_step.size">step.size</code></td>
<td>
<p>This is the size of steps in the sliding window</p>
</td></tr>
<tr><td><code id="WinCalcD_+3A_boot">boot</code></td>
<td>
<p>This indicates whether or not bootstrapping should be performed to estimate variance</p>
</td></tr>
<tr><td><code id="WinCalcD_+3A_replicate">replicate</code></td>
<td>
<p>Number of replicates to be used in estimating variance</p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function is just an extension of CalcD and calculates D statistic for windows. </p>


<h3>Value</h3>

<p>Returns a table with the number of each type of site, Z scores and p-values for each window in the genome
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon</p>


<h3>References</h3>

<p>http://coleoguy.github.io/
</p>
<p>Durand, Eric Y., et al. Testing for ancient admixture between closely related populations. Molecular biology and evolution 28.8 (2011): 2239-2252.
</p>
<p>Eaton, D. A. R., and R. H. Ree. 2013. Inferring phylogeny and introgression using RADseq data: An example from flowering plants (Pedicularis: Orobanchaceae). Syst. Biol. 62:689-706</p>


<h3>Examples</h3>

<pre><code class='language-R'>WinCalcD(alignment = system.file("1.fasta", package = "evobiR"), 
         win.size=100, step.size=50, boot = TRUE, replicate=10)
  </code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
