<!DOCTYPE html><html lang="en"><head><title>Help for package evolved</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {evolved}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#birds_spp'><p>Birds species list</p></a></li>
<li><a href='#calcFossilDivTT'><p>Calculate paleo diversity curves through different methods</p></a></li>
<li><a href='#checkAndFixUltrametric'><p>Find and fix small rounding errors in ultrametric trees</p></a></li>
<li><a href='#countSeqDiffs'><p>Counting protein sequence differences</p></a></li>
<li><a href='#cytOxidase'><p>Cytochrome Oxidase sequences</p></a></li>
<li><a href='#data_whales'><p>Whale body size and speciation rates</p></a></li>
<li><a href='#dinos_fossil'><p>Occurrence of dinosaur fossils</p></a></li>
<li><a href='#estimateSpeciation'><p>Estimate speciation assuming a pure-birth process</p></a></li>
<li><a href='#fitCRBD'><p>Fit a constant-rate birth-death process to a phylogeny</p></a></li>
<li><a href='#lttPlot'><p>Make a lineage through time (LTT) plot</p></a></li>
<li><a href='#NatSelSim'><p>Simulating natural selection through time in a bi-allelic gene</p></a></li>
<li><a href='#OneGenHWSim'><p>Simulating one generation of genotypes under Hardy-Weinberg equilibrium</p></a></li>
<li><a href='#plotNatSel'><p>Plot NatSelSim output</p></a></li>
<li><a href='#plotPaintedWhales'><p>Plotting the whale phylogeny and coloring its clades</p></a></li>
<li><a href='#plotProteinSeq'><p>Plot protein sequence(s)</p></a></li>
<li><a href='#plotRawFossilOccs'><p>Plot a literal interpretation of a fossil record</p></a></li>
<li><a href='#plotWFDrift'><p>Plot WFDriftSim output</p></a></li>
<li><a href='#ProteinSeq'><p>Details, generics, and methods for the <code>ProteinSeq</code> class</p></a></li>
<li><a href='#simulateBirthDeathRich'><p>Simulating richness through birth-death processes</p></a></li>
<li><a href='#simulateTree'><p>Simulating a phylogenetic trees through the birth-death process</p></a></li>
<li><a href='#timeseries_fossil'><p>Fossil Time series</p></a></li>
<li><a href='#WFDriftSim'><p>Simulating generations of genetic drift in a Wright–Fisher (WF) population</p></a></li>
<li><a href='#whale_phylo'><p>Whale Phylogeny</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Open Software for Teaching Evolutionary Biology at Multiple
Scales Through Virtual Inquiries</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>"Evolutionary Virtual Education" - 'evolved' - provides multiple tools to help educators (especially at the graduate level or in advanced undergraduate level courses) apply inquiry-based learning in general evolution classes. In particular, the tools provided include functions that simulate evolutionary processes (e.g., genetic drift, natural selection within a single locus) or concepts (e.g. Hardy-Weinberg equilibrium, phylogenetic distribution of traits). More than only simulating, the package also provides tools for students to analyze (e.g., measuring, testing, visualizing) datasets with characteristics that are common to many fields related to evolutionary biology. Importantly, the package is heavily oriented towards providing tools for inquiry-based learning - where students follow scientific practices to actively construct knowledge. For additional details, see package's vignettes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, diversitree, phytools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0), LSD, paleobioDB,
BAMMtools, plot3D</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/Auler-J/evolved&amp;gt;">https://github.com/Auler-J/evolved&gt;</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Auler-J/evolved/issues">https://github.com/Auler-J/evolved/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-26 15:47:16 UTC; matheus</td>
</tr>
<tr>
<td>Author:</td>
<td>Matheus Januario <a href="https://orcid.org/0000-0002-6480-7095"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph, cre],
  Jennifer Auler <a href="https://orcid.org/0000-0001-7576-9058"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Andressa Viol [aut],
  Daniel Rabosky [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matheus Januario &lt;januarioml.eco@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-27 12:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='birds_spp'>Birds species list</h2><span id='topic+birds_spp'></span>

<h3>Description</h3>

<p>Birds species list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>birds_spp
</code></pre>


<h3>Format</h3>

<p>A <code>vector</code> containing the names of almost all (i.e., 9993) extant bird species, following Jetz et al. (2012) taxonomy. This dataset is part of the package and is
licensed under the Creative Commons Attribution 4.0 International License (CC BY 4.0).
</p>


<h3>Source</h3>

<p>Actual file downloaded from <a href="https://vertlife.org/data/">https://vertlife.org/data/</a>
</p>


<h3>References</h3>

<p>Jetz, W., Thomas, G. H., Joy, J. B., Hartmann, K., &amp; Mooers, A. O. (2012). The global diversity of birds in space and time. Nature, 491(7424), 444-448.
</p>

<hr>
<h2 id='calcFossilDivTT'>Calculate paleo diversity curves through different methods</h2><span id='topic+calcFossilDivTT'></span>

<h3>Description</h3>

<p><code>calcFossilDivTT</code> calculates fossil diversity through time using
different methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcFossilDivTT(
  data,
  tax.lvl = "species",
  method = "rangethrough",
  bin.reso = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calcFossilDivTT_+3A_data">data</code></td>
<td>
<p>A <code>data.frame</code> containing the columns: <code>max_ma</code>,
<code>min_ma</code> and  the name provided in <code>tax.lvl</code>. <code>max_ma</code> and
<code>min_ma</code> are respectively the early and late bounds of rock layer's age.
<code>tax.lvl</code> column is the taxonomic level of the data. Any additional
columns are ignored.</p>
</td></tr>
<tr><td><code id="calcFossilDivTT_+3A_tax.lvl">tax.lvl</code></td>
<td>
<p>A <code>character</code> giving the taxonomic in which
calculations will be based on (default value is <code>"species"</code>).
This must refer to the column names in <code>data</code>.</p>
</td></tr>
<tr><td><code id="calcFossilDivTT_+3A_method">method</code></td>
<td>
<p>A <code>character</code> string setting the method which should be used.
Could be either <code>"rangethrough"</code> or <code>"stdmethod"</code>, which will
respectively calculate diversity using the range through or the standard
methods (Foote &amp; Miller, 2007)</p>
</td></tr>
<tr><td><code id="calcFossilDivTT_+3A_bin.reso">bin.reso</code></td>
<td>
<p>A <code>numeric</code> assigning the resolution (length) of the
time bin to consider in calculations. Default value is <code>1</code> (which in
most cases - e.g. those following the Paleobiology Database default
timescale - will equate to one million years)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> containing the diversity (column <code>div</code>) of
the chosen taxonomic level through time, with calculation based on
<code>method</code>. If <code>"method = rangethrough"</code>, the time moments are the
layer boundaries given in <code>data</code>.
If <code>"method = stdmethod"</code>, the time moments are evenly-space bins  with
length equal to <code>bin.reso</code>, starting at the earliest bound in the
dataset.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Foote, M., Miller, A. I., Raup, D. M., &amp; Stanley, S. M. (2007). Principles
of paleontology. Macmillan.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Loading data
data("dinos_fossil")

# Using function:
div1 &lt;- calcFossilDivTT(dinos_fossil, method = "stdmethod")
div2 &lt;- calcFossilDivTT(dinos_fossil, method = "stdmethod", bin.reso = 10)

# Comparing different bins sizes in the standard method
plot(x=div1$age, y=div1$div, type="l", 
     xlab = "Time (Mya)", ylab = "Richness", 
     xlim=rev(range(div1$age)), col="red") 
lines(x=div2$age, y=div2$div, col="blue")

# Comparing different methods:
div3 &lt;- calcFossilDivTT(dinos_fossil, method = "rangethrough")
plot(x=div1$age, y=div1$div, type="l", 
     xlab = "Time (Mya)", ylab = "Richness", 
     xlim=rev(range(div1$age)), col="red") 
lines(x=div3$age, y=div3$div, col="blue")

</code></pre>

<hr>
<h2 id='checkAndFixUltrametric'>Find and fix small rounding errors in ultrametric trees</h2><span id='topic+checkAndFixUltrametric'></span>

<h3>Description</h3>

<p><code>checkAndFixUltrametric</code> finds and correct small numerical errors that
might appear in ultrametric trees that where created through simulations.
This function should never be used as a formal statistical method to make a
tree ultrametric, as it was designed just to correct small rounding errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkAndFixUltrametric(phy)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkAndFixUltrametric_+3A_phy">phy</code></td>
<td>
<p>A <code>phylo</code> object, following terminology from package
<code>ape</code> in which function will operate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A check and fixed <code>phylo</code> object.
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Paradis, E. (2012). Analysis of Phylogenetics and Evolution with R (Vol. 2).
New York: Springer.
</p>
<p>Popescu, A. A., Huber, K. T., &amp; Paradis, E. (2012). ape 3.0: New tools for
distance-based phylogenetics and evolutionary analysis in R. Bioinformatics,
28(11), 1536-1537.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>S &lt;- 1
E &lt;- 0
set.seed(1)
phy &lt;- simulateTree(pars = c(S, E), max.taxa = 6, max.t = 5)
phy$edge.length[1] &lt;- phy$edge.length[1]+0.1
ape::is.ultrametric(phy)
phy &lt;- checkAndFixUltrametric(phy)
ape::is.ultrametric(phy)

</code></pre>

<hr>
<h2 id='countSeqDiffs'>Counting protein sequence differences</h2><span id='topic+countSeqDiffs'></span>

<h3>Description</h3>

<p><code>countSeqDiffs</code> counts the number of protein differences among two sequences of proteins within the same &quot;ProteinSeq&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countSeqDiffs(x, taxon1, taxon2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="countSeqDiffs_+3A_x">x</code></td>
<td>
<p>A &quot;ProteinSeq&quot; object containing proteins from <code>taxon1</code>
and <code>taxon2</code>.</p>
</td></tr>
<tr><td><code id="countSeqDiffs_+3A_taxon1">taxon1</code></td>
<td>
<p>A character giving the common name of the first species that
will be compared. Must be a name present in <code>x</code>.</p>
</td></tr>
<tr><td><code id="countSeqDiffs_+3A_taxon2">taxon2</code></td>
<td>
<p>A character giving the common name of the second species that
will be compared. Must be a name present in <code>x</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A integer giving the number of protein differences between
<code>taxon1</code> and <code>taxon2</code>.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Dan Rabosky, Jennifer Auler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>countSeqDiffs(cytOxidase, "human", "chimpanzee")

countSeqDiffs(cytOxidase, "human", "cnidaria")

countSeqDiffs(cytOxidase, "chimpanzee", "cnidaria")

</code></pre>

<hr>
<h2 id='cytOxidase'>Cytochrome Oxidase sequences</h2><span id='topic+cytOxidase'></span>

<h3>Description</h3>

<p><code>cytOxidase</code> is a set of homologous protein sequences from the GENE
cytochrome oxidase SUBUNIT 1 gene. This mitochondrial gene, often known as
CO1 (“see-oh-one”),  plays a key role in cellular respiration. C01 contains
approximately 513 aminoacids and has been used by previous studies for
reconstructing phylogenetic trees and estimating divergence times in
Metazoaria by assuming a molecular clock. Its 5' partition is used for the
‘Barcoding of Life’ initiative, for instance. This dataset is part of the
package and is licensed under the Creative Commons Attribution 4.0
International License (CC BY 4.0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytOxidase
</code></pre>


<h3>Format</h3>

<p>A  object of class &quot;ProteinSeq&quot; with 17 entries, each representing
a different animal species
</p>

<dl>
<dt>names</dt><dd><p>Organism's popular name or a taxonomic classification</p>
</dd>
<dt>sequence</dt><dd><p>Aminoacid sequence of length 513</p>
</dd>
</dl>



<h3>Details</h3>

<p>The object of class &quot;ProteinSeq&quot; is structured as a named
<code>vector</code> of 17 different animal species, with each individual component
being a sequence of 513 aminoacids.
</p>


<h3>Source</h3>

<p>Amino acid sequences were originally downloaded from genebank and later
curated and aligned by Daniel L. Rabosky.
</p>

<hr>
<h2 id='data_whales'>Whale body size and speciation rates</h2><span id='topic+data_whales'></span>

<h3>Description</h3>

<p>Data on the body size of many cetacean species and species-specific
speciation rates. This dataset is part of the package and is licensed
under the Creative Commons Attribution 4.0 International License (CC BY 4.0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_whales
</code></pre>


<h3>Format</h3>

<p>A <code>data.frame</code> with 75 rows and 4 columns.
</p>

<dl>
<dt>species</dt><dd><p>Whale species</p>
</dd>
<dt>log_mass</dt><dd><p>Log of body mass (grams)</p>
</dd>
<dt>S</dt><dd><p>Species-specific speciation rate</p>
</dd>
<dt>color</dt><dd><p>Suggested color to be used for the tip's clade</p>
</dd>
</dl>



<h3>Details</h3>

<p>Species follow taxonomy from Steeman et al (2009).
Species-specific speciation rates from Rabosky 2014 &amp; Rabosky
et al, 2014. Mass data from PanTHERIA (Jones et al, 2009).
</p>


<h3>Source</h3>

<p>Compilation of many primary sources (see details).
</p>


<h3>References</h3>

<p>Jones, K. E., Bielby, J., Cardillo, M., Fritz, S. A., O'Dell, J., Orme, C. D. L., ... &amp; Purvis, A. (2009). PanTHERIA: a species‐level database of life history, ecology, and geography of extant and recently extinct mammals: Ecological Archives E090‐184. Ecology, 90(9), 2648-2648.
</p>
<p>Rabosky, D. L. (2014). Automatic detection of key innovations, rate shifts, and diversity-dependence on phylogenetic trees. PLoS one, 9(2), e89543.
</p>
<p>Rabosky, D. L., Grundler, M., Anderson, C., Title, P., Shi, J. J., Brown, J. W., ... &amp; Larson, J. G. (2014). BAMM tools: an R package for the analysis of evolutionary dynamics on phylogenetic trees. Methods in Ecology and Evolution, 5(7), 701-707.
</p>
<p>Steeman, M. E., Hebsgaard, M. B., Fordyce, R. E., Ho, S. Y., Rabosky, D. L., Nielsen, R., ... &amp; Willerslev, E. (2009). Radiation of extant cetaceans driven by restructuring of the oceans. Systematic biology, 58(6), 573-585.
</p>

<hr>
<h2 id='dinos_fossil'>Occurrence of dinosaur fossils</h2><span id='topic+dinos_fossil'></span>

<h3>Description</h3>

<p>Many dinosaur (including avian species) fossil occurrences from different
moments of the geological past. Much information (i.e., extra columns) was
removed from the original dataset to make it more compact, but it can be
fully accessed by the data URL. This dataset is part of the package and is
licensed under the Creative Commons Attribution 4.0 International License
(CC BY 4.0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dinos_fossil
</code></pre>


<h3>Format</h3>

<p>A <code>data.frame</code> containing 15527 rows and 13 columns
</p>

<dl>
<dt>phylum</dt><dd><p>Organism phylum</p>
</dd>
<dt>class</dt><dd><p>Organism taxonomic class</p>
</dd>
<dt>order</dt><dd><p>Organism taxonomic order</p>
</dd>
<dt>family</dt><dd><p>Organism taxonomic family</p>
</dd>
<dt>genus</dt><dd><p>Organism genus</p>
</dd>
<dt>species</dt><dd><p>Organism specific name</p>
</dd>
<dt>early_interval</dt><dd><p>Earlier known geological period of occurrence</p>
</dd>
<dt>late_interval</dt><dd><p>Later known geological period of occurrence</p>
</dd>
<dt>max_ma</dt><dd><p>Occurrence's oldest time boundary in million years</p>
</dd>
<dt>min_ma</dt><dd><p>Occurrence's newest time boundary in million years</p>
</dd>
<dt>midpoint</dt><dd><p>Midpoint between max_ma and min_ma</p>
</dd>
<dt>lng</dt><dd><p>Longitude of place where occurrence was found. Follows decimal degree format.</p>
</dd>
<dt>lat</dt><dd><p>Latitude of place where occurrence was found. Follows decimal degree format.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The Paleobiology Database (downloaded on 2022-03-11). <br />
Data URL: http://paleobiodb.org/data1.2/occs/list.csv?datainfo&amp;rowcount&amp;base_name=Dinosauria&amp;show=full,classext,genus,subgenus,acconly,ident,img,etbasis,strat,lith,env,timebins,timecompare,resgroup,ref,ent,entname,crmod <br />
</p>

<hr>
<h2 id='estimateSpeciation'>Estimate speciation assuming a pure-birth process</h2><span id='topic+estimateSpeciation'></span>

<h3>Description</h3>

<p><code>estimateSpeciation</code> Estimates the speciation rate assuming a
constant-rate, pure-birth model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimateSpeciation(phy)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimateSpeciation_+3A_phy">phy</code></td>
<td>
<p>A <code>phylo</code> object, following terminology from package
<code>ape</code> in which function will operate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>numeric</code> with the estimated speciation rate.
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Yule G.U. 1925. A mathematical theory of evolution, based on the conclusions of Dr. JC Willis, FRS. Philosophical Transactions of the Royal Society of London. Series B, Containing Papers of a Biological Character. 213:21–87.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>S &lt;- 1
E &lt;- 0
set.seed(1)
phy &lt;- simulateTree(pars = c(S, E), max.taxa = 6, max.t = 5)
estimateSpeciation(phy)

</code></pre>

<hr>
<h2 id='fitCRBD'>Fit a constant-rate birth-death process to a phylogeny</h2><span id='topic+fitCRBD'></span>

<h3>Description</h3>

<p><code>fitCRBD</code> fits a constant-rate birth-death process to a phylogeny in the
format of <code>ape</code> package's <code>phylo</code> object. Optimization is based on
likelihood functions made with <code>diversitree</code>. This function is
basically a wrapper for the <code>diversitree</code>'s <code>make.bd</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitCRBD(phy, n.opt = 5, l.min = 0.001, l.max = 5, max.bad = 200)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitCRBD_+3A_phy">phy</code></td>
<td>
<p>A <code>phylo</code> object, following terminology from package
<code>ape</code>, in which function will operate.</p>
</td></tr>
<tr><td><code id="fitCRBD_+3A_n.opt">n.opt</code></td>
<td>
<p>Number of optimizations that will be tried by function.</p>
</td></tr>
<tr><td><code id="fitCRBD_+3A_l.min">l.min</code></td>
<td>
<p>Lower bound for optimization. Default value is <code>0.001</code>.</p>
</td></tr>
<tr><td><code id="fitCRBD_+3A_l.max">l.max</code></td>
<td>
<p>Upper bound for optimization. Default value is <code>5</code>.</p>
</td></tr>
<tr><td><code id="fitCRBD_+3A_max.bad">max.bad</code></td>
<td>
<p>Maximum number of unsuccessful optimization attempts. Default
value is <code>200</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>numeric</code> with the best estimates of speciation <code>S</code>
and extinction <code>E</code> rates.
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Paradis, E. (2012). Analysis of Phylogenetics and Evolution with R (Vol. 2).
New York: Springer.
</p>
<p>Popescu, A. A., Huber, K. T., &amp; Paradis, E. (2012). ape 3.0: New tools for
distance-based phylogenetics and evolutionary analysis in R. Bioinformatics,
28(11), 1536-1537.
</p>
<p>FitzJohn, R. G. (2010). Analysing diversification with diversitree.
R Package. ver, 9-2.
</p>
<p>FitzJohn, R. G. (2012). Diversitree: comparative phylogenetic analyses of
diversification in R. Methods in Ecology and Evolution, 3(6), 1084-1092.
</p>


<h3>See Also</h3>

<p>see help page from <code>diversitree::make.bd</code> and
<code>stats::optim</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
S &lt;- 0.1
E &lt;- 0.1
set.seed(1)
phy &lt;- simulateTree(pars = c(S, E), max.taxa = 30, max.t = 8)
fitCRBD(phy)

</code></pre>

<hr>
<h2 id='lttPlot'>Make a lineage through time (LTT) plot</h2><span id='topic+lttPlot'></span>

<h3>Description</h3>

<p><code>lttPlot</code> plots the lineage through time (LTT) of a <code>phylo</code> object.
It also adds a reference line connecting the edges of the graph.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lttPlot(
  phy,
  lwd = 1,
  col = "red",
  plot = TRUE,
  rel.time = FALSE,
  add = FALSE,
  knitr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lttPlot_+3A_phy">phy</code></td>
<td>
<p>A <code>phylo</code> object, as specified by the <code>ape</code> package.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_lwd">lwd</code></td>
<td>
<p>Line width.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_col">col</code></td>
<td>
<p>Line color.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_plot">plot</code></td>
<td>
<p>A <code>logical</code> indicating with calculations should be plotted.
If <code>FALSE</code>, function returns a list of the calculated points.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_rel.time">rel.time</code></td>
<td>
<p>A <code>logical</code> indicating how the time scale should be
shown. If <code>FALSE</code> (default), plots the absolute time since <code>phy</code>'s
crown age. If  <code>TRUE</code>, plots time as a relative proportion between
crown age and furthest tip from root.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_add">add</code></td>
<td>
<p>A <code>logical</code> indicating if plot should be added to
pre-existing plot. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="lttPlot_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots the sum of alive lineages per point in time, and adds a red
line as a reference of expectation under pure birth. If <code>plot = FALSE</code>,
a list the richness of each point in time, and <code>phy</code>'s crown age.
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Paradis, E. (2012). Analysis of Phylogenetics and Evolution with R (Vol. 2).
New York: Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
S &lt;- 1
E &lt;- 0
set.seed(1)
phy &lt;- simulateTree(pars = c(S, E), max.taxa = 20, max.t = 5)
lttPlot(phy, knitr = TRUE)
lttPlot(phy, plot = FALSE, knitr = TRUE)

</code></pre>

<hr>
<h2 id='NatSelSim'>Simulating natural selection through time in a bi-allelic gene</h2><span id='topic+NatSelSim'></span>

<h3>Description</h3>

<p><code>NatSelSim</code> simulates natural selection in a bi-allelic gene through
<code>n.gen</code> generations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NatSelSim(
  w11 = 1,
  w12 = 1,
  w22 = 0.9,
  p0 = 0.5,
  n.gen = 10,
  plot.type = "animateall",
  print.data = FALSE,
  knitr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NatSelSim_+3A_w11">w11</code></td>
<td>
<p>Number giving the fitness of genotype A1A1. Values will be
normalized if any genotype fitness exceeds one.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_w12">w12</code></td>
<td>
<p>Number giving the fitness of genotype A1A2. Values will be
normalized if any genotype fitness exceeds one.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_w22">w22</code></td>
<td>
<p>Number giving the fitness of genotype A2A2. Values will be
normalized if any genotype fitness exceeds one.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_p0">p0</code></td>
<td>
<p>Initial (time = 0) allelic frequency of A1.
A2's initial allelic frequency is <code>1-p0</code>.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_n.gen">n.gen</code></td>
<td>
<p>Number of generation that will be simulated.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_plot.type">plot.type</code></td>
<td>
<p>String indicating if plot should be animated.
The default, &quot;animateall&quot; animate all possible panels.
Other options are &quot;static&quot; (no animation), &quot;animate1&quot;, &quot;animate3&quot;, or
&quot;animate4&quot;. Users can animate each panel individually (using
<code>plot.type = "animateX"</code>, with X being the panel which one wants to
animate (so options are  &quot;animate1&quot;,  &quot;animate3&quot;,  and &quot;animate4&quot;
(see return for more info).</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_print.data">print.data</code></td>
<td>
<p>Logical indicating whether all
simulation results should be returned as a <code>data.frame</code>. Default value
is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="NatSelSim_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If any value of fitness (i.e., <code>w11</code>, <code>w12</code>,
<code>w22</code>) is larger than one, fitness is interpreted as absolute fitness
and values are re-normalized.
</p>


<h3>Value</h3>

<p>If <code>print.data = TRUE</code>, it returns a <code>data.frame</code>
containing the number of individuals for each genotype through time. The
plots done by the function shows (1) Allele frequency change through time.
(2) The adaptive landscape (which remains static during the whole simulation,
so can't be animated), (3) Time series of mean population fitness,
and (4) Time series of genotypic population frequencies.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Jennifer Auler, Dan Rabosky
</p>


<h3>References</h3>

<p>Fisher, R. A. (1930). The Fundamental Theorem of Natural Selection. In: The genetical theory of natural selection. The Clarendon Press
</p>
<p>Plutynski, A. (2006). What was Fisher’s fundamental theorem of natural selection and what was it for?. Studies in History and Philosophy of Science Part C: Studies in History and Philosophy of Biological and Biomedical Sciences, 37(1), 59-82.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#using the default values (w11=1, w12=1, w22=0.9, p0=0.5, n.gen=10)
NatSelSim()

# Continuing a simulation for extra time:
# Run the first simulation
sim1=NatSelSim(w11 = .4, w12 = .5, w22 = .4, p0 = 0.35, 
n.gen = 5, plot.type = "static", print.data = TRUE, knitr = TRUE)

# Then take the allelic frequency form the first sim:
new_p0 &lt;- (sim1$AA[nrow(sim1)] + sim1$Aa[nrow(sim1)]*1/2) 
# and use as p0 for a second one:

NatSelSim(w11 = .4, w12 = .5, w22 = .4, p0 = new_p0, n.gen = 5, plot.type = "static", knitr = TRUE)


</code></pre>

<hr>
<h2 id='OneGenHWSim'>Simulating one generation of genotypes under Hardy-Weinberg equilibrium</h2><span id='topic+OneGenHWSim'></span>

<h3>Description</h3>

<p><code>OneGenHWSim</code> creates <code>n.sim</code> simulations of one
generation of genotypes under Hardy-Weinberg equilibrium for a
bi-allelic loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OneGenHWSim(n.ind = 50, p = 0.5, n.sim = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="OneGenHWSim_+3A_n.ind">n.ind</code></td>
<td>
<p>Integer indicating the census size of the simulated populations.
If decimals are inserted, they will be rounded.</p>
</td></tr>
<tr><td><code id="OneGenHWSim_+3A_p">p</code></td>
<td>
<p>Numerical between zero and one that indicates A1's allele
frequency. A2's allele frequency is assumed to be <code>1-p</code>.</p>
</td></tr>
<tr><td><code id="OneGenHWSim_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations to be made. If decimals are inserted,
they will be rounded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> containing the number of individuals for each
genotype.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Dan Rabosky, Jennifer Auler
</p>


<h3>References</h3>

<p>Hardy, G. H. (1908). Mendelian proportions in a mixed
population. Science, 28, 49–50.
</p>
<p>Weinberg, W. (1908). Uber den Nachweis der Vererbung beim Menschen.
Jahreshefte des Vereins fur vaterlandische Naturkunde in Wurttemberg,
Stuttgart 64:369–382. [On the demonstration of inheritance in humans].
Translation by R. A. Jameson printed in D. L. Jameson (Ed.), (1977).
Benchmark papers in genetics, Volume 8: Evolutionary genetics (pp.
115–125). Stroudsburg, PA: Dowden, Hutchinson &amp; Ross.
</p>
<p>Mayo, O. (2008). A century of Hardy–Weinberg equilibrium. Twin Research
and Human Genetics, 11(3), 249-256.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#using the default values (n.ind = 50, p = 0.5, n.sim = 100):
OneGenHWSim()

#Simulating with a already fixed allele:
OneGenHWSim(n.ind = 50, p = 1)

# Testing if the simulation works:
A1freq &lt;- .789 #any value could work
n.simul &lt;- 100
simulations &lt;- OneGenHWSim(n.ind = n.simul, n.sim = n.simul, p = A1freq)

#expected:
c(A1freq^2, 2*A1freq*(1-A1freq), (1-A1freq)^2)

#simulated:
apply(X = simulations, MARGIN = 2, FUN = function(x){mean(x)/n.simul})

</code></pre>

<hr>
<h2 id='plotNatSel'>Plot NatSelSim output</h2><span id='topic+plotNatSel'></span>

<h3>Description</h3>

<p>Plot NatSelSim output
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNatSel(
  gen.HW = gen.HW,
  p.t = p.t,
  w.t = w.t,
  t = t,
  W.gntp = c(w11, w12, w22),
  plot.type = "animateall",
  knitr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotNatSel_+3A_gen.hw">gen.HW</code></td>
<td>
<p>Dataframe with A1A1, A1A2 and A2A2 genotypic
frequencies in each generation (nrows = NGen)</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_p.t">p.t</code></td>
<td>
<p>Allelic frequency through time</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_w.t">w.t</code></td>
<td>
<p>Mean population fitness through time</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_t">t</code></td>
<td>
<p>time</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_w.gntp">W.gntp</code></td>
<td>
<p>Initial genotypic fitness</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_plot.type">plot.type</code></td>
<td>
<p>String indicating if plot should be animated. The default, &quot;animateall&quot;, animate all possible panels. Other options are &quot;static&quot;, &quot;animate1&quot;, &quot;animate3&quot;, or &quot;animate4&quot;.</p>
</td></tr>
<tr><td><code id="plotNatSel_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot of NatSelSim's output (see <code>NatSelSim</code>'s help page for
details).
</p>

<hr>
<h2 id='plotPaintedWhales'>Plotting the whale phylogeny and coloring its clades</h2><span id='topic+plotPaintedWhales'></span>

<h3>Description</h3>

<p><code>plotPaintedWhales</code> plots the phylogeny from Steeman et al (2011), coloring the Dolphins (Delphinidae), porpoises (Phocoenidae), the Mysticetes, the baleen whales (Balaenopteridae), and the Beaked whales (Ziphiidae).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPaintedWhales(
  show.legend = TRUE,
  direction = "rightwards",
  knitr = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPaintedWhales_+3A_show.legend">show.legend</code></td>
<td>
<p>Logical indicating if clade legend should be shown.</p>
</td></tr>
<tr><td><code id="plotPaintedWhales_+3A_direction">direction</code></td>
<td>
<p>Phylogeny plotting direction. Should be set to &quot;rightwards&quot;</p>
</td></tr>
<tr><td><code id="plotPaintedWhales_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.
(the default) or &quot;leftwards&quot;.</p>
</td></tr>
<tr><td><code id="plotPaintedWhales_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code>phytools::plotSimmap</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The whale phylogeny, with branch lengths being colored by a major
whale taxonomic group.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Steeman, M. E., Hebsgaard, M. B., Fordyce, R. E., Ho, S. Y., Rabosky, D. L., Nielsen, R., ... &amp; Willerslev, E. (2009). Radiation of extant cetaceans driven by restructuring of the oceans. Systematic biology, 58(6), 573-585.
</p>


<h3>See Also</h3>

<p>help page from <code>phytools::plotSimmap</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plotPaintedWhales(knitr = TRUE)

</code></pre>

<hr>
<h2 id='plotProteinSeq'>Plot protein sequence(s)</h2><span id='topic+plotProteinSeq'></span>

<h3>Description</h3>

<p><code>plotProteinSeq</code> draws the sequences of proteins within the same &quot;ProteinSeq&quot; object. In this format, more similar sequences will have
similar banding patterns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProteinSeq(x, taxon.to.plot, knitr = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotProteinSeq_+3A_x">x</code></td>
<td>
<p>A &quot;ProteinSeq&quot; object containing proteins from <code>taxon1</code>
and <code>taxon2</code>.</p>
</td></tr>
<tr><td><code id="plotProteinSeq_+3A_taxon.to.plot">taxon.to.plot</code></td>
<td>
<p>A character vector providing the common name of the species that will be plotted. Must be a name present in <code>x</code>.</p>
</td></tr>
<tr><td><code id="plotProteinSeq_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A draw of the protein sequence(s) provided. Colors refer to
specific amino acids (&quot;R&quot;, &quot;W&quot;, &quot;I&quot;, &quot;F&quot;, &quot;S&quot;, &quot;T&quot;, &quot;N&quot;, &quot;H&quot;, &quot;K&quot;, &quot;D&quot;, &quot;G&quot;, &quot;L&quot;, &quot;Y&quot;, &quot;V&quot;, &quot;M&quot;, &quot;A&quot;, &quot;E&quot;, &quot;P&quot;, &quot;Q&quot;, &quot;C&quot;)&quot;, &quot;gaps/space in the sequence (&quot;-&quot;), ambiguous amino acid (&quot;B&quot; - often representing either asparagine (&quot;N&quot;) or aspartic acid (&quot;D&quot;)), or another marker for  ambiguous amino acid (&quot;X&quot;).
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Jennifer Auler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cytOxidase)
plotProteinSeq(cytOxidase, c("human", "chimpanzee", "cnidaria"), knitr = TRUE)

</code></pre>

<hr>
<h2 id='plotRawFossilOccs'>Plot a literal interpretation of a fossil record</h2><span id='topic+plotRawFossilOccs'></span>

<h3>Description</h3>

<p><code>plotRawFossilOccs</code> calculates and plots the early and late boundaries
associated with each taxa in a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRawFossilOccs(
  data,
  tax.lvl = NULL,
  sort = TRUE,
  use.midpoint = TRUE,
  return.ranges = FALSE,
  knitr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotRawFossilOccs_+3A_data">data</code></td>
<td>
<p>A <code>data.frame</code> containing fossil data on the age (early and
late bounds of rock layer, respectively labeled as <code>max_ma</code> and
<code>min_ma</code>) and the taxonomic level asked in <code>tax_lv</code>.</p>
</td></tr>
<tr><td><code id="plotRawFossilOccs_+3A_tax.lvl">tax.lvl</code></td>
<td>
<p>A <code>character</code> giving the taxonomic in which
calculations will be based on, which must refer to the column names in
<code>data</code>. If <code>NULL</code> (default value), the function plots every
individual occurrences in <code>data</code>.</p>
</td></tr>
<tr><td><code id="plotRawFossilOccs_+3A_sort">sort</code></td>
<td>
<p><code>logical</code> indicating if taxa should be sorted by their
<code>max_ma</code> values (default value is <code>TRUE</code>). Otherwise (i.e., if
<code>FALSE</code>), function will follow the order of taxa (or occurrences)
inputted in <code>data</code>.</p>
</td></tr>
<tr><td><code id="plotRawFossilOccs_+3A_use.midpoint">use.midpoint</code></td>
<td>
<p><code>logical</code> indicating if function should use
occurrence midpoints (between <code>max_ma</code> and <code>min_ma</code>) as
occurrence temporal boundaries, a method commonly employed in paleobiology
to remove noise related to extremely coarse temporal resolution due to
stratification. This argument is only used if a <code>tax.lvl</code> is provided.</p>
</td></tr>
<tr><td><code id="plotRawFossilOccs_+3A_return.ranges">return.ranges</code></td>
<td>
<p><code>logical</code> indicating if ranges calculated by
function should be return as a <code>data.frame</code>. If <code>tax.lvl</code> is
<code>NULL</code>, the function don't calculate ranges and so it has nothing
to return.</p>
</td></tr>
<tr><td><code id="plotRawFossilOccs_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a pile of the max-min temporal ranges of the chosen
<code>tax.lvl</code>. This usually will be stratigraphic ranges for occurrences
(so there is no attempt to estimate &quot;true&quot; ranges), and if
<code>tax.lvl = NULL</code> (the default), occurrences are drawn as ranges of
stratigraphic resolution (= the fossil dating imprecision). If
<code>return.ranges = TRUE</code>, it returns a <code>data.frame</code> containing the
diversity (column <code>div</code>) of the chosen taxonomic level, through time.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Jennifer Auler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("dinos_fossil")
oldpar &lt;- par(no.readonly = TRUE) 
par(mfrow=c(1,2))
plotRawFossilOccs(dinos_fossil, tax.lvl = "species", knitr = TRUE)
plotRawFossilOccs(dinos_fossil, tax.lvl = "genus", knitr = TRUE)
par(oldpar)

</code></pre>

<hr>
<h2 id='plotWFDrift'>Plot WFDriftSim output</h2><span id='topic+plotWFDrift'></span>

<h3>Description</h3>

<p>Plot WFDriftSim output
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotWFDrift(p.through.time, plot.type = plot, knitr = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotWFDrift_+3A_p.through.time">p.through.time</code></td>
<td>
<p>Matrix with n.gen columns and n.sim lines</p>
</td></tr>
<tr><td><code id="plotWFDrift_+3A_plot.type">plot.type</code></td>
<td>
<p>String. Options are &quot;static&quot; or &quot;animate&quot;</p>
</td></tr>
<tr><td><code id="plotWFDrift_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A static or animated plot of populations under genetic drift through time
</p>


<h3>Examples</h3>

<pre><code class='language-R'>store_p = WFDriftSim(Ne = 5, n.gen = 10, p0=.2, n.sim=5, plot = "none", print.data = TRUE)
plotWFDrift(store_p, "static")
</code></pre>

<hr>
<h2 id='ProteinSeq'>Details, generics, and methods for the <code>ProteinSeq</code> class</h2><span id='topic+ProteinSeq'></span><span id='topic+is.ProteinSeq'></span><span id='topic+print.ProteinSeq'></span><span id='topic+summary.ProteinSeq'></span><span id='topic+head.ProteinSeq'></span><span id='topic+tail.ProteinSeq'></span>

<h3>Description</h3>

<p>The <code>ProteinSeq</code> class is an input for the functions <code>countSeqDiffs</code> and <code>is.ProteinSeq</code>. It consists of a character vector. Each entry in this vector represents the aminoacid (the protein components coded by a gene) sequence, for a given aligned protein sequence. The object must be a character, named vector, with the names typically corresponding to the species (name could be scientific or common name) from which every sequence came. The characters within the vector must correspond to valid aminoacid symbols (i.e. capitalized letters or deletion &quot;_&quot; symbols). Particularly, the following symbols relate to amino acids: <code>"A", "C", "D", "E", "F", "G", "H", "I", "K", "L", "M", "N", "P", "Q", "R", "S", "T", "V", "W", "Y"</code>.
</p>
<p>Importantly, the symbol <code>"_"</code> means an indel (insertion or deletion), and the symbols <code>"X", "B", "Z", "J"</code> should be considered as ambiguous site readings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.ProteinSeq(x)

## S3 method for class 'ProteinSeq'
print(x, ...)

## S3 method for class 'ProteinSeq'
summary(object, ...)

## S3 method for class 'ProteinSeq'
head(x, n = 20, ...)

## S3 method for class 'ProteinSeq'
tail(x, n = 20, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ProteinSeq_+3A_x">x</code></td>
<td>
<p>an object of the class <code>ProteinSeq</code></p>
</td></tr>
<tr><td><code id="ProteinSeq_+3A_...">...</code></td>
<td>
<p>arguments to be passed to or from other methods.
</p>
<p>#' @return Shows the last <code>n</code> elements of a <code>ProteinSeq</code> object.</p>
</td></tr>
<tr><td><code id="ProteinSeq_+3A_object">object</code></td>
<td>
<p>an object of the class <code>ProteinSeq</code></p>
</td></tr>
<tr><td><code id="ProteinSeq_+3A_n">n</code></td>
<td>
<p>number of aminoacids to be shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>is.ProteinSeq</code> A <code>ProteinSeq</code> must be a list containing
multiple vectors made of characters (usually letters that code to Amino Acids,
deletions, etc). All of these must have the correct length (i.e. same as all
the others) and their relative positions should match (i.e., the object must
contain <em>alligned</em> Amino acide sequences).
</p>


<h3>Value</h3>

<p>A logical indicating if object <code>x</code> is of
class <code>ProteinSeq</code>
</p>
<p><code>print.ProteinSeq</code> Prints a brief summary of a
<code>print.ProteinSeq</code> containing the number of sequences and
the length of the alignment. See more details of the format in
<code>??ProteinSeq</code>.
</p>
<p>Same as <code>print.ProteinSeq</code>.
</p>
<p>Shows the first <code>n</code> elements of a <code>ProteinSeq</code> object.
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>

<hr>
<h2 id='simulateBirthDeathRich'>Simulating richness through birth-death processes</h2><span id='topic+simulateBirthDeathRich'></span>

<h3>Description</h3>

<p><code>simulateBirthDeathRich</code> calculates the number of species at a certain
point in time, following a birth-death process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateBirthDeathRich(t, S = NULL, E = NULL, K = NULL, R = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateBirthDeathRich_+3A_t">t</code></td>
<td>
<p>Point in time which richness will be simulated.</p>
</td></tr>
<tr><td><code id="simulateBirthDeathRich_+3A_s">S</code></td>
<td>
<p>A numeric representing the per-capita speciation rate (in number
of events per lineage per million years). Must be larger than <code>E</code>.</p>
</td></tr>
<tr><td><code id="simulateBirthDeathRich_+3A_e">E</code></td>
<td>
<p>A numeric representing the per-capita extinction rate (in number
of events per lineage per million years). Must be smaller than <code>S</code>.</p>
</td></tr>
<tr><td><code id="simulateBirthDeathRich_+3A_k">K</code></td>
<td>
<p>A numeric representing the extinction fraction (i.e.,
<code>K</code> = <code>E</code> / <code>S</code>). Must be either zero or a positive
which is number smaller than one.</p>
</td></tr>
<tr><td><code id="simulateBirthDeathRich_+3A_r">R</code></td>
<td>
<p>A numeric representing the per-capita Net Diversification
rate (i.e., <code>R</code> = <code>S</code> - <code>E</code>). Must be a positive number.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function only accepts as inputs <code>S</code> and <code>E</code>, or
<code>K</code> and <code>R</code>.
</p>


<h3>Value</h3>

<p>The number of simulated species (i.e., the richness).
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Daniel Rabosky, Jennifer Auler
</p>


<h3>References</h3>

<p>Raup, D. M. (1985). Mathematical models of cladogenesis.
Paleobiology, 11(1), 42-52.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># running a single simulation:
SS &lt;- 0.40
EE &lt;- 0.09
tt &lt;- 10 #in Mya
simulateBirthDeathRich(t = tt, S = SS, E = EE)

#running many simulations and graphing results:
nSim &lt;- 1000
res &lt;- vector()
for(i in 1:nSim){
  res &lt;- c(res, 
  simulateBirthDeathRich(t = tt, S = SS, E = EE))
}
plot(table(res)/length(res),
     xlab="Richness", ylab="Probability")
</code></pre>

<hr>
<h2 id='simulateTree'>Simulating a phylogenetic trees through the birth-death process</h2><span id='topic+simulateTree'></span>

<h3>Description</h3>

<p><code>simulateTree</code> uses a birth-death process to simulate a phylogenetic
tree, following the format of <code>ape</code> package's <code>phylo</code> object. The
function is basically a wrapper for the <code>diversitree</code>'s <code>tree.bd</code>
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateTree(
  pars,
  max.taxa = Inf,
  max.t,
  min.taxa = 2,
  include.extinct = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateTree_+3A_pars">pars</code></td>
<td>
<p><code>numeric</code> vector with the simulation parameters: speciation
(first slot) and extinction (second slot) rates, respectively. Should follow
any formats stated in the function <code>tree.bd</code> from the
<code>diversitree</code> package.</p>
</td></tr>
<tr><td><code id="simulateTree_+3A_max.taxa">max.taxa</code></td>
<td>
<p>Maximum number of taxa to include in the tree. If
<code>Inf</code>, then the tree will be evolved until <code>max.t</code> time has passed.</p>
</td></tr>
<tr><td><code id="simulateTree_+3A_max.t">max.t</code></td>
<td>
<p>Maximum length to evolve the phylogeny over. If equal to
<code>Inf</code>, then the tree will evolve until <code>max.taxa</code>
extant taxa are present.</p>
</td></tr>
<tr><td><code id="simulateTree_+3A_min.taxa">min.taxa</code></td>
<td>
<p>Minimum number of taxa to include in the tree.</p>
</td></tr>
<tr><td><code id="simulateTree_+3A_include.extinct">include.extinct</code></td>
<td>
<p>A <code>logical</code> indicating if extinct taxa should
be included in the final phylogeny.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>see help page from <code>diversitree::tree.bd</code>
</p>


<h3>Value</h3>

<p>A <code>phylo</code> object
</p>


<h3>Author(s)</h3>

<p>Daniel Rabosky, Matheus Januario, Jennifer Auler
</p>


<h3>References</h3>

<p>Paradis, E. (2012). Analysis of Phylogenetics and Evolution with R (Vol. 2).
New York: Springer.
</p>
<p>Popescu, A. A., Huber, K. T., &amp; Paradis, E. (2012). ape 3.0: New tools for
distance-based phylogenetics and evolutionary analysis in R. Bioinformatics,
28(11), 1536-1537.
</p>
<p>FitzJohn, R. G. (2010). Analysing diversification with diversitree.
R Packag. ver, 9-2.
</p>
<p>FitzJohn, R. G. (2012). Diversitree: comparative phylogenetic analyses of
diversification in R. Methods in Ecology and Evolution, 3(6), 1084-1092.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
S &lt;- 1
E &lt;- 0
set.seed(1)
phy &lt;- simulateTree(pars = c(S, E), max.taxa = 6, max.t=Inf)
ape::plot.phylo(phy)
ape::axisPhylo()

# alternatively, we can stop the simulation using time:
set.seed(42)
phy2 &lt;- simulateTree(pars = c(S, E), max.t=7)
ape::plot.phylo(phy2)
ape::axisPhylo()

</code></pre>

<hr>
<h2 id='timeseries_fossil'>Fossil Time series</h2><span id='topic+timeseries_fossil'></span>

<h3>Description</h3>

<p>Values of clade diversity for many clades of organisms (note some clades
are nested within other clades in the dataset). This dataset is part of the
package and is licensed =under the Creative Commons Attribution 4.0
International License (CC BY 4.0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeseries_fossil
</code></pre>


<h3>Format</h3>

<p>A <code>data.frame</code> with 598 rows and 6 columns.
</p>

<dl>
<dt>clade</dt><dd><p>Time series clade</p>
</dd>
<dt>source</dt><dd><p>Primary source of the Time series</p>
</dd>
<dt>stem_age</dt><dd><p>Stem age of clade</p>
</dd>
<dt>rel_time</dt><dd><p>Geological relative time (in Million years ago relative to present)</p>
</dd>
<dt>time_ma</dt><dd><p>Geological time in million years since clade stem age</p>
</dd>
<dt>richness</dt><dd><p>Number of species at given geological time</p>
</dd>
</dl>



<h3>Details</h3>

<p>Legend: <br />
anth = Anthozoa (Cnidaria); <br />
art = Articulata (Crinoidea, Echinodermata); <br />
biv = Bivalvia (Mollusca); <br />
bryo = Bryozoa (Lophotrochozoa, Ectoprocta); <br />
ceph = Cephalopoda (Mollusca); <br />
chon = Chondrocytes (Chordata); <br />
crin = Crinoidea (Echinodermata); <br />
dinosauria = Dinosauria (Chordata); <br />
ech = Echinoidea (Echinodermata); <br />
foram = Foraminifera (Retaria); <br />
gast = Gastropoda (Mollusca); <br />
graptoloids = Graptolites (Graptolithina); <br />
ling = Ligulata (Brachiopoda); <br />
ostr = Ostracoda (Crustacea, Arthropoda); <br />
tril = Trilobita (Arthropoda). <br />
</p>


<h3>Source</h3>

<p>Data originally compiled from many primary sources. Organized, curated by, and downloaded from, Rabosky &amp; Benson (2021).
</p>


<h3>References</h3>

<p>Rabosky, D. L., &amp; Benson, R. B. (2021). Ecological and biogeographic drivers of biodiversity cannot be resolved using clade age-richness data. Nature Communications, 12(1), 2945.
</p>

<hr>
<h2 id='WFDriftSim'>Simulating generations of genetic drift in a Wright–Fisher (WF) population</h2><span id='topic+WFDriftSim'></span>

<h3>Description</h3>

<p><code>WFDriftSim</code> simulates genetic drift of diploid Wright–Fisher
populations with a given effective population size through a certain number of
generations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WFDriftSim(
  Ne,
  n.gen,
  p0 = 0.5,
  n.sim = 1,
  plot.type = "animate",
  print.data = FALSE,
  knitr = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WFDriftSim_+3A_ne">Ne</code></td>
<td>
<p>Number giving the effective population size of the population</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_n.gen">n.gen</code></td>
<td>
<p>Number of generations to be simulated.</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_p0">p0</code></td>
<td>
<p>Initial frequency of a given allele. As the simulated organism is
diploid, the other alleles frequency will be <code>1-(p0)</code>. Default value
is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of simulations to be made. If decimals are inserted,
they will be rounded. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_plot.type">plot.type</code></td>
<td>
<p>Character indicating if simulations should be plotted as colored
lines. Each color represents a different population. If
<code>plot.type = "animate"</code> (default value) it animates each generation
individually. If <code>plot.type = "static"</code> it plots all lines rapidly. If
<code>plot.type = "none"</code> nothing is plotted.</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_print.data">print.data</code></td>
<td>
<p>Logical indicating whether all simulation results should be
returned as a <code>data.frame</code>. Default value is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="WFDriftSim_+3A_knitr">knitr</code></td>
<td>
<p>Logical indicating if plot is intended to show up in RMarkdown files made by the <code>Knitr</code> R package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The effective population size (<code>Ne</code>) is strongly connected
with the rate of genetic drift (for details, see Waples, 2022).
</p>


<h3>Value</h3>

<p>If <code>plot.type = "static"</code> or <code>"animate"</code>, plots the
timeseries of all simulations, with each line+color referring to a
different simulation. Note that if many simulations (generally more
than 20) are simulated, colors might be cycled and different simulation
will have the same color. If <code>print.data = TRUE</code>, returns a
<code>data.frame</code> with the simulation results.
</p>


<h3>Author(s)</h3>

<p>Matheus Januario, Dan Rabosky, Jennifer Auler
</p>


<h3>References</h3>

<p>Fisher RA (1922) On the dominance ratio. Proc. R. Soc. Edinb 42:321–341
</p>
<p>Kimura M (1955) Solution of a process of random genetic drift with a
continuous model. PNAS–USA 41(3):144–150
</p>
<p>Tran, T. D., Hofrichter, J., &amp; Jost, J. (2013). An introduction to the
mathematical structure of the Wright–Fisher model of population genetics.
Theory in Biosciences, 132(2), 73-82. [good for the historical review,
math can be challenging]
</p>
<p>Waples, R. S. (2022). What is Ne, anyway?. Journal of Heredity.
</p>
<p>Wright S (1931) Evolution in Mendelian populations. Genetics 16:97–159
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Default values:
WFDriftSim(Ne = 5, n.gen = 10, knitr = TRUE)

#A population which has already fixed one of the alleles:
WFDriftSim(Ne = 5, n.gen = 10, p0=1, knitr = TRUE)

#Many populations::
WFDriftSim(Ne = 5, n.gen = 10, p0=0.2, n.sim=10, knitr = TRUE)

######## continuing a previous simulation:
n.gen_1stsim &lt;- 10 # number of gens in the 1st sim:
sim1 &lt;- WFDriftSim(Ne = 5, n.gen = n.gen_1stsim, p0=.2, n.sim=10, 
plot.type = "none", print.data = TRUE, knitr = TRUE)
n.gen_2ndsim &lt;-7 # number of gens in the 2nd sim:
# now, note how we assigned p0:
sim2 &lt;- WFDriftSim(Ne = 5, n.gen = n.gen_2ndsim, p0=sim1[,ncol(sim1)], 
plot.type = "static", n.sim=10, print.data = TRUE, knitr = TRUE)

# if we want to merge both simulations, then we have to:
# remove first column of 2nd sim (because it repeats
# the last column of the 1st sim)
sim2 &lt;- sim2[,-1]

# re-name 2nd sim columns:
colnames(sim2) &lt;- paste0("gen", (n.gen_1stsim+1):(n.gen_1stsim+n.gen_2ndsim))

#finally, merging both rounds of simulations:
all_sims &lt;- cbind(sim1, sim2)
head(all_sims)


</code></pre>

<hr>
<h2 id='whale_phylo'>Whale Phylogeny</h2><span id='topic+whale_phylo'></span>

<h3>Description</h3>

<p>An ultrametric phylogenetic tree of the living cetaceans. Phylogeny
generated by Steeman et al (2009). This dataset is part of the package and is licensed
under the Creative Commons Attribution 4.0 International License (CC BY 4.0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>whale_phylo
</code></pre>


<h3>Format</h3>

<p>An ultrametric <code>phylo</code> object with 87 tips
</p>


<h3>Source</h3>

<p>Original phylogeny generation by Steeman et al (2001).
File obtained from Rabosky et al, 2014.
</p>


<h3>References</h3>

<p>Rabosky, D. L., Grundler, M., Anderson, C., Title, P., Shi, J. J., Brown, J. W., ... &amp; Larson, J. G. (2014). BAMM tools: an R package for the analysis of evolutionary dynamics on phylogenetic trees. Methods in Ecology and Evolution, 5(7), 701-707.
</p>
<p>Steeman, M. E., Hebsgaard, M. B., Fordyce, R. E., Ho, S. Y., Rabosky, D. L., Nielsen, R., ... &amp; Willerslev, E. (2009). Radiation of extant cetaceans driven by restructuring of the oceans. Systematic biology, 58(6), 573-585.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
