<!DOCTYPE html><html lang="en"><head><title>Help for package expDB</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {expDB}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dbAddDesigns'><p>Add design for a trial</p></a></li>
<li><a href='#dbAddFertilization'><p>Insert or Update fertilization into expDB</p></a></li>
<li><a href='#dbAddGene'><p>Add gene information into database</p></a></li>
<li><a href='#dbAddGeneAllele'><p>Add gene allele information into database</p></a></li>
<li><a href='#dbAddGenotype'><p>Add genotypes into expDB</p></a></li>
<li><a href='#dbAddIrrigatons'><p>Insert or Update irrigation into expDB</p></a></li>
<li><a href='#dbAddLog'><p>Add log from expDB</p></a></li>
<li><a href='#dbAddMethods'><p>Insert or Update methods into expDB</p></a></li>
<li><a href='#dbAddMets'><p>Insert and update met into expDB</p></a></li>
<li><a href='#dbAddNodes'><p>Add nodes into expDB</p></a></li>
<li><a href='#dbAddPhenotype'><p>Add design for a trial</p></a></li>
<li><a href='#dbAddResearcher'><p>Insert and update researcher into expDB</p></a></li>
<li><a href='#dbAddSites'><p>Insert or update site into expDB</p></a></li>
<li><a href='#dbAddSource'><p>Insert and update source into expDB</p></a></li>
<li><a href='#dbAddTraits'><p>Insert or update trait into expDB</p></a></li>
<li><a href='#dbAddTrials'><p>Insert or Update trial into expDB</p></a></li>
<li><a href='#dbAddTrialSoil'><p>Add soil for a trial</p></a></li>
<li><a href='#dbAddWeather'><p>Add weather records into expDB</p></a></li>
<li><a href='#dbAppendTable'><p>Append a table into db and check the column name</p></a></li>
<li><a href='#dbExportMet'><p>Export trials weather records to met file</p></a></li>
<li><a href='#dbGenotypeCheckName'><p>Check genotype names</p></a></li>
<li><a href='#dbGetDryWeightPerStem'><p>Get the dry weight per stem</p></a></li>
<li><a href='#dbGetFertilization'><p>Get fertilization from database</p></a></li>
<li><a href='#dbGetFieldMaturity'><p>Estimation of maturity</p></a></li>
<li><a href='#dbGetFieldPopulation'><p>Estimation of plant populations</p></a></li>
<li><a href='#dbGetFieldStemNumber'><p>Estimation of stem number per unit area</p></a></li>
<li><a href='#dbGetFieldTillerNumber'><p>Estimation of tiller number per unit area</p></a></li>
<li><a href='#dbGetGene'><p>Get the gene information</p></a></li>
<li><a href='#dbGetGenotype'><p>Get the genotype information</p></a></li>
<li><a href='#dbGetIrrigation'><p>Get irrigation from database</p></a></li>
<li><a href='#dbGetLog'><p>Get log from expDB</p></a></li>
<li><a href='#dbGetMetInfo'><p>Get met information</p></a></li>
<li><a href='#dbGetOrganFinalLeafNumber'><p>Get the final leaf number</p></a></li>
<li><a href='#dbGetOrganHaunIndex'><p>Get Haun Index</p></a></li>
<li><a href='#dbGetPhenotype'><p>Get phenotype values through a group of conditions</p></a></li>
<li><a href='#dbGetPlantFlowering'><p>Estimation of flowering time</p></a></li>
<li><a href='#dbGetPlantHeading'><p>Estimation of heading time</p></a></li>
<li><a href='#dbGetPlantStemElongation'><p>Estimation of stem elongation</p></a></li>
<li><a href='#dbGetPlantStemNumber'><p>Estimation of stem number per plant</p></a></li>
<li><a href='#dbGetPlantTillerNumber'><p>Estimation of tiller number per plant</p></a></li>
<li><a href='#dbGetSites'><p>Get site into expDB</p></a></li>
<li><a href='#dbGetSource'><p>Get source from expDB</p></a></li>
<li><a href='#dbGetTraits'><p>Get trait list</p></a></li>
<li><a href='#dbGetTrials'><p>Get trials by a groups of conditions.</p></a></li>
<li><a href='#dbGetWeather'><p>Get weather records from expDB</p></a></li>
<li><a href='#dbGetZadoksStage'><p>Obtain the key phenology stage</p></a></li>
<li><a href='#dbImportXLSX'><p>Import data from excel file</p></a></li>
<li><a href='#dbInsertUpdateByRow'><p>Insert new rows or update existing rows to a specific table according to a specific column (unique) by each row</p></a></li>
<li><a href='#dbListTrials'><p>List all trials</p></a></li>
<li><a href='#expdbConnect'><p>Connect to expDB</p></a></li>
<li><a href='#expdbCreateDB'><p>create to expDB</p></a></li>
<li><a href='#expdbDisconnect'><p>Didconnect to expDB</p></a></li>
<li><a href='#getIdByUniqueIndex'><p>Get index id by unique_columns.</p></a></li>
<li><a href='#harvestQuadratDetail'><p>Process quadrat (detail) harvestQuadratDetail</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Database for Experiment Dataset</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bangyou Zheng &lt;bangyou.zheng@csiro.au&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A SQLite database is designed to store all information 
  of experiment-based data including metadata, experiment design, 
  managements, phenotypic values and climate records. The dataset can be
  imported from an excel file.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://expdb.bangyou.me/">https://expdb.bangyou.me/</a>, <a href="https://github.com/byzheng/expdb">https://github.com/byzheng/expdb</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/byzheng/expdb/issues">https://github.com/byzheng/expdb/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>DBI, RSQLite, utils, readxl, png, grid, tibble, lubridate,
dplyr, methods, weaana (&ge; 0.1.1), tidyr, tidyverse, magrittr,
stats, reshape2, stringr, rlang</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-08 05:00:13 UTC; zhe00a</td>
</tr>
<tr>
<td>Author:</td>
<td>Bangyou Zheng [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-08 17:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='dbAddDesigns'>Add design for a trial</h2><span id='topic+dbAddDesigns'></span>

<h3>Description</h3>

<p>Add design for a trial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddDesigns(con, data, extra_design = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddDesigns_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddDesigns_+3A_data">data</code></td>
<td>
<p>Trial design</p>
</td></tr>
<tr><td><code id="dbAddDesigns_+3A_extra_design">extra_design</code></td>
<td>
<p>The extra columns for design</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddFertilization'>Insert or Update fertilization into expDB</h2><span id='topic+dbAddFertilization'></span>

<h3>Description</h3>

<p>Insert or Update fertilization into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddFertilization(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddFertilization_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddFertilization_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddGene'>Add gene information into database</h2><span id='topic+dbAddGene'></span>

<h3>Description</h3>

<p>Add gene information into database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddGene(con, genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddGene_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddGene_+3A_genes">genes</code></td>
<td>
<p>A data.frame of genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return values
</p>

<hr>
<h2 id='dbAddGeneAllele'>Add gene allele information into database</h2><span id='topic+dbAddGeneAllele'></span>

<h3>Description</h3>

<p>Add gene allele information into database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddGeneAllele(con, genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddGeneAllele_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddGeneAllele_+3A_genes">genes</code></td>
<td>
<p>A data.frame of genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return values
</p>

<hr>
<h2 id='dbAddGenotype'>Add genotypes into expDB</h2><span id='topic+dbAddGenotype'></span>

<h3>Description</h3>

<p>Add genotypes into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddGenotype(con, genotypes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddGenotype_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddGenotype_+3A_genotypes">genotypes</code></td>
<td>
<p>A string vector of genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return values
</p>

<hr>
<h2 id='dbAddIrrigatons'>Insert or Update irrigation into expDB</h2><span id='topic+dbAddIrrigatons'></span>

<h3>Description</h3>

<p>Insert or Update irrigation into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddIrrigatons(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddIrrigatons_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddIrrigatons_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddLog'>Add log from expDB</h2><span id='topic+dbAddLog'></span>

<h3>Description</h3>

<p>Add log from expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddLog(con, msg, date = format(Sys.time(), format = "%Y-%m-%d"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddLog_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddLog_+3A_msg">msg</code></td>
<td>
<p>Add message into expdb</p>
</td></tr>
<tr><td><code id="dbAddLog_+3A_date">date</code></td>
<td>
<p>Create time of message</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return values
</p>

<hr>
<h2 id='dbAddMethods'>Insert or Update methods into expDB</h2><span id='topic+dbAddMethods'></span>

<h3>Description</h3>

<p>Insert or Update methods into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddMethods(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddMethods_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddMethods_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddMets'>Insert and update met into expDB</h2><span id='topic+dbAddMets'></span>

<h3>Description</h3>

<p>Insert and update met into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddMets(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddMets_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddMets_+3A_data">data</code></td>
<td>
<p>Met design</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddNodes'>Add nodes into expDB</h2><span id='topic+dbAddNodes'></span>

<h3>Description</h3>

<p>Add nodes into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddNodes(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddNodes_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddNodes_+3A_data">data</code></td>
<td>
<p>phenotype value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddPhenotype'>Add design for a trial</h2><span id='topic+dbAddPhenotype'></span>

<h3>Description</h3>

<p>Add design for a trial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddPhenotype(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddPhenotype_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddPhenotype_+3A_data">data</code></td>
<td>
<p>phenotype value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddResearcher'>Insert and update researcher into expDB</h2><span id='topic+dbAddResearcher'></span>

<h3>Description</h3>

<p>Insert and update researcher into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddResearcher(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddResearcher_+3A_con">con</code></td>
<td>
<p>A connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddResearcher_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddSites'>Insert or update site into expDB</h2><span id='topic+dbAddSites'></span>

<h3>Description</h3>

<p>Insert or update site into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddSites(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddSites_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddSites_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddSource'>Insert and update source into expDB</h2><span id='topic+dbAddSource'></span>

<h3>Description</h3>

<p>Insert and update source into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddSource(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddSource_+3A_con">con</code></td>
<td>
<p>A connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddSource_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddTraits'>Insert or update trait into expDB</h2><span id='topic+dbAddTraits'></span>

<h3>Description</h3>

<p>Insert or update trait into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddTraits(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddTraits_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddTraits_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddTrials'>Insert or Update trial into expDB</h2><span id='topic+dbAddTrials'></span>

<h3>Description</h3>

<p>Insert or Update trial into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddTrials(con, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddTrials_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddTrials_+3A_data">data</code></td>
<td>
<p>A data frame includes all columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddTrialSoil'>Add soil for a trial</h2><span id='topic+dbAddTrialSoil'></span>

<h3>Description</h3>

<p>Add soil for a trial
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddTrialSoil(
  con,
  data,
  units = list(thickness = "cm", no3 = "kg/ha", nh4 = "kg/ha")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddTrialSoil_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddTrialSoil_+3A_data">data</code></td>
<td>
<p>Soil profiles for trials</p>
</td></tr>
<tr><td><code id="dbAddTrialSoil_+3A_units">units</code></td>
<td>
<p>a list for the unit</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAddWeather'>Add weather records into expDB</h2><span id='topic+dbAddWeather'></span>

<h3>Description</h3>

<p>Add weather records into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAddWeather(con, data, name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAddWeather_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAddWeather_+3A_data">data</code></td>
<td>
<p>A string character for the path to met file,
a WeaAna object, or a data frame.</p>
</td></tr>
<tr><td><code id="dbAddWeather_+3A_name">name</code></td>
<td>
<p>The met name in the database 
if data is a data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

<hr>
<h2 id='dbAppendTable'>Append a table into db and check the column name</h2><span id='topic+dbAppendTable'></span>

<h3>Description</h3>

<p>Append a table into db and check the column name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbAppendTable(con, table, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbAppendTable_+3A_con">con</code></td>
<td>
<p>A connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbAppendTable_+3A_table">table</code></td>
<td>
<p>The target table name</p>
</td></tr>
<tr><td><code id="dbAppendTable_+3A_data">data</code></td>
<td>
<p>A data frame to write into table</p>
</td></tr>
</table>

<hr>
<h2 id='dbExportMet'>Export trials weather records to met file</h2><span id='topic+dbExportMet'></span>

<h3>Description</h3>

<p>Export trials weather records to met file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbExportMet(con, output, na = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbExportMet_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbExportMet_+3A_output">output</code></td>
<td>
<p>The folder of output files</p>
</td></tr>
<tr><td><code id="dbExportMet_+3A_na">na</code></td>
<td>
<p>The character for missing value with default NA</p>
</td></tr>
<tr><td><code id="dbExportMet_+3A_...">...</code></td>
<td>
<p>All other arguments to define range of export trials.
All trials will be export if there are not arguments.
Supported arguments include trial (or trialcode)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Write weather records into files. No return values.
</p>

<hr>
<h2 id='dbGenotypeCheckName'>Check genotype names</h2><span id='topic+dbGenotypeCheckName'></span>

<h3>Description</h3>

<p>Check genotype names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGenotypeCheckName(con, genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGenotypeCheckName_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGenotypeCheckName_+3A_genotype">genotype</code></td>
<td>
<p>The genotype name will be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with check genotype names
</p>

<hr>
<h2 id='dbGetDryWeightPerStem'>Get the dry weight per stem</h2><span id='topic+dbGetDryWeightPerStem'></span>

<h3>Description</h3>

<p>Get the dry weight per stem
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetDryWeightPerStem(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetDryWeightPerStem_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetDryWeightPerStem_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetDryWeightPerStem_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected dry weight per stem
</p>

<hr>
<h2 id='dbGetFertilization'>Get fertilization from database</h2><span id='topic+dbGetFertilization'></span>

<h3>Description</h3>

<p>Get fertilization from database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetFertilization(con, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetFertilization_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetFertilization_+3A_...">...</code></td>
<td>
<p>Other arguments to specify meta data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for fertilization information
</p>

<hr>
<h2 id='dbGetFieldMaturity'>Estimation of maturity</h2><span id='topic+dbGetFieldMaturity'></span>

<h3>Description</h3>

<p>Estimation of maturity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetFieldMaturity(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetFieldMaturity_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetFieldMaturity_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetFieldMaturity_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected maturity time
</p>

<hr>
<h2 id='dbGetFieldPopulation'>Estimation of plant populations</h2><span id='topic+dbGetFieldPopulation'></span>

<h3>Description</h3>

<p>Estimation of plant populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetFieldPopulation(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetFieldPopulation_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetFieldPopulation_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetFieldPopulation_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected field population
</p>

<hr>
<h2 id='dbGetFieldStemNumber'>Estimation of stem number per unit area</h2><span id='topic+dbGetFieldStemNumber'></span>

<h3>Description</h3>

<p>Estimation of stem number per unit area
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetFieldStemNumber(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetFieldStemNumber_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetFieldStemNumber_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetFieldStemNumber_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected stem number
</p>

<hr>
<h2 id='dbGetFieldTillerNumber'>Estimation of tiller number per unit area</h2><span id='topic+dbGetFieldTillerNumber'></span>

<h3>Description</h3>

<p>Estimation of tiller number per unit area
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetFieldTillerNumber(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetFieldTillerNumber_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetFieldTillerNumber_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetFieldTillerNumber_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected tiller number
</p>

<hr>
<h2 id='dbGetGene'>Get the gene information</h2><span id='topic+dbGetGene'></span>

<h3>Description</h3>

<p>Get the gene information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetGene(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetGene_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with all gene information
</p>

<hr>
<h2 id='dbGetGenotype'>Get the genotype information</h2><span id='topic+dbGetGenotype'></span>

<h3>Description</h3>

<p>Get the genotype information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetGenotype(con, name_only = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetGenotype_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetGenotype_+3A_name_only">name_only</code></td>
<td>
<p>Only return the name of genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with genotype information or a vector with genotype name 
if name_only = TRUE.
</p>

<hr>
<h2 id='dbGetIrrigation'>Get irrigation from database</h2><span id='topic+dbGetIrrigation'></span>

<h3>Description</h3>

<p>Get irrigation from database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetIrrigation(con, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetIrrigation_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetIrrigation_+3A_...">...</code></td>
<td>
<p>Other arguments to specify meta data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame for irrigation information
</p>

<hr>
<h2 id='dbGetLog'>Get log from expDB</h2><span id='topic+dbGetLog'></span>

<h3>Description</h3>

<p>Get log from expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetLog(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetLog_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with all logs
</p>

<hr>
<h2 id='dbGetMetInfo'>Get met information</h2><span id='topic+dbGetMetInfo'></span>

<h3>Description</h3>

<p>Get met information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetMetInfo(con, name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetMetInfo_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetMetInfo_+3A_name">name</code></td>
<td>
<p>The met name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for met information
</p>

<hr>
<h2 id='dbGetOrganFinalLeafNumber'>Get the final leaf number</h2><span id='topic+dbGetOrganFinalLeafNumber'></span>

<h3>Description</h3>

<p>The final leaf number is first retrived from trait
&quot;O_FinalLeafNumber&quot;, then calculated from trait
&quot;O_HaunIndex&quot; if &quot;O_FinalLeafNumber&quot; is not observed.
Final leaf number equals the maximum value of &quot;O_HaunIndex&quot;, which
should be an integer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetOrganFinalLeafNumber(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetOrganFinalLeafNumber_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetOrganFinalLeafNumber_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments</p>
</td></tr>
<tr><td><code id="dbGetOrganFinalLeafNumber_+3A_...">...</code></td>
<td>
<p>Arguments to specific trials.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected final leaf number
</p>

<hr>
<h2 id='dbGetOrganHaunIndex'>Get Haun Index</h2><span id='topic+dbGetOrganHaunIndex'></span>

<h3>Description</h3>

<p>The Haun Index is retrived from trait &quot;O_HaunIndex&quot;, extending the final observationa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetOrganHaunIndex(con, trials = NULL, avg = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetOrganHaunIndex_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetOrganHaunIndex_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments</p>
</td></tr>
<tr><td><code id="dbGetOrganHaunIndex_+3A_avg">avg</code></td>
<td>
<p>Whether to calculate the average value</p>
</td></tr>
<tr><td><code id="dbGetOrganHaunIndex_+3A_...">...</code></td>
<td>
<p>Arguments to specific trials.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected Haun Index
</p>

<hr>
<h2 id='dbGetPhenotype'>Get phenotype values through a group of conditions</h2><span id='topic+dbGetPhenotype'></span>

<h3>Description</h3>

<p>Get phenotype values through a group of conditions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPhenotype(
  con,
  traits = NULL,
  direction = "long",
  tt = FALSE,
  gene = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPhenotype_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPhenotype_+3A_traits">traits</code></td>
<td>
<p>A list of traits. All traits will be returned if NULL</p>
</td></tr>
<tr><td><code id="dbGetPhenotype_+3A_direction">direction</code></td>
<td>
<p>One of 'long' or 'wide' for reshape function</p>
</td></tr>
<tr><td><code id="dbGetPhenotype_+3A_tt">tt</code></td>
<td>
<p>Whether to calculate thermal time</p>
</td></tr>
<tr><td><code id="dbGetPhenotype_+3A_gene">gene</code></td>
<td>
<p>Whether to get gene information</p>
</td></tr>
<tr><td><code id="dbGetPhenotype_+3A_...">...</code></td>
<td>
<p>All other arguments to define range of export trials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for selected phenotypic values
</p>

<hr>
<h2 id='dbGetPlantFlowering'>Estimation of flowering time</h2><span id='topic+dbGetPlantFlowering'></span>

<h3>Description</h3>

<p>Estimation of flowering time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPlantFlowering(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPlantFlowering_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPlantFlowering_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetPlantFlowering_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected flowering time
</p>

<hr>
<h2 id='dbGetPlantHeading'>Estimation of heading time</h2><span id='topic+dbGetPlantHeading'></span>

<h3>Description</h3>

<p>Estimation of heading time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPlantHeading(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPlantHeading_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPlantHeading_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetPlantHeading_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected heading time
</p>

<hr>
<h2 id='dbGetPlantStemElongation'>Estimation of stem elongation</h2><span id='topic+dbGetPlantStemElongation'></span>

<h3>Description</h3>

<p>Estimation of stem elongation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPlantStemElongation(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPlantStemElongation_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPlantStemElongation_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetPlantStemElongation_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected stem elongation stage
</p>

<hr>
<h2 id='dbGetPlantStemNumber'>Estimation of stem number per plant</h2><span id='topic+dbGetPlantStemNumber'></span>

<h3>Description</h3>

<p>Estimation of stem number per plant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPlantStemNumber(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPlantStemNumber_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPlantStemNumber_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetPlantStemNumber_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected stem number for individual plant
</p>

<hr>
<h2 id='dbGetPlantTillerNumber'>Estimation of tiller number per plant</h2><span id='topic+dbGetPlantTillerNumber'></span>

<h3>Description</h3>

<p>Estimation of tiller number per plant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetPlantTillerNumber(con, trials = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetPlantTillerNumber_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetPlantTillerNumber_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetPlantTillerNumber_+3A_...">...</code></td>
<td>
<p>Arguments pass to dbGetTrials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected tiller number for individual plant
</p>

<hr>
<h2 id='dbGetSites'>Get site into expDB</h2><span id='topic+dbGetSites'></span>

<h3>Description</h3>

<p>Get site into expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetSites(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetSites_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for all sites in the data base
</p>

<hr>
<h2 id='dbGetSource'>Get source from expDB</h2><span id='topic+dbGetSource'></span>

<h3>Description</h3>

<p>Get source from expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetSource(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetSource_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for all source in the data base
</p>

<hr>
<h2 id='dbGetTraits'>Get trait list</h2><span id='topic+dbGetTraits'></span>

<h3>Description</h3>

<p>Get trait list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetTraits(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetTraits_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for all traits in the data base
</p>

<hr>
<h2 id='dbGetTrials'>Get trials by a groups of conditions.</h2><span id='topic+dbGetTrials'></span>

<h3>Description</h3>

<p>Get trials by a groups of conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetTrials(con, design = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetTrials_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetTrials_+3A_design">design</code></td>
<td>
<p>Whether include design</p>
</td></tr>
<tr><td><code id="dbGetTrials_+3A_...">...</code></td>
<td>
<p>All other arguments to define range of export trials.
All trials will be export if there are not arguments.
Supported arguments include trial (or trialcode)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected trials
</p>

<hr>
<h2 id='dbGetWeather'>Get weather records from expDB</h2><span id='topic+dbGetWeather'></span>

<h3>Description</h3>

<p>Get weather records from expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetWeather(con, name, format = "data_frame", na = NA_character_, tz = "UTC")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetWeather_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetWeather_+3A_name">name</code></td>
<td>
<p>The met name</p>
</td></tr>
<tr><td><code id="dbGetWeather_+3A_format">format</code></td>
<td>
<p>The format of export dataset.</p>
</td></tr>
<tr><td><code id="dbGetWeather_+3A_na">na</code></td>
<td>
<p>The character for missing value with default NA</p>
</td></tr>
<tr><td><code id="dbGetWeather_+3A_tz">tz</code></td>
<td>
<p>Time zone applied for hourly temperature</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame for all weather records
</p>

<hr>
<h2 id='dbGetZadoksStage'>Obtain the key phenology stage</h2><span id='topic+dbGetZadoksStage'></span>

<h3>Description</h3>

<p>Obtain the key phenology stage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbGetZadoksStage(con, trials, key_stage)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbGetZadoksStage_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbGetZadoksStage_+3A_trials">trials</code></td>
<td>
<p>A data.frame to specify trials. If not NULL, other arguments
will be ignored.</p>
</td></tr>
<tr><td><code id="dbGetZadoksStage_+3A_key_stage">key_stage</code></td>
<td>
<p>The key zadoks stage</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for selected Zadoks stage
</p>

<hr>
<h2 id='dbImportXLSX'>Import data from excel file</h2><span id='topic+dbImportXLSX'></span>

<h3>Description</h3>

<p>Import data from excel file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbImportXLSX(con, xlsx, ignore_genotype = TRUE, ignore_trait = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbImportXLSX_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbImportXLSX_+3A_xlsx">xlsx</code></td>
<td>
<p>The path to excel file</p>
</td></tr>
<tr><td><code id="dbImportXLSX_+3A_ignore_genotype">ignore_genotype</code></td>
<td>
<p>Ignore genotype tables when importing</p>
</td></tr>
<tr><td><code id="dbImportXLSX_+3A_ignore_trait">ignore_trait</code></td>
<td>
<p>Ignore trait table when importing</p>
</td></tr>
<tr><td><code id="dbImportXLSX_+3A_...">...</code></td>
<td>
<p>Other arguments. Supported arguments include
</p>

<ul>
<li><p>extra_design: Extra columns in the experiment design.
</p>
</li>
<li><p>tz: The time zone for the hourly climates.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>No return values
</p>

<hr>
<h2 id='dbInsertUpdateByRow'>Insert new rows or update existing rows to a specific table according to a specific column (unique) by each row</h2><span id='topic+dbInsertUpdateByRow'></span>

<h3>Description</h3>

<p>Insert new rows or update existing rows to a specific table according to a specific column (unique) by each row
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbInsertUpdateByRow(con, table, data, unique_col = "name")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbInsertUpdateByRow_+3A_con">con</code></td>
<td>
<p>A connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="dbInsertUpdateByRow_+3A_table">table</code></td>
<td>
<p>The target table name</p>
</td></tr>
<tr><td><code id="dbInsertUpdateByRow_+3A_data">data</code></td>
<td>
<p>A data frame to write into table</p>
</td></tr>
<tr><td><code id="dbInsertUpdateByRow_+3A_unique_col">unique_col</code></td>
<td>
<p>A character vector to indentify each row in the table</p>
</td></tr>
</table>

<hr>
<h2 id='dbListTrials'>List all trials</h2><span id='topic+dbListTrials'></span>

<h3>Description</h3>

<p>List all trials
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbListTrials(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbListTrials_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame for all trials in the data base
</p>

<hr>
<h2 id='expdbConnect'>Connect to expDB</h2><span id='topic+expdbConnect'></span>

<h3>Description</h3>

<p>Connect to expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expdbConnect(filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expdbConnect_+3A_filename">filename</code></td>
<td>
<p>The filename of SQLite</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a connection object as produced by dbConnect
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
con &lt;- connect('filename')

## End(Not run)
</code></pre>

<hr>
<h2 id='expdbCreateDB'>create to expDB</h2><span id='topic+expdbCreateDB'></span>

<h3>Description</h3>

<p>create to expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expdbCreateDB(filename, system_traits = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expdbCreateDB_+3A_filename">filename</code></td>
<td>
<p>The filename of new expDB</p>
</td></tr>
<tr><td><code id="expdbCreateDB_+3A_system_traits">system_traits</code></td>
<td>
<p>Whether to import system traits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a connection object as produced by dbConnect
</p>

<hr>
<h2 id='expdbDisconnect'>Didconnect to expDB</h2><span id='topic+expdbDisconnect'></span>

<h3>Description</h3>

<p>Didconnect to expDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expdbDisconnect(con)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expdbDisconnect_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
con &lt;- connect('filename')
disconnect(con)

## End(Not run)
</code></pre>

<hr>
<h2 id='getIdByUniqueIndex'>Get index id by unique_columns.</h2><span id='topic+getIdByUniqueIndex'></span>

<h3>Description</h3>

<p>Get index id by unique_columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getIdByUniqueIndex(
  con,
  table,
  data,
  unique_col = "name",
  data_col = unique_col,
  ignore_case = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getIdByUniqueIndex_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="getIdByUniqueIndex_+3A_table">table</code></td>
<td>
<p>the table name</p>
</td></tr>
<tr><td><code id="getIdByUniqueIndex_+3A_data">data</code></td>
<td>
<p>A data frame to write into table</p>
</td></tr>
<tr><td><code id="getIdByUniqueIndex_+3A_unique_col">unique_col</code></td>
<td>
<p>A character vector to identify each row in the table</p>
</td></tr>
<tr><td><code id="getIdByUniqueIndex_+3A_data_col">data_col</code></td>
<td>
<p>The column names in the data.frame</p>
</td></tr>
<tr><td><code id="getIdByUniqueIndex_+3A_ignore_case">ignore_case</code></td>
<td>
<p>Whether ignore_case</p>
</td></tr>
</table>


<h3>Value</h3>

<p>id of unique_col
</p>

<hr>
<h2 id='harvestQuadratDetail'>Process quadrat (detail) harvestQuadratDetail</h2><span id='topic+harvestQuadratDetail'></span>

<h3>Description</h3>

<p>Process quadrat (detail) harvestQuadratDetail
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harvestQuadratDetail(con, records)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="harvestQuadratDetail_+3A_con">con</code></td>
<td>
<p>a connection object as produced by dbConnect</p>
</td></tr>
<tr><td><code id="harvestQuadratDetail_+3A_records">records</code></td>
<td>
<p>Phenotype records</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return values
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
