<!DOCTYPE html><html lang="en"><head><title>Help for package ExpGenetic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ExpGenetic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#basepreplot'><p>Plot the base frequency distribution diagram for small RNA (sRNA)</p></a></li>
<li><a href='#Countfilter'><p>Filtering out lowly expressed genes based on count</p></a></li>
<li><a href='#F1_miRNA_count'><p>Count table of miRNAs in F1 (F1: the polyploid progeny).</p></a></li>
<li><a href='#F1_miRNA_rpm'><p>RPM table of miRNAs in F1 (F1: the polyploid progeny).</p></a></li>
<li><a href='#F1_sRNA_seq'><p>All sRNA sequences in F1 (F1: the polyploid progeny).</p></a></li>
<li><a href='#Get12Bins'><p>Non-additive expression analysis</p></a></li>
<li><a href='#GetDAtable'><p>Non-additive expression analysis</p></a></li>
<li><a href='#GetDEdata'><p>Get the results of differential expression analysis.</p></a></li>
<li><a href='#lenplot'><p>Plot the length distribution diagram for small RNAs (sRNAs)</p></a></li>
<li><a href='#mirnapredata'><p>Base frequency distribution of small RNA (sRNA)</p></a></li>
<li><a href='#P1_miRNA_count'><p>Count table of miRNAs in P1 (P1: one of the parents).</p></a></li>
<li><a href='#P1_miRNA_rpm'><p>RPM table of miRNAs in P1 (P1: one of the parents).</p></a></li>
<li><a href='#P1_sRNA_seq'><p>All sRNA sequences in P1 (P1: one of the parents).</p></a></li>
<li><a href='#P2_miRNA_count'><p>Count table of miRNAs in P2 (P2: one of the parents).</p></a></li>
<li><a href='#P2_miRNA_rpm'><p>RPM table of miRNAs in P2 (P2: one of the parents).</p></a></li>
<li><a href='#P2_sRNA_seq'><p>All sRNA sequences in P2 (P2: one of the parents).</p></a></li>
<li><a href='#polyDESeq'><p>Make a Triangle Diagram</p></a></li>
<li><a href='#Rpmfilter'><p>Filtering out lowly expressed genes based on RPM</p></a></li>
<li><a href='#srnapredata'><p>Length distribution of small RNAs (sRNAs)</p></a></li>
<li><a href='#VennData'><p>Get the details of the Venn Diagram</p></a></li>
<li><a href='#VennPlot'><p>Make a three-set Venn Diagram</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Non-Additive Expression Analysis of Hybrid Offspring</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Three functional modules, including genetic features, differential expression analysis and non-additive expression analysis were integrated into the package. And the package is suitable for RNA-seq and small RNA sequencing data. Besides, two methods of non-additive expression analysis were provided. One is the calculation of the additive (a) and dominant (d), the other is the evaluation of expression level dominance by comparing the total expression of the gene in hybrid offspring with the expression level in parents. For non-additive expression analysis of RNA-seq data, it is only applicable to hybrid offspring (including two sub-genomes) species for the time being.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>DESeq2 (&ge; 1.34.0), futile.logger (&ge; 1.4.3), ggplot2 (&ge;
3.3.6), ggsci (&ge; 2.9), plyr (&ge; 1.8.7), VennDiagram (&ge; 1.7.3)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-08 08:27:24 UTC; wuyuqing</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuqing Wu <a href="https://orcid.org/0000-0002-6333-0926"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuqing Wu &lt;wuyuqing0104@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-09 07:13:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='basepreplot'>Plot the base frequency distribution diagram for small RNA (sRNA)</h2><span id='topic+basepreplot'></span>

<h3>Description</h3>

<p>Plot the base frequency distribution diagram for small RNA (sRNA)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basepreplot(sRNAdata, width = 0.6, font_size = 10, title_size = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="basepreplot_+3A_srnadata">sRNAdata</code></td>
<td>
<p>A data frame. Base frequency distribution of sRNAs.</p>
</td></tr>
<tr><td><code id="basepreplot_+3A_width">width</code></td>
<td>
<p>A numeric. Bar width, and default is 0.6.</p>
</td></tr>
<tr><td><code id="basepreplot_+3A_font_size">font_size</code></td>
<td>
<p>A numeric. Size of axis ticks and legend item labels, and default is 10.</p>
</td></tr>
<tr><td><code id="basepreplot_+3A_title_size">title_size</code></td>
<td>
<p>A numeric. Size of axis titles and legend titles, and default is 12.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Base frequency distribution plot of sRNAs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#F1
F1_miRNA &lt;- F1_miRNA_count[,1]
F1_bf &lt;- mirnapredata(sRNAseq = F1_miRNA)
basepreplot(sRNAdata = F1_bf)
</code></pre>

<hr>
<h2 id='Countfilter'>Filtering out lowly expressed genes based on count</h2><span id='topic+Countfilter'></span>

<h3>Description</h3>

<p>Regarding the criteria for filtering out lowly expressed genes, no less than the count threshold in all replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Countfilter(
  P1_count,
  P2_count,
  F1_count,
  type,
  homoeologs,
  count_threshold = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Countfilter_+3A_p1_count">P1_count</code></td>
<td>
<p>A data frame. The count table of genes in P1 species. For the count table, the first column is the gene identifier, and other columns are read counts of the gene in each biological replicate.</p>
</td></tr>
<tr><td><code id="Countfilter_+3A_p2_count">P2_count</code></td>
<td>
<p>A data frame. The count table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="Countfilter_+3A_f1_count">F1_count</code></td>
<td>
<p>A data frame. The count table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="Countfilter_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="Countfilter_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes within the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="Countfilter_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 5 (the count values in all replicates).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1; P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>A data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Count5result &lt;- Countfilter(P1_count = P1_miRNA_count,
                            P2_count = P2_miRNA_count,
                            F1_count = F1_miRNA_count,
                            type = "sRNA", count_threshold = 5)
</code></pre>

<hr>
<h2 id='F1_miRNA_count'>Count table of miRNAs in F1 (F1: the polyploid progeny).</h2><span id='topic+F1_miRNA_count'></span>

<h3>Description</h3>

<p>Count table of miRNAs in F1 species. The &quot;F1&quot; represents the polyploid progeny.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(F1_miRNA_count)
#                sequence     BF1.1    BF1.2    BF1.3
#1  TTTGGATTGAAGGGAGCTCTA     20233     6388    16732
#2  TTTCCAAATGTAGACAAAGCA     19909     5157    16076
#3    TCCCAAATGTAGACAAAGC        82       33      103
#4 CTTTGTCTATCGTTTGGAAAAG      2367     1040     3203
#5  TTGGACTGAAGGGAGCTCCTT        34        9       21
#6  TCGGACCAGGCTTCATTCCCC      3281      607     1289
</code></pre>

<hr>
<h2 id='F1_miRNA_rpm'>RPM table of miRNAs in F1 (F1: the polyploid progeny).</h2><span id='topic+F1_miRNA_rpm'></span>

<h3>Description</h3>

<p>RPM table of miRNAs in F1 species. The &quot;F1&quot; represents the polyploid progeny.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(F1_miRNA_rpm)
#                sequence     BF1.1    BF1.2     BF1.3
#1  TTTGGATTGAAGGGAGCTCTA   1512.16  1086.35   2032.97
#2  TTTCCAAATGTAGACAAAGCA   1487.94   877.01   1953.27
#3    TCCCAAATGTAGACAAAGC      6.13     5.61     12.51
#4 CTTTGTCTATCGTTTGGAAAAG    176.90   176.86    389.17
#5  TTGGACTGAAGGGAGCTCCTT      2.54     1.53      2.55
#6  TCGGACCAGGCTTCATTCCCC    245.21   103.23    156.62
</code></pre>

<hr>
<h2 id='F1_sRNA_seq'>All sRNA sequences in F1 (F1: the polyploid progeny).</h2><span id='topic+F1_sRNA_seq'></span>

<h3>Description</h3>

<p>All sRNA sequences in F1 (F1: the polyploid progeny).
</p>

<hr>
<h2 id='Get12Bins'>Non-additive expression analysis</h2><span id='topic+Get12Bins'></span>

<h3>Description</h3>

<p>Rapp et al. proposed the classification of 12 expression patterns in allopolyploids, including additivity (I, XII), ELD (II, XI, IV, IX), transgressive down-regulation (III, VII, X) and transgressive up-regulation (V, VI, VIII).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get12Bins(
  P1_count,
  P2_count,
  F1_count,
  type,
  homoeologs,
  count_threshold = 5,
  Pvalue = 0.05,
  log2FC = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Get12Bins_+3A_p1_count">P1_count</code></td>
<td>
<p>A data frame. The count table of genes in P1 species. For the count table, the first column is the gene identifier, and other columns are the corresponding expression levels of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_p2_count">P2_count</code></td>
<td>
<p>A data frame. The count table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_f1_count">F1_count</code></td>
<td>
<p>A data frame. The count table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 5 (the count values in all replicates).</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_pvalue">Pvalue</code></td>
<td>
<p>A numeric. The P value of differential expression analysis using DESeq2. Default is 0.05.</p>
</td></tr>
<tr><td><code id="Get12Bins_+3A_log2fc">log2FC</code></td>
<td>
<p>A numeric. The log2-transformed expression fold of differential expression analysis using DESeq2. Default is 1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>pv11: P value of differential expression analysis using DESeq2. Parental P1 was used as the control group and F1 was used as the treatment group. pv12: P value of differential expression analysis using DESeq2. Parental P2 was used as the control group and F1 was used as the treatment group. pv21: P value of differential expression analysis using DESeq2. Parental P1 was used as the control group and P2 was used as the treatment group. Besides, &quot;fc&quot; represents the log2FoldChange of differential expression analysis.
</p>


<h3>Value</h3>

<p>A data frame. Classification results of non-additive analysis based on the ELD method.
</p>


<h3>References</h3>

<p>Rapp RA, Udall JA, Wendel JF. Genomic expression dominance in allopolyploids. BMC Biol. 2009 May 1;7:18.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
miRNA_12bin &lt;- Get12Bins(P1_count = P1_miRNA_count,
                         P2_count = P2_miRNA_count,
                         F1_count = F1_miRNA_count,type = "sRNA")

</code></pre>

<hr>
<h2 id='GetDAtable'>Non-additive expression analysis</h2><span id='topic+GetDAtable'></span>

<h3>Description</h3>

<p>About the classification method based on |d/a|, the additive (a) and dominant (d) values were calculated by the expression level of each miRNA. Edwards et al. proposed that the &quot;|d/a|&quot; can be used as the criterion to estimate the expression patterns of miRNAs. Specific classification criteria are as follows, |d/a| &lt;= 0.2, additivity; |d/a| &gt; 0.2 and |d/a| &lt;= 0.8, partial dominance; |d/a| &gt; 0.8 and |d/a| &lt;= 1.2, dominance; |d/a| &gt; 1.2, overdominance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetDAtable(P1_RPM, P2_RPM, F1_RPM, type, homoeologs, rpm_threshold = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetDAtable_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P1 species. For the RPM table, the first column is the gene identifier, and other columns are the RPM values of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="GetDAtable_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 1 (the average RPM of all replicates).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1; P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>A data frame. Classification results of non-additive expression analysis based on |d/a|.
</p>


<h3>References</h3>

<p>Edwards MD, Stuber CW, Wendel JF. Molecular-marker-facilitated investigations of quantitative-trait loci in maize. I. Numbers, genomic distribution and types of gene action. Genetics. 1987 May;116(1):113-25.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DAresult &lt;- GetDAtable(P1_RPM = P1_miRNA_rpm,
                       P2_RPM = P2_miRNA_rpm,
                       F1_RPM = F1_miRNA_rpm,
                       type = "sRNA", rpm_threshold = 1)
</code></pre>

<hr>
<h2 id='GetDEdata'>Get the results of differential expression analysis.</h2><span id='topic+GetDEdata'></span>

<h3>Description</h3>

<p>Extract the results of differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetDEdata(
  P1_count,
  P2_count,
  F1_count,
  output_type,
  type,
  homoeologs,
  count_threshold = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetDEdata_+3A_p1_count">P1_count</code></td>
<td>
<p>A data frame. The count table of genes in P1 species. For the count table, the first column is the gene identifier, and other columns are the corresponding expression levels of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_p2_count">P2_count</code></td>
<td>
<p>A data frame. The count table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_f1_count">F1_count</code></td>
<td>
<p>A data frame. The count table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_output_type">output_type</code></td>
<td>
<p>A character. &quot;F1_vs_P1&quot;, &quot;F1_vs_P2&quot; or &quot;P2_vs_P1&quot;.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="GetDEdata_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 5 (the count values in all replicates).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>F1_vs_P1: Results of differential expression analysis using DESeq2. Parental P1 was used as the control group and F1 was used as the treatment group. If the log2FoldChange of a gene is positive, it means that the expression level of the gene in F1 is higher than that in P1. F1_vs_P2: Results of differential expression analysis using DESeq2. Parental P2 was used as the control group and F1 was used as the treatment group. P2_vs_P1: Results of differential expression analysis using DESeq2. Parental P1 was used as the control group and P2 was used as the treatment group.
</p>


<h3>Value</h3>

<p>A data frame. Differential expression analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
P2_vs_P1 &lt;- GetDEdata(P1_count = P1_miRNA_count,
                      P2_count = P2_miRNA_count,
                      F1_count = F1_miRNA_count,
                      output_type = "P2_vs_P1", type="sRNA")

</code></pre>

<hr>
<h2 id='lenplot'>Plot the length distribution diagram for small RNAs (sRNAs)</h2><span id='topic+lenplot'></span>

<h3>Description</h3>

<p>There are two types of pictures: bar plot (type = &quot;bar&quot;) and line plot (type = &quot;line&quot;). For the bar plot, the Y-axis displays the proportion of sRNAs in a certain length, the X-axis represents sRNAs in different length. And for line plot, the Y-axis displays the abundance of sRNAs in a certain length, the X-axis represents sRNAs in different length.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lenplot(sRNAdata, type, width = 0.6, font_size = 10, title_size = 12)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lenplot_+3A_srnadata">sRNAdata</code></td>
<td>
<p>A data frame. Frequency distribution of sRNAs in different length.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_type">type</code></td>
<td>
<p>A character. &quot;bar&quot; or &quot;line&quot;.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_width">width</code></td>
<td>
<p>A numeric. Bar width, and default is 0.6. if the type is &quot;line&quot;, the parameter does not need to be given.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_font_size">font_size</code></td>
<td>
<p>A numeric. Size of axis ticks and legend item labels, and default is 10.</p>
</td></tr>
<tr><td><code id="lenplot_+3A_title_size">title_size</code></td>
<td>
<p>A numeric. Size of axis titles and legend titles, and default is 12.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Length distribution plot of sRNAs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#P1(B.napus)
B.napu_sRNA &lt;- srnapredata(sRNAseq = P1_sRNA_seq,Group = "B.napus(AACC)")
#P2(B.rapa)
B.rapa_sRNA &lt;- srnapredata(sRNAseq = P2_sRNA_seq,Group = "B.rapa(AA)")
#F1(B.napus X B.rapa)
B.nr_sRNA &lt;- srnapredata(sRNAseq = F1_sRNA_seq,Group = "B.napus x B.rapa(AAAACC)")
#intergrate these data for length distribution plot
sRNA_data &lt;- rbind(B.napu_sRNA,B.rapa_sRNA,B.nr_sRNA)
#plot
lenplot(sRNAdata = sRNA_data,type = "line")
lenplot(sRNAdata = sRNA_data,type = "bar")
</code></pre>

<hr>
<h2 id='mirnapredata'>Base frequency distribution of small RNA (sRNA)</h2><span id='topic+mirnapredata'></span>

<h3>Description</h3>

<p>Get the base frequency distribution table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mirnapredata(sRNAseq)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mirnapredata_+3A_srnaseq">sRNAseq</code></td>
<td>
<p>Character. All sRNA sequences in vector format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame. The output consists of three columns, i.e., base, base frequency and position.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#F1
F1_miRNA &lt;- F1_miRNA_count[,1]
F1_bf &lt;- mirnapredata(sRNAseq = F1_miRNA)
#output result
head(F1_bf)
#  Base Frequency Position
#1    A        32        1
#2    C        27        1
#3    G        31        1
#4    T       115        1
#5    A        27        2
#6    C        50        2
</code></pre>

<hr>
<h2 id='P1_miRNA_count'>Count table of miRNAs in P1 (P1: one of the parents).</h2><span id='topic+P1_miRNA_count'></span>

<h3>Description</h3>

<p>Count table of miRNAs in P1 species. The &quot;P1&quot; represents one of parents.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(P1_miRNA_count)
#                sequence   Bnapus.1   Bnapus.2   Bnapus.3
#1  TTTGGATTGAAGGGAGCTCTA      29848      12094      10685
#2  TTAGATTCACGCACAAACTCG        986        571        456
#3  TGAAGCTGCCAGCATGATCTA       3152       1436       1091
#4 CTTTGTCTATCGTTTGGAAAAG       2449       1307       1116
#5  GATCATGTTCGCAGTTTCACC       1364        650        656
#6  TTTCCAAATGTAGACAAAGCA      11658       3914       4123
</code></pre>

<hr>
<h2 id='P1_miRNA_rpm'>RPM table of miRNAs in P1 (P1: one of the parents).</h2><span id='topic+P1_miRNA_rpm'></span>

<h3>Description</h3>

<p>RPM table of miRNAs in P1 species. The &quot;P1&quot; represents one of parents.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(P1_miRNA_rpm)
#               sequence    Brapa.1    Brapa.2    Brapa.3
#1 TTTGGATTGAAGGGAGCTCTA    1641.18    1116.03    1014.37
#2 TGAAGCTGCCAGCATGATCTA     129.33     103.23     103.68
#3 TTTCCAAATGTAGACAAAGCA     905.23     920.57    1180.51
#4 TCGGACCAGGCTTCATCCCCC      24.71      14.38      15.03
#5 AGAATCTTGATGATGCTGCAG      48.64      41.09      41.60
#6 TTGACAGAAGAAAGAGAGCAC      86.96      81.23      67.41
</code></pre>

<hr>
<h2 id='P1_sRNA_seq'>All sRNA sequences in P1 (P1: one of the parents).</h2><span id='topic+P1_sRNA_seq'></span>

<h3>Description</h3>

<p>All sRNA sequences in P1 (P1: one of the parents).
</p>

<hr>
<h2 id='P2_miRNA_count'>Count table of miRNAs in P2 (P2: one of the parents).</h2><span id='topic+P2_miRNA_count'></span>

<h3>Description</h3>

<p>Count table of miRNAs in P2 species. The &quot;P2&quot; represents one of parents.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(P2_miRNA_count)
#                sequence  Bnapus.1  Bnapus.2  Bnapus.3
#1  TTTGGATTGAAGGGAGCTCTA     29848     12094     10685
#2  TTAGATTCACGCACAAACTCG       986       571       456
#3  TGAAGCTGCCAGCATGATCTA      3152      1436      1091
#4 CTTTGTCTATCGTTTGGAAAAG      2449      1307      1116
#5  GATCATGTTCGCAGTTTCACC      1364       650       656
#6  TTTCCAAATGTAGACAAAGCA     11658      3914      4123
</code></pre>

<hr>
<h2 id='P2_miRNA_rpm'>RPM table of miRNAs in P2 (P2: one of the parents).</h2><span id='topic+P2_miRNA_rpm'></span>

<h3>Description</h3>

<p>RPM table of miRNAs in P2 species. The &quot;P2&quot; represents one of parents.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(P2_miRNA_rpm)
#                sequence   Bnapus.1   Bnapus.2   Bnapus.3
#1  TTTGGATTGAAGGGAGCTCTA    1804.35    1362.88    1439.22
#2  TTAGATTCACGCACAAACTCG      59.60      64.35      61.42
#3  TGAAGCTGCCAGCATGATCTA     190.54     161.82     146.95
#4 CTTTGTCTATCGTTTGGAAAAG     148.04     147.29     150.32
#5  GATCATGTTCGCAGTTTCACC      82.46      73.25      88.36
#6  TTTCCAAATGTAGACAAAGCA     704.74     441.07     555.35
</code></pre>

<hr>
<h2 id='P2_sRNA_seq'>All sRNA sequences in P2 (P2: one of the parents).</h2><span id='topic+P2_sRNA_seq'></span>

<h3>Description</h3>

<p>All sRNA sequences in P2 (P2: one of the parents).
</p>

<hr>
<h2 id='polyDESeq'>Make a Triangle Diagram</h2><span id='topic+polyDESeq'></span>

<h3>Description</h3>

<p>The count matrix of different species as the input data to perform differential expression analysis using DESeq2. And the number of differentially expressed genes between any two species is marked on the triangle diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polyDESeq(
  P1_count,
  P2_count,
  F1_count,
  P1_name,
  P2_name,
  F1_name,
  type,
  homoeologs,
  count_threshold = 5,
  Pvalue = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="polyDESeq_+3A_p1_count">P1_count</code></td>
<td>
<p>A data frame. The count table of genes in P1 species. For the count table, the first column is the gene identifier, and other columns are the read counts of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_p2_count">P2_count</code></td>
<td>
<p>A data frame. The count table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_f1_count">F1_count</code></td>
<td>
<p>A data frame. The count table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_p1_name">P1_name</code></td>
<td>
<p>A character. Category names of P1 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_p2_name">P2_name</code></td>
<td>
<p>A character. Category names of P2 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_f1_name">F1_name</code></td>
<td>
<p>A character. Category names of F1 species.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_count_threshold">count_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 5 (the count values in all replicates).</p>
</td></tr>
<tr><td><code id="polyDESeq_+3A_pvalue">Pvalue</code></td>
<td>
<p>A numeric. Threshold for significance test in differential expression analysis. Default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1;P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>Triangle Diagram
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
polyDESeq(P1_count = P1_miRNA_count,
          P2_count = P2_miRNA_count,
          F1_count = F1_miRNA_count,
          P1_name = "B.napus(AACC)",
          P2_name = "B.rapa(AA)",
          F1_name = "B.napus x B.rapa (AAAACC)",type="sRNA")

</code></pre>

<hr>
<h2 id='Rpmfilter'>Filtering out lowly expressed genes based on RPM</h2><span id='topic+Rpmfilter'></span>

<h3>Description</h3>

<p>Regarding the criteria for filtering out lowly expressed genes, no less than the RPM threshold in all replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Rpmfilter(P1_RPM, P2_RPM, F1_RPM, type, homoeologs, rpm_threshold = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Rpmfilter_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P1 species. For the RPM table, the first column is the gene identifier (e.g. sequences of sRNA, Gene ID), and other columns are the RPM values of the gene in each biological replicate.</p>
</td></tr>
<tr><td><code id="Rpmfilter_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="Rpmfilter_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="Rpmfilter_+3A_type">type</code></td>
<td>
<p>A character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="Rpmfilter_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes within the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="Rpmfilter_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 1 (the average RPM of all replicates).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1; P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>A data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Rpm1result &lt;- Rpmfilter(P1_RPM = P1_miRNA_rpm,
                         P2_RPM = P2_miRNA_rpm,
                         F1_RPM = F1_miRNA_rpm,
                         type = "sRNA", rpm_threshold = 1)
</code></pre>

<hr>
<h2 id='srnapredata'>Length distribution of small RNAs (sRNAs)</h2><span id='topic+srnapredata'></span>

<h3>Description</h3>

<p>Get the length distribution of sRNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srnapredata(sRNAseq, Group)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="srnapredata_+3A_srnaseq">sRNAseq</code></td>
<td>
<p>Character. All sRNA sequences in vector format.</p>
</td></tr>
<tr><td><code id="srnapredata_+3A_group">Group</code></td>
<td>
<p>Character. Group name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame. The output consists of three columns, i.e., length, frequency and group name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#P1(B.napus)
B.napu_sRNA &lt;- srnapredata(sRNAseq = P1_sRNA_seq, Group = "B.napus(AACC)")
#P2(B.rapa)
B.rapa_sRNA &lt;- srnapredata(sRNAseq = P2_sRNA_seq, Group = "B.rapa(AA)")
#F1(B.napus X B.rapa)
B.nr_sRNA &lt;- srnapredata(sRNAseq = F1_sRNA_seq, Group = "B.napus x B.rapa(AAAACC)")
#intergrate these data for length distribution plot
sRNA_data &lt;- rbind(B.napu_sRNA, B.rapa_sRNA, B.nr_sRNA)
#output result
head(sRNA_data)
#  Length Frequency         Group
#1     15         8 B.napus(AACC)
#2     16         7 B.napus(AACC)
#3     17        13 B.napus(AACC)
#4     18        16 B.napus(AACC)
#5     19        25 B.napus(AACC)
#6     20        33 B.napus(AACC)
</code></pre>

<hr>
<h2 id='VennData'>Get the details of the Venn Diagram</h2><span id='topic+VennData'></span>

<h3>Description</h3>

<p>Get the information for each region of the venn diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VennData(
  P1_RPM,
  P2_RPM,
  F1_RPM,
  type,
  homoeologs,
  rpm_threshold = 1,
  output_file = "venn_list"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VennData_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P1 species. For the RPM table, the first column is the gene identifier, and other columns are the RPM values of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="VennData_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="VennData_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="VennData_+3A_type">type</code></td>
<td>
<p>Character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="VennData_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="VennData_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 1 (the average RPM of all replicates).</p>
</td></tr>
<tr><td><code id="VennData_+3A_output_file">output_file</code></td>
<td>
<p>&quot;venn_list&quot;, &quot;P1_specific&quot;, &quot;P2_specific&quot;, &quot;F1_specific&quot;, or &quot;all_common&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1; P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>A data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#output_file = "venn_list"
venn_list &lt;- VennData(P1_RPM = P1_miRNA_rpm,
                      P2_RPM = P2_miRNA_rpm,
                      F1_RPM = F1_miRNA_rpm,
                      type="sRNA",rpm_threshold = 1,
                      output_file = "venn_list")
##output_file = "P1_specific"
P1_specific &lt;- VennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          type="sRNA",rpm_threshold = 1,
                          output_file = "P1_specific")
##output_file = "P2_specific"
P2_specific &lt;- VennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          type="sRNA",rpm_threshold = 1,
                          output_file = "P2_specific")
##output_file = "F1_specific"
F1_specific &lt;- VennData(P1_RPM = P1_miRNA_rpm,
                          P2_RPM = P2_miRNA_rpm,
                          F1_RPM = F1_miRNA_rpm,
                          type="sRNA",rpm_threshold = 1,
                          output_file = "F1_specific")
##output_file = "all_common"
all_common &lt;- VennData(P1_RPM = P1_miRNA_rpm,
                         P2_RPM = P2_miRNA_rpm,
                         F1_RPM = F1_miRNA_rpm,
                         type="sRNA",rpm_threshold = 1,
                         output_file = "all_common")
</code></pre>

<hr>
<h2 id='VennPlot'>Make a three-set Venn Diagram</h2><span id='topic+VennPlot'></span>

<h3>Description</h3>

<p>This function creates a Venn Diagram to display the overlap of expressed genes between three sets (parents and progeny).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VennPlot(
  P1_RPM,
  P2_RPM,
  F1_RPM,
  P1_name,
  P2_name,
  F1_name,
  type,
  homoeologs,
  rpm_threshold = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VennPlot_+3A_p1_rpm">P1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P1 species. For the RPM table, the first column is the gene identifier, and other columns are the RPM values of the genes in each biological replicate.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_p2_rpm">P2_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in P2 species.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_f1_rpm">F1_RPM</code></td>
<td>
<p>A data frame. The RPM table of genes in F1 species.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_p1_name">P1_name</code></td>
<td>
<p>Character. Category names of P1 species.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_p2_name">P2_name</code></td>
<td>
<p>Character. Category names of P2 species.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_f1_name">F1_name</code></td>
<td>
<p>Character. Category names of F1 species.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_type">type</code></td>
<td>
<p>Character. &quot;sRNA&quot; or &quot;mRNA&quot;.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_homoeologs">homoeologs</code></td>
<td>
<p>A data frame. Orthologous relationships of genes in the parental species and their progeny. Only required when the 'type' is 'mRNA'.</p>
</td></tr>
<tr><td><code id="VennPlot_+3A_rpm_threshold">rpm_threshold</code></td>
<td>
<p>A numeric. Threshold for filtering out the lowly expressed genes. The default is 1 (the average RPM of all replicates).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'homoeologs' table contains the orthologs pairs. In detail, the first column is the group name (unique) of homoeologs among three species (Parents: P1; P2, Progeny: F1), the second column is the Gene ID of P1, the third column is the Gene ID of P2. And the fourth column and fifth columns are the identifier of F1 orthologs derived from P1 and P2 ancestors, respectively (e.g. &quot;Homoeolog1	BraA01t00004Z	BolC01g000040.2J BnA01g0000030.1 BnC01g0424620.1&quot;).
</p>


<h3>Value</h3>

<p>Venn diagram.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#miRNA
VennPlot(P1_RPM = P1_miRNA_rpm,
         P2_RPM = P2_miRNA_rpm,
         F1_RPM = F1_miRNA_rpm,
         P1_name = "B.napus(AACC)",
         P2_name = "B.rapa(AA)",
         F1_name = "B.napus x B.rapa(AAAACC)",type="sRNA")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
