<!DOCTYPE html><html><head><title>Help for package FactoMineR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FactoMineR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FactoMineR-package'><p>Multivariate Exploratory Data Analysis and Data Mining with R</p></a></li>
<li><a href='#AovSum'><p>Analysis of variance with the contrasts sum (the sum of the coefficients is 0)</p></a></li>
<li><a href='#autoLab'><p>Function to better position the labels on the graphs</p></a></li>
<li><a href='#CA'><p>Correspondence Analysis (CA)</p></a></li>
<li><a href='#CaGalt'><p>Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt)</p></a></li>
<li><a href='#catdes'><p>Categories description</p></a></li>
<li><a href='#children'><p>Children (data)</p></a></li>
<li><a href='#coeffRV'><p>Calculate the RV coefficient and test its significance</p></a></li>
<li><a href='#condes'><p>Continuous variable description</p></a></li>
<li><a href='#coord.ellipse'><p>Construct confidence ellipses</p></a></li>
<li><a href='#decathlon'><p>Performance in decathlon (data)</p></a></li>
<li><a href='#descfreq'><p>Description of frequencies</p></a></li>
<li><a href='#dimdesc'><p>Dimension description</p></a></li>
<li><a href='#DMFA'><p>Dual Multiple Factor Analysis (DMFA)</p></a></li>
<li><a href='#ellipseCA'><p>Draw confidence ellipses in CA</p></a></li>
<li><a href='#estim_ncp'><p>Estimate the number of components in Principal Component Analysis</p></a></li>
<li><a href='#FAMD'><p>Factor Analysis for Mixed Data</p></a></li>
<li><a href='#footsize'><p>footsize</p></a></li>
<li><a href='#geomorphology'><p>geomorphology(data)</p></a></li>
<li><a href='#GPA'><p>Generalised Procrustes Analysis</p></a></li>
<li><a href='#graph.var'><p>Make graph of variables</p></a></li>
<li><a href='#HCPC'><p>Hierarchical Clustering on Principle Components (HCPC)</p></a></li>
<li><a href='#health'><p>health (data)</p></a></li>
<li><a href='#HMFA'><p>Hierarchical Multiple Factor Analysis</p></a></li>
<li><a href='#hobbies'><p>hobbies (data)</p></a></li>
<li><a href='#JO'><p>Number of medals in athletism during olympic games per country</p></a></li>
<li><a href='#LinearModel'><p>Linear Model with AIC or BIC selection, and with the contrasts</p>
sum (the sum of the coefficients is 0) if any categorical variables</a></li>
<li><a href='#MCA'><p>Multiple Correspondence Analysis (MCA)</p></a></li>
<li><a href='#meansComp'><p>Perform pairwise means comparisons</p></a></li>
<li><a href='#MFA'><p>Multiple Factor Analysis (MFA)</p></a></li>
<li><a href='#milk'><p>milk</p></a></li>
<li><a href='#mortality'><p>The cause of mortality in France in 1979 and 2006</p></a></li>
<li><a href='#PCA'><p>Principal Component Analysis (PCA)</p></a></li>
<li><a href='#plot.CA'><p>Draw the Correspondence Analysis (CA) graphs</p></a></li>
<li><a href='#plot.CaGalt'><p>Draw the Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) graphs</p></a></li>
<li><a href='#plot.catdes'><p>Plots for description of clusters (catdes)</p></a></li>
<li><a href='#plot.DMFA'><p>Draw the Dual Multiple Factor Analysis (DMFA) graphs</p></a></li>
<li><a href='#plot.FAMD'><p>Draw the Multiple Factor Analysis for Mixt Data graphs</p></a></li>
<li><a href='#plot.GPA'><p>Draw the General Procrustes Analysis (GPA) map</p></a></li>
<li><a href='#plot.HCPC'><p>Plots for Hierarchical Classification on Principle Components (HCPC) results</p></a></li>
<li><a href='#plot.HMFA'><p>Draw the Hierarchical Multiple Factor Analysis (HMFA) graphs</p></a></li>
<li><a href='#plot.MCA'><p>Draw the Multiple Correspondence Analysis (MCA) graphs</p></a></li>
<li><a href='#plot.meansComp'><p>Draw the means comparisons</p></a></li>
<li><a href='#plot.MFA'><p>Draw the Multiple Factor Analysis (MFA) graphs</p></a></li>
<li><a href='#plot.PCA'><p>Draw the Principal Component Analysis (PCA) graphs</p></a></li>
<li><a href='#plotellipses'><p>Draw confidence ellipses around the categories</p></a></li>
<li><a href='#plotGPApartial'><p>Draw an interactive General Procrustes Analysis (GPA) map</p></a></li>
<li><a href='#plotMFApartial'><p>Plot an interactive Multiple Factor Analysis (MFA) graph</p></a></li>
<li><a href='#poison'><p>Poison</p></a></li>
<li><a href='#poison.text'><p>Poison</p></a></li>
<li><a href='#poulet'><p>Genomic data for chicken</p></a></li>
<li><a href='#predict.CA'><p>Predict projection for new rows with Correspondence Analysis</p></a></li>
<li><a href='#predict.FAMD'><p>Predict projection for new rows with Factor Analysis of Mixed Data</p></a></li>
<li><a href='#predict.LinearModel'><p>Predict method for Linear Model Fits</p></a></li>
<li><a href='#predict.MCA'><p>Predict projection for new rows with Multiple Correspondence Analysis</p></a></li>
<li><a href='#predict.MFA'><p>Predict projection for new rows with Multiple Factor Analysis</p></a></li>
<li><a href='#predict.PCA'><p>Predict projection for new rows with Principal Component Analysis</p></a></li>
<li><a href='#prefpls'><p>Scatter plot and additional variables with quality of representation contour lines</p></a></li>
<li><a href='#print.AovSum'><p>Print the AovSum results</p></a></li>
<li><a href='#print.CA'><p>Print the Correspondance Analysis (CA) results</p></a></li>
<li><a href='#print.CaGalt'><p>Print the Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) results</p></a></li>
<li><a href='#print.catdes'><p>Print the catdes results</p></a></li>
<li><a href='#print.condes'><p>Print the condes results</p></a></li>
<li><a href='#print.FAMD'><p>Print the Multiple Factor Analysis of mixt Data (FAMD) results</p></a></li>
<li><a href='#print.GPA'><p>Print the Generalized Procrustes Analysis (GPA) results</p></a></li>
<li><a href='#print.HCPC'><p>Print the Hierarchical Clustering on Principal Components (HCPC) results</p></a></li>
<li><a href='#print.HMFA'><p>Print the Hierarchical Multiple Factor Analysis results</p></a></li>
<li><a href='#print.LinearModel'><p>Print the LinearModel results</p></a></li>
<li><a href='#print.MCA'><p>Print the Multiple Correspondance Analysis (MCA) results</p></a></li>
<li><a href='#print.MFA'><p>Print the Multiple Factor Analysis results</p></a></li>
<li><a href='#print.PCA'><p>Print the Principal Component Analysis (PCA) results</p></a></li>
<li><a href='#reconst'><p>Reconstruction of the data from the PCA, CA or MFA results</p></a></li>
<li><a href='#RegBest'><p>Select variables in multiple linear regression</p></a></li>
<li><a href='#senso'><p>senso</p></a></li>
<li><a href='#simule'><p>Simulate by bootstrap</p></a></li>
<li><a href='#summary.CA'><p>Printing summeries of ca objects</p></a></li>
<li><a href='#summary.CaGalt'><p>Printing summaries of CaGalt objects</p></a></li>
<li><a href='#summary.FAMD'><p>Printing summeries of FAMD objects</p></a></li>
<li><a href='#summary.MCA'><p>Printing summeries of MCA objects</p></a></li>
<li><a href='#summary.MFA'><p>Printing summaries of MFA objects</p></a></li>
<li><a href='#summary.PCA'><p>Printing summeries of PCA objects</p></a></li>
<li><a href='#svd.triplet'><p>Singular Value Decomposition of a Matrix</p></a></li>
<li><a href='#tab.disjonctif'><p>Make a disjonctif table</p></a></li>
<li><a href='#tab.disjonctif.prop'><p>Make a disjunctive table when missing values are present</p></a></li>
<li><a href='#tea'><p>tea (data)</p></a></li>
<li><a href='#textual'><p>Text mining</p></a></li>
<li><a href='#wine'><p>Wine</p></a></li>
<li><a href='#write.infile'><p>Print in a file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.10</td>
</tr>
<tr>
<td>Title:</td>
<td>Multivariate Exploratory Data Analysis and Data Mining</td>
</tr>
<tr>
<td>Author:</td>
<td>Francois Husson, Julie Josse, Sebastien Le, Jeremy Mazet</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francois Husson &lt;francois.husson@institut-agro.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>car,cluster,DT,ellipse,emmeans,flashClust,graphics,grDevices,lattice,leaps,MASS,multcompView,scatterplot3d,stats,utils,ggplot2,ggrepel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>missMDA,knitr,Factoshiny,markdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Exploratory data analysis methods to summarize, visualize and describe datasets. The main principal component methods are available, those with the largest potential in terms of applications: principal component analysis (PCA) when variables are quantitative, correspondence analysis (CA) and multiple correspondence analysis (MCA) when variables are categorical, Multiple Factor Analysis when variables are structured in groups, etc. and hierarchical cluster analysis. F. Husson, S. Le and J. Pages (2017).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://factominer.free.fr">http://factominer.free.fr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/husson/FactoMineR">https://github.com/husson/FactoMineR</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-29 18:22:13 UTC; husson</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-29 19:52:39 UTC</td>
</tr>
</table>
<hr>
<h2 id='FactoMineR-package'>Multivariate Exploratory Data Analysis and Data Mining with R
</h2><span id='topic+FactoMineR-package'></span><span id='topic+FactoMineR'></span>

<h3>Description</h3>

<p>The method proposed in this package are exploratory mutlivariate methods such as principal component analysis, correspondence analysis or clustering.
</p>


<h3>Details</h3>

<p>FactoMineR is a package for exploratory multivariate data analysis. The package Factoshiny gives an interface to use most of the functions of FactoMineR.
</p>


<h3>Author(s)</h3>

<p>Francois Husson, Julie Josse, Sebastien Le, Jeremy Mazet
</p>
<p>Maintainer: <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>
</p>


<h3>References</h3>

<p>Le, S., Josse, J. &amp; Husson, F. (2008). FactoMineR: An R Package for Multivariate Analysis. Journal of Statistical Software. 25(1). pp. 1-18.
<a href="https://www.jstatsoft.org/v25/i01/">https://www.jstatsoft.org/v25/i01/</a>
</p>
<p>A website: <a href="http://factominer.free.fr/">http://factominer.free.fr/</a>
</p>
<p>Some videos: <a href="https://www.youtube.com/playlist?list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu">https://www.youtube.com/playlist?list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu</a>
</p>

<hr>
<h2 id='AovSum'>Analysis of variance with the contrasts sum (the sum of the coefficients is 0)</h2><span id='topic+AovSum'></span>

<h3>Description</h3>

<p>Analysis of variance with the contrasts sum (the sum of the coefficients is 0)
<br />
Test for all the coefficients<br />
Handle missing values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AovSum(formula, data, na.action = na.omit, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AovSum_+3A_formula">formula</code></td>
<td>
<p>the formula for the model 'y~x1+x2+x1:x2'</p>
</td></tr>
<tr><td><code id="AovSum_+3A_data">data</code></td>
<td>
<p>a data-frame</p>
</td></tr>
<tr><td><code id="AovSum_+3A_na.action">na.action</code></td>
<td>
<p>(where relevant) information returned by model.frame 
on the special handling of NAs.</p>
</td></tr>
<tr><td><code id="AovSum_+3A_...">...</code></td>
<td>
<p>other arguments, cf the function <code>lm</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Retourne des objets
</p>
<table>
<tr><td><code>Ftest</code></td>
<td>
<p>a table with the F-tests</p>
</td></tr>
<tr><td><code>Ttest</code></td>
<td>
<p>a table with the t-tests</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+aov">aov</a></code>, <code><a href="stats.html#topic+lm">lm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example two-way anova
data(senso)
res &lt;- AovSum(Score~ Product + Day , data=senso)
res

## Example two-way anova with interaction
data(senso)
res2 &lt;- AovSum(Score~ Product + Day + Product : Day, data=senso)
res2

## Example ancova
data(footsize)
res3 &lt;- AovSum(footsize ~ size + sex + size : sex, data=footsize)
res3
</code></pre>

<hr>
<h2 id='autoLab'>Function to better position the labels on the graphs</h2><span id='topic+autoLab'></span>

<h3>Description</h3>

<p>Function to better position the labels on the graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoLab(x, y = NULL, labels = seq(along = x), cex = 1,
                       method = c("SANN", "GA"),
                       allowSmallOverlap = FALSE,
                       trace = FALSE, shadotext = FALSE,
                       doPlot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoLab_+3A_x">x</code></td>
<td>
<p>the x-coordinates</p>
</td></tr>
<tr><td><code id="autoLab_+3A_y">y</code></td>
<td>
<p>the y-coordinates</p>
</td></tr>
<tr><td><code id="autoLab_+3A_labels">labels</code></td>
<td>
<p>the labels</p>
</td></tr>
<tr><td><code id="autoLab_+3A_cex">cex</code></td>
<td>
<p>cex</p>
</td></tr>
<tr><td><code id="autoLab_+3A_method">method</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="autoLab_+3A_allowsmalloverlap">allowSmallOverlap</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="autoLab_+3A_trace">trace</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="autoLab_+3A_shadotext">shadotext</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="autoLab_+3A_doplot">doPlot</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="autoLab_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>See the text function
</p>

<hr>
<h2 id='CA'>Correspondence Analysis (CA)</h2><span id='topic+CA'></span>

<h3>Description</h3>

<p>Performs Correspondence Analysis (CA) including supplementary row and/or column points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CA(X, ncp = 5, row.sup = NULL, col.sup = NULL, 
    quanti.sup=NULL, quali.sup = NULL, graph = TRUE, 
	axes = c(1,2), row.w = NULL, excl=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CA_+3A_x">X</code></td>
<td>
<p>a data frame or a table with <em>n</em> rows and <em>p</em> columns, i.e. a contingency table</p>
</td></tr>
<tr><td><code id="CA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="CA_+3A_row.sup">row.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary rows</p>
</td></tr>
<tr><td><code id="CA_+3A_col.sup">col.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary columns</p>
</td></tr>
<tr><td><code id="CA_+3A_quanti.sup">quanti.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary continuous variables</p>
</td></tr>
<tr><td><code id="CA_+3A_quali.sup">quali.sup</code></td>
<td>
<p>a vector indicating the indexes of the categorical supplementary variables</p>
</td></tr>
<tr><td><code id="CA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="CA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="CA_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, a vector of 1 and each row has a weight equals to its margin); the weights are given only for the active rows</p>
</td></tr>
<tr><td><code id="CA_+3A_excl">excl</code></td>
<td>
<p>numeric vector indicating the indexes of the &quot;junk&quot; columns (default is NULL). Useful for MCA with excl argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>col</code></td>
<td>
<p>a list of matrices with all the results for the column variable (coordinates, square cosine, contributions, inertia)</p>
</td></tr>
<tr><td><code>row</code></td>
<td>
<p>a list of matrices with all the results for the row variable (coordinates, square cosine, contributions, inertia)</p>
</td></tr>
<tr><td><code>col.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary column points (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>row.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary row points (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quanti.sup</code></td>
<td>
<p>if quanti.sup is not NULL, a matrix containing the results for the supplementary continuous variables (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quali.sup</code></td>
<td>
<p>if quali.sup is not NULL, a list of matrices with all the results for the supplementary categorical variables (coordinates of each categories of each variables, v.test which is a criterion with a Normal distribution, square correlation ratio)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>a list with some statistics</p>
</td></tr>
</table>
<p>Returns the row and column points factor map.<br />
The plot may be improved using the argument autolab, modifying the size of the labels or selecting some elements thanks to the <code><a href="#topic+plot.CA">plot.CA</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>,Jeremy Mazet</p>


<h3>References</h3>

<p>Benzecri, J.-P. (1992) <em>Correspondence Analysis Handbook</em>, New-York : Dekker <br />
Benzecri, J.-P. (1980) <em>L'analyse des donnees tome 2 : l'analyse des correspondances</em>, Paris : Bordas <br />
Greenacre, M.J. (1993) <em>Correspondence Analysis in Practice</em>, London : Academic Press<br />
Husson, F., Le, S. and Pages, J. (2009). Analyse de donnees avec R, <em>Presses Universitaires de Rennes</em>.<br />
Husson, F., Le, S. and Pages, J. (2010). Exploratory Multivariate Analysis by Example Using R, <em>Chapman and Hall</em>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.CA">print.CA</a></code>, <code><a href="#topic+summary.CA">summary.CA</a></code>, <code><a href="#topic+ellipseCA">ellipseCA</a></code>, <code><a href="#topic+plot.CA">plot.CA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code>,<br />
<a href="https://www.youtube.com/watch?v=Y-bXsUNX5hA&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=4">Video showing how to perform CA with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(children)
res.ca &lt;- CA (children, row.sup = 15:18, col.sup = 6:8)
summary(res.ca)
## Ellipses for all the active elements
ellipseCA(res.ca)
## Ellipses around some columns only
ellipseCA(res.ca,ellipse="col",col.col.ell=c(rep("blue",2),rep("transparent",3)),
     invisible=c("row.sup","col.sup"))

## Not run: 
## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(children)

## End(Not run)
</code></pre>

<hr>
<h2 id='CaGalt'>Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt)</h2><span id='topic+CaGalt'></span>

<h3>Description</h3>

<p>Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) aims at expanding correspondence analysis on an aggregated lexical table to the case of several quantitative and categorical variables with the objective of establishing a typology of the variables and a typology of the frequencies from their mutual relationships. To avoid the instability issued from multicollinearity among the contextual variables and limit the influence of noisy measurements, the contextual variables are substituted by their principal components. Validation tests in the form of confidence ellipses for the frequencies and the variables are also proposed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CaGalt(Y, X, type="s", conf.ellip=FALSE, nb.ellip=100, level.ventil=0,
  sx=NULL, graph=TRUE, axes=c(1,2))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CaGalt_+3A_y">Y</code></td>
<td>
<p>a data frame with n rows (individuals) and p columns (frequencies)</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_x">X</code></td>
<td>
<p>a data frame with n rows (individuals) and k columns (quantitative or categorical variables)</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_type">type</code></td>
<td>
<p>the type of variables: &quot;c&quot; or &quot;s&quot; for quantitative variables and &quot;n&quot; for categorical variables. The difference is that for &quot;s&quot; variables are scaled to unit variance (by default, variables are scaled to unit variance)</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_conf.ellip">conf.ellip</code></td>
<td>
<p>boolean (FALSE by default), if TRUE, draw confidence ellipses around the frequencies and the variables when &quot;graph&quot; is TRUE</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_nb.ellip">nb.ellip</code></td>
<td>
<p>number of bootstrap samples to compute the confidence ellipses (by default 100)</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_level.ventil">level.ventil</code></td>
<td>
<p>proportion corresponding to the level under which the category is ventilated; by default, 0 and no ventilation is done. Available only when type is equal to &quot;n&quot;</p>
</td></tr>	
<tr><td><code id="CaGalt_+3A_sx">sx</code></td>
<td>
<p>number of principal components kept from the principal axes analysis of the contextual variables (by default is NULL and all principal components are kept)</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="CaGalt_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including: 
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices containing all the results for the individuals (coordinates, square cosine)</p>
</td></tr>  
<tr><td><code>freq</code></td>
<td>
<p>a list of matrices containing all the results for the frequencies (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>quanti.var</code></td>
<td>
<p>a list of matrices containing all the results for the quantitative variables (coordinates, correlation between variables and axes, square cosine)</p>
</td></tr>
<tr><td><code>quali.var</code></td>
<td>
<p>a list of matrices containing all the results for the categorical variables (coordinates of each categories of each variables, square cosine)</p>
</td></tr>
<tr><td><code>ellip</code></td>
<td>
<p>a list of matrices containing the coordinates of the frequencies and variables for replicated samples from which the confidence ellipses are constructed</p>
</td></tr> 
</table>
<p>Returns the individuals, the frequencies and the variables factor map. If there are more than 50 frequencies, the first 50 frequencies that have the highest contribution on the 2 dimensions of your plot are drawn.
The plots may be improved using the argument autolab, modifying the size of the labels or selecting some elements thanks to the plot.CaGalt function. 
</p>


<h3>Author(s)</h3>

<p>Belchin Kostov <a href="mailto:badriyan@clinic.ub.es">badriyan@clinic.ub.es</a>, Monica Becue-Bertaut, Francois Husson</p>


<h3>References</h3>

<p>Becue-Bertaut, M., Pages, J. and Kostov, B. (2014). Untangling the influence of several contextual variables on
the respondents'\ lexical choices. A statistical approach.<em>SORT</em>
Becue-Bertaut, M. and Pages, J. (2014). Correspondence analysis of textual data involving contextual information:
Ca-galt on principal components.<em>Advances in Data Analysis and Classification</em>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.CaGalt">print.CaGalt</a></code>, <code><a href="#topic+summary.CaGalt">summary.CaGalt</a></code>, <code><a href="#topic+plot.CaGalt">plot.CaGalt</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
###Example with categorical variables
data(health)
res.cagalt&lt;-CaGalt(Y=health[,1:115],X=health[,116:118],type="n")

## End(Not run)
</code></pre>

<hr>
<h2 id='catdes'>Categories description</h2><span id='topic+catdes'></span>

<h3>Description</h3>

<p>Description of the categories of one factor by categorical variables and/or by quantitative variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>catdes(donnee,num.var,proba = 0.05, row.w = NULL, na.method="NA")</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="catdes_+3A_donnee">donnee</code></td>
<td>
<p>a data frame made up of at least one categorical variables and a set of quantitative variables and/or categorical variables</p>
</td></tr>
<tr><td><code id="catdes_+3A_num.var">num.var</code></td>
<td>
<p>the indice of the variable to characterized</p>
</td></tr>
<tr><td><code id="catdes_+3A_proba">proba</code></td>
<td>
<p>the significance threshold considered to characterized the category (by default 0.05)</p>
</td></tr>
<tr><td><code id="catdes_+3A_row.w">row.w</code></td>
<td>
<p>a vector of integers corresponding to an optional row weights (by default, a vector of 1 for uniform row weights)</p>
</td></tr>
<tr><td><code id="catdes_+3A_na.method">na.method</code></td>
<td>
<p>a boolean that says how to manage missing values. If ; if na.method=&quot;NA&quot; a new category is considered for the categorical variable; if na.method=&quot;na.omit&quot; the missing values are deleted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>test.chi</code></td>
<td>
<p>The categorical variables which characterized the factor are listed in ascending order (from the one which characterized the most
the factor to the one which significantly characterized with the proba <code>proba</code></p>
</td></tr>
<tr><td><code>category</code></td>
<td>
<p>description of each category of the <code>num.var</code> by each category of all the categorical variables</p>
</td></tr>
<tr><td><code>quanti.var</code></td>
<td>
<p>the global description of the <code>num.var</code> variable by the quantitative variables with the square correlation coefficient and the p-value of the F-test in a one-way analysis of variance (assuming the hypothesis of homoscedsticity)</p>
</td></tr>
<tr><td><code>quanti</code></td>
<td>
<p>the description of each category of the <code>num.var</code> variable by the quantitative variables.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Husson, F., Le, S. and Pages, J. (2010). Exploratory Multivariate Analysis by Example Using R, <em>Chapman and Hall</em>.
Lebart, L., Morineau, A. and Piron, M. (1995) Statistique exploratoire multidimensionnelle, <em>Dunod</em>.<br />
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plot.catdes">plot.catdes</a></code>, <code><a href="#topic+condes">condes</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
catdes(wine, num.var=2)
</code></pre>

<hr>
<h2 id='children'>Children (data)</h2><span id='topic+children'></span>

<h3>Description</h3>

<p>The data used here is a contingency table that summarizes the answers given
by different categories of people to the 
following question : according to you, 
what are the reasons that can make hesitate a woman or a couple to have 
children?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(children)</code></pre>


<h3>Format</h3>

<p>A data frame with 18 rows and 8 columns. Rows represent the different reasons mentioned,
columns represent the different categories (education, age) people belong to.
</p>


<h3>Source</h3>

<p>Traitements Statistiques des Enquetes (D. Grange, L. Lebart, eds.) Dunod, 1993
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(children)
res.ca &lt;- CA (children, row.sup = 15:18, col.sup = 6:8)
</code></pre>

<hr>
<h2 id='coeffRV'>Calculate the RV coefficient and test its significance</h2><span id='topic+coeffRV'></span>

<h3>Description</h3>

<p>Calculate the RV coefficient and test its significance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeffRV(X, Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeffRV_+3A_x">X</code></td>
<td>
<p>a matrix with <em>n</em> rows (individuals) and <em>p</em> numerous columns (variables)</p>
</td></tr>
<tr><td><code id="coeffRV_+3A_y">Y</code></td>
<td>
<p>a matrix with <em>n</em> rows (individuals) and <em>p</em> numerous columns (variables)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the RV coefficient between <code>X</code> and <code>Y</code>. It returns also the standardized RV, 
the expectation, the variance and the skewness under the permutation
distribution. These moments are used to approximate the exact
distribution of the RV statistic with the Pearson type III approximation and the p-value associated to this test is given.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr><td><code>RV</code></td>
<td>
<p>the RV coefficient between the two matrices</p>
</td></tr>
<tr><td><code>RVs</code></td>
<td>
<p>the standardized RV coefficients</p>
</td></tr>
<tr><td><code>mean</code></td>
<td>
<p>the mean of the RV permutation distribution</p>
</td></tr>
<tr><td><code>variance</code></td>
<td>
<p>the variance of the RV permutation distribution</p>
</td></tr>
<tr><td><code>skewness</code></td>
<td>
<p>the skewness of the RV permutation distribution</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value associated to the test of the significativity of the RV coefficient (with the Pearson type III approximation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Julie Josse, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Escouffier, Y. (1973) <em>Le traitement des variables vectorielles</em>. Biometrics <b>29</b> 751&ndash;760.<br />
Josse, J., Husson, F., Pag\'es, J. (2007) <em>Testing the significance of the RV coefficient</em>. 
Computational Statististics and Data Analysis. <b>53</b> 82&ndash;91.<br />
Kazi-Aoual, F., Hitier, S., Sabatier, R., Lebreton, J.-D., (1995) Refined approximations to permutations tests 
for multivariate inference. Computational Statistics and Data Analysis, <b>20</b>, 643&ndash;656 <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X &lt;- wine[,3:7]
Y &lt;- wine[,11:20]
coeffRV(X,Y)
</code></pre>

<hr>
<h2 id='condes'>Continuous variable description</h2><span id='topic+condes'></span>

<h3>Description</h3>

<p>Description continuous by quantitative variables and/or by categorical variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>condes(donnee,num.var,weights=NULL,proba = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="condes_+3A_donnee">donnee</code></td>
<td>
<p>a data frame made up of at least one quantitative variable and a set of quantitative variables and/or categorical variables</p>
</td></tr>
<tr><td><code id="condes_+3A_num.var">num.var</code></td>
<td>
<p>the number of the variable to characterized</p>
</td></tr>
<tr><td><code id="condes_+3A_weights">weights</code></td>
<td>
<p>weights for the individuals; if NULL, all individuals has a weight equals to 1; the sum of the weights can be equal to 1 and then the weights will be multiplied by the number of individuals, the sum can be greater than the number of individuals</p>
</td></tr>
<tr><td><code id="condes_+3A_proba">proba</code></td>
<td>
<p>the significance threshold considered to characterized the category (by default 0.05)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>quanti</code></td>
<td>
<p>the description of the <code>num.var</code> variable by the quantitative variables. The variables are sorted in ascending order  (from the one which characterized the most to the one which significantly characterized with the proba <code>proba</code>)</p>
</td></tr>
<tr><td><code>quali</code></td>
<td>
<p>The categorical variables which characterized the continuous variables are listed in ascending order</p>
</td></tr>
<tr><td><code>category</code></td>
<td>
<p>description of the continuous variable <code>num.var</code> by each category of all the categorical variables</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+catdes">catdes</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
condes(decathlon, num.var=3)
</code></pre>

<hr>
<h2 id='coord.ellipse'>Construct confidence ellipses</h2><span id='topic+coord.ellipse'></span>

<h3>Description</h3>

<p>Construct confidence ellipses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord.ellipse (coord.simul, centre = NULL, axes = c(1, 2), 
    level.conf = 0.95, npoint = 100, bary = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coord.ellipse_+3A_coord.simul">coord.simul</code></td>
<td>
<p>a data frame containing the coordinates of the individuals
for which the confidence ellipses are constructed. This data frame can contain more than 2 variables;
the variables taken into account are chosen after. The first column must be a factor which allows
to associate one row to an ellipse. The simule object of the result of the simule function
correspond to a data frame.</p>
</td></tr>
<tr><td><code id="coord.ellipse_+3A_centre">centre</code></td>
<td>
<p>a data frame whose columns are the same than those of the coord.simul, and with the coordinates
of the centre of each ellipse. This parameter is optional and NULL by default; in this case,
the centre of the ellipses is calculated from the data</p>
</td></tr>
<tr><td><code id="coord.ellipse_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components of coord.simul that are taken into account</p>
</td></tr>
<tr><td><code id="coord.ellipse_+3A_level.conf">level.conf</code></td>
<td>
<p>confidence level used to construct the ellipses. By default, 0.95</p>
</td></tr>
<tr><td><code id="coord.ellipse_+3A_npoint">npoint</code></td>
<td>
<p>number of points used to draw the ellipses</p>
</td></tr>
<tr><td><code id="coord.ellipse_+3A_bary">bary</code></td>
<td>
<p>boolean, if bary = TRUE, the coordinates of the ellipse around the barycentre of individuals are calculated</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>res</code></td>
<td>
<p>a data frame with (npoint times the number of ellipses) rows and three columns. The first column
is the factor of coord.simul, the two others columns  give the coordinates of the ellipses on the two dimensions chosen.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the parameters of the function chosen</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet</p>


<h3>See Also</h3>

 <p><code><a href="#topic+simule">simule</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup = 13,graph=FALSE)
aa &lt;- cbind.data.frame(decathlon[,13],res.pca$ind$coord)
bb &lt;- coord.ellipse(aa,bary=TRUE)
plot(res.pca,habillage=13,ellipse=bb)

## To automatically draw ellipses around the barycentres of all the categorical variables
plotellipses(res.pca)
</code></pre>

<hr>
<h2 id='decathlon'>Performance in decathlon (data)</h2><span id='topic+decathlon'></span>

<h3>Description</h3>

<p>The data used here refer to athletes' performance during two sporting events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(decathlon)</code></pre>


<h3>Format</h3>

<p>A data frame with 41 rows and 13 columns: the first ten columns corresponds to 
the performance of the athletes for the 10 events of the decathlon. 
The columns 11 and 12 correspond respectively to the rank and the points obtained. 
The last column is a categorical variable
corresponding to the sporting event (2004 Olympic Game or 2004 Decastar)
</p>


<h3>Source</h3>

<p>Department of statistics and computer science, Agrocampus Rennes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup=13)
</code></pre>

<hr>
<h2 id='descfreq'>Description of frequencies</h2><span id='topic+descfreq'></span>

<h3>Description</h3>

<p>Description of the rows of a contingency table or of groups of rows of a contingency table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descfreq(donnee, by.quali = NULL, proba = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="descfreq_+3A_donnee">donnee</code></td>
<td>
<p>a data frame corresponding to a contingency table (quantitative data)</p>
</td></tr>
<tr><td><code id="descfreq_+3A_by.quali">by.quali</code></td>
<td>
<p>a factor used to merge the data from different rows of the contingency table; by default NULL and each row is characterized</p>
</td></tr>
<tr><td><code id="descfreq_+3A_proba">proba</code></td>
<td>
<p>the significance threshold considered to characterized the category (by default 0.05)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the characterization of each rows or each group of the <code>by.quali</code>. A test corresponding to the hypergeometric distribution is performed and the probability to observe a more extreme value than the one observed is calculated. For each row (or category), each of the columns characterising the row are sorted in ascending order of p-value.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Lebart, L., Morineau, A. and Piron, M. (1995) Statistique exploratoire multidimensionnelle, <em>Dunod</em>.<br />
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+catdes">catdes</a></code>, <code><a href="#topic+condes">condes</a></code>, <code><a href="#topic+textual">textual</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(children)
descfreq(children[1:14,1:5])    ## desc of rows
descfreq(t(children[1:14,1:5])) ## desc of columns
</code></pre>

<hr>
<h2 id='dimdesc'>Dimension description</h2><span id='topic+dimdesc'></span>

<h3>Description</h3>

<p>This function is designed to point out the variables and the categories that are the most characteristic according 
to each dimension obtained by a Factor Analysis. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dimdesc(res, axes = 1:3, proba = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dimdesc_+3A_res">res</code></td>
<td>
<p>an object of class PCA, MCA, CA, MFA or HMFA</p>
</td></tr>
<tr><td><code id="dimdesc_+3A_axes">axes</code></td>
<td>
<p>a vector with the dimensions to describe</p>
</td></tr>
<tr><td><code id="dimdesc_+3A_proba">proba</code></td>
<td>
<p>the significance threshold considered to characterized the dimension (by default 0.05)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>quanti</code></td>
<td>
<p>the description of the dimensions by the quantitative variables. The variables are sorted.</p>
</td></tr>
<tr><td><code>quali</code></td>
<td>
<p>the description of the dimensions by the categorical variables</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Husson, F., Le, S. and Pages, J. (2010). Exploratory Multivariate Analysis by Example Using R, <em>Chapman and Hall</em>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code>, <code><a href="#topic+CA">CA</a></code>, <code><a href="#topic+MCA">MCA</a></code>, <code><a href="#topic+MFA">MFA</a></code>, <code><a href="#topic+HMFA">HMFA</a></code>,<br />
<a href="https://www.youtube.com/watch?v=Uhw-1NilmAk&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=9">Video showing how to use this function</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup=13, graph=FALSE)
dimdesc(res.pca)
</code></pre>

<hr>
<h2 id='DMFA'>Dual Multiple Factor Analysis (DMFA)</h2><span id='topic+DMFA'></span>

<h3>Description</h3>

<p>Performs Dual Multiple Factor Analysis (DMFA) with supplementary individuals, supplementary quantitative 
variables and supplementary categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DMFA(don, num.fact = ncol(don), scale.unit = TRUE, ncp = 5, 
    quanti.sup = NULL, quali.sup = NULL, graph = TRUE, axes=c(1,2))</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DMFA_+3A_don">don</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns (numeric variables)</p>
</td></tr>
<tr><td><code id="DMFA_+3A_num.fact">num.fact</code></td>
<td>
<p>the number of the categorical variable which allows to make the group of individuals</p>
</td></tr>
<tr><td><code id="DMFA_+3A_scale.unit">scale.unit</code></td>
<td>
<p>a boolean, if TRUE (value set by default) then data are scaled to unit variance</p>
</td></tr>
<tr><td><code id="DMFA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="DMFA_+3A_quanti.sup">quanti.sup</code></td>
<td>
<p>a vector indicating the indexes of the quantitative supplementary variables</p>
</td></tr>
<tr><td><code id="DMFA_+3A_quali.sup">quali.sup</code></td>
<td>
<p>a vector indicating the indexes of the categorical supplementary variables</p>
</td></tr>
<tr><td><code id="DMFA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="DMFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>a list of matrices containing all the results for the active variables (coordinates, correlation between variables and axes, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices containing all the results for the active individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary individuals (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quanti.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary quantitative variables (coordinates, correlation between variables and axes)</p>
</td></tr>
<tr><td><code>quali.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary categorical variables (coordinates of each categories of each variables, and v.test which is a criterion with a Normal distribution)</p>
</td></tr>
<tr><td><code>svd</code></td>
<td>
<p>the result of the singular value decomposition</p>
</td></tr>
<tr><td><code>var.partiel</code></td>
<td>
<p>a list with the partial coordinate of the variables for each group</p>
</td></tr>
<tr><td><code>cor.dim.gr</code></td>
<td>
</td></tr>
<tr><td><code>Xc</code></td>
<td>
<p>a list with the data centered by group</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>a list with the results for the groups (cordinate, normalized coordinates, cos2)</p>
</td></tr>
<tr><td><code>Cov</code></td>
<td>
<p>a list with the covariance matrices for each group</p>
</td></tr>
</table>
<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+plot.DMFA">plot.DMFA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example with the famous Fisher's iris data
res.dmfa = DMFA ( iris, num.fact = 5)
</code></pre>

<hr>
<h2 id='ellipseCA'>Draw confidence ellipses in CA</h2><span id='topic+ellipseCA'></span>

<h3>Description</h3>

<p>Draw confidence ellipses in CA around rows and/or columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ellipseCA (x, ellipse=c("col","row"), method="multinomial", nbsample=100,
    axes=c(1,2), xlim=NULL, ylim=NULL, col.row="blue", col.col="red",
	col.row.ell=col.row, col.col.ell=col.col, 
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ellipseCA_+3A_x">x</code></td>
<td>
<p>an object of class CA</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_ellipse">ellipse</code></td>
<td>
<p>a vector of character that defines which ellipses are drawn</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_method">method</code></td>
<td>
<p>the method to construct ellipses (see details below)</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_nbsample">nbsample</code></td>
<td>
<p>number of samples drawn to evaluate the stability of the points</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_col.row">col.row</code></td>
<td>
<p>a color for the rows points</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_col.col">col.col</code></td>
<td>
<p>a color for columns points</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_col.row.ell">col.row.ell</code></td>
<td>
<p>a color for the ellipses of rows points (the color &quot;transparent&quot; can be used if an ellipse should not be drawn)</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_col.col.ell">col.col.ell</code></td>
<td>
<p>a color for the ellipses of columns points (the color &quot;transparent&quot; can be used if an ellipse should not be drawn)</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="ellipseCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from the plot.CA function, such as title, invisible, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With <code>method="multinomial"</code>, the table X with the active elements is taken as a reference. Then new data tables are drawn in the following way: 
N (the sum of X) values are drawn from a multinomial distribution with theoretical frequencies equals to the values in the cells divided by N.<br />
</p>
<p>With <code>method="boot"</code>, the values are bootstrapped row by row: Ni (the sum of row i in the X table) values are taken in a vector with Nij equals to column j (with j varying from 1 to J).
</p>
<p>Thus <code>nbsample</code> new datasets are drawn and projected as supplementary rows and/or supplementary columns. Then confidence ellipses are drawn for each elements thanks to the <code>nbsample</code> supplementary points.
</p>


<h3>Value</h3>

<p>Returns the factor map with the joint plot of CA with ellipses around some elements.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Lebart, L., Morineau, A. and Piron, M. (1995) Statistique exploratoire multidimensionnelle, <em>Dunod</em>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plot.CA">plot.CA</a></code>, <code><a href="#topic+CA">CA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(children)
res.ca &lt;- CA (children, col.sup = 6:8, row.sup = 15:18)
## Ellipses for all the active elements
ellipseCA(res.ca)
## Ellipses around some columns only
ellipseCA(res.ca,ellipse="col",col.col.ell=c(rep("red",2),rep("transparent",3)),
     invisible=c("row.sup","col.sup"))
</code></pre>

<hr>
<h2 id='estim_ncp'>Estimate the number of components in Principal Component Analysis</h2><span id='topic+estim_ncp'></span>

<h3>Description</h3>

<p>Estimate the number of components in PCA .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estim_ncp(X, ncp.min=0, ncp.max=NULL, scale=TRUE, method="GCV")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estim_ncp_+3A_x">X</code></td>
<td>
<p>a data frame with continuous variables</p>
</td></tr>
<tr><td><code id="estim_ncp_+3A_ncp.min">ncp.min</code></td>
<td>
<p>minimum number of dimensions to interpret, by default 0</p>
</td></tr>
<tr><td><code id="estim_ncp_+3A_ncp.max">ncp.max</code></td>
<td>
<p>maximum number of dimensions to interpret, by default NULL which corresponds to the number of columns minus 2</p>
</td></tr>
<tr><td><code id="estim_ncp_+3A_scale">scale</code></td>
<td>
<p>a boolean, if TRUE (value set by default) then data are scaled to unit variance</p>
</td></tr>
<tr><td><code id="estim_ncp_+3A_method">method</code></td>
<td>
<p>method used to estimate the number of components, &quot;GCV&quot; for the generalized cross-validation approximation or &quot;Smooth&quot; for the smoothing method (by default &quot;GCV&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns ncp the best number of dimensions to use (find the minimum or the first local minimum) and the 
mean error for each dimension tested
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Julie Josse<a href="mailto:Julie.Josse@agrocampus-ouest.fr">Julie.Josse@agrocampus-ouest.fr</a></p>


<h3>References</h3>

<p>Josse, J. and Husson, F. (2012). Selecting the number of components in PCA using cross-validation approximations. Computational Statistics and Data Analysis, 56, 1869-1879.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
nb.dim &lt;- estim_ncp(decathlon[,1:10],scale=TRUE)
</code></pre>

<hr>
<h2 id='FAMD'>Factor Analysis for Mixed Data</h2><span id='topic+FAMD'></span>

<h3>Description</h3>

<p>FAMD is a principal component method dedicated to explore data with both continuous and categorical variables. It can be seen roughly as a mixed between PCA and MCA. More precisely, the continuous variables are scaled to unit variance and the categorical variables are transformed into a disjunctive data table (crisp coding) and then scaled using the specific scaling of MCA. This ensures to balance the influence of both continous and categorical variables in the analysis. It means that both variables are on a equal foot to determine the dimensions of variability. This method allows one to study the similarities between individuals taking into account mixed variables and to study the relationships between all the variables. It also provides graphical outputs such as the representation of the individuals, the correlation circle for the continuous variables and representations of the categories of the categorical variables, and also specific graphs to visulaize the associations between both type of variables. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FAMD (base, ncp = 5, graph = TRUE, sup.var = NULL, 
    ind.sup = NULL, axes = c(1,2), row.w = NULL, tab.disj = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FAMD_+3A_base">base</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns</p>
</td></tr>
<tr><td><code id="FAMD_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="FAMD_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="FAMD_+3A_ind.sup">ind.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary individuals</p>
</td></tr>
<tr><td><code id="FAMD_+3A_sup.var">sup.var</code></td>
<td>
<p>a vector indicating the indexes of the supplementary variables</p>
</td></tr>
<tr><td><code id="FAMD_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="FAMD_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, uniform row weights); the weights are given only for the active individuals</p>
</td></tr>
<tr><td><code id="FAMD_+3A_tab.disj">tab.disj</code></td>
<td>
<p>object obtained from the imputeFAMD function of the missMDA package that allows to handle missing values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>a list of matrices containing all the results for the variables considered as group (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices with all the results for the individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>quali.var</code></td>
<td>
<p>a list of matrices with all the results for the categorical variables (coordinates, square cosine, contributions, v.test)</p>
</td></tr>
<tr><td><code>quanti.var</code></td>
<td>
<p>a list of matrices with all the results for the quantitative variables (coordinates, correlation, square cosine, contributions)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>a list with some statistics</p>
</td></tr>
</table>
<p>Returns the individuals factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Pages J. (2004). Analyse factorielle de donnees mixtes. Revue Statistique Appliquee.  LII (4). pp. 93-111.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.FAMD">print.FAMD</a></code>, <code><a href="#topic+summary.FAMD">summary.FAMD</a></code>, <code><a href="#topic+plot.FAMD">plot.FAMD</a></code>,<br />
<a href="https://www.youtube.com/watch?v=FKB96VGUgUE&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=8">Video showing how to perform FAMD with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(geomorphology)
res &lt;- FAMD(geomorphology)
summary(res)

## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(geomorphology)

### with missing values
require(missMDA)
data(ozone)
res.impute &lt;- imputeFAMD(ozone, ncp=3) 
res.afdm &lt;- FAMD(ozone,tab.disj=res.impute$tab.disj) 

## End(Not run)
</code></pre>

<hr>
<h2 id='footsize'>footsize</h2><span id='topic+footsize'></span>

<h3>Description</h3>

<p>Dataset for the covariance analysis (a quantitative variable explained by quantitative (continuous) and qualitative (categorical) variables)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(footsize)</code></pre>


<h3>Format</h3>

<p>Dataset with 84 rows and 3 columns:
footsize, size and sex</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(footsize)
res3 &lt;- AovSum (footsize ~ size + sex + size :sex, data=footsize)
res3
</code></pre>

<hr>
<h2 id='geomorphology'>geomorphology(data)</h2><span id='topic+geomorphology'></span>

<h3>Description</h3>

<p>The data used here concern a geomorphology analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(geomorphology)</code></pre>


<h3>Format</h3>

<p>A data frame with 75 rows and 11 columns. Rows represent the individuals,
columns represent the different questions. 10 variables are quantitative and one variable is qualitative.
The dataset is analysed in: http://www.sciencedirect.com/science/article/pii/S0169555X11006362
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(geomorphology)
res &lt;- FAMD(geomorphology)
plot(res,choix="ind",habillage=4)

## End(Not run)
</code></pre>

<hr>
<h2 id='GPA'>Generalised Procrustes Analysis</h2><span id='topic+GPA'></span>

<h3>Description</h3>

<p>Performs Generalised Procrustes Analysis (GPA) that takes into account missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GPA(df, tolerance=10^-10, nbiteration=200, scale=TRUE, 
    group, name.group = NULL, graph = TRUE, axes = c(1,2))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GPA_+3A_df">df</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns (quantitative varaibles)</p>
</td></tr>
<tr><td><code id="GPA_+3A_tolerance">tolerance</code></td>
<td>
<p>a threshold with respect to which the algorithm stops, i.e. when the difference between 
the GPA loss function at step <em>n</em> and <em>n+1</em> is less than <code>tolerance</code></p>
</td></tr>
<tr><td><code id="GPA_+3A_nbiteration">nbiteration</code></td>
<td>
<p>the maximum number of iterations until the algorithm stops</p>
</td></tr>
<tr><td><code id="GPA_+3A_scale">scale</code></td>
<td>
<p>a boolean, if TRUE (which is the default value) scaling is required</p>
</td></tr>
<tr><td><code id="GPA_+3A_group">group</code></td>
<td>
<p>a vector indicating the number of variables in each group</p>
</td></tr>
<tr><td><code id="GPA_+3A_name.group">name.group</code></td>
<td>
<p>a vector indicating the name of the groups (the groups are successively named 
group.1, group.2 and so on, by default)</p>
</td></tr>
<tr><td><code id="GPA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="GPA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a Generalised Procrustes Analysis (GPA) that takes into account missing values:
some data frames of <code>df</code> may have non described or non evaluated rows, i.e. rows with missing 
values only.<br />
The algorithm used here is the one developed by Commandeur.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr><td><code>RV</code></td>
<td>
<p>a matrix of RV coefficients between partial configurations</p>
</td></tr>
<tr><td><code>RVs</code></td>
<td>
<p>a matrix of standardized RV coefficients between partial configurations</p>
</td></tr>
<tr><td><code>simi</code></td>
<td>
<p>a matrix of Procrustes similarity indexes between partial configurations</p>
</td></tr>
<tr><td><code>scaling</code></td>
<td>
<p>a vector of isotropic scaling factors</p>
</td></tr>
<tr><td><code>dep</code></td>
<td>
<p>an array of initial partial configurations </p>
</td></tr>
<tr><td><code>consensus</code></td>
<td>
<p>a matrix of consensus configuration</p>
</td></tr>
<tr><td><code>Xfin</code></td>
<td>
<p>an array of partial configurations after transformations</p>
</td></tr>
<tr><td><code>correlations</code></td>
<td>
<p>correlation matrix between initial partial configurations and consensus dimensions</p>
</td></tr>
<tr><td><code>PANOVA</code></td>
<td>
<p>a list of &quot;Procrustes Analysis of Variance&quot; tables, per assesor (config), per product(objet), per dimension (dimension)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Elisabeth Morand</p>


<h3>References</h3>

<p>Commandeur, J.J.F (1991) <em>Matching configurations</em>.DSWO press, Leiden University.<br />
Dijksterhuis, G. &amp; Punter, P. (1990) Interpreting generalized procrustes analysis &quot;Analysis of Variance&quot; tables, 
<em>Food Quality and Preference</em>, <b>2</b>, 255&ndash;265<br />
Gower, J.C (1975) Generalized Procrustes analysis, <em>Psychometrika</em>, <b>40</b>, 33&ndash;50 <br />
Kazi-Aoual, F., Hitier, S., Sabatier, R., Lebreton, J.-D., (1995) Refined approximations to permutations tests 
for multivariate inference. Computational Statistics and Data Analysis, <b>20</b>, 643&ndash;656 <br />
Qannari, E.M., MacFie, H.J.H, Courcoux, P. (1999) Performance indices
and isotropic scaling factors in sensory profiling, <em>Food Quality and Preference</em>, <b>10</b>, 17&ndash;21
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res.gpa &lt;- GPA(wine[,-(1:2)], group=c(5,3,10,9,2),
    name.group=c("olf","vis","olfag","gust","ens"))

### If you want to construct the partial points for some individuals only
plotGPApartial (res.gpa)

## End(Not run)
</code></pre>

<hr>
<h2 id='graph.var'>Make graph of variables</h2><span id='topic+graph.var'></span>

<h3>Description</h3>

<p>Plot the graphs of the variables after  a Factor Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph.var(x, axes = c(1, 2), 
    xlim = NULL, ylim = NULL, col.sup = "blue", 
    col.var = "black", draw="all", label=draw, lim.cos2.var = 0.1,
    cex = 1, title = NULL, new.plot = TRUE, ...)
    </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph.var_+3A_x">x</code></td>
<td>
<p>an object of class PCA, MCA, MFA or HMFA</p>
</td></tr>
<tr><td><code id="graph.var_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="graph.var_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="graph.var_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="graph.var_+3A_col.sup">col.sup</code></td>
<td>
<p>a color for the quantitative supplementary variables</p>
</td></tr>
<tr><td><code id="graph.var_+3A_col.var">col.var</code></td>
<td>
<p>a color for the variables</p>
</td></tr>
<tr><td><code id="graph.var_+3A_draw">draw</code></td>
<td>
<p>a list of character for the variables which are drawn (by default, all the variables are drawn). You can draw all the active variables
by putting &quot;var&quot; and/or all the supplementary variables by putting &quot;quanti.sup&quot; and/or a list with the names of the variables which should be drawn</p>
</td></tr>
<tr><td><code id="graph.var_+3A_label">label</code></td>
<td>
<p>a list of character for the variables which are labelled (by default, all the drawn variables are labelled). You can label all the active variables
by putting &quot;var&quot; and/or all the supplementary variables by putting &quot;quanti.sup&quot; and/or a list with the names of the variables which should be labelled</p>
</td></tr>
<tr><td><code id="graph.var_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under the variables are not drawn</p>
</td></tr>
<tr><td><code id="graph.var_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="graph.var_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="graph.var_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="graph.var_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code>, <code><a href="#topic+MFA">MFA</a></code>, <code><a href="#topic+MCA">MCA</a></code>, <code><a href="#topic+DMFA">DMFA</a></code>, <code><a href="#topic+HMFA">HMFA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup = 13, graph = FALSE)
graph.var (res.pca, draw = c("var","Points"), 
    label = c("Long.jump", "Points"))
</code></pre>

<hr>
<h2 id='HCPC'>Hierarchical Clustering on Principle Components (HCPC)</h2><span id='topic+HCPC'></span>

<h3>Description</h3>

<p>Performs an agglomerative hierarchical clustering on results from a
factor analysis. It is possible to cut the tree by clicking at the
suggested (or an other) level. Results include paragons, description
of the clusters, graphics.</p>


<h3>Usage</h3>

<pre><code class='language-R'>HCPC(res, nb.clust=0, consol=TRUE, iter.max=10, min=3, 
  max=NULL, metric="euclidean", method="ward", order=TRUE,
  graph.scale="inertia", nb.par=5, graph=TRUE, proba=0.05, 
  cluster.CA="rows",kk=Inf,description=TRUE,...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HCPC_+3A_res">res</code></td>
<td>
<p>Either the result of a factor analysis or a dataframe.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_nb.clust">nb.clust</code></td>
<td>
<p>an integer. If 0, the tree is cut at the level the user clicks
on. If -1, the tree is automatically cut at the suggested level (see
details). If a (positive) integer, the tree is cut with nb.cluters clusters.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_consol">consol</code></td>
<td>
<p>a boolean. If TRUE, a k-means consolidation is performed (consolidation cannot be performed if kk is used and equals a number).</p>
</td></tr>
<tr><td><code id="HCPC_+3A_iter.max">iter.max</code></td>
<td>
<p>An integer. The maximum number of iterations for the consolidation.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_min">min</code></td>
<td>
<p>an integer. The least possible number of clusters suggested.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_max">max</code></td>
<td>
<p>an integer. The higher possible number of clusters suggested; by default the minimum between 10 and the number of individuals divided by 2.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_metric">metric</code></td>
<td>
<p>The metric used to built the tree. See <code><a href="cluster.html#topic+agnes">agnes</a></code> for details.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_method">method</code></td>
<td>
<p>The method used to built the tree. See <code><a href="cluster.html#topic+agnes">agnes</a></code> for details.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_order">order</code></td>
<td>
<p>A boolean. If TRUE, clusters are ordered following their center
coordinate on the first axis.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_graph.scale">graph.scale</code></td>
<td>
<p>A character string. By default &quot;inertia&quot; and the height of the tree corresponds
to the inertia gain, else &quot;sqrt-inertia&quot; the square root of the inertia gain.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_nb.par">nb.par</code></td>
<td>
<p>An integer. The number of edited paragons.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_graph">graph</code></td>
<td>
<p>If TRUE, graphics are displayed. If FALSE, no graph are displayed.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_proba">proba</code></td>
<td>
<p>The probability used to select axes and variables in
catdes (see <code><a href="#topic+catdes">catdes</a></code> for details.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_cluster.ca">cluster.CA</code></td>
<td>
<p>A string equals to &quot;rows&quot; or &quot;columns&quot; for the clustering 
of Correspondence Analysis results.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_kk">kk</code></td>
<td>
<p>An integer corresponding to the number of clusters used in a Kmeans 
preprocessing before the hierarchical clustering;   the top of the hierarchical 
tree is then constructed from this partition. This is very useful if the number 
of individuals is high. Note that consolidation cannot be performed if kk is different 
from Inf and some graphics are not drawn. Inf is used by default and 
no preprocessing is done, all the graphical outputs are then given.</p>
</td></tr>
<tr><td><code id="HCPC_+3A_description">description</code></td>
<td>
<p>boolean; if TRUE the clusters are characterized by the variables and the dimensions</p>
</td></tr>
<tr><td><code id="HCPC_+3A_...">...</code></td>
<td>
<p>Other arguments from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function first built a hierarchical tree. Then the sum of the
within-cluster inertia are calculated for each partition. The suggested
partition is the one with the higher relative loss of inertia
(i(clusters n+1)/i(cluster n)).
</p>
<p>The absolute loss of inertia (i(cluster n)-i(cluster n+1)) is plotted
with the tree.  
</p>
<p>If the ascending clustering is constructed from a data-frame with a lot of 
rows (individuals), it is possible to first perform a partition with 
kk clusters and then construct the tree from the (weighted) kk clusters. 
</p>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>data.clust</code></td>
<td>
<p>The original data with a supplementary column called
clust containing the partition.</p>
</td></tr>
<tr><td><code>desc.var</code></td>
<td>
<p>The description of the classes by the variables.
See <code><a href="#topic+catdes">catdes</a></code> for details or <code><a href="#topic+descfreq">descfreq</a></code> if clustering
is performed on CA results.</p>
</td></tr>
<tr><td><code>desc.axes</code></td>
<td>
<p>The description of the classes by the factors (axes). 
See <code><a href="#topic+catdes">catdes</a></code> for details.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>A list or parameters and internal objects. <code>call$t</code> gives the results for the hierarchical tree; 
<code>call$bw.before.consol</code> and <code>call$bw.after.consol</code> give the between inertia before consolidation (i.e. for the clustering obtained
from the hierarchical tree) and after the consolidation with Kmeans.</p>
</td></tr>
<tr><td><code>desc.ind</code></td>
<td>
<p>The paragons (para) and the more typical individuals
of each cluster. See details.</p>
</td></tr>
</table>
<p>Returns the tree and a barplot of the inertia gains, the individual
factor map with the tree (3D), the factor map with individuals coloured
by cluster (2D).
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Guillaume Le Ray, Quentin Molto</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plot.HCPC">plot.HCPC</a></code>, <code><a href="#topic+catdes">catdes</a></code>,<br />
<a href="https://www.youtube.com/watch?v=4XrgWmN9erg&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=7">Video showing how to perform clustering with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(iris)
# Principal Component Analysis:
res.pca &lt;- PCA(iris[,1:4], graph=FALSE)
# Clustering, auto nb of clusters:
hc &lt;- HCPC(res.pca, nb.clust=-1)

### Construct a hierarchical tree from a partition (with 10 clusters)
### (useful when the number of individuals is very important)
hc2 &lt;- HCPC(iris[,1:4], kk=10, nb.clust=-1)

## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(iris[,1:4])

## End(Not run)
</code></pre>

<hr>
<h2 id='health'>health (data)</h2><span id='topic+health'></span>

<h3>Description</h3>

<p>In 1989-1990 the Valencian Institute of Public Health (IVESP) conducted a survey to better know the attitudes and opinions related to health for the non-expert population. The first question included in the questionnaire &quot;What does health mean to you?&quot; required free and spontaneous answers. A priori, the variables Age group (under 21, 21-35, 36-50 and over 50), Health condition (poor, fair, good and very good health) and Gender were considered as possibly conditioning the respondents' viewpoint on health.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(health)
</code></pre>


<h3>Format</h3>

<p>A data frame with 392 rows and 118 columns. Rows represent the individuals (respondents), columns represent the words used at least 10 times to answer the open-ended question (columns 1 to 115) and respondents' characteristics (age, health condition and gender)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(health)
res.cagalt&lt;-CaGalt(Y=health[,1:115],X=health[,116:118],type="n")

## End(Not run)
</code></pre>

<hr>
<h2 id='HMFA'>Hierarchical Multiple Factor Analysis</h2><span id='topic+HMFA'></span>

<h3>Description</h3>

<p>Performs a hierarchical multiple factor analysis, using an object of class <code>list</code> of <code>data.frame</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HMFA(X,H,type = rep("s", length(H[[1]])), ncp = 5, graph = TRUE,
    axes = c(1,2), name.group = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HMFA_+3A_x">X</code></td>
<td>
<p>a <code>data.frame</code></p>
</td></tr>
<tr><td><code id="HMFA_+3A_h">H</code></td>
<td>
<p>a list with one vector for each hierarchical level; in each vector the number of variables or the number of group
constituting the group</p>
</td></tr>
<tr><td><code id="HMFA_+3A_type">type</code></td>
<td>
<p>the type of variables in each group in the first partition; three possibilities: &quot;c&quot; or &quot;s&quot; for quantitative variables
(the difference is that for &quot;s&quot;, the variables are scaled in the program), &quot;n&quot; for categorical variables; by default, all the variables
are quantitative and the variables are scaled unit</p>
</td></tr>
<tr><td><code id="HMFA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="HMFA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="HMFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="HMFA_+3A_name.group">name.group</code></td>
<td>
<p>a list of vector containing the name of the groups for each level of the hierarchy (by default, NULL and the 
group are named L1.G1, L1.G2 and so on)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>a list with first a list of matrices with the coordinates of the groups for each level and second a matrix with the canonical correlation (correlation between the coordinates of the individuals and the partial points))</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices with all the results for the active individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>quanti.var</code></td>
<td>
<p>a list of matrices with all the results for the quantitative variables (coordinates, correlation between variables and axes)</p>
</td></tr>
<tr><td><code>quali.var</code></td>
<td>
<p>a list of matrices with all the results for the supplementary categorical variables (coordinates of each categories of each variables, and v.test which is a criterion with a Normal distribution)</p>
</td></tr>
<tr><td><code>partial</code></td>
<td>
<p>a list of arrays with the coordinates of the partial points for each partition</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sebastien Le, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Le Dien, S. &amp; Pages, J. (2003) Hierarchical Multiple factor analysis: application to the comparison of sensory profiles, <em>Food Quality and Preferences</em>, <b>18 (6)</b>, 453-464.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.HMFA">print.HMFA</a></code>, <code><a href="#topic+plot.HMFA">plot.HMFA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data(wine)
hierar &lt;- list(c(2,5,3,10,9,2), c(4,2))
res.hmfa &lt;- HMFA(wine, H = hierar, type=c("n",rep("s",5)))
</code></pre>

<hr>
<h2 id='hobbies'>hobbies (data)</h2><span id='topic+hobbies'></span>

<h3>Description</h3>

<p>The data used here concern a questionnaire on hobbies.
We asked to 8403 individuals how answer questions about their hobbies (18 questions). The following four variables were used to label the individuals: sex (male, female), age (15-25, 26-35, 36-45, 46-55, 56-65, 66-75, 76-85, 86-100), marital status (single, married, widowed,divorced, remarried), profession (manual labourer, unskilled worker, technician, foreman, senior management, employee, other). And finally, a quantitative variable indicates the number of hobbies practised out of the 18 possible choices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tea)</code></pre>


<h3>Format</h3>

<p>A data frame with 8403 rows and 23 columns. Rows represent the individuals,
columns represent the different questions. The first 18 questions are active ones,
and the 4 following ones are supplementary categorical variables and the 23th is a supplementary quantitative variable (the number of activities)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hobbies)
## Not run: 
res.mca &lt;- MCA(hobbies,quali.sup=19:22,quanti.sup=23,method="Burt")
plot(res.mca,invisible=c("ind","quali.sup"),hab="quali") ### active var. only
plot(res.mca,invisible=c("var","quali.sup"),cex=.5,label="none") ### individuals only
plot(res.mca,invisible=c("ind","var"),hab="quali") ### supp. qualitative var. only

dimdesc(res.mca)
plotellipses(res.mca,keepvar=1:4)

## End(Not run)

</code></pre>

<hr>
<h2 id='JO'>Number of medals in athletism during olympic games per country</h2><span id='topic+JO'></span>

<h3>Description</h3>

<p>This data frame is a contengency table with the athletism events (in row) and the coutries (in columns). Each cell gives the number of medals obtained during the 5 olympis games from 1992 to 2008 (Barcelona 1992, Atlanta 1996, Sydney 2000, Athens 2004, Beijing 2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(JO)</code></pre>


<h3>Format</h3>

<p>A data frame with the 24 events in athletism and in colum the 58 countries who obtained at least on medal
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(JO)
res.ca &lt;- CA(JO)
res.ca &lt;- CA(JO, axes = 3:4)

## End(Not run)
</code></pre>

<hr>
<h2 id='LinearModel'>Linear Model with AIC or BIC selection, and with the contrasts 
sum (the sum of the coefficients is 0) if any categorical variables</h2><span id='topic+LinearModel'></span>

<h3>Description</h3>

<p>Linear Model with AIC or BIC selection, and with the contrasts sum (the sum of the coefficients is 0) if any categorical variables <br />
Test for all the coefficients<br />
Handle missing values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LinearModel(formula, data, na.action = na.omit, type = c("III","II", 3, 2), 
       selection=c("none","aic","bic"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LinearModel_+3A_formula">formula</code></td>
<td>
<p>the formula for the model 'y~x1+x2+x1:x2'</p>
</td></tr>
<tr><td><code id="LinearModel_+3A_data">data</code></td>
<td>
<p>a data-frame</p>
</td></tr>
<tr><td><code id="LinearModel_+3A_na.action">na.action</code></td>
<td>
<p>(where relevant) information returned by model.frame on the special handling of NAs.</p>
</td></tr>
<tr><td><code id="LinearModel_+3A_type">type</code></td>
<td>
<p>type of test, &quot;III&quot;, &quot;II&quot;, 3 or 2. Roman numerals are equivalent to the 
corresponding Arabic numerals.</p>
</td></tr>
<tr><td><code id="LinearModel_+3A_selection">selection</code></td>
<td>
<p>a string that defines the model selection according to
&quot;BIC&quot; for Bayesian Information Criterion or &quot;AIC&quot; for Akaike Information Criterion; 
&quot;none&quot;, by defaut, means that there is no selection.</p>
</td></tr>
<tr><td><code id="LinearModel_+3A_...">...</code></td>
<td>
<p>other arguments, cf the function <code>lm</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Anova function of the package car is used to calculate the F-tests.
</p>
<p>The t-tests are obtained using the contrasts &quot;contr.sum&quot; which means that 'sum to zero contrasts'.
</p>
<p>A stepwise procedure (using both backword and forward selections) is performed to select a 
model if selection=&quot;AIC&quot; or selection=&quot;BIC&quot;.</p>


<h3>Value</h3>

<p>The outouts
</p>
<table>
<tr><td><code>Ftest</code></td>
<td>
<p>a table with the F-tests</p>
</td></tr>
<tr><td><code>Ttest</code></td>
<td>
<p>a table with the t-tests</p>
</td></tr>
<tr><td><code>lmResult</code></td>
<td>
<p>the summary of the function lm</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>the matched call</p>
</td></tr>
<tr><td><code>lmResultComp</code></td>
<td>
<p>the summary of the lm function for the complete model (given only if a selection is performed)</p>
</td></tr>
<tr><td><code>callComp</code></td>
<td>
<p>the matched call for the complete model (given only if a selection is performed)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+aov">aov</a></code>, <code><a href="stats.html#topic+lm">lm</a></code>, <code><a href="#topic+print.LinearModel">print.LinearModel</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example two-way anova
data(senso)
res &lt;- LinearModel(Score~ Product + Day , data=senso, selection="none")
res
## Perform means comparison
meansComp(res,~Product)

## Example two-way anova with interaction
data(senso)
res2 &lt;- LinearModel(Score~ Product + Day + Product : Day, data=senso, selection="none")
res2
meansComp(res,~Product:Day)

## Example two-way anova with selection
data(senso)
res2 &lt;- LinearModel(Score~ Product + Day + Product : Day, data=senso, selection="BIC")
res2

## Example ancova
data(footsize)
res3 &lt;- LinearModel(footsize ~ size + sex + size : sex, data=footsize)
res3
</code></pre>

<hr>
<h2 id='MCA'>Multiple Correspondence Analysis (MCA)</h2><span id='topic+MCA'></span>

<h3>Description</h3>

<p>Performs Multiple Correspondence Analysis (MCA) with supplementary individuals, supplementary quantitative 
variables and supplementary categorical variables.<br />
Performs also Specific Multiple Correspondence Analysis with supplementary categories and supplementary categorical variables.<br />
Missing values are treated as an additional level, categories which are rare can be ventilated
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MCA(X, ncp = 5, ind.sup = NULL, quanti.sup = NULL, 
    quali.sup = NULL, excl=NULL, graph = TRUE, 
	level.ventil = 0, axes = c(1,2), row.w = NULL, 
	method="Indicator", na.method="NA", tab.disj=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MCA_+3A_x">X</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns (categorical variables)</p>
</td></tr>
<tr><td><code id="MCA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="MCA_+3A_ind.sup">ind.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary individuals</p>
</td></tr>
<tr><td><code id="MCA_+3A_quanti.sup">quanti.sup</code></td>
<td>
<p>a vector indicating the indexes of the quantitative supplementary variables</p>
</td></tr>
<tr><td><code id="MCA_+3A_quali.sup">quali.sup</code></td>
<td>
<p>a vector indicating the indexes of the categorical supplementary variables</p>
</td></tr>
<tr><td><code id="MCA_+3A_excl">excl</code></td>
<td>
<p>vector indicating the indexes of the &quot;junk&quot; categories (default is NULL), it can be a vector of the names of the categories or a vector of the indexes in the disjunctive data table</p>
</td></tr>
<tr><td><code id="MCA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="MCA_+3A_level.ventil">level.ventil</code></td>
<td>
<p>a proportion corresponding to the level under which the category is ventilated; by default, 0 and no ventilation is done</p>
</td></tr>
<tr><td><code id="MCA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="MCA_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, a vector of 1 for uniform row weights); the weights are given only for the active individuals</p>
</td></tr>
<tr><td><code id="MCA_+3A_method">method</code></td>
<td>
<p>a string corresponding to the name of the method used: &quot;Indicator&quot; (by default) is the CA on the Indicator matrix, &quot;Burt&quot; is the CA on the Burt table. For Burt and the Indicator, the graph of the individuals and the graph of the categories are given</p>
</td></tr>
<tr><td><code id="MCA_+3A_na.method">na.method</code></td>
<td>
<p>a string corresponding to the name of the method used if there are missing values; available methods are &quot;NA&quot; or &quot;Average&quot; (by default, &quot;NA&quot;)</p>
</td></tr>
<tr><td><code id="MCA_+3A_tab.disj">tab.disj</code></td>
<td>
<p>optional data.frame corresponding to the disjunctive table used for the analysis; it corresponds to a disjunctive table obtained from imputation method (see package missMDA).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>a list of matrices containing all the results for the active variables (coordinates, square cosine, contributions, v.test, square correlation ratio)</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices containing all the results for the active individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary individuals (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quanti.sup</code></td>
<td>
<p>a matrix containing the coordinates of the supplementary quantitative variables (the correlation between a variable and an axis is equal to the variable coordinate on the axis)</p>
</td></tr>
<tr><td><code>quali.sup</code></td>
<td>
<p>a list of matrices with all the results for the supplementary categorical variables (coordinates of each categories of each variables, square cosine and v.test which is a criterion with a Normal distribution, square correlation ratio)</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>a list with some statistics</p>
</td></tr>
</table>
<p>Returns the graphs of the individuals and categories and the graph with the variables.<br />
The plots may be improved using the argument autolab, modifying the size of the labels or selecting some elements thanks to the <code><a href="#topic+plot.MCA">plot.MCA</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Julie Josse, Jeremy Mazet</p>


<h3>References</h3>

<p>Husson, F., Le, S. and Pages, J. (2010). Exploratory Multivariate Analysis by Example Using R, <em>Chapman and Hall</em>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+plotellipses">plotellipses</a></code>, <code><a href="#topic+summary.MCA">summary.MCA</a></code>,<code><a href="#topic+print.MCA">print.MCA</a></code>, <code><a href="#topic+plot.MCA">plot.MCA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code>,<br />
<a href="https://www.youtube.com/watch?v=reG8Y9ZgcaQ&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=5">Video showing how to perform MCA with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Tea example
 data(tea)
 res.mca &lt;- MCA(tea,quanti.sup=19,quali.sup=20:36)
 summary(res.mca)
 plot(res.mca,invisible=c("var","quali.sup","quanti.sup"),cex=0.7)
 plot(res.mca,invisible=c("ind","quali.sup","quanti.sup"),cex=0.8)
 plot(res.mca,invisible=c("quali.sup","quanti.sup"),cex=0.8)
 dimdesc(res.mca)
 plotellipses(res.mca,keepvar=1:4)
 plotellipses(res.mca,keepvar="Tea")

## Hobbies example
data(hobbies)
res.mca &lt;- MCA(hobbies,quali.sup=19:22,quanti.sup=23)
plot(res.mca,invisible=c("ind","quali.sup"),hab="quali") 
plot(res.mca,invisible=c("var","quali.sup"),cex=.5,label="none") 
plot(res.mca,invisible=c("ind","var"),hab="quali")
dimdesc(res.mca)
plotellipses(res.mca,keepvar=1:4)

## Specific MCA: some categories are supplementary
data (poison)
res &lt;- MCA (poison[,3:8],excl=c(1,3))

## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(tea)

## Example with missing values : use the missMDA package
require(missMDA)
data(vnf)
completed &lt;- imputeMCA(vnf,ncp=2)
res.mca &lt;- MCA(vnf,tab.disj=completed$tab.disj)

## End(Not run)
</code></pre>

<hr>
<h2 id='meansComp'>Perform pairwise means comparisons</h2><span id='topic+meansComp'></span>

<h3>Description</h3>

<p>Perform means comparisons and give groups of means that are not significantly different.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meansComp(object, spec, graph=TRUE, ...) 
    </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meansComp_+3A_object">object</code></td>
<td>
<p>A fitted model object that is supported, such as the result of a call to LinearModel, lm or aov.</p>
</td></tr>
<tr><td><code id="meansComp_+3A_spec">spec</code></td>
<td>
<p>A character vector specifying the names of the predictors over which means comparisons are desired. specs may also be a formula or a list (optionally named) of valid specs. Use of formulas is described in the Overview section below.</p>
</td></tr>
<tr><td><code id="meansComp_+3A_graph">graph</code></td>
<td>
<p>Boolean; plot the graph to compare the means.</p>
</td></tr>
<tr><td><code id="meansComp_+3A_...">...</code></td>
<td>
<p>other arguments, cf the function <code>emmeans</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+plot.meansComp">plot.meansComp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(senso)
  res &lt;- LinearModel(Score~ Product + Day , data=senso, selection="none")
  meansComp(res,~Product)
  
## Not run: 
  ## and with the sidak correction
  meansComp(res,~Product,adjust="sidak")

## End(Not run)</code></pre>

<hr>
<h2 id='MFA'>Multiple Factor Analysis (MFA)</h2><span id='topic+MFA'></span>

<h3>Description</h3>

<p>Performs Multiple Factor Analysis in the sense of Escofier-Pages with supplementary individuals and supplementary groups of variables.
Groups of variables can be quantitative, categorical or contingency tables.<br />
Specific Multiple Fac tor Analysis can be performed using the argument excl.<br />
Missing values in numeric variables are replaced by the column mean.<br />
Missing values in categorical variables are treated as an additional level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MFA (base, group, type = rep("s",length(group)), excl = NULL, 
    ind.sup = NULL, ncp = 5, name.group = NULL,  
    num.group.sup = NULL, graph = TRUE, weight.col.mfa = NULL, 
    row.w = NULL, axes = c(1,2), tab.comp=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MFA_+3A_base">base</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns (variables)</p>
</td></tr>
<tr><td><code id="MFA_+3A_group">group</code></td>
<td>
<p>a vector with the number of variables in each group</p>
</td></tr>
<tr><td><code id="MFA_+3A_type">type</code></td>
<td>
<p>the type of variables in each group; four possibilities: &quot;c&quot; or &quot;s&quot; for quantitative variables
(the difference is that for &quot;s&quot; variables are scaled to unit variance), &quot;n&quot; for categorical variables
&quot;m&quot; for group of mixed variables  and &quot;f&quot; for frequencies (from a contingency tables); by default,
all variables are quantitative and scaled to unit variance</p>
</td></tr>
<tr><td><code id="MFA_+3A_excl">excl</code></td>
<td>
<p>an argument that may possible to exclude categories of active variables of categorical variable groups. 
NULL by default, it is a list with indexes of categories that are excluded per group</p>
</td></tr>
<tr><td><code id="MFA_+3A_ind.sup">ind.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary individuals</p>
</td></tr>
<tr><td><code id="MFA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="MFA_+3A_name.group">name.group</code></td>
<td>
<p>a vector containing the name of the groups (by default, NULL and the group are named group.1, group.2 and so on)</p>
</td></tr>
<tr><td><code id="MFA_+3A_num.group.sup">num.group.sup</code></td>
<td>
<p>the indexes of the illustrative groups (by default, NULL and no group are illustrative)</p>
</td></tr>
<tr><td><code id="MFA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="MFA_+3A_weight.col.mfa">weight.col.mfa</code></td>
<td>
<p>vector of weights, useful for HMFA method (by default, NULL and an MFA is performed)</p>
</td></tr>
<tr><td><code id="MFA_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, a vector of 1 for uniform row weights); the weights are given only for the active individuals</p>
</td></tr>
<tr><td><code id="MFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="MFA_+3A_tab.comp">tab.comp</code></td>
<td>
<p>object obtained from the imputeMFA function of the missMDA package that allows to handle missing values</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>summary.quali</code></td>
<td>
<p>a summary of the results for the categorical variables</p>
</td></tr>
<tr><td><code>summary.quanti</code></td>
<td>
<p>a summary of the results for the quantitative variables</p>
</td></tr>
<tr><td><code>separate.analyses</code></td>
<td>
<p>the results for the separate analyses</p>
</td></tr>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>a list of matrices containing all the results for the groups (Lg and RV coefficients, coordinates, square cosine, contributions, distance to the origin, the correlations between each group and each factor)</p>
</td></tr>
<tr><td><code>rapport.inertie</code></td>
<td>
<p>inertia ratio</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices containing all the results for the active individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary individuals (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quanti.var</code></td>
<td>
<p>a list of matrices containing all the results for the quantitative variables (coordinates, correlation between variables and axes, contribution, cos2)</p>
</td></tr>
<tr><td><code>quali.var</code></td>
<td>
<p>a list of matrices containing all the results for categorical variables (coordinates of each categories of each variables, contribution and v.test which is a criterion with a Normal distribution)</p>
</td></tr>
<tr><td><code>freq</code></td>
<td>
<p>a list of matrices containing all the results for the frequencies (coordinates, contribution, cos2)</p>
</td></tr>
<tr><td><code>quanti.var.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary quantitative variables (coordinates, correlation between variables and axes, cos2)</p>
</td></tr>
<tr><td><code>quali.var.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary categorical variables (coordinates of each categories of each variables, cos2 and v.test which is a criterion with a Normal distribution)</p>
</td></tr>
<tr><td><code>freq.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary frequencies (coordinates, cos2)</p>
</td></tr>
<tr><td><code>partial.axes</code></td>
<td>
<p>a list of matrices containing all the results for the partial axes (coordinates, correlation between variables and axes, correlation between partial axes)</p>
</td></tr>
<tr><td><code>global.pca</code></td>
<td>
<p>the result of the analysis when it is considered as a unique weighted PCA</p>
</td></tr>
</table>
<p>Returns the individuals factor map, the variables factor map and the groups factor map.<br />
The plots may be improved using the argument autolab, modifying the size of the labels or selecting some elements thanks to the <code><a href="#topic+plot.MFA">plot.MFA</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, J. Mazet</p>


<h3>References</h3>

<p>Escofier, B. and Pages, J. (1994) Multiple Factor Analysis (AFMULT package). <em>Computational Statistics and Data Analysis</em>, 18, 121-140.<br />
Becue-Bertaut, M. and Pages, J. (2008) Multiple factor analysis and clustering of a mixture of quantitative, categorical and frequency data. <em>Computational Statistice and Data Analysis</em>, 52, 3255-3268.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.MFA">print.MFA</a></code>, <code><a href="#topic+summary.MFA">summary.MFA</a></code>, <code><a href="#topic+plot.MFA">plot.MFA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code>,<br />
<a href="https://www.youtube.com/watch?v=g5_hM93e8HM&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=6">Video showing how to perform MFA with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res &lt;- MFA(wine, group=c(2,5,3,10,9,2), type=c("n",rep("s",5)),
    ncp=5, name.group=c("orig","olf","vis","olfag","gust","ens"),
    num.group.sup=c(1,6))
summary(res)
barplot(res$eig[,1],main="Eigenvalues",names.arg=1:nrow(res$eig))

#### Confidence ellipses around categories per variable
plotellipses(res)
plotellipses(res,keepvar="Label") ## for 1 variable

#### Interactive graph
liste = plotMFApartial(res)
plot(res,choix="ind",habillage = "Terroir")

###Example with groups of categorical variables
data (poison)
MFA(poison, group=c(2,2,5,6), type=c("s","n","n","n"),
    name.group=c("desc","desc2","symptom","eat"),
    num.group.sup=1:2)

###Example with groups of frequency tables
data(mortality)
res&lt;-MFA(mortality,group=c(9,9),type=c("f","f"),
    name.group=c("1979","2006"))

## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(wine)

### with missing values
require(missMDA)
data(orange)
res.impute &lt;- imputeMFA(orange, group=c(5,3), type=rep("s",2),ncp=2) 
res.mfa &lt;- MFA(res.impute$completeObs,group=c(5,3),type=rep("s",2)) 

## End(Not run)
</code></pre>

<hr>
<h2 id='milk'>milk</h2><span id='topic+milk'></span>

<h3>Description</h3>

<p>Dataset to illustrate the selection of variables in regression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(milk)</code></pre>


<h3>Format</h3>

<p>Dataset with 85 rows and 6 columns :
85 milks described by the 5 variables: density, fat content, protein, casein, dry, yield</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(milk)
res = RegBest(y=milk[,6],x=milk[,-6])
res$best
</code></pre>

<hr>
<h2 id='mortality'>The cause of mortality in France in 1979 and 2006</h2><span id='topic+mortality'></span>

<h3>Description</h3>

<p>The cause of mortality in France in 1979 and 2006.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mortality)</code></pre>


<h3>Format</h3>

<p>A data frame with 62 rows (the different causes of death) and 18 columns. Each column corresponds to an age interval (15-24, 25-34, 35-44, 45-54, 55-64, 65-74, 75-84, 85-94, 95 and more) in a year. The 9 first columns correspond to data in 1979 and the 9 last columns to data in 2006.
In each cell, the counts of deaths for a cause of death in an age interval (in a year) is given.
</p>


<h3>Source</h3>

<p>Centre d'epidemiologie sur les causes medicales
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mortality)

## Not run: 
res&lt;-MFA(mortality,group=c(9,9),type=c("f","f"),
    name.group=c("1979","2006"))

plot(res,choix="freq",invisible="ind",graph.type = "classic")
lines(res$freq$coord[1:9,1],res$freq$coord[1:9,2],col="red")
lines(res$freq$coord[10:18,1],res$freq$coord[10:18,2],col="green")    
    
## End(Not run)
</code></pre>

<hr>
<h2 id='PCA'>Principal Component Analysis (PCA)</h2><span id='topic+PCA'></span>

<h3>Description</h3>

<p>Performs Principal Component Analysis (PCA) with supplementary individuals, supplementary quantitative 
variables and supplementary categorical variables.<br />
Missing values are replaced by the column mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA(X, scale.unit = TRUE, ncp = 5, ind.sup = NULL, 
    quanti.sup = NULL, quali.sup = NULL, row.w = NULL, 
    col.w = NULL, graph = TRUE, axes = c(1,2))</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA_+3A_x">X</code></td>
<td>
<p>a data frame with <em>n</em> rows (individuals) and <em>p</em> columns (numeric variables)</p>
</td></tr>
<tr><td><code id="PCA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions kept in the results (by default 5)</p>
</td></tr>
<tr><td><code id="PCA_+3A_scale.unit">scale.unit</code></td>
<td>
<p>a boolean, if TRUE (value set by default) then data are scaled to unit variance</p>
</td></tr>
<tr><td><code id="PCA_+3A_ind.sup">ind.sup</code></td>
<td>
<p>a vector indicating the indexes of the supplementary individuals</p>
</td></tr>
<tr><td><code id="PCA_+3A_quanti.sup">quanti.sup</code></td>
<td>
<p>a vector indicating the indexes of the quantitative supplementary variables</p>
</td></tr>
<tr><td><code id="PCA_+3A_quali.sup">quali.sup</code></td>
<td>
<p>a vector indicating the indexes of the categorical supplementary variables</p>
</td></tr>
<tr><td><code id="PCA_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, a vector of 1 for uniform row weights); the weights are given only for the active individuals</p>
</td></tr>
<tr><td><code id="PCA_+3A_col.w">col.w</code></td>
<td>
<p>an optional column weights (by default, uniform column weights); the weights are given only for the active variables</p>
</td></tr>
<tr><td><code id="PCA_+3A_graph">graph</code></td>
<td>
<p>boolean, if TRUE a graph is displayed</p>
</td></tr>
<tr><td><code id="PCA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>eig</code></td>
<td>
<p>a matrix containing all the eigenvalues, the percentage of variance and the cumulative percentage of variance</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>a list of matrices containing all the results for the active variables (coordinates, correlation between variables and axes, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind</code></td>
<td>
<p>a list of matrices containing all the results for the active individuals (coordinates, square cosine, contributions)</p>
</td></tr>
<tr><td><code>ind.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary individuals (coordinates, square cosine)</p>
</td></tr>
<tr><td><code>quanti.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary quantitative variables (coordinates, correlation between variables and axes)</p>
</td></tr>
<tr><td><code>quali.sup</code></td>
<td>
<p>a list of matrices containing all the results for the supplementary categorical variables (coordinates of each categories of each variables, v.test which is a criterion with a Normal distribution, and eta2 which is the square correlation corefficient between a qualitative variable and a dimension)</p>
</td></tr>
</table>
<p>Returns the individuals factor map and the variables factor map.<br />
The plots may be improved using the argument autolab, modifying the size of the labels or selecting some elements thanks to the <code><a href="#topic+plot.PCA">plot.PCA</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Jeremy Mazet</p>


<h3>References</h3>

<p>Husson, F., Le, S. and Pages, J. (2010). Exploratory Multivariate Analysis by Example Using R, <em>Chapman and Hall</em>.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+print.PCA">print.PCA</a></code>, <code><a href="#topic+summary.PCA">summary.PCA</a></code>, <code><a href="#topic+plot.PCA">plot.PCA</a></code>, <code><a href="#topic+dimdesc">dimdesc</a></code>,<br />
<a href="https://www.youtube.com/watch?v=CTSbxU6KLbM&amp;list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&amp;index=3">Video showing how to perform PCA with FactoMineR</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup=13)
## plot of the eigenvalues
## barplot(res.pca$eig[,1],main="Eigenvalues",names.arg=1:nrow(res.pca$eig))
summary(res.pca)
plot(res.pca,choix="ind",habillage=13)
## Not run: 
## To describe the dimensions
dimdesc(res.pca, axes = 1:2)

## To draw ellipses around the categories of the 13th variable (which is categorical)
plotellipses(res.pca,13)

## Graphical interface
require(Factoshiny)
res &lt;- Factoshiny(decathlon)

## Example with missing data
## use package missMDA
require(missMDA)
data(orange)
nb &lt;- estim_ncpPCA(orange,ncp.min=0,ncp.max=5,method.cv="Kfold",nbsim=50)
imputed &lt;- imputePCA(orange,ncp=nb$ncp)
res.pca &lt;- PCA(imputed$completeObs)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.CA'>Draw the Correspondence Analysis (CA) graphs</h2><span id='topic+plot.CA'></span>

<h3>Description</h3>

<p>Draw the Correspondence Analysis (CA) graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CA'
plot(x, axes = c(1, 2),
    xlim = NULL, ylim = NULL, 
	invisible = c("none","row","col","row.sup","col.sup","quali.sup"), 
	choix = c("CA","quanti.sup"), col.row="blue", col.col="red", 
	col.row.sup="darkblue", col.col.sup="darkred", 
    col.quali.sup="magenta", col.quanti.sup="blue",
	label = c("all","none","row","row.sup","col","col.sup","quali.sup"), 
    title = NULL, palette = NULL, autoLab = c("auto","yes","no"), 
	new.plot=FALSE, selectRow = NULL, selectCol = NULL,
	unselect = 0.7, shadowtext = FALSE, habillage = "none",
	legend = list(bty = "y", x = "topleft"), 
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CA_+3A_x">x</code></td>
<td>
<p>an object of class CA</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_invisible">invisible</code></td>
<td>
<p>string indicating if some points should be unlabelled (&quot;row&quot;, &quot;col&quot;, &quot;row.sup&quot;, &quot;col.sup&quot;,&quot;quali.sup&quot;)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;CA&quot; for the CA map, &quot;quanti.sup&quot; for the supplementary quantitative variables)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.row">col.row</code></td>
<td>
<p>a color for the rows points</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.col">col.col</code></td>
<td>
<p>a color for columns points</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.row.sup">col.row.sup</code></td>
<td>
<p>a color for the supplementary rows points</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.col.sup">col.col.sup</code></td>
<td>
<p>a color for supplementary columns points</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.quali.sup">col.quali.sup</code></td>
<td>
<p>a color for the supplementary categorical variables</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_col.quanti.sup">col.quanti.sup</code></td>
<td>
<p>a color for the supplementary quantitative variables</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_label">label</code></td>
<td>
<p>a list of character for the elements which are labelled (by default, all the elements are labelled (&quot;row&quot;, &quot;row.sup&quot;, &quot;col&quot;, &quot;col.sup&quot;,&quot;quali.sup&quot;)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_selectrow">selectRow</code></td>
<td>
<p>a selection of the rows that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_selectcol">selectCol</code></td>
<td>
<p>a selection of the columns that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if <code>unselect=1</code> the transparceny is total and the elements are not drawn, if <code>unselect=0</code> the elements are drawn as usual but without any label) or may be a color (for example <code>unselect="grey60"</code>)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_habillage">habillage</code></td>
<td>
<p>color the individuals among a categorical variable (give the number of the categorical supplementary variable or its name)</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_legend">legend</code></td>
<td>
<p>a list of arguments that defines the legend if needed (when individuals are drawn according to a variable); see the arguments of the function <code>legend</code></p>
</td></tr>
<tr><td><code id="plot.CA_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="plot.CA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>autoLab = "yes"</code> is time-consuming if there are many labels that overlap. In this case, you can modify the size of the characters in order to have less overlapping, using for example cex=0.7.<br />
The <code>selectRow</code> and <code>selectCol</code> arguments can be used in order to select a part of the elements that are drawn.
For example, you can use:<br />
<code>selectRow = 1:5</code> and then the rows 1 to 5 are drawn.<br /> 
<code>select = c("name1","name5")</code> and then the rows that have the names name1 and name5 are drawn.<br /> 
<code>select = "coord 10"</code> and then the 10 rows (10 active and 10 supplementaries) that have the highest (squared) coordinates on the 2 chosen dimensions are drawn.<br /> 
<code>select = "contrib 10"</code> and then the 10 rows (10 active) that have the highest contribution on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "cos2 5"</code> and then the 5 rows (5 actives and 5 supplementaries) that have the highest cos2 on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "dist 8"</code> and then the 8 rows (8 actives and 8 supplementaries) that have the highest distance to the center of gravity are drawn.<br /> 
</p>


<h3>Value</h3>

<p>Returns the factor map with the joint plot of CA.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+CA">CA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(children)
res.ca &lt;- CA (children, col.sup = 6:8, row.sup = 15:18)

## select rows and columns that have a cos2 greater than 0.8
plot(res.ca, selectCol="cos2 0.8", selectRow="cos2 0.8")

## Not run: 
## You can modify the ggplot graphs as ususal with ggplot2
require(ggplot2)
gr &lt;- plot(res.ca)
gr + theme(panel.grid.major = element_blank(),
   plot.title=element_text(size=14, color="blue"),
   axis.title = element_text(size=12, color="red"))

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.CaGalt'>Draw the Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) graphs</h2><span id='topic+plot.CaGalt'></span>

<h3>Description</h3>

<p>Plot the graphs for a Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CaGalt'
plot(x, axes = c(1, 2), choix = c("ind", "freq", "quali.var", "quanti.var"), 
  conf.ellip = FALSE, contr.ellipse = 3, xlim = NULL, ylim = NULL, col.ind = "black", 
  col.freq = "darkred", col.quali = "blue", col.quanti = "blue", label = TRUE, 
  lim.cos2.var = 0, title = NULL, palette = NULL, 
  autoLab = c("auto", "yes", "no"), new.plot = FALSE, select = NULL, 
  unselect = 0.7, shadowtext = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CaGalt_+3A_x">x</code></td>
<td>
<p>an object of class CaGalt</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;ind&quot; for the individuals, &quot;freq&quot; for the frequencies, &quot;quali.var&quot; for the categorical variables, &quot;quanti.var&quot; for the quantitative variables)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_conf.ellip">conf.ellip</code></td>
<td>
<p>boolean (FALSE by default), if TRUE, draw ellipses around the frequencies and the variables</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_contr.ellipse">contr.ellipse</code></td>
<td>
<p>the confidence ellipses were drawn for the frequencies with a contribution higher than X times of mean contribution on the 2 dimensions of your plot (by default 3)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_col.ind">col.ind</code></td>
<td>
<p>a color for the individuals (by default &quot;black&quot;)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_col.freq">col.freq</code></td>
<td>
<p>a color for the frequencies (by default &quot;darkred&quot;)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_col.quali">col.quali</code></td>
<td>
<p>a color for the categories of categorical variables (by default &quot;blue&quot;)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_col.quanti">col.quanti</code></td>
<td>
<p>a color for the quantitative variables (by default &quot;blue&quot;)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_label">label</code></td>
<td>
<p>the labels are drawn (by default TRUE)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under the variables are not drawn</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_autolab">autoLab</code></td>
<td>
<p>if autoLab=&quot;auto&quot;, autoLab is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_select">select</code></td>
<td>
<p>a selection of the elements that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if unselect=1 the transparency is total and the elements are not drawn, if unselect=0 the elements are drawn as usual but without any label) or may be a color (for example unselect=&quot;grey60&quot;)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program)</p>
</td></tr>
<tr><td><code id="plot.CaGalt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument autoLab = &quot;yes&quot; is time-consuming if there are many labels that overlap. In this case, you can modify the size of the characters in order to have less overlapping, using for example cex=0.7.
The select argument can be used in order to select a part of the elements (individuals if you draw the graph of individuals, or variables if you draw the graph of variabless) that are drawn. For example, you can use:
select = 1:5 and then the elements 1:5 are drawn.
select = c(&quot;name1&quot;,&quot;name5&quot;) and then the elements that have the names name1 and name5 are drawn.
select = &quot;coord 10&quot; and then the 10 elements that have the highest (squared) coordinates on the 2 chosen dimensions are drawn.
select = &quot;contrib 10&quot; and then the 10 elements that have the highest contribution on the 2 dimensions of your plot are drawn (available only when frequencies are drawn).
select = &quot;cos2 5&quot; and then the 5 elements that have the highest cos2 on the 2 dimensions of your plot are drawn.
</p>


<h3>Value</h3>

<p>Returns the individuals, the frequencies and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Belchin Kostov <a href="mailto:badriyan@clinic.ub.es">badriyan@clinic.ub.es</a>, Monica Becue-Bertaut, Francois Husson</p>


<h3>See Also</h3>

 <p><code><a href="#topic+CaGalt">CaGalt</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(health)
res.cagalt&lt;-CaGalt(Y=health[,1:115],X=health[,116:118],type="n")
plot(res.cagalt,choix="quali.var",conf.ellip=TRUE,axes=c(1,4))

## Selection of some individuals,categories and frequencies
plot(res.cagalt,choix="freq",col.freq="darkgreen",cex=1.5,select="contrib 10") 
plot(res.cagalt,choix="ind",select="coord 10") 
plot(res.cagalt,choix="quali.var",select="cos2 0.5") 

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.catdes'>Plots for description of clusters (catdes)</h2><span id='topic+plot.catdes'></span>

<h3>Description</h3>

<p>Plots a graph from a catdes output.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'catdes'
plot(x, show="all",output=c("figure","dt") , level=0.01, sort=NULL,
   col.upper="indianred2", col.lower="royalblue1", numchar = 10,
   barplot = FALSE,cex.names=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.catdes_+3A_x">x</code></td>
<td>
<p>A catdes object, see <code><a href="#topic+catdes">catdes</a></code> for details.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_show">show</code></td>
<td>
<p>a string. If &quot;quali&quot;, only the categorical variables are
used. If &quot;quanti&quot;, only the the quantitative variables are used. If
&quot;all&quot;, both quali and quanti are used. If &quot;quanti.var&quot; is used the characterization of the quantitative variables is given; if &quot;test.chi2&quot; is used the characterization of the qualitative variables is given.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_output">output</code></td>
<td>
<p>string: &quot;dt&quot; for a datatable or &quot;figure&quot; for a figure</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_level">level</code></td>
<td>
<p>a positive float. Indicates a critical value the p-value.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_sort">sort</code></td>
<td>
<p>NULL (default) or an integer between 1 and the number of clusters or a character (the name of a group). If it is an integer or the name of a group, the features are sorted according to their significances in the construction of the given cluster.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_col.upper">col.upper</code></td>
<td>
<p>The color used for under-represented features.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_col.lower">col.lower</code></td>
<td>
<p>The color used for over-represented features.</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_numchar">numchar</code></td>
<td>
<p>number of characters for the labels</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_barplot">barplot</code></td>
<td>
<p>a boolean; if true a barplot per category is drawn, else a table</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_cex.names">cex.names</code></td>
<td>
<p>the magnification to be used for the names</p>
</td></tr>
<tr><td><code id="plot.catdes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if barplot is true, a barplot is drawn per category with variables that significantly describe the category.<br />
If barplot is false; it returns a grid. The rows stand for the clusters and the columns for the significant variables. 
A cell colored in col.lower (resp. col.upper) i.e. by default in blue (resp. red) for a quantitative variable means that the average value of the variable in the given cluster is significantly lower (resp. higher) than in the overall data. 
A cell colored in col.lower (resp. col.upper) for a categorical variable means that the given value of the variable is significantly under-represented (resp. over-represented) in the given cluster than in the overall data. 
The degree of transparency of the color also indicates the significance of the difference between the behavior of the variable in the given cluster and in the overall data. Indeed, the more transparent the cell is, the less significant the difference is. 
</p>


<h3>Author(s)</h3>

<p>Guillaume Le Ray, Camille Chanial, Elise Dumas, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+catdes">catdes</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res.c=catdes(wine, num.var=2)
plot(res.c)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.DMFA'>Draw the Dual Multiple Factor Analysis (DMFA) graphs</h2><span id='topic+plot.DMFA'></span>

<h3>Description</h3>

<p>Plot the graphs for a Principal Component Analysis (DMFA) with supplementary individuals, supplementary quantitative 
variables and supplementary categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DMFA'
plot(x, axes = c(1, 2), choix = "ind", label="all",
    lim.cos2.var = 0., xlim=NULL, ylim=NULL, title = NULL,
    palette = NULL, new.plot = FALSE, 
	autoLab = c("auto","yes","no"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.DMFA_+3A_x">x</code></td>
<td>
<p>an object of class DMFA</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;ind&quot; for the individuals, &quot;var&quot; for the variables)</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_label">label</code></td>
<td>
<p>a list of character for the elements which are labelled (by default, all the elements are labelled (&quot;ind&quot;, ind.sup&quot;, &quot;quali&quot;, &quot;var&quot;, &quot;quanti.sup&quot;))</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under the variables are not drawn</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.DMFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map, the partial variables representation and the groups factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+DMFA">DMFA</a></code></p>

<hr>
<h2 id='plot.FAMD'>Draw the Multiple Factor Analysis for Mixt Data graphs</h2><span id='topic+plot.FAMD'></span>

<h3>Description</h3>

<p>It provides the graphical outputs associated with the principal component method for mixed data: FAMD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FAMD'
plot(x, choix = c("ind","var","quanti","quali"), axes = c(1, 2), 
    lab.var = TRUE, lab.ind = TRUE, habillage = "none", col.lab = FALSE,
    col.hab = NULL, invisible = NULL, lim.cos2.var = 0., xlim = NULL,
    ylim = NULL, title = NULL, palette=NULL, autoLab = c("auto","yes","no"), 
	new.plot = FALSE, select = NULL, unselect = 0.7, shadowtext = FALSE, 
	legend = list(bty = "y", x = "topleft"),
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.FAMD_+3A_x">x</code></td>
<td>
<p>an object of class FAMD</p>
</td></tr> 
<tr><td><code id="plot.FAMD_+3A_choix">choix</code></td>
<td>
<p>a string corresponding to the graph that you want to do (&quot;ind&quot; for the individual or categorical
variables graph, &quot;var&quot; for all the variables (quantitative and categorical), &quot;quanti&quot; for the correlation circle)</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_lab.var">lab.var</code></td>
<td>
<p>boolean indicating if the labelled of the variables should be drawn on the map</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_lab.ind">lab.ind</code></td>
<td>
<p>boolean indicating if the labelled of the individuals should be drawn on the map</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;ind&quot;, one color is used for each individual else if it is the name 
or the position of a categorical variable, it colors 
according to the different categories of this variable</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_col.lab">col.lab</code></td>
<td>
<p>boolean indicating if the labelled should be colored</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_col.hab">col.hab</code></td>
<td>
<p>vector indicating the colors to use to labelled the rows or columns elements chosen in habillage</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_invisible">invisible</code></td>
<td>
<p>list of string; for choix =&quot;ind&quot;, the individuals can be omit (invisible = &quot;ind&quot;),
or supplementary individuals (invisible=&quot;ind.sup&quot;) or the centerg of gravity of the categorical variables 
(invisible= &quot;quali&quot;); if invisible = c(&quot;ind&quot;,&quot;ind.sup&quot;), just the centers of gravity are drawn</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under the variables are not drawn</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_select">select</code></td>
<td>
<p>a selection of the elements that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if <code>unselect=1</code> the transparceny is total and the elements are not drawn, if <code>unselect=0</code> the elements are drawn as usual but without any label) or may be a color (for example <code>unselect="grey60"</code>)</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program)</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_legend">legend</code></td>
<td>
<p>a list of arguments that defines the legend if needed (when individuals are drawn according to a variable); see the arguments of the function <code>legend</code></p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="plot.FAMD_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+FAMD">FAMD</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(geomorphology)
res &lt;- FAMD(geomorphology)
plot(res,choix="ind",habillage=4)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.GPA'>Draw the General Procrustes Analysis (GPA) map</h2><span id='topic+plot.GPA'></span>

<h3>Description</h3>

<p>Draw the General Procrustes Analysis (GPA) map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GPA'
plot(x, axes = c(1, 2), 
    lab.ind.moy = TRUE, habillage = "ind",
    partial = "all", chrono = FALSE, xlim = NULL, ylim = NULL, 
    cex = 1, title = NULL, palette = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.GPA_+3A_x">x</code></td>
<td>
<p>an object of class GPA</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_lab.ind.moy">lab.ind.moy</code></td>
<td>
<p>boolean, if TRUE, the label of the mean points are drawn</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;ind&quot;, one color is used for each individual;
if &quot;group&quot; the individuals are colored according to the group</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_partial">partial</code></td>
<td>
<p>list of the individuals or of the center of gravity for which the partial points should be drawn (by
default, partial = &quot;none&quot; and no partial points are drawn)</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_chrono">chrono</code></td>
<td>
<p>boolean, if TRUE, the partial points of a same point are linked (useful when groups correspond to different moment)</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.GPA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the General Procrustes Analysis map.
</p>


<h3>Author(s)</h3>

<p>Elisabeth Morand, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+GPA">GPA</a></code></p>

<hr>
<h2 id='plot.HCPC'>Plots for Hierarchical Classification on Principle Components (HCPC) results</h2><span id='topic+plot.HCPC'></span>

<h3>Description</h3>

<p>Plots graphs from a HCPC result: tree, barplot of inertia gains and
first factor map with or without the tree, in 2 or 3 dimensions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HCPC'
plot(x, axes=c(1,2), choice="3D.map", rect=TRUE, 
  draw.tree=TRUE, ind.names=TRUE, t.level="all", title=NULL,
  new.plot=FALSE, max.plot=15, tree.barplot=TRUE,
  centers.plot=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.HCPC_+3A_x">x</code></td>
<td>
<p>A HCPC object, see <code><a href="#topic+HCPC">HCPC</a></code> for details.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_axes">axes</code></td>
<td>
<p>a two integers vector.Defines the axes of the factor map
to plot.</p>
</td></tr>  
<tr><td><code id="plot.HCPC_+3A_choice">choice</code></td>
<td>
<p>A string. &quot;tree&quot; plots the tree. &quot;bar&quot; plots bars of
inertia gains. &quot;map&quot; plots a factor map, individuals colored
by cluster. &quot;3D.map&quot; plots the same factor map, individuals colored by cluster,
the tree above.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_rect">rect</code></td>
<td>
<p>a boolean. If TRUE, rectangles are drawn around clusters
if choice =&quot;tree&quot;.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_tree.barplot">tree.barplot</code></td>
<td>
<p>a boolean. If TRUE, the barplot of intra inertia
losses is added on the tree graph.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_draw.tree">draw.tree</code></td>
<td>
<p>A boolean. If TRUE, the tree is projected on the
factor map if choice =&quot;map&quot;.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_ind.names">ind.names</code></td>
<td>
<p>A boolean. If TRUE, the individuals names are added
on the factor map when choice=&quot;3D.map&quot; or choice=&quot;map&quot;</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_t.level">t.level</code></td>
<td>
<p>Either a positive integer or a string. A positive
integer indicates the starting level to plot the tree on the map when
draw.tree=TRUE. If &quot;all&quot;, the whole tree is ploted. If &quot;centers&quot;, it
draws the tree starting t the centers of the clusters.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_title">title</code></td>
<td>
<p>a string. Title of the graph. NULL by default and a title is automatically defined</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_centers.plot">centers.plot</code></td>
<td>
<p>a boolean. If TRUE, the centers of clusters are
drawn on the 3D factor maps.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_new.plot">new.plot</code></td>
<td>
<p>a boolean. If TRUE, the plot is done in a new window.</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_max.plot">max.plot</code></td>
<td>
<p>The max for the bar plot</p>
</td></tr>
<tr><td><code id="plot.HCPC_+3A_...">...</code></td>
<td>
<p>Other arguments from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the chosen plot.
</p>


<h3>Author(s)</h3>

<p>Guillaume Le Ray, Quentin Molto, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+HCPC">HCPC</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
# Clustering, auto nb of clusters:
res.hcpc=HCPC(iris[1:4], nb.clust=3)
# 3D graph from a different point of view:
plot(res.hcpc, choice="3D.map", angle=60)
</code></pre>

<hr>
<h2 id='plot.HMFA'>Draw the Hierarchical Multiple Factor Analysis (HMFA) graphs</h2><span id='topic+plot.HMFA'></span>

<h3>Description</h3>

<p>Draw the Hierarchical Multiple Factor Analysis (HMFA) graphs</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HMFA'
plot(x, axes = c(1,2),num=6, choix = "ind", 
    lab.grpe = TRUE, lab.var = TRUE, lab.ind.moy = TRUE, 
    invisible = NULL, lim.cos2.var = 0., 
    xlim = NULL, ylim = NULL, cex = 1, title = NULL, new.plot = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.HMFA_+3A_x">x</code></td>
<td>
<p>an object of class HMFA</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_num">num</code></td>
<td>
<p>number of grpahs in a same windows</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_choix">choix</code></td>
<td>
<p>a string corresponding to the graph that you want to do (&quot;ind&quot; for the individual or categorical
variables graph, &quot;var&quot; for the quantitative variables graph, &quot;axes&quot; for the graph of the partial axes,
&quot;group&quot; for the groups representation)</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_lab.grpe">lab.grpe</code></td>
<td>
<p>boolean, if TRUE, the label of the groups are drawn</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_lab.var">lab.var</code></td>
<td>
<p>boolean, if TRUE, the label of the variables are drawn</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_lab.ind.moy">lab.ind.moy</code></td>
<td>
<p>boolean, if TRUE, the label of the mean points are drawn</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_invisible">invisible</code></td>
<td>
<p>list of string; for choix =&quot;ind&quot;, the individuals can be omit (invisible = &quot;ind&quot;),
or the centers of gravity of the categorical variables 
(invisible= &quot;quali&quot;)</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under with the points are not drawn</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.HMFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Jeremy Mazet, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+HMFA">HMFA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
hierar &lt;- list(c(2,5,3,10,9,2), c(4,2))
res.hmfa &lt;- HMFA(wine, H = hierar, type=c("n",rep("s",5)), graph = FALSE)
plot(res.hmfa, invisible="quali")
plot(res.hmfa, invisible="ind")
</code></pre>

<hr>
<h2 id='plot.MCA'>Draw the Multiple Correspondence Analysis (MCA) graphs</h2><span id='topic+plot.MCA'></span>

<h3>Description</h3>

<p>Draw the Multiple Correspondence Analysis (MCA) graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MCA'
plot(x, axes = c(1, 2), choix=c("ind","var","quanti.sup"),
    xlim = NULL, ylim = NULL, 
    invisible = c("none","ind","var","ind.sup","quali.sup","quanti.sup"),
    col.ind = "black", col.var = "red", col.quali.sup = "darkgreen",
    col.ind.sup = "blue", col.quanti.sup = "blue",
    label = c("all","none","ind","var","ind.sup","quali.sup","quanti.sup"),
    title = NULL, habillage = "none", 
    palette = NULL, autoLab = c("auto","yes","no"), new.plot = FALSE, 
    select = NULL, selectMod = NULL, unselect = 0.7, 
	shadowtext = FALSE, legend = list(bty = "y", x = "topleft"), 
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MCA_+3A_x">x</code></td>
<td>
<p>an object of class MCA</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;ind&quot; for the individuals and the categories, &quot;var&quot; for the variables, &quot;quanti.sup&quot; for the supplementary quantitative variables)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_invisible">invisible</code></td>
<td>
<p>string indicating if some points should not be drawn (&quot;ind&quot;, &quot;var&quot;, &quot;ind.sup&quot;, &quot;quali.sup&quot;, &quot;quanti.sup&quot;)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_col.ind">col.ind</code></td>
<td>
<p>a color for the individuals, if color =&quot;none&quot; the label is not written</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_col.var">col.var</code></td>
<td>
<p>a color for the categories of categorical variables, if color =&quot;none&quot; the label is not written</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_col.quali.sup">col.quali.sup</code></td>
<td>
<p>a color for the categorical supplementary variables, if color =&quot;none&quot; the label is not written</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_col.ind.sup">col.ind.sup</code></td>
<td>
<p>a color for the supplementary individuals only if there is not habillage, if color =&quot;none&quot; the label is not written</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_col.quanti.sup">col.quanti.sup</code></td>
<td>
<p>a color for the supplementary quantitative variables, if color =&quot;none&quot; the label is not written</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_label">label</code></td>
<td>
<p>print the labels of the points; &quot;all&quot; print all the labels; may be a vector with &quot;ind&quot; (for the individuals),&quot;ind.sup&quot; (for the supplementary individuals),&quot;var&quot; (for the active categories), &quot;quali.sup&quot; &quot;var&quot; (for the supplementary categories)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;none&quot;, one color is used for the individual, another one for the categorical variables;
if &quot;quali&quot;, one color is used for each categorical variables; else if it is the position of a categorical variable, it colors 
according to the different categories of this variable</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_select">select</code></td>
<td>
<p>a selection of the elements that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_selectmod">selectMod</code></td>
<td>
<p>a selection of the categories that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if <code>unselect=1</code> the transparceny is total and the elements are not drawn, if <code>unselect=0</code> the elements are drawn as usual but without any label) or may be a color (for example <code>unselect="grey60"</code>)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program)</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_legend">legend</code></td>
<td>
<p>a list of arguments that defines the legend if needed (when individuals are drawn according to a variable); see the arguments of the function <code>legend</code></p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="plot.MCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>autoLab = "yes"</code> is time-consuming if there are many labels that overlap. In this case, you can modify the size of the characters in order to have less overlapping, using for example cex=0.7.<br />
The <code>select</code> argument can be used in order to select a part of the elements (individuals if you draw the graph of individuals, or variables if you draw the graph of variabless) that are drawn.
For example, you can use:<br />
<code>select = 1:5</code> and then the elements 1:5 are drawn.<br /> 
<code>select = c("name1","name5")</code> and then the elements that have the names name1 and name5 are drawn.<br /> 
<code>select = "coord 10"</code> and then the 10 elements that have the highest (squared) coordinates on the 2 chosen dimensions are drawn.<br /> 
<code>select = "contrib 10"</code> and then the 10 elements that have the highest contribution on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "cos2 5"</code> and then the 5 elements that have the highest cos2 on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "dist 8"</code> and then the 8 elements that have the highest distance to the center of gravity are drawn.<br /> 
The <code>selectMod</code> argument can be used in order to select the categories that are drawn.
</p>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MCA">MCA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data (poison)
res.mca = MCA (poison, quali.sup = 3:4, quanti.sup = 1:2, graph=FALSE)
plot(res.mca,invisible=c("var","quali.sup"))
plot(res.mca,invisible="ind")
plot(res.mca,choix="var")
plot(res.mca,invisible=c("ind"), selectMod="cos2 10")
## Not run: 
plot(res.mca, selectMod="cos2 5", select="cos2 5")

## You can modify the ggplot graphs as ususal with ggplot2
require(ggplot2)
gr &lt;- plot(res.mca)
gr + theme(panel.grid.major = element_blank(),
   plot.title=element_text(size=14, color="blue"),
   axis.title = element_text(size=12, color="red"))

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.meansComp'>Draw the means comparisons</h2><span id='topic+plot.meansComp'></span>

<h3>Description</h3>

<p>Plot the graphs for the means comparisons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meansComp'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.meansComp_+3A_x">x</code></td>
<td>
<p>an object of class meansComp.</p>
</td></tr>
<tr><td><code id="plot.meansComp_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as ggplot, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+meansComp">meansComp</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(senso)
  res &lt;- LinearModel(Score~ Product + Day , data=senso, selection="none")
  meansComp(res,~Product)
  
## Not run: 
  ## and with the sidak correction
  meansComp(res,~Product,adjust="sidak")

## End(Not run)</code></pre>

<hr>
<h2 id='plot.MFA'>Draw the Multiple Factor Analysis (MFA) graphs</h2><span id='topic+plot.MFA'></span>

<h3>Description</h3>

<p>Draw the Multiple Factor Analysis (MFA) graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MFA'
plot(x, axes = c(1, 2), choix = c("ind","var","group","axes","freq"), 
    ellipse=NULL, ellipse.par=NULL,
    lab.grpe=TRUE, lab.var=TRUE, lab.ind=TRUE, 
    lab.par=FALSE, lab.col=TRUE, ncp=2, habillage="group", col.hab=NULL, 
    invisible = c("none","ind","ind.sup","quanti","quanti.sup",
	"quali","quali.sup","row","row.sup","col","col.sup"), 
	partial = NULL, lim.cos2.var = 0., 
    chrono = FALSE, xlim = NULL, ylim = NULL, 
    title = NULL, palette = NULL, 
	autoLab = c("auto","yes","no"), new.plot = FALSE, 
	select = NULL, unselect = 0.7, shadowtext = FALSE, 
	legend = list(bty = "y", x = "topleft"),
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MFA_+3A_x">x</code></td>
<td>
<p>an object of class MFA</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_choix">choix</code></td>
<td>
<p>a string corresponding to the graph that you want to do (&quot;ind&quot; for the individual or categorical
variables graph, &quot;var&quot; for the quantitative variables graph, &quot;freq&quot; for the frequence or contingency tables,
&quot;axes&quot; for the graph of the partial axes, &quot;group&quot; for the groups representation)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_ellipse">ellipse</code></td>
<td>
<p>boolean (NULL by default), if not null, draw ellipses around the individuals, and use the results of <code>coord.ellipse</code></p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_ellipse.par">ellipse.par</code></td>
<td>
<p>boolean (NULL by default), if not null, draw ellipses around the partial individuals, and use the results of <code>coord.ellipse</code></p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lab.grpe">lab.grpe</code></td>
<td>
<p>boolean, if TRUE, the labels of the groups are drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lab.var">lab.var</code></td>
<td>
<p>boolean, if TRUE, the labels of the variables are drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lab.ind">lab.ind</code></td>
<td>
<p>boolean, if TRUE, the labels of the mean points are drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lab.par">lab.par</code></td>
<td>
<p>boolean, if TRUE, the labels of the partial points are drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lab.col">lab.col</code></td>
<td>
<p>boolean, if TRUE, the labels of the columns for the contingency tables are drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_ncp">ncp</code></td>
<td>
<p>number of principal components drawn for the separate analyses for the graph of the partial axes</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;ind&quot;, one color is used for each individual;
if &quot;group&quot; the individuals are colored according to the group; else if it is the name or the position of a categorical variable, it colors 
according to the different categories of this variable</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_col.hab">col.hab</code></td>
<td>
<p>the colors to use. By default, colors are chosen</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_invisible">invisible</code></td>
<td>
<p>list of string; for choix =&quot;ind&quot;, the individuals can be omit (invisible = &quot;ind&quot;),
or supplementary individuals (invisible=&quot;ind.sup&quot;) or the center of gravity of the categorical variables 
(invisible= &quot;quali&quot; or &quot;quali.sup&quot; for the supplementary categories); 
if invisible = c(&quot;ind&quot;,&quot;ind.sup&quot;), just the centers of gravity are drawn; if choix=&quot;var&quot;, invisible=&quot;quanti&quot; suppress the
active variable and invisible = &quot;quanti.sup&quot; suppress the supplementary variables</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_partial">partial</code></td>
<td>
<p>list of the individuals or of the center of gravity for which the partial points should be drawn (by
default, partial = NULL and no partial points are drawn)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under with the points are not drawn</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_chrono">chrono</code></td>
<td>
<p>boolean, if TRUE, the partial points of a same point are linked (useful when groups correspond to different moment)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_select">select</code></td>
<td>
<p>a selection of the elements that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if <code>unselect=1</code> the transparceny is total and the elements are not drawn, if <code>unselect=0</code> the elements are drawn as usual but without any label) or may be a color (for example <code>unselect="grey60"</code>)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program)</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_legend">legend</code></td>
<td>
<p>a list of arguments that defines the legend if needed (when individuals are drawn according to a variable); see the arguments of the function <code>legend</code></p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="plot.MFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>autoLab = "yes"</code> is time-consuming if there are many labels that overlap. In this case, you can modify the size of the characters in order to have less overlapping, using for example cex=0.7.<br />
The <code>select</code> argument can be used in order to select a part of the elements (individuals if you draw the graph of individuals, or variables if you draw the graph of variabless) that are drawn.
For example, you can use:<br />
<code>select = 1:5</code> and then the elements 1:5 are drawn.<br /> 
<code>select = c("name1","name5")</code> and then the elements that have the names name1 and name5 are drawn.<br /> 
<code>select = "coord 10"</code> and then the 10 elements (individuals, variables, frequencies) that have the highest (squared) coordinates on the 2 chosen dimensions are drawn.<br /> 
<code>select = "contrib 10"</code> and then the 10 elements (individuals, variables, frequencies) that have the highest contribution on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "cos2 5"</code> and then the 5 elements (individuals, variables, frequencies) that have the highest cos2 on the 2 dimensions of your plot are drawn.<br /> 
<code>selectMod = "contrib 5"</code>, the 5 categories that contribute the most to the two dimensions are drawn.<br /> 
<code>selectMod = "cos2 0.5"</code>, the categories that have a cos2 greater than 0.5 on the two dimensions are drawn.<br /> 
<code>selectMod = "v.test 2"</code>, the categories that have a v.test higher than the value on one of the two dimensions are drawn.<br /> 
</p>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Jeremy Mazet</p>


<h3>See Also</h3>

 <p><code><a href="#topic+MFA">MFA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res &lt;- MFA(wine,group=c(2,5,3,10,9,2),type=c("n",rep("s",5)),ncp=5,
    name.group=c("orig","olf","vis","olfag","gust","ens"),
    num.group.sup=c(1,6),graph=FALSE)
plot(res, choix = "ind")
plot(res, choix = "ind", partial="all")
plot(res, choix = "ind", habillage="Label")
plot(res, choix = "var", habillage="group")
plot(res, choix = "axes")

data(wine)
res &lt;- MFA(wine, group=c(2,5,3,10,9,2), type=c("n",rep("s",5)),
    ncp=5, name.group=c("orig","olf","vis","olfag","gust","ens"),
    num.group.sup=c(1,6))
summary(res)
barplot(res$eig[,1],main="Eigenvalues",names.arg=1:nrow(res$eig))

#### Confidence ellipses around categories per variable
plotellipses(res)
plotellipses(res,keepvar="Label") ## for 1 variable

#### Interactive graph
liste = plotMFApartial(res)
plot(res,choix="ind",habillage = "Terroir")

###Example with groups of categorical variables
data (poison)
MFA(poison, group=c(2,2,5,6), type=c("s","n","n","n"),
    name.group=c("desc","desc2","symptom","eat"),
    num.group.sup=1:2)

###Example with groups of frequency tables
data(mortality)
res&lt;-MFA(mortality,group=c(9,9),type=c("f","f"),
    name.group=c("1979","2006"))

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.PCA'>Draw the Principal Component Analysis (PCA) graphs</h2><span id='topic+plot.PCA'></span>

<h3>Description</h3>

<p>Plot the graphs for a Principal Component Analysis (PCA) with supplementary individuals, supplementary quantitative 
variables and supplementary categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
plot(x, axes = c(1, 2), choix = c("ind","var","varcor"),
    ellipse = NULL, xlim = NULL, ylim = NULL, habillage="none", 
    col.hab = NULL, col.ind="black", col.ind.sup="blue", 
    col.quali="magenta", col.quanti.sup="blue", col.var="black",
    label = c("all","none","ind","ind.sup","quali","var","quanti.sup"),
	invisible = c("none","ind","ind.sup","quali","var","quanti.sup"), 
    lim.cos2.var = 0., title = NULL, palette=NULL,
    autoLab = c("auto","yes","no"), new.plot = FALSE, select = NULL, 
	unselect = 0.7, shadowtext = FALSE, legend = list(bty = "y", x = "topleft"),
	graph.type = c("ggplot","classic"), ggoptions = NULL, ...)
    </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PCA_+3A_x">x</code></td>
<td>
<p>an object of class PCA</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;ind&quot; for the individuals, &quot;var&quot; for the variables, &quot;varcor&quot; for a graph with the correlation circle when <code>scale.unit=FALSE</code>)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ellipse">ellipse</code></td>
<td>
<p>boolean (NULL by default), if not null, draw ellipses around the individuals, and use the results of <code>coord.ellipse</code></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_habillage">habillage</code></td>
<td>
<p>give no color for the individuals (&quot;none&quot;), a color for each individual (&quot;ind&quot;),  or color the individuals among a categorical variable (give the number of the categorical variable)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.hab">col.hab</code></td>
<td>
<p>a vector with the color to use for the individuals</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.ind">col.ind</code></td>
<td>
<p>a color for the individuals only if there is not habillage</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.ind.sup">col.ind.sup</code></td>
<td>
<p>a color for the supplementary individuals only if there is not habillage</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.quali">col.quali</code></td>
<td>
<p>a color for the categories of categorical variables only  if there is not habillage</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.quanti.sup">col.quanti.sup</code></td>
<td>
<p>a color for the quantitative supplementary variables</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.var">col.var</code></td>
<td>
<p>a color for the variables</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_label">label</code></td>
<td>
<p>a list of character for the elements which are labelled (by default, all the elements are labelled (&quot;ind&quot;, ind.sup&quot;, &quot;quali&quot;, &quot;var&quot;, &quot;quanti.sup&quot;))</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_invisible">invisible</code></td>
<td>
<p>string indicating if some points should not be drawn (&quot;ind&quot;, &quot;ind.sup&quot; or &quot;quali&quot; for the individual graph and &quot;var&quot; or &quot;quanti.sup&quot; for
the correlation circle graph)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lim.cos2.var">lim.cos2.var</code></td>
<td>
<p>value of the square cosinus under the variables are not drawn</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;yes&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;yes&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_new.plot">new.plot</code></td>
<td>
<p>boolean, if TRUE, a new graphical device is created; only used when <code>graph.type="classic"</code></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_select">select</code></td>
<td>
<p>a selection of the elements that are drawn; see the details section</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_unselect">unselect</code></td>
<td>
<p>may be either a value between 0 and 1 that gives the transparency of the unselected objects (if <code>unselect=1</code> the transparceny is total and the elements are not drawn, if <code>unselect=0</code> the elements are drawn as usual but without any label) or may be a color (for example <code>unselect="grey60"</code>)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_shadowtext">shadowtext</code></td>
<td>
<p>boolean; if true put a shadow on the labels (rectangles are written under the labels which may lead to difficulties to modify the graph with another program); only used when <code>graph.type="classic"</code></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_legend">legend</code></td>
<td>
<p>a list of arguments that defines the legend if needed (when individuals are drawn according to a variable); see the arguments of the function <code>legend</code></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ggoptions">ggoptions</code></td>
<td>
<p>a list that gives the graph options when grah.type=&quot;ggplot&quot; is used. See the optines and the default values in the details section</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>autoLab = "yes"</code> is time-consuming if there are many labels that overlap. In this case, you can modify the size of the characters in order to have less overlapping, using for example cex=0.7.<br />
The <code>select</code> argument can be used in order to select a part of the elements (individuals if you draw the graph of individuals, or variables if you draw the graph of variables) that are drawn.
For example, you can use:<br />
<code>select = 1:5</code> and then the elements 1:5 are drawn.<br /> 
<code>select = c("name1","name5")</code> and then the elements that have the names name1 and name5 are drawn.<br /> 
<code>select = "coord 10"</code> and then the 10 elements that have the highest (squared) coordinates on the 2 chosen dimensions are drawn.<br /> 
<code>select = "contrib 10"</code> and then the 10 elements that have the highest contribution on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "cos2 5"</code> and then the 5 elements that have the highest cos2 on the 2 dimensions of your plot are drawn.<br /> 
<code>select = "dist 8"</code> and then the 8 elements that have the highest distance to the center of gravity are drawn.<br /> 
<br />
<code>ggoptions</code> is a list that gives some ggplot2 options when the graph.type=&quot;ggplot&quot; is used. Use for instance ggoptions(list(size=3,title.size=10,bg.color=&quot;orange&quot;)) if you want to modify the size of the points and labels, the title size and the background color.<br />
Below you can see the options and the default values: <br />
size = 4, #label size (point size = size/3)<br />
point.shape = 19, #points shape<br />
line.lty = 2, #origin linetypes (0=&quot;blank&quot;, 1=&quot;solid&quot;, 2=&quot;dashed&quot;, 3=&quot;dotted&quot;,...)<br />
line.lwd = 0, #origin lines width<br />
line.color = &quot;black&quot;, #origin lines color<br />
segment.lty = 1, #arrow linetypes (0=&quot;blank&quot;, 1=&quot;solid&quot;, 2=&quot;dashed&quot;, 3=&quot;dotted&quot;,...)<br />
segment.lwd = 0, #arrow width<br />
circle.lty = 1, #circle linetypes (0=&quot;blank&quot;, 1=&quot;solid&quot;, 2=&quot;dashed&quot;, 3=&quot;dotted&quot;,...)<br />
circle.lwd = 0, #circle width<br />
circle.color = &quot;black&quot;, #circle color<br />
low.col.quanti = &quot;blue&quot;, #for quantitative variables, low color to be used<br />
high.col.quanti = &quot;red3&quot;, #for quantitative variables, high color to be used<br />
</p>


<h3>Value</h3>

<p>Returns the individuals factor map and the variables factor map.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup = 13)
plot(res.pca, habillage = 13, cex=0.8)
## Not run: 
plot(res.pca, habillage = "cos2")
plot(res.pca, habillage = "100m")
plot(res.pca, habillage = c("Competition","100m"))

## End(Not run)
## To automatically draw ellipses around the barycentres of the categorical variables
plotellipses(res.pca)

## Not run: 
## Selection of some individuals
plot(res.pca,select="contrib 7") # plot the 7 individuals with the highest contribution 
plot(res.pca,select="cos2 0.8")  # plot the individuals with cos2 greater than 0.8
plot(res.pca,select="cos2 5")    # plot the 5 individuals with the highest cos2 
plot(res.pca,choix="var",select="cos2 0.6")  # plot the variables with cos2 greater than 0.6

plot(res.pca,habillage="100m",
   ggoptions=list(low.col.quanti="grey90",high.col.quanti="grey10"),legend=list(x="bottom"))

## You can modify the ggplot graphs as ususal with ggplot2
require(ggplot2)
gr &lt;- plot(res.pca)
gr + theme(panel.grid.major = element_blank(),
   plot.title=element_text(size=14, color="blue"),
   axis.title = element_text(size=12, color="red"))

## To draw classical R graphs
plot(res.pca, graph.type = "classic")

## End(Not run)
</code></pre>

<hr>
<h2 id='plotellipses'>Draw confidence ellipses around the categories</h2><span id='topic+plotellipses'></span>

<h3>Description</h3>

<p>Draw confidence ellipses around the categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotellipses(model, keepvar = "all", axes = c(1, 2), means=TRUE, level = 0.95, 
    magnify = 2, cex = 1, pch = 20, pch.means=15, type = c("g","p"), 
    keepnames = TRUE, namescat = NULL, xlim=NULL, ylim=NULL, lwd=1, 
    label="all", autoLab=c("auto","yes","no"), 
	graph.type = c("ggplot","classic"), ...)
	</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotellipses_+3A_model">model</code></td>
<td>
<p>an object of class MCA or PCA or MFA</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_keepvar">keepvar</code></td>
<td>
<p>a boolean or numeric vector of
indexes of variables or a character vector of names of variables. If
keepvar is &quot;all&quot;, &quot;quali&quot; or &quot;quali.sup&quot; variables which are plotted
are all the categorical variables, only those which are used to
compute the dimensions (active variables) or only the supplementary
categorical variables. If keepvar is a numeric vector of indexes or a
character vector of names of variables, only relevant variables are plotted.</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_means">means</code></td>
<td>
<p>boolean which indicates if the confidence ellipses are
for (the coordinates of) the means of the
categories (the empirical variance is divided  by the number of
observations) or for (the coordinates of) the observations of the
categories</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_level">level</code></td>
<td>
<p>the confidence level for the ellipses</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_magnify">magnify</code></td>
<td>
<p>numeric which control how the level names are
magnified. A value of 2 means that the level names have character
expansion equal to two times cex</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_pch">pch</code></td>
<td>
<p>plotting character for coordinates, cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_pch.means">pch.means</code></td>
<td>
<p>plotting character for means, cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_type">type</code></td>
<td>
<p>cf. function <code><a href="lattice.html#topic+xyplot">xyplot</a></code> in the <span class="pkg">lattice</span> package</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_keepnames">keepnames</code></td>
<td>
<p>a boolean or numeric vector of
indexes of variables or a character vector of names of variables. If keepnames is TRUE, names of levels are taken
from the (modified) dataset extracted from modele, if FALSE trimming
names is done. When trimming, names of levels are taken from the
(modified) dataset extracted from modele, then, the corresponding
number of characters of names of original variables plus 1 is
removed. If keepnames is a vector of indexes or names, trimming is done on
all variables excepted whose in keepnames</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_namescat">namescat</code></td>
<td>
<p>a vector giving for each observation the value of
categorical variable, each variable are stacked under each other. If
NULL, names are taken from the (modified) dataset extracted from
modele</p>
</td></tr> 
<tr><td><code id="plotellipses_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_lwd">lwd</code></td>
<td>
<p>The line width, a positive number, defaulting to 1</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_label">label</code></td>
<td>
<p>a list of character for the elements which are labelled (by default, &quot;all&quot;, you can use &quot;none&quot;, &quot;ind&quot;, ind.sup&quot;))</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_autolab">autoLab</code></td>
<td>
<p>if <code>autoLab="auto"</code>, <code>autoLab</code> is equal to &quot;y&quot; if there are less than 50 elements and &quot;no&quot; otherwise; if &quot;y&quot;, the labels of the drawn elements are placed in a &quot;good&quot; way (can be time-consuming if many elements), and if &quot;no&quot; the elements are placed quickly but may overlap</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_graph.type">graph.type</code></td>
<td>
<p>a character that gives the type of graph used: &quot;ggplot&quot; or &quot;classic&quot;</p>
</td></tr>
<tr><td><code id="plotellipses_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a graph with the ellipses.
If only one variable is chosen, the graph is different.
</p>


<h3>Author(s)</h3>

<p>Pierre-Andre Cornillon, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MCA">MCA</a></code>, <code><a href="#topic+PCA">PCA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(poison)
res.mca = MCA(poison, quali.sup = 3:4, quanti.sup = 1:2)
plotellipses(res.mca)
plotellipses(res.mca,keepvar=3:6)

## End(Not run)

data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup=13)
plotellipses(res.pca,keepvar=13)
</code></pre>

<hr>
<h2 id='plotGPApartial'>Draw an interactive General Procrustes Analysis (GPA) map</h2><span id='topic+plotGPApartial'></span>

<h3>Description</h3>

<p>Draw an interactive General Procrustes Analysis (GPA) map. The graph is interactive and clicking on a point will draw the partial points,
if you click on a point for which the partial points are yet drawn, the partial points are deleted.
To stop the interactive plot, click on the title (or in the top of the graph)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotGPApartial(x, axes = c(1, 2), 
    lab.ind.moy = TRUE, habillage = "ind",
    chrono = FALSE, draw.partial = NULL, 
    xlim = NULL, ylim = NULL, cex = 1, title = NULL, palette = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotGPApartial_+3A_x">x</code></td>
<td>
<p>an object of class GPA</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_lab.ind.moy">lab.ind.moy</code></td>
<td>
<p>boolean, if TRUE, the label of the mean points are drawn</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;ind&quot;, one color is used for each individual;
if &quot;group&quot; the individuals are colored according to the group</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_chrono">chrono</code></td>
<td>
<p>boolean, if TRUE, the partial points of a same point are linked (useful when groups correspond to different moment)</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_draw.partial">draw.partial</code></td>
<td>
<p>data frame of a boolean variable for all the individuals and all the centers of gravity and with for which the partial points should be drawn (by
default, NULL and no partial points are drawn)</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plotGPApartial_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the General Procrustes Analysis map.
</p>


<h3>Author(s)</h3>

<p>Elisabeth Morand, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+GPA">GPA</a></code></p>

<hr>
<h2 id='plotMFApartial'>Plot an interactive Multiple Factor Analysis (MFA) graph</h2><span id='topic+plotMFApartial'></span>

<h3>Description</h3>

<p>Draw an interactive Multiple Factor Analysis (MFA) graphs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMFApartial(x, axes = c(1, 2), 
    lab.ind = TRUE, lab.par = FALSE, habillage = "group",
    chrono = FALSE, col.hab = NULL, invisible = NULL, 
    draw.partial = NULL, xlim = NULL, ylim = NULL, 
    cex = 1, title = NULL, palette = NULL, ...)
</code></pre>


<h3>Arguments</h3>

 
<table>
<tr><td><code id="plotMFApartial_+3A_x">x</code></td>
<td>
<p>an object of class MFA</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_axes">axes</code></td>
<td>
<p>a length 2 vector specifying the components to plot</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_lab.ind">lab.ind</code></td>
<td>
<p>boolean, if TRUE, the label of the mean points are drawn</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_lab.par">lab.par</code></td>
<td>
<p>boolean, if TRUE, the label of the partial points are drawn</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_habillage">habillage</code></td>
<td>
<p>string corresponding to the color which are used. If &quot;group&quot;, one color is used for each group of variables;
if &quot;quali&quot; the individuals are colored according to one categorical variable; if &quot;group&quot; the individuals are colored according to the group</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_chrono">chrono</code></td>
<td>
<p>boolean, if TRUE, the partial points of a same point are linked (useful when groups correspond to different moment)</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_col.hab">col.hab</code></td>
<td>
<p>the colors to use. By default, colors are chosen</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_invisible">invisible</code></td>
<td>
<p>list of string; for choix =&quot;ind&quot;, the individuals can be omit (invisible = &quot;ind&quot;),
or supplementary individuals (invisible=&quot;ind.sup&quot;) or the centerg of gravity of the categorical variables 
(invisible= &quot;quali&quot;); if invisible = c(&quot;ind&quot;,&quot;ind.sup&quot;), just the centers of gravity are drawn</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_draw.partial">draw.partial</code></td>
<td>
<p>data frame of a boolean variable for all the individuals and all the centers of gravity and with for which the partial points should be drawn (by
default, NULL and no partial points are drawn)</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_xlim">xlim</code></td>
<td>
<p>range for the plotted 'x' values, defaulting to the range of the finite values of 'x'</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_ylim">ylim</code></td>
<td>
<p>range for the plotted 'y' values, defaulting to the range of the finite values of 'y'</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_cex">cex</code></td>
<td>
<p>cf. function <code><a href="graphics.html#topic+par">par</a></code> in the <span class="pkg">graphics</span> package</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_palette">palette</code></td>
<td>
<p>the color palette used to draw the points. By default colors are chosen. If you want to define the colors : palette=palette(c(&quot;black&quot;,&quot;red&quot;,&quot;blue&quot;)); or you can use: palette=palette(rainbow(30)), or in black and white for example: palette=palette(gray(seq(0,.9,len=25)))</p>
</td></tr>
<tr><td><code id="plotMFApartial_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Draw a graph with the individuals and the centers of gravity. The graph is interactive and clicking on a point will draw the partial points,
if you click on a point for which the partial points are yet drawn, the partial points are deleted.
To stop the interactive plot, click on the title (or in the top of the graph).<br />
Return the names of the points for which the partial points are drawn.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MFA">MFA</a></code>, <code><a href="#topic+plot.MFA">plot.MFA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res.wine = MFA(wine,group=c(2,5,3,10,9,2),type=c("n",rep("s",5)),ncp=5,
    name.group=c("orig","olf","vis","olfag","gust","ens"),
    num.group.sup=c(1,6),graph=FALSE)
liste = plotMFApartial(res.wine)
plot(res.wine, partial = liste)

## End(Not run)
</code></pre>

<hr>
<h2 id='poison'>Poison</h2><span id='topic+poison'></span>

<h3>Description</h3>

<p>The data used here refer to a survey carried out on a sample of children of primary school
who suffered from food poisoning. They were asked about their symptoms and about what they ate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poison)</code></pre>


<h3>Format</h3>

<p>A data frame with 55 rows and 15 columns.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(poison)
res.mca &lt;- MCA(poison, quanti.sup = 1:2, quali.sup=c(3,4))

## End(Not run)
</code></pre>

<hr>
<h2 id='poison.text'>Poison</h2><span id='topic+poison.text'></span>

<h3>Description</h3>

<p>The data used here refer to a survey carried out on a sample of children of primary school
who suffered from food poisoning. They were asked about their symptoms and about what they ate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poison)</code></pre>


<h3>Format</h3>

<p>A data frame with 55 rows and 3 columns (the sex, if they are sick or not, and a textual variable with their symptom and what they eat).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(poison.text)
res.text &lt;- textual(poison.text, num.text = 3, contingence.by = c(1,2))
## Contingence table for the sex variable, the sich variable and the couple
## of variable sick-sex
res.text2 &lt;- textual(poison.text, num.text = 3, contingence.by = list(1,2,c(1,2)))
</code></pre>

<hr>
<h2 id='poulet'>Genomic data for chicken</h2><span id='topic+poulet'></span>

<h3>Description</h3>

<p>Genomic data for chicken</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poulet)</code></pre>


<h3>Format</h3>

<p>A data frame with 43 chickens and 7407 variables.  
A factor with levels J16 J16R16 J16R5 J48 J48R24 N<br />
And many continuous variables corresponding to the gene expression
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(poulet)
res.pca = PCA(poulet,quali.sup=1, graph=FALSE)
plot(res.pca)
plot(res.pca,habillage=1,label="quali",
    palette=palette(c("black","red","blue","darkgreen","purple","orange")))
dimdesc(res.pca)
## Dessine des ellipses autour des centres de gravite
plotellipses(res.pca)
## End(Not run)
</code></pre>

<hr>
<h2 id='predict.CA'>Predict projection for new rows with Correspondence Analysis</h2><span id='topic+predict.CA'></span>

<h3>Description</h3>

<p>Predict the projection of new rows with Correspondence Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CA'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.CA_+3A_object">object</code></td>
<td>
<p>an object of class CA</p>
</td></tr>
<tr><td><code id="predict.CA_+3A_newdata">newdata</code></td>
<td>
<p>A data frame or a matrix in which to look for variables with which to predict. newdata must contain columns with the same names as the original data.</p>
</td></tr>
<tr><td><code id="predict.CA_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+CA">CA</a></code></p>

<hr>
<h2 id='predict.FAMD'>Predict projection for new rows with Factor Analysis of Mixed Data</h2><span id='topic+predict.FAMD'></span>

<h3>Description</h3>

<p>Predict the projection of new rows with Factor Analysis of Mixed Data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FAMD'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.FAMD_+3A_object">object</code></td>
<td>
<p>an object of class FAMD</p>
</td></tr>
<tr><td><code id="predict.FAMD_+3A_newdata">newdata</code></td>
<td>
<p>A data frame or a matrix in which to look for variables with which to predict. newdata must contain columns with the same names as the original data.</p>
</td></tr>
<tr><td><code id="predict.FAMD_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+FAMD">FAMD</a></code></p>

<hr>
<h2 id='predict.LinearModel'>Predict method for Linear Model Fits</h2><span id='topic+predict.LinearModel'></span>

<h3>Description</h3>

<p>Predicted values based on LinearModel object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LinearModel'
predict(object, newdata, interval = c("none", "confidence", "prediction"),
        level = 0.95, type = c("response", "terms"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.LinearModel_+3A_object">object</code></td>
<td>
<p>Object of class inheriting from &quot;LinearModel&quot;</p>
</td></tr>
<tr><td><code id="predict.LinearModel_+3A_newdata">newdata</code></td>
<td>
<p>An optional data frame in which to look for variables with which to predict. If omitted, the fitted values are used.</p>
</td></tr>
<tr><td><code id="predict.LinearModel_+3A_interval">interval</code></td>
<td>
<p>Type of interval calculation. Can be abbreviated.</p>
</td></tr>
<tr><td><code id="predict.LinearModel_+3A_level">level</code></td>
<td>
<p>Tolerance/confidence level.</p>
</td></tr>
<tr><td><code id="predict.LinearModel_+3A_type">type</code></td>
<td>
<p>Type of prediction (response or model term). Can be abbreviated.</p>
</td></tr>
<tr><td><code id="predict.LinearModel_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods such as <code>lm</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the help of <code><a href="stats.html#topic+predict.lm">predict.lm</a></code> function.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+LinearModel">LinearModel</a></code></p>

<hr>
<h2 id='predict.MCA'>Predict projection for new rows with Multiple Correspondence Analysis</h2><span id='topic+predict.MCA'></span>

<h3>Description</h3>

<p>Predict the projection of new rows with Multiple Correspondence Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MCA'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.MCA_+3A_object">object</code></td>
<td>
<p>an object of class MCA</p>
</td></tr>
<tr><td><code id="predict.MCA_+3A_newdata">newdata</code></td>
<td>
<p>A data frame or a matrix in which to look for variables with which to predict. newdata must contain columns with the same names as the original data.</p>
</td></tr>
<tr><td><code id="predict.MCA_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MCA">MCA</a></code></p>

<hr>
<h2 id='predict.MFA'>Predict projection for new rows with Multiple Factor Analysis</h2><span id='topic+predict.MFA'></span>

<h3>Description</h3>

<p>Predict the projection of new rows with Multiple Factor Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MFA'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.MFA_+3A_object">object</code></td>
<td>
<p>an object of class MFA</p>
</td></tr>
<tr><td><code id="predict.MFA_+3A_newdata">newdata</code></td>
<td>
<p>A data frame or a matrix in which to look for variables with which to predict. newdata must contain columns with the same names as the original data.</p>
</td></tr>
<tr><td><code id="predict.MFA_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MFA">MFA</a></code></p>

<hr>
<h2 id='predict.PCA'>Predict projection for new rows with Principal Component Analysis</h2><span id='topic+predict.PCA'></span>

<h3>Description</h3>

<p>Predict the projection of new rows with Principal Component Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.PCA_+3A_object">object</code></td>
<td>
<p>an object of class PCA</p>
</td></tr>
<tr><td><code id="predict.PCA_+3A_newdata">newdata</code></td>
<td>
<p>A data frame or a matrix in which to look for variables with which to predict. newdata must contain columns with the same names as the original data.</p>
</td></tr>
<tr><td><code id="predict.PCA_+3A_...">...</code></td>
<td>
<p>Other options.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code></p>

<hr>
<h2 id='prefpls'>Scatter plot and additional variables with quality of representation contour lines</h2><span id='topic+prefpls'></span>

<h3>Description</h3>

<p>This function is useful to interpret the usual graphs $(x,y)$ with additional
quantitative variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prefpls(donnee, var1 = 1, var2 = 2, firstvar = 3, 
    lastvar = ncol(donnee), levels = c(0.2,0.4,0.6,0.7,0.8,0.9,1), 
    asp = 1, nbchar = max(nchar(colnames(donnee))), title = NULL,
    choix="var")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prefpls_+3A_donnee">donnee</code></td>
<td>
<p>a data frame made up of quantitative variables</p>
</td></tr>  
<tr><td><code id="prefpls_+3A_var1">var1</code></td>
<td>
<p>the position of the variable corresponding to the x-axis</p>
</td></tr>
<tr><td><code id="prefpls_+3A_var2">var2</code></td>
<td>
<p>the position of the variable corresponding to the y-axis</p>
</td></tr>
<tr><td><code id="prefpls_+3A_firstvar">firstvar</code></td>
<td>
<p>the position of the first endogenous variable</p>
</td></tr>
<tr><td><code id="prefpls_+3A_lastvar">lastvar</code></td>
<td>
<p>the position of the last endogenous variable (by default the last column of <code>donnee</code>)</p>
</td></tr>
<tr><td><code id="prefpls_+3A_levels">levels</code></td>
<td>
<p>a list of the levels displayed in the graph of variables</p>
</td></tr>
<tr><td><code id="prefpls_+3A_asp">asp</code></td>
<td>
<p>aspect ratio for the graph of the individuals</p>
</td></tr>
<tr><td><code id="prefpls_+3A_nbchar">nbchar</code></td>
<td>
<p>the number of characters used for the labels of the variables</p>
</td></tr>
<tr><td><code id="prefpls_+3A_title">title</code></td>
<td>
<p>string corresponding to the title of the graph you draw (by default NULL and a title is chosen)</p>
</td></tr>
<tr><td><code id="prefpls_+3A_choix">choix</code></td>
<td>
<p>the graph to plot (&quot;ind&quot; for the individuals, &quot;var&quot; for the variables)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is very useful when there is a strong correlation between two variables <em>x</em> and <em>y</em>
</p>


<h3>Value</h3>

<p>A scatter plot of the invividuals<br />
A graph with additional variables and the quality of representation contour lines.
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>References</h3>

<p>Husson, F. &amp; Pages, J. (2005). Scatter plot and additional variables. <em>Journal of applied statistics</em></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
prefpls(decathlon[,c(11,12,1:10)])
</code></pre>

<hr>
<h2 id='print.AovSum'>Print the AovSum results</h2><span id='topic+print.AovSum'></span>

<h3>Description</h3>

<p>Print the results of the ANOVA obtained by the function AovSum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'AovSum'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.AovSum_+3A_x">x</code></td>
<td>
<p>an object of class AovSum</p>
</td></tr>
<tr><td><code id="print.AovSum_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Vincent Guyader, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+AovSum">AovSum</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(senso)
res &lt;- AovSum(Score~ Product + Day , data=senso)
res

## End(Not run)
</code></pre>

<hr>
<h2 id='print.CA'>Print the Correspondance Analysis (CA) results</h2><span id='topic+print.CA'></span>

<h3>Description</h3>

<p>Print the Correspondance Analysis (CA) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.CA_+3A_x">x</code></td>
<td>
<p>an object of class CA</p>
</td></tr>
<tr><td><code id="print.CA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.CA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.CA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+CA">CA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.CaGalt'>Print the Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) results</h2><span id='topic+print.CaGalt'></span>

<h3>Description</h3>

<p>Print the Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CaGalt'
## S3 method for class 'CaGalt'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.CaGalt_+3A_x">x</code></td>
<td>
<p>an object of class CaGalt</p>
</td></tr>
<tr><td><code id="print.CaGalt_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to. If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.CaGalt_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL)</p>
</td></tr>
<tr><td><code id="print.CaGalt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Belchin Kostov <a href="mailto:badriyan@clinic.ub.es">badriyan@clinic.ub.es</a>, Monica Becue-Bertaut, Francois Husson</p>


<h3>See Also</h3>

 <p><code><a href="#topic+CaGalt">CaGalt</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(health)
res.cagalt&lt;-CaGalt(Y=health[,1:115],X=health[,116:118],type="n")
print(res.cagalt)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.catdes'>Print the catdes results</h2><span id='topic+print.catdes'></span>

<h3>Description</h3>

<p>Print the results of the function catdes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'catdes'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.catdes_+3A_x">x</code></td>
<td>
<p>an object of class catdes</p>
</td></tr>
<tr><td><code id="print.catdes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Vincent Guyader, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+catdes">catdes</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res &lt;- catdes(wine, num.var=2)
print(res)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.condes'>Print the condes results</h2><span id='topic+print.condes'></span>

<h3>Description</h3>

<p>Print the results of the function condes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'condes'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.condes_+3A_x">x</code></td>
<td>
<p>an object of class condes</p>
</td></tr>
<tr><td><code id="print.condes_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+condes">condes</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(wine)
res &lt;- condes(wine, num.var=3)
print(res)

## End(Not run)
</code></pre>

<hr>
<h2 id='print.FAMD'>Print the Multiple Factor Analysis of mixt Data (FAMD) results</h2><span id='topic+print.FAMD'></span>

<h3>Description</h3>

<p>Print the Multiple Factor Analysis of mixt Data (FAMD) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FAMD'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.FAMD_+3A_x">x</code></td>
<td>
<p>an object of class FAMD</p>
</td></tr>
<tr><td><code id="print.FAMD_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.FAMD_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.FAMD_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+FAMD">FAMD</a></code></p>

<hr>
<h2 id='print.GPA'>Print the Generalized Procrustes Analysis (GPA) results</h2><span id='topic+print.GPA'></span>

<h3>Description</h3>

<p>Print the Generalized Procrustes Analysis (GPA) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GPA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.GPA_+3A_x">x</code></td>
<td>
<p>an object of class GPA</p>
</td></tr>
<tr><td><code id="print.GPA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.GPA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.GPA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Elisabeth Morand, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+GPA">GPA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.HCPC'>Print the Hierarchical Clustering on Principal Components (HCPC) results</h2><span id='topic+print.HCPC'></span>

<h3>Description</h3>

<p>Print the Hierarchical Clustering on Principal Components (HCPC) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HCPC'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.HCPC_+3A_x">x</code></td>
<td>
<p>an object of class HCPC</p>
</td></tr>
<tr><td><code id="print.HCPC_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.HCPC_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.HCPC_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+HCPC">HCPC</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.HMFA'>Print the Hierarchical Multiple Factor Analysis results</h2><span id='topic+print.HMFA'></span>

<h3>Description</h3>

<p>Print the Hierarchical Multiple Factor Analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'HMFA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.HMFA_+3A_x">x</code></td>
<td>
<p>an object of class HMFA</p>
</td></tr>
<tr><td><code id="print.HMFA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.HMFA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.HMFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sebastien Le, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+HMFA">HMFA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.LinearModel'>Print the LinearModel results</h2><span id='topic+print.LinearModel'></span>

<h3>Description</h3>

<p>Print the results of the ANOVA obtained by the function LinearModel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LinearModel'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.LinearModel_+3A_x">x</code></td>
<td>
<p>an object of class LinearModel</p>
</td></tr>
<tr><td><code id="print.LinearModel_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gives the results of the LinearModel function. If a model selection is performed, the global F-test
for the complete model is first given, then all the results are given for the selected model
(global F-test, the F-tests for main effects and interaction, the t-tests)
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+LinearModel">LinearModel</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(senso)
res &lt;- LinearModel(Score~ Product + Day , data=senso)
res

res2 &lt;- LinearModel(Score~ Product + Day , data=senso, selection="BIC")
res2

## End(Not run)
</code></pre>

<hr>
<h2 id='print.MCA'>Print the Multiple Correspondance Analysis (MCA) results</h2><span id='topic+print.MCA'></span>

<h3>Description</h3>

<p>Print the Multiple Correspondance Analysis (spMCA) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MCA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.MCA_+3A_x">x</code></td>
<td>
<p>an object of class MCA</p>
</td></tr>
<tr><td><code id="print.MCA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.MCA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.MCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MCA">MCA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.MFA'>Print the Multiple Factor Analysis results</h2><span id='topic+print.MFA'></span>

<h3>Description</h3>

<p>Print the Multiple Factor Analysis results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MFA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.MFA_+3A_x">x</code></td>
<td>
<p>an object of class MFA</p>
</td></tr>
<tr><td><code id="print.MFA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.MFA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.MFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MFA">MFA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>

<hr>
<h2 id='print.PCA'>Print the Principal Component Analysis (PCA) results</h2><span id='topic+print.PCA'></span>

<h3>Description</h3>

<p>Print the Principal Component Analysis (PCA) results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
print(x, file = NULL, sep = ";", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.PCA_+3A_x">x</code></td>
<td>
<p>an object of class PCA</p>
</td></tr>
<tr><td><code id="print.PCA_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to.  If NULL (the default), the results are not printed in a file</p>
</td></tr>
<tr><td><code id="print.PCA_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print (if the argument file is not NULL</p>
</td></tr>
<tr><td><code id="print.PCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet, Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code>, <code><a href="#topic+write.infile">write.infile</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup = 13)
print(res.pca, file="c:/essai.csv", sep = ";")

## End(Not run)
</code></pre>

<hr>
<h2 id='reconst'>Reconstruction of the data from the PCA, CA or MFA results</h2><span id='topic+reconst'></span>

<h3>Description</h3>

<p>Reconstruct the data from the PCA, CA or MFA results.<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconst(res, ncp=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reconst_+3A_res">res</code></td>
<td>
<p>an object of class PCA, CA or MFA</p>
</td></tr>
<tr><td><code id="reconst_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions used to reconstitute the data (by default NULL and the number of dimensions calculated for the PCA, CA or MFA is used)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the number of individuals and the number of variables used for the PCA, CA or MFA
</p>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a>, Julie Josse<a href="mailto:Julie.Josse@agrocampus-ouest.fr">Julie.Josse@agrocampus-ouest.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code>,<code><a href="#topic+CA">CA</a></code>, <code><a href="#topic+MFA">MFA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup=13, graph=FALSE)
rec &lt;- reconst(res.pca,ncp=2)
</code></pre>

<hr>
<h2 id='RegBest'>Select variables in multiple linear regression</h2><span id='topic+RegBest'></span>

<h3>Description</h3>

<p>Find an optimal submodel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RegBest(y,x, int = TRUE, wt=NULL, na.action = na.omit,
    method=c("r2","Cp", "adjr2"), nbest=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RegBest_+3A_y">y</code></td>
<td>
<p>A response vector</p>
</td></tr>
<tr><td><code id="RegBest_+3A_x">x</code></td>
<td>
<p>A matrix of predictors</p>
</td></tr>
<tr><td><code id="RegBest_+3A_int">int</code></td>
<td>
<p>Add an intercept to the model</p>
</td></tr>
<tr><td><code id="RegBest_+3A_wt">wt</code></td>
<td>
<p>Optional weight vector</p>
</td></tr>
<tr><td><code id="RegBest_+3A_na.action">na.action</code></td>
<td>
<p>Handling missing values</p>
</td></tr>
<tr><td><code id="RegBest_+3A_method">method</code></td>
<td>
<p>Calculate R-squared, adjusted R-squared or Cp to select the model. By default a the F-test on the r-square is used</p>
</td></tr>
<tr><td><code id="RegBest_+3A_nbest">nbest</code></td>
<td>
<p>number of best models for each set of explained variables (by default 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the objects
</p>
<table>
<tr><td><code>all</code></td>
<td>
<p>gives all the <code>nbest</code> best models for a given number of variables</p>
</td></tr>
<tr><td><code>best</code></td>
<td>
<p>the best model</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+lm">lm</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(milk)
res = RegBest(y=milk[,6],x=milk[,-6])
res$best
</code></pre>

<hr>
<h2 id='senso'>senso</h2><span id='topic+senso'></span>

<h3>Description</h3>

<p>Dataset to illustrate one-way and Two-way analysis of variance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(senso)</code></pre>


<h3>Format</h3>

<p>Dataset with 45 rows and 3 columns:
Score, Product and Day</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example of 2-way analysis of variance
data(senso)
res &lt;- AovSum (Score~ Product + Day, data=senso)
res

## Example of 2-way analysis of variance with interaction
data(senso)
res2 &lt;- AovSum (Score~ Product + Day + Product : Day, data=senso)
res2

</code></pre>

<hr>
<h2 id='simule'>Simulate by bootstrap</h2><span id='topic+simule'></span>

<h3>Description</h3>

<p>Simulate by bootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simule(data, nb.simul)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simule_+3A_data">data</code></td>
<td>
<p>A data frame from which the rows are the original data from which the simualte data are calculated (by the average
of a bootstrap sample. The columns corresponds to the variables for which the simulation should be done. The first column must
be a factor allowing to group the rows. A bootstrap simulation is done for each level of this factor.</p>
</td></tr>
<tr><td><code id="simule_+3A_nb.simul">nb.simul</code></td>
<td>
<p>The number of simulations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The simulation is independently done for each level of the factor. The number of rows can be different for each levels.
</p>


<h3>Value</h3>

<table>
<tr><td><code>mean</code></td>
<td>
<p>Data.frame with all the levels of the factor variable, and for each variable, the mean of the original data.</p>
</td></tr>
<tr><td><code>simul</code></td>
<td>
<p>Data.frame with all the levels of the factor variable, and for each variable, the nb.simul bootstrap simulations.</p>
</td></tr>
<tr><td><code>simul.mean</code></td>
<td>
<p>Data.frame with all the levels of the factor variable, and for each variable, the mean of the simulated data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jeremy Mazet</p>

<hr>
<h2 id='summary.CA'>Printing summeries of ca objects</h2><span id='topic+summary.CA'></span>

<h3>Description</h3>

<p>Printing summaries of correspondence analysis objects</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CA'
summary(object, nb.dec = 3, nbelements=10,
   ncp = 3, align.names=TRUE, file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.CA_+3A_object">object</code></td>
<td>
<p>an object of class CA</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_nbelements">nbelements</code></td>
<td>
<p>number of elements written (rows, columns, ...) ; use <code>nbelements = Inf</code> if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions printed</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.CA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+CA">CA</a></code></p>

<hr>
<h2 id='summary.CaGalt'>Printing summaries of CaGalt objects</h2><span id='topic+summary.CaGalt'></span>

<h3>Description</h3>

<p>Printing summaries of Correspondence Analysis on Generalised Aggregated Lexical Table (CaGalt) objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CaGalt'
summary(object, nb.dec = 3, nbelements=10, nbind = nbelements, 
  ncp = 3, align.names=TRUE, file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.CaGalt_+3A_object">object</code></td>
<td>
<p>an object of class CaGalt</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of printed decimals</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_nbelements">nbelements</code></td>
<td>
<p>number of written elements (variables, categories, frequencies); use nbelements = Inf if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_nbind">nbind</code></td>
<td>
<p>number of written elements; use nbind = Inf to have the results for all the individuals and nbind = 0 if you do not want the results for individuals</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_ncp">ncp</code></td>
<td>
<p>number of printed dimensions</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.CaGalt_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Belchin Kostov <a href="mailto:badriyan@clinic.ub.es">badriyan@clinic.ub.es</a>, Monica Becue-Bertaut, Francois Husson</p>


<h3>See Also</h3>

 <p><code><a href="#topic+CaGalt">CaGalt</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(health)
res.cagalt&lt;-CaGalt(Y=health[,1:115],X=health[,116:118],type="n")
summary(res.cagalt)

## End(Not run)
</code></pre>

<hr>
<h2 id='summary.FAMD'>Printing summeries of FAMD objects</h2><span id='topic+summary.FAMD'></span>

<h3>Description</h3>

<p>Printing summaries of factor analysis on mixed data objects</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'FAMD'
summary(object, nb.dec = 3, nbelements=10,
    nbind=nbelements, ncp = 3, align.names=TRUE , file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.FAMD_+3A_object">object</code></td>
<td>
<p>an object of class FAMD</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_nbelements">nbelements</code></td>
<td>
<p>number of elements written (variables, categories, ...);
use <code>nbelements = Inf</code> if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_nbind">nbind</code></td>
<td>
<p>number of individuals written (individuals and supplementary individuals, ...);
use <code>nbind = Inf</code> to have the results for all the individuals and <code>nbind = 0</code> 
if you do not want the results for individuals</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions printed</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.FAMD_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+FAMD">FAMD</a></code></p>

<hr>
<h2 id='summary.MCA'>Printing summeries of MCA objects</h2><span id='topic+summary.MCA'></span>

<h3>Description</h3>

<p>Printing summaries of multiple correspondence analysis objects</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MCA'
summary(object, nb.dec = 3, nbelements=10,
    nbind=nbelements, ncp = 3, align.names=TRUE, file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.MCA_+3A_object">object</code></td>
<td>
<p>an object of class MCA</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_nbelements">nbelements</code></td>
<td>
<p>number of elements written (variables, categories, ...);
use <code>nbelements = Inf</code> if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_nbind">nbind</code></td>
<td>
<p>number of individuals written (individuals and supplementary individuals, ...);
use <code>nbind = Inf</code> to have the results for all the individuals and <code>nbind = 0</code> 
if you do not want the results for individuals</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions printed</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.MCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MCA">MCA</a></code></p>

<hr>
<h2 id='summary.MFA'>Printing summaries of MFA objects</h2><span id='topic+summary.MFA'></span>

<h3>Description</h3>

<p>Printing summaries of multiple factor analysis objects</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MFA'
summary(object, nb.dec = 3, nbelements=10,
    nbind = nbelements, ncp = 3, align.names=TRUE, file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.MFA_+3A_object">object</code></td>
<td>
<p>an object of class MFA</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_nbelements">nbelements</code></td>
<td>
<p>number of elements written (groups, variables, categories, ...);
use <code>nbelements = Inf</code> if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_nbind">nbind</code></td>
<td>
<p>number of individuals written (individuals and supplementary individuals, ...);
use <code>nbind = Inf</code> to have the results for all the individuals and <code>nbind = 0</code> 
if you do not want the results for individuals</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions printed</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.MFA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+MFA">MFA</a></code></p>

<hr>
<h2 id='summary.PCA'>Printing summeries of PCA objects</h2><span id='topic+summary.PCA'></span>

<h3>Description</h3>

<p>Printing summaries of principal component analysis objects</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
summary(object, nb.dec = 3, nbelements=10,
   nbind = nbelements, ncp = 3, align.names=TRUE, file="", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.PCA_+3A_object">object</code></td>
<td>
<p>an object of class PCA</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_nbelements">nbelements</code></td>
<td>
<p>number of elements written (variables, categories, ...);
use <code>nbelements = Inf</code> if you want to have all the elements</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_nbind">nbind</code></td>
<td>
<p>number of individuals written (individuals and supplementary individuals, ...);
use <code>nbind = Inf</code> to have the results for all the individuals and <code>nbind = 0</code> 
if you do not want the results for individuals</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_ncp">ncp</code></td>
<td>
<p>number of dimensions printed</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_align.names">align.names</code></td>
<td>
<p>boolean, if TRUE the names of the objects are written using the same number of characters</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_file">file</code></td>
<td>
<p>a connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="summary.PCA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods, such as cex, cex.main, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+PCA">PCA</a></code></p>

<hr>
<h2 id='svd.triplet'>Singular Value Decomposition of a Matrix</h2><span id='topic+svd.triplet'></span>

<h3>Description</h3>

<p>Compute the singular-value decomposition of a rectangular matrix with weights for rows and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>svd.triplet(X, row.w=NULL, col.w=NULL, ncp=Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="svd.triplet_+3A_x">X</code></td>
<td>
<p>a data matrix</p>
</td></tr>
<tr><td><code id="svd.triplet_+3A_row.w">row.w</code></td>
<td>
<p>vector with the weights of each row (NULL by default and the weights are uniform)</p>
</td></tr>
<tr><td><code id="svd.triplet_+3A_col.w">col.w</code></td>
<td>
<p>vector with the weights of each column (NULL by default and the weights are uniform)</p>
</td></tr>
<tr><td><code id="svd.triplet_+3A_ncp">ncp</code></td>
<td>
<p>the number of components kept for the outputs</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>vs</code></td>
<td>
<p> a vector containing the singular values of 'x';</p>
</td></tr>
<tr><td><code>u</code></td>
<td>
<p>a matrix whose columns contain the left singular vectors of 'x';</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>a matrix whose columns contain the right singular vectors of 'x'.</p>
</td></tr>
</table>


<h3>See Also</h3>

 <p><code><a href="base.html#topic+svd">svd</a></code></p>

<hr>
<h2 id='tab.disjonctif'>Make a disjonctif table</h2><span id='topic+tab.disjonctif'></span>

<h3>Description</h3>

<p>Make a disjonctif table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tab.disjonctif(tab)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tab.disjonctif_+3A_tab">tab</code></td>
<td>
<p>a data frame with factors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The disjonctif table</p>

<hr>
<h2 id='tab.disjonctif.prop'>Make a disjunctive table when missing values are present</h2><span id='topic+tab.disjonctif.prop'></span>

<h3>Description</h3>

<p>Create a disjunctive table. The missing values are replaced by the proportion of the category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tab.disjonctif.prop(tab,seed=NULL,row.w=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tab.disjonctif.prop_+3A_tab">tab</code></td>
<td>
<p>a data frame with factors</p>
</td></tr>
<tr><td><code id="tab.disjonctif.prop_+3A_row.w">row.w</code></td>
<td>
<p>an optional row weights (by default, a vector of 1 for uniform row weights)</p>
</td></tr>
<tr><td><code id="tab.disjonctif.prop_+3A_seed">seed</code></td>
<td>
<p>a single value, interpreted as an integer for the set.seed function (if seed = NULL, missing values are initially imputed by the mean of each variable)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The disjonctif table.prop</p>

<hr>
<h2 id='tea'>tea (data)</h2><span id='topic+tea'></span>

<h3>Description</h3>

<p>The data used here concern a questionnaire on tea.
We asked to 300 individuals how they drink tea (18 questions), what are their product's perception (12 questions) and some personal details (4 questions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tea)</code></pre>


<h3>Format</h3>

<p>A data frame with 300 rows and 36 columns. Rows represent the individuals,
columns represent the different questions. The first 18 questions are active ones,
the 19th is a supplementary quantitative variable (the age) and the last variables 
are supplementary categorical variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(tea)
res.mca=MCA(tea,quanti.sup=19,quali.sup=20:36)
plot(res.mca,invisible=c("var","quali.sup","quanti.sup"),cex=0.7)
plot(res.mca,invisible=c("ind","quali.sup","quanti.sup"),cex=0.8)
plot(res.mca,invisible=c("quali.sup","quanti.sup"),cex=0.8)
dimdesc(res.mca)
plotellipses(res.mca,keepvar=1:4)

## make a hierarchical clustering: click on the tree to define the number of clusters
## HCPC(res.mca)

## End(Not run)
</code></pre>

<hr>
<h2 id='textual'>Text mining</h2><span id='topic+textual'></span>

<h3>Description</h3>

<p>Calculates the number of occurence of each words and a contingence table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>textual(tab, num.text, contingence.by=1:ncol(tab), 
    maj.in.min = TRUE, sep.word=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="textual_+3A_tab">tab</code></td>
<td>
<p>a data frame with one textual variable</p>
</td></tr>
<tr><td><code id="textual_+3A_num.text">num.text</code></td>
<td>
<p>indice of the textual variable</p>
</td></tr>
<tr><td><code id="textual_+3A_contingence.by">contingence.by</code></td>
<td>
<p>a list with the indices of the variables for which a contingence 
table is calculated by default a contingence table is calculated for all the variables 
(except the textual one). A contingence table can also be calculated for couple of variables.
If <code>contingence.by</code> is equal to num.text, then the contingence table is calculated 
for each row of the data table</p>
</td></tr>
<tr><td><code id="textual_+3A_maj.in.min">maj.in.min</code></td>
<td>
<p>boolean, if TRUE majuscule are transformed in minuscule</p>
</td></tr>
<tr><td><code id="textual_+3A_sep.word">sep.word</code></td>
<td>
<p>a string with all the characters which correspond to separator of words</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list including:
</p>
<table>
<tr><td><code>cont.table</code></td>
<td>
<p>the contingence table with in rows the categories of the categorical variables 
(or the couple of categories), and in column the words, 
and in each cell the number of occurence</p>
</td></tr>
<tr><td><code>nb.words</code></td>
<td>
<p>a data.frame with all the words and for each word, the number of lists in which 
it is present, and the number of occurence</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+CA">CA</a></code>, <code><a href="#topic+descfreq">descfreq</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(poison.text)
res.text &lt;- textual(poison.text, num.text = 3, contingence.by = 1)
descfreq(res.text$cont.table)
## Contingence table for the couple of variable sick-sex
res.text2 &lt;- textual(poison.text, num.text = 3, contingence.by = list(c(1,2)))
descfreq(res.text2$cont.table)
## Contingence table for sex, sick and the couple of variable sick-sex
res.text2 &lt;- textual(poison.text, num.text = 3, contingence.by = list(1,2,c(1,2)))
</code></pre>

<hr>
<h2 id='wine'>Wine</h2><span id='topic+wine'></span>

<h3>Description</h3>

<p>The data used here refer to 21 wines of Val de Loire.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(wine)</code></pre>


<h3>Format</h3>

<p>A data frame with 21 rows (the number of wines) and 31 columns: the first column corresponds to the label of origin,
the second column corresponds to the soil, and the others correspond to sensory
descriptors.
</p>


<h3>Source</h3>

<p>Centre de recherche INRA d'Angers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)

## Example of PCA
res.pca = PCA(wine,ncp=5, quali.sup = 1:2)

## Not run: 
## Example of MCA
res.mca = MCA(wine,ncp=5, quanti.sup = 3:ncol(wine))

## Example of MFA
res.mfa = MFA(wine,group=c(2,5,3,10,9,2),type=c("n",rep("s",5)),ncp=5,
    name.group=c("orig","olf","vis","olfag","gust","ens"),
    num.group.sup=c(1,6),graph=FALSE)
plotellipses(res.mfa)
plotellipses(res.mfa,keepvar="Label") ## for 1 variable

## End(Not run)
</code></pre>

<hr>
<h2 id='write.infile'>Print in a file</h2><span id='topic+write.infile'></span>

<h3>Description</h3>

<p>Print in a file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.infile(X, file, sep=";", append = FALSE, nb.dec=4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.infile_+3A_x">X</code></td>
<td>
<p>an object of class list, data.frame, matrix, ...</p>
</td></tr>
<tr><td><code id="write.infile_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to print to</p>
</td></tr>
<tr><td><code id="write.infile_+3A_sep">sep</code></td>
<td>
<p>character string to insert between the objects to print 
(if the argument file is not NULL)</p>
</td></tr>
<tr><td><code id="write.infile_+3A_append">append</code></td>
<td>
<p>logical. If TRUE output will be
appended to file; otherwise, it will overwrite the contents
of file.</p>
</td></tr>
<tr><td><code id="write.infile_+3A_nb.dec">nb.dec</code></td>
<td>
<p>number of decimal printed, by default 4</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Francois Husson <a href="mailto:francois.husson@institut-agro.fr">francois.husson@institut-agro.fr</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(decathlon)
res.pca &lt;- PCA(decathlon, quanti.sup = 11:12, quali.sup = 13)
write.infile(res.pca, file="c:/essai.csv", sep = ";")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
