<!DOCTYPE html><html lang="en"><head><title>Help for package falconx</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {falconx}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biasMatrix'><p>Bias Matrix</p></a></li>
<li><a href='#falconx'><p>Finding Allele-Specific Copy Number in Whole-Exome Sequencing Data</p></a></li>
<li><a href='#getASCN.x'><p>Getting Allele-specific DNA Copy Number</p></a></li>
<li><a href='#getChangepoints.x'><p>Getting Change-points</p></a></li>
<li><a href='#pos'><p>Position (bp)</p></a></li>
<li><a href='#readMatrix'><p>Reads Matrix</p></a></li>
<li><a href='#tauhat'><p>Estimated Break Points</p></a></li>
<li><a href='#view'><p>Viewing Data with Allele-specific Copy Number</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-02-16</td>
</tr>
<tr>
<td>Title:</td>
<td>Finding Allele-Specific Copy Number in Whole-Exome Sequencing
Data</td>
</tr>
<tr>
<td>Author:</td>
<td>Hao Chen and Nancy R. Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hao Chen &lt;hxchen@ucdavis.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a method for Allele-specific DNA Copy Number profiling for whole-Exome sequencing data.  Given the allele-specific coverage and site biases at the variant loci, this program segments the genome into regions of homogeneous allele-specific copy number.  It requires, as input, the read counts for each variant allele in a pair of case and control samples, as well as the site biases. For detection of somatic mutations, the case and control samples can be the tumor and normal sample from the same individual.  The implemented method is based on the paper: Chen, H., Jiang, Y., Maxwell, K., Nathanson, K. and Zhang, N. (under review). Allele-specific copy number estimation by whole Exome sequencing.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-02-24 17:15:18 UTC; haochen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-02-24 23:43:44</td>
</tr>
</table>
<hr>
<h2 id='biasMatrix'>Bias Matrix</h2><span id='topic+biasMatrix'></span>

<h3>Description</h3>

<p>This is a data frame with two columns and the column names are &quot;sN&quot;, &quot;sT&quot;.  They are the site-specific bias in total coverage for normal (control) sample and tumor (case) sample, respectively.
</p>

<hr>
<h2 id='falconx'>Finding Allele-Specific Copy Number in Whole-Exome Sequencing Data</h2><span id='topic+falconx'></span>

<h3>Description</h3>

<p>This library contains a set of tools for allele-specific DNA copy number profiling using whole exome sequencing.  Given the allele-specific coverage and site biases at the variant loci, this program segments the genome into regions of homogeneous allele-specific copy number.  It requires, as input, the read counts for each variant allele in a pair of case and control samples, as well as the site biases. For detection of somatic mutations, the case and control samples can be the tumor and normal sample from the same individual.
</p>


<h3>Author(s)</h3>

<p>Hao Chen and Nancy R. Zhang
</p>
<p>Maintainer: Hao Chen (hxchen@ucdavis.edu)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getChangepoints.x">getChangepoints.x</a></code>, <code><a href="#topic+getASCN.x">getASCN.x</a></code>, <code><a href="#topic+view">view</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Example) 
 # tauhat = getChangepoints.x(readMatrix, biasMatrix)  # uncomment this to run the function. 
cn = getASCN.x(readMatrix, biasMatrix, tauhat=tauhat)
 # cn$tauhat would give the indices of change-points.  
 # cn$ascn would give the estimated allele-specific copy numbers for each segment.
 # cn$Haplotype[[i]] would give the estimated haplotype for the major chromosome in segment i 
 # if this segment has different copy numbers on the two homologous chromosomes.
view(cn)
</code></pre>

<hr>
<h2 id='getASCN.x'>Getting Allele-specific DNA Copy Number</h2><span id='topic+getASCN.x'></span>

<h3>Description</h3>

<p>Given a set of breakpoints where parent-specific copy number changes, this function estimates the parent-specific copy number for each segment, and the haplotype for the major chromosome on segments where the two homologous chromosomes have different copy numbers.  You are recommended to specify the parameter &quot;rdep&quot;, the case-control genome-wide average coverage ratio.  Usually, a good estimate of rdep is (total mapped reads in tumor)/(total mapped reads in normal).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getASCN.x(readMatrix, biasMatrix, tauhat=NULL, threshold=0.15, COri=c(0.95,1.05), 
error=1e-5, maxIter=1000, independence=TRUE, pos=NULL, readlength=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getASCN.x_+3A_readmatrix">readMatrix</code></td>
<td>
<p>A data frame with four columns and the column names are &quot;AN&quot;, &quot;BN&quot;, &quot;AT&quot;
and &quot;BT&quot;. They are A-allele coverage in the tumor (case) sample, B-allele coverage
in the tumor (case) sample, A-allele coverage in the normal (control) sample,
and B-allele coverage in the normal (control) sample, respectively.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_biasmatrix">biasMatrix</code></td>
<td>
<p>A data frame with two columns and the column names are &quot;sN&quot;, &quot;sT&quot;.  They are the site-specific bias in total coverage for normal (control) sample and tumor (case) sample, respectively.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_tauhat">tauhat</code></td>
<td>
<p>The estimated break points.  If it is not specified (NULL), then this function will first estimate the break points by calling the function &quot;getChangepoints.x&quot;, and then estimate the parent-specific DNA copy number for each segment.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_threshold">threshold</code></td>
<td>
<p>The estimated copy number are set to be 1 if it differs from 1 by less than this threshold.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_cori">COri</code>, <code id="getASCN.x_+3A_error">error</code>, <code id="getASCN.x_+3A_maxiter">maxIter</code></td>
<td>
<p>Parameters used in estimating the success probabilities of the mixed binomial distribution.  See the manuscript by Chen and Zhang for more details.  &quot;pOri&quot; provides the initial success probabilities.  The two values in pOri needs to be different.  &quot;error&quot; provides the stopping criterion.  &quot;maxIter&quot; is the maximum iterating steps if the stopping criterion is not achieved.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_independence">independence</code></td>
<td>
<p>The model assumes reads are conditionally independent.  If this argument is FALSE, the pruning approach will be performed.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_pos">pos</code></td>
<td>
<p>The locations (in base pair) of the heterozygous sites. This information is needed when &quot;independence=FALSE&quot;.</p>
</td></tr>
<tr><td><code id="getASCN.x_+3A_readlength">readlength</code></td>
<td>
<p>The length of read if the data is from single-end sequencing, and the maximum span of read pairs if the data if from paired-end sequencing.  This information is needed when &quot;independence=FALSE&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>tauhat</code></td>
<td>
<p>A vector holding the estimated break points in terms of the index in the coverage vectors.</p>
</td></tr>
<tr><td><code>ascn</code></td>
<td>
<p>The estimated parent-specific copy numbers in the segments between the break points in tauhat.</p>
</td></tr>
<tr><td><code>Haplotype</code></td>
<td>
<p>The estimated haplotype for the major chromosome (the chromosome has a higher copy number compared to its homologous chromosome) on segments where the two homologous chromosomes have different copy numbers.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+getChangepoints.x">getChangepoints.x</a></code>, <code><a href="#topic+view">view</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Example) 
cn = getASCN.x(readMatrix, biasMatrix, tauhat=tauhat)
 # cn$tauhat would give the indices of change-points.  
 # cn$ascn would give the estimated allele-specific copy numbers for each segment.
 # cn$Haplotype[[i]] would give the estimated haplotype for the major chromosome in segment i 
 # if this segment has different copy numbers on the two homologous chromosomes.
</code></pre>

<hr>
<h2 id='getChangepoints.x'>Getting Change-points</h2><span id='topic+getChangepoints.x'></span>

<h3>Description</h3>

<p>This function estimates the change-points where one or both parent-specific copy numbers change.  It uses a circular binary segmentation approach to find change-points in a binomial mixture process.  The output of the function is the set of locations of the break points.  If the whole genome is analyzed, it is recommended to run this function chromosome by chromosome, and the runs on different chromosomes can be done in parallel to shorten the running time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getChangepoints.x(readMatrix, biasMatrix, verbose=TRUE, COri=c(0.95,1.05), error=1e-5, 
maxIter=1000, independence=TRUE, pos=NULL, readlength=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getChangepoints.x_+3A_readmatrix">readMatrix</code></td>
<td>
<p>A data frame with four columns and the column names are &quot;AN&quot;, &quot;BN&quot;, &quot;AT&quot;
and &quot;BT&quot;. They are A-allele coverage in the tumor (case) sample, B-allele coverage
in the tumor (case) sample, A-allele coverage in the normal (control) sample,
and B-allele coverage in the normal (control) sample, respectively.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_biasmatrix">biasMatrix</code></td>
<td>
<p>A data frame with two columns and the column names are &quot;sN&quot;, &quot;sT&quot;.  They are the site-specific bias in total coverage for normal (control) sample and tumor (case) sample, respectively.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_verbose">verbose</code></td>
<td>
<p>Provide progress messages if it is TRUE.  This argument is TRUE by default.  Set it to be FALSE if you want to turn off the progress messages.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_cori">COri</code>, <code id="getChangepoints.x_+3A_error">error</code>, <code id="getChangepoints.x_+3A_maxiter">maxIter</code></td>
<td>
<p>Parameters used in estimating the success probabilities of the mixed binomial distribution.  See the manuscript by Chen and Zhang for more details.  &quot;pOri&quot; provides the initial success probabilities.  The two values in pOri needs to be different.  &quot;error&quot; provides the stopping criterion.  &quot;maxIter&quot; is the maximum iterating steps if the stopping criterion is not achieved.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_independence">independence</code></td>
<td>
<p>The model assumes reads are conditionally independent.  If this argument is FALSE, the pruning approach will be performed.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_pos">pos</code></td>
<td>
<p>The locations (in base pair) of the heterozygous sites. This information is needed when &quot;independence=FALSE&quot;.</p>
</td></tr>
<tr><td><code id="getChangepoints.x_+3A_readlength">readlength</code></td>
<td>
<p>The length of read if the data is from single-end sequencing, and the maximum span of read pairs if the data if from paired-end sequencing.  This information is needed when &quot;independence=FALSE&quot;.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+getASCN.x">getASCN.x</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Example) 
 # tauhat = getChangepoints.x(readMatrix, biasMatrix)  # uncomment this to run the function. 
</code></pre>

<hr>
<h2 id='pos'>Position (bp)</h2><span id='topic+pos'></span>

<h3>Description</h3>

<p>This is a vector containing position (in base pair) of each heterozygous site for the read count data in &quot;Example.rda&quot;.  
</p>

<hr>
<h2 id='readMatrix'>Reads Matrix</h2><span id='topic+readMatrix'></span>

<h3>Description</h3>

<p>This is a data frame with four columns and the column names are &quot;AN&quot;, &quot;BN&quot;, &quot;AT&quot;
and &quot;BT&quot;. They are A-allele coverage in the tumor (case) sample, B-allele coverage
in the tumor (case) sample, A-allele coverage in the normal (control) sample,
and B-allele coverage in the normal (control) sample, respectively.</p>

<hr>
<h2 id='tauhat'>Estimated Break Points</h2><span id='topic+tauhat'></span>

<h3>Description</h3>

<p>This is a vector containing the estimated break points that one would get by calling the function getChangepoints.x(readMatrix, biasMatrix) with &quot;readMatrix&quot; and &quot;biasMatrix&quot; from &quot;Example.rda&quot;.
</p>

<hr>
<h2 id='view'>Viewing Data with Allele-specific Copy Number</h2><span id='topic+view'></span>

<h3>Description</h3>

<p>This function generates three plots:  The first plots the A-allele frequencies of the case (black) sample overlayed onto those of the control (gray) sample; the second plots the relative depth of the case over control adjusted by the ratio of total mapped 
reads, i.e. P*(read count in tumor)/(read count in normal), where P=(total reads mapped in normal)/(total reads mapped in tumor); the third plots the estimated 
parent-specific DNA copy numbers. </p>


<h3>Usage</h3>

<pre><code class='language-R'> view(output, pos=NULL, rdep=NULL, plot="all", independence=TRUE, ...) </code></pre>


<h3>Arguments</h3>

 
<table role = "presentation">
<tr><td><code id="view_+3A_output">output</code></td>
<td>
<p>The output from calling function &quot;getASCN.x&quot;.</p>
</td></tr>
<tr><td><code id="view_+3A_pos">pos</code></td>
<td>
<p>A vector of the base positions for the SNPs.  If this information is not provided, the x-axis of the plots will simply be the SNP ordering.  If this information is provided, the x-axis of the plots will be the position information.</p>
</td></tr> 
<tr><td><code id="view_+3A_rdep">rdep</code></td>
<td>
<p>The relative depth of the case sample over the control sample.  If it is not specified (NULL), then the value median(AT+BT)/median(AN+BN) will be used.</p>
</td></tr>
<tr><td><code id="view_+3A_plot">plot</code></td>
<td>
<p>This argument determines what to plot.  By default, this function gives all three plots described above (&quot;all&quot;).  You can also plot each one individually if you set this argument to either of &quot;Afreq&quot;, &quot;RelativeCoverage&quot; or &quot;ASCN&quot;.</p>
</td></tr>
<tr><td><code id="view_+3A_independence">independence</code></td>
<td>
<p>If argument &quot;pos&quot; is specified, when &quot;independence=FALSE&quot;, the pruned positions will be used.</p>
</td></tr>
<tr><td><code id="view_+3A_...">...</code></td>
<td>
<p>Arguments from plot can be passed along.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+getASCN.x">getASCN.x</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Example) 
cn = getASCN.x(readMatrix, biasMatrix, tauhat=tauhat)
view(cn)

# to view with position as the x-axis 
view(cn, pos=pos)

# to view the plot for only showing A-allele frequency of the case (black) sample overlayed 
# onto those of the control (gray) sample
par(mfrow=c(1,1))
view(cn, plot="Afreq")

# to view the relative depth of the case over control adjusted by the ratio of total mapped 
# reads in fixed size bins
par(mfrow=c(1,1))
view(cn, plot="RelativeCoverage")

# to view the estimated allele-specific DNA copy numbers
par(mfrow=c(1,1))
view(cn, plot="ASCN")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
