<!DOCTYPE html><html><head><title>Help for package FamEvent</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FamEvent}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FamEvent-package'>
<p>Family age-at-onset data simulation and penetrance estimation</p></a></li>
<li><a href='#carrierprob'>
<p>Compute mutation carrier probabilities for individuals with missing gentoypes</p></a></li>
<li><a href='#fampower'>
<p>Simulation-based power calculation for genetic effect</p></a></li>
<li><a href='#LSfam'><p>Ontario Lynch Syndrom families</p></a></li>
<li><a href='#penetrance'>
<p>Penetrance function and confidence intervals</p></a></li>
<li><a href='#penetrance_cmp'>
<p>Estimating Penetrances for competing risk models and confidence intervals</p></a></li>
<li><a href='#penmodel'><p>Fit a penetrance model</p></a></li>
<li><a href='#penmodel_cmp'><p>Fit a penetrance model for competing risks data</p></a></li>
<li><a href='#penmodelEM'><p>EM algorithm for estimating the penetrance model with missing genotypes</p></a></li>
<li><a href='#penplot'>
<p>Plot penetrance functions</p></a></li>
<li><a href='#penplot_cmp'>
<p>Plot penetrance functions from competing risk models</p></a></li>
<li><a href='#plot.penmodel'>
<p>Plot method for <code>penmodel</code></p></a></li>
<li><a href='#plot.penmodel_cmp'>
<p>Plot method for <code>penmodel_cmp</code></p></a></li>
<li><a href='#plot.simfam'><p>Plot method for <code>simfam</code> or Plot pedigrees</p></a></li>
<li><a href='#plot.simfam_cmp'><p>Plot method for <code>simfam_cmp</code> or Plot pedigrees</p></a></li>
<li><a href='#plot.simfam_tvc'><p>Plot method for <code>simfam_tvc</code> or Plot pedigrees</p></a></li>
<li><a href='#plot.simfam2'><p>Plot method for <code>simfam2</code> or Plot pedigrees</p></a></li>
<li><a href='#print.penmodel'>
<p>Print method for <code>penmodel</code>.</p></a></li>
<li><a href='#print.penmodel_cmp'>
<p>Print method for <code>penmodel_cmp</code>.</p></a></li>
<li><a href='#print.summary.penmodel'>
<p>Print method for <code>summary.penmodel</code> of a fitted penetrance model.</p></a></li>
<li><a href='#print.summary.penmodel_cmp'>
<p>Print method for <code>summary.penmodel_cmp</code> of a fitted competing risks penetrance model.</p></a></li>
<li><a href='#simfam'>
<p>Generate familial time-to-event data</p></a></li>
<li><a href='#simfam_cmp'>
<p>Generate familial competing risks data</p></a></li>
<li><a href='#simfam_tvc'>
<p>Generate familial time-to-event data with a time-varying covariate</p></a></li>
<li><a href='#simfam2'>
<p>Generate familial time-to-event data with Kinship or IBD matrices.</p></a></li>
<li><a href='#summary.penmodel'>
<p>Summary method for class <code>penmodel</code></p></a></li>
<li><a href='#summary.penmodel_cmp'>
<p>Summary method for class <code>penmodel_cmp</code></p></a></li>
<li><a href='#summary.simfam'>
<p>Summary method for <code>simfam</code></p></a></li>
<li><a href='#summary.simfam_cmp'>
<p>Summary method for <code>simfam_cmp</code></p></a></li>
<li><a href='#summary.simfam_tvc'>
<p>Summary method for <code>simfam_tvc</code></p></a></li>
<li><a href='#summary.simfam2'>
<p>Summary method for <code>simfam2</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Family Age-at-Onset Data Simulation and Penetrance Estimation</td>
</tr>
<tr>
<td>Version:</td>
<td>3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-07-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Yun-Hee Choi, Karen Kopciuk, Wenqing He, Laurent Briollais</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yun-Hee Choi &lt;yun-hee.choi@schulich.uwo.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulates age-at-onset traits associated with a segregating major gene in family data 
     obtained from population-based, clinic-based, or multi-stage designs. Appropriate ascertainment 
     correction is utilized to estimate age-dependent penetrance functions either parametrically from 
     the fitted model or nonparametrically from the data. The Expectation and Maximization algorithm 
     can infer missing genotypes and carrier probabilities estimated from family's genotype and
     phenotype information or from a fitted model. Plot functions include pedigrees of simulated 
     families and predicted penetrance curves based on specified parameter values.
     For more information see Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An 
     R Package for Generating and Modeling Time-to-Event Data in Family Designs, 
     Journal of Statistical Software 97 (7), 1-30.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2.0)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), survival</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, stats, kinship2, truncnorm, eha, pracma, cmprsk,
matrixcalc</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-02 18:49:20 UTC; yun-heechoi</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-02 19:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='FamEvent-package'>
Family age-at-onset data simulation and penetrance estimation
</h2><span id='topic+FamEvent-package'></span>

<h3>Description</h3>

<p>Family-based studies are used to characterize the disease risk associated
with being a carrier of a major gene. When the disease risk can vary with 
age of onset, penetrance or disease risk functions need to provide age-dependent
estimates of this disease risk over lifetime.
This FamEvent package can generate age-at-onset data in the context of familial studies,
with correction for ascertainment (selection) bias arising from a specified
study design based on proband's mutation and disease statuses.
Possible study designs are: <code>"pop"</code> for population-based design where
families are ascertained through affected probands, <code>"pop+"</code> are similar to
<code>"pop"</code> but probands are also known mutation carriers, <code>"cli"</code> for
clinic-based design that includes affected probands with at least one
parent and one sib affected, <code>"cli+"</code> are similar to <code>"cli"</code> but probands are
also known mutation carrriers. And <code>"twostage"</code> for two-stage design that randomly samples families from
the population in the first stage and oversamples high risk families that includes at least two
affected members in the family at the second stage.
</p>
<p>Ages at disease onset are generated specific to family members' gender and
mutation status according to the specified study design with residual
familial correlations induced by a shared frailty, correlated frailties with Kinship matrix, or a second gene.
</p>
<p>For estimating age at onset risks with family data, an ascertainment
corrected prospective likelihood approach is used to account for the
population or clinic-based study designs while a composite likelihood
approach is used for the two-stage sampling design. The Expectation and
Maximization (EM) algorithm has been implemented for inferring missing
genotypes conditional on observed genotypes and phenotypes in the families. 
For family members who have missing genotypes, their carrier 
probabilities are obtained either from the fitted model or from Mendelian 
transmission probabilities. 
</p>
<p>This package also
provides functions to plot the age-dependent penetrance curves estimated
parametrically from the fitted model or non-parametrically from the data,
pedigree plots of simulated families and penetrance function curves for
carriers and non-carriers of a major and second gene based on specified
parameter values.
</p>
<p><b>In summary, this package facilitates the following:</b>
</p>
<p>1. Family data generations from 1) a shared frailty model or 2) a correlated frailty model with Kinship and/or IBD matrix; 3) a competing risk model.
</p>
<p>2. Covariates considered: sex, gene, options for additional covariates.
</p>
<p>3. Time-varying covariate generation based on permanent exposure (PE) or the Cox-Oaks model (CO).
</p>
<p>3. Model estimation for shared frailty models and competing risk models. 
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi, Karen Kopciuk, Laurent Briollais, Wenqing He
</p>
<p>Maintainer: Yun-Hee Choi &lt; yun-hee.choi@schulich.uwo.ca &gt;
</p>


<h3>References</h3>

<p>Choi, Y.-H., Jung, H., Buys, S., Daly, M., John, E.M., Hopper, J., Andrulis, I., Terry, M.B., Briollais, L. (2021) A Competing Risks Model with Binary Time Varying Covariates for Estimation of Breast Cancer Risks in BRCA1 Families, Statistical Methods in Medical Research 30 (9), 2165-2183. https://doi.org/10.1177/09622802211008945.
</p>
<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a></code>, <code><a href="#topic+summary.simfam">summary.simfam</a></code>, <code><a href="#topic+plot.simfam">plot.simfam</a></code>, 
<code><a href="#topic+simfam2">simfam2</a></code>, <code><a href="#topic+summary.simfam2">summary.simfam2</a></code>, <code><a href="#topic+plot.simfam2">plot.simfam2</a></code>,
<code><a href="#topic+simfam_tvc">simfam_tvc</a></code>, <code><a href="#topic+summary.simfam_tvc">summary.simfam_tvc</a></code>, 
<code><a href="#topic+plot.simfam_tvc">plot.simfam_tvc</a></code>, 
<code><a href="#topic+penplot">penplot</a></code>, <code><a href="#topic+carrierprob">carrierprob</a></code>, <code><a href="#topic+penmodel">penmodel</a></code>, <code><a href="#topic+penmodelEM">penmodelEM</a></code>, 
<code><a href="#topic+print.penmodel">print.penmodel</a></code>, <code><a href="#topic+summary.penmodel">summary.penmodel</a></code>,<code><a href="#topic+print.summary.penmodel">print.summary.penmodel</a></code>, <code><a href="#topic+plot.penmodel">plot.penmodel</a></code>,
<code><a href="#topic+simfam_cmp">simfam_cmp</a></code>, <code><a href="#topic+summary.simfam_cmp">summary.simfam_cmp</a></code>, <code><a href="#topic+plot.simfam_cmp">plot.simfam_cmp</a></code>, <code><a href="#topic+penplot_cmp">penplot_cmp</a></code>, <code><a href="#topic+penmodel_cmp">penmodel_cmp</a></code>,  
<code><a href="#topic+print.penmodel_cmp">print.penmodel_cmp</a></code>, <code><a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a></code>, 
<code><a href="#topic+print.summary.penmodel_cmp">print.summary.penmodel_cmp</a></code>, <code><a href="#topic+plot.penmodel_cmp">plot.penmodel_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Example1:  Simulate family data 
set.seed(4321)
fam &lt;- simfam(N.fam = 100, design = "pop+", variation = "none", base.dist = "Weibull", 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), allelefreq = 0.02)

# summary of simulated family data
summary(fam) 

# Pedigree plots for family 1 and 2
plot(fam, famid = c(1,2))

# penetrance function plots given model parameter values for Weibull baseline 
penplot(base.parms = c(0.01, 3), vbeta = c(-1.3, 2.35), base.dist = "Weibull", 
        variation = "none", agemin = 20)

# model fit of family data 
fit &lt;- penmodel(Surv(time, status) ~ gender + mgene, cluster = "famID", design = "pop+",
       parms=c(0.01, 3, -1.13, 2.35), data = fam, base.dist = "Weibull", robust = TRUE)

# summary of estimated model parameters and penetrance estimates 
summary(fit)

# penetrance curves useful for model checking 
plot(fit) 

## Example 2: Simulate family data from a correlated frailty model 
#  with Kinship and IBD matrices given pedigree data.

# Inputdata and IBD matrix should be provided; 
# Inputdata was generated as an example using simfam.

data &lt;- simfam(N.fam = 10, design = "noasc", variation = "none",
         base.dist = "Weibull", base.parms = c(0.016, 3), vbeta = c(1, 1))

IBDmatrix &lt;- diag(1, dim(data)[1])
data &lt;- data[ , c(1:7, 11, 14)]

fam2 &lt;- simfam2(inputdata = data, IBD = IBDmatrix, design = "pop", 
        variation = c("kinship","IBD"), depend = c(1, 1), 
        base.dist = "Weibull", base.parms = c(0.016, 3),
        var_names = c("gender", "mgene"), vbeta = c(1,1),
        agemin=20) 

summary(fam2)

### Example 3: Simulate correlated competing risks family data 
set.seed(4321)
fam3 &lt;- simfam_cmp(N.fam = 200, design = "pop+", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "cgamma", depend=c(1, 2, 0.5), 
       allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
       vbeta = list(c(-1.13, 2.35), c(-1, 2)))

# summary of simulated family data
summary(fam3) 

# Pedigree plots for family 1 
plot(fam3, famid = 1)

# penetrance function plot for event 1 given model parameter values for Weibull baseline 
penplot_cmp(event = 1, base.parms = list(c(0.01, 3), c(0.01, 3)), 
          vbeta = list(c(-1.3, 2.35), c(-1, 2)), base.dist = "Weibull", 
          variation = "frailty", frailty.dist = "cgamma", 
          depend=c(1,2,0.5), agemin = 20)


# Fitting shared correlated gamma frailty Penetrance model for simulated competing risk data

fit3 &lt;- penmodel_cmp(
        formula1 = Surv(time, status==1) ~ gender + mgene, 
        formula2 = Surv(time, status==2) ~ gender + mgene,
        cluster = "famID", gvar = "mgene", design = "pop+",  
        parms = list(c(0.01, 3, -1, 2), c(0.01, 3, -1, 2), c(0.5, 1, 0.5)),
        base.dist = "Weibull", frailty.dist = "cgamma", data = fam2, robust = TRUE)

# Summary of the model parameter estimates from the model fit
summary(fit3)

# Plot the lifetime penetrance curves with 95
# gender and mutation status groups along with their nonparametric penetrance curves  
# based on data excluding probands. 

plot(fit3, add.CIF = TRUE, conf.int = TRUE, MC = 100)

### Example 4: Simulate family data with a time-varying covariate 
set.seed(4321)

fam4 &lt;- simfam_tvc(N.fam = 10, design = "pop", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "gamma", depend = 1, 
       add.tvc = TRUE, tvc.type = "CO", tvc.range = c(30,60), 
       tvc.parms = c(1, 0.1, 0), allelefreq = 0.02, 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))

summary(fam4)


## End(Not run)


</code></pre>

<hr>
<h2 id='carrierprob'>
Compute mutation carrier probabilities for individuals with missing gentoypes
</h2><span id='topic+carrierprob'></span>

<h3>Description</h3>

<p>Computes model- or data-based carrier probabilities for individuals with missing genotypes based on the observed mutation status of family members and the individual's phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carrierprob(condition = "geno", method = "data", fit = NULL, data, mode = "dominant", 
q = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carrierprob_+3A_condition">condition</code></td>
<td>

<p>Choice of conditional information to use for computing the carrier probability. Possible choices are <code>"geno"</code> for using observed genotypes and <code>"geno+pheno"</code> for using both observed genotype and phenotype information in the calculation of the carrier probability.
</p>
</td></tr>
<tr><td><code id="carrierprob_+3A_method">method</code></td>
<td>

<p>Choice of methods to calculate the carrier probability. Possible choices are <code>"data"</code> for empirical calculation of the carrier probabilities based on data, 
<code>"mendelian"</code> using Mendelian transmission probabilities based on observed carriers within families, or <code>"model"</code> using the parametric model fit; see details below. Default is <code>"data"</code>.
If <code>method = "data"</code>, only <code>data</code> is required to be specified.  
</p>
</td></tr>
<tr><td><code id="carrierprob_+3A_fit">fit</code></td>
<td>

<p>An object of class <code>penmodel</code>, a fitted model by <code>penmodelEM</code> function for inferring missing mutation statuses in the family.
</p>
</td></tr>
<tr><td><code id="carrierprob_+3A_data">data</code></td>
<td>

<p>Family data that includes missing genotypes using the same data format generated by the function <code>simfam</code>.
</p>
</td></tr>
<tr><td><code id="carrierprob_+3A_mode">mode</code></td>
<td>

<p>Choice of modes of inheritance when using <code>method="model"</code>. Possible choices are <code>"dominant"</code> for dominant model or <code>"recessive"</code> for recessive model. Default is <code>"dominant"</code>.
</p>
</td></tr>
<tr><td><code id="carrierprob_+3A_q">q</code></td>
<td>

<p>Frequency of the disease causing allele when using <code>method="model"</code>. The value should be between 0 and 1. If <code>NULL</code>, the estimated allele frequency from data will be used. Default value  is <code>0.02</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>method="model"</code> along with the choice of <code>condition="geno+pheno"</code>, the carrier probability for individual <code class="reqn">i</code> is calculated by conditioning on her/his observed phenotype and carrier statuses of family members 
 
P(X<sub>i</sub> = 1 | Y<sub>i</sub> , X<sup>o</sup> )  =
P(Y<sub>i</sub> | X<sub>i</sub> = 1) * P(X<sub>i</sub> = 1 | X<sup>o</sup>) /
(P(Y<sub>i</sub> | X<sub>i</sub> = 1) * P(X<sub>i</sub> = 1 | X<sup>o</sup>) + P(Y<sub>i</sub> | X<sub>i</sub> = 0) * P(X<sub>i</sub> = 0 | X<sup>o</sup>)) 
where X<sub>i</sub> indicates the unknown carrier status of individual <code class="reqn">i</code> and X<sup>o</sup> represents the observed carrier statuses in his or her family members; Y<sub>i</sub> represents the observed phenotype t<sub>i</sub>, &delta;<sub>i</sub> of individual <code class="reqn">i</code> in terms of age at onset t<sub>i</sub> and disease status indicator &delta;<sub>i</sub> with 1 used for affected individuals and 0 for unaffected individuals.
When <code>method="mendelian"</code> along with the choice of <code>condition="geno"</code>, the carrier probability is calculated based on Mendelian laws of genetic transmission with a fixed allele frequency.
</p>


<h3>Value</h3>

<p>Returns a data frame with a vector of carrier probabilities called <code>carrp.geno</code> when <code>condition="geno"</code> or <code>carrp.pheno</code> when <code>condtion="geno+pheno"</code> added after the last column of the family data.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a>, <a href="#topic+penmodelEM">penmodelEM</a>, <a href="#topic+plot.simfam">plot.simfam</a>, <a href="#topic+summary.simfam">summary.simfam</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulated family data with 30% of members missing their genetic information.

set.seed(4321)
fam &lt;- simfam(N.fam = 100, design = "pop+", base.dist = "Weibull", mrate = 0.3,
       base.parms = c(0.01,3), vbeta = c(-1.13, 2.35), agemin = 20)
 
# EM algorithm for fitting family data with missing genotypes assuming a Weibull
# baseline hazard and dominant mode of Mendelian inheritance for a major gene.

fitEM &lt;- penmodelEM(Surv(time, status) ~ gender + mgene, cluster = "famID", gvar = "mgene", 
        parms = c(0.01, 3, -1.13, 2.35), data = fam, design = "pop+", base.dist = "Weibull", 
        method = "mendelian", mode = "dominant")

# Carrier probability obtained by conditioning on the observed genotypes and phenotype, 
# assuming a dominant Mendelian mode of inheritance

fam.added &lt;- carrierprob(condition = "geno+pheno", method = "model", fit = fitEM, 
             data = fam, mode = "dominant", q = 0.02)

# pedigree plot for family 1 displaying carrier probabilities

plot.simfam(fam.added, famid = 1)

</code></pre>

<hr>
<h2 id='fampower'>
Simulation-based power calculation for genetic effect
</h2><span id='topic+fampower'></span>

<h3>Description</h3>

<p>Computes the power of detecting genetic effect in the penetrance model based on a family-based simulation study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fampower(N.fam, N.sim, effectsize, beta.sex, alpha = 0.05, side = 2, design = "pop", 
variation = "none", interaction = FALSE, depend = NULL, base.dist = "Weibull", 
frailty.dist = NULL, base.parms, allelefreq = c(0.02, 0.2), dominant.m = TRUE, 
dominant.s = TRUE, mrate = 0, hr = 0, probandage = c(45, 2), agemin = 20, agemax = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fampower_+3A_n.fam">N.fam</code></td>
<td>

<p>Number of families to generate.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_n.sim">N.sim</code></td>
<td>

<p>Number of simulations.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_effectsize">effectsize</code></td>
<td>

<p>Effect size of the major mutated gene (<code>beta.gene</code>) to detect under the alternative hypothesis. When <code>interaction=TRUE</code>, both the main and interaction effects should be specified, <code>effectsize = c(beta.gene, beta.int).</code>
</p>
</td></tr>
<tr><td><code id="fampower_+3A_beta.sex">beta.sex</code></td>
<td>

<p>Gender effect that is fixed in the model.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_alpha">alpha</code></td>
<td>

<p>Significance level. Default value is 0.05.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_side">side</code></td>
<td>

<p>Number of sides for the alternative hypothesis. Possible choices are 1 for one-sided test and 2 for two-sided test. Default value is 2.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_design">design</code></td>
<td>

<p>Family based study design used in the simulations. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code> or <code>"twostage"</code>, where <code>"pop"</code> is for the population-based design that families are ascertained by affected probands, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sibling affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands and <code>"twostage"</code> for two-stage design that randomly samples families from the population in the first stage and oversamples high risk families in the second stage that includes at least two affected members in the family. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_interaction">interaction</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the interaction between gender and mutation status is allowed, otherwise no interaction is allowed. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_depend">depend</code></td>
<td>

<p>Variance of the frailty distribution. Dependence within families increases with depend value.  Default value is <code>NULL</code>. Value <code> &gt; 0 </code> should be specified when <code>variation = "frailty"</code>. 
</p>
</td></tr>
<tr><td><code id="fampower_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution.  Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code> <code>"gamma",</code> or <code>"logBurr"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution.  Possible choices are: <code>"gamma"</code> for gamma distribution or <code>"lognormal"</code> for log normal distribution when <code>variation = "frailty"</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_base.parms">base.parms</code></td>
<td>

<p>Vector of parameter values for baseline hazard function. 
</p>
<p><code>base.parms = c(lambda, rho)</code>, where lambda and rho are the shape and scale parameters, respectively. If <code>base.dist = "logBurr"</code> is chosen, three parameters should be specified for <code>base.parms = c(lambda, rho, eta)</code>. 
</p>
</td></tr>
<tr><td><code id="fampower_+3A_allelefreq">allelefreq</code></td>
<td>

<p>Vector of population allele frequencies of major and second disease gene alleles. Frequencies must be between 0 and 1. Default frequencies are 0.02 for major gene allele and 0.2 for second gene allele, <code>allelefreq = c(0.02, 0.2)</code>.</p>
</td></tr>
<tr><td><code id="fampower_+3A_dominant.m">dominant.m</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of the major gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="fampower_+3A_dominant.s">dominant.s</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of the second gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="fampower_+3A_mrate">mrate</code></td>
<td>

<p>Proportion of missing genotypes, value between 0 and 1. Default value is 0.</p>
</td></tr>
<tr><td><code id="fampower_+3A_hr">hr</code></td>
<td>

<p>Proportion of high risk families, which include at least two affected members, to be sampled from the two stage sampling. This value should be specified when <code>design = "twostage"</code>. Default value is 0. Value should lie between 0 and 1.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_probandage">probandage</code></td>
<td>

<p>Vector of mean and standard deviation for the proband age. Default values are mean of 45 years and standard deviation of 2 years, <code>probandage = c(45, 2)</code>.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="fampower_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 100 years of age.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The power of testing <code class="reqn">H_0: \beta_{gene} = 0</code> vs. <code class="reqn">H_1: \beta_{gene} = </code> <code>effectsize</code> is obtained by the proportion of times the null hypothesis is rejected out of the <code>N.sim</code> simulations.
</p>
<p>When <code>interaction = TRUE</code>, the powers of both the main effect of mutated gend and the interaction effect of mutated gene and gender will be computed.
</p>


<h3>Value</h3>

<p>Returns 
</p>
<table>
<tr><td><code>power</code></td>
<td>
<p>Power of detecting the genetic effect.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    
## Example 1: obtain the power for testing the genetic effect 
# based on 50 POP families simulated using 100 simulations
## Not run: 
set.seed(4321)
fampower(N.fam = 50, N.sim = 100, effectsize = 1, beta.sex = 0.8, alpha = 0.05, side = 2, 
design = "pop+", variation = "none", base.dist = "Weibull", allelefreq = 0.02, 
base.parms = c(0.01, 3))
## End(Not run)
    
## Example 2: obtain the power for both the main and interaction effects  
# based on 50 POP families simulated using 100 simulations
## Not run: 
set.seed(4321)
fampower(N.fam = 50, N.sim = 100, effectsize = c(1.5, 1), beta.sex = 0.8, alpha = 0.05, 
side = 2, interaction = TRUE,  design = "pop+", variation = "none", base.dist = "Weibull", 
allelefreq = 0.02, base.parms = c(0.01, 3))
## End(Not run)
</code></pre>

<hr>
<h2 id='LSfam'>Ontario Lynch Syndrom families</h2><span id='topic+LSfam'></span>

<h3>Description</h3>

<p>Data from 32 Lynch Syndrome families segregating mismatch repair mutations selected from the Ontario Familial Colorectal Cancer Registry that includes 765 individuals, both probands and relatives. The families were ascertained throughout affected and mutation carrier probands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("LSfam")</code></pre>


<h3>Format</h3>

<p>A data frame with 765 observations on the following 11 variables.
</p>

<dl>
<dt><code>famID</code></dt><dd><p>Family identification (ID) numbers.</p>
</dd>
<dt><code>indID</code></dt><dd><p>Individuals ID numbers.</p>
</dd>
<dt><code>fatherID</code></dt><dd><p>Father ID numbers.</p>
</dd>
<dt><code>motherID</code></dt><dd><p>Mother ID numbers.</p>
</dd>
<dt><code>gender</code></dt><dd><p>Gender indicators: 1 for male, 0 for female.</p>
</dd>
<dt><code>status</code></dt><dd><p>Disease statuses: 1 for affected, 0 for unaffected.</p>
</dd>
<dt><code>time</code></dt><dd><p>Ages at diagnosis of colorectal cancer for the affected or ages of last follow-up for the unaffected.</p>
</dd>
<dt><code>currentage</code></dt><dd><p>Current ages in years.</p>
</dd>
<dt><code>mgene</code></dt><dd><p>MLH1 or MSH2 mutation indicators: 1 for mutated gene carriers, 0 for mutated gene noncarriers, or <code>NA</code> if missing.</p>
</dd>
<dt><code>proband</code></dt><dd><p>Proband indicators: 1 for proband, 0 for non-proband.</p>
</dd>
<dt><code>relation</code></dt><dd><p>Family members' relationship with the proband.
Relation codes:
</p>

<table>
<tr>
 <td style="text-align: left;">
    1  </td><td style="text-align: left;"> Proband (self) </td>
</tr>
<tr>
 <td style="text-align: left;">
    2  </td><td style="text-align: left;"> Brother or sister </td>
</tr>
<tr>
 <td style="text-align: left;">
    3  </td><td style="text-align: left;"> Son or daughter </td>
</tr>
<tr>
 <td style="text-align: left;">
    4  </td><td style="text-align: left;"> Parent </td>
</tr>
<tr>
 <td style="text-align: left;">
    5  </td><td style="text-align: left;"> Nephew or niece </td>
</tr>
<tr>
 <td style="text-align: left;">
    6  </td><td style="text-align: left;"> Spouse </td>
</tr>
<tr>
 <td style="text-align: left;">
    7  </td><td style="text-align: left;"> Brother or sister in law </td>
</tr>
<tr>
 <td style="text-align: left;">
    8  </td><td style="text-align: left;"> Paternal grandparent </td>
</tr>
<tr>
 <td style="text-align: left;">
    9  </td><td style="text-align: left;"> Paternal uncle or aunt </td>
</tr>
<tr>
 <td style="text-align: left;">
    10  </td><td style="text-align: left;"> Paternal cousin </td>
</tr>
<tr>
 <td style="text-align: left;">
    11  </td><td style="text-align: left;"> Maternal grandparent </td>
</tr>
<tr>
 <td style="text-align: left;">
    12  </td><td style="text-align: left;"> Maternal uncle or aunt </td>
</tr>
<tr>
 <td style="text-align: left;">
    13  </td><td style="text-align: left;"> Maternal cousin  </td>
</tr>
<tr>
 <td style="text-align: left;">
    14  </td><td style="text-align: left;"> Son or daughter in law </td>
</tr>
<tr>
 <td style="text-align: left;">
    15  </td><td style="text-align: left;"> Grandchild </td>
</tr>
<tr>
 <td style="text-align: left;">
    16  </td><td style="text-align: left;"> Uncle's or aunt's spouse. </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

</dd>
</dl>



<h3>References</h3>

<p>Choi, Y.-H., Cotterchio, M., McKeown-Eyssen, G., Neerav, M., Bapat, B., Boyd, K., Gallinger, S.,  McLaughlin, J., Aronson, M., and Briollais, L. (2009). Penetrance of Colorectal Cancer among MLH1/ MSH2 Carriers Participating in the Colorectal Cancer Familial Registry in Ontario, Hereditary Cancer in Clinical Practice, 7:14.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(LSfam)

# Summary of LSfam
summary.simfam(LSfam)

# Pedigree plot for the first family
plot.simfam(LSfam)

# Assign minimum age for fitting penmodel
attr(LSfam, "agemin") &lt;- 18

fit &lt;- penmodelEM(Surv(time, status) ~ gender + mgene, cluster = "famID", 
       parms = c(0.05, 2, 1, 3), data = LSfam[!is.na(LSfam$time) &amp; LSfam$time &gt; 18, ], 
       method = "mendelian", base.dist = "Weibull", design = "pop+", robust = TRUE)

summary(fit)

penetrance(fit, fixed = c(1, 1), age = c(50, 60, 70), CI = TRUE, MC = 100)

</code></pre>

<hr>
<h2 id='penetrance'>
Penetrance function and confidence intervals
</h2><span id='topic+penetrance'></span>

<h3>Description</h3>

<p>Estimates the cumulative disease risks (penetrances) and confidence intervals at given age(s) based on the fitted penetrance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  penetrance(fit, fixed, age, CI = TRUE, MC = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penetrance_+3A_fit">fit</code></td>
<td>

<p>An object class of <code>'penmodel'</code>, a fitted model by <code><a href="#topic+penmodel">penmodel</a></code> or <code><a href="#topic+penmodelEM">penmodelEM</a></code> functions.
</p>
</td></tr>
<tr><td><code id="penetrance_+3A_fixed">fixed</code></td>
<td>

<p>Vector of fixed values of the covariates used for penetrance calculation.
</p>
</td></tr>
<tr><td><code id="penetrance_+3A_age">age</code></td>
<td>

<p>Vector of ages used for penetrance calculation.
</p>
</td></tr>
<tr><td><code id="penetrance_+3A_ci">CI</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the 95% confidence interval will be obtained using a Monte-Carlo method, 
otherwise no confidence interval will be provided. Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="penetrance_+3A_mc">MC</code></td>
<td>

<p>Number of simulated samples used to calculate confidence intervals with a Monte-Carlo method. 
If <code>MC=0</code>, no confidence intervals will be calculated. Default value is 100.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The penetrance function is defined as the probability of developing a disease by age <code class="reqn">t</code> given fixed values of covariates <code class="reqn">x</code>, 
</p>
<p style="text-align: center;"><code class="reqn"> P(T &lt; t | x) = 1 - S(t; x),</code>
</p>

<p>where <code class="reqn">t</code> is greater than the minimum age t<sub>0</sub> and <code class="reqn">S(t; x)</code>  is the survival distribution based on a proportional hazards model with a specified baseline hazard distribution.
</p>
<p>The proportional hazards model is specified as:
h(t|x) = h<sub>0</sub>(t) exp(&beta;*x),
where h<sub>0</sub>(t) is the baseline hazards function, <code class="reqn">x</code> is the vector of covariates and &beta; is the vector of corresponding regression coefficients.
</p>
<p>Calculations of standard errors of the penetrance estimates and 95% confidence intervals (CIs) for the penetrance at a given age are based on Monte-Carlo simulations of the estimated penetrance model. 
</p>
<p>A multivariate normal distribution is assumed for the parameter estimates, and <code>MC = n</code> sets of the parameters are generated from the multivariate normal distribution with the parameter estimates and their variance-covariance matrix.
For each simulated set, a penetrance estimate is calculated at a given age by substituting the simulated parameters into the penetrance function.
</p>
<p>The standard error of the penetrance estimate at a given age is calculated by the standard deviation of penetrance estimates obtained from <code class="reqn">n</code> simulations.
</p>
<p>The 95% CI for the penetrance at a given age is calculated using the 2.5th and 97.5th percentiles of the penetrance estimates obtained from <code class="reqn">n</code> simulations.
</p>


<h3>Value</h3>

<p>Returns the following values:
</p>
<table>
<tr><td><code>age</code></td>
<td>
<p>Ages at which the penetrances are calculated.</p>
</td></tr>
<tr><td><code>penetrance</code></td>
<td>
<p>Penetrance estimates at given ages.</p>
</td></tr>
<tr><td><code>lower</code></td>
<td>
<p>Lower limit of the 95% confidence interval; simulation-based 2.5th percentile of the penetrance estimates.</p>
</td></tr>
<tr><td><code>upper</code></td>
<td>
<p>Upper limit of the 95% confidence interval; simulation-based 97.5th percentile of the penetrance estimates.</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Simulation-based standard errors of the penetrance estimates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a>, <a href="#topic+penmodel">penmodel</a>, <a href="#topic+penmodelEM">penmodelEM</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  set.seed(4321)
  fam &lt;- simfam(N.fam = 100, design = "pop+", base.dist = "Weibull", allelefreq = 0.02, 
         base.parms = c(0.01,3), vbeta = c(-1.13, 2.35))
	
  fit &lt;- penmodel(Surv(time, status) ~ gender +  mgene, cluster = "famID", 
         parms = c(0.01, 3, -1.13, 2.35),  data = fam, base.dist = "Weibull", design = "pop+")
 
 # Compute penetrance estimates for male carriers at age 40, 50, 60, and 70 and
 # their 95% CIs based on 100 Monte Carlo simulations.
 
 penetrance(fit, fixed = c(1,1), age = c(40, 50, 60, 70), CI = TRUE, MC = 100)

</code></pre>

<hr>
<h2 id='penetrance_cmp'>
Estimating Penetrances for competing risk models and confidence intervals
</h2><span id='topic+penetrance_cmp'></span>

<h3>Description</h3>

<p>Estimates the cumulative disease risks (penetrances) and confidence intervals for the event of interest in the presence of competing event given fixed values of covariates based on the fitted competing risk model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  penetrance_cmp(fit, event = 1, fixed, age, CI = TRUE, MC = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penetrance_cmp_+3A_fit">fit</code></td>
<td>

<p>An object class of <code>'penmodel_cmp'</code>, a fitted model by <code><a href="#topic+penmodel_cmp">penmodel_cmp</a></code> function.
</p>
</td></tr>
<tr><td><code id="penetrance_cmp_+3A_event">event</code></td>
<td>

<p>Event of interest (either 1 or 2) for penetrance estimation. Default value is 1.
</p>
</td></tr>
<tr><td><code id="penetrance_cmp_+3A_fixed">fixed</code></td>
<td>

<p>list of vectors of fixed values of the covariates for both events used for penetrance calculation.
</p>
</td></tr>
<tr><td><code id="penetrance_cmp_+3A_age">age</code></td>
<td>

<p>Vector of ages used for penetrance calculation.
</p>
</td></tr>
<tr><td><code id="penetrance_cmp_+3A_ci">CI</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the 95% confidence interval will be obtained using a Monte-Carlo method, otherwise no confidence interval will be provided. Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="penetrance_cmp_+3A_mc">MC</code></td>
<td>

<p>Number of simulated samples used to calculate confidence intervals with a Monte-Carlo method. 
If <code>MC=0</code>, no confidence intervals will be calculated. Default value is 100.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The cause-specific hazard for event <code class="reqn">j</code> is specified as:
h<sub>j</sub>(t|x) = h<sub>0j</sub>(t) exp(&beta;<sub>j</sub> *x<sub>j</sub>),
where h<sub>0j</sub>(t) is the baseline hazards function for event <code class="reqn">j</code>, x<sub>j</sub> is the vector of covariates associated with event <code class="reqn">j</code> and &beta;<sub>j</sub> is the vector of corresponding regression coefficients, <code class="reqn">j = 1, 2</code>. 
</p>
<p>The penetrance function for event <code class="reqn">j</code> in the presence of competing risks based on cause-specific hazards (with no frailties assumed) model is defined as the probability of developing an event of interest by age <code class="reqn">t</code> given fixed values of covariates <code class="reqn">x</code> in the following form: 
 P(T < t, d = j |x) = &int;<sub>t0</sub><sup>t</sup> h<sub>j</sub>(u|x<sub>j</sub>)exp(-H<sub>1</sub>(u|x<sub>1</sub>)-H<sub>2</sub>(u|x<sub>2</sub>))du,
where t<sub>0</sub> is the minimum age of onset, <code class="reqn">d</code> is the type of event which takes <code class="reqn">(j = 1, 2)</code>.
</p>
<p>The shared frailty competing risks model is:

h<sub>j</sub>(t|z<sub>j</sub>, x) = z<sub>j</sub>*h<sub>0j</sub>(t) exp(&beta;<sub>j</sub> *x<sub>j</sub>),
where z<sub>j</sub> is the shared frailty for event <code class="reqn">j</code> within families whose distribution is specified by <code>frailty.dist</code>.
</p>
<p>The penetrance function for event <code class="reqn">j</code> from the shared frailty competing risks model is obtained by integrating over the frailty distribution of G(z1, z2),

P(T < t, d = j | x) = &int;<sub>t0</sub><sup>t</sup>
&int; &int; h<sub>j</sub>(u|x<sub>j</sub>, z<sub>j</sub>) exp(-H<sub>1</sub>(u|x<sub>1</sub> z<sub>j</sub>)-H<sub>2</sub>(u|x<sub>2</sub>, z<sub>j</sub>)) dG(z<sub>1</sub> dz<sub>2</sub>)du, 
where t<sub>0</sub> is the minimum age of onset, <code class="reqn">d</code> is the type of event which takes <code class="reqn">(j = 1, 2)</code>. 
</p>
<p>See Choi et al. (2021) for more details about the penetrance functions.
</p>
<p>Calculations of standard errors of the penetrance estimates and 95% confidence intervals (CIs) for the penetrance at a given age are based on Monte-Carlo simulations of the estimated penetrance model. A multivariate normal distribution is assumed for the parameter estimates, and <code>MC = n</code> sets of the parameters are generated from the multivariate normal distribution with the parameter estimates and their variance-covariance matrix.
For each simulated set, a penetrance estimate is calculated at a given age by substituting the simulated parameters into the penetrance function.
</p>
<p>The standard error of the penetrance estimate at a given age is calculated by the standard deviation of penetrance estimates obtained from <code class="reqn">n</code> simulations.
</p>
<p>The 95% CI for the penetrance at a given age is calculated using the 2.5th and 97.5th percentiles of the penetrance estimates obtained from <code class="reqn">n</code> simulations.
</p>


<h3>Value</h3>

<p>Returns the following values:
</p>
<table>
<tr><td><code>age</code></td>
<td>
<p>Ages at which the penetrances are calculated.</p>
</td></tr>
<tr><td><code>penetrance</code></td>
<td>
<p>Penetrance estimates at given ages.</p>
</td></tr>
<tr><td><code>lower</code></td>
<td>
<p>Lower limit of the 95% confidence interval; simulation-based 2.5th percentile of the penetrance estimates.</p>
</td></tr>
<tr><td><code>upper</code></td>
<td>
<p>Upper limit of the 95% confidence interval; simulation-based 97.5th percentile of the penetrance estimates.</p>
</td></tr>
<tr><td><code>se</code></td>
<td>
<p>Simulation-based standard errors of the penetrance estimates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Jung, H., Buys, S., Daly, M., John, E.M., Hopper, J., Andrulis, I., Terry, M.B., Briollais, L. (2021) A Competing Risks Model with Binary Time Varying Covariates for Estimation of Breast Cancer Risks in BRCA1 Families, Statistical Methods in Medical Research 30 (9), 2165-2183. https://doi.org/10.1177/09622802211008945.
</p>
<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_cmp">simfam_cmp</a>, <a href="#topic+penmodel_cmp">penmodel_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(4321)
fam2 &lt;- simfam_cmp(N.fam = 200, design = "pop+", variation = "frailty", competing = TRUE,
                base.dist = "Weibull", frailty.dist = "cgamma", depend=c(2, 2, 2), 
                base.parms = list(c(0.01, 3), c(0.01, 3)), 
                vbeta = list(c(-1.13, 2.35), c(-1, 2)), allelefreq = 0.02)

fit2 &lt;- penmodel_cmp(Surv(time, status==1) ~ gender + mgene, 
                   Surv(time, status==2) ~ gender + mgene, 
                   cluster = "famID", gvar = "mgene", frailty.dist = "cgamma",
                   parms = list(c(0.01, 3, -1, 2.3), c(0.01, 3, -1, 2), c(2, 2, 2)), 
                   data = fam2, design = "pop+", base.dist = "Weibull", 
                   agemin = NULL, robust = TRUE)

# Compute penetrance estimates for event 1 for male carriers at age 40, 50, 60, 70 and
# their 95
 
penetrance_cmp(fit2, event = 1, fixed = list(c(1,1), c(1,1)), 
             age = c(40, 50, 60, 70), CI = TRUE, MC = 200)

## End(Not run)
</code></pre>

<hr>
<h2 id='penmodel'>Fit a penetrance model</h2><span id='topic+penmodel'></span>

<h3>Description</h3>

<p>Fits a penetrance model for family data based on a prospective likelihood with ascertainment correction and provides model parameter estimates. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penmodel(formula, cluster = "famID", gvar = "mgene", parms, cuts = NULL, data, 
design = "pop", base.dist = "Weibull", frailty.dist = "none",
agemin = NULL, robust = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penmodel_+3A_formula">formula</code></td>
<td>
<p>A formula expression as for other regression models. The response should be a survival object as returned by the <code>Surv</code> function. See the documentation for <code>Surv</code>, <code>lm</code> and <code>formula</code> for details.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_cluster">cluster</code></td>
<td>

<p>Name of cluster variable. Default is <code>"famID"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_gvar">gvar</code></td>
<td>
<p>Name of genetic variable. Default is <code>"mgene"</code>.</p>
</td></tr>
<tr><td><code id="penmodel_+3A_parms">parms</code></td>
<td>

<p>Vector of initial values for the parameters in the model including baseline parameters and regression coefficients. <code>parms = c(baseparm, coef)</code>, where <code>baseparm</code> includes the initial values for baseline parameters used for <code>base.dist</code>, and <code>coef</code> includes the initial values for regression coefficients for the variables specified in <code>formula</code>. If frailty.dist is specified, the initial value of the frailty parameter should be specified <code>parms = c(baseparm, coef, k)</code>, where <code>k</code> the initial value for the frailty parameter. 
See details for the baseline parameters.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_cuts">cuts</code></td>
<td>
 
<p>Vector of cut points that define the intervals where the hazard function is constant. The <code>cuts</code> should be specified when <code>base.dist="piecewise"</code> and must be strictly positive and finite. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_data">data</code></td>
<td>

<p>Data frame generated from <code><a href="#topic+simfam">simfam</a></code> or data frame containing variables named in the <code>formula</code> and specific variables: <code>famID</code>, <code>indID</code>, <code>gender</code>, <code>currentage</code>, <code>mgene</code>, <code>time</code>, <code>status</code> and <code>weight</code> with <code>attr(data,"agemin")</code> specified.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_design">design</code></td>
<td>

<p>Study design of the family data.  Possible choices are: <code>"pop", "pop+", "cli"</code>, <code>"cli+"</code> or <code>"twostage"</code>, where <code>"pop"</code> is for the population-based design with affected probands whose mutation status can be either carrier or non-carrier, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands, and <code>"twostage"</code> is for the two-stage design with oversampling of high risks families. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distributions to fit. Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code>, <code>"gamma"</code>, <code>"logBurr"</code>, or <code>"piecewise"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution. Possible choices are: <code>"gamma"</code>, <code>"lognormal"</code>, or <code>"none"</code>. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_+3A_robust">robust</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the robust &lsquo;sandwich&rsquo; standard errors and variance-covariance matrix are provided, otherwise the conventional standard errors and variance-covariance matrix are provided.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>frailty.dist = "none"</code>, the following penetrance model is fitted to family data with a specified baseline hazard distribution 
  h(t|x<sub>s</sub>, x<sub>g</sub>) = h<sub>0</sub>(t - t<sub>0</sub>)  exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>),  
where h<sub>0</sub>(t) is the baseline hazards function specified by <code>base.dist</code>, which depends on the shape and scale parameters, &lambda; and &rho;; x<sub>s</sub> indicates male (1) and female (0) and x<sub>g</sub> indicates carrier (1) or non-carrier (0) of a gene of interest (major gene). Additional covariates can be added to <code>formula</code> in the model.
</p>
<p>When <code>frailty.dist</code> is specified as either <code>"gamma"</code> or <code>"lognormal"</code>, the follwoing shared frailty model is fitted to family data 

 h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), 
where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, and Z represents a frailty shared within families whose distribution is specified by <code>frailty.dist</code>.
</p>
<p><em>Choice of frailty distributions</em>
</p>
<p><code>frailty.dist = "gamma"</code> assumes <code class="reqn">Z</code> follows Gamma(<code class="reqn">k</code>, <code class="reqn">1/k</code>). 
</p>
<p><code>frailty.dist = "lognormal"</code> assumes <code class="reqn">Z</code> follows log-normal distribution with mean 0 and variance <code class="reqn">1/k</code>.
</p>
<p><code>frailty.dist = "none"</code> shares no frailties within families and assumes independence among family members. 
</p>
<p>For family data arising from population- or clinic-based study designs (<code>design="pop", "pop+"</code>, <code>"cli"</code>, or <code>"cli+"</code>), the parameters of the penetrance model are estimated using the ascertainment-corrected prospective likelihood approach (Choi, Kopciuk and Briollais, 2008).
</p>
<p>For family data arising from a two-stage study design (<code>design="twostage"</code>), model parameters are estimated  using the composite likelihood approach (Choi and Briollais, 2011)
</p>
<p>Note that the baseline parameters include <code>lambda</code> and <code>rho</code>, which represent the scale and shape parameters, respectively, and <code>eta</code>, additional parameter to specify for <code>"logBurr"</code> distribution. For the <code>"lognormal"</code> baseline distribution, <code>lambda</code> and <code>rho</code> represent the location and scale parameters for the normally distributed logarithm, where <code>lambda</code> can take any real values and <code>rho</code> &gt; 0. For the other baselinse distributions, <code>lambda</code> &gt; 0, <code>rho</code> &gt; 0, and <code>eta</code> &gt; 0. When a piecewise constant distribution is specified for the baseline hazards, <code>base.dist="piecewise"</code>, <code>baseparm</code> should specify the initial interval-constant values, one more than the cut points specified by<code>cuts</code>.
</p>
<p>Transformed baseline parameters are used for estimation; log transformation is applied to both scale and shape parameters (<code class="reqn">\lambda, \rho</code>) for <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code> and <code>"gamma"</code> baselines, to (<code class="reqn">\lambda, \rho, \eta</code>) for <code>"logBurr"</code> and to the piecewise constant parameters for a <code>piecewise</code> baseline hazard. For <code>"lognormal"</code> baseline distribution, the log transformation is applied only to <code class="reqn">\rho</code>, not to <code class="reqn">\lambda</code>, which represents the location parameter for the normally distributed logarithm.
</p>
<p>Calculations of penetrance estimates and their standard errors and 95% confidence intervals at given ages can be obtained by <code><a href="#topic+penetrance">penetrance</a></code> function via Monte-Carlo simulations of the estimated penetrance model.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'penmodel'</code>, including the following elements:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>

<p>Parameter estimates of transformed baseline parameters and regression coefficients.
</p>
</td></tr>
<tr><td><code>varcov</code></td>
<td>

<p>Variance-covariance matrix of parameter estimates obtained from the inverse of Hessian matrix. 
</p>
</td></tr>
<tr><td><code>varcov.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; variance-covariance matrix of parameter estimates when <code>robust=TRUE</code>. 
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>Standard errors of parameter estimates obtained from the inverse of Hessian matrix.
</p>
</td></tr>
<tr><td><code>se.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; standard errors of parameter estimates when <code>robust=TRUE</code>.
</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Loglikelihood value for the fitted penetrance model.</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion (AIC) value of the model; AIC = 2*k - 2*logLik, where k is the number of parameters used in the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodelEM">penmodelEM</a></code>, <code><a href="#topic+simfam">simfam</a></code>, <code><a href="#topic+penplot">penplot</a></code>, <code><a href="#topic+print.penmodel">print.penmodel</a></code>,
<code><a href="#topic+summary.penmodel">summary.penmodel</a></code>, <code><a href="#topic+print.summary.penmodel">print.summary.penmodel</a></code>, <code><a href="#topic+plot.penmodel">plot.penmodel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Family data simulated from population-based design using a Weibull baseline hazard 

set.seed(4321)
fam &lt;- simfam(N.fam = 200, design = "pop+", variation = "none", base.dist = "Weibull", 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), agemin = 20, allelefreq = 0.02)
 
# Penetrance model fit for simulated family data

fit &lt;- penmodel(Surv(time, status) ~ gender + mgene, cluster = "famID", design = "pop+",
       parms = c(0.01, 3, -1.13, 2.35), data = fam, base.dist = "Weibull")

# Summary of the model parameter estimates from the model fit

summary(fit)

# Plot the lifetime penetrance curves with 95% CIs from the model fit for specific  
# gender and mutation status groups along with their nonparametric penetrance curves  
# based on data excluding probands. 

plot(fit, add.KM = TRUE, conf.int = TRUE, MC = 100)
</code></pre>

<hr>
<h2 id='penmodel_cmp'>Fit a penetrance model for competing risks data</h2><span id='topic+penmodel_cmp'></span>

<h3>Description</h3>

<p>Fits a competing risks model for family data with ascertainment correction and provides model parameter estimates. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penmodel_cmp(formula1, formula2, cluster = "famID", gvar = "mgene", 
parms, cuts = NULL, data,  design = "pop", base.dist = "Weibull", 
frailty.dist = "none", agemin = NULL, robust = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penmodel_cmp_+3A_formula1">formula1</code></td>
<td>
<p>A formula expression for event 1 as for other regression models. The response should be a survival object as returned by the <code>Surv</code> function. See the documentation for <code>Surv</code>, <code>lm</code> and <code>formula</code> for details.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_formula2">formula2</code></td>
<td>
<p>A formula expression for event 2 as for other regression models. The response should be a survival object as returned by the <code>Surv</code> function. See the documentation for <code>Surv</code>, <code>lm</code> and <code>formula</code> for details.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_cluster">cluster</code></td>
<td>

<p>Name of cluster variable. Default is <code>"famID"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_gvar">gvar</code></td>
<td>
<p>Name of genetic variable. Default is <code>"mgene"</code>.</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_parms">parms</code></td>
<td>

<p>list of Vectors of initial values for the parameters in each model including baseline parameters and regression coefficients and frailty parameters. 
</p>
<p><code>parms = list(c(baseparm1, coef1), c(baseparms2, coef2), c(k1, k2)</code>, where <code>baseparm1</code> and <code>baseparm2</code> include the initial values for baseline parameters used for each <code>base.dist</code>; <code>coef1</code> and <code>coef2</code> include the initial values for regression coefficients for the variables specified in <code>formula1</code> and <code>formula2</code>, respectively, and <code>c(k1, k2)</code> are the initial values for frailty parameters used for the specified frailty.distribution.
See Details for more details.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_cuts">cuts</code></td>
<td>
 
<p>Vector of cut points that define the intervals when <code>base.dist="piecewise"</code> is specified and must be strictly positive and finite and greater than <code>agemin</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_data">data</code></td>
<td>

<p>Data frame generated from <code><a href="#topic+simfam">simfam</a></code> or data frame containing variables named in the <code>formula</code> and specific variables: <code>famID</code>, <code>indID</code>, <code>gender</code>, <code>currentage</code>, <code>mgene</code>, <code>time</code>, <code>status</code> and <code>weight</code> with <code>attr(data,"agemin")</code> specified.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_design">design</code></td>
<td>

<p>Study design of the family data.  Possible choices are: <code>"pop"</code> and <code>"pop+"</code>, where <code>"pop"</code> is for the population-based design with affected probands whose mutation status can be either carrier or non-carrier, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_base.dist">base.dist</code></td>
<td>

<p>Vector of two baseline hazard distributions to be fitted for competing events. Possible choices for each event are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code>, <code>"gamma"</code>, <code>"logBurr"</code>, or <code>"piecewise"</code>. If only one distribution is specified, the same distribution will be assumed for both events. A vector of two distributions should be specified if different baseline distributions are assumed for different events.
Default is <code>c("Weibull", "Weibull")</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution to fit a shared frailty model for competing events. Possible choices are: <code>"gamma"</code> for independent gamma, <code>"lognormal"</code> for independent log-normal, <code>"cgamma"</code> for correlated gamma, <code>"clognormal"</code> for correlated log-normal distributions, or <code>"none"</code> can be chosen to ignoring frailties in the model assuming no residual familial correlation given covariates. 
Default is <code>"gamma")</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penmodel_cmp_+3A_robust">robust</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the robust &lsquo;sandwich&rsquo; standard errors and variance-covariance matrix are provided, otherwise the conventional standard errors and variance-covariance matrix are provided.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The shared frailty comepting risks model is fitted to family data with specified baseline hazard distributions and frailty distribution
</p>
<p>Event 1:
</p>

<p style="text-align:center"> h<sub>1</sub>(t|X,Z) = h<sub>01</sub>(t - t<sub>0</sub>) Z<sub>1</sub> exp(&beta;<sub>s1</sub> * x<sub>s</sub> + &beta;<sub>g1</sub> * x<sub>g</sub>), </p>
<p>Event 2:
</p>

<p style="text-align:center"> h<sub>2</sub>(t|X,Z) = h<sub>02</sub>(t - t<sub>0</sub>) Z<sub>2</sub> exp(&beta;<sub>s2</sub> * x<sub>s</sub> + &beta;<sub>g2</sub> * x<sub>g</sub>), </p>
<p>where h<sub>01</sub>(t) and h<sub>02</sub>(t) are the baseline hazard functions for event 1 and event 2, respectively, which can be specified by <code>base.dist</code>. t<sub>0</sub> is a minimum age of disease onset, Z<sub>1</sub> and Z<sub>2</sub> are frailties shared within families for each event and follow either a gamma, log-normal, correlateg gamma, or correlated log-normal distributions, which can be specified by <code>frailty.dist</code>. x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively. Additional covariates can be added to <code>formula1</code> for event 1 and <code>formula2</code> for event 2 in the model. 
</p>
<p><em>Choice of frailty distributions for competing risk models</em>
</p>
<p><code>frailty.dist = "gamma"</code> shares the frailties within families generated from a gamma distribution independently for each competing event, where
Z<sub>j</sub> follows Gamma(k<sub>j</sub>, 1/k<sub>j</sub>). 
</p>
<p><code>frailty.dist = "lognormal"</code> shares the frailties within families generated from a log-normal distribution independently for each competing event, where 
Z<sub>j</sub> follows log-normal distribution with mean 0 and variance 1/k<sub>j</sub>.
</p>
<p><code>frailty.dist = "cgamma"</code> shares the frailties within families generated from a correlated gamma distribution to allow the frailties between two events to be correlated, where the correlated gamma frailties (Z<sub>1</sub>, Z<sub>2</sub>) are generated with three independent gamma frailties (Y<sub>0</sub>, Y<sub>1</sub>, Y<sub>2</sub>) as follows:
</p>
Z<sub>1</sub> = k<sub>0</sub>/(k<sub>0</sub> + k<sub>1</sub>) Y<sub>0</sub> + Y<sub>1</sub> <p>;
</p>
Z<sub>2</sub> = k<sub>0</sub>/(k<sub>0</sub> + k<sub>2</sub>) Y<sub>0</sub> + Y<sub>2</sub> <p>,
</p>
<p>where Y<sub>0</sub> from Gamma(k<sub>0</sub>, 1/k<sub>0</sub>),
Y<sub>1</sub> from Gamma(k<sub>1</sub>, 1/(k<sub>0</sub> + k<sub>1</sub>)),
Y<sub>2</sub> from Gamma(k<sub>2</sub>, 1/(k<sub>0</sub> + k<sub>2</sub>)).
</p>
<p><code>frailty.dist = "clognormal"</code> shares the frailties within families generated from a correlated log-normal distribution where 
log(Z<sub>j</sub>) follows  a normal distribution with mean 0, variance 1/k<sub>j</sub> and correlation between two events k<sub>0</sub>.
</p>
<p><code>depend</code> should specify the values of related frailty parameters: <code>c(k1, k2)</code> with <code>frailty.dist = "gamma"</code> or <code>frailty.dist = "lognormal"</code>; <code>c(k1, k2, k0)</code> for <code>frailty.dist = "cgamma"</code> or <code>frailty.dist = "clognormal"</code>. 
</p>
<p>More details about the competing risks model for family data arising from population-based study designs (<code>design="pop", "pop+"</code> and their inference procedure based on the ascertainment-corrected likelihood approach can be found in Choi et al., 2021.
</p>
<p>Note that the baseline parameters include <code>lambda</code> and <code>rho</code>, which represent the scale and shape parameters, respectively, and <code>eta</code>, additional parameter to specify for <code>"logBurr"</code> distribution. For the <code>"lognormal"</code> baseline distribution, <code>lambda</code> and <code>rho</code> represent the location and scale parameters for the normally distributed logarithm, where <code>lambda</code> can take any real values and <code>rho</code> &gt; 0. For the other baselinse distributions, <code>lambda</code> &gt; 0, <code>rho</code> &gt; 0, and <code>eta</code> &gt; 0. When a piecewise constant distribution is specified for the baseline hazards, <code>base.dist="piecewise"</code>, <code>baseparm</code> should specify the initial interval-constant values, one more than the cut points specified by<code>cuts</code>.
</p>
<p>Transformed baseline parameters are used for estimation; log transformation is applied to both scale and shape parameters (<code class="reqn">\lambda, \rho</code>) for <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code> and <code>"gamma"</code> baselines, to (<code class="reqn">\lambda, \rho, \eta</code>) for <code>"logBurr"</code> and to the piecewise constant parameters for a <code>piecewise</code> baseline hazard. For <code>"lognormal"</code> baseline distribution, the log transformation is applied only to <code class="reqn">\rho</code>, not to <code class="reqn">\lambda</code>, which represents the location parameter for the normally distributed logarithm.
</p>
<p>Calculations of penetrance estimates and their standard errors and 95% confidence intervals at given ages can be obtained by <code><a href="#topic+penetrance">penetrance</a></code> function via Monte-Carlo simulations of the estimated penetrance model.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'penmodel_cmp'</code>, including the following elements:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>

<p>Parameter estimates of transformed baseline parameters and regression coefficients.
</p>
</td></tr>
<tr><td><code>varcov</code></td>
<td>

<p>Variance-covariance matrix of parameter estimates obtained from the inverse of Hessian matrix. 
</p>
</td></tr>
<tr><td><code>varcov.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; variance-covariance matrix of parameter estimates when <code>robust=TRUE</code>. 
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>Standard errors of parameter estimates obtained from the inverse of Hessian matrix.
</p>
</td></tr>
<tr><td><code>se.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; standard errors of parameter estimates when <code>robust=TRUE</code>.
</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Loglikelihood value for the fitted penetrance model.</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion (AIC) value of the model; AIC = 2*k - 2*logLik, where k is the number of parameters used in the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Jung, H., Buys, S., Daly, M., John, E.M., Hopper, J., Andrulis, I., Terry, M.B., Briollais, L. (2021) A Competing Risks Model with Binary Time Varying Covariates for Estimation of Breast Cancer Risks in BRCA1 Families, Statistical Methods in Medical Research 30 (9), 2165-2183. https://doi.org/10.1177/09622802211008945.
</p>
<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_cmp">simfam_cmp</a></code>, <code><a href="#topic+penplot_cmp">penplot_cmp</a></code>, <code><a href="#topic+print.penmodel_cmp">print.penmodel_cmp</a></code>,
<code><a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a></code>, <code><a href="#topic+print.summary.penmodel_cmp">print.summary.penmodel_cmp</a></code>, <code><a href="#topic+plot.penmodel_cmp">plot.penmodel_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Competing risk family data simulated from population-based design 
# using Weibull baseline hazards with gamma frailty distribution.
## Not run: 
set.seed(4321)
fam1 &lt;- simfam_cmp(N.fam = 200, design = "pop+", variation = "frailty", 
              base.dist = "Weibull", frailty.dist = "cgamma", depend=c(0.5, 1, 0.5), 
              allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
              vbeta = list(c(-1.13, 2.35), c(-1, 2)))

# Fitting shared gamma frailty Penetrance model for simulated competing risk data

fit1 &lt;- penmodel_cmp(
        formula1 = Surv(time, status==1) ~ gender + mgene, 
        formula2 = Surv(time, status==2) ~ gender + mgene,
        cluster = "famID", gvar = "mgene", design = "pop+",  
        parms = list(c(0.01, 3, -1, 2), c(0.01, 3, -1, 2), c(0.5, 1)),
        base.dist = "Weibull", frailty.dist = "gamma", data = fam1, robust = TRUE)

# Fitting shared correlated gamma frailty Penetrance model for simulated competing risk data

fit2 &lt;- penmodel_cmp(
        formula1 = Surv(time, status==1) ~ gender + mgene, 
        formula2 = Surv(time, status==2) ~ gender + mgene,
        cluster = "famID", gvar = "mgene", design = "pop+",  
        parms = list(c(0.01, 3, -1, 2), c(0.01, 3, -1, 2), c(0.5, 1, 0.5)),
        base.dist = "Weibull", frailty.dist = "cgamma", data = fam1, robust = TRUE)

# Summary of the model parameter estimates from the model fit

summary(fit1)

# Plot the lifetime penetrance curves with 95
# gender and mutation status groups along with their nonparametric penetrance curves  
# based on data excluding probands. 

plot(fit1, add.CIF = TRUE, conf.int = TRUE, MC = 100)

## End(Not run)
</code></pre>

<hr>
<h2 id='penmodelEM'>EM algorithm for estimating the penetrance model with missing genotypes</h2><span id='topic+penmodelEM'></span>

<h3>Description</h3>

<p>Fits a penetrance model for family data with missing genotypes via the EM algorithm and provides model parameter estimates. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penmodelEM(formula, cluster = "famID", gvar = "mgene", parms, cuts = NULL, data, 
design = "pop", base.dist = "Weibull", agemin = NULL, robust = FALSE, method = "data", 
mode = "dominant", q = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penmodelEM_+3A_formula">formula</code></td>
<td>
<p>A formula expression as for other regression models. The response should be a survival object as returned by the <code>Surv</code> function. See the documentation for <code>Surv</code>, <code>lm</code> and <code>formula</code> for details.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_cluster">cluster</code></td>
<td>

<p>Name of cluster variable. Default is <code>"famID"</code>.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_gvar">gvar</code></td>
<td>
<p>Name of genetic variable. Default is <code>"mgene"</code>.</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_parms">parms</code></td>
<td>

<p>Vector of initial values for the parameters in the model including baseline parameters and regression coefficients. <code>parms = c(baseparm, coef)</code>, where <code>baseparm</code> includes the initial values for baseline parameters used for <code>base.dist</code>, and <code>coef</code> includes the initial values for regression coefficients for the variables specified in <code>formula</code>. See details for the baseline parameters.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_cuts">cuts</code></td>
<td>
 
<p>Vector of cuts that define the intervals where the hazard function is constant. The <code>cuts</code> should be specified <code>base.dist="piecewise"</code> and must be strictly positive and finite. Default is <code>NULL</code>.
</p>
</td></tr>	
<tr><td><code id="penmodelEM_+3A_data">data</code></td>
<td>

<p>Data frame generated from <code><a href="#topic+simfam">simfam</a></code> or data frame containing specific variables: <code>famID</code>, <code>indID</code>, <code>gender</code>, <code>currentage</code>, <code>mgene</code>, <code>time</code>, <code>status</code> and <code>weight</code> with <code>attr(data,"agemin")</code> specified.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_design">design</code></td>
<td>

<p>Study design of the family data. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code> or <code>"twostage"</code>, where <code>"pop"</code> is for the population-based design with affected probands whose mutation status can be either carrier or non-carrier, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands, and <code>"twostage"</code> is for the two-stage design with oversampling of high risks families. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distributions to fit. Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code>, <code>"gamma"</code>, <code>"logBurr"</code>, or <code>"piecewise"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_robust">robust</code></td>
<td>

<p>Logical; if <code>TRUE</code>, the robust &lsquo;sandwich&rsquo; standard errors and variance-covariance matrix are provided, otherwise the conventional standard errors and variance-covariance matrix are provided.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_method">method</code></td>
<td>

<p>Choice of methods for calculating the carrier probabilities for individuals with missing mutation status. Possible choices are <code>"data"</code> for empirical calculation of the carrier probabilities based on the observed carriers' statuses in the entire sample, specific to generation and proband's mutation status or <code>"mendelian"</code> for calculating carrier probabilities based on Mendelian transmission probabilies with the given allele frequency and mutation statuses observed in the family. Default is <code>"data"</code>.
</p>
<p>If <code>method = "mendelian"</code>, specify both <code>mode</code> of the inheritance and the allele frequency <code>q</code>. 
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_mode">mode</code></td>
<td>

<p>Choice of modes of inheritance for calculating carrier probabilies for individuals with missing mutation status. Possible choices are <code>"dominant"</code> for dominant model or <code>"recessive"</code> for recessive model. Default is <code>"dominant"</code>.
</p>
</td></tr>
<tr><td><code id="penmodelEM_+3A_q">q</code></td>
<td>

<p>Frequency of the disease causing allele used for calculating carrier pobabilities. The value should be between 0 and 1. If <code>NULL</code>, the estimated allele frequency from data will be used. Default value  is <code>0.02</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The expectation and maximization (EM) algorithm is applied for making inference about the missing genotypes. In the expectation step, for individuals with unknown carrier status, we first compute their carrier probabilities given their family's observed phenotype and genotype information based on current estimates of parameters &theta; as follows,
w<sub>fi</sub> = P(X<sub>fi</sub> = 1 | Y<sub>fi</sub>, X<sub>f</sub><sup>o</sup>), 
where  X<sub>fi</sub> represents the mutation carrier status and Y<sub>fi</sub> represents the phenotype in terms of age at onset t<sub>fi</sub> and disease status &delta;<sub>fi</sub> for individual i, i = 1, ..., n<sub>f</sub>, in family <code class="reqn">f, f = 1, ..., n,</code> and X<sub>f</sub><sup>o</sup> represents the observed genotypes in family <code class="reqn">f</code>.
</p>
<p>Then, we obtain the conditional expectation of the log-likelihood function (<code class="reqn">\ell</code>) of the complete data given the observed data as a weighted log-likelihood, which has the form
 E<sub>&theta;</sub> [ l (&theta; | Y, X<sup>o</sup>) ] =  &sum;<sub>f</sub> &sum;<sub>i</sub> l <sub> fi</sub> (&theta; | X<sub>fi</sub> = 1) * w<sub>fi</sub> +  l<sub>fi</sub> (&theta; | X<sub>fi</sub> = 0) * ( 1 - w<sub>fi</sub> ). 
In the maximization step, the updated parameter estimates are obtained by maximizing the weighted log likelihood computed in the E-step. These expectation and maximization steps iterate until convergence to obtain the maximum likelihood estimates. See more details in Choi and Briollais (2011) or Choi et al. (2014).
</p>
<p>Note that the baseline parameters include <code>lambda</code> and <code>rho</code>, which represent the scale and shape parameters, respectively, and <code>eta</code>, additional parameter to specify for <code>"logBurr"</code> distribution. For the <code>"lognormal"</code> baseline distribution, <code>lambda</code> and <code>rho</code> represent the location and scale parameters for the normally distributed logarithm, where <code>lambda</code> can take any real values and <code>rho</code> &gt; 0. For the other baselinse distributions, <code>lambda</code> &gt; 0, <code>rho</code> &gt; 0, and <code>eta</code> &gt; 0. When a piecewise constant distribution is specified for the baseline hazards, <code>base.dist="piecewise"</code>, <code>baseparm</code> should specify the initial interval-constant values, one more than the cut points specified by<code>cuts</code>.
</p>
<p>Transformed baseline parameters are used for estimation; log transformation is applied to both scale and shape parameters (<code class="reqn">\lambda, \rho</code>) for <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code> and <code>"gamma"</code> baselines, to (<code class="reqn">\lambda, \rho, \eta</code>) for <code>"logBurr"</code> and to the piecewise constant parameters for a <code>piecewise</code> baseline hazard. For <code>"lognormal"</code> baseline distribution, the log transformation is applied only to <code class="reqn">\rho</code>, not to <code class="reqn">\lambda</code>, which represents the location parameter for the normally distributed logarithm.
</p>
<p>Calculations of penetrance estimates and their standard errors and 95% confidence intervals at given ages can be obtained by <code><a href="#topic+penetrance">penetrance</a></code> function via Monte-Carlo simulations of the estimated penetrance model.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'penmodel'</code>, including the following elements:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>

<p>Parameter estimates of transformed baseline parameters and regression coefficients.
</p>
</td></tr>
<tr><td><code>varcov</code></td>
<td>

<p>Variance-covariance matrix of parameter estimates obtained from the inverse of Hessian matrix. 
</p>
</td></tr>
<tr><td><code>varcov.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; variance-covariance matrix of parameter estimates when <code>robust=TRUE</code>. 
</p>
</td></tr>
<tr><td><code>se</code></td>
<td>

<p>Standard errors of parameter estimates obtained from the inverse of Hessian matrix.
</p>
</td></tr>
<tr><td><code>se.robust</code></td>
<td>

<p>Robust &lsquo;sandwich&rsquo; standard errors of parameter estimates when <code>robust=TRUE</code>.
</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Loglikelihood value for the fitted penetrance model.</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion (AIC) value of the model; AIC = 2*k - 2*logLik, where k is the number of parameters used in the model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H. and Briollais, L. (2011) An EM composite likelihood approach for multistage sampling of family data with missing genetic covariates, Statistica Sinica 21, 231-253.
</p>
<p>Choi, Y.-H., Briollais, L., Green, J., Parfrey, P., and Kopciuk, K. (2014) Estimating successive cancer risks in Lynch Syndrome families using a progressive three-state model, Statistics in Medicine 33, 618-638.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a></code>, <code><a href="#topic+penmodel">penmodel</a></code>, <code><a href="#topic+print.penmodel">print.penmodel</a></code>, <code><a href="#topic+summary.penmodel">summary.penmodel</a></code>,
<code><a href="#topic+print.summary.penmodel">print.summary.penmodel</a></code>, <code><a href="#topic+plot.penmodel">plot.penmodel</a></code>, <code><a href="#topic+carrierprob">carrierprob</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Family data simulated with 20% of members missing their genetic information.

set.seed(4321)
fam &lt;- simfam(N.fam = 100, design = "pop+", base.dist = "Weibull", base.parms = c(0.01, 3),
       vbeta = c(1, 2), agemin = 20, allelefreq = 0.02, mrate = 0.2)
 
# EM algorithm for fitting family data with missing genotypes

fit &lt;- penmodelEM(Surv(time, status) ~ gender + mgene, cluster = "famID", gvar = "mgene", 
       parms = c(0.01, 3, 1, 2), data = fam, design="pop+", robust = TRUE, 
       base.dist = "Weibull", method = "mendelian", mode = "dominant", q = 0.02)

# Summary of the model parameter estimates from the model fit by penmodelEM 

summary(fit)

# Plot the lifetime penetrance curves from model fit for gender and 
# mutation status groups along with their nonparametric penetrance curves 
# based on observed data excluding probands.
 
plot(fit)

</code></pre>

<hr>
<h2 id='penplot'>
Plot penetrance functions
</h2><span id='topic+penplot'></span>

<h3>Description</h3>

<p>Plots the penetrance functions given the values of baseline parameters and regression coefficients and choices of baseline and frailty distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penplot(base.parms, vbeta, cuts = NULL, variation = "none", base.dist = "Weibull", 
frailty.dist = NULL, depend = 1, agemin = 20, agemax = 80, print = TRUE, 
col = c("blue","red","blue","red"),  lty = c(1, 1, 2, 2), add.legend = TRUE, 
add.title = TRUE, x = "topleft", y = NULL, xlab = "Age at onset", ylab = "Penetrance", 
ylim = NULL, main = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penplot_+3A_base.parms">base.parms</code></td>
<td>

<p>Vector of parameter values for the specified baseline hazard function: <code>base.parms = c(lambda, rho)</code> should be specified for <code>base.dist = "Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"gamma"</code>, and <code>"lognormal"</code>, <code>c(lambda, rho, eta)</code> for <code>base.dist = "logBurr"</code>, or interval constant hazard values for the intervals produced by <code>cuts</code> for <code>base.dist = "piecewise"</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_vbeta">vbeta</code></td>
<td>

<p>Vector of regression coefficients for gender and majorgene, <code>vbeta = c(beta.s, beta.g)</code>. If <code>variation = "secondgene"</code>, regression coefficients for gender, major gene and second gene, <code>vbeta = c(beta.s, beta.g1, beta.g2)</code>, should be specified.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_cuts">cuts</code></td>
<td>

<p>Vector of cut points defining the intervals where the hazard function is constant. The <code>cuts</code> should be specified when <code>base.dist = "piecewise"</code> and must be strictly positive and finite. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution. Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code>,  <code>"gamma"</code>, or <code>"piecewise"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution. Possible choices are <code>"gamma"</code> for gamma distribution or <code>"lognormal"</code> for log normal distributions when <code>variation = "frailty"</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_depend">depend</code></td>
<td>

<p>Variance of the frailty distribution. Dependence within families increases with <code>depend</code> value.  Default value is 1. 
</p>
</td></tr>
<tr><td><code id="penplot_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset. Default is 80 years of age.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_print">print</code></td>
<td>

<p>Logical; if <code>TRUE</code>, prints the penetrance values by age 70 obtained from the assumed model for male carriers, female carriers, male noncarrers, and female noncarriers.	 Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_col">col</code></td>
<td>

<p>Colors of lines for male carriers, female carriers, male noncarrers, and female noncarriers. Default is <code>c("blue", "red", "blue", "red")</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_lty">lty</code></td>
<td>

<p>Types of lines for male carriers, female carriers, male noncarriers, and female noncarriers. Default is <code>c(1, 1, 2, 2)</code>.
</p>
</td></tr>
<tr><td><code id="penplot_+3A_add.legend">add.legend</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays legend in the plot. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_add.title">add.title</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays title in the plot. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_x">x</code>, <code id="penplot_+3A_y">y</code></td>
<td>
<p>Position of legend; see <a href="graphics.html#topic+legend">legend</a>. Defaults are <code>x = "topleft", y = NULL</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x-axis. Default is <code>"Age at onset"</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y-axis. Default is <code>"Penetrance"</code>.</p>
</td></tr>
<tr><td><code id="penplot_+3A_ylim">ylim</code></td>
<td>
<p>Limits of the y-axis. Default is <code>NULL</code>. If <code>NULL</code>, <code>ylim</code> will be automatically determined. </p>
</td></tr>
<tr><td><code id="penplot_+3A_main">main</code></td>
<td>
<p>Main title of the plot. Default is <code>NULL</code>. If <code>NULL</code>, the title will be automatically created.</p>
</td></tr>
<tr><td><code id="penplot_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Proportional hazard models</em>
</p>
<p>The penetrance model conditional on the covariates   X = c(x<sub>s</sub>, x<sub>g</sub>) is assumed to have the following hazard function:
 h(t|X) = h<sub>0</sub>(t - t<sub>0</sub>) exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>),
where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p>The penetrance function for the penetrance model has the form,
1 - exp(- H<sup>0</sup>(t - t<sub>0</sub>) * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub> )), 
where H<sub>0</sub>(t) is the cumulative baseline hazard function.
</p>
<p><em>Shared frailty models</em>
</p>
<p>The penetrance model conditional on the frailty <code class="reqn">Z</code> and covariates 
  X = c(x<sub>s</sub>, x<sub>g</sub>) is assumed to have the following hazard function:
  h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>),
where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p>For example, when using a Weibull distribution for baseline hazard and a gamma distribution for frailty, the penetrance function has the form
1 - (1 + &lambda;<sup>&rho;</sup> * (t - t<sub>0</sub>)<sup>&rho;</sup> * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>)/&kappa;)<sup>-&kappa;</sup>.
</p>
<p><em>Two-gene models</em>
</p>
<p>The penetrance curve for the two-gene model is generated by
1 - exp(- H<sup>0</sup>(t - t<sub>0</sub>) * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>1</sub> * x<sub>1</sub> + &beta;<sub>2</sub> * x<sub>2</sub>)), 
where H<sub>0</sub>(t) is the cumulative baseline hazard function,
x<sub>1</sub> indicates carrior (1) or non-carrior (0) of a major gene and x<sub>2</sub> indicates carrior (1) or non-carrior (0) of a second gene.
When plotting with the two-gene model, the plot will generate separate curves for mutation carriers and noncarriers, and separate curves for the second gene carriers and noncarriers.
</p>


<h3>Value</h3>

<p>Displays plots of the penetrance functions and returns the following values:
</p>
<table>
<tr><td><code>pen70</code></td>
<td>
<p>Penetrance estimates by age 70 specific to gender and mutation-status subgroups. </p>
</td></tr>
<tr><td><code>x.age</code></td>
<td>
<p>Vector of ages of onset ranging from <code>agemin</code> to <code>agemax</code> years</p>
</td></tr>
<tr><td><code>pen</code></td>
<td>
<p>Penetrance estimates computed at each age of <code>x.age</code>; if <code>variation = "none"</code> or <code>"frailty"</code>, it includes subgroups specific to gender and mutation status for major gene. If <code>variation = "secondgene"</code>, it includes subgroups specific to gender and both mutation statuses for major gene and second gene.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a>, <a href="#topic+plot.penmodel">plot.penmodel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Penetrance function curves based on Weibull baseline hazard function

penplot(base.parms = c(0.01,3), vbeta = c(0.5, 2), variation = "none", base.dist = "Weibull", 
		agemin = 20, ylim = c(0,1))
</code></pre>

<hr>
<h2 id='penplot_cmp'>
Plot penetrance functions from competing risk models
</h2><span id='topic+penplot_cmp'></span>

<h3>Description</h3>

<p>Plots the penetrance functions from competing risk models given the values of baseline parameters and regression coefficients and choices of baseline and frailty distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penplot_cmp(event, base.parms, vbeta, cuts = NULL, variation = "none", 
          base.dist = "Weibull", frailty.dist = NULL, depend = c(1, 1), 
          agemin = 20, agemax = 80, print = TRUE, 
          col = c("blue","red","blue","red"),  lty = c(1, 1, 2, 2), 
          add.legend = TRUE, add.title = TRUE, x = "topleft", 
          y = NULL, xlab = "Age at onset", ylab = "Penetrance", 
          ylim = NULL, main = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penplot_cmp_+3A_event">event</code></td>
<td>
<p>Event of interest for penetrance function: either 1 or 2. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_base.parms">base.parms</code></td>
<td>

<p>List of vectors of parameter values for the specified baseline hazard functions for both events. For example, <code>base.parms = list(c(lambda1, rho1), c(lambda2, rho2))</code> should be specified when <code>base.dist = c("Weibull", "Weibull")</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_vbeta">vbeta</code></td>
<td>

<p>List of vectors of regression coefficients for gender and majorgene, <code>vbeta = list(c(beta1.s, beta1.g), c(beta2.s, beta2.g))</code> for both events. If <code>variation = "secondgene"</code>, regression coefficients for gender, major gene and second gene, <code>vbeta = list(c(beta1.s, beta1.g1, beta1.g2), c(beta2.s, beta2.g1, beta2.g2))</code>, should be specified for each event.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_cuts">cuts</code></td>
<td>

<p>Vector of cut points defining the intervals when <code>base.dist = "piecewise"</code> is specified and must be strictly positive and finite and greater than <code>agemin</code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_base.dist">base.dist</code></td>
<td>

<p>Vector of two baseline hazard distributions chosen for competing events. Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code>,  <code>"gamma"</code>, or <code>"piecewise"</code>.If only one distribution is specified, the same distribution will be assumed for both events. A vector of two distributions should be specified if different baseline distributions are assumed for different events. Default is <code>c("Weibull", "Weibull")</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution. Possible choices are <code>"gamma"</code> for independent gamma, <code>"lognormal"</code> for independent log-normal, <code>"cgamma"</code> for correlated gamma, <code>"clognormal"</code> for correlated log-normal distribution when <code>variation = "frailty"</code> or <code>"none" or NULL</code> when no frailty distribution is assumed. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_depend">depend</code></td>
<td>

<p>Vector of frailty parameter values assumed for specified frailty distribution.  They  corresponds inverse of variance of the frailty distribution. Dependence within families decreases with <code>depend</code> value.  Default value is <code>c(1,1)</code>. 
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset. Default is 80 years of age.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_print">print</code></td>
<td>

<p>Logical; if <code>TRUE</code>, prints the penetrance values by age 70 obtained from the assumed model for male carriers, female carriers, male noncarrers, and female noncarriers.	 Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_col">col</code></td>
<td>

<p>Colors of lines for male carriers, female carriers, male noncarrers, and female noncarriers. Default is <code>c("blue", "red", "blue", "red")</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_lty">lty</code></td>
<td>

<p>Types of lines for male carriers, female carriers, male noncarriers, and female noncarriers. Default is <code>c(1, 1, 2, 2)</code>.
</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_add.legend">add.legend</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays legend in the plot. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_add.title">add.title</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays title in the plot. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_x">x</code>, <code id="penplot_cmp_+3A_y">y</code></td>
<td>
<p>Position of legend; see <a href="graphics.html#topic+legend">legend</a>. Defaults are <code>x = "topleft", y = NULL</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x-axis. Default is <code>"Age at Onset"</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y-axis. Default is <code>"Penetrance"</code>.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_ylim">ylim</code></td>
<td>
<p>Limits of the y-axis. Default is <code>NULL</code>. If <code>NULL</code>, <code>ylim</code> will be automatically determined.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_main">main</code></td>
<td>
<p>Main title of the plot. Default is <code>NULL</code>. If <code>NULL</code>, the title will be automatically created.</p>
</td></tr>
<tr><td><code id="penplot_cmp_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Cause-specific proportional hazard models</em>
The penetrance models for competing events conditional on the covariates   X = c(x<sub>s</sub>, x<sub>g</sub>) are assumed to have the following hazard functions for event <code class="reqn">j = 1, 2</code>:
 h<sub>j</sub>(t|X) = h<sub>0j</sub>(t - t<sub>0</sub>) exp(&beta;<sub>js</sub> * x<sub>s</sub> + &beta;<sub>jg</sub> * x<sub>g</sub>), 
where h<sub>0j</sub>(t) is the baseline hazard function for event <code class="reqn">j</code>, <code class="reqn">j=1,2</code>, t<sub>0j</sub> is a minimum age of disease onset, x<sub>s</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a major gene of interest, respectively.
</p>
<p>The penetrance function for the penetrance model has the form, 1 - exp(- H<sup>0</sup>(t - t<sub>0</sub>) * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub> )), 
where H<sub>0</sub>(t) is the cumulative baseline hazard function.
</p>
<p><em>Shared frailty models</em>
</p>
<p>The penetrance model conditional on the frailty <code class="reqn">Z</code> and covariates 
  X = c(x<sub>s</sub>, x<sub>g</sub>) is assumed to have the following hazard function:
 h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>),
where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p>For example, when using a Weibull distribution for baseline hazard and a gamma distribution for frailty, the penetrance function has the form
 1 - (1 + &lambda;<sup>&rho;</sup> * (t - t<sub>0</sub>)<sup>&rho;</sup> * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>)/&kappa;)<sup>-&kappa;</sup>.
</p>
<p><em>Two-gene models</em>
</p>
<p>The penetrance curve for the two-gene model is generated by
1 - exp(- H<sup>0</sup>(t - t<sub>0</sub>) * exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>1</sub> * x<sub>1</sub> + &beta;<sub>2</sub> * x<sub>2</sub>)), 
where H<sub>0</sub>(t) is the cumulative baseline hazard function,
x<sub>1</sub> indicates carrior (1) or non-carrior (0) of a major gene and x<sub>2</sub> indicates carrior (1) or non-carrior (0) of a second gene.
When plotting with the two-gene model, the plot will generate separate curves for mutation carriers and noncarriers, and separate curves for the second gene carriers and noncarriers.
</p>


<h3>Value</h3>

<p>Displays plots of the penetrance functions and returns the following values:
</p>
<table>
<tr><td><code>pen70</code></td>
<td>
<p>Penetrance estimates by age 70 specific to gender and mutation-status subgroups. </p>
</td></tr>
<tr><td><code>x.age</code></td>
<td>
<p>Vector of ages of onset ranging from <code>agemin</code> to <code>agemax</code> years</p>
</td></tr>
<tr><td><code>pen</code></td>
<td>
<p>Penetrance estimates computed at each age of <code>x.age</code>; if <code>variation = "none"</code> or <code>"frailty"</code>, it includes subgroups specific to gender and mutation status for major gene. If <code>variation = "secondgene"</code>, it includes subgroups specific to gender and both mutation statuses for major gene and second gene.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_cmp">simfam_cmp</a>, <a href="#topic+plot.penmodel_cmp">plot.penmodel_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Penetrance function curves for event 1 
# based on Weibull baselines (no frailty)
penplot_cmp(event=1, base.parms = list(c(0.01,3), c(0.01, 3)), 
        vbeta = list(c(-1, 2), c(-1, 1)), variation = "none", 
        base.dist = "Weibull", agemin = 20, ylim = c(0,1))

# Penetrance function curves for event 1 
# based on gamma frailty and Weibull baselines
penplot_cmp(event=1, base.parms = list(c(0.01,3), c(0.01, 3)), 
        vbeta = list(c(-1, 2), c(-1, 1)), depend=c(2, 2), 
        variation = "frailty", frailty.dist="gamma", base.dist = "Weibull", 
        agemin = 20, ylim = c(0,1))

# Penetrance function curves for event 1 
# based on correlated gamma frailty and Weibull baselines
penplot_cmp(event=1, base.parms = list(c(0.01,3), c(0.01, 3)), 
        vbeta = list(c(-1, 2), c(-1, 1)), depend=c(2, 2, 0.2), 
        variation = "frailty", frailty.dist="cgamma", 
        base.dist = "Weibull", agemin = 20, ylim = c(0,1))

# Penetrance function curves for event 1 
# based on correlated lognormal frailty and Weibull baselines
penplot_cmp(event=1, base.parms = list(c(0.01,3), c(0.01, 3)), 
        vbeta = list(c(-1, 2), c(-1, 1)), depend=c(2, 2, 0.2), 
        variation = "frailty", frailty.dist="clognormal", 
        base.dist = "Weibull", agemin = 20, ylim = c(0,1))
</code></pre>

<hr>
<h2 id='plot.penmodel'>
Plot method for <code>penmodel</code>
</h2><span id='topic+plot.penmodel'></span>

<h3>Description</h3>

<p>Plots penetrance curves estimated from the fitted penetrance model and overlays non-parametric penetrance curves estimated from the data without proabands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel'
plot(x, agemax = 80, print = TRUE, mark.time = FALSE, conf.int = FALSE, 
add.KM = TRUE, MC = 100, col = c("blue", "red", "blue", "red"), lty = c(1, 1, 2, 2), 
add.legend = TRUE, add.title = TRUE, xpos = "topleft", ypos = NULL, 
xlab = "Age at onset", ylab = "Penetrance", ylim = NULL, main = NULL,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.penmodel_+3A_x">x</code></td>
<td>

<p>An object class of <code>'penmodel'</code>, a fitted model by <code><a href="#topic+penmodel">penmodel</a></code> or <code><a href="#topic+penmodelEM">penmodelEM</a></code> functions.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 80 years of age.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_print">print</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays parameter estimates and penetrance estimates by age 70.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_mark.time">mark.time</code></td>
<td>
<p>Logical; if <code>TRUE</code>, curves are marked at each censoring time, otherwise, no labeling is done.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_conf.int">conf.int</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays 95% confidence intervals for both parametric and non-parametric penetrance estimates for each subgroup and returns their <code>lower</code> and <code>upper</code> limits.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_add.km">add.KM</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays Kaplan-Meier curves from data.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_mc">MC</code></td>
<td>

<p>Number of simulated samples used to calculate confidence intervals with a Monte-Carlo method. 
If <code>MC = 0</code>, no confidence intervals will be calculated. Default value is 100.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_col">col</code></td>
<td>

<p>Colors of lines for male carriers, female carriers, male noncarrers, and female noncarriers. Default is <code>c("blue", "red", "blue", "red")</code>.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_lty">lty</code></td>
<td>

<p>Types of lines for male carriers, female carriers, male noncarriers, and female noncarriers. Default is <code>c(1, 1, 2, 2)</code>.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_add.legend">add.legend</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays a legend in the plot.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_add.title">add.title</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays a title in the plot.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_xpos">xpos</code>, <code id="plot.penmodel_+3A_ypos">ypos</code></td>
<td>

<p>Position of legend; see <a href="graphics.html#topic+legend">legend</a>. Defaults are <code>xpos = "topleft", ypos = NULL</code>.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x-axis. Default is <code>"Age at onset"</code>.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y-axis. Default is <code>"Penetrance"</code>.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the y-axis. Default is <code>NULL</code>. If <code>NULL</code>, <code>ylim</code> will be automatically determined.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_main">main</code></td>
<td>
<p>Main title of the plot. Default is <code>NULL</code>. If <code>NULL</code>, the title will be automatically created.</p>
</td></tr>
<tr><td><code id="plot.penmodel_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 95% confidence intervals for the parametric penetrance curves are
obtained based on simulations of the parameters, assuming a multivariate normal distribution for the estimated
parameters with their variance-covariance matrix. See <code><a href="#topic+penetrance">penetrance</a></code> for more details.
</p>


<h3>Value</h3>

<p>Returns the following summary values:
</p>
<table>
<tr><td><code>coefficients</code></td>
<td>
<p>Parameter estimates of transformed baseline parameters (<code class="reqn">\lambda, \rho</code>) and regression coefficients for gender and mutation status (<code class="reqn">\beta_s, \beta_g</code>).</p>
</td></tr>
<tr><td><code>pen70</code></td>
<td>
<p> Penetrance estimates by age 70, specific to gender and mutation-status subgroups.</p>
</td></tr>
<tr><td><code>x.age</code></td>
<td>
<p>Vector of ages of onsest ranging from <code>agemin</code> to <code>agemax</code> years</p>
</td></tr>
<tr><td><code>pen</code></td>
<td>
<p>Penetrance estimates at each age in <code>x.age</code>, specific to gender and mutation-status subgroups.</p>
</td></tr>
<tr><td><code>lower</code></td>
<td>
<p>Lower limits of 95% confidence interval estimates for penetrance at each age in <code>x.age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
<tr><td><code>upper</code></td>
<td>
<p>Upper limits of 95% confidence interval estimates for penetrance at each age in <code>x.age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel">penmodel</a>, <a href="#topic+print.penmodel">print.penmodel</a>, <a href="#topic+penmodelEM">penmodelEM</a>, <a href="#topic+summary.penmodel">summary.penmodel</a>,<a href="#topic+print.summary.penmodel">print.summary.penmodel</a></code>, 
<code><a href="#topic+simfam">simfam</a>	</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulated family data  

set.seed(4321)
fam &lt;- simfam(N.fam = 300, design = "pop+", base.dist = "Weibull", variation = "none",
       base.parms = c(0.01,3), vbeta = c(-1.13, 2.35), allelefreq = 0.02, agemin = 20)
 
# Fit family data

fit &lt;- penmodel(Surv(time, status) ~ gender + mgene, cluster = "famID", design = "pop+",
parms = c(0.01, 3, -1.13, 2.35), data = fam, base.dist = "Weibull", robust = TRUE)       

# Plot penetrance function curves with 95% CIs

plot(fit, agemax = 80, conf.int = TRUE)

</code></pre>

<hr>
<h2 id='plot.penmodel_cmp'>
Plot method for <code>penmodel_cmp</code>
</h2><span id='topic+plot.penmodel_cmp'></span>

<h3>Description</h3>

<p>Plots penetrance curves for each event estimated from the fitted competing risks model and overlays non-parametric cumulative incidence curves estimated from the data without proabands.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel_cmp'
plot(x, agemax = 80, print = TRUE, conf.int = FALSE, 
add.CIF = TRUE, MC = 100, col = c("blue", "red", "blue", "red"), lty = c(1, 1, 2, 2), 
xlab = "Age at onset", ylab = "Penetrance", ylim = NULL,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.penmodel_cmp_+3A_x">x</code></td>
<td>

<p>An object class of <code>'penmodel'</code>, a fitted model by <code><a href="#topic+penmodel">penmodel</a></code> or <code><a href="#topic+penmodelEM">penmodelEM</a></code> functions.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 80 years of age.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_print">print</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays parameter estimates and penetrance estimates by age 70.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_conf.int">conf.int</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays 95% confidence intervals for both parametric and non-parametric penetrance estimates for each subgroup and returns their <code>lower</code> and <code>upper</code> limits.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_add.cif">add.CIF</code></td>
<td>
<p>Logical; if <code>TRUE</code>, displays cumulative incidence curves from competing event data.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_mc">MC</code></td>
<td>

<p>Number of simulated samples used to calculate confidence intervals with a Monte-Carlo method. 
If <code>MC = 0</code>, no confidence intervals will be calculated. Default value is 100.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_col">col</code></td>
<td>

<p>Colors of lines for male carriers, female carriers, male noncarrers, and female noncarriers. Default is <code>c("blue", "red", "blue", "red")</code>.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_lty">lty</code></td>
<td>

<p>Types of lines for male carriers, female carriers, male noncarriers, and female noncarriers. Default is <code>c(1, 1, 2, 2)</code>.
</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_xlab">xlab</code></td>
<td>
<p>Title for the x-axis. Default is <code>"Age at onset"</code>.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_ylab">ylab</code></td>
<td>
<p>Title for the y-axis. Default is <code>"Penetrance"</code>.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_ylim">ylim</code></td>
<td>
<p>Limits for the y-axis. Default is <code>NULL</code>. If <code>NULL</code>, <code>ylim</code> will be automatically determined.</p>
</td></tr>
<tr><td><code id="plot.penmodel_cmp_+3A_...">...</code></td>
<td>
<p>Other parameters to be passed through to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 95% confidence intervals for the parametric penetrance curves are
obtained based on simulations of the parameters, assuming a multivariate normal distribution for the estimated
parameters with their variance-covariance matrix. See <code><a href="#topic+penetrance_cmp">penetrance_cmp</a></code> for more details.
</p>


<h3>Value</h3>

<p>Returns the following summary values:
</p>
<table>
<tr><td><code>coefficients</code></td>
<td>
<p>Parameter estimates from the competing risks model.</p>
</td></tr>
<tr><td><code>pen70</code></td>
<td>
<p>Penetrance estimates by age 70, specific to gender and mutation-status subgroups.</p>
</td></tr>
<tr><td><code>age</code></td>
<td>
<p>Vector of ages of onsest ranging from <code>agemin</code> to <code>agemax</code> years</p>
</td></tr>
<tr><td><code>pen1</code></td>
<td>
<p>Penetrance estimates for event 1 at each age in <code>age</code>, specific to gender and mutation-status subgroups.</p>
</td></tr>
<tr><td><code>pen2</code></td>
<td>
<p>Penetrance estimates for event 2 at each age in <code>age</code>, specific to gender and mutation-status subgroups.</p>
</td></tr>
<tr><td><code>lower1</code></td>
<td>
<p>Lower limits of 95% confidence interval estimates for penetrance for event 1 at each age in <code>age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
<tr><td><code>upper1</code></td>
<td>
<p>Upper limits of 95% confidence interval estimates for penetrance for event 1 at each age in <code>x.age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
<tr><td><code>lower2</code></td>
<td>
<p>Lower limits of 95% confidence interval estimates for penetrance for event 2 at each age in <code>age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
<tr><td><code>upper2</code></td>
<td>
<p>Upper limits of 95% confidence interval estimates for penetrance for event 2 at each age in <code>x.age</code>, specific to gender and mutation status subgroups.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel_cmp">penmodel_cmp</a>, <a href="#topic+print.penmodel_cmp">print.penmodel_cmp</a>, <a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a>,
<a href="#topic+print.summary.penmodel_cmp">print.summary.penmodel_cmp</a>, <a href="#topic+simfam_cmp">simfam_cmp</a>	</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# Simulate family data  
set.seed(4321)
fam2 &lt;- simfam_cmp(N.fam = 500, design = "pop+", variation = "frailty", 
                  base.dist = "Weibull", frailty.dist = "cgamma", depend=c(2, 2, 2), 
                  allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
                  vbeta = list(c(-1.13, 2.35),c(-1, 2)))

# Fit family data
fit2 &lt;- penmodel_cmp(formula1 = Surv(time, status==1)~ gender + mgene, 
                   formula2 = Surv(time, status==2)~ gender + mgene, 
                   cluster = "famID", gvar = "mgene", frailty.dist = "cgamma",
                   parms=list(c(0.01, 3, -1, 2.3), c(0.01, 3, -1, 2), c(2, 2, 2)), 
                   data=fam2, design="pop+", base.dist="Weibull", robust=TRUE)

# Plot penetrance function curves with 95
plot(fit2, conf.int=TRUE, MC=200, ylim=c(0, 0.7))


## End(Not run)


</code></pre>

<hr>
<h2 id='plot.simfam'>Plot method for <code>simfam</code> or Plot pedigrees</h2><span id='topic+plot.simfam'></span>

<h3>Description</h3>

<p>Provides pedigree plots for specified families generated from <code>simfam</code> function with option to save plots into a pdf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam'
plot(x, famid, pdf = FALSE, file = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.simfam_+3A_x">x</code></td>
<td>
<p>An object of class <code>'simfam'</code> created by <code><a href="#topic+simfam">simfam</a></code> function or a data frame that has class attributes <code>c("simfam", "data.frame")</code>.
</p>
</td></tr>
<tr><td><code id="plot.simfam_+3A_famid">famid</code></td>
<td>
<p>List of family IDs to plot. Default is the first family in given data set.</p>
</td></tr>
<tr><td><code id="plot.simfam_+3A_pdf">pdf</code></td>
<td>
<p>Logical; if <code>TRUE</code>, pedigree plots are saved in a pdf file. If <code>FALSE</code>, plot pedigrees on current plotting device. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_+3A_file">file</code></td>
<td>
<p>File name to save the pedigree plots; Default file name is <code>"pedigreeplot.pdf"</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to the plot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Argument <code>x</code> can be a data frame that contains <code>famID</code>, <code>indID</code>, <code>fatherID</code>,  <code>motherID</code>, <code>gender</code> (1 for male, 0 for female), <code>status</code> (1 for affected, 0 for non-affected), <code>mgene</code> (1 for mutation carrier, 0 for non-carrier, <code>NA</code> for missing), and <code>proband</code> (1 for proband, 0 for non-proband) and should have class attributes <code>class(x) &lt;- c("simfam", "data.frame")</code>.
</p>
<p>Optionally, the data frame can contain a column named <code>carrp.geno</code> or <code>carrp.pheno</code> to replace missing values in <code>mgene</code> with their carrier probabilities.
</p>


<h3>Value</h3>

<p>Returns pedigree plots for specified families created by <code><a href="#topic+simfam">simfam</a></code> function or for the data frame provided along with the affection and carrier mutation statuses of family members. Probands from each pedigree are indicated using red color.
</p>
<p>When object inlcudes <code>carrp.geno</code> and/or <code>carrp.pheno</code> generated by <code><a href="#topic+carrierprob">carrierprob</a></code> function, the <code>plot</code> function displays the carrier probabilities for those with missing carrier status.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a>, <a href="#topic+summary.simfam">summary.simfam</a>, <a href="#topic+carrierprob">carrierprob</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulated family data

set.seed(4321)
fam &lt;- simfam(N.fam = 200, design = "pop+", base.dist = "Weibull", allelefreq = 0.02, 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), agemin = 20)
 
# Pedigree plots for first three simulated families

plot(fam, famid = c(1:3))

</code></pre>

<hr>
<h2 id='plot.simfam_cmp'>Plot method for <code>simfam_cmp</code> or Plot pedigrees</h2><span id='topic+plot.simfam_cmp'></span>

<h3>Description</h3>

<p>Provides pedigree plots for specified families generated from <code>simfam_cmp</code> function with option to save plots into a pdf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam_cmp'
plot(x, famid, pdf = FALSE, file = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.simfam_cmp_+3A_x">x</code></td>
<td>
<p>An object of class <code>'simfam'</code> created by <code><a href="#topic+simfam">simfam</a></code> function or a data frame that has class attributes <code>c("simfam", "data.frame")</code>.
</p>
</td></tr>
<tr><td><code id="plot.simfam_cmp_+3A_famid">famid</code></td>
<td>
<p>List of family IDs to plot. Default is the first family in given data set.</p>
</td></tr>
<tr><td><code id="plot.simfam_cmp_+3A_pdf">pdf</code></td>
<td>
<p>Logical; if <code>TRUE</code>, pedigree plots are saved in a pdf file. If <code>FALSE</code>, plot pedigrees on current plotting device. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_cmp_+3A_file">file</code></td>
<td>
<p>File name to save the pedigree plots; Default file name is <code>"pedigreeplot.pdf"</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_cmp_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to the plot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Argument <code>x</code> can be a data frame that contains <code>famID</code>, <code>indID</code>, <code>fatherID</code>,  <code>motherID</code>, <code>gender</code> (1 for male, 0 for female), <code>status</code> (1 for affected by event 1, 2 for affected by event 2, 0 for non-affected), <code>mgene</code> (1 for mutation carrier, 0 for non-carrier, <code>NA</code> for missing), and <code>proband</code> (1 for proband, 0 for non-proband) and should have class attributes <code>class(x) &lt;- c("simfam", "data.frame")</code>.
</p>
<p>Optionally, the data frame can contain a column named <code>carrp.geno</code> or <code>carrp.pheno</code> to replace missing values in <code>mgene</code> with their carrier probabilities.
</p>


<h3>Value</h3>

<p>Returns pedigree plots for specified families created by <code><a href="#topic+simfam_cmp">simfam_cmp</a></code> function or for the data frame provided along with the affection and carrier mutation statuses of family members. Probands from each pedigree are indicated using red color.
</p>
<p>When object inlcudes <code>carrp.geno</code> and/or <code>carrp.pheno</code> generated by <code><a href="#topic+carrierprob">carrierprob</a></code> function, the <code>plot</code> function displays the carrier probabilities for those with missing carrier status.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a>, <a href="#topic+summary.simfam">summary.simfam</a>, <a href="#topic+carrierprob">carrierprob</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulated competing risk data from gamma frailty model based on pop+ design

set.seed(4321)
fam &lt;- simfam_cmp(N.fam = 10, design = "pop+", variation = "frailty", 
                  base.dist = "Weibull", frailty.dist = "gamma", depend=c(2, 2), 
                  allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
                  vbeta = list(c(-1.13, 2.35),c(-1, 2)))

# Pedigree plots for first three simulated families

plot(fam, famid = 1)

</code></pre>

<hr>
<h2 id='plot.simfam_tvc'>Plot method for <code>simfam_tvc</code> or Plot pedigrees</h2><span id='topic+plot.simfam_tvc'></span>

<h3>Description</h3>

<p>Provides pedigree plots for specified families generated from <code>simfam_tvc</code> function with option to save plots into a pdf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam_tvc'
plot(x, famid, pdf = FALSE, file = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.simfam_tvc_+3A_x">x</code></td>
<td>
<p>An object of class <code>'simfam_tvc'</code> created by <code><a href="#topic+simfam_tvc">simfam_tvc</a></code> function or a data frame that has class attributes <code>c("simfam_tvc", "data.frame")</code>.
</p>
</td></tr>
<tr><td><code id="plot.simfam_tvc_+3A_famid">famid</code></td>
<td>
<p>List of family IDs to plot. Default is the first family in given data set.</p>
</td></tr>
<tr><td><code id="plot.simfam_tvc_+3A_pdf">pdf</code></td>
<td>
<p>Logical; if <code>TRUE</code>, pedigree plots are saved in a pdf file. If <code>FALSE</code>, plot pedigrees on current plotting device. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_tvc_+3A_file">file</code></td>
<td>
<p>File name to save the pedigree plots; Default file name is <code>"pedigreeplot.pdf"</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam_tvc_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to the plot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Argument <code>x</code> can be a data frame that contains <code>famID</code>, <code>indID</code>, <code>fatherID</code>,  <code>motherID</code>, <code>gender</code> (1 for male, 0 for female), <code>status</code> (1 for affected, 0 for non-affected), <code>mgene</code> (1 for mutation carrier, 0 for non-carrier, <code>NA</code> for missing), and <code>proband</code> (1 for proband, 0 for non-proband) and should have class attributes <code>class(x) &lt;- c("simfam", "data.frame")</code>.
</p>
<p>Optionally, the data frame can contain a column named <code>carrp.geno</code> or <code>carrp.pheno</code> to replace missing values in <code>mgene</code> with their carrier probabilities.
</p>


<h3>Value</h3>

<p>Returns pedigree plots for specified families created by <code><a href="#topic+simfam_tvc">simfam_tvc</a></code> function or for the data frame provided along with the affection and carrier mutation statuses of family members. Probands from each pedigree are indicated using red color.
</p>
<p>When object inlcudes <code>carrp.geno</code> and/or <code>carrp.pheno</code> generated by <code><a href="#topic+carrierprob">carrierprob</a></code> function, the <code>plot</code> function displays the carrier probabilities for those with missing carrier status.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_tvc">simfam_tvc</a>, <a href="#topic+summary.simfam_tvc">summary.simfam_tvc</a>, <a href="#topic+carrierprob">carrierprob</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Simulated family data
set.seed(4321)
fam &lt;- simfam_tvc(N.fam = 10, design = "pop", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "gamma", depend = 1, 
       add.tvc = TRUE, tvc.type = "CO", tvc.range = c(30,60), 
       tvc.parms = c(1, 0.1, 0), allelefreq = 0.02, 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))
 
# Pedigree plots for first three simulated families

plot(fam, famid = c(1:2))

</code></pre>

<hr>
<h2 id='plot.simfam2'>Plot method for <code>simfam2</code> or Plot pedigrees</h2><span id='topic+plot.simfam2'></span>

<h3>Description</h3>

<p>Provides pedigree plots for specified families generated from <code>simfam2</code> function with option to save plots into a pdf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam2'
plot(x, famid, pdf = FALSE, file = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.simfam2_+3A_x">x</code></td>
<td>
<p>An object of class <code>'simfam2'</code> created by <code><a href="#topic+simfam2">simfam2</a></code> function or a data frame that has class attributes <code>c("simfam2", "data.frame")</code>.
</p>
</td></tr>
<tr><td><code id="plot.simfam2_+3A_famid">famid</code></td>
<td>
<p>List of family IDs to plot. Default is the first family in given data set.</p>
</td></tr>
<tr><td><code id="plot.simfam2_+3A_pdf">pdf</code></td>
<td>
<p>Logical; if <code>TRUE</code>, pedigree plots are saved in a pdf file. If <code>FALSE</code>, plot pedigrees on current plotting device. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam2_+3A_file">file</code></td>
<td>
<p>File name to save the pedigree plots; Default file name is <code>"pedigreeplot.pdf"</code>.</p>
</td></tr>
<tr><td><code id="plot.simfam2_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to the plot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Argument <code>x</code> can be a data frame that contains <code>famID</code>, <code>indID</code>, <code>fatherID</code>,  <code>motherID</code>, <code>gender</code> (1 for male, 0 for female), <code>status</code> (1 for affected, 0 for non-affected), <code>mgene</code> (1 for mutation carrier, 0 for non-carrier, <code>NA</code> for missing), and <code>proband</code> (1 for proband, 0 for non-proband) and should have class attributes <code>class(x) &lt;- c("simfam", "data.frame")</code>.
</p>
<p>Optionally, the data frame can contain a column named <code>carrp.geno</code> or <code>carrp.pheno</code> to replace missing values in <code>mgene</code> with their carrier probabilities.
</p>


<h3>Value</h3>

<p>Returns pedigree plots for specified families created by <code><a href="#topic+simfam2">simfam2</a></code> function or for the data frame provided along with the affection and carrier mutation statuses of family members. Probands from each pedigree are indicated using red color.
</p>
<p>When object inlcudes <code>carrp.geno</code> and/or <code>carrp.pheno</code> generated by <code><a href="#topic+carrierprob">carrierprob</a></code> function, the <code>plot</code> function displays the carrier probabilities for those with missing carrier status.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam2">simfam2</a>, <a href="#topic+summary.simfam2">summary.simfam2</a>, <a href="#topic+carrierprob">carrierprob</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(4321)

data &lt;- simfam(N.fam = 10, design = "noasc", variation = "none",
         base.dist = "Weibull", base.parms = c(0.016, 3), vbeta = c(1, 1))

IBDmatrix &lt;- diag(1, dim(data)[1])
data &lt;- data[ , c(1:7, 11, 14)]

fam2 &lt;- simfam2(inputdata = data, IBD = IBDmatrix, design = "pop", 
        variation = c("kinship","IBD"), depend = c(1, 1), 
        base.dist = "Weibull", base.parms = c(0.016, 3),
        var_names = c("gender", "mgene"), vbeta = c(1,1),
        agemin=20) 

plot(fam2, famid = c(1:2))

</code></pre>

<hr>
<h2 id='print.penmodel'>
Print method for <code>penmodel</code>.
</h2><span id='topic+print.penmodel'></span>

<h3>Description</h3>

<p>Prints a summary of parameter estimates of a fitted penetrance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.penmodel_+3A_x">x</code></td>
<td>

<p>An object class of <code>'penmodel'</code>, a fitted model by <code><a href="#topic+penmodel">penmodel</a></code> or <code><a href="#topic+penmodelEM">penmodelEM</a></code> functions.
</p>
</td></tr>
<tr><td><code id="print.penmodel_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.penmodel_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a short summary of the model and model fit.
</p>
<p>Returns an object of class <code>'penmodel'</code>.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel">penmodel</a>, <a href="#topic+penmodelEM">penmodelEM</a>, <a href="#topic+summary.penmodel">summary.penmodel</a>, <a href="#topic+print.summary.penmodel">print.summary.penmodel</a>, <a href="#topic+plot.penmodel">plot.penmodel</a></code>
</p>

<hr>
<h2 id='print.penmodel_cmp'>
Print method for <code>penmodel_cmp</code>.
</h2><span id='topic+print.penmodel_cmp'></span>

<h3>Description</h3>

<p>Prints a summary of parameter estimates of a fitted competing risk penetrance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel_cmp'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.penmodel_cmp_+3A_x">x</code></td>
<td>

<p>An object class of <code>'penmodel_cmp'</code>, a fitted model by <code><a href="#topic+penmodel_cmp">penmodel_cmp</a></code> function.
</p>
</td></tr>
<tr><td><code id="print.penmodel_cmp_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.penmodel_cmp_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a short summary of the model and model fit.
</p>
<p>Returns an object of class <code>'penmodel_cmp'</code>.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel_cmp">penmodel_cmp</a>, <a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a>, <a href="#topic+print.summary.penmodel_cmp">print.summary.penmodel_cmp</a>, <a href="#topic+plot.penmodel_cmp">plot.penmodel_cmp</a></code>
</p>

<hr>
<h2 id='print.summary.penmodel'>
Print method for <code>summary.penmodel</code> of a fitted penetrance model.
</h2><span id='topic+print.summary.penmodel'></span>

<h3>Description</h3>

<p>Prints a short summary of parameter and penetrance estimates of a <code>'summary.penmodel'</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.penmodel'
print(x, digits = max(3, getOption("digits") - 3),
  signif.stars=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.summary.penmodel_+3A_x">x</code></td>
<td>

<p>An object class of <code>'summary.penmodel'</code>, a result of a call to <code><a href="#topic+summary.penmodel">summary.penmodel</a></code>.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_+3A_signif.stars">signif.stars</code></td>
<td>
<p>Logical; if <code>TRUE</code>, provides stars to hightlight significant p-values. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a summary of parameter estimates, their standard errors, <code class="reqn">t</code>-statistics and corresponding two-sided <code class="reqn">p</code>-values and additionally indicates significance stars if <code>signif.stars</code> is <code>TRUE</code>.
</p>
<p>Also prints penetrance estimates by age 70 specific to gender and mutation-status subgroups along with their standard errors and 95% confidence intervals.
</p>
<p>Returns an object of class <code>'summary.penmodel'</code>.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel">penmodel</a>, <a href="#topic+penmodelEM">penmodelEM</a>, <a href="#topic+print.penmodel">print.penmodel</a>, <a href="#topic+summary.penmodel">summary.penmodel</a></code>
</p>

<hr>
<h2 id='print.summary.penmodel_cmp'>
Print method for <code>summary.penmodel_cmp</code> of a fitted competing risks penetrance model.
</h2><span id='topic+print.summary.penmodel_cmp'></span>

<h3>Description</h3>

<p>Prints a short summary of parameter and penetrance estimates of a <code>'summary.penmodel_cmp'</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.penmodel_cmp'
print(x, digits = max(3, getOption("digits") - 3),
  signif.stars=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.summary.penmodel_cmp_+3A_x">x</code></td>
<td>

<p>An object class of <code>'summary.penmodel_cmp'</code>, a result of a call to <code><a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a></code>.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_cmp_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_cmp_+3A_signif.stars">signif.stars</code></td>
<td>
<p>Logical; if <code>TRUE</code>, provides stars to hightlight significant p-values. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="print.summary.penmodel_cmp_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a summary of parameter estimates, their standard errors, <code class="reqn">t</code>-statistics and corresponding two-sided <code class="reqn">p</code>-values and additionally indicates significance stars if <code>signif.stars</code> is <code>TRUE</code>.
</p>
<p>Also prints penetrance estimates for each event by age 70 specific to gender and mutation-status subgroups along with their standard errors and 95% confidence intervals.
</p>
<p>Returns an object of class <code>'summary.penmodel_cmp'</code>.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel_cmp">penmodel_cmp</a>, <a href="#topic+print.penmodel_cmp">print.penmodel_cmp</a>, <a href="#topic+summary.penmodel_cmp">summary.penmodel_cmp</a></code>
</p>

<hr>
<h2 id='simfam'>
Generate familial time-to-event data
</h2><span id='topic+simfam'></span>

<h3>Description</h3>

<p>Generates familial time-to-event data for specified study design, genetic model and source of residual familial correlation; the generated data frame also contains family structure (individual's id, father id, mother id, relationship to proband, generation), gender, current age, genotypes of major or second genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simfam(N.fam, design = "pop", variation = "none", interaction = FALSE, 
       add.x = FALSE, x.dist = NULL, x.parms = NULL, depend = NULL, 
       base.dist = "Weibull", frailty.dist = NULL, base.parms = c(0.016, 3), 
       vbeta = c(1, 1), 
       allelefreq = 0.02, dominant.m = TRUE, dominant.s = TRUE,
       mrate = 0, hr = 0, probandage = c(45, 2), agemin = 20, agemax = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simfam_+3A_n.fam">N.fam</code></td>
<td>

<p>Number of families to generate.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_design">design</code></td>
<td>

<p>Family based study design used in the simulations. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code>, <code>"twostage"</code>, or <code>"noasc"</code>, where <code>"pop"</code> is for the population-based design that families are ascertained by affected probands, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands,  <code>"twostage"</code> for two-stage design that randomly samples families from the population in the first stage and oversamples high risk families in the second stage that include at least two affected members in the family, and <code>"noasc"</code> for no ascertainment correction that families are from simple random sampling. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, <code>"kinship"</code> for correlated frailties within families with kinship matrix, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_interaction">interaction</code></td>
<td>

<p>Logical; if <code>TRUE</code>, allows the interaction between gender and mutation status. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_add.x">add.x</code></td>
<td>

<p>Logical; if <code>TRUE</code>, generates a covariate in addition to gender and majorgene. Default is <code>FALSE</code>. 
</p>
</td></tr>
<tr><td><code id="simfam_+3A_x.dist">x.dist</code></td>
<td>

<p>Distribution of the covairate. Possible choices to generate the covariate are: <code>"normal"</code> from normal distribution and <code>"binomial"</code> from binominal distribution. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_x.parms">x.parms</code></td>
<td>

<p>Parameter values for the specified distribution of the covariate. <code>x.parms = c(mean, sd)</code> should be specified for <code>x.dist = "normal"</code> and <code>x.parms = c(size, probs)</code> should be specified for <code>"binomial"</code>; when <code>size = 1</code>, it generates binary covariate. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_depend">depend</code></td>
<td>

<p>Inverse of variance of the frailty distribution. Dependence within families decreases with depend value.  Default is <code>NULL</code>. Value should be specified as a positive real number when <code>variation = "frailty"</code> or <code>variation = "kinship"</code>. 
</p>
</td></tr>
<tr><td><code id="simfam_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution.  Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code> <code>"gamma"</code>, <code>"logBurr"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution.  Possible choices are: <code>"gamma"</code> or <code>"lognormal"</code> when <code>variation="frailty"</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_base.parms">base.parms</code></td>
<td>

<p>Vector of parameter values for the specified baseline hazard function. <code>base.parms = c(lambda, rho)</code> should be specified for <code>base.dist = "Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"gamma"</code>, and <code>"lognormal"</code>. For <code>base.dist = "logBurr"</code>, three parameters should be specified <code>base.parms = c(lambda, rho, eta)</code>. 
Default value is <code>base.parms = c(0.016, 3)</code> for <code>base.dist = "Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_vbeta">vbeta</code></td>
<td>

<p>Vector of regression coefficients for gender, majorgene, interaction between gender and majorgene (if <code>interaction = TRUE</code>), secondgene (if <code>variation = "secondgene"</code>), and additional covariate (if <code>add.x = TRUE</code>).</p>
</td></tr>
<tr><td><code id="simfam_+3A_allelefreq">allelefreq</code></td>
<td>

<p>Population allele frequencies of major disease gene. Value should be between 0 and 1. 
Vector of population allele frequencies of major and second disease genes should be provided when <code>variation = "secondgene"</code>. Default value is <code>allelefreq = 0.02</code>.</p>
</td></tr>
<tr><td><code id="simfam_+3A_dominant.m">dominant.m</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of major gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_+3A_dominant.s">dominant.s</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of second gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_+3A_mrate">mrate</code></td>
<td>

<p>Proportion of missing genotypes, value between 0 and 1. Default value is 0.</p>
</td></tr>
<tr><td><code id="simfam_+3A_hr">hr</code></td>
<td>

<p>Proportion of high risk families, which include at least two affected members, to be sampled from the two stage sampling. This value should be specified when <code>design="twostage"</code>. Default value is 0. Value should lie between 0 and 1.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_probandage">probandage</code></td>
<td>

<p>Vector of mean and standard deviation for the proband age. Default values are mean of 45 years and standard deviation of 2 years, <code>probandage = c(45, 2)</code>.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="simfam_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 100 years of age.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>design</code> argument defines the type of family based design to be simulated. Two variants of the population-based and clinic-based design can be chosen: <code>"pop"</code> when proband is affected, <code>"pop+"</code> when proband is affected mutation carrier, <code>"cli"</code> when proband is affected and at least one parent and one sibling are affected, <code>"cli+"</code> when proband is affected mutation-carrier and at least one parent and one sibling are affected. The two-stage design, <code>"twostage"</code>, is used to oversample high risk families, where the proportion of high risks families to include in the sample is specified by <code>hr</code>. High risk families often include multiple (at least two) affected members in the family. <code>design = "noasc"</code> is to be used for no ascertainment correction.
</p>
<p>The ages at onset are generated from the following penetrance models depending on the choice of <code>variation = "none", "frailty", "secondgene", "kinship".</code>. When <code>variation = "none"</code>, the ages at onset are independently generated from the proportional hazard model conditional on the gender and carrier status of major gene mutation,   X = c(x<sub>s</sub>, x<sub>g</sub>).  
The ages at onset correlated within families are generated from the shared frailty model (<code>variation = "frailty"</code>) , the correlated shared frailty model with kinship matrix (<code>variation = "kinship"</code>), or the two-gene model (<code>variation = "secondene"</code>), where the residual familial correlation is induced by a frailty or a second gene, respectively, shared within the family.
</p>
<p><em>The proportional hazard model</em>
</p>
 <p style="text-align:center"> h(t|X) = h<sub>0</sub>(t - t<sub>0</sub>) exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The shared frailty model</em>
</p>
 <p style="text-align:center"> h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, <code class="reqn">Z</code> represents a frailty shared within families and follows either a gamma or log-normal distribution, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The correlated shared frailty model with kinship matrix</em>
</p>
 <p style="text-align:center"> h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, <code class="reqn">Z</code> represents a vector of frailties following a multivariate log-normal distribution with mean <code class="reqn">0</code> and variance <code class="reqn">2*K/depend</code>, where <code class="reqn">K</code> represents the kinship matrix, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The two-gene model</em>
</p>
 <p style="text-align:center"> h(t|X) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>1</sub> * x<sub>2</sub> + &beta;<sub>2</sub> * x<sub>2</sub>), </p>
<p>where  x<sub>1</sub>, x<sub>2</sub> indicate carriers (1) and non-carriers (0) of a major gene and of second gene mutation, respectively.
</p>
<p>The current ages for each generation are simulated assuming normal distributions. However, the probands' ages are generated using a left truncated normal distribution as their ages cannot be less than the minimum age of onset. The average age difference between each generation and their parents is specified as 20 years apart. 
</p>
<p>Note that simulating family data under the clinic-based designs (<code>"cli"</code> or <code>"cli+"</code>) or the two-stage design can be slower since the ascertainment criteria for the high risk families are difficult to meet in such settings. Especially, <code>"cli"</code> design could be slower than <code>"cli+"</code> design since the proband's mutation status is randomly selected from a disease population in <code>"cli"</code> design, so his/her family members are less likely to be mutation carriers and have less chance to be affected, whereas the probands are all mutation carriers, their family members have higher chance to be carriers and affected by disease. Therefore, <code>"cli"</code> design requires more iterations to sample high risk families than <code>"cli+"</code> design. 
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'simfam'</code>, a data frame which contains:
</p>
<table>
<tr><td><code>famID</code></td>
<td>
<p>Family identification (ID) numbers.</p>
</td></tr>
<tr><td><code>indID</code></td>
<td>
<p>Individual ID numbers.</p>
</td></tr>
<tr><td><code>gender</code></td>
<td>
<p>Gender indicators: 1 for males, 0 for females.</p>
</td></tr>
<tr><td><code>motherID</code></td>
<td>
<p>Mother ID numbers.</p>
</td></tr>
<tr><td><code>fatherID</code></td>
<td>
<p>Father ID numbers.</p>
</td></tr>
<tr><td><code>proband</code></td>
<td>
<p>Proband indicators: 1 if the individual is the proband, 0 otherwise.</p>
</td></tr>
<tr><td><code>generation</code></td>
<td>
<p>Individuals generation: 1=parents of probands,2=probands and siblings, 3=children of probands and siblings.</p>
</td></tr>
<tr><td><code>majorgene</code></td>
<td>
<p>Genotypes of major gene: 1=AA, 2=Aa, 3=aa where A is disease gene.</p>
</td></tr>
<tr><td><code>secondgene</code></td>
<td>
<p>Genotypes of second gene: 1=BB, 2=Bb, 3=bb where B is disease gene.</p>
</td></tr>
<tr><td><code>ageonset</code></td>
<td>
<p>Ages at disease onset in years.</p>
</td></tr>
<tr><td><code>currentage</code></td>
<td>
<p>Current ages in years.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Ages at disease onset for the affected or ages of last follow-up for the unaffected. </p>
</td></tr>
<tr><td><code>status</code></td>
<td>
<p>Disease statuses: 1 for affected, 0 for unaffected (censored).</p>
</td></tr>
<tr><td><code>mgene</code></td>
<td>
<p>Major gene mutation indicators: 1 for mutated gene carriers, 0 for mutated gene noncarriers, or <code>NA</code> if missing.</p>
</td></tr>
<tr><td><code>relation</code></td>
<td>
<p>Family members' relationship with the proband:
</p>

<table>
<tr>
 <td style="text-align: left;">
    1  </td><td style="text-align: left;"> Proband (self) </td>
</tr>
<tr>
 <td style="text-align: left;">
    2  </td><td style="text-align: left;"> Brother or sister </td>
</tr>
<tr>
 <td style="text-align: left;">
    3  </td><td style="text-align: left;"> Son or daughter </td>
</tr>
<tr>
 <td style="text-align: left;">
    4  </td><td style="text-align: left;"> Parent </td>
</tr>
<tr>
 <td style="text-align: left;">
    5  </td><td style="text-align: left;"> Nephew or niece </td>
</tr>
<tr>
 <td style="text-align: left;">
    6  </td><td style="text-align: left;"> Spouse </td>
</tr>
<tr>
 <td style="text-align: left;">
    7  </td><td style="text-align: left;"> Brother or sister in law
  </td>
</tr>

</table>

</td></tr>
<tr><td><code>fsize</code></td>
<td>
<p>Family size including parents, siblings and children of the proband and the siblings.</p>
</td></tr>
<tr><td><code>naff</code></td>
<td>
<p>Number of affected members in family.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Sampling weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi, Wenqing He
</p>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.simfam">summary.simfam</a>, <a href="#topic+plot.simfam">plot.simfam</a>, <a href="#topic+penplot">penplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Example 1: simulate family data from a population-based design using
#  a Weibull distribution for the baseline hazard and inducing 
#  residual familial correlation through a shared gamma frailty.

set.seed(4321)
fam &lt;- simfam(N.fam = 10, design = "pop+", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "gamma", depend = 1, 
       allelefreq = 0.02, base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))

head(fam) 

## Not run: 
  famID indID gender motherID fatherID proband generation majorgene secondgene
1     1     1      1        0        0       0          1         2          0
2     1     2      0        0        0       0          1         2          0
3     1     3      0        2        1       1          2         2          0
4     1     4      1        0        0       0          0         3          0
5     1     9      0        3        4       0          3         2          0
6     1    10      1        3        4       0          3         3          0
   ageonset currentage     time status mgene relation fsize naff weight
1 103.76925   69.19250 69.19250      0     1        4    18    2      1
2  64.88982   67.31119 64.88982      1     1        4    18    2      1
3  45.84891   47.57119 45.84891      1     1        1    18    2      1
4 269.71990   47.37403 47.37403      0     0        6    18    2      1
5  69.78355   27.80081 27.80081      0     1        3    18    2      1
6 192.09392   25.34148 25.34148      0     0        3    18    2      1

## End(Not run)

summary(fam)

plot(fam, famid = c(1:2)) # pedigree plots for families with IDs = 1 and 2

## Example 2: simulate family data from a two-stage design to include 
#  30% of high risk families in the sample. 

set.seed(4321)
fam &lt;- simfam(N.fam = 50, design = "twostage", variation = "none", base.dist = "Weibull", 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), hr = 0.3, allelefreq = 0.02)

summary(fam)

## Example 3: simulate family data from a correlated frailty model with kinship matrix 

set.seed(4321)
fam &lt;- simfam(N.fam = 50, design = "pop", variation = "kinship", base.dist = "Weibull", 
       frailty.dist = "lognormal", base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), 
       depend = 1, allelefreq = 0.02)

summary(fam)

</code></pre>

<hr>
<h2 id='simfam_cmp'>
Generate familial competing risks data
</h2><span id='topic+simfam_cmp'></span>

<h3>Description</h3>

<p>Generates familial competing risks data for specified study design, genetic model and source of residual familial correlation; the generated data frame has the same family structure as that <code>simfam</code> function, including individual's id, father id, mother id, relationship to proband, generation, gender, current age, genotypes of major or second genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simfam_cmp(N.fam, design = "pop+", variation = "none", interaction = FALSE, 
         depend = NULL, base.dist = c("Weibull", "Weibull"), frailty.dist = "none", 
         base.parms = list(c(0.016, 3), c(0.016, 3)), 
         vbeta = list(c(-1.13, 2.35), c(-1, 2)), allelefreq = 0.02, dominant.m = TRUE, 
         dominant.s = TRUE, mrate = 0, hr = 0, probandage = c(45, 2), 
         agemin = 20, agemax = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simfam_cmp_+3A_n.fam">N.fam</code></td>
<td>

<p>Number of families to generate.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_design">design</code></td>
<td>

<p>Family based study design used in the simulations. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code> or <code>"twostage"</code>, where <code>"pop"</code> is for the population-based design that families are ascertained by affected probands, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands and <code>"twostage"</code> for two-stage design that randomly samples families from the population in the first stage and oversamples high risk families in the second stage that include at least two affected members in the family. Default is <code>"pop+"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_interaction">interaction</code></td>
<td>

<p>Logical; if <code>TRUE</code>, allows the interaction between gender and mutation status. Two logical values should be specified for each competing event; if only one logical value is provided, the same logical value will be assumed for both events. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_depend">depend</code></td>
<td>

<p>Two values shoud be specified for each competing event when <code>frailty.dist = "gamma"</code> or <code>frailty.dist = "lognormal"</code>, three values should be specified with <code>frailty.dist = "cgamma"</code> or <code>frailty.dist = "clognormal"</code>. The first two values represent the inverse of the variance for each competing event and the third value represents the correlation between the two events.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution.  Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code> <code>"gamma"</code>, <code>"logBurr"</code>. Default is <code>"Weibull"</code>. Two distributions should be specified for each competing event. If only one distribution is specified, the same distribution will be assumed for both events.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution. Possible choices are <code>"gamma"</code> for independent gamma, <code>"lognormal"</code> for independent lognormal, <code>"cgamma"</code> for correlated gamma, or <code>"clognormal"</code> for correlated lognormal distribution. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_base.parms">base.parms</code></td>
<td>

<p>The list of two vectors of baseline parameters for each event should be specified. For example, <code>base.parms=list(c(lambda1, rho1), c(lambda2, rho2))</code> should be specified for <code>base.dist=c("Weibull", "Weibull")</code>.
</p>
<p>Two parameters <code>base.parms=c(lambda, rho)</code> should be specified for <code>base.dist="Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"gamma"</code>, and <code>"lognormal"</code>, and three parameters should be specified <code>base.parms = c(lambda, rho, eta)</code> for <code>base.dist="logBurr"</code>. 
</p>
</td></tr>  	
<tr><td><code id="simfam_cmp_+3A_vbeta">vbeta</code></td>
<td>

<p>List of two vectors of regression coefficients for each event should be specified. 
Each vector contains regression coefficients for gender, majorgene, interaction between gender and majorgene (if <code>interaction = TRUE</code>), and secondgene (if <code>variation = "secondgene"</code>). 
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_allelefreq">allelefreq</code></td>
<td>

<p>Population allele frequencies of major disease gene. Value should be between 0 and 1. 
Vector of population allele frequencies for major and second disease genes should be provided when <code>variation = "secondgene"</code>. Default value is <code>allelefreq = 0.02</code>.</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_dominant.m">dominant.m</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of major gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_dominant.s">dominant.s</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of second gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_mrate">mrate</code></td>
<td>

<p>Proportion of missing genotypes, value between 0 and 1. Default value is 0.</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_hr">hr</code></td>
<td>

<p>Proportion of high risk families, which include at least two affected members, to be sampled from the two stage sampling. This value should be specified when <code>design="twostage"</code>. Default value is 0. Value should lie between 0 and 1.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_probandage">probandage</code></td>
<td>

<p>Vector of mean and standard deviation for the proband age. Default values are mean of 45 years and standard deviation of 2 years, <code>probandage = c(45, 2)</code>.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="simfam_cmp_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 100 years of age.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Competing risk model</em>
</p>
<p>Event 1:
</p>

<p style="text-align:center"> h<sub>1</sub>(t|X,Z) = h<sub>01</sub>(t - t<sub>0</sub>) Z<sub>1</sub> exp(&beta;<sub>s1</sub> * x<sub>s</sub> + &beta;<sub>g1</sub> * x<sub>g</sub>), </p>
<p>Event 2:
</p>

<p style="text-align:center"> h<sub>2</sub>(t|X,Z) = h<sub>02</sub>(t - t<sub>0</sub>) Z<sub>2</sub> exp(&beta;<sub>s2</sub> * x<sub>s</sub> + &beta;<sub>g2</sub> * x<sub>g</sub>), </p>
<p>where h<sub>01</sub>(t) and h<sub>02</sub>(t) are the baseline hazard functions for event 1 and event 2, respectively, t<sub>0</sub> is a minimum age of disease onset, Z<sub>1</sub> and Z<sub>2</sub> are frailties shared within families for each event and follow either a gamma, log-normal, correlateg gamma, or correlated log-normal distributions, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>Choice of frailty distributions for competing risk models</em>
</p>
<p><code>frailty.dist = "gamma"</code> shares the frailties within families generated from a gamma distribution independently for each competing event, where
Z<sub>j</sub> follows Gamma(k<sub>j</sub>, 1/k<sub>j</sub>). 
</p>
<p><code>frailty.dist = "lognormal"</code> shares the frailties within families generated from a log-normal distribution independently for each competing event, where 
Z<sub>j</sub> follows log-normal distribution with mean 0 and variance (1/k<sub>j</sub>.
</p>
<p><code>frailty.dist = "cgamma"</code> shares the frailties within families generated from a correlated gamma distribution to allow the frailties between two events to be correlated, where the correlated gamma frailties (Z<sub>1</sub>, Z<sub>2</sub>) are generated with three independent gamma frailties (Y<sub>0</sub>, Y<sub>1</sub>, Y<sub>2</sub>) as follows:
</p>
Z<sub>1</sub> = k<sub>0</sub>/(k<sub>0</sub> + k<sub>1</sub>) Y<sub>0</sub> + Y<sub>1</sub> 
Z<sub>2</sub> = k<sub>0</sub>/(k<sub>0</sub> + k<sub>2</sub>) Y<sub>0</sub> + Y<sub>2</sub> 
<p>where Y<sub>0</sub> from Gamma(k<sub>0</sub>, 1/k<sub>0</sub>);
</p>
Y<sub>1</sub><p> from Gamma(k<sub>1</sub>, 1/(k<sub>0</sub> + k<sub>1</sub>));
</p>
Y<sub>2</sub><p> from Gamma(k<sub>2</sub>, 1/(k<sub>0</sub> + k<sub>2</sub>)).
</p>
<p><code>frailty.dist = "clognormal"</code> shares the frailties within families generated from a correlated log-normal distribution where 
log(Z<sub>j</sub>) follows  a normal distribution with mean 0, variance 1/k<sub>j</sub> and correlation between two events k<sub>0</sub>.
</p>
<p><code>depend</code> should specify the values of related frailty parameters: <code>c(k1, k2)</code> with <code>frailty.dist = "gamma"</code> or <code>frailty.dist = "lognormal"</code>; <code>c(k1, k2, k0)</code> for <code>frailty.dist = "cgamma"</code> or <code>frailty.dist = "clognormal"</code>. 
</p>
<p>The current ages for each generation are simulated assuming normal distributions. However, the probands' ages are generated using a left truncated normal distribution as their ages cannot be less than the minimum age of onset. The average age difference between each generation and their parents is specified as 20 years apart. 
</p>
<p>The <code>design</code> argument defines the type of family based design to be simulated. Two variants of the population-based and clinic-based design can be chosen: <code>"pop"</code> when proband is affected, <code>"pop+"</code> when proband is affected mutation carrier, <code>"cli"</code> when proband is affected and at least one parent and one sibling are affected, <code>"cli+"</code> when proband is affected mutation-carrier and at least one parent and one sibling are affected. The two-stage design, <code>"twostage"</code>, is used to oversample high risk families, where the proportion of high risks families to include in the sample is specified by <code>hr</code>. High risk families often include multiple (at least two) affected members in the family.
</p>
<p>Note that simulating family data under the clinic-based designs (<code>"cli"</code> or <code>"cli+"</code>) or the two-stage design can be slower since the ascertainment criteria for the high risk families are difficult to meet in such settings. Especially, <code>"cli"</code> design could be slower than <code>"cli+"</code> design since the proband's mutation status is randomly selected from a disease population in <code>"cli"</code> design, so his/her family members are less likely to be mutation carriers and have less chance to be affected, whereas the probands are all mutation carriers, their family members have higher chance to be carriers and affected by disease. Therefore, <code>"cli"</code> design requires more iterations to sample high risk families than <code>"cli+"</code> design. 
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'simfam'</code>, a data frame which contains:
</p>
<table>
<tr><td><code>famID</code></td>
<td>
<p>Family identification (ID) numbers.</p>
</td></tr>
<tr><td><code>indID</code></td>
<td>
<p>Individual ID numbers.</p>
</td></tr>
<tr><td><code>gender</code></td>
<td>
<p>Gender indicators: 1 for males, 0 for females.</p>
</td></tr>
<tr><td><code>motherID</code></td>
<td>
<p>Mother ID numbers.</p>
</td></tr>
<tr><td><code>fatherID</code></td>
<td>
<p>Father ID numbers.</p>
</td></tr>
<tr><td><code>proband</code></td>
<td>
<p>Proband indicators: 1 if the individual is the proband, 0 otherwise.</p>
</td></tr>
<tr><td><code>generation</code></td>
<td>
<p>Individuals generation: 1=parents of probands,2=probands and siblings, 3=children of probands and siblings.</p>
</td></tr>
<tr><td><code>majorgene</code></td>
<td>
<p>Genotypes of major gene: 1=AA, 2=Aa, 3=aa where A is disease gene.</p>
</td></tr>
<tr><td><code>secondgene</code></td>
<td>
<p>Genotypes of second gene: 1=BB, 2=Bb, 3=bb where B is disease gene.</p>
</td></tr>
<tr><td><code>ageonset</code></td>
<td>
<p>Ages at disease onset in years.</p>
</td></tr>
<tr><td><code>currentage</code></td>
<td>
<p>Current ages in years.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Ages at disease onset for the affected or ages of last follow-up for the unaffected. </p>
</td></tr>
<tr><td><code>status</code></td>
<td>
<p>Disease statuses: 1 for affected by event 1, 2 for affected by event 2, 0 for unaffected (censored).
</p>
</td></tr>
<tr><td><code>mgene</code></td>
<td>
<p>Major gene mutation indicators: 1 for mutated gene carriers, 0 for mutated gene noncarriers, or <code>NA</code> if missing.</p>
</td></tr>
<tr><td><code>relation</code></td>
<td>
<p>Family members' relationship with the proband:
</p>

<table>
<tr>
 <td style="text-align: left;">
    1  </td><td style="text-align: left;"> Proband (self) </td>
</tr>
<tr>
 <td style="text-align: left;">
    2  </td><td style="text-align: left;"> Brother or sister </td>
</tr>
<tr>
 <td style="text-align: left;">
    3  </td><td style="text-align: left;"> Son or daughter </td>
</tr>
<tr>
 <td style="text-align: left;">
    4  </td><td style="text-align: left;"> Parent </td>
</tr>
<tr>
 <td style="text-align: left;">
    5  </td><td style="text-align: left;"> Nephew or niece </td>
</tr>
<tr>
 <td style="text-align: left;">
    6  </td><td style="text-align: left;"> Spouse </td>
</tr>
<tr>
 <td style="text-align: left;">
    7  </td><td style="text-align: left;"> Brother or sister in law
  </td>
</tr>

</table>

</td></tr>
<tr><td><code>fsize</code></td>
<td>
<p>Family size including parents, siblings and children of the proband and the siblings.</p>
</td></tr>
<tr><td><code>naff</code></td>
<td>
<p>Number of affected members by either event 1 or 2 within family.</p>
</td></tr>
<tr><td><code>df1</code></td>
<td>
<p>Number of affected members by event 1 within family.</p>
</td></tr>
<tr><td><code>df2</code></td>
<td>
<p>Number of affected members by event 2 within family.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Sampling weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07.
</p>
<p>Choi, Y.-H., Jung, H., Buys, S., Daly, M., John, E.M., Hopper, J., Andrulis, I., Terry, M.B., Briollais, L. (2021) A Competing Risks Model with Binary Time Varying Covariates for Estimation of Breast Cancer Risks in BRCA1 Families, Statistical Methods in Medical Research 30 (9), 2165-2183. https://doi.org/10.1177/09622802211008945.
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.simfam_cmp">summary.simfam_cmp</a>, <a href="#topic+plot.simfam_cmp">plot.simfam_cmp</a>, <a href="#topic+penplot_cmp">penplot_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Example 1: simulate competing risk family data from pop+ design using
#  Weibull distribution for both baseline hazards and inducing 
#  residual familial correlation through a correlated gamma frailty.

set.seed(4321)
fam &lt;- simfam_cmp(N.fam = 10, design = "pop+", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "cgamma", depend=c(1, 2, 0.5), 
       allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
       vbeta = list(c(-1.13, 2.35), c(-1, 2)))


head(fam) 

## Not run: 
  famID indID gender motherID fatherID proband generation majorgene secondgene  ageonset
1     1     1      1        0        0       0          1         3          0 124.23752
2     1     2      0        0        0       0          1         2          0  54.66936
3     1     3      0        2        1       1          2         2          0  32.75208
4     1     4      1        0        0       0          0         3          0 136.44926
5     1    11      1        3        4       0          3         3          0  71.53672
6     1    12      1        3        4       0          3         3          0 152.47073
  currentage     time status true_status mgene relation fsize naff df1 df2 weight
1   65.30602 65.30602      0           2     0        4    25    2   1   1      1
2   68.62107 54.66936      1           1     1        4    25    2   1   1      1
3   47.07842 32.75208      2           2     1        1    25    2   1   1      1
4   45.09295 45.09295      0           2     0        6    25    2   1   1      1
5   25.32819 25.32819      0           1     0        3    25    2   1   1      1
6   22.95059 22.95059      0           2     0        3    25    2   1   1      1

## End(Not run)

summary(fam)

plot(fam, famid = 1) # pedigree plots for family with ID = 1

</code></pre>

<hr>
<h2 id='simfam_tvc'>
Generate familial time-to-event data with a time-varying covariate
</h2><span id='topic+simfam_tvc'></span>

<h3>Description</h3>

<p>Generates familial time-to-event data with a time-varying covariate for specified study design, genetic model and source of residual familial correlation; the generated data frame also contains family structure (individual's id, father id, mother id, relationship to proband, generation), gender, current age, genotypes of major or second genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simfam_tvc(N.fam, design = "pop", variation = "none", interaction = FALSE, 
       add.x = FALSE, x.dist = NULL, x.parms = NULL, depend = NULL, 
       add.tvc = FALSE, tvc.type = "PE", tvc.range = NULL, tvc.parms = 1,
       base.dist = "Weibull", frailty.dist = NULL, base.parms = c(0.016, 3), 
       vbeta = c(1, 1), 
       allelefreq = 0.02, dominant.m = TRUE, dominant.s = TRUE,
       mrate = 0, hr = 0, probandage = c(45, 2), agemin = 20, agemax = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simfam_tvc_+3A_n.fam">N.fam</code></td>
<td>

<p>Number of families to generate.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_design">design</code></td>
<td>

<p>Family based study design used in the simulations. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code>, <code>"twostage"</code>, or <code>"noasc"</code>, where <code>"pop"</code> is for the population-based design that families are ascertained by affected probands, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands,  <code>"twostage"</code> for two-stage design that randomly samples families from the population in the first stage and oversamples high risk families in the second stage that include at least two affected members in the family, and <code>"noasc"</code> for no ascertainment correction that families are from simple random sampling. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"frailty"</code> for frailty shared within families, <code>"secondgene"</code> for second gene variation, <code>"kinship"</code> for correlated frailties within families with kinship matrix, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_interaction">interaction</code></td>
<td>

<p>Logical; if <code>TRUE</code>, allows the interaction between gender and mutation status. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_add.x">add.x</code></td>
<td>

<p>Logical; if <code>TRUE</code>, generates a covariate in addition to gender and majorgene. Default is <code>FALSE</code>. 
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_x.dist">x.dist</code></td>
<td>

<p>Distribution of the covairate. Possible choices to generate the covariate are: <code>"normal"</code> from normal distribution and <code>"binomial"</code> from binominal distribution. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_x.parms">x.parms</code></td>
<td>

<p>Parameter values for the specified distribution of the covariate. <code>x.parms = c(mean, sd)</code> should be specified for <code>x.dist = "normal"</code> and <code>x.parms = c(size, probs)</code> should be specified for <code>"binomial"</code>; when <code>size = 1</code>, it generates binary covariate. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_depend">depend</code></td>
<td>

<p>Inverse of variance of the frailty distribution. Dependence within families decreases with depend value.  Default is <code>NULL</code>. Value should be specified as a positive real number when <code>variation = "frailty"</code> or <code>variation = "kinship"</code>. 
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_add.tvc">add.tvc</code></td>
<td>

<p>Logical; if <code>TRUE</code>, generates a time varying covariate. Default is <code>FALSE</code>. 
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_tvc.type">tvc.type</code></td>
<td>

<p>Choice of time-varying covariate model. Possible choices are: <code>"PE"</code> and <code>"CO"</code>.
Default is <code>"PE"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_tvc.range">tvc.range</code></td>
<td>

<p>Range of ages at which the time-varying covariate occurs. 
Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_tvc.parms">tvc.parms</code></td>
<td>

<p>Vector of parameter values used for the time-varying covariate model. 
Default value is 1.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution.  Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code> <code>"gamma"</code>, <code>"logBurr"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_frailty.dist">frailty.dist</code></td>
<td>

<p>Choice of frailty distribution.  Possible choices are: <code>"gamma"</code> or <code>"lognormal"</code> when <code>variation="frailty"</code>. Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_base.parms">base.parms</code></td>
<td>

<p>Vector of parameter values for the specified baseline hazard function. <code>base.parms = c(lambda, rho)</code> should be specified for <code>base.dist = "Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"gamma"</code>, and <code>"lognormal"</code>. For <code>base.dist = "logBurr"</code>, three parameters should be specified <code>base.parms = c(lambda, rho, eta)</code>. 
Default value is <code>base.parms = c(0.016, 3)</code> for <code>base.dist = "Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_vbeta">vbeta</code></td>
<td>

<p>Vector of regression coefficients for gender, majorgene, interaction between gender and majorgene (if <code>interaction = TRUE</code>), secondgene (if <code>variation = "secondgene"</code>), and additional covariate (if <code>add.x = TRUE</code>).</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_allelefreq">allelefreq</code></td>
<td>

<p>Population allele frequencies of major disease gene. Value should be between 0 and 1. 
Vector of population allele frequencies of major and second disease genes should be provided when <code>variation = "secondgene"</code>. Default value is <code>allelefreq = 0.02</code>.</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_dominant.m">dominant.m</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of major gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_dominant.s">dominant.s</code></td>
<td>
<p>Logical; if <code>TRUE</code>, the genetic model of second gene is dominant, otherwise recessive.</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_mrate">mrate</code></td>
<td>

<p>Proportion of missing genotypes, value between 0 and 1. Default value is 0.</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_hr">hr</code></td>
<td>

<p>Proportion of high risk families, which include at least two affected members, to be sampled from the two stage sampling. This value should be specified when <code>design="twostage"</code>. Default value is 0. Value should lie between 0 and 1.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_probandage">probandage</code></td>
<td>

<p>Vector of mean and standard deviation for the proband age. Default values are mean of 45 years and standard deviation of 2 years, <code>probandage = c(45, 2)</code>.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is 20 years of age.
</p>
</td></tr>
<tr><td><code id="simfam_tvc_+3A_agemax">agemax</code></td>
<td>

<p>Maximum age of disease onset or maximum age. Default is 100 years of age.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><b>Time-varying covariate</b>
</p>
<p>When <code>add.tvc = TRUE</code>, the time at which the time-varying covariate (TVC) occurs, <code>tvc.age</code>, is generated from a uniform distribution with the range specified by <code>tvc.range</code>. A vector of minimum and maximum ages for the TVC should be specified in <code>tve.range</code>. When <code>tvc.range = NULL</code>, <code>agemin</code> and <code>agemax</code> are used as the range. In addition, <code>tvc.type</code> should be either <code>"PE"</code> or <code>"CO"</code> and the parameter values for the specified TVC type should be provided in <code>tvc.parms</code>.
</p>
<p><code>tvc.type = "PE"</code> represents a permanent exposure model for TVC which assumes that the effect of the TVC stays constant after <code>tvc.age</code>. The <code>tvc.parms</code> for the PE model should be specified as a single value, which represents log hazard ratio.
</p>
<p><code>tvc.type = "CO"</code> represents the Cox and Oaks model for TVC which assumes that the effect of the TVC decays exponentially over time in the form
 &beta; exp(-(t - t*) &eta;) + &eta;<sub>0</sub>,
where t* is the time at which the TVC occurs.
The <code>tvc.parms</code> for the CO model should be specified by a vector of three parameters consisting of <code>c(beta, eta, eta0)</code>.
</p>
<p><b>Family-based study design</b>
</p>
<p>The <code>design</code> argument defines the type of family based design to be simulated. Two variants of the population-based and clinic-based design can be chosen: <code>"pop"</code> when proband is affected, <code>"pop+"</code> when proband is affected mutation carrier, <code>"cli"</code> when proband is affected and at least one parent and one sibling are affected, <code>"cli+"</code> when proband is affected mutation-carrier and at least one parent and one sibling are affected. The two-stage design, <code>"twostage"</code>, is used to oversample high risk families, where the proportion of high risks families to include in the sample is specified by <code>hr</code>. High risk families often include multiple (at least two) affected members in the family. <code>design = "noasc"</code> is to be used for no ascertainment correction.
</p>
<p><b>Penetrance model</b>
</p>
<p>The ages at onset are generated from the following penetrance models depending on the choice of <code>variation = "none", "frailty", "secondgene", "kinship".</code>. When <code>variation = "none"</code>, the ages at onset are independently generated from the proportional hazard model conditional on the gender and carrier status of major gene mutation,   X = c(x<sub>s</sub>, x<sub>g</sub>).  
</p>
<p>The ages at onset correlated within families are generated from the shared frailty model (<code>variation = "frailty"</code>) , the correlated shared frailty model with kinship matrix (<code>variation = "kinship"</code>), or the two-gene model (<code>variation = "secondene"</code>), where the residual familial correlation is induced by a frailty or a second gene, respectively, shared within the family.
</p>
<p><em>The proportional hazard model</em>
</p>
 <p style="text-align:center"> h(t|X) = h<sub>0</sub>(t - t<sub>0</sub>) exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The shared frailty model</em>
</p>
 <p style="text-align:center"> h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, <code class="reqn">Z</code> represents a frailty shared within families and follows either a gamma or log-normal distribution, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The correlated shared frailty model with kinship matrix</em>
</p>
 <p style="text-align:center"> h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>g</sub> * x<sub>g</sub>), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, <code class="reqn">Z</code> represents a vector of frailties following a multivariate log-normal distribution with mean <code class="reqn">0</code> and variance <code class="reqn">2*K*depend</code>, where <code class="reqn">K</code> represents the kinship matrix, x<sub>x</sub> and x<sub>g</sub> indicate male (1) or female (0) and carrier (1) or non-carrier (0) of a main gene of interest, respectively.
</p>
<p><em>The two-gene model</em>
</p>
 <p style="text-align:center"> h(t|X) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp(&beta;<sub>s</sub> * x<sub>s</sub> + &beta;<sub>1</sub> * x<sub>2</sub> + &beta;<sub>2</sub> * x<sub>2</sub>), </p>
<p>where  x<sub>1</sub>, x<sub>2</sub> indicate carriers (1) and non-carriers (0) of a major gene and of second gene mutation, respectively.
</p>
<p>The current ages for each generation are simulated assuming normal distributions. However, the probands' ages are generated using a left truncated normal distribution as their ages cannot be less than the minimum age of onset. The average age difference between each generation and their parents is specified as 20 years apart. 
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'simfam'</code>, a data frame which contains:
</p>
<table>
<tr><td><code>famID</code></td>
<td>
<p>Family identification (ID) numbers.</p>
</td></tr>
<tr><td><code>indID</code></td>
<td>
<p>Individual ID numbers.</p>
</td></tr>
<tr><td><code>gender</code></td>
<td>
<p>Gender indicators: 1 for males, 0 for females.</p>
</td></tr>
<tr><td><code>motherID</code></td>
<td>
<p>Mother ID numbers.</p>
</td></tr>
<tr><td><code>fatherID</code></td>
<td>
<p>Father ID numbers.</p>
</td></tr>
<tr><td><code>proband</code></td>
<td>
<p>Proband indicators: 1 if the individual is the proband, 0 otherwise.</p>
</td></tr>
<tr><td><code>generation</code></td>
<td>
<p>Individuals generation: 1=parents of probands,2=probands and siblings, 3=children of probands and siblings.</p>
</td></tr>
<tr><td><code>majorgene</code></td>
<td>
<p>Genotypes of major gene: 1=AA, 2=Aa, 3=aa where A is disease gene.</p>
</td></tr>
<tr><td><code>secondgene</code></td>
<td>
<p>Genotypes of second gene: 1=BB, 2=Bb, 3=bb where B is disease gene.</p>
</td></tr>
<tr><td><code>ageonset</code></td>
<td>
<p>Ages at disease onset in years.</p>
</td></tr>
<tr><td><code>currentage</code></td>
<td>
<p>Current ages in years.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Ages at disease onset for the affected or ages of last follow-up for the unaffected. </p>
</td></tr>
<tr><td><code>status</code></td>
<td>
<p>Disease statuses: 1 for affected, 0 for unaffected (censored).</p>
</td></tr>
<tr><td><code>mgene</code></td>
<td>
<p>Major gene mutation indicators: 1 for mutated gene carriers, 0 for mutated gene noncarriers, or <code>NA</code> if missing.</p>
</td></tr>
<tr><td><code>newx</code></td>
<td>
<p>Additional covariate when <code>add.x = TRUE</code>.</p>
</td></tr>
<tr><td><code>tvc.age</code></td>
<td>
<p>Age at which the time-varying covariate occurs when <code>add.tvc = TRUE</code>.</p>
</td></tr>
<tr><td><code>tvc.status</code></td>
<td>
<p>TVC status: 1 if <code>tvc.age &lt; time</code>, 0 otherwise.</p>
</td></tr>
<tr><td><code>relation</code></td>
<td>
<p>Family members' relationship with the proband:
</p>

<table>
<tr>
 <td style="text-align: left;">
    1  </td><td style="text-align: left;"> Proband (self) </td>
</tr>
<tr>
 <td style="text-align: left;">
    2  </td><td style="text-align: left;"> Brother or sister </td>
</tr>
<tr>
 <td style="text-align: left;">
    3  </td><td style="text-align: left;"> Son or daughter </td>
</tr>
<tr>
 <td style="text-align: left;">
    4  </td><td style="text-align: left;"> Parent </td>
</tr>
<tr>
 <td style="text-align: left;">
    5  </td><td style="text-align: left;"> Nephew or niece </td>
</tr>
<tr>
 <td style="text-align: left;">
    6  </td><td style="text-align: left;"> Spouse </td>
</tr>
<tr>
 <td style="text-align: left;">
    7  </td><td style="text-align: left;"> Brother or sister in law
  </td>
</tr>

</table>

</td></tr>
<tr><td><code>fsize</code></td>
<td>
<p>Family size including parents, siblings and children of the proband and the siblings.</p>
</td></tr>
<tr><td><code>naff</code></td>
<td>
<p>Number of affected members in family.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Sampling weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.simfam_tvc">summary.simfam_tvc</a>, <a href="#topic+plot.simfam_tvc">plot.simfam_tvc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Example: simulate family data with TVC based on CO model.

set.seed(4321)
fam &lt;- simfam_tvc(N.fam = 10, design = "pop", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "gamma", depend = 1, 
       add.tvc = TRUE, tvc.type = "CO", tvc.range = c(30,60), 
       tvc.parms = c(1, 0.1, 0), allelefreq = 0.02, 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))

## Not run: 
&gt; head(fam) 
  famID indID gender motherID fatherID proband generation majorgene secondgene ageonset
1     1     1      1        0        0       0          1         2          0 61.80566
2     1     2      0        0        0       0          1         3          0 61.56996
3     1     3      0        2        1       1          2         2          0 39.42050
4     1     4      1        0        0       0          0         3          0 90.17320
5     1    13      0        3        4       0          3         3          0 51.49538
6     1    14      0        3        4       0          3         3          0 75.97238
  currentage     time status mgene  tvc.age tvc.status relation fsize naff weight
1   68.26812 61.80566      1     1 59.16387          1        4    29    3      1
2   68.60174 61.56996      1     0 39.45786          1        4    29    3      1
3   47.05410 39.42050      1     1 35.01941          1        1    29    3      1
4   44.86501 44.86501      0     0 58.67013          0        6    29    3      1
5   22.73075 22.73075      0     0 30.19254          0        3    29    3      1
6   22.71399 22.71399      0     0 40.66258          0        3    29    3      1

&gt; summary(fam) 
Study design:           pop: population-based study with affected probands 
Baseline distribution:  Weibull 
Frailty distribution:   gamma 
Number of families:                     10 
Average number of affected per family:  3.1 
Average number of carriers per family:  3.4 
Average family size:                    16.3 
Average age of onset for affected:      48.19 
Average number of TVC event per family: 4 
Sampling weights used:                  1

## End(Not run)


</code></pre>

<hr>
<h2 id='simfam2'>
Generate familial time-to-event data with Kinship or IBD matrices.
</h2><span id='topic+simfam2'></span>

<h3>Description</h3>

<p>Generate familial time-to-event data from correlated fraily model with Kinship or/and IBD matrices given pedigree data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simfam2(inputdata = NULL, IBD = NULL, design = "pop", variation = "none", depend = NULL, 
base.dist = "Weibull", base.parms = c(0.016, 3), var_names = c("gender", "mgene"), 
vbeta = c(1, 1), agemin = 20, hr = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simfam2_+3A_inputdata">inputdata</code></td>
<td>

<p>Dataframe contains variables <code>famID, indID, gender, motherID, fatherID, proband</code>, 
<code>generation, currentage</code> and other variables to be used in generating time-to-event data.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_ibd">IBD</code></td>
<td>

<p>IBD matrix
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_design">design</code></td>
<td>

<p>Family based study design used in the simulations. Possible choices are: <code>"pop"</code>, <code>"pop+"</code>, <code>"cli"</code>, <code>"cli+"</code>, <code>"twostage"</code>, or <code>"noasc"</code>, where <code>"pop"</code> is for the population-based design that families are ascertained by affected probands, <code>"pop+"</code> is similar to <code>"pop"</code> but with mutation carrier probands, <code>"cli"</code> is for the clinic-based design that includes affected probands with at least one parent and one sib affected, <code>"cli+"</code> is similar to <code>"cli"</code> but with mutation carrier probands,  <code>"twostage"</code> for two-stage design that randomly samples families from the population in the first stage and oversamples high risk families in the second stage that include at least two affected members in the family, and <code>"noasc"</code> for no ascertainment correction that families are from simple random sampling. Default is <code>"pop"</code>.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_variation">variation</code></td>
<td>

<p>Source of residual familial correlation.  Possible choices are: <code>"kinship"</code> for correlated frailties within families generated by kinship matrix, <code>"IBD"</code> for correlated frailties by IBD matrix, <code>c("kinship", "IBD")</code> by both kinship and IBD matrices, or <code>"none"</code> for no residual familial correlation. Default is <code>"none"</code>.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_depend">depend</code></td>
<td>

<p>Inverse of variance for the frailty distribution. A single value should be specified when <code>variation = "IBD"</code> or <code>variation = "kinship"</code> or a vector of two values when <code>variation = c("kinship", "IBD")</code>, where the first element corresponds to kinship matrix and the second element corresponds to IBD matrix.
Default is <code>NULL</code>.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_base.dist">base.dist</code></td>
<td>

<p>Choice of baseline hazard distribution.  Possible choices are: <code>"Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"lognormal"</code> <code>"gamma"</code>, <code>"logBurr"</code>. Default is <code>"Weibull"</code>.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_base.parms">base.parms</code></td>
<td>

<p>Vector of parameter values for the specified baseline hazard function. <code>base.parms = c(lambda, rho)</code> should be specified for <code>base.dist = "Weibull"</code>, <code>"loglogistic"</code>, <code>"Gompertz"</code>, <code>"gamma"</code>, and <code>"lognormal"</code>. For <code>base.dist = "logBurr"</code>, three parameters should be specified <code>base.parms = c(lambda, rho, eta)</code>. 
Default value is <code>base.parms = c(0.016, 3)</code> for <code>base.dist = "Weibull"</code>.
</p>
</td></tr>  	
<tr><td><code id="simfam2_+3A_var_names">var_names</code></td>
<td>

<p>Names of variables to be used in generating time-to-event data. Specified variables should be part of <code>inputdata</code>.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_vbeta">vbeta</code></td>
<td>

<p>Vector of regression coefficients for the variables specified by <code>var_names</code>. 
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_hr">hr</code></td>
<td>

<p>Proportion of high risk families, which include at least two affected members, to be sampled from the two stage sampling. This value should be specified when <code>design="twostage"</code>. Default value is 0. Value should lie between 0 and 1.
</p>
</td></tr>
<tr><td><code id="simfam2_+3A_agemin">agemin</code></td>
<td>

<p>Minimum age of disease onset or minimum age. Default is 20 years of age.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ages at onset are generated from the correlated frailties and covariates using the following model:
</p>
<p><em>The correlated shared frailty model with kinship and/or IBD matrices</em>
</p>
 <p style="text-align:center"> h(t|X,Z) = h<sub>0</sub>(t - t<sub>0</sub>) Z exp( X*vbeta ), </p>
<p>where h<sub>0</sub>(t) is the baseline hazard function, t<sub>0</sub> is a minimum age of disease onset, <code class="reqn">Z</code> represents a vector of frailties following a multivariate log-normal distribution with mean <code class="reqn">0</code> and variance <code class="reqn">2*K*sig1 + D*sig2</code>, where <code class="reqn">K</code> represents the kinship matrix and D is IBD matrix, <code class="reqn">sig1</code> and <code class="reqn">sig2</code> are variance components related to each matrix and their values are specified by <code>depend = c(1/sig1, 1/sig2)</code>, and <code class="reqn">X</code> represents a vector of variables whose names are specified by <code>var_names</code>, and \beta is a vector of corresponding coefficients whose values are specified by <code>vbeta</code>. 
</p>
<p>The variance structure of the frailties shared within families is chosen by either <code>variation = "kinship"</code> or <code>"IBD"</code> matrix or both <code>variation = c("kinship", "IBD")</code>. 
</p>
<p>When <code>variation = "none"</code>, the ages at onset are independently generated from the proportional hazard model conditional on the covariates <code class="reqn">X</code>. 
</p>
<p>The <code>design</code> argument defines the type of family based design to be simulated. Two variants of the population-based and clinic-based design can be chosen: <code>"pop"</code> when proband is affected, <code>"pop+"</code> when proband is affected mutation carrier, <code>"cli"</code> when proband is affected and at least one parent and one sibling are affected, <code>"cli+"</code> when proband is affected mutation-carrier and at least one parent and one sibling are affected. The two-stage design, <code>"twostage"</code>, is used to oversample high risk families, where the proportion of high risks families to include in the sample is specified by <code>hr</code>. High risk families often include multiple (at least two) affected members in the family. <code>design = "noasc"</code> is to be used for no ascertainment correction.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>'simfam'</code>, a data frame which contains <code>inputdata</code> and the following: 
</p>
<table>
<tr><td><code>ageonset</code></td>
<td>
<p>Ages at disease onset in years.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Ages at disease onset for the affected or ages of last follow-up for the unaffected. </p>
</td></tr>
<tr><td><code>status</code></td>
<td>
<p>Disease statuses: 1 for affected, 0 for unaffected (censored).</p>
</td></tr>
<tr><td><code>fsize</code></td>
<td>
<p>Family size including parents, siblings and children of the proband and the siblings.</p>
</td></tr>
<tr><td><code>naff</code></td>
<td>
<p>Number of affected members in family.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Sampling weights.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Choi, Y.-H., Briollais, L., He, W. and Kopciuk, K. (2021) FamEvent: An R Package for Generating and Modeling Time-to-Event Data in Family Designs, Journal of Statistical Software 97 (7), 1-30. doi:10.18637/jss.v097.i07
</p>
<p>Choi, Y.-H., Kopciuk, K. and Briollais, L. (2008) Estimating Disease Risk Associated Mutated Genes in Family-Based Designs, Human Heredity 66, 238-251.
</p>
<p>Choi, Y.-H. and Briollais (2011) An EM Composite Likelihood Approach for Multistage Sampling of Family Data with Missing Genetic Covariates, Statistica Sinica 21, 231-253.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.simfam2">summary.simfam2</a>, <a href="#topic+plot.simfam">plot.simfam</a>, <a href="#topic+penplot">penplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Example: simulate family data from a population-based design using
#  a Weibull distribution for the baseline hazard and inducing 
#  residual familial correlation through kinship and IBD matrices.

# Inputdata and IBD matrix should be provided; 
# simuated inputdata as an example here;

data &lt;- simfam(N.fam = 10, design = "noasc", variation = "none",
         base.dist = "Weibull", base.parms = c(0.016, 3), vbeta = c(1, 1))

IBDmatrix &lt;- diag(1, dim(data)[1])
data &lt;- data[ , c(1:7, 11, 14)]

fam2 &lt;- simfam2(inputdata = data, IBD = IBDmatrix, design = "pop", 
        variation = c("kinship","IBD"), depend = c(1, 1), 
        base.dist = "Weibull", base.parms = c(0.016, 3),
        var_names = c("gender", "mgene"), vbeta = c(1,1),
        agemin=20) 

head(fam2)

summary(fam2)

</code></pre>

<hr>
<h2 id='summary.penmodel'>
Summary method for class <code>penmodel</code>
</h2><span id='topic+summary.penmodel'></span>

<h3>Description</h3>

<p>Provides a summary of a fitted penetrance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel'
summary(object, correlation=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.penmodel_+3A_object">object</code></td>
<td>

<p>An object class of <code>'penmodel'</code>, a fitted model by <code><a href="#topic+penmodel">penmodel</a></code> or <code><a href="#topic+penmodelEM">penmodelEM</a></code> functions.
</p>
</td></tr>
<tr><td><code id="summary.penmodel_+3A_correlation">correlation</code></td>
<td>
<p>Logical; if <code>TRUE</code>, returns the correlation matrix of the estimated parameters. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="summary.penmodel_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the object of class <code>'summary.penmodel'</code>, including the following summary values:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>
<p>List of parameter estimates of transformed baseline parameters and regression coefficients, their standard errors, their robust standard errors if <code>robust=TRUE</code> was selected when fitting the penetrance model, <code class="reqn">t</code>-statistics and corresponding two-sided <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code>varcov</code></td>
<td>
<p>Variance-covariance matrix of the parameter estimates.</p>
</td></tr>
<tr><td><code>varcov.robust</code></td>
<td>
<p>Robust variance-covariance matrix of the parameter estimates if <code>robust = TRUE</code> was selected when fitting the penetrance model.</p>
</td></tr>
<tr><td><code>correlation</code></td>
<td>
<p>Correlation matrix obtained from the variance-covariance matrix.</p>
</td></tr>
<tr><td><code>correlation.robust</code></td>
<td>
<p>Correlation matrix obtained from the robust variance-covariance matrix if <code>robust = TRUE</code> was selected when fitting the penetrance model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel">penmodel</a>, <a href="#topic+penmodelEM">penmodelEM</a>, <a href="#topic+print.penmodel">print.penmodel</a>, <a href="#topic+print.summary.penmodel">print.summary.penmodel</a> <a href="#topic+plot.penmodel">plot.penmodel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated family data 

set.seed(4321)
fam &lt;- simfam(N.fam = 200, design = "pop+", variation = "none", base.dist = "Weibull", 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35), agemin = 20, allelefreq = 0.02)
 
# Penetrance model fit for the simulated family data

fit &lt;- penmodel(Surv(time, status) ~ gender + mgene, cluster = "famID", 
       parms=c(0.01, 3, -1.13, 2.35), data = fam, design = "pop+", base.dist = "Weibull")

# Summary of the model parameter and penetrance estimates from model fit

summary(fit)

## Not run: 
Estimates: 
            Estimate Std. Error t value Pr(&gt;|t|)  
log(lambda)   -4.531    0.08583 -52.793  0.01206 *
log(rho)       1.113    0.04688  23.737  0.02680 *
gender        -1.302    0.19233  -6.768  0.09339 .
mgene          2.349    0.23825   9.859  0.06436 .
Signif. codes:   0 '***'  0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1 

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.penmodel_cmp'>
Summary method for class <code>penmodel_cmp</code>
</h2><span id='topic+summary.penmodel_cmp'></span>

<h3>Description</h3>

<p>Provides a summary of a fitted competing risks penetrance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'penmodel_cmp'
summary(object, correlation=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.penmodel_cmp_+3A_object">object</code></td>
<td>

<p>An object class of <code>'penmodel_cmp'</code>, a fitted model by <code><a href="#topic+penmodel_cmp">penmodel_cmp</a></code> function.
</p>
</td></tr>
<tr><td><code id="summary.penmodel_cmp_+3A_correlation">correlation</code></td>
<td>
<p>Logical; if <code>TRUE</code>, returns the correlation matrix of the estimated parameters. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="summary.penmodel_cmp_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the object of class <code>'summary.penmodel_cmp'</code>, including the following summary values:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>
<p>List of parameter estimates of transformed baseline parameters and regression coefficients, their standard errors, their robust standard errors if <code>robust=TRUE</code> was selected when fitting the penetrance model, <code class="reqn">t</code>-statistics and corresponding two-sided <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code>varcov</code></td>
<td>
<p>Variance-covariance matrix of the parameter estimates.</p>
</td></tr>
<tr><td><code>varcov.robust</code></td>
<td>
<p>Robust variance-covariance matrix of the parameter estimates if <code>robust = TRUE</code> was selected when fitting the penetrance model.</p>
</td></tr>
<tr><td><code>correlation</code></td>
<td>
<p>Correlation matrix obtained from the variance-covariance matrix.</p>
</td></tr>
<tr><td><code>correlation.robust</code></td>
<td>
<p>Correlation matrix obtained from the robust variance-covariance matrix if <code>robust = TRUE</code> was selected when fitting the penetrance model.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+penmodel_cmp">penmodel_cmp</a>, <a href="#topic+print.penmodel_cmp">print.penmodel_cmp</a>, <a href="#topic+print.summary.penmodel_cmp">print.summary.penmodel_cmp</a> <a href="#topic+plot.penmodel_cmp">plot.penmodel_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulated family completing risks data 

## Not run: 
set.seed(4321)
fam1 &lt;- simfam_cmp(N.fam = 300, design = "pop+", variation = "frailty", competing=TRUE,
              base.dist = "Weibull", frailty.dist = "gamma", depend=c(0.5, 1), 
              allelefreq = 0.02, base.parms = list(c(0.01, 3), c(0.01, 3)), 
              vbeta = list(c(-1.13, 2.35),c(-1, 2)))

# Penetrance model fit for the simulated family data

fit &lt;- penmodel_cmp(
       formula1 = Surv(time, status==1) ~ gender + mgene, 
       formula2 = Surv(time, status==2) ~ gender + mgene, 
       cluster = "famID", 
       parms = list(c(0.01, 3, -1.13, 2.35), c(0.01, 3, -1, 2)),
       data = fam1, design = "pop+", base.dist = "Weibull")

# Summary of the model parameter and penetrance estimates from model fit

summary(fit)

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.simfam'>
Summary method for <code>simfam</code>
</h2><span id='topic+summary.simfam'></span>

<h3>Description</h3>

<p>Provides a summary of simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam'
summary(object, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.simfam_+3A_object">object</code></td>
<td>

<p>An object class of <code>'simfam'</code> generated from <code>simfam</code> function
</p>
</td></tr>
<tr><td><code id="summary.simfam_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="summary.simfam_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Displays a summary of simulated data and returns the following values:
</p>
<table>
<tr><td><code>num.fam</code></td>
<td>
<p>Number of families simulated.</p>
</td></tr>
<tr><td><code>avg.num.affected</code></td>
<td>
<p>Average number of affected individuals per family.</p>
</td></tr>
<tr><td><code>avg.num.carriers</code></td>
<td>
<p>Average number of mutation carriers per family.</p>
</td></tr>
<tr><td><code>avg.family.size</code></td>
<td>
<p>Average family size.</p>
</td></tr>
<tr><td><code>ave.ageonset</code></td>
<td>
<p>Average age of onset for affected individuals.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam">simfam</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(4321)
fam &lt;- simfam(N.fam = 50, design = "pop", variation = "none", base.dist = "Weibull", 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))
       
summary(fam)
## Not run: 
Study design:                           pop 
Baseline distribution:                  Weibull 
Number of families:                     50 
Average number of affected per family:  1.24 
Average number of carriers per family:  1.3 
Average family size:                    17.02 
Average age of onset for affected:      40.08 

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.simfam_cmp'>
Summary method for <code>simfam_cmp</code>
</h2><span id='topic+summary.simfam_cmp'></span>

<h3>Description</h3>

<p>Provides a summary of simulated data from <code>simfam_cmp</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam_cmp'
summary(object, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.simfam_cmp_+3A_object">object</code></td>
<td>

<p>An object class of <code>'simfam'</code> generated from <code>simfam_cmp</code> function
</p>
</td></tr>
<tr><td><code id="summary.simfam_cmp_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="summary.simfam_cmp_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Displays a summary of simulated data and returns the following values:
</p>
<table>
<tr><td><code>num.fam</code></td>
<td>
<p>Number of families simulated.</p>
</td></tr>
<tr><td><code>avg.num.affected1</code></td>
<td>
<p>Average number of affected individuals by event 1 per family.</p>
</td></tr>
<tr><td><code>avg.num.affected2</code></td>
<td>
<p>Average number of affected individuals by event 2 per family.</p>
</td></tr>
<tr><td><code>avg.num.carriers</code></td>
<td>
<p>Average number of mutation carriers per family.</p>
</td></tr>
<tr><td><code>avg.family.size</code></td>
<td>
<p>Average family size.</p>
</td></tr>
<tr><td><code>ave.ageonset1</code></td>
<td>
<p>Average age of onset for affected individuals by event 1.</p>
</td></tr>
<tr><td><code>ave.ageonset2</code></td>
<td>
<p>Average age of onset for affected individuals by event 2.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_cmp">simfam_cmp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(4321)
fam &lt;- simfam_cmp(N.fam = 50, design = "pop+", variation = "none", 
        base.dist = "Weibull", 
        base.parms = list(c(0.01, 3), c(0.01, 3)), 
        vbeta = list(c(-1.13, 2.35), c(-1,2)))
       
summary(fam)
## Not run: 
Study design:                           pop+ 
Baseline distribution for event 1:      Weibull 
Baseline distribution for event 2:      Weibull 
Number of families:                     50 
Average number of event 1 per family:   1.24 
Average number of event 2 per family:   0.7 
Average number of carriers per family:  5.54 
Average family size:                    15.58 
Average age of onset for event 1:       42.59 
Average age of onset for event 2:       43.72 

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.simfam_tvc'>
Summary method for <code>simfam_tvc</code>
</h2><span id='topic+summary.simfam_tvc'></span>

<h3>Description</h3>

<p>Provides a summary of simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam_tvc'
summary(object, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.simfam_tvc_+3A_object">object</code></td>
<td>

<p>An object class of <code>'simfam_tvc'</code> generated from <code>simfam_tvc</code> function
</p>
</td></tr>
<tr><td><code id="summary.simfam_tvc_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="summary.simfam_tvc_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Displays a summary of simulated data and returns the following values:
</p>
<table>
<tr><td><code>num.fam</code></td>
<td>
<p>Number of families simulated.</p>
</td></tr>
<tr><td><code>avg.num.affected</code></td>
<td>
<p>Average number of affected individuals per family.</p>
</td></tr>
<tr><td><code>avg.num.carriers</code></td>
<td>
<p>Average number of mutation carriers per family.</p>
</td></tr>
<tr><td><code>avg.family.size</code></td>
<td>
<p>Average family size.</p>
</td></tr>
<tr><td><code>ave.ageonset</code></td>
<td>
<p>Average age of onset for affected individuals.</p>
</td></tr>
<tr><td><code>ave.num.tvc</code></td>
<td>
<p>Average number of TVC events per family.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam_tvc">simfam_tvc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(4321)
fam &lt;- simfam_tvc(N.fam = 10, design = "pop", variation = "frailty", 
       base.dist = "Weibull", frailty.dist = "gamma", depend = 1, 
       add.tvc = TRUE, tvc.type = "CO", tvc.range = c(30,60), 
       tvc.parms = c(1, 0.1, 0), allelefreq = 0.02, 
       base.parms = c(0.01, 3), vbeta = c(-1.13, 2.35))

summary(fam)
## Not run: 
Study design:           pop: population-based study with affected probands 
Baseline distribution:  Weibull 
Frailty distribution:   gamma 
Number of families:                     10 
Average number of affected per family:  3.1 
Average number of carriers per family:  3.4 
Average family size:                    16.3 
Average age of onset for affected:      48.19 
Average number of TVC event per family: 4 

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.simfam2'>
Summary method for <code>simfam2</code>
</h2><span id='topic+summary.simfam2'></span>

<h3>Description</h3>

<p>Provides a summary of simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'simfam2'
summary(object, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.simfam2_+3A_object">object</code></td>
<td>

<p>An object class of <code>'simfam2'</code> generated from <code>simfam2</code> function
</p>
</td></tr>
<tr><td><code id="summary.simfam2_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to use when printing.</p>
</td></tr>
<tr><td><code id="summary.simfam2_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Displays a summary of simulated data and returns the following values:
</p>
<table>
<tr><td><code>num.fam</code></td>
<td>
<p>Number of families simulated.</p>
</td></tr>
<tr><td><code>avg.num.affected</code></td>
<td>
<p>Average number of affected individuals per family.</p>
</td></tr>
<tr><td><code>avg.num.carriers</code></td>
<td>
<p>Average number of mutation carriers per family.</p>
</td></tr>
<tr><td><code>avg.family.size</code></td>
<td>
<p>Average family size.</p>
</td></tr>
<tr><td><code>ave.ageonset</code></td>
<td>
<p>Average age of onset for affected individuals.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yun-Hee Choi	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simfam2">simfam2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(4321)

data &lt;- simfam(N.fam = 10, design = "noasc", variation = "none",
         base.dist = "Weibull", base.parms = c(0.016, 3), vbeta = c(1, 1))

IBDmatrix &lt;- diag(1, dim(data)[1])
data &lt;- data[ , c(1:7, 11, 14)]

fam2 &lt;- simfam2(inputdata = data, IBD = IBDmatrix, design = "pop", 
        variation = c("kinship","IBD"), depend = c(1, 1), 
        base.dist = "Weibull", base.parms = c(0.016, 3),
        var_names = c("gender", "mgene"), vbeta = c(1,1),
        agemin=20) 

summary(fam2)
## Not run: 
Study design:          pop 
Baseline distribution: Weibull 
Frailty distribution:  lognormal with kinship and IBD matrices 
Number of families:                     50 
Average number of affected per family:  1.24 
Average number of carriers per family:  1.3 
Average family size:                    17.02 
Average age of onset for affected:      40.08 

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
