<!DOCTYPE html><html lang="en"><head><title>Help for package famSKATRC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {famSKATRC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#famSKAT_RC'>
<p>famSKAT_RC</p></a></li>
<li><a href='#famSKATRC'>
<p>Family Sequence Kernel Association Test for Rare and Common Variants</p></a></li>
<li><a href='#process_data'>
<p>Process Data</p></a></li>
<li><a href='#sample.ped.geno'>
<p>Sample Pedigree Genotype Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Family Sequence Kernel Association Test for Rare and Common
Variants</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Khalid B. Kunji [aut, cre],
  Mohamad Saad [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Khalid B. Kunji &lt;kkunji@hbku.edu.qa&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Description:</td>
<td>FamSKAT-RC is a family-based association kernel test for both 
    rare and common variants. This test is general and several special cases 
    are known as other methods: famSKAT, which only focuses on rare variants 
    in family-based data, SKAT, which focuses on rare variants in 
    population-based data (unrelated individuals), and SKAT-RC, which focuses 
    on both rare and common variants in population-based data. When one 
    applies famSKAT-RC and sets the value of phi to 1, famSKAT-RC becomes 
    famSKAT. When one applies famSKAT-RC and set the value of phi to 1 and the 
    kinship matrix to the identity matrix, famSKAT-RC becomes SKAT. When one 
    applies famSKAT-RC and set the kinship matrix (fullkins) to the identity 
    matrix (and phi is not equal to 1), famSKAT-RC becomes SKAT-RC. We also 
    include a small sample synthetic pedigree to demonstrate the method with.  
    For more details see Saad M and Wijsman EM (2014) &lt;<a href="https://doi.org/10.1002%2Fgepi.21844">doi:10.1002/gepi.21844</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a>,
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25132070">https://www.ncbi.nlm.nih.gov/pubmed/25132070</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm, kinship2, coxme, bdsmatrix</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.1)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>KeepSource:</td>
<td>TRUE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-11-08 19:24:55 UTC; kunji</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-11-09 15:01:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='famSKAT_RC'>
famSKAT_RC
</h2><span id='topic+famSKAT_RC'></span>

<h3>Description</h3>

<p>FamSKAT-RC is a family-based association kernel test for both rare and common variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>famSKAT_RC (PHENO, genotypes, id, fullkins, covariates=NULL, sqrtweights_c,
            sqrtweights_r, binomialimpute=FALSE, acc=NULL, maf, phi)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="famSKAT_RC_+3A_pheno">PHENO</code></td>
<td>

<p>The vector of the phenotype values. The missing values must be represented by NA.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_genotypes">genotypes</code></td>
<td>

<p>The genotype matrix. Its size should be N * P, where N is the number of individuals and P is the number of SNPs.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_id">id</code></td>
<td>

<p>The vector of individual IDs to be included in the test. These IDs are present in the kinship matrix.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_fullkins">fullkins</code></td>
<td>

<p>The full kinship matrix that contains all individuals.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_covariates">covariates</code></td>
<td>

<p>A matrix of possible covariates.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_sqrtweights_c">sqrtweights_c</code></td>
<td>

<p>The weight function to be assigned for common variants. An example is: function(maf) ifelse(maf&gt;0, dbeta(maf,0.5,0.5), 0).
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_sqrtweights_r">sqrtweights_r</code></td>
<td>

<p>The weight function to be assigned for rare variants. An example is: function(maf) ifelse(maf&gt;0, dbeta(maf,1,25), 0).
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_binomialimpute">binomialimpute</code></td>
<td>

<p>If TRUE, then impute missing genotypes using a binomial distribution (e.g. rbinom(10, 2, MAF), if 10 genotypes are missing).
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_acc">acc</code></td>
<td>

<p>The accuracy used in the Davies approximation. For example, acc= 1e-06.
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_maf">maf</code></td>
<td>

<p>A MAF threshold used to define rare and common variants (e.g. maf=0.01).
</p>
</td></tr>
<tr><td><code id="famSKAT_RC_+3A_phi">phi</code></td>
<td>

<p>This parameter indicates the contribution portion of rare variants. For example, a value of phi=0.5 means that the contribution of rare and common variants to the test is equaland a value of phi=1 means that only rare variants contribute to the test. You can vary the phi values as you wish and you can also provide a grid of different values (e.g. phi = c(0,0.2,0.5,0.9)). In this case, four p-values will be obtained, one for each value of phi.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>FamSKAT-RC is a family-based association kernel test for both rare and common variants. This test is general and several special cases are known as other methods: famSKAT, which only focuses on rare variants in family-based data, SKAT, which focuses on rare variants in population-based data (unrelated individuals), and SKAT-RC, which focuses on both rare and common variants in population-based data. When one applies famSKAT-RC and sets the value of phi to 1, famSKAT-RC becomes famSKAT. When one applies famSKAT-RC and set the value of phi to 1 and the kinship matrix to the identity matrix, famSKAT-RC becomes SKAT. When one applies famSKAT-RC and set the kinship matrix (fullkins) to the identity matrix (and phi is not equal to 1), famSKAT-RC becomes SKAT-RC.
</p>


<h3>Value</h3>

<p>The test statistic p-value.
</p>
<table role = "presentation">
<tr><td><code>temp_pvalues</code></td>
<td>
<p>Some Description</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(kinship2)
sample.ped.geno &lt;- process_data()
KIN = kinship(sample.ped.geno$IID, sample.ped.geno$FA, sample.ped.geno$MO)
IID = sample.ped.geno$IID
wuweights_r &lt;- function(maf) ifelse(maf&gt;0, dbeta(maf,1,25), 0)
wuweights_c &lt;- function(maf) ifelse(maf&gt;0, dbeta(maf,0.5,0.5), 0)
P_VALUES &lt;- famSKAT_RC(PHENO=sample.ped.geno[,"Phenotype"],genotypes=as.matrix(
            sample.ped.geno[,7:ncol(sample.ped.geno)]), binomialimpute=TRUE,
            id=IID,fullkins=KIN,maf=0.05, sqrtweights_c=wuweights_c,
            sqrtweights_r=wuweights_r, phi = c(0,0.2,0.5,0.9))
print(P_VALUES)
</code></pre>

<hr>
<h2 id='famSKATRC'>
Family Sequence Kernel Association Test for Rare and Common Variants
</h2><span id='topic+famSKATRC'></span>

<h3>Description</h3>

<p>FamSKAT-RC is a family-based association kernel test for both 
rare and common variants. This test is general and several special cases 
are known as other methods: famSKAT, which only focuses on rare variants 
in family-based data, SKAT, which focuses on rare variants in 
population-based data (unrelated individuals), and SKAT-RC, which focuses 
on both rare and common variants in population-based data. When one 
applies famSKAT-RC and sets the value of phi to 1, famSKAT-RC becomes 
famSKAT. When one applies famSKAT-RC and set the value of phi to 1 and the 
kinship matrix to the identity matrix, famSKAT-RC becomes SKAT. When one 
applies famSKAT-RC and set the kinship matrix (fullkins) to the identity 
matrix (and phi is not equal to 1), famSKAT-RC becomes SKAT-RC. We also 
include a small sample synthetic pedigree to demonstrate the method with.  
For more details see Saad M and Wijsman EM (2014) &lt;doi:10.1002/gepi.21844&gt;.
</p>


<h3>Author(s)</h3>

<p>Khalid B. Kunji [aut, cre], Mohamad Saad [aut]
</p>
<p>Maintainer: Khalid B. Kunji &lt;kkunji@hbku.edu.qa&gt;
</p>


<h3>References</h3>

<p>Saad M and Wijsman EM (2014) Combining family- and population-based imputation data for association analysis of rare and common variants in large pedigrees.
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4190076/
</p>

<hr>
<h2 id='process_data'>
Process Data
</h2><span id='topic+process_data'></span>

<h3>Description</h3>

<p>Processes the raw data included to produce data identical to the (also included)
sample.ped.geno data set.  This can be used for processing your own pedigrees with
SNP data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_data(Data = read.table(system.file("extdata", "data",
                                            package = "famSKATRC"), header = TRUE))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process_data_+3A_data">Data</code></td>
<td>

<p>A string, the path to the location of the data file you are processing, formatted as
the included example, which can be loaded with:
<code>read.table(system.file("extdata", "data", package = "famSKATRC"), header = TRUE)</code>
and can be found in your filesystem at: system.file(&quot;extdata&quot;, &quot;data&quot;,
package = &quot;famSKATRC&quot;)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the data frame with completed preprocessing changes for famSKATRC.
Mainly reworking IDs so there are not duplicates.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+famSKATRC">famSKATRC</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sample.ped.geno &lt;- process_data()
## The function is currently defined as
function(Data = read.table(system.file("extdata", "data",
                                                package = "famSKATRC"), header = TRUE))
{
  Data[ , "IID"] = paste(Data[ , "FID"]  , Data[ , "IID"]  ,sep=".")
  Data[Data[,"FA"]!=0 , "FA"] = paste(Data[Data[,"FA"]!=0 , "FID"], Data[Data[,"FA"]!=0,
                                                                        "FA"]  ,sep=".")
  Data[Data[,"FA"]!=0 , "MO"] = paste(Data[Data[,"FA"]!=0 , "FID"], Data[Data[,"FA"]!=0,
                                                                        "MO"]  ,sep=".")
  return(Data)
}
</code></pre>

<hr>
<h2 id='sample.ped.geno'>
Sample Pedigree Genotype Data
</h2><span id='topic+sample.ped.geno'></span>

<h3>Description</h3>

<p>A sample pedigree file with SNP data, already processed by the process_data()
function.  The raw data is also included in the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sample.ped.geno")</code></pre>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 36 variables.
</p>

<dl>
<dt><code>FID</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>IID</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>FA</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>MO</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>SEX</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Phenotype</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs15</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs17</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs21</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs23</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs24</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs26</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rs30</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>A pedigree file with 20 individuals from two families.  The first column is a family ID,
the second an individual ID, the 3rd gives their father, the 4th their mother, and the
5th their sex.  The 6th column gives their phenotype and columns 7 through 36 give their
genotype, 30 SNP loci.
</p>


<h3>Source</h3>

<p>This is synthetically generated data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sample.ped.geno)
library(kinship2)
sample.ped.geno$FA[sample.ped.geno$FA == "0"] &lt;- NA
sample.ped.geno$MO[sample.ped.geno$MO == "0"] &lt;- NA
sample.ped.geno$Phenotype[sample.ped.geno$Phenotype &gt;= 0] &lt;- 1
sample.ped.geno$Phenotype[sample.ped.geno$Phenotype &lt; 0] &lt;- 0
ped.list &lt;- pedigree(id = sample.ped.geno$IID, dadid = sample.ped.geno$FA,
                     momid = sample.ped.geno$MO, sex = sample.ped.geno$SEX,
                     famid = sample.ped.geno$FID,
                     affected = sample.ped.geno$Phenotype)
ped1 &lt;- ped.list[1]
ped2 &lt;- ped.list[2]
plot(ped1)
plot(ped2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
