<!DOCTYPE html><html><head><title>Help for package FAMT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FAMT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FAMT-package'><p>Factor Analysis for Multiple Testing (FAMT) : simultaneous tests under dependence in high-dimensional data</p></a></li>
<li><a href='#annotations'><p>Gene annotations data frame</p></a></li>
<li><a href='#as.FAMTdata'><p>Create a 'FAMTdata' object from an expression, covariates and annotations dataset</p></a></li>
<li><a href='#covariates'><p>Covariates data frame</p></a></li>
<li><a href='#defacto'><p>FAMT factors description</p></a></li>
<li><a href='#emfa'><p>Factor Analysis model adjustment with the EM algorithm</p></a></li>
<li><a href='#expression'><p>Gene expressions data frame</p></a></li>
<li><a href='#modelFAMT'><p>The FAMT complete multiple testing procedure</p></a></li>
<li><a href='#nbfactors'><p>Estimation of the optimal number of factors of the FA model</p></a></li>
<li><a href='#pi0FAMT'><p>Estimation of the Proportion of True Null Hypotheses</p></a></li>
<li><a href='#raw.pvalues'><p>Calculation of classical multiple testing statistics and p-values</p></a></li>
<li><a href='#residualsFAMT'><p>Calculation of residual under null hypothesis</p></a></li>
<li><a href='#summaryFAMT'><p>Summary of a FAMTdata or a FAMTmodel</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Factor Analysis for Multiple Testing (FAMT) : Simultaneous Tests
under Dependence in High-Dimensional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-06</td>
</tr>
<tr>
<td>Author:</td>
<td>David Causeur, Chloe Friguet, Magalie Houee-Bigot, Maela Kloareg</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Causeur &lt;David.Causeur@agrocampus-ouest.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>mnormt, impute</td>
</tr>
<tr>
<td>Description:</td>
<td>The method proposed in this package takes into account the impact of dependence on the multiple testing procedures for high-throughput data as proposed by Friguet et al. (2009). The common information shared by all the variables is modeled by a factor analysis structure. The number of factors considered in the model is chosen to reduce the false discoveries variance in multiple tests. The model parameters are estimated thanks to an EM algorithm. Adjusted tests statistics are derived, as well as the associated p-values. The proportion of true null hypotheses (an important parameter when controlling the false discovery rate) is also estimated from the FAMT model. Graphics are proposed to interpret and describe the factors.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://famt.free.fr/">http://famt.free.fr/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-09 09:34:57 UTC; causeur</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-09 10:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='FAMT-package'>Factor Analysis for Multiple Testing (FAMT) : simultaneous tests under dependence in high-dimensional data
</h2><span id='topic+FAMT-package'></span><span id='topic+FAMT'></span>

<h3>Description</h3>

<p>The method proposed in this package takes into account the impact of dependence on multiple testing procedures for high-throughput data as proposed by Friguet et al. (2009). The common information shared by all the variables is modeled by a factor analysis structure. The number of factors considered in the model is chosen to reduce the variance of the number of false discoveries. The model parameters are estimated thanks to an EM algorithm. Factor-adjusted tests statistics are derived, as well as the associated p-values. The proportion of true null hypotheses (an important parameter when controlling the false discovery rate) is also estimated from the FAMT model. Diagnostic plots are proposed to interpret and describe the factors.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> FAMT</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2010-05-03</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code> function creates a single R object containing the data stored: 
- in one mandatory data-frame: the 'expression' dataset with m rows (if m tests) and n columns (n is the sample size) containing the observations of the responses.
- and two optional data-frames: the 'covariates' dataset with n rows and at least 2 columns, one giving the specification to match 'expression' and 'covariates' and the other one containing the observations of at least one covariate. The optional dataset, 'annotations' can be provided to help interpreting the factors: with m rows and at least one column to identify the variables (ID).
</p>
<p>The whole multiple testing procedure is provided in a single function, <code><a href="#topic+modelFAMT">modelFAMT</a></code>, but you can also choose to apply the procedure step by step, using the functions : 
</p>
<p><code><a href="#topic+nbfactors">nbfactors</a></code> (Estimation of the optimal number of factors)
<code><a href="#topic+emfa">emfa</a></code> (EM fitting of the Factor Analysis model).
</p>
<p>The <code><a href="#topic+modelFAMT">modelFAMT</a></code> also provides the individual test statistics and corresponding p-values like the <code><a href="#topic+raw.pvalues">raw.pvalues</a></code> function.
</p>
<p>A function <code><a href="#topic+summaryFAMT">summaryFAMT</a></code> provides some key elements of classical summaries either on 'FAMTdata' or 'FAMTmodel'.
</p>
<p>The estimation of the proportion of true null hypotheses from a 'FAMTmodel' is done by the function <code><a href="#topic+pi0FAMT">pi0FAMT</a></code>. 
</p>
<p>The <code><a href="#topic+defacto">defacto</a></code> function provides diagnostic plots to interpret and describe the factors.
</p>


<h3>Author(s)</h3>

<p>David Causeur, Chloe Friguet, Magalie Houee-Bigot, Maela Kloareg.
</p>
<p>Maintainer: <a href="mailto:David.Causeur@agrocampus-ouest.fr">David.Causeur@agrocampus-ouest.fr</a>
</p>


<h3>References</h3>

<p>Causeur D., Friguet C., Houee-Bigot M., Kloareg M. (2011). Factor Analysis for Multiple Testing (FAMT): An R Package for Large-Scale Significance Testing Under Dependence. Journal of Statistical Software, 40(14),1-19. 
<a href="https://www.jstatsoft.org/v40/i14">https://www.jstatsoft.org/v40/i14</a>
</p>
<p>Friguet C., Kloareg M. and Causeur D. (2009). A factor model approach to multiple testing under dependence. Journal of the American Statistical Association, 104:488, p.1406-1415
</p>
<p><a href="http://famt.free.fr/">http://famt.free.fr/</a>
</p>

<hr>
<h2 id='annotations'>Gene annotations data frame
</h2><span id='topic+annotations'></span>

<h3>Description</h3>

<p>A data frame with 6 columns describing the 9893 genes, which expressions are stored in the 'expression' dataset, in terms of functional categories, oligonucleotide size and location on the microarray. See also <code><a href="#topic+expression">expression</a></code>, <code><a href="#topic+covariates">covariates</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(annotations)</code></pre>


<h3>Format</h3>

<p>A data frame with 9893 observations on the following 6 variables.
</p>

<dl>
<dt><code>ID</code></dt><dd><p>Gene identification</p>
</dd>
<dt><code>Name</code></dt><dd><p>Gene annotation (functional categories) (character)</p>
</dd>
<dt><code>Block</code></dt><dd><p>Location on the microarray(factor)</p>
</dd>
<dt><code>Column</code></dt><dd><p>Location on the microarray (factor)</p>
</dd>
<dt><code>Row</code></dt><dd><p>Location on the microarray (factor)</p>
</dd>
<dt><code>Length</code></dt><dd><p>Oligonucleotide size (numeric vector)</p>
</dd>
</dl>



<h3>Source</h3>

<p>UMR Genetique Animale - INRA/AGROCAMPUS OUEST - Rennes, France.</p>


<h3>References</h3>

<p>Blum Y., Le Mignon G., Lagarrigue S. and Causeur S. (2010) - A factor model to analyze heterogeneity in gene expression, BMC Bioinformatics, 11:368.
</p>
<p>Le Mignon, G. and Desert, C. and Pitel, F. and Leroux, S. and Demeure, O. and Guernec, G. and Abasht, B. and Douaire, M. and Le Roy, P. and Lagarrigue S. (2009) - Using trancriptome profling to characterize QTL regions on chicken chromosome 5. BMC Genomics, 10:575.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(annotations)
dim(annotations) 
summary(annotations)
</code></pre>

<hr>
<h2 id='as.FAMTdata'>Create a 'FAMTdata' object from an expression, covariates and annotations dataset</h2><span id='topic+as.FAMTdata'></span>

<h3>Description</h3>

<p>The function creates a 'FAMTdata' object containing the expression, the covariates and the annotations dataset if provided. The function checks the consistency of dataframes between them. Then missing values of expression can be imputed.</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.FAMTdata(expression, covariates = NULL, annotations = NULL, idcovar = 1, 
idannot = NULL, na.action=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.FAMTdata_+3A_expression">expression</code></td>
<td>
<p>An expression data frame with genes in rows and arrays in columns. The arrays are identified by the column names.</p>
</td></tr>
<tr><td><code id="as.FAMTdata_+3A_covariates">covariates</code></td>
<td>
<p>An optional data frame with arrays in rows, and covariates in columns. One column must contain the array identification (NULL by default).</p>
</td></tr>
<tr><td><code id="as.FAMTdata_+3A_annotations">annotations</code></td>
<td>
<p>An optional data frame containing informations on the genes (NULL by default)</p>
</td></tr>
<tr><td><code id="as.FAMTdata_+3A_idcovar">idcovar</code></td>
<td>
<p>The column number corresponding to the array identification in the covariates data frame (1 by default)</p>
</td></tr>
<tr><td><code id="as.FAMTdata_+3A_idannot">idannot</code></td>
<td>
<p>The column number corresponding to the gene identification in annotations data frame (NULL by default)</p>
</td></tr>
<tr><td><code id="as.FAMTdata_+3A_na.action">na.action</code></td>
<td>
<p>If TRUE (default value), missing expression data are imputed using nearest neighbor averaging (<code>impute.knn</code> function of 'impute' package).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code> function creates a single R object containing the data stored: 
- in one mandatory data-frame: the 'expression' dataset with m rows (if m tests) and n columns (n is the sample size) containing the observations of the responses.
- and two optional data frames: the 'covariates' dataset with n rows and at least 2 columns, one giving the specification to match 'expression' and 'covariates' and the other one containing the observations of at least one covariate. The optional dataset,'annotations', can be provided to help interpreting the factors: with m rows and at least one column to identify the variables (ID).
</p>


<h3>Value</h3>

<table>
<tr><td><code>expression</code></td>
<td>
<p>The expression data frame</p>
</td></tr>
<tr><td><code>covariates</code></td>
<td>
<p>The optional covariates data frame</p>
</td></tr>
<tr><td><code>annotations</code></td>
<td>
<p>The optional data frame containing annotations. The genes annotations such as the functional categories should be in a character form, not in a factor form.</p>
</td></tr>
<tr><td><code>idcovar</code></td>
<td>
<p>The column number corresponding to the array identification in the covariate data frame (which should correspond to the column names in 'expression')</p>
</td></tr>
<tr><td><code>na.expr</code></td>
<td>
<p>Rows and columns of expression with missing values</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The class of the data produced with the <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code> function is called 'FAMTdata'. We advise to carry out a summary of FAMT data with the function <code><a href="#topic+summaryFAMT">summaryFAMT</a></code>.</p>


<h3>Author(s)</h3>

<p> David Causeur </p>


<h3>See Also</h3>

<p><code><a href="#topic+summaryFAMT">summaryFAMT</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># The data are divided into one mandatory data-frame, the gene expressions, 
#  and two optional datasets: the covariates, and the annotations.

# The expression dataset with 9893 rows (genes) and 43 columns (arrays)
#  containing the observations of the responses.
# The covariates dataset with 43 rows (arrays) and 6 columns: 
#  the second column gives the specification to match 'expression' 
#  and 'covariates' (array identification), the other ones contain
#  the observations of covariates.
# The annotations dataset contains 9893 rows (genes) and 
#  6 columns to help interpreting the factors, the first one (ID) 
#  identifies the variables (genes). 

data(expression)
data(covariates)
data(annotations)

# Create the 'FAMTdata'
############################################
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)
# 'FAMTdata' summary
summaryFAMT(chicken)

</code></pre>

<hr>
<h2 id='covariates'>Covariates data frame
</h2><span id='topic+covariates'></span>

<h3>Description</h3>

<p>A data frame with 6 covariates in columns and 43 arrays in rows, describing the arrays of the expression dataset. See also <code><a href="#topic+expression">expression</a></code>, <code><a href="#topic+annotations">annotations</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covariates)</code></pre>


<h3>Format</h3>

<p>A data frame with 43 observations on the following 6 variables.
</p>

<dl>
<dt><code>AfClass</code></dt><dd><p>a factor with levels <code>F</code> (Fat) <code>L</code> (Lean) <code>NC</code> (Intermediate) giving the abdominal fatness class</p>
</dd>
<dt><code>ArrayName</code></dt><dd><p>Identifying the arrays (character)</p>
</dd>
<dt><code>Mere</code></dt><dd><p>a factor with 8 levels giving the dam of the offsprings</p>
</dd>
<dt><code>Lot</code></dt><dd><p>a factor with 4 levels giving the hatch</p>
</dd>
<dt><code>Pds9s</code></dt><dd><p>a numeric vector giving the body weight</p>
</dd>
<dt><code>Af</code></dt><dd><p>a numeric vector giving the abdominal fatness, the experimental condition of main interest in this example</p>
</dd>
</dl>



<h3>Source</h3>

<p>UMR Genetique Animale - INRA/AGROCAMPUS OUEST - Rennes, France.</p>


<h3>References</h3>

<p>Blum Y., Le Mignon G., Lagarrigue S. and Causeur S. (2010) - A factor model to analyze heterogeneity in gene expression, BMC Bioinformatics, 11:368.
</p>
<p>Le Mignon, G. and Desert, C. and Pitel, F. and Leroux, S. and Demeure, O. and Guernec, G. and Abasht, B. and Douaire, M. and Le Roy, P. and Lagarrigue S. (2009) - Using trancriptome profling to characterize QTL regions on chicken chromosome 5. BMC Genomics, 10:575.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(covariates)
dim(covariates) 
summary(covariates)
</code></pre>

<hr>
<h2 id='defacto'>FAMT factors description</h2><span id='topic+defacto'></span>

<h3>Description</h3>

<p>This function helps the user to describe and interpret the factors using some available external information on either genes or arrays. Diagnostic plots are provided.</p>


<h3>Usage</h3>

<pre><code class='language-R'>defacto(model, plot = TRUE, axes = c(1, 2), select.covar = NULL, 
select.annot = NULL, lim.b = 0.01, lab = TRUE, cex = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="defacto_+3A_model">model</code></td>
<td>
<p>'FAMTmodel' object</p>
</td></tr>
<tr><td><code id="defacto_+3A_plot">plot</code></td>
<td>
<p>Boolean (TRUE by default). If TRUE, diagnostic plots are provided (unless the 'FAMTmodel' has less than one factor).</p>
</td></tr>
<tr><td><code id="defacto_+3A_axes">axes</code></td>
<td>
<p>Vector of length 2, specifying the factors to plot.</p>
</td></tr>
<tr><td><code id="defacto_+3A_select.covar">select.covar</code></td>
<td>
<p>Selection of external covariates. If NULL (default value), the function takes all covariates except the array identifiers and those used in the model.</p>
</td></tr>
<tr><td><code id="defacto_+3A_select.annot">select.annot</code></td>
<td>
<p>Selection of external annotations. If NULL (default value), the function takes all the available factors in 'annotations'.</p>
</td></tr>
<tr><td><code id="defacto_+3A_lim.b">lim.b</code></td>
<td>
<p>Proportion of variables with the highest loadings for each factor to appear on plots or in tables (0.01 by default).</p>
</td></tr>
<tr><td><code id="defacto_+3A_lab">lab</code></td>
<td>
<p>Boolean. If TRUE (default value), array names are labeled on the figure</p>
</td></tr>
<tr><td><code id="defacto_+3A_cex">cex</code></td>
<td>
<p>A numerical value giving the amount by which plotting text and symbol should be enlarged relative to the default (1 is the default value)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>loadings</code></td>
<td>
<p>highest loadings (B matrix) for each factor. The proportion of loading is determined by &quot;lim.b&quot;</p>
</td></tr>
<tr><td><code>covariates</code></td>
<td>
<p>Matrix of p-values for the tests of linear relationships between scores on each factor (rows) and external covariates (columns).</p>
</td></tr>
<tr><td><code>annotations</code></td>
<td>
<p>Matrix of p-values for the tests of linear relationships between loadings of each factor (rows) and external annotations (columns).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Causeur, Maela Kloareg</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+modelFAMT">modelFAMT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## FAMT data
data(expression)
data(covariates)
data(annotations)

# Create the 'FAMTdata'
############################################
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)
# 'FAMTdata' summary 
## Not run: summaryFAMT(chicken)

# FAMT complete multiple testing procedure
############################################
model = modelFAMT(chicken,x=c(3,6),test=6,nbf=3)
# summary on the 'FAMT model'
## Not run: summaryFAMT(model)

# Factors description
############################################
chicken.defacto = defacto(model,axes=1:2,select.covar=4:5,select.annot=3:6,
cex=0.6)
</code></pre>

<hr>
<h2 id='emfa'>Factor Analysis model adjustment with the EM algorithm</h2><span id='topic+emfa'></span>

<h3>Description</h3>

<p>A function to fit a Factor Analysis model with the EM algorithm.</p>


<h3>Usage</h3>

<pre><code class='language-R'>emfa(data, nbf, x = 1, test = x[1], pvalues = NULL, min.err = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="emfa_+3A_data">data</code></td>
<td>
<p>'FAMTdata' object, see <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code></p>
</td></tr>
<tr><td><code id="emfa_+3A_nbf">nbf</code></td>
<td>
<p>Number of factors of the FA model, see <code><a href="#topic+nbfactors">nbfactors</a></code></p>
</td></tr>
<tr><td><code id="emfa_+3A_x">x</code></td>
<td>
<p>Column number(s) corresponding to the experimental condition and the optional covariates (1 by default) in the covariates data frame.</p>
</td></tr>
<tr><td><code id="emfa_+3A_test">test</code></td>
<td>
<p>Column number corresponding to the experimental condition (x[1] by default) on which the test is performed.</p>
</td></tr>
<tr><td><code id="emfa_+3A_pvalues">pvalues</code></td>
<td>
<p>p-values of the individual tests. If NULL, the classical procedure is applied (see <code><a href="#topic+raw.pvalues">raw.pvalues</a></code>)</p>
</td></tr>
<tr><td><code id="emfa_+3A_min.err">min.err</code></td>
<td>
<p>Stopping criterion value for iterations in EM algorithm (default value: 0.001)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to use this function, the number of factors is needed (otherwise, use <code><a href="#topic+nbfactors">nbfactors</a></code>). 
</p>


<h3>Value</h3>

<table>
<tr><td><code>B</code></td>
<td>
<p>Estimation of the loadings</p>
</td></tr>
<tr><td><code>Psi</code></td>
<td>
<p>Estimation of Psi</p>
</td></tr>
<tr><td><code>Factors</code></td>
<td>
<p>Scores of the individuals on the factors</p>
</td></tr>
<tr><td><code>commonvar</code></td>
<td>
<p>Proportion of genes common variance (modeled on the factors)</p>
</td></tr>
<tr><td><code>SelectHo</code></td>
<td>
<p>Vector of row numbers corresponding to the non-significant genes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Causeur</p>


<h3>References</h3>

<p>Friguet C., Kloareg M. and Causeur D. (2009). A factor model approach to multiple testing under dependence. Journal of the American Statistical Association, 104:488, p.1406-1415 </p>


<h3>See Also</h3>

<p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+nbfactors">nbfactors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Reading 'FAMTdata'
data(expression)
data(covariates)
data(annotations)
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)

# EM fitting of the Factor Analysis model
chicken.emfa = emfa(chicken,nbf=3,x=c(3,6),test=6)
chicken.emfa$commonvar
</code></pre>

<hr>
<h2 id='expression'>Gene expressions data frame
</h2><span id='topic+expression'></span>

<h3>Description</h3>

<p>This dataset concerns hepatic transciptome profiles of 43 half sib male chickens selected for their variability on abdominal fatness (AF). Genes are in rows (9893 genes) and arrays in columns (43 arrays).</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(expression)</code></pre>


<h3>Format</h3>

<p>A data frame with 9893 genes on 43 arrays. 

</p>


<h3>Source</h3>

<p>UMR Genetique Animale - INRA/AGROCAMPUS OUEST - Rennes, France.</p>


<h3>References</h3>

<p>Blum Y., Le Mignon G., Lagarrigue S. and Causeur S. (2010) - A factor model to analyze heterogeneity in gene expression, BMC Bioinformatics, 11:368.
</p>
<p>Le Mignon, G. and Desert, C. and Pitel, F. and Leroux, S. and Demeure, O. and Guernec, G. and Abasht, B. and Douaire, M. and Le Roy, P. and Lagarrigue S. (2009) - Using trancriptome profling to characterize QTL regions on chicken chromosome 5. BMC Genomics, 10:575.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expression)
dim(expression)
summary(expression)
</code></pre>

<hr>
<h2 id='modelFAMT'>The FAMT complete multiple testing procedure
</h2><span id='topic+modelFAMT'></span>

<h3>Description</h3>

<p>This function implements the whole FAMT procedure (including <code><a href="#topic+nbfactors">nbfactors</a></code> and <code><a href="#topic+emfa">emfa</a></code>). The number of factors considered in the model is chosen to reduce the variance of the number of the false discoveries. The model parameters are estimated using an EM algorithm. Factor-adjusted tests statistics are derived, as well as the corresponding p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>modelFAMT(data, x = 1, test = x[1], nbf = NULL, maxnbfactors = 8, 
min.err = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="modelFAMT_+3A_data">data</code></td>
<td>
<p>'FAMTdata' object, see <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code></p>
</td></tr>
<tr><td><code id="modelFAMT_+3A_x">x</code></td>
<td>
<p>Column number(s) corresponding to the experimental condition and the optional covariates (1 by default) in the covariates data frame.</p>
</td></tr>
<tr><td><code id="modelFAMT_+3A_test">test</code></td>
<td>
<p>Column number corresponding to the experimental condition (x[1] by default) one which the test is performed.</p>
</td></tr>
<tr><td><code id="modelFAMT_+3A_nbf">nbf</code></td>
<td>
<p>The number of factors of the FA model (NULL by default). If NULL, the function estimates the optimal nbf (see <code><a href="#topic+nbfactors">nbfactors</a></code>)</p>
</td></tr>
<tr><td><code id="modelFAMT_+3A_maxnbfactors">maxnbfactors</code></td>
<td>
<p>The maximum number of factors (8 by default)</p>
</td></tr>
<tr><td><code id="modelFAMT_+3A_min.err">min.err</code></td>
<td>
<p>Stopping criterion value for iterations (default value:0.001)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>adjpval</code></td>
<td>
<p>Vector of FAMT factor-adjusted p-values</p>
</td></tr>
<tr><td><code>adjtest</code></td>
<td>
<p>Vector of FAMT factor-adjusted F statistics</p>
</td></tr>
<tr><td><code>adjdata</code></td>
<td>
<p>Factor-adjusted FAMT data</p>
</td></tr>
<tr><td><code>FA</code></td>
<td>
<p>Estimation of the FA model parameters</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>Vector of classical p-values</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>Column number(s) corresponding to the experimental condition and the optional covariates in the covariates data frame</p>
</td></tr>
<tr><td><code>test</code></td>
<td>
<p>Column number corresponding to the experimental condition on which the test is performed</p>
</td></tr>
<tr><td><code>nbf</code></td>
<td>
<p>The number of factors used to fit the FA model</p>
</td></tr>
<tr><td><code>idcovar</code></td>
<td>
<p>The column number used for the array identification in the 'covariates' data frame</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The user can perform individual test statistics putting the number of factors (<code>nbf</code>) equal to zero.  
The result of this function is a 'FAMTmodel'. It is used as argument in other functions of the package : <code><a href="#topic+summaryFAMT">summaryFAMT</a></code>, <code><a href="#topic+pi0FAMT">pi0FAMT</a></code> or <code><a href="#topic+defacto">defacto</a></code>. 
We advise to carry out a summary of FAMT model with the function <code><a href="#topic+summaryFAMT">summaryFAMT</a></code>.
</p>


<h3>Author(s)</h3>

<p>David Causeur</p>


<h3>References</h3>

<p>Friguet C., Kloareg M. and Causeur D. (2009). A factor model approach to multiple testing under dependence. Journal of the American Statistical Association, 104:488, p.1406-1415 </p>


<h3>See Also</h3>

<p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+raw.pvalues">raw.pvalues</a></code>, <code><a href="#topic+nbfactors">nbfactors</a></code>, <code><a href="#topic+emfa">emfa</a></code>, <code><a href="#topic+summaryFAMT">summaryFAMT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Reading 'FAMTdata'
data(expression)
data(covariates)
data(annotations)

chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)

# Classical method with modelFAMT 
## Not run: modelpval=modelFAMT(chicken,x=c(3,6),test=6,nbf=0)
## Not run: summaryFAMT(modelpval)

# FAMT complete multiple testing procedure
# when the optimal number of factors is unknown
## Not run: model = modelFAMT(chicken,x=c(3,6),test=6)

# when the optimal number of factors has already been estimated 
 model = modelFAMT(chicken,x=c(3,6),test=6,nbf=3)

summaryFAMT(model)
hist(model$adjpval)
## End(Not run)
</code></pre>

<hr>
<h2 id='nbfactors'>Estimation of the optimal number of factors of the FA model</h2><span id='topic+nbfactors'></span>

<h3>Description</h3>

<p>The optimal number of factors of the FA model is estimated to minimize the variance of the number of false positives (see Friguet et al., 2009).</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbfactors(data, x = 1, test = x[1], pvalues = NULL, maxnbfactors = 8, 
diagnostic.plot = FALSE, min.err = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbfactors_+3A_data">data</code></td>
<td>
<p>'FAMTdata' object, see <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code></p>
</td></tr>
<tr><td><code id="nbfactors_+3A_x">x</code></td>
<td>
<p>Column number(s) corresponding to the experimental condition and the optional covariates (1 by default) in the covariates data frame</p>
</td></tr>
<tr><td><code id="nbfactors_+3A_test">test</code></td>
<td>
<p>Column number corresponding to the experimental condition (x[1] by default) on which the test is performed</p>
</td></tr>
<tr><td><code id="nbfactors_+3A_pvalues">pvalues</code></td>
<td>
<p>Vector of p-values for the individual tests. If NULL, the classical procedure is applied (see <code><a href="#topic+raw.pvalues">raw.pvalues</a></code>)</p>
</td></tr>
<tr><td><code id="nbfactors_+3A_maxnbfactors">maxnbfactors</code></td>
<td>
<p>The maximum number of factors for the FA model (8 by default)</p>
</td></tr>
<tr><td><code id="nbfactors_+3A_diagnostic.plot">diagnostic.plot</code></td>
<td>
<p>boolean (FALSE by default). If TRUE, the values of the variance inflation criteria for each number of factors are plotted</p>
</td></tr>
<tr><td><code id="nbfactors_+3A_min.err">min.err</code></td>
<td>
<p>Stopping criterion value for iterations (default value : 0.001)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>optimalnbfactors</code></td>
<td>
<p>Optimal number of factors of the FA model (an elbow criterion is used)</p>
</td></tr>
<tr><td><code>criterion</code></td>
<td>
<p>Variance criterion for each number of factors</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> David Causeur </p>


<h3>References</h3>

<p> Friguet C., Kloareg M. and Causeur D. (2009). A factor model approach to multiple testing under dependence. Journal of the American Statistical Association, 104:488, p.1406-1415 </p>


<h3>See Also</h3>

<p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+emfa">emfa</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
## Reading 'FAMTdata'
data(expression)
data(covariates)
data(annotations)
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)

# Estimation of the number of factors 
## Not run: nbfactors(chicken,x=c(3,6),test=6)

# Estimation of the number of factors with a graph of variance inflation 
# criterion
## Not run: nbfactors(chicken,x=c(3,6),test=6, diagnostic.plot=TRUE)

</code></pre>

<hr>
<h2 id='pi0FAMT'>Estimation of the Proportion of True Null Hypotheses
</h2><span id='topic+pi0FAMT'></span>

<h3>Description</h3>

<p>A function to estimate the proportion <code>pi0</code> of true null hypotheses from a 'FAMTmodel' (see also function &quot;pval.estimate.eta0&quot; in package &quot;fdrtool&quot;).</p>


<h3>Usage</h3>

<pre><code class='language-R'>pi0FAMT(model, method = c("smoother", "density"), 
diagnostic.plot = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pi0FAMT_+3A_model">model</code></td>
<td>
<p>'FAMTmodel' object (see <code><a href="#topic+modelFAMT">modelFAMT</a></code>)</p>
</td></tr>
<tr><td><code id="pi0FAMT_+3A_method">method</code></td>
<td>
<p>algorithm used to estimate the proportion of null p-values. Available options are &quot;density&quot; and &quot;smoother&quot; (as described in Friguet and Causeur, 2010)</p>
</td></tr>
<tr><td><code id="pi0FAMT_+3A_diagnostic.plot">diagnostic.plot</code></td>
<td>
<p>if TRUE the histogram of the p-values with the estimate of <code>pi0</code> horizontal line is plotted. With the &quot;smoother&quot; method, an additional graph is displayed showing the spline curve used to estimate <code>pi0</code>. With the &quot;density&quot; method, the estimated convex density of the p-values is plotted onto the histogram</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The quantity <code>pi0</code>, i.e. the proportion of null hypotheses, is an important parameter when controlling the false discovery rate (FDR). A conservative choice is <code>pi0</code> = 1 but a choice closer to the true value will increase efficiency and power - see Benjamini and Hochberg (1995, 2000), Black(2004) and Storey (2002) for details.
The function <code>pi0FAMT</code> provides 2 algorithms to estimate this proportion. The &quot;density&quot; method is based on Langaas et al. (2005)'s approach where the density of p-values f(p) is first estimated considering f as a convex function, and the estimation of <code>pi0</code> is got for p=1. The &quot;smoother&quot; method uses the smoothing spline approach proposed by Storey and Tibshirani(2003).
</p>


<h3>Value</h3>

<p><code>pi0</code>The estimated proportion <code>pi0</code> of null hypotheses.</p>


<h3>Author(s)</h3>

<p> Chloe Friguet &amp; David Causeur </p>


<h3>References</h3>

<p>Friguet C. and Causeur D. (2010) Estimation of the proportion of true null hypohteses in high-dimensional data under dependence. Submitted.
</p>
<p>&quot;density&quot; procedure: Langaas et al (2005) Estimating the proportion of true null hypotheses, with application to DNA microarray data. JRSS. B, 67, 555-572.
</p>
<p>&quot;smoother&quot; procedure: Storey, J. D., and R. Tibshirani (2003) Statistical significance for genome-wide experiments. Proc. Nat. Acad. Sci. USA, 100, 9440-9445. </p>


<h3>See Also</h3>

<p><code><a href="#topic+modelFAMT">modelFAMT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Reading 'FAMTdata'
data(expression)
data(covariates)
data(annotations)
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)

# FAMT complete multiple testing procedure
model = modelFAMT(chicken,x=c(3,6),test=6,nbf=3)

# Estimation of the Proportion of True Null Hypotheses
# "density" method 
## Not run:  pi0FAMT(model,method="density",diagnostic.plot=TRUE)

# "smoother" method
pi0FAMT(model,method="smoother",diagnostic.plot=TRUE)

</code></pre>

<hr>
<h2 id='raw.pvalues'>Calculation of classical multiple testing statistics and p-values</h2><span id='topic+raw.pvalues'></span>

<h3>Description</h3>

<p>Calculates for each gene expression, the Fisher test statistics and the corresponding p-value for H0: the gene expression does not depend on the experimental condition in a model with possible covariates.</p>


<h3>Usage</h3>

<pre><code class='language-R'>raw.pvalues(data, x = 1, test = x[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raw.pvalues_+3A_data">data</code></td>
<td>
<p>'FAMTdata' object, see <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code></p>
</td></tr>
<tr><td><code id="raw.pvalues_+3A_x">x</code></td>
<td>
<p>Column number(s) corresponding to the experimental condition and the optional covariates (1 by default) in the 'covariates' data frame.</p>
</td></tr>
<tr><td><code id="raw.pvalues_+3A_test">test</code></td>
<td>
<p>Column number corresponding to the experimental condition (x[1] by default) of interest in the multiple testing procedure.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>pval</code></td>
<td>
<p>Vector containing the p-values</p>
</td></tr>
<tr><td><code>test</code></td>
<td>
<p>Vector containing the F statistics</p>
</td></tr>
<tr><td><code>resdf</code></td>
<td>
<p>Residual degrees of freedom</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Causeur</p>


<h3>See Also</h3>

 <p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expression)
data(covariates)
data(annotations)

# Create the 'FAMTdata'
############################################
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)
# 'FAMTdata' summary
summaryFAMT(chicken)

# Calculation of classical p-values
############################################
# test on the 6th covariate: 
rawpval = raw.pvalues(chicken,x=6)
hist(rawpval$pval)

# with a supplementary covariate (third column of the covariates data frame)
## Not run: rawpval = raw.pvalues(chicken,x=c(3,6),test=6)
## Not run: hist(rawpval$pval)
</code></pre>

<hr>
<h2 id='residualsFAMT'>Calculation of residual under null hypothesis 
</h2><span id='topic+residualsFAMT'></span>

<h3>Description</h3>

<p>internal function</p>

<hr>
<h2 id='summaryFAMT'>Summary of a FAMTdata or a FAMTmodel</h2><span id='topic+summaryFAMT'></span>

<h3>Description</h3>

<p>The function produces summaries of 'FAMTdata' or 'FAMTmodel'. The function involves a specific method depending on the class of the main argument.
</p>
<p>If the main argument is a 'FAMTdata' object, the function provides, for the 'expression file', the number of tests (which corresponds to the number of genes or rows), the sample size (which is the number of arrays or columns). The function provides classical summaries for 'covariates' and 'annotations' data (see summary in <code><a href="#topic+FAMT-package">FAMT-package</a></code>). 
</p>
<p>If the argument is a 'FAMTmodel', the function provides the numbers of rejected genes using classical and FAMT analyses, the annotation characteristics of significant genes, and the estimated proportion of true null hypotheses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryFAMT(obj, pi0 = NULL, alpha = 0.15, info = c("ID", "Name"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summaryFAMT_+3A_obj">obj</code></td>
<td>
<p>'FAMTdata' or 'FAMTmodel', see also <code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+modelFAMT">modelFAMT</a></code></p>
</td></tr>
<tr><td><code id="summaryFAMT_+3A_pi0">pi0</code></td>
<td>
<p>Proportion of tests under H0. NULL, by default, it is estimated.</p>
</td></tr>
<tr><td><code id="summaryFAMT_+3A_alpha">alpha</code></td>
<td>
<p>Type I levels for the control of the false discovery rate (0.15 by default) if the first argument is 'FAMTmodel' (it can be a single value or a vector).</p>
</td></tr>
<tr><td><code id="summaryFAMT_+3A_info">info</code></td>
<td>
<p>Names of the columns containing the genes identification and array names in the original data frames, necessary if the first argument is 'FAMTmodel'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If the argument is a 'FAMTdata': 
a list with components expression:
</p>
<table>
<tr><td><code>expression$'Number of tests'</code></td>
<td>
<p>Number of genes</p>
</td></tr>
<tr><td><code>expression$'Sample size'</code></td>
<td>
<p>Number of arrays</p>
</td></tr>
<tr><td><code>covariates</code></td>
<td>
<p>Classical summary of covariates</p>
</td></tr>
<tr><td><code>annotations</code></td>
<td>
<p>Classical summary of annotations</p>
</td></tr>
</table>
<p>If the argument is a 'FAMTmodel':
</p>
<table>
<tr><td><code>nbreject</code></td>
<td>
<p>Matrix giving the numbers of rejected genes with the classical analysis and with the FAMT analysis for the given Type I levels alpha.</p>
</td></tr>
<tr><td><code>DE</code></td>
<td>
<p>Identification of the significant genes by their annotations.</p>
</td></tr>
<tr><td><code>pi0</code></td>
<td>
<p>Estimation of the proportion of true null hypotheses, estimated with the &quot;smoother&quot; method, see <code><a href="#topic+pi0FAMT">pi0FAMT</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>David Causeur
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.FAMTdata">as.FAMTdata</a></code>, <code><a href="#topic+modelFAMT">modelFAMT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Reading 'FAMTdata'
data(expression)
data(covariates)
data(annotations)
chicken = as.FAMTdata(expression,covariates,annotations,idcovar=2)

## Summary of a 'FAMTdata'
#############################################
summaryFAMT(chicken)

## Summary of a 'FAMTmodel'
#############################################
# FAMT complete multiple testing procedure 
model = modelFAMT(chicken,x=c(3,6),test=6,nbf=3)
summaryFAMT(model)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
