<!DOCTYPE html><html><head><title>Help for package fastglmpca</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fastglmpca}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fit_glmpca_pois'><p>Fit Poisson GLM-PCA Model to Count Data</p></a></li>
<li><a href='#fitted.glmpca_pois_fit'><p>Get Fitted Values for GLM-PCA Model Fit</p></a></li>
<li><a href='#generate_glmpca_data_pois'><p>Generate Data from a GLMCPA Model</p></a></li>
<li><a href='#pbmc_facs'><p>Mixture of 10 FACS-purified PBMC Single-Cell RNA-seq data</p></a></li>
<li><a href='#set_fastglmpca_threads'><p>Set up Multithreading for fastglmpca</p></a></li>
<li><a href='#summary.glmpca_pois_fit'><p>Summarize GLM-PCA Model Fit</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1-103</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-23</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast Algorithms for Generalized Principal Component Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements fast, scalable optimization algorithms for
    fitting generalized principal components analysis (GLM-PCA) models,
    as described in "A Generalization of Principal Components
    Analysis to the Exponential Family" Collins M, Dasgupta S, Schapire RE
    (2002, ISBN:9780262271738), and subsequently "Feature Selection 
    and Dimension Reduction for Single-Cell RNA-Seq Based on a Multinomial
    Model" Townes FW, Hicks SC, Aryee MJ, Irizarry RA (2019)
    &lt;<a href="https://doi.org/10.1186%2Fs13059-019-1861-6">doi:10.1186/s13059-019-1861-6</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/stephenslab/fastglmpca">https://github.com/stephenslab/fastglmpca</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/stephenslab/fastglmpca/issues">https://github.com/stephenslab/fastglmpca/issues</a></td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, Matrix, MatrixExtra, stats, distr, daarem, Rcpp (&ge;
1.0.8), RcppParallel (&ge; 5.1.5)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppParallel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, ggplot2, cowplot</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-24 18:17:13 UTC; eweine</td>
</tr>
<tr>
<td>Author:</td>
<td>Eric Weine [aut, cre],
  Peter Carbonetto [aut],
  Matthew Stephens [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Eric Weine &lt;ericweine15@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-31 14:40:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='fit_glmpca_pois'>Fit Poisson GLM-PCA Model to Count Data</h2><span id='topic+fit_glmpca_pois'></span><span id='topic+fit_glmpca_pois_control_default'></span><span id='topic+init_glmpca_pois'></span>

<h3>Description</h3>

<p>Fit a Poisson GLM-PCA model by maximum-likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_glmpca_pois(
  Y,
  K,
  fit0 = init_glmpca_pois(Y, K),
  verbose = TRUE,
  control = list()
)

fit_glmpca_pois_control_default()

init_glmpca_pois(
  Y,
  K,
  U,
  V,
  X = numeric(0),
  Z = numeric(0),
  B = numeric(0),
  W = numeric(0),
  fixed_b_cols = numeric(0),
  fixed_w_cols = numeric(0),
  col_size_factor = TRUE,
  row_intercept = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_glmpca_pois_+3A_y">Y</code></td>
<td>
<p>The n x m matrix of counts; all entries of <code>Y</code> should
be non-negative. It can be a sparse matrix (class
<code>"dgCMatrix"</code>) or dense matrix (class <code>"matrix"</code>).</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_k">K</code></td>
<td>
<p>Integer 1 or greater specifying the rank of the matrix
factorization. This should only be provided if the initial fit
(<code>fit0</code>) is not.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_fit0">fit0</code></td>
<td>
<p>Initial model fit. It should be an object of class
&ldquo;glmpca_fit_pois&rdquo;, such as an output from
<code>init_glmpca_pois</code> or a previous call to
<code>fit_glmpca_pois</code>.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_verbose">verbose</code></td>
<td>
<p>If <code>verbose = TRUE</code>, information about the
algorithm's progress is printed after each update.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_control">control</code></td>
<td>
<p>List of control parameters to modify behavior of
the optimization algorithm; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_u">U</code></td>
<td>
<p>An optional argument giving the initial estimate of the
loadings matrix. It should be an n x K matrix, where n is the
number of rows in the counts matrix <code>Y</code>, and K &gt; 0 is the rank
of the matrix factorization. When <code>U</code> and <code>V</code> are not
provided, input argument <code>K</code> should be specified instead.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_v">V</code></td>
<td>
<p>An optional argument giving is the initial estimate of the
factors matrix. It should be a m x K matrix, where m is the number
of columns in the counts matrix <code>Y</code>, and K &gt; 0 is the rank of
the matrix factorization. When <code>U</code> and <code>V</code> are not
provided, input argument <code>K</code> should be specified instead.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_x">X</code></td>
<td>
<p>Optional argument giving row covariates of the count
matrix <code>Y</code>. It should be an n x nx matrix, where nx is
the number of row covariates.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_z">Z</code></td>
<td>
<p>Optional argument giving column covariates of the count
matrix <code>Y</code>. It should be an m x nz matrix, where nz is the
number of column covariates.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_b">B</code></td>
<td>
<p>Optional argument giving the initial estimates for the
coefficients of the row covariates. It should be an m x nx matrix,
where nx is the number of row covariates.
This argument is ignored if X is not provided.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_w">W</code></td>
<td>
<p>Optional argument giving the initial estimates for the
coefficients of the column covariates.  It should be an n x nz matrix,
where nz is the number of column covariates.
This argument is ignored if Z is not provided.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_fixed_b_cols">fixed_b_cols</code></td>
<td>
<p>Optional numeric vector specifying which
columns of <code>B</code> (if any) should be fixed during
optimization. This argument is ignored if X is not provided.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_fixed_w_cols">fixed_w_cols</code></td>
<td>
<p>Optional numeric vector specifying which
columns of <code>W</code> (if any) should be fixed during
optimization. This argument is ignored if Z is not provided.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_col_size_factor">col_size_factor</code></td>
<td>
<p>If <code>col_size_factor = TRUE</code>, add a
fixed factor accounting for average differences in Poisson rates
across columns of <code>Y</code>. Setting <code>col_size_factor = TRUE</code>
and <code>row_intercept = TRUE</code> is intended to replicate the
default behavior of <code>glmpca</code>.</p>
</td></tr>
<tr><td><code id="fit_glmpca_pois_+3A_row_intercept">row_intercept</code></td>
<td>
<p>If <code>row_intercept = TRUE</code>, add a fixed
factor accounting for average differences in Poisson rates across
rows of <code>Y</code>. Setting <code>col_size_factor = TRUE</code>
and <code>row_intercept = TRUE</code> is intended to replicate the
default behavior of <code>glmpca</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In generalized principal component analysis (GLM-PCA)
based on a Poisson likelihood, the counts <code class="reqn">y_{ij}</code> stored in an
<code class="reqn">n \times m</code> matrix <code class="reqn">Y</code> are modeled as </p>
<p style="text-align: center;"><code class="reqn">y_{ij}
\sim Pois(\lambda_{ij}),</code>
</p>
<p> in which the logarithm of each rate
parameter <code class="reqn">\lambda_{ij}</code> is defined as a linear combination of
rank-K matrices to be estimated from the data: </p>
<p style="text-align: center;"><code class="reqn">\log
\lambda_{ij} = (UDV')_{ij},</code>
</p>
<p> where <code class="reqn">U</code> and <code class="reqn">V</code> are
orthogonal matrices of dimension <code class="reqn">n \times K</code> and <code class="reqn">m
\times K</code>, respectively, and <code class="reqn">D</code> is a diagonal <code class="reqn">K
\times K</code> matrix in which the entries along its diagonal are
positive and decreasing. <code class="reqn">K</code> is a tuning parameter specifying
the rank of the matrix factorization. This is the same as the
low-rank matrix decomposition underlying PCA (that is, the singular
value decomposition), but because we are not using a linear
(Gaussian) model, this is called &ldquo;generalized PCA&rdquo; or
&ldquo;GLM PCA&rdquo;.
</p>
<p>To allow for additional components that may be fixed,
<code>fit_glmpca_pois</code> can also fit the more general model
</p>
<p style="text-align: center;"><code class="reqn">\log \lambda_{ij} = (UDV' + XB' + WZ')_{ij},</code>
</p>
<p> in which
<code class="reqn">X</code>, <code class="reqn">Z</code> are fixed matrices of dimension <code class="reqn">n \times
n_x</code> and <code class="reqn">m \times n_z</code>, respectively, and
<code class="reqn">B</code>, <code class="reqn">W</code> are matrices of dimension <code class="reqn">m \times n_x</code> and <code class="reqn">n \times n_z</code> to be estimated from the data.
</p>
<p><code>fit_glmpca_pois</code> computes maximum-likelihood estimates (MLEs)
of <code class="reqn">U</code>, <code class="reqn">V</code>, <code class="reqn">D</code>, <code class="reqn">B</code> and <code class="reqn">W</code> satistifying the
orthogonality constraints for <code class="reqn">U</code> and <code class="reqn">V</code> and the
additional constraints on <code class="reqn">D</code> that the entries are positive and
decreasing. This is accomplished by iteratively fitting a series of
Poisson GLMs, where each of these individual Poissons GLMs is fitted
using a fast &ldquo;cyclic co-ordinate descent&rdquo; (CCD) algorithm.
</p>
<p>The <code>control</code> argument is a list in which any of the following
named components will override the default optimization algorithm
settings (as they are defined by
<code>fit_glmpca_pois_control_default</code>). Additional control
arguments not listed here can be used to control the behaviour of
<code><a href="daarem.html#topic+fpiter">fpiter</a></code> or <code><a href="daarem.html#topic+daarem">daarem</a></code>; see
the help accompanying these functions for details.
</p>

<dl>
<dt><code>use_daarem</code></dt><dd><p>If <code>use_daarem = TRUE</code>, the updates
are accelerated using DAAREM; see <code><a href="daarem.html#topic+daarem">daarem</a></code> for
details.</p>
</dd>
<dt><code>tol</code></dt><dd><p>This is the value of the &ldquo;tol&rdquo; control
argument for <code><a href="daarem.html#topic+fpiter">fpiter</a></code> or
<code><a href="daarem.html#topic+daarem">daarem</a></code> that determines when to stop the
optimization. In brief, the optimization stops when the change in
the estimates or in the log-likelihood between two successive
updates is less than &ldquo;tol&rdquo;.</p>
</dd>
<dt><code>maxiter</code></dt><dd><p>This is the value of the &ldquo;maxiter&rdquo;
control argument for <code><a href="daarem.html#topic+fpiter">fpiter</a></code> or
<code><a href="daarem.html#topic+daarem">daarem</a></code>. In brief, it sets the upper limit on
the number of CCD updates.</p>
</dd>
<dt><code>convtype</code></dt><dd><p>This is the value of the &ldquo;convtype&rdquo;
control argument for <code><a href="daarem.html#topic+daarem">daarem</a></code>. It determines
whether the stopping criterion is based on the change in the
estimates or the change in the log-likelihood between two
successive updates.</p>
</dd>
<dt><code>mon.tol</code></dt><dd><p>This is the value of the &ldquo;mon.tol&rdquo;
control argument for <code><a href="daarem.html#topic+daarem">daarem</a></code>. This setting
determines to what extent the monotonicity condition can be
violated.</p>
</dd>
<dt><code>num_ccd_iter</code></dt><dd><p>Number of co-ordinate descent updates to
be made to parameters at each iteration of the algorithm.</p>
</dd>
<dt><code>line_search</code></dt><dd><p>If <code>line_search = TRUE</code>, a
backtracking line search is performed at each iteration of CCD to
guarantee improvement in the objective (the log-likelihood).</p>
</dd>
<dt><code>ls_alpha</code></dt><dd><p>alpha parameter for backtracking line search.
(Should be a number between 0 and 0.5, typically a number near
zero.)</p>
</dd>
<dt><code>ls_beta</code></dt><dd><p>beta parameter for backtracking line search
controlling the rate at which the step size is decreased.
(Should be a number between 0 and 0.5.)</p>
</dd>
<dt><code>calc_deriv</code></dt><dd><p>If <code>calc_deriv = TRUE</code>, the maximum
gradient of <code class="reqn">U</code> and <code class="reqn">V</code> is calculated and stored after each
update. This may be useful for assessing convergence of the
optimization, though increases overhead.</p>
</dd>
<dt><code>calc_max_diff</code></dt><dd><p>If <code>calc_max_diff = TRUE</code>, the
largest change in <code class="reqn">U</code> and <code class="reqn">V</code> after each update is
calculated and stored. This may be useful for monitoring progress
of the optimization algorithm.</p>
</dd>
<dt><code>orthonormalize</code></dt><dd><p>If <code>orthonormalize = TRUE</code>, the
matrices <code class="reqn">U</code> and <code class="reqn">V</code> are made to be orthogonal after each
update step. This improves the speed of convergence without the
DAAREM acceleration; however, should not be used when
<code>use_daarem = TRUE</code>.</p>
</dd></dl>

<p>You may use function <code><a href="#topic+set_fastglmpca_threads">set_fastglmpca_threads</a></code> to adjust
the number of threads used in performing the updates.
</p>


<h3>Value</h3>

<p>An object capturing the state of the model fit. It contains
estimates of <code class="reqn">U</code>, <code class="reqn">V</code> and <code class="reqn">D</code> (stored as matrices
<code>U</code>, <code>V</code> and a vector of diagonal entries <code>d</code>,
analogous to the <code><a href="base.html#topic+svd">svd</a></code> return value); the other
parameters (<code class="reqn">X</code>, <code class="reqn">B</code>, <code class="reqn">Z</code>, <code class="reqn">W</code>); the log-likelihood
achieved (<code>loglik</code>); information about which columns of
<code class="reqn">B</code> and <code class="reqn">W</code> are fixed (<code>fixed_b_cols</code>,
<code>fixed_w_cols</code>); and a data frame <code>progress</code> storing
information about the algorithm's progress after each update.
</p>


<h3>References</h3>

<p>Townes, F. W., Hicks, S. C., Aryee, M. J. and Irizarry,
R. A. (2019). Feature selection and dimension reduction for
single-cell RNA-Seq based on a multinomial model. <em>Genome Biology</em>
<b>20</b>, 295. <a href="https://doi.org/10.1186/s13059-019-1861-6">doi:10.1186/s13059-019-1861-6</a>
</p>
<p>Collins, M., Dasgupta, S. and Schapire, R. E. (2002). A
generalization of principal components analysis to the exponential
family. In <em>Advances in Neural Information Processing Systems</em> 14.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fit_glmpca_pois">fit_glmpca_pois</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
n &lt;- 200
p &lt;- 100
K &lt;- 3
dat  &lt;- generate_glmpca_data_pois(n,p,K)
fit0 &lt;- init_glmpca_pois(dat$Y,K)
fit  &lt;- fit_glmpca_pois(dat$Y,fit0 = fit0)

</code></pre>

<hr>
<h2 id='fitted.glmpca_pois_fit'>Get Fitted Values for GLM-PCA Model Fit</h2><span id='topic+fitted.glmpca_pois_fit'></span>

<h3>Description</h3>

<p><code>fitted</code> method for the
&ldquo;glmpca_pois_fit&rdquo; class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'glmpca_pois_fit'
fitted(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitted.glmpca_pois_fit_+3A_object">object</code></td>
<td>
<p>An object of class &ldquo;glmpca_fit&rdquo;,
typically the result of calling <code><a href="#topic+fit_glmpca_pois">fit_glmpca_pois</a></code>.</p>
</td></tr>
<tr><td><code id="fitted.glmpca_pois_fit_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the generic
<code>fitted</code> method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An n x p matrix of fitted means. Calculated as
</p>
<p style="text-align: center;"><code class="reqn">exp(UDV')</code>
</p>
<p> using the <code>fit</code> object.
</p>

<hr>
<h2 id='generate_glmpca_data_pois'>Generate Data from a GLMCPA Model</h2><span id='topic+generate_glmpca_data_pois'></span>

<h3>Description</h3>

<p>Generate data from a GLMPCA model with a specified
rank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_glmpca_data_pois(n, p, K, link = c("log", "log1p"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_glmpca_data_pois_+3A_n">n</code></td>
<td>
<p>Number of rows (genes).</p>
</td></tr>
<tr><td><code id="generate_glmpca_data_pois_+3A_p">p</code></td>
<td>
<p>Number of columns (cells).</p>
</td></tr>
<tr><td><code id="generate_glmpca_data_pois_+3A_k">K</code></td>
<td>
<p>Rank of the underlying mean structure</p>
</td></tr>
<tr><td><code id="generate_glmpca_data_pois_+3A_link">link</code></td>
<td>
<p>Character vector describing the link between the product 
of the loading and factors and the mean of the data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function assumes that each column of the data is generated
from a multinomial distribution. Let </p>
<p style="text-align: center;"><code class="reqn">Y_j</code>
</p>
<p> denote column j of the 
generated data matrix. First, we set </p>
<p style="text-align: center;"><code class="reqn">sum(Y_j)</code>
</p>
<p> equal to a value 
generated from a </p>
<p style="text-align: center;"><code class="reqn">Uniform(50, 5000)</code>
</p>
<p> distribution. Then,
we generate </p>
<p style="text-align: center;"><code class="reqn">L</code>
</p>
<p> and </p>
<p style="text-align: center;"><code class="reqn">F</code>
</p>
<p> from mixture distributions,
and calculate </p>
<p style="text-align: center;"><code class="reqn">H = exp(L'F)</code>
</p>
<p>. Then, we generate the individual
elements of </p>
<p style="text-align: center;"><code class="reqn">Y_j</code>
</p>
<p> from a multinomial model where the probability
for each individual element is just </p>
<p style="text-align: center;"><code class="reqn">H_j</code>
</p>
<p> normalized.
</p>


<h3>Value</h3>

<p>list with the following components
</p>

<ul>
<li><p> LL - loadings of underlying mean structure. A K x n matrix
</p>
</li>
<li><p> FF - factors of underlying mean structure. A K x p matrix
</p>
</li>
<li><p> Y - n x p matrix of generated data.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1)
sim_data &lt;- generate_glmpca_data_pois(1000, 500, 1)

</code></pre>

<hr>
<h2 id='pbmc_facs'>Mixture of 10 FACS-purified PBMC Single-Cell RNA-seq data</h2><span id='topic+pbmc_facs'></span>

<h3>Description</h3>

<p>These data are a selection of the reference
transcriptome profiles generated via single-cell RNA sequencing
(RNA-seq) of 10 bead-enriched subpopulations of PBMCs (Donor A),
described in Zheng <em>et al</em> (2017). The data are unique
molecular identifier (UMI) counts for 16,791 genes in 3,774 cells.
(Genes with no expression in any of the cells were removed.) Since
the majority of the UMI counts are zero, they are efficiently
stored as a 16,791 x 3774 sparse matrix. These data are used in the
vignette illustrating how &lsquo;fastglmpca&rsquo; can be used to
analyze single-cell RNA-seq data. Data for a separate set of 1,000
cells is provided as a &ldquo;test set&rdquo; to evaluate out-of-sample
predictions.
</p>


<h3>Format</h3>

<p><code>pbmc_facs</code> is a list with the following elements:
</p>

<dl>
<dt>counts</dt><dd><p>16,791 x 3,774 sparse matrix of UMI counts, with
rows corresponding to genes and columns corresponding to
cells (samples). It is an object of class <code>"dgCMatrix"</code>).</p>
</dd>
<dt>counts_test</dt><dd><p>UMI counts for an additional test set of 100
cells.</p>
</dd>
<dt>samples</dt><dd><p>Data frame containing information about the
samples, including cell barcode and source FACS population
(&ldquo;celltype&rdquo; and &ldquo;facs_subpop&rdquo;).</p>
</dd>
<dt>samples_test</dt><dd><p>Sample information for the additional test
set of 100 cells.</p>
</dd>
<dt>genes</dt><dd><p>Data frame containing information and the genes,
including gene symbol and Ensembl identifier.</p>
</dd>
<dt>fit</dt><dd><p>GLM-PCA model that was fit to the UMI count data in
the vignette.</p>
</dd></dl>



<h3>Source</h3>

<p><a href="https://www.10xgenomics.com/resources/datasets">https://www.10xgenomics.com/resources/datasets</a>
</p>


<h3>References</h3>

<p>G. X. Y. Zheng <em>et al</em> (2017). Massively parallel digital
transcriptional profiling of single cells. <em>Nature Communications</em>
<b>8</b>, 14049. <a href="https://doi.org/10.1038/ncomms14049">doi:10.1038/ncomms14049</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(Matrix)
data(pbmc_facs)
cat(sprintf("Number of genes: %d\n",nrow(pbmc_facs$counts)))
cat(sprintf("Number of cells: %d\n",ncol(pbmc_facs$counts)))
cat(sprintf("Proportion of counts that are non-zero: %0.1f%%.\n",
            100*mean(pbmc_facs$counts &gt; 0)))

</code></pre>

<hr>
<h2 id='set_fastglmpca_threads'>Set up Multithreading for fastglmpca</h2><span id='topic+set_fastglmpca_threads'></span>

<h3>Description</h3>

<p>Initialize RcppParallel multithreading using a
pre-specified number of threads, or using the default number of
threads when <code>n</code> is not specified or is NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_fastglmpca_threads(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_fastglmpca_threads_+3A_n">n</code></td>
<td>
<p>The requested number of threads.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of threads to be used.
</p>

<hr>
<h2 id='summary.glmpca_pois_fit'>Summarize GLM-PCA Model Fit</h2><span id='topic+summary.glmpca_pois_fit'></span><span id='topic+print.summary.glmpca_pois_fit'></span>

<h3>Description</h3>

<p><code>summary</code> method for objects of class
&ldquo;glmpcan_fit_pois&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'glmpca_pois_fit'
summary(object, ...)

## S3 method for class 'summary.glmpca_pois_fit'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.glmpca_pois_fit_+3A_object">object</code></td>
<td>
<p>An object of class &ldquo;glmpca_fit&rdquo;,
typically the result of calling <code><a href="#topic+fit_glmpca_pois">fit_glmpca_pois</a></code>.</p>
</td></tr>
<tr><td><code id="summary.glmpca_pois_fit_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the generic
<code>summary</code> or <code>print.summary</code> method.</p>
</td></tr>
<tr><td><code id="summary.glmpca_pois_fit_+3A_x">x</code></td>
<td>
<p>An object of class &ldquo;summary.glmpca_fit&rdquo;,
usually the result of a call to <code>summary.glmpca_fit</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>summary</code> returns a vector of basic statistics
summarizing the model fit.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
