<!DOCTYPE html><html><head><title>Help for package fastqcr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fastqcr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fastqc'><p>Run FastQC Tool</p></a></li>
<li><a href='#fastqc_install'><p>Install FastQC Tool</p></a></li>
<li><a href='#qc_aggregate'><p>Aggregate FastQC Reports for Multiple Samples</p></a></li>
<li><a href='#qc_fails'><p>Inspect Problems in Aggregated FastQC Reports</p></a></li>
<li><a href='#qc_plot'><p>Plot FastQC Results</p></a></li>
<li><a href='#qc_plot_collection'><p>Plot FastQC Results of multiple samples</p></a></li>
<li><a href='#qc_read'><p>Read FastQC Data</p></a></li>
<li><a href='#qc_read_collection'><p>Read a collection of FastQC data files</p></a></li>
<li><a href='#qc_report'><p>Build a QC Report</p></a></li>
<li><a href='#qc_unzip'><p>Unzip Files in the FastQC Result Directory</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quality Control of Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-02-18</td>
</tr>
<tr>
<td>Description:</td>
<td>'FASTQC' is the most widely used tool for evaluating the quality of high throughput sequencing data.  
    It produces, for each sample, an html report and a compressed file containing the raw data. 
    If you have hundreds of samples, you are not going to open up each 'HTML' page. 
    You need some way of looking at these data in aggregate. 
    'fastqcr' Provides helper functions to easily parse, aggregate and analyze 
    'FastQC' reports for large numbers of samples. It provides a convenient solution for building 
    a 'Multi-QC' report, as well as, a 'one-sample' report with result interpretations.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, grid, gridExtra, ggplot2, magrittr, readr (&ge; 1.3.0),
rmarkdown (&ge; 1.4), rvest, tibble, tidyr, scales, stats, utils,
xml2, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://rpkgs.datanovia.com/fastqcr/index.html">https://rpkgs.datanovia.com/fastqcr/index.html</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kassambara/fastqcr/issues">https://github.com/kassambara/fastqcr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Collate:</td>
<td>'utilities.R' 'fastqc.R' 'fastqc_install.R' 'qc_aggregate.R'
'qc_plot.R' 'qc_plot_collection.R' 'qc_problems.R' 'qc_read.R'
'qc_read_collection.R' 'qc_report.R' 'qc_unzip.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-18 09:47:57 UTC; kassambara</td>
</tr>
<tr>
<td>Author:</td>
<td>Alboukadel Kassambara [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alboukadel Kassambara &lt;alboukadel.kassambara@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-18 10:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='fastqc'>Run FastQC Tool</h2><span id='topic+fastqc'></span>

<h3>Description</h3>

<p>Run FastQC Tool
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastqc(
  fq.dir = getwd(),
  qc.dir = NULL,
  threads = 4,
  fastqc.path = "~/bin/FastQC/fastqc"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastqc_+3A_fq.dir">fq.dir</code></td>
<td>
<p>path to the directory containing fastq files. Default is the
current working directory.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_qc.dir">qc.dir</code></td>
<td>
<p>path to the FastQC result directory. If NULL, a directory
named fastqc_results is created in the current working directory.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_threads">threads</code></td>
<td>
<p>the number of threads to be used. Default is 4.</p>
</td></tr>
<tr><td><code id="fastqc_+3A_fastqc.path">fastqc.path</code></td>
<td>
<p>path to fastqc program</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Create a directory containing the reports
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Run FastQC: generates a QC directory
fastqc(fq.dir)

## End(Not run)
</code></pre>

<hr>
<h2 id='fastqc_install'>Install FastQC Tool</h2><span id='topic+fastqc_install'></span>

<h3>Description</h3>

<p>Install the FastQC Tool. To be used only on Unix system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastqc_install(url, dest.dir = "~/bin")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fastqc_install_+3A_url">url</code></td>
<td>
<p>url to download the latest version. If missing, the function will
try to install the latest version from
<a href="https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc">https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc</a>.</p>
</td></tr>
<tr><td><code id="fastqc_install_+3A_dest.dir">dest.dir</code></td>
<td>
<p>destination directory to install the tool.</p>
</td></tr>
</table>

<hr>
<h2 id='qc_aggregate'>Aggregate FastQC Reports for Multiple Samples</h2><span id='topic+qc_aggregate'></span><span id='topic+summary.qc_aggregate'></span><span id='topic+qc_stats'></span>

<h3>Description</h3>

<p>Aggregate multiple FastQC reports into a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_aggregate(qc.dir = ".", progressbar = TRUE)

## S3 method for class 'qc_aggregate'
summary(object, ...)

qc_stats(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_aggregate_+3A_qc.dir">qc.dir</code></td>
<td>
<p>path to the FastQC result directory to scan.</p>
</td></tr>
<tr><td><code id="qc_aggregate_+3A_progressbar">progressbar</code></td>
<td>
<p>logical value. If TRUE, shows a progress bar.</p>
</td></tr>
<tr><td><code id="qc_aggregate_+3A_object">object</code></td>
<td>
<p>an object of class qc_aggregate.</p>
</td></tr>
<tr><td><code id="qc_aggregate_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><strong>qc_aggregate()</strong> returns an object of class qc_aggregate
which is a (tibble) data frame with the following column names:
</p>

<ul>
<li><p> sample: sample names
</p>
</li>
<li><p> module: fastqc modules
</p>
</li>
<li><p> status: fastqc module status for each sample
</p>
</li>
<li><p> tot.seq: total sequences (i.e.: the number of reads)
</p>
</li>
<li><p> seq.length: sequence length
</p>
</li>
<li><p> pct.gc: % of GC content
</p>
</li>
<li><p> pct.dup: % of duplicate reads
</p>
</li></ul>

</li>
<li> <p><strong>summary</strong>: Generates a summary of qc_aggregate.
Returns a data frame with the following columns:
</p>

<ul>
<li><p> module: fastqc modules
</p>
</li>
<li><p> nb_samples: the number of samples tested
</p>
</li>
<li><p> nb_pass, nb_fail, nb_warn: the number of samples that passed, failed and warned, respectively.
</p>
</li>
<li><p> failed, warned: the name of samples that failed and warned, respectively.
</p>
</li></ul>

</li>
<li> <p><strong>qc_stats</strong>: returns a data frame containing general statistics of fastqc reports.
columns are: sample, pct.dup, pct.gc, tot.seq and seq.length.
</p>
</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>qc_aggregate()</code>: Aggregate FastQC Reports for Multiple Samples
</p>
</li>
<li> <p><code>qc_stats()</code>: Creates general statistics of fastqc reports.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'># Demo QC dir
qc.dir &lt;- system.file("fastqc_results", package = "fastqcr")
qc.dir

# List of files in the directory
list.files(qc.dir)

# Aggregate the report
qc &lt;- qc_aggregate(qc.dir, progressbar = FALSE)
qc

# Generates a summary of qc_aggregate
summary(qc)

# General statistics of fastqc reports.
qc_stats(qc)

</code></pre>

<hr>
<h2 id='qc_fails'>Inspect Problems in Aggregated FastQC Reports</h2><span id='topic+qc_fails'></span><span id='topic+qc_warns'></span><span id='topic+qc_problems'></span>

<h3>Description</h3>

<p>Inspect problems in aggregated FastQC reports.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_fails(object, element = c("sample", "module"), compact = TRUE)

qc_warns(object, element = c("sample", "module"), compact = TRUE)

qc_problems(
  object,
  element = c("sample", "module"),
  name = NULL,
  status = c("FAIL", "WARN"),
  compact = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_fails_+3A_object">object</code></td>
<td>
<p>an object of class qc_aggregate.</p>
</td></tr>
<tr><td><code id="qc_fails_+3A_element">element</code></td>
<td>
<p>character vector specifying which element to check for 
inspecting problems. Allowed values are one of c(&quot;sample&quot;, &quot;module&quot;). 
Default is &quot;sample&quot;. </p>
 <ul>
<li><p> If &quot;sample&quot;, shows samples with more 
failed and/or warned modules </p>
</li>
<li><p> If &quot;module&quot;, shows moduled that failed 
and/or warned in the most samples </p>
</li></ul>
</td></tr>
<tr><td><code id="qc_fails_+3A_compact">compact</code></td>
<td>
<p>logical value. If TRUE, returns a compact output format; 
otherwise, returns a stretched format.</p>
</td></tr>
<tr><td><code id="qc_fails_+3A_name">name</code></td>
<td>
<p>character vector containing the names of modules and/or samples of
interest. See <a href="#topic+qc_read">qc_read</a> for valid module names. If name specified, a 
stretched output format is returned by default unless you explicitly indicate
compact = TRUE.</p>
</td></tr>
<tr><td><code id="qc_fails_+3A_status">status</code></td>
<td>
<p>character vector specifying the module status. Allowed values 
includes one or the combination of c(&quot;FAIL&quot;, &quot;WARN&quot;). If status = &quot;FAIL&quot;, 
only modules with failed status are returned.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><strong>qc_problems(), qc_fails(), qc_warns()</strong>: returns a tibble (data frame) containing samples
that had one or more modules with failure or warning.
The format and the interpretation of the results depend on the argument 'element',
which value is one of c(&quot;sample&quot;, &quot;module&quot;).
</p>

<ul>
<li> <p><strong>If element = &quot;sample&quot; (default)</strong>, results are samples with failed and/or warned modules. 
The results contain the following columns: sample (sample names),
nb_problems (the number of modules with problems), module (the name of modules with problems).
</p>
</li>
<li> <p><strong>If element = &quot;module&quot;</strong>, results are modules that failed and/or warned in the most samples. 
The results contain the following columns:
module (the name of module with problems), nb_problems (the number of samples with problems),
sample (the name of samples with problems)
</p>
</li></ul>

</li></ul>



<h3>Functions</h3>


<ul>
<li> <p><code>qc_fails()</code>: Displays which samples had one or more failed modules. Use
qc_fails(qc, &quot;module&quot;) to see which modules failed in the most samples.
</p>
</li>
<li> <p><code>qc_warns()</code>: Displays which samples had one or more warned modules. Use
qc_warns(qc, &quot;module&quot;) to see which modules warned in the most samples.
</p>
</li>
<li> <p><code>qc_problems()</code>: Union of <code>qc_fails()</code> and <code>qc_warns()</code>. 
Display which samples or modules that failed or warned.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'># Demo QC dir
qc.dir &lt;- system.file("fastqc_results", package = "fastqcr")
qc.dir
# List of files in the directory
list.files(qc.dir)

# Aggregate the report
qc &lt;- qc_aggregate(qc.dir, progressbar = FALSE)

# Display samples with failed modules
qc_fails(qc)
qc_fails(qc, compact = FALSE)

# Display samples with warned modules
qc_warns(qc)

# Module failed in the most samples
qc_fails(qc, "module")
qc_fails(qc, "module", compact = FALSE)

# Specify a module of interest
qc_problems(qc, "module",  name = "Per sequence GC content")

</code></pre>

<hr>
<h2 id='qc_plot'>Plot FastQC Results</h2><span id='topic+qc_plot'></span><span id='topic+print.qctable'></span>

<h3>Description</h3>

<p>Plot FastQC data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_plot(qc, modules = "all")

## S3 method for class 'qctable'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_plot_+3A_qc">qc</code></td>
<td>
<p>An object of class qc_read or a path to the sample zipped fastqc result file.</p>
</td></tr>
<tr><td><code id="qc_plot_+3A_modules">modules</code></td>
<td>
<p>Character vector containing the names of fastqc modules for
which you want to import the data. Default is all. Allowed values include
one or the combination of:
</p>

<ul>
<li><p> &quot;Summary&quot;,
</p>
</li>
<li><p> &quot;Basic Statistics&quot;,
</p>
</li>
<li><p> &quot;Per base sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per sequence quality scores&quot;,
</p>
</li>
<li><p> &quot;Per base sequence content&quot;,
</p>
</li>
<li><p> &quot;Per sequence GC content&quot;,
</p>
</li>
<li><p> &quot;Per base N content&quot;,
</p>
</li>
<li><p> &quot;Sequence Length Distribution&quot;,
</p>
</li>
<li><p> &quot;Sequence Duplication Levels&quot;,
</p>
</li>
<li><p> &quot;Overrepresented sequences&quot;,
</p>
</li>
<li><p> &quot;Adapter Content&quot;,
</p>
</li>
<li><p> &quot;Kmer Content&quot;
</p>
</li></ul>

<p>Partial match of module names allowed. For example,
you can use modules = &quot;GC content&quot;, instead of the full names modules = &quot;Per sequence GC content&quot;.</p>
</td></tr>
<tr><td><code id="qc_plot_+3A_x">x</code></td>
<td>
<p>an object of class qctable.</p>
</td></tr>
<tr><td><code id="qc_plot_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of ggplots containing the plot for specified modules..
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Demo file
qc.file &lt;- system.file("fastqc_results", "S1_fastqc.zip",  package = "fastqcr")
qc.file
# Read all modules
qc &lt;- qc_read(qc.file)

# Plot per sequence GC content
qc_plot(qc, "Per sequence GC content")

# Per base sequence quality
qc_plot(qc, "Per base sequence quality")

# Per sequence quality scores
qc_plot(qc, "Per sequence quality scores")

# Per base sequence content
qc_plot(qc, "Per base sequence content")

# Sequence duplication levels
qc_plot(qc, "Sequence duplication levels")


</code></pre>

<hr>
<h2 id='qc_plot_collection'>Plot FastQC Results of multiple samples</h2><span id='topic+qc_plot_collection'></span>

<h3>Description</h3>

<p>Plot FastQC data of multiple samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_plot_collection(qc, modules = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_plot_collection_+3A_qc">qc</code></td>
<td>
<p>An object of class qc_read_collection or a path to the sample zipped fastqc result files.</p>
</td></tr>
<tr><td><code id="qc_plot_collection_+3A_modules">modules</code></td>
<td>
<p>Character vector containing the names of fastqc modules for
which you want to import the data. Default is all. Allowed values include
one or the combination of:
</p>

<ul>
<li><p> &quot;Summary&quot;,
</p>
</li>
<li><p> &quot;Basic Statistics&quot;,
</p>
</li>
<li><p> &quot;Per base sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per sequence quality scores&quot;,
</p>
</li>
<li><p> &quot;Per base sequence content&quot;,
</p>
</li>
<li><p> &quot;Per sequence GC content&quot;,
</p>
</li>
<li><p> &quot;Per base N content&quot;,
</p>
</li>
<li><p> &quot;Sequence Length Distribution&quot;,
</p>
</li>
<li><p> &quot;Sequence Duplication Levels&quot;,
</p>
</li>
<li><p> &quot;Overrepresented sequences&quot;,
</p>
</li>
<li><p> &quot;Adapter Content&quot;,
</p>
</li>
<li><p> &quot;Kmer Content&quot;
</p>
</li></ul>

<p>Partial match of module names allowed. For example,
you can use modules = &quot;GC content&quot;, instead of the full names modules = &quot;Per sequence GC content&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of ggplots containing the plot for specified modules..
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ahmed, <a href="mailto:mahmoud.s.fahmy@students.kasralainy.edu.eg">mahmoud.s.fahmy@students.kasralainy.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qc.dir &lt;- system.file("fastqc_results", package = "fastqcr")
qc.files &lt;- list.files(qc.dir, full.names = TRUE)[1:2]
nb_samples &lt;- length(qc.files)

# read specific modules in all files
# To read all modules, specify: modules = "all"
modules &lt;- c(
  "Per sequence GC content",
  "Per base sequence quality",
 "Per sequence quality scores"
)
qc &lt;- qc_read_collection(qc.files, sample_names = paste('S', 1:nb_samples, sep = ''),
       modules = modules)

# Plot per sequence GC content
qc_plot_collection(qc, "Per sequence GC content")

# Per base sequence quality
qc_plot_collection(qc, "Per base sequence quality")

# Per sequence quality scores
qc_plot_collection(qc, "Per sequence quality scores")


</code></pre>

<hr>
<h2 id='qc_read'>Read FastQC Data</h2><span id='topic+qc_read'></span>

<h3>Description</h3>

<p>Read FastQC data into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_read(file, modules = "all", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_read_+3A_file">file</code></td>
<td>
<p>Path to the file to be imported. Can be the path to either :
</p>

<ul>
<li><p> the fastqc zipped file (e.g.: 'path/to/samplename_fastqc.zip'). No need to unzip,
</p>
</li>
<li><p> or the unzipped folder name (e.g.: 'path/to/samplename_fastqc'),
</p>
</li>
<li><p> or the sample name (e.g.: 'path/to/samplename' )
</p>
</li>
<li><p> or the fastqc_data.txt file,
</p>
</li></ul>
</td></tr>
<tr><td><code id="qc_read_+3A_modules">modules</code></td>
<td>
<p>Character vector containing the names of FastQC modules for
which you want to import/inspect the data. Default is all. Allowed values include
one or the combination of:
</p>

<ul>
<li><p> &quot;Summary&quot;,
</p>
</li>
<li><p> &quot;Basic Statistics&quot;,
</p>
</li>
<li><p> &quot;Per base sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per tile sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per sequence quality scores&quot;,
</p>
</li>
<li><p> &quot;Per base sequence content&quot;,
</p>
</li>
<li><p> &quot;Per sequence GC content&quot;,
</p>
</li>
<li><p> &quot;Per base N content&quot;,
</p>
</li>
<li><p> &quot;Sequence Length Distribution&quot;,
</p>
</li>
<li><p> &quot;Sequence Duplication Levels&quot;,
</p>
</li>
<li><p> &quot;Overrepresented sequences&quot;,
</p>
</li>
<li><p> &quot;Adapter Content&quot;,
</p>
</li>
<li><p> &quot;Kmer Content&quot;
</p>
</li></ul>

<p>Partial match of module names allowed. For example,
you can use modules = &quot;GC content&quot;, instead of the full names modules = &quot;Per sequence GC content&quot;.</p>
</td></tr>
<tr><td><code id="qc_read_+3A_verbose">verbose</code></td>
<td>
<p>logical value. If TRUE, print filename when reading.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of tibbles containing the data for specified modules.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Demo file
qc.file &lt;- system.file("fastqc_results", "S1_fastqc.zip",  package = "fastqcr")
qc.file
# Read all modules
qc_read(qc.file)

# Read a specified module
qc_read(qc.file,"Per base sequence quality")

</code></pre>

<hr>
<h2 id='qc_read_collection'>Read a collection of FastQC data files</h2><span id='topic+qc_read_collection'></span>

<h3>Description</h3>

<p>A wrapper function around <a href="#topic+qc_read">qc_read</a> to read multiple FastQC data files at once.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_read_collection(files, sample_names, modules = "all", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_read_collection_+3A_files">files</code></td>
<td>
<p>A <code>character</code> vector of paths to the files to be imported.</p>
</td></tr>
<tr><td><code id="qc_read_collection_+3A_sample_names">sample_names</code></td>
<td>
<p>A <code>character</code> vector of length equals that of the first argument <code>files</code></p>
</td></tr>
<tr><td><code id="qc_read_collection_+3A_modules">modules</code></td>
<td>
<p>Character vector containing the names of FastQC modules for
which you want to import/inspect the data. Default is all. Allowed values include
one or the combination of:
</p>

<ul>
<li><p> &quot;Summary&quot;,
</p>
</li>
<li><p> &quot;Basic Statistics&quot;,
</p>
</li>
<li><p> &quot;Per base sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per tile sequence quality&quot;,
</p>
</li>
<li><p> &quot;Per sequence quality scores&quot;,
</p>
</li>
<li><p> &quot;Per base sequence content&quot;,
</p>
</li>
<li><p> &quot;Per sequence GC content&quot;,
</p>
</li>
<li><p> &quot;Per base N content&quot;,
</p>
</li>
<li><p> &quot;Sequence Length Distribution&quot;,
</p>
</li>
<li><p> &quot;Sequence Duplication Levels&quot;,
</p>
</li>
<li><p> &quot;Overrepresented sequences&quot;,
</p>
</li>
<li><p> &quot;Adapter Content&quot;,
</p>
</li>
<li><p> &quot;Kmer Content&quot;
</p>
</li></ul>

<p>Partial match of module names allowed. For example,
you can use modules = &quot;GC content&quot;, instead of the full names modules = &quot;Per sequence GC content&quot;.</p>
</td></tr>
<tr><td><code id="qc_read_collection_+3A_verbose">verbose</code></td>
<td>
<p>logical value. If TRUE, print filename when reading.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>list</code> of <code>tibbles</code> containing the data of specified modules form each file.
</p>


<h3>Author(s)</h3>

<p>Mahmoud Ahmed, <a href="mailto:mahmoud.s.fahmy@students.kasralainy.edu.eg">mahmoud.s.fahmy@students.kasralainy.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># extract paths to the demo files
qc.dir &lt;- system.file("fastqc_results", package = "fastqcr")
qc.files &lt;- list.files(qc.dir, full.names = TRUE)[1:2]
nb_samples &lt;- length(qc.files)


# read a specified module in all files
# To read all modules, specify: modules = "all"
qc &lt;- qc_read_collection(qc.files, 
    sample_names = paste('S', 1:nb_samples, sep = ''),
    modules = "Per base sequence quality")

</code></pre>

<hr>
<h2 id='qc_report'>Build a QC Report</h2><span id='topic+qc_report'></span>

<h3>Description</h3>

<p>Create an HTML file containing FastQC reports of one or multiple 
files. Inputs can be either a directory containing multiple FastQC reports 
or a single sample FastQC report.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_report(
  qc.path,
  result.file,
  experiment = NULL,
  interpret = FALSE,
  template = NULL,
  preview = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_report_+3A_qc.path">qc.path</code></td>
<td>
<p>path to the FastQC reports. Allowed values include: </p>
 
<ul>
<li><p> A path to a directory containing multiple zipped FastQC reports, </p>
</li>
<li>
<p>Or a single sample zipped FastQC report.  Partial match is allowed for sample name.</p>
</li></ul>
</td></tr>
<tr><td><code id="qc_report_+3A_result.file">result.file</code></td>
<td>
<p>path to the result file prefix (e.g., path/to/qc-result).
Don't add the file extension.</p>
</td></tr>
<tr><td><code id="qc_report_+3A_experiment">experiment</code></td>
<td>
<p>text specifying a short description of the experiment. For 
example experiment = &quot;RNA sequencing of colon cancer cell lines&quot;.</p>
</td></tr>
<tr><td><code id="qc_report_+3A_interpret">interpret</code></td>
<td>
<p>logical value. If TRUE, adds the interpretation of each 
module.</p>
</td></tr>
<tr><td><code id="qc_report_+3A_template">template</code></td>
<td>
<p>a character vector specifying the path to an Rmd template. 
file.</p>
</td></tr>
<tr><td><code id="qc_report_+3A_preview">preview</code></td>
<td>
<p>logical value. If TRUE, shows a preview of the report.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Demo QC Directory
qc.path &lt;- system.file("fastqc_results", package = "fastqcr")
qc.path

# List of files in the directory
list.files(qc.path)

# Multi QC report
qc_report(qc.path, result.file = "~/Desktop/result")

# QC Report of one sample with plot interpretation
 qc.file &lt;- system.file("fastqc_results", "S1_fastqc.zip", package = "fastqcr")
 qc_report(qc.file, result.file = "~/Desktop/result",
   interpret = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='qc_unzip'>Unzip Files in the FastQC Result Directory</h2><span id='topic+qc_unzip'></span>

<h3>Description</h3>

<p>Unzip all files in the FastQC result directory. Default is the
current working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qc_unzip(qc.dir = ".", rm.zip = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qc_unzip_+3A_qc.dir">qc.dir</code></td>
<td>
<p>Path to the FastQC result directory.</p>
</td></tr>
<tr><td><code id="qc_unzip_+3A_rm.zip">rm.zip</code></td>
<td>
<p>logical. If TRUE, remove zipped files after extraction. Default
is TRUE.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
qc_unzip("FASTQC")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
