<!DOCTYPE html><html lang="en"><head><title>Help for package fastJT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fastJT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fastJT-package'><p>Efficient Jonckheere-Terpstra Test Statistics</p></a></li>
<li><a href='#fastJT'>
<p>Compute the Jonckheere-Terpstra Test Statistics</p></a></li>
<li><a href='#fastJT.select'>
<p>Conduct k-Fold Cross-Validation based on Jonckheere-Terpstra Test Statistics</p></a></li>
<li><a href='#pvalues'>
<p>Compute P-values Based on Jonckheere-Terpstra Test Statistics</p></a></li>
<li><a href='#summary.fastJT'>
<p>Summarize Jonckheere-Terpstra Test Statistics and P-Values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Efficient Jonckheere-Terpstra Test Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-04</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiaxing Lin, Alexander Sibley, Ivo Shterev, and Kouros Owzar</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Sibley &lt;dcibioinformatics@duke.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This 'Rcpp'-based package implements highly efficient functions for the calculation of the Jonckheere-Terpstra statistic. It can be used for a variety of applications, including feature selection in machine learning problems, or to conduct genome-wide association studies (GWAS) with multiple quantitative phenotypes. The code leverages 'OpenMP' directives for multi-core computing to reduce overall processing time. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.3)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-05 00:04:20 UTC; zs68</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-10 15:10:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='fastJT-package'>Efficient Jonckheere-Terpstra Test Statistics
</h2><span id='topic+fastJT-package'></span>

<h3>Description</h3>

<p>This <code>Rcpp</code>-based package implements highly efficient functions for the calculation of the Jonkheere-Terpstra statistic. It can be used for a variety of applications, including feature selection in machine learning problems, or to conduct genome-wide association studies (GWAS) with multiple quantitative phenotypes. The code leverages <code>OpenMP</code> directives for multi-core computing to reduce overall processing time.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> fastJT</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.6</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-11-04</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Please see the example function calls below, or refer to the individual function documentation or the included vignette for more information. The package vignette serves as a tutorial for using this package. The technical details are provided in the reference cited below. Specifically, the calculation of the standardized test statistics employs the null variance equation as defined by Hollander and Wolfe (1999, eq. 6.19) to account for ties in the data.
</p>


<h3>Author(s)</h3>

<p>Jiaxing Lin, Alexander Sibley, Ivo Shterev, and Kouros Owzar
</p>
<p>Maintainer: Alexander Sibley &lt;dcibioinformatics@duke.edu&gt;
</p>


<h3>References</h3>

<p>Hollander, M. and Wolfe, D. A. (1999) <em>Nonparametric Statistical Methods</em>. New York: Wiley, 2nd edition.
</p>


<h3>See Also</h3>

<p><code>Rcpp</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate dummy data	
num_sample  &lt;- 100
num_marker  &lt;- 10
num_feature &lt;- 500
set.seed(12345)
Data &lt;- matrix(rnorm(num_sample*num_marker), num_sample, num_marker)
Features &lt;- matrix(rbinom(num_sample*num_feature, 2, 0.5), num_sample, num_feature)
colnames(Data) &lt;- paste0("Var:",1:num_marker)
colnames(Features) &lt;- paste0("Ftr:",1:num_feature)
   
res &lt;- fastJT(Y=Data, X=Features, outTopN=15)
res
res &lt;- fastJT.select(Y=Data, X=Features, cvMesh=NULL, kFold=5, 
                     selCrit=NULL, outTopN=5, numThreads=1)
res
</code></pre>

<hr>
<h2 id='fastJT'>
Compute the Jonckheere-Terpstra Test Statistics
</h2><span id='topic+fastJT'></span>

<h3>Description</h3>

<p>A method to compute the Jonckheere-Terpstra test statistics for large numbers of dependent and independent variables, with optional multi-threaded execution. The calculation of the standardized test statistics employs the null variance equation as defined by Hollander and Wolfe (1999, eq. 6.19) to account for ties in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastJT(Y, X, outTopN=15L, numThreads=1L, standardized=TRUE)  	
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fastJT_+3A_y">Y</code></td>
<td>
<p>A matrix of continuous values, representing dependent variables, e.g. marker levels or other observed values. Row names should be sample IDs, and column names should be variable names. Required.
</p>
</td></tr>
<tr><td><code id="fastJT_+3A_x">X</code></td>
<td>
<p>A matrix of integer values, representing independent variables, e.g. SNP counts or other classification features. Row names should be sample IDs, and column names should be feature IDs. Required.
</p>
</td></tr>
<tr><td><code id="fastJT_+3A_outtopn">outTopN</code></td>
<td>
<p>An integer to indicate the number of top independent variables to be reported for each dependent variable, based on the standardized Jonckheere-Terpstra test statistics. Optional. The default value is <code>15L</code>. If set to <code>NA</code>, all results are returned.
</p>
</td></tr>
<tr><td><code id="fastJT_+3A_numthreads">numThreads</code></td>
<td>
<p>A integer to indicate the number of threads used in the computation. Optional. The default value is <code>1L</code> (sequential computation).
</p>
</td></tr>
<tr><td><code id="fastJT_+3A_standardized">standardized</code></td>
<td>
<p>A boolean to specify whether to return standardized Jonckheere-Terpstra statistics (<code>TRUE</code>) or non-standardized statistics (<code>FALSE</code>). Optional. The default value is <code>TRUE</code>.
</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A <code>list</code> with two objects
</p>
<table role = "presentation">
<tr><td><code>J</code></td>
<td>
<p>A matrix of the standardized/non-standardized Jonckheere-Terpstra test statistics,
depending on the value of the <code>standardized</code> argument.
</p>
</td></tr>
<tr><td><code>XIDs</code></td>
<td>
<p>If <code>outTopN</code> was specified, this object is a matrix of the column IDs of <code>X</code> corresponding to the top standardized Jonckheere-Terpstra test statistics for each dependent variable. Otherwise this is a vector of the column IDs of <code>X</code>.
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Rows (samples) are assumed to be in the same order in <code>X</code> and <code>Y</code>.
</p>


<h3>References</h3>

<p>Hollander, M. and Wolfe, D. A. (1999) <em>Nonparametric Statistical Methods</em>. New York: Wiley, 2nd edition.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate dummy data	
num_sample &lt;- 100
num_marker &lt;- 10
num_SNP		 &lt;- 500
set.seed(12345)
Mark &lt;- matrix(rnorm(num_sample*num_marker), num_sample, num_marker)
Geno &lt;- matrix(rbinom(num_sample*num_SNP, 2, 0.5), num_sample, num_SNP)
colnames(Mark) &lt;- paste0("Mrk:",1:num_marker)
colnames(Geno) &lt;- paste0("SNP:",1:num_SNP)

res &lt;- fastJT(Y=Mark, X=Geno, outTopN=5)
res
res &lt;- fastJT(Y=Mark, X=Geno, outTopN=NA)
head(res) 
</code></pre>

<hr>
<h2 id='fastJT.select'>
Conduct k-Fold Cross-Validation based on Jonckheere-Terpstra Test Statistics
</h2><span id='topic+fastJT.select'></span>

<h3>Description</h3>

<p>A method to conduct <code>k</code>-fold cross-validation based the Jonckheere-Terpstra test statistics for large numbers of dependent and independent variables, with optional multi-threaded execution. The data are divided into <code>k</code> subsets, one of which is withheld while the remaining <code>k</code>-1 subsets are used to test the features. The process is repeated <code>k</code> times, with each of the subsets withheld exactly once as the validation data. The calculation of the standardized test statistics employs the null variance equation as defined by Hollander and Wolfe (1999, eq. 6.19) to account for ties in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastJT.select(Y, X, cvMesh=NULL, kFold=10L, selCrit=NULL, outTopN=15L, numThreads=1L)  	
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fastJT.select_+3A_y">Y</code></td>
<td>
<p>A matrix of continuous values, representing dependent variables, e.g. marker levels or other observed values. Row names should be sample IDs, and column names should be variable names. Required.
</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_x">X</code></td>
<td>
<p>A matrix of integer values, representing independent variables, e.g. SNP counts or other classification features. Row names should be sample IDs, and column names should be feature IDs. Required.
</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_cvmesh">cvMesh</code></td>
<td>
<p>A user-defined function to specify how to separate the data into training and testing parts. The inputs of this function are a vector representing the sample IDs and <code>kFold</code>, an integer representing the number of folds for cross validation. The output of this function is a list of <code>kFold</code> vectors of sample IDs forming the testing subset for each fold. The default value is <code>NULL</code>, and if no function is specified, the data are partitioned sequentially into <code>kFold</code> equal sized subsets. Optional.
</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_kfold">kFold</code></td>
<td>
<p>An integer to indicate the number of folds. Optional. The default value is 10.
</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_selcrit">selCrit</code></td>
<td>
<p>A user-defined function to specify the criteria for selecting the top features. The inputs of this function are <code>J</code>, the matrix of statistics resulting from <code>fastJT</code>, and <code>P</code>, the matrix of p-values from <code>pvalues(J)</code>. The output is a data frame containing the selected feature ID for each trait of interest. Optional. The default value is <code>NULL</code>, and if no function is specified, the features with the largest standardized Jonckheere-Terpstra test statistics are selected.</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_outtopn">outTopN</code></td>
<td>
<p>An integer to indicate the number of top hits to be returned when <code>selCrit=NULL</code>. Unused if <code>selCrit!=NULL</code>. Optional. The default value is <code>15L</code>.
</p>
</td></tr>
<tr><td><code id="fastJT.select_+3A_numthreads">numThreads</code></td>
<td>
<p>A integer to indicate the number of threads to be used in the computation. Optional. The default value is <code>1L</code> (sequential computation).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Three <code>lists</code> of length <code>kFold</code>
</p>
<table role = "presentation">
<tr><td><code>J</code></td>
<td>
<p>A <code>list</code> of matrices of standardized Jonckheere-Terpstra test statistics, one for each cross validation.</p>
</td></tr>
<tr><td><code>Pval</code></td>
<td>
<p>A <code>list</code> of matrices of p-values, one for each cross validation.</p>
</td></tr>
<tr><td><code>XIDs</code></td>
<td>
<p>A <code>list</code> of matrices of the selected feature IDs, one for each cross validation.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Rows (samples) are assumed to be in the same order in <code>X</code> and <code>Y</code>.
</p>


<h3>References</h3>

<p>Hollander, M. and Wolfe, D. A. (1999) <em>Nonparametric Statistical Methods</em>. New York: Wiley, 2nd edition.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastJT">fastJT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate dummy data	
num_sample  &lt;- 100
num_marker  &lt;- 10
num_feature &lt;- 500
set.seed(12345)
Data &lt;- matrix(rnorm(num_sample*num_marker), num_sample, num_marker)
Features &lt;- matrix(rbinom(num_sample*num_feature, 2, 0.5), num_sample, num_feature)
colnames(Data) &lt;- paste0("Var:",1:num_marker)
colnames(Features) &lt;- paste0("Ftr:",1:num_feature)

res &lt;- fastJT.select(Y=Data, X=Features, cvMesh=NULL, kFold=5, 
                     selCrit=NULL, outTopN=5, numThreads=1)
res 
</code></pre>

<hr>
<h2 id='pvalues'>
Compute P-values Based on Jonckheere-Terpstra Test Statistics
</h2><span id='topic+pvalues'></span>

<h3>Description</h3>

<p>Method to compute the p-values for results from <code>fastJT</code>. <code>fastJT</code> must be run with <code>standardized=TRUE</code> in order to use this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalues(object)  	
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pvalues_+3A_object">object</code></td>
<td>
<p>A <code>fastJT</code> object that is the return of method <code>fastJT</code>. Required.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of p-values with the same dimensions as the standardized statistics from <code>fastJT</code>. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastJT">fastJT</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate dummy data	
num_patient &lt;- 100
num_marker 	&lt;- 10
num_SNP     &lt;- 500
set.seed(12345)
Mark &lt;- matrix(rnorm(num_patient*num_marker), num_patient, num_marker)
Geno &lt;- matrix(rbinom(num_patient*num_SNP, 2, 0.5), num_patient, num_SNP)
colnames(Mark) &lt;- paste0("Mrk:",1:num_marker)
colnames(Geno) &lt;- paste0("SNP:",1:num_SNP)

res &lt;- fastJT(Y=Mark, X=Geno, outTopN=5)
pvalues(res)
res &lt;- fastJT(Y=Mark, X=Geno, outTopN=NA)
pvalues(res)
</code></pre>

<hr>
<h2 id='summary.fastJT'>
Summarize Jonckheere-Terpstra Test Statistics and P-Values
</h2><span id='topic+summary.fastJT'></span>

<h3>Description</h3>

<p>Summary method for fastJT results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fastJT'
summary(object, Y2Print=1:10, X2Print=1:10, printP=TRUE, 
               outTopN=NA, subObj=FALSE, ...)	
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.fastJT_+3A_object">object</code></td>
<td>
<p>A <code>fastJT</code> object that is the return of method <code>fastJT</code>. Required.
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_y2print">Y2Print</code></td>
<td>
<p>Either a numeric or character vector that indicates the desired dependent variables to print. The default is 1:10. Set to <code>NA</code> to print full results. Optional.
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_x2print">X2Print</code></td>
<td>
<p>Either a numeric or character vector that indicates the desired independent variables to print. If <code>outTopN=NA</code> in function <code>fastJT</code>, i.e., the results are not sorted, both numeric and character vectors can be used to set the print range. The default range is 1:10. Set to <code>NA</code> to print full results. If <code>outTopN!=NA</code> in function <code>fastJT</code>, the range of <code>X2Print</code> refers to the range of top normalized statistics computed in <code>fastJT</code>, and only numeric vectors can be used. Optional.
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_printp">printP</code></td>
<td>
<p>A boolean indicating whether to print the p-values (<code>TRUE</code>) or the standardized statistics (<code>FALSE</code>). The default value is <code>TRUE</code>. Optional.
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_outtopn">outTopN</code></td>
<td>
<p>An integer specifying the number of top hits to print in the summary, if the statistics were not sorted during the evaluation of the statistics (i.e., if <code>outTopN=NA</code> in function <code>fastJT</code>). Optional. The default value is <code>NA</code>.
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_subobj">subObj</code></td>
<td>
<p>A boolean indicating whether to return a <code>fastJT</code> object subset per the requested summary (<code>TRUE</code>). Optional. The default value is <code>FALSE</code> (nothing is returned).
</p>
</td></tr>
<tr><td><code id="summary.fastJT_+3A_...">...</code></td>
<td>
<p>Additional arguments affecting the summary produced.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>subObj=TRUE</code>, this method returns a <code>fastJT</code> object matching the subset of statistics from <code>object</code> that are being printed. For example, if <code>object</code> is not sorted by top hits, <code>summary(object, outTopN=10, subObj=TRUE)</code> will print the summary and return a subset of <code>object</code> that contains only the top 10 independent variables for each dependent variable. If <code>subObj=FALSE</code>, the summary is printed but no values are returned.
</p>


<h3>Note</h3>

<p>This function prints a matrix or paired columns of independent variable IDs and statistics/p-values to the log.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fastJT">fastJT</a></code>, <code><a href="#topic+pvalues">pvalues</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate dummy data	
num_patient &lt;- 100
num_marker 	&lt;- 10
num_SNP     &lt;- 500
set.seed(12345);
Mark &lt;- matrix(rnorm(num_patient*num_marker), num_patient, num_marker)
Geno &lt;- matrix(rbinom(num_patient*num_SNP, 2, 0.5), num_patient, num_SNP)
colnames(Mark) &lt;- paste0("Mrk:",1:num_marker)
colnames(Geno) &lt;- paste0("SNP:",1:num_SNP)

res &lt;- fastJT(Y=Mark, X=Geno, outTopN=5)
summary(res, Y2Print=c("Mrk:1","Mrk:2"), X2Print=1:5, printP=FALSE)
summary(res, Y2Print=NA, X2Print=1:5, printP=FALSE)

res &lt;- fastJT(Y=Mark, X=Geno, outTopN=NA)
summary(res, Y2Print=1:10, X2Print=1:10, printP=TRUE)
summary(res, Y2Print=c("Mrk:1","Mrk:2"), X2Print=c("SNP:1","SNP:2"), printP=TRUE) 

res &lt;- fastJT(Y=Mark, X=Geno, outTopN=NA, standardized=FALSE)
summary(res, outTopN=10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
