<!DOCTYPE html><html><head><title>Help for package fbar</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fbar}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#decompose_metabolites'><p>Decompose a metabolite table into the metabolite stub itself and the compartment it is in</p></a></li>
<li><a href='#ecoli_core'><p>A small E. coli model, created from a number of sources.</p></a></li>
<li><a href='#expanded_to_glpk'><p>Parse a long format metabolic model to a glpk model</p></a></li>
<li><a href='#expanded_to_gurobi'><p>Parse a long format metabolic model to a Gurobi model</p></a></li>
<li><a href='#expanded_to_reactiontbl'><p>Convert intermediate expanded format back to a reaction table</p></a></li>
<li><a href='#expanded_to_ROI'><p>Parse a long format metabolic model to an ROI model</p></a></li>
<li><a href='#fbar'><p>fbar: Flux Balance Analysis in R with a tidy data approach</p></a></li>
<li><a href='#find_flux_variability_df'><p>Given a metabolic model as a data frame, return a new data frame with fluxes and variability</p></a></li>
<li><a href='#find_fluxes_df'><p>Given a metabolic model as a data frame, return a new data frame with fluxes</p></a></li>
<li><a href='#gene_associate'><p>Apply gene expressions to reaction table</p></a></li>
<li><a href='#gene_eval'><p>Function to estimate the expression levels of gene sets</p></a></li>
<li><a href='#get_BiGG'><p>Download a model from a BiGG json file</p></a></li>
<li><a href='#iJO1366'><p>A full size E. coli model.</p></a></li>
<li><a href='#nutrient_types'><p>A subset of exchange reactions annotated to indicate typical availability</p></a></li>
<li><a href='#parse_met_list'><p>Internal function: Expand half reaction equations into a long form</p></a></li>
<li><a href='#reactiontbl_to_expanded'><p>Parse a reaction table to an intermediate, long format</p></a></li>
<li><a href='#reactiontbl_to_gurobi'><p>Parse reaction table to Gurobi format</p></a></li>
<li><a href='#recompose_metabolites'><p>Merge metabolite stub and compartment to form an id</p></a></li>
<li><a href='#split_on_arrow'><p>Internal function: Splitting reaction equation into substrate and product</p></a></li>
<li><a href='#validate_expanded'><p>Validate an expanded model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>An Extensible Approach to Flux Balance Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-09-02</td>
</tr>
<tr>
<td>Description:</td>
<td>A toolkit for Flux Balance Analysis and related
    metabolic modeling techniques. Functions are provided for: parsing
    models in tabular format, converting parsed metabolic models to input
    formats for common linear programming solvers, and
    evaluating and applying gene-protein-reaction mappings. In addition, there
    are wrappers to parse a model, select a solver, find the metabolic fluxes,
    and return the results applied to the original model. Compared to other
    packages in this field, this package puts a much heavier focus on
    providing reusable components that can be used in the design of new
    implementation of new techniques, in particular those that involve large
    parameter sweeps. For a background on the theory, see What is Flux Balance 
    Analysis &lt;<a href="https://doi.org/10.1038%2Fnbt.1614">doi:10.1038/nbt.1614</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://maxconway.github.io/fbar/">http://maxconway.github.io/fbar/</a>,
<a href="https://github.com/maxconway/fbar">https://github.com/maxconway/fbar</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/maxconway/fbar/issues">https://github.com/maxconway/fbar/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, dplyr, magrittr, Matrix, purrr, rlang, ROI,
ROI.plugin.ecos, stringr, tibble, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, jsonlite, curl, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-02 13:16:23 UTC; max</td>
</tr>
<tr>
<td>Author:</td>
<td>Max Conway [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Max Conway &lt;conway.max1@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-02 13:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='decompose_metabolites'>Decompose a metabolite table into the metabolite stub itself and the compartment it is in</h2><span id='topic+decompose_metabolites'></span>

<h3>Description</h3>

<p>Decompose a metabolite table into the metabolite stub itself and the compartment it is in
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decompose_metabolites(
  met_table,
  compartment_regex = "(\\[[a-zA-Z0-9]+]$)|(_[a-zA-Z]$)"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decompose_metabolites_+3A_met_table">met_table</code></td>
<td>
<p>A metabolite table, with one column, <code>met</code></p>
</td></tr>
<tr><td><code id="decompose_metabolites_+3A_compartment_regex">compartment_regex</code></td>
<td>
<p>Regular expression to identify compartments in model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a metabolite table with the columns <code>chemical</code> and <code>compartment</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ecoli_core)

mod &lt;- reactiontbl_to_expanded(ecoli_core)

decompose_metabolites(mod$mets)

recompose_metabolites(decompose_metabolites(mod$mets))
</code></pre>

<hr>
<h2 id='ecoli_core'>A small E. coli model, created from a number of sources.</h2><span id='topic+ecoli_core'></span>

<h3>Description</h3>

<p>A small E. coli model, created from a number of sources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecoli_core
</code></pre>


<h3>Format</h3>

<p>A data frame with 95 rows and 7 columns:
</p>

<dl>
<dt>abbreviation</dt><dd><p>an abbreviated reaction name, acts as the reaction id</p>
</dd>
<dt>lowbnd</dt><dd><p>lower bound on the reaction rate</p>
</dd>
<dt>uppbnd</dt><dd><p>upper bound on the reaction rate</p>
</dd>
<dt>obj_coef</dt><dd><p>identifies a reaction (or reactions) for which the maximum possible rate should be found</p>
</dd>
<dt>equation</dt><dd><p>reaction equation</p>
</dd>
<dt>officialName</dt><dd><p>full reaction name</p>
</dd>
<dt>geneAssociation</dt><dd><p>A boolean combination of genes which control the reaction</p>
</dd>
<dt>subsystem</dt><dd><p>an indicator of reaction function</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://bigg.ucsd.edu">http://bigg.ucsd.edu</a>, <a href="https://www.asmscience.org/content/journal/ecosalplus/10.1128/ecosalplus.10.2.1">Reconstruction and Use of Microbial Metabolic Networks: the Core Escherichia coli Metabolic Model as an Educational Guide</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/21988831/">A comprehensive genome-scale reconstruction of Escherichia coli metabolism&ndash;2011.</a>
</p>

<hr>
<h2 id='expanded_to_glpk'>Parse a long format metabolic model to a glpk model</h2><span id='topic+expanded_to_glpk'></span>

<h3>Description</h3>

<p><strong>This function is deprecated. <code>ROI.plugin.glpk</code> is recommended instead.</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expanded_to_glpk(reactions_expanded)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expanded_to_glpk_+3A_reactions_expanded">reactions_expanded</code></td>
<td>
<p>A list of data frames as output by <code>reactiontbl_to_expanded</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This parses the long format produced by <code>reactiontbl_to_expanded</code> to a glpk model.
</p>
<p>To install the Rglpk package in Linux, run <code>sudo apt-get install libglpk-dev</code> in a terminal, and then run <code>install.packages('Rglpk')</code> in R.
</p>
<p>The <code>reaction_table</code> must have columns:
</p>

<ul>
<li> <p><code>abbreviation</code>,
</p>
</li>
<li> <p><code>equation</code>,
</p>
</li>
<li> <p><code>uppbnd</code>,
</p>
</li>
<li> <p><code>lowbnd</code>, and
</p>
</li>
<li> <p><code>obj_coef</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list suitable for input to Rglpk
</p>


<h3>See Also</h3>

<p>Other parsing_and_conversion: 
<code><a href="#topic+expanded_to_ROI">expanded_to_ROI</a>()</code>,
<code><a href="#topic+expanded_to_gurobi">expanded_to_gurobi</a>()</code>,
<code><a href="#topic+reactiontbl_to_expanded">reactiontbl_to_expanded</a>()</code>,
<code><a href="#topic+reactiontbl_to_gurobi">reactiontbl_to_gurobi</a>()</code>
</p>

<hr>
<h2 id='expanded_to_gurobi'>Parse a long format metabolic model to a Gurobi model</h2><span id='topic+expanded_to_gurobi'></span>

<h3>Description</h3>

<p><strong>This function is deprecated. <code>github.com/Fl0Sch/ROI.plugin.gurobi</code> is recommended instead.</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expanded_to_gurobi(reactions_expanded)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expanded_to_gurobi_+3A_reactions_expanded">reactions_expanded</code></td>
<td>
<p>A list of data frames as output by <code>expand_reactions</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Used as the second half of <code><a href="#topic+reactiontbl_to_gurobi">reactiontbl_to_gurobi</a></code>, this parses the long format produced by <code>reactiontbl_to_expanded</code> to a Gurobi model
</p>
<p>For installation instructions for Gurobi, refer to the Gurobi website: <a href="https://www.gurobi.com/">https://www.gurobi.com/</a>.
</p>
<p>The <code>reaction_table</code> must have columns:
</p>

<ul>
<li> <p><code>abbreviation</code>,
</p>
</li>
<li> <p><code>equation</code>,
</p>
</li>
<li> <p><code>uppbnd</code>,
</p>
</li>
<li> <p><code>lowbnd</code>, and
</p>
</li>
<li> <p><code>obj_coef</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list suitable for input to Gurobi.
</p>


<h3>See Also</h3>

<p>Other parsing_and_conversion: 
<code><a href="#topic+expanded_to_ROI">expanded_to_ROI</a>()</code>,
<code><a href="#topic+expanded_to_glpk">expanded_to_glpk</a>()</code>,
<code><a href="#topic+reactiontbl_to_expanded">reactiontbl_to_expanded</a>()</code>,
<code><a href="#topic+reactiontbl_to_gurobi">reactiontbl_to_gurobi</a>()</code>
</p>

<hr>
<h2 id='expanded_to_reactiontbl'>Convert intermediate expanded format back to a reaction table</h2><span id='topic+expanded_to_reactiontbl'></span>

<h3>Description</h3>

<p>Useful for saving a new or edited model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expanded_to_reactiontbl(expanded)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expanded_to_reactiontbl_+3A_expanded">expanded</code></td>
<td>
<p>A list of data frames: </p>

<ul>
<li> <p><code>rxns</code>, which has one row per reaction, 
</p>
</li>
<li> <p><code>mets</code>, which has one row for each metabolite, and 
</p>
</li>
<li> <p><code>stoich</code>, which has one row for each time a metabolite appears in a reaction.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame describing the metabolic model.
</p>

<hr>
<h2 id='expanded_to_ROI'>Parse a long format metabolic model to an ROI model</h2><span id='topic+expanded_to_ROI'></span>

<h3>Description</h3>

<p>This parses the long format produced by <code>reactiontbl_to_expanded</code> to an ROI model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expanded_to_ROI(reactions_expanded)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expanded_to_ROI_+3A_reactions_expanded">reactions_expanded</code></td>
<td>
<p>A list of data frames as output by <code>reactiontbl_to_expanded</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>To solve models using ROI, you will need a solver plugin for ROI. Probably the easiest one to install is ROI.plugin.glpk.
To install this in Linux, run <code>sudo apt-get install libglpk-dev</code> in a terminal, and then run <code>install.packages('ROI.plugin.glpk')</code> in R.
</p>
<p>The <code>reaction_table</code> must have columns:
</p>

<ul>
<li> <p><code>abbreviation</code>,
</p>
</li>
<li> <p><code>equation</code>,
</p>
</li>
<li> <p><code>uppbnd</code>,
</p>
</li>
<li> <p><code>lowbnd</code>, and
</p>
</li>
<li> <p><code>obj_coef</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list suitable for input to ROI.
</p>


<h3>See Also</h3>

<p>Other parsing_and_conversion: 
<code><a href="#topic+expanded_to_glpk">expanded_to_glpk</a>()</code>,
<code><a href="#topic+expanded_to_gurobi">expanded_to_gurobi</a>()</code>,
<code><a href="#topic+reactiontbl_to_expanded">reactiontbl_to_expanded</a>()</code>,
<code><a href="#topic+reactiontbl_to_gurobi">reactiontbl_to_gurobi</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(ecoli_core)
library(dplyr)
try(library(ROI.plugin.ecos)) # make a solver available to ROI

roi_model &lt;- ecoli_core %&gt;%
  reactiontbl_to_expanded %&gt;%
  expanded_to_ROI
  
if(length(ROI::ROI_applicable_solvers(roi_model))&gt;=1){
  roi_result &lt;- ROI::ROI_solve(roi_model)
  
  ecoli_core_with_flux &lt;- ecoli_core %&gt;%
    mutate(flux = roi_result[['solution']])
}

## End(Not run)
</code></pre>

<hr>
<h2 id='fbar'>fbar: Flux Balance Analysis in R with a tidy data approach</h2><span id='topic+fbar'></span>

<h3>Description</h3>

<p>fbar is a simple, easy to use Flux Balance Analysis package with a tidy data approach. 
Just <code>data_frames</code> and the occasional <code>list</code>, no new classes to learn. 
The focus is on simplicity and speed. 
Models are expected as a flat table, and results can be simply appended to the table. 
This makes this package very suitable for use in pipelines with pre- and post- processing of models and results, so that it works well as a backbone for customized methods. 
Loading, parsing and evaluating a model takes around 0.1s, which, together with the straightforward data structures used, makes this library very suitable for large parameter sweeps.
</p>


<h3>Details</h3>

<p>For a list of functions in the package, see <code>vignette('Introduction','fbar')</code>
</p>

<hr>
<h2 id='find_flux_variability_df'>Given a metabolic model as a data frame, return a new data frame with fluxes and variability</h2><span id='topic+find_flux_variability_df'></span>

<h3>Description</h3>

<p>This function calculates fluxes <code>folds</code> times with shuffled versions of the metabolic model.
This is designed to detect and quantify underdetermined fluxes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_flux_variability_df(reaction_table, folds = 10, do_minimization = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_flux_variability_df_+3A_reaction_table">reaction_table</code></td>
<td>
<p>a data frame representing the metabolic model</p>
</td></tr>
<tr><td><code id="find_flux_variability_df_+3A_folds">folds</code></td>
<td>
<p>number of times to calculate fluxes</p>
</td></tr>
<tr><td><code id="find_flux_variability_df_+3A_do_minimization">do_minimization</code></td>
<td>
<p>toggle to uniformly minimize all non-objective fluxes after finding the objective</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses ROI, so to solve models, you will need a solver plugin for ROI. Probably the easiest one to install is ROI.plugin.glpk.
To install this in Linux, run <code>sudo apt-get install libglpk-dev</code> in a terminal, and then run <code>install.packages('ROI.plugin.glpk')</code> in R.
</p>


<h3>Value</h3>

<p>reaction_table with two added columns: sd (the standard deviation of fluxes found) and flux (a typical flux) from this distribution
</p>

<hr>
<h2 id='find_fluxes_df'>Given a metabolic model as a data frame, return a new data frame with fluxes</h2><span id='topic+find_fluxes_df'></span>

<h3>Description</h3>

<p>Given a metabolic model as a data frame, return a new data frame with fluxes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_fluxes_df(reaction_table, do_minimization = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_fluxes_df_+3A_reaction_table">reaction_table</code></td>
<td>
<p>a data frame representing the metabolic model</p>
</td></tr>
<tr><td><code id="find_fluxes_df_+3A_do_minimization">do_minimization</code></td>
<td>
<p>toggle to uniformly minimize all non-objective fluxes after finding the objective</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses ROI, so to solve models, you will need a solver plugin for ROI. Probably the easiest one to install is ROI.plugin.glpk.
To install this in Linux, run <code>sudo apt-get install libglpk-dev</code> in a terminal, and then run <code>install.packages('ROI.plugin.glpk')</code> in R.
</p>


<h3>Value</h3>

<p>The input data frame with a new numeric column, &quot;<code>flux</code>&quot;.
</p>


<h3>See Also</h3>

<p>find_fluxes_vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(ecoli_core)
ecoli_core_with_flux &lt;- find_fluxes_df(ecoli_core)

## End(Not run)
</code></pre>

<hr>
<h2 id='gene_associate'>Apply gene expressions to reaction table</h2><span id='topic+gene_associate'></span>

<h3>Description</h3>

<p>A convenience function that uses <code><a href="#topic+gene_eval">gene_eval</a></code> and a custom function to apply new upper and lower bounds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_associate(
  reaction_table,
  gene_table,
  expression_flux_function = function(x) {     (1 + log(x)/stats::sd(x)^2)^sign(x - 1)
    }
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_associate_+3A_reaction_table">reaction_table</code></td>
<td>
<p>A data frame describing the metabolic model.</p>
</td></tr>
<tr><td><code id="gene_associate_+3A_gene_table">gene_table</code></td>
<td>
<p>A data frame showing gene presence</p>
</td></tr>
<tr><td><code id="gene_associate_+3A_expression_flux_function">expression_flux_function</code></td>
<td>
<p>a function to convert from gene set expression to flux</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the reaction_table, with a new column, present, and altered upper and lower bounds
</p>


<h3>Warning</h3>

<p>This function relies on <code><a href="#topic+gene_eval">gene_eval</a></code>, which uses <code><a href="base.html#topic+eval">eval</a></code> to evaluate gene expression sets. 
This gives flexibility, but means that malicious code in the <code>gene_sets</code> argument could get evaluated.
<code>gene_sets</code> is evaluated in a restricted environment, but there might be a way around this, so you might want to check for anything suspicious in this argument manually.
For more information, read the code.
</p>


<h3>See Also</h3>

<p>gene_eval
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iJO1366)
library(dplyr)

gene_table = tibble(name = iJO1366$geneAssociation %&gt;%
stringr::str_split('and|or|\\s|\\(|\\)') %&gt;%
  purrr::flatten_chr() %&gt;%
  unique,
presence = 1) %&gt;%
  filter(name != '', !is.na(name))

gene_associate(reaction_table = iJO1366 %&gt;%
                 mutate(geneAssociation = geneAssociation %&gt;%
                          stringr::str_replace_all('and', '&amp;') %&gt;%
                          stringr::str_replace_all('or', '|')
                 ),
               gene_table = gene_table
)
</code></pre>

<hr>
<h2 id='gene_eval'>Function to estimate the expression levels of gene sets</h2><span id='topic+gene_eval'></span>

<h3>Description</h3>

<p>Function to estimate the expression levels of gene sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_eval(gene_sets, genes, presences)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_eval_+3A_gene_sets">gene_sets</code></td>
<td>
<p>A list of gene set strings: names of genes punctuated with <code>&amp;</code>, <code>|</code> and brackets.</p>
</td></tr>
<tr><td><code id="gene_eval_+3A_genes">genes</code></td>
<td>
<p>A list of gene names</p>
</td></tr>
<tr><td><code id="gene_eval_+3A_presences">presences</code></td>
<td>
<p>A list of gene presences, the same length as <code>genes</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector the same length as <code>gene_sets</code>, with the the calculated combined gene expression levels.
</p>
<p>This function evaluates the gene sets in the context of the gene presences. 
It can take booleans, or numbers, in which case it associates <code>&amp;</code> with finding the minimum, and <code>|</code> with finding the maximum.
</p>


<h3>Warning</h3>

<p>This function uses <code><a href="base.html#topic+eval">eval</a></code> to evaluate gene expression sets. 
This gives flexibility, but means that malicious code in the <code>gene_sets</code> argument could get evaluated.
<code>gene_sets</code> is evaluated in a restricted environment, but there might be a way around this, so you might want to check for anything suspicious in this argument manually.
For more information, read the code.
</p>


<h3>See Also</h3>

<p>gene_associate
</p>

<hr>
<h2 id='get_BiGG'>Download a model from a BiGG json file</h2><span id='topic+get_BiGG'></span>

<h3>Description</h3>

<p>Download a model from a BiGG json file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_BiGG(address)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_BiGG_+3A_address">address</code></td>
<td>
<p>An address to download from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A model in expanded format
</p>

<hr>
<h2 id='iJO1366'>A full size E. coli model.</h2><span id='topic+iJO1366'></span>

<h3>Description</h3>

<p>A full size E. coli model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iJO1366
</code></pre>


<h3>Format</h3>

<p>A data frame with 2,583 rows and 10 columns:
</p>

<dl>
<dt>abbreviation</dt><dd><p>an abbreviated reaction name, acts as the reaction id</p>
</dd>
<dt>lowbnd</dt><dd><p>lower bound on the reaction rate</p>
</dd>
<dt>uppbnd</dt><dd><p>upper bound on the reaction rate</p>
</dd>
<dt>obj_coef</dt><dd><p>identifies a reaction (or reactions) for which the maximum possible rate should be found</p>
</dd>
<dt>equation</dt><dd><p>reaction equation</p>
</dd>
<dt>officialName</dt><dd><p>full reaction name</p>
</dd>
<dt>geneAssociation</dt><dd><p>A boolean combination of genes which control the reaction</p>
</dd>
<dt>subsystem</dt><dd><p>an indicator of reaction function</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="http://bigg.ucsd.edu">http://bigg.ucsd.edu</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/21988831/">A comprehensive genome-scale reconstruction of Escherichia coli metabolism&ndash;2011.</a>
</p>

<hr>
<h2 id='nutrient_types'>A subset of exchange reactions annotated to indicate typical availability</h2><span id='topic+nutrient_types'></span>

<h3>Description</h3>

<p>A subset of exchange reactions annotated to indicate typical availability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nutrient_types
</code></pre>


<h3>Format</h3>

<p>A data frame with 25 rows and 2 columns:
</p>

<dl>
<dt>abbreviation</dt><dd><p>an exchange reaction id</p>
</dd>
<dt>nutrient_type</dt><dd><p>the nutrient availability, one of 'micro', 'macro' or 'substrate'</p>
</dd>
</dl>


<hr>
<h2 id='parse_met_list'>Internal function: Expand half reaction equations into a long form</h2><span id='topic+parse_met_list'></span>

<h3>Description</h3>

<p>Internal function: Expand half reaction equations into a long form
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_met_list(mets)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parse_met_list_+3A_mets">mets</code></td>
<td>
<p>Character vector of halves of reaction equations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>date_frame</code> with columns: </p>

<dl>
<dt>stoich</dt><dd><p>the stoichiometric coefficient</p>
</dd>
<dt>met</dt><dd><p>the metabolite</p>
</dd>
</dl>


<hr>
<h2 id='reactiontbl_to_expanded'>Parse a reaction table to an intermediate, long format</h2><span id='topic+reactiontbl_to_expanded'></span>

<h3>Description</h3>

<p>The long format can also be suitable for manipulating equations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactiontbl_to_expanded(reaction_table, regex_arrow = "&lt;?[-=]+&gt;")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reactiontbl_to_expanded_+3A_reaction_table">reaction_table</code></td>
<td>
<p>A data frame describing the metabolic model.</p>
</td></tr>
<tr><td><code id="reactiontbl_to_expanded_+3A_regex_arrow">regex_arrow</code></td>
<td>
<p>Regular expression for the arrow splitting sides of the reaction equation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>reaction_table</code> must have columns:
</p>

<ul>
<li> <p><code>abbreviation</code>,
</p>
</li>
<li> <p><code>equation</code>,
</p>
</li>
<li> <p><code>uppbnd</code>,
</p>
</li>
<li> <p><code>lowbnd</code>, and
</p>
</li>
<li> <p><code>obj_coef</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list of data frames: </p>

<ul>
<li> <p><code>rxns</code>, which has one row per reaction, 
</p>
</li>
<li> <p><code>mets</code>, which has one row for each metabolite, and 
</p>
</li>
<li> <p><code>stoich</code>, which has one row for each time a metabolite appears in a reaction.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other parsing_and_conversion: 
<code><a href="#topic+expanded_to_ROI">expanded_to_ROI</a>()</code>,
<code><a href="#topic+expanded_to_glpk">expanded_to_glpk</a>()</code>,
<code><a href="#topic+expanded_to_gurobi">expanded_to_gurobi</a>()</code>,
<code><a href="#topic+reactiontbl_to_gurobi">reactiontbl_to_gurobi</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data(ecoli_core)
library(dplyr)
try(library(ROI.plugin.ecos)) # make a solver available to ROI

roi_model &lt;- ecoli_core %&gt;%
  reactiontbl_to_expanded %&gt;%
  expanded_to_ROI
  
if(length(ROI::ROI_applicable_solvers(roi_model))&gt;=1){
  roi_result &lt;- ROI::ROI_solve(roi_model)
  
  ecoli_core_with_flux &lt;- ecoli_core %&gt;%
    mutate(flux = roi_result[['solution']])
}

## End(Not run)
</code></pre>

<hr>
<h2 id='reactiontbl_to_gurobi'>Parse reaction table to Gurobi format</h2><span id='topic+reactiontbl_to_gurobi'></span>

<h3>Description</h3>

<p><strong>This function is deprecated. <code>github.com/Fl0Sch/ROI.plugin.gurobi</code> is recommended instead.</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactiontbl_to_gurobi(reaction_table, regex_arrow = "&lt;?[-=]+&gt;")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reactiontbl_to_gurobi_+3A_reaction_table">reaction_table</code></td>
<td>
<p>A data frame describing the metabolic model.</p>
</td></tr>
<tr><td><code id="reactiontbl_to_gurobi_+3A_regex_arrow">regex_arrow</code></td>
<td>
<p>Regular expression for the arrow splitting sides of the reaction equation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parses a reaction table to give a list in Gurobi's input format.
This function is a shorthand for <code><a href="#topic+reactiontbl_to_expanded">reactiontbl_to_expanded</a></code> followed by <code><a href="#topic+expanded_to_gurobi">expanded_to_gurobi</a></code>.
</p>
<p>The <code>reaction_table</code> must have columns:
</p>

<ul>
<li> <p><code>abbreviation</code>,
</p>
</li>
<li> <p><code>equation</code>,
</p>
</li>
<li> <p><code>uppbnd</code>,
</p>
</li>
<li> <p><code>lowbnd</code>, and
</p>
</li>
<li> <p><code>obj_coef</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list suitable for input to Gurobi.
</p>


<h3>See Also</h3>

<p>Other parsing_and_conversion: 
<code><a href="#topic+expanded_to_ROI">expanded_to_ROI</a>()</code>,
<code><a href="#topic+expanded_to_glpk">expanded_to_glpk</a>()</code>,
<code><a href="#topic+expanded_to_gurobi">expanded_to_gurobi</a>()</code>,
<code><a href="#topic+reactiontbl_to_expanded">reactiontbl_to_expanded</a>()</code>
</p>

<hr>
<h2 id='recompose_metabolites'>Merge metabolite stub and compartment to form an id</h2><span id='topic+recompose_metabolites'></span>

<h3>Description</h3>

<p>Merge metabolite stub and compartment to form an id
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recompose_metabolites(
  expanded_metabolites,
  before_signifier = "_",
  after_signifier = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recompose_metabolites_+3A_expanded_metabolites">expanded_metabolites</code></td>
<td>
<p>a metabolite table as created by <code><a href="#topic+decompose_metabolites">decompose_metabolites</a></code></p>
</td></tr>
<tr><td><code id="recompose_metabolites_+3A_before_signifier">before_signifier</code></td>
<td>
<p>a string that is inserted before the compartment identifier</p>
</td></tr>
<tr><td><code id="recompose_metabolites_+3A_after_signifier">after_signifier</code></td>
<td>
<p>a string that is inserted after the compartment identifier</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A merged metabolite table with one column, <code>met</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ecoli_core)

mod &lt;- reactiontbl_to_expanded(ecoli_core)

decompose_metabolites(mod$mets)

recompose_metabolites(decompose_metabolites(mod$mets))
</code></pre>

<hr>
<h2 id='split_on_arrow'>Internal function: Splitting reaction equation into substrate and product</h2><span id='topic+split_on_arrow'></span>

<h3>Description</h3>

<p>Internal function: Splitting reaction equation into substrate and product
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_on_arrow(equations, regex_arrow = "&lt;?[-=]+&gt;")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_on_arrow_+3A_equations">equations</code></td>
<td>
<p>Character vector of reaction equations.</p>
</td></tr>
<tr><td><code id="split_on_arrow_+3A_regex_arrow">regex_arrow</code></td>
<td>
<p>Regular expression for the arrow splitting sides of the reaction equation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data_frame</code>, with columns: </p>

<dl>
<dt>reversible</dt><dd><p>boolean, is reaction reversible</p>
</dd>
<dt>before</dt><dd><p>the left hand side of the reaction string</p>
</dd>
<dt>after</dt><dd><p>the right hand side of the reaction string</p>
</dd>
</dl>


<hr>
<h2 id='validate_expanded'>Validate an expanded model</h2><span id='topic+validate_expanded'></span>

<h3>Description</h3>

<p>Validate an expanded model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_expanded(reactions_expanded)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validate_expanded_+3A_reactions_expanded">reactions_expanded</code></td>
<td>
<p>the expanded model to check</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
