<!DOCTYPE html><html lang="en"><head><title>Help for package FBN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {FBN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FBN-package'>
<p>FISH Based Normalization and Copy Number Inference of SNP Microarray Data</p></a></li>
<li><a href='#FBN.histogramMaxima'><p> Local maxima of the histogram of SNP microarray data</p></a></li>
<li><a href='#FBN.kmeans'><p> K-Means clustering of SNP microarray data</p></a></li>
<li><a href='#FBN.valueCenter'><p> The SNP normalization function</p></a></li>
<li><a href='#FBNormalization'><p> FISH Based Normalization and Copy Number Inference of</p>
SNP Microarray Data</a></li>
<li><a href='#FISHprobes'><p>Sample data for FBN package</p></a></li>
<li><a href='#hmcls'><p>Sample data for FBN package</p></a></li>
<li><a href='#meanFilter'><p> 1D Mean Filter</p></a></li>
<li><a href='#medianFilter'><p> 1D Median Filter</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>FISH Based Normalization and Copy Number Inference of SNP
Microarray Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Adrian Andronache &lt;adi.andronache@gmail.com&gt;, Luca Agnelli &lt;luca.agnelli@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luca Agnelli  &lt;luca.agnelli@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Normalizes the data from a file containing the raw values of the SNP probes of microarray data by using the FISH probes and their corresponding copy number.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-16 16:37:05 UTC; agnelli</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-17 23:50:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='FBN-package'>
FISH Based Normalization and Copy Number Inference of SNP Microarray Data
</h2><span id='topic+FBN-package'></span><span id='topic+FBN'></span>

<h3>Description</h3>

<p>Normalizes the data from a file containing the raw values of the SNP probes of microarray data
by using the FISH probes and their corresponding CNs.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> FBN</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.5.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2012-03-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>To start using the FBN package, call <code>library(FBN)</code>.
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>References</h3>

<p>Agnelli L et al. (2009), &quot;A SNP Microarray and FISH-Based Procedure to Detect Allelic Imbalances in Multiple Myeloma: an Integrated Genomics Approach Reveals a Wide Gene Dosage Effect&quot;, <em>Genes Chrom Cancer</em>
</p>

<hr>
<h2 id='FBN.histogramMaxima'> Local maxima of the histogram of SNP microarray data </h2><span id='topic+FBN.histogramMaxima'></span>

<h3>Description</h3>

<p>Finds at most 6 of the local maxima of the histogram of the <code>inputData</code> representing the SNP microarray data. 
Firstly, the function finds all local maxima onto the histogram, 
and finally removes those that are closer than <code>minSpan</code>. 
The histogram is estimated with equi-spaced breaks (also the default), 
defined by <code>breaksData</code> (see the documentation of <code><a href="graphics.html#topic+hist">hist</a></code> for more details). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FBN.histogramMaxima(inputData, minSpan, breaksData)
FBN.histogramMaxima(inputData, minSpan = 0.2, breaksData = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FBN.histogramMaxima_+3A_inputdata">inputData</code></td>
<td>
<p> A vector of SNP microarray values for which the local maxima of the histogram are desired </p>
</td></tr>
<tr><td><code id="FBN.histogramMaxima_+3A_minspan">minSpan</code></td>
<td>
<p> The minimum distance separating consecutive local maxima. If <code>minSpan</code> is negative, then a smoothing filter is applied on the histogram to reduce its noise, and therefore, estimates the most significative maxima. </p>
</td></tr>
<tr><td><code id="FBN.histogramMaxima_+3A_breaksdata">breaksData</code></td>
<td>
<p> One of: </p>
 
<ul>
<li><p>a vector giving the breakpoints between histogram cells,
</p>
</li>
<li><p>a single number giving the number of cells for the histogram,
</p>
</li>
<li><p>a character string naming an algorithm to compute the number of cells (see the Details section of <code><a href="graphics.html#topic+hist">hist</a></code>),
</p>
</li>
<li><p>a function to compute the number of cells.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function has been designed based on SNP microarray data and FISH resolution. As in FISH analyses high numbers of signals do not allow a correct identification of discrete CNs, an empirical number of no more than 6 local maxima is therefore imposed. These maxima are used to initialize the k-means algorithm to determine the CN clusters. 
</p>


<h3>Value</h3>

<p>Returns a vector containing at most 6 values of the <code>inputData</code> in which the histogram shows local maxima.
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+hist">hist</a></code>, <code><a href="#topic+FBNormalization">FBNormalization</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>require(stats)
require(graphics)

x = c(rnorm(1000, 1, .2), rnorm(1000, 2, .2))
y = FBN.histogramMaxima(x, minSpan = .1)
h = hist(x)
par(new = TRUE)
plot(y,vector(mode=mode(y), length = length(y)), xlim = c(min(h$breaks), 
 	max(h$breaks)), ylim = c(0,max(h$counts)), xlab = NA, ylab = NA, 
 	col = 'red' )


x = c(1, 2, 2, 3, 4, 5, 5, 5, 6, 7, 8, 9, 10, 10, 10, 11)
y = FBN.histogramMaxima(x, minSpan = 3, breaksData = 100)
h = hist(x, 100)
par(new = TRUE)
plot(y,vector(mode=mode(y), length = length(y)), xlim = c(min(h$breaks), 
 	max(h$breaks)), ylim = c(0,max(h$counts)), xlab = NA, ylab = NA, 
 	col = 'red' )
</code></pre>

<hr>
<h2 id='FBN.kmeans'> K-Means clustering of SNP microarray data </h2><span id='topic+FBN.kmeans'></span>

<h3>Description</h3>

<p>Performs a k-means clustering of SNP microarray data. Returns clusters of values as being putatively characterized by different CN. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FBN.kmeans(inputData, minSpan, breaksData)
FBN.kmeans(inputData = NULL, minSpan = 0.2, breaksData = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FBN.kmeans_+3A_inputdata">inputData</code></td>
<td>
<p> A vector of values containig the SNP microarray data </p>
</td></tr>
<tr><td><code id="FBN.kmeans_+3A_minspan">minSpan</code></td>
<td>
<p> The minimum distance separating consecutive local maxima that are to be detected on 
the histogram of the <code>inputData</code>. These maxima are used to initialize the k-means clustering process.
For more details concerning the local maxima detection, check the documentation of <code><a href="#topic+FBN.histogramMaxima">FBN.histogramMaxima</a></code></p>
</td></tr>
<tr><td><code id="FBN.kmeans_+3A_breaksdata">breaksData</code></td>
<td>
<p> One of: </p>

<ul>
<li><p>a vector giving the breakpoints between histogram cells,
</p>
</li>
<li><p>a single number giving the number of cells for the histogram,
</p>
</li>
<li><p>a character string naming an algorithm to compute the number of cells (see Details section of <code><a href="graphics.html#topic+hist">hist</a></code>),
</p>
</li>
<li><p>a function to compute the number of cells.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This fuction takes as input the vector of raw SNP microarray values, and perform a k-means clustering 
trying to identify the groups of raw values characterized by different CNs. The clustering process is initialized 
with the local maxima detected on the histogram of the input data (see the documentation of <code><a href="#topic+FBN.histogramMaxima">FBN.histogramMaxima</a></code>).
To increase the robustness of the clustering process and to remove possible small or noisy clusters,
a double filtering is done: firstly, removing those clusters populated by less than 1% values from the entire <code>inputData</code>, and then, due to putative noisy histograms, merging those clusters whose centers are closer than 0.2 in nominal values.   
</p>


<h3>Value</h3>

<p>An object of class <code><a href="stats.html#topic+kmeans">kmeans</a></code>
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
require(stats)
require(graphics)
x = c(rnorm(1000, 1, .2), rnorm(1000, 2, .2))
y = FBN.kmeans(x, minSpan = .001)
h = hist(x)
par(new = TRUE)
plot(y$centers,vector(mode=mode(y$centers), length = length(y$centers)), 
	xlim = c(min(h$breaks), max(h$breaks)), ylim = c(0,max(h$counts)), 
	xlab = NA, ylab = NA, col = 'red' )
	
</code></pre>

<hr>
<h2 id='FBN.valueCenter'> The SNP normalization function </h2><span id='topic+FBN.valueCenter'></span>

<h3>Description</h3>

<p>Normalization of the raw SNP microarray values, by multiplication (on linear scale) or addition (in log scale) 
of all the raw SNP values with the normalization factor. 
The normalization factor is estimated such that it brings the <code>normalizingValue</code> of the raw 
SNP values onto the <code>nominalValueCN</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FBN.valueCenter(inputData, normalizingValue, nominalValueCN, 
	logScale)
FBN.valueCenter(inputData = NULL, normalizingValue = NULL, 
	nominalValueCN = 2, logScale = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FBN.valueCenter_+3A_inputdata">inputData</code></td>
<td>
<p> The vector of raw SNP values, as they come out from, e.g. Circular Binary Segmentation in <code>DNAcopy</code> package from Bioconductor  </p>
</td></tr>
<tr><td><code id="FBN.valueCenter_+3A_normalizingvalue">normalizingValue</code></td>
<td>
<p> The value representing the center of the cluster identified as having a certain CN </p>
</td></tr>
<tr><td><code id="FBN.valueCenter_+3A_nominalvaluecn">nominalValueCN</code></td>
<td>
<p> The nominal value representing a certain CN on which the <code>normalizingValue</code> has to be brought. </p>
</td></tr>
<tr><td><code id="FBN.valueCenter_+3A_logscale">logScale</code></td>
<td>
<p> A logical value, specifying wether the data is on linear (<code>FALSE</code>) or logarithmic scale (<code>TRUE</code>).  </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>nominalValueCN</code> is a real value representing the CN, e.g. CN<code class="reqn">=2</code> has a <code>nominalValueCN</code> of 2, 
but all other CN<code class="reqn">=n</code> (<code class="reqn">n</code> != 2) will have a <code>nominalValueCN</code> different from <code class="reqn">n</code>. 
Such <code>nominalValueCN</code> is identified by the <code><a href="#topic+FBN.kmeans">FBN.kmeans</a></code> function. 
</p>


<h3>Value</h3>

<p>Returns a vector containing the normalized values of the <code>inputData</code>
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+FBN.kmeans">FBN.kmeans</a></code>, <code><a href="#topic+FBNormalization">FBNormalization</a></code>  </p>


<h3>Examples</h3>

<pre><code class='language-R'>require(stats)
require(graphics)
x = c(rnorm(1000, 1, .1), rnorm(1000, 1.5, .1))
y = FBN.valueCenter(x, normalizingValue = 1, nominalValueCN = 2, 
	logScale = FALSE)
par(mfrow = c(2, 1), new = FALSE)
h = hist(x)
par(new = TRUE)
plot(1, 0, col = 'red', xlim = c(min(h$breaks), max(h$breaks)), 
	ylim = c(0,max(h$counts)), xlab = NA, ylab = NA)
par(new = FALSE)
h = hist(y)
par(new = TRUE)
plot(2, 0, col = 'red', xlim = c(min(h$breaks), max(h$breaks)), 
	ylim = c(0,max(h$counts)), xlab = NA, ylab = NA)
</code></pre>

<hr>
<h2 id='FBNormalization'> FISH Based Normalization and Copy Number Inference of 
SNP Microarray Data </h2><span id='topic+FBNormalization'></span>

<h3>Description</h3>

<p>Normalizes the data from a file containing the raw values of the SNP probes of microarrray data
by using the FISH probes and their corresponding CNs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FBNormalization(rawDataFileName, fishProbesFileName, 
	normDataFileName, debugFlag, plotFlag, plotAndSaveFlag)
FBNormalization(rawDataFileName = NULL, 
								fishProbesFileName = NULL, 
								normDataFileName = NULL,	
								debugFlag = FALSE, 
								plotFlag = FALSE, 
								plotAndSaveFlag = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FBNormalization_+3A_rawdatafilename">rawDataFileName</code></td>
<td>
<p> The file containig the raw values of the SNP probes. It should be .txt format, 
tab delimited, containing on each column the SNP probes data of the different samples. 
The structure of the input file should be as follows: </p>

<ul>
<li><p> a column containing the identification name of the SNP probes, entitled 'SNP',
</p>
</li>
<li><p> a column containing the chromosome number of the SNP probes, entitled 'chrom',
</p>
</li>
<li><p> a column containing the physical positon of the SNP probes, entitled 'physical.position',
</p>
</li>
<li><p> a column containing the cytoband of the SNP probe, entitled 'cytoband',
</p>
</li>
<li><p> 'n' columns containing the values of the SNP probes, entitled with the custom identification name of the samples.
</p>
</li></ul>
</td></tr>
<tr><td><code id="FBNormalization_+3A_fishprobesfilename">fishProbesFileName</code></td>
<td>
<p> The file containing the FISH probes information. It should be .txt format, 
tab delimited, containing a separate row for each available FISH probe and on each column the various 
CN data revealed by FISH of the different samples. The structure of the input file should be as follows: </p>

<ul>
<li><p> each row should represent one FISH probe
</p>
</li>
<li><p> one column containing the name of the BAC clone, entitled 'BACclone';
</p>
</li>
<li><p> one column containing the chromosome number where the FISH probe is located, entitled 'chromosome';
</p>
</li>
<li><p> one column containing the cytoband location of the Fish probe, entitled 'cytoband';
</p>
</li>
<li><p> one column containing the start position of the physical localization of the FISH probe, entitled 'start.loc';
</p>
</li>
<li><p> one column containing the end position of the physical localization of the FISH probe, entitled 'end.loc';
</p>
</li>
<li><p> 'n' columns containing the values of the CN coresponding to the FISH probe, entitled with the custom identification name of the samples.
</p>
</li></ul>
</td></tr>
<tr><td><code id="FBNormalization_+3A_normdatafilename">normDataFileName</code></td>
<td>
<p> The name of the output file where the normalized data will be written. Also, the function writes
a file starting with the same name as <code>normDataFileName</code> and ending with <em>'_thresholds.txt'</em>, containing the
thresholds values that can be used to infer CNs to the normalized data. </p>
</td></tr>
<tr><td><code id="FBNormalization_+3A_debugflag">debugFlag</code></td>
<td>
<p> Logical value, specifying wether the function should run in debug mode. If <code>TRUE</code>, it plots the various histograms and the detected centers of the clusters in the different samples. </p>
</td></tr>
<tr><td><code id="FBNormalization_+3A_plotflag">plotFlag</code></td>
<td>
<p> Logical value, specifying wether the function should plot the histograms of the individual data. If <code>TRUE</code>, it plots the various histograms and the detected centers of the clusters in the different samples. </p>
</td></tr>
<tr><td><code id="FBNormalization_+3A_plotandsaveflag">plotAndSaveFlag</code></td>
<td>
<p> Logical value, specifying wether the function should plot and save the various plots of the histograms. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>For further detalis, see Supporting Information in <cite>Agnelli L et al. (2009), &quot;A SNP Microarray and FISH-Based Procedure to Detect Allelic Imbalances in Multiple Myeloma: an Integrated Genomics Approach Reveals a Wide Gene Dosage Effect&quot;, <em>Genes Chrom Cancer</em></cite>
</p>


<h3>Value</h3>

<p>No value is returned. The function writes on the disk the output files as previously described. 
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## set path to FBN package data directory
rawDataFileName = './../data/hmcls.txt'
fishProbesFileName = './../data/FISHprobes.txt'
normDataFileName= 'hmcls_NORM.txt'
FBNormalization(rawDataFileName, fishProbesFileName, normDataFileName, 
	debugFlag = FALSE)
</code></pre>

<hr>
<h2 id='FISHprobes'>Sample data for FBN package</h2><span id='topic+FISHprobes'></span>

<h3>Description</h3>

<p>Four FISH probes information for the two sample Human Myeloma Cell Lines (AMO1 and NCI-H929).</p>


<h3>Format</h3>

<p>tab delimited .txt file</p>


<h3>References</h3>

<p>Agnelli L et al. (2009), &quot;A SNP Microarray and FISH-Based Procedure to Detect Allelic Imbalances in Multiple Myeloma: an Integrated Genomics Approach Reveals a Wide Gene Dosage Effect&quot;, <em>Genes Chrom Cancer</em>
</p>

<hr>
<h2 id='hmcls'>Sample data for FBN package</h2><span id='topic+hmcls'></span>

<h3>Description</h3>

<p>Two Human Myeloma Cell Lines (AMO1 and NCI-H929) profiled on GeneChip Human Mapping 250K NspI arrays.</p>


<h3>Format</h3>

<p>image data .rda</p>


<h3>References</h3>

<p>Agnelli L et al. (2009), &quot;A SNP Microarray and FISH-Based Procedure to Detect Allelic Imbalances in Multiple Myeloma: an Integrated Genomics Approach Reveals a Wide Gene Dosage Effect&quot;, <em>Genes Chrom Cancer</em>
</p>

<hr>
<h2 id='meanFilter'> 1D Mean Filter </h2><span id='topic+meanFilter'></span>

<h3>Description</h3>

<p>1-dimensional mean filter with a specified <code>windowSize</code> of the <code>inputData</code></p>


<h3>Usage</h3>

<pre><code class='language-R'>meanFilter(inputData, windowSize)
meanFilter(inputData = NULL, windowSize = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meanFilter_+3A_inputdata">inputData</code></td>
<td>
<p> The vector of values to be filtered </p>
</td></tr>
<tr><td><code id="meanFilter_+3A_windowsize">windowSize</code></td>
<td>
<p> The half-size of the filtering window (default <code>windowSize</code> = 3) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Classical implementation of a mean filter, using a sliding window. By default, the half-size of the sliding window is set to 3 unless otherwise specified. 
</p>


<h3>Value</h3>

<p>The output data has the same size of the input data. If the window half-size is smaller or equal to 1, then the input data is passed directly to the output data.
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- meanFilter(c(0, 0, 0, 1, 1, 1, 0, 0, 1, 0))
x &lt;- meanFilter(c(0, 0, 0, 1, 0, 0, 0, 0, 1, 0), windowSize = 5)
</code></pre>

<hr>
<h2 id='medianFilter'> 1D Median Filter </h2><span id='topic+medianFilter'></span>

<h3>Description</h3>

<p>1-dimensinal median filter with a specified <code>windowSize</code> of the <code>inputData</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>medianFilter(inputData, windowSize)
medianFilter(inputData = NULL, windowSize = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="medianFilter_+3A_inputdata">inputData</code></td>
<td>
<p> The vector of values to be filtered </p>
</td></tr>
<tr><td><code id="medianFilter_+3A_windowsize">windowSize</code></td>
<td>
<p> The half-size of the filtering window (default <code>windowSize</code> = 3) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Classical implementation of a median filter, using a sliding window. By default, the half-size of the sliding window is set to 3 unless otherwise specified. 
</p>


<h3>Value</h3>

<p>The output data has the same size of the input data. If the window half-size is smaller or equal to 1, then the input data is passed directly to the output data.
</p>


<h3>Author(s)</h3>

<p> Adrian Andronache <a href="mailto:adi.andronache@gmail.com">adi.andronache@gmail.com</a> <br />
Luca Agnelli <a href="mailto:luca.agnelli@gmail.com">luca.agnelli@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- medianFilter(c(0, 0, 0, 1, 1, 1, 0, 0, 1, 0))
x &lt;- medianFilter(c(0, 0, 0, 1, 0, 0, 0, 0, 1, 0), windowSize = 5)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
