<!DOCTYPE html><html><head><title>Help for package fcfdr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fcfdr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#binary_cfdr'><p>Perform cFDR leveraging binary auxiliary covariates</p></a></li>
<li><a href='#corr_plot'><p>Violin plot of p-values for quantiles of q</p></a></li>
<li><a href='#flexible_cfdr'><p>Perform Flexible cFDR</p></a></li>
<li><a href='#log10pv_plot'><p>Plot -log10(p) against -log10(v) and colour by q</p></a></li>
<li><a href='#match_ind_maf'><p>Function to downsample independent SNPs to match MAF distribution of whole set.</p></a></li>
<li><a href='#parameters_in_locfdr'><p>parameters_in_locfdr</p></a></li>
<li><a href='#pv_plot'><p>Plot p against v and colour by q</p></a></li>
<li><a href='#stratified_qqplot'><p>Stratified Q-Q plot.</p></a></li>
<li><a href='#T1D_application_data'><p>Data for T1D application</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Flexible cFDR</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to implement the Flexible cFDR (Hutchinson et al. (2021) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1009853">doi:10.1371/journal.pgen.1009853</a>&gt;) and Binary cFDR (Hutchinson et al. (2021) &lt;<a href="https://doi.org/10.1101%2F2021.10.21.465274">doi:10.1101/2021.10.21.465274</a>&gt;) methodologies to leverage auxiliary data from arbitrary distributions, for example functional genomic data, with GWAS p-values to generate re-weighted p-values. </td>
</tr>
<tr>
<td>Imports:</td>
<td>locfdr, MASS, ggplot2, cowplot, fields, dplyr, spatstat.geom,
polyCub, hexbin, bigsplines, data.table, grDevices, Hmisc</td>
</tr>
<tr>
<td>Suggests:</td>
<td>stats, knitr, rmarkdown, digest, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-03 14:23:05 UTC; anna</td>
</tr>
<tr>
<td>Author:</td>
<td>Anna Hutchinson <a href="https://orcid.org/0000-0002-9224-4410"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Chris Wallace [aut],
  Thomas Willis [ctb, aut],
  James Liley [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anna Hutchinson &lt;annahutchinson1995@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-07 09:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='binary_cfdr'>Perform cFDR leveraging binary auxiliary covariates</h2><span id='topic+binary_cfdr'></span>

<h3>Description</h3>

<p>Perform cFDR leveraging binary auxiliary covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_cfdr(p, q, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binary_cfdr_+3A_p">p</code></td>
<td>
<p>p-values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="binary_cfdr_+3A_q">q</code></td>
<td>
<p>binary auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="binary_cfdr_+3A_group">group</code></td>
<td>
<p>group membership of each SNP for leave-one-out procedure (vector of length n) (e.g. chromosome number or LD block)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of p, q and v values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the parameters_in_locfdr() function to extract the parameters estimated by
# the locfdr function.

# generate p
set.seed(2)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
q &lt;- rbinom(n, 1, 0.1)

group &lt;- c(rep("A", n/2), rep("B", n/2)) 

binary_cfdr(p, q, group)

</code></pre>

<hr>
<h2 id='corr_plot'>Violin plot of p-values for quantiles of q</h2><span id='topic+corr_plot'></span>

<h3>Description</h3>

<p>Violin plot of p-values for quantiles of q
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr_plot(p, q, ylim = c(0, 1.5))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr_plot_+3A_p">p</code></td>
<td>
<p>p values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_q">q</code></td>
<td>
<p>auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="corr_plot_+3A_ylim">ylim</code></td>
<td>
<p>y-axis limits (-log10)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Can be used to investigate the relationship between p and q
</p>
<p>If this shows a non-monotonic relationship then the cFDR framework should not be used
</p>
<p>(because e.g. cFDR cannot simultaneously shrink v-values for high p and low p)
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the corr_plot() function to visualise the relationship between p and q.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

corr_plot(p, q)

</code></pre>

<hr>
<h2 id='flexible_cfdr'>Perform Flexible cFDR</h2><span id='topic+flexible_cfdr'></span>

<h3>Description</h3>

<p>Performs Flexible cFDR for continuous auxiliary covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flexible_cfdr(
  p,
  q,
  indep_index,
  res_p = 300,
  res_q = 500,
  nxbin = 1000,
  gridp = 50,
  splinecorr = TRUE,
  dist_thr = 0.5,
  locfdr_df = 10,
  plot = TRUE,
  maf = NULL,
  check_indep_cor = TRUE,
  enforce_p_q_cor = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flexible_cfdr_+3A_p">p</code></td>
<td>
<p>p-values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_q">q</code></td>
<td>
<p>continuous auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_indep_index">indep_index</code></td>
<td>
<p>indices of independent SNPs</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_res_p">res_p</code></td>
<td>
<p>number of grid points in x-direction (<code>p</code>) for KDE estimation</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_res_q">res_q</code></td>
<td>
<p>number of grid points in y-direction (<code>q</code>) for KDE estimation</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_nxbin">nxbin</code></td>
<td>
<p>number of bins in x-direction (<code>p</code>) for hex-binning</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_gridp">gridp</code></td>
<td>
<p>number of data points required in a KDE grid point for left-censoring</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_splinecorr">splinecorr</code></td>
<td>
<p>logical value for whether spline correction should be implemented</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_dist_thr">dist_thr</code></td>
<td>
<p>distance threshold for spline correction</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_locfdr_df">locfdr_df</code></td>
<td>
<p><code>df</code> parameter in locfdr function</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_plot">plot</code></td>
<td>
<p>logical value for whether to produce plots to assess KDE fit</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_maf">maf</code></td>
<td>
<p>minor allele frequencies for SNPs to which <code>p</code> and <code>q</code> relate (optional and used to perform MAF matching)</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_check_indep_cor">check_indep_cor</code></td>
<td>
<p>check that sign of the correlation between <code>p</code> and <code>q</code> is the same in the independent subset as in the whole</p>
</td></tr>
<tr><td><code id="flexible_cfdr_+3A_enforce_p_q_cor">enforce_p_q_cor</code></td>
<td>
<p>if <code>p</code> and <code>q</code> are negatively correlated, flip the sign on <code>q</code> values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>maf</code> is specified, then the independent SNPs will be down-sampled to match the minor allele frequency distribution.
</p>


<h3>Value</h3>

<p>List of length two: (1) data.frame of p-values, q-values and v-values (2) data.frame of auxiliary data (q_low used for left censoring, how many data-points were left censored and/or spline corrected)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# this is a long running example
 
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the flexible_cfdr() function to generate v-values using default parameter values.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

n_indep &lt;- n

flexible_cfdr(p, q, indep_index = 1:n_indep)


</code></pre>

<hr>
<h2 id='log10pv_plot'>Plot -log10(p) against -log10(v) and colour by q</h2><span id='topic+log10pv_plot'></span>

<h3>Description</h3>

<p>Plot -log10(p) against -log10(v) and colour by q
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log10pv_plot(p, q, v, axis_lim = c(0, 20))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log10pv_plot_+3A_p">p</code></td>
<td>
<p>p values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="log10pv_plot_+3A_q">q</code></td>
<td>
<p>auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="log10pv_plot_+3A_v">v</code></td>
<td>
<p>v values from cFDR</p>
</td></tr>
<tr><td><code id="log10pv_plot_+3A_axis_lim">axis_lim</code></td>
<td>
<p>Optional axis limits</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Can be used to visualise the results from Flexible cFDR
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# this is a long running example

# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the flexible_cfdr() function to generate v-values and then the log10pv_plot() function
# to visualise the results.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

n_indep &lt;- n

res &lt;- flexible_cfdr(p, q, indep_index = 1:n_indep)

log10pv_plot(p = res[[1]]$p, q = res[[1]]$q, v = res[[1]]$v)


</code></pre>

<hr>
<h2 id='match_ind_maf'>Function to downsample independent SNPs to match MAF distribution of whole set.</h2><span id='topic+match_ind_maf'></span>

<h3>Description</h3>

<p>Matches MAF distribution of independent set of SNPs to MAF distribution of whole set of SNPs to avoid MAF-based confounding.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_ind_maf(maf, indep_index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_ind_maf_+3A_maf">maf</code></td>
<td>
<p>minor allele frequencies of (all) SNPs</p>
</td></tr>
<tr><td><code id="match_ind_maf_+3A_indep_index">indep_index</code></td>
<td>
<p>indices of independent SNPs</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Must supply maf values from the whole data set, not just the independent SNPs.
</p>


<h3>Value</h3>

<p>indices of independent SNP in chosen in sample
</p>

<hr>
<h2 id='parameters_in_locfdr'>parameters_in_locfdr</h2><span id='topic+parameters_in_locfdr'></span>

<h3>Description</h3>

<p>parameters_in_locfdr
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parameters_in_locfdr(
  p,
  q,
  indep_index,
  res_p = 300,
  res_q = 500,
  maf = NULL,
  check_indep_cor = TRUE,
  enforce_p_q_cor = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parameters_in_locfdr_+3A_p">p</code></td>
<td>
<p>p values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_q">q</code></td>
<td>
<p>continuous auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_indep_index">indep_index</code></td>
<td>
<p>indices of independent SNPs</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_res_p">res_p</code></td>
<td>
<p>resolution for p</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_res_q">res_q</code></td>
<td>
<p>resolution for q</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_maf">maf</code></td>
<td>
<p>minor allele frequencies for SNPs to which <code>p</code> and <code>q</code> relate (optional and used to perform MAF matching)</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_check_indep_cor">check_indep_cor</code></td>
<td>
<p>check that sign of the correlation between <code>p</code> and <code>q</code> is the same in the independent subset as in the whole</p>
</td></tr>
<tr><td><code id="parameters_in_locfdr_+3A_enforce_p_q_cor">enforce_p_q_cor</code></td>
<td>
<p>if <code>p</code> and <code>q</code> are negatively correlated, flip the sign on <code>q</code> values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of values used as input into <code>locfdr::locfdr</code> function intrinsically in <code>flexible_cfdr</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the parameters_in_locfdr() function to extract the parameters estimated by
# the locfdr function.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

n_indep &lt;- n

parameters_in_locfdr(p, q, indep_index = 1:n_indep)


</code></pre>

<hr>
<h2 id='pv_plot'>Plot p against v and colour by q</h2><span id='topic+pv_plot'></span>

<h3>Description</h3>

<p>Plot p against v and colour by q
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pv_plot(p, q, v, axis_lim = c(0, 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pv_plot_+3A_p">p</code></td>
<td>
<p>p values for principal trait (vector of length n)</p>
</td></tr>
<tr><td><code id="pv_plot_+3A_q">q</code></td>
<td>
<p>auxiliary data values (vector of length n)</p>
</td></tr>
<tr><td><code id="pv_plot_+3A_v">v</code></td>
<td>
<p>v values from cFDR</p>
</td></tr>
<tr><td><code id="pv_plot_+3A_axis_lim">axis_lim</code></td>
<td>
<p>Optional axis limits</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Can be used to visualise the results from Flexible cFDR
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # this is a long running example
 
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the flexible_cfdr() function to generate v-values and then the pv_plot() function
# to visualise the results.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

n_indep &lt;- n

res &lt;- flexible_cfdr(p, q, indep_index = 1:n_indep)

pv_plot(p = res[[1]]$p, q = res[[1]]$q, v = res[[1]]$v)


</code></pre>

<hr>
<h2 id='stratified_qqplot'>Stratified Q-Q plot.</h2><span id='topic+stratified_qqplot'></span>

<h3>Description</h3>

<p>Stratified Q-Q plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stratified_qqplot(
  data_frame,
  prin_value_label,
  cond_value_label = NULL,
  thresholds = c(1, 0.1, 0.01, 0.001, 1e-04)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stratified_qqplot_+3A_data_frame">data_frame</code></td>
<td>
<p><code>data.frame</code> containing p-values and auxiliary data values</p>
</td></tr>
<tr><td><code id="stratified_qqplot_+3A_prin_value_label">prin_value_label</code></td>
<td>
<p>label of principal p-value column in <code>data_frame</code></p>
</td></tr>
<tr><td><code id="stratified_qqplot_+3A_cond_value_label">cond_value_label</code></td>
<td>
<p>label of conditional trait column in <code>data_frame</code></p>
</td></tr>
<tr><td><code id="stratified_qqplot_+3A_thresholds">thresholds</code></td>
<td>
<p>threshold values to define strata</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Can be used to investigate the relationship between p and q
</p>
<p>Note that this function does not do the heavy lifting of styling the plot's aesthetics.
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing GWAS p-values for a related trait).
# We use the stratified_qqplot() function to examine the relationship between p and q

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
zq &lt;- c(rnorm(n1p, sd=4), rnorm(n-n1p, sd=1.2))
q &lt;- 2*pnorm(-abs(zq))

df &lt;- data.frame(p, q)

stratified_qqplot(data_frame = df, prin_value_label = "p", cond_value_label = "q")

</code></pre>

<hr>
<h2 id='T1D_application_data'>Data for T1D application</h2><span id='topic+T1D_application_data'></span>

<h3>Description</h3>

<p>A data.frame containing the rsID, chromosome (CHR19) and base pair position (BP19) in hg19,
reference allele (REF), alternative allele (ALLT), type 1 diabetes GWAS p-value (T1D_pval),
minor allele frequency (MAF), LDAK weight (LDAK_weight), rheumatoid arthritis GWAS p-value (RA_pval),
binary regulatory factor binding site overlap (DGF), average H3K27ac fold change value
in T1D-relevant cell types (H3K27ac) for 113,543 SNPs in the T1D GWAS (https://www.nature.com/articles/ng.3245)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T1D_application_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 113543 rows and 11 variables:
</p>


<h3>Details</h3>

<p>Minor allele frequencies estimated from the CEU sub-population samples
in the 1000 Genomes Project Phase 3 data set. Missing values were replaced
by drawing samples from the empirical distribution of MAFs
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
