<!DOCTYPE html><html><head><title>Help for package fcros</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {fcros}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#fcros-package'><p>A Method to Search for Differentially Expressed Genes and to Detect Recurrent</p>
Chromosomal Copy Number Aberrations</a></li>
<li><a href='#bott'><p>Example of sequencing data to test the rankReads function.</p></a></li>
<li><a href='#calcSRmat'><p>Calculation of the sorted rank matrix from the dataset</p></a></li>
<li><a href='#calcSRmatMod'><p>Calculation of the sorted rank matrix from the dataset</p></a></li>
<li><a href='#cghData'><p>Example of aCGH &quot;data file&quot; for the fcros package.</p></a></li>
<li><a href='#cghInfo'><p>Example of aCGH &quot;info file&quot; for the fcros package.</p></a></li>
<li><a href='#chrMerge'><p>Using a C code for merging chromosome segments</p></a></li>
<li><a href='#chrPlot'><p>Plot a chromosome data</p></a></li>
<li><a href='#chrPlot2'><p>Plot a chromosome segmentation results</p></a></li>
<li><a href='#chrSegment'><p>Segmentation of a chromosome data</p></a></li>
<li><a href='#dataSummary'><p>Summarization of the detection results for a list of chromosomes</p></a></li>
<li><a href='#fc2Calc'><p>Calculation of fold change using pairwise comparison values</p></a></li>
<li><a href='#fcros'><p>Search for differentially expressed genes/probes</p></a></li>
<li><a href='#fcros2'><p>Search for differentially expressed genes/probes</p></a></li>
<li><a href='#fcrosFCmat'><p>Calculation of a matrix of fold changes using pairwise comparisons</p></a></li>
<li><a href='#fcrosMod'><p>Search for differentially expressed genes or to detect</p>
recurrent copy number aberration probes</a></li>
<li><a href='#fcrosRead'><p>Read a tab delimited text file</p></a></li>
<li><a href='#fcrosTopN'><p>Search for the top N changed genes or probes</p></a></li>
<li><a href='#fcrosTtest'><p>Student t-test for detecting differentially expressed genes</p></a></li>
<li><a href='#fcrosWrite'><p>Writing the fcros() or pfco() results in a tab delimited text file</p></a></li>
<li><a href='#fdata'><p>Example of dataset for the fcros package.</p></a></li>
<li><a href='#fvalTopN'><p>Search for the top N changed genes or probes using f-values</p></a></li>
<li><a href='#fvalVolcanoPlot'><p>Performs a volcano plot of the FCROS/PFCO statistics</p></a></li>
<li><a href='#histoPlot'><p>Histogram plot function of the fcros package results</p></a></li>
<li><a href='#moyStdCalc'><p>Calculation of the mean and the standard deviation of the full</p>
or reduced matrix of sorted ranks</a></li>
<li><a href='#pfco'><p>Searching for differentially expressed genes/probes using an</p>
approach based on the Perron-Frobenius theorem</a></li>
<li><a href='#pfcoMod'><p>Searching for differentially expressed genes or detecting</p>
recurrent copy number aberration probes using
an approach based on the Perron-Frobenius theorem</a></li>
<li><a href='#pvalTopN'><p>Search for the top N changed genes or probes using p-values</p></a></li>
<li><a href='#pvalVolcanoPlot'><p>Performs a volcano plot of the FCROS/PFCO statistics</p></a></li>
<li><a href='#rankReads'><p>This function computes a score to assess the significance of sequencing values</p></a></li>
<li><a href='#rmatCalc'><p>Calculation of the FC matrix using pairwise comparisons</p></a></li>
<li><a href='#rmatTrim'><p>Calculation of the reduced matrix containing sorted rank values</p></a></li>
<li><a href='#scoreThr'><p>Determine numerically a threshold for the ranking score</p></a></li>
<li><a href='#tcnReads'><p>Performs a total count normalization of reads</p></a></li>
<li><a href='#tprobaCalc'><p>Calculation of the Student one sample test probabilities</p></a></li>
<li><a href='#varBeta'><p>Compute variance of a beta distribution from data</p></a></li>
<li><a href='#voomReads'><p>Transformation of read count values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.6.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-28</td>
</tr>
<tr>
<td>Title:</td>
<td>A Method to Search for Differentially Expressed Genes and to
Detect Recurrent Chromosomal Copy Number Aberrations</td>
</tr>
<tr>
<td>Author:</td>
<td>Doulaye Dembele</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Doulaye Dembele &lt;doulaye@igbmc.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>A fold change rank based method is presented to search for genes with changing
             expression and to detect recurrent chromosomal copy number aberrations. This 
             method may be useful for high-throughput biological data (micro-array, sequencing, ...).
             Probabilities are associated with genes or probes in the data set and there is no
             problem of multiple tests when using this method. For array-based comparative genomic
             hybridization data, segmentation results are obtained by merging the significant
             probes detected.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-31 12:25:38 UTC; doulaye</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-31 12:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='fcros-package'>A Method to Search for Differentially Expressed Genes and to Detect Recurrent
Chromosomal Copy Number Aberrations</h2><span id='topic+fcros-package'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change rank ordering
statistics to search for differentially expressed genes or to
detect recurrent chromosomal copy number aberrations. This package
can be used for two biological conditions high-throughput dataset 
(microarray, RNA-seq, ...), for expression profiling dataset over time
without replicates or for cytogenetics dataset (aCGH, Sequencing).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
        Package: </td><td style="text-align: left;"> fcros</td>
</tr>
<tr>
 <td style="text-align: left;">
        Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
        Version: </td><td style="text-align: left;"> 1.6.1</td>
</tr>
<tr>
 <td style="text-align: left;">
        Date: </td><td style="text-align: left;"> 2019-05-28</td>
</tr>
<tr>
 <td style="text-align: left;">
        License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>

<p>Package fcros has the following functions:
</p>

<table>
<tr>
 <td style="text-align: left;">
         fcros(): </td><td style="text-align: left;"> The function to use with a dataset from two biological condition samples. The</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  dataset should be in a single table. The function fcros() performs a pairwise</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  conparison of samples to obtain a matrix of fold changes. The fold changes</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcros2(): </td><td style="text-align: left;"> The function to use with datasets from two biological biological conditions. The</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  datasets should be in two separate tables as inputs. The</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  function fcros2() performs a pairwise comparison of samples from each table</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> to
                  obtain fold changes. The fold changes are sorted, their rank values are</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                   combined and then used to associate statistic with genes/probes.</td>
</tr>
<tr>
 <td style="text-align: left;">
         pfco(): </td><td style="text-align: left;"> The function to use with a dataset from two biological condition samples. The</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  dataset should be in a single table. The function pfco() performs a pairwise</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  conparison of samples to obtain a matrix of fold changes. The fold changes</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes</td>
</tr>
<tr>
 <td style="text-align: left;">
                  </td><td style="text-align: left;"> using a singular value decomposition.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosMod(): </td><td style="text-align: left;"> This function uses fold changes or ratios matrix as input to associate
                  statistic</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> with genes/probes.</td>
</tr>
<tr>
 <td style="text-align: left;">
         pfcoMod(): </td><td style="text-align: left;"> This function uses fold changes or ratios matrix as input to associate
                  statistic</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> with genes/probes using a singular value decomposition.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosFCmat(): </td><td style="text-align: left;"> This function allows to compute a matrix of fold changes using</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                   pairwise comparisons of the two biological condition samples in a dataset.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosTtest(): </td><td style="text-align: left;"> This function allows to use the Student t-test to calculate p-values</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  for the genes in a dataset.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosRead(): </td><td style="text-align: left;"> This function allows to read a tab delimited text file to be use as an</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> 
                  input for the function fcros(), fcros2() or fcrosMod().</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosWrite(): </td><td style="text-align: left;"> This function allows to save the results obtained using the function
                  fcros(),</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> fcros2() or fcrosMod() in a tab delimited text file.</td>
</tr>
<tr>
 <td style="text-align: left;">
         fcrosTopN(): </td><td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
 <td style="text-align: left;">
         fvalTopN(): </td><td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
 <td style="text-align: left;">
         pvalTopN(): </td><td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
 <td style="text-align: left;">
         histoPlot(): </td><td style="text-align: left;"> This function plots on the screen the histogram of the FCROS statistics
                  </td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> obtained using the results of the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  fcrosMod() or pfcoMod()</td>
</tr>
<tr>
 <td style="text-align: left;">
         fvalVolcanoPlot(): </td><td style="text-align: left;"> This function performs a volcano plot of the results obtained</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  using the function fcros(), fcros2(), pfco(), fcrosMod() or pfcomod()</td>
</tr>
<tr>
 <td style="text-align: left;">
         pvalVolcanoPlot(): </td><td style="text-align: left;"> This function performs a volcano plot of the results obtained</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  using the function fcros(), fcros2(), pfco(), fcrosMod() or pfcoMod()</td>
</tr>
<tr>
 <td style="text-align: left;">
         chrSummary(): </td><td style="text-align: left;"> This function summarizes detection results by chromosome</td>
</tr>
<tr>
 <td style="text-align: left;">
         chrSegment(): </td><td style="text-align: left;"> This function segments a chromosome data</td>
</tr>
<tr>
 <td style="text-align: left;">
         chrPlot(): </td><td style="text-align: left;"> This function performs a plot of the chromosome probes data</td>
</tr>
<tr>
 <td style="text-align: left;">
         chrPlot2(): </td><td style="text-align: left;"> This function performs a plot of the chromosome segmentation results</td>
</tr>
<tr>
 <td style="text-align: left;">
         voomReads(): </td><td style="text-align: left;"> This function performs a transformation of the read counts</td>
</tr>
<tr>
 <td style="text-align: left;">
         tcnReads(): </td><td style="text-align: left;"> This function performs a total count normalization of reads</td>
</tr>
<tr>
 <td style="text-align: left;">
         rankReads(): </td><td style="text-align: left;"> The function to use with a dataset from two biological condition samples. 
                  The</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> dataset should be in a single table. The function rankReads() performs a</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> 
                  pairwise conparison of samples to obtain a matrix of fold changes. Small uniform</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  values are added to read counts. This is repeated nrun time. The fold changes</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes.</td>
</tr>
<tr>
 <td style="text-align: left;">
         scoreThr(): </td><td style="text-align: left;"> Using the log10 transformed score values obtained with the rankReads(),
                  this function</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> computes numerically the inflection point value given lower
                  and upper bound</td>
</tr>
<tr>
 <td style="text-align: left;"></td><td style="text-align: left;"> values for the slope region.
    </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Doulaye Dembele
Maintainer: Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed 
genes, BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462<br />
</p>
<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt, trim.opt);

   # perform Volcano plot
   fvalVolcanoPlot(af, thr = 0.01)

   # save fcros values in a file
   fcrosWrite(af, file = "test2delete_values.txt");

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(c(0,11), ncol = 1);
   id.up &lt;- matrix(c(rep(0,11)), ncol = 1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown+1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup+1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);

   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20),
          ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t, data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20),
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='bott'>Example of sequencing data to test the rankReads function.</h2><span id='topic+bott'></span>

<h3>Description</h3>

<p>This is a subset of data taken from the Bottomly dataset
see http://bowtie-bio.sourceforge.net/recount/. The complete dataset has
36,536 rows or ENSEMBL identifiers (genes) and 21 columns (samples). For the
&quot;bott&quot; data, the first 5,000 rows, first 3 and last 3 samples were used.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bott)</code></pre>


<h3>Format</h3>

<p>A data frame with 5,000 rows and 7 columns.
</p>

<dl>
<dt><code>gene</code>:</dt><dd><p>ENSEMBL ID</p>
</dd>
<dt><code>SRX033480</code>:</dt><dd><p>sequencing values for the first B6 mouse</p>
</dd>
<dt><code>SRX033488</code>:</dt><dd><p>sequencing values for the second B6 mouse</p>
</dd>
<dt><code>SRX033481</code>:</dt><dd><p>sequencing values for the third B6 mouse</p>
</dd>
<dt><code>SRX033493</code>:</dt><dd><p>sequencing values for the first D2 mouse</p>
</dd>
<dt><code>SRX033486</code>:</dt><dd><p>sequencing values for the second D2 mouse</p>
</dd>
<dt><code>SRX033494</code>:</dt><dd><p>sequencing values for the third B6 mouse</p>
</dd>
</dl>



<h3>Details</h3>

<p>&quot;bott&quot; is a subset of a complet dataset obtained using a single 
RNA-seq reads from C57BL/6J (B6) and DBA/J2 (D2) mice.</p>


<h3>References</h3>

<p>Bottomly et al. Evaluating Gene Expression in C57BL/6J and DBA/J2
Mouse Striatum Using RNA-seq and Microarrays, PLoS One, 6(3)e17820, 2011</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(bott)
   summary(bott)
</code></pre>

<hr>
<h2 id='calcSRmat'>Calculation of the sorted rank matrix from the dataset</h2><span id='topic+calcSRmat'></span>

<h3>Description</h3>

<p>This is an internal function used to calculate the sorted
rank matrix. It is used in the functions: fcros() and pfco().</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcSRmat(xdata, cont, test, log2.opt=0, trim.opt=0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcSRmat_+3A_xdata">xdata</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes: <code>xdata</code>.</p>
</td></tr>
<tr><td><code id="calcSRmat_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...).</p>
</td></tr>
<tr><td><code id="calcSRmat_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...).</p>
</td></tr>
<tr><td><code id="calcSRmat_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the matrix &quot;xdata&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="calcSRmat_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene are not
used for computing its statistics &quot;ri&quot;, i.e. the interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kaster P, Fold change rank ordering statistics:
a new method for detecting differentially expressed 
genes, BMC bioinformatics, 2014, 15:14</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='calcSRmatMod'>Calculation of the sorted rank matrix from the dataset</h2><span id='topic+calcSRmatMod'></span>

<h3>Description</h3>

<p>This is an internal function used to calculate the sorted
rank matrix. It is used in the functions: fcrosMod() and pfcoMod().</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcSRmatMod(xdata, samp, log2.opt=0, trim.opt=0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcSRmatMod_+3A_xdata">xdata</code></td>
<td>
<p> A matrix containing fold changes or ratios from a biological
dataset to process for searching differentially expressed
genes or for detecting recurrent copy number aberrations 
regions: <code>fcMat</code>.</p>
</td></tr>
<tr><td><code id="calcSRmatMod_+3A_samp">samp</code></td>
<td>
<p> A vector of sample label names which should appear in the columns
of the matrix fcMat: <code>samp</code>.</p>
</td></tr>
<tr><td><code id="calcSRmatMod_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
values in the matrix &quot;fcMat&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="calcSRmatMod_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values for each gene 
are not used for computing the statistic &quot;ri&quot;, i.e. the 
interquartile range rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kaster P, Fold change rank ordering statistics:
a new method for detecting differentially expressed 
genes, BMC bioinformatics, 2014, 15:14</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='cghData'>Example of aCGH &quot;data file&quot; for the fcros package.</h2><span id='topic+cghData'></span>

<h3>Description</h3>

<p>This dataset is part of the study performed and published 
by Sircoulomb et al. 2010, BMC Cancer, 10:539.
For our illustration, we used the first 10 patients'
data and for only 3 chromosomes: 7, 8 and 9. The complete data are
available from the Gene Expression Omnibus website under the accession 
number GSE17907.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cghData)</code></pre>


<h3>Format</h3>

<p>A data frame with 33,3613 observations for 10 samples.
</p>

<dl>
<dt><code>Probes</code>:</dt><dd><p>a text for the unique probe ID</p>
</dd>
<dt><code>GSM447252</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447253</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447254</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447255</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447256</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447257</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447258</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447259</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447260</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
<dt><code>GSM447261</code>:</dt><dd><p>a numeric vector with log2 change values</p>
</dd>
</dl>



<h3>Details</h3>

<p>This is a subset of a complete dataset obtained using array 
Comparative Genomic Hybridization technology. Agilent 244K design arrays 
have been used to monitor breast cancer patients.</p>


<h3>References</h3>

<p>F Sircoulomb, I Bekhouche, P Finetti, J Adelaide, AB Hamida, 
J Bonansea, S Raynaud, C Innocenti, E Charafe-Jauffret, C Tarpin, FB Ayed,
P Viens, J Jacquemier, F Bertucci, D Birnbaum and M Chaffanet; Genome 
profiling of ERBB2-amplifief breast cancers. BMC Cancer, 2010, 10:539</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(cghData)

   summary(cghData)

   #  perform boxplot of data sample values
   boxplot(cghData[,2:11])
</code></pre>

<hr>
<h2 id='cghInfo'>Example of aCGH &quot;info file&quot; for the fcros package.</h2><span id='topic+cghInfo'></span>

<h3>Description</h3>

<p>This dataset is part of the Agilent 244K design array
probes description file which is used in the publicatin of Sircoulomb et al. 
2010, BMC Cancer, 10:539.
For our illustration, we used only 3 chromosomes (7, 8 and 9) data. The 
complete data are available in the file &quot;GPL9158-3352.txt&quot;, see the Gene 
Expression Omnibus website and accession number GSE17907.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cghInfo)</code></pre>


<h3>Format</h3>

<p>A data frame with 33,3613 rows for 7 columns with information on probes.
</p>

<dl>
<dt><code>Index</code>:</dt><dd><p>a numeric used for the probe</p>
</dd>
<dt><code>ProbeName</code>:</dt><dd><p>a text for the unique probe ID</p>
</dd>
<dt><code>GeneSymbol</code>:</dt><dd><p>a text with the gene symbol associated with the probe</p>
</dd>
<dt><code>Chromosome</code>:</dt><dd><p>a text with the chromosome index associated with the probe</p>
</dd>
<dt><code>Start</code>:</dt><dd><p>a start position value for the sequence associated with the probe</p>
</dd>
<dt><code>End</code>:</dt><dd><p>an end position value for the sequence associated with the probe</p>
</dd>
<dt><code>Cytoband</code>:</dt><dd><p>a text for the cytoband associated with the probe</p>
</dd>
</dl>



<h3>Details</h3>

<p>This is a part of information obtained from the Agilent 244K design array
probes description file.</p>


<h3>References</h3>

<p>F Sircoulomb, I Bekhouche, P Finetti, J Adelaide, AB Hamida,
J Bonansea, S Raynaud, C Innocenti, E Charafe-Jauffret, C Tarpin, FB Ayed,
P Viens, J Jacquemier, F Bertucci, D Birnbaum and M Chaffanet; Genome 
profiling of ERBB2-amplifief breast cancers. BMC Cancer, 2010, 10:539</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(cghInfo)

   summary(cghInfo)
</code></pre>

<hr>
<h2 id='chrMerge'>Using a C code for merging chromosome segments</h2><span id='topic+chrMerge'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code while merging
chromosome segments in the segmentation step.</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrMerge(nbSeg, idStart, idEnd, lBound, uBound, segVal, segProba,
                         fcall, L2R, nd, dm, sigma)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrMerge_+3A_nbseg">nbSeg</code></td>
<td>
<p> Number of current segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_idstart">idStart</code></td>
<td>
<p> Position indexes of the first probes for segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_idend">idEnd</code></td>
<td>
<p> Positions indexes of the last probes for segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_lbound">lBound</code></td>
<td>
<p> Lower bound position for segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_ubound">uBound</code></td>
<td>
<p> Upper position for segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_segval">segVal</code></td>
<td>
<p> Change values associated with segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_segproba">segProba</code></td>
<td>
<p> Probabilities associated with segments</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_fcall">fcall</code></td>
<td>
<p> Detection status associated with probes</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_l2r">L2R</code></td>
<td>
<p> Change values associated with probes</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_nd">nd</code></td>
<td>
<p> Number of acceptable non-detection between two significant of a segment</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_dm">dm</code></td>
<td>
<p> Average distance between two consecutive probes of the chromosome</p>
</td></tr>
<tr><td><code id="chrMerge_+3A_sigma">sigma</code></td>
<td>
<p> Standard deviation of the residual observations, see reference</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='chrPlot'>Plot a chromosome data</h2><span id='topic+chrPlot'></span>

<h3>Description</h3>

<p>This function generates a picture using a chromosome data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrPlot(chrData, thr = 0.05, deb = 100, fin = 1e10)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrPlot_+3A_chrdata">chrData</code></td>
<td>
<p> A chromosome data obtained from an output of the function
dataSummary(): <code>xinfo2</code> = dataSummary(af, xinfo, chromosomes, alpha)<br />
<code>idx</code> = which(xinfo2$xinfo.s$Chromosome == &quot;chr1&quot;)<br />
<code>chrData</code> = xinfo2$xinfo.s[idx, ]</p>
</td></tr>
<tr><td><code id="chrPlot_+3A_thr">thr</code></td>
<td>
<p> The probability threshold leading to the selection of the significant probes:
<code>thr</code> = 0.05</p>
</td></tr>
<tr><td><code id="chrPlot_+3A_deb">deb</code></td>
<td>
<p> This parameter allows to specify the start position of the chromosome region
for plotting. It can be used for zooming. Negative value will lead to the plot
of all chromosome data. <code>deb</code> = 100</p>
</td></tr>
<tr><td><code id="chrPlot_+3A_fin">fin</code></td>
<td>
<p> This parameter allows to specify the end position of the chromosome region for
plot. It can be used for zooming. Negative value will lead to the plot of all
chromosome data. <code>thr</code> = 1e7</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function generates a picture on the screen</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # load CGH data and info files
    data(cghData)
    rownames(cghData) &lt;- cghData[,1];
    data(cghInfo)
    noms &lt;- colnames(cghData)
    m &lt;- length(noms)
    samp  &lt;- noms[2:m]

    # associate statistics with probes
    af &lt;- fcrosMod(cghData, samp, log2.opt = 0, trim.opt = 0.25)

    chromosomes = c(7:9)
    alpha &lt;- 0.05

    # summarize results for each chromosome
    xinfo2 &lt;- dataSummary(af, cghInfo, chromosomes, alpha)

    # focused on chromosome 7 data
    idx &lt;- which(xinfo2$xinfo.s$Chromosome == "7")
    chrData &lt;- xinfo2$xinfo.s[idx, ]

    # Plot chromosome 7 data
    chrPlot(chrData, thr = alpha)
</code></pre>

<hr>
<h2 id='chrPlot2'>Plot a chromosome segmentation results</h2><span id='topic+chrPlot2'></span>

<h3>Description</h3>

<p>This function generates a picture. It uses a chromosome data and the 
output results of the segmentation function chrSegment().</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrPlot2(chrData, chrSeg, deb = 100, fin = 1e10)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrPlot2_+3A_chrdata">chrData</code></td>
<td>
<p> A chromosome data obtained from an output of the function
dataSummary(): <code>xinfo2</code> = dataSummary(af, xinfo, chromosomes, alpha)<br />
<code>idx</code> = which(xinfo2$xinfo.s$Chromosome == &quot;chr1&quot;)<br />
<code>chrData</code> = xinfo2$xinfo.s[idx, ]</p>
</td></tr>
<tr><td><code id="chrPlot2_+3A_chrseg">chrSeg</code></td>
<td>
<p> An output object of the function chrSegment():
<code>chrSeg</code> = chrSegment(chrData, nd = 10)</p>
</td></tr>
<tr><td><code id="chrPlot2_+3A_deb">deb</code></td>
<td>
<p> This parameter allows to specify the start position of the chromosome region 
for plotting. It can be used for zooming. Negative value will lead to the plot
of all chromosome data. <code>deb</code> = 100</p>
</td></tr>
<tr><td><code id="chrPlot2_+3A_fin">fin</code></td>
<td>
<p> This parameter allows to specify the end position of the chromosome region for 
plot. It can be used for zooming. Negative value will lead to the plot of all
chromosome data. <code>thr</code> = 1e7</p>
</td></tr>
</table>


<h3>Value</h3>

<p>  This function generates a picture on the screen</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # load CGH data and info files
    data(cghData)
    rownames(cghData) &lt;- cghData[,1];
    data(cghInfo)
    noms &lt;- colnames(cghData)
    m &lt;- length(noms)
    samp  &lt;- noms[2:m]

    # associate statistics with probes in the dataset
    af &lt;- pfcoMod(cghData, samp, log2.opt = 0, trim.opt = 0.25)

    chromosomes &lt;- c(7:9)
    alpha &lt;- 0.05

    # summarize results for each chromosome
    xinfo2 &lt;- dataSummary(af, cghInfo, chromosomes, alpha)

    # focused on chromosome 7 data
    idx &lt;- which(xinfo2$xinfo.s$Chromosome == "7")
    chrData &lt;- xinfo2$xinfo.s[idx, ]

    # segment chromosome 7 data
    chrSeg &lt;- chrSegment(chrData, nd = 15)

    # plot chromosome 7 results
    op &lt;- par(mfrow = c(2,1))
    chrPlot(chrData, thr = alpha, deb =-1, fin = 3.5e7)
    chrPlot2(chrData, chrSeg, -1, fin = 3.5e7)
    par(op)
</code></pre>

<hr>
<h2 id='chrSegment'>Segmentation of a chromosome data</h2><span id='topic+chrSegment'></span>

<h3>Description</h3>

<p>This function allows to segment a chromosome data</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrSegment(chrData, nd = 10)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrSegment_+3A_chrdata">chrData</code></td>
<td>
<p> A chromosome data obtained from an output of the function
dataSummary(): <code>xinfo2</code> = dataSummary(af, xinfo, chromosomes, alpha)<br />
<code>idx</code> = which(xinfo2$xinfo.s$Chromosome == &quot;chr1&quot;)<br />
<code>chrData</code> = xinfo2$xinfo.s[idx, ]</p>
</td></tr>
<tr><td><code id="chrSegment_+3A_nd">nd</code></td>
<td>
<p> The acceptable number of non-detected probes which can separate two
significant probes in a segment. Default setting value is 10: <code>nd</code> = 10</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data frame containing 6 information for each segment
</p>
<table>
<tr><td><code>idStart</code></td>
<td>
<p> The start position indexes associated with segments</p>
</td></tr>
<tr><td><code>idEnd</code></td>
<td>
<p> The End position indexes associated with segments</p>
</td></tr>
<tr><td><code>lBounds</code></td>
<td>
<p> The lower bound positions associated with segments</p>
</td></tr>
<tr><td><code>uBounds</code></td>
<td>
<p> The upper bound positions associated with segments</p>
</td></tr>
<tr><td><code>segL2R</code></td>
<td>
<p> The change values associated with segments</p>
</td></tr>
<tr><td><code>segProba</code></td>
<td>
<p> The probabilities associated with segments</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # load CGH data and info files
    data(cghData)
    rownames(cghData) &lt;- cghData[,1]
    data(cghInfo)
    noms &lt;- colnames(cghData)
    m &lt;- length(noms)
    samp  &lt;- noms[2:m]

    # associate statistics with probes in the dataset
    af &lt;- pfcoMod(cghData, samp, log2.opt = 0, trim.opt = 0.25)

    chromosomes = c(7:9)
    alpha &lt;- 0.05

    # summarize results for each chromosome
    xinfo2 &lt;- dataSummary(af, cghInfo, chromosomes, alpha)

    # focused on chromosome 7 data
    idx &lt;- which(xinfo2$xinfo.s$Chromosome == "7")
    chrData &lt;- xinfo2$xinfo.s[idx, ]

    # segment chromosome 7 data
    chrSeg &lt;- chrSegment(chrData, nd = 15)

    # show first 10 segment results
    chrSeg[1:10,]
</code></pre>

<hr>
<h2 id='dataSummary'>Summarization of the detection results for a list of chromosomes</h2><span id='topic+dataSummary'></span>

<h3>Description</h3>

<p> From an outpout object of the function fcrosMod() or pfcoMod(),
the chromosomes information object, the list of chromosomes and a threshold, 
this function creates two objects containing ordered chromosome data 
and summary results.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataSummary(af, xinfo, chromosomes = c(1:22,"X","Y"), alpha = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dataSummary_+3A_af">af</code></td>
<td>
<p> An output object of the function fcrosMod() or pfcoMod():<br />
<code>af</code> = fcrosMod(xdata, samp, log2.opt, trim.opt)<br />
<code>af</code> = pfcoMod(xdata, samp, log2.opt, trim.opt)</p>
</td></tr>
<tr><td><code id="dataSummary_+3A_xinfo">xinfo</code></td>
<td>
<p> A data frame containing chromosomes information (probe name,
gene symbol, chromosome index, start position, end position
and the cytoband). These information should appear with the labels
ProbeName, GeneSymbol, Chromosome, Start, End, Cytoband. Additional
information may be used. Only labels Chromosome, Start and End are 
mandotory.</p>
</td></tr>
<tr><td><code id="dataSummary_+3A_chromosomes">chromosomes</code></td>
<td>
<p> A list of chromosomes. Default setting is a list with all
chromosomes:<br /> 
<code>chromosomes</code> = (1:22,&quot;X&quot;,&quot;Y&quot;)</p>
</td></tr>
<tr><td><code id="dataSummary_+3A_alpha">alpha</code></td>
<td>
<p> A threshold allowing to select significant probes based on probabilities.
Default setting is 0.05 (5% of error) <code>thr</code> = 0.05</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>    # load CGH data and info files
    data(cghData)
    rownames(cghData) &lt;- cghData[,1]
    data(cghInfo)
    noms &lt;- colnames(cghData)
    m &lt;- length(noms)
    samp  &lt;- noms[2:m]

    # associate statistics with probes in the dataset
    af &lt;- pfcoMod(cghData, samp, log2.opt = 0, trim.opt = 0.25)

    chromosomes = c(7:9)
    alpha &lt;- 0.05

    # summarize results for each chromosome
    xinfo2 &lt;- dataSummary(af, cghInfo, chromosomes, alpha)

    # display the number of significant probes for each chromosome
    xinfo2$chrSumm
</code></pre>

<hr>
<h2 id='fc2Calc'>Calculation of fold change using pairwise comparison values</h2><span id='topic+fc2Calc'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code to
calculate fold changes using pairwise comparison of samples.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fc2Calc(rvect, n, m, idx, m2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fc2Calc_+3A_rvect">rvect</code></td>
<td>
<p> Vector containing the full or reduced matrix with the pairwise
comparison of samples results</p>
</td></tr>
<tr><td><code id="fc2Calc_+3A_n">n</code></td>
<td>
<p> Number of genes or probes in the dataset</p>
</td></tr>
<tr><td><code id="fc2Calc_+3A_m">m</code></td>
<td>
<p> Number of columns of the full or reduced matrix of pairwise 
comparison of samples results</p>
</td></tr>
<tr><td><code id="fc2Calc_+3A_idx">idx</code></td>
<td>
<p> Indexes of the columns to keep in the pairwise comparison
of samples</p>
</td></tr>
<tr><td><code id="fc2Calc_+3A_m2">m2</code></td>
<td>
<p> Number of columns in the full or reduced matrix of 
comparison of samples</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='fcros'>Search for differentially expressed genes/probes</h2><span id='topic+fcros'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change rank ordering
statistics for detecting differentially expressed genes in a
dataset.
This function should be used with two biological conditions
dataset (microarray or RNA-seq, ...).
Using pairwise combinations of samples from the
two biological conditions, fold changes (FC) are calculated.
For each combination, the FC obtained are sorted in increasing
order and corresponding rank values are associated with genes. 
Then, a statistic is assigned to the robust average ordered rank
values for each gene/probe.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcros(xdata, cont, test, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcros_+3A_xdata">xdata</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes. The rownames of xdata are used for the output idnames.</p>
</td></tr>
<tr><td><code id="fcros_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...).</p>
</td></tr>
<tr><td><code id="fcros_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...).</p>
</td></tr>
<tr><td><code id="fcros_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the matrix &quot;xdata&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="fcros_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene are not
used for computing its statistics &quot;ri&quot;, i.e. the interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Label names appearing in the parameters &quot;cont&quot; and &quot;test&quot; should match
with some label names in the columns of the data matrix &quot;xdata&quot;. It is not 
necessary to use all label names appearing in the columns of the dataset matrix.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 9 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of rank values associated with genes.
These values are rank values statistics leading to f-values
and p-values.</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for genes in the dataset. These fold changes are
calculated as a ratio of averages from the test and the control
samples. Non log scale values are used in the calculation.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples.  Non log scale values are used 
in the calculation.</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes using
the &quot;mean&quot; and the &quot;standard deviation&quot; (&quot;sd&quot;) of the statistics &quot;ri&quot;.
The &quot;mean&quot; and &quot;sd&quot; are used as a normal distribution parameters.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. These values are obtained
from the fold change rank values and one sample t-test.</p>
</td></tr>
<tr><td><code>bounds</code></td>
<td>
<p>Two values, which are the lower and the upper bounds or
the minimum and the maximum values of the non standardized &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>Three values, which are the estimates for the parameters &quot;delta&quot;
(average difference between consecutive ordered average of rank values)
&quot;mean&quot; (mean value of &quot;ri&quot;) and the standard deviation (&quot;sd&quot;)
of &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params_t</code></td>
<td>
<p>Three values which are theoretical levels for 
parameters &quot;delta&quot;, &quot;mean&quot; and &quot;sd&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed 
genes, BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];
   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0, 1);
   id.up &lt;- matrix(0, 1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);


   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20), 
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20), 
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='fcros2'>Search for differentially expressed genes/probes</h2><span id='topic+fcros2'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change or ratio rank
ordering statistics for detecting differentially expressed 
genes. This function should be used with dataset in two separate
tables and from two biological conditions datasets (microarray,
RNA-seq, ...). Pairwise combinations of samples from
the two biological conditions are used to obtain a matrix of
fold changes. For each combination, the FCs obtained are
sorted in an increasing order and the corresponding rank values are
associated with genes/probes. 
Then, a statistic is associated with each gene/probe.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcros2(xdata1, xdata2, cont, test, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcros2_+3A_xdata1">xdata1</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes. The rownames of xdata1 are used for the output idnames.</p>
</td></tr>
<tr><td><code id="fcros2_+3A_xdata2">xdata2</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process fordetecting differentially expressed
genes: <code>xdata2</code>. </p>
</td></tr>
<tr><td><code id="fcros2_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcros2_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcros2_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the tables &quot;xdata1&quot; and &quot;xdata2&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="fcros2_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and  the upper rank values for a gene
are not used for computing  the statistics &quot;ri&quot;, i.e. the 
interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The label names appearing in the parameters &quot;cont&quot; and &quot;test&quot; should 
match some label names in the columns of the data matrices &quot;xdata1&quot; and &quot;xdata2&quot;. 
It is not necessary to use all column label names appearing in matrices &quot;xdata1&quot; 
and &quot;xdata2&quot;. However, it is assumed that the same genes (same IDs or symbol) are 
used in xdata1 and xdata2.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 9 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of the IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of rank values associated with genes in the
datasets. These values are rank statistics leading to f-values
and p-values.</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for genes. These fold changes are
calculated as a ratio of averages from the test and control
samples. Non log scale values are used in the calculation.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples using the pairwise comparisons.
Non log scale values are used in the calculation.</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes. These 
values are obtained using the &quot;mean&quot; and the
&quot;standard deviation&quot; (&quot;sd&quot;) of the statistics &quot;ri&quot;. The &quot;mean&quot; and
&quot;sd&quot; are used as a normal distribution parameters.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. These values are obtained
from the f-values.</p>
</td></tr>
<tr><td><code>bounds</code></td>
<td>
<p>Two values which are the lower and the upper bound or the
minimum and the maximum values of non standardized &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>Three values, which are the estimates for the parameters &quot;delta&quot;
(average difference between consecutive ordered average of rank) 
&quot;mean&quot; (mean value of variable &quot;ri&quot;) and the standard deviation (&quot;sd&quot;)
of variable &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params_t</code></td>
<td>
<p>Three values, which are the theoretical levels for
parameters &quot;delta&quot;, &quot;mean&quot; and &quot;sd&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes, 
BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont02", "cont03", "cont04", "cont05",
             "cont06", "cont07", "cont08", "cont09", "cont10");
   test &lt;- c("test01", "test02", "test03", "test04", "test05",
             "test06", "test07", "test08", "test09", "test10");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcros2()
   xdata1 &lt;- fdata[,c(2:5, 12:17)];
   xdata2 &lt;- fdata[,c(6:11, 18:21)];
   rownames(xdata1) &lt;- fdata[,1];
   rownames(xdata2) &lt;- fdata[,1];

   af2 &lt;- fcros2(xdata1, xdata2, cont, test, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af2, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0,1);
   id.up &lt;- matrix(0,1);
   n &lt;- length(af2$FC);
   f.value &lt;- af2$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);

   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20),
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20),
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='fcrosFCmat'>Calculation of a matrix of fold changes using pairwise comparisons</h2><span id='topic+fcrosFCmat'></span>

<h3>Description</h3>

<p>This function is used internally by fcros2() to compute a matrix 
of fold changes using pairwise comparisons of a two biological conditions dataset.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosFCmat(xdata, cont, test, log2.opt=0, trim.opt=0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosFCmat_+3A_xdata">xdata</code></td>
<td>
<p> A table containing a two biological conditions
dataset to process for obtaining a matrix of fold changes
that results from pairwise comparisons of samples.
The rownames of xdata are used for the output idnames.</p>
</td></tr>
<tr><td><code id="fcrosFCmat_+3A_cont">cont</code></td>
<td>
<p> A vector containing label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcrosFCmat_+3A_test">test</code></td>
<td>
<p> A vector containing label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcrosFCmat_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
the data values in matrix &quot;xdata&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="fcrosFCmat_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene
are not used for computing the statistic &quot;ri&quot; or &quot;u1&quot;, i.e. the
interquartile rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The label names appearing in the parameters &quot;cont&quot; and &quot;test&quot; should match
some label names of the columns in the table &quot;xdata&quot;. The dataset &quot;xdata&quot; can
contain other label names not used.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 3 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>fcMat</code></td>
<td>
<p>A matrix of fold changes associated with genes. This matrix is
obtained using pairwise comparisons of samples
in the dataset &quot;xdata&quot;.</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for genes in the dataset &quot;xdata&quot;. These
fold changes are calculated as a ratio of averages from the
test and the control samples. Non log scale values are used
in this calculation.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes in the dataset &quot;xdata&quot;. These
fold changes are calculated as a trimmed mean of the fold
changes obtained using pairs of samples. Non log scale values
are used in this calculation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed 
genes, BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcrosFCmat()
   fc &lt;- fcrosFCmat(fdata, cont, test, log2.opt);

   # plot histogram of the fold change (log2 scale) in dataset
   hist(log2(fc$FC),nclass = 50);
</code></pre>

<hr>
<h2 id='fcrosMod'>Search for differentially expressed genes or to detect
recurrent copy number aberration probes</h2><span id='topic+fcrosMod'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change rank ordering
statistics to search for differentially expressed genes or to 
detect chromosomal recurrent copy number aberration probes. This
function should be used with a matrix of fold changes or ratios
from biological dataset (microarray, RNA-seq, ...). The function
fcrosMod() is an extention of the function fcros() to a
dataset which does not contain replicate samples or to a
dataset with one biological condition dataset. Statistics
are associated with genes/probes to characterize their change levels.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosMod(fcMat, samp, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosMod_+3A_fcmat">fcMat</code></td>
<td>
<p> A matrix containing fold changes or ratios from a biological
dataset to process for searching differentially expressed
genes or for detecting recurrent copy number aberrations 
regions. The rownames of fcMat are used for the output idnames.</p>
</td></tr>
<tr><td><code id="fcrosMod_+3A_samp">samp</code></td>
<td>
<p> A vector of sample label names which should appear in the columns
of the matrix fcMat: <code>samp</code>.</p>
</td></tr>
<tr><td><code id="fcrosMod_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
values in the matrix &quot;fcMat&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="fcrosMod_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values for each gene 
are not used for computing the statistic &quot;ri&quot;, i.e. the 
interquartile range rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The label names appearing in the parameter &quot;samp&quot; should
match some label names of the columns in the data matrix &quot;xdata&quot;. It is not 
necessary to use all label names appearing in the columns of the dataset matrix.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 8 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of ordered rank values associated with genes in the
dataset. These values are rank statistics leading to the f-values
and the p-values.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes in matrix &quot;fcMat&quot;. These fold
changes are calculated as a trimed mean of the values in
&quot;fcMat&quot;. Non log scale values are used in this calculation.</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes using
the &quot;mean&quot; and the &quot;standard deviation&quot; (&quot;sd&quot;) of values in &quot;ri&quot;. 
The &quot;mean&quot; and &quot;sd&quot; are used as a normal distribution parameters.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. The p-values are obtained
using a one sample t-test on the fold change rank values.</p>
</td></tr>
<tr><td><code>bounds</code></td>
<td>
<p>Two values, which are the lower and the upper bounds or
the minimum and the maximum values of the non standardized &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>Three values, which are the estimates for the parameters &quot;delta&quot;
(average difference between consecutive ordered average of 
rank) &quot;mean&quot; (mean value of the &quot;ri&quot;) and the standard deviation
(&quot;sd&quot;) of the &quot;ri&quot;.</p>
</td></tr>
<tr><td><code>params_t</code></td>
<td>
<p>Three values, which are theoretical levels for
the parameters &quot;delta&quot;, &quot;mean&quot; and &quot;sd&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcrosMod()
   fc &lt;- fcrosFCmat(fdata, cont, test, log2.opt, trim.opt);
   m &lt;- ncol(fc$fcMat)
   samp &lt;- paste("Col",as.character(1:m), sep = "");
   fc.val &lt;- cbind(data.frame(fc$fcMat))
   colnames(fc.val) &lt;- samp
   rownames(fc.val) &lt;- fdata[,1]

   af &lt;- fcrosMod(fc.val, samp, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0,1);
   id.up &lt;- matrix(0,1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);


   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1, 2:21], type = "l", col = "blue", xlim = c(1,20),
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i, 2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20), 
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='fcrosRead'>Read a tab delimited text file</h2><span id='topic+fcrosRead'></span>

<h3>Description</h3>

<p>This function can be used to read a tab delimited text
file to be used as input for the functions fcros(), fcros2(), fcrosMod()
pfco() or pfcoMod().
You may also use other functions (read.csv(),
read.csv2(), read.delim() or read.delim2()) to read your dataset.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosRead(filename)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosRead_+3A_filename">filename</code></td>
<td>
<p> &quot;filename&quot; is a tab delimited text file to read.
The first line of this file should contain labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output is a matrix or a table containing data.</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele, doulaye@igbmc.fr</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # generate random dataset of size 100 x 5
  xdata &lt;- matrix(c(rep(0,600)), ncol = 6);

  xdata[,2:6] &lt;- matrix(rnorm(500,0,1), ncol = 5);

  x0 &lt;- matrix(NA, 100, 0);

  for (i in 1:100) {
      x0[i] &lt;- paste("Obs_",i, sep = "");
  }

  # set column names
  colnames(xdata) &lt;- c("Index", "Col_1", "Col_2", "Col_3", "Col_4", "Col_5");

  # save data in a text file
  write.table(xdata, file = "test2delete.txt", quote = FALSE, sep = "\t",
         eol = "\n", col.names = TRUE, row.names = FALSE);

  # now used fcros.read() to open the file
  mydata &lt;- fcrosRead(file = "test2delete.txt");

  summary(mydata)
</code></pre>

<hr>
<h2 id='fcrosTopN'>Search for the top N changed genes or probes</h2><span id='topic+fcrosTopN'></span>

<h3>Description</h3>

<p>This function allows to seach for the top N 
differentially expressed genes or changed probes.
It uses the output results obtained using one of the following functions
fcros(), fcros2(), fcrosMod(), pfco() or pfcoMod().</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosTopN(af, topN)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosTopN_+3A_af">af</code></td>
<td>
<p> This is an output object obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod().</p>
</td></tr>
<tr><td><code id="fcrosTopN_+3A_topn">topN</code></td>
<td>
<p> The expected number of the top DE genes in the dataset used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data frame containing 2 components
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>Two threshold values for the down- and the up-regulated allowing
to have the top N DE genes</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>The indexes of the top N DE genes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # now select top 10 down and/or up regulated genes
   top10 &lt;- fcrosTopN(af, 10);

   # display thresholds
   top10$alpha
   
   # display index of top10 genes
   fdata[top10$index, 1]
   
   # display fvalue of the top10 genes
   (af$f.value)[top10$index]
</code></pre>

<hr>
<h2 id='fcrosTtest'>Student t-test for detecting differentially expressed genes</h2><span id='topic+fcrosTtest'></span>

<h3>Description</h3>

<p>The function uses the basic R t.test() function to perform 
the Student t-test. It should be
used for two biological conditions dataset
(microarray, or RNA-seq). The Fold changes, statistics and p-values
are returned for each gene in the dataset.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosTtest(xdata, cont, test, log2.opt = 0)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosTtest_+3A_xdata">xdata</code></td>
<td>
<p> A table containing a two biological conditions
dataset to process for detecting differentially expressed genes.
The rownames of xdata are used for the output idnames.</p>
</td></tr>
<tr><td><code id="fcrosTtest_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcrosTtest_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...)</p>
</td></tr>
<tr><td><code id="fcrosTtest_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the matrix &quot;xdata&quot; are expressed in a log2 scale: <code>log2.opt</code> = 0</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Label names appearing in the parameters &quot;cont&quot; and &quot;test&quot; should match
column label names of the data matrix &quot;xdata&quot;. It is not necessary to use all
column label names of the dataset &quot;xdata&quot;.</p>


<h3>Value</h3>

     
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for the genes in the dataset.</p>
</td></tr>
<tr><td><code>stat</code></td>
<td>
<p>The Student t-test statistics associated with genes.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The Student t-test p-values associated with genes.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcrosTtest()
   at &lt;- fcrosTtest(fdata, cont, test, log2.opt);
   
   # now select some differentially expressed genes
   id.de  &lt;- matrix(0, 1);
   n &lt;- length(at$FC);
   for (i in 1:n) {
       if ((at$p.value)[i] &lt;= 0.0005) { id.de &lt;- rbind(id.de, i); }
   }

   data.de &lt;- fdata[id.de, ];
   nde &lt;- nrow(data.de);

   # now plot the DE genes
   t &lt;- 1:20;
   plot(t, data.de[1, 2:21], type = "l", col = "blue", xlim = c(1,20),
        ylim = c(0,18), main = "Down- and up-regulated genes");
   for (i in 2:nde) {
       lines(t, data.de[i,2:21], type = "l", col = "blue")
   }
</code></pre>

<hr>
<h2 id='fcrosWrite'>Writing the fcros() or pfco() results in a tab delimited text file</h2><span id='topic+fcrosWrite'></span>

<h3>Description</h3>

<p>This function creates a tab-delimited text file with the
results of the function fcros(), fcros2(), fcrosMod(), pfco() or pfcoMod().
The results are the values associated with the parameters
&quot;idnames&quot;, &quot;ri&quot;, &quot;FC&quot;, &quot;FC2&quot;, &quot;f.value&quot; and &quot;p.value&quot;.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fcrosWrite(af, file = "fcrosResults.txt", thr = 1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fcrosWrite_+3A_af">af</code></td>
<td>
<p> An output object of the functions fcros(), fcros2(), fcrosMod(), 
pfco()<br />
or pfcoMod():
<code>af = fcros(xdata, cont, test, log2.opt, trim.opt)</code><br />
<code>af = pfco(xdata, cont, test, log2.opt, trim.opt)</code></p>
</td></tr>
<tr><td><code id="fcrosWrite_+3A_file">file</code></td>
<td>
<p> The output file name:
<code>file = "fcrosResults.txt"</code></p>
</td></tr>
<tr><td><code id="fcrosWrite_+3A_thr">thr</code></td>
<td>
<p> A threshold allowing to filter data based on p-values. Default
setting is 1 (no filtering) <code>thr = 1</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function creates and saves a tab-delimited text file on the disk.</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(fdata);

    rownames(fdata) &lt;- fdata[,1];
    cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
    test &lt;- c("test01", "test02", "test08", "test09", "test05");
    log2.opt &lt;- 0;
    trim.opt &lt;- 0.25;
    af &lt;- fcros(fdata, cont, test, log2.opt, trim.opt);
    af2 &lt;- pfco(fdata, cont, test, log2.opt, trim.opt);

    fcrosWrite(af, file = "test2delete_values.txt");
    fcrosWrite(af2, file = "test2delete2_values.txt");
</code></pre>

<hr>
<h2 id='fdata'>Example of dataset for the fcros package.</h2><span id='topic+fdata'></span>

<h3>Description</h3>

<p>This is a synthetic dataset generated using the &quot;madsim&quot; R package 
(available from the CRAN web site).
Two biological conditions data matrix using 10,000 genes and 20
samples (10 control and 10 test samples) were simulated.
108 and 102 genes of this
dataset are down- and up-regulated, respectively.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fdata)</code></pre>


<h3>Format</h3>

<p>A data frame with 10,000 observations on the following 22 variables.
</p>

<dl>
<dt><code>index</code>:</dt><dd><p>a text for the genes, unique ID</p>
</dd>
<dt><code>cont01</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 01</p>
</dd>
<dt><code>cont02</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 02</p>
</dd>
<dt><code>cont03</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 03</p>
</dd>
<dt><code>cont04</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 04</p>
</dd>
<dt><code>cont05</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 05</p>
</dd>
<dt><code>cont06</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 06</p>
</dd>
<dt><code>cont07</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 07</p>
</dd>
<dt><code>cont08</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 08</p>
</dd>
<dt><code>cont09</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 09</p>
</dd>
<dt><code>cont10</code>:</dt><dd><p>a numeric vector with log2 intensities for control sample 10</p>
</dd>
<dt><code>test01</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 01</p>
</dd>
<dt><code>test02</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 02</p>
</dd>
<dt><code>test03</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 03</p>
</dd>
<dt><code>test04</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 04</p>
</dd>
<dt><code>test05</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 05</p>
</dd>
<dt><code>test06</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 06</p>
</dd>
<dt><code>test07</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 07</p>
</dd>
<dt><code>test08</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 08</p>
</dd>
<dt><code>test09</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 09</p>
</dd>
<dt><code>test10</code>:</dt><dd><p>a numeric vector with log2 intensities for test sample 10</p>
</dd>
<dt><code>DE_status</code>:</dt><dd><p>a numeric vector with values -1, 0 and 1. Value 0 is used for
no change genes, while -1 and 1 are used for down- and up-regulated genes, respectively.</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset is obtained using the microarray data simulation model
implemented in the package &quot;madsim&quot;. A real microarray data, &quot;madsim_test&quot;, was
used as seed and the number of the control and the test samples were set to 10.
The parameter &quot;sdn&quot; was set to 0.3 and all the other parameters in the madsim package
were set to their default settings.</p>


<h3>References</h3>

<p>Dembele D, A flexible microarray data simulation model.
Microarrays, 2013, v.2, n.2, pp.115-130</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata)

   rownames(fdata) &lt;- fdata[,1];

   op &lt;- par(mfrow = c(2,1));

   #  perform MA plot using samples "cont01" and "cont05"
   A1 &lt;- 0.5*(fdata$cont01 + fdata$cont05);
   M1 &lt;- fdata$cont05 - fdata$cont01;

   plot(A1, M1, col="red", xlim=c(2,18), ylim=c(-5,5),
            main="MA plot with two control samples");
   lines(x = c(2,18), y = c(0,0), col = "blue")
   lines(x = c(2,18), y = c(-1,-1), col = "blue")
   lines(x = c(2,18), y = c(1,1), col = "blue")

   #  perform MA plot using samples "cont01" and "test05"
   A2 &lt;- 0.5*(fdata$cont01 + fdata$test05);
   M2 &lt;- fdata$test05 - fdata$cont01;

   plot(A2, M2, col="red", xlim=c(2,18), ylim=c(-5,5),
            main="MA plot with one control and one test samples");
   lines(x = c(2,18), y = c(0,0), col = "blue")
   lines(x = c(2,18), y = c(-1,-1), col = "blue")
   lines(x = c(2,18), y = c(1,1), col = "blue")
   par(op)
</code></pre>

<hr>
<h2 id='fvalTopN'>Search for the top N changed genes or probes using f-values</h2><span id='topic+fvalTopN'></span>

<h3>Description</h3>

<p>This function allows to seach for the top N 
differentially expressed genes or changed probes.
It uses the f-values obtained using one of the following functions
fcros(), fcros2(), fcrosMod(), pfco() or pfcoMod().</p>


<h3>Usage</h3>

<pre><code class='language-R'>fvalTopN(fval, topN)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fvalTopN_+3A_fval">fval</code></td>
<td>
<p> This is a f-values vector obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod(): <code>fval = af$f.value</code></p>
</td></tr>
<tr><td><code id="fvalTopN_+3A_topn">topN</code></td>
<td>
<p> The expected number of the top DE genes/probes in the dataset used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data frame containing 2 components
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>Two threshold values for the down- and the up-regulated allowing
to have the top N DE genes</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>The indexes of the top N DE genes / probes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform pfco()
   af &lt;- pfco(fdata, cont, test, log2.opt);
   
   # now select top 10 down and/or up regulated genes
   top10 &lt;- fvalTopN(af$f.value, 10);

   # display thresholds
   top10$alpha
   
   # display index of top10 genes
   fdata[top10$index, 1]
   
   # display fvalue of the top10 genes
   (af$f.value)[top10$index]
</code></pre>

<hr>
<h2 id='fvalVolcanoPlot'>Performs a volcano plot of the FCROS/PFCO statistics</h2><span id='topic+fvalVolcanoPlot'></span>

<h3>Description</h3>

<p>This function allows to have a volcano like plot using the output 
results of the function fcros(), fcros2(), fcrosMod(), pfco() or pfcoMod():
p-values versus robust fold changes (FC2). 
The p-value are transformed using -log10(), while FC2 are transformed using log2().</p>


<h3>Usage</h3>

<pre><code class='language-R'>fvalVolcanoPlot(af, thr = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fvalVolcanoPlot_+3A_af">af</code></td>
<td>
<p> This is an object obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod():
<code>af = fcros(xdata, cont, test)</code></p>
</td></tr>
<tr><td><code id="fvalVolcanoPlot_+3A_thr">thr</code></td>
<td>
<p> The threshold to obtain the DE genes in the dataset (red plots):
<code>thr = 0.05</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function displays on the screen a volcano like plot using the f-values
and the robust fold changes (FC2).</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # Volcano plot
   fvalVolcanoPlot(af, thr = 0.01);
</code></pre>

<hr>
<h2 id='histoPlot'>Histogram plot function of the fcros package results</h2><span id='topic+histoPlot'></span>

<h3>Description</h3>

<p>This function allows to have a histogram plot. It uses the 
statistics &quot;ri&quot; or &quot;u1&quot; obtained using one of the following functions: fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod().</p>


<h3>Usage</h3>

<pre><code class='language-R'>histoPlot(af, nbins = 50)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="histoPlot_+3A_af">af</code></td>
<td>
<p> This is an object obtained using the function fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod()</p>
</td></tr>
<tr><td><code id="histoPlot_+3A_nbins">nbins</code></td>
<td>
<p> This parameter is used for the number of bins in
the histogram. Default setting is 50: <code>nbins = 50</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function plots a histogram on the screen.</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics:
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros() and pfco()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   af2 &lt;- pfco(fdata, cont, test, log2.opt);

   # Histogram plots
   op &lt;- par(mfrow = c(1,2))
      histoPlot(af);
      histoPlot(af2);
   par(op);
</code></pre>

<hr>
<h2 id='moyStdCalc'>Calculation of the mean and the standard deviation of the full
or reduced matrix of sorted ranks</h2><span id='topic+moyStdCalc'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code in the
calculation of the mean and the standard deviation of the full or reduced 
matrix with sorted rank values. The calculations are performed for each row.</p>


<h3>Usage</h3>

<pre><code class='language-R'>moyStdCalc(rvect, n, m)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moyStdCalc_+3A_rvect">rvect</code></td>
<td>
<p> Vector containing the full or reduced matrix with sorted rank values</p>
</td></tr>
<tr><td><code id="moyStdCalc_+3A_n">n</code></td>
<td>
<p> Number of genes or probes in the dataset</p>
</td></tr>
<tr><td><code id="moyStdCalc_+3A_m">m</code></td>
<td>
<p> Number of columns of the full or reduced matrix of sorted rank values</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='pfco'>Searching for differentially expressed genes/probes using an
approach based on the Perron-Frobenius theorem</h2><span id='topic+pfco'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change and the Perron
theorem for detecting differentially expressed genes in a
dataset.
This function should be used with two biological conditions
dataset (microarray or RNA-seq, ...).
Using pairwise combinations of samples from the
two biological conditions, fold changes (FC) are calculated.
For each combination, the FC obtained are sorted in increasing
order and corresponding rank values are associated with genes.
Then, a statistic is assigned to the robust average ordered rank
values for each gene/probe.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pfco(xdata, cont, test, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pfco_+3A_xdata">xdata</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes. The rownames of xdata are used for the output idnames.</p>
</td></tr>
<tr><td><code id="pfco_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...).</p>
</td></tr>
<tr><td><code id="pfco_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...).</p>
</td></tr>
<tr><td><code id="pfco_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the matrix &quot;xdata&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="pfco_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene are not
used for computing its statistics &quot;ri&quot;, i.e. the interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Label names appearing in the parameter &quot;samp&quot; should match
with some label names in the columns of the data matrix &quot;xdata&quot;. It is not 
necessary to use all label names appearing in the columns of the dataset matrix.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 9 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of rank values associated with genes.
These values are rank values statistics leading to f-values
and p-values.</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for genes in the dataset. These fold changes are
calculated as a ratio of averages from the test and the control
samples. Non log scale values are used in the calculation.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples.  Non log scale values are used 
in the calculation.</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes using
the &quot;mean&quot; and the &quot;standard deviation&quot; (&quot;sd&quot;) of the statistics &quot;ri&quot;.
The &quot;mean&quot; and &quot;sd&quot; are used as a normal distribution parameters.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. These values are obtained
using a one sample Student t-test on the fold change rank values.</p>
</td></tr>
<tr><td><code>comp</code></td>
<td>
<p>Singular values.</p>
</td></tr>
<tr><td><code>comp.w</code></td>
<td>
<p>Singular values weights.</p>
</td></tr>
<tr><td><code>comp.wcum</code></td>
<td>
<p>Cumulative sum of the singular values weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform pfco()
   af &lt;- pfco(fdata, cont, test, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0, 1);
   id.up &lt;- matrix(0, 1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);


   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20), 
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20), 
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='pfcoMod'>Searching for differentially expressed genes or detecting
recurrent copy number aberration probes using
an approach based on the Perron-Frobenius theorem</h2><span id='topic+pfcoMod'></span>

<h3>Description</h3>

<p>Implementation of a method based on fold change rank and the Perron 
theorem to search for differentially expressed genes or to
detect chromosomal recurrent copy number aberration probes. This
function should be used with a matrix of fold changes or ratios
from biological dataset (microarray, RNA-seq, ...). The function
pfcoMod() is an extention of the function pfco() to a
dataset which does not contain replicate samples or to a
dataset with one biological condition dataset. Statistics
are associated with genes/probes to characterize their change levels.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pfcoMod(fcMat, samp, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pfcoMod_+3A_fcmat">fcMat</code></td>
<td>
<p> A matrix containing fold changes or ratios from a biological
dataset to process for searching differentially expressed
genes or for detecting recurrent copy number aberrations 
regions. The rownames of fcMat are used for the output idnames.</p>
</td></tr>
<tr><td><code id="pfcoMod_+3A_samp">samp</code></td>
<td>
<p> A vector of sample label names which should appear in the columns
of the matrix fcMat: <code>samp</code>.</p>
</td></tr>
<tr><td><code id="pfcoMod_+3A_log2.opt">log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
values in the matrix &quot;fcMat&quot; are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td></tr>
<tr><td><code id="pfcoMod_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values for each gene 
are not used for computing the statistic &quot;ri&quot;, i.e. the 
interquartile range rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The label names appearing in the parameter &quot;samp&quot; should
match some label names of the columns in the data matrix &quot;xdata&quot;. It is not 
necessary to use all label names appearing in the columns of the dataset matrix.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 8 components
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of ordered rank values associated with genes in the
dataset. These values are rank statistics leading to the f-values
and the p-values.</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes in matrix &quot;fcMat&quot;. These fold
changes are calculated as a trimed mean of the values in
&quot;fcMat&quot;. Non log scale values are used in this calculation.</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes using
the &quot;mean&quot; and the &quot;standard deviation&quot; (&quot;sd&quot;) of values in &quot;ri&quot;. 
The &quot;mean&quot; and &quot;sd&quot; are used as a normal distribution parameters.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. The p-values are obtained
from the fold change ranks using a one sample t-test.</p>
</td></tr>
<tr><td><code>comp</code></td>
<td>
<p>Singular values.</p>
</td></tr>
<tr><td><code>comp.w</code></td>
<td>
<p>Singular values weights.</p>
</td></tr>
<tr><td><code>comp.wcum</code></td>
<td>
<p>Cumulative sum of the singular values weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform pfcoMod()
   fc &lt;- fcrosFCmat(fdata, cont, test, log2.opt, trim.opt);
   m &lt;- ncol(fc$fcMat)
   samp &lt;- paste("Col",as.character(1:m), sep = "");
   fc.val &lt;- cbind(data.frame(fc$fcMat))
   colnames(fc.val) &lt;- samp
   rownames(fc.val) &lt;- fdata[,1]

   af &lt;- pfcoMod(fc.val, samp, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0,1);
   id.up &lt;- matrix(0,1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);


   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1, 2:21], type = "l", col = "blue", xlim = c(1,20),
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i, 2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20), 
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>

<hr>
<h2 id='pvalTopN'>Search for the top N changed genes or probes using p-values</h2><span id='topic+pvalTopN'></span>

<h3>Description</h3>

<p>This function allows to seach for the top N 
differentially expressed genes or changed probes.
It uses the p-value. These p-values can be obtained using one of
the following functions fcros(), fcros2(), fcrosMod(), pfco(),
pfcoMod() or another statistical method.</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalTopN(pval, topN)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalTopN_+3A_pval">pval</code></td>
<td>
<p> This is vector with p-values obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco(), pfcoMod() or another statistical method:
<code>pval = af$p.value</code></p>
</td></tr>
<tr><td><code id="pvalTopN_+3A_topn">topN</code></td>
<td>
<p> The expected number of the top DE genes/probes in the dataset used:
<code>topN</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data frame containing 2 components
</p>
<table>
<tr><td><code>alpha</code></td>
<td>
<p>Two threshold values for the down- and the up-regulated allowing
to have the top N DE genes</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>The indexes of the top N DE genes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1]
   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # now select top 10 down and/or up regulated genes
   top10 &lt;- pvalTopN(af$p.value, 12);

   # display thresholds
   top10$alpha
   
   # display index of top10 genes
   fdata[top10$index, 1]
   
   # display fvalue of the top10 genes
   (af$f.value)[top10$index]
</code></pre>

<hr>
<h2 id='pvalVolcanoPlot'>Performs a volcano plot of the FCROS/PFCO statistics</h2><span id='topic+pvalVolcanoPlot'></span>

<h3>Description</h3>

<p>This function allows to have a volcano like plot using the output 
results of the function fcros(), fcros2(), fcrosMod(), pfco() or pfcomod():
p-values versus robust fold changes (FC2). 
The p-value are transformed using -log10(), while FC2 are transformed using log2().</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalVolcanoPlot(af, thr = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalVolcanoPlot_+3A_af">af</code></td>
<td>
<p> This is an object obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod():
<code>af = fcros(xdata, cont, test)</code></p>
</td></tr>
<tr><td><code id="pvalVolcanoPlot_+3A_thr">thr</code></td>
<td>
<p> The threshold to obtain the DE genes in the dataset (red plots):
<code>thr = 0.05</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function displays on the screen a volcano like plot using the p-values
and the robust fold changes (FC2).</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(fdata);

   rownames(fdata) &lt;- fdata[,1]
   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # Volcano plot
   pvalVolcanoPlot(af, thr = 1e-6);
</code></pre>

<hr>
<h2 id='rankReads'>This function computes a score to assess the significance of sequencing values</h2><span id='topic+rankReads'></span>

<h3>Description</h3>

<p>Implementation of two methods based (1) on the coefficient of variation
or (2) on the fold change rank ordering statistics for detecting genes
with significant sequencing values (gwssv). A score is obtained for each
gene and a threshold allows to select the number of gwssv.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rankReads(xdata, cont, test, meth=0, Ttimes=10, err=0.1, trim.opt=0,
                        rseed=60)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rankReads_+3A_xdata">xdata</code></td>
<td>
<p> A matrix or a table containing sequencing dataset.
The rownames of xdata is used for the output idnames.</p>
</td></tr>
<tr><td><code id="rankReads_+3A_cont">cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c(&quot;cont01&quot;, &quot;cont02&quot;, ...).</p>
</td></tr>
<tr><td><code id="rankReads_+3A_test">test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c(&quot;test01&quot;, &quot;test02&quot;, &quot;test03&quot;, ...).</p>
</td></tr>
<tr><td><code id="rankReads_+3A_meth">meth</code></td>
<td>
<p> This parameter allows to specify the approach to use. The value
0 (defaul) means the coefficient of variation is used. When
non-zero value is given, the fcros method used: <code>meth</code> = 0</p>
</td></tr>
<tr><td><code id="rankReads_+3A_ttimes">Ttimes</code></td>
<td>
<p> The number of perturbed data to use.
The value 10 (default) means that the dataset is used 20 times
and small uniform values are added at each time:
<code>Ttimes</code> = 10</p>
</td></tr>
<tr><td><code id="rankReads_+3A_err">err</code></td>
<td>
<p> This is the amount of the uniform values to add to count values.
The value 0.1 (default) is used:
<code>err</code> = 0.1</p>
</td></tr>
<tr><td><code id="rankReads_+3A_trim.opt">trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene are not
used for computing its statistics &quot;ri&quot;, i.e. the inter-quartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td></tr>
<tr><td><code id="rankReads_+3A_rseed">rseed</code></td>
<td>
<p> This value allow to set the computer random generation seed
value in order to be able to have the same results for runs
performed at different times: <code>rseed</code> = 58</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Label names appearing in the parameters &quot;cont&quot; and &quot;test&quot; should match
with some label names in the columns of the data matrix &quot;xdata&quot;. It is not
necessary to use all label names appearing in the columns of the dataset matrix.
For a general purpose dataset, one of these parametere can be empty.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 10 components when meth=1
and 3 components when meth=0
</p>
<table>
<tr><td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td></tr>
<tr><td><code>score</code></td>
<td>
<p>coefficient of variation (meth=0) or Fisher-Snedecor test p-value
(meth=1). Smaller (higher) values are associated with genes with significant
(non significant) sequencing values.</p>
</td></tr>
<tr><td><code>moy</code></td>
<td>
<p>trimmed means associated with genes (when meth=0).</p>
</td></tr>
<tr><td><code>ri</code></td>
<td>
<p>The average of rank values associated with genes when meth=1.
These values are rank values statistics leading to f-values
and p-values (when meth=1).</p>
</td></tr>
<tr><td><code>FC</code></td>
<td>
<p>The fold changes for genes in the dataset. These fold changes are
calculated as a ratio of averages from the test and the control
samples. Non log scale values are used in the calculation (when meth=1).</p>
</td></tr>
<tr><td><code>FC2</code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples.  Non log scale values are used
in the calculation (when meth=1).</p>
</td></tr>
<tr><td><code>f.value</code></td>
<td>
<p>The f-values are probabilities associated with genes using
the &quot;mean&quot; and the &quot;standard deviation&quot; (&quot;sd&quot;) of the statistics &quot;ri&quot;.
The &quot;mean&quot; and &quot;sd&quot; are used as a normal distribution parameters 
(when meth=1).</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>The p-values associated with genes. These values are obtained
from the fold change rank values and one sample t-test (when meth=1).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, manuscript under preparation</p>


<h3>Examples</h3>

<pre><code class='language-R'>   data(bott);
   cont &lt;- c("SRX033480", "SRX033488", "SRX033481");
   test &lt;- c("SRX033493", "SRX033486", "SRX033494");
   n &lt;- nrow(bott);

   x2 &lt;- tcnReads(bott[,c(cont,test)])
   idx.ok &lt;- (apply(x2, 1, sum) != 0)
   xdata &lt;- x2[,c(cont,test)]
   rownames(xdata) &lt;- bott[,1]
   idx.ok &lt;- (apply(x2, 1, sum) != 0)
   tt2 &lt;- sum(idx.ok)

   raf10.cv &lt;- rankReads(xdata, cont, test, meth=0)
   raf10.pv &lt;- rankReads(xdata, cont, test, meth=1)
   score.cv &lt;- -log10(sort(raf10.cv$score))
   score.pv &lt;- -log10(sort(raf10.pv$score))
   tmp &lt;- scoreThr(score.cv, 2500, 3500)
   tmp

   tmp &lt;- scoreThr(score.pv, 2500, 3500)
   tmp

   op &lt;- par(mfrow = c(1,2))
   plot(score.cv, xlab = "index of genes",
      ylab = "-log10(sorted(score)", main = "rs.cv", type = "l",
      col = "blue", panel.first = grid())
   plot(score.pv, xlab = "index of genes",
      ylab = "-log10(sorted(score)", main = "rs.pv", type = "l",
      col = "blue", panel.first = grid())
   par(op)
</code></pre>

<hr>
<h2 id='rmatCalc'>Calculation of the FC matrix using pairwise comparisons</h2><span id='topic+rmatCalc'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code to calculate a
vector form of the matrix of fold changes using pairwise comparison of data samples.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmatCalc(fvect, n, m1, m2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmatCalc_+3A_fvect">fvect</code></td>
<td>
<p> Two biological conditions dataset matrix</p>
</td></tr>
<tr><td><code id="rmatCalc_+3A_n">n</code></td>
<td>
<p> Number of genes or probes in the dataset</p>
</td></tr>
<tr><td><code id="rmatCalc_+3A_m1">m1</code></td>
<td>
<p> Number of samples in the first biological condition</p>
</td></tr>
<tr><td><code id="rmatCalc_+3A_m2">m2</code></td>
<td>
<p> Number of samples in the second biological condition</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kaster P, Fold change rank ordering statistics:
a new method for detecting differentially expressed 
genes, BMC bioinformatics, 2014, 15:14<br />
</p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='rmatTrim'>Calculation of the reduced matrix containing sorted rank values</h2><span id='topic+rmatTrim'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code for
calculating the reduced matrix of sorted rank values.</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmatTrim(rvect, n, m, idx, m2)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rmatTrim_+3A_rvect">rvect</code></td>
<td>
<p> Vector containing the full matrix with sorted rank values</p>
</td></tr>
<tr><td><code id="rmatTrim_+3A_n">n</code></td>
<td>
<p> Number of genes or probes in the dataset</p>
</td></tr>
<tr><td><code id="rmatTrim_+3A_m">m</code></td>
<td>
<p> Number of columns of the full matrix of sorted rank values</p>
</td></tr>
<tr><td><code id="rmatTrim_+3A_idx">idx</code></td>
<td>
<p> Indexes of the columns to keep in the reduced matrix</p>
</td></tr>
<tr><td><code id="rmatTrim_+3A_m2">m2</code></td>
<td>
<p> Number of columns of the reduced matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='scoreThr'>Determine numerically a threshold for the ranking score</h2><span id='topic+scoreThr'></span>

<h3>Description</h3>

<p>Given a log 10 transformed of the sorted ranking score values
(dscore), this function determines numerically a threshold (inflection
point) for significance of sequencing value level. User should specify the 
lower and the upper bounds of the slope region containing the inflection point.</p>


<h3>Usage</h3>

<pre><code class='language-R'>scoreThr(dscore, deb, fin)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scoreThr_+3A_dscore">dscore</code></td>
<td>
<p> transformed score values obtained using the function
rankReads()</p>
</td></tr>
<tr><td><code id="scoreThr_+3A_deb">deb</code></td>
<td>
<p> integer for lower bound of the slope region containing the
inflection point. User should perfom a plot to determine this
value</p>
</td></tr>
<tr><td><code id="scoreThr_+3A_fin">fin</code></td>
<td>
<p> integer for upper bound of the slope region containing the
inflection point. User should perfom a plot to determine this
value</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns two values: &ldquo;pos&rdquo; and &ldquo;thr&rdquo; which are the
index or the number of genes with significant values and the score
(-log10(score)) threshold
</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, manuscript under preparation</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='tcnReads'>Performs a total count normalization of reads</h2><span id='topic+tcnReads'></span>

<h3>Description</h3>

<p>Given a data table x with count reads, one column for each
sample, this function adjust values in such a way they become comparable
between samples. User can specify the maximum value for total count reads
to be used for each sample.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tcnReads(x, maxVal=0)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcnReads_+3A_x">x</code></td>
<td>
<p> Data table with count reads, one column for one sample</p>
</td></tr>
<tr><td><code id="tcnReads_+3A_maxval">maxVal</code></td>
<td>
<p> Target total number reads for each sample. The value 0 
(default) means that the median total of samples is
used: <code>maxVal</code> = 0</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data table of the same size as input
</p>
<table>
<tr><td><code>x2</code></td>
<td>
<p>Data table of the same size as x</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018<br />
</p>
<p>Dembele D, manuscript under preparation</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='tprobaCalc'>Calculation of the Student one sample test probabilities</h2><span id='topic+tprobaCalc'></span>

<h3>Description</h3>

<p>This is an internal function for using a C code to
perform a Student one sample test for each row of the full or
reduced matrix with sorted rank values.</p>


<h3>Usage</h3>

<pre><code class='language-R'>tprobaCalc(moy, std, n, dl, em)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tprobaCalc_+3A_moy">moy</code></td>
<td>
<p> Vector containing average of rank values for rows</p>
</td></tr>
<tr><td><code id="tprobaCalc_+3A_std">std</code></td>
<td>
<p> Vector containing standard deviation of rank values for rows</p>
</td></tr>
<tr><td><code id="tprobaCalc_+3A_n">n</code></td>
<td>
<p> Number of genes or probes in the dataset</p>
</td></tr>
<tr><td><code id="tprobaCalc_+3A_dl">dl</code></td>
<td>
<p> Degree of freedom in the test. This is equal to the number of 
the columns in the full or reduced matrix of sorted rank values 
minus one</p>
</td></tr>
<tr><td><code id="tprobaCalc_+3A_em">em</code></td>
<td>
<p> Expected average rank values for each row</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='varBeta'>Compute variance of a beta distribution from data</h2><span id='topic+varBeta'></span>

<h3>Description</h3>

<p>This is an internal function. 
Given a vector with values between 0 and 1 and assumed to
come from a beta distribution, this function return the variance of 
the distribution. A trim parameter allow to have a robust value</p>


<h3>Usage</h3>

<pre><code class='language-R'>varBeta(x, trim.opt)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varBeta_+3A_x">x</code></td>
<td>
<p> vector with components between 0 and 1</p>
</td></tr>
<tr><td><code id="varBeta_+3A_trim.opt">trim.opt</code></td>
<td>
<p> value between 0 and 0.5 used as trim parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns one value equalt to the variance of the best
beta distribution of entry x
</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, manuscript under preparation</p>


<h3>Examples</h3>

<pre><code class='language-R'>#    data(fdata);
</code></pre>

<hr>
<h2 id='voomReads'>Transformation of read count values</h2><span id='topic+voomReads'></span>

<h3>Description</h3>

<p>This function allows to transform the count values
associated with Sequencing reads. The purpose of this transformation
is to allow applying normal-based microarray-like statistical methods
to RNA-seq read counts. Log2 values are returned.</p>


<h3>Usage</h3>

<pre><code class='language-R'>voomReads(x, Rm=1e+06)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="voomReads_+3A_x">x</code></td>
<td>
<p> This is a read counts matrix</p>
</td></tr>
<tr><td><code id="voomReads_+3A_rm">Rm</code></td>
<td>
<p> A constant used in the transformation</p>
</td></tr>
</table>


<h3>Value</h3>

<p> This function returns a data matrix of the same size as input 
matrix x. The values of this matrix are expressed in log2 scale.
</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Charity W Law, Yunshun Chen, Wei Shi and Gordon K Smyth,
voom: precision weights unlock linear model analysis tools
for RNA-seq read counts,Genome Biology, 2014, 15R29</p>


<h3>Examples</h3>

<pre><code class='language-R'>#   data(fdata);
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
